<!DOCTYPE html><html><head><title>Help for package FDRsamplesize2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FDRsamplesize2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FDRsamplesize2-package'><p>FDRsamplesize2: Computing Power and Sample Size for the False Discovery Rate in Multiple Applications</p></a></li>
<li><a href='#alpha.power.fdr'><p>Compute p-value threshold for given the proportion pi0 of tests with a true null</p></a></li>
<li><a href='#average.power.coxph'><p>Compute the average power of many Cox regression models</p></a></li>
<li><a href='#average.power.fisher'><p>Compute average power of many Fisher's exact tests</p></a></li>
<li><a href='#average.power.hart'><p>Compute average power for RNA-seq experiments assuming Negative Binomial distribution</p></a></li>
<li><a href='#average.power.li'><p>Compute average power for RNA-Seq experiments assuming Poisson distribution</p></a></li>
<li><a href='#average.power.oneway'><p>Compute average power of many one-way ANOVA tests</p></a></li>
<li><a href='#average.power.ranksum'><p>Compute average power of rank-sum tests</p></a></li>
<li><a href='#average.power.signrank'><p>Compute average power of many signed-rank tests</p></a></li>
<li><a href='#average.power.signtest'><p>Compute average power of many sign tests</p></a></li>
<li><a href='#average.power.t.test'><p>Compute average power of many t-tests</p></a></li>
<li><a href='#average.power.tcorr'><p>Compute average power of many t-tests for non-zero correlation</p></a></li>
<li><a href='#average.power.twoprop'><p>Computer average power of many two proportion z-tests</p></a></li>
<li><a href='#fdr.avepow'><p>Compute FDR and average power for a given sample size and effect size vector</p></a></li>
<li><a href='#fdr.power.alpha'><p>Compute FDR for given p-value threshold, average power and proportion of tests with a true null</p></a></li>
<li><a href='#find.sample.size'><p>Determines the sample size needed to achieve the desired FDR and average power</p></a></li>
<li><a href='#n.fdr.coxph'><p>Sample size calculation for the Cox proportional hazards regression model</p></a></li>
<li><a href='#n.fdr.fisher'><p>Sample size calculation for Fisher's Exact tests</p></a></li>
<li><a href='#n.fdr.negbin'><p>Sample size calculation for Negative Binomial data</p></a></li>
<li><a href='#n.fdr.oneway'><p>Sample size calculation for one-way ANOVA</p></a></li>
<li><a href='#n.fdr.poisson'><p>Sample size calculation for Poisson data</p></a></li>
<li><a href='#n.fdr.ranksum'><p>Sample size calculation for rank-sum tests</p></a></li>
<li><a href='#n.fdr.signrank'><p>Sample size calculation for signed-rank tests</p></a></li>
<li><a href='#n.fdr.signtest'><p>Sample size calculation for sign tests</p></a></li>
<li><a href='#n.fdr.tcorr'><p>Sample size calculation for t-tests for non-zero correlation</p></a></li>
<li><a href='#n.fdr.ttest'><p>Sample size calculation for t-tests</p></a></li>
<li><a href='#n.fdr.twoprop'><p>Sample size calculation for comparing two proportions</p></a></li>
<li><a href='#power.cox'><p>Compute the power of a single-predictor Cox regression model</p></a></li>
<li><a href='#power.fisher'><p>Compute power for Fisher's exact test</p></a></li>
<li><a href='#power.hart'><p>Compute power for RNA-seq experiments assuming Negative Binomial distribution</p></a></li>
<li><a href='#power.li'><p>Compute power for RNA-Seq experiments assuming Poisson distribution</p></a></li>
<li><a href='#power.oneway'><p>Compute power of one-way ANOVA</p></a></li>
<li><a href='#power.ranksum'><p>Compute power of the rank-sum test</p></a></li>
<li><a href='#power.signrank'><p>Compute power of the signed-rank test</p></a></li>
<li><a href='#power.signtest'><p>Compute power of the sign test</p></a></li>
<li><a href='#power.tcorr'><p>Compute power of the t-test for non-zero correlation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Computing Power and Sample Size for the False Discovery Rate in
Multiple Applications</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yonghui Ni &lt;Yonghui.Ni@STJUDE.ORG&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Defines a collection of functions to compute average power and sample size for studies that use the false discovery rate as the final measure of statistical significance. A three-rectangle approximation method of a p-value histogram is proposed to derive a formula to compute the statistical power for analyses that involve the FDR. The methodology paper of this package is under review.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-26 22:25:36 UTC; yni54</td>
</tr>
<tr>
<td>Author:</td>
<td>Yonghui Ni [aut, cre],
  Stanley Pounds [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-27 03:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='FDRsamplesize2-package'>FDRsamplesize2: Computing Power and Sample Size for the False Discovery Rate in Multiple Applications</h2><span id='topic+FDRsamplesize2'></span><span id='topic+FDRsamplesize2-package'></span>

<h3>Description</h3>

<p>Defines a collection of functions to compute average power and sample size for studies that use the false discovery rate as the final measure of statistical significance. A three-rectangle approximation method of a p-value histogram is proposed to derive a formula to compute the statistical power for analyses that involve the FDR. The methodology paper of this package is under review.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Yonghui Ni <a href="mailto:Yonghui.Ni@STJUDE.ORG">Yonghui.Ni@STJUDE.ORG</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Stanley Pounds <a href="mailto:Stanley.Pounds@STJUDE.ORG">Stanley.Pounds@STJUDE.ORG</a>
</p>
</li></ul>


<hr>
<h2 id='alpha.power.fdr'>Compute p-value threshold for given the proportion pi0 of tests with a true null</h2><span id='topic+alpha.power.fdr'></span>

<h3>Description</h3>

<p>Given the proportion pi0 of tests with a true null, find the p-value threshold that results in a desired FDR and average power.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha.power.fdr(fdr, pwr, pi0, method = "HH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alpha.power.fdr_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="alpha.power.fdr_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="alpha.power.fdr_+3A_pi0">pi0</code></td>
<td>
<p>the proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code id="alpha.power.fdr_+3A_method">method</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height) , &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To get the fixed p-value threshold for multiple testing procedure, 4 approximation methods are provided, they are Benjamini &amp; Hochberg procedure (1995), Jung's formula (2005),
method of using p-value histogram height (HH) and method of using p-value histogram mean (HM). For last two methods' details, see Ni Y, Onar-Thomas A, Pounds S.  &quot;Computing Power and Sample Size for the False Discovery Rate in Multiple Applications&quot;
</p>


<h3>Value</h3>

<p>The fixed p-value threshold for multiple testing procedure
</p>


<h3>References</h3>

<p>Pounds S and Cheng C, &quot;Sample size determination for the false discovery rate.&quot; Bioinformatics 21.23 (2005): 4263-4271.
</p>
<p>Gadbury GL, et al. (2004) Power and sample size estimation in high dimensional biology. Statistical Methods in Medical Research 13(4):325-38.
</p>
<p>Jung,Sin-Ho.&quot;Sample size for FDR-control in microarray data analysis.&quot; Bioinformatics 21.14 (2005): 3097-3104.
</p>
<p>Ni Y, Seffernick A, Onar-Thomas A, Pounds S.  &quot;Computing Power and Sample Size for the False Discovery Rate in Multiple Applications&quot;, Manuscript.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alpha.power.fdr(fdr = 0.1, pwr = 0.9, pi0=0.9, method = "HH")
</code></pre>

<hr>
<h2 id='average.power.coxph'>Compute the average power of many Cox regression models</h2><span id='topic+average.power.coxph'></span>

<h3>Description</h3>

<p>Compute the average power of many Cox regression models for a given number of events, p-value threshold, vector of effect sizes (log hazard ratio),and variance of predictor variables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power.coxph(n, alpha, logHR, v)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power.coxph_+3A_n">n</code></td>
<td>
<p>number of events (scalar)</p>
</td></tr>
<tr><td><code id="average.power.coxph_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="average.power.coxph_+3A_loghr">logHR</code></td>
<td>
<p>log hazard ratio (vector)</p>
</td></tr>
<tr><td><code id="average.power.coxph_+3A_v">v</code></td>
<td>
<p>variance of predictor variable (vector)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>References</h3>

<p>Hsieh, FY and Lavori, Philip W (2000) Sample-size calculations for the Cox proportional hazards regression model with non-binary covariates. Controlled Clinical Trials 21(6):552-560.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+power.cox">power.cox</a></code> for more details about power calculation of single-predictor Cox regression model. The power calculation is based on asymptotic normal approximation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>logHR = log(rep(c(1, 2),c(900, 100)));
v = rep(1, 1000);
average.power.coxph(n = 50, alpha = 0.05, logHR = logHR, v = v)
</code></pre>

<hr>
<h2 id='average.power.fisher'>Compute average power of many Fisher's exact tests</h2><span id='topic+average.power.fisher'></span>

<h3>Description</h3>

<p>Compute average power of many Fisher's exact tests
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power.fisher(p1, p2, n, alpha, alternative)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power.fisher_+3A_p1">p1</code></td>
<td>
<p>probability in one group (vector)</p>
</td></tr>
<tr><td><code id="average.power.fisher_+3A_p2">p2</code></td>
<td>
<p>probability in other group (vector)</p>
</td></tr>
<tr><td><code id="average.power.fisher_+3A_n">n</code></td>
<td>
<p>per-group sample size</p>
</td></tr>
<tr><td><code id="average.power.fisher_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold</p>
</td></tr>
<tr><td><code id="average.power.fisher_+3A_alternative">alternative</code></td>
<td>
<p>one- or two-sided test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>See Also</h3>

<p><code><a href="#topic+power.fisher">power.fisher</a></code> for more details about power calculation of Fisher's exact test
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234);
p1 = sample(seq(0,0.5,0.1),5,replace = TRUE);
p2 = sample(seq(0.5,1,0.1),5,replace = TRUE);
average.power.fisher(p1 = p1,p2 = p2,n = 20,alpha = 0.05,alternative = "two.sided")
</code></pre>

<hr>
<h2 id='average.power.hart'>Compute average power for RNA-seq experiments assuming Negative Binomial distribution</h2><span id='topic+average.power.hart'></span>

<h3>Description</h3>

<p>Compute average power for RNA-seq experiments assuming Negative Binomial distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power.hart(n, alpha, log.fc, mu, sig)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power.hart_+3A_n">n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td></tr>
<tr><td><code id="average.power.hart_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="average.power.hart_+3A_log.fc">log.fc</code></td>
<td>
<p>log fold-change (vector), usual null hypothesis is log.fc=0</p>
</td></tr>
<tr><td><code id="average.power.hart_+3A_mu">mu</code></td>
<td>
<p>read depth per gene (vector, same length as log.fc)</p>
</td></tr>
<tr><td><code id="average.power.hart_+3A_sig">sig</code></td>
<td>
<p>coefficient of variation (CV) per gene (vector, same length as log.fc)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The power function is based on equation (1) of Hart et al (2013). It assumes a Negative Binomial model for RNA-seq read counts and equal sample size per group.
</p>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>References</h3>

<p>SN Hart, TM Therneau, Y Zhang, GA Poland, and J-P Kocher (2013). Calculating Sample Size Estimates for RNA Sequencing Data. Journal of Computational Biology 20: 970-978.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+power.hart">power.hart</a></code> for more details about power calculation of data under Negative Binomial distribution. The power calculation is based on asymptotic normal approximation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>logFC = log(rep(c(1,2),c(900,100)));
mu = rep(5,1000);
sig = rep(0.6,1000);
average.power.hart(n = 50, alpha = 0.05,log.fc = logFC, mu = mu, sig = sig)
</code></pre>

<hr>
<h2 id='average.power.li'>Compute average power for RNA-Seq experiments assuming Poisson distribution</h2><span id='topic+average.power.li'></span>

<h3>Description</h3>

<p>Use the formula of Li et al (2013) to compute power for comparing RNA-seq expression across two groups assuming the Poisson distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power.li(n, alpha, rho, mu0, w, type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power.li_+3A_n">n</code></td>
<td>
<p>per-group sample size</p>
</td></tr>
<tr><td><code id="average.power.li_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="average.power.li_+3A_rho">rho</code></td>
<td>
<p>fold-change, usual null hypothesis is that rho=1 (vector)</p>
</td></tr>
<tr><td><code id="average.power.li_+3A_mu0">mu0</code></td>
<td>
<p>average count in control group (vector)</p>
</td></tr>
<tr><td><code id="average.power.li_+3A_w">w</code></td>
<td>
<p>ratio of the total number of reads mapped between the two groups (scalar or vector)</p>
</td></tr>
<tr><td><code id="average.power.li_+3A_type">type</code></td>
<td>
<p>type of test: &quot;w&quot; for Wald, &quot;s&quot; for score, &quot;lw&quot; for log-transformed Wald, &quot;ls&quot; for log-transformed score</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the average power for a series of two-sided tests defined by the input parameters. The power is based on the sample size formulas in equations (10-13) of Li et al (2013). Also, note that the null.effect is set to 1 in the examples because the usual null hypothesis is that the fold-change = 1.
</p>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>References</h3>

<p>C-I Li, P-F Su, Y Guo, and Y Shyr (2013). Sample size calculation for differential expression analysis of RNA-seq data under Poisson distribution. Int J Comput Biol Drug Des 6(4).&lt;doi:10.1504/IJCBDD.2013.056830&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+power.li">power.li</a></code> for more details about power calculation of data under Poisson distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rho = rep(c(1,1.25),c(900,100));
mu0 = rep(5,1000);
w = rep(0.5,1000);
average.power.li(n = 50, alpha = 0.05, rho = rho, mu0 = mu0, w = w, type = "w")
</code></pre>

<hr>
<h2 id='average.power.oneway'>Compute average power of many one-way ANOVA tests</h2><span id='topic+average.power.oneway'></span>

<h3>Description</h3>

<p>Compute average power of many one-way ANOVA tests
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power.oneway(n, alpha, theta, k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power.oneway_+3A_n">n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td></tr>
<tr><td><code id="average.power.oneway_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="average.power.oneway_+3A_theta">theta</code></td>
<td>
<p>sum of ((group mean - overall mean)/stdev)^2 across all groups for each hypothesis test(vector)</p>
</td></tr>
<tr><td><code id="average.power.oneway_+3A_k">k</code></td>
<td>
<p>the number of groups to be compared</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>See Also</h3>

<p><code><a href="#topic+power.oneway">power.oneway</a></code> for more details about power calculation of one-way ANOVA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>theta=rep(c(2,0),c(100,900));
average.power.oneway(n = 50, alpha = 0.05, theta = theta, k = 2)
</code></pre>

<hr>
<h2 id='average.power.ranksum'>Compute average power of rank-sum tests</h2><span id='topic+average.power.ranksum'></span>

<h3>Description</h3>

<p>Compute average power of rank-sum tests
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power.ranksum(n, alpha, p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power.ranksum_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="average.power.ranksum_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="average.power.ranksum_+3A_p">p</code></td>
<td>
<p>Pr(Y&gt;X), as in Noether (JASA 1987)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>See Also</h3>

<p><code><a href="#topic+power.ranksum">power.ranksum</a></code> for more details about power calculation of rank-sum test. The power calculation is based on asymptotic normal approximation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = rep(c(0.8,0.5),c(100,900));
average.power.ranksum(n = 50, alpha = 0.05, p=p)
</code></pre>

<hr>
<h2 id='average.power.signrank'>Compute average power of many signed-rank tests</h2><span id='topic+average.power.signrank'></span>

<h3>Description</h3>

<p>Compute average power of many signed-rank tests
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power.signrank(n, alpha, p1, p2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power.signrank_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="average.power.signrank_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="average.power.signrank_+3A_p1">p1</code></td>
<td>
<p>Pr(X&gt;0), as in Noether (JASA 1987)</p>
</td></tr>
<tr><td><code id="average.power.signrank_+3A_p2">p2</code></td>
<td>
<p>Pr(X+X'&gt;0), as in Noether (JASA 1987)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>See Also</h3>

<p><code><a href="#topic+power.signrank">power.signrank</a></code> for more details about power calculation of signed-rank test. The power calculation is based on asymptotic normal approximation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p1 = rep(c(0.8,0.5),c(100,900));
p2 = rep(c(0.8,0.5),c(100,900));
average.power.signrank(n = 50, alpha = 0.05, p1 = p1, p2 = p2)
</code></pre>

<hr>
<h2 id='average.power.signtest'>Compute average power of many sign tests</h2><span id='topic+average.power.signtest'></span>

<h3>Description</h3>

<p>Compute average power of many sign tests
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power.signtest(n, alpha, p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power.signtest_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="average.power.signtest_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="average.power.signtest_+3A_p">p</code></td>
<td>
<p>Pr(Y&gt;X), as in Noether (JASA 1987)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>See Also</h3>

<p><code><a href="#topic+power.signtest">power.signtest</a></code> for more details about power calculation of sign test. The power calculation is based on asymptotic normal approximation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = rep(c(0.8,0.5),c(100,900));
average.power.signtest(n = 50, alpha = 0.05, p=p)
</code></pre>

<hr>
<h2 id='average.power.t.test'>Compute average power of many t-tests</h2><span id='topic+average.power.t.test'></span>

<h3>Description</h3>

<p>Compute average power of many t-tests; Uses classical power formula for t-test; Assumes equal variance and sample size
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power.t.test(
  n,
  alpha,
  delta,
  sigma = 1,
  type = "two.sample",
  alternative = "two.sided"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power.t.test_+3A_n">n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td></tr>
<tr><td><code id="average.power.t.test_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="average.power.t.test_+3A_delta">delta</code></td>
<td>
<p>difference of population means (vector)</p>
</td></tr>
<tr><td><code id="average.power.t.test_+3A_sigma">sigma</code></td>
<td>
<p>standard deviation (vector or scalar, default=1)</p>
</td></tr>
<tr><td><code id="average.power.t.test_+3A_type">type</code></td>
<td>
<p>type of t-test: &quot;two.sample&quot;, &quot;one.sample&quot;</p>
</td></tr>
<tr><td><code id="average.power.t.test_+3A_alternative">alternative</code></td>
<td>
<p>one- or two-sided test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>Examples</h3>

<pre><code class='language-R'>d = rep(c(2,0),c(100,900));
average.power.t.test(n = 20, alpha = 0.05,delta = d)
</code></pre>

<hr>
<h2 id='average.power.tcorr'>Compute average power of many t-tests for non-zero correlation</h2><span id='topic+average.power.tcorr'></span>

<h3>Description</h3>

<p>Compute average power of many t-tests for non-zero correlation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power.tcorr(n, alpha, rho)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power.tcorr_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="average.power.tcorr_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="average.power.tcorr_+3A_rho">rho</code></td>
<td>
<p>population correlation coefficient (vector)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For many applications, the null.effect is rho = 0
</p>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>See Also</h3>

<p><code><a href="#topic+power.tcorr">power.tcorr</a></code> for more details about power calculation of t-test for non-zero correlation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rho = rep(c(0.3,0),c(100,900));
average.power.tcorr(n = 50, alpha = 0.05, rho = rho)
</code></pre>

<hr>
<h2 id='average.power.twoprop'>Computer average power of many two proportion z-tests</h2><span id='topic+average.power.twoprop'></span>

<h3>Description</h3>

<p>Computer average power of many two proportion z-tests.The power calculation of two proportion z-test is based on asymptotic normal approximation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power.twoprop(n, alpha, p1, p2, alternative)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power.twoprop_+3A_n">n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td></tr>
<tr><td><code id="average.power.twoprop_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="average.power.twoprop_+3A_p1">p1</code></td>
<td>
<p>probability in one group (vector)</p>
</td></tr>
<tr><td><code id="average.power.twoprop_+3A_p2">p2</code></td>
<td>
<p>probability in other group (vector)</p>
</td></tr>
<tr><td><code id="average.power.twoprop_+3A_alternative">alternative</code></td>
<td>
<p>one- or two-sided test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234);
p1 = sample(seq(0,0.5,0.1),40,replace = TRUE);
p2 = sample(seq(0.5,1,0.1),40,replace = TRUE);
average.power.twoprop(n = 30, alpha = 0.05, p1 = p1,p2 = p2,alternative="two.sided")
</code></pre>

<hr>
<h2 id='fdr.avepow'>Compute FDR and average power for a given sample size and effect size vector</h2><span id='topic+fdr.avepow'></span>

<h3>Description</h3>

<p>For a given fixed sample size and effect size vector,compute FDR and average power as a function of the p-value threshold alpha.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fdr.avepow(n, avepow.func, null.hypo, alpha = 1:100/1000, method = "BH", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fdr.avepow_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="fdr.avepow_+3A_avepow.func">avepow.func</code></td>
<td>
<p>function to compute average power</p>
</td></tr>
<tr><td><code id="fdr.avepow_+3A_null.hypo">null.hypo</code></td>
<td>
<p>string to evaluate null hypothesis</p>
</td></tr>
<tr><td><code id="fdr.avepow_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold(s) to consider</p>
</td></tr>
<tr><td><code id="fdr.avepow_+3A_method">method</code></td>
<td>
<p>method to estimate proportion pi0 of tests with a true null hypothesis, including: &quot;HH&quot; (p-value histogram height) , &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
<tr><td><code id="fdr.avepow_+3A_...">...</code></td>
<td>
<p>additional arguments, including effect size vector for average power function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>input sample size</p>
</td></tr>
<tr><td><code>avepow.func</code></td>
<td>
<p>average power function</p>
</td></tr>
<tr><td><code>null.hypo</code></td>
<td>
<p>null hypothesis string</p>
</td></tr>
<tr><td><code>pi0</code></td>
<td>
<p>computed value of pi0</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height) , &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
<tr><td><code>other.args</code></td>
<td>
<p>additional arguments</p>
</td></tr>
<tr><td><code>res.tbl</code></td>
<td>
<p>table of alpha, fdr, and average power</p>
</td></tr>
</table>


<h3>References</h3>

<p>Pounds S and Cheng C, &quot;Sample size determination for the false discovery rate.&quot; Bioinformatics 21.23 (2005): 4263-4271.
</p>
<p>Gadbury GL, et al. (2004) Power and sample size estimation in high dimensional biology. Statistical Methods in Medical Research 13(4):325-38.
</p>
<p>Jung,Sin-Ho.&quot;Sample size for FDR-control in microarray data analysis.&quot; Bioinformatics 21.14 (2005): 3097-3104.
</p>
<p>Ni Y, Seffernick A, Onar-Thomas A, Pounds S.  &quot;Computing Power and Sample Size for the False Discovery Rate in Multiple Applications&quot;, Manuscript.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n = 50; # number of events
logHR = rep(c(0,0.5),c(950,50));
v = rep(1,length(logHR));  # variance of predictor variable (vector)
res = fdr.avepow(n,average.power.coxph,"logHR==0",logHR=logHR,v=v);
res$pi0;
head(res$res.tbl)
</code></pre>

<hr>
<h2 id='fdr.power.alpha'>Compute FDR for given p-value threshold, average power and proportion of tests with a true null</h2><span id='topic+fdr.power.alpha'></span>

<h3>Description</h3>

<p>Compute the FDR for given values of the p-value threshold alpha, average power, and proportion pi0 of tests with a true null hypothesis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fdr.power.alpha(alpha, pwr, pi0, method = "HH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fdr.power.alpha_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (vector)</p>
</td></tr>
<tr><td><code id="fdr.power.alpha_+3A_pwr">pwr</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code id="fdr.power.alpha_+3A_pi0">pi0</code></td>
<td>
<p>actual proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code id="fdr.power.alpha_+3A_method">method</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height), &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>FDR
</p>


<h3>References</h3>

<p>Pounds S and Cheng C, &quot;Sample size determination for the false discovery rate.&quot; Bioinformatics 21.23 (2005): 4263-4271.
</p>
<p>Gadbury GL, et al. (2004) Power and sample size estimation in high dimensional biology. Statistical Methods in Medical Research 13(4):325-38.
</p>
<p>Jung,Sin-Ho.&quot;Sample size for FDR-control in microarray data analysis.&quot; Bioinformatics 21.14 (2005): 3097-3104.
</p>
<p>Ni Y, Seffernick A, Onar-Thomas A, Pounds S.  &quot;Computing Power and Sample Size for the False Discovery Rate in Multiple Applications&quot;, Manuscript.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alpha = 1:100/1000;
pwr = rep(0.8,length(alpha));
pi0 = 0.95;
fdr.power.alpha(alpha,pwr,pi0,method="HH")
</code></pre>

<hr>
<h2 id='find.sample.size'>Determines the sample size needed to achieve the desired FDR and average power</h2><span id='topic+find.sample.size'></span>

<h3>Description</h3>

<p>Determines the sample size needed to achieve the desired FDR and average power by given the proportion of true null hypothesis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.sample.size(alpha, pwr, avepow.func, n0 = 3, n1 = 6, max.its = 50, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find.sample.size_+3A_alpha">alpha</code></td>
<td>
<p>the fixed p-value threshold  (scalar numeric)</p>
</td></tr>
<tr><td><code id="find.sample.size_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="find.sample.size_+3A_avepow.func">avepow.func</code></td>
<td>
<p>an R function to compute average power</p>
</td></tr>
<tr><td><code id="find.sample.size_+3A_n0">n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code id="find.sample.size_+3A_n1">n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
<tr><td><code id="find.sample.size_+3A_max.its">max.its</code></td>
<td>
<p>maximum number of iterations</p>
</td></tr>
<tr><td><code id="find.sample.size_+3A_...">...</code></td>
<td>
<p>additional arguments to average power function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>a sample size estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>Note</h3>

<p>For the test with power calculation based on asymptotic normal approximation, we suggest checking <code>FDRsamplesize2</code> calculation by simulation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Here, calculating the sample size for the study involving many sign tests
average.power.signtest;
p.adj = 0.001;
p = rep(c(0.8,0.5), c(100,9900));
find.sample.size(alpha = p.adj, pwr = 0.8, avepow.func = average.power.signtest, p = p)
</code></pre>

<hr>
<h2 id='n.fdr.coxph'>Sample size calculation for the Cox proportional hazards regression model</h2><span id='topic+n.fdr.coxph'></span>

<h3>Description</h3>

<p>Find number of events needed to have a desired false discovery rate and average power for a large number of Cox regression models with non-binary covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr.coxph(fdr, pwr, logHR, v, pi0.hat = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr.coxph_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.coxph_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.coxph_+3A_loghr">logHR</code></td>
<td>
<p>log hazard ratio (vector)</p>
</td></tr>
<tr><td><code id="n.fdr.coxph_+3A_v">v</code></td>
<td>
<p>variance of predictor variable (vector)</p>
</td></tr>
<tr><td><code id="n.fdr.coxph_+3A_pi0.hat">pi0.hat</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height), &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>number of events estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>desired.fdr</code></td>
<td>
<p>desired FDR</p>
</td></tr>
<tr><td><code>input.pi0</code></td>
<td>
<p>proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>Note</h3>

<p>For the test with power calculation based on asymptotic normal approximation, we suggest checking <code>FDRsamplesize2</code> calculation by simulation.
</p>


<h3>References</h3>

<p>Hsieh, FY and Lavori, Philip W (2000) Sample-size calculations for the Cox proportional hazards regression model with non-binary covariates. Controlled Clinical Trials 21(6):552-560.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>log.HR=log(rep(c(1,2),c(900,100)))
v=rep(1,1000)
n.fdr.coxph(fdr=0.1, pwr=0.8,logHR=log.HR, v=v, pi0.hat="BH")
</code></pre>

<hr>
<h2 id='n.fdr.fisher'>Sample size calculation for Fisher's Exact tests</h2><span id='topic+n.fdr.fisher'></span>

<h3>Description</h3>

<p>Find the sample size needed to have a desired false discovery rate and average power for a large number of Fisher's exact tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr.fisher(fdr, pwr, p1, p2, alternative = "two.sided", pi0.hat = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr.fisher_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.fisher_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.fisher_+3A_p1">p1</code></td>
<td>
<p>probability in one group (vector)</p>
</td></tr>
<tr><td><code id="n.fdr.fisher_+3A_p2">p2</code></td>
<td>
<p>probability in other group (vector)</p>
</td></tr>
<tr><td><code id="n.fdr.fisher_+3A_alternative">alternative</code></td>
<td>
<p>one- or two-sided test</p>
</td></tr>
<tr><td><code id="n.fdr.fisher_+3A_pi0.hat">pi0.hat</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height) , &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>per-group sample size estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>desired.fdr</code></td>
<td>
<p>desired FDR</p>
</td></tr>
<tr><td><code>input.pi0</code></td>
<td>
<p>proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234);
p1 = sample(seq(0,0.5,0.1),10,replace = TRUE);
p2 = sample(seq(0.5,1,0.1),10,replace = TRUE);
n.fdr.fisher(fdr = 0.1, pwr = 0.8, p1 = p1, p2 = p2, alternative = "two.sided", pi0.hat = "BH")
</code></pre>

<hr>
<h2 id='n.fdr.negbin'>Sample size calculation for Negative Binomial data</h2><span id='topic+n.fdr.negbin'></span>

<h3>Description</h3>

<p>Find the sample size needed to have a desired false discovery rate and average power for a large number of Negative Binomial comparisons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr.negbin(fdr, pwr, log.fc, mu, sig, pi0.hat = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr.negbin_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.negbin_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.negbin_+3A_log.fc">log.fc</code></td>
<td>
<p>log fold-change (vector), usual null hypothesis is log.fc=0</p>
</td></tr>
<tr><td><code id="n.fdr.negbin_+3A_mu">mu</code></td>
<td>
<p>read depth per gene (vector, same length as log.fc)</p>
</td></tr>
<tr><td><code id="n.fdr.negbin_+3A_sig">sig</code></td>
<td>
<p>coefficient of variation (CV) per gene (vector, same length as log.fc)</p>
</td></tr>
<tr><td><code id="n.fdr.negbin_+3A_pi0.hat">pi0.hat</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height), &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>per-group sample size estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>desired.fdr</code></td>
<td>
<p>desired FDR</p>
</td></tr>
<tr><td><code>input.pi0</code></td>
<td>
<p>proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>Note</h3>

<p>For the test with power calculation based on asymptotic normal approximation, we suggest checking <code>FDRsamplesize2</code> calculation by simulation.
</p>


<h3>References</h3>

<p>SN Hart, TM Therneau, Y Zhang, GA Poland, and J-P Kocher (2013). Calculating Sample Size Estimates for RNA Sequencing Data. Journal of Computational Biology 20: 970-978.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>logFC = log(rep(c(1,2),c(900,100)));
mu = rep(5,1000);
sig = rep(0.6,1000);
n.fdr.negbin(fdr = 0.1, pwr = 0.8, log.fc = logFC, mu = mu, sig = sig, pi0.hat = "BH")
</code></pre>

<hr>
<h2 id='n.fdr.oneway'>Sample size calculation for one-way ANOVA</h2><span id='topic+n.fdr.oneway'></span>

<h3>Description</h3>

<p>Find the sample size needed to have a desired false discovery rate and average power for a large number of one-way ANOVA tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr.oneway(fdr, pwr, theta, k, pi0.hat = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr.oneway_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.oneway_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.oneway_+3A_theta">theta</code></td>
<td>
<p>sum of ((group mean - overall mean)/stdev)^2 across all groups for each hypothesis test (vector)</p>
</td></tr>
<tr><td><code id="n.fdr.oneway_+3A_k">k</code></td>
<td>
<p>the number of groups to be compared</p>
</td></tr>
<tr><td><code id="n.fdr.oneway_+3A_pi0.hat">pi0.hat</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height), &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>per-group sample size estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>desired.fdr</code></td>
<td>
<p>desired FDR</p>
</td></tr>
<tr><td><code>input.pi0</code></td>
<td>
<p>proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>theta=rep(c(2,0),c(100,900));
n.fdr.oneway(fdr = 0.1, pwr = 0.8, theta = theta, k = 2, pi0.hat = "BH")
</code></pre>

<hr>
<h2 id='n.fdr.poisson'>Sample size calculation for Poisson data</h2><span id='topic+n.fdr.poisson'></span>

<h3>Description</h3>

<p>Find the sample size needed to have a desired false discovery rate and average power for a large number of two-group comparisons under Poisson distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr.poisson(fdr, pwr, rho, mu0, w, type, pi0.hat = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr.poisson_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.poisson_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.poisson_+3A_rho">rho</code></td>
<td>
<p>fold-change, usual null hypothesis is that rho=1 (vector)</p>
</td></tr>
<tr><td><code id="n.fdr.poisson_+3A_mu0">mu0</code></td>
<td>
<p>average count in control group (vector)</p>
</td></tr>
<tr><td><code id="n.fdr.poisson_+3A_w">w</code></td>
<td>
<p>ratio of the total number of reads mapped between the two groups</p>
</td></tr>
<tr><td><code id="n.fdr.poisson_+3A_type">type</code></td>
<td>
<p>type of test: &quot;w&quot; for Wald, &quot;s&quot; for score, &quot;lw&quot; for log-transformed Wald, &quot;ls&quot; for log-transformed score.</p>
</td></tr>
<tr><td><code id="n.fdr.poisson_+3A_pi0.hat">pi0.hat</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height) , &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>per-group sample size estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>desired.fdr</code></td>
<td>
<p>desired FDR</p>
</td></tr>
<tr><td><code>input.pi0</code></td>
<td>
<p>proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>References</h3>

<p>C-I Li, P-F Su, Y Guo, and Y Shyr (2013). Sample size calculation for differential expression analysis of RNA-seq data under Poisson distribution. Int J Comput Biol Drug Des 6(4).&lt;doi:10.1504/IJCBDD.2013.056830&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rho = rep(c(1,1.25),c(900,100));
mu0 = rep(5,1000);
w = rep(0.5,1000);
n.fdr.poisson(fdr = 0.1, pwr = 0.8, rho = rho, mu0 = mu0, w = w, type = "w", pi0.hat = "BH")
</code></pre>

<hr>
<h2 id='n.fdr.ranksum'>Sample size calculation for rank-sum tests</h2><span id='topic+n.fdr.ranksum'></span>

<h3>Description</h3>

<p>Find the sample size needed to have a desired false discovery rate and average power for a large number of rank-sum tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr.ranksum(fdr, pwr, p, pi0.hat = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr.ranksum_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.ranksum_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.ranksum_+3A_p">p</code></td>
<td>
<p>Pr(Y&gt;X), as in Noether (JASA 1987)</p>
</td></tr>
<tr><td><code id="n.fdr.ranksum_+3A_pi0.hat">pi0.hat</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height), &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>sample size estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>desired.fdr</code></td>
<td>
<p>desired FDR</p>
</td></tr>
<tr><td><code>input.pi0</code></td>
<td>
<p>proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>References</h3>

<p>Noether, Gottfried E (1987) Sample size determination for some common nonparametric tests. Journal of the American Statistical Association, 82:645-647.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = rep(c(0.8,0.5),c(100,900));
n.fdr.ranksum(fdr = 0.1, pwr = 0.8, p = p, pi0.hat = "BH")
</code></pre>

<hr>
<h2 id='n.fdr.signrank'>Sample size calculation for signed-rank tests</h2><span id='topic+n.fdr.signrank'></span>

<h3>Description</h3>

<p>Find the sample size needed to have a desired false discovery rate and average power for a large number of signed-rank tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr.signrank(fdr, pwr, p1, p2, pi0.hat = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr.signrank_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.signrank_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.signrank_+3A_p1">p1</code></td>
<td>
<p>Pr(X&gt;0), as in Noether (JASA 1987)</p>
</td></tr>
<tr><td><code id="n.fdr.signrank_+3A_p2">p2</code></td>
<td>
<p>Pr(X+X'&gt;0), as in Noether (JASA 1987)</p>
</td></tr>
<tr><td><code id="n.fdr.signrank_+3A_pi0.hat">pi0.hat</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height) , &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>sample size estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>desired.fdr</code></td>
<td>
<p>desired FDR</p>
</td></tr>
<tr><td><code>input.pi0</code></td>
<td>
<p>proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>References</h3>

<p>Noether, Gottfried E (1987) Sample size determination for some common nonparametric tests. Journal of the American Statistical Association, 82:645-647.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p1 = rep(c(0.8,0.5),c(100,900));
p2 = rep(c(0.8,0.5),c(100,900));
n.fdr.signrank(fdr = 0.1, pwr = 0.8, p1 = p1, p2 = p2, pi0.hat = "BH")
</code></pre>

<hr>
<h2 id='n.fdr.signtest'>Sample size calculation for sign tests</h2><span id='topic+n.fdr.signtest'></span>

<h3>Description</h3>

<p>Find the sample size needed to have a desired false discovery rate and average power for a large number of sign tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr.signtest(fdr, pwr, p, pi0.hat = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr.signtest_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.signtest_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.signtest_+3A_p">p</code></td>
<td>
<p>Pr(X&gt;0), as in Noether (JASA 1987)</p>
</td></tr>
<tr><td><code id="n.fdr.signtest_+3A_pi0.hat">pi0.hat</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height), &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>sample size estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>desired.fdr</code></td>
<td>
<p>desired FDR</p>
</td></tr>
<tr><td><code>input.pi0</code></td>
<td>
<p>proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>Note</h3>

<p>For the test with power calculation based on asymptotic normal approximation, we suggest checking <code>FDRsamplesize2</code> calculation by simulation.
</p>


<h3>References</h3>

<p>Noether, Gottfried E (1987) Sample size determination for some common nonparametric tests. Journal of the American Statistical Association, 82:645-647.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = rep(c(0.8, 0.5), c(100, 900));
n.fdr.signtest(fdr = 0.1, pwr = 0.8, p = p, pi0.hat = "BH")
</code></pre>

<hr>
<h2 id='n.fdr.tcorr'>Sample size calculation for t-tests for non-zero correlation</h2><span id='topic+n.fdr.tcorr'></span>

<h3>Description</h3>

<p>Find the sample size needed to have a desired false discovery rate and average power for a large number of t-tests for non-zero correlation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr.tcorr(fdr, pwr, rho, pi0.hat = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr.tcorr_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.tcorr_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.tcorr_+3A_rho">rho</code></td>
<td>
<p>population correlation coefficient (vector)</p>
</td></tr>
<tr><td><code id="n.fdr.tcorr_+3A_pi0.hat">pi0.hat</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height), &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>sample size estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>desired.fdr</code></td>
<td>
<p>desired FDR</p>
</td></tr>
<tr><td><code>input.pi0</code></td>
<td>
<p>proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>rho = rep(c(0.3,0),c(100,900));
n.fdr.tcorr(fdr = 0.1, pwr = 0.8, rho = rho, pi0.hat="BH")
</code></pre>

<hr>
<h2 id='n.fdr.ttest'>Sample size calculation for t-tests</h2><span id='topic+n.fdr.ttest'></span>

<h3>Description</h3>

<p>Find the sample size needed to have a desired false discovery rate and average power for a large number of t-tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr.ttest(
  fdr,
  pwr,
  delta,
  sigma = 1,
  type = "two.sample",
  pi0.hat = "BH",
  alternative = "two.sided"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr.ttest_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.ttest_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.ttest_+3A_delta">delta</code></td>
<td>
<p>difference of population means (vector)</p>
</td></tr>
<tr><td><code id="n.fdr.ttest_+3A_sigma">sigma</code></td>
<td>
<p>standard deviation (vector or scalar)</p>
</td></tr>
<tr><td><code id="n.fdr.ttest_+3A_type">type</code></td>
<td>
<p>type of t-test</p>
</td></tr>
<tr><td><code id="n.fdr.ttest_+3A_pi0.hat">pi0.hat</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height), &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
<tr><td><code id="n.fdr.ttest_+3A_alternative">alternative</code></td>
<td>
<p>one- or two-sided test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>sample size (per group) estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>desired.fdr</code></td>
<td>
<p>desired FDR</p>
</td></tr>
<tr><td><code>input.pi0</code></td>
<td>
<p>proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>d = rep(c(2,0),c(100,900));
n.fdr.ttest(fdr = 0.1, pwr = 0.8, delta = d)
</code></pre>

<hr>
<h2 id='n.fdr.twoprop'>Sample size calculation for comparing two proportions</h2><span id='topic+n.fdr.twoprop'></span>

<h3>Description</h3>

<p>Find the sample size needed to have a desired false discovery rate and average power for a large number of two-group comparisons using the two proportion z-test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr.twoprop(fdr, pwr, p1, p2, alternative = "two.sided", pi0.hat = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr.twoprop_+3A_fdr">fdr</code></td>
<td>
<p>desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.twoprop_+3A_pwr">pwr</code></td>
<td>
<p>desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="n.fdr.twoprop_+3A_p1">p1</code></td>
<td>
<p>probability in one group (vector)</p>
</td></tr>
<tr><td><code id="n.fdr.twoprop_+3A_p2">p2</code></td>
<td>
<p>probability in other group (vector)</p>
</td></tr>
<tr><td><code id="n.fdr.twoprop_+3A_alternative">alternative</code></td>
<td>
<p>one- or two-sided test</p>
</td></tr>
<tr><td><code id="n.fdr.twoprop_+3A_pi0.hat">pi0.hat</code></td>
<td>
<p>method to estimate proportion <code>pi0</code> of tests with true null, including: &quot;HH&quot; (p-value histogram height), &quot;HM&quot; (p-value histogram mean), &quot;BH&quot; (Benjamini &amp; Hochberg 1995), &quot;Jung&quot; (Jung 2005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>per-group sample size estimate</p>
</td></tr>
<tr><td><code>computed.avepow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>desired.avepow</code></td>
<td>
<p>desired average power</p>
</td></tr>
<tr><td><code>desired.fdr</code></td>
<td>
<p>desired FDR</p>
</td></tr>
<tr><td><code>input.pi0</code></td>
<td>
<p>proportion of tests with a true null hypothesis</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>fixed p-value threshold for multiple testing procedure</p>
</td></tr>
<tr><td><code>n.its</code></td>
<td>
<p>number of iteration</p>
</td></tr>
<tr><td><code>max.its</code></td>
<td>
<p>maximum number of iteration, default is 50</p>
</td></tr>
<tr><td><code>n0</code></td>
<td>
<p>lower limit for initial sample size range</p>
</td></tr>
<tr><td><code>n1</code></td>
<td>
<p>upper limit for initial sample size range</p>
</td></tr>
</table>


<h3>Note</h3>

<p>For the test with power calculation based on asymptotic normal approximation, we suggest checking <code>FDRsamplesize2</code> calculation by simulation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234);
p1 = sample(seq(0,0.5,0.1),40,replace = TRUE);
p2 = sample(seq(0.5,1,0.1),40,replace = TRUE);
n.fdr.twoprop(fdr = 0.1, pwr = 0.8, p1 = p1, p2 = p2, alternative = "two.sided", pi0.hat = "BH")
</code></pre>

<hr>
<h2 id='power.cox'>Compute the power of a single-predictor Cox regression model</h2><span id='topic+power.cox'></span>

<h3>Description</h3>

<p>Use the formula of Hsieh and Lavori (2000) to compute the power of a single-predictor Cox model, which is based on asymptotic normal approximation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.cox(n, alpha, logHR, v)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.cox_+3A_n">n</code></td>
<td>
<p>number of events (scalar)</p>
</td></tr>
<tr><td><code id="power.cox_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.cox_+3A_loghr">logHR</code></td>
<td>
<p>log hazard ratio (vector)</p>
</td></tr>
<tr><td><code id="power.cox_+3A_v">v</code></td>
<td>
<p>variance of predictor variable (vector)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of power estimates for two-sided test
</p>


<h3>References</h3>

<p>Hsieh, FY and Lavori, Philip W (2000) Sample-size calculations for the Cox proportional hazards regression model with non-binary covariates. Controlled Clinical Trials 21(6):552-560.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>logHR = log(rep(c(1, 2),c(900, 100)));
v = rep(1, 1000);
res = power.cox(n = 50,alpha = 0.05,logHR = logHR, v = v)
</code></pre>

<hr>
<h2 id='power.fisher'>Compute power for Fisher's exact test</h2><span id='topic+power.fisher'></span>

<h3>Description</h3>

<p>Compute power for Fisher's exact test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.fisher(p1, p2, n, alpha, alternative)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.fisher_+3A_p1">p1</code></td>
<td>
<p>probability in one group (scalar)</p>
</td></tr>
<tr><td><code id="power.fisher_+3A_p2">p2</code></td>
<td>
<p>probability in other group (scalar)</p>
</td></tr>
<tr><td><code id="power.fisher_+3A_n">n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.fisher_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.fisher_+3A_alternative">alternative</code></td>
<td>
<p>one- or two-sided test, must be one of &quot;greater&quot;, &quot;less&quot;, or &quot;two.sided&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Power estimate for one- or two-sided tests
</p>


<h3>Examples</h3>

<pre><code class='language-R'>power.fisher(p1 = 0.5, p2 = 0.9, n=20, alpha = 0.05, alternative = 'two.sided')
</code></pre>

<hr>
<h2 id='power.hart'>Compute power for RNA-seq experiments assuming Negative Binomial distribution</h2><span id='topic+power.hart'></span>

<h3>Description</h3>

<p>Use the formula of Hart et al (2013) to compute power for comparing RNA-seq expression across two groups assuming a Negative Binomial distribution. The power calculation is based on asymptotic normal approximation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.hart(n, alpha, log.fc, mu, sig)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.hart_+3A_n">n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.hart_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.hart_+3A_log.fc">log.fc</code></td>
<td>
<p>log fold-change (vector), usual null hypothesis is log.fc=0</p>
</td></tr>
<tr><td><code id="power.hart_+3A_mu">mu</code></td>
<td>
<p>read depth per gene (vector, same length as log.fc)</p>
</td></tr>
<tr><td><code id="power.hart_+3A_sig">sig</code></td>
<td>
<p>coefficient of variation (CV) per gene (vector, same length as log.fc)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is based on equation (1) of Hart et al (2013). It assumes a Negative Binomial model for RNA-seq read counts and equal sample size per group.
</p>


<h3>Value</h3>

<p>Vector of power estimates for the set of two-sided tests
</p>


<h3>References</h3>

<p>SN Hart, TM Therneau, Y Zhang, GA Poland, and J-P Kocher (2013). Calculating Sample Size Estimates for RNA Sequencing Data. Journal of Computational Biology 20: 970-978.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n.hart = 2*(qnorm(0.975)+qnorm(0.9))^2*(1/20+0.6^2)/(log(2)^2)   # Equation (6) of Hart et al
power.hart(n.hart,0.05,log(2),20,0.6)                            # Recapitulate 90% power
</code></pre>

<hr>
<h2 id='power.li'>Compute power for RNA-Seq experiments assuming Poisson distribution</h2><span id='topic+power.li'></span>

<h3>Description</h3>

<p>Use the formula of Li et al (2013) to compute power for comparing RNA-seq expression across two groups assuming the Poisson distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.li(n, alpha, rho, mu0, w, type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.li_+3A_n">n</code></td>
<td>
<p>per-group sample size</p>
</td></tr>
<tr><td><code id="power.li_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.li_+3A_rho">rho</code></td>
<td>
<p>fold-change, usual null hypothesis is that rho=1 (vector)</p>
</td></tr>
<tr><td><code id="power.li_+3A_mu0">mu0</code></td>
<td>
<p>average count in control group</p>
</td></tr>
<tr><td><code id="power.li_+3A_w">w</code></td>
<td>
<p>ratio of the total number of reads mapped between the two groups</p>
</td></tr>
<tr><td><code id="power.li_+3A_type">type</code></td>
<td>
<p>type of test: &quot;w&quot; for Wald, &quot;s&quot; for score, &quot;lw&quot; for log-transformed Wald, &quot;ls&quot; for log-transformed score</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the power for each of a series of two-sided tests defined by the input parameters. The power is based on the sample size formulas in equations (10-13) of Li et al (2013). Also, note that the null.effect is set to 1 in the examples because the usual null hypothesis is that the fold-change = 1.
</p>


<h3>Value</h3>

<p>Vector of power estimates for two-sided tests
</p>


<h3>References</h3>

<p>C-I Li, P-F Su, Y Guo, and Y Shyr (2013). Sample size calculation for differential expression analysis of RNA-seq data under Poisson distribution. Int J Comput Biol Drug Des 6(4). &lt;doi:10.1504/IJCBDD.2013.056830&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>power.li(n = 88, alpha = 0.05, rho = 1.25, mu0 = 5, w = 0.5,type = "w")
# recapitulate 80% power in Table 1 of Li et al (2013)
</code></pre>

<hr>
<h2 id='power.oneway'>Compute power of one-way ANOVA</h2><span id='topic+power.oneway'></span>

<h3>Description</h3>

<p>Compute power of one-way ANOVA; Uses classical power formula for ANOVA; Assumes equal variance and sample size
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.oneway(n, alpha, theta, k = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.oneway_+3A_n">n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.oneway_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.oneway_+3A_theta">theta</code></td>
<td>
<p>sum of ((group mean - overall mean)/stdev)^2 across all groups for each hypothesis test(vector)</p>
</td></tr>
<tr><td><code id="power.oneway_+3A_k">k</code></td>
<td>
<p>the number of groups to be compared, default k=2</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For many applications, the null effect is zero for the parameter theta described above
</p>


<h3>Value</h3>

<p>Vector of power estimates for test of equal means
</p>


<h3>Examples</h3>

<pre><code class='language-R'>theta=rep(c(2,0),c(100,900));
res = power.oneway(n = 50, alpha = 0.05, theta = theta, k = 2)
</code></pre>

<hr>
<h2 id='power.ranksum'>Compute power of the rank-sum test</h2><span id='topic+power.ranksum'></span>

<h3>Description</h3>

<p>Compute power of rank-sum test; Uses formula of Noether (JASA 1987), which is based on asymptotic normal approximation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.ranksum(n, alpha, p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.ranksum_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.ranksum_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.ranksum_+3A_p">p</code></td>
<td>
<p>Pr(Y&gt;X), as in Noether (JASA 1987)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In most applications, the null effect size will be designated by p = 0.5
</p>


<h3>Value</h3>

<p>Vector of power estimates for two-sided tests
</p>


<h3>References</h3>

<p>Noether, Gottfried E (1987) Sample size determination for some common nonparametric tests. Journal of the American Statistical Association, 82:645-647.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = rep(c(0.8,0.5),c(100,900))
res = power.ranksum(n = 50, alpha = 0.5, p=p)
</code></pre>

<hr>
<h2 id='power.signrank'>Compute power of the signed-rank test</h2><span id='topic+power.signrank'></span>

<h3>Description</h3>

<p>Use the Noether (1987) formula to compute the power of the signed-rank test, which is based on asymptotic normal approximation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.signrank(n, alpha, p1, p2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.signrank_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.signrank_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.signrank_+3A_p1">p1</code></td>
<td>
<p>Pr(X&gt;0), as in Noether (JASA 1987)</p>
</td></tr>
<tr><td><code id="power.signrank_+3A_p2">p2</code></td>
<td>
<p>Pr(X+X'&gt;0), as in Noether (JASA 1987)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In most applications, the null effect size will be designated by p1 = p2 = 0.5
</p>


<h3>Value</h3>

<p>Vector of power estimates for two-sided tests
</p>


<h3>References</h3>

<p>Noether, Gottfried E (1987) Sample size determination for some common nonparametric tests. Journal of the American Statistical Association, 82:645-647.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p1 = rep(c(0.8,0.5),c(100,900));
p2 = rep(c(0.8,0.5),c(100,900));
res = power.signrank(n = 50, alpha = 0.05, p1 = p1, p2 = p2)
</code></pre>

<hr>
<h2 id='power.signtest'>Compute power of the sign test</h2><span id='topic+power.signtest'></span>

<h3>Description</h3>

<p>Use the Noether (1987) formula to compute the power of the sign test, which is based on asymptotic normal approximation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.signtest(n, alpha, p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.signtest_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.signtest_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.signtest_+3A_p">p</code></td>
<td>
<p>Pr(X&gt;0), as in Noether (JASA 1987)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In most applications, the null effect size will be designated by p = 0.5
</p>


<h3>Value</h3>

<p>Vector of power estimates for two-sided tests
</p>


<h3>References</h3>

<p>Noether, Gottfried E (1987) Sample size determination for some common nonparametric tests. Journal of the American Statistical Association, 82:645-647.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = rep(c(0.8,0.5),c(100,900));
res = power.signtest(n = 50, alpha = 0.05, p = p)
</code></pre>

<hr>
<h2 id='power.tcorr'>Compute power of the t-test for non-zero correlation</h2><span id='topic+power.tcorr'></span>

<h3>Description</h3>

<p>Compute power of the t-test for non-zero correlation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.tcorr(n, alpha, rho)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.tcorr_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.tcorr_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.tcorr_+3A_rho">rho</code></td>
<td>
<p>population correlation coefficient (vector)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For many applications, the null.effect is rho = 0
</p>


<h3>Value</h3>

<p>Vector of power estimates for two-sided tests
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rho = rep(c(0.3,0),c(100,900));
res = power.tcorr(n = 50, alpha = 0.05, rho = rho)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
