<!DOCTYPE html><html><head><title>Help for package FDRsampsize</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FDRsampsize}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FDRsampsize-package'><p>An R package to Perform Power and Sample Size Calculations for Microarray Studies</p></a></li>
<li><a href='#afdr'><p>Compute the anticipated FDR</p></a></li>
<li><a href='#alpha.fdr'><p>Find the fixed p-value threshold that controls the FDR at a specified level</p></a></li>
<li><a href='#alpha.power'><p>Find the p-value threshold that gives a specified average power</p></a></li>
<li><a href='#average.power'><p>Compute average power for a given sample size</p></a></li>
<li><a href='#fdr.power'><p>Compute the average power at a specific FDR control level</p></a></li>
<li><a href='#fdr.sampsize'><p>Determine sample size required to achieve a desired average power while controlling the FDR at a specified level.</p></a></li>
<li><a href='#n.fdr'><p>Find the sample size that meets desired FDR and power criteria</p></a></li>
<li><a href='#pi.star'><p>Compute the anticipated null proportion estimate</p></a></li>
<li><a href='#power.cox'><p>Compute the power of a single-predictor Cox regression model</p></a></li>
<li><a href='#power.hart'><p>Compute Power for RNA-seq Experiments Assuming Negative Binomial Distribution.</p></a></li>
<li><a href='#power.li'><p>Compute Power for RNA-Seq Experiments Assuming Poisson Distribution</p></a></li>
<li><a href='#power.onesampt'><p>Compute power of the one-sample t-test</p></a></li>
<li><a href='#power.oneway'><p>Compute power of one-way ANOVA</p></a></li>
<li><a href='#power.ranksum'><p>Compute power of the rank-sum test</p></a></li>
<li><a href='#power.signtest'><p>Compute power of the sign test</p></a></li>
<li><a href='#power.tcorr'><p>Compute Power of the t-test for non-zero correlation</p></a></li>
<li><a href='#power.twosampt'><p>Compute power of the two-samples t-test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Compute Sample Size that Meets Requirements for Average Power
and FDR</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Stan Pounds &lt;stanley.pounds@stjude.org&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Stan Pounds &lt;stanley.pounds@stjude.org&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-01-06</td>
</tr>
<tr>
<td>Description:</td>
<td>Defines a collection of functions to compute average power and sample size for studies that use the false discovery rate as the final measure of statistical significance.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-01-15 01:53:06 UTC; spounds</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-01-15 10:28:09</td>
</tr>
</table>
<hr>
<h2 id='FDRsampsize-package'>An R package to Perform Power and Sample Size Calculations for Microarray Studies</h2><span id='topic+FDRsampsize-package'></span><span id='topic+FDRsampsize-package'></span><span id='topic+FDRsampsize'></span>

<h3>Description</h3>

<p> A general approach to performing power and sample size calculations for microarray
studies has been developed in the literature.  However, the software associated with those articles
implements the approach only for studies that will perform the t-test or one-way ANOVA to compare 
gene expression across two or more groups.  Here, we describe a set of R routines that implement 
the general method for power and sample size calculations for a wider variety of statistical tests.  
These routines accept the name of a function that computes the power for the statistical test of 
interest and thus have the flexibility to perform calculations for virtually any statistical test 
with a known power formula.</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
 Package: </td><td style="text-align: left;"> FDRsampsize</td>
</tr>
<tr>
 <td style="text-align: left;">
 Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
 Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
 Date: </td><td style="text-align: left;"> 2016-01-06</td>
</tr>
<tr>
 <td style="text-align: left;">
 License: </td><td style="text-align: left;"> GPL(&gt;=2) </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Stan Pounds &lt;stanley.pounds@stjude.org&gt; </p>


<h3>References</h3>

<p>A Onar-Thomas, S Pounds. FDRsampsize: An R package to Perform Generalized Power and Sample Size Calculations for Planning Studies that use the False Discovery Rate to Measure Significance, Manuscript 2016.
</p>
<p>Pounds, Stan, and Cheng Cheng. &quot;Sample size determination for the false discovery rate.&quot; Bioinformatics 21.23 (2005): 4263-4271.
</p>
<p>Jung, Sin-Ho. &quot;Sample size for FDR-control in microarray data analysis.&quot; Bioinformatics 21.14 (2005): 3097-3104.</p>

<hr>
<h2 id='afdr'>Compute the anticipated FDR</h2><span id='topic+afdr'></span>

<h3>Description</h3>

<p>Compute the anticipated FDR for given sample size, p-value threshold, and effect size.</p>


<h3>Usage</h3>

<pre><code class='language-R'>afdr (n, alpha, pow.func, eff.size, lam = 0.95, eps = 1e-04, 
    ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="afdr_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="afdr_+3A_alpha">alpha</code></td>
<td>
<p>p-value cut-off (scalar)</p>
</td></tr>
<tr><td><code id="afdr_+3A_pow.func">pow.func</code></td>
<td>
<p>an R function that computes statistical power</p>
</td></tr>
<tr><td><code id="afdr_+3A_eff.size">eff.size</code></td>
<td>
<p>effect size vector</p>
</td></tr>
<tr><td><code id="afdr_+3A_lam">lam</code></td>
<td>
<p>p-value at which to evaluate ensemble PDF (for pi.star)</p>
</td></tr>
<tr><td><code id="afdr_+3A_eps">eps</code></td>
<td>
<p>epsilon for numerical differentiation</p>
</td></tr>
<tr><td><code id="afdr_+3A_...">...</code></td>
<td>
<p>additional agruments for the functions</p>
</td></tr> </table>


<h3>Details</h3>

<p>The aFDR is defined by Pounds and Cheng (2005) as the anticipated false discovery rate
incurred by performing all tests with p-value threshold alpha given the same size
effect size and power function.</p>


<h3>Value</h3>

<p>the aFDR</p>


<h3>References</h3>

<p>Pounds, Stan, and Cheng Cheng. &quot;Sample size determination for the false discovery rate.&quot; Bioinformatics 21.23 (2005): 4263-4271.
</p>
<p>Jung, Sin-Ho. &quot;Sample size for FDR-control in microarray data analysis.&quot; Bioinformatics 21.14 (2005): 3097-3104.</p>


<h3>Examples</h3>

<pre><code class='language-R'> afdr(n=50,alpha=0.01,pow.func=power.twosampt,eff.size=rep(c(1,0),c(100,900)))
 </code></pre>

<hr>
<h2 id='alpha.fdr'>Find the fixed p-value threshold that controls the FDR at a specified level</h2><span id='topic+alpha.fdr'></span>

<h3>Description</h3>

<p>Find the p-value threshold that satisfies an FDR requirement (if such a threshold exists)</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha.fdr (fdr, n, pow.func, eff.size, null.effect, lam = 0.95, 
    eps = 1e-04, tol = 1e-04, ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alpha.fdr_+3A_fdr">fdr</code></td>
<td>
<p>Desired FDR, scalar</p>
</td></tr>
<tr><td><code id="alpha.fdr_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="alpha.fdr_+3A_pow.func">pow.func</code></td>
<td>
<p>an R function to compute statistical power</p>
</td></tr>
<tr><td><code id="alpha.fdr_+3A_eff.size">eff.size</code></td>
<td>
<p>effect size vector</p>
</td></tr>
<tr><td><code id="alpha.fdr_+3A_null.effect">null.effect</code></td>
<td>
<p>value of effect size that corresponds to the null hypothesis</p>
</td></tr>
<tr><td><code id="alpha.fdr_+3A_lam">lam</code></td>
<td>
<p>the lambda parameter in computing the pi0 (proportion of tests with a true null) estimate of Storey (2002)</p>
</td></tr>
<tr><td><code id="alpha.fdr_+3A_eps">eps</code></td>
<td>
<p>epsilon for numerical differentiation</p>
</td></tr>
<tr><td><code id="alpha.fdr_+3A_tol">tol</code></td>
<td>
<p>tolerance for bisection solution to alpha</p>
</td></tr>
<tr><td><code id="alpha.fdr_+3A_...">...</code></td>
<td>
<p>additional agruments for the functions</p>
</td></tr> </table>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table>
<tr><td><code>fdr</code></td>
<td>
<p>the FDR at that alpha</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>the determined alpha</p>
</td></tr>
<tr><td><code>OK</code></td>
<td>
<p>indicates if the requirement is met</p>
</td></tr>
</table>


<h3>References</h3>

<p>A Onar-Thomas, S Pounds &quot;FDRsampsize: An R package to Perform Generalized Power and Sample Size Calculations for Planning Studies that use the False Discovery Rate to Measure Significance&quot;, Manuscript 2015.
</p>
<p>Pounds, Stan, and Cheng Cheng. &quot;Sample size determination for the false discovery rate.&quot; Bioinformatics 21.23 (2005): 4263-4271.
</p>
<p>Jung, Sin-Ho. &quot;Sample size for FDR-control in microarray data analysis.&quot; Bioinformatics 21.14 (2005): 3097-3104.</p>


<h3>Examples</h3>

<pre><code class='language-R'> alpha.fdr(fdr=0.1,n=50,pow.func=power.twosampt,
           eff.size=rep(0:1,c(900,100)),null.effect=0)
 </code></pre>

<hr>
<h2 id='alpha.power'>Find the p-value threshold that gives a specified average power</h2><span id='topic+alpha.power'></span>

<h3>Description</h3>

<p>Find p-value cut-off that yields desired average power given n and effect size</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha.power (ave.pow, n, pow.func, eff.size, null.effect, tol = 1e-06, 
    ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alpha.power_+3A_ave.pow">ave.pow</code></td>
<td>
<p>desired average power (scalar)</p>
</td></tr>
<tr><td><code id="alpha.power_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="alpha.power_+3A_pow.func">pow.func</code></td>
<td>
<p>name of R function to compute statistical power</p>
</td></tr>
<tr><td><code id="alpha.power_+3A_eff.size">eff.size</code></td>
<td>
<p>effect size vector</p>
</td></tr>
<tr><td><code id="alpha.power_+3A_null.effect">null.effect</code></td>
<td>
<p>value of effect size that corresponds to null hypothesis</p>
</td></tr>
<tr><td><code id="alpha.power_+3A_tol">tol</code></td>
<td>
<p>tolerance for bisection solution to alpha</p>
</td></tr>
<tr><td><code id="alpha.power_+3A_...">...</code></td>
<td>
<p>additional agruments for the functions</p>
</td></tr> </table>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table>
<tr><td><code>alpha</code></td>
<td>
<p>desired value of alpha</p>
</td></tr>
<tr><td><code>ave.pow</code></td>
<td>
<p>average power at that alpha</p>
</td></tr>
</table>


<h3>References</h3>

<p>A Onar-Thomas, S Pounds. &quot;FDRsampsize: An R package to Perform Generalized Power and Sample Size Calculations for Planning Studies that use the False Discovery Rate to Measure Significance&quot;, Manuscript 2015.
Pounds, Stan, and Cheng Cheng. &quot;Sample size determination for the false discovery rate.&quot; Bioinformatics 21.23 (2005): 4263-4271.
</p>
<p>Jung, Sin-Ho. &quot;Sample size for FDR-control in microarray data analysis.&quot; Bioinformatics 21.14 (2005): 3097-3104.</p>


<h3>Examples</h3>

<pre><code class='language-R'> alpha.power(ave.pow=0.8,n=50,pow.func=power.twosampt,
             eff.size=rep(0:1,c(900,100)),null.effect=0)
 </code></pre>

<hr>
<h2 id='average.power'>Compute average power for a given sample size</h2><span id='topic+average.power'></span>

<h3>Description</h3>

<p>Compute average power for given sample size, effect size, and p-value threshold</p>


<h3>Usage</h3>

<pre><code class='language-R'>average.power (n, alpha, pow.func, eff.size, null.effect, ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average.power_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="average.power_+3A_alpha">alpha</code></td>
<td>
<p>p-value cut off (scalar)</p>
</td></tr>
<tr><td><code id="average.power_+3A_pow.func">pow.func</code></td>
<td>
<p>an R function to compute statistical power</p>
</td></tr>
<tr><td><code id="average.power_+3A_eff.size">eff.size</code></td>
<td>
<p>effect size vector</p>
</td></tr>
<tr><td><code id="average.power_+3A_null.effect">null.effect</code></td>
<td>
<p>value of effect size that corresponds to null hypothesis</p>
</td></tr>
<tr><td><code id="average.power_+3A_...">...</code></td>
<td>
<p>additional agruments for the functions</p>
</td></tr> </table>


<h3>Value</h3>

<p>average power (scalar)</p>


<h3>References</h3>

<p>Pounds, Stan, and Cheng Cheng. &quot;Sample size determination for the false discovery rate.&quot; Bioinformatics 21.23 (2005): 4263-4271.
Gadbury GL, et al. (2004) Power and sample size estimation in high dimensional biology.  Statistical Methods in Medical Research 13(4):325-38.
Jung, Sin-Ho. &quot;Sample size for FDR-control in microarray data analysis.&quot; Bioinformatics 21.14 (2005): 3097-3104.</p>


<h3>Examples</h3>

<pre><code class='language-R'> average.power(n=50,alpha=0.01,pow.func=power.twosampt,
               eff.size=rep(0:1,c(900,100)),null.effect=0)
 </code></pre>

<hr>
<h2 id='fdr.power'>Compute the average power at a specific FDR control level</h2><span id='topic+fdr.power'></span>

<h3>Description</h3>

<p>Compute the average power at a specific level of FDR control for a given effect size and sample size</p>


<h3>Usage</h3>

<pre><code class='language-R'>fdr.power (fdr, n, pow.func, eff.size, null.effect, lam = 0.95, 
    eps = 1e-04, tol = 1e-04, ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fdr.power_+3A_fdr">fdr</code></td>
<td>
<p>Desired FDR, scalar</p>
</td></tr>
<tr><td><code id="fdr.power_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="fdr.power_+3A_pow.func">pow.func</code></td>
<td>
<p>name of R function to compute statistical power</p>
</td></tr>
<tr><td><code id="fdr.power_+3A_eff.size">eff.size</code></td>
<td>
<p>effect size vector; will be provided as the third argument of pow.func</p>
</td></tr>
<tr><td><code id="fdr.power_+3A_null.effect">null.effect</code></td>
<td>
<p>value of effect size that corresponds to null hypothesis</p>
</td></tr>
<tr><td><code id="fdr.power_+3A_lam">lam</code></td>
<td>
<p>name of R function to compute statistical power</p>
</td></tr>
<tr><td><code id="fdr.power_+3A_eps">eps</code></td>
<td>
<p>epsilon for numerical differentiation</p>
</td></tr>
<tr><td><code id="fdr.power_+3A_tol">tol</code></td>
<td>
<p>tolerance for bisection solution to alpha</p>
</td></tr>
<tr><td><code id="fdr.power_+3A_...">...</code></td>
<td>
<p>additional agruments for the functions</p>
</td></tr> </table>


<h3>Value</h3>

<p>average power (scalar) of the tests with a false null hypothesis</p>


<h3>References</h3>

<p>A Onar-Thomas, S Pounds &quot;FDRsampsize: An R package to Perform Generalized Power and Sample Size Calculations for Planning Studies that use the False Discovery Rate to Measure Significance&quot;, Manuscript 2016.
</p>
<p>Gadbury GL, et al. (2004) Power and sample size estimation in high dimensional biology.  Statistical Methods in Medical Research 13(4):325-38.
</p>
<p>Pounds S and Cheng C (2005) Sample size determination for the false discovery rate.  Bioinformatics 21(23): 4263-71.</p>


<h3>Examples</h3>

<pre><code class='language-R'> fdr.power(fdr=0.10,n=50,pow.func=power.twosampt,
           eff.size=rep(0:1,c(900,100)),null.effect=0)
</code></pre>

<hr>
<h2 id='fdr.sampsize'>Determine sample size required to achieve a desired average power while controlling the FDR at a specified level.</h2><span id='topic+fdr.sampsize'></span><span id='topic+print.fdr.sampsize'></span>

<h3>Description</h3>

<p>Determines the sample size needed to achieve a desired average power while controlling the FDR at a specified level.</p>


<h3>Usage</h3>

<pre><code class='language-R'>fdr.sampsize (fdr, ave.pow, pow.func, eff.size, null.effect, max.n = 500, 
    min.n = 5, tol = 1e-05, eps = 1e-05, lam = 0.95, ...) 
## S3 method for class 'fdr.sampsize'
print(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fdr.sampsize_+3A_fdr">fdr</code></td>
<td>
<p>Desired FDR (scalar numeric)</p>
</td></tr>
<tr><td><code id="fdr.sampsize_+3A_ave.pow">ave.pow</code></td>
<td>
<p>Desired average power (scalar numeric)</p>
</td></tr>
<tr><td><code id="fdr.sampsize_+3A_pow.func">pow.func</code></td>
<td>
<p>Character string name of function to compute power; must accept n, alpha, and eff.size as its first three arguments.  Other optional arguments may also be provided.</p>
</td></tr>
<tr><td><code id="fdr.sampsize_+3A_eff.size">eff.size</code></td>
<td>
<p>Numeric vector of effect sizes; interally, this will be provided as the third argument of pow.func</p>
</td></tr>
<tr><td><code id="fdr.sampsize_+3A_null.effect">null.effect</code></td>
<td>
<p>Scalar value of the effect size under the null hypothesis.  This may be 0 or 1 for tests that respectively use differences or ratios for comparisons.</p>
</td></tr>
<tr><td><code id="fdr.sampsize_+3A_max.n">max.n</code></td>
<td>
<p>Maximum n to consider</p>
</td></tr>
<tr><td><code id="fdr.sampsize_+3A_min.n">min.n</code></td>
<td>
<p>Minimum n to consider</p>
</td></tr>
<tr><td><code id="fdr.sampsize_+3A_tol">tol</code></td>
<td>
<p>Tolerance for bisection calculations</p>
</td></tr>
<tr><td><code id="fdr.sampsize_+3A_eps">eps</code></td>
<td>
<p>Epsilon for numerical differentiation</p>
</td></tr>
<tr><td><code id="fdr.sampsize_+3A_lam">lam</code></td>
<td>
<p>Lambda for computing anticipated pi0 estimate, see Storey 2002.</p>
</td></tr>
<tr><td><code id="fdr.sampsize_+3A_x">x</code></td>
<td>
<p>result of the fdr.sampsize function</p>
</td></tr>
<tr><td><code id="fdr.sampsize_+3A_...">...</code></td>
<td>
<p>additional arguments for pow.func</p>
</td></tr> </table>


<h3>Details</h3>

<p>This function checks the technical conditions regarding whether the desired FDR can be
achieved by min.n or max.n before calling n.fdr.  Thus, for most applications,
fdr.sampsize should be used instead of n.fdr.</p>


<h3>Value</h3>

<p><code>fdr.sampsiz</code>e returns an object of class 'FDRsampsize', which is a list with the following components:
</p>
<table>
<tr><td><code>OK</code></td>
<td>
<p>indicates if the requirement is met</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>the computed sample size</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>the p-value threshold that gives the desired FDR</p>
</td></tr>
<tr><td><code>fdr.hat</code></td>
<td>
<p>anticipated value of the FDR estimate given n and effect size </p>
</td></tr>
<tr><td><code>act.fdr</code></td>
<td>
<p>actual expected FDR given n and effect size</p>
</td></tr>
<tr><td><code>ave.pow</code></td>
<td>
<p>average power</p>
</td></tr>
<tr><td><code>act.pi</code></td>
<td>
<p>actual value of pi0, the proportion of tests with a true null hypothesis.</p>
</td></tr>
<tr><td><code>pi.hat</code></td>
<td>
<p>expected value of the pi0 estimate</p>
</td></tr>
<tr><td><code>eff.size</code></td>
<td>
<p>input effect size vector</p>
</td></tr>
</table>


<h3>References</h3>

<p>A Onar-Thomas, S Pounds. &quot;FDRsampsize: An R package to Perform Generalized Power and Sample Size Calculations for Planning Studies that use the False Discovery Rate to Measure Significance&quot;, Manuscript 2015.
</p>
<p>Pounds, Stan, and Cheng Cheng. &quot;Sample size determination for the false discovery rate.&quot; Bioinformatics 21.23 (2005): 4263-4271.
</p>
<p>Jung, Sin-Ho. &quot;Sample size for FDR-control in microarray data analysis.&quot; Bioinformatics 21.14 (2005): 3097-3104.</p>


<h3>Examples</h3>

<pre><code class='language-R'> power.twosampt             # show the power.cox function
 res=fdr.sampsize(fdr=0.1,
                  ave.pow=0.8,
                  pow.func=power.twosampt,
                  eff.size=rep(c(1,0),c(10,990)),
                  null.effect=0)
 res
 </code></pre>

<hr>
<h2 id='n.fdr'>Find the sample size that meets desired FDR and power criteria</h2><span id='topic+n.fdr'></span>

<h3>Description</h3>

<p>Find smallest sample size that meets requirements for average power and FDR</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.fdr (ave.pow, fdr, pow.func, eff.size, null.effect, lam = 0.95, 
    eps = 1e-04, n0 = 5, n1 = 500, tol = 1e-06, ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n.fdr_+3A_ave.pow">ave.pow</code></td>
<td>
<p>required average power (scalar)</p>
</td></tr>
<tr><td><code id="n.fdr_+3A_fdr">fdr</code></td>
<td>
<p>required FDR (scalar)</p>
</td></tr>
<tr><td><code id="n.fdr_+3A_pow.func">pow.func</code></td>
<td>
<p>name of R function that computes statistical power</p>
</td></tr>
<tr><td><code id="n.fdr_+3A_eff.size">eff.size</code></td>
<td>
<p>effect size vector</p>
</td></tr>
<tr><td><code id="n.fdr_+3A_null.effect">null.effect</code></td>
<td>
<p>Value of effect size that indicates null</p>
</td></tr>
<tr><td><code id="n.fdr_+3A_lam">lam</code></td>
<td>
<p>p-value at which to evaluate ensemble PDF</p>
</td></tr>
<tr><td><code id="n.fdr_+3A_eps">eps</code></td>
<td>
<p>epsilon for numerical differentiation</p>
</td></tr>
<tr><td><code id="n.fdr_+3A_n0">n0</code></td>
<td>
<p>smallest sample size to be considered for bisection</p>
</td></tr>
<tr><td><code id="n.fdr_+3A_n1">n1</code></td>
<td>
<p>maximum sample size to be considered for bisection</p>
</td></tr>
<tr><td><code id="n.fdr_+3A_tol">tol</code></td>
<td>
<p>tolerance for solving for alpha in iterations</p>
</td></tr>
<tr><td><code id="n.fdr_+3A_...">...</code></td>
<td>
<p>additional agruments for the functions</p>
</td></tr> </table>


<h3>Details</h3>

<p>This performs the sample size calculation without checking whether the
minimum or maximum sample size satisfy the desired requirements.  The fdr.sampsize
function checks these conditions and then calls n.fdr.  Thus, many users will
may prefer to use the sampsize.fdr function instead of n.fdr.</p>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>a sample size estimate</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>the p-value cut-off</p>
</td></tr>
<tr><td><code>fdr.hat</code></td>
<td>
<p>an approximation to the expected value of the FDR estimate given n </p>
</td></tr>
<tr><td><code>ave.pow</code></td>
<td>
<p>the average power</p>
</td></tr>
<tr><td><code>fdr.act</code></td>
<td>
<p>the actual FDR given n</p>
</td></tr>
<tr><td><code>pi.hat</code></td>
<td>
<p>expected value of the pi.hat estimator given n</p>
</td></tr>
<tr><td><code>act.pi</code></td>
<td>
<p>actual pi0</p>
</td></tr>
</table>


<h3>References</h3>

<p>A Onar-Thomas, S Pounds. &quot;FDRsampsize: An R package to Perform Generalized Power and Sample Size Calculations for Planning Studies that use the False Discovery Rate to Measure Significance&quot;, Manuscript 2015.
</p>
<p>Pounds, Stan, and Cheng Cheng. &quot;Sample size determination for the false discovery rate.&quot; Bioinformatics 21.23 (2005): 4263-4271.
</p>
<p>Jung, Sin-Ho. &quot;Sample size for FDR-control in microarray data analysis.&quot; Bioinformatics 21.14 (2005): 3097-3104.</p>

<hr>
<h2 id='pi.star'>Compute the anticipated null proportion estimate</h2><span id='topic+pi.star'></span>

<h3>Description</h3>

<p>Compute an approximation of the expected value of the null proportion estimate given the sample size and effect size.</p>


<h3>Usage</h3>

<pre><code class='language-R'>pi.star (n, pow.func, eff.size, lam = 0.95, eps = 1e-04, ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pi.star_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="pi.star_+3A_pow.func">pow.func</code></td>
<td>
<p>an R function to compute statistical power</p>
</td></tr>
<tr><td><code id="pi.star_+3A_eff.size">eff.size</code></td>
<td>
<p>effect size vector</p>
</td></tr>
<tr><td><code id="pi.star_+3A_lam">lam</code></td>
<td>
<p>p-value at which to numerically evaluate p-value pdf (scalar)</p>
</td></tr>
<tr><td><code id="pi.star_+3A_eps">eps</code></td>
<td>
<p>epsilon for numerical differentiation</p>
</td></tr>
<tr><td><code id="pi.star_+3A_...">...</code></td>
<td>
<p>additional agruments for the functions</p>
</td></tr> </table>


<h3>Value</h3>

<p>scalar value for approximated E(pi.hat)</p>


<h3>References</h3>

<p>#&gt; Pounds, Stan, and Cheng Cheng. &quot;Sample size determination for the false discovery rate.&quot; Bioinformatics 21.23 (2005): 4263-4271.</p>

<hr>
<h2 id='power.cox'>Compute the power of a single-predictor Cox regression model</h2><span id='topic+power.cox'></span>

<h3>Description</h3>

<p>Use the formula of Hseih and Lavori (2000) to compute the power of a single-predictor Cox model.</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.cox (n, alpha, logHR, v) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.cox_+3A_n">n</code></td>
<td>
<p>number of events (scalar)</p>
</td></tr>
<tr><td><code id="power.cox_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.cox_+3A_loghr">logHR</code></td>
<td>
<p>log hazard ratio (vector)</p>
</td></tr>
<tr><td><code id="power.cox_+3A_v">v</code></td>
<td>
<p>variance of predictor variable (vector)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of power estimates for two-sided test</p>


<h3>References</h3>

<p>Hsieh, FY and Lavori, Philip W (2000) Sample-size calculations for the Cox proportional hazards regression model with nonbinary covariates.  Controlled Clinical Trials 21(6):552-560.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> power.cox             # show the power.cox function
 res=fdr.sampsize(fdr=0.1,
                  ave.pow=0.8,
                  pow.func=power.cox,
                  eff.size=rep(c(log(2),0),c(100,900)),
                  null.effect=0,
                  v=1)
 res
</code></pre>

<hr>
<h2 id='power.hart'>Compute Power for RNA-seq Experiments Assuming Negative Binomial Distribution.</h2><span id='topic+power.hart'></span>

<h3>Description</h3>

<p>Use the formula of Hart et al (2013) to compute power for comparing RNA-seq expression across two groups assuming a negative binomial distribution.</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.hart (n, alpha, log.fc, mu, sig) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.hart_+3A_n">n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.hart_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.hart_+3A_log.fc">log.fc</code></td>
<td>
<p>log fold-change (vector), usual null hypothesis is log.fc=0</p>
</td></tr>
<tr><td><code id="power.hart_+3A_mu">mu</code></td>
<td>
<p>read depth per gene (vector, same length as log.fc)</p>
</td></tr>
<tr><td><code id="power.hart_+3A_sig">sig</code></td>
<td>
<p>coefficient of variation (CV) per gene (vector, same length as log.fc)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is based on equation (1) of Hart et al (2013).  It
assumes a negative binomial model for RNA-seq read counts and 
equal sample size per group.</p>


<h3>Value</h3>

<p>vector of power estimates for the set of two-sided tests</p>


<h3>References</h3>

<p>SN Hart, TM Therneau, Y Zhang, GA Poland, and J-P Kocher (2013).  
Calculating Sample Size Estimates for RNA Sequencing Data.  
Journal of Computational Biology 20: 970-978.</p>


<h3>Examples</h3>

<pre><code class='language-R'> power.hart       # show the power function
 n.hart=2*(qnorm(0.975)+qnorm(0.9))^2*(1/20+0.6^2)/(log(2)^2) # Equation 6 of Hart et al
 power.hart(n.hart,0.05,log(2),20,0.6)                        # Recapitulate 90% power  
 res=fdr.sampsize(fdr=0.1,
                  ave.pow=0.8,
                  pow.func=power.hart,
                  eff.size=rep(c(log(2),0),c(100,900)),
                  null.effect=0,mu=5,sig=1)
 res
</code></pre>

<hr>
<h2 id='power.li'>Compute Power for RNA-Seq Experiments Assuming Poisson Distribution</h2><span id='topic+power.li'></span>

<h3>Description</h3>

<p>Use the formula of Li et al (2013) to compute power for comparing RNA-seq expression across two groups assuming the Poisson distribution.</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.li (n, alpha, rho, mu0, w = 1, type = "w") 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.li_+3A_n">n</code></td>
<td>
<p>per-group sample size</p>
</td></tr>
<tr><td><code id="power.li_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold</p>
</td></tr>  
<tr><td><code id="power.li_+3A_rho">rho</code></td>
<td>
<p>fold-change, usual null hypothesis is that rho=1</p>
</td></tr>
<tr><td><code id="power.li_+3A_mu0">mu0</code></td>
<td>
<p>average count in control group</p>
</td></tr>
<tr><td><code id="power.li_+3A_w">w</code></td>
<td>
<p>ratio of total number of </p>
</td></tr>  
<tr><td><code id="power.li_+3A_type">type</code></td>
<td>
<p>type of test: &quot;w&quot; for Wald, &quot;s&quot; for score, &quot;lw&quot; for log-transformed Wald, &quot;ls&quot; for log-transformed score.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the  power for each of a series of two-sided
tests defined by the input parameters.  The power is based on the
sample size formulas in equations 10-13 of Li et al (2013).
Also, note that the null.effect is set to 1 in the examples because
the usual null hypothesis is that the fold-change = 1.</p>


<h3>Value</h3>

<p>vector of power estimates for two-sided tests</p>


<h3>References</h3>

<p>C-I Li, P-F Su, Y Guo, and Y Shyr (2013).  
Sample size calculation for differential expression analysis of
RNA-seq data under Poisson distribution.
Int J Comput Biol Drug Des 6(4).  doi:10.1504/IJCBDD.2013.056830</p>


<h3>Examples</h3>

<pre><code class='language-R'> power.li      # show the power function
 power.li(88,0.05,1.25,5,0.5,"w")  # recapitulate 80% power in Table 1 of Li et al (2013)
 res=fdr.sampsize(fdr=0.1,
                  ave.pow=0.8,
                  pow.func=power.li,
                  eff.size=rep(c(1.5,1),c(100,900)),
                  null.effect=1,
                  mu0=5,w=1,type="w")
 res
</code></pre>

<hr>
<h2 id='power.onesampt'>Compute power of the one-sample t-test</h2><span id='topic+power.onesampt'></span>

<h3>Description</h3>

<p>Estimate power of the one-sample t-test;Uses classical power formula for one-sample t-test</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.onesampt (n, alpha, delta, sigma = 1) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.onesampt_+3A_n">n</code></td>
<td>
<p>number of events (scalar)</p>
</td></tr>
<tr><td><code id="power.onesampt_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.onesampt_+3A_delta">delta</code></td>
<td>
<p>difference of actual mean from null mean (vector)</p>
</td></tr>
<tr><td><code id="power.onesampt_+3A_sigma">sigma</code></td>
<td>
<p>standard deviation (vector or scalar, default=1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of power estimates for two-sided test</p>


<h3>Examples</h3>

<pre><code class='language-R'> power.onesampt        # show the power function
 res=fdr.sampsize(fdr=0.1,
                  ave.pow=0.8,
                  pow.func=power.onesampt,
                  eff.size=rep(c(2,0),c(100,900)),
                  null.effect=0,
                  sigma=1)
 res
</code></pre>

<hr>
<h2 id='power.oneway'>Compute power of one-way ANOVA</h2><span id='topic+power.oneway'></span>

<h3>Description</h3>

<p>Compute power of one-way ANOVA;Uses classical power formula for ANOVA;Assumes equal variance and sample size </p>


<h3>Usage</h3>

<pre><code class='language-R'>power.oneway (n, alpha, theta, k = 2) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.oneway_+3A_n">n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.oneway_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.oneway_+3A_theta">theta</code></td>
<td>
<p>sum of ((group mean - overall mean)/stdev)^2 across all groups for each hypothesis test (vector)</p>
</td></tr>
<tr><td><code id="power.oneway_+3A_k">k</code></td>
<td>
<p>the number of groups to be compared, default k=2</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For many applications, the null effect is zero for the parameter theta described above.</p>


<h3>Value</h3>

<p>vector of power estimates for test of equal means</p>


<h3>Examples</h3>

<pre><code class='language-R'> power.oneway        # show the power function
 res=fdr.sampsize(fdr=0.1,
                  ave.pow=0.8,
                  pow.func=power.oneway,
                  eff.size=rep(c(2,0),c(100,900)),
                  null.effect=0,
                  k=3)
 res
</code></pre>

<hr>
<h2 id='power.ranksum'>Compute power of the rank-sum test</h2><span id='topic+power.ranksum'></span>

<h3>Description</h3>

<p>Compute power of rank-sum test;Uses formula of Noether (JASA 1987)</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.ranksum (n, alpha, p) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.ranksum_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.ranksum_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.ranksum_+3A_p">p</code></td>
<td>
<p>Pr (Y&gt;X), as in Noether (JASA 1987)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In most applications, the null effect size will be designated by p = 0.5, which indicates that
Thus, in the example below, the argument null.effect=0.5 is specified in the call to fdr.sampsize.</p>


<h3>Value</h3>

<p>vector of power estimates for two-sided tests</p>


<h3>References</h3>

<p>Noether, Gottfried E (1987) Sample size determination for some common nonparametric tests.  
Journal of the American Statistical Association, 82:645-647.</p>


<h3>Examples</h3>

<pre><code class='language-R'> power.ranksum        # show the power function
 res=fdr.sampsize(fdr=0.1,
                  ave.pow=0.8,
                  pow.func=power.ranksum,
                  eff.size=rep(c(0.8,0.5),c(100,900)),
                  null.effect=0.5)
 res
</code></pre>

<hr>
<h2 id='power.signtest'>Compute power of the sign test</h2><span id='topic+power.signtest'></span>

<h3>Description</h3>

<p>Use the Noether (1987) formula to compute the power of the sign test</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.signtest (n, alpha, p) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.signtest_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.signtest_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.signtest_+3A_p">p</code></td>
<td>
<p>Pr (Y&gt;X), as in Noether (JASA 1987)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In most applications, the null effect size will be designated by p = 0.5 instead of p = 0.
Thus, in the call to fdr.sampsize, we specify null.effect=0.5 in the example below.</p>


<h3>Value</h3>

<p>vector of power estimates for two-sided tests</p>


<h3>References</h3>

<p>Noether, Gottfried E (1987) Sample size determination for some common nonparametric tests.  
Journal of the American Statistical Association, 82:645-647.</p>


<h3>Examples</h3>

<pre><code class='language-R'> power.signtest        # show the power function
 res=fdr.sampsize(fdr=0.1,
                  ave.pow=0.8,
                  pow.func=power.signtest,
                  eff.size=rep(c(0.8,0.5),c(100,900)),
                  null.effect=0.5)
 res
</code></pre>

<hr>
<h2 id='power.tcorr'>Compute Power of the t-test for non-zero correlation</h2><span id='topic+power.tcorr'></span>

<h3>Description</h3>

<p>Estimate power of t-test for non-zero correlation;Uses classical power formula for t-test</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.tcorr (n, alpha, rho) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.tcorr_+3A_n">n</code></td>
<td>
<p>sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.tcorr_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.tcorr_+3A_rho">rho</code></td>
<td>
<p>population correlation coefficient (vector)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For many applications, the null.effect is rho=0.</p>


<h3>Value</h3>

<p>vector of power estimates for two-sided tests</p>


<h3>Examples</h3>

<pre><code class='language-R'> power.tcorr        # show the power function
 res=fdr.sampsize(fdr=0.1,
                  ave.pow=0.8,
                  pow.func=power.tcorr,
                  eff.size=rep(c(0.3,0),c(100,900)),
                  null.effect=0)
 res
</code></pre>

<hr>
<h2 id='power.twosampt'>Compute power of the two-samples t-test</h2><span id='topic+power.twosampt'></span>

<h3>Description</h3>

<p>Estimate power of the two-samples t-test;Uses classical power formula for two-sample t-test;Assumes equal variance and sample size </p>


<h3>Usage</h3>

<pre><code class='language-R'>power.twosampt (n, alpha, delta, sigma = 1) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.twosampt_+3A_n">n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td></tr>
<tr><td><code id="power.twosampt_+3A_alpha">alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td></tr>
<tr><td><code id="power.twosampt_+3A_delta">delta</code></td>
<td>
<p>difference between population means (vector)</p>
</td></tr>
<tr><td><code id="power.twosampt_+3A_sigma">sigma</code></td>
<td>
<p>standard deviation (vector or scalar)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For many applications, the null.effect is zero difference of means.</p>


<h3>Value</h3>

<p>vector of power estimates for two-sided test</p>


<h3>Examples</h3>

<pre><code class='language-R'> power.twosampt        # show the power function
 res=fdr.sampsize(fdr=0.1,
                  ave.pow=0.8,
                  pow.func=power.twosampt,
                  eff.size=rep(c(2,0),c(100,900)),
                  null.effect=0,
                  sigma=1)
 res
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
