<!DOCTYPE html><html><head><title>Help for package FFTrees</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FFTrees}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_fft_df'><p>Add an FFT definition to tree definitions</p></a></li>
<li><a href='#add_nodes'><p>Add nodes to an FFT definition</p></a></li>
<li><a href='#add_stats'><p>Add decision statistics to data (based on frequency counts of a 2x2 classification outcomes)</p></a></li>
<li><a href='#blood'><p>Blood donation data</p></a></li>
<li><a href='#breastcancer'><p>Physiological data of patients tested for breast cancer</p></a></li>
<li><a href='#car'><p>Car acceptability data</p></a></li>
<li><a href='#classtable'><p>Compute classification statistics for binary prediction and criterion (e.g.; truth) vectors</p></a></li>
<li><a href='#comp_pred'><p>Fit and predict competing classification algorithms</p></a></li>
<li><a href='#contraceptive'><p>Contraceptive use data</p></a></li>
<li><a href='#creditapproval'><p>Credit approval data</p></a></li>
<li><a href='#describe_data'><p>Describe data</p></a></li>
<li><a href='#drop_nodes'><p>Drop a node from an FFT definition</p></a></li>
<li><a href='#edit_nodes'><p>Edit nodes in an FFT definition</p></a></li>
<li><a href='#fact_clean'><p>Clean factor variables in prediction data</p></a></li>
<li><a href='#fertility'><p>Fertility data</p></a></li>
<li><a href='#FFTrees'><p>Main function to create and apply fast-and-frugal trees (FFTs)</p></a></li>
<li><a href='#fftrees_apply'><p>Apply an FFT to data and generate accuracy statistics</p></a></li>
<li><a href='#fftrees_create'><p>Create an object of class <code>FFTrees</code></p></a></li>
<li><a href='#fftrees_cuerank'><p>Calculate thresholds that optimize some statistic (goal) for cues in data</p></a></li>
<li><a href='#fftrees_define'><p>Create FFT definitions</p></a></li>
<li><a href='#fftrees_ffttowords'><p>Describe a fast-and-frugal tree (FFT) in words</p></a></li>
<li><a href='#fftrees_fitcomp'><p>Fit competitive algorithms</p></a></li>
<li><a href='#fftrees_grow_fan'><p>Grow fast-and-frugal trees (FFTs) using the <code>fan</code> algorithms</p></a></li>
<li><a href='#fftrees_ranktrees'><p>Rank FFTs by current goal</p></a></li>
<li><a href='#fftrees_threshold_factor_grid'><p>Perform a grid search over factor and return accuracy statistics for a given factor cue</p></a></li>
<li><a href='#fftrees_threshold_numeric_grid'><p>Perform a grid search over thresholds and return accuracy statistics for a given numeric cue</p></a></li>
<li><a href='#fftrees_wordstofftrees'><p>Convert a verbal description of an FFT into an <code>FFTrees</code> object</p></a></li>
<li><a href='#FFTrees.guide'><p>Open the <strong>FFTrees</strong> package guide</p></a></li>
<li><a href='#flip_exits'><p>Flip exits in an FFT definition</p></a></li>
<li><a href='#forestfires'><p>Forest fires data</p></a></li>
<li><a href='#get_best_tree'><p>Select the best tree (from current set of FFTs)</p></a></li>
<li><a href='#get_exit_type'><p>Get exit type (from a vector <code>x</code> of FFT exit descriptions)</p></a></li>
<li><a href='#get_fft_df'><p>Get FFT definitions (from an <code>FFTrees</code> object <code>x</code>)</p></a></li>
<li><a href='#heart.cost'><p>Cue costs for the heartdisease data</p></a></li>
<li><a href='#heart.test'><p>Heart disease testing data</p></a></li>
<li><a href='#heart.train'><p>Heart disease training data</p></a></li>
<li><a href='#heartdisease'><p>Heart disease data</p></a></li>
<li><a href='#inwords'><p>Provide a verbal description of an FFT</p></a></li>
<li><a href='#iris.v'><p>Iris data</p></a></li>
<li><a href='#mushrooms'><p>Mushrooms data</p></a></li>
<li><a href='#plot.FFTrees'><p>Plot an <code>FFTrees</code> object</p></a></li>
<li><a href='#predict.FFTrees'><p>Predict classification outcomes or probabilities from data</p></a></li>
<li><a href='#print.FFTrees'><p>Print basic information of fast-and-frugal trees (FFTs)</p></a></li>
<li><a href='#read_fft_df'><p>Read an FFT definition from tree definitions</p></a></li>
<li><a href='#reorder_nodes'><p>Reorder nodes in an FFT definition</p></a></li>
<li><a href='#select_nodes'><p>Select nodes from an FFT definition</p></a></li>
<li><a href='#showcues'><p>Visualize cue accuracies (as points in ROC space)</p></a></li>
<li><a href='#sonar'><p>Sonar data</p></a></li>
<li><a href='#summary.FFTrees'><p>Summarize an <code>FFTrees</code> object</p></a></li>
<li><a href='#titanic'><p>Titanic survival data</p></a></li>
<li><a href='#voting'><p>Voting data</p></a></li>
<li><a href='#wine'><p>Wine tasting data</p></a></li>
<li><a href='#write_fft_df'><p>Write an FFT definition to tree definitions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Generate, Visualise, and Evaluate Fast-and-Frugal Decision Trees</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-06-06</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hansjoerg Neth &lt;h.neth@uni.kn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Create, visualize, and test fast-and-frugal decision trees (FFTs) using the algorithms and methods described by Phillips, Neth, Woike &amp; Gaissmaier (2017), &lt;<a href="https://doi.org/10.1017%2FS1930297500006239">doi:10.1017/S1930297500006239</a>&gt;. 
    FFTs are simple and transparent decision trees for solving binary classification problems. 
    FFTs can be preferable to more complex algorithms because they require very little information, are easy to understand and communicate, and are robust against overfitting.</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>caret, rpart, randomForest, e1071, cli, dplyr, knitr,
magrittr, scales, stringr, testthat, tibble, tidyselect</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, spelling</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/publicdomain/zero/1.0/legalcode">CC0</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://CRAN.R-project.org/package=FFTrees">https://CRAN.R-project.org/package=FFTrees</a>,
<a href="https://github.com/ndphillips/FFTrees/">https://github.com/ndphillips/FFTrees/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ndphillips/FFTrees/issues">https://github.com/ndphillips/FFTrees/issues</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-05 22:02:01 UTC; hneth</td>
</tr>
<tr>
<td>Author:</td>
<td>Nathaniel Phillips
    <a href="https://orcid.org/0000-0002-8969-7013"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Hansjoerg Neth <a href="https://orcid.org/0000-0001-5427-3141"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Jan Woike <a href="https://orcid.org/0000-0002-6816-121X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Wolfgang Gaissmaier
    <a href="https://orcid.org/0000-0001-6273-178X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-05 23:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_fft_df'>Add an FFT definition to tree definitions</h2><span id='topic+add_fft_df'></span>

<h3>Description</h3>

<p><code>add_fft_df</code> adds the definition(s) of
one or more FFT(s) (in the multi-line format of an <code>FFTrees</code> object)
or a single FFT (as a tidy data frame)
to the multi-line FFT definitions of an <code>FFTrees</code> object.
</p>
<p><code>add_fft_df</code> allows for collecting and combining
(sets of) tree definitions after
manipulating them with other tree trimming functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_fft_df(fft, ffts_df = NULL, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_fft_df_+3A_fft">fft</code></td>
<td>
<p>A (set of) FFT definition(s)
(in the multi-line format of an <code>FFTrees</code> object)
or one FFT definition
(as a data frame in tidy format, with one row per node).</p>
</td></tr>
<tr><td><code id="add_fft_df_+3A_ffts_df">ffts_df</code></td>
<td>
<p>A set of FFT definitions (as a data frame,
usually from an <code>FFTrees</code> object,
with suitable variable names to pass <code>verify_ffts_df</code>.
Default: <code>ffts_df = NULL</code>.</p>
</td></tr>
<tr><td><code id="add_fft_df_+3A_quiet">quiet</code></td>
<td>
<p>Hide feedback messages (as logical)?
Default: <code>quiet = FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A (set of) FFT definition(s) in the one line
FFT definition format used by an <code>FFTrees</code> object
(as a data frame).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_fft_df">get_fft_df</a></code> for getting the FFT definitions of an <code>FFTrees</code> object;
<code><a href="#topic+read_fft_df">read_fft_df</a></code> for reading one FFT definition from tree definitions;
<code><a href="#topic+write_fft_df">write_fft_df</a></code> for writing one FFT to tree definitions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other tree definition and manipulation functions: 
<code><a href="#topic+add_nodes">add_nodes</a>()</code>,
<code><a href="#topic+drop_nodes">drop_nodes</a>()</code>,
<code><a href="#topic+edit_nodes">edit_nodes</a>()</code>,
<code><a href="#topic+flip_exits">flip_exits</a>()</code>,
<code><a href="#topic+get_fft_df">get_fft_df</a>()</code>,
<code><a href="#topic+read_fft_df">read_fft_df</a>()</code>,
<code><a href="#topic+reorder_nodes">reorder_nodes</a>()</code>,
<code><a href="#topic+select_nodes">select_nodes</a>()</code>,
<code><a href="#topic+write_fft_df">write_fft_df</a>()</code>
</p>

<hr>
<h2 id='add_nodes'>Add nodes to an FFT definition</h2><span id='topic+add_nodes'></span>

<h3>Description</h3>

<p><code>add_nodes</code> allows adding
one or more <code>nodes</code> to an existing FFT definition
(in the tidy data frame format).
</p>
<p><code>add_nodes</code> allows to directly set and change the value(s) of
<code>class</code>, <code>cue</code>, <code>direction</code>, <code>threshold</code>, and <code>exit</code>,
in an FFT definition for the specified <code>nodes</code>.
</p>
<p>There is only rudimentary verification for plausible entries.
Importantly, however, as <code>add_nodes</code> is ignorant of <code>data</code>,
the values of its variables are not validated for a specific set of data.
</p>
<p>Values in <code>nodes</code> refer to their new position in the final FFT.
Duplicate values of <code>nodes</code> are ignored (and only the last
entry is used).
</p>
<p>When a new exit node is added, the exit type of a former final node
is set to the signal value (i.e., <code>exit_types[2]</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_nodes(
  fft,
  nodes = NA,
  class = NA,
  cue = NA,
  direction = NA,
  threshold = NA,
  exit = NA,
  quiet = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_nodes_+3A_fft">fft</code></td>
<td>
<p>One FFT definition
(as a data frame in tidy format, with one row per node).</p>
</td></tr>
<tr><td><code id="add_nodes_+3A_nodes">nodes</code></td>
<td>
<p>The FFT nodes to be added (as an integer vector).
Values refer to their new position in the final FFT
(i.e., after adding all <code>nodes</code> to <code>fft</code>).
Default: <code>nodes = NA</code>.</p>
</td></tr>
<tr><td><code id="add_nodes_+3A_class">class</code></td>
<td>
<p>The class values of <code>nodes</code> (as character).</p>
</td></tr>
<tr><td><code id="add_nodes_+3A_cue">cue</code></td>
<td>
<p>The cue names of <code>nodes</code> (as character).</p>
</td></tr>
<tr><td><code id="add_nodes_+3A_direction">direction</code></td>
<td>
<p>The direction values of <code>nodes</code> (as character).</p>
</td></tr>
<tr><td><code id="add_nodes_+3A_threshold">threshold</code></td>
<td>
<p>The threshold values of <code>nodes</code> (as character).</p>
</td></tr>
<tr><td><code id="add_nodes_+3A_exit">exit</code></td>
<td>
<p>The exit values of <code>nodes</code> (as values from <code>exit_types</code>).</p>
</td></tr>
<tr><td><code id="add_nodes_+3A_quiet">quiet</code></td>
<td>
<p>Hide feedback messages (as logical)?
Default: <code>quiet = FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>One FFT definition
(as a data frame in tidy format, with one row per node).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+drop_nodes">drop_nodes</a></code> for deleting nodes from an FFT definition;
<code><a href="#topic+edit_nodes">edit_nodes</a></code> for editing nodes in an FFT definition;
<code><a href="#topic+flip_exits">flip_exits</a></code> for reversing exits in an FFT definition;
<code><a href="#topic+reorder_nodes">reorder_nodes</a></code> for reordering nodes of an FFT definition;
<code><a href="#topic+select_nodes">select_nodes</a></code> for selecting nodes in an FFT definition;
<code><a href="#topic+get_fft_df">get_fft_df</a></code> for getting the FFT definitions of an <code>FFTrees</code> object;
<code><a href="#topic+read_fft_df">read_fft_df</a></code> for reading one FFT definition from tree definitions;
<code><a href="#topic+add_fft_df">add_fft_df</a></code> for adding FFTs to tree definitions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other tree definition and manipulation functions: 
<code><a href="#topic+add_fft_df">add_fft_df</a>()</code>,
<code><a href="#topic+drop_nodes">drop_nodes</a>()</code>,
<code><a href="#topic+edit_nodes">edit_nodes</a>()</code>,
<code><a href="#topic+flip_exits">flip_exits</a>()</code>,
<code><a href="#topic+get_fft_df">get_fft_df</a>()</code>,
<code><a href="#topic+read_fft_df">read_fft_df</a>()</code>,
<code><a href="#topic+reorder_nodes">reorder_nodes</a>()</code>,
<code><a href="#topic+select_nodes">select_nodes</a>()</code>,
<code><a href="#topic+write_fft_df">write_fft_df</a>()</code>
</p>

<hr>
<h2 id='add_stats'>Add decision statistics to data (based on frequency counts of a 2x2 classification outcomes)</h2><span id='topic+add_stats'></span>

<h3>Description</h3>

<p><code>add_stats</code> assumes the input of the 4 essential classification outcomes
(as frequency counts in a data frame <code>"data"</code> with variable names <code>"hi"</code>, <code>"fa"</code>, <code>"mi"</code>, and <code>"cr"</code>)
and uses them to compute various decision accuracy measures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_stats(
  data,
  correction = 0.25,
  sens.w = NULL,
  my.goal = NULL,
  my.goal.fun = NULL,
  cost.outcomes = NULL,
  cost.each = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_stats_+3A_data">data</code></td>
<td>
<p>A data frame with 4 frequency counts (as integer values, named <code>"hi"</code>, <code>"fa"</code>, <code>"mi"</code>, and <code>"cr"</code>).</p>
</td></tr>
<tr><td><code id="add_stats_+3A_correction">correction</code></td>
<td>
<p>numeric. Correction added to all counts for calculating <code>dprime</code>.
Default: <code>correction = .25</code>.</p>
</td></tr>
<tr><td><code id="add_stats_+3A_sens.w">sens.w</code></td>
<td>
<p>numeric. Sensitivity weight (for computing weighted accuracy, <code>wacc</code>).
Default: <code>sens.w = NULL</code> (to ensure that values are passed by calling function).</p>
</td></tr>
<tr><td><code id="add_stats_+3A_my.goal">my.goal</code></td>
<td>
<p>Name of an optional, user-defined goal (as character string).
Default: <code>my.goal = NULL</code>.</p>
</td></tr>
<tr><td><code id="add_stats_+3A_my.goal.fun">my.goal.fun</code></td>
<td>
<p>User-defined goal function (with 4 arguments <code>hi fa mi cr</code>).
Default: <code>my.goal.fun = NULL</code>.</p>
</td></tr>
<tr><td><code id="add_stats_+3A_cost.outcomes">cost.outcomes</code></td>
<td>
<p>list. A list of length 4 named <code>"hi"</code>, <code>"fa"</code>, <code>"mi"</code>, <code>"cr"</code>, and
specifying the costs of a hit, false alarm, miss, and correct rejection, respectively.
E.g.; <code>cost.outcomes = listc("hi" = 0, "fa" = 10, "mi" = 20, "cr" = 0)</code> means that a
false alarm and miss cost 10 and 20 units, respectively, while correct decisions incur no costs.
Default: <code>cost.outcomes = NULL</code> (to ensure that values are passed by calling function).</p>
</td></tr>
<tr><td><code id="add_stats_+3A_cost.each">cost.each</code></td>
<td>
<p>numeric. An optional fixed cost added to all outputs (e.g., the cost of using the cue).
Default: <code>cost.each = NULL</code> (to ensure that values are passed by calling function).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Providing numeric values for <code>cost.each</code> (as a vector) and <code>cost.outcomes</code> (as a named list)
allows computing cost information for the counts of corresponding classification decisions.
</p>


<h3>Value</h3>

<p>A data frame with variables of computed accuracy and cost measures (but dropping inputs).
</p>

<hr>
<h2 id='blood'>Blood donation data</h2><span id='topic+blood'></span>

<h3>Description</h3>

<p>Data taken from the Blood Transfusion Service Center in Hsin-Chu City in Taiwan
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blood
</code></pre>


<h3>Format</h3>

<p>A data frame containing 748 rows and 5 columns.
</p>

<dl>
<dt>recency</dt><dd><p>Months since last donation</p>
</dd>
<dt>frequency</dt><dd><p>Total number of donations</p>
</dd>
<dt>total</dt><dd><p>Total blood donated (in c.c.)</p>
</dd>
<dt>time</dt><dd><p>Months since first donation</p>
</dd>
<dt>donation.crit</dt><dd><p><em>Criterion</em>: Did the person donate blood (in March 2007)?
</p>
<p>Values: <code>0</code>/no vs. <code>1</code>/yes (76.2% vs.\ 23.8%).</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Blood+Transfusion+Service+Center">https://archive.ics.uci.edu/ml/datasets/Blood+Transfusion+Service+Center</a>
</p>
<p>Original owner and donor:
</p>
<p>Prof. I-Cheng Yeh
</p>
<p>Department of Information Management
</p>
<p>Chung-Hua University
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='breastcancer'>Physiological data of patients tested for breast cancer</h2><span id='topic+breastcancer'></span>

<h3>Description</h3>

<p>Physiological data of patients tested for breast cancer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>breastcancer
</code></pre>


<h3>Format</h3>

<p>A data frame containing 699 patients (rows) and 9 variables (columns).
</p>

<dl>
<dt>thickness</dt><dd><p>Clump Thickness</p>
</dd>
<dt>cellsize.unif</dt><dd><p>Uniformity of Cell Size</p>
</dd>
<dt>cellshape.unif</dt><dd><p>Uniformity of Cell Shape</p>
</dd>
<dt>adhesion</dt><dd><p>Marginal Adhesion</p>
</dd>
<dt>epithelial</dt><dd><p>Single Epithelial Cell Size</p>
</dd>
<dt>nuclei.bare</dt><dd><p>Bare Nuclei</p>
</dd>
<dt>chromatin</dt><dd><p>Bland Chromatin</p>
</dd>
<dt>nucleoli</dt><dd><p>Normal Nucleoli</p>
</dd>
<dt>mitoses</dt><dd><p>Mitoses</p>
</dd>
<dt>diagnosis</dt><dd><p><em>Criterion</em>: Absence/presence of breast cancer.
</p>
<p>Values: <code>FALSE</code> vs. <code>TRUE</code> (65.0% vs.\ 35.0%).</p>
</dd>
</dl>



<h3>Details</h3>

<p>We made the following enhancements to the original data for improved usability:
</p>

<ul>
<li><p>The ID number of the cases was excluded.
</p>
</li>
<li><p>The numeric criterion with value &quot;2&quot; for benign and &quot;4&quot; for malignant was converted to logical TRUE/FALSE.
</p>
</li>
<li><p>16 cases were excluded because they contained NAs.
</p>
</li></ul>

<p>Other than that, the data remains consistent with the original dataset.
</p>


<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+(Original)">https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+(Original)</a>
</p>
<p>Original creator:
</p>
<p>Dr. William H. Wolberg (physician)
</p>
<p>University of Wisconsin Hospitals
</p>
<p>Madison, Wisconsin, USA
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='car'>Car acceptability data</h2><span id='topic+car'></span>

<h3>Description</h3>

<p>A dataset on car evaluations based on basic features, derived from a simple hierarchical decision model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>car
</code></pre>


<h3>Format</h3>

<p>A data frame containing 1728 cars (rows) and 7 variables (columns).
</p>

<dl>
<dt>buying.price</dt><dd><p>price for buying the car, Factor (high, low, med, vhigh)</p>
</dd>
<dt>maint.price</dt><dd><p>price of the maintenance, Factor (high, low, med, vhigh)</p>
</dd>
<dt>doors</dt><dd><p>number of doors, Factor (2, 3, 4, 5more)</p>
</dd>
<dt>persons</dt><dd><p>capacity in terms of persons to carry, Factor (2, 4, more)</p>
</dd>
<dt>luggage</dt><dd><p>the size of luggage boot, Factor (big, med, small)</p>
</dd>
<dt>safety</dt><dd><p>estimated safety of the car, Factor (high, low, med)</p>
</dd>
<dt>acceptability</dt><dd><p><em>Criterion</em>: Category of acceptability rating.
</p>
<p>Values: <code>unacc</code>/ <code>vgood</code>/ <code>good</code>/ <code>acc</code></p>
</dd>
</dl>



<h3>Details</h3>

<p>The criterion variable is a car's <code>acceptability</code> rating.
</p>
<p>The <em>criterion</em> for this dataset has not yet been binarized. Before using it with an <em>FFTree</em>, this necessary prerequisite step should be completed based on individual preferences.
</p>


<h3>Source</h3>

<p><a href="http://archive.ics.uci.edu/ml/datasets/Car+Evaluation">http://archive.ics.uci.edu/ml/datasets/Car+Evaluation</a>
</p>
<p>Original creator and donor:
</p>
<p>Marko Bohanec and Blaz Zupan
</p>


<h3>References</h3>

<p>Bohanec, M., Rajkovic, V. (1990): Expert system for decision making.
<em>Sistemica</em>, <em>1</em> (1), 145&ndash;157.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='classtable'>Compute classification statistics for binary prediction and criterion (e.g.; truth) vectors</h2><span id='topic+classtable'></span>

<h3>Description</h3>

<p>The main input are 2 logical vectors of prediction and criterion values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>classtable(
  prediction_v = NULL,
  criterion_v = NULL,
  correction = 0.25,
  sens.w = NULL,
  cost.outcomes = NULL,
  cost_v = NULL,
  my.goal = NULL,
  my.goal.fun = NULL,
  quiet_mis = FALSE,
  na_prediction_action = "ignore"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="classtable_+3A_prediction_v">prediction_v</code></td>
<td>
<p>logical. A logical vector of predictions.</p>
</td></tr>
<tr><td><code id="classtable_+3A_criterion_v">criterion_v</code></td>
<td>
<p>logical. A logical vector of (TRUE) criterion values.</p>
</td></tr>
<tr><td><code id="classtable_+3A_correction">correction</code></td>
<td>
<p>numeric. Correction added to all counts for calculating <code>dprime</code>.
Default: <code>correction = .25</code>.</p>
</td></tr>
<tr><td><code id="classtable_+3A_sens.w">sens.w</code></td>
<td>
<p>numeric. Sensitivity weight parameter (from 0 to 1, for computing <code>wacc</code>).
Default: <code>sens.w = NULL</code> (to ensure that values are passed by calling function).</p>
</td></tr>
<tr><td><code id="classtable_+3A_cost.outcomes">cost.outcomes</code></td>
<td>
<p>list. A list of length 4 with names 'hi', 'fa', 'mi', and 'cr' specifying
the costs of a hit, false alarm, miss, and correct rejection, respectively.
For instance, <code>cost.outcomes = listc("hi" = 0, "fa" = 10, "mi" = 20, "cr" = 0)</code> means that
a false alarm and miss cost 10 and 20, respectively, while correct decisions have no cost.
Default: <code>cost.outcomes = NULL</code> (to ensure that values are passed by calling function).</p>
</td></tr>
<tr><td><code id="classtable_+3A_cost_v">cost_v</code></td>
<td>
<p>numeric. Additional cost value of each decision (as an optional vector of numeric values).
Typically used to include the cue cost of each decision (as a constant for the current level of an FFT).
Default: <code>cost_v = NULL</code> (to ensure that values are passed by calling function).</p>
</td></tr>
<tr><td><code id="classtable_+3A_my.goal">my.goal</code></td>
<td>
<p>Name of an optional, user-defined goal (as character string). Default: <code>my.goal = NULL</code>.</p>
</td></tr>
<tr><td><code id="classtable_+3A_my.goal.fun">my.goal.fun</code></td>
<td>
<p>User-defined goal function (with 4 arguments <code>hi fa mi cr</code>). Default: <code>my.goal.fun = NULL</code>.</p>
</td></tr>
<tr><td><code id="classtable_+3A_quiet_mis">quiet_mis</code></td>
<td>
<p>A logical value passed to hide/show <code>NA</code> user feedback
(usually <code>x$params$quiet$mis</code> of the calling function).
Default: <code>quiet_mis = FALSE</code> (i.e., show user feedback).</p>
</td></tr>
<tr><td><code id="classtable_+3A_na_prediction_action">na_prediction_action</code></td>
<td>
<p>What happens when no prediction is possible? (Experimental and currently unused.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The primary confusion matrix is computed by <code><a href="caret.html#topic+confusionMatrix">confusionMatrix</a></code> of the <strong>caret</strong> package.
</p>

<hr>
<h2 id='comp_pred'>Fit and predict competing classification algorithms</h2><span id='topic+comp_pred'></span>

<h3>Description</h3>

<p><code>comp_pred</code> provides a wrapper for running (i.e., fit or predict)
alternative classification algorithms to data
(i.e., <code>data.train</code> or <code>data.test</code>, respectively).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comp_pred(
  formula,
  data.train,
  data.test = NULL,
  algorithm = NULL,
  model = NULL,
  sens.w = NULL,
  new.factors = "exclude",
  quiet_mis = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comp_pred_+3A_formula">formula</code></td>
<td>
<p>A formula (usually <code>x$formula</code>, for an <code>FFTrees</code> object <code>x</code>).</p>
</td></tr>
<tr><td><code id="comp_pred_+3A_data.train">data.train</code></td>
<td>
<p>A training dataset (as a data frame).</p>
</td></tr>
<tr><td><code id="comp_pred_+3A_data.test">data.test</code></td>
<td>
<p>A testing dataset (as a data frame).</p>
</td></tr>
<tr><td><code id="comp_pred_+3A_algorithm">algorithm</code></td>
<td>
<p>A character string specifying an algorithm in the set:
</p>

<ul>
<li><p><code>"lr"</code>: Logistic regression (using <code><a href="stats.html#topic+glm">glm</a></code> from <strong>stats</strong> with <code>family = "binomial"</code>);
</p>
</li>
<li><p><code>"rlr"</code>: Regularized logistic regression (currently not supported);
</p>
</li>
<li><p><code>"cart"</code>: Decision trees (using <code>rpart</code> from <strong>rpart</strong>);
</p>
</li>
<li><p><code>"svm"</code>: Support vector machines (using <code>svm</code> from <strong>e1071</strong>);
</p>
</li>
<li><p><code>"rf"</code>: Random forests (using <code>randomForest</code> from <strong>randomForest</strong>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="comp_pred_+3A_model">model</code></td>
<td>
<p>An optional existing model (as a <code>model</code>), to be applied to the test data.</p>
</td></tr>
<tr><td><code id="comp_pred_+3A_sens.w">sens.w</code></td>
<td>
<p>Sensitivity weight parameter (numeric, from <code>0</code> to <code>1</code>), required to compute <code>wacc</code>.</p>
</td></tr>
<tr><td><code id="comp_pred_+3A_new.factors">new.factors</code></td>
<td>
<p>What should be done if new factor values are discovered in the test set (as a character string)?
Available options:
</p>

<ul>
<li><p><code>"exclude"</code>: exclude case (i.e., remove these cases, used by default);
</p>
</li>
<li><p><code>"base"</code>: predict the base rate of the criterion.
</p>
</li></ul>
</td></tr>
<tr><td><code id="comp_pred_+3A_quiet_mis">quiet_mis</code></td>
<td>
<p>A logical value passed to hide/show <code>NA</code> user feedback
(usually <code>x$params$quiet$mis</code> of the calling function).
Default: <code>quiet_mis = FALSE</code> (i.e., show user feedback).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The range of competing algorithms currently available includes
logistic regression (<code>stats::glm</code>),
CART (<code>rpart::rpart</code>),
support vector machines (<code>e1071::svm</code>), and
random forests (<code>randomForest::randomForest</code>).
</p>
<p>The current support for handling missing data (or <code>NA</code> values) is only rudimentary.
When enabled (via the global options <code>allow_NA_pred</code> or <code>allow_NA_crit</code>),
any rows in <code>data.train</code> or <code>data.test</code> with incomplete cases are being removed
prior to fitting or predicting a model (by using <code>na.omit</code> from <strong>stats</strong>).
See the specifications of each model for more sophisticated ways of handling missing data.
</p>

<hr>
<h2 id='contraceptive'>Contraceptive use data</h2><span id='topic+contraceptive'></span>

<h3>Description</h3>

<p>A subset of the 1987 National Indonesia Contraceptive Prevalence Survey.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contraceptive
</code></pre>


<h3>Format</h3>

<p>A data frame containing 1473 cases (rows) and 10 variables (columns).
</p>

<dl>
<dt>wife.age</dt><dd><p>Wife's age, Numeric</p>
</dd>
<dt>wife.edu</dt><dd><p>Wife's education, Nummeric, (1=low, 2, 3, 4=high)</p>
</dd>
<dt>hus.ed</dt><dd><p>Husband's education, Nummeric, (1=low, 2, 3, 4=high)</p>
</dd>
<dt>children</dt><dd><p>Number of children ever born, Numeric</p>
</dd>
<dt>wife.rel</dt><dd><p>Wife's religion, Numeric, (0=Non-Islam, 1=Islam)</p>
</dd>
<dt>wife.work</dt><dd><p>Wife's now working?, Nummeric, (0=Yes, 1=No)</p>
</dd>
<dt>hus.occ</dt><dd><p>Husband's occupation, Nummeric, (1, 2, 3, 4)</p>
</dd>
<dt>sol</dt><dd><p>Standard-of-living index, Nummeric, (1=low, 2, 3, 4=high)</p>
</dd>
<dt>media</dt><dd><p>Media exposure, Numeric, (0=Good, 1=Not good)</p>
</dd>
<dt>cont.crit</dt><dd><p><em>Criterion</em>: Use of a contraceptive (as logical).
</p>
<p>Values: <code>FALSE</code> vs. <code>TRUE</code> (42.7% vs. 57.3%).</p>
</dd>
</dl>



<h3>Details</h3>

<p>The samples describe married women who were either not pregnant
or do not know if they were pregnant at the time of the interview.
</p>
<p>The problem consists in predicting a woman's current contraceptive method choice
(here: binarized <code>cont.crit</code>)
based on her demographic and socio-economic characteristics.
</p>
<p>We made the following enhancements to the original data for improved usability:
</p>

<ul>
<li><p>The criterion was binarized from a class attribute variable with three levels (<code>1=No-use</code>, <code>2=Long-term</code>, <code>3=Short-term</code>) , into a logical variable with two levels (<code>TRUE</code> vs. <code>FALSE</code>).
</p>
</li></ul>

<p>Other than that, the data remains consistent with the original dataset.
</p>


<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Contraceptive+Method+Choice">https://archive.ics.uci.edu/ml/datasets/Contraceptive+Method+Choice</a>
</p>
<p>Original creator and donor:
</p>
<p>Tjen-Sien Lim
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='creditapproval'>Credit approval data</h2><span id='topic+creditapproval'></span>

<h3>Description</h3>

<p>This data reports predictors and the result of credit card applications.
Its attribute names and values have been changed to symbols to protect confidentiality.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>creditapproval
</code></pre>


<h3>Format</h3>

<p>A data frame containing 690 cases (rows) and 15 variables (columns).
</p>

<dl>
<dt>c.1</dt><dd><p>categorical: b, a</p>
</dd>
<dt>c.2</dt><dd><p>continuous</p>
</dd>
<dt>c.3</dt><dd><p>continuous</p>
</dd>
<dt>c.4</dt><dd><p>categorical: u, y, l, t</p>
</dd>
<dt>c.5</dt><dd><p>categorical: g, p, gg</p>
</dd>
<dt>c.6</dt><dd><p>categorical: c, d, cc, i, j, k, m, r, q, w, x, e, aa, ff</p>
</dd>
<dt>c.7</dt><dd><p>categorical: v, h, bb, j, n, z, dd, ff, o</p>
</dd>
<dt>c.8</dt><dd><p>continuous</p>
</dd>
<dt>c.9</dt><dd><p>categorical: t, f</p>
</dd>
<dt>c.10</dt><dd><p>categorical: t, f</p>
</dd>
<dt>c.11</dt><dd><p>continuous</p>
</dd>
<dt>c.12</dt><dd><p>categorical: t, f</p>
</dd>
<dt>c.13</dt><dd><p>categorical: g, p, s</p>
</dd>
<dt>c.14</dt><dd><p>continuous</p>
</dd>
<dt>c.15</dt><dd><p>continuous</p>
</dd>
<dt>crit</dt><dd><p><em>Criterion</em>: Credit approval.
</p>
<p>Values: <code>TRUE</code> (+)  vs. <code>FALSE</code> (-) (44.5% vs. 55.5%).</p>
</dd>
</dl>



<h3>Details</h3>

<p>This dataset contains a mix of attributes &ndash; continuous, nominal with small Ns,
and nominal with larger Ns. There are also a few missing values.
</p>
<p>We made the following enhancements to the original data for improved usability:
</p>

<ul>
<li><p>Any missing values, denoted as &quot;?&quot; in the dataset, were transformed into NAs.
</p>
</li>
<li><p>Binary factor variables with exclusive &quot;t&quot; and &quot;f&quot; values were converted to logical TRUE/FALSE vectors.
</p>
</li></ul>

<p>Other than that, the data remains consistent with the original dataset.
</p>


<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Credit+Approval">https://archive.ics.uci.edu/ml/datasets/Credit+Approval</a>
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='describe_data'>Describe data</h2><span id='topic+describe_data'></span>

<h3>Description</h3>

<p>Calculate key descriptive statistics for a given set of data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>describe_data(data, data_name, criterion_name, baseline_value)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="describe_data_+3A_data">data</code></td>
<td>
<p>A data frame with a criterion variable <code>criterion_name</code>.</p>
</td></tr>
<tr><td><code id="describe_data_+3A_data_name">data_name</code></td>
<td>
<p>A character string specifying a name for the data.</p>
</td></tr>
<tr><td><code id="describe_data_+3A_criterion_name">criterion_name</code></td>
<td>
<p>A character string specifying the criterion name.</p>
</td></tr>
<tr><td><code id="describe_data_+3A_baseline_value">baseline_value</code></td>
<td>
<p>The value in <code>criterion_name</code> denoting the baseline
(e.g., <code>TRUE</code> or <code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the descriptive statistics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(heartdisease)
describe_data(heartdisease, "heartdisease",
              criterion_name = "diagnosis",
              baseline_value = TRUE)

</code></pre>

<hr>
<h2 id='drop_nodes'>Drop a node from an FFT definition</h2><span id='topic+drop_nodes'></span>

<h3>Description</h3>

<p><code>drop_nodes</code> deletes
one or more <code>nodes</code> from an existing FFT definition
(by removing the corresponding rows from the FFT definition
in the tidy data frame format).
</p>
<p>When dropping the final node,
the last remaining node becomes the new final node
(i.e., gains a second exit).
</p>
<p>Duplicates in <code>nodes</code> are dropped only once
(rather than incrementally) and <code>nodes</code> not in
the range <code>1:nrow(fft)</code> are ignored.
Dropping all nodes yields an error.
</p>
<p><code>drop_nodes</code> is the inverse function of <code><a href="#topic+select_nodes">select_nodes</a></code>.
Inserting new nodes is possible by <code><a href="#topic+add_nodes">add_nodes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drop_nodes(fft, nodes = NA, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drop_nodes_+3A_fft">fft</code></td>
<td>
<p>One FFT definition
(as a data frame in tidy format, with one row per node).</p>
</td></tr>
<tr><td><code id="drop_nodes_+3A_nodes">nodes</code></td>
<td>
<p>The FFT nodes to drop (as an integer vector).
Default: <code>nodes = NA</code>.</p>
</td></tr>
<tr><td><code id="drop_nodes_+3A_quiet">quiet</code></td>
<td>
<p>Hide feedback messages (as logical)?
Default: <code>quiet = FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>One FFT definition
(as a data frame in tidy format, with one row per node).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_nodes">add_nodes</a></code> for adding nodes to an FFT definition;
<code><a href="#topic+edit_nodes">edit_nodes</a></code> for editing nodes in an FFT definition;
<code><a href="#topic+select_nodes">select_nodes</a></code> for selecting nodes in an FFT definition;
<code><a href="#topic+get_fft_df">get_fft_df</a></code> for getting the FFT definitions of an <code>FFTrees</code> object;
<code><a href="#topic+read_fft_df">read_fft_df</a></code> for reading one FFT definition from tree definitions;
<code><a href="#topic+add_fft_df">add_fft_df</a></code> for adding FFTs to tree definitions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other tree definition and manipulation functions: 
<code><a href="#topic+add_fft_df">add_fft_df</a>()</code>,
<code><a href="#topic+add_nodes">add_nodes</a>()</code>,
<code><a href="#topic+edit_nodes">edit_nodes</a>()</code>,
<code><a href="#topic+flip_exits">flip_exits</a>()</code>,
<code><a href="#topic+get_fft_df">get_fft_df</a>()</code>,
<code><a href="#topic+read_fft_df">read_fft_df</a>()</code>,
<code><a href="#topic+reorder_nodes">reorder_nodes</a>()</code>,
<code><a href="#topic+select_nodes">select_nodes</a>()</code>,
<code><a href="#topic+write_fft_df">write_fft_df</a>()</code>
</p>

<hr>
<h2 id='edit_nodes'>Edit nodes in an FFT definition</h2><span id='topic+edit_nodes'></span>

<h3>Description</h3>

<p><code>edit_nodes</code> allows manipulating
one or more <code>nodes</code> from an existing FFT definition
(in the tidy data frame format).
</p>
<p><code>edit_nodes</code> allows to directly set and change the value(s) of
<code>class</code>, <code>cue</code>, <code>direction</code>, <code>threshold</code>, and <code>exit</code>,
in an FFT definition for the specified <code>nodes</code>.
</p>
<p>There is only rudimentary verification for plausible entries.
Importantly, however, as <code>edit_nodes</code> is ignorant of <code>data</code>,
the values of its variables are not validated for a specific set of data.
</p>
<p>Repeated changes of a node are possible
(by repeating the corresponding integer value in <code>nodes</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edit_nodes(
  fft,
  nodes = NA,
  class = NA,
  cue = NA,
  direction = NA,
  threshold = NA,
  exit = NA,
  quiet = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edit_nodes_+3A_fft">fft</code></td>
<td>
<p>One FFT definition
(as a data frame in tidy format, with one row per node).</p>
</td></tr>
<tr><td><code id="edit_nodes_+3A_nodes">nodes</code></td>
<td>
<p>The FFT nodes to be edited (as an integer vector).
Default: <code>nodes = NA</code>.</p>
</td></tr>
<tr><td><code id="edit_nodes_+3A_class">class</code></td>
<td>
<p>The class values of <code>nodes</code> (as character).</p>
</td></tr>
<tr><td><code id="edit_nodes_+3A_cue">cue</code></td>
<td>
<p>The cue names of <code>nodes</code> (as character).</p>
</td></tr>
<tr><td><code id="edit_nodes_+3A_direction">direction</code></td>
<td>
<p>The direction values of <code>nodes</code> (as character).</p>
</td></tr>
<tr><td><code id="edit_nodes_+3A_threshold">threshold</code></td>
<td>
<p>The threshold values of <code>nodes</code> (as character).</p>
</td></tr>
<tr><td><code id="edit_nodes_+3A_exit">exit</code></td>
<td>
<p>The exit values of <code>nodes</code> (as values from <code>exit_types</code>).</p>
</td></tr>
<tr><td><code id="edit_nodes_+3A_quiet">quiet</code></td>
<td>
<p>Hide feedback messages (as logical)?
Default: <code>quiet = FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>One FFT definition
(as a data frame in tidy format, with one row per node).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_nodes">add_nodes</a></code> for adding nodes to an FFT definition;
<code><a href="#topic+drop_nodes">drop_nodes</a></code> for deleting nodes from an FFT definition;
<code><a href="#topic+flip_exits">flip_exits</a></code> for reversing exits in an FFT definition;
<code><a href="#topic+reorder_nodes">reorder_nodes</a></code> for reordering nodes of an FFT definition;
<code><a href="#topic+select_nodes">select_nodes</a></code> for selecting nodes in an FFT definition;
<code><a href="#topic+get_fft_df">get_fft_df</a></code> for getting the FFT definitions of an <code>FFTrees</code> object;
<code><a href="#topic+read_fft_df">read_fft_df</a></code> for reading one FFT definition from tree definitions;
<code><a href="#topic+add_fft_df">add_fft_df</a></code> for adding FFTs to tree definitions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other tree definition and manipulation functions: 
<code><a href="#topic+add_fft_df">add_fft_df</a>()</code>,
<code><a href="#topic+add_nodes">add_nodes</a>()</code>,
<code><a href="#topic+drop_nodes">drop_nodes</a>()</code>,
<code><a href="#topic+flip_exits">flip_exits</a>()</code>,
<code><a href="#topic+get_fft_df">get_fft_df</a>()</code>,
<code><a href="#topic+read_fft_df">read_fft_df</a>()</code>,
<code><a href="#topic+reorder_nodes">reorder_nodes</a>()</code>,
<code><a href="#topic+select_nodes">select_nodes</a>()</code>,
<code><a href="#topic+write_fft_df">write_fft_df</a>()</code>
</p>

<hr>
<h2 id='fact_clean'>Clean factor variables in prediction data</h2><span id='topic+fact_clean'></span>

<h3>Description</h3>

<p>Clean factor variables in prediction data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fact_clean(data.train, data.test, show.warning = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fact_clean_+3A_data.train">data.train</code></td>
<td>
<p>A training dataset</p>
</td></tr>
<tr><td><code id="fact_clean_+3A_data.test">data.test</code></td>
<td>
<p>A testing dataset</p>
</td></tr>
<tr><td><code id="fact_clean_+3A_show.warning">show.warning</code></td>
<td>
<p>logical</p>
</td></tr>
</table>

<hr>
<h2 id='fertility'>Fertility data</h2><span id='topic+fertility'></span>

<h3>Description</h3>

<p>This dataset describes a sample of 100 volunteers providing a semen sample that was analyzed according to the WHO 2010 criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fertility
</code></pre>


<h3>Format</h3>

<p>A data frame containing 100 rows and 10 columns.
</p>

<dl>
<dt>season</dt><dd><p>Season in which the analysis was performed. (winter, spring, summer, fall)</p>
</dd>
<dt>age</dt><dd><p>Age at the time of analysis</p>
</dd>
<dt>child.dis</dt><dd><p>Childish diseases (ie , chicken pox, measles, mumps, polio) (yes(1), no(0)) </p>
</dd>
<dt>trauma</dt><dd><p>Accident or serious trauma (yes(1), no(0))</p>
</dd>
<dt>surgery</dt><dd><p>Surgical intervention (yes(1), no(0))</p>
</dd>
<dt>fevers</dt><dd><p>High fevers in the last year (less than three months ago(-1), more than three months ago (0), no. (1))</p>
</dd>
<dt>alcohol</dt><dd><p>Frequency of alcohol consumption (several times a day, every day, several times a week, once a week, hardly ever or never)</p>
</dd>
<dt>smoking</dt><dd><p>Smoking habit (never(-1), occasional (0)) daily (1))</p>
</dd>
<dt>sitting</dt><dd><p>Number of hours spent sitting per day</p>
</dd>
<dt>diagnosis</dt><dd><p><em>Criterion</em>: Diagnosis normal (TRUE) vs. altered (FALSE) (88.0% vs.\ 22.0%).</p>
</dd>
</dl>



<h3>Details</h3>

<p>Sperm concentration are related to socio-demographic data, environmental factors, health status, and life habits.
</p>
<p>We made the following enhancements to the original data for improved usability:
</p>

<ul>
<li><p>The criterion was redefined from a factor variable with two levels
(<code>N=Normal</code>, <code>O=Altered</code>) into a logical variable (<code>TRUE</code> vs. <code>FALSE</code>).
</p>
</li></ul>

<p>Other than that, the data remains consistent with the original dataset.
</p>


<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Fertility">https://archive.ics.uci.edu/ml/datasets/Fertility</a>
</p>
<p>Original contributors:
</p>
<p>David Gil
Lucentia Research Group
Department of Computer Technology
University of Alicante
</p>
<p>Jose Luis Girela
Department of Biotechnology
University of Alicante
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='FFTrees'>Main function to create and apply fast-and-frugal trees (FFTs)</h2><span id='topic+FFTrees'></span>

<h3>Description</h3>

<p><code>FFTrees</code> is the workhorse function of the <strong>FFTrees</strong> package for creating fast-and-frugal trees (FFTs).
</p>
<p>FFTs are decision algorithms for solving binary classification tasks, i.e., they predict the values of a binary criterion variable based on 1 or multiple predictor variables (cues).
</p>
<p>Using <code>FFTrees</code> on <code>data</code> usually generates a range of FFTs and corresponding summary statistics (as an <code>FFTrees</code> object)
that can then be printed, plotted, and examined further.
</p>
<p>The criterion and predictor variables are specified in <code><a href="stats.html#topic+formula">formula</a></code> notation.
Based on the settings of <code>data</code> and <code>data.test</code>, FFTs are trained on a (required) training dataset
(given the set of current <code>goal</code> values) and evaluated on (or predict) an (optional) test dataset.
</p>
<p>If an existing <code>FFTrees</code> object <code>object</code> or <code>tree.definitions</code> are provided as inputs,
no new FFTs are created.
When both arguments are provided, <code>tree.definitions</code> take priority over the FFTs in an existing <code>object</code>.
Specifically,
</p>

<ul>
<li><p>If <code>tree.definitions</code> are provided, these are assigned to the FFTs of <code>x</code>.
</p>
</li>
<li><p>If no <code>tree.definitions</code> are provided, but an existing <code>FFTrees</code> object <code>object</code> is provided,
the trees from <code>object</code> are assigned to the FFTs of <code>x</code>.
</p>
</li></ul>

<p>Create and evaluate fast-and-frugal trees (FFTs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FFTrees(
  formula = NULL,
  data = NULL,
  data.test = NULL,
  algorithm = "ifan",
  train.p = 1,
  goal = NULL,
  goal.chase = NULL,
  goal.threshold = NULL,
  max.levels = NULL,
  numthresh.method = "o",
  numthresh.n = 10,
  repeat.cues = TRUE,
  stopping.rule = "exemplars",
  stopping.par = 0.1,
  sens.w = 0.5,
  cost.outcomes = NULL,
  cost.cues = NULL,
  main = NULL,
  decision.labels = c("False", "True"),
  my.goal = NULL,
  my.goal.fun = NULL,
  my.tree = NULL,
  object = NULL,
  tree.definitions = NULL,
  do.comp = TRUE,
  do.cart = TRUE,
  do.lr = TRUE,
  do.rf = TRUE,
  do.svm = TRUE,
  quiet = list(ini = TRUE, fin = FALSE, mis = FALSE, set = TRUE),
  comp = NULL,
  force = NULL,
  rank.method = NULL,
  rounding = NULL,
  store.data = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FFTrees_+3A_formula">formula</code></td>
<td>
<p>A formula. A <code><a href="stats.html#topic+formula">formula</a></code> specifying a binary criterion variable (as logical) as a function of 1 or more predictor variables (cues).</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_data">data</code></td>
<td>
<p>A data frame. A dataset used for training (fitting) FFTs and alternative algorithms.
<code>data</code> must contain the binary criterion variable specified in <code>formula</code> and potential predictors (which can be categorical or numeric variables).</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_data.test">data.test</code></td>
<td>
<p>A data frame. An optional dataset used for model testing (prediction) with the same structure as data.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_algorithm">algorithm</code></td>
<td>
<p>A character string. The algorithm used to create FFTs. Can be <code>'ifan'</code>, <code>'dfan'</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_train.p">train.p</code></td>
<td>
<p>numeric. What percentage of the data to use for training when <code>data.test</code> is not specified?
For example, <code>train.p = .50</code> will randomly split <code>data</code> into a 50% training set and a 50% test set.
Default: <code>train.p = 1</code> (i.e., using <em>all</em> data for training).</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_goal">goal</code></td>
<td>
<p>A character string indicating the statistic to maximize when <em>selecting trees</em>:
<code>"acc"</code> = overall accuracy, <code>"bacc"</code> = balanced accuracy, <code>"wacc"</code> = weighted accuracy,
<code>"dprime"</code> = discriminability, <code>"cost"</code> = costs (based on <code>cost.outcomes</code> and <code>cost.cues</code>).</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_goal.chase">goal.chase</code></td>
<td>
<p>A character string indicating the statistic to maximize when <em>constructing trees</em>:
<code>"acc"</code> = overall accuracy, <code>"bacc"</code> = balanced accuracy, <code>"wacc"</code> = weighted accuracy,
<code>"dprime"</code> = discriminability, <code>"cost"</code> = costs (based on <code>cost.outcomes</code> and <code>cost.cues</code>).</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_goal.threshold">goal.threshold</code></td>
<td>
<p>A character string indicating the criterion to maximize when <em>optimizing cue thresholds</em>:
<code>"acc"</code> = overall accuracy, <code>"bacc"</code> = balanced accuracy, <code>"wacc"</code> = weighted accuracy,
<code>"dprime"</code> = discriminability, <code>"cost"</code> = costs (based only on <code>cost.outcomes</code>, as <code>cost.cues</code> are constant per cue).
All default goals are set in <code><a href="#topic+fftrees_create">fftrees_create</a></code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_max.levels">max.levels</code></td>
<td>
<p>integer. The maximum number of nodes (or levels) considered for an FFT.
As all combinations of possible exit structures are considered, larger values of <code>max.levels</code> will create larger sets of FFTs.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_numthresh.method">numthresh.method</code></td>
<td>
<p>How should thresholds for numeric cues be determined (as character)?
<code>"o"</code> will optimize thresholds (for <code>goal.threshold</code>), while <code>"m"</code> will use the median.
Default: <code>numthresh.method = "o"</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_numthresh.n">numthresh.n</code></td>
<td>
<p>The number of numeric thresholds to try (as integer).
Default: <code>numthresh.n = 10</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_repeat.cues">repeat.cues</code></td>
<td>
<p>May cues occur multiple times within a tree (as logical)?
Default: <code>repeat.cues = TRUE</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_stopping.rule">stopping.rule</code></td>
<td>
<p>A character string indicating the method to stop growing trees.
Available options are:
</p>

<ul>
<li><p><code>"exemplars"</code>: A tree grows until only a small proportion of unclassified exemplars remain;
</p>
</li>
<li><p><code>"levels"</code>: A tree grows until a certain level is reached;
</p>
</li>
<li><p><code>"statdelta"</code>: A tree grows until the change in the criterion statistic <code>goal.chase</code> exceeds some threshold level.
(This setting is currently experimental and includes the first level beyond threshold.
As tree statistics can be non-monotonic, this option may yield inconsistent results.)
</p>
</li></ul>

<p>All stopping methods use <code>stopping.par</code> to set a numeric threshold value.
Default: <code>stopping.rule = "exemplars"</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_stopping.par">stopping.par</code></td>
<td>
<p>numeric. A numeric parameter indicating the criterion value for the current <code>stopping.rule</code>.
For stopping.rule <code>"levels"</code>, this is the number of desired levels (as an integer).
For stopping rule <code>"exemplars"</code>, this is the smallest proportion of exemplars allowed in the last level.
For stopping.rule <code>"statdelta"</code>, this is the minimum required change (in the <code>goal.chase</code> value) to include a level.
Default: <code>stopping.par = .10</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_sens.w">sens.w</code></td>
<td>
<p>A numeric value from <code>0</code> to <code>1</code> indicating how to weight
sensitivity relative to specificity when optimizing <em>weighted</em> accuracy (e.g., <code>goal = 'wacc'</code>).
Default: <code>sens.w = .50</code> (i.e., <code>wacc</code> corresponds to <code>bacc</code>).</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_cost.outcomes">cost.outcomes</code></td>
<td>
<p>A list of length 4 specifying the cost value for one of the 4 possible classification outcomes.
The list elements must be named <code>'hi'</code>, <code>'fa'</code>, <code>'mi'</code>, and <code>'cr'</code>
(for specifying the costs of a hit, false alarm, miss, and correct rejection, respectively) and provide a numeric cost value.
E.g.; <code>cost.outcomes = listc("hi" = 0, "fa" = 10, "mi" = 20, "cr" = 0)</code> imposes false alarm and miss costs of <code>10</code> and <code>20</code> units, respectively, while correct decisions have no costs.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_cost.cues">cost.cues</code></td>
<td>
<p>A list containing the cost of each cue (in some common unit).
Each list element must have a name corresponding to a cue (i.e., a variable in <code>data</code>), and should be a single (positive numeric) value.
Cues in <code>data</code> that are not present in <code>cost.cues</code> are assumed to have no costs (i.e., a cost value of <code>0</code>).</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_main">main</code></td>
<td>
<p>string. An optional label for the dataset. Passed on to other functions, like <code><a href="#topic+plot.FFTrees">plot.FFTrees</a></code>, and <code><a href="#topic+print.FFTrees">print.FFTrees</a></code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_decision.labels">decision.labels</code></td>
<td>
<p>A vector of strings of length 2 for the text labels for negative and positive decision/prediction outcomes
(i.e., left vs. right, noise vs. signal, 0 vs. 1, respectively, as character).
E.g.; <code>decision.labels = c("Healthy", "Diseased")</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_my.goal">my.goal</code></td>
<td>
<p>The name of an optimization measure defined by <code>my.goal.fun</code> (as a character string).
Example: <code>my.goal = "my_acc"</code> (see <code>my.goal.fun</code> for corresponding function).
Default: <code>my.goal = NULL</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_my.goal.fun">my.goal.fun</code></td>
<td>
<p>The definition of an outcome measure to optimize, defined as a function
of the frequency counts of the 4 basic classification outcomes <code>hi, fa, mi, cr</code>
(i.e., an R function with 4 arguments <code>hi, fa, mi, cr</code>).
Example: <code>my.goal.fun = function(hi, fa, mi, cr){(hi + cr)/(hi + fa + mi + cr)}</code> (i.e., accuracy).
Default: <code>my.goal.fun = NULL</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_my.tree">my.tree</code></td>
<td>
<p>A verbal description of an FFT, i.e., an &quot;FFT in words&quot; (as character string).
For example, <code>my.tree = "If age &gt; 20, predict TRUE. If sex = {m}, predict FALSE. Otherwise, predict TRUE."</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_object">object</code></td>
<td>
<p>An optional existing <code>FFTrees</code> object.
When specified, no new FFTs are fitted, but existing trees are applied to <code>data</code> and <code>data.test</code>.
When <code>formula</code>, <code>data</code> or <code>data.test</code> are not specified, the current values of <code>object</code> are used.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_tree.definitions">tree.definitions</code></td>
<td>
<p>An optional <code>data.frame</code> of hard-coded FFT definitions (in the format of <code>x$trees$definitions</code> of an <code>FFTrees</code> object <code>x</code>).
If specified, no new FFTs are being fitted (i.e., <code>algorithm</code> and functions for evaluating cues and creating FFTs are skipped).
Instead, the tree definitions provided are used to re-evaluate the current <code>FFTrees</code> object on current data.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_do.comp">do.comp</code>, <code id="FFTrees_+3A_do.lr">do.lr</code>, <code id="FFTrees_+3A_do.cart">do.cart</code>, <code id="FFTrees_+3A_do.svm">do.svm</code>, <code id="FFTrees_+3A_do.rf">do.rf</code></td>
<td>
<p>Should alternative algorithms be used for comparison (as logical)?
All options are set to <code>TRUE</code> by default. Available options correspond to:
</p>

<ul>
<li><p><code>do.lr</code>: Logistic regression (LR, using <code><a href="stats.html#topic+glm">glm</a></code> from <strong>stats</strong> with <code>family = "binomial"</code>);
</p>
</li>
<li><p><code>do.cart</code>: Classification and regression trees (CART, using <code>rpart</code> from <strong>rpart</strong>);
</p>
</li>
<li><p><code>do.svm</code>: Support vector machines (SVM, using <code>svm</code> from <strong>e1071</strong>);
</p>
</li>
<li><p><code>do.rf</code>: Random forests (RF, using <code>randomForest</code> from <strong>randomForest</strong>.
</p>
</li></ul>

<p>Specifying <code>do.comp = FALSE</code> sets all available options to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_quiet">quiet</code></td>
<td>
<p>A list of 4 logical arguments: Should detailed progress reports be suppressed?
Setting list elements to <code>FALSE</code> is helpful when diagnosing errors.
Default: <code>quiet = list(ini = TRUE, fin = FALSE, mis = FALSE, set = TRUE)</code>,
for initial vs. final steps, missing cases, and parameter settings, respectively.
Providing a single logical value sets all elements to <code>TRUE</code> or <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="FFTrees_+3A_comp">comp</code>, <code id="FFTrees_+3A_force">force</code>, <code id="FFTrees_+3A_rank.method">rank.method</code>, <code id="FFTrees_+3A_rounding">rounding</code>, <code id="FFTrees_+3A_store.data">store.data</code>, <code id="FFTrees_+3A_verbose">verbose</code></td>
<td>
<p>Deprecated arguments (unused or replaced, to be retired in future releases).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>FFTrees</code> object with the following elements:
</p>

<dl>
<dt>criterion_name</dt><dd><p>The name of the binary criterion variable (as character).</p>
</dd>
<dt>cue_names</dt><dd><p>The names of all potential predictor variables (cues) in the data (as character).</p>
</dd>
<dt>formula</dt><dd><p>The <code><a href="stats.html#topic+formula">formula</a></code> specified when creating the FFTs.</p>
</dd>
<dt>trees</dt><dd><p>A list of FFTs created, with further details contained in <code>n</code>, <code>best</code>, <code>definitions</code>, <code>inwords</code>, <code>stats</code>, <code>level_stats</code>, and <code>decisions</code>.</p>
</dd>
<dt>data</dt><dd><p>The original training and test data (if available).</p>
</dd>
<dt>params</dt><dd><p>A list of defined control parameters (e.g.; <code>algorithm</code>, <code>goal</code>, <code>sens.w</code>, as well as various thresholds, stopping rule, and cost parameters).</p>
</dd>
<dt>competition</dt><dd><p>Models and classification statistics for competitive classification algorithms:
Logistic regression (<code>lr</code>), classification and regression trees (<code>cart</code>), random forests (<code>rf</code>), and support vector machines (<code>svm</code>).</p>
</dd>
<dt>cues</dt><dd><p>A list of cue information, with further details contained in <code>thresholds</code> and <code>stats</code>.</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+print.FFTrees">print.FFTrees</a></code> for printing FFTs;
<code><a href="#topic+plot.FFTrees">plot.FFTrees</a></code> for plotting FFTs;
<code><a href="#topic+summary.FFTrees">summary.FFTrees</a></code> for summarizing FFTs;
<code><a href="#topic+inwords">inwords</a></code> for obtaining a verbal description of FFTs;
<code><a href="#topic+showcues">showcues</a></code> for plotting cue accuracies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# 1. Create fast-and-frugal trees (FFTs) for heart disease:
heart.fft &lt;- FFTrees(formula = diagnosis ~ .,
                     data = heart.train,
                     data.test = heart.test,
                     main = "Heart Disease",
                     decision.labels = c("Healthy", "Diseased")
                     )

# 2. Print a summary of the result:
heart.fft  # same as:
# print(heart.fft, data = "train", tree = "best.train")

# 3. Plot an FFT applied to training data:
plot(heart.fft)  # same as:
# plot(heart.fft, what = "all", data = "train", tree = "best.train")

# 4. Apply FFT to (new) testing data:
plot(heart.fft, data = "test")            # predict for Tree 1
plot(heart.fft, data = "test", tree = 2)  # predict for Tree 2

# 5. Predict classes and probabilities for new data:
predict(heart.fft, newdata = heartdisease)
predict(heart.fft, newdata = heartdisease, type = "prob")

# 6. Create a custom tree (from verbal description) with my.tree:
custom.fft &lt;- FFTrees(
  formula = diagnosis ~ .,
  data = heartdisease,
  my.tree = "If age &lt; 50, predict False.
             If sex = 1, predict True.
             If chol &gt; 300, predict True, otherwise predict False.",
  main = "My custom FFT")

# Plot the (pretty bad) custom tree:
plot(custom.fft)

</code></pre>

<hr>
<h2 id='fftrees_apply'>Apply an FFT to data and generate accuracy statistics</h2><span id='topic+fftrees_apply'></span>

<h3>Description</h3>

<p><code>fftrees_apply</code> applies a fast-and-frugal tree (FFT, as an <code>FFTrees</code> object)
to a dataset (of type <code>mydata</code>) and generates corresponding accuracy statistics
(on cue levels and for trees).
</p>
<p><code>fftrees_apply</code> is called internally by the main <code><a href="#topic+FFTrees">FFTrees</a></code> function
(with <code>mydata = "train"</code> and &mdash; if test data exists &mdash; <code>mydata = "test"</code>).
Alternatively, <code>fftrees_apply</code> is called when predicting outcomes for new data
by <code><a href="#topic+predict.FFTrees">predict.FFTrees</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fftrees_apply(x, mydata = NULL, newdata = NULL, fin_NA_pred = "majority")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fftrees_apply_+3A_x">x</code></td>
<td>
<p>An object with FFT definitions which are to be applied to current data (as an <code>FFTrees</code> object).</p>
</td></tr>
<tr><td><code id="fftrees_apply_+3A_mydata">mydata</code></td>
<td>
<p>The type of data to which the FFT should be applied (as character, either <code>"train"</code> or <code>"test"</code>).</p>
</td></tr>
<tr><td><code id="fftrees_apply_+3A_newdata">newdata</code></td>
<td>
<p>New data to which an FFT should be applied (as a data frame).</p>
</td></tr>
<tr><td><code id="fftrees_apply_+3A_fin_na_pred">fin_NA_pred</code></td>
<td>
<p>What outcome should be predicted if the <em>final</em> node in a tree has a cue value of <code>NA</code>
(as character)? Valid options are:
</p>

<dl>
<dt>'noise'</dt><dd><p>predict <code>FALSE</code> (0/left/signal) for all corresponding cases</p>
</dd>
<dt>'signal'</dt><dd><p>predict <code>TRUE</code> (1/right/noise) for all corresponding cases</p>
</dd>
<dt>'majority'</dt><dd><p>predict the more common criterion value (i.e., <code>TRUE</code> if base rate <code>p(TRUE) &gt; .50</code> in 'train' data) for all corresponding cases</p>
</dd>
<dt>'baseline'</dt><dd><p>flip a random coin that is biased by the criterion baseline <code>p(TRUE)</code> (in 'train' data) for all corresponding cases</p>
</dd>
<dt>'dnk'</dt><dd><p>yet ToDo: abstain from classifying / decide to 'do not know' / defer (i.e., tertium datur)</p>
</dd>
</dl>

<p>Default: <code>fin_NA_pred = "majority"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified <code>FFTrees</code> object (with lists in <code>x$trees</code> containing information on FFT decisions and statistics).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>

<hr>
<h2 id='fftrees_create'>Create an object of class <code>FFTrees</code></h2><span id='topic+fftrees_create'></span>

<h3>Description</h3>

<p><code>fftrees_create</code> creates an <code>FFTrees</code> object.
</p>
<p><code>fftrees_create</code> is called internally by the main <code><a href="#topic+FFTrees">FFTrees</a></code> function.
Its main purpose is to verify and store various parameters
(e.g., to denote algorithms, goals, thresholds) to be used in maximization processes
and for evaluation purposes (e.g., <code>sens.w</code> and cost values).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fftrees_create(
  formula = NULL,
  data = NULL,
  data.test = NULL,
  algorithm = NULL,
  goal = NULL,
  goal.chase = NULL,
  goal.threshold = NULL,
  max.levels = NULL,
  numthresh.method = NULL,
  numthresh.n = NULL,
  repeat.cues = NULL,
  stopping.rule = NULL,
  stopping.par = NULL,
  sens.w = NULL,
  cost.outcomes = NULL,
  cost.cues = NULL,
  main = NULL,
  decision.labels = NULL,
  my.goal = NULL,
  my.goal.fun = NULL,
  my.tree = NULL,
  do.comp = TRUE,
  do.lr = TRUE,
  do.svm = TRUE,
  do.cart = TRUE,
  do.rf = TRUE,
  quiet = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fftrees_create_+3A_formula">formula</code></td>
<td>
<p>A formula (with a binary criterion variable).</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_data">data</code></td>
<td>
<p>Training data (as data frame).</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_data.test">data.test</code></td>
<td>
<p>Data for testing models/prediction (as data frame).</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_algorithm">algorithm</code></td>
<td>
<p>Algorithm for growing FFTs (<code>"ifan"</code> or <code>"dfan"</code>) (as character string).</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_goal">goal</code></td>
<td>
<p>Measure used to select FFTs (as character string).</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_goal.chase">goal.chase</code></td>
<td>
<p>Measure used to optimize FFT creation (as character string).</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_goal.threshold">goal.threshold</code></td>
<td>
<p>Measure used to optimize cue thresholds (as character string).</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_max.levels">max.levels</code></td>
<td>
<p>integer.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_numthresh.method">numthresh.method</code></td>
<td>
<p>string.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_numthresh.n">numthresh.n</code></td>
<td>
<p>integer.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_repeat.cues">repeat.cues</code></td>
<td>
<p>logical.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_stopping.rule">stopping.rule</code></td>
<td>
<p>string.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_stopping.par">stopping.par</code></td>
<td>
<p>numeric.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_sens.w">sens.w</code></td>
<td>
<p>numeric.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_cost.outcomes">cost.outcomes</code></td>
<td>
<p>list.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_cost.cues">cost.cues</code></td>
<td>
<p>list.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_main">main</code></td>
<td>
<p>string.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_decision.labels">decision.labels</code></td>
<td>
<p>string.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_my.goal">my.goal</code></td>
<td>
<p>The name of an optimization measure defined by <code>my.goal.fun</code> (as a character string).
Example: <code>my.goal = "my_acc"</code> (see <code>my.goal.fun</code> for corresponding function).
Default: <code>my.goal = NULL</code>.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_my.goal.fun">my.goal.fun</code></td>
<td>
<p>The definition of an outcome measure to optimize, defined as a function
of the frequency counts of the 4 basic classification outcomes <code>hi, fa, mi, cr</code>
(i.e., an R function with 4 arguments <code>hi, fa, mi, cr</code>).
Example: <code>my.goal.fun = function(hi, fa, mi, cr){(hi + cr)/(hi + fa + mi + cr)}</code> (i.e., accuracy).
Default: <code>my.goal.fun = NULL</code>.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_my.tree">my.tree</code></td>
<td>
<p>A verbal description of an FFT, i.e., an &quot;FFT in words&quot; (as character string).
For example, <code>my.tree = "If age &gt; 20, predict TRUE. If sex = {m}, predict FALSE. Otherwise, predict TRUE."</code>.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_do.comp">do.comp</code></td>
<td>
<p>logical.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_do.lr">do.lr</code></td>
<td>
<p>logical.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_do.svm">do.svm</code></td>
<td>
<p>logical.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_do.cart">do.cart</code></td>
<td>
<p>logical.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_do.rf">do.rf</code></td>
<td>
<p>logical.</p>
</td></tr>
<tr><td><code id="fftrees_create_+3A_quiet">quiet</code></td>
<td>
<p>A list of logical elements.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>FFTrees</code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fftrees_define">fftrees_define</a></code> for defining FFTs;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>

<hr>
<h2 id='fftrees_cuerank'>Calculate thresholds that optimize some statistic (goal) for cues in data</h2><span id='topic+fftrees_cuerank'></span>

<h3>Description</h3>

<p><code>fftrees_cuerank</code> takes an <code>FFTrees</code> object <code>x</code> and
optimizes its <code>goal.threshold</code> (from <code>x$params</code>) for all cues in
<code>newdata</code> (of type <code>data</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fftrees_cuerank(x = NULL, newdata = NULL, data = "train", rounding = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fftrees_cuerank_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object.</p>
</td></tr>
<tr><td><code id="fftrees_cuerank_+3A_newdata">newdata</code></td>
<td>
<p>A dataset with cues to be ranked (as data frame).</p>
</td></tr>
<tr><td><code id="fftrees_cuerank_+3A_data">data</code></td>
<td>
<p>The type of data with cues to be ranked (as character: <code>'train'</code>, <code>'test'</code>, or <code>'dynamic'</code>).
Default: <code>data = 'train'</code>.</p>
</td></tr>
<tr><td><code id="fftrees_cuerank_+3A_rounding">rounding</code></td>
<td>
<p>integer. An integer value indicating the decimal digit
to which non-integer numeric cue thresholds are to be rounded.
Default: <code>rounding = NULL</code> (i.e., no rounding).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>fftrees_cuerank</code> creates a data frame <code>cuerank_df</code>
that is added to <code>x$cues$stats</code>.
</p>
<p>Note that the cue directions and thresholds computed by <strong>FFTrees</strong>
always predict positive criterion values (i.e., <code>TRUE</code> or signal,
rather than <code>FALSE</code> or noise).
Using these thresholds for negative exits (i.e., for predicting instances of
<code>FALSE</code> or noise) usually requires a reversal (e.g., negating cue direction).
</p>
<p><code>fftrees_cuerank</code> is called (twice) by the <code>fftrees_grow_fan</code> algorithm
to grow fast-and-frugal trees (FFTs).
</p>


<h3>Value</h3>

<p>A modified <code>FFTrees</code> object (with cue rank information
for the current <code>data</code> type in <code>x$cues$stats</code>).
</p>

<hr>
<h2 id='fftrees_define'>Create FFT definitions</h2><span id='topic+fftrees_define'></span>

<h3>Description</h3>

<p><code>fftrees_define</code> defines fast-and-frugal trees (FFTs)
either from the definitions provided or by applying algorithms (when no definitions are provided),
and returns a modified <code>FFTrees</code> object that contains those definitions.
</p>
<p>In most use cases, <code>fftrees_define</code> passes a new <code>FFTrees</code> object <code>x</code> either
to <code><a href="#topic+fftrees_grow_fan">fftrees_grow_fan</a></code> (to create new FFTs by applying algorithms to data) or
to <code><a href="#topic+fftrees_wordstofftrees">fftrees_wordstofftrees</a></code> (if <code>my.tree</code> is specified).
</p>
<p>If an existing <code>FFTrees</code> object <code>object</code> or <code>tree.definitions</code> are provided as inputs,
no new FFTs are created.
When both arguments are provided, <code>tree.definitions</code> take priority over the FFTs in an existing <code>object</code>.
Specifically,
</p>

<ul>
<li><p>If <code>tree.definitions</code> are provided, these are assigned to the FFTs of <code>x</code>.
</p>
</li>
<li><p>If no <code>tree.definitions</code> are provided, but an existing <code>FFTrees</code> object <code>object</code> is provided,
the trees from <code>object</code> are assigned to the FFTs of <code>x</code>.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>fftrees_define(x, object = NULL, tree.definitions = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fftrees_define_+3A_x">x</code></td>
<td>
<p>The current <code>FFTrees</code> object (to be changed and returned).</p>
</td></tr>
<tr><td><code id="fftrees_define_+3A_object">object</code></td>
<td>
<p>An existing <code>FFTrees</code> object (with tree definitions).</p>
</td></tr>
<tr><td><code id="fftrees_define_+3A_tree.definitions">tree.definitions</code></td>
<td>
<p>A <code>data.frame</code>. An optional hard-coded definition of FFTs (in the same format as in an <code>FFTrees</code> object).
If specified, no new FFTs are created, but the tree definitions in <code>object</code> or <code>x</code> are replaced by the tree definitions provided
and the current object is re-evaluated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>FFTrees</code> object with tree definitions.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fftrees_create">fftrees_create</a></code> for creating <code>FFTrees</code> objects;
<code><a href="#topic+fftrees_grow_fan">fftrees_grow_fan</a></code> for creating FFTs by applying algorithms to data;
<code><a href="#topic+fftrees_wordstofftrees">fftrees_wordstofftrees</a></code> for creating FFTs from verbal descriptions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>

<hr>
<h2 id='fftrees_ffttowords'>Describe a fast-and-frugal tree (FFT) in words</h2><span id='topic+fftrees_ffttowords'></span>

<h3>Description</h3>

<p><code>fftrees_ffttowords</code> provides a verbal description
of tree definition (as defined in an <code>FFTrees</code> object).
Thus, <code>fftrees_ffttowords</code> translates an abstract FFT definition
into natural language output.
</p>
<p><code>fftrees_ffttowords</code> is the complement function to
<code><a href="#topic+fftrees_wordstofftrees">fftrees_wordstofftrees</a></code>, which parses a verbal description
of an FFT into the abstract tree definition of an <code>FFTrees</code> object.
</p>
<p>The final sentence (or tree node) of the FFT's description
always predicts positive criterion values (i.e., <code>TRUE</code> instances) first,
before predicting negative criterion values (i.e., <code>FALSE</code> instances).
Note that this may require a reversal of exit directions,
if the final cue predicted <code>FALSE</code> instances.
</p>
<p>Note that the cue directions and thresholds computed by <strong>FFTrees</strong>
always predict positive criterion values (i.e., <code>TRUE</code> or signal,
rather than <code>FALSE</code> or noise).
Using these thresholds for negative exits (i.e., for predicting instances of
<code>FALSE</code> or noise) usually requires a reversal (e.g., negating cue direction).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fftrees_ffttowords(x = NULL, mydata = "train", digits = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fftrees_ffttowords_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object created with <code><a href="#topic+FFTrees">FFTrees</a></code>.</p>
</td></tr>
<tr><td><code id="fftrees_ffttowords_+3A_mydata">mydata</code></td>
<td>
<p>The type of data to which a tree is being applied (as character string &quot;train&quot; or &quot;test&quot;).
Default: <code>mydata = "train"</code>.</p>
</td></tr>
<tr><td><code id="fftrees_ffttowords_+3A_digits">digits</code></td>
<td>
<p>How many digits to round numeric values (as integer)?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified <code>FFTrees</code> object <code>x</code> with
<code>x$trees$inwords</code> containing a list of string vectors.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fftrees_wordstofftrees">fftrees_wordstofftrees</a></code> for converting a verbal description
of an FFT into an <code>FFTrees</code> object;
<code><a href="#topic+fftrees_create">fftrees_create</a></code> for creating <code>FFTrees</code> objects;
<code><a href="#topic+fftrees_grow_fan">fftrees_grow_fan</a></code> for creating FFTs by applying algorithms to data;
<code><a href="#topic+print.FFTrees">print.FFTrees</a></code> for printing FFTs;
<code><a href="#topic+plot.FFTrees">plot.FFTrees</a></code> for plotting FFTs;
<code><a href="#topic+summary.FFTrees">summary.FFTrees</a></code> for summarizing FFTs;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
heart.fft &lt;- FFTrees(diagnosis ~ .,
  data = heartdisease,
  decision.labels = c("Healthy", "Disease")
)

inwords(heart.fft)

</code></pre>

<hr>
<h2 id='fftrees_fitcomp'>Fit competitive algorithms</h2><span id='topic+fftrees_fitcomp'></span>

<h3>Description</h3>

<p><code>fftrees_fitcomp</code> fits competitive algorithms for binary classification tasks
(e.g., LR, CART, RF, SVM) to the data and parameters specified in an <code>FFTrees</code> object.
</p>
<p><code>fftrees_fitcomp</code> is called by the main <code><a href="#topic+FFTrees">FFTrees</a></code> function
when creating FFTs from and applying them to data (unless <code>do.comp = FALSE</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fftrees_fitcomp(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fftrees_fitcomp_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>

<hr>
<h2 id='fftrees_grow_fan'>Grow fast-and-frugal trees (FFTs) using the <code>fan</code> algorithms</h2><span id='topic+fftrees_grow_fan'></span>

<h3>Description</h3>

<p><code>fftrees_grow_fan</code> is called by <code><a href="#topic+fftrees_define">fftrees_define</a></code>
to create new FFTs by applying the <code>fan</code> algorithms
(specifically, either <code>ifan</code> or <code>dfan</code>) to data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fftrees_grow_fan(x, repeat.cues = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fftrees_grow_fan_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object.</p>
</td></tr>
<tr><td><code id="fftrees_grow_fan_+3A_repeat.cues">repeat.cues</code></td>
<td>
<p>Can cues be considered/used repeatedly (as logical)?
Default: <code>repeat.cues = TRUE</code>, but only relevant when using the <code>dfan</code> algorithm.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+fftrees_create">fftrees_create</a></code> for creating <code>FFTrees</code> objects;
<code><a href="#topic+fftrees_define">fftrees_define</a></code> for defining FFTs;
<code><a href="#topic+fftrees_grow_fan">fftrees_grow_fan</a></code> for creating FFTs by applying algorithms to data;
<code><a href="#topic+fftrees_wordstofftrees">fftrees_wordstofftrees</a></code> for creating FFTs from verbal descriptions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>

<hr>
<h2 id='fftrees_ranktrees'>Rank FFTs by current goal</h2><span id='topic+fftrees_ranktrees'></span>

<h3>Description</h3>

<p><code>fftrees_ranktrees</code> ranks trees in an <code>FFTrees</code> object <code>x</code>
based on the current goal (either <code>"cost"</code> or as specified in <code>x$params$goal</code>).
</p>
<p><code>fftrees_ranktrees</code> is called by the main <code><a href="#topic+FFTrees">FFTrees</a></code> function
when creating FFTs from and applying them to (training) data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fftrees_ranktrees(x, data = "train")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fftrees_ranktrees_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object.</p>
</td></tr>
<tr><td><code id="fftrees_ranktrees_+3A_data">data</code></td>
<td>
<p>The type of data to be used (as character).
Default: <code>data = "train"</code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>

<hr>
<h2 id='fftrees_threshold_factor_grid'>Perform a grid search over factor and return accuracy statistics for a given factor cue</h2><span id='topic+fftrees_threshold_factor_grid'></span>

<h3>Description</h3>

<p>Perform a grid search over factor and return accuracy statistics for a given factor cue
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fftrees_threshold_factor_grid(
  thresholds = NULL,
  cue_v = NULL,
  criterion_v = NULL,
  directions = "=",
  goal.threshold = NULL,
  sens.w = NULL,
  my.goal = NULL,
  my.goal.fun = NULL,
  cost.each = NULL,
  cost.outcomes = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fftrees_threshold_factor_grid_+3A_thresholds">thresholds</code></td>
<td>
<p>numeric. A vector of factor thresholds to consider.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_factor_grid_+3A_cue_v">cue_v</code></td>
<td>
<p>numeric. Feature/cue values.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_factor_grid_+3A_criterion_v">criterion_v</code></td>
<td>
<p>logical. A logical vector of (TRUE) criterion values.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_factor_grid_+3A_directions">directions</code></td>
<td>
<p>character. Character vector of threshold directions to consider.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_factor_grid_+3A_goal.threshold">goal.threshold</code></td>
<td>
<p>A character string indicating the criterion to maximize when <em>optimizing cue thresholds</em>:
<code>"acc"</code> = overall accuracy, <code>"bacc"</code> = balanced accuracy, <code>"wacc"</code> = weighted accuracy,
<code>"dprime"</code> = discriminability, <code>"cost"</code> = costs (based only on <code>cost.outcomes</code>, as <code>cost.cues</code> are constant per cue).
Default: <code>goal.threshold = "bacc"</code>.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_factor_grid_+3A_sens.w">sens.w</code></td>
<td>
<p>numeric. Sensitivity weight parameter (from <code>0</code> to <code>1</code>, for computing <code>wacc</code>).
Default: <code>sens.w = .50</code>.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_factor_grid_+3A_my.goal">my.goal</code></td>
<td>
<p>Name of an optional, user-defined goal (as character string). Default: <code>my.goal = NULL</code>.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_factor_grid_+3A_my.goal.fun">my.goal.fun</code></td>
<td>
<p>User-defined goal function (with 4 arguments <code>hi fa mi cr</code>). Default: <code>my.goal.fun = NULL</code>.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_factor_grid_+3A_cost.each">cost.each</code></td>
<td>
<p>numeric. A constant cost value to add to each value (e.g., the cost of the cue).</p>
</td></tr>
<tr><td><code id="fftrees_threshold_factor_grid_+3A_cost.outcomes">cost.outcomes</code></td>
<td>
<p>list. A list of length 4 with names 'hi', 'fa', 'mi', and 'cr' specifying
the costs of a hit, false alarm, miss, and correct rejection, respectively, in some common currency.
For instance, <code>cost.outcomes = listc("hi" = 0, "fa" = 10, "mi" = 20, "cr" = 0)</code> means that
a false alarm and miss cost <code>10</code> and <code>20</code> units, respectively, while correct decisions have no cost.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing accuracy statistics for factor thresholds.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fftrees_threshold_numeric_grid">fftrees_threshold_numeric_grid</a></code> for numeric cues.
</p>

<hr>
<h2 id='fftrees_threshold_numeric_grid'>Perform a grid search over thresholds and return accuracy statistics for a given numeric cue</h2><span id='topic+fftrees_threshold_numeric_grid'></span>

<h3>Description</h3>

<p>Perform a grid search over thresholds and return accuracy statistics for a given numeric cue
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fftrees_threshold_numeric_grid(
  thresholds,
  cue_v,
  criterion_v,
  directions = c("&gt;", "&lt;="),
  goal.threshold = NULL,
  sens.w = NULL,
  my.goal = NULL,
  my.goal.fun = NULL,
  cost.each = NULL,
  cost.outcomes = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fftrees_threshold_numeric_grid_+3A_thresholds">thresholds</code></td>
<td>
<p>numeric. A vector of thresholds to consider.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_numeric_grid_+3A_cue_v">cue_v</code></td>
<td>
<p>numeric. Feature values.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_numeric_grid_+3A_criterion_v">criterion_v</code></td>
<td>
<p>logical. A logical vector of (TRUE) criterion values.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_numeric_grid_+3A_directions">directions</code></td>
<td>
<p>character. Possible directions to consider.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_numeric_grid_+3A_goal.threshold">goal.threshold</code></td>
<td>
<p>A character string indicating the criterion to maximize when <em>optimizing cue thresholds</em>:
<code>"acc"</code> = overall accuracy, <code>"bacc"</code> = balanced accuracy, <code>"wacc"</code> = weighted accuracy,
<code>"dprime"</code> = discriminability, <code>"cost"</code> = costs (based only on <code>cost.outcomes</code>, as <code>cost.cues</code> are constant per cue).
Default: <code>goal.threshold = "bacc"</code>.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_numeric_grid_+3A_sens.w">sens.w</code></td>
<td>
<p>numeric. Sensitivity weight parameter (from <code>0</code> to <code>1</code>, for computing <code>wacc</code>).
Default: <code>sens.w = .50</code>.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_numeric_grid_+3A_my.goal">my.goal</code></td>
<td>
<p>Name of an optional, user-defined goal (as character string). Default: <code>my.goal = NULL</code>.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_numeric_grid_+3A_my.goal.fun">my.goal.fun</code></td>
<td>
<p>User-defined goal function (with 4 arguments <code>hi fa mi cr</code>). Default: <code>my.goal.fun = NULL</code>.</p>
</td></tr>
<tr><td><code id="fftrees_threshold_numeric_grid_+3A_cost.each">cost.each</code></td>
<td>
<p>numeric. A constant cost value to add to each value (e.g., the cost of the cue).</p>
</td></tr>
<tr><td><code id="fftrees_threshold_numeric_grid_+3A_cost.outcomes">cost.outcomes</code></td>
<td>
<p>list. A list of length 4 with names 'hi', 'fa', 'mi', and 'cr' specifying
the costs of a hit, false alarm, miss, and correct rejection, respectively, in some common currency.
For instance, <code>cost.outcomes = listc("hi" = 0, "fa" = 10, "mi" = 20, "cr" = 0)</code> means that
a false alarm and miss cost <code>10</code> and <code>20</code> units, respectively, while correct decisions have no cost.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing accuracy statistics for numeric thresholds.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fftrees_threshold_factor_grid">fftrees_threshold_factor_grid</a></code> for factor cues.
</p>

<hr>
<h2 id='fftrees_wordstofftrees'>Convert a verbal description of an FFT into an <code>FFTrees</code> object</h2><span id='topic+fftrees_wordstofftrees'></span>

<h3>Description</h3>

<p><code>fftrees_wordstofftrees</code> converts a verbal description
of an FFT (provided as a string of text) into
a tree definition (of an <code>FFTrees</code> object).
Thus, <code>fftrees_wordstofftrees</code> provides a simple
natural language parser for FFTs.
</p>
<p><code>fftrees_wordstofftrees</code> is the complement function to
<code><a href="#topic+fftrees_ffttowords">fftrees_ffttowords</a></code>, which converts an abstract tree definition
(of an <code>FFTrees</code> object) into a verbal description
(i.e., provides natural language output).
</p>
<p>To increase robustness, the parsing of <code>fftrees_wordstofftrees</code>
allows for lower- or uppercase spellings (but not typographical variants)
and ignores the else-part of the final sentence (i.e., the part
beginning with &quot;otherwise&quot;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fftrees_wordstofftrees(x, my.tree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fftrees_wordstofftrees_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object.</p>
</td></tr>
<tr><td><code id="fftrees_wordstofftrees_+3A_my.tree">my.tree</code></td>
<td>
<p>A character string. A verbal description (as a string of text) defining an FFT.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>FFTrees</code> object with a new tree definition as described by <code>my.tree</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fftrees_ffttowords">fftrees_ffttowords</a></code> for converting FFTs into verbal descriptions;
<code><a href="#topic+print.FFTrees">print.FFTrees</a></code> for printing FFTs;
<code><a href="#topic+plot.FFTrees">plot.FFTrees</a></code> for plotting FFTs;
<code><a href="#topic+summary.FFTrees">summary.FFTrees</a></code> for summarizing FFTs;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>

<hr>
<h2 id='FFTrees.guide'>Open the <strong>FFTrees</strong> package guide</h2><span id='topic+FFTrees.guide'></span>

<h3>Description</h3>

<p>Open the <strong>FFTrees</strong> package guide
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FFTrees.guide()
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>

<hr>
<h2 id='flip_exits'>Flip exits in an FFT definition</h2><span id='topic+flip_exits'></span>

<h3>Description</h3>

<p><code>flip_exits</code> reverses the exits of
one or more <code>nodes</code> from an existing FFT definition
(in the tidy data frame format).
</p>
<p><code>flip_exits</code> alters the value(s) of the non-final
exits specified in <code>nodes</code> (from 0 to 1, or from 1 to 0).
By contrast, exits of final <code>nodes</code> remain unchanged.
</p>
<p>Duplicates in <code>nodes</code> are flipped only once
(rather than repeatedly) and <code>nodes</code> not in
the range <code>1:nrow(fft)</code> are ignored.
</p>
<p><code>flip_exits</code> is a more specialized function
than <code><a href="#topic+edit_nodes">edit_nodes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flip_exits(fft, nodes = NA, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="flip_exits_+3A_fft">fft</code></td>
<td>
<p>One FFT definition
(as a data frame in tidy format, with one row per node).</p>
</td></tr>
<tr><td><code id="flip_exits_+3A_nodes">nodes</code></td>
<td>
<p>The FFT nodes whose exits are to be flipped (as an integer vector).
Default: <code>nodes = NA</code>.</p>
</td></tr>
<tr><td><code id="flip_exits_+3A_quiet">quiet</code></td>
<td>
<p>Hide feedback messages (as logical)?
Default: <code>quiet = FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>One FFT definition
(as a data frame in tidy format, with one row per node).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_nodes">add_nodes</a></code> for adding nodes to an FFT definition;
<code><a href="#topic+edit_nodes">edit_nodes</a></code> for editing nodes in an FFT definition;
<code><a href="#topic+drop_nodes">drop_nodes</a></code> for deleting nodes from an FFT definition;
<code><a href="#topic+reorder_nodes">reorder_nodes</a></code> for reordering nodes of an FFT definition;
<code><a href="#topic+select_nodes">select_nodes</a></code> for selecting nodes in an FFT definition;
<code><a href="#topic+get_fft_df">get_fft_df</a></code> for getting the FFT definitions of an <code>FFTrees</code> object;
<code><a href="#topic+read_fft_df">read_fft_df</a></code> for reading one FFT definition from tree definitions;
<code><a href="#topic+add_fft_df">add_fft_df</a></code> for adding FFTs to tree definitions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other tree definition and manipulation functions: 
<code><a href="#topic+add_fft_df">add_fft_df</a>()</code>,
<code><a href="#topic+add_nodes">add_nodes</a>()</code>,
<code><a href="#topic+drop_nodes">drop_nodes</a>()</code>,
<code><a href="#topic+edit_nodes">edit_nodes</a>()</code>,
<code><a href="#topic+get_fft_df">get_fft_df</a>()</code>,
<code><a href="#topic+read_fft_df">read_fft_df</a>()</code>,
<code><a href="#topic+reorder_nodes">reorder_nodes</a>()</code>,
<code><a href="#topic+select_nodes">select_nodes</a>()</code>,
<code><a href="#topic+write_fft_df">write_fft_df</a>()</code>
</p>

<hr>
<h2 id='forestfires'>Forest fires data</h2><span id='topic+forestfires'></span>

<h3>Description</h3>

<p>A dataset of forest fire statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>forestfires
</code></pre>


<h3>Format</h3>

<p>A data frame containing 517 rows and 13 columns.
</p>

<dl>
<dt>X</dt><dd><p>Integer -x-axis spatial coordinate within the Montesinho park map: 1 to 9</p>
</dd>
<dt>Y</dt><dd><p>Integer - y-axis spatial coordinate within the Montesinho park map: 2 to 9</p>
</dd>
<dt>month</dt><dd><p>Factor - month of the year: &quot;jan&quot; to &quot;dec&quot; </p>
</dd>
<dt>day</dt><dd><p>Factor -day of the week: &quot;mon&quot; to &quot;sun&quot;</p>
</dd>
<dt>FFMC</dt><dd><p>Numeric -FFMC index from the FWI system: 18.7 to 96.20</p>
</dd>
<dt>DMC</dt><dd><p>Numeric - DMC index from the FWI system: 1.1 to 291.3 </p>
</dd>
<dt>DC</dt><dd><p>Numeric - DC index from the FWI system: 7.9 to 860.6 </p>
</dd>
<dt>ISI</dt><dd><p>Numeric - ISI index from the FWI system: 0.0 to 56.10</p>
</dd>
<dt>temp</dt><dd><p>Numeric - temperature in Celsius degrees: 2.2 to 33.30</p>
</dd>
<dt>RH</dt><dd><p>Numeric - relative humidity in percent: 15.0 to 100 </p>
</dd>
<dt>wind</dt><dd><p>Numeric - wind speed in km/h: 0.40 to 9.40 </p>
</dd>
<dt>rain</dt><dd><p>Numeric - outside rain in mm/m2 : 0.0 to 6.4 </p>
</dd>
<dt>fire.crit</dt><dd><p><em>Criterion</em>: Was there a fire (greater than 1.00 ha)?
</p>
<p>Values: <code>TRUE</code> (yes) vs. <code>FALSE</code> (no) (47.0% vs. 53.0%).</p>
</dd>
</dl>



<h3>Details</h3>

<p>We made the following enhancements to the original data for improved usability:
</p>

<ul>
<li><p>The criterion was redefined from a numeric variable that indicated the number of hectares that burned in a fire into a logical variable (<code>TRUE</code> (for values &gt;1) vs. <code>FALSE</code> (for values &lt;=1)).
</p>
</li></ul>

<p>Other than that, the data remains consistent with the original dataset.
</p>


<h3>Source</h3>

<p><a href="http://archive.ics.uci.edu/ml/datasets/Forest+Fires">http://archive.ics.uci.edu/ml/datasets/Forest+Fires</a>
</p>
<p>Original creator:
Prof. Paulo Cortez and Anbal Morais
Department of Information Systems
University of Minho, Portugal
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='get_best_tree'>Select the best tree (from current set of FFTs)</h2><span id='topic+get_best_tree'></span>

<h3>Description</h3>

<p><code>get_best_tree</code> selects (looks up and identifies) the best tree (as an integer)
from the set (or &ldquo;fan&rdquo;) of FFTs contained in the current <code>FFTrees</code> object <code>x</code>,
an existing type of <code>data</code> ('train' or 'test'), and
a <code>goal</code> for which corresponding statistics are available
in the designated <code>data</code> type (in <code>x$trees$stats</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_best_tree(x, data, goal, my.goal.max = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_best_tree_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object.</p>
</td></tr>
<tr><td><code id="get_best_tree_+3A_data">data</code></td>
<td>
<p>The type of data to consider (as character: either 'train' or 'test').</p>
</td></tr>
<tr><td><code id="get_best_tree_+3A_goal">goal</code></td>
<td>
<p>A goal (as character) to be maximized or minimized when selecting a tree
from an existing <code>FFTrees</code> object <code>x</code> (with existing <code>x$trees$stats</code>).</p>
</td></tr>
<tr><td><code id="get_best_tree_+3A_my.goal.max">my.goal.max</code></td>
<td>
<p>Default direction for user-defined <code>my.goal</code> (as logical):
Should <code>my.goal</code> be maximized?
Default: <code>my.goal.max = TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Importantly, <code>get_best_tree</code> only identifies and selects the 'tree' <em>identifier</em>
(as an integer) from the set of <em>existing</em> trees with known statistics,
rather than creating new trees or computing new cue thresholds.
More specifically, <code>goal</code> is used for identifying and selecting the 'tree'
identifier (as an integer) of the best FFT from an existing set of FFTs, but not for
computing new cue thresholds (see <code>goal.threshold</code> and <code>fftrees_cuerank()</code>) or
creating new trees (see <code>goal.chase</code> and <code>fftrees_ranktrees()</code>).
</p>


<h3>Value</h3>

<p>An integer denoting the <code>tree</code> that maximizes/minimizes <code>goal</code> in <code>data</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other utility functions: 
<code><a href="#topic+get_exit_type">get_exit_type</a>()</code>,
<code><a href="#topic+get_fft_df">get_fft_df</a>()</code>
</p>

<hr>
<h2 id='get_exit_type'>Get exit type (from a vector <code>x</code> of FFT exit descriptions)</h2><span id='topic+get_exit_type'></span>

<h3>Description</h3>

<p><code>get_exit_type</code> checks and converts a vector <code>x</code>
of FFT exit descriptions into exits of an FFT
that correspond to the current options of
<code>exit_types</code> (as a global constant).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_exit_type(x, verify = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_exit_type_+3A_x">x</code></td>
<td>
<p>A vector of FFT exit descriptions.</p>
</td></tr>
<tr><td><code id="get_exit_type_+3A_verify">verify</code></td>
<td>
<p>A flag to turn verification on/off (as logical).
Default: <code>verify = TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>get_exit_type</code> also verifies that the exit types conform to an FFT
(e.g., only the exits of the final node are bi-directional).
</p>


<h3>Value</h3>

<p>A vector of <code>exit_types</code> (or an error).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other utility functions: 
<code><a href="#topic+get_best_tree">get_best_tree</a>()</code>,
<code><a href="#topic+get_fft_df">get_fft_df</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_exit_type(c(0, 1, .5))
get_exit_type(c(FALSE,   " True ",  2/4))
get_exit_type(c("noise", "signal", "final"))
get_exit_type(c("left",  "right",  "both"))

</code></pre>

<hr>
<h2 id='get_fft_df'>Get FFT definitions (from an <code>FFTrees</code> object <code>x</code>)</h2><span id='topic+get_fft_df'></span>

<h3>Description</h3>

<p><code>get_fft_df</code> gets the FFT definitions
of an <code>FFTrees</code> object <code>x</code>
(as a <code>data.frame</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_fft_df(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_fft_df_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The FFTs in the <code>data.frame</code> returned
are represented in the one-line per FFT definition format
used by an <code>FFTrees</code> object.
</p>
<p>In addition to looking up <code>x$trees$definitions</code>,
<code>get_fft_df</code> verifies that the FFT definitions
are valid (given current settings).
</p>


<h3>Value</h3>

<p>A set of FFT definitions (as a <code>data.frame</code>/<code>tibble</code>,
in the one-line per FFT definition format used by an <code>FFTrees</code> object).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_fft_df">read_fft_df</a></code> for reading one FFT definition from tree definitions;
<code><a href="#topic+write_fft_df">write_fft_df</a></code> for writing one FFT to tree definitions;
<code><a href="#topic+add_fft_df">add_fft_df</a></code> for adding FFTs to tree definitions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other utility functions: 
<code><a href="#topic+get_best_tree">get_best_tree</a>()</code>,
<code><a href="#topic+get_exit_type">get_exit_type</a>()</code>
</p>
<p>Other tree definition and manipulation functions: 
<code><a href="#topic+add_fft_df">add_fft_df</a>()</code>,
<code><a href="#topic+add_nodes">add_nodes</a>()</code>,
<code><a href="#topic+drop_nodes">drop_nodes</a>()</code>,
<code><a href="#topic+edit_nodes">edit_nodes</a>()</code>,
<code><a href="#topic+flip_exits">flip_exits</a>()</code>,
<code><a href="#topic+read_fft_df">read_fft_df</a>()</code>,
<code><a href="#topic+reorder_nodes">reorder_nodes</a>()</code>,
<code><a href="#topic+select_nodes">select_nodes</a>()</code>,
<code><a href="#topic+write_fft_df">write_fft_df</a>()</code>
</p>

<hr>
<h2 id='heart.cost'>Cue costs for the heartdisease data</h2><span id='topic+heart.cost'></span>

<h3>Description</h3>

<p>This data further characterizes the variables (cues) in the <code><a href="#topic+heartdisease">heartdisease</a></code> dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heart.cost
</code></pre>


<h3>Format</h3>

<p>A list of length 13 containing the cost of each cue in the <code><a href="#topic+heartdisease">heartdisease</a></code> dataset (in dollars).
Each list element is a single (positive numeric) value.
</p>


<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/machine-learning-databases/heart-disease/costs/">https://archive.ics.uci.edu/ml/machine-learning-databases/heart-disease/costs/</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+heartdisease">heartdisease</a></code> dataset.
</p>
<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='heart.test'>Heart disease testing data</h2><span id='topic+heart.test'></span>

<h3>Description</h3>

<p>Testing data for a <code><a href="#topic+heartdisease">heartdisease</a></code> data.
This subset is used to test the prediction performance of a model trained on the <code><a href="#topic+heart.train">heart.train</a></code> data.
The dataset <code><a href="#topic+heartdisease">heartdisease</a></code> contains both datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heart.test
</code></pre>


<h3>Format</h3>

<p>A data frame containing 153 rows and 14 columns (see <code><a href="#topic+heartdisease">heartdisease</a></code> for details).
</p>


<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Heart+Disease">https://archive.ics.uci.edu/ml/datasets/Heart+Disease</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+heartdisease">heartdisease</a></code> dataset.
</p>
<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='heart.train'>Heart disease training data</h2><span id='topic+heart.train'></span>

<h3>Description</h3>

<p>Training data for a binary prediction model (here: FFT) on (a subset of) the <code><a href="#topic+heartdisease">heartdisease</a></code> data.
The complementary subset for model testing is <code><a href="#topic+heart.test">heart.test</a></code>.
The data in <code><a href="#topic+heartdisease">heartdisease</a></code> contains both subsets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heart.train
</code></pre>


<h3>Format</h3>

<p>A data frame containing 150 rows and 14 columns (see <code><a href="#topic+heartdisease">heartdisease</a></code> for details).
</p>


<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Heart+Disease">https://archive.ics.uci.edu/ml/datasets/Heart+Disease</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+heartdisease">heartdisease</a></code> dataset.
</p>
<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='heartdisease'>Heart disease data</h2><span id='topic+heartdisease'></span>

<h3>Description</h3>

<p>A dataset predicting the <code>diagnosis</code> of 303 patients tested for heart disease.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heartdisease
</code></pre>


<h3>Format</h3>

<p>A data frame containing 303 rows and 14 columns, with the following variables:
</p>

<dl>
<dt>diagnosis</dt><dd><p>True value of binary criterion: TRUE = Heart disease, FALSE = No Heart disease</p>
</dd>
<dt>age</dt><dd><p>Age (in years)</p>
</dd>
<dt>sex</dt><dd><p>Sex, 1 = male, 0 = female</p>
</dd>
<dt>cp</dt><dd><p>Chest pain type: ta = typical angina, aa = atypical angina, np = non-anginal pain, a = asymptomatic</p>
</dd>
<dt>trestbps</dt><dd><p>Resting blood pressure (in mm Hg on admission to the hospital)</p>
</dd>
<dt>chol</dt><dd><p>Serum cholestoral in mg/dl</p>
</dd>
<dt>fbs</dt><dd><p>Fasting blood sugar &gt; 120 mg/dl: 1 = true, 0 = false </p>
</dd>
<dt>restecg</dt><dd><p>Resting electrocardiographic results. &quot;normal&quot; = normal, &quot;abnormal&quot; = having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of &gt; 0.05 mV), &quot;hypertrophy&quot; = showing probable or definite left ventricular hypertrophy by Estes' criteria.</p>
</dd>
<dt>thalach</dt><dd><p>Maximum heart rate achieved</p>
</dd>
<dt>exang</dt><dd><p>Exercise induced angina: 1 = yes, 0 = no</p>
</dd>
<dt>oldpeak</dt><dd><p>ST depression induced by exercise relative to rest </p>
</dd>
<dt>slope</dt><dd><p>The slope of the peak exercise ST segment. </p>
</dd>
<dt>ca</dt><dd><p>Number of major vessels (0-3) colored by flourosopy </p>
</dd>
<dt>thal</dt><dd><p>&quot;normal&quot; = normal, &quot;fd&quot; = fixed defect, &quot;rd&quot; = reversible defect</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Heart+Disease">https://archive.ics.uci.edu/ml/datasets/Heart+Disease</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+heart.cost">heart.cost</a></code> dataset for cost information.
</p>
<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='inwords'>Provide a verbal description of an FFT</h2><span id='topic+inwords'></span>

<h3>Description</h3>

<p><code>inwords</code> generates and provides a verbal description
of a fast-and-frugal tree (FFT) from an <code>FFTrees</code> object.
</p>
<p>When <code>data</code> remains unspecified, <code>inwords</code> will only look up <code>x$trees$inwords</code>.
When <code>data</code> is set to either &quot;train&quot; or &quot;test&quot;, <code>inwords</code> first employs
<code><a href="#topic+fftrees_ffttowords">fftrees_ffttowords</a></code> to re-generate the verbal descriptions of FFTs in <code>x</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inwords(x, data = NULL, tree = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="inwords_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object.</p>
</td></tr>
<tr><td><code id="inwords_+3A_data">data</code></td>
<td>
<p>The type of data to which a tree is being applied (as character string &quot;train&quot; or &quot;test&quot;).
Default: <code>data = NULL</code> will only look up <code>x$trees$inwords</code>.</p>
</td></tr>
<tr><td><code id="inwords_+3A_tree">tree</code></td>
<td>
<p>The tree to display (as an integer).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A verbal description of an FFT (as a character string).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fftrees_ffttowords">fftrees_ffttowords</a></code> for converting FFTs into verbal descriptions;
<code><a href="#topic+print.FFTrees">print.FFTrees</a></code> for printing FFTs;
<code><a href="#topic+plot.FFTrees">plot.FFTrees</a></code> for plotting FFTs;
<code><a href="#topic+summary.FFTrees">summary.FFTrees</a></code> for summarizing FFTs;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>

<hr>
<h2 id='iris.v'>Iris data</h2><span id='topic+iris.v'></span>

<h3>Description</h3>

<p>A famous dataset from R.A. Fisher (1936) simplified to predict only the virginica class (i.e., as a binary classification problem).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iris.v
</code></pre>


<h3>Format</h3>

<p>A data frame containing 150 rows and 4 columns.
</p>

<dl>
<dt>sep.len</dt><dd><p>sepal length in cm</p>
</dd>
<dt>sep.wid</dt><dd><p>sepal width in cm</p>
</dd>
<dt>pet.len</dt><dd><p>petal length in cm</p>
</dd>
<dt>pet.wid</dt><dd><p>petal width in cm</p>
</dd>
<dt>virginica</dt><dd><p><em>Criterion</em>: Does an iris belong to the class &quot;virginica&quot;?
</p>
<p>Values: <code>TRUE</code> vs. <code>FALSE</code> (33.33% vs.66.67%).</p>
</dd>
</dl>



<h3>Details</h3>

<p>To improve usability, we made the following changes:
</p>

<ul>
<li><p>The criterion was binarized from a factor variable with three levels
(<code>Iris-setosa</code>, <code>Iris-versicolor</code>, <code>Iris-virginica</code>),
into a logical variable (i.e., <code>TRUE</code> for all instances of <code>Iris-virginica</code>
and <code>FALSE</code> for the two other levels).
</p>
</li></ul>

<p>Other than that, the data remains consistent with the original dataset.
</p>


<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Iris">https://archive.ics.uci.edu/ml/datasets/Iris</a>
</p>


<h3>References</h3>

<p>Fisher, R.A. (1936): The use of multiple measurements in taxonomic problems.
Annual Eugenics, 7, Part II, pp. 179&ndash;188.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='mushrooms'>Mushrooms data</h2><span id='topic+mushrooms'></span>

<h3>Description</h3>

<p>Data describing poisonous vs. non-poisonous mushrooms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mushrooms
</code></pre>


<h3>Format</h3>

<p>A data frame containing 8,124 rows and 23 columns.
</p>
<p>See <a href="http://archive.ics.uci.edu/ml/machine-learning-databases/mushroom/agaricus-lepiota.names">http://archive.ics.uci.edu/ml/machine-learning-databases/mushroom/agaricus-lepiota.names</a> for column descriptions.
</p>

<dl>
<dt>poisonous</dt><dd><p><em>Criterion</em>: Is the mushroom poisonous?
</p>
<p>Values: <code>TRUE</code> (poisonous) vs. <code>FALSE</code> (eatable) (48.2% vs.\ 52.8%).</p>
</dd>
<dt>cshape</dt><dd><p>cap-shape, character (bell=b, conical=c, convex=x, flat=f, knobbed=k, sunken=s)</p>
</dd>
<dt>csurface</dt><dd><p>cap-surface, character (fibrous=f, grooves=g, scaly=y, smooth=s)</p>
</dd>
<dt>ccolor</dt><dd><p>cap-color, character (brown=n, buff=b, cinnamon=c, gray=g, green=r, pink=p, purple=u, red=e, white=w, yellow=y)</p>
</dd>
<dt>bruises</dt><dd><p>Are there bruises? logical (TRUE/FALSE)</p>
</dd>
<dt>odor</dt><dd><p>character (almond=a, anise=l, creosote=c, fishy=y, foul=f, musty=m, none=n, pungent=p, spicy=s) </p>
</dd>
<dt>gattach</dt><dd><p>gill-attachment, character (attached=a, descending=d, free=f, notched=n)</p>
</dd>
<dt>gspace</dt><dd><p>gill-spacing, character (close=c, crowded=w, distant=d)</p>
</dd>
<dt>gsize</dt><dd><p>gill-size, character (broad=b, narrow=n)</p>
</dd>
<dt>gcolor</dt><dd><p>gill-color, character (black=k, brown=n, buff=b, chocolate=h, gray=g, green=r, orange=o, pink=p, purple=u, red=e, white=w, yellow=y)</p>
</dd>
<dt>sshape</dt><dd><p>stalk-shape, character (enlarging=e, tapering=t)</p>
</dd>
<dt>sroot</dt><dd><p>stalk-root, character (bulbous=b ,club=c, cup=u, equal=e, rhizomorphs=z, rooted=r)</p>
</dd>
<dt>ssaring</dt><dd><p>stalk-surface-above-ring, character (fibrous=f, scaly=y, silky=k, smooth=s)</p>
</dd>
<dt>ssbring</dt><dd><p>stalk-surface-below-ring, character (fibrous=f, scaly=y, silky=k, smooth=s)</p>
</dd>
<dt>scaring</dt><dd><p>stalk-color-above-ring, character (brown=n, buff=b, cinnamon=c, gray=g, orange=o, pink=p, red=e, white=w, yellow=y)</p>
</dd>
<dt>scbring</dt><dd><p>stalk-color-below-ring, character (brown=n, buff=b, cinnamon=c, gray=g, orange=o, pink=p, red=e, white=w, yellow=y)</p>
</dd>
<dt>vtype</dt><dd><p>veil-type, character (partial=p, universal=u)</p>
</dd>
<dt>vcolor</dt><dd><p>veil-color, character (brown=n, orange=o, white=w, yellow=y)</p>
</dd>
<dt>ringnum</dt><dd><p>character (none=n, one=o, two=t)</p>
</dd>
<dt>ringtype</dt><dd><p>character (cobwebby=c, evanescent=e, flaring=f, large=l, none=n, pendant=p, sheathing=s, zone=z)</p>
</dd>
<dt>sporepc</dt><dd><p>spore-print-color, character (black=k, brown=n, buff=b, chocolate=h, green=r, orange=o, purple=u, white=w, yellow=y)</p>
</dd>
<dt>population</dt><dd><p>character(abundant=a, clustered=c, numerous=n, scattered=s, several=v, solitary=y)</p>
</dd>
<dt>habitat</dt><dd><p>character (grasses=g, leaves=l, meadows=m, paths=p, urban=u, waste=w, woods=d)</p>
</dd>
</dl>



<h3>Details</h3>

<p>This dataset includes descriptions of hypothetical samples corresponding to 23 species of gilled mushrooms
in the Agaricus and Lepiota Family. Each species is classified as <code>poisonous</code> (True or False).
The Guide clearly states that there is no simple rule for determining the edibility of a mushroom;
no rule like &ldquo;leaflets three, let it be&rdquo; for Poisonous Oak and Ivy.
</p>
<p>We made the following enhancements to the original data for improved usability:
</p>

<ul>
<li><p>Any missing values, denoted as &quot;?&quot; in the dataset, were transformed into NAs.
</p>
</li>
<li><p>Binary factor variables with exclusive &quot;t&quot; and &quot;f&quot; values were converted to logical <code>TRUE/FALSE</code> vectors.
</p>
</li>
<li><p>The binary factor <em>criterion</em> variable with exclusive &quot;p&quot; and &quot;e&quot; values was converted to a logical <code>TRUE/FALSE</code> vector.
</p>
</li></ul>

<p>Other than that, the data remains consistent with the original dataset.
</p>


<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Mushroom">https://archive.ics.uci.edu/ml/datasets/Mushroom</a>
</p>


<h3>References</h3>

<p>Mushroom records drawn from The Audubon Society Field Guide to North American Mushrooms (1981).
G.H. Lincoff (Pres.), New York: A.A. Knopf.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='plot.FFTrees'>Plot an <code>FFTrees</code> object</h2><span id='topic+plot.FFTrees'></span>

<h3>Description</h3>

<p><code>plot.FFTrees</code> visualizes an <code>FFTrees</code> object created by the <code><a href="#topic+FFTrees">FFTrees</a></code> function.
</p>
<p><code>plot.FFTrees</code> is the main plotting function of the <strong>FFTrees</strong> package and
called when evaluating the generic <code><a href="base.html#topic+plot">plot</a></code> on an <code>FFTrees</code> object.
</p>
<p><code>plot.FFTrees</code> visualizes a selected FFT, key data characteristics, and various aspects of classification performance.
</p>
<p>As <code>x</code> may not contain test data, <code>plot.FFTrees</code> by default plots the performance characteristics
for training data (i.e., fitting), rather than for test data (i.e., for prediction).
When test data is available, specifying <code>data = "test"</code> plots prediction performance.
</p>
<p>Whenever the sensitivity weight (<code>sens.w</code>) is set to its default of <code>sens.w = 0.50</code>,
a level shows <em>balanced</em> accuracy (<code>bacc</code>). If, however, <code>sens.w</code> deviates from its default,
the level shows the tree's <em>weighted</em> accuracy value (<code>wacc</code>) and the current <code>sens.w</code> value (below the level).
</p>
<p>Many aspects of the plot (e.g., its panels) and the FFT's appearance (e.g., labels of its nodes and exits)
can be customized by setting corresponding arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FFTrees'
plot(
  x = NULL,
  data = "train",
  what = "all",
  tree = 1,
  main = NULL,
  cue.labels = NULL,
  decision.labels = NULL,
  cue.cex = NULL,
  threshold.cex = NULL,
  decision.cex = 1,
  comp = TRUE,
  show.header = NULL,
  show.tree = NULL,
  show.confusion = NULL,
  show.levels = NULL,
  show.roc = NULL,
  show.icons = NULL,
  show.iconguide = NULL,
  hlines = TRUE,
  label.tree = NULL,
  label.performance = NULL,
  n.per.icon = NULL,
  level.type = "bar",
  which.tree = NULL,
  decision.names = NULL,
  stats = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.FFTrees_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object created by the <code><a href="#topic+FFTrees">FFTrees</a></code> function.</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_data">data</code></td>
<td>
<p>The type of data in <code>x</code> to be plotted (as a string) or a test dataset (as a data frame).
</p>

<ul>
<li><p>A valid data string must be either <code>'train'</code> (for fitting performance) or <code>'test'</code> (for prediction performance).
</p>
</li>
<li><p>For a valid data frame, the specified tree is evaluated and plotted for this data (as 'test' data),
but the global <code>FFTrees</code> object <code>x</code> remains unchanged unless it is re-assigned.
</p>
</li></ul>

<p>By default, <code>data = 'train'</code> (as <code>x</code> may not contain test data).</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_what">what</code></td>
<td>
<p>What should be plotted (as a character string)? Valid options are:
</p>

<dl>
<dt>'all'</dt><dd><p>Plot the tree diagram with all corresponding guides and performance statistics, but excluding cue accuracies.</p>
</dd>
<dt>'cues'</dt><dd><p>Plot only the marginal accuracy of cues in ROC space.
Note that cue accuracies are <em>not</em> shown when calling <code>what = 'all'</code> and use the <code><a href="#topic+showcues">showcues</a></code> function.</p>
</dd>
<dt>'icontree'</dt><dd><p>Plot tree diagram with icon arrays on exit nodes.
Consider also setting <code>n.per.icon</code> and <code>show.iconguide</code>.</p>
</dd>
<dt>'tree'</dt><dd><p>Plot only the tree diagram.</p>
</dd>
<dt>'roc'</dt><dd><p>Plot only the performance of tree(s) (and comparison algorithms) in ROC space.</p>
</dd>
</dl>

<p>Default: <code>what = 'all'</code>.</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_tree">tree</code></td>
<td>
<p>The tree to be plotted (as an integer, only valid when the corresponding tree argument is non-empty).
Default: <code>tree = 1</code>.
To plot the best training or best test tree with respect to the <code>goal</code> specified during FFT construction,
use <code>'best.train'</code> or <code>'best.test'</code>, respectively.</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_main">main</code></td>
<td>
<p>The main plot label (as a character string).</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_cue.labels">cue.labels</code></td>
<td>
<p>An optional string of labels for the cues / nodes (as character vector).</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_decision.labels">decision.labels</code></td>
<td>
<p>A character vector of length 2 indicating the content-specific names for noise and signal predictions/exits.</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_cue.cex">cue.cex</code></td>
<td>
<p>The size of the cue labels (as numeric).</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_threshold.cex">threshold.cex</code></td>
<td>
<p>The size of the threshold labels (as numeric).</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_decision.cex">decision.cex</code></td>
<td>
<p>The size of the decision labels (as numeric).</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_comp">comp</code></td>
<td>
<p>Should the performance of competitive algorithms (e.g.; logistic regression, random forests, etc.)
be shown in the ROC plot (if available, as logical)?</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_show.header">show.header</code></td>
<td>
<p>Show header with basic data properties (in top panel, as logical)?</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_show.tree">show.tree</code></td>
<td>
<p>Show nodes and exits of FFT (in middle panel, as logical)?</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_show.confusion">show.confusion</code></td>
<td>
<p>Show a 2x2 confusion matrix (in bottom panel, as logical)?</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_show.levels">show.levels</code></td>
<td>
<p>Show performance levels (in bottom panel, as logical)?</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_show.roc">show.roc</code></td>
<td>
<p>Show ROC curve (in bottom panel, as logical)?</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_show.icons">show.icons</code></td>
<td>
<p>Show exit cases as icon arrays (in middle panel, as logical)?</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_show.iconguide">show.iconguide</code></td>
<td>
<p>Show icon guide (in middle panel, as logical)?</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_hlines">hlines</code></td>
<td>
<p>Show horizontal panel separation lines (as logical)?
Default: <code>hlines = TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_label.tree">label.tree</code></td>
<td>
<p>A label for the FFT (optional, as character string).</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_label.performance">label.performance</code></td>
<td>
<p>A label for the performance section (optional, as character string).</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_n.per.icon">n.per.icon</code></td>
<td>
<p>The number of cases represented by each icon (as numeric).</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_level.type">level.type</code></td>
<td>
<p>The type of performance levels to be drawn at the bottom (as character string, either <code>"bar"</code> or <code>"line"</code>.
Default: <code>level.type = "bar"</code>.</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_which.tree">which.tree</code></td>
<td>
<p>Deprecated argument. Use <code>tree</code> instead.</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_decision.names">decision.names</code></td>
<td>
<p>Deprecated argument. Use <code>decision.labels</code> instead.</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_stats">stats</code></td>
<td>
<p>Deprecated argument. Should statistical information be plotted (as logical)?
Use <code>what = "all"</code> to include performance statistics
and <code>what = "tree"</code> to plot only a tree diagram.</p>
</td></tr>
<tr><td><code id="plot.FFTrees_+3A_...">...</code></td>
<td>
<p>Graphical parameters (passed to text of panel titles,
to <code><a href="#topic+showcues">showcues</a></code> when <code>what = 'cues'</code>, or
to <code><a href="graphics.html#topic+title">title</a></code> when <code>what = 'roc'</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An invisible <code>FFTrees</code> object <code>x</code>
and a plot visualizing and describing an FFT (as side effect).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+showcues">showcues</a></code> for plotting cue accuracies;
<code><a href="#topic+print.FFTrees">print.FFTrees</a></code> for printing FFTs;
<code><a href="#topic+summary.FFTrees">summary.FFTrees</a></code> for summarizing FFTs;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other plot functions: 
<code><a href="#topic+showcues">showcues</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create FFTs (for heartdisease data):
heart_fft &lt;- FFTrees(formula = diagnosis ~ .,
                     data = heart.train)

# Visualize the default FFT (Tree #1, what = 'all'):
plot(heart_fft, main = "Heart disease",
     decision.labels = c("Absent", "Present"))

# Visualize cue accuracies (in ROC space):
plot(heart_fft, what = "cues",  main = "Cue accuracies for heart disease data")

# Visualize tree diagram with icon arrays on exit nodes:
plot(heart_fft, what = "icontree", n.per.icon = 2,
     main = "Diagnosing heart disease")

# Visualize performance comparison in ROC space:
plot(heart_fft, what = "roc", main = "Performance comparison for heart disease data")

# Visualize predictions of FFT #2 (for new test data) with custom options:
plot(heart_fft, tree = 2, data = heart.test,
     main = "Predicting heart disease",
     cue.labels = c("1. thal?", "2. cp?", "3. ca?", "4. exang"),
     decision.labels = c("ok", "sick"), n.per.icon = 2,
     show.header = TRUE, show.confusion = FALSE, show.levels = FALSE, show.roc = FALSE,
     hlines = FALSE, font = 3, col = "steelblue")

# # For details, see
# vignette("FFTrees_plot", package = "FFTrees")


</code></pre>

<hr>
<h2 id='predict.FFTrees'>Predict classification outcomes or probabilities from data</h2><span id='topic+predict.FFTrees'></span>

<h3>Description</h3>

<p><code>predict.FFTrees</code> predicts binary classification outcomes or their probabilities from <code>newdata</code>
for an <code>FFTrees</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FFTrees'
predict(
  object = NULL,
  newdata = NULL,
  tree = 1,
  type = "class",
  sens.w = NULL,
  method = "laplace",
  data = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.FFTrees_+3A_object">object</code></td>
<td>
<p>An <code>FFTrees</code> object created by the <code><a href="#topic+FFTrees">FFTrees</a></code> function.</p>
</td></tr>
<tr><td><code id="predict.FFTrees_+3A_newdata">newdata</code></td>
<td>
<p>dataframe. A data frame of test data.</p>
</td></tr>
<tr><td><code id="predict.FFTrees_+3A_tree">tree</code></td>
<td>
<p>integer. Which tree in the object should be used? By default, <code>tree = 1</code> is used.</p>
</td></tr>
<tr><td><code id="predict.FFTrees_+3A_type">type</code></td>
<td>
<p>string. What should be predicted? Can be <code>"class"</code>, which returns a vector of class predictions, <code>"prob"</code> which returns a matrix of class probabilities,
or <code>"both"</code> which returns a matrix with both class and probability predictions.</p>
</td></tr>
<tr><td><code id="predict.FFTrees_+3A_sens.w">sens.w</code>, <code id="predict.FFTrees_+3A_data">data</code></td>
<td>
<p>deprecated</p>
</td></tr>
<tr><td><code id="predict.FFTrees_+3A_method">method</code></td>
<td>
<p>string. Method of calculating class probabilities. Either 'laplace', which applies the Laplace correction, or 'raw' which applies no correction.</p>
</td></tr>
<tr><td><code id="predict.FFTrees_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to <code>predict</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either a logical vector of predictions, or a matrix of class probabilities.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+print.FFTrees">print.FFTrees</a></code> for printing FFTs;
<code><a href="#topic+plot.FFTrees">plot.FFTrees</a></code> for plotting FFTs;
<code><a href="#topic+summary.FFTrees">summary.FFTrees</a></code> for summarizing FFTs;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create training and test data:
set.seed(100)
breastcancer &lt;- breastcancer[sample(nrow(breastcancer)), ]
breast.train &lt;- breastcancer[1:150, ]
breast.test  &lt;- breastcancer[151:303, ]

# Create an FFTrees object from the training data:
breast.fft &lt;- FFTrees(
  formula = diagnosis ~ .,
  data = breast.train
)

# Predict classification outcomes for test data:
breast.fft.pred &lt;- predict(breast.fft,
  newdata = breast.test
)

# Predict class probabilities for test data:
breast.fft.pred &lt;- predict(breast.fft,
  newdata = breast.test,
  type = "prob"
)

</code></pre>

<hr>
<h2 id='print.FFTrees'>Print basic information of fast-and-frugal trees (FFTs)</h2><span id='topic+print.FFTrees'></span>

<h3>Description</h3>

<p><code>print.FFTrees</code> prints basic information on FFTs for an <code>FFTrees</code> object <code>x</code>.
</p>
<p>As <code>x</code> may not contain test data, <code>print.FFTrees</code> by default prints the performance characteristics for training data (i.e., fitting), rather than for test data (i.e., for prediction).
When test data is available, specify <code>data = "test"</code> to print prediction performance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FFTrees'
print(x = NULL, tree = 1, data = "train", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.FFTrees_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object created by <code><a href="#topic+FFTrees">FFTrees</a></code>.</p>
</td></tr>
<tr><td><code id="print.FFTrees_+3A_tree">tree</code></td>
<td>
<p>The tree to be printed (as an integer, only valid when the corresponding tree argument is non-empty).
Default: <code>tree = 1</code>.
To print the best training or best test tree with respect to the <code>goal</code> specified during FFT construction,
use <code>"best.train"</code> or <code>"best.test"</code>, respectively.</p>
</td></tr>
<tr><td><code id="print.FFTrees_+3A_data">data</code></td>
<td>
<p>The type of data in <code>x</code> to be printed (as a string) or a test dataset (as a data frame).
</p>

<ul>
<li><p>A valid data string must be either <code>'train'</code> (for fitting performance) or <code>'test'</code> (for prediction performance).
</p>
</li>
<li><p>For a valid data frame, the specified tree is evaluated and printed for this data (as 'test' data),
but the global <code>FFTrees</code> object <code>x</code> remains unchanged unless it is re-assigned.
</p>
</li></ul>

<p>By default, <code>data = 'train'</code> (as <code>x</code> may not contain test data).</p>
</td></tr>
<tr><td><code id="print.FFTrees_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code>print</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An invisible <code>FFTrees</code> object <code>x</code>
and summary information on an FFT printed to the console (as side effect).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.FFTrees">plot.FFTrees</a></code> for plotting FFTs;
<code><a href="#topic+summary.FFTrees">summary.FFTrees</a></code> for summarizing FFTs;
<code><a href="#topic+inwords">inwords</a></code> for obtaining a verbal description of FFTs;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>

<hr>
<h2 id='read_fft_df'>Read an FFT definition from tree definitions</h2><span id='topic+read_fft_df'></span>

<h3>Description</h3>

<p><code>read_fft_df</code> reads and returns
the definition of a single FFT (as a tidy data frame)
from the multi-line FFT definitions of an <code>FFTrees</code> object.
</p>
<p><code>read_fft_df</code> allows reading individual tree definitions
to manipulate them with other tree trimming functions.
</p>
<p><code><a href="#topic+write_fft_df">write_fft_df</a></code> provides the inverse functionality.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_fft_df(ffts_df, tree = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_fft_df_+3A_ffts_df">ffts_df</code></td>
<td>
<p>A set of FFT definitions (as a data frame,
usually from an <code>FFTrees</code> object,
with suitable variable names to pass <code>verify_ffts_df</code>.</p>
</td></tr>
<tr><td><code id="read_fft_df_+3A_tree">tree</code></td>
<td>
<p>The ID of the to-be-selected FFT (as an integer),
corresponding to a tree in <code>ffts_df</code>.
Default: <code>tree = 1</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>One FFT definition
(as a data frame in tidy format, with one row per node).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_fft_df">get_fft_df</a></code> for getting the FFT definitions of an <code>FFTrees</code> object;
<code><a href="#topic+write_fft_df">write_fft_df</a></code> for writing one FFT to tree definitions;
<code><a href="#topic+add_fft_df">add_fft_df</a></code> for adding FFTs to tree definitions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other tree definition and manipulation functions: 
<code><a href="#topic+add_fft_df">add_fft_df</a>()</code>,
<code><a href="#topic+add_nodes">add_nodes</a>()</code>,
<code><a href="#topic+drop_nodes">drop_nodes</a>()</code>,
<code><a href="#topic+edit_nodes">edit_nodes</a>()</code>,
<code><a href="#topic+flip_exits">flip_exits</a>()</code>,
<code><a href="#topic+get_fft_df">get_fft_df</a>()</code>,
<code><a href="#topic+reorder_nodes">reorder_nodes</a>()</code>,
<code><a href="#topic+select_nodes">select_nodes</a>()</code>,
<code><a href="#topic+write_fft_df">write_fft_df</a>()</code>
</p>

<hr>
<h2 id='reorder_nodes'>Reorder nodes in an FFT definition</h2><span id='topic+reorder_nodes'></span>

<h3>Description</h3>

<p><code>reorder_nodes</code> allows reordering
the <code>nodes</code> in an existing FFT definition
(in the tidy data frame format).
</p>
<p><code>reorder_nodes</code> allows to directly set and change the node
order in an FFT definition by specifying <code>nodes</code>.
</p>
<p>When a former non-final node becomes a final node,
the exit type of the former final node
is set to the signal value (i.e., <code>exit_types[2]</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reorder_nodes(fft, order = NA, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reorder_nodes_+3A_fft">fft</code></td>
<td>
<p>One FFT definition
(as a data frame in tidy format, with one row per node).</p>
</td></tr>
<tr><td><code id="reorder_nodes_+3A_order">order</code></td>
<td>
<p>The desired node order (as an integer vector).
The values of <code>order</code> must be a permutation of <code>1:nrow(fft)</code>.
Default: <code>order = NA</code>.</p>
</td></tr>
<tr><td><code id="reorder_nodes_+3A_quiet">quiet</code></td>
<td>
<p>Hide feedback messages (as logical)?
Default: <code>quiet = FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>One FFT definition
(as a data frame in tidy format, with one row per node).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_nodes">add_nodes</a></code> for adding nodes to an FFT definition;
<code><a href="#topic+edit_nodes">edit_nodes</a></code> for editing nodes in an FFT definition;
<code><a href="#topic+drop_nodes">drop_nodes</a></code> for deleting nodes from an FFT definition;
<code><a href="#topic+flip_exits">flip_exits</a></code> for reversing exits in an FFT definition;
<code><a href="#topic+select_nodes">select_nodes</a></code> for selecting nodes in an FFT definition;
<code><a href="#topic+get_fft_df">get_fft_df</a></code> for getting the FFT definitions of an <code>FFTrees</code> object;
<code><a href="#topic+read_fft_df">read_fft_df</a></code> for reading one FFT definition from tree definitions;
<code><a href="#topic+add_fft_df">add_fft_df</a></code> for adding FFTs to tree definitions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other tree definition and manipulation functions: 
<code><a href="#topic+add_fft_df">add_fft_df</a>()</code>,
<code><a href="#topic+add_nodes">add_nodes</a>()</code>,
<code><a href="#topic+drop_nodes">drop_nodes</a>()</code>,
<code><a href="#topic+edit_nodes">edit_nodes</a>()</code>,
<code><a href="#topic+flip_exits">flip_exits</a>()</code>,
<code><a href="#topic+get_fft_df">get_fft_df</a>()</code>,
<code><a href="#topic+read_fft_df">read_fft_df</a>()</code>,
<code><a href="#topic+select_nodes">select_nodes</a>()</code>,
<code><a href="#topic+write_fft_df">write_fft_df</a>()</code>
</p>

<hr>
<h2 id='select_nodes'>Select nodes from an FFT definition</h2><span id='topic+select_nodes'></span>

<h3>Description</h3>

<p><code>select_nodes</code> selects
one or more <code>nodes</code> from an existing FFT definition
(by filtering the corresponding row(s) from the FFT definition
in the tidy data frame format).
</p>
<p>When not selecting the final node,
the last selected node becomes the new final node
(i.e., gains a second exit).
</p>
<p>Duplicates in <code>nodes</code> are selected only once
(rather than incrementally) and <code>nodes</code> not in
the range <code>1:nrow(fft)</code> are ignored.
</p>
<p><code>select_nodes</code> is the inverse function
of <code><a href="#topic+drop_nodes">drop_nodes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_nodes(fft, nodes = NA, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_nodes_+3A_fft">fft</code></td>
<td>
<p>One FFT definition
(as a data frame in tidy format, with one row per node).</p>
</td></tr>
<tr><td><code id="select_nodes_+3A_nodes">nodes</code></td>
<td>
<p>The FFT nodes to select (as an integer vector).
Default: <code>nodes = NA</code>.</p>
</td></tr>
<tr><td><code id="select_nodes_+3A_quiet">quiet</code></td>
<td>
<p>Hide feedback messages (as logical)?
Default: <code>quiet = FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>One FFT definition
(as a data frame in tidy format, with one row per node).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_nodes">add_nodes</a></code> for adding nodes to an FFT definition;
<code><a href="#topic+drop_nodes">drop_nodes</a></code> for deleting nodes from an FFT definition;
<code><a href="#topic+edit_nodes">edit_nodes</a></code> for editing nodes in an FFT definition;
<code><a href="#topic+flip_exits">flip_exits</a></code> for reversing exits in an FFT definition;
<code><a href="#topic+reorder_nodes">reorder_nodes</a></code> for reordering nodes of an FFT definition;
<code><a href="#topic+get_fft_df">get_fft_df</a></code> for getting the FFT definitions of an <code>FFTrees</code> object;
<code><a href="#topic+read_fft_df">read_fft_df</a></code> for reading one FFT definition from tree definitions;
<code><a href="#topic+add_fft_df">add_fft_df</a></code> for adding FFTs to tree definitions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other tree definition and manipulation functions: 
<code><a href="#topic+add_fft_df">add_fft_df</a>()</code>,
<code><a href="#topic+add_nodes">add_nodes</a>()</code>,
<code><a href="#topic+drop_nodes">drop_nodes</a>()</code>,
<code><a href="#topic+edit_nodes">edit_nodes</a>()</code>,
<code><a href="#topic+flip_exits">flip_exits</a>()</code>,
<code><a href="#topic+get_fft_df">get_fft_df</a>()</code>,
<code><a href="#topic+read_fft_df">read_fft_df</a>()</code>,
<code><a href="#topic+reorder_nodes">reorder_nodes</a>()</code>,
<code><a href="#topic+write_fft_df">write_fft_df</a>()</code>
</p>

<hr>
<h2 id='showcues'>Visualize cue accuracies (as points in ROC space)</h2><span id='topic+showcues'></span>

<h3>Description</h3>

<p><code>showcues</code> plots the cue accuracies of an <code>FFTrees</code> object
created by the <code><a href="#topic+FFTrees">FFTrees</a></code> function (as points in ROC space).
</p>
<p>If the optional arguments <code>cue.accuracies</code> and <code>alt.goal</code> are specified,
their values take precedence over the corresponding settings of an <code>FFTrees</code> object <code>x</code>
(but do not change <code>x</code>).
</p>
<p><code>showcues</code> is called when the main <code><a href="#topic+plot.FFTrees">plot.FFTrees</a></code> function is set to <code>what = "cues"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showcues(
  x = NULL,
  cue.accuracies = NULL,
  alt.goal = NULL,
  main = NULL,
  top = 5,
  quiet = list(ini = TRUE, fin = FALSE, set = TRUE),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="showcues_+3A_x">x</code></td>
<td>
<p>An <code>FFTrees</code> object created by the <code><a href="#topic+FFTrees">FFTrees</a></code> function.</p>
</td></tr>
<tr><td><code id="showcues_+3A_cue.accuracies">cue.accuracies</code></td>
<td>
<p>An optional data frame specifying cue accuracies directly (without specifying <code>FFTrees</code> object <code>x</code>).</p>
</td></tr>
<tr><td><code id="showcues_+3A_alt.goal">alt.goal</code></td>
<td>
<p>An optional alternative goal to sort the current cue accuracies (without using the goal of <code>FFTrees</code> object <code>x</code>).</p>
</td></tr>
<tr><td><code id="showcues_+3A_main">main</code></td>
<td>
<p>A main plot title (as character string).</p>
</td></tr>
<tr><td><code id="showcues_+3A_top">top</code></td>
<td>
<p>How many of the top cues should be highlighted (as an integer)?</p>
</td></tr>
<tr><td><code id="showcues_+3A_quiet">quiet</code></td>
<td>
<p>Should user feedback messages be suppressed (as a list of 3 logical arguments)?
Default: <code>quiet = list(ini = TRUE, fin = FALSE, set = FALSE)</code>.</p>
</td></tr>
<tr><td><code id="showcues_+3A_...">...</code></td>
<td>
<p>Graphical parameters (passed to <code><a href="base.html#topic+plot">plot</a></code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot showing cue accuracies (of an <code>FFTrees</code> object) (as points in ROC space).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+print.FFTrees">print.FFTrees</a></code> for printing FFTs;
<code><a href="#topic+plot.FFTrees">plot.FFTrees</a></code> for plotting FFTs;
<code><a href="#topic+summary.FFTrees">summary.FFTrees</a></code> for summarizing FFTs;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other plot functions: 
<code><a href="#topic+plot.FFTrees">plot.FFTrees</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create fast-and-frugal trees (FFTs) for heart disease:
heart.fft &lt;- FFTrees(formula = diagnosis ~ .,
                     data = heart.train,
                     data.test = heart.test,
                     main = "Heart Disease",
                     decision.labels = c("Healthy", "Diseased")
                     )

# Show cue accuracies (in ROC space):
showcues(heart.fft,
         main = "Predicting heart disease")

</code></pre>

<hr>
<h2 id='sonar'>Sonar data</h2><span id='topic+sonar'></span>

<h3>Description</h3>

<p>The file contains patterns of sonar signals bounced off a metal cylinder or bounced off a roughly cylindrical rock at various angles and under various conditions.
The transmitted sonar signal is a frequency-modulated chirp, rising in frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sonar
</code></pre>


<h3>Format</h3>

<p>A data frame containing 208 rows and 60 columns.
</p>

<dl>
<dt>V1</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V2</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V3</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V4</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V5</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V6</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V7</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V8</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V9</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V10</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V11</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V12</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V13</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V14</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V15</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V16</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V17</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V18</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V19</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V20</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V21</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V22</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V23</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V24</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V25</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V26</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V27</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V28</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V29</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V30</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V31</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V32</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V33</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V34</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V35</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V36</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V37</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V38</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V39</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V40</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V41</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V42</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V43</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V44</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V45</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V46</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V47</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V48</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V49</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V50</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V51</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V52</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V53</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V54</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V55</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V56</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V57</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V58</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V59</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>V60</dt><dd><p>Number in the range 0.0 to 1.0 that represents the energy within a particular frequency band, integrated over a certain period of time.</p>
</dd>
<dt>mine.crit</dt><dd><p><em>Criterion</em>: Did a sonar signal bounce off a metal cylinder (or a rock)?
</p>
<p>Values: <code>TRUE</code> (metal cylinder) vs. <code>FALSE</code> (rock) (53.37% vs.\ 46.63%).</p>
</dd>
</dl>



<h3>Details</h3>

<p>We made the following enhancements to the original data for improved usability:
</p>

<ul>
<li><p>The binary factor <em>criterion</em> variable with exclusive &quot;m&quot; and &quot;r&quot; values was converted to a logical <code>TRUE/FALSE</code> vector.
</p>
</li></ul>

<p>Other than that, the data remains consistent with the original dataset.
</p>


<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Connectionist+Bench+(Sonar,+Mines+vs.+Rocks)">https://archive.ics.uci.edu/ml/datasets/Connectionist+Bench+(Sonar,+Mines+vs.+Rocks)</a>
</p>


<h3>References</h3>

<p>Gorman, R. P., and Sejnowski, T. J. (1988). &quot;Analysis of Hidden Units in a Layered Network Trained to Classify Sonar Targets&quot; in Neural Networks, Vol. 1, pp. 75-89.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='summary.FFTrees'>Summarize an <code>FFTrees</code> object</h2><span id='topic+summary.FFTrees'></span>

<h3>Description</h3>

<p><code>summary.FFTrees</code> summarizes key contents of an <code>FFTrees</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FFTrees'
summary(object, tree = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.FFTrees_+3A_object">object</code></td>
<td>
<p>An <code>FFTrees</code> object.</p>
</td></tr>
<tr><td><code id="summary.FFTrees_+3A_tree">tree</code></td>
<td>
<p>The tree to summarize (as an integer, but may be a vector).
If <code>tree = NULL</code> (as per default) or exceeding the possible range <code>1:object$trees$n</code>,
information on all trees in <code>object</code> is returned.</p>
</td></tr>
<tr><td><code id="summary.FFTrees_+3A_...">...</code></td>
<td>
<p>Additional arguments (currently ignored).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given an <code>FFTrees</code> object <code>x</code>,
<code>summary.FFTrees</code> selects key parameters from <code>x$params</code>
and provides the definitions and performance statistics for <code>tree</code> from <code>x$trees</code>.
Inspect and query <code>x</code> for additional details.
</p>
<p><code>summary.FFTrees</code> returns an invisible list containing two elements:
</p>

<ol>
<li> <p><code>definitions</code> and corresponding performance measures of <code>tree</code>s;
</p>
</li>
<li> <p><code>stats</code> on decision frequencies, derived probabilities, and costs (separated by <code>train</code> and <code>test</code>).
</p>
</li></ol>

<p>A header prints descriptive information of the <code>FFTrees</code> object (to the console):
Its <code>main</code> title, number of trees (<code>object$trees$n</code>), and the name of the criterion variable (<code>object$criterion_name</code>).
</p>
<p>Per default, information on all available trees is shown and returned.
Specifying <code>tree</code> filters the output list elements for the corresponding tree(s).
When only a single <code>tree</code> is specified, the printed header includes a verbal description of
the corresponding tree.
</p>
<p>While <code>summary.FFTrees</code> provides key details about the specified <code>tree</code>(s),
the individual decisions (stored in <code>object$trees$decisions</code>) are not shown or returned.
</p>


<h3>Value</h3>

<p>An invisible list with elements containing the <code>definitions</code> and performance <code>stats</code>
of the FFT(s) specified by <code>tree</code>(s).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+print.FFTrees">print.FFTrees</a></code> for printing FFTs;
<code><a href="#topic+plot.FFTrees">plot.FFTrees</a></code> for plotting FFTs;
<code><a href="#topic+inwords">inwords</a></code> for obtaining a verbal description of FFTs;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>

<hr>
<h2 id='titanic'>Titanic survival data</h2><span id='topic+titanic'></span>

<h3>Description</h3>

<p>Data indicating who survived on the Titanic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>titanic
</code></pre>


<h3>Format</h3>

<p>A data frame containing 2,201 rows and 4 columns.
</p>

<dl>
<dt>class</dt><dd><p>Factor - Class (first, second, third, or crew)</p>
</dd>
<dt>age</dt><dd><p>Factor - Age group (child or adult)</p>
</dd>
<dt>sex</dt><dd><p>Factor - Sex (male or female)</p>
</dd>
<dt>survived</dt><dd><p>Logical - Whether the passenger survived (TRUE) or not (FALSE)</p>
</dd>
</dl>



<h3>Details</h3>

<p>See <code><a href="datasets.html#topic+Titanic">Titanic</a></code> of the R <strong>datasets</strong> package for details and
the same data (in a 4-dimensional <code>table</code>).
</p>


<h3>Source</h3>

<p><a href="https://www.encyclopedia-titanica.org">https://www.encyclopedia-titanica.org</a>
</p>


<h3>References</h3>

<p>Dawson, Robert J. MacG. (1995), The Unusual Episode Data Revisited. Journal of Statistics Education, 3.
doi: 10.1080/10691898.1995.11910499.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+voting">voting</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='voting'>Voting data</h2><span id='topic+voting'></span>

<h3>Description</h3>

<p>A dataset of votes for each of the U.S. House of Representatives Congressmen on the 16 key votes identified by the CQA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>voting
</code></pre>


<h3>Format</h3>

<p>A data frame containing 435 rows and 16 columns.
</p>

<dl>
<dt>handicapped</dt><dd><p>handicapped-infants, logical (TRUE, FALSE)</p>
</dd>
<dt>water</dt><dd><p>water-project-cost-sharing, logical (TRUE, FALSE)</p>
</dd>
<dt>adoption</dt><dd><p>adoption-of-the-budget-resolution, logical (TRUE, FALSE)</p>
</dd>
<dt>physician</dt><dd><p>physician-fee-freeze, logical (TRUE, FALSE)</p>
</dd>
<dt>elsalvador</dt><dd><p>el-salvador-aid, logical (TRUE, FALSE)</p>
</dd>
<dt>religionschool</dt><dd><p>religious-groups-in-schools, logical (TRUE, FALSE)</p>
</dd>
<dt>satellite</dt><dd><p>anti-satellite-test-ban, logical (TRUE, FALSE)</p>
</dd>
<dt>nicaraguan</dt><dd><p>aid-to-nicaraguan-contras, logical (TRUE, FALSE)</p>
</dd>
<dt>mxmissile</dt><dd><p>mxmissile, logical (TRUE, FALSE)</p>
</dd>
<dt>immigration</dt><dd><p>immigration, logical (TRUE, FALSE)</p>
</dd>
<dt>synfuels</dt><dd><p>synfuels-corporation-cutback, logical (TRUE, FALSE)</p>
</dd>
<dt>education</dt><dd><p>education-spending, logical (TRUE, FALSE)</p>
</dd>
<dt>superfund</dt><dd><p>superfund-right-to-sue, logical (TRUE, FALSE)</p>
</dd>
<dt>crime</dt><dd><p>crime, logical (TRUE, FALSE)</p>
</dd>
<dt>dutyfree</dt><dd><p>duty-free-exports, logical (TRUE, FALSE)</p>
</dd>
<dt>southafrica</dt><dd><p>export-administration-act-south-africa, logical (TRUE, FALSE)</p>
</dd>
<dt>party.crit</dt><dd><p><em>Criterion</em>: Where the voters democratic (or republican) congressmen?
</p>
<p>Values: <code>TRUE</code> (democrat) / <code>FALSE</code> (republican) (61.52% vs. 38.48%).</p>
</dd>
</dl>



<h3>Details</h3>

<p>The CQA lists nine different types of votes: voted for, paired for, and announced for (these three simplified to yea),
voted against, paired against, and announced against (these three simplified to nay),
voted present, voted present to avoid conflict of interest, and did not vote or otherwise make a position known
(these three simplified to an unknown disposition).
</p>
<p>We made the following enhancements to the original data for improved usability:
</p>

<ul>
<li><p>Any missing values, denoted as &quot;?&quot; in the dataset, were transformed into NAs.
</p>
</li>
<li><p>Binary factor variables with exclusive &quot;y&quot; and &quot;n&quot; values were converted to logical TRUE/FALSE vectors.
</p>
</li>
<li><p>The binary character <em>criterion</em> variable with exclusive &quot;democrat&quot; and &quot;republican&quot; values was converted to a logical <code>TRUE/FALSE</code> vector.
</p>
</li></ul>

<p>Other than that, the data remains consistent with the original dataset.
</p>


<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records">https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records</a>
</p>


<h3>References</h3>

<p>Congressional Quarterly Almanac, 98th Congress, 2nd session 1984,
Volume XL: Congressional Quarterly Inc.
Washington, D.C., 1985.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+wine">wine</a></code>
</p>

<hr>
<h2 id='wine'>Wine tasting data</h2><span id='topic+wine'></span>

<h3>Description</h3>

<p>Chemical and tasting data from wines in North Portugal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wine
</code></pre>


<h3>Format</h3>

<p>A data frame containing 6497 rows and 13 columns.
</p>

<dl>
<dt>fixed.acidity</dt><dd><p>fixed acidity (nummeric)</p>
</dd>
<dt>volatile.acidity</dt><dd><p>volatile acidity (nummeric)</p>
</dd>
<dt>citric.acid</dt><dd><p>citric acid (nummeric)</p>
</dd>
<dt>residual.sugar</dt><dd><p>residual sugar (nummeric)</p>
</dd>
<dt>chlorides</dt><dd><p>chlorides (nummeric)</p>
</dd>
<dt>free.sulfur.dioxide</dt><dd><p>free sulfur dioxide (nummeric)</p>
</dd>
<dt>total.sulfur.dioxide</dt><dd><p>total sulfur dioxide (nummeric)</p>
</dd>
<dt>density</dt><dd><p>density (nummeric)</p>
</dd>
<dt>pH</dt><dd><p>PH Value (nummeric)</p>
</dd>
<dt>sulphates</dt><dd><p>Sulphates (nummeric)</p>
</dd>
<dt>alcohol</dt><dd><p>Alcohol (nummeric)</p>
</dd>
<dt>quality</dt><dd><p>Quality (nummeric, score between 0 and 10)</p>
</dd>
<dt>type</dt><dd><p><em>Criterion</em>: Is the wine <code>red</code> or <code>white</code>? (24.61% vs.75.39%)</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="http://archive.ics.uci.edu/ml/datasets/Wine+Quality">http://archive.ics.uci.edu/ml/datasets/Wine+Quality</a>
</p>


<h3>References</h3>

<p>P. Cortez, A. Cerdeira, F. Almeida, T. Matos and J. Reis.
Modeling wine preferences by data mining from physicochemical properties.
In Decision Support Systems, <em>Elsevier</em>, 47(4):547-553. ISSN: 0167-9236.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+blood">blood</a></code>,
<code><a href="#topic+breastcancer">breastcancer</a></code>,
<code><a href="#topic+car">car</a></code>,
<code><a href="#topic+contraceptive">contraceptive</a></code>,
<code><a href="#topic+creditapproval">creditapproval</a></code>,
<code><a href="#topic+fertility">fertility</a></code>,
<code><a href="#topic+forestfires">forestfires</a></code>,
<code><a href="#topic+heart.cost">heart.cost</a></code>,
<code><a href="#topic+heart.test">heart.test</a></code>,
<code><a href="#topic+heart.train">heart.train</a></code>,
<code><a href="#topic+heartdisease">heartdisease</a></code>,
<code><a href="#topic+iris.v">iris.v</a></code>,
<code><a href="#topic+mushrooms">mushrooms</a></code>,
<code><a href="#topic+sonar">sonar</a></code>,
<code><a href="#topic+titanic">titanic</a></code>,
<code><a href="#topic+voting">voting</a></code>
</p>

<hr>
<h2 id='write_fft_df'>Write an FFT definition to tree definitions</h2><span id='topic+write_fft_df'></span>

<h3>Description</h3>

<p><code>write_fft_df</code> writes
the definition of a single FFT (as a tidy data frame)
into the one-line FFT definition used by an <code>FFTrees</code> object.
</p>
<p><code>write_fft_df</code> allows turning individual tree definitions
into the one-line FFT definition format
used by an <code>FFTrees</code> object.
</p>
<p><code><a href="#topic+read_fft_df">read_fft_df</a></code> provides the inverse functionality.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_fft_df(fft, tree = -99L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write_fft_df_+3A_fft">fft</code></td>
<td>
<p>One FFT definition
(as a data frame in tidy format, with one row per node).</p>
</td></tr>
<tr><td><code id="write_fft_df_+3A_tree">tree</code></td>
<td>
<p>The ID of the to-be-written FFT (as an integer).
Default: <code>tree = -99L</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An FFT definition in the one line
FFT definition format used by an <code>FFTrees</code> object
(as a data frame).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_fft_df">get_fft_df</a></code> for getting the FFT definitions of an <code>FFTrees</code> object;
<code><a href="#topic+read_fft_df">read_fft_df</a></code> for reading one FFT definition from tree definitions;
<code><a href="#topic+add_fft_df">add_fft_df</a></code> for adding FFTs to tree definitions;
<code><a href="#topic+FFTrees">FFTrees</a></code> for creating FFTs from and applying them to data.
</p>
<p>Other tree definition and manipulation functions: 
<code><a href="#topic+add_fft_df">add_fft_df</a>()</code>,
<code><a href="#topic+add_nodes">add_nodes</a>()</code>,
<code><a href="#topic+drop_nodes">drop_nodes</a>()</code>,
<code><a href="#topic+edit_nodes">edit_nodes</a>()</code>,
<code><a href="#topic+flip_exits">flip_exits</a>()</code>,
<code><a href="#topic+get_fft_df">get_fft_df</a>()</code>,
<code><a href="#topic+read_fft_df">read_fft_df</a>()</code>,
<code><a href="#topic+reorder_nodes">reorder_nodes</a>()</code>,
<code><a href="#topic+select_nodes">select_nodes</a>()</code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
