<!DOCTYPE html><html lang="en"><head><title>Help for package FGRepo</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FGRepo}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cytoband_grch37'><p>Cytoband information (GRCh37)</p></a></li>
<li><a href='#cytoband_grch38'><p>Cytoband information (GRCh38)</p></a></li>
<li><a href='#gene_names_grch37'><p>Gene Names (GRCh37)</p></a></li>
<li><a href='#gene_names_grch38'><p>Gene Names (GRCh38)</p></a></li>
<li><a href='#GWASCat_graph'><p>GWAScatalog graph</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Functional Genomics Repository for POST-GWAS Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-11-07</td>
</tr>
<tr>
<td>Author:</td>
<td>Alireza Ani [aut, cre],
  Ahmad Vaez [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alireza Ani &lt;a.ani@umcg.nl&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of datasets essential for functional genomic analysis. Gene names, gene positions, cytoband information, sourced from Ensembl and phenotypes association graph prepared from GWAScatalog are included. Data is available in both GRCh37 and 38 builds. These datasets facilitate a wide range of genomic studies, including the identification of genetic variants, exploration of genomic features, and post-GWAS functional analysis. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-07 06:27:56 UTC; Alireza</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-08 15:10:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='cytoband_grch37'>Cytoband information (GRCh37)</h2><span id='topic+cytoband_grch37'></span>

<h3>Description</h3>

<p>A dataset containing cytoband information based on GRCh37.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cytoband_grch37
</code></pre>


<h3>Format</h3>

<p>A data frame with 4 columns:
</p>

<dl>
<dt>chr</dt><dd><p>chromosome</p>
</dd>
<dt>start</dt><dd><p>start position</p>
</dd>
<dt>end</dt><dd><p>end position</p>
</dd>
<dt>band</dt><dd><p>band name</p>
</dd>
</dl>



<h3>Source</h3>

<p>Ensembl, <a href="https://www.ensembl.org">https://www.ensembl.org</a>
</p>


<h3>References</h3>

<p>Harrison, PW, et al. (2024). Ensembl 2024. Nucleic Acids Research, 52, Pages D891–D899. <a href="https://doi.org/10.1093/nar/gkad1049">doi:10.1093/nar/gkad1049</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cytoband_grch37)
</code></pre>

<hr>
<h2 id='cytoband_grch38'>Cytoband information (GRCh38)</h2><span id='topic+cytoband_grch38'></span>

<h3>Description</h3>

<p>A dataset containing cytoband information based on GRCh38.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cytoband_grch38
</code></pre>


<h3>Format</h3>

<p>A data frame with 4 columns:
</p>

<dl>
<dt>chr</dt><dd><p>chromosome</p>
</dd>
<dt>start</dt><dd><p>start position</p>
</dd>
<dt>end</dt><dd><p>end position</p>
</dd>
<dt>band</dt><dd><p>band name</p>
</dd>
</dl>



<h3>Source</h3>

<p>Ensembl, <a href="https://www.ensembl.org">https://www.ensembl.org</a>
</p>


<h3>References</h3>

<p>Harrison, PW, et al. (2024). Ensembl 2024. Nucleic Acids Research, 52, Pages D891–D899. <a href="https://doi.org/10.1093/nar/gkad1049">doi:10.1093/nar/gkad1049</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cytoband_grch38)
</code></pre>

<hr>
<h2 id='gene_names_grch37'>Gene Names (GRCh37)</h2><span id='topic+gene_names_grch37'></span>

<h3>Description</h3>

<p>A dataset containing gene names based on GRCh37.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_names_grch37
</code></pre>


<h3>Format</h3>

<p>A data frame with 6 columns:
</p>

<dl>
<dt>id</dt><dd><p>Gene identifier</p>
</dd>
<dt>name</dt><dd><p>Gene name</p>
</dd>
<dt>chr</dt><dd><p>Chromosome</p>
</dd>
<dt>start</dt><dd><p>Start position</p>
</dd>
<dt>end</dt><dd><p>End position</p>
</dd>
<dt>type</dt><dd><p>Gene type</p>
</dd>
</dl>



<h3>Source</h3>

<p>Ensembl, <a href="https://www.ensembl.org">https://www.ensembl.org</a>
</p>


<h3>References</h3>

<p>Harrison, PW, et al. (2024). Ensembl 2024. Nucleic Acids Research, 52, Pages D891–D899. <a href="https://doi.org/10.1093/nar/gkad1049">doi:10.1093/nar/gkad1049</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gene_names_grch37)
</code></pre>

<hr>
<h2 id='gene_names_grch38'>Gene Names (GRCh38)</h2><span id='topic+gene_names_grch38'></span>

<h3>Description</h3>

<p>A dataset containing gene names based on GRCh38.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_names_grch38
</code></pre>


<h3>Format</h3>

<p>A data frame with 6 columns:
</p>

<dl>
<dt>id</dt><dd><p>Gene identifier</p>
</dd>
<dt>name</dt><dd><p>Gene name</p>
</dd>
<dt>chr</dt><dd><p>Gene name</p>
</dd>
<dt>start</dt><dd><p>Gene name</p>
</dd>
<dt>end</dt><dd><p>Gene name</p>
</dd>
<dt>type</dt><dd><p>Gene name</p>
</dd>
</dl>



<h3>Source</h3>

<p>Ensembl, <a href="https://www.ensembl.org">https://www.ensembl.org</a>
</p>


<h3>References</h3>

<p>Harrison, PW, et al. (2024). Ensembl 2024. Nucleic Acids Research, 52, Pages D891–D899. <a href="https://doi.org/10.1093/nar/gkad1049">doi:10.1093/nar/gkad1049</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gene_names_grch38)
</code></pre>

<hr>
<h2 id='GWASCat_graph'>GWAScatalog graph</h2><span id='topic+GWASCat_graph'></span>

<h3>Description</h3>

<p>A graph object of GWAScatalog data. The association between variants and phenotypes are indicated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GWASCat_graph
</code></pre>


<h3>Format</h3>

<p>An object of igraph class.
</p>


<h3>Source</h3>

<p>GWAS Catalog, <a href="https://www.ebi.ac.uk/gwas">https://www.ebi.ac.uk/gwas</a>
</p>


<h3>References</h3>

<p>Sollis, E, et al. (2022).  The NHGRI-EBI GWAS Catalog: knowledgebase and deposition resource. Nucleic Acids Research, 51, Pages D977–D985. <a href="https://doi.org/10.1093/nar/gkac1010">doi:10.1093/nar/gkac1010</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GWASCat_graph)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
