<!DOCTYPE html><html lang="en-GB"><head><title>Help for package FieldSimR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FieldSimR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#compsym_asr_input'><p>Simulate genetic values based on a compound symmetry model for GxE interaction - 'AlphaSimR'</p>
input parameters</a></li>
<li><a href='#compsym_asr_output'><p>Simulate genetic values based on a compound symmetry model for GxE interaction -</p>
Simulation with 'AlphaSimR'</a></li>
<li><a href='#error_df_bivar'><p>Plot errors - Example data frame</p></a></li>
<li><a href='#field_trial_error'><p>Simulate plot errors in plant breeding field trials</p></a></li>
<li><a href='#group_cor_mat'><p>Simulate a reduced rank correlation matrix with multiple groups</p></a></li>
<li><a href='#gv_df_unstr'><p>Genetic values - Example data frame</p></a></li>
<li><a href='#make_phenotypes'><p>Generate phenotypes - Combine genetic values and plot errors</p></a></li>
<li><a href='#multi_asr_input'><p>Simulate genetic values based on a multiplicative model for GxE interaction - 'AlphaSimR'</p>
input parameters</a></li>
<li><a href='#multi_asr_output'><p>Simulate genetic values based on a multiplicative model for GxE interaction -</p>
Simulation with 'AlphaSimR'</a></li>
<li><a href='#plot_effects'><p>Graphics for plot effects</p></a></li>
<li><a href='#plot_hist'><p>Histogram of values</p></a></li>
<li><a href='#plot_matrix'><p>Graphics for matrices</p></a></li>
<li><a href='#qq_plot'><p>Q-Q plot</p></a></li>
<li><a href='#rand_cor_mat'><p>Simulate a random correlation matrix</p></a></li>
<li><a href='#rand_diag_mat'><p>Simulate a random diagonal variance matrix</p></a></li>
<li><a href='#sample_met'><p>Sample environments from a target population</p></a></li>
<li><a href='#sample_variogram'><p>Sample variogram</p></a></li>
<li><a href='#skew_diag_mat'><p>Simulate a skewed diagonal variance matrix</p></a></li>
<li><a href='#struc_cor_mat'><p>Simulate a structured correlation matrix with reduced rank</p></a></li>
<li><a href='#theoretical_variogram'><p>Theoretical variogram</p></a></li>
<li><a href='#unstr_asr_input'><p>Simulate genetic values based on an unstructured model for GxE interaction - 'AlphaSimR' input</p>
parameters</a></li>
<li><a href='#unstr_asr_output'><p>Simulate genetic values based on an unstructured model for GxE interaction -</p>
Simulation with 'AlphaSimR'</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Simulation of Plot Errors and Phenotypes in Plant Breeding Field
Trials</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-08-30</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Christian Werner &lt;werner.christian@proton.me&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulates plot data in multi-environment field trials with one or more traits. 
  Its core function generates plot errors that capture spatial trend, random error (noise), 
  and extraneous variation, which are combined at a user-defined ratio. 
  Phenotypes can be generated by combining the plot errors with simulated genetic values that capture 
  genotype-by-environment (GxE) interaction using wrapper functions for the R package 'AlphaSimR'.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/crWerner/fieldsimr">https://github.com/crWerner/fieldsimr</a>,
<a href="https://crwerner.github.io/fieldsimr/">https://crwerner.github.io/fieldsimr/</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, interp, lattice, Matrix, mbend, cluster, grDevices,
RColorBrewer</td>
</tr>
<tr>
<td>Suggests:</td>
<td>AlphaSimR, knitr, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/crWerner/fieldsimr/issues">https://github.com/crWerner/fieldsimr/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-08-30 14:30:26 UTC; CWERNER</td>
</tr>
<tr>
<td>Author:</td>
<td>Christian Werner <a href="https://orcid.org/0000-0001-9400-5061"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Daniel Tolhurst <a href="https://orcid.org/0000-0002-4787-080X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Jon Bancic [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-08-30 15:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='compsym_asr_input'>Simulate genetic values based on a compound symmetry model for GxE interaction - 'AlphaSimR'
input parameters</h2><span id='topic+compsym_asr_input'></span>

<h3>Description</h3>

<p>Creates a list of input parameters for
<a href="https://CRAN.R-project.org/package=AlphaSimR">'AlphaSimR'</a> to simulate
genetic values in multiple environments for one or more traits based on a compound symmetry
model for genotype-by-environment (GxE) interaction. <br />
This function utilises the ability of 'AlphaSimR' to simulate correlated traits.
The wrapper function <code>compsym_asr_input()</code> is used to specify the input parameters required in 'AlphaSimR'.
After simulating the genetic values, the wrapper function <a href="#topic+compsym_asr_output">compsym_asr_output</a> can be used to
generate a data frame with output values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compsym_asr_input(
  ntraits = 1,
  nenvs = 2,
  mean = 0,
  var = 1,
  prop.main = 0.5,
  corA = NULL,
  meanDD = NULL,
  varDD = NULL,
  prop.mainDD = NULL,
  corDD = NULL,
  relAA = NULL,
  prop.mainAA = NULL,
  corAA = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compsym_asr_input_+3A_ntraits">ntraits</code></td>
<td>
<p>Number of traits to be simulated.</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_nenvs">nenvs</code></td>
<td>
<p>Number of environments to be simulated (minimum of two).</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_mean">mean</code></td>
<td>
<p>A vector of mean genetic values for each environment-within-trait combination.
If only one value is specified, all combinations will be assigned the same mean.</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_var">var</code></td>
<td>
<p>A vector of genetic variances for each trait. <br />
<strong>Note:</strong> When <code>useVarA = TRUE</code> is specified in 'AlphaSimR' (default), the values in
<code>var</code> represent the additive genetic variances, otherwise they represent the
total (additive + non-additive) genetic variances.</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_prop.main">prop.main</code></td>
<td>
<p>A vector defining the proportion of main effect variance for each trait.
If only one value is specified, all traits will be assigned the same proportion. <br />
<strong>Note:</strong> <code>0 &lt; prop.main &lt; 1</code>.</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_cora">corA</code></td>
<td>
<p>A matrix of additive genetic correlations between traits. By default, a diagonal
matrix is constructed.</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_meandd">meanDD</code></td>
<td>
<p>A vector of mean dominance degrees for each environment-within-trait combination
(similar to <code>mean</code>). If only one value is specified, all combinations will be assigned
the same mean. By default, <code>meanDD = NULL</code> and dominance is not simulated.</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_vardd">varDD</code></td>
<td>
<p>A vector of dominance degree variances for each trait.</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_prop.maindd">prop.mainDD</code></td>
<td>
<p>A vector defining the proportion of dominance degree main effect
variance for each trait (similar to <code>prop.main</code>).
If only one value is specified, all traits will be assigned the same proportion. <br />
<strong>Note:</strong> <code>0 &lt; prop.mainDD &lt; 1</code>.</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_cordd">corDD</code></td>
<td>
<p>A matrix of dominance degree correlations between traits (similar
to <code>corA</code>). If not specified and dominance is simulated, a diagonal matrix is constructed.</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_relaa">relAA</code></td>
<td>
<p>A vector defining the relative magnitude of additive-by-additive (epistatic) variance
to additive genetic variance for each trait, that is in a diploid organism with
allele frequency of 0.5.
If only one value is specified, all traits will be assigned the same relative magnitude.</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_prop.mainaa">prop.mainAA</code></td>
<td>
<p>A vector defining the proportion of epistatic main effect variance for each
trait (similar to <code>prop.main</code>). If only one value is specified, all traits will be assigned the
same proportion. <br />
<strong>Note:</strong> <code>0 &lt; prop.mainAA &lt; 1</code>.</p>
</td></tr>
<tr><td><code id="compsym_asr_input_+3A_coraa">corAA</code></td>
<td>
<p>A matrix of epistatic correlations between traits (similar to
<code>corA</code>). If not specified and epistasis is simulated, a diagonal matrix is constructed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The compound symmetry model assumes the same genetic variance for each environment
and the same genetic covariance between each pair of environments. New functionality
is being implemented which relaxes the former assumption
(also see <a href="#topic+unstr_asr_output">unstr_asr_output</a>).
</p>
<p><strong>Note:</strong> 'AlphaSimR' can simulate different biological effects (see:
<a href="https://gaynorr.github.io/AlphaSimR/reference/SimParam.html">SimParam</a>).
</p>

<ul>
<li><p> For additive traits use <code>addTraitA()</code>.
</p>
</li>
<li><p> For additive + dominance traits use <code>addTraitAD()</code>.
</p>
</li>
<li><p> For additive + epistatic traits use <code>addTraitAE()</code>.
</p>
</li>
<li><p> For additive + dominance + epistatic traits use <code>addTraitADE()</code>.
</p>
</li></ul>

<p>Check the <code>useVarA</code> argument of these functions when simulating non-additive traits.
</p>


<h3>Value</h3>

<p>A list with input parameters for 'AlphaSimR', which are used to simulate
correlated genetic values based on a compound symmetry model for GxE interaction.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate genetic values with 'AlphaSimR' for two additive + dominance traits
# in two environments based on a compound symmetry model.

# 1. Define the genetic architecture of the simulated traits.
# Mean genetic values and mean dominance degrees.
mean &lt;- c(4.9, 5.4, 235.2, 228.5) # Trait 1 x 2 environments, Trait 2 x 2 environments
meanDD &lt;- c(0.4, 0.4, 0.1, 0.1) # Trait 1 and 2, same value for both environments

# Additive genetic variances and dominance degree variances.
var &lt;- c(0.08, 13) # Different values for Traits 1 and 2
varDD &lt;- 0.2 # Same value for Traits 1 and 2

# Proportion of additive and dominance degree main effect variances.
prop.main &lt;- c(0.4, 0.6) # Different values for Traits 1 and 2
prop.mainDD &lt;- 0.4 # Same value for Traits 1 and 2

# Additive and dominance degree correlations between the two simulated traits.
corA &lt;- matrix(c(
  1.0, 0.5,
  0.5, 1.0
), ncol = 2)
corDD &lt;- diag(2) # Assuming independence

input_asr &lt;- compsym_asr_input(
  ntraits = 2,
  nenvs = 2,
  mean = mean,
  var = var,
  prop.main = prop.main,
  corA = corA,
  meanDD = meanDD,
  varDD = varDD,
  prop.mainDD = prop.mainDD,
  corDD = corDD
)

</code></pre>

<hr>
<h2 id='compsym_asr_output'>Simulate genetic values based on a compound symmetry model for GxE interaction -
Simulation with 'AlphaSimR'</h2><span id='topic+compsym_asr_output'></span>

<h3>Description</h3>

<p>Creates a data frame of simulated genetic values in multiple environments for one or more traits
based on a compound symmetry model for genotype-by-environment (GxE) interaction. The wrapper function
<code>compsym_asr_output()</code> requires an <a href="https://CRAN.R-project.org/package=AlphaSimR">'AlphaSimR'</a>
population object generated with <a href="#topic+compsym_asr_input">compsym_asr_input</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compsym_asr_output(pop, ntraits = 1, nenvs, nreps = 1, return.effects = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compsym_asr_output_+3A_pop">pop</code></td>
<td>
<p>An <a href="https://CRAN.R-project.org/package=AlphaSimR">'AlphaSimR'</a> population object
(<a href="https://gaynorr.github.io/AlphaSimR/reference/Pop-class.html">Pop-class</a> or
<a href="https://gaynorr.github.io/AlphaSimR/reference/HybridPop-class.html">HybridPop-class</a>)
generated with <a href="#topic+compsym_asr_input">compsym_asr_input</a>.</p>
</td></tr>
<tr><td><code id="compsym_asr_output_+3A_ntraits">ntraits</code></td>
<td>
<p>Number of traits specified in <a href="#topic+compsym_asr_input">compsym_asr_input</a>.</p>
</td></tr>
<tr><td><code id="compsym_asr_output_+3A_nenvs">nenvs</code></td>
<td>
<p>Number of environments specified in <a href="#topic+compsym_asr_input">compsym_asr_input</a>.</p>
</td></tr>
<tr><td><code id="compsym_asr_output_+3A_nreps">nreps</code></td>
<td>
<p>A vector defining the number of replicates in each environment. If only one value
is specified, all environments will be assigned the same number.</p>
</td></tr>
<tr><td><code id="compsym_asr_output_+3A_return.effects">return.effects</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), a list is returned with additional
entries containing the genotype main effects and GxE interaction effects for each trait.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns 'env', genotype 'id', and 'rep', followed by the
simulated genetic values for each trait. When <code>return.effects = TRUE</code>, a list is returned with
additional entries containing the genotype main effects and GxE interaction effects for each trait.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate genetic values with 'AlphaSimR' for two additive + dominance traits
# in two environments based on a compound symmetry model.

# 1. Define the genetic architecture of the simulated traits.
# Mean genetic values and mean dominance degrees.
mean &lt;- c(4.9, 5.4, 235.2, 228.5) # Trait 1 x 2 environments, Trait 2 x 2 environments
meanDD &lt;- c(0.4, 0.4, 0.1, 0.1) # Trait 1 and 2, same value for both environments

# Additive genetic variances and dominance degree variances.
var &lt;- c(0.08, 13) # Different values for Traits 1 and 2
varDD &lt;- 0.2 # Same value for Traits 1 and 2

# Proportion of additive and dominance degree main effect variances.
prop.main &lt;- c(0.4, 0.6) # Different values for Traits 1 and 2
prop.mainDD &lt;- 0.4 # Same value for Traits 1 and 2

# Additive and dominance degree correlations between the two simulated traits.
corA &lt;- matrix(c(
  1.0, 0.5,
  0.5, 1.0
), ncol = 2)
corDD &lt;- diag(2) # Assuming independence

input_asr &lt;- compsym_asr_input(
  ntraits = 2,
  nenvs = 2,
  mean = mean,
  var = var,
  prop.main = prop.main,
  corA = corA,
  meanDD = meanDD,
  varDD = varDD,
  prop.mainDD = prop.mainDD,
  corDD = corDD
)


# 2. Use input_asr to simulate genetic values with 'AlphaSimR' based on a
# compound symmetry model.

library("AlphaSimR")
FOUNDERPOP &lt;- quickHaplo(
  nInd = 10,
  nChr = 1,
  segSites = 20
)

SP &lt;- SimParam$new(FOUNDERPOP)



SP$addTraitAD(
  nQtlPerChr = 20,
  mean = input_asr$mean,
  var = input_asr$var,
  corA = input_asr$corA,
  meanDD = input_asr$meanDD,
  varDD = input_asr$varDD,
  corDD = input_asr$corDD,
  useVarA = TRUE
)

# By default, the variances in 'var' represent additive genetic variances.
# When useVarA = FALSE, the values represent total genetic variances.

pop &lt;- newPop(FOUNDERPOP)


# 3. Create a data frame with simulated genetic values for the two traits in
# the two environments, with two replicates of each genotype.

gv_ls &lt;- compsym_asr_output(
  pop = pop,
  ntraits = 2,
  nenvs = 2,
  nreps = 2,
  return.effects = TRUE
)

</code></pre>

<hr>
<h2 id='error_df_bivar'>Plot errors - Example data frame</h2><span id='topic+error_df_bivar'></span>

<h3>Description</h3>

<p>An example data frame with simulated plot errors for two traits in three environments.
Environments 1 and 2 comprise two blocks, while Environment 3 comprises three blocks.
The blocks are aligned in the column direction (side-by-side) and comprise 5 columns and 20 rows.
The data frame was generated using the function <a href="#topic+field_trial_error">field_trial_error</a> with
bivariate interpolation. The simulation is demonstrated in the vignette
<a href="https://crwerner.github.io/fieldsimr/articles/spatial_error_demo.html">Simulation of plot errors and phenotypes in a plant breeding field trial</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>error_df_bivar
</code></pre>


<h3>Format</h3>

<p>A data frame with 700 rows and 6 columns:
</p>

<dl>
<dt>env</dt><dd><p>Environment number</p>
</dd>
<dt>block</dt><dd><p>Block number</p>
</dd>
<dt>col</dt><dd><p>Column number</p>
</dd>
<dt>row</dt><dd><p>Row number</p>
</dd>
<dt>e.Trait1</dt><dd><p>Simulated plot errors for Trait 1</p>
</dd>
<dt>e.Trait2</dt><dd><p>Simulated plot errors for Trait 2</p>
</dd>
</dl>


<hr>
<h2 id='field_trial_error'>Simulate plot errors in plant breeding field trials</h2><span id='topic+field_trial_error'></span>

<h3>Description</h3>

<p>Creates a data frame of simulated plot errors in multi-environment field trials for one or
more traits. The plot errors capture spatial trend, random error (noise), and extraneous variation.
Spatial trend is simulated using bivariate interpolation or a separable first-order
autoregressive (AR1) process. Random error is simulated using an independent process. Extraneous
variation is simulated using random or zig-zag ordering between neighbouring columns and/or rows.
The three error components are combined at a user-defined ratio. <br />
Correlated plot errors can be simulated between traits by setting different correlation structures
for each error component. A separable structure is assumed between traits and plots within environments,
but different error variances can be specified for each environment-within-trait combination.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>field_trial_error(
  ntraits = 1,
  nenvs = 1,
  nblocks = 2,
  block.dir = "col",
  ncols = 10,
  nrows = 20,
  varR = 1,
  ScorR = NULL,
  RcorR = NULL,
  EcorR = NULL,
  spatial.model = "Bivariate",
  complexity = NULL,
  plot.length = 8,
  plot.width = 2,
  col.cor = 0.5,
  row.cor = 0.7,
  prop.spatial = 0.5,
  ext.ord = "random",
  ext.dir = "row",
  prop.ext = 0,
  return.effects = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="field_trial_error_+3A_ntraits">ntraits</code></td>
<td>
<p>Number of traits to be simulated.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_nenvs">nenvs</code></td>
<td>
<p>Number of environments to be simulated.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_nblocks">nblocks</code></td>
<td>
<p>A vector defining the number of blocks in each environment.
If only one value is specified, all environments will be assigned the same number.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_block.dir">block.dir</code></td>
<td>
<p>A vector defining the block direction in each environment. Use
'col' for side-by-side (default), 'row' for above-and-below, or
NA if only one block is simulated. If only one value is specified, all environments
will be assigned the same block direction.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_ncols">ncols</code></td>
<td>
<p>A vector defining the number of columns in each environment. If only one
value is specified, all environments will be assigned the same number.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_nrows">nrows</code></td>
<td>
<p>A vector defining the number of rows in each environment. If only one
value is specified, all environments will be assigned the same number.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_varr">varR</code></td>
<td>
<p>A vector of error variances for each environment-within-trait combination. If only
one value is specified, all combinations will be assigned the same error variance.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_scorr">ScorR</code></td>
<td>
<p>A matrix of spatial error correlations between traits. If not specified and
spatial trend is simulated, a diagonal matrix is constructed.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_rcorr">RcorR</code></td>
<td>
<p>A matrix of random error correlations between traits. If not specified and
random error is simulated, a diagonal matrix is constructed.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_ecorr">EcorR</code></td>
<td>
<p>A matrix of extraneous error correlations between traits. If not specified and
extraneous variation is simulated, a diagonal matrix is constructed. <br />
<strong>Note:</strong> the same correlation between traits is used for the column and row errors.
Currently only implemented when <code>ext.ord = "random"</code>.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_spatial.model">spatial.model</code></td>
<td>
<p>A character string defining the model used to simulate spatial trend.
Use 'Bivariate' for bivariate interpolation (default) or 'AR1' for a separable first-order
autoregressive process. Bivariate interpolation is implemented with the <code>interp</code>
function of the R package <a href="https://CRAN.R-project.org/package=interp">'interp'</a>.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_complexity">complexity</code></td>
<td>
<p>A vector defining the complexity of the simulated spatial trend in each
environment when <code>spatial.model = "Bivariate"</code>. If only one value is specified,
all environments will be assigned the same complexity.
If not specified and <code>spatial.model = "Bivariate"</code>, the complexity is set to half the maximum
number of columns and rows in each environment.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_plot.length">plot.length</code></td>
<td>
<p>A vector of plot lengths for each environment (column direction). If only one value is specified,
all environments will be assigned the same plot length. Only required when <code>spatial.model = "Bivariate"</code>.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_plot.width">plot.width</code></td>
<td>
<p>A vector of plot widths for each environment (row direction). If only one value is specified,
all environments will be assigned the same plot width. Only required when <code>spatial.model = "Bivariate"</code>.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_col.cor">col.cor</code></td>
<td>
<p>A vector of column autocorrelations for each environment.
If only one value is specified, all environments will be assigned the same column autocorrelation.
Only required when <code>spatial.model = "AR1"</code>.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_row.cor">row.cor</code></td>
<td>
<p>A vector of row autocorrelations for each environment.
If only one value is specified, all environments will be assigned the same row
autocorrelation. Only required when <code>spatial.model = "AR1"</code>.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_prop.spatial">prop.spatial</code></td>
<td>
<p>A vector defining the proportion of spatial trend for each environment-within-trait
combination. If only one value is specified, all combinations will be assigned the proportion.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_ext.ord">ext.ord</code></td>
<td>
<p>A character string defining the method used to simulate extraneous variation.
Use 'random' (default) for random variation between neighbouring columns and/or rows or
'zig-zag' for alternating positive and negative values.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_ext.dir">ext.dir</code></td>
<td>
<p>A vector defining the direction of extraneous variation for each environment.
Use 'row' (default) for row variation, 'col' for column variation, 'both' for variation in both directions,
or NA if no extraneous variation is simulated. When <code>ext.dir = "both"</code>, half the variance is assigned to the columns and
half is assigned to the rows. If only one value is specified, all environments will be
assigned the same direction.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_prop.ext">prop.ext</code></td>
<td>
<p>A vector defining the proportion of extraneous variation for each environment-within-trait
combination. If only one value is specified, all combinations will be assigned the same proportion.</p>
</td></tr>
<tr><td><code id="field_trial_error_+3A_return.effects">return.effects</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), a list is returned with additional entries
containing the spatial, random, and extraneous error terms for each trait.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns 'env', 'block', 'col', and 'row', followed by the
simulated plot errors for each trait. When <code>return.effects = TRUE</code>, a list is returned with additional entries
containing the spatial, random, and extraneous error terms for each trait.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate plot errors for two traits in two environments using an AR1 model
# for spatial variation.

# Error variances for the four environment-within-trait combinations.
varR &lt;- c(0.2, 0.4, 10, 15) # Trait 1 x 2 environments, Trait 2 x 2 environments

# Spatial error correlations between the two simulated traits.
ScorR &lt;- matrix(c(
  1.0, 0.2,
  0.2, 1.0
), ncol = 2)

error_ls &lt;- field_trial_error(
  ntraits = 2,
  nenvs = 2,
  nblocks = 2,
  block.dir = "row",
  ncols = 10,
  nrows = 20,
  varR = varR,
  ScorR = ScorR,
  spatial.model = "AR1",
  col.cor = 0.5,
  row.cor = 0.7,
  prop.spatial = 0.4,
  ext.ord = "zig-zag",
  ext.dir = "row",
  prop.ext = 0.2,
  return.effects = TRUE
)

</code></pre>

<hr>
<h2 id='group_cor_mat'>Simulate a reduced rank correlation matrix with multiple groups</h2><span id='topic+group_cor_mat'></span>

<h3>Description</h3>

<p>Creates a symmetric correlation matrix with user-defined structure, rank and groupings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group_cor_mat(
  n = c(5, 5),
  within.cor = 0.5,
  between.cor = 0.2,
  range = NULL,
  rank = 4,
  skew = 0,
  pos.def = FALSE,
  small.positive = NULL,
  return.groups = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="group_cor_mat_+3A_n">n</code></td>
<td>
<p>A vector defining the size of each group.</p>
</td></tr>
<tr><td><code id="group_cor_mat_+3A_within.cor">within.cor</code></td>
<td>
<p>A vector defining the baseline correlation within each group.
If only one value is supplied, all groups will be assigned the same correlation. <br />
<strong>Note:</strong> <code>-1 &lt; within.cor &lt; 1</code>.</p>
</td></tr>
<tr><td><code id="group_cor_mat_+3A_between.cor">between.cor</code></td>
<td>
<p>A scalar defining the baseline correlation between groups. <br />
<strong>Note:</strong> <code>between.cor &lt;= within.cor</code>.</p>
</td></tr>
<tr><td><code id="group_cor_mat_+3A_range">range</code></td>
<td>
<p>A scalar defining the range of correlations around the baseline. By default,
<code>range = 1 - max(within.cor)</code> which ensures the matrix is positive semi-definite.
<strong>Note:</strong> <code>max(within.cor) + range &lt;= 1</code>.</p>
</td></tr>
<tr><td><code id="group_cor_mat_+3A_rank">rank</code></td>
<td>
<p>A scalar defining the rank of the correlation matrix.</p>
</td></tr>
<tr><td><code id="group_cor_mat_+3A_skew">skew</code></td>
<td>
<p>A scalar defining the skewness imposed on the correlations.
<strong>Note:</strong> <code>-1 &lt; skew &lt; 1</code>.</p>
</td></tr>
<tr><td><code id="group_cor_mat_+3A_pos.def">pos.def</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), the function <code>bend</code> of the R package
<a href="https://cran.r-project.org/package=mbend">'mbend'</a> is used to
bend a non-positive (semi)-definite matrix to be positive (semi)-definite.</p>
</td></tr>
<tr><td><code id="group_cor_mat_+3A_small.positive">small.positive</code></td>
<td>
<p>Argument passed to <code>bend</code> when <code>pos.def = TRUE</code> (default is 1e-8).
Eigenvalues smaller than <code>small.positive</code> are replaced by this. <br />
<strong>Note:</strong> <code>0 &lt; small.positive &lt; 0.1</code>.</p>
</td></tr>
<tr><td><code id="group_cor_mat_+3A_return.groups">return.groups</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), a list is returned with additional
entries containing the members of each group.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A symmetric correlation matrix with defined rank and groupings.
When <code>pos.def = TRUE</code>, the correlation matrix is guaranteed to be positive (semi)-definite.
When <code>return.groups = TRUE</code>, a list is returned with additional entries containing the group members.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate and visualise a correlation matrix with 2 groups containing 5 and 10 members,
# correlations of 0.4 within groups and 0 between groups and rank equal to 4
cor_ls &lt;- group_cor_mat(
  n = c(5, 10),
  within.cor = 0.4,
  between.cor = 0,
  rank = 4,
  return.groups = TRUE
)

plot_matrix(
  mat = cor_ls$cor.mat,
  group.df = cor_ls$group.df,
  order = TRUE
)

</code></pre>

<hr>
<h2 id='gv_df_unstr'>Genetic values - Example data frame</h2><span id='topic+gv_df_unstr'></span>

<h3>Description</h3>

<p>An example data frame with simulated genetic values of 100 genotypes for two traits in three environments.
Environments 1 and 2 comprise two replicates of each genotype, while Environment 3 comprises three replicates.
The data frame was generated using the wrapper functions
<a href="#topic+unstr_asr_input">unstr_asr_input</a> and <a href="#topic+unstr_asr_output">unstr_asr_output</a>, which simulate correlated
genetic values with <a href="https://CRAN.R-project.org/package=AlphaSimR">'AlphaSimR'</a>. The simulation is
demonstrated in the vignette
<a href="https://crwerner.github.io/fieldsimr/articles/unstructured_GxE_demo.html">Simulation of genetic values based on an unstructured model for GxE interaction</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gv_df_unstr
</code></pre>


<h3>Format</h3>

<p>A data frame with 700 rows and 5 columns:
</p>

<dl>
<dt>env</dt><dd><p>Environment number</p>
</dd>
<dt>id</dt><dd><p>Genotype identifier</p>
</dd>
<dt>rep</dt><dd><p>Replicate number</p>
</dd>
<dt>gv.Trait1</dt><dd><p>Simulated genetic values for Trait 1</p>
</dd>
<dt>gv.Trait2</dt><dd><p>Simulated genetic values for Trait 2</p>
</dd>
</dl>


<hr>
<h2 id='make_phenotypes'>Generate phenotypes - Combine genetic values and plot errors</h2><span id='topic+make_phenotypes'></span>

<h3>Description</h3>

<p>Creates a data frame of phenotypes by combining genetic values with plot errors
generated with the function <a href="#topic+field_trial_error">field_trial_error</a>.
Requires genetic values generated with the functions <a href="#topic+compsym_asr_output">compsym_asr_output</a>
or <a href="#topic+unstr_asr_output">unstr_asr_output</a>,
or any data frame matching the description below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_phenotypes(
  gv.df,
  error.df,
  design.df = NULL,
  randomise = TRUE,
  return.effects = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_phenotypes_+3A_gv.df">gv.df</code></td>
<td>
<p>A data frame of genetic values. Must contain the columns 'env', genotype 'id', 'rep',
and the genetic values for each trait.</p>
</td></tr>
<tr><td><code id="make_phenotypes_+3A_error.df">error.df</code></td>
<td>
<p>A data frame of plot errors. Must contain the columns 'env', 'block',
'col', 'row', and the plot errors for each trait.</p>
</td></tr>
<tr><td><code id="make_phenotypes_+3A_design.df">design.df</code></td>
<td>
<p>A optional data frame of frequencies for generating incomplete block designs.
Must contain the columns 'env', 'id', and 'nreps' indicating the number of replicates per individual for each environment.</p>
</td></tr>
<tr><td><code id="make_phenotypes_+3A_randomise">randomise</code></td>
<td>
<p>When <code>TRUE</code> (default), genotypes are randomly allocated to plots according to
a randomized complete (or incomplete) block design.<br />
<strong>Note:</strong> Other experimental designs are being implemented and should be generated externally.</p>
</td></tr>
<tr><td><code id="make_phenotypes_+3A_return.effects">return.effects</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), a list is returned with additional
entries containing the genetic values and plot errors for each trait.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns 'env', 'block', 'column', 'row', genotype 'id', 'rep', and
the phenotypes for each trait. When <code>return.effects = TRUE</code>, a list is returned with additional
entries containing the genetic values and plot errors for each trait.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate and visualise phenotypes by combining the genetic values and plot errors provided
# in the two example data frames gv_df_unstr and error_df_bivar.

pheno_ls &lt;- make_phenotypes(
  gv.df = gv_df_unstr,
  error.df = error_df_bivar,
  randomise = TRUE,
  return.effects = TRUE
)

plot_effects(
  df = pheno_ls$pheno.df[pheno_ls$pheno.df$env == 1, ],
  effect = "y.Trait1",
  labels = TRUE,
)

</code></pre>

<hr>
<h2 id='multi_asr_input'>Simulate genetic values based on a multiplicative model for GxE interaction - 'AlphaSimR'
input parameters</h2><span id='topic+multi_asr_input'></span>

<h3>Description</h3>

<p>Creates a list of input parameters for
<a href="https://CRAN.R-project.org/package=AlphaSimR">'AlphaSimR'</a> to simulate
genetic values in multiple environments for one or more traits based on a (reduced rank)
multiplicative model for genotype-by-environment (GxE) interaction. <br />
This function utilises the ability of 'AlphaSimR' to simulate correlated traits.
The wrapper function <code>multi_asr_input()</code> is used to specify the input parameters required in 'AlphaSimR'.
After simulating the genetic values, the wrapper function <a href="#topic+multi_asr_output">multi_asr_output</a> can be used to
generate a data frame with output values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi_asr_input(
  ntraits = 1,
  nenvs = 2,
  mean = 0,
  var = 1,
  corA = NULL,
  nterms = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multi_asr_input_+3A_ntraits">ntraits</code></td>
<td>
<p>Number of traits to be simulated.</p>
</td></tr>
<tr><td><code id="multi_asr_input_+3A_nenvs">nenvs</code></td>
<td>
<p>Number of environments to be simulated (minimum of two).</p>
</td></tr>
<tr><td><code id="multi_asr_input_+3A_mean">mean</code></td>
<td>
<p>A vector of mean genetic values for each trait or each environment-within-trait combination.
If only one value is specified, all combinations will be assigned the same mean.</p>
</td></tr>
<tr><td><code id="multi_asr_input_+3A_var">var</code></td>
<td>
<p>A vector of additive genetic variances for each trait or each environment-within-trait combination.
If only one value is specified, all combinations will be assigned the same variance.</p>
</td></tr>
<tr><td><code id="multi_asr_input_+3A_cora">corA</code></td>
<td>
<p>A matrix of additive genetic correlations between environment-within-trait
combinations. By default, a diagonal matrix is constructed.</p>
</td></tr>
<tr><td><code id="multi_asr_input_+3A_nterms">nterms</code></td>
<td>
<p>A scalar defining the number of multiplicative terms to be simulated. By default,
the number of terms is set to the number of environment-within-trait combinations.
<strong>Note:</strong> when <code>nterms</code> is less than the number of environment-within-trait combinations,
the values in <code>mean</code> will be approximated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently supports additive traits only, but other (non-additive) traits are being implemented.
</p>


<h3>Value</h3>

<p>A list with input parameters for 'AlphaSimR', which are used to simulate
correlated genetic values based on a multiplicative model for GxE interaction.
Covariates are also supplied for use in <a href="#topic+multi_asr_output">multi_asr_output</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate genetic values with 'AlphaSimR' for two additive traits in two
# environments based on a multiplicative model with three terms.

# 1. Define the genetic architecture of the simulated traits.
# Mean genetic values.
mean &lt;- c(5, 240) # Trait 1, Trait 2

# Additive genetic variances.
var &lt;- c(0.086, 0.12, 15.1, 8.5) # Trait 1 x 2 environments, Trait 2 x 2 environments

# Additive genetic correlations between the two simulated traits.
TcorA &lt;- matrix(c(
  1.0, 0.6,
  0.6, 1.0
), ncol = 2)

# Additive genetic correlations between the two simulated environments.
EcorA &lt;- matrix(c(
  1.0, 0.2,
  0.2, 1.0
), ncol = 2)

# Construct separable additive genetic correlation matrix.
corA &lt;- kronecker(TcorA, EcorA)

input_asr &lt;- multi_asr_input(
  ntraits = 2,
  nenvs = 2,
  mean = mean,
  var = var,
  corA = corA,
  nterms = 3
)

</code></pre>

<hr>
<h2 id='multi_asr_output'>Simulate genetic values based on a multiplicative model for GxE interaction -
Simulation with 'AlphaSimR'</h2><span id='topic+multi_asr_output'></span>

<h3>Description</h3>

<p>Creates a data frame of simulated genetic values in multiple environments for one or more traits
based on a (reduced rank) multiplicative model for genotype-by-environment (GxE) interaction.
This function requires an <a href="https://CRAN.R-project.org/package=AlphaSimR">'AlphaSimR'</a>
population object generated with <a href="#topic+multi_asr_input">multi_asr_input</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi_asr_output(
  pop,
  ntraits = 1,
  nenvs,
  nreps = 1,
  cov.mat,
  return.effects = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multi_asr_output_+3A_pop">pop</code></td>
<td>
<p>An <a href="https://CRAN.R-project.org/package=AlphaSimR">'AlphaSimR'</a> population object
(<a href="https://gaynorr.github.io/AlphaSimR/reference/Pop-class.html">Pop-class</a> or
<a href="https://gaynorr.github.io/AlphaSimR/reference/HybridPop-class.html">HybridPop-class</a>)
generated with <a href="#topic+multi_asr_input">multi_asr_input</a>.</p>
</td></tr>
<tr><td><code id="multi_asr_output_+3A_ntraits">ntraits</code></td>
<td>
<p>Number of traits specified in <a href="#topic+multi_asr_input">multi_asr_input</a>.</p>
</td></tr>
<tr><td><code id="multi_asr_output_+3A_nenvs">nenvs</code></td>
<td>
<p>Number of environments specified in <a href="#topic+multi_asr_input">multi_asr_input</a>.</p>
</td></tr>
<tr><td><code id="multi_asr_output_+3A_nreps">nreps</code></td>
<td>
<p>A vector defining the number of replicates in each environment. If only one value
is specified, all environments will be assigned the same number.</p>
</td></tr>
<tr><td><code id="multi_asr_output_+3A_cov.mat">cov.mat</code></td>
<td>
<p>A matrix of covariates that will be used to construct the genetic values, typically generated
with <a href="#topic+multi_asr_input">multi_asr_input</a>.</p>
</td></tr>
<tr><td><code id="multi_asr_output_+3A_return.effects">return.effects</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), a list is returned with additional
entries containing the genotype slopes for each multiplicative term.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns 'env', genotype 'id', and 'rep', followed by the
simulated genetic values for each trait. When <code>return.effects = TRUE</code>, a list is returned with
additional entries containing the genotype slopes for each multiplicative term.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate genetic values with 'AlphaSimR' for two additive traits in two
# environments based on a multiplicative model with three terms.

# 1. Define the genetic architecture of the simulated traits.
# Mean genetic values.
mean &lt;- c(5, 240) # Trait 1, Trait 2

# Additive genetic variances.
var &lt;- c(0.086, 0.12, 15.1, 8.5) # Trait 1 x 2 environments, Trait 2 x 2 environments

# Additive genetic correlations between the two simulated traits.
TcorA &lt;- matrix(c(
  1.0, 0.6,
  0.6, 1.0
), ncol = 2)

# Additive genetic correlations between the two simulated environments.
EcorA &lt;- matrix(c(
  1.0, 0.2,
  0.2, 1.0
), ncol = 2)

# Construct separable additive genetic correlation matrix
corA &lt;- kronecker(TcorA, EcorA)

input_asr &lt;- multi_asr_input(
  ntraits = 2,
  nenvs = 2,
  mean = mean,
  var = var,
  corA = corA,
  nterms = 3
)


# 2. Use input_asr to simulate genetic values in 'AlphaSimR' based on a
# multiplicative model with three terms.

library("AlphaSimR")
FOUNDERPOP &lt;- quickHaplo(
  nInd = 10,
  nChr = 1,
  segSites = 20
)

SP &lt;- SimParam$new(FOUNDERPOP)



SP$addTraitA(
  nQtlPerChr = 20,
  mean = input_asr$mean,
  var = input_asr$var,
  corA = input_asr$corA
)

pop &lt;- newPop(FOUNDERPOP)


# 3. Create a data frame with simulated genetic values for the two traits in the two
# environments, with two replicates of each genotype.

# The covariates are obtained from input_asr.

gv_ls &lt;- multi_asr_output(
  pop = pop,
  ntraits = 2,
  nenvs = 2,
  nreps = 2,
  cov.mat = input_asr$cov.mat,
  return.effects = TRUE
)

</code></pre>

<hr>
<h2 id='plot_effects'>Graphics for plot effects</h2><span id='topic+plot_effects'></span>

<h3>Description</h3>

<p>Creates a graphical field array for a set of plot effects (e.g., phenotypes, genetic values, or plot errors).
Requires a data frame generated with the functions <a href="#topic+field_trial_error">field_trial_error</a> or <a href="#topic+make_phenotypes">make_phenotypes</a>,
or any data frame matching the description below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_effects(df, effect, blocks = TRUE, labels = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_effects_+3A_df">df</code></td>
<td>
<p>A data frame with the columns 'col', 'row', and the effects to be plotted.</p>
</td></tr>
<tr><td><code id="plot_effects_+3A_effect">effect</code></td>
<td>
<p>The name of the effects to be plotted.</p>
</td></tr>
<tr><td><code id="plot_effects_+3A_blocks">blocks</code></td>
<td>
<p>When <code>TRUE</code> (default), the field array is split into blocks.
This requires an additional column 'block' in the data frame.</p>
</td></tr>
<tr><td><code id="plot_effects_+3A_labels">labels</code></td>
<td>
<p>When <code>TRUE</code> (default), column and row labels are displayed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A graphical field array with x- and y-axes displaying the column and row numbers,
and colour gradient ranging from red (low value) to green (high value).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Display the simulated plot errors in the example data frame 'error_df_bivar'
# for Trait 1 in Environment 1.

error_df &lt;- error_df_bivar[error_df_bivar$env == 1, ]

plot_effects(
  df = error_df,
  effect = "e.Trait1",
  labels = TRUE,
)

</code></pre>

<hr>
<h2 id='plot_hist'>Histogram of values</h2><span id='topic+plot_hist'></span>

<h3>Description</h3>

<p>Creates a histogram of user-defined values (e.g., effects, correlations, or covariances).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_hist(df, value = NULL, bins = 30, density = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_hist_+3A_df">df</code></td>
<td>
<p>A data frame or vector with the values to be plotted.</p>
</td></tr>
<tr><td><code id="plot_hist_+3A_value">value</code></td>
<td>
<p>The name of the values to be plotted. Ignored when 'df' is a vector.</p>
</td></tr>
<tr><td><code id="plot_hist_+3A_bins">bins</code></td>
<td>
<p>Argument passed to <code>ggplot2</code> (default is <code>30</code>). Controls the number
of bins in the histogram.</p>
</td></tr>
<tr><td><code id="plot_hist_+3A_density">density</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), a density curve is superimposed
onto the histogram.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A histogram with x- and y-axes displaying the values and their frequency, respectively.
When <code>density = TRUE</code>, a density curve is superimposed onto the histogram.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Histogram of the simulated plot errors in the example data frame 'error_df_bivar'
# for Trait 1 in Environment 1.
error_df &lt;- error_df_bivar[error_df_bivar$env == 1, ]
plot_hist(
  df = error_df,
  value = "e.Trait1",
  density = TRUE
)

</code></pre>

<hr>
<h2 id='plot_matrix'>Graphics for matrices</h2><span id='topic+plot_matrix'></span>

<h3>Description</h3>

<p>Creates a heatmap of a symmetric matrix (e.g., correlation or covariance matrix).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_matrix(mat, order = FALSE, group.df = NULL, labels = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_matrix_+3A_mat">mat</code></td>
<td>
<p>A symmetric matrix.</p>
</td></tr>
<tr><td><code id="plot_matrix_+3A_order">order</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), the function <code>agnes</code> of the R package
<a href="https://cran.r-project.org/package=cluster">'cluster'</a> is used with default arguments to
order the matrix based on a dendrogram.</p>
</td></tr>
<tr><td><code id="plot_matrix_+3A_group.df">group.df</code></td>
<td>
<p>An optional data frame with columns containing the variable names followed by the group numbers.
When supplied, the heatmap is split into groups and then ordered (when <code>order = TRUE</code>).</p>
</td></tr>
<tr><td><code id="plot_matrix_+3A_labels">labels</code></td>
<td>
<p>When <code>TRUE</code> (default), variable labels are displayed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap with x- and y-axes displaying the variable numbers,
and colour gradient ranging from blue (low value) to red (high value).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Display a random correlation matrix.

cor_mat &lt;- rand_cor_mat(
  n = 10,
  min.cor = -1,
  max.cor = 1
)

# Define groups.
group_df &lt;- data.frame(variable = 1:10, group = c(1, 1, 1, 1, 2, 2, 2, 3, 3, 4))

plot_matrix(
  mat = cor_mat,
  group.df = group_df,
  order = TRUE,
  labels = TRUE
)

</code></pre>

<hr>
<h2 id='qq_plot'>Q-Q plot</h2><span id='topic+qq_plot'></span>

<h3>Description</h3>

<p>Creates a normal quantile-quantile (Q-Q) plot for a set of effects (e.g., phenotypes, genetic values, or plot errors).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qq_plot(df, effect, labels = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qq_plot_+3A_df">df</code></td>
<td>
<p>A data frame or vector with the effects to be plotted.</p>
</td></tr>
<tr><td><code id="qq_plot_+3A_effect">effect</code></td>
<td>
<p>The name of the effects to be plotted. Ignored when 'df' is a vector.</p>
</td></tr>
<tr><td><code id="qq_plot_+3A_labels">labels</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), column and row labels are displayed.
This requires additional columns 'col' and 'row' in the data frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Q-Q plot with x- and y-axes displaying the theoretical and sample quantiles of
the effects, respectively.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Q-Q plot of the simulated plot errors in the example data frame 'error_df_bivar'
# for Trait 1 in Environment 1.

error_df &lt;- error_df_bivar[error_df_bivar$env == 1, ]

qq &lt;- qq_plot(
  df = error_df,
  effect = "e.Trait1",
  labels = TRUE
)

# Q-Q plot
qq

# Extract the data frame with the theoretical and sample quantiles of the
# user-defined effects.
qq_df &lt;- qq$data

</code></pre>

<hr>
<h2 id='rand_cor_mat'>Simulate a random correlation matrix</h2><span id='topic+rand_cor_mat'></span>

<h3>Description</h3>

<p>Creates a symmetric <code>n x n</code> correlation matrix with user-defined minimum and maximum
correlations based on a continuous uniform distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rand_cor_mat(
  n = 5,
  min.cor = -1,
  max.cor = 1,
  pos.def = FALSE,
  small.positive = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rand_cor_mat_+3A_n">n</code></td>
<td>
<p>A scalar defining the dimensions of the correlation matrix.</p>
</td></tr>
<tr><td><code id="rand_cor_mat_+3A_min.cor">min.cor</code></td>
<td>
<p>A scalar defining the minimum correlation.</p>
</td></tr>
<tr><td><code id="rand_cor_mat_+3A_max.cor">max.cor</code></td>
<td>
<p>A scalar defining the maximum correlation. <br />
<strong>Note:</strong> <code>-1 &lt; min.cor &lt; max.cor &lt; 1</code>.</p>
</td></tr>
<tr><td><code id="rand_cor_mat_+3A_pos.def">pos.def</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), the function <code>bend</code> of the R package
<a href="https://cran.r-project.org/package=mbend">'mbend'</a> is used to
bend a non-positive (semi)-definite matrix to be positive (semi)-definite.</p>
</td></tr>
<tr><td><code id="rand_cor_mat_+3A_small.positive">small.positive</code></td>
<td>
<p>Argument passed to <code>bend</code> when <code>pos.def = TRUE</code> (default is 1e-8).
Eigenvalues smaller than <code>small.positive</code> are replaced by this. <br />
<strong>Note:</strong> <code>0 &lt; small.positive &lt; 0.1</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A symmetric <code>n x n</code> correlation matrix. When <code>pos.def = TRUE</code>,
the correlation matrix is guaranteed to be positive (semi)-definite.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate and visualise a random correlation matrix with 10 columns and rows.
cor_mat &lt;- rand_cor_mat(
  n = 10,
  min.cor = -0.2,
  max.cor = 0.8,
  pos.def = TRUE
)

plot_matrix(
  mat = cor_mat,
  order = TRUE
)

</code></pre>

<hr>
<h2 id='rand_diag_mat'>Simulate a random diagonal variance matrix</h2><span id='topic+rand_diag_mat'></span>

<h3>Description</h3>

<p>Creates a diagonal <code>n x n</code> variance matrix with user-defined minimum and maximum
variances based on a continuous uniform distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rand_diag_mat(n = 5, min.var = 0, max.var = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rand_diag_mat_+3A_n">n</code></td>
<td>
<p>A scalar defining the dimensions of the variance matrix.</p>
</td></tr>
<tr><td><code id="rand_diag_mat_+3A_min.var">min.var</code></td>
<td>
<p>A scalar defining the minimum variance.</p>
</td></tr>
<tr><td><code id="rand_diag_mat_+3A_max.var">max.var</code></td>
<td>
<p>A scalar defining the maximum variance. <br />
<strong>Note:</strong> <code>0 &lt; min.var &lt; max.var</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A diagonal <code>n x n</code> variance matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate a random diagonal matrix with 10 columns and rows.
diag_mat &lt;- rand_diag_mat(
  n = 10,
  min.var = 0,
  max.var = 0.2
)

</code></pre>

<hr>
<h2 id='sample_met'>Sample environments from a target population</h2><span id='topic+sample_met'></span>

<h3>Description</h3>

<p>Creates a list of environments sampled from a population with user-defined sample size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_met(
  ntraits = 1,
  nenvs = 1000,
  nsamples = 10,
  sample.size = 20,
  replace = TRUE,
  cov.mat = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sample_met_+3A_ntraits">ntraits</code></td>
<td>
<p>A scalar defining the number of traits.</p>
</td></tr>
<tr><td><code id="sample_met_+3A_nenvs">nenvs</code></td>
<td>
<p>A scalar defining the number of environments in the target population.</p>
</td></tr>
<tr><td><code id="sample_met_+3A_nsamples">nsamples</code></td>
<td>
<p>A scalar defining the number of samples to be taken.</p>
</td></tr>
<tr><td><code id="sample_met_+3A_sample.size">sample.size</code></td>
<td>
<p>A vector defining the number of environments in each sample.
When only one value is specified, all samples will be assigned the same number.</p>
</td></tr>
<tr><td><code id="sample_met_+3A_replace">replace</code></td>
<td>
<p>When <code>TRUE</code> (default), samples are taken with replacement.
Ignored when <code>nsamples = 1</code>.</p>
</td></tr>
<tr><td><code id="sample_met_+3A_cov.mat">cov.mat</code></td>
<td>
<p>An optional matrix of environmental covariates for one or more traits.
When supplied, the covariates are sampled and printed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with elements given by the sample of environments taken from the target population.
When <code>cov.mat</code> is supplied, additional entries are given containing the sampled environmental
covariates for each trait.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample environments from a target population of 1000, with each sample containing 20 environments.
cov_ls &lt;- sample_met(
  nenvs = 1000,
  nsamples = 10,
  sample.size = 20,
  replace = TRUE
)

</code></pre>

<hr>
<h2 id='sample_variogram'>Sample variogram</h2><span id='topic+sample_variogram'></span>

<h3>Description</h3>

<p>Creates a sample variogram for a set of effects (e.g., plot errors).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_variogram(df, effect, min.np = 30)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sample_variogram_+3A_df">df</code></td>
<td>
<p>A data frame with the columns 'col', 'row', and the effects to be plotted.</p>
</td></tr>
<tr><td><code id="sample_variogram_+3A_effect">effect</code></td>
<td>
<p>The name of the effects to be plotted.</p>
</td></tr>
<tr><td><code id="sample_variogram_+3A_min.np">min.np</code></td>
<td>
<p>Minimum number of pairs for which semivariances are displayed (default is 30).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sample variogram with x- and y-axes displaying the row and
column displacements, and the z-axis displaying the average semivariances (variogram ordinates)
for the effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample variogram of plot errors simulated using a separable first order
# autoregressive (AR1) process.

error_df &lt;- field_trial_error(
  ntraits = 1,
  nenvs = 1,
  spatial.model = "AR1"
)

variogram &lt;- sample_variogram(
  df = error_df,
  effect = "e.Trait1"
)

# Sample variogram
variogram

# Extract the data frame with the column and row displacements, and the
# average semivariances.
variogram_df &lt;- variogram$data

</code></pre>

<hr>
<h2 id='skew_diag_mat'>Simulate a skewed diagonal variance matrix</h2><span id='topic+skew_diag_mat'></span>

<h3>Description</h3>

<p>Creates a diagonal <code>n x n</code> variance matrix with user-defined skewness
based on a gamma or inverse gamma distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>skew_diag_mat(n = 5, shape = 1.5, scale = 1, inverse = FALSE, mean.var = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="skew_diag_mat_+3A_n">n</code></td>
<td>
<p>A scalar defining the dimensions of the variance matrix.</p>
</td></tr>
<tr><td><code id="skew_diag_mat_+3A_shape">shape</code></td>
<td>
<p>A scalar defining the shape of the distribution.</p>
</td></tr>
<tr><td><code id="skew_diag_mat_+3A_scale">scale</code></td>
<td>
<p>A scalar defining the scale of the distribution.</p>
</td></tr>
<tr><td><code id="skew_diag_mat_+3A_inverse">inverse</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), the variances are
sampled from the inverse gamma distribution instead of the gamma distribution.</p>
</td></tr>
<tr><td><code id="skew_diag_mat_+3A_mean.var">mean.var</code></td>
<td>
<p>An optional scalar defining the mean variance.
.  When supplied, the variances are scaled to achieve the defined mean.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A diagonal <code>n x n</code> variance matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate a random diagonal matrix with 10 columns and rows, and negatively skewed variances
# scaled to a mean of 0.1.
diag_mat &lt;- skew_diag_mat(
  n = 10,
  shape = 1.5,
  scale = 1,
  mean.var = 0.1
)

</code></pre>

<hr>
<h2 id='struc_cor_mat'>Simulate a structured correlation matrix with reduced rank</h2><span id='topic+struc_cor_mat'></span>

<h3>Description</h3>

<p>Creates a symmetric <code>n x n</code> correlation matrix with user-defined structure and rank.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>struc_cor_mat(
  n = 5,
  base.cor = 0.5,
  range = NULL,
  rank = 3,
  skew = 0,
  base.mat = NULL,
  pos.def = FALSE,
  small.positive = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="struc_cor_mat_+3A_n">n</code></td>
<td>
<p>A scalar defining the dimensions of the correlation matrix.</p>
</td></tr>
<tr><td><code id="struc_cor_mat_+3A_base.cor">base.cor</code></td>
<td>
<p>A scalar defining the baseline correlation. <br />
<strong>Note:</strong> <code>-1 &lt; base.cor &lt; 1</code>.</p>
</td></tr>
<tr><td><code id="struc_cor_mat_+3A_range">range</code></td>
<td>
<p>A scalar defining the range of correlations around the baseline. By default,
<code>range = 1 - base.cor</code> which ensures the matrix is positive semi-definite with defined rank.
<strong>Note:</strong> <code>base.cor + range &lt;= 1</code>.</p>
</td></tr>
<tr><td><code id="struc_cor_mat_+3A_rank">rank</code></td>
<td>
<p>A scalar defining the rank of the correlation matrix.</p>
</td></tr>
<tr><td><code id="struc_cor_mat_+3A_skew">skew</code></td>
<td>
<p>A scalar defining the skewness imposed on the correlations.
<strong>Note:</strong> <code>-1 &lt; skew &lt; 1</code>.</p>
</td></tr>
<tr><td><code id="struc_cor_mat_+3A_base.mat">base.mat</code></td>
<td>
<p>An optional <code>n x n</code> base correlation matrix.
When supplied, <code>base.cor</code> and <code>skew</code> are ignored and noise is simulated based on <code>rank</code>.</p>
</td></tr>
<tr><td><code id="struc_cor_mat_+3A_pos.def">pos.def</code></td>
<td>
<p>When <code>TRUE</code> (default is <code>FALSE</code>), the function <code>bend</code> of the R package
<a href="https://cran.r-project.org/package=mbend">'mbend'</a> is used to
bend a non-positive (semi)-definite matrix to be positive (semi)-definite.</p>
</td></tr>
<tr><td><code id="struc_cor_mat_+3A_small.positive">small.positive</code></td>
<td>
<p>Argument passed to <code>bend</code> when <code>pos.def = TRUE</code> (default is 1e-8).
Eigenvalues smaller than <code>small.positive</code> are replaced by this. <br />
<strong>Note:</strong> <code>0 &lt; small.positive &lt; 0.1</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A symmetric <code>n x n</code> correlation matrix with defined rank.
When <code>pos.def = TRUE</code>, the correlation matrix is guaranteed to be positive (semi)-definite.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate and visualise a correlation matrix with 10 columns and rows, rank equal to 4 and
# negatively skewed correlations.
cor_mat &lt;- struc_cor_mat(
  n = 10,
  base.cor = 0.3,
  range = 0.7,
  rank = 4,
  skew = -0.5
)

plot_matrix(
  mat = cor_mat,
  order = TRUE
)

</code></pre>

<hr>
<h2 id='theoretical_variogram'>Theoretical variogram</h2><span id='topic+theoretical_variogram'></span>

<h3>Description</h3>

<p>Creates a theoretical variogram for a separable first order autoregressive (AR1) process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theoretical_variogram(
  ncols = 10,
  nrows = 20,
  varR = 1,
  col.cor = 0.5,
  row.cor = 0.7,
  prop.spatial = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="theoretical_variogram_+3A_ncols">ncols</code></td>
<td>
<p>A scalar defining the number of columns.</p>
</td></tr>
<tr><td><code id="theoretical_variogram_+3A_nrows">nrows</code></td>
<td>
<p>A scalar defining the number of rows.</p>
</td></tr>
<tr><td><code id="theoretical_variogram_+3A_varr">varR</code></td>
<td>
<p>A scalar defining the error variance.</p>
</td></tr>
<tr><td><code id="theoretical_variogram_+3A_col.cor">col.cor</code></td>
<td>
<p>A scalar defining the column autocorrelation,</p>
</td></tr>
<tr><td><code id="theoretical_variogram_+3A_row.cor">row.cor</code></td>
<td>
<p>A scalar defining the row autocorrelation.</p>
</td></tr>
<tr><td><code id="theoretical_variogram_+3A_prop.spatial">prop.spatial</code></td>
<td>
<p>A scalar defining the proportion of spatial trend.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A theoretical variogram with x- and y-axes displaying the row and column displacements,
and the z-axis displaying the semivariances (variogram ordinates) for a separable autoregressive process.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Theoretical variogram for a field trial with 10 columns and 20 rows, based
# on column and row autocorrelations of 0.5 and 0.7, and a proportion of
# spatial trend of 0.5. The remaining proportion represents random error.

variogram &lt;- theoretical_variogram(
  ncols = 10,
  nrows = 20,
  varR = 1,
  col.cor = 0.5,
  row.cor = 0.7,
  prop.spatial = 0.5
)

# Theoretical variogram
variogram

# Extract the data frame with the column and row displacements, and the
# theoretical semivariances.
variogram_df &lt;- variogram$data

</code></pre>

<hr>
<h2 id='unstr_asr_input'>Simulate genetic values based on an unstructured model for GxE interaction - 'AlphaSimR' input
parameters</h2><span id='topic+unstr_asr_input'></span>

<h3>Description</h3>

<p>Creates a list of input parameters for
<a href="https://CRAN.R-project.org/package=AlphaSimR">'AlphaSimR'</a> to simulate
genetic values in multiple environments for one or more traits based on an unstructured
model for genotype-by-environment (GxE) interaction. <br />
This function utilises the ability of 'AlphaSimR' to simulate correlated traits.
The wrapper function <code>unstr_asr_input()</code> is used to specify the input parameters required in 'AlphaSimR',
and can handle separable and non-separable structures between traits and
environments (see below).
After simulating the genetic values, the wrapper function <a href="#topic+unstr_asr_output">unstr_asr_output</a> can be used to
generate a data frame with output values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unstr_asr_input(
  ntraits = 1,
  nenvs = 2,
  mean = 0,
  var = 1,
  Tvar = NULL,
  Evar = NULL,
  corA = NULL,
  TcorA = NULL,
  EcorA = NULL,
  meanDD = NULL,
  varDD = NULL,
  TvarDD = NULL,
  EvarDD = NULL,
  corDD = NULL,
  TcorDD = NULL,
  EcorDD = NULL,
  relAA = NULL,
  TrelAA = NULL,
  ErelAA = NULL,
  corAA = NULL,
  TcorAA = NULL,
  EcorAA = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="unstr_asr_input_+3A_ntraits">ntraits</code></td>
<td>
<p>Number of traits to be simulated.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_nenvs">nenvs</code></td>
<td>
<p>Number of environments to be simulated (minimum of two).</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_mean">mean</code></td>
<td>
<p>A vector of mean genetic values for each environment-within-trait combination.
If only one value is specified, all combinations will be assigned the same mean.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_var">var</code></td>
<td>
<p>A vector of genetic variances for each environment-within-trait combination.
If only one value is specified, all combinations will be assigned
the same variance. <br />
<strong>Alternatively</strong>, if a separable structure between traits and environments is desired,
<code>Tvar</code> and <code>Evar</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_tvar">Tvar</code></td>
<td>
<p>A vector of genetic variances for each trait. Must be provided in combination with
<code>Evar</code>. <br />
<strong>Alternatively</strong>, <code>var</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_evar">Evar</code></td>
<td>
<p>A vector of genetic variances for each environment. Must be provided in
combination with <code>Tvar</code>. <br />
<strong>Alternatively</strong>, <code>var</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_cora">corA</code></td>
<td>
<p>A matrix of additive genetic correlations between environment-within-trait
combinations. By default, a diagonal matrix is constructed. <br />
<strong>Alternatively</strong>, <code>TcorA</code> and <code>EcorA</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_tcora">TcorA</code></td>
<td>
<p>A matrix of additive genetic correlations between traits. Must be provided in
combination with <code>EcorA</code>. <br />
<strong>Alternatively</strong>, <code>corA</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_ecora">EcorA</code></td>
<td>
<p>A matrix of additive genetic correlations between environments.
Must be provided in combination with <code>TcorA</code>. <br />
<strong>Alternatively</strong>, <code>corA</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_meandd">meanDD</code></td>
<td>
<p>A vector of mean dominance degrees for each environment-within-trait combination
(similar to <code>mean</code>). If only one value is specified, all
combinations will be assigned the same mean. By default, <code>meanDD = NULL</code> and dominance
is not simulated.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_vardd">varDD</code></td>
<td>
<p>A vector of dominance degree variances for each environment-within-trait
combination (similar to <code>var</code>). If only one value is specified, all
combinations will be assigned the same variance. <br />
<strong>Alternatively</strong>, if a separable structure between traits and environments is desired,
<code>TvarDD</code> and <code>EvarDD</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_tvardd">TvarDD</code></td>
<td>
<p>A vector of dominance degree variances for each trait (similar to <code>Tvar</code>).
Must be provided in combination with <code>EvarDD</code>. <br />
<strong>Alternatively</strong>, <code>varDD</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_evardd">EvarDD</code></td>
<td>
<p>A vector of dominance degree variances for each environment (similar to
<code>Evar</code>). Must be provided in combination with <code>TvarDD</code>. <br />
<strong>Alternatively</strong>, <code>varDD</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_cordd">corDD</code></td>
<td>
<p>A matrix of dominance degree correlations between environment-within-trait
combinations (similar to <code>corA</code>). If not specified and dominance is simulated, a
diagonal matrix is constructed. <br />
<strong>Alternatively</strong>, <code>TcorDD</code> and <code>EcorDD</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_tcordd">TcorDD</code></td>
<td>
<p>A matrix of dominance degree correlations between traits (similar to
<code>TcorA</code>). Must be provided in combination with <code>EcorDD</code>. <br />
<strong>Alternatively</strong>, <code>corDD</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_ecordd">EcorDD</code></td>
<td>
<p>A matrix of dominance degree correlations between environments (similar to
<code>EcorA</code>). Must be provided in combination with <code>TcorDD</code>. <br />
<strong>Alternatively</strong>, <code>corDD</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_relaa">relAA</code></td>
<td>
<p>A vector defining the relative magnitude of additive-by-additive (epistatic) variance
to additive genetic variance for each environment-within-trait combination,
that is in a diploid organism with allele frequency of 0.5. If only one value is specified,
all environment-within-trait combinations will be assigned the same value. By default,
<code>relAA = NULL</code> and epistasis is not simulated. <br />
<strong>Alternatively</strong>, if a separable structure between traits and environments is desired,
<code>TrelAA</code> and <code>ErelAA</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_trelaa">TrelAA</code></td>
<td>
<p>A vector defining the relative magnitude of epistatic variance
to additive genetic variance for each trait. Must be provided in combination
with <code>ErelAA</code>. <br />
<strong>Alternatively</strong>, <code>relAA</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_erelaa">ErelAA</code></td>
<td>
<p>A vector defining the relative magnitude of epistatic variance
to additive genetic variance for each environment. Must be provided in
combination with <code>TrelAA</code>. <br />
<strong>Alternatively</strong>, <code>relAA</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_coraa">corAA</code></td>
<td>
<p>A matrix of epistatic correlations between environment-within-trait
combinations (similar to <code>corA</code>). If not specified and epistasis is simulated, a
diagonal matrix is constructed. <br />
<strong>Alternatively</strong>, <code>TcorAA</code> and <code>EcorAA</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_tcoraa">TcorAA</code></td>
<td>
<p>A matrix of epistatic correlations between traits (similar to <code>TcorA</code>).
Must be provided in combination with <code>EcorAA</code>. <br />
<strong>Alternatively</strong>, <code>corAA</code> can be specified.</p>
</td></tr>
<tr><td><code id="unstr_asr_input_+3A_ecoraa">EcorAA</code></td>
<td>
<p>A matrix of epistatic correlations between environments (similar to
<code>EcorA</code>). Must be provided in combination with <code>TcorAA</code>. <br />
<strong>Alternatively</strong>, <code>corAA</code> can be specified.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>unstr_asr_input</code> can handle separable and non-separable structures between traits and
environments.
</p>

<ul>
<li><p> For separable structures, provide (1) <code>Tvar</code> &amp; <code>Evar</code>, and (2)
<code>TcorA</code> &amp; <code>EcorA</code>.
</p>
</li>
<li><p> For non-separable structures, provide (1) <code>var</code>, and (2) <code>corA</code>. <br />
</p>
</li></ul>

<p><strong>Note:</strong> 'AlphaSimR' can simulate different biological effects (see:
<a href="https://gaynorr.github.io/AlphaSimR/reference/SimParam.html">SimParam</a>).
</p>

<ul>
<li><p> For additive traits use <code>addTraitA()</code>.
</p>
</li>
<li><p> For additive + dominance traits use <code>addTraitAD()</code>.
</p>
</li>
<li><p> For additive + epistatic traits use <code>addTraitAE()</code>.
</p>
</li>
<li><p> For additive + dominance + epistatic traits use <code>addTraitADE()</code>.
</p>
</li></ul>

<p>Check the <code>useVarA</code> argument of these functions when simulating non-additive traits.
</p>


<h3>Value</h3>

<p>A list with input parameters for 'AlphaSimR', which are used to simulate
correlated genetic values based on an unstructured model for GxE interaction.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate genetic values with 'AlphaSimR' for two additive + dominance traits
# in two environments based on an unstructured model.

# 1. Define the genetic architecture of the simulated traits.
# Mean genetic values and mean dominance degrees.
mean &lt;- c(4.9, 5.4, 235.2, 228.5) # Trait 1 x 2 environments, Trait 2 x 2 environments
meanDD &lt;- c(0.4, 0.4, 0.1, 0.1) # Trait 1 and 2, same value for both environments

# Additive genetic variances and dominance degree variances.
var &lt;- c(0.086, 0.12, 15.1, 8.5) # Trait 1 x 2 environments, Trait 2 x 2 environments
varDD &lt;- 0.2 # Same value for all environment-within-trait combinations

# Additive genetic correlations between the two simulated traits.
TcorA &lt;- matrix(c(
  1.0, 0.6,
  0.6, 1.0
), ncol = 2)

# Additive genetic correlations between the two simulated environments.
EcorA &lt;- matrix(c(
  1.0, 0.2,
  0.2, 1.0
), ncol = 2)

# Dominance degree correlations between the four environment-within-trait combinations.
corDD &lt;- diag(4) # Assuming independence

input_asr &lt;- unstr_asr_input(
  ntraits = 2,
  nenvs = 2,
  mean = mean,
  var = var,
  TcorA = TcorA,
  EcorA = EcorA,
  meanDD = meanDD,
  varDD = varDD,
  corDD = corDD
)

</code></pre>

<hr>
<h2 id='unstr_asr_output'>Simulate genetic values based on an unstructured model for GxE interaction -
Simulation with 'AlphaSimR'</h2><span id='topic+unstr_asr_output'></span>

<h3>Description</h3>

<p>Creates a data frame of simulated genetic values in multiple environments for one or more traits
based on an unstructured model for genotype-by-environment (GxE) interaction. The wrapper function
<code>unstr_asr_output</code> requires an <a href="https://CRAN.R-project.org/package=AlphaSimR">'AlphaSimR'</a>
population object generated with <a href="#topic+unstr_asr_input">unstr_asr_input</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unstr_asr_output(pop, ntraits = 1, nenvs, nreps = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="unstr_asr_output_+3A_pop">pop</code></td>
<td>
<p>An <a href="https://CRAN.R-project.org/package=AlphaSimR">'AlphaSimR'</a> population object
(<a href="https://gaynorr.github.io/AlphaSimR/reference/Pop-class.html">Pop-class</a> or
<a href="https://gaynorr.github.io/AlphaSimR/reference/HybridPop-class.html">HybridPop-class</a>)
generated with <a href="#topic+unstr_asr_input">unstr_asr_input</a>.</p>
</td></tr>
<tr><td><code id="unstr_asr_output_+3A_ntraits">ntraits</code></td>
<td>
<p>Number of simulated traits specified in <a href="#topic+unstr_asr_input">unstr_asr_input</a>.</p>
</td></tr>
<tr><td><code id="unstr_asr_output_+3A_nenvs">nenvs</code></td>
<td>
<p>Number of simulated environments specified in <a href="#topic+unstr_asr_input">unstr_asr_input</a>.</p>
</td></tr>
<tr><td><code id="unstr_asr_output_+3A_nreps">nreps</code></td>
<td>
<p>A vector defining the number of replicates in each environment. If only one value
is specified, all environments will be assigned the same number.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns 'env', genotype 'id', and 'rep', followed by the
simulated genetic values for each trait.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate genetic values with 'AlphaSimR' for two additive + dominance traits
# in two environments based on an unstructured model.

# 1. Define the genetic architecture of the simulated traits.
# Mean genetic values and mean dominance degrees.
mean &lt;- c(4.9, 5.4, 235.2, 228.5) # Trait 1 x 2 environments, Trait 2 x 2 environments
meanDD &lt;- c(0.4, 0.4, 0.1, 0.1) # Trait 1 and 2, same value for both environments

# Additive genetic variances and dominance degree variances.
var &lt;- c(0.086, 0.12, 15.1, 8.5) # Trait 1 x 2 environments, Trait 2 x 2 environments
varDD &lt;- 0.2 # Same value for all environment-within-trait combinations

# Additive genetic correlations between the two simulated traits.
TcorA &lt;- matrix(c(
  1.0, 0.6,
  0.6, 1.0
), ncol = 2)

# Additive genetic correlations between the two simulated environments.
EcorA &lt;- matrix(c(
  1.0, 0.2,
  0.2, 1.0
), ncol = 2)

# Dominance degree correlations between the four environment-within-trait combinations.
corDD &lt;- diag(4) # Assuming independence

input_asr &lt;- unstr_asr_input(
  ntraits = 2,
  nenvs = 2,
  mean = mean,
  var = var,
  TcorA = TcorA,
  EcorA = EcorA,
  meanDD = meanDD,
  varDD = varDD,
  corDD = corDD
)


# 2. Use input_asr to simulate genetic values with 'AlphaSimR' based on an
# unstructured model.

library("AlphaSimR")
FOUNDERPOP &lt;- quickHaplo(
  nInd = 10,
  nChr = 1,
  segSites = 20
)

SP &lt;- SimParam$new(FOUNDERPOP)



SP$addTraitAD(
  nQtlPerChr = 20,
  mean = input_asr$mean,
  var = input_asr$var,
  corA = input_asr$corA,
  meanDD = input_asr$meanDD,
  varDD = input_asr$varDD,
  corDD = input_asr$corDD,
  useVarA = TRUE
)

# By default, the variances in 'var' represent additive genetic variances.
# When useVarA = FALSE, the values represent total genetic variances.

pop &lt;- newPop(FOUNDERPOP)


# 3. Create a data frame with simulated genetic values for the two traits in
# the two environments, with two replicates of each genotype.

gv_df &lt;- unstr_asr_output(
  pop = pop,
  ntraits = 2,
  nenvs = 2,
  nreps = 2
)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
