<!DOCTYPE html><html><head><title>Help for package file2meco</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {file2meco}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#check_match_table'><p>Replace the names use match table</p></a></li>
<li><a href='#check_sample_table'><p>Read sample table</p></a></li>
<li><a href='#CHOCOPhlAn_taxonomy'><p>The CHOCOPhlAn_taxonomy data</p></a></li>
<li><a href='#file2meco'><p>Introduction to file2meco package (<a href="https://github.com/ChiLiubio/file2meco">https://github.com/ChiLiubio/file2meco</a>)</p></a></li>
<li><a href='#humann2meco'><p>Transform 'HUMAnN' metagenomic results to 'microtable' object.</p></a></li>
<li><a href='#meco2phyloseq'><p>Transform 'microtable' object of 'microeco' package to the 'phyloseq' object of 'phyloseq' package.</p></a></li>
<li><a href='#MetaCyc_pathway_map'><p>The MetaCyc_pathway_map data</p></a></li>
<li><a href='#metacyc_pathway_website'><p>Get the website for a 'MetaCyc' pathway name</p></a></li>
<li><a href='#mpa2meco'><p>Transform metagenomic classification results of 'mpa' format to 'microtable' object.</p></a></li>
<li><a href='#ncyc_map'><p>The ncyc_map data</p></a></li>
<li><a href='#ncyc2meco'><p>Transform 'NCycDB' or 'PCycDB' metagenomic abundance to 'microtable' object.</p></a></li>
<li><a href='#pcyc_map'><p>The pcyc_map data</p></a></li>
<li><a href='#phyloseq2meco'><p>Transform the 'phyloseq' object of 'phyloseq' package to 'microtable' object of 'microeco' package.</p></a></li>
<li><a href='#qiime1meco'><p>Transform 'QIIME' results to 'microtable' object.</p></a></li>
<li><a href='#qiime2meco'><p>Transform 'QIIME2' results to 'microtable' object.</p></a></li>
<li><a href='#vs2meco'><p>Transform viromescan results to 'microtable' object.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Transform Files to 'microtable' Object with 'microeco' Package</td>
</tr>
<tr>
<td>Version:</td>
<td>0.7.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Chi Liu [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chi Liu &lt;liuchi0426@126.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Transform output files of some tools to the 'microtable' object of 'microtable' class in 'microeco' package. The 'microtable' class is the basic class in 'microeco' package and is necessary for the downstream microbial community data analysis.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ChiLiubio/file2meco">https://github.com/ChiLiubio/file2meco</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>microeco (&ge; 1.2.0), ape, magrittr, dplyr, tidyr, yaml, rhdf5,
Matrix</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Biostrings, seqinr, phyloseq, readxl</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-07 05:45:06 UTC; Chi</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-07 06:20:02 UTC</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
</table>
<hr>
<h2 id='check_match_table'>Replace the names use match table</h2><span id='topic+check_match_table'></span>

<h3>Description</h3>

<p>Replace the names use match table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_match_table(match_table = NULL, abund_new = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_match_table_+3A_match_table">match_table</code></td>
<td>
<p>default NULL; character or data.frame; matching table used.</p>
</td></tr>
<tr><td><code id="check_match_table_+3A_abund_new">abund_new</code></td>
<td>
<p>default NULL; data.frame; the abundance table used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>new abundance table.
</p>

<hr>
<h2 id='check_sample_table'>Read sample table</h2><span id='topic+check_sample_table'></span>

<h3>Description</h3>

<p>Read sample table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_sample_table(sample_table = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_sample_table_+3A_sample_table">sample_table</code></td>
<td>
<p>default NULL; character or data.frame; matching table used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sample information table.
</p>

<hr>
<h2 id='CHOCOPhlAn_taxonomy'>The CHOCOPhlAn_taxonomy data</h2><span id='topic+CHOCOPhlAn_taxonomy'></span>

<h3>Description</h3>

<p>The CHOCOPhlAn_taxonomy data is used for the parsing the 'HUMAaN' metagenomic results and add the taxonomy hierarchical information to the 'tax_table'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(CHOCOPhlAn_taxonomy)
</code></pre>

<hr>
<h2 id='file2meco'>Introduction to file2meco package (<a href="https://github.com/ChiLiubio/file2meco">https://github.com/ChiLiubio/file2meco</a>)</h2><span id='topic+file2meco'></span>

<h3>Description</h3>

<p>For the detailed tutorial on the file2meco package, please follow the tutorial link in the github repository
(<a href="https://github.com/ChiLiubio/file2meco">https://github.com/ChiLiubio/file2meco</a>)
</p>
<p>Please open the help document by using <code>help</code> function or by clicking the following links collected:<br />
<code><a href="#topic+qiime1meco">qiime1meco</a></code><br />
<code><a href="#topic+qiime2meco">qiime2meco</a></code><br />
<code><a href="#topic+humann2meco">humann2meco</a></code><br />
<code><a href="#topic+mpa2meco">mpa2meco</a></code><br />
<code><a href="#topic+ncyc2meco">ncyc2meco</a></code><br />
<code><a href="#topic+phyloseq2meco">phyloseq2meco</a></code><br />
<code><a href="#topic+meco2phyloseq">meco2phyloseq</a></code><br />
<code><a href="#topic+vs2meco">vs2meco</a></code><br />
</p>
<p>To report bugs or discuss questions, please use Github Issues (<a href="https://github.com/ChiLiubio/file2meco/issues">https://github.com/ChiLiubio/file2meco/issues</a>).
Before creating a new issue, 
please read the guideline (<a href="https://chiliubio.github.io/microeco_tutorial/notes.html#github-issues">https://chiliubio.github.io/microeco_tutorial/notes.html#github-issues</a>).
</p>
<p>To cite file2meco package in publications, please run the following command to get the reference: <code>citation("file2meco")</code><br />
</p>
<p>Reference:<br />
Liu, C., Li, X., Mansoldo, F.R.P., An, J., Kou, Y., Zhang, X., Wang, J., Zeng, J., Vermelho, A.B., Yao, M., 2022. 
Microbial habitat specificity largely affects microbial co-occurrence patterns and functional profiles in wetland soils. Geoderma 418, 115866.
</p>

<hr>
<h2 id='humann2meco'>Transform 'HUMAnN' metagenomic results to 'microtable' object.</h2><span id='topic+humann2meco'></span>

<h3>Description</h3>

<p>Transform 'HUMAnN' metagenomic results to microtable object, reference: Franzosa et al. (2018) &lt;doi:10.1038/s41592-018-0176-y&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>humann2meco(
  feature_table,
  db = c("MetaCyc", "KEGG", "gene")[1],
  sample_table = NULL,
  match_table = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="humann2meco_+3A_feature_table">feature_table</code></td>
<td>
<p>file path of 'HUMAnN' output abundance table; Please see the example.</p>
</td></tr>
<tr><td><code id="humann2meco_+3A_db">db</code></td>
<td>
<p>default &quot;MetaCyc&quot;; one of &quot;MetaCyc&quot;, &quot;KEGG&quot; or &quot;gene&quot;; &quot;MetaCyc&quot; or &quot;KEGG&quot; means the input feature table is pathway abundance.
&quot;gene&quot; represents the abundance of genes, such as 'eggNOG', 'KO' and 'EC'. 
When using &quot;gene&quot;, the generated tax_table has only taxonomic lineages and gene name, no higher functional levels.</p>
</td></tr>
<tr><td><code id="humann2meco_+3A_sample_table">sample_table</code></td>
<td>
<p>default NULL; sample metadata table; If provided, must be one of the several types of formats: <br />
1) comma seperated file with the suffix csv or tab seperated file with suffix tsv or txt; <br />
2) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed; <br />
3) <code>data.frame</code> object from R.</p>
</td></tr>
<tr><td><code id="humann2meco_+3A_match_table">match_table</code></td>
<td>
<p>default NULL; a two column table used to replace the sample names in feature table; Must be two columns without column names;
The first column must be raw sample names same with those in feature table, 
the second column must be new sample names same with the rownames in sample_table; Please also see the example files.
If provided, must be one of the several types of formats: <br />
1) comma seperated file with the suffix csv or tab seperated file with suffix tsv/txt; <br />
2) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed; <br />
3) <code>data.frame</code> object from R.</p>
</td></tr>
<tr><td><code id="humann2meco_+3A_...">...</code></td>
<td>
<p>parameter passed to <code>microtable$new</code> function of <code>microeco</code> package, such as <code>auto_tidy</code> parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>microtable</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(file2meco)
library(microeco)
library(magrittr)
sample_file_path &lt;- system.file("extdata", "example_metagenome_sample_info.tsv", 
  package="file2meco")
match_file_path &lt;- system.file("extdata", "example_metagenome_match_table.tsv", package="file2meco")
# MetaCyc pathway examples
# use the raw data files stored inside the package for MetaCyc pathway database based analysis
abund_file_path &lt;- system.file("extdata", "example_HUMAnN_MetaCyc_abund.tsv", package="file2meco")
# the default db is "MetaCyc"
humann2meco(abund_file_path, db = "MetaCyc")
humann2meco(abund_file_path, db = "MetaCyc", sample_table = sample_file_path, 
  match_table = match_file_path)
test &lt;- humann2meco(abund_file_path, db = "MetaCyc", sample_table = sample_file_path, 
  match_table = match_file_path)
test$tidy_dataset()
# rel = FALSE sum original abundance instead of relative abundance
test$cal_abund(select_cols = 1:3, rel = FALSE)
test$taxa_abund$Superclass1 %&lt;&gt;% .[!grepl("unclass", rownames(.)), ]
# use_percentage = FALSE disable percentage for relative abundance
test1 &lt;- trans_abund$new(test, taxrank = "Superclass1", ntaxa = 10, use_percentage = FALSE)
# reassign ylab title instead of default 'Relative Abundance'
test1$ylabname &lt;- "Abundance (RPK)"
# bar_type = "notfull" show original abundance instead of normalized 0-1
test1$plot_bar(facet = "Group", bar_type = "notfull")
# select both function and taxa
test$cal_abund(select_cols = c("Superclass1", "Phylum", "Genus"), rel = TRUE)
test1 &lt;- trans_abund$new(test, taxrank = "Phylum", ntaxa = 10, delete_taxonomy_lineage = TRUE)
test1$plot_bar(facet = "Group")
test$taxa_abund$Phylum %&lt;&gt;% .[!grepl("unclass", rownames(.)), ]
test1 &lt;- trans_abund$new(test, taxrank = "Phylum", ntaxa = 10, delete_taxonomy_lineage = FALSE)
test1$plot_bar(facet = "Group")
# functional biomarker
test$cal_abund(select_cols = 1:3, rel = TRUE)
test$taxa_abund$Superclass1 %&lt;&gt;% .[!grepl("unclass", rownames(.)), ]
test$taxa_abund$Superclass2 %&lt;&gt;% .[!grepl("unclass", rownames(.)), ]
test$taxa_abund$pathway %&lt;&gt;% .[!grepl("unclass", rownames(.)), ]
test1 &lt;- trans_diff$new(test, method = "lefse", group = "Group")
test1$plot_diff_bar(use_number = 1:20)
# taxa biomarker
test$cal_abund(select_cols = 4:9, rel = TRUE)
test$taxa_abund$Phylum %&lt;&gt;% .[!grepl("unclass", rownames(.)), ]
test1 &lt;- trans_diff$new(test, method = "lefse", group = "Group", p_adjust_method = "none")
test1$plot_diff_bar(threshold = 2)
#############################################################
# KEGG pathway examples
abund_file_path &lt;- system.file("extdata", "example_HUMAnN_KEGG_abund.tsv", package="file2meco")
humann2meco(abund_file_path, db = "KEGG")
test &lt;- humann2meco(abund_file_path, db = "KEGG", 
  sample_table = sample_file_path, match_table = match_file_path)
test$tax_table %&lt;&gt;% subset(Level.1 != "unclassified")
test$tidy_dataset()
test$cal_abund(select_cols = 1:3, rel = FALSE)
# use_percentage = FALSE disable percentage for relative abundance
test1 &lt;- trans_abund$new(test, taxrank = "Level.2", ntaxa = 10, use_percentage = FALSE)
# or use ggplot2::ylab to change ylab title
test1$ylabname &lt;- "Abundance (RPK)"
test1$plot_bar(facet = "Group", bar_type = "notfull")
# select both function and taxa
test$cal_abund(select_cols = c("Level.1", "Phylum", "Genus"), rel = TRUE)
test1 &lt;- trans_abund$new(test, taxrank = "Phylum", ntaxa = 10, delete_taxonomy_lineage = FALSE)
test1$plot_bar(facet = "Group")
# functional biomarker
test$cal_abund(select_cols = 1:3, rel = TRUE)
test1 &lt;- trans_diff$new(test, method = "lefse", group = "Group")
test1$plot_diff_bar(threshold = 3)
# taxa biomarker
test$cal_abund(select_cols = 4:9, rel = TRUE)
test1 &lt;- trans_diff$new(test, method = "lefse", group = "Group", p_adjust_method = "none")
test1$plot_diff_bar(threshold = 2)

</code></pre>

<hr>
<h2 id='meco2phyloseq'>Transform 'microtable' object of 'microeco' package to the 'phyloseq' object of 'phyloseq' package.</h2><span id='topic+meco2phyloseq'></span>

<h3>Description</h3>

<p>Transform 'microtable' object of 'microeco' package to the 'phyloseq' object of 'phyloseq' package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meco2phyloseq(dataset)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meco2phyloseq_+3A_dataset">dataset</code></td>
<td>
<p>a microtable object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>phyloseq object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(microeco)
data("dataset")
meco2phyloseq(dataset)

## End(Not run)
</code></pre>

<hr>
<h2 id='MetaCyc_pathway_map'>The MetaCyc_pathway_map data</h2><span id='topic+MetaCyc_pathway_map'></span>

<h3>Description</h3>

<p>The MetaCyc_pathway_map data is a manually curated 'MetaCyc' pathway hierarchical structure data. 
It is used for the parsing the 'HUMAaN' metagenomic abundance table associated with 'MetaCyc' database.
Currently, only superclass 1, 2 and the pathway are used in this data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(MetaCyc_pathway_map)
</code></pre>

<hr>
<h2 id='metacyc_pathway_website'>Get the website for a 'MetaCyc' pathway name</h2><span id='topic+metacyc_pathway_website'></span>

<h3>Description</h3>

<p>Get the website for a 'MetaCyc' pathway name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metacyc_pathway_website(pathway = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metacyc_pathway_website_+3A_pathway">pathway</code></td>
<td>
<p>default NULL; character vector; one or more MetaCyc pathway names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>metacyc_pathway_website("FOLSYN-PWY")
</code></pre>

<hr>
<h2 id='mpa2meco'>Transform metagenomic classification results of 'mpa' format to 'microtable' object.</h2><span id='topic+mpa2meco'></span>

<h3>Description</h3>

<p>Transform the classification results of mpa (MetaPhlAn) format to microtable object,
such as MetaPhlAn and Kraken2 results. Kraken2 results can be obtained by merge_metaphlan_tables.py from MetaPhlAn or 
combine_mpa.py from KrakenTools (https://ccb.jhu.edu/software/krakentools/).
The algorithm of Kraken2 determines that the abundance of a taxon is not equal to the sum of abundances of taxa in its subordinate lineage.
So the default tables in taxa_abund of return microtable object are extracted from the abundances of raw file. 
It is totally different with the return taxa_abund of cal_abund function, 
which sums the abundances of taxa at different taxonomic levels based on the taxonomic table and the otu_table 
(i.e., taxa abundance table at a specified level, e.g., 's__').
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mpa2meco(
  feature_table,
  sample_table = NULL,
  match_table = NULL,
  use_level = "s__",
  rel = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mpa2meco_+3A_feature_table">feature_table</code></td>
<td>
<p>'mpa' format abundance table, see the example.</p>
</td></tr>
<tr><td><code id="mpa2meco_+3A_sample_table">sample_table</code></td>
<td>
<p>default NULL; sample metadata table; If provided, must be one of the several types of formats: 
1) comma seperated file with the suffix csv or tab seperated file with suffix tsv/txt; 
2) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed; 
3) <code>data.frame</code> object from R.</p>
</td></tr>
<tr><td><code id="mpa2meco_+3A_match_table">match_table</code></td>
<td>
<p>default NULL; a two column table used to replace the sample names in abundance table; Must be two columns without column names;
The first column must be raw sample names same with those in feature table, 
the second column must be new sample names same with the rownames in sample_table; Please also see the example files.</p>
</td></tr>
<tr><td><code id="mpa2meco_+3A_use_level">use_level</code></td>
<td>
<p>default &quot;s__&quot;; the prefix parsed for the otu_table and tax_table; must be one of 'd__', 'k__', 'p__', 'c__', 'o__', 'f__', 'g__' and 's__'.</p>
</td></tr>
<tr><td><code id="mpa2meco_+3A_rel">rel</code></td>
<td>
<p>default FALSE; Whether convert the original abundance to relative abundance in generated taxa_abund list. 
If TRUE, all the data.frame objects in taxa_abund list have relative abundance (0-1).</p>
</td></tr>
<tr><td><code id="mpa2meco_+3A_...">...</code></td>
<td>
<p>parameter passed to microtable$new function of microeco package, such as auto_tidy parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>microtable object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(microeco)
library(file2meco)
library(magrittr)
# use Kraken2 file stored inside the package
abund_file_path &lt;- system.file("extdata", "example_kraken2_merge.txt", package="file2meco")
mpa2meco(abund_file_path)
# add sample information table
sample_file_path &lt;- system.file("extdata", "example_metagenome_sample_info.tsv", 
  package="file2meco")
# sample names are different between abund_file_path and sample_file_path; 
# use a matching table to adjust them
match_file_path &lt;- system.file("extdata", "example_metagenome_match_table.tsv", package="file2meco")
test &lt;- mpa2meco(abund_file_path, sample_table = sample_file_path, 
  match_table = match_file_path, use_level = "s__", rel = TRUE)
# make the taxonomy standard for the following analysis
test$tax_table %&lt;&gt;% tidy_taxonomy
test$tidy_dataset()
# calculate taxa_abund with specified level instead of raw kraken results
test1 &lt;- clone(test)
test1$cal_abund(rel = TRUE)
identical(test$taxa_abund$Kingdom, test1$taxa_abund$Kingdom)

</code></pre>

<hr>
<h2 id='ncyc_map'>The ncyc_map data</h2><span id='topic+ncyc_map'></span>

<h3>Description</h3>

<p>The ncyc_map data is used for the parsing the 'NCycDB' metagenomic results and add the N cycle pathway information to the 'tax_table' of 'microtable' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ncyc_map)
</code></pre>

<hr>
<h2 id='ncyc2meco'>Transform 'NCycDB' or 'PCycDB' metagenomic abundance to 'microtable' object.</h2><span id='topic+ncyc2meco'></span>

<h3>Description</h3>

<p>Transform 'NCycDB' or 'PCycDB' metagenomic abundance to microtable object. 
The function can identify the mapping database according to the gene names of input feature abundance table.
Reference: Qichao et al. (2019) &lt;doi: 10.1093/bioinformatics/bty741&gt; and Zeng et al. (2022) &lt;doi: 10.1186/s40168-022-01292-1&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncyc2meco(feature_table, sample_table = NULL, match_table = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ncyc2meco_+3A_feature_table">feature_table</code></td>
<td>
<p>'NCycDB' or 'PCycDB' output abundance table, see the example file.</p>
</td></tr>
<tr><td><code id="ncyc2meco_+3A_sample_table">sample_table</code></td>
<td>
<p>default NULL; sample metadata table; If provided, must be one of the several types of formats: 
1) comma seperated file with the suffix csv or tab seperated file with suffix tsv/txt; 
2) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed; 
3) <code>data.frame</code> object from R.
A file path must be tab or comma seperated file, generally, a file with suffix &quot;tsv&quot; or &quot;csv&quot;.</p>
</td></tr>
<tr><td><code id="ncyc2meco_+3A_match_table">match_table</code></td>
<td>
<p>default NULL; a two column table used to replace the sample names in abundance table; Must be two columns without column names;
The first column must be raw sample names same with those in feature table, 
the second column must be new sample names same with the rownames in sample_table; Please also see the example files.
A file path must be tab or comma seperated file, e.g. a file with suffix &quot;tsv&quot; or &quot;csv&quot;.</p>
</td></tr>
<tr><td><code id="ncyc2meco_+3A_...">...</code></td>
<td>
<p>parameter passed to microtable$new function of microeco package, such as auto_tidy parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>microtable object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# NCycDB
abund_file_path &lt;- system.file("extdata", "example_Ncyc_table.tsv", package="file2meco")
sample_file_path &lt;- system.file("extdata", "example_metagenome_sample_info.tsv", 
  package="file2meco")
match_file_path &lt;- system.file("extdata", "example_metagenome_match_table.tsv", package="file2meco")
library(microeco)
library(file2meco)
library(magrittr)
ncyc2meco(abund_file_path)
test &lt;- ncyc2meco(abund_file_path, sample_table = sample_file_path, 
  match_table = match_file_path)
test$tidy_dataset()
# use split_group = TRUE to calculate the pathway abundance with multipe map correspondance
test$cal_abund(select_cols = 1, rel = TRUE, split_group = TRUE, split_column = "Pathway")
test$taxa_abund$Pathway %&lt;&gt;% .[!grepl("unclass", rownames(.)), ]
test1 &lt;- trans_abund$new(test, taxrank = "Pathway")
test1$plot_bar(bar_type = "notfull")
# for gene abundance, no splitting on the Pathway
test$cal_abund(select_cols = 1:2, rel = TRUE, split_group = FALSE)
test$taxa_abund$Gene %&lt;&gt;% .[!grepl("unclass", rownames(.)), ]
test1 &lt;- trans_abund$new(test, taxrank = "Gene")
test1$plot_bar(bar_type = "notfull")

# PCycDB
abund_file_path &lt;- system.file("extdata", "example_Pcyc_table.tsv", package="file2meco")
test &lt;- ncyc2meco(abund_file_path)
test$tidy_dataset()
# show pathway abundance
test$cal_abund(select_cols = 1, rel = TRUE, split_group = TRUE, split_by = "&amp;&amp;", 
    split_column = "Pathway")
test$taxa_abund$Pathway %&lt;&gt;% .[!grepl("unclass|Others", rownames(.)), ]
test1 &lt;- trans_abund$new(test, taxrank = "Pathway")
test1$plot_bar(bar_type = "notfull")
# show gene abundance
test$cal_abund(select_cols = 2, rel = TRUE, split_group = FALSE)
test$taxa_abund$Gene %&lt;&gt;% .[!grepl("unclass", rownames(.)), ]
test1 &lt;- trans_abund$new(test, taxrank = "Gene")
test1$plot_bar(bar_type = "notfull")

</code></pre>

<hr>
<h2 id='pcyc_map'>The pcyc_map data</h2><span id='topic+pcyc_map'></span>

<h3>Description</h3>

<p>The pcyc_map data is used for the parsing the 'PCycDB' metagenomic results and add the P cycle pathway information to the 'tax_table' of 'microtable' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pcyc_map)
</code></pre>

<hr>
<h2 id='phyloseq2meco'>Transform the 'phyloseq' object of 'phyloseq' package to 'microtable' object of 'microeco' package.</h2><span id='topic+phyloseq2meco'></span>

<h3>Description</h3>

<p>Transform the 'phyloseq' object of 'phyloseq' package to 'microtable' object of 'microeco' package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyloseq2meco(physeq, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyloseq2meco_+3A_physeq">physeq</code></td>
<td>
<p>a phyloseq object.</p>
</td></tr>
<tr><td><code id="phyloseq2meco_+3A_...">...</code></td>
<td>
<p>parameter passed to microtable$new function of microeco package, such as auto_tidy parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>microtable object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(phyloseq)
data("GlobalPatterns")
phyloseq2meco(GlobalPatterns)

## End(Not run)
</code></pre>

<hr>
<h2 id='qiime1meco'>Transform 'QIIME' results to 'microtable' object.</h2><span id='topic+qiime1meco'></span>

<h3>Description</h3>

<p>Transform 'QIIME' results to microtable object. 
The QIIME results refer in particular to the files of qiime1 software.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qiime1meco(
  feature_table,
  sample_table = NULL,
  match_table = NULL,
  phylo_tree = NULL,
  rep_fasta = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qiime1meco_+3A_feature_table">feature_table</code></td>
<td>
<p>the otu table generated from 'QIIME'. Taxonomic information should be in the end of the file.</p>
</td></tr>
<tr><td><code id="qiime1meco_+3A_sample_table">sample_table</code></td>
<td>
<p>default NULL; sample metadata table; If provided, must be one of the several types of formats: 
1) comma seperated file with the suffix csv or tab seperated file with suffix tsv/txt; 
2) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed; 
3) <code>data.frame</code> object from R.</p>
</td></tr>
<tr><td><code id="qiime1meco_+3A_match_table">match_table</code></td>
<td>
<p>default NULL; a two column table used to replace the sample names in feature table; Must be two columns without column names;
The first column must be raw sample names same with those in feature table, 
the second column must be new sample names same with the rownames in sample_table; Please also see the example files.
If provided, must be one of the several types of formats: <br />
1) comma seperated file with the suffix csv or tab seperated file with suffix tsv/txt; <br />
2) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed; <br />
3) <code>data.frame</code> object from R.</p>
</td></tr>
<tr><td><code id="qiime1meco_+3A_phylo_tree">phylo_tree</code></td>
<td>
<p>default NULL; the phylogenetic tree; generally, a file with suffix &quot;tre&quot;.</p>
</td></tr>
<tr><td><code id="qiime1meco_+3A_rep_fasta">rep_fasta</code></td>
<td>
<p>default NULL; the representative sequences; a fasta file, generally with suffix &quot;fasta&quot; or &quot;fna&quot; or &quot;fa&quot;.</p>
</td></tr>
<tr><td><code id="qiime1meco_+3A_...">...</code></td>
<td>
<p>parameter passed to microtable$new function of microeco package, such as <code>auto_tidy</code> parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>microtable</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# use the raw data files stored inside the package
otu_file_path &lt;- system.file("extdata", "otu_table_raw.txt", package="file2meco")
sample_file_path &lt;- system.file("extdata", "sample_info.csv", package="file2meco")
phylo_file_path &lt;- system.file("extdata", "rep_phylo.tre", package="file2meco")
rep_fasta_path &lt;- system.file("extdata", "rep.fna", package="file2meco")
qiime1meco(otu_file_path, sample_table = sample_file_path)
qiime1meco(otu_file_path, sample_table = sample_file_path, 
  phylo_tree = phylo_file_path)
qiime1meco(otu_file_path, sample_table = sample_file_path, 
  phylo_tree = phylo_file_path, rep_fasta = rep_fasta_path)

## End(Not run)
</code></pre>

<hr>
<h2 id='qiime2meco'>Transform 'QIIME2' results to 'microtable' object.</h2><span id='topic+qiime2meco'></span>

<h3>Description</h3>

<p>Transform 'QIIME2' qza results to microtable object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qiime2meco(
  feature_table,
  sample_table = NULL,
  match_table = NULL,
  taxonomy_table = NULL,
  phylo_tree = NULL,
  rep_fasta = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qiime2meco_+3A_feature_table">feature_table</code></td>
<td>
<p>the ASV abundance data with qza format, such as the <code>'data2_table.qza'</code> in the example.</p>
</td></tr>
<tr><td><code id="qiime2meco_+3A_sample_table">sample_table</code></td>
<td>
<p>default NULL; the sample metadata table; four types of formats are available: <br />
1) q2-type tab seperated file of QIIME2, such as the <code>'sample-metadata.tsv'</code> in the example;<br />
2) comma seperated file with the suffix csv or tab seperated file with suffix tsv or txt;<br />
3) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed;<br />
4) <code>data.frame</code> object from R.</p>
</td></tr>
<tr><td><code id="qiime2meco_+3A_match_table">match_table</code></td>
<td>
<p>default NULL; a two column table used to replace the sample names in feature table; Must be two columns without column names;
The first column must be raw sample names same with those in feature table, 
the second column must be new sample names same with the rownames in sample_table; Please also see the example files.
If provided, must be one of the several types of formats: <br />
1) comma seperated file with the suffix csv or tab seperated file with suffix tsv or txt; <br />
2) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed; <br />
3) <code>data.frame</code> object from R.</p>
</td></tr>
<tr><td><code id="qiime2meco_+3A_taxonomy_table">taxonomy_table</code></td>
<td>
<p>default NULL; the taxonomy assignment data with qza format, such as the <code>'taxonomy.qza'</code> in the example.</p>
</td></tr>
<tr><td><code id="qiime2meco_+3A_phylo_tree">phylo_tree</code></td>
<td>
<p>default NULL; the phylogenetic tree with qza format, such as the <code>'tree.qza'</code> in the example.</p>
</td></tr>
<tr><td><code id="qiime2meco_+3A_rep_fasta">rep_fasta</code></td>
<td>
<p>default NULL; the representative sequences with qza format, such as the <code>'dada2_rep_set.qza'</code> in the example.</p>
</td></tr>
<tr><td><code id="qiime2meco_+3A_...">...</code></td>
<td>
<p>parameter passed to <code>microtable$new</code> function of <code>microeco</code> package, such as <code>auto_tidy</code> parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>microtable</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# The data files is downloaded from https://docs.qiime2.org/2020.8/tutorials/pd-mice/ 
#   and stored inside the package.
abund_file_path &lt;- system.file("extdata", "dada2_table.qza", package="file2meco")
sample_file_path &lt;- system.file("extdata", "sample-metadata.tsv", package="file2meco")
taxonomy_file_path &lt;- system.file("extdata", "taxonomy.qza", package="file2meco")
qiime2meco(abund_file_path)
qiime2meco(abund_file_path, sample_table = sample_file_path)
qiime2meco(abund_file_path, sample_table = sample_file_path, 
  taxonomy_table = taxonomy_file_path)

## End(Not run)
</code></pre>

<hr>
<h2 id='vs2meco'>Transform viromescan results to 'microtable' object.</h2><span id='topic+vs2meco'></span>

<h3>Description</h3>

<p>Transform the results of viromescan software to microtable object.
The output of viromescan is single file for each sample.
All the results are needed to be merged and adjusted (for several chaotic taxonomy).
The input should be the 'count' tables at Species level, i.e. Species_level_results-Counts.txt.
For more details, please see the reference &lt;DOI: 10.1186/s12864-016-2446-3&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vs2meco(input_dir, sample_table = NULL, match_table = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vs2meco_+3A_input_dir">input_dir</code></td>
<td>
<p>the input directory, containing all the result folders for each sample. Each folder should be named by the sample name.</p>
</td></tr>
<tr><td><code id="vs2meco_+3A_sample_table">sample_table</code></td>
<td>
<p>default NULL; sample metadata table; If provided, must be one of the several types of formats:<br />
1) comma seperated file with the suffix csv or tab seperated file with suffix tsv/txt;<br />
2) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed;<br />
3) <code>data.frame</code> object from R.</p>
</td></tr>
<tr><td><code id="vs2meco_+3A_match_table">match_table</code></td>
<td>
<p>default NULL; a two column table used to replace the sample names in abundance table; Must be two columns without column names;
The first column must be raw sample names same with those in feature table, 
the second column must be new sample names same with the rownames in sample_table; Please also see the example files.</p>
</td></tr>
<tr><td><code id="vs2meco_+3A_...">...</code></td>
<td>
<p>parameter passed to microtable$new function of microeco package, such as auto_tidy parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>microtable object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(microeco)
library(file2meco)
# use viromescan directory inside the package
dir_path &lt;- system.file("extdata", "viromescan", package="file2meco")
d1 &lt;- vs2meco(dir_path)
d1$cal_abund()
# d1$taxa_abund$Family is same with the percentage output of viromescan at 
# Family level, i.e. Family_level_results-%.txt file
d1$cal_abund(rel = FALSE)
# d1$taxa_abund$Family is same with the count output of viromescan at 
# Family level, i.e. Family_level_results-Counts.txt file

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
