<!DOCTYPE html><html><head><title>Help for package FILEST</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FILEST}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cbind_bigmatrix'><p>Combind two matrices by column for big data, internally used for</p>
parallelization</a></li>
<li><a href='#create.template.setting'><p>Create a template for a setting file of function filest</p></a></li>
<li><a href='#demo.filest'><p>Demonstration the filest function</p></a></li>
<li><a href='#filest'><p>Simulate data for multiple populations</p></a></li>
<li><a href='#FILEST-package'><p>Fine-Level Structure Simulator</p></a></li>
<li><a href='#rbind_bigmatrix'><p>Combind two matrices by row for big data, internally used for</p>
parallelization</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Fine-Level Structure Simulator</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>A population genetic simulator, which is able to generate synthetic datasets for single-nucleotide polymorphisms (SNP) for multiple populations. The genetic distances among populations can be set according to the Fixation Index (Fst) as explained in Balding and Nichols (1995) &lt;<a href="https://doi.org/10.1007%2FBF01441146">doi:10.1007/BF01441146</a>&gt;. This tool is able to simulate outlying individuals and missing SNPs can be specified. For Genome-wide association study (GWAS), disease status can be set in desired level according risk ratio.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.4)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>KRIS (&ge; 1.1.1),rARPACK,grDevices,stats,utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://gitlab.com/kris.ccp/filest/-/issues">https://gitlab.com/kris.ccp/filest/-/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://gitlab.com/kris.ccp/filest">https://gitlab.com/kris.ccp/filest</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-25 11:07:57 UTC; kris</td>
</tr>
<tr>
<td>Author:</td>
<td>Kridsadakorn Chaichoompu [aut, cre],
  Kristel Van Steen [aut],
  Fentaw Abegaz [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kridsadakorn Chaichoompu &lt;kridsadakorn@biostatgen.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-25 12:50:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='cbind_bigmatrix'>Combind two matrices by column for big data, internally used for
parallelization</h2><span id='topic+cbind_bigmatrix'></span>

<h3>Description</h3>

<p>Combind two matrices by column for big data, internally used for
parallelization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cbind_bigmatrix(a, b)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cbind_bigmatrix_+3A_a">a</code></td>
<td>
<p>The first matrix</p>
</td></tr>
<tr><td><code id="cbind_bigmatrix_+3A_b">b</code></td>
<td>
<p>The second matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The combined matrix by column
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rbind_bigmatrix">rbind_bigmatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- matrix(c(1,2,0,1,2,2,1,2,0,0,1,2,1,2,2,2),ncol=4)
Y &lt;- matrix(c(2,1,1,0,1,0,0,1,1,2,2,0,0,1,1,0),ncol=4)
Z &lt;- cbind_bigmatrix(X,Y)
print(Z)

</code></pre>

<hr>
<h2 id='create.template.setting'>Create a template for a setting file of function filest</h2><span id='topic+create.template.setting'></span>

<h3>Description</h3>

<p>Create a template for a setting file of function filest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create.template.setting(out.file, no.setting = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create.template.setting_+3A_out.file">out.file</code></td>
<td>
<p>An absolute path to a new setting file</p>
</td></tr>
<tr><td><code id="create.template.setting_+3A_no.setting">no.setting</code></td>
<td>
<p>A number of simulated settings</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An output directory if suggessfully created. Null if a setting file
can't be created.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Create 2 simulated settings

output &lt;- file.path(tempdir(),"example_setting.txt")
res &lt;- create.template.setting(out.file = output, no.setting = 2)
print(res)

</code></pre>

<hr>
<h2 id='demo.filest'>Demonstration the filest function</h2><span id='topic+demo.filest'></span>

<h3>Description</h3>

<p>This function generates the setting file and demonstrate how to use <code><a href="#topic+filest">filest</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo.filest()
</code></pre>


<h3>Value</h3>

<p>The output directory
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#To run this function, simply call demo.filest()
demo.filest()

</code></pre>

<hr>
<h2 id='filest'>Simulate data for multiple populations</h2><span id='topic+filest'></span>

<h3>Description</h3>

<p>The output files are saved to the specified directory according to <code>out</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filest(setting, out, thread = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filest_+3A_setting">setting</code></td>
<td>
<p>An absolute path to a setting file</p>
</td></tr>
<tr><td><code id="filest_+3A_out">out</code></td>
<td>
<p>An absolute path for output files</p>
</td></tr>
<tr><td><code id="filest_+3A_thread">thread</code></td>
<td>
<p>A number to specify a maximum thread to be run in parallel</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes the specific input file containing the settings
for simulations. It allows multiple settings for several simulation within
one file. The simulation-setting file must be a text file. The line started
with &quot;&ndash;&quot; indicates the parameters for simulation, and the line started with
&quot;#&quot; are comments. Empty lines are allowed in the setting file. The parameters
in the setting file are listed below:
</p>

<ul>
<li> <p><code>--setting</code>	A name of setting
</p>
</li>
<li> <p><code>--population</code> A list that indicates the numbers of population
separated by comma
</p>
</li>
<li> <p><code>--fst</code>	A list that indicates the Fst values separated by comma.
Each Fst value represents a genetic distance of that particular population
and the first population. The Fst values for the first population and the
second population should be the same values, otherwise they will be summed up
and devided by two.
</p>
</li>
<li> <p><code>--case</code> A list that indicates the ratio values of cases separated
by comma
</p>
</li>
<li> <p><code>--outlier</code>	A list that indicates the logical values (0/1) whether
that population are outliers, separated by comma
</p>
</li>
<li> <p><code>--marker</code> A number of SNPs
</p>
</li>
<li> <p><code>--replicate</code>	A number of replicates
</p>
</li>
<li> <p><code>--riskratio</code> A number of replicates
</p>
</li>
<li> <p><code>--no.case.snp</code>	A number of case SNPs
</p>
</li>
<li> <p><code>--pc</code> A logical value (TRUE/FALSE) whether PCs will be calculated.
</p>
</li>
<li> <p><code>--fulloutput</code> A logical value (TRUE/FALSE) whether all
information will be exported.
</p>
</li></ul>



<h3>Value</h3>

<p>NULL if done successfully. NA if output directory can't be created.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Check and run the demo from demo.filest()
demo.filest()

#Here is the code for demo.filest()
txt &lt;- "--setting=example1\n"
txt &lt;- paste0(txt, "--population=100,100\n")
txt &lt;- paste0(txt, "--fst=0.01,0.01\n")
txt &lt;- paste0(txt, "--case=0,0\n")
txt &lt;- paste0(txt, "--outlier=0,0\n")
txt &lt;- paste0(txt, "--marker=1000\n")
txt &lt;- paste0(txt, "--replicate=1\n")
txt &lt;- paste0(txt, "--riskratio=1\n")
txt &lt;- paste0(txt, "--no.case.snp=0\n")
txt &lt;- paste0(txt, "--pc=TRUE\n")
txt &lt;- paste0(txt, "--missing=0\n")
txt &lt;- paste0(txt, "--fulloutput=TRUE\n")

outdir &lt;- file.path(tempdir())

settingfile &lt;- file.path(outdir, "example1.txt")
fo &lt;- file(settingfile,"w")
for (i in txt){ write(i,fo)}
close(fo)

filest(setting = settingfile, out = outdir, thread = 1)

</code></pre>

<hr>
<h2 id='FILEST-package'>Fine-Level Structure Simulator</h2><span id='topic+FILEST'></span><span id='topic+FILEST-package'></span>

<h3>Description</h3>

<p>A population genetic simulator, which is able to generate synthetic datasets
for single-nucleotide polymorphisms (SNP) for multiple populations. The
genetic distances among populations can be set according to the Fixation
Index (Fst). This tool is able to simulate outlying individuals and missing
SNPs can be specified. For Genome-wide association study (GWAS), disease
status can be set in desired level according risk ratio.
</p>


<h3>Details</h3>

<p>The R package <span class="pkg">FILEST</span> requires <span class="pkg">KRIS</span> and <span class="pkg">rARPACK</span>.
</p>
<p>Here is the list of functions in the R package <span class="pkg">FILEST</span>:
</p>

<ul>
<li> <p><code><a href="#topic+cbind_bigmatrix">cbind_bigmatrix</a></code>
</p>
</li>
<li> <p><code><a href="#topic+create.template.setting">create.template.setting</a></code>
</p>
</li>
<li> <p><code><a href="#topic+demo.filest">demo.filest</a></code>
</p>
</li>
<li> <p><code><a href="#topic+filest">filest</a></code>
</p>
</li>
<li> <p><code><a href="#topic+rbind_bigmatrix">rbind_bigmatrix</a></code>
</p>
</li></ul>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Kridsadakorn Chaichoompu <a href="mailto:kridsadakorn@biostatgen.org">kridsadakorn@biostatgen.org</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Kristel Van Steen
</p>
</li>
<li><p> Fentaw Abegaz
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://gitlab.com/kris.ccp/filest">https://gitlab.com/kris.ccp/filest</a>
</p>
</li>
<li><p> Report bugs at <a href="https://gitlab.com/kris.ccp/filest/-/issues">https://gitlab.com/kris.ccp/filest/-/issues</a>
</p>
</li></ul>


<hr>
<h2 id='rbind_bigmatrix'>Combind two matrices by row for big data, internally used for
parallelization</h2><span id='topic+rbind_bigmatrix'></span>

<h3>Description</h3>

<p>Combind two matrices by row for big data, internally used for
parallelization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rbind_bigmatrix(a, b)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rbind_bigmatrix_+3A_a">a</code></td>
<td>
<p>The first matrix</p>
</td></tr>
<tr><td><code id="rbind_bigmatrix_+3A_b">b</code></td>
<td>
<p>The second matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The combined matrix by row
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cbind_bigmatrix">cbind_bigmatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- matrix(c(1,2,0,1,2,2,1,2,0,0,1,2,1,2,2,2),ncol=4)
Y &lt;- matrix(c(2,1,1,0,1,0,0,1,1,2,2,0,0,1,1,0),ncol=4)
Z &lt;- rbind_bigmatrix(X,Y)
print(Z)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
