<!DOCTYPE html><html lang="en"><head><title>Help for package findGSEP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {findGSEP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#error_minimize'><p>Minimize the Error for K-mer Frequency Fitting</p></a></li>
<li><a href='#error_minimize_raw'><p>Minimize the Error for Raw K-mer Frequency Fitting</p></a></li>
<li><a href='#error_minimize2'><p>Tuning Final Fitting for Heterozygous Genomes</p></a></li>
<li><a href='#error_minimize2_raw'><p>Tuning Final Fitting for Raw Heterozygous Genomes</p></a></li>
<li><a href='#error_minimize3'><p>Minimize the Error for Remaining K-mer Frequency</p></a></li>
<li><a href='#error_minimize3_raw'><p>Minimize the Error for Raw Remaining K-mer Frequency</p></a></li>
<li><a href='#filter_peaks'><p>Filter Peaks from K-mer Histogram</p></a></li>
<li><a href='#filter_peaks_raw'><p>Filter Peaks from Raw K-mer Histogram</p></a></li>
<li><a href='#findGSE_raw'><p>Estimate Genome Size Using Raw K-mer Frequencies</p></a></li>
<li><a href='#findGSE_sp'><p>Estimate Genome Size Using K-mer Frequencies</p></a></li>
<li><a href='#findGSEP'><p>Estimate genome size of polyploid species using k-mer frequencies.</p></a></li>
<li><a href='#get_het_pos'><p>Filter Peaks from K-mer Histogram</p></a></li>
<li><a href='#initial_count_recover'><p>Recover Initial K-mer Count</p></a></li>
<li><a href='#initial_count_recover_raw'><p>Recover Initial Raw K-mer Count</p></a></li>
<li><a href='#initialize_start_time'><p>Initialize Start Time</p></a></li>
<li><a href='#kmer_count_modify'><p>Modify K-mer Frequency Before Fitting</p></a></li>
<li><a href='#kmer_count_modify_raw'><p>Modify Raw K-mer Frequency Before Fitting</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimate Genome Size of Polyploid Species Using k-Mer
Frequencies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Laiyi Fu &lt;fulaiyi321@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides tools to estimate the genome size of polyploid species using k-mer frequencies. This package includes functions to process k-mer frequency data and perform genome size estimation by fitting k-mer frequencies with a normal distribution model. It supports handling of complex polyploid genomes and offers various options for customizing the estimation process. The basic method 'findGSE' is detailed in Sun, Hequan, et al. (2018) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtx637">doi:10.1093/bioinformatics/btx637</a>&gt;. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 4.2.0),RColorBrewer,ggplot2,grDevices</td>
</tr>
<tr>
<td>Imports:</td>
<td>pracma, scales, dplyr,</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/sperfu/findGSEP">https://github.com/sperfu/findGSEP</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/sperfu/findGSEP/issues">https://github.com/sperfu/findGSEP/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown,</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-25 14:42:01 UTC; fly</td>
</tr>
<tr>
<td>Author:</td>
<td>Laiyi Fu [aut, cre],
  Hequan Sun [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-26 17:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='error_minimize'>Minimize the Error for K-mer Frequency Fitting</h2><span id='topic+error_minimize'></span>

<h3>Description</h3>

<p>This function minimizes the error for k-mer frequency fitting by adjusting the mean, standard deviation, and scaling factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>error_minimize(
  tooptimize,
  x,
  end,
  xfit,
  xfit_left,
  xfit_right,
  d,
  min_valid_pos,
  itr
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="error_minimize_+3A_tooptimize">tooptimize</code></td>
<td>
<p>A numeric vector containing the scale factors to optimize.</p>
</td></tr>
<tr><td><code id="error_minimize_+3A_x">x</code></td>
<td>
<p>A numeric vector representing the histogram replicated from <code>d</code>.</p>
</td></tr>
<tr><td><code id="error_minimize_+3A_end">end</code></td>
<td>
<p>An integer indicating the right-side position for fitting.</p>
</td></tr>
<tr><td><code id="error_minimize_+3A_xfit">xfit</code></td>
<td>
<p>x-values to get the skew normal distribution</p>
</td></tr>
<tr><td><code id="error_minimize_+3A_xfit_left">xfit_left</code></td>
<td>
<p>A numeric value for the left-side position to calculate initial mean and standard deviation.</p>
</td></tr>
<tr><td><code id="error_minimize_+3A_xfit_right">xfit_right</code></td>
<td>
<p>A numeric value for the right-side position to calculate initial mean and standard deviation.</p>
</td></tr>
<tr><td><code id="error_minimize_+3A_d">d</code></td>
<td>
<p>A data frame representing the observed k-mer frequencies that will be fitted.</p>
</td></tr>
<tr><td><code id="error_minimize_+3A_min_valid_pos">min_valid_pos</code></td>
<td>
<p>An integer indicating the left-side position from which the observed k-mer frequencies will be fitted.</p>
</td></tr>
<tr><td><code id="error_minimize_+3A_itr">itr</code></td>
<td>
<p>An integer representing the iteration count.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value representing the minimized error.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
tooptimize &lt;- c(1, 1, 1, 1)
x &lt;- rnorm(100)
end &lt;- 100
xfit &lt;- seq(min(x), max(x), length=end)
xfit_left &lt;- min(x)
xfit_right &lt;- max(x)
d &lt;- data.frame(V1=1:100, V2=rnorm(100))
min_valid_pos &lt;- 10
itr &lt;- 100
error &lt;- error_minimize(tooptimize, x, end, xfit, xfit_left, xfit_right, d, min_valid_pos, itr)
print(error)

</code></pre>

<hr>
<h2 id='error_minimize_raw'>Minimize the Error for Raw K-mer Frequency Fitting</h2><span id='topic+error_minimize_raw'></span>

<h3>Description</h3>

<p>This function minimizes the error for raw k-mer frequency fitting by adjusting the mean, standard deviation, and scaling factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>error_minimize_raw(
  tooptimize,
  x,
  end,
  xfit,
  xfit_left,
  xfit_right,
  d,
  min_valid_pos,
  itr
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="error_minimize_raw_+3A_tooptimize">tooptimize</code></td>
<td>
<p>A numeric vector containing the scale factors to optimize.</p>
</td></tr>
<tr><td><code id="error_minimize_raw_+3A_x">x</code></td>
<td>
<p>A numeric vector representing the histogram replicated from <code>d</code>.</p>
</td></tr>
<tr><td><code id="error_minimize_raw_+3A_end">end</code></td>
<td>
<p>An integer indicating the right-side position for fitting.</p>
</td></tr>
<tr><td><code id="error_minimize_raw_+3A_xfit">xfit</code></td>
<td>
<p>x-values to get the skew normal distribution</p>
</td></tr>
<tr><td><code id="error_minimize_raw_+3A_xfit_left">xfit_left</code></td>
<td>
<p>A numeric value for the left-side position to calculate initial mean and standard deviation.</p>
</td></tr>
<tr><td><code id="error_minimize_raw_+3A_xfit_right">xfit_right</code></td>
<td>
<p>A numeric value for the right-side position to calculate initial mean and standard deviation.</p>
</td></tr>
<tr><td><code id="error_minimize_raw_+3A_d">d</code></td>
<td>
<p>A data frame representing the observed k-mer frequencies that will be fitted.</p>
</td></tr>
<tr><td><code id="error_minimize_raw_+3A_min_valid_pos">min_valid_pos</code></td>
<td>
<p>An integer indicating the left-side position from which the observed k-mer frequencies will be fitted.</p>
</td></tr>
<tr><td><code id="error_minimize_raw_+3A_itr">itr</code></td>
<td>
<p>An integer representing the iteration count.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value representing the minimized error.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tooptimize &lt;- c(1, 1, 1, 1)
x &lt;- rnorm(100)
end &lt;- 100
xfit &lt;- seq(min(x), max(x), length=end)
xfit_left &lt;- min(x)
xfit_right &lt;- max(x)
d &lt;- data.frame(V1=1:100, V2=rnorm(100))
min_valid_pos &lt;- 10
itr &lt;- 100
error &lt;- error_minimize_raw(tooptimize, x, end, xfit, xfit_left, xfit_right, d, min_valid_pos, itr)
print(error)

</code></pre>

<hr>
<h2 id='error_minimize2'>Tuning Final Fitting for Heterozygous Genomes</h2><span id='topic+error_minimize2'></span>

<h3>Description</h3>

<p>This function tunes the final fitting for heterozygous genomes by adjusting the delta values for heterozygous and homozygous regions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>error_minimize2(tooptimize, h_het, h_hom, h_target)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="error_minimize2_+3A_tooptimize">tooptimize</code></td>
<td>
<p>A numeric vector containing the scale factors to optimize.</p>
</td></tr>
<tr><td><code id="error_minimize2_+3A_h_het">h_het</code></td>
<td>
<p>A numeric vector representing the raw fitting for the heterozygous region.</p>
</td></tr>
<tr><td><code id="error_minimize2_+3A_h_hom">h_hom</code></td>
<td>
<p>A numeric vector representing the raw fitting for the homozygous region.</p>
</td></tr>
<tr><td><code id="error_minimize2_+3A_h_target">h_target</code></td>
<td>
<p>A numeric vector representing the target k-mer frequency.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value representing the minimized difference.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
tooptimize &lt;- c(0.5)
h_het &lt;- rnorm(100)
h_hom &lt;- rnorm(100)
h_target &lt;- rnorm(100)
diff &lt;- error_minimize2(tooptimize, h_het, h_hom, h_target)
print(diff)

</code></pre>

<hr>
<h2 id='error_minimize2_raw'>Tuning Final Fitting for Raw Heterozygous Genomes</h2><span id='topic+error_minimize2_raw'></span>

<h3>Description</h3>

<p>This function tunes the final fitting for raw heterozygous genomes by adjusting the delta values for heterozygous and homozygous regions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>error_minimize2_raw(tooptimize, h_het, h_hom, h_target)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="error_minimize2_raw_+3A_tooptimize">tooptimize</code></td>
<td>
<p>A numeric vector containing the scale factors to optimize.</p>
</td></tr>
<tr><td><code id="error_minimize2_raw_+3A_h_het">h_het</code></td>
<td>
<p>A numeric vector representing the raw fitting for the heterozygous region.</p>
</td></tr>
<tr><td><code id="error_minimize2_raw_+3A_h_hom">h_hom</code></td>
<td>
<p>A numeric vector representing the raw fitting for the homozygous region.</p>
</td></tr>
<tr><td><code id="error_minimize2_raw_+3A_h_target">h_target</code></td>
<td>
<p>A numeric vector representing the target k-mer frequency.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value representing the minimized difference.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tooptimize &lt;- c(0.5)
h_het &lt;- rnorm(100)
h_hom &lt;- rnorm(100)
h_target &lt;- rnorm(100)
diff &lt;- error_minimize2_raw(tooptimize, h_het, h_hom, h_target)
print(diff)

</code></pre>

<hr>
<h2 id='error_minimize3'>Minimize the Error for Remaining K-mer Frequency</h2><span id='topic+error_minimize3'></span>

<h3>Description</h3>

<p>This function minimizes the error for the remaining k-mer frequency by adjusting the scaling factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>error_minimize3(
  tooptimize,
  x,
  end,
  xfit_left,
  xfit_right,
  d,
  min_valid_pos,
  itr,
  meanfit,
  sdfit
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="error_minimize3_+3A_tooptimize">tooptimize</code></td>
<td>
<p>A numeric vector containing the scale factors to optimize.</p>
</td></tr>
<tr><td><code id="error_minimize3_+3A_x">x</code></td>
<td>
<p>A numeric vector representing the histogram.</p>
</td></tr>
<tr><td><code id="error_minimize3_+3A_end">end</code></td>
<td>
<p>An integer indicating the right-side position for fitting.</p>
</td></tr>
<tr><td><code id="error_minimize3_+3A_xfit_left">xfit_left</code></td>
<td>
<p>A numeric value for the left-side position to calculate initial mean and standard deviation.</p>
</td></tr>
<tr><td><code id="error_minimize3_+3A_xfit_right">xfit_right</code></td>
<td>
<p>A numeric value for the right-side position to calculate initial mean and standard deviation.</p>
</td></tr>
<tr><td><code id="error_minimize3_+3A_d">d</code></td>
<td>
<p>A data frame representing the observed k-mer frequencies that will be fitted.</p>
</td></tr>
<tr><td><code id="error_minimize3_+3A_min_valid_pos">min_valid_pos</code></td>
<td>
<p>An integer indicating the left-side position from which the observed k-mer frequencies will be fitted.</p>
</td></tr>
<tr><td><code id="error_minimize3_+3A_itr">itr</code></td>
<td>
<p>An integer representing the iteration count.</p>
</td></tr>
<tr><td><code id="error_minimize3_+3A_meanfit">meanfit</code></td>
<td>
<p>A numeric value representing the initial mean.</p>
</td></tr>
<tr><td><code id="error_minimize3_+3A_sdfit">sdfit</code></td>
<td>
<p>A numeric value representing the initial standard deviation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value representing the minimized error.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
tooptimize &lt;- c(1, 1, 1, 1)
x &lt;- rnorm(100)
end &lt;- 100
xfit &lt;- seq(min(x), max(x), length=end)
xfit_left &lt;- 20
xfit_right &lt;- 80
d &lt;- data.frame(V1=1:100, V2=rnorm(100))
min_valid_pos &lt;- 10
itr &lt;- 100
meanfit &lt;- 18
sdfit &lt;- 4.21
error &lt;- error_minimize3(tooptimize, x, end, xfit_left, xfit_right, d,
min_valid_pos, itr, meanfit, sdfit)
print(error)

</code></pre>

<hr>
<h2 id='error_minimize3_raw'>Minimize the Error for Raw Remaining K-mer Frequency</h2><span id='topic+error_minimize3_raw'></span>

<h3>Description</h3>

<p>This function minimizes the error for the raw remaining k-mer frequency by adjusting the scaling factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>error_minimize3_raw(
  tooptimize,
  x,
  end,
  xfit_left,
  xfit_right,
  d,
  min_valid_pos,
  itr
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="error_minimize3_raw_+3A_tooptimize">tooptimize</code></td>
<td>
<p>A numeric vector containing the scale factors to optimize.</p>
</td></tr>
<tr><td><code id="error_minimize3_raw_+3A_x">x</code></td>
<td>
<p>A numeric vector representing the histogram.</p>
</td></tr>
<tr><td><code id="error_minimize3_raw_+3A_end">end</code></td>
<td>
<p>An integer indicating the right-side position for fitting.</p>
</td></tr>
<tr><td><code id="error_minimize3_raw_+3A_xfit_left">xfit_left</code></td>
<td>
<p>A numeric value for the left-side position to calculate initial mean and standard deviation.</p>
</td></tr>
<tr><td><code id="error_minimize3_raw_+3A_xfit_right">xfit_right</code></td>
<td>
<p>A numeric value for the right-side position to calculate initial mean and standard deviation.</p>
</td></tr>
<tr><td><code id="error_minimize3_raw_+3A_d">d</code></td>
<td>
<p>A data frame representing the observed k-mer frequencies that will be fitted.</p>
</td></tr>
<tr><td><code id="error_minimize3_raw_+3A_min_valid_pos">min_valid_pos</code></td>
<td>
<p>An integer indicating the left-side position from which the observed k-mer frequencies will be fitted.</p>
</td></tr>
<tr><td><code id="error_minimize3_raw_+3A_itr">itr</code></td>
<td>
<p>An integer representing the iteration count.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value representing the minimized error.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tooptimize &lt;- c(1, 1, 1, 1)
x &lt;- rnorm(100)
end &lt;- 100
xfit &lt;- seq(min(x), max(x), length=end)
xfit_left &lt;- min(x) + 1
xfit_right &lt;- max(x) - 1
d &lt;- data.frame(V1=1:100, V2=rnorm(100))
min_valid_pos &lt;- 10
itr &lt;- 100

error &lt;- error_minimize3_raw(tooptimize, x, end, xfit_left, xfit_right, d,
min_valid_pos, itr)
print(error)

</code></pre>

<hr>
<h2 id='filter_peaks'>Filter Peaks from K-mer Histogram</h2><span id='topic+filter_peaks'></span>

<h3>Description</h3>

<p>This function filters peaks from a k-mer histogram to find a major peak with enough support information on both sides.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_peaks(peaks, histo)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_peaks_+3A_peaks">peaks</code></td>
<td>
<p>A data frame containing the peaks from the histogram.</p>
</td></tr>
<tr><td><code id="filter_peaks_+3A_histo">histo</code></td>
<td>
<p>A data frame representing the k-mer histogram.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with filtered peaks.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>peaks &lt;- data.frame(V1=1:20, V2=sample(1:10, 20, replace=TRUE))
histo &lt;- data.frame(V1=1:100, V2=sample(1:10, 100, replace=TRUE))
filtered_peaks &lt;- filter_peaks(peaks, histo)
print(filtered_peaks)
</code></pre>

<hr>
<h2 id='filter_peaks_raw'>Filter Peaks from Raw K-mer Histogram</h2><span id='topic+filter_peaks_raw'></span>

<h3>Description</h3>

<p>This function filters peaks from a raw k-mer histogram to find a major peak with enough support information on both sides.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_peaks_raw(peaks, histo)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_peaks_raw_+3A_peaks">peaks</code></td>
<td>
<p>A data frame containing the peaks from the histogram.</p>
</td></tr>
<tr><td><code id="filter_peaks_raw_+3A_histo">histo</code></td>
<td>
<p>A data frame representing the raw k-mer histogram.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with filtered peaks.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>peaks &lt;- data.frame(V1=1:20, V2=sample(1:10, 20, replace=TRUE))
histo &lt;- data.frame(V1=1:100, V2=sample(1:10, 100, replace=TRUE))
filtered_peaks &lt;- filter_peaks_raw(peaks, histo)
print(filtered_peaks)

</code></pre>

<hr>
<h2 id='findGSE_raw'>Estimate Genome Size Using Raw K-mer Frequencies</h2><span id='topic+findGSE_raw'></span>

<h3>Description</h3>

<p>This function estimates the genome size of a species using raw k-mer frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findGSE_raw(histo = "", sizek = 0, outdir = "", exp_hom = 0, species = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findGSE_raw_+3A_histo">histo</code></td>
<td>
<p>A character string specifying the path to the histogram file.</p>
</td></tr>
<tr><td><code id="findGSE_raw_+3A_sizek">sizek</code></td>
<td>
<p>An integer indicating the size of k used to generate the histogram.</p>
</td></tr>
<tr><td><code id="findGSE_raw_+3A_outdir">outdir</code></td>
<td>
<p>A character string specifying the output directory. If not specify, will use tempdir() as output directory.</p>
</td></tr>
<tr><td><code id="findGSE_raw_+3A_exp_hom">exp_hom</code></td>
<td>
<p>A numeric value representing the expected average k-mer coverage for the homozygous regions.</p>
</td></tr>
<tr><td><code id="findGSE_raw_+3A_species">species</code></td>
<td>
<p>A character string specifying the species name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the estimated genome size and other fitting parameters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

histo &lt;- "sample1.histo"
sizek &lt;- 21
outdir &lt;- tempdir()
exp_hom &lt;- 200
species &lt;- ""
fit_lists &lt;- findGSE_raw(histo, sizek,output_dir, exp_hom, species)

## End(Not run)
</code></pre>

<hr>
<h2 id='findGSE_sp'>Estimate Genome Size Using K-mer Frequencies</h2><span id='topic+findGSE_sp'></span>

<h3>Description</h3>

<p>This function estimates the genome size of a species using k-mer frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findGSE_sp(
  histo = "",
  sizek = 0,
  outdir = "",
  exp_hom = 0,
  species = "",
  ploidy_ind = 2,
  avg_cov = 0,
  left_fit_ratio = 0.835,
  meanfit_old = 0,
  sdfit_old = 0,
  scale_flag = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findGSE_sp_+3A_histo">histo</code></td>
<td>
<p>A character string specifying the path to the histogram file.</p>
</td></tr>
<tr><td><code id="findGSE_sp_+3A_sizek">sizek</code></td>
<td>
<p>An integer indicating the size of k used to generate the histogram.</p>
</td></tr>
<tr><td><code id="findGSE_sp_+3A_outdir">outdir</code></td>
<td>
<p>A character string specifying the output directory. If not specify, will use tempdir() as output directory.</p>
</td></tr>
<tr><td><code id="findGSE_sp_+3A_exp_hom">exp_hom</code></td>
<td>
<p>A numeric value representing the expected average k-mer coverage for the homozygous regions.</p>
</td></tr>
<tr><td><code id="findGSE_sp_+3A_species">species</code></td>
<td>
<p>A character string specifying the species name.</p>
</td></tr>
<tr><td><code id="findGSE_sp_+3A_ploidy_ind">ploidy_ind</code></td>
<td>
<p>An integer indicating the ploidy index (default is 2).</p>
</td></tr>
<tr><td><code id="findGSE_sp_+3A_avg_cov">avg_cov</code></td>
<td>
<p>A numeric value representing the average coverage.</p>
</td></tr>
<tr><td><code id="findGSE_sp_+3A_left_fit_ratio">left_fit_ratio</code></td>
<td>
<p>A numeric value for the left fit ratio (default is 0.835).</p>
</td></tr>
<tr><td><code id="findGSE_sp_+3A_meanfit_old">meanfit_old</code></td>
<td>
<p>A numeric value representing the previous mean fit.</p>
</td></tr>
<tr><td><code id="findGSE_sp_+3A_sdfit_old">sdfit_old</code></td>
<td>
<p>A numeric value representing the previous standard deviation fit.</p>
</td></tr>
<tr><td><code id="findGSE_sp_+3A_scale_flag">scale_flag</code></td>
<td>
<p>A logical value indicating whether to apply scaling (default is FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the estimated genome size and other fitting parameters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

histo &lt;- "sample1.histo"
sizek &lt;- 21
outdir &lt;- tempdir()
exp_hom &lt;- 200
species &lt;- ""
ploidy_ind &lt;- 2
avg_cov &lt;- 0
left_fit_ratio &lt;- 0.835
meanfit_old &lt;- 0
sdfit_old &lt;- 0
scale_flag &lt;- FALSE
fit_lists &lt;- findGSE_sp(path, samples, sizek, exp_hom, ploidy, range_left,
 range_right, xlimit, ylimit, output_dir)

## End(Not run)
</code></pre>

<hr>
<h2 id='findGSEP'>Estimate genome size of polyploid species using k-mer frequencies.</h2><span id='topic+findGSEP'></span>

<h3>Description</h3>

<p>findGSEP is a function for multiple polyploidy
genome size estimation by fitting k-mer frequencies iteratively
with a normal distribution model.
</p>
<p>To use findGSEP, one needs to prepare a histo file,
which contains two tab-separated columns.
The first column gives frequencies at which k-mers occur in reads,
while the second column gives counts of such distinct k-mers.
Parameters k and related histo file are required for any estimation.
</p>
<p>Dependencies (R library) required: pracma, fGarch, etc. - see DESCRIPTION for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findGSEP(
  path,
  samples,
  sizek,
  exp_hom,
  ploidy,
  range_left,
  range_right,
  xlimit,
  ylimit,
  output_dir = "outfile"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findGSEP_+3A_path">path</code></td>
<td>
<p>is the histo file location (mandatory).</p>
</td></tr>
<tr><td><code id="findGSEP_+3A_samples">samples</code></td>
<td>
<p>is the histo file name (mandatory)</p>
</td></tr>
<tr><td><code id="findGSEP_+3A_sizek">sizek</code></td>
<td>
<p>is the size of k used to generate the histo file (mandatory).
K is involved in calculating heterzygosity if the genome is heterozygous.</p>
</td></tr>
<tr><td><code id="findGSEP_+3A_exp_hom">exp_hom</code></td>
<td>
<p>a rough average k-mer coverage for finding the homozygous regions.
In general, one can get peaks in the k-mer frequencies file, but has to
determine which one is for the homozygous regions, and which one is for the
heterozygous regions. It is optional, however, it must be provided
if one wants to estimate size for a heterozygous genome.
VALUE for exp_hom must satisfy fp &lt; VALUE &lt; 2*fp, where fp is the freq for homozygous peak.
If not provided, 0 by default assumes the genome is homozygous.</p>
</td></tr>
<tr><td><code id="findGSEP_+3A_ploidy">ploidy</code></td>
<td>
<p>is the number of ploidy. (mandatory).</p>
</td></tr>
<tr><td><code id="findGSEP_+3A_range_left">range_left</code></td>
<td>
<p>is the left range for estimation, default is exp_hom*0.2, normally do not need
to change this. (optional).</p>
</td></tr>
<tr><td><code id="findGSEP_+3A_range_right">range_right</code></td>
<td>
<p>is the right range for estimation, default is exp_hom*0.2, normally do not need
to change this. (optional).</p>
</td></tr>
<tr><td><code id="findGSEP_+3A_xlimit">xlimit</code></td>
<td>
<p>is the x-axis range, if not given, then it will automatically calculate a proper range,
normally do not need to change this. (optional).</p>
</td></tr>
<tr><td><code id="findGSEP_+3A_ylimit">ylimit</code></td>
<td>
<p>is the y-axis range, if not given, then it will automatically calculate a proper range,
normally do not need to change this. (optional).</p>
</td></tr>
<tr><td><code id="findGSEP_+3A_output_dir">output_dir</code></td>
<td>
<p>is the path to write output files (optional).
If not specify, will use tempdir() as output directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects. The function generates PDF, PNG, and CSV files in the specified output directory.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
test_histo &lt;- system.file("extdata","example.histo",package = "findGSEP")
path &lt;- dirname(test_histo)
samples &lt;- basename(test_histo)
sizek &lt;- 21
exp_hom &lt;- 200
ploidy &lt;- 3
range_left &lt;- exp_hom*0.2
range_right &lt;- exp_hom*0.2
xlimit &lt;- -1
ylimit &lt;- -1
output_dir &lt;- tempdir()

findGSEP(path, samples, sizek, exp_hom, ploidy, range_left, range_right, xlimit, ylimit, output_dir)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_het_pos'>Filter Peaks from K-mer Histogram</h2><span id='topic+get_het_pos'></span>

<h3>Description</h3>

<p>This function filters peaks from a k-mer histogram to find a major peak with enough support information on both sides.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_het_pos(histo_data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_het_pos_+3A_histo_data">histo_data</code></td>
<td>
<p>A data frame representing the k-mer histogram.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with filtered peaks.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
x &lt;- seq(-10, 10, length.out = 100)
y &lt;- dnorm(x)
histo_data &lt;- data.frame(V1 = 1:100, V2 = y * 10)
get_het_pos(histo_data)

</code></pre>

<hr>
<h2 id='initial_count_recover'>Recover Initial K-mer Count</h2><span id='topic+initial_count_recover'></span>

<h3>Description</h3>

<p>This function recovers the initial k-mer count, making the k-mer frequency consecutive.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initial_count_recover(d0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="initial_count_recover_+3A_d0">d0</code></td>
<td>
<p>A data frame representing the initial k-mer count from software like Jellyfish.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with recovered k-mer counts.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>d0 &lt;- data.frame(V1 = c(1, 2, 4), V2 = c(100, 200, 300))
dr &lt;- initial_count_recover(d0)
</code></pre>

<hr>
<h2 id='initial_count_recover_raw'>Recover Initial Raw K-mer Count</h2><span id='topic+initial_count_recover_raw'></span>

<h3>Description</h3>

<p>This function recovers the initial raw k-mer count, making the k-mer frequency consecutive.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initial_count_recover_raw(d0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="initial_count_recover_raw_+3A_d0">d0</code></td>
<td>
<p>A data frame representing the initial raw k-mer count from software like Jellyfish.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with recovered raw k-mer counts.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>d0 &lt;- data.frame(V1 = c(1, 2, 4), V2 = c(100, 200, 300))
dr &lt;- initial_count_recover_raw(d0)

</code></pre>

<hr>
<h2 id='initialize_start_time'>Initialize Start Time</h2><span id='topic+initialize_start_time'></span>

<h3>Description</h3>

<p>This function initializes the start time for a process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initialize_start_time()
</code></pre>


<h3>Value</h3>

<p>The start time of the process.
</p>

<hr>
<h2 id='kmer_count_modify'>Modify K-mer Frequency Before Fitting</h2><span id='topic+kmer_count_modify'></span>

<h3>Description</h3>

<p>This function modifies the k-mer frequency before fitting, adjusting either the left or right part based on the peak position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kmer_count_modify(start, end, left_right, histx)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kmer_count_modify_+3A_start">start</code></td>
<td>
<p>An integer indicating the starting position (does not include the peak position).</p>
</td></tr>
<tr><td><code id="kmer_count_modify_+3A_end">end</code></td>
<td>
<p>An integer indicating the ending position (does not include the peak position).</p>
</td></tr>
<tr><td><code id="kmer_count_modify_+3A_left_right">left_right</code></td>
<td>
<p>An integer indicating whether to modify the left part (0) or the right part (1).</p>
</td></tr>
<tr><td><code id="kmer_count_modify_+3A_histx">histx</code></td>
<td>
<p>A data frame representing the k-mer histogram.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified data frame with adjusted k-mer frequencies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histx &lt;- data.frame(V1 = 1:10, V2 = c(100, 200, 300, 400, 500, 400, 300, 200, 100, 50))
histx_new &lt;- kmer_count_modify(3, 7, 0, histx)
</code></pre>

<hr>
<h2 id='kmer_count_modify_raw'>Modify Raw K-mer Frequency Before Fitting</h2><span id='topic+kmer_count_modify_raw'></span>

<h3>Description</h3>

<p>This function modifies the raw k-mer frequency before fitting, adjusting either the left or right part based on the peak position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kmer_count_modify_raw(start, end, left_right, histx)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kmer_count_modify_raw_+3A_start">start</code></td>
<td>
<p>An integer indicating the starting position (does not include the peak position).</p>
</td></tr>
<tr><td><code id="kmer_count_modify_raw_+3A_end">end</code></td>
<td>
<p>An integer indicating the ending position (does not include the peak position).</p>
</td></tr>
<tr><td><code id="kmer_count_modify_raw_+3A_left_right">left_right</code></td>
<td>
<p>An integer indicating whether to modify the left part (0) or the right part (1).</p>
</td></tr>
<tr><td><code id="kmer_count_modify_raw_+3A_histx">histx</code></td>
<td>
<p>A data frame representing the raw k-mer histogram.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified data frame with adjusted raw k-mer frequencies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>histx &lt;- data.frame(V1 = 1:10, V2 = c(100, 200, 300, 400, 500, 400, 300, 200, 100, 50))
histx_new &lt;- kmer_count_modify_raw(3, 7, 0, histx)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
