<!DOCTYPE html><html><head><title>Help for package FinePop</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FinePop}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#clip.genepop'><p>Remove designated markers from a GENEPOP file.</p></a></li>
<li><a href='#DJ'><p>Jost's D</p></a></li>
<li><a href='#EBDJ'><p>Empirical Bayes estimator of Jost's D</p></a></li>
<li><a href='#EBFST'><p>Empirical Bayes estimator of Fst.</p></a></li>
<li><a href='#EBGstH'><p>Empirical Bayes estimator of Hedrick's G'st</p></a></li>
<li><a href='#FinePop-package'><p>Fine-Scale Population Analysis</p></a></li>
<li><a href='#FstBoot'><p>Bootstrap sampler of Fst</p></a></li>
<li><a href='#FstEnv'><p>Regression analysis of environmental factors on genetic differentiation</p></a></li>
<li><a href='#GstH'><p>Hedrick's G'st</p></a></li>
<li><a href='#GstN'><p>Nei's Gst.</p></a></li>
<li><a href='#GstNC'><p>Nei and Chesser's Gst</p></a></li>
<li><a href='#herring'><p>An example dataset of Atlantic herring.</p></a></li>
<li><a href='#jsmackerel'><p>An example dataset of Japanese Spanich mackerel in GENEPOP and frequency format.</p></a></li>
<li><a href='#read.frequency'><p>Create an allele (haplotype) frequency data object of populations from a frequency format file.</p></a></li>
<li><a href='#read.genepop'><p>Create a genotype data object of populations from a GENEPOP format file.</p></a></li>
<li><a href='#thetaWC.pair'><p>Weir and Cockerham's theta adapted for pairwise Fst.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Fine-Scale Population Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.5.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-15</td>
</tr>
<tr>
<td>Author:</td>
<td>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Reiichiro Nakamichi &lt;nakamichi_reiichiro33@fra.go.jp&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Statistical tool set for population genetics. The package provides following functions: 1) empirical Bayes estimator of Fst and other measures of genetic differentiation, 2) regression analysis of environmental effects on genetic differentiation using bootstrap method, 3) interfaces to read and manipulate 'GENEPOP' format data files and allele/haplotype frequency format files.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2.0)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ape</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-15 09:41:07 UTC; nakamichi</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-15 10:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='clip.genepop'>Remove designated markers from a GENEPOP file.</h2><span id='topic+clip.genepop'></span>

<h3>Description</h3>

<p>This function reads a GENEPOP file (Rousset 2008), remove designated markers, and write a GENEPOP file of clipped data. The user can directly designate the names of the markers to be removed. The user also can set the filtering threshold of major allele frequency. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clip.genepop(infile, outfile, remove.list = NULL, major.af = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clip.genepop_+3A_infile">infile</code></td>
<td>
<p>A character value specifying the name of the GENEPOP file to be clipped.</p>
</td></tr>
<tr><td><code id="clip.genepop_+3A_outfile">outfile</code></td>
<td>
<p>A character value specifying the name of the clipped GENEPOP file.</p>
</td></tr>
<tr><td><code id="clip.genepop_+3A_remove.list">remove.list</code></td>
<td>
<p>A character value or vector specifying the names of the markers to be removed. The names must be included in the target GENEPOP file.</p>
</td></tr>
<tr><td><code id="clip.genepop_+3A_major.af">major.af</code></td>
<td>
<p>A numeric value specifying the threshold of major allele frequency for marker removal. Markers with major allele frequencies higher than this value will be removed. This value must be between 0 and 1.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi</p>


<h3>References</h3>

<p>Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.genepop.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.genepop.file, sep="\n")

# Remove markers designated by their names
clipped_by_name.jsm.genepop.file &lt;- tempfile()
clip.genepop(infile=jsm.genepop.file,
             outfile=clipped_by_name.jsm.genepop.file,
             remove.list=c("Sni21","Sni26"))

# Remove markers with high major allele frequencies (in this example, &gt; 0.5)
clipped_by_af.jsm.genepop.file &lt;- tempfile()
clip.genepop(infile=jsm.genepop.file,
             outfile=clipped_by_af.jsm.genepop.file,
             major.af=0.5)

# Remove markers both by their names and by major allele frequencies
clipped_by_both.jsm.genepop.file &lt;- tempfile()
clip.genepop(infile=jsm.genepop.file,
             outfile=clipped_by_both.jsm.genepop.file,
             remove.list=c("Sni21","Sni26"), major.af=0.5)

# See four text files in temporary directory.
#  jsm.genepop.file                 : original data of five markers
#  clipped_by_name.jsm.genepop.file : clipped data by marker names
#  clipped_by_af.jsm.genepop.file   : clipped data by allele frequency
#  clipped_by_both.jsm.genepop.file : clipped data by both names and frequency
</code></pre>

<hr>
<h2 id='DJ'>Jost's D</h2><span id='topic+DJ'></span>

<h3>Description</h3>

<p>This function estimates pairwise D (Jost 2008) among subpopulations from a GENEPOP data object (Rousset 2008). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DJ(popdata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DJ_+3A_popdata">popdata</code></td>
<td>
<p>Population data object created by read.genepop function from a GENEPOP file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of estimated pairwise Jost's D.
</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Jost L (2008) Gst and its relatives do not measure differentiation. <em>Molecular Ecology</em>, 17, 4015-4026.<br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.genepop">read.genepop</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.genepop(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# Jost's D estimation
result.DJ &lt;- DJ(popdata)
print(as.dist(result.DJ))
</code></pre>

<hr>
<h2 id='EBDJ'>Empirical Bayes estimator of Jost's D</h2><span id='topic+EBDJ'></span>

<h3>Description</h3>

<p>This function estimates pairwise D (Jost 2008) among subpopulations using empirical Bayes method (Kitada et al. 2007). This function accepts two types of data object, GENEPOP data (Rousset 2008) and allele (haplotype) frequency data (Kitada et al. 2007). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBDJ(popdata, num.iter=100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBDJ_+3A_popdata">popdata</code></td>
<td>
<p>Genotype data object of populations created by read.genepop function from a GENEPOP file. Allele (haplotype) frequency data object created by read.frequency function from a frequency format file also is acceptable.</p>
</td></tr>
<tr><td><code id="EBDJ_+3A_num.iter">num.iter</code></td>
<td>
<p>A positive integer value specifying the number of iterations in empirical Bayes simulation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Frequency format file is a plain text file containing allele (haplotype) count data. This format is mainly for mitochondrial DNA (mtDNA) haplotype frequency data, however nuclear DNA (nDNA) data also is applicable. In the data object created by read.frequency function, &quot;number of samples&quot; means haplotype count. Therefore, it equals the number of individuals in mtDNA data, however it is the twice of the number of individuals in nDNA data. First part of the frequency format file is the number of subpopulations, second part is the number of loci, and latter parts are [<em>population</em> x <em>allele</em>] matrices of the observed allele (haplotype) counts at each locus. Two examples of frequency format files are attached in this package. See <code><a href="#topic+jsmackerel">jsmackerel</a></code>.
</p>


<h3>Value</h3>

<p>Matrix of estimated pairwise Jost's D.
</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Jost L (2008) Gst and its relatives do not measure differentiation. <em>Molecular Ecology</em>, 17, 4015-4026.<br /><br />
Kitada S, Kitakado T, Kishino H (2007) Empirical Bayes inference of pairwise FST and its distribution in the genome. <em>Genetics</em>, 177, 861-873.<br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.genepop">read.genepop</a></code>, <code><a href="#topic+read.frequency">read.frequency</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.genepop(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# Jost's D estimation
result.EBDJ &lt;- EBDJ(popdata)
print(as.dist(result.EBDJ))
</code></pre>

<hr>
<h2 id='EBFST'>Empirical Bayes estimator of Fst.</h2><span id='topic+EBFST'></span>

<h3>Description</h3>

<p>This function estimates global/pairwise Fst among subpopulations using empirical Bayes method (Kitada et al. 2007, 2017). Preciseness of estimated pairwise Fst is evaluated by bootstrap method. This function accepts two types of data object, GENEPOP data (Rousset 2008) and allele (haplotype) frequency data (Kitada et al. 2007). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBFST(popdata, num.iter = 100, locus = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBFST_+3A_popdata">popdata</code></td>
<td>
<p>Genotype data object of populations created by read.genepop function from a GENEPOP file. Allele (haplotype) frequency data object created by read.frequency function from a frequency format file also is acceptable.</p>
</td></tr>
<tr><td><code id="EBFST_+3A_num.iter">num.iter</code></td>
<td>
<p>A positive integer value specifying the number of iterations in empirical Bayes simulation.</p>
</td></tr>
<tr><td><code id="EBFST_+3A_locus">locus</code></td>
<td>
<p>A Logical argument indicating whether locus-specific Fst values should be calculated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Frequency format file is a plain text file containing allele (haplotype) count data. This format is mainly for mitochondrial DNA (mtDNA) haplotype frequency data, however nuclear DNA (nDNA) data also is applicable. In the data object created by read.frequency function, &quot;number of samples&quot; means haplotype count. Therefore, it equals the number of individuals in mtDNA data, however it is the twice of the number of individuals in nDNA data. First part of the frequency format file is the number of subpopulations, second part is the number of loci, and latter parts are [<em>population</em> x <em>allele</em>] matrices of the observed allele (haplotype) counts at each locus. Two examples of frequency format files are attached in this package. See <code><a href="#topic+jsmackerel">jsmackerel</a></code>.
</p>


<h3>Value</h3>

<p>global:
</p>
<table>
<tr><td><code>theta</code></td>
<td>
<p>Estimated gene flow rate.</p>
</td></tr>
<tr><td><code>fst</code></td>
<td>
<p>Estimated genome-wide global Fst.</p>
</td></tr>
<tr><td><code>fst.locus</code></td>
<td>
<p>Estimated locus-specific global Fst. (If locus = TRUE)</p>
</td></tr>
</table>
<p>pairwise:
</p>
<table>
<tr><td><code>fst</code></td>
<td>
<p>Estimated genome-wide pairwise Fst.</p>
</td></tr>
<tr><td><code>fst.boot</code></td>
<td>
<p>Bootstrap mean of estimated Fst.</p>
</td></tr>
<tr><td><code>fst.boot.sd</code></td>
<td>
<p>Bootstrap standard deviation of estimated Fst.</p>
</td></tr>
<tr><td><code>fst.locus</code></td>
<td>
<p>Estimated locus-specific pairwise Fst. (If locus = TRUE)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Kitada S, Kitakado T, Kishino H (2007) Empirical Bayes inference of pairwise FST and its distribution in the genome. <em>Genetics</em>, 177, 861-873.<br /><br />
Kitada S, Nakamichi R, Kishino H (2017) The empirical Bayes estimators of fine-scale population structure in high gene flow species. <em>Mol. Ecol. Resources</em>, DOI: 10.1111/1755-0998.12663 <br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.genepop">read.genepop</a></code>, <code><a href="#topic+read.frequency">read.frequency</a></code>,
<code><a href="stats.html#topic+as.dist">as.dist</a></code>, <code><a href="stats.html#topic+as.dendrogram">as.dendrogram</a></code>,
<code><a href="stats.html#topic+hclust">hclust</a></code>, <code><a href="stats.html#topic+cmdscale">cmdscale</a></code>, <code><a href="ape.html#topic+nj">nj</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.genepop(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# Fst estimation
result.eb &lt;- EBFST(popdata)
ebfst &lt;- result.eb$pairwise$fst
ebfst.d &lt;- as.dist(ebfst)
print(ebfst.d)

# dendrogram
ebfst.hc &lt;- hclust(ebfst.d,method="average")
plot(as.dendrogram(ebfst.hc), xlab="",ylab="",main="", las=1)

# MDS plot
mds &lt;- cmdscale(ebfst.d)
plot(mds, type="n", xlab="",ylab="")
text(mds[,1],mds[,2], popdata$pop_names)

# NJ tree
library(ape)
ebfst.nj &lt;- nj(ebfst.d)
plot(ebfst.nj,type="u",main="",sub="")
</code></pre>

<hr>
<h2 id='EBGstH'>Empirical Bayes estimator of Hedrick's G'st</h2><span id='topic+EBGstH'></span>

<h3>Description</h3>

<p>This function estimates pairwise G'st (Hedrick 2005) among subpopulations using empirical Bayes method (Kitada et al. 2007). This function accepts two types of data object, GENEPOP data (Rousset 2008) and allele (haplotype) frequency data (Kitada et al. 2007). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBGstH(popdata, num.iter = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBGstH_+3A_popdata">popdata</code></td>
<td>
<p>Genotype data object of populations created by read.genepop function from a GENEPOP file. Allele (haplotype) frequency data object created by read.frequency function from a frequency format file also is acceptable.</p>
</td></tr>
<tr><td><code id="EBGstH_+3A_num.iter">num.iter</code></td>
<td>
<p>A positive integer value specifying the number of iterations in empirical Bayes simulation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Frequency format file is a plain text file containing allele (haplotype) count data. This format is mainly for mitochondrial DNA (mtDNA) haplotype frequency data, however nuclear DNA (nDNA) data also is applicable. In the data object created by read.frequency function, &quot;number of samples&quot; means haplotype count. Therefore, it equals the number of individuals in mtDNA data, however it is the twice of the number of individuals in nDNA data. First part of the frequency format file is the number of subpopulations, second part is the number of loci, and latter parts are [<em>population</em> x <em>allele</em>] matrices of the observed allele (haplotype) counts at each locus. Two examples of frequency format files are attached in this package. See <code><a href="#topic+jsmackerel">jsmackerel</a></code>.
</p>


<h3>Value</h3>

<p>Matrix of estimated pairwise Hedrick's G'st.
</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Hedrick P (2005) A standardized genetic differentiation measure. <em>Evolution</em>, 59, 1633-1638.<br /><br />
Kitada S, Kitakado T, Kishino H (2007) Empirical Bayes inference of pairwise FST and its distribution in the genome. <em>Genetics</em>, 177, 861-873.<br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.genepop">read.genepop</a></code>, <code><a href="#topic+read.frequency">read.frequency</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.genepop(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# Hedrick's G'st estimation
result.EBGstH &lt;- EBGstH(popdata)
print(as.dist(result.EBGstH))
</code></pre>

<hr>
<h2 id='FinePop-package'>Fine-Scale Population Analysis</h2><span id='topic+FinePop-package'></span>

<h3>Description</h3>

<p>Statistical tool set for population genetics. The package provides following functions: 1) empirical Bayes estimator of Fst and other measures of genetic differentiation, 2) regression analysis of environmental effects on genetic differentiation using bootstrap method, 3) interfaces to read and manipulate 'GENEPOP' format data files and allele/haplotype frequency format files.</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada<br />
Maintainer: Reiichiro Nakamichi &lt;nakamichi_reiichiro33@fra.go.jp&gt;
</p>


<h3>References</h3>

<p>Kitada S, Kitakado T, Kishino H (2007) Empirical Bayes inference of pairwise FST and its distribution in the genome. <em>Genetics</em>, 177, 861-873.<br /><br />
Kitada S, Nakamichi R, Kishino H (2017) The empirical Bayes estimators of fine-scale population structure in high gene flow species. <em>Mol. Ecol. Resources</em>, DOI: 10.1111/1755-0998.12663 <br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.<br /><br />
</p>

<hr>
<h2 id='FstBoot'>Bootstrap sampler of Fst</h2><span id='topic+FstBoot'></span>

<h3>Description</h3>

<p>This function provides bootstrapped estimators of Fst to evaluate the environmental effects on the genetic diversity. See Details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FstBoot(popdata, fst.method = "EBFST", bsrep = 100, log.bs = F, locus = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FstBoot_+3A_popdata">popdata</code></td>
<td>
<p>Genotype data object of populations created by read.genepop function from a GENEPOP file.</p>
</td></tr>
<tr><td><code id="FstBoot_+3A_fst.method">fst.method</code></td>
<td>
<p>A character value specifying the Fst estimation method to be used. Currently, &quot;EBFST&quot;, &quot;EBGstH&quot;, &quot;EBDJ&quot;, &quot;GstN&quot;, &quot;GstNC&quot;, &quot;GstH&quot;, &quot;DJ&quot; and &quot;thetaWC.pair&quot; are available.</p>
</td></tr>
<tr><td><code id="FstBoot_+3A_bsrep">bsrep</code></td>
<td>
<p>A positive integer value specifying the trial times of bootstrapping.</p>
</td></tr>
<tr><td><code id="FstBoot_+3A_log.bs">log.bs</code></td>
<td>
<p>A logical value specifying whether the bootstrapped data of each trial should be saved. If TRUE, GENEPOP format files named &quot;gtdata_bsXXX.txt&quot; (XXX=trial number) are saved in the working directory.</p>
</td></tr>
<tr><td><code id="FstBoot_+3A_locus">locus</code></td>
<td>
<p>A Logical argument indicating whether locus-specific Fst values should be calculated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>FinePop provides a method for regression analyses of the pairwise Fst values against geographical distance and the differences to examine the effect of environmental variables on population differentiation (Kitada et. al 2017). First, <code><a href="#topic+FstBoot">FstBoot</a></code> function resamples locations with replacement, and then, we also resample the member individuals with replacement from the sampled populations. It calculates pairwise Fst for each bootstrap sample. Second, <code><a href="#topic+FstEnv">FstEnv</a></code> function estimates regression coefficients (<code><a href="stats.html#topic+lm">lm</a></code> function) of for the Fst values for each iteration. It then computes the standard deviation of the regression coefficients, Z-values and P-values of each regression coefficient. All possible model combinations for the environmental explanatory variables were examined, including their interactions. The best fit model with the minimum information criterion (TIC, Takeuchi 1976, Burnham &amp; Anderson 2002) is selected. Performance for detecting environmental effects on population structuring is evaluated by the R2 value.
</p>


<h3>Value</h3>

<table>
<tr><td><code>bs.pop.list</code></td>
<td>
<p>List of subpopulations in bootstrapped data</p>
</td></tr>
<tr><td><code>bs.fst.list</code></td>
<td>
<p>List of genome-wide pairwise Fst matrices for bootstrapped data.</p>
</td></tr>
<tr><td><code>org.fst</code></td>
<td>
<p>Genome-wide pairwise Fst matrix for original data.</p>
</td></tr>
<tr><td><code>bs.fst.list.locus</code></td>
<td>
<p>List of locus-specific pairwise Fst matrices for bootstrapped data. (If locus = TRUE)</p>
</td></tr>
<tr><td><code>org.fst.locus</code></td>
<td>
<p>Locus-specific pairwise Fst matrix for original data. (If locus = TRUE)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Burnham KP, Anderson DR (2002) <em>Model Selection and Multimodel Inference: A Practical Information-Theoretic Approach.</em> Springer, New York.<br /><br />
Kitada S, Nakamichi R, Kishino H (2017) The empirical Bayes estimators of fine-scale population structure in high gene flow species. <em>Mol. Ecol. Resources</em>, DOI: 10.1111/1755-0998.12663. <br /><br />
Takeuchi K (1976) Distribution of information statistics and criteria for adequacy of Models. <em>Mathematical Science</em>, 153, 12-18 (in Japanese).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.genepop">read.genepop</a></code>, <code><a href="#topic+FstEnv">FstEnv</a></code>,
<code><a href="#topic+EBFST">EBFST</a></code>, <code><a href="#topic+EBGstH">EBGstH</a></code>,<code><a href="#topic+EBDJ">EBDJ</a></code>,
<code><a href="#topic+GstN">GstN</a></code>, <code><a href="#topic+GstNC">GstNC</a></code>, <code><a href="#topic+GstH">GstH</a></code>,
<code><a href="#topic+DJ">DJ</a></code>, <code><a href="#topic+thetaWC.pair">thetaWC.pair</a></code>,
<code><a href="#topic+herring">herring</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of genotypic and environmental dataset
data(herring)

# Data bootstrapping and Fst estimation
# fstbs &lt;- FstBoot(herring$popdata)

# Effects of environmental factors on genetic differentiation
# fstenv &lt;- FstEnv(fstbs, herring$environment, herring$distance)

# Since these calculations are too heavy, pre-caluculated results are included in this dataset.
fstbs &lt;- herring$fst.bootstrap
fstenv &lt;- herring$fst.env
summary(fstenv)
</code></pre>

<hr>
<h2 id='FstEnv'>Regression analysis of environmental factors on genetic differentiation</h2><span id='topic+FstEnv'></span>

<h3>Description</h3>

<p>This function provides linear regression analysis for Fst against environmental factors to evaluate the environmental effects on the genetic diversity. See Details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FstEnv(fst.bs, environment, distance = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FstEnv_+3A_fst.bs">fst.bs</code></td>
<td>
<p>Bootstrap samples of pairwise Fst matrices provided by FstBoot function from a GENEPOP file.</p>
</td></tr>
<tr><td><code id="FstEnv_+3A_environment">environment</code></td>
<td>
<p>A table object of environmental factors. Rows are subpopulations, and columns are environmental factors. Names of subpopulations (row names) must be same as those in fst.bs.</p>
</td></tr>
<tr><td><code id="FstEnv_+3A_distance">distance</code></td>
<td>
<p>A square matrix of distance among subpopulations (omittable). Names of subpopulations (row/column names) must be same as those in fst.bs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>FinePop provides a method for regression analyses of the pairwise Fst values against geographical distance and the differences to examine the effect of environmental variables on population differentiation (Kitada et. al 2017). First, <code><a href="#topic+FstBoot">FstBoot</a></code> function resamples locations with replacement, and then, we also resample the member individuals with replacement from the sampled populations. It calculates pairwise Fst for each bootstrap sample. Second, <code><a href="#topic+FstEnv">FstEnv</a></code> function estimates regression coefficients (<code><a href="stats.html#topic+lm">lm</a></code> function) of for the Fst values for each iteration. It then computes the standard deviation of the regression coefficients, Z-values and P-values of each regression coefficient. All possible model combinations for the environmental explanatory variables were examined, including their interactions. The best fit model with the minimum Takeuchi information criterion (TIC, Takeuchi 1976, Burnham &amp; Anderson 2002) is selected. Performance for detecting environmental effects on population structuring is evaluated by the R2 value.
</p>


<h3>Value</h3>

<p>A list of regression result:
</p>
<table>
<tr><td><code>model</code></td>
<td>
<p>Evaluated model of environmental factors on genetic differentiation.</p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p>Estimated coefficient, standard deviation, Z value and p value of each factor.</p>
</td></tr>
<tr><td><code>TIC</code></td>
<td>
<p>Takeuchi information criterion.</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>coefficient of determination.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Burnham KP, Anderson DR (2002) <em>Model Selection and Multimodel Inference: A Practical Information-Theoretic Approach.</em> Springer, New York.<br /><br />
Kitada S, Nakamichi R, Kishino H (2017) The empirical Bayes estimators of fine-scale population structure in high gene flow species. <em>Mol. Ecol. Resources</em>, DOI: 10.1111/1755-0998.12663. <br /><br />
Takeuchi K (1976) Distribution of information statistics and criteria for adequacy of Models. <em>Mathematical Science</em>, 153, 12-18 (in Japanese).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FstBoot">FstBoot</a></code>, <code><a href="#topic+read.genepop">read.genepop</a></code>,
<code><a href="stats.html#topic+lm">lm</a></code>, <code><a href="#topic+herring">herring</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of genotypic and environmental dataset
data(herring)

# Data bootstrapping and Fst estimation
# fstbs &lt;- FstBoot(herring$popdata)

# Effects of environmental factors on genetic differentiation
# fstenv &lt;- FstEnv(fstbs, herring$environment, herring$distance)

# Since these calculations are too heavy, pre-calculated results are included in this dataset.
fstbs &lt;- herring$fst.bootstrap
fstenv &lt;- herring$fst.env
summary(fstenv)
</code></pre>

<hr>
<h2 id='GstH'>Hedrick's G'st</h2><span id='topic+GstH'></span>

<h3>Description</h3>

<p>This function estimates pairwise G'st (Hedrick 2005) among subpopulations from a GENEPOP data object (Rousset 2008). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GstH(popdata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GstH_+3A_popdata">popdata</code></td>
<td>
<p>Population data object created by read.genepop function from a GENEPOP file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of estimated pairwise Hedrick's G'st.
</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Hedrick P (2005) A standardized genetic differentiation measure. <em>Evolution</em>, 59, 1633-1638.<br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.genepop">read.genepop</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.genepop(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# Hedrick's G'st estimation
result.GstH &lt;- GstH(popdata)
print(as.dist(result.GstH))
</code></pre>

<hr>
<h2 id='GstN'>Nei's Gst.</h2><span id='topic+GstN'></span>

<h3>Description</h3>

<p>This function estimates pairwise Gst among subpopulations (Nei 1973) from a GENEPOP data object (Rousset 2008). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GstN(popdata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GstN_+3A_popdata">popdata</code></td>
<td>
<p>Population data object created by read.genepop function from a GENEPOP file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of estimated pairwise Gst.
</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Nei M (1973) Analysis of Gene Diversity in Subdivided Populations. <em>Proc. Nat. Acad. Sci.</em>, 70, 3321-3323.<br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.genepop">read.genepop</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.genepop(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# Gst estimation
result.gstN &lt;- GstN(popdata)
print(as.dist(result.gstN))
</code></pre>

<hr>
<h2 id='GstNC'>Nei and Chesser's Gst</h2><span id='topic+GstNC'></span>

<h3>Description</h3>

<p>This function estimates pairwise Gst among subpopulations (Nei&amp;Chesser 1983) from a GENEPOP data object (Rousset 2008). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GstNC(popdata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GstNC_+3A_popdata">popdata</code></td>
<td>
<p>Population data object created by read.genepop function from a GENEPOP file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of estimated pairwise Gst.
</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Nei M, Chesser RK (1983) Estimation of fixation indices and gene diversity. <em>Annals of Human Genetics</em>, 47, 253-259.<br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.genepop">read.genepop</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.genepop(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# Gst estimation
result.gstNC &lt;- GstNC(popdata)
print(as.dist(result.gstNC))
</code></pre>

<hr>
<h2 id='herring'>An example dataset of Atlantic herring.</h2><span id='topic+herring'></span>

<h3>Description</h3>

<p>An example of a genetic data for Atlantic herring population (Limborg et al. 2012). It contains genotypic information of 281 SNPs from 18 subpopulations of 607 individuals. GENEPOP format (Rousset 2008) text file is available. Subpopulation names, environmental factors (temperature and salinity) at each subpopulation and geographic distance (shortest ocean path) among subpopulations also are attached.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("herring")</code></pre>


<h3>Format</h3>

<p>$ genepop       : Genotypic information of 281 SNPs in GENEPOP format text data.<br />
$ popname       : Names of subpopulations.<br />
$ environment   : Table of temperature and salinity at each subpopulation.<br />
$ distance      : Matrix of geographic distance (shortest ocean path) among subpopulations.<br />
$ popdata       : Genotype data object of this herring data created by <code><a href="#topic+read.genepop">read.genepop</a></code> function.<br />
$ fst.bootstrap : Bootstrapped Fst estimations of this herring data generated by <code><a href="#topic+FstBoot">FstBoot</a></code> function.<br />
$ fst.env       : Regression analysis of environmental effects on genetic differentiation of this herring data generated by <code><a href="#topic+FstEnv">FstEnv</a></code> function.
</p>


<h3>References</h3>

<p>Limborg MT, Helyar SJ, de Bruyn M et al. (2012) Environmental selection on transcriptome-derived SNPs in a high gene flow marine fish, the Atlantic herring (Clupea harengus). <em>Molecular Ecology</em>, 21, 3686-3703. <br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.genepop">read.genepop</a></code>, <code><a href="#topic+FstBoot">FstBoot</a></code>, <code><a href="#topic+FstEnv">FstEnv</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(herring)
ah.genepop.file &lt;- tempfile()
ah.popname.file &lt;- tempfile()
cat(herring$genepop, file=ah.genepop.file, sep="\n")
cat(herring$popname, file=ah.popname.file, sep=" ")

# See two text files in temporary directory.
#  ah.genepop.file  : GENEPOP format file of 281SNPs in 18 subpopulations
#  ah.popname.file  : plain text file of subpopulation names

print(herring$environment)

# herring$popdata = read.genepop(genepop="AH_genepop.txt", popname="AH_popname.txt")
# herring$fst.bootstrap = FstBoot(herring$popdata)
# herring$fst.env = FstEnv(herring$fst.bootstrap, herring$environment, herring$distance)
</code></pre>

<hr>
<h2 id='jsmackerel'>An example dataset of Japanese Spanich mackerel in GENEPOP and frequency format.</h2><span id='topic+jsmackerel'></span>

<h3>Description</h3>

<p>An example of a genetic data for a Japanese Spanish mackerel population (Nakajima et al. 2014). It contains genotypic information of 5 microsatellite markers and mtDNA D-loop region from 8 subpopulations of 715 individuals. GENEPOP format (Rousset 2008) and frequency format (Kitada et al. 2007) text files are available. Name list of subpopulations also is attached.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("jsmackerel")</code></pre>


<h3>Format</h3>

<p>$ MS.genepop: Genotypic information of 5 microsatellites in GENEPOP format text data.<br />
$ MS.freq: Allele frequency of 5 microsatellites  in frequency format text data.<br />
$ mtDNA.freq: Haplotype frequency of mtDNA D-loop region in frequency format text data.<br />
$ popname: Names of subpopulations.
</p>


<h3>Details</h3>

<p>Frequency format file is a plain text file containing allele (haplotype) count data. This format is mainly for mitochondrial DNA (mtDNA) haplotype frequency data, however nuclear DNA (nDNA) data also is applicable. In the data object created by read.frequency function, &quot;number of samples&quot; means haplotype count. Therefore, it equals the number of individuals in mtDNA data, however it is the twice of the number of individuals in nDNA data. First part of the frequency format file is the number of subpopulations, second part is the number of loci, and latter parts are [<em>population</em> x <em>allele</em>] matrices of the observed allele (haplotype) counts at each locus. Two examples of frequency format files are attached in this package.
</p>


<h3>References</h3>

<p>Nakajima K et al. (2014) Genetic effects of marine stock enhancement: a case study based on the highly piscivorous Japanese Spanish mackerel. <em>Canadian Journal of Fisheries and Aquatic Sciences</em>, 71, 301-314.<br /><br />
Kitada S, Kitakado T, Kishino H (2007) Empirical Bayes inference of pairwise FST and its distribution in the genome. <em>Genetics</em>, 177, 861-873.<br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.genepop">read.genepop</a></code>, <code><a href="#topic+read.frequency">read.frequency</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jsmackerel)

jsm.ms.genepop.file &lt;- tempfile()
jsm.ms.freq.file &lt;- tempfile()
jsm.mt.freq.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$MS.freq, file=jsm.ms.freq.file, sep="\n")
cat(jsmackerel$mtDNA.freq, file=jsm.mt.freq.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# See four text files in your working directory.
#  jsm.ms.genepop.file  : GENEPOP format file of microsatellite data
#  jsm.ms.freq.file     : frequency format file of microsatellite data
#  jsm.mt.freq.file  : frequency format file of mtDNA D-loop region data
#  jsm.popname.file     : plain text file of subpopulation names

</code></pre>

<hr>
<h2 id='read.frequency'>Create an allele (haplotype) frequency data object of populations from a frequency format file.</h2><span id='topic+read.frequency'></span>

<h3>Description</h3>

<p>This function reads a frequency format file (Kitada et al. 2007) and parse it into an R data object. This data object provides a summary of allele (haplotype) frequency in each population and marker status. This data object is used by EBFST function of this package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.frequency(frequency, popname = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.frequency_+3A_frequency">frequency</code></td>
<td>
<p>A character value specifying the name of the frequency format file to be analyzed.</p>
</td></tr>
<tr><td><code id="read.frequency_+3A_popname">popname</code></td>
<td>
<p>A character value specifying the name of the plain text file containing the names of subpopulations to be analyzed. This text file must not contain other than subpopulation names. The names must be separated by spaces, tabs or line breaks. If this argument is omitted, serial numbers will be assigned as subpopulation names.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Frequency format file is a plain text file containing allele (haplotype) count data. This format is mainly for a mitochondrial DNA (mtDNA) haplotype frequency data, however nuclear DNA (nDNA) data also is applicable. In the data object created by read.frequency function, &quot;number of samples&quot; means haplotype count. Therefore, it equals the number of individuals in mtDNA data, however it is the twice of the number of individuals in nDNA data. First part of the frequency format file is the number of subpopulations, second part is the number of loci, and latter parts are [<em>population</em> x <em>allele</em>] matrices of the observed allele (haplotype) counts at each locus. Two examples of frequency format files are attached in this package. See <code><a href="#topic+jsmackerel">jsmackerel</a></code>.
</p>


<h3>Value</h3>

<table>
<tr><td><code>npops</code></td>
<td>
<p>Number of subpopulations.</p>
</td></tr>
<tr><td><code>pop_sizes</code></td>
<td>
<p>Number of samples in each subpopulation.</p>
</td></tr>
<tr><td><code>pop_names</code></td>
<td>
<p>Names of subpopulations.</p>
</td></tr>
<tr><td><code>nloci</code></td>
<td>
<p>Number of loci.</p>
</td></tr>
<tr><td><code>loci_names</code></td>
<td>
<p>Names of loci.</p>
</td></tr>
<tr><td><code>all_alleles</code></td>
<td>
<p>A list of alleles (haplotypes) at each locus.</p>
</td></tr>
<tr><td><code>nalleles</code></td>
<td>
<p>Number of alleles (haplotypes) at each locus.</p>
</td></tr>
<tr><td><code>indtyp</code></td>
<td>
<p>Number of genotyped samples in each subpopulation at each locus.</p>
</td></tr>
<tr><td><code>obs_allele_num</code></td>
<td>
<p>Observed allele (haplotype) counts at each locus in each subpopulation.</p>
</td></tr>
<tr><td><code>allele_freq</code></td>
<td>
<p>Observed allele (haplotype) frequencies at each locus in each subpopulation.</p>
</td></tr>
<tr><td><code>call_rate</code></td>
<td>
<p>Rate of genotyped samples at each locus.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Kitada S, Kitakado T, Kishino H (2007) Empirical Bayes inference of pairwise FST and its distribution in the genome. <em>Genetics</em>, 177, 861-873.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+jsmackerel">jsmackerel</a></code>, <code><a href="#topic+EBFST">EBFST</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of frequency format file
data(jsmackerel)
jsm.mt.freq.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$mtDNA.freq, file=jsm.mt.freq.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Read frequency format file with subpopulation names
# Prepare your frequency format file and population name file in the working directory
# Replace "jsm.mt.freq.file" and "jsm.popname.file" by your file names.
popdata.mt &lt;- read.frequency(frequency=jsm.mt.freq.file, popname=jsm.popname.file)

# Read frequency file without subpopulation names
popdata.mt.noname &lt;- read.frequency(frequency=jsm.mt.freq.file)
</code></pre>

<hr>
<h2 id='read.genepop'>Create a genotype data object of populations from a GENEPOP format file.</h2><span id='topic+read.genepop'></span>

<h3>Description</h3>

<p>This function reads a GENEPOP format file (Rousset 2008) and parse it into an R data object. This data object provides a summary of genotype/haplotype of each sample, allele frequency in each population, and marker status. This data object is used in downstream analysis of this package. This function is a &quot;lite&quot; and faster version of readGenepop function in diveRsity package (Keenan 2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.genepop(genepop, popname = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.genepop_+3A_genepop">genepop</code></td>
<td>
<p>A character value specifying the name of the GENEPOP file to be analyzed.</p>
</td></tr>
<tr><td><code id="read.genepop_+3A_popname">popname</code></td>
<td>
<p>A character value specifying the name of the plain text file containing the names of subpopulations to be analyzed. This text file must not contain other than subpopulation names. The names must be separated by spaces, tabs or line breaks. If this argument is omitted, serial numbers will be assigned as subpopulation names.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>npops</code></td>
<td>
<p>Number of subpopulations.</p>
</td></tr>
<tr><td><code>pop_sizes</code></td>
<td>
<p>Number of samples in each subpopulation.</p>
</td></tr>
<tr><td><code>pop_names</code></td>
<td>
<p>Names of subpopulations.</p>
</td></tr>
<tr><td><code>nloci</code></td>
<td>
<p>Number of loci.</p>
</td></tr>
<tr><td><code>loci_names</code></td>
<td>
<p>Names of loci.</p>
</td></tr>
<tr><td><code>all_alleles</code></td>
<td>
<p>A list of alleles at each locus.</p>
</td></tr>
<tr><td><code>nalleles</code></td>
<td>
<p>Number of alleles at each locus.</p>
</td></tr>
<tr><td><code>indtyp</code></td>
<td>
<p>Number of genotyped samples in each subpopulation at each locus.</p>
</td></tr>
<tr><td><code>ind_names</code></td>
<td>
<p>Names of samples in each subpopulation.</p>
</td></tr>
<tr><td><code>pop_alleles</code></td>
<td>
<p>Genotypes of each sample at each locus in haploid designation.</p>
</td></tr>
<tr><td><code>pop_list</code></td>
<td>
<p>Genotypes of each sample at each locus in diploid designation.</p>
</td></tr>
<tr><td><code>obs_allele_num</code></td>
<td>
<p>Observed allele counts at each locus in each subpopulation.</p>
</td></tr>
<tr><td><code>allele_freq</code></td>
<td>
<p>Observed allele frequencies at each locus in each subpopulation.</p>
</td></tr>
<tr><td><code>call_rate</code></td>
<td>
<p>Rate of genotyped samples at each locus.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi</p>


<h3>References</h3>

<p>Keenan K (2015) diveRsity: A Comprehensive, General Purpose Population Genetics Analysis Package. <a href="https://github.com/kkeenan02/diveRsity">https://github.com/kkeenan02/diveRsity</a><br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Read GENEPOP file with subpopulation names
# Prepare your GENEPOP file and population name file in the working directory
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.genepop(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# Read GENEPOP file without subpopulation names
popdata.noname &lt;- read.genepop(genepop=jsm.ms.genepop.file)
</code></pre>

<hr>
<h2 id='thetaWC.pair'>Weir and Cockerham's theta adapted for pairwise Fst.</h2><span id='topic+thetaWC.pair'></span>

<h3>Description</h3>

<p>This function estimates Fst between population pairs based on Weir and Cockerham's theta (Weir &amp; Cockerham 1984) adapted for pairwise comparison from a GENEPOP data object (Rousset 2008). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>thetaWC.pair(popdata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="thetaWC.pair_+3A_popdata">popdata</code></td>
<td>
<p>Population data object created by read.genepop function from a GENEPOP file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Weir and Cockerham (1984) derived an unbiased estimator of a coancestry coefficient (theta) based on a random effect model. It expresses the extent of genetic heterogeneity within the population. The second stage common approach is to investigate the detailed pattern of the population structure, based on a measure of genetic difference between pairs of subpopulations (demes). We call this by pairwise Fst. This function follows the formula of Weir and Cockerham's theta with the sample size <em>r</em> = 2. Given the pair, our finite sample correction multiplies <em>a</em> of Weir &amp; Cockerham's theta by (<em>r</em> - 1) / <em>r</em> (equation 2 in p.1359 of Weir &amp; Cockerham 1984).
</p>


<h3>Value</h3>

<p>Matrix of estimated pairwise Fst by theta with finite sample correction.
</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.<br /><br />
Weir BS, Cockerham CC (1984) Estimating F-statistics for the analysis of population structure. <em>Evolution</em>, 38, 1358-1370.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.genepop">read.genepop</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.genepop(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# theta estimation
result.theta.pair &lt;- thetaWC.pair(popdata)
print(as.dist(result.theta.pair))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
