<!DOCTYPE html><html><head><title>Help for package FinePop2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FinePop2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FinePop2-package'><p>Fine-Scale Population Analysis (Rewrite for Gene-Trait-Environment Interaction Analysis)</p></a></li>
<li><a href='#clip.GENEPOP'><p>Remove designated markers from a GENEPOP file.</p></a></li>
<li><a href='#globalFST'><p>Genome-wide global FST (Weir &amp; Cockerham 1984).</p></a></li>
<li><a href='#GLS'><p>Generalized least squres for regression analysis considering auto correlation.</p></a></li>
<li><a href='#herring'><p>An example dataset of Atlantic herring.</p></a></li>
<li><a href='#jsmackerel'><p>An example dataset of Japanese Spanich mackerel in GENEPOP and frequency format.</p></a></li>
<li><a href='#locus_specificFST'><p>Locus-specific global FST (Kitada et al. 2007, 2017).</p></a></li>
<li><a href='#pop_pairwiseFST'><p>Genome-wide poppulation-paiwise FST (Nei &amp; Chesser 1983).</p></a></li>
<li><a href='#pop_specificFST'><p>Genome-wide population-specific FST (Weir &amp; Goudet 2017).</p></a></li>
<li><a href='#read.GENEPOP'><p>Create a genotype data object of populations from a GENEPOP format file.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Fine-Scale Population Analysis (Rewrite for
Gene-Trait-Environment Interaction Analysis)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-02-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Reiichiro Nakamichi, Shuichi Kitada, Hirohisa Kishino</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Reiichiro Nakamichi &lt;nakamichi@affrc.go.jp&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Statistical tool set for population genetics. The package provides following functions: 1) estimators of genetic differentiation (FST), 2) regression analysis of environmental effects on genetic differentiation using generalized least squares (GLS) method, 3) interfaces to read and manipulate 'GENEPOP' format data files). For more information, see Kitada, Nakamichi and Kishino (2020) &lt;<a href="https://doi.org/10.1101%2F2020.01.30.927186">doi:10.1101/2020.01.30.927186</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2.0)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-02-20 14:55:40 UTC; nakamichi</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-02-28 10:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='FinePop2-package'>Fine-Scale Population Analysis (Rewrite for Gene-Trait-Environment Interaction Analysis)</h2><span id='topic+FinePop2-package'></span>

<h3>Description</h3>

<p>Statistical tool set for population genetics. The package provides following functions: 1) estimators of genetic differentiation (FST), 2) regression analysis of environmental effects on genetic differentiation using generalized least squares (GLS) method, 3) interfaces to read and manipulate 'GENEPOP' format data files). For more information, see Kitada, Nakamichi and Kishino (2020) &lt;doi:10.1101/2020.01.30.927186&gt;.</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Shuichi Kitada, Hirohisa Kishino<br />
Maintainer: Reiichiro Nakamichi &lt;nakamichi@affrc.go.jp&gt;
</p>


<h3>References</h3>

<p>Kitada S, Nakamichi R, Kishino H (2017) The empirical Bayes estimators of fine-scale population structure in high gene flow species. <em>Mol. Ecol. Resources</em>, 17, 1210-1222.<br />
Nei M, Chesser RK (1983) Estimation of fixation indices and gene diversity. <em>Annals of Human Genetics</em>, 47, 253-259.<br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.<br />
Weir BS, Cockerham CC (1984) Estimating F-statistics for the analysis of population structure. <em>Evolution</em>, 38, 1358-1370.<br />
Weir BS, Goudet J (2017) A Unified Characterization of Population Structure and Relatedness. <em>Genetics</em>, 206, 2085-2103.
</p>

<hr>
<h2 id='clip.GENEPOP'>Remove designated markers from a GENEPOP file.</h2><span id='topic+clip.GENEPOP'></span>

<h3>Description</h3>

<p>This function reads a GENEPOP file (Rousset 2008), remove designated markers, and write a GENEPOP file of clipped data. The user can directly designate the names of the markers to be removed. The user also can set the filtering threshold of major allele frequency. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clip.GENEPOP(infile, outfile, remove.list = NULL, major.af = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clip.GENEPOP_+3A_infile">infile</code></td>
<td>
<p>A character value specifying the name of the GENEPOP file to be clipped.</p>
</td></tr>
<tr><td><code id="clip.GENEPOP_+3A_outfile">outfile</code></td>
<td>
<p>A character value specifying the name of the clipped GENEPOP file.</p>
</td></tr>
<tr><td><code id="clip.GENEPOP_+3A_remove.list">remove.list</code></td>
<td>
<p>A character value or vector specifying the names of the markers to be removed. The names must be included in the target GENEPOP file.</p>
</td></tr>
<tr><td><code id="clip.GENEPOP_+3A_major.af">major.af</code></td>
<td>
<p>A numeric value specifying the threshold of major allele frequency for marker removal. Markers with major allele frequencies higher than this value will be removed. This value must be between 0 and 1.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi</p>


<h3>References</h3>

<p>Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.genepop.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.genepop.file, sep="\n")

# Remove markers designated by their names
clipped_by_name.jsm.genepop.file &lt;- tempfile()
clip.GENEPOP(infile=jsm.genepop.file,
             outfile=clipped_by_name.jsm.genepop.file,
             remove.list=c("Sni21","Sni26"))

# Remove markers with high major allele frequencies (in this example, &gt; 0.5)
clipped_by_af.jsm.genepop.file &lt;- tempfile()
clip.GENEPOP(infile=jsm.genepop.file,
             outfile=clipped_by_af.jsm.genepop.file,
             major.af=0.5)

# Remove markers both by their names and by major allele frequencies
clipped_by_both.jsm.genepop.file &lt;- tempfile()
clip.GENEPOP(infile=jsm.genepop.file,
             outfile=clipped_by_both.jsm.genepop.file,
             remove.list=c("Sni21","Sni26"), major.af=0.5)

# See four text files in temporary directory.
#  jsm.genepop.file                 : original data of five markers
#  clipped_by_name.jsm.genepop.file : clipped data by marker names
#  clipped_by_af.jsm.genepop.file   : clipped data by allele frequency
#  clipped_by_both.jsm.genepop.file : clipped data by both names and frequency
</code></pre>

<hr>
<h2 id='globalFST'>Genome-wide global FST (Weir &amp; Cockerham 1984).</h2><span id='topic+globalFST'></span>

<h3>Description</h3>

<p>This function estimates genom-wide global FST based on Weir and Cockerham's theta (Weir &amp; Cockerham 1984) from a GENEPOP data object (Rousset 2008). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>globalFST(popdata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="globalFST_+3A_popdata">popdata</code></td>
<td>
<p>Population data object created by <code>read.GENEPOP</code> function from a GENEPOP file.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>fst</code></td>
<td>
<p>Estimated genome-wide global FST</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>Standard error of estimated FST</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Shuichi Kitada, Hirohisa Kishino</p>


<h3>References</h3>

<p>Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.<br /><br />
Weir BS, Cockerham CC (1984) Estimating F-statistics for the analysis of population structure. <em>Evolution</em>, 38, 1358-1370.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.GENEPOP">read.GENEPOP</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory.
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.GENEPOP(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# theta estimation
result.globalFST &lt;- globalFST(popdata)
print(result.globalFST)
</code></pre>

<hr>
<h2 id='GLS'>Generalized least squres for regression analysis considering auto correlation.</h2><span id='topic+GLS'></span>

<h3>Description</h3>

<p>This function provides a multiple regression analysis considering auto correlation of response variable using generalized least squres method (Aitken 1934). It supports <code><a href="stats.html#topic+lm">lm</a></code> like format of <code>model</code>.  A typical model has the form <code>response ~ terms</code>. Terms specification supports only <code>first + second</code> form. Cross term specification of <code>first * second</code> form is not supported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLS(model, data, omega = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GLS_+3A_model">model</code></td>
<td>
<p>Symbolic description of the model to be fitted.</p>
</td></tr>
<tr><td><code id="GLS_+3A_data">data</code></td>
<td>
<p>Data frame containing variables in the <code>model</code>.</p>
</td></tr>
<tr><td><code id="GLS_+3A_omega">omega</code></td>
<td>
<p>A numeric matrix of auto correlation of responce variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>coefficients</code></td>
<td>
<p>Estimated coefficient, standard error, Z value and p value of each factor.</p>
</td></tr>
<tr><td><code>variance</code></td>
<td>
<p>Variance-covariance matrix of estimated coefficients.</p>
</td></tr>
<tr><td><code>logL</code></td>
<td>
<p>Log likelihood of fitted model.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Shuichi Kitada, Hirohisa Kishino</p>


<h3>References</h3>

<p>Aitken AC (1934) On Least-squares and Linear Combinations of Observations. <em>Proceedings of the Royal Society of Edinburgh</em>, 55, 42-48.</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm">lm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example data of Atlantic herring
data(herring)
ah.genepop.file &lt;- tempfile()
ah.popname.file &lt;- tempfile()
cat(herring$genepop, file=ah.genepop.file, sep="\n")
cat(herring$popname, file=ah.popname.file, sep=" ")

# Data load
popdata &lt;- read.GENEPOP(ah.genepop.file, ah.popname.file)

# Pop-specific FST and correlation among populations
fst.popsp &lt;- pop_specificFST(popdata, cov=TRUE)
cov.fst.popsp &lt;- fst.popsp$cov
sd.fst.popsp &lt;- sqrt(diag(cov.fst.popsp))
cov2.fst.popsp &lt;- apply(cov.fst.popsp, 2, function(x){x / sd.fst.popsp})
cor.fst.popsp &lt;- apply(cov2.fst.popsp, 1, function(x){x / sd.fst.popsp})

# Pop-pairwise FST and population structure
fst.poppair &lt;- pop_pairwiseFST(popdata)
fst.md &lt;- cmdscale(fst.poppair)

# GLS analysis of FST and environmental factors
test.data &lt;- data.frame(fst=fst.md[,1], herring$environment)
GLS(fst~., scale(test.data), omega=cor.fst.popsp)
</code></pre>

<hr>
<h2 id='herring'>An example dataset of Atlantic herring.</h2><span id='topic+herring'></span>

<h3>Description</h3>

<p>An example of a genetic data for Atlantic herring population (Limborg et al. 2012). It contains genotypic information of 281 SNPs from 18 subpopulations of 607 individuals. GENEPOP format (Rousset 2008) text file is available. Subpopulation names, environmental factors (longitude, latitude, temperature and salinity) at each subpopulation are attached.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("herring")</code></pre>


<h3>Format</h3>

<p>$ genepop       : Genotypic information of 281 SNPs in GENEPOP format text data.<br />
$ popname       : Names of subpopulations.<br />
$ environment   : Table of temperature and salinity at each subpopulation.<br />
</p>


<h3>References</h3>

<p>Limborg MT, Helyar SJ, de Bruyn M et al. (2012) Environmental selection on transcriptome-derived SNPs in a high gene flow marine fish, the Atlantic herring (Clupea harengus). <em>Molecular Ecology</em>, 21, 3686-3703. <br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.GENEPOP">read.GENEPOP</a></code>, <code><a href="#topic+GLS">GLS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(herring)
ah.genepop.file &lt;- tempfile()
ah.popname.file &lt;- tempfile()
cat(herring$genepop, file=ah.genepop.file, sep="\n")
cat(herring$popname, file=ah.popname.file, sep=" ")

# See two text files in temporary directory.
#  ah.genepop.file  : GENEPOP format file of 281SNPs in 18 subpopulations
#  ah.popname.file  : plain text file of subpopulation names

print(herring$environment)
</code></pre>

<hr>
<h2 id='jsmackerel'>An example dataset of Japanese Spanich mackerel in GENEPOP and frequency format.</h2><span id='topic+jsmackerel'></span>

<h3>Description</h3>

<p>An example of a genetic data for a Japanese Spanish mackerel population (Nakajima et al. 2014). It contains genotypic information of 5 microsatellite markers from 8 subpopulations of 715 individuals. GENEPOP format (Rousset 2008) text files are available. Name list of subpopulations also is attached.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("jsmackerel")</code></pre>


<h3>Format</h3>

<p>$ MS.genepop: Genotypic information of 5 microsatellites in GENEPOP format text data.<br />
$ popname: Names of subpopulations.
</p>


<h3>References</h3>

<p>Nakajima K et al. (2014) Genetic effects of marine stock enhancement: a case study based on the highly piscivorous Japanese Spanish mackerel. <em>Canadian Journal of Fisheries and Aquatic Sciences</em>, 71, 301-314.<br /><br />
Kitada S, Kitakado T, Kishino H (2007) Empirical Bayes inference of pairwise FST and its distribution in the genome. <em>Genetics</em>, 177, 861-873.<br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.GENEPOP">read.GENEPOP</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# See two text files in temporary directory.
#  jsm.ms.genepop.file  : GENEPOP format file of microsatellite data
#  jsm.popname.file     : plain text file of subpopulation names
</code></pre>

<hr>
<h2 id='locus_specificFST'>Locus-specific global FST (Kitada et al. 2007, 2017).</h2><span id='topic+locus_specificFST'></span>

<h3>Description</h3>

<p>This function estimates locus-specific global FST among subpopulations using empirical Bayes method (Kitada et al. 2007, 2017) from a GENEPOP data object (Rousset 2008). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locus_specificFST(popdata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="locus_specificFST_+3A_popdata">popdata</code></td>
<td>
<p>Population data object created by <code>read.GENEPOP</code> function from a GENEPOP file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated locus-specific global FST.</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Shuichi Kitada, Hirohisa Kishino</p>


<h3>References</h3>

<p>Kitada S, Kitakado T, Kishino H (2007) Empirical Bayes inference of pairwise FST and its distribution in the genome. <em>Genetics</em>, 177, 861-873.<br /><br />
Kitada S, Nakamichi R, Kishino H (2017) The empirical Bayes estimators of fine-scale population structure in high gene flow species. <em>Mol. Ecol. Resources</em>, DOI: 10.1111/1755-0998.12663 <br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.GENEPOP">read.GENEPOP</a></code>, <code><a href="FinePop.html#topic+read.frequency">read.frequency</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory.
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.GENEPOP(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# FST estimation
locspFST &lt;- locus_specificFST(popdata)
print(locspFST)
</code></pre>

<hr>
<h2 id='pop_pairwiseFST'>Genome-wide poppulation-paiwise FST (Nei &amp; Chesser 1983).</h2><span id='topic+pop_pairwiseFST'></span>

<h3>Description</h3>

<p>This function estimates genome-wide poppulation-paiwise FST among subpopulations based on Nei and Chesser's corrected GST (Nei&amp;Chesser 1983) from a GENEPOP data object (Rousset 2008). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pop_pairwiseFST(popdata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pop_pairwiseFST_+3A_popdata">popdata</code></td>
<td>
<p>Population data object created by <code>read.GENEPOP</code> function from a GENEPOP file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated genome-wide population-pairwise FST.</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Shuichi Kitada, Hirohisa Kishino</p>


<h3>References</h3>

<p>Nei M, Chesser RK (1983) Estimation of fixation indices and gene diversity. <em>Annals of Human Genetics</em>, 47, 253-259.<br /><br />
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.GENEPOP">read.GENEPOP</a></code>,
<code><a href="stats.html#topic+as.dist">as.dist</a></code>, <code><a href="stats.html#topic+as.dendrogram">as.dendrogram</a></code>,
<code><a href="stats.html#topic+hclust">hclust</a></code>, <code><a href="stats.html#topic+cmdscale">cmdscale</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory.
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.GENEPOP(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# GST estimation
result.poppairFST &lt;- pop_pairwiseFST(popdata)
poppairFST.d &lt;- as.dist(result.poppairFST)
print(poppairFST.d)

# dendrogram
poppairFST.hc &lt;- hclust(poppairFST.d,method="average")
plot(as.dendrogram(poppairFST.hc), xlab="",ylab="",main="", las=1)

# MDS plot
mds &lt;- cmdscale(poppairFST.d)
plot(mds, type="n", xlab="",ylab="")
text(mds[,1],mds[,2], popdata$pop_names)
</code></pre>

<hr>
<h2 id='pop_specificFST'>Genome-wide population-specific FST (Weir &amp; Goudet 2017).</h2><span id='topic+pop_specificFST'></span>

<h3>Description</h3>

<p>This function estimates genome-wide poppulation-specific FST based on Weir and Goudet's Method (Weir&amp;Goudet 2017) from a GENEPOP data object (Rousset 2008). Missing genotype values in the GENEPOP file (&quot;0000&quot; or &quot;000000&quot;) are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pop_specificFST(popdata, cov = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pop_specificFST_+3A_popdata">popdata</code></td>
<td>
<p>Population data object created by <code>read.GENEPOP</code> function from a GENEPOP file.</p>
</td></tr>
<tr><td><code id="pop_specificFST_+3A_cov">cov</code></td>
<td>
<p>A Logical argument indicating whether variance-covariance matrix of estimated FST should be calculated.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>fst</code></td>
<td>
<p>Estimated genome-wide population-specific FST and standard error.</p>
</td></tr>
<tr><td><code>cov</code></td>
<td>
<p>Variance-covariance matrix of estimated FST</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Shuichi Kitada, Hirohisa Kishino</p>


<h3>References</h3>

<p>Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.<br /><br />
Weir BS, Goudet J (2017) A Unified Characterization of Population Structure and Relatedness. <em>Genetics</em>, 206, 2085-2103.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.GENEPOP">read.GENEPOP</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory.
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.GENEPOP(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# FST estimation
result.popspFST &lt;- pop_specificFST(popdata)
print(result.popspFST$fst)
</code></pre>

<hr>
<h2 id='read.GENEPOP'>Create a genotype data object of populations from a GENEPOP format file.</h2><span id='topic+read.GENEPOP'></span>

<h3>Description</h3>

<p>This function reads a GENEPOP format file (Rousset 2008) and parse it into an R data object. This data object provides a summary of genotype/haplotype of each sample, allele frequency in each population, and marker status. This data object is used in downstream analysis of this package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.GENEPOP(genepop, popname = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.GENEPOP_+3A_genepop">genepop</code></td>
<td>
<p>A character value specifying the name of the GENEPOP file to be analyzed.</p>
</td></tr>
<tr><td><code id="read.GENEPOP_+3A_popname">popname</code></td>
<td>
<p>A character value specifying the name of the plain text file containing the names of subpopulations to be analyzed. This text file must not contain other than subpopulation names. The names must be separated by spaces, tabs or line breaks. If this argument is omitted, serial numbers will be assigned as subpopulation names.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>num_pop</code></td>
<td>
<p>Number of subpopulations.</p>
</td></tr>
<tr><td><code>pop_sizes</code></td>
<td>
<p>Number of samples in each subpopulation.</p>
</td></tr>
<tr><td><code>pop_names</code></td>
<td>
<p>Names of subpopulations.</p>
</td></tr>
<tr><td><code>ind_names</code></td>
<td>
<p>Names of samples in each subpopulation.</p>
</td></tr>
<tr><td><code>num_loci</code></td>
<td>
<p>Number of loci.</p>
</td></tr>
<tr><td><code>loci_names</code></td>
<td>
<p>Names of loci.</p>
</td></tr>
<tr><td><code>num_allele</code></td>
<td>
<p>Number of alleles at each locus.</p>
</td></tr>
<tr><td><code>allele_list</code></td>
<td>
<p>A list of alleles at each locus.</p>
</td></tr>
<tr><td><code>ind_count</code></td>
<td>
<p>Observed count of genotyped samples in each subpopulation at each locus.</p>
</td></tr>
<tr><td><code>allele_count</code></td>
<td>
<p>Observed count of genotyped alleles in each subpopulation at each locus.</p>
</td></tr>
<tr><td><code>allele_freq</code></td>
<td>
<p>Observed allele frequencies in each subpopulation at each locus.</p>
</td></tr>
<tr><td><code>genotype</code></td>
<td>
<p>Genotypes of each sample at each locus in haploid designation.</p>
</td></tr>
<tr><td><code>call_rate_loci</code></td>
<td>
<p>Call rate of each locus (rate of genotyped samples at each locus).</p>
</td></tr>
<tr><td><code>call_rate_ind</code></td>
<td>
<p>Call rate of each sample (rate of genotyped markers for each sample).</p>
</td></tr>
<tr><td><code>He</code></td>
<td>
<p>Expected heterozigosity in each subpopulation.</p>
</td></tr>
<tr><td><code>Ho</code></td>
<td>
<p>Observed heterozigosity in each subpopulation.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi</p>


<h3>References</h3>

<p>Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Read GENEPOP file with subpopulation names.
# Prepare your GENEPOP file and population name file in the working directory.
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.GENEPOP(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# Read GENEPOP file without subpopulation names.
popdata.noname &lt;- read.GENEPOP(genepop=jsm.ms.genepop.file)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
