<!DOCTYPE html><html><head><title>Help for package FiRE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FiRE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FiRE'>
<p>Constructor for Class</p></a></li>
<li><a href='#FiRE-package'>
<p>Finder of Rare Entities (FiRE) - To assign rareness/ outlierness score to every sample</p></a></li>
<li><a href='#fit'>
<p>Hash samples in bins</p></a></li>
<li><a href='#sample_data'><p>Preprocessed 293T–Jurkat scRNA-seq dataset</p></a></li>
<li><a href='#sample_label'><p>Labels of preprocessed 293T–Jurkat scRNA-seq dataset</p></a></li>
<li><a href='#score'>
<p>Compute score on hashed sample.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Finder of Rare Entities (FiRE)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>The algorithm assigns rareness/ outlierness score to every sample in voluminous datasets.
    The algorithm makes multiple estimations of the proximity between a pair of samples, in low-dimensional spaces. To compute proximity, FiRE uses Sketching, a variant of locality sensitive hashing. For more details: Jindal, A., Gupta, P., Jayadeva and Sengupta, D., 2018. Discovery of rare cells from voluminous single cell expression data. Nature Communications, 9(1), p.4719. &lt;<a href="https://doi.org/10.1038%2Fs41467-018-07234-6">doi:10.1038/s41467-018-07234-6</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/princethewinner/FiRE">https://github.com/princethewinner/FiRE</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/princethewinner/FiRE/issues">https://github.com/princethewinner/FiRE/issues</a></td>
</tr>
<tr>
<td>Note:</td>
<td>While reporting issues on github start subject line with
[cran-package]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, Rcpp (&ge; 0.12.19)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, BH</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Author:</td>
<td>Prashant Gupta [aut, cre],
  Aashi Jindal [aut],
  Jayadeva [aut],
  Debarka Sengupta [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Prashant Gupta &lt;prashant10991@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-06 08:46:17 UTC; prashant</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-06 09:20:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='FiRE'>
Constructor for Class
</h2><span id='topic+Rcpp_FiRE-class'></span>

<h3>Description</h3>

<p>Constructor to create an instance of FiRE class. There are two ways in which constructor can be created.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="FiRE_+3A_l">L</code></td>
<td>
<p>Number of estimators.</p>
</td></tr>
<tr><td><code id="FiRE_+3A_m">M</code></td>
<td>
<p>Number of dimensions to be sampled per estimator.</p>
</td></tr>
<tr><td><code id="FiRE_+3A_h">H</code></td>
<td>
<p>Number of bins. Default=1017881</p>
</td></tr>
<tr><td><code id="FiRE_+3A_seed">seed</code></td>
<td>
<p>seed for random number generator. Default=0</p>
</td></tr>
<tr><td><code id="FiRE_+3A_verbose">verbose</code></td>
<td>
<p>verbose level. Default=1</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For usage see example.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
     L &lt;- 100
     M &lt;- 50
     H &lt;- 107881
     seed &lt;- 0
     verbose &lt;- 1

     ## Creating class object with required arguments
     model &lt;- new(FiRE::FiRE, L, M)

     ## Creating class object with all arguments
     model &lt;- new(FiRE::FiRE, L, M, H, seed, verbose)

</code></pre>

<hr>
<h2 id='FiRE-package'>
Finder of Rare Entities (FiRE) - To assign rareness/ outlierness score to every sample
</h2><span id='topic+FiRE'></span><span id='topic+Rcpp_FiRE'></span>

<h3>Description</h3>

<p>This package sorts samples as per their rareness/ oultierness. Instead of dichotomized decisions, FiRE assigns rareness/ outlierness score to every sample. These scores can then be used to identify rare samples or outliers, with varying degrees of rareness. FiRE takes multiple estimations of the proximity between a pair of samples, in low-dimensional spaces to compute scores. </p>


<h3>Details</h3>

<p>FiRE is written in c++ and wrapped with Rcpp to create an R interface. To use FiRE, an object with the number of estimators (L), number of dimensions to be sampled per estimator (M), number of bins per estimator (H = 1017881), seed for random number generator (seed = 0) and verbose level (verbose = 0/1) needs to be created. Once the object is created, <code>fit</code> function needs to be called to hash all the samples into bins. Once hashing is done, <code>score</code> function needs to be called to retrieve score for every sample. Sample commands may be seen in the Examples section. The resulting model has four model parameters which may be accessed, dimensions (d) of size M sampled per estimator, thresholds (ths) of size M per d, weights(w) of size M generated per estimator and bin (b) of size H per estimator.
</p>


<h3>Author(s)</h3>

<p>Prashant Gupta, prashant10991@gmail.com
</p>
<p>Aashi Jindal, jindal.aashi21@gmail.com
</p>
<p>Jayadeva, iitjd4@gmail.com
</p>
<p>Debarka Sengupta, debarka@gmail.com
</p>
<p>Maintainer: Prashant Gupta &lt;prashant10991@gmail.com&gt;
</p>


<h3>References</h3>

<p>[1]Jindal, A., Gupta, P., Jayadeva and Sengupta, D., 2018. Discovery of rare cells from voluminous single cell expression data. Nature Communications, 9(1), p.4719.
</p>
<p>[2]Wang, Z., Dong, W., Josephson, W., Lv, Q., Charikar, M. and Li, K., 2007, June. Sizing sketches: a rank-based analysis for similarity search. In ACM SIGMETRICS Performance Evaluation Review (Vol. 35, No. 1, pp. 157-168). ACM.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
     ## L &lt;- number of estimators
     ## M &lt;- Number of dims to sample
     ## H &lt;- Number of bins
     ## seed &lt;- seed for random number generator
     ## verbose &lt;- verbose level

     library('FiRE')
     data(sample_data) #Samples * Features

     data(sample_label)
     ## Samples with label '1' represent abundant,
     ## while samples with label '2' represent rare.

     ## Saving rownames and colnames
     rnames &lt;- rownames(sample_data)
     cnames &lt;- colnames(sample_data)

     ## Converting data.frame to matrix
     sample_data &lt;- as.matrix(sample_data)
     sample_label &lt;- as.matrix(sample_label)

     L &lt;- 100 # Number of estimators
     M &lt;- 50 # Dims to be sampled

     # Model creation without optional parameter
     model &lt;- new(FiRE::FiRE, L, M)

     ## There are 3 more optional parameters they can be passed as
     ## model &lt;- new(FiRE::FiRE, L, M, H, seed, verbose)

     ## Hashing all samples
     model$fit(sample_data)

     ## Computing score of each sample
     rareness_score &lt;- model$score(sample_data)

     ## Apply IQR-based criteria to identify rare samples for further downstream analysis.
     q3 &lt;- quantile(rareness_score, 0.75)
     iqr &lt;- IQR(rareness_score)
     th &lt;- q3 + (1.5*iqr)

     ## Select indexes that satisfy IQR-based thresholding criteria.
     indIqr &lt;- which(rareness_score &gt;= th)

     ## Create a vector for binary predictions
     predictions &lt;- rep(1, dim(sample_data)[1])
     predictions[indIqr] &lt;- 2 #Replace predictions for rare samples with '2'.

     ## To access model parameters
     ## Parameters - generated weights
     # model$w

     ## Parameters - sample dimensions
     # model$d

     ## Parameters - generated thresholds
     # model$ths

     ## Parameters - Bins
     # model$b
</code></pre>

<hr>
<h2 id='fit'>
Hash samples in bins
</h2><span id='topic+fit'></span>

<h3>Description</h3>

<p>Hashing samples into bins, using Sketching as hash function. This function repeats hashing process <code>L</code> times by sampling <code>M</code> dimensions at a time and hashes samples into one of <code>H</code> bins.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_+3A_data">data</code></td>
<td>
<p>On which rarity score needs to be computed. Required to be a <code>matrix</code>. (can be converted to matrix with <code>as.matrix</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For usage see example.
</p>


<h3>Note</h3>

<p>This function does not do any preprocessing, so preprocessed data must be passed if required.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 

     ## Creating class object with required arguments
     model &lt;- new(FiRE::FiRE, L, M)
     model$fit(data)

  
## End(Not run)
</code></pre>

<hr>
<h2 id='sample_data'>Preprocessed 293T–Jurkat scRNA-seq dataset</h2><span id='topic+sample_data'></span>

<h3>Description</h3>

<p>This data set has been generated from 293T and Jurkat cell data containing a total of ~3200 cells,
with an almost equal number of representative transcriptomes of each type.
The cells were mixed in vitro at equal proportions.
Authors of the study resolved the cell types bioinformatically exploiting their SNV profiles.
With this data, we mimicked the rare cell phenomenon by bioinformatically diluting
Jurkat cell proportion to 2.5% in the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sample_data)</code></pre>


<h3>Format</h3>

<p>A matrix containing 1580 observations (in rows), with 1000 features per observation (in columns).</p>


<h3>Source</h3>

<p>10xgenomics, https://support.10xgenomics.com/single-cell-gene-expression/datasets</p>


<h3>References</h3>

<p>Zheng, G. X. et al.
Massively parallel digital transcriptional profiling of single cells.
<em>Nat. Commun.</em> 8, 14049 (2017).
</p>

<hr>
<h2 id='sample_label'>Labels of preprocessed 293T–Jurkat scRNA-seq dataset</h2><span id='topic+sample_label'></span>

<h3>Description</h3>

<p>This contains labels for 1580 observations. There are two different clusters,
one is abundant (denoted by 1) and the other is rare (denoted by 2).
Rare cluster consists of Jurkat cells, which form 2.5% of the total population.
The abundant cluster consists of 293T cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sample_label)</code></pre>


<h3>Format</h3>

<p>A vector containing 1580 observations.</p>


<h3>Source</h3>

<p>10xgenomics, https://support.10xgenomics.com/single-cell-gene-expression/datasets</p>


<h3>References</h3>

<p>Zheng, G. X. et al.
Massively parallel digital transcriptional profiling of single cells.
<em>Nat. Commun.</em> 8, 14049 (2017).
</p>

<hr>
<h2 id='score'>
Compute score on hashed sample.
</h2><span id='topic+score'></span>

<h3>Description</h3>

<p>Once hashing is done using <code>fit</code>, compute score via proximity estimation.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="score_+3A_data">data</code></td>
<td>
<p>On which rarity score needs to be computed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For usage see example.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 

     ## Creating class object with required arguments
     model &lt;- new(FiRE::FiRE, L, M)
     model$fit(data)
     score &lt;- model$score(data)

  
## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
