<!DOCTYPE html><html><head><title>Help for package fishtree</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {fishtree}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fishtree-package'><p>fishtree: Interface to the Fish Tree of Life API</p></a></li>
<li><a href='#fishtree_alignment'><p>Get aligned sequences from the Fish Tree of Life</p></a></li>
<li><a href='#fishtree_complete_phylogeny'><p>Get complete (stochastically-resolved) phylogenies from the Fish Tree of Life</p></a></li>
<li><a href='#fishtree_phylogeny'><p>Get a phylogeny from the Fish Tree of Life</p></a></li>
<li><a href='#fishtree_rogues'><p>Get rogue taxa that break the monophyly of defined taxa</p></a></li>
<li><a href='#fishtree_taxonomy'><p>Get taxonomies and other data from the Fish Tree of Life</p></a></li>
<li><a href='#fishtree_tip_rates'><p>Get tip rates for the Fish Tree of Life</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Interface to the Fish Tree of Life API</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.4</td>
</tr>
<tr>
<td>Description:</td>
<td>An interface to the Fish Tree of Life API to download taxonomies,
    phylogenies, fossil calibrations, and diversification rate information for
    ray-finned fishes.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-2-Clause">BSD_2_clause</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://fishtreeoflife.org/">https://fishtreeoflife.org/</a>, <a href="https://github.com/jonchang/fishtree">https://github.com/jonchang/fishtree</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jonchang/fishtree/issues">https://github.com/jonchang/fishtree/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape (&ge; 5.2), jsonlite (&ge; 1.5), memoise, parallel, rlang (&ge;
0.4.1), utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>diversitree, geiger, hisse (&ge; 1.9.10), knitr, markdown,
phytools, picante, rfishbase, rmarkdown, spelling, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-28 04:06:30 UTC; jonchang</td>
</tr>
<tr>
<td>Author:</td>
<td>Jonathan Chang <a href="https://orcid.org/0000-0002-3811-1254"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Michael E Alfaro [ctb],
  Daniel L Rabosky [ctb],
  Stephen A Smith [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jonathan Chang &lt;me@jonathanchang.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-31 00:50:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='fishtree-package'>fishtree: Interface to the Fish Tree of Life API</h2><span id='topic+fishtree'></span><span id='topic+fishtree-package'></span>

<h3>Description</h3>

<p>fishtree makes it easy to access phylogenetic information about ray-finned
fishes with R. Download taxonomies, phylogenies, sequence matrices, and
diversification rate data with a simple set of functions.
</p>


<h3>Details</h3>

<p>Implementation note: this package makes calls over the network and caches
its (sometimes large) results for faster loading. Because of this,
long-running R instances could use a lot of memory.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Jonathan Chang <a href="mailto:me@jonathanchang.org">me@jonathanchang.org</a> (<a href="https://orcid.org/0000-0002-3811-1254">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Michael E Alfaro [contributor]
</p>
</li>
<li><p> Daniel L Rabosky [contributor]
</p>
</li>
<li><p> Stephen A Smith [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://fishtreeoflife.org/">https://fishtreeoflife.org/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/jonchang/fishtree/issues">https://github.com/jonchang/fishtree/issues</a>
</p>
</li></ul>


<hr>
<h2 id='fishtree_alignment'>Get aligned sequences from the Fish Tree of Life</h2><span id='topic+fishtree_alignment'></span>

<h3>Description</h3>

<p>Retrieves an aligned sequence via the Fish Tree of Life API. If neither 'species' nor 'rank' are specified, returns the entire sequence matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fishtree_alignment(species, rank, split = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fishtree_alignment_+3A_species">species</code></td>
<td>
<p>(Optionally) subset the results based on a vector of species names.</p>
</td></tr>
<tr><td><code id="fishtree_alignment_+3A_rank">rank</code></td>
<td>
<p>(Optionally) subset the results based on the supplied taxonomic rank.</p>
</td></tr>
<tr><td><code id="fishtree_alignment_+3A_split">split</code></td>
<td>
<p>Splits the output into a list by gene locus.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class '&quot;DNAbin&quot;', or a named list of the same if 'split = TRUE&ldquo;
</p>


<h3>References</h3>

<p>Rabosky, D. L., Chang, J., Title, P. O., Cowman, P. F., Sallan, L., Friedman, M., Kashner, K., Garilao, C., Near, T. J., Coll, M., Alfaro, M. E. (2018). An inverse latitudinal gradient in speciation rate for marine fishes. Nature, 559(7714), 392–395. doi:10.1038/s41586-018-0273-1
</p>


<h3>See Also</h3>

<p><a href="ape.html#topic+DNAbin">DNAbin</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
surgeon_dna &lt;- fishtree_alignment(rank = "Acanthuridae", split = TRUE)
surgeon_dna[[1]]
par(mfrow = c(9, 3), mar = c(0.5, 0.5, 1, 0.5), xaxt = "n", yaxt = "n")
for (gene in names(surgeon_dna)) {
  image(surgeon_dna[[gene]], legend = FALSE, show.labels = FALSE)
  title(gene)
}

## End(Not run)
</code></pre>

<hr>
<h2 id='fishtree_complete_phylogeny'>Get complete (stochastically-resolved) phylogenies from the Fish Tree of Life</h2><span id='topic+fishtree_complete_phylogeny'></span>

<h3>Description</h3>

<p>Retrieves a complete phylogeny generated by stochastic polytomy resolution via the Fish Tree of Life API. If neither 'species' nor 'rank' are specified, returns the entire phylogeny. See Rabosky et al. (2018) and Chang et al. (2019) for details on how these phylogenies were built using stochastic polytomy resolution. WARNING: These phylogenies should generally not be used for downstream analyses of trait evolution. See Rabosky (2015) for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fishtree_complete_phylogeny(
  species,
  rank,
  mc.cores = getOption("mc.cores", 1L)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fishtree_complete_phylogeny_+3A_species">species</code></td>
<td>
<p>(Optionally) subset the results based on a vector of species names.</p>
</td></tr>
<tr><td><code id="fishtree_complete_phylogeny_+3A_rank">rank</code></td>
<td>
<p>(Optionally) subset the results based on the supplied taxonomic rank.</p>
</td></tr>
<tr><td><code id="fishtree_complete_phylogeny_+3A_mc.cores">mc.cores</code></td>
<td>
<p>Number of cores to use in <a href="parallel.html#topic+mclapply">mclapply</a> when subsetting the tree (default '1')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class '&quot;multiPhylo&quot;' that should probably not be used for analyses of trait evolution, including (but not limited to) <a href="ape.html#topic+pic">pic</a>, <a href="ape.html#topic+ace">ace</a>, <a href="ape.html#topic+corBrownian">corBrownian</a>, <a href="diversitree.html#topic+make.bisse">make.bisse</a>, or <a href="hisse.html#topic+hisse">hisse</a>.
</p>


<h3>References</h3>

<p>Rabosky, D. L. (2015). No substitute for real data: A cautionary note on the use of phylogenies from birth-death polytomy resolvers for downstream comparative analyses. Evolution, 69(12), 3207–3216. doi:10.1111/evo.12817
</p>
<p>Rabosky, D. L., Chang, J., Title, P. O., Cowman, P. F., Sallan, L., Friedman, M., Kashner, K., Garilao, C., Near, T. J., Coll, M., Alfaro, M. E. (2018). An inverse latitudinal gradient in speciation rate for marine fishes. Nature, 559(7714), 392–395. doi:10.1038/s41586-018-0273-1
</p>
<p>Chang, J., Rabosky, D. L., &amp; Alfaro, M. E. (2019). Estimating diversification rates on incompletely-sampled phylogenies: theoretical concerns and practical solutions. Systematic Biology. doi:10.1093/sysbio/syz081
</p>
<p>Enhanced polytomy resolution strengthens evidence for global gradient in speciation rate for marine fishes. <a href="https://fishtreeoflife.org/rabosky-et-al-2018-update/">https://fishtreeoflife.org/rabosky-et-al-2018-update/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tree &lt;- fishtree_complete_phylogeny(rank = "Acanthuridae")
sampled_tips &lt;- fishtree_phylogeny(rank = "Acanthuridae")$tip.label
all_tips &lt;- tree[[1]]$tip.label
new_tips &lt;- setdiff(all_tips, sampled_tips)
par(mfrow = c(2,2))
for (ii in 1:4) {
  plot(tree[[ii]], show.tip.label = FALSE, no.margin = TRUE)
  ape::tiplabels(pch = 19, col = ifelse(tree[[ii]]$tip.label %in% new_tips, "red", NA))
}

## End(Not run)
</code></pre>

<hr>
<h2 id='fishtree_phylogeny'>Get a phylogeny from the Fish Tree of Life</h2><span id='topic+fishtree_phylogeny'></span>

<h3>Description</h3>

<p>Retrieves a phylogeny via the Fish Tree of Life API. If neither 'species' nor 'rank' are specified, returns the entire phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fishtree_phylogeny(
  species,
  rank,
  type = c("chronogram", "phylogram", "chronogram_mrca", "phylogram_mrca")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fishtree_phylogeny_+3A_species">species</code></td>
<td>
<p>(Optionally) subset the results based on a vector of species names.</p>
</td></tr>
<tr><td><code id="fishtree_phylogeny_+3A_rank">rank</code></td>
<td>
<p>(Optionally) subset the results based on the supplied taxonomic rank.</p>
</td></tr>
<tr><td><code id="fishtree_phylogeny_+3A_type">type</code></td>
<td>
<p>Either '&quot;chronogram&quot;' or '&quot;phylogram&quot;'. A chronogram has branch lengths proportional to units of time, while a phylogram has branch lengths proportional to the amount of character change. When retrieving a phylogeny by rank, and that rank is not recovered as monophyletic, acceptable types also include '&quot;chronogram_mrca&quot;' and '&quot;phylogram_mrca&quot;', which returns a tree with *all* species descending from the common ancestor of species in the specified rank.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For maximum interoperability, 'species' considers spaces and underscores equivalently. Internally, the phylogenies use underscores.
</p>


<h3>Value</h3>

<p>An object of class '&quot;phylo&quot;'.
</p>


<h3>References</h3>

<p>Rabosky, D. L., Chang, J., Title, P. O., Cowman, P. F., Sallan, L., Friedman, M., Kashner, K., Garilao, C., Near, T. J., Coll, M., Alfaro, M. E. (2018). An inverse latitudinal gradient in speciation rate for marine fishes. Nature, 559(7714), 392–395. doi:10.1038/s41586-018-0273-1
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fishtree_rogues">fishtree_rogues</a></code>, <code><a href="ape.html#topic+read.tree">read.tree</a></code>, <code><a href="phytools.html#topic+force.ultrametric">force.ultrametric</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get a phylogeny for a taxonomic rank
surgeons &lt;- fishtree_phylogeny(rank = "Acanthuridae")

# Get a phylogeny for only certain species
genomic_fish &lt;- c("Oryzias latipes", "Tetraodon nigroviridis",
                  "Gasterosteus aculeatus", "Danio rerio")
fishtree_phylogeny(species = genomic_fish)

# Chronograms may not be ultrametric due to numerical precision issues
# Consider using phytools::force.ultrametric
ape::is.ultrametric(surgeons)
ape::is.ultrametric(surgeons, tol = 0.00001)

# Difference between MRCA trees and regular trees
gobies_mrca &lt;- fishtree_phylogeny(rank = "Gobiidae", type = "chronogram_mrca")
gobies &lt;- fishtree_phylogeny(rank = "Gobiidae", type = "chronogram")
# MRCA trees will have more tips for non-monophyletic groups
length(gobies_mrca$tip.label) &gt; length(gobies$tip.label)
# Drop rogue tips in the MRCA tree
rogue_gobies &lt;- fishtree_rogues("Gobiidae")
pruned_gobies &lt;- ape::drop.tip(gobies_mrca, rogue_gobies)
# Now the trees are identical
setequal(gobies$tip.label, pruned_gobies$tip.label)

## End(Not run)
</code></pre>

<hr>
<h2 id='fishtree_rogues'>Get rogue taxa that break the monophyly of defined taxa</h2><span id='topic+fishtree_rogues'></span>

<h3>Description</h3>

<p>For groups that were recovered as paraphyletic in the phylogenetic analysis,
uses the Fish Tree of Life API to identify which species prevented that clade
from being recovered as monophyletic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fishtree_rogues(rank)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fishtree_rogues_+3A_rank">rank</code></td>
<td>
<p>the (possibly paraphyletic) rank for which rogue or intruder species should be identified.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of species names, potentially empty.
</p>


<h3>References</h3>

<p>Rabosky, D. L., Chang, J., Title, P. O., Cowman, P. F., Sallan, L., Friedman, M., Kashner, K., Garilao, C., Near, T. J., Coll, M., Alfaro, M. E. (2018). An inverse latitudinal gradient in speciation rate for marine fishes. Nature, 559(7714), 392–395. doi:10.1038/s41586-018-0273-1
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
fishtree_rogues("Gobiidae")   # several rogue taxa!
fishtree_rogues("Labridae")   # nice and monophlyetic

## End(Not run)
</code></pre>

<hr>
<h2 id='fishtree_taxonomy'>Get taxonomies and other data from the Fish Tree of Life</h2><span id='topic+fishtree_taxonomy'></span>

<h3>Description</h3>

<p>Retrieves taxonomic and other information from the Fish Tree of Life API.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fishtree_taxonomy(ranks = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fishtree_taxonomy_+3A_ranks">ranks</code></td>
<td>
<p>One or more taxonomic ranks to retrieve.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list, with components containing data on the specified taxa. If 'ranks'
is unspecified, a data frame with all valid taxa is returned instead.
</p>


<h3>References</h3>

<p>Rabosky, D. L., Chang, J., Title, P. O., Cowman, P. F., Sallan, L., Friedman, M., Kashner, K., Garilao, C., Near, T. J., Coll, M., Alfaro, M. E. (2018). An inverse latitudinal gradient in speciation rate for marine fishes. Nature, 559(7714), 392–395. doi:10.1038/s41586-018-0273-1
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tax &lt;- fishtree_taxonomy(rank = "Labridae")
n_total &lt;- length(tax$Labridae$species)
n_sampl &lt;- length(tax$Labridae$sampled_species)
paste("There are", n_sampl, "sampled species out of", n_total, "in wrasses.")

## End(Not run)
</code></pre>

<hr>
<h2 id='fishtree_tip_rates'>Get tip rates for the Fish Tree of Life</h2><span id='topic+fishtree_tip_rates'></span>

<h3>Description</h3>

<p>Downloads tip rates for the entire Fish Tree of Life, or for a specified subset. Tip rates can be thought of as an
instantaneous speciation or extinction rate; for example, a higher tip-specific speciation rate might imply that
a lineage is more likely to split a new lineage at the present time. See Title (2019) in references for details.
If neither 'species' nor 'rank' are specified, returns the entire set of tip-specific diversification rates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fishtree_tip_rates(species, rank, sampled_only = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fishtree_tip_rates_+3A_species">species</code></td>
<td>
<p>(Optionally) subset the results based on a vector of species names.</p>
</td></tr>
<tr><td><code id="fishtree_tip_rates_+3A_rank">rank</code></td>
<td>
<p>(Optionally) subset the results based on the supplied taxonomic rank.</p>
</td></tr>
<tr><td><code id="fishtree_tip_rates_+3A_sampled_only">sampled_only</code></td>
<td>
<p>Restricts the returned dataset to only those species that have genetic data available. Defaults to 'TRUE'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame. Columns ending with '.tv' indicate time-variable BAMM runs; those ending in '.tc' are time-constant runs. The 'dr' column refers to the DR statistic, while 'lambda' and 'mu' are speciation and extinction, respectively.
</p>


<h3>References</h3>

<p>DR rates (supplement, section 1.2.2): Jetz, W., Thomas, G. H., Joy, J. B., Hartmann, K., &amp; Mooers, A. O. (2012). The global diversity of birds in space and time. Nature, 491(7424), 444–448. doi:10.1038/nature11631
</p>
<p>Interpreting tip rates: Title, P. O., &amp; Rabosky, D. L. (2019). Tip rates, phylogenies and diversification: What are we estimating, and how good are the estimates? Methods in Ecology and Evolution, 10(6), 821–834. doi:10.1111/2041-210x.13153
</p>
<p>BAMM rates: Rabosky, D. L. (2014). Automatic Detection of Key Innovations, Rate Shifts, and Diversity-Dependence on Phylogenetic Trees. PLoS ONE, 9(2), e89543. doi:10.1371/journal.pone.0089543
</p>
<p>Rabosky, D. L., Chang, J., Title, P. O., Cowman, P. F., Sallan, L., Friedman, M., Kashner, K., Garilao, C., Near, T. J., Coll, M., Alfaro, M. E. (2018). An inverse latitudinal gradient in speciation rate for marine fishes. Nature, 559(7714), 392–395. doi:10.1038/s41586-018-0273-1
</p>
<p>Enhanced polytomy resolution strengthens evidence for global gradient in speciation rate for marine fishes. <a href="https://fishtreeoflife.org/rabosky-et-al-2018-update/">https://fishtreeoflife.org/rabosky-et-al-2018-update/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get cichlid rates and trees
rates &lt;- fishtree_tip_rates(rank = "Cichlidae")
tree &lt;- fishtree_phylogeny(rank = "Cichlidae")

# Plot tree and extract plotting data
plot(tree, show.tip.label = FALSE)
obj &lt;- get("last_plot.phylo", ape::.PlotPhyloEnv)

# Generate a color ramp
ramp &lt;- grDevices::colorRamp(c("black", "red"), bias = 10)
tiporder &lt;- match(rates$species, gsub("_", " ", tree$tip.label))
scaled_rates &lt;- rates$lambda.tv / max(rates$lambda.tv, na.rm = TRUE)
tipcols &lt;- apply(ramp(scaled_rates), 1, function(x) do.call(rgb, as.list(x / 255)))

# Place colored bars
for (ii in 1:length(tiporder)) {
    tip &lt;- tiporder[ii]
    lines(x = c(obj$xx[tip] + 0.5, obj$xx[tip] + 0.5 + scaled_rates[ii]),
          y = rep(obj$yy[tip], 2),
          col = tipcols[ii])
}

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
