<!DOCTYPE html><html lang="en"><head><title>Help for package FLASHMM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FLASHMM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#contrast.matrix'><p>Construct Contrast Matrix</p></a></li>
<li><a href='#lmm'><p>Fitting Linear Mixed-effects Models</p></a></li>
<li><a href='#lmmfit'><p>Fitting Linear Mixed-effects Models</p></a></li>
<li><a href='#lmmtest'><p>Testing Fixed Effects and Contrasts of the Fixed Effects</p></a></li>
<li><a href='#simuRNAseq'><p>Simulating Multi-sample Multi-cell-type scRNA-seq Dataset based on Negative Binomial Distribution</p></a></li>
<li><a href='#sslmm'><p>Computing Summary-level Data from Individual-level Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Fast and Scalable Single Cell Differential Expression Analysis
using Mixed-Effects Models</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A fast and scalable linear mixed-effects model (LMM) estimation algorithm 
    for analysis of single-cell differential expression. The algorithm uses 
    summary-level statistics and requires less computer memory to fit the LMM. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, MASS, Matrix</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, bookdown, rmarkdown, devtools, BiocManager,
SingleCellExperiment, ExperimentHub</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/BaderLab/FLASHMM">https://github.com/BaderLab/FLASHMM</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/BaderLab/FLASHMM/issues">https://github.com/BaderLab/FLASHMM/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-11 21:13:07 UTC; changjiangxu</td>
</tr>
<tr>
<td>Author:</td>
<td>Changjiang Xu [aut, cre],
  Gary Bader [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Changjiang Xu &lt;changjiang.xu@utoronto.ca&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-11 21:30:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='contrast.matrix'>Construct Contrast Matrix</h2><span id='topic+contrast.matrix'></span>

<h3>Description</h3>

<p>Construct the contrast matrix to make various comparsions of different treatments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contrast.matrix(contrast, model.matrix.names)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="contrast.matrix_+3A_contrast">contrast</code></td>
<td>
<p>A vector of character strings specifying the various comparisons, which are the expressions constituted by model.matrix.names.</p>
</td></tr>
<tr><td><code id="contrast.matrix_+3A_model.matrix.names">model.matrix.names</code></td>
<td>
<p>Column names of model (design) matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix which columns correspond to contrasts.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>model_variables &lt;- c("A", "B", "C", "D")
contrast &lt;- c("AvsB" = "A-B", "AvsC" = "A-C", 'AvsB.C.D'= "A-(B+C+D)/3")
contrast.matrix(contrast, model_variables)

</code></pre>

<hr>
<h2 id='lmm'>Fitting Linear Mixed-effects Models</h2><span id='topic+lmm'></span>

<h3>Description</h3>

<p>lmm is used to fit linear mixed-effects models (LMM) based on summary-level data. The LMM parameters are estimated by restricted maximum likelihood (REML) with Fisher scoring (FS) gradient descent algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lmm(
  XX,
  XY,
  ZX,
  ZY,
  ZZ,
  Ynorm,
  n,
  d = ncol(ZZ),
  theta0 = NULL,
  method = "REML-FS",
  max.iter = 50,
  epsilon = 1e-05,
  output.cov = TRUE,
  output.RE = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lmm_+3A_xx">XX</code></td>
<td>
<p>t(X)%*%X, X is a design matrix for fixed effects.</p>
</td></tr>
<tr><td><code id="lmm_+3A_xy">XY</code></td>
<td>
<p>t(Y%*%X), Y is a features-by-samples matrix of observed responses (genes-by-cells expression matrix for scRNA-seq).</p>
</td></tr>
<tr><td><code id="lmm_+3A_zx">ZX</code></td>
<td>
<p>t(Z)%*%X, Z = [Z1, ..., Zk],  a design matrix for k random factors (variables).</p>
</td></tr>
<tr><td><code id="lmm_+3A_zy">ZY</code></td>
<td>
<p>t(Y%*%Z).</p>
</td></tr>
<tr><td><code id="lmm_+3A_zz">ZZ</code></td>
<td>
<p>t(Z)%*%Z.</p>
</td></tr>
<tr><td><code id="lmm_+3A_ynorm">Ynorm</code></td>
<td>
<p>Norms for features (each row in Y), that is, Ynorm = rowSums(Y*Y).</p>
</td></tr>
<tr><td><code id="lmm_+3A_n">n</code></td>
<td>
<p>Numbers of samples (cells in scRNA-seq), nrow(X).</p>
</td></tr>
<tr><td><code id="lmm_+3A_d">d</code></td>
<td>
<p>A vector of (m1,...,mk), mi = ncol(Zi), number of columns in Zi. m1 + ... + mk = ncol(Z), number of columns in Z.</p>
</td></tr>
<tr><td><code id="lmm_+3A_theta0">theta0</code></td>
<td>
<p>A vector of initial values of the variance components, (s1, ...,sk, s_(k+1)), si = sigma_i^2, the variance component of the i-th type random effects. s_(k+1) = sigma^2, the variance component of model residual error.</p>
</td></tr>
<tr><td><code id="lmm_+3A_method">method</code></td>
<td>
<p>The REML with Fisher scoring (FS) iterative algorithm, REML-FS.</p>
</td></tr>
<tr><td><code id="lmm_+3A_max.iter">max.iter</code></td>
<td>
<p>The maximal number of iterations for the iterative algorithm.</p>
</td></tr>
<tr><td><code id="lmm_+3A_epsilon">epsilon</code></td>
<td>
<p>Positive convergence tolerance. If the absolute value of the first partial derivative of log likelihood is less than epsilon, the iterations converge.</p>
</td></tr>
<tr><td><code id="lmm_+3A_output.cov">output.cov</code></td>
<td>
<p>If TRUE, output the covariance matrices for the estimated coefficients, which are needed for testing contrasts.</p>
</td></tr>
<tr><td><code id="lmm_+3A_output.re">output.RE</code></td>
<td>
<p>If TRUE, output the best linear unbiased prediction (BLUP) of the random effects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<p>dlogL First partial derivatives of log-likelihoods for each feature (gene).
</p>
<p>niter Nmbers of iterations for each feature (gene).
</p>
<p>coef A matrix of estimated coefficients (fixed effects), each column corresponds to a feature (gene) and each row one covariate.
</p>
<p>se A matrix of the standard errors of the estimated coefficients.
</p>
<p>t A matrix of t-values for the fixed effects, equal to coef/se.
</p>
<p>df Degrees of freedom.
</p>
<p>p A matrix of two-sided p-values for the fixed effects.
</p>
<p>cov A array of covariance matrices of the estimated coefficients (fixed effects).
</p>
<p>theta A matrix of the estimated variance components, each column corresponds to a feature (gene) and each row one variance component. The last row is the variance component of the residual error.
</p>
<p>se.theta Standard errors of the estimated theta.
</p>
<p>RE A matrix of the best linear unbiased prediction (BLUP) of random effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Generate data: X, Y, and Z.
set.seed(2024)

n &lt;- 1e3
m &lt;- 10
Y &lt;- matrix(rnorm(m*n), m, n)
rownames(Y) &lt;- paste0("Gene", 1:nrow(Y))

trt &lt;- sample(c("A", "B"), n, replace = TRUE)
X &lt;- model.matrix(~ 0 + trt)

q &lt;- 20
sam &lt;- rep(NA, n)
sam[trt == "A"] &lt;- paste0("A", sample.int(q/2, sum(trt == "A"), replace = TRUE))
sam[trt == "B"] &lt;- paste0("B", sample.int(q/2, sum(trt == "B"), replace = TRUE))
Z &lt;- model.matrix(~ 0 + sam)
d &lt;- ncol(Z)

#Fit LMM by summary-level data
#Compute and store the summary-level data:
n &lt;- nrow(X)
XX &lt;- t(X)%*%X
XY &lt;- t(Y%*%X)
ZX &lt;- t(Z)%*%X
ZY &lt;- t(Y%*%Z)
ZZ &lt;- t(Z)%*%Z
Ynorm &lt;- rowSums(Y*Y)
fit &lt;- lmm(XX, XY, ZX, ZY, ZZ, Ynorm = Ynorm, n = n, d = d)
str(fit)

</code></pre>

<hr>
<h2 id='lmmfit'>Fitting Linear Mixed-effects Models</h2><span id='topic+lmmfit'></span>

<h3>Description</h3>

<p>lmmfit, a wrapper function of lmm, fits linear mixed-effects models (LMM) by sample-level data. The LMM parameters are estimated by restricted maximum likelihood (REML) with Fisher scoring (FS) gradient descent algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lmmfit(
  Y,
  X,
  Z,
  d,
  theta0 = NULL,
  nBlocks = ceiling(nrow(Y) * ncol(Y) * 1e-08),
  method = "REML-FS",
  max.iter = 50,
  epsilon = 1e-05,
  output.cov = TRUE,
  output.RE = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lmmfit_+3A_y">Y</code></td>
<td>
<p>A features-by-samples matrix of responses (genes-by-cells matrix of gene expressions for scRNA-seq).</p>
</td></tr>
<tr><td><code id="lmmfit_+3A_x">X</code></td>
<td>
<p>A design matrix for fixed effects, with rows corresponding to the columns of Y.</p>
</td></tr>
<tr><td><code id="lmmfit_+3A_z">Z</code></td>
<td>
<p>A design matrix for random effects, with rows corresponding to the columns of Y. Z = [Z1, ..., Zk], and Zi, i=1,...,k, is the design matrix for the i-th type random factor.</p>
</td></tr>
<tr><td><code id="lmmfit_+3A_d">d</code></td>
<td>
<p>A vector of (m1,...,mk), mi = ncol(Zi), number of columns in Zi. m1 + ... + mk = ncol(Z), number of columns in Z.</p>
</td></tr>
<tr><td><code id="lmmfit_+3A_theta0">theta0</code></td>
<td>
<p>A vector of initial values of the variance components, (s1, ...,sk, s_(k+1)), si = sigma_i^2, the variance component of the i-th type random effects. s_(k+1) = sigma^2, the variance component of model residual error.</p>
</td></tr>
<tr><td><code id="lmmfit_+3A_nblocks">nBlocks</code></td>
<td>
<p>Number of blocks, used for blocking a big data to reduce the storage required in computing.</p>
</td></tr>
<tr><td><code id="lmmfit_+3A_method">method</code></td>
<td>
<p>The REML with Fisher scoring (FS) iterative algorithm, REML-FS.</p>
</td></tr>
<tr><td><code id="lmmfit_+3A_max.iter">max.iter</code></td>
<td>
<p>The maximal number of iterations for the iterative algorithm.</p>
</td></tr>
<tr><td><code id="lmmfit_+3A_epsilon">epsilon</code></td>
<td>
<p>Positive convergence tolerance. If the absolute value of the first partial derivative of log likelihood is less than epsilon, the iterations converge.</p>
</td></tr>
<tr><td><code id="lmmfit_+3A_output.cov">output.cov</code></td>
<td>
<p>If TRUE, output the covariance matrices for the estimated coefficients, which are needed for testing contrasts.</p>
</td></tr>
<tr><td><code id="lmmfit_+3A_output.re">output.RE</code></td>
<td>
<p>If TRUE, output the best linear unbiased prediction (BLUP) of the random effects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<p>dlogL First partial derivatives of log-likelihoods for each feature (gene).
</p>
<p>niter Nmbers of iterations for each feature (gene).
</p>
<p>coef A matrix of estimated coefficients (fixed effects), each column corresponds to a feature (gene) and each row one covariate.
</p>
<p>se A matrix of the standard errors of the estimated coefficients.
</p>
<p>t A matrix of t-values for the fixed effects, equal to coef/se.
</p>
<p>df Degrees of freedom.
</p>
<p>p A matrix of two-sided p-values for the fixed effects.
</p>
<p>cov A array of covariance matrices of the estimated coefficients (fixed effects).
</p>
<p>theta A matrix of the estimated variance components, each column corresponds to a feature (gene) and each row one variance component. The last row is the variance component of the residual error.
</p>
<p>se.theta Standard errors of the estimated theta.
</p>
<p>RE A matrix of the best linear unbiased prediction (BLUP) of random effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Generate data: X, Y, and Z.
set.seed(2024)

n &lt;- 1e3
m &lt;- 10
Y &lt;- matrix(rnorm(m*n), m, n)
rownames(Y) &lt;- paste0("Gene", 1:nrow(Y))

trt &lt;- sample(c("A", "B"), n, replace = TRUE)
X &lt;- model.matrix(~ 0 + trt)

q &lt;- 20
sam &lt;- rep(NA, n)
sam[trt == "A"] &lt;- paste0("A", sample.int(q/2, sum(trt == "A"), replace = TRUE))
sam[trt == "B"] &lt;- paste0("B", sample.int(q/2, sum(trt == "B"), replace = TRUE))
Z &lt;- model.matrix(~ 0 + sam)
d &lt;- ncol(Z)

#Fit LMM by the cell-level data
fit &lt;- lmmfit(Y, X, Z, d = d)
str(fit)

#Hypothesis testing
lmmtest(fit)
lmmtest(fit, index = 2)
lmmtest(fit, contrast = cbind("B-A" = c(-1, 1)))

</code></pre>

<hr>
<h2 id='lmmtest'>Testing Fixed Effects and Contrasts of the Fixed Effects</h2><span id='topic+lmmtest'></span>

<h3>Description</h3>

<p>lmmtest is used to test fixed effects or contrasts of fixed effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lmmtest(
  fit,
  index,
  contrast = NULL,
  alternative = c("two.sided", "less", "greater")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lmmtest_+3A_fit">fit</code></td>
<td>
<p>A result of lmmfit/lmm, which contains
coef (estimates of fixed effects), a matrix with rows representing the fixed effects and columns the different response variables in the model,
cov (covariance matrix of the fixed effects), an array of three dimensions for different response variables in the model,
df (residual degree of freedom in the linear model).</p>
</td></tr>
<tr><td><code id="lmmtest_+3A_index">index</code></td>
<td>
<p>A vector of integers or characters indicating which fixed effects are to be tested. By default index consists of all of the fixed effects. Ignored if contrast is not NULL.</p>
</td></tr>
<tr><td><code id="lmmtest_+3A_contrast">contrast</code></td>
<td>
<p>A matrix with columns corresponding to contrasts of the fixed effects to be tested.</p>
</td></tr>
<tr><td><code id="lmmtest_+3A_alternative">alternative</code></td>
<td>
<p>A character string specifying the alternative hypothesis, one of &quot;two.sided&quot;, &quot;greater&quot; or &quot;less&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of coefficients, t-values and p-values, in which the rows correspond to the features (genes) and the columns the fixed effects (covariates). .
</p>

<hr>
<h2 id='simuRNAseq'>Simulating Multi-sample Multi-cell-type scRNA-seq Dataset based on Negative Binomial Distribution</h2><span id='topic+simuRNAseq'></span>

<h3>Description</h3>

<p>simuRNAseq simulates scRNA-seq data with multiple subjects (samples), multiple clusters (cell-types) and two treatments (conditions) based on a negative binomial (NB) distribution using a reference data as background or control.
The reference data consisting of genes-by-cells counts matrix is used to estimate the NB dispersion and means for the genes to be simulated. The simulated genes are randomly selected from the reference data.
The NB dispersion are estimated by the method-of-moments estimate (MME).
The NB means for the background in the control are estimated by the sample mean. The NB means for the differentially expressed (DE) genes are given by the sample mean plus a log-fold change (logFC).
The simulated cells are randomly selected from the meta data that specifies subjects, cell-types and treatments for the cells.
The meta data consists of samples, clusters of cell types, and treatments, which can be generated either from reference data or randomly. If not provided, it will be randomly generated.
A random seed is recommended to be specified by set.seed before simulating data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simuRNAseq(
  counts,
  nGenes = nrow(counts),
  nCells = ncol(counts),
  metadata = NULL,
  samples.nested = TRUE,
  nsam = 25,
  ncls = 10,
  ntrt = 2,
  trt = NULL,
  nDEgenes = ncls,
  nDEgenesType,
  pDEgenesType = NULL,
  adjust.library.size = TRUE,
  direction = c("both", "up", "down"),
  minbeta = 0.25,
  maxbeta = 1,
  var.randomeffects = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simuRNAseq_+3A_counts">counts</code></td>
<td>
<p>A genes-by-cells matrix of reference counts. If missing, counts is generated by a negative binomial distribution.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_ngenes">nGenes</code></td>
<td>
<p>Number of genes to be simulated.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_ncells">nCells</code></td>
<td>
<p>Number of cells to be simulated.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_metadata">metadata</code></td>
<td>
<p>The meta data consisting of 4 columns: sam (sample labels), cls (cluster lables of cell types), trt (treatments or conditions) and libsize (library size or total counts per cell), which is randomly generated if not provided.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_samples.nested">samples.nested</code></td>
<td>
<p>If TRUE, when metadata is not provided, each simulated subject (sample) belongs to only one condition (either treatment or control), that is, the subject is nested in a condition (treatment).</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_nsam">nsam</code></td>
<td>
<p>Number of subjects (individuals).</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_ncls">ncls</code></td>
<td>
<p>Number of clusters (cell-types).</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_ntrt">ntrt</code></td>
<td>
<p>Number of treatments (only one condition is treated).</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_trt">trt</code></td>
<td>
<p>Treatment, specifying which condition is treated.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_ndegenes">nDEgenes</code></td>
<td>
<p>Total number of DE genes.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_ndegenestype">nDEgenesType</code></td>
<td>
<p>Number of DE genes specific to a cell type, named by cell cluster labels.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_pdegenestype">pDEgenesType</code></td>
<td>
<p>Proportion of DE genes in a cell-type. Default NULL means equal proportion.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_adjust.library.size">adjust.library.size</code></td>
<td>
<p>If TRUE, adjust library sizes using the reference counts.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_direction">direction</code></td>
<td>
<p>Specify if the DE genes are up- and/or down-regulated.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_minbeta">minbeta</code></td>
<td>
<p>Lower bound of the DE gene logFC.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_maxbeta">maxbeta</code></td>
<td>
<p>Upper bound of the DE gene logFC. minbeta &lt; maxbeta. If direction = &quot;both&quot;, minbeta*maxbeta &gt; 0. If direction = &quot;up&quot;, minbeta &gt; 0. If direction = &quot;down&quot;, maxbeta &lt; 0.</p>
</td></tr>
<tr><td><code id="simuRNAseq_+3A_var.randomeffects">var.randomeffects</code></td>
<td>
<p>Variance of random effects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ref.mean.dispersion: A data frame of the reference counts' means and dispersion,
</p>
<p>metadata: Meta data consisting of 4 columns: sam (sample labels), cls (cluster lables of cell types), trt (two treatments/conditions) and libsize (library sizes).
</p>
<p>counts: A genes-by-cells matrix of the simulated counts.
</p>
<p>DEgenes: A data frame of DE genes consisting of 3 columns: gene, beta (effect), and cluster to which the gene is specific.
</p>
<p>treatment: The condition treated.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Simulate a multi-sample multi-cell-type scRNA-seq dataset.
set.seed(2412)
dat &lt;- simuRNAseq(nGenes = 50, nCells = 1000, nsam = 25, ncls = 4, ntrt = 2, nDEgenes = 6)
str(dat)
table(dat$metadata[, c("sam", "trt")]) #The samples are nested in a condition.

#Analyze differentially expressed (DE) genes specific to a cell-type using LMM.
Y &lt;- log(dat$counts + 1) #expressions (log-transformed counts)
X &lt;- model.matrix(~ 0 + log(libsize) + cls + cls:trt, data = dat$metadata)
Z &lt;- model.matrix(~ 0 + sam, data = dat$metadata)
d &lt;- ncol(Z)

#Fit LMM using cell-level data.
fit &lt;- lmmfit(Y, X, Z, d = d)

#Fit LMM using summary-level data.
#Compute and store the summary-level data:
n &lt;- nrow(X)
XX &lt;- t(X)%*%X
XY &lt;- t(Y%*%X)
ZX &lt;- t(Z)%*%X
ZY &lt;- t(Y%*%Z)
ZZ &lt;- t(Z)%*%Z
Ynorm &lt;- rowSums(Y*Y)
fitss &lt;- lmm(XX, XY, ZX, ZY, ZZ, Ynorm = Ynorm, n = n, d = d)
identical(fit, fitss)

#Hypothesis testing
test &lt;- lmmtest(fit)
head(test)

#The DE genes specific to a cell-type.
tail(test[, grep(":", colnames(test))])
#The true DE genes
dat$DEgenes

</code></pre>

<hr>
<h2 id='sslmm'>Computing Summary-level Data from Individual-level Data</h2><span id='topic+sslmm'></span>

<h3>Description</h3>

<p>sslmm can be used to compute the correlation-related summary statistics (summary-level data) for lmm function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sslmm(X, Y, Z)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sslmm_+3A_x">X</code></td>
<td>
<p>A design matrix for fixed effects, with rows corresponding to the columns of Y.</p>
</td></tr>
<tr><td><code id="sslmm_+3A_y">Y</code></td>
<td>
<p>A features-by-samples matrix of responses (genes-by-cells matrix of gene expressions for scRNA-seq).</p>
</td></tr>
<tr><td><code id="sslmm_+3A_z">Z</code></td>
<td>
<p>A design matrix for random effects, with rows corresponding to the columns of Y.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of summary statistics:
XX = t(X)%*%X,
XY = t(X)%*%t(Y),
ZX = t(Z)%*%X,
ZY = t(Z)%*%t(Y),
ZZ = t(Z)%*%Z,
Ynorm = rowSums(Y*Y) and n = nrow(X).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n &lt;- 1e3
set.seed(2024)
p &lt;- 2
X &lt;- matrix(rnorm(p*n), n, p)
colnames(X) &lt;- paste0("X", 1:ncol(X))
m &lt;- 3
Y &lt;- matrix(rnorm(m*n), m, n)
rownames(Y) &lt;- paste0("Y", 1:nrow(Y))
q &lt;- 4
Z &lt;- gl(q, n/q, labels = letters[1:q])
Z &lt;- model.matrix(~ 0 + Z)
sslmm(X, Y, Z)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
