<!DOCTYPE html><html lang="en"><head><title>Help for package Fluidigm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Fluidigm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Fluidigm'><p>Fluidigm</p></a></li>
<li><a href='#calculatePairwiseSimilarities'><p>Run plink to Calculate Pairwise Similarities</p></a></li>
<li><a href='#estimateErrors'><p>Estimate Errors in 'PLINK&rdquo; ped files</p></a></li>
<li><a href='#fluidigm2PLINK'><p>Convert Fluidigm Output to 'PLINK' Format</p></a></li>
<li><a href='#fluidigmAnalysisWrapper'><p>Run the Fluidigm Analysis Script Together</p></a></li>
<li><a href='#getPairwiseSimilarityLoci'><p>Get Pairwise Similarity Loci</p></a></li>
<li><a href='#similarityMatrix'><p>Calculate the similarity matrix</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Handling Fluidigm Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Designed to streamline the process of analyzing genotyping data from Fluidigm machines, this package offers a suite of tools for data handling and analysis. It includes functions for converting Fluidigm data to format used by 'PLINK', estimating errors, calculating pairwise similarities, determining pairwise similarity loci, and generating a similarity matrix.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>perl (&gt;= 5.10.0), plink (&gt;=1.07)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table (&ge; 1.9.4), graphics, grDevices, reshape (&ge;
0.8.8), stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat, digest</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-11 15:50:51 UTC; ejo138</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel Fischer <a href="https://orcid.org/0000-0002-7513-683X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Helena Johansson [ctb, fnd],
  Robert Ekblom <a href="https://orcid.org/0000-0003-2222-1966"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel Fischer &lt;Daniel.Fischer@luke.fi&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-12 19:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Fluidigm'>Fluidigm</h2><span id='topic+Fluidigm'></span><span id='topic+Fluidigm-package'></span>

<h3>Description</h3>

<p>Designed to streamline the process of analyzing genotyping data from Fluidigm machines, this package offers a suite of tools for data handling and analysis. It includes functions for converting Fluidigm data to format used by 'PLINK', estimating errors, calculating pairwise similarities, determining pairwise similarity loci, and generating a similarity matrix.
</p>


<h3>Title</h3>

<p>Comprehensive Analysis of Fluidigm Genotyping Data
</p>


<h3>Description</h3>

<p>A suite of tools designed to streamline the process of analyzing genotyping data from Fluidigm machines.
It includes functions for converting Fluidigm data to 'PLINK' format, estimating errors, calculating pairwise
similarities, determining pairwise similarity loci, and generating a similarity matrix.
</p>


<h3>Details</h3>

<p>The package provides a comprehensive analysis pipeline for Fluidigm genotyping data. It starts by converting
the raw data from the Fluidigm machine into a format that can be used with the 'PLINK' software. It then estimates
errors in the data, calculates pairwise similarities between genotypes, determines pairwise similarity loci,
and generates a similarity matrix. The package is designed to make it easier and more efficient for researchers
to extract meaningful insights from their genotyping studies.
</p>


<h3>References</h3>


<ul>
<li><p>Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender D, Maller J, Sklar P, de Bakker PIW, Daly MJ &amp; Sham PC (2007)
<em>PLINK: a toolset for whole-genome association and population-based linkage analysis.</em> American Journal of Human Genetics, 81.
</p>
</li>
<li><p>Purcell, Shaun. PLINK. <a href="https://zzz.bwh.harvard.edu/plink/">https://zzz.bwh.harvard.edu/plink/</a></p>
</li></ul>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Daniel Fischer <a href="mailto:Daniel.Fischer@luke.fi">Daniel.Fischer@luke.fi</a> (<a href="https://orcid.org/0000-0002-7513-683X">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Robert Ekblom (<a href="https://orcid.org/0000-0003-2222-1966">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Helena Johansson [contributor, funder]
</p>
</li></ul>


<hr>
<h2 id='calculatePairwiseSimilarities'>Run plink to Calculate Pairwise Similarities</h2><span id='topic+calculatePairwiseSimilarities'></span>

<h3>Description</h3>

<p>This function serves as a wrapper to the 'PLINK' software, which is a free, open-source whole genome association analysis toolset.
It specifically uses 'PLINK' to calculate pairwise similarities between genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculatePairwiseSimilarities(
  file,
  db = NA,
  map = NA,
  out = NA,
  sexing = TRUE,
  verbose = TRUE,
  verbosity = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculatePairwiseSimilarities_+3A_file">file</code></td>
<td>
<p>A string representing the path to the filtered ped/map file pair (without ped/map file extension).</p>
</td></tr>
<tr><td><code id="calculatePairwiseSimilarities_+3A_db">db</code></td>
<td>
<p>A string representing the path to an existing genotype database. If not provided, the function will proceed with the existing data.</p>
</td></tr>
<tr><td><code id="calculatePairwiseSimilarities_+3A_map">map</code></td>
<td>
<p>A string representing the filepath to PlateDnoY.map file. If not provided, the function will use the map file with the same name as the ped file.</p>
</td></tr>
<tr><td><code id="calculatePairwiseSimilarities_+3A_out">out</code></td>
<td>
<p>A string representing the path to the output. If not provided, the output will be written to a file with the same name as the input file, appended with &quot;_oDB&quot;.</p>
</td></tr>
<tr><td><code id="calculatePairwiseSimilarities_+3A_sexing">sexing</code></td>
<td>
<p>A logical value indicating whether the function should try to perform sexing. Default is FALSE.</p>
</td></tr>
<tr><td><code id="calculatePairwiseSimilarities_+3A_verbose">verbose</code></td>
<td>
<p>A logical value indicating whether the output should be verbose. Default is TRUE.</p>
</td></tr>
<tr><td><code id="calculatePairwiseSimilarities_+3A_verbosity">verbosity</code></td>
<td>
<p>An integer representing the level of verbosity. Set to a higher number for more detailed output. Default is 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function first checks the input parameters and sets default values if necessary. It then constructs and executes a PLINK command to merge
the genotype output with the existing genotype database if one is provided. Finally, it calculates pairwise similarities for all samples (and
database individuals) using another PLINK command. If the 'sexing' parameter is set to TRUE, the function will also attempt to determine the
sex of the individuals.
</p>


<h3>Value</h3>

<p>A list containing the following elements:
gensim, a matrix indicating if genotypes are called correctly for replicates and/or if genotypes are missing
summs, a matrix with summary statistics
</p>


<h3>References</h3>


<ul>
<li><p>Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender D, Maller J, Sklar P, de Bakker PIW, Daly MJ &amp; Sham PC (2007)
<em>PLINK: a toolset for whole-genome association and population-based linkage analysis.</em> American Journal of Human Genetics, 81.
</p>
</li>
<li><p>Purcell, Shaun. PLINK. <a href="https://zzz.bwh.harvard.edu/plink/">https://zzz.bwh.harvard.edu/plink/</a></p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

  outdir &lt;- tempdir()

  calculatePairwiseSimilarities(file=file.path(outdir, "example_data.csv.GOOD"))

## End(Not run)

</code></pre>

<hr>
<h2 id='estimateErrors'>Estimate Errors in 'PLINK&rdquo; ped files</h2><span id='topic+estimateErrors'></span>

<h3>Description</h3>

<p>This function processes the 'PLINK' ped files and estimates errors. It also performs sex assignment and species marker analysis if required.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimateErrors(
  file,
  outdir = NA,
  db = NA,
  appendSamplesToDB = FALSE,
  keep.rep = 1,
  y.marker = NA,
  x.marker = NA,
  sp.marker = NA,
  plots = TRUE,
  neg_controls = NA,
  allele_error = 5,
  marker_dropout = 15,
  no_marker = 50,
  male.y = 3,
  male.hetX = 0,
  female.y = 0,
  female.Xtot = 8,
  female.hetXtot = 3,
  warning.noYtot = 2,
  warning.noHetXtot = 3,
  sexing = FALSE,
  verbose = TRUE,
  verbosity = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimateErrors_+3A_file">file</code></td>
<td>
<p>A string. Path to the ped input file.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_outdir">outdir</code></td>
<td>
<p>A string specifying the output folder. If left empty the original folder path of the input file will be used.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_db">db</code></td>
<td>
<p>A string. Name of the used database. Default is NA.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_appendsamplestodb">appendSamplesToDB</code></td>
<td>
<p>A logical. Should new samples be added to database? Default is FALSE.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_keep.rep">keep.rep</code></td>
<td>
<p>A numeric. Keep only this n-fold replicates, default n=1.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_y.marker">y.marker</code></td>
<td>
<p>A vector. Y markers for sexing. Default is NA.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_x.marker">x.marker</code></td>
<td>
<p>A vector. X markers for sexing. Default is NA.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_sp.marker">sp.marker</code></td>
<td>
<p>A vector. Markers used for species-identification. Default is NA.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_plots">plots</code></td>
<td>
<p>A logical. Should plots be created? Default is TRUE.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_neg_controls">neg_controls</code></td>
<td>
<p>A vector. Names of negative controls. Default is NA.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_allele_error">allele_error</code></td>
<td>
<p>A numeric. Threshold for RERUN on Allele errors. Default is 5.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_marker_dropout">marker_dropout</code></td>
<td>
<p>A numeric. Threshold for RERUN on Marker dropout. Default is 15.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_no_marker">no_marker</code></td>
<td>
<p>A numeric. Number of markers. Default is 50.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_male.y">male.y</code></td>
<td>
<p>A numeric. Threshold for sexing, male y-chromosome markers. Default is 3.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_male.hetx">male.hetX</code></td>
<td>
<p>A numeric. Threshold for sexing, heterozygote x-chr markers. Default is 0.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_female.y">female.y</code></td>
<td>
<p>A numeric. Threshold for sexing, female y-chromosome markers. Default is 0.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_female.xtot">female.Xtot</code></td>
<td>
<p>A numeric. Threshold for sexing, total female x-chr markers. Default is 8.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_female.hetxtot">female.hetXtot</code></td>
<td>
<p>A numeric. Threshold for sexing, heterozygote x-chr markers. Default is 3.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_warning.noytot">warning.noYtot</code></td>
<td>
<p>A numeric. Threshold for sexing, when should warning be triggered. Default is 2.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_warning.nohetxtot">warning.noHetXtot</code></td>
<td>
<p>A numeric. Threshold for sexing, when should warning be triggered. Default is 3.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_sexing">sexing</code></td>
<td>
<p>A logical. Should sexing be performed? Default is FALSE.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_verbose">verbose</code></td>
<td>
<p>A logical or numeric. Should the output be verbose? Default is TRUE.</p>
</td></tr>
<tr><td><code id="estimateErrors_+3A_verbosity">verbosity</code></td>
<td>
<p>A numeric. Level of verbosity, set to higher number for more details. Default is 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function processes the PLINK ped files and estimates errors. It checks if the first and second run of each sample have the same genotype and if both
replicates are identical. It also performs sex assignment based on the provided Y and X markers. If species marker is provided, it performs species marker
analysis. The function creates a consensus PED for all &quot;GOOD&quot; samples and exports it along with a .map file without the Y-markers. It also creates a
database file, if provided.
</p>


<h3>Value</h3>

<p>A list containing the following elements:
gensim, a matrix indicating if genotypes are called correctly for replicates and/or if genotypes are missing
summs, a matrix with summary statistics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
    # outdir is here the output directory from the fluidigm2PLINK function

    # Estimate the errors with sexing applied
    estimateErrors(file=file.path(outdir, "example_data.csv.ped"),
                   keep.rep = 2)

    # Estimate the errors and apply sexing with y and x markers defined
    estimateErrors(file=file.path(outdir, "example_data.csv.ped"),
                   keep.rep = 2,
                   sexing=TRUE,
                   y.marker = c("Y_scaffoldY158711_762",
                                "Y_scaffoldY42647_3017",
                                "Y_scaffoldY42656_3986"),
                   x.marker = c("X_scaffold11905_7659",
                                "X_scaffold17088_4621",
                                "X_scaffold1915_14108",
                                "X_scaffold4825_648",
                                "X_scaffold5374_1437",
                                "X_scaffold10171:3154"))

## End(Not run)
</code></pre>

<hr>
<h2 id='fluidigm2PLINK'>Convert Fluidigm Output to 'PLINK' Format</h2><span id='topic+fluidigm2PLINK'></span>

<h3>Description</h3>

<p>This function takes a Fluidigm output file and converts it into a 'PLINK' format. The 'PLINK' ped/map format is a widely used genetic
variation data format. This function is useful for researchers who want to analyze Fluidigm data using tools that accept
'PLINK' format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fluidigm2PLINK(
  file = NA,
  map = NA,
  out = NA,
  outdir = NA,
  plots = TRUE,
  rearrange = TRUE,
  missing.geno = "0 0",
  fixNames = TRUE,
  overwrite = FALSE,
  verbose = TRUE,
  verbosity = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fluidigm2PLINK_+3A_file">file</code></td>
<td>
<p>A string specifying the path to the input file in CSV format.</p>
</td></tr>
<tr><td><code id="fluidigm2PLINK_+3A_map">map</code></td>
<td>
<p>A string specifying the filepath to the map-file that should be used.</p>
</td></tr>
<tr><td><code id="fluidigm2PLINK_+3A_out">out</code></td>
<td>
<p>A string specifying the output file name. If left empty, the original basename of the input file will be used.</p>
</td></tr>
<tr><td><code id="fluidigm2PLINK_+3A_outdir">outdir</code></td>
<td>
<p>A string specifying the output folder. If left empty the original folder path of the input file will be used.</p>
</td></tr>
<tr><td><code id="fluidigm2PLINK_+3A_plots">plots</code></td>
<td>
<p>A logical indicating whether additional figures for conversion should be plotted. Default is TRUE.</p>
</td></tr>
<tr><td><code id="fluidigm2PLINK_+3A_rearrange">rearrange</code></td>
<td>
<p>A logical indicating whether the ped/map output should be rearranged in order of provided map file. Default is TRUE.</p>
</td></tr>
<tr><td><code id="fluidigm2PLINK_+3A_missing.geno">missing.geno</code></td>
<td>
<p>A character string specifying how missing values should be coded. Default is &quot;0 0&quot;.</p>
</td></tr>
<tr><td><code id="fluidigm2PLINK_+3A_fixnames">fixNames</code></td>
<td>
<p>A logical indicating whether whitespaces from sample names should be automatically removed. Default is TRUE.</p>
</td></tr>
<tr><td><code id="fluidigm2PLINK_+3A_overwrite">overwrite</code></td>
<td>
<p>A logical indicating wheter the original map file should be overwritten or not. Default FALSE</p>
</td></tr>
<tr><td><code id="fluidigm2PLINK_+3A_verbose">verbose</code></td>
<td>
<p>A logical or numerical value indicating whether the output should be verbose. Default is TRUE.</p>
</td></tr>
<tr><td><code id="fluidigm2PLINK_+3A_verbosity">verbosity</code></td>
<td>
<p>A numerical value indicating the level of verbosity. Set to a higher number for more details. Default is 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function first checks the input parameters and then imports the Fluidigm data from the CSV file. It creates a new MAP file
based on the information provided in the given map file. The function then creates a PED file and exports both files.
If requested, the function also generates plots for genotyping success and additional summary statistics.
</p>
<p>This function uses the PLINK software. For more information about PLINK, please refer to the official documentation.
</p>


<h3>Value</h3>

<p>A ped/map file pair and optional diagnostic plots.
</p>


<h3>References</h3>

<p>PLINK: Whole genome data analysis toolset - Harvard University
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  file_path_csv &lt;- system.file("extdata", "example_data.csv", package = "Fluidigm")
  file_path_map &lt;- system.file("extdata", "example_data_withY.map", package = "Fluidigm")
  outdir &lt;- tempdir()

  fluidigm2PLINK(file=file_path_csv, map=file_path_map, outdir=outdir)

</code></pre>

<hr>
<h2 id='fluidigmAnalysisWrapper'>Run the Fluidigm Analysis Script Together</h2><span id='topic+fluidigmAnalysisWrapper'></span>

<h3>Description</h3>

<p>This function serves as a wrapper for the entire analysis pipeline. It takes a Fluidigm input file and performs several operations including conversion to 'PLINK' format, error estimation, calculation of pairwise similarities, determination of pairwise similarity loci, and calculation of the similarity matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fluidigmAnalysisWrapper(
  file,
  out = NA,
  outdir = NA,
  db = NA,
  appendSamplesToDB = FALSE,
  map = NA,
  keep.rep = 1,
  neg_controls = NA,
  y.marker = NA,
  x.marker = NA,
  sp.marker = NA,
  plots = TRUE,
  allele_error = 5,
  marker_dropout = 15,
  no_marker = 50,
  male.y = 3,
  male.hetX = 0,
  female.y = 0,
  female.Xtot = 8,
  female.hetXtot = 3,
  warning.noYtot = 2,
  warning.noHetXtot = 3,
  rearrange = TRUE,
  group = NA,
  fixNames = TRUE,
  sexing = TRUE,
  similarity = 0.85,
  verbose = TRUE,
  verbosity = 1,
  missing.geno = "0 0",
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fluidigmAnalysisWrapper_+3A_file">file</code></td>
<td>
<p>A string specifying the path to the Fluidigm input file.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_out">out</code></td>
<td>
<p>A string specifying the output file name. If left empty, the original basename of the input file will be used.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_outdir">outdir</code></td>
<td>
<p>A string specifying the output folder. If left empty the original folder path of the input file will be used.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_db">db</code></td>
<td>
<p>A string specifying the filepath to the database file. If not provided, the function will proceed with the existing data.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_appendsamplestodb">appendSamplesToDB</code></td>
<td>
<p>A logical indicating whether new samples should be added to the database. Default is FALSE.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_map">map</code></td>
<td>
<p>A string specifying the filepath to the PlateDnoY.map file. If not provided, the function will use the map file with the same name as the ped file.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_keep.rep">keep.rep</code></td>
<td>
<p>A numeric value indicating the number of replicates to keep. Default is 1.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_neg_controls">neg_controls</code></td>
<td>
<p>A vector specifying the names of negative controls. Default is NA.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_y.marker">y.marker</code></td>
<td>
<p>A vector specifying the Y markers for sexing. Default is NA.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_x.marker">x.marker</code></td>
<td>
<p>A vector specifying the X markers for sexing. Default is NA.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_sp.marker">sp.marker</code></td>
<td>
<p>A vector specifying the markers used for species identification. Default is NA.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_plots">plots</code></td>
<td>
<p>A logical indicating whether plots should be created. Default is TRUE.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_allele_error">allele_error</code></td>
<td>
<p>A numeric value specifying the threshold for RERUN on Allele errors. Default is 5.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_marker_dropout">marker_dropout</code></td>
<td>
<p>A numeric value specifying the threshold for RERUN on Marker dropout. Default is 15.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_no_marker">no_marker</code></td>
<td>
<p>A numeric value specifying the number of markers. Default is 50.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_male.y">male.y</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, male y-chromosome markers. Default is 3.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_male.hetx">male.hetX</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, heterozygote x-chr markers. Default is 0.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_female.y">female.y</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, female y-chromosome markers. Default is 0.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_female.xtot">female.Xtot</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, total female x-chr markers. Default is 8.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_female.hetxtot">female.hetXtot</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, heterozygote x-chr markers. Default is 3.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_warning.noytot">warning.noYtot</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, when should warning be triggered. Default is 2.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_warning.nohetxtot">warning.noHetXtot</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, when should warning be triggered. Default is 3.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_rearrange">rearrange</code></td>
<td>
<p>A logical indicating whether the ped/map output should be rearranged in order of provided map file. Default is TRUE.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_group">group</code></td>
<td>
<p>A string specifying the sample identifier for statistics. Default is NA.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_fixnames">fixNames</code></td>
<td>
<p>A logical indicating whether whitespaces from sample names should be automatically removed. Default is TRUE.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_sexing">sexing</code></td>
<td>
<p>A logical indicating whether sexing should be performed. Default is FALSE.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_similarity">similarity</code></td>
<td>
<p>Similarity threshold. Default: 0.85.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_verbose">verbose</code></td>
<td>
<p>A logical or numerical value indicating whether the output should be verbose. Default is TRUE.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_verbosity">verbosity</code></td>
<td>
<p>A numerical value indicating the level of verbosity. Set to a higher number for more details. Default is 1.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_missing.geno">missing.geno</code></td>
<td>
<p>A character string specifying how missing values should be coded. Default is &quot;0 0&quot;.</p>
</td></tr>
<tr><td><code id="fluidigmAnalysisWrapper_+3A_overwrite">overwrite</code></td>
<td>
<p>A logical indicating wheter the original map file should be overwritten or not. Default FALSE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function first checks the input parameters and sets default values if necessary. It then runs the following functions in order:
</p>

<ul>
<li><p> fluidigm2PLINK: Converts the Fluidigm data to PLINK format.
</p>
</li>
<li><p> estimateErrors: Estimates errors in the PLINK ped files.
</p>
</li>
<li><p> calculatePairwiseSimilarities: Calculates pairwise similarities between genotypes.
</p>
</li>
<li><p> getPairwiseSimilarityLoci: Determines the loci of pairwise similarities.
</p>
</li>
<li><p> similarityMatrix: Calculates the similarity matrix.
The function prints a completion message when all operations are done.
</p>
</li></ul>



<h3>Value</h3>

<p>A list containing the following elements:
gensim, a matrix indicating if genotypes are called correctly for replicates and/or if genotypes are missing
summs, a matrix with summary statistics
</p>


<h3>References</h3>


<ul>
<li><p>Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender D, Maller J, Sklar P, de Bakker PIW, Daly MJ &amp; Sham PC (2007)
<em>PLINK: a toolset for whole-genome association and population-based linkage analysis.</em> American Journal of Human Genetics, 81.
</p>
</li>
<li><p>Purcell, Shaun. PLINK. <a href="https://zzz.bwh.harvard.edu/plink/">https://zzz.bwh.harvard.edu/plink/</a></p>
</li></ul>



<h3>See Also</h3>


<ul>
<li><p><code><a href="#topic+fluidigm2PLINK">fluidigm2PLINK</a></code>: Converts the Fluidigm data to PLINK format.
</p>
</li>
<li><p><code><a href="#topic+estimateErrors">estimateErrors</a></code>: Estimates errors in the PLINK ped files.
</p>
</li>
<li><p><code><a href="#topic+calculatePairwiseSimilarities">calculatePairwiseSimilarities</a></code>: Calculates pairwise similarities between genotypes.
</p>
</li>
<li><p><code><a href="#topic+getPairwiseSimilarityLoci">getPairwiseSimilarityLoci</a></code>: Determines the loci of pairwise similarities.
</p>
</li>
<li><p><code><a href="#topic+similarityMatrix">similarityMatrix</a></code>: Calculates the similarity matrix.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  fluidigmAnalysisWrapper(file="path/to/your/file.csv", map="path/to/your/mapfile.map")

## End(Not run)

</code></pre>

<hr>
<h2 id='getPairwiseSimilarityLoci'>Get Pairwise Similarity Loci</h2><span id='topic+getPairwiseSimilarityLoci'></span>

<h3>Description</h3>

<p>This function is a wrapper to a perl script that determines the loci of pairwise similarities. It is designed to work with data from the 'PLINK' software, which is commonly used in bioinformatics for whole genome association analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPairwiseSimilarityLoci(file, verbose = TRUE, verbosity = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getPairwiseSimilarityLoci_+3A_file">file</code></td>
<td>
<p>A string. This is the path to the previously created PLINK ped file (with .ped extension). The ped file contains genotype information in a format that can be used for further analysis.</p>
</td></tr>
<tr><td><code id="getPairwiseSimilarityLoci_+3A_verbose">verbose</code></td>
<td>
<p>A logical. If TRUE, the function will print detailed messages during its execution to help you understand what it's doing at each step. Default is TRUE.</p>
</td></tr>
<tr><td><code id="getPairwiseSimilarityLoci_+3A_verbosity">verbosity</code></td>
<td>
<p>An integer. This parameter controls the level of verbosity. The higher the number, the more detailed the messages. Default is 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function first checks the input parameters. It then constructs the output file names and the command to run the perl script.
The command is executed using the system function. If the 'verbose' parameter is set to TRUE, the function will print a message
when it has finished running.
</p>
<p>The perl script was written by Doug Scofield, see references.
</p>


<h3>Value</h3>

<p>This function does not return a value in the R environment. Instead, it creates an output file with the
'.pairs' extension in the same directory as the input file. This output file contains the results of the pairwise
similarity loci analysis.
</p>


<h3>References</h3>

<p>The original code this function is based on can be found at: GitHub https://github.com/douglasgscofield/bioinfo/blob/main/scripts/plink-pairwise-loci.pl
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  outdir &lt;- tempdir()
  getPairwiseSimilarityLoci(file = file.path(outdir, "example_data.csv.GOOD"))

## End(Not run)

</code></pre>

<hr>
<h2 id='similarityMatrix'>Calculate the similarity matrix</h2><span id='topic+similarityMatrix'></span>

<h3>Description</h3>

<p>Performs pairwise similarity analysis on genotypic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>similarityMatrix(
  file = NA,
  mibs.file = NA,
  pairs.file = NA,
  ped.file = NA,
  group = NA,
  plots = TRUE,
  similarity = 0.85,
  verbose = TRUE,
  verbosity = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="similarityMatrix_+3A_file">file</code></td>
<td>
<p>Input file path. Default: NA.</p>
</td></tr>
<tr><td><code id="similarityMatrix_+3A_mibs.file">mibs.file</code></td>
<td>
<p>MIBS input file path. Default: NA.</p>
</td></tr>
<tr><td><code id="similarityMatrix_+3A_pairs.file">pairs.file</code></td>
<td>
<p>PAIRS input file path. Default: NA.</p>
</td></tr>
<tr><td><code id="similarityMatrix_+3A_ped.file">ped.file</code></td>
<td>
<p>PED input file path. Default: NA.</p>
</td></tr>
<tr><td><code id="similarityMatrix_+3A_group">group</code></td>
<td>
<p>Sample identifier for statistics. Default: NA.</p>
</td></tr>
<tr><td><code id="similarityMatrix_+3A_plots">plots</code></td>
<td>
<p>Should plots be created? Default: TRUE.</p>
</td></tr>
<tr><td><code id="similarityMatrix_+3A_similarity">similarity</code></td>
<td>
<p>Similarity threshold. Default: 0.85.</p>
</td></tr>
<tr><td><code id="similarityMatrix_+3A_verbose">verbose</code></td>
<td>
<p>Should output be verbose? Default: TRUE.</p>
</td></tr>
<tr><td><code id="similarityMatrix_+3A_verbosity">verbosity</code></td>
<td>
<p>Verbosity level. Default: 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads genotype data, performs pairwise similarity calculations, generates plots, and outputs data for further analysis.
</p>


<h3>Value</h3>

<p>Does not return a value. Creates output files in the same directory as the input files.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
     similarityMatrix(file = file.path(outdir, "example_data.csv.GOOD"))

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
