<!DOCTYPE html><html lang="en"><head><title>Help for package forensIT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {forensIT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#forensIT'><p>forensIT: Information Theory Tools for Forensic Analysis</p></a></li>
<li><a href='#buildEnsembleCPTs'><p>buildEnsembleCPTs</p></a></li>
<li><a href='#buildEnsembleITValues'><p>buildEnsembleITValues</p></a></li>
<li><a href='#compareBnetPopGenoPDFs'><p>Compare population and Bayesian network genotype probability density functions # nolint</p></a></li>
<li><a href='#crossH'><p>Cross entropy</p></a></li>
<li><a href='#distKL'><p>distKL: KL distribution obtained for specific relative contributor</p></a></li>
<li><a href='#elimLangeGoradia'><p>Eliminate Mendelian errors using Lange-Goradia algorithm</p></a></li>
<li><a href='#exportPed'><p>Export a pedigree to a file</p></a></li>
<li><a href='#genotypeProbs'><p>Genotype probabilities</p></a></li>
<li><a href='#genotypeProbTable'><p>Genotype Probability Table</p></a></li>
<li><a href='#genotypeProbTable_bis'><p>genotypeProbTable_bis</p></a></li>
<li><a href='#getAllelesFromGenotypes'><p>getAllelesFromGenotypes</p></a></li>
<li><a href='#H'><p>Entropy of a discrete probability distribution</p></a></li>
<li><a href='#index2Genotypes2'><p>index2Genotypes2</p></a></li>
<li><a href='#index2Genotypes2.pedtools'><p>index2Genotypes</p></a></li>
<li><a href='#KLd'><p>KL divergence</p></a></li>
<li><a href='#KLde'><p>KL divergence</p></a></li>
<li><a href='#perMarkerKLs'><p>perMarkerKLs</p></a></li>
<li><a href='#plotKL'><p>Plot KL distances.</p></a></li>
<li><a href='#Px'><p>Px</p></a></li>
<li><a href='#runIT'><p>runIT</p></a></li>
<li><a href='#simLR'><p>Simulate LR</p></a></li>
<li><a href='#simME'><p>simME: output from simMinimalEnsemble considering an uncle</p></a></li>
<li><a href='#simMinimalEnsemble'><p>simMinimalEnsemble</p></a></li>
<li><a href='#simTestIDMarkers'><p>Simulate testID markers</p></a></li>
<li><a href='#strsplit2'><p>strsplit2</p></a></li>
<li><a href='#trioCheckFast'><p>trioCheckFast</p></a></li>
<li><a href='#unidimKLplot'><p>unidimKLplot: KL distributions presented in the same units (Log10(LR))</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Information Theory Tools for Forensic Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>The 'forensIT' package is a comprehensive statistical toolkit tailored for handling missing person cases. By leveraging information theory metrics, it enables accurate assessment of kinship, particularly when limited genetic evidence is available. With a focus on optimizing statistical power, 'forensIT' empowers investigators to effectively prioritize family members, enhancing the reliability and efficiency of missing person investigations.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, mispitools, forrel, pedprobr, dplyr, tidyr, magrittr,
fbnet, foreach, hrbrthemes, gtools, reshape2, pedtools,
iterators, doParallel</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-04 14:51:19 UTC; franco</td>
</tr>
<tr>
<td>Author:</td>
<td>Franco Marsico [aut, cre],
  Ariel Chernomoretz [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Franco Marsico &lt;franco.lmarsico@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-04 15:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='forensIT'>forensIT: Information Theory Tools for Forensic Analysis</h2><span id='topic+forensIT-package'></span><span id='topic+forensIT'></span>

<h3>Description</h3>

<p>The 'forensIT' package, available on CRAN, is a comprehensive statistical toolkit  tailored for handling missing person cases. By leveraging information theory metrics, it enables accurate assessment of kinship, particularly when limited genetic evidence is available.   With a focus on optimizing statistical power, 'forensIT' empowers investigators to effectively prioritize family members, enhancing the reliability and efficiency of missing person investigations. Experience the power of information theory in kinship testing with the user-friendly 'forensIT' package, freely accessible on CRAN. # nolint
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Franco Marsico <a href="mailto:franco.lmarsico@gmail.com">franco.lmarsico@gmail.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Ariel Chernomoretz <a href="mailto:ariel@df.uba.ar">ariel@df.uba.ar</a>
</p>
</li></ul>


<hr>
<h2 id='buildEnsembleCPTs'>buildEnsembleCPTs</h2><span id='topic+buildEnsembleCPTs'></span>

<h3>Description</h3>

<p>Build ensemble of CPTs from a list of simulations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildEnsembleCPTs(lsimu, lminimalProbGenoMOI)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildEnsembleCPTs_+3A_lsimu">lsimu</code></td>
<td>
<p>list of simulations</p>
</td></tr>
<tr><td><code id="buildEnsembleCPTs_+3A_lminimalprobgenomoi">lminimalProbGenoMOI</code></td>
<td>
<p>list of minimal probabilities of genotypes given MOI # nolint</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of CPTs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(forrel)
library(mispitools)
freqs &lt;- lapply(getfreqs(Argentina)[1:15], function(x) {x[x!=0]})
fam  &lt;- linearPed(2)
fam  &lt;- addChildren(fam, father =  1, mother =  2)
fam  &lt;- pedtools::setMarkers(fam, locusAttributes = freqs)
ped  &lt;- profileSim(fam, N = 1, ids = c(6)  , numCores = 1,seed=123)
lsimEnsemble  &lt;- simTestIDMarkers(ped,2,numSim=5,seed=123)
lensembleIT   &lt;- buildEnsembleITValues(lsimu=lsimEnsemble,ITtab=simME$ITtable,bFullIT = TRUE)
lensembleCPTs &lt;- buildEnsembleCPTs(lsimu=lsimEnsemble,lminimalProbGenoMOI=simME$lprobGenoMOI)
</code></pre>

<hr>
<h2 id='buildEnsembleITValues'>buildEnsembleITValues</h2><span id='topic+buildEnsembleITValues'></span>

<h3>Description</h3>

<p>Build ensemble of IT values from a list of simulations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildEnsembleITValues(
  lsimu = lsimulation,
  ITtab = sim$ITtable,
  bFullIT = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildEnsembleITValues_+3A_lsimu">lsimu</code></td>
<td>
<p>list of simulations</p>
</td></tr>
<tr><td><code id="buildEnsembleITValues_+3A_ittab">ITtab</code></td>
<td>
<p>IT table</p>
</td></tr>
<tr><td><code id="buildEnsembleITValues_+3A_bfullit">bFullIT</code></td>
<td>
<p>boolean to return full IT table</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of IT values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(forrel)
library(mispitools)
freqs &lt;- lapply(getfreqs(Argentina)[1:15], function(x) {x[x!=0]})
fam  &lt;- linearPed(2)
fam  &lt;- addChildren(fam, father =  1, mother =  2)
fam  &lt;- pedtools::setMarkers(fam, locusAttributes = freqs)
ped  &lt;- profileSim(fam, N = 1, ids = c(6)  , numCores = 1,seed=123)
lsimEnsemble  &lt;- simTestIDMarkers(ped,2,numSim=5,seed=123)
lensembleIT   &lt;- buildEnsembleITValues(lsimu=lsimEnsemble,ITtab=simME$ITtable,bFullIT = TRUE)
</code></pre>

<hr>
<h2 id='compareBnetPopGenoPDFs'>Compare population and Bayesian network genotype probability density functions # nolint</h2><span id='topic+compareBnetPopGenoPDFs'></span>

<h3>Description</h3>

<p>Compare population and Bayesian network genotype probability density functions # nolint
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareBnetPopGenoPDFs(lprobTable)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compareBnetPopGenoPDFs_+3A_lprobtable">lprobTable</code></td>
<td>
<p>list of probability tables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of KL divergences
</p>

<hr>
<h2 id='crossH'>Cross entropy</h2><span id='topic+crossH'></span>

<h3>Description</h3>

<p>Cross entropy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crossH(px, py, epsilon = 1e-20)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="crossH_+3A_px">px</code></td>
<td>
<p>probability distribution</p>
</td></tr>
<tr><td><code id="crossH_+3A_py">py</code></td>
<td>
<p>probability distribution</p>
</td></tr>
<tr><td><code id="crossH_+3A_epsilon">epsilon</code></td>
<td>
<p>small number to avoid log(0)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>cross entropy
</p>

<hr>
<h2 id='distKL'>distKL: KL distribution obtained for specific relative contributor</h2><span id='topic+distKL'></span>

<h3>Description</h3>

<p>distKL: KL distribution obtained for specific relative contributor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distKL(ped, missing, relative, frequency, numsims = 100, cores = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="distKL_+3A_ped">ped</code></td>
<td>
<p>Reference pedigree. It could be an input from read_fam() function or a pedigree built with pedtools. # nolint</p>
</td></tr>
<tr><td><code id="distKL_+3A_missing">missing</code></td>
<td>
<p>Missing person</p>
</td></tr>
<tr><td><code id="distKL_+3A_relative">relative</code></td>
<td>
<p>Selected relative.</p>
</td></tr>
<tr><td><code id="distKL_+3A_frequency">frequency</code></td>
<td>
<p>Allele frequency database.</p>
</td></tr>
<tr><td><code id="distKL_+3A_numsims">numsims</code></td>
<td>
<p>Number of simulated genotypes.</p>
</td></tr>
<tr><td><code id="distKL_+3A_cores">cores</code></td>
<td>
<p>Enables parallelization.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class data.frame with KLs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(forrel)
x = linearPed(2)
x = setMarkers(x, locusAttributes = NorwegianFrequencies[1:2])
x = profileSim(x, N = 1, ids = 2)
distKL(ped = x, missing = 5, relative = 1, cores = 1,
frequency = NorwegianFrequencies[1:2], numsims = 3)
</code></pre>

<hr>
<h2 id='elimLangeGoradia'>Eliminate Mendelian errors using Lange-Goradia algorithm</h2><span id='topic+elimLangeGoradia'></span>

<h3>Description</h3>

<p>Eliminate Mendelian errors using Lange-Goradia algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>elimLangeGoradia(ped, iMarker = 1, bitera = TRUE, bverbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="elimLangeGoradia_+3A_ped">ped</code></td>
<td>
<p>pedigree</p>
</td></tr>
<tr><td><code id="elimLangeGoradia_+3A_imarker">iMarker</code></td>
<td>
<p>index of marker to be used</p>
</td></tr>
<tr><td><code id="elimLangeGoradia_+3A_bitera">bitera</code></td>
<td>
<p>iterate until no more errors are found</p>
</td></tr>
<tr><td><code id="elimLangeGoradia_+3A_bverbose">bverbose</code></td>
<td>
<p>print progress</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pedigree with Mendelian errors eliminated
</p>

<hr>
<h2 id='exportPed'>Export a pedigree to a file</h2><span id='topic+exportPed'></span>

<h3>Description</h3>

<p>Export a pedigree to a file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exportPed(ped, fname, iMarker = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="exportPed_+3A_ped">ped</code></td>
<td>
<p>pedigree</p>
</td></tr>
<tr><td><code id="exportPed_+3A_fname">fname</code></td>
<td>
<p>file name</p>
</td></tr>
<tr><td><code id="exportPed_+3A_imarker">iMarker</code></td>
<td>
<p>index of marker to be used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pedigree with Mendelian errors eliminated
</p>

<hr>
<h2 id='genotypeProbs'>Genotype probabilities</h2><span id='topic+genotypeProbs'></span>

<h3>Description</h3>

<p>Calculate genotype probabilities from parental probabilities
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotypeProbs(probP, probM)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genotypeProbs_+3A_probp">probP</code></td>
<td>
<p>vector of parental probabilities</p>
</td></tr>
<tr><td><code id="genotypeProbs_+3A_probm">probM</code></td>
<td>
<p>vector of parental probabilities</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of genotype probabilities
</p>

<hr>
<h2 id='genotypeProbTable'>Genotype Probability Table</h2><span id='topic+genotypeProbTable'></span>

<h3>Description</h3>

<p>Genotype Probability Table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotypeProbTable(bbn1, resQQ, bplot = FALSE, numMarkers = 4, lLoci)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genotypeProbTable_+3A_bbn1">bbn1</code></td>
<td>
<p>Bayesian network</p>
</td></tr>
<tr><td><code id="genotypeProbTable_+3A_resqq">resQQ</code></td>
<td>
<p>results from bn</p>
</td></tr>
<tr><td><code id="genotypeProbTable_+3A_bplot">bplot</code></td>
<td>
<p>boolean to plot</p>
</td></tr>
<tr><td><code id="genotypeProbTable_+3A_nummarkers">numMarkers</code></td>
<td>
<p>number of markers</p>
</td></tr>
<tr><td><code id="genotypeProbTable_+3A_lloci">lLoci</code></td>
<td>
<p>list of loci</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotype Probability Table
</p>

<hr>
<h2 id='genotypeProbTable_bis'>genotypeProbTable_bis</h2><span id='topic+genotypeProbTable_bis'></span>

<h3>Description</h3>

<p>function to calculate the probability of genotypes given the MOI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotypeProbTable_bis(bbn1, resQQ, bplot = FALSE, numMarkers = 4, freq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genotypeProbTable_bis_+3A_bbn1">bbn1</code></td>
<td>
<p>bayesian network</p>
</td></tr>
<tr><td><code id="genotypeProbTable_bis_+3A_resqq">resQQ</code></td>
<td>
<p>list of results from the inference</p>
</td></tr>
<tr><td><code id="genotypeProbTable_bis_+3A_bplot">bplot</code></td>
<td>
<p>plot results</p>
</td></tr>
<tr><td><code id="genotypeProbTable_bis_+3A_nummarkers">numMarkers</code></td>
<td>
<p>number of markers</p>
</td></tr>
<tr><td><code id="genotypeProbTable_bis_+3A_freq">freq</code></td>
<td>
<p>allele frequencies</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of genotype probabilities
</p>

<hr>
<h2 id='getAllelesFromGenotypes'>getAllelesFromGenotypes</h2><span id='topic+getAllelesFromGenotypes'></span>

<h3>Description</h3>

<p>Get alleles from genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAllelesFromGenotypes(g)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAllelesFromGenotypes_+3A_g">g</code></td>
<td>
<p>genotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>alleles
</p>

<hr>
<h2 id='H'>Entropy of a discrete probability distribution</h2><span id='topic+H'></span>

<h3>Description</h3>

<p>Entropy of a discrete probability distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>H(px, epsilon = 1e-20, normalized = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="H_+3A_px">px</code></td>
<td>
<p>probability distribution</p>
</td></tr>
<tr><td><code id="H_+3A_epsilon">epsilon</code></td>
<td>
<p>small number to avoid log(0)</p>
</td></tr>
<tr><td><code id="H_+3A_normalized">normalized</code></td>
<td>
<p>boolean to normalize entropy</p>
</td></tr>
</table>


<h3>Value</h3>

<p>entropy
</p>

<hr>
<h2 id='index2Genotypes2'>index2Genotypes2</h2><span id='topic+index2Genotypes2'></span>

<h3>Description</h3>

<p>index2Genotypes2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>index2Genotypes2(ped, id, iMarker, alleleSet)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="index2Genotypes2_+3A_ped">ped</code></td>
<td>
<p>pedigree</p>
</td></tr>
<tr><td><code id="index2Genotypes2_+3A_id">id</code></td>
<td>
<p>individual id</p>
</td></tr>
<tr><td><code id="index2Genotypes2_+3A_imarker">iMarker</code></td>
<td>
<p>marker index</p>
</td></tr>
<tr><td><code id="index2Genotypes2_+3A_alleleset">alleleSet</code></td>
<td>
<p>allele set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genotypes
</p>

<hr>
<h2 id='index2Genotypes2.pedtools'>index2Genotypes</h2><span id='topic+index2Genotypes2.pedtools'></span>

<h3>Description</h3>

<p>index2Genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>index2Genotypes2.pedtools(ped, id, iMarker, alleleSet)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="index2Genotypes2.pedtools_+3A_ped">ped</code></td>
<td>
<p>pedigree</p>
</td></tr>
<tr><td><code id="index2Genotypes2.pedtools_+3A_id">id</code></td>
<td>
<p>individual id</p>
</td></tr>
<tr><td><code id="index2Genotypes2.pedtools_+3A_imarker">iMarker</code></td>
<td>
<p>marker index</p>
</td></tr>
<tr><td><code id="index2Genotypes2.pedtools_+3A_alleleset">alleleSet</code></td>
<td>
<p>allele set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genotypes
</p>

<hr>
<h2 id='KLd'>KL divergence</h2><span id='topic+KLd'></span>

<h3>Description</h3>

<p>KL divergence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KLd(ppx, ppy, epsilon = 1e-20, bsigma = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="KLd_+3A_ppx">ppx</code></td>
<td>
<p>probability distribution</p>
</td></tr>
<tr><td><code id="KLd_+3A_ppy">ppy</code></td>
<td>
<p>probability distribution</p>
</td></tr>
<tr><td><code id="KLd_+3A_epsilon">epsilon</code></td>
<td>
<p>small number to avoid log(0)</p>
</td></tr>
<tr><td><code id="KLd_+3A_bsigma">bsigma</code></td>
<td>
<p>boolean to compute sigma</p>
</td></tr>
</table>


<h3>Value</h3>

<p>KL divergence
</p>

<hr>
<h2 id='KLde'>KL divergence</h2><span id='topic+KLde'></span>

<h3>Description</h3>

<p>KL divergence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KLde(px, py, epsilon = 1e-20)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="KLde_+3A_px">px</code></td>
<td>
<p>probability distribution</p>
</td></tr>
<tr><td><code id="KLde_+3A_py">py</code></td>
<td>
<p>probability distribution</p>
</td></tr>
<tr><td><code id="KLde_+3A_epsilon">epsilon</code></td>
<td>
<p>small number to avoid log(0)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>KL divergence
</p>

<hr>
<h2 id='perMarkerKLs'>perMarkerKLs</h2><span id='topic+perMarkerKLs'></span>

<h3>Description</h3>

<p>perMarkerKLs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perMarkerKLs(ped, MP, frequency)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="perMarkerKLs_+3A_ped">ped</code></td>
<td>
<p>Reference pedigree.</p>
</td></tr>
<tr><td><code id="perMarkerKLs_+3A_mp">MP</code></td>
<td>
<p>missing person</p>
</td></tr>
<tr><td><code id="perMarkerKLs_+3A_frequency">frequency</code></td>
<td>
<p>Allele frequency database.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class data.frame with KLs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(forrel)
x = linearPed(2)
plot(x)
x = setMarkers(x, locusAttributes = NorwegianFrequencies[1:5])
x = profileSim(x, N = 1, ids = 2)
perMarkerKLs(x, MP = 5 , NorwegianFrequencies[1:5])
</code></pre>

<hr>
<h2 id='plotKL'>Plot KL distances.</h2><span id='topic+plotKL'></span>

<h3>Description</h3>

<p>Plot KL distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotKL(res)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotKL_+3A_res">res</code></td>
<td>
<p>output from distKL function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A scatterplot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(forrel)
x = linearPed(2)
plot(x)
x = setMarkers(x, locusAttributes = NorwegianFrequencies[1:5])
x = profileSim(x, N = 1, ids = 2)
res &lt;- distKL(ped = x, missing = 5, relative = 1,
cores = 1, frequency = NorwegianFrequencies[1:5], numsims = 5)
plotKL(res)
</code></pre>

<hr>
<h2 id='Px'>Px</h2><span id='topic+Px'></span>

<h3>Description</h3>

<p>Px
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Px(p1, p0, dbg = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Px_+3A_p1">p1</code></td>
<td>
<p>probability distribution</p>
</td></tr>
<tr><td><code id="Px_+3A_p0">p0</code></td>
<td>
<p>probability distribution</p>
</td></tr>
<tr><td><code id="Px_+3A_dbg">dbg</code></td>
<td>
<p>boolean to compute sigma</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Px
</p>

<hr>
<h2 id='runIT'>runIT</h2><span id='topic+runIT'></span>

<h3>Description</h3>

<p>run information theory (IT) metrics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runIT(
  lped = NULL,
  freqs,
  QP,
  dbg,
  numCores,
  bOnlyIT = FALSE,
  lprobg_ped = NULL,
  bsigma = FALSE,
  blog = FALSE,
  dep = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="runIT_+3A_lped">lped</code></td>
<td>
<p>list of pedigree objects</p>
</td></tr>
<tr><td><code id="runIT_+3A_freqs">freqs</code></td>
<td>
<p>list of allele frequencies</p>
</td></tr>
<tr><td><code id="runIT_+3A_qp">QP</code></td>
<td>
<p>QP</p>
</td></tr>
<tr><td><code id="runIT_+3A_dbg">dbg</code></td>
<td>
<p>debug</p>
</td></tr>
<tr><td><code id="runIT_+3A_numcores">numCores</code></td>
<td>
<p>number of cores</p>
</td></tr>
<tr><td><code id="runIT_+3A_bonlyit">bOnlyIT</code></td>
<td>
<p>boolean to only run IT</p>
</td></tr>
<tr><td><code id="runIT_+3A_lprobg_ped">lprobg_ped</code></td>
<td>
<p>list of probG</p>
</td></tr>
<tr><td><code id="runIT_+3A_bsigma">bsigma</code></td>
<td>
<p>boolean to compute sigma</p>
</td></tr>
<tr><td><code id="runIT_+3A_blog">blog</code></td>
<td>
<p>boolean to write log</p>
</td></tr>
<tr><td><code id="runIT_+3A_dep">dep</code></td>
<td>
<p>check fbnet dependency</p>
</td></tr>
</table>


<h3>Value</h3>

<p>runIT
</p>

<hr>
<h2 id='simLR'>Simulate LR</h2><span id='topic+simLR'></span>

<h3>Description</h3>

<p>Simulate LR
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simLR(
  lprobg_ped,
  numSim = 10000,
  epsilon = 1e-20,
  bplot = FALSE,
  bLRs = FALSE,
  seed = 123457
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simLR_+3A_lprobg_ped">lprobg_ped</code></td>
<td>
<p>list of probability distributions</p>
</td></tr>
<tr><td><code id="simLR_+3A_numsim">numSim</code></td>
<td>
<p>number of simulations</p>
</td></tr>
<tr><td><code id="simLR_+3A_epsilon">epsilon</code></td>
<td>
<p>small number to avoid log(0)</p>
</td></tr>
<tr><td><code id="simLR_+3A_bplot">bplot</code></td>
<td>
<p>boolean to plot</p>
</td></tr>
<tr><td><code id="simLR_+3A_blrs">bLRs</code></td>
<td>
<p>boolean to return LRs</p>
</td></tr>
<tr><td><code id="simLR_+3A_seed">seed</code></td>
<td>
<p>seed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>LRs
</p>

<hr>
<h2 id='simME'>simME: output from simMinimalEnsemble considering an uncle</h2><span id='topic+simME'></span>

<h3>Description</h3>

<p>simME: output from simMinimalEnsemble considering an uncle
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simME
</code></pre>


<h3>Format</h3>

<p>A list with minimalEnsemble of genotypes
</p>

<hr>
<h2 id='simMinimalEnsemble'>simMinimalEnsemble</h2><span id='topic+simMinimalEnsemble'></span>

<h3>Description</h3>

<p>It performs simulations of minimal ensembles of genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simMinimalEnsemble(
  ped,
  QP,
  testID,
  freqs,
  numCores = 1,
  seed = 123457,
  bVerbose = TRUE,
  bJustGetNumber = FALSE,
  bdbg = FALSE,
  dep = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simMinimalEnsemble_+3A_ped">ped</code></td>
<td>
<p>pedigree</p>
</td></tr>
<tr><td><code id="simMinimalEnsemble_+3A_qp">QP</code></td>
<td>
<p>QP</p>
</td></tr>
<tr><td><code id="simMinimalEnsemble_+3A_testid">testID</code></td>
<td>
<p>test ID</p>
</td></tr>
<tr><td><code id="simMinimalEnsemble_+3A_freqs">freqs</code></td>
<td>
<p>frequencies</p>
</td></tr>
<tr><td><code id="simMinimalEnsemble_+3A_numcores">numCores</code></td>
<td>
<p>number of cores</p>
</td></tr>
<tr><td><code id="simMinimalEnsemble_+3A_seed">seed</code></td>
<td>
<p>seed</p>
</td></tr>
<tr><td><code id="simMinimalEnsemble_+3A_bverbose">bVerbose</code></td>
<td>
<p>boolean to print information</p>
</td></tr>
<tr><td><code id="simMinimalEnsemble_+3A_bjustgetnumber">bJustGetNumber</code></td>
<td>
<p>boolean to just get the number of runs</p>
</td></tr>
<tr><td><code id="simMinimalEnsemble_+3A_bdbg">bdbg</code></td>
<td>
<p>boolean to debug</p>
</td></tr>
<tr><td><code id="simMinimalEnsemble_+3A_dep">dep</code></td>
<td>
<p>check dependency fbnet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of results
</p>

<hr>
<h2 id='simTestIDMarkers'>Simulate testID markers</h2><span id='topic+simTestIDMarkers'></span>

<h3>Description</h3>

<p>Simulate testID markers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simTestIDMarkers(ped, testID, numSim = 10, seed = 123457)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simTestIDMarkers_+3A_ped">ped</code></td>
<td>
<p>pedigree</p>
</td></tr>
<tr><td><code id="simTestIDMarkers_+3A_testid">testID</code></td>
<td>
<p>test ID</p>
</td></tr>
<tr><td><code id="simTestIDMarkers_+3A_numsim">numSim</code></td>
<td>
<p>number of simulations</p>
</td></tr>
<tr><td><code id="simTestIDMarkers_+3A_seed">seed</code></td>
<td>
<p>seed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of simulations
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(forrel)
library(mispitools)
freqs &lt;- lapply(getfreqs(Argentina)[1:15], function(x) {x[x!=0]})
fam  &lt;- linearPed(2)
fam  &lt;- addChildren(fam, father =  1, mother =  2)
fam  &lt;- pedtools::setMarkers(fam, locusAttributes = freqs)
ped  &lt;- profileSim(fam, N = 1, ids = c(6)  , numCores = 1,seed=123)
lsimEnsemble  &lt;- simTestIDMarkers(ped,2,numSim=5,seed=123)
</code></pre>

<hr>
<h2 id='strsplit2'>strsplit2</h2><span id='topic+strsplit2'></span>

<h3>Description</h3>

<p>strsplit2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strsplit2(x, split)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="strsplit2_+3A_x">x</code></td>
<td>
<p>character vector</p>
</td></tr>
<tr><td><code id="strsplit2_+3A_split">split</code></td>
<td>
<p>character</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix
</p>

<hr>
<h2 id='trioCheckFast'>trioCheckFast</h2><span id='topic+trioCheckFast'></span>

<h3>Description</h3>

<p>Check for Mendelian errors in trios
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trioCheckFast(ffa, mmo, oof)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trioCheckFast_+3A_ffa">ffa</code></td>
<td>
<p>father's alleles</p>
</td></tr>
<tr><td><code id="trioCheckFast_+3A_mmo">mmo</code></td>
<td>
<p>mother's alleles</p>
</td></tr>
<tr><td><code id="trioCheckFast_+3A_oof">oof</code></td>
<td>
<p>offspring's alleles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if there is a Mendelian error
</p>

<hr>
<h2 id='unidimKLplot'>unidimKLplot: KL distributions presented in the same units (Log10(LR))</h2><span id='topic+unidimKLplot'></span>

<h3>Description</h3>

<p>unidimKLplot: KL distributions presented in the same units (Log10(LR))
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unidimKLplot(res)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="unidimKLplot_+3A_res">res</code></td>
<td>
<p>output from distKL function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A scatterplot.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
