<!DOCTYPE html><html><head><title>Help for package FourgameteP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FourgameteP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FourgameteP'><p>FourGamete Test</p></a></li>
<li><a href='#FGf'><p>Four-gamete Test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>FourGamete Test</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Milton T Drott</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Milton T Drott &lt;mtd66@cornell.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The four-gamete test is based on the infinite-sites model which assumes that the probability of the same mutation occurring twice (recurrent or parallel mutations) and the probability of a mutation back to the original state (reverse mutations) are close to zero. Without these types of mutations, the only explanation for observing the four dilocus genotypes (example below) is recombination (Hudson and Kaplan 1985, Genetics 111:147-164). Thus, the presence of all four gametes is also called phylogenetic incompatibility.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-09-30 18:55:13 UTC; mtd</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-10-08 10:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='FourgameteP'>FourGamete Test</h2><span id='topic+FourgameteP'></span><span id='topic+FourgameteP-package'></span>

<h3>Description</h3>

<p>The four-gamete test is based on the infinite-sites model which assumes that the probability of the same mutation occurring twice (recurrent or parallel mutations) and the probability of a mutation back to the original state (reverse mutations) are close to zero. Without these types of mutations, the only explanation for observing the four dilocus genotypes (example below) is recombination (Hudson and Kaplan 1985, Genetics 111:147-164). Thus, the presence of all four gametes is also called phylogenetic incompatibility.
</p>
<p>FourgameteP contains a single function: '<strong>FGf()</strong>'.
</p>
<p>This function determines if, across all pairwise comparisons, it is possible to find all four gametes
</p>
<p><strong>Example:</strong>
</p>

<ul>
<li> <p>...............Locus1....Locus2
</p>
</li>
<li> <p> Gamete 1.......A.........A
</p>
</li>
<li> <p> Gamete 2.......A.........B
</p>
</li>
<li> <p> Gamete 3.......B.........A
</p>
</li>
<li> <p> Gamete 4.......B.........B
</p>
</li></ul>

<p>While the example above indicates two alleles at each of two loci, FourGamete will output all possible allele combinations between two loci.
</p>


<h3>Details</h3>

<p>Please make sure that your data fits all of the following requirements:
</p>

<ul>
<li><p>1) Data must represent alleles at haploid loci (maximum of 26 loci)
</p>
</li>
<li><p>2) Rows are individuals and columns are loci
</p>
</li>
<li><p>3) Loci must labeled in the top row with names containing no spaces
</p>
</li>
<li><p>4) Do not include metadata columns (e.g., individual names or other 
</p>
</li>
<li><p>5) Alleles at a locus must be numerical
</p>
</li>
<li><p>6) Missing data should be coded &quot;0&quot;
</p>
</li>
<li><p>7) Files should be saved as tab-delimited text
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Milton T Drott <a href="mailto:mtd66@cornell.edu">mtd66@cornell.edu</a>
</p>

<hr>
<h2 id='FGf'>Four-gamete Test</h2><span id='topic+FGf'></span>

<h3>Description</h3>

<p>The function FGf allows users to quickly assess genotypic data for phylogenetic incompatibility, a sign of recombination, using the four-gamete test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FGf(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FGf_+3A_x">x</code></td>
<td>
<p>A properly formatted data frame. For details see '?FourgameteP'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>All data output files are organized based on names in the example: 
</p>

<ul>
<li> <p>...............Locus1....Locus2
</p>
</li>
<li> <p> Gamete 1.......A.........A
</p>
</li>
<li> <p> Gamete 2.......A.........B
</p>
</li>
<li> <p> Gamete 3.......B.........A
</p>
</li>
<li> <p> Gamete 4.......B.........B
</p>
</li></ul>

<p>Columns &quot;Locus1&quot; and &quot;Locus2&quot; in data files indicate which two columns (loci) of your data were being compared.
</p>
<p>In data output file <strong>&quot;TrueUnique&quot;</strong>, the column &quot;Locus1A&quot; is the first allele &quot;A&quot; in the column &quot;Locus1&quot; while &quot;Locus2B&quot; would indicate the second allele (B) present in the &quot;Locus2&quot; column. Together they make up four dilocus genotypes (or the four gametes): (Locus1A,Locus2A), (Locus1A, Locus2B), (Locus1B,Locus2A), and (Locus1B,Locus2B)
</p>
<p>The &quot;FourGametes?&quot; column indicates if the program was able to find all four gametes. A value of &quot;TRUE&quot; indicates phylogenetic incompatibility between the tested loci. A value of “FALSE” indicates that only three or fewer gametes were found.
</p>
<p><strong>&quot;FinalResults&quot;</strong> This file contains all locus pair comparisons and if the program was able to find any ANY combination of alleles at a locus pair for which the four gametes were found (TRUE) or if  it was only able to find three or fewer of the gametes (FALSE).
</p>
<p><strong>&quot;TrueUnique&quot;</strong> This file contains all of the possible combination of alleles at all locus pairs for which the four gametes can be found (TRUE).
</p>
<p>The number of monomorphic loci is stored in the value <strong>&quot;MonoLoci&quot;</strong> and their identity in <strong>&quot;NamesOfMonoLoci&quot;</strong>.
</p>
<p>The total number of comparisons between loci that are not monomorphic is stored in <strong>&quot;Comparisons&quot;</strong>
</p>
<p>The number of these comparisons that were TRUE or False is stored in <strong>&quot;ComparisonsTrue&quot;</strong> or in <strong>&quot;ComparisonsFalse&quot;</strong> respectively.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n = c(2, 3, 3, 2)
c = c(2, 2, 4, 4)
d = c(2, 5, 5, 3)
df = data.frame(n, c, d)
FGf(df)



</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
