<!DOCTYPE html><html><head><title>Help for package FSAtools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FSAtools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add.model'>
<p>Object constructor for binary predictors</p></a></li>
<li><a href='#align.fsa'>
<p>Aligns peaks using size ladder</p></a></li>
<li><a href='#classify'>
<p>Apply the binary predictor to FSA peaks</p></a></li>
<li><a href='#designFile'>
<p>Process interface's design file</p></a></li>
<li><a href='#export.attr'>
<p>Print an attribute of a 'fsa' object to a file</p></a></li>
<li><a href='#filter.fsa'>
<p>Applies filter() to a &quot;fsa&quot; object</p></a></li>
<li><a href='#fusions.process'>
<p>LD-RTPCR fusion identification by Sanger</p></a></li>
<li><a href='#generic.log'>
<p>Wrapper for <code>generic.process</code></p></a></li>
<li><a href='#generic.process'>
<p>Processing multiple FSA files</p></a></li>
<li><a href='#genotype.fsa'>
<p>Calls alleles for a SNP genotyping experiment</p></a></li>
<li><a href='#multiplot'>
<p>Wrapper to layout</p></a></li>
<li><a href='#peaks.fsa'>
<p>Get maximal value in ranges</p></a></li>
<li><a href='#plot.fsa'>
<p>Plot method for &quot;fsa&quot; objects</p></a></li>
<li><a href='#plot.fsaModel'>
<p>Plot method for &quot;fsaModel&quot; objects</p></a></li>
<li><a href='#print.fsa'>
<p>Print method for &quot;fsa&quot; objects</p></a></li>
<li><a href='#read.abif'><p>Read ABIF formatted files</p></a></li>
<li><a href='#read.fsa'>
<p>Imports a .fsa file from Applied Biosystems</p></a></li>
<li><a href='#read.sanger'>
<p>Imports a .ab1 file from Applied Biosystems corresponding to Sanger sequencing</p></a></li>
<li><a href='#train'>
<p>Training function for binary predictors</p></a></li>
<li><a href='#wav2RGB'>
<p>Converts light wavelengths to RGB colors</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Fragment Analysis and Capillary Sequencing Tool Kit</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-18</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://bioinformatics.ovsa.fr/FSAtools">https://bioinformatics.ovsa.fr/FSAtools</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/maressyl/R.FSAtools/issues">https://github.com/maressyl/R.FSAtools/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>A flexible and interfaced framework for importing, processing and ploting Applied Biosystems data files. Application to Reverse-Transcriptase Multiplex Ligation-dependent Probe Amplification (RT-MLPA) gene-expression profiling and classification is illustrated in Mareschal, Ruminy et al (2015) &lt;<a href="https://doi.org/10.1016%2Fj.jmoldx.2015.01.007">doi:10.1016/j.jmoldx.2015.01.007</a>&gt;. Gene-fusion detection and Sanger sequencing are illustrated in Mareschal, Palau et al (2021) &lt;<a href="https://doi.org/10.1182%2Fbloodadvances.2020002517">doi:10.1182/bloodadvances.2020002517</a>&gt;. Examples are provided for genotyping applications as well.</td>
</tr>
<tr>
<td>Note:</td>
<td>FSAtools replaces and generalizes the former MLPA package from
the same authors.</td>
</tr>
<tr>
<td>Depends:</td>
<td>graphics, grDevices, stats, utils, R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Biostrings, parallel, tcltk, tools</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-18 14:00:53 UTC; sylvain</td>
</tr>
<tr>
<td>Author:</td>
<td>Sylvain Mareschal [aut, cre],
  Philippe Ruminy [dtc, ctb],
  Jean R. Lobry [ctb],
  Fabrice Jardin [ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sylvain Mareschal &lt;mareschal@ovsa.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-18 18:12:35 UTC</td>
</tr>
</table>
<hr>
<h2 id='add.model'>
Object constructor for binary predictors
</h2><span id='topic+add.model'></span>

<h3>Description</h3>

<p>This function aggregates the data required to predict class in <code><a href="#topic+classify">classify</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  add.model(x, model, groupMeans, groupSDs, groupNames, groupColors, threshold,
    geneNames, geneTs, geneMs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.model_+3A_x">x</code></td>
<td>

<p>An object of class <code>fsa</code> to which add a model.
</p>
</td></tr>
<tr><td><code id="add.model_+3A_model">model</code></td>
<td>

<p>A <code>list</code> aggregating the model parameters. Can be provided instead of the argument vectors below individually.
</p>
</td></tr>
<tr><td><code id="add.model_+3A_groupmeans">groupMeans</code></td>
<td>

<p>Numeric vector of length 2, the means of the scores in each group as computed on a training series. If <code>model</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="add.model_+3A_groupsds">groupSDs</code></td>
<td>

<p>Numeric vector of length 2, the standard deviations of the scores in each group as computed on a training series. If <code>model</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="add.model_+3A_groupnames">groupNames</code></td>
<td>

<p>Character vector of length 2, the names of the group described in <code>groupMeans</code>, <code>groupSDs</code> and <code>groupColors</code>. If <code>model</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="add.model_+3A_groupcolors">groupColors</code></td>
<td>

<p>Character vector of length 2, the colors to use to plot each group (see <code><a href="graphics.html#topic+par">par</a></code> for allowed values). If <code>model</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="add.model_+3A_threshold">threshold</code></td>
<td>

<p>Single numeric vector, the confidence threshold to use for prediction (a call will be made only if it is at least at this level of certainty). If <code>model</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="add.model_+3A_genenames">geneNames</code></td>
<td>

<p>Character vector, the names of the genes whose expression is to be used. If <code>model</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="add.model_+3A_genets">geneTs</code></td>
<td>

<p>Numeric vector, for each gene in <code>geneNames</code>, the statistic of a <code><a href="stats.html#topic+t.test">t.test</a></code> comparing its expression between the two groups in a training series. If <code>model</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="add.model_+3A_genems">geneMs</code></td>
<td>

<p>Numeric vector, for each gene in <code>geneNames</code>, the mean expression in the whole training series. If <code>model</code> is provided, this vector will be used instead.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an S3 object of class <code>fsaModel</code>.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classify">classify</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Example FSA file provided
  fsa &lt;- read.fsa(system.file("extdata/fsa_GEP/A5918.fsa", package="FSAtools"))
  
  # Add model from design file
  design &lt;- designFile(system.file("extdata/design_GEP.conf", package="FSAtools"))
  fsa &lt;- add.model(fsa, model=design$GLOBALS$MODEL)
  
  # Observe model
  print(attr(fsa, "model"))
  plot(attr(fsa, "model"))
</code></pre>

<hr>
<h2 id='align.fsa'>
Aligns peaks using size ladder
</h2><span id='topic+align.fsa'></span>

<h3>Description</h3>

<p>This function adds to a <code>fsa</code> object a linear regression model allowing the raw time indexes to be converted into base pair sizes, using a known size markers ladder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  align.fsa(x, channel = "ROX", fullLadder = c(50, 60, 90, 100, 120, 150, 160, 180, 190,
    200, 220, 240, 260, 280, 290, 300, 320, 340, 360, 380, 400), useLadder = c(50, 60, 
    90, 100, 120), outThreshold = 0.15, noiseLevel = 10, surePeaks = 5,
	leakingRatios = c(-1, 10), trim = c("forward", "backward", "none"),
	maskOffScale = FALSE, rMin = 0.999, rescue = FALSE, ylim = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="align.fsa_+3A_x">x</code></td>
<td>

<p>An object of class <code>fsa</code>, as returned by <code><a href="#topic+read.fsa">read.fsa</a></code>
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_channel">channel</code></td>
<td>

<p>Single character value, the name of the channel used for size markers.
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_fullladder">fullLadder</code></td>
<td>

<p>Integer vector, the size markers used in the assay (in base pairs).
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_useladder">useLadder</code></td>
<td>

<p>Integer vector, the size markers to use for the alignment (using only size markers nearing the expected size for the experimental peaks may achieve a more precise alignment). They must be present in <code>fullLadder</code>. If <code>NULL</code>, <code>fullLadder</code> will be used entirely.
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_outthreshold">outThreshold</code></td>
<td>

<p>Single numeric value, maximal distance from the computed size-marker intensity for a peak to be considered as a size-marker. If lower than 1, it is considered as a proportion of the size-marker intensity computed from sure peaks.
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_noiselevel">noiseLevel</code></td>
<td>

<p>Single numeric value, minimal intensity for a local maximum to be considered as a peak.
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_maskoffscale">maskOffScale</code></td>
<td>

<p>Single logical value, whether to mask indexes with off-scale values in any channel to limit side-effects or not.
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_surepeaks">surePeaks</code></td>
<td>

<p>Single integer value, amount of peaks to use to compute size-marker intensity. They are selected at the end of the profile, as most artefacts are observed ate the beginning. Consider to reduce this value if your assay was prematurely ended.
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_leakingratios">leakingRatios</code></td>
<td>

<p>Numeric vector of length two, defining the thresholds to consider a marker peak as a leakage from another channel. A leakage is expected to show at least one channel with a negative value below the absolute value of the marker channel (first ratio, -1) and another channel with a positive value higher than 10 times the absolute value of the marker channel (second ratio, 10).
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_trim">trim</code></td>
<td>

<p>Single character value, defining how to behave when more/less peaks than expected are read. &quot;forward&quot; will keep first peaks and adjust discarding the last ones, &quot;backward&quot; will keep last peaks and adjust discarding the first ones, and &quot;none&quot; will generate an error.
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_rmin">rMin</code></td>
<td>

<p>Single numeric value, minimum adjusted r squared value (see <code><a href="stats.html#topic+summary.lm">summary.lm</a></code>) to consider an alignment as &quot;good&quot;. Poor alignments raise a warning, and may be due to artefactual peaks in the size-marker channel or errors in <code>fullLadder</code> definition. Consider lowering <code>outThreshold</code> and raising <code>noiseLevel</code> to minimize artefact selection.
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_rescue">rescue</code></td>
<td>

<p>Single logical value, whether to plot a &quot;rescue&quot; profile or not. Rescue profiles are calls to <code><a href="#topic+plot.fsa">plot.fsa</a></code> on which diverse additionnal data is drawn to help diagnose alignment problems.
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_ylim">ylim</code></td>
<td>

<p>To be passed to <code><a href="#topic+plot.fsa">plot.fsa</a></code> for the alignment rescue plot, if enabled (see <code>rescue</code>).
</p>
</td></tr>
<tr><td><code id="align.fsa_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="#topic+plot.fsa">plot.fsa</a></code> for the alignment rescue plot, if enabled (see <code>rescue</code>).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the object of class <code>fsa</code> provided with updated <code><a href="base.html#topic+attributes">attributes</a></code> :
</p>
<table>
<tr><td><code>ladderModel</code></td>
<td>
<p>A numeric vector of linear regression coefficients to use to convert raw indexes into base pairs.</p>
</td></tr>
<tr><td><code>ladderExact</code></td>
<td>
<p>A named numeric vector of raw indexes at which size markers were detected.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal, Philippe Ruminy
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Example FSA file provided
  fsa &lt;- read.fsa(system.file("extdata/fsa_GEP/A5918.fsa", package="FSAtools"))
  
  # Plot subset of the profile (time index)
  plot(fsa, units="index", xlim=c(4000,5000))
  
  # Align using full ladder
  fullLadder &lt;- c(
    50, 60, 90, 100, 120, 150, 160, 180, 190, 200, 220,
    240, 260, 280, 290, 300, 320, 340, 360, 380, 400
  )
  fsa &lt;- align.fsa(fsa, fullLadder=fullLadder)
  
  # Plot subset of the profile (base pairs)
  plot(fsa, units="bp", xlim=c(80,130))
</code></pre>

<hr>
<h2 id='classify'>
Apply the binary predictor to FSA peaks
</h2><span id='topic+classify'></span>

<h3>Description</h3>

<p>Predict to which class the sample is most likely to belong, using a modified LPS model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  classify(x, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="classify_+3A_x">x</code></td>
<td>

<p>A <code>fsa</code> object with <code>peaks</code> and <code>model</code> attributes.
</p>
</td></tr>
<tr><td><code id="classify_+3A_plot">plot</code></td>
<td>

<p>Single logical value, whether to plot a visual representation of the prediction or not.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list :
</p>
<table>
<tr><td><code>score</code></td>
<td>
<p>The raw score used to make the prediction.</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>The probability to belong to each of the two groups.</p>
</td></tr>
<tr><td><code>class</code></td>
<td>
<p>The final prediction, as a group name. May be <code>NA</code> if no probability passes the model threshold.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>References</h3>

<p>Radmacher MD, McShane LM, Simon R. <cite>A paradigm for class prediction using gene expression profiles.</cite> J Comput Biol. 2002;9(3):505-11.
</p>
<p>Wright G, Tan B, Rosenwald A, Hurt EH, Wiestner A, Staudt LM. <cite>A gene expression-based method to diagnose clinically distinct subgroups of diffuse large B cell lymphoma.</cite> Proc Natl Acad Sci U S A. 2003 Aug 19;100(17):9991-6.
</p>
<p>Bohers E, Mareschal S, Bouzelfen A, Marchand V, Ruminy P, Maingonnat C, Menard AL, Etancelin P, Bertrand P, Dubois S, Alcantara M, Bastard C, Tilly H, Jardin F. <cite>Targetable activating mutations are very frequent in GCB and ABC diffuse large B-cell lymphoma.</cite> Genes Chromosomes Cancer. 2014 Feb;53(2):144-53.  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fsa">read.fsa</a></code>, <code><a href="#topic+peaks.fsa">peaks.fsa</a></code>, <code><a href="#topic+add.model">add.model</a></code>, <code><a href="#topic+generic.process">generic.process</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Example FSA file provided
  fsa &lt;- read.fsa(system.file("extdata/fsa_GEP/A5918.fsa", package="FSAtools"))
  fsa &lt;- align.fsa(fsa)
  
  # Add model from design file
  design &lt;- designFile(system.file("extdata/design_GEP.conf", package="FSAtools"))
  fsa &lt;- add.model(fsa, model=design$GLOBALS$MODEL)
  
  # Add peak heights
  fsa &lt;- peaks.fsa(fsa, peaks=design$GLOBALS$PEAKS)
  
  # Classify sample
  fsa &lt;- classify(fsa, plot=TRUE)
  print(attr(fsa, "classification"))
</code></pre>

<hr>
<h2 id='designFile'>
Process interface's design file
</h2><span id='topic+designFile'></span>

<h3>Description</h3>

<p>This function is a slave for <code><a href="#topic+generic.process">generic.process</a></code>. It process a design file and returns its processed elements as a list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  designFile(fileName)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="designFile_+3A_filename">fileName</code></td>
<td>

<p>Single character value, the path and name of a design file to process.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Design files are text file split in multiple sections. Each section starts with a &quot;[NAME]&quot; line and ends when the next section begins. Lines starting with a <code>#</code> sign are ignored, as well as blank lines.
</p>
<p>Standard sections refer to an existing R function (the section name is expected to match the function name in a case-sensitive manner), each line in the section setting an argument to call this function : the first value is the argument name, then after a tabulation come one or many values separated by tabulations. Multiple values will be aggregated into a vector, and <code>type.convert</code> will try to guess the correct type.
</p>
<p>One or many modifiers can be added in the section name after the function name followed by the <code>:</code> sign and separated by commas (e.g. <code>file.remove:first,nowarn</code>). Modifiers have the following effects :
</p>

<dl>
<dt>first</dt><dd><p>The function will be called only once, while processing the first .fsa file.</p>
</dd>
<dt>last</dt><dd><p>The function will be called only once, while processing the last .fsa file.</p>
</dd>
<dt>nowarn</dt><dd><p>The function will be embedded inside <code>suppressWarnings</code> to silently ignore warnings.</p>
</dd>
</dl>

<p><code>$NAME</code> and <code>@NAME</code> can be used while setting an argument to refer to global variable <code>NAME</code>. Use <code>$</code> reference for character variables, which will be replaced with <code>gsub</code> and can thus be combined (e.g. <code>$NAME.txt</code>). Use <code>@</code> reference to obtain the raw R variable, regardless of its type. Default globals are :
</p>

<dl>
<dt>FILE_PATH</dt><dd><p>The full name and path of the .fsa file currently considered in the loop.</p>
</dd>
<dt>FILE_DIR</dt><dd><p>The parent directory of the .fsa file currently considered in the loop.</p>
</dd>
<dt>FILE_NAME</dt><dd><p>The base name (without path) of the .fsa file currently considered in the loop.</p>
</dd>
<dt>OBJECT</dt><dd><p>The last object of class 'fsa' returned by any function called in the pipeline.</p>
</dd>
<dt>OUTPUT_PATH</dt><dd><p>The full name and path defined by <code>output</code> while calling <code><a href="#topic+generic.process">generic.process</a></code>.</p>
</dd>
<dt>OUTPUT_DIR</dt><dd><p>The parent directory of <code>output</code> while calling <code><a href="#topic+generic.process">generic.process</a></code>.</p>
</dd>
<dt>OUTPUT_NAME</dt><dd><p>The base name (without path) defined by <code>output</code> while calling <code><a href="#topic+generic.process">generic.process</a></code>.</p>
</dd>
</dl>

<p>Sections with full uppercase names (only letters and <code>_</code> are allowed) will define a new global variable with matching name. The global will be a named list, each line in the section defining a vector (the first value being the name for the vector in the list). The <code>table</code> modifier can be used while defining globals (e.g. <code>PEAKS:table</code>), to request instead the section to be parsed as a TSV file (one row in the section is one row in the table, columns are separated by tabulations, see <code><a href="utils.html#topic+read.table">read.table</a></code>). The first row will be used as column names and the first column as row names.
</p>
<p>A DESIGN global (<code>[DESIGN]</code> section) is strongly recommended to keep design file self-explained, the following elements are suggested :
</p>

<dl>
<dt>author</dt><dd><p>The name of the design author (for human readers only).</p>
</dd>
<dt>purpose</dt><dd><p>The description of the design (for human readers only).</p>
</dd>
<dt>FSAtools</dt><dd><p>Version of the FSAtools package for which the design was created (separated with dots).</p>
</dd>
<dt>updated</dt><dd><p>Date of the last design update (YYYY-MM-DD).</p>
</dd>
</dl>

<p>Please refer to the two provided working examples to help building your own designs. Full processing with the examples are described in <code><a href="#topic+generic.process">generic.process</a></code> <code>examples</code> section, direct access to the example design files is shown below.
</p>


<h3>Value</h3>

<p>Returns a multi-level <code>list</code>, with a direct children per function to call and an extra <code>GLOBALS</code> element.
</p>
<p>Children are named according to the function to call, thus multiple children can have the same name.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generic.process">generic.process</a></code>, <code><a href="#topic+generic.interface">generic.interface</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Example file provided
  file &lt;- system.file("extdata/design_GEP.conf", package="FSAtools")
  design &lt;- designFile(file)
  
  # Alignment rescue design provided
  file &lt;- system.file("extdata/design_SNP.conf", package="FSAtools")
  design &lt;- designFile(file)
</code></pre>

<hr>
<h2 id='export.attr'>
Print an attribute of a 'fsa' object to a file
</h2><span id='topic+export.attr'></span>

<h3>Description</h3>

<p>Adds the content of an attribute of a 'fsa' object to a CSV file, either appending new rows or columns.
</p>
<p>Typically used in <code><a href="#topic+generic.process">generic.process</a></code> via the design file to export numeric data during the processing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  export.attr(x, attr, file, meta = character(0), sep = "\t", dec = ".", quote = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export.attr_+3A_x">x</code></td>
<td>

<p>The <code>fsa</code> object whose attribute is to be printed.
</p>
</td></tr>
<tr><td><code id="export.attr_+3A_attr">attr</code></td>
<td>

<p>Single character value, the name of the attribute to print.
</p>
</td></tr>
<tr><td><code id="export.attr_+3A_file">file</code></td>
<td>

<p>Single character value, the path and name to the file to create or update.
</p>
</td></tr>
<tr><td><code id="export.attr_+3A_meta">meta</code></td>
<td>

<p>Character vector, the names of <code>x</code> meta-data fields to export as extra columns.
</p>
</td></tr>
<tr><td><code id="export.attr_+3A_sep">sep</code></td>
<td>

<p>To be passed to <code><a href="utils.html#topic+write.table">write.table</a></code>.
</p>
</td></tr>
<tr><td><code id="export.attr_+3A_dec">dec</code></td>
<td>

<p>To be passed to <code><a href="utils.html#topic+write.table">write.table</a></code>.
</p>
</td></tr>
<tr><td><code id="export.attr_+3A_quote">quote</code></td>
<td>

<p>To be passed to <code><a href="utils.html#topic+write.table">write.table</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>New data will be appended below the content of the file with extra 'meta' columns, row names will be added only if the file was empty.
</p>


<h3>Value</h3>

<p>Invisibly returns <code>TRUE</code> on success.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fsa">read.fsa</a></code>, <code><a href="#topic+peaks.fsa">peaks.fsa</a></code>, <code><a href="#topic+genotype.closest.fsa">genotype.closest.fsa</a></code>, <code><a href="#topic+genotype.ratio.fsa">genotype.ratio.fsa</a></code>, <code><a href="#topic+classify">classify</a></code>
</p>

<hr>
<h2 id='filter.fsa'>
Applies filter() to a &quot;fsa&quot; object
</h2><span id='topic+filter.fsa'></span>

<h3>Description</h3>

<p>Replaces the requested column of values by the output of <code><a href="stats.html#topic+filter">filter</a></code>, possibly after masking values out of a specified index or bp range.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  filter.fsa(x, channel, ..., from = NA, to = NA, units = "bp")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter.fsa_+3A_x">x</code></td>
<td>

<p>An object of class <code>fsa</code>, as returned by <code><a href="#topic+read.fsa">read.fsa</a></code>
</p>
</td></tr>
<tr><td><code id="filter.fsa_+3A_channel">channel</code></td>
<td>

<p>Single character value, the name of the channel used for size markers.
</p>
</td></tr>
<tr><td><code id="filter.fsa_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="stats.html#topic+filter">filter</a></code>.
</p>
</td></tr>
<tr><td><code id="filter.fsa_+3A_from">from</code></td>
<td>

<p>Single numeric value, the starting offset (integer index or numeric bp) to consider. No subsetting will be applied if <code>NA</code>.
</p>
</td></tr>
<tr><td><code id="filter.fsa_+3A_to">to</code></td>
<td>

<p>Single numeric value, the last offset (integer index or numeric bp) to consider. No subsetting will be applied if <code>NA</code>.
</p>
</td></tr>
<tr><td><code id="filter.fsa_+3A_units">units</code></td>
<td>

<p>Either &quot;index&quot; or &quot;bp&quot;, defining the unit of <code>from</code> and <code>to</code>. Notice <code>x</code> must have been processed by <code><a href="#topic+align.fsa">align.fsa</a></code> to use &quot;bp&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>x</code>, with updated content.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fsa">read.fsa</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Example FSA file provided
  fsa &lt;- read.fsa(system.file("extdata/fsa_GEP/A5918.fsa", package="FSAtools"))
  fsa &lt;- align.fsa(fsa)
  
  # Profile before filtering
  plot(fsa)
  
  # Plot subset of the profile (base pairs)
  fsa &lt;- filter.fsa(fsa, channel="ROX", filter=20, from=40, to=140, units="bp")
  
  # Profile after filtering
  plot(fsa)
</code></pre>

<hr>
<h2 id='fusions.process'>
LD-RTPCR fusion identification by Sanger
</h2><span id='topic+fusions.process'></span>

<h3>Description</h3>

<p>Automatically interpret gene fusions found by Sanger sequencing using the Ligation-Dependent PCR protocol. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  fusions.process(input, design, output, sheet = NA, cores = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fusions.process_+3A_input">input</code></td>
<td>

<p>Single character value, the path to the directory containing the AB1 files to process.
</p>
</td></tr>
<tr><td><code id="fusions.process_+3A_design">design</code></td>
<td>

<p>Data.frame describing all possible fusions (see Details).
</p>
</td></tr>
<tr><td><code id="fusions.process_+3A_output">output</code></td>
<td>

<p>Single character value, the path to a directory in which to produce output files (will be created if doesn't yet exists).
</p>
</td></tr>
<tr><td><code id="fusions.process_+3A_sheet">sheet</code></td>
<td>

<p>Single character value, the name and path of a CSV file describing the files to process. 3 columns are expected: <code>ID</code> which gives a simpler sample name to use in outputs, <code>way</code> which defines if sequencing was 'forward' or 'reverse', and <code>file</code> which gives the file name and path relative to the <code>input</code> argument.
</p>
</td></tr>
<tr><td><code id="fusions.process_+3A_cores">cores</code></td>
<td>

<p>Single integer value, the amount of CPUs to use on the local machine to parallelize the computation. If <code>NA</code>, a guess will be made. If 1, computation will not use the <code>parallel</code> package at all but only loop over samples.
</p>
</td></tr>
<tr><td><code id="fusions.process_+3A_...">...</code></td>
<td>

<p>Further arguments are passed to <code>fusions.process.core</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>design</code> must contain one row for each possible combination of a left primer with a right primer, whether this fusion is expected and relevant or not.
</p>
<p>Expected columns in <code>design</code> are (excluding extra columns required with <code>extra</code>) :
</p>

<dl>
<dt>left.name</dt><dd><p>Character, the name of the left primer.</p>
</dd>
<dt>left.seq</dt><dd><p>Character (uppercase), the sequence of the left primer (gene-specific part only).</p>
</dd>
<dt>left.unileft</dt><dd><p>Character (uppercase), the sequence of the left universal primer used for amplification.</p>
</dd>
<dt>left.symbol</dt><dd><p>Character, the symbol of the gene targeted by the left primer.</p>
</dd>
<dt>left.GRCh38</dt><dd><p>Character, the genomic coordinates of the last base of the left primer (chromosome:position:strand).</p>
</dd>
<dt>left.GRCh38_band</dt><dd><p>Character, the cytogenetic location of the gene targeted by the left primer.</p>
</dd>
<dt>right.name</dt><dd><p>Character, the name of the right primer.</p>
</dd>
<dt>right.seq</dt><dd><p>Character (uppercase), the sequence of the right primer (gene-specific part only).</p>
</dd>
<dt>right.uniright</dt><dd><p>Character (uppercase), the sequence of the right universal primer used for amplification.</p>
</dd>
<dt>right.symbol</dt><dd><p>Character, the symbol of the gene targeted by the right primer.</p>
</dd>
<dt>right.GRCh38</dt><dd><p>Character, the genomic coordinates of the last base of the right primer (chromosome:position:strand).</p>
</dd>
<dt>right.GRCh38_band</dt><dd><p>Character, the cytogenetic location of the gene targeted by the right primer.</p>
</dd>
<dt>seq_forward</dt><dd><p>Character (uppercase), the complete sequence expected in forward sequencing (concatenation of <code>left.unileft</code>, <code>left.seq</code>, <code>right.seq</code>, <code>right.uniright</code> and the right tail, if any).</p>
</dd>
<dt>seq_reverse</dt><dd><p>Character (uppercase), the complete sequence expected in reverse sequencing (reverse complement of a concatenation of the left tail, if any, <code>left.unileft</code>, <code>left.seq</code>, <code>right.seq</code>, <code>right.uniright</code>).</p>
</dd>
</dl>

<p>Please contact the authors to obtain a relevant design object.
</p>


<h3>Value</h3>

<p>Invisibly returns the aggregated table of top results for all samples.
</p>
<p>Various files are produced, in location set by the <code>output</code> argument :
</p>

<dl>
<dt>Top.csv</dt><dd><p>The aggregated table of top results for all samples.</p>
</dd>
<dt>*.pdf</dt><dd><p>One plot for each sample, showing the sequencing profile and the best alignments found.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generic.process">generic.process</a></code>
</p>

<hr>
<h2 id='generic.log'>
Wrapper for <code>generic.process</code>
</h2><span id='topic+generic.log'></span>

<h3>Description</h3>

<p>This function is mainly a wrapper for <code><a href="#topic+generic.process">generic.process</a></code>, diverting messages, warnings and errors to a more readable log file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  generic.log(..., logFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generic.log_+3A_...">...</code></td>
<td>

<p>Arguments to be passed to <code><a href="#topic+generic.process">generic.process</a></code>.
</p>
</td></tr>
<tr><td><code id="generic.log_+3A_logfile">logFile</code></td>
<td>

<p>Single character value, the path and name of the log file where to divert output.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either an <code>error</code> object if one occured, an integer number of warnings which happened during the (otherwise successfull) processing or <code>TRUE</code> if everything went fine.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generic.process">generic.process</a></code>, <code><a href="#topic+generic.interface">generic.interface</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Working in temporary directory
  output &lt;- sprintf("%s/GEP", tempdir())
  logFile &lt;- sprintf("%s.log", output)
  
  # Direct analysis
  generic.process(
    input = system.file("extdata/fsa_GEP", package="FSAtools"),
    design = system.file("extdata/design_GEP.conf", package="FSAtools"),
    output = output
  )
  
  # Logged analysis (check logFile)
  generic.log(
    input = system.file("extdata/fsa_GEP", package="FSAtools"),
    design = system.file("extdata/design_GEP.conf", package="FSAtools"),
    output = output,
    logFile = logFile
  )
</code></pre>

<hr>
<h2 id='generic.process'>
Processing multiple FSA files
</h2><span id='topic+generic.process'></span><span id='topic+generic.interface'></span>

<h3>Description</h3>

<p><code>generic.process</code> handles the whole analysis of a series of .fsa files according to the pipeline described in the user-provided design file, generating tabular and graphical profiles.
</p>
<p><code>generic.interface</code> summons a Tcl-Tk interface to call <code>generic.process</code> interactively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  generic.process(input, design, output, include = NULL, exclude = NULL,
     progressBar = NULL)
  generic.interface()
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generic.process_+3A_input">input</code></td>
<td>

<p>Single character value, the path to a directory containing .fsa files to analyse. Notice it will be explored recursively, so sub-directories are allowed.
</p>
</td></tr>
<tr><td><code id="generic.process_+3A_design">design</code></td>
<td>

<p>Single character value, the path to a design file, as handled by <code><a href="#topic+designFile">designFile</a></code>.
</p>
</td></tr>
<tr><td><code id="generic.process_+3A_output">output</code></td>
<td>

<p>Single character value, the path to a &quot;.pdf&quot; or &quot;.log&quot; file that will be created during the analysis.
</p>
</td></tr>
<tr><td><code id="generic.process_+3A_include">include</code></td>
<td>

<p>Single character value, a regular expression files (with relative path) in <code>input</code> must match to be processed (ignored if <code>NULL</code>).
</p>
</td></tr>
<tr><td><code id="generic.process_+3A_exclude">exclude</code></td>
<td>

<p>Single character value, a regular expression files (with relative path) in <code>input</code> must not match to be processed (ignored if <code>NULL</code>).
</p>
</td></tr>
<tr><td><code id="generic.process_+3A_progressbar">progressBar</code></td>
<td>

<p>A <code>ttkprogressbar</code> to increment during the processing, or <code>NULL</code>. This argument is only provided to connect <code>GEP.interface</code> and <code>GEP.process</code>, thus it should be ignored.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The content of the analysis pipeline is fully controlled by the design file, see <code><a href="#topic+designFile">designFile</a></code> for details and the <code>examples</code> section below for two working examples provided in the package.
</p>
<p>More generally, <code>generic.process</code> loops over the list of .fsa files in the <code>input</code> directory and calls the requested functions one after the other, updating the <code>fsa</code> object at each step.
</p>


<h3>Value</h3>

<p>Return nothing. <code>generic.process</code> raise errors, warnings and messages which are intercepted by <code>generic.interface</code> and redirected to the log file (<code>output</code>.log).
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>References</h3>

<p>Mareschal, Ruminy et al (2015) &lt;doi:10.1016/j.jmoldx.2015.01.007&gt; &quot;Accurate Classification of Germinal Center B-Cell-Like/Activated B-Cell-Like Diffuse Large B-Cell Lymphoma Using a Simple and Rapid Reverse Transcriptase-Multiplex Ligation-Dependent Probe Amplification Assay: A CALYM Study&quot;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+designFile">designFile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ### EXAMPLE 1 : Gene expression (RT-MLPA) ###
  
  # Working in temporary directory
  output &lt;- sprintf("%s/GEP", tempdir())
  
  # See files before analysis
  dir(system.file("extdata", package="FSAtools"))
  
  # Launch analysis in package directory
  generic.process(
    input = system.file("extdata/fsa_GEP", package="FSAtools"),
    design = system.file("extdata/design_GEP.conf", package="FSAtools"),
    output = output
  )
  
  # List resulting files
  dir(dirname(output), full.names=TRUE)
  
  
  ### EXAMPLE 2 : Genotyping ###
  
  # Working in temporary directory
  output &lt;- sprintf("%s/SNP", tempdir())
  
  # See files before analysis
  dir(system.file("extdata", package="FSAtools"))
  
  # Launch analysis in package directory
  generic.process(
    input = system.file("extdata/fsa_SNP", package="FSAtools"),
    design = system.file("extdata/design_SNP.conf", package="FSAtools"),
    output = output
  )
  
  # List resulting files
  dir(dirname(output), full.names=TRUE)
</code></pre>

<hr>
<h2 id='genotype.fsa'>
Calls alleles for a SNP genotyping experiment
</h2><span id='topic+genotype.closest.fsa'></span><span id='topic+genotype.ratio.fsa'></span><span id='topic+genotype.N1.fsa'></span>

<h3>Description</h3>

<p>Calls alleles in experiments where fragments of different sizes are expected according to the allele.
</p>
<p><code>genotype.closest.fsa</code> selects for each allele the closest expected normalized peak height among the 0, 1 and 2 copy values provided in the design.
</p>
<p><code>genotype.ratio.fsa</code> calls an allele as absent, heterozygous or homozygous using fixed thresholds on the proportion of signal from a locus.
</p>
<p><code>genotype.N1.fsa</code> calls an allele as present if it exceeds a given proportion of the expected signal for 1 copy, as provided in the design. When a single allele is called present for a locus, it is considered homozygous.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  genotype.closest.fsa(x)
  genotype.ratio.fsa(x, homo = 0.85, hetero = c(0.3, 0.7))
  genotype.N1.fsa(x, threshold = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genotype.fsa_+3A_x">x</code></td>
<td>

<p>The <code>fsa</code> object to use, which must have a <code>peaks</code> attribute (see <code><a href="#topic+peaks.fsa">peaks.fsa</a></code>). Peaks are expected to be named according to the <code>LOCUS_ALLELE</code> pattern. <code>genotype.closest.fsa()</code> expected optional N0, N1 and N2 columns with expected normalized heights for 0, 1 and 2 copies of the allele.
</p>
</td></tr>
<tr><td><code id="genotype.fsa_+3A_homo">homo</code></td>
<td>

<p>Single numeric value, the ratio of the considered allele signal over all signal from this locus to call a homozygous allele. Similarly alleles below <code>1 - homo</code> will be called &quot;absent&quot;.
</p>
</td></tr>
<tr><td><code id="genotype.fsa_+3A_hetero">hetero</code></td>
<td>

<p>Numeric vector of length two, minimum and maximum ratios of the considered allele signal over all signal from this locus to call a heterozygous allele.
</p>
</td></tr>
<tr><td><code id="genotype.fsa_+3A_threshold">threshold</code></td>
<td>

<p>Single numeric value, the ratio of the considered allele signal over the expected value for 1n to call an allele.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>x</code>, with a new or updated <code>genotypes</code> attribute, a <code>data.frame</code> with a row for each locus :
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>The concatenation of the two called alleles, possibly with <code>?</code>.</p>
</td></tr>
<tr><td><code>alleles</code></td>
<td>
<p>The comma-separated list of ratios observed for all alleles (<code>genotype.ratio.fsa()</code> only).</p>
</td></tr>
</table>
<p>A <code>calls</code> vector attribute is also set, corresponding to the <code>call</code> column only.
</p>


<h3>Note</h3>

<p>All three functions assume peaks to be named according to the following convention : &quot;ALLELE - LOCUS&quot;.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generic.process">generic.process</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Using a design file
  design &lt;- designFile(system.file("extdata/design_SNP.conf", package="FSAtools"))
  
  # Example FSA file provided
  fsa &lt;- read.fsa(system.file("extdata/fsa_SNP/A7840.fsa", package="FSAtools"))
  fsa &lt;- align.fsa(
    fsa,
    channel = design$align.fsa$channel,
    outThreshold = design$align.fsa$outThreshold,
    useLadder = design$align.fsa$useLadder
  )
  fsa &lt;- peaks.fsa(fsa, peaks=design$GLOBALS$PEAKS)
  
  # Genotype
  fsa &lt;- genotype.ratio.fsa(fsa)
  print(attr(fsa, "genotypes"))
  print(attr(fsa, "calls"))
</code></pre>

<hr>
<h2 id='multiplot'>
Wrapper to layout
</h2><span id='topic+multiplot'></span>

<h3>Description</h3>

<p>Calls <code><a href="graphics.html#topic+layout">layout</a></code> using atomic arguments and convenient defaults, mainly to be included in <code><a href="#topic+generic.process">generic.process</a></code> design files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  multiplot(nrow, ncol, widths = rep.int(1, ncol), heights = rep.int(1, nrow),
    indexes = 1:(nrow * ncol), byrow = FALSE, respect = FALSE, cex = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiplot_+3A_nrow">nrow</code></td>
<td>

<p>Single integer value, the amount of rows into which divide the screen for multiple plots.
</p>
</td></tr>
<tr><td><code id="multiplot_+3A_ncol">ncol</code></td>
<td>

<p>Single integer value, the amount of columns into which divide the screen for multiple plots.
</p>
</td></tr>
<tr><td><code id="multiplot_+3A_widths">widths</code></td>
<td>

<p>To be passed to <code><a href="graphics.html#topic+layout">layout</a></code>.
</p>
</td></tr>
<tr><td><code id="multiplot_+3A_heights">heights</code></td>
<td>

<p>To be passed to <code><a href="graphics.html#topic+layout">layout</a></code>.
</p>
</td></tr>
<tr><td><code id="multiplot_+3A_indexes">indexes</code></td>
<td>

<p>Integer vector, the ordering of plots while building the <code>mat</code> matrix for <code><a href="graphics.html#topic+layout">layout</a></code>.
</p>
</td></tr>
<tr><td><code id="multiplot_+3A_byrow">byrow</code></td>
<td>

<p>Single logical value, whether to fill the <code>mat</code> matrix for <code><a href="graphics.html#topic+layout">layout</a></code> with <code>indexes</code> by row or by column.
</p>
</td></tr>
<tr><td><code id="multiplot_+3A_respect">respect</code></td>
<td>

<p>To be passed to <code><a href="graphics.html#topic+layout">layout</a></code>.
</p>
</td></tr>
<tr><td><code id="multiplot_+3A_cex">cex</code></td>
<td>

<p>Single numeric value, if not <code>NA</code> <code><a href="graphics.html#topic+par">par</a></code> will be called to force this value, as large layout automatically change it.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns <code>TRUE</code> on success.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generic.process">generic.process</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  multiplot(nrow=1, ncol=3, widths=c(1,2,2))
  plot(1:5)
  plot(1:5)
  plot(1:5)
</code></pre>

<hr>
<h2 id='peaks.fsa'>
Get maximal value in ranges
</h2><span id='topic+peaks.fsa'></span>

<h3>Description</h3>

<p>Look for the maximal value in one or many ranges, typically for peak detection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  peaks.fsa(x, peaks, names, size.min, size.max, channels, colors,
    logTransform = FALSE, lowThreshold = 1000, noiseRange = c(-10, 0))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peaks.fsa_+3A_x">x</code></td>
<td>

<p>An aligned object of class <code>fsa</code>, as returned by <code><a href="#topic+align.fsa">align.fsa</a></code>.
</p>
</td></tr>
<tr><td><code id="peaks.fsa_+3A_peaks">peaks</code></td>
<td>

<p>A <code>data.frame</code> with one row for each peak to consider. Can be provided instead of individual <code>names</code> (corresponding to <code>peaks</code> row names instead of a column), <code>size.min</code>, <code>size.max</code>, <code>channels</code> and <code>colors</code> argument vectors.
</p>
</td></tr>
<tr><td><code id="peaks.fsa_+3A_names">names</code></td>
<td>

<p>Character vector, the names to give to the peaks (duplicated values should be avoided). If <code>peaks</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="peaks.fsa_+3A_size.min">size.min</code></td>
<td>

<p>Numeric vector, the minimal size (in base pairs) to look for the corresponding peak. If <code>peaks</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="peaks.fsa_+3A_size.max">size.max</code></td>
<td>

<p>Numeric vector, the maximal size (in base pairs) to look for the corresponding peak. If <code>peaks</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="peaks.fsa_+3A_channels">channels</code></td>
<td>

<p>Character vector, the name of the channel in <code>x</code> in which to look for the corresponding peak. If <code>peaks</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="peaks.fsa_+3A_colors">colors</code></td>
<td>

<p>Vector defining the color to use in future plots to highlight the corresponding peak. If <code>peaks</code> is provided, this vector will be used instead.
</p>
</td></tr>
<tr><td><code id="peaks.fsa_+3A_logtransform">logTransform</code></td>
<td>

<p>Single logical value, whether to apply log transformation (base 2) to normalized values (previously floored to 0 and summed with 1) or not.
</p>
</td></tr>
<tr><td><code id="peaks.fsa_+3A_lowthreshold">lowThreshold</code></td>
<td>

<p>Single numeric value, threshold for which &quot;low profile&quot; warnings are called if all peaks are lower.
</p>
</td></tr>
<tr><td><code id="peaks.fsa_+3A_noiserange">noiseRange</code></td>
<td>

<p>Numeric vector of length 2, defining the range (relative to the starting range of the first peak defined in <code>ranges</code>) in which measure the noise (in bp). If the noise peak is 20 percent greater than the first peak, a warning is raised as the accuracy of the measure may be compromised.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>x</code>, with a new or updated <code>peaks</code> attribute, a <code>data.frame</code> with a row for each range :
</p>
<table>
<tr><td><code>size.min</code></td>
<td>
<p>User-provided argument.</p>
</td></tr>
<tr><td><code>size.max</code></td>
<td>
<p>User-provided argument.</p>
</td></tr>
<tr><td><code>channels</code></td>
<td>
<p>User-provided argument.</p>
</td></tr>
<tr><td><code>colors</code></td>
<td>
<p>User-provided argument.</p>
</td></tr>
<tr><td><code>size</code></td>
<td>
<p>Size at which the maximum was found, in base pairs.</p>
</td></tr>
<tr><td><code>height</code></td>
<td>
<p>Maximum found, in fluorescence units.</p>
</td></tr>
<tr><td><code>offScale</code></td>
<td>
<p>Is there any off-scale value in the range ?</p>
</td></tr>
<tr><td><code>normalized</code></td>
<td>
<p>Current peak's height divided by the mean of all peak heights.</p>
</td></tr>
</table>
<p>A <code>normalized</code> vector attribute is also set, corresponding to the <code>normalized</code> column only.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generic.process">generic.process</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Example FSA file provided
  fsa &lt;- read.fsa(system.file("extdata/fsa_GEP/A5918.fsa", package="FSAtools"))
  fsa &lt;- align.fsa(fsa)
  
  # Single custom interval
  fsa &lt;- peaks.fsa(
    fsa,
    names = "IRF4",
    size.min = 86.2,
    size.max = 87.5,
    channels = "6-FAM",
    colors = "blue"
  )
  print(attr(fsa, "peaks"))
  
  # Using a design file
  design &lt;- designFile(system.file("extdata/design_GEP.conf", package="FSAtools"))
  fsa &lt;- peaks.fsa(fsa, peaks=design$GLOBALS$PEAKS)
  print(attr(fsa, "peaks"))
</code></pre>

<hr>
<h2 id='plot.fsa'>
Plot method for &quot;fsa&quot; objects
</h2><span id='topic+plot.fsa'></span>

<h3>Description</h3>

<p>Plots a <code>fsa</code> object. For each selected channel, a line is drawn bewteen measured fluorescence intensities (y axis) along the electrophoresis time (x axis).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fsa'
plot(x, units = NA, channels = NA, chanColors = NA, ladder = TRUE,
    offScaleCol = "#FF0000", offScalePch = "+", offScaleCex = 0.4, bg = "white",
    fg = "black", title = "", title.adj = 0, title.line = NA, xlab = NA,
    ylab = "Intensity", xlim = NA, ylim = NA, xaxt = "s", yaxt = "s", bty = "o",
    xaxp = NA, nticks = 5, all.bp = TRUE, peaks.alpha = 48L, peaks.srt = 30,
    peaks.adj = c(0, 0), peaks.cex = 1.3, peaks.font = 2, legend.x = "topleft", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.fsa_+3A_x">x</code></td>
<td>

<p>The <code>fsa</code> object to plot.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_units">units</code></td>
<td>

<p>Single character value, the unit to use on x axis. &quot;index&quot; uses the raw index contained in files, &quot;bp&quot; usess base pair estimations but needs the object to be aligned first using <code>align.fsa</code>. <code>NA</code> will select &quot;bp&quot; if <code>x</code> is aligned, &quot;index&quot; elsewhere.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_channels">channels</code></td>
<td>

<p>Character or integer vector, the channels to plot. If <code>NA</code>, all channels are selected.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_chancolors">chanColors</code></td>
<td>

<p>Character vector defining colors to use to plot channels. Can be named according to channel names stored in <code>x</code>, or parallel with <code>channels</code> (first color for first channel, etc, no recycling). If <code>NA</code>, colors stored in <code>x</code> are used. See the <code>col</code> argument in <code><a href="graphics.html#topic+par">par</a></code> for further details on allowed values.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_ladder">ladder</code></td>
<td>

<p>Single logical value, whether to add an x axis with size ladder peaks or not. Raises a warning if <code>x</code> was not aligned before plotting.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_offscalecol">offScaleCol</code></td>
<td>

<p>To be passed to <code><a href="graphics.html#topic+points">points</a></code> for off-scale value plot (see <code><a href="graphics.html#topic+par">par</a></code> for allowed values).
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_offscalepch">offScalePch</code></td>
<td>

<p>To be passed to <code><a href="graphics.html#topic+points">points</a></code> for off-scale value plot (see <code><a href="graphics.html#topic+par">par</a></code> for allowed values).
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_offscalecex">offScaleCex</code></td>
<td>

<p>To be passed to <code><a href="graphics.html#topic+points">points</a></code> for off-scale value plot (see <code><a href="graphics.html#topic+par">par</a></code> for allowed values).
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_bg">bg</code></td>
<td>

<p>See <code><a href="graphics.html#topic+par">par</a></code> for further details.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_fg">fg</code></td>
<td>

<p>See <code><a href="graphics.html#topic+par">par</a></code> for further details. This value is also used for col.axis, col.lab, col.main and col.sub graphical parameters.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_title">title</code></td>
<td>

<p>Single character value, the main title to print on the plot.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_title.adj">title.adj</code></td>
<td>

<p>To be passed as <code>adj</code> to <code><a href="graphics.html#topic+title">title</a></code>.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_title.line">title.line</code></td>
<td>

<p>To be passed as <code>line</code> to <code><a href="graphics.html#topic+title">title</a></code>.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_xlab">xlab</code></td>
<td>

<p>See <code><a href="graphics.html#topic+plot">plot</a></code> for further details. If <code>NA</code>, <code>units</code> is used.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_ylab">ylab</code></td>
<td>

<p>See <code><a href="graphics.html#topic+plot">plot</a></code> for further details.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_xlim">xlim</code></td>
<td>

<p>See <code><a href="graphics.html#topic+plot">plot</a></code> for further details. If <code>NA</code>, <code>x</code> range is used.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_ylim">ylim</code></td>
<td>

<p>See <code><a href="graphics.html#topic+plot">plot</a></code> for further details. If <code>NA</code>, <code>x</code> range is used.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_xaxt">xaxt</code></td>
<td>

<p>See <code><a href="graphics.html#topic+par">par</a></code> for further details.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_yaxt">yaxt</code></td>
<td>

<p>See <code><a href="graphics.html#topic+par">par</a></code> for further details.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_bty">bty</code></td>
<td>

<p>See <code><a href="graphics.html#topic+par">par</a></code> for further details.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_xaxp">xaxp</code></td>
<td>

<p>See <code><a href="graphics.html#topic+par">par</a></code> for further details. If <code>NA</code>, a suitable value is computed.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_nticks">nticks</code></td>
<td>

<p>Single integer value. When <code>xaxp</code> is <code>NA</code> and <code>units</code> is &quot;bp&quot;, this values fixes the interval between X axis labels.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_all.bp">all.bp</code></td>
<td>

<p>Single logical value, whether to force an unlabeled axis tick at each bp when <code>units</code> is &quot;bp&quot; or not.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_peaks.alpha">peaks.alpha</code></td>
<td>

<p>Single integer value, the alpha channel to add to peak colors to make a background (255 is no transparency at all, 0 is invisible).
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_peaks.srt">peaks.srt</code></td>
<td>

<p>To be passed as <code>srt</code> to <code><a href="graphics.html#topic+text">text</a></code> while printing peak names.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_peaks.adj">peaks.adj</code></td>
<td>

<p>To be passed as <code>adj</code> to <code><a href="graphics.html#topic+text">text</a></code> while printing peak names.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_peaks.cex">peaks.cex</code></td>
<td>

<p>To be passed as <code>cex</code> to <code><a href="graphics.html#topic+text">text</a></code> while printing peak names.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_peaks.font">peaks.font</code></td>
<td>

<p>To be passed as <code>font</code> to <code><a href="graphics.html#topic+text">text</a></code> while printing peak names.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_legend.x">legend.x</code></td>
<td>

<p>To be passed as <code>x</code> to <code><a href="graphics.html#topic+legend">legend</a></code>.
</p>
</td></tr>
<tr><td><code id="plot.fsa_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="graphics.html#topic+plot">plot</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns <code>TRUE</code> on success.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fsa">read.fsa</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Example FSA file provided
  fsa &lt;- read.fsa(system.file("extdata/fsa_GEP/A5918.fsa", package="FSAtools"))
  
  # Plot whole profile
  plot(fsa)
  
  # Plot subset of the profile (time index)
  plot(fsa, units="index", xlim=c(4000,5000))
  
  # Plot subset of the profile (base pairs)
  fsa &lt;- align.fsa(fsa)
  plot(fsa, units="bp", xlim=c(80,130))
</code></pre>

<hr>
<h2 id='plot.fsaModel'>
Plot method for &quot;fsaModel&quot; objects
</h2><span id='topic+plot.fsaModel'></span>

<h3>Description</h3>

<p>Plots a <code>fsaModel</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fsaModel'
plot(x, xlab = "Score", lwd = 3, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.fsaModel_+3A_x">x</code></td>
<td>

<p>The <code>fsaModel</code> object to plot.
</p>
</td></tr>
<tr><td><code id="plot.fsaModel_+3A_xlab">xlab</code></td>
<td>

<p>To be passed to <code><a href="graphics.html#topic+plot">plot</a></code>.
</p>
</td></tr>
<tr><td><code id="plot.fsaModel_+3A_lwd">lwd</code></td>
<td>

<p>To be passed to <code><a href="graphics.html#topic+plot">plot</a></code>.
</p>
</td></tr>
<tr><td><code id="plot.fsaModel_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="graphics.html#topic+plot">plot</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns <code>TRUE</code> on success.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+train">train</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Example FSA file provided
  fsa &lt;- read.fsa(system.file("extdata/fsa_GEP/A5918.fsa", package="FSAtools"))
  
  # Add model from design file
  design &lt;- designFile(system.file("extdata/design_GEP.conf", package="FSAtools"))
  fsa &lt;- add.model(fsa, model=design$GLOBALS$MODEL)
  
  # Plot model
  plot(attr(fsa, "model"))
</code></pre>

<hr>
<h2 id='print.fsa'>
Print method for &quot;fsa&quot; objects
</h2><span id='topic+print.fsa'></span>

<h3>Description</h3>

<p>Prints a short summary of an <code>fsa</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fsa'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.fsa_+3A_x">x</code></td>
<td>

<p>The <code>fsa</code> object to print.
</p>
</td></tr>
<tr><td><code id="print.fsa_+3A_...">...</code></td>
<td>

<p>Currently ignored.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns <code>TRUE</code> on success.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fsa">read.fsa</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Example FSA file provided
  fsa &lt;- read.fsa(system.file("extdata/fsa_GEP/A5918.fsa", package="FSAtools"))
  print(fsa)
  
  # Aligned version
  fsa &lt;- align.fsa(fsa)
  print(fsa)
</code></pre>

<hr>
<h2 id='read.abif'>Read ABIF formatted files</h2><span id='topic+read.abif'></span>

<h3>Description</h3>

<p>ABIF stands for Applied Biosystem Inc. Format, a binary format modeled after TIFF format.
Corresponding files usually have an <code>*.ab1</code> or <code>*.fsa</code> extension.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  read.abif(filename, max.bytes.in.file = file.info(filename)$size,
    pied.de.pilote = 1.2, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.abif_+3A_filename">filename</code></td>
<td>
<p>The name of the file.</p>
</td></tr>
<tr><td><code id="read.abif_+3A_max.bytes.in.file">max.bytes.in.file</code></td>
<td>
<p>The size in bytes of the file, defaulting to what is returned by <code><a href="base.html#topic+file.info">file.info</a></code></p>
</td></tr>	
<tr><td><code id="read.abif_+3A_pied.de.pilote">pied.de.pilote</code></td>
<td>
<p>Safety factor: the argument <code>n</code> to <code><a href="base.html#topic+readBin">readBin</a></code> is set as <code>pied.de.pilote*max.bytes.in.file</code>.</p>
</td></tr>	
<tr><td><code id="read.abif_+3A_verbose">verbose</code></td>
<td>
<p>logical [FALSE]. If TRUE verbose mode is on.</p>
</td></tr>	
</table>


<h3>Details</h3>

<p>All data are imported into memory, there is no attempt to read items on the fly.
</p>


<h3>Value</h3>

<p>A list with three components: <code>Header</code> which is a list that contains various low-level information, among which <code>numelements</code> is the number of elements in the directory and <code>dataoffset</code> the offset to find the location of the directory. <code>Directory</code> is a data.frame for the directory of the file with the number of row being the number of elements in the directory and the 7 columns describing various low-level information about the elements. <code>Data</code> is a list with the number of components equal to the number of elements in the directory.
</p>


<h3>Note</h3>

<p>This function and the current help page were duplicated from the <span class="pkg">seqinr</span> package in its 3.0-7 version (available on the CRAN under GPL 2 licensing).
</p>


<h3>Author(s)</h3>

<p>J.R. Lobry, 'bool' type implemented by Sylvain Mareschal
</p>


<h3>References</h3>

 
<p>Charif, D. and Lobry, J.R. (2007) <cite>Structural approaches to sequence evolution: Molecules, networks, populations</cite> ISBN 978-3-540-35305-8, pp 207-232.
</p>
<p>Anonymous (2006) Applied Biosystem Genetic Analysis Data File Format.
Available at <a href="https://projects.nfstc.org/workshops/resources/articles/ABIF_File_Format.pdf">https://projects.nfstc.org/workshops/resources/articles/ABIF_File_Format.pdf</a>.
Last visited on 2020-07-10.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+readBin">readBin</a></code> which is used here to import the binary file and <code><a href="base.html#topic+file.info">file.info</a></code> to get the size of the file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Example FSA file provided
  rawFsa &lt;- read.abif(system.file("extdata/fsa_GEP/A5918.fsa", package="FSAtools"))
</code></pre>

<hr>
<h2 id='read.fsa'>
Imports a .fsa file from Applied Biosystems
</h2><span id='topic+read.fsa'></span>

<h3>Description</h3>

<p>This function parses a FSA file holding fragment analysis data, using <span class="pkg">seqinr</span> package's <code><a href="#topic+read.abif">read.abif</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  read.fsa(file, lowess = TRUE, lowess.top = 200, processed = FALSE, meta.extra = NULL,
    quiet = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.fsa_+3A_file">file</code></td>
<td>

<p>Single character value, the name and path of the file to parse.
</p>
</td></tr>
<tr><td><code id="read.fsa_+3A_lowess">lowess</code></td>
<td>

<p>Single logical value, whether to apply <code><a href="stats.html#topic+lowess">lowess</a></code> on intensities to smooth time-related biases or not.
</p>
</td></tr>
<tr><td><code id="read.fsa_+3A_lowess.top">lowess.top</code></td>
<td>

<p>Single numeric value, values flagged as &quot;off-scale&quot; or above this threshold will be replaced by <code>lowess.top</code> to compute the lowess smooth, in order to limit the impact of high and wide peaks.
</p>
</td></tr>
<tr><td><code id="read.fsa_+3A_processed">processed</code></td>
<td>

<p>Single logical value, whether to use processed DATA values (as stored in sets 9 to 12, not always available) rather than raw values (sets 1 to 4). If <code>NA</code>, processed ones will be used as long as they are available, else raw ones will be used instead.
</p>
</td></tr>
<tr><td><code id="read.fsa_+3A_meta.extra">meta.extra</code></td>
<td>

<p>Named character vector, defining which extra fields to extract to populate the <code>runMetaData</code> attribute. The vector names define the human-readable names to use in output, the vector values provide the 4 uppercase letter code to extract (all values will be gathered in a vector if the code is used several times). See the reference provided in <code><a href="#topic+read.abif">read.abif</a></code> for existing codes in the ABIF file format.
</p>
</td></tr>
<tr><td><code id="read.fsa_+3A_quiet">quiet</code></td>
<td>

<p>Single logical value, whether to print FSA meta-data read from the file or not.
</p>
</td></tr>
<tr><td><code id="read.fsa_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to <code><a href="#topic+read.abif">read.abif</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A S3 object of class <code>fsa</code>
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.abif">read.abif</a></code>, <code><a href="#topic+generic.process">generic.process</a></code>, <code><a href="#topic+plot.fsa">plot.fsa</a></code>, <code><a href="#topic+read.sanger">read.sanger</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Example FSA file provided
  fsa &lt;- read.fsa(system.file("extdata/fsa_GEP/A5918.fsa", package="FSAtools"))
  print(fsa)
</code></pre>

<hr>
<h2 id='read.sanger'>
Imports a .ab1 file from Applied Biosystems corresponding to Sanger sequencing
</h2><span id='topic+read.sanger'></span>

<h3>Description</h3>

<p>This function parses a FSA/AB1 file using <code><a href="#topic+read.fsa">read.fsa</a></code>, with few adjustments for Sanger sequencing experiments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  read.sanger(file, channelOrder = NULL, guess.threshold = 0.3, processed = NA,
     lowess = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.sanger_+3A_file">file</code></td>
<td>

<p>Single character value, the name and path of the file to parse.
</p>
</td></tr>
<tr><td><code id="read.sanger_+3A_channelorder">channelOrder</code></td>
<td>

<p>Character vector, providing 'A', 'C', 'G' and 'T' in the order of the used channels. If <code>NULL</code>, a guess will be attempted based on the called sequence.
</p>
</td></tr>
<tr><td><code id="read.sanger_+3A_guess.threshold">guess.threshold</code></td>
<td>

<p>Single numeric value, setting the tolerance to use for channel guessing validation. Lower values mean higher chances to get an error for channel guessing failure.
</p>
</td></tr>
<tr><td><code id="read.sanger_+3A_processed">processed</code></td>
<td>

<p>To be passed to <code><a href="#topic+read.fsa">read.fsa</a></code>.
</p>
</td></tr>
<tr><td><code id="read.sanger_+3A_lowess">lowess</code></td>
<td>

<p>To be passed to <code><a href="#topic+read.fsa">read.fsa</a></code>.
</p>
</td></tr>
<tr><td><code id="read.sanger_+3A_...">...</code></td>
<td>

<p>To be passed to <code><a href="#topic+read.fsa">read.fsa</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A S3 object of class <code>fsa</code>
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fsa">read.fsa</a></code>, <code><a href="#topic+read.abif">read.abif</a></code>
</p>

<hr>
<h2 id='train'>
Training function for binary predictors
</h2><span id='topic+train'></span>

<h3>Description</h3>

<p>This function build a model from data to predict class in <code><a href="#topic+classify">classify</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  train(peakMatrix, group, filter.p = 0.05, groupColors = c("red", "blue"),
    threshold = 0.9)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="train_+3A_peakmatrix">peakMatrix</code></td>
<td>

<p>Numeric matrix of normalized peak heights with samples in rows and peaks in columns.
</p>
</td></tr>
<tr><td><code id="train_+3A_group">group</code></td>
<td>

<p>Two-level factor defining the group of every samples in <code>peaks</code>.
</p>
</td></tr>
<tr><td><code id="train_+3A_filter.p">filter.p</code></td>
<td>

<p>Single numeric value, if not <code>NA</code> only genes for which the t-test p is lower than this will be used in the model.
</p>
</td></tr>
<tr><td><code id="train_+3A_groupcolors">groupColors</code></td>
<td>

<p>Vector of length two, defining the colors to use to represent the two groups in future plots.
</p>
</td></tr>
<tr><td><code id="train_+3A_threshold">threshold</code></td>
<td>

<p>Single numeric value, the likelihood threshold above which make a call when classifying (classification will return <code>NA</code> is this threshold is met for none of the two groups).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an S3 object of class <code>fsaModel</code>.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add.model">add.model</a></code>, <code><a href="#topic+classify">classify</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # Underlying truth for pseudo-data (10 genes)
  geneNames &lt;- paste("gene", LETTERS[1:10], sep=".")
  geneMean &lt;- abs(rnorm(10))
  groupShift &lt;- rnorm(10, sd=0.1)
  
  # Generate pseudo-data for 50 samples
  mtx &lt;- NULL
  for(g in 1:10) {
    x &lt;- rnorm(n=50, mean=geneMean[g], sd=0.1)
    x[1:25] &lt;- x[1:25] + groupShift[g]
    x[26:50] &lt;- x[26:50] - groupShift[g]
    mtx &lt;- cbind(mtx, x)
  }
  colnames(mtx) &lt;- geneNames
  rownames(mtx) &lt;- c(
    paste("group1", 1:25, sep="."),
    paste("group2", 26:50, sep=".")
  )
  
  # Train model
  group &lt;- c(
    rep("group1", 25),
    rep("group2", 25)
  )
  model &lt;- train(mtx, group)
  plot(model)
  
  # Compare model to truth
  i &lt;- match(geneNames, model$geneNames)
  out &lt;- data.frame(
    gene = geneNames,
    true.M = geneMean,
    model.M = model$geneMs[i],
    true.shift = groupShift,
    model.T = model$geneTs[i]
  )
  print(out)
</code></pre>

<hr>
<h2 id='wav2RGB'>
Converts light wavelengths to RGB colors
</h2><span id='topic+wav2RGB'></span>

<h3>Description</h3>

<p>Converts wavelengths in nanometers into corresponding visible colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  wav2RGB(wav)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wav2RGB_+3A_wav">wav</code></td>
<td>

<p>Numeric vector of wavelengths (in nanometers) to convert into colors.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character vector of the same length as <code>wav</code>, with an RGB color for each wavelength. Wavelengths out of visible ranges return black.
</p>


<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>References</h3>

<p><a href="http://codingmess.blogspot.fr/2009/05/conversion-of-wavelength-in-nanometers.html">http://codingmess.blogspot.fr/2009/05/conversion-of-wavelength-in-nanometers.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  wv &lt;- seq(from=300, to=800, by=10)
  plot(x=wv, y=rep(1, length(wv)), col=wav2RGB(wv), pch=19)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
