<!DOCTYPE html><html><head><title>Help for package fst4pg</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {fst4pg}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fst4pg-package'><p>fst4pg: Genetic Distance Segmentation for Population Genetics</p></a></li>
<li><a href='#BuildFreqNbG'><p>Convert the Freq and NbGamete tables into a list.</p></a></li>
<li><a href='#Compute_Denominator'><p>Computation of the numerator of the moment estimator</p></a></li>
<li><a href='#Compute_Nominator'><p>Computation of the numerator of the moment estimator</p></a></li>
<li><a href='#ContrastGraphSummary'><p>ContrastGraphSummary</p></a></li>
<li><a href='#ContrastSummary'><p>ContrastSummary</p></a></li>
<li><a href='#ContrastTopRegions'><p>ContrastTopRegions</p></a></li>
<li><a href='#DF4Plot1Prof'><p>DF4Plot1Prof</p></a></li>
<li><a href='#Freq'><p>Frequencies of the American and European HGDP populations</p></a></li>
<li><a href='#Freq.filt'><p>Filtering markers based on allelic frequencies</p></a></li>
<li><a href='#HeatMap'><p>HeatMap</p></a></li>
<li><a href='#HudsonFst.gw'><p>HudsonFst.gw</p></a></li>
<li><a href='#HudsonFst.m'><p>HudsonFst.m</p></a></li>
<li><a href='#HudsonFst.plot'><p>Plot Fst values along chromosomes</p></a></li>
<li><a href='#HudsonFst.prof'><p>HudsonFst.prof</p></a></li>
<li><a href='#Info'><p>Marker information</p></a></li>
<li><a href='#MakeProfile.op'><p>MakeProfile.op</p></a></li>
<li><a href='#MakeProfile.sn_nomemory'><p>MakeProfile.sn_nomemory</p></a></li>
<li><a href='#MergeRegion'><p>MergeRegion</p></a></li>
<li><a href='#NbGamete'><p>Number of gametes of the American and European HGDP populations</p></a></li>
<li><a href='#Plot1Prof'><p>Plot1Prof</p></a></li>
<li><a href='#ProfilingSummary'><p>ProfilingSummary</p></a></li>
<li><a href='#Ratio_Average'><p>Computation of the Fst moment estimator</p></a></li>
<li><a href='#RawPlot'><p>RawPlot</p></a></li>
<li><a href='#Summarise1Profile'><p>Summarise1Profile</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Genetic Distance Segmentation for Population Genetics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-06-15</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides efficient methods to compute local and genome wide
    genetic distances (corresponding to the so called Hudson Fst
    parameters) through moment method, perform chromosome segmentation
    into homogeneous Fst genomic regions, and selection sweep detection
    for multi-population comparison. When multiple profile segmentation is
    required, the procedure can be parallelized using the future package.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, fpopw, furrr, future, ggplot2, gplots, graphics,
grDevices, purrr, rlang, scales, stats, stringr, tibble, tidyr,
utils</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-30 16:24:08 UTC; smoka</td>
</tr>
<tr>
<td>Author:</td>
<td>Tristan Mary-Huard
    <a href="https://orcid.org/0000-0002-3839-9067"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Guillem Rigaill <a href="https://orcid.org/0000-0002-7176-7511"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tristan Mary-Huard &lt;tristan.mary-huard@agroparistech.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-07-03 18:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='fst4pg-package'>fst4pg: Genetic Distance Segmentation for Population Genetics</h2><span id='topic+fst4pg'></span><span id='topic+fst4pg-package'></span>

<h3>Description</h3>

<p>Provides efficient methods to compute local and genome wide genetic distances (corresponding to the so called Hudson Fst parameters) through moment method, perform chromosome segmentation into homogeneous Fst genomic regions, and selection sweep detection for multi-population comparison. When multiple profile segmentation is required, the procedure can be parallelized using the future package.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Tristan Mary-Huard <a href="mailto:tristan.mary-huard@agroparistech.fr">tristan.mary-huard@agroparistech.fr</a> (<a href="https://orcid.org/0000-0002-3839-9067">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Guillem Rigaill <a href="mailto:guillem.rigaill@inrae.fr">guillem.rigaill@inrae.fr</a> (<a href="https://orcid.org/0000-0002-7176-7511">ORCID</a>)
</p>
</li></ul>


<hr>
<h2 id='BuildFreqNbG'>Convert the Freq and NbGamete tables into a list.</h2><span id='topic+BuildFreqNbG'></span>

<h3>Description</h3>

<p>The function builds a list where each element corresponds to a population present in both Freq and NbGametes (all other
populations being discarded). Each element consists of a data.frame with 2 columns, Freq and NbGamete.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BuildFreqNbG(Freq, NbGamete)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BuildFreqNbG_+3A_freq">Freq</code></td>
<td>
<p>A data.frame or matrix of frequencies where each row corresponds to a marker, each column corresponds to a population,</p>
</td></tr>
<tr><td><code id="BuildFreqNbG_+3A_nbgamete">NbGamete</code></td>
<td>
<p>A data.frame or matrix of number of gametes where each row corresponds to a marker, and each column corresponds to a population</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of data.frames, each corresponding to a population.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load the HGDP data 
data(Freq);data(NbGamete)
FreqNbG &lt;- BuildFreqNbG(Freq,NbGamete)
</code></pre>

<hr>
<h2 id='Compute_Denominator'>Computation of the numerator of the moment estimator</h2><span id='topic+Compute_Denominator'></span>

<h3>Description</h3>

<p>Computation of the numerator of the moment estimator
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Compute_Denominator(p1, p2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Compute_Denominator_+3A_p1">p1</code></td>
<td>
<p>numeric, frequencies in population 1</p>
</td></tr>
<tr><td><code id="Compute_Denominator_+3A_p2">p2</code></td>
<td>
<p>numeric, frequencies in population 2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with the denominators of the Fst moment estimator
</p>

<hr>
<h2 id='Compute_Nominator'>Computation of the numerator of the moment estimator</h2><span id='topic+Compute_Nominator'></span>

<h3>Description</h3>

<p>Computation of the numerator of the moment estimator
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Compute_Nominator(p1, p2, n1, n2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Compute_Nominator_+3A_p1">p1</code></td>
<td>
<p>numeric, frequencies in population 1</p>
</td></tr>
<tr><td><code id="Compute_Nominator_+3A_p2">p2</code></td>
<td>
<p>numeric, frequencies in population 2</p>
</td></tr>
<tr><td><code id="Compute_Nominator_+3A_n1">n1</code></td>
<td>
<p>numeric, number of gametes in population 1</p>
</td></tr>
<tr><td><code id="Compute_Nominator_+3A_n2">n2</code></td>
<td>
<p>numeric, number of gametes in population 2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with the  numerators of the Fst moment estimator
</p>

<hr>
<h2 id='ContrastGraphSummary'>ContrastGraphSummary</h2><span id='topic+ContrastGraphSummary'></span>

<h3>Description</h3>

<p>Display mean ratio and/or number of selection graphs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ContrastGraphSummary(
  CS,
  Info,
  Ratio.thres,
  Coef = 1,
  CutNbSel = NULL,
  CutMeanRatio = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ContrastGraphSummary_+3A_cs">CS</code></td>
<td>
<p>a contrast summary, as provided by function <code>ContrastSummary</code></p>
</td></tr>
<tr><td><code id="ContrastGraphSummary_+3A_info">Info</code></td>
<td>
<p>a data.frame providing information about markers</p>
</td></tr>
<tr><td><code id="ContrastGraphSummary_+3A_ratio.thres">Ratio.thres</code></td>
<td>
<p>a numeric value, regions exhibiting Fst levels whose ratio with the reference level is 
higher than Ratio.thres will be highlighted.</p>
</td></tr>
<tr><td><code id="ContrastGraphSummary_+3A_coef">Coef</code></td>
<td>
<p>a scalar, controling font sizes for the graph, optional</p>
</td></tr>
<tr><td><code id="ContrastGraphSummary_+3A_cutnbsel">CutNbSel</code></td>
<td>
<p>a scalar, providing a y-value for an horizontal line on the NbSel graph</p>
</td></tr>
<tr><td><code id="ContrastGraphSummary_+3A_cutmeanratio">CutMeanRatio</code></td>
<td>
<p>a scalar, providing a y-value for an horizontal line on the MeanRatio graph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>two ggplots objects, called NbSel and MeanRatio, respectively.
</p>

<hr>
<h2 id='ContrastSummary'>ContrastSummary</h2><span id='topic+ContrastSummary'></span>

<h3>Description</h3>

<p>Summarize multiple Fst profiles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ContrastSummary(PS, RefLevel, Ratio.thres = 3, NbSnp.min = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ContrastSummary_+3A_ps">PS</code></td>
<td>
<p>a list of profile summaries, as provided by the <code>ProfilingSummary</code> function</p>
</td></tr>
<tr><td><code id="ContrastSummary_+3A_reflevel">RefLevel</code></td>
<td>
<p>a list of reference (i.e. baseline) Fst levels</p>
</td></tr>
<tr><td><code id="ContrastSummary_+3A_ratio.thres">Ratio.thres</code></td>
<td>
<p>a numeric value, regions exhibiting Fst levels whose ratio with the reference level is 
higher than Ratio.thres will be highlighted.</p>
</td></tr>
<tr><td><code id="ContrastSummary_+3A_nbsnp.min">NbSnp.min</code></td>
<td>
<p>an integer. The minimum number of markers required to highlight a region</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The full example execution takes a few seconds.
# data(Freq);data(NbGamete)
# FreqNbG &lt;- BuildFreqNbG(Freq,NbGamete)
# HFst.m &lt;- HudsonFst.m(FreqNbG)

## Two sets of populations to contrast
# Contrast &lt;- list(America=c("Colombian","Maya"),Europe=c("Tuscan","Italian"))
# Profiles &lt;- HudsonFst.prof(HFst.m,Contrast=Contrast)
# PS &lt;- ProfilingSummary(Profiles,Info) 

# RefLevel &lt;- rapply(Profiles,median,classes = "numeric",how='list')
# Ratio.thres &lt;- 3
# NbSnp.min &lt;- 1
# CS &lt;- ContrastSummary(PS, RefLevel,
#                       Ratio.thres=Ratio.thres,
#                       NbSnp.min=NbSnp.min)
</code></pre>

<hr>
<h2 id='ContrastTopRegions'>ContrastTopRegions</h2><span id='topic+ContrastTopRegions'></span>

<h3>Description</h3>

<p>ContrastTopRegions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ContrastTopRegions(CS, Crit, Info, Thres, Simplify = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ContrastTopRegions_+3A_cs">CS</code></td>
<td>
<p>a list of contrast summaries as obtained from function <code>ContrastSummary</code></p>
</td></tr>
<tr><td><code id="ContrastTopRegions_+3A_crit">Crit</code></td>
<td>
<p>a string providing the name of the variable to use to select regions</p>
</td></tr>
<tr><td><code id="ContrastTopRegions_+3A_info">Info</code></td>
<td>
<p>a data.frame providing information about markers</p>
</td></tr>
<tr><td><code id="ContrastTopRegions_+3A_thres">Thres</code></td>
<td>
<p>the threshold to be used on the Crit variable</p>
</td></tr>
<tr><td><code id="ContrastTopRegions_+3A_simplify">Simplify</code></td>
<td>
<p>a boolean specifying whether the results should be displayed 
as a list (by-default option) or as a single data.frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame or a list of data.frames
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The full example execution takes a few seconds.
# data(Freq);data(NbGamete)
# FreqNbG &lt;- BuildFreqNbG(Freq,NbGamete)
# HFst.m &lt;- HudsonFst.m(FreqNbG)

## Two sets of populations to contrast
# Contrast &lt;- list(America=c("Colombian","Maya"),Europe=c("Tuscan","Italian"))
# Profiles &lt;- HudsonFst.prof(HFst.m,Contrast=Contrast)
# PS &lt;- ProfilingSummary(Profiles,Info) 

# RefLevel &lt;- rapply(Profiles,median,classes = "numeric",how='list')
# Ratio.thres &lt;- 3
# NbSnp.min &lt;- 1
# CS &lt;- ContrastSummary(PS, RefLevel,
#                       Ratio.thres=Ratio.thres,
#                       NbSnp.min=NbSnp.min)
# NbSel.thres &lt;- 2
# TopRegions &lt;- ContrastTopRegions(CS = CS,Crit = 'NbSel',Info = Info,
#                                 Thres = NbSel.thres, Simplify=TRUE)
</code></pre>

<hr>
<h2 id='DF4Plot1Prof'>DF4Plot1Prof</h2><span id='topic+DF4Plot1Prof'></span>

<h3>Description</h3>

<p>Shape the data for Fst profile representation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DF4Plot1Prof(Info, HF, FstProf, Coord = NULL, Threshold = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DF4Plot1Prof_+3A_info">Info</code></td>
<td>
<p>a data.frame providing information about markers</p>
</td></tr>
<tr><td><code id="DF4Plot1Prof_+3A_hf">HF</code></td>
<td>
<p>a data.frame with 2 columns Fst and Weight, as obtained from the <code>HudsonFst.m</code> function</p>
</td></tr>
<tr><td><code id="DF4Plot1Prof_+3A_fstprof">FstProf</code></td>
<td>
<p>an Fst profile, as obtained from the <code>HudsonFst.prof</code> function</p>
</td></tr>
<tr><td><code id="DF4Plot1Prof_+3A_coord">Coord</code></td>
<td>
<p>a vector with the minimum and maximum coordinates (i.e. positions along the genome), 
providing the range of the genomic region that will be plotted, optional.</p>
</td></tr>
<tr><td><code id="DF4Plot1Prof_+3A_threshold">Threshold</code></td>
<td>
<p>a numeric value. Markers belonging to regions whose Fst profile is higher than threshold
will be highlighted. Optional.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame that can be used as an input for function <code>Plot1Prof</code>
</p>

<hr>
<h2 id='Freq'>Frequencies of the American and European HGDP populations</h2><span id='topic+Freq'></span>

<h3>Description</h3>

<p><code>Freq</code> is a data.frame containing the frequencies of 49,636 markers located on chromosome 1, 
for the 13 American and European populations described in the Stanford HGDP dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Freq
</code></pre>


<h3>Format</h3>

<p>A data.frame
</p>

<hr>
<h2 id='Freq.filt'>Filtering markers based on allelic frequencies</h2><span id='topic+Freq.filt'></span>

<h3>Description</h3>

<p>Filtering markers based on allelic frequencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Freq.filt(FreqNbG, Maf = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Freq.filt_+3A_freqnbg">FreqNbG</code></td>
<td>
<p>a list of data.frames (one per population) with 2 columns: Freq and NbGamete</p>
</td></tr>
<tr><td><code id="Freq.filt_+3A_maf">Maf</code></td>
<td>
<p>a numerci value for the thresholding of minor allelic frequencies</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of positions to be removed
</p>

<hr>
<h2 id='HeatMap'>HeatMap</h2><span id='topic+HeatMap'></span>

<h3>Description</h3>

<p>Make a frequency heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HeatMap(
  Min,
  Max,
  chr = NULL,
  Info,
  FreqNbG,
  Dir = NULL,
  Weights = NULL,
  Weight.thres = 0.05,
  NbAdjM = 0,
  Subsets = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HeatMap_+3A_min">Min</code></td>
<td>
<p>the starting position value of the region</p>
</td></tr>
<tr><td><code id="HeatMap_+3A_max">Max</code></td>
<td>
<p>the end position value of the region</p>
</td></tr>
<tr><td><code id="HeatMap_+3A_chr">chr</code></td>
<td>
<p>a string providing the chromosome name, optional</p>
</td></tr>
<tr><td><code id="HeatMap_+3A_info">Info</code></td>
<td>
<p>a data.frame providing information about markers</p>
</td></tr>
<tr><td><code id="HeatMap_+3A_freqnbg">FreqNbG</code></td>
<td>
<p>a list of data.frames (one per population) with two columns: Freq and NbGamete</p>
</td></tr>
<tr><td><code id="HeatMap_+3A_dir">Dir</code></td>
<td>
<p>a string providing the name of the directory where the graph should be saved, optional</p>
</td></tr>
<tr><td><code id="HeatMap_+3A_weights">Weights</code></td>
<td>
<p>a vector of weights associated with each marker, optional</p>
</td></tr>
<tr><td><code id="HeatMap_+3A_weight.thres">Weight.thres</code></td>
<td>
<p>a numeric value. Markers with weights lower than this threshold
will be discarded from the graphical representation. Optional.</p>
</td></tr>
<tr><td><code id="HeatMap_+3A_nbadjm">NbAdjM</code></td>
<td>
<p>an integer providing the number of markers before and after the highlighted regions 
that should be added to the graphical representation, optional.</p>
</td></tr>
<tr><td><code id="HeatMap_+3A_subsets">Subsets</code></td>
<td>
<p>a list of character vectors with the population names, optional.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap where rows correspond to markers, and columns to populations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The full example execution takes a few seconds.
# data(Freq);data(NbGamete)
# FreqNbG &lt;- BuildFreqNbG(Freq,NbGamete)
# HFst.m &lt;- HudsonFst.m(FreqNbG)

## Two sets of populations to contrast
# Contrast &lt;- list(America=c("Colombian","Maya"),Europe=c("Tuscan","Italian"))
# Profiles &lt;- HudsonFst.prof(HFst.m,Contrast=Contrast)
# PS &lt;- ProfilingSummary(Profiles,Info) 

# RefLevel &lt;- rapply(Profiles,median,classes = "numeric",how='list')
# Ratio.thres &lt;- 3
# NbSnp.min &lt;- 1
# CS &lt;- ContrastSummary(PS, RefLevel,
#                       Ratio.thres=Ratio.thres,
#                       NbSnp.min=NbSnp.min)
# NbSel.thres &lt;- 2
# TopRegions &lt;- ContrastTopRegions(CS = CS,Crit = 'NbSel',Info = Info,
#                                  Thres = NbSel.thres, Simplify=TRUE)
# HeatMap(Min = TopRegions[1,]$Start,
#         Max = TopRegions[1,]$End,
#         chr = TopRegions[1,]$Chromosome,
#         Info = Info,
#         FreqNbG = FreqNbG,
#         Subsets = Contrast)                                
</code></pre>

<hr>
<h2 id='HudsonFst.gw'>HudsonFst.gw</h2><span id='topic+HudsonFst.gw'></span>

<h3>Description</h3>

<p>Compute genome-wide Hudson Fst moment estimator
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HudsonFst.gw(HFst.m, Mat = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HudsonFst.gw_+3A_hfst.m">HFst.m</code></td>
<td>
<p>a list of data.frames as obtained with function <code>HudsonFst.m</code></p>
</td></tr>
<tr><td><code id="HudsonFst.gw_+3A_mat">Mat</code></td>
<td>
<p>boolean, should the result be output as a matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>By default a matrix of Hudson Fst coefficients, a vector otherwise.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Freq);data(NbGamete)
FreqNbG &lt;- BuildFreqNbG(Freq,NbGamete)
HFst.m &lt;- HudsonFst.m(FreqNbG)
HFst.chr &lt;- HudsonFst.gw(HFst.m)
</code></pre>

<hr>
<h2 id='HudsonFst.m'>HudsonFst.m</h2><span id='topic+HudsonFst.m'></span>

<h3>Description</h3>

<p>Compute Hudson Fst moment estimator at marker level
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HudsonFst.m(FreqNbG)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HudsonFst.m_+3A_freqnbg">FreqNbG</code></td>
<td>
<p>a list of data.frames (one per population) with 2 columns: Freq and NbGamete</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of data.frames with 2 columns: Fst and Weight.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load the FreqNbG object build from the HGDP data
data(Freq);data(NbGamete)
FreqNbG &lt;- BuildFreqNbG(Freq,NbGamete)
HFst.m &lt;- HudsonFst.m(FreqNbG)
</code></pre>

<hr>
<h2 id='HudsonFst.plot'>Plot Fst values along chromosomes</h2><span id='topic+HudsonFst.plot'></span>

<h3>Description</h3>

<p>Plot Fst values along chromosomes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HudsonFst.plot(
  Info,
  HFst.m,
  HFst.prof = NULL,
  Coord = NULL,
  Ref = NULL,
  Threshold = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HudsonFst.plot_+3A_info">Info</code></td>
<td>
<p>a data.frame providing information about markers</p>
</td></tr>
<tr><td><code id="HudsonFst.plot_+3A_hfst.m">HFst.m</code></td>
<td>
<p>a data.frame with 2 columns, Fst and Weight, as provided by the <code>HudsonFst.m</code> function</p>
</td></tr>
<tr><td><code id="HudsonFst.plot_+3A_hfst.prof">HFst.prof</code></td>
<td>
<p>a data.frame corresponding to one item of the output of the <code>HudsonFst.prof</code> function</p>
</td></tr>
<tr><td><code id="HudsonFst.plot_+3A_coord">Coord</code></td>
<td>
<p>a vector with the minimum and maximum coordinates (i.e. positions along the genome)
providing the range of the genomic region that will be plotted.</p>
</td></tr>
<tr><td><code id="HudsonFst.plot_+3A_ref">Ref</code></td>
<td>
<p>a value to plot a reference line</p>
</td></tr>
<tr><td><code id="HudsonFst.plot_+3A_threshold">Threshold</code></td>
<td>
<p>a value to plot a threshold line</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The full example execution takes a few seconds.
data(Freq);data(NbGamete)
FreqNbG &lt;- BuildFreqNbG(Freq,NbGamete)
HFst.m &lt;- HudsonFst.m(FreqNbG)
TwoPops &lt;- list(First="Colombian",Second="Tuscan")
HFst.prof &lt;- HudsonFst.prof(HFst.m,Contrast=TwoPops)

## Plot the raw data

HudsonFst.plot(Info,HFst.m$Colombian_Tuscan)

## Plot the raw data and the segmentation

HudsonFst.plot(Info,HFst.m$Colombian_Tuscan,HFst.prof$Colombian_Tuscan)

## Add a background/reference level

RefLevel &lt;- median(HFst.prof$Colombian_Tuscan)
HudsonFst.plot(Info,HFst.m$Colombian_Tuscan,HFst.prof$Colombian_Tuscan,
               Ref=RefLevel)

## Add a threshold 

Threshold &lt;- 3*RefLevel
HudsonFst.plot(Info,HFst.m$Colombian_Tuscan,HFst.prof$Colombian_Tuscan,
               Ref=RefLevel,Threshold = Threshold)

</code></pre>

<hr>
<h2 id='HudsonFst.prof'>HudsonFst.prof</h2><span id='topic+HudsonFst.prof'></span>

<h3>Description</h3>

<p>Perform FST profiling between pairs of pops, as requested by Contrast. If no contrast is provided, all pairs are considered
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HudsonFst.prof(
  HFst.m,
  Contrast = NULL,
  Kmax = 100,
  NbSegCrit = "biggest.S3IB",
  parallel = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HudsonFst.prof_+3A_hfst.m">HFst.m</code></td>
<td>
<p>A list of data.frame with two columns each, Fst and Weight, 
as provided by the <code>HudsonFst.m</code> function</p>
</td></tr>
<tr><td><code id="HudsonFst.prof_+3A_contrast">Contrast</code></td>
<td>
<p>a list of two vectors with the names of the populations to be contrasted</p>
</td></tr>
<tr><td><code id="HudsonFst.prof_+3A_kmax">Kmax</code></td>
<td>
<p>maximum number of breakpoints to be considered</p>
</td></tr>
<tr><td><code id="HudsonFst.prof_+3A_nbsegcrit">NbSegCrit</code></td>
<td>
<p>the criterion used for the choice of the number of segments</p>
</td></tr>
<tr><td><code id="HudsonFst.prof_+3A_parallel">parallel</code></td>
<td>
<p>a boolean, should the profiling be parallelized (using future) or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a smoothed profile
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Freq);data(NbGamete)
FreqNbG &lt;- BuildFreqNbG(Freq,NbGamete)
HFst.m &lt;- HudsonFst.m(FreqNbG)

## Two population analysis
TwoPops &lt;- list(First="Colombian",Second="Tuscan")
HFst.prof &lt;- HudsonFst.prof(HFst.m,Contrast=TwoPops)

## The full example execution takes a few seconds.
## Two sets of populations to contrast

Contrast &lt;- list(America=c("Colombian","Maya"),Europe=c("Tuscan","Italian"))
Profiles &lt;- HudsonFst.prof(HFst.m,Contrast=Contrast)

## For larger lists and/or larger marker sets, 
## use the future package for parallel computation:

future::plan("multisession",workers=4)
Profiles &lt;- HudsonFst.prof(HFst.m,Contrast=Contrast)
future::plan("default")

</code></pre>

<hr>
<h2 id='Info'>Marker information</h2><span id='topic+Info'></span>

<h3>Description</h3>

<p><code>Info</code> is a data.frame describing the markers located on chromosome 1 in the Stanford 
HGDP dataset. Each row corresponds to a marker, and the 5 columns provide information about
the marker name, its chromosome membership, its position, and its reference and alternative alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Info
</code></pre>


<h3>Format</h3>

<p>A data.frame
</p>

<hr>
<h2 id='MakeProfile.op'>MakeProfile.op</h2><span id='topic+MakeProfile.op'></span>

<h3>Description</h3>

<p>Perform segmentation on a given dataset and returns the segmented profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MakeProfile.op(DF, coef.pen.value = 1, sd.y = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MakeProfile.op_+3A_df">DF</code></td>
<td>
<p>a data.frame with two columns, Fst and Weights, as provided by the <code>HudsonFst.m</code> function</p>
</td></tr>
<tr><td><code id="MakeProfile.op_+3A_coef.pen.value">coef.pen.value</code></td>
<td>
<p>coef to use for penaly 2*coef.pen.value*log(n)</p>
</td></tr>
<tr><td><code id="MakeProfile.op_+3A_sd.y">sd.y</code></td>
<td>
<p>a numeric value corresponding to the (estimated) standard deviation of the signal. If NULL (default) the value is automatically estimated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a smoothed profile
</p>

<hr>
<h2 id='MakeProfile.sn_nomemory'>MakeProfile.sn_nomemory</h2><span id='topic+MakeProfile.sn_nomemory'></span>

<h3>Description</h3>

<p>Perform segmentation on a given dataset and returns the segmented profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MakeProfile.sn_nomemory(DF, Kmax, method = "biggest.S3IB", sd.y = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MakeProfile.sn_nomemory_+3A_df">DF</code></td>
<td>
<p>a data.frame with two columns, Fst and Weights, as provided by the <code>HudsonFst.m</code> function</p>
</td></tr>
<tr><td><code id="MakeProfile.sn_nomemory_+3A_kmax">Kmax</code></td>
<td>
<p>max number of changes used for model selection (check with crops)</p>
</td></tr>
<tr><td><code id="MakeProfile.sn_nomemory_+3A_method">method</code></td>
<td>
<p>a string, the name of the criterion used for model selection: (1) &quot;givenVariance&quot; = using the penalty of Lebarbier 2005 given a estimator of the variance, (2) &quot;biggest.S3IB&quot; = biggest=TRUE in saut taken from S3IB, (3) &quot;notbiggest.S3IB&quot;
To be chosen amongs3ib.jump, s3ib.nojump, pre, ddse (capushe) or jump (capushe)</p>
</td></tr>
<tr><td><code id="MakeProfile.sn_nomemory_+3A_sd.y">sd.y</code></td>
<td>
<p>a numeric value corresponding to the (estimated) standard deviation of the signal. If NULL (default) the value is automatically estimated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a smoothed profile
</p>

<hr>
<h2 id='MergeRegion'>MergeRegion</h2><span id='topic+MergeRegion'></span>

<h3>Description</h3>

<p>Merge adjacent top regions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MergeRegion(DT, Crit)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MergeRegion_+3A_dt">DT</code></td>
<td>
<p>a data.frame</p>
</td></tr>
<tr><td><code id="MergeRegion_+3A_crit">Crit</code></td>
<td>
<p>a string corresponding to the name of the criterion used for selecting the top regions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a simplified data.frame (tibble)
</p>

<hr>
<h2 id='NbGamete'>Number of gametes of the American and European HGDP populations</h2><span id='topic+NbGamete'></span>

<h3>Description</h3>

<p><code>NbGamete</code> is a data.frame containing the number of gametes collected 
for the 13 American and European populations described in the Stanford HGDP dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NbGamete
</code></pre>


<h3>Format</h3>

<p>A data.frame
</p>

<hr>
<h2 id='Plot1Prof'>Plot1Prof</h2><span id='topic+Plot1Prof'></span>

<h3>Description</h3>

<p>Display the graphical representation of an Fst profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot1Prof(DF, Title = "", Range = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot1Prof_+3A_df">DF</code></td>
<td>
<p>a data.frame, as provided by function <code>DF4Plot1Prof</code></p>
</td></tr>
<tr><td><code id="Plot1Prof_+3A_title">Title</code></td>
<td>
<p>a string providing a title for the graph, optional.</p>
</td></tr>
<tr><td><code id="Plot1Prof_+3A_range">Range</code></td>
<td>
<p>a vector with the minimum and maximum values for the y-axis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>

<hr>
<h2 id='ProfilingSummary'>ProfilingSummary</h2><span id='topic+ProfilingSummary'></span>

<h3>Description</h3>

<p>Summary of Fst profiles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ProfilingSummary(FstProfiles, SnpInfo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ProfilingSummary_+3A_fstprofiles">FstProfiles</code></td>
<td>
<p>a list of Fst profiles as obtained from function <code>HudsonFst.prof</code></p>
</td></tr>
<tr><td><code id="ProfilingSummary_+3A_snpinfo">SnpInfo</code></td>
<td>
<p>a data.frame providing information about markers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of data.frame. Each data.frame summarizes a Fst profile, in terms of number of segments, 
Start and End positions, length (i.e. number of markers)  and Fst level of each segment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Freq);data(NbGamete);data(Info)
FreqNbG &lt;- BuildFreqNbG(Freq,NbGamete)
HFst.m &lt;- HudsonFst.m(FreqNbG)
TwoPops &lt;- list(First="Colombian",Second="Tuscan")
HFst.prof &lt;- HudsonFst.prof(HFst.m,Contrast=TwoPops)
PS &lt;- ProfilingSummary(HFst.prof,Info)
</code></pre>

<hr>
<h2 id='Ratio_Average'>Computation of the Fst moment estimator</h2><span id='topic+Ratio_Average'></span>

<h3>Description</h3>

<p>Computation of the Fst moment estimator
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ratio_Average(Nominator, Denominator)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ratio_Average_+3A_nominator">Nominator</code></td>
<td>
<p>numeric, numerators of the Fst moment estimator</p>
</td></tr>
<tr><td><code id="Ratio_Average_+3A_denominator">Denominator</code></td>
<td>
<p>numeric, denominators of the Fst moment estimator</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with the global Fst estimator
</p>

<hr>
<h2 id='RawPlot'>RawPlot</h2><span id='topic+RawPlot'></span>

<h3>Description</h3>

<p>Plot the Fst estimates along a (portion of) chromosome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RawPlot(Info, HF, Coord = NULL, Title = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RawPlot_+3A_info">Info</code></td>
<td>
<p>a data.frame providing information about markers</p>
</td></tr>
<tr><td><code id="RawPlot_+3A_hf">HF</code></td>
<td>
<p>a data.frame with 2 columns, Fst and Weight, as provided by the <code>HudsonFst.m</code> function</p>
</td></tr>
<tr><td><code id="RawPlot_+3A_coord">Coord</code></td>
<td>
<p>a vector with the minimum and maximum coordinates (i.e. positions along the genome), 
providing the range of the genomic region that will be plotted.</p>
</td></tr>
<tr><td><code id="RawPlot_+3A_title">Title</code></td>
<td>
<p>a string providing a title for the graph.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object.
</p>

<hr>
<h2 id='Summarise1Profile'>Summarise1Profile</h2><span id='topic+Summarise1Profile'></span>

<h3>Description</h3>

<p>Summarise1Profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Summarise1Profile(profile, snpinfo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Summarise1Profile_+3A_profile">profile</code></td>
<td>
<p>a vector, corresponding to the Fst profile of a pair of populations</p>
</td></tr>
<tr><td><code id="Summarise1Profile_+3A_snpinfo">snpinfo</code></td>
<td>
<p>a data.frame, providing information about markers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame that combines both objects
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
