<!DOCTYPE html><html><head><title>Help for package fsthet</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {fsthet}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#allele.counts'><p>This counts the number of alleles at a locus.</p></a></li>
<li><a href='#bins'><p>Example heterozygosity bins from fsthet.</p></a></li>
<li><a href='#calc.actual.fst'><p>This calcualtes global Fsts from a genepop dataframe.</p></a></li>
<li><a href='#calc.allele.freq'><p>This calculates allele frequencies.</p></a></li>
<li><a href='#calc.betahat'><p>This calculates beta-hat, the Fst value used in Lositan.</p></a></li>
<li><a href='#calc.exp.het'><p>This calculates expected heterozygosities.</p></a></li>
<li><a href='#calc.fst'><p>This calculates Fst.</p></a></li>
<li><a href='#calc.theta'><p>This calculates theta.</p></a></li>
<li><a href='#ci.means'><p>This calculates the average confidence intervals from multiple bootstrap outputs.</p></a></li>
<li><a href='#cis'><p>Example dataframe of smoothed quantiles from fsthet</p></a></li>
<li><a href='#cis.list'><p>Example list of CI matrices from bootstrap output from numerical simulations</p></a></li>
<li><a href='#fhetboot'><p>This is a wrapper to run the bootstrapping and plot the confidence intervals and significant loci.</p></a></li>
<li><a href='#find.outliers'><p>This identifies all of the SNPs outside of the smoothed quantiles in the dataset.</p></a></li>
<li><a href='#find.quantiles'><p>Generates quantiles from binned Fst values</p></a></li>
<li><a href='#fst.boot'><p>This is the major bootstrapping function to calculate confidence intervals.</p></a></li>
<li><a href='#fst.boot.means'><p>Calculates mean values within the bins.</p></a></li>
<li><a href='#fst.boot.onecol'><p>This bootstraps across all individuals to calculate a bootstrapped Fst for a randomly-sampled locus.</p></a></li>
<li><a href='#fst.options.print'><p>This prints the options for choosing an Fst calculation.</p></a></li>
<li><a href='#fsthet'><p>This is a wrapper to generate and plot the smoothed quantiles and identify outliers.</p></a></li>
<li><a href='#fsts'><p>Example fst calculations from a genepop file.</p></a></li>
<li><a href='#fsts.beta'><p>Example fst calculations from a genepop file.</p></a></li>
<li><a href='#fsts.betahat'><p>Example fst calculations from a genepop file.</p></a></li>
<li><a href='#fsts.theta'><p>Example fst calculations from a genepop file.</p></a></li>
<li><a href='#gpop'><p>Example genepop file from numerical simulations</p></a></li>
<li><a href='#make.bins'><p>This sorts Fst values into a designated number of overlapping heterozygosity bins.</p></a></li>
<li><a href='#my.read.genepop'><p>This reads a genepop file into R</p></a></li>
<li><a href='#p.boot'><p>Calculates mean values within the bins.</p></a></li>
<li><a href='#plotting.cis'><p>This plots a dataframe of fsts with bootstrapped confidence intervals.</p></a></li>
<li><a href='#quant.out'><p>Example fsthet output based on numerical simulations</p></a></li>
<li><a href='#remove.spaces'><p>This removes spaces from a character vector</p></a></li>
<li><a href='#var.fst'><p>This calculates Cockerham &amp; Weir's Beta.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Title:</td>
<td>Fst-Heterozygosity Smoothed Quantiles</td>
</tr>
<tr>
<td>Description:</td>
<td>A program to generate smoothed quantiles for the Fst-heterozygosity distribution. Designed for use with large numbers of loci (e.g., genome-wide SNPs). The best case for analyzing the Fst-heterozygosity distribution is when many populations (&gt;10) have been sampled. See Flanagan &amp; Jones (2017) &lt;<a href="https://doi.org/10.1093%2Fjhered%2Fesx048">doi:10.1093/jhered/esx048</a>&gt;.</td>
</tr>
<tr>
<td>Author:</td>
<td>Sarah P. Flanagan and Adam G. Jones</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sarah P. Flanagan &lt;spflanagan.phd@gmail.com&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-03-20 16:14:29 UTC; sarah</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-03-20 18:27:53 UTC</td>
</tr>
</table>
<hr>
<h2 id='allele.counts'>This counts the number of alleles at a locus. </h2><span id='topic+allele.counts'></span>

<h3>Description</h3>

<p>This counts the number of times each allele occurs at a locus from a list of genotypes (the sum of all the counts is 2*number of individuals).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allele.counts(genotypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="allele.counts_+3A_genotypes">genotypes</code></td>
<td>
<p>A list of genotypes.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>AlleleCounts</code></td>
<td>
<p>The number of times each allele is recorded at the locus.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  #create a random sample of genotypes
  genotypes&lt;-sample(c("0101","0102","0202"),50,replace=TRUE)
  counts&lt;-allele.counts(genotypes)
</code></pre>

<hr>
<h2 id='bins'>Example heterozygosity bins from fsthet.</h2><span id='topic+bins'></span>

<h3>Description</h3>

<p>This is a list with a data.frame of bins (the lower and upper bounds for each heterozygosity bin) and a list of fsts that fall into each bin, with the name of each set of Fst values being the upper heterozygosity bound from the data.frame of bins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bins</code></pre>


<h3>Format</h3>

<p>list</p>


<h3>Source</h3>

<p>bins&lt;-make.bins(fsts)
</p>


<h3>References</h3>

<p>See Flanagan &amp; Jones
</p>

<hr>
<h2 id='calc.actual.fst'>This calcualtes global Fsts from a genepop dataframe.</h2><span id='topic+calc.actual.fst'></span>

<h3>Description</h3>

<p>This calcualtes global Fsts from a genepop dataframe.
This does not include bootstrapping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.actual.fst(df, fst.choice="fst")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc.actual.fst_+3A_df">df</code></td>
<td>
<p>Provide the genepop dataframe (from my.read.genepop).</p>
</td></tr>
<tr><td><code id="calc.actual.fst_+3A_fst.choice">fst.choice</code></td>
<td>
<p>Specify which type of fst calculation should be used. See fst.options.print for the choices.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>fsts</code></td>
<td>
<p>This returns a dataframe with Locus, Ht, and Fst characters.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  gpop&lt;-data.frame(popinfo=c(rep("POP 1", 20),rep("POP 2", 20)),ind.names=c(1:20,1:20),
    loc0=sample(c("0101","0102","0202"),40,replace=TRUE))
  fsts&lt;-calc.actual.fst(gpop)
  ## Not run: 
    gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
    gpop&lt;-my.read.genepop(gfile)
    fsts&lt;-calc.actual.fst(gpop)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='calc.allele.freq'>This calculates allele frequencies. </h2><span id='topic+calc.allele.freq'></span>

<h3>Description</h3>

<p>This calculates allele frequencies from a list of genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.allele.freq(genotypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc.allele.freq_+3A_genotypes">genotypes</code></td>
<td>
<p>A list of genotypes.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>obs.af</code></td>
<td>
<p>A list of observed allele frequencies in the genotypes list.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  #create a random sample of genotypes
  genotypes&lt;-sample(c("0101","0102","0202"),50,replace=TRUE)
  af&lt;-calc.allele.freq(genotypes)
</code></pre>

<hr>
<h2 id='calc.betahat'>This calculates beta-hat, the Fst value used in Lositan. </h2><span id='topic+calc.betahat'></span>

<h3>Description</h3>

<p>This calculates Weir &amp; Cockerham (1993)'s beta-hat.
Beaumont &amp; Nichols (1996) used this formulation in FDIST2 (and is implemented in Lositan)
See the vignette for details on the calculation of beta.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.betahat(df, i)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc.betahat_+3A_df">df</code></td>
<td>
<p>A dataframe containing the genepop information, where the first column is the population ID.</p>
</td></tr>
<tr><td><code id="calc.betahat_+3A_i">i</code></td>
<td>
<p>Column number containing genotype information.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>ht</code></td>
<td>
<p>HB (or 1-F1). This is a single numerical value.</p>
</td></tr>
<tr><td><code>fst</code></td>
<td>
<p>The calculated betahat value ((F0-F1)/(1-F1))for this locus.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>   gpop&lt;-data.frame(popinfo=c(rep("POP 1", 20),rep("POP 2", 20)),ind.names=c(1:20,1:20))
     for(i in 1:40){
      gpop[1:20,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
      gpop[21:40,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
     }
  bh&lt;-calc.betahat(gpop, 3) #calculate betahat for the SNP
  
    gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
    gpop&lt;-my.read.genepop(gfile)
    beta1&lt;-calc.betahat(gpop,3) #calculate betahat for the first SNP
  
</code></pre>

<hr>
<h2 id='calc.exp.het'>This calculates expected heterozygosities. </h2><span id='topic+calc.exp.het'></span>

<h3>Description</h3>

<p>This calculates expected heterozygosities from a list of allele frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.exp.het(af)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc.exp.het_+3A_af">af</code></td>
<td>
<p>is a list of allele frequencies.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>ht</code></td>
<td>
<p>The expected heterozygosity under Hardy-Weinberg expectations. This is a single numerical value.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  #create a random sample of genotypes
  genotypes&lt;-sample(c("0101","0102","0202"),50,replace=TRUE)
  af&lt;-calc.allele.freq(genotypes)
  hs&lt;-calc.exp.het(af)
</code></pre>

<hr>
<h2 id='calc.fst'>This calculates Fst. </h2><span id='topic+calc.fst'></span>

<h3>Description</h3>

<p>This calculates Fst. 
The caluclation is done as (Ht-Hs)/Ht, 
where Ht is the expected heterozygosity for all populations 
and Hs is the expected heterozygosity for each population.
This calculation is used in bootstrapping functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.fst(df, i)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc.fst_+3A_df">df</code></td>
<td>
<p>A dataframe containing the genepop information, where the first column is the population ID.</p>
</td></tr>
<tr><td><code id="calc.fst_+3A_i">i</code></td>
<td>
<p>Column number containing genotype information.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>ht</code></td>
<td>
<p>The expected heterozygosity under Hardy-Weinberg expectations. This is a single numerical value.</p>
</td></tr>
<tr><td><code>fst</code></td>
<td>
<p>The calculated Fst value for this locus.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  gpop&lt;-data.frame(popinfo=c(rep("POP 1", 20),rep("POP 2", 20)),ind.names=c(1:20,1:20),
    loc0=sample(c("0101","0102","0202"),40,replace=TRUE))
  fst1&lt;-calc.fst(gpop,3)
  ## Not run: 
    gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
    gpop&lt;-my.read.gpop(gfile)
    fst1&lt;-calc.fst(gpop,3) #calculate fst for the first SNP
  
## End(Not run)
</code></pre>

<hr>
<h2 id='calc.theta'>This calculates theta.</h2><span id='topic+calc.theta'></span>

<h3>Description</h3>

<p>This calculates Weir (1990)'s theta.
See the vignette for details on the calculation of beta.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.theta(df, i)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc.theta_+3A_df">df</code></td>
<td>
<p>A dataframe containing the genepop information, where the first column is the population ID.</p>
</td></tr>
<tr><td><code id="calc.theta_+3A_i">i</code></td>
<td>
<p>Column number containing genotype information.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>ht</code></td>
<td>
<p>T2. This is a single numerical value.</p>
</td></tr>
<tr><td><code>fst</code></td>
<td>
<p>The calculated theta value (T1/T2) for this locus.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  gpop&lt;-data.frame(popinfo=c(rep("POP 1", 20),rep("POP 2", 20)),ind.names=c(1:20,1:20),
    loc0=sample(c("0101","0102","0202"),40,replace=TRUE))
  theta1&lt;-calc.theta(gpop, 3)
  ## Not run: 
    gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
    gpop&lt;-my.read.gpop(gfile)
    theta1&lt;-calc.theta(gpop,3) #calculate theta for the first SNP
  
## End(Not run)
</code></pre>

<hr>
<h2 id='ci.means'>This calculates the average confidence intervals from multiple bootstrap outputs. </h2><span id='topic+ci.means'></span>

<h3>Description</h3>

<p>This calculates the mean upper and lower confidence intervals from a list of bootstrap CI matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ci.means(boot.out.list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ci.means_+3A_boot.out.list">boot.out.list</code></td>
<td>
<p>A list of matrices. Each matrix is the CIs from fst.boot (boot.out[[3]]).</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>avg.cil</code></td>
<td>
<p>A list of the average lower CI values</p>
</td></tr>
<tr><td><code>avg.ciu</code></td>
<td>
<p>A list of the average upper CI values</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  
  ## Not run: 
    gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
    gpop&lt;-my.read.genepop(gfile)
    quant.out&lt;-fst.boot(gpop, bootstrap = FALSE)
    quant.list&lt;-ci.means(quant.out[[3]])
  
## End(Not run)
</code></pre>

<hr>
<h2 id='cis'>Example dataframe of smoothed quantiles from fsthet</h2><span id='topic+cis'></span>

<h3>Description</h3>

<p>Example list of data.frames with smoothed quantiles from fsthet output from numerical simulations
The data were generated using a numerical analysis with Nm = 10, 75 demes, and 5 population samples taken. No selection was imposed.
This is a list with a single data.frame containing values from 95 percent smoothed quantiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cis</code></pre>


<h3>Format</h3>

<p>list</p>


<h3>Source</h3>

<p>Ninety-five percent smoothed quantiles, using the dataframe gpop.</p>


<h3>References</h3>

<p>See Flanagan &amp; Jones
</p>

<hr>
<h2 id='cis.list'>Example list of CI matrices from bootstrap output from numerical simulations</h2><span id='topic+cis.list'></span>

<h3>Description</h3>

<p>Example list of CI data.frames from fsthet output from numerical simulations
The data were generated using a numerical analysis with Nm = 10, 75 demes, and 5 population samples taken. No selection was imposed.
This is a list of data.frames containing values from 99 percent and 95 percent smoothed quantiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cis.list</code></pre>


<h3>Format</h3>

<p>list</p>


<h3>Source</h3>

<p>From multiple smoothed quantile alpha thresholds, using the dataframe gpop.</p>


<h3>References</h3>

<p>See Flanagan &amp; Jones
</p>

<hr>
<h2 id='fhetboot'>This is a wrapper to run the bootstrapping and plot the confidence intervals and significant loci.</h2><span id='topic+fhetboot'></span>

<h3>Description</h3>

<p>This calcualtes global Fsts from a genepop dataframe and then does:
p-value calculations
plots the Heterozygosity-Fst relationship with smoothed CIs
outputs the loci lying outside the confidence intervals.
Returns a data frame containing Locus ID, Ht, Fst, P-value, a Benjamini-Hochberg-corrected P-value, and a true/false value of whether it's an outlier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	fhetboot(gpop, fst.choice="fst", alpha=0.05,nreps=10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fhetboot_+3A_gpop">gpop</code></td>
<td>
<p>Provide the genepop dataframe (from my.read.genepop).</p>
</td></tr>
<tr><td><code id="fhetboot_+3A_fst.choice">fst.choice</code></td>
<td>
<p>Specify which type of fst calculation should be used. See fst.options.print for the choices.</p>
</td></tr>
<tr><td><code id="fhetboot_+3A_alpha">alpha</code></td>
<td>
<p>The alpha value for the confidence intervals and the p-value adjustment calculations (default is 0.05).</p>
</td></tr>
<tr><td><code id="fhetboot_+3A_nreps">nreps</code></td>
<td>
<p>The number of bootstrap replicates to use. The default is 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>fsts</code></td>
<td>
<p>This returns a dataframe with Locus, Ht, Fst, P-value, correcte P-value, and True/False of whether it's an outlier.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  
  ## Not run: 
    gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
    gpop&lt;-my.read.genepop(gfile)
    out.dat&lt;-fhetboot(gpop, fst.choice="fst", alpha=0.05,nreps=10)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='find.outliers'>This identifies all of the SNPs outside of the smoothed quantiles in the dataset.</h2><span id='topic+find.outliers'></span>

<h3>Description</h3>

<p>This identifies all of the SNPs outside of the smoothed quantiles in the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.outliers(df, boot.out, ci.df = NULL, file.name = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find.outliers_+3A_df">df</code></td>
<td>
<p>Provide the dataframe with Ht and Fst values.</p>
</td></tr>
<tr><td><code id="find.outliers_+3A_boot.out">boot.out</code></td>
<td>
<p>Bootstrap output. You must provide this.</p>
</td></tr>
<tr><td><code id="find.outliers_+3A_ci.df">ci.df</code></td>
<td>
<p>List of confidence intervals. You may provide this in addition to bootstrap output to save a small amount of time.</p>
</td></tr>
<tr><td><code id="find.outliers_+3A_file.name">file.name</code></td>
<td>
<p>You may provide a file name to output the outliers to a csv file. Otherwise, the function will only return the outliers.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>out</code></td>
<td>
<p>A list of the outlier loci</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  
  ## Not run: 
    gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
    gpop&lt;-my.read.gpop(gfile)
    fsts&lt;-calc.actual.fst(gpop)
    boot.out&lt;-as.data.frame(t(replicate(10, fst.boot(gpop))))
    outliers&lt;-find.outliers(fsts,boot.out)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='find.quantiles'>Generates quantiles from binned Fst values</h2><span id='topic+find.quantiles'></span>

<h3>Description</h3>

<p>This takes the output from make.bins and calculates the smoothed quantiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.quantiles(bins,bin.fst,ci=0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find.quantiles_+3A_bins">bins</code></td>
<td>
<p>A dataframe containing with upper and lower het and Fst values for each bin (output from make.bins).</p>
</td></tr>
<tr><td><code id="find.quantiles_+3A_bin.fst">bin.fst</code></td>
<td>
<p> A list with the Fst values for each bin (output from make.bins). </p>
</td></tr>
<tr><td><code id="find.quantiles_+3A_ci">ci</code></td>
<td>
<p>A value for the confidence intervals alpha (default is 0.05).</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>fst.CI</code></td>
<td>
<p>A list of data.frames, one for each ci value with the upper and lower Fst quantiles for each Heterozygosity bin.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>gpop&lt;-data.frame(popinfo=c(rep("POP 1", 20),rep("POP 2", 20)),ind.names=c(1:20,1:20))
     for(i in 1:40){
      gpop[1:20,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
      gpop[21:40,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
     }
  fsts&lt;-calc.actual.fst(gpop)
  nloci&lt;-(ncol(gpop)-2)
  boot.out&lt;-as.data.frame(t(replicate(nloci, fst.boot.onecol(gpop,"fst"))))
  bins&lt;-make.bins(boot.out,25,Ht.name="V1",Fst.name="V2")
  fst.CI&lt;-find.quantiles(bins$bins,bins$bin.fst)

</code></pre>

<hr>
<h2 id='fst.boot'>This is the major bootstrapping function to calculate confidence intervals. </h2><span id='topic+fst.boot'></span>

<h3>Description</h3>

<p>This randomly samples all of the loci, with replacement 
(so if you have 200 loci, it will choose 200 loci to calculate Fst for, but some may be sampled multiply)
It makes use of fst.boot.onerow.
To calculate the confidence intervals, this function bins the Fst values based on heterozygosity values.
The bins are overlapping and each bin is the width of smooth.rate. 
The Fst value which separates the top 100*(ci/2) and bottom 100*(ci/2) percent in each bin are the upper and lower CIs.
This function can be slow. We recommend running it 10 times to generate confidence intervals for analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fst.boot(df,fst.choice="fst",ci=0.05,num.breaks=25, bootstrap = TRUE,min.per.bin=20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fst.boot_+3A_df">df</code></td>
<td>
<p>A dataframe containing the genepop information, where the first column is the population ID.</p>
</td></tr>
<tr><td><code id="fst.boot_+3A_fst.choice">fst.choice</code></td>
<td>

<p>A character defining which fst calculation is to be used. See fst.options.print() for the choices.</p>
</td></tr>
<tr><td><code id="fst.boot_+3A_ci">ci</code></td>
<td>
<p>A value for the confidence intervals alpha (default is 0.05).</p>
</td></tr>
<tr><td><code id="fst.boot_+3A_num.breaks">num.breaks</code></td>
<td>
<p>The number of breaks used to create bins (default is 25)</p>
</td></tr>
<tr><td><code id="fst.boot_+3A_bootstrap">bootstrap</code></td>
<td>
<p>A TRUE/FALSE statement telling the program whether to bootstrap and then determine the bins or to calculate bins and confidence intervals from the empirical dataset without bootstrapping. The default is TRUE, which means bootstrapping occurs.</p>
</td></tr>
<tr><td><code id="fst.boot_+3A_min.per.bin">min.per.bin</code></td>
<td>
<p>The minimum number of loci that are required for a bin to be retained. Default is 20.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Fsts</code></td>
<td>
<p>The bootstrapped Fst and Ht values</p>
</td></tr>
<tr><td><code>Bins</code></td>
<td>
<p>A dataframe containing the bins start and stop Ht values.</p>
</td></tr>
<tr><td><code>fst.CI</code></td>
<td>
<p>A list of dataframes containing the lower and upper confidence intervals' Ht values.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  gpop&lt;-data.frame(popinfo=c(rep("POP 1", 20),rep("POP 2", 20)),ind.names=c(1:20,1:20))
   for(i in 1:40){
    gpop[1:20,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
    gpop[21:40,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
   }
  fsts&lt;-calc.actual.fst(gpop)
  quant.out&lt;-as.data.frame(t(replicate(1, fst.boot(gpop,bootstrap=FALSE))))
  ## Not run: 
    gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
    gpop&lt;-my.read.genepop(gfile)
    fsts&lt;-calc.actual.fst(gpop)
    quant.out&lt;-as.data.frame(t(replicate(1, fst.boot(gpop,bootstrap=FALSE))))
  
## End(Not run)
</code></pre>

<hr>
<h2 id='fst.boot.means'>Calculates mean values within the bins. </h2><span id='topic+fst.boot.means'></span>

<h3>Description</h3>

<p>This calculates mean heterozygosity and Fst values for each bin used in bootstrapping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fst.boot.means(boot.out)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fst.boot.means_+3A_boot.out">boot.out</code></td>
<td>
<p>The first item in the output lists from fst.boot (aka boot.out[[1]]. </p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>bmu</code></td>
<td>
<p>A dataframe containing four columns:
heterozygosity
Fst
the number of loci in the bin
the lower Ht value for the bin
and the upper Ht value for the bin.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  gpop&lt;-data.frame(popinfo=c(rep("POP 1", 20),rep("POP 2", 20)),ind.names=c(1:20,1:20))
   for(i in 1:40){
    gpop[1:20,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
    gpop[21:40,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
   }
    fsts&lt;-calc.actual.fst(gpop)
  boot.out&lt;-as.data.frame(t(replicate(1, fst.boot(gpop))))
  outliers&lt;-find.outliers(fsts,boot.out)
## Not run: 
  gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
  gpop&lt;-my.read.genepop(gfile)
  fsts&lt;-calc.actual.fst(gpop)
  boot.out&lt;-as.data.frame(t(replicate(10, fst.boot(gpop))))
  outliers&lt;-find.outliers(fsts,boot.out)

## End(Not run)
</code></pre>

<hr>
<h2 id='fst.boot.onecol'>This bootstraps across all individuals to calculate a bootstrapped Fst for a randomly-sampled locus. </h2><span id='topic+fst.boot.onecol'></span>

<h3>Description</h3>

<p>This calculates Fst using calc.fst.
It randomly selects a column containing genotype information for all individuals.
It then calculates Fst and Ht for that locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fst.boot.onecol(df, fst.choice)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fst.boot.onecol_+3A_df">df</code></td>
<td>
<p>A dataframe containing the genepop information, where the first column is the population ID.</p>
</td></tr>
<tr><td><code id="fst.boot.onecol_+3A_fst.choice">fst.choice</code></td>
<td>

<p>A character defining which fst calculation is to be used. 
The three options are:
Wright's Fst (Wright, wright, WRIGHT, W, w)
Weir and Cockerham 1993's beta (WeirCockerham,weircockerham,wc,WC)
Corrected Weir and Cockerham 1993's beta from Beaumont and Nichols 1996 (WeirCockerhamCorrected, weircockerhamcorrected,corrected,wcc,WCC)</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>ht.fst</code></td>
<td>
<p>A vector containin Ht and Fst</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>gpop&lt;-data.frame(popinfo=c(rep("POP 1", 20),rep("POP 2", 20)),ind.names=c(1:20,1:20))
   for(i in 1:40){
    gpop[1:20,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
    gpop[21:40,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
   }
   fsts&lt;-calc.actual.fst(gpop)
  nloci&lt;-(ncol(gpop)-2)
  boot.out&lt;-as.data.frame(t(replicate(nloci, fst.boot.onecol(gpop,"fst"))))
## Not run: 
  gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
  gpop&lt;-my.read.genepop(gfile)
  fsts&lt;-calc.actual.fst(gpop)
  nloci&lt;-(ncol(gpop)-2)
  boot.out&lt;-as.data.frame(t(replicate(nloci, fst.boot.onecol(gpop,"fst"))))
  outliers&lt;-find.outliers(fsts,boot.out)

## End(Not run)
</code></pre>

<hr>
<h2 id='fst.options.print'>This prints the options for choosing an Fst calculation. </h2><span id='topic+fst.options.print'></span>

<h3>Description</h3>

<p>This prints the options for choosing an Fst calculation. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fst.options.print()
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>fst.options.print()
</code></pre>

<hr>
<h2 id='fsthet'>This is a wrapper to generate and plot the smoothed quantiles and identify outliers.</h2><span id='topic+fsthet'></span>

<h3>Description</h3>

<p>This calcualtes global Fsts from a genepop dataframe and then does:
calculates smoothed quantiles
plots the Heterozygosity-Fst relationship with smoothed quantiles
outputs the loci lying outside the quantiles.
Returns a data frame containing Locus ID, Ht, Fst, and a true/false value of whether it's an outlier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	fsthet(gpop, fst.choice="fst", alpha=0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fsthet_+3A_gpop">gpop</code></td>
<td>
<p>Provide the genepop dataframe (from my.read.genepop).</p>
</td></tr>
<tr><td><code id="fsthet_+3A_fst.choice">fst.choice</code></td>
<td>
<p>Specify which type of fst calculation should be used. See fst.options.print for the choices.</p>
</td></tr>
<tr><td><code id="fsthet_+3A_alpha">alpha</code></td>
<td>
<p>The alpha value for the quantiles (default is 0.05 to generate 95 percent quantiles).</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>fsts</code></td>
<td>
<p>This returns a dataframe with Locus, Ht, Fst, and True/False of whether it's an outlier.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  
## Not run: 
  gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
  gpop&lt;-my.read.gpop(gfile)
  out.dat&lt;-fsthet(gpop)

## End(Not run)
</code></pre>

<hr>
<h2 id='fsts'>Example fst calculations from a genepop file.</h2><span id='topic+fsts'></span>

<h3>Description</h3>

<p>Example fst calculations from a genepop file.
The original data were generated by using a numerical analysis with Nm = 10, 75 demes, and 5 population samples taken. No selection was imposed.
The fsts were calculated using calc.actual.fst(gpop)
This file contains a dataframe with 2000 columns and 3 rows. The first column is the Locus ID, the second column is the Ht for that locus, and the third column is the Fst for that locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fsts</code></pre>


<h3>Format</h3>

<p>data.frame</p>


<h3>Source</h3>

<p>Generated by numerical analysis</p>


<h3>References</h3>

<p>See Flanagan &amp; Jones
</p>

<hr>
<h2 id='fsts.beta'>Example fst calculations from a genepop file.</h2><span id='topic+fsts.beta'></span>

<h3>Description</h3>

<p>Example fst calculations using beta (fst.choice=&quot;var&quot;) from a genepop file.
The original data were generated by using a numerical analysis with Nm = 10, 75 demes, and 5 population samples taken. No selection was imposed.
The fsts were calculated using calc.actual.fst(gpop,fst.choice=&quot;var&quot;)
This file contains a dataframe with 2000 columns and 3 rows. The first column is the Locus ID, the second column is the Ht for that locus, and the third column is the Fst for that locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fsts.beta</code></pre>


<h3>Format</h3>

<p>data.frame</p>


<h3>Source</h3>

<p>Generated by numerical analysis</p>


<h3>References</h3>

<p>See Flanagan &amp; Jones
</p>

<hr>
<h2 id='fsts.betahat'>Example fst calculations from a genepop file.</h2><span id='topic+fsts.betahat'></span>

<h3>Description</h3>

<p>Example fst calculations using betahat (fst.choice=&quot;betahat&quot;) from a genepop file.
The original data were generated by using a numerical analysis with Nm = 10, 75 demes, and 5 population samples taken. No selection was imposed.
The fsts were calculated using calc.actual.fst(gpop,fst.choice=&quot;betahat&quot;)
This file contains a dataframe with 2000 columns and 3 rows. The first column is the Locus ID, the second column is the Ht for that locus, and the third column is the Fst for that locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fsts.betahat</code></pre>


<h3>Format</h3>

<p>data.frame</p>


<h3>Source</h3>

<p>Generated by numerical analysis</p>


<h3>References</h3>

<p>See Flanagan &amp; Jones
</p>

<hr>
<h2 id='fsts.theta'>Example fst calculations from a genepop file.</h2><span id='topic+fsts.theta'></span>

<h3>Description</h3>

<p>Example fst calculations using theta (fst.choice=&quot;theta&quot;) from a genepop file.
The original data were generated by using a numerical analysis with Nm = 10, 75 demes, and 5 population samples taken. No selection was imposed.
The fsts were calculated using calc.actual.fst(gpop,fst.choice=&quot;theta&quot;)
This file contains a dataframe with 2000 columns and 3 rows. The first column is the Locus ID, the second column is the Ht for that locus, and the third column is the Fst for that locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fsts.theta</code></pre>


<h3>Format</h3>

<p>data.frame</p>


<h3>Source</h3>

<p>Generated by numerical analysis</p>


<h3>References</h3>

<p>See Flanagan &amp; Jones
</p>

<hr>
<h2 id='gpop'>Example genepop file from numerical simulations</h2><span id='topic+gpop'></span>

<h3>Description</h3>

<p>Example genepop file from numerical simulations.
It was generated by using a numerical analysis with Nm = 10, 75 demes, and 5 population samples taken. No selection was imposed.
This file contains a dataframe with 2002 columns and 250 rows. The first two columns are the population name and the individual name. The remaining columns are genotypes for each locus (one column per locus). Each row is an individual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gpop</code></pre>


<h3>Format</h3>

<p>data.frame</p>


<h3>Source</h3>

<p>Generated by numerical analysis</p>


<h3>References</h3>

<p>See Flanagan &amp; Jones
</p>

<hr>
<h2 id='make.bins'>This sorts Fst values into a designated number of overlapping heterozygosity bins. </h2><span id='topic+make.bins'></span>

<h3>Description</h3>

<p>This breaks up Fst values into a designated number of overlapping heterozygosity bins.
It returns a list containing a data.frame called bins a list called bin.fst with the Fst values for each of the Het categories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make.bins(fsts,num.breaks=25, Ht.name="Ht", Fst.name="Fst",min.per.bin=20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make.bins_+3A_fsts">fsts</code></td>
<td>
<p>A dataframe containing at least the columns with heterozygosity and Fst values.</p>
</td></tr>
<tr><td><code id="make.bins_+3A_num.breaks">num.breaks</code></td>
<td>
<p> The number of breaks used to create bins (default is 25)</p>
</td></tr>
<tr><td><code id="make.bins_+3A_ht.name">Ht.name</code></td>
<td>
<p>Provide the name of the column with the heterozygosity values, unless the column is named &quot;Ht&quot;.</p>
</td></tr>
<tr><td><code id="make.bins_+3A_fst.name">Fst.name</code></td>
<td>
<p>Provide the name of the column with the Fst values, unless the column is named &quot;Fst&quot;.</p>
</td></tr>
<tr><td><code id="make.bins_+3A_min.per.bin">min.per.bin</code></td>
<td>
<p>If you have a smaller dataset, you can change the minimum number of loci required to be in each bin. Default is 20.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>list(bins</code>, <code>bin.fst)</code></td>
<td>
<p>A list with a data.frame called bins with the upper and lower Fst and Ht values
and a list called bin.fst with the Fst values for each of the Het categories.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  gpop&lt;-data.frame(popinfo=c(rep("POP 1", 20),rep("POP 2", 20)),ind.names=c(1:20,1:20))
     for(i in 1:40){
      gpop[1:20,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
      gpop[21:40,(i+2)]&lt;-sample(c("0101","0102","0202"),20,replace=TRUE)
     }
  fsts&lt;-calc.actual.fst(gpop)
  nloci&lt;-(ncol(gpop)-2)
  boot.out&lt;-as.data.frame(t(replicate(nloci, fst.boot.onecol(gpop,"fst"))))
  bins&lt;-make.bins(boot.out,25,Ht.name="V1",Fst.name="V2")
  ## Not run: 
    gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
    gpop&lt;-my.read.gpop(gfile)
    fsts&lt;-calc.actual.fst(gpop)
    nloci&lt;-(ncol(gpop)-2)
    boot.out&lt;-as.data.frame(t(replicate(nloci, fst.boot.onecol(gpop))))
    make.bins(boot.out,25)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='my.read.genepop'>This reads a genepop file into R</h2><span id='topic+my.read.genepop'></span>

<h3>Description</h3>

<p>This reads a genepop file into R. 
It was adapted from a similar functionin adegenet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>my.read.genepop(file, ncode = 2L, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="my.read.genepop_+3A_file">file</code></td>
<td>
<p>is the filename of the genpop file.</p>
</td></tr>
<tr><td><code id="my.read.genepop_+3A_quiet">quiet</code></td>
<td>
<p>If quiet = FALSE updates will be printed. 
If quiet = T status updates will not be printed.
</p>
</td></tr>
<tr><td><code id="my.read.genepop_+3A_ncode">ncode</code></td>
<td>
<p>Do not change this argument.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>res</code></td>
<td>
<p>A dataframe with the Population ID in column 1, the Individual ID in column 2, 
and the genotypes in columns following that. There is one row per individual.</p>
</td></tr>
</table>


<h3>References</h3>

<p><a href="http://adegenet.r-forge.r-project.org/">http://adegenet.r-forge.r-project.org/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
  gpop&lt;-my.read.genepop(gfile)
  
</code></pre>

<hr>
<h2 id='p.boot'>Calculates mean values within the bins. </h2><span id='topic+p.boot'></span>

<h3>Description</h3>

<p>This calculates mean heterozygosity and Fst values for each bin used in bootstrapping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p.boot(actual.fsts, boot.out,boot.means=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="p.boot_+3A_actual.fsts">actual.fsts</code></td>
<td>
<p>The first item in the output lists from fst.boot. </p>
</td></tr>
<tr><td><code id="p.boot_+3A_boot.out">boot.out</code></td>
<td>
<p>The output from a bootstrapping run. Either supply this or boot.means. </p>
</td></tr>
<tr><td><code id="p.boot_+3A_boot.means">boot.means</code></td>
<td>
<p>The output from fst.boot.means. Either supply this or bootstrapping output. </p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>pvals</code></td>
<td>
<p>A numeric containing uncorrected p-values for each locus. The names attribute are the locus names.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  
  ## Not run: 
    gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
    gpop&lt;-my.read.genepop(gfile)
    fsts&lt;-calc.actual.fst(gpop)
    boot.out&lt;-as.data.frame(t(replicate(10, fst.boot(gpop))))
    boot.pvals&lt;-p.boot(fsts,boot.out=boot.out)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='plotting.cis'>This plots a dataframe of fsts with bootstrapped confidence intervals.</h2><span id='topic+plotting.cis'></span>

<h3>Description</h3>

<p>This plots a dataframe of fsts with bootstrapped confidence intervals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotting.cis(df,boot.out,ci.df=NULL,sig.list=NULL,Ht.name="Ht",Fst.name="Fst",
	ci.col="red", pt.pch=1,file.name=NULL,sig.col=ci.col,make.file=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotting.cis_+3A_df">df</code></td>
<td>
<p>A dataframe of Fst and Ht values. It must have at least two columns, one named &quot;Ht&quot; and one named &quot;Fst&quot;. Or you must pass the column names to the function</p>
</td></tr>
<tr><td><code id="plotting.cis_+3A_boot.out">boot.out</code></td>
<td>
<p>Bootstrap output. You must either provide this or a list of confidence interval values.</p>
</td></tr>
<tr><td><code id="plotting.cis_+3A_ci.df">ci.df</code></td>
<td>
<p>Data frame of confidence intervals. You must either provide this or bootstrap output.</p>
</td></tr>
<tr><td><code id="plotting.cis_+3A_sig.list">sig.list</code></td>
<td>
<p>List of significant locus names (this acts as a way to highlight particular loci). This is optional and colors some of the points using the same shape as pt.pch and the color of sig.col (default sig.color is same as ci.col).</p>
</td></tr>
<tr><td><code id="plotting.cis_+3A_ht.name">Ht.name</code></td>
<td>
<p>Provide the name of the column with the heterozygosity values, unless the column is named &quot;Ht&quot;.</p>
</td></tr>
<tr><td><code id="plotting.cis_+3A_fst.name">Fst.name</code></td>
<td>
<p>Provide the name of the column with the Fst values, unless the column is named &quot;Fst&quot;.</p>
</td></tr>
<tr><td><code id="plotting.cis_+3A_ci.col">ci.col</code></td>
<td>
<p>You can input the colors of the confidence intervals to be plotted. First is the 95 percent CI, second is the 99 percent CI. Defaults are &quot;red&quot; and &quot;gold&quot;.</p>
</td></tr>
<tr><td><code id="plotting.cis_+3A_pt.pch">pt.pch</code></td>
<td>
<p>You can change the point shape here. Default is 1 (open circles)</p>
</td></tr>
<tr><td><code id="plotting.cis_+3A_sig.col">sig.col</code></td>
<td>
<p>The color of the significant loci, if that option is taken. The default is the same color as the confidence interval.</p>
</td></tr>
<tr><td><code id="plotting.cis_+3A_file.name">file.name</code></td>
<td>
<p>You can provide the filename. If not provided, default is &quot;OutlierLoci&quot; in the current directory.</p>
</td></tr>
<tr><td><code id="plotting.cis_+3A_make.file">make.file</code></td>
<td>
<p>A boolean value (TRUE or FALSE). If TRUE, a file will be created with the plot. If FALSE, the plot will be made in R only (and can be further annotated).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'> gpop&lt;-data.frame(popinfo=c(rep("POP 1", 20),rep("POP 2", 20)),ind.names=c(1:20,1:20),
                  loc0=sample(c("0101","0102","0202"),40,replace=TRUE),
                  loc1=sample(c("0101","0102","0202"),40,replace=TRUE))
  fsts&lt;-calc.actual.fst(gpop)
  bins&lt;-make.bins(fsts)
  cis&lt;-find.quantiles(bins = bins$bins,bin.fst = bins$bin.fst)
  quant.list&lt;-cis$CI0.95
  plotting.cis(df=fsts,ci.df=quant.list,make.file=FALSE)
  ## Not run: 
  load(fsts)
  bins&lt;-make.bins(fsts)
  cis&lt;-find.quantiles(bins = bins$bins,bin.fst = bins$bin.fst)
  quant.list&lt;-cis$CI0.95
  plotting.cis(df=fsts,ci.df=quant.list,make.file=FALSE)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='quant.out'>Example fsthet output based on numerical simulations</h2><span id='topic+quant.out'></span>

<h3>Description</h3>

<p>Example fsthet output based on numerical simulations
Allelic information was generated by using a numerical analysis with Nm = 10, 75 demes, and 5 population samples taken. No selection was imposed.
This is a list of three structures. The first is a data.frame containing the Ht and Fst values. The second is a data.frame of the bins with the lower heterozygosity values and the upper heterozygosity values for each bin. The third is a list of data.frames with the lower (Low) and upper (Upp) Fst values for each bin (the bins are in &quot;LowHet&quot; and &quot;UppHet&quot; columns.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quant.out</code></pre>


<h3>Format</h3>

<p>list</p>


<h3>Source</h3>

<p>Smoothed quantiles generated from the dataframe gpop.</p>


<h3>References</h3>

<p>See Flanagan &amp; Jones
</p>

<hr>
<h2 id='remove.spaces'>This removes spaces from a character vector</h2><span id='topic+remove.spaces'></span>

<h3>Description</h3>

<p>This removes spaces from a before and after words in a character vector. 
It was adapted from a similar function in adegenet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove.spaces(charvec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove.spaces_+3A_charvec">charvec</code></td>
<td>
<p>is a vector of characters containing spaces to be removed.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>charvec</code></td>
<td>
<p>A vector of characters without spaces</p>
</td></tr>
</table>


<h3>References</h3>

<p><a href="http://adegenet.r-forge.r-project.org/">http://adegenet.r-forge.r-project.org/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>charvec&lt;-c("this ", " is"," a"," test")
remove.spaces(charvec)
</code></pre>

<hr>
<h2 id='var.fst'>This calculates Cockerham &amp; Weir's Beta. </h2><span id='topic+var.fst'></span>

<h3>Description</h3>

<p>This calculates Weir &amp; Cockerham (1993)'s Fst. 
The caluclation is based on variance in allele frequencies.
See the vignette for details on the calculation of beta.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>var.fst(df, i)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="var.fst_+3A_df">df</code></td>
<td>
<p>A dataframe containing the genepop information, where the first column is the population ID.</p>
</td></tr>
<tr><td><code id="var.fst_+3A_i">i</code></td>
<td>
<p>Column number containing genotype information.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>ht</code></td>
<td>
<p>2pbar(1-pbar). This is a single numerical value.</p>
</td></tr>
<tr><td><code>fst</code></td>
<td>
<p>The calculated beta value for this locus.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  gpop&lt;-data.frame(popinfo=c(rep("POP 1", 20),rep("POP 2", 20)),ind.names=c(1:20,1:20),
                  loc0=sample(c("0101","0102","0202"),40,replace=TRUE),
                  loc1=sample(c("0101","0102","0202"),40,replace=TRUE))
  var1&lt;-var.fst(gpop,3) 
  ## Not run: 
    gfile&lt;-system.file("extdata", "example.genepop.txt",package = 'fsthet')
    gpop&lt;-my.read.gpop(gfile)
    var1&lt;-var.fst(gpop,3) #calculate variance-based for the first SNP
  
## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
