<!DOCTYPE html><html lang="en"><head><title>Help for package FSTpackage</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FSTpackage}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FST.example'>
<p>Data example for FSTest (tests for genetic association allowing for multiple functional annotation scores)</p></a></li>
<li><a href='#FST.GeneSet.test'>
<p>Test the association between an quantitative/dichotomous outcome variable and a large gene-set by a score type test allowing for multiple functional annotation scores.</p></a></li>
<li><a href='#FST.prelim'>
<p>The preliminary data management for FST (functional score tests)</p></a></li>
<li><a href='#FST.SummaryStat.test'>
<p>Using summary statistics to test the association between an quantitative/dichotomous outcome variable and a gene by a score type test allowing for multiple functional annotation scores.</p></a></li>
<li><a href='#FST.test'>
<p>Test the association between an quantitative/dichotomous outcome variable and a gene by a score type test allowing for multiple functional annotation scores.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Unified Sequence-Based Association Tests Allowing for Multiple
Functional Annotation Scores</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-12-14</td>
</tr>
<tr>
<td>Author:</td>
<td>Zihuai He</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zihuai He &lt;zihuai@umich.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for sequencing studies allowing for multiple functional annotation scores. Score type tests and an efficient perturbation method are used for individual gene/large gene-set/genome wide analysis. Only summary statistics are needed.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), CompQuadForm, SKAT, Matrix, MASS, mvtnorm</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-06-27 12:07:00 UTC; statzihuai</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-06-27 16:00:43 UTC</td>
</tr>
</table>
<hr>
<h2 id='FST.example'>
Data example for FSTest (tests for genetic association allowing for multiple functional annotation scores)
</h2><span id='topic+FST.example'></span>

<h3>Description</h3>

<p>The dataset contains outcome variable Y, covariate X, genotype data G, functional scores Z and gene-set ID for each variable GeneSetID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(FST.example)</code></pre>

<hr>
<h2 id='FST.GeneSet.test'>
Test the association between an quantitative/dichotomous outcome variable and a large gene-set by a score type test allowing for multiple functional annotation scores.
</h2><span id='topic+FST.GeneSet.test'></span>

<h3>Description</h3>

<p>Once the preliminary work is done using &quot;FST.prelim()&quot;, this function tests a specifc gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FST.GeneSet.test(result.prelim,G,Z,GeneSetID,Gsub.id=NULL,weights=NULL,
B=5000,impute.method='fixed')
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FST.GeneSet.test_+3A_result.prelim">result.prelim</code></td>
<td>

<p>The output of function &quot;FST.prelim()&quot;
</p>
</td></tr>
<tr><td><code id="FST.GeneSet.test_+3A_g">G</code></td>
<td>

<p>Genetic variants in the target gene, an n*p matrix where n is the subject ID and p is the total number of genetic variables. Note that the number of rows in G should be same as the number of subjects. ***The column name should be the variable name, in order to be matched with the GeneSetID.
</p>
</td></tr>
<tr><td><code id="FST.GeneSet.test_+3A_z">Z</code></td>
<td>

<p>Functional annotation scores, an p*q matrix where p is the total number of genetic variables and q is the number of functional annotation scores. Note that the first column in Z should be all 1 if the users want the original weights of SKAT/burden test to be included.
</p>
</td></tr>
<tr><td><code id="FST.GeneSet.test_+3A_genesetid">GeneSetID</code></td>
<td>

<p>A p*2 matrix indicating the genes in which the variables are located, where the first column is the genes' name and the second column is the variables' name.
</p>
</td></tr>
<tr><td><code id="FST.GeneSet.test_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an n dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X and time.
</p>
</td></tr>
<tr><td><code id="FST.GeneSet.test_+3A_weights">weights</code></td>
<td>

<p>A numeric vector of weights for genetic variants (The length should be same as the number of genetic variants in the set). These weights are usually based on minor allele frequencies. The default is NULL, where the beta(1,25) weights are applied.
</p>
</td></tr>
<tr><td><code id="FST.GeneSet.test_+3A_b">B</code></td>
<td>

<p>Number of Bootstrap replicates. The default is 5000.
</p>
</td></tr>
<tr><td><code id="FST.GeneSet.test_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;Best guess&quot; uses the genotype with highest probability.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>n.marker</code></td>
<td>

<p>number of heterozygous SNPs in the SNP set.
</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>

<p>P-value of the set based generalized score type test.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## FST.prelim does the preliminary data management.
# Input: Y, X (covariates)
## FST.test tests a region.
# Input: G (genetic variants), Z (functional annotation scores) and result of FST.prelim

library(FSTpackage)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by d matrix
# G: genotype matrix, n by p matrix where n is the total number of subjects
# Z: functional annotation matrix, p by q matrix

data(FST.example)
Y&lt;-FST.example$Y;X&lt;-FST.example$X;G&lt;-FST.example$G;Z&lt;-FST.example$Z;GeneSetID&lt;-FST.example$GeneSetID

# Preliminary data management
result.prelim&lt;-FST.prelim(Y,X=X,out_type='D')

# test with 5000 bootstrap replicates
result&lt;-FST.GeneSet.test(result.prelim,G,Z,GeneSetID,B=5000)
</code></pre>

<hr>
<h2 id='FST.prelim'>
The preliminary data management for FST (functional score tests)
</h2><span id='topic+FST.prelim'></span>

<h3>Description</h3>

<p>Before testing a specific gene using a score type test, this function does the preliminary data management, such as fitting the model under the null hypothesis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FST.prelim(Y, X=NULL, id=NULL, out_type="C")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FST.prelim_+3A_y">Y</code></td>
<td>

<p>The outcome variable, an n*1 matrix where n is the total number of observations
</p>
</td></tr>
<tr><td><code id="FST.prelim_+3A_x">X</code></td>
<td>

<p>An n*d covariates matrix where d is the total number of covariates.
</p>
</td></tr>
<tr><td><code id="FST.prelim_+3A_id">id</code></td>
<td>

<p>The subject id. This is used to match the genotype matrix. The default is NULL, where the a matched phenotype and genotype matrix is assumed.
</p>
</td></tr>
<tr><td><code id="FST.prelim_+3A_out_type">out_type</code></td>
<td>

<p>Type of outcome variable. Can be either &quot;C&quot; for continuous or &quot;D&quot; for dichotomous. The default is &quot;C&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list used for function FST.test().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FSTpackage)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by d matrix
# G: genotype matrix, n by p matrix where n is the total number of subjects
# Z: functional annotation matrix, p by q matrix

data(FST.example)
Y&lt;-FST.example$Y;X&lt;-FST.example$X;G&lt;-FST.example$G

# Preliminary data management
result.prelim&lt;-FST.prelim(Y,X=X)

</code></pre>

<hr>
<h2 id='FST.SummaryStat.test'>
Using summary statistics to test the association between an quantitative/dichotomous outcome variable and a gene by a score type test allowing for multiple functional annotation scores.
</h2><span id='topic+FST.SummaryStat.test'></span>

<h3>Description</h3>

<p>This function tests a specific gene using summary statistics (score vector and its covariance matrix)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FST.SummaryStat.test(score,Sigma,Z,weights,B=5000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FST.SummaryStat.test_+3A_score">score</code></td>
<td>

<p>The score vector of length p, where p is the total number of genetic variables.
</p>
</td></tr>
<tr><td><code id="FST.SummaryStat.test_+3A_sigma">Sigma</code></td>
<td>

<p>The p*p covariance matrix of the score vector
</p>
</td></tr>
<tr><td><code id="FST.SummaryStat.test_+3A_z">Z</code></td>
<td>

<p>Functional annotation scores, an p*q matrix where p is the total number of genetic variables and q is the number of functional annotation scores. Note that the first column in Z should be all 1 if the users want the original weights of SKAT/burden test to be included.
</p>
</td></tr>
<tr><td><code id="FST.SummaryStat.test_+3A_weights">weights</code></td>
<td>

<p>A numeric vector of weights for genetic variants (The length should be same as the number of genetic variants in the set.). These weights are usually based on minor allele frequencies.
</p>
</td></tr>
<tr><td><code id="FST.SummaryStat.test_+3A_b">B</code></td>
<td>

<p>Number of Bootstrap replicates. The default is 5000.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>p.value</code></td>
<td>

<p>P-value of the set based generalized score type test.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## FST.SummaryStat.test tests a region.
# Input: score (a score vector), Sigma (the covariance matrix of the score vector)

library(FSTpackage)

data(FST.example)
score&lt;-FST.example$score;Sigma&lt;-FST.example$Sigma;Z&lt;-FST.example$Z;weights&lt;-FST.example$weights

# test with 5000 bootstrap replicates
result&lt;-FST.SummaryStat.test(score,Sigma,Z,weights,B=5000)
</code></pre>

<hr>
<h2 id='FST.test'>
Test the association between an quantitative/dichotomous outcome variable and a gene by a score type test allowing for multiple functional annotation scores.
</h2><span id='topic+FST.test'></span>

<h3>Description</h3>

<p>Once the preliminary work is done using &quot;FST.prelim()&quot;, this function tests a specifc gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FST.test(result.prelim,G,Z,Gsub.id=NULL,weights=NULL,B=5000,impute.method='fixed')
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FST.test_+3A_result.prelim">result.prelim</code></td>
<td>

<p>The output of function &quot;FST.prelim()&quot;
</p>
</td></tr>
<tr><td><code id="FST.test_+3A_g">G</code></td>
<td>

<p>Genetic variants in the target gene, an n*p matrix where n is the subject ID and p is the total number of genetic variables. Note that the number of rows in G should be same as the number of subjects.
</p>
</td></tr>
<tr><td><code id="FST.test_+3A_z">Z</code></td>
<td>

<p>Functional annotation scores, an p*q matrix where p is the total number of genetic variables and q is the number of functional annotation scores. Note that the first column in Z should be all 1 if the users want the original weights of SKAT/burden test to be included.
</p>
</td></tr>
<tr><td><code id="FST.test_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an n dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X and time.
</p>
</td></tr>
<tr><td><code id="FST.test_+3A_weights">weights</code></td>
<td>

<p>A numeric vector of weights for genetic variants (The length should be same as the number of genetic variants in the set.). These weights are usually based on minor allele frequencies. The default is NULL, where the beta(1,25) weights are applied.
</p>
</td></tr>
<tr><td><code id="FST.test_+3A_b">B</code></td>
<td>

<p>Number of Bootstrap replicates. The default is 5000.
</p>
</td></tr>
<tr><td><code id="FST.test_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;Best guess&quot; uses the genotype with highest probability.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>n.marker</code></td>
<td>

<p>number of heterozygous SNPs in the SNP set.
</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>

<p>P-value of the set based generalized score type test.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## FST.prelim does the preliminary data management.
# Input: Y, X (covariates)
## FST.test tests a region.
# Input: G (genetic variants), Z (functional annotation scores) and result of FST.prelim

library(FSTpackage)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by d matrix
# G: genotype matrix, n by p matrix where n is the total number of subjects
# Z: functional annotation matrix, p by q matrix

data(FST.example)
Y&lt;-FST.example$Y;X&lt;-FST.example$X;G&lt;-FST.example$G;Z&lt;-FST.example$Z

# Preliminary data management
result.prelim&lt;-FST.prelim(Y,X=X,out_type='D')

# test with 5000 bootstrap replicates
result&lt;-FST.test(result.prelim,G,Z,B=5000)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
