<!DOCTYPE html><html><head><title>Help for package ftrCOOL</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ftrCOOL}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AA2Binary'><p>Amino Acid To Binary (AA2Binary)</p></a></li>
<li><a href='#AAindex'><p>Amino Acid Index (AAindex)</p></a></li>
<li><a href='#AAKpartComposition'><p>Amino Acid to K Part Composition (AAKpartComposition)</p></a></li>
<li><a href='#AAutoCor'><p>Amino Acid Autocorrelation-Autocovariance (AAutoCor)</p></a></li>
<li><a href='#AESNN3'><p>Learn from alignments (AESNN3)</p></a></li>
<li><a href='#alphabetCheck'><p>AlphabetCheck</p></a></li>
<li><a href='#ANF_DNA'><p>Accumulated Nucleotide Frequency (ANF_DNA)</p></a></li>
<li><a href='#ANF_RNA'><p>Accumulated riboNucleotide Frequency (ANF_RNA)</p></a></li>
<li><a href='#APAAC'><p>Amphiphilic Pseudo-Amino Acid Composition(series) (APAAC)</p></a></li>
<li><a href='#APkNUCdi_DNA'><p>Amphiphilic Pseudo-k Nucleotide Composition-di(series) (APkNUCdi_DNA)</p></a></li>
<li><a href='#APkNUCdi_RNA'><p>Amphiphilic Pseudo-k riboNucleotide Composition-di(series) (APkNUCdi_RNA)</p></a></li>
<li><a href='#APkNUCTri_DNA'><p>Amphiphilic Pseudo-k Nucleotide Composition-Tri(series) (APkNUCTri_DNA)</p></a></li>
<li><a href='#ASA'><p>Accessible Solvent Accessibility (ASA)</p></a></li>
<li><a href='#ASDC'><p>Adaptive skip dipeptide composition (ASDC)</p></a></li>
<li><a href='#ASDC_DNA'><p>Adaptive skip dinucleotide composition_DNA) (ASDC_DNA)</p></a></li>
<li><a href='#ASDC_RNA'><p>Adaptive skip di-ribonucleotide composition) (ASDC_RNA)</p></a></li>
<li><a href='#AutoCorDiNUC_DNA'><p>Di Nucleotide Autocorrelation-Autocovariance (AutoCorDiNUC_DNA)</p></a></li>
<li><a href='#AutoCorDiNUC_RNA'><p>Di riboNucleotide Autocorrelation-Autocovariance (AutoCorDiNUC_RNA)</p></a></li>
<li><a href='#AutoCorTriNUC_DNA'><p>Tri Nucleotide Autocorrelation-Autocovariance (AutoCorTriNUC_DNA)</p></a></li>
<li><a href='#binary_3bit_T1'><p>Binary - 3bit - Type1 (binary_3bit_T1)</p></a></li>
<li><a href='#binary_3bit_T2'><p>Binary - 3bit - Type2 (binary_3bit_T2)</p></a></li>
<li><a href='#binary_3bit_T3'><p>Binary - 3bit - Type3 (binary_3bit_T3)</p></a></li>
<li><a href='#binary_3bit_T4'><p>Binary - 3bit - Type4 (binary_3bit_T4)</p></a></li>
<li><a href='#binary_3bit_T5'><p>Binary - 3bit - Type5 (binary_3bit_T5)</p></a></li>
<li><a href='#binary_3bit_T6'><p>Binary - 3bit - Type6 (binary_3bit_T6)</p></a></li>
<li><a href='#binary_3bit_T7'><p>Binary - 3bit - Type7 (binary_3bit_T7)</p></a></li>
<li><a href='#binary_5bit_T1'><p>Binary - 5bit - Type1 (binary_5bit_T1)</p></a></li>
<li><a href='#binary_5bit_T2'><p>Binary - 5bit - Type2 (binary_5bit_T2)</p></a></li>
<li><a href='#binary_6bit'><p>Binary - 6bit (binary_6bit)</p></a></li>
<li><a href='#BLOSUM62'><p>Blosum62 (BLOSUM62)</p></a></li>
<li><a href='#CkSAApair'><p>Composition of k-Spaced Amino Acids pairs (CkSAApair)</p></a></li>
<li><a href='#CkSGAApair'><p>Composition of k-Spaced Grouped Amino Acids pairs (CkSGAApair)</p></a></li>
<li><a href='#CkSNUCpair_DNA'><p>Composition of k-Spaced Nucleotides Pairs (CkSNUCpair_DNA)</p></a></li>
<li><a href='#CkSNUCpair_RNA'><p>Composition of k-Spaced riboNucleotides Pairs (CkSNUCpair_RNA)</p></a></li>
<li><a href='#codonAdaptionIndex'><p>Codon Adaption Index (codonAdaptionIndex)</p></a></li>
<li><a href='#CodonFraction'><p>Codon Fraction (CodonFraction)</p></a></li>
<li><a href='#CodonUsage_DNA'><p>Codon Usage in DNA (CodonUsage_DNA)</p></a></li>
<li><a href='#CodonUsage_RNA'><p>Codon Usage in RNA (CodonUsage_RNA)</p></a></li>
<li><a href='#conjointTriad'><p>Conjoint Triad (conjointTriad)</p></a></li>
<li><a href='#conjointTriadKS'><p>k-Spaced Conjoint Triad (conjointTriadKS)</p></a></li>
<li><a href='#CTD'><p>Composition_Transition_Distribution (CTD)</p></a></li>
<li><a href='#CTDC'><p>CTD Composition (CTDC)</p></a></li>
<li><a href='#CTDD'><p>CTD Distribution (CTDD)</p></a></li>
<li><a href='#CTDT'><p>CTD Transition (CTDT)</p></a></li>
<li><a href='#DDE'><p>Dipeptide Deviation from Expected Mean value (DDE)</p></a></li>
<li><a href='#DiNUC2Binary_DNA'><p>Dinucleotide To Binary DNA (DiNUC2Binary_DNA)</p></a></li>
<li><a href='#DiNUC2Binary_RNA'><p>Di riboNucleotide To Binary RNA (DiNUC2Binary_RNA)</p></a></li>
<li><a href='#DiNUCindex_DNA'><p>Di Nucleotide Index (DiNUCindex_DNA)</p></a></li>
<li><a href='#DiNUCindex_RNA'><p>Di riboNucleotide Index (DiNUCindex_RNA)</p></a></li>
<li><a href='#DisorderB'><p>disorder Binary (DisorderB)</p></a></li>
<li><a href='#DisorderC'><p>disorder Content (DisorderC)</p></a></li>
<li><a href='#DisorderS'><p>disorder Simple (DisorderS)</p></a></li>
<li><a href='#DistancePair'><p>PseAAC of distance-pairs and reduced alphabet (DistancePair)</p></a></li>
<li><a href='#DPCP_DNA'><p>Dinucleotide physicochemical properties (DPCP_DNA)</p></a></li>
<li><a href='#DPCP_RNA'><p>Di-ribonucleotide physicochemical properties (DPCP_RNA)</p></a></li>
<li><a href='#EAAComposition'><p>Enhanced Amino Acid Composition (EAAComposition)</p></a></li>
<li><a href='#EffectiveNumberCodon'><p>Effective Number of Codon (EffectiveNumberCodon)</p></a></li>
<li><a href='#EGAAComposition'><p>Enhanced Grouped Amino Acid Composition (EGAAComposition)</p></a></li>
<li><a href='#EIIP'><p>Electron-Ion Interaction Pseudopotentials (EIIP)</p></a></li>
<li><a href='#ENUComposition_DNA'><p>Enhanced Nucleotide Composition (ENUComposition_DNA)</p></a></li>
<li><a href='#ENUComposition_RNA'><p>Enhanced riboNucleotide Composition (ENUComposition_RNA)</p></a></li>
<li><a href='#ExpectedValKmerNUC_DNA'><p>Expected Value for K-mer Nucleotide (ExpectedValKmerNUC_DNA)</p></a></li>
<li><a href='#ExpectedValKmerNUC_RNA'><p>Expected Value for K-mer riboNucleotide (ExpectedValKmerNUC_RNA)</p></a></li>
<li><a href='#ExpectedValueAA'><p>Expected Value for each Amino Acid (ExpectedValueAA)</p></a></li>
<li><a href='#ExpectedValueGAA'><p>Expected Value for Grouped Amino Acid (ExpectedValueGAA)</p></a></li>
<li><a href='#ExpectedValueGKmerAA'><p>Expected Value for Grouped K-mer Amino Acid(ExpectedValueGKmerAA)</p></a></li>
<li><a href='#ExpectedValueKmerAA'><p>Expected Value for K-mer Amino Acid (ExpectedValueKmerAA)</p></a></li>
<li><a href='#fa.read'><p>Fasta File Reader (fa.read)</p></a></li>
<li><a href='#fickettScore'><p>Fickett Score (fickettScore)</p></a></li>
<li><a href='#G_Ccontent_DNA'><p>G_C content in DNA (G_Ccontent_DNA)</p></a></li>
<li><a href='#G_Ccontent_RNA'><p>G_C content in RNA (G_Ccontent_RNA)</p></a></li>
<li><a href='#GAAKpartComposition'><p>Grouped Amino Acid K Part Composition (GAAKpartComposition)</p></a></li>
<li><a href='#GrpDDE'><p>Group Dipeptide Deviation from Expected Mean (GrpDDE)</p></a></li>
<li><a href='#kAAComposition'><p>k Amino Acid Composition (kAAComposition)</p></a></li>
<li><a href='#kGAAComposition'><p>k Grouped Amino Acid Composition (kGAAComposition)</p></a></li>
<li><a href='#KNN_DNA'><p>K-Nearest Neighbor_DNA (KNN_DNA)</p></a></li>
<li><a href='#KNN_RNA'><p>K-Nearest Neighbor_RNA (KNN_RNA)</p></a></li>
<li><a href='#KNNPeptide'><p>K-Nearest Neighbor for Peptides (KNNPeptide)</p></a></li>
<li><a href='#KNNProtein'><p>K-Nearest Neighbor for Protein (KNNProtein)</p></a></li>
<li><a href='#kNUComposition_DNA'><p>k Nucleotide Composition (kNUComposition_DNA)</p></a></li>
<li><a href='#kNUComposition_RNA'><p>k riboNucleotide Composition (kNUComposition_RNA)</p></a></li>
<li><a href='#LocalPoSpKAAF'><p>Local Position Specific k Amino Acids Frequency (LocalPoSpKAAF)</p></a></li>
<li><a href='#LocalPoSpKNUCF_DNA'><p>Local Position Specific k Nucleotide Frequency (LocalPoSpKNUCF_DNA)</p></a></li>
<li><a href='#LocalPoSpKNUCF_RNA'><p>Local Position Specific k riboNucleotide Frequency (LocalPoSpKNUCF_RNA)</p></a></li>
<li><a href='#maxORF'><p>Maximum Open Reading Frame in DNA (maxORF)</p></a></li>
<li><a href='#maxORF_RNA'><p>Maximum Open Reading Frame in RNA (maxORF_RNA)</p></a></li>
<li><a href='#maxORFlength_DNA'><p>Maximum Open Reading Frame length in DNA (maxORFlength_DNA)</p></a></li>
<li><a href='#maxORFlength_RNA'><p>Maximum Open Reading Frame length in RNA (maxORFlength_RNA)</p></a></li>
<li><a href='#Mismatch_DNA'><p>Mismatch_DNA (Mismatch_DNA)</p></a></li>
<li><a href='#Mismatch_RNA'><p>Mismatch_RNA (Mismatch_RNA)</p></a></li>
<li><a href='#MMI_DNA'><p>Multivariate Mutual Information_DNA (MMI_DNA)</p></a></li>
<li><a href='#MMI_RNA'><p>Multivariate Mutual Information_RNA (MMI_RNA)</p></a></li>
<li><a href='#nameKmer'><p>naming Kmer (nameKmer)</p></a></li>
<li><a href='#NCP_DNA'><p>Nucleotide Chemical Property (NCP_DNA)</p></a></li>
<li><a href='#NCP_RNA'><p>riboNucleotide Chemical Property (NCP_RNA)</p></a></li>
<li><a href='#needleman'><p>Needleman-Wunsch (needleman)</p></a></li>
<li><a href='#nonStandardSeq'><p>nonStandard sequence (nonStandardSeq)</p></a></li>
<li><a href='#NUC2Binary_DNA'><p>Nucleotide To Binary (NUC2Binary_DNA)</p></a></li>
<li><a href='#NUC2Binary_RNA'><p>riboNucleotide To Binary (NUC2Binary_RNA)</p></a></li>
<li><a href='#NUCKpartComposition_DNA'><p>Nucleotide to K Part Composition (NUCKpartComposition_DNA)</p></a></li>
<li><a href='#NUCKpartComposition_RNA'><p>riboNucleotide to K Part Composition (NUCKpartComposition_RNA)</p></a></li>
<li><a href='#OPF_10bit'><p>Overlapping Property Features_10bit (OPF_10bit)</p></a></li>
<li><a href='#OPF_7bit_T1'><p>Overlapping property features_7bit_T1 (OPF_7bit_T1)</p></a></li>
<li><a href='#OPF_7bit_T2'><p>Overlapping property features_7bit_T2 (OPF_7bit_T2)</p></a></li>
<li><a href='#OPF_7bit_T3'><p>Overlapping property features_7bit_T3 (OPF_7bit_T3)</p></a></li>
<li><a href='#PCPseDNC'><p>Parallel Correlation Pseudo Dinucleotide Composition (PCPseDNC)</p></a></li>
<li><a href='#PS2_DNA'><p>Position-specific of two nucleotide_DNA (PS2_DNA)</p></a></li>
<li><a href='#PS2_RNA'><p>Position-specific of two nucleotide_RNA (PS2_RNA)</p></a></li>
<li><a href='#PS3_DNA'><p>Position-specific of three nucleotide_DNA (PS3_DNA)</p></a></li>
<li><a href='#PS3_RNA'><p>Position-specific of three ribonucleotide_RNA (PS3_RNA)</p></a></li>
<li><a href='#PS4_DNA'><p>Position-specific of four nucleotide_DNA (PS4_DNA)</p></a></li>
<li><a href='#PS4_RNA'><p>Position-specific of four ribonucleotide (PS4_RNA)</p></a></li>
<li><a href='#PSEAAC'><p>Pseudo-Amino Acid Composition (Parallel) (PSEAAC)</p></a></li>
<li><a href='#PseEIIP'><p>Pseudo Electron-Ion Interaction Pseudopotentials of Trinucleotide (PseEIIP)</p></a></li>
<li><a href='#PSEkNUCdi_DNA'><p>Pseudo k Nucleotide Composition-Di(Parallel) (PSEkNUCdi_DNA)</p></a></li>
<li><a href='#PSEkNUCdi_RNA'><p>Pseudo k riboNucleotide Composition-Di(Parallel) (PSEkNUCdi_RNA)</p></a></li>
<li><a href='#PSEkNUCTri_DNA'><p>Pseudo k Nucleotide Composition-Tri(Parallel) (PSEkNUCTri_RNA)</p></a></li>
<li><a href='#PseKRAAC_T1'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-1 (PseKRAAC_T1)</p></a></li>
<li><a href='#PseKRAAC_T10'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-10 (PseKRAAC_T10)</p></a></li>
<li><a href='#PseKRAAC_T11'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-11 (PseKRAAC_T11)</p></a></li>
<li><a href='#PseKRAAC_T12'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-12 (PseKRAAC_T12)</p></a></li>
<li><a href='#PseKRAAC_T13'><p>Pseudo K_tuple Reduced Amino Acid Composition Type_13 (PseKRAAC_T13)</p></a></li>
<li><a href='#PseKRAAC_T14'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-14 (PseKRAAC_T14)</p></a></li>
<li><a href='#PseKRAAC_T15'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-15 (PseKRAAC_T15)</p></a></li>
<li><a href='#PseKRAAC_T16'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-16 (PseKRAAC_T16)</p></a></li>
<li><a href='#PseKRAAC_T2'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-2 (PseKRAAC_T2)</p></a></li>
<li><a href='#PseKRAAC_T3A'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-3A (PseKRAAC_T3A)</p></a></li>
<li><a href='#PseKRAAC_T3B'><p>Pseudo K_tuple Reduced Amino Acid Composition Type_3B (PseKRAAC_T3B)</p></a></li>
<li><a href='#PseKRAAC_T4'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-4 (PseKRAAC_T4)</p></a></li>
<li><a href='#PseKRAAC_T5'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-5 (PseKRAAC_T5)</p></a></li>
<li><a href='#PseKRAAC_T6A'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-6A (PseKRAAC_T6A)</p></a></li>
<li><a href='#PseKRAAC_T6B'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-6B (PseKRAAC_T6B)</p></a></li>
<li><a href='#PseKRAAC_T7'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-7 (PseKRAAC_T7)</p></a></li>
<li><a href='#PseKRAAC_T8'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-8 (PseKRAAC_T8)</p></a></li>
<li><a href='#PseKRAAC_T9'><p>Pseudo K_tuple Reduced Amino Acid Composition Type-9 (PseKRAAC_T9)</p></a></li>
<li><a href='#PSSM'><p>Position-Specific Scoring Matrix (PSSM)</p></a></li>
<li><a href='#PSTNPds'><p>Position-Specific Trinucleotide Propensity based on double-strand (PSTNPds)</p></a></li>
<li><a href='#PSTNPss_DNA'><p>Position-Specific Trinucleotide Propensity based on single-strand DNA (PSTNPss_DNA)</p></a></li>
<li><a href='#PSTNPss_RNA'><p>Position-Specific Tri-ribonucleotide Propensity based on single-strand RNA (PSTNPss_RNA)</p></a></li>
<li><a href='#QSOrder'><p>Quasi Sequence Order (QSOrder)</p></a></li>
<li><a href='#readASAdir'><p>Read Directory of Accessible Solvent accessibility predicted files (readASAdir)</p></a></li>
<li><a href='#readDisDir'><p>Read disorder predicted Directory (readDisDir)</p></a></li>
<li><a href='#readPSSMdir'><p>Read PSSM Directory (readPSSMdir)</p></a></li>
<li><a href='#readss2Dir'><p>Read ss2 predicted Directory (readss2Dir)</p></a></li>
<li><a href='#readTorsionDir'><p>Read Directory of Torsion predicted files (readTorsionDir)</p></a></li>
<li><a href='#revComp'><p>reverseCompelement (revComp)</p></a></li>
<li><a href='#SAAC'><p>Splitted Amino Acid Composition (SAAC)</p></a></li>
<li><a href='#SGAAC'><p>Splitted Group Amino Acid Composition (SGAAC)</p></a></li>
<li><a href='#SOCNumber'><p>Sequence Order Coupling Number (SOCNumber)</p></a></li>
<li><a href='#SSEB'><p>Secondary Structure Elements Binary (SSEB)</p></a></li>
<li><a href='#SSEC'><p>Secondary Structure Elements Composition (SSEC)</p></a></li>
<li><a href='#SSES'><p>Secondary Structure Elements Simple (SSES)</p></a></li>
<li><a href='#TorsionAngle'><p>Torsion Angle (TorsionAngle)</p></a></li>
<li><a href='#TPCP_DNA'><p>Trinucleotide physicochemical properties (TPCP_DNA)</p></a></li>
<li><a href='#TriNUCindex_DNA'><p>Tri Nucleotide Index (TriNucIndex)</p></a></li>
<li><a href='#Zcurve12bit_DNA'><p>Z_curve_12bit_DNA (Zcurve12bit_DNA)</p></a></li>
<li><a href='#Zcurve12bit_RNA'><p>Z_curve_12bit_RNA (Zcurve12bit_RNA)</p></a></li>
<li><a href='#Zcurve144bit_DNA'><p>Z_curve_144bit_DNA (Zcurve144bit_DNA)</p></a></li>
<li><a href='#Zcurve144bit_RNA'><p>Z_curve_144bit_RNA (Zcurve144bit_RNA)</p></a></li>
<li><a href='#Zcurve36bit_DNA'><p>Z_curve_36bit_DNA (Zcurve36bit_DNA)</p></a></li>
<li><a href='#Zcurve36bit_RNA'><p>Z_curve_36bit_RNA (Zcurve36bit_RNA)</p></a></li>
<li><a href='#Zcurve48bit_DNA'><p>Z_curve_48bit_DNA (Zcurve48bit_DNA)</p></a></li>
<li><a href='#Zcurve48bit_RNA'><p>Z_curve_48bit_RNA (Zcurve48bit_RNA)</p></a></li>
<li><a href='#Zcurve9bit_DNA'><p>Z_curve_9bit_DNA (Zcurve9bit_DNA)</p></a></li>
<li><a href='#Zcurve9bit_RNA'><p>Z_curve_9bit_RNA (Zcurve9bit_RNA)</p></a></li>
<li><a href='#zSCALE'><p>Z-SCALE (zSCALE)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Feature Extraction from Biological Sequences</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Sare Amerifar</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sare Amerifar &lt;sare.ameri.01@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Extracts features from biological sequences. It contains most features which are presented in related work and also includes features which have never been introduced before. It extracts numerous features from nucleotide and peptide sequences. Each feature converts the input sequences to discrete numbers in order to use them as predictors in machine learning models. There are many features and information which are hidden inside a sequence. Utilizing the package, users can convert biological sequences to discrete models based on chosen properties. References: 'iLearn' 'Z. Chen et al.' (2019) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbz041">doi:10.1093/bib/bbz041</a>&gt;. 'iFeature' 'Z. Chen et al.' (2018) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbty140">doi:10.1093/bioinformatics/bty140</a>&gt;. <a href="https://CRAN.R-project.org/package=rDNAse">https://CRAN.R-project.org/package=rDNAse</a>. 'PseKRAAC' 'Y. Zuo et al.' 'PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition' (2017) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw564">doi:10.1093/bioinformatics/btw564</a>&gt;. 'iDNA6mA-PseKNC' 'P. Feng et al.' 'iDNA6mA-PseKNC: Identifying DNA N6-methyladenosine sites by incorporating nucleotide physicochemical properties into PseKNC' (2019) &lt;<a href="https://doi.org/10.1016%2Fj.ygeno.2018.01.005">doi:10.1016/j.ygeno.2018.01.005</a>&gt;. 'I. Dubchak et al.' 'Prediction of protein folding class using global description of amino acid sequence' (1995) &lt;<a href="https://doi.org/10.1073%2Fpnas.92.19.8700">doi:10.1073/pnas.92.19.8700</a>&gt;. 'W. Chen et al.' 'Identification and analysis of the N6-methyladenosine in the Saccharomyces cerevisiae transcriptome' (2015) &lt;<a href="https://doi.org/10.1038%2Fsrep13859">doi:10.1038/srep13859</a>&gt;. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-29 22:52:57 UTC; sareameri</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-29 23:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='AA2Binary'>Amino Acid To Binary (AA2Binary)</h2><span id='topic+AA2Binary'></span>

<h3>Description</h3>

<p>This function transforms an amino acid to a binary format.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AA2Binary(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AA2Binary_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="AA2Binary_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each amino acid is represented by a string containing 20 characters(0-1). For example, A = ALANIN = &quot;1000000...0&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 20 logical entries. For example, A = ALANIN = c(T,F,F,F,F,F,F,...F)
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 20 numerals. For example, A = ALANIN = c(1,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="AA2Binary_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="AA2Binary_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="AA2Binary_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*20.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-AA2Binary(seqs = ptmSeqsVect, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='AAindex'>Amino Acid Index (AAindex)</h2><span id='topic+AAindex'></span>

<h3>Description</h3>

<p>This function converts the amino acids of a sequence to a list of physicochemical properties in the aaIndex file.
For each amino acid, the function uses a numeric vector which shows the aaIndex of the amino acid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AAindex(
  seqs,
  selectedAAidx = 1:554,
  standardized = TRUE,
  threshold = 1,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AAindex_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="AAindex_+3A_selectedaaidx">selectedAAidx</code></td>
<td>
<p>AAindex function works based on physicochemical properties. Users select the properties by their ids
or indexes in aaIndex2 file.</p>
</td></tr>
<tr><td><code id="AAindex_+3A_standardized">standardized</code></td>
<td>
<p>is a logical parameter. If it is set to TRUE, amino acid indices will be in the standard format.
The default value is TRUE.</p>
</td></tr>
<tr><td><code id="AAindex_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In selectedAAidx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="AAindex_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="AAindex_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="AAindex_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this function each amino acid is converted to a numeric vector. Elements of the vector represent a
physicochemical property for the amino acid.
In the aaIndex database, there are 554 amino acid indices. Users can choose the desired aaindex by specifying aaindexes through their ids or indexes in the aaIndex file, via selectedAAidx parameter.
</p>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length)*(number of selected amino acid indexes)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-AAindex(seqs = ptmSeqsVect, selectedAAidx=1:5,outFormat="mat")

ad&lt;-paste0(dir,"/aaidx.txt")
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
AAindex(seqs = filePrs, selectedAAidx=1:5,standardized=TRUE,threshold=1,outFormat="txt"
,outputFileDist=ad)

unlink("dir", recursive = TRUE)

</code></pre>

<hr>
<h2 id='AAKpartComposition'>Amino Acid to K Part Composition (AAKpartComposition)</h2><span id='topic+AAKpartComposition'></span>

<h3>Description</h3>

<p>In this function, each sequence is divided into k equal partitions.
The length of each part is equal to ceiling(l(lenght of the sequence)/k).
The last part can have a different length containing the residual amino acids.
The amino acid composition is calculated for each part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AAKpartComposition(seqs, k = 3, normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AAKpartComposition_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="AAKpartComposition_+3A_k">k</code></td>
<td>
<p>is an integer value. Each sequence should be divided to k partition(s).</p>
</td></tr>
<tr><td><code id="AAKpartComposition_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="AAKpartComposition_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a feature matrix with k*20 number of columns. The number of rows is equal to the number of
sequences.
</p>


<h3>Note</h3>

<p>Warning: The length of all sequences should be greater than k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-AAKpartComposition(seqs=filePrs,k=5,normalized=FALSE)

</code></pre>

<hr>
<h2 id='AAutoCor'>Amino Acid Autocorrelation-Autocovariance (AAutoCor)</h2><span id='topic+AAutoCor'></span>

<h3>Description</h3>

<p>It creates the feature matrix for each function in autocorelation
(i.e., Moran, Greay, NormalizeMBorto) or autocovariance (i.e., AC, CC,ACC). The user can
select any combination of the functions too. In this case, the final matrix will contain
features of each selected function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AAutoCor(
  seqs,
  selectedAAidx = list(c("CIDH920105", "BHAR880101", "CHAM820101", "CHAM820102",
    "CHOC760101", "BIGC670101", "CHAM810101", "DAYM780201")),
  maxlag = 3,
  threshold = 1,
  type = c("Moran", "Geary", "NormalizeMBorto", "AC", "CC", "ACC"),
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AAutoCor_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="AAutoCor_+3A_selectedaaidx">selectedAAidx</code></td>
<td>
<p>Function takes as input the physicochemical properties. Users select the properties by their ids
or indices in the aaIndex2 file. This parameter could be a vector or a list of amino acid indices.
The default values of the vector are the 'CIDH920105','BHAR880101','CHAM820101','CHAM820102','CHOC760101','BIGC670101','CHAM810101','DAYM780201'
ids in the aaIndex2 file.</p>
</td></tr>
<tr><td><code id="AAutoCor_+3A_maxlag">maxlag</code></td>
<td>
<p>This parameter shows the maximum gap between two amino acids. The gaps change from 1 to maxlag (the maximum lag).</p>
</td></tr>
<tr><td><code id="AAutoCor_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In selectedAAidx, indices with a correlation higher than the threshold will be deleted.
The default value is 1.</p>
</td></tr>
<tr><td><code id="AAutoCor_+3A_type">type</code></td>
<td>
<p>could be 'Moran', 'Greay', 'NormalizeMBorto', 'AC', 'CC', or 'ACC'. Also, it could be any combination of them.</p>
</td></tr>
<tr><td><code id="AAutoCor_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For CC and AAC autocovriance functions, which consider the covariance of the
two physicochemical properties, we have provided users with the ability to categorize
their selected properties in a list.
The binary combination of each group will be taken into account.
Note: If all the features are in a group or selectedAAidx parameter is a vector,
the binary combination will be calculated for all the physicochemical properties.
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of columns in the matrix changes depending on the chosen autocorrelation or autocovariance types and nlag parameter.
The output is a matrix. The number of rows shows the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-AAutoCor(seqs=filePrs,maxlag=20,threshold=0.9,
type=c("Moran","Geary","NormalizeMBorto","AC"))

mat2&lt;-AAutoCor(seqs=filePrs,maxlag=20,threshold=0.9,selectedAAidx=
list(c('CIDH920105','BHAR880101','CHAM820101','CHAM820102'),c('CHOC760101','BIGC670101')
,c('CHAM810101','DAYM780201')),type=c("AC","CC","ACC"))

</code></pre>

<hr>
<h2 id='AESNN3'>Learn from alignments (AESNN3)</h2><span id='topic+AESNN3'></span>

<h3>Description</h3>

<p>This function replace each amino acid of the sequence with a three-dimensional vector. Values are taken from the three hidden units of the neural network trained on structure alignments.
The AESNN3 function can be applied to encode peptides of equal length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AESNN3(seqs, label = c(), outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AESNN3_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="AESNN3_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="AESNN3_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="AESNN3_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length)*(5)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat parameter for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Lin K, May AC, Taylor WR. Amino acid encoding schemes from protein structure alignments: multi-dimensional vectors to describe residue types. J Theor Biol (2002).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-AESNN3(seqs = ptmSeqsVect,outFormat="mat")
</code></pre>

<hr>
<h2 id='alphabetCheck'>AlphabetCheck</h2><span id='topic+alphabetCheck'></span>

<h3>Description</h3>

<p>This function checks the alphabets in a sequence. If one of the following conditions hold, the sequence will be deleted:
1. A peptide sequence containing non-standard amino acids,
2. A DNA sequence with an alphabet other than A, C, G, or T,
3. An RNA sequence having an alphabet other than A, C, G, or U.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alphabetCheck(sequences, alphabet = "aa", label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alphabetCheck_+3A_sequences">sequences</code></td>
<td>
<p>is a string vector. Each element is a peptide, protein, DNA, or RNA sequences.</p>
</td></tr>
<tr><td><code id="alphabetCheck_+3A_alphabet">alphabet</code></td>
<td>
<p>This parameter shows the alphabet of sequences. If it is set to 'aa', it indicates the
alphabet of amino acids. When it is 'dna', it shows the nucleotide alphabet and in case it equals 'rna', it represents ribonucleotide alphabet.</p>
</td></tr>
<tr><td><code id="alphabetCheck_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'alphabetCheck' returns a list with two elements. The first element is a vector which contains
valid sequences. The second element is a vector which contains the labels of the sequences (if any exists).
</p>


<h3>Note</h3>

<p>This function receives a sequence vector and the label of sequences (if any).
It deletes sequences (and their labels) containing non-standard alphabets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
seq&lt;-alphabetCheck(sequences=c("AGDFLIAACNMLKIVYT","ADXVGAJK"),alphabet="aa")

</code></pre>

<hr>
<h2 id='ANF_DNA'>Accumulated Nucleotide Frequency (ANF_DNA)</h2><span id='topic+ANF_DNA'></span>

<h3>Description</h3>

<p>This function replaces nucleotides with a four-length vector.
The first three elements represent the nucleotides and
the forth holds the frequency of the nucleotide from the beginning of the sequence until the position of the nucleotide in the sequence.
'A' will be replaced with c(1, 1, 1, freq), 'C' with c(0, 1, 0, freq),'G' with c(1, 0, 0, freq), and 'T' with c(0, 0, 1, freq).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ANF_DNA(seqs, outFormat = "mat", outputFileDist = "", label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ANF_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="ANF_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="ANF_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
<tr><td><code id="ANF_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length)*(4)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Chen, W., Tran, H., Liang, Z. et al. Identification and analysis of the N6-methyladenosine in the Saccharomyces cerevisiae transcriptome. Sci Rep 5, 13859 (2015).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-ANF_DNA(seqs = LNC50Nuc,outFormat="mat")


</code></pre>

<hr>
<h2 id='ANF_RNA'>Accumulated riboNucleotide Frequency (ANF_RNA)</h2><span id='topic+ANF_RNA'></span>

<h3>Description</h3>

<p>This function replaces ribonucleotides with a four-length vector.
The first three elements represent the ribonucleotides and
the forth holds the frequency of the ribonucleotide from the beginning of the sequence until the position of the ribonucleotide in the sequence.
'A' will be replaced with c(1, 1, 1, freq), 'C' with c(0, 1, 0, freq),'G' with c(1, 0, 0, freq), and 'U' with c(0, 0, 1, freq).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ANF_RNA(seqs, outFormat = "mat", outputFileDist = "", label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ANF_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="ANF_RNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="ANF_RNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
<tr><td><code id="ANF_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length)*(4)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Chen, W., Tran, H., Liang, Z. et al. Identification and analysis of the N6-methyladenosine in the Saccharomyces cerevisiae transcriptome. Sci Rep 5, 13859 (2015).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-ANF_RNA(seqs = fileLNC,outFormat="mat")

</code></pre>

<hr>
<h2 id='APAAC'>Amphiphilic Pseudo-Amino Acid Composition(series) (APAAC)</h2><span id='topic+APAAC'></span>

<h3>Description</h3>

<p>This function calculates the amphiphilic pseudo amino acid composition (Series)
for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>APAAC(
  seqs,
  aaIDX = c("ARGP820101", "HOPT810101"),
  lambda = 30,
  w = 0.05,
  l = 1,
  threshold = 1,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="APAAC_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="APAAC_+3A_aaidx">aaIDX</code></td>
<td>
<p>is a vector of Ids or indexes of the user-selected physicochemical properties in the aaIndex2 database.
The default values of the vector are the hydrophobicity ids and hydrophilicity ids
in the amino acid index file.</p>
</td></tr>
<tr><td><code id="APAAC_+3A_lambda">lambda</code></td>
<td>
<p>is a tuning parameter. Its value indicates the maximum number of spaces between amino acid pairs. The number
changes from 1 to lambda.</p>
</td></tr>
<tr><td><code id="APAAC_+3A_w">w</code></td>
<td>
<p>(weight) is a tuning parameter. It changes in from 0 to 1. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="APAAC_+3A_l">l</code></td>
<td>
<p>This parameter keeps the value of l in lmer composition. The lmers form the first 20^l elements of the APAAC descriptor.</p>
</td></tr>
<tr><td><code id="APAAC_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In aaIDX, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="APAAC_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the pseudo amino acid composition for each physicochemical property.
We have provided users with the ability to choose among different properties (i.e., not confined to hydrophobicity or hydrophilicity).
</p>


<h3>Value</h3>

<p>A feature matrix such that the number of columns is 20^l+(number of chosen aaIndex*lambda) and the number of rows equals the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-APAAC(seqs=filePrs,l=2,lambda=3,threshold=1)

</code></pre>

<hr>
<h2 id='APkNUCdi_DNA'>Amphiphilic Pseudo-k Nucleotide Composition-di(series) (APkNUCdi_DNA)</h2><span id='topic+APkNUCdi_DNA'></span>

<h3>Description</h3>

<p>This function calculates the amphiphilic pseudo k nucleotide composition(Di) (Series)
for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>APkNUCdi_DNA(
  seqs,
  selectedIdx = c("Rise", "Roll", "Shift", "Slide", "Tilt", "Twist"),
  lambda = 3,
  w = 0.05,
  l = 2,
  ORF = FALSE,
  reverseORF = TRUE,
  threshold = 1,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="APkNUCdi_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="APkNUCdi_DNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>is a vector of Ids or indices of the desired physicochemical properties of dinucleotides.
Users can choose the desired indices by their ids or their names in the DI_DNA index file.
The default value of this parameter is a vector with (&quot;Rise&quot;, &quot;Roll&quot;, &quot;Shift&quot;, &quot;Slide&quot;, &quot;Tilt&quot;, &quot;Twist&quot;) ids.</p>
</td></tr>
<tr><td><code id="APkNUCdi_DNA_+3A_lambda">lambda</code></td>
<td>
<p>is a tuning parameter. This integer value shows the maximum limit of spaces between dinucleotide pairs. The Number of spaces
changes from 1 to lambda.</p>
</td></tr>
<tr><td><code id="APkNUCdi_DNA_+3A_w">w</code></td>
<td>
<p>(weight) is a tuning parameter. It changes in the range of 0 to 1. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="APkNUCdi_DNA_+3A_l">l</code></td>
<td>
<p>This parameter keeps the value of l in lmer composition. The lmers form the first 4^l elements of the APkNCdi descriptor.</p>
</td></tr>
<tr><td><code id="APkNUCdi_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="APkNUCdi_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="APkNUCdi_DNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 to 1]. In selectedIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="APkNUCdi_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the pseudo nucleotide composition for each physicochemical property of dinucleotides.
We have provided users with the ability to choose among the 148 properties in the di-nucleotide index database.
</p>


<h3>Value</h3>

<p>It is a feature matrix. The number of columns is 4^l+(number of the chosen indices*lambda) and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-APkNUCdi_DNA(seqs=fileLNC,ORF=TRUE,threshold=1)

</code></pre>

<hr>
<h2 id='APkNUCdi_RNA'>Amphiphilic Pseudo-k riboNucleotide Composition-di(series) (APkNUCdi_RNA)</h2><span id='topic+APkNUCdi_RNA'></span>

<h3>Description</h3>

<p>This function calculates the amphiphilic pseudo k ribonucleotide composition(Di) (Series)
for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>APkNUCdi_RNA(
  seqs,
  selectedIdx = c("Rise (RNA)", "Roll (RNA)", "Shift (RNA)", "Slide (RNA)",
    "Tilt (RNA)", "Twist (RNA)"),
  lambda = 3,
  w = 0.05,
  l = 2,
  ORF = FALSE,
  reverseORF = TRUE,
  threshold = 1,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="APkNUCdi_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="APkNUCdi_RNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>is a vector of Ids or indices of the desired physicochemical properties of di-ribonucleotides.
Users can choose the desired indices by their ids or their names in the DI_RNA index file.
The default value of this parameter is a vector with (&quot;Rise (RNA)&quot;, &quot;Roll (RNA)&quot;, &quot;Shift (RNA)&quot;, &quot;Slide (RNA)&quot;, &quot;Tilt (RNA)&quot;,&quot;Twist (RNA)&quot;) ids.</p>
</td></tr>
<tr><td><code id="APkNUCdi_RNA_+3A_lambda">lambda</code></td>
<td>
<p>is a tuning parameter. This integer value shows the maximum limit of spaces between di-ribonucleotide pairs. The Number of spaces
changes from 1 to lambda.</p>
</td></tr>
<tr><td><code id="APkNUCdi_RNA_+3A_w">w</code></td>
<td>
<p>(weight) is a tuning parameter. It changes in the range of 0 to 1. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="APkNUCdi_RNA_+3A_l">l</code></td>
<td>
<p>This parameter keeps the value of l in lmer composition. The lmers form the first 4^l elements of the APkNCdi descriptor.</p>
</td></tr>
<tr><td><code id="APkNUCdi_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="APkNUCdi_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="APkNUCdi_RNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 to 1]. In selectedIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="APkNUCdi_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the pseudo ribonucleotide composition for each physicochemical property of di-ribonucleotides.
We have provided users with the ability to choose among the 22 properties in the di-ribonucleotide index database.
</p>


<h3>Value</h3>

<p>It is a feature matrix. The number of columns is 4^l+(number of the chosen indices*lambda) and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-APkNUCdi_RNA(seqs=fileLNC,ORF=TRUE,threshold=0.8)

</code></pre>

<hr>
<h2 id='APkNUCTri_DNA'>Amphiphilic Pseudo-k Nucleotide Composition-Tri(series) (APkNUCTri_DNA)</h2><span id='topic+APkNUCTri_DNA'></span>

<h3>Description</h3>

<p>This function calculates the amphiphilic pseudo k nucleotide composition(Tri) (Series)
for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>APkNUCTri_DNA(
  seqs,
  selectedIdx = c("Dnase I", "Bendability (DNAse)"),
  lambda = 3,
  w = 0.05,
  l = 3,
  ORF = FALSE,
  reverseORF = TRUE,
  threshold = 1,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="APkNUCTri_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="APkNUCTri_DNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>is a vector of Ids or indices of the desired physicochemical properties of trinucleotides.
Users can choose the desired indices by their ids or their names in the TRI_DNA index file.
The default value of the parameter is a vector with (&quot;Dnase I&quot;, &quot;Bendability (DNAse)&quot;) ids.</p>
</td></tr>
<tr><td><code id="APkNUCTri_DNA_+3A_lambda">lambda</code></td>
<td>
<p>is a tuning parameter. This integer value shows the maximum limit of spaces between trinucleotide pairs. The Number of spaces
changes from 1 to lambda.</p>
</td></tr>
<tr><td><code id="APkNUCTri_DNA_+3A_w">w</code></td>
<td>
<p>(weight) is a tuning parameter. It changes in the range of 0 to 1. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="APkNUCTri_DNA_+3A_l">l</code></td>
<td>
<p>This parameter keeps the value of l in lmer composition. The lmers form the first 4^l of the APkNCTri descriptor.</p>
</td></tr>
<tr><td><code id="APkNUCTri_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="APkNUCTri_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="APkNUCTri_DNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In selectedIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="APkNUCTri_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the pseudo nucleotide composition for each physicochemical property of trinucleotides.
We have provided users with the ability to choose among the 12 properties in the tri-nucleotide index database.
</p>


<h3>Value</h3>

<p>It is a feature matrix. The number of columns is 4^l+(number of the chosen indices*lambda) and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-APkNUCTri_DNA(seqs=fileLNC,l=3,threshold=1)

</code></pre>

<hr>
<h2 id='ASA'>Accessible Solvent Accessibility (ASA)</h2><span id='topic+ASA'></span>

<h3>Description</h3>

<p>ASA represents an amino acid by a numeric value.
This function extracts the ASA from the output of SPINE-X software which predicts ASA for each amino acid in a peptide or protein sequence.
The output of SPINE-X is a tab-delimited file. ASAs are in the 11th column of the file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ASA(dirPath, outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ASA_+3A_dirpath">dirPath</code></td>
<td>
<p>Path of the directory which contains all output files of SPINE-X. Each file belongs to a sequence.</p>
</td></tr>
<tr><td><code id="ASA_+3A_outformat">outFormat</code></td>
<td>
<p>It can take two values: 'mat' (which stands for matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="ASA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>It shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same lengths such that the number of columns is equal to the length of the sequences
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
ad&lt;-paste0(dir,"/asa.txt")


PredASAdir&lt;-system.file("testForder",package="ftrCOOL")
PredASAdir&lt;-paste0(PredASAdir,"/ASAdir/")
ASA(PredASAdir,outFormat="txt",outputFileDist=ad)

unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='ASDC'>Adaptive skip dipeptide composition (ASDC)</h2><span id='topic+ASDC'></span>

<h3>Description</h3>

<p>This descriptor sufficiently considers the correlation information present not only between adjacent residues but also between intervening residues.
This function calculates frequency of pair amino acids omitting gaps between them. Then this function normalizes each value through dividing each frequency by summition(frequencies).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ASDC(seqs, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ASDC_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="ASDC_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 400 (all posible amino acid pairs).
</p>


<h3>References</h3>

<p>Wei L, Zhou C, Chen H, Song J, Su R. ACPred-FL: a sequence-based predictor using effective feature representation to improve the prediction of anti-cancer peptides. Bioinformatics (2018).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-ASDC(seqs=filePrs)

</code></pre>

<hr>
<h2 id='ASDC_DNA'>Adaptive skip dinucleotide composition_DNA) (ASDC_DNA)</h2><span id='topic+ASDC_DNA'></span>

<h3>Description</h3>

<p>This descriptor sufficiently considers the correlation information present not only between adjacent nucleotides but also between intervening nucleotides
This function calculates frequency of pair nucleotides omitting gaps between them. Then this function normalizes each value through dividing each frequency by summition(frequencies).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ASDC_DNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ASDC_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="ASDC_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="ASDC_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="ASDC_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 16 (All posible nucleotide pairs).
</p>


<h3>References</h3>

<p>Wei L, Zhou C, Chen H, Song J, Su R. ACPred-FL: a sequence-based predictor using effective feature representation to improve the prediction of anti-cancer peptides. Bioinformatics (2018).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
fileLNC&lt;-fa.read(file=fileLNC,alphabet="dna")[1:5]
mat1&lt;-ASDC_DNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)

</code></pre>

<hr>
<h2 id='ASDC_RNA'>Adaptive skip di-ribonucleotide composition) (ASDC_RNA)</h2><span id='topic+ASDC_RNA'></span>

<h3>Description</h3>

<p>This descriptor sufficiently considers the correlation information present not only between adjacent ribo
ribonucleotides but also between intervening nucleotides
This function calculates frequency of pair ribonucleotides omitting gaps between them. Then this function normalizes each value through dividing each frequency by summition(frequencies).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ASDC_RNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ASDC_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="ASDC_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="ASDC_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="ASDC_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 16 (All posible ribonucleotide pairs).
</p>


<h3>References</h3>

<p>Wei L, Zhou C, Chen H, Song J, Su R. ACPred-FL: a sequence-based predictor using effective feature representation to improve the prediction of anti-cancer peptides. Bioinformatics (2018).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
fileLNC&lt;-fa.read(file=paste0(ptmSeqsADR,"/testSeq2RNA51.txt"),alphabet="rna")
mat1&lt;-ASDC_RNA(seqs=fileLNC)

</code></pre>

<hr>
<h2 id='AutoCorDiNUC_DNA'>Di Nucleotide Autocorrelation-Autocovariance (AutoCorDiNUC_DNA)</h2><span id='topic+AutoCorDiNUC_DNA'></span>

<h3>Description</h3>

<p>It creates the feature matrix for each function in autocorelation
(i.e., Moran, Greay, NormalizeMBorto) or autocovariance (i.e., AC, CC,ACC). The user can
select any combination of the functions too. In this case, the final matrix will contain
features of each selected function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AutoCorDiNUC_DNA(
  seqs,
  selectedIdx = c("Rise", "Roll", "Shift", "Slide", "Tilt", "Twist"),
  maxlag = 3,
  threshold = 1,
  type = c("Moran", "Geary", "NormalizeMBorto", "AC", "CC", "ACC"),
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AutoCorDiNUC_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="AutoCorDiNUC_DNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>function takes as input the physicochemical properties. Users select the properties by their ids
or indices in the DI_DNA file. This parameter could be a vector or a list of dinucleotide indices.
The default value of this parameter is a vector with (&quot;Rise&quot;, &quot;Roll&quot;, &quot;Shift&quot;, &quot;Slide&quot;, &quot;Tilt&quot;, &quot;Twist&quot;) ids.</p>
</td></tr>
<tr><td><code id="AutoCorDiNUC_DNA_+3A_maxlag">maxlag</code></td>
<td>
<p>This parameter shows the maximum gap between two dinucleotide pairs. The gaps change from 1 to maxlag (the maximum lag).</p>
</td></tr>
<tr><td><code id="AutoCorDiNUC_DNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 to 1]. In selectedIdx, indices with a correlation higher than the threshold will be deleted.The default value is 1.</p>
</td></tr>
<tr><td><code id="AutoCorDiNUC_DNA_+3A_type">type</code></td>
<td>
<p>could be 'Moran', 'Greay', 'NormalizeMBorto', 'AC', 'CC', or 'ACC'. Also, it could be any combination of them.</p>
</td></tr>
<tr><td><code id="AutoCorDiNUC_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For CC and AAC autocovriance functions, which consider the covariance of the
two physicochemical properties, we have provided users with the ability to categorize
their selected properties in a list.
The binary combination of each group will be taken into account.
Note: If all the features are in a group or selectedAAidx parameter is a vector,
the binary combination will be calculated for all the physicochemical properties.
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of columns in the matrix changes depending on the chosen autocorrelation or autocovariance types and nlag parameter.
The output is a matrix. The number of rows shows the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")

mat2&lt;-AutoCorDiNUC_DNA(seqs=fileLNC,selectedIdx=list(10,c(1,3),6:13,c(2:7))
,maxlag=15,type="CC")

</code></pre>

<hr>
<h2 id='AutoCorDiNUC_RNA'>Di riboNucleotide Autocorrelation-Autocovariance (AutoCorDiNUC_RNA)</h2><span id='topic+AutoCorDiNUC_RNA'></span>

<h3>Description</h3>

<p>It creates the feature matrix for each function in autocorelation
(i.e., Moran, Greay, NormalizeMBorto) or autocovariance (i.e., AC, CC,ACC). The user can
select any combination of the functions too. In this case, the final matrix will contain
features of each selected function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AutoCorDiNUC_RNA(
  seqs,
  selectedIdx = c("Rise (RNA)", "Roll (RNA)", "Shift (RNA)", "Slide (RNA)",
    "Tilt (RNA)", "Twist (RNA)"),
  maxlag = 3,
  threshold = 1,
  type = c("Moran", "Geary", "NormalizeMBorto", "AC", "CC", "ACC"),
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AutoCorDiNUC_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleic acid(RNA) sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a RNA sequence.</p>
</td></tr>
<tr><td><code id="AutoCorDiNUC_RNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>function takes as input the physicochemical properties. Users select the properties by their ids
or indices in the DI_RNA file. This parameter could be a vector or a list of di-ribonucleic acid indices.
The default value of this parameter is a vector with (&quot;Rise (RNA)&quot;, &quot;Roll (RNA)&quot;, &quot;Shift (RNA)&quot;, &quot;Slide (RNA)&quot;, &quot;Tilt (RNA)&quot;,&quot;Twist (RNA)&quot;) ids.</p>
</td></tr>
<tr><td><code id="AutoCorDiNUC_RNA_+3A_maxlag">maxlag</code></td>
<td>
<p>This parameter shows the maximum gap between two di-ribonucleotide pairs. The gaps change from 1 to maxlag (the maximum lag).</p>
</td></tr>
<tr><td><code id="AutoCorDiNUC_RNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 to 1]. In selectedIdx, indices with a correlation higher than the threshold will be deleted.The default value is 1.</p>
</td></tr>
<tr><td><code id="AutoCorDiNUC_RNA_+3A_type">type</code></td>
<td>
<p>could be 'Moran', 'Greay', 'NormalizeMBorto', 'AC', 'CC', or 'ACC'. Also, it could be any combination of them.</p>
</td></tr>
<tr><td><code id="AutoCorDiNUC_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For CC and AAC autocovriance functions, which consider the covariance of the
two physicochemical properties, we have provided users with the ability to categorize
their selected properties in a list.
The binary combination of each group will be taken into account.
Note: If all the features are in a group or selectedAAidx parameter is a vector,
the binary combination will be calculated for all the physicochemical properties.
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of columns in the matrix changes depending on the chosen autocorrelation or autocovariance types and nlag parameter.
The output is a matrix. The number of rows shows the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
fileLNC&lt;-fa.read(fileLNC,alphabet="rna")
fileLNC&lt;-fileLNC[1:20]
mat1&lt;-AutoCorDiNUC_RNA(seqs=fileLNC,maxlag=20,type=c("Moran"))

</code></pre>

<hr>
<h2 id='AutoCorTriNUC_DNA'>Tri Nucleotide Autocorrelation-Autocovariance (AutoCorTriNUC_DNA)</h2><span id='topic+AutoCorTriNUC_DNA'></span>

<h3>Description</h3>

<p>It creates the feature matrix for each function in autocorelation
(i.e., Moran, Greay, NormalizeMBorto) or autocovariance (i.e., AC, CC,ACC). The user can
select any combination of the functions too. In this case, the final matrix will contain
features of each selected function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AutoCorTriNUC_DNA(
  seqs,
  selectedNucIdx = c("Dnase I", "Bendability (DNAse)"),
  maxlag = 3,
  threshold = 1,
  type = c("Moran", "Geary", "NormalizeMBorto", "AC", "CC", "ACC"),
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AutoCorTriNUC_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="AutoCorTriNUC_DNA_+3A_selectednucidx">selectedNucIdx</code></td>
<td>
<p>function takes as input the physicochemical properties. Users select the properties by their ids
or indices in the TRI_DNA file. This parameter could be a vector or a list of trinucleotide indices.
The default value of this parameter is a vector with  (&quot;Dnase I&quot;, &quot;Bendability (DNAse)&quot;) ids.</p>
</td></tr>
<tr><td><code id="AutoCorTriNUC_DNA_+3A_maxlag">maxlag</code></td>
<td>
<p>This parameter shows the maximum gap between two tri-nucleotide pairs. The gaps change from 1 to maxlag (the maximum lag).</p>
</td></tr>
<tr><td><code id="AutoCorTriNUC_DNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 to 1]. In selectedNucIdx, indices with a correlation higher than the threshold will be deleted.The default value is 1.</p>
</td></tr>
<tr><td><code id="AutoCorTriNUC_DNA_+3A_type">type</code></td>
<td>
<p>could be 'Moran', 'Greay', 'NormalizeMBorto', 'AC', 'CC', or 'ACC'. Also, it could be any combination of them.</p>
</td></tr>
<tr><td><code id="AutoCorTriNUC_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For CC and AAC autocovriance functions, which consider the covariance of the
two physicochemical properties, we have provided users with the ability to categorize
their selected properties in a list.
The binary combination of each group will be taken into account.
Note: If all the features are in a group or selectedAAidx parameter is a vector,
the binary combination will be calculated for all the physicochemical properties.
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of columns in the matrix changes depending on the chosen autocorrelation or autocovariance types and nlag parameter.
The output is a matrix. The number of rows shows the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat1&lt;-AutoCorTriNUC_DNA(seqs=fileLNC,selectedNucIdx=c(1:7),maxlag=20,type=c("Moran","Geary"))

mat2&lt;-AutoCorTriNUC_DNA(seqs=fileLNC,selectedNucIdx=list(c(1,3),6:10,c(2:7)),
maxlag=15,type=c("AC","CC"))

</code></pre>

<hr>
<h2 id='binary_3bit_T1'>Binary - 3bit - Type1 (binary_3bit_T1)</h2><span id='topic+binary_3bit_T1'></span>

<h3>Description</h3>

<p>This group of functions(binary_3bit_T1-T7) categorizes amino acids in 3 groups based on the type.
Then represent group of amino acids by a three dimentional vector.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_3bit_T1(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binary_3bit_T1_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="binary_3bit_T1_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each amino acid is represented by a string containing 20 characters(0-1). For example, A = ALANIN = &quot;1000000...0&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 20 logical entries. For example, A = ALANIN = c(T,F,F,F,F,F,F,...F)
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 20 numerals. For example, A = ALANIN = c(1,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="binary_3bit_T1_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="binary_3bit_T1_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="binary_3bit_T1_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*3.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-binary_3bit_T1(seqs = ptmSeqsVect, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='binary_3bit_T2'>Binary - 3bit - Type2 (binary_3bit_T2)</h2><span id='topic+binary_3bit_T2'></span>

<h3>Description</h3>

<p>This group of functions(binary_3bit_T1-T7) categorizes amino acids in 3 groups based on the type.
Then represent group of amino acids by a three dimentional vector.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_3bit_T2(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binary_3bit_T2_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="binary_3bit_T2_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each amino acid is represented by a string containing 20 characters(0-1). For example, A = ALANIN = &quot;1000000...0&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 20 logical entries. For example, A = ALANIN = c(T,F,F,F,F,F,F,...F)
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 20 numerals. For example, A = ALANIN = c(1,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="binary_3bit_T2_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="binary_3bit_T2_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="binary_3bit_T2_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*3.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-binary_3bit_T2(seqs = ptmSeqsVect, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='binary_3bit_T3'>Binary - 3bit - Type3 (binary_3bit_T3)</h2><span id='topic+binary_3bit_T3'></span>

<h3>Description</h3>

<p>This group of functions(binary_3bit_T1-T7) categorizes amino acids in 3 groups based on the type.
Then represent group of amino acids by a three dimentional vector.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_3bit_T3(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binary_3bit_T3_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="binary_3bit_T3_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each amino acid is represented by a string containing 20 characters(0-1). For example, A = ALANIN = &quot;1000000...0&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 20 logical entries. For example, A = ALANIN = c(T,F,F,F,F,F,F,...F)
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 20 numerals. For example, A = ALANIN = c(1,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="binary_3bit_T3_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="binary_3bit_T3_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="binary_3bit_T3_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*3.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-binary_3bit_T3(seqs = ptmSeqsVect, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='binary_3bit_T4'>Binary - 3bit - Type4 (binary_3bit_T4)</h2><span id='topic+binary_3bit_T4'></span>

<h3>Description</h3>

<p>This group of functions(binary_3bit_T1-T7) categorizes amino acids in 3 groups based on the type.
Then represent group of amino acids by a three dimentional vector.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_3bit_T4(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binary_3bit_T4_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="binary_3bit_T4_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each amino acid is represented by a string containing 20 characters(0-1). For example, A = ALANIN = &quot;1000000...0&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 20 logical entries. For example, A = ALANIN = c(T,F,F,F,F,F,F,...F)
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 20 numerals. For example, A = ALANIN = c(1,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="binary_3bit_T4_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="binary_3bit_T4_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="binary_3bit_T4_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*3.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-binary_3bit_T4(seqs = ptmSeqsVect, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='binary_3bit_T5'>Binary - 3bit - Type5 (binary_3bit_T5)</h2><span id='topic+binary_3bit_T5'></span>

<h3>Description</h3>

<p>This group of functions(binary_3bit_T1-T7) categorizes amino acids in 3 groups based on the type.
Then represent group of amino acids by a three dimentional vector.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_3bit_T5(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binary_3bit_T5_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="binary_3bit_T5_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each amino acid is represented by a string containing 20 characters(0-1). For example, A = ALANIN = &quot;1000000...0&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 20 logical entries. For example, A = ALANIN = c(T,F,F,F,F,F,F,...F)
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 20 numerals. For example, A = ALANIN = c(1,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="binary_3bit_T5_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="binary_3bit_T5_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="binary_3bit_T5_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*3.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-binary_3bit_T5(seqs = ptmSeqsVect, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='binary_3bit_T6'>Binary - 3bit - Type6 (binary_3bit_T6)</h2><span id='topic+binary_3bit_T6'></span>

<h3>Description</h3>

<p>This group of functions(binary_3bit_T1-T7) categorizes amino acids in 3 groups based on the type.
Then represent group of amino acids by a three dimentional vector.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_3bit_T6(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binary_3bit_T6_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="binary_3bit_T6_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each amino acid is represented by a string containing 20 characters(0-1). For example, A = ALANIN = &quot;1000000...0&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 20 logical entries. For example, A = ALANIN = c(T,F,F,F,F,F,F,...F)
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 20 numerals. For example, A = ALANIN = c(1,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="binary_3bit_T6_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="binary_3bit_T6_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="binary_3bit_T6_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*3.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-binary_3bit_T6(seqs = ptmSeqsVect, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='binary_3bit_T7'>Binary - 3bit - Type7 (binary_3bit_T7)</h2><span id='topic+binary_3bit_T7'></span>

<h3>Description</h3>

<p>This group of functions(binary_3bit_T1-T7) categorizes amino acids in 3 groups based on the type.
Then represent group of amino acids by a three dimentional vector.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_3bit_T7(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binary_3bit_T7_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="binary_3bit_T7_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each amino acid is represented by a string containing 20 characters(0-1). For example, A = ALANIN = &quot;1000000...0&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 20 logical entries. For example, A = ALANIN = c(T,F,F,F,F,F,F,...F)
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 20 numerals. For example, A = ALANIN = c(1,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="binary_3bit_T7_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="binary_3bit_T7_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="binary_3bit_T7_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*3.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-binary_3bit_T7(seqs = ptmSeqsVect, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='binary_5bit_T1'>Binary - 5bit - Type1 (binary_5bit_T1)</h2><span id='topic+binary_5bit_T1'></span>

<h3>Description</h3>

<p>This function categorizes amino acids in 5 groups.
Then represent group of amino acids by a 5 dimentional vector i.e.e1, e2, e3, e4, e5.
e1=G, A, V, L, M, I, e2=F, Y, W, e3=K, R, H, e4=D, E, e5=S, T, C, P, N, Q. e1 is ecoded by 10000
e2 is encoded by 01000 and ... and e5 is encoded by 00001.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_5bit_T1(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binary_5bit_T1_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="binary_5bit_T1_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each amino acid is represented by a string containing 20 characters(0-1). For example, A = ALANIN = &quot;1000000...0&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 20 logical entries. For example, A = ALANIN = c(T,F,F,F,F,F,F,...F)
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 20 numerals. For example, A = ALANIN = c(1,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="binary_5bit_T1_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="binary_5bit_T1_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="binary_5bit_T1_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*5.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-binary_5bit_T1(seqs = ptmSeqsVect, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='binary_5bit_T2'>Binary - 5bit - Type2 (binary_5bit_T2)</h2><span id='topic+binary_5bit_T2'></span>

<h3>Description</h3>

<p>The idea behind this function is: We have 20 amino acids and we can show them with at least 5 bits.
A is encoded by (00011), C (00101), D (00110), E (00111), F(01001), G (01010), H (01011), I (01100), K (01101), L (01110), M (10001), N (10010), P (10011), Q (10100), R (10101), S (10110), T (11000), V (11001), W (11010), Y (11100).
This function transforms an amino acid to a binary format.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_5bit_T2(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binary_5bit_T2_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="binary_5bit_T2_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each amino acid is represented by a string containing 20 characters(0-1). For example, A = ALANIN = &quot;1000000...0&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 20 logical entries. For example, A = ALANIN = c(T,F,F,F,F,F,F,...F)
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 20 numerals. For example, A = ALANIN = c(1,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="binary_5bit_T2_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="binary_5bit_T2_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="binary_5bit_T2_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*5.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-binary_5bit_T2(seqs = ptmSeqsVect, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='binary_6bit'>Binary - 6bit (binary_6bit)</h2><span id='topic+binary_6bit'></span>

<h3>Description</h3>

<p>This function categorizes amino acids in 6 groups.
Then represent group of amino acids by a 6 dimentional vector i.e.e1, e2, e3, e4, e5, e6.
e1=H, R, K, e2=D, E, N, D, e3=C, e4=S, T, P, A, G, e5=M, I, L, V, e6=F, Y, W.
e1 is ecoded by 100000 e2 is encoded by 010000 and ... and e6 is encoded by 000001.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_6bit(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binary_6bit_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="binary_6bit_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each amino acid is represented by a string containing 20 characters(0-1). For example, A = ALANIN = &quot;1000000...0&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 20 logical entries. For example, A = ALANIN = c(T,F,F,F,F,F,F,...F)
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 20 numerals. For example, A = ALANIN = c(1,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="binary_6bit_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="binary_6bit_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="binary_6bit_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*6.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-binary_6bit(seqs = ptmSeqsVect, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='BLOSUM62'>Blosum62 (BLOSUM62)</h2><span id='topic+BLOSUM62'></span>

<h3>Description</h3>

<p>This function creates a 20-dimentional numeric vector for each amino acid of a sequence. Each entry of the vector contains
the similarity score of the amino acid with other amino acids including itself. The score is extracted from the Blosum62 matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BLOSUM62(seqs, label = c(), outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BLOSUM62_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="BLOSUM62_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="BLOSUM62_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="BLOSUM62_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length)*20
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
filePr&lt;-system.file("extdata/protein.fasta",package="ftrCOOL")
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

ad&lt;-paste0(dir,"/blosum62.txt")
vect&lt;-BLOSUM62(seqs = filePr,outFormat="mat")
BLOSUM62(seqs = filePrs,outFormat="txt",outputFileDist=ad)

unlink("dir", recursive = TRUE)


</code></pre>

<hr>
<h2 id='CkSAApair'>Composition of k-Spaced Amino Acids pairs (CkSAApair)</h2><span id='topic+CkSAApair'></span>

<h3>Description</h3>

<p>This function calculates the composition of k-spaced amino acid pairs. In other words, it
computes the frequency of all amino acid pairs with k spaces.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CkSAApair(seqs, rng = 3, upto = FALSE, normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CkSAApair_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="CkSAApair_+3A_rng">rng</code></td>
<td>
<p>This parameter can be a number or a vector. Each element of the vector shows the number of spaces between amino acid pairs.
For each k in the rng vector, a new vector (whose size is 400) is created which contains the frequency of pairs with k gaps.</p>
</td></tr>
<tr><td><code id="CkSAApair_+3A_upto">upto</code></td>
<td>
<p>It is a logical parameter. The default value is FALSE. If rng is a number and upto is set to TRUE, rng is converted
to a vector with values from [0 to rng].</p>
</td></tr>
<tr><td><code id="CkSAApair_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="CkSAApair_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 400*(length of rng vector).
</p>


<h3>Note</h3>

<p>'upto' is enabled only when rng is a number and not a vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-CkSAApair(seqs=filePrs,rng=2,upto=TRUE,normalized=TRUE)

mat2&lt;-CkSAApair(seqs=filePrs,rng=c(1,3,5))

</code></pre>

<hr>
<h2 id='CkSGAApair'>Composition of k-Spaced Grouped Amino Acids pairs (CkSGAApair)</h2><span id='topic+CkSGAApair'></span>

<h3>Description</h3>

<p>In this function, amino acids are first grouped into a category which is defined by the user.
Later, the composition of the k-spaced grouped amino acids is computed.
Please note that this function differs from <a href="#topic+CkSAApair">CkSAApair</a> which works on individual amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CkSGAApair(
  seqs,
  rng = 3,
  upto = FALSE,
  normalized = TRUE,
  Grp = "locFus",
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CkSGAApair_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="CkSGAApair_+3A_rng">rng</code></td>
<td>
<p>This parameter can be a number or a vector. Each element of the vector shows the number of spaces between amino acid pairs.
For each k in the rng vector, a new vector (whose size is (number of categorizes)^2) is created which contains the frequency of pairs with k gaps.</p>
</td></tr>
<tr><td><code id="CkSGAApair_+3A_upto">upto</code></td>
<td>
<p>It is a logical parameter. The default value is FALSE. If rng is a number and upto is set to TRUE, rng is converted
to a vector with values from [1 to rng].</p>
</td></tr>
<tr><td><code id="CkSGAApair_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="CkSGAApair_+3A_grp">Grp</code></td>
<td>
<p>is a list of vectors containig amino acids. Each vector represents a category. Users can define a customized amino acid grouping, provided that the sum of all amino acids is 20 and there is no repeated amino acid in the groups.
Also, users can choose 'cTriad'(conjointTriad), 'locFus', or 'aromatic'. Each option provides specific information about the type of an amino acid grouping.</p>
</td></tr>
<tr><td><code id="CkSGAApair_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Column names in the feature matrix follow G(?ss?). For example, G(1ss2) means Group1**Group2, where
'*' is a wild character.
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. Row length is equal to the number of sequences and
the number of columns is ((number of categorizes)^2)*(length of rng vector).
</p>


<h3>Note</h3>

<p>'upto' is enabled only when rng is a number and not a vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-CkSGAApair(seqs=filePrs,rng=2,upto=TRUE,Grp="aromatic")

mat2&lt;-CkSGAApair(seqs=filePrs,rng=c(1,3,5),upto=FALSE,Grp=
list(Grp1=c("G","A","V","L","M","I","F","Y","W"),Grp2=c("K","R","H","D","E")
,Grp3=c("S","T","C","P","N","Q")))

</code></pre>

<hr>
<h2 id='CkSNUCpair_DNA'>Composition of k-Spaced Nucleotides Pairs (CkSNUCpair_DNA)</h2><span id='topic+CkSNUCpair_DNA'></span>

<h3>Description</h3>

<p>This function calculates the composition of k-spaced nucleotide pairs. In other words, it
computes the frequency of all nucleotide pairs with k spaces.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CkSNUCpair_DNA(
  seqs,
  rng = 3,
  upto = FALSE,
  ORF = FALSE,
  reverseORF = TRUE,
  normalized = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CkSNUCpair_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="CkSNUCpair_DNA_+3A_rng">rng</code></td>
<td>
<p>This parameter can be a number or a vector. Each element of the vector shows the number of spaces between nucleotide pairs.
For each k in the rng vector, a new vector (whose size is 16) is created which contains the frequency of pairs with k gaps.</p>
</td></tr>
<tr><td><code id="CkSNUCpair_DNA_+3A_upto">upto</code></td>
<td>
<p>It is a logical parameter. The default value is FALSE. If rng is a number and upto is set to TRUE, rng is converted
to a vector with values from [0 to rng].</p>
</td></tr>
<tr><td><code id="CkSNUCpair_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="CkSNUCpair_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="CkSNUCpair_DNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="CkSNUCpair_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 16*(length of rng vector).
</p>


<h3>Note</h3>

<p>'upto' is enabled only when rng is a number and not a vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat1&lt;-CkSNUCpair_DNA(seqs=fileLNC,rng=2,upto=TRUE,ORF=TRUE,reverseORF=FALSE)
mat2&lt;-CkSNUCpair_DNA(seqs=fileLNC,rng=c(1,3,5))

</code></pre>

<hr>
<h2 id='CkSNUCpair_RNA'>Composition of k-Spaced riboNucleotides Pairs (CkSNUCpair_RNA)</h2><span id='topic+CkSNUCpair_RNA'></span>

<h3>Description</h3>

<p>This function calculates the composition of k-spaced ribonucleotide pairs. In other words, it
computes the frequency of all ribonucleotide pairs with k spaces.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CkSNUCpair_RNA(
  seqs,
  rng = 3,
  upto = FALSE,
  ORF = FALSE,
  reverseORF = TRUE,
  normalized = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CkSNUCpair_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="CkSNUCpair_RNA_+3A_rng">rng</code></td>
<td>
<p>This parameter can be a number or a vector. Each element of the vector shows the number of spaces between ribonucleotide pairs.
For each k in the rng vector, a new vector (whose size is 16) is created which contains the frequency of pairs with k gaps.</p>
</td></tr>
<tr><td><code id="CkSNUCpair_RNA_+3A_upto">upto</code></td>
<td>
<p>It is a logical parameter. The default value is FALSE. If rng is a number and upto is set to TRUE, rng is converted
to a vector with values from [0 to rng].</p>
</td></tr>
<tr><td><code id="CkSNUCpair_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="CkSNUCpair_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="CkSNUCpair_RNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="CkSNUCpair_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 16*(length of rng vector).
</p>


<h3>Note</h3>

<p>'upto' is enabled only when rng is a number and not a vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat1&lt;-CkSNUCpair_RNA(seqs=fileLNC,rng=2,upto=TRUE,ORF=TRUE,reverseORF=FALSE)
mat2&lt;-CkSNUCpair_RNA(seqs=fileLNC,rng=c(1,3,5))

</code></pre>

<hr>
<h2 id='codonAdaptionIndex'>Codon Adaption Index (codonAdaptionIndex)</h2><span id='topic+codonAdaptionIndex'></span>

<h3>Description</h3>

<p>This function calculates the codon adaption index for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>codonAdaptionIndex(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="codonAdaptionIndex_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="codonAdaptionIndex_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="codonAdaptionIndex_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="codonAdaptionIndex_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature vector. The length of the vector is equal to the number of sequences.
Each entry in the vector contains the value of the codon adaption index.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-codonAdaptionIndex(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)

</code></pre>

<hr>
<h2 id='CodonFraction'>Codon Fraction (CodonFraction)</h2><span id='topic+CodonFraction'></span>

<h3>Description</h3>

<p>This function calculates the codon fraction for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CodonFraction(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CodonFraction_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="CodonFraction_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="CodonFraction_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="CodonFraction_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A feature matrix such that the number of columns is 4^3 and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-CodonFraction(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)

</code></pre>

<hr>
<h2 id='CodonUsage_DNA'>Codon Usage in DNA (CodonUsage_DNA)</h2><span id='topic+CodonUsage_DNA'></span>

<h3>Description</h3>

<p>This function calculates the codon usage for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CodonUsage_DNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CodonUsage_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="CodonUsage_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="CodonUsage_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="CodonUsage_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A feature matrix such that the number of columns is 4^3 and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-CodonUsage_DNA(fileLNC,ORF=TRUE,reverseORF=FALSE)

</code></pre>

<hr>
<h2 id='CodonUsage_RNA'>Codon Usage in RNA (CodonUsage_RNA)</h2><span id='topic+CodonUsage_RNA'></span>

<h3>Description</h3>

<p>This function calculates the codon usage for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CodonUsage_RNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CodonUsage_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="CodonUsage_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="CodonUsage_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="CodonUsage_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A feature matrix such that the number of columns is 4^3 and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-CodonUsage_RNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)

</code></pre>

<hr>
<h2 id='conjointTriad'>Conjoint Triad (conjointTriad)</h2><span id='topic+conjointTriad'></span>

<h3>Description</h3>

<p>This function calculates the grouped tripeptide composition with the conjoint triad grouping type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>conjointTriad(seqs, normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="conjointTriad_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="conjointTriad_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="conjointTriad_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows equals to the number of sequences and
the number of columns is 7^3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-conjointTriad(seqs=filePrs)

</code></pre>

<hr>
<h2 id='conjointTriadKS'>k-Spaced Conjoint Triad (conjointTriadKS)</h2><span id='topic+conjointTriadKS'></span>

<h3>Description</h3>

<p>This function calculates the grouped tripeptide composition with conjoint
triad grouping type. For each k, it creates a 7^3 feature vector.
K is the space between the first and the second amino acids
and the second and the third amino acids of the tripeptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>conjointTriadKS(seqs, rng = 3, upto = FALSE, normalized = FALSE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="conjointTriadKS_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="conjointTriadKS_+3A_rng">rng</code></td>
<td>
<p>This parameter can be a number or a vector. Each element of the vector shows the number of spaces between the first and the second amino acids
and the second and the third amino acids of the tripeptide.
For each k in the rng vector, a new vector (whose size is 7^3) is created which contains the frequency of tri-amino acid with k gaps.</p>
</td></tr>
<tr><td><code id="conjointTriadKS_+3A_upto">upto</code></td>
<td>
<p>It is a logical parameter. The default value is FALSE. If rng is a number and upto is set to TRUE, rng is converted
to a vector with values from 0 to rng.</p>
</td></tr>
<tr><td><code id="conjointTriadKS_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="conjointTriadKS_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A tripeptide with k spaces looks like AA1(ss..s)AA2(ss..s)AA3. AA stands for amino acids and s means space.
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (7^3)*(length rng vector).
</p>


<h3>Note</h3>

<p>'upto' is enabled only when rng is a number and not a vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-conjointTriadKS(filePrs,rng=2,upto=TRUE,normalized=TRUE)

mat2&lt;-conjointTriadKS(filePrs,rng=c(1,3,5))

</code></pre>

<hr>
<h2 id='CTD'>Composition_Transition_Distribution (CTD)</h2><span id='topic+CTD'></span>

<h3>Description</h3>

<p>This function calculates the composition, transition, and distribution for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTD(seqs, normalized = FALSE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTD_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="CTD_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="CTD_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output is a combination of three different matrices: Composition, Transition, and Distribution.
You can obtain any of the three matrices by executing the corresponding function, i.e., <a href="#topic+CTDC">CTDC</a>, <a href="#topic+CTDT">CTDT</a>, and <a href="#topic+CTDD">CTDD</a>.
</p>


<h3>References</h3>

<p>Dubchak, Inna, et al. &quot;Prediction of protein folding class using global description of amino acid sequence.&quot; Proceedings of the National Academy of Sciences 92.19 (1995): 8700-8704.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
CTDtotal&lt;-CTD(seqs=filePrs,normalized=FALSE)

</code></pre>

<hr>
<h2 id='CTDC'>CTD Composition (CTDC)</h2><span id='topic+CTDC'></span>

<h3>Description</h3>

<p>This function computes the composition part of <a href="#topic+CTD">CTD</a>. Thirteen properties are defined in this function. Each
property categorizes the amino acids of the sequences into three groups. The grouped amino acid composition is calculated for each property.
For more information, please check the references.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTDC(seqs, normalized = FALSE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTDC_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="CTDC_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="CTDC_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix.
The number of rows is equal to the number of sequences and the number of columns is 3*7, where three is the number of groups and thirteen is the number of properties.
</p>


<h3>References</h3>

<p>Dubchak, Inna, et al. &quot;Prediction of protein folding class using global description of amino acid sequence.&quot; Proceedings of the National Academy of Sciences 92.19 (1995): 8700-8704.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
CTD_C&lt;-CTDC(seqs=filePrs,normalized=FALSE,label=c())

</code></pre>

<hr>
<h2 id='CTDD'>CTD Distribution (CTDD)</h2><span id='topic+CTDD'></span>

<h3>Description</h3>

<p>This function computes the distribution part of <a href="#topic+CTD">CTD</a>. It calculates fifteen values for each property.
For more information, please check the references.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTDD(seqs, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTDD_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="CTDD_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and the number of columns is 15*7.
</p>


<h3>References</h3>

<p>Dubchak, Inna, et al. &quot;Prediction of protein folding class using global description of amino acid sequence.&quot; Proceedings of the National Academy of Sciences 92.19 (1995): 8700-8704.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
CTD_D&lt;-CTDD(seqs=filePrs)

</code></pre>

<hr>
<h2 id='CTDT'>CTD Transition (CTDT)</h2><span id='topic+CTDT'></span>

<h3>Description</h3>

<p>This function computes the transition part of <a href="#topic+CTD">CTD</a>. Thirteen properties are defined in this function. Each
property categorizes the amino acids of a sequence into three groups. For each property, the grouped amino acid transition (i.e., transitions 1-2, 1-3, and 2-3)
is calculated. For more information, please check the references.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTDT(seqs, normalized = FALSE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTDT_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="CTDT_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="CTDT_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix.
The number of rows is equal to the number of sequences and the number of columns is 3*7,
where three is the number of transition types (i.e., 1-2, 1-3, and 2-3) and thirteen is the number of properties.
</p>


<h3>References</h3>

<p>Dubchak, Inna, et al. &quot;Prediction of protein folding class using global description of amino acid sequence.&quot; Proceedings of the National Academy of Sciences 92.19 (1995): 8700-8704.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
CTD_T&lt;-CTDT(seqs=filePrs,normalized=FALSE)

</code></pre>

<hr>
<h2 id='DDE'>Dipeptide Deviation from Expected Mean value (DDE)</h2><span id='topic+DDE'></span>

<h3>Description</h3>

<p>This function computes the dipeptide deviation from the expected mean value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DDE(seqs, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DDE_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="DDE_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A feature matrix with 20^2=400 number of columns. The number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-DDE(seqs=filePrs)

</code></pre>

<hr>
<h2 id='DiNUC2Binary_DNA'>Dinucleotide To Binary DNA (DiNUC2Binary_DNA)</h2><span id='topic+DiNUC2Binary_DNA'></span>

<h3>Description</h3>

<p>This function transforms a dinucleotide to a binary number with four bits which is enough to represent all the possible types of dinucleotides.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DiNUC2Binary_DNA(
  seqs,
  binaryType = "numBin",
  outFormat = "mat",
  outputFileDist = "",
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DiNUC2Binary_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="DiNUC2Binary_DNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin' (String binary): each dinucleotide is represented by a string containing 4 characters(0-1). For example, AA = &quot;0000&quot;   AC=&quot;0001&quot;   ...  TT=&quot;1111&quot;
'logicBin' (logical value): Each dinucleotide is represented by a vector containing 4 logical entries. For example, AA = c(F,F,F,F)   AC=c(F,F,F,T)  ...    TT=c(T,T,T,T)
'numBin' (numeric bin): Each dinucleotide is represented by a numeric (i.e., integer) vector containing 4 numeric entries. For example, AA = c(0,0,0,0)   AC = c(0,0,0,1)  ...  TT = c(1,1,1,1)</p>
</td></tr>
<tr><td><code id="DiNUC2Binary_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="DiNUC2Binary_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
<tr><td><code id="DiNUC2Binary_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the (length of the sequences-1).
Otherwise, it is equal to (length of the sequences-1)*4.
If outFormat is 'txt', all binary values will be written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-DiNUC2Binary_DNA(seqs = LNC50Nuc, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='DiNUC2Binary_RNA'>Di riboNucleotide To Binary RNA (DiNUC2Binary_RNA)</h2><span id='topic+DiNUC2Binary_RNA'></span>

<h3>Description</h3>

<p>This function transforms a di-ribonucleotide to a binary number with four bits which is enough to represent all the possible types of di-ribonucleotides.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DiNUC2Binary_RNA(
  seqs,
  binaryType = "numBin",
  outFormat = "mat",
  outputFileDist = "",
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DiNUC2Binary_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="DiNUC2Binary_RNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin' (String binary): each di-ribonucleotide is represented by a string containing 4 characters(0-1). For example, AA = &quot;0000&quot;   AC=&quot;0001&quot;   ...  TT=&quot;1111&quot;
'logicBin' (logical value): Each di-ribonucleotide is represented by a vector containing 4 logical entries. For example, AA = c(F,F,F,F)   AC=c(F,F,F,T)  ...    TT=c(T,T,T,T)
'numBin' (numeric bin): Each di-ribonucleotide is represented by a numeric (i.e., integer) vector containing 4 numeric entries. For example, AA = c(0,0,0,0)   AC = c(0,0,0,1)  ...  TT = c(1,1,1,1)</p>
</td></tr>
<tr><td><code id="DiNUC2Binary_RNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="DiNUC2Binary_RNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
<tr><td><code id="DiNUC2Binary_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is (length of the sequences-1).
Otherwise, it is equal to (length of the sequences-1)*4.
If outFormat is 'txt', all binary values will be written to a 'txt' file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-DiNUC2Binary_RNA(seqs = fileLNC, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='DiNUCindex_DNA'>Di Nucleotide Index (DiNUCindex_DNA)</h2><span id='topic+DiNUCindex_DNA'></span>

<h3>Description</h3>

<p>This function replaces dinucleotides in a sequence with their physicochemical properties in the dinucleotide index file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DiNUCindex_DNA(
  seqs,
  selectedIdx = c("Rise", "Roll", "Shift", "Slide", "Tilt", "Twist"),
  threshold = 1,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DiNUCindex_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="DiNUCindex_DNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>DiNUCindex_DNA function works based on physicochemical properties. Users, select the properties by their ids
or indexes in DI_DNA index file.
The default value of this parameter is a vector with (&quot;Rise&quot;, &quot;Roll&quot;, &quot;Shift&quot;, &quot;Slide&quot;, &quot;Tilt&quot;, &quot;Twist&quot;) entries.</p>
</td></tr>
<tr><td><code id="DiNUCindex_DNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In selectedIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="DiNUCindex_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="DiNUCindex_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="DiNUCindex_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are 148 physicochemical indexes in the dinucleotide database.
</p>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length-1)*(number of selected di-nucleotide indexes)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana1.fa",package="ftrCOOL")
vect&lt;-DiNUCindex_DNA(seqs = fileLNC,outFormat="mat")
</code></pre>

<hr>
<h2 id='DiNUCindex_RNA'>Di riboNucleotide Index (DiNUCindex_RNA)</h2><span id='topic+DiNUCindex_RNA'></span>

<h3>Description</h3>

<p>This function replaces di-ribonucleotides in a sequence with their physicochemical properties in the di-ribonucleotide index file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DiNUCindex_RNA(
  seqs,
  selectedIdx = c("Rise (RNA)", "Roll (RNA)", "Shift (RNA)", "Slide (RNA)",
    "Tilt (RNA)", "Twist (RNA)"),
  threshold = 1,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DiNUCindex_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="DiNUCindex_RNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>DiNucIndex function works based on physicochemical properties. Users, select the properties by their ids
or indexes in DI_RNA file.
The default value of this parameter is a vector with (&quot;Rise (RNA)&quot;, &quot;Roll (RNA)&quot;, &quot;Shift (RNA)&quot;, &quot;Slide (RNA)&quot;, &quot;Tilt (RNA)&quot;,&quot;Twist (RNA)&quot;) entries.</p>
</td></tr>
<tr><td><code id="DiNUCindex_RNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In selectedIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="DiNUCindex_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="DiNUCindex_RNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="DiNUCindex_RNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are 22 physicochemical indexes in the di-ribonucleotide database.
</p>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length-1)*(number of selected di-ribonucleotide indexes)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
vect&lt;-DiNUCindex_RNA(seqs = fileLNC,outFormat="mat")
</code></pre>

<hr>
<h2 id='DisorderB'>disorder Binary (DisorderB)</h2><span id='topic+DisorderB'></span>

<h3>Description</h3>

<p>This function extracts the ordered and disordered amino acids in protein or peptide sequences. The input to the function is provided by VSL2 software. Also, the function
converts order amino acids to '10' and disorder amino acids to '01'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DisorderB(
  dirPath,
  binaryType = "numBin",
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DisorderB_+3A_dirpath">dirPath</code></td>
<td>
<p>Path of the directory which contains all output files of VSL2. Each file belongs to a sequence.</p>
</td></tr>
<tr><td><code id="DisorderB_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin' (String binary): each amino acid is represented by a string containing 2 characters(0-1). order = &quot;10&quot;   disorder=&quot;01&quot;.
'logicBin' (logical value): Each amino acid is represented by a vector containing 2 logical entries.  order = c(TRUE,FALSE)   disorder=c(FALSE,TRUE).
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 2 numeric entries.   order = c(1,0)   disorder=c(0,1).</p>
</td></tr>
<tr><td><code id="DisorderB_+3A_outformat">outFormat</code></td>
<td>
<p>It can take two values: 'mat' (which stands for matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="DisorderB_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>It shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*2.
If outFormat is 'txt', all binary values will be written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()

PredDisdir&lt;-system.file("testForder",package="ftrCOOL")
PredDisdir&lt;-paste0(PredDisdir,"/Disdir/")
ad1&lt;-paste0(dir,"/disorderB.txt")

DisorderB(PredDisdir,binaryType="numBin",outFormat="txt",outputFileDist=ad1)

unlink("dir", recursive = TRUE)

</code></pre>

<hr>
<h2 id='DisorderC'>disorder Content (DisorderC)</h2><span id='topic+DisorderC'></span>

<h3>Description</h3>

<p>This function extracts ordered and disordered amino acids in protein or peptide sequences. The input to the function is provided by VSL2 software.
Also, the function returns number of order and disorder amino acids in the sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DisorderC(dirPath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DisorderC_+3A_dirpath">dirPath</code></td>
<td>
<p>Path of the directory which contains all output files of VSL2. Each file belongs to a sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a feature matrix with 2 columns. The number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
PredDisdir&lt;-system.file("testForder",package="ftrCOOL")
PredDisdir&lt;-paste0(PredDisdir,"/Disdir/")

mat&lt;-DisorderC(PredDisdir)

</code></pre>

<hr>
<h2 id='DisorderS'>disorder Simple (DisorderS)</h2><span id='topic+DisorderS'></span>

<h3>Description</h3>

<p>This function extracts ordered and disordered amino acids in protein or peptide sequences.
The input to the function is provided by VSL2 software. The function
represent order amino acids by 'O' and disorder amino acids by 'D'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DisorderS(dirPath, outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DisorderS_+3A_dirpath">dirPath</code></td>
<td>
<p>Path of the directory which contains all output files of VSL2. Each file belongs to a sequence.</p>
</td></tr>
<tr><td><code id="DisorderS_+3A_outformat">outFormat</code></td>
<td>
<p>It can take two values: 'mat' (which stands for matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="DisorderS_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>It shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same lengths such that the number of columns is equal to the length of the sequences
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()

PredDisdir&lt;-system.file("testForder",package="ftrCOOL")
PredDisdir&lt;-paste0(PredDisdir,"/Disdir/")
ad1&lt;-paste0(dir,"/disorderS.txt")

DisorderS(PredDisdir, outFormat="txt",outputFileDist=ad1)

unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='DistancePair'>PseAAC of distance-pairs and reduced alphabet (DistancePair)</h2><span id='topic+DistancePair'></span>

<h3>Description</h3>

<p>In this function, first amino acids are grouped into a category which is one of 'cp13', 'cp14', 'cp19', 'cp20'. Users choose one of these terms to categorize amino acids.
Then DistancePair function computes frequencies of all grouped residues and also all grouped-paired residues with [0,rng] distance. 'rng'
is a parameter which already was set by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DistancePair(seqs, rng = 3, normalized = TRUE, Grp = "cp14", label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DistancePair_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="DistancePair_+3A_rng">rng</code></td>
<td>
<p>This parameter is a number. It shows maximum number of spaces between amino acid pairs.
For each k in the rng vector, a new vector (whose size is (number of categorizes)^2) is created which contains the frequency of pairs with k gaps.</p>
</td></tr>
<tr><td><code id="DistancePair_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="DistancePair_+3A_grp">Grp</code></td>
<td>
<p>for this parameter users can choose between these items: 'cp13', 'cp14', 'cp19', or 'cp20'.</p>
</td></tr>
<tr><td><code id="DistancePair_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. Row length is equal to the number of sequences and
the number of columns is (number of categorizes)+((number of categorizes)^2)*(rng+1).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-DistancePair(seqs=filePrs,rng=2,Grp="cp14")
</code></pre>

<hr>
<h2 id='DPCP_DNA'>Dinucleotide physicochemical properties (DPCP_DNA)</h2><span id='topic+DPCP_DNA'></span>

<h3>Description</h3>

<p>This function replaces dinucleotides in a sequence with their physicochemical properties which is multiplied by normalized frequency of that di-nucleotide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DPCP_DNA(
  seqs,
  selectedIdx = c("Rise", "Roll", "Shift", "Slide", "Tilt", "Twist"),
  threshold = 1,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DPCP_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="DPCP_DNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>DPCP_DNA function works based on physicochemical properties. Users, select the properties by their ids
or indexes in DI_DNA index file.
The default value of this parameter is a vector with (&quot;Rise&quot;, &quot;Roll&quot;, &quot;Shift&quot;, &quot;Slide&quot;, &quot;Tilt&quot;, &quot;Twist&quot;) entries.</p>
</td></tr>
<tr><td><code id="DPCP_DNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In selectedIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="DPCP_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="DPCP_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="DPCP_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are 148 physicochemical indexes in the dinucleotide database.
</p>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length-1)*(number of selected di-nucleotide indexes)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana1.fa",package="ftrCOOL")
vect&lt;-DPCP_DNA(seqs = fileLNC,outFormat="mat")
</code></pre>

<hr>
<h2 id='DPCP_RNA'>Di-ribonucleotide physicochemical properties (DPCP_RNA)</h2><span id='topic+DPCP_RNA'></span>

<h3>Description</h3>

<p>This function replaces di-ribonucleotides in a sequence with their physicochemical properties which is multiplied by normalized frequency of that di-ribonucleotide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DPCP_RNA(
  seqs,
  selectedIdx = c("Rise (RNA)", "Roll (RNA)", "Shift (RNA)", "Slide (RNA)",
    "Tilt (RNA)", "Twist (RNA)"),
  threshold = 1,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DPCP_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="DPCP_RNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>DiNucIndex function works based on physicochemical properties. Users, select the properties by their ids
or indexes in DI_RNA file.
The default value of this parameter is a vector with (&quot;Rise (RNA)&quot;, &quot;Roll (RNA)&quot;, &quot;Shift (RNA)&quot;, &quot;Slide (RNA)&quot;, &quot;Tilt (RNA)&quot;,&quot;Twist (RNA)&quot;) entries.</p>
</td></tr>
<tr><td><code id="DPCP_RNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In selectedAAidx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="DPCP_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="DPCP_RNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="DPCP_RNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are 22 physicochemical indexes in the di-ribonucleotide database.
</p>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length-1)*(number of selected di-ribonucleotide indexes)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
vect&lt;-DPCP_RNA(seqs = fileLNC,outFormat="mat")
</code></pre>

<hr>
<h2 id='EAAComposition'>Enhanced Amino Acid Composition (EAAComposition)</h2><span id='topic+EAAComposition'></span>

<h3>Description</h3>

<p>This function slides a window over the input sequence(s).
Also, it computes the composition of amino acids that appears within the limits of the window.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EAAComposition(
  seqs,
  winSize = 50,
  overLap = TRUE,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EAAComposition_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="EAAComposition_+3A_winsize">winSize</code></td>
<td>
<p>is a number which shows the size of the window.</p>
</td></tr>
<tr><td><code id="EAAComposition_+3A_overlap">overLap</code></td>
<td>
<p>This parameter shows how the window
moves over the sequence. If overlap is set to FALSE, the window slides over the sequence in such a way that every time the window moves, it covers a unique portion of the sequence.
Otherwise, portions of the sequence which appear within the window limits have &quot;winSize-1&quot; amino acids in common.</p>
</td></tr>
<tr><td><code id="EAAComposition_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="EAAComposition_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="EAAComposition_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Column names in the output matrix are Wi(aa), where aa shows an amino acid type (&quot;A&quot;, &quot;C&quot;, &quot;D&quot;,..., &quot;Y&quot;) and
i indicates the number of times that the window has moved over the sequence(s).
</p>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (20 * number of partitions displayed by the window)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>
<p>When overlap is FALSE, the last partition represented by the window may have a different length with other parts.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. &quot;iFeature: a python package and web server for features extraction and selection from protein and peptide sequences.&quot; Bioinformatics 34.14 (2018): 2499-2502.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-EAAComposition(seqs = ptmSeqsVect,winSize=50, overLap=FALSE,outFormat='mat')

ad&lt;-paste0(dir,"/EaaCompos.txt")
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
EAAComposition(seqs = filePrs,winSize=50, overLap=FALSE,outFormat="txt"
,outputFileDist=ad)

unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='EffectiveNumberCodon'>Effective Number of Codon (EffectiveNumberCodon)</h2><span id='topic+EffectiveNumberCodon'></span>

<h3>Description</h3>

<p>This function calculates the effective number of codon for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EffectiveNumberCodon(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EffectiveNumberCodon_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="EffectiveNumberCodon_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="EffectiveNumberCodon_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="EffectiveNumberCodon_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature vector. The length of the vector is equal to the number of sequences.
Each entry in the vector contains the effective number of codon.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
vect&lt;-EffectiveNumberCodon(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)

</code></pre>

<hr>
<h2 id='EGAAComposition'>Enhanced Grouped Amino Acid Composition (EGAAComposition)</h2><span id='topic+EGAAComposition'></span>

<h3>Description</h3>

<p>In this function, amino acids are first grouped into user-defined categories. Then, enhanced grouped amino acid composition is computed.
For details about the enhanced feature, please refer to function <a href="#topic+EAAComposition">EAAComposition</a>.
Please note that this function differs from function <a href="#topic+EAAComposition">EAAComposition</a> which works on individual amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EGAAComposition(
  seqs,
  winSize = 50,
  overLap = TRUE,
  Grp = "locFus",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EGAAComposition_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="EGAAComposition_+3A_winsize">winSize</code></td>
<td>
<p>shows the size of sliding window. It is a numeric value.</p>
</td></tr>
<tr><td><code id="EGAAComposition_+3A_overlap">overLap</code></td>
<td>
<p>This parameter shows how the window
moves on the sequence. If the overlap is set to TRUE, the next window would have distance 1 with
the previous window. Otherwise, the next window will start from the next amino acid after the previous window.
There is no overlap between the next and previous windows.</p>
</td></tr>
<tr><td><code id="EGAAComposition_+3A_grp">Grp</code></td>
<td>
<p>is a list of vectors containig amino acids. Each vector represents a category. Users can define a customized amino acid grouping, provided that the sum of all amino acids is 20 and there is no repeated amino acid in the groups.
Also, users can choose 'cTriad'(conjointTriad), 'locFus', or 'aromatic'. Each option provides specific information about the type of an amino acid grouping.</p>
</td></tr>
<tr><td><code id="EGAAComposition_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="EGAAComposition_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="EGAAComposition_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is ((number of categorizes) * (number of windows))
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat1&lt;-EGAAComposition(seqs = ptmSeqsVect,winSize=20,overLap=FALSE,Grp="locFus")

mat2&lt;-EGAAComposition(seqs = ptmSeqsVect,winSize=30,overLap=FALSE,Grp=
list(Grp1=c("G","A","V","L","M","I","F","Y","W"),Grp2=c("K","R","H","D","E")
,Grp3=c("S","T","C","P","N","Q")),outFormat="mat")

ad&lt;-paste0(dir,"/EGrpaaCompos.txt")
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
EGAAComposition(seqs = filePrs,winSize=20,Grp="cTriad",outFormat="txt"
,outputFileDist=ad)


unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='EIIP'>Electron-Ion Interaction Pseudopotentials (EIIP)</h2><span id='topic+EIIP'></span>

<h3>Description</h3>

<p>This function replaces each nucleotide in the input sequence with its electron-ion interaction value.
The resulting sequence is represented by a feature vector whose length is equal to the length of the sequence.
Please check the references for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EIIP(seqs, outFormat = "mat", outputFileDist = "", label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EIIP_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="EIIP_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="EIIP_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
<tr><td><code id="EIIP_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is equal to the length of the sequences
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat parameter for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. &quot;iLearn: an integrated platform and meta-learner for feature engineering, machine-learning analysis and modeling of DNA, RNA and protein sequence data.&quot; Briefings in bioinformatics 21.3 (2020): 1047-1057.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-EIIP(seqs = LNC50Nuc,outFormat="mat")
</code></pre>

<hr>
<h2 id='ENUComposition_DNA'>Enhanced Nucleotide Composition (ENUComposition_DNA)</h2><span id='topic+ENUComposition_DNA'></span>

<h3>Description</h3>

<p>This function slides a window over the input sequence(s).
Also, it computes the composition of nucleotides that appears within the limits of the window.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ENUComposition_DNA(
  seqs,
  winSize = 50,
  overLap = TRUE,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ENUComposition_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="ENUComposition_DNA_+3A_winsize">winSize</code></td>
<td>
<p>is a number which shows the size of the window.</p>
</td></tr>
<tr><td><code id="ENUComposition_DNA_+3A_overlap">overLap</code></td>
<td>
<p>This parameter shows how the window
moves on the sequence. If the overlap is set to TRUE, the next window would have distance 1 with
the previous window. Otherwise, the next window will start from the next nucleotide after the previous window.
There is no overlap between the next and previous windows.</p>
</td></tr>
<tr><td><code id="ENUComposition_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="ENUComposition_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="ENUComposition_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (4 * number of partitions displayed by the window)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-ENUComposition_DNA(seqs = LNC50Nuc, winSize=20,outFormat="mat")

ad&lt;-paste0(dir,"/ENUCcompos.txt")
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
ENUComposition_DNA(seqs = fileLNC,outFormat="txt",winSize=20
,outputFileDist=ad,overLap=FALSE)


unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='ENUComposition_RNA'>Enhanced riboNucleotide Composition (ENUComposition_RNA)</h2><span id='topic+ENUComposition_RNA'></span>

<h3>Description</h3>

<p>This function slides a window over the input sequence(s).
Also, it computes the composition of ribonucleotides that appears within the limits of the window.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ENUComposition_RNA(
  seqs,
  winSize = 50,
  overLap = TRUE,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ENUComposition_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="ENUComposition_RNA_+3A_winsize">winSize</code></td>
<td>
<p>is a number which shows the size of the window.</p>
</td></tr>
<tr><td><code id="ENUComposition_RNA_+3A_overlap">overLap</code></td>
<td>
<p>This parameter shows how the window
moves on the sequence. If the overlap is set to TRUE, the next window would have distance 1 with
the previous window. Otherwise, the next window will start from the next ribonucleotide after the previous window.
There is no overlap between the next and previous windows.</p>
</td></tr>
<tr><td><code id="ENUComposition_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="ENUComposition_RNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="ENUComposition_RNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (4 * number of partitions displayed by the window)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-ENUComposition_RNA(seqs = fileLNC, winSize=20,outFormat="mat")

ad&lt;-paste0(dir,"/ENUCcompos.txt")
ENUComposition_RNA(seqs = fileLNC,outFormat="txt",winSize=20
,outputFileDist=ad,overLap=FALSE)


unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='ExpectedValKmerNUC_DNA'>Expected Value for K-mer Nucleotide (ExpectedValKmerNUC_DNA)</h2><span id='topic+ExpectedValKmerNUC_DNA'></span>

<h3>Description</h3>

<p>This function is introduced by this package for the first time.
It computes the expected value for each k-mer in a sequence.
ExpectedValue(k-mer) = freq(k-mer) / ( freq(nucleotide1) * freq(nucleotide2)  * ... * freq(nucleotidek) )
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExpectedValKmerNUC_DNA(
  seqs,
  k = 4,
  ORF = FALSE,
  reverseORF = TRUE,
  normalized = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExpectedValKmerNUC_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValKmerNUC_DNA_+3A_k">k</code></td>
<td>
<p>is an integer value. The default is four.</p>
</td></tr>
<tr><td><code id="ExpectedValKmerNUC_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="ExpectedValKmerNUC_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="ExpectedValKmerNUC_DNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValKmerNUC_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (4^k).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-ExpectedValKmerNUC_DNA(seqs=fileLNC,k=4,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='ExpectedValKmerNUC_RNA'>Expected Value for K-mer riboNucleotide (ExpectedValKmerNUC_RNA)</h2><span id='topic+ExpectedValKmerNUC_RNA'></span>

<h3>Description</h3>

<p>This function is introduced by this package for the first time.
It computes the expected value for each k-mer in a sequence.
ExpectedValue(k-mer) = freq(k-mer) / ( freq(ribonucleotide1) * freq(ribonucleotide2)  * ... * freq(ribonucleotidek) )
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExpectedValKmerNUC_RNA(
  seqs,
  k = 4,
  ORF = FALSE,
  reverseORF = TRUE,
  normalized = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExpectedValKmerNUC_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValKmerNUC_RNA_+3A_k">k</code></td>
<td>
<p>is an integer value. The default is four.</p>
</td></tr>
<tr><td><code id="ExpectedValKmerNUC_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="ExpectedValKmerNUC_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="ExpectedValKmerNUC_RNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValKmerNUC_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (4^k).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-ExpectedValKmerNUC_RNA(seqs=fileLNC,k=4,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='ExpectedValueAA'>Expected Value for each Amino Acid (ExpectedValueAA)</h2><span id='topic+ExpectedValueAA'></span>

<h3>Description</h3>

<p>This function is introduced by this package for the first time.
It computes the expected value for each k-mer in a sequence.
ExpectedValue(k-mer) = freq(k-mer) / (c_1 * c_2  * ... * c_k),
where c_i is the number of codons that encrypt the i'th amino acid in the k-mer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExpectedValueAA(seqs, k = 2, normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExpectedValueAA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValueAA_+3A_k">k</code></td>
<td>
<p>is an integer value. The default is two.</p>
</td></tr>
<tr><td><code id="ExpectedValueAA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValueAA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 20^k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-ExpectedValueAA(seqs=filePrs,k=2,normalized=FALSE)
</code></pre>

<hr>
<h2 id='ExpectedValueGAA'>Expected Value for Grouped Amino Acid (ExpectedValueGAA)</h2><span id='topic+ExpectedValueGAA'></span>

<h3>Description</h3>

<p>This function is introduced by this package for the first time.
In this function, amino acids are first grouped into user-defined categories.
Later, the expected value of grouped amino acids is computed.
Please note that this function differs from Function <a href="#topic+ExpectedValueAA">ExpectedValueAA</a> which works on individual amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExpectedValueGAA(seqs, k = 3, Grp = "locFus", normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExpectedValueGAA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValueGAA_+3A_k">k</code></td>
<td>
<p>is an integer value. The default is three.</p>
</td></tr>
<tr><td><code id="ExpectedValueGAA_+3A_grp">Grp</code></td>
<td>
<p>is a list of vectors containig amino acids. Each vector represents a category. Users can define a customized amino acid grouping, provided that the sum of all amino acids is 20 and there is no repeated amino acid in the groups.
Also, users can choose 'cTriad'(conjointTriad), 'locFus', or 'aromatic'. Each option provides specific information about the type of an amino acid grouping.</p>
</td></tr>
<tr><td><code id="ExpectedValueGAA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValueGAA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>for more information about ExpectedValueGAA, please refer to function ExpectedValueKmer.
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (number of categories)^k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-ExpectedValueGAA(seqs=filePrs,k=2,Grp="locFus")

mat2&lt;-ExpectedValueGAA(seqs=filePrs,k=1,Grp=
list(Grp1=c("G","A","V","L","M","I","F","Y","W"),Grp2=c("K","R","H","D","E")
,Grp3=c("S","T","C","P","N","Q")))
</code></pre>

<hr>
<h2 id='ExpectedValueGKmerAA'>Expected Value for Grouped K-mer Amino Acid(ExpectedValueGKmerAA)</h2><span id='topic+ExpectedValueGKmerAA'></span>

<h3>Description</h3>

<p>This function is introduced by this package for the first time.
In this function, amino acids are first grouped into user-defined categories.
Later, the expected value of grouped k-mer is computed.
Please note that this function differs from Function <a href="#topic+ExpectedValueKmerAA">ExpectedValueKmerAA</a> which works on individual amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExpectedValueGKmerAA(
  seqs,
  k = 2,
  Grp = "locFus",
  normalized = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExpectedValueGKmerAA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValueGKmerAA_+3A_k">k</code></td>
<td>
<p>is an integer. The default value is two.</p>
</td></tr>
<tr><td><code id="ExpectedValueGKmerAA_+3A_grp">Grp</code></td>
<td>
<p>is a list of vectors containig amino acids. Each vector represents a category. Users can define a customized amino acid grouping, provided that the sum of all amino acids is 20 and there is no repeated amino acid in the groups.
Also, users can choose 'cTriad'(conjointTriad), 'locFus', or 'aromatic'. Each option provides specific information about the type of an amino acid grouping.</p>
</td></tr>
<tr><td><code id="ExpectedValueGKmerAA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValueGKmerAA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (number of categorizes)^k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-ExpectedValueGKmerAA(seqs=filePrs,k=2,Grp="locFus")

mat2&lt;-ExpectedValueGKmerAA(seqs=filePrs,k=1,Grp=
list(Grp1=c("G","A","V","L","M","I","F","Y","W"),Grp2=c("K","R","H","D","E")
,Grp3=c("S","T","C","P","N","Q")))
</code></pre>

<hr>
<h2 id='ExpectedValueKmerAA'>Expected Value for K-mer Amino Acid (ExpectedValueKmerAA)</h2><span id='topic+ExpectedValueKmerAA'></span>

<h3>Description</h3>

<p>This function computes the expected value of each k-mer by dividing the frequency of the
kmer to multiplying frequency of each amino acid of the k-mer in the sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExpectedValueKmerAA(seqs, k = 2, normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExpectedValueKmerAA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValueKmerAA_+3A_k">k</code></td>
<td>
<p>is an integer value and it shows the size of kmer in the kmer composition. The default value is 2.</p>
</td></tr>
<tr><td><code id="ExpectedValueKmerAA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="ExpectedValueKmerAA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ExpectedValue(k-mer) = freq(k-mer) / ( freq(aminoacid1) * freq(aminoacid2)  * ... * freq(aminoacidk) )
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows equals the number of sequences and
the number of columns if upto set false, is 20^k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-ExpectedValueKmerAA(filePrs,k=2,normalized=FALSE)
</code></pre>

<hr>
<h2 id='fa.read'>Fasta File Reader (fa.read)</h2><span id='topic+fa.read'></span>

<h3>Description</h3>

<p>This function reads a FASTA file. Each sequence starts with '&gt;' in the file.
This is a general function which can be applied to all types of sequences (i.e., protein/peptide, dna, and rna).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fa.read(file, legacy.mode = TRUE, seqonly = FALSE, alphabet = "aa")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fa.read_+3A_file">file</code></td>
<td>
<p>The address of the FASTA file.</p>
</td></tr>
<tr><td><code id="fa.read_+3A_legacy.mode">legacy.mode</code></td>
<td>
<p>comments all lines which start with &quot;;&quot;.</p>
</td></tr>
<tr><td><code id="fa.read_+3A_seqonly">seqonly</code></td>
<td>
<p>if it is set to true, the function will return sequences with no description.</p>
</td></tr>
<tr><td><code id="fa.read_+3A_alphabet">alphabet</code></td>
<td>
<p>is a vector which contains amino acid, RNA, or DNA alphabets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a string vector such that each element is a sequence.
</p>


<h3>References</h3>

<p>https://cran.r-project.org/web/packages/rDNAse/index.html
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
sequenceVectLNC&lt;-fa.read(file=fileLNC,alphabet="dna")

filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
sequenceVectPRO&lt;-fa.read(file=filePrs,alphabet="aa")
</code></pre>

<hr>
<h2 id='fickettScore'>Fickett Score (fickettScore)</h2><span id='topic+fickettScore'></span>

<h3>Description</h3>

<p>This function calculates the ficket score of each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fickettScore(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fickettScore_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="fickettScore_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="fickettScore_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="fickettScore_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature vector. The length of the vector is equal to the number of sequences.
Each entry in the vector contains the value of the fickett score.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
vect&lt;-fickettScore(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='G_Ccontent_DNA'>G_C content in DNA (G_Ccontent_DNA)</h2><span id='topic+G_Ccontent_DNA'></span>

<h3>Description</h3>

<p>This function calculates G-C content of each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>G_Ccontent_DNA(
  seqs,
  ORF = FALSE,
  reverseORF = TRUE,
  normalized = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="G_Ccontent_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="G_Ccontent_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="G_Ccontent_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="G_Ccontent_DNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="G_Ccontent_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature vector. The length of the vector is equal to the number of sequences.
Each entry in the vector contains G-C content of a sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
vect&lt;-G_Ccontent_DNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='G_Ccontent_RNA'>G_C content in RNA (G_Ccontent_RNA)</h2><span id='topic+G_Ccontent_RNA'></span>

<h3>Description</h3>

<p>This function calculates G-C content of each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>G_Ccontent_RNA(
  seqs,
  ORF = FALSE,
  reverseORF = TRUE,
  normalized = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="G_Ccontent_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="G_Ccontent_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="G_Ccontent_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="G_Ccontent_RNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="G_Ccontent_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a feature vector. The length of the vector is equal to the number of sequences.
Each entry in the vector contains G-C content of a sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
vect&lt;-G_Ccontent_RNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='GAAKpartComposition'>Grouped Amino Acid K Part Composition (GAAKpartComposition)</h2><span id='topic+GAAKpartComposition'></span>

<h3>Description</h3>

<p>In this function, amino acids are first grouped into user-defined categories.
Later, the composition of the grouped amino acid k part is computed.
Please note that this function differs from <a href="#topic+AAKpartComposition">AAKpartComposition</a> which works on individual amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GAAKpartComposition(
  seqs,
  k = 5,
  normalized = TRUE,
  Grp = "locFus",
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GAAKpartComposition_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="GAAKpartComposition_+3A_k">k</code></td>
<td>
<p>is an integer. Each sequence should be divided to k partition(s).</p>
</td></tr>
<tr><td><code id="GAAKpartComposition_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="GAAKpartComposition_+3A_grp">Grp</code></td>
<td>
<p>is a list of vectors containig amino acids. Each vector represents a category. Users can define a customized amino acid grouping, provided that the sum of all amino acids is 20 and there is no repeated amino acid in the groups.
Also, users can choose 'cTriad'(conjointTriad), 'locFus', or 'aromatic'. Each option provides specific information about the type of an amino acid grouping.</p>
</td></tr>
<tr><td><code id="GAAKpartComposition_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a feature matrix with k*(number of categorizes) number of columns. The number of rows is equal to the number of
sequences.
</p>


<h3>Note</h3>

<p>Warning: The length of all sequences should be greater than k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-GAAKpartComposition(seqs=filePrs,k=5,Grp="aromatic")

mat2&lt;-GAAKpartComposition(seqs=filePrs,k=3,normalized=FALSE,Grp=
list(Grp1=c("G","A","V","L","M","I","F","Y","W"),Grp2=c("K","R","H","D","E")
,Grp3=c("S","T","C","P","N","Q")))
</code></pre>

<hr>
<h2 id='GrpDDE'>Group Dipeptide Deviation from Expected Mean (GrpDDE)</h2><span id='topic+GrpDDE'></span>

<h3>Description</h3>

<p>This function is introduced by this package for the first time.
In this function, amino acids are first grouped into user-defined categories.
Later, <a href="#topic+DDE">DDE</a> is applied to grouped amino acids.
Please note that this function differs from <a href="#topic+DDE">DDE</a> which works on individual amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GrpDDE(seqs, Grp = "locFus", label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GrpDDE_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="GrpDDE_+3A_grp">Grp</code></td>
<td>
<p>is a list of vectors containig amino acids. Each vector represents a category. Users can define a customized amino acid grouping, provided that the sum of all amino acids is 20 and there is no repeated amino acid in the groups.
Also, users can choose 'cTriad'(conjointTriad), 'locFus', or 'aromatic'. Each option provides specific information about the type of an amino acid grouping.</p>
</td></tr>
<tr><td><code id="GrpDDE_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A feature matrix with (number of categorizes)^2 number of columns. The number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-GrpDDE(seqs=filePrs,Grp="aromatic")

mat2&lt;-GrpDDE(seqs=filePrs,Grp=
list(Grp1=c("G","A","V","L","M","I","F","Y","W"),Grp2=c("K","R","H","D","E")
,Grp3=c("S","T","C","P","N","Q")))
</code></pre>

<hr>
<h2 id='kAAComposition'>k Amino Acid Composition (kAAComposition)</h2><span id='topic+kAAComposition'></span>

<h3>Description</h3>

<p>This function calculates the frequency of all k-mers in the sequence(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kAAComposition(seqs, rng = 3, upto = FALSE, normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kAAComposition_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="kAAComposition_+3A_rng">rng</code></td>
<td>
<p>This parameter can be a number or a vector. Each entry of the vector holds the value of k in the k-mer composition.</p>
</td></tr>
<tr><td><code id="kAAComposition_+3A_upto">upto</code></td>
<td>
<p>It is a logical parameter. The default value is FALSE. If rng is a number and upto is set to TRUE, rng is converted
to a vector with values from 1 to rng.</p>
</td></tr>
<tr><td><code id="kAAComposition_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="kAAComposition_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns depends on rng vector. For each value k in the vector, (20)^k columns are created in the matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-kAAComposition(seqs=filePrs,rng=3,upto=TRUE)
mat2&lt;-kAAComposition(seqs=filePrs,rng=c(1,3),upto=TRUE)
</code></pre>

<hr>
<h2 id='kGAAComposition'>k Grouped Amino Acid Composition (kGAAComposition)</h2><span id='topic+kGAAComposition'></span>

<h3>Description</h3>

<p>In this function, amino acids are first grouped into user-defined categories.
Later, the composition of the k grouped amino acids is computed.
Please note that this function differs from <a href="#topic+kAAComposition">kAAComposition</a> which works on individual amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kGAAComposition(
  seqs,
  rng = 3,
  upto = FALSE,
  normalized = TRUE,
  Grp = "locFus",
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kGAAComposition_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="kGAAComposition_+3A_rng">rng</code></td>
<td>
<p>This parameter can be a number or a vector. Each entry of the vector holds the value of k in the k-mer composition.
For each k in the rng vector, a new vector (whose size is 20^k) is created which contains the frequency of k-mers.</p>
</td></tr>
<tr><td><code id="kGAAComposition_+3A_upto">upto</code></td>
<td>
<p>It is a logical parameter. The default value is FALSE. If rng is a number and upto is set to TRUE, rng is converted
to a vector with values from 1 to rng.</p>
</td></tr>
<tr><td><code id="kGAAComposition_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="kGAAComposition_+3A_grp">Grp</code></td>
<td>
<p>is a list of vectors containig amino acids. Each vector represents a category. Users can define a customized amino acid grouping, provided that the sum of all amino acids is 20 and there is no repeated amino acid in the groups.
Also, users can choose 'cTriad'(conjointTriad), 'locFus', or 'aromatic'. Each option provides specific information about the type of an amino acid grouping.</p>
</td></tr>
<tr><td><code id="kGAAComposition_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>for more details, please refer to <a href="#topic+kAAComposition">kAAComposition</a>
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is ((number of categorizes)^k)*(length of rng vector).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-CkSGAApair(seqs=filePrs,rng=2,upto=TRUE,Grp="aromatic")

mat2&lt;-CkSGAApair(seqs=filePrs,rng=c(1,3,5),Grp=
list(Grp1=c("G","A","V","L","M","I","F","Y","W"),Grp2=c("K","R","H","D","E")
,Grp3=c("S","T","C","P","N","Q")))
</code></pre>

<hr>
<h2 id='KNN_DNA'>K-Nearest Neighbor_DNA (KNN_DNA)</h2><span id='topic+KNN_DNA'></span>

<h3>Description</h3>

<p>This function is like <a href="#topic+KNNPeptide">KNNPeptide</a> with the difference that similarity score is computed by Needleman-Wunsch algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KNN_DNA(seqs, trainSeq, percent = 30, labeltr = c(), label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KNN_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="KNN_DNA_+3A_trainseq">trainSeq</code></td>
<td>
<p>is a fasta file with nucleotide sequences. Each sequence starts
with a '&gt;' character. Also it could be a string vector such that each element is a nucleotide sequence. Eaxh sequence in the training set
is associated with a label. The label is found in the parameret labeltr.</p>
</td></tr>
<tr><td><code id="KNN_DNA_+3A_percent">percent</code></td>
<td>
<p>determines the threshold which is used to identify sequences (in the training set) which are similar to the input sequence.</p>
</td></tr>
<tr><td><code id="KNN_DNA_+3A_labeltr">labeltr</code></td>
<td>
<p>This parameter is a vector whose length is equivalent to the number of sequences in the training set. It shows class of
each sequence in the trainig set.</p>
</td></tr>
<tr><td><code id="KNN_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix such that number of columns is number of classes multiplied by percent and number of rows is equal to the number of the sequences.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. &quot;iFeature: a python package and web server for features extraction and selection from protein and peptide sequences.&quot; Bioinformatics 34.14 (2018): 2499-2502.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
seqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/testData51.txt"),alphabet="dna")

posSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/posData51.txt"),alphabet="dna")
negSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/negData51.txt"),alphabet="dna")


trainSeq&lt;-c(posSeqs,negSeqs)

labelPos&lt;-rep(1,length(posSeqs))
labelNeg&lt;-rep(0,length(negSeqs))

labeltr&lt;-c(labelPos,labelNeg)

KNN_DNA(seqs=seqs,trainSeq=trainSeq,percent=5,labeltr=labeltr)

</code></pre>

<hr>
<h2 id='KNN_RNA'>K-Nearest Neighbor_RNA (KNN_RNA)</h2><span id='topic+KNN_RNA'></span>

<h3>Description</h3>

<p>This function is like <a href="#topic+KNNPeptide">KNNPeptide</a> with the difference that similarity score is computed by Needleman-Wunsch algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KNN_RNA(seqs, trainSeq, percent = 30, labeltr = c(), label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KNN_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="KNN_RNA_+3A_trainseq">trainSeq</code></td>
<td>
<p>is a fasta file with ribonucleotide sequences. Each sequence starts
with a '&gt;' character. Also it could be a string vector such that each element is a ribonucleotide sequence. Eaxh sequence in the training set
is associated with a label. The label is found in the parameret labeltr.</p>
</td></tr>
<tr><td><code id="KNN_RNA_+3A_percent">percent</code></td>
<td>
<p>determines the threshold which is used to identify sequences (in the training set) which are similar to the input sequence.</p>
</td></tr>
<tr><td><code id="KNN_RNA_+3A_labeltr">labeltr</code></td>
<td>
<p>This parameter is a vector whose length is equivalent to the number of sequences in the training set. It shows class of
each sequence in the trainig set.</p>
</td></tr>
<tr><td><code id="KNN_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix such that number of columns is number of classes multiplied by percent and number of rows is equal to the number of the sequences.
</p>


<h3>References</h3>

<p>Wei,L., Su,R., Luan,S., Liao,Z., Manavalan,B., Zou,Q. and Shi,X. Iterative feature representations improve N4-methylcytosine site prediction. Bioinformatics,  (2019).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
posSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/pos2RNA51.txt"),alphabet="rna")
negSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/neg2RNA51.txt"),alphabet="rna")
seqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/testSeq2RNA51.txt"),alphabet="rna")


trainSeq&lt;-c(posSeqs,negSeqs)


labelPos&lt;-rep(1,length(posSeqs))
labelNeg&lt;-rep(0,length(negSeqs))

labeltr&lt;-c(labelPos,labelNeg)

KNN_RNA(seqs=seqs,trainSeq=trainSeq,percent=10,labeltr=labeltr)

</code></pre>

<hr>
<h2 id='KNNPeptide'>K-Nearest Neighbor for Peptides (KNNPeptide)</h2><span id='topic+KNNPeptide'></span>

<h3>Description</h3>

<p>This function needs an extra training data set and a label. We compute the similarity score of each input sequence with all sequences in the training data set.
We use the BLOSUM62 matrix to compute the similarity score. The label shows the class of each sequence in the training data set.
KNNPeptide finds the label of 1
It reports the frequency of each class for each k
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KNNPeptide(seqs, trainSeq, percent = 30, label = c(), labeltr = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KNNPeptide_+3A_seqs">seqs</code></td>
<td>
<p>is a fasta file with amino acids sequences. Each sequence starts
with a '&gt;' character or it is a string vector such that each element is a peptide or protein sequence.</p>
</td></tr>
<tr><td><code id="KNNPeptide_+3A_trainseq">trainSeq</code></td>
<td>
<p>is a fasta file with amino acids sequences. Each sequence starts
with a '&gt;' character. Also it could be a string vector such that each element is a peptide sequence. Eaxh sequence in the training set
is associated with a label. The label is found in the parameret labeltr.</p>
</td></tr>
<tr><td><code id="KNNPeptide_+3A_percent">percent</code></td>
<td>
<p>determines the threshold which is used to identify sequences (in the training set) which are similar to the input sequence.</p>
</td></tr>
<tr><td><code id="KNNPeptide_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="KNNPeptide_+3A_labeltr">labeltr</code></td>
<td>
<p>This parameter is a vector whose length is equivalent to the number of sequences in the training set. It shows class of
each sequence in the trainig set.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix such that number of columns is number of classes multiplied by percent and number of rows is equal to the number of the sequences.
</p>


<h3>Note</h3>

<p>This function is usable for amino acid sequences with the same length in both training data set and the set of sequences.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. &quot;iFeature: a python package and web server for features extraction and selection from protein and peptide sequences.&quot; Bioinformatics 34.14 (2018): 2499-2502.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])

posSeqs&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/poSeqPTM101.csv"))[,2])
negSeqs&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/negSeqPTM101.csv"))[,2])

posSeqs&lt;-posSeqs[1:10]
negSeqs&lt;-negSeqs[1:10]

trainSeq&lt;-c(posSeqs,negSeqs)

labelPos&lt;-rep(1,length(posSeqs))
labelNeg&lt;-rep(0,length(negSeqs))

labeltr&lt;-c(labelPos,labelNeg)

KNNPeptide(seqs=ptmSeqsVect,trainSeq=trainSeq,percent=10,labeltr=labeltr)

</code></pre>

<hr>
<h2 id='KNNProtein'>K-Nearest Neighbor for Protein (KNNProtein)</h2><span id='topic+KNNProtein'></span>

<h3>Description</h3>

<p>This function is like <a href="#topic+KNNPeptide">KNNPeptide</a> with the difference that similarity score is computed by Needleman-Wunsch algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KNNProtein(seqs, trainSeq, percent = 30, labeltr = c(), label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KNNProtein_+3A_seqs">seqs</code></td>
<td>
<p>is a fasta file with amino acids sequences. Each sequence starts
with a '&gt;' character. Also it could be a string vector such that each element is a protein sequence.</p>
</td></tr>
<tr><td><code id="KNNProtein_+3A_trainseq">trainSeq</code></td>
<td>
<p>is a fasta file with amino acids sequences. Each sequence starts
with a '&gt;' character. Also it could be a string vector such that each element is a protein sequence. Eaxh sequence in the training set
is associated with a label. The label is found in the parameret labeltr.</p>
</td></tr>
<tr><td><code id="KNNProtein_+3A_percent">percent</code></td>
<td>
<p>determines the threshold which is used to identify sequences (in the training set) which are similar to the input sequence.</p>
</td></tr>
<tr><td><code id="KNNProtein_+3A_labeltr">labeltr</code></td>
<td>
<p>This parameter is a vector whose length is equivalent to the number of sequences in the training set. It shows class of
each sequence in the trainig set.</p>
</td></tr>
<tr><td><code id="KNNProtein_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix such that number of columns is number of classes multiplied by percent and number of rows is equal to the number of the sequences.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. &quot;iFeature: a python package and web server for features extraction and selection from protein and peptide sequences.&quot; Bioinformatics 34.14 (2018): 2499-2502.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
ptmSeqsVect&lt;-ptmSeqsVect[1:2]
ptmSeqsVect&lt;-sapply(ptmSeqsVect,function(seq){substr(seq,1,31)})

posSeqs&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/poSeqPTM101.csv"))[,2])
negSeqs&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/negSeqPTM101.csv"))[,2])

posSeqs&lt;-posSeqs[1:3]
negSeqs&lt;-negSeqs[1:3]

posSeqs&lt;-sapply(posSeqs,function(seq){substr(seq,1,31)})
negSeqs&lt;-sapply(negSeqs,function(seq){substr(seq,1,31)})

trainSeq&lt;-c(posSeqs,negSeqs)

labelPos&lt;-rep(1,length(posSeqs))
labelNeg&lt;-rep(0,length(negSeqs))

labeltr&lt;-c(labelPos,labelNeg)

mat&lt;-KNNProtein(seqs=ptmSeqsVect,trainSeq=trainSeq,percent=5,labeltr=labeltr)
</code></pre>

<hr>
<h2 id='kNUComposition_DNA'>k Nucleotide Composition (kNUComposition_DNA)</h2><span id='topic+kNUComposition_DNA'></span>

<h3>Description</h3>

<p>This function calculates the frequency of all k-mers in the sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kNUComposition_DNA(
  seqs,
  rng = 3,
  reverse = FALSE,
  upto = FALSE,
  normalized = TRUE,
  ORF = FALSE,
  reverseORF = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kNUComposition_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="kNUComposition_DNA_+3A_rng">rng</code></td>
<td>
<p>This parameter can be a number or a vector. Each entry of the vector holds the value of k in the k-mer composition.
For each k in the rng vector, a new vector (whose size is 4^k) is created which contains the frequency of kmers.</p>
</td></tr>
<tr><td><code id="kNUComposition_DNA_+3A_reverse">reverse</code></td>
<td>
<p>It is a logical parameter which assumes the reverse complement of the sequence.</p>
</td></tr>
<tr><td><code id="kNUComposition_DNA_+3A_upto">upto</code></td>
<td>
<p>It is a logical parameter. The default value is FALSE. If rng is a number and upto is set to TRUE, rng is converted
to a vector with values from 1 to rng.</p>
</td></tr>
<tr><td><code id="kNUComposition_DNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="kNUComposition_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="kNUComposition_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="kNUComposition_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns depends on the rng vector. For each value k in the vector, (4)^k columns are created in the matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-kNUComposition_DNA(seqs=fileLNC,rng=c(1,3))

</code></pre>

<hr>
<h2 id='kNUComposition_RNA'>k riboNucleotide Composition (kNUComposition_RNA)</h2><span id='topic+kNUComposition_RNA'></span>

<h3>Description</h3>

<p>This function calculates the frequency of all k-mers in the sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kNUComposition_RNA(
  seqs,
  rng = 3,
  reverse = FALSE,
  upto = FALSE,
  normalized = TRUE,
  ORF = FALSE,
  reverseORF = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kNUComposition_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="kNUComposition_RNA_+3A_rng">rng</code></td>
<td>
<p>This parameter can be a number or a vector. Each entry of the vector holds the value of k in the k-mer composition.
For each k in the rng vector, a new vector (whose size is 4^k) is created which contains the frequency of kmers.</p>
</td></tr>
<tr><td><code id="kNUComposition_RNA_+3A_reverse">reverse</code></td>
<td>
<p>It is a logical parameter which assumes the reverse complement of the sequence.</p>
</td></tr>
<tr><td><code id="kNUComposition_RNA_+3A_upto">upto</code></td>
<td>
<p>It is a logical parameter. The default value is FALSE. If rng is a number and upto is set to TRUE, rng is converted
to a vector with values from 1 to rng.</p>
</td></tr>
<tr><td><code id="kNUComposition_RNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="kNUComposition_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="kNUComposition_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="kNUComposition_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns depends on the rng vector. For each value k in the vector, (4)^k columns are created in the matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-kNUComposition_RNA(seqs=fileLNC,rng=c(1,3))

</code></pre>

<hr>
<h2 id='LocalPoSpKAAF'>Local Position Specific k Amino Acids Frequency (LocalPoSpKAAF)</h2><span id='topic+LocalPoSpKAAF'></span>

<h3>Description</h3>

<p>For each sequence, this function creates a feature vector denoted as
(f1,f2, f3, …, fN), where fi = freq(i'th k-mer of the sequence) / i.
It should be applied to sequences with the same length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LocalPoSpKAAF(seqs, k = 2, label = c(), outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LocalPoSpKAAF_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="LocalPoSpKAAF_+3A_k">k</code></td>
<td>
<p>is a numeric value which holds the value of k in the k-mers.</p>
</td></tr>
<tr><td><code id="LocalPoSpKAAF_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="LocalPoSpKAAF_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="LocalPoSpKAAF_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length-k+1)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-LocalPoSpKAAF(seqs = ptmSeqsVect, k=2,outFormat="mat")

ad&lt;-paste0(dir,"/LocalPoSpKaaF.txt")
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
LocalPoSpKAAF(seqs = filePrs, k=1,outFormat="txt"
,outputFileDist=ad)

unlink("dir", recursive = TRUE)

</code></pre>

<hr>
<h2 id='LocalPoSpKNUCF_DNA'>Local Position Specific k Nucleotide Frequency (LocalPoSpKNUCF_DNA)</h2><span id='topic+LocalPoSpKNUCF_DNA'></span>

<h3>Description</h3>

<p>For each sequence, this function creates a feature vector denoted as
(f1,f2, f3, …, fN), where fi = freq(i'th k-mer of the sequence) / i.
It should be applied to sequences with the same length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LocalPoSpKNUCF_DNA(
  seqs,
  k = 2,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LocalPoSpKNUCF_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="LocalPoSpKNUCF_DNA_+3A_k">k</code></td>
<td>
<p>is a numeric value which holds the value of k in the k-mers.</p>
</td></tr>
<tr><td><code id="LocalPoSpKNUCF_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="LocalPoSpKNUCF_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="LocalPoSpKNUCF_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length-k+1)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-LocalPoSpKNUCF_DNA(seqs = LNC50Nuc, k=2,outFormat="mat")

ad&lt;-paste0(dir,"/LocalPoSpKnucF.txt")
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
LocalPoSpKNUCF_DNA(seqs = fileLNC,k=1,outFormat="txt"
,outputFileDist=ad)

unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='LocalPoSpKNUCF_RNA'>Local Position Specific k riboNucleotide Frequency (LocalPoSpKNUCF_RNA)</h2><span id='topic+LocalPoSpKNUCF_RNA'></span>

<h3>Description</h3>

<p>For each sequence, this function creates a feature vector denoted as
(f1,f2, f3, …, fN), where fi = freq(i'th k-mer of the sequence) / i.
It should be applied to sequences with the same length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LocalPoSpKNUCF_RNA(
  seqs,
  k = 2,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LocalPoSpKNUCF_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="LocalPoSpKNUCF_RNA_+3A_k">k</code></td>
<td>
<p>is a numeric value which holds the value of k in the k-mers.</p>
</td></tr>
<tr><td><code id="LocalPoSpKNUCF_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="LocalPoSpKNUCF_RNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="LocalPoSpKNUCF_RNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length-k+1)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-LocalPoSpKNUCF_RNA(seqs = fileLNC, k=2,outFormat="mat")

ad&lt;-paste0(dir,"/LocalPoSpKnucF.txt")
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
LocalPoSpKNUCF_RNA(seqs = fileLNC,k=1,outFormat="txt"
,outputFileDist=ad)

unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='maxORF'>Maximum Open Reading Frame in DNA (maxORF)</h2><span id='topic+maxORF'></span>

<h3>Description</h3>

<p>This function gets a sequence as the input.
If reverse is true, the function extracts the max Open Reading Frame in the sequence and its reverse complement (hint: Six frames).
Otherwise, only the sequence is searched (hint: Three frames).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maxORF(seqs, reverse = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maxORF_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="maxORF_+3A_reverse">reverse</code></td>
<td>
<p>It is a logical parameter which assumes the reverse complement of the sequence.</p>
</td></tr>
<tr><td><code id="maxORF_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing a subsequence for each given sequences. The subsequence is the maximum ORF of the sequence.
</p>


<h3>Note</h3>

<p>If a sequence does not contain ORF, the function deletes the sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
ORF&lt;-maxORF(seqs=fileLNC,reverse=FALSE)
</code></pre>

<hr>
<h2 id='maxORF_RNA'>Maximum Open Reading Frame in RNA (maxORF_RNA)</h2><span id='topic+maxORF_RNA'></span>

<h3>Description</h3>

<p>This function gets a sequence as the input.
If reverse is true, the function extracts the max Open Reading Frame in the sequence and its reverse complement (hint: Six frames).
Otherwise, only the sequence is searched (hint: Three frames).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maxORF_RNA(seqs, reverse = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maxORF_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="maxORF_RNA_+3A_reverse">reverse</code></td>
<td>
<p>It is a logical parameter which assumes the reverse complement of the sequence.</p>
</td></tr>
<tr><td><code id="maxORF_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing a subsequence for each given sequences. The subsequence is the maximum ORF of the sequence.
</p>


<h3>Note</h3>

<p>If a sequence does not contain ORF, the function deletes the sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
ORF&lt;-maxORF_RNA(seqs=fileLNC,reverse=FALSE)
</code></pre>

<hr>
<h2 id='maxORFlength_DNA'>Maximum Open Reading Frame length in DNA (maxORFlength_DNA)</h2><span id='topic+maxORFlength_DNA'></span>

<h3>Description</h3>

<p>This function returns the length of the maximum Open Reading Frame for each sequence.
If reverse is FALSE, ORF region will be searched in a sequence.
Otherwise, it will be searched both in the sequence and its reverse complement.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maxORFlength_DNA(seqs, reverse = TRUE, normalized = FALSE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maxORFlength_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="maxORFlength_DNA_+3A_reverse">reverse</code></td>
<td>
<p>It is a logical parameter which assumes the reverse complement of the sequence.</p>
</td></tr>
<tr><td><code id="maxORFlength_DNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="maxORFlength_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing the lengths of maximum ORFs for each sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
vect&lt;-maxORFlength_DNA(seqs=fileLNC,reverse=TRUE,normalized=TRUE)

</code></pre>

<hr>
<h2 id='maxORFlength_RNA'>Maximum Open Reading Frame length in RNA (maxORFlength_RNA)</h2><span id='topic+maxORFlength_RNA'></span>

<h3>Description</h3>

<p>This function returns the length of the maximum Open Reading Frame for each sequence.
If reverse is FALSE, ORF region will be searched in a sequence.
Otherwise, it will be searched both in the sequence and its reverse complement.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maxORFlength_RNA(seqs, reverse = TRUE, normalized = FALSE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maxORFlength_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="maxORFlength_RNA_+3A_reverse">reverse</code></td>
<td>
<p>It is a logical parameter which assumes the reverse complement of the sequence.</p>
</td></tr>
<tr><td><code id="maxORFlength_RNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="maxORFlength_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing the lengths of maximum ORFs for each sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
vect&lt;-maxORFlength_RNA(seqs=fileLNC,reverse=TRUE,normalized=TRUE)

</code></pre>

<hr>
<h2 id='Mismatch_DNA'>Mismatch_DNA (Mismatch_DNA)</h2><span id='topic+Mismatch_DNA'></span>

<h3>Description</h3>

<p>This function also calculates the frequencies of all k-mers in the sequence but alows maximum m mismatch.
m&lt;k.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Mismatch_DNA(seqs, k = 3, m = 2, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Mismatch_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="Mismatch_DNA_+3A_k">k</code></td>
<td>
<p>This parameter can be a number which shows kmer.</p>
</td></tr>
<tr><td><code id="Mismatch_DNA_+3A_m">m</code></td>
<td>
<p>This parametr shows muximum number of mismatches.</p>
</td></tr>
<tr><td><code id="Mismatch_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns depends on the rng vector. For each value k in the vector, (4)^k columns are created in the matrix.
</p>


<h3>References</h3>

<p>Liu, B., Gao, X. and Zhang, H. BioSeq-Analysis2.0: an updated platform for analyzing DNA, RNA and protein sequences at sequence level and residue level based on machine learning approaches. Nucleic Acids Res (2019).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-Mismatch_DNA(seqs=fileLNC)


</code></pre>

<hr>
<h2 id='Mismatch_RNA'>Mismatch_RNA (Mismatch_RNA)</h2><span id='topic+Mismatch_RNA'></span>

<h3>Description</h3>

<p>This function also calculates the frequencies of all k-mers in the sequence but alows maximum m mismatch.
m&lt;k.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Mismatch_RNA(seqs, k = 3, m = 2, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Mismatch_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="Mismatch_RNA_+3A_k">k</code></td>
<td>
<p>This parameter can be a number which shows kmer.</p>
</td></tr>
<tr><td><code id="Mismatch_RNA_+3A_m">m</code></td>
<td>
<p>This parametr shows muximum number of mismatches.</p>
</td></tr>
<tr><td><code id="Mismatch_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns depends on the rng vector. For each value k in the vector, (4)^k columns are created in the matrix.
</p>


<h3>References</h3>

<p>Liu, B., Gao, X. and Zhang, H. BioSeq-Analysis2.0: an updated platform for analyzing DNA, RNA and protein sequences at sequence level and residue level based on machine learning approaches. Nucleic Acids Res (2019).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-Mismatch_RNA(seqs=fileLNC)


</code></pre>

<hr>
<h2 id='MMI_DNA'>Multivariate Mutual Information_DNA (MMI_DNA)</h2><span id='topic+MMI_DNA'></span>

<h3>Description</h3>

<p>MMI computes mutual information based on 2-mers T2 =  AA, AC, AG, AT, CC, CG, CT, GG, GT, TT
and 3-mers T3 = AAA, AAC, AAG, AAT, ACC, ACG, ACT, AGG, AGT, ATT, CCC, CCG, CCT, CGG, CGT, CTT, GGG, GGT, GTT and TTT
for more information please check the reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MMI_DNA(seqs, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MMI_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="MMI_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It is a feature matrix. The number of columns is 30 and the number of rows is equal to the number of sequences.
</p>


<h3>References</h3>

<p>Zhen Chen, Pei Zhao, Chen Li, Fuyi Li, Dongxu Xiang, Yong-Zi Chen, Tatsuya Akutsu, Roger J Daly, Geoffrey I Webb, Quanzhi Zhao, Lukasz Kurgan, Jiangning Song. iLearnPlus: a comprehensive and automated machine-learning platform for nucleic acid and protein sequence analysis, prediction and visualization, Nucleic Acids Research (2021).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-MMI_DNA(seqs=fileLNC)
</code></pre>

<hr>
<h2 id='MMI_RNA'>Multivariate Mutual Information_RNA (MMI_RNA)</h2><span id='topic+MMI_RNA'></span>

<h3>Description</h3>

<p>MMI computes mutual information based on 2-mers T2 =  AA, AC, AG, AU, CC, CG, CU, GG, GU, U
and 3-mers T3 = AAA, AAC, AAG, AAU, ACC, ACG, ACU, AGG, AGU, AUU, CCC, CCG, CCU, CGG, CGU, CUU, GGG, GGU, GUU and UUU
for more information please check the reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MMI_RNA(seqs, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MMI_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="MMI_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It is a feature matrix. The number of columns is 30 and the number of rows is equal to the number of sequences.
</p>


<h3>References</h3>

<p>Zhen Chen, Pei Zhao, Chen Li, Fuyi Li, Dongxu Xiang, Yong-Zi Chen, Tatsuya Akutsu, Roger J Daly, Geoffrey I Webb, Quanzhi Zhao, Lukasz Kurgan, Jiangning Song. iLearnPlus: a comprehensive and automated machine-learning platform for ribonucleic acid and protein sequence analysis, prediction and visualization, Nucleic Acids Research (2021).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-MMI_RNA(seqs=fileLNC)
</code></pre>

<hr>
<h2 id='nameKmer'>naming Kmer (nameKmer)</h2><span id='topic+nameKmer'></span>

<h3>Description</h3>

<p>This function creates all possible k-combinations of the given alphabets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nameKmer(k = 3, type = "aa", num = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nameKmer_+3A_k">k</code></td>
<td>
<p>is a numeric value.</p>
</td></tr>
<tr><td><code id="nameKmer_+3A_type">type</code></td>
<td>
<p>can be one of &quot;aa&quot;, &quot;rna&quot;, &quot;dna&quot;, or &quot;num&quot;.</p>
</td></tr>
<tr><td><code id="nameKmer_+3A_num">num</code></td>
<td>
<p>When type is set to &quot;num&quot;, it shows the numeric alphabet( 1,..,,num).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a string vector of length (20^k for 'aa' type), (4^k for 'dna' type), (4^k for 'rna' type),
and (num^k for 'num' type).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
all_kmersAA&lt;-nameKmer(k=2,type="aa")

all_kmersDNA&lt;-nameKmer(k=3,type="dna")

all_kmersNUM&lt;-nameKmer(k=3,type="num",num=2)
</code></pre>

<hr>
<h2 id='NCP_DNA'>Nucleotide Chemical Property (NCP_DNA)</h2><span id='topic+NCP_DNA'></span>

<h3>Description</h3>

<p>This function replaces nucleotides with a three-length vector.
The vector represent the nucleotides such that
'A' will be replaced with c(1, 1, 1), 'C' with c(0, 1, 0),'G' with c(1, 0, 0), and 'T' with c(0, 0, 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NCP_DNA(
  seqs,
  binaryType = "numBin",
  outFormat = "mat",
  outputFileDist = "",
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NCP_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="NCP_DNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each nucleotide is represented by a string containing 4 characters(0-1). A = &quot;0001&quot; , C = &quot;0010&quot; , G = &quot;0100&quot; , T = &quot;1000&quot;
'logicBin'(logical value): Each nucleotide is represented by a vector containing 4 logical entries. A = c(F,F,F,T) , ... , T = c(T,F,F,F)
'numBin' (numeric bin): Each nucleotide is represented by a numeric (i.e., integer) vector containing 4 numerals. A = c(0,0,0,1) , ... , T = c(1,0,0,0)</p>
</td></tr>
<tr><td><code id="NCP_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="NCP_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
<tr><td><code id="NCP_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*3.
If outFormat is 'txt', all binary values will be written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. &quot;iLearn: an integrated platform and meta-learner for feature engineering, machine-learning analysis and modeling of DNA, RNA and protein sequence data.&quot; Briefings in bioinformatics 21.3 (2020): 1047-1057.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-NCP_DNA(seqs = LNC50Nuc,binaryType="strBin",outFormat="mat")

ad&lt;-paste0(dir,"/NCP.txt")
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
NCP_DNA(seqs = fileLNC,binaryType="numBin",outFormat="txt",outputFileDist=ad)

unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='NCP_RNA'>riboNucleotide Chemical Property (NCP_RNA)</h2><span id='topic+NCP_RNA'></span>

<h3>Description</h3>

<p>This function replaces ribonucleotides with a three-length vector.
The vector represent the ribonucleotides such that
'A' will be replaced with c(1, 1, 1), 'C' with c(0, 1, 0),'G' with c(1, 0, 0), and 'U' with c(0, 0, 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NCP_RNA(
  seqs,
  binaryType = "numBin",
  outFormat = "mat",
  outputFileDist = "",
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NCP_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="NCP_RNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each ribonucleotide is represented by a string containing 4 characters(0-1). A = &quot;0001&quot; , C = &quot;0010&quot; , G = &quot;0100&quot; , T = &quot;1000&quot;
'logicBin'(logical value): Each ribonucleotide is represented by a vector containing 4 logical entries. A = c(F,F,F,T) , ... , T = c(T,F,F,F)
'numBin' (numeric bin): Each ribonucleotide is represented by a numeric (i.e., integer) vector containing 4 numerals. A = c(0,0,0,1) , ... , T = c(1,0,0,0)</p>
</td></tr>
<tr><td><code id="NCP_RNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="NCP_RNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
<tr><td><code id="NCP_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*3.
If outFormat is 'txt', all binary values will be written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. &quot;iLearn: an integrated platform and meta-learner for feature engineering, machine-learning analysis and modeling of DNA, RNA and protein sequence data.&quot; Briefings in bioinformatics 21.3 (2020): 1047-1057.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-NCP_RNA(seqs = fileLNC,binaryType="strBin",outFormat="mat")

ad&lt;-paste0(dir,"/NCP.txt")
NCP_RNA(seqs = fileLNC,binaryType="numBin",outFormat="txt",outputFileDist=ad)
unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='needleman'>Needleman-Wunsch (needleman)</h2><span id='topic+needleman'></span>

<h3>Description</h3>

<p>This function works based on Needleman-Wunsch algorithm which computes similarity score of two sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>needleman(seq1, seq2, gap = -1, mismatch = -1, match = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="needleman_+3A_seq1">seq1</code></td>
<td>
<p>(sequence1) is a string.</p>
</td></tr>
<tr><td><code id="needleman_+3A_seq2">seq2</code></td>
<td>
<p>(sequence2) is a string.</p>
</td></tr>
<tr><td><code id="needleman_+3A_gap">gap</code></td>
<td>
<p>The penalty for gaps in sequence alignment. Usually, it is a negative value.</p>
</td></tr>
<tr><td><code id="needleman_+3A_mismatch">mismatch</code></td>
<td>
<p>The penalty for the mismatch in the sequence alignment. Usually, it is a negative value.</p>
</td></tr>
<tr><td><code id="needleman_+3A_match">match</code></td>
<td>
<p>A score for the match in sequence alignment. Usually, it is a positive value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a number which indicates the similarity between sequence1 and sequence2.
</p>


<h3>References</h3>

<p>https://gist.github.com/juliuskittler/ed53696ac1e590b413aac2dddf0457f6
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
simScore&lt;-needleman(seq1="Hello",seq2="Hello",gap=-1,mismatch=-2,match=1)
</code></pre>

<hr>
<h2 id='nonStandardSeq'>nonStandard sequence (nonStandardSeq)</h2><span id='topic+nonStandardSeq'></span>

<h3>Description</h3>

<p>This function returns sequences which contain at least one non-standard alphabet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nonStandardSeq(file, legacy.mode = TRUE, seqonly = FALSE, alphabet = "aa")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nonStandardSeq_+3A_file">file</code></td>
<td>
<p>The address of fasta file which contains all the sequences.</p>
</td></tr>
<tr><td><code id="nonStandardSeq_+3A_legacy.mode">legacy.mode</code></td>
<td>
<p>It comments all lines starting with &quot;;&quot;</p>
</td></tr>
<tr><td><code id="nonStandardSeq_+3A_seqonly">seqonly</code></td>
<td>
<p>If it is set to true, the function returns sequences with no description.</p>
</td></tr>
<tr><td><code id="nonStandardSeq_+3A_alphabet">alphabet</code></td>
<td>
<p>It is a vector which contains the amino acid, RNA, or DNA alphabets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a string vector. Each element of the vector is a sequence which contains
at least one non-standard alphabet.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
nonStandardPrSeq&lt;-nonStandardSeq(file = filePrs,alphabet="aa")

fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
nonStandardNUCSeq&lt;-nonStandardSeq(file = filePrs, alphabet="dna")
</code></pre>

<hr>
<h2 id='NUC2Binary_DNA'>Nucleotide To Binary (NUC2Binary_DNA)</h2><span id='topic+NUC2Binary_DNA'></span>

<h3>Description</h3>

<p>This function transforms a nucleotide to a binary format.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NUC2Binary_DNA(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NUC2Binary_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="NUC2Binary_DNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each nucleotide is represented by a string containing 4 characters(0-1). A = &quot;0001&quot; , C = &quot;0010&quot; , G = &quot;0100&quot; , T = &quot;1000&quot;
'logicBin'(logical value): Each nucleotide is represented by a vector containing 4 logical entries. A = c(F,F,F,T) , ... , T = c(T,F,F,F)
'numBin' (numeric bin): Each nucleotide is represented by a numeric (i.e., integer) vector containing 4 numerals. A = c(0,0,0,1) , ... , T = c(1,0,0,0)</p>
</td></tr>
<tr><td><code id="NUC2Binary_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="NUC2Binary_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="NUC2Binary_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*4.
If outFormat is 'txt', all binary values will be written to a 'txt' file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat parameter for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-NUC2Binary_DNA(seqs = LNC50Nuc,outFormat="mat")

ad&lt;-paste0(dir,"/NUC2Binary.txt")
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
NUC2Binary_DNA(seqs = fileLNC,binaryType="numBin",outFormat="txt",outputFileDist=ad)

unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='NUC2Binary_RNA'>riboNucleotide To Binary (NUC2Binary_RNA)</h2><span id='topic+NUC2Binary_RNA'></span>

<h3>Description</h3>

<p>This function transforms a ribonucleotide to a binary format.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NUC2Binary_RNA(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NUC2Binary_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="NUC2Binary_RNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each ribonucleotide is represented by a string containing 4 characters(0-1). A = &quot;0001&quot; , C = &quot;0010&quot; , G = &quot;0100&quot; , U = &quot;1000&quot;
'logicBin'(logical value): Each ribonucleotide is represented by a vector containing 4 logical entries. A = c(F,F,F,T) , ... , U = c(T,F,F,F)
'numBin' (numeric bin): Each ribonucleotide is represented by a numeric (i.e., integer) vector containing 4 numerals. A = c(0,0,0,1) , ... , U = c(1,0,0,0)</p>
</td></tr>
<tr><td><code id="NUC2Binary_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="NUC2Binary_RNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="NUC2Binary_RNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*4.
If outFormat is 'txt', all binary values will be written to a 'txt' file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat parameter for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-NUC2Binary_RNA(seqs = fileLNC,outFormat="mat")

ad&lt;-paste0(dir,"/NUC2Binary.txt")
NUC2Binary_RNA(seqs = fileLNC,binaryType="numBin",outFormat="txt",outputFileDist=ad)

unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='NUCKpartComposition_DNA'>Nucleotide to K Part Composition (NUCKpartComposition_DNA)</h2><span id='topic+NUCKpartComposition_DNA'></span>

<h3>Description</h3>

<p>In this function, each sequence is divided into k equal partitions.
The length of each part is equal to ceiling(l(lenght of the sequence)/k).
The last part can have a different length containing the residual nucleotides.
The nucleotide composition is calculated for each part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NUCKpartComposition_DNA(
  seqs,
  k = 5,
  ORF = FALSE,
  reverseORF = TRUE,
  normalized = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NUCKpartComposition_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="NUCKpartComposition_DNA_+3A_k">k</code></td>
<td>
<p>is an integer value. Each sequence should be divided to k partition(s).</p>
</td></tr>
<tr><td><code id="NUCKpartComposition_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="NUCKpartComposition_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="NUCKpartComposition_DNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="NUCKpartComposition_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a feature matrix with k*4 number of columns. The number of rows is equal to the number of
sequences.
</p>


<h3>Note</h3>

<p>Warning: The length of all sequences should be greater than k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-NUCKpartComposition_DNA(seqs=fileLNC,k=5,ORF=TRUE,reverseORF=FALSE,normalized=FALSE)
</code></pre>

<hr>
<h2 id='NUCKpartComposition_RNA'>riboNucleotide to K Part Composition (NUCKpartComposition_RNA)</h2><span id='topic+NUCKpartComposition_RNA'></span>

<h3>Description</h3>

<p>In this function, each sequence is divided into k equal partitions.
The length of each part is equal to ceiling(l(lenght of the sequence)/k).
The last part can have a different length containing the residual ribonucleotides.
The ribonucleotide composition is calculated for each part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NUCKpartComposition_RNA(
  seqs,
  k = 5,
  ORF = FALSE,
  reverseORF = TRUE,
  normalized = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NUCKpartComposition_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="NUCKpartComposition_RNA_+3A_k">k</code></td>
<td>
<p>is an integer value. Each sequence should be divided to k partition(s).</p>
</td></tr>
<tr><td><code id="NUCKpartComposition_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="NUCKpartComposition_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="NUCKpartComposition_RNA_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="NUCKpartComposition_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a feature matrix with k*4 number of columns. The number of rows is equal to the number of
sequences.
</p>


<h3>Note</h3>

<p>Warning: The length of all sequences should be greater than k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-NUCKpartComposition_RNA(seqs=fileLNC,k=5,ORF=TRUE,reverseORF=FALSE,normalized=FALSE)
</code></pre>

<hr>
<h2 id='OPF_10bit'>Overlapping Property Features_10bit (OPF_10bit)</h2><span id='topic+OPF_10bit'></span>

<h3>Description</h3>

<p>This group of functions (OPF Group) categorize amino acids in different groups based on the type.
This function includes 10 amino acid properties. OPF_10bit substitutes each amino acid with a 10-dimensional vector.
Each element of the vector shows if that amino acid locates in a special property category or not. '0' means that amino acid is not located in that property group and '1' means it is located.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OPF_10bit(seqs, label = c(), outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OPF_10bit_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="OPF_10bit_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="OPF_10bit_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="OPF_10bit_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
Number of columns for this feature matrix is equal to (length of the sequences)*10 and number of rows is equal to the number of sequences.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Wei,L., Zhou,C., Chen,H., Song,J. and Su,R. ACPred-FL: a sequence-based predictor using effective feature representation to improve the prediction of anti-cancer peptides. Bioinformatics (2018).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-OPF_10bit(seqs = ptmSeqsVect,outFormat="mat")

</code></pre>

<hr>
<h2 id='OPF_7bit_T1'>Overlapping property features_7bit_T1 (OPF_7bit_T1)</h2><span id='topic+OPF_7bit_T1'></span>

<h3>Description</h3>

<p>This group of functions (OPF Group) categorize amino acids in different groups based on the type.
This function includes 7 amino acid properties. OPF_7bit_T1 substitutes each amino acid with a 7-dimensional vector.
Each element of the vector shows if that amino acid locates in a special property category or not. '0' means that amino acid is not located in that property group and '1' means it is located.
The only difference between OPF_7bit type1, type2, and type3 is in localization of amino acids in the properties groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OPF_7bit_T1(seqs, label = c(), outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OPF_7bit_T1_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="OPF_7bit_T1_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="OPF_7bit_T1_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="OPF_7bit_T1_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
Number of columns for this feature matrix is equal to (length of the sequences)*7 and number of rows is equal to the number of sequences.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Wei,L., Zhou,C., Chen,H., Song,J. and Su,R. ACPred-FL: a sequence-based predictor using effective feature representation to improve the prediction of anti-cancer peptides. Bioinformatics (2018).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-OPF_7bit_T1(seqs = ptmSeqsVect,outFormat="mat")

</code></pre>

<hr>
<h2 id='OPF_7bit_T2'>Overlapping property features_7bit_T2 (OPF_7bit_T2)</h2><span id='topic+OPF_7bit_T2'></span>

<h3>Description</h3>

<p>This group of functions (OPF Group) categorize amino acids in different groups based on the type.
This function includes 7 amino acid properties. OPF_7bit_T2 substitutes each amino acid with a 7-dimensional vector.
Each element of the vector shows if that amino acid locates in a special property category or not. '0' means that amino acid is not located in that property group and '1' means it is located.
The only difference between OPF_7bit type1, type2, and type3 is in localization of amino acids in the properties groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OPF_7bit_T2(seqs, label = c(), outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OPF_7bit_T2_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="OPF_7bit_T2_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="OPF_7bit_T2_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="OPF_7bit_T2_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
Number of columns for this feature matrix is equal to (length of the sequences)*7 and number of rows is equal to the number of sequences.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Wei,L., Zhou,C., Chen,H., Song,J. and Su,R. ACPred-FL: a sequence-based predictor using effective feature representation to improve the prediction of anti-cancer peptides. Bioinformatics (2018).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-OPF_7bit_T2(seqs = ptmSeqsVect,outFormat="mat")

</code></pre>

<hr>
<h2 id='OPF_7bit_T3'>Overlapping property features_7bit_T3 (OPF_7bit_T3)</h2><span id='topic+OPF_7bit_T3'></span>

<h3>Description</h3>

<p>This group of functions (OPF Group) categorize amino acids in different groups based on the type.
This function includes 7 amino acid properties. OPF_7bit_T3 substitutes each amino acid with a 7-dimensional vector.
Each element of the vector shows if that amino acid locates in a special property category or not. '0' means that amino acid is not located in that property group and '1' means it is located.
The only difference between OPF_7bit type1, type2, and type3 is in localization of amino acids in the properties groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OPF_7bit_T3(seqs, label = c(), outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OPF_7bit_T3_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="OPF_7bit_T3_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="OPF_7bit_T3_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="OPF_7bit_T3_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
Number of columns for this feature matrix is equal to (length of the sequences)*7 and number of rows is equal to the number of sequences.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Wei,L., Zhou,C., Chen,H., Song,J. and Su,R. ACPred-FL: a sequence-based predictor using effective feature representation to improve the prediction of anti-cancer peptides. Bioinformatics (2018).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-OPF_7bit_T3(seqs = ptmSeqsVect,outFormat="mat")

</code></pre>

<hr>
<h2 id='PCPseDNC'>Parallel Correlation Pseudo Dinucleotide Composition (PCPseDNC)</h2><span id='topic+PCPseDNC'></span>

<h3>Description</h3>

<p>This function works like <a href="#topic+PSEkNUCdi_DNA">PSEkNUCdi_DNA</a> except that the default value of selectedIdx parameter is different.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCPseDNC(
  seqs,
  selectedIdx = c("Base stacking", "Protein induced deformability", "B-DNA twist",
    "A-philicity", "Propeller twist", "Duplex stability:(freeenergy)",
    "DNA denaturation", "Bending stiffness", "Protein DNA twist", "Aida_BA_transition",
    "Breslauer_dG", "Breslauer_dH", "Electron_interaction", "Hartman_trans_free_energy",
    "Helix-Coil_transition", "Lisser_BZ_transition", "Polar_interaction",
    "SantaLucia_dG", "SantaLucia_dS", "Sarai_flexibility", "Stability", "Sugimoto_dG",
    "Sugimoto_dH", "Sugimoto_dS", "Duplex tability(disruptenergy)",     
    "Stabilising energy of Z-DNA", "Breslauer_dS", "Ivanov_BA_transition",
    "SantaLucia_dH", "Stacking_energy", "Watson-Crick_interaction",
    "Dinucleotide GC Content", "Rise", "Roll", "Shift", "Slide", "Tilt", "Twist"),
  lambda = 3,
  w = 0.05,
  l = 2,
  ORF = FALSE,
  reverseORF = TRUE,
  threshold = 1,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PCPseDNC_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="PCPseDNC_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>is a vector of Ids or indices of the desired physicochemical properties of dinucleotides.
Users can choose the desired indices by their ids or their names in the DI_DNA index file. Default value of
this parameter is a vector with (&quot;Base stacking&quot;,&quot;Protein induced deformability&quot;,&quot;B-DNA twist&quot;,&quot;A-philicity&quot;,
&quot;Propeller twist&quot;,&quot;Duplex stability:(freeenergy)&quot;,&quot;DNA denaturation&quot;,&quot;Bending stiffness&quot;,
&quot;Protein DNA twist&quot;,&quot;Aida_BA_transition&quot;,&quot;Breslauer_dG&quot;,&quot;Breslauer_dH&quot;,&quot;Electron_interaction&quot;,
&quot;Hartman_trans_free_energy&quot;,&quot;Helix-Coil_transition&quot;,&quot;Lisser_BZ_transition&quot;,&quot;Polar_interaction&quot;,
&quot;SantaLucia_dG&quot;,&quot;SantaLucia_dS&quot;,&quot;Sarai_flexibility&quot;,&quot;Stability&quot;,&quot;Sugimoto_dG&quot;,
&quot;Sugimoto_dH&quot;,&quot;Sugimoto_dS&quot;,&quot;Duplex tability(disruptenergy)&quot;,&quot;Stabilising energy of Z-DNA&quot;,
&quot;Breslauer_dS&quot;,&quot;Ivanov_BA_transition&quot;,&quot;SantaLucia_dH&quot;,&quot;Stacking_energy&quot;,&quot;Watson-Crick_interaction&quot;,&quot;Dinucleotide GC Content&quot;,
&quot;Rise&quot;, &quot;Roll&quot;, &quot;Shift&quot;, &quot;Slide&quot;, &quot;Tilt&quot;, &quot;Twist&quot;) entries.</p>
</td></tr>
<tr><td><code id="PCPseDNC_+3A_lambda">lambda</code></td>
<td>
<p>is a tuning parameter. This integer value shows the maximum limit of spaces between dinucleotide pairs. The Number of spaces
changes from 1 to lambda.</p>
</td></tr>
<tr><td><code id="PCPseDNC_+3A_w">w</code></td>
<td>
<p>(weight) is a tuning parameter. It changes in the range of 0 to 1. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="PCPseDNC_+3A_l">l</code></td>
<td>
<p>This parameter keeps the value of l in lmer composition. The lmers form the first 4^l elements of the APkNCdi descriptor.</p>
</td></tr>
<tr><td><code id="PCPseDNC_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="PCPseDNC_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="PCPseDNC_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In selectedIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="PCPseDNC_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the pseudo nucleotide composition for each physicochemical property of di-nucleotides.
We have provided users with the ability to choose among the 148 properties in the di-nucleotide index database.
</p>


<h3>Value</h3>

<p>a feature matrix such that the number of columns is 4^l+lambda and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-PSEkNUCdi_DNA(seqs=fileLNC,l=2,ORF=TRUE,threshold=0.8)
</code></pre>

<hr>
<h2 id='PS2_DNA'>Position-specific of two nucleotide_DNA (PS2_DNA)</h2><span id='topic+PS2_DNA'></span>

<h3>Description</h3>

<p>This function transforms each di-nucleotide of the sequence to a binary format.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PS2_DNA(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PS2_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="PS2_DNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each di-nucleotide is represented by a string containing 16 characters(0-1). For example, 'AA' = &quot;1000000000000000&quot;, 'AC' = &quot;0100000000000000&quot;, ..., 'TT'= &quot;0000000000000001&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 16 logical entries. For example, 'AA' = c(T,F,F,F,F,F,F,F,F,F,F,F,F,F,F,F), ...
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 16 numerals. For example, 'AA' = c(1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0)</p>
</td></tr>
<tr><td><code id="PS2_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="PS2_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="PS2_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences-1)*16.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Zhen Chen, Pei Zhao, Chen Li, Fuyi Li, Dongxu Xiang, Yong-Zi Chen, Tatsuya Akutsu, Roger J Daly, Geoffrey I Webb, Quanzhi Zhao, Lukasz Kurgan, Jiangning Song, iLearnPlus: a comprehensive and automated machine-learning platform for nucleic acid and protein sequence analysis, prediction and visualization, Nucleic Acids Research, (2021).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-PS2_DNA(seqs = LNC50Nuc,outFormat="mat")




</code></pre>

<hr>
<h2 id='PS2_RNA'>Position-specific of two nucleotide_RNA (PS2_RNA)</h2><span id='topic+PS2_RNA'></span>

<h3>Description</h3>

<p>This function transforms each di-ribonucleotide of the sequence to a binary format.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PS2_RNA(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PS2_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="PS2_RNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each di-nucleotide is represented by a string containing 16 characters(0-1). For example, 'AA' = &quot;1000000000000000&quot;, 'AC' = &quot;0100000000000000&quot;, ..., 'TT'= &quot;0000000000000001&quot;
'logicBin'(logical value): Each amino acid is represented by a vector containing 16 logical entries. For example, 'AA' = c(T,F,F,F,F,F,F,F,F,F,F,F,F,F,F,F), ...
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 16 numerals. For example, 'AA' = c(1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0)</p>
</td></tr>
<tr><td><code id="PS2_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="PS2_RNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="PS2_RNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences-1)*16.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Zhen Chen, Pei Zhao, Chen Li, Fuyi Li, Dongxu Xiang, Yong-Zi Chen, Tatsuya Akutsu, Roger J Daly, Geoffrey I Webb, Quanzhi Zhao, Lukasz Kurgan, Jiangning Song, iLearnPlus: a comprehensive and automated machine-learning platform for nucleic acid and protein sequence analysis, prediction and visualization, Nucleic Acids Research, (2021).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-PS2_RNA(seqs = fileLNC, binaryType="numBin",outFormat="mat")

</code></pre>

<hr>
<h2 id='PS3_DNA'>Position-specific of three nucleotide_DNA (PS3_DNA)</h2><span id='topic+PS3_DNA'></span>

<h3>Description</h3>

<p>This function transforms each tri-nucleotide of the sequence to a binary format.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PS3_DNA(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PS3_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="PS3_DNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each di-nucleotide is represented by a string containing 64 characters (63 times '0' and one '1'). For example, 'AAA' = &quot;1000000000000000...0&quot;, ....
'logicBin'(logical value): Each amino acid is represented by a vector containing 64 logical entries (63 times F and one T). For example, 'AA' = c(T,F,F,F,F,F,F,F,F,F,F,F,F,F,F,F,...,F), ...
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 64 numerals (63 times '0' and one '1'). For example, 'AA' = c(1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="PS3_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="PS3_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="PS3_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences-2)*64.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Zhen Chen, Pei Zhao, Chen Li, Fuyi Li, Dongxu Xiang, Yong-Zi Chen, Tatsuya Akutsu, Roger J Daly, Geoffrey I Webb, Quanzhi Zhao, Lukasz Kurgan, Jiangning Song, iLearnPlus: a comprehensive and automated machine-learning platform for nucleic acid and protein sequence analysis, prediction and visualization, Nucleic Acids Research, (2021).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-PS3_DNA(seqs = LNC50Nuc,outFormat="mat")




</code></pre>

<hr>
<h2 id='PS3_RNA'>Position-specific of three ribonucleotide_RNA (PS3_RNA)</h2><span id='topic+PS3_RNA'></span>

<h3>Description</h3>

<p>This function transforms each tri-ribonucleotide of the sequence to a binary format.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PS3_RNA(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PS3_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="PS3_RNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each di-ribonucleotide is represented by a string containing 64 characters (63 times '0' and one '1'). For example, 'AAA' = &quot;1000000000000000...0&quot;, ....
'logicBin'(logical value): Each amino acid is represented by a vector containing 64 logical entries (63 times F and one T). For example, 'AA' = c(T,F,F,F,F,F,F,F,F,F,F,F,F,F,F,F,...,F), ...
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 64 numerals (63 times '0' and one '1'). For example, 'AA' = c(1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="PS3_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="PS3_RNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="PS3_RNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences-2)*64.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Zhen Chen, Pei Zhao, Chen Li, Fuyi Li, Dongxu Xiang, Yong-Zi Chen, Tatsuya Akutsu, Roger J Daly, Geoffrey I Webb, Quanzhi Zhao, Lukasz Kurgan, Jiangning Song, iLearnPlus: a comprehensive and automated machine-learning platform for nucleic acid and protein sequence analysis, prediction and visualization, Nucleic Acids Research, (2021).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-PS3_RNA(seqs = fileLNC, binaryType="numBin",outFormat="mat")




</code></pre>

<hr>
<h2 id='PS4_DNA'>Position-specific of four nucleotide_DNA (PS4_DNA)</h2><span id='topic+PS4_DNA'></span>

<h3>Description</h3>

<p>This function transforms each 4-nucleotide of the sequence to a binary format.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PS4_DNA(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PS4_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="PS4_DNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each di-nucleotide is represented by a string containing 256 characters (255 times '0' and one '1'). For example, 'AAA' = &quot;1000000000000000...0&quot;, ....
'logicBin'(logical value): Each amino acid is represented by a vector containing 256 logical entries (255 times F and one T). For example, 'AA' = c(T,F,F,F,F,F,F,F,F,F,F,F,F,F,F,F,...,F), ...
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 256 numerals (255 times '0' and one '1'). For example, 'AA' = c(1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="PS4_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="PS4_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="PS4_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences-3)*256.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Zhen Chen, Pei Zhao, Chen Li, Fuyi Li, Dongxu Xiang, Yong-Zi Chen, Tatsuya Akutsu, Roger J Daly, Geoffrey I Webb, Quanzhi Zhao, Lukasz Kurgan, Jiangning Song, iLearnPlus: a comprehensive and automated machine-learning platform for nucleic acid and protein sequence analysis, prediction and visualization, Nucleic Acids Research, (2021).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-PS4_DNA(seqs = LNC50Nuc,outFormat="mat")




</code></pre>

<hr>
<h2 id='PS4_RNA'>Position-specific of four ribonucleotide (PS4_RNA)</h2><span id='topic+PS4_RNA'></span>

<h3>Description</h3>

<p>This function transforms each 4-ribonucleotide of the sequence to a binary format.
The type of the binary format is determined by the binaryType parameter.
For details about each format, please refer to the description of the binaryType parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PS4_RNA(
  seqs,
  binaryType = "numBin",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PS4_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="PS4_RNA_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each di-ribonucleotide is represented by a string containing 256 characters (255 times '0' and one '1'). For example, 'AAA' = &quot;1000000000000000...0&quot;, ....
'logicBin'(logical value): Each amino acid is represented by a vector containing 256 logical entries (255 times F and one T). For example, 'AA' = c(T,F,F,F,F,F,F,F,F,F,F,F,F,F,F,F,...,F), ...
'numBin' (numeric bin): Each amino acid is represented by a numeric (i.e., integer) vector containing 256 numerals (255 times '0' and one '1'). For example, 'AA' = c(1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,...,0)</p>
</td></tr>
<tr><td><code id="PS4_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="PS4_RNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="PS4_RNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences-3)*256.
If outFormat is 'txt', all binary values will be written to a the output is written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Zhen Chen, Pei Zhao, Chen Li, Fuyi Li, Dongxu Xiang, Yong-Zi Chen, Tatsuya Akutsu, Roger J Daly, Geoffrey I Webb, Quanzhi Zhao, Lukasz Kurgan, Jiangning Song, iLearnPlus: a comprehensive and automated machine-learning platform for nucleic acid and protein sequence analysis, prediction and visualization, Nucleic Acids Research, (2021).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-PS4_RNA(seqs = fileLNC, binaryType="numBin",outFormat="mat")




</code></pre>

<hr>
<h2 id='PSEAAC'>Pseudo-Amino Acid Composition (Parallel) (PSEAAC)</h2><span id='topic+PSEAAC'></span>

<h3>Description</h3>

<p>This function calculates the pseudo amino acid composition (parallel)
for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PSEAAC(
  seqs,
  aaIDX = c("ARGP820101", "HOPT810101", "Mass"),
  lambda = 30,
  w = 0.05,
  l = 1,
  threshold = 1,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PSEAAC_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PSEAAC_+3A_aaidx">aaIDX</code></td>
<td>
<p>is a vector of Ids or indexes of the user-selected physicochemical properties in the aaIndex2 database.
The default values of the vector are the hydrophobicity ids and hydrophilicity ids and Mass of residual
in the amino acid index file.</p>
</td></tr>
<tr><td><code id="PSEAAC_+3A_lambda">lambda</code></td>
<td>
<p>is a tuning parameter. Its value indicates the maximum number of spaces between amino acid pairs. The number
changes from 1 to lambda.</p>
</td></tr>
<tr><td><code id="PSEAAC_+3A_w">w</code></td>
<td>
<p>(weight) is a tuning parameter. It changes in from 0 to 1. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="PSEAAC_+3A_l">l</code></td>
<td>
<p>This parameter keeps the value of l in lmer composition. The lmers form the first 20^l elements of the APAAC descriptor.</p>
</td></tr>
<tr><td><code id="PSEAAC_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. It deletes aaIndexes which have a correlation
bigger than the threshold. The default value is 1.</p>
</td></tr>
<tr><td><code id="PSEAAC_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A feature matrix such that the number of columns is 20^l+(lambda) and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-PSEAAC(seqs=filePrs,l=2)
</code></pre>

<hr>
<h2 id='PseEIIP'>Pseudo Electron-Ion Interaction Pseudopotentials of Trinucleotide (PseEIIP)</h2><span id='topic+PseEIIP'></span>

<h3>Description</h3>

<p>This function calculates the pseudo electron-ion interaction for each sequence.
It creates a feature vector for each sequence.
The vector contains a value for each for each tri-nucleotide.
The value is computed by multiplying the aggregate value of electron-ion interaction of each nucleotide
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseEIIP(seqs, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseEIIP_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="PseEIIP_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix which the number of rows is equal to the number of sequences and the
number of columns is 4^3=64.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. &quot;iLearn: an integrated platform and meta-learner for feature engineering, machine-learning analysis and modeling of DNA, RNA and protein sequence data.&quot; Briefings in bioinformatics 21.3 (2020): 1047-1057.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-PseEIIP(seqs = LNC50Nuc)
</code></pre>

<hr>
<h2 id='PSEkNUCdi_DNA'>Pseudo k Nucleotide Composition-Di(Parallel) (PSEkNUCdi_DNA)</h2><span id='topic+PSEkNUCdi_DNA'></span>

<h3>Description</h3>

<p>This function calculates the pseudo-k nucleotide composition(Di) (Parallel)
for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PSEkNUCdi_DNA(
  seqs,
  selectedIdx = c("Rise", "Roll", "Shift", "Slide", "Tilt", "Twist"),
  lambda = 3,
  w = 0.05,
  l = 2,
  ORF = FALSE,
  reverseORF = TRUE,
  threshold = 1,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PSEkNUCdi_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_DNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>is a vector of Ids or indices of the desired physicochemical properties of dinucleotides.
Users can choose the desired indices by their ids or their names in the DI_DNA file.
The default values of the parameter is a vector with (&quot;Rise&quot;, &quot;Roll&quot;, &quot;Shift&quot;, &quot;Slide&quot;, &quot;Tilt&quot;, &quot;Twist&quot;) ids.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_DNA_+3A_lambda">lambda</code></td>
<td>
<p>is a tuning parameter. This integer value shows the maximum limit of spaces between dinucleotide pairs. The Number of spaces
changes from 1 to lambda.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_DNA_+3A_w">w</code></td>
<td>
<p>(weight) is a tuning parameter. It changes in the range of 0 to 1. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_DNA_+3A_l">l</code></td>
<td>
<p>This parameter keeps the value of l in lmer composition. The lmers form the first 4^l elements of the APkNCdi descriptor.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_DNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In selectedIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the pseudo nucleotide composition for each physicochemical property of di-nucleotides.
We have provided users with the ability to choose among the 148 properties in the di-nucleotide index database.
</p>


<h3>Value</h3>

<p>a feature matrix such that the number of columns is 4^l+lambda and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-PSEkNUCdi_DNA(seqs=fileLNC,l=2,ORF=TRUE,threshold=0.8)
</code></pre>

<hr>
<h2 id='PSEkNUCdi_RNA'>Pseudo k riboNucleotide Composition-Di(Parallel) (PSEkNUCdi_RNA)</h2><span id='topic+PSEkNUCdi_RNA'></span>

<h3>Description</h3>

<p>This function calculates the pseudo-k ribonucleotide composition(Di) (Parallel)
for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PSEkNUCdi_RNA(
  seqs,
  selectedIdx = c("Rise (RNA)", "Roll (RNA)", "Shift (RNA)", "Slide (RNA)",
    "Tilt (RNA)", "Twist (RNA)"),
  lambda = 3,
  w = 0.05,
  l = 2,
  ORF = FALSE,
  reverseORF = TRUE,
  threshold = 1,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PSEkNUCdi_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_RNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>is a vector of Ids or indices of the desired physicochemical properties of di-ribonucleotides.
Users can choose the desired indices by their ids or their names in the DI_RNA peoperties file.
The default value of this parameter is a vector with (&quot;Rise (RNA)&quot;, &quot;Roll (RNA)&quot;, &quot;Shift (RNA)&quot;, &quot;Slide (RNA)&quot;, &quot;Tilt (RNA)&quot;,&quot;Twist (RNA)&quot;) ids.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_RNA_+3A_lambda">lambda</code></td>
<td>
<p>is a tuning parameter. This integer value shows the maximum limit of spaces between di-ribonucleotide pairs. The Number of spaces
changes from 1 to lambda.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_RNA_+3A_w">w</code></td>
<td>
<p>(weight) is a tuning parameter. It changes in the range of 0 to 1. The default value is 0.5.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_RNA_+3A_l">l</code></td>
<td>
<p>This parameter keeps the value of l in lmer composition. The lmers form the first 4^l elements of the APkNCdi descriptor.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_RNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In selectedIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="PSEkNUCdi_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the pseudo ribonucleotide composition for each physicochemical property of di-ribonucleotides.
We have provided users with the ability to choose among the 22 properties in the di-ribonucleotide index database.
</p>


<h3>Value</h3>

<p>a feature matrix such that the number of columns is 4^l+lambda and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-PSEkNUCdi_RNA(seqs=fileLNC,l=2,ORF=TRUE,threshold=0.8)
</code></pre>

<hr>
<h2 id='PSEkNUCTri_DNA'>Pseudo k Nucleotide Composition-Tri(Parallel) (PSEkNUCTri_RNA)</h2><span id='topic+PSEkNUCTri_DNA'></span>

<h3>Description</h3>

<p>This function calculates pseudo-k nucleotide composition(Tri) (Parallel)
for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PSEkNUCTri_DNA(
  seqs,
  selectedIdx = c("Dnase I", "Bendability (DNAse)"),
  lambda = 3,
  w = 0.05,
  l = 3,
  ORF = FALSE,
  reverseORF = TRUE,
  threshold = 1,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PSEkNUCTri_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="PSEkNUCTri_DNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>is a vector of Ids or indices of the desired physicochemical properties of trinucleotides.
Users can choose the desired indices by their ids or their names in the TRI_DNA index file.
The default value of this parameter is a vector with (&quot;Dnase I&quot;, &quot;Bendability (DNAse)&quot;) ids.</p>
</td></tr>
<tr><td><code id="PSEkNUCTri_DNA_+3A_lambda">lambda</code></td>
<td>
<p>is a tuning parameter. This integer value shows the maximum limit of spaces between Tri-nucleotide pairs. The Number of spaces
changes from 1 to lambda.</p>
</td></tr>
<tr><td><code id="PSEkNUCTri_DNA_+3A_w">w</code></td>
<td>
<p>(weight) is a tuning parameter. It can take a value in the range 0 to 1. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="PSEkNUCTri_DNA_+3A_l">l</code></td>
<td>
<p>This parameter keeps the value of l in lmer composition. The lmers form the first 4^l elements of the APkNCTri descriptor.</p>
</td></tr>
<tr><td><code id="PSEkNUCTri_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="PSEkNUCTri_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="PSEkNUCTri_DNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between (0 , 1]. In selectedIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="PSEkNUCTri_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the pseudo nucleotide composition for each physicochemical property of trinucleotides.
We have provided users with the ability to choose among the 12 properties in the tri-nucleotide index database.
</p>


<h3>Value</h3>

<p>a feature matrix such that the number of columns is 4^l+lambda and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-PSEkNUCTri_DNA(seqs=fileLNC, l=2,ORF=TRUE,threshold=0.8)
</code></pre>

<hr>
<h2 id='PseKRAAC_T1'>Pseudo K_tuple Reduced Amino Acid Composition Type-1 (PseKRAAC_T1)</h2><span id='topic+PseKRAAC_T1'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type1(PseKRAAC_T1) contains Grp 2 to 20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T1(
  seqs,
  type = "gap",
  Grp = 5,
  GapOrLambdaValue = 2,
  k = 2,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T1_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T1_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T1_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T1_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T1_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T1_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
2=c(&quot;CMFILVWY&quot;, &quot;AGTSNQDEHRKP&quot;),
3=c(&quot;CMFILVWY&quot;, &quot;AGTSP&quot;, &quot;NQDEHRK&quot;),
4=c(&quot;CMFWY&quot;, &quot;ILV&quot;, &quot;AGTS&quot;, &quot;NQDEHRKP&quot;),
5=c(&quot;WFYH&quot;, &quot;MILV&quot;, &quot;CATSP&quot;, &quot;G&quot;, &quot;NQDERK&quot;),
6=c(&quot;WFYH&quot;, &quot;MILV&quot;, &quot;CATS&quot;, &quot;P&quot;, &quot;G&quot;, &quot;NQDERK&quot;),
7=c(&quot;WFYH&quot;, &quot;MILV&quot;, &quot;CATS&quot;, &quot;P&quot;, &quot;G&quot;, &quot;NQDE&quot;, &quot;RK&quot;),
8=c(&quot;WFYH&quot;, &quot;MILV&quot;, &quot;CA&quot;, &quot;NTS&quot;, &quot;P&quot;, &quot;G&quot;, &quot;DE&quot;, &quot;QRK&quot;),
9=c(&quot;WFYH&quot;, &quot;MI&quot;, &quot;LV&quot;, &quot;CA&quot;, &quot;NTS&quot;, &quot;P&quot;, &quot;G&quot;, &quot;DE&quot;, &quot;QRK&quot;),
10=c(&quot;WFY&quot;, &quot;ML&quot;, &quot;IV&quot;, &quot;CA&quot;, &quot;TS&quot;, &quot;NH&quot;, &quot;P&quot;, &quot;G&quot;, &quot;DE&quot;, &quot;QRK&quot;),
11=c(&quot;WFY&quot;, &quot;ML&quot;, &quot;IV&quot;, &quot;CA&quot;, &quot;TS&quot;, &quot;NH&quot;, &quot;P&quot;, &quot;G&quot;, &quot;D&quot;, &quot;QE&quot;, &quot;RK&quot;),
12=c(&quot;WFY&quot;, &quot;ML&quot;, &quot;IV&quot;, &quot;C&quot;, &quot;A&quot;, &quot;TS&quot;, &quot;NH&quot;, &quot;P&quot;, &quot;G&quot;, &quot;D&quot;, &quot;QE&quot;, &quot;RK&quot;),
13=c(&quot;WFY&quot;, &quot;ML&quot;, &quot;IV&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;S&quot;, &quot;NH&quot;, &quot;P&quot;, &quot;G&quot;, &quot;D&quot;, &quot;QE&quot;, &quot;RK&quot;),
14=c(&quot;WFY&quot;, &quot;ML&quot;, &quot;IV&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;S&quot;, &quot;NH&quot;, &quot;P&quot;, &quot;G&quot;, &quot;D&quot;, &quot;QE&quot;, &quot;R&quot;, &quot;K&quot;),
15=c(&quot;WFY&quot;, &quot;ML&quot;, &quot;IV&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;S&quot;, &quot;N&quot;, &quot;H&quot;, &quot;P&quot;, &quot;G&quot;, &quot;D&quot;, &quot;QE&quot;, &quot;R&quot;, &quot;K&quot;),
16=c(&quot;W&quot;, &quot;FY&quot;, &quot;ML&quot;, &quot;IV&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;S&quot;, &quot;N&quot;, &quot;H&quot;, &quot;P&quot;, &quot;G&quot;, &quot;D&quot;, &quot;QE&quot;, &quot;R&quot;, &quot;K&quot;),
17=c(&quot;W&quot;, &quot;FY&quot;, &quot;ML&quot;, &quot;IV&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;S&quot;, &quot;N&quot;, &quot;H&quot;, &quot;P&quot;, &quot;G&quot;, &quot;D&quot;, &quot;Q&quot;, &quot;E&quot;, &quot;R&quot;, &quot;K&quot;),
18=c(&quot;W&quot;, &quot;FY&quot;, &quot;M&quot;, &quot;L&quot;, &quot;IV&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;S&quot;, &quot;N&quot;, &quot;H&quot;, &quot;P&quot;, &quot;G&quot;, &quot;D&quot;, &quot;Q&quot;, &quot;E&quot;, &quot;R&quot;, &quot;K&quot;),
19=c(&quot;W&quot;, &quot;F&quot;, &quot;Y&quot;, &quot;M&quot;, &quot;L&quot;, &quot;IV&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;S&quot;, &quot;N&quot;, &quot;H&quot;, &quot;P&quot;, &quot;G&quot;, &quot;D&quot;, &quot;Q&quot;, &quot;E&quot;, &quot;R&quot;, &quot;K&quot;),
20=c(&quot;W&quot;, &quot;F&quot;, &quot;Y&quot;, &quot;M&quot;, &quot;L&quot;, &quot;I&quot;, &quot;V&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;S&quot;, &quot;N&quot;, &quot;H&quot;, &quot;P&quot;, &quot;G&quot;, &quot;D&quot;, &quot;Q&quot;, &quot;E&quot;, &quot;R&quot;, &quot;K&quot;)
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T1(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T1(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T10'>Pseudo K_tuple Reduced Amino Acid Composition Type-10 (PseKRAAC_T10)</h2><span id='topic+PseKRAAC_T10'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type10(PseKRAAC_T10) contains Grp 2-20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T10(
  seqs,
  type = "gap",
  Grp = 5,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T10_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T10_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T10_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T10_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T10_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T10_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
2=c('CMFILVWY', 'AGTSNQDEHRKP'),
3=c('CMFILVWY', 'AGTSP', 'NQDEHRK'),
4=c('CMFWY', 'ILV', 'AGTS', 'NQDEHRKP'),
5=c('FWYH', 'MILV', 'CATSP', 'G', 'NQDERK'),
6=c('FWYH', 'MILV', 'CATS', 'P', 'G', 'NQDERK'),
7=c('FWYH', 'MILV', 'CATS', 'P', 'G', 'NQDE', 'RK'),
8=c('FWYH', 'MILV', 'CA', 'NTS', 'P', 'G', 'DE', 'QRK'),
9=c('FWYH', 'ML', 'IV', 'CA', 'NTS', 'P', 'G', 'DE', 'QRK'),
10=c('FWY', 'ML', 'IV', 'CA', 'TS', 'NH', 'P', 'G', 'DE', 'QRK'),
11=c('FWY', 'ML', 'IV', 'CA', 'TS', 'NH', 'P', 'G', 'D', 'QE', 'RK'),
12=c('FWY', 'ML', 'IV', 'C', 'A', 'TS', 'NH', 'P', 'G', 'D', 'QE', 'RK'),
13=c('FWY', 'ML', 'IV', 'C', 'A', 'T', 'S', 'NH', 'P', 'G', 'D', 'QE', 'RK'),
14=c('FWY', 'ML', 'IV', 'C', 'A', 'T', 'S', 'NH', 'P', 'G', 'D', 'QE', 'R', 'K'),
15=c('FWY', 'ML', 'IV', 'C', 'A', 'T', 'S', 'N', 'H', 'P', 'G', 'D', 'QE', 'R', 'K'),
16=c('W', 'FY', 'ML', 'IV', 'C', 'A', 'T', 'S', 'N', 'H', 'P', 'G', 'D', 'QE', 'R', 'K'),
17=c('W', 'FY', 'ML', 'IV', 'C', 'A', 'T', 'S', 'N', 'H', 'P', 'G', 'D', 'Q', 'E', 'R', 'K'),
18=c('W', 'FY', 'M', 'L', 'IV', 'C', 'A', 'T', 'S', 'N', 'H', 'P', 'G', 'D', 'Q', 'E', 'R', 'K'),
19=c('W', 'F', 'Y', 'M', 'L', 'IV', 'C', 'A', 'T', 'S', 'N', 'H', 'P', 'G', 'D', 'Q', 'E', 'R', 'K'),
20=c('W', 'F', 'Y', 'M', 'L', 'I', 'V', 'C', 'A', 'T', 'S', 'N', 'H', 'P', 'G', 'D', 'Q', 'E', 'R', 'K')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T10(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T10(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T11'>Pseudo K_tuple Reduced Amino Acid Composition Type-11 (PseKRAAC_T11)</h2><span id='topic+PseKRAAC_T11'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type11(PseKRAAC_T11) contains Grp 2-20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T11(
  seqs,
  type = "gap",
  Grp = 5,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T11_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T11_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T11_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T11_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T11_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T11_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
2=c('CFYWMLIV', 'GPATSNHQEDRK'),
3=c('CFYWMLIV', 'GPATS', 'NHQEDRK'),
4=c('CFYW', 'MLIV', 'GPATS', 'NHQEDRK'),
5=c('CFYW', 'MLIV', 'G', 'PATS', 'NHQEDRK'),
6=c('CFYW', 'MLIV', 'G', 'P', 'ATS', 'NHQEDRK'),
7=c('CFYW', 'MLIV', 'G', 'P', 'ATS', 'NHQED', 'RK'),
8=c('CFYW', 'MLIV', 'G', 'P', 'ATS', 'NH', 'QED', 'RK'),
9=c('CFYW', 'ML', 'IV', 'G', 'P', 'ATS', 'NH', 'QED', 'RK'),
10=c('C', 'FYW', 'ML', 'IV', 'G', 'P', 'ATS', 'NH', 'QED', 'RK'),
11=c('C', 'FYW', 'ML', 'IV', 'G', 'P', 'A', 'TS', 'NH', 'QED', 'RK'),
12=c('C', 'FYW', 'ML', 'IV', 'G', 'P', 'A', 'TS', 'NH', 'QE', 'D', 'RK'),
13=c('C', 'FYW', 'ML', 'IV', 'G', 'P', 'A', 'T', 'S', 'NH', 'QE', 'D', 'RK'),
14=c('C', 'FYW', 'ML', 'IV', 'G', 'P', 'A', 'T', 'S', 'N', 'H', 'QE', 'D', 'RK'),
15=c('C', 'FYW', 'ML', 'IV', 'G', 'P', 'A', 'T', 'S', 'N', 'H', 'QE', 'D', 'R', 'K'),
16=c('C', 'FY', 'W', 'ML', 'IV', 'G', 'P', 'A', 'T', 'S', 'N', 'H', 'QE', 'D', 'R', 'K'),
17=c('C', 'FY', 'W', 'ML', 'IV', 'G', 'P', 'A', 'T', 'S', 'N', 'H', 'Q', 'E', 'D', 'R', 'K'),
18=c('C', 'FY', 'W', 'M', 'L', 'IV', 'G', 'P', 'A', 'T', 'S', 'N', 'H', 'Q', 'E', 'D', 'R', 'K'),
19=c('C', 'F', 'Y', 'W', 'M', 'L', 'IV', 'G', 'P', 'A', 'T', 'S', 'N', 'H', 'Q', 'E', 'D', 'R', 'K'),
20=c('C', 'F', 'Y', 'W', 'M', 'L', 'I', 'V', 'G', 'P', 'A', 'T', 'S', 'N', 'H', 'Q', 'E', 'D', 'R', 'K')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T11(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T11(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T12'>Pseudo K_tuple Reduced Amino Acid Composition Type-12 (PseKRAAC_T12)</h2><span id='topic+PseKRAAC_T12'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type12(PseKRAAC_T12) contains Grp 2-18,20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T12(
  seqs,
  type = "gap",
  Grp = 5,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T12_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T12_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T12_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T12_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T12_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T12_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
2=c('IVMLFWYC', 'ARNDQEGHKPST'),
3=c('IVLMFWC', 'YA', 'RNDQEGHKPST'),
4=c('IVLMFW', 'C', 'YA', 'RNDQEGHKPST'),
5=c('IVLMFW', 'C', 'YA', 'G', 'RNDQEHKPST'),
6=c('IVLMF', 'WY', 'C', 'AH', 'G', 'RNDQEKPST'),
7=c('IVLMF', 'WY', 'C', 'AH', 'GP', 'R', 'NDQEKST'),
8=c('IVLMF', 'WY', 'C', 'A', 'G', 'R', 'Q', 'NDEHKPST'),
9=c('IVLMF', 'WY', 'C', 'A', 'G', 'P', 'H', 'K', 'RNDQEST'),
10=c('IVLM', 'F', 'W', 'Y', 'C', 'A', 'H', 'G', 'RN', 'DQEKPST'),
11=c('IVLMF', 'W', 'Y', 'C', 'A', 'H', 'G', 'R', 'N', 'Q', 'DEKPST'),
12=c('IVLM', 'F', 'W', 'Y', 'C', 'A', 'H', 'G', 'N', 'Q', 'T', 'RDEKPS'),
13=c('IVLM', 'F', 'W', 'Y', 'C', 'A', 'H', 'G', 'N', 'Q', 'P', 'R', 'DEKST'),
14=c('IVLM', 'F', 'W', 'Y', 'C', 'A', 'H', 'G', 'N', 'Q', 'P', 'R', 'K', 'DEST'),
15=c('IVLM', 'F', 'W', 'Y', 'C', 'A', 'H', 'G', 'N', 'Q', 'P', 'R', 'K', 'D', 'EST'),
16=c('IVLM', 'F', 'W', 'Y', 'C', 'A', 'H', 'G', 'N', 'Q', 'P', 'R', 'K', 'S', 'T', 'DE'),
17=c('IVL', 'M', 'F', 'W', 'Y', 'C', 'A', 'H', 'G', 'N', 'Q', 'P', 'R', 'K', 'S', 'T', 'DE'),
18=c('IVL', 'M', 'F', 'W', 'Y', 'C', 'A', 'H', 'G', 'N', 'Q', 'P', 'R', 'K', 'S', 'T', 'D', 'E'),
20=c('I', 'V', 'L', 'M', 'F', 'W', 'Y', 'C', 'A', 'H', 'G', 'N', 'Q', 'P', 'R', 'K', 'S', 'T', 'D', 'E')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T12(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T12(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T13'>Pseudo K_tuple Reduced Amino Acid Composition Type_13 (PseKRAAC_T13)</h2><span id='topic+PseKRAAC_T13'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type13(PseKRAAC_T13) contains Grp 4,12,17,20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T13(
  seqs,
  type = "gap",
  Grp = 4,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T13_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T13_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T13_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T13_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T13_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T13_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
4=c('ADKERNTSQ', 'YFLIVMCWH', 'G', 'P'),
12=c('A', 'D', 'KER', 'N', 'TSQ', 'YF', 'LIVM', 'C', 'W', 'H', 'G', 'P'),
17=c('A', 'D', 'KE', 'R', 'N', 'T', 'S', 'Q', 'Y', 'F', 'LIV', 'M', 'C', 'W', 'H', 'G', 'P'),
20=c('A', 'D', 'K', 'E', 'R', 'N', 'T', 'S', 'Q', 'Y', 'F', 'L', 'I', 'V', 'M', 'C', 'W', 'H', 'G', 'P')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T13(seqs=filePrs,type="gap",Grp=17,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T13(seqs=filePrs,type="lambda",Grp=17,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T14'>Pseudo K_tuple Reduced Amino Acid Composition Type-14 (PseKRAAC_T14)</h2><span id='topic+PseKRAAC_T14'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type14(PseKRAAC_T14) contains Grp 2-20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T14(
  seqs,
  type = "gap",
  Grp = 2,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T14_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T14_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T14_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T14_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T14_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T14_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
2=c('ARNDCQEGHKPST', 'ILMFWYV'),
3=c('ARNDQEGHKPST', 'C', 'ILMFWYV'),
4=c('ARNDQEGHKPST', 'C', 'ILMFYV', 'W'),
5=c('AGPST', 'RNDQEHK', 'C', 'ILMFYV', 'W'),
6=c('AGPST', 'RNDQEK', 'C', 'H', 'ILMFYV', 'W'),
7=c('ANDGST', 'RQEK', 'C', 'H', 'ILMFYV', 'P', 'W'),
8=c('ANDGST', 'RQEK', 'C', 'H', 'ILMV', 'FY', 'P', 'W'),
9=c('AGST', 'RQEK', 'ND', 'C', 'H', 'ILMV', 'FY', 'P', 'W'),
10=c('AGST', 'RK', 'ND', 'C', 'QE', 'H', 'ILMV', 'FY', 'P', 'W'),
11=c('AST', 'RK', 'ND', 'C', 'QE', 'G', 'H', 'ILMV', 'FY', 'P', 'W'),
12=c('AST', 'RK', 'ND', 'C', 'QE', 'G', 'H', 'IV', 'LM', 'FY', 'P', 'W'),
13=c('AST', 'RK', 'N', 'D', 'C', 'QE', 'G', 'H', 'IV', 'LM', 'FY', 'P', 'W'),
14=c('AST', 'RK', 'N', 'D', 'C', 'Q', 'E', 'G', 'H', 'IV', 'LM', 'FY', 'P', 'W'),
15=c('A', 'RK', 'N', 'D', 'C', 'Q', 'E', 'G', 'H', 'IV', 'LM', 'FY', 'P', 'ST', 'W'),
16=c('A', 'RK', 'N', 'D', 'C', 'Q', 'E', 'G', 'H', 'IV', 'LM', 'F', 'P', 'ST', 'W', 'Y'),
17=c('A', 'R', 'N', 'D', 'C', 'Q', 'E', 'G', 'H', 'IV', 'LM', 'K', 'F', 'P', 'ST', 'W', 'Y'),
18=c('A', 'R', 'N', 'D', 'C', 'Q', 'E', 'G', 'H', 'IV', 'LM', 'K', 'F', 'P', 'S', 'T', 'W', 'Y'),
19=c('A', 'R', 'N', 'D', 'C', 'Q', 'E', 'G', 'H', 'IV', 'L', 'K', 'M', 'F', 'P', 'S', 'T', 'W', 'Y'),
20=c('A', 'R', 'N', 'D', 'C', 'Q', 'E', 'G', 'H', 'I', 'V', 'L', 'K', 'M', 'F', 'P', 'S', 'T', 'W', 'Y')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T14(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T14(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T15'>Pseudo K_tuple Reduced Amino Acid Composition Type-15 (PseKRAAC_T15)</h2><span id='topic+PseKRAAC_T15'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type15(PseKRAAC_T15) contains Grp 2-16,20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T15(
  seqs,
  type = "gap",
  Grp = 2,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T15_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T15_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T15_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T15_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T15_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T15_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
</p>
<p>Grp2=c('MFILVAW', 'CYQHPGTSNRKDE'),
Grp3=c('MFILVAW', 'CYQHPGTSNRK', 'DE'),
Grp4=c('MFILV', 'ACW', 'YQHPGTSNRK', 'DE'),
Grp5=c('MFILV', 'ACW', 'YQHPGTSN', 'RK', 'DE'),
Grp6=c('MFILV', 'A', 'C', 'WYQHPGTSN', 'RK', 'DE'),
Grp7=c('MFILV', 'A', 'C', 'WYQHP', 'GTSN', 'RK', 'DE'),
Grp8=c('MFILV', 'A', 'C', 'WYQHP', 'G', 'TSN', 'RK', 'DE'),
Grp9=c('MF', 'ILV', 'A', 'C', 'WYQHP', 'G', 'TSN', 'RK', 'DE'),
Grp10=c('MF', 'ILV', 'A', 'C', 'WYQHP', 'G', 'TSN', 'RK', 'D', 'E'),
Grp11=c('MF', 'IL', 'V', 'A', 'C', 'WYQHP', 'G', 'TSN', 'RK', 'D', 'E'),
Grp12=c('MF', 'IL', 'V', 'A', 'C', 'WYQHP', 'G', 'TS', 'N', 'RK', 'D', 'E'),
Grp13=c('MF', 'IL', 'V', 'A', 'C', 'WYQHP', 'G', 'T', 'S', 'N', 'RK', 'D', 'E'),
Grp14=c('MF', 'I', 'L', 'V', 'A', 'C', 'WYQHP', 'G', 'T', 'S', 'N', 'RK', 'D', 'E'),
Grp15=c('MF', 'IL', 'V', 'A', 'C', 'WYQ', 'H', 'P', 'G', 'T', 'S', 'N', 'RK', 'D', 'E'),
Grp16=c('MF', 'I', 'L', 'V', 'A', 'C', 'WYQ', 'H', 'P', 'G', 'T', 'S', 'N', 'RK', 'D', 'E'),
Grp20=c('M', 'F', 'I', 'L', 'V', 'A', 'C', 'W', 'Y', 'Q', 'H', 'P', 'G', 'T', 'S', 'N', 'R', 'K', 'D', 'E')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T15(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T15(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T16'>Pseudo K_tuple Reduced Amino Acid Composition Type-16 (PseKRAAC_T16)</h2><span id='topic+PseKRAAC_T16'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type16(PseKRAAC_T16) contains Grp 2-16,20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T16(
  seqs,
  type = "gap",
  Grp = 2,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T16_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T16_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T16_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T16_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T16_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T16_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
</p>
<p>2=c('IMVLFWY', 'GPCASTNHQEDRK'),
3=c('IMVLFWY', 'GPCAST', 'NHQEDRK'),
4=c('IMVLFWY', 'G', 'PCAST', 'NHQEDRK'),
5=c('IMVL', 'FWY', 'G', 'PCAST', 'NHQEDRK'),
6=c('IMVL', 'FWY', 'G', 'P', 'CAST', 'NHQEDRK'),
7=c('IMVL', 'FWY', 'G', 'P', 'CAST', 'NHQED', 'RK'),
8=c('IMV', 'L', 'FWY', 'G', 'P', 'CAST', 'NHQED', 'RK'),
9=c('IMV', 'L', 'FWY', 'G', 'P', 'C', 'AST', 'NHQED', 'RK'),
10=c('IMV', 'L', 'FWY', 'G', 'P', 'C', 'A', 'STNH', 'RKQE', 'D'),
11=c('IMV', 'L', 'FWY', 'G', 'P', 'C', 'A', 'STNH', 'RKQ', 'E', 'D'),
12=c('IMV', 'L', 'FWY', 'G', 'P', 'C', 'A', 'ST', 'N', 'HRKQ', 'E', 'D'),
13=c('IMV', 'L', 'F', 'WY', 'G', 'P', 'C', 'A', 'ST', 'N', 'HRKQ', 'E', 'D'),
14=c('IMV', 'L', 'F', 'WY', 'G', 'P', 'C', 'A', 'S', 'T', 'N', 'HRKQ', 'E', 'D'),
15=c('IMV', 'L', 'F', 'WY', 'G', 'P', 'C', 'A', 'S', 'T', 'N', 'H', 'RKQ', 'E', 'D'),
16=c('IMV', 'L', 'F', 'W', 'Y', 'G', 'P', 'C', 'A', 'S', 'T', 'N', 'H', 'RKQ', 'E', 'D'),
20=c('I', 'M', 'V', 'L', 'F', 'W', 'Y', 'G', 'P', 'C', 'A', 'S', 'T', 'N', 'H', 'R', 'K', 'Q', 'E', 'D')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T16(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T16(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T2'>Pseudo K_tuple Reduced Amino Acid Composition Type-2 (PseKRAAC_T2)</h2><span id='topic+PseKRAAC_T2'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type2(PseKRAAC_T2) contains Grp 2-6,8,15,20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T2(
  seqs,
  type = "gap",
  Grp = 2,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T2_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T2_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T2_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T2_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T2_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T2_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
</p>
<p>2=c('LVIMCAGSTPFYW', 'EDNQKRH'),
3=c('LVIMCAGSTP', 'FYW', 'EDNQKRH'),
4=c('LVIMC', 'AGSTP', 'FYW', 'EDNQKRH'),
5=c('LVIMC', 'AGSTP', 'FYW', 'EDNQ', 'KRH'),
6=c('LVIM', 'AGST', 'PHC', 'FYW', 'EDNQ', 'KR'),
8=c('LVIMC', 'AG', 'ST', 'P', 'FYW', 'EDNQ', 'KR', 'H'),
15=c('LVIM', 'C', 'A', 'G', 'S', 'T', 'P', 'FY', 'W', 'E', 'D', 'N', 'Q', 'KR', 'H'),
20=c('L', 'V', 'I', 'M', 'C', 'A', 'G', 'S', 'T', 'P', 'F', 'Y', 'W', 'E', 'D', 'N', 'Q', 'K', 'R', 'H')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T2(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T2(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T3A'>Pseudo K_tuple Reduced Amino Acid Composition Type-3A (PseKRAAC_T3A)</h2><span id='topic+PseKRAAC_T3A'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type3 contain two type: type3A and type3B. 'PseKRAAC_T3A' contains Grp 2-20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T3A(
  seqs,
  type = "gap",
  Grp = 2,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T3A_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T3A_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T3A_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T3A_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T3A_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T3A_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
Grp2=c('AGSPDEQNHTKRMILFYVC', 'W'),
Grp3=c('AGSPDEQNHTKRMILFYV', 'W', 'C'),
Grp4=c('AGSPDEQNHTKRMIV', 'W', 'YFL', 'C'),
Grp5=c('AGSPDEQNHTKR', 'W', 'YF', 'MIVL', 'C'),
Grp6=c('AGSP', 'DEQNHTKR', 'W', 'YF', 'MIL', 'VC'),
Grp7=c('AGP', 'DEQNH', 'TKRMIV', 'W', 'YF', 'L', 'CS'),
Grp8=c('AG', 'DEQN', 'TKRMIV', 'HY', 'W', 'L', 'FP', 'CS'),
Grp9=c('AG', 'P', 'DEQN', 'TKRMI', 'HY', 'W', 'F', 'L', 'VCS'),
Grp10=c('AG', 'P', 'DEQN', 'TKRM', 'HY', 'W', 'F', 'I', 'L', 'VCS'),
Grp11=c('AG', 'P', 'DEQN', 'TK', 'RI', 'H', 'Y', 'W', 'F', 'ML', 'VCS'),
Grp12=c('FAS', 'P', 'G', 'DEQ', 'NL', 'TK', 'R', 'H', 'W', 'Y', 'IM', 'VC'),
Grp13=c('FAS', 'P', 'G', 'DEQ', 'NL', 'T', 'K', 'R', 'H', 'W', 'Y', 'IM', 'VC'),
Grp14=c('FA', 'P', 'G', 'T', 'DE', 'QM', 'NL', 'K', 'R', 'H', 'W', 'Y', 'IV', 'CS'),
Grp15=c('FAS', 'P', 'G', 'T', 'DE', 'Q', 'NL', 'K', 'R', 'H', 'W', 'Y', 'M', 'I', 'VC'),
Grp16=c('FA', 'P', 'G', 'ST', 'DE', 'Q', 'N', 'K', 'R', 'H', 'W', 'Y', 'M', 'L', 'I', 'VC'),
Grp17=c('FA', 'P', 'G', 'S', 'T', 'DE', 'Q', 'N', 'K', 'R', 'H', 'W', 'Y', 'M', 'L', 'I', 'VC'),
Grp18=c('FA', 'P', 'G', 'S', 'T', 'DE', 'Q', 'N', 'K', 'R', 'H', 'W', 'Y', 'M', 'L', 'I', 'V', 'C'),
Grp19=c('FA', 'P', 'G', 'S', 'T', 'D', 'E', 'Q', 'N', 'K', 'R', 'H', 'W', 'Y', 'M', 'L', 'I', 'V', 'C'),
Grp20=c('F', 'A', 'P', 'G', 'S', 'T', 'D', 'E', 'Q', 'N', 'K', 'R', 'H', 'W', 'Y', 'M', 'L', 'I', 'V', 'C')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T3A(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T3A(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T3B'>Pseudo K_tuple Reduced Amino Acid Composition Type_3B (PseKRAAC_T3B)</h2><span id='topic+PseKRAAC_T3B'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type3 contain two type: type3A and type3B. 'PseKRAAC_T3B' contains Grp 2-20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T3B(
  seqs,
  type = "gap",
  Grp = 2,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T3B_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T3B_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T3B_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T3B_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T3B_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T3B_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
2=c('HRKQNEDSTGPACVIM', 'LFYW'),
3=c('HRKQNEDSTGPACVIM', 'LFY', 'W'),
4=c('HRKQNEDSTGPA', 'CIV', 'MLFY', 'W'),
5=c('HRKQNEDSTGPA', 'CV', 'IML', 'FY', 'W'),
6=c('HRKQNEDSTPA', 'G', 'CV', 'IML', 'FY', 'W'),
7=c('HRKQNEDSTA', 'G', 'P', 'CV', 'IML', 'FY', 'W'),
8=c('HRKQSTA', 'NED', 'G', 'P', 'CV', 'IML', 'FY', 'W'),
9=c('HRKQ', 'NED', 'ASTG', 'P', 'C', 'IV', 'MLF', 'Y', 'W'),
10=c('RKHSA', 'Q', 'NED', 'G', 'P', 'C', 'TIV', 'MLF', 'Y', 'W'),
11=c('RKQ', 'NG', 'ED', 'AST', 'P', 'C', 'IV', 'HML', 'F', 'Y', 'W'),
12=c('RKQ', 'ED', 'NAST', 'G', 'P', 'C', 'IV', 'H', 'ML', 'F', 'Y', 'W'),
13=c('RK', 'QE', 'D', 'NG', 'HA', 'ST', 'P', 'C', 'IV', 'ML', 'F', 'Y', 'W'),
14=c('R', 'K', 'QE', 'D', 'NG', 'HA', 'ST', 'P', 'C', 'IV', 'ML', 'F', 'Y', 'W'),
15=c('R', 'K', 'QE', 'D', 'NG', 'HA', 'ST', 'P', 'C', 'IV', 'M', 'L', 'F', 'Y', 'W'),
16=c('R', 'K', 'Q', 'E', 'D', 'NG', 'HA', 'ST', 'P', 'C', 'IV', 'M', 'L', 'F', 'Y', 'W'),
17=c('R', 'K', 'Q', 'E', 'D', 'NG', 'HA', 'S', 'T', 'P', 'C', 'IV', 'M', 'L', 'F', 'Y', 'W'),
18=c('R', 'K', 'Q', 'E', 'D', 'NG', 'HA', 'S', 'T', 'P', 'C', 'I', 'V', 'M', 'L', 'F', 'Y', 'W'),
19=c('R', 'K', 'Q', 'E', 'D', 'NG', 'H', 'A', 'S', 'T', 'P', 'C', 'I', 'V', 'M', 'L', 'F', 'Y', 'W'),
20=c('R', 'K', 'Q', 'E', 'D', 'N', 'G', 'H', 'A', 'S', 'T', 'P', 'C', 'I', 'V', 'M', 'L', 'F', 'Y', 'W')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T3B(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T3B(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T4'>Pseudo K_tuple Reduced Amino Acid Composition Type-4 (PseKRAAC_T4)</h2><span id='topic+PseKRAAC_T4'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type4(PseKRAAC_T4) contains Grp 5,8,9,11,13,20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T4(
  seqs,
  type = "gap",
  Grp = 5,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T4_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T4_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T4_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T4_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T4_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T4_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
5=c('G', 'IVFYW', 'ALMEQRK', 'P', 'NDHSTC'),
8=c('G', 'IV', 'FYW', 'ALM', 'EQRK', 'P', 'ND', 'HSTC'),
9=c('G', 'IV', 'FYW', 'ALM', 'EQRK', 'P', 'ND', 'HS', 'TC'),
11=c('G', 'IV', 'FYW', 'A', 'LM', 'EQRK', 'P', 'ND', 'HS', 'T', 'C'),
13=c('G', 'IV', 'FYW', 'A', 'L', 'M', 'E', 'QRK', 'P', 'ND', 'HS', 'T', 'C'),
20=c('G', 'I', 'V', 'F', 'Y', 'W', 'A', 'L', 'M', 'E', 'Q', 'R', 'K', 'P', 'N', 'D', 'H', 'S', 'T', 'C')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T4(seqs=filePrs,type="gap",Grp=8,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T4(seqs=filePrs,type="lambda",Grp=8,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T5'>Pseudo K_tuple Reduced Amino Acid Composition Type-5 (PseKRAAC_T5)</h2><span id='topic+PseKRAAC_T5'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type5(PseKRAAC_T5) contains Grp 3,4,8,10,15,20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T5(
  seqs,
  type = "gap",
  Grp = 4,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T5_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T5_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T5_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T5_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T5_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T5_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
3=c('FWYCILMVAGSTPHNQ', 'DE', 'KR'),
4=c('FWY', 'CILMV', 'AGSTP', 'EQNDHKR'),
8=c('FWY', 'CILMV', 'GA', 'ST', 'P', 'EQND', 'H', 'KR'),
10=c('G', 'FYW', 'A', 'ILMV', 'RK', 'P', 'EQND', 'H', 'ST', 'C'),
15=c('G', 'FY', 'W', 'A', 'ILMV', 'E', 'Q', 'RK', 'P', 'N', 'D', 'H', 'S', 'T', 'C'),
20=c('G', 'I', 'V', 'F', 'Y', 'W', 'A', 'L', 'M', 'E', 'Q', 'R', 'K', 'P', 'N', 'D', 'H', 'S', 'T', 'C')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T5(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T5(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T6A'>Pseudo K_tuple Reduced Amino Acid Composition Type-6A (PseKRAAC_T6A)</h2><span id='topic+PseKRAAC_T6A'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type6 contain two type: type6A and type6B. 'PseKRAAC_T6A' contains Grp 4,5,20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T6A(
  seqs,
  type = "gap",
  Grp = 5,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T6A_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T6A_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T6A_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T6A_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T6A_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T6A_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
4=c('AGPST', 'CILMV', 'DEHKNQR', 'FYW'),
5=c('AHT', 'CFILMVWY', 'DE', 'GP', 'KNQRS'),
20=c('A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T6A(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T6A(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T6B'>Pseudo K_tuple Reduced Amino Acid Composition Type-6B (PseKRAAC_T6B)</h2><span id='topic+PseKRAAC_T6B'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type6 contain two type: type6A and type6B. 'PseKRAAC_T6B' contains Grp 5.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T6B(
  seqs,
  type = "gap",
  Grp = 5,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T6B_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T6B_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T6B_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T6B_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T6B_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T6B_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
5=c('AEHKQRST', 'CFILMVWY', 'DN', 'G', 'P')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T6B(seqs=filePrs,type="gap",Grp=5,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T6B(seqs=filePrs,type="lambda",Grp=5,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T7'>Pseudo K_tuple Reduced Amino Acid Composition Type-7 (PseKRAAC_T7)</h2><span id='topic+PseKRAAC_T7'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type7(PseKRAAC_T7) contains Grp 2-20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T7(
  seqs,
  type = "gap",
  Grp = 5,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T7_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T7_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T7_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T7_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T7_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T7_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
Grp2=c('C', 'MFILVWYAGTSNQDEHRKP'),
Grp3=c('C', 'MFILVWYAKR', 'GTSNQDEHP'),
Grp4=c('C', 'KR', 'MFILVWYA', 'GTSNQDEHP'),
Grp5=c('C', 'KR', 'MFILVWYA', 'DE', 'GTSNQHP'),
Grp6=c('C', 'KR', 'WYA', 'MFILV', 'DE', 'GTSNQHP'),
Grp7=c('C', 'KR', 'WYA', 'MFILV', 'DE', 'QH', 'GTSNP'),
Grp8=c('C', 'KR', 'WYA', 'MFILV', 'D', 'E', 'QH', 'GTSNP'),
Grp9=c('C', 'KR', 'WYA', 'MFILV', 'D', 'E', 'QH', 'TP', 'GSN'),
Grp10=c('C', 'KR', 'WY', 'A', 'MFILV', 'D', 'E', 'QH', 'TP', 'GSN'),
Grp11=c('C', 'K', 'R', 'WY', 'A', 'MFILV', 'D', 'E', 'QH', 'TP', 'GSN'),
Grp12=c('C', 'K', 'R', 'WY', 'A', 'MFILV', 'D', 'E', 'QH', 'TP', 'GS', 'N'),
Grp13=c('C', 'K', 'R', 'W', 'Y', 'A', 'MFILV', 'D', 'E', 'QH', 'TP', 'GS', 'N'),
Grp14=c('C', 'K', 'R', 'W', 'Y', 'A', 'FILV', 'M', 'D', 'E', 'QH', 'TP', 'GS', 'N'),
Grp15=c('C', 'K', 'R', 'W', 'Y', 'A', 'FILV', 'M', 'D', 'E', 'Q', 'H', 'TP', 'GS', 'N'),
Grp16=c('C', 'K', 'R', 'W', 'Y', 'A', 'FILV', 'M', 'D', 'E', 'Q', 'H', 'TP', 'G', 'S', 'N'),
Grp17=c('C', 'K', 'R', 'W', 'Y', 'A', 'FI', 'LV', 'M', 'D', 'E', 'Q', 'H', 'TP', 'G', 'S', 'N'),
Grp18=c('C', 'K', 'R', 'W', 'Y', 'A', 'FI', 'LV', 'M', 'D', 'E', 'Q', 'H', 'T', 'P', 'G', 'S', 'N'),
Grp19=c('C', 'K', 'R', 'W', 'Y', 'A', 'F', 'I', 'LV', 'M', 'D', 'E', 'Q', 'H', 'T', 'P', 'G', 'S', 'N'),
Grp20=c('C', 'K', 'R', 'W', 'Y', 'A', 'F', 'I', 'L', 'V', 'M', 'D', 'E', 'Q', 'H', 'T', 'P', 'G', 'S', 'N')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T7(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T7(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T8'>Pseudo K_tuple Reduced Amino Acid Composition Type-8 (PseKRAAC_T8)</h2><span id='topic+PseKRAAC_T8'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type8(PseKRAAC_T8) contains Grp 2-20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T8(
  seqs,
  type = "gap",
  Grp = 5,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T8_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T8_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T8_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T8_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T8_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T8_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
Grp2=c('ADEGKNPQRST', 'CFHILMVWY'),
Grp3=c('ADEGNPST', 'CHKQRW', 'FILMVY'),
Grp4=c('AGNPST', 'CHWY', 'DEKQR', 'FILMV'),
Grp5=c('AGPST', 'CFWY', 'DEN', 'HKQR', 'ILMV'),
Grp6=c('APST', 'CW', 'DEGN', 'FHY', 'ILMV', 'KQR'),
Grp7=c('AGST', 'CW', 'DEN', 'FY', 'HP', 'ILMV', 'KQR'),
Grp8=c('AST', 'CG', 'DEN', 'FY', 'HP', 'ILV', 'KQR', 'MW'),
Grp9=c('AST', 'CW', 'DE', 'FY', 'GN', 'HQ', 'ILV', 'KR', 'MP'),
Grp10=c('AST', 'CW', 'DE', 'FY', 'GN', 'HQ', 'IV', 'KR', 'LM', 'P'),
Grp11=c('AST', 'C', 'DE', 'FY', 'GN', 'HQ', 'IV', 'KR', 'LM', 'P', 'W'),
Grp12=c('AST', 'C', 'DE', 'FY', 'G', 'HQ', 'IV', 'KR', 'LM', 'N', 'P', 'W'),
Grp13=c('AST', 'C', 'DE', 'FY', 'G', 'H', 'IV', 'KR', 'LM', 'N', 'P', 'Q', 'W'),
Grp14=c('AST', 'C', 'DE', 'FL', 'G', 'H', 'IV', 'KR', 'M', 'N', 'P', 'Q', 'W', 'Y'),
Grp15=c('AST', 'C', 'DE', 'F', 'G', 'H', 'IV', 'KR', 'L', 'M', 'N', 'P', 'Q', 'W', 'Y'),
Grp16=c('AT', 'C', 'DE', 'F', 'G', 'H', 'IV', 'KR', 'L', 'M', 'N', 'P', 'Q', 'S', 'W', 'Y'),
Grp17=c('AT', 'C', 'DE', 'F', 'G', 'H', 'IV', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'W', 'Y'),
Grp18=c('A', 'C', 'DE', 'F', 'G', 'H', 'IV', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'W', 'Y'),
Grp19=c('A', 'C', 'D', 'E', 'F', 'G', 'H', 'IV', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'W', 'Y'),
Grp20=c('A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'V', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'W', 'Y')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T8(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T8(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PseKRAAC_T9'>Pseudo K_tuple Reduced Amino Acid Composition Type-9 (PseKRAAC_T9)</h2><span id='topic+PseKRAAC_T9'></span>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type9(PseKRAAC_T9) contains Grp 2-20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PseKRAAC_T9(
  seqs,
  type = "gap",
  Grp = 5,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PseKRAAC_T9_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T9_+3A_type">type</code></td>
<td>
<p>This parameter has two valid value &quot;lambda&quot; and &quot;gap&quot;. &quot;lambda&quot;
calls lambda_model function and &quot;gap&quot; calls gap_model function.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T9_+3A_grp">Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T9_+3A_gaporlambdavalue">GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T9_+3A_k">k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td></tr>
<tr><td><code id="PseKRAAC_T9_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Groups:
Grp2=c('ADEGKNPQRST', 'CFHILMVWY'),
Grp3=c('ADEGNPST', 'CHKQRW', 'FILMVY'),
Grp4=c('AGNPST', 'CHWY', 'DEKQR', 'FILMV'),
Grp5=c('AGPST', 'CFWY', 'DEN', 'HKQR', 'ILMV'),
Grp6=c('APST', 'CW', 'DEGN', 'FHY', 'ILMV', 'KQR'),
Grp7=c('AGST', 'CW', 'DEN', 'FY', 'HP', 'ILMV', 'KQR'),
Grp8=c('AST', 'CG', 'DEN', 'FY', 'HP', 'ILV', 'KQR', 'MW'),
Grp9=c('AST', 'CW', 'DE', 'FY', 'GN', 'HQ', 'ILV', 'KR', 'MP'),
Grp10=c('AST', 'CW', 'DE', 'FY', 'GN', 'HQ', 'IV', 'KR', 'LM', 'P'),
Grp11=c('AST', 'C', 'DE', 'FY', 'GN', 'HQ', 'IV', 'KR', 'LM', 'P', 'W'),
Grp12=c('AST', 'C', 'DE', 'FY', 'G', 'HQ', 'IV', 'KR', 'LM', 'N', 'P', 'W'),
Grp13=c('AST', 'C', 'DE', 'FY', 'G', 'H', 'IV', 'KR', 'LM', 'N', 'P', 'Q', 'W'),
Grp14=c('AST', 'C', 'DE', 'FL', 'G', 'H', 'IV', 'KR', 'M', 'N', 'P', 'Q', 'W', 'Y'),
Grp15=c('AST', 'C', 'DE', 'F', 'G', 'H', 'IV', 'KR', 'L', 'M', 'N', 'P', 'Q', 'W', 'Y'),
Grp16=c('AT', 'C', 'DE', 'F', 'G', 'H', 'IV', 'KR', 'L', 'M', 'N', 'P', 'Q', 'S', 'W', 'Y'),
Grp17=c('AT', 'C', 'DE', 'F', 'G', 'H', 'IV', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'W', 'Y'),
Grp18=c('A', 'C', 'DE', 'F', 'G', 'H', 'IV', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'W', 'Y'),
Grp19=c('A', 'C', 'D', 'E', 'F', 'G', 'H', 'IV', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'W', 'Y'),
Grp20=c('A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'V', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'W', 'Y')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. &quot;PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition.&quot; Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T9(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T9(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>

<hr>
<h2 id='PSSM'>Position-Specific Scoring Matrix (PSSM)</h2><span id='topic+PSSM'></span>

<h3>Description</h3>

<p>This functions receives as input PSSM matrices (which are created by PSI-BLAST software)
and converts them into feature vectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PSSM(dirPath, outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PSSM_+3A_dirpath">dirPath</code></td>
<td>
<p>Path of the directory which contains all output files of PSI-BLAST. Each file belongs to a sequence.</p>
</td></tr>
<tr><td><code id="PSSM_+3A_outformat">outFormat</code></td>
<td>
<p>It can take two values: 'mat' (which stands for matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="PSSM_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>It shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length)*(20)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
ad&lt;-paste0(dir,"/pssm.txt")

PSSMdir&lt;-system.file("testForder",package="ftrCOOL")
PSSMdir&lt;-paste0(PSSMdir,"/PSSMdir/")
mat&lt;-PSSM(PSSMdir,outFormat="txt",outputFileDist=ad)


unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='PSTNPds'>Position-Specific Trinucleotide Propensity based on double-strand (PSTNPds)</h2><span id='topic+PSTNPds'></span>

<h3>Description</h3>

<p>This function works like <a href="#topic+PSTNPss_DNA">PSTNPss_DNA</a> except that it considers T as A and G as C. So it
converts Ts in the sequence to A and Gs to C. Then, it works with 2 alphabets A and C.
For more details refer to <a href="#topic+PSTNPss_DNA">PSTNPss_DNA</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PSTNPds(seqs, pos, neg, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PSTNPds_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="PSTNPds_+3A_pos">pos</code></td>
<td>
<p>is a fasta file containing nucleotide sequences. Each sequence starts
with '&gt;'. Also, the value of this parameter can be a string vector.
The sequences are positive sequences in the training model.</p>
</td></tr>
<tr><td><code id="PSTNPds_+3A_neg">neg</code></td>
<td>
<p>is a fasta file containing nucleotide sequences. Each sequence starts
with '&gt;'. Also, the value of this parameter can be a string vector.
The sequences are negative sequences in the training model.</p>
</td></tr>
<tr><td><code id="PSTNPds_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equal to the number of sequences.
It shows the class of each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a feature matrix. The number of columns is equal to the length of sequences minus two
and the number of rows is equal to the number of sequences.
</p>


<h3>Note</h3>

<p>The length of the sequences in positive and negative data sets and the input sets
should be equal.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. &quot;iLearn: an integrated platform and meta-learner for feature engineering, machine-learning analysis and modeling of DNA, RNA and protein sequence data.&quot; Briefings in bioinformatics 21.3 (2020): 1047-1057.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")

posSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/posData.txt"),alphabet="dna")
negSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/negData.txt"),alphabet="dna")
seqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/testData.txt"),alphabet="dna")


PSTNPds(seqs=seqs,pos=posSeqs[1],neg=negSeqs[1])



</code></pre>

<hr>
<h2 id='PSTNPss_DNA'>Position-Specific Trinucleotide Propensity based on single-strand DNA (PSTNPss_DNA)</h2><span id='topic+PSTNPss_DNA'></span>

<h3>Description</h3>

<p>The inputs to this function are positive and negative data sets and a set of sequences.
The output of the function is a matrix of feature vectors.
The number of rows of the output matrix is equal to the number of sequences.
The feature vector for an input sequence with length L is [u(1),u(2),...u(L-2)].
For each input sequence, u(1) is calculated by subtracting the frequency of sequences
(which start with the same trinucleotides as the input sequence) in the positive set
with those starting with the same trinucleotide in the negative set.
We compute u(i) like u(1) with the exception that
instead of the first trinucleotide, the ith trinucletide is considered.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PSTNPss_DNA(seqs, pos, neg, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PSTNPss_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="PSTNPss_DNA_+3A_pos">pos</code></td>
<td>
<p>is a fasta file containing nucleotide sequences. Each sequence starts
with '&gt;'. Also, the value of this parameter can be a string vector.
The sequences are positive sequences in the training model.</p>
</td></tr>
<tr><td><code id="PSTNPss_DNA_+3A_neg">neg</code></td>
<td>
<p>is a fasta file containing nucleotide sequences. Each sequence starts
with '&gt;'. Also, the value of this parameter can be a string vector.</p>
</td></tr>
<tr><td><code id="PSTNPss_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equal to the number of sequences.
It shows the class of each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a feature matrix. The number of columns is equal to the length of sequences minus two
and the number of rows is equal to the number of sequences.
</p>


<h3>Note</h3>

<p>The length of the sequences in positive and negative data sets and the input sets
should be equal.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")

posSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/posDNA.txt"),alphabet="dna")
negSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/negDNA.txt"),alphabet="dna")
seqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/DNA_testing.txt"),alphabet="dna")


mat=PSTNPss_DNA(seqs=seqs,pos=posSeqs,neg=negSeqs)



</code></pre>

<hr>
<h2 id='PSTNPss_RNA'>Position-Specific Tri-ribonucleotide Propensity based on single-strand RNA (PSTNPss_RNA)</h2><span id='topic+PSTNPss_RNA'></span>

<h3>Description</h3>

<p>The inputs to this function are positive and negative data sets and a set of sequences.
The output of the function is a matrix of feature vectors.
The number of rows of the output matrix is equal to the number of sequences.
The feature vector for an input sequence with length L is [u(1),u(2),...u(L-2)].
For each input sequence, u(1) is calculated by subtracting the frequency of sequences
(which start with the same tri-ribonucleotides as the input sequence) in the positive set
with those starting with the same tri-ribonucleotide in the negative set.
We compute u(i) like u(1) with the exception that
instead of the first tri-ribonucleotide, the ith tri-ribonucletide is considered.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PSTNPss_RNA(seqs, pos, neg, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PSTNPss_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="PSTNPss_RNA_+3A_pos">pos</code></td>
<td>
<p>is a fasta file containing ribonucleotide sequences. Each sequence starts
with '&gt;'. Also, the value of this parameter can be a string vector.
The sequences are positive sequences in the training model</p>
</td></tr>
<tr><td><code id="PSTNPss_RNA_+3A_neg">neg</code></td>
<td>
<p>is a fasta file containing ribonucleotide sequences. Each sequence starts
with '&gt;'. Also, the value of this parameter can be a string vector.</p>
</td></tr>
<tr><td><code id="PSTNPss_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equal to the number of sequences.
It shows the class of each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a feature matrix. The number of columns is equal to the length of sequences minus two
and the number of rows is equal to the number of sequences.
</p>


<h3>Note</h3>

<p>The length of the sequences in positive and negative data sets and the input sets
should be equal.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")


posSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/pos2RNA.txt"),alphabet="rna")
negSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/neg2RNA.txt"),alphabet="rna")
seqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/testSeq2RNA.txt"),alphabet="rna")


PSTNPss_RNA(seqs=seqs,pos=posSeqs,neg=negSeqs)



</code></pre>

<hr>
<h2 id='QSOrder'>Quasi Sequence Order (QSOrder)</h2><span id='topic+QSOrder'></span>

<h3>Description</h3>

<p>This function computes the quasi-sequence-order for sequences.
It is for amino acid pairs with d distances (d can be any number between 1 and 20).
First, it calculates the frequencies of each amino acid (&quot;A&quot;, &quot;C&quot;,..., &quot;Y&quot;).
Then, it normalizes the frequencies by dividing the frequency of an amino acid to the frequency of all amino acids
plus the sum of tau values which is multiplied by W. tau values are given by function <a href="#topic+SOCNumber">SOCNumber</a>.
For d bigger than 20, it computes tau for d in the range &quot;1 to (nlag-20) * W&quot; and normalizes them like before.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QSOrder(seqs, nlag = 25, W = 0.1, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QSOrder_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="QSOrder_+3A_nlag">nlag</code></td>
<td>
<p>is a numeric value which shows the maximum distance between two amino acids.
Distances can be 1, 2, ..., or nlag.</p>
</td></tr>
<tr><td><code id="QSOrder_+3A_w">W</code></td>
<td>
<p>(weight) is a tuning parameter.</p>
</td></tr>
<tr><td><code id="QSOrder_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please find details about tau in function <a href="#topic+SOCNumber">SOCNumber</a>.
</p>


<h3>Value</h3>

<p>It returns a feature matrix which the number of rows equals to the number of sequences and the number of columns is (nlag*2).
For each distance d, there are two values. One value for Granthman and another one for Schneider distance.
</p>


<h3>Note</h3>

<p>For d between 21 to nlag, the function calculates tau values for (d-20) to (nlag-20).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat&lt;-QSOrder(seqs=filePrs,nlag=25)

</code></pre>

<hr>
<h2 id='readASAdir'>Read Directory of Accessible Solvent accessibility predicted files (readASAdir)</h2><span id='topic+readASAdir'></span>

<h3>Description</h3>

<p>This function reads a directory that contains the output files of SPINE-X.
It gets the directory path as the input and returns a list of vectors. Each vector includes the ASA predicted value for amino acids of the sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readASAdir(dirPath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readASAdir_+3A_dirpath">dirPath</code></td>
<td>
<p>path of the directory which contains all the output files of SPINE-X. Each file belongs to a sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of vectors with all the predicted ASA value for each amino acid. The length of the
list is the number of files(sequences) and the length of each vector is (length of sequence(i))
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


PredASAdir&lt;-system.file("testForder",package="ftrCOOL")
PredASAdir&lt;-paste0(PredASAdir,"/ASAdir/")
PredVectASA&lt;-readASAdir(PredASAdir)

</code></pre>

<hr>
<h2 id='readDisDir'>Read disorder predicted Directory (readDisDir)</h2><span id='topic+readDisDir'></span>

<h3>Description</h3>

<p>This function reads a directory that contains the output VSL2 files.
It gets the directory path as the input and returns a list of vectors. Each vector includes the disorder/order type for the amino acids of the sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readDisDir(dirPath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readDisDir_+3A_dirpath">dirPath</code></td>
<td>
<p>the path of a directory which contains all the VSL2 output files.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of vectors with all the predicted disorder/order type for each amino acid. The length of the
list is equal to the number of files(sequences) and the length of each vector is the length of the sequence(i).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
PredDisdir&lt;-system.file("testForder",package="ftrCOOL")
PredDisdir&lt;-paste0(PredDisdir,"/Disdir/")
listPredVect&lt;-readDisDir(PredDisdir)

</code></pre>

<hr>
<h2 id='readPSSMdir'>Read PSSM Directory (readPSSMdir)</h2><span id='topic+readPSSMdir'></span>

<h3>Description</h3>

<p>This function reads a directory that contains the output psi-blast.
It gets the directory path as the input and returns a list of vectors. Each vector includes the type for the amino acids of the sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readPSSMdir(dirPath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readPSSMdir_+3A_dirpath">dirPath</code></td>
<td>
<p>the path of a directory which contains all the VSL2 output files.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of vectors with all the predicted disorder/order type for each amino acid. The length of the
list is equal to the number of files(sequences) and the length of each vector is the length of the sequence(i).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
pssmDir&lt;-system.file("testForder",package="ftrCOOL")
pssmDir&lt;-paste0(pssmDir,"/PSSMdir/")
listPredVect&lt;-readPSSMdir(pssmDir)

</code></pre>

<hr>
<h2 id='readss2Dir'>Read ss2 predicted Directory (readss2Dir)</h2><span id='topic+readss2Dir'></span>

<h3>Description</h3>

<p>This function reads a directory that contains the output files of PSIPRED
It gets the directory path as the input and returns a list of vectors. Each vector contains the secondary structure of the amino acids in a peptide/protein sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readss2Dir(dirPath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readss2Dir_+3A_dirpath">dirPath</code></td>
<td>
<p>The path of the directory which contains all predss2 files. Each file belongs to a sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a list of vectors with all the predicted secondary structure for each amino acid. The length of the
list is the number of files(sequences) and the length of each vector is (length sequence(i))
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

PredSS2dir&lt;-system.file("testForder",package="ftrCOOL")
PredSS2dir&lt;-paste0(PredSS2dir,"/ss2Dir/")
listPredVect&lt;-readss2Dir(PredSS2dir)

</code></pre>

<hr>
<h2 id='readTorsionDir'>Read Directory of Torsion predicted files (readTorsionDir)</h2><span id='topic+readTorsionDir'></span>

<h3>Description</h3>

<p>This function reads a directory that contains the output files of SPINE-X.
It gets the directory path as the input and returns a list of vectors. Each vector includes the phi and psi angle
of the amino acids of the sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readTorsionDir(dirPath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readTorsionDir_+3A_dirpath">dirPath</code></td>
<td>
<p>The path of the directory which contains all output files of SPINE-X. Each file belongs to a sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a list of vectors with all the predicted phi and psi angles for each amino acid. The length of the
list is the number of files(sequences) and the length of each vector is (2(phi-psi)*length sequence(i)).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

PredTorsioNdir&lt;-system.file("testForder",package="ftrCOOL")
PredTorsioNdir&lt;-paste0(PredTorsioNdir,"/TorsioNdir/")
PredVectASA&lt;-readTorsionDir(PredTorsioNdir)

</code></pre>

<hr>
<h2 id='revComp'>reverseCompelement (revComp)</h2><span id='topic+revComp'></span>

<h3>Description</h3>

<p>This function returns the reverse compelement of a dna sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>revComp(seq, outputType = "str")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="revComp_+3A_seq">seq</code></td>
<td>
<p>is a dna sequence.</p>
</td></tr>
<tr><td><code id="revComp_+3A_outputtype">outputType</code></td>
<td>
<p>this parameter can take two values: 'char' or 'str'. If outputType is
'str', the reverse complement sequence of the input sequence is returned as a string.
Otherwise, a vector of characters which represent the reverse complement is returned.
Default value is 'str'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The reverse complement of the input sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
Seq&lt;-ptmSeqsVect[1]
revCompSeq&lt;-revComp(seq=Seq,outputType="char")
</code></pre>

<hr>
<h2 id='SAAC'>Splitted Amino Acid Composition (SAAC)</h2><span id='topic+SAAC'></span>

<h3>Description</h3>

<p>This function splits the input sequence into three parts. The first part is N-terminal
and the third part is C-terminal and middle part contains all amino acids between these two part.
N-terminal will be determined by the first numNterm amino acid in the sequences and C-terminal is
determined by numCterm of the last amino acids in the sequence. Users should enter numNterm and numCterm
parameters. Their default value is 25.
The function calculates <a href="#topic+kAAComposition">kAAComposition</a> for each of the three parts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SAAC(seqs, k = 1, numNterm = 5, numCterm = 5, normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SAAC_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="SAAC_+3A_k">k</code></td>
<td>
<p>shows which type of amino acid composition applies to the parts.
For example, the amino acid composition is applied when k=1 and when k=2, the dipeptide Composition is applied.</p>
</td></tr>
<tr><td><code id="SAAC_+3A_numnterm">numNterm</code></td>
<td>
<p>shows how many amino acids should be considered for N-terminal.</p>
</td></tr>
<tr><td><code id="SAAC_+3A_numcterm">numCterm</code></td>
<td>
<p>shows how many amino acids should be considered for C-terminal.</p>
</td></tr>
<tr><td><code id="SAAC_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="SAAC_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a feature matrix. The number of rows is equal to the number of sequences.
The number of columns is (3*(20^k)).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-SAAC(seqs=filePrs,k=1,numNterm=15,numCterm=15)
</code></pre>

<hr>
<h2 id='SGAAC'>Splitted Group Amino Acid Composition (SGAAC)</h2><span id='topic+SGAAC'></span>

<h3>Description</h3>

<p>In this function, amino acids are first grouped into a user-defined category.
Later, the splitted amino Acid composition is computed.
Please note that this function differs from <a href="#topic+SAAC">SAAC</a> which works on individual amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SGAAC(
  seqs,
  k = 1,
  numNterm = 25,
  numCterm = 25,
  Grp = "locFus",
  normalized = TRUE,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SGAAC_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="SGAAC_+3A_k">k</code></td>
<td>
<p>shows which type of amino acid composition applies to the parts.
For example, the amino acid composition is applied when k=1 and when k=2, the dipeptide Composition is applied.</p>
</td></tr>
<tr><td><code id="SGAAC_+3A_numnterm">numNterm</code></td>
<td>
<p>shows how many amino acids should be considered for N-terminal.</p>
</td></tr>
<tr><td><code id="SGAAC_+3A_numcterm">numCterm</code></td>
<td>
<p>shows how many amino acids should be considered for C-terminal.</p>
</td></tr>
<tr><td><code id="SGAAC_+3A_grp">Grp</code></td>
<td>
<p>is a list of vectors containig amino acids. Each vector represents a category. Users can define a customized amino acid grouping, provided that the sum of all amino acids is 20 and there is no repeated amino acid in the groups.
Also, users can choose 'cTriad'(conjointTriad), 'locFus', or 'aromatic'. Each option provides specific information about the type of an amino acid grouping.</p>
</td></tr>
<tr><td><code id="SGAAC_+3A_normalized">normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td></tr>
<tr><td><code id="SGAAC_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a feature matrix. The number of rows is equal to the number of sequences.
The number of columns is 3*((number of groups)^k).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-SGAAC(seqs=filePrs,k=1,numNterm=15,numCterm=15,Grp="aromatic")

</code></pre>

<hr>
<h2 id='SOCNumber'>Sequence Order Coupling Number (SOCNumber)</h2><span id='topic+SOCNumber'></span>

<h3>Description</h3>

<p>This function uses dissimilarity matrices Grantham and Schneider to compute the dissimilarity between amino acid pairs.
The distance between amino acid pairs is determined by d which varies between 1 to nlag.
For each d, it computes the sum of the dissimilarities of all amino acid pairs. The sum shows the value of tau for a value d.
The feature vector contains the values of taus for both matrices. Thus, the length of the feature vector is equal to nlag*2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SOCNumber(seqs, nlag = 30, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SOCNumber_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="SOCNumber_+3A_nlag">nlag</code></td>
<td>
<p>is a numeric value which shows the maximum distance between two amino acids.
Distances can be 1, 2, ..., or nlag. Defult is 30.</p>
</td></tr>
<tr><td><code id="SOCNumber_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (nlag*2). For each distance d, there are two values. One value for Granthman and another one for Schneider distance.
</p>


<h3>Note</h3>

<p>When d=1, the pairs of amino acids have no gap and when d=2, there is one gap between the amino acid pairs in the sequence.
It will repeat likewise for other values of d.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat&lt;-SOCNumber(seqs=filePrs,nlag=25)
</code></pre>

<hr>
<h2 id='SSEB'>Secondary Structure Elements Binary (SSEB)</h2><span id='topic+SSEB'></span>

<h3>Description</h3>

<p>This function works based on the output of PSIPRED which predicts the secondary structure of the amino acids in a sequence.
The output of the PSIPRED is a tab-delimited file which contains the secondary structure in the third column.
SSEB gives a binary number (i.e., '001'='H','010'=E','100'='C') for each amino acid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SSEB(dirPath, binaryType = "numBin", outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SSEB_+3A_dirpath">dirPath</code></td>
<td>
<p>Path of the directory which contains all output files of PSIPRED. Each file belongs to a sequence.</p>
</td></tr>
<tr><td><code id="SSEB_+3A_binarytype">binaryType</code></td>
<td>
<p>It can take any of the following values: ('strBin','logicBin','numBin').
'strBin'(String binary): each structure is represented by a string containing 3 characters(0-1). Helix  = &quot;001&quot; , Extended = &quot;010&quot; , coil = &quot;100&quot;.
'logicBin'(logical value): Each structure is represented by a vector containing 3 logical entries. Helix  = c(FALSE,FALSE,TRUE) , Extended = c(FALSE,TRUE,FALSE) , Coil = c(TRUE,FALSE,FALSE).
'numBin' (numeric bin): Each structure is represented by a numeric (i.e., integer) vector containing 3 numerals. Helix  = c(0,0,1) , Extended = c(0,1,0) , coil = c(1,0,0).</p>
</td></tr>
<tr><td><code id="SSEB_+3A_outformat">outFormat</code></td>
<td>
<p>It can take two values: 'mat' (which stands for matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="SSEB_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>It shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function converts each amino acid to a 3-bit value, such that 2 bits are 0 and
1 bit is 1. The position of 1 shows the type of the secondary structure of the amino acids in the protein/peptide.
In this function, '001' is used to show Helix structure, '010' to show Extended structure and '100' to show coil structure.
</p>


<h3>Value</h3>

<p>The output is different depending on the outFormat parameter ('mat' or 'txt').
If outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and if binaryType is 'strBin', the number of columns is the length of the sequences.
Otherwise, it is equal to (length of the sequences)*3.
If outFormat is 'txt', all binary values will be written to a tab-delimited file. Each line in the file shows the binary format of a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in the outFormat parameter for sequences with different lengths.
Warning: If the outFormat is set to 'mat' for sequences with different lengths, it returns an error.
It is noteworthy that 'txt' format is not usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
ad&lt;-paste0(dir,"/SSEB.txt")

Predss2dir&lt;-system.file("testForder",package="ftrCOOL")
Predss2dir&lt;-paste0(Predss2dir,"/ss2Dir/")
mat&lt;-SSEB(Predss2dir,binaryType="numBin",outFormat="txt",outputFileDist=ad)

unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='SSEC'>Secondary Structure Elements Composition (SSEC)</h2><span id='topic+SSEC'></span>

<h3>Description</h3>

<p>This function works based on the output of PSIPRED which predicts the secondary structure of the amino acids in a sequence.
The output of the PSIPRED is a tab-delimited file which contains the secondary structure in the third column.
SSEC returns the frequency of the secondary structures (i.e., Helix, Extended, Coil) of the sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SSEC(dirPath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SSEC_+3A_dirpath">dirPath</code></td>
<td>
<p>Path of the directory which contains all output files of PSIPRED. Each file belongs to a sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a feature matrix which the number of rows is the number of sequences and
the number of columns is 3. The first column shows the number of amino acids which participate
in the coil structure. The second column shows the number of amino acids in the extended structure
and the last column shows the number of amino acids in the helix structure.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

Predss2dir&lt;-system.file("testForder",package="ftrCOOL")
Predss2dir&lt;-paste0(Predss2dir,"/ss2Dir/")
mat&lt;-SSEC(Predss2dir)

</code></pre>

<hr>
<h2 id='SSES'>Secondary Structure Elements Simple (SSES)</h2><span id='topic+SSES'></span>

<h3>Description</h3>

<p>This function works based on the output of PSIPRED which predicts the secondary structure of the amino acids in a sequence.
The output of the PSIPRED is a tab-delimited file which contains the secondary structure in the third column.
The function represent amino acids in the helix structure by 'H', amino acids in the extended structure by 'E', and amino acids in the coil structure by 'C'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SSES(dirPath, outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SSES_+3A_dirpath">dirPath</code></td>
<td>
<p>Path of the directory which contains all output files of PSIPRED. Each file belongs to a sequence.</p>
</td></tr>
<tr><td><code id="SSES_+3A_outformat">outFormat</code></td>
<td>
<p>It can take two values: 'mat' (which stands for matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="SSES_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>It shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same lengths such that the number of columns is equal to the length of the sequences
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for the sequences with the same lengths. However,
the users can use 'txt' option in the outFormat parameter for sequences with different lengths.
Warning: If the outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when the output format is 'txt', the label information is not displayed in the text file.
It is noteworthy that, 'txt' format is not usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
ad&lt;-paste0(dir,"/simpleSSE.txt")

Predss2dir&lt;-system.file("testForder",package="ftrCOOL")
Predss2dir&lt;-paste0(Predss2dir,"/ss2Dir/")
mat&lt;-SSES(Predss2dir,outFormat="txt",outputFileDist=ad)

unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='TorsionAngle'>Torsion Angle (TorsionAngle)</h2><span id='topic+TorsionAngle'></span>

<h3>Description</h3>

<p>The inputs to this function are phi and psi angles of each amino acid in the sequence. We use the output of SPINE-X software to obtain the angles.
Further, the TA function replaces each amino acid of the sequence with a vector. The vector contain two elements: The phi and psi angles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TorsionAngle(dirPath, outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TorsionAngle_+3A_dirpath">dirPath</code></td>
<td>
<p>Path of the directory which contains all output files of SPINE-X. Each file belongs to a sequence.</p>
</td></tr>
<tr><td><code id="TorsionAngle_+3A_outformat">outFormat</code></td>
<td>
<p>It can take two values: 'mat' (which stands for matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="TorsionAngle_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>It shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is differnet depending on the outFormat parameter ('mat' or 'txt').
If the outFormat is set to 'mat', it returns a feature matrix for sequences with the same lengths.
The number of rows is equal to the number of sequences and the number of columns
is (length of the sequence)*2.
If the outFormat is set to 'txt', all binary values will be writen in a 'txt' file.
Each row belongs to a sequence.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat parameter for sequences with different lengths.
Warning: If the outFormat is set to 'mat' for sequences with different lengths, it returns an error.
It is noteworthy that 'txt' format is not usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dir = tempdir()
ad&lt;-paste0(dir,"/ta.txt")

PredTorsioNdir&lt;-system.file("testForder",package="ftrCOOL")
PredTorsioNdir&lt;-paste0(PredTorsioNdir,"/TorsioNdir/")
mat&lt;-TorsionAngle(PredTorsioNdir,outFormat="txt",outputFileDist=ad)

unlink("dir", recursive = TRUE)
</code></pre>

<hr>
<h2 id='TPCP_DNA'>Trinucleotide physicochemical properties (TPCP_DNA)</h2><span id='topic+TPCP_DNA'></span>

<h3>Description</h3>

<p>This function replaces trinucleotides in a sequence with their physicochemical properties which is multiplied by normalized frequency of that tri-nucleotide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TPCP_DNA(
  seqs,
  selectedIdx = c("Dnase I", "Bendability (DNAse)"),
  threshold = 1,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TPCP_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="TPCP_DNA_+3A_selectedidx">selectedIdx</code></td>
<td>
<p>TPCP_DNA function works based on physicochemical properties. Users, select the properties by their ids
or indexes in TRI_DNA index file.
The default values of the vector are the  ids in &quot;Dnase I&quot;, &quot;Bendability (DNAse)&quot;.</p>
</td></tr>
<tr><td><code id="TPCP_DNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between 0 to 1. In selectedIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="TPCP_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="TPCP_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="TPCP_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are 12 physicochemical indexes in the trinucleotide database.
</p>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length-2)*(number of selected trinucleotide properties)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana1.fa",package="ftrCOOL")
vect&lt;-TPCP_DNA(seqs = fileLNC,threshold=1,outFormat="mat")
</code></pre>

<hr>
<h2 id='TriNUCindex_DNA'>Tri Nucleotide Index (TriNucIndex)</h2><span id='topic+TriNUCindex_DNA'></span>

<h3>Description</h3>

<p>This function replaces trinucleotides in a sequence with their physicochemical properties in the trinucleotide index file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TriNUCindex_DNA(
  seqs,
  selectedNucIdx = c("Dnase I", "Bendability (DNAse)"),
  threshold = 1,
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TriNUCindex_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="TriNUCindex_DNA_+3A_selectednucidx">selectedNucIdx</code></td>
<td>
<p>TriNucIndex function works based on physicochemical properties. Users, select the properties by their ids
or indexes in TRI_DNA index file.
The default values of the vector are the  ids in &quot;Dnase I&quot;, &quot;Bendability (DNAse)&quot;.</p>
</td></tr>
<tr><td><code id="TriNUCindex_DNA_+3A_threshold">threshold</code></td>
<td>
<p>is a number between 0 to 1. In selectedNucIdx, indices with a correlation
higher than the threshold will be deleted. The default value is 1.</p>
</td></tr>
<tr><td><code id="TriNUCindex_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="TriNUCindex_DNA_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="TriNUCindex_DNA_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are 12 physicochemical indexes in the trinucleotide database.
</p>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length-2)*(number of selected trinucleotide properties)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat parameter for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana1.fa",package="ftrCOOL")
vect&lt;-TriNUCindex_DNA(seqs = fileLNC,threshold=1,outFormat="mat")
</code></pre>

<hr>
<h2 id='Zcurve12bit_DNA'>Z_curve_12bit_DNA (Zcurve12bit_DNA)</h2><span id='topic+Zcurve12bit_DNA'></span>

<h3>Description</h3>

<p>These group of functions (Zcurve (9, 12, 36, 48, 144)_bit) function calculates the Z-curves. Z-curves are based on freqiencies of nucleotides, di-nucleotides, or tri-nucleotides and their positions on the sequences.
For more information about the methods please refer to reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zcurve12bit_DNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zcurve12bit_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="Zcurve12bit_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="Zcurve12bit_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="Zcurve12bit_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 12.
</p>


<h3>References</h3>

<p>Gao,F. and Zhang,C.T. Comparison of various algorithms for recognizing short coding sequences of human genes. Bioinformatics, (2004).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-Zcurve12bit_DNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='Zcurve12bit_RNA'>Z_curve_12bit_RNA (Zcurve12bit_RNA)</h2><span id='topic+Zcurve12bit_RNA'></span>

<h3>Description</h3>

<p>These group of functions (Zcurve (9, 12, 36, 48, 144)_bit) function calculates the Z-curves. Z-curves are based on freqiencies of ribonucleotides, di-ribonucleotides, or tri-ribonucleotides and their positions on the sequences.
For more information about the methods please refer to reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zcurve12bit_RNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zcurve12bit_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="Zcurve12bit_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="Zcurve12bit_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="Zcurve12bit_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 12.
</p>


<h3>References</h3>

<p>Gao,F. and Zhang,C.T. Comparison of various algorithms for recognizing short coding sequences of human genes. Bioinformatics, (2004).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-Zcurve12bit_RNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='Zcurve144bit_DNA'>Z_curve_144bit_DNA (Zcurve144bit_DNA)</h2><span id='topic+Zcurve144bit_DNA'></span>

<h3>Description</h3>

<p>These group of functions (Zcurve (9, 12, 36, 48, 144)_bit) function calculates the Z-curves. Z-curves are based on freqiencies of nucleotides, di-nucleotides, or tri-nucleotides and their positions on the sequences.
For more information about the methods please refer to reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zcurve144bit_DNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zcurve144bit_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="Zcurve144bit_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="Zcurve144bit_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="Zcurve144bit_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 144.
</p>


<h3>References</h3>

<p>Gao,F. and Zhang,C.T. Comparison of various algorithms for recognizing short coding sequences of human genes. Bioinformatics, (2004).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-Zcurve144bit_DNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='Zcurve144bit_RNA'>Z_curve_144bit_RNA (Zcurve144bit_RNA)</h2><span id='topic+Zcurve144bit_RNA'></span>

<h3>Description</h3>

<p>These group of functions (Zcurve (9, 12, 36, 48, 144)_bit) function calculates the Z-curves. Z-curves are based on freqiencies of ribonucleotides, di-ribonucleotides, or tri-ribonucleotides and their positions on the sequences.
For more information about the methods please refer to reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zcurve144bit_RNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zcurve144bit_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="Zcurve144bit_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="Zcurve144bit_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="Zcurve144bit_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 144.
</p>


<h3>References</h3>

<p>Gao,F. and Zhang,C.T. Comparison of various algorithms for recognizing short coding sequences of human genes. Bioinformatics, (2004).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-Zcurve144bit_RNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='Zcurve36bit_DNA'>Z_curve_36bit_DNA (Zcurve36bit_DNA)</h2><span id='topic+Zcurve36bit_DNA'></span>

<h3>Description</h3>

<p>These group of functions (Zcurve (9, 12, 36, 48, 144)_bit) function calculates the Z-curves. Z-curves are based on freqiencies of nucleotides, di-nucleotides, or tri-nucleotides and their positions on the sequences.
For more information about the methods please refer to reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zcurve36bit_DNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zcurve36bit_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="Zcurve36bit_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="Zcurve36bit_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="Zcurve36bit_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 36.
</p>


<h3>References</h3>

<p>Gao,F. and Zhang,C.T. Comparison of various algorithms for recognizing short coding sequences of human genes. Bioinformatics, (2004).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-Zcurve36bit_DNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='Zcurve36bit_RNA'>Z_curve_36bit_RNA (Zcurve36bit_RNA)</h2><span id='topic+Zcurve36bit_RNA'></span>

<h3>Description</h3>

<p>These group of functions (Zcurve (9, 12, 36, 48, 144)_bit) function calculates the Z-curves. Z-curves are based on freqiencies of ribonucleotides, di-ribonucleotides, or tri-ribonucleotides and their positions on the sequences.
For more information about the methods please refer to reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zcurve36bit_RNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zcurve36bit_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="Zcurve36bit_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="Zcurve36bit_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="Zcurve36bit_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 36.
</p>


<h3>References</h3>

<p>Gao,F. and Zhang,C.T. Comparison of various algorithms for recognizing short coding sequences of human genes. Bioinformatics, (2004).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-Zcurve36bit_RNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='Zcurve48bit_DNA'>Z_curve_48bit_DNA (Zcurve48bit_DNA)</h2><span id='topic+Zcurve48bit_DNA'></span>

<h3>Description</h3>

<p>These group of functions (Zcurve (9, 12, 36, 48, 144)_bit) function calculates the Z-curves. Z-curves are based on freqiencies of nucleotides, di-nucleotides, or tri-nucleotides and their positions on the sequences.
For more information about the methods please refer to reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zcurve48bit_DNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zcurve48bit_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="Zcurve48bit_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="Zcurve48bit_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="Zcurve48bit_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 48.
</p>


<h3>References</h3>

<p>Gao,F. and Zhang,C.T. Comparison of various algorithms for recognizing short coding sequences of human genes. Bioinformatics, (2004).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-Zcurve48bit_DNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='Zcurve48bit_RNA'>Z_curve_48bit_RNA (Zcurve48bit_RNA)</h2><span id='topic+Zcurve48bit_RNA'></span>

<h3>Description</h3>

<p>These group of functions (Zcurve (9, 12, 36, 48, 144)_bit) function calculates the Z-curves. Z-curves are based on freqiencies of ribo
ribonucleotides, di-ribonucleotides, or tri-ribonucleotides and their positions on the sequences.
For more information about the methods please refer to reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zcurve48bit_RNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zcurve48bit_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="Zcurve48bit_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="Zcurve48bit_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="Zcurve48bit_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 48.
</p>


<h3>References</h3>

<p>Gao,F. and Zhang,C.T. Comparison of various algorithms for recognizing short coding sequences of human genes. Bioinformatics, (2004).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-Zcurve48bit_RNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='Zcurve9bit_DNA'>Z_curve_9bit_DNA (Zcurve9bit_DNA)</h2><span id='topic+Zcurve9bit_DNA'></span>

<h3>Description</h3>

<p>These group of functions (Zcurve (9, 12, 36, 48, 144)_bit) function calculates the Z-curves. Z-curves are based on freqiencies of nucleotides, di-nucleotides, or tri-nucleotides and their positions on the sequences.
For more information about the methods please refer to reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zcurve9bit_DNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zcurve9bit_DNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td></tr>
<tr><td><code id="Zcurve9bit_DNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="Zcurve9bit_DNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="Zcurve9bit_DNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 9.
</p>


<h3>References</h3>

<p>Gao,F. and Zhang,C.T. Comparison of various algorithms for recognizing short coding sequences of human genes. Bioinformatics, (2004).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-Zcurve9bit_DNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='Zcurve9bit_RNA'>Z_curve_9bit_RNA (Zcurve9bit_RNA)</h2><span id='topic+Zcurve9bit_RNA'></span>

<h3>Description</h3>

<p>These group of functions (Zcurve (9, 12, 36, 48, 144)_bit) function calculates the Z-curves. Z-curves are based on freqiencies of ribo
ribonucleotides, di-ribonucleotides, or tri-ribonucleotides and their positions on the sequences.
For more information about the methods please refer to reference part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zcurve9bit_RNA(seqs, ORF = FALSE, reverseORF = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zcurve9bit_RNA_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td></tr>
<tr><td><code id="Zcurve9bit_RNA_+3A_orf">ORF</code></td>
<td>
<p>(Open Reading Frame) is a logical parameter. If it is set to true, ORF region of each sequence is considered instead of the original sequence (i.e., 3-frame).</p>
</td></tr>
<tr><td><code id="Zcurve9bit_RNA_+3A_reverseorf">reverseORF</code></td>
<td>
<p>is a logical parameter. It is enabled only if ORF is true.
If reverseORF is true, ORF region will be searched in the sequence and also in the reverse complement of the sequence (i.e., 6-frame).</p>
</td></tr>
<tr><td><code id="Zcurve9bit_RNA_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is 9.
</p>


<h3>References</h3>

<p>Gao,F. and Zhang,C.T. Comparison of various algorithms for recognizing short coding sequences of human genes. Bioinformatics, (2004).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-Zcurve9bit_RNA(seqs=fileLNC,ORF=TRUE,reverseORF=FALSE)
</code></pre>

<hr>
<h2 id='zSCALE'>Z-SCALE (zSCALE)</h2><span id='topic+zSCALE'></span>

<h3>Description</h3>

<p>This function converts the amino acids of a sequence to five physicochemical descriptor variables which were developed by Sandberg et al. in 1998.
The Z-SCALE function can be applied to encode peptides of equal length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zSCALE(seqs, label = c(), outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zSCALE_+3A_seqs">seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td></tr>
<tr><td><code id="zSCALE_+3A_label">label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td></tr>
<tr><td><code id="zSCALE_+3A_outformat">outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td></tr>
<tr><td><code id="zSCALE_+3A_outputfiledist">outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length)*(5)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat parameter for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-zSCALE(seqs = ptmSeqsVect,outFormat="mat")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
