<!DOCTYPE html><html><head><title>Help for package FunctanSNP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FunctanSNP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#coefSNPreg'><p>Extract coefficients from an SNPlm or SNPinter object</p></a></li>
<li><a href='#plotGVF'><p>Plot the genetic variant function</p></a></li>
<li><a href='#plotRawdata'><p>Visualization of the sequence (genotypes) data</p></a></li>
<li><a href='#plotSNPreg'><p>Plot coefficient functions and residuals from an SNPlm or SNPinter object</p></a></li>
<li><a href='#predictSNPreg'><p>Make predictions from an SNPlm or SNPinter object</p></a></li>
<li><a href='#simData1'><p>Generate simulated data for <code>SNPlm</code></p></a></li>
<li><a href='#simData2'><p>Generate simulated data for <code>SNPinter</code> and <code>SNPcvinter</code></p></a></li>
<li><a href='#simX'><p>Generate the sequence (genotypes) data containing only 0, 1 and 2</p></a></li>
<li><a href='#SNPcvinter'><p>Cross-validation for <code>SNPinter</code></p></a></li>
<li><a href='#SNPgvf'><p>Transform the sequence (genotypes) data into the genetic variant function</p></a></li>
<li><a href='#SNPinter'><p>Revised partially functional interaction regression analysis for the sequence (genotypes) data</p></a></li>
<li><a href='#SNPlm'><p>Revised functional linear regression analysis for the sequence (genotypes) data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Functional Analysis (with Interactions) for Dense SNP Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rui Ren &lt;xmurr@stu.xmu.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of revised functional regression models for multiple genetic variation data, such as single nucleotide polymorphism (SNP) data, which provides revised functional linear regression models, partially functional interaction regression analysis with penalty-based techniques and corresponding drawing functions, etc.(Ruzong Fan, Yifan Wang, James L. Mills, Alexander F. Wilson, Joan E. Bailey-Wilson, and Momiao Xiong (2013) &lt;<a href="https://doi.org/10.1002%2Fgepi.21757">doi:10.1002/gepi.21757</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>caret, glmnet, lava, MASS, stats, fda, funData, graphics</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-18 09:02:10 UTC; renrui</td>
</tr>
<tr>
<td>Author:</td>
<td>Rui Ren [aut, cre],
  Kuangnan Fang [aut],
  Qingzhao Zhang [aut],
  Shuangge Ma [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-18 12:28:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='coefSNPreg'>Extract coefficients from an SNPlm or SNPinter object</h2><span id='topic+coefSNPreg'></span>

<h3>Description</h3>

<p>This functions predicts coefficients and related coefficient functions from a fitted &quot;SNPlm&quot; or &quot;SNPinter&quot; object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coefSNPreg(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coefSNPreg_+3A_x">x</code></td>
<td>
<p>an &quot;SNPlm&quot;/&quot;SNPinter&quot; object obtained by <code><a href="#topic+SNPlm">SNPlm</a></code>/<code><a href="#topic+SNPinter">SNPinter</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An &quot;coef.SNPreg&quot; object that contains the list of the following items.
</p>

<ul>
<li><p>alpha: estimated intercept value.
</p>
</li>
<li><p>gamma: estimated coefficients of the scalar covariates.
</p>
</li>
<li><p>b: estimated coefficients of the chosen basis functions for the genetic effect function beta(t) (and interaction items betak(t), if x is an &quot;SNPinter&quot; object).
</p>
</li>
<li><p>betat: an &quot;fd&quot; object, representing the estimated genetic effect function beta(t).
</p>
</li></ul>



<h3>See Also</h3>

<p>See Also as <code><a href="#topic+SNPlm">SNPlm</a></code>, <code><a href="#topic+SNPinter">SNPinter</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 300
m &lt;- 30
simdata1 &lt;- simData1(n, m, seed = 123)
SNPlmres &lt;- SNPlm(y = simdata1$y, z = simdata1$z,
                  location = simdata1$location, X = simdata1$X,
                  type1 = "Bspline", type2 = "Bspline", nbasis1 = 5,
                  nbasis2 = 5, params1 = 4, params2 = 4,
                  intercept = FALSE, Plot = FALSE)
coef1 &lt;- coefSNPreg(x = SNPlmres)
coef1$alpha ###intercept
coef1$gamma ###coefficients of scalar variables
coef1$b     ###coefficients of basis functions

simdata2 &lt;- simData2(n, m, seed = 123)
lambda1 &lt;- 0.05
lambda2 &lt;- sqrt(3)*lambda1
eta &lt;- 0
SNPinterres &lt;- SNPinter(y = simdata2$y, z = simdata2$z,
                        location = simdata2$location, X = simdata2$X,
                        lambda1, lambda2, eta, type1 = "Bspline", nbasis1 = 5,
                        params1 = 4, Bsplines = 5, norder = 4, intercept = TRUE,
                        eps = 1e-2, maxstep = 1e2, Plot = FALSE)
coef2 &lt;- coefSNPreg(x = SNPinterres)
coef2$alpha ###intercept
coef2$gamma ###coefficients of scalar variables
coef2$b     ###coefficients of basis functions

</code></pre>

<hr>
<h2 id='plotGVF'>Plot the genetic variant function</h2><span id='topic+plotGVF'></span>

<h3>Description</h3>

<p>Plot the estimated genetic variant function obtained by SNPgvf or a given functional data object(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotGVF(x, y = NULL, Lfdobj = 0, href = TRUE, titles = NULL,
  xlim = NULL, ylim = NULL, xlab = NULL, ylab = NULL,
  ask = FALSE, nx = NULL, axes = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotGVF_+3A_x">x</code></td>
<td>
<p>functional data object(s) to be plotted, such as the estimated genetic variant function obtained by SNPgvf.</p>
</td></tr>
<tr><td><code id="plotGVF_+3A_y">y</code></td>
<td>
<p>sequence of points at which to evaluate the functions 'x' and plot on the horizontal axis. Defaults to seq(rangex[1], rangex[2], length = nx).</p>
</td></tr>
<tr><td><code id="plotGVF_+3A_lfdobj">Lfdobj</code></td>
<td>
<p>either a nonnegative integer or a linear differential operator object. If present, the derivative or the value of applying the operator is plotted rather than the functions themselves.</p>
</td></tr>
<tr><td><code id="plotGVF_+3A_href">href</code></td>
<td>
<p>a logical variable: If TRUE, add a horizontal reference line at 0.</p>
</td></tr>
<tr><td><code id="plotGVF_+3A_titles">titles</code></td>
<td>
<p>a vector of strings for identifying curves</p>
</td></tr>
<tr><td><code id="plotGVF_+3A_xlim">xlim</code></td>
<td>
<p>a vector of length 2 containing axis limits for the horizontal axis.</p>
</td></tr>
<tr><td><code id="plotGVF_+3A_ylim">ylim</code></td>
<td>
<p>a vector of length 2 containing axis limits for the vertical axis.</p>
</td></tr>
<tr><td><code id="plotGVF_+3A_xlab">xlab</code></td>
<td>
<p>a label for the horizontal axis.</p>
</td></tr>
<tr><td><code id="plotGVF_+3A_ylab">ylab</code></td>
<td>
<p>a label for the vertical axis.</p>
</td></tr>
<tr><td><code id="plotGVF_+3A_ask">ask</code></td>
<td>
<p>a logical value: If TRUE, each curve is shown separately, and the plot advances with a mouse click</p>
</td></tr>
<tr><td><code id="plotGVF_+3A_nx">nx</code></td>
<td>
<p>the number of points to use to define the plot. The default is usually enough, but for a highly variable function more may be required.</p>
</td></tr>
<tr><td><code id="plotGVF_+3A_axes">axes</code></td>
<td>
<p>Either a logical or a list or NULL.
</p>

<ul>
<li><p>logical whether axes should be drawn on the plot
</p>
</li>
<li><p>list a list used to create custom axes used to create axes via x$axes[[1]] and x$axes[-1]. The primary example of this uses list(&quot;axesIntervals&quot;, ...)
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>plot the estimated genetic variant function.
</p>


<h3>See Also</h3>

<p>See Also as <code><a href="#topic+SNPgvf">SNPgvf</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 20
m &lt;- 50
simdata &lt;- simX(n, m, seed = 1, d.ratio = 0)
X &lt;- simdata$X
location &lt;- simdata$location
SNPgvfres &lt;- SNPgvf(location, X, type = "Bspline", nbasis = 5, params = 4, Plot = FALSE)
plotGVF(SNPgvfres)

</code></pre>

<hr>
<h2 id='plotRawdata'>Visualization of the sequence (genotypes) data</h2><span id='topic+plotRawdata'></span>

<h3>Description</h3>

<p>Display the sequence data, such as the data generated by simX. The horizontal axis represents the sampling sites, and the vertical axis represents the sequence values containing only 0, 1 and 2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRawdata(location, X, color = NULL, pch = 16, cex = 0.9)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotRawdata_+3A_location">location</code></td>
<td>
<p>a numeric vector defining the sampling sites of the sequence data.</p>
</td></tr>
<tr><td><code id="plotRawdata_+3A_x">X</code></td>
<td>
<p>a matrix specifying the sequence data, with the number of rows equal to the number of samples.</p>
</td></tr>
<tr><td><code id="plotRawdata_+3A_color">color</code></td>
<td>
<p>A specification for the default plotting color. See graphical parameters (<code><a href="graphics.html#topic+par">par</a></code>) for more details.</p>
</td></tr>
<tr><td><code id="plotRawdata_+3A_pch">pch</code></td>
<td>
<p>Either an integer specifying a symbol or a single character to be used as the default in plotting points. See graphical parameters (<code><a href="graphics.html#topic+par">par</a></code>) for more details.</p>
</td></tr>
<tr><td><code id="plotRawdata_+3A_cex">cex</code></td>
<td>
<p>A numerical value giving the amount by which plotting text and symbols should be magnified relative to the default. See graphical parameters (<code><a href="graphics.html#topic+par">par</a></code>) for more details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>show the sequence data.
</p>


<h3>See Also</h3>

<p>See Also as <code><a href="#topic+simX">simX</a></code>, <code><a href="#topic+SNPgvf">SNPgvf</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 20
m &lt;- 50
simdata &lt;- simX(n, m, seed = 1, d.ratio = 0)
X &lt;- simdata$X
location &lt;- simdata$location
plotRawdata(location, X = X[1:2, ])

</code></pre>

<hr>
<h2 id='plotSNPreg'>Plot coefficient functions and residuals from an SNPlm or SNPinter object</h2><span id='topic+plotSNPreg'></span>

<h3>Description</h3>

<p>Produces a residual diagram or a plot of the genetic effect function beta(t) (and interaction items betak(t), if the entry x is an &quot;SNPinter&quot; object) for a fitted &quot;SNPlm&quot; or &quot;SNPinter&quot; object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSNPreg(x, type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSNPreg_+3A_x">x</code></td>
<td>
<p>an &quot;SNPlm&quot;/&quot;SNPinter&quot; object obtained by <code><a href="#topic+SNPlm">SNPlm</a></code>/<code><a href="#topic+SNPinter">SNPinter</a></code>.</p>
</td></tr>
<tr><td><code id="plotSNPreg_+3A_type">type</code></td>
<td>
<p>character, &quot;beta&quot; or &quot;residuals&quot; type, if &quot;beta&quot;, plot the genetic effect function beta0(t) (and interaction items betak(t), if the entry x is an &quot;SNPinter&quot; object), if &quot;residuals&quot;, show the residual plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>show the residual plot or coefficient functions.
</p>


<h3>See Also</h3>

<p>See Also as <code><a href="#topic+SNPlm">SNPlm</a></code>, <code><a href="#topic+SNPinter">SNPinter</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 300
m &lt;- 30
simdata1 &lt;- simData1(n, m, seed = 123)
SNPlmres &lt;- SNPlm(y = simdata1$y, z = simdata1$z,
                  location = simdata1$location, X = simdata1$X,
                  type1 = "Bspline", type2 = "Bspline", nbasis1 = 5,
                  nbasis2 = 5, params1 = 4, params2 = 4,
                  intercept = FALSE, Plot = FALSE)
plotSNPreg(x = SNPlmres, type = "beta")
plotSNPreg(x = SNPlmres, type = "residuals")

simdata2 &lt;- simData2(n, m, seed = 123)
lambda1 &lt;- 0.05
lambda2 &lt;- sqrt(3)*lambda1
eta &lt;- 0
SNPinterres &lt;- SNPinter(y = simdata2$y, z = simdata2$z,
                        location = simdata2$location, X = simdata2$X,
                        lambda1, lambda2, eta, type1 = "Bspline", nbasis1 = 5,
                        params1 = 4, Bsplines = 5, norder = 4, intercept = TRUE,
                        eps = 1e-2, maxstep = 1e2, Plot = FALSE)
plotSNPreg(x = SNPinterres, type = "beta")
plotSNPreg(x = SNPinterres, type = "residuals")

</code></pre>

<hr>
<h2 id='predictSNPreg'>Make predictions from an SNPlm or SNPinter object</h2><span id='topic+predictSNPreg'></span>

<h3>Description</h3>

<p>This functions predicts fitted values for newly entered data from a fitted &quot;SNPlm&quot; or &quot;SNPinter&quot; object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predictSNPreg(x, newz, newX)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predictSNPreg_+3A_x">x</code></td>
<td>
<p>an &quot;SNPlm&quot;/&quot;SNPinter&quot; object obtained by <code><a href="#topic+SNPlm">SNPlm</a></code>/<code><a href="#topic+SNPinter">SNPinter</a></code>.</p>
</td></tr>
<tr><td><code id="predictSNPreg_+3A_newz">newz</code></td>
<td>
<p>matrix of new values for z at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predictSNPreg_+3A_newx">newX</code></td>
<td>
<p>matrix of new values for X at which predictions are to be made.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p>predicted.values:  the predicted mean values.
</p>
</li></ul>



<h3>See Also</h3>

<p>See Also as <code><a href="#topic+SNPlm">SNPlm</a></code>, <code><a href="#topic+SNPinter">SNPinter</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 300
m &lt;- 30
simdata1 &lt;- simData1(n, m, seed = 123)
SNPlmres &lt;- SNPlm(y = simdata1$y[1:200], z = simdata1$z[1:200, ],
                  location = simdata1$location, X = simdata1$X[1:200, ],
                  type1 = "Bspline", type2 = "Bspline", nbasis1 = 5,
                  nbasis2 = 5, params1 = 4, params2 = 4, intercept = FALSE,
                  Plot = FALSE)
predict.values1 &lt;- predictSNPreg(x = SNPlmres, newz = simdata1$z[-(1:200), ],
                                 newX = simdata1$X[-(1:200), ])

simdata2 &lt;- simData2(n, m, seed = 123)
lambda1 &lt;- 0.05
lambda2 &lt;- sqrt(3)*lambda1
eta &lt;- 0
SNPinterres &lt;- SNPinter(y = simdata2$y[1:200], z = simdata2$z[1:200, ],
                        location = simdata2$location, X = simdata2$X[1:200, ],
                        lambda1, lambda2, eta, type1 = "Bspline",
                        nbasis1 = 5, params1 = 4, Bsplines = 5, norder = 4,
                        intercept = TRUE, eps = 1e-2, maxstep = 1e2, Plot = FALSE)
predict.values2 &lt;- predictSNPreg(x = SNPinterres, newz = simdata2$z[-(1:200), ],
                                 newX = simdata2$X[-(1:200), ])

</code></pre>

<hr>
<h2 id='simData1'>Generate simulated data for <code><a href="#topic+SNPlm">SNPlm</a></code></h2><span id='topic+simData1'></span>

<h3>Description</h3>

<p>Generate simulated data for users to apply the method SNPlm, including response variable y, scalar variable Z and sequence (genotypes) data X.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simData1(n, m, seed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simData1_+3A_n">n</code></td>
<td>
<p>an interger variable specifying the number of samples to be generated.</p>
</td></tr>
<tr><td><code id="simData1_+3A_m">m</code></td>
<td>
<p>an interger variable specifying the sequence length of each sample.</p>
</td></tr>
<tr><td><code id="simData1_+3A_seed">seed</code></td>
<td>
<p>an integer variable specifying the random seed used for random sequence generation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An &quot;simData1&quot; object that contains the list of the following items.
</p>

<ul>
<li><p>y: a numeric vector representing the response variables.
</p>
</li>
<li><p>z: a matrix representing the scalar covariates, with the number of rows equal to the number of samples.
</p>
</li>
<li><p>location: a numeric vector defining the sampling sites of the sequence (genotypes) data.
</p>
</li>
<li><p>X: a matrix representing the sequence data, with the number of rows equal to the number of samples.
</p>
</li>
<li><p>beta: an &quot;fd&quot; object, representing the genetic effect function.
</p>
</li></ul>



<h3>See Also</h3>

<p>See Also as <code><a href="#topic+simX">simX</a></code>, <code><a href="#topic+SNPlm">SNPlm</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 300
m &lt;- 30
simdata1 &lt;- simData1(n, m, seed = 123)

</code></pre>

<hr>
<h2 id='simData2'>Generate simulated data for <code><a href="#topic+SNPinter">SNPinter</a></code> and <code><a href="#topic+SNPcvinter">SNPcvinter</a></code></h2><span id='topic+simData2'></span>

<h3>Description</h3>

<p>Generate simulated data for users to apply the method SNPinter and SNPcvinter, including response variable y, scalar variable Z and sequence (genotypes) data X.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simData2(n, m, seed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simData2_+3A_n">n</code></td>
<td>
<p>an interger variable specifying the number of samples to be generated.</p>
</td></tr>
<tr><td><code id="simData2_+3A_m">m</code></td>
<td>
<p>an interger variable specifying the sequence length of each sample.</p>
</td></tr>
<tr><td><code id="simData2_+3A_seed">seed</code></td>
<td>
<p>an integer variable specifyinging the random seed used for random sequence generation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An &quot;simData2&quot; object that contains the list of the following items.
</p>

<ul>
<li><p>y: a numeric vector representing the response variables.
</p>
</li>
<li><p>z: a matrix representing the scalar covariates, with the number of rows equal to the number of samples.
</p>
</li>
<li><p>location: a numeric vector defining the sampling sites of the sequence (genotypes) data.
</p>
</li>
<li><p>X: a matrix representing the sequence data, with the number of rows equal to the number of samples.
</p>
</li>
<li><p>beta: an &quot;fd&quot; object specifying the genetic effect function.
</p>
</li></ul>



<h3>See Also</h3>

<p>See Also as <code><a href="#topic+simData1">simData1</a></code>, <code><a href="#topic+SNPinter">SNPinter</a></code>, <code><a href="#topic+SNPcvinter">SNPcvinter</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 300
m &lt;- 30
simdata2 &lt;- simData2(n, m, seed = 123)

</code></pre>

<hr>
<h2 id='simX'>Generate the sequence (genotypes) data containing only 0, 1 and 2</h2><span id='topic+simX'></span>

<h3>Description</h3>

<p>This function provides a method for generating sequence data containing only 0, 1 and 2, which can be used to simulate the generation of sequence genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simX(n, m, seed = 1, d.ratio = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simX_+3A_n">n</code></td>
<td>
<p>an interger variable specifying the number of samples to be generated.</p>
</td></tr>
<tr><td><code id="simX_+3A_m">m</code></td>
<td>
<p>an interger variable specifying the sequence length of each sample.</p>
</td></tr>
<tr><td><code id="simX_+3A_seed">seed</code></td>
<td>
<p>an integer variable specifying the random seed used for random sequence generation.</p>
</td></tr>
<tr><td><code id="simX_+3A_d.ratio">d.ratio</code></td>
<td>
<p>a numeric variable between 0 and 1 indicating the deletion ratio of sample sequences, default value is 0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An &quot;simX&quot; object that contains the list of the following items.
</p>

<ul>
<li><p>location: a numeric vector defining the sampling sites of the sequence data.
</p>
</li>
<li><p>X: a matrix with n rows and m columns representing the sequence data.
</p>
</li></ul>



<h3>See Also</h3>

<p>See Also as <code><a href="#topic+plotRawdata">plotRawdata</a></code>, <code><a href="#topic+SNPgvf">SNPgvf</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 2
m &lt;- 50
simdata1 &lt;- simX(n, m, seed = 1, d.ratio = 0)
simdata2 &lt;- simX(n, m, seed = 1, d.ratio = 0.3)
plotRawdata(location = simdata1$location, X = simdata1$X)
plotRawdata(location = simdata2$location, X = simdata2$X)

</code></pre>

<hr>
<h2 id='SNPcvinter'>Cross-validation for <code><a href="#topic+SNPinter">SNPinter</a></code></h2><span id='topic+SNPcvinter'></span>

<h3>Description</h3>

<p>Performs K-fold cross validation for the revised partially functional interaction regression analysis over a grid of values for the regularization parameter lambda1 and lambda2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNPcvinter(y, z, location, X, K, lambda1, lambda2 = NULL, eta = 0,
  type1, nbasis1, params1, Bsplines = 20, norder = 4,
  intercept = FALSE, eps = 1e-05, maxstep = 1e+05, Plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNPcvinter_+3A_y">y</code></td>
<td>
<p>a numeric vector defining the response variables.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_z">z</code></td>
<td>
<p>a matrix defining the scalar covariates, with the number of rows equal to the number of samples.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_location">location</code></td>
<td>
<p>a numeric vector defining the sampling sites of the sequence data.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_x">X</code></td>
<td>
<p>a matrix specifying the sequence (genotypes) data, with the number of rows equal to the number of samples.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_k">K</code></td>
<td>
<p>an integer specifying the number of cross-validation folds, default is 5.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_lambda1">lambda1</code></td>
<td>
<p>a numeric vector specifying the sparsity penalty parameter to be determined.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_lambda2">lambda2</code></td>
<td>
<p>a numeric vector specifying the group sparsity penalty parameter to be determined.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_eta">eta</code></td>
<td>
<p>a numeric vector specifying the penalty parameter for smoothing analysis.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_type1">type1</code></td>
<td>
<p>a character specifying the type of the basis functions that constitutes the genetic variation function. The options are &quot;Bspline&quot;, &quot;Exponential&quot;, &quot;Fourier&quot;, &quot;Monomial&quot;, and &quot;Power&quot;.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_nbasis1">nbasis1</code></td>
<td>
<p>an integer specifying the number of basis functions that constitutes the genetic variation function.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_params1">params1</code></td>
<td>
<p>in addition to rangeval1 (a vector of length 2 giving the lower and upper limits of the range of permissible values for the genetic variation function) and nbasis1, all bases have one or two parameters unique to that basis type or shared with one other;
</p>

<ul>
<li><p>bspline: Argument norder = the order of the spline, which is one more than the degree of the polynomials used. This defaults to 4, which gives cubic splines.
</p>
</li>
<li><p>exponential: Argument ratevec. In fda_2.0.2, this defaulted to 1. In fda_2.0.3, it will default to 0:1.
</p>
</li>
<li><p>fourier: Argument period defaults to diff(rangeval).
</p>
</li>
<li><p>monomial/power: Argument exponents. Default = 0:(nbasis-1). For monomial bases, exponents must be distinct nonnegative integers. For power bases, they must be distinct real numbers.
</p>
</li></ul>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_bsplines">Bsplines</code></td>
<td>
<p>an integer specifying the number of basis functions that constitutes the genetic effect function.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_norder">norder</code></td>
<td>
<p>an integer specifying the order of bsplines that constitutes the genetic effect function, which is one higher than their degree. The default of 4 gives cubic splines.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_intercept">intercept</code></td>
<td>
<p>should intercept(s) be fitted (TRUE) or set to zero (default = FALSE).</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_eps">eps</code></td>
<td>
<p>a numeric variable specifying the threshold at which the algorithm terminates, default is 1e-5.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_maxstep">maxstep</code></td>
<td>
<p>a numeric variable specifying the maximum iteration steps, default is 1e5.</p>
</td></tr>
<tr><td><code id="SNPcvinter_+3A_plot">Plot</code></td>
<td>
<p>should the estimated genetic effect function beta0(t) and interaction items betak(t) be plotted (TRUE) or not (default = FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An &quot;SNPcvinter&quot; object that contains the list of the following items.
</p>

<ul>
<li><p>lambda1Select: a numeric value of the sparsity penalty parameter selected by cross validation.
</p>
</li>
<li><p>lambda2Select: a numeric value of the group sparsity penalty parameter selected by cross validation.
</p>
</li>
<li><p>etaSelect: a numeric value of the smoothing parameter selected by cross validation.
</p>
</li>
<li><p>alpha: estimated intercept value..
</p>
</li>
<li><p>gamma: estimated coefficients of the scalar covariates.
</p>
</li>
<li><p>b: estimated coefficients of the chosen basis functions for the genetic effect function beta0(t) and interaction items betak(t).
</p>
</li>
<li><p>betat: an &quot;fd&quot; object, representing the estimated genetic effect function beta(t) and interaction items betak(t).
</p>
</li>
<li><p>residuals: the residuals, that is response minus fitted values.
</p>
</li>
<li><p>fitted.values:  the fitted mean values.
</p>
</li>
<li><p>lambda1: a numeric vector specifying the sparsity penalty parameter for cross validation.
</p>
</li>
<li><p>lambda2: a numeric vector specifying the group sparsity penalty parameter for cross validation.
</p>
</li>
<li><p>eta: a numeric vector specifying the smoothing parameter for cross validation.
</p>
</li>
<li><p>CVerror: a numeric vector, containing the mean square errors on testing set during cross validation.
</p>
</li></ul>



<h3>See Also</h3>

<p>See Also as <code><a href="#topic+simData2">simData2</a></code>, <code><a href="#topic+SNPinter">SNPinter</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 300
m &lt;- 30
simdata2 &lt;- simData2(n, m, seed = 123)
y &lt;- simdata2$y
z &lt;- simdata2$z
location &lt;- simdata2$location
X &lt;- simdata2$X
lambda1 &lt;- c(0.01, 0.05, 0.1)
lambda2 &lt;- sqrt(3)*lambda1
SNPcvinterres &lt;- SNPcvinter(y, z, location, X, K = 3, lambda1, lambda2, eta = 0,
                            type1 = "Bspline", nbasis1 = 5, params1 = 4, Bsplines = 5,
                            norder = 4, intercept = TRUE, eps = 1e-2, maxstep = 1e2, Plot = TRUE)
SNPcvinterres$lambda1Select
SNPcvinterres$lambda2Select
SNPcvinterres$alpha
SNPcvinterres$gamma
SNPcvinterres$b

</code></pre>

<hr>
<h2 id='SNPgvf'>Transform the sequence (genotypes) data into the genetic variant function</h2><span id='topic+SNPgvf'></span>

<h3>Description</h3>

<p>This function conducts the ordinary linear square smoothing analysis and models the genotypes of an individual (such as the sequence data generated by simX containing only 0, 1, 2) as the genetic variant function X(t).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNPgvf(location, X, type, nbasis, params, Plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNPgvf_+3A_location">location</code></td>
<td>
<p>a numeric vector defining the sampling sites of the sequence data.</p>
</td></tr>
<tr><td><code id="SNPgvf_+3A_x">X</code></td>
<td>
<p>a matrix specifyinging the sequence data, with the number of rows equal to the number of samples.</p>
</td></tr>
<tr><td><code id="SNPgvf_+3A_type">type</code></td>
<td>
<p>a character specifying the type of the basis functions. The options are &quot;Bspline&quot;, &quot;Exponential&quot;, &quot;Fourier&quot;, &quot;Monomial&quot;, and &quot;Power&quot;.</p>
</td></tr>
<tr><td><code id="SNPgvf_+3A_nbasis">nbasis</code></td>
<td>
<p>an integer specifying the number of basis functions.</p>
</td></tr>
<tr><td><code id="SNPgvf_+3A_params">params</code></td>
<td>
<p>in addition to rangeval (a vector of length 2 giving the lower and upper limits of the range of permissible values for the function) and nbasis, all bases have one or two parameters unique to that basis type or shared with one other:
</p>

<ul>
<li><p>bspline: Argument norder = the order of the spline, which is one more than the degree of the polynomials used. This defaults to 4, which gives cubic splines.
</p>
</li>
<li><p>exponential: Argument ratevec. In fda_2.0.2, this defaulted to 1. In fda_2.0.3, it will default to 0:1.
</p>
</li>
<li><p>fourier: Argument period defaults to diff(rangeval).
</p>
</li>
<li><p>monomial/power: Argument exponents. Default = 0:(nbasis-1). For monomial bases, exponents must be distinct nonnegative integers. For power bases, they must be distinct real numbers.
</p>
</li></ul>
</td></tr>
<tr><td><code id="SNPgvf_+3A_plot">Plot</code></td>
<td>
<p>should the estimated genetic variant function X(t) be plotted (TRUE) or not (default = FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An 'fd' object that contains the estimated genetic variant function.
</p>


<h3>See Also</h3>

<p>See Also as <code><a href="#topic+simX">simX</a></code>, <code><a href="#topic+plotRawdata">plotRawdata</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 20
m &lt;- 50
simdata &lt;- simX(n, m, seed = 1, d.ratio = 0)
X &lt;- simdata$X
location &lt;- simdata$location
SNPgvfres &lt;- SNPgvf(location, X, type = "Bspline", nbasis = 5, params = 4, Plot = FALSE)
plotRawdata(location, X)
plotGVF(SNPgvfres)

</code></pre>

<hr>
<h2 id='SNPinter'>Revised partially functional interaction regression analysis for the sequence (genotypes) data</h2><span id='topic+SNPinter'></span>

<h3>Description</h3>

<p>This function conducts joint analysis, which includes all scalar covariates Z, genetic variant function X(t), and their interactions in a partially functional interaction regression model. In addition, this function identifies the relevant genetic variation sites with a local sparsity penalty-based method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNPinter(y, z, location, X, lambda1, lambda2 = NULL, eta, type1, nbasis1,
  params1, Bsplines = 20, norder = 4, intercept = FALSE,
  eps = 1e-05, maxstep = 1e+05, Plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNPinter_+3A_y">y</code></td>
<td>
<p>a numeric vector defining the response variables.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_z">z</code></td>
<td>
<p>a matrix defining the scalar covariates, with the number of rows equal to the number of samples.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_location">location</code></td>
<td>
<p>a numeric vector defining the sampling sites of the sequence data.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_x">X</code></td>
<td>
<p>a matrix specifying the sequence (genotypes) data, with the number of rows equal to the number of samples.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_lambda1">lambda1</code></td>
<td>
<p>a numeric vector specifying the sparsity penalty parameter to be determined.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_lambda2">lambda2</code></td>
<td>
<p>a numeric vector specifying the group sparsity penalty parameter to be determined.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_eta">eta</code></td>
<td>
<p>a numeric vector specifying the penalty parameter for smoothing analysis.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_type1">type1</code></td>
<td>
<p>a character specifying the type of the basis functions that constitutes the genetic variation function. The options are &quot;Bspline&quot;, &quot;Exponential&quot;, &quot;Fourier&quot;, &quot;Monomial&quot;, and &quot;Power&quot;.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_nbasis1">nbasis1</code></td>
<td>
<p>an integer specifying the number of basis functions that constitutes the genetic variation function.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_params1">params1</code></td>
<td>
<p>in addition to rangeval1 (a vector of length 2 giving the lower and upper limits of the range of permissible values for the genetic variation function) and nbasis1, all bases have one or two parameters unique to that basis type or shared with one other;
</p>

<ul>
<li><p>bspline: Argument norder = the order of the spline, which is one more than the degree of the polynomials used. This defaults to 4, which gives cubic splines.
</p>
</li>
<li><p>exponential: Argument ratevec. In fda_2.0.2, this defaulted to 1. In fda_2.0.3, it will default to 0:1.
</p>
</li>
<li><p>fourier: Argument period defaults to diff(rangeval).
</p>
</li>
<li><p>monomial/power: Argument exponents. Default = 0:(nbasis-1). For monomial bases, exponents must be distinct nonnegative integers. For power bases, they must be distinct real numbers.
</p>
</li>
<li><p>para: some relevant parameters of &quot;SNPlm&quot; model.
</p>
</li></ul>
</td></tr>
<tr><td><code id="SNPinter_+3A_bsplines">Bsplines</code></td>
<td>
<p>an integer specifying the number of basis functions that constitutes the genetic effect function.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_norder">norder</code></td>
<td>
<p>an integer specifying the order of bsplines that constitutes the genetic effect function, which is one higher than their degree. The default of 4 gives cubic splines.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_intercept">intercept</code></td>
<td>
<p>should intercept(s) be fitted (TRUE) or set to zero (default = FALSE).</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_eps">eps</code></td>
<td>
<p>a numeric variable specifying the threshold at which the algorithm terminates, default is 1e-5.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_maxstep">maxstep</code></td>
<td>
<p>a numeric variable specifying the maximum iteration steps, default is 1e5.</p>
</td></tr>
<tr><td><code id="SNPinter_+3A_plot">Plot</code></td>
<td>
<p>should the estimated genetic effect function beta0(t) and interaction items betak(t) be plotted (TRUE) or not (default = FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An &quot;SNPinter&quot; object that contains the list of the following items.
</p>

<ul>
<li><p>alpha: estimated intercept value..
</p>
</li>
<li><p>gamma: estimated coefficients of the scalar covariates.
</p>
</li>
<li><p>b: estimated coefficients of the chosen basis functions for the genetic effect function beta0(t) and interaction items betak(t).
</p>
</li>
<li><p>betat: an &quot;fd&quot; object, representing the estimated genetic effect function beta(t) and interaction items betak(t).
</p>
</li>
<li><p>residuals: the residuals, that is response minus fitted values.
</p>
</li>
<li><p>fitted.values:  the fitted mean values.
</p>
</li></ul>



<h3>See Also</h3>

<p>See Also as <code><a href="#topic+simData1">simData1</a></code>, <code><a href="#topic+SNPcvinter">SNPcvinter</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 300
m &lt;- 30
simdata2 &lt;- simData2(n, m, seed = 123)
y &lt;- simdata2$y
z &lt;- simdata2$z
location &lt;- simdata2$location
X &lt;- simdata2$X
lambda1 &lt;- 0.05
lambda2 &lt;- sqrt(3)*lambda1
eta &lt;- 0
SNPinterres &lt;- SNPinter(y, z, location, X, lambda1, lambda2, eta, type1 = "Bspline",
                        nbasis1 = 5, params1 = 4, Bsplines = 5, norder = 4, intercept = TRUE,
                        eps = 1e-2, maxstep = 1e2, Plot = TRUE)
SNPinterres$alpha
SNPinterres$gamma
SNPinterres$b

</code></pre>

<hr>
<h2 id='SNPlm'>Revised functional linear regression analysis for the sequence (genotypes) data</h2><span id='topic+SNPlm'></span>

<h3>Description</h3>

<p>This function models the genetic effect of genetic variants by relating the genetic variant function to the phenotype adjusting for covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNPlm(y, z, location, X, type1, type2, nbasis1, nbasis2, params1, params2,
  intercept = FALSE, Plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNPlm_+3A_y">y</code></td>
<td>
<p>a numeric vector specifying the response variables.</p>
</td></tr>
<tr><td><code id="SNPlm_+3A_z">z</code></td>
<td>
<p>a matrix specifying the scalar covariates, with the number of rows equal to the number of samples.</p>
</td></tr>
<tr><td><code id="SNPlm_+3A_location">location</code></td>
<td>
<p>a numeric vector defining the sampling sites of the sequence data.</p>
</td></tr>
<tr><td><code id="SNPlm_+3A_x">X</code></td>
<td>
<p>a matrix specifying the sequence data, with the number of rows equal to the number of samples.</p>
</td></tr>
<tr><td><code id="SNPlm_+3A_type1">type1</code></td>
<td>
<p>a character specifying the type of the basis functions that constitutes the genetic variation function. The options are &quot;Bspline&quot;, &quot;Exponential&quot;, &quot;Fourier&quot;, &quot;Monomial&quot;, and &quot;Power&quot;.</p>
</td></tr>
<tr><td><code id="SNPlm_+3A_type2">type2</code></td>
<td>
<p>a character specifying the type of the basis functions that constitutes the genetic effect function. The options are &quot;Bspline&quot;, &quot;Exponential&quot;, &quot;Fourier&quot;, &quot;Monomial&quot;, and &quot;Power&quot;.</p>
</td></tr>
<tr><td><code id="SNPlm_+3A_nbasis1">nbasis1</code></td>
<td>
<p>an integer specifying the number of basis functions that constitutes the genetic variation function.</p>
</td></tr>
<tr><td><code id="SNPlm_+3A_nbasis2">nbasis2</code></td>
<td>
<p>an integer specifying the number of basis functions that constitutes the genetic effect function.</p>
</td></tr>
<tr><td><code id="SNPlm_+3A_params1">params1</code></td>
<td>
<p>in addition to rangeval1 (a vector of length 2 giving the lower and upper limits of the range of permissible values for the genetic variation function) and nbasis1, all bases have one or two parameters unique to that basis type or shared with one other;</p>
</td></tr>
<tr><td><code id="SNPlm_+3A_params2">params2</code></td>
<td>
<p>in addition to rangeval1 (a vector of length 2 giving the lower and upper limits of the range of permissible values for the genetic effect function) and nbasis1, all bases have one or two parameters unique to that basis type or shared with one other;
</p>

<ul>
<li><p>bspline: Argument norder = the order of the spline, which is one more than the degree of the polynomials used. This defaults to 4, which gives cubic splines.
</p>
</li>
<li><p>exponential: Argument ratevec. In fda_2.0.2, this defaulted to 1. In fda_2.0.3, it will default to 0:1.
</p>
</li>
<li><p>fourier: Argument period defaults to diff(rangeval).
</p>
</li>
<li><p>monomial/power: Argument exponents. Default = 0:(nbasis-1). For monomial bases, exponents must be distinct nonnegative integers. For power bases, they must be distinct real numbers.
</p>
</li></ul>
</td></tr>
<tr><td><code id="SNPlm_+3A_intercept">intercept</code></td>
<td>
<p>should intercept(s) be fitted (TRUE) or set to zero (default = FALSE).</p>
</td></tr>
<tr><td><code id="SNPlm_+3A_plot">Plot</code></td>
<td>
<p>should the estimated genetic effect function beta(t) be plotted (TRUE) or not (default = FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An &quot;SNPlm&quot; object that contains the list of the following items.
</p>

<ul>
<li><p>alpha: estimated intercept value.
</p>
</li>
<li><p>gamma: estimated coefficients of the scalar covariates.
</p>
</li>
<li><p>b: estimated coefficients of the chosen basis functions for the genetic effect function beta(t).
</p>
</li>
<li><p>betat: an &quot;fd&quot; object, representing the estimated genetic effect function beta(t).
</p>
</li>
<li><p>residuals: the residuals, that is response minus fitted values.
</p>
</li>
<li><p>fitted.values:  the fitted mean values.
</p>
</li>
<li><p>terms: the terms object used.
</p>
</li>
<li><p>model: if requested (the default), the model frame used.
</p>
</li>
<li><p>para: some relevant parameters of &quot;SNPlm&quot; model.
</p>
</li></ul>



<h3>See Also</h3>

<p>See Also as <code><a href="#topic+SNPgvf">SNPgvf</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(FunctanSNP)
n &lt;- 300
m &lt;- 30
simdata1 &lt;- simData1(n, m, seed = 123)
y &lt;- simdata1$y
z &lt;- simdata1$z
location &lt;- simdata1$location
X &lt;- simdata1$X
SNPlmres &lt;- SNPlm(y, z, location, X, type1 = "Bspline", type2 = "Bspline", nbasis1 = 5,
                  nbasis2 = 5, params1 = 4, params2 = 4, intercept = FALSE, Plot = TRUE)
SNPlmres$alpha
SNPlmres$gamma
SNPlmres$b
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
