<!DOCTYPE html><html><head><title>Help for package FuncDiv</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FuncDiv}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alpha_div_contrib'><p>Main function for computing contributional <strong>alpha</strong> diversity</p></a></li>
<li><a href='#beta_div_contrib'><p>Main function for computing contributional <strong>beta</strong> diversity</p></a></li>
<li><a href='#compute_alpha_div'><p>Convenience function for running default alpha diversity metrics on a single vector input</p></a></li>
<li><a href='#contrib_to_multitab'><p>Utility function to convert from contributional to multi-table input objects</p></a></li>
<li><a href='#func_abun_crossproduct'><p>Utility function to get community-wide function abundance table</p></a></li>
<li><a href='#FuncDiv_alpha_metrics'><p>List object containing the functions to compute the default alpha diversity metrics</p></a></li>
<li><a href='#multitab_to_contrib'><p>Utility function to convert from multi-table objects to contributional table</p></a></li>
<li><a href='#subset_func_and_abun_tables'><p>Utility function to subset function copy number and taxonomic abundance tables</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Compute Contributional Diversity Metrics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Compute alpha and beta contributional diversity metrics,
    which is intended for linking taxonomic and functional microbiome
    data. See 'GitHub' repository for the tutorial:
    <a href="https://github.com/gavinmdouglas/FuncDiv/wiki">https://github.com/gavinmdouglas/FuncDiv/wiki</a>. Citation: Gavin M.
    Douglas, Sunu Kim, Morgan G. I. Langille, B. Jesse Shapiro (2023)
    &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtac809">doi:10.1093/bioinformatics/btac809</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/AGPL-3">AGPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, collapse, data.table, parallel, parallelDist, Rcpp,
RcppParallel, RcppXPtrUtils, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, RcppParallel</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-07 16:26:31 UTC; gavin</td>
</tr>
<tr>
<td>Author:</td>
<td>Gavin Douglas <a href="https://orcid.org/0000-0001-5164-6707"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gavin Douglas &lt;gavinmdouglas@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-08 19:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='alpha_div_contrib'>Main function for computing contributional <strong>alpha</strong> diversity</h2><span id='topic+alpha_div_contrib'></span>

<h3>Description</h3>

<p>Based on joint taxa-function input data (i.e., contributional data), a dataframe
will be returned for each specified metric, which will contain the metric values for all function and sample combinations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha_div_contrib(
  metrics,
  func_tab = NULL,
  abun_tab = NULL,
  contrib_tab = NULL,
  in_tree = NULL,
  ncores = 1,
  replace_NA = FALSE,
  custom_metric_functions = NULL,
  samp_colname = "sample",
  func_colname = "function.",
  taxon_colname = "taxon",
  abun_colname = "taxon_abun"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alpha_div_contrib_+3A_metrics">metrics</code></td>
<td>
<p>alpha diversity metrics to compute. Must either be names of functions in <code>FuncDiv_alpha_metrics</code>, or alternatively in <code>custom_metric_functions</code>, if specified.</p>
</td></tr>
<tr><td><code id="alpha_div_contrib_+3A_func_tab">func_tab</code></td>
<td>
<p>data.frame object containing function copy numbers, with rows as functions and columns as taxa. Required if <code>abun_tab</code> is specified, and is mutually exclusive with <code>contrib_tab</code>.</p>
</td></tr>
<tr><td><code id="alpha_div_contrib_+3A_abun_tab">abun_tab</code></td>
<td>
<p>data.frame object containing taxonomic abundances across samples, with rows as taxa and columns as samples. Required if <code>func_tab</code> is specified, and is mutually exclusive with <code>contrib_tab</code>.</p>
</td></tr>
<tr><td><code id="alpha_div_contrib_+3A_contrib_tab">contrib_tab</code></td>
<td>
<p>data.frame object containing combined taxa abundances and function copy numbers across taxa. Must contain columns corresponding to the sample ids, function ids, taxa ids, and taxa
abundances within samples. These column names are specified by the <code>samp_colname</code>, <code>func_colname</code>, <code>taxon_colname</code>, and <code>abun_colname</code>, respectively. Mutually exclusive with <code>abun_tab</code> and <code>func_tab</code>.</p>
</td></tr>
<tr><td><code id="alpha_div_contrib_+3A_in_tree">in_tree</code></td>
<td>
<p>phylo object to use if <code>faiths_pd</code> is specified.</p>
</td></tr>
<tr><td><code id="alpha_div_contrib_+3A_ncores">ncores</code></td>
<td>
<p>integer indicating number of cores to use for parallelizable steps.</p>
</td></tr>
<tr><td><code id="alpha_div_contrib_+3A_replace_na">replace_NA</code></td>
<td>
<p>Boolean vector of length one, indicating whether all NA's in the output of all metrics should be converted to 0's. Note that this done automatically done for <code>richness</code> either way.</p>
</td></tr>
<tr><td><code id="alpha_div_contrib_+3A_custom_metric_functions">custom_metric_functions</code></td>
<td>
<p>List object containing custom alpha diversity metric functions. This overrides <code>FuncDiv_alpha_metrics</code> when specified. The list element names must correspond to at
least the names indicated by the <code>metrics</code> parameter.</p>
</td></tr>
<tr><td><code id="alpha_div_contrib_+3A_samp_colname">samp_colname</code></td>
<td>
<p>sample id column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
<tr><td><code id="alpha_div_contrib_+3A_func_colname">func_colname</code></td>
<td>
<p>function id column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
<tr><td><code id="alpha_div_contrib_+3A_taxon_colname">taxon_colname</code></td>
<td>
<p>taxon id column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
<tr><td><code id="alpha_div_contrib_+3A_abun_colname">abun_colname</code></td>
<td>
<p>taxonomic abundance (within each sample) column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Input data can be either a separate function copy number and taxonomic abundance table, or a joint contributional table.
By default, specified metrics must be one of <code>names(FuncDiv_alpha_metrics)</code>. However, custom alpha diversity metric functions
can be specified with the <code>custom_metric_functions</code> parameter.
</p>
<p>Note that the taxonomic abundances can be relative abundance, read counts, or transformed in another way. However, note that some default metrics
are only compatible with count data (see <code>?FuncDiv_alpha_metrics</code>).
</p>


<h3>Value</h3>

<p>a list, containing one dataframe for each specified alpha diversity metric.
In each dataframe, rows are functions and samples are columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># First, simulate some (non-realistic) data.
set.seed(123)
test_tree &lt;- ape::rtree(100)
test_abun &lt;- data.frame(matrix(rnorm(500), nrow = 100, ncol = 5))
rownames(test_abun) &lt;- test_tree$tip.label
colnames(test_abun) &lt;- c("sample1", "sample2", "sample3", "sample4", "sample5")
test_abun[test_abun &lt; 0] &lt;- 0
test_func &lt;- data.frame(matrix(sample(c(0L, 1L), 200, replace = TRUE),
                               nrow = 2, ncol = 100))
colnames(test_func) &lt;- test_tree$tip.label
rownames(test_func) &lt;- c("func1", "func2")

# Compute alpha diversity, based on (observed) richness, Faith's phylogenetic
# diversity, and the Gini-Simpson Index.
contrib_alpha &lt;- alpha_div_contrib(metrics = c("richness",  "faiths_pd", "gini_simpson_index"),
                                   func_tab = test_func,
                                   abun_tab = test_abun,
                                   in_tree = test_tree,
                                   ncores = 1)

# Print out computed Gini-Simpson Index values.
contrib_alpha$gini_simpson_index

</code></pre>

<hr>
<h2 id='beta_div_contrib'>Main function for computing contributional <strong>beta</strong> diversity</h2><span id='topic+beta_div_contrib'></span>

<h3>Description</h3>

<p>Based on joint taxa-function input data (i.e., contributional data), the beta diversity (i.e., inter-sample distance or divergence)
will be computed for the subset of taxa encoding each individual function separately. A large List object containing all these tables
can be returned, or alternatively these tables will be written to the disk as plain-text files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beta_div_contrib(
  metrics = NULL,
  func_tab = NULL,
  abun_tab = NULL,
  contrib_tab = NULL,
  in_tree = NULL,
  func_ids = NULL,
  return_objects = FALSE,
  write_outfiles = FALSE,
  outdir = NULL,
  ncores = 1,
  samp_colname = "sample",
  func_colname = "function.",
  taxon_colname = "taxon",
  abun_colname = "taxon_abun"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="beta_div_contrib_+3A_metrics">metrics</code></td>
<td>
<p>beta diversity metrics to compute. Must be default metric computed by <code>parallelDist::parDist</code> or one of &quot;weighted_unifrac&quot;, &quot;unweighted_unifrac&quot;, or &quot;jensen_shannon_div&quot;.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_func_tab">func_tab</code></td>
<td>
<p>data.frame object containing function copy numbers, with rows as functions and columns as taxa. Required if <code>abun_tab</code> is specified, and is mutually exclusive with <code>contrib_tab</code>.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_abun_tab">abun_tab</code></td>
<td>
<p>data.frame object containing taxonomic abundances across samples, with rows as taxa and columns as samples. Required if <code>func_tab</code> is specified, and is mutually exclusive with <code>contrib_tab</code>.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_contrib_tab">contrib_tab</code></td>
<td>
<p>data.frame object containing combined taxa abundances and function copy numbers across taxa. Must contain columns corresponding to the sample ids, function ids, taxa ids, and taxa
abundances within samples. These column names are specified by the <code>samp_colname</code>, <code>func_colname</code>, <code>taxon_colname</code>, and <code>abun_colname</code>, respectively.Mutually exclusive with <code>abun_tab</code> and <code>func_tab</code>.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_in_tree">in_tree</code></td>
<td>
<p>phylo object to use if <code>weighted_unifrac</code> or <code>unweighted_unifrac</code> are specified.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_func_ids">func_ids</code></td>
<td>
<p>character vector specifying subset of function ids to include for analysis. Will analyze all functions present if this is not specified.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_return_objects">return_objects</code></td>
<td>
<p>Boolean vector of length one, specifying whether function should return a list of all output distance tables (nested by metric name, and then by function id). Incompatible with <code>write_outfiles</code>.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_write_outfiles">write_outfiles</code></td>
<td>
<p>Boolean vector of length one, specifying whether function write all distance tables to plain-text files in the specified <code>outdir</code> location. Incompatible with <code>return_objects</code>.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_outdir">outdir</code></td>
<td>
<p>character vector of length one, indicating where to save output files if <code>write_outfiles = TRUE</code>.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_ncores">ncores</code></td>
<td>
<p>integer indicating number of cores to use for parallelizable steps.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_samp_colname">samp_colname</code></td>
<td>
<p>sample id column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_func_colname">func_colname</code></td>
<td>
<p>function id column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_taxon_colname">taxon_colname</code></td>
<td>
<p>taxon id column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
<tr><td><code id="beta_div_contrib_+3A_abun_colname">abun_colname</code></td>
<td>
<p>taxonomic abundance (within each sample) column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Input data can be either a separate function copy number and taxonomic abundance table, or a joint contributional table.
Metrics must be one of &quot;weighted_unifrac&quot;, &quot;unweighted_unifrac&quot;, &quot;jensen_shannon_div&quot;, or a default metric available through the <code>parallelDist::parDist</code> function. See <code>?parallelDist::parDist</code> for a description of all default metrics.
</p>
<p>The taxonomic abundances will be converted to relative abundances prior to computing inter-sample distances.
</p>


<h3>Value</h3>

<p>differs depending on the <code>return_objects</code> and <code>write_outfiles</code> parameters.
</p>
<p>If <code>return_objects = TRUE</code>, then a nested List will be returned.
Each specific beta diversity metric will be the first level, and the functions are the second level
(e.g., contrib_beta$binary$func2).
</p>
<p>If <code>write_outfiles</code> then a character vector will be returned, indicating where the output tables were written.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># First, simulate some (non-realistic) data.
set.seed(123)
test_tree &lt;- ape::rtree(100)
test_abun &lt;- data.frame(matrix(rnorm(500), nrow = 100, ncol = 5))
rownames(test_abun) &lt;- test_tree$tip.label
colnames(test_abun) &lt;- c("sample1", "sample2", "sample3", "sample4", "sample5")
test_abun[test_abun &lt; 0] &lt;- 0
test_func &lt;- data.frame(matrix(sample(c(0L, 1L), 200, replace = TRUE),
                               nrow = 2, ncol = 100))
colnames(test_func) &lt;- test_tree$tip.label
rownames(test_func) &lt;- c("func1", "func2")

# Compute beta diversity, based on Weighted UniFrac and Jaccard distances
# (i.e., "binary").
contrib_beta &lt;- beta_div_contrib(metrics = c("weighted_unifrac", "binary"),
                                 func_tab = test_func,
                                 abun_tab = test_abun,
                                 in_tree = test_tree,
                                 return_objects = TRUE,
                                 ncores = 1)

# Parse beta diversity distance list value for a specific function (func2) and
# distance metric (Jaccard).
contrib_beta$binary$func2

</code></pre>

<hr>
<h2 id='compute_alpha_div'>Convenience function for running default alpha diversity metrics on a single vector input</h2><span id='topic+compute_alpha_div'></span>

<h3>Description</h3>

<p>This is a simple wrapper for <code>FuncDiv_alpha_metrics</code>, and you can see more details with <code>?FuncDiv_alpha_metrics</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_alpha_div(x, metric, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_alpha_div_+3A_x">x</code></td>
<td>
<p>input vector. Either class numeric (representing abundance of categories [e.g., microbes]) or character (indicating which taxa are present, which is required for <code>faiths_pd</code>).</p>
</td></tr>
<tr><td><code id="compute_alpha_div_+3A_metric">metric</code></td>
<td>
<p>alpha diversity metric to compute. Must be one of <code>names(FuncDiv_alpha_metrics)</code>.</p>
</td></tr>
<tr><td><code id="compute_alpha_div_+3A_...">...</code></td>
<td>
<p>included so that functions with single arguments will not throw errors if <code>tree</code> is included (and ignored). This should be a phylo object to use in case of <code>faiths_pd</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions all have a single input: a numeric vector containing taxa abundances within a given sample.
The exception is for <code>faiths_pd</code>, which expects a character vector of taxa labels that are present, as well as a tree (phylo object),
which must contain all these specified taxa labels as tip labels.
</p>


<h3>Value</h3>

<p>numeric vector with alpha diversity value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Most metrics just require an input vector of abundances.
test_abun &lt;- c(0, NA, 1, 2, 10, 4)
compute_alpha_div(x = test_abun, metric = "richness")

# Note that the input for computing Faith's PD is different.
# Get a randomly generated tree:
test_tree &lt;- ape::rtree(n = 50)
test_present_tips &lt;- c('t1', 't2', 't3')
compute_alpha_div(x = test_present_tips, metric = "faiths_pd", tree = test_tree)

</code></pre>

<hr>
<h2 id='contrib_to_multitab'>Utility function to convert from contributional to multi-table input objects</h2><span id='topic+contrib_to_multitab'></span>

<h3>Description</h3>

<p>Converts from contributional-type table (i.e., a single, long table with joint taxa/function information) to separate taxa abundance and function copy number tables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contrib_to_multitab(
  contrib_tab,
  samp_colname = "sample",
  func_colname = "function.",
  abun_colname = "taxon_abun",
  taxon_colname = "taxon",
  copy.num_colname = "genome_function_count"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="contrib_to_multitab_+3A_contrib_tab">contrib_tab</code></td>
<td>
<p>data.frame object containing combined taxa abundances and function copy numbers across taxa. Must contain columns corresponding to the sample ids, function ids, taxa ids, and taxa
abundances within samples. These column names are specified by the <code>samp_colname</code>, <code>func_colname</code>, <code>taxon_colname</code>, <code>abun_colname</code>, and <code>copy.num_colname</code>, respectively.</p>
</td></tr>
<tr><td><code id="contrib_to_multitab_+3A_samp_colname">samp_colname</code></td>
<td>
<p>sample id column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
<tr><td><code id="contrib_to_multitab_+3A_func_colname">func_colname</code></td>
<td>
<p>function id column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
<tr><td><code id="contrib_to_multitab_+3A_abun_colname">abun_colname</code></td>
<td>
<p>taxonomic abundance (within each sample) column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
<tr><td><code id="contrib_to_multitab_+3A_taxon_colname">taxon_colname</code></td>
<td>
<p>taxon id column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
<tr><td><code id="contrib_to_multitab_+3A_copy.num_colname">copy.num_colname</code></td>
<td>
<p>function copy number column name of <code>contrib_tab</code> input data.frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with taxon abundance (<code>taxon_abun</code>) and function copy number (<code>function_copy_num</code>) data.frames as separate elements.
</p>

<hr>
<h2 id='func_abun_crossproduct'>Utility function to get community-wide function abundance table</h2><span id='topic+func_abun_crossproduct'></span>

<h3>Description</h3>

<p>Takes in table of function copy numbers across taxa and table of taxa abundances across samples.
I.e., it represents the multiplication of the function copy numbers by the abundances of the taxa within each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>func_abun_crossproduct(func_tab, abun_tab)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="func_abun_crossproduct_+3A_func_tab">func_tab</code></td>
<td>
<p>data.frame object containing function copy numbers, with rows as functions and columns as taxa.</p>
</td></tr>
<tr><td><code id="func_abun_crossproduct_+3A_abun_tab">abun_tab</code></td>
<td>
<p>data.frame object containing taxonomic abundances across samples, with rows as taxa and columns as samples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame representing the <em>unnormalized</em> community-wide abundances of functions across samples.
</p>

<hr>
<h2 id='FuncDiv_alpha_metrics'>List object containing the functions to compute the default alpha diversity metrics</h2><span id='topic+FuncDiv_alpha_metrics'></span>

<h3>Description</h3>

<p>These functions are used by the <code>alpha_div_contrib</code> function to compute contributional diversity,
but can be used for any arbitrary input vector as well to compute standard alpha diversity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FuncDiv_alpha_metrics
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 15.
</p>


<h3>Details</h3>

<p>The metrics were primarily taken from definitions provided by <code>scikit-bio</code> Python package, as well as the <code>vegan</code> and <code>picante</code> R packages.
The functions are provided as elements of this list, so that it is more convenient to call them programatically.
All available alpha diversity metrics can be seen by typing <code>names(FuncDiv_alpha_metrics)</code>.
The code to compute each metric can be inspected for each function, for instance, for richness, by typing: <code>FuncDiv_alpha_metrics$richness</code>.
</p>
<p>These functions all have a single input: a numeric vector containing taxa abundances within a given sample.
The exception is for <code>faiths_pd</code>, which expects a character vector of taxa labels that are present, as well as a tree (phylo object),
which must contain all these specified taxa labels as tip labels.
</p>
<p>Note that not all these metrics are appropriate for relative abundance data. In particular, these metrics expect count data (e.g., read counts)
corresponding to the number of occurrences of each category (e.g., each microbe): <code>menhinicks_richness</code>, <code>mcintoshs_evenness</code>, <code>mcintoshs_dominance</code>,
<code>margalefs_richness</code>, and <code>fishers_alpha</code>.
</p>


<h3>Value</h3>

<p>numeric vector with alpha diversity value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Most metrics just require an input vector of abundances.
test_abun &lt;- c(0, NA, 1, 2, 10, 4)
FuncDiv_alpha_metrics[["richness"]](test_abun)

# Note that the input for computing Faith's PD is different.
# Get a randomly generated tree:
test_tree &lt;- ape::rtree(n = 50)
test_present_tips &lt;- c('t1', 't2', 't3')
FuncDiv_alpha_metrics[["faiths_pd"]](test_present_tips, test_tree)

</code></pre>

<hr>
<h2 id='multitab_to_contrib'>Utility function to convert from multi-table objects to contributional table</h2><span id='topic+multitab_to_contrib'></span>

<h3>Description</h3>

<p>Converts from separate taxa abundance and function copy number table input style to contributional-type table (i.e., a single, long table with joint taxa/function information).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multitab_to_contrib(
  func_tab,
  abun_tab,
  ncores = 1,
  samp_colname = "sample",
  func_colname = "function.",
  abun_colname = "taxon_abun",
  taxon_colname = "taxon",
  copy.num_colname = "genome_function_count"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multitab_to_contrib_+3A_func_tab">func_tab</code></td>
<td>
<p>data.frame object containing function copy numbers, with rows as functions and columns as taxa.</p>
</td></tr>
<tr><td><code id="multitab_to_contrib_+3A_abun_tab">abun_tab</code></td>
<td>
<p>data.frame object containing taxonomic abundances across samples, with rows as taxa and columns as samples.</p>
</td></tr>
<tr><td><code id="multitab_to_contrib_+3A_ncores">ncores</code></td>
<td>
<p>integer specifying number of cores to use for parallizable steps.</p>
</td></tr>
<tr><td><code id="multitab_to_contrib_+3A_samp_colname">samp_colname</code></td>
<td>
<p>sample id column name of <code>contrib_tab</code> output data.frame.</p>
</td></tr>
<tr><td><code id="multitab_to_contrib_+3A_func_colname">func_colname</code></td>
<td>
<p>function id column name of <code>contrib_tab</code> output data.frame.</p>
</td></tr>
<tr><td><code id="multitab_to_contrib_+3A_abun_colname">abun_colname</code></td>
<td>
<p>taxonomic abundance (within each sample) column name of <code>contrib_tab</code> output data.frame.</p>
</td></tr>
<tr><td><code id="multitab_to_contrib_+3A_taxon_colname">taxon_colname</code></td>
<td>
<p>taxon id column name of <code>contrib_tab</code> output data.frame.</p>
</td></tr>
<tr><td><code id="multitab_to_contrib_+3A_copy.num_colname">copy.num_colname</code></td>
<td>
<p>function copy number (within each taxa) column name of <code>contrib_tab</code> output data.frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame in contributional format (i.e., single, long-format version of both input tables).
</p>

<hr>
<h2 id='subset_func_and_abun_tables'>Utility function to subset function copy number and taxonomic abundance tables</h2><span id='topic+subset_func_and_abun_tables'></span>

<h3>Description</h3>

<p>The input tables will be returned except subset to the same taxa ids. Any functions and / or samples that are totally absent after this step will be dropped.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset_func_and_abun_tables(func_table, abun_table, func_ids = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset_func_and_abun_tables_+3A_func_table">func_table</code></td>
<td>
<p>data.frame object containing function copy numbers, with rows as functions and columns as taxa.</p>
</td></tr>
<tr><td><code id="subset_func_and_abun_tables_+3A_abun_table">abun_table</code></td>
<td>
<p>data.frame object containing taxonomic abundances across samples, with rows as taxa and columns as samples.</p>
</td></tr>
<tr><td><code id="subset_func_and_abun_tables_+3A_func_ids">func_ids</code></td>
<td>
<p>optional character vector of function ids to retain (all other rows of <code>func_tab</code> will be removed).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing subsetted function and abundance data.frames as separate elements.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
