<!DOCTYPE html><html><head><title>Help for package fwsim</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {fwsim}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fwsim'><p>Fisher-Wright Population Simulation</p></a></li>
<li><a href='#fwsim-package'>
<p>Haplotype tools</p></a></li>
<li><a href='#init_mutmodel'>
<p>init_mutmodel</p></a></li>
<li><a href='#mutmodel'><p>Mutation model logic</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.3.4</td>
</tr>
<tr>
<td>Title:</td>
<td>Fisher-Wright Population Simulation</td>
</tr>
<tr>
<td>Author:</td>
<td>Mikkel Meyer Andersen and Poul Svante Eriksen</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mikkel Meyer Andersen &lt;mikl@math.aau.dk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulates a population under the Fisher-Wright model (fixed or stochastic population size) with a one-step neutral mutation process (stepwise mutation model, logistic mutation model and exponential mutation model supported). The stochastic population sizes are random Poisson distributed and different kinds of population growth are supported. For the stepwise mutation model, it is possible to specify locus and direction specific mutation rate (in terms of upwards and downwards mutation rate). Intermediate generations can be saved in order to study e.g. drift.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11), methods</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++11</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-10-05 13:00:56 UTC; mikl</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-10-05 13:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='fwsim'>Fisher-Wright Population Simulation</h2><span id='topic+fwsim'></span><span id='topic+fwsim_fixed'></span><span id='topic+print.fwsim'></span><span id='topic+summary.fwsim'></span><span id='topic+plot.fwsim'></span>

<h3>Description</h3>

<p>This package provides tools to simulate a population under the Fisher-Wright model with a stepwise neutral mutation process on <code class="reqn">r</code> loci, where mutations on loci happen independently. The population sizes are either fixed (traditional/original Fisher-Wright model) or random Poisson distributed with exponential growth supported. Intermediate generations can be saved in order to study e.g. drift.
</p>
<p>For stochastic population sizes:
Model described in detail at <a href="http://arxiv.org/abs/1210.1773">http://arxiv.org/abs/1210.1773</a>. Let <code class="reqn">M</code> be the population size at generation <code class="reqn">i</code> and <code class="reqn">N</code> the population size at generation <code class="reqn">i + 1</code>.
Then we assume that <code class="reqn">N</code> conditionally on <code class="reqn">M</code> is <code class="reqn">\mbox{Poisson}(\alpha M)</code> distributed for <code class="reqn">\alpha &gt; 0</code> (<code class="reqn">\alpha &gt; 1</code> gives expected growth and <code class="reqn">0 &lt; \alpha &lt; 1</code> gives expected decrease).
</p>
<p>For each haplotype <code class="reqn">x</code> occuring <code class="reqn">m</code> times in the <code class="reqn">i</code>'th generation, the number
of children <code class="reqn">n</code> is <code class="reqn">\mbox{Poisson}(\alpha m)</code> distributed independently of other haplotypes. 
It then follows that the sum of the number of haplotypes follows a <code class="reqn">\mbox{Poisson}(\alpha M)</code> distribution (as just stated in the previous paragraph) and that <code class="reqn">n</code> conditionally on <code class="reqn">N</code> follows a <code class="reqn">\mbox{Binomial}(N, m/M)</code> as expected.
</p>
<p>The mutation model can be e.g. the stepwise neutral mutation model. See <code><a href="#topic+init_mutmodel">init_mutmodel</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fwsim(G, H0, N0, mutmodel, alpha = 1.0, SNP = FALSE, 
  save_generations = NULL, progress = TRUE, trace = FALSE, ensure_children = FALSE, ...)
fwsim_fixed(G, H0, N0, mutmodel, SNP = FALSE, 
  save_generations = NULL, progress = TRUE, trace = FALSE, ...)
## S3 method for class 'fwsim'
print(x, ...)
## S3 method for class 'fwsim'
summary(object, ...)
## S3 method for class 'fwsim'
plot(x, which = 1L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fwsim_+3A_g">G</code></td>
<td>
<p> number of generations to evolve (integer, remember postfix L).</p>
</td></tr>
<tr><td><code id="fwsim_+3A_h0">H0</code></td>
<td>
<p> haplotypes of the initial population. Must be a vector or matrix (if more than one initial haplotype). The number of loci is the length or number of columns of <code>H0</code>.</p>
</td></tr>
<tr><td><code id="fwsim_+3A_n0">N0</code></td>
<td>
<p> count of the <code>H0</code> haplotypes. The i'th element is the count of the haplotype <code>H0[i, ]</code>. <code>sum(N0)</code> is the size of initial population.</p>
</td></tr>
<tr><td><code id="fwsim_+3A_mutmodel">mutmodel</code></td>
<td>
<p> a <code>mutmodel</code> object created with <code><a href="#topic+init_mutmodel">init_mutmodel</a></code>. Alternatively, a numeric vector of length r of mutation probabilities (this will create a stepwise mutation model with r loci divide the mutation probabilities evenly between upwards and downwards mutation).</p>
</td></tr>
<tr><td><code id="fwsim_+3A_alpha">alpha</code></td>
<td>
<p> vector of length 1 or <code>G</code> of growth factors (1 correspond to expected constant population size). If length 1, the value is reused in creating a vector of length <code>G</code>.</p>
</td></tr>
<tr><td><code id="fwsim_+3A_snp">SNP</code></td>
<td>
<p> to make alleles modulus 2 to immitate SNPs.</p>
</td></tr>
<tr><td><code id="fwsim_+3A_save_generations">save_generations</code></td>
<td>
<p> to save intermediate populations. <code>NULL</code> means that no intermediate population will be saved. Else, a vector of the generation numbers to save.</p>
</td></tr>
<tr><td><code id="fwsim_+3A_progress">progress</code></td>
<td>
<p> whether to print progress of the evolution.</p>
</td></tr>
<tr><td><code id="fwsim_+3A_trace">trace</code></td>
<td>
<p> whether to print trace of the evolution (more verbose than <code>progress</code>).</p>
</td></tr>
<tr><td><code id="fwsim_+3A_ensure_children">ensure_children</code></td>
<td>
<p> Ensures that every generation has at least one child; implemented by getting <code class="reqn">\mbox{Poisson}(\alpha M) + 1</code> children.</p>
</td></tr>
<tr><td><code id="fwsim_+3A_x">x</code></td>
<td>
<p>A <code>fwsim</code> object.</p>
</td></tr>
<tr><td><code id="fwsim_+3A_object">object</code></td>
<td>
<p>A <code>fwsim</code> object.</p>
</td></tr>
<tr><td><code id="fwsim_+3A_which">which</code></td>
<td>
<p>A number specifying the plot (currently only 1: the actual population sizes vs the expected sizes).</p>
</td></tr>
<tr><td><code id="fwsim_+3A_...">...</code></td>
<td>
<p> not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>fwsim</code> object with elements
</p>
<table>
<tr><td><code>pars</code></td>
<td>
<p> the parameters used for the simulation</p>
</td></tr>
<tr><td><code>saved_populations</code></td>
<td>
<p> a list of haplotypes in the intermediate populations</p>
</td></tr>
<tr><td><code>population</code></td>
<td>
<p> haplotypes in the end population after <code>G</code> generations</p>
</td></tr>
<tr><td><code>pop_sizes</code></td>
<td>
<p> the population size for each generation</p>
</td></tr>
<tr><td><code>expected_pop_sizes</code></td>
<td>
<p> the expected population size for each generation</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mikkel Meyer Andersen &lt;mikl@math.aau.dk&gt; and Poul Svante Eriksen</p>


<h3>Examples</h3>

<pre><code class='language-R'># SMM (stepwise mutation model) example
set.seed(1)
fit &lt;- fwsim(G = 100L, H0 = c(0L, 0L, 0L), N0 = 10000L, 
  mutmodel = c(Loc1 = 0.001, Loc2 = 0.002, Loc3 = 0.003))
summary(fit)
fit

# SMM (stepwise mutation model) example
H0 &lt;- matrix(c(0L, 0L, 0L), 1L, 3L, byrow = TRUE)

mutmodel &lt;- init_mutmodel(modeltype = 1L, 
  mutpars = matrix(c(c(0.003, 0.001), rep(0.004, 2), rep(0.001, 2)), 
                   ncol = 3, 
                   dimnames = list(NULL, c("DYS19", "DYS389I", "DYS391"))))
mutmodel

set.seed(1)
fit &lt;- fwsim(G = 100L, H0 = H0, N0 = 10000L, mutmodel = mutmodel)

xtabs(N ~ DYS19 + DYS389I, fit$population)
plot(1L:fit$pars$G, fit$pop_sizes, type = "l", 
  ylim = range(range(fit$pop_sizes), range(fit$expected_pop_sizes)))
points(1L:fit$pars$G, fit$expected_pop_sizes, type = "l", col = "red")

set.seed(1)
fit_fixed &lt;- fwsim_fixed(G = 100L, H0 = H0, N0 = 10000L, mutmodel = mutmodel)

# LMM (logistic mutation model) example
mutpars.locus1 &lt;- c(0.149,   2.08,    18.3,   0.149,   0.374,   27.4) # DYS19
mutpars.locus2 &lt;- c(0.500,   1.18,    18.0,   0.500,   0.0183,  349)  # DYS389I
mutpars.locus3 &lt;- c(0.0163,  17.7,    11.1,   0.0163,  0.592,   14.1) # DYS391
mutpars &lt;- matrix(c(mutpars.locus1, mutpars.locus2, mutpars.locus3), ncol = 3)
colnames(mutpars) &lt;- c("DYS19", "DYS389I", "DYS391")
mutmodel &lt;- init_mutmodel(modeltype = 2L, mutpars = mutpars)
mutmodel

set.seed(1)
H0_LMM &lt;- matrix(c(15L, 13L, 10L), 1L, 3L, byrow = TRUE)
fit_LMM &lt;- fwsim(G = 100L, H0 = H0_LMM, N0 = 10000L, mutmodel = mutmodel)
xtabs(N ~ DYS19 + DYS389I, fit_LMM$population)
</code></pre>

<hr>
<h2 id='fwsim-package'>
Haplotype tools
</h2><span id='topic+fwsim-package'></span>

<h3>Description</h3>

<p>Tools for analysing haplotypes, e.g. population simulation.
</p>


<h3>Author(s)</h3>

<p>Mikkel Meyer Andersen
</p>
<p>Maintainer: Mikkel Meyer Andersen &lt;mikl@math.aau.dk&gt;
</p>

<hr>
<h2 id='init_mutmodel'>
init_mutmodel
</h2><span id='topic+init_mutmodel'></span><span id='topic+print.mutmodel'></span>

<h3>Description</h3>

<p>Method to initialise a mutation model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_mutmodel(modeltype = 1, mutpars = NULL, ...)
## S3 method for class 'mutmodel'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="init_mutmodel_+3A_modeltype">modeltype</code></td>
<td>
<pre>1: SMM (traditional single-step mutation model).
2: LMM (Logistic mutation model introduced in Jochens (2011) 
   'Empirical Evaluation Reveals Best Fit of a Logistic Mutation 
   Model for Human Y-Chromosomal Microsatellites').
3: Exponential mutation model (unpublished).</pre></td></tr>
<tr><td><code id="init_mutmodel_+3A_mutpars">mutpars</code></td>
<td>
<p>A matrix specifying the mutation parameters for each locus. Rows are parameters and columns are loci. If a vector, the same values are used for all loci.</p>
</td></tr>
<tr><td><code id="init_mutmodel_+3A_x">x</code></td>
<td>
<p>A <code>mutmodel</code> object.</p>
</td></tr>
<tr><td><code id="init_mutmodel_+3A_...">...</code></td>
<td>
<p> not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Mutation parameters for each locus. </p>
<pre>
    Mutmodel 1 (SMM): 2 parameters per locus
      P(i -&gt; i-1) = mu\_d
      P(i -&gt; i+1) = mu\_u
      P(i -&gt; i)   = 1 - P(i -&gt; i-1) - P(i -&gt; i+1)
                  = 1 - mu\_d - mu\_u

      mutpars[1, locus]: mu\_d
      mutpars[2, locus]: mu\_u

    Mutmodel 2 (LMM): 6 parameters per locus
      P(i -&gt; i-1) = gamma\_d / (1 + exp(alpha\_d*(beta\_d - i)))
      P(i -&gt; i+1) = gamma\_u / (1 + exp(alpha\_u*(beta\_u - i)))
      P(i -&gt; i)   = 1 - P(i -&gt; i-1) - P(i -&gt; i+1)

      mutpars[1, locus]: gamma\_d
      mutpars[2, locus]: alpha\_d
      mutpars[3, locus]: beta\_d
      mutpars[4, locus]: gamma\_u
      mutpars[5, locus]: alpha\_u
      mutpars[6, locus]: beta\_u

   Mutmodel 3 (EMM): 4 parameters per locus
      P(i -&gt; i-1) = 1/((1 + exp(a + b*i))*(1 + exp(alpha + beta*i)))
      P(i -&gt; i+1) = exp(alpha+beta*i)/((1+exp(a+b*i))*(1+exp(alpha+beta*i)))
      P(i -&gt; i)   = 1 - P(i -&gt; i-1) - P(i -&gt; i+1)
                  = exp(a + b*i)/(1 + exp(a + b*i))

      mutpars[1, locus]: a
      mutpars[2, locus]: b
      mutpars[3, locus]: alpha
      mutpars[4, locus]: beta
  </pre>


<h3>Value</h3>

<p>A <code>mutmodel</code> object (a list with entires <code>modeltype</code> and <code>mutpars</code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mutpars &lt;- matrix(c(c(0.003, 0.001), rep(0.004, 2), rep(0.001, 2)), ncol = 3)
colnames(mutpars) &lt;- c("DYS19", "DYS389I", "DYS391")
mutmodel &lt;- init_mutmodel(modeltype = 1L, mutpars = mutpars)
mutmodel

mutpars.locus1 &lt;- c(0.149,   2.08,    18.3,   0.149,   0.374,   27.4) # DYS19
mutpars.locus2 &lt;- c(0.500,   1.18,    18.0,   0.500,   0.0183,  349)  # DYS389I
mutpars.locus3 &lt;- c(0.0163,  17.7,    11.1,   0.0163,  0.592,   14.1) # DYS391
mutpars &lt;- matrix(c(mutpars.locus1, mutpars.locus2, mutpars.locus3), ncol = 3)
colnames(mutpars) &lt;- c("DYS19", "DYS389I", "DYS391")
mutmodel &lt;- init_mutmodel(modeltype = 2L, mutpars = mutpars)
mutmodel
</code></pre>

<hr>
<h2 id='mutmodel'>Mutation model logic</h2><span id='topic+mutmodel_not_mut'></span><span id='topic+mutmodel_dw_mut'></span><span id='topic+mutmodel_up_mut'></span><span id='topic+approx_stationary_dist'></span>

<h3>Description</h3>

<p>Functions for mutation model logic, e.g. probability of downwards and upwards mutations etc.</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutmodel_not_mut(mutmodel, locus, alleles)
mutmodel_dw_mut(mutmodel, locus, alleles)
mutmodel_up_mut(mutmodel, locus, alleles)
approx_stationary_dist(mutmodel, alleles)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutmodel_+3A_mutmodel">mutmodel</code></td>
<td>
<p> a <code>mutmodel</code> object created with <code><a href="#topic+init_mutmodel">init_mutmodel</a></code>.</p>
</td></tr>
<tr><td><code id="mutmodel_+3A_locus">locus</code></td>
<td>
<p> the locus of interest (integer, remember postfix L).</p>
</td></tr>
<tr><td><code id="mutmodel_+3A_alleles">alleles</code></td>
<td>
<p> vector of integers (remember postfix L) of the alleles of interest.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Mutation probabilities for locus <code>locus</code> in mutation model <code>mutmodel</code> at alleleles <code>alleles</code>.
</p>


<h3>Author(s)</h3>

<p>Mikkel Meyer Andersen &lt;mikl@math.aau.dk&gt; and Poul Svante Eriksen</p>


<h3>Examples</h3>

<pre><code class='language-R'>mutpars.locus1 &lt;- c(0.149,   2.08,    18.3,   0.149,   0.374,   27.4) # DYS19
mutpars.locus2 &lt;- c(0.500,   1.18,    18.0,   0.500,   0.0183,  349)  # DYS389I
mutpars.locus3 &lt;- c(0.0163,  17.7,    11.1,   0.0163,  0.592,   14.1)  # DYS391
mutpars &lt;- matrix(c(mutpars.locus1, mutpars.locus2, mutpars.locus3), ncol = 3)
colnames(mutpars) &lt;- c("DYS19", "DYS389I", "DYS391")
mutmodel &lt;- init_mutmodel(modeltype = 2L, mutpars = mutpars)

mutmodel_not_mut(mutmodel, locus = 1L, alleles = 10L:20L)
mutmodel_dw_mut(mutmodel, locus = 1L, alleles = 10L:20L)
mutmodel_up_mut(mutmodel, locus = 1L, alleles = 10L:20L)

statdists &lt;- approx_stationary_dist(mutmodel, alleles = 5L:20L)
bp &lt;- barplot(statdists, beside = TRUE)
text(bp, 0.02, round(statdists, 1), cex = 1, pos = 3) 
text(bp, 0, rep(rownames(statdists), ncol(mutmodel$mutpars)), cex = 1, pos = 3)


mutpars &lt;- matrix(c(c(0.003, 0.001), rep(0.004, 2), rep(0.001, 2)), ncol = 3)
colnames(mutpars) &lt;- c("DYS19", "DYS389I", "DYS391")
mutmodel &lt;- init_mutmodel(modeltype = 1L, mutpars = mutpars)
mutmodel
statdists &lt;- approx_stationary_dist(mutmodel, alleles = 5L:20L)
statdists

bp &lt;- barplot(statdists, beside = TRUE)
text(bp, 0.02, round(statdists, 1), cex = 1, pos = 3) 
text(bp, 0, rep(rownames(statdists), ncol(mutmodel$mutpars)), cex = 1, pos = 3)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
