<!DOCTYPE html><html lang="en"><head><title>Help for package g3viz</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {g3viz}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#g3Lollipop'><p>Render g3lollipop diagram for the given mutation data</p></a></li>
<li><a href='#g3Lollipop-shiny'><p>Shiny bindings for g3Lollipop</p></a></li>
<li><a href='#g3Lollipop.options'><p>G3Lollipop plot options</p></a></li>
<li><a href='#g3Lollipop.theme'><p>G3Lollipop chart options of built-in themes.</p></a></li>
<li><a href='#getMutationsFromCbioportal'><p>Query cancer genomic mutation data from cBioPortal</p></a></li>
<li><a href='#guessMAFColumnName'><p>Guess column name for MAF file</p></a></li>
<li><a href='#hgnc2pfam'><p>Map from Hugo symbol to Pfam domains</p></a></li>
<li><a href='#hgnc2pfam.df'><p>Mapping table between gene.symbol, uniprot.id, and pfam</p></a></li>
<li><a href='#hgnc2uniprot'><p>Mapping from Hugo symbol to UniProt IDs</p></a></li>
<li><a href='#mapMutationTypeToMutationClass'><p>Map from mutation type (aka, variant classification) to mutation class</p></a></li>
<li><a href='#mutation.table.df'><p>Default mapping table between mutation type (aka, variant classification) to mutation class</p></a></li>
<li><a href='#parseProteinChange'><p>Extract <em>amino_acid_position</em> from <em>Protein_Change</em></p></a></li>
<li><a href='#readMAF'><p>Read MAF file</p></a></li>
<li><a href='#uniprot2pfam'><p>From UniProt ID to Pfam-A domain composition</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Interactively Visualize Genetic Mutation Data using a
Lollipop-Diagram</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xin Guo &lt;g3viz.group@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Interface for 'g3-lollipop' 'JavaScript' library.
    Visualize genetic mutation data using an interactive lollipop diagram in 'RStudio' or your web browser.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>jsonlite, stringr, httr2, AnnotationDbi, org.Hs.eg.db,
htmlwidgets</td>
</tr>
<tr>
<td>Suggests:</td>
<td>shiny (&ge; 1.0.0), knitr, rmarkdown, kableExtra</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/G3viz/g3viz">https://github.com/G3viz/g3viz</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/G3viz/g3viz/issues">https://github.com/G3viz/g3viz/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-04 05:56:50 UTC; xguo</td>
</tr>
<tr>
<td>Author:</td>
<td>Xin Guo [aut, cre],
  Rener Zhang [ctb],
  Ruining Zhu [ctb],
  Feng Zhao [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-04 09:00:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='g3Lollipop'>Render g3lollipop diagram for the given mutation data</h2><span id='topic+g3Lollipop'></span>

<h3>Description</h3>

<p>Render g3lollipop diagram for the given mutation data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>g3Lollipop(
  mutation.dat,
  gene.symbol,
  uniprot.id = NA,
  gene.symbol.col = "Hugo_Symbol",
  aa.pos.col = "AA_Position",
  protein.change.col = c("Protein_Change", "HGVSp_Short"),
  factor.col = "Mutation_Class",
  plot.options = g3Lollipop.options(),
  save.png.btn = TRUE,
  save.svg.btn = TRUE,
  btn.style = NA,
  output.filename = "output"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="g3Lollipop_+3A_mutation.dat">mutation.dat</code></td>
<td>
<p>Input genomic mutation data frame</p>
</td></tr>
<tr><td><code id="g3Lollipop_+3A_gene.symbol">gene.symbol</code></td>
<td>
<p>HGNC primary gene symbol</p>
</td></tr>
<tr><td><code id="g3Lollipop_+3A_uniprot.id">uniprot.id</code></td>
<td>
<p>UniProt ID, in case that the specified gene symbol links to
multiple UniProt entries (isoforms). For example, <em>AKAP7</em> gene has two
isoforms in <a href="https://www.uniprot.org/">UniProt</a>,
<a href="https://www.uniprot.org/uniprot/O43687">O43687</a> and
<a href="https://www.uniprot.org/uniprot/Q9P0M2">Q9P0M2</a>.</p>
</td></tr>
<tr><td><code id="g3Lollipop_+3A_gene.symbol.col">gene.symbol.col</code></td>
<td>
<p>Column name of Hugo gene symbols (e.g., TP53). Default
<em>Hugo_Symbol</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop_+3A_aa.pos.col">aa.pos.col</code></td>
<td>
<p>Column name of the parsed amino-acid change position.
Default <em>AA_Position</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop_+3A_protein.change.col">protein.change.col</code></td>
<td>
<p>Column name of protein change information (e.g.,
p.K960R, G658S, L14Sfs*15). Default is a list of <em>Protein_Change</em>,
<em>HGVSp_Short</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop_+3A_factor.col">factor.col</code></td>
<td>
<p>column of classes in the plot legend. IF <code>NA</code>, use
parsed <em>Mutation_Class</em> column, otherwise, use specified.  Default
<code>NA</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop_+3A_plot.options">plot.options</code></td>
<td>
<p>g3lollipop diagram options in list format. Check
<code><a href="#topic+g3Lollipop.options">g3Lollipop.options</a></code></p>
</td></tr>
<tr><td><code id="g3Lollipop_+3A_save.png.btn">save.png.btn</code></td>
<td>
<p>If add <em>save-as-png</em> button to the diagram. Default
<code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop_+3A_save.svg.btn">save.svg.btn</code></td>
<td>
<p>If add <em>save-as-svg</em> button to the diagram. Default
<code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop_+3A_btn.style">btn.style</code></td>
<td>
<p>button style, including browser default button style, and
two built-in styles, <em>blue</em> or <em>gray</em>. Default <code>NA</code>,
indicating browser default.</p>
</td></tr>
<tr><td><code id="g3Lollipop_+3A_output.filename">output.filename</code></td>
<td>
<p>Specify output file name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lollipop diagram for the given mutation data. The chart is interactive
within either Shiny applications or Rmd documents under the bindings.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# system mutation data
maf.file &lt;- system.file("extdata", "TCGA.BRCA.varscan.somatic.maf.gz", package = "g3viz")
# read in MAF file
mutation.dat &lt;- readMAF(maf.file)

# use built-in chart theme
chart.options &lt;- g3Lollipop.theme(theme.name = "default",
                                  title.text = "PIK3CA gene (default theme)")
# generate chart
g3Lollipop(mutation.dat,
           gene.symbol = "PIK3CA",
           plot.options = chart.options,
           btn.style = "blue",
           output.filename = "default_theme")

</code></pre>

<hr>
<h2 id='g3Lollipop-shiny'>Shiny bindings for g3Lollipop</h2><span id='topic+g3Lollipop-shiny'></span><span id='topic+g3LollipopOutput'></span><span id='topic+renderG3Lollipop'></span>

<h3>Description</h3>

<p>Output and render functions for using g3viz lollipop diagram within Shiny
applications and interactive Rmd documents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>g3LollipopOutput(outputId, width = "100%", height = "520px")

renderG3Lollipop(expr, env = parent.frame(), quoted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="g3Lollipop-shiny_+3A_outputid">outputId</code></td>
<td>
<p>output variable to read from</p>
</td></tr>
<tr><td><code id="g3Lollipop-shiny_+3A_width">width</code>, <code id="g3Lollipop-shiny_+3A_height">height</code></td>
<td>
<p>Must be a valid CSS unit (like <code>'100%'</code>,
<code>'400px'</code>, <code>'auto'</code>) or a number, which will be coerced to a
string and have <code>'px'</code> appended.</p>
</td></tr>
<tr><td><code id="g3Lollipop-shiny_+3A_expr">expr</code></td>
<td>
<p>An expression that generates a g3-lollipop</p>
</td></tr>
<tr><td><code id="g3Lollipop-shiny_+3A_env">env</code></td>
<td>
<p>The environment in which to evaluate <code>expr</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop-shiny_+3A_quoted">quoted</code></td>
<td>
<p>Is <code>expr</code> a quoted expression (with <code>quote()</code>)? This
is useful if you want to save an expression in a variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value returned. It is the binding which enables interactive functions
within Shiny applications and Rmd documents.
</p>

<hr>
<h2 id='g3Lollipop.options'>G3Lollipop plot options</h2><span id='topic+g3Lollipop.options'></span>

<h3>Description</h3>

<p>G3Lollipop plot options
</p>


<h3>Usage</h3>

<pre><code class='language-R'>g3Lollipop.options(
  chart.width = 800,
  chart.type = "circle",
  chart.margin = list(left = 40, right = 20, top = 15, bottom = 25),
  chart.background = "transparent",
  transition.time = 600,
  y.axis.label = "# of mutations",
  axis.label.font = "normal 12px Arial",
  axis.label.color = "#4f4f4f",
  axis.label.alignment = "middle",
  axis.label.dy = "-2em",
  y.axis.line.color = "#c4c8ca",
  y.axis.line.style = "dash",
  y.axis.line.width = 1,
  y.max.range.ratio = 1.1,
  legend.margin = list(left = 10, right = 0, top = 5, bottom = 5),
  legend.interactive = TRUE,
  legend.title = NA,
  lollipop.track.height = 420,
  lollipop.track.background = "rgb(233,233,233)",
  lollipop.pop.min.size = 2,
  lollipop.pop.max.size = 12,
  lollipop.pop.info.limit = 8,
  lollipop.pop.info.color = "#EEE",
  lollipop.pop.info.dy = "0.35em",
  lollipop.line.color = "rgb(42,42,42)",
  lollipop.line.width = 0.5,
  lollipop.circle.color = "wheat",
  lollipop.circle.width = 0.5,
  lollipop.label.ratio = 1.4,
  lollipop.label.min.font.size = 10,
  lollipop.color.scheme = "accent",
  highlight.text.angle = "90",
  title.text = "",
  title.font = "normal 16px Arial",
  title.color = "#424242",
  title.alignment = "middle",
  title.dy = "0.35em",
  anno.height = 30,
  anno.margin = list(top = 4, bottom = 0),
  anno.background = "transparent",
  anno.bar.fill = "#e5e3e1",
  anno.bar.margin = list(top = 2, bottom = 2),
  domain.color.scheme = "category10",
  domain.margin = list(top = 0, bottom = 0),
  domain.text.font = "normal 11px Arial",
  domain.text.color = "#f2f2f2",
  brush = TRUE,
  brush.selection.background = "#666",
  brush.selection.opacity = 0.2,
  brush.border.color = "#969696",
  brush.handler.color = "#333",
  brush.border.width = 1,
  legend = TRUE,
  tooltip = TRUE,
  zoom = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="g3Lollipop.options_+3A_chart.width">chart.width</code></td>
<td>
<p>chart width. Default 800.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_chart.type">chart.type</code></td>
<td>
<p><em>pie</em> or <em>circle</em>. Default <em>circle</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_chart.margin">chart.margin</code></td>
<td>
<p>specify chart margin in _list_ format.<br />
Default <code>list(left = 40, right = 20, top = 15, bottom = 25)</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_chart.background">chart.background</code></td>
<td>
<p>chart background. Default <em>transparent</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_transition.time">transition.time</code></td>
<td>
<p>animation transition time when clicking lollipop pops to show labels (in millisecond).  Default 600.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_y.axis.label">y.axis.label</code></td>
<td>
<p>Y-axis label text. Default <em>&quot;# of mutations&quot;</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_axis.label.font">axis.label.font</code></td>
<td>
<p>css font style shorthand
(<em>font-style font-variant font-weight font-size/line-height font-family</em>).
Default <em>&quot;normal 12px Arial&quot;</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_axis.label.color">axis.label.color</code></td>
<td>
<p>axis label text color. Default <em>#4f4f4f</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_axis.label.alignment">axis.label.alignment</code></td>
<td>
<p>axis label text alignment (start/end/middle).  Default <em>middle</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_axis.label.dy">axis.label.dy</code></td>
<td>
<p>text adjustment of axis label text.  Default <em>-2em</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_y.axis.line.color">y.axis.line.color</code></td>
<td>
<p>color of y-axis in-chart lines (ticks).  Default <em>#c4c8ca</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_y.axis.line.style">y.axis.line.style</code></td>
<td>
<p>style of y-axis in-chart lines (ticks), &quot;dash&quot; or &quot;line&quot;.  Default <em>dash</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_y.axis.line.width">y.axis.line.width</code></td>
<td>
<p>width of y-axis in-chart lines (ticks).  Default 1.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_y.max.range.ratio">y.max.range.ratio</code></td>
<td>
<p>ratio of y-axis range to data value range.  Default 1.1.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_legend.margin">legend.margin</code></td>
<td>
<p>legend margin in <em>list</em>.  Default <code>list(left = 10, right = 0, top = 5, bottom = 5)</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_legend.interactive">legend.interactive</code></td>
<td>
<p>legend interactive mode.  Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title.  If <code>NA</code>, <em>factor.col</em> in <code><a href="#topic+g3Lollipop">g3Lollipop</a></code> is used.
Default is <code>NA</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.track.height">lollipop.track.height</code></td>
<td>
<p>height of lollipop track. Default 420.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.track.background">lollipop.track.background</code></td>
<td>
<p>background of lollipop track.  Default <em>rgb(244,244,244)</em></p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.pop.min.size">lollipop.pop.min.size</code></td>
<td>
<p>lollipop pop minimal size.  Default 2.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.pop.max.size">lollipop.pop.max.size</code></td>
<td>
<p>lollipop pop maximal size. Default 12.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.pop.info.limit">lollipop.pop.info.limit</code></td>
<td>
<p>threshold of lollipop pop size to show count information in middle of pop.  Default 8.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.pop.info.color">lollipop.pop.info.color</code></td>
<td>
<p>lollipop pop information text color. Default <em>#EEE</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.pop.info.dy">lollipop.pop.info.dy</code></td>
<td>
<p>y-axis direction text adjustment of lollipop pop information. Default <em>-0.35em</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.line.color">lollipop.line.color</code></td>
<td>
<p>lollipop line color. Default <em>rgb(42,42,42)</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.line.width">lollipop.line.width</code></td>
<td>
<p>lollipop line width. Default 0.5.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.circle.color">lollipop.circle.color</code></td>
<td>
<p>lollipop circle border color. Default <em>wheat</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.circle.width">lollipop.circle.width</code></td>
<td>
<p>lollipop circle border width. Default 0.5.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.label.ratio">lollipop.label.ratio</code></td>
<td>
<p>lollipop click-out label font size to circle size ratio.  Default 1.4.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.label.min.font.size">lollipop.label.min.font.size</code></td>
<td>
<p>lollipop click-out label minimal font size.  Default 10.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_lollipop.color.scheme">lollipop.color.scheme</code></td>
<td>
<p>color scheme to fill lollipop pops.  Default <em>accent</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_highlight.text.angle">highlight.text.angle</code></td>
<td>
<p>pop-on-click highlight text angle.  Default 90.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_title.text">title.text</code></td>
<td>
<p>title of chart. Default is empty.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_title.font">title.font</code></td>
<td>
<p>font of chart title. Default <em>normal 16px Arial</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_title.color">title.color</code></td>
<td>
<p>color of chart title. Default <em>#424242</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_title.alignment">title.alignment</code></td>
<td>
<p>text alignment of chart title (start/middle/end). Default <em>middle</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_title.dy">title.dy</code></td>
<td>
<p>text adjustment of chart title.  Default <em>0.35em</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_anno.height">anno.height</code></td>
<td>
<p>height of protein structure annotation track.  Default 30.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_anno.margin">anno.margin</code></td>
<td>
<p>margin of protein structure annotation track.  Default <code>list(top = 4, bottom = 0)</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_anno.background">anno.background</code></td>
<td>
<p>background of protein structure annotation track.  Default <em>transparent</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_anno.bar.fill">anno.bar.fill</code></td>
<td>
<p>background of protein bar in protein structure annotation track.  Default <em>#e5e3e1</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_anno.bar.margin">anno.bar.margin</code></td>
<td>
<p>margin of protein bar in protein structure annotation track.  Default <code>list(top = 2, bottom = 2)</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_domain.color.scheme">domain.color.scheme</code></td>
<td>
<p>color scheme of protein domains.  Default <em>category10</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_domain.margin">domain.margin</code></td>
<td>
<p>margin of protein domains.  Default <code>list(top = 0, bottom = 0)</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_domain.text.font">domain.text.font</code></td>
<td>
<p>domain label text font in shorthand format.  Default <em>normal 11px Arial</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_domain.text.color">domain.text.color</code></td>
<td>
<p>domain label text color.  Default <em>#f2f2f2</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_brush">brush</code></td>
<td>
<p>if show brush.  Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_brush.selection.background">brush.selection.background</code></td>
<td>
<p>background color of selection brush. Default <em>#666</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_brush.selection.opacity">brush.selection.opacity</code></td>
<td>
<p>background opacity of selection brush.  Default 0.2.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_brush.border.color">brush.border.color</code></td>
<td>
<p>border color of selection brush.  Default <em>#969696</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_brush.handler.color">brush.handler.color</code></td>
<td>
<p>color of left and right handlers of selection brush.  Default <em>#333</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_brush.border.width">brush.border.width</code></td>
<td>
<p>border width of selection brush.  Default 1.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_legend">legend</code></td>
<td>
<p>if show legend.  Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_tooltip">tooltip</code></td>
<td>
<p>if show tooltip.  Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.options_+3A_zoom">zoom</code></td>
<td>
<p>if enable zoom feature.  Default <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with g3Lollipop plot options
</p>

<hr>
<h2 id='g3Lollipop.theme'>G3Lollipop chart options of built-in themes.</h2><span id='topic+g3Lollipop.theme'></span>

<h3>Description</h3>

<p>G3Lollipop chart options of built-in themes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>g3Lollipop.theme(
  theme.name = "default",
  title.text = "",
  y.axis.label = "# of mutations",
  legend.title = NA
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="g3Lollipop.theme_+3A_theme.name">theme.name</code></td>
<td>
<p>theme name, including <em>default</em>, <em>cbioportal</em>,
<em>nature</em>, <em>nature2</em>, <em>dark</em>, <em>blue</em>, <em>ggplot2</em>, and <em>simple</em>. Default
<em>default</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.theme_+3A_title.text">title.text</code></td>
<td>
<p>title of chart. Default is empty.</p>
</td></tr>
<tr><td><code id="g3Lollipop.theme_+3A_y.axis.label">y.axis.label</code></td>
<td>
<p>Y-axis label text. Default <em>&quot;# of mutations&quot;</em>.</p>
</td></tr>
<tr><td><code id="g3Lollipop.theme_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title.  If <code>NA</code>, <em>factor.col</em> in
<code><a href="#topic+g3Lollipop">g3Lollipop</a></code> is used. Default is <code>NA</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with g3Lollipop plot options
</p>

<hr>
<h2 id='getMutationsFromCbioportal'>Query cancer genomic mutation data from cBioPortal</h2><span id='topic+getMutationsFromCbioportal'></span>

<h3>Description</h3>

<p>Retrieve and parse mutation data from cBioPortal by the given
cBioPortal cancer study ID and the gene symbol.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMutationsFromCbioportal(
  study.id,
  gene.symbol,
  output.file = NA,
  mutation.type.to.class.df = NA
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getMutationsFromCbioportal_+3A_study.id">study.id</code></td>
<td>
<p>cbioprotal study ID</p>
</td></tr>
<tr><td><code id="getMutationsFromCbioportal_+3A_gene.symbol">gene.symbol</code></td>
<td>
<p>HGNC gene symbol.</p>
</td></tr>
<tr><td><code id="getMutationsFromCbioportal_+3A_output.file">output.file</code></td>
<td>
<p>if specified, output to a file in <em>CSV</em> format.
Default is <code>NA</code>.</p>
</td></tr>
<tr><td><code id="getMutationsFromCbioportal_+3A_mutation.type.to.class.df">mutation.type.to.class.df</code></td>
<td>
<p>mapping table from mutation type to class.
See <code><a href="#topic+mapMutationTypeToMutationClass">mapMutationTypeToMutationClass</a></code> for details. Default
<code>NA</code>, which indicates to use default mappings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with columns
</p>

<dl>
<dt>Hugo_Symbol</dt><dd><p>Hugo gene symbol</p>
</dd>
<dt>Protein_Change</dt><dd><p>Protein change information (cBioportal uses <em>HGVSp</em> format)</p>
</dd>
<dt>Sample_ID</dt><dd><p>Sample ID</p>
</dd>
<dt>Mutation_Type</dt><dd><p>mutation type, aka, variant classification.</p>
</dd>
<dt>Chromosome</dt><dd><p>chromosome</p>
</dd>
<dt>Start_Position</dt><dd><p>start position</p>
</dd>
<dt>End_Position</dt><dd><p>end position</p>
</dd>
<dt>Reference_Allele</dt><dd><p>reference allele</p>
</dd>
<dt>Variant_Allele</dt><dd><p>variant allele</p>
</dd>
<dt>Mutation_Class</dt><dd><p>mutation class (e.g., Truncating/Missense/Inframe/Other)</p>
</dd>
<dt>AA_Position</dt><dd><p>amino-acid position of the variant; if the variant is not in
protein-coding region, <code>NA</code></p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>
# Usage:

</code></pre>

<hr>
<h2 id='guessMAFColumnName'>Guess column name for MAF file</h2><span id='topic+guessMAFColumnName'></span>

<h3>Description</h3>

<p>Guess column name for MAF file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>guessMAFColumnName(maf.df, alt.column.names)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="guessMAFColumnName_+3A_maf.df">maf.df</code></td>
<td>
<p>MAF data frame</p>
</td></tr>
<tr><td><code id="guessMAFColumnName_+3A_alt.column.names">alt.column.names</code></td>
<td>
<p>a vector of alternative column names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if hit one alternative column name, return the name; otherwise, return <code>NA</code>
</p>

<hr>
<h2 id='hgnc2pfam'>Map from Hugo symbol to Pfam domains</h2><span id='topic+hgnc2pfam'></span>

<h3>Description</h3>

<p>Mapping from Hugo symbol to Pfam-A domain composition.
If the given Hugo symbol has multiple UniProt ID mappings,
and <code>guess == TRUE</code>,
the longest UniProt protein is selected. Return is either a list of a JSON.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hgnc2pfam(hgnc.symbol, guess = TRUE, uniprot.id = NA, output.format = "json")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hgnc2pfam_+3A_hgnc.symbol">hgnc.symbol</code></td>
<td>
<p>primary Hugo symbol</p>
</td></tr>
<tr><td><code id="hgnc2pfam_+3A_guess">guess</code></td>
<td>
<p>if the given Hugo symbol links to multiple UniProt IDs,
choose the longest one (<code>guess == TRUE</code>);
otherwise <code>NA</code> (<code>guess == FALSE</code>).  Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="hgnc2pfam_+3A_uniprot.id">uniprot.id</code></td>
<td>
<p>UniProt ID, in case that gene symbol maps to multiple UniProt entries.</p>
</td></tr>
<tr><td><code id="hgnc2pfam_+3A_output.format">output.format</code></td>
<td>
<p>output format: JSON or list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list or a JSON with attributes:
<em>symbol</em>, <em>uniprot</em>, <em>length</em>, and a list of <em>Pfam</em> entries, including
<em>hmm.acc</em>, <em>hmm.name</em>, <em>start</em>, <em>end</em>, and <em>type</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># general usage
hgnc2pfam("TP53")
hgnc2pfam("TP53", output.format = "json")
hgnc2pfam("TP53", output.format = "list")
hgnc2pfam("TP53", output.format = "json", uniprot.id = "P04637") # OK

# for gene mapping to multiple UniProt enties
hgnc2pfam("GNAS", guess = TRUE)
hgnc2pfam("GNAS", guess = FALSE)
hgnc2pfam("GNAS", output.format = "list")
hgnc2pfam("GNAS", output.format = "list", uniprot.id = "P84996")
## Not run: 
hgnc2pfam("GNAS", output.format = "list", uniprot.id = "P84997") # , returns FALSE

## End(Not run)

hgnc2pfam("PRAMEF9", output.format = "list") # no Pfam mappings

</code></pre>

<hr>
<h2 id='hgnc2pfam.df'>Mapping table between gene.symbol, uniprot.id, and pfam</h2><span id='topic+hgnc2pfam.df'></span>

<h3>Description</h3>

<p>A dataset containing the mapping table between Hugo symbol, UniProt ID, and
Pfam ACC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hgnc2pfam.df
</code></pre>


<h3>Format</h3>

<p>A data frame with columns:
</p>

<dl>
<dt>symbol</dt><dd><p>Gene symbol</p>
</dd>
<dt>uniprot</dt><dd><p>UniProt ID</p>
</dd>
<dt>length</dt><dd><p>protein length</p>
</dd>
<dt>start</dt><dd><p>starting position of Pfam domain</p>
</dd>
<dt>end</dt><dd><p>ending position of Pfam domain</p>
</dd>
<dt>hmm.acc</dt><dd><p>Pfam accession number</p>
</dd>
<dt>hmm.name</dt><dd><p>Pfam name</p>
</dd>
<dt>type</dt><dd><p>Pfam type, i.e., domain/family/motif/repeat/disordered/coiled-coil</p>
</dd>
</dl>



<h3>Source</h3>

<p>Pfam (v31.0) and UniProt
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hgnc2pfam.df
</code></pre>

<hr>
<h2 id='hgnc2uniprot'>Mapping from Hugo symbol to UniProt IDs</h2><span id='topic+hgnc2uniprot'></span>

<h3>Description</h3>

<p>Mapping from Hugo Symbol to UniProt ID using internal mapping table.
Return a data frame with columns <em>symbol</em> (Hugo symbol), <em>uniprot</em> (UniProt ID),
and <em>length</em> (protein length).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hgnc2uniprot(hgnc.symbol)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hgnc2uniprot_+3A_hgnc.symbol">hgnc.symbol</code></td>
<td>
<p>primary HUGO symbol</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with columns <em>symbol</em> (Hugo symbol), <em>uniprot</em> (UniProt ID),
and <em>length</em> (protein length).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># maps to single UniProt entry
hgnc2uniprot("TP53")

# maps to multiple UniProt entries
hgnc2uniprot("GNAS")
hgnc2uniprot("AKAP7")

</code></pre>

<hr>
<h2 id='mapMutationTypeToMutationClass'>Map from mutation type (aka, variant classification) to mutation class</h2><span id='topic+mapMutationTypeToMutationClass'></span>

<h3>Description</h3>

<p>Map from mutation type (aka, variant classification) to mutation class.
Default mappings are as follows,
</p>

<ul>
<li><p> Missense
</p>

<ul>
<li> <p><em>Missense_Mutation</em> &mdash; a point mutation in which a single nucleotide change results in a codon that codes for a different amino acid
See <a href="https://en.wikipedia.org/wiki/Missense_mutation">https://en.wikipedia.org/wiki/Missense_mutation</a>.
</p>
</li></ul>

</li>
<li><p> Inframe
</p>

<ul>
<li> <p><em>In_Frame_Del</em> &mdash; a deletion that keeps the sequence in frame
</p>
</li>
<li> <p><em>In_Frame_Ins</em> &mdash; an insertion that keeps the sequence in frame
</p>
</li>
<li> <p><em>Silent</em> &mdash; variant is in coding region of the chosen transcript, but protein structure is identical (i.e., a synonymous mutation)
</p>
</li>
<li> <p><em>Targeted_Region</em> &mdash; targeted region
</p>
</li></ul>

</li>
<li><p> Truncating
</p>

<ul>
<li> <p><em>Frame_Shift</em> &mdash; a variant caused by indels of a number of nucleotides in a DNA sequence that is not divisible by three.
See <a href="https://en.wikipedia.org/wiki/Frameshift_mutation">https://en.wikipedia.org/wiki/Frameshift_mutation</a>.
</p>
</li>
<li> <p><em>Frame_Shift_Ins</em> &mdash; a variant caused by insertion that moves the coding sequence out of frame.
See <a href="https://en.wikipedia.org/wiki/Frameshift_mutation">https://en.wikipedia.org/wiki/Frameshift_mutation</a>.
</p>
</li>
<li> <p><em>Frame_Shift_Del</em> &mdash; a variant caused by deletion that moves the coding sequence out of frame.
See <a href="https://en.wikipedia.org/wiki/Frameshift_mutation">https://en.wikipedia.org/wiki/Frameshift_mutation</a>.
</p>
</li>
<li> <p><em>Nonsense_Mutation</em> &mdash; a premature stop codon that is created by the variant.
See <a href="https://en.wikipedia.org/wiki/Nonsense_mutation">https://en.wikipedia.org/wiki/Nonsense_mutation</a>.
</p>
</li>
<li> <p><em>Nonstop_Mutation</em> &mdash; a variant that removes stop codon.
</p>
</li>
<li> <p><em>Splice_Site</em> &mdash; a variant that is within two bases of a splice site.
</p>
</li>
<li> <p><em>Splice_Region</em> &mdash; a variant that is within splice region.
</p>
</li></ul>

</li>
<li><p> Other
</p>

<ul>
<li> <p><em>5'UTR</em> &mdash; a variant that is on the 5'UTR for the chosen transcript.
</p>
</li>
<li> <p><em>3'UTR</em> &mdash; a variant that is on the 3'UTR for the chosen transcript.
</p>
</li>
<li> <p><em>5'Flank</em> &mdash; a variant that is upstream of the chosen transcript (generally within 3kb).
</p>
</li>
<li> <p><em>3'Flank</em> &mdash; a variant that is downstream of the chosen transcript (generally within 3kb).
</p>
</li>
<li> <p><em>Fusion</em> &mdash; a gene fusion.
</p>
</li>
<li> <p><em>IGR</em> &mdash; an intergenic region. Does not overlap any transcript.
</p>
</li>
<li> <p><em>Intron</em> &mdash; a variant that lies between exons within the bounds of the chosen transcript.
</p>
</li>
<li> <p><em>Translation_Start_Site</em> &mdash; a variant that is in translation start site.
</p>
</li>
<li> <p><em>De_novo_Start_InFrame</em> &mdash; a novel start codon that is created by the given variant using the chosen transcript.
However, it is in frame relative to the coded protein.
</p>
</li>
<li> <p><em>De_novo_Start_OutOfFrame</em> &mdash; a novel start codon that is created by the given variant using the chosen transcript.
However, it is out of frame relative to the coded protein.
</p>
</li>
<li> <p><em>Start_Codon_SNP</em> &mdash; a point mutation that overlaps the start codon.
</p>
</li>
<li> <p><em>Start_Codon_Ins</em> &mdash; an insertion that overlaps the start codon.
</p>
</li>
<li> <p><em>Start_Codon_Del</em> &mdash; a deletion that overlaps the start codon.
</p>
</li>
<li> <p><em>RNA</em> &mdash; a variant that lies on one of the RNA transcripts.
</p>
</li>
<li> <p><em>lincRNA</em> &mdash; a variant that lies on one of the lincRNAs.
</p>
</li>
<li> <p><em>Unknown</em> &mdash; Unknown
</p>
</li></ul>

</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>mapMutationTypeToMutationClass(
  mutation.type.vec,
  mutation.type.to.class.df = NA
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mapMutationTypeToMutationClass_+3A_mutation.type.vec">mutation.type.vec</code></td>
<td>
<p>a vector of mutation type information</p>
</td></tr>
<tr><td><code id="mapMutationTypeToMutationClass_+3A_mutation.type.to.class.df">mutation.type.to.class.df</code></td>
<td>
<p>A mapping table from mutation type (header <em>Mutation_Type</em>)
to mutation class (header <em>Mutation_Class</em>).
Default <code>NA</code>, which indicates to use default mappings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of mapped mutation class information
</p>

<hr>
<h2 id='mutation.table.df'>Default mapping table between mutation type (aka, variant classification) to mutation class</h2><span id='topic+mutation.table.df'></span>

<h3>Description</h3>

<p>A dataset containing the mapping table between genomic mutation type (aka, variant classification) to mutation class.
See <code><a href="#topic+mapMutationTypeToMutationClass">mapMutationTypeToMutationClass</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation.table.df
</code></pre>


<h3>Format</h3>

<p>A data frame with three columns:
</p>

<dl>
<dt>Mutation_Type</dt><dd><p>Mutation type, aka, variant classification</p>
</dd>
<dt>Mutation_Class</dt><dd><p>mutation class</p>
</dd>
<dt>Short_Name</dt><dd><p>short name of mutation type</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>mutation.table.df
</code></pre>

<hr>
<h2 id='parseProteinChange'>Extract <em>amino_acid_position</em> from <em>Protein_Change</em></h2><span id='topic+parseProteinChange'></span>

<h3>Description</h3>

<p>Parse <em>amino_acid_position</em> according to HGVSp_short format. <br />
For example, <em>p.Q16Rfs*28</em>, amino-acid position is 16.
See <a href="http://varnomen.hgvs.org/recommendations/protein/">http://varnomen.hgvs.org/recommendations/protein/</a> or <a href="https://www.hgvs.org/mutnomen/recs-prot.html">https://www.hgvs.org/mutnomen/recs-prot.html</a>. <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseProteinChange(protein.change.vec, mutation.class.vec)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parseProteinChange_+3A_protein.change.vec">protein.change.vec</code></td>
<td>
<p>a vector of strings with protein change information, usually in HGVSp_short format.</p>
</td></tr>
<tr><td><code id="parseProteinChange_+3A_mutation.class.vec">mutation.class.vec</code></td>
<td>
<p>a vector of strings with mutation class (or so-called variant classification) information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of parsed amino-acid position
</p>

<hr>
<h2 id='readMAF'>Read MAF file</h2><span id='topic+readMAF'></span>

<h3>Description</h3>

<p>Read mutation information from MAF file. For MAF format specification, see
<a href="https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/">https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readMAF(
  maf.file,
  gene.symbol.col = "Hugo_Symbol",
  variant.class.col = c("Variant_Classification", "Mutation_Type"),
  protein.change.col = c("Protein_Change", "HGVSp_Short"),
  if.parse.aa.pos = TRUE,
  if.parse.mutation.class = TRUE,
  mutation.class.col = "Mutation_Class",
  aa.pos.col = "AA_Position",
  mutation.type.to.class.df = NA,
  sep = "\t",
  quote = "",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readMAF_+3A_maf.file">maf.file</code></td>
<td>
<p>MAF file name.  Gnuzipped input file allowed, with &quot;.gz&quot; file
extension.</p>
</td></tr>
<tr><td><code id="readMAF_+3A_gene.symbol.col">gene.symbol.col</code></td>
<td>
<p>Column name of Hugo gene symbols (e.g., TP53). Default
<em>Hugo_Symbol</em>.</p>
</td></tr>
<tr><td><code id="readMAF_+3A_variant.class.col">variant.class.col</code></td>
<td>
<p>Column name for variant class information (e.g.,
<em>Missense_Mutation</em>, <em>Nonsense_Mutation</em>). Default is the first
match of <em>Variant_Classification</em> or <em>Mutation_Type</em>.</p>
</td></tr>
<tr><td><code id="readMAF_+3A_protein.change.col">protein.change.col</code></td>
<td>
<p>Column name for protein change information (e.g.,
p.K960R, G658S, L14Sfs*15). Default is the first match of
<em>Protein_Change</em> or <em>HGVSp_Short</em>.</p>
</td></tr>
<tr><td><code id="readMAF_+3A_if.parse.aa.pos">if.parse.aa.pos</code></td>
<td>
<p>if parse amino-acid position of mutations. Default is
<code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="readMAF_+3A_if.parse.mutation.class">if.parse.mutation.class</code></td>
<td>
<p>if parse mutation class from mutation type
(variant classification) information. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="readMAF_+3A_mutation.class.col">mutation.class.col</code></td>
<td>
<p>Column name of the parsed mutation class. Default
<em>Mutation_Class</em>.</p>
</td></tr>
<tr><td><code id="readMAF_+3A_aa.pos.col">aa.pos.col</code></td>
<td>
<p>Column name of the parsed amino-acid change position.
Default <em>AA_Position</em>.</p>
</td></tr>
<tr><td><code id="readMAF_+3A_mutation.type.to.class.df">mutation.type.to.class.df</code></td>
<td>
<p>mapping table from mutation type to class.
<code><a href="#topic+mapMutationTypeToMutationClass">mapMutationTypeToMutationClass</a></code> for details. Default <code>NA</code>,
which indicates to use default mappoings.</p>
</td></tr>
<tr><td><code id="readMAF_+3A_sep">sep</code></td>
<td>
<p>separator of columns. Default <code>sep = "\t"</code>.</p>
</td></tr>
<tr><td><code id="readMAF_+3A_quote">quote</code></td>
<td>
<p>the set of quoting characters. To disable quoting altogether,
use <code>quote = ""</code>.  Default <code>quote = ""</code>.</p>
</td></tr>
<tr><td><code id="readMAF_+3A_...">...</code></td>
<td>
<p>additional parameters pass to <code><a href="utils.html#topic+read.table">read.table</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing MAF information, plus optional columns of the
parsed <em>Mutation_Class</em> and <em>Protein_Position</em>.
</p>

<hr>
<h2 id='uniprot2pfam'>From UniProt ID to Pfam-A domain composition</h2><span id='topic+uniprot2pfam'></span>

<h3>Description</h3>

<p>Map from UniProt ID to Pfam-A domain composition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uniprot2pfam(uniprot.id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uniprot2pfam_+3A_uniprot.id">uniprot.id</code></td>
<td>
<p>UniProt ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with columns
</p>

<ul>
<li> <p><em>uniprot</em> &mdash; UniProt ID
</p>
</li>
<li> <p><em>length</em> &mdash; protein length
</p>
</li>
<li> <p><em>hmm.acc</em> &mdash; accession number of Pfam HMM model, e.g., PF08563
</p>
</li>
<li> <p><em>hmm.name</em> &mdash; Pfam name, e.g., P53_TAD
</p>
</li>
<li> <p><em>start</em> &mdash; Pfam domain start position
</p>
</li>
<li> <p><em>end</em> &mdash; Pfam domain end position
</p>
</li>
<li> <p><em>type</em> &mdash; Pfam type, including domain/motif/family
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>uniprot2pfam("Q5VWM5")  # PRAMEF9; PRAMEF15
uniprot2pfam("P04637")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
