<!DOCTYPE html><html lang="en"><head><title>Help for package GAGBLUP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GAGBLUP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bin'><p>Binning Genotypes for Dimensional Reduction</p></a></li>
<li><a href='#bins'><p>binned genotypes</p></a></li>
<li><a href='#gagblup'><p>Genetic algorithm assisted genomic best liner unbiased prediction for genomic selection</p></a></li>
<li><a href='#genotype'><p>Genotype in Numeric Format</p></a></li>
<li><a href='#phenotype'><p>Phenotypic data of 410 hybrids</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genetic Algorithm Assisted Genomic Best Liner Unbiased
Prediction</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs genetic algorithm (Scrucca, L (2013) &lt;<a href="https://doi.org/10.18637%2Fjss.v053.i04">doi:10.18637/jss.v053.i04</a>&gt;) 
             assisted genomic best liner unbiased prediction for genomic selection.
             It also provides a binning method in natural population for genomic selection
             under the principle of linkage disequilibrium for dimensional reduction.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>GA, foreach, doParallel</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-05 02:39:04 UTC; Dell</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-05 16:50:08 UTC</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuxiang Zhang &lt;yuxiangzhang_99@foxmail.com&gt;</td>
</tr>
<tr>
<td>Author:</td>
<td>Yuxiang Zhang [aut, cre],
  Yang Xu [aut],
  Yanru Cui [ctb],
  Chenwu Xu [ctb],
  Shizhong Xu [ctb]</td>
</tr>
</table>
<hr>
<h2 id='bin'>Binning Genotypes for Dimensional Reduction</h2><span id='topic+bin'></span>

<h3>Description</h3>

<p>Binning the original genotypes into bins for dimensional reduction 
under the principle of linkage disequilibrium.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bin(genotype = genotype, binvar = 0.15)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bin_+3A_genotype">genotype</code></td>
<td>
<p>a matrix for genotypes in numeric format, coded as 1, 0 and -1, with individuals in rows and markers in cols.</p>
</td></tr>
<tr><td><code id="bin_+3A_binvar">binvar</code></td>
<td>
<p>a hyper-parameter between 0 and 1, the closer to 0, the fewer bins yields.
Users can choose binvar based on the required number of bins, default is 0.15.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with following information is returned:
$bins_genotypes  binned genotypes
$bins_range  start and stop of each bin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data from GAGBLUP package
data(genotype)
## binning genotypes
bins &lt;- bin(genotype,0.2)
 
</code></pre>

<hr>
<h2 id='bins'>binned genotypes</h2><span id='topic+bins'></span>

<h3>Description</h3>

<p>Binned genotypes of 410 maize hybrids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bins
</code></pre>


<h3>Format</h3>

<p>A data frame with 410 rows and 810 variables.
</p>

<hr>
<h2 id='gagblup'>Genetic algorithm assisted genomic best liner unbiased prediction for genomic selection</h2><span id='topic+gagblup'></span>

<h3>Description</h3>

<p>Performs genomic selection with genetic algorithm assisted genomic best liner unbiased prediction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gagblup(
  genotype,
  phenotype,
  fit_fun = "HAT",
  maxiter = 2000,
  nfold = 10,
  nTimes = 1,
  seed = 123,
  n_core = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gagblup_+3A_genotype">genotype</code></td>
<td>
<p>a matrix for genotypes in numeric format, with individuals in rows and markers in cols.</p>
</td></tr>
<tr><td><code id="gagblup_+3A_phenotype">phenotype</code></td>
<td>
<p>a vector of phenotype, missing (NA) values are not allowed.</p>
</td></tr>
<tr><td><code id="gagblup_+3A_fit_fun">fit_fun</code></td>
<td>
<p>the fitness function. There are four options: fitness = &quot;AIC&quot;/&quot;BIC&quot;/&quot;FIT&quot;/&quot;HAT&quot;, default is &quot;HAT&quot;</p>
</td></tr>
<tr><td><code id="gagblup_+3A_maxiter">maxiter</code></td>
<td>
<p>max number of iterations for GAGBLUP, default is 2000</p>
</td></tr>
<tr><td><code id="gagblup_+3A_nfold">nfold</code></td>
<td>
<p>the number of folds. Default is 10.</p>
</td></tr>
<tr><td><code id="gagblup_+3A_ntimes">nTimes</code></td>
<td>
<p>the number of independent replicates for the cross-validation. Default is 1.</p>
</td></tr>
<tr><td><code id="gagblup_+3A_seed">seed</code></td>
<td>
<p>the random number. Default is 123.</p>
</td></tr>
<tr><td><code id="gagblup_+3A_n_core">n_core</code></td>
<td>
<p>the number of CPU to be used, default is 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with following information is returned:
$R2  the squared pearson correlation coefficient between the true value and the predicted value,
$predicted_value  the predicted value and the true value of the phenotype,
$marker_selection  a vector of the selected markers, with 1 indicates selected, 0 indicates not selected.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load example data from GAGBLUP package
data(phenotype)
data(bins)
phenotype &lt;- phenotype[1:200,3]
result &lt;- gagblup(bins[1:200,],phenotype,fit_fun='HAT',maxiter=1,nfold=2,nTimes=1,seed=123,n_core=1)
</code></pre>

<hr>
<h2 id='genotype'>Genotype in Numeric Format</h2><span id='topic+genotype'></span>

<h3>Description</h3>

<p>Genotypic data of 410 maize hybrids with 4906 SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotype
</code></pre>


<h3>Format</h3>

<p>A data frame with 410 rows and 4906 variables.
</p>

<hr>
<h2 id='phenotype'>Phenotypic data of 410 hybrids</h2><span id='topic+phenotype'></span>

<h3>Description</h3>

<p>This dataset contains phenotypic data of 410 hybrids for grain yield in maize.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phenotype
</code></pre>


<h3>Format</h3>

<p>A data frame with 410 rows and 3 variables:
</p>

<dl>
<dt><code>M</code></dt><dd><p>The names of male parents.</p>
</dd>
<dt><code>F</code></dt><dd><p>The names of female parents.</p>
</dd>
<dt><code>GY</code></dt><dd><p>The grain yield of hybrids.</p>
</dd>
</dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
