<!DOCTYPE html><html><head><title>Help for package GaMaBioMD</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GaMaBioMD}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alignment_info'><p>Sequence Alignment and Analysis</p></a></li>
<li><a href='#bubble_plot_count'><p>Bubble Plot Count Function</p></a></li>
<li><a href='#bubble_plot_percentage'><p>Bubble Plot Percentage Function</p></a></li>
<li><a href='#clustering_average_similarity'><p>Perform Hierarchical Clustering for average similarity matrix</p></a></li>
<li><a href='#clustering_percent_similarity'><p>Perform Hierarchical Clustering for Percent Similarity</p></a></li>
<li><a href='#compute_average_similarity_matrix'><p>Compute Average Similarity Matrix</p></a></li>
<li><a href='#data_sampling'><p>Samples data from each SampleID group if specified, otherwise uses the final data.</p></a></li>
<li><a href='#expand_accession_ranges'><p>This function expands accession number ranges, creating a data frame with sample and accession numbers.</p></a></li>
<li><a href='#generate_heatmaps'><p>Generate Heatmaps</p></a></li>
<li><a href='#get_sequence_information'><p>Retrieves sequence information for a given list of accessions.</p></a></li>
<li><a href='#preprocess_for_alignment'><p>Preprocesses data for sequence alignment.</p></a></li>
<li><a href='#SampleID_vs_NumSequences'><p>Plots the number of sequences and the probability of sequences per SampleID.</p></a></li>
<li><a href='#tree_average_similarity'><p>Generate Phylogenetic Tree and Color Palette for Average Similarity</p></a></li>
<li><a href='#tree_percent_similarity'><p>Generate Phylogenetic Tree and Color Palette for Percent Similarity Matrix</p></a></li>
<li><a href='#write_fasta'><p>Writes a data frame to a FASTA file.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Diversity Analysis for Sequence Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Sandip Garai [aut, cre],
  Shrikant Mantri [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sandip Garai &lt;sandipnicksandy@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The full form is Garai and Mantri Biological Material Diversity. It is an R package designed for the calculation of biological diversity using sequence data. It simplifies the process by requiring only sample IDs and accession numbers. Whether you're analyzing genetic or microbial diversity, It provides efficient tools for diversity analysis. Serially one should go for the functions as presented here expand_accession_ranges(), get_sequence_information(), preprocess_for_alignment(), write_fasta(), SampleID_vs_NumSequences(), data_sampling(), alignment_info(), compute_average_similarity_matrix(), generate_heatmaps(), clustering_average_similarity(), clustering_percent_similarity(), bubble_plot_count(), bubble_plot_percentage(), tree_average_similarity(), tree_percent_similarity(). Till date there are total 15 functions. More details can be found in Faith (1992) &lt;<a href="https://doi.org/10.1016%2F0006-3207%2892%2991201-3">doi:10.1016/0006-3207(92)91201-3</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>traits, dplyr, magrittr, ggplot2, Biostrings, heatmaply,
reshape2, dendextend, stats, ape</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-12 13:38:56 UTC; user</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-12 14:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='alignment_info'>Sequence Alignment and Analysis</h2><span id='topic+alignment_info'></span>

<h3>Description</h3>

<p>Sequence Alignment and Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alignment_info(data, type = "global", verbose = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alignment_info_+3A_data">data</code></td>
<td>
<p>A data frame with 'SequenceID' and 'Sequence' columns.</p>
</td></tr>
<tr><td><code id="alignment_info_+3A_type">type</code></td>
<td>
<p>A character string specifying the type of sequence alignment ('global', 'local', etc.).</p>
</td></tr>
<tr><td><code id="alignment_info_+3A_verbose">verbose</code></td>
<td>
<p>An integer indicating the level of verbosity (0, 1, or 2).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing matrices and results from sequence alignment.
</p>


<h3>References</h3>

<p>Faith, D. P. (1992). Conservation evaluation and phylogenetic diversity. Biological conservation, 61(1), 1-10.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

# If you want to sample 10% from each SampleID group:
sampled_data &lt;- data_sampling(final_data, sample_proportion = 0.1)

alignment_results &lt;- alignment_info(final_data, type = "global", verbose = 1)

# Access the resulting data frames
score_matrix &lt;- alignment_results$score_matrix
normalized_score_matrix &lt;- alignment_results$normalized_score_matrix

total_aligned_positions_matrix &lt;- alignment_results$total_aligned_positions_matrix
number_of_matching_positions_matrix &lt;- alignment_results$number_of_matching_positions_matrix

percent_similarity_matrix &lt;- alignment_results$percent_similarity_matrix

alignment_results_list &lt;- alignment_results$alignment_results_list

alignment_info_matrix &lt;- alignment_results$alignment_info_matrix

output_directory &lt;- tempdir()

# Save the list of alignment results to an RDS file
saveRDS(alignment_results_list, file.path(output_directory, "alignment_results_list.rds"))

# Save matrices to files
write.table(score_matrix, file.path(output_directory, "score_matrix.txt"), sep = "\t")

</code></pre>

<hr>
<h2 id='bubble_plot_count'>Bubble Plot Count Function</h2><span id='topic+bubble_plot_count'></span>

<h3>Description</h3>

<p>This function generates a bubble plot using ggplot2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bubble_plot_count(
  clustered_data,
  title,
  x_label,
  y_label,
  size_label,
  color_label
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bubble_plot_count_+3A_clustered_data">clustered_data</code></td>
<td>
<p>A data frame containing clustered data.</p>
</td></tr>
<tr><td><code id="bubble_plot_count_+3A_title">title</code></td>
<td>
<p>The title of the plot.</p>
</td></tr>
<tr><td><code id="bubble_plot_count_+3A_x_label">x_label</code></td>
<td>
<p>The label for the x-axis.</p>
</td></tr>
<tr><td><code id="bubble_plot_count_+3A_y_label">y_label</code></td>
<td>
<p>The label for the y-axis.</p>
</td></tr>
<tr><td><code id="bubble_plot_count_+3A_size_label">size_label</code></td>
<td>
<p>The label for the size variable.</p>
</td></tr>
<tr><td><code id="bubble_plot_count_+3A_color_label">color_label</code></td>
<td>
<p>The label for the color variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object representing the bubble plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

# If you want to sample 10% from each SampleID group:
sampled_data &lt;- data_sampling(final_data, sample_proportion = 0.1)

alignment_results &lt;- alignment_info(final_data, type = "global", verbose = 1)

# Access the resulting data frames
score_matrix &lt;- alignment_results$score_matrix
normalized_score_matrix &lt;- alignment_results$normalized_score_matrix

total_aligned_positions_matrix &lt;- alignment_results$total_aligned_positions_matrix
number_of_matching_positions_matrix &lt;- alignment_results$number_of_matching_positions_matrix

percent_similarity_matrix &lt;- alignment_results$percent_similarity_matrix

alignment_results_list &lt;- alignment_results$alignment_results_list

alignment_info_matrix &lt;- alignment_results$alignment_info_matrix

output_directory &lt;- tempdir()

# Save the list of alignment results to an RDS file
saveRDS(alignment_results_list, file.path(output_directory, "alignment_results_list.rds"))

# Save matrices to files
write.table(score_matrix, file.path(output_directory, "score_matrix.txt"), sep = "\t")
average_percent_similarity &lt;- compute_average_similarity_matrix(percent_similarity_matrix)
print(average_percent_similarity)

output_directory &lt;- tempdir()
width_inch &lt;- 8
height_inch &lt;- 6
dpi &lt;- 300

clustering_result &lt;- clustering_average_similarity(average_percent_similarity)

# Extract the dendrogram and clustered data
dend_colored &lt;- clustering_result$dendrogram
clustered_data &lt;- clustering_result$clustered_data
Cluster_SampleID_Percentage &lt;- clustering_result$Cluster_SampleID_Percentage
Cluster_TotalPercentage &lt;- clustering_result$Cluster_TotalPercentage

tiff_file &lt;- file.path(output_directory, "6. hierarchical_clustering_dendrogram_colored.tiff")

# Save the dendrogram as a TIFF image
tiff(tiff_file, width = width_inch, height = height_inch, units = "in", res = dpi)
plot(dend_colored, main = "Colored Hierarchical Clustering Dendrogram")
dev.off()

# Save the clustered data frame to a CSV file
write.csv(clustered_data, file.path(output_directory, "7. clustered_data.csv"), row.names = FALSE)

# Example usage with clustered_data
clustered_data &lt;- clustered_data # Load or generate your clustered data
bubble_plot_count &lt;- bubble_plot_count(clustered_data = clustered_data,
                                      title = "Bubble Plot of Clusters",
                                      x_label = "Clusters",
                                      y_label = "Sample ID",
                                      size_label = "Count",
                                      color_label = "Sample ID")

# Save the bubble plot as a TIFF image
output_directory &lt;- tempdir()
width_inch &lt;- 8
height_inch &lt;- 6
dpi &lt;- 300

# Set the file name for the TIFF image
tiff_file &lt;- file.path(output_directory, "bubble_plot_count.tiff")

# Open the TIFF device
tiff(tiff_file, width = width_inch, height = height_inch, units = "in", res = dpi)

# Print and save the bubble plot
print(bubble_plot_count)

# Close the TIFF device
dev.off()

</code></pre>

<hr>
<h2 id='bubble_plot_percentage'>Bubble Plot Percentage Function</h2><span id='topic+bubble_plot_percentage'></span>

<h3>Description</h3>

<p>This function generates a bubble plot using ggplot2 based on percentage data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bubble_plot_percentage(
  Cluster_SampleID_Percentage,
  title,
  x_label,
  y_label,
  size_label,
  color_label
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bubble_plot_percentage_+3A_cluster_sampleid_percentage">Cluster_SampleID_Percentage</code></td>
<td>
<p>A data frame containing cluster, sample ID, and percentage data.</p>
</td></tr>
<tr><td><code id="bubble_plot_percentage_+3A_title">title</code></td>
<td>
<p>The title of the plot.</p>
</td></tr>
<tr><td><code id="bubble_plot_percentage_+3A_x_label">x_label</code></td>
<td>
<p>The label for the x-axis.</p>
</td></tr>
<tr><td><code id="bubble_plot_percentage_+3A_y_label">y_label</code></td>
<td>
<p>The label for the y-axis.</p>
</td></tr>
<tr><td><code id="bubble_plot_percentage_+3A_size_label">size_label</code></td>
<td>
<p>The label for the size variable.</p>
</td></tr>
<tr><td><code id="bubble_plot_percentage_+3A_color_label">color_label</code></td>
<td>
<p>The label for the color variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object representing the bubble plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

# If you want to sample 10% from each SampleID group:
sampled_data &lt;- data_sampling(final_data, sample_proportion = 0.1)

alignment_results &lt;- alignment_info(final_data, type = "global", verbose = 1)

# Access the resulting data frames
score_matrix &lt;- alignment_results$score_matrix
normalized_score_matrix &lt;- alignment_results$normalized_score_matrix

total_aligned_positions_matrix &lt;- alignment_results$total_aligned_positions_matrix
number_of_matching_positions_matrix &lt;- alignment_results$number_of_matching_positions_matrix

percent_similarity_matrix &lt;- alignment_results$percent_similarity_matrix

alignment_results_list &lt;- alignment_results$alignment_results_list

alignment_info_matrix &lt;- alignment_results$alignment_info_matrix

output_directory &lt;- tempdir()

# Save the list of alignment results to an RDS file
saveRDS(alignment_results_list, file.path(output_directory, "alignment_results_list.rds"))

# Save matrices to files
write.table(score_matrix, file.path(output_directory, "score_matrix.txt"), sep = "\t")
average_percent_similarity &lt;- compute_average_similarity_matrix(percent_similarity_matrix)
print(average_percent_similarity)

output_directory &lt;- tempdir()
width_inch &lt;- 8
height_inch &lt;- 6
dpi &lt;- 300

clustering_result &lt;- clustering_percent_similarity(percent_similarity_matrix)

# Extract the dendrogram and clustered data
dend_colored &lt;- clustering_result$dendrogram
clustered_data &lt;- clustering_result$clustered_data
Cluster_SampleID_Percentage &lt;- clustering_result$Cluster_SampleID_Percentage
Cluster_TotalPercentage &lt;- clustering_result$Cluster_TotalPercentage

tiff_file &lt;- file.path(output_directory, "6. hierarchical_clustering_dendrogram_colored.tiff")

# Save the dendrogram as a TIFF image
tiff(tiff_file, width = width_inch, height = height_inch, units = "in", res = dpi)
plot(dend_colored, main = "Colored Hierarchical Clustering Dendrogram")
dev.off()

# Save the clustered data frame to a CSV file
write.csv(clustered_data, file.path(output_directory, "7. clustered_data.csv"), row.names = FALSE)

# Example usage with Cluster_SampleID_Percentage
Cluster_SampleID_Percentage &lt;- Cluster_SampleID_Percentage
bubble_plot_percentage &lt;- bubble_plot_percentage(Cluster_SampleID_Percentage,
                                                title = "Bubble Plot",
                                                x_label = "Clusters",
                                                y_label = "Sample ID",
                                                size_label = "Percentage",
                                                color_label = "Sample ID")

# Save the bubble plot as a TIFF image
output_directory &lt;- tempdir()
width_inch &lt;- 8
height_inch &lt;- 6
dpi &lt;- 300

# Set the file name for the TIFF image
tiff_file &lt;- file.path(output_directory, "bubble_plot_percentage.tiff")

# Open the TIFF device
tiff(tiff_file, width = width_inch, height = height_inch, units = "in", res = dpi)

# Print and save the bubble plot
print(bubble_plot_percentage)

# Close the TIFF device
dev.off()

</code></pre>

<hr>
<h2 id='clustering_average_similarity'>Perform Hierarchical Clustering for average similarity matrix</h2><span id='topic+clustering_average_similarity'></span>

<h3>Description</h3>

<p>This function performs hierarchical clustering based on a similarity matrix and provides additional information
such as a colored dendrogram, clustered data, sample percentage within clusters, and total percentage for each cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clustering_average_similarity(
  similarity_matrix,
  num_clusters_option = 10,
  cut_height_option = 0.2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clustering_average_similarity_+3A_similarity_matrix">similarity_matrix</code></td>
<td>
<p>A matrix containing similarity values between SampleIDs.</p>
</td></tr>
<tr><td><code id="clustering_average_similarity_+3A_num_clusters_option">num_clusters_option</code></td>
<td>
<p>The desired number of clusters. Default is 10.</p>
</td></tr>
<tr><td><code id="clustering_average_similarity_+3A_cut_height_option">cut_height_option</code></td>
<td>
<p>The height at which the dendrogram should be cut to obtain clusters. Default is 0.2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the dendrogram, clustered data, sample percentage within clusters, and total percentage for each cluster.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

# If you want to sample 10% from each SampleID group:
sampled_data &lt;- data_sampling(final_data, sample_proportion = 0.1)

alignment_results &lt;- alignment_info(final_data, type = "global", verbose = 1)

# Access the resulting data frames
score_matrix &lt;- alignment_results$score_matrix
normalized_score_matrix &lt;- alignment_results$normalized_score_matrix

total_aligned_positions_matrix &lt;- alignment_results$total_aligned_positions_matrix
number_of_matching_positions_matrix &lt;- alignment_results$number_of_matching_positions_matrix

percent_similarity_matrix &lt;- alignment_results$percent_similarity_matrix

alignment_results_list &lt;- alignment_results$alignment_results_list

alignment_info_matrix &lt;- alignment_results$alignment_info_matrix

output_directory &lt;- tempdir()

# Save the list of alignment results to an RDS file
saveRDS(alignment_results_list, file.path(output_directory, "alignment_results_list.rds"))

# Save matrices to files
write.table(score_matrix, file.path(output_directory, "score_matrix.txt"), sep = "\t")
average_percent_similarity &lt;- compute_average_similarity_matrix(percent_similarity_matrix)
print(average_percent_similarity)

output_directory &lt;- tempdir()
width_inch &lt;- 8
height_inch &lt;- 6
dpi &lt;- 300

clustering_result &lt;- clustering_average_similarity(average_percent_similarity)

# Extract the dendrogram and clustered data
dend_colored &lt;- clustering_result$dendrogram
clustered_data &lt;- clustering_result$clustered_data
Cluster_SampleID_Percentage &lt;- clustering_result$Cluster_SampleID_Percentage
Cluster_TotalPercentage &lt;- clustering_result$Cluster_TotalPercentage

tiff_file &lt;- file.path(output_directory, "6. hierarchical_clustering_dendrogram_colored.tiff")

# Save the dendrogram as a TIFF image
tiff(tiff_file, width = width_inch, height = height_inch, units = "in", res = dpi)
plot(dend_colored, main = "Colored Hierarchical Clustering Dendrogram")
dev.off()

# Save the clustered data frame to a CSV file
write.csv(clustered_data, file.path(output_directory, "7. clustered_data.csv"), row.names = FALSE)

</code></pre>

<hr>
<h2 id='clustering_percent_similarity'>Perform Hierarchical Clustering for Percent Similarity</h2><span id='topic+clustering_percent_similarity'></span>

<h3>Description</h3>

<p>This function performs hierarchical clustering based on a percent similarity matrix and provides additional information
such as a colored dendrogram, clustered data, sample percentage within clusters, and total percentage for each cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clustering_percent_similarity(
  similarity_matrix,
  num_clusters_option = 10,
  cut_height_option = 0.2,
  add_to_final_data = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clustering_percent_similarity_+3A_similarity_matrix">similarity_matrix</code></td>
<td>
<p>A matrix containing percent similarity values between SampleIDs.</p>
</td></tr>
<tr><td><code id="clustering_percent_similarity_+3A_num_clusters_option">num_clusters_option</code></td>
<td>
<p>The desired number of clusters. Default is 10.</p>
</td></tr>
<tr><td><code id="clustering_percent_similarity_+3A_cut_height_option">cut_height_option</code></td>
<td>
<p>The height at which the dendrogram should be cut to obtain clusters. Default is 0.2.</p>
</td></tr>
<tr><td><code id="clustering_percent_similarity_+3A_add_to_final_data">add_to_final_data</code></td>
<td>
<p>A logical value indicating whether to add the clustering results to the final data. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the dendrogram, clustered data, sample percentage within clusters, and total percentage for each cluster.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

# If you want to sample 10% from each SampleID group:
sampled_data &lt;- data_sampling(final_data, sample_proportion = 0.1)

alignment_results &lt;- alignment_info(final_data, type = "global", verbose = 1)

# Access the resulting data frames
score_matrix &lt;- alignment_results$score_matrix
normalized_score_matrix &lt;- alignment_results$normalized_score_matrix

total_aligned_positions_matrix &lt;- alignment_results$total_aligned_positions_matrix
number_of_matching_positions_matrix &lt;- alignment_results$number_of_matching_positions_matrix

percent_similarity_matrix &lt;- alignment_results$percent_similarity_matrix

alignment_results_list &lt;- alignment_results$alignment_results_list

alignment_info_matrix &lt;- alignment_results$alignment_info_matrix

output_directory &lt;- tempdir()

# Save the list of alignment results to an RDS file
saveRDS(alignment_results_list, file.path(output_directory, "alignment_results_list.rds"))

# Save matrices to files
write.table(score_matrix, file.path(output_directory, "score_matrix.txt"), sep = "\t")
average_percent_similarity &lt;- compute_average_similarity_matrix(percent_similarity_matrix)
print(average_percent_similarity)

output_directory &lt;- tempdir()
width_inch &lt;- 8
height_inch &lt;- 6
dpi &lt;- 300

clustering_result &lt;- clustering_percent_similarity(percent_similarity_matrix)

# Extract the dendrogram and clustered data
dend_colored &lt;- clustering_result$dendrogram
clustered_data &lt;- clustering_result$clustered_data
Cluster_SampleID_Percentage &lt;- clustering_result$Cluster_SampleID_Percentage
Cluster_TotalPercentage &lt;- clustering_result$Cluster_TotalPercentage

tiff_file &lt;- file.path(output_directory, "6. hierarchical_clustering_dendrogram_colored.tiff")

# Save the dendrogram as a TIFF image
tiff(tiff_file, width = width_inch, height = height_inch, units = "in", res = dpi)
plot(dend_colored, main = "Colored Hierarchical Clustering Dendrogram")
dev.off()

# Save the clustered data frame to a CSV file
write.csv(clustered_data, file.path(output_directory, "7. clustered_data.csv"), row.names = FALSE)

</code></pre>

<hr>
<h2 id='compute_average_similarity_matrix'>Compute Average Similarity Matrix</h2><span id='topic+compute_average_similarity_matrix'></span>

<h3>Description</h3>

<p>This function computes the average similarity matrix based on the percent similarity matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_average_similarity_matrix(percent_similarity_matrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_average_similarity_matrix_+3A_percent_similarity_matrix">percent_similarity_matrix</code></td>
<td>
<p>A matrix containing percent similarity values between sequences.
Rows and columns should be labeled with SequenceIDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing average similarity values between SampleIDs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

# If you want to sample 10% from each SampleID group:
sampled_data &lt;- data_sampling(final_data, sample_proportion = 0.1)

alignment_results &lt;- alignment_info(final_data, type = "global", verbose = 1)

# Access the resulting data frames
score_matrix &lt;- alignment_results$score_matrix
normalized_score_matrix &lt;- alignment_results$normalized_score_matrix

total_aligned_positions_matrix &lt;- alignment_results$total_aligned_positions_matrix
number_of_matching_positions_matrix &lt;- alignment_results$number_of_matching_positions_matrix

percent_similarity_matrix &lt;- alignment_results$percent_similarity_matrix

alignment_results_list &lt;- alignment_results$alignment_results_list

alignment_info_matrix &lt;- alignment_results$alignment_info_matrix

output_directory &lt;- tempdir()

# Save the list of alignment results to an RDS file
saveRDS(alignment_results_list, file.path(output_directory, "alignment_results_list.rds"))

# Save matrices to files
write.table(score_matrix, file.path(output_directory, "score_matrix.txt"), sep = "\t")
average_percent_similarity &lt;- compute_average_similarity_matrix(percent_similarity_matrix)
print(average_percent_similarity)

</code></pre>

<hr>
<h2 id='data_sampling'>Samples data from each SampleID group if specified, otherwise uses the final data.</h2><span id='topic+data_sampling'></span>

<h3>Description</h3>

<p>This function takes a data frame with 'SampleID' and 'SequenceID' columns and
either returns the original data frame (if sample_proportion is NULL) or
samples a specified proportion from each SampleID group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_sampling(final_data, sample_proportion = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="data_sampling_+3A_final_data">final_data</code></td>
<td>
<p>A data frame with 'SampleID' and 'SequenceID' columns.</p>
</td></tr>
<tr><td><code id="data_sampling_+3A_sample_proportion">sample_proportion</code></td>
<td>
<p>Proportion of data to sample from each SampleID group.
If NULL, the original data frame is returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame either with the original data or sampled data based on the specified proportion.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data # use final_data

# If you want to sample 10% from each SampleID group:
sampled_data &lt;- data_sampling(final_data, sample_proportion = 0.1)

</code></pre>

<hr>
<h2 id='expand_accession_ranges'>This function expands accession number ranges, creating a data frame with sample and accession numbers.</h2><span id='topic+expand_accession_ranges'></span>

<h3>Description</h3>

<p>This function expands accession number ranges, creating a data frame with sample and accession numbers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expand_accession_ranges(accession_ranges)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expand_accession_ranges_+3A_accession_ranges">accession_ranges</code></td>
<td>
<p>A named list where each element represents an accession range.
The names of the list elements should correspond to sample names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns 'sample' and 'accession'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of defining accession number ranges.
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015250",
  WRU8 = c("AF245628", "AF353208 to AF353210"),
  WPU13 = "L11934 to L11940",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333086")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)
</code></pre>

<hr>
<h2 id='generate_heatmaps'>Generate Heatmaps</h2><span id='topic+generate_heatmaps'></span>

<h3>Description</h3>

<p>This function generates interactive and static heatmaps based on the average similarity matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_heatmaps(
  average_percent_similarity,
  html_title = "Heatmap of Average Similarity",
  tiff_title = "Heatmap of Average Similarity",
  xlab = "SampleID",
  ylab = "SampleID",
  width_inch = 8,
  height_inch = 6,
  dpi = 300
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_heatmaps_+3A_average_percent_similarity">average_percent_similarity</code></td>
<td>
<p>A matrix containing average similarity values between SampleIDs.
Rows and columns should be labeled with SampleIDs.</p>
</td></tr>
<tr><td><code id="generate_heatmaps_+3A_html_title">html_title</code></td>
<td>
<p>Title for the interactive heatmap (HTML version).</p>
</td></tr>
<tr><td><code id="generate_heatmaps_+3A_tiff_title">tiff_title</code></td>
<td>
<p>Title for the static heatmap (TIFF version).</p>
</td></tr>
<tr><td><code id="generate_heatmaps_+3A_xlab">xlab</code></td>
<td>
<p>Label for the x-axis.</p>
</td></tr>
<tr><td><code id="generate_heatmaps_+3A_ylab">ylab</code></td>
<td>
<p>Label for the y-axis.</p>
</td></tr>
<tr><td><code id="generate_heatmaps_+3A_width_inch">width_inch</code></td>
<td>
<p>Width of the heatmap in inches.</p>
</td></tr>
<tr><td><code id="generate_heatmaps_+3A_height_inch">height_inch</code></td>
<td>
<p>Height of the heatmap in inches.</p>
</td></tr>
<tr><td><code id="generate_heatmaps_+3A_dpi">dpi</code></td>
<td>
<p>Dots per inch for the static heatmap.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the file names of the generated heatmaps.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

# If you want to sample 10% from each SampleID group:
sampled_data &lt;- data_sampling(final_data, sample_proportion = 0.1)

alignment_results &lt;- alignment_info(final_data, type = "global", verbose = 1)

# Access the resulting data frames
score_matrix &lt;- alignment_results$score_matrix
normalized_score_matrix &lt;- alignment_results$normalized_score_matrix

total_aligned_positions_matrix &lt;- alignment_results$total_aligned_positions_matrix
number_of_matching_positions_matrix &lt;- alignment_results$number_of_matching_positions_matrix

percent_similarity_matrix &lt;- alignment_results$percent_similarity_matrix

alignment_results_list &lt;- alignment_results$alignment_results_list

alignment_info_matrix &lt;- alignment_results$alignment_info_matrix

output_directory &lt;- tempdir()

# Save the list of alignment results to an RDS file
saveRDS(alignment_results_list, file.path(output_directory, "alignment_results_list.rds"))

# Save matrices to files
write.table(score_matrix, file.path(output_directory, "score_matrix.txt"), sep = "\t")
average_percent_similarity &lt;- compute_average_similarity_matrix(percent_similarity_matrix)
print(average_percent_similarity)

output_directory &lt;- tempdir()
width_inch &lt;- 8
height_inch &lt;- 6
dpi &lt;- 300

heatmap_files &lt;- generate_heatmaps(average_percent_similarity)

# Save the interactive heatmap as an HTML file ####
html &lt;- file.path(output_directory, "5. heatmap.html")
# htmlwidgets should be installed and loaded
# htmlwidgets::saveWidget(heatmap_files$html, file = html)

# save the TIFFE images ####

# heatmap_tiff_file1
tiff1 &lt;- file.path(output_directory, "5. heatmap1.tiff")
tiff(tiff1, width = width_inch, height = height_inch, units = "in", res = dpi)
heatmap(as.matrix(average_percent_similarity), main = "Heatmap of Average Similarity")

# Close the TIFF device
dev.off()

# heatmap_tiff_file2
tiff2 &lt;- file.path(output_directory, "5. heatmap2.tiff")

# Open the TIFF device
tiff(tiff2, width = width_inch, height = height_inch, units = "in", res = dpi)
print(heatmap_files$tiff2)

# Close the TIFF device
dev.off()

# heatmap_tiff_file3
# tiff3 &lt;- file.path(output_directory, "5. heatmap3.tiff")

# Open the TIFF device and create the heatmap.2 with hierarchical clustering dendrogram
# gplots should be installed and loaded
# gplots::heatmap.2(as.matrix(average_percent_similarity),
#          dendrogram = "row",
#          Colv = "Rowv",
#          scale = "row",
#          main = "Heatmap of Average Similarity")

# Close the TIFF device
# dev.off()

</code></pre>

<hr>
<h2 id='get_sequence_information'>Retrieves sequence information for a given list of accessions.</h2><span id='topic+get_sequence_information'></span>

<h3>Description</h3>

<p>Retrieves sequence information for a given list of accessions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sequence_information(accession, remove_dot_1 = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_sequence_information_+3A_accession">accession</code></td>
<td>
<p>accession number.</p>
</td></tr>
<tr><td><code id="get_sequence_information_+3A_remove_dot_1">remove_dot_1</code></td>
<td>
<p>Logical. If TRUE, removes '.1' from accession numbers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing sequence information for the given accessions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)

</code></pre>

<hr>
<h2 id='preprocess_for_alignment'>Preprocesses data for sequence alignment.</h2><span id='topic+preprocess_for_alignment'></span>

<h3>Description</h3>

<p>This function merges sample and accession information with sequence information,
filters out rows with missing sequences, and extracts relevant columns for the final data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preprocess_for_alignment(sam_acc, seq_info)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preprocess_for_alignment_+3A_sam_acc">sam_acc</code></td>
<td>
<p>A data frame containing sample and accession information.</p>
</td></tr>
<tr><td><code id="preprocess_for_alignment_+3A_seq_info">seq_info</code></td>
<td>
<p>A data frame containing sequence information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the resulting data frames: 'merged_data', 'main_data', 'final_data'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

</code></pre>

<hr>
<h2 id='SampleID_vs_NumSequences'>Plots the number of sequences and the probability of sequences per SampleID.</h2><span id='topic+SampleID_vs_NumSequences'></span>

<h3>Description</h3>

<p>This function takes a data frame with 'SampleID' and 'SequenceID' columns and
creates two bar plots. The first plot shows the number of sequences per SampleID,
and the second plot shows the probability of sequences per SampleID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SampleID_vs_NumSequences(final_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SampleID_vs_NumSequences_+3A_final_data">final_data</code></td>
<td>
<p>A data frame with 'SampleID' and 'SequenceID' columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two ggplot2 plots: 'plot_num_sequences' and 'plot_prob'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

# Example usage
plots &lt;- SampleID_vs_NumSequences(final_data)

output_directory &lt;- tempdir()
# Set the file name for the TIFF images
tiff_file_num_sequences &lt;- file.path(output_directory, "0. SampleID_vs_NumSequences.tiff")
tiff_file_prob &lt;- file.path(output_directory, "0. SampleID_vs_Probability.tiff")

# Set the width, height, and DPI parameters
width_inch &lt;- 8
height_inch &lt;- 8
dpi &lt;- 300

# Open the TIFF devices and save the plots
tiff(tiff_file_num_sequences, width = width_inch, height = height_inch, units = "in", res = dpi)
print(plots$plot_num_sequences)
dev.off()

tiff(tiff_file_prob, width = width_inch, height = height_inch, units = "in", res = dpi)
print(plots$plot_prob)
dev.off()

</code></pre>

<hr>
<h2 id='tree_average_similarity'>Generate Phylogenetic Tree and Color Palette for Average Similarity</h2><span id='topic+tree_average_similarity'></span>

<h3>Description</h3>

<p>This function generates a Neighbor-Joining phylogenetic tree and a color palette based on the average similarity matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_average_similarity(similarity_matrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree_average_similarity_+3A_similarity_matrix">similarity_matrix</code></td>
<td>
<p>A matrix containing pairwise similarities between samples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the phylogenetic tree and a color palette.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

# If you want to sample 10% from each SampleID group:
sampled_data &lt;- data_sampling(final_data, sample_proportion = 0.1)

alignment_results &lt;- alignment_info(final_data, type = "global", verbose = 1)

# Access the resulting data frames
score_matrix &lt;- alignment_results$score_matrix
normalized_score_matrix &lt;- alignment_results$normalized_score_matrix

total_aligned_positions_matrix &lt;- alignment_results$total_aligned_positions_matrix
number_of_matching_positions_matrix &lt;- alignment_results$number_of_matching_positions_matrix

percent_similarity_matrix &lt;- alignment_results$percent_similarity_matrix

alignment_results_list &lt;- alignment_results$alignment_results_list

alignment_info_matrix &lt;- alignment_results$alignment_info_matrix

output_directory &lt;- tempdir()

# Save the list of alignment results to an RDS file
saveRDS(alignment_results_list, file.path(output_directory, "alignment_results_list.rds"))

# Save matrices to files
write.table(score_matrix, file.path(output_directory, "score_matrix.txt"), sep = "\t")
average_percent_similarity &lt;- compute_average_similarity_matrix(percent_similarity_matrix)
print(average_percent_similarity)

output_directory &lt;- tempdir()
width_inch &lt;- 8
height_inch &lt;- 6
dpi &lt;- 300

clustering_result &lt;- clustering_average_similarity(average_percent_similarity)

# Extract the dendrogram and clustered data
dend_colored &lt;- clustering_result$dendrogram
clustered_data &lt;- clustering_result$clustered_data
Cluster_SampleID_Percentage &lt;- clustering_result$Cluster_SampleID_Percentage
Cluster_TotalPercentage &lt;- clustering_result$Cluster_TotalPercentage

tiff_file &lt;- file.path(output_directory, "6. hierarchical_clustering_dendrogram_colored.tiff")

# Save the dendrogram as a TIFF image
tiff(tiff_file, width = width_inch, height = height_inch, units = "in", res = dpi)
plot(dend_colored, main = "Colored Hierarchical Clustering Dendrogram")
dev.off()

# Save the clustered data frame to a CSV file
write.csv(clustered_data, file.path(output_directory, "7. clustered_data.csv"), row.names = FALSE)

# Example usage with similarity_matrix
result &lt;- tree_average_similarity(average_percent_similarity)
tree &lt;- result$tree
color_palette &lt;- result$color_palette

output_directory &lt;- tempdir()

tree_newick &lt;- file.path(output_directory, "phylogenetic_tree_nj.nwk")

# Save the phylogenetic tree as a Newick file
# ape::write.tree(tree, file = tree_newick)

# Save the phylogenetic tree as a TIFF image
width_inch &lt;- 8
height_inch &lt;- 6
dpi &lt;- 300

# Set the file name for the TIFF image
tiff_file &lt;- file.path(output_directory, "phylogenetic_tree.tiff")

# Open the TIFF device
tiff(tiff_file, width = width_inch, height = height_inch, units = "in", res = dpi)

# Plot the phylogenetic tree vertically
plot(tree, main = "Neighbor-Joining Tree", cex = 1, direction = "downward")

# Close the TIFF device
dev.off()

# Set the file name for the TIFF image with rainbow-colored branches
tiff_file_rainbow &lt;- file.path(output_directory, "phylogenetic_tree_rainbow.tiff")

# Open the TIFF device
tiff(tiff_file_rainbow, width = width_inch, height = height_inch, units = "in", res = dpi)

# Plot the phylogenetic tree vertically with rainbow-colored branches
plot(tree, main = "NJ Tree", cex = 1, direction = "downward", tip.color = color_palette)

# Close the TIFF device
dev.off()

</code></pre>

<hr>
<h2 id='tree_percent_similarity'>Generate Phylogenetic Tree and Color Palette for Percent Similarity Matrix</h2><span id='topic+tree_percent_similarity'></span>

<h3>Description</h3>

<p>This function generates a Neighbor-Joining phylogenetic tree and a color palette based on the percent similarity matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_percent_similarity(percent_similarity_matrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree_percent_similarity_+3A_percent_similarity_matrix">percent_similarity_matrix</code></td>
<td>
<p>A matrix containing pairwise percent similarities between samples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the phylogenetic tree and a color palette.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

# If you want to sample 10% from each SampleID group:
sampled_data &lt;- data_sampling(final_data, sample_proportion = 0.1)

alignment_results &lt;- alignment_info(final_data, type = "global", verbose = 1)

# Access the resulting data frames
score_matrix &lt;- alignment_results$score_matrix
normalized_score_matrix &lt;- alignment_results$normalized_score_matrix

total_aligned_positions_matrix &lt;- alignment_results$total_aligned_positions_matrix
number_of_matching_positions_matrix &lt;- alignment_results$number_of_matching_positions_matrix

percent_similarity_matrix &lt;- alignment_results$percent_similarity_matrix

alignment_results_list &lt;- alignment_results$alignment_results_list

alignment_info_matrix &lt;- alignment_results$alignment_info_matrix

output_directory &lt;- tempdir()

# Save the list of alignment results to an RDS file
saveRDS(alignment_results_list, file.path(output_directory, "alignment_results_list.rds"))

# Save matrices to files
write.table(score_matrix, file.path(output_directory, "score_matrix.txt"), sep = "\t")
average_percent_similarity &lt;- compute_average_similarity_matrix(percent_similarity_matrix)
print(average_percent_similarity)

output_directory &lt;- tempdir()
width_inch &lt;- 8
height_inch &lt;- 6
dpi &lt;- 300

clustering_result &lt;- clustering_percent_similarity(percent_similarity_matrix)

# Extract the dendrogram and clustered data
dend_colored &lt;- clustering_result$dendrogram
clustered_data &lt;- clustering_result$clustered_data
Cluster_SampleID_Percentage &lt;- clustering_result$Cluster_SampleID_Percentage
Cluster_TotalPercentage &lt;- clustering_result$Cluster_TotalPercentage

tiff_file &lt;- file.path(output_directory, "6. hierarchical_clustering_dendrogram_colored.tiff")

# Save the dendrogram as a TIFF image
tiff(tiff_file, width = width_inch, height = height_inch, units = "in", res = dpi)
plot(dend_colored, main = "Colored Hierarchical Clustering Dendrogram")
dev.off()

# Save the clustered data frame to a CSV file
write.csv(clustered_data, file.path(output_directory, "7. clustered_data.csv"), row.names = FALSE)

# Example usage with percent_similarity_matrix
result &lt;- tree_percent_similarity(percent_similarity_matrix)
tree &lt;- result$tree
color_palette &lt;- result$color_palette

tree_newick &lt;- file.path(output_directory, "phylogenetic_tree_nj.nwk")

# Save the phylogenetic tree as a Newick file
# ape::write.tree(tree, file = tree_newick)

# Save the phylogenetic tree as a TIFF image
width_inch &lt;- 8
height_inch &lt;- 6
dpi &lt;- 300

# Set the file name for the TIFF image
tiff_file &lt;- file.path(output_directory, "phylogenetic_tree.tiff")

# Open the TIFF device
tiff(tiff_file, width = width_inch, height = height_inch, units = "in", res = dpi)

# Plot the phylogenetic tree vertically
plot(tree, main = "Neighbor-Joining Tree", cex = 1, direction = "downward")

# Close the TIFF device
dev.off()

# Set the file name for the TIFF image with rainbow-colored branches
tiff_file_rainbow &lt;- file.path(output_directory, "phylogenetic_tree_rainbow.tiff")

# Open the TIFF device
tiff(tiff_file_rainbow, width = width_inch, height = height_inch, units = "in", res = dpi)

# Plot the phylogenetic tree vertically with rainbow-colored branches
plot(tree, main = "NJ Tree", cex = 1, direction = "downward", tip.color = color_palette)

# Close the TIFF device
dev.off()

</code></pre>

<hr>
<h2 id='write_fasta'>Writes a data frame to a FASTA file.</h2><span id='topic+write_fasta'></span>

<h3>Description</h3>

<p>This function takes a data frame with 'SequenceID' and 'Sequence' columns
and writes the contents to a FASTA file. Each row in the data frame corresponds
to a sequence in the FASTA file, with the 'SequenceID' used as the header line
and the 'Sequence' as the sequence line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_fasta(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write_fasta_+3A_data">data</code></td>
<td>
<p>A data frame with 'SequenceID' and 'Sequence' columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector representing the contents of the FASTA file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
accession_ranges &lt;- list(
  SRU1 = "AJ240966 to AJ240970",
  STU2 = "AB015240 to AB015245",
  WPU13 = "L11934 to L11939",
  INU20 = c("AF277467 to AF277470", "AF333080 to AF333085")
)

# Use the function to expand accession ranges
sam_acc &lt;- expand_accession_ranges(accession_ranges)
print(sam_acc)

# 2 get_sequence_information
accessions_to_query &lt;- sam_acc$accession
seq_info &lt;- get_sequence_information(accessions_to_query, remove_dot_1 = TRUE)
print(seq_info)
result &lt;- preprocess_for_alignment(sam_acc, seq_info)

# Access the resulting data frames
merged_data &lt;- result$merged_data
main_data &lt;- result$main_data
final_data &lt;- result$final_data

data &lt;- final_data

# Call the function
fasta_content &lt;- write_fasta(data)

# Print or use the `fasta_content` as needed
print(fasta_content)

output_directory &lt;- tempdir()
# Specify the output file path
output_file_path &lt;- file.path(output_directory, "output.fasta")

# Open a connection to the output file
output_file &lt;- file(output_file_path, "w")

# Write the content to the file
writeLines(fasta_content, output_file)

# Close the output file
close(output_file)

# Print a message indicating successful file creation
warning("FASTA file has been created and saved at:", output_file_path, "\n")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
