<!DOCTYPE html><html><head><title>Help for package gamclass</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gamclass}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gamclass-package'>
<p>Functions and Data for a Course in Modern Regression</p></a></li>
<li><a href='#addhlines'>
<p>Add horizontal lines to plot.</p></a></li>
<li><a href='#airAccs'>
<p>Aircraft Crash data</p></a></li>
<li><a href='#bomregions2018'><p>Australian and Related Historical Annual Climate Data, by Region</p></a></li>
<li><a href='#bronchitis'><p>Chronic bronchitis in a sample of men in Cardiff</p></a></li>
<li><a href='#bssBYcut'>
<p>Between group SS for <code>y</code>, for all possible splits on values of <code>x</code></p></a></li>
<li><a href='#compareModels'><p>Compare accuracy of alternative classification methods</p></a></li>
<li><a href='#confusion'>
<p>Given actual and predicted group assignments, give the confusion matrix</p></a></li>
<li><a href='#coralPval'>
<p>P-values from biological expression array data</p></a></li>
<li><a href='#cvalues'>
<p>Historical speed of light measurements</p></a></li>
<li><a href='#CVcluster'>
<p>Cross-validation estimate of predictive accuracy for clustered data</p></a></li>
<li><a href='#CVgam'>
<p>Cross-validation estimate of accuracy from GAM model fit</p></a></li>
<li><a href='#eventCounts'>
<p>Tabulate vector of dates by specified time event</p></a></li>
<li><a href='#FARS'>
<p>US fatal road accident data for automobiles, 1998 to 2010</p></a></li>
<li><a href='#fars2007'>
<p>US Fatal Road Accident Data, 2007 and 2008</p></a></li>
<li><a href='#frontDeaths'>
<p>Safety Device effectiveness Measures, by Year</p></a></li>
<li><a href='#gamRF'>
<p>Random forest fit to residuals from GAM model</p></a></li>
<li><a href='#german'>
<p>German credit scoring data</p></a></li>
<li><a href='#greatLakesM'>
<p>Monthly Great Lake heights: 1918 - 2019</p></a></li>
<li><a href='#ldaErr'><p>Calculate Error Rates for Linear Discriminant Model</p></a></li>
<li><a href='#loti'>
<p>Global temperature anomalies</p></a></li>
<li><a href='#plotFars'>
<p>Plot Protection Device Effectiveness Measure Against Year</p></a></li>
<li><a href='#relDeaths'>
<p>Yearly Driver deaths, as Fraction of Deaths for All Years</p></a></li>
<li><a href='#RFcluster'>
<p>Random forests estimate of predictive accuracy for clustered data</p></a></li>
<li><a href='#rfErr'><p>Calculate Error Rates for randomForest model</p></a></li>
<li><a href='#rpartErr'><p>Calculate Error Rates for rpart model</p></a></li>
<li><a href='#simreg'>
<p>Simulate (repeated) regression calculations</p></a></li>
<li><a href='#tabFarsDead'>
<p>Extract ratio of ratios estimate of safety device effectiveness, from</p>
the <code>Fars</code> dataset.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Functions and Data for a Course on Modern Regression and
Classification</td>
</tr>
<tr>
<td>Version:</td>
<td>0.62.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-21</td>
</tr>
<tr>
<td>Author:</td>
<td>John Maindonald</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>John Maindonald &lt;john@statsresearch.co.nz&gt;</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>sp, kernlab, mlbench, car, mgcv, DAAG, MASS,
knitr,prettydoc,rmarkdown,bookdown</td>
</tr>
<tr>
<td>Imports:</td>
<td>rpart, randomForest, lattice, latticeExtra, methods</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr,rmarkdown,bookdown</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions and data are provided that support a course
        that emphasizes statistical issues of inference and generalizability.  
        The functions are designed to make it straightforward to illustrate
        the use of cross-validation, the training/test approach, simulation, 
        and model-based estimates of accuracy.  Methods considered are
        Generalized Additive Modeling, Linear and Quadratic Discriminant
        Analysis, Tree-based methods, and Random Forests. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-21 08:35:17 UTC; johnm1</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-21 09:12:42 UTC</td>
</tr>
</table>
<hr>
<h2 id='gamclass-package'>
Functions and Data for a Course in Modern Regression
</h2><span id='topic+gamclass-package'></span><span id='topic+gamclass'></span>

<h3>Description</h3>

<p>For purposes of this package, modern regression extends to include
classification and multivariate exploration.  
A strong focus is on
methods described in Wood (2017) &lt;doi:10.1201/9781315370279&gt;
</p>


<h3>Details</h3>

<p>Functions are mostly designed to facilitate a variety of cross-validation
and bootstrap calculations.
</p>


<h3>Author(s)</h3>

<p>John Maindonald
</p>
<p>Maintainer: jhmaindonald@gmail.com
</p>


<h3>References</h3>

<p>Venables, W N, &amp; Ripley, B D (2013). Modern applied
statistics with S-PLUS. Springer Science &amp; Business Media.
</p>
<p>Wood, S N (2017) Generalized Additive Models: An Introduction
with R (2nd edition). Chapman and Hall/CRC.
</p>
<p><a href="https://github.com/jhmaindonald/gamclass">https://github.com/jhmaindonald/gamclass</a>
</p>

<hr>
<h2 id='addhlines'>
Add horizontal lines to plot.
</h2><span id='topic+addhlines'></span>

<h3>Description</h3>

<p>This is designed for adding horizontal lines that show predicted
values to a plot of observed values versus x-values,
in <code><a href="rpart.html#topic+rpart">rpart</a></code> regression. Where predicted values change
between two successive x-values lines are extended to the midway
point. This reflects the way that <code><a href="rpart.html#topic+predict.rpart">predict.rpart</a></code>
handles predictions for new data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addhlines(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addhlines_+3A_x">x</code></td>
<td>

<p>Vector of predictor variable values.
</p>
</td></tr>
<tr><td><code id="addhlines_+3A_y">y</code></td>
<td>

<p>Vector of predicted values.
</p>
</td></tr>
<tr><td><code id="addhlines_+3A_...">...</code></td>
<td>

<p>Additional graphics parameters, for passing through to the
<code>lines()</code>
function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Lines are added to the current graph.
</p>


<h3>Author(s)</h3>

<p>John Maindonald
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(34, 18, 45, 18, 27, 24, 34, 20, 24, 28, 21, 18)
y &lt;- c(14, 11, 12, 9, 4, 11, 6, 9, 4, 10, 9, 2)
hat &lt;- c(10.5, 7.75, 10.5, 7.75, 7, 7, 10.5, 7.75, 7, 10.5, 7, 7.75)
plot(x, y)
addhlines(x, hat, lwd=2, col="gray")

## The function is currently defined as
function(x,y, ...){
  ordx &lt;- order(x)
  xo &lt;- x[ordx]
  yo &lt;- y[ordx]
  breaks &lt;- diff(yo)!=0
  xh &lt;- c(xo[1],0.5*(xo[c(FALSE,breaks)]+xo[c(breaks, FALSE)]))
  yh &lt;- yo[c(TRUE, breaks)]
  y3 &lt;- x3 &lt;- numeric(3*length(xh)-1)
  loc1 &lt;- seq(from=1, to=length(x3), by=3)
  x3[loc1] &lt;- xh
  x3[loc1+1]&lt;- c(xh[-1], max(x))
  x3[loc1[-length(loc1)]+2] &lt;- NA
  y3[loc1[-length(loc1)]+2] &lt;- NA
  y3[loc1] &lt;- yh
  y3[loc1+1] &lt;- yh
  lines(x3,y3, ...)
}
</code></pre>

<hr>
<h2 id='airAccs'>
Aircraft Crash data
</h2><span id='topic+airAccs'></span>

<h3>Description</h3>

<p>Aircraft Crash Data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(airAccs)
</code></pre>


<h3>Format</h3>

<p>A data frame with 5666 observations on the following 7 variables.
</p>

<dl>
<dt><code>Date</code></dt><dd><p>Date of Accident</p>
</dd>
<dt><code>location</code></dt><dd><p>Location of accident</p>
</dd>
<dt><code>operator</code></dt><dd><p>Aircraft operator</p>
</dd>
<dt><code>planeType</code></dt><dd><p>Aircraft type</p>
</dd>
<dt><code>Dead</code></dt><dd><p>Number of deaths</p>
</dd>
<dt><code>Aboard</code></dt><dd><p>Number aboard</p>
</dd>
<dt><code>Ground</code></dt><dd><p>Deaths on ground</p>
</dd>    
</dl>



<h3>Details</h3>

<p>For details of inclusion criteria, see
<a href="https://www.planecrashinfo.com/database.htm">https://www.planecrashinfo.com/database.htm</a>
</p>


<h3>Source</h3>

<p><a href="https://www.planecrashinfo.com/database.htm">https://www.planecrashinfo.com/database.htm</a>
</p>


<h3>References</h3>

<p><a href="https://www.planecrashinfo.com/reference.htm">https://www.planecrashinfo.com/reference.htm</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(airAccs)
str(airAccs)
</code></pre>

<hr>
<h2 id='bomregions2018'>Australian and Related Historical Annual Climate Data, by Region</h2><span id='topic+bomregions2018'></span>

<h3>Description</h3>

<p>Australian regional temperature data, Australian regional rainfall
data, and Annual SOI, are given for the years 1900-2018.  The regional 
rainfall and temperature data are area-weighted averages for the 
respective regions.  The Southern Oscillation Index (SOI) is the 
difference in barometric pressure at sea level between Tahiti and Darwin.
Data through to 2021, including also the Indian Ocean Dipole, is available in the file <code>DAAG::bomregions2021</code>
.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("bomregions2018")</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Year</dt><dd><p>Year</p>
</dd>
<dt>seAVt</dt><dd><p>Southeastern region average temperature (degrees C)</p>
</dd>
<dt>southAVt</dt><dd><p>Southern temperature</p>
</dd>
<dt>eastAVt</dt><dd><p>Eastern temperature</p>
</dd>
<dt>northAVt</dt><dd><p>Northern temperature</p>
</dd>
<dt>swAVt</dt><dd><p>Southwestern temperature</p>
</dd>
<dt>qldAVt</dt><dd><p> temperature</p>
</dd>
<dt>nswAVt</dt><dd><p> temperature</p>
</dd>
<dt>ntAVt</dt><dd><p> temperature</p>
</dd>
<dt>saAVt</dt><dd><p> temperature</p>
</dd>
<dt>tasAVt</dt><dd><p> temperature</p>
</dd>
<dt>vicAVt</dt><dd><p> temperature</p>
</dd>
<dt>waAVt</dt><dd><p> temperature</p>
</dd>
<dt>mdbAVt</dt><dd><p>Murray-Darling basin temperature</p>
</dd>
<dt>ausAVt</dt><dd><p>Australian average temperature, area-weighted mean</p>
</dd>
<dt>seRain</dt><dd><p>Southeast Australian annual rainfall (mm)</p>
</dd>
<dt>southRain</dt><dd><p>Southern rainfall</p>
</dd>
<dt>eastRain</dt><dd><p>Eastern rainfall</p>
</dd>
<dt>northRain</dt><dd><p>Northern rainfall</p>
</dd>
<dt>swRain</dt><dd><p>Southwest rainfall</p>
</dd>
<dt>qldRain</dt><dd><p>Queensland rainfall</p>
</dd>
<dt>nswRain</dt><dd><p>NSW rainfall</p>
</dd>
<dt>ntRain</dt><dd><p>Northern Territory rainfall</p>
</dd> 
<dt>saRain</dt><dd><p>South Australian rainfall</p>
</dd> 
<dt>tasRain</dt><dd><p>Tasmanian rainfall</p>
</dd>
<dt>vicRain</dt><dd><p>Victorian rainfall</p>
</dd>
<dt>waRain</dt><dd><p>West Australian rainfall</p>
</dd> 
<dt>mdbRain</dt><dd><p>Murray-Darling basin rainfall</p>
</dd>
<dt>ausRain</dt><dd><p>Australian average rainfall, area weighted</p>
</dd>
<dt>SOI</dt><dd><p>Annual average Southern Oscillation Index</p>
</dd>
<dt>sunspot</dt><dd><p>Annual average sunspot counts</p>
</dd>
<dt>co2mlo</dt><dd><p>Moana Loa CO2 concentrations, from 1959</p>
</dd>
<dt>co2law</dt><dd><p>Moana Loa CO2 concentrations, 1900 to 1978</p>
</dd>
<dt>CO2</dt><dd><p>CO2 concentrations, composite series</p>
</dd>
<dt>avDMI</dt><dd><p>Annual average Dipole Mode Index, for the Indian Ocean Dipole</p>
</dd>
</dl>



<h3>Source</h3>

<p>Australian Bureau of Meteorology web pages:
</p>
<p><a href="http://www.bom.gov.au/climate/change/index.shtml">http://www.bom.gov.au/climate/change/index.shtml</a>
</p>
<p>The SOI data are from <a href="http://www.bom.gov.au/climate/enso/#tabs=SOI">http://www.bom.gov.au/climate/enso/#tabs=SOI</a>.
</p>
<p>The CO2 series <code>co2law</code>, for Law Dome ice core data. is from
<a href="https://data.ess-dive.lbl.gov/portals/CDIAC">https://data.ess-dive.lbl.gov/portals/CDIAC</a>.
</p>
<p>The CO2 series <code>co2mlo</code> is from Dr. Pieter Tans, NOAA/ESRL
(<a href="https://gml.noaa.gov/ccgg/trends/">https://gml.noaa.gov/ccgg/trends/</a>)
</p>
<p>The series <code>CO2</code> is a composite series, obtained by adding 0.46 to
he Law data for 1900 to 1958, then following this with the Moana Loa
data that is avaiable from 1959.  The addition of 0.46 is designed so
that the averages from the two series agree for the period 1959 to
1968
</p>
<p>Sunspot data is from <a href="http://www.sidc.be/silso/datafiles">http://www.sidc.be/silso/datafiles</a>
</p>


<h3>References</h3>

<p>D.M. Etheridge, L.P. Steele, R.L. Langenfelds, R.J. Francey,
J.-M. Barnola and V.I. Morgan, 1998, <em>Historical CO2 records from the
Law Dome DE08, DE08-2, and DSS ice cores</em>, in Trends: A Compendium of
Data on Global Change, on line at Carbon Dioxide Information Analysis
Center, Oak Ridge National Laboratory, U.S. Department of Energy, Oak
Ridge, Tenn., U.S.A.
</p>
<p>Lavery, B., Joung, G. and Nicholls, N. 1997. An extended high-quality
historical rainfall dataset for Australia. Australian Meteorological
Magazine, 46, 27-38.
</p>
<p>Nicholls, N., Lavery, B., Frederiksen, C.\ and Drosdowsky, W. 1996.
Recent apparent changes in relationships between the El Nino &ndash;
southern oscillation and Australian rainfall and temperature.
Geophysical Research Letters 23: 3357-3360.
</p>
<p>SIDC-team, World Data Center for the Sunspot Index, Royal Observatory of
Belgium, Monthly Report on the International Sunspot Number, online
catalogue of the sunspot index: <a href="http://www.sidc.be/silso/datafiles">http://www.sidc.be/silso/datafiles</a>,
1900-2011
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(ts(bomregions2018[, c("mdbRain","SOI")], start=1900),
     panel=function(y,...)panel.smooth(bomregions2018$Year, y,...))
avrain &lt;- bomregions2018[,"mdbRain"]
xbomsoi &lt;- with(bomregions2018, data.frame(Year=Year, SOI=SOI,
                cuberootRain=avrain^0.33))
xbomsoi$trendSOI &lt;- lowess(xbomsoi$SOI, f=0.1)$y
xbomsoi$trendRain &lt;- lowess(xbomsoi$cuberootRain, f=0.1)$y
xbomsoi$detrendRain &lt;-
  with(xbomsoi, cuberootRain - trendRain + mean(trendRain))
xbomsoi$detrendSOI &lt;-
  with(xbomsoi, SOI - trendSOI + mean(trendSOI))
## Plot time series avrain and SOI: ts object xbomsoi
plot(ts(xbomsoi[, c("cuberootRain","SOI")], start=1900),
     panel=function(y,...)panel.smooth(xbomsoi$Year, y,...),
     xlab = "Year", main="", ylim=list(c(250, 800),c(-20,25)))
par(mfrow=c(1,2))
rainpos &lt;- pretty(xbomsoi$cuberootRain^3, 6)
plot(cuberootRain ~ SOI, data = xbomsoi,
     ylab = "Rainfall (cube root scale)", yaxt="n")
axis(2, at = rainpos^0.33, labels=paste(rainpos))
mtext(side = 3, line = 0.8, "A", adj = -0.025)
with(xbomsoi, lines(lowess(cuberootRain ~ SOI, f=0.75)))
plot(detrendRain ~ detrendSOI, data = xbomsoi,
     xlab="Detrended SOI", ylab = "Detrended rainfall", yaxt="n")
axis(2, at = rainpos^0.33, labels=paste(rainpos))
with(xbomsoi, lines(lowess(detrendRain ~ detrendSOI, f=0.75)))
mtext(side = 3, line = 0.8, "B", adj = -0.025)
par(mfrow=c(1,1))
</code></pre>

<hr>
<h2 id='bronchitis'>Chronic bronchitis in a sample of men in Cardiff</h2><span id='topic+bronchitis'></span>

<h3>Description</h3>

<p>The data consist of observations on three variables
for each of 212 men in a sample of Cardiff enumeration districts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bronchitis</code></pre>


<h3>Format</h3>

<p>A data.frame of 212 obs of 3 variables:
</p>

<dl>
<dt><code>cig</code></dt><dd><p>numeric, the number of cigarettes per day</p>
</dd>
<dt><code>poll</code></dt><dd><p>numeric, the smoke level in the locality</p>
</dd>
<dt><code>r</code></dt><dd><p>integer, 1= respondent suffered from chronic bronchitis</p>
</dd>
<dt><code>rfac</code></dt><dd><p>factor, with levels <code>abs</code> (<code>r</code>=0), and <code>abs</code> (<code>r</code>=0)</p>
</dd>
</dl>



<h3>Note</h3>

<p>See p.224 in SMIR</p>


<h3>Source</h3>

<p>This copy of the dataset was copied from version 0.02 of the <span class="pkg">SMIR</span>
package, which in turn obtained it from Jones (1975).</p>


<h3>References</h3>

<p>Jones, K. (1975), <em>A geographical contribution to the aetiology of
chronic bronchitis</em>, Unpublished BSc dissertation,
University of Southampton. Published in
Wrigley, N. (1976). <em>Introduction to the use of
logit models in geography</em>, Geo.Abstracts Ltd, CATMOG
10, University of East Anglia, Norwich.
</p>
<p>Murray Aitkin, Brian Francis, John Hinde and Ross Darnell (2009). SMIR: Companion to
Statistical Modelling in R (SMIR). Oxford University Press.	
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bronchit)
</code></pre>

<hr>
<h2 id='bssBYcut'>
Between group SS for <code>y</code>, for all possible splits on values of <code>x</code>
</h2><span id='topic+bssBYcut'></span>

<h3>Description</h3>

<p>Each point of separation between successve values of <code>x</code> is used
in turn to create two groups of observations.  The between group sum
of squares for <code>y</code> is calculated for each such split.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bssBYcut(x, y, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bssBYcut_+3A_x">x</code></td>
<td>

<p>Variable (numeric) used to define splits. Observations with <code>x</code>
values less than the cut point go into the first group, while those
with values <code>&gt;=</code> the cut point go into the second group.
</p>
</td></tr>
<tr><td><code id="bssBYcut_+3A_y">y</code></td>
<td>

<p>Variable for which BSS values are to be calculated.
</p>
</td></tr>
<tr><td><code id="bssBYcut_+3A_data">data</code></td>
<td>

<p>Data frame with columns <code>x</code> and <code>y</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with columns:
</p>
<table>
<tr><td><code>xOrd</code></td>
<td>
<p>Cut points for splits.</p>
</td></tr>
<tr><td><code>comp2</code></td>
<td>
<p>Between groups sum of squares</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>J H Maindonald
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xy &lt;- bssBYcut(weight, height, women)
with(xy, xy[which.max(bss), ])

## The function is currently defined as
function (x, y, data)
{
    xnam &lt;- deparse(substitute(x))
    ynam &lt;- deparse(substitute(y))
    xv &lt;- data[, xnam]
    yv &lt;- data[, ynam]
    sumss &lt;- function(x, y, cut) {
        av &lt;- mean(y)
        left &lt;- x &lt; cut
        sum(left) * (mean(y[left]) - av)^2 + sum(!left) * (mean(y[!left]) -
            av)^2
    }
    xOrd &lt;- unique(sort(xv))[-1]
    bss &lt;- numeric(length(xOrd))
    for (i in 1:length(xOrd)) {
        bss[i] &lt;- sumss(xv, yv, xOrd[i])
    }
    list(xOrd = xOrd, bss = bss)
  }
</code></pre>

<hr>
<h2 id='compareModels'>Compare accuracy of alternative classification methods</h2><span id='topic+compareModels'></span>

<h3>Description</h3>

<p>Compare, between models, probabilities that the models assign to
membership in the correct group or class. Probabilites should be
estimated from cross-validation or from bootstrap out-of-bag data
or preferably for test data that are completely separate from the
data used to dervive the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareModels(groups, estprobs = list(lda = NULL, rf = NULL),
            gpnames = NULL, robust = TRUE, print = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compareModels_+3A_groups">groups</code></td>
<td>
<p>Factor that specifies the groups</p>
</td></tr>
<tr><td><code id="compareModels_+3A_estprobs">estprobs</code></td>
<td>

<p>List whose elements (with names that identify the models) are
matrices that give for each observation (row) estimated
probabilities of membership for each of the groups (columns).</p>
</td></tr>
<tr><td><code id="compareModels_+3A_gpnames">gpnames</code></td>
<td>
<p>Character: names for groups, if different from
<code>levels(groups)</code></p>
</td></tr>
<tr><td><code id="compareModels_+3A_robust">robust</code></td>
<td>
<p>Logical, <code>TRUE</code> or <code>FALSE</code></p>
</td></tr>
<tr><td><code id="compareModels_+3A_print">print</code></td>
<td>
<p>Logical.  Should results be printed?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The estimated probabilities are compared directly,  under normal
distribution assumptions.  An effect is fitted for each observation,
plus an effect for the method.  Comparison on a logit scale may
sometimes be preferable.  An option to allow this is scheduled for
incorporation in a later version.
</p>


<h3>Value</h3>

<table>
<tr><td><code>modelAVS</code></td>
<td>
<p>Average accuracies for models</p>
</td></tr>
<tr><td><code>modelSE</code></td>
<td>
<p>Approximate average SE for comparing models</p>
</td></tr>
<tr><td><code>gpAVS</code></td>
<td>
<p>Average accuracies for groups</p>
</td></tr>
<tr><td><code>gpSE</code></td>
<td>
<p>Approximate average SE for comparing groups</p>
</td></tr>
<tr><td><code>obsEff</code></td>
<td>
<p>Effects assigned to individual observations</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The analysis estimates effects due to model and group (<code>gp</code>),
after accounting for differences between observations.
</p>


<h3>Author(s)</h3>

<p>John Maindonald
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(MASS)
library(DAAG)
library(randomForest)
ldahat &lt;- lda(species ~ length+breadth, data=cuckoos, CV=TRUE)$posterior
qdahat &lt;- qda(species ~ length+breadth, data=cuckoos, CV=TRUE)$posterior
rfhat &lt;- predict(randomForest(species ~ length+breadth, data=cuckoos),
                 type="prob")
compareModels(groups=cuckoos$species, estprobs=list(lda=ldahat,
              qda=qdahat, rf=rfhat), robust=FALSE)
</code></pre>

<hr>
<h2 id='confusion'>
Given actual and predicted group assignments, give the confusion matrix
</h2><span id='topic+confusion'></span>

<h3>Description</h3>

<p>Given actual and predicted group assignments, give the confusion matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>confusion(actual, predicted, gpnames = NULL, rowcol=c("actual", "predicted"),
printit = c("overall","confusion"), prior = NULL, digits=3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="confusion_+3A_actual">actual</code></td>
<td>

<p>Actual (prior) group assigments
</p>
</td></tr>
<tr><td><code id="confusion_+3A_predicted">predicted</code></td>
<td>

<p>Predicted group assigments.
</p>
</td></tr>
<tr><td><code id="confusion_+3A_gpnames">gpnames</code></td>
<td>

<p>Names for groups, if different from <code>levels(actual)</code>
</p>
</td></tr>
<tr><td><code id="confusion_+3A_rowcol">rowcol</code></td>
<td>
<p>For predicted categories to appear as rows,
specify <code>rowcol="predicted"</code>
</p>
</td></tr>
<tr><td><code id="confusion_+3A_printit">printit</code></td>
<td>

<p>Character vector. Print <code>"overall"</code>, or <code>"confusion"</code> matrix, or both.
</p>
</td></tr>
<tr><td><code id="confusion_+3A_prior">prior</code></td>
<td>

<p>Prior probabilities for groups, if different from the relative group
frequencies
</p>
</td></tr>
<tr><td><code id="confusion_+3A_digits">digits</code></td>
<td>
<p>Number of decimal digits to display in printed output
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Predicted group assignments should be estimated from cross-validation
or from bootstrap out-of-bag data.  Better still, work with assignments
for test data that are completely separate from the data used to
dervive the model.
</p>


<h3>Value</h3>

<p>A list with elements overall (overall accuracy), confusion (confusion
matrix) and prior (prior used for calculation of overall accuracy)
</p>


<h3>Author(s)</h3>

<p>John H Maindonald
</p>


<h3>References</h3>

<p>Maindonald and Braun: 'Data Analysis and Graphics Using R',
3rd edition 2010, Section 12.2.2
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(MASS)
library(DAAG)
cl &lt;- lda(species ~ length+breadth, data=cuckoos, CV=TRUE)$class
confusion(cl, cuckoos$species)

## The function is currently defined as
function (actual, predicted, gpnames = NULL,
            rowcol = c("actual", "predicted"),
            printit = c("overall","confusion"),
            prior = NULL, digits = 3) 
{
  if (is.null(gpnames)) 
    gpnames &lt;- levels(actual)
  if (is.logical(printit)){
    if(printit)printit &lt;- c("overall","confusion")
    else printit &lt;- ""
  }
  tab &lt;- table(actual, predicted)
  acctab &lt;- t(apply(tab, 1, function(x) x/sum(x)))
  dimnames(acctab) &lt;- list(Actual = gpnames, `Predicted (cv)` = gpnames)
  if (is.null(prior)) {
    relnum &lt;- table(actual)
    prior &lt;- relnum/sum(relnum)
    acc &lt;- sum(tab[row(tab) == col(tab)])/sum(tab)
  }
  else {
    acc &lt;- sum(prior * diag(acctab))
  }
  names(prior) &lt;- gpnames
  if ("overall"%in%printit) {
    cat("Overall accuracy =", round(acc, digits), "\n")
    if(is.null(prior)){
      cat("This assumes the following prior frequencies:", 
          "\n")
      print(round(prior, digits))
    }
  }
  if ("confusion"%in%printit) {
    cat("\nConfusion matrix", "\n")
    print(round(acctab, digits))
  }
  invisible(list(overall=acc, confusion=acctab, prior=prior))
}
</code></pre>

<hr>
<h2 id='coralPval'>
P-values from biological expression array data
</h2><span id='topic+coralPval'></span>

<h3>Description</h3>

<p>P-values were calculated for each of 3072 genes, for data that
compared expression values between post-settlement coral larvae
and pre-settlement coral larvae.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("coralPval")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:3072, 1] 8.60e-01 3.35e-08 3.96e-01 2.79e-01 6.36e-01 ...
</p>


<h3>Details</h3>

<p>t-statistics, and hence p-values, were derived from five replicate
two-colour micro-array slides.  Details are in a vignette that
accompanies the <span class="pkg">DAAGbio</span> package.
</p>


<h3>Source</h3>

<p>See the <code>?DAAGbio::coralRG</code>
</p>


<h3>References</h3>

<p>Grasso, L. C.; Maindonald, J.; Rudd, S.; Hayward, D. C.; Saint, R.;
Miller, D. J.; and Ball, E. E., 2008. Microarray analysis identifies
candidate genes for key roles in coral development. BMC Genomics, 9:540.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## From p-values, calculate Benjamini-Hochberg false discrimination rates
fdr &lt;- p.adjust(gamclass::coralPval, method='BH')
## Number of genes identified as differentially expressed for FDR = 0.01
sum(fdr&lt;=0.01)
</code></pre>

<hr>
<h2 id='cvalues'>
Historical speed of light measurements
</h2><span id='topic+cvalues'></span>

<h3>Description</h3>

<p>Measurements made beteween 1675 and 1972
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cvalues
</code></pre>


<h3>Format</h3>

<p>A data frame with 9 observations on the following 3 variables.
</p>

<dl>
<dt><code>Year</code></dt><dd><p>Year of measurement</p>
</dd>
<dt><code>speed</code></dt><dd><p>estimated speed in meters per second</p>
</dd>
<dt><code>error</code></dt><dd><p>measurement error, as estimated by experimenter(s)</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://en.wikipedia.org/wiki/Speed_of_light">https://en.wikipedia.org/wiki/Speed_of_light</a> accessed 2011/12/22
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cvalues)
</code></pre>

<hr>
<h2 id='CVcluster'>
Cross-validation estimate of predictive accuracy for clustered data
</h2><span id='topic+CVcluster'></span>

<h3>Description</h3>

<p>This function adapts cross-validation to work with clustered
categorical outcome data. For example, there may be multiple
observations on individuals (clusters). It requires a
fitting function that accepts a model formula.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CVcluster(formula, id, data, na.action=na.omit, nfold = 15, FUN = MASS::lda,
              predictFUN=function(x, newdata, ...)predict(x, newdata, ...)$class,
              printit = TRUE, cvparts = NULL, seed = 29)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CVcluster_+3A_formula">formula</code></td>
<td>
<p>Model formula
</p>
</td></tr>
<tr><td><code id="CVcluster_+3A_id">id</code></td>
<td>

<p>numeric, identifies clusters
</p>
</td></tr>
<tr><td><code id="CVcluster_+3A_data">data</code></td>
<td>

<p>data frame that supplies the data
</p>
</td></tr>
<tr><td><code id="CVcluster_+3A_na.action">na.action</code></td>
<td>
<p><code>na.fail</code> (default) or <code>na.omit</code>
</p>
</td></tr>
<tr><td><code id="CVcluster_+3A_nfold">nfold</code></td>
<td>

<p>Number of cross-validation folds
</p>
</td></tr>
<tr><td><code id="CVcluster_+3A_fun">FUN</code></td>
<td>

<p><code>function</code> that fits the model
</p>
</td></tr>
<tr><td><code id="CVcluster_+3A_predictfun">predictFUN</code></td>
<td>

<p><code>function</code> that gives predicted values
</p>
</td></tr>
<tr><td><code id="CVcluster_+3A_printit">printit</code></td>
<td>

<p>Should summary information be printed?
</p>
</td></tr>
<tr><td><code id="CVcluster_+3A_cvparts">cvparts</code></td>
<td>

<p>Use, if required, to specify the precise folds used for the
cross-validation.  The comparison between different models
will be more accurate if the same folds are used.
</p>
</td></tr>
<tr><td><code id="CVcluster_+3A_seed">seed</code></td>
<td>

<p>Set seed, if required, so that results are exactly reproducible
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>class</code></td>
<td>
<p>Predicted values from cross-validation</p>
</td></tr>
<tr><td><code>CVaccuracy</code></td>
<td>
<p>Cross-validation estimate of accuracy</p>
</td></tr>
<tr><td><code>confusion</code></td>
<td>
<p>Confusion matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Maindonald
</p>


<h3>References</h3>

<p><a href="https://maths-people.anu.edu.au/~johnm/nzsr/taws.html">https://maths-people.anu.edu.au/~johnm/nzsr/taws.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(requireNamespace('mlbench')&amp;requireNamespace('MASS')){
data('Vowel',package='mlbench')
acc &lt;- CVcluster(formula=Class ~., id = V1, data = Vowel, nfold = 15, FUN = MASS::lda,
              predictFUN=function(x, newdata, ...)predict(x, newdata, ...)$class,
                     printit = TRUE, cvparts = NULL, seed = 29)
}
</code></pre>

<hr>
<h2 id='CVgam'>
Cross-validation estimate of accuracy from GAM model fit
</h2><span id='topic+CVgam'></span>

<h3>Description</h3>

<p>The cross-validation estimate of accuracy is sufficiently independent of
the available model fitting criteria (including Generalized
Cross-validation) that it provides a useful check on the extent of
downward bias in the estimated standard error of residual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CVgam(formula, data, nfold = 10, debug.level = 0, method = "GCV.Cp",
              printit = TRUE, cvparts = NULL, gamma = 1, seed = 29)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CVgam_+3A_formula">formula</code></td>
<td>
<p>Model formula, for passing to the <code>gam()</code> function
</p>
</td></tr>
<tr><td><code id="CVgam_+3A_data">data</code></td>
<td>

<p>data frame that supplies the data
</p>
</td></tr>
<tr><td><code id="CVgam_+3A_nfold">nfold</code></td>
<td>

<p>Number of cross-validation folds
</p>
</td></tr>
<tr><td><code id="CVgam_+3A_debug.level">debug.level</code></td>
<td>

<p>See <code><a href="mgcv.html#topic+gam">gam</a></code> for details
</p>
</td></tr>
<tr><td><code id="CVgam_+3A_method">method</code></td>
<td>

<p>Fit method for GAM model.  See <code><a href="mgcv.html#topic+gam">gam</a></code> for details
</p>
</td></tr>
<tr><td><code id="CVgam_+3A_printit">printit</code></td>
<td>

<p>Should summary information be printed?
</p>
</td></tr>
<tr><td><code id="CVgam_+3A_cvparts">cvparts</code></td>
<td>

<p>Use, if required, to specify the precise folds used for the
cross-validation.  The comparison between different models
will be more accurate if the same folds are used.
</p>
</td></tr>
<tr><td><code id="CVgam_+3A_gamma">gamma</code></td>
<td>

<p>See <code><a href="mgcv.html#topic+gam">gam</a></code> for details.
</p>
</td></tr>
<tr><td><code id="CVgam_+3A_seed">seed</code></td>
<td>

<p>Set seed, if required, so that results are exactly reproducible
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>fitted</code></td>
<td>
<p>fitted values</p>
</td></tr>
<tr><td><code>resid</code></td>
<td>
<p>residuals</p>
</td></tr>
<tr><td><code>cvscale</code></td>
<td>
<p>scale parameter from cross-validation</p>
</td></tr>
<tr><td><code>scale.gam</code></td>
<td>
<p>scale parameter from function <code>gam</code></p>
</td></tr>
</table>
<p>The scale parameter from cross-validation is the error mean square)
</p>


<h3>Author(s)</h3>

<p>John Maindonald
</p>


<h3>References</h3>

<p><a href="https://maths-people.anu.edu.au/~johnm/nzsr/taws.html">https://maths-people.anu.edu.au/~johnm/nzsr/taws.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(sp)){
library(mgcv)
data(meuse)
meuse$ffreq &lt;- factor(meuse$ffreq)
CVgam(formula=log(zinc)~s(elev) + s(dist) + ffreq + soil,
              data = meuse, nfold = 10, debug.level = 0, method = "GCV.Cp",
              printit = TRUE, cvparts = NULL, gamma = 1, seed = 29)
}
</code></pre>

<hr>
<h2 id='eventCounts'>
Tabulate vector of dates by specified time event
</h2><span id='topic+eventCounts'></span>

<h3>Description</h3>

<p>For example, dates may be dates of plane crashes.  For purposes
of analysis, this function tabulates number of crash events
per event of time, for each successive specified event.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eventCounts(data, dateCol="Date", from = NULL, to = NULL,
          by = "1 month", categoryCol=NULL, takeOnly=NULL, prefix="n_")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eventCounts_+3A_data">data</code></td>
<td>

<p>Data frame that should include any columns whose names appear in other
function arguments.
</p>
</td></tr>
<tr><td><code id="eventCounts_+3A_datecol">dateCol</code></td>
<td>

<p>Name of column that holds vector of dates
</p>
</td></tr>
<tr><td><code id="eventCounts_+3A_from">from</code></td>
<td>

<p>Starting date. If <code>NULL</code> set to first date given. If supplied,
any rows earlier than <code>from</code> will be omitted.  Similarly, rows
later than any supplied date <code>to</code> will be omitted.
</p>
</td></tr>
<tr><td><code id="eventCounts_+3A_to">to</code></td>
<td>

<p>Final date, for which numbers of events are to be tallied. If
<code>NULL</code> set to final date given.
</p>
</td></tr>
<tr><td><code id="eventCounts_+3A_by">by</code></td>
<td>

<p>Time event to be used; e.g. &quot;1 day&quot;, or &quot;1 week&quot;, or &quot;4 weeks&quot;,
or &quot;1 month&quot;, or &quot;1 quarter&quot;, or &quot;1 year&quot;, or &quot;10 years&quot;.
</p>
</td></tr>
<tr><td><code id="eventCounts_+3A_categorycol">categoryCol</code></td>
<td>

<p>If not <code>NULL</code> create one column of counts for each level (or if
not a factor, unique value).
</p>
</td></tr>
<tr><td><code id="eventCounts_+3A_takeonly">takeOnly</code></td>
<td>

<p>If not <code>NULL</code>, a charater string that when deparsed and executed
will return a vector of logicals.
</p>
</td></tr>
<tr><td><code id="eventCounts_+3A_prefix">prefix</code></td>
<td>

<p>If <code>categoryCol</code> is not <code>NULL</code>, a prefix for the names of the
columns of counts.  Otherwise (<code>categoryCol=NULL</code>) a name for the
column of counts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame, with columns <code>Date</code> (the first day of the
event for which events are given), and other column(s) that
hols counts of events.
</p>


<h3>Author(s)</h3>

<p>John Maindonald
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+cut">cut</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>crashDate &lt;- as.Date(c("1908-09-17","1912-07-12","1913-08-06",
                       "1913-09-09","1913-10-17"))
df &lt;- data.frame(date=crashDate)
byYears &lt;- eventCounts(data=df, dateCol="date",
                       from=as.Date("1908-01-01"),
                       by="1 year")
</code></pre>

<hr>
<h2 id='FARS'>
US fatal road accident data for automobiles, 1998 to 2010
</h2><span id='topic+FARS'></span>

<h3>Description</h3>

<p>Data are from the US FARS (Fatality Analysis Recording System) archive
that is intended to include every accident in which there was at least
one fatality.  Data are limited to vehicles where the front seat
passenger seat was occupied.  Values are given for selected variables
only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FARS</code></pre>


<h3>Format</h3>

<p>A data frame with 134332 observations on the following 18 variables.
</p>

<dl>
<dt><code>caseid</code></dt><dd><p>a character vector. &ldquo;state:casenum:vnum&rdquo;</p>
</dd>  
<dt><code>state</code></dt><dd><p>a numeric vector. See the FARS website for details</p>
</dd>
<dt><code>age</code></dt><dd><p>a numeric vector; 998=not reported; 999=not known. Cases 
with <code>age</code> &lt; 16 have been omitted</p>
</dd>
<dt><code>airbag</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>injury</code></dt><dd><p>a numeric vector; 4 indicates death.  Blanks, unknown,
and &ldquo;Died prior to accident&rdquo; have been omitted</p>
</dd>
<dt><code>Restraint</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>sex</code></dt><dd><p>1=male, 2=female, 9=unknown</p>
</dd>
<dt><code>inimpact</code></dt><dd><p>a numeric vector; direction of initial impact.
Categories 1 to 12 describe clock positions, so that 1,11, and 12
relate to near frontal impacts; 0 is not a collision; 13: top;
14: undercarriage. 18, introduced in 2005 has been omitted, as have
404 values in additional categories for 2010.  99 denotes a missing
value.</p>
</dd>
<dt><code>modelyr</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>airbagAvail</code></dt><dd><p>a factor with levels <code>no</code> <code>yes</code> <code>NA-code</code></p>
</dd>
<dt><code>airbagDeploy</code></dt><dd><p>a factor with levels <code>no</code> <code>yes</code> <code>NA-code</code></p>
</dd>
<dt><code>D_injury</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>D_airbagAvail</code></dt><dd><p>a factor with levels <code>no</code> <code>yes</code> <code>NA-code</code></p>
</dd>
<dt><code>D_airbagDeploy</code></dt><dd><p>a factor with levels <code>no</code> <code>yes</code> <code>NA-code</code></p>
</dd>
<dt><code>D_Restraint</code></dt><dd><p>a factor with levels <code>no</code> <code>yes</code> <code>NA-code</code></p>
</dd>
<dt><code>year</code></dt><dd><p>year of accident</p>
</dd>
</dl>



<h3>Details</h3>

<p>Data is for automabiles where the right passenger seat was occupied,
with one observation for each such passenger.  Observations for vehicles
where the most harmful event was a fire or explosion or immersion
or gas inhalation, or where someone fell or jumped from the vehicle,
are omitted.  Data are limited to vehicle body types 1 to 19,48,49,61,
or 62.  This excludes large trucks, pickup trucks, vans and buses.
The 2009 and 2010 data does not include information on whether
airbags were installed.
</p>


<h3>Note</h3>

<p>The papers given as references demonstrate the use of Fatal
Accident Recording System data to assess the effectiveness of airbags
(even differences between different types of airbags) and seatbelts.
Useful results can be obtained by matching driver mortality, with and
without airbags, to mortality rates for right front seat passengers
in cars without passenger airbags.
</p>


<h3>Source</h3>

<p><a href="http://www-fars.nhtsa.dot.gov/Main/index.aspx">http://www-fars.nhtsa.dot.gov/Main/index.aspx</a>
</p>


<h3>References</h3>

<p><a href="https://maths-people.anu.edu.au/~johnm/nzsr/taws.html">https://maths-people.anu.edu.au/~johnm/nzsr/taws.html</a>
</p>
<p>Olson CM, Cummings P, Rivara FP. 2006. Association of first- and
second-generation air bags with front occupant death in car crashes: a
matched cohort study. Am J Epidemiol 164:161-169
</p>
<p>Cummings, P; McKnight, B, 2010. Accounting for vehicle, crash, and occupant
characteristics in traffic crash studies. Injury Prevention 16: 363-366
</p>
<p>Braver, ER; Shardell, M; Teoh, ER, 2010. <em>How have changes in air
bag designs affected frontal crash mortality?</em> Ann Epidemiol 20:499-510.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(FARS)
</code></pre>

<hr>
<h2 id='fars2007'>
US Fatal Road Accident Data, 2007 and 2008
</h2><span id='topic+fars2007'></span><span id='topic+fars2008'></span>

<h3>Description</h3>

<p>Data are included on variables that may be relevant to assessing
airbag and seatbelt effectiveness in preventing fatal injury.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fars2007
fars2008
</code></pre>


<h3>Format</h3>

<p>A data frame with 24179 observations on the following 24 variables.
</p>

<dl>
<dt><code>state</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>casenum</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>vnum</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>pnum</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>lightcond</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>numfatal</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>age</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>airbag</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>injury</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ptype</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>restraint</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>seatpos</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>sex</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>body</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>inimpact</code></dt><dd><p>A numeric vector; numbers 1 to 12 give clockface
directions of initial impact.  Values in these datasets are limited 
to 11, 12 and 1;Â i.e., near frontal impact</p>
</dd>
<dt><code>mhevent</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>numoccs</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>travspd</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>modelyr</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Data is for automabiles where a passenger seat was occupied,
with one observation for each such passenger.</p>


<h3>Source</h3>

<p><a href="http://www-fars.nhtsa.dot.gov/Main/index.aspx">http://www-fars.nhtsa.dot.gov/Main/index.aspx</a></p>


<h3>References</h3>

<p><a href="https://maths-people.anu.edu.au/~johnm/nzsr/taws.html">https://maths-people.anu.edu.au/~johnm/nzsr/taws.html</a>
</p>
<p>Olson CM, Cummings P, Rivara FP. 2006. Association of first- and
second-generation air bags with front occupant death in car crashes: a
matched cohort study. Am J Epidemiol 164:161-169
</p>
<p>Cummings, P; McKnight, B, 2010. Accounting for vehicle, crash, and occupant
characteristics in traffic crash studies. Injury Prevention 16: 363-366
</p>
<p>Braver, ER; Shardell, M; Teoh, ER, 2010. <em>How have changes in air
bag designs affected frontal crash mortality?</em> Ann Epidemiol 20:499-510.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(fars2007)
str(fars2007)
</code></pre>

<hr>
<h2 id='frontDeaths'>
Safety Device effectiveness Measures, by Year 
</h2><span id='topic+frontDeaths'></span><span id='topic+sideDeaths'></span><span id='topic+rearDeaths'></span><span id='topic+otherDeaths'></span>

<h3>Description</h3>

<p>Safety devices may be airbags or seatbelts.  For airbags, alternatives
are to use &lsquo;airbag installed&rsquo; or &lsquo;airbag deployed&rsquo; as the
criterion. Ratio of driver deaths to passenger deaths are calculated
for driver with device and for driver without device, in both cases for
passenger without device.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("frontDeaths")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 3
$ airbagAvail : num [1:13, 1:2, 1:4] 1068 1120 1089 1033 940 ...
..- attr(*, &quot;dimnames&quot;)=List of 3
.. ..$ years        : chr [1:13] &quot;1998&quot; &quot;1999&quot; &quot;2000&quot; &quot;2001&quot; ...
.. ..$ D_airbagAvail: chr [1:2] &quot;no&quot; &quot;yes&quot;
.. ..$ injury       : chr [1:4] &quot;P_injury&quot; &quot;D_injury&quot; &quot;tot&quot; &quot;prop&quot;
$ airbagDeploy: num [1:13, 1:2, 1:4] 1133 1226 1196 1151 1091 ...
..- attr(*, &quot;dimnames&quot;)=List of 3
.. ..$ years        : chr [1:13] &quot;1998&quot; &quot;1999&quot; &quot;2000&quot; &quot;2001&quot; ...
.. ..$ D_airbagAvail: chr [1:2] &quot;no&quot; &quot;yes&quot;
.. ..$ injury       : chr [1:4] &quot;P_injury&quot; &quot;D_injury&quot; &quot;tot&quot; &quot;prop&quot;
$ restraint   : num [1:13, 1:2, 1:4] 780 783 735 714 741 645 634 561 558 494 ...
..- attr(*, &quot;dimnames&quot;)=List of 3
.. ..$ years        : chr [1:13] &quot;1998&quot; &quot;1999&quot; &quot;2000&quot; &quot;2001&quot; ...
.. ..$ D_airbagAvail: chr [1:2] &quot;no&quot; &quot;yes&quot;
.. ..$ injury       : chr [1:4] &quot;P_injury&quot; &quot;D_injury&quot; &quot;tot&quot; &quot;prop&quot;
</p>


<h3>Source</h3>

<p>See <code><a href="#topic+FARS">FARS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(frontDeaths)
## maybe str(frontDeaths) ; plot(frontDeaths) ...
</code></pre>

<hr>
<h2 id='gamRF'>
Random forest fit to residuals from GAM model
</h2><span id='topic+gamRF'></span>

<h3>Description</h3>

<p>Fit model using <code>gam()</code> from mgcv, then use random forest
regression with residuals.  Check perfomance of this hybrid model
for predictions to <code>newdata</code>, if supplied.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamRF(formlist, yvar, data, newdata = NULL, rfVars, method = "GCV.Cp", 
    printit = TRUE, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gamRF_+3A_formlist">formlist</code></td>
<td>

<p>List of rght hand sides of formulae for GAM models.
</p>
</td></tr>
<tr><td><code id="gamRF_+3A_yvar">yvar</code></td>
<td>

<p>Character string holding y-variable name.
</p>
</td></tr>
<tr><td><code id="gamRF_+3A_data">data</code></td>
<td>

<p>Data
</p>
</td></tr>
<tr><td><code id="gamRF_+3A_newdata">newdata</code></td>
<td>

<p>Optionally, supply test data.
</p>
</td></tr>
<tr><td><code id="gamRF_+3A_rfvars">rfVars</code></td>
<td>

<p>Names of explanatory variables for the <code>randomForest</code> model.
</p>
</td></tr>
<tr><td><code id="gamRF_+3A_method">method</code></td>
<td>

<p>Smoothing parameter estimation method for use of <code>gam()</code>. See
<code><a href="mgcv.html#topic+gam">gam</a></code>.
</p>
</td></tr>
<tr><td><code id="gamRF_+3A_printit">printit</code></td>
<td>

<p>Should a summary of results (error rates) be printed?
</p>
</td></tr>
<tr><td><code id="gamRF_+3A_seed">seed</code></td>
<td>

<p>Set a seed to make result repeatable.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of test data accuracies for the hybrid models (one for each
element of <code>formlist</code>), plus test error mean square and OOB error
mean square for the use of <code>randomForest()</code>.
</p>


<h3>Note</h3>

<p>The best results are typically obtained when a relatively low
degree of freedom GAM model is used.  It seems advisable to
use those variables for the GAM fit that seem likely to be
similar in their effect irrespective of geographic location.
</p>


<h3>Author(s)</h3>

<p>John Maindonald &lt;john.maindonald@anu.edu.au&gt;
</p>


<h3>References</h3>

<p>J. Li, A. D. Heap, A. Potter and J. J. Daniell. 2011. Application
of Machine Learning Methods to Spatial Interpolation of Environmental
Variables. Environmental Modelling and Software 26: 1647-1656.
DOI: 10.1016/j.envsoft.2011.07.004.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CVgam">CVgam</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(length(find.package("sp", quiet=TRUE))&gt;0){
data("meuse", package="sp")
meuse &lt;- within(meuse, {levels(soil) &lt;- c("1","2","2")
                        ffreq &lt;- as.numeric(ffreq)
                        loglead &lt;- log(lead)}
)
form &lt;- ~ dist + elev + ffreq + soil
rfVars &lt;- c("dist", "elev", "soil", "ffreq", "x", "y")
## Select 90 out of 155 rows
sub &lt;- sample(1:nrow(meuse), 90)
meuseOut &lt;- meuse[-sub,]
meuseIn &lt;- meuse[sub,]
gamRF(formlist=list("lm"=form), yvar="loglead", rfVars=rfVars,
                    data=meuseIn, newdata=meuseOut)
}

## The function is currently defined as
function (formlist, yvar, data, newdata = NULL, rfVars, method = "GCV.Cp", 
    printit = TRUE, seed = NULL) 
{   if(!is.null(seed))set.seed(seed)
    errRate &lt;- numeric(length(formlist)+2)
    names(errRate) &lt;- c(names(formlist), "rfTest", "rfOOB")
    ytrain &lt;- data[, yvar]
    xtrain &lt;- data[, rfVars]
    xtest &lt;- newdata[, rfVars]
    ytest = newdata[, yvar]
    res.rf &lt;- randomForest(x = xtrain, y = ytrain, 
                           xtest=xtest, 
                           ytest=ytest)
    errRate["rfOOB"] &lt;- mean(res.rf$mse)
    errRate["rfTest"] &lt;- mean(res.rf$test$mse)    
    GAMhat &lt;- numeric(nrow(data))
    for(nam in names(formlist)){
      form &lt;- as.formula(paste(c(yvar, paste(formlist[[nam]])), collapse=" "))
      train.gam &lt;- gam(form, data = data, method = method)
      res &lt;- resid(train.gam)
      cvGAMms &lt;- sum(res^2)/length(res)
      if (!all(rfVars %in% names(newdata))) {
        missNam &lt;- rfVars[!(rfVars %in% names(newdata))]
        stop(paste("The following were not found in 'newdata':", 
                   paste(missNam, collapse = ", ")))
      }
      GAMtesthat &lt;- predict(train.gam, newdata = newdata)
      GAMtestres &lt;- ytest - GAMtesthat
      Gres.rf &lt;- randomForest(x = xtrain, y = res, xtest = xtest, 
                              ytest = GAMtestres)
      errRate[nam] &lt;- mean(Gres.rf$test$mse)
    }
    if (printit) 
        print(round(errRate, 4))
    invisible(errRate)
}
</code></pre>

<hr>
<h2 id='german'>
German credit scoring data
</h2><span id='topic+german'></span>

<h3>Description</h3>

<p>See website for details of data attributes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>german
</code></pre>


<h3>Format</h3>

<p>A data frame with 1000 observations on the following 21 variables.
</p>

<dl>
<dt><code>V1</code></dt><dd><p>a factor with levels <code>A11</code> <code>A12</code> <code>A13</code> <code>A14</code></p>
</dd>
<dt><code>V2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V3</code></dt><dd><p>a factor with levels <code>A30</code> <code>A31</code> <code>A32</code> <code>A33</code> <code>A34</code></p>
</dd>
<dt><code>V4</code></dt><dd><p>a factor with levels <code>A40</code> <code>A41</code> <code>A410</code> <code>A42</code> <code>A43</code> <code>A44</code> <code>A45</code> <code>A46</code> <code>A48</code> <code>A49</code></p>
</dd>
<dt><code>V5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V6</code></dt><dd><p>a factor with levels <code>A61</code> <code>A62</code> <code>A63</code> <code>A64</code> <code>A65</code></p>
</dd>
<dt><code>V7</code></dt><dd><p>a factor with levels <code>A71</code> <code>A72</code> <code>A73</code> <code>A74</code> <code>A75</code></p>
</dd>
<dt><code>V8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V9</code></dt><dd><p>a factor with levels <code>A91</code> <code>A92</code> <code>A93</code> <code>A94</code></p>
</dd>
<dt><code>V10</code></dt><dd><p>a factor with levels <code>A101</code> <code>A102</code> <code>A103</code></p>
</dd>
<dt><code>V11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V12</code></dt><dd><p>a factor with levels <code>A121</code> <code>A122</code> <code>A123</code> <code>A124</code></p>
</dd>
<dt><code>V13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V14</code></dt><dd><p>a factor with levels <code>A141</code> <code>A142</code> <code>A143</code></p>
</dd>
<dt><code>V15</code></dt><dd><p>a factor with levels <code>A151</code> <code>A152</code> <code>A153</code></p>
</dd>
<dt><code>V16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V17</code></dt><dd><p>a factor with levels <code>A171</code> <code>A172</code> <code>A173</code> <code>A174</code></p>
</dd>
<dt><code>V18</code></dt><dd><p>a factor with levels <code>good</code> <code>bad</code></p>
</dd>
<dt><code>V19</code></dt><dd><p>a factor with levels <code>A191</code> <code>A192</code></p>
</dd>
<dt><code>V20</code></dt><dd><p>a factor with levels <code>A201</code> <code>A202</code></p>
</dd>
<dt><code>V21</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>700 good and 300 bad credits with 20 predictor variables. Data from 1973 to 1975. Stratified sample from actual credits with bad credits heavily oversampled. A cost matrix can be used.
</p>


<h3>Source</h3>

<p><a href="http://archive.ics.uci.edu/datasets">http://archive.ics.uci.edu/datasets</a>
</p>


<h3>References</h3>

<p>GrÃ¶mping, U. (2019). South German Credit Data: Correcting a Widely 
Used Data Set. Report 4/2019, Reports in Mathematics, Physics and 
Chemistry, Department II, Beuth University of Applied Sciences Berlin.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(german)
</code></pre>

<hr>
<h2 id='greatLakesM'>
Monthly Great Lake heights: 1918 - 2019
</h2><span id='topic+greatLakesM'></span>

<h3>Description</h3>

<p>Heights, in meters, are for the lakes Erie,  Michigan/Huron, 
Ontario and St Clair
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(greatLakesM)</code></pre>


<h3>Format</h3>

<p>The format is:
'data.frame':	1212 obs. of  7 variables:
$ month         : Factor w/ 12 levels &quot;apr&quot;,&quot;aug&quot;,&quot;dec&quot;,..: 5 4 8 1 9 7 6 2 12 11 ...
$ year          : int  1918 1918 1918 1918 1918 1918 1918 1918 1918 1918 ...
$ Superior      : num  183 183 183 183 183 ...
$ Michigan.Huron: num  177 177 177 177 177 ...
$ St..Clair     : num  175 175 175 175 175 ...
$ Erie          : num  174 174 174 174 174 ...
$ Ontario       : num  74.7 74.7 74.9 75.1 75.1 ...
</p>


<h3>Details</h3>

<p>For more details, go to the website that is the source of the data.
</p>


<h3>Source</h3>

<p><a href="https://www.lre.usace.army.mil/Missions/Great-Lakes-Information/Great-Lakes-Information-2/Water-Level-Data/">https://www.lre.usace.army.mil/Missions/Great-Lakes-Information/Great-Lakes-Information-2/Water-Level-Data/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(greatLakesM)
mErie &lt;- ts(greatLakesM[,'Erie'], start=1918, frequency=12)
greatLakes &lt;- aggregate(greatLakesM[,-(1:2)], by=list(greatLakesM$year), 
                        FUN=mean)
names(greatLakes)[1] &lt;- 'year'
## maybe str(greatLakesM)
</code></pre>

<hr>
<h2 id='ldaErr'>Calculate Error Rates for Linear Discriminant Model</h2><span id='topic+ldaErr'></span>

<h3>Description</h3>

<p>Given an lda model object, calculate training set error,
leave-one-out cross-validation error, and test set error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldaErr(train.lda, train, test, group = "type")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ldaErr_+3A_train.lda">train.lda</code></td>
<td>
<p>Fitted lda model object.</p>
</td></tr>
<tr><td><code id="ldaErr_+3A_train">train</code></td>
<td>
<p>Training set data frame.</p>
</td></tr>
<tr><td><code id="ldaErr_+3A_test">test</code></td>
<td>
<p>Test set data frame.</p>
</td></tr>
<tr><td><code id="ldaErr_+3A_group">group</code></td>
<td>
<p>Factor that identifies groups in training data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector that holds leave-one-out, training, and test error
rates
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(spam, package='kernlab')
spam[,-58] &lt;- scale(spam[,-58])
nr &lt;- sample(1:nrow(spam))
spam01 &lt;- spam[nr[1:3601],]     ## Use for training,
spam2 &lt;- spam[nr[3602:4601],]   ## Test
spam01.lda &lt;- lda(type~., data=spam01)
ldaRates &lt;- ldaErr(train.lda=spam01.lda, train=spam01, test=spam2, group="type")

## End(Not run)

</code></pre>

<hr>
<h2 id='loti'>
Global temperature anomalies
</h2><span id='topic+loti'></span>

<h3>Description</h3>

<p>GISS (Goddard Institute for Space Studies) Land-Ocean Temperature 
Index (LOTI) data for the years 1880 to 2019, giving anomalies in 
0.01 degrees Celsius, from the 1951 - 1980 average.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loti
</code></pre>


<h3>Format</h3>

<p>A data frame with 140 observations on the following 19 variables.
</p>

<dl>
<dt><code>Year</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Jan</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Feb</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Mar</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Apr</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>May</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Jun</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Jul</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Aug</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sep</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Oct</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Nov</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Dec</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>JtoD</code></dt><dd><p>Jan-Dec averages</p>
</dd>
<dt><code>D.N</code></dt><dd><p>Dec-Nov averages</p>
</dd>
<dt><code>DJF</code></dt><dd><p>Dec-Jan-Feb averages</p>
</dd>
<dt><code>MAM</code></dt><dd><p>Mar-Apr-May</p>
</dd>
<dt><code>JJA</code></dt><dd><p>Jun-Jul-Aug</p>
</dd>
<dt><code>SON</code></dt><dd><p>Sept-Oct-Nov</p>
</dd>
<dt><code>JtoD2011</code></dt><dd><p>January to December average, from 
data accessed in 2011</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data are the Combined Land-Surface Air and Sea-Surface Water Temperature 
Anomalies (Land-Ocean Temperature Index, LOTI), in 0.01 degrees Celsius, from
<a href="https://data.giss.nasa.gov/gistemp/tabledata_v4/GLB.Ts+dSST.txt">https://data.giss.nasa.gov/gistemp/tabledata_v4/GLB.Ts+dSST.txt</a>
Data in the column <code>JtoD2011</code> was accessed 2011-09-06.
</p>
<p>Also available is a CSV file, with anomalies in degrees Celsius.
</p>


<h3>References</h3>

<p>GISTEMP Team, 2020: GISS Surface Temperature Analysis (GISTEMP), version 4. 
NASA Goddard Institute for Space Studies. Dataset accessed 2020-11-13 at 
<a href="https://data.giss.nasa.gov/gistemp/">https://data.giss.nasa.gov/gistemp/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(loti)
plot(JtoD ~ Year, data=loti)
## Add 11 point moving average
ma11 &lt;- filter(loti$JtoD, rep(1,11)/11, sides=2)
lines(loti$Year, ma11)
</code></pre>

<hr>
<h2 id='plotFars'>
Plot Protection Device Effectiveness Measure Against Year 
</h2><span id='topic+plotFars'></span>

<h3>Description</h3>

<p>Devices may be airbags or seatbelts.  For airbags, alternatives are 
to use &ldquo;airbag installed&rdquo; or &ldquo;airbag deployed&rdquo; as 
the criterion.  The plot shows, for each of the specified features, 
the ratio of driver death rate (or other outcome, e.g., deathÂ or injury)
with feature, to rate without feature, in both cases for passenger 
without feature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotFars(tabDeaths=gamclass::frontDeaths,
          statistics = c("airbagAvail", "airbagDeploy", "restraint"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotFars_+3A_tabdeaths">tabDeaths</code></td>
<td>
<p>List, containing (as a minimum) three-dimensional arrays
with the names specified in the argument <code>statistics</code>, such as is
returned by the function <code><a href="#topic+tabFarsDead">tabFarsDead</a></code></p>
</td></tr>
<tr><td><code id="plotFars_+3A_statistics">statistics</code></td>
<td>

<p>Vector of character: names of the sublists, which contain information on
the deathrates
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The name <code>injury</code> is used, with <code>frontDeaths</code> or <code>sideDeaths</code>
or <code>rearDeaths</code> or <code>otherDeaths</code> as the first argument, to refer to
deaths.  The function <code>tabFarsDeaths</code> allows the option of returning an
object, suitable for using as first argument, that treats <code>injury</code> as
death or serious injury.
</p>


<h3>Value</h3>

<p>A graphics object is returned
</p>


<h3>Note</h3>

<p>Note that the &ldquo;airbag deployed&rdquo; statistic is not a useful measure
of airbag effectiveness.  At its most effective, the airbag will deploy
only when the accident is sufficiently serious that deployment will
reduce the risk of serious injury and/or accident. The with/without
deployment comparison compares, in part, serious accidents with less
serious accidents.</p>


<h3>Author(s)</h3>

<p>John Maindonald
</p>

<hr>
<h2 id='relDeaths'>
Yearly Driver deaths, as Fraction of Deaths for All Years
</h2><span id='topic+relDeaths'></span>

<h3>Description</h3>

<p>The four list elements are for four positions of initial impact.
Each list element is a 13 by 3 <code style="white-space: pre;">&#8288;years&#8288;</code> by &ldquo;safety device&rdquo; matrix
that gives the proportion, for that device in year, of the total over 
<var>years</var>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("relDeaths")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 4
$ front: num [1:13, 1:3] 0.559 0.548 0.544 0.577 0.574 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:13] &quot;1998&quot; &quot;1999&quot; &quot;2000&quot; &quot;2001&quot; ...
.. ..$ : chr [1:3] &quot;airbagAvail&quot; &quot;airbagDeploy&quot; &quot;restraint&quot;
$ side : num [1:13, 1:3] 0.36 0.366 0.367 0.35 0.348 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:13] &quot;1998&quot; &quot;1999&quot; &quot;2000&quot; &quot;2001&quot; ...
.. ..$ : chr [1:3] &quot;airbagAvail&quot; &quot;airbagDeploy&quot; &quot;restraint&quot;
$ rear : num [1:13, 1:3] 0.0507 0.0558 0.0575 0.0498 0.0522 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:13] &quot;1998&quot; &quot;1999&quot; &quot;2000&quot; &quot;2001&quot; ...
.. ..$ : chr [1:3] &quot;airbagAvail&quot; &quot;airbagDeploy&quot; &quot;restraint&quot;
$ other: num [1:13, 1:3] 0.0312 0.0304 0.0313 0.0237 0.0254 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:13] &quot;1998&quot; &quot;1999&quot; &quot;2000&quot; &quot;2001&quot; ...
.. ..$ : chr [1:3] &quot;airbagAvail&quot; &quot;airbagDeploy&quot; &quot;restraint&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(relDeaths)
## maybe str(relDeaths) ; plot(relDeaths) ...
</code></pre>

<hr>
<h2 id='RFcluster'>
Random forests estimate of predictive accuracy for clustered data
</h2><span id='topic+RFcluster'></span>

<h3>Description</h3>

<p>This function adapts random forests to work (albeit clumsily and
inefficiently) with clustered categorical outcome data. For
example, there may be multiple observations on individuals (clusters).
Predictions are made fof the OOB (out of bag) clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RFcluster(formula, id, data, nfold = 15,
              ntree=500, progress=TRUE, printit = TRUE, seed = 29)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RFcluster_+3A_formula">formula</code></td>
<td>
<p>Model formula
</p>
</td></tr>
<tr><td><code id="RFcluster_+3A_id">id</code></td>
<td>

<p>numeric, identifies clusters
</p>
</td></tr>
<tr><td><code id="RFcluster_+3A_data">data</code></td>
<td>

<p>data frame that supplies the data
</p>
</td></tr>
<tr><td><code id="RFcluster_+3A_nfold">nfold</code></td>
<td>
<p>numeric, number of folds
</p>
</td></tr>
<tr><td><code id="RFcluster_+3A_ntree">ntree</code></td>
<td>
<p>numeric, number of trees (number of bootstrap samples)
</p>
</td></tr>
<tr><td><code id="RFcluster_+3A_progress">progress</code></td>
<td>

<p>Print information on progress of calculations
</p>
</td></tr>
<tr><td><code id="RFcluster_+3A_printit">printit</code></td>
<td>

<p>Print summary information on accuracy
</p>
</td></tr>
<tr><td><code id="RFcluster_+3A_seed">seed</code></td>
<td>
<p>Set seed, if required, so that results are exactly reproducible</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Bootstrap samples are taken of observations in the in-bag clusters.
Predictions are made for all observations in the OOB clusters.
</p>


<h3>Value</h3>

<table>
<tr><td><code>class</code></td>
<td>
<p>Predicted values from cross-validation</p>
</td></tr>
<tr><td><code>OOBaccuracy</code></td>
<td>
<p>Cross-validation estimate of accuracy</p>
</td></tr>
<tr><td><code>confusion</code></td>
<td>
<p>Confusion matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Maindonald
</p>


<h3>References</h3>

<p><a href="https://maths-people.anu.edu.au/~johnm/nzsr/taws.html">https://maths-people.anu.edu.au/~johnm/nzsr/taws.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(mlbench)
library(randomForest)
data(Vowel)
RFcluster(formula=Class ~., id = V1, data = Vowel, nfold = 15,
              ntree=500, progress=TRUE, printit = TRUE, seed = 29)

## End(Not run)
</code></pre>

<hr>
<h2 id='rfErr'>Calculate Error Rates for randomForest model</h2><span id='topic+rfErr'></span>

<h3>Description</h3>

<p>Given an randomForest model object, calculate training set error,
out-of-bag (OOB) error, and test set error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rfErr(train.rf, train, test, group = "type")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rfErr_+3A_train.rf">train.rf</code></td>
<td>
<p>Fitted randomForest model object.</p>
</td></tr>
<tr><td><code id="rfErr_+3A_train">train</code></td>
<td>
<p>Training set data frame.</p>
</td></tr>
<tr><td><code id="rfErr_+3A_test">test</code></td>
<td>
<p>Test set data frame.</p>
</td></tr>
<tr><td><code id="rfErr_+3A_group">group</code></td>
<td>
<p>Factor that identifies groups</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector that holds training set error,
out-of-bag (OOB) error, and test set error rates.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(spam, package='kernlab')
spam[,-58] &lt;- scale(spam[,-58])
nr &lt;- sample(1:nrow(spam))
spam01 &lt;- spam[nr[1:3601],]     ## Use for training,
spam2 &lt;- spam[nr[3602:4601],]   ## Test
spam01.rf &lt;- randomForest(type ~ ., data=spam01)
rfRates &lt;- rfErr(train.rf=spam01.rf, train=spam01, test=spam2,
                 group='type')

## End(Not run)

</code></pre>

<hr>
<h2 id='rpartErr'>Calculate Error Rates for rpart model</h2><span id='topic+rpartErr'></span>

<h3>Description</h3>

<p>Given an rpart model object, calculate training set error,
10-fold cross-validation error, and test set error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rpartErr(train.rp, train, test, group = "type")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rpartErr_+3A_train.rp">train.rp</code></td>
<td>
<p>Fitted lda model object.</p>
</td></tr>
<tr><td><code id="rpartErr_+3A_train">train</code></td>
<td>
<p>Training set data frame.</p>
</td></tr>
<tr><td><code id="rpartErr_+3A_test">test</code></td>
<td>
<p>Test set data frame.</p>
</td></tr>
<tr><td><code id="rpartErr_+3A_group">group</code></td>
<td>
<p>Factor that identifies groups</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector that holds training set error,
10-fold cross-validation error, and test set error
rates.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(spam, package='kernlab')
spam[,-58] &lt;- scale(spam[,-58])
nr &lt;- sample(1:nrow(spam))
spam01 &lt;- spam[nr[1:3601],]     ## Use for training,
## if holdout not needed
spam2 &lt;- spam[nr[3602:4601],]   ## Test
spam01.rp &lt;- rpart(type~., data=spam01, cp=0.0001)
rpRates &lt;- rpartErr(train.rp=spam01.rp, train=spam01, test=spam2,
                    group='type')

## End(Not run)

</code></pre>

<hr>
<h2 id='simreg'>
Simulate (repeated) regression calculations
</h2><span id='topic+simreg'></span><span id='topic+bootreg'></span>

<h3>Description</h3>

<p>Derive parameter estimates and standard errors by simulation, or
by bootstrap resampling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simreg(formula, data, nsim = 1000)
bootreg(formula, data, nboot = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simreg_+3A_formula">formula</code></td>
<td>

<p>Model formula
</p>
</td></tr>
<tr><td><code id="simreg_+3A_data">data</code></td>
<td>

<p>Data frame from which names in formula can be taken
</p>
</td></tr>
<tr><td><code id="simreg_+3A_nsim">nsim</code></td>
<td>
<p>Number of repeats of the simulation (<code>simreg</code>)</p>
</td></tr>
<tr><td><code id="simreg_+3A_nboot">nboot</code></td>
<td>
<p>Number of bootstrap resamples (<code>bootreg</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of coefficients from repeated simulations, or from bootstrap
resamples.  For <code>simreg</code> there is one row for each repeat
of the simulation.  For <code>bootreg</code> there is one row for each
resample.
</p>


<h3>Note</h3>

<p>Note that <code>bootreg</code> uses the simplest
possible form of bootstrap.  For any except very large datasets,
standard errors may be substantial under-estimates
</p>


<h3>Author(s)</h3>

<p>John Maindonald
</p>


<h3>References</h3>

<p><a href="https://maths-people.anu.edu.au/~johnm/nzsr/taws.html">https://maths-people.anu.edu.au/~johnm/nzsr/taws.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xy &lt;- data.frame(x=rnorm(100), y=rnorm(100))
simcoef &lt;- simreg(formula = y~x, data = xy, nsim = 100)
bootcoef &lt;- bootreg(formula = y~x, data = xy, nboot = 100)
</code></pre>

<hr>
<h2 id='tabFarsDead'>
Extract ratio of ratios estimate of safety device effectiveness, from
the <code>Fars</code> dataset.
</h2><span id='topic+tabFarsDead'></span>

<h3>Description</h3>

<p>Safety devices may be airbags or seatbelts.  For airbags, alternatives
are to use &lsquo;airbag installed&rsquo; or &lsquo;airbag deployed&rsquo; as the
criterion. Ratio of driver deaths to passenger deaths are calculated
for driver with device and for driver without device, in both cases for
passenger without device, and the ratio of these ratios calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tabFarsDead(dset=gamclass::FARS, fatal = 4, 
            restrict=expression(age&gt;=16&amp;age&lt;998&amp;inimpact%in%c(11,12,1)),
            statistics = c("airbagAvail", "airbagDeploy", "Restraint"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tabFarsDead_+3A_dset">dset</code></td>
<td>
<p>data frame containing data</p>
</td></tr>
<tr><td><code id="tabFarsDead_+3A_fatal">fatal</code></td>
<td>

<p>numeric: 4 for fatal injury, or <code>c(3,4)</code> for incapacitating or
fatal injury
</p>
</td></tr>
<tr><td><code id="tabFarsDead_+3A_statistics">statistics</code></td>
<td>

<p>Vector of character: ratio of rates variables that will be tabulated
</p>
</td></tr>
<tr><td><code id="tabFarsDead_+3A_restrict">restrict</code></td>
<td>
<p>Expression restricting values as specified</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the &lsquo;airbag deployed&rsquo; statistic is not a useful measure
of airbag effectiveness.  At its most effective, the airbag will deploy
only when the accident is sufficiently serious that deployment will
reduce the risk of serious injury and/or accident. The with/without
deployment comparison compares, in part, serious accidents with less
serious accidents.
</p>


<h3>Value</h3>

<p>A list with elements
</p>
<table>
<tr><td><code>airbagAvail</code></td>
<td>
<p>a multiway table with margins <code>yrs</code>,
<code>airbagAvail</code>, and a third margin with levels
<code>P_injury</code>, <code>D_injury</code>, <code>tot</code>, and <code>prop</code></p>
</td></tr>
<tr><td><code>airbagDeploy</code></td>
<td>
<p>a multiway table with margins <code>yrs</code>,
<code>airbagDeploy</code>, and a third margin with levels
<code>P_injury</code>, <code>D_injury</code>, <code>tot</code>, and <code>prop</code></p>
</td></tr>
<tr><td><code>Restraint</code></td>
<td>
<p>a multiway table with margins <code>yrs</code>,
<code>Restraint</code>, and a third margin <code>injury</code> with levels
<code>P_injury</code>, <code>D_injury</code>, <code>tot</code>, and <code>prop</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Maindonald
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tabDeaths &lt;- tabFarsDead()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
