<!DOCTYPE html><html><head><title>Help for package GANPA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GANPA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GANPA-package'><p>Gene Association Network-based Pathway Analysis</p></a></li>
<li><a href='#GSE.Test.Main'><p>Gene-weighted pathway significance analysis</p></a></li>
<li><a href='#weight.gsets.test'><p>Weight genes in pathways with a gene association network</p></a></li>
<li><a href='#weight.gsets.with.msprot'>
<p>Weight genes in pathways with a gene association network after correction for multi-subunit proteins</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Association Network-Based Pathway Analysis (GANPA)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhaoyuan Fang, Weidong Tian and Hongbin Ji</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zhaoyuan Fang &lt;fangzhaoyuan@sibs.ac.cn&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>GANPAdata</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, stats, grDevices, utils</td>
</tr>
<tr>
<td>Description:</td>
<td>A network-based gene weighting algorithm for pathway enrichment analysis, using either RNA-seq or microarray data. Zhaoyuan Fang, Weidong Tian and Hongbin Ji (2012) &lt;<a href="https://doi.org/10.1038%2Fcr.2011.149">doi:10.1038/cr.2011.149</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-11 00:55:20 UTC; fangzy</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-11 04:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='GANPA-package'>Gene Association Network-based Pathway Analysis</h2><span id='topic+GANPA-package'></span><span id='topic+GANPA'></span>

<h3>Description</h3>

<p>This package implements a network-based gene weighting algorithm for pathways, as well as a gene-weighted gene set analysis approach for microarray data pathway analysis.
</p>


<h3>Author(s)</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji
</p>
<p>Maintainer: Zhaoyuan Fang <a href="mailto:fangzhaoyuan@sibs.ac.cn">fangzhaoyuan@sibs.ac.cn</a>
</p>


<h3>References</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji. A Network-Based Gene Weighting Approach for Pathway Analysis. <em>Submitted</em>.
</p>
<p>Aravind Subramanian, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, Scott L. Pomeroy, Todd R. Golub, Eric S. Lander, and Jill P. Mesirov. Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles. <em>PNAS</em> 2005 102(43): 15545-15550. 
</p>

<hr>
<h2 id='GSE.Test.Main'>Gene-weighted pathway significance analysis</h2><span id='topic+GSE.Test.Main'></span>

<h3>Description</h3>

<p>Test the significance of pathways in microarray experiments. This includes a network-based gene weighting algorithm for pathways. Classical and gene-weighted versions of gene set analysis approaches are both used. When required, this function also corrects for gene weighting biases caused by multiple-subunit protein.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSE.Test.Main(gExprs.obj, gsets, gNET, check.exprs = TRUE, msp.groups, 
			  size.min = 15, size.max = 500, permN = 1000, randN = 30, 
			  permFDR.cutoff = 0.5, output.label = "", msp.correction = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSE.Test.Main_+3A_gexprs.obj">gExprs.obj</code></td>
<td>
<p>Gene expression experiment data object.</p>
</td></tr>
<tr><td><code id="GSE.Test.Main_+3A_gsets">gsets</code></td>
<td>
<p>A list of gene sets.</p>
</td></tr>
<tr><td><code id="GSE.Test.Main_+3A_gnet">gNET</code></td>
<td>
<p>A gene association network stored in a list.</p>
</td></tr>
<tr><td><code id="GSE.Test.Main_+3A_check.exprs">check.exprs</code></td>
<td>
<p>Logical (TRUE by default). Check and correct the missing values and scaling in the gExprs.obj. If the scale is natural, it will be converted to log2.</p>
</td></tr>
<tr><td><code id="GSE.Test.Main_+3A_msp.groups">msp.groups</code></td>
<td>
<p>A list of multi-subunit proteins.</p>
</td></tr>
<tr><td><code id="GSE.Test.Main_+3A_size.min">size.min</code></td>
<td>
<p>Minimum size of gene sets used for analysis. By default 15 genes.</p>
</td></tr>
<tr><td><code id="GSE.Test.Main_+3A_size.max">size.max</code></td>
<td>
<p>Maximum size of gene sets used for analysis. By default 500 genes.</p>
</td></tr>
<tr><td><code id="GSE.Test.Main_+3A_permn">permN</code></td>
<td>
<p>Sample permutation times. By default 1000 times.</p>
</td></tr>
<tr><td><code id="GSE.Test.Main_+3A_randn">randN</code></td>
<td>
<p>Gene randomization times. Can be set smaller (say, 30) if you do not care randomization-based significance so as to be faster.</p>
</td></tr>
<tr><td><code id="GSE.Test.Main_+3A_permfdr.cutoff">permFDR.cutoff</code></td>
<td>
<p>Sample permutation FDR cutoff. A number between 0 and 1. Set it larger if wish to see the significance of more gene sets.</p>
</td></tr>
<tr><td><code id="GSE.Test.Main_+3A_output.label">output.label</code></td>
<td>
<p>A label to name output files, e.g. &quot;P53.C2&quot;.</p>
</td></tr>
<tr><td><code id="GSE.Test.Main_+3A_msp.correction">msp.correction</code></td>
<td>
<p>Logical (TRUE). Whether to do a correction for multi-subunit proteins in gene weighting.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It will write analysis results to .csv files.</p>


<h3>Author(s)</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji
</p>


<h3>References</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji. A Network-Based Gene Weighting Approach for Pathway Analysis. <em>Submitted</em>. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not to run
# library(GANPAdata)
# data("gExprs.p53", "gsets.msigdb.pnas", "gNET", "msp.groups", 
#      package="GANPAdata")
# GSE.Test.Main(gExprs.obj=gExprs.p53, gsets=gsets.msigdb.pnas, 
#      gNET=gNET, check.exprs=TRUE, msp.groups=msp.groups, 
#      size.min=15, size.max=500, permN=1000, randN=30, 
#      permFDR.cutoff=0.5, output.label="P53\_C2", msp.correction=TRUE)
</code></pre>

<hr>
<h2 id='weight.gsets.test'>Weight genes in pathways with a gene association network</h2><span id='topic+weight.gsets.test'></span>

<h3>Description</h3>

<p>Given a gene functional association network, the gene weights in a list of pathways are assigned, with genes not present in the network assigned basic weights. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weight.gsets.test(isets, gsets)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="weight.gsets.test_+3A_isets">isets</code></td>
<td>
<p>A gene association network stored in a list.</p>
</td></tr>
<tr><td><code id="weight.gsets.test_+3A_gsets">gsets</code></td>
<td>
<p>Pathways stored in the form of a list of gene sets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of named numeric vectors storing gene weights, with 
gene names in the vector names.
</p>


<h3>Author(s)</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji</p>


<h3>References</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji. 
A Network-Based Gene Weighting Approach for Pathway Analysis. 
<em>Submitted</em>. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+weight.gsets.with.msprot">weight.gsets.with.msprot</a></code>
</p>

<hr>
<h2 id='weight.gsets.with.msprot'>
Weight genes in pathways with a gene association network after correction for multi-subunit proteins
</h2><span id='topic+weight.gsets.with.msprot'></span>

<h3>Description</h3>

<p>Given a gene functional association network and a list of multi-subunit proteins, the gene weights in a list of pathways are assigned with a procedure of correction for multi-subunit proteins. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weight.gsets.with.msprot(gsets, isets.multi, msp.groups)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="weight.gsets.with.msprot_+3A_gsets">gsets</code></td>
<td>
<p>A gene association network stored in a list.</p>
</td></tr>
<tr><td><code id="weight.gsets.with.msprot_+3A_isets.multi">isets.multi</code></td>
<td>
<p>Pathways stored in the form of a list of gene sets.</p>
</td></tr>
<tr><td><code id="weight.gsets.with.msprot_+3A_msp.groups">msp.groups</code></td>
<td>
<p>A list of multi-subunit proteins.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of named numeric vectors storing gene weights, with gene names in the vector names. 
</p>


<h3>Author(s)</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji
</p>


<h3>References</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji. A Network-Based Gene Weighting Approach for Pathway Analysis. <em>Submitted</em>. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+weight.gsets.test">weight.gsets.test</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
