<!DOCTYPE html><html><head><title>Help for package GANPAdata</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GANPAdata}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GANPAdata-package'>
<p>The GANPA Datasets Package</p></a></li>
<li><a href='#gExprs.p53'>
<p>Gene expression data for the P53 dataset</p></a></li>
<li><a href='#gNET'>
<p>gNET: A comprehensive gene functional association network</p></a></li>
<li><a href='#gsets.msigdb.pnas'>
<p>Functional Gene Sets Used in GSEA PNAS Publication</p></a></li>
<li><a href='#msp.groups'>
<p>A List of Human Multi-subunit Proteins</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>The GANPA Datasets Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2011-05-26</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhaoyuan Fang, Weidong Tian and Hongbin Ji</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zhaoyuan Fang &lt;fangzhaoyuan@sibs.ac.cn&gt;</td>
</tr>
<tr>
<td>Suggests:</td>
<td>GANPA</td>
</tr>
<tr>
<td>Description:</td>
<td>This is a dataset package for GANPA, which implements a
        network-based gene weighting approach to pathway analysis. This
        package includes data useful for GANPA, such as a functional
        association network, pathways, an expression dataset and
        multi-subunit proteins.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2012-10-29 08:57:06 UTC; ripley</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2012-10-29 08:57:06</td>
</tr>
</table>
<hr>
<h2 id='GANPAdata-package'>
The GANPA Datasets Package
</h2><span id='topic+GANPAdata-package'></span><span id='topic+GANPAdata'></span>

<h3>Description</h3>

<p>This is a dataset package for GANPA, which implements a network-based gene weighting approach to pathway analysis. This package includes data useful for GANPA, such as a functional association network, pathways, an expression dataset and multi-subunit proteins.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> GANPAdata</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2011-05-26</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji
</p>
<p>Maintainer: Zhaoyuan Fang <a href="mailto:fangzhaoyuan@sibs.ac.cn">fangzhaoyuan@sibs.ac.cn</a>
</p>


<h3>References</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji. A Network-Based Gene Weighting Approach for Pathway Analysis. <em>Submitted</em>.
</p>
<p>Aravind Subramanian, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, Scott L. Pomeroy, Todd R. Golub, Eric S. Lander, and Jill P. Mesirov. Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles. <em>PNAS</em> 2005 102(43): 15545-15550.
</p>

<hr>
<h2 id='gExprs.p53'>
Gene expression data for the P53 dataset
</h2><span id='topic+gExprs.p53'></span>

<h3>Description</h3>

<p>The P53 dataset is consisted of gene expression profiling of 17 P53-wildtype (WT) and 33 P53-mutated (MUT) cancer cell lines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gExprs.p53)</code></pre>


<h3>Format</h3>

<p>A List of 2 matrices. 
</p>
<p>$ gExprs    : num [1:10100, 1:50]
</p>
<p>$ sampleinfo: chr [1:50, 1:2]
</p>


<h3>Source</h3>

<p>http://www.broadinstitute.org/gsea/index.jsp
</p>


<h3>References</h3>

<p>Aravind Subramanian, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, Scott L. Pomeroy, Todd R. Golub, Eric S. Lander, and Jill P. Mesirov. Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles. <em>PNAS</em> 2005 102(43): 15545-15550.
</p>
<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji. A Network-Based Gene Weighting Approach for Pathway Analysis. <em>Submitted</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gExprs.p53)
</code></pre>

<hr>
<h2 id='gNET'>
gNET: A comprehensive gene functional association network
</h2><span id='topic+gNET'></span>

<h3>Description</h3>

<p>gNET was constructed from protein-protein interactions (PPI), co-annotation in GO Biological Process (BP), and co-expression in large scale gene expression microarray data. Several limits have been applied to these data sources to achieve specificity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gNET)</code></pre>


<h3>Format</h3>

<p>A List of 16979 character vectors.</p>


<h3>References</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji. A Network-Based Gene Weighting Approach for Pathway Analysis. <em>Submitted</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gNET)
</code></pre>

<hr>
<h2 id='gsets.msigdb.pnas'>
Functional Gene Sets Used in GSEA PNAS Publication
</h2><span id='topic+gsets.msigdb.pnas'></span>

<h3>Description</h3>

<p>A list of 522 C2 functional gene sets, see reference for details. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gsets.msigdb.pnas)</code></pre>


<h3>Format</h3>

<p>A List of 522 character vectors.</p>


<h3>Source</h3>

<p>http://www.broadinstitute.org/gsea/index.jsp
</p>


<h3>References</h3>

<p>Aravind Subramanian, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, Scott L. Pomeroy, Todd R. Golub, Eric S. Lander, and Jill P. Mesirov. Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles. <em>PNAS</em> 2005 102(43): 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gsets.msigdb.pnas)
</code></pre>

<hr>
<h2 id='msp.groups'>
A List of Human Multi-subunit Proteins
</h2><span id='topic+msp.groups'></span>

<h3>Description</h3>

<p>A list of genes encoding 82 multi-subunit proteins were extracted from human genes. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(msp.groups)</code></pre>


<h3>Format</h3>

<p>A List of 82 character vectors.</p>


<h3>Details</h3>

<p>A multi-subunit protein is named in the format of Multi.XXX where XXX is a member gene picked arbitrarily from the subunit gene group. Note this naming rule is just for convenience. 
</p>


<h3>References</h3>

<p>Zhaoyuan Fang, Weidong Tian and Hongbin Ji. A Network-Based Gene Weighting Approach for Pathway Analysis. <em>Submitted</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(msp.groups)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
