<!DOCTYPE html><html lang="en"><head><title>Help for package GARCOM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GARCOM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gene_annot_counts'><p>gene annotation counts</p></a></li>
<li><a href='#gene_pos_counts'><p>gene position counts</p></a></li>
<li><a href='#genecoord'><p>genecoord</p></a></li>
<li><a href='#recodedgen'><p>recodedgen</p></a></li>
<li><a href='#sample_chr21_vcf'><p>sample_chr21_vcf</p></a></li>
<li><a href='#snpgene'><p>snpgene</p></a></li>
<li><a href='#snppos'><p>snppos</p></a></li>
<li><a href='#vcf_counts_annot'><p>gene annotation counts using VCF data</p></a></li>
<li><a href='#vcf_counts_SNP_genecoords'><p>VCF gene position counts</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene and Region Counting of Mutations ("GARCOM")</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Gene and Region Counting of Mutations (GARCOM) package computes mutation (or alleles) counts per gene per individuals based on gene annotation or genomic base pair boundaries. It comes with features to accept data formats in plink(.raw) and VCF. It provides users flexibility to extract and filter individuals, mutations and genes of interest.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table (&ge; 1.12.8), stats, vcfR (&ge; 1.12.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-12-23 11:37:43 UTC; sanjeevsariya</td>
</tr>
<tr>
<td>Author:</td>
<td>Sanjeev Sariya [aut, cre, cph],
  Giuseppe Tosto [aut, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sanjeev Sariya &lt;sanjeevsariya@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-23 11:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='gene_annot_counts'>gene annotation counts</h2><span id='topic+gene_annot_counts'></span>

<h3>Description</h3>

<p>The function returns a matrix with allelic counts per gene per individual for annotated SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_annot_counts(dt_gen,dt_snpgene,keep_indiv=NULL,
extract_SNP=NULL,filter_gene=NULL,
impute_missing=FALSE,impute_method="mean")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gene_annot_counts_+3A_dt_gen">dt_gen</code></td>
<td>
<p>a dataframe for genetic data that follows PLINK format (.raw)</p>
</td></tr>
<tr><td><code id="gene_annot_counts_+3A_dt_snpgene">dt_snpgene</code></td>
<td>
<p>a dataframe that contains SNP and annotated gene with SNP and GENE as column name</p>
</td></tr>
<tr><td><code id="gene_annot_counts_+3A_keep_indiv">keep_indiv</code></td>
<td>
<p>an option to specify individuals to retain. Mutation counts will be provided for individuals included in the list only. Default is all individuals. Provide list of individuals in a vector.</p>
</td></tr>
<tr><td><code id="gene_annot_counts_+3A_extract_snp">extract_SNP</code></td>
<td>
<p>an option to specify SNPs for which mutation counts are needed. Mutation counts will be provided for SNPs provided in the list only. Default all SNPs are used. Provide list of SNPs in a vector.</p>
</td></tr>
<tr><td><code id="gene_annot_counts_+3A_filter_gene">filter_gene</code></td>
<td>
<p>an option to filter in a list of Genes. Mutation counts will be provided for genes specifed in the list only. Default is all genes. Provide list of genes in a vector.</p>
</td></tr>
<tr><td><code id="gene_annot_counts_+3A_impute_missing">impute_missing</code></td>
<td>
<p>an option to impute missing genotypes. Default is FALSE.</p>
</td></tr>
<tr><td><code id="gene_annot_counts_+3A_impute_method">impute_method</code></td>
<td>
<p>an option to specify imptuation method. Default method is imputation to the mean. Alternatively imputation can be carried out by median. Function accepts method in quotes: &quot;mean&quot; or &quot;median&quot;. Data are rounded to the second decimal places (e.g. 0.1234 will become 0.12).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Inputs needed are recoded genetic data formatted in PLINK format (.raw) and SNP-gene annotation data. The first six columns of the input genetic data follow standard PLINK .raw format. Column names as FID, IID, PAT, MAT, SEX and PHENOTYPE followed by SNP information as recoded by the PLINK software. SNP-gene data has two columns: GENE and SNP names. The function returns allelic counts per gene per sample (where each row represents a gene and each column represents an individual starting with the second column where first column contains gene information).
</p>


<h3>Value</h3>

<p>Returns an object of data.table class as an output with allelic gene counts within each sample where each row corresponds to gene and column to individual IDs from column second. The first column contains gene names.
</p>


<h3>Author(s)</h3>

<p>Sanjeev Sariya
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Package provides sample data that are loaded with package loading. 

data(recodedgen) #PLINK raw formatted data of 10 individiduals with 10 SNPs

data(snpgene) #SNP and its respective GENE annotated. 
#Here 10 SNPs are shown annotated in five genes. 
#A SNP can be annotated in multiple genes. 

gene_annot_counts(recodedgen,snpgene) #run the function

#subset Genes
gene_annot_counts(recodedgen,snpgene,filter_gene=c("GENE1","GENE2"))

#Subset individuals
gene_annot_counts(recodedgen, snpgene,keep_indiv=c("IID_sample1","IID_sample8"))

#subset with genes and samples
gene_annot_counts(recodedgen,snpgene,filter_gene=c("GENE1","GENE2"),
keep_indiv=c("IID_sample1","IID_sample8"))

#impute missing using default method. 

gene_annot_counts(recodedgen,snpgene,impute_missing=TRUE)

#Subset on individuals and impute for missing values. Default as mean
gene_annot_counts(recodedgen,snpgene,impute_missing=TRUE,
keep_indiv=c("IID_sample1","IID_sample2","IID_sample10"))

#impute using median method
gene_annot_counts(recodedgen,snpgene,impute_missing=TRUE,impute_method="median")

#end not RUN

</code></pre>

<hr>
<h2 id='gene_pos_counts'>gene position counts</h2><span id='topic+gene_pos_counts'></span>

<h3>Description</h3>

<p>Function returns matrix with allelic counts per gene per individual for SNP and gene coordinates as inputs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_pos_counts(dt_gen,dt_snp,dt_gene,keep_indiv=NULL,
extract_SNP=NULL,filter_gene=NULL,
impute_missing=FALSE,impute_method="mean")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gene_pos_counts_+3A_dt_gen">dt_gen</code></td>
<td>
<p>a dataframe for genetic data that follows PLINK format (.raw)</p>
</td></tr>
<tr><td><code id="gene_pos_counts_+3A_dt_snp">dt_snp</code></td>
<td>
<p>a dataframe for SNP information with SNP BP as column names.</p>
</td></tr>
<tr><td><code id="gene_pos_counts_+3A_dt_gene">dt_gene</code></td>
<td>
<p>a dataframe for gene boundaries with CHR START END GENE as column names. Where CHR should be integer 1-22. START and END column should be integer. GENE column contains gene names</p>
</td></tr>
<tr><td><code id="gene_pos_counts_+3A_keep_indiv">keep_indiv</code></td>
<td>
<p>an option to specify individuals to retain. Mutation counts will be provided for individuals provided in the list only. Default is all individuals.</p>
</td></tr>
<tr><td><code id="gene_pos_counts_+3A_extract_snp">extract_SNP</code></td>
<td>
<p>an option to specify SNPs for which mutation counts are needed. Mutation counts will be provided for SNPs included in the list only. Default is all SNPs.</p>
</td></tr>
<tr><td><code id="gene_pos_counts_+3A_filter_gene">filter_gene</code></td>
<td>
<p>an option to filter in Genes. Mutation counts will be provided for genes included in the list only. Default is all genes.</p>
</td></tr>
<tr><td><code id="gene_pos_counts_+3A_impute_missing">impute_missing</code></td>
<td>
<p>an option to impute missing genotypes. Default is FALSE.</p>
</td></tr>
<tr><td><code id="gene_pos_counts_+3A_impute_method">impute_method</code></td>
<td>
<p>an option to specify method to specify imptuation method. Default method is impute to the mean. Alternatively imputation can be carried out by median. Function accepts method in quotes: &quot;mean&quot; or &quot;median&quot;. Data are rounded to the second decimal places (e.g. 0.1234 will become 0.12.).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Inputs needed are: recoded genetic data formatted in PLINK format, SNP name with BP (position) and gene name with START and END position. The first six columns of the input genetic data follow standard PLINK .raw format. Column names as FID, IID, PAT, MAT, SEX and PHENOTYPE followed by SNP information as recoded by the PLINK software. The function returns allelic counts per gene per sample (where each row represents a gene and each column represents an individual starting with the second column where first column contains gene information).
</p>


<h3>Value</h3>

<p>Returns an object of data.table class as an output with allelic gene counts within each sample where each row corresponds to gene and column to individual IDs from column second. The first column contains gene names.
</p>


<h3>Author(s)</h3>

<p>Sanjeev Sariya
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Package provides sample data that are loaded with package loading. 
#not RUN
data(recodedgen) #PLINK raw formatted data of 10 individiduals with 10 SNPs

data(genecoord) #gene coordinates with START, END, CHR and GENE names. 
#Five genes with start and end genomic coordinates

data(snppos) #SNP and BP column names with SNP names and SNP genomic location in BP. 
#10 SNPs with genomic location

gene_pos_counts(recodedgen, snppos, genecoord) #run the function

#subset individuals
gene_pos_counts(recodedgen, snppos, genecoord,keep_indiv=c("IID_sample2","IID_sample4"))

#subset genes
gene_pos_counts(recodedgen,snppos,genecoord,filter_gene=c("GENE1","GENE2")) 

#subset genes and individual iids
gene_pos_counts(recodedgen,snppos,genecoord,filter_gene=c("GENE1","GENE2"),
keep_indiv=c("IID_sample10","IID_sample4")) 

##impute by mean
gene_pos_counts(recodedgen,snppos,genecoord,impute_missing=TRUE,impute_method="mean")

#end not RUN

</code></pre>

<hr>
<h2 id='genecoord'>genecoord</h2><span id='topic+genecoord'></span>

<h3>Description</h3>

<p>sample data for gene base pair boundaries
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genecoord
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 5 rows and 3 columns.
</p>


<h3>Details</h3>

<p>genecoord is example data provided with the GARCOM package. It has 3 columns and 5 rows.
Column names are GENE, START and END where GENE column contains gene name, START and END indicate
start BP and end BP respectively.
</p>

<hr>
<h2 id='recodedgen'>recodedgen</h2><span id='topic+recodedgen'></span>

<h3>Description</h3>

<p>sample genetic data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recodedgen
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 10 rows and 16 columns.
</p>


<h3>Details</h3>

<p>recodedgen is sample genetic data provided with the GARCOM package. It has with 10 rows and 16 columns.
Where the first 6 columns are FID, IID, PAT, MAT, SEX and PHENOTYPE which are inherent from the
PLINK (recode) output. Columns followed by PHENOTYPE are SNP names which are suffixed with _A, or _C
or _T or _G. Each SNP column may have 0, 1, 2 or NA value. Where NA represents missingness.
</p>

<hr>
<h2 id='sample_chr21_vcf'>sample_chr21_vcf</h2><span id='topic+sample_chr21_vcf'></span>

<h3>Description</h3>

<p>VCF data hg19 build 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_chr21_vcf
</code></pre>


<h3>Format</h3>

<p>VCF sample
</p>


<h3>Details</h3>

<p>sample_chr21_vcf is sample VCF file provided for CHR21 for of ten individuals 
contains 128 SNPs
</p>

<hr>
<h2 id='snpgene'>snpgene</h2><span id='topic+snpgene'></span>

<h3>Description</h3>

<p>sample SNP-gene annotation data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpgene
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 10 rows and 2 columns.
</p>


<h3>Details</h3>

<p>snpgene is sample SNP-Gene annotation data provided with the GARCOM package. It has 10 rows and 2 columns.
Column names are GENE and SNP, where GENE column contains GENE names and SNP column
contains SNP name that is annotated with the GENE
</p>

<hr>
<h2 id='snppos'>snppos</h2><span id='topic+snppos'></span>

<h3>Description</h3>

<p>sample data for SNP coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snppos
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 10 rows and 2 columns.
</p>


<h3>Details</h3>

<p>snppos is sample SNP-BP data provided with the GARCOM package. It has 2 columns and 10 rows. Column names are SNP and BP.
where SNP columns contains SNP names BP column contains position of the SNP
</p>

<hr>
<h2 id='vcf_counts_annot'>gene annotation counts using VCF data</h2><span id='topic+vcf_counts_annot'></span>

<h3>Description</h3>

<p>Function returns a matrix with allelic (reference) counts per gene per individual for SNP-gene annotation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf_counts_annot(
  vcf_data,
  df_snpgene,
  keep_indiv = NULL,
  extract_SNP = NULL,
  filter_gene = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vcf_counts_annot_+3A_vcf_data">vcf_data</code></td>
<td>
<p>an object of vcfR class</p>
</td></tr>
<tr><td><code id="vcf_counts_annot_+3A_df_snpgene">df_snpgene</code></td>
<td>
<p>a data frame that contains SNP and annotated gene with SNP and GENE as column name</p>
</td></tr>
<tr><td><code id="vcf_counts_annot_+3A_keep_indiv">keep_indiv</code></td>
<td>
<p>an option to specify individuals to retain. Mutation counts will be provided for individuals included in the list only. Default is all individuals. Provide list of individuals in a vector.</p>
</td></tr>
<tr><td><code id="vcf_counts_annot_+3A_extract_snp">extract_SNP</code></td>
<td>
<p>an option to specify SNPs for which mutation counts are needed. Mutation counts will be provided for SNPs included in the list only. Default is all SNPs.</p>
</td></tr>
<tr><td><code id="vcf_counts_annot_+3A_filter_gene">filter_gene</code></td>
<td>
<p>an option to filter in a list of Genes. Mutation counts will be provided for genes specifed in the list only. Default is all genes. Provide list of genes in a vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Inputs needed are a vcf data and a data frame of SNP-gene annotation. The function returns a matrix of allelic counts (reference) per gene per sample (where each row represents a gene and each column represents an individual starting with the second column where first column contains gene information).
</p>


<h3>Value</h3>

<p>Returns an matrix of data.table class as an output with allelic (reference) gene counts within each sample where each row corresponds to gene and column to individual IDs from column second. The first column contains gene names.
</p>


<h3>Author(s)</h3>

<p>Sanjeev Sariya
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
vcf_counts_annot(vcf,df_snpgene_test)

## End(Not run)

</code></pre>

<hr>
<h2 id='vcf_counts_SNP_genecoords'>VCF gene position counts</h2><span id='topic+vcf_counts_SNP_genecoords'></span>

<h3>Description</h3>

<p>Function returns a matrix with allelic counts per gene per individual for SNP and gene coordinates as inputs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf_counts_SNP_genecoords(
  vcf_data,
  df_snppos,
  df_genecoords,
  keep_indiv = NULL,
  extract_SNP = NULL,
  filter_gene = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vcf_counts_SNP_genecoords_+3A_vcf_data">vcf_data</code></td>
<td>
<p>an object of vcfR class</p>
</td></tr>
<tr><td><code id="vcf_counts_SNP_genecoords_+3A_df_snppos">df_snppos</code></td>
<td>
<p>a dataframe for SNP information with SNP BP as column names.</p>
</td></tr>
<tr><td><code id="vcf_counts_SNP_genecoords_+3A_df_genecoords">df_genecoords</code></td>
<td>
<p>a dataframe for gene boundaries with CHR START END GENE as column names. Where CHR should be integer 1-22. START and END column should be integer. GENE column contains gene names</p>
</td></tr>
<tr><td><code id="vcf_counts_SNP_genecoords_+3A_keep_indiv">keep_indiv</code></td>
<td>
<p>an option to specify individuals to retain. Mutation counts will be provided for individuals included in the list only. Default is all individuals. Provide list of individuals in a vector.</p>
</td></tr>
<tr><td><code id="vcf_counts_SNP_genecoords_+3A_extract_snp">extract_SNP</code></td>
<td>
<p>an option to specify SNPs for which mutation counts are needed. Mutation counts will be provided for SNPs included in the list only. Default is all SNPs.</p>
</td></tr>
<tr><td><code id="vcf_counts_SNP_genecoords_+3A_filter_gene">filter_gene</code></td>
<td>
<p>an option to filter in Genes. Mutation counts will be provided for genes included in the list only. Default is all genes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an matrix of data.table class as an output with allelic (reference) gene counts within each sample where each row corresponds to gene and column to individual IDs from column second. The first column contains gene names.
</p>


<h3>Author(s)</h3>

<p>Sanjeev Sariya
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
vcf_counts_SNP_genecoords(vcf_data_test,df_snppos_test,df_genecoords_test)

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
