<!DOCTYPE html><html><head><title>Help for package gausscov</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gausscov}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#abcq'><p>American Business Cycle</p></a></li>
<li><a href='#boston'><p>Boston data</p></a></li>
<li><a href='#decode'><p>Decodes the number of a subset selected by fasb.R to give the  covariates</p></a></li>
<li><a href='#f1st'><p>Stepwise selection of covariates</p></a></li>
<li><a href='#f2st'><p>Repeated stepwise selection of covariates</p></a></li>
<li><a href='#f3st'><p>Stepwise selection of covariates</p></a></li>
<li><a href='#f3sti'><p>Selection of covariates with given excluded covariates</p></a></li>
<li><a href='#fasb'><p>Calculates all subsets  where each included covariate is significant.</p></a></li>
<li><a href='#fgeninter'><p>Generation of interactions</p></a></li>
<li><a href='#fgentrig'><p>Generation of sine and cosine functions</p></a></li>
<li><a href='#fgr1st'><p>Calculates a dependence graph using Gaussian stepwise selection</p></a></li>
<li><a href='#flag'><p>Calculation of lagged covariates</p></a></li>
<li><a href='#fpval'><p> Calculates the regression coefficients, the P-values and the standard P-values for the chosen subset ind</p></a></li>
<li><a href='#fundr'><p>Converts directed into an undirected graph</p></a></li>
<li><a href='#leukeia'><p>Leukemia data set</p></a></li>
<li><a href='#mel-temp'><p>Melbourne minimum temperature</p></a></li>
<li><a href='#redwine'><p>Redwine data</p></a></li>
<li><a href='#simgpval'><p>Simulates Gaussian P-values</p></a></li>
<li><a href='#snspt'><p>Sunspot data</p></a></li>
<li><a href='#vardata'><p>USA economics data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-3-8</td>
</tr>
<tr>
<td>Title:</td>
<td>The Gaussian Covariate Method for Variable Selection</td>
</tr>
<tr>
<td>Author:</td>
<td>Laurie Davies [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Laurie Davies &lt;pldavies44@cantab.net&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The standard linear regression theory whether frequentist or Bayesian is based on an 'assumed (revealed?) truth' (John Tukey) attitude to models.  This is reflected in the language of statistical inference which involves a concept of truth, for example confidence intervals, hypothesis testing  and consistency. The motivation behind this package was to remove the word true from the theory and practice of linear regression  and to replace it by approximation. The approximations considered are the least squares approximations. An approximation is called valid if it contains no irrelevant covariates. This is operationalized using the concept of a Gaussian P-value which is the probability that pure Gaussian noise is better in term of least squares than the covariate. The precise definition given in the paper, it  is intuitive and requires only four simple equations. Its overwhelming advantage compared with a standard F P-value is that is is exact and valid whatever the data. In contrast F P-values are only valid for specially designed simulations. Given this a valid approximation is one where all the Gaussian P-values are less than a threshold p0 specified by the statistician, in this package with the default value 0.01. This approximations approach is not only much simpler it is overwhelmingly better than the standard model based approach. The  will be demonstrated using six real data sets, four from high dimensional regression and two from vector autoregression. The simplicity and superiority of Gaussian P-values derive from their universal exactness and validity. This is in complete contrast to standard F P-values which are valid only for carefully designed simulations. The function f1st is the most important function. It is a greedy forward selection procedure which results in either just one or no approximations which may however not be valid. If the size is less than than a threshold with default value 21 then an all subset procedure is called which returns  the best valid subset. A good default start is  f1st(y,x,kmn=15) The best function for returning multiple  approximations is  f3st which repeatedly calls f1st.  For more information see the web site below and the accompanying papers: L. Davies and L. Duembgen, "Covariate Selection Based on a Model-free Approach to Linear Regression with Exact Probabilities", 2022, &lt;<a href="https://doi.org/10.48550/arXiv.2202.01553">doi:10.48550/arXiv.2202.01553</a>&gt;. L. Davies, "An Approximation Based Theory of Linear Regression", 2024, &lt;<a href="https://doi.org/10.48550/arXiv.2402.09858">doi:10.48550/arXiv.2402.09858</a>&gt;.</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), stats</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-19 16:01:39 UTC; laurie</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-19 16:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='abcq'>American Business Cycle</h2><span id='topic+abcq'></span>

<h3>Description</h3>

<p>The 22 variables are quarterly data from  1919-1941 and 1947-1983 of the variables GNP72, CPRATE, CORPYIELD, M1, M2, BASE, CSTOCK, WRICE67, PRODUR72, NONRES72, IRES72, DBUSI72, CDUR72, CNDUR72, XPT72, MPT72, GOVPUR72,NCSPDE72, NCSBS72, NCSCON72, CCSPDE72 and CCSBS72. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>abcq</code></pre>


<h3>Format</h3>

<p>A matrix of size  240 x 22</p>


<h3>Source</h3>

<p>http://data.nber.org/data/abc/</p>

<hr>
<h2 id='boston'>Boston data</h2><span id='topic+boston'></span>

<h3>Description</h3>

<p>This data set is part of the MASS package.  The 14 columns are:
</p>
<p>crim per capita crime rate by town
</p>
<p>zn proportion of residential land zoned for lots over 25.000 sq.ft.
</p>
<p>indus proportion of non-residential business acres per town
</p>
<p>chas Charles River dummy variable (=1 if tract bounds rive; 0
otherwise)
</p>
<p>nox nitrogen oxides concentration (parts per 10 million)
</p>
<p>rm average number of rooms per dwelling
</p>
<p>age proportion of owner-occupied units built prior to 1940
</p>
<p>dis weighted mean of distances to five Boston employment centres
</p>
<p>rad index of accessibility to radial highways
</p>
<p>tax full-value property-tax rate per $10,000
</p>
<p>ptration pupil-teacher ration by town
</p>
<p>black 100(Bk-0.63)^2 where Bk is the proportion of blacks by town
</p>
<p>lstat lower status of the population (percent)
</p>
<p>medv median value of owner occupies homes in $1000s.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boston</code></pre>


<h3>Format</h3>

<p>A 506 x 14 matrix.</p>


<h3>Source</h3>

<p>R package MASS
https://cran.r-project.org/web/packages/available_packages_by_name.html</p>


<h3>References</h3>

<p>MASS Support Functions and Datasets for Venables and Ripley's MASS
</p>

<hr>
<h2 id='decode'>Decodes the number of a subset selected by fasb.R to give the  covariates</h2><span id='topic+decode'></span>

<h3>Description</h3>

<p>Decodes the number of a subset selected by fasb.R to give the  covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decode(ns, k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="decode_+3A_ns">ns</code></td>
<td>
<p>The number of the subset</p>
</td></tr>
<tr><td><code id="decode_+3A_k">k</code></td>
<td>
<p>The number of covariates</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ind The list of covariates
</p>
<p>set A binary vector giving the covariates
</p>


<h3>Examples</h3>

<pre><code class='language-R'>a&lt;- decode(19,8)
</code></pre>

<hr>
<h2 id='f1st'>Stepwise selection of covariates</h2><span id='topic+f1st'></span>

<h3>Description</h3>

<p>Stepwise selection of covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f1st(y,x,p0=0.01,kmn=0,kmx=0,kex=0,mx=21,sub=T,inr=T,xinr=F,qq=-1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="f1st_+3A_y">y</code></td>
<td>
<p>Dependent variable</p>
</td></tr>
<tr><td><code id="f1st_+3A_x">x</code></td>
<td>
<p>Covariates</p>
</td></tr>
<tr><td><code id="f1st_+3A_p0">p0</code></td>
<td>
<p>The P-value cut-off</p>
</td></tr>
<tr><td><code id="f1st_+3A_kmn">kmn</code></td>
<td>
<p>The minimum number of included covariates irrespective of cut-off P-value</p>
</td></tr>
<tr><td><code id="f1st_+3A_kmx">kmx</code></td>
<td>
<p>The maximum number of included covariates irrespective of cut-off P-value.</p>
</td></tr>
<tr><td><code id="f1st_+3A_kex">kex</code></td>
<td>
<p>The excluded covariates</p>
</td></tr>
<tr><td><code id="f1st_+3A_mx">mx</code></td>
<td>
<p>The maximum number covariates for an all subset search</p>
</td></tr>
<tr><td><code id="f1st_+3A_sub">sub</code></td>
<td>
<p>Logical if TRUE best subset selected</p>
</td></tr>
<tr><td><code id="f1st_+3A_inr">inr</code></td>
<td>
<p>Logical if TRUE include intercept if not present</p>
</td></tr>
<tr><td><code id="f1st_+3A_xinr">xinr</code></td>
<td>
<p>Logical if TRUE intercept already present</p>
</td></tr>
<tr><td><code id="f1st_+3A_qq">qq</code></td>
<td>
<p>The number of covariates to choose from. If qq=-1 the number of covariates of x is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pv In order the included covariates, the regression coefficient values, the Gaussian P-values, the standard P-values.
</p>
<p>res The residuals
</p>
<p>stpv The covariates in order of selection and Gaussian P-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(boston)
bostint&lt;-fgeninter(boston[,1:13],2)[[1]]
a&lt;-f1st(boston[,14],bostint,kmn=10,sub=TRUE)
</code></pre>

<hr>
<h2 id='f2st'>Repeated stepwise selection of covariates</h2><span id='topic+f2st'></span>

<h3>Description</h3>

<p>Repeated stepwise selection of covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f2st(y,x,p0=0.01,kmn=0,kmx=0,kex=0,mx=21,lm=9^9,sub=T,inr=T,xinr=F,qq=-1)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="f2st_+3A_y">y</code></td>
<td>
<p>Dependent variable</p>
</td></tr>
<tr><td><code id="f2st_+3A_x">x</code></td>
<td>
<p>Covariates</p>
</td></tr>
<tr><td><code id="f2st_+3A_p0">p0</code></td>
<td>
<p>The P-value cut-off</p>
</td></tr>
<tr><td><code id="f2st_+3A_kmn">kmn</code></td>
<td>
<p>The minimum number of included covariates irrespective of cut-off P-value</p>
</td></tr>
<tr><td><code id="f2st_+3A_kmx">kmx</code></td>
<td>
<p>The maximum number of included covariates irrespective of cut-off P-value.</p>
</td></tr>
<tr><td><code id="f2st_+3A_kex">kex</code></td>
<td>
<p>The excluded covariates</p>
</td></tr>
<tr><td><code id="f2st_+3A_mx">mx</code></td>
<td>
<p>The maximum number of covariates for an all subset search</p>
</td></tr>
<tr><td><code id="f2st_+3A_lm">lm</code></td>
<td>
<p>The maximum number of linear approximations</p>
</td></tr>
<tr><td><code id="f2st_+3A_sub">sub</code></td>
<td>
<p>Logical if TRUE select the best subset</p>
</td></tr>
<tr><td><code id="f2st_+3A_inr">inr</code></td>
<td>
<p>Logical if TRUE include an intercept</p>
</td></tr>
<tr><td><code id="f2st_+3A_xinr">xinr</code></td>
<td>
<p>Logical if TRUE intercept already included</p>
</td></tr>
<tr><td><code id="f2st_+3A_qq">qq</code></td>
<td>
<p>The number of covariates to choose from. If qq=-1 the number of covariates of x is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pv   In order the linear approximation, the included covariates, the Gaussian P-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(boston)
bostint&lt;-fgeninter(boston[,1:13],2)[[1]]
a&lt;-f2st(boston[,14],bostint,lm=3,sub=FALSE)
</code></pre>

<hr>
<h2 id='f3st'>Stepwise selection of covariates</h2><span id='topic+f3st'></span>

<h3>Description</h3>

<p>Stepwise selection of covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f3st(y,x,m,p0=0.01,kmn=0,kmx=0,kex=0,mx=21,sub=T,inr=T,xinr=F,qq=-1,kexmx=100)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="f3st_+3A_y">y</code></td>
<td>
<p>Dependent variable</p>
</td></tr>
<tr><td><code id="f3st_+3A_x">x</code></td>
<td>
<p>Covariates</p>
</td></tr>
<tr><td><code id="f3st_+3A_m">m</code></td>
<td>
<p>The number of iterations</p>
</td></tr>
<tr><td><code id="f3st_+3A_p0">p0</code></td>
<td>
<p>The P-value cut-off</p>
</td></tr>
<tr><td><code id="f3st_+3A_kmn">kmn</code></td>
<td>
<p>The minimum number of included covariates irrespective of cut-off P-value</p>
</td></tr>
<tr><td><code id="f3st_+3A_kmx">kmx</code></td>
<td>
<p>The maximum number of included covariates irrespective of cut-off P-value.</p>
</td></tr>
<tr><td><code id="f3st_+3A_kex">kex</code></td>
<td>
<p>The excluded covariates</p>
</td></tr>
<tr><td><code id="f3st_+3A_mx">mx</code></td>
<td>
<p>The maximum number covariates for an all subset search</p>
</td></tr>
<tr><td><code id="f3st_+3A_sub">sub</code></td>
<td>
<p>Logical if TRUE best subset selected</p>
</td></tr>
<tr><td><code id="f3st_+3A_inr">inr</code></td>
<td>
<p>Logical if TRUE include intercept if not present</p>
</td></tr>
<tr><td><code id="f3st_+3A_xinr">xinr</code></td>
<td>
<p>Logical if TRUE intercept already present</p>
</td></tr>
<tr><td><code id="f3st_+3A_qq">qq</code></td>
<td>
<p>The number of covariates to choose from. If qq=-1 the number of covariates of x is used.</p>
</td></tr>
<tr><td><code id="f3st_+3A_kexmx">kexmx</code></td>
<td>
<p>The maximum number of covariates in an approximation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>covch The sum of squared residuals and the selected covariates ordered in increasing size of sum of squared residuals.
</p>
<p>lai  The number of rows of covch
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukemia)
a&lt;-f3st(leukemia[[1]],leukemia[[2]],m=2,kmn=5,sub=TRUE,kexmx=10)
</code></pre>

<hr>
<h2 id='f3sti'>Selection of covariates with given excluded covariates</h2><span id='topic+f3sti'></span>

<h3>Description</h3>

<p>Selection of covariates with given excluded covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f3sti(y,x,covch,ind,m,p0=0.01,kmn=0,kmx=0, kex=0,mx=21,sub=T,inr=F,xinr=F,qq=-1,kexmx=100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="f3sti_+3A_y">y</code></td>
<td>
<p>Dependent variable</p>
</td></tr>
<tr><td><code id="f3sti_+3A_x">x</code></td>
<td>
<p>Covariates</p>
</td></tr>
<tr><td><code id="f3sti_+3A_covch">covch</code></td>
<td>
<p>Sum of squared residuals and selected covariates</p>
</td></tr>
<tr><td><code id="f3sti_+3A_ind">ind</code></td>
<td>
<p>The excluded covariates</p>
</td></tr>
<tr><td><code id="f3sti_+3A_m">m</code></td>
<td>
<p>Number of iterations</p>
</td></tr>
<tr><td><code id="f3sti_+3A_p0">p0</code></td>
<td>
<p>The P-value cut-off</p>
</td></tr>
<tr><td><code id="f3sti_+3A_kmn">kmn</code></td>
<td>
<p>The minimum number of included covariates irrespective of cut-off P-value</p>
</td></tr>
<tr><td><code id="f3sti_+3A_kmx">kmx</code></td>
<td>
<p>The maximum number of included covariates irrespective of cut-off P-value.</p>
</td></tr>
<tr><td><code id="f3sti_+3A_kex">kex</code></td>
<td>
<p>The excluded covariates</p>
</td></tr>
<tr><td><code id="f3sti_+3A_mx">mx</code></td>
<td>
<p>The maximum number covariates for an all subset search</p>
</td></tr>
<tr><td><code id="f3sti_+3A_sub">sub</code></td>
<td>
<p>Logical if TRUE best subset selected</p>
</td></tr>
<tr><td><code id="f3sti_+3A_inr">inr</code></td>
<td>
<p>Logical if TRUE include intercept if not present</p>
</td></tr>
<tr><td><code id="f3sti_+3A_xinr">xinr</code></td>
<td>
<p>Logical if TRUE intercept already present</p>
</td></tr>
<tr><td><code id="f3sti_+3A_qq">qq</code></td>
<td>
<p>The number of covariates to choose from. If qq=-1 the number of covariates of x is used.</p>
</td></tr>
<tr><td><code id="f3sti_+3A_kexmx">kexmx</code></td>
<td>
<p>The maximum number of covariates in an approximation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ind1 The excluded covariates
</p>
<p>covch The sum of squared residuals and the selected covariates ordered in increasing size of sum of squared residuals
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukemia)
covch=c(2.023725,1182,1219,2888,0)
covch&lt;-matrix(covch,nrow=1,ncol=5)
ind&lt;-c(1182,1219,2888)
ind&lt;-matrix(ind,nrow=3,ncol=1)
m&lt;-1
a&lt;-f3sti(leukemia[[1]],leukemia[[2]],covch,ind,m,kexmx=5)
</code></pre>

<hr>
<h2 id='fasb'>Calculates all subsets  where each included covariate is significant.</h2><span id='topic+fasb'></span>

<h3>Description</h3>

<p>The subset are ordered according to the sum of squared residuals. Subsets can be decoded with decode.R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fasb(y,x,p0=0.01,ind=0,inr=T,xinr=F,qq=-1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fasb_+3A_y">y</code></td>
<td>
<p>The dependent variable</p>
</td></tr>
<tr><td><code id="fasb_+3A_x">x</code></td>
<td>
<p>The covariates</p>
</td></tr>
<tr><td><code id="fasb_+3A_p0">p0</code></td>
<td>
<p>Cut-off p-value for significance</p>
</td></tr>
<tr><td><code id="fasb_+3A_ind">ind</code></td>
<td>
<p>The indices of a subset of covariates for which all subsets are to be considered</p>
</td></tr>
<tr><td><code id="fasb_+3A_inr">inr</code></td>
<td>
<p>If TRUE to include intercept</p>
</td></tr>
<tr><td><code id="fasb_+3A_xinr">xinr</code></td>
<td>
<p>If TRUE intercept already included</p>
</td></tr>
<tr><td><code id="fasb_+3A_qq">qq</code></td>
<td>
<p>The number of covariates from which to choose. Equals number of covariates minus length of ind if qq=-1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nv Coded List of subsets with number of covariates and sum of squared residuals
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(redwine)
nvv&lt;-fasb(redwine[,12],redwine[,1:11])
</code></pre>

<hr>
<h2 id='fgeninter'>Generation of interactions</h2><span id='topic+fgeninter'></span>

<h3>Description</h3>

<p>Generates all interactions of degree at most ord 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fgeninter(x,ord)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fgeninter_+3A_x">x</code></td>
<td>
<p>Covariates</p>
</td></tr>
<tr><td><code id="fgeninter_+3A_ord">ord</code></td>
<td>
<p>Order of interactions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>xx  All interactions of order at most ord.
</p>
<p>intx Decomposes a given interaction covariate of xx
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(boston)
bostint&lt;-fgeninter(boston[,1:13],2)[[1]]
</code></pre>

<hr>
<h2 id='fgentrig'>Generation of sine and cosine functions</h2><span id='topic+fgentrig'></span>

<h3>Description</h3>

<p>Generates sin(pi*j*(1:n)/n) (odd) and cos(pi*j*(1:n)/n) (even) for j=1,...,m for a given sample size n.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fgentrig(n,m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fgentrig_+3A_n">n</code></td>
<td>
<p>Sample size</p>
</td></tr>
<tr><td><code id="fgentrig_+3A_m">m</code></td>
<td>
<p>Maximum order of sine and cosine functions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>x  The functions sin(pi*j*(1:n)/n) (odd) and cos(pi*j*(1:n)/n) (even) for j=1,...,m.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trig&lt;-fgentrig(36,36)
</code></pre>

<hr>
<h2 id='fgr1st'>Calculates a dependence graph using Gaussian stepwise selection</h2><span id='topic+fgr1st'></span>

<h3>Description</h3>

<p>Calculates an independence graph using Gaussian stepwise selection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fgr1st(x,p0=0.01,ind=0,kmn=0,kmx=0,mx=21,nedge=10^5,inr=T,xinr=F,qq=-1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fgr1st_+3A_x">x</code></td>
<td>
<p>The matrix of covariates</p>
</td></tr>
<tr><td><code id="fgr1st_+3A_p0">p0</code></td>
<td>
<p>Cut-off P-value</p>
</td></tr>
<tr><td><code id="fgr1st_+3A_ind">ind</code></td>
<td>
<p>Restricts the dependent nodes to this subset</p>
</td></tr>
<tr><td><code id="fgr1st_+3A_kmn">kmn</code></td>
<td>
<p>The minimum number selected variables for each node irrespective of cut-off P-value</p>
</td></tr>
<tr><td><code id="fgr1st_+3A_kmx">kmx</code></td>
<td>
<p>The maximum number selected variables for each node irrespective of cut-off P-value</p>
</td></tr>
<tr><td><code id="fgr1st_+3A_mx">mx</code></td>
<td>
<p>Maximum number of selected covariates for each node for all subset search</p>
</td></tr> 
<tr><td><code id="fgr1st_+3A_nedge">nedge</code></td>
<td>
<p>Maximum number of edges</p>
</td></tr>
<tr><td><code id="fgr1st_+3A_inr">inr</code></td>
<td>
<p>Logical, if TRUE include an intercept</p>
</td></tr>
<tr><td><code id="fgr1st_+3A_xinr">xinr</code></td>
<td>
<p>Logical, if TRUE intercept already included</p>
</td></tr>
<tr><td><code id="fgr1st_+3A_qq">qq</code></td>
<td>
<p>The number of covariates to choose from. If qq=-1 the number of covariates of x is used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ned Number of edges
</p>
<p>edg List of edges together with P-values for each  edge and proportional reduction of sum of squared residuals.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(boston)
a&lt;-fgr1st(boston[,1:13],ind=3:6)
</code></pre>

<hr>
<h2 id='flag'>Calculation of lagged covariates</h2><span id='topic+flag'></span>

<h3>Description</h3>

<p>Calculation of lagged covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flag(x,n,i,lag,inr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="flag_+3A_x">x</code></td>
<td>
<p>The covariates</p>
</td></tr>
<tr><td><code id="flag_+3A_n">n</code></td>
<td>
<p>The sample size</p>
</td></tr>
<tr><td><code id="flag_+3A_i">i</code></td>
<td>
<p>The dependent variable</p>
</td></tr>
<tr><td><code id="flag_+3A_lag">lag</code></td>
<td>
<p>The maximum lag</p>
</td></tr>
<tr><td><code id="flag_+3A_inr">inr</code></td>
<td>
<p>If true then intersect included</p>
</td></tr>
</table>


<h3>Value</h3>

<p>y The ith covariate of x without a lag, the dependent variable.
</p>
<p>xl The covariates with lags from 1 :lag starting with the first covariate.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(abcq)
abcql&lt;-flag(abcq,240,1,16,TRUE)
a&lt;-f1st(abcql[[1]],abcql[[2]])
</code></pre>

<hr>
<h2 id='fpval'> Calculates the regression coefficients, the P-values and the standard P-values for the chosen subset ind</h2><span id='topic+fpval'></span>

<h3>Description</h3>

<p>Calculates the regression coefficients, the P-values and the standard P-values for the chosen subset ind.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fpval(y,x,ind,inr=T,xinr=F,qq=-1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fpval_+3A_y">y</code></td>
<td>
<p>The dependent variable</p>
</td></tr>
<tr><td><code id="fpval_+3A_x">x</code></td>
<td>
<p>The covariates</p>
</td></tr>
<tr><td><code id="fpval_+3A_ind">ind</code></td>
<td>
<p>The indices of the subset of the covariates whose P-values are required</p>
</td></tr>
<tr><td><code id="fpval_+3A_inr">inr</code></td>
<td>
<p>Logical If TRUE intercept to be included</p>
</td></tr>
<tr><td><code id="fpval_+3A_xinr">xinr</code></td>
<td>
<p>If TRUE intercept already included</p>
</td></tr>
<tr><td><code id="fpval_+3A_qq">qq</code></td>
<td>
<p>The total number of covariates from which ind was chosen. If qq=-1 the number of covariates of x minus length ind plus 1  is taken.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>apv In order the subset ind, the regression coefficients, the Gaussian P-values, the standard P-values and the proportion of sum of squares explained.
</p>
<p>res The residuals
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(boston)
a&lt;-fpval(boston[,14],boston[,1:13],c(1,2,4:6,8:13))
</code></pre>

<hr>
<h2 id='fundr'>Converts directed into an undirected graph</h2><span id='topic+fundr'></span>

<h3>Description</h3>

<p>Conversion of a directed graph into an undirected graph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fundr(gr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fundr_+3A_gr">gr</code></td>
<td>
<p>A directed graph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gr The undirected graph
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(boston)
grb&lt;-fgr1st(boston[,1:13])
grbu&lt;-fundr(grb[[2]][,1:2])
</code></pre>

<hr>
<h2 id='leukeia'>Leukemia data set</h2><span id='topic+leukemia'></span>

<h3>Description</h3>

<p>Dataset of <code class="reqn">n=72</code> persons indicating presence or absence of leukemia (variable 3572) and <code class="reqn">q=3571</code> gene expressions of the 72 persons (variables 1 to 3571)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(leukemia)</code></pre>


<h3>Format</h3>


<dl>
<dt>y</dt><dd><p>0-1 data of individuals with and without leukemia.</p>
</dd>
<dt>x</dt><dd><p>covariates of the level of 3571 genes.</p>
</dd>
</dl>



<h3>Source</h3>

<p>http://stat.ethz.ch/~dettling/bagboost.html</p>


<h3>References</h3>

<p>Boosting for tumor classification with gene expression
data. Dettling, M. and Buehlmann, P. Bioinformatics, 2003,19(9):1061&ndash;1069.
</p>

<hr>
<h2 id='mel-temp'>Melbourne minimum temperature</h2><span id='topic+mel_temp'></span>

<h3>Description</h3>

<p>The daily minimum temperature in Melbourne for the years 1981-1990.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mel_temp</code></pre>


<h3>Format</h3>

<p>A vector  of length 3650</p>


<h3>Source</h3>

<p>https://www.kaggle.com/paulbrabban/daily-minimum-temperatures-in-melbourne
</p>

<hr>
<h2 id='redwine'>Redwine data</h2><span id='topic+redwine'></span>

<h3>Description</h3>

<p>The subjective quality of wine on an integer scale from 1-10 (variable 12) together with 11 physicochemical properties
</p>


<h3>Usage</h3>

<pre><code class='language-R'>redwine</code></pre>


<h3>Format</h3>

<p>A matrix of size  1599 x 12</p>


<h3>Source</h3>

<p>https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/
</p>


<h3>References</h3>

<p> Modeling wine preferences by data mining from
physicochemical properties, Cortez, P., Cerdeira, A., Almeida, F.,
Matos, T., and Reis, J., Decision Support Systems, Elsevier, 2009,47(4):547&ndash;553.
</p>

<hr>
<h2 id='simgpval'>Simulates Gaussian P-values</h2><span id='topic+simgpval'></span>

<h3>Description</h3>

<p>Simulates Gaussian P-values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simgpval(y,x,i,nsim,qq=-1,plt=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simgpval_+3A_y">y</code></td>
<td>
<p>Dependent variable</p>
</td></tr>
<tr><td><code id="simgpval_+3A_x">x</code></td>
<td>
<p>Covariates</p>
</td></tr>
<tr><td><code id="simgpval_+3A_i">i</code></td>
<td>
<p>The chosen covariate</p>
</td></tr>
<tr><td><code id="simgpval_+3A_nsim">nsim</code></td>
<td>
<p>The number of simulations</p>
</td></tr>
<tr><td><code id="simgpval_+3A_qq">qq</code></td>
<td>
<p>The total number of covariates available</p>
</td></tr>
<tr><td><code id="simgpval_+3A_plt">plt</code></td>
<td>
<p>Logical, if TRUE the F P-values of the Gaussian covariates are ordered and plotted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pvg P-value of x_i and relative frequency
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snspt)
snspt&lt;-matrix(snspt,nrow=3253,ncol=1)
a&lt;-flag(snspt,3253,1,12,inr=FALSE)
simgpval(a[[1]],a[[2]],7,10,plt=FALSE)
</code></pre>

<hr>
<h2 id='snspt'>Sunspot data</h2><span id='topic+snspt'></span>

<h3>Description</h3>

<p>The average number of sunspots each month from January 1749 to January 2020
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snspt</code></pre>


<h3>Format</h3>

<p>A vector of size  3253</p>


<h3>Source</h3>

<p>WDC-SILSO, Royal Observatory of Belgium, Brussels</p>

<hr>
<h2 id='vardata'>USA economics data</h2><span id='topic+vardata'></span>

<h3>Description</h3>

<p> United States economic data taken from the FRED-MD macroeconomic database with the NAs removed.182 indices each of length 256</p>


<h3>Usage</h3>

<pre><code class='language-R'>vardata</code></pre>


<h3>Format</h3>

<p>A matrix of size  256 X 182</p>


<h3>Source</h3>

<p>https://research.stlouisfed.org/econ/mccracken/fred-databases</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
