<!DOCTYPE html><html lang="en"><head><title>Help for package GB5mcPred</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GB5mcPred}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Pred5mc'><p>Pred5mc</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gradient Boosting Algorithm for Predicting Methylation States</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Dipro Sinha [aut, cre],
  Sunil Archak [aut],
  Dwijesh Chandra Mishra [aut],
  Tanwy Dasmandal [aut],
  Md Yeasin [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dipro Sinha &lt;diprosinha@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>DNA methylation of 5-methylcytosine (5mC) is the result of a multi-step, enzyme-dependent process. Predicting these sites in-vitro is laborious, time consuming as well as costly. This ' Gb5mC-Pred ' package is an in-silico pipeline for predicting DNA sequences containing the 5mC sites. It uses a machine learning approach which uses Stochastic Gradient Boosting approach for prediction of the sequences with 5mC sites. This package has been developed by using the concept of Navarez and Roxas (2022) &lt;<a href="https://doi.org/10.1109%2FTCBB.2021.3082184">doi:10.1109/TCBB.2021.3082184</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, devtools, tidyverse, seqinr, Biostrings,
splitstackshape, entropy, party, stringr, tibble, doParallel,
parallel, e1071, caret, randomForest, gbm, foreach, ftrCOOL,
iterators</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-07-10 17:22:06 UTC; YEASIN</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-07-11 15:40:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='Pred5mc'>Pred5mc</h2><span id='topic+Pred5mc'></span>

<h3>Description</h3>

<p>Prediction of sequences with 5mc sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pred5mc(Fastafile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Pred5mc_+3A_fastafile">Fastafile</code></td>
<td>
<p>Sequence file (.fasta format)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>MethStatus: Sequences with their methylation state (methylated or non-methylated)
</p>


<h3>References</h3>

<p>Chen, W., Lv, H., Nie, F., &amp; Lin, H. (2019). i6mA-Pred: identifying DNA N6-methyladenine 	sites in the rice genome. Bioinformatics, 35(16), 2796-2800.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(GB5mcPred)
data&lt;-system.file("exdata/test.fasta", package = "GB5mcPred")
pred&lt;-Pred5mc(Fastafile=data)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
