<!DOCTYPE html><html><head><title>Help for package gbs2ploidy</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gbs2ploidy}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gbs2ploidy-package'>
<p>Inference of Ploidy from (Genotyping-by-Sequencing) GBS Data</p></a></li>
<li><a href='#dat'>
<p>Simulated allele counts</p></a></li>
<li><a href='#estploidy'>
<p>Discriminate cytotypes using GBS data</p></a></li>
<li><a href='#estprops'>
<p>Estimate allelic proportions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Inference of Ploidy from (Genotyping-by-Sequencing) GBS Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-12-01</td>
</tr>
<tr>
<td>Author:</td>
<td>Zachariah Gompert</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zachariah Gompert &lt;zach.gompert@usu.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), MASS, rjags</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for inference of ploidy from (Genotyping-by-sequencing) GBS data, including a function to infer allelic ratios and allelic proportions in a Bayesian framework. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-12-01 19:46:38 UTC; zgompert</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-12-01 21:32:40</td>
</tr>
</table>
<hr>
<h2 id='gbs2ploidy-package'>
Inference of Ploidy from (Genotyping-by-Sequencing) GBS Data
</h2><span id='topic+gbs2ploidy-package'></span><span id='topic+gbs2ploidy'></span>

<h3>Description</h3>

<p>Functions for inference of ploidy from (Genotyping-by-sequencing) GBS data, including a function to infer allelic ratios and allelic proportions in a Bayesian framework. 
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> gbs2ploidy</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Inference of Ploidy from (Genotyping-by-Sequencing) GBS Data</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-12-01</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Zachariah Gompert</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Zachariah Gompert &lt;zach.gompert@usu.edu&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R (&gt;= 2.10), MASS, rjags</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Functions for inference of ploidy from (Genotyping-by-sequencing) GBS data, including a function to infer allelic ratios and allelic proportions in a Bayesian framework. </td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Index of help topics:
</p>
<pre>
dat                     Simulated allele counts
estploidy               Discriminate cytotypes using GBS data
estprops                Estimate allelic proportions
gbs2ploidy-package      Inference of Ploidy from
                        (Genotyping-by-Sequencing) GBS Data
</pre>
<p>A typical analysis will begin by estimating allelic proportions using the <code>estprops</code> function. This is done in a Bayesian framework and is the most computationally intensive part of the analysis (i.e., depending on the size of the data set, this might take a day or more). This function depends on <code>rjags</code>, which means the user needs to install the stand-alone program <code>JAGS</code> as well. Principal component analysis and discriminant analysis are then used to obtain cytotype assignment probabilities via the <code>estploidy</code> function. This can be done with or without a training set of individuals with known ploidies.
</p>


<h3>Author(s)</h3>

<p>Zachariah Gompert
</p>
<p>Maintainer: Zachariah Gompert &lt;zach.gompert@usu.edu&gt;
</p>


<h3>References</h3>

<p>Gompert Z. and Mock K. (XXXX) Detection of individual ploidy levels with genotyping-by-sequencing (GBS) analysis. Molecular Ecology Resources, submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load a simulated data set
data(dat)
## Not run: 
## obtain posterior estimates of allelic proportions; short chains are used for 
## the example, we recommend increasing this to at least 1000 MCMC steps with a
## 500 step burnin
props&lt;-estprops(cov1=t(dat[[1]]),cov2=t(dat[[2]]),mcmc.steps=20,mcmc.burnin=5,
    mcmc.thin=2)

## calculate observed heterozygosity and depth of coverage from the allele count
## data
hx&lt;-apply(is.na(dat[[1]]+dat[[2]])==FALSE,1,mean)
dx&lt;-apply(dat[[1]]+dat[[2]],1,mean,na.rm=TRUE)

## run estploidy without using known ploidy data
pl&lt;-estploidy(alphas=props,het=hx,depth=dx,train=FALSE,pl=NA,set=NA,nclasses=2,
    ids=dat[[3]],pcs=1:2)

## boxplots to visualize posterior assignment probabilities by true ploidy 
## (which is known because these are simulated data)
boxplot(pl$pp[,1] ~ dat[[3]],ylab="assignment probability",xlab="ploidy")

## run estploidy with a training data set with known ploidy; the data set is 
## split into 100 individuals with known ploidy and 100 that are used for 
## inference
truep&lt;-dat[[3]]
trn&lt;-sort(sample(1:200,100,replace=FALSE))
truep[-trn]&lt;-NA
plt&lt;-estploidy(alphas=props,het=hx,depth=dx,train=TRUE,pl=truep,set=trn,
    nclasses=2,ids=dat[[3]],pcs=1:2)

## boxplots to visualize posterior assignment probabilities for individuals that
## were not part of the training set by true ploidy (which is known because 
## these are simulated data)
boxplot(plt$pp[,1] ~ dat[[3]][-trn],ylab="assignment probability",xlab="ploidy")

## End(Not run)
</code></pre>

<hr>
<h2 id='dat'>
Simulated allele counts
</h2><span id='topic+dat'></span>

<h3>Description</h3>

<p><code>dat</code> is a simulated data set meant to mimic GBS data. It is a list with three components. The first two components are N (number of individuals) by P (number of SNPs) matrixes with allele counts for the first and second allele at each locus, respectively. The third component is a numeric vector that gives the true ploidy for each individual (2 = diploid, 4 = tetraploid).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("dat")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 3
$ : int [1:200, 1:10000] NA NA 7 4 NA NA NA NA NA NA ...
$ : int [1:200, 1:10000] NA NA 2 5 NA NA NA NA NA NA ...
$ : num [1:200] 4 2 4 2 2 2 2 2 2 2 ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)
str(dat)
</code></pre>

<hr>
<h2 id='estploidy'>
Discriminate cytotypes using GBS data
</h2><span id='topic+estploidy'></span>

<h3>Description</h3>

<p>Use principal component analysis (PCA) and discriminant analysis (DA) to assign individuals to different cytotype groups based on GBS data. This function can be run with or without a training set of individuals with known cytotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estploidy(alphas = NA, het = NA, depth = NA, train = FALSE, pl = NA, set = NA, 
    nclasses = 2, ids = NA, pcs = 1:2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estploidy_+3A_alphas">alphas</code></td>
<td>

<p>a list generated by the <code>estprops</code> function that contains posterior estimates of allelic proportions.
</p>
</td></tr>
<tr><td><code id="estploidy_+3A_het">het</code></td>
<td>

<p>a numeric vector with the observed heterozygosity for each individual, that is the proportion of SNPs at which the individual was heterozygous.
</p>
</td></tr>
<tr><td><code id="estploidy_+3A_depth">depth</code></td>
<td>

<p>a numeric vector with the mean number of reads per SNP for each individual (all SNPs or just heterozygous SNPs).
</p>
</td></tr>
<tr><td><code id="estploidy_+3A_train">train</code></td>
<td>

<p>a boolean specifying whether or not a training set with known ploidy should be used.
</p>
</td></tr>
<tr><td><code id="estploidy_+3A_pl">pl</code></td>
<td>

<p>a vector of known ploidies with one entry per individual (use &lsquo;NA&rsquo; for individuals with unknown ploidy); only used if <code>train == TRUE</code>.
</p>
</td></tr>
<tr><td><code id="estploidy_+3A_set">set</code></td>
<td>

<p>indixes for the training set; only used if <code>train == TRUE</code>.
</p>
</td></tr>
<tr><td><code id="estploidy_+3A_nclasses">nclasses</code></td>
<td>

<p>the number of cyotypes expected.
</p>
</td></tr>
<tr><td><code id="estploidy_+3A_ids">ids</code></td>
<td>

<p>names or other IDs for individuals.
</p>
</td></tr>
<tr><td><code id="estploidy_+3A_pcs">pcs</code></td>
<td>

<p>a vector giving the PC to use for DA.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Assignment probabilities to different cytotype groups are obtained using PCA and DA, as described in Gompert &amp; Mock (XXXX). Residual hetorzygosity (from regressing <code>het</code> on <code>depth</code>) and point estimates (posterior medians) for allelic proportions are first ordinated using PCA (on the centered covariance matrix). The first <code>pcs</code> PCs are retained for DA. If a training set is not provided, k-means clustering is used to obtain initial classifications (with <code>nclasses</code> groups), which are then used for leave-one-out cross-validation with DA. In this case, PC loadings and discriminant scores should be evaluated to associate groups from k-means clustering with specific cytotypes (Gompert &amp; Mock, XXXX). If a training set is provided, it is used to define groups for DA and to estimate assignment probabilities without k-means clustering. 
</p>


<h3>Value</h3>

<p><code>estploidy</code> returns a list with three components:
</p>
<table>
<tr><td><code>pp</code></td>
<td>
<p>A matrix with assignment probabilities for each individual (rows) to each group (columns); the first column gives the <code>ids</code> provided by the user. Only individuals that were not part of the training set are included.</p>
</td></tr>
<tr><td><code>pcwghts</code></td>
<td>
<p>A matrix with the variable loadings (PC weights) from the ordination of residual heterozygosity and allelic proportions. Columns correspond with PCs in ascending order (i.e., the PC with the largest eigenvalue is first).</p>
</td></tr>
<tr><td><code>pcscrs</code></td>
<td>
<p>A matrix of PC scores from the ordination of residual heterozygosity and allelic proportions. Columns correspond with PCs in ascending order (i.e., the PC with the largest eigenvalue is first).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zachariah Gompert
</p>


<h3>References</h3>

<p>Gompert Z. and Mock K. (XXXX) Detection of individual ploidy levels with genotyping-by-sequencing (GBS) analysis. Molecular Ecology Resources, submitted.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+estprops">estprops</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load a simulated data set
data(dat)
## Not run: 
## obtain posterior estimates of allelic proportions; short chains are used for 
## the example, we recommend increasing this to at least 1000 MCMC steps with a
## 500 step burnin
props&lt;-estprops(cov1=t(dat[[1]]),cov2=t(dat[[2]]),mcmc.steps=20,mcmc.burnin=5,
    mcmc.thin=1)

## calculate observed heterozygosity and depth of coverage from the allele count
## data
hx&lt;-apply(is.na(dat[[1]]+dat[[2]])==FALSE,1,mean)
dx&lt;-apply(dat[[1]]+dat[[2]],1,mean,na.rm=TRUE)

## run estploidy without using known ploidy data
pl&lt;-estploidy(alphas=props,het=hx,depth=dx,train=FALSE,pl=NA,set=NA,nclasses=2,
    ids=dat[[3]],pcs=1:2)

## boxplots to visualize posterior assignment probabilities by true ploidy 
## (which is known because these are simulated data)
boxplot(pl$pp[,1] ~ dat[[3]],ylab="assignment probability",xlab="ploidy")

## run estploidy with a training data set with known ploidy; the data set is 
## split into 100 individuals with known ploidy and 100 that are used for 
## inference
truep&lt;-dat[[3]]
trn&lt;-sort(sample(1:200,100,replace=FALSE))
truep[-trn]&lt;-NA
plt&lt;-estploidy(alphas=props,het=hx,depth=dx,train=TRUE,pl=truep,set=trn,
    nclasses=2,ids=dat[[3]],pcs=1:2)

## boxplots to visualize posterior assignment probabilities for individuals that
## were not part of the training set by true ploidy (which is known because 
## these are simulated data)
boxplot(plt$pp[,1] ~ dat[[3]][-trn],ylab="assignment probability",xlab="ploidy")

## End(Not run)
</code></pre>

<hr>
<h2 id='estprops'>
Estimate allelic proportions
</h2><span id='topic+estprops'></span>

<h3>Description</h3>

<p>This functions uses Markov chain Monte Carlo to obtain Bayesian estimates of allelic proportions, which denote that proportion of heterozygous GBS SNPs with different allelic ratios.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estprops(cov1 = NA, cov2 = NA, props = c(0.25, 0.33, 0.5, 0.66, 0.75), 
    mcmc.nchain = 2, mcmc.steps = 10000, mcmc.burnin = 1000, mcmc.thin = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estprops_+3A_cov1">cov1</code></td>
<td>

<p>a P (number of SNPs) by N (number of individuals) matrix with read counts for the first allele (e.g., the non-reference allele). Numeric values should be provided for heterygous SNPs only, homozygous SNPs should be coded as missing data (i.e., &lsquo;NA&rsquo;).
</p>
</td></tr>
<tr><td><code id="estprops_+3A_cov2">cov2</code></td>
<td>

<p>a P (number of SNPs) by N (number of individuals) matrix with read counts for second allele (e.g., the reference allele). Numeric values should be provided for heterygous SNPs only, homozygous SNPs should be coded as missing data (i.e., &lsquo;NA&rsquo;).
</p>
</td></tr>
<tr><td><code id="estprops_+3A_props">props</code></td>
<td>

<p>a vector containing valid allelic proportions given the expected cyotypes present in the sample.
</p>
</td></tr>
<tr><td><code id="estprops_+3A_mcmc.nchain">mcmc.nchain</code></td>
<td>

<p>number of chains for MCMC.
</p>
</td></tr>
<tr><td><code id="estprops_+3A_mcmc.steps">mcmc.steps</code></td>
<td>

<p>number of post burnin iterations for each chain.
</p>
</td></tr>
<tr><td><code id="estprops_+3A_mcmc.burnin">mcmc.burnin</code></td>
<td>

<p>number of iterations to discard from each chain as a burnin.
</p>
</td></tr>
<tr><td><code id="estprops_+3A_mcmc.thin">mcmc.thin</code></td>
<td>

<p>thinning interval for MCMC.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Allelic proportions are inferred from the allele counts based on the Bayesian model described in Gompert \&amp; Mock (XXXX). Please consult this publication for a detailed description of the model. Users can modify the vector of possible allelic proportions based on expectations for their data set. For example, true allelic proportions for diploids, triploids and tetraploids are 1:1 (0.5), 1:2 or 2:1 (0.33 or 0.66), and 1:3, 2:2, or 3:1 (0.25, 0.5, or 0.75), respectively.	
</p>


<h3>Value</h3>

<p><code>estprops</code> returns a list with one component per individual. Components summarize the posterior distributions for allelic proportions. Rows correspond to different allelic proportions (as defined by &lsquo;props&rsquo;) and columns give the 2.5th, 25th, 50th, 75th, and 97.5th quantiles of the posterior distribution for each parameter.
</p>


<h3>Author(s)</h3>

<p>Zachariah Gompert
</p>


<h3>References</h3>

<p>Gompert Z. and Mock K. (XXXX) Detection of individual ploidy levels with genotyping-by-sequencing (GBS) analysis. Molecular Ecology Resources, submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load a simulated data set
data(dat)
## Not run: 
## obtain posterior estimates of allelic proportions; short chains are used for 
## the example, we recommend increasing this to at least 1000 MCMC steps with a 
## 500 step burnin
props&lt;-estprops(cov1=t(dat[[1]]),cov2=t(dat[[2]]),mcmc.steps=20,mcmc.burnin=5,
    mcmc.thin=1)

## plot point estimates and 95
## allelic proportions for the first nine individuals
par(mfrow=c(3,3))
for(i in 1:9){
    plot(props[[i]][,3],ylim=c(0,1),axes=FALSE,xlab="ratios",ylab="proportions")
	axis(1,at=1:5,c("1:3","1:2","1:1","2:1","3:1"))
	axis(2)
	box()
	segments(1:5,props[[i]][,1],1:5,props[[i]][,5])
	title(main=paste("true ploidy =",dat[[3]][i]))
}

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
