<!DOCTYPE html><html lang="en-GB"><head><title>Help for package GCalignR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GCalignR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GCalignR'><p>GCalignR: A Package to Align Gas Chromatography Peaks Based on Retention Times</p></a></li>
<li><a href='#align_chromatograms'><p>Aligning peaks based on retention times</p></a></li>
<li><a href='#align_peaks'><p>align peaks individually among chromatograms</p></a></li>
<li><a href='#align_peaks_fast'><p>align peaks individually among chromatograms</p></a></li>
<li><a href='#aligned_peak_data'><p>Aligned Gas-Chromatography data</p></a></li>
<li><a href='#as.data.frame.GCalign'><p>Output aligned data in form of a data frame for each variable</p></a></li>
<li><a href='#blank_substraction'><p>Subtraction of blank readings from sample readings</p></a></li>
<li><a href='#check_input'><p>Check input prior to processing in GCalignR</p></a></li>
<li><a href='#choose_optimal_reference'><p>Select the optimal reference for full alignments of peak lists</p></a></li>
<li><a href='#draw_chromatogram'><p>Visualise peak lists as a pseudo-chromatogram</p></a></li>
<li><a href='#find_peaks'><p>Detect local maxima in time series</p></a></li>
<li><a href='#gc_heatmap'><p>Visualises peak alignments in form of a heatmap</p></a></li>
<li><a href='#linear_transformation'><p>Full Alignment of Peak Lists by linear retention time correction.</p></a></li>
<li><a href='#merge_redundant_rows'><p>Merge redundant rows</p></a></li>
<li><a href='#norm_peaks'><p>Normalisation of peak abundancies</p></a></li>
<li><a href='#peak_data'><p>Gas-chromatography data for Antarctic Fur Seals (<em>Arctocephalus gazella</em>)</p></a></li>
<li><a href='#peak_factors'><p>Grouping factors corresponding to gas-chromatography data of Antarctic Fur Seals (<em>Arctocephalus gazella</em>)</p></a></li>
<li><a href='#peak_interspace'><p>Estimate the observed space between peaks within chromatograms</p></a></li>
<li><a href='#plot.GCalign'><p>Plot diagnostics for an GCalign Object</p></a></li>
<li><a href='#print.GCalign'><p>Summarising Peak Alignments with GCalignR</p></a></li>
<li><a href='#read_empower2'><p>Import data from single EMPOWER2 HPLC files</p></a></li>
<li><a href='#read_peak_list'><p>Read content of a text file and convert it to a list</p></a></li>
<li><a href='#remove_blanks'><p>Remove peaks present in negative control samples</p></a></li>
<li><a href='#remove_singletons'><p>Remove singletons</p></a></li>
<li><a href='#simple_chroma'><p>Simulate simple chromatograms</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Simple Peak Alignment for Gas-Chromatography Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-07-03</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Description:</td>
<td>Aligns peak based on peak retention times and matches homologous peaks
    across samples. The underlying alignment procedure comprises three sequential steps.
    (1) Full alignment of samples by linear transformation of retention times to 
    maximise similarity among homologous peaks (2) Partial alignment of peaks within 
    a user-defined retention time window to cluster homologous peaks (3) Merging rows
    that are likely representing homologous substances (i.e. no sample shows peaks in 
    both rows and the rows have similar retention time means). The algorithm is described in detail
    in Ottensmann et al., 2018 &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0198311">doi:10.1371/journal.pone.0198311</a>&gt;. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2 (&ge; 2.2.1), graphics, stats, readr, reshape2, stringr,
utils, pbapply, methods, tibble</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> | file LICENSE [expanded from: GPL (&ge; 2) | file LICENSE]</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, pander, rmarkdown, testthat, vegan (&ge; 2.4.2)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mottensmann/GCalignR/issues">https://github.com/mottensmann/GCalignR/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mottensmann/GCalignR">https://github.com/mottensmann/GCalignR</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-03 17:46:30 UTC; meino</td>
</tr>
<tr>
<td>Author:</td>
<td>Meinolf Ottensmann
    <a href="https://orcid.org/0000-0003-3112-6928"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Martin Stoffel [aut],
  Hazel J. Nichols [aut],
  Joseph I. Hoffman [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Meinolf Ottensmann &lt;meinolf.ottensmann@web.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-03 18:00:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='GCalignR'>GCalignR: A Package to Align Gas Chromatography Peaks Based on Retention Times</h2><span id='topic+GCalignR-package'></span><span id='topic+GCalignR'></span>

<h3>Description</h3>

<p><strong><code>GCalignR</code></strong> contains the functions listed below. Follow the links to access the documentation of each function.
</p>
<p><code><a href="#topic+align_chromatograms">align_chromatograms</a></code> executes all alignment steps.
</p>
<p><code><a href="#topic+as.data.frame.GCalign">as.data.frame.GCalign</a></code> exports aligned data to data frames.
</p>
<p><code><a href="#topic+check_input">check_input</a></code> tests the input data for formatting issues.
</p>
<p><code><a href="#topic+draw_chromatogram">draw_chromatogram</a></code> visualises peak lists in form of a chromatogram.
</p>
<p><code><a href="#topic+find_peaks">find_peaks</a></code> detects and calculates peak heights in chromatograms. Not intended to be used for peak integration in empirical data. Used for illustration purposes only.
</p>
<p><code><a href="#topic+gc_heatmap">gc_heatmap</a></code> visualises aligned datasets using heatmaps that can be customised.
</p>
<p><code><a href="#topic+norm_peaks">norm_peaks</a></code> allows to compute the relative abundance of peaks with samples.
</p>
<p><code><a href="#topic+peak_interspace">peak_interspace</a></code> gives a histogram of the distance between peaks within samples over the whole dataset.
</p>
<p><code><a href="#topic+read_peak_list">read_peak_list</a></code> reads the content of a text file and converts it to a list.
</p>
<p><code><a href="#topic+remove_blanks">remove_blanks</a></code> removes peaks resembling contaminations from aligned datasets.
</p>
<p><code><a href="#topic+remove_singletons">remove_singletons</a></code> removes peaks that are unique for one individual sample.
</p>
<p><code><a href="#topic+simple_chroma">simple_chroma</a></code> creates simple chromatograms for testing and illustration purposes.
</p>


<h3>Details</h3>

<p>More details on the package are found in the vignettes that can be accessed via <code>browseVignettes("GCalignR")</code>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Meinolf Ottensmann <a href="mailto:meinolf.ottensmann@web.de">meinolf.ottensmann@web.de</a> (<a href="https://orcid.org/0000-0003-3112-6928">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Martin Stoffel <a href="mailto:martin.adam.stoffel@gmail.com">martin.adam.stoffel@gmail.com</a>
</p>
</li>
<li><p> Hazel J. Nichols
</p>
</li>
<li><p> Joseph I. Hoffman
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/mottensmann/GCalignR">https://github.com/mottensmann/GCalignR</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/mottensmann/GCalignR/issues">https://github.com/mottensmann/GCalignR/issues</a>
</p>
</li></ul>


<hr>
<h2 id='align_chromatograms'>Aligning peaks based on retention times</h2><span id='topic+align_chromatograms'></span>

<h3>Description</h3>

<p>This is the core function of <code><a href="#topic+GCalignR">GCalignR</a></code> to align peak data. The input data is a peak list. Read through the documentation below and take a look at the vignettes for a thorough introduction. Three parameters <code>max_linear_shift</code>, <code>max_diff_peak2mean</code> and <code>min_diff_peak2peak</code> are required as well as the column name of the peak retention time variable <code>rt_col_name</code>. Arguments are described among optional parameters below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_chromatograms(
  data,
  sep = "\t",
  rt_col_name = NULL,
  write_output = NULL,
  rt_cutoff_low = NULL,
  rt_cutoff_high = NULL,
  reference = NULL,
  max_linear_shift = 0.02,
  max_diff_peak2mean = 0.02,
  min_diff_peak2peak = 0.08,
  blanks = NULL,
  delete_single_peak = FALSE,
  remove_empty = FALSE,
  permute = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_chromatograms_+3A_data">data</code></td>
<td>
<p>Dataset containing peaks that need to be aligned and matched. For every peak a arbitrary number of numerical variables can be included (e.g. peak height, peak area) in addition to the mandatory retention time. The standard format is a tab-delimited text file according to the following layout: (1) The first row contains sample names, the (2) second row column names of the corresponding peak lists. Starting with the third row, peak lists are included for every sample that needs to be incorporated in the dataset. Here, a peak list contains data for individual peaks in rows, whereas columns specify variables in the order given in the second row of the text file. Peak lists of individual samples are concatenated horizontally and need to be of the same width (i.e. the same number of columns in consistent order). Alternatively, the input may be a list of data frames. Each data frame contains the peak data for a single individual. Variables (i.e.columns) are named consistently across data frames. The names of elements in the list are used as sample identifiers. Cells may be filled with numeric or integer values but no factors or characters are allowed. NA and 0 may be used to indicate empty rows.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_sep">sep</code></td>
<td>
<p>The field separator character. The default is tab separated (<code>sep = '\t'</code>).
See the &quot;sep&quot; argument in <code><a href="utils.html#topic+read.table">read.table</a></code> for details.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_rt_col_name">rt_col_name</code></td>
<td>
<p>A character giving the name of the column containing the retention times. The decimal separator needs to be a point.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_write_output">write_output</code></td>
<td>
<p>A character vector of variable names. For each variable a text file containing the aligned dataset is written to the working directory. Vector elements need to correspond to column names of data.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_rt_cutoff_low">rt_cutoff_low</code></td>
<td>
<p>A numeric value giving a retention time threshold. Peaks with retention time below the threshold are removed in a preprocessing step.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_rt_cutoff_high">rt_cutoff_high</code></td>
<td>
<p>A numeric value giving a retention time threshold. Peaks with retention time above the threshold are removed in a preprocessing step.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_reference">reference</code></td>
<td>
<p>A character giving the name of sample from the dataset. By default, a sample is automatically selected from the dataset using the function <code><a href="#topic+choose_optimal_reference">choose_optimal_reference</a></code>. The reference is used for the full alignment of peak lists by linear transformation.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_max_linear_shift">max_linear_shift</code></td>
<td>
<p>Numeric value giving the window size considered in the full alignment. Usually, the amplitude of linear drift is small in typical GC-FID datasets. Therefore, the default value of 0.05 minutes is adequate for most datasets. Increase this value if the drift amplitude is larger.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_max_diff_peak2mean">max_diff_peak2mean</code></td>
<td>
<p>Numeric value defining the allowed deviation of the retention time of a given peak from the mean of the corresponding row (i.e. scored substance). This parameter reflects the retention time range in which peaks across samples are still matched as homologous peaks (i.e. substance). Peaks with retention times exceeding the threshold are sorted into a different row.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_min_diff_peak2peak">min_diff_peak2peak</code></td>
<td>
<p>Numeric value defining the expected minimum difference in retention times among homologous peaks (i.e. substance). Rows that differ less in the mean retention time, are therefore merged if every sample contains either one or none of the respective compounds. This parameter is a major determinant in the classification of distinct peaks. Therefore careful consideration is required to adjust this setting to your needs (e.g. the resolution of your gas-chromatography pipeline). Large values may cause to merge truly different substances with similar retention times, if those are not simultaneously occurring within at least one individual, which might occur by chance for small sample sizes. By default set to 0.2 minutes.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_blanks">blanks</code></td>
<td>
<p>Character vector of names of negative controls. Substances found in any of the blanks will be removed from the aligned dataset, before the blanks are deleted from the aligned data as well. This is an optional filtering step.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_delete_single_peak">delete_single_peak</code></td>
<td>
<p>Boolean, determining whether substances that occur in just one sample are removed or not.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_remove_empty">remove_empty</code></td>
<td>
<p>Boolean, allows to remove samples which lack any peak after the alignment finished. By default FALSE</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_permute">permute</code></td>
<td>
<p>Boolean, by default a random permutation of samples is conducted prior for each row-wise alignment step. Setting this parameter to FALSE causes alignment of the dataset as it is.
</p>
<p>order of samples is constantly randomised during the alignment. Allows to prevent this behaviour for maximal repeatability if needed.</p>
</td></tr>
<tr><td><code id="align_chromatograms_+3A_...">...</code></td>
<td>
<p>optional arguments passed to methods, see <code><a href="graphics.html#topic+barplot">barplot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function aligns and matches homologous peaks across samples using a three-step algorithm based on user-defined parameters that are explained in the next section. In brief: <strong>(1)</strong> A full alignment of peak retention times is conducted to correct for systematic linear drift of retention times among homologous peaks from run to run. Thereby a coarse alignment is achieved that maximises the similarity of retention times across homologous peaks prior to a <strong>(2)</strong> partial alignment and matching of peaks. This and the next step in the alignment is based on a retention time matrix that contains all samples as columns and peak retention times in rows. The goal is to match homologous peaks within the same row that represents a chemical substance. Here, peaks are recognised as homologous when the retention time matches within a user-defined error span (see <code>max_diff_peak2mean</code>) that approximates the expected retention time uncertainty. Here, the position of every peak in the matrix is evaluated in comparison to similar peaks across the complete dataset and homologous peaks are gradually grouped together row by row. After all peaks were matched, a <strong>(3)</strong> adjacent rows of similar retention time are scanned to detect redundancies. A pair of rows is identified as redundant and merged if mean retention times are closer than specified by <code>min_diff_peak2peak</code> and single samples only contain peak in one but not both rows. Therefore, this step allows to match peaks that are associated with higher variance than allowed during the previous step. Several optional processing steps are available, ranging from the removal of peaks representing contaminations (requires to include blanks as a control) to the removal of uninformative peaks that are present in just one sample (so called singletons).
</p>


<h3>Value</h3>

<p>Returns an object of class &quot;GCalign&quot; that is a a list containing several objects that are listed below. Note, that the objects &quot;heatmap_input&quot; and &quot;Logfile&quot; are best inspected by calling the provided functions <code>gc_heatmap</code> and <code>print</code>.
</p>
<table role = "presentation">
<tr><td><code>aligned</code></td>
<td>
<p>Aligned Gas Chromatography peak data subdivided into individual data frames for every variable. Samples are represented by columns, rows specify homologous peaks. The first column of every data frame is comprised of the mean retention time of the respective peak (i.e. row). Retention times of samples resemble the values of the raw data. Internally, linear adjustments are considered where appropriate</p>
</td></tr>
<tr><td><code>heatmap_input</code></td>
<td>
<p>Used internally to create heatmaps of the aligned data</p>
</td></tr>
<tr><td><code>Logfile</code></td>
<td>
<p>A protocol of the alignment process.</p>
</td></tr>
<tr><td><code>input_list</code></td>
<td>
<p>Input data in form of a list of data frames.</p>
</td></tr>
<tr><td><code>aligned_list</code></td>
<td>
<p>Aligned data in form of a list of data frames.</p>
</td></tr>
<tr><td><code>input_matrix</code></td>
<td>
<p>List of matrices. Each matrix contains the input data for a variable</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com) &amp; Meinolf Ottensmann (meinolf.ottensmann@web.de)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load example dataset
data("peak_data")
## Subset for faster processing
peak_data &lt;- peak_data[1:3]
peak_data &lt;- lapply(peak_data, function(x) x[1:50,])
## align data with default settings
out &lt;- align_chromatograms(peak_data, rt_col_name = "time")

</code></pre>

<hr>
<h2 id='align_peaks'>align peaks individually among chromatograms</h2><span id='topic+align_peaks'></span>

<h3>Description</h3>

<p><strong>align_peaks</strong> allows to align similar peaks across samples
so that shared peaks are consistently located at the the same location (i.e.
defined as the same substance). The order of chromatograms (i.e. data.frames
in <code>gc_peak_list</code>) is randomized before each run of the alignment of
algorithm (if randomisation is not needed, this behaviour can be changed by setting <strong>permute = FALSE</strong>). The main principle of this function is to reduce the variance in
retention times within rows, thereby peaks of similar retention time are
grouped together. Peaks that deviate significantly from the mean retention times
of the other samples are shifted to another row. At the start of a row the
first two samples are compared and separated if required, then all other
samples are included consecutively. If <code>iterations &gt; 1</code> the whole
algorithm is repeated accordingly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_peaks(
  gc_peak_list,
  max_diff_peak2mean = 0.02,
  iterations = 1,
  rt_col_name,
  permute = TRUE,
  R = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_peaks_+3A_gc_peak_list">gc_peak_list</code></td>
<td>
<p>List of data.frames. Each data.frame contains GC-data
(e.g. retention time, peak area, peak height) of one sample. Variables are
stored in columns. Rows represent distinct peaks. Retention time is a
required variable.</p>
</td></tr>
<tr><td><code id="align_peaks_+3A_max_diff_peak2mean">max_diff_peak2mean</code></td>
<td>
<p>Numeric value defining the allowed deviation of the retention time of a given peak from the mean of the corresponding row (i.e. scored substance). This parameter reflects the retention time range in which peaks across samples are still matched as homologous peaks (i.e. substance). Peaks with retention times exceeding the threshold are sorted into a different row.</p>
</td></tr>
<tr><td><code id="align_peaks_+3A_rt_col_name">rt_col_name</code></td>
<td>
<p>A character giving the name of the column containing the retention times. The decimal separator needs to be a point.</p>
</td></tr>
<tr><td><code id="align_peaks_+3A_permute">permute</code></td>
<td>
<p>Boolean, by default a random permutation of samples is conducted prior for each row-wise alignment step. Setting this parameter to FALSE causes alignment of the dataset as it is.
</p>
<p>order of samples is constantly randomised during the alignment. Allows to prevent this behaviour for maximal repeatability if needed.</p>
</td></tr>
<tr><td><code id="align_peaks_+3A_r">R</code></td>
<td>
<p>integer indicating the current iteration of the alignment step.
Created by <a href="#topic+align_chromatograms">align_chromatograms</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each row the retention time of every sample is compared to the
mean retention time of all previously examined samples within the same row.
Starting with the second sample a comparison is done between the first and the
second sample, then between the third and the two first ones and so on.
Whenever the current sample shows a deviation from the mean retention time of
the previous samples a shift will either move this sample to the next row
(i.e. retention time above average) or all other samples will be moved to the
next row (i.e. retention time below average). If the retention time of the
sample in evaluation shows no deviation within <strong>-max_diff_peak2mean</strong>:
<strong>max_diff_peak2mean</strong> around the mean retention time no shifting is done
and the algorithm proceeds with the following sample.
</p>


<h3>Value</h3>

<p>a list of data.frames containing GC-data with aligned peaks.
</p>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com) &amp; Meinolf Ottensmann
(meinolf.ottensmann@web.de)
</p>

<hr>
<h2 id='align_peaks_fast'>align peaks individually among chromatograms</h2><span id='topic+align_peaks_fast'></span>

<h3>Description</h3>

<p><strong>align_peaks_fast</strong> is a fast implementation to sort peaks strictly based on the exact retention time values across samples. See <a href="#topic+align_peaks">align_peaks</a> for details on the standard approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_peaks_fast(gc_peak_list, rt_col_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_peaks_fast_+3A_gc_peak_list">gc_peak_list</code></td>
<td>
<p>List of data.frames. Each data.frame contains GC-data
(e.g. retention time, peak area, peak height) of one sample. Variables are
stored in columns. Rows represent distinct peaks. Retention time is a
required variable.</p>
</td></tr>
<tr><td><code id="align_peaks_fast_+3A_rt_col_name">rt_col_name</code></td>
<td>
<p>A character giving the name of the column containing the retention times. The decimal separator needs to be a point.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a matrix with the number of rows corresponding to the unique number of retention times across the dataset. Then, for each sample peaks are mapped to the corresponding row (=retention time) of the matrix and in parallel <code>gc_peak_list</code> is updated.
</p>


<h3>Value</h3>

<p>a list of data.frames containing GC-data with aligned peaks.
</p>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com) &amp; Meinolf Ottensmann
(meinolf.ottensmann@web.de)
</p>

<hr>
<h2 id='aligned_peak_data'>Aligned Gas-Chromatography data</h2><span id='topic+aligned_peak_data'></span>

<h3>Description</h3>

<p>This is in example of an aligned gas-chromatography dataset processed with <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>. The raw data is accessible within this package as <strong>peak_data.RData</strong> and is comprised of 41 Mother-Pup pairs of Antarctic Fur Seals (<em>Arctocephalus gazella</em>) sampled from two different colonies at Bird Island, South Georgia. In addition two blanks are included.
</p>


<h3>Format</h3>

<p>Object of class &quot;GCalign&quot; including three lists. List <strong>&quot;aligned&quot;</strong> includes data.frames
for all variables present in the raw data (&quot;time&quot; and &quot;area&quot;). The list <strong>&quot;heatmap_input&quot;</strong> holds data frames with retention times of the input data, linearly adjusted retention times as well as the final output, were peaks are aligned among samples. This file is primarily used in <code><a href="#topic+gc_heatmap">gc_heatmap</a></code>. The list <strong>&quot;Logfile&quot;</strong> summarises the alignment process and the data structure before, during and after running <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>. For a convenient overview use <code><a href="#topic+print.GCalign">print.GCalign</a></code>.
</p>


<h3>Source</h3>

<p>http://www.pnas.org/content/suppl/2015/08/05/1506076112.DCSupplemental/pnas.1506076112.sd02.xlsx
</p>


<h3>References</h3>

<p>Stoffel, M.A.; Caspers, B.A.; Forcada, J.; Giannakara, A.; Baier, M.; Eberhart-Phillips, L.; Mueller, C.; Hoffman, J.I. (2015): Chemical fingerprints encode mother-offspring similarity, colony membership, relatedness, and genetic quality in fur seals. In: Proceedings of the National Academy of Sciences of the United States of America 112 (36), S. E5005-12. DOI: 10.1073/pnas.1506076112.
</p>

<hr>
<h2 id='as.data.frame.GCalign'>Output aligned data in form of a data frame for each variable</h2><span id='topic+as.data.frame.GCalign'></span>

<h3>Description</h3>

<p>Based on an object of class &quot;GCalign&quot; that was created using <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>, a list of data frames for each variable in the dataset is returned. Within data frames rows represent substances and columns are variables (i.e. substances).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GCalign'
as.data.frame(x, row.names = NULL, optional = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as.data.frame.GCalign_+3A_x">x</code></td>
<td>
<p>An object of class &quot;GCalign&quot;. See <code><a href="#topic+align_chromatograms">align_chromatograms</a></code> for details.</p>
</td></tr>
<tr><td><code id="as.data.frame.GCalign_+3A_row.names">row.names</code></td>
<td>
<p><code>NULL</code> or a character vector giving the row
names for the data frame.  Missing values are not allowed.</p>
</td></tr>
<tr><td><code id="as.data.frame.GCalign_+3A_optional">optional</code></td>
<td>
<p>logical. If <code>TRUE</code>, setting row names and
converting column names (to syntactic names: see
<code><a href="base.html#topic+make.names">make.names</a></code>) is optional.  Note that all of <span class="rlang"><b>R</b></span>'s
<span class="pkg">base</span> package <code>as.data.frame()</code> methods use
<code>optional</code> only for column names treatment, basically with the
meaning of <code><a href="base.html#topic+data.frame">data.frame</a>(*, check.names = !optional)</code>.
See also the <code>make.names</code> argument of the <code>matrix</code> method.</p>
</td></tr>
<tr><td><code id="as.data.frame.GCalign_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed to or from methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Meinolf Ottensmann (meinolf.ottensmann@web.de) &amp; Martin Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("aligned_peak_data")
out &lt;- as.data.frame(x = aligned_peak_data)

</code></pre>

<hr>
<h2 id='blank_substraction'>Subtraction of blank readings from sample readings</h2><span id='topic+blank_substraction'></span>

<h3>Description</h3>

<p>For each substance that is present in blanks, samples are corrected by subtraction of the respective quantity. If more than one sample is submitted, abundances are averaged. This procedure is sensitive to differences in the total concentration of samples and should be applied to samples where the preparation yields comparable concentrations for each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blank_substraction(input = NULL, blanks = NULL, conc_col_name = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="blank_substraction_+3A_input">input</code></td>
<td>
<p>A GCalign Object</p>
</td></tr>
<tr><td><code id="blank_substraction_+3A_blanks">blanks</code></td>
<td>
<p>Character vector of names of negative controls.</p>
</td></tr>
<tr><td><code id="blank_substraction_+3A_conc_col_name">conc_col_name</code></td>
<td>
<p>If the input is a GCalign object the variable containing the abundance values needs to be specified.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Substances that are present in one or more blanks are identified in the aligned dataset, then the mean abundance is calculated for the blanks and the corresponding value is subtracted from each sample. If the control contains higher concentration (i.e. blank substation creates negative abundances) warnings will be shown and the respective value will be set to zero
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run
#out &lt;- blank_substraction(aligned_peak_data, blanks = "M2", conc_col_name = "area")


</code></pre>

<hr>
<h2 id='check_input'>Check input prior to processing in GCalignR</h2><span id='topic+check_input'></span>

<h3>Description</h3>

<p>Checks input files for common formatting problems.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_input(data, plot = FALSE, sep = "\t", message = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_input_+3A_data">data</code></td>
<td>
<p>Dataset containing peaks that need to be aligned and matched. For every peak a arbitrary number of numerical variables can be included (e.g. peak height, peak area) in addition to the mandatory retention time. The standard format is a tab-delimited text file according to the following layout: (1) The first row contains sample names, the (2) second row column names of the corresponding peak lists. Starting with the third row, peak lists are included for every sample that needs to be incorporated in the dataset. Here, a peak list contains data for individual peaks in rows, whereas columns specify variables in the order given in the second row of the text file. Peak lists of individual samples are concatenated horizontally and need to be of the same width (i.e. the same number of columns in consistent order). Alternatively, the input may be a list of data frames. Each data frame contains the peak data for a single individual. Variables (i.e.columns) are named consistently across data frames. The names of elements in the list are used as sample identifiers. Cells may be filled with numeric or integer values but no factors or characters are allowed. NA and 0 may be used to indicate empty rows.</p>
</td></tr>
<tr><td><code id="check_input_+3A_plot">plot</code></td>
<td>
<p>Boolean specifying if the distribution of peak numbers is plotted.</p>
</td></tr>
<tr><td><code id="check_input_+3A_sep">sep</code></td>
<td>
<p>The field separator character. The default is tab separated (<code>sep = '\t'</code>).
See the &quot;sep&quot; argument in <code><a href="utils.html#topic+read.table">read.table</a></code> for details.</p>
</td></tr>
<tr><td><code id="check_input_+3A_message">message</code></td>
<td>
<p>Boolean determining if passing all checks is indicated by a message.</p>
</td></tr>
<tr><td><code id="check_input_+3A_...">...</code></td>
<td>
<p>optional arguments passed to methods, see <code><a href="graphics.html#topic+barplot">barplot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sample names should contain just letters, numbers and underscores and no whitespaces.
Each sample has to contain the same number of columns, one of which is the retention
time and the others are arbitrary variables in consistent order across samples. Retention times are expected to be numeric, i.e. they are only allowed to contain numbers from 0-9 and &quot;.&quot; as the only decimal character. Have a look at the vignettes for examples.
</p>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com) &amp; Meinolf Ottensmann
(meinolf.ottensmann@web.de)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## gc-data
data("peak_data")
## Checks format
check_input(peak_data)
## Includes a barplot of peak numbers in the raw data
check_input(peak_data, plot = TRUE)

</code></pre>

<hr>
<h2 id='choose_optimal_reference'>Select the optimal reference for full alignments of peak lists</h2><span id='topic+choose_optimal_reference'></span>

<h3>Description</h3>

<p>Full alignments of peak lists require the specification of a fixed reference to which all other samples are aligned to. This function provides an simple algorithm to find the most suitable sample among a dataset. The so defined reference can be used for full alignments using <code><a href="#topic+linear_transformation">linear_transformation</a></code>. The functions is evoked internally by <code><a href="#topic+align_chromatograms">align_chromatograms</a></code> if no reference was specified by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>choose_optimal_reference(data = NULL, rt_col_name = NULL, sep = "\t")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="choose_optimal_reference_+3A_data">data</code></td>
<td>
<p>Dataset containing peaks that need to be aligned and matched. For every peak a arbitrary number of numerical variables can be included (e.g. peak height, peak area) in addition to the mandatory retention time. The standard format is a tab-delimited text file according to the following layout: (1) The first row contains sample names, the (2) second row column names of the corresponding peak lists. Starting with the third row, peak lists are included for every sample that needs to be incorporated in the dataset. Here, a peak list contains data for individual peaks in rows, whereas columns specify variables in the order given in the second row of the text file. Peak lists of individual samples are concatenated horizontally and need to be of the same width (i.e. the same number of columns in consistent order). Alternatively, the input may be a list of data frames. Each data frame contains the peak data for a single individual. Variables (i.e.columns) are named consistently across data frames. The names of elements in the list are used as sample identifiers. Cells may be filled with numeric or integer values but no factors or characters are allowed. NA and 0 may be used to indicate empty rows.</p>
</td></tr>
<tr><td><code id="choose_optimal_reference_+3A_rt_col_name">rt_col_name</code></td>
<td>
<p>A character giving the name of the column containing the retention times. The decimal separator needs to be a point.</p>
</td></tr>
<tr><td><code id="choose_optimal_reference_+3A_sep">sep</code></td>
<td>
<p>The field separator character. The default is tab separated (<code>sep = '\t'</code>).
See the &quot;sep&quot; argument in <code><a href="utils.html#topic+read.table">read.table</a></code> for details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Every sample is considered in determining the optimal reference in comparison to all other samples by estimating the similarity to all other samples. For a reference-sample pair, the deviation in retention times between all reference peaks and the always nearest peak in the sample is summed up and divided by the number of reference peaks. The calculated value is a similarity score that converges to zero the more similar reference and sample are. For every potential reference, the median score of all pair-wise comparisons is used as a similarity proxy. The optimal sample is then defined by the minimum value among these scores. This functions is used internally in <code><a href="#topic+align_chromatograms">align_chromatograms</a></code> to select a reference if non was specified by the user.
</p>


<h3>Value</h3>

<p>A list with following elements
</p>
<table role = "presentation">
<tr><td><code>sample</code></td>
<td>
<p>Name of the sample with the highest average similarity to all other samples</p>
</td></tr>
<tr><td><code>score</code></td>
<td>
<p>Median number of shared peaks with other samples</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com) &amp; Meinolf Ottensmann (meinolf.ottensmann@web.de)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## 1.) input is a list
## using a list of samples
data("peak_data")
## subset for faster processing
peak_data &lt;- peak_data[1:3]
choose_optimal_reference(peak_data, rt_col_name = "time")

</code></pre>

<hr>
<h2 id='draw_chromatogram'>Visualise peak lists as a pseudo-chromatogram</h2><span id='topic+draw_chromatogram'></span>

<h3>Description</h3>

<p>Creates a graphical representation of one or multiple peak lists in the form of a pseudo- chromatogram. Peaks are represented by Gaussian distributions centred at the peak retention time. The peak height is arbitrary and does not reflect any measured peak intensity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_chromatogram(
  data = NULL,
  rt_col_name = NULL,
  conc_col_name = NULL,
  width = 0.1,
  step = NULL,
  sep = "\t",
  breaks = NULL,
  rt_limits = NULL,
  samples = NULL,
  show_num = FALSE,
  show_rt = FALSE,
  plot = TRUE,
  shape = c("gaussian", "stick"),
  legend.position = "bottom"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="draw_chromatogram_+3A_data">data</code></td>
<td>
<p>The input data can be either a GCalignR input file or an GCalign object. See <code><a href="#topic+align_chromatograms">align_chromatograms</a></code> for details on both.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_rt_col_name">rt_col_name</code></td>
<td>
<p>A character giving the name of the column containing the retention times. The decimal separator needs to be a point.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_conc_col_name">conc_col_name</code></td>
<td>
<p>Character, denoting a variable used to scale the peak height (e.g., peak area or peak height.)</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_width">width</code></td>
<td>
<p>Numeric value giving the standard deviation of Gaussian peaks. Decrease this value to separate overlapping peaks within samples. Default is 0.01.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_step">step</code></td>
<td>
<p>character allowing to visualise different steps of the alignment when a GCalign object is used. By default the aligned data is shown.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_sep">sep</code></td>
<td>
<p>The field separator character. The default is tab separated (<code>sep = '\t'</code>).
See the &quot;sep&quot; argument in <code><a href="utils.html#topic+read.table">read.table</a></code> for details.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_breaks">breaks</code></td>
<td>
<p>A numeric vector giving the breakpoints between ticks on the x axis.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_rt_limits">rt_limits</code></td>
<td>
<p>A numeric vector of length two giving min and max values or retention times to plot.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_samples">samples</code></td>
<td>
<p>A character vector of sample names to draw chromatograms of a subset.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_show_num">show_num</code></td>
<td>
<p>Boolean indicating whether sample numbers are drawn on top of each peak.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_show_rt">show_rt</code></td>
<td>
<p>Boolean indicating whether peak retention times are drawn on top of each peak.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_plot">plot</code></td>
<td>
<p>Boolean indicating if the plot is printed.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_shape">shape</code></td>
<td>
<p>A character determining the shape of peaks. Peaks are approximated as &quot;gaussian&quot; by default. Alternatively, peaks can be visualised as &quot;sticks&quot;.</p>
</td></tr>
<tr><td><code id="draw_chromatogram_+3A_legend.position">legend.position</code></td>
<td>
<p>See <code><a href="ggplot2.html#topic+theme">theme</a></code> for options of legend positions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Peaks from the are depicted as Gaussian distributions. If the data is an &quot;GCalign&quot; object that was processed with <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>, chromatograms can be drawn for the dataset prior to alignment (<strong>&quot;input&quot;</strong>), after correcting linear drift (<strong>&quot;shifted&quot;</strong>) or after the complete alignment was conducted (<strong>&quot;aligned&quot;</strong>). In the latter case, retention times refer to the mean retention time of a homologous peaks scored among samples and do not reflect any between-sample variation anymore. Depending on the range of retention times and the distance among substances the peak width can be adjusted to enable a better visual separation of peaks by changing the value of parameter <code>width</code>. Note, homologous peaks (= exactly matching retention time) will overlap completely and only the last sample plotted will be visible. Hence, the number of samples can be printed on top of each peak. The function returns a list containing the ggplot object along with the internally used data frame to allow for maximum control in adapting the plot (see examples section in this document).
</p>


<h3>Value</h3>

<p>A list containing the data frame created for plotting and the ggplot object. See <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>.
</p>


<h3>Author(s)</h3>

<p>Meinolf Ottensmann (meinolf.ottensmann@web.de) &amp; Martin Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load data
path &lt;- (system.file("extdata", "simulated_peak_data.txt", package = "GCalignR"))
## run with defaults
x &lt;- draw_chromatogram(data = path, rt_col_name = "rt")
## Customise and split samples in panels
x &lt;- draw_chromatogram(data = path, rt_col_name = "rt", samples = c("A2","A4"),
 plot = FALSE, show_num = FALSE)
x[["ggplot"]] + ggplot2::facet_wrap(~ sample, nrow = 2)
## plot without numbers
x &lt;- draw_chromatogram(data = path, show_num = FALSE, rt_col_name = "rt")
</code></pre>

<hr>
<h2 id='find_peaks'>Detect local maxima in time series</h2><span id='topic+find_peaks'></span>

<h3>Description</h3>

<p>Detects peaks in a vector and calculates the peak height. This function is only appropriate for symmetric gaussian peaks and does not take into account any baseline correction as it required in 'real word' data. Therefore, it does not substitute sophisticated peak detection and integration tools and is only used for illustration purposes in our vignettes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_peaks(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find_peaks_+3A_df">df</code></td>
<td>
<p>A data frame containing x and y coordinates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing x and y coordinates of peaks.
</p>


<h3>Author(s)</h3>

<p>Meinolf Ottensmann (meinolf.ottensmann@web.de) &amp; Martin Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## create df
df &lt;- data.frame(x = 1:1000, y = dnorm(1:1000,300,20))
## plot
with(df, plot(x,y))
## detect peak
find_peaks(df)

</code></pre>

<hr>
<h2 id='gc_heatmap'>Visualises peak alignments in form of a heatmap</h2><span id='topic+gc_heatmap'></span>

<h3>Description</h3>

<p>The goal of aligning peaks is to match homologous peaks that are thought to represent homologous substances in the same row across samples, although peaks have slightly different retention times across samples. This function makes it possible to evaluate the alignment quickly by inspecting the (i) distribution of peaks across samples, (ii) the variation for each homologous peak (column) as well as (iii) patterns that might hint at splitting peaks across rows. The mean retention time per homologous peak is here defined as the &quot;true&quot;  retention time and deviations of individual peaks can be seen by a large deviation in the retention time to the mean. Subsetting of the retention time range (i.e. selecting peaks by the mean retention time) and samples (by name or by position) allow to quickly inspect regions of interest. Two types of heatmaps are available, a binary heatmap allows to determine if the retention time of single samples deviates by more than the user defined threshold from the mean. Optionally, a discrete heatmap allows to check deviations quantitatively. Large deviation can have multiple reasons. The most likely explanation is given by the fact that adjacent rows were merged as specified by the value <code>min_diff_peak2peak</code> in <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>. Here clear cases, in which peaks of multiple samples have been grouped in either of two or more rows can be merged and cause relatively high variation in peak retention times.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gc_heatmap(
  object = NULL,
  algorithm_step = c("aligned", "shifted", "input"),
  substance_subset = NULL,
  legend_type = c("legend", "colourbar"),
  samples_subset = NULL,
  type = c("binary", "discrete"),
  threshold = NULL,
  label_size = NULL,
  show_legend = TRUE,
  main_title = NULL,
  label = c("y", "xy", "x", "none")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gc_heatmap_+3A_object">object</code></td>
<td>
<p>Object of class &quot;GCalign&quot;, the output of a call to <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>.</p>
</td></tr>
<tr><td><code id="gc_heatmap_+3A_algorithm_step">algorithm_step</code></td>
<td>
<p>Character indicating which step of the algorithm is plotted. Either &quot;input&quot;, &quot;shifted&quot; or &quot;aligned&quot; specifying the raw, linearly shifted or aligned data respectively. Default is the heatmap for the aligned dataset.</p>
</td></tr>
<tr><td><code id="gc_heatmap_+3A_substance_subset">substance_subset</code></td>
<td>
<p>A vector of integers containing indices of substances in ascending order of retention times to plot.</p>
</td></tr>
<tr><td><code id="gc_heatmap_+3A_legend_type">legend_type</code></td>
<td>
<p>A character specifying how to present deviations of retention times from the mean. Either in form of discrete steps or on a gradient scale using 'legend' or 'colourbar' respectively. Changes are only possible when <code>type = "discrete"</code></p>
</td></tr>
<tr><td><code id="gc_heatmap_+3A_samples_subset">samples_subset</code></td>
<td>
<p>A vector indicating which samples are plotted on the heatmap by giving either indices or names of samples.</p>
</td></tr>
<tr><td><code id="gc_heatmap_+3A_type">type</code></td>
<td>
<p>A character specifying whether a deviations of retention times are shown 'binary' (i.e. in comparison to the threshold value) or on a 'discrete' scale with respect to the mean retention time.</p>
</td></tr>
<tr><td><code id="gc_heatmap_+3A_threshold">threshold</code></td>
<td>
<p>A numeric value denoting the threshold above which the deviation of individual peak retention times from the mean retention time of the respective substance are highlighted in heatmaps. By default, the value of parameter <code>max_diff_peak2mean</code> (see <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>) that was used in aligning the data is used.</p>
</td></tr>
<tr><td><code id="gc_heatmap_+3A_label_size">label_size</code></td>
<td>
<p>An integer determining the size of labels on y and x axis. By default a fitting label_size is calculate (beta!) to compromise between readability and messiness due to a potentially large number of substances and samples.</p>
</td></tr>
<tr><td><code id="gc_heatmap_+3A_show_legend">show_legend</code></td>
<td>
<p>Boolean determining whether a legend is included or not.</p>
</td></tr>
<tr><td><code id="gc_heatmap_+3A_main_title">main_title</code></td>
<td>
<p>Character giving the title of the heatmap. If not specified, titles are generated automatically.</p>
</td></tr>
<tr><td><code id="gc_heatmap_+3A_label">label</code></td>
<td>
<p>Character determining if labels are shown on axes. Depending on the number of peaks and/or samples, labels are difficult to read. Use subsets instead. Possible option are &quot;xy&quot;, &quot;x&quot;, &quot;y&quot; or &quot;none&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class &quot;ggplot&quot;
</p>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com) &amp;
Meinolf Ottensmann (meinolf.ottensmann@web.de)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 ## aligned gc-dataset
 data("aligned_peak_data")
 ## Default settings: The final output is plotted
 gc_heatmap(aligned_peak_data, algorithm_step = "aligned")

 ## Plot the input data
 gc_heatmap(aligned_peak_data,algorithm_step = "input")

 ## Plot a subset of the first 50 scored substances
 gc_heatmap(aligned_peak_data,algorithm_step="aligned",substance_subset = 1:50)

 ## Plot specific samples, apply a stricter threshold
 gc_heatmap(aligned_peak_data,samples_subset = c("M2","P7","M13","P13"),threshold = 0.02)

</code></pre>

<hr>
<h2 id='linear_transformation'>Full Alignment of Peak Lists by linear retention time correction.</h2><span id='topic+linear_transformation'></span>

<h3>Description</h3>

<p>Shifts all peaks within samples to maximise the similarity to a reference sample. For optimal results, a sufficient number of shared peaks are required to find a optimal solution. A reference needs to be specified, for instance using <code><a href="#topic+choose_optimal_reference">choose_optimal_reference</a></code>. Linear shifts are evaluated within a user-defined window in discrete steps. The highest similarity score defines the shift that will be applied. If more than a single shift step yields to the same similarity score, the smallest absolute value wins in order to avoid overcompensation. The functions is envoked internally by <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linear_transformation(
  gc_peak_list,
  reference,
  max_linear_shift = 0.05,
  step_size = 0.01,
  rt_col_name,
  Logbook = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="linear_transformation_+3A_gc_peak_list">gc_peak_list</code></td>
<td>
<p>List of data.frames. Each data.frame contains GC-data
(e.g. retention time, peak area, peak height) of one sample. Variables are
stored in columns. Rows represent distinct peaks. Retention time is a
required variable.</p>
</td></tr>
<tr><td><code id="linear_transformation_+3A_reference">reference</code></td>
<td>
<p>A character giving the name of a sample included in the dataset. All samples are aligned to the reference.</p>
</td></tr>
<tr><td><code id="linear_transformation_+3A_max_linear_shift">max_linear_shift</code></td>
<td>
<p>Numeric value giving the window size considered in the full alignment. Usually, the amplitude of linear drift is small in typical GC-FID datasets. Therefore, the default value of 0.05 minutes is adequate for most datasets. Increase this value if the drift amplitude is larger.</p>
</td></tr>
<tr><td><code id="linear_transformation_+3A_step_size">step_size</code></td>
<td>
<p>Integer giving the step size in which linear shifts are evaluated between <code>max_linear_shift</code> and <code>-max_linear_shift</code>.</p>
</td></tr>
<tr><td><code id="linear_transformation_+3A_rt_col_name">rt_col_name</code></td>
<td>
<p>A character giving the name of the column containing the retention times. The decimal separator needs to be a point.</p>
</td></tr>
<tr><td><code id="linear_transformation_+3A_logbook">Logbook</code></td>
<td>
<p>A list. If present, a summary of the applied linear shifts in full alignments of peak lists is appended to the list. If not specified, a list will be created automatically.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A similarity score is calculated as the sum of deviations in retention times between all reference peaks and the closest peak in the sample. The principle idea is that the appropriate linear transformation will reduce the deviation in retention time between homologous peaks, whereas all other peaks should deviate randomly. Among all considered shifts, the minimum deviation score is selected for subsequent full alignment by shifting all peaks of the sample by the same value.
</p>


<h3>Value</h3>

<p>A list containing two items.
</p>
<table role = "presentation">
<tr><td><code>chroma_aligned</code></td>
<td>
<p>List containing the transformed data</p>
</td></tr>
<tr><td><code>Logbook</code></td>
<td>
<p>Logbook, record of the applied shifts</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com) &amp;
Meinolf Ottensmann (meinolf.ottensmann@web.de)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- peak_data[1:10]
dat &lt;- lapply(dat, function(x) x[1:50,])
x &lt;- linear_transformation(gc_peak_list = dat, reference = "C2", rt_col_name = "time")

</code></pre>

<hr>
<h2 id='merge_redundant_rows'>Merge redundant rows</h2><span id='topic+merge_redundant_rows'></span>

<h3>Description</h3>

<p>Sometimes, redundant rows (i.e. groups of resembling a homologous peak) remain in an aligned dataset. This is the case when two or more adjacent rows exhibit a difference in the mean retention time that is greater than <code>min_diff_peak2peak</code>, the parameter that determines a threshold below that redundancy is checked within <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>. Therefore, this function allows to raise the threshold for a post processing step that groups the homologous peaks together without the need of repeating a potentially time-consuming alignment with adjusted parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_redundant_rows(data, min_diff_peak2peak = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_redundant_rows_+3A_data">data</code></td>
<td>
<p>An object of class &quot;GCalign&quot;. See <code><a href="#topic+align_chromatograms">align_chromatograms</a></code> for details.</p>
</td></tr>
<tr><td><code id="merge_redundant_rows_+3A_min_diff_peak2peak">min_diff_peak2peak</code></td>
<td>
<p>A numerical giving a threshold in minutes below which rows of similar retention time are checked for redundancy.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Based on the value of parameter <code>threshold</code>, possibly redundant rows are identified by comparing mean retention times. Next, rows are checked for redundancy. When one or more samples contain peaks in a pair of compared rows, no redundancy is existent and the pair is skipped.
</p>


<h3>Value</h3>

<p>a list of two items
</p>
<table role = "presentation">
<tr><td><code>GCalign</code></td>
<td>
<p>input data with updated input to <code><a href="#topic+gc_heatmap">gc_heatmap</a></code></p>
</td></tr>
<tr><td><code>peak_list</code></td>
<td>
<p>a list of data frames containing the updated dataset</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Meinolf Ottensmann (meinolf.ottensmann@web.de) &amp; Martin Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load example dataset
data("peak_data")
## Subset for faster processing
peak_data &lt;- peak_data[1:3]
peak_data &lt;- lapply(peak_data, function(x) x[1:50,])
## align data whith strict parameters
out &lt;- align_chromatograms(peak_data, rt_col_name = "time",
max_diff_peak2mean = 0.01, min_diff_peak2peak = 0.02)
## relax threshold to merge redundant rows
out2 &lt;- merge_redundant_rows(data = out, min_diff_peak2peak = 0.05)

</code></pre>

<hr>
<h2 id='norm_peaks'>Normalisation of peak abundancies</h2><span id='topic+norm_peaks'></span>

<h3>Description</h3>

<p>Calculates the relative abundance of a peak by normalising an intensity measure with regard to the cumulative abundance of all peaks that are present within an individual sample. The desired measure of peak abundance needs to be included in a column of the input dataset aligned with <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>norm_peaks(
  data,
  conc_col_name = NULL,
  rt_col_name = NULL,
  out = c("data.frame", "list")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="norm_peaks_+3A_data">data</code></td>
<td>
<p>Object of class GCalign created with <code><a href="#topic+align_chromatograms">align_chromatograms</a></code> or a list of data frames that contain peak list of individual samples.</p>
</td></tr>
<tr><td><code id="norm_peaks_+3A_conc_col_name">conc_col_name</code></td>
<td>
<p>Character giving the name of a column in <code>data</code> containing a variable describing the abundance of peaks (e.g. peak area or peak height).</p>
</td></tr>
<tr><td><code id="norm_peaks_+3A_rt_col_name">rt_col_name</code></td>
<td>
<p>A character giving the name of the column containing the retention times. The decimal separator needs to be a point.</p>
</td></tr>
<tr><td><code id="norm_peaks_+3A_out">out</code></td>
<td>
<p>character defining the format of the returned data. Either &quot;List&quot; or &quot;data.frame&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depending on <code>out</code> either a list of data frame or a single data frame were rows represent samples and columns relative peak abundances. Abundances are given as percentages.
</p>
<p>@author Martin Stoffel (martin.adam.stoffel@gmail.com) &amp;
Meinolf Ottensmann (meinolf.ottensmann@web.de)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## aligned gc-dataset
data("aligned_peak_data")
## returns normalised peak area
norm_peaks(data = aligned_peak_data, conc_col_name = "area", rt_col_name = "time")

</code></pre>

<hr>
<h2 id='peak_data'>Gas-chromatography data for Antarctic Fur Seals (<em>Arctocephalus gazella</em>)</h2><span id='topic+peak_data'></span>

<h3>Description</h3>

<p>This is an example of a typical gas-chromatography output file, listing a number of peaks with their
respective retention times and abundance measures. Peaks were detected using Xcalibur 2.0.5 (Thermo Fisher Scientific).
The data consists of 41 mother-pup pairs of two different colonies from Bird Island, South Georgia. In addition two blanks (i.e. negative controls) are included.
</p>


<h3>Format</h3>

<p>A <code>list</code> of <code>data.frame</code>'s. Each <code>data.frame</code> contains gas-chromatography peak data of a single sample.
The variables within each <code>data.frame</code> are: &quot;time&quot; (peak retention time) and &quot;area&quot; (integral of the peak curve).
Each list element i.e. each <code>data.frame</code>is named with the unique sample ID.
</p>


<h3>Source</h3>

<p>http://www.pnas.org/content/suppl/2015/08/05/1506076112.DCSupplemental/pnas.1506076112.sd02.xlsx
</p>


<h3>References</h3>

<p>Stoffel, M.A.; Caspers, B.A.; Forcada, J.; Giannakara, A.; Baier, M.; Eberhart-Phillips, L.; Mueller, C.; Hoffman, J.I. (2015): Chemical fingerprints encode mother-offspring similarity, colony membership, relatedness, and genetic quality in fur seals. In: Proceedings of the National Academy of Sciences of the United States of America 112 (36), S. E5005-12. DOI: 10.1073/pnas.1506076112.
</p>

<hr>
<h2 id='peak_factors'>Grouping factors corresponding to gas-chromatography data of Antarctic Fur Seals (<em>Arctocephalus gazella</em>)</h2><span id='topic+peak_factors'></span>

<h3>Description</h3>

<p>List of factors corresponding to samples in <code><a href="#topic+peak_data">peak_data</a></code>
</p>


<h3>Format</h3>

<p>A data frame where columns represent factors, rows are samples.
</p>


<h3>Source</h3>

<p>http://www.pnas.org/content/suppl/2015/08/05/1506076112.DCSupplemental/pnas.1506076112.sd02.xlsx
</p>


<h3>References</h3>

<p>Stoffel, M.A.; Caspers, B.A.; Forcada, J.; Giannakara, A.; Baier, M.; Eberhart-Phillips, L.; Mueller, C.; Hoffman, J.I. (2015): Chemical fingerprints encode mother-offspring similarity, colony membership, relatedness, and genetic quality in fur seals. In: Proceedings of the National Academy of Sciences of the United States of America 112 (36), S. E5005-12. DOI: 10.1073/pnas.1506076112.
</p>

<hr>
<h2 id='peak_interspace'>Estimate the observed space between peaks within chromatograms</h2><span id='topic+peak_interspace'></span>

<h3>Description</h3>

<p>The parameter <code>min_diff_peak2peak</code> is a major determinant in the alignment of a dataset with <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>. This function helps to infer a suitable value based on the input data. The underlying assumption here is that distinct peaks within a separated by a larger gap than homologous peaks across samples. Tightly spaced peaks within a sample will appear on the left side of the plotted distribution and can indicate the presence of split peaks in the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peak_interspace(
  data,
  rt_col_name = NULL,
  sep = "\t",
  quantiles = NULL,
  quantile_range = c(0, 1),
  by_sample = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="peak_interspace_+3A_data">data</code></td>
<td>
<p>Dataset containing peaks that need to be aligned and matched. For every peak a arbitrary number of numerical variables can be included (e.g. peak height, peak area) in addition to the mandatory retention time. The standard format is a tab-delimited text file according to the following layout: (1) The first row contains sample names, the (2) second row column names of the corresponding peak lists. Starting with the third row, peak lists are included for every sample that needs to be incorporated in the dataset. Here, a peak list contains data for individual peaks in rows, whereas columns specify variables in the order given in the second row of the text file. Peak lists of individual samples are concatenated horizontally and need to be of the same width (i.e. the same number of columns in consistent order). Alternatively, the input may be a list of data frames. Each data frame contains the peak data for a single individual. Variables (i.e.columns) are named consistently across data frames. The names of elements in the list are used as sample identifiers. Cells may be filled with numeric or integer values but no factors or characters are allowed. NA and 0 may be used to indicate empty rows.</p>
</td></tr>
<tr><td><code id="peak_interspace_+3A_rt_col_name">rt_col_name</code></td>
<td>
<p>A character giving the name of the column containing the retention times. The decimal separator needs to be a point.</p>
</td></tr>
<tr><td><code id="peak_interspace_+3A_sep">sep</code></td>
<td>
<p>The field separator character. The default is tab separated (<code>sep = '\t'</code>).
See the &quot;sep&quot; argument in <code><a href="utils.html#topic+read.table">read.table</a></code> for details.</p>
</td></tr>
<tr><td><code id="peak_interspace_+3A_quantiles">quantiles</code></td>
<td>
<p>A numeric vector. Specified quantiles are calculated from the distribution.</p>
</td></tr>
<tr><td><code id="peak_interspace_+3A_quantile_range">quantile_range</code></td>
<td>
<p>A numeric vector of length two that allows to subset an arbitrary interquartile range.</p>
</td></tr>
<tr><td><code id="peak_interspace_+3A_by_sample">by_sample</code></td>
<td>
<p>A logical that allows to calculate peak interspaces individually for each sample. By default all samples are combined to give the global distribution of next-peak differences in retention times. When <code>by_sample = TRUE</code>, a series of plots (one for each sample) is created and a keystroke is required to proceed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing summary statistics of the peak interspace distribution
</p>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com) &amp; Meinolf Ottensmann (meinolf.ottensmann@web.de)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## plotting with defaults
peak_interspace(data = peak_data, rt_col_name = "time")
## plotting up to the 0.95 quantile
peak_interspace(data = peak_data,rt_col_name = "time",quantile_range = c(0,0.95))
## return the 0.1 quantile
peak_interspace(data = peak_data,rt_col_name = "time", quantiles = 0.1)

</code></pre>

<hr>
<h2 id='plot.GCalign'>Plot diagnostics for an GCalign Object</h2><span id='topic+plot.GCalign'></span>

<h3>Description</h3>

<p>Visualises the aligned data based on four diagnostic plots. One plot shows the distribution of peak numbers per sample in the raw data and after alignment. A second plot gives the distribution of linear shifts that were applied in order to conduct a full alignment of samples with respect to reference. A third sample gives a distribution of the variation in retention times of homologous peaks. The fourth plot shows a frequency distribution of peaks shared among samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GCalign'
plot(
  x,
  which_plot = c("all", "shifts", "variation", "peak_numbers", "peaks_shared"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.GCalign_+3A_x">x</code></td>
<td>
<p>Object of class GCalign, created with <code><a href="#topic+align_chromatograms">align_chromatograms</a></code></p>
</td></tr>
<tr><td><code id="plot.GCalign_+3A_which_plot">which_plot</code></td>
<td>
<p>A character defining which plot is created. Options are &quot;shifts&quot;, &quot;variation&quot;, &quot;peak_numbers&quot; and &quot;peaks_shared&quot;. By default all four are created.</p>
</td></tr>
<tr><td><code id="plot.GCalign_+3A_...">...</code></td>
<td>
<p>Optional arguments passed on to methods. See
<code><a href="graphics.html#topic+graphics">plot</a></code>, <code><a href="graphics.html#topic+graphics">hist</a></code> and <code><a href="graphics.html#topic+graphics">barplot</a></code>. Note that optional arguments are not passed on when plotting all figures.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depending on the selected plot a data frame containing the data source of the respective plot is returned. If all plots are created, no output will be returned.
</p>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com) &amp; Meinolf Ottensmann
(meinolf.ottensmann@web.de)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## GCalign object
data("aligned_peak_data")

## All plots are shown by default
plot(aligned_peak_data)

## Distribution of peak numbers
plot(aligned_peak_data, which_plot = "peak_numbers")

## variation of retention times
plot(aligned_peak_data, which_plot = "variation")

</code></pre>

<hr>
<h2 id='print.GCalign'>Summarising Peak Alignments with GCalignR</h2><span id='topic+print.GCalign'></span>

<h3>Description</h3>

<p>print method for class &quot;GCalign&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GCalign'
print(x, write_text_file = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.GCalign_+3A_x">x</code></td>
<td>
<p>Object of class GCalign, created with <code><a href="#topic+align_chromatograms">align_chromatograms</a></code></p>
</td></tr>
<tr><td><code id="print.GCalign_+3A_write_text_file">write_text_file</code></td>
<td>
<p>A boolean allowing to write a text file.</p>
</td></tr>
<tr><td><code id="print.GCalign_+3A_...">...</code></td>
<td>
<p>Optional arguments passed on to methods are currently not supported.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com) &amp; Meinolf Ottensmann
(meinolf.ottensmann@web.de)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## GCalign object
data("aligned_peak_data")
## print summary
print(aligned_peak_data)

</code></pre>

<hr>
<h2 id='read_empower2'>Import data from single EMPOWER2 HPLC files</h2><span id='topic+read_empower2'></span>

<h3>Description</h3>

<p>reads output files of the EMPOWER 2 SOFTWARE (Waters). Input files must contain data of single samples deposited within the same directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_empower2(
  path = NULL,
  pattern = ".txt",
  sep = "\t",
  skip = 2,
  id = "SampleName"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_empower2_+3A_path">path</code></td>
<td>
<p>path to a folder containing input files</p>
</td></tr>
<tr><td><code id="read_empower2_+3A_pattern">pattern</code></td>
<td>
<p>pattern used to select files. By default &quot;.txt&quot;</p>
</td></tr>
<tr><td><code id="read_empower2_+3A_sep">sep</code></td>
<td>
<p>The field separator character. The default is tab separated (<code>sep = '\t'</code>).
See the &quot;sep&quot; argument in <code><a href="utils.html#topic+read.table">read.table</a></code> for details.</p>
</td></tr>
<tr><td><code id="read_empower2_+3A_skip">skip</code></td>
<td>
<p>rows to skip before reading data</p>
</td></tr>
<tr><td><code id="read_empower2_+3A_id">id</code></td>
<td>
<p>column containing sample name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of data frames (each corresponding to a sample)
</p>

<hr>
<h2 id='read_peak_list'>Read content of a text file and convert it to a list</h2><span id='topic+read_peak_list'></span>

<h3>Description</h3>

<p>Reads the content of text file that is formatted as described in <code><a href="#topic+align_chromatograms">align_chromatograms</a></code> and converts it to a list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_peak_list(data, sep = "\t", rt_col_name, check = T)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_peak_list_+3A_data">data</code></td>
<td>
<p>A text file containing a peak list. See <code><a href="#topic+align_chromatograms">align_chromatograms</a></code> for details.</p>
</td></tr>
<tr><td><code id="read_peak_list_+3A_sep">sep</code></td>
<td>
<p>The field separator character. The default is tab separated (<code>sep = '\t'</code>).
See the &quot;sep&quot; argument in <code><a href="utils.html#topic+read.table">read.table</a></code> for details.</p>
</td></tr>
<tr><td><code id="read_peak_list_+3A_rt_col_name">rt_col_name</code></td>
<td>
<p>A character giving the name of the column containing the retention times. The decimal separator needs to be a point.</p>
</td></tr>
<tr><td><code id="read_peak_list_+3A_check">check</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of data frames containing peak data for every sample of <code>data</code>.
</p>


<h3>Author(s)</h3>

<p>Meinolf Ottensmann (meinolf.ottensmann@web.de) &amp; Martin Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "simulated_peak_data.txt", package = "GCalignR")
x &lt;- read_peak_list(data = path, rt_col_name = "rt")

</code></pre>

<hr>
<h2 id='remove_blanks'>Remove peaks present in negative control samples</h2><span id='topic+remove_blanks'></span>

<h3>Description</h3>

<p>Removes peaks that are present in blanks (i.e. negative control samples) to eliminate contaminations in the aligned data. Afterwards, blanks are deleted itself. This function is only applicable when blanks were not discarded during a previous alignment using <code><a href="#topic+align_chromatograms">align_chromatograms</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_blanks(data, blanks)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="remove_blanks_+3A_data">data</code></td>
<td>
<p>An object of class &quot;GCalign&quot;. See <code><a href="#topic+align_chromatograms">align_chromatograms</a></code> for details. Alternatively, a list of data frames. Whereby each data frame contains the peak list for an individual sample.</p>
</td></tr>
<tr><td><code id="remove_blanks_+3A_blanks">blanks</code></td>
<td>
<p>Character vector of names of negative controls. Substances found in any of the blanks will be removed from the aligned dataset, before the blanks are deleted from the aligned data as well. This is an optional filtering step.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of data frames for each individual.
</p>


<h3>Author(s)</h3>

<p>Meinolf Ottensmann (meinolf.ottensmann@web.de) &amp; Martin Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("peak_data")
## subset for faster processing
data &lt;- lapply(peak_data[1:5], function(x) x[20:35,])
x &lt;- align_chromatograms(data, rt_col_name = "time")
out &lt;- remove_blanks(data = x, blanks = c("C2","C3"))
## number of deleted peaks
nrow(x[["aligned_list"]][["M2"]]) - nrow(out[["M2"]])

</code></pre>

<hr>
<h2 id='remove_singletons'>Remove singletons</h2><span id='topic+remove_singletons'></span>

<h3>Description</h3>

<p>Identifies and removes singletons (i.e. peaks that are unique for one sample) from the aligned dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_singletons(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="remove_singletons_+3A_data">data</code></td>
<td>
<p>An object of class &quot;GCalign&quot;. See <code><a href="#topic+align_chromatograms">align_chromatograms</a></code> for details. Alternatively, a list of data frames. Whereby each data frame contains the peak list for an individual sample.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of data frames for each individual.
</p>


<h3>Author(s)</h3>

<p>Meinolf Ottensmann (meinolf.ottensmann@web.de) &amp; Martin Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("peak_data")
## subset for faster processing
data &lt;- lapply(peak_data[1:5], function(x) x[20:35,])
x &lt;- align_chromatograms(data, rt_col_name = "time")
out &lt;- remove_singletons(data = x)

</code></pre>

<hr>
<h2 id='simple_chroma'>Simulate simple chromatograms</h2><span id='topic+simple_chroma'></span>

<h3>Description</h3>

<p>Creates chromatograms with user defined peaks for illustrative purposes. Linear drift is applied in sample order if more than one sample is created. See parameters of the function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simple_chroma(
  peaks = c(10, 13, 25, 37, 50),
  N = 1,
  min = 0,
  max = 30,
  Names = NULL,
  sd = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simple_chroma_+3A_peaks">peaks</code></td>
<td>
<p>A numeric vector giving the retention times on which gaussian distribution, defining peaks, are centered. If more than one sample is generated <code>N &gt; 1</code>, <code>peaks</code> defines a population of peaks, from which samples are generated.</p>
</td></tr>
<tr><td><code id="simple_chroma_+3A_n">N</code></td>
<td>
<p>An integer giving the number of chromatograms to create. By default <code>N = 1</code>.</p>
</td></tr>
<tr><td><code id="simple_chroma_+3A_min">min</code></td>
<td>
<p>A numeric giving the minimum retention time.</p>
</td></tr>
<tr><td><code id="simple_chroma_+3A_max">max</code></td>
<td>
<p>A numeric giving the maximum retention time.</p>
</td></tr>
<tr><td><code id="simple_chroma_+3A_names">Names</code></td>
<td>
<p>A character vector giving sample names. If not specified, names are generated automatically.</p>
</td></tr>
<tr><td><code id="simple_chroma_+3A_sd">sd</code></td>
<td>
<p>A numeric vector of the same length as peaks giving the standard deviation of each peak. Only supported if N = 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing x and y coordinates and corresponding sample names.
</p>


<h3>Author(s)</h3>

<p>Meinolf Ottensmann (meinolf.ottensmann@web.de) &amp; Martin Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## create a chromatogram
x &lt;- simple_chroma(peaks = c(5,10,15), N = 1, min = 0, max = 30, Names = "MyChroma")
## plot chromatogram
with(x, plot(x,y, xlab = "time", ylab = "intensity"))

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
