<!DOCTYPE html><html lang="en"><head><title>Help for package GCPBayes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GCPBayes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CS'><p>Continuous Spike</p></a></li>
<li><a href='#DNAJC1'><p>Gene DNAJC1 from BCAC and Epithyr  studies</p></a></li>
<li><a href='#DS'><p>Dirac Spike</p></a></li>
<li><a href='#e2_Monte_Carlo_EM'><p>Internal: e2_Monte_Carlo_EM</p></a></li>
<li><a href='#GCPBayes'><p>GCPBayes Package</p></a></li>
<li><a href='#HS'><p>Hierarchical Spike</p></a></li>
<li><a href='#MCMCplot'><p>MCMC plot</p></a></li>
<li><a href='#PARP2'><p>Gene PARP2 from CECILE study</p></a></li>
<li><a href='#PARP2_summary'><p>Summary statistics of gene PARP2 from CECILE study</p></a></li>
<li><a href='#Simulated_individual'><p>Simulated individual level data</p></a></li>
<li><a href='#Simulated_individual_survival'><p>Simulated individual level survival data</p></a></li>
<li><a href='#Simulated_summary'><p>Simulated summary statistics for K=5 traits</p></a></li>
<li><a href='#summaryCS'><p>Summary function of Continuous Spike</p></a></li>
<li><a href='#summaryDS'><p>Summary function of Dirac Spike</p></a></li>
<li><a href='#summaryHS'><p>Summary function of Hierarchical Spike</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Meta-Analysis of Pleiotropic Effects Using Group
Structure</td>
</tr>
<tr>
<td>Version:</td>
<td>4.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Run a Gibbs sampler for a multivariate Bayesian sparse group selection model with Dirac, continuous and hierarchical spike prior for detecting pleiotropy on the traits. This package is designed for summary statistics containing estimated regression coefficients and its estimated covariance matrix. The methodology is available from: Baghfalaki, T., Sugier, P. E., Truong, T., Pettitt, A. N., Mengersen, K., &amp; Liquet, B. (2021) &lt;<a href="https://doi.org/10.1002%2Fsim.8855">doi:10.1002/sim.8855</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/tbaghfalaki/GCPBayes">https://github.com/tbaghfalaki/GCPBayes</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2.0)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, mvtnorm, invgamma, gdata, truncnorm, postpack, wiqid,
Rcpp (&ge; 1.0.9)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-14 13:05:55 UTC; y_asgari</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-14 21:40:02 UTC</td>
</tr>
<tr>
<td>Author:</td>
<td>Yazdan Asgari [aut, cre],
  Taban Baghfalaki [aut],
  Benoit Liquet [aut],
  Pierre-Emmanuel Sugier [aut],
  Mohammed Sedki [aut],
  Therese Truong [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yazdan Asgari &lt;yazdan.asgari@inserm.fr&gt;</td>
</tr>
</table>
<hr>
<h2 id='CS'>Continuous Spike</h2><span id='topic+CS'></span>

<h3>Description</h3>

<p>Run a Gibbs sampler for a multivariate Bayesian sparse group selection model with Continuous spike prior for detecting pleiotropic effects on the traits. This function is designed for summary statistics containing estimated regression coefficients and their estimated covariance matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CS(
  Betah,
  Sigmah,
  kappa0,
  tau20,
  zeta0,
  m,
  K,
  niter = 1000,
  burnin = 500,
  nthin = 2,
  nchains = 2,
  a1 = a1,
  a2 = a2,
  c1 = c1,
  c2 = c2,
  sigma2 = 10^-3,
  snpnames = snpnames,
  genename = genename
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CS_+3A_betah">Betah</code></td>
<td>
<p>A list containing m-dimensional vectors of the regression coefficients for K studies.</p>
</td></tr>
<tr><td><code id="CS_+3A_sigmah">Sigmah</code></td>
<td>
<p>A list containing the positive definite covariance matrices (m*m-dimensional) which is the estimated covariance matrices of K studies.</p>
</td></tr>
<tr><td><code id="CS_+3A_kappa0">kappa0</code></td>
<td>
<p>Initial value for kappa (its dimension is equal to nchains).</p>
</td></tr>
<tr><td><code id="CS_+3A_tau20">tau20</code></td>
<td>
<p>Initial value for tau2 (its dimension is equal to nchains).</p>
</td></tr>
<tr><td><code id="CS_+3A_zeta0">zeta0</code></td>
<td>
<p>Initial value for zeta.</p>
</td></tr>
<tr><td><code id="CS_+3A_m">m</code></td>
<td>
<p>Number of variables in the group.</p>
</td></tr>
<tr><td><code id="CS_+3A_k">K</code></td>
<td>
<p>Number of traits.</p>
</td></tr>
<tr><td><code id="CS_+3A_niter">niter</code></td>
<td>
<p>Number of iterations for the Gibbs sampler.</p>
</td></tr>
<tr><td><code id="CS_+3A_burnin">burnin</code></td>
<td>
<p>Number of burn-in iterations.</p>
</td></tr>
<tr><td><code id="CS_+3A_nthin">nthin</code></td>
<td>
<p>The lag of the iterations used for the posterior analysis is defined (or thinning rate).</p>
</td></tr>
<tr><td><code id="CS_+3A_nchains">nchains</code></td>
<td>
<p>Number of Markov chains, when nchains&gt;1, the function calculates the Gelman-Rubin convergence statistic, as modified by Brooks and Gelman (1998).</p>
</td></tr>
<tr><td><code id="CS_+3A_a1">a1</code>, <code id="CS_+3A_a2">a2</code></td>
<td>
<p>Hyperparameters of kappa. Default is a1=0.1 and a2=0.1.</p>
</td></tr>
<tr><td><code id="CS_+3A_c1">c1</code>, <code id="CS_+3A_c2">c2</code></td>
<td>
<p>Hyperparameters of tau2. Default is c1=0.1 and c2=0.1.</p>
</td></tr>
<tr><td><code id="CS_+3A_sigma2">sigma2</code></td>
<td>
<p>Variance of spike (multivariate normal distribution with a diagonal covariance matrix with small variance) representing the null effect distribution. Default is 10^-3.</p>
</td></tr>
<tr><td><code id="CS_+3A_snpnames">snpnames</code></td>
<td>
<p>Names of variables for the group.</p>
</td></tr>
<tr><td><code id="CS_+3A_genename">genename</code></td>
<td>
<p>Name of group.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let betah_k, k=1,...,K be a m-dimensional vector of the regression coefficients for the kth study and Sigmah_k  be its estimated covariance matrix. The hierarchical set-up of  CS prior, by considering summary statistics (betah_k and Sigmah_k, k=1,...,K) as the input of the method, is given by:
</p>
<p>betah_k ~ (1 - zeta_k) N_m(0,sigma2 I_m) + zeta_k N_m(0,tau2 I_m ),
</p>
<p>zeta_k ~  Ber(kappa),
</p>
<p>kappa ~ Beta(a_1,a_2),
</p>
<p>tau2 ~ inverseGamma (c_1,c_2).
</p>


<h3>Value</h3>


<ul>
<li><p> mcmcchain: The list of simulation output for all parameters.
</p>
</li>
<li><p> Summary: Summary statistics for regression coefficients in each study.
</p>
</li>
<li><p> Criteria: genename, snpnames, PPA, log10BF, lBFDR, theta.
</p>
</li>
<li><p> Indicator: A table containing m rows of binary indicators for each study, the number of studies with nonzero signal and having pleiotropic effect by credible interval (CI). The first K columns show nonzero signals, K+1 th column includes the number of studies with nonzero signal and the last column shows an indicator for having pleiotropic effect of each SNP.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Taban Baghfalaki.
</p>


<h3>References</h3>

<p>Baghfalaki, T., Sugier, P. E., Truong, T., Pettitt, A. N., Mengersen, K., &amp; Liquet, B. (2021). Bayesian meta analysis models for cross cancer genomic investigation of pleiotropic effects using group structure. Statistics in Medicine, 40(6), 1498-1518.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>############################# Gene DNAJC1 ###############################################
data(DNAJC1)
Breast &lt;- DNAJC1$Breast
Thyroid &lt;- DNAJC1$Thyroid
genename &lt;- "DNAJC1"
snpnames &lt;- Breast$snp
Betah &lt;- list(Breast$beta, Thyroid$beta)
Sigmah &lt;- list(diag(Breast$se^2), diag(Thyroid$se^2))
K &lt;- 2
m &lt;- 14

pvalue &lt;- matrix(0, K, m)
for (k in 1:K) {
  pvalue[k, ] &lt;- 2 * pnorm(-abs(Betah[[k]] / sqrt(diag(Sigmah[[k]]))))
}

zinit &lt;- rep(0, K)
for (j in 1:K) {
  index &lt;- 1:m
  PVALUE &lt;- p.adjust(pvalue[j, ])
  SIGNALS &lt;- index[PVALUE &lt; 0.05]
  modelf1 &lt;- rep(0, m)
  modelf1[SIGNALS] &lt;- 1
  if (max(modelf1) == 1) (zinit[j] &lt;- 1)
}


RES &lt;- CS(Betah, Sigmah,
  kappa0 = 0.5, tau20 = 1, zeta0 = zinit,
  m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1, a1 = 0.1, a2 = 0.1,
  c1 = 0.1, c2 = 0.1, sigma2 = 10^-3, snpnames = snpnames, genename = genename
)
## Not run: 
  print(RES)


  RES1 &lt;- CS(Betah, Sigmah,
    kappa0 = c(0.2, 0.5), tau20 = c(1, 2), zeta0 = zinit,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, c1 = 0.1, c2 = 0.1, sigma2 = 10^-3, snpnames, genename
  )
  print(RES1)
  ################### Simulated summary level data with K=5 ###############################
  data(Simulated_summary)
  genename &lt;- Simulated_summary$genename
  snpnames &lt;- Simulated_summary$snpnames
  Betah &lt;- Simulated_summary$simBeta
  Sigmah &lt;- Simulated_summary$simSIGMA
  K &lt;- 5
  m &lt;- 10

  pvalue &lt;- matrix(0, K, m)
  for (k in 1:K) {
    pvalue[k, ] &lt;- 2 * pnorm(-abs(Betah[[k]] / sqrt(diag(Sigmah[[k]]))))
  }

  zinit &lt;- rep(0, K)
  for (j in 1:K) {
    index &lt;- 1:m
    PVALUE &lt;- p.adjust(pvalue[j, ])
    SIGNALS &lt;- index[PVALUE &lt; 0.05]
    modelf1 &lt;- rep(0, m)
    modelf1[SIGNALS] &lt;- 1
    if (max(modelf1) == 1) (zinit[j] &lt;- 1)
  }


  RES &lt;- CS(Betah, Sigmah,
    kappa0 = 0.5, tau20 = 1, zeta0 = zinit,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1, a1 = 0.1, a2 = 0.1,
    c1 = 0.1, c2 = 0.1, sigma2 = 10^-3, snpnames = snpnames, genename = genename
  )
  print(RES)


  RES1 &lt;- CS(Betah, Sigmah,
    kappa0 = c(0.2, 0.5), tau20 = c(1, 2), zeta0 = zinit,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, c1 = 0.1, c2 = 0.1, sigma2 = 10^-3, snpnames, genename
  )
  print(RES1)
  ################################### Gene PARP2 ##########################################
  library(BhGLM)
  data(PARP2)
  Breast &lt;- PARP2$Breast
  Thyroid &lt;- PARP2$Thyroid
  genename &lt;- "PARP2"
  snpnames &lt;- c("rs3093872", "rs3093921", "rs1713411", "rs3093926", "rs3093930", "rs878156")


  Fit1 &lt;- BhGLM::bglm(y1 ~ ., family = binomial(link = "logit"), data = Breast)
  Betah1 &lt;- Fit1$coefficients[-1]
  Sigmah1 &lt;- cov(coef(arm::sim(Fit1)))[-1, -1]

  Fit2 &lt;- BhGLM::bglm(y2 ~ ., family = binomial(link = "logit"), data = Thyroid)
  Betah2 &lt;- Fit2$coefficients[-1]
  Sigmah2 &lt;- cov(coef(arm::sim(Fit2)))[-1, -1]

  Betah &lt;- list(Betah1, Betah2)
  Sigmah &lt;- list(Sigmah1, Sigmah2)
  K &lt;- 2
  m &lt;- 6

  pvalue &lt;- matrix(0, K, m)
  for (k in 1:K) {
    pvalue[k, ] &lt;- 2 * pnorm(-abs(Betah[[k]] / sqrt(diag(Sigmah[[k]]))))
  }

  zinit &lt;- rep(0, K)
  for (j in 1:K) {
    index &lt;- 1:m
    PVALUE &lt;- p.adjust(pvalue[j, ])
    SIGNALS &lt;- index[PVALUE &lt; 0.05]
    modelf1 &lt;- rep(0, m)
    modelf1[SIGNALS] &lt;- 1
    if (max(modelf1) == 1) (zinit[j] &lt;- 1)
  }


  RES &lt;- CS(Betah, Sigmah,
    kappa0 = 0.5, tau20 = 1, zeta0 = zinit,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1, a1 = 0.1, a2 = 0.1,
    c1 = 0.1, c2 = 0.1, sigma2 = 10^-3, snpnames = snpnames, genename = genename
  )
  print(RES)

  RES1 &lt;- CS(Betah, Sigmah,
    kappa0 = c(0.2, 0.5), tau20 = c(1, 2), zeta0 = zinit,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, c1 = 0.1, c2 = 0.1, sigma2 = 10^-3, snpnames, genename
  )
  print(RES1)
  ########### Simulated individual level data with K=3 and continuous phynotype ###########
  library(BhGLM)
  data(Simulated_individual)
  Study1 &lt;- Simulated_individual$Study1
  Study2 &lt;- Simulated_individual$Study2
  Study3 &lt;- Simulated_individual$Study3
  K &lt;- 3
  m &lt;- 30
  genename &lt;- "Simulated"
  snpnames &lt;- sprintf("SNP%s", seq(1:m))


  Fit1 &lt;- BhGLM::bglm(Y1 ~ ., family = gaussian, data = data.frame(Study1))
  Betah1 &lt;- Fit1$coefficients[-1]
  Sigmah1 &lt;- cov(coef(arm::sim(Fit1)))[-1, -1]

  Fit2 &lt;- BhGLM::bglm(Y2 ~ ., family = gaussian, data = data.frame(Study2))
  Betah2 &lt;- Fit2$coefficients[-1]
  Sigmah2 &lt;- cov(coef(arm::sim(Fit2)))[-1, -1]

  Fit3 &lt;- BhGLM::bglm(Y3 ~ ., family = gaussian, data = data.frame(Study3))
  Betah3 &lt;- Fit3$coefficients[-1]
  Sigmah3 &lt;- cov(coef(arm::sim(Fit3)))[-1, -1]

  Betah &lt;- list(Betah1, Betah2, Betah3)
  Sigmah &lt;- list(Sigmah1, Sigmah2, Sigmah3)


  pvalue &lt;- matrix(0, K, m)
  for (k in 1:K) {
    pvalue[k, ] &lt;- 2 * pnorm(-abs(Betah[[k]] / sqrt(diag(Sigmah[[k]]))))
  }

  zinit &lt;- rep(0, K)
  for (j in 1:K) {
    index &lt;- 1:m
    PVALUE &lt;- p.adjust(pvalue[j, ])
    SIGNALS &lt;- index[PVALUE &lt; 0.05]
    modelf1 &lt;- rep(0, m)
    modelf1[SIGNALS] &lt;- 1
    if (max(modelf1) == 1) (zinit[j] &lt;- 1)
  }

  RES &lt;- CS(Betah, Sigmah,
    kappa0 = 0.5, tau20 = 1, zeta0 = zinit,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1, a1 = 0.1, a2 = 0.1,
    c1 = 0.1, c2 = 0.1, sigma2 = 10^-3, snpnames = snpnames, genename = genename
  )
  print(RES)

  RES1 &lt;- CS(Betah, Sigmah,
    kappa0 = c(0.2, 0.5), tau20 = c(1, 2), zeta0 = zinit,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, c1 = 0.1, c2 = 0.1, sigma2 = 10^-3, snpnames, genename
  )
  print(RES1)

  ########### Simulated individual level data with K=2 and gene expression data ###########
  library(BhGLM)
  data(Simulated_individual_survival)
  Study1 &lt;- Simulated_individual_survival$Study1
  Study2 &lt;- Simulated_individual_survival$Study2
  K &lt;- 2
  m &lt;- 10
  genename &lt;- "Simulated"
  snpnames &lt;- sprintf("G%s", seq(1:m))


  Fit1 &lt;- BhGLM::bcoxph(Study1$T ~ Study1$X)
  Betah1 &lt;- Fit1$coefficients
  Sigmah1 &lt;- Fit1$var


  Fit2 &lt;- BhGLM::bcoxph(Study2$T ~ Study2$X)
  Betah2 &lt;- Fit2$coefficients
  Sigmah2 &lt;- Fit2$var

  Betah &lt;- list(Betah1, Betah2)
  Sigmah &lt;- list(Sigmah1, Sigmah2)

  pvalue &lt;- matrix(0, K, m)
  for (k in 1:K) {
    pvalue[k, ] &lt;- 2 * pnorm(-abs(Betah[[k]] / sqrt(diag(Sigmah[[k]]))))
  }

  zinit &lt;- rep(0, K)
  for (j in 1:K) {
    index &lt;- 1:m
    PVALUE &lt;- p.adjust(pvalue[j, ])
    SIGNALS &lt;- index[PVALUE &lt; 0.05]
    modelf1 &lt;- rep(0, m)
    modelf1[SIGNALS] &lt;- 1
    if (max(modelf1) == 1) (zinit[j] &lt;- 1)
  }


  RES1 &lt;- CS(Betah, Sigmah,
    kappa0 = c(0.2, 0.5), tau20 = c(1, 2), zeta0 = zinit,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, c1 = 0.1, c2 = 0.1, sigma2 = 10^-3, snpnames, genename
  )
  print(RES1)

## End(Not run)
</code></pre>

<hr>
<h2 id='DNAJC1'>Gene DNAJC1 from BCAC and Epithyr  studies</h2><span id='topic+DNAJC1'></span>

<h3>Description</h3>

<p>The summary statistics data for DNAJC1 protein coding gene including beta and standard
error for pleiotropy investigation of breast and thyroid cancers.
The summary statistics of breast and thyroid  cancer are extracted from BCAC and Epithyr  (Baghgalaki et al., 2021b) studies, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DNAJC1
</code></pre>


<h3>Format</h3>

<p>A list which contains two matrices for the summary statistics of each study.
</p>

<dl>
<dt>Breast</dt><dd><p>Summary statitics of breast cancer including the name of SNPs, beta and se</p>
</dd>
<dt>Thyroid</dt><dd><p>Summary statitics of thyroid cancer including the name of SNPs, beta and se</p>
</dd>
</dl>



<h3>References</h3>


<ol>
<li>
<p>Baghfalaki, T., Sugier, Y. Asgari, P. E., Truong, &amp; Liquet, B. (2021). GCPBayes:  An R Package for Studying Cross-Phenotype Genetic Associations with Group-Level Bayesian Meta-Analysis. <em>Submitted</em>.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+GCPBayes">GCPBayes</a></code>
</p>

<hr>
<h2 id='DS'>Dirac Spike</h2><span id='topic+DS'></span>

<h3>Description</h3>

<p>Run a Gibbs sampler for a multivariate Bayesian sparse group selection model with Dirac spike prior for detecting pleiotropic effects on the traits. This function is designed for summary statistics containing estimated regression coefficients and their estimated covariance matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DS(
  Betah,
  Sigmah,
  kappa0,
  sigma20,
  m,
  K,
  niter = 1000,
  burnin = 500,
  nthin = 2,
  nchains = 2,
  a1 = 0.1,
  a2 = 0.1,
  d1 = 0.1,
  d2 = 0.1,
  snpnames,
  genename
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DS_+3A_betah">Betah</code></td>
<td>
<p>A list containing m-dimensional vectors of the regression coefficients for K studies.</p>
</td></tr>
<tr><td><code id="DS_+3A_sigmah">Sigmah</code></td>
<td>
<p>A list containing the positive definite covariance matrices (m*m-dimensional) which is the estimated covariance matrices of K studies.</p>
</td></tr>
<tr><td><code id="DS_+3A_kappa0">kappa0</code></td>
<td>
<p>Initial value for kappa (its dimension is equal to nchains).</p>
</td></tr>
<tr><td><code id="DS_+3A_sigma20">sigma20</code></td>
<td>
<p>Initial value for sigma2 (its dimension is equal to nchains).</p>
</td></tr>
<tr><td><code id="DS_+3A_m">m</code></td>
<td>
<p>Number of variables in the group.</p>
</td></tr>
<tr><td><code id="DS_+3A_k">K</code></td>
<td>
<p>Number of traits.</p>
</td></tr>
<tr><td><code id="DS_+3A_niter">niter</code></td>
<td>
<p>Number of iterations for the Gibbs sampler.</p>
</td></tr>
<tr><td><code id="DS_+3A_burnin">burnin</code></td>
<td>
<p>Number of burn-in iterations.</p>
</td></tr>
<tr><td><code id="DS_+3A_nthin">nthin</code></td>
<td>
<p>The lag of the iterations used for the posterior analysis is defined (or thinning rate).</p>
</td></tr>
<tr><td><code id="DS_+3A_nchains">nchains</code></td>
<td>
<p>Number of Markov chains, when nchains&gt;1, the function calculates the Gelman-Rubin convergence statistic, as modified by Brooks and Gelman (1998).</p>
</td></tr>
<tr><td><code id="DS_+3A_a1">a1</code>, <code id="DS_+3A_a2">a2</code></td>
<td>
<p>Hyperparameters of kappa. Default is a1=0.1 and a2=0.1.</p>
</td></tr>
<tr><td><code id="DS_+3A_d1">d1</code>, <code id="DS_+3A_d2">d2</code></td>
<td>
<p>Hyperparameters of sigma2. Default is d1=0.1 and d2=0.1.</p>
</td></tr>
<tr><td><code id="DS_+3A_snpnames">snpnames</code></td>
<td>
<p>Names of variables for the group.</p>
</td></tr>
<tr><td><code id="DS_+3A_genename">genename</code></td>
<td>
<p>Name of group.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let betah_k, k=1,...,K be a m-dimensional vector of the regression coefficients for the kth study and Sigmah_k  be its estimated covariance matrix. The hierarchical set-up of  DS prior, by considering summary statistics (betah_k and Sigmah_k, k=1,...,K) as the input of the method, is given by:
</p>
<p>betah _k ~ (1 - kappa) delta_0(betah_k) + kappa N_m(0,sigma2 I_m ),
</p>
<p>kappa ~ Beta(a_1,a_2),
</p>
<p>sigma2 ~ inverseGamma (d_1,d_2).
</p>
<p>where delta_0(betah_k) denotes a point mass at 0, such that delta_0(betah_k)=1 if beta_k=0 and  delta_0(betah_k)=0 if  at least one of the $m$ components of beta_k is non-zero.
</p>


<h3>Value</h3>


<ul>
<li><p> mcmcchain: The list of simulation output for all parameters.
</p>
</li>
<li><p> Summary: Summary statistics for regression coefficients in each study.
</p>
</li>
<li><p> Criteria: genename, snpnames, PPA, log10BF, lBFDR, theta.
</p>
</li>
<li><p> Indicator: A table containing m rows of binary indicators for each study, the number of studies with nonzero signal and having pleiotropic effect by credible interval (CI). The first K columns show nonzero signals, K+1 th column includes the number of studies with nonzero signal and the last column shows an indicator for having pleiotropic effect of each SNP.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Taban Baghfalaki.
</p>


<h3>References</h3>


<ol>
<li>
<p>Baghfalaki, T., Sugier, P. E., Truong, T., Pettitt, A. N., Mengersen, K., &amp; Liquet, B. (2021). Bayesian meta analysis models for cross cancer genomic investigation of pleiotropic effects using group structure. <em>Statistics in Medicine</em>, <strong>40</strong>(6), 1498-1518.
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>############################# Gene DNAJC1 ###############################################
data(DNAJC1)
Breast &lt;- DNAJC1$Breast
Thyroid &lt;- DNAJC1$Thyroid
genename &lt;- "DNAJC1"
snpnames &lt;- Breast$snp
Betah &lt;- list(Breast$beta, Thyroid$beta)
Sigmah &lt;- list(diag(Breast$se^2), diag(Thyroid$se^2))
K &lt;- 2
m &lt;- 14

RES &lt;- DS(Betah, Sigmah,
  kappa0 = 0.5, sigma20 = 1,
  m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1,
  a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename
)
## Not run: 
  print(RES)

  RES1 &lt;- DS(Betah, Sigmah,
    kappa0 = c(0.2, 0.5), sigma20 = c(1, 2),
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename
  )
  print(RES1)
  ################### Simulated summary level data with K=5 ###############################
  data(Simulated_summary)
  genename &lt;- Simulated_summary$genename
  snpnames &lt;- Simulated_summary$snpnames
  Betah &lt;- Simulated_summary$simBeta
  Sigmah &lt;- Simulated_summary$simSIGMA
  K &lt;- 5
  m &lt;- 10

  RES &lt;- DS(Betah, Sigmah,
    kappa0 = 0.5, sigma20 = 1,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename
  )
  print(RES)

  RES1 &lt;- DS(Betah, Sigmah,
    kappa0 = c(0.2, 0.5), sigma20 = c(1, 2),
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename
  )
  print(RES1)
  ################################### Gene PARP2 ##########################################
  library(BhGLM)
  data(PARP2)
  Breast &lt;- PARP2$Breast
  Thyroid &lt;- PARP2$Thyroid
  genename &lt;- "PARP2"
  snpnames &lt;- c("rs3093872", "rs3093921", "rs1713411", "rs3093926", "rs3093930", "rs878156")


  Fit1 &lt;- BhGLM::bglm(y1 ~ ., family = binomial(link = "logit"), data = Breast)
  Betah1 &lt;- Fit1$coefficients[-1]
  Sigmah1 &lt;- cov(coef(arm::sim(Fit1)))[-1, -1]


  Fit2 &lt;- BhGLM::bglm(y2 ~ ., family = binomial(link = "logit"), data = Thyroid)
  Betah2 &lt;- Fit2$coefficients[-1]
  Sigmah2 &lt;- cov(coef(arm::sim(Fit2)))[-1, -1]

  Betah &lt;- list(Betah1, Betah2)
  Sigmah &lt;- list(Sigmah1, Sigmah2)
  K &lt;- 2
  m &lt;- 6

  RES &lt;- DS(Betah, Sigmah,
    kappa0 = 0.5, sigma20 = 1,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename
  )
  print(RES)

  RES1 &lt;- DS(Betah, Sigmah,
    kappa0 = c(0.2, 0.5), sigma20 = c(1, 2),
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename
  )
  print(RES1)
  ########### Simulated individual level data with K=3 and continuous phynotype ###########
  library(BhGLM)
  data(Simulated_individual)
  Study1 &lt;- Simulated_individual$Study1
  Study2 &lt;- Simulated_individual$Study2
  Study3 &lt;- Simulated_individual$Study3
  K &lt;- 3
  m &lt;- 30
  genename &lt;- "Simulated"
  snpnames &lt;- sprintf("SNP%s", seq(1:m))


  Fit1 &lt;- BhGLM::bglm(Y1 ~ ., family = gaussian, data = data.frame(Study1))
  Betah1 &lt;- Fit1$coefficients[-1]
  Sigmah1 &lt;- cov(coef(arm::sim(Fit1)))[-1, -1]


  Fit2 &lt;- BhGLM::bglm(Y2 ~ ., family = gaussian, data = data.frame(Study2))
  Betah2 &lt;- Fit2$coefficients[-1]
  Sigmah2 &lt;- cov(coef(arm::sim(Fit2)))[-1, -1]


  Fit3 &lt;- BhGLM::bglm(Y3 ~ ., family = gaussian, data = data.frame(Study3))
  Betah3 &lt;- Fit3$coefficients[-1]
  Sigmah3 &lt;- cov(coef(arm::sim(Fit3)))[-1, -1]

  Betah &lt;- list(Betah1, Betah2, Betah3)
  Sigmah &lt;- list(Sigmah1, Sigmah2, Sigmah3)


  RES &lt;- DS(Betah, Sigmah,
    kappa0 = 0.5, sigma20 = 1,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename
  )
  print(RES)


  RES1 &lt;- DS(Betah, Sigmah,
    kappa0 = c(0.2, 0.5), sigma20 = c(1, 2),
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename
  )
  print(RES1)

  ########### Simulated individual level data with K=2 and gene expression data ###########
  library(BhGLM)
  data(Simulated_individual_survival)
  Study1 &lt;- Simulated_individual_survival$Study1
  Study2 &lt;- Simulated_individual_survival$Study2
  K &lt;- 2
  m &lt;- 10
  genename &lt;- "Simulated"
  snpnames &lt;- sprintf("G%s", seq(1:m))


  Fit1 &lt;- BhGLM::bcoxph(Study1$T ~ Study1$X)
  Betah1 &lt;- Fit1$coefficients
  Sigmah1 &lt;- Fit1$var


  Fit2 &lt;- BhGLM::bcoxph(Study2$T ~ Study2$X)
  Betah2 &lt;- Fit2$coefficients
  Sigmah2 &lt;- Fit2$var

  Betah &lt;- list(Betah1, Betah2)
  Sigmah &lt;- list(Sigmah1, Sigmah2)



  RES1 &lt;- DS(Betah, Sigmah,
    kappa0 = c(0.2, 0.5), sigma20 = c(1, 2),
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename
  )
  print(RES1)

## End(Not run)
</code></pre>

<hr>
<h2 id='e2_Monte_Carlo_EM'>Internal: e2_Monte_Carlo_EM</h2><span id='topic+e2_Monte_Carlo_EM'></span>

<h3>Description</h3>

<p>Internal: e2_Monte_Carlo_EM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>e2_Monte_Carlo_EM(
  Betah,
  Sigmah,
  kappa0 = kappa0,
  kappastar0 = kappastar0,
  sigma20 = sigma20,
  s20 = s20,
  m,
  K,
  a1 = a1,
  a2 = a2,
  d1 = d1,
  d2 = d2,
  c1 = c1,
  c2 = c2,
  e2 = e2,
  snpnames,
  genename
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="e2_Monte_Carlo_EM_+3A_betah">Betah</code></td>
<td>
<p>A matrix of dimension K*m represents the regression coefficients. Each row of this matrix includes the regression coefficients for each trait.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_sigmah">Sigmah</code></td>
<td>
<p>A symmetric block-diagonal matrix of dimension K*m is used. Each block of this matrix shows a positive definite covariance matrix which is an estimated covariance matrix of each trait.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_kappa0">kappa0</code></td>
<td>
<p>Initial value for kappa.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_kappastar0">kappastar0</code></td>
<td>
<p>Initial value for kappastar.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_sigma20">sigma20</code></td>
<td>
<p>Initial value for sigma2.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_s20">s20</code></td>
<td>
<p>Initial value for s2.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_m">m</code></td>
<td>
<p>Number of variables in the group.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_k">K</code></td>
<td>
<p>Number of traits.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_a1">a1</code>, <code id="e2_Monte_Carlo_EM_+3A_a2">a2</code></td>
<td>
<p>Hyperparameters of kappa. Default is a1=0.1 and a2=0.1.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_d1">d1</code>, <code id="e2_Monte_Carlo_EM_+3A_d2">d2</code></td>
<td>
<p>Hyperparameters of sigma2. Default is d1=0.1 and d2=0.1.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_c1">c1</code>, <code id="e2_Monte_Carlo_EM_+3A_c2">c2</code></td>
<td>
<p>Hyperparameters of kappastar. Default is c1=0.1 and c2=0.1.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_e2">e2</code></td>
<td>
<p>Initial value for doing Monte Carlo EM algorithm to estimate hyperparameter of s2.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_snpnames">snpnames</code></td>
<td>
<p>Names of variables for the group.</p>
</td></tr>
<tr><td><code id="e2_Monte_Carlo_EM_+3A_genename">genename</code></td>
<td>
<p>Name of group.</p>
</td></tr>
</table>

<hr>
<h2 id='GCPBayes'>GCPBayes Package</h2><span id='topic+GCPBayes'></span>

<h3>Description</h3>

<p>Run a Gibbs sampler for a multivariate Bayesian sparse group selection model with Dirac, continuous and hierarchical spike prior for detecting pleiotropic effects on multiple traits. This package is designed for summary statistics containing estimated regression coefficients and their estimated covariance matrices.
</p>


<h3>Author(s)</h3>

<p>Taban Baghfalaki <a href="mailto:t.baghfalaki@gmail.com">t.baghfalaki@gmail.com</a>, <a href="mailto:t.baghfalaki@modares.ac.ir">t.baghfalaki@modares.ac.ir</a>
</p>


<h3>References</h3>


<ol>
<li>
<p>Baghfalaki, T., Sugier, P. E., Truong, T., Pettitt, A. N., Mengersen, K., &amp; Liquet, B. (2021). Bayesian meta analysis models for cross cancer genomic investigation of pleiotropic effects using group structure. <em>Statistics in Medicine</em>, <strong>40</strong>(6), 1498-1518.
</p>
</li>
<li>
<p>Baghfalaki, T., Sugier, Y. Asgari, P. E., Truong, &amp; Liquet, B. (2021). GCPBayes:  An R Package for Studying Cross-Phenotype Genetic Associations with Group-Level Bayesian Meta-Analysis. <em>Submitted</em>.
</p>
</li></ol>


<hr>
<h2 id='HS'>Hierarchical Spike</h2><span id='topic+HS'></span>

<h3>Description</h3>

<p>Run a Gibbs sampler for a multivariate Bayesian sparse group selection model with hierarchical spike prior for detecting pleiotropic effects on the traits. This function is designed for summary statistics containing estimated regression coefficients and their estimated covariance matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HS(
  Betah,
  Sigmah,
  kappa0 = kappa0,
  kappastar0 = kappastar0,
  sigma20 = sigma20,
  s20 = s20,
  m,
  K,
  niter = 1000,
  burnin = 500,
  nthin = 2,
  nchains = 2,
  a1 = 0.1,
  a2 = 0.1,
  d1 = 0.1,
  d2 = 0.1,
  c1 = 1,
  c2 = 1,
  e2 = 1,
  snpnames,
  genename
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HS_+3A_betah">Betah</code></td>
<td>
<p>A list containing m-dimensional vectors of the regression coefficients for K studies.</p>
</td></tr>
<tr><td><code id="HS_+3A_sigmah">Sigmah</code></td>
<td>
<p>A list containing the positive definite covariance matrices (m*m-dimensional) which is the estimated covariance matrices of K studies.</p>
</td></tr>
<tr><td><code id="HS_+3A_kappa0">kappa0</code></td>
<td>
<p>Initial value for kappa (its dimension is equal to nchains).</p>
</td></tr>
<tr><td><code id="HS_+3A_kappastar0">kappastar0</code></td>
<td>
<p>Initial value for kappastar (its dimension is equal to nchains).</p>
</td></tr>
<tr><td><code id="HS_+3A_sigma20">sigma20</code></td>
<td>
<p>Initial value for sigma2 (its dimension is equal to nchains).</p>
</td></tr>
<tr><td><code id="HS_+3A_s20">s20</code></td>
<td>
<p>Initial value for s2 (its dimension is equal to nchains).</p>
</td></tr>
<tr><td><code id="HS_+3A_m">m</code></td>
<td>
<p>Number of variables in the group.</p>
</td></tr>
<tr><td><code id="HS_+3A_k">K</code></td>
<td>
<p>Number of traits.</p>
</td></tr>
<tr><td><code id="HS_+3A_niter">niter</code></td>
<td>
<p>Number of iterations for the Gibbs sampler.</p>
</td></tr>
<tr><td><code id="HS_+3A_burnin">burnin</code></td>
<td>
<p>Number of burn-in iterations.</p>
</td></tr>
<tr><td><code id="HS_+3A_nthin">nthin</code></td>
<td>
<p>The lag of the iterations used for the posterior analysis is defined (or thinning rate).</p>
</td></tr>
<tr><td><code id="HS_+3A_nchains">nchains</code></td>
<td>
<p>Number of Markov chains, when nchains&gt;1, the function calculates the Gelman-Rubin convergence statistic, as modified by Brooks and Gelman (1998).</p>
</td></tr>
<tr><td><code id="HS_+3A_a1">a1</code>, <code id="HS_+3A_a2">a2</code></td>
<td>
<p>Hyperparameters of kappa. Default is a1=0.1 and a2=0.1.</p>
</td></tr>
<tr><td><code id="HS_+3A_d1">d1</code>, <code id="HS_+3A_d2">d2</code></td>
<td>
<p>Hyperparameters of sigma2. Default is d1=0.1 and d2=0.1.</p>
</td></tr>
<tr><td><code id="HS_+3A_c1">c1</code>, <code id="HS_+3A_c2">c2</code></td>
<td>
<p>Hyperparameters of kappastar. Default is c1=0.1 and c2=0.1.</p>
</td></tr>
<tr><td><code id="HS_+3A_e2">e2</code></td>
<td>
<p>Initial value for doing Monte Carlo EM algorithm to estimate hyperparameter of s2.</p>
</td></tr>
<tr><td><code id="HS_+3A_snpnames">snpnames</code></td>
<td>
<p>Names of variables for the group.</p>
</td></tr>
<tr><td><code id="HS_+3A_genename">genename</code></td>
<td>
<p>Name of group.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For considering the HS prior, a reparameterization of betah_k is considered as betah_k = V_k^0.5 b_k,  V_k^0.5 = diag(tau_k1,...,tau_km)$. Therfore, we have the following hirarchical model:
</p>
<p>b_k ~ (1 - kappa) delta_0(b_k) + kappa N_m(0,sigma2 I_m ),
</p>
<p>tau_kj ~ (1 - kappa^*) delta_0(tau_kj) + kappa TN(0,s2),
</p>
<p>kappa ~ Beta(a_1,a_2),
</p>
<p>kappa^* ~ Beta(c_1,c_2),
</p>
<p>sigma2 ~ inverseGamma (d_1,d_2).
</p>
<p>s2 ~ inverseGamma (e_1,e_2).
</p>
<p>where delta_0(betah_k) denotes a point mass at 0, such that delta_0(betah_k)=1 if beta_k=0 and  delta_0(betah_k)=0 if  at least one of the $m$ components of beta_k is non-zero and TN(0,s2)  denotes a univariate truncated normal distribution at zero with mean 0 and variance s2.
</p>


<h3>Value</h3>


<ul>
<li><p> mcmcchain: The list of simulation output for all parameters.
</p>
</li>
<li><p> Summary: Summary statistics for regression coefficients in each study.
</p>
</li>
<li><p> Indicator 1: A table containing m rows of binary indicators for each study, the number of studies with nonzero signal and having pleiotropic effect by credible interval (CI). The first K columns show nonzero signals, K+1 th column includes the number of studies with nonzero signal and the last column shows an indicator for having pleiotropic effect of each SNP.
</p>
</li>
<li><p> Indicator 2: A table containing m rows of binary indicators for each study, the number of studies with nonzero signal and having pleiotropic effect by median. The first K columns show nonzero signals, K+1 th column includes the number of studies with nonzero signal and the last column shows an indicator for having pleiotropic effect of each SNP.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Taban Baghfalaki.
</p>


<h3>References</h3>

<p>Baghfalaki, T., Sugier, P. E., Truong, T., Pettitt, A. N., Mengersen, K., &amp; Liquet, B. (2021). Bayesian meta analysis models for cross cancer genomic investigation of pleiotropic effects using group structure. Statistics in Medicine, 40(6), 1498-1518.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>############################# PARP2_summary ###############################################
data(PARP2_summary)
Breast &lt;- PARP2_summary$Breast
Thyroid &lt;- PARP2_summary$Thyroid
Betah &lt;- list(Breast$beta, Thyroid$beta)
Sigmah &lt;- list(diag(Breast$se), diag(Thyroid$se))
genename &lt;- "PARP2"
snpnames &lt;- Breast$snp
K &lt;- 2
m &lt;- 6

RES &lt;- HS(Betah, Sigmah,
  kappa0 = 0.5, kappastar0 = 0.5, sigma20 = 1, s20 = 1,
  m = m, K = K, niter = 1000, burnin = 500, nthin = 1, nchains = 1,
  a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, c1 = 1, c2 = 1, e2 = 1, snpnames, genename
)
## Not run: 
  print(RES)
  ############################# Gene DNAJC1 ###############################################
  data(DNAJC1)
  Breast &lt;- DNAJC1$Breast
  Thyroid &lt;- DNAJC1$Thyroid
  genename &lt;- "DNAJC1"
  snpnames &lt;- Breast$snp
  Betah &lt;- list(Breast$beta, Thyroid$beta)
  Sigmah &lt;- list(diag(Breast$se^2), diag(Thyroid$se^2))
  K &lt;- 2
  m &lt;- 14

  RES &lt;- HS(Betah, Sigmah,
    kappa0 = 0.5, kappastar0 = 0.5, sigma20 = 1, s20 = 1,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 1, nchains = 1,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, c1 = 1, c2 = 1, e2 = 1, snpnames, genename
  )
  print(RES)

  RES1 &lt;- HS(Betah, Sigmah,
    kappa0 = c(0.5, 0.3), kappastar0 = c(0.5, 0.3), sigma20 = c(2, 1), s20 = c(1, 2),
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, c1 = 1, c2 = 1, e2 = 1, snpnames, genename
  )
  print(RES1)
  ################### Simulated summary level data with K=5 ###############################
  data(Simulated_summary)
  genename &lt;- Simulated_summary$genename
  snpnames &lt;- Simulated_summary$snpnames
  Betah &lt;- Simulated_summary$simBeta
  Sigmah &lt;- Simulated_summary$simSIGMA
  K &lt;- 5
  m &lt;- 10

  RES &lt;- HS(Betah, Sigmah,
    kappa0 = 0.5, kappastar0 = 0.5, sigma20 = 1, s20 = 1,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, c1 = 1, c2 = 1, e2 = 1, snpnames, genename
  )
  print(RES)

  RES1 &lt;- HS(Betah, Sigmah,
    kappa0 = c(0.5, 0.3), kappastar0 = c(0.5, 0.3), sigma20 = c(2, 1), s20 = c(1, 2),
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, c1 = 1, c2 = 1, e2 = 1, snpnames, genename
  )
  print(RES1)
  ################################### Gene PARP2 ##########################################
  library(BhGLM)
  data(PARP2)
  Breast &lt;- PARP2$Breast
  Thyroid &lt;- PARP2$Thyroid
  genename &lt;- "PARP2"
  snpnames &lt;- c("rs3093872", "rs3093921", "rs1713411", "rs3093926", "rs3093930", "rs878156")


  Fit1 &lt;- BhGLM::bglm(y1 ~ ., family = binomial(link = "logit"), data = Breast)
  Betah1 &lt;- Fit1$coefficients[-1]
  Sigmah1 &lt;- cov(coef(arm::sim(Fit1)))[-1, -1]


  Fit2 &lt;- BhGLM::bglm(y2 ~ ., family = binomial(link = "logit"), data = Thyroid)
  Betah2 &lt;- Fit2$coefficients[-1]
  Sigmah2 &lt;- cov(coef(arm::sim(Fit2)))[-1, -1]

  Betah &lt;- list(Betah1, Betah2)
  Sigmah &lt;- list(Sigmah1, Sigmah2)
  K &lt;- 2
  m &lt;- 6


  RES &lt;- HS(Betah, Sigmah,
    kappa0 = 0.5, kappastar0 = 0.5, sigma20 = 1, s20 = 1,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, c1 = 1, c2 = 1, e2 = 1, snpnames, genename
  )
  print(RES)

  RES1 &lt;- HS(Betah, Sigmah,
    kappa0 = c(0.5, 0.3), kappastar0 = c(0.5, 0.3), sigma20 = c(2, 1), s20 = c(1, 2),
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, c1 = 1, c2 = 1, e2 = 1, snpnames, genename
  )
  print(RES1)
  ########### Simulated individual level data with K=3 and continuous phynotype ###########
  library(BhGLM)
  data(Simulated_individual)
  Study1 &lt;- Simulated_individual$Study1
  Study2 &lt;- Simulated_individual$Study2
  Study3 &lt;- Simulated_individual$Study3
  K &lt;- 3
  m &lt;- 30
  genename &lt;- "Simulated"
  snpnames &lt;- sprintf("SNP%s", seq(1:m))


  Fit1 &lt;- BhGLM::bglm(Y1 ~ ., family = gaussian, data = data.frame(Study1))
  Betah1 &lt;- Fit1$coefficients[-1]
  Sigmah1 &lt;- cov(coef(arm::sim(Fit1)))[-1, -1]


  Fit2 &lt;- BhGLM::bglm(Y2 ~ ., family = gaussian, data = data.frame(Study2))
  Betah2 &lt;- Fit2$coefficients[-1]
  Sigmah2 &lt;- cov(coef(arm::sim(Fit2)))[-1, -1]


  Fit3 &lt;- BhGLM::bglm(Y3 ~ ., family = gaussian, data = data.frame(Study3))
  Betah3 &lt;- Fit3$coefficients[-1]
  Sigmah3 &lt;- cov(coef(arm::sim(Fit3)))[-1, -1]

  Betah &lt;- list(Betah1, Betah2, Betah3)
  Sigmah &lt;- list(Sigmah1, Sigmah2, Sigmah3)


  RES &lt;- HS(Betah, Sigmah,
    kappa0 = 0.5, kappastar0 = 0.5, sigma20 = 1, s20 = 1,
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, c1 = 1, c2 = 1, e2 = 1, snpnames, genename
  )
  print(RES)

  RES1 &lt;- HS(Betah, Sigmah,
    kappa0 = c(0.5, 0.3), kappastar0 = c(0.5, 0.3), sigma20 = c(2, 1), s20 = c(1, 2),
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, c1 = 1, c2 = 1, e2 = 1, snpnames, genename
  )
  print(RES1)

  ########### Simulated individual level data with K=2 and gene expression data ###########
  library(BhGLM)
  data(Simulated_individual_survival)
  Study1 &lt;- Simulated_individual_survival$Study1
  Study2 &lt;- Simulated_individual_survival$Study2
  K &lt;- 2
  m &lt;- 10
  genename &lt;- "Simulated"
  snpnames &lt;- sprintf("G%s", seq(1:m))


  Fit1 &lt;- BhGLM::bcoxph(Study1$T ~ Study1$X)
  Betah1 &lt;- Fit1$coefficients
  Sigmah1 &lt;- Fit1$var


  Fit2 &lt;- BhGLM::bcoxph(Study2$T ~ Study2$X)
  Betah2 &lt;- Fit2$coefficients
  Sigmah2 &lt;- Fit2$var

  Betah &lt;- list(Betah1, Betah2)
  Sigmah &lt;- list(Sigmah1, Sigmah2)

  RES1 &lt;- HS(Betah, Sigmah,
    kappa0 = c(0.5, 0.3), kappastar0 = c(0.5, 0.3), sigma20 = c(2, 1), s20 = c(1, 2),
    m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
    a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, c1 = 1, c2 = 1, e2 = 1, snpnames, genename
  )
  print(RES1)

## End(Not run)
</code></pre>

<hr>
<h2 id='MCMCplot'>MCMC plot</h2><span id='topic+MCMCplot'></span>

<h3>Description</h3>

<p>Trace plot, density plot and ACF plot for the output of CS/DS/HS. The plot is able to draw at most ten SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MCMCplot(
  Result = Result,
  k = k,
  nchains = nchains,
  whichsnps = whichsnps,
  betatype = "l",
  acftype = "correlation",
  dencol = "white",
  denlty = 1,
  denbg = "white"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MCMCplot_+3A_result">Result</code></td>
<td>
<p>All the generated results by CS/DS/HS function.</p>
</td></tr>
<tr><td><code id="MCMCplot_+3A_k">k</code></td>
<td>
<p>The number of study for drawing plots, k=1,2,...,K.</p>
</td></tr>
<tr><td><code id="MCMCplot_+3A_nchains">nchains</code></td>
<td>
<p>Number of Markov chains run in Result.</p>
</td></tr>
<tr><td><code id="MCMCplot_+3A_whichsnps">whichsnps</code></td>
<td>
<p>The name of SNPs.</p>
</td></tr>
<tr><td><code id="MCMCplot_+3A_betatype">betatype</code></td>
<td>
<p>The type of plot desired. The following values are possible: &quot;p&quot; for points, &quot;l&quot; for lines, &quot;b&quot; for both points and lines, &quot;c&quot; for empty points joined by lines, &quot;o&quot; for overplotted points and lines, &quot;s&quot; and &quot;S&quot; for stair steps and &quot;h&quot; for histogram-like vertical lines. Finally, &quot;n&quot; does not produce any points or lines.</p>
</td></tr>
<tr><td><code id="MCMCplot_+3A_acftype">acftype</code></td>
<td>
<p>String giving the type of ACF to be computed. Allowed values are &quot;correlation&quot; (the default), &quot;covariance&quot; or &quot;partial&quot;. Will be partially matched.</p>
</td></tr>
<tr><td><code id="MCMCplot_+3A_dencol">dencol</code></td>
<td>
<p>The color for filling the density plot.</p>
</td></tr>
<tr><td><code id="MCMCplot_+3A_denlty">denlty</code></td>
<td>
<p>The line type to be used in the density plot.</p>
</td></tr>
<tr><td><code id="MCMCplot_+3A_denbg">denbg</code></td>
<td>
<p>The color to be used for the background of the density plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Trace plot, density plot and ACF plot for the output of CS/DS/HS for checking convergence of MCMC chains.
</p>


<h3>Author(s)</h3>

<p>Taban Baghfalaki.
</p>


<h3>References</h3>

<p>Baghfalaki, T., Sugier, P. E., Truong, T., Pettitt, A. N., Mengersen, K., &amp; Liquet, B. (2021). Bayesian meta analysis models for cross cancer genomic investigation of pleiotropic effects using group structure. Statistics in Medicine, 40(6), 1498-1518.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#############################Gene DNAJC1 ###############################################
data(DNAJC1)
Breast &lt;- DNAJC1$Breast
Thyroid &lt;- DNAJC1$Thyroid
genename &lt;- "DNAJC1"
snpnames &lt;- Breast$snp
Betah &lt;- list(Breast$beta, Thyroid$beta)
Sigmah &lt;- list(diag(Breast$se^2), diag(Thyroid$se^2))
K &lt;- 2
m &lt;- 14


RES1 &lt;- DS(Betah, Sigmah,
           kappa0 = 0.5, sigma20 = 1,
           m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1,
           a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename
)


MCMCplot(Result = RES1, k = 2, nchains = 1, whichsnps = sample(snpnames, 7),
                     betatype = "l",
                     acftype = "correlation",
                     dencol = "white", denlty = 1, denbg = "white")
###################Simulated summary level data with K=5 ###############################
## Not run: 
data(Simulated_summary)
genename &lt;- Simulated_summary$genename
snpnames &lt;- Simulated_summary$snpnames
Betah &lt;- Simulated_summary$simBeta
Sigmah &lt;- Simulated_summary$simSIGMA
K &lt;- 5
m &lt;- 10

RES1 &lt;- DS(Betah, Sigmah,
 kappa0 = c(0.2, 0.5), sigma20 = c(1, 2),
 m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 2,
 a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename)

MCMCplot(Result = RES1, k = 3, nchains = 2, whichsnps = sample(snpnames, 3),
         betatype = "l",
         acftype = "partial",
         dencol = "blue", denlty = 1, denbg = "black")

RES1 &lt;- DS(Betah, Sigmah,
 kappa0 = c(0.2, 0.5, 0.6), sigma20 = c(1, 2, 1.5),
 m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 3,
 a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename)

MCMCplot(Result = RES1, k = 3, nchains = 3, whichsnps = sample(snpnames, 5),
         betatype = "l",
         acftype = "partial",
         dencol = "white", denlty = 1, denbg = "white")
#############################Gene DNAJC1 ###############################################
pvalue &lt;- matrix(0, K, m)
for (k in 1:K) {
  pvalue[k, ] &lt;- 2 * pnorm(-abs(Betah[[k]] / sqrt(diag(Sigmah[[k]]))))
}

zinit &lt;- rep(0, K)
for (j in 1:K) {
  index &lt;- 1:m
  PVALUE &lt;- p.adjust(pvalue[j, ])
 SIGNALS &lt;- index[PVALUE &lt; 0.05]
 modelf1 &lt;- rep(0, m)
 modelf1[SIGNALS] &lt;- 1
  if (max(modelf1) == 1) (zinit[j] &lt;- 1)
}

RES &lt;- CS(Betah, Sigmah,
  kappa0 = 0.5, tau20 = 1, zeta0 = zinit,
 m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1, a1 = 0.1, a2 = 0.1,
c1 = 0.1, c2 = 0.1, sigma2 = 10^-3, snpnames = snpnames, genename = genename)

MCMCplot(Result = RES1, k = 1, nchains = 1, whichsnps = sample(snpnames, 7),
         betatype = "l",
         acftype = "correlation",
         dencol = "white", denlty = 1, denbg = "white")
###################################Gene PARP2 ##########################################
library(BhGLM)
data(PARP2)
Breast &lt;- PARP2$Breast
Thyroid &lt;- PARP2$Thyroid
genename &lt;- "PARP2"
snpnames &lt;- c("rs3093872", "rs3093921", "rs1713411", "rs3093926", "rs3093930", "rs878156")


Fit1 &lt;- BhGLM::bglm(y1~ ., family=binomial(link="logit"),data=Breast)
Betah1 &lt;-  Fit1$coefficients[-1]
Sigmah1 &lt;- cov(coef(arm::sim(Fit1)))[-1,-1]

Fit2 &lt;- BhGLM::bglm(y2~ ., family=binomial(link="logit"),data=Thyroid)
Betah2 &lt;-  Fit2$coefficients[-1]
Sigmah2 &lt;- cov(coef(arm::sim(Fit2)))[-1,-1]

Betah &lt;- list(Betah1,Betah2)
Sigmah &lt;- list(Sigmah1,Sigmah2)
K &lt;- 2
m &lt;- 6


RES1 &lt;- DS(Betah, Sigmah, kappa0=c(0.2,0.5), sigma20=c(1,2),
          m=m, K=K, niter=1000, burnin=500, nthin=1, nchains=2,
          a1=0.1, a2=0.1, d1=0.1, d2=0.1, snpnames, genename)

MCMCplot(Result=RES1, k=1, nchains=2, whichsnps=snpnames,
         betatype = "l",
         acftype = "correlation",
         dencol = "red", denlty = 1, denbg = "white")


RES1 &lt;- DS(Betah, Sigmah, kappa0=c(0.2,0.5), sigma20=c(1,2),
          m=m, K=K, niter=2000, burnin=1000, nthin=2, nchains=2,
          a1=0.1, a2=0.1, d1=0.1, d2=0.1, snpnames, genename)

MCMCplot(Result=RES1, k=1, nchains=2, whichsnps=snpnames,
         betatype = "l",
         acftype = "correlation",
         dencol = "white", denlty = 1, denbg = "white")

## End(Not run)

</code></pre>

<hr>
<h2 id='PARP2'>Gene PARP2 from CECILE study</h2><span id='topic+PARP2'></span>

<h3>Description</h3>

<p>A list containing the individual level data for gene PARP2 including genotypes and phenotypes for pleiotropy
investigation of breast and thyroid cancers. It is from
CECILE study, a French population-based case-control study on breast cancer and from the French
studies included in the EPITHYR consortium on thyroid cancer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PARP2
</code></pre>


<h3>Format</h3>

<p>A list which contains two matrices.
</p>

<dl>
<dt>Breast</dt><dd><p>Summary statitics of breast cancer including the name of SNPs, beta and se</p>
</dd>
<dt>Thyroid</dt><dd><p>Summary statitics of thyroid cancer including the name of SNPs, beta and se</p>
</dd>
</dl>



<h3>References</h3>


<ol>
<li>
<p>Baghfalaki, T., Sugier, P. E., Truong, T., Pettitt, A. N., Mengersen, K., &amp; Liquet, B. (2021). Bayesian meta analysis models for cross cancer genomic investigation of pleiotropic effects using group structure. <em>Statistics in Medicine</em>, <strong>40</strong>(6), 1498-1518.
</p>
</li>
<li>
<p>Baghfalaki, T., Sugier, Y. Asgari, P. E., Truong, &amp; Liquet, B. (2021). GCPBayes:  An R Package for Studying Cross-Phenotype Genetic Associations with Group-Level Bayesian Meta-Analysis. <em>Submitted</em>.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+GCPBayes">GCPBayes</a></code>
</p>

<hr>
<h2 id='PARP2_summary'>Summary statistics of gene PARP2 from CECILE study</h2><span id='topic+PARP2_summary'></span>

<h3>Description</h3>

<p>The summary statistics data for PARP2 protein coding gene including beta and standard error for pleiotropy investigation of breast and thyroid cancers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PARP2_summary
</code></pre>


<h3>Format</h3>

<p>A list which contains two matrices for the summary statistics of each study.
</p>

<dl>
<dt>Breast</dt><dd><p>Summary statitics of breast cancer including the name of SNPs, beta and se</p>
</dd>
<dt>Thyroid</dt><dd><p>Summary statitics of thyroid cancer including the name of SNPs, beta and se</p>
</dd>
</dl>



<h3>References</h3>


<ol>
<li>
<p>Baghfalaki, T., Sugier, P. E., Truong, T., Pettitt, A. N., Mengersen, K., &amp; Liquet, B. (2021). Bayesian meta analysis models for cross cancer genomic investigation of pleiotropic effects using group structure. <em>Statistics in Medicine</em>, <strong>40</strong>(6), 1498-1518.
</p>
</li>
<li>
<p>Baghfalaki, T., Sugier, Y. Asgari, P. E., Truong, &amp; Liquet, B. (2021). GCPBayes:  An R Package for Studying Cross-Phenotype Genetic Associations with Group-Level Bayesian Meta-Analysis. <em>Submitted</em>.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+GCPBayes">GCPBayes</a></code>
</p>

<hr>
<h2 id='Simulated_individual'>Simulated individual level data</h2><span id='topic+Simulated_individual'></span>

<h3>Description</h3>

<p>A list containing the individual level data including genotypes and phenotypes for pleiotropy investigation of three studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Simulated_individual
</code></pre>


<h3>Format</h3>

<p>A list which contains the name of gene, the name of the SNPs, the vectors and the matrices for the summary statistics of each study.
</p>

<dl>
<dt>Study1</dt><dd><p>The inividual level data including genotypes and phenotypes of Study 1</p>
</dd>
<dt>Study2</dt><dd><p>The inividual level data including genotypes and phenotypes of Study 2</p>
</dd>
<dt>Study3</dt><dd><p>The inividual level data including genotypes and phenotypes of Study 3</p>
</dd>
</dl>



<h3>References</h3>

<p>'
</p>
<ol>
<li>
<p>Baghfalaki, T., Sugier, Y. Asgari, P. E., Truong, &amp; Liquet, B. (2021). GCPBayes:  An R Package for Studying Cross-Phenotype Genetic Associations with Group-Level Bayesian Meta-Analysis. <em>Submitted</em>.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+GCPBayes">GCPBayes</a></code>
</p>

<hr>
<h2 id='Simulated_individual_survival'>Simulated individual level survival data</h2><span id='topic+Simulated_individual_survival'></span>

<h3>Description</h3>

<p>A list containing the individual level data including gene expression data for survival outcomes for pleiotropy investigation of two studies.
#' @name Simulated_individual_survival
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Simulated_individual_survival
</code></pre>


<h3>Format</h3>

<p>A list which contains the name of gene, the name of the SNPs, the vectors and the matrices for the summary statistics of each study.
</p>

<dl>
<dt>Study1</dt><dd><p>The inividual level data including survival outcomes and gene expression data of Study 1</p>
</dd>
<dt>Study2</dt><dd><p>The inividual level data including survival outcomes and gene expression data of Study 2</p>
</dd>
</dl>



<h3>References</h3>

<p>'
</p>
<ol>
<li>
<p>Baghfalaki, T., Sugier, Y. Asgari, P. E., Truong, &amp; Liquet, B. (2021). GCPBayes:  An R Package for Studying Cross-Phenotype Genetic Associations with Group-Level Bayesian Meta-Analysis. <em>Submitted</em>.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+GCPBayes">GCPBayes</a></code>
</p>

<hr>
<h2 id='Simulated_summary'>Simulated summary statistics for K=5 traits</h2><span id='topic+Simulated_summary'></span>

<h3>Description</h3>

<p>A list containing the summary statistics including regression coefficients and covariance matrices for K=5 studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Simulated_summary
</code></pre>


<h3>Format</h3>

<p>A list which contains the name of gene, the name of the SNPs, the vectors and the matrices for the summary statistics of each study.
</p>

<dl>
<dt>genename</dt><dd><p>The name of gene</p>
</dd>
<dt>snpnames</dt><dd><p>The name of the SNPs</p>
</dd>
<dt>simBeta</dt><dd><p>The regression coeffiecnts of the studies</p>
</dd>
<dt>simSIGMA</dt><dd><p>The covariance matrices for the studies</p>
</dd>
</dl>



<h3>References</h3>

<p>'
</p>
<ol>
<li>
<p>Baghfalaki, T., Sugier, Y. Asgari, P. E., Truong, &amp; Liquet, B. (2021). GCPBayes:  An R Package for Studying Cross-Phenotype Genetic Associations with Group-Level Bayesian Meta-Analysis. <em>Submitted</em>.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+GCPBayes">GCPBayes</a></code>
</p>

<hr>
<h2 id='summaryCS'>Summary function of Continuous Spike</h2><span id='topic+summaryCS'></span>

<h3>Description</h3>

<p>summaryCS is a generic function used to produce result summaries of the results of CS function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summaryCS(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summaryCS_+3A_object">object</code></td>
<td>
<p>a result of a call to CS</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Name of Gene: the component from object.
</p>
</li>
<li><p> Number of SNPs: the component from object.
</p>
</li>
<li><p> Name of SNPs: the component from object.
</p>
</li>
<li><p> log10BF: the component from object.
</p>
</li>
<li><p> lBFDR: the component from object.
</p>
</li>
<li><p> theta: the component from object.
</p>
</li>
<li><p> Significance based on CI: the component from object.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Taban Baghfalaki.
</p>


<h3>References</h3>

<p>Baghfalaki, T., Sugier, P. E., Truong, T., Pettitt, A. N., Mengersen, K., &amp; Liquet, B. (2021). Bayesian meta analysis models for cross cancer genomic investigation of pleiotropic effects using group structure. Statistics in Medicine, 40(6), 1498-1518.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>############################# Gene DNAJC1 ###############################################
data(DNAJC1)
Breast &lt;- DNAJC1$Breast
Thyroid &lt;- DNAJC1$Thyroid
genename &lt;- "DNAJC1"
snpnames &lt;- Breast$snp
Betah &lt;- list(Breast$beta, Thyroid$beta)
Sigmah &lt;- list(diag(Breast$se^2), diag(Thyroid$se^2))
K &lt;- 2
m &lt;- 14

pvalue &lt;- matrix(0, K, m)
for (k in 1:K) {
  pvalue[k, ] &lt;- 2 * pnorm(-abs(Betah[[k]] / sqrt(diag(Sigmah[[k]]))))
}

zinit &lt;- rep(0, K)
for (j in 1:K) {
  index &lt;- 1:m
  PVALUE &lt;- p.adjust(pvalue[j, ])
  SIGNALS &lt;- index[PVALUE &lt; 0.05]
  modelf1 &lt;- rep(0, m)
  modelf1[SIGNALS] &lt;- 1
  if (max(modelf1) == 1) (zinit[j] &lt;- 1)
}


RES &lt;- CS(Betah, Sigmah,
  kappa0 = 0.5, tau20 = 1, zeta0 = zinit,
  m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1, a1 = 0.1, a2 = 0.1,
  c1 = 0.1, c2 = 0.1, sigma2 = 10^-3, snpnames = snpnames, genename = genename
)
summaryCS(RES)
</code></pre>

<hr>
<h2 id='summaryDS'>Summary function of Dirac Spike</h2><span id='topic+summaryDS'></span>

<h3>Description</h3>

<p>summaryDS is a generic function used to produce result summaries of the results of DS function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summaryDS(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summaryDS_+3A_object">object</code></td>
<td>
<p>a result of a call to DS</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Name of Gene: the component from object.
</p>
</li>
<li><p> Number of SNPs: the component from object.
</p>
</li>
<li><p> Name of SNPs: the component from object.
</p>
</li>
<li><p> log10BF: the component from object.
</p>
</li>
<li><p> lBFDR: the component from object.
</p>
</li>
<li><p> theta: the component from object.
</p>
</li>
<li><p> Significance based on CI: the component from object.
</p>
</li>
<li><p> Significance based on median thresholding: the component from object.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Taban Baghfalaki.
</p>


<h3>References</h3>

<p>Baghfalaki, T., Sugier, P. E., Truong, T., Pettitt, A. N., Mengersen, K., &amp; Liquet, B. (2021). Bayesian meta analysis models for cross cancer genomic investigation of pleiotropic effects using group structure. Statistics in Medicine, 40(6), 1498-1518.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>############################# Gene DNAJC1 ###############################################
data(DNAJC1)
Breast &lt;- DNAJC1$Breast
Thyroid &lt;- DNAJC1$Thyroid
genename &lt;- "DNAJC1"
snpnames &lt;- Breast$snp
Betah &lt;- list(Breast$beta, Thyroid$beta)
Sigmah &lt;- list(diag(Breast$se^2), diag(Thyroid$se^2))
K &lt;- 2
m &lt;- 14


RES &lt;- DS(Betah, Sigmah,
  kappa0 = 0.5, sigma20 = 1,
  m = m, K = K, niter = 2000, burnin = 1000, nthin = 2, nchains = 1,
  a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, snpnames, genename
)
summaryDS(RES)
</code></pre>

<hr>
<h2 id='summaryHS'>Summary function of Hierarchical Spike</h2><span id='topic+summaryHS'></span>

<h3>Description</h3>

<p>summaryHS is a generic function used to produce result summaries of the results of HS function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summaryHS(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summaryHS_+3A_object">object</code></td>
<td>
<p>a result of a call to DS</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Name of Gene: the component from object.
</p>
</li>
<li><p> Number of SNPs: the component from object.
</p>
</li>
<li><p> Name of SNPs: the component from object.
</p>
</li>
<li><p> Pleiotropic effect based on CI: the component from object.
</p>
</li>
<li><p> Pleiotropic effect based on median thresholding: the component from object.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Taban Baghfalaki.
</p>


<h3>References</h3>

<p>Baghfalaki, T., Sugier, P. E., Truong, T., Pettitt, A. N., Mengersen, K., &amp; Liquet, B. (2021). Bayesian meta analysis models for cross cancer genomic investigation of pleiotropic effects using group structure. Statistics in Medicine, 40(6), 1498-1518.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>############################# PARP2_summary ###############################################
data(PARP2_summary)
Breast &lt;- PARP2_summary$Breast
Thyroid &lt;- PARP2_summary$Thyroid
Betah &lt;- list(Breast$beta, Thyroid$beta)
Sigmah &lt;- list(diag(Breast$se), diag(Thyroid$se))
genename &lt;- "PARP2"
snpnames &lt;- Breast$snp
K &lt;- 2
m &lt;- 6


RES &lt;- HS(Betah, Sigmah,
  kappa0 = 0.5, kappastar0 = 0.5, sigma20 = 1, s20 = 1,
  m = m, K = K, niter = 1000, burnin = 500, nthin = 1, nchains = 1,
  a1 = 0.1, a2 = 0.1, d1 = 0.1, d2 = 0.1, c1 = 1, c2 = 1, e2 = 1, snpnames, genename
)
summaryHS(RES)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
