<!DOCTYPE html><html><head><title>Help for package gdm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gdm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculate.gdm.deviance'><p>Calculate GDM Deviance for Observed &amp; Predicted Dissimilarities</p></a></li>
<li><a href='#createsitepair'><p>Create Site-Pair Table</p></a></li>
<li><a href='#formatsitepair'><p>Combines Biological and Environmental Data to Produce a GDM-formatted</p>
Site-Pair Table</a></li>
<li><a href='#gdm'><p>Fit a Generalized Dissimilarity Model to Tabular Site-Pair Data</p></a></li>
<li><a href='#gdm-package'>
<p>Overview of the functions in the gdm package</p></a></li>
<li><a href='#gdm.crossvalidation'><p>Cross-Validation Assessment of a Fitted GDM</p></a></li>
<li><a href='#gdm.partition.deviance'><p>Perform Deviance Partitioning of a Fitted GDM</p></a></li>
<li><a href='#gdm.single.crossvalidation'><p>Single GDM Cross-Validation Test, Internal Function</p></a></li>
<li><a href='#gdm.transform'><p>Transform Environmental Data Using a Fitted Generalized Dissimilarity Model</p></a></li>
<li><a href='#gdm.varImp'><p>Assess Predictor Importance and Quantify Model Significance in a Fitted</p>
Generalized Dissimilarity Model.</a></li>
<li><a href='#gdmDissim'><p>An example biological dissimilarity matrix</p></a></li>
<li><a href='#isplineExtract'><p>Extract I-spline Values From a Fitted Generalized Dissimilarity</p>
Model.</a></li>
<li><a href='#permutateSitePair'><p>Permutate Site-Pair Table Rows, Internal Function</p></a></li>
<li><a href='#plot.gdm'><p>Plot Model Fit and I-splines from a Fitted Generalized Dissimilarity</p>
Model.</a></li>
<li><a href='#plotUncertainty'><p>Plot I-splines With Error Bands Using Bootstrapping.</p></a></li>
<li><a href='#predict.gdm'><p>Predict Biological Dissimilarities Between Sites or Times Using a</p>
Fitted Generalized Dissimilarity Model</a></li>
<li><a href='#southwest'><p>Species and Environmental Data from Southwestern Australia.</p></a></li>
<li><a href='#subsample.sitepair'><p>Remove Sites at Random from a Site-Pair Table</p></a></li>
<li><a href='#summary.gdm'><p>Summarize a Fitted Generalized Dissimilarity Model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Generalized Dissimilarity Modeling</td>
</tr>
<tr>
<td>Version:</td>
<td>1.5.0-9.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-11-17</td>
</tr>
<tr>
<td>Description:</td>
<td>A toolkit with functions to fit, plot, summarize, and apply Generalized Dissimilarity Models. Mokany K, Ware C, Woolley SNC, Ferrier S, Fitzpatrick MC (2022) &lt;<a href="https://doi.org/10.1111%2Fgeb.13459">doi:10.1111/geb.13459</a>&gt; Ferrier S, Manion G, Elith J, Richardson K (2007) &lt;<a href="https://doi.org/10.1111%2Fj.1472-4642.2007.00341.x">doi:10.1111/j.1472-4642.2007.00341.x</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://mfitzpatrick.al.umces.edu/gdm/">https://mfitzpatrick.al.umces.edu/gdm/</a>,
<a href="https://github.com/fitzLab-AL/gdm/">https://github.com/fitzLab-AL/gdm/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/fitzLab-AL/gdm/issues/">https://github.com/fitzLab-AL/gdm/issues/</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>parallel, methods, raster, Rcpp, reshape2, vegan, doParallel,
foreach, pbapply</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-28 21:52:51 UTC; mfitzpatrick</td>
</tr>
<tr>
<td>Author:</td>
<td>Matt Fitzpatrick <a href="https://orcid.org/0000-0003-1911-8407"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Karel Mokany <a href="https://orcid.org/0000-0003-4199-3697"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Glenn Manion [aut],
  Diego Nieto-Lugilde
    <a href="https://orcid.org/0000-0003-4135-2881"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Simon Ferrier <a href="https://orcid.org/0000-0001-7884-2388"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Matthew Lisk [ctb],
  Chris Ware [ctb],
  Skip Woolley [ctb],
  Tom Harwood [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matt Fitzpatrick &lt;mfitzpatrick@umces.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-01 15:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculate.gdm.deviance'>Calculate GDM Deviance for Observed &amp; Predicted Dissimilarities</h2><span id='topic+calculate.gdm.deviance'></span>

<h3>Description</h3>

<p>Calculate GDM deviance for observed &amp; predicted dissimilarities.
Can be used for assessing cross-validation data. Translated from the c++
function CalcGDMDevianceDouble() in the file NNLS_Double.cpp from the
GDM R package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate.gdm.deviance(predDiss, obsDiss)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate.gdm.deviance_+3A_preddiss">predDiss</code></td>
<td>
<p>(float) A vector of predicted dissimilarity values, of same
length as obsDiss.</p>
</td></tr>
<tr><td><code id="calculate.gdm.deviance_+3A_obsdiss">obsDiss</code></td>
<td>
<p>(float) A vector of observed dissimilarity values, of same
length as predDiss.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single value (float) being the deviance.
</p>

<hr>
<h2 id='createsitepair'>Create Site-Pair Table</h2><span id='topic+createsitepair'></span>

<h3>Description</h3>

<p>Creates a site-pair table from the lower half of a site-by-site distance
(dissimilarity) matrix. This function is called from the
<code><a href="#topic+formatsitepair">formatsitepair</a></code> function and not needed by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createsitepair(dist, spdata, envInfo, dXCol, dYCol, siteCol, weightsType,
custWeights)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createsitepair_+3A_dist">dist</code></td>
<td>
<p>The lower half of a site-by-site distance (dissimilarity) matrix,
provided by the <code><a href="#topic+formatsitepair">formatsitepair</a></code> function.</p>
</td></tr>
<tr><td><code id="createsitepair_+3A_spdata">spdata</code></td>
<td>
<p>Input species data, the same as the bioData input to the
<code><a href="#topic+formatsitepair">formatsitepair</a></code> function.</p>
</td></tr>
<tr><td><code id="createsitepair_+3A_envinfo">envInfo</code></td>
<td>
<p>Input environmental data. Only accepts data tables as input.
If the environmental data for <code><a href="#topic+formatsitepair">formatsitepair</a></code> are rasters, the
data would have been extracted into table format within
<code><a href="#topic+formatsitepair">formatsitepair</a></code>.</p>
</td></tr>
<tr><td><code id="createsitepair_+3A_dxcol">dXCol</code></td>
<td>
<p>Input x coordinate, the same as the XColumn input to the
<code><a href="#topic+formatsitepair">formatsitepair</a></code> function.</p>
</td></tr>
<tr><td><code id="createsitepair_+3A_dycol">dYCol</code></td>
<td>
<p>Input y coordinate, the same as the YColumn input to the
<code><a href="#topic+formatsitepair">formatsitepair</a></code> function.</p>
</td></tr>
<tr><td><code id="createsitepair_+3A_sitecol">siteCol</code></td>
<td>
<p>Site column, taken from either the species or environmental
tables.</p>
</td></tr>
<tr><td><code id="createsitepair_+3A_weightstype">weightsType</code></td>
<td>
<p>The method of determining the site-pair weights used in
model fitting.</p>
</td></tr>
<tr><td><code id="createsitepair_+3A_custweights">custWeights</code></td>
<td>
<p>Custom weights, as a vector, if given by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A site-pair table with appropriate distance (dissimilarity) and
weight columns used for fitting GDM.
</p>


<h3>Note</h3>

<p>This function is called from the <code><a href="#topic+formatsitepair">formatsitepair</a></code> function
and not needed by the user.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+formatsitepair">formatsitepair</a></code>
</p>

<hr>
<h2 id='formatsitepair'>Combines Biological and Environmental Data to Produce a GDM-formatted
Site-Pair Table</h2><span id='topic+formatsitepair'></span>

<h3>Description</h3>

<p>This function takes input biological data and environmental,
geographic, and other predictor data and builds a site-pair table required
for fitting a Generalized Dissimilarity Model using the <code><a href="#topic+gdm">gdm</a></code>
function. NOTE: x-y coordinates of sites MUST be present in either the
biological or the environmental data. Site coordinates ideally should be in a
projected coordinate system (i.e., not longitude-latitude) to ensure proper
calculation of geographic distances. <br />
</p>
<p>The input biological data can be in one of the following four formats. Note
that the general term &quot;species&quot; is used, but any classification of biological
entities (e.g. functional types, haplotypes, etc) can be used as long as an
appropriate distance metric is also supplied (see &quot;dist&quot; argument): <br />
</p>

<ol>
<li><p> site-by-species matrix <br />
</p>
</li>
<li><p> x, y, species list <br />
</p>
</li>
<li><p> site-by-site biological distance (dissimilarity) matrix <br />
</p>
</li>
<li><p> an existing site-pair table (see Details) <br />
</p>
</li></ol>

<p>Predictor data can be provided in three formats: <br />
</p>

<ol>
<li><p> a site-by-predictor matrix with a column for each predictor variable
and a row for each site <br />
</p>
</li>
<li><p> a raster stack, with one raster for each predictor variable <br />
</p>
</li>
<li><p> one or more site-by-site distance matrices using the &quot;distPreds&quot;
argument (see below).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>formatsitepair(bioData, bioFormat, dist="bray", abundance=FALSE, siteColumn=NULL,
XColumn, YColumn, sppColumn=NULL, abundColumn=NULL, sppFilter=0, predData,
distPreds=NULL, weightType="equal", custWeights=NULL, sampleSites=1, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="formatsitepair_+3A_biodata">bioData</code></td>
<td>
<p>The input biological (the response variable) data table, in
one of the four formats defined above (see Details).</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_bioformat">bioFormat</code></td>
<td>
<p>An integer code specifying the format of bioData. Acceptable
values are 1, 2, 3, or 4 (see Details).</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_dist">dist</code></td>
<td>
<p>Default = &quot;bray&quot;. A character code indicating the metric to
quantify pairwise site distances / dissimilarities. Calls the
<code><a href="vegan.html#topic+vegdist">vegdist</a></code> function from the <span class="pkg">vegan</span> package to
calculate dissimilarity and therefore accepts any method available from
<code><a href="vegan.html#topic+vegdist">vegdist</a></code>.</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_abundance">abundance</code></td>
<td>
<p>Default = FALSE. Indicates whether the biological data are
abundance data (TRUE) or presence-absence (0, 1) data (FALSE).</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_sitecolumn">siteColumn</code></td>
<td>
<p>The name of the column in either the biological or
environmental data table containing a unique site identifier. If a site column
is provided in both the biological and environmental data, the site column name
must be the same in both tables.</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_xcolumn">XColumn</code></td>
<td>
<p>The name of the column containing x-coordinates of sites.
X-coordinates can be provided in either the biological or environmental data
tables, but MUST be in at least one of them. If an x-coordinate column is
provided in both the biological and environmental data, the column name must
be identical. Site coordinates ideally should be in a projected coordinate
system (i.e., not longitude-latitude) to ensure proper calculation of
geographic distances. Note that if you are using rasters, they must be in the
same coordinate system as the site coordinates.</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_ycolumn">YColumn</code></td>
<td>
<p>The name of the column containing y-coordinates of sample sites.
Y-coordinates can be provided in either the biological or environmental data
tables, but MUST be in at least one of them. If a y-coordinate column is
provided in both the biological and environmental data, the column name must
be identical. Site coordinates ideally should be in a projected coordinate
system (i.e., not longitude-latitude) to ensure proper calculation of
geographic distances. Note that if you are using rasters, they must be in the
same coordinate system as the site coordinates.</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_sppcolumn">sppColumn</code></td>
<td>
<p>Only used if bioFormat = 2 (x, y, species list). The name of
the column containing unique name / identifier for each species.</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_abundcolumn">abundColumn</code></td>
<td>
<p>If abundance = TRUE, this parameter identifies the column
containing the measure of abundance at each site. Only used if bioFormat = 2
(i.e., x, y, species list), though in the case of abundance data, the table
would have four columns: x, y, species, abundance.</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_sppfilter">sppFilter</code></td>
<td>
<p>Default = 0. To account for limited sampling effort at some
sites, sppFilter removes all sites at which the number of recorded species
(i.e., observed species richness) is less than the specified value. For
example, if sppFilter = 5, all sites with fewer than 5 recorded species
will be removed.</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_preddata">predData</code></td>
<td>
<p>The environmental predictor data. Accepts either a
site-by-predictor table or a raster stack.</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_distpreds">distPreds</code></td>
<td>
<p>An optional list of distance matrices to be used as predictors
in combination with predData. For example, a site-by-site dissimilarity matrix
for one biological group (e.g., trees) can be used as a predictor for another
group (e.g., ferns). Each distance matrix must have as the first column the
names of the sites (therefore the matrix will not be square). The name of the
column containing the site names should have the same name as that provided
for the siteColumn argument. Site IDs are required here to ensure correct ordering
of sites in the construction of the site-pair table. Note that the formatsitepair
function will not accept only distances matrices as predictors
(i.e., at least one predictor variable is required). If you wish to fit GDM
using only distance matrices, provide one fake predictor (e.g., with all sites
having the same value), plus site and coordinate columns if needed. The s1 and
s2 columns for this fake variable can then be removed by hand before fitting the GDM.</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_weighttype">weightType</code></td>
<td>
<p>Default = &quot;equal&quot;. Defines the weighting for sites. Can be
either: (1) &quot;equal&quot; (weights for all sites set = 1), (2) &quot;richness&quot; (each
site weighted according to number of species recorded), or (3) &quot;custom&quot;
(user defined). If weightType=&quot;custom&quot;, the user must provide a vector of
site weights equal to the number of rows in the full site-pair table (i.e.,
before species filtering (sppFilter argument) or sub-sampling is taken into
account (sampleSites argument)).</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_custweights">custWeights</code></td>
<td>
<p>A two column matrix or data frame of user-defined site
weights. The first column should be the site name and should be named the same
as that provided for the siteColumn argument. The second column should be numeric
weight values and should be named &quot;weights&quot;. The weight values represent the
importance of each site in model fitting, and the values in the output
site-pair table is an average of the two sites in each site-pair. Required
when weightType = &quot;custom&quot;. Ignored otherwise.</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_samplesites">sampleSites</code></td>
<td>
<p>Default = 1. A number between 0-1 indicating the fraction
of sites to be used to construct the site-pair table. This argument can be
used to reduce the number of sites to overcome possible memory limitations
when fitting models with very large numbers of sites.</p>
</td></tr>
<tr><td><code id="formatsitepair_+3A_verbose">verbose</code></td>
<td>
<p>Default = FALSE. If TRUE, summary of information regarding
dimensions of the site-pair table will be printed that can be useful for diagnostics.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>bioData and bioFormat:
The function accepts biological data in the following formats:
</p>
<p>bioData = site-by-species matrix; bioFormat = 1: assumes that the response
data are provided with a site ID column (specified by siteCol) and, optionally,
two columns for the x &amp; y coordinates of the sites. All remaining columns
contain the biological data, with a column for each biological entity (most
commonly species). In the case that a raster stack is provided for the
environmental data (predData), x-y coordinates MUST be provided in bioData
to allow extraction of the environmental data at site locations. The x-y
coordinates will be intersected with the raster stack and, if the number of
unique cells intersected by the points is less than the number of unique site
IDs (i.e. multiple sites fall within a single cell), the function will use
the raster cell as the site ID and aggregate sites accordingly. Therefore,
model fitting will be sensitive to raster cell size. If the environmental
data are in tabular format, they should have the same number of sites
(i.e., same number of rows) as bioData. The x-y coordinate and site ID
columns must have the same names in bioData and predData.
</p>
<p>bioData = x, y, species list (optionally a fourth column with abundance can
be provided); bioFormat = 2: assumes a table of 3 or 4 columns, the first two
being the x &amp; y coordinates of species records, the third (sppCol) being the
name / identifier of the species observed at that location, and optionally a
fourth column indicating a measure of abundance.  If an abundance column is
not provided, presence-only data are assumed. In the case that a raster stack
is provided for the environmental data (predData), the x-y coordinates will
be intersected with the raster stack and, if the number of unique cells
intersected by the points is less than the number of unique site IDs
(i.e. multiple sites fall within a single cell), the function will use the
raster cell as the site ID and aggregate sites accordingly. Therefore, model
fitting will be sensitive to raster cell size.
</p>
<p>bioData = site-pair table; bioFormat = 4: with an already created site-pair
table, this option allows the user to add one or more distance matrices (see
distPreds above) to the existing site-pair table and/or sub-sample the
site-pair table (see sample above). If the site-pair table was not created
using the formatsitepair function, the user will need to ensure the order of
the sites matches that in other tables being provided to the function.
</p>
<p>NOTES: (1) The function assumes that the x-y coordinates and the raster stack
(if used) are in the same coordinate system. No checking is performed to
confirm this is the case. (2) The function assumes that the association between
the provided site and x-y coordinate columns are singular and unique.
Therefore, the function will fail should a given site has more than one sets of
coordinates associated with it, as well as multiple sites being given the
exact same coordinates.
</p>


<h3>Value</h3>

<p>A formatted site-pair table containing the response (biological
distance or dissimilarity), predictors, and weights as required for fitting
Generalized Dissimilarity Models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## tabular data
# start with the southwest data table
 head(southwest)
 sppData &lt;- southwest[c(1,2,13,14)]
 envTab &lt;- southwest[c(2:ncol(southwest))]

#########table type 1
## site-species table without coordinates
testData1a &lt;- reshape2::dcast(sppData, site~species)
##site-species table with coordinates
coords &lt;- unique(sppData[, 2:ncol(sppData)])
testData1b &lt;- merge(testData1a, coords, by="site")
## site-species, table-table
exFormat1a &lt;- formatsitepair(testData1a, 1, siteColumn="site", XColumn="Long",
YColumn="Lat", predData=envTab)

#' # next, let's try environmental raster data
## not run
# rastFile &lt;- system.file("./extdata/swBioclims.grd", package="gdm")
# envRast &lt;- stack(rastFile)

## site-species, table-raster
## not run
# exFormat1b &lt;- formatsitepair(testData1b, 1, siteColumn="site", XColumn="Long",
# YColumn="Lat", predData=envRast)

#########table type 2
## site xy spp list, table-table
exFormat2a &lt;- formatsitepair(sppData, 2, XColumn="Long", YColumn="Lat",
sppColumn="species", siteColumn="site", predData=envTab)

## site xy spp list, table-raster
## not run
# exFormat2b &lt;- formatsitepair(sppData, 2, XColumn="Long", YColumn="Lat",
# sppColumn="species", siteColumn="site", predData=envRast)

#########table type 3
## It is possible to format a site-pair table by starting
# with a pre-calculated matrix of biological distances
dim(gdmDissim) #square pairwise distance matrix
gdmDissim[1:5, 1:5]
# need to add a site ID column
site &lt;- unique(sppData$site)
gdmDissim &lt;- cbind(site, gdmDissim)
# now we can format the table:
exFormat3 &lt;- formatsitepair(gdmDissim, 3, XColumn="Long", YColumn="Lat",
predData=envTab, siteColumn="site")

#########table type 4
## adds a predictor matrix to an existing site-pair table, in this case,
## predData needs to be provided, but is not actually used
exFormat4 &lt;- formatsitepair(exFormat2a, 4, predData=envTab, siteColumn="site",
                            distPreds=list(as.matrix(gdmDissim)))

</code></pre>

<hr>
<h2 id='gdm'>Fit a Generalized Dissimilarity Model to Tabular Site-Pair Data</h2><span id='topic+gdm'></span>

<h3>Description</h3>

<p>The gdm function is used to fit a generalized dissimilarity model to tabular
site-pair data formatted as follows using the <code><a href="#topic+formatsitepair">formatsitepair</a></code>
function: distance, weights, s1.xCoord, s1.yCoord, s2.xCoord, s2.yCoord,
s1.Pred1, s1.Pred2, ...,s1.PredN, s2.Pred1, s2.Pred2, ..., s2.PredN. The
distance column contains the response variable must be any ratio-based
dissimilarity (distance) measure between Site 1 and Site 2. The weights column
defines any weighting to be applied during fitting of the model. If equal
weighting is required, then all entries in this column should be set to 1.0
(default). The third and fourth columns, s1.xCoord and s1.yCoord, represent
the spatial coordinates of the first site in the site pair (s1). The fifth
and sixth columns, s2.xCoord and s2.yCoord, represent the coordinates of the
second site (s2). Note that the first six columns are REQUIRED, even if you
do not intend to use geographic distance as a predictor (in which case these
columns can be loaded with dummy data if the actual coordinates are
unknown - though that would be weird, no?). The next N*2 columns contain values
for N predictors for Site 1, followed by values for the same N predictors for
Site 2. <br />
</p>
<p>The following is an example of a GDM input table header with three
environmental predictors (Temp, Rain, Bedrock): <br />
</p>
<p>distance, weights, s1.xCoord, s1.yCoord, s2.xCoord, s2.yCoord, s1.Temp,
s1.Rain, s1.Bedrock, s2.Temp, s2.Rain, s2.Bedrock
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdm(data, geo=FALSE, splines=NULL, knots=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gdm_+3A_data">data</code></td>
<td>
<p>A data frame containing the site pairs to be used to fit the GDM
(obtained using the <code><a href="#topic+formatsitepair">formatsitepair</a></code> function). The
observed response data must be located in the first column. The weights to
be applied to each site pair must be located in the second column. If geo
is TRUE, then the s1.xCoord, s1.yCoord and s2.xCoord, s2.yCoord columns
will be used to calculate the geographic distance between site pairs for
inclusion as the geographic predictor term in the model. Site coordinates
ideally should be in a projected coordinate system (i.e., not longitude-latitude)
to ensure proper calculation of geographic distances. If geo is FALSE
(default), then the s1.xCoord, s1.yCoord, s2.xCoord and s2.yCoord data
columns must still be included, but are ignored in fitting the model.
Columns containing the predictor data for Site 1, and the predictor data
for Site 2, follow.</p>
</td></tr>
<tr><td><code id="gdm_+3A_geo">geo</code></td>
<td>
<p>Set to TRUE if geographic distance between sites is to be included
as a model term. Set to FALSE if geographic distance is to be omitted from
the model. Default is FALSE.</p>
</td></tr>
<tr><td><code id="gdm_+3A_splines">splines</code></td>
<td>
<p>An optional vector of the number of I-spline basis functions
to be used for each predictor in fitting the model. If supplied, it must
have the same length as the number of predictors (including geographic
distance if geo is TRUE). If this vector is not provided (splines=NULL),
then a default of 3 basis functions is used for all predictors.</p>
</td></tr>
<tr><td><code id="gdm_+3A_knots">knots</code></td>
<td>
<p>An optional vector of knots in  <em>units of the predictor
variables</em> to be used in the fitting process. If knots are supplied and
splines=NULL, then the knots argument must have the same length as the
number of predictors * n, where n is the number of knots (default=3). If both
knots and the number of splines are supplied, then the length of the knots
argument must be the same as the sum  of the values in the splines vector.
Note that the default values for knots when the default three I-spline basis
functions are 0 (minimum), 50 (median), and 100 (maximum) quantiles.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gdm returns a gdm model object. The function
<code><a href="#topic+summary.gdm">summary.gdm</a></code> can be used to obtain or print a synopsis of the
results. A gdm model object is a list containing at least the following
components:
</p>
 <dl>
<dt>dataname</dt><dd><p>The name of the table used as the data argument
to the model.</p>
</dd> <dt>geo</dt><dd><p>Whether geographic distance was used as a
predictor in the model.</p>
</dd> <dt>gdmdeviance</dt><dd><p>The deviance of the fitted GDM
model.</p>
</dd> <dt>nulldeviance</dt><dd><p> The deviance of the null model.</p>
</dd>
<dt>explained</dt><dd><p>The percentage of null deviance explained by the fitted
GDM model.</p>
</dd> <dt>intercept</dt><dd><p>The fitted value for the intercept term in the
model.</p>
</dd> <dt>predictors</dt><dd><p>A list of the names of the predictors that were
used to fit the model, in order of the amount of turnover associated with
each predictor (based on the sum of the I-spline coefficients).</p>
</dd> <dt>coefficients</dt><dd><p>A list of the coefficients for
each spline for each of the predictors considered in model fitting.</p>
</dd>
<dt>knots</dt><dd><p>A vector of the knots derived from the x data (or user
defined), for each predictor.</p>
</dd> <dt>splines</dt><dd><p> A vector of the number of
I-spline basis functions used for each predictor.</p>
</dd> <dt>creationdate</dt><dd><p> The
date and time of model creation.</p>
</dd> <dt>observed</dt><dd><p>The observed response for
each site pair (from data column 1).</p>
</dd> <dt>predicted</dt><dd><p> The predicted
response for each site pair, from the fitted model (after applying the link
function).</p>
</dd> <dt>ecological</dt><dd><p> The linear predictor (ecological distance)
for each site pair, from the fitted model (before applying the link
function).</p>
</dd> </dl>



<h3>References</h3>

<p>Ferrier S, Manion G, Elith J, Richardson, K (2007) Using
generalized dissimilarity modelling to analyse and predict patterns of beta
diversity in regional biodiversity assessment. <em>Diversity &amp;
Distributions</em> 13, 252-264.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+formatsitepair">formatsitepair</a>, <a href="#topic+summary.gdm">summary.gdm</a>,
  <a href="#topic+plot.gdm">plot.gdm</a>, <a href="#topic+predict.gdm">predict.gdm</a>, <a href="#topic+gdm.transform">gdm.transform</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ##fit table environmental data
 # format site-pair table using the southwest data table
 head(southwest)
 sppData &lt;- southwest[c(1,2,13,14)]
 envTab &lt;- southwest[c(2:ncol(southwest))]

 sitePairTab &lt;- formatsitepair(sppData, 2, XColumn="Long", YColumn="Lat", sppColumn="species",
                               siteColumn="site", predData=envTab)

 ##fit table GDM
 gdmTabMod &lt;- gdm(sitePairTab, geo=TRUE)
 summary(gdmTabMod)

 ##fit raster environmental data
 ##sets up site-pair table
 rastFile &lt;- system.file("./extdata/swBioclims.grd", package="gdm")
 envRast &lt;- raster::stack(rastFile)

 ##environmental raster data
 sitePairRast &lt;- formatsitepair(sppData, 2, XColumn="Long",
                                YColumn="Lat", sppColumn="species",
                                siteColumn="site", predData=envRast)
 ##sometimes raster data returns NA in the site-pair table, these rows will
 ##have to be removed before fitting gdm
 sitePairRast &lt;- na.omit(sitePairRast)

 ##fit raster GDM
 gdmRastMod &lt;- gdm(sitePairRast, geo=TRUE)
 summary(gdmRastMod)

</code></pre>

<hr>
<h2 id='gdm-package'>
Overview of the functions in the gdm package
</h2><span id='topic+gdm-package'></span>

<h3>Description</h3>

<p>Generalized Dissimilarity Modeling is a statistical technique for modelling variation in biodiversity between pairs of geographical locations or through time. The <span class="pkg">gdm</span> package provides functions to fit, evaluate, summarize, and plot Generalized Dissimilarity Models and to make predictions (across space and/or through time) and map biological patterns by transforming environmental predictor variables.
</p>


<h3>Details</h3>

<p>The functions in the <span class="pkg">gdm</span> package provide the tools necessary for fitting GDMs, including functions to prepare biodiversity and environmental data. Major functionality includes:
</p>

<ul>
<li><p> Formatting various types of biodiversity and environmental data to <span class="pkg">gdm</span>'s site-pair format used in model fitting
</p>
</li>
<li><p> Fitting GDMs using geographic and environmental distances between sites
</p>
</li>
<li><p> Plotting fitted functions &amp; extracting I-spline values
</p>
</li>
<li><p> Estimating predictor importance using matrix permutation and predictor contributions using deviance paritioning
</p>
</li>
<li><p> Using cross-validation to evaluate models
</p>
</li>
<li><p> Predicting pairwise dissimiliarites between sites or times and transforming environmental predictors to biological importance and mapping these patterns.
</p>
</li></ul>

<p>To see the preferable citation of the package, type <code>citation("gdm")</code>.
</p>


<h3>I. Formatting input data</h3>

<p>GDM fits biological distances to pairwise site geographical and environmental distances. Most users will need to first format their data to <span class="pkg">gdm</span>'s site-pair table format:
</p>

<table>
<tr>
 <td style="text-align: left;">
    <code><a href="#topic+formatsitepair">formatsitepair</a></code></td><td style="text-align: left;"> To convert biodiversity and environmental data to site-pair format </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>II. Model fitting, evaluation, and summary</h3>


<table>
<tr>
 <td style="text-align: left;">
    <code><a href="#topic+gdm">gdm</a></code></td><td style="text-align: left;"> To fit a GDM</td>
</tr>
<tr>
 <td style="text-align: left;">
    <code><a href="#topic+gdm.crossvalidation">gdm.crossvalidation</a></code></td><td style="text-align: left;"> To evaluate a GDM</td>
</tr>
<tr>
 <td style="text-align: left;">
    <code><a href="#topic+gdm.partition.deviance">gdm.partition.deviance</a></code></td><td style="text-align: left;"> To asses predictor contributions to deviance explained</td>
</tr>
<tr>
 <td style="text-align: left;">
    <code><a href="#topic+gdm.varImp">gdm.varImp</a></code></td><td style="text-align: left;"> To asses model significance and predictor importance</td>
</tr>
<tr>
 <td style="text-align: left;">
    <code><a href="base.html#topic+summary">summary</a></code></td><td style="text-align: left;"> To summarize a GDM</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>III. Model prediction and transformation of environmental data</h3>


<table>
<tr>
 <td style="text-align: left;">
    <code><a href="stats.html#topic+predict">predict</a></code></td><td style="text-align: left;"> To predict biological dissimilarities between sites in space or between time periods</td>
</tr>
<tr>
 <td style="text-align: left;">
    <code><a href="#topic+gdm.transform">gdm.transform</a></code></td><td style="text-align: left;"> To transform each environmental predictor to biological importance</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>IV. Plotting model output and fitted functions</h3>


<table>
<tr>
 <td style="text-align: left;">
    <code><a href="graphics.html#topic+plot">plot</a></code></td><td style="text-align: left;"> To plot model fit and I-splines</td>
</tr>
<tr>
 <td style="text-align: left;">
    <code><a href="#topic+isplineExtract">isplineExtract</a></code></td><td style="text-align: left;"> To extract I-spline values to allow for custom plotting</td>
</tr>
<tr>
 <td style="text-align: left;">
    <code><a href="#topic+plotUncertainty">plotUncertainty</a></code></td><td style="text-align: left;"> To estimate and plot model sensitivity using bootstrapping</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>The <span class="pkg">gdm</span> development team is Matt Fitzpatrick and Karel Mokany. The R package is based on code originally developed by Glenn Manion under the direction of Simon Ferrier. Where others have contributed to individual functions, credits are provided in function help pages.
</p>
<p>The maintainer of the R version of <span class="pkg">gdm</span> is Matt Fitzpatrick &lt;mfitzpatrick@umces.edu&gt;.
</p>

<hr>
<h2 id='gdm.crossvalidation'>Cross-Validation Assessment of a Fitted GDM</h2><span id='topic+gdm.crossvalidation'></span>

<h3>Description</h3>

<p>Undertake a cross-validation assessment of a GDM fit using all
the predictors included in the formated GDM input table (spTable). The
cross-validation is run using a specified proportion (train.proportion) of
the randomly selected sites included in spTable to train the model, with the
remaining sites being used to test the performance of the model predictions.
The test is repeated a specified number of times (n.crossvalid.tests), with
a unique random sample taken each time. Outputs are a number of
cross-validation test metrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdm.crossvalidation(spTable, train.proportion=0.9, n.crossvalid.tests=1,
geo=FALSE, splines=NULL, knots=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gdm.crossvalidation_+3A_sptable">spTable</code></td>
<td>
<p>(dataframe) A dataframe holding the GDM input table for model
fitting.</p>
</td></tr>
<tr><td><code id="gdm.crossvalidation_+3A_train.proportion">train.proportion</code></td>
<td>
<p>(float) The proportion of sites in 'spTable' to use
in training the GDM, with the remaining proportion used to test the model.
(default = 0.9)</p>
</td></tr>
<tr><td><code id="gdm.crossvalidation_+3A_n.crossvalid.tests">n.crossvalid.tests</code></td>
<td>
<p>(integer) The number of cross-validation sets to
use in testing the GDM. (default = 1)</p>
</td></tr>
<tr><td><code id="gdm.crossvalidation_+3A_geo">geo</code></td>
<td>
<p>(boolean) Geographic distance to be used in model fitting
(default = FALSE).</p>
</td></tr>
<tr><td><code id="gdm.crossvalidation_+3A_splines">splines</code></td>
<td>
<p>(vector) An optional vector of the number of I-spline basis
functions to be used for each predictor in fitting the model.</p>
</td></tr>
<tr><td><code id="gdm.crossvalidation_+3A_knots">knots</code></td>
<td>
<p>(vector) An optional vector of knots in units of the predictor
variables to be used in the fitting process.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List, providing cross-validation statistics. These are metrics that describe how well the model fit using the
sitepair training table predicts the dissimilarities in the sitepair testing table. Metrics provided include:
'Deviance.Explained' (the deviance explained for the training data);
'Test.Deviance.Explained' (the deviance explained for the test data);
'Mean.Error';
'Mean.Absolute.Error';
'Root.Mean.Squre.Error';
'Obs.Pred.Correlation' (Pearson's correlation coefficient between observed and predicted values);
'Equalised.RMSE' (the average root mean square error across bands of observed dissimilarities (0.05 dissimialrity units));
'Error.by.Observed.Value' (the average root mean square error and number of observations within bands of observed dissimilarities (0.05 dissimialrity units)).
</p>

<hr>
<h2 id='gdm.partition.deviance'>Perform Deviance Partitioning of a Fitted GDM</h2><span id='topic+gdm.partition.deviance'></span>

<h3>Description</h3>

<p>Partitions deviance explained from GDM into different
user specified components - most typically environment versus space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdm.partition.deviance(sitePairTable, varSets=list(), partSpace=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gdm.partition.deviance_+3A_sitepairtable">sitePairTable</code></td>
<td>
<p>A correctly formatted site-pair table from
<code><a href="#topic+formatsitepair">formatsitepair</a></code>.</p>
</td></tr>
<tr><td><code id="gdm.partition.deviance_+3A_varsets">varSets</code></td>
<td>
<p>A list in which each element is a vector of variable names
across which deviance partitioning is to be performed, excluding
geographic distance (which is set by the partSpace argument). Variable names
must match those used to build the site-pair table. See example.</p>
</td></tr>
<tr><td><code id="gdm.partition.deviance_+3A_partspace">partSpace</code></td>
<td>
<p>Whether or not to perform the partitioning using
geographic space. Default=TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe summarizing deviance partitioning results.
</p>


<h3>Author(s)</h3>

<p>Matt Fitzpatrick and Karel Mokany
</p>


<h3>Examples</h3>

<pre><code class='language-R'># set up site-pair table using the southwest data set
sppData &lt;- southwest[c(1,2,13,14)]
envTab &lt;- southwest[c(2:ncol(southwest))]
sitePairTab &lt;- formatsitepair(sppData, 2, XColumn="Long", YColumn="Lat",
sppColumn="species", siteColumn="site", predData=envTab)

# EXAMPLE - Partition two groups of variables
# Make list of variable sets for partitioning
varSet &lt;- vector("list", 2)

# now, name the variable groups for partitioning
# note you do not need to add "space" as this is only needed
# for environmental variables

# two groups (soils &amp; climate)
names(varSet) &lt;- c("soil", "climate")

# lastly, add variable names for
varSet$soil &lt;- c("awcA", "phTotal", "sandA", "shcA", "solumDepth")
varSet$climate &lt;- c("bio5", "bio6", "bio15", "bio18", "bio19")
varSet

# run the function to partition soils, climate, and space (partSpace=TRUE)
scgPart &lt;- gdm.partition.deviance(sitePairTab, varSet, partSpace=TRUE)

# EXAMPLE - Partition three groups of variables
# Make list of variable sets for partitioning
varSet &lt;- vector("list", 3)
names(varSet) &lt;- c("soil", "temp", "precip")
varSet$soil &lt;- c("awcA", "phTotal", "sandA", "shcA", "solumDepth")
varSet$temp &lt;- c("bio5", "bio6")
varSet$precip &lt;- c("bio15", "bio18", "bio19")

# partition soils, temperature, and precip
# note we can't also partition space given the function's limit to a
# maximum of three variable sets, so we set partSpace=FALSE
scPart &lt;- gdm.partition.deviance(sitePairTab, varSet, partSpace=FALSE)

</code></pre>

<hr>
<h2 id='gdm.single.crossvalidation'>Single GDM Cross-Validation Test, Internal Function</h2><span id='topic+gdm.single.crossvalidation'></span>

<h3>Description</h3>

<p>Undertake a cross-validation assessment of a GDM, using a single
training and testing dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdm.single.crossvalidation(spTable_train, spTable_test, geo=FALSE,
splines=NULL, knots=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gdm.single.crossvalidation_+3A_sptable_train">spTable_train</code></td>
<td>
<p>(dataframe) A dataframe holding the GDM input table for
model fitting.</p>
</td></tr>
<tr><td><code id="gdm.single.crossvalidation_+3A_sptable_test">spTable_test</code></td>
<td>
<p>(dataframe) A dataframe holding the GDM input table for
model testing, having identical column names to 'spTable_train' but using
different site-pairs.</p>
</td></tr>
<tr><td><code id="gdm.single.crossvalidation_+3A_geo">geo</code></td>
<td>
<p>(boolean) Geographic distance to be used in model fitting
(default = FALSE).</p>
</td></tr>
<tr><td><code id="gdm.single.crossvalidation_+3A_splines">splines</code></td>
<td>
<p>(vector) An optional vector of the number of I-spline basis
functions to be used for each predictor in fitting the model.</p>
</td></tr>
<tr><td><code id="gdm.single.crossvalidation_+3A_knots">knots</code></td>
<td>
<p>(vector) An optional vector of knots in units of the predictor
variables to be used in the fitting process.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List, providing cross-validation statistics. These are metrics that
describe how well the model fit using the sitepair training table predicts
the dissimilarities in the sitepair testing table. Metrics provided include:
'Deviance.Explained' (the deviance explained for the training data);
'Test.Deviance.Explained' (the deviance explained for the test data);
'Mean.Error';
'Mean.Absolute.Error';
'Root.Mean.Squre.Error';
'Obs.Pred.Correlation' (Pearson's correlation coefficient between observed and predicted values);
'Equalised.RMSE' (the average root mean square error across bands of observed dissimilarities (0.05 dissimialrity units));
'Error.by.Observed.Value' (the average root mean square error and number of observations within bands of observed dissimilarities (0.05 dissimialrity units)).
</p>

<hr>
<h2 id='gdm.transform'>Transform Environmental Data Using a Fitted Generalized Dissimilarity Model</h2><span id='topic+gdm.transform'></span>

<h3>Description</h3>

<p>This function transforms geographic and environmental predictors using (1) the
fitted functions from a model object returned from <code><a href="#topic+gdm">gdm</a></code> and (2) a
data frame or raster stack containing predictor data for a set of sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdm.transform(model, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gdm.transform_+3A_model">model</code></td>
<td>
<p>A gdm model object resulting from a call to <code><a href="#topic+gdm">gdm</a></code>.</p>
</td></tr>
<tr><td><code id="gdm.transform_+3A_data">data</code></td>
<td>
<p>Either (i) a data frame containing values for each predictor variable in the model, formatted as follows: X, Y, var1, var2, var3, ..., varN or (ii) a raster stack with one layer per predictor variable used in the model, excluding X and Y (rasters for x- and y-coordinates are built automatically from the input rasters if the model was fit with geo=T). The order of the columns (data frame) or raster layers (raster stack) MUST be the same as the order of the predictors in the site-pair table used in model fitting. There is currently no checking to ensure that the order of the variables to be transformed are the same as those in the site-pair table used in model fitting. If geographic distance was not used as a predictor in model fitting, the x- and y-columns need to be removed from the data to be transformed. Output is provided in the same format as the input data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gdm.transform returns either a data frame with the same number of rows as the input data frame or a raster stack, depending on the format of the input data. If the model uses geographic distance as a predictor the output object will contain columns or layers for the transformed X and Y values for each site. The transformed environmental data will be in the remaining columns or layers.
</p>


<h3>References</h3>

<p>Ferrier S, Manion G, Elith J, Richardson, K (2007) Using generalized dissimilarity modelling to analyse and predict patterns of beta diversity in regional biodiversity assessment. <em>Diversity &amp; Distributions</em> 13, 252-264.
</p>
<p>Fitzpatrick MC, Keller SR (2015) Ecological genomics meets community-level modeling of biodiversity: Mapping the genomic landscape of current and future environmental adaptation. <em>Ecology Letters</em> 18: 1-16
</p>


<h3>Examples</h3>

<pre><code class='language-R'># start with the southwest data set
# grab the columns with xy, site ID, and species data
sppTab &lt;- southwest[, c("species", "site", "Lat", "Long")]

##fit gdm using rasters
rastFile &lt;- system.file("./extdata/swBioclims.grd", package="gdm")
envRast &lt;- raster::stack(rastFile)
sitePairRast &lt;- formatsitepair(sppTab, 2, XColumn="Long", YColumn="Lat", sppColumn="species",
                               siteColumn="site", predData=envRast)
##remove NA values
sitePairRast &lt;- na.omit(sitePairRast)

##fit raster GDM
gdmRastMod &lt;- gdm(sitePairRast, geo=TRUE)

##raster input, raster output
transRasts &lt;- gdm.transform(gdmRastMod, envRast)

# map biological patterns
rastDat &lt;- raster::sampleRandom(transRasts, 10000)
pcaSamp &lt;- prcomp(rastDat)

# note the use of the 'index' argument
pcaRast &lt;- raster::predict(transRasts, pcaSamp, index=1:3)

# scale rasters
pcaRast[[1]] &lt;- (pcaRast[[1]]-pcaRast[[1]]@data@min) /
  (pcaRast[[1]]@data@max-pcaRast[[1]]@data@min)*255
pcaRast[[2]] &lt;- (pcaRast[[2]]-pcaRast[[2]]@data@min) /
  (pcaRast[[2]]@data@max-pcaRast[[2]]@data@min)*255
pcaRast[[3]] &lt;- (pcaRast[[3]]-pcaRast[[3]]@data@min) /
  (pcaRast[[3]]@data@max-pcaRast[[3]]@data@min)*255

raster::plotRGB(pcaRast, r=1, g=2, b=3)

</code></pre>

<hr>
<h2 id='gdm.varImp'>Assess Predictor Importance and Quantify Model Significance in a Fitted
Generalized Dissimilarity Model.</h2><span id='topic+gdm.varImp'></span>

<h3>Description</h3>

<p>This function uses matrix permutation to perform model and
predictor significance testing and to estimate predictor importance in a
generalized dissimilarity model. The function can be run in parallel on
multicore machines to reduce computation time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdm.varImp(spTable, geo, splines = NULL, knots = NULL,
predSelect = FALSE, nPerm = 50, pValue=0.05, parallel = FALSE, cores = 2,
sampleSites = 1, sampleSitePairs = 1, outFile = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gdm.varImp_+3A_sptable">spTable</code></td>
<td>
<p>A site-pair table, same as used to fit a <code><a href="#topic+gdm">gdm</a></code>.</p>
</td></tr>
<tr><td><code id="gdm.varImp_+3A_geo">geo</code></td>
<td>
<p>Similar to the <code><a href="#topic+gdm">gdm</a></code> geo argument. The only
difference is that the geo argument does not have a default in this function.</p>
</td></tr>
<tr><td><code id="gdm.varImp_+3A_splines">splines</code></td>
<td>
<p>Same as the <code><a href="#topic+gdm">gdm</a></code> splines argument. Note that
the current implementation requires that all predictors have the same number of
splines.</p>
</td></tr>
<tr><td><code id="gdm.varImp_+3A_knots">knots</code></td>
<td>
<p>Same as the <code><a href="#topic+gdm">gdm</a></code> knots argument.</p>
</td></tr>
<tr><td><code id="gdm.varImp_+3A_predselect">predSelect</code></td>
<td>
<p>Set to TRUE to perform predictor selection using matrix
permutation and backward elimination. Default is FALSE. When predSelect = FALSE
results will be returned only for a model fit with all predictors.</p>
</td></tr>
<tr><td><code id="gdm.varImp_+3A_nperm">nPerm</code></td>
<td>
<p>Number of permutations to use to estimate p-values. Default is 50.</p>
</td></tr>
<tr><td><code id="gdm.varImp_+3A_pvalue">pValue</code></td>
<td>
<p>The p-value to use for predictor selection / elimination. Default is 0.05.</p>
</td></tr>
<tr><td><code id="gdm.varImp_+3A_parallel">parallel</code></td>
<td>
<p>Whether or not to run the matrix permutations and model
fitting in parallel. Parallel processing is highly recommended when either
(i) the nPerms argument is large (&gt;100) or (ii) a large number of site-pairs
(and / or variables) are being used in model fitting (note computation demand
can be reduced using subsampling - see next arguments). The default is FALSE.</p>
</td></tr>
<tr><td><code id="gdm.varImp_+3A_cores">cores</code></td>
<td>
<p>When the parallel argument is set to TRUE, the number of cores
to be registered for parallel processing. Must be &lt;= the number of cores in
the machine running the function. There is no benefit to setting the number of
cores greater than the number of predictors in the model.</p>
</td></tr>
<tr><td><code id="gdm.varImp_+3A_samplesites">sampleSites</code></td>
<td>
<p>The fraction (0-1, though a value of 0 would be silly,
wouldn't it?) of <em>sites to retain</em> from the full site-pair table. If
less than 1, this argument will completely remove a fraction of sites such
that they are not used in the permutation routines.</p>
</td></tr>
<tr><td><code id="gdm.varImp_+3A_samplesitepairs">sampleSitePairs</code></td>
<td>
<p>The fraction (0-1) of <em>site-pairs (i.e., rows)
to retain</em> from the full site-pair table - in other words, all sites will
be used in the permutation routines (assuming sampleSites = 1), but not
all <em>site-pair combinations</em>. In the case where both the sampleSites
and the sampleSitePairs argument have values less than 1, sites first will
be removed using the sampleSites argument, followed by removal of site-pairs
using the sampleSitePairs argument. Note that the number of site-pairs
removed is based on the fraction of the resulting site-pair table after
sites have been removed, not on the size of the full site-pair table.</p>
</td></tr>
<tr><td><code id="gdm.varImp_+3A_outfile">outFile</code></td>
<td>
<p>An optional character string to write the object returned by
the function to disk as an .RData object (&quot;.RData&quot; is not required as part
of the file name). The .RData object will contain a single list with the
name of &quot;outObject&quot;. The default is NULL, meaning that no file will be written.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To test model significance, first a model is fit using all predictors and
un-permuted environmental data. Any predictor for which the sum of the I-spline
coefficients sum to zero is preemptively removed. Next, the environmental data are permuted
nPerm times (by randomizing the order of the rows) and a GDM is fit to each
permuted table. Model significance is determined by comparing the deviance
explained by GDM fit to the un-permuted table to the distribution of deviance
explained values from GDM fit to the nPerm permuted tables. To assess predictor
significance, this process is repeated for each predictor individually (i.e.,
only the data for the predictor being tested is permuted rather than the entire
environmental table). Predictor importance is quantified as the percent change
in deviance explained between a model fit with and without that predictor permuted. If
predSelect=TRUE, this process continues by next permutating the site-pair
table nPerm times, but removing one predictor at a time and reassessing
predictor importance and significance. At each step, the least important
predictor is dropped (backward elimination) and the process continues until
all non-significant predictors are removed, with significance level being set
by the user and the pValue argument.
</p>


<h3>Value</h3>

<p>A list of four tables. The first table summarizes full model deviance,
percent deviance explained by the full model, the p-value of the full model,
and the number of permutations used to calculate the statistics for each
fitted model (i.e., the full model and each model with predictors removed in
succession during the backward elimination procedure if predSelect=T). The
remaining three tables summarize (1) predictor importance, (2) predictor
significance, and (3) the number of permutations used to calculate the
statistics for that model, which is provided because some GDMs may fail
to converge for some permutations / predictor combinations and you might want to
know how many permutations were used when calculating statistics. Or maybe you
don't, you decide.
</p>
<p>Predictor importance is measured as the percent decrease in deviance explained
between the full model and the deviance explained by a model fit with that predictor
permuted. Significance is estimated using the bootstrapped p-value when the
predictor has been permuted. For most cases, the number of permutations will
equal the nPerm argument. However, the value may be less should any of the models
fit to them permuted tables fail to converge.
</p>
<p>If predSelect=FALSE, the tables will have values only in the first column.
</p>


<h3>Author(s)</h3>

<p>Matt Fitzpatrick and Karel Mokany
</p>


<h3>References</h3>

<p>Ferrier S, Manion G, Elith J, Richardson, K (2007) Using
generalized dissimilarity modelling to analyse and predict patterns of
beta diversity in regional biodiversity assessment. <em>Diversity &amp;
Distributions</em> 13, 252-264.
</p>
<p>Fitzpatrick, MC, Sanders NJ, Ferrier S, Longino JT, Weiser MD, and RR Dunn.
2011. Forecasting the Future of Biodiversity: a Test of Single- and
Multi-Species Models for Ants in North America. <em>Ecography</em> 34: 836-47.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##fit table environmental data
##set up site-pair table using the southwest data set
sppData &lt;- southwest[c(1,2,13,14)]
envTab &lt;- southwest[c(2:ncol(southwest))]
sitePairTab &lt;- formatsitepair(sppData, 2, XColumn="Long", YColumn="Lat",
sppColumn="species", siteColumn="site", predData=envTab)

## not run
#modTest &lt;- gdm.varImp(sitePairTab, geo=T, nPerm=50, parallel=T, cores=10, predSelect=T)
#barplot(sort(modTest$`Predictor Importance`[,1], decreasing=T))

</code></pre>

<hr>
<h2 id='gdmDissim'>An example biological dissimilarity matrix</h2><span id='topic+gdmDissim'></span>

<h3>Description</h3>

<p>Pairwise Bray-Curtis dissimilarity calculated using the species
occurrence data from the <code><a href="#topic+southwest">southwest</a></code> data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdmDissim
</code></pre>


<h3>Format</h3>

<p>A data frame with 94 rows and 94 columns:
</p>

<hr>
<h2 id='isplineExtract'>Extract I-spline Values From a Fitted Generalized Dissimilarity
Model.</h2><span id='topic+isplineExtract'></span>

<h3>Description</h3>

<p>Extracts the I-spline values from a gdm object. There is one
I-spline for each predictor that has at least one non-zero coefficient in
the fitted model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isplineExtract(model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isplineExtract_+3A_model">model</code></td>
<td>
<p>A gdm object from <code><a href="#topic+gdm">gdm</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two items. The first item contains the x-values (actual
values of the predictors) of the I-splines and the second item contains the
y-values (partial ecological distances) of the fitted I-splines.
</p>


<h3>References</h3>

<p>Ferrier S, Manion G, Elith J, Richardson, K (2007) Using
generalized dissimilarity modelling to analyse and predict patterns of beta
diversity in regional biodiversity assessment. <em>Diversity &amp; Distributions</em>
13, 252-264.
</p>
<p>Fitzpatrick MC, Sanders NJ, Normand S, Svenning J-C, Ferrier S, Gove AD,
Dunn RR (2013). Environmental and historical imprints on beta diversity: insights
from variation in rates of species turnover along gradients. Proceedings of the
Royal Society: Series B 280, art. 1768
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##set up site-pair table using the southwest data set
sppData &lt;- southwest[, c(1,2,14,13)]
envTab &lt;- southwest[, c(2:ncol(southwest))]
sitePairTab &lt;- formatsitepair(sppData, 2, XColumn="Long", YColumn="Lat", sppColumn="species",
                              siteColumn="site", predData=envTab)

##create GDM
gdmMod &lt;- gdm(sitePairTab, geo=TRUE)

##extracts splines
exSplines &lt;- isplineExtract(gdmMod)

##plot spline(s)
#spline for winter precip (bio19)
plot(exSplines[[1]][,"bio19"], exSplines[[2]][,"bio19"], type="l",
     lwd=3, xlab="Winter precipitation (mm)", ylab="Partial Ecological Distance")

</code></pre>

<hr>
<h2 id='permutateSitePair'>Permutate Site-Pair Table Rows, Internal Function</h2><span id='topic+permutateSitePair'></span>

<h3>Description</h3>

<p>A function which randomizes the rows of a given site-pair table. This function
is called from the <code><a href="#topic+gdm.varImp">gdm.varImp</a></code> function and not needed by the user.
be called directly by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>permutateSitePair(spTab, siteVarTab, indexTab, vNames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="permutateSitePair_+3A_sptab">spTab</code></td>
<td>
<p>A site-pair table.</p>
</td></tr>
<tr><td><code id="permutateSitePair_+3A_sitevartab">siteVarTab</code></td>
<td>
<p>A site x variable table.</p>
</td></tr>
<tr><td><code id="permutateSitePair_+3A_indextab">indexTab</code></td>
<td>
<p>A table of index values for the site-pair table.</p>
</td></tr>
<tr><td><code id="permutateSitePair_+3A_vnames">vNames</code></td>
<td>
<p>Vector of variable names in both the site-pair and
site x variable tables.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new site-pair table with rows randomized.
</p>


<h3>Note</h3>

<p>This function is called from the <code><a href="#topic+gdm.varImp">gdm.varImp</a></code> function
and not needed by the user.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gdm.varImp">gdm.varImp</a></code>
</p>

<hr>
<h2 id='plot.gdm'>Plot Model Fit and I-splines from a Fitted Generalized Dissimilarity
Model.</h2><span id='topic+plot.gdm'></span>

<h3>Description</h3>

<p>plot is used to plot the I-splines and fit of a generalized
dissimilarity model created using the <code><a href="#topic+gdm">gdm</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gdm'
plot(x, plot.layout = c(2, 2), plot.color = "blue",
  plot.linewidth = 2, include.rug = FALSE, rug.sitepair = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.gdm_+3A_x">x</code></td>
<td>
<p>A gdm model object returned from <code><a href="#topic+gdm">gdm</a></code>.</p>
</td></tr>
<tr><td><code id="plot.gdm_+3A_plot.layout">plot.layout</code></td>
<td>
<p>This argument specifies the row and column layout for the
plots, including: (1) a single page plot of observed response data against
the raw linear predictor (ecological distance) from the model, and (2) a
single page plot of the observed response against the predicted response
from the model, i.e. after applying the link function, 1.0 - exp(-y), to the
linear predictor, and (3) the I-splines fitted to the individual predictors.
Default is 2 rows by 2 columns. To produce one predictor plot per page set
plot.layout to c(1,1). The first two model plots are always produced on a
single page each and therefore the layout parameter affects only the layout
of the I-spline plots for those predictors that featured in the model fitting
process (i.e., predictors with all-zero I-spline coefficients are not plotted).</p>
</td></tr>
<tr><td><code id="plot.gdm_+3A_plot.color">plot.color</code></td>
<td>
<p>Color of the data points that are plotted for the overall plots.</p>
</td></tr>
<tr><td><code id="plot.gdm_+3A_plot.linewidth">plot.linewidth</code></td>
<td>
<p>The line width for the regression line over-plotted in
the two overall plots to optimize the display of the line over the data points.</p>
</td></tr>
<tr><td><code id="plot.gdm_+3A_include.rug">include.rug</code></td>
<td>
<p>Whether or not to include a rug plot of the predictor
values used to fit the gdm in the I-spline plots. When set to TRUE, a s
ite-pair table must be supplied for the rug.sitepair argument. Default is FALSE.</p>
</td></tr>
<tr><td><code id="plot.gdm_+3A_rug.sitepair">rug.sitepair</code></td>
<td>
<p>A site-pair table used to add a rug plot of the predictor
values used to fit the gdm in the I-spline plots. This should be the same
site-pair table used to fit the gdm model being plotted. The function does
not check whether the supplied site-pair table matches that used in model fitting.</p>
</td></tr>
<tr><td><code id="plot.gdm_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot returns NULL. Use <code><a href="#topic+summary.gdm">summary.gdm</a></code> to obtain a
synopsis of the model object.
</p>


<h3>References</h3>

<p>Ferrier S, Manion G, Elith J, Richardson, K (2007) Using
generalized dissimilarity modelling to analyse and predict patterns of
beta diversity in regional biodiversity assessment.
<em>Diversity &amp; Distributions</em> 13:252-264.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+isplineExtract">isplineExtract</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##set up site-pair table using the southwest data set
sppData &lt;- southwest[c(1,2,13,14)]
envTab &lt;- southwest[c(2:ncol(southwest))]
sitePairTab &lt;- formatsitepair(sppData, 2, XColumn="Long", YColumn="Lat",
                              sppColumn="species", siteColumn="site",
                              predData=envTab)

##create GDM
gdmMod &lt;- gdm(sitePairTab, geo=TRUE)

##plot GDM
plot(gdmMod, plot.layout=c(3,3))

</code></pre>

<hr>
<h2 id='plotUncertainty'>Plot I-splines With Error Bands Using Bootstrapping.</h2><span id='topic+plotUncertainty'></span>

<h3>Description</h3>

<p>This function estimates uncertainty in the fitted I-splines
by fitting many GDMs using a subsample of the data. The function can run in parallel
on multicore machines to reduce computation time (recommended for large number
of iterations). I-spline plots with error bands (+/- one standard deviation)
are produced showing (1) the variance of I-spline coefficients and (2) a rug plot
indicating how sites used in model fitting are distributed along each gradient.
Function result optionally can be saved to disk as a csv for custom plotting, etc.
The result output table will have 6 columns per predictor, three each for the
x and y values containing the lower bound, full model, and upper bound.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotUncertainty(spTable, sampleSites, bsIters, geo=FALSE,
splines=NULL, knots=NULL, splineCol="blue", errCol="grey80",
plot.linewidth=2.0, plot.layout=c(2,2), parallel=FALSE, cores=2, save=FALSE,
fileName="gdm.plotUncertainy.csv")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotUncertainty_+3A_sptable">spTable</code></td>
<td>
<p>A site-pair table, same as used to fit a <code><a href="#topic+gdm">gdm</a></code>.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_samplesites">sampleSites</code></td>
<td>
<p>The fraction (0-1) of sites to retain from the full
site-pair table when subsampling.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_bsiters">bsIters</code></td>
<td>
<p>The number of bootstrap iterations to perform.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_geo">geo</code></td>
<td>
<p>Same as the <code><a href="#topic+gdm">gdm</a></code> geo argument.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_splines">splines</code></td>
<td>
<p>Same as the <code><a href="#topic+gdm">gdm</a></code> splines argument.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_knots">knots</code></td>
<td>
<p>Same as the <code><a href="#topic+gdm">gdm</a></code> knots argument.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_splinecol">splineCol</code></td>
<td>
<p>The color of the plotted mean spline. The default is &quot;blue&quot;.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_errcol">errCol</code></td>
<td>
<p>The color of shading for the error bands (+/- one standard
deviation around the mean line). The default is &quot;grey80&quot;.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_plot.linewidth">plot.linewidth</code></td>
<td>
<p>The line width of the plotted mean spline line. The
default is 2.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_plot.layout">plot.layout</code></td>
<td>
<p>Same as the <code><a href="#topic+plot.gdm">plot.gdm</a></code> plot.layout argument.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_parallel">parallel</code></td>
<td>
<p>Perform the uncertainty assessment using multiple
cores? Default = FALSE.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_cores">cores</code></td>
<td>
<p>When the parallel argument is set to TRUE, the number of
cores to be registered for the foreach loop. Must be &lt;= the number of cores
in the machine running the function.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_save">save</code></td>
<td>
<p>Save the function result (e.g., for custom plotting)? Default=FALSE.</p>
</td></tr>
<tr><td><code id="plotUncertainty_+3A_filename">fileName</code></td>
<td>
<p>Name of the csv file to save the data frame that contains the function
result. Default = gdm.plotUncertainy.csv. Ignored if save=FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plotUncertainty returns NULL. Saves a csv to disk if save=TRUE.
</p>


<h3>References</h3>

<p>Shryock, D. F., C. A. Havrilla, L. A. DeFalco, T. C. Esque,
N. A. Custer, and T. E. Wood. 2015. Landscape genomics of <em>Sphaeralcea ambigua</em>
in the Mojave Desert: a multivariate, spatially-explicit approach to guide
ecological restoration. <em>Conservation Genetics</em> 16:1303-1317.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.gdm">plot.gdm</a>, <a href="#topic+formatsitepair">formatsitepair</a>, <a href="#topic+subsample.sitepair">subsample.sitepair</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##set up site-pair table using the southwest data set
sppData &lt;- southwest[c(1,2,13,14)]
envTab &lt;- southwest[c(2:ncol(southwest))]
sitePairTab &lt;- formatsitepair(sppData, 2, XColumn="Long", YColumn="Lat",
                              sppColumn="species", siteColumn="site", predData=envTab)

##plot GDM uncertainty using one core
#not run
#plotUncertainty(sitePairTab, sampleSites=0.70, bsIters=5, geo=TRUE, plot.layout=c(3,3))

##plot GDM uncertainty in parallel
#not run
#plotUncertainty(sitePairTab, sampleSites=0.70, bsIters=50, geo=TRUE, plot.layout=c(3,3),
                 #parallel=T, cores=10)

</code></pre>

<hr>
<h2 id='predict.gdm'>Predict Biological Dissimilarities Between Sites or Times Using a
Fitted Generalized Dissimilarity Model</h2><span id='topic+predict.gdm'></span>

<h3>Description</h3>

<p>This function predicts biological distances between sites or times using a
model object returned from <code><a href="#topic+gdm">gdm</a></code>. Predictions between site
pairs require a data frame containing the values of predictors for pairs
of locations, formatted as follows: distance, weights, s1.X, s1.Y, s2.X,
s2.Y, s1.Pred1, s1.Pred2, ..., s1.PredN, s2.Pred1, s2.Pred2, ..., s2.PredN, ...,
Predictions of biological change through time require two raster stacks or
bricks for environmental conditions at two time periods, each with a
layer for each environmental predictor in the fitted model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gdm'
predict(object, data, time=FALSE, predRasts=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.gdm_+3A_object">object</code></td>
<td>
<p>A gdm model object resulting from a call to <code><a href="#topic+gdm">gdm</a></code>.</p>
</td></tr>
<tr><td><code id="predict.gdm_+3A_data">data</code></td>
<td>
<p>Either a data frame containing the values of predictors for pairs
of sites, in the same format and structure as used to fit the model using
<code><a href="#topic+gdm">gdm</a></code> or a raster stack if a prediction of biological change
through time is needed.
</p>
<p>For a data frame, the first two columns - distance and weights - are required
by the function but are not used in the prediction and can therefore be filled
with dummy data (e.g. all zeros). If geo is TRUE, then the s1.X, s1.Y and s2.X,
s2.Y columns will be used for calculating the geographical distance between
each site for inclusion of the geographic predictor term into the GDM model.
If geo is FALSE, then the s1.X, s1.Y, s2.X and s2.Y data columns are ignored.
However these columns are still REQUIRED and can be filled with dummy data
(e.g. all zeroes). The remaining columns are for N predictors for Site 1 and
followed by N predictors for Site 2. The order of the columns must match those
in the site-pair table used to fit the model.
</p>
<p>A raster stack should be provided only when time=T and should contain one
layer for each environmental predictor in the same order as the columns in
the site-pair table used to fit the model.</p>
</td></tr>
<tr><td><code id="predict.gdm_+3A_time">time</code></td>
<td>
<p>TRUE/FALSE: Is the model prediction for biological change through time?</p>
</td></tr>
<tr><td><code id="predict.gdm_+3A_predrasts">predRasts</code></td>
<td>
<p>A raster stack characterizing environmental conditions for a
different time in the past or future, with the same extent, resolution, and
layer order as the data object. Required only if time=T.</p>
</td></tr>
<tr><td><code id="predict.gdm_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>predict returns either a response vector with the same length as the
number of rows in the input data frame or a raster depicting change through time across the study region.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gdm.transform">gdm.transform</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##set up site-pair table using the southwest data set
sppData &lt;- southwest[, c(1,2,14,13)]
envTab &lt;- southwest[, c(2:ncol(southwest))]

# remove soils (no rasters for these)
envTab &lt;- envTab[,-c(2:6)]
sitePairTab &lt;- formatsitepair(sppData, 2, XColumn="Long", YColumn="Lat", sppColumn="species",
                             siteColumn="site", predData=envTab)

gdmMod &lt;- gdm(sitePairTab, geo=TRUE)

##predict GDM
predDiss &lt;- predict(gdmMod, sitePairTab)

##time example
rastFile &lt;- system.file("./extdata/swBioclims.grd", package="gdm")
envRast &lt;- raster::stack(rastFile)

##make some fake climate change data
futRasts &lt;- envRast
##reduce winter precipitation by 25%
futRasts[[3]] &lt;- futRasts[[3]]*0.75

timePred &lt;- predict(gdmMod, envRast, time=TRUE, predRasts=futRasts)
raster::plot(timePred)

</code></pre>

<hr>
<h2 id='southwest'>Species and Environmental Data from Southwestern Australia.</h2><span id='topic+southwest'></span>

<h3>Description</h3>

<p>A data set containing species occurrence and associated environmental
data at 94 sites in southwestern Australia.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>southwest
</code></pre>


<h3>Format</h3>

<p>A data frame with 29364 rows and 14 variables:
</p>

<dl>
<dt>species</dt><dd><p>species name</p>
</dd>
<dt>site</dt><dd><p>site name</p>
</dd>
<dt>awcA</dt><dd><p>plant-available water capacity in soil horizon A</p>
</dd>
<dt>phTotal</dt><dd><p>soil pH</p>
</dd>
<dt>sandA</dt><dd><p>percent sand content in soil horizon A</p>
</dd>
<dt>shcA</dt><dd><p>saturated hydraulic conductivity in soil horizon A</p>
</dd>
<dt>solumDepth</dt><dd><p>soil depth to unweathered parent material</p>
</dd>
<dt>bio5</dt><dd><p>maximum temperature of the coldest month</p>
</dd>
<dt>bio6</dt><dd><p>minimum temperature of the coldest month</p>
</dd>
<dt>bio15</dt><dd><p>precipitation seasonality</p>
</dd>
<dt>bio18</dt><dd><p>precipitation of warmest quarter</p>
</dd>
<dt>bio19</dt><dd><p>precipitation of coldest quarter</p>
</dd>
<dt>Lat</dt><dd><p>latitude</p>
</dd>
<dt>Long</dt><dd><p>longitude</p>
</dd>
</dl>


<hr>
<h2 id='subsample.sitepair'>Remove Sites at Random from a Site-Pair Table</h2><span id='topic+subsample.sitepair'></span>

<h3>Description</h3>

<p>Randomly selects a number of sites from a given site-pair table
and removes them from the site-pair table. It will remove all instances of
the sites randomly selected to be removed in both s1 and s2 positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsample.sitepair(spTable, sampleSites)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsample.sitepair_+3A_sptable">spTable</code></td>
<td>
<p>A site-pair table, same as used to fit a <code><a href="#topic+gdm">gdm</a></code>.</p>
</td></tr>
<tr><td><code id="subsample.sitepair_+3A_samplesites">sampleSites</code></td>
<td>
<p>The fraction (0-1, though a value of 0 would be silly,
wouldn't it?) of <em>sites to retain</em> from the full site-pair table. If
less than 1, this argument will completely remove a fraction of sites such
that they are not used in the permutation routines.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A site-pair table, such as one created by <code><a href="#topic+formatsitepair">formatsitepair</a></code>,
ideally smaller than the one given. In the very rare case where the function
determines not to remove any sites, or should the sampleSites argument be 1,
then the function will return the full site-pair table.
</p>


<h3>Note</h3>

<p>This function removes sites, not just site-pairs (rows) from the
site-pair table. This function is called from several of the other functions
within the gdm package, including the <code><a href="#topic+plotUncertainty">plotUncertainty</a></code> and
<code><a href="#topic+gdm.varImp">gdm.varImp</a></code> functions, for the purposes of subsampling the sites
in the site-pair table.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+formatsitepair">formatsitepair</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##set up site-pair table using the southwest data set
sppData &lt;- southwest[c(1,2,13,14)]
envTab &lt;- southwest[c(2:ncol(southwest))]
sitePairTab &lt;- formatsitepair(sppData, 2, XColumn="Long", YColumn="Lat", sppColumn="species",
                              siteColumn="site", predData=envTab)

subsample.sitepair(sitePairTab, sampleSites=0.7)

</code></pre>

<hr>
<h2 id='summary.gdm'>Summarize a Fitted Generalized Dissimilarity Model</h2><span id='topic+summary.gdm'></span>

<h3>Description</h3>

<p>This function summarizes the gdm model object returned from <code><a href="#topic+gdm">gdm</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gdm'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.gdm_+3A_object">object</code></td>
<td>
<p>A gdm model object resulting from a call to <code><a href="#topic+gdm">gdm</a></code>.</p>
</td></tr>
<tr><td><code id="summary.gdm_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>summary prints its output to the R Console window and returns no value.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gdm">gdm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##set up site-pair table using the southwest data set
sppData &lt;- southwest[, c(1,2,14,13)]
envTab &lt;- southwest[, c(2:ncol(southwest))]
sitePairTab &lt;- formatsitepair(sppData, 2, XColumn="Long", YColumn="Lat", sppColumn="species",
                              siteColumn="site", predData=envTab)

##create GDM
gdmMod &lt;- gdm(sitePairTab, geo=TRUE)

##summary of GDM
summary(gdmMod)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
