<!DOCTYPE html><html><head><title>Help for package GEEaSPU</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GEEaSPU}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GEEaSPU-package'>
<p>Adaptive Association Tests for Multiple Phenotypes using Generalized Estimating Equations (GEE)</p></a></li>
<li><a href='#GEEaSPU'>
<p>An adaptive sum of powered score (SPU) test for single SNP and mutiple phenotypes</p></a></li>
<li><a href='#GEEaSPUpath'>
<p>An adaptive sum of powered score (SPU) test for gene-set (pathway) and mutiple phenotypes</p></a></li>
<li><a href='#GEEaSPUset'>
<p>An adaptive sum of powered score (SPU) test for SNP-set (gene) and mutiple phenotypes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Adaptive Association Tests for Multiple Phenotypes using
Generalized Estimating Equations (GEE)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-08-03</td>
</tr>
<tr>
<td>Author:</td>
<td>Junghi Kim and Wei Pan</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junghi Kim &lt;junghikim0@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides adaptive association tests for SNP level, gene level and pathway level analyses.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GNU General Public License (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), gee, stats</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.6)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-08-04 04:01:40 UTC; junghikim</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-08-04 11:47:48</td>
</tr>
</table>
<hr>
<h2 id='GEEaSPU-package'>
Adaptive Association Tests for Multiple Phenotypes using Generalized Estimating Equations (GEE)
</h2><span id='topic+GEEaSPU-package'></span>

<h3>Description</h3>

<p>Provides adaptive association tests for SNP level, gene level and pathway level analyses.
</p>


<h3>Details</h3>

<p>Functions for adaptive association tests including GEEaSPU, 
GEEaSPUset and GEEaSPUpath. These methods use a weighting scheme for 
testing associations with multiple phenotypes. GEEaSPU function computes a SNP
level p value; GEEaSPUset function can be used for the SNP-set (gene) level
association test, while GEEaSPUpath function is for the gene-set (pathway) level analysis.
</p>


<h3>Author(s)</h3>

<p>Junghi Kim and Wei Pan
</p>


<h3>References</h3>

<p>Kim J, Zhang Y, and Pan W (2016). &quot;Powerful and adaptive testing for multi-trait and multi-SNP associations with GWAS and sequencing data.&quot; <em>Genetics</em> 203(2), 715&ndash;731.
</p>
<p>Zhang Y, Xu Z, Shen, X and  Pan W (2014). &quot;Testing for association with multiple traits in generalized estimation equations, with application to neuroimaging data.&quot; <em>NeuroImage</em> 96, 309&ndash;325.
</p>

<hr>
<h2 id='GEEaSPU'>
An adaptive sum of powered score (SPU) test for single SNP and mutiple phenotypes
</h2><span id='topic+GEEaSPU'></span>

<h3>Description</h3>

<p>Tests whether a SNP is associated with multivariate outcomes;
provides a series of p-values of GEE-SPU(pow) and GEE-aSPU tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GEEaSPU(pheno, geno, Z = NULL, model = "gaussian", corstr = "independence", 
        pow = c(1:8, Inf), n.perm = 1000, null.type = "perm", score.test = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GEEaSPU_+3A_pheno">pheno</code></td>
<td>
<p>a numeric phenotype matrix with each row as a different individual and each column as a separate phenotype.</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_geno">geno</code></td>
<td>
<p>a numeric  vector with each element for an individual genotype.</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_z">Z</code></td>
<td>
<p>a numeric covariate matrix with each row as a different individual and each column as a covariated to be adjested.</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_model">model</code></td>
<td>
<p>a character string specifying the model of the phenotypes. 
Models supported are &quot;gaussian&quot; for a quantitative trait and &quot;binomial&quot; for a binary trait (default = &quot;gaussian&quot;).
</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_corstr">corstr</code></td>
<td>
<p>a character string specifying the correlation structure of phenotypes. The following are permitted: &quot;independence&quot;, &quot;fixed&quot;, &quot;stat_M_dep&quot;, &quot;non_stat_M_dep&quot;, &quot;exchangeable&quot;, &quot;AR-M&quot; and &quot;unstructured&quot; (default = &quot;independence&quot;).
</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_pow">pow</code></td>
<td>
<p>a vector of the power weight to be used at a trait level (default = c(1:8, Inf)).</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_n.perm">n.perm</code></td>
<td>
<p>a numeric value of number of null statistics (default = 1000).</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_null.type">null.type</code></td>
<td>
<p>a character string specifying how to generate null statistics; &quot;perm&quot; is used when null statistics are generated using permutations and  
&quot;sim&quot; is used when null statistics are generated using simulations (default = &quot;perm&quot;).</p>
</td></tr>
<tr><td><code id="GEEaSPU_+3A_score.test">score.test</code></td>
<td>
<p>a logical value indicating whether to include GEEaSPU-Score test along with GEE-Score test (default = FALSE). If TRUE, it computes p-values of GEEaSPU-Score and GEE-Score as well as GEEaSPU test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adaptive association tests for single SNP and multiple phenotypes using GEE.
</p>


<h3>Value</h3>

<p>a vector of p-values from GEE-SPU(pow) tests and GEE-aSPU test.
</p>


<h3>Note</h3>

<p>When large SNP-set (namely large gene size) or large number of phenotypes are included, the permuation based test (null.type = &quot;perm&quot;) is recommended.
</p>
<p>An option &quot;binomial&quot; model only supports the option, null.type = &quot;sim&quot;.
</p>


<h3>Author(s)</h3>

<p>Junghi Kim and Wei Pan
</p>


<h3>References</h3>

<p>Kim J, Zhang Y, and Pan W (2016). &quot;Powerful and adaptive testing for multi-trait and multi-SNP associations with GWAS and sequencing data.&quot; <em>Genetics</em> 203(2), 715&ndash;731.
</p>
<p>Zhang Y, Xu Z, Shen, X and  Pan W (2014). &quot;Testing for association with multiple traits in generalized estimation equations, with application to neuroimaging data.&quot; <em>NeuroImage</em>, 96, 309&ndash;325.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># -- simulating phenotypes
# -- n.subjects: number of subjects
# -- n.traits: number of phenotypes
# -- Sigma: covariance matrix of phenotypes (e.g. AR(1))

set.seed(136)
n.subjects &lt;- 100
n.traits &lt;- 3
sigma &lt;- 2; rho &lt;- 0.5
Sigma0 &lt;- diag(n.traits); 
Sigma &lt;- sigma * rho^abs(row(Sigma0) - col(Sigma0))
eS &lt;- eigen(Sigma, symmetric = TRUE)
ev &lt;- eS$values
X &lt;- matrix(rnorm(n.traits * n.subjects), n.subjects)
pheno &lt;- X %*% diag(sqrt(pmax(ev, 0)), ncol(Sigma)) %*% eS$vectors


# -- simulating genotype
geno &lt;- rbinom(n = n.subjects, size = 2, prob = 0.2)


# -- Computing the p-value of GEEaSPU test with the permutation based method
Pvl &lt;- GEEaSPU(pheno = pheno, geno = geno, Z = NULL, pow = c(1,2,4,Inf), 
        n.perm = 1000, null.type = "perm", score.test = FALSE)


# -- Each element of Pvl is a p value of GEE-SPU(pow) in order
# -- The last element of Pvl is a p value of GEE-aSPU test
Pvl
Pvl[length(Pvl)]


# &gt; Pvl
#     SPU.1     SPU.2     SPU.4   SPU.Inf      aSPU 
# 0.1890000 0.4070000 0.3520000 0.3040000 0.2917083 


# &gt; Pvl[length(Pvl)]
#      aSPU 
# 0.2917083
</code></pre>

<hr>
<h2 id='GEEaSPUpath'>
An adaptive sum of powered score (SPU) test for gene-set (pathway) and mutiple phenotypes
</h2><span id='topic+GEEaSPUpath'></span>

<h3>Description</h3>

<p>Tests whether gene-set (pathway) is associated with multivariate outcomes;
provides a series of p-values of GEE-SPU(pow, pow2, pow3) and GEEaSPUpath tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GEEaSPUpath(pheno, geno, nSNPs, Z = NULL, corstr = "independence", 
           pow = c(1,2,4,8), pow2 = c(1,2,4,8), pow3 = c(1,2,4,8), n.perm = 1000)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GEEaSPUpath_+3A_pheno">pheno</code></td>
<td>
<p>a numeric phenotype matrix with each row as a different individual and each column as a separate phenotype.</p>
</td></tr>
<tr><td><code id="GEEaSPUpath_+3A_geno">geno</code></td>
<td>
<p>a numeric genotype matrix with each row as a different individual and each column as a snp; the SNPs (with the number stored in nSNPs) from one gene are 
stored consecutively from the first gene.</p>
</td></tr>
<tr><td><code id="GEEaSPUpath_+3A_nsnps">nSNPs</code></td>
<td>
<p>A numeric vector, whose length matches to the total number of genes; 
each element of vector indicate the number of SNPs in each gene.</p>
</td></tr>
<tr><td><code id="GEEaSPUpath_+3A_z">Z</code></td>
<td>
<p>a numeric covariate matrix with each row as a different individual and each column as a covariated to be adjested.</p>
</td></tr>
<tr><td><code id="GEEaSPUpath_+3A_corstr">corstr</code></td>
<td>
<p>a character string specifying the correlation structure of phenotypes. The following are permitted: &quot;independence&quot;, &quot;fixed&quot;, &quot;stat_M_dep&quot;, &quot;non_stat_M_dep&quot;, &quot;exchangeable&quot;, &quot;AR-M&quot; and &quot;unstructured&quot; (default = &quot;independence&quot;).
</p>
</td></tr>
<tr><td><code id="GEEaSPUpath_+3A_pow">pow</code></td>
<td>
<p>a vector of the power weight to be used at a SNP level (default = c(1,2,4,8)).</p>
</td></tr>
<tr><td><code id="GEEaSPUpath_+3A_pow2">pow2</code></td>
<td>
<p>a vector of the power weight to be used at a trait level (default = c(1,2,4,8)).</p>
</td></tr>
<tr><td><code id="GEEaSPUpath_+3A_pow3">pow3</code></td>
<td>
<p>a vector of the power weight to be used at a gene level (default = c(1,2,4,8)).</p>
</td></tr>
<tr><td><code id="GEEaSPUpath_+3A_n.perm">n.perm</code></td>
<td>
<p>a numeric value of number of null statistics (default = 1000).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adaptive association tests for gene-set (pathway) and multiple phenotypes using GEE.
</p>


<h3>Value</h3>

<p>a vector of p-values from GEE-SPU(pow, pow2, pow3) tests and GEE-aSPUpath test.
</p>


<h3>Note</h3>

<p>GEEaSPUpath function only supports a case for a quantitative trait (model = &quot;gaussian&quot;) and a permutation based test (null.type = &quot;perm&quot;). 
</p>


<h3>Author(s)</h3>

<p>Junghi Kim and Wei Pan
</p>


<h3>References</h3>

<p>Kim J, Zhang Y, and Pan W (2016). &quot;Powerful and adaptive testing for multi-trait and multi-SNP associations with GWAS and sequencing data.&quot; <em>Genetics</em>, 203(2), 715&ndash;731.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># -- simulating phenotypes
# -- n.subjects: number of subjects
# -- n.traits: number of phenotypes
# -- Sigma: covariance matrix of phenotypes (e.g. AR(1))

set.seed(136)
n.subjects &lt;- 100
n.traits &lt;- 3
sigma &lt;- 2; rho &lt;- 0.5
Sigma0 &lt;- diag(n.traits)
Sigma &lt;- sigma * rho^abs(row(Sigma0) - col(Sigma0))
eS &lt;- eigen(Sigma, symmetric = TRUE)
ev &lt;- eS$values
X &lt;- matrix(rnorm(n.subjects * n.traits), n.subjects)
pheno &lt;- X %*% diag(sqrt(pmax(ev, 0)), ncol(Sigma)) %*% eS$vectors


# -- simulating genotype
# -- Assume we have two genes each of which has 3 and 5 SNPs respectively.
# -- n.geno1: number of SNPs included in the gene1
# -- n.geno2: number of SNPs included in the gene2
# -- nSNPs &lt;- c(3,5)

n.geno1 &lt;- 3
n.geno2 &lt;- 5
maf1 &lt;- 0.2
maf2 &lt;- 0.4
gene1 &lt;- matrix(rbinom(n = n.subjects*n.geno1, size = 2, prob = maf1), ncol = n.geno1)
gene2 &lt;- matrix(rbinom(n = n.subjects*n.geno2, size = 2, prob = maf2), ncol = n.geno2)
geno &lt;- cbind(gene1, gene2)


# -- Computing the p-value of GEEaSPUpath test
Pvl &lt;- GEEaSPUpath(pheno = pheno, geno = geno, nSNPs = c(3,5), Z = NULL, 
        corstr = "independence", pow = c(1,4,8), pow2 = c(1,4,8), 
        pow3 = c(1,4,8), n.perm = 1000)


# -- Each element of Pvl is a p value of GEE-SPU(pow,pow2,pow3) in order
# -- The last element of Pvl is a p value of GEE-aSPUpath test
Pvl
Pvl[length(Pvl)]


# &gt; Pvl
# SPU.1.1.1  SPU.1.1.4  SPU.1.1.8  SPU.1.4.1  SPU.1.4.4  SPU.1.4.8  SPU.1.8.1  SPU.1.8.4 
# 0.00900000 0.05600000 0.07000000 0.06200000 0.08300000 0.11200000 0.06100000 0.08200000 
#  SPU.1.8.8  SPU.4.1.1  SPU.4.1.4  SPU.4.1.8  SPU.4.4.1  SPU.4.4.4  SPU.4.4.8  SPU.4.8.1 
# 0.10600000 0.58100000 0.54300000 0.49200000 0.62400000 0.64000000 0.62700000 0.64900000 
#  SPU.4.8.4  SPU.4.8.8  SPU.8.1.1  SPU.8.1.4  SPU.8.1.8  SPU.8.4.1  SPU.8.4.4  SPU.8.4.8 
# 0.67100000 0.67500000 0.58300000 0.53700000 0.48100000 0.63400000 0.64600000 0.63800000 
#  SPU.8.8.1  SPU.8.8.4  SPU.8.8.8   aSPUpath 
# 0.66000000 0.68100000 0.67900000 0.04395604 


# &gt; Pvl[length(Pvl)]
#   aSPUpath 
# 0.04395604 
</code></pre>

<hr>
<h2 id='GEEaSPUset'>
An adaptive sum of powered score (SPU) test for SNP-set (gene) and mutiple phenotypes
</h2><span id='topic+GEEaSPUset'></span>

<h3>Description</h3>

<p>Tests whether SNP-set (gene) is associated with multivariate outcomes;
provides a series of p-values of GEE-SPU(pow, pow2) and GEEaSPUset tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GEEaSPUset(pheno, geno, Z = NULL, model = "gaussian", corstr = "independence", 
          pow = c(1,2,4,8), pow2 = c(1,2,4,8), n.perm = 1000, 
          null.type = "perm", score.test = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GEEaSPUset_+3A_pheno">pheno</code></td>
<td>
<p>a numeric phenotype matrix with each row as a different individual and each column as a separate phenotype.</p>
</td></tr>
<tr><td><code id="GEEaSPUset_+3A_geno">geno</code></td>
<td>
<p>a numeric genotype matrix with each row as a different individual and each column as a snp.</p>
</td></tr>
<tr><td><code id="GEEaSPUset_+3A_z">Z</code></td>
<td>
<p>a numeric covariate matrix with each row as a different individual and each column as a covariated to be adjested.</p>
</td></tr>
<tr><td><code id="GEEaSPUset_+3A_model">model</code></td>
<td>
<p>a character string specifying the model of the phenotypes. 
Models supported are &quot;gaussian&quot; for a quantitative trait and &quot;binomial&quot; for a binary trait (default = &quot;gaussian&quot;).
</p>
</td></tr>
<tr><td><code id="GEEaSPUset_+3A_corstr">corstr</code></td>
<td>
<p>a character string specifying the correlation structure of phenotypes. The following are permitted: &quot;independence&quot;, &quot;fixed&quot;, &quot;stat_M_dep&quot;, &quot;non_stat_M_dep&quot;, &quot;exchangeable&quot;, &quot;AR-M&quot; and &quot;unstructured&quot; (default = &quot;independence&quot;).
</p>
</td></tr>
<tr><td><code id="GEEaSPUset_+3A_pow">pow</code></td>
<td>
<p>a vector of the power weight to be used at a SNP level (default = c(1,2,4,8)).</p>
</td></tr>
<tr><td><code id="GEEaSPUset_+3A_pow2">pow2</code></td>
<td>
<p>a vector of the power weight to be used at a trait level (default = c(1,2,4,8)).</p>
</td></tr>
<tr><td><code id="GEEaSPUset_+3A_n.perm">n.perm</code></td>
<td>
<p>a numeric value of number of null statistics (default = 1000).</p>
</td></tr>
<tr><td><code id="GEEaSPUset_+3A_null.type">null.type</code></td>
<td>
<p>a character string specifying how to generate null statistics; &quot;perm&quot; is used when null statistics are generated using permutations and  
&quot;sim&quot; is used when null statistics are generated using simulations (default = &quot;perm&quot;).</p>
</td></tr>
<tr><td><code id="GEEaSPUset_+3A_score.test">score.test</code></td>
<td>
<p>a logical value indicating whether to include GEEaSPU-Score test along with GEE-Score test (default = FALSE). If TRUE, it computes p-values of GEEaSPU-Score and GEE-Score as well as GEEaSPU test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adaptive association tests for SNP-set (gene) and multiple phenotypes using GEE.
</p>


<h3>Value</h3>

<p>a vector of p-values from GEE-SPU(pow, pow2) tests and GEE-aSPUset test.
</p>


<h3>Note</h3>

<p>When large SNP-set (namely large gene size) or large number of phenotypes are included, 
the permuation based test (null.type = &quot;perm&quot;) is recommended.
</p>
<p>An option &quot;binomial&quot; model only supports the option, null.type=&quot;sim&quot;.
</p>


<h3>Author(s)</h3>

<p>Junghi Kim and Wei Pan
</p>


<h3>References</h3>

<p>Kim J, Zhang Y, and Pan W (2016). &quot;Powerful and adaptive testing for multi-trait and multi-SNP associations with GWAS and sequencing data.&quot; <em>Genetics</em>, 203(2), 715&ndash;731.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># -- simulating phenotypes
# -- n.subjects: number of subjects
# -- n.traits: number of phenotypes
# -- Sigma: covariance matrix of phenotypes (e.g. AR(1))

set.seed(136)
n.subjects &lt;- 100
n.traits &lt;- 3
sigma &lt;- 2; rho &lt;- 0.5
Sigma0 &lt;- diag(n.traits); 
Sigma &lt;- sigma * rho^abs(row(Sigma0) - col(Sigma0))
eS &lt;- eigen(Sigma, symmetric = TRUE)
ev &lt;- eS$values
X &lt;- matrix(rnorm(n.subjects * n.traits), n.subjects)
pheno &lt;- X %*% diag(sqrt(pmax(ev, 0)), ncol(Sigma)) %*% eS$vectors


# -- simulating genotype
# -- n.geno: number of SNPs included in the SNP set/gene

n.geno &lt;- 3
maf &lt;- 0.2
geno &lt;- matrix(rbinom(n = n.subjects * n.geno, size = 2, prob = maf), ncol = n.geno)


# -- Computing the p-value of GEEaSPUpath test with the permutation based method
Pvl &lt;- GEEaSPUset(pheno = pheno, geno = geno, Z = NULL, model = "gaussian", 
        corstr = "independence", pow = c(1,4,Inf), pow2 = c(1,4,Inf), n.perm = 1000,         
        null.type = "perm", score.test = FALSE)


# -- Each element of Pvl is a p value of GEE-SPU(pow,pow2) in order
# -- The last element of Pvl is a p value of GEE-aSPUset test
Pvl
Pvl[length(Pvl)]


# &gt; Pvl
#     SPU.1.1     SPU.1.4   SPU.1.Inf     SPU.4.1     SPU.4.4   SPU.4.Inf   SPU.Inf.1 
#  0.01400000  0.08800000  0.07200000  0.53000000  0.41000000  0.32100000  0.55100000 
#   SPU.Inf.4 SPU.Inf.Inf     aSPUset 
#  0.48700000  0.41000000  0.04095904 
 
 
# &gt; Pvl[length(Pvl)]
#    aSPUset 
# 0.04095904
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
