<!DOCTYPE html><html lang="en"><head><title>Help for package geeVerse</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {geeVerse}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#compile_result'><p>Compile Results from qpgee()</p></a></li>
<li><a href='#CVfit'><p>Cross-Validation for Generalized Estimating Equations (GEE)</p></a></li>
<li><a href='#generateData'><p>Generate Data for Simulation</p></a></li>
<li><a href='#PGEE'><p>PGEE accelerated with RCpp</p></a></li>
<li><a href='#predict.qpgee'><p>Predict method for qpgee model objects</p></a></li>
<li><a href='#qpgee'><p>Quantile Penalized Generalized Estimating Equations with Auto Selected Penalty level</p></a></li>
<li><a href='#qpgee.est'><p>Quantile Penalized Generalized Estimating Equations (QPGEE) Estimation Function</p></a></li>
<li><a href='#Siga_cov'><p>Generate Covariance Matrix</p></a></li>
<li><a href='#simuGene'><p>A Simulated Genetic Data from HapGen2</p></a></li>
<li><a href='#yeastG1'><p>A Subset of Yeast Cell Cycle Gene Expression Data (G1 Phase)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Comprehensive Analysis of High Dimensional Longitudinal Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.2</td>
</tr>
<tr>
<td>Description:</td>
<td>To provide a comprehensive analysis of high dimensional longitudinal
    data,this package provides analysis for any combination of 1) simultaneous
    variable selection and estimation, 2) mean regression or quantile regression
    for heterogeneous data, 3) cross-sectional or longitudinal data, 4) balanced
    or imbalanced data, 5) moderate, high or even ultra-high dimensional data,
    via computationally efficient implementations of penalized generalized
    estimating equations.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>mvtnorm, quantreg, Rcpp (&ge; 0.10.2), doParallel, foreach</td>
</tr>
<tr>
<td>Suggests:</td>
<td>methods</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-12 05:40:07 UTC; tianhaizu</td>
</tr>
<tr>
<td>Author:</td>
<td>Tianhai Zu [aut, cre],
  Brittany Green [aut, ctb],
  Yan Yu [aut, ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tianhai Zu &lt;zuti@mail.uc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-12 10:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='compile_result'>Compile Results from qpgee()</h2><span id='topic+compile_result'></span>

<h3>Description</h3>

<p>This function reports correct percentage, TP, FP, MSE and MAD from a (list of)
fitted qpgee model comparing to the true betas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compile_result(qpgee_results, beta0, threshold = 10^-3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compile_result_+3A_qpgee_results">qpgee_results</code></td>
<td>
<p>A (list of) fitted qpgee model.</p>
</td></tr>
<tr><td><code id="compile_result_+3A_beta0">beta0</code></td>
<td>
<p>True beta used in true data generation process.</p>
</td></tr>
<tr><td><code id="compile_result_+3A_threshold">threshold</code></td>
<td>
<p>Integer, the threshold to determine whether a esimated beta should
be consider as 0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector contains correct percentage, TP, FP, MSE and MAD and its standard
error if Monte Carlo simulations.
</p>

<hr>
<h2 id='CVfit'>Cross-Validation for Generalized Estimating Equations (GEE)</h2><span id='topic+CVfit'></span>

<h3>Description</h3>

<p>This function performs k-fold cross-validation for model selection in the context
of Generalized Estimating Equations (GEE). It is designed to evaluate the performance
of different models specified by a range of lambda values, choosing the one that
minimizes the cross-validation criterion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CVfit(
  formula,
  id,
  data,
  family,
  scale.fix,
  scale.value,
  fold,
  lambda.vec,
  pindex,
  eps,
  maxiter,
  tol,
  corstr = "independence",
  ncore = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CVfit_+3A_formula">formula</code></td>
<td>
<p>an object of class <code>"formula"</code> (or one that can be coerced to that class):
a symbolic description of the model to be fitted.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_id">id</code></td>
<td>
<p>a vector which identifies the cluster/group for each observation.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_data">data</code></td>
<td>
<p>an optional data frame containing the variables in the model.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_family">family</code></td>
<td>
<p>a description of the error distribution and link function to be used in the model.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_scale.fix">scale.fix</code></td>
<td>
<p>logical; if <code>TRUE</code>, the scale parameter is fixed to <code>scale.value</code>.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_scale.value">scale.value</code></td>
<td>
<p>the value of the scale parameter when <code>scale.fix</code> is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_fold">fold</code></td>
<td>
<p>the number of folds to be used in the cross-validation.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_lambda.vec">lambda.vec</code></td>
<td>
<p>a vector of lambda values for which the cross-validation error will be calculated.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_pindex">pindex</code></td>
<td>
<p>an optional numeric vector specifying a parameter index.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_eps">eps</code></td>
<td>
<p>the threshold for convergence criteria.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_maxiter">maxiter</code></td>
<td>
<p>the maximum number of iterations for the convergence of the algorithm.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_tol">tol</code></td>
<td>
<p>the tolerance level for the convergence of the algorithm.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_corstr">corstr</code></td>
<td>
<p>the correlation structure used.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_ncore">ncore</code></td>
<td>
<p>if greater than 1, the function will use parallel computation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this is a re-implemented version with parallel computing.
</p>


<h3>Value</h3>

<p>An object of class <code>"CVfit"</code>, which is a list containing:
</p>

<dl>
<dt><code>fold</code></dt><dd><p>The number of folds used in the cross-validation.</p>
</dd>
<dt><code>lam.vect</code></dt><dd><p>The vector of lambda values tested.</p>
</dd>
<dt><code>cv.vect</code></dt><dd><p>The cross-validation error for each lambda.</p>
</dd>
<dt><code>lam.opt</code></dt><dd><p>The lambda value that resulted in the minimum cross-validation error.</p>
</dd>
<dt><code>cv.min</code></dt><dd><p>The minimum cross-validation error.</p>
</dd>
<dt><code>call</code></dt><dd><p>The matched call.</p>
</dd>
</dl>


<hr>
<h2 id='generateData'>Generate Data for Simulation</h2><span id='topic+generateData'></span>

<h3>Description</h3>

<p>This function generates simulated data including the predictor matrix 'X' and the response vector 'y',
based on the specified parameters. The function allows for the simulation of data under different settings
of correlation, distribution, and the number of observations and subjects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateData(
  nsub,
  nobs,
  p,
  beta0,
  rho,
  correlation = "AR1",
  dis = "normal",
  ka = 0,
  SNPs = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generateData_+3A_nsub">nsub</code></td>
<td>
<p>Integer, the number of subjects.</p>
</td></tr>
<tr><td><code id="generateData_+3A_nobs">nobs</code></td>
<td>
<p>Integer or numeric vector, the number of observations per subject.</p>
</td></tr>
<tr><td><code id="generateData_+3A_p">p</code></td>
<td>
<p>Integer, the number of predictors.</p>
</td></tr>
<tr><td><code id="generateData_+3A_beta0">beta0</code></td>
<td>
<p>Numeric vector, initial coefficients for the first few predictors.</p>
</td></tr>
<tr><td><code id="generateData_+3A_rho">rho</code></td>
<td>
<p>Numeric, the correlation coefficient used in generating correlated errors.</p>
</td></tr>
<tr><td><code id="generateData_+3A_correlation">correlation</code></td>
<td>
<p>Character, the correlation of correlation structure (default is autoregressive).</p>
</td></tr>
<tr><td><code id="generateData_+3A_dis">dis</code></td>
<td>
<p>Character, the distribution of errors (&quot;normal&quot; or &quot;t&quot;).</p>
</td></tr>
<tr><td><code id="generateData_+3A_ka">ka</code></td>
<td>
<p>1 for heterogeneous errors and 0 for homogeneous errors.</p>
</td></tr>
<tr><td><code id="generateData_+3A_snps">SNPs</code></td>
<td>
<p>User can provide simulated or real SNPs for genetic data simulation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two elements: 'X', the matrix of predictors, and 'y', the response vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim_data &lt;- generateData(nsub = 100, nobs = rep(10, 100),  p = 200,
                         beta0 = c(rep(1,7),rep(0,193)), rho = 0.6, correlation = "AR1",
                          dis = "normal", ka = 1)
</code></pre>

<hr>
<h2 id='PGEE'>PGEE accelerated with RCpp</h2><span id='topic+PGEE'></span>

<h3>Description</h3>

<p>A function to fit penalized generalized estimating equation model.
This function was re-wrote partly with RCPP and RCPPEigen for better computation efficiency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PGEE(
  formula,
  id,
  data,
  na.action = NULL,
  family = gaussian(link = "identity"),
  corstr = "independence",
  Mv = NULL,
  beta_int = NULL,
  R = NULL,
  scale.fix = TRUE,
  scale.value = 1,
  lambda,
  pindex = NULL,
  eps = 10^-6,
  maxiter = 30,
  tol = 10^-3,
  silent = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PGEE_+3A_formula">formula</code></td>
<td>
<p>A formula expression <code>response ~ predictors</code>;</p>
</td></tr>
<tr><td><code id="PGEE_+3A_id">id</code></td>
<td>
<p>A vector for identifying subjects/clusters.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_data">data</code></td>
<td>
<p>A data frame which stores the variables in <code>formula</code> with <code>id</code> variable.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_na.action">na.action</code></td>
<td>
<p>A function to remove missing values from the data. Only <code>na.omit</code> is allowed here.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_family">family</code></td>
<td>
<p>A <code>family</code> object: a list of functions and expressions for defining <code>link</code> and
<code>variance</code> functions. Families supported in <code>PGEE</code> are <code>binomial</code>, <code>gaussian</code>, <code>gamma</code> and
<code>poisson</code>. The <code>links</code>, which are not available in <code>gee</code>, is not available here. The default family
is <code>gaussian</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_corstr">corstr</code></td>
<td>
<p>A character string, which specifies the correlation of correlation structure.
Structures supported in <code>PGEE</code> are <code>"AR-1"</code>,<code>"exchangeable"</code>, <code>"fixed"</code>, <code>"independence"</code>,
<code>"stat_M_dep"</code>,<code>"non_stat_M_dep"</code>, and <code>"unstructured"</code>. The default <code>corstr</code> correlation is
<code>"independence"</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_mv">Mv</code></td>
<td>
<p>If either <code>"stat_M_dep"</code>, or <code>"non_stat_M_dep"</code> is specified in <code>corstr</code>, then this
assigns a numeric value for <code>Mv</code>. Otherwise, the default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_beta_int">beta_int</code></td>
<td>
<p>User specified initial values for regression parameters. The default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_r">R</code></td>
<td>
<p>If <code>corstr = "fixed"</code> is specified, then <code>R</code> is a square matrix of dimension maximum cluster
size containing the user specified correlation. Otherwise, the default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_scale.fix">scale.fix</code></td>
<td>
<p>A logical variable; if true, the scale parameter is fixed at the value of <code>scale.value</code>.
The default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_scale.value">scale.value</code></td>
<td>
<p>If <code>scale.fix = TRUE</code>, this assigns a numeric value to which the scale parameter should be
fixed. The default value is 1.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_lambda">lambda</code></td>
<td>
<p>A numerical value for the penalization parameter of the scad function, which is estimated via
cross-validation.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_pindex">pindex</code></td>
<td>
<p>An index vector showing the parameters which are not subject to penalization. The default value
is <code>NULL</code>. However, in case of a model with intercept, the intercept parameter should be never penalized.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_eps">eps</code></td>
<td>
<p>A numerical value for the epsilon used in minorization-maximization algorithm. The default value is
<code>10^-6</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_maxiter">maxiter</code></td>
<td>
<p>The number of iterations that is used in the estimation algorithm. The default value is <code>25</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_tol">tol</code></td>
<td>
<p>The tolerance level that is used in the estimation algorithm. The default value is <code>10^-3</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_silent">silent</code></td>
<td>
<p>A logical variable; if false, the regression parameter estimates at each iteration are
printed. The default value is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a PGEE object, which includes:
fitted coefficients - the fitted single index coefficients with unit norm and first component being non negative
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#generate data
set.seed(2021)
sim_data &lt;- generateData(nsub = 100, nobs = rep(10, 100),  p = 100,
                         c(rep(1,7),rep(0,93)), rho = 0.6, correlation = "AR1",
                          dis = "normal", ka = 1)

X=sim_data$X
y=sim_data$y
id = rep(1:100, each = 10)
data = data.frame(X,y,id)

PGEE_fit = PGEE("y ~.-id-1",id = id, data = data,corstr = "exchangeable",lambda=0.01)
PGEE_fit$coefficients
</code></pre>

<hr>
<h2 id='predict.qpgee'>Predict method for qpgee model objects</h2><span id='topic+predict.qpgee'></span>

<h3>Description</h3>

<p>This function makes predictions from a &quot;qpgee&quot; model object. When 'newdata' is not provided,
it returns predictions using the original data the model was fitted on. If 'newdata' is supplied
(through '...'), it uses this new data for prediction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'qpgee'
predict(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.qpgee_+3A_object">object</code></td>
<td>
<p>A &quot;qpgee&quot; model object.</p>
</td></tr>
<tr><td><code id="predict.qpgee_+3A_...">...</code></td>
<td>
<p>Additional arguments to the function. Can include 'newdata', a dataframe
containing the new data to predict on. The structure of 'newdata' should match that of the data
the model was originally fitted with, specifically in terms of the variables it contains.
Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If 'newdata' is not supplied, returns a vector of predictions based on the fitted values
and handling of NAs specified in the model object. If 'newdata' is supplied, returns a vector
of predictions for the new data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage:
sim_data &lt;- generateData(nsub = 100, nobs = rep(10, 100),  p = 100,
                         beta0 = c(rep(1,7),rep(0,93)), rho = 0.6, correlation = "AR1",
                          dis = "normal", ka = 1)

X=sim_data$X
y=sim_data$y

#fit qpgee
qpgee.fit = qpgee(X,y,tau=0.5,nobs=rep(10, 100),lambda = 0.1)
predict(qpgee.fit)

</code></pre>

<hr>
<h2 id='qpgee'>Quantile Penalized Generalized Estimating Equations with Auto Selected Penalty level</h2><span id='topic+qpgee'></span>

<h3>Description</h3>

<p>This function automatically select the penalty level by going through a list
of lambdas, and select the best level of penalty with high-dimensional BIC
(HBIC) or cross-validation (CV).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpgee(
  x,
  y,
  tau = 0.5,
  method = "HBIC",
  ncore = 1,
  nobs = rep(1, length(y)),
  correlation = "exchangeable",
  lambda = NULL,
  intercept = FALSE,
  f0 = NULL,
  betaint = NULL,
  max_it = 100,
  cutoff = 10^-4
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qpgee_+3A_x">x</code></td>
<td>
<p>A matrix of predictors.</p>
</td></tr>
<tr><td><code id="qpgee_+3A_y">y</code></td>
<td>
<p>A numeric vector of response variables.</p>
</td></tr>
<tr><td><code id="qpgee_+3A_tau">tau</code></td>
<td>
<p>The quantile to be estimated (default is 0.5, the median).</p>
</td></tr>
<tr><td><code id="qpgee_+3A_method">method</code></td>
<td>
<p>The criterion to select level of penalty. Currently it only
supports &quot;HBIC&quot;.</p>
</td></tr>
<tr><td><code id="qpgee_+3A_ncore">ncore</code></td>
<td>
<p>A numeric value specifying how many core to use.</p>
</td></tr>
<tr><td><code id="qpgee_+3A_nobs">nobs</code></td>
<td>
<p>A numeric vector indicating the number of observations per subject.</p>
</td></tr>
<tr><td><code id="qpgee_+3A_correlation">correlation</code></td>
<td>
<p>A string specifying the working correlation structure.
Options include &quot;exchangeable&quot; (Exchangeable), &quot;AR1&quot; (Autoregressive),
&quot;Tri&quot; (Tri-diagonal), and &quot;exchangeable&quot; (Independent).</p>
</td></tr>
<tr><td><code id="qpgee_+3A_lambda">lambda</code></td>
<td>
<p>A vector of penalty parameter for regularization. If not provided,
a grid will be provided by this function.</p>
</td></tr>
<tr><td><code id="qpgee_+3A_intercept">intercept</code></td>
<td>
<p>Whether to include an intercept when estimating.</p>
</td></tr>
<tr><td><code id="qpgee_+3A_f0">f0</code></td>
<td>
<p>estimated conditional error distributions.</p>
</td></tr>
<tr><td><code id="qpgee_+3A_betaint">betaint</code></td>
<td>
<p>Initial values for the beta coefficients. If NULL,
non-longitudinal quantile regression is used for initialization.</p>
</td></tr>
<tr><td><code id="qpgee_+3A_max_it">max_it</code></td>
<td>
<p>Maximum number of iterations (default is 100).</p>
</td></tr>
<tr><td><code id="qpgee_+3A_cutoff">cutoff</code></td>
<td>
<p>Threshold for coefficient shrinkage (default is 0.1).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<table role = "presentation">
<tr><td><code>beta</code></td>
<td>
<p>Estimated beta coefficients.</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>Fitted values of the linear predictor.</p>
</td></tr>
<tr><td><code>R</code></td>
<td>
<p>Estimated working correlation matrix.</p>
</td></tr>
<tr><td><code>X_selected</code></td>
<td>
<p>Indices of selected predictors.</p>
</td></tr>
<tr><td><code>mcl</code></td>
<td>
<p>Mean check loss.</p>
</td></tr>
<tr><td><code>hbic</code></td>
<td>
<p>Hannan-Quinn Information Criterion value.</p>
</td></tr>
<tr><td><code>converge</code></td>
<td>
<p>Boolean indicating whether the algorithm converged.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage:

sim_data &lt;- generateData(nsub = 20, nobs = rep(10, 20),  p = 20,
                         beta0 = c(rep(1,5),rep(0,15)), rho = 0.1, correlation = "AR1",
                          dis = "normal", ka = 1)

X=sim_data$X
y=sim_data$y

#fit qpgee with auto selected lambda
qpgee.fit = qpgee(X,y,tau=0.5,nobs=rep(10, 20),ncore=1)
qpgee.fit$beta

</code></pre>

<hr>
<h2 id='qpgee.est'>Quantile Penalized Generalized Estimating Equations (QPGEE) Estimation Function</h2><span id='topic+qpgee.est'></span>

<h3>Description</h3>

<p>This function implements Quantile Penalized Generalized Estimating Equations
(QPGEE) for longitudinal data analysis. It estimates parameters using a
penalized quantile regression approach within a GEE framework, allowing for
different working correlation structures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpgee.est(
  x,
  y,
  tau = 0.5,
  nobs = rep(1, length(y)),
  correlation = "exchangeable",
  lambda = 0.1,
  intercept = FALSE,
  betaint = NULL,
  f0 = NULL,
  max_it = 100,
  cutoff = 10^-4
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qpgee.est_+3A_x">x</code></td>
<td>
<p>A matrix of predictors.</p>
</td></tr>
<tr><td><code id="qpgee.est_+3A_y">y</code></td>
<td>
<p>A numeric vector of response variables.</p>
</td></tr>
<tr><td><code id="qpgee.est_+3A_tau">tau</code></td>
<td>
<p>The quantile to be estimated (default is 0.5, the median).</p>
</td></tr>
<tr><td><code id="qpgee.est_+3A_nobs">nobs</code></td>
<td>
<p>A numeric vector indicating the number of observations per subject.</p>
</td></tr>
<tr><td><code id="qpgee.est_+3A_correlation">correlation</code></td>
<td>
<p>A string specifying the working correlation structure.
Options include &quot;exchangeable&quot; (Exchangeable), &quot;AR1&quot; (Autoregressive),
&quot;Tri&quot; (Tri-diagonal), &quot;independence&quot; (Independent), and &quot;unstructured&quot;.</p>
</td></tr>
<tr><td><code id="qpgee.est_+3A_lambda">lambda</code></td>
<td>
<p>The penalty parameter for regularization (default is 0.1).</p>
</td></tr>
<tr><td><code id="qpgee.est_+3A_intercept">intercept</code></td>
<td>
<p>Whether to include an intercept when estimating.</p>
</td></tr>
<tr><td><code id="qpgee.est_+3A_betaint">betaint</code></td>
<td>
<p>Initial values for the beta coefficients. If NULL,
non-longitudinal quantile regression is used for initialization.</p>
</td></tr>
<tr><td><code id="qpgee.est_+3A_f0">f0</code></td>
<td>
<p>estimated conditional error distributions.</p>
</td></tr>
<tr><td><code id="qpgee.est_+3A_max_it">max_it</code></td>
<td>
<p>Maximum number of iterations (default is 100).</p>
</td></tr>
<tr><td><code id="qpgee.est_+3A_cutoff">cutoff</code></td>
<td>
<p>Threshold for coefficient shrinkage (default is 0.1).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<table role = "presentation">
<tr><td><code>beta</code></td>
<td>
<p>Estimated beta coefficients.</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>Fitted values of the linear predictor.</p>
</td></tr>
<tr><td><code>R</code></td>
<td>
<p>Estimated working correlation matrix.</p>
</td></tr>
<tr><td><code>X_selected</code></td>
<td>
<p>Indices of selected predictors.</p>
</td></tr>
<tr><td><code>mcl</code></td>
<td>
<p>Mean check loss.</p>
</td></tr>
<tr><td><code>hbic</code></td>
<td>
<p>Hannan-Quinn Information Criterion value.</p>
</td></tr>
<tr><td><code>converge</code></td>
<td>
<p>Boolean indicating whether the algorithm converged.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage:
sim_data &lt;- generateData(nsub = 100, nobs = rep(10, 100),  p = 100,
                         beta0 = c(rep(1,7),rep(0,93)), rho = 0.6, correlation = "AR1",
                          dis = "normal", ka = 1)

X=sim_data$X
y=sim_data$y

#fit qpgee
qpgee.fit = qpgee.est(X,y,tau=0.5,nobs=rep(10, 100))
qpgee.fit$beta

</code></pre>

<hr>
<h2 id='Siga_cov'>Generate Covariance Matrix</h2><span id='topic+Siga_cov'></span>

<h3>Description</h3>

<p>This function generates a covariance matrix based on the specified correlation structure.
The function supports &quot;compound symmetry&quot; (cs) and &quot;autoregressive&quot; (ar) correlation structures,
as well as an identity matrix as the default option when neither &quot;cs&quot; nor &quot;AR1&quot; is specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Siga_cov(rho, correlation, nt)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Siga_cov_+3A_rho">rho</code></td>
<td>
<p>Numeric, the correlation coefficient used for generating the covariance matrix.
For &quot;cs&quot; or &quot;exchangeable&quot;, it represents the common correlation between any two observations.
For &quot;AR1&quot;, it represents the correlation between two consecutive observations,
with the correlation decreasing for observations further apart.</p>
</td></tr>
<tr><td><code id="Siga_cov_+3A_correlation">correlation</code></td>
<td>
<p>Character, specifies the correlation of correlation structure for the covariance matrix.
Options are &quot;cs&quot; or &quot;exchangeable&quot; for compound symmetry, &quot;AR1&quot; for autoregressive, and any other input
will result in an identity matrix.</p>
</td></tr>
<tr><td><code id="Siga_cov_+3A_nt">nt</code></td>
<td>
<p>Integer, the dimension of the square covariance matrix (number of time points or observations).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A square matrix of dimension 'nt' representing the specified covariance structure.
</p>

<hr>
<h2 id='simuGene'>A Simulated Genetic Data from HapGen2</h2><span id='topic+simuGene'></span>

<h3>Description</h3>

<p>The 'simuGene' dataset contains 500 SNPs simulated  data from
a commonly used tool for genetic data, HapGen2. We re-sampled existing genotype
data to create this simulated data. The genotype data we resample from is the
publicly available 1000 Genomes Project data. More specifically, we use resampled
from chromosome 14.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simuGene
</code></pre>


<h3>Format</h3>

<p>A data frame with 1000 rows (subjects) and 500 columns (SNPs).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(simuGene)
head(simuGene)
</code></pre>

<hr>
<h2 id='yeastG1'>A Subset of Yeast Cell Cycle Gene Expression Data (G1 Phase)</h2><span id='topic+yeastG1'></span>

<h3>Description</h3>

<p>The 'yeastG1' dataset contains gene expression data from the yeast cell
cycle during the G1 phase.
The original dataset (Spellman et al. 1998) includes expression levels for 6178 genes measured at 18 time points.
And this is a subset of 283 cell-cycled-regularized genes observed over 4 time
points at G1 stage and the standardized binding probabilities of a total of 96 TFs
obtained from
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yeastG1
</code></pre>


<h3>Format</h3>

<p>A data frame with 1132 rows and 99 columns.
</p>
<p>The dataset contains gene expression levels for the following transcription factors:
ABF1, ACE2, ADR1, ARG80, ARG81, ARO80, ASH1, BAS1, CAD1, CBF1, CIN5, CRZ1,
CUP9, DAL81, DAL82, DIG1, DOT6, FHL1, FKH1, FKH2, FZF1, GAL4, GAT1, GAT3,
GCN4, GCR1, GCR2, GLN3, GRF10.Pho2., GTS1, HAL9, HAP2, HAP3, HAP4, HAP5,
HIR1, HIR2, HMS1, HSF1, IME4, INO2, INO4, IXR1, LEU3, MAC1, MAL13, MATa1,
MBP1, MCM1, MET31, MET4, MIG1, MOT3, MSN1, MSN4, MSS11, MTH1, NDD1, NRG1,
PDR1, PHD1, PHO4, PUT3, RAP1, RCS1, REB1, RFX1, RGM1, RLM1, RME1, ROX1,
RPH1, RTG1, RTG3, SFP1, SIG1, SIP4, SKN7, SMP1, SOK2, SRD1, STB1, STE12,
STP1, STP2, SUM1, SWI4, SWI5, SWI6, YAP1, YAP5, YAP6, YFL044C, YJL206C,
ZAP1, ZMS1
</p>


<h3>Source</h3>

<p>Spellman, P. T., Sherlock, G., Zhang, M. Q., Iyer, V. R., Anders, K., Eisen, M. B., ... &amp; Futcher, B. (1998).
Comprehensive identification of cell cycle-regulated genes of the yeast Saccharomyces cerevisiae by microarray hybridization.
Molecular biology of the cell, 9(12), 3273-3297.
</p>
<p>Wang, L., Zhou, J., and Qu, A. (2012). Penalized generalized estimating equations for high-dimensional longitudinal data anaysis. <em>Biometrics</em>, <b>68</b>, 353&ndash;360.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(yeastG1)
head(yeastG1)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
