<!DOCTYPE html><html><head><title>Help for package GEInter</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GEInter}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AR'><p>The covariance matrix with an autoregressive (AR) structure among variables</p></a></li>
<li><a href='#Augmented.data'><p>Accommodating missingness in environmental measurements in gene-environment interaction analysis</p></a></li>
<li><a href='#bic.BLMCP'><p>BIC for BLMCP</p></a></li>
<li><a href='#bic.PTReg'><p>BIC for PTReg</p></a></li>
<li><a href='#BLMCP'><p>Accommodating missingness in environmental measurements in gene-environment interaction</p>
analysis: penalized estimation and selection</a></li>
<li><a href='#coef.bic.BLMCP'><p>Extract coefficients from a &quot;bic.BLMCP&quot; object</p></a></li>
<li><a href='#coef.bic.PTReg'><p>Extract coefficients from a &quot;bic.PTReg&quot; object</p></a></li>
<li><a href='#coef.BLMCP'><p>Extract coefficients from a &quot;BLMCP&quot; object</p></a></li>
<li><a href='#coef.PTReg'><p>Extract coefficients from a &quot;PTReg&quot; object</p></a></li>
<li><a href='#coef.RobSBoosting'><p>Extract coefficients from a &quot;RobSBoosting&quot; object</p></a></li>
<li><a href='#HNSCC'><p>A data frame containing the TCGA head and neck squamous cell carcinoma (HNSCC) data.</p></a></li>
<li><a href='#Miss.boosting'><p>Robust gene-environment interaction analysis approach via sparse boosting, where the</p>
missingness in environmental measurements is effectively accommodated using multiple
imputation approach</a></li>
<li><a href='#plot.bic.BLMCP'><p>Plot coefficients from a &quot;bic.BLMCP&quot; object</p></a></li>
<li><a href='#plot.bic.PTReg'><p>Plot coefficients from a &quot;bic.PTReg&quot; object</p></a></li>
<li><a href='#plot.BLMCP'><p>Plot coefficients from a &quot;BLMCP&quot; object</p></a></li>
<li><a href='#plot.Miss.boosting'><p>Plot coefficients from a &quot;Miss.boosting&quot; object</p></a></li>
<li><a href='#plot.PTReg'><p>Plot coefficients from a &quot;PTReg&quot; object</p></a></li>
<li><a href='#plot.RobSBoosting'><p>Plot coefficients from a &quot;RobSBoosting&quot; object</p></a></li>
<li><a href='#predict.bic.BLMCP'><p>Make predictions from a &quot;bic.BLMCP&quot; object.</p></a></li>
<li><a href='#predict.bic.PTReg'><p>Make predictions from a &quot;bic.PTReg&quot; object</p></a></li>
<li><a href='#predict.BLMCP'><p>Make predictions from a &quot;BLMCP&quot; object</p></a></li>
<li><a href='#predict.Miss.boosting'><p>Make predictions from a &quot;Miss.boosting&quot; object</p></a></li>
<li><a href='#predict.PTReg'><p>Make predictions from a &quot;PTReg&quot; object</p></a></li>
<li><a href='#predict.RobSBoosting'><p>Make predictions from a &quot;RobSBoosting&quot; object</p></a></li>
<li><a href='#PTReg'><p>Robust gene-environment interaction analysis using penalized trimmed regression</p></a></li>
<li><a href='#QPCorr.matrix'><p>Robust identification of gene-environment interactions using a</p>
quantile partial correlation approach</a></li>
<li><a href='#QPCorr.pval'><p>P-values of the &quot;QPCorr.matrix&quot; obtained using a permutation approach</p></a></li>
<li><a href='#Rob_data'><p>A matrix containing the simulated data for <code>RobSBoosting</code> and <code>Miss.boosting</code> methods</p></a></li>
<li><a href='#RobSBoosting'><p>Robust semiparametric gene-environment interaction analysis using sparse boosting</p></a></li>
<li><a href='#simulated_data'><p>Simulated data for generating response</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Robust Gene-Environment Interaction Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xing Qin &lt;qin.xing@163.sufe.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Description: For the risk, progression, and response to treatment of many complex diseases, it has been increasingly recognized that gene-environment interactions play important roles beyond the main genetic and environmental effects. In practical interaction analyses, outliers in response variables and covariates are not uncommon. In addition, missingness in environmental factors is routinely encountered in epidemiological studies. The developed package consists of five robust approaches to address the outliers problems, among which two approaches can also accommodate missingness in environmental factors. Both continuous and right censored responses are considered. The proposed approaches are based on penalization and sparse boosting techniques for identifying important interactions, which are realized using efficient algorithms. Beyond the gene-environment analysis, the developed package can also be adopted to conduct analysis on interactions between other types of low-dimensional and high-dimensional data. (Mengyun Wu et al (2017), &lt;<a href="https://doi.org/10.1080%2F00949655.2018.1523411">doi:10.1080/00949655.2018.1523411</a>&gt;; Mengyun Wu et al (2017), &lt;<a href="https://doi.org/10.1002%2Fgepi.22055">doi:10.1002/gepi.22055</a>&gt;; Yaqing Xu et al (2018), &lt;<a href="https://doi.org/10.1080%2F00949655.2018.1523411">doi:10.1080/00949655.2018.1523411</a>&gt;; Yaqing Xu et al (2019), &lt;<a href="https://doi.org/10.1016%2Fj.ygeno.2018.07.006">doi:10.1016/j.ygeno.2018.07.006</a>&gt;; Mengyun Wu et al (2021), &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtab318">doi:10.1093/bioinformatics/btab318</a>&gt;). </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, splines, pcaPP, Hmisc, survival, quantreg, reshape2,
ggplot2, stats, graphics</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-12 12:37:07 UTC; qx</td>
</tr>
<tr>
<td>Author:</td>
<td>Mengyun Wu [aut],
  Xing Qin [aut, cre],
  Shuangge Ma [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-19 06:50:19 UTC</td>
</tr>
</table>
<hr>
<h2 id='AR'>The covariance matrix with an autoregressive (AR) structure among variables</h2><span id='topic+AR'></span>

<h3>Description</h3>

<p>The covariance matrix with an AR structure among variables, where the marginal variances are 1 and the <code>j</code>th and <code>k</code>th variables have correlation coefficient <code>rho^abs(j-k)</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AR(rho, p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AR_+3A_rho">rho</code></td>
<td>
<p>The correlation coefficient indicating the AR relationship between the variables.</p>
</td></tr>
<tr><td><code id="AR_+3A_p">p</code></td>
<td>
<p>The dimension of variables.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A covariance matrix.
</p>

<hr>
<h2 id='Augmented.data'>Accommodating missingness in environmental measurements in gene-environment interaction analysis</h2><span id='topic+Augmented.data'></span>

<h3>Description</h3>

<p>We consider the scenario with missingness in environmental (E) measurements. Our approach
consists of two steps. We first develop a nonparametric kernel-based data augmentation
approach to accommodate missingness. Then, we adopt a penalization approach <code>BLMCP</code>
for regularized estimation and selection of important interactions and main genetic (G) effects,
where the &quot;main effects-interactions&quot; hierarchical structure is respected.
As E variables are usually preselected and have a low dimension, selection is not conducted on E
variables. With a well-designed weighting scheme, a nice &quot;byproduct&quot; is that the proposed
approach enjoys a certain robustness property.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Augmented.data(G, E, Y, h, family = c("continuous", "survival"), E_type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Augmented.data_+3A_g">G</code></td>
<td>
<p>Input matrix of <code>p</code> genetic measurements consisting of <code>n</code> rows. Each row is an observation vector.</p>
</td></tr>
<tr><td><code id="Augmented.data_+3A_e">E</code></td>
<td>
<p>Input matrix of <code>q</code> environmental risk factors. Each row is an observation vector.</p>
</td></tr>
<tr><td><code id="Augmented.data_+3A_y">Y</code></td>
<td>
<p>Response variable. A quantitative vector for <code>family="continuous"</code>.
For <code>family="survival"</code>, <code>Y</code> should be a two-column matrix with the first column
being the log(survival time) and the second column being the censoring indicator.
The indicator is a binary variable, with &quot;1&quot; indicating dead, and &quot;0&quot; indicating right censored.</p>
</td></tr>
<tr><td><code id="Augmented.data_+3A_h">h</code></td>
<td>
<p>The bandwidths of the kernel functions with the first and second elements corresponding
to the discrete and continuous E factors.</p>
</td></tr>
<tr><td><code id="Augmented.data_+3A_family">family</code></td>
<td>
<p>Response type of <code>Y</code> (see above).</p>
</td></tr>
<tr><td><code id="Augmented.data_+3A_e_type">E_type</code></td>
<td>
<p>A vector indicating the type of each E factor, with &quot;ED&quot; representing discrete E factor, and &quot;EC&quot; representing continuous E factor.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>E_w</code></td>
<td>
<p>The augmented data corresponding to <code>E</code>.</p>
</td></tr>
<tr><td><code>G_w</code></td>
<td>
<p>The augmented data corresponding to <code>G</code>.</p>
</td></tr>
<tr><td><code>y_w</code></td>
<td>
<p>The augmented data corresponding to response <code>y</code>.</p>
</td></tr>
<tr><td><code>weight</code></td>
<td>
<p>The weights of the augmented observation data for accommodating missingness and also
right censoring if <code>family="survival"</code>.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Mengyun Wu, Yangguang Zang, Sanguo Zhang, Jian Huang, and Shuangge Ma.
<em>Accommodating missingness in environmental measurements in gene-environment interaction
analysis. Genetic Epidemiology, 41(6):523-554, 2017.</em><br /> Jin Liu, Jian Huang, Yawei Zhang, Qing
Lan, Nathaniel Rothman, Tongzhang Zheng, and Shuangge Ma.
<em>Identification of gene-environment interactions in cancer studies using penalization.
Genomics, 102(4):189-194, 2013.</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100)
sigmaG=AR(0.3,50)
G=MASS::mvrnorm(100,rep(0,50),sigmaG)
E=matrix(rnorm(100*5),100,5)
E[,2]=E[,2]&gt;0
E[,3]=E[,3]&gt;0
alpha=runif(5,2,3)
beta=matrix(0,5+1,50)
beta[1,1:7]=runif(7,2,3)
beta[2:4,1]=runif(3,2,3)
beta[2:3,2]=runif(2,2,3)
beta[5,3]=runif(1,2,3)

# continuous with Normal error N(0,4)
y1=simulated_data(G=G,E=E,alpha=alpha,beta=beta,error=rnorm(100,0,4),family="continuous")

# survival with Normal error N(0,1)
y2=simulated_data(G,E,alpha,beta,rnorm(100,0,1),family="survival",0.7,0.9)

# generate E measurements with missingness
miss_label1=c(2,6,8,15)
miss_label2=c(4,6,8,16)
E1=E2=E;E1[miss_label1,1]=NA;E2[miss_label2,1]=NA

# continuous
data_new1&lt;-Augmented.data(G,E1,y1,h=c(0.5,1), family="continuous",
E_type=c("EC","ED","ED","EC","EC"))
fit1&lt;-BLMCP(data_new1$G_w, data_new1$E_w, data_new1$y_w, data_new1$weight,
lambda1=0.025,lambda2=0.06,gamma1=3,gamma2=3,max_iter=200)
coef1=coef(fit1)
y1_hat=predict(fit1,E[c(1,2),],G[c(1,2),])
plot(fit1)

## survival
data_new2&lt;-Augmented.data(G,E2,y2, h=c(0.5,1), family="survival",
E_type=c("EC","ED","ED","EC","EC"))
fit2&lt;-BLMCP(data_new2$G_w, data_new2$E_w, data_new2$y_w, data_new2$weight,
lambda1=0.04,lambda2=0.05,gamma1=3,gamma2=3,max_iter=200)
coef2=coef(fit2)
y2_hat=predict(fit2,E[c(1,2),],G[c(1,2),])
plot(fit2)
</code></pre>

<hr>
<h2 id='bic.BLMCP'>BIC for BLMCP</h2><span id='topic+bic.BLMCP'></span>

<h3>Description</h3>

<p>Selects a point along the regularization path of a fitted <code>BLMCP</code> object according to
the BIC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bic.BLMCP(
  G,
  E,
  Y,
  weight = NULL,
  lambda1_set = NULL,
  lambda2_set = NULL,
  nlambda1 = 20,
  nlambda2 = 20,
  gamma1 = 6,
  gamma2 = 6,
  max_iter = 200
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bic.BLMCP_+3A_g">G</code></td>
<td>
<p>Input matrix of <code>p</code> genetic (G) measurements consisting of <code>n</code> rows.
Each row is an observation vector.</p>
</td></tr>
<tr><td><code id="bic.BLMCP_+3A_e">E</code></td>
<td>
<p>Input matrix of <code>q</code> environmental (E) risk factors. Each row is an observation
vector.</p>
</td></tr>
<tr><td><code id="bic.BLMCP_+3A_y">Y</code></td>
<td>
<p>Response variable. A quantitative vector for continuous response. For survival response, <code>Y</code> should be a two-column matrix with the first column being
the log(survival time) and the second column being the censoring indicator. The indicator is a
binary variable, with &quot;1&quot; indicating dead, and &quot;0&quot; indicating right censored.</p>
</td></tr>
<tr><td><code id="bic.BLMCP_+3A_weight">weight</code></td>
<td>
<p>Observation weights.</p>
</td></tr>
<tr><td><code id="bic.BLMCP_+3A_lambda1_set">lambda1_set</code></td>
<td>
<p>A user supplied lambda sequence for group minimax concave penalty (MCP),
where each main G effect and its corresponding interactions are regarded as a group.</p>
</td></tr>
<tr><td><code id="bic.BLMCP_+3A_lambda2_set">lambda2_set</code></td>
<td>
<p>A user supplied lambda sequence for MCP accommodating interaction selection.</p>
</td></tr>
<tr><td><code id="bic.BLMCP_+3A_nlambda1">nlambda1</code></td>
<td>
<p>The number of lambda1 values.</p>
</td></tr>
<tr><td><code id="bic.BLMCP_+3A_nlambda2">nlambda2</code></td>
<td>
<p>The number of lambda2 values.</p>
</td></tr>
<tr><td><code id="bic.BLMCP_+3A_gamma1">gamma1</code></td>
<td>
<p>The regularization parameter of the group MCP penalty.</p>
</td></tr>
<tr><td><code id="bic.BLMCP_+3A_gamma2">gamma2</code></td>
<td>
<p>The regularization parameter of the MCP penalty.</p>
</td></tr>
<tr><td><code id="bic.BLMCP_+3A_max_iter">max_iter</code></td>
<td>
<p>Maximum number of iterations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with S3 class <code>"bic.BLMCP"</code> is returned, which is a list with the ingredients of the BIC fit.
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>The call that produced this object.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The matrix of the coefficients for main E effects, each column corresponds to one
combination of (lambda1,lambda2).</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>The coefficients for main G effects and G-E interactions, each column corresponds to
one combination of (lambda1,lambda2). For each column, the first element is the first G effect and
the second to (<code>q+1</code>) elements are the interactions for the first G factor, and so on.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>The number of nonzeros for each value of (lambda1,lambda2).</p>
</td></tr>
<tr><td><code>BIC</code></td>
<td>
<p>Bayesian Information Criterion for each value of (lambda1,lambda2).</p>
</td></tr>
<tr><td><code>alpha_estimate</code></td>
<td>
<p>Final alpha estimate using Bayesian Information Criterion.</p>
</td></tr>
<tr><td><code>beta_estimate</code></td>
<td>
<p>Final beta estimate using Bayesian Information Criterion.</p>
</td></tr>
<tr><td><code>lambda_combine</code></td>
<td>
<p>The matrix of (lambda1, lambda2), with the first column being the values of
lambda1, the second being the values of lambda2.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Mengyun Wu, Yangguang Zang, Sanguo Zhang, Jian Huang, and Shuangge Ma.
<em>Accommodating missingness in environmental measurements in gene-environment interaction
analysis. Genetic Epidemiology, 41(6):523-554, 2017.</em><br /> Jin Liu, Jian Huang, Yawei Zhang, Qing
Lan, Nathaniel Rothman, Tongzhang Zheng, and Shuangge Ma.
<em>Identification of gene-environment interactions in cancer studies using penalization.
Genomics, 102(4):189-194, 2013.</em>
</p>


<h3>See Also</h3>

<p><code>predict</code>, <code>coef</code> and <code>plot</code> methods,
and the <code>BLMCP</code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100)
sigmaG=AR(0.3,50)
G=MASS::mvrnorm(150,rep(0,50),sigmaG)
E=matrix(rnorm(150*5),150,5)
E[,2]=E[,2]&gt;0;E[,3]=E[,3]&gt;0
alpha=runif(5,2,3)
beta=matrix(0,5+1,50);beta[1,1:8]=runif(8,2,3)
beta[2:4,1]=runif(3,2,3)
beta[2:3,2]=runif(2,2,3)
beta[5,3]=runif(1,2,3)

# continuous with Normal error
y1=simulated_data(G=G,E=E,alpha=alpha,beta=beta,error=rnorm(150),family="continuous")

# survival with Normal error
y2=simulated_data(G,E,alpha,beta,rnorm(150,0,1),family="survival",0.8,1)

# continuous
fit1&lt;-bic.BLMCP(G,E,y1,weight=NULL,lambda1_set=NULL,lambda2_set=NULL,
nlambda1=10,nlambda2=10,gamma1=6,gamma2=6,max_iter=200)
coef1=coef(fit1)
y1_hat=predict(fit1,E,G)
plot(fit1)


## survival
fit2&lt;-bic.BLMCP(G,E,y2,weight=NULL,lambda1_set=NULL,lambda2_set=NULL,
nlambda1=20,nlambda2=20,gamma1=6,gamma2=6,max_iter=200)
coef2=coef(fit2)
y2_hat=predict(fit2,E,G)
plot(fit2)

</code></pre>

<hr>
<h2 id='bic.PTReg'>BIC for PTReg</h2><span id='topic+bic.PTReg'></span>

<h3>Description</h3>

<p>Selects a point along the regularization path of a fitted <code>PTReg</code> object according to
the BIC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bic.PTReg(
  G,
  E,
  Y,
  lambda1_set,
  lambda2_set,
  gamma1,
  gamma2,
  max_init,
  h = NULL,
  tau = 0.4,
  mu = 2.5,
  family = c("continuous", "survival")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bic.PTReg_+3A_g">G</code></td>
<td>
<p>Input matrix of <code>p</code> genetic (G) measurements consisting of <code>n</code> rows.
Each row is an observation vector.</p>
</td></tr>
<tr><td><code id="bic.PTReg_+3A_e">E</code></td>
<td>
<p>Input matrix of <code>q</code> environmental (E) risk factors. Each row is an
observation vector.</p>
</td></tr>
<tr><td><code id="bic.PTReg_+3A_y">Y</code></td>
<td>
<p>Response variable. A quantitative vector for <code>family="continuous"</code>.
For <code>family="survival"</code>, <code>Y</code> should be a two-column matrix with the first column being
the log(survival time) and the second column being the censoring indicator. The indicator is a
binary variable, with &quot;1&quot; indicating dead, and &quot;0&quot; indicating right censored.</p>
</td></tr>
<tr><td><code id="bic.PTReg_+3A_lambda1_set">lambda1_set</code></td>
<td>
<p>A user supplied lambda sequence for minimax concave penalty (MCP)
accommodating main G effect selection.</p>
</td></tr>
<tr><td><code id="bic.PTReg_+3A_lambda2_set">lambda2_set</code></td>
<td>
<p>A user supplied lambda sequence for MCP accommodating interaction selection.</p>
</td></tr>
<tr><td><code id="bic.PTReg_+3A_gamma1">gamma1</code></td>
<td>
<p>The regularization parameter of the MCP penalty corresponding to G effects.</p>
</td></tr>
<tr><td><code id="bic.PTReg_+3A_gamma2">gamma2</code></td>
<td>
<p>The regularization parameter of the MCP penalty corresponding to G-E interactions.</p>
</td></tr>
<tr><td><code id="bic.PTReg_+3A_max_init">max_init</code></td>
<td>
<p>The number of initializations.</p>
</td></tr>
<tr><td><code id="bic.PTReg_+3A_h">h</code></td>
<td>
<p>The number of the trimmed samples if the parameter <code>mu</code> is not given.</p>
</td></tr>
<tr><td><code id="bic.PTReg_+3A_tau">tau</code></td>
<td>
<p>The threshold value used in stability selection.</p>
</td></tr>
<tr><td><code id="bic.PTReg_+3A_mu">mu</code></td>
<td>
<p>The parameter for screening outliers with extreme absolute residuals if the number of
the trimmed samples <code>h</code> is not given.</p>
</td></tr>
<tr><td><code id="bic.PTReg_+3A_family">family</code></td>
<td>
<p>Response type of <code>Y</code> (see above).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with S3 class <code>"bic.PTReg"</code> is returned, which is a list with the ingredients of the BIC fit.
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>The call that produced this object.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The matrix of the coefficients for main E effects, each column corresponds to one
combination of (lambda1,lambda2).</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>The coefficients for main G effects and G-E interactions, each column corresponds to
one combination of (lambda1,lambda2). For each column, the first element is the first G effect and
the second to (<code>q+1</code>) elements are the interactions for the first G factor, and so on.</p>
</td></tr>
<tr><td><code>intercept</code></td>
<td>
<p>Matrix of the intercept estimate, each column corresponds to one combination of
(lambda1,lambda2).</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>The number of nonzeros for each value of (lambda1,lambda2).</p>
</td></tr>
<tr><td><code>BIC</code></td>
<td>
<p>Bayesian Information Criterion for each value of (lambda1,lambda2).</p>
</td></tr>
<tr><td><code>family</code></td>
<td>
<p>The same as input <code>family</code>.</p>
</td></tr>
<tr><td><code>intercept_estimate</code></td>
<td>
<p>Final intercept estimate using Bayesian Information Criterion.</p>
</td></tr>
<tr><td><code>alpha_estimate</code></td>
<td>
<p>Final alpha estimate using Bayesian Information Criterion.</p>
</td></tr>
<tr><td><code>beta_estimate</code></td>
<td>
<p>Final beta estimate using Bayesian Information Criterion.</p>
</td></tr>
<tr><td><code>lambda_combine</code></td>
<td>
<p>Matrix of (lambda1, lambda2), with the first column being the values of
lambda1, the second being the values of lambda2.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Yaqing Xu, Mengyun Wu, Shuangge Ma, and Syed Ejaz Ahmed.
<em>Robust gene-environment interaction analysis using penalized trimmed regression. Journal of
Statistical Computation and Simulation, 88(18):3502-3528, 2018.</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sigmaG&lt;-AR(rho=0.3,p=30)
sigmaE&lt;-AR(rho=0.3,p=3)
set.seed(300)
G=MASS::mvrnorm(150,rep(0,30),sigmaG)
EC=MASS::mvrnorm(150,rep(0,2),sigmaE[1:2,1:2])
ED = matrix(rbinom((150),1,0.6),150,1)
E=cbind(EC,ED)
alpha=runif(3,0.8,1.5)
beta=matrix(0,4,30)
beta[1,1:4]=runif(4,1,1.5)
beta[2,c(1,2)]=runif(2,1,1.5)
lambda1_set=lambda2_set=c(0.2,0.25,0.3,0.35,0.4,0.5)


#continuous response with outliers/contaminations in response variable
y1=simulated_data(G,E,alpha,beta,error=c(rnorm(140),rcauchy(10,0,5)),family="continuous")
fit1&lt;-bic.PTReg(G,E,y1,lambda1_set,lambda2_set,gamma1=6,gamma2=6,
max_init=50,tau=0.6,mu=2.5,family="continuous")
coefficients1=coefficients(fit1)
y_predict=predict(fit1,E,G)
plot(fit1)

# survival with Normal error
y2=simulated_data(G,E,alpha,beta,rnorm(150,0,1),family="survival",0.7,0.9)
fit2&lt;-bic.PTReg(G,E,y2,lambda1_set,lambda2_set,gamma1=6,gamma2=6,
max_init=50,tau=0.6,mu=2.5,family="survival")
coefficients2=coefficients(fit2)
y_predict=predict(fit2,E,G)
plot(fit2)

</code></pre>

<hr>
<h2 id='BLMCP'>Accommodating missingness in environmental measurements in gene-environment interaction
analysis: penalized estimation and selection</h2><span id='topic+BLMCP'></span>

<h3>Description</h3>

<p>The joint gene-environment (G-E) interaction analysis approach developed in Liu et al, 2013.
To accommodate &quot;main effects, interactions&quot; hierarchy, two types of penalty, group minimax
concave penalty (MCP) and MCP are adopted. Specifically, for each G factor, its main effect
and corresponding G-E interactions are regarded as a group, where the group MCP is imposed to
identify whether this G factor has any effect at all. In addition, the MCP is imposed on the
interaction terms to further identify important interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BLMCP(
  G,
  E,
  Y,
  weight = NULL,
  lambda1,
  lambda2,
  gamma1 = 6,
  gamma2 = 6,
  max_iter = 200
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BLMCP_+3A_g">G</code></td>
<td>
<p>Input matrix of <code>p</code> G measurements consisting of <code>n</code> rows. Each row is an
observation vector.</p>
</td></tr>
<tr><td><code id="BLMCP_+3A_e">E</code></td>
<td>
<p>Input matrix of <code>q</code> environmental risk factors. Each row is an observation
vector.</p>
</td></tr>
<tr><td><code id="BLMCP_+3A_y">Y</code></td>
<td>
<p>Response variable. A quantitative vector for continuous response. For survival response, <code>Y</code> should be a two-column matrix with the first column being
the log(survival time) and the second column being the censoring indicator. The indicator is a
binary variable, with &quot;1&quot; indicating dead, and &quot;0&quot; indicating right censored.</p>
</td></tr>
<tr><td><code id="BLMCP_+3A_weight">weight</code></td>
<td>
<p>Observation weights.</p>
</td></tr>
<tr><td><code id="BLMCP_+3A_lambda1">lambda1</code></td>
<td>
<p>A user supplied lambda for group MCP, where each main G effect and its
corresponding interactions are regarded as a group.</p>
</td></tr>
<tr><td><code id="BLMCP_+3A_lambda2">lambda2</code></td>
<td>
<p>A user supplied lambda for MCP accommodating interaction selection.</p>
</td></tr>
<tr><td><code id="BLMCP_+3A_gamma1">gamma1</code></td>
<td>
<p>The regularization parameter of the group MCP penalty.</p>
</td></tr>
<tr><td><code id="BLMCP_+3A_gamma2">gamma2</code></td>
<td>
<p>The regularization parameter of the MCP penalty.</p>
</td></tr>
<tr><td><code id="BLMCP_+3A_max_iter">max_iter</code></td>
<td>
<p>Maximum number of iterations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with S3 class <code>"BLMCP"</code> is returned, which is a list with the following components.
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>The call that produced this object.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The matrix of the coefficients for main E effects.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>The matrix of the regression coefficients for all main G effects (the first row)
and interactions.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>The number of nonzeros.</p>
</td></tr>
<tr><td><code>BIC</code></td>
<td>
<p>Bayesian Information Criterion.</p>
</td></tr>
<tr><td><code>aa</code></td>
<td>
<p>The indicator representing whether the algorithm reaches convergence.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Mengyun Wu, Yangguang Zang, Sanguo Zhang, Jian Huang, and Shuangge Ma.
<em>Accommodating missingness in environmental measurements in gene-environment interaction
analysis. Genetic Epidemiology, 41(6):523-554, 2017.</em><br />  Jin Liu, Jian Huang, Yawei Zhang,
Qing Lan, Nathaniel Rothman, Tongzhang Zheng, and Shuangge Ma.
<em>Identification of gene-environment interactions in cancer studies using penalization.
Genomics, 102(4):189-194, 2013.</em>
</p>


<h3>See Also</h3>

<p><code>predict</code>, and <code>coef</code>, and <code>plot</code>, and <code>bic.BLMCP</code> and
<code>Augmentated.data</code> methods.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100)
sigmaG=AR(0.3,100)
G=MASS::mvrnorm(250,rep(0,100),sigmaG)
E=matrix(rnorm(250*5),250,5)
E[,2]=E[,2]&gt;0;E[,3]=E[,3]&gt;0
alpha=runif(5,2,3)
beta=matrix(0,5+1,100);beta[1,1:8]=runif(8,2,3)
beta[2:4,1]=runif(3,2,3);beta[2:3,2]=runif(2,2,3);beta[5,3]=runif(1,2,3)

# continuous with Normal error
y1=simulated_data(G,E,alpha,beta,error=rnorm(250),family="continuous")
fit1&lt;-BLMCP(G,E,y1,weight=NULL,lambda1=0.05,lambda2=0.06,gamma1=3,gamma2=3,max_iter=200)
coef1=coef(fit1)
y1_hat=predict(fit1,E,G)
plot(fit1)

# survival with Normal error
y2=simulated_data(G,E,alpha,beta,rnorm(250,0,1),family="survival",0.7,0.9)
fit2&lt;-BLMCP(G,E,y2,weight=NULL,lambda1=0.05,lambda2=0.06,gamma1=3,gamma2=3,max_iter=200)
coef2=coef(fit2)
y2_hat=predict(fit2,E,G)
plot(fit2)
</code></pre>

<hr>
<h2 id='coef.bic.BLMCP'>Extract coefficients from a &quot;bic.BLMCP&quot; object</h2><span id='topic+coef.bic.BLMCP'></span>

<h3>Description</h3>

<p>This function extracts the coefficients of main effects and interactions
from a BIC BLMCP model, using the stored <code>"bic.BLMCP"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bic.BLMCP'
coef(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.bic.BLMCP_+3A_object">object</code></td>
<td>
<p>Fitted <code>"bic.BLMCP"</code> model object.</p>
</td></tr>
<tr><td><code id="coef.bic.BLMCP_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to get coefficients.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the ... argument which is passed on to the <code>coef</code>
method for <code>bic.BLMCP</code> objects.
</p>
<table>
<tr><td><code>alpha</code></td>
<td>
<p>The matrix of the coefficients for main environmental effects.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>The matrix of the regression coefficients for all main genetic effects (the first row)
and interactions.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Mengyun Wu, Yangguang Zang, Sanguo Zhang, Jian Huang, and Shuangge Ma.
<em>Accommodating missingness in environmental measurements in gene-environment interaction
analysis. Genetic Epidemiology, 41(6):523-554, 2017.</em><br />  Jin Liu, Jian Huang, Yawei Zhang, Qing
Lan, Nathaniel Rothman, Tongzhang Zheng, and Shuangge Ma.
<em>Identification of gene-environment interactions in cancer studies using penalization.
Genomics, 102(4):189-194, 2013.</em>
</p>


<h3>See Also</h3>

<p><code>bic.BLMCP</code>, and <code>predict</code>, and <code>plot</code> methods, and the
<code>BLMCP</code> function.
</p>

<hr>
<h2 id='coef.bic.PTReg'>Extract coefficients from a &quot;bic.PTReg&quot; object</h2><span id='topic+coef.bic.PTReg'></span>

<h3>Description</h3>

<p>This function extracts the coefficients of main effects and interactions from a BIC PTReg model,
using the stored <code>"bic.PTReg"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bic.PTReg'
coef(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.bic.PTReg_+3A_object">object</code></td>
<td>
<p>Fitted &quot;bic.PTReg&quot; model object.</p>
</td></tr>
<tr><td><code id="coef.bic.PTReg_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to get coefficients.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the ... argument which is passed on to the <code>coef</code>
method for <code>bic.PTReg</code> objects.
</p>
<table>
<tr><td><code>intercept</code></td>
<td>
<p>The intercept estimate.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The matrix of the coefficients for main environmental effects.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>The matrix of the regression coefficients for all main genetic effects (the first row) and interactions.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Yaqing Xu, Mengyun Wu, Shuangge Ma, and Syed Ejaz Ahmed.
<em>Robust gene-environment interaction analysis using penalized trimmed regression. Journal of
Statistical Computation and Simulation, 88(18):3502-3528, 2018.</em>
</p>


<h3>See Also</h3>

<p><code>bic.PTReg</code>, and <code>predict</code>, and <code>plot</code> methods, and
<code>PTReg</code>.
</p>

<hr>
<h2 id='coef.BLMCP'>Extract coefficients from a &quot;BLMCP&quot; object</h2><span id='topic+coef.BLMCP'></span>

<h3>Description</h3>

<p>This function extracts the coefficients of main effects and interactions from a BLMCP model,
using the stored <code>"BLMCP"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BLMCP'
coef(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.BLMCP_+3A_object">object</code></td>
<td>
<p>Fitted <code>"BLMCP"</code> model object.</p>
</td></tr>
<tr><td><code id="coef.BLMCP_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to get coefficients.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the ... argument which is passed on to the <code>coef</code>
method for <code>BLMCP</code> objects.
</p>
<table>
<tr><td><code>alpha</code></td>
<td>
<p>The matrix of the coefficients for main environmental effects.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>The matrix of the regression coefficients for all main genetic effects (the first row)
and interactions.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Mengyun Wu, Yangguang Zang, Sanguo Zhang, Jian Huang, and Shuangge Ma.
<em>Accommodating missingness in environmental measurements in gene-environment interaction analysis. Genetic Epidemiology, 41(6):523-554, 2017.</em><br /> Jin Liu, Jian Huang, Yawei Zhang, Qing Lan, Nathaniel Rothman, Tongzhang Zheng, and Shuangge Ma.
<em>Identification of gene-environment interactions in cancer studies using penalization.
Genomics, 102(4):189-194, 2013.</em>
</p>


<h3>See Also</h3>

<p><code>BLMCP</code>, and <code>predict</code>, <code>plot</code> methods, and
<code>bic.BLMCP</code>.
</p>

<hr>
<h2 id='coef.PTReg'>Extract coefficients from a &quot;PTReg&quot; object</h2><span id='topic+coef.PTReg'></span>

<h3>Description</h3>

<p>This function extracts main effect and interaction coefficients from a PTReg model, using the
stored <code>"PTReg"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PTReg'
coef(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.PTReg_+3A_object">object</code></td>
<td>
<p>Fitted <code>"PTReg"</code> model object.</p>
</td></tr>
<tr><td><code id="coef.PTReg_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to get coefficients.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the ... argument which is passed on to the <code>coef</code>
method for <code>PTReg</code> objects.
</p>
<table>
<tr><td><code>intercept</code></td>
<td>
<p>The intercept estimate.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The matrix of the coefficients for main environmental effects.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>The matrix of the regression coefficients for all main genetic effects (the first row) and interactions.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Yaqing Xu, Mengyun Wu, Shuangge Ma, and Syed Ejaz Ahmed.
<em>Robust gene-environment interaction analysis using penalized trimmed regression. Journal of
Statistical Computation and Simulation, 88(18):3502-3528, 2018.</em>
</p>


<h3>See Also</h3>

<p><code>PTReg</code>, and <code>predict</code> methods, and
<code>bic.PTReg</code>.
</p>

<hr>
<h2 id='coef.RobSBoosting'>Extract coefficients from a &quot;RobSBoosting&quot; object</h2><span id='topic+coef.RobSBoosting'></span>

<h3>Description</h3>

<p>This function extracts coefficients from a RobSBoosting model, using the stored
<code>"RobSBoosting"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RobSBoosting'
coef(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.RobSBoosting_+3A_object">object</code></td>
<td>
<p>Fitted <code>"RobSBoosting"</code> model object.</p>
</td></tr>
<tr><td><code id="coef.RobSBoosting_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to get coefficients.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>intercept</code></td>
<td>
<p>The intercept estimate.</p>
</td></tr>
<tr><td><code>unique_variable</code></td>
<td>
<p>A matrix with two columns that represents the variables that are selected
for the model after removing the duplicates, since the <code>loop_time</code> iterations of the method
may produce variables that are repeatedly selected into the model. Here, the first and second
columns correspond to the indexes of environmental (E) factors and genetic (G) factors. For
example, (1, 0) represents that this variable is the first E factor, and (1,2) represents that the
variable is the interaction between the first E factor and second G factor.</p>
</td></tr>
<tr><td><code>unique_coef</code></td>
<td>
<p>Coefficients corresponding to <code>unique_variable</code>. Here, the coefficients
are simple regression coefficients for the linear effect (discrete E factor, G factor, and their
interaction), and B spline coefficients for the nonlinear effect (continuous E factor, and
corresponding G-E interaction).</p>
</td></tr>
<tr><td><code>unique_knots</code></td>
<td>
<p>A list of knots corresponding to <code>unique_variable</code>. Here, when the type
of <code>unique_variable</code> is discrete E factor, G factor, or their interaction, knot will be NULL,
and knots will be B spline otherwise.</p>
</td></tr>
<tr><td><code>unique_Boundary.knots</code></td>
<td>
<p>A list of boundary knots corresponding to <code>unique_variable</code>.</p>
</td></tr>
<tr><td><code>unique_vtype</code></td>
<td>
<p>A vector representing the variable type of <code>unique_variable</code>. Here, &quot;EC&quot;
stands for continuous E effect, &quot;ED&quot; for discrete E effect, &quot;G&quot; for G effect, &quot;EC-G&quot; for the
interaction between &quot;EC&quot; and &quot;G&quot;, and &quot;ED-G&quot; for the interaction between &quot;ED&quot; and &quot;G&quot;.</p>
</td></tr>
<tr><td><code>estimation_results</code></td>
<td>
<p>A list of estimation results for each variable. Here, the first
<code>q</code> elemnets are for the E effects, the (<code>q+1</code>) element
is for the first G effect and the (<code>q+2</code>) to (<code>2q+1</code>) elements are for the interactions
corresponding to the first G factor, and so on.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Mengyun Wu and Shuangge Ma.
<em>Robust semiparametric gene-environment interaction analysis using sparse boosting.
Statistics in Medicine, 38(23):4625-4641, 2019.</em>
</p>


<h3>See Also</h3>

<p><code>RobSBoosting</code>, and <code>predict</code>, and <code>plot</code> methods.
</p>

<hr>
<h2 id='HNSCC'>A data frame containing the TCGA head and neck squamous cell carcinoma (HNSCC) data.</h2><span id='topic+HNSCC'></span>

<h3>Description</h3>

<p>A data frame containing the 7 environmental (E)
effects (the first 7 columns), 2000 genetic (G) effects (column 8 to column 2007), logarithm of survival time
(column 2008), and censoring indicator (column 2009). All of them can be downloaded  from TCGA Provisional using the
R package <code>cgdsr</code>. See details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HNSCC)
</code></pre>


<h3>Format</h3>

<p>A data frame with 484 rows and 2009 variables.
</p>


<h3>Details</h3>

<p>There are seven E effects, namely alcohol consumption frequency (ACF), smoking pack
years (SPY), age, gender, PN, PT, and ICD O3 site. For G effects, 2,000 gene
expressions are considered. Among 484 subjects, 343 subjects have missingness in ACF and/or SPY.
For G effects, we analyze mRNA gene expressions. A total of 18,409 gene expression
measurements are available, then prescreening is conducted using marginal Cox models, finally,
the top 2,000 genes with the smallest p-values are selected for downstream analysis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(HNSCC)
E=as.matrix(HNSCC[,1:7])
G=as.matrix(HNSCC[,8:2007])
Y=as.matrix(HNSCC[,2008:2009])

fit&lt;-Miss.boosting(G,E,Y,im_time=10,loop_time=1000,v=0.25,num.knots=5,degree=3,tau=0.3,
family="survival",E_type=c(rep("EC",3),rep("ED",4)))
plot(fit)

</code></pre>

<hr>
<h2 id='Miss.boosting'>Robust gene-environment interaction analysis approach via sparse boosting, where the
missingness in environmental measurements is effectively accommodated using multiple
imputation approach</h2><span id='topic+Miss.boosting'></span>

<h3>Description</h3>

<p>This gene-environment analysis approach includes three steps to accommodate both missingness
in environmental (E) measurements and long-tailed or contaminated outcomes. At the first step,
the  multiple imputation approach based on sparse boosting method is developed to accommodate
missingness in E measurements, where we use <code>NA</code> to represent those E measurments which
are missing. Here a semiparametric model is assumed to accommodate nonlinear effects, where we
model continuous E factors in a nonlinear way, and discrete E factors in a linear way. For
estimating the nonlinear functions, the B spline expansion is adopted. At the second step, for
each imputed data, we develop <code>RobSBoosting</code> approach for identifying important main E
and genetic (G) effects, and G-E interactions, where the Huber loss function and Qn estimator are
adopted to accommodate long-tailed distribution/data contamination (see <code>RobSBoosting</code>).
At the third step, the identification results from Step 2 are combined based on stability
selection technique.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Miss.boosting(
  G,
  E,
  Y,
  im_time = 10,
  loop_time = 500,
  num.knots = c(2),
  Boundary.knots,
  degree = c(2),
  v = 0.1,
  tau,
  family = c("continuous", "survival"),
  knots = NULL,
  E_type
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Miss.boosting_+3A_g">G</code></td>
<td>
<p>Input matrix of <code>p</code> genetic measurements consisting of <code>n</code> rows. Each row
is an observation vector.</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_e">E</code></td>
<td>
<p>Input matrix of <code>q</code> environmental risk factors. Each row is an observation
vector.</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_y">Y</code></td>
<td>
<p>Response variable. A quantitative vector for <code>family</code>=&quot;continuous&quot;. For
<code>family</code>=&quot;survival&quot;, <code>Y</code> should be a two-column matrix with the first column being
the log(survival time) and the second column being the censoring indicator. The indicator is a
binary variable, with &quot;1&quot; indicating dead, and &quot;0&quot; indicating right censored.</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_im_time">im_time</code></td>
<td>
<p>Number of imputation for accommodating missingness
in E variables.</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_loop_time">loop_time</code></td>
<td>
<p>Number of iterations of the sparse boosting.</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_num.knots">num.knots</code></td>
<td>
<p>Numbers of knots for the B spline basis.</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_boundary.knots">Boundary.knots</code></td>
<td>
<p>The boundary of knots for the B spline basis.</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_degree">degree</code></td>
<td>
<p>Degree for the B spline basis.</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_v">v</code></td>
<td>
<p>The step size used in the sparse boosting process. Default is 0.1.</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_tau">tau</code></td>
<td>
<p>Threshold used in the stability selection at the third step.</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_family">family</code></td>
<td>
<p>Response type of <code>Y</code> (see above).</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_knots">knots</code></td>
<td>
<p>List of knots for the B spline basis. Default is NULL and knots can be generated
with the given <code>num.knots, degree</code> and <code>Boundary.knots</code>.</p>
</td></tr>
<tr><td><code id="Miss.boosting_+3A_e_type">E_type</code></td>
<td>
<p>A vector indicating the type of each E factor, with &quot;ED&quot; representing discrete E factor, and &quot;EC&quot; representing continuous E factor.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with S3 class <code>"Miss.boosting"</code> is returned, which is a list with the following components
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>The call that produced this object.</p>
</td></tr>
<tr><td><code>alpha0</code></td>
<td>
<p>A vector with each element indicating whether the corresponding E factor is
selected.</p>
</td></tr>
<tr><td><code>beta0</code></td>
<td>
<p>A vector with each element indicating whether the corresponding G factor or G-E
interaction is selected. The first element is the first G effect and the second to
(<code>q+1</code>) elements are the interactions for the first G factor, and so on.</p>
</td></tr>
<tr><td><code>intercept</code></td>
<td>
<p>The intercept estimate.</p>
</td></tr>
<tr><td><code>unique_variable</code></td>
<td>
<p>A matrix with two columns that represents the variables that are
selected for the model after removing the duplicates, since the <code>loop_time</code> iterations of
the method may produce variables that are repeatedly selected into the model. Here, the first
and second columns correspond to the indexes of E factors and G factors. For example, (1, 0)
represents that this variable is the first E factor, and (1,2) represents that the variable is
the interaction between the first E factor and second G factor.</p>
</td></tr>
<tr><td><code>unique_coef</code></td>
<td>
<p>Coefficients corresponding to <code>unique_variable</code>. Here, the coefficients are simple regression coefficients for the linear effect (discrete E factor, G factor, and
their interaction), and B spline coefficients for the nonlinear effect (continuous E factor,
and corresponding G-E interaction).</p>
</td></tr>
<tr><td><code>unique_knots</code></td>
<td>
<p>A list of knots corresponding to <code>unique_variable</code>. Here, when the
type of <code>unique_variable</code> is discrete E factor, G factor or their interaction, knot will
be NULL, and knots will be B spline otherwise.</p>
</td></tr>
<tr><td><code>unique_Boundary.knots</code></td>
<td>
<p>A list of boundary knots corresponding to
<code>unique_variable</code>.</p>
</td></tr>
<tr><td><code>unique_vtype</code></td>
<td>
<p>A vector representing the variable type of <code>unique_variable</code>.
Here, &quot;EC&quot; stands for continuous E effect, &quot;ED&quot; for discrete E effect,
&quot;G&quot;  for genetic factor variable, &quot;EC-G&quot; for the interaction between &quot;EC&quot; and &quot;G&quot;,
and &quot;ED-G&quot; for the interaction between &quot;ED&quot; and &quot;G&quot;.</p>
</td></tr>
<tr><td><code>degree</code></td>
<td>
<p>Degree for the B spline basis.</p>
</td></tr>
<tr><td><code>NorM</code></td>
<td>
<p>The values of B spline basis.</p>
</td></tr>
<tr><td><code>E_type</code></td>
<td>
<p>The type of E effects.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Mengyun Wu and Shuangge Ma.
<em>Robust semiparametric gene-environment interaction analysis using sparse boosting.
Statistics in Medicine, 38(23):4625-4641, 2019.</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Rob_data)
G=Rob_data[,1:20];E=Rob_data[,21:24]
Y=Rob_data[,25];Y_s=Rob_data[,26:27]
knots=list();Boundary.knots=matrix(0,(20+4),2)
for (i in 1:4){
  knots[[i]]=c(0,1)
  Boundary.knots[i,]=c(0,1)
}
E2=E1=E

##continuous
E1[7,1]=NA
fit1&lt;-Miss.boosting(G,E1,Y,im_time=1,loop_time=100,num.knots=c(2),Boundary.knots,
degree=c(2),v=0.1,tau=0.3,family="continuous",knots=knots,E_type=c("EC","EC","ED","ED"))
y1_hat=predict(fit1,matrix(E1[1,],nrow=1),matrix(G[1,],nrow=1))
plot(fit1)


##survival
E2[4,1]=NA
fit2&lt;-Miss.boosting(G,E2,Y_s,im_time=2,loop_time=200,num.knots=c(2),Boundary.knots,
degree=c(2),v=0.1,tau=0.3,family="survival",knots,E_type=c("EC","EC","ED","ED"))
y2_hat=predict(fit2,matrix(E1[1,],nrow=1),matrix(G[1,],nrow=1))
plot(fit2)

</code></pre>

<hr>
<h2 id='plot.bic.BLMCP'>Plot coefficients from a &quot;bic.BLMCP&quot; object</h2><span id='topic+plot.bic.BLMCP'></span>

<h3>Description</h3>

<p>Draw a heatmap for estimated coefficients in a fitted
<code>"bic.BLMCP"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bic.BLMCP'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.bic.BLMCP_+3A_x">x</code></td>
<td>
<p>Fitted <code>"bic.BLMCP"</code> model.</p>
</td></tr>
<tr><td><code id="plot.bic.BLMCP_+3A_...">...</code></td>
<td>
<p>Other graphical parameters to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap for estimated coefficients.
</p>


<h3>References</h3>

<p>Mengyun Wu, Yangguang Zang, Sanguo Zhang, Jian Huang, and Shuangge Ma.
<em>Accommodating missingness in environmental measurements in gene-environment interaction
analysis. Genetic Epidemiology, 41(6):523-554, 2017.</em><br />  Jin Liu, Jian Huang, Yawei Zhang,
Qing Lan, Nathaniel Rothman, Tongzhang Zheng, and Shuangge Ma.
<em>Identification of gene-environment interactions in cancer studies using penalization.
Genomics, 102(4):189-194, 2013.</em>
</p>


<h3>See Also</h3>

<p><code>predict</code>, <code>coef</code> and <code>BLMCP</code> methods.
</p>

<hr>
<h2 id='plot.bic.PTReg'>Plot coefficients from a &quot;bic.PTReg&quot; object</h2><span id='topic+plot.bic.PTReg'></span>

<h3>Description</h3>

<p>Draw a heatmap for estimated coefficients in a fitted
<code>"bic.PTReg"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bic.PTReg'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.bic.PTReg_+3A_x">x</code></td>
<td>
<p>Fitted <code>"bic.PTReg"</code> model.</p>
</td></tr>
<tr><td><code id="plot.bic.PTReg_+3A_...">...</code></td>
<td>
<p>Other graphical parameters to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap for estimated coefficients.
</p>


<h3>References</h3>

<p>Yaqing Xu, Mengyun Wu, Shuangge Ma, and Syed Ejaz Ahmed.
<em>Robust gene-environment interaction analysis using penalized trimmed regression. Journal
of Statistical Computation and Simulation, 88(18):3502-3528, 2018.</em>
</p>


<h3>See Also</h3>

<p><code>bic.PTReg</code>, and <code>predict</code>, and <code>coef</code>
methods.
</p>

<hr>
<h2 id='plot.BLMCP'>Plot coefficients from a &quot;BLMCP&quot; object</h2><span id='topic+plot.BLMCP'></span>

<h3>Description</h3>

<p>Draw a heatmap for estimated coefficients in a fitted
<code>"BLMCP"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BLMCP'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.BLMCP_+3A_x">x</code></td>
<td>
<p>Fitted <code>"BLMCP"</code> model.</p>
</td></tr>
<tr><td><code id="plot.BLMCP_+3A_...">...</code></td>
<td>
<p>Other graphical parameters to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap for estimated coefficients.
</p>


<h3>References</h3>

<p>Mengyun Wu, Yangguang Zang, Sanguo Zhang, Jian Huang, and Shuangge Ma.
<em>Accommodating missingness in environmental measurements in gene-environment interaction
analysis. Genetic Epidemiology, 41(6):523-554, 2017.</em>
</p>
<p>Jin Liu, Jian Huang, Yawei Zhang, Qing Lan, Nathaniel Rothman, Tongzhang Zheng, and
Shuangge Ma.
<em>Identification of gene-environment interactions in cancer studies using penalization. Genomics, 102(4):189-194, 2013.</em>
</p>


<h3>See Also</h3>

<p><code>BLMCP</code>, and  <code>predict</code> and <code>coef</code>
methods.
</p>

<hr>
<h2 id='plot.Miss.boosting'>Plot coefficients from a &quot;Miss.boosting&quot; object</h2><span id='topic+plot.Miss.boosting'></span>

<h3>Description</h3>

<p>Draw plots for estimated parameters in a fitted
<code>"Miss.boosting"</code> object, including a heatmap for discrete
environmental (E) effects, and selected genetic (G) effects and G-E interactions, and plots for each
of selected continuous E (EC) effect and interactions between EC and G.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Miss.boosting'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Miss.boosting_+3A_x">x</code></td>
<td>
<p>Fitted <code>"Miss.boosting"</code> model.</p>
</td></tr>
<tr><td><code id="plot.Miss.boosting_+3A_...">...</code></td>
<td>
<p>Other graphical parameters to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap for estimated coefficients.
</p>


<h3>References</h3>

<p>Mengyun Wu and Shuangge Ma.
<em>Robust semiparametric gene-environment interaction analysis using sparse boosting.
Statistics in Medicine, 38(23):4625-4641, 2019.</em>
</p>


<h3>See Also</h3>

<p><code>Miss.boosting</code>, and <code>predict</code> methods.
</p>

<hr>
<h2 id='plot.PTReg'>Plot coefficients from a &quot;PTReg&quot; object</h2><span id='topic+plot.PTReg'></span>

<h3>Description</h3>

<p>Draw a heatmap for estimated coefficients in a fitted
<code>"PTReg"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PTReg'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.PTReg_+3A_x">x</code></td>
<td>
<p>Fitted <code>"PTReg"</code> model.</p>
</td></tr>
<tr><td><code id="plot.PTReg_+3A_...">...</code></td>
<td>
<p>Other graphical parameters to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap for estimated coefficients.
</p>


<h3>References</h3>

<p>Yaqing Xu, Mengyun Wu, Shuangge Ma, and Syed Ejaz Ahmed.
<em>Robust gene-environment interaction analysis using penalized trimmed regression. Journal
of Statistical Computation and Simulation, 88(18):3502-3528, 2018.</em>
</p>


<h3>See Also</h3>

<p><code>PTReg</code>, and <code>predict</code>, and <code>coef</code>
methods.
</p>

<hr>
<h2 id='plot.RobSBoosting'>Plot coefficients from a &quot;RobSBoosting&quot; object</h2><span id='topic+plot.RobSBoosting'></span>

<h3>Description</h3>

<p>Draw plots for estimated parameters in a fitted
<code>"RobSBoosting"</code> object, including a heatmap for discrete
environmental (E) effects, and selected genetic (G) effects and G-E interactions, and plots for
each of selected continuous E (EC) effect and interactions between EC and G.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RobSBoosting'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.RobSBoosting_+3A_x">x</code></td>
<td>
<p>Fitted <code>"RobSBoosting"</code> model.</p>
</td></tr>
<tr><td><code id="plot.RobSBoosting_+3A_...">...</code></td>
<td>
<p>Other graphical parameters to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots for estimated coefficients.
</p>


<h3>References</h3>

<p>Mengyun Wu and Shuangge Ma.
<em>Robust semiparametric gene-environment interaction analysis using sparse boosting.
Statistics in Medicine, 38(23):4625-4641, 2019.</em>
</p>


<h3>See Also</h3>

<p><code>RobSBoosting</code>, <code>predict</code> and <code>coef</code>
methods.
</p>

<hr>
<h2 id='predict.bic.BLMCP'>Make predictions from a &quot;bic.BLMCP&quot; object.</h2><span id='topic+predict.bic.BLMCP'></span>

<h3>Description</h3>

<p>This function makes predictions from a BIC BLMCP model, using the stored
<code>"bic.BLMCP"</code> object.  This function makes it easier to use the results of BIC to
make a prediction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bic.BLMCP'
predict(object, newE, newG, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.bic.BLMCP_+3A_object">object</code></td>
<td>
<p>Fitted <code>"bic.BLMCP"</code> object.</p>
</td></tr>
<tr><td><code id="predict.bic.BLMCP_+3A_newe">newE</code></td>
<td>
<p>Matrix of new values for <code>E</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.bic.BLMCP_+3A_newg">newG</code></td>
<td>
<p>Matrix of new values for <code>G</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.bic.BLMCP_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to predict.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the ... argument which is passed
on to the <code>predict</code> method for <code>BLMCP</code> objects.
</p>


<h3>References</h3>

<p>Mengyun Wu, Yangguang Zang, Sanguo Zhang, Jian Huang, and Shuangge Ma.
<em>Accommodating missingness in environmental measurements in gene-environment interaction
analysis. Genetic Epidemiology, 41(6):523-554, 2017.</em>
</p>
<p>Jin Liu, Jian Huang, Yawei Zhang, Qing Lan, Nathaniel Rothman, Tongzhang Zheng, and
Shuangge Ma.
<em>Identification of gene-environment interactions in cancer studies using penalization. Genomics, 102(4):189-194, 2013.</em><br />
<a href="http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC3869641&amp;blobtype=pdf">http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC3869641&amp;blobtype=pdf</a>
</p>


<h3>See Also</h3>

<p><code>coef</code>, and <code>plot</code> and
<code>bic.BLMCP</code> methods, and <code>BLMCP</code>.
</p>

<hr>
<h2 id='predict.bic.PTReg'>Make predictions from a &quot;bic.PTReg&quot; object</h2><span id='topic+predict.bic.PTReg'></span>

<h3>Description</h3>

<p>This function makes predictions from a BIC PTReg model, using the stored <code>"bic.PTReg"</code>
object.  This function makes it easier to use the results of BIC to make
a prediction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bic.PTReg'
predict(object, newE, newG, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.bic.PTReg_+3A_object">object</code></td>
<td>
<p>Fitted <code>"bic.PTReg"</code> object.</p>
</td></tr>
<tr><td><code id="predict.bic.PTReg_+3A_newe">newE</code></td>
<td>
<p>Matrix of new values for <code>E</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.bic.PTReg_+3A_newg">newG</code></td>
<td>
<p>Matrix of new values for <code>G</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.bic.PTReg_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to predict.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the ... argument which is passed
on to the <code>predict</code> method for <code>PTReg</code> objects.
</p>


<h3>References</h3>

<p>Yaqing Xu, Mengyun Wu, Shuangge Ma, and Syed Ejaz Ahmed.
<em>Robust gene-environment interaction analysis using penalized trimmed regression. Journal
of Statistical Computation and Simulation, 88(18):3502-3528, 2018.</em>
</p>


<h3>See Also</h3>

<p><code>bic.PTReg</code>, and <code>coef</code>, and <code>plot</code> methods, and <code>PTReg</code>.
</p>

<hr>
<h2 id='predict.BLMCP'>Make predictions from a &quot;BLMCP&quot; object</h2><span id='topic+predict.BLMCP'></span>

<h3>Description</h3>

<p>This function makes predictions from a  BLMCP
model, using the stored <code>"BLMCP"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BLMCP'
predict(object, newE, newG, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.BLMCP_+3A_object">object</code></td>
<td>
<p>Fitted <code>"BLMCP"</code> object.</p>
</td></tr>
<tr><td><code id="predict.BLMCP_+3A_newe">newE</code></td>
<td>
<p>Matrix of new values for <code>E</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.BLMCP_+3A_newg">newG</code></td>
<td>
<p>Matrix of new values for <code>G</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.BLMCP_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to predict.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the ... argument which is passed
on to the <code>predict</code> method for <code>BLMCP</code> objects.
</p>


<h3>References</h3>

<p>Mengyun Wu, Yangguang Zang, Sanguo Zhang, Jian Huang, and Shuangge Ma.
<em>Accommodating missingness in environmental measurements in gene-environment interaction
analysis. Genetic Epidemiology, 41(6):523-554, 2017.</em><br />  Jin Liu, Jian Huang, Yawei Zhang,
Qing Lan, Nathaniel Rothman, Tongzhang Zheng, and Shuangge Ma.
<em>Identification of gene-environment interactions in cancer studies using penalization. Genomics, 102(4):189-194, 2013.</em>
</p>


<h3>See Also</h3>

<p><code>BLMCP</code>, <code>coef</code>, and <code>plot</code> methods, and <code>bic.BLMCP</code> method.
</p>

<hr>
<h2 id='predict.Miss.boosting'>Make predictions from a &quot;Miss.boosting&quot; object</h2><span id='topic+predict.Miss.boosting'></span>

<h3>Description</h3>

<p>This function makes predictions from a  Miss.boosting model, using the stored
<code>"Miss.boosting"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Miss.boosting'
predict(object, newE, newG, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.Miss.boosting_+3A_object">object</code></td>
<td>
<p>Fitted <code>"Miss.boosting"</code> object.</p>
</td></tr>
<tr><td><code id="predict.Miss.boosting_+3A_newe">newE</code></td>
<td>
<p>Matrix of new values for <code>E</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.Miss.boosting_+3A_newg">newG</code></td>
<td>
<p>Matrix of new values for <code>G</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.Miss.boosting_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to predict.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the ... argument which is passed
on to the <code>predict</code> method for <code>Miss.boosting</code> objects.
</p>


<h3>References</h3>

<p>Mengyun Wu and Shuangge Ma.
<em>Robust semiparametric gene-environment interaction analysis using sparse boosting.
Statistics in Medicine, 38(23):4625-4641, 2019.</em>
</p>


<h3>See Also</h3>

<p><code>Miss.boosting</code>, and <code>plot</code> methods.
</p>

<hr>
<h2 id='predict.PTReg'>Make predictions from a &quot;PTReg&quot; object</h2><span id='topic+predict.PTReg'></span>

<h3>Description</h3>

<p>This function makes predictions from a PTReg model, using the stored <code>"PTReg"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PTReg'
predict(object, newE, newG, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.PTReg_+3A_object">object</code></td>
<td>
<p>Fitted <code>"PTReg"</code> object.</p>
</td></tr>
<tr><td><code id="predict.PTReg_+3A_newe">newE</code></td>
<td>
<p>Matrix of new values for <code>E</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.PTReg_+3A_newg">newG</code></td>
<td>
<p>Matrix of new values for <code>G</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.PTReg_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to predict.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the ... argument which is passed
on to the <code>predict</code> method for <code>PTReg</code> objects.
</p>


<h3>References</h3>

<p>Yaqing Xu, Mengyun Wu, Shuangge Ma, and Syed Ejaz Ahmed.
<em>Robust gene-environment interaction analysis using penalized trimmed regression. Journal
of Statistical Computation and Simulation, 88(18):3502-3528, 2018.</em>
</p>


<h3>See Also</h3>

<p><code>PTReg</code>, <code>coef</code> and <code>plot</code> methods, and <code>bic.PTReg</code>.
</p>

<hr>
<h2 id='predict.RobSBoosting'>Make predictions from a &quot;RobSBoosting&quot; object</h2><span id='topic+predict.RobSBoosting'></span>

<h3>Description</h3>

<p>This function makes predictions from a  RobSBoosting
model, using the stored <code>"RobSBoosting"</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RobSBoosting'
predict(object, newE, newG, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.RobSBoosting_+3A_object">object</code></td>
<td>
<p>Fitted <code>"RobSBoosting"</code> object.</p>
</td></tr>
<tr><td><code id="predict.RobSBoosting_+3A_newe">newE</code></td>
<td>
<p>Matrix of new values for <code>E</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.RobSBoosting_+3A_newg">newG</code></td>
<td>
<p>Matrix of new values for <code>G</code> at which predictions are to be
made.</p>
</td></tr>
<tr><td><code id="predict.RobSBoosting_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to predict.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the ... argument which is passed
on to the <code>predict</code> method for <code>RobSBoosting</code> objects.
</p>


<h3>References</h3>

<p>Mengyun Wu and Shuangge Ma.
<em>Robust semiparametric gene-environment interaction analysis using sparse boosting. Statistics in Medicine, 38(23):4625-4641, 2019.</em>
</p>


<h3>See Also</h3>

<p><code>RobSBoosting</code>, <code>coef</code>, and <code>plot</code> methods.
</p>

<hr>
<h2 id='PTReg'>Robust gene-environment interaction analysis using penalized trimmed regression</h2><span id='topic+PTReg'></span>

<h3>Description</h3>

<p>Gene-environment interaction analysis using penalized trimmed regression, which is robust to
outliers in both predictor and response spaces. The objective function is based on trimming
technique, where the samples with extreme absolute residuals are trimmed. A decomposition
framework is adopted for accommodating &quot;main effects-interactions&quot; hierarchy, and minimax
concave penalty (MCP) is adopted for regularized estimation and interaction (and main genetic
effect) selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PTReg(
  G,
  E,
  Y,
  lambda1,
  lambda2,
  gamma1 = 6,
  gamma2 = 6,
  max_init,
  h = NULL,
  tau = 0.4,
  mu = 2.5,
  family = c("continuous", "survival")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PTReg_+3A_g">G</code></td>
<td>
<p>Input matrix of <code>p</code> genetic (G) measurements consisting of <code>n</code> rows. Each
row is an observation vector.</p>
</td></tr>
<tr><td><code id="PTReg_+3A_e">E</code></td>
<td>
<p>Input matrix of <code>q</code> environmental (E) risk factors. Each row is an observation
vector.</p>
</td></tr>
<tr><td><code id="PTReg_+3A_y">Y</code></td>
<td>
<p>Response variable. A quantitative vector for <code>family="continuous"</code>. For
<code>family="survival"</code>, <code>Y</code> should be a two-column matrix with the first column being
the log(survival time) and the second column being the censoring indicator. The indicator is a
binary variable, with &quot;1&quot; indicating dead, and &quot;0&quot; indicating right censored.</p>
</td></tr>
<tr><td><code id="PTReg_+3A_lambda1">lambda1</code></td>
<td>
<p>A user supplied lambda for MCP accommodating main G effect selection.</p>
</td></tr>
<tr><td><code id="PTReg_+3A_lambda2">lambda2</code></td>
<td>
<p>A user supplied lambda for MCP accommodating G-E interaction selecton.</p>
</td></tr>
<tr><td><code id="PTReg_+3A_gamma1">gamma1</code></td>
<td>
<p>The regularization parameter of the MCP penalty corresponding to G effects.</p>
</td></tr>
<tr><td><code id="PTReg_+3A_gamma2">gamma2</code></td>
<td>
<p>The regularization parameter of the MCP penalty corresponding to G-E
interactions.</p>
</td></tr>
<tr><td><code id="PTReg_+3A_max_init">max_init</code></td>
<td>
<p>The number of initializations.</p>
</td></tr>
<tr><td><code id="PTReg_+3A_h">h</code></td>
<td>
<p>The number of the trimmed samples if the parameter <code>mu</code> is not given.</p>
</td></tr>
<tr><td><code id="PTReg_+3A_tau">tau</code></td>
<td>
<p>The threshold value used in stability selection.</p>
</td></tr>
<tr><td><code id="PTReg_+3A_mu">mu</code></td>
<td>
<p>The parameter for screening outliers with extreme absolute residuals if the number
of the trimmed samples <code>h</code> is not given.</p>
</td></tr>
<tr><td><code id="PTReg_+3A_family">family</code></td>
<td>
<p>Response type of <code>Y</code> (see above).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with S3 class <code>"PTReg"</code> is returned, which is a list with the following components.
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>The call that produced this object.</p>
</td></tr>
<tr><td><code>intercept</code></td>
<td>
<p>The intercept estimate.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The matrix of the coefficients for main E effects.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>The matrix of the regression coefficients for all main G effects (the first row)
and interactions.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>The number of nonzeros.</p>
</td></tr>
<tr><td><code>BIC</code></td>
<td>
<p>Bayesian Information Criterion.</p>
</td></tr>
<tr><td><code>select_sample</code></td>
<td>
<p>Selected samples where samples with extreme absolute residuals are
trimmed.</p>
</td></tr>
<tr><td><code>family</code></td>
<td>
<p>The same as input <code>family</code>.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Yaqing Xu, Mengyun Wu, Shuangge Ma, and Syed Ejaz Ahmed.
<em>Robust gene-environment interaction analysis using penalized trimmed regression. Journal
of Statistical Computation and Simulation, 88(18):3502-3528, 2018.</em>
</p>


<h3>See Also</h3>

<p><code>coef</code>, <code>predict</code>, and <code>plot</code> methods, and <code>bic.PTReg</code> method.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sigmaG&lt;-AR(rho=0.3,p=30)
sigmaE&lt;-AR(rho=0.3,p=3)
set.seed(300)
G=MASS::mvrnorm(150,rep(0,30),sigmaG)
EC=MASS::mvrnorm(150,rep(0,2),sigmaE[1:2,1:2])
ED = matrix(rbinom((150),1,0.6),150,1)
E=cbind(EC,ED)
alpha=runif(3,0.8,1.5)
beta=matrix(0,4,30)
beta[1,1:4]=runif(4,1,1.5)
beta[2,c(1,2)]=runif(2,1,1.5)


#continuous response
y1=simulated_data(G=G,E=E,alpha=alpha,beta=beta,error=c(rnorm(130),
rcauchy(20,0,5)),family="continuous")
fit1&lt;-PTReg(G=G,E=E,y1,lambda1=0.3,lambda2=0.3,gamma1=6,gamma2=6,
max_init=50,h=NULL,tau=0.6,mu=2.5,family="continuous")
coef1=coef(fit1)
y_hat1=predict(fit1,E,G)
plot(fit1)

# survival response
y2=simulated_data(G,E,alpha,beta,rnorm(150,0,1),
family="survival",0.7,0.9)
fit2&lt;-PTReg(G=G,E=E,y2,lambda1=0.3,lambda2=0.3,gamma1=6,gamma2=6,
max_init=50,h=NULL,tau=0.6,mu=2.5,family="survival")
coef2=coef(fit2)
y_hat2=predict(fit2,E,G)
plot(fit2)


</code></pre>

<hr>
<h2 id='QPCorr.matrix'>Robust identification of gene-environment interactions using a
quantile partial correlation approach</h2><span id='topic+QPCorr.matrix'></span>

<h3>Description</h3>

<p>A robust gene-environment interaction identification approach using the quantile partial
correlation technique. This approach is a marginal analysis approach built on the quantile
regression technique, which can accommodate long-tailed or contaminated outcomes. For response
with right censoring, Kaplan-Meier (KM) estimator-based weights are adopted to easily
accommodate censoring. In addition, it adopts partial correlation to identify important
interactions while properly controlling for the main genetic (G) and environmental (E) effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QPCorr.matrix(G, E, Y, tau, w = NULL, family = c("continuous", "survival"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QPCorr.matrix_+3A_g">G</code></td>
<td>
<p>Input matrix of <code>p</code> G measurements consisting of <code>n</code> rows. Each row is an
observation vector.</p>
</td></tr>
<tr><td><code id="QPCorr.matrix_+3A_e">E</code></td>
<td>
<p>Input matrix of <code>q</code> E risk factors. Each row is an observation vector.</p>
</td></tr>
<tr><td><code id="QPCorr.matrix_+3A_y">Y</code></td>
<td>
<p>Response variable. A quantitative vector for <code>family="continuous"</code>. For
<code>family="survival"</code>, <code>Y</code> should be a two-column matrix with the first column being
the log(survival time) and the second column being the censoring indicator. The indicator is a
binary variable, with &quot;1&quot; indicating dead, and &quot;0&quot; indicating right censored.</p>
</td></tr>
<tr><td><code id="QPCorr.matrix_+3A_tau">tau</code></td>
<td>
<p>Quantile.</p>
</td></tr>
<tr><td><code id="QPCorr.matrix_+3A_w">w</code></td>
<td>
<p>Weight for accommodating censoring if <code>family</code>=&quot;survival&quot;. Default is NULL and a
Kaplan-Meier estimator-based weight is used.</p>
</td></tr>
<tr><td><code id="QPCorr.matrix_+3A_family">family</code></td>
<td>
<p>Response type of <code>Y</code> (see above).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of (censored) quantile partial correlations for interactions.
</p>


<h3>References</h3>

<p>Yaqing Xu, Mengyun Wu, Qingzhao Zhang, and Shuangge Ma.
<em>Robust identification of gene-environment interactions for prognosis using a quantile
partial correlation approach. Genomics, 111(5):1115-1123, 2019.</em>
</p>


<h3>See Also</h3>

<p><code>QPCorr.pval</code> method.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alpha=matrix(0,5,1)
alpha[1:2]=1
beta=matrix(0,6,100)
beta[1,1:5]=1
beta[2:3,1:5]=2
beta[4:6,6:7]=2
sigmaG&lt;-AR(rho=0.3,100)
sigmaE&lt;-AR(rho=0.3,5)
G&lt;-MASS::mvrnorm(200,rep(0,100),sigmaG)
E&lt;-MASS::mvrnorm(200,rep(0,5),sigmaE)
e1&lt;-rnorm(200*.05,50,1);e2&lt;-rnorm(200*.05,-50,1);e3&lt;-rnorm(200*.9)
e&lt;-c(e1,e2,e3)

# continuous
y1=simulated_data(G=G,E=E,alpha=alpha,beta=beta,error=e,family="continuous")
cpqcorr_stat1&lt;-QPCorr.matrix(G,E,y1,tau=0.5,w=NULL,family="continuous")

# survival
y2=simulated_data(G,E,alpha,beta,rnorm(200,0,1),family="survival",0.7,0.9)
cpqcorr_stat&lt;-QPCorr.matrix(G,E,y2,tau=0.5,w=NULL,family="survival")

</code></pre>

<hr>
<h2 id='QPCorr.pval'>P-values of the &quot;QPCorr.matrix&quot; obtained using a permutation approach</h2><span id='topic+QPCorr.pval'></span>

<h3>Description</h3>

<p>P-values of the <code>"QPCorr.matrix "</code> obtained using a permutation approach, the
interactions with smaller p-values are regarded as more important.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QPCorr.pval(
  G,
  E,
  Y,
  tau,
  w = NULL,
  permutation_t = 1000,
  family = c("continuous", "survival")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QPCorr.pval_+3A_g">G</code></td>
<td>
<p>Input matrix of <code>p</code> genetic (G) measurements consisting of <code>n</code> rows. Each row is an observation vector.</p>
</td></tr>
<tr><td><code id="QPCorr.pval_+3A_e">E</code></td>
<td>
<p>Input matrix of <code>q</code> environmental (E) risk factors, each row is an observation
vector.</p>
</td></tr>
<tr><td><code id="QPCorr.pval_+3A_y">Y</code></td>
<td>
<p>Response variable. A quantitative vector for <code>family="continuous"</code>. For
<code>family="survival"</code>, <code>Y</code> should be a two-column matrix with the first column being
the log(survival time) and the second column being the censoring indicator. The indicator is a
binary variable, with &quot;1&quot; indicating dead, and &quot;0&quot; indicating right censored.</p>
</td></tr>
<tr><td><code id="QPCorr.pval_+3A_tau">tau</code></td>
<td>
<p>Quantile.</p>
</td></tr>
<tr><td><code id="QPCorr.pval_+3A_w">w</code></td>
<td>
<p>Weight for accommodating censoring if <code>family="survival"</code>. Default is NULL and a
Kaplan-Meier estimator-based weight is used.</p>
</td></tr>
<tr><td><code id="QPCorr.pval_+3A_permutation_t">permutation_t</code></td>
<td>
<p>Number of permutation.</p>
</td></tr>
<tr><td><code id="QPCorr.pval_+3A_family">family</code></td>
<td>
<p>Response type of <code>Y</code> (see above).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of p-value, with the element in the <code>i</code>th row and the <code>j</code> column
represents the p-value of the (censored) quantile partial correlation corresponding to the
<code>i</code>th E and the <code>j</code>th G.
</p>


<h3>References</h3>

<p>Yaqing Xu, Mengyun Wu, Qingzhao Zhang, and Shuangge Ma.
<em>Robust identification of gene-environment interactions for prognosis using a quantile
partial correlation approach. Genomics, 111(5):1115-1123, 2019.</em>
</p>


<h3>See Also</h3>

<p><code>QPCorr.matrix </code> method.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n=50
alpha=matrix(0,5,1)
alpha[1:2]=1
beta=matrix(0,6,20)
beta[1,1:4]=1
beta[2:3,1:4]=2
sigmaG&lt;-AR(rho=0.3,20)
sigmaE&lt;-AR(rho=0.3,5)
G&lt;-MASS::mvrnorm(n,rep(0,20),sigmaG)
E&lt;-MASS::mvrnorm(n,rep(0,5),sigmaE)
e1&lt;-rnorm(n*.05,50,1);e2&lt;-rnorm(n*.05,-50,1);e3&lt;-rnorm((n-length(e1)-length(e2)))
e&lt;-c(e1,e2,e3)


# continuous
y1=simulated_data(G=G,E=E,alpha=alpha,beta=beta,error=e,family="continuous")
cpqcorr_pvalue1&lt;-QPCorr.pval(G,E,y1,tau=0.5,permutation_t=500,family="continuous")

# survival
y2=simulated_data(G,E,alpha,beta,rnorm(n,0,1),family="survival",0.7,0.9)
cpqcorr_pvalue2&lt;-QPCorr.pval(G,E,y2,tau=0.5,permutation_t=500,family="survival")

</code></pre>

<hr>
<h2 id='Rob_data'>A matrix containing the simulated data for <code>RobSBoosting</code> and <code>Miss.boosting</code> methods</h2><span id='topic+Rob_data'></span>

<h3>Description</h3>

<p>A matrix containing the simulated genetic (G) effects (the first 20 columns), environmental (E)
effects (column 21 to column 24), continuous response (column 25), logarithm of survival time
(column 26), and censoring indicator (column 27).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Rob_data)
</code></pre>


<h3>Format</h3>

<p>A matrix with 100 rows and 27 variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Rob_data)
</code></pre>

<hr>
<h2 id='RobSBoosting'>Robust semiparametric gene-environment interaction analysis using sparse boosting</h2><span id='topic+RobSBoosting'></span>

<h3>Description</h3>

<p>Robust semiparametric gene-environment interaction analysis using sparse boosting. Here a
semiparametric model is assumed to accommodate nonlinear effects, where we model continuous
environmental (E) factors in a nonlinear way, and discrete E factors and all genetic (G)
factors in a linear way. For estimating the nonlinear functions, the B spline expansion is
adopted. The Huber loss function and Qn estimator are adopted to accommodate long-tailed
distribution/data contamination. For model estimation and selection of relevant variables, we
adopt an effective sparse boosting approach, where the strong hierarchy is respected.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RobSBoosting(
  G,
  E,
  Y,
  loop_time,
  num.knots = NULL,
  Boundary.knots = NULL,
  degree = 1,
  v = 0.1,
  family = c("continuous", "survival"),
  knots = NULL,
  E_type
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RobSBoosting_+3A_g">G</code></td>
<td>
<p>Input matrix of <code>p</code> genetic measurements consisting of <code>n</code> rows. Each row
is an observation vector.</p>
</td></tr>
<tr><td><code id="RobSBoosting_+3A_e">E</code></td>
<td>
<p>Input matrix of <code>q</code> environmental risk factors, each row is an observation
vector.</p>
</td></tr>
<tr><td><code id="RobSBoosting_+3A_y">Y</code></td>
<td>
<p>Response variable. A quantitative vector for <code>family="continuous"</code>. For
<code>family="survival"</code>, <code>Y</code> should be a two-column matrix with the first column being
the log(survival time) and the second column being the censoring indicator. The indicator is a
binary variable, with &quot;1&quot; indicating dead, and &quot;0&quot; indicating right censored.</p>
</td></tr>
<tr><td><code id="RobSBoosting_+3A_loop_time">loop_time</code></td>
<td>
<p>Number of iterations of the sparse boosting.</p>
</td></tr>
<tr><td><code id="RobSBoosting_+3A_num.knots">num.knots</code></td>
<td>
<p>Numbers of knots for the B spline basis.</p>
</td></tr>
<tr><td><code id="RobSBoosting_+3A_boundary.knots">Boundary.knots</code></td>
<td>
<p>The boundary of knots for the B spline basis.</p>
</td></tr>
<tr><td><code id="RobSBoosting_+3A_degree">degree</code></td>
<td>
<p>Degree for the B spline basis.</p>
</td></tr>
<tr><td><code id="RobSBoosting_+3A_v">v</code></td>
<td>
<p>The step size used in the sparse boosting process. Default is 0.1.</p>
</td></tr>
<tr><td><code id="RobSBoosting_+3A_family">family</code></td>
<td>
<p>Response type of <code>Y</code> (see above).</p>
</td></tr>
<tr><td><code id="RobSBoosting_+3A_knots">knots</code></td>
<td>
<p>List of knots for the B spline basis. Default is NULL and knots can be generated
with the given <code>num.knots</code>, <code>degree</code> and <code>Boundary.knots</code>.</p>
</td></tr>
<tr><td><code id="RobSBoosting_+3A_e_type">E_type</code></td>
<td>
<p>A vector indicating the type of each E factor, with &quot;ED&quot; representing discrete E factor, and &quot;EC&quot; representing continuous E factor.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with S3 class <code>"RobSBoosting"</code> is returned, which is a list with the following components.
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>The call that produced this object.</p>
</td></tr>
<tr><td><code>max_t</code></td>
<td>
<p>The stopping iteration time of the sparse boosting.</p>
</td></tr>
<tr><td><code>spline_result</code></td>
<td>
<p>A list of length <code>max_t</code> that includes the estimation results of
each iteration.</p>
</td></tr>
<tr><td><code>BIC</code></td>
<td>
<p>A vector of length max_t that includes Bayesian Information Criterion based on the
Huber's prediction error.</p>
</td></tr>
<tr><td><code>variable</code></td>
<td>
<p>A vector of length max_t that includes the index of selected variable in each
iteration.</p>
</td></tr>
<tr><td><code>id</code></td>
<td>
<p>The iteration time with the smallest BIC.</p>
</td></tr>
<tr><td><code>variable_pair</code></td>
<td>
<p>A matrix with two columns that include the set of variables that can
potentially enter the regression model at the stopping iteration time. Here, the first and
second columns correspond to the indexes of E factors and G factors. For example, (1, 0)
represents that this variable is the first E factor, and (1,2) represents that the variable is
the interaction between the first E factor and second G factor. </p>
</td></tr>
<tr><td><code>v_type</code></td>
<td>
<p>A vector whose length is the number of rows of <code>variable_pair</code>, with each
element representing the variable type of the corresponding row of <code>variable_pair</code>. Here,
&quot;EC&quot; stands for continuous E effect, &quot;ED&quot; for discrete E effect, and &quot;G&quot;  for G effect, &quot;EC-G&quot;
for the interaction between &quot;EC&quot; and &quot;G&quot;, &quot;ED-G&quot; for the interaction between &quot;ED&quot; and &quot;G&quot;.</p>
</td></tr>
<tr><td><code>family</code></td>
<td>
<p>The same as input <code>family</code>.</p>
</td></tr>
<tr><td><code>degree</code></td>
<td>
<p>Degree for the B spline basis.</p>
</td></tr>
<tr><td><code>v</code></td>
<td>
<p>The step size used in the sparse boosting process.</p>
</td></tr>
<tr><td><code>NorM</code></td>
<td>
<p>The values of B spline basis.</p>
</td></tr>
<tr><td><code>estimation_results</code></td>
<td>
<p>A list of estimation results for each variable. Here, the first
<code>q</code> elemnets are for the E effects, the (<code>q+1</code>) element
is for the first G effect and the (<code>q+2</code>) to (<code>2q+1</code>) elements are for the interactions
corresponding to the first G factor, and so on.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Mengyun Wu and Shuangge Ma.
<em>Robust semiparametric gene-environment interaction analysis using sparse boosting.
Statistics in Medicine, 38(23):4625-4641, 2019.</em>
</p>


<h3>See Also</h3>

<p><code>bs</code> method for B spline expansion, <code>coef</code>, <code>predict</code>, and <code>plot</code> methods, and <code>Miss.boosting</code>
method.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Rob_data)
G=Rob_data[,1:20];E=Rob_data[,21:24]
Y=Rob_data[,25];Y_s=Rob_data[,26:27]
knots = list();Boundary.knots = matrix(0, 24, 2)
for(i in 1:4) {
  knots[[i]] = c(0, 1)
  Boundary.knots[i, ] = c(0, 1)
  }

#continuous
fit1= RobSBoosting(G,E,Y,loop_time = 80,num.knots = 2,Boundary.knots=Boundary.knots,
degree = 2,family = "continuous",knots = knots,E_type=c("EC","EC","ED","ED"))
coef1 = coef(fit1)
predict1=predict(fit1,newE=E[1:2,],newG=G[1:2,])
plot(fit1)


#survival
fit2= RobSBoosting(G,E,Y_s,loop_time = 200, num.knots = 2, Boundary.knots=Boundary.knots,
family = "survival", knots = knots,E_type=c("EC","EC","ED","ED"))
coef2 = coef(fit2)
predict2=predict(fit2,newE=E[1:2,],newG=G[1:2,])
plot(fit2)

</code></pre>

<hr>
<h2 id='simulated_data'>Simulated data for generating response</h2><span id='topic+simulated_data'></span>

<h3>Description</h3>

<p>Generate simulated response.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulated_data(
  G,
  E,
  alpha,
  beta,
  error,
  family = c("continuous", "survival"),
  a1 = NULL,
  a2 = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulated_data_+3A_g">G</code></td>
<td>
<p>Input matrix of <code>p</code> genetic (G) measurements consisting of <code>n</code> rows. Each
row is an observation vector.</p>
</td></tr>
<tr><td><code id="simulated_data_+3A_e">E</code></td>
<td>
<p>Input matrix of <code>q</code> environmental (E) risk factors. Each row is an observation
vector.</p>
</td></tr>
<tr><td><code id="simulated_data_+3A_alpha">alpha</code></td>
<td>
<p>Matrix of the true coefficients for main E effects.</p>
</td></tr>
<tr><td><code id="simulated_data_+3A_beta">beta</code></td>
<td>
<p>Matrix of the true regression coefficients for all main G effects (the first row) and interactions.</p>
</td></tr>
<tr><td><code id="simulated_data_+3A_error">error</code></td>
<td>
<p>Error terms.</p>
</td></tr>
<tr><td><code id="simulated_data_+3A_family">family</code></td>
<td>
<p>Type of the response variable. If <code>family="continuous"</code>, a quantitative
vector is generated. If <code>family="survival"</code>, a two-column matrix with the first column
being the log(survival time) and the second column being the censoring indicator is
generated.The indicator is a binary variable, with &quot;1&quot; indicating dead, and &quot;0&quot; indicating
right censored.</p>
</td></tr>
<tr><td><code id="simulated_data_+3A_a1">a1</code></td>
<td>
<p>If <code>family</code>=&quot;survival&quot;, we generate the censoring time from a uniform
distribution where <code>a1</code> is the left endpoint.</p>
</td></tr>
<tr><td><code id="simulated_data_+3A_a2">a2</code></td>
<td>
<p>If <code>family</code>=&quot;survival&quot;, we generate the censoring time from a uniform
distribution where <code>a2</code> is the right endpoint.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Response variable. A quantitative vector for <code>family="continuous"</code>. For
<code>family="survival"</code>, it would be a two-column matrix with the first column being the
log(survival time) and the second column being the censoring indicator. The indicator
is a binary variable, with &quot;1&quot; indicating dead, and &quot;0&quot; indicating right censored.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
