<!DOCTYPE html><html lang="en"><head><title>Help for package GenAlgo</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GenAlgo}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gaTourResults'><p>Results of a Genetic Algorithm</p></a></li>
<li><a href='#GenAlg'>
<p>A generic Genetic Algorithm for feature selection</p></a></li>
<li><a href='#GenAlg-class'><p>Class &quot;GenAlg&quot;</p></a></li>
<li><a href='#GenAlg-tools'><p>Utility functions for selection and mutation in genetic algorithms</p></a></li>
<li><a href='#maha'>
<p>Compute the (squared) Mahalanobis distance between two groups of vectors</p></a></li>
<li><a href='#tourData09'><p>Tour de France 2009</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>2.2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-10-13</td>
</tr>
<tr>
<td>Title:</td>
<td>Classes and Methods to Use Genetic Algorithms for Feature
Selection</td>
</tr>
<tr>
<td>Author:</td>
<td>Kevin R. Coombes</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kevin R. Coombes &lt;krc@silicovore.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, stats, MASS, oompaBase (&ge; 3.0.1), ClassDiscovery</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Biobase, xtable</td>
</tr>
<tr>
<td>Description:</td>
<td>Defines classes and methods that can be used
  to implement genetic algorithms for feature selection.  The idea is
  that we want to select a fixed number of features to combine into a
  linear classifier that can predict a binary outcome, and can use a
  genetic algorithm heuristically to select an optimal set of features.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (== 2.0)</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Microarray, Clustering</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://oompa.r-forge.r-project.org/">http://oompa.r-forge.r-project.org/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-10-15 15:15:55 UTC; KRC</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-10-15 17:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='gaTourResults'>Results of a Genetic Algorithm</h2><span id='topic+gaTourResults'></span><span id='topic+diversity'></span><span id='topic+fitter'></span><span id='topic+meanfit'></span><span id='topic+recurse'></span>

<h3>Description</h3>

<p>We ran a genetic algorithm to find the optimal 'fantasy' team for the
competition run by the Versus broadcasting network for the 2009 Tour
de France. In order to make the vignette run in a timely fashion,
we saved the results in this data object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gaTourResults)
</code></pre>


<h3>Format</h3>

<p>There are four objects in the data file. The first is <code>recurse</code>, which
is an object of the <code><a href="#topic+GenAlg-class">GenAlg-class</a></code> representing the final
generation. The other three objects are all numeric vector of length 1100:
<code>diversity</code> contains the average population diversity at each generation,
<code>fitter</code> contains the maximum fitness, and <code>meanfit</code> contains the
mean fitness.
</p>


<h3>Source</h3>

<p>Kevin R. Coombes</p>

<hr>
<h2 id='GenAlg'>
A generic Genetic Algorithm for feature selection
</h2><span id='topic+GenAlg'></span><span id='topic+newGeneration'></span><span id='topic+popDiversity'></span>

<h3>Description</h3>

<p>These functions allow you to initialize (<code>GenAlg</code>) and iterate
(<code>newGeneration</code>) a genetic algorithm to perform feature
selection for binary class prediction in the context of gene
expression microarrays or other high-throughput technologies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenAlg(data, fitfun, mutfun, context, pm=0.001, pc=0.5, gen=1)
newGeneration(ga)
popDiversity(ga)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GenAlg_+3A_data">data</code></td>
<td>

<p>The initial population of potential solutions, in the form of a data
matrix with one individual per row.</p>
</td></tr>
<tr><td><code id="GenAlg_+3A_fitfun">fitfun</code></td>
<td>

<p>A function to compute the fitness of an individual solution. Must take
two input arguments: a vector of indices into rows of the population
matrix, and a <code>context</code> list within which any other items required
by the function can be resolved. Must return a real number; higher values
indicate better fitness, with the maximum fitness occurring at the optimal
solution to the underlying numerical problem.</p>
</td></tr>
<tr><td><code id="GenAlg_+3A_mutfun">mutfun</code></td>
<td>

<p>A function to mutate individual alleles in the population. Must take two
arguments: the starting allele and a <code>context</code> list as in the
fitness function.</p>
</td></tr>
<tr><td><code id="GenAlg_+3A_context">context</code></td>
<td>

<p>A list of additional data required to perform mutation or to compute
fitness. This list is passed along as the second argument when
<code>fitfun</code> and <code>mutfun</code> are called.</p>
</td></tr>
<tr><td><code id="GenAlg_+3A_pm">pm</code></td>
<td>

<p>A real value between <code>0</code> and <code>1</code>, representing the probability
that an individual allele will be mutated.</p>
</td></tr>
<tr><td><code id="GenAlg_+3A_pc">pc</code></td>
<td>

<p>A real value between <code>0</code> and <code>1</code>, representing the probability
that crossover will occur during reproduction.</p>
</td></tr>
<tr><td><code id="GenAlg_+3A_gen">gen</code></td>
<td>

<p>An integer identifying the current generation.</p>
</td></tr>
<tr><td><code id="GenAlg_+3A_ga">ga</code></td>
<td>

<p>An object of class <code>GenAlg</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Both the <code>GenAlg</code> generator and the <code>newGeneration</code> functions
return a <code><a href="#topic+GenAlg-class">GenAlg-class</a></code> object. The <code>popDiversity</code> function
returns a real number representing the average diversity of the population.
Here diversity is defined by the number of alleles (selected features) that
differ in two individuals.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
P. Roebuck <a href="mailto:proebuck@mdanderson.org">proebuck@mdanderson.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GenAlg-class">GenAlg-class</a></code>,
<code><a href="#topic+GenAlg-tools">GenAlg-tools</a></code>,
<code><a href="#topic+maha">maha</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># generate some fake data
nFeatures &lt;- 1000
nSamples &lt;- 50
fakeData &lt;- matrix(rnorm(nFeatures*nSamples), nrow=nFeatures, ncol=nSamples)
fakeGroups &lt;- sample(c(0,1), nSamples, replace=TRUE)
myContext &lt;- list(dataset=fakeData, gps=fakeGroups)

# initialize population
n.individuals &lt;- 200
n.features &lt;- 9
y &lt;- matrix(0, n.individuals, n.features)
for (i in 1:n.individuals) {
  y[i,] &lt;- sample(1:nrow(fakeData), n.features)
}

# set up the genetic algorithm
my.ga &lt;- GenAlg(y, selectionFitness, selectionMutate, myContext, 0.001, 0.75)

# advance one generation
my.ga &lt;- newGeneration(my.ga)

</code></pre>

<hr>
<h2 id='GenAlg-class'>Class &quot;GenAlg&quot;</h2><span id='topic+GenAlg-class'></span><span id='topic+as.data.frame+2CGenAlg-method'></span><span id='topic+as.matrix+2CGenAlg-method'></span><span id='topic+summary+2CGenAlg-method'></span>

<h3>Description</h3>

<p>Objects of the <code>GenAlg</code> class represent one step (population) in the
evolution of a genetic algorithm. This algorithm has been customized to
perform feature selection for the class prediction problem.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'GenAlg'
as.data.frame(x, row.names=NULL, optional=FALSE, ...)
## S4 method for signature 'GenAlg'
as.matrix(x, ...)
## S4 method for signature 'GenAlg'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GenAlg-class_+3A_object">object</code></td>
<td>
<p>object of class <code>GenAlg</code></p>
</td></tr>
<tr><td><code id="GenAlg-class_+3A_x">x</code></td>
<td>
<p>object of class <code>GenAlg</code></p>
</td></tr>
<tr><td><code id="GenAlg-class_+3A_row.names">row.names</code></td>
<td>
<p>character vector giving the row names for the data frame,
or <code>NULL</code></p>
</td></tr>
<tr><td><code id="GenAlg-class_+3A_optional">optional</code></td>
<td>
<p>logical scalar. If <code>TRUE</code>, setting row names and
converting column names to syntactic names is optional.</p>
</td></tr>
<tr><td><code id="GenAlg-class_+3A_...">...</code></td>
<td>
<p>extra arguments for generic routines</p>
</td></tr>
</table>


<h3>Objects from the Class</h3>

<p>Objects should be created by calls to the <code><a href="#topic+GenAlg">GenAlg</a></code> generator;
they will also be created automatically as a result of applying the function
<code><a href="#topic+newGeneration">newGeneration</a></code> to an existing <code>GenAlg</code> object.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code>:</dt><dd><p>The initial population of potential solutions,
in the form of a data matrix with one individual per row.</p>
</dd>
<dt><code>fitfun</code>:</dt><dd><p>A function to compute the fitness of an
individual solution. Must take two input arguments: a vector of
indices into the rows of the population matrix, and a <code>context</code>
list within which any other items required by the function can be
resolved. Must return a real number; higher values indicate better
fitness, with the maximum fitness occurring at the optimal solution
to the underlying numerical problem.</p>
</dd> 
<dt><code>mutfun</code>:</dt><dd><p>A function to mutate individual alleles in the
population. Must take two arguments: the starting allele and a
<code>context</code> list as in the fitness function.</p>
</dd>
<dt><code>p.mutation</code>:</dt><dd><p>numeric scalar between <code>0</code> and <code>1</code>,
representing the probability that an individual allele will be mutated.</p>
</dd>
<dt><code>p.crossover</code>:</dt><dd><p>numeric scalar between <code>0</code> and <code>1</code>,
representing the probability that crossover will occur during
reproduction.</p>
</dd>
<dt><code>generation</code>:</dt><dd><p>integer scalar identifying the current
generation.</p>
</dd>
<dt><code>fitness</code>:</dt><dd><p>numeric vector containing the fitness of
all individuals in the population.</p>
</dd>
<dt><code>best.fit</code>:</dt><dd><p>A numeric value; the maximum fitness.</p>
</dd>
<dt><code>best.individual</code>:</dt><dd><p>A matrix (often with one row)
containing the individual(s) achieving the maximum fitness. </p>
</dd>
<dt><code>context</code>:</dt><dd><p>A list of additional data required to perform
mutation or to compute fitness. This list is passed along as the
second argument when <code>fitfun</code> and <code>mutfun</code> are called.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>as.data.frame</dt><dd><p><code>signature(x = "GenAlg")</code>: Converts the
<code>GenAlg</code> object into a data frame.  The first column contains
the fitness ; remaining columns contain three selected features,
given as integer indices into the rows of the original data matrix. </p>
</dd>
<dt>as.matrix</dt><dd><p><code>signature(x = "GenAlg")</code>: Converts the GenAlg
object into a matrix, following the conventions of <code>as.data.frame</code>.</p>
</dd>
<dt>summary</dt><dd><p><code>signature(object = "GenAlg")</code>: Print a summary
of the GenAlg object.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
P. Roebuck <a href="mailto:proebuck@mdanderson.org">proebuck@mdanderson.org</a>
</p>


<h3>References</h3>

<p>David Goldberg.<br />
&quot;Genetic Algorithms in Search, Optimization and Machine Learning.&quot;<br />
Addison-Wesley, 1989.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GenAlg">GenAlg</a></code>,
<code><a href="#topic+GenAlg-tools">GenAlg-tools</a></code>,
<code><a href="#topic+maha">maha</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("GenAlg")
</code></pre>

<hr>
<h2 id='GenAlg-tools'>Utility functions for selection and mutation in genetic algorithms</h2><span id='topic+GenAlg-tools'></span><span id='topic+simpleMutate'></span><span id='topic+selectionFitness'></span><span id='topic+selectionMutate'></span>

<h3>Description</h3>

<p>These functions implement specific forms of mutation and fitness
that can be used in genetic algorithms for feature selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simpleMutate(allele, context)
selectionMutate(allele, context)
selectionFitness(arow, context)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GenAlg-tools_+3A_allele">allele</code></td>
<td>

<p>In the <code>simpleMutate</code> function, <code>allele</code> is a binary
vector filled with 0's and 1's.  In the <code>selectionMutate</code>
function, <code>allele</code> is an integer (which is silently ignored;
see Details). 
</p>
</td></tr>
<tr><td><code id="GenAlg-tools_+3A_arow">arow</code></td>
<td>

<p>A vector of integer indices identifying the rows (features) to be
selected from the <code>context$dataset</code> matrix.
</p>
</td></tr>
<tr><td><code id="GenAlg-tools_+3A_context">context</code></td>
<td>

<p>A list or data frame containing auxiliary information that is needed
to resolve references from the mutation or fitness code.  In both 
<code>selectionMutate</code> and <code>selectionFitness</code>, <code>context</code>
must contain a <code>dataset</code> component that is either a matrix or a
data frame.  In <code>selectionFitness</code>, the <code>context</code> must
also include a grouping factor (with two levels) called <code>gps</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions represent 'callbacks'. They can be used in the
function <code><a href="#topic+GenAlg">GenAlg</a></code>, which creates objects. They will then
be called repeatedly (for each individual in the population) each time
the genetic algorithm is updated to the next generation.
</p>
<p>The <code>simpleMutate</code> function assumes that chromosomes are binary
vectors, so alleles simply take on the value 0 or 1. A mutation of an
allele, therefore, flips its state between those two possibilities.
</p>
<p>The <code>selectionMutate</code> and <code>selectionFitness</code> functions, by
contrast, are specialized to perform feature selection assuming a
fixed number K of features, with a goal of learning how to
distinguish between two different groups of samples. We assume that
the underlying data consists of a data frame (or matrix), with the
rows representing features (such as genes) and the columns
representing samples. In addition, there must be a grouping vector
(or factor) that assigns all of the sample columns to one of two
possible groups. These data are collected into a list,
<code>context</code>, containing a <code>dataset</code> matrix and a <code>gps</code>
factor. An individual member of the population of potential
solutions is encoded as a length K vector of indices into the rows
of the <code>dataset</code>. An individual <code>allele</code>, therefore, is a
single index identifying a row of the <code>dataset</code>. When mutating
it, we assume that it can be changed into any other possible allele;
i.e., any other row number. To compute the fitness, we use the
Mahalanobis distance between the centers of the two groups defined by
the <code>gps</code> factor.
</p>


<h3>Value</h3>

<p>Both <code>selectionMutate</code> and <code>simpleMutate</code> return an integer
value; in the simpler case, the value is guaranteed to be a 0 or 1.
The <code>selectionFitness</code> function returns a real number.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
P. Roebuck <a href="mailto:proebuck@mdanderson.org">proebuck@mdanderson.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GenAlg">GenAlg</a></code>,
<code><a href="#topic+GenAlg-class">GenAlg-class</a></code>,
<code><a href="#topic+maha">maha</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># generate some fake data
nFeatures &lt;- 1000
nSamples &lt;- 50
fakeData &lt;- matrix(rnorm(nFeatures*nSamples), nrow=nFeatures, ncol=nSamples)
fakeGroups &lt;- sample(c(0,1), nSamples, replace=TRUE)
myContext &lt;- list(dataset=fakeData, gps=fakeGroups)

# initialize population
n.individuals &lt;- 200
n.features &lt;- 9
y &lt;- matrix(0, n.individuals, n.features)
for (i in 1:n.individuals) {
  y[i,] &lt;- sample(1:nrow(fakeData), n.features)
}

# set up the genetic algorithm
my.ga &lt;- GenAlg(y, selectionFitness, selectionMutate, myContext, 0.001, 0.75)

# advance one generation
my.ga &lt;- newGeneration(my.ga)

</code></pre>

<hr>
<h2 id='maha'>
Compute the (squared) Mahalanobis distance between two groups of vectors
</h2><span id='topic+maha'></span>

<h3>Description</h3>

<p>The Mahalanobis distance between two groups of vectors 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maha(data, groups, method = "mve")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="maha_+3A_data">data</code></td>
<td>
<p>A matrix with columns representing features (or variables)
and rows representing independent samples</p>
</td></tr>
<tr><td><code id="maha_+3A_groups">groups</code></td>
<td>

<p>A factor or logical vector with length equal to the number of rows
(samples) in the <code>data</code> matrix</p>
</td></tr>
<tr><td><code id="maha_+3A_method">method</code></td>
<td>

<p>A character string determining the method that should be used to
estimate the covariance matrix. The default value of &quot;<code>mve</code>&quot;
uses the <a href="MASS.html#topic+cov.mve">cov.mve</a> function from the MASS package. The
other valid option is &quot;var&quot;, which uses the <code><a href="stats.html#topic+var">var</a></code>
function from the standard <code>stats</code> package.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Mahalanobis distance between two groups of vectors is the distance
between their centers, computed in the equivalent of a principal
component space that accounts for different variances.
</p>


<h3>Value</h3>

<p>Returns a numeric vector of length 1.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
P. Roebuck <a href="mailto:proebuck@mdanderson.org">proebuck@mdanderson.org</a>
</p>


<h3>References</h3>

<p>Mardia, K. V. and Kent, J. T. and Bibby, J. M.<br />
<em>Multivariate Analysis</em>.<br />
Academic Press, Reading, MA 1979, pp. 213&ndash;254.
</p>


<h3>See Also</h3>

<p><code><a href="MASS.html#topic+cov.mve">cov.mve</a></code>,
<code><a href="stats.html#topic+var">var</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>nFeatures &lt;- 40
nSamples &lt;- 2*10
dataset &lt;- matrix(rnorm(nSamples*nFeatures), ncol=nSamples)
groups &lt;- factor(rep(c("A", "B"), each=10))
maha(dataset, groups)
</code></pre>

<hr>
<h2 id='tourData09'>Tour de France 2009</h2><span id='topic+tourData09'></span>

<h3>Description</h3>

<p>Each row represents the performance of a rider in the
2009 Tour de France; the name and team of the rider are used as the
row names. The four columns are the <code>Cost</code> (to include on a team
in the Versus fantasy challenge), <code>Scores</code> (based on daily
finishing position), <code>JerseyBonus</code> (for any days spent in one of
the three main leader jerseys), and <code>Total</code> (the sum of Scores
and JerseyBonus).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tourData09)
</code></pre>


<h3>Format</h3>

<p>A data frame with 102 rows and 4 columns.
</p>


<h3>Source</h3>

<p> The data were collected in 2009 from the web site
<code>http://www.versus.com/tdfgames</code>, which appears to no longer
exist.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
