<!DOCTYPE html><html lang="en"><head><title>Help for package genepi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {genepi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#genepi-package'>
<p>Functions for some genetic epidemiology methods.</p></a></li>
<li><a href='#haploOR-internal'><p>Internal Functions for haplotypeOddsRatio</p></a></li>
<li><a href='#haplotypeOddsRatio'>
<p>Calculate haplotype disease risk.</p></a></li>
<li><a href='#twoStagePower'>
<p>Calculate the power of a two stage design for GWAS</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genetic Epidemiology Design and Inference</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.0.0), stats</td>
</tr>
<tr>
<td>Author:</td>
<td>Venkatraman E. Seshan</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Venkatraman E. Seshan &lt;seshanv@mskcc.org&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Package for Genetic Epidemiologic Methods Developed at MSKCC. It contains functions to calculate haplotype specific odds ratio and the power of two stage design for GWAS studies. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-31 16:07:06 UTC; seshanv</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-31 16:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='genepi-package'>
Functions for some genetic epidemiology methods.
</h2><span id='topic+genepi-package'></span><span id='topic+genepi'></span>

<h3>Description</h3>

<p>These functions provide code for genetic epidemiology methods
developed at MSKCC.  They currently include estimating haplotype
disease risk and two stage designs for GWAS.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
    Package: </td><td style="text-align: left;"> genepi</td>
</tr>
<tr>
 <td style="text-align: left;">
    Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
    Version: </td><td style="text-align: left;"> 1.0.3</td>
</tr>
<tr>
 <td style="text-align: left;">
    Date: </td><td style="text-align: left;"> 2023-08-31</td>
</tr>
<tr>
 <td style="text-align: left;">
    License: </td><td style="text-align: left;"> GPL version 2 or later</td>
</tr>
<tr>
 <td style="text-align: left;">
    LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>

<p>There are two functions <code>haplotypeOddsRatio</code> and
<code>twoStagePower</code> in this package.
</p>
<p>Package will be archived and functions added to clinfun package
</p>


<h3>Author(s)</h3>

<p>Venkatraman E. Seshan
</p>
<p>Maintainer: Venkatraman E. Seshan &lt;seshanv@mskcc.org&gt;
</p>


<h3>References</h3>

<p>Satagopan JM, Venkatraman ES, Begg CB. (2004) Two-stage designs for
gene-disease association studies with sample size constraints.
<em>Biometrics</em> 
</p>
<p>Venkatraman ES, Mitra N, Begg CB. (2004) A method for evaluating the
impact of individual haplotypes on disease incidence in molecular
epidemiology studies. <em>Stat Appl Genet Mol Biol.</em> v3:Article27.
</p>

<hr>
<h2 id='haploOR-internal'>Internal Functions for haplotypeOddsRatio</h2><span id='topic+genhaplopairs'></span><span id='topic+hwehaplofreq'></span><span id='topic+probonecopy'></span>

<h3>Description</h3>

<p>Internal functions needed for haplotypeOddsRatio.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genhaplopairs(n)
hwehaplofreq(n, two2k, g1idx, g2idx, g1tbl, hpair, tol=1e-8)
probonecopy(n, hf0, g1idx, g1tbl, hpair)
</code></pre>


<h3>Details</h3>

<p>These functions are not to be called by the user.
</p>

<hr>
<h2 id='haplotypeOddsRatio'>
Calculate haplotype disease risk.
</h2><span id='topic+haplotypeOddsRatio'></span><span id='topic+print.haploOR'></span>

<h3>Description</h3>

<p>Haplotype disease risk is calculated resolving haplotype ambiguity and
adjusting for covariates and population stratification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplotypeOddsRatio(formula, gtypevar, data, stratvar=NULL, nsim=100, tol=1e-8)
## S3 method for class 'haploOR'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haplotypeOddsRatio_+3A_formula">formula</code></td>
<td>

<p>The formula for logistic regression without the haplotype variable.
</p>
</td></tr>
<tr><td><code id="haplotypeOddsRatio_+3A_gtypevar">gtypevar</code></td>
<td>

<p>The variable names in the data frame corresponding to the loci of
interest.  Each variables counts the number of mutant genotypes a
subject has at that locus.  Legal values are 0, 1, 2 &amp; NA.
</p>
</td></tr>
<tr><td><code id="haplotypeOddsRatio_+3A_data">data</code></td>
<td>

<p>The name of the dataframe being analyzed.  It should have all the
variables in the formula as well as those in genotype and stratvar.
</p>
</td></tr>
<tr><td><code id="haplotypeOddsRatio_+3A_stratvar">stratvar</code></td>
<td>

<p>Name of the stratification variable.  This is used to account for
population stratification.  The haplotype frequencies are estimated
within each stratum.
</p>
</td></tr>
<tr><td><code id="haplotypeOddsRatio_+3A_nsim">nsim</code></td>
<td>

<p>Variance should be inflated to account for inferred ambiguous
haplotypes.  The estimates are recalculated by simulating the
disease haplotype copy number and variance added to average.
</p>
</td></tr>
<tr><td><code id="haplotypeOddsRatio_+3A_tol">tol</code></td>
<td>
<p>Tolerance limit for the EM algorithm convergence.</p>
</td></tr>
<tr><td><code id="haplotypeOddsRatio_+3A_x">x</code></td>
<td>
<p>Object of class haploOR.</p>
</td></tr>
<tr><td><code id="haplotypeOddsRatio_+3A_...">...</code></td>
<td>
<p>Other print options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This implements the method in the reference below.
</p>


<h3>Value</h3>

<p>It is a list of class haploOR
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>The function call that produced this output.</p>
</td></tr>
<tr><td><code>coef</code></td>
<td>
<p>Table with estimated coefficients, standard error,
Z-statistic and p-value.</p>
</td></tr>
<tr><td><code>var</code></td>
<td>
<p>Covariance matrix of the estimated log odds-ratiios.</p>
</td></tr>
<tr><td><code>deviance</code></td>
<td>
<p>Average of the simulated deviances.  Its theoretical
properties are unknown.</p>
</td></tr>
<tr><td><code>aic</code></td>
<td>
<p>Average of the simulated aic.</p>
</td></tr>
<tr><td><code>null.deviance</code></td>
<td>
<p>Deviance of null model.</p>
</td></tr>
<tr><td><code>df.null</code></td>
<td>
<p>Degrees of freedom of null model.</p>
</td></tr>
<tr><td><code>df.residual</code></td>
<td>
<p>Degrees of freedom of full model.</p>
</td></tr>
</table>
<p>The &quot;print&quot; method formats the results into a user-friendly table.
</p>


<h3>Author(s)</h3>

<p>Venkatraman E. Seshan
</p>


<h3>References</h3>

<p>Venkatraman ES, Mitra N, Begg CB. (2004) A method for evaluating the
impact of individual haplotypes on disease incidence in molecular
epidemiology studies. <em>Stat Appl Genet Mol Biol.</em> v3:Article27.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulated data with 2 loci haplotypes 1=00, 2=01, 3=10, 4=11
# control haplotype probabilities p[i]  i=1,2,3,4
# haplotype pairs (i&lt;=j) i=j: probs = p[i]^2 ; i&lt;j: p[i]*p[j]
p &lt;- c(0.25, 0.2, 0.2, 0.35)
p0 &lt;- rep(0, 10)
l &lt;- 0
for(i in 1:4) {for(j in i:4) {l &lt;- l+1; p0[l] &lt;- 2*p[i]*p[j]/(1+1*(i==j))}}
controls &lt;- as.numeric(cut(runif(1000), cumsum(c(0,p0)), labels=1:10))
# case probabilities disease haplotype is 11
or &lt;- c(2, 5)
p1 &lt;- p0*c(1,1,1,2,1,1,2,1,2,8); p1 &lt;- p1/sum(p1)
cases &lt;- as.numeric(cut(runif(1000), cumsum(c(0,p1)), labels=1:10))
# now pool them together and set up the data frame
dat &lt;- data.frame(status=rep(0:1, c(1000, 1000)))
# number of copies of mutant variant for locus 1
dat$gtype1 &lt;- c(0,0,1,1,0,1,1,2,2,2)[c(controls, cases)]
# number of copies of mutant variant for locus 2
dat$gtype2 &lt;- c(0,1,0,1,2,1,2,0,1,2)[c(controls, cases)]
# true number of copies of disease haplotype
dat$hcn &lt;- c(0,0,0,1,0,0,1,0,1,2)[c(controls, cases)]
# model with genotypes only
haplotypeOddsRatio(status ~ 1, c("gtype1","gtype2"), dat)
# model from the logistic fit using the number of copies of disease haplotype
glm(status ~ factor(hcn), dat, family=binomial)
</code></pre>

<hr>
<h2 id='twoStagePower'>
Calculate the power of a two stage design for GWAS
</h2><span id='topic+twoStagePower'></span>

<h3>Description</h3>

<p>Calculate the power of a two stage design for GWAS under sample size
or cost constraints.  Implements methods in the refereces below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twoStagePower(n=NULL, Cost=NULL, m=5000, mu=0.045, mu.loc=0.5, p=0.10,
            f=NULL, relcost=1, true.needed=1, rho=0, rho0=0, nsim=2000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="twoStagePower_+3A_n">n</code></td>
<td>
<p>The maximum sample size for the study.</p>
</td></tr>
<tr><td><code id="twoStagePower_+3A_cost">Cost</code></td>
<td>

<p>Maximum available resource.  One of <code>Cost</code> or <code>n</code> must be
specified.
</p>
</td></tr>
<tr><td><code id="twoStagePower_+3A_m">m</code></td>
<td>

<p>The number of markers in the study. Default is 5000.  It will take a
a long time to compute power for very large numbers e.g. 100000
</p>
</td></tr>
<tr><td><code id="twoStagePower_+3A_mu">mu</code></td>
<td>

<p>The mean vector for the markers that are associated with endpoint.
</p>
</td></tr>
<tr><td><code id="twoStagePower_+3A_mu.loc">mu.loc</code></td>
<td>

<p>The locations of the true markers.  Since the chromosome is mapped
to the unit interval (0,1) the numbers should be between 0 and 1.
</p>
</td></tr>
<tr><td><code id="twoStagePower_+3A_p">p</code></td>
<td>

<p>The proportion of markers taken to the second stage.  The default is
0.1 which is found to be optimal.
</p>
</td></tr> 
<tr><td><code id="twoStagePower_+3A_f">f</code></td>
<td>

<p>The fraction of Cost or sample size allocated to the first stage.
If not specified it uses 0.75 for the Cost constraint scenario and
0.5 for the sample size contraint scenario.
</p>
</td></tr>
<tr><td><code id="twoStagePower_+3A_relcost">relcost</code></td>
<td>

<p>Specifies how expensive it is to genotype in the second stage
compared to the first stage.
</p>
</td></tr>
<tr><td><code id="twoStagePower_+3A_true.needed">true.needed</code></td>
<td>

<p>The number of markers selected in the end.  Can be a maximum of
length of mu.loc (or mu).
</p>
</td></tr>
<tr><td><code id="twoStagePower_+3A_rho">rho</code>, <code id="twoStagePower_+3A_rho0">rho0</code></td>
<td>
<p>correlation between markers</p>
</td></tr>
<tr><td><code id="twoStagePower_+3A_nsim">nsim</code></td>
<td>
<p>Number of Monte Carlo replications to compute power.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This implements the method in the reference below.
</p>


<h3>Value</h3>

<p>It returns the power as a single numeric value
</p>


<h3>Author(s)</h3>

<p>Jaya M. Satagopan &amp; Venkatraman E. Seshan
</p>


<h3>References</h3>

<p>Satagopan JM, Venkatraman ES, Begg CB. (2004) Two-stage designs for
gene-disease association studies with sample size constraints.
<em>Biometrics</em> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>twoStagePower(n=1000)
twoStagePower(Cost=1000)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
