<!DOCTYPE html><html lang="en"><head><title>Help for package geneNR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {geneNR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#geneSNP'><p>Identifies Candidate Genes based on identified Single Nucleotide Ploymorphisms (SNPs) from Genome Wide Association Stuides (GWAS) Analysis</p></a></li>
<li><a href='#sample_data_rice'><p>Sample Data</p></a></li>
<li><a href='#sample_data_wheat'><p>Sample Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Automated Gene Identification for Post-GWAS Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Facilitates the post-Genome Wide Association Studies (GWAS) analysis of identifying candidate genes within user-defined search window, based on the identified Single Nucleotide Polymorphisms (SNPs) as given by Mazumder AK (2024) &lt;<a href="https://doi.org/10.1038%2Fs41598-024-66903-3">doi:10.1038/s41598-024-66903-3</a>&gt;. It supports candidate gene analysis for wheat and rice. Just import your GWAS result as explained in the sample_data file and the function does all the manual search and retrieve candidate genes for you, while exporting the results into ready-to-use output.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/licenses/by/4.0">CC BY 4.0</a></td>
</tr>
<tr>
<td>Copyright:</td>
<td>(C) 2025 Nirmalaruban</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Imports:</td>
<td>readr, stringr, utils, httr, rvest, xml2, writexl</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, devtools</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-07 18:37:22 UTC; nirma</td>
</tr>
<tr>
<td>Author:</td>
<td>Rajamani Nirmalaruban [aut, cre, cph],
  R. Suvitha [aut],
  Rajbir Yadav [aut],
  Meda Alekya [aut],
  Amit Kumar Mazumder [aut],
  Subramani Sugumar [aut],
  Prashanth babu [aut],
  Manjeet Kumar [aut],
  Kiran B Gaikwad [aut],
  Naresh Kumar Bainsla [aut],
  S. Bhaskar Reddy [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rajamani Nirmalaruban &lt;nirmalaruban97@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-10 14:50:34 UTC</td>
</tr>
</table>
<hr>
<h2 id='geneSNP'>Identifies Candidate Genes based on identified Single Nucleotide Ploymorphisms (SNPs) from Genome Wide Association Stuides (GWAS) Analysis</h2><span id='topic+geneSNP'></span>

<h3>Description</h3>

<p>Identifies Candidate Genes based on identified Single Nucleotide Ploymorphisms (SNPs) from Genome Wide Association Stuides (GWAS) Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneSNP(data_file, upstream = 1e+06, downstream = 1e+06, crop = "wheat")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geneSNP_+3A_data_file">data_file</code></td>
<td>
<p>The input data in .csv format. (sample_data_wheat or sample_data_rice for demo purpose)</p>
</td></tr>
<tr><td><code id="geneSNP_+3A_upstream">upstream</code></td>
<td>
<p>The search window upstream of the current position of the SNP. (default: 1000000)</p>
</td></tr>
<tr><td><code id="geneSNP_+3A_downstream">downstream</code></td>
<td>
<p>The search window downstream of the current position of the SNP. (default: 1000000)</p>
</td></tr>
<tr><td><code id="geneSNP_+3A_crop">crop</code></td>
<td>
<p>Either &quot;wheat&quot; or &quot;rice&quot;. (default: wheat)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing traits, SNP, gene_id, gene_size, and gene_type.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(system.file("extdata", "precomputed_sample_results.rda", package = "geneNR"))
message(sample_results)

result &lt;- geneSNP("sample_data_wheat", 10000, 10000, crop = "wheat")
result &lt;- geneSNP("sample_data_rice", 10000, 10000, crop = "rice")

</code></pre>

<hr>
<h2 id='sample_data_rice'>Sample Data</h2><span id='topic+sample_data_rice'></span>

<h3>Description</h3>

<p>A dataset containing sample data related to genetic markers and associated traits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_data_rice
</code></pre>


<h3>Format</h3>

<p>A data frame with columns:
</p>

<dl>
<dt>SNP</dt><dd><p>SNP identifier, character.</p>
</dd>
<dt>Chr</dt><dd><p>Chromosome location, character.</p>
</dd>
<dt>Pos</dt><dd><p>Position on the chromosome, numeric.</p>
</dd>
<dt>traits</dt><dd><p>Associated traits, character.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Basha FTM, Sar P, Bhowmick PK, Mahato A, Bisht DS, Iquebal MA, Chakraborty K, Banerjee A, Verma BC, Bhaduri D, Kumar J, Ngangkham U, Saha S, Priyamedha, Mandal NP, Roy S. Genome-wide association study identified QTLs and genes underlying early seedling vigour in aus rice (Oryza sativa L.). Mol Genet Genomics. 2024 Dec 3;299(1):112. doi: 10.1007/s00438-024-02204-8. PMID: 39625651.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sample_data_rice)    #lazy loading
</code></pre>

<hr>
<h2 id='sample_data_wheat'>Sample Data</h2><span id='topic+sample_data_wheat'></span>

<h3>Description</h3>

<p>A dataset containing sample data related to genetic markers and associated traits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_data_wheat
</code></pre>


<h3>Format</h3>

<p>A data frame with columns:
</p>

<dl>
<dt>SNP</dt><dd><p>SNP identifier, character.</p>
</dd>
<dt>Chr</dt><dd><p>Chromosome location, character.</p>
</dd>
<dt>Pos</dt><dd><p>Position on the chromosome, numeric.</p>
</dd>
<dt>traits</dt><dd><p>Associated traits, character.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Generated for demonstration purposes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sample_data_wheat)    #lazy loading
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
