<!DOCTYPE html><html lang="en-US"><head><title>Help for package geneviewer</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {geneviewer}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BRCA1_splice_variants'><p>BRCA1 Splice Variants</p></a></li>
<li><a href='#erythromycin_BlastP'><p>Erythromycin BlastP results</p></a></li>
<li><a href='#erythromycin_cluster'><p>Erythromycin Gene Cluster Data</p></a></li>
<li><a href='#gbk_features_to_df'><p>Convert GenBank Features to Data Frame</p></a></li>
<li><a href='#GC_align'><p>Align gene clusters</p></a></li>
<li><a href='#GC_annotation'><p>Add Annotations to a GC_chart</p></a></li>
<li><a href='#GC_chart'><p>Create a GC Chart Visualization</p></a></li>
<li><a href='#GC_cluster'><p>Modify Cluster Settings</p></a></li>
<li><a href='#GC_clusterFooter'><p>Add a Footer to Each Cluster in a GC Chart</p></a></li>
<li><a href='#GC_clusterLabel'><p>Set or Update Cluster Labels for a GC Chart</p></a></li>
<li><a href='#GC_clusterTitle'><p>Update cluster Title of a GC Chart Cluster</p></a></li>
<li><a href='#GC_color'><p>Update Color Scheme in Gene Chart</p></a></li>
<li><a href='#GC_coordinates'><p>Modify Coordinates in a GC Chart</p></a></li>
<li><a href='#GC_genes'><p>Modify Gene Characteristics within a Chart</p></a></li>
<li><a href='#GC_grid'><p>Update Grid Display of a GC Chart Cluster</p></a></li>
<li><a href='#GC_labels'><p>Add Labels to Each Cluster in a GC Chart</p></a></li>
<li><a href='#GC_legend'><p>Set Legend for a Gene Chart</p></a></li>
<li><a href='#GC_links'><p>Add Links to GC Chart</p></a></li>
<li><a href='#GC_normalize'><p>Normalize Gene Clusters in a Genomic Chart</p></a></li>
<li><a href='#GC_scale'><p>Update Scale of a GC Chart Cluster</p></a></li>
<li><a href='#GC_scaleBar'><p>Update Scale Bar of a GC Chart Cluster</p></a></li>
<li><a href='#GC_sequence'><p>Update Sequence Display of a GC Chart Cluster</p></a></li>
<li><a href='#GC_title'><p>Add Title to a GC_chart</p></a></li>
<li><a href='#GC_tooltip'><p>Set Tooltip for a Gene Chart</p></a></li>
<li><a href='#GC_trackMouse'><p>Track Mouse Movement in a GC_chart</p></a></li>
<li><a href='#GC_transcript'><p>Modify Transcript Characteristics within a Chart</p></a></li>
<li><a href='#genbank_to_fasta'><p>Convert GenBank to FASTA Format</p></a></li>
<li><a href='#geneviewer-shiny'><p>Shiny bindings for geneviewer</p></a></li>
<li><a href='#get_introns'><p>Calculate Intron Positions Based on Exons</p></a></li>
<li><a href='#hs_dystrophin_transcripts'><p>Human Dystrophin Transcripts Data</p></a></li>
<li><a href='#human_hox_genes'><p>Human HOX Gene Cluster Data</p></a></li>
<li><a href='#mummer_alignment'><p>Perform Sequence Alignment Using MUMmer</p></a></li>
<li><a href='#ophA_clusters'><p>ophA Gene Cluster from Omphalotus olearius</p></a></li>
<li><a href='#protein_blast'><p>Perform Protein BLAST Analysis Within Specified Clusters</p></a></li>
<li><a href='#read_bed'><p>Read BED Files</p></a></li>
<li><a href='#read_fasta'><p>Read Protein Sequences from FASTA Files</p></a></li>
<li><a href='#read_gbk'><p>Read Data from GenBank Files</p></a></li>
<li><a href='#read_gff'><p>Read GFF Files</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Cluster Visualizations</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.10</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Niels van der Velden &lt;n.s.j.vandervelden@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides tools for plotting gene clusters and transcripts by 
  importing data from GenBank, FASTA, and GFF files. It performs BLASTP and 
  MUMmer alignments [Altschul et al. (1990) &lt;<a href="https://doi.org/10.1016%2FS0022-2836%2805%2980360-2">doi:10.1016/S0022-2836(05)80360-2</a>&gt;;
  Delcher et al. (1999) &lt;<a href="https://doi.org/10.1093%2Fnar%2F27.11.2369">doi:10.1093/nar/27.11.2369</a>&gt;] and displays results on 
  gene arrow maps. Extensive customization options are available, including 
  legends, labels, annotations, scales, colors, tooltips, and more.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/nvelden/geneviewer">https://github.com/nvelden/geneviewer</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/nvelden/geneviewer/issues">https://github.com/nvelden/geneviewer/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>htmlwidgets, fontawesome, magrittr, dplyr, tidyr, rlang</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, shiny, biomaRt, pwalign, Biostrings,
testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-09 15:43:46 UTC; nielsvandervelden</td>
</tr>
<tr>
<td>Author:</td>
<td>Niels van der Velden [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-10 14:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='BRCA1_splice_variants'>BRCA1 Splice Variants</h2><span id='topic+BRCA1_splice_variants'></span>

<h3>Description</h3>

<p>This dataset contains detailed information on five different splice variants
of the BRCA1 gene. It includes data on the types of transcript features
(exon, intron, UTR), chromosomal positions, strand orientations, and
transcript lengths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BRCA1_splice_variants
</code></pre>


<h3>Format</h3>

<p>A data frame with 119 observations and 6 variables:
</p>

<dl>
<dt>ensembl_transcript_id</dt><dd><p>Unique Ensembl transcript identifiers. A character vector.</p>
</dd>
<dt>type</dt><dd><p>Type of the transcript feature, indicating whether it is an &quot;exon&quot;, or &quot;UTR&quot;. A character vector.</p>
</dd>
<dt>start</dt><dd><p>The start position of the feature on the chromosome. A numeric vector.</p>
</dd>
<dt>end</dt><dd><p>The end position of the feature on the chromosome. A numeric vector.</p>
</dd>
<dt>strand</dt><dd><p>The strand orientation of the feature, with -1 indicating the reverse strand. An integer vector.</p>
</dd>
<dt>transcript_length</dt><dd><p>The total length of the transcript in base pairs. An integer vector.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Derived from the Ensembl website
</p>

<hr>
<h2 id='erythromycin_BlastP'>Erythromycin BlastP results</h2><span id='topic+erythromycin_BlastP'></span>

<h3>Description</h3>

<p>This dataset contains detailed information on genes involved in the
biosynthesis of Erythromycin, an antibiotic produced by the bacterium
Saccharopolyspora erythraea and several homologous gene clusters identified
by antiSMASH. It includes gene identifiers, chromosomal positions,
orientations, and annotations regarding their products and functions, as well
as similarity and identity scores from BlastP analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>erythromycin_BlastP
</code></pre>


<h3>Format</h3>

<p>A data frame with 148 observations and 16 variables:
</p>

<dl>
<dt>protein_id</dt><dd><p>Unique protein identifiers. A character vector.</p>
</dd>
<dt>region</dt><dd><p>The chromosomal region of the gene, indicating start and end positions and strand. A character vector.</p>
</dd>
<dt>translation</dt><dd><p>Amino acid sequence of the protein encoded by the gene. A character vector.</p>
</dd>
<dt>cluster</dt><dd><p>Identifier of the gene cluster to which the gene belongs. A character vector.</p>
</dd>
<dt>strand</dt><dd><p>The strand orientation (&quot;forward&quot; or &quot;complement&quot;) of the gene. A character vector.</p>
</dd>
<dt>start</dt><dd><p>The start position of the gene on the chromosome. A numeric vector.</p>
</dd>
<dt>end</dt><dd><p>The end position of the gene on the chromosome. A numeric vector.</p>
</dd>
<dt>rowID</dt><dd><p>A unique identifier for each row in the dataset. An integer vector.</p>
</dd>
<dt>identity</dt><dd><p>The identity score from BlastP analysis, representing the percentage of identical matches. A numeric vector.</p>
</dd>
<dt>similarity</dt><dd><p>The similarity score from BlastP analysis, often reflecting the functional or structural similarity. A numeric vector.</p>
</dd>
<dt>BlastP</dt><dd><p>Reference to the protein_id after BlastP comparison, or NA if not applicable. A character vector.</p>
</dd>
<dt>score</dt><dd><p>Score assigned based on the BlastP analysis, quantifying the match quality. A numeric vector.</p>
</dd>
<dt>Gene</dt><dd><p>Gene name or identifier if available, otherwise NA. A character vector.</p>
</dd>
<dt>Position</dt><dd><p>Formatted string indicating the gene's position and orientation on the chromosome. A character vector.</p>
</dd>
<dt>Product</dt><dd><p>Description of the gene product. A character vector.</p>
</dd>
<dt>Functions</dt><dd><p>Functional categorization of the gene. A character vector.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Derived from BlastP analysis of Saccharopolyspora erythraea genes
involved in Erythromycin production.
</p>

<hr>
<h2 id='erythromycin_cluster'>Erythromycin Gene Cluster Data</h2><span id='topic+erythromycin_cluster'></span>

<h3>Description</h3>

<p>This dataset contains information about various genes involved in the
biosynthesis of Erythromycin, an antibiotic produced by the bacterium
Saccharopolyspora erythraea. It includes details such as gene identifiers,
start and end positions of genes, and their associated functions and products.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>erythromycin_cluster
</code></pre>


<h3>Format</h3>

<p>A data frame with 23 rows and 6 columns:
</p>

<dl>
<dt>Identifiers</dt><dd><p>Unique identifiers for the genes. A character vector.</p>
</dd>
<dt>Start</dt><dd><p>Start positions of the genes on the chromosome. A numeric vector.</p>
</dd>
<dt>End</dt><dd><p>End positions of the genes on the chromosome. A numeric vector.</p>
</dd>
<dt>Strand</dt><dd><p>Strand orientation of the gene, either &quot;+&quot; or &quot;-&quot;. A character vector.</p>
</dd>
<dt>Product</dt><dd><p>Description of the gene products. A character vector.</p>
</dd>
<dt>Functions</dt><dd><p>Functional categorization of the genes. A character vector.</p>
</dd>
</dl>


<hr>
<h2 id='gbk_features_to_df'>Convert GenBank Features to Data Frame</h2><span id='topic+gbk_features_to_df'></span>

<h3>Description</h3>

<p>This function processes a list of GenBank features (loaded by read_gbk())
and converts selected features into a data frame. It supports processing
multiple gene clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gbk_features_to_df(
  gbk_list,
  feature = "CDS",
  keys = NULL,
  process_region = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gbk_features_to_df_+3A_gbk_list">gbk_list</code></td>
<td>
<p>A list of lists where each sub-list contains GenBank
features for a specific gene cluster. Each sub-list is expected to have a
named list of features, with each feature being a character vector.</p>
</td></tr>
<tr><td><code id="gbk_features_to_df_+3A_feature">feature</code></td>
<td>
<p>A string specifying the feature type to extract from each gene
cluster's FEATURE list (e.g., &quot;CDS&quot; or &quot;gene&quot;). Defaults to &quot;CDS&quot;.</p>
</td></tr>
<tr><td><code id="gbk_features_to_df_+3A_keys">keys</code></td>
<td>
<p>An optional vector of strings representing specific keys within
the feature to retain in the final data frame. If 'NULL' (the default), all
keys within the specified feature are included.</p>
</td></tr>
<tr><td><code id="gbk_features_to_df_+3A_process_region">process_region</code></td>
<td>
<p>A boolean flag; when set to 'TRUE' (the default),
special processing is performed on the 'region' key (if present) to extract
'strand', 'start', and 'end' information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame where each row corresponds to a feature from the input
list. The data frame includes a 'cluster' column indicating the source gbk
file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gbk &lt;- read_gbk("path/to/genbank_file.gbk")
df &lt;- gbk_features_to_df(gbk)

# To extract only specific keys within the "CDS" feature
df &lt;- gbk_features_to_df(gbk, feature = "CDS", keys = c("gene", "region"))

# To disable special processing of the 'region' key
df &lt;- gbk_features_to_df(gbk, process_region = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='GC_align'>Align gene clusters</h2><span id='topic+GC_align'></span>

<h3>Description</h3>

<p>This function aligns clusters based on a specified gene id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_align(GC_chart, id_column, id, align = "left")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_align_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A chart object containing genomic data along with clustering
information.</p>
</td></tr>
<tr><td><code id="GC_align_+3A_id_column">id_column</code></td>
<td>
<p>The name of the column that contains the gene identifiers.</p>
</td></tr>
<tr><td><code id="GC_align_+3A_id">id</code></td>
<td>
<p>The specific identifier of the gene to be aligned.</p>
</td></tr>
<tr><td><code id="GC_align_+3A_align">align</code></td>
<td>
<p>The alignment method for the gene. Valid values are
&quot;left&quot;, &quot;right&quot;, or &quot;center&quot;. Defaults to &quot;left&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The modified 'GC_chart' object with updated genomic coordinates.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 90, 130, 170, 210),
  end = c(40, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 2', 'Gene 3', 'Gene 4', 'Gene 5'),
  group = c('A', 'B', 'C', 'C', 'A'),
  cluster =  c(1, 1, 1, 2, 2)
)


GC_chart(genes_data, group = "group", cluster = "cluster", height = "150px") %&gt;%
  GC_align("group", "A", align = "left") %&gt;%
  GC_legend(FALSE)

</code></pre>

<hr>
<h2 id='GC_annotation'>Add Annotations to a GC_chart</h2><span id='topic+GC_annotation'></span>

<h3>Description</h3>

<p>This function adds annotations to specified clusters within a GC chart.
Annotations can be of various types and are positioned based on provided
coordinates. The types of annotations available are: text, textMarker, line,
arrow, symbol, rectangle, promoter, and terminator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_annotation(GC_chart, type = "textAnnotation", cluster = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_annotation_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A GC chart object to which the annotations will be added.</p>
</td></tr>
<tr><td><code id="GC_annotation_+3A_type">type</code></td>
<td>
<p>Character vector specifying the type of annotations to add. The
default is &quot;text&quot;.</p>
</td></tr>
<tr><td><code id="GC_annotation_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector specifying the clusters to which
annotations should be added.</p>
</td></tr>
<tr><td><code id="GC_annotation_+3A_...">...</code></td>
<td>
<p>Additional parameters for customization of annotations, depending
on the type.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated GC chart object with added annotations.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GC_trackMouse">GC_trackMouse</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 50, 90, 130, 170, 210),
  end = c(40, 80, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 2', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 1, 2, 2, 2)
)

# Adding annotations to a GC chart
GC_chart(genes_data, cluster = "cluster", group = "group", height = "220px") %&gt;%
  GC_annotation(
    type = "textMarker",
    cluster = 1,
    position = 24,
    text = "Gene 1",
    arrowSize = 8
  ) %&gt;%
  GC_annotation(
    type = "text",
    text = "feature 1",
    x = 91,
    y = 71
  ) %&gt;%
  GC_annotation(
    type = "symbol",
    symbol = "triangle",
    x = 95,
    y = 64,
    size = 10,
    rotation = 180
  ) %&gt;%
  GC_annotation(
    type = "terminator",
    x = 81
  ) %&gt;%
  GC_annotation(
    type = "promoter",
    x = 49
  ) %&gt;%
  # Convenience function to track mouse position on hoover
  GC_trackMouse()

</code></pre>

<hr>
<h2 id='GC_chart'>Create a GC Chart Visualization</h2><span id='topic+GC_chart'></span>

<h3>Description</h3>

<p>Generates an interactive GC chart for genomic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_chart(
  data,
  start = "start",
  end = "end",
  cluster = NULL,
  group = NULL,
  strand = NULL,
  width = "100%",
  height = "400px",
  style = list(),
  elementId = NULL,
  save_button = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_chart_+3A_data">data</code></td>
<td>
<p>Data frame containing genomic information or the file path to a
folder containing '.gbk' files. When providing a file path, the data is
loaded and processed into a data frame internally.</p>
</td></tr>
<tr><td><code id="GC_chart_+3A_start">start</code></td>
<td>
<p>Column name that indicates start positions. Default is &quot;start&quot;.</p>
</td></tr>
<tr><td><code id="GC_chart_+3A_end">end</code></td>
<td>
<p>Column name that indicates end positions. Default is &quot;end&quot;.</p>
</td></tr>
<tr><td><code id="GC_chart_+3A_cluster">cluster</code></td>
<td>
<p>Optional column name used for clustering purposes. Default is
NULL.</p>
</td></tr>
<tr><td><code id="GC_chart_+3A_group">group</code></td>
<td>
<p>Column name used for gene grouping to influence color
aesthetics.</p>
</td></tr>
<tr><td><code id="GC_chart_+3A_strand">strand</code></td>
<td>
<p>Optional column name indicating strand orientation. Acceptable
values include 1, 'forward', 'sense', or '+' to represent the forward
strand, and -1, 0, 'reverse', 'antisense', &quot;complement&quot; or '-' to represent
the reverse strand. Default is NULL, meaning strand information is not
used.</p>
</td></tr>
<tr><td><code id="GC_chart_+3A_width">width</code></td>
<td>
<p>Width specification for the chart, such as '100%' or 500.
Default is unspecified.</p>
</td></tr>
<tr><td><code id="GC_chart_+3A_height">height</code></td>
<td>
<p>Height specification for the chart, such as '400px' or 300.
Default is unspecified.</p>
</td></tr>
<tr><td><code id="GC_chart_+3A_style">style</code></td>
<td>
<p>A list of CSS styles to be applied to the chart container. Each
element of the list should be a valid CSS property-value pair. For example,
list(backgroundColor = &quot;white&quot;, border = &quot;2px solid black&quot;). Default is an
empty list.</p>
</td></tr>
<tr><td><code id="GC_chart_+3A_elementid">elementId</code></td>
<td>
<p>Optional identifier string for the widget. Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_chart_+3A_save_button">save_button</code></td>
<td>
<p>Logical, whether to include a save button. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A GC chart widget.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 50, 90, 130, 170, 210),
  end = c(40, 80, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 2', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 1, 2, 2, 2)
)
# Load from data.frame
GC_chart(genes_data, group = "group", cluster = "cluster", height = "200px") %&gt;%
GC_labels("name")

# Load from folder containing .gbk files
# file_path &lt;- "~/path/to/folder/"
# GC_chart(file_path) %&gt;%

</code></pre>

<hr>
<h2 id='GC_cluster'>Modify Cluster Settings</h2><span id='topic+GC_cluster'></span>

<h3>Description</h3>

<p>This function can switch prevention of gene overlap on, adjust the spacing
between tracks and alter the styling of specified clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_cluster(
  GC_chart,
  separate_strands = NULL,
  strand_spacing = NULL,
  prevent_gene_overlap = NULL,
  overlap_spacing = NULL,
  cluster = NULL,
  style = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_cluster_+3A_gc_chart">GC_chart</code></td>
<td>
<p>The gene chart object to be modified.</p>
</td></tr>
<tr><td><code id="GC_cluster_+3A_separate_strands">separate_strands</code></td>
<td>
<p>Logical, indicating whether to vertically separate
forward and reverse genes.</p>
</td></tr>
<tr><td><code id="GC_cluster_+3A_strand_spacing">strand_spacing</code></td>
<td>
<p>Numeric, specifies the spacing between genes on
different strands. Used only if 'separate_strands' is TRUE.</p>
</td></tr>
<tr><td><code id="GC_cluster_+3A_prevent_gene_overlap">prevent_gene_overlap</code></td>
<td>
<p>Logical, indicating whether to vertically separate
overlapping genes.</p>
</td></tr>
<tr><td><code id="GC_cluster_+3A_overlap_spacing">overlap_spacing</code></td>
<td>
<p>Numeric, specifies the spacing between overlapping genes
Used only if 'prevent_gene_overlap' is TRUE.</p>
</td></tr>
<tr><td><code id="GC_cluster_+3A_cluster">cluster</code></td>
<td>
<p>Optional; used to specify which clusters in the chart should have tooltips.</p>
</td></tr>
<tr><td><code id="GC_cluster_+3A_style">style</code></td>
<td>
<p>A list of CSS styles to be applied to the gene track.
Each element of the list should be a valid CSS property-value
pair. For example, list(backgroundColor = &quot;red&quot;, color = &quot;white&quot;).</p>
</td></tr>
<tr><td><code id="GC_cluster_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the underlying functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the modified gene chart object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(1, 10, 200, 220, 600),
  end = c(10, 150, 180, 400, 400),
  name = c('Gene 1', 'Gene 2', 'Gene 3', 'Gene 4', 'Gene 5'),
  group = c('A', 'A', 'A', 'A', 'A')
)


GC_chart(genes_data, group = "group", height = "150px") %&gt;%
  GC_cluster(separate_strands=TRUE, strand_spacing = 0) %&gt;%
  GC_legend(FALSE)

</code></pre>

<hr>
<h2 id='GC_clusterFooter'>Add a Footer to Each Cluster in a GC Chart</h2><span id='topic+GC_clusterFooter'></span>

<h3>Description</h3>

<p>This function allows you to add a footer to all or specific clusters within a GC chart.
You can specify titles, subtitles, and control the display and styling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_clusterFooter(
  GC_chart,
  title = NULL,
  subtitle = NULL,
  show = TRUE,
  cluster = NULL,
  subtitleFont = list(),
  titleFont = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_clusterFooter_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A GC chart object that the footers will be added to.</p>
</td></tr>
<tr><td><code id="GC_clusterFooter_+3A_title">title</code></td>
<td>
<p>Character vector or NULL. The title to be displayed in the footer.
Multiple titles can be provided for different clusters, and they will be recycled
if there are more clusters than titles. Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_clusterFooter_+3A_subtitle">subtitle</code></td>
<td>
<p>Character vector or NULL. Subtitles to accompany the main titles.
Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_clusterFooter_+3A_show">show</code></td>
<td>
<p>Logical vector. Controls the visibility of each footer.
Default is TRUE for all clusters.</p>
</td></tr>
<tr><td><code id="GC_clusterFooter_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector specifying which clusters should have footers added or updated.
If NULL, all clusters will be updated. Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_clusterFooter_+3A_subtitlefont">subtitleFont</code></td>
<td>
<p>List, styling options for the subtitle.</p>
</td></tr>
<tr><td><code id="GC_clusterFooter_+3A_titlefont">titleFont</code></td>
<td>
<p>List, styling options for the title.</p>
</td></tr>
<tr><td><code id="GC_clusterFooter_+3A_...">...</code></td>
<td>
<p>Additional arguments for further customization of the footers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A GC chart object with updated footer settings for each specified cluster.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 90, 130, 170, 210),
  end = c(40, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 2, 2, 2)
)

# Add a simple footer with subtitle to all clusters
GC_chart(genes_data, cluster = "cluster", group = "group") %&gt;%
  GC_clusterFooter(
    title = "Cluster Footer",
    subtitle = "Cluster subtitle"
  )
# Add styling to the title and subtitle
GC_chart(genes_data, cluster = "cluster", group = "group") %&gt;%
  GC_clusterFooter(
    title = "This is a footer",
    subtitle = "Subtitle for the footer",
    spacing = 15,
    show = TRUE,
    cluster = 1,
    x = 6,
    y = -20,
    align = "center", # left / right
    spacing = 12,
    titleFont = list(
      fontSize = "12px",
      fontWeight = "bold",
      fontFamily = "sans-serif",
      fill = "black",
      cursor = "default"
    ),
    subtitleFont = list(
      fill = "grey",
      fontSize = "10px",
      fontStyle = "normal",
      fontFamily = "sans-serif",
      cursor = "default"
    )
  )

</code></pre>

<hr>
<h2 id='GC_clusterLabel'>Set or Update Cluster Labels for a GC Chart</h2><span id='topic+GC_clusterLabel'></span>

<h3>Description</h3>

<p>This function allows you to set or update the labels for specified clusters
within a GC chart. It provides flexibility in terms of the title, visibility,
width, position, and other additional customization options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_clusterLabel(
  GC_chart,
  title = NULL,
  show = TRUE,
  width = "100px",
  cluster = NULL,
  position = "left",
  wrapLabel = TRUE,
  wrapOptions = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_clusterLabel_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A GC chart object.</p>
</td></tr>
<tr><td><code id="GC_clusterLabel_+3A_title">title</code></td>
<td>
<p>Character vector. The title for the cluster label. Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_clusterLabel_+3A_show">show</code></td>
<td>
<p>Logical. Whether to show the cluster label. Default is TRUE.</p>
</td></tr>
<tr><td><code id="GC_clusterLabel_+3A_width">width</code></td>
<td>
<p>Character. The width of the cluster label. Default is &quot;100px&quot;.</p>
</td></tr>
<tr><td><code id="GC_clusterLabel_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector. Clusters in the GC chart to update.
Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_clusterLabel_+3A_position">position</code></td>
<td>
<p>Character. Position of the label, either &quot;left&quot; or &quot;right&quot;.
Default is &quot;left&quot;.</p>
</td></tr>
<tr><td><code id="GC_clusterLabel_+3A_wraplabel">wrapLabel</code></td>
<td>
<p>Logical. Indicates whether the label should be wrapped.
Default is TRUE.</p>
</td></tr>
<tr><td><code id="GC_clusterLabel_+3A_wrapoptions">wrapOptions</code></td>
<td>
<p>List. Specifies the wrapping options.
Default is an empty List.</p>
</td></tr>
<tr><td><code id="GC_clusterLabel_+3A_...">...</code></td>
<td>
<p>Additional customization arguments for the cluster label,
such as 'fontSize', 'fontStyle', 'fontWeight', 'fontFamily', 'cursor', etc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated GC chart with new or modified cluster labels.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 90, 130, 170, 210),
  end = c(40, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 2, 2, 2)
)

# Set cluster labels
GC_chart(genes_data, cluster ="cluster", group = "group", height = "200px") %&gt;%
GC_clusterLabel(title = unique(genes_data$cluster))

# Set label for a specific cluster
GC_chart(genes_data, cluster ="cluster", group = "group", height = "200px") %&gt;%
GC_clusterLabel(title = "Cluster 1", cluster = 1)

# Style labels
GC_chart(genes_data, cluster ="cluster", group = "group", height = "200px") %&gt;%
  GC_clusterLabel(
    title = c("Cluster 1", "Cluster 2"),
    width = "100px",
    x = 0,
    y = 0,
    position = "left",
    wrapLabel = TRUE,
    wrapOptions = list(
      dyAdjust = 0,
      lineHeightEms = 1.05,
      lineHeightSquishFactor =  1,
      splitOnHyphen =  TRUE,
      centreVertically = TRUE
     ),
    fontSize = "12px",
    fontStyle = "normal",
    fontWeight = "bold",
    fontFamily = "sans-serif",
    cursor = "default"
  )

</code></pre>

<hr>
<h2 id='GC_clusterTitle'>Update cluster Title of a GC Chart Cluster</h2><span id='topic+GC_clusterTitle'></span>

<h3>Description</h3>

<p>Modify the cluster title and subtitle of specified clusters within a GC chart and adjust
the display settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_clusterTitle(
  GC_chart,
  title = NULL,
  subtitle = NULL,
  subtitleFont = list(),
  titleFont = list(),
  show = TRUE,
  height = NULL,
  cluster = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_clusterTitle_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A GC chart object.</p>
</td></tr>
<tr><td><code id="GC_clusterTitle_+3A_title">title</code></td>
<td>
<p>Character vector. Titles to set for the clusters.</p>
</td></tr>
<tr><td><code id="GC_clusterTitle_+3A_subtitle">subtitle</code></td>
<td>
<p>Character vector. Subtitles to set for the clusters.</p>
</td></tr>
<tr><td><code id="GC_clusterTitle_+3A_subtitlefont">subtitleFont</code></td>
<td>
<p>List. Settings for the subtitle font.</p>
</td></tr>
<tr><td><code id="GC_clusterTitle_+3A_titlefont">titleFont</code></td>
<td>
<p>List. Settings for the title font.</p>
</td></tr>
<tr><td><code id="GC_clusterTitle_+3A_show">show</code></td>
<td>
<p>Logical. Whether to display the title. Default is TRUE.</p>
</td></tr>
<tr><td><code id="GC_clusterTitle_+3A_height">height</code></td>
<td>
<p>Character. Height for the title (e.g., &quot;40px&quot;).</p>
</td></tr>
<tr><td><code id="GC_clusterTitle_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector. Clusters in the GC chart to update.</p>
</td></tr>
<tr><td><code id="GC_clusterTitle_+3A_...">...</code></td>
<td>
<p>Additional customization arguments for title and subtitle.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated GC chart with new title settings.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 50, 90, 130, 170, 210),
  end = c(40, 80, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 2', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 1, 2, 2, 2)
)

# Basic usage
GC_chart(genes_data, cluster ="cluster", group = "group", height = "400px") %&gt;%
GC_labels("name") %&gt;%
GC_clusterTitle(
  title = "Cluster 1 Data",
  subtitle = "Detailed View",
  show = TRUE,
  cluster = 1
)

# Customizing title style
GC_chart(genes_data, cluster = "cluster", group = "group", height = "400px") %&gt;%
  GC_labels("name") %&gt;%
  GC_clusterTitle(
    title = "Cluster 1 Data",
    subtitle = "Detailed View",
    show = TRUE,
    cluster = 1,
    x = 0,
    y = 5,
    align = "center",
    spacing = 20,
    titleFont = list(
      fontSize = "16px",
      fontStyle = "normal",
      fontWeight = "bold",
      textDecoration = "normal",
      fontFamily = "sans-serif",
      cursor = "default",
      fill = "black"
      # Any other CSS styles
    ),
    subtitleFont = list(
      fontSize = "14px",
      fontStyle = "normal",
      fontWeight = "bold",
      textDecoration = "normal",
      fontFamily = "sans-serif",
      cursor = "default",
      fill = "black"
      # Any other CSS styles
    )
  )
</code></pre>

<hr>
<h2 id='GC_color'>Update Color Scheme in Gene Chart</h2><span id='topic+GC_color'></span>

<h3>Description</h3>

<p>This function updates the color scheme of the legend and genes in a gene chart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_color(GC_chart, colorScheme = NULL, customColors = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_color_+3A_gc_chart">GC_chart</code></td>
<td>
<p>The gene chart object to be modified.</p>
</td></tr>
<tr><td><code id="GC_color_+3A_colorscheme">colorScheme</code></td>
<td>
<p>Optional; character or NULL, the name of a predefined
color scheme to apply to the genes.Acceptable values include D3.js's built-in
color schemes like &quot;schemeCategory10&quot;, &quot;schemeAccent&quot;, &quot;schemeTableau10&quot;.</p>
</td></tr>
<tr><td><code id="GC_color_+3A_customcolors">customColors</code></td>
<td>
<p>Either NULL, a list of color values, or a named list of
color values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the gene chart object with updated color settings for the genes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 50, 90, 130, 170, 210),
  end = c(40, 80, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 2', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 1, 2, 2, 2)
)

GC_chart(genes_data, cluster = "cluster", group = "group", height = "100px") %&gt;%
  GC_color(colorScheme = "schemeCategory10")

GC_chart(genes_data, cluster = "cluster", group = "group", height = "100px") %&gt;%
  GC_color(customColors = c("red", "orange", "green"))

GC_chart(genes_data, cluster = "cluster", group = "group", height = "100px") %&gt;%
  GC_color(customColors = list(A = "yellow", B = "pink", C =  "purple"))


</code></pre>

<hr>
<h2 id='GC_coordinates'>Modify Coordinates in a GC Chart</h2><span id='topic+GC_coordinates'></span>

<h3>Description</h3>

<p>This function updates a GC chart by modifying the coordinates settings. It
allows for showing or hiding tick values, applying custom tick values for the
top and bottom axes, and supports several other customizations for specific
or all clusters in the chart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_coordinates(
  GC_chart,
  show = TRUE,
  tickValuesTop = NULL,
  tickValuesBottom = NULL,
  ticksFormat = NULL,
  tickStyle = list(),
  textStyle = list(),
  cluster = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_coordinates_+3A_gc_chart">GC_chart</code></td>
<td>
<p>The GC chart object to be modified.</p>
</td></tr>
<tr><td><code id="GC_coordinates_+3A_show">show</code></td>
<td>
<p>Logical, whether to show the tick values or not. Can be a single
value or a vector.</p>
</td></tr>
<tr><td><code id="GC_coordinates_+3A_tickvaluestop">tickValuesTop</code></td>
<td>
<p>Numeric vector or NULL, custom tick values to be used at
the top of the cluster. If NULL, the default tick values are used.</p>
</td></tr>
<tr><td><code id="GC_coordinates_+3A_tickvaluesbottom">tickValuesBottom</code></td>
<td>
<p>Numeric vector or NULL, custom tick values to be used
at the bottom of the cluster. If NULL, the default tick values are used.</p>
</td></tr>
<tr><td><code id="GC_coordinates_+3A_ticksformat">ticksFormat</code></td>
<td>
<p>Format for tick labels. Default is &quot;,.0f&quot;.</p>
</td></tr>
<tr><td><code id="GC_coordinates_+3A_tickstyle">tickStyle</code></td>
<td>
<p>List, styling options for the ticks.</p>
</td></tr>
<tr><td><code id="GC_coordinates_+3A_textstyle">textStyle</code></td>
<td>
<p>List, styling options for the text.
the coordinate modifications. If NULL, applies to all clusters.</p>
</td></tr>
<tr><td><code id="GC_coordinates_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector or NULL; specifies which clusters
to generate coordinates for.
If NULL, labels will be applied to all clusters. Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_coordinates_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the coordinate options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the GC chart object with updated coordinates.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 90, 130, 170, 210),
  end = c(40, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 2', 'Gene 3', 'Gene 4', 'Gene 5'),
  cluster = c(1, 1, 2, 2, 2)
)

# Add coordinates to all clusters
GC_chart(genes_data, cluster = "cluster", group = "name", height = "200px") %&gt;%
GC_coordinates()

# Modify coordinates of a specific cluster
GC_chart(genes_data, cluster = "cluster", group = "name", height = "200px") %&gt;%
GC_coordinates() %&gt;%
GC_coordinates(
  cluster = 2,
  show = TRUE,
  tickValuesTop = c(130, 170, 210, 240),
  tickValuesBottom = c(160, 200),
  ticksFormat = ",.0f", # ".2s",
  rotate = -45,
  yPositionTop = 55,
  yPositionBottom = 45,
  overlapThreshold = 20,
  tickStyle = list(
  stroke = "black",
  strokeWidth = 1,
  lineLength = 6
  ),
  textStyle = list(
  fill = "black",
  fontSize = "12px",
  fontFamily = "Arial",
  cursor = "default"
  )
)

</code></pre>

<hr>
<h2 id='GC_genes'>Modify Gene Characteristics within a Chart</h2><span id='topic+GC_genes'></span>

<h3>Description</h3>

<p>This function updates a gene chart with specific characteristics for genes
based on the given parameters. It can show/hide genes, apply a color scheme,
assign custom colors, filter by cluster, and accept additional options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_genes(
  GC_chart,
  group = NULL,
  marker = NULL,
  marker_size = NULL,
  show = TRUE,
  colorScheme = NULL,
  customColors = NULL,
  cluster = NULL,
  itemStyle = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_genes_+3A_gc_chart">GC_chart</code></td>
<td>
<p>The gene chart object to be modified.</p>
</td></tr>
<tr><td><code id="GC_genes_+3A_group">group</code></td>
<td>
<p>Column name used for gene grouping to influence color
aesthetics.</p>
</td></tr>
<tr><td><code id="GC_genes_+3A_marker">marker</code></td>
<td>
<p>Character or NULL, type of marker to represent genes on the chart.
Allowed values are 'arrow', 'boxarrow', 'box', 'cbox', and 'rbox'.</p>
</td></tr>
<tr><td><code id="GC_genes_+3A_marker_size">marker_size</code></td>
<td>
<p>Character or NULL, size category of the marker
('small', 'medium', 'large').</p>
</td></tr>
<tr><td><code id="GC_genes_+3A_show">show</code></td>
<td>
<p>Logical, whether to show the genes or not.</p>
</td></tr>
<tr><td><code id="GC_genes_+3A_colorscheme">colorScheme</code></td>
<td>
<p>Character or NULL, the name of the color scheme to use.</p>
</td></tr>
<tr><td><code id="GC_genes_+3A_customcolors">customColors</code></td>
<td>
<p>List or NULL, custom colors to apply to the genes.</p>
</td></tr>
<tr><td><code id="GC_genes_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character, the specific cluster to filter genes by.</p>
</td></tr>
<tr><td><code id="GC_genes_+3A_itemstyle">itemStyle</code></td>
<td>
<p>List, a list of styles to apply to individual items in the
chart.</p>
</td></tr>
<tr><td><code id="GC_genes_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the gene options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the modified gene chart object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 90, 130, 170, 210),
  end = c(40, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 2, 2, 2)
)

# Change the appearance of a specific gene cluster
GC_chart(genes_data, cluster = "cluster", group = "group", height = "200px") %&gt;%
  GC_genes(
    group = "group",
    show = TRUE,
    marker = "arrow",
    marker_size = "medium",
    colorScheme = NULL, # One of D3.js build in colorSchemes
                        # (eg. "schemeCategory10",
                        # "schemeAccent", "schemeTableau10")
    customColors = NULL, # A vector of color names
    prevent_overlap = FALSE,
    gene_overlap_spacing = 40,
    cluster = 1, # Specify a specific cluster
    x = 1,
    y = 50,
    stroke = "black",
    strokeWidth = 1,
    arrowheadWidth = NULL,
    arrowheadHeight = NULL,
    arrowHeight = NULL,
    markerHeight = NULL # overwrites marker_size
   )

# Change the appearance of a specific gene
GC_chart(genes_data, cluster = "cluster", group = "group", height = "200px") %&gt;%
GC_genes(
  cluster = 2,
  itemStyle = list(list(index = 2, fill = "red", stroke = "blue")
  )
 )

</code></pre>

<hr>
<h2 id='GC_grid'>Update Grid Display of a GC Chart Cluster</h2><span id='topic+GC_grid'></span>

<h3>Description</h3>

<p>Modify the grid display of specified clusters within a GC chart. This
function allows users to adjust the margins, width, and height of the grid
for each cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_grid(
  GC_chart,
  margin = NULL,
  width = NULL,
  height = NULL,
  direction = "column",
  cluster = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_grid_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A GC chart object.</p>
</td></tr>
<tr><td><code id="GC_grid_+3A_margin">margin</code></td>
<td>
<p>A list specifying top, right, bottom, and left margins.</p>
</td></tr>
<tr><td><code id="GC_grid_+3A_width">width</code></td>
<td>
<p>Numeric or character. Width of the grid. If numeric, will be
considered as percentage.</p>
</td></tr>
<tr><td><code id="GC_grid_+3A_height">height</code></td>
<td>
<p>Numeric. Height of the grid.</p>
</td></tr>
<tr><td><code id="GC_grid_+3A_direction">direction</code></td>
<td>
<p>Character. Layout direction of the grid, either &quot;column&quot;
(default) for vertical or &quot;row&quot; for horizontal.</p>
</td></tr>
<tr><td><code id="GC_grid_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector. Clusters in the GC chart to update.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated GC chart with new grid display settings.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 90, 130, 170, 210),
  end = c(40, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 2, 2, 2)
)

# Set Margin of clusters
GC_chart(genes_data, cluster ="cluster", group = "group", height = "200px") %&gt;%
GC_grid(margin = list(left = "50px", right = "0px"))

# Set height of a specific cluster
GC_chart(genes_data, cluster ="cluster", group = "group", height = "200px") %&gt;%
GC_grid(height = "120px", cluster = 2)

</code></pre>

<hr>
<h2 id='GC_labels'>Add Labels to Each Cluster in a GC Chart</h2><span id='topic+GC_labels'></span>

<h3>Description</h3>

<p>This function adds labels to each cluster within a GC chart. It provides the
option to show or hide the labels and supports customization of label properties.
It can automatically pick up group names as labels from the 'GC_chart' object if not provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_labels(
  GC_chart,
  label = NULL,
  show = TRUE,
  cluster = NULL,
  itemStyle = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_labels_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A 'GC chart' object to which labels will be added.</p>
</td></tr>
<tr><td><code id="GC_labels_+3A_label">label</code></td>
<td>
<p>Character vector, logical, or NULL. Specific labels for the clusters.
If NULL and 'GC_chart' has group names, those will be used as labels.</p>
</td></tr>
<tr><td><code id="GC_labels_+3A_show">show</code></td>
<td>
<p>Logical; controls the visibility of labels. Default is 'TRUE'.</p>
</td></tr>
<tr><td><code id="GC_labels_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector or NULL; specifies which clusters should be labeled.
If NULL, labels will be applied to all clusters. Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_labels_+3A_itemstyle">itemStyle</code></td>
<td>
<p>List, a list of styles to apply to individual items in the chart.</p>
</td></tr>
<tr><td><code id="GC_labels_+3A_...">...</code></td>
<td>
<p>Additional arguments for further customization of the labels.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'GC chart' object with updated label settings for each specified cluster.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 90, 130, 170, 210),
  end = c(40, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 2, 2, 2)
)

# Add labels to all clusters
GC_chart(genes_data, cluster = "cluster", group = "group", height = "200px") %&gt;%
GC_labels()

# Add labels and styling
GC_chart(genes_data, cluster = "cluster", group = "group", height = "200px") %&gt;%
GC_labels(
 label = "group",
 show = TRUE,
 x = 0,
 y = 50,
 dy = "-1.2em",
 dx = "0em",
 rotate = 0,
 adjustLabels = TRUE, # Rotate labels to prevent overlap
 fontSize = "12px",
 fontStyle = "italic",
 fill = "black",
 fontFamily = "sans-serif",
 textAnchor = "middle",
 cursor = "default",
)

# Alter style of a specific label
GC_chart(genes_data, cluster = "cluster", group = "group", height = "200px") %&gt;%
GC_labels(label = "group") %&gt;%
GC_labels(
  cluster = 1,
  itemStyle = list(
    list(index = 0, fill = "red", fontSize = "14px", fontWeight = "bold")
  )
)

</code></pre>

<hr>
<h2 id='GC_legend'>Set Legend for a Gene Chart</h2><span id='topic+GC_legend'></span>

<h3>Description</h3>

<p>This function configures the legend for a gene chart. It allows toggling the
legend's visibility, setting a background color, and assigning custom labels
for the legend entries. The function can also handle additional
customizations through various options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_legend(
  GC_chart,
  group = NULL,
  show = TRUE,
  backgroundColor = "#0000",
  order = list(),
  position = "bottom",
  style = list(),
  legendOptions = list(),
  legendTextOptions = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_legend_+3A_gc_chart">GC_chart</code></td>
<td>
<p>The gene chart object to be modified.</p>
</td></tr>
<tr><td><code id="GC_legend_+3A_group">group</code></td>
<td>
<p>Optional; character or NULL, specifies the groups to include in
the legend. If NULL, groups are taken from the 'group' attribute of the
'GC_chart' object.</p>
</td></tr>
<tr><td><code id="GC_legend_+3A_show">show</code></td>
<td>
<p>Logical, specifies whether to display the legend.</p>
</td></tr>
<tr><td><code id="GC_legend_+3A_backgroundcolor">backgroundColor</code></td>
<td>
<p>String, the background color of the legend.</p>
</td></tr>
<tr><td><code id="GC_legend_+3A_order">order</code></td>
<td>
<p>Optional; list, specifies the order of the legend labels.</p>
</td></tr>
<tr><td><code id="GC_legend_+3A_position">position</code></td>
<td>
<p>Character. Position of the legend, either &quot;top&quot; or &quot;bottom&quot;.
Default is &quot;bottom&quot;.</p>
</td></tr>
<tr><td><code id="GC_legend_+3A_style">style</code></td>
<td>
<p>A list of CSS styles to be applied to the chart container. Each
element of the list should be a valid CSS property-value pair. For example,
list(backgroundColor = &quot;white&quot;, border = &quot;2px solid black&quot;). Default is an
empty list.</p>
</td></tr>
<tr><td><code id="GC_legend_+3A_legendoptions">legendOptions</code></td>
<td>
<p>List, additional options for the legend.</p>
</td></tr>
<tr><td><code id="GC_legend_+3A_legendtextoptions">legendTextOptions</code></td>
<td>
<p>List, additional text options for the legend.</p>
</td></tr>
<tr><td><code id="GC_legend_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the legend configuration.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the modified gene chart object with the legend configured.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 90, 130, 170, 210),
  end = c(40, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'B', 'B', 'A', 'C'),
  cluster = c('A', 'A', 'A', 'B', 'B')
)

# Customize legend
GC_chart(genes_data, cluster = "cluster", group = "group", height = "200px") %&gt;%
  GC_legend(
    position = "top", #bottom
    orientation = "horizontal", #vertical
    x = 0,
    y = 0,
    adjustHeight = TRUE,
    backgroundColor = "#0000",
    order = list(),
    positions = "bottom",
    style = list(
      backgroundColor = "#0000"
      # Any other CSS styles
    ),
    legendOptions = list(
      cursor = "pointer",
      colorScheme = NULL,
      customColors = NULL # c("red", "green", "orange")
      # Additional styles
    ),
    legendTextOptions = list(
      cursor = "pointer",
      textAnchor = "start",
      dy = ".35em",
      fontSize = "12px",
      fontFamily = "sans-serif",
      fill = "black"
      # Additional styles
    )
  )

</code></pre>

<hr>
<h2 id='GC_links'>Add Links to GC Chart</h2><span id='topic+GC_links'></span>

<h3>Description</h3>

<p>Add links generated by 'get_links' to a 'GC_chart' object. Links are added
to the graph by their respective rowIDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_links(
  GC_chart,
  group = NULL,
  data = NULL,
  value1 = NULL,
  value2 = NULL,
  cluster = NULL,
  curve = TRUE,
  measure = "identity",
  show_links = TRUE,
  label = TRUE,
  normal_color = "#969696",
  inverted_color = "#d62728",
  use_group_colors = FALSE,
  color_bar = TRUE,
  colorBarOptions = list(),
  linkWidth = NULL,
  linkStyle = list(),
  labelStyle = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_links_+3A_gc_chart">GC_chart</code></td>
<td>
<p>Gene chart object.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_group">group</code></td>
<td>
<p>The name of the column in the data to create value pairs from.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_data">data</code></td>
<td>
<p>data.frame containing linking data.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_value1">value1</code></td>
<td>
<p>Optional vector of group values to generate links for.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_value2">value2</code></td>
<td>
<p>Optional vector of group values to generate links for.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector or NULL; specifies which clusters.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_curve">curve</code></td>
<td>
<p>Logical; if 'TRUE', links are curved, otherwise straight.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_measure">measure</code></td>
<td>
<p>Character; specifies which measure to use for link color intensity. Should be &quot;identity&quot;, &quot;similarity&quot;, or &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_show_links">show_links</code></td>
<td>
<p>Logical; if 'TRUE', links are shown, otherwise hidden.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_label">label</code></td>
<td>
<p>Logical; if 'TRUE', shows measure labels on the links, otherwise hidden.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_normal_color">normal_color</code></td>
<td>
<p>Color for the links in their normal state.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_inverted_color">inverted_color</code></td>
<td>
<p>Color for inverted links.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_use_group_colors">use_group_colors</code></td>
<td>
<p>Logical; if 'TRUE', color links by group.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_color_bar">color_bar</code></td>
<td>
<p>Logical; if 'TRUE', the color bar is displayed.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_colorbaroptions">colorBarOptions</code></td>
<td>
<p>List of options to customize the color bar appearance.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_linkwidth">linkWidth</code></td>
<td>
<p>Numeric; specifies the width of the links. A value of '1'
represents full width, and '0' represents no width.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_linkstyle">linkStyle</code></td>
<td>
<p>A list of CSS styles to apply to the links.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_labelstyle">labelStyle</code></td>
<td>
<p>A list of CSS styles specifically for the labels.</p>
</td></tr>
<tr><td><code id="GC_links_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the links.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Modified 'GC_chart' object with added links.
</p>


<h3>See Also</h3>

<p>* [get_links()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Add links between all groups in each cluster
genes_data &lt;- data.frame(
  start = c(10, 90, 130, 170, 240, 250, 300, 340, 380, 420),
  end = c(40, 120, 160, 200, 210, 270, 330, 370, 410, 450),
  name = c('Gene 1', 'Gene 2', 'Gene 3', 'Gene 4', 'Gene 5',
           'Gene 6', 'Gene 7', 'Gene 8', 'Gene 9', 'Gene 10'),
  group = c('A', 'B', 'C', 'A', 'B', 'C', 'A', 'B', 'C', 'D'),
  identity = c(NA, NA, NA, 50, 40, 100, 60, 65, 20, NA),
  similarity = c(NA, NA, NA, 40, 30, 90, 50, 55, 10, NA),
  cluster = c(1, 1, 1, 2, 2, 2, 3, 3, 3, 3)
)
GC_chart(genes_data,
         cluster = "cluster",
         height = "200px") %&gt;%
  GC_links(
    group = "group",
    value1 = "A",
    value2 = "B",
    measure = "identity",
    label = FALSE) %&gt;%
  GC_labels(label = "group")

# Add links between group A of cluster 1 and A and B of cluster 2
GC_chart(genes_data,
         cluster = "cluster",
         height = "200px") %&gt;%
         GC_labels(label = "group") %&gt;%
         GC_links(group = "group",
                   value1 = c("A", "A"),
                   value2 = c("B", "A"),
                   label = FALSE,
                   cluster = c(1,2))

# Style links and color bar
GC_chart(genes_data,
         cluster = "cluster",
         height = "200px"
) %&gt;%
  GC_links(
    group = "group",
    data = NULL,
    curve = TRUE,
    measure = "identity",
    show_links = TRUE,
    label = TRUE,
    normal_color = "#1f77b4",
    inverted_color = "#d62728",
    use_group_colors = FALSE,
    color_bar = TRUE,
    colorBarOptions = list(
      x = 0,
      y = 24,
      width = 10,
      height = 60,
      labelOptions = list(
        fontSize = 8,
        xOffset = 2,
        yOffset = 0
        # Any other CSS style
      ),
      titleOptions = list(
        fontSize = 10,
        xOffset = 2,
        yOffset = 0
        # Any other CSS style
      ),
      barOptions = list(
        stroke = "#000",
        strokeWidth = 0.5,
        opacity = 1
        # Any other CSS style
      )
    ),
    linkWidth = 1,
    linkStyle = list(
      stroke = "black",
      strokeWidth = 0.5,
      fillOpacity = 0.4
      # Any other CSS style
    ),
    labelStyle = list(
      fontSize = "8px"
      # Any other CSS style
    )
  ) %&gt;%
  GC_labels(label = "group", cluster = 1) %&gt;%
  GC_clusterLabel()
</code></pre>

<hr>
<h2 id='GC_normalize'>Normalize Gene Clusters in a Genomic Chart</h2><span id='topic+GC_normalize'></span>

<h3>Description</h3>

<p>This function normalizes the genomic coordinates of a set of genes within a
cluster, ensuring that the genes are evenly spaced along the entire range of
the cluster. The function allows for the option to preserve the original gene
lengths and to introduce customized gaps between the genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_normalize(
  GC_chart,
  group = NULL,
  cluster = NULL,
  normalize = TRUE,
  preserve_gene_length = TRUE,
  gap = NULL,
  custom_gaps = list()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_normalize_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A chart object containing genomic data and clustering
information. The chart object must include a 'series' component with data
for each cluster.</p>
</td></tr>
<tr><td><code id="GC_normalize_+3A_group">group</code></td>
<td>
<p>Column name containing gene names to apply custom gaps to.</p>
</td></tr>
<tr><td><code id="GC_normalize_+3A_cluster">cluster</code></td>
<td>
<p>A vector of cluster names to be normalized. If 'NULL', all
clusters in the 'GC_chart' will be normalized.</p>
</td></tr>
<tr><td><code id="GC_normalize_+3A_normalize">normalize</code></td>
<td>
<p>A logical value indicating whether to normalize the gene
positions within each cluster. Default is 'TRUE'.</p>
</td></tr>
<tr><td><code id="GC_normalize_+3A_preserve_gene_length">preserve_gene_length</code></td>
<td>
<p>A logical vector indicating whether to preserve
the original gene lengths for each cluster. If a single value is provided,
it is recycled for all clusters.</p>
</td></tr>
<tr><td><code id="GC_normalize_+3A_gap">gap</code></td>
<td>
<p>A numeric vector specifying the proportion of the total length to
be used as the gap between genes in each cluster. The value must be between
0 and 1. If a single value is provided, it is recycled for all clusters. If
'NULL', the gaps are calculated based on the actual spacing between the
genes in the original data.</p>
</td></tr>
<tr><td><code id="GC_normalize_+3A_custom_gaps">custom_gaps</code></td>
<td>
<p>A named list where each name corresponds to a gene name and
each value is a numeric factor by which to adjust the gap after that gene.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The modified 'GC_chart' object with normalized gene coordinates for
the specified clusters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 90, 130, 10, 90, 130),
  end = c(40, 120, 160, 40, 120, 160),
  name = c('Gene 1', 'Gene 2', 'Gene 3','Gene 1', 'Gene 2', 'Gene 3'),
  group = c('A', 'B', 'C', 'A', 'B', 'C'),
  cluster =  c(1, 1, 1, 2, 2, 2)
)

GC_chart(genes_data, group = "group", cluster = "cluster", height = "150px") %&gt;%
  GC_normalize(
    cluster = 2,
    preserve_gene_length = TRUE,
    gap = NULL
 )

</code></pre>

<hr>
<h2 id='GC_scale'>Update Scale of a GC Chart Cluster</h2><span id='topic+GC_scale'></span>

<h3>Description</h3>

<p>Modify the scale settings for specified clusters within a GC chart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_scale(
  GC_chart,
  cluster = NULL,
  start = NULL,
  end = NULL,
  padding = 2,
  hidden = FALSE,
  breaks = list(),
  tickValues = NULL,
  reverse = FALSE,
  axis_position = "bottom",
  axis_type = "position",
  y = NULL,
  scale_breaks = FALSE,
  scale_break_threshold = 20,
  scale_break_padding = 1,
  ticksCount = 10,
  ticksFormat = NULL,
  tickStyle = list(),
  textStyle = list(),
  lineStyle = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_scale_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A GC chart object.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector specifying clusters in the GC
chart to update.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_start">start</code></td>
<td>
<p>Numeric vector indicating the starting point for the scale.
Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_end">end</code></td>
<td>
<p>Numeric vector indicating the end point for the scale.
Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_padding">padding</code></td>
<td>
<p>Numeric value or percentage string indicating the padding on
either side of the scale. The value can be a number or a string in the
format of '2%'. Default is 2.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_hidden">hidden</code></td>
<td>
<p>Logical flag indicating whether the axis is hidden. Default is
FALSE.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_breaks">breaks</code></td>
<td>
<p>List specifying settings for the scale breaks. Default is an
empty list ().</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_tickvalues">tickValues</code></td>
<td>
<p>Numeric vector or NULL, custom tick values to be used at
the top of the cluster. If NULL, the default tick values are used.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_reverse">reverse</code></td>
<td>
<p>Logical flag indicating whether to reverse the scale for the
corresponding cluster. Default is FALSE.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_axis_position">axis_position</code></td>
<td>
<p>Character string indicating the type of the axis ('top' or
bottom'). Default is 'bottom'.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_axis_type">axis_type</code></td>
<td>
<p>Character string indicating the type of the axis ('position' or
'range'). Default is 'position'.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_y">y</code></td>
<td>
<p>Numeric value from 1 to 100 indicating the y-position of the x-axis.
Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_scale_breaks">scale_breaks</code></td>
<td>
<p>Logical flag indicating if scale breaks should be
employed. Default is FALSE.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_scale_break_threshold">scale_break_threshold</code></td>
<td>
<p>Numeric value indicating the threshold
percentage for determining scale breaks. Default is 20.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_scale_break_padding">scale_break_padding</code></td>
<td>
<p>Numeric value indicating the padding on either
side of a scale break. Default is 1.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_tickscount">ticksCount</code></td>
<td>
<p>Numeric value indicating the number of ticks on the scale.
Default is 10.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_ticksformat">ticksFormat</code></td>
<td>
<p>Format for tick labels; depends on axis_type, defaulting
to &quot;,.0f&quot; for 'position' or &quot;.2s&quot; for 'range' when NULL.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_tickstyle">tickStyle</code></td>
<td>
<p>List specifying the style for the ticks.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_textstyle">textStyle</code></td>
<td>
<p>List specifying the style for the tick text.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_linestyle">lineStyle</code></td>
<td>
<p>List specifying the style for the axis line.</p>
</td></tr>
<tr><td><code id="GC_scale_+3A_...">...</code></td>
<td>
<p>Additional arguments for scale settings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated GC chart with new scale settings.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(100, 1000, 2000),
  end = c(150, 1500, 2500),
  name = c('Gene 4', 'Gene 5', 'Gene 6'),
  group = c('B', 'A', 'C'),
  cluster = c(2, 2, 2)
)
#Example usage with all custom options
GC_chart(genes_data, cluster = "cluster", group = "group", height = "200px") %&gt;%
  GC_scale(
    start = 1,
    end = 2600,
    padding = 2,
    hidden = FALSE,
    reverse = FALSE,
    axis_position = "bottom",
    # breaks = list(
    #  list(start = 160, end = 900),
    #  list(start = 1600, end = 1900)
    # ),
    # tickValues = c(1, 2600),
    scale_breaks = TRUE,
    scale_break_threshold = 20,
    scale_break_padding = 1,
    ticksCount = 10,
    ticksFormat = ",.0f",
    y = NULL,
    tickStyle =
      list(
        stroke = "grey",
        strokeWidth = 1,
        lineLength = 6
        # Any other CSS styles
      ),
    textStyle =
      list(
        fill = "black",
        fontSize = "10px",
        fontFamily = "Arial",
        cursor = "default"
        # Any other CSS styles
      ),
    lineStyle = list(
      stroke = "grey",
      strokeWidth = 1
      # Any other CSS styles
    )
  ) %&gt;%
  GC_legend(FALSE)
</code></pre>

<hr>
<h2 id='GC_scaleBar'>Update Scale Bar of a GC Chart Cluster</h2><span id='topic+GC_scaleBar'></span>

<h3>Description</h3>

<p>Modify the scale bar settings for specified clusters within a GC chart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_scaleBar(
  GC_chart,
  show = TRUE,
  cluster = NULL,
  scaleBarLineStyle = list(),
  scaleBarTickStyle = list(),
  labelStyle = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_scaleBar_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A GC chart object.</p>
</td></tr>
<tr><td><code id="GC_scaleBar_+3A_show">show</code></td>
<td>
<p>Logical. Whether to show the scale bar.</p>
</td></tr>
<tr><td><code id="GC_scaleBar_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector. Clusters in the GC chart to update.</p>
</td></tr>
<tr><td><code id="GC_scaleBar_+3A_scalebarlinestyle">scaleBarLineStyle</code></td>
<td>
<p>List of style options for the scale bar line.</p>
</td></tr>
<tr><td><code id="GC_scaleBar_+3A_scalebartickstyle">scaleBarTickStyle</code></td>
<td>
<p>List of style options for the scale bar ticks.</p>
</td></tr>
<tr><td><code id="GC_scaleBar_+3A_labelstyle">labelStyle</code></td>
<td>
<p>List of style options for the scale bar label.</p>
</td></tr>
<tr><td><code id="GC_scaleBar_+3A_...">...</code></td>
<td>
<p>Additional arguments for scale bar settings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated GC chart with new scale bar settings.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
 start = c(1000, 9000, 13000, 17000, 21000),
 end = c(4000, 12000, 16000, 20000, 24000),
 name = c('Gene 1', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
 group = c('A', 'B', 'B', 'A', 'C'),
 cluster = c(1, 1, 2, 2, 2)
)

# Set scale bar for individual clusters
GC_chart(genes_data, cluster ="cluster", group = "group") %&gt;%
GC_scaleBar(cluster = 1, title = "1 kb", scaleBarUnit = 1000) %&gt;%
GC_scaleBar(cluster = 2, title = "2 kb", scaleBarUnit = 2000)

# Style scale bar
GC_chart(genes_data, cluster ="cluster", group = "group", height = "400px") %&gt;%
  GC_scaleBar(
    title = "1kb",
    scaleBarUnit = 1000,
    cluster = NULL,
    x = 0,
    y = 50,
    labelStyle = list(
      labelPosition =  "left",
      fontSize = "10px",
      fontFamily = "sans-serif",
      fill = "red", # Text color
      cursor = "default"
      # Any other CSS style for the label
    ),
    textPadding = -2,
    scaleBarLineStyle = list(
      stroke = "black",
      strokeWidth = 1
      # Any other CSS style for the line
    ),
    scaleBarTickStyle = list(
      stroke = "black",
      strokeWidth = 1
      # Any other CSS style for the tick
    )
  )

</code></pre>

<hr>
<h2 id='GC_sequence'>Update Sequence Display of a GC Chart Cluster</h2><span id='topic+GC_sequence'></span>

<h3>Description</h3>

<p>Modify the sequence display and break markers of specified clusters within a GC chart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_sequence(
  GC_chart,
  show = TRUE,
  cluster = NULL,
  y = 50,
  sequenceStyle = list(),
  markerStyle = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_sequence_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A GC chart object.</p>
</td></tr>
<tr><td><code id="GC_sequence_+3A_show">show</code></td>
<td>
<p>Logical, whether to display the sequence (default is TRUE).</p>
</td></tr>
<tr><td><code id="GC_sequence_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector specifying clusters to update.</p>
</td></tr>
<tr><td><code id="GC_sequence_+3A_y">y</code></td>
<td>
<p>Vertical position of the sequence line (default is 50).</p>
</td></tr>
<tr><td><code id="GC_sequence_+3A_sequencestyle">sequenceStyle</code></td>
<td>
<p>A list of styling options for the sequence line.</p>
</td></tr>
<tr><td><code id="GC_sequence_+3A_markerstyle">markerStyle</code></td>
<td>
<p>A list of styling options for the sequence break markers.</p>
</td></tr>
<tr><td><code id="GC_sequence_+3A_...">...</code></td>
<td>
<p>Additional customization arguments for sequence display.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function allows customization of the sequence line and break markers in a GC chart.
It offers options to adjust the sequence line ('sequenceStyle') and break markers ('markerStyle').
The 'y' parameter can be used to set the vertical position of the sequence.
</p>


<h3>Value</h3>

<p>An updated GC chart with modified sequence display settings.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 90, 130, 170, 210),
  end = c(40, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 2, 2, 2)
)

# Basic usage
GC_chart(genes_data, cluster ="cluster", group = "group", height = "200px") %&gt;%
GC_labels("name") %&gt;%
GC_sequence(show = TRUE, y = 50, cluster = NULL)

# Customize sequence and marker styles
GC_chart(genes_data, cluster="cluster", group = "group", height = "200px") %&gt;%
  GC_scale(hidden = TRUE, scale_breaks = TRUE) %&gt;%
  GC_sequence(
    start = NULL,
    end = NULL,
    sequenceStyle = list(
      stroke = "blue",
      strokeWidth = 1
      # Any other CSS style
    ),
    markerStyle = list(
      stroke = "blue",
      strokeWidth = 1,
      gap = 3,
      tiltAmount = 5
      # Any other CSS style
    )
  ) %&gt;%
  GC_legend(FALSE)

</code></pre>

<hr>
<h2 id='GC_title'>Add Title to a GC_chart</h2><span id='topic+GC_title'></span>

<h3>Description</h3>

<p>Modify the cluster title and subtitle of specified clusters within a GC chart and adjust
the display settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_title(
  GC_chart,
  title = NULL,
  subtitle = NULL,
  style = list(),
  subtitleFont = list(),
  titleFont = list(),
  show = TRUE,
  height = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_title_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A GC chart object.</p>
</td></tr>
<tr><td><code id="GC_title_+3A_title">title</code></td>
<td>
<p>Character vector. Titles to set for the clusters.</p>
</td></tr>
<tr><td><code id="GC_title_+3A_subtitle">subtitle</code></td>
<td>
<p>Character vector. Subtitles to set for the clusters.</p>
</td></tr>
<tr><td><code id="GC_title_+3A_style">style</code></td>
<td>
<p>A list of CSS styles to be applied to the chart container. Each
element of the list should be a valid CSS property-value pair. For example,
list(backgroundColor = &quot;white&quot;, border = &quot;2px solid black&quot;).
Default is an empty list.</p>
</td></tr>
<tr><td><code id="GC_title_+3A_subtitlefont">subtitleFont</code></td>
<td>
<p>List. Settings for the subtitle font.</p>
</td></tr>
<tr><td><code id="GC_title_+3A_titlefont">titleFont</code></td>
<td>
<p>List. Settings for the title font.</p>
</td></tr>
<tr><td><code id="GC_title_+3A_show">show</code></td>
<td>
<p>Logical. Whether to display the title. Default is TRUE.</p>
</td></tr>
<tr><td><code id="GC_title_+3A_height">height</code></td>
<td>
<p>Character. Height for the title (e.g., &quot;50px&quot;).</p>
</td></tr>
<tr><td><code id="GC_title_+3A_...">...</code></td>
<td>
<p>Additional customization arguments for title and subtitle.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated GC chart with new title settings.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 50, 90, 130, 170, 210),
  end = c(40, 80, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 2', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 1, 2, 2, 2)
)

# Basic usage
GC_chart(genes_data, cluster ="cluster", group = "group", height = "400px") %&gt;%
GC_labels("name") %&gt;%
GC_title(
  title = "Cluster 1 Data",
  subtitle = "Detailed View",
  show = TRUE
)

# Customizing title style
GC_chart(genes_data, cluster = "cluster", group = "group", height = "500px") %&gt;%
  GC_labels("name") %&gt;%
  GC_title(
    title = "Cluster 1 Data",
    subtitle = "Detailed View",
    show = TRUE,
    height = "50px",
    cluster = 1,
    x = 0,
    y = 25, # height / 2
    align = "center",
    spacing = 20,
    style = list(
      backgroundColor = "#0000"
      # Any other CSS styles
    ),
    titleFont = list(
      fontSize = "16px",
      fontStyle = "normal",
      fontWeight = "bold",
      textDecoration = "normal",
      fontFamily = "sans-serif",
      cursor = "default",
      fill = "black"
      # Any other CSS styles
    ),
    subtitleFont = list(
      fontSize = "14px",
      fontStyle = "normal",
      fontWeight = "bold",
      textDecoration = "normal",
      fontFamily = "sans-serif",
      cursor = "default",
      fill = "black"
      # Any other CSS styles
    )
  )

</code></pre>

<hr>
<h2 id='GC_tooltip'>Set Tooltip for a Gene Chart</h2><span id='topic+GC_tooltip'></span>

<h3>Description</h3>

<p>This function configures the tooltip for a gene chart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_tooltip(
  GC_chart,
  formatter = "&lt;b&gt;Start:&lt;/b&gt; {start}&lt;br&gt;&lt;b&gt;End:&lt;/b&gt; {end}",
  show = TRUE,
  cluster = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_tooltip_+3A_gc_chart">GC_chart</code></td>
<td>
<p>The gene chart object to be modified.</p>
</td></tr>
<tr><td><code id="GC_tooltip_+3A_formatter">formatter</code></td>
<td>
<p>A character string defining the HTML content of the tooltip. It can
include placeholders like <code>{start}</code> and <code>{end}</code> which will be replaced by actual
data values. The default value shows start and end data.</p>
</td></tr>
<tr><td><code id="GC_tooltip_+3A_show">show</code></td>
<td>
<p>Logical, whether to display the tooltip or not.</p>
</td></tr>
<tr><td><code id="GC_tooltip_+3A_cluster">cluster</code></td>
<td>
<p>Optional; used to specify which clusters in the chart should have tooltips.</p>
</td></tr>
<tr><td><code id="GC_tooltip_+3A_...">...</code></td>
<td>
<p>Additional arguments that can be used to further customize the tooltip.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the gene chart object with the tooltip configured.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set tooltip
genes_data &lt;- data.frame(
  start = c(10, 90, 130, 170, 210),
  end = c(40, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 2', 'Gene 3', 'Gene 4', 'Gene 5'),
  group = c('A', 'B', 'B', 'A', 'C')
)

# Add tooltips to the gene chart
GC_chart(genes_data, group = "group", height = "200px") %&gt;%
GC_tooltip(formatter = " &lt;b&gt;Start:&lt;/b&gt; {start}&lt;br&gt;&lt;b&gt;end:&lt;/b&gt; {end}")

</code></pre>

<hr>
<h2 id='GC_trackMouse'>Track Mouse Movement in a GC_chart</h2><span id='topic+GC_trackMouse'></span>

<h3>Description</h3>

<p>This function enables or disables mouse tracking on specified clusters within
a GC chart. When enabled, the x and y coordinates of the mouse are displayed
which can be used to place annotations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_trackMouse(GC_chart, show = TRUE, cluster = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_trackMouse_+3A_gc_chart">GC_chart</code></td>
<td>
<p>A GC chart object to which the annotations will be added.</p>
</td></tr>
<tr><td><code id="GC_trackMouse_+3A_show">show</code></td>
<td>
<p>Logical, specifies whether to track the mouse or not.</p>
</td></tr>
<tr><td><code id="GC_trackMouse_+3A_cluster">cluster</code></td>
<td>
<p>Numeric or character vector specifying the clusters to which
annotations should be added.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated GC chart object with mouse tracking settings.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GC_annotation">GC_annotation</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genes_data &lt;- data.frame(
  start = c(10, 50, 90, 130, 170, 210),
  end = c(40, 80, 120, 160, 200, 240),
  name = c('Gene 1', 'Gene 2', 'Gene 3', 'Gene 4', 'Gene 5', 'Gene 6'),
  group = c('A', 'A', 'B', 'B', 'A', 'C'),
  cluster = c(1, 1, 1, 2, 2, 2)
)

# Enable mouse tracking
GC_chart(genes_data, cluster = "cluster", group = "group", height = "220px") %&gt;%
GC_trackMouse()

</code></pre>

<hr>
<h2 id='GC_transcript'>Modify Transcript Characteristics within a Chart</h2><span id='topic+GC_transcript'></span>

<h3>Description</h3>

<p>This function updates the chart with specific characteristics for transcripts
based on the given parameters. It can show/hide transcripts, apply a color
scheme, assign custom colors, filter by cluster, and accept additional
options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_transcript(
  GC_chart,
  transcript = NULL,
  type = NULL,
  strand = NULL,
  group = NULL,
  selection = NULL,
  show = TRUE,
  colorScheme = NULL,
  customColors = NULL,
  styleExons = list(),
  styleIntrons = list(),
  styleUTRs = list(),
  itemStyleExons = list(),
  itemStyleIntrons = list(),
  itemStyleUTRs = list(),
  labelOptions = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_transcript_+3A_gc_chart">GC_chart</code></td>
<td>
<p>The chart object to be modified.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_transcript">transcript</code></td>
<td>
<p>Optional column name used for clustering transcript data.
Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_type">type</code></td>
<td>
<p>Column name identifying the feature type (&quot;UTR&quot;,
&quot;exon&quot;). Default is NULL.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_strand">strand</code></td>
<td>
<p>Optional column name indicating strand orientation. Acceptable
values include 1, 'forward', 'sense', or '+' to represent the forward
strand, and -1, 0, 'reverse', 'antisense', &quot;complement&quot; or '-' to represent
the reverse strand. Default is NULL, meaning strand information is not
used.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_group">group</code></td>
<td>
<p>Optional column name used for transcript grouping to influence
color aesthetics.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_selection">selection</code></td>
<td>
<p>Numeric or character, the specific transcript to filter
transcripts by.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_show">show</code></td>
<td>
<p>Logical, whether to show the transcripts or not.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_colorscheme">colorScheme</code></td>
<td>
<p>Character or NULL, the name of the color scheme to use.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_customcolors">customColors</code></td>
<td>
<p>List or NULL, custom colors to apply to the genes.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_styleexons">styleExons</code></td>
<td>
<p>List, styles to apply to exons in the chart.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_styleintrons">styleIntrons</code></td>
<td>
<p>List, styles to apply to introns in the chart.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_styleutrs">styleUTRs</code></td>
<td>
<p>List, styles to apply to UTRs in the chart.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_itemstyleexons">itemStyleExons</code></td>
<td>
<p>List, a list of styles to apply to individual exons in
the chart.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_itemstyleintrons">itemStyleIntrons</code></td>
<td>
<p>List, a list of styles to apply to individual introns
in the chart.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_itemstyleutrs">itemStyleUTRs</code></td>
<td>
<p>List, a list of styles to apply to individual UTRs in the
chart.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_labeloptions">labelOptions</code></td>
<td>
<p>List, options for styling labels such as font size, color,
and position.</p>
</td></tr>
<tr><td><code id="GC_transcript_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the gene options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the modified gene chart object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>transcript_data &lt;- data.frame(
  transcript = c("transcript1", "transcript1", "transcript1", "transcript1",
                 "transcript2", "transcript2", "transcript2"),
  type = c("5_utr", "exon", "exon", "3_utr",
           "5_utr", "exon", "3_utr"),
  start = c(1, 101, 201, 301,
            1, 101, 301),
  end = c(50, 150, 250, 350,
          50, 150, 350),
  strand = rep("forward", 7)
)

# All default transcript settings
GC_chart(
  transcript_data,
  start = "start",
  end = "end",
  height = "200px"
) %&gt;%
  GC_transcript(
    transcript = "transcript",
    strand = "strand",
    type = "type",
    group = NULL,
    show = TRUE,
    selection = NULL,
    colorScheme = NULL,
    customColors = NULL,
    styleExons = list(
      show = TRUE,
      strokeWidth = 0,
      cursor = "default",
      marker = "box",
      markerSize = "medium",
      arrowheadWidth = NULL,
      arrowheadHeight = NULL,
      markerHeight = NULL,
      cornerRadius = NULL
      # Any other CSS style
    ),
    styleIntrons = list(
      show = TRUE,
      strokeWidth = 1,
      fill = "none",
      cursor = "default",
      marker = "intron",
      markerSize = "medium",
      arrowheadWidth = NULL,
      arrowheadHeight = NULL,
      markerHeight = NULL,
      cornerRadius = NULL
      # Any other CSS style
    ),
    styleUTRs = list(
      show = TRUE,
      fontSize = "10px",
      fontStyle = "normal",
      fontFamily = "sans-serif",
      cursor = "default",
      color = "black",
      fill = "#FFF",
      strokeWidth = 1,
      marker = "box",
      markerSize = "medium",
      arrowheadWidth = NULL,
      arrowheadHeight = NULL,
      markerHeight = NULL,
      cornerRadius = NULL
      # Any other CSS style
    ),
    labelOptions = list(
      show = TRUE,
      xOffset = 2,
      yOffset = 0,
      fontSize = "12px",
      fontStyle = "normal",
      fontWeight = "normal",
      fontFamily = "sans-serif",
      cursor = "default",
      color = "black"
    ),
    itemStyleExons = list(),
    itemStyleIntrons = list(),
    itemStyleUTRs = list()
  )
# Change the appearance of a specific intron
GC_chart(transcript_data,
         start = "start",
         end = "end",
         height = "200px"
) %&gt;%
  GC_transcript(
    transcript = "transcript",
    type = "type",
    selection = 2,
    itemStyleExons = list(list(index = 0, fill = "red")
    )
  )
</code></pre>

<hr>
<h2 id='genbank_to_fasta'>Convert GenBank to FASTA Format</h2><span id='topic+genbank_to_fasta'></span>

<h3>Description</h3>

<p>This function reads a GenBank file, extracts the sequence and
writes it to a new file in FASTA format. It parses the DEFINITION and VERSION
for the header and sequences from the ORIGIN section.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genbank_to_fasta(path, output_dir = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genbank_to_fasta_+3A_path">path</code></td>
<td>
<p>Path to the GenBank file.</p>
</td></tr>
<tr><td><code id="genbank_to_fasta_+3A_output_dir">output_dir</code></td>
<td>
<p>Optional path for the output FASTA file. If NULL, the
output is saved in the same directory as the input file with the same base
name but with a .fasta extension.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None explicitly, but writes the FASTA formatted data to a file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Path to the example GenBank file in the package
gbk_file &lt;- system.file("extdata", "BGC0000001.gbk", package = "geneviewer")

# Convert the GenBank file to FASTA format
genbank_to_fasta(gbk_file)



</code></pre>

<hr>
<h2 id='geneviewer-shiny'>Shiny bindings for geneviewer</h2><span id='topic+geneviewer-shiny'></span><span id='topic+GC_chartOutput'></span><span id='topic+renderGC_chart'></span>

<h3>Description</h3>

<p>Output and render functions for using geneviewer within Shiny applications and
interactive Rmd documents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_chartOutput(outputId, width = "100%", height = "400px")

renderGC_chart(expr, env = parent.frame(), quoted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geneviewer-shiny_+3A_outputid">outputId</code></td>
<td>
<p>Output variable to read from.</p>
</td></tr>
<tr><td><code id="geneviewer-shiny_+3A_width">width</code></td>
<td>
<p>height Must be a valid CSS unit (like '100
or a number, which will be coerced to a string and have 'px' appended.</p>
</td></tr>
<tr><td><code id="geneviewer-shiny_+3A_height">height</code></td>
<td>
<p>Height of the output widget, must be a valid CSS unit (like
'100
have 'px' appended.</p>
</td></tr>
<tr><td><code id="geneviewer-shiny_+3A_expr">expr</code></td>
<td>
<p>An expression that generates a GC chart.</p>
</td></tr>
<tr><td><code id="geneviewer-shiny_+3A_env">env</code></td>
<td>
<p>The environment in which to evaluate expr.</p>
</td></tr>
<tr><td><code id="geneviewer-shiny_+3A_quoted">quoted</code></td>
<td>
<p>Is expr a quoted expression (with quote())? This is useful if
you want to save an expression in a variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>GC_chart widget that can be placed in the UI.
</p>


<h3>See Also</h3>

<p>[GC_chart()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  library(shiny)
  library(geneviewer)

  ui &lt;- fluidPage(
    titlePanel("Omphalotin Gene Cluster Visualization"),
    mainPanel(
      GC_chartOutput("gcChart", width = "100%", height = "500px")
    )
  )

  server &lt;- function(input, output) {
    output$gcChart &lt;- renderGC_chart({
      GC_chart(
        ophA_clusters, # Ensure 'ophA_clusters' data is defined or available
        cluster = "cluster",
        group = "class"
      ) %&gt;%
      GC_title(title = c("&lt;i&gt;O. olearius&lt;/i&gt;", "&lt;i&gt;D. bispora&lt;/i&gt;")) %&gt;%
      GC_labels("name") %&gt;%
      GC_legend(position = "bottom") %&gt;%
      GC_scaleBar() %&gt;%
      GC_clusterLabel(title = "ophA")
    })
  }

  shinyApp(ui = ui, server = server)
}
</code></pre>

<hr>
<h2 id='get_introns'>Calculate Intron Positions Based on Exons</h2><span id='topic+get_introns'></span>

<h3>Description</h3>

<p>This function takes a dataset containing transcript information and calculates
the start and end positions of introns based on the positions of exons. It
generates a combined dataframe including both the original exons and the
calculated introns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_introns(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_introns_+3A_data">data</code></td>
<td>
<p>A dataframe containing 'start' and 'end' positions of transcripts.
Optional columns are 'type' (exon or UTR), 'transcript' (containing unique
transcript IDs), and 'strand' (indicating the direction, forward or reverse,
of each transcript).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with the original exons and the calculated introns, sorted
by transcript and start position. Each row includes the 'start', 'end',
'type' (exon, UTR or intron), 'transcript', and 'strand' for each segment.
</p>

<hr>
<h2 id='hs_dystrophin_transcripts'>Human Dystrophin Transcripts Data</h2><span id='topic+hs_dystrophin_transcripts'></span>

<h3>Description</h3>

<p>This dataset contains the Exon positions on human Dystrophin transcripts
'Dp427p2', 'Dp260-2', 'Dp140', 'Dp116', and 'Dp71'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hs_dystrophin_transcripts
</code></pre>


<h3>Format</h3>

<p>A data frame with 202 rows and 5 columns:
</p>

<dl>
<dt>transcript</dt><dd><p>Transcript names. A character vector representing
the names of the transcripts.</p>
</dd>
<dt>type</dt><dd><p>Transcript type.</p>
</dd>
<dt>start</dt><dd><p>Start positions. An integer vector showing the starting
positions of each Exon.</p>
</dd>
<dt>end</dt><dd><p>End positions. An integer vector showing the ending
positions of each Exon.</p>
</dd>
<dt>length</dt><dd><p>Transcript lengths. An integer vector
indicating the length of each transcript.</p>
</dd>
</dl>


<hr>
<h2 id='human_hox_genes'>Human HOX Gene Cluster Data</h2><span id='topic+human_hox_genes'></span>

<h3>Description</h3>

<p>This dataset represents the positions and clusters of various human HOX
genes. HOX genes are a group of related genes that control the body plan of
an embryo along the head-tail axis. The dataset includes genes from HOXA,
HOXB, HOXC, and HOXD clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>human_hox_genes
</code></pre>


<h3>Format</h3>

<p>A data frame with 39 rows and 6 columns:
</p>

<dl>
<dt>symbol</dt><dd><p>HGNC symbols for the HOX genes. A character vector.</p>
</dd>
<dt>chromosome_name</dt><dd><p>Chromosome number where each gene is located. A numeric vector.</p>
</dd>
<dt>start</dt><dd><p>Start positions of the genes on the chromosome. A numeric vector.</p>
</dd>
<dt>end</dt><dd><p>End positions of the genes on the chromosome. A numeric vector.</p>
</dd>
<dt>strand</dt><dd><p>-1 for reverse and 1 for forward.</p>
</dd>
<dt>cluster</dt><dd><p>Cluster identification as a character vector,
specifying the HOX gene cluster (HOXA, HOXB, HOXC, HOXD).</p>
</dd>
<dt>name</dt><dd><p>Simplified names for the HOX genes. A character vector.</p>
</dd>
</dl>


<hr>
<h2 id='mummer_alignment'>Perform Sequence Alignment Using MUMmer</h2><span id='topic+mummer_alignment'></span>

<h3>Description</h3>

<p>This function orchestrates the alignment of sequences in a specified
directory using MUMmer, a tool for aligning large DNA or protein sequences.
It can handle GenBank and FASTA file formats. and performs checks to ensure
necessary files are present.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mummer_alignment(
  path,
  cluster = NULL,
  maptype = "many-to-many",
  seqtype = "protein",
  mummer_options = "",
  filter_options = "",
  remove_files = TRUE,
  output_dir = tempdir()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mummer_alignment_+3A_path">path</code></td>
<td>
<p>The directory containing the sequence files.</p>
</td></tr>
<tr><td><code id="mummer_alignment_+3A_cluster">cluster</code></td>
<td>
<p>Optional vector of cluster names to consider for alignment. If
NULL, clusters are inferred from file names. The order of names determines
the alignment sequence.</p>
</td></tr>
<tr><td><code id="mummer_alignment_+3A_maptype">maptype</code></td>
<td>
<p>The type of mapping to perform; &quot;many-to-many&quot; or
&quot;one-to-one&quot;. &quot;many-to-many&quot; allows for multiple matches between clusters,
&quot;one-to-one&quot; restricts alignments to unique matches between a pair.</p>
</td></tr>
<tr><td><code id="mummer_alignment_+3A_seqtype">seqtype</code></td>
<td>
<p>The type of sequences, either &quot;protein&quot; or &quot;nucleotide&quot;.</p>
</td></tr>
<tr><td><code id="mummer_alignment_+3A_mummer_options">mummer_options</code></td>
<td>
<p>Additional command line options for MUMmer. To see all
available options, you can run 'nucmer &ndash;help' or 'promer &ndash;help' in the
terminal depending on whether you are aligning nucleotide or protein
sequences.</p>
</td></tr>
<tr><td><code id="mummer_alignment_+3A_filter_options">filter_options</code></td>
<td>
<p>Additional options for filtering MUMmer results. To
view all filtering options, run 'delta-filter &ndash;help' in the terminal.</p>
</td></tr>
<tr><td><code id="mummer_alignment_+3A_remove_files">remove_files</code></td>
<td>
<p>Logical indicating whether to remove intermediate files
generated during the process, defaults to TRUE.</p>
</td></tr>
<tr><td><code id="mummer_alignment_+3A_output_dir">output_dir</code></td>
<td>
<p>Optional directory for output files; defaults to
<code>tempdir()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame combining all alignment results, or NULL if errors occur
during processing.
</p>


<h3>References</h3>

<p>Kurtz S, Phillippy A, Delcher AL, Smoot M, Shumway M, Antonescu
C, Salzberg SL (2004). Versatile and open software for comparing large
genomes. Genome Biology, 5(R12).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Basic alignment with default options
mummer_alignment(
  path = "/path/to/sequences",
  maptype = "many-to-many",
  seqtype = "protein"
)

# Alignment with specific MUMmer options
mummer_alignment(
  path = "/path/to/sequences",
  maptype = "one-to-one",
  seqtype = "protein",
  mummer_options = "--maxgap=500 --mincluster=100",
  filter_options = "-i 90"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='ophA_clusters'>ophA Gene Cluster from Omphalotus olearius</h2><span id='topic+ophA_clusters'></span>

<h3>Description</h3>

<p>This dataset represents the ophA gene cluster from Omphalotus olearius and
Dendrothele bispora, involved in the production of omphalotin, a cyclic
N-methylated peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ophA_clusters
</code></pre>


<h3>Format</h3>

<p>A data frame with 17 rows and 5 columns:
</p>

<dl>
<dt>name</dt><dd><p>Gene names. A character vector.</p>
</dd>
<dt>start</dt><dd><p>Start positions of the genes. An integer vector.</p>
</dd>
<dt>end</dt><dd><p>End positions of the genes. An integer vector.</p>
</dd>
<dt>class</dt><dd><p>Classifications of the genes. A character vector indicating
the type of protein or function associated with each gene.</p>
</dd>
<dt>cluster</dt><dd><p>Cluster identification as a character vector, specifying whether
the gene belongs to the ophA or dbophA gene cluster.</p>
</dd>
</dl>



<h3>References</h3>

<p>van der Velden NS et al. Autocatalytic backbone N-methylation in
a family of ribosomal peptide natural products. Nat Chem Biol. 2017
Aug;13(8):833-835. doi: 10.1038/nchembio.2393. Epub 2017 Jun 5. PMID:
28581484.
</p>

<hr>
<h2 id='protein_blast'>Perform Protein BLAST Analysis Within Specified Clusters</h2><span id='topic+protein_blast'></span>

<h3>Description</h3>

<p>This function conducts a BLAST analysis for protein sequences within
specified clusters. It generates all possible protein combinations between a
query cluster and other clusters, performs pairwise alignments, calculates
sequence identity and similarity, and filters results based on a minimum
identity threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>protein_blast(
  data,
  query,
  id = "protein_id",
  start = "start",
  end = "end",
  cluster = "cluster",
  genes = NULL,
  identity = 30,
  parallel = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="protein_blast_+3A_data">data</code></td>
<td>
<p>A dataframe or a character vector specifying the path to .gbk
files. When a character vector is provided, it is interpreted as file paths
to .gbk files which are then read and processed. The dataframe must contain
columns for unique protein identifiers, cluster identifiers, protein
sequences, and the start and end positions of each gene.</p>
</td></tr>
<tr><td><code id="protein_blast_+3A_query">query</code></td>
<td>
<p>The name of the query cluster to be used for BLAST comparisons.</p>
</td></tr>
<tr><td><code id="protein_blast_+3A_id">id</code></td>
<td>
<p>The name of the column that contains the gene identifiers. Defaults
to &quot;protein_id&quot;.</p>
</td></tr>
<tr><td><code id="protein_blast_+3A_start">start</code></td>
<td>
<p>The name of the column specifying the start positions of genes.
Defaults to &quot;start&quot;.</p>
</td></tr>
<tr><td><code id="protein_blast_+3A_end">end</code></td>
<td>
<p>The name of the column specifying the end positions of genes.
Defaults to &quot;end&quot;.</p>
</td></tr>
<tr><td><code id="protein_blast_+3A_cluster">cluster</code></td>
<td>
<p>The name of the column specifying the cluster names. Defaults
to &quot;cluster&quot;.</p>
</td></tr>
<tr><td><code id="protein_blast_+3A_genes">genes</code></td>
<td>
<p>An optional vector of gene identifiers to include in the
analysis. Defaults to NULL.</p>
</td></tr>
<tr><td><code id="protein_blast_+3A_identity">identity</code></td>
<td>
<p>Minimum identity threshold for BLAST hits to be considered
significant. Defaults to 30.</p>
</td></tr>
<tr><td><code id="protein_blast_+3A_parallel">parallel</code></td>
<td>
<p>Logical indicating whether to use parallel processing for
alignments. Defaults to TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified version of the input 'data' dataframe, including
additional columns for BLAST results (identity, similarity).
</p>


<h3>Note</h3>

<p>This function relies on the Biostrings and pwalign package for sequence
alignment and the dplyr package for data manipulation. Ensure these packages
are installed and loaded into your R session.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
path_to_folder &lt;- "path/to/gbk/folder/"
data_updated &lt;- protein_blast(
                         path_to_folder,
                         id = "protein_id",
                         query = "cluster A",
                         identity = 30
                         )

## End(Not run)

</code></pre>

<hr>
<h2 id='read_bed'>Read BED Files</h2><span id='topic+read_bed'></span>

<h3>Description</h3>

<p>This function reads BED files from a specified directory or file path and
combines them into a single data frame. BED files use 0-based coordinate starts,
while this function transforms the coordinates to 1-based during import.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_bed(path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_bed_+3A_path">path</code></td>
<td>
<p>A character string specifying the directory containing BED files
or the file path to a single BED file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can read multiple BED files from a directory or a
single BED file from a specified path. It adds a 'filename' column with the
name of the file, and combines the data frames from all files into one.
</p>


<h3>Value</h3>

<p>A data frame combining data from the BED files.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Read BED files from a directory
bed_data &lt;- read_bed("path/to/directory")

# Read a single BED file
bed_data &lt;- read_bed("path/to/file.bed")

## End(Not run)
</code></pre>

<hr>
<h2 id='read_fasta'>Read Protein Sequences from FASTA Files</h2><span id='topic+read_fasta'></span>

<h3>Description</h3>

<p>This function reads protein sequences from the specified FASTA file or all
FASTA files within a directory. It specifically looks for metadata in the
FASTA headers with key-value pairs separated by an equals sign '='. For
example, from the header '&gt;protein1 [gene=scnD] [protein=ScnD]', it extracts
'gene' as the key and 'scnD' as its value, and similarly for other key-value
pairs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_fasta(fasta_path, sequence = TRUE, keys = NULL, file_extension = "fasta")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_fasta_+3A_fasta_path">fasta_path</code></td>
<td>
<p>Path to the FASTA file or directory containing FASTA files.</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_sequence">sequence</code></td>
<td>
<p>Logical; if 'TRUE', the protein sequences are included in the
returned data frame.</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_keys">keys</code></td>
<td>
<p>An optional vector of strings representing specific keys within
the fasta header to retain in the final data frame. If 'NULL' (the default),
all keys within the specified feature are included.</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_file_extension">file_extension</code></td>
<td>
<p>Extension of the FASTA files to be read from the
directory (default is 'fasta').</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Biostrings package is required to run this function.
</p>


<h3>Value</h3>

<p>A data frame with columns for each piece of information extracted from
the FASTA headers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Read sequences from a single FASTA file
sequences_df &lt;- read_fasta("path/to/single_file.fasta")

# Read all sequences from a directory of FASTA files
sequences_df &lt;- read_fasta("path/to/directory/", file_extension = "fa")

# Read sequences and include the protein sequences in the output
sequences_df &lt;- read_fasta("path/to/directory/", sequence = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='read_gbk'>Read Data from GenBank Files</h2><span id='topic+read_gbk'></span>

<h3>Description</h3>

<p>This function reads data from a single GenBank file or directory with
GenBank files. It allows selective extraction of information by specifying
sections and features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_gbk(path, sections = NULL, features = NULL, origin = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_gbk_+3A_path">path</code></td>
<td>
<p>A string representing the file path to the target GenBank (.gbk)
file or directory.</p>
</td></tr>
<tr><td><code id="read_gbk_+3A_sections">sections</code></td>
<td>
<p>An optional vector of strings representing the names of
specific sections within the GenBank file to extract (e.g., &quot;LOCUS&quot;,
&quot;DEFINITION&quot;, &quot;ACCESSION&quot;, &quot;VERSION&quot;). If 'NULL' (the default), the
function extracts all available sections.</p>
</td></tr>
<tr><td><code id="read_gbk_+3A_features">features</code></td>
<td>
<p>An optional vector of strings indicating specific feature
types to extract from the FEATURES section of the GenBank file (e.g.,
&quot;CDS&quot;, &quot;gene&quot;, &quot;mRNA&quot;). If 'NULL' (the default), the function extracts all
feature types present in the FEATURES section.</p>
</td></tr>
<tr><td><code id="read_gbk_+3A_origin">origin</code></td>
<td>
<p>A boolean flag; when set to 'TRUE' (the default), the origin
sequence data is included in the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the contents of the specified sections and features
of the GenBank file. Each section and feature is returned as a separate
list element.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Path to example GenBank file in the package
genbank_file &lt;- system.file(
  "extdata",
  "BGC0000001.gbk",
  package = "geneviewer"
  )

# Read all data from the example GenBank file
gbk_data &lt;- read_gbk(genbank_file)

# Read only specific sections from the example GenBank file
gbk_data &lt;- read_gbk(genbank_file, sections = c("LOCUS", "DEFINITION"))

# Read specific features from the FEATURES section of the example GenBank file
gbk_data &lt;- read_gbk(genbank_file, features = c("gene", "CDS"))

# Read data without the origin sequence
gbk_data &lt;- read_gbk(genbank_file, origin = FALSE)

</code></pre>

<hr>
<h2 id='read_gff'>Read GFF Files</h2><span id='topic+read_gff'></span>

<h3>Description</h3>

<p>This function reads GFF files from a specified directory or file path and
combines them into a single data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_gff(path, fields = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_gff_+3A_path">path</code></td>
<td>
<p>A character string specifying the directory containing GFF files
or the file path to a single GFF file.</p>
</td></tr>
<tr><td><code id="read_gff_+3A_fields">fields</code></td>
<td>
<p>An optional vector of character strings specifying the fields
to extract from the GFF files.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can read multiple GFF files from a directory or a
single GFF file. It processes each file, extracts the specified fields (if
provided), adds a 'name' column with the filename, and combines the data
frames from all files into one.
</p>


<h3>Value</h3>

<p>A data frame containing the combined data from the GFF files.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Read GFF files from a directory
gff_data &lt;- read_gff("path/to/directory")

# Read a single GFF file
gff_data &lt;- read_gff("path/to/file.gff")

# Read specific fields from GFF files
gff_data &lt;- read_gff(
"path/to/directory",
fields = c("seqid", "start", "end", "attributes")
)

## End(Not run)
</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>magrittr</dt><dd><p><code><a href="magrittr.html#topic+pipe">%&gt;%</a></code></p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
