<!DOCTYPE html><html><head><title>Help for package geneExpressionFromGEO</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {geneExpressionFromGEO}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dec_two'><p>Function that returns numeric values with 2 decimal numbers.</p></a></li>
<li><a href='#getGeneExpressionFromGEO'><p>Function that reads in the GEO code of a dataset, and returns the gene expression dataframe.</p></a></li>
<li><a href='#readUrl'><p>Function that reads in a URL to check and verifies if it exists (function written by Janko Thyson @rappster and taken from https://stackoverflow.com/a/12195574 )</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Title:</td>
<td>Easily Downloads a Gene Expression Dataset from a GEO Code and
Retrieves the Gene Symbols of Its Probesets</td>
</tr>
<tr>
<td>Version:</td>
<td>0.9</td>
</tr>
<tr>
<td>Description:</td>
<td>A function that reads in the GEO code of a gene expression dataset, retrieves its data from GEO, (optionally) retrieves the gene symbols of the dataset, and returns a simple dataframe table containing all the data. Platforms available: GPL11532, GPL23126, GPL6244, GPL8300, GPL80, GPL96, GPL570, GPL571, GPL20115, GPL1293,  GPL6102, GPL6104, GPL6883, GPL6884, GPL13497, GPL14550, GPL17077, GPL6480. GEO: Gene Expression Omnibus. ID: identifier code. The GEO datasets are downloaded from the URL <a href="https://ftp.ncbi.nlm.nih.gov/geo/series/">https://ftp.ncbi.nlm.nih.gov/geo/series/</a>. More information can be found in the following manuscript: Davide Chicco, "geneExpressionFromGEO: an R package to facilitate data reading from Gene Expression Omnibus (GEO)". Microarray Data Analysis, Methods in Molecular Biology, volume 2401, chapter 12, pages 187-194, Springer Protocols, 2021, &lt;<a href="https://doi.org/10.1007%2F978-1-0716-1839-4_12">doi:10.1007/978-1-0716-1839-4_12</a>&gt;.</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/davidechicco/geneExpressionFromGEO/issues">https://github.com/davidechicco/geneExpressionFromGEO/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/davidechicco/geneExpressionFromGEO">https://github.com/davidechicco/geneExpressionFromGEO</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>BiocManager, xml2, Biobase, annotate, GEOquery, markdown,
R.utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-12-15 14:48:25 UTC; dave</td>
</tr>
<tr>
<td>Author:</td>
<td>Davide Chicco <a href="https://orcid.org/0000-0001-9655-7142"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Davide Chicco &lt;davidechicco@davidechicco.it&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-12-16 08:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='dec_two'>Function that returns numeric values with 2 decimal numbers.</h2><span id='topic+dec_two'></span>

<h3>Description</h3>

<p>Function that returns numeric values with 2 decimal numbers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dec_two(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dec_two_+3A_x">x</code></td>
<td>
<p>input numeric value with N decimal numbers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric value with 2 decimal numbers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>aaa &lt;- dec_two(8.31232)
</code></pre>

<hr>
<h2 id='getGeneExpressionFromGEO'>Function that reads in the GEO code of a dataset, and returns the gene expression dataframe.</h2><span id='topic+getGeneExpressionFromGEO'></span>

<h3>Description</h3>

<p>Function that reads in the GEO code of a dataset, and returns the gene expression dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGeneExpressionFromGEO(datasetGeoCode, retrieveGeneSymbols, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGeneExpressionFromGEO_+3A_datasetgeocode">datasetGeoCode</code></td>
<td>
<p>the GEO code of a dataset.</p>
</td></tr>
<tr><td><code id="getGeneExpressionFromGEO_+3A_retrievegenesymbols">retrieveGeneSymbols</code></td>
<td>
<p>a boolean flag stating if the function should retrieve the gene symbols or not.</p>
</td></tr>
<tr><td><code id="getGeneExpressionFromGEO_+3A_verbose">verbose</code></td>
<td>
<p>a boolean flag stating if helping messages should be printed or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a gene expression dataset.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geneExpressionDF1 &lt;- getGeneExpressionFromGEO("GSE3268", FALSE, FALSE)
</code></pre>

<hr>
<h2 id='readUrl'>Function that reads in a URL to check and verifies if it exists (function written by Janko Thyson @rappster and taken from https://stackoverflow.com/a/12195574 )</h2><span id='topic+readUrl'></span>

<h3>Description</h3>

<p>Function that reads in a URL to check and verifies if it exists (function written by Janko Thyson @rappster and taken from https://stackoverflow.com/a/12195574 )
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readUrl(url)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readUrl_+3A_url">url</code></td>
<td>
<p>the URL of a webpage</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output of a webpage verification check
</p>


<h3>Examples</h3>

<pre><code class='language-R'>y &lt;- readUrl("http://stat.ethz.ch/R-manual/R-devel/library/base/html/connections.html")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
