<!DOCTYPE html><html lang="en"><head><title>Help for package geneSLOPE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {geneSLOPE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#clump_snps'><p>Clumping procedure for SLOPE</p></a></li>
<li><a href='#clumpingResult'><p>clumpingResult class</p></a></li>
<li><a href='#create_lambda'><p>Lambda sequences for SLOPE</p></a></li>
<li><a href='#geneSLOPE'><p>Genome-Wide Association Study with SLOPE</p></a></li>
<li><a href='#gui_geneSLOPE'><p>GUI for GWAS with SLOPE</p></a></li>
<li><a href='#identify_clump'><p>identify_clump</p></a></li>
<li><a href='#identify_clump.clumpingResult'><p>Identify clump number in <code>clumpingResult</code> class plot</p></a></li>
<li><a href='#identify_clump.selectionResult'><p>Identify clump number in <code>selectionResult</code> class plot</p></a></li>
<li><a href='#phenotypeData'><p>phenotypeData class</p></a></li>
<li><a href='#plot.clumpingResult'><p>Plot <code>clumpingResult</code> class object</p></a></li>
<li><a href='#plot.selectionResult'><p>Plot selectionResult class object</p></a></li>
<li><a href='#print.clumpingResult'><p>Print <code>clumpingResult</code> class object</p></a></li>
<li><a href='#print.phenotypeData'><p>Print phenotypeData class object</p></a></li>
<li><a href='#print.screeningResult'><p>Print function for class screeningResult class</p></a></li>
<li><a href='#print.selectionResult'><p>Print selectionResult class object</p></a></li>
<li><a href='#read_phenotype'><p>Read phenotype from .fam file</p></a></li>
<li><a href='#screen_snps'><p>Reading and screening SNPs from .raw file and</p></a></li>
<li><a href='#screeningResult'><p>screeningResult class</p></a></li>
<li><a href='#select_snps'><p>GWAS with SLOPE</p></a></li>
<li><a href='#selectionResult'><p>selectionResult class</p></a></li>
<li><a href='#summary.clumpingResult'><p>Summary clumpingResult class object</p></a></li>
<li><a href='#summary.phenotypeData'><p>Summary phenotypeData class object</p></a></li>
<li><a href='#summary.screeningResult'><p>Summary function for class screeningResult</p></a></li>
<li><a href='#summary.selectionResult'><p>Summary selectionResult class object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genome-Wide Association Study with SLOPE</td>
</tr>
<tr>
<td>Version:</td>
<td>0.38.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-07</td>
</tr>
<tr>
<td>Description:</td>
<td>Genome-wide association study (GWAS) performed with SLOPE,
    short for Sorted L-One Penalized Estimation, a
    method for estimating the vector of coefficients in a linear model.
    In the first step of GWAS, single nucleotide polymorphisms (SNPs) are 
    clumped according to their correlations and distances. 
    Then, SLOPE is performed on the data where each clump has
    one representative.
    Malgorzata Bogdan, Ewout van den Berg, Chiara Sabatti, Weijie Su 
    and Emmanuel Candes (2014) 
    "SLOPE - Adaptive Variable Selection via Convex Optimization" 
    &lt;<a href="https://doi.org/10.48550/arXiv.1407.3824">doi:10.48550/arXiv.1407.3824</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/psobczyk/geneSLOPE">https://github.com/psobczyk/geneSLOPE</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/psobczyk/geneSLOPE/issues">https://github.com/psobczyk/geneSLOPE/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.3), SLOPE</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, bigmemory, grid, utils, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>shiny, knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-14 07:46:51 UTC; piotr</td>
</tr>
<tr>
<td>Author:</td>
<td>Damian Brzyski [aut],
  Christine Peterson [aut],
  Emmanuel J. Candes [aut],
  Malgorzata Bogdan [aut],
  Chiara Sabatti [aut],
  Piotr Sobczyk [cre, aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Piotr Sobczyk &lt;pj.sobczyk@gmail.com&gt;</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-16 09:12:37 UTC</td>
</tr>
</table>
<hr>
<h2 id='clump_snps'>Clumping procedure for SLOPE</h2><span id='topic+clump_snps'></span>

<h3>Description</h3>

<p>Clumping procedure performed on SNPs, columns of matrix <code>X</code>, from
object of class <code><a href="#topic+screeningResult">screeningResult</a></code>,
which is an output of function <code><a href="#topic+screen_snps">screen_snps</a></code>.
SNPs are clustered based on their correlations. For details see package vignette.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clump_snps(screenResult, rho = 0.5, pValues = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clump_snps_+3A_screenresult">screenResult</code></td>
<td>
<p>object of class <code><a href="#topic+screeningResult">screeningResult</a></code>.</p>
</td></tr>
<tr><td><code id="clump_snps_+3A_rho">rho</code></td>
<td>
<p>numeric, minimal correlation between two SNPs to be assigned to one clump.</p>
</td></tr>
<tr><td><code id="clump_snps_+3A_pvalues">pValues</code></td>
<td>
<p>numeric vector, p-values for SNPs computed outside geneSLOPE,
eg. with EMMAX.</p>
</td></tr>
<tr><td><code id="clump_snps_+3A_verbose">verbose</code></td>
<td>
<p>logical, if TRUE (default) progress bar is shown.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code><a href="#topic+clumpingResult">clumpingResult</a></code>. 
See the class documentation for details.
</p>

<hr>
<h2 id='clumpingResult'>clumpingResult class</h2><span id='topic+clumpingResult'></span>

<h3>Description</h3>

<p>A result of procedure for snp clumping produced by <code><a href="#topic+clump_snps">clump_snps</a></code>
</p>


<h3>Details</h3>

<p>Always a named list of eleven elements
</p>

<ol>
<li> <p><code>X</code> numeric matrix, consists of one snp representative for each clump
</p>
</li>
<li> <p><code>y</code> numeric vector, phenotype
</p>
</li>
<li> <p><code>SNPnumber</code> numeric vector, which columns in SNP matrix <code>X_all</code>
are related to clumps representatives
</p>
</li>
<li> <p><code>SNPclumps</code> list of numeric vectors, which columns in SNP matrix
<code>X_all</code> are related to clump members
</p>
</li>
<li> <p><code>X_info</code> data.frame, mapping information about SNPs from .map file.
Copied from the result of screening procedure.
</p>
</li>
<li> <p><code>selectedSnpsNumbers</code> numeric vector, which rows of <code>X_info</code>
matrix are related to selected clump representatives
</p>
</li>
<li> <p><code>X_all</code> numeric matrix, all the snps that passed screening procedure
</p>
</li>
<li> <p><code>numberOfSnps</code> numeric, total number of SNPs before screening procedure
</p>
</li>
<li> <p><code>selectedSnpsNumbersScreening</code> numeric vector, which rows of <code>X_info</code>
data.frame are related to snps that passed screening
</p>
</li>
<li> <p><code>pVals</code> numeric vector, p-values from marginal tests for each snp
</p>
</li>
<li> <p><code>pValMax</code> numeric, p-value used in screening procedure
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+screeningResult">screeningResult</a></code> <code><a href="#topic+clump_snps">clump_snps</a></code>
</p>

<hr>
<h2 id='create_lambda'>Lambda sequences for SLOPE</h2><span id='topic+create_lambda'></span>

<h3>Description</h3>

<p>Computes <code class="reqn">\lambda</code> sequences for SLOPE according to several pre-defined methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_lambda(n, p, fdr = 0.2, method = c("bhq", "gaussian"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_lambda_+3A_n">n</code></td>
<td>
<p>number of observations</p>
</td></tr>
<tr><td><code id="create_lambda_+3A_p">p</code></td>
<td>
<p>number of variables</p>
</td></tr>
<tr><td><code id="create_lambda_+3A_fdr">fdr</code></td>
<td>
<p>target False Discovery Rate (FDR)</p>
</td></tr>
<tr><td><code id="create_lambda_+3A_method">method</code></td>
<td>
<p>method to use for computing <code class="reqn">\lambda</code> (see Details)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following methods for computing <code class="reqn">\lambda</code> are supported:
</p>

<ul>
<li> <p><code>bhq</code>: Computes sequence inspired by Benjamini-Hochberg (BHq)
procedure
</p>
</li>
<li> <p><code>gaussian</code>: Computes modified BHq sequence inspired by
Gaussian designs
</p>
</li></ul>


<hr>
<h2 id='geneSLOPE'>Genome-Wide Association Study with SLOPE</h2><span id='topic+geneSLOPE'></span>

<h3>Description</h3>

<p>Package geneSLOPE performes genome-wide association study (GWAS) with <span class="pkg">SLOPE</span>,
short for Sorted L-One Penalized Estimation. SLOPE is a
method for estimating the vector of coefficients in linear model. For details
about it see references.
</p>


<h3>Details</h3>

<p>GWAS is split into three steps:
</p>

<ul>
<li><p> In the first step data is read using <span class="pkg">bigmemory</span> package and immediatly
screened using marginal tests for each Single Nucleotide Polymorphism (<em>SNP</em>)
</p>
</li>
<li><p> SNPs are clumped based on their correlations
</p>
</li>
<li><p> SLOPE is performed on data where each clump has
one representative (therefore we ensure that variables in linear model
are not strongly correlated)
</p>
</li></ul>

<p>Version: 0.38.2
</p>


<h3>Author(s)</h3>


<p>Malgorzata Bogdan, Damian Brzyski, Emmanuel J. Candes, Christine Peterson, Chiara Sabatti, Piotr Sobczyk
</p>
<p>Maintainer: Piotr Sobczyk <a href="mailto:pj.sobczyk@gmail.com">pj.sobczyk@gmail.com</a>

</p>


<h3>References</h3>

<p><em>SLOPE &ndash; Adaptive Variable Selection via Convex Optimization</em>,
Malgorzata Bogdan, Ewout van den Berg, Chiara Sabatti,
Weijie Su and Emmanuel Candes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
famFile &lt;- system.file("extdata", "plinkPhenotypeExample.fam", package = "geneSLOPE")
mapFile &lt;- system.file("extdata", "plinkMapExample.map", package = "geneSLOPE")
snpsFile &lt;- system.file("extdata", "plinkDataExample.raw", package = "geneSLOPE")
phe &lt;- read_phenotype(filename = famFile)
screening.result &lt;- screen_snps(snpsFile, mapFile, phe, pValMax = 0.05, chunkSize = 1e2)
clumping.result &lt;- clump_snps(screening.result, rho = 0.3, verbose = TRUE)
slope.result &lt;- select_snps(clumping.result, fdr=0.1)



if(interactive()) { gui_geneSLOPE() }
</code></pre>

<hr>
<h2 id='gui_geneSLOPE'>GUI for GWAS with SLOPE</h2><span id='topic+gui_geneSLOPE'></span>

<h3>Description</h3>

<p>A graphical user interface for performing Genome-wide
Association Study with SLOPE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gui_geneSLOPE()
</code></pre>


<h3>Details</h3>

<p>requires installing <span class="pkg"><a href="shiny.html#topic+shiny">shiny</a></span> package
</p>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='identify_clump'>identify_clump</h2><span id='topic+identify_clump'></span>

<h3>Description</h3>

<p>identify_clump
</p>


<h3>Usage</h3>

<pre><code class='language-R'>identify_clump(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="identify_clump_+3A_x">x</code></td>
<td>
<p>appropriate class object</p>
</td></tr>
<tr><td><code id="identify_clump_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Enable interactive selection of snps in plot.
Return clump number.
</p>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='identify_clump.clumpingResult'>Identify clump number in <code><a href="#topic+clumpingResult">clumpingResult</a></code> class plot</h2><span id='topic+identify_clump.clumpingResult'></span>

<h3>Description</h3>

<p>Identify clump number in <code><a href="#topic+clumpingResult">clumpingResult</a></code> class plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clumpingResult'
identify_clump(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="identify_clump.clumpingResult_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+clumpingResult">clumpingResult</a></code> class object</p>
</td></tr>
<tr><td><code id="identify_clump.clumpingResult_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='identify_clump.selectionResult'>Identify clump number in <code><a href="#topic+selectionResult">selectionResult</a></code> class plot</h2><span id='topic+identify_clump.selectionResult'></span>

<h3>Description</h3>

<p>Identify clump number in <code><a href="#topic+selectionResult">selectionResult</a></code> class plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'selectionResult'
identify_clump(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="identify_clump.selectionResult_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+selectionResult">selectionResult</a></code> class object</p>
</td></tr>
<tr><td><code id="identify_clump.selectionResult_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='phenotypeData'>phenotypeData class</h2><span id='topic+phenotypeData'></span>

<h3>Description</h3>

<p>Phenotype data
</p>


<h3>Details</h3>

<p>Always a named list of two elements
</p>

<ol>
<li> <p><code>y</code> numeric vector, phenotype
</p>
</li>
<li> <p><code>yInfo</code> data.frame, additional information about observations
provied in .fam file
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+read_phenotype">read_phenotype</a></code>
</p>

<hr>
<h2 id='plot.clumpingResult'>Plot <code><a href="#topic+clumpingResult">clumpingResult</a></code> class object</h2><span id='topic+plot.clumpingResult'></span>

<h3>Description</h3>

<p>Plot <code><a href="#topic+clumpingResult">clumpingResult</a></code> class object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clumpingResult'
plot(x, chromosomeNumber = NULL, clumpNumber = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.clumpingResult_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+clumpingResult">clumpingResult</a></code> class object</p>
</td></tr>
<tr><td><code id="plot.clumpingResult_+3A_chromosomenumber">chromosomeNumber</code></td>
<td>
<p>optional parameter, only selected chromosome will be plotted</p>
</td></tr>
<tr><td><code id="plot.clumpingResult_+3A_clumpnumber">clumpNumber</code></td>
<td>
<p>optional parameter, only SNPs from selected clump will be plotted</p>
</td></tr>
<tr><td><code id="plot.clumpingResult_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='plot.selectionResult'>Plot selectionResult class object</h2><span id='topic+plot.selectionResult'></span>

<h3>Description</h3>

<p>Plot selectionResult class object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'selectionResult'
plot(x, chromosomeNumber = NULL, clumpNumber = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.selectionResult_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+selectionResult">selectionResult</a></code> class object</p>
</td></tr>
<tr><td><code id="plot.selectionResult_+3A_chromosomenumber">chromosomeNumber</code></td>
<td>
<p>optional parameter, only selected chromosome will be plotted</p>
</td></tr>
<tr><td><code id="plot.selectionResult_+3A_clumpnumber">clumpNumber</code></td>
<td>
<p>optional parameter, only SNPs from selected clump will be plotted</p>
</td></tr>
<tr><td><code id="plot.selectionResult_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='print.clumpingResult'>Print <code><a href="#topic+clumpingResult">clumpingResult</a></code> class object</h2><span id='topic+print.clumpingResult'></span>

<h3>Description</h3>

<p>Print <code><a href="#topic+clumpingResult">clumpingResult</a></code> class object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clumpingResult'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.clumpingResult_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+clumpingResult">clumpingResult</a></code> class object</p>
</td></tr>
<tr><td><code id="print.clumpingResult_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='print.phenotypeData'>Print phenotypeData class object</h2><span id='topic+print.phenotypeData'></span>

<h3>Description</h3>

<p>Print phenotypeData class object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phenotypeData'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.phenotypeData_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+phenotypeData">phenotypeData</a></code> class object</p>
</td></tr>
<tr><td><code id="print.phenotypeData_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='print.screeningResult'>Print function for class screeningResult class</h2><span id='topic+print.screeningResult'></span>

<h3>Description</h3>

<p>Print function for class screeningResult class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'screeningResult'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.screeningResult_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+screeningResult">screeningResult</a></code> class object</p>
</td></tr>
<tr><td><code id="print.screeningResult_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='print.selectionResult'>Print selectionResult class object</h2><span id='topic+print.selectionResult'></span>

<h3>Description</h3>

<p>Print selectionResult class object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'selectionResult'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.selectionResult_+3A_x">x</code></td>
<td>
<p>selectionResult class object</p>
</td></tr>
<tr><td><code id="print.selectionResult_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='read_phenotype'>Read phenotype from .fam file</h2><span id='topic+read_phenotype'></span>

<h3>Description</h3>

<p>Reading phenotype data from file. It is assumed, that data is given in .fam file.
In this format, first column is family id (FID), second is individual id (IID),
third is Paternal individual ID (PAT), fourth is  Maternal individual ID (MAT),
fifth is SEX and sixth and last is PHENOTYPE.
If file has only four columns, then it is assumed that PAT and MAT columns are missing.
If there is only one column, then it is assumed that only phenotype is provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_phenotype(filename, sep = " ", header = FALSE, stringAsFactors = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_phenotype_+3A_filename">filename</code></td>
<td>
<p>character, name of file with phenotype.</p>
</td></tr>
<tr><td><code id="read_phenotype_+3A_sep">sep</code></td>
<td>
<p>character, field seperator in file.</p>
</td></tr>
<tr><td><code id="read_phenotype_+3A_header">header</code></td>
<td>
<p>logical, does first row of file contain variables names.</p>
</td></tr>
<tr><td><code id="read_phenotype_+3A_stringasfactors">stringAsFactors</code></td>
<td>
<p>logical, should character vectors be converted to factors?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code><a href="#topic+phenotypeData">phenotypeData</a></code>. 
See the class documentation for details.
</p>

<hr>
<h2 id='screen_snps'>Reading and screening SNPs from .raw file and</h2><span id='topic+screen_snps'></span>

<h3>Description</h3>

<p>Reading .raw file that was previously exported from PLINK - see details.
Additional information about SNP mapping is read from .map file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>screen_snps(
  rawFile,
  mapFile = "",
  phenotype,
  pValMax = 0.05,
  chunkSize = 100,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="screen_snps_+3A_rawfile">rawFile</code></td>
<td>
<p>character, name of .raw file.</p>
</td></tr>
<tr><td><code id="screen_snps_+3A_mapfile">mapFile</code></td>
<td>
<p>character, name of .map file.</p>
</td></tr>
<tr><td><code id="screen_snps_+3A_phenotype">phenotype</code></td>
<td>
<p>numeric vector or an object of class <code><a href="#topic+phenotypeData">phenotypeData</a></code>.</p>
</td></tr>
<tr><td><code id="screen_snps_+3A_pvalmax">pValMax</code></td>
<td>
<p>numeric, p-value threshold value used for screening.</p>
</td></tr>
<tr><td><code id="screen_snps_+3A_chunksize">chunkSize</code></td>
<td>
<p>integer, number of snps that will be processed together.
The bigger the chunkSize is, the faster function works but
computer might run out of RAM.</p>
</td></tr>
<tr><td><code id="screen_snps_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE (default) information about progress is printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Exporting data from PLINK</strong>
To import data to R, it needs to be exported from
PLINK using the option &quot;&ndash;recodeAD&quot;
The PLINK command should therefore look like
<code>plink --file input --recodeAD --out output</code>.
For more information, please refer to:
<a href="https://zzz.bwh.harvard.edu/plink/dataman.shtml">https://zzz.bwh.harvard.edu/plink/dataman.shtml</a>
</p>


<h3>Value</h3>

<p>object of class <code><a href="#topic+screeningResult">screeningResult</a></code>.
</p>

<hr>
<h2 id='screeningResult'>screeningResult class</h2><span id='topic+screeningResult'></span>

<h3>Description</h3>

<p>A result of procedure for snp clumping produced by <code><a href="#topic+screen_snps">screen_snps</a></code>
</p>


<h3>Details</h3>

<p>Always a named list of eight elements
</p>

<ol>
<li> <p><code>X</code> numeric matrix, consists of snps that passed screening
</p>
</li>
<li> <p><code>y</code> numeric vector, phenotype
</p>
</li>
<li> <p><code>X_info</code> data.frame, SNP info from .map file
</p>
</li>
<li> <p><code>pVals</code> numeric vector, p-values from marginal tests for each snp
</p>
</li>
<li> <p><code>numberOfSnps</code> numeric, total number of SNPs in .raw file
</p>
</li>
<li> <p><code>selectedSnpsNumbers</code> numeric vector, which rows of <code>X_info</code>
data.frame are related to snps that passed screening
</p>
</li>
<li> <p><code>pValMax</code> numeric, p-value used in screening procedure
</p>
</li>
<li> <p><code>phenotypeInfo</code> data.frame, additional information about observations
provied in <code><a href="#topic+phenotypeData">phenotypeData</a></code> object
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+phenotypeData">phenotypeData</a></code> <code><a href="#topic+screen_snps">screen_snps</a></code>
</p>

<hr>
<h2 id='select_snps'>GWAS with SLOPE</h2><span id='topic+select_snps'></span>

<h3>Description</h3>

<p>Performs GWAS with SLOPE on given snp matrix and phenotype.
At first clumping procedure is performed. Highly correlated
(that is stronger than parameter <em>rho</em>) snps are clustered.
Then SLOPE is used on snp matrix which contains
one representative for each clump.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_snps(
  clumpingResult,
  fdr = 0.1,
  type = c("slope", "smt"),
  lambda = "gaussian",
  sigma = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="select_snps_+3A_clumpingresult">clumpingResult</code></td>
<td>
<p>object of class <code><a href="#topic+clumpingResult">clumpingResult</a></code>, 
clumpProcedure output.</p>
</td></tr>
<tr><td><code id="select_snps_+3A_fdr">fdr</code></td>
<td>
<p>numeric, False Discovery Rate for SLOPE.</p>
</td></tr>
<tr><td><code id="select_snps_+3A_type">type</code></td>
<td>
<p>method for snp selection. slope (default value) is SLOPE
on clump representatives, smt is the Benjamini-Hochberg procedure on
single marker test p-values for clump representatives.</p>
</td></tr>
<tr><td><code id="select_snps_+3A_lambda">lambda</code></td>
<td>
<p>lambda for SLOPE. See <code><a href="#topic+create_lambda">create_lambda</a></code>.</p>
</td></tr>
<tr><td><code id="select_snps_+3A_sigma">sigma</code></td>
<td>
<p>numeric, sigma for SLOPE.</p>
</td></tr>
<tr><td><code id="select_snps_+3A_verbose">verbose</code></td>
<td>
<p>logical, if TRUE progress bar is printed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code><a href="#topic+selectionResult">selectionResult</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
famFile &lt;- system.file("extdata", "plinkPhenotypeExample.fam", package = "geneSLOPE")
mapFile &lt;- system.file("extdata", "plinkMapExample.map", package = "geneSLOPE")
snpsFile &lt;- system.file("extdata", "plinkDataExample.raw", package = "geneSLOPE")
phe &lt;- read_phenotype(filename = famFile)
screening.result &lt;- screen_snps(snpsFile, mapFile, phe, pValMax = 0.05, chunkSize = 1e2)
clumping.result &lt;- clump_snps(screening.result, rho = 0.3, verbose = TRUE)
slope.result &lt;- select_snps(clumping.result, fdr=0.1)

</code></pre>

<hr>
<h2 id='selectionResult'>selectionResult class</h2><span id='topic+selectionResult'></span>

<h3>Description</h3>

<p>A result of applying SLOPE to matrix of SNPs obtained by
clumping produced. Result of function <code><a href="#topic+select_snps">select_snps</a></code>
</p>


<h3>Details</h3>

<p>Always a named list of eighteen elements
</p>

<ol>
<li> <p><code>X</code> numeric matrix, consists of one snp representative for each clump
selected by SLOPE
</p>
</li>
<li> <p><code>effects</code> numeric vector, coefficients in linear model build on
snps selected by SLOPE
</p>
</li>
<li> <p><code>R2</code> numeric, value of R-squared in linear model build on
snps selected by SLOPE
</p>
</li>
<li> <p><code>selectedSNPs</code> which columns in matrix <code>X_all</code>
are related to snps selected by SLOPE
</p>
</li>
<li> <p><code>y</code> selectedClumps list of numeric vectors, which columns in SNP matrix
<code>X_all</code> are related to clump members selected by SLOPE
</p>
</li>
<li> <p><code>lambda</code> numeric vector, lambda values used by SLOPE procedure
</p>
</li>
<li> <p><code>y</code> numeric vector, phenotype
</p>
</li>
<li> <p><code>clumpRepresentatives</code> numeric vector, which columns in SNP matrix <code>X_all</code>
are related to clumps representatives
</p>
</li>
<li> <p><code>clumps</code> list of numeric vectors, which columns in SNP matrix
<code>X_all</code> are related to clump members
</p>
</li>
<li> <p><code>X_info</code> data.frame, mapping information about SNPs from .map file.
Copied from the result of clumping procedure
</p>
</li>
<li> <p><code>X_clumps</code> numeric matrix, consists of one snp representative for each clump
</p>
</li>
<li> <p><code>X_all</code> numeric matrix, all the snps that passed screening procedure
</p>
</li>
<li> <p><code>selectedSnpsNumbers</code> numeric vector, which rows of <code>X_info</code>
data.frame are related to snps that were selected by SLOPE
</p>
</li>
<li> <p><code>clumpingRepresentativesNumbers</code> numeric vector, which rows of <code>X_info</code>
data.frame are related to snps that are clump represenatives
</p>
</li>
<li> <p><code>screenedSNPsNumbers</code> numeric vector, which rows of <code>X_info</code>
data.frame are related to snps that passed screening
</p>
</li>
<li> <p><code>numberOfSnps</code> numeric, total number of SNPs before screening procedure
</p>
</li>
<li> <p><code>pValMax</code> numeric, p-value used in screening procedure
</p>
</li>
<li> <p><code>fdr</code> numeric, false discovery rate used by <code>SLOPE</code>
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+screeningResult">screeningResult</a></code> <code><a href="#topic+clumpingResult">clumpingResult</a></code>
<code><a href="#topic+select_snps">select_snps</a></code> <code><a href="SLOPE.html#topic+SLOPE">SLOPE</a></code>
</p>

<hr>
<h2 id='summary.clumpingResult'>Summary clumpingResult class object</h2><span id='topic+summary.clumpingResult'></span>

<h3>Description</h3>

<p>Summary clumpingResult class object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clumpingResult'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.clumpingResult_+3A_object">object</code></td>
<td>
<p><code><a href="#topic+clumpingResult">clumpingResult</a></code> class object</p>
</td></tr>
<tr><td><code id="summary.clumpingResult_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='summary.phenotypeData'>Summary phenotypeData class object</h2><span id='topic+summary.phenotypeData'></span>

<h3>Description</h3>

<p>Summary phenotypeData class object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phenotypeData'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.phenotypeData_+3A_object">object</code></td>
<td>
<p><code><a href="#topic+phenotypeData">phenotypeData</a></code> class object</p>
</td></tr>
<tr><td><code id="summary.phenotypeData_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='summary.screeningResult'>Summary function for class screeningResult</h2><span id='topic+summary.screeningResult'></span>

<h3>Description</h3>

<p>Summary function for class screeningResult
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'screeningResult'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.screeningResult_+3A_object">object</code></td>
<td>
<p><code><a href="#topic+screeningResult">screeningResult</a></code> class object</p>
</td></tr>
<tr><td><code id="summary.screeningResult_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='summary.selectionResult'>Summary selectionResult class object</h2><span id='topic+summary.selectionResult'></span>

<h3>Description</h3>

<p>Summary selectionResult class object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'selectionResult'
summary(object, clumpNumber = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.selectionResult_+3A_object">object</code></td>
<td>
<p><code><a href="#topic+selectionResult">selectionResult</a></code> class object</p>
</td></tr>
<tr><td><code id="summary.selectionResult_+3A_clumpnumber">clumpNumber</code></td>
<td>
<p>number of clump to be summarized</p>
</td></tr>
<tr><td><code id="summary.selectionResult_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods. They are ignored in this function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
