<!DOCTYPE html><html lang="en"><head><title>Help for package GeneNMF</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GeneNMF}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dropMetaPrograms'><p>Drop meta-programs</p></a></li>
<li><a href='#findVariableFeatures_wfilters'><p>Find variable features</p></a></li>
<li><a href='#getDataMatrix'><p>Extract data matrix from Seurat object</p></a></li>
<li><a href='#getMetaPrograms'><p>Extract consensus gene programs (meta-programs)</p></a></li>
<li><a href='#getNMFgenes'><p>Get list of genes for each NMF program</p></a></li>
<li><a href='#multiNMF'><p>Run NMF on a list of Seurat objects</p></a></li>
<li><a href='#multiPCA'><p>Run PCA on a list of Seurat objects</p></a></li>
<li><a href='#plotMetaPrograms'><p>Visualizations for meta-programs</p></a></li>
<li><a href='#runGSEA'><p>Run Gene set enrichment analysis</p></a></li>
<li><a href='#runNMF'><p>Compute NMF as a low-dim embedding for Seurat</p></a></li>
<li><a href='#sampleObj'><p>Sample dataset to test GeneNMF installation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Non-Negative Matrix Factorization for Single-Cell Omics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.6.2</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of methods to extract gene programs from single-cell gene expression data using non-negative matrix factorization (NMF). 'GeneNMF' contains functions to directly interact with the 'Seurat' toolkit and derive interpretable gene program signatures.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>RcppML, Matrix, stats, methods, utils, Seurat (&ge; 4.3.0),
cluster, lsa, irlba, pheatmap, dendextend, viridis</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, fgsea, msigdbr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/carmonalab/GeneNMF">https://github.com/carmonalab/GeneNMF</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/carmonalab/GeneNMF/issues">https://github.com/carmonalab/GeneNMF/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-11 16:29:53 UTC; mass</td>
</tr>
<tr>
<td>Author:</td>
<td>Massimo Andreatta <a href="https://orcid.org/0000-0002-8036-2647"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Santiago Carmona <a href="https://orcid.org/0000-0002-2495-0671"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Massimo Andreatta &lt;massimo.andreatta@unil.ch&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-12 13:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='dropMetaPrograms'>Drop meta-programs</h2><span id='topic+dropMetaPrograms'></span>

<h3>Description</h3>

<p>Remove meta-programs from the GeneNMF results. This can be desirable if
one or more MPs are redundant or have poor metrics (e.g. low sample coverage
or low silhouette score).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dropMetaPrograms(mp.res, dropMP = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dropMetaPrograms_+3A_mp.res">mp.res</code></td>
<td>
<p>The meta-programs object generated by <code><a href="#topic+getMetaPrograms">getMetaPrograms</a></code></p>
</td></tr>
<tr><td><code id="dropMetaPrograms_+3A_dropmp">dropMP</code></td>
<td>
<p>A vector with the names of the MPs to be dropped</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input meta-program object, minus the dropped MPs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
data(sampleObj)
geneNMF_programs &lt;- multiNMF(list(sampleObj), k=5)
geneNMF_metaprograms &lt;- getMetaPrograms(geneNMF_programs, nMP=3)
geneNMF_metaprograms_filtered &lt;- dropMetaPrograms(geneNMF_metaprograms, dropMP = c("MP2")) 

</code></pre>

<hr>
<h2 id='findVariableFeatures_wfilters'>Find variable features</h2><span id='topic+findVariableFeatures_wfilters'></span>

<h3>Description</h3>

<p>Select highly variable genes (HVG) from an expression matrix. Genes from a blocklist
(e.g. cell cycling genes, mitochondrial genes) can be excluded from the list of
variable genes, as well as genes with very low or very high average expression
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findVariableFeatures_wfilters(
  obj,
  nfeatures = 2000,
  genesBlockList = NULL,
  min.exp = 0.01,
  max.exp = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findVariableFeatures_wfilters_+3A_obj">obj</code></td>
<td>
<p>A Seurat object containing an expression matrix</p>
</td></tr>
<tr><td><code id="findVariableFeatures_wfilters_+3A_nfeatures">nfeatures</code></td>
<td>
<p>Number of top HVG to be returned</p>
</td></tr>
<tr><td><code id="findVariableFeatures_wfilters_+3A_genesblocklist">genesBlockList</code></td>
<td>
<p>Optionally takes a vector or list of vectors of gene names.
These genes will be ignored for HVG detection. This is useful to mitigate effect
of genes associated with technical artifacts or batch effects
(e.g. mitochondrial, heat-shock response). If set to 'NULL' no genes will be excluded</p>
</td></tr>
<tr><td><code id="findVariableFeatures_wfilters_+3A_min.exp">min.exp</code></td>
<td>
<p>Minimum average normalized expression for HVG. If lower, the gene will be excluded</p>
</td></tr>
<tr><td><code id="findVariableFeatures_wfilters_+3A_max.exp">max.exp</code></td>
<td>
<p>Maximum average normalized expression for HVG. If higher, the gene will be excluded</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the input Seurat object <code>obj</code> with the calculated highly 
variable features accessible through <code>VariableFeatures(obj)</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sampleObj)
sampleObj &lt;- findVariableFeatures_wfilters(sampleObj, nfeatures=100)

</code></pre>

<hr>
<h2 id='getDataMatrix'>Extract data matrix from Seurat object</h2><span id='topic+getDataMatrix'></span>

<h3>Description</h3>

<p>Get the gene expression matrix from a Seurat object, optionally centered
and/or subset on highly variable genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDataMatrix(
  obj,
  assay = "RNA",
  slot = "data",
  hvg = NULL,
  center = FALSE,
  scale = FALSE,
  non_negative = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDataMatrix_+3A_obj">obj</code></td>
<td>
<p>Seurat object</p>
</td></tr>
<tr><td><code id="getDataMatrix_+3A_assay">assay</code></td>
<td>
<p>Get data matrix from this assay</p>
</td></tr>
<tr><td><code id="getDataMatrix_+3A_slot">slot</code></td>
<td>
<p>Get data matrix from this slot (=layer)</p>
</td></tr>
<tr><td><code id="getDataMatrix_+3A_hvg">hvg</code></td>
<td>
<p>List of variable genes to subset the matrix. If NULL, uses
all genes</p>
</td></tr>
<tr><td><code id="getDataMatrix_+3A_center">center</code></td>
<td>
<p>Whether to center the data matrix</p>
</td></tr>
<tr><td><code id="getDataMatrix_+3A_scale">scale</code></td>
<td>
<p>Whether to scale the data matrix</p>
</td></tr>
<tr><td><code id="getDataMatrix_+3A_non_negative">non_negative</code></td>
<td>
<p>Enforce non-negative values for NMF</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a sparse data matrix (cells per genes), subset 
according to the given parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sampleObj)
matrix &lt;- getDataMatrix(sampleObj)

</code></pre>

<hr>
<h2 id='getMetaPrograms'>Extract consensus gene programs (meta-programs)</h2><span id='topic+getMetaPrograms'></span>

<h3>Description</h3>

<p>Run it over a list of NMF models obtained using <code><a href="#topic+multiNMF">multiNMF</a></code>; it will 
determine gene programs that are consistently observed across samples 
and values of k.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMetaPrograms(
  nmf.res,
  nMP = 10,
  specificity.weight = 5,
  weight.explained = 0.5,
  max.genes = 200,
  metric = c("cosine", "jaccard"),
  hclust.method = "ward.D2",
  min.confidence = 0.5,
  remove.empty = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getMetaPrograms_+3A_nmf.res">nmf.res</code></td>
<td>
<p>A list of NMF models obtained from <code><a href="#topic+multiNMF">multiNMF</a></code></p>
</td></tr>
<tr><td><code id="getMetaPrograms_+3A_nmp">nMP</code></td>
<td>
<p>Total number of meta-programs</p>
</td></tr>
<tr><td><code id="getMetaPrograms_+3A_specificity.weight">specificity.weight</code></td>
<td>
<p>A parameter controlling how specific gene
should be for each program. 'specificity.weight=0' no constraint on
specificity, and positive values impose increasing specificity.</p>
</td></tr>
<tr><td><code id="getMetaPrograms_+3A_weight.explained">weight.explained</code></td>
<td>
<p>Fraction of NMF weights explained by selected
genes. For example if weight.explained=0.5, all genes that together
account for 50% of NMF weights are used to return program signatures.</p>
</td></tr>
<tr><td><code id="getMetaPrograms_+3A_max.genes">max.genes</code></td>
<td>
<p>Max number of genes for each programs</p>
</td></tr>
<tr><td><code id="getMetaPrograms_+3A_metric">metric</code></td>
<td>
<p>Metric to calculate pairwise similarity between programs</p>
</td></tr>
<tr><td><code id="getMetaPrograms_+3A_hclust.method">hclust.method</code></td>
<td>
<p>Method to build similarity tree between individual programs</p>
</td></tr>
<tr><td><code id="getMetaPrograms_+3A_min.confidence">min.confidence</code></td>
<td>
<p>Percentage of programs in which a gene is seen 
(out of programs in the corresponding program tree branch/cluster), to be 
retained in the consensus metaprograms</p>
</td></tr>
<tr><td><code id="getMetaPrograms_+3A_remove.empty">remove.empty</code></td>
<td>
<p>Whether to remove meta-programs with no genes above 
confidence threshold</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with i) 'metaprograms.genes' top genes for each 
meta-program; ii) 'metaprograms.metrics' dataframe with meta-programs 
statistics: a) freq. of samples where the MP is present, b) average 
silhouette width, c) mean similarity (cosine or Jaccard), d) number of genes in MP, 
e) number of gene programs in MP; iii) 'programs.similarity': matrix of 
similarities (Jaccard or cosine) between meta-programs; iv) 'programs.tree': 
hierarchical clustering of meta-programs (hclust tree); v) 
'programs.clusters': meta-program identity for each program
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
data(sampleObj)
geneNMF_programs &lt;- multiNMF(list(sampleObj), k=5)
geneNMF_metaprograms &lt;- getMetaPrograms(geneNMF_programs, nMP=3)

</code></pre>

<hr>
<h2 id='getNMFgenes'>Get list of genes for each NMF program</h2><span id='topic+getNMFgenes'></span>

<h3>Description</h3>

<p>Run it over a list of NMF models obtained using <code>multiNMF()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNMFgenes(
  nmf.res,
  specificity.weight = 5,
  weight.explained = 0.5,
  max.genes = 200
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getNMFgenes_+3A_nmf.res">nmf.res</code></td>
<td>
<p>A list of NMF models obtained using <code>multiNMF()</code></p>
</td></tr>
<tr><td><code id="getNMFgenes_+3A_specificity.weight">specificity.weight</code></td>
<td>
<p>A parameter controlling how specific gene
should be for each program. 'specificity.weight=0' no constraint on
specificity, and positive values impose increasing specificity.</p>
</td></tr>
<tr><td><code id="getNMFgenes_+3A_weight.explained">weight.explained</code></td>
<td>
<p>Fraction of NMF weights explained by selected
genes. For example if weight.explained=0.5, all genes that together
account for 50% of NMF weights are used to return program signatures.</p>
</td></tr>
<tr><td><code id="getNMFgenes_+3A_max.genes">max.genes</code></td>
<td>
<p>Max number of genes for each program</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of top genes for each gene program found
by <code>multiNMF()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
data(sampleObj)
geneNMF_programs &lt;- multiNMF(list(sampleObj), k=5)
geneNMF_genes &lt;- getNMFgenes(geneNMF_programs)

</code></pre>

<hr>
<h2 id='multiNMF'>Run NMF on a list of Seurat objects</h2><span id='topic+multiNMF'></span>

<h3>Description</h3>

<p>Given a list of Seurat objects, run non-negative matrix factorization on 
each sample individually, over a range of target NMF components (k).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiNMF(
  obj.list,
  assay = "RNA",
  slot = "data",
  k = 5:6,
  hvg = NULL,
  nfeatures = 2000,
  L1 = c(0, 0),
  min.exp = 0.01,
  max.exp = 3,
  center = FALSE,
  scale = FALSE,
  min.cells.per.sample = 10,
  hvg.blocklist = NULL,
  seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiNMF_+3A_obj.list">obj.list</code></td>
<td>
<p>A list of Seurat objects</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_assay">assay</code></td>
<td>
<p>Get data matrix from this assay</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_slot">slot</code></td>
<td>
<p>Get data matrix from this slot (=layer)</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_k">k</code></td>
<td>
<p>Number of target components for NMF (can be a vector)</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_hvg">hvg</code></td>
<td>
<p>List of pre-calculated variable genes to subset the matrix.
If hvg=NULL it calculates them automatically</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_nfeatures">nfeatures</code></td>
<td>
<p>Number of HVG, if calculate_hvg=TRUE</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_l1">L1</code></td>
<td>
<p>L1 regularization term for NMF</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_min.exp">min.exp</code></td>
<td>
<p>Minimum average log-expression value for retaining genes</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_max.exp">max.exp</code></td>
<td>
<p>Maximum average log-expression value for retaining genes</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_center">center</code></td>
<td>
<p>Whether to center the data matrix</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_scale">scale</code></td>
<td>
<p>Whether to scale the data matrix</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_min.cells.per.sample">min.cells.per.sample</code></td>
<td>
<p>Minimum numer of cells per sample (smaller 
samples will be ignored)</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_hvg.blocklist">hvg.blocklist</code></td>
<td>
<p>Optionally takes a vector or list of vectors of gene
names. These genes will be ignored for HVG detection. This is useful
to mitigateeffect of genes associated with technical artifacts and
batch effects (e.g. mitochondrial), and to exclude TCR and BCR 
adaptive immune(clone-specific) receptors. If set to 'NULL' no genes 
will be excluded</p>
</td></tr>
<tr><td><code id="multiNMF_+3A_seed">seed</code></td>
<td>
<p>Random seed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of NMF programs, one for each sample and for each
value of 'k'. The format of each program in the list follosw the
structure of <code><a href="RcppML.html#topic+nmf">nmf</a></code> factorization models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
data(sampleObj)
geneNMF_programs &lt;- multiNMF(list(sampleObj), k=5)

</code></pre>

<hr>
<h2 id='multiPCA'>Run PCA on a list of Seurat objects</h2><span id='topic+multiPCA'></span>

<h3>Description</h3>

<p>Given a list of Seurat objects, run non-negative PCA factorization on 
each sample individually.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiPCA(
  obj.list,
  assay = "RNA",
  slot = "data",
  k = 4:5,
  hvg = NULL,
  nfeatures = 500,
  min.exp = 0.01,
  max.exp = 3,
  min.cells.per.sample = 10,
  center = FALSE,
  scale = FALSE,
  hvg.blocklist = NULL,
  seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiPCA_+3A_obj.list">obj.list</code></td>
<td>
<p>A list of Seurat objects</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_assay">assay</code></td>
<td>
<p>Get data matrix from this assay</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_slot">slot</code></td>
<td>
<p>Get data matrix from this slot (=layer)</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_k">k</code></td>
<td>
<p>Number of target components for PCA</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_hvg">hvg</code></td>
<td>
<p>List of pre-calculated variable genes to subset the matrix.
If hvg=NULL it calculates them automatically</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_nfeatures">nfeatures</code></td>
<td>
<p>Number of HVG, if calculate_hvg=TRUE</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_min.exp">min.exp</code></td>
<td>
<p>Minimum average log-expression value for retaining genes</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_max.exp">max.exp</code></td>
<td>
<p>Maximum average log-expression value for retaining genes</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_min.cells.per.sample">min.cells.per.sample</code></td>
<td>
<p>Minimum numer of cells per sample (smaller 
samples will be ignored)</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_center">center</code></td>
<td>
<p>Whether to center the data matrix</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_scale">scale</code></td>
<td>
<p>Whether to scale the data matrix</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_hvg.blocklist">hvg.blocklist</code></td>
<td>
<p>Optionally takes a vector or list of vectors of gene
names. These genes will be ignored for HVG detection. This is useful
to mitigateeffect of genes associated with technical artifacts and
batch effects (e.g. mitochondrial), and to exclude TCR and BCR 
adaptive immune(clone-specific) receptors. If set to 'NULL' no genes 
will be excluded</p>
</td></tr>
<tr><td><code id="multiPCA_+3A_seed">seed</code></td>
<td>
<p>Random seed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of non-negative PCA programs, one for each sample.
The format of each program in the list follows the
structure of <code><a href="RcppML.html#topic+nmf">nmf</a></code> factorization models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
data(sampleObj)
geneNMF_programs &lt;- multiPCA(list(sampleObj), k=5)

</code></pre>

<hr>
<h2 id='plotMetaPrograms'>Visualizations for meta-programs</h2><span id='topic+plotMetaPrograms'></span>

<h3>Description</h3>

<p>Generates a clustered heatmap for meta-program similarities (by Jaccard 
index or Cosine similarity). This function is intended to be run on the object
generated by <code><a href="#topic+getMetaPrograms">getMetaPrograms</a></code>, which contains a pre-calculated 
tree of pairwise similarities between clusters (as a 'hclust' object).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMetaPrograms(
  mp.res,
  similarity.cutoff = c(0, 1),
  scale = "none",
  downsample = 1000,
  showtree = TRUE,
  palette = viridis(100, option = "A", direction = -1),
  annotation_colors = NULL,
  main = "Clustered Heatmap",
  show_rownames = FALSE,
  show_colnames = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotMetaPrograms_+3A_mp.res">mp.res</code></td>
<td>
<p>The meta-programs object generated by <code><a href="#topic+getMetaPrograms">getMetaPrograms</a></code></p>
</td></tr>
<tr><td><code id="plotMetaPrograms_+3A_similarity.cutoff">similarity.cutoff</code></td>
<td>
<p>Min and max values for similarity metric</p>
</td></tr>
<tr><td><code id="plotMetaPrograms_+3A_scale">scale</code></td>
<td>
<p>Heatmap rescaling (passed to pheatmap as 'scale')</p>
</td></tr>
<tr><td><code id="plotMetaPrograms_+3A_downsample">downsample</code></td>
<td>
<p>Limit max number of samples in heatmap, to avoid overloading
the graphics</p>
</td></tr>
<tr><td><code id="plotMetaPrograms_+3A_showtree">showtree</code></td>
<td>
<p>Whether to plot the hierarchical clustering tree</p>
</td></tr>
<tr><td><code id="plotMetaPrograms_+3A_palette">palette</code></td>
<td>
<p>Heatmap color palette (passed to pheatmap as 'color')</p>
</td></tr>
<tr><td><code id="plotMetaPrograms_+3A_annotation_colors">annotation_colors</code></td>
<td>
<p>Color palette for MP annotations</p>
</td></tr>
<tr><td><code id="plotMetaPrograms_+3A_main">main</code></td>
<td>
<p>Heatmap title</p>
</td></tr>
<tr><td><code id="plotMetaPrograms_+3A_show_rownames">show_rownames</code></td>
<td>
<p>Whether to display individual program names as rows</p>
</td></tr>
<tr><td><code id="plotMetaPrograms_+3A_show_colnames">show_colnames</code></td>
<td>
<p>Whether to display individual program names as cols</p>
</td></tr>
<tr><td><code id="plotMetaPrograms_+3A_...">...</code></td>
<td>
<p>Additional parameters for pheatmap</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a clustered heatmap of MP similarities, in ggplot2 format
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
data(sampleObj)
geneNMF_programs &lt;- multiNMF(list(sampleObj), k=5)
geneNMF_metaprograms &lt;- getMetaPrograms(geneNMF_programs, nMP=3)
plotMetaPrograms(geneNMF_metaprograms)

</code></pre>

<hr>
<h2 id='runGSEA'>Run Gene set enrichment analysis</h2><span id='topic+runGSEA'></span>

<h3>Description</h3>

<p>Utility function to run Gene Set Enrichment Analysis (GSEA) against gene 
sets from MSigDB. Note: this is an optional function, which is conditional
to the installation of suggested packages <code>fgsea</code> and <code>msigdbr</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runGSEA(
  genes,
  universe = NULL,
  category = "H",
  subcategory = NULL,
  species = "Homo sapiens",
  pval.thr = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="runGSEA_+3A_genes">genes</code></td>
<td>
<p>A vector of genes</p>
</td></tr>
<tr><td><code id="runGSEA_+3A_universe">universe</code></td>
<td>
<p>Background universe of gene symbols (passed on to <code>fgsea::fora</code>)</p>
</td></tr>
<tr><td><code id="runGSEA_+3A_category">category</code></td>
<td>
<p>GSEA main category (e.g. &quot;H&quot; or &quot;C5&quot;)</p>
</td></tr>
<tr><td><code id="runGSEA_+3A_subcategory">subcategory</code></td>
<td>
<p>GSEA subcategory</p>
</td></tr>
<tr><td><code id="runGSEA_+3A_species">species</code></td>
<td>
<p>Species for GSEA analysis. For a list of the available species,
type <code>msigdbr::msigdbr_species()</code></p>
</td></tr>
<tr><td><code id="runGSEA_+3A_pval.thr">pval.thr</code></td>
<td>
<p>Min p-value to include results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a table of enriched gene programs from GSEA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sampleObj)
geneset &lt;- c("BANK1","CD22","CD79A","CD19","IGHD","IGHG3","IGHM")
#test is conditional on availability of suggested packages
if (requireNamespace("fgsea", quietly=TRUE) &amp;
   requireNamespace("msigdbr", quietly=TRUE)) {
   gsea_res &lt;- runGSEA(geneset,
       universe=rownames(sampleObj),
       category = "C8")
}

</code></pre>

<hr>
<h2 id='runNMF'>Compute NMF as a low-dim embedding for Seurat</h2><span id='topic+runNMF'></span>

<h3>Description</h3>

<p>Compute NMF embeddings for single-cell dataset, and store them in the Seurat data
structure. They can be used as an alternative to PCA for downstream analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runNMF(
  obj,
  assay = "RNA",
  slot = "data",
  k = 10,
  new.reduction = "NMF",
  seed = 123,
  L1 = c(0, 0),
  hvg = NULL,
  center = FALSE,
  scale = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="runNMF_+3A_obj">obj</code></td>
<td>
<p>A seurat object</p>
</td></tr>
<tr><td><code id="runNMF_+3A_assay">assay</code></td>
<td>
<p>Get data matrix from this assay</p>
</td></tr>
<tr><td><code id="runNMF_+3A_slot">slot</code></td>
<td>
<p>Get data matrix from this slot (=layer)</p>
</td></tr>
<tr><td><code id="runNMF_+3A_k">k</code></td>
<td>
<p>Number of components for low-dim representation</p>
</td></tr>
<tr><td><code id="runNMF_+3A_new.reduction">new.reduction</code></td>
<td>
<p>Name of new dimensionality reduction</p>
</td></tr>
<tr><td><code id="runNMF_+3A_seed">seed</code></td>
<td>
<p>Random seed</p>
</td></tr>
<tr><td><code id="runNMF_+3A_l1">L1</code></td>
<td>
<p>L1 regularization term for NMF</p>
</td></tr>
<tr><td><code id="runNMF_+3A_hvg">hvg</code></td>
<td>
<p>Which genes to use for the reduction</p>
</td></tr>
<tr><td><code id="runNMF_+3A_center">center</code></td>
<td>
<p>Whether to center the data matrix</p>
</td></tr>
<tr><td><code id="runNMF_+3A_scale">scale</code></td>
<td>
<p>Whether to scale the data matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a Seurat object with a new dimensionality reduction (NMF)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sampleObj)
sampleObj &lt;- runNMF(sampleObj, k=8)
</code></pre>

<hr>
<h2 id='sampleObj'>Sample dataset to test GeneNMF installation</h2><span id='topic+sampleObj'></span>

<h3>Description</h3>

<p>A Seurat object containing single-cell transcriptomes
(scRNA-seq) for 50 cells and 20729 genes.
Single-cell UMI counts were normalized using a standard log-normalization:
counts for each cell were divided by the total counts for that cell and
multiplied by 10,000, then natural-log transformed using 'log1p'. <br /><br />
This a subsample of 25 predicted B cells and 25 predicted NK cells from
the large scRNA-seq PBMC dataset published
by Hao et al. (<a href="https://doi.org/10.1016/j.cell.2021.04.048">doi:10.1016/j.cell.2021.04.048</a>) and
available as UMI counts at
<a href="https://atlas.fredhutch.org/data/nygc/multimodal/pbmc_multimodal.h5seurat">https://atlas.fredhutch.org/data/nygc/multimodal/pbmc_multimodal.h5seurat</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleObj
</code></pre>


<h3>Format</h3>

<p>A sparse matrix of 50 cells and 20729 genes.
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.1016/j.cell.2021.04.048">doi:10.1016/j.cell.2021.04.048</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
