<!DOCTYPE html><html><head><title>Help for package GeneticSubsetter</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GeneticSubsetter}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CoreSetOptimizer'>
<p>Subset Optimization</p></a></li>
<li><a href='#CoreSetter'>
<p>Genotype Subsetting</p></a></li>
<li><a href='#CoreSetterCombined'>
<p>Genotype Subsetting-Alternative Method</p></a></li>
<li><a href='#CoreSetterPoly'>
<p>Genotype Subsetting for Autopolyploids and Polymorphic Markers</p></a></li>
<li><a href='#GeneticSubsetter'><p>Genetic Subsetter</p></a></li>
<li><a href='#genotypes'>
<p>Sample Barley Genotypes</p></a></li>
<li><a href='#genotypes.poly'>
<p>Sample Barley Genotypes</p></a></li>
<li><a href='#HET'>
<p>Heterozygosity Calculator</p></a></li>
<li><a href='#Mat'>
<p>Kinship Matrix Creator</p></a></li>
<li><a href='#MTK'>
<p>MTK calculator</p></a></li>
<li><a href='#MtkCalc'>
<p>MTK calculator (Depreciated)</p></a></li>
<li><a href='#PicCalc'>
<p>PIC Calculator (Depreciated)</p></a></li>
<li><a href='#SubsetterMTK'><p>Genotype Subsetting with PIC - Method One (Depreciated)</p></a></li>
<li><a href='#SubsetterPIC'>
<p>Genotype Subsetting with PIC - Method One (Depreciated)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identify Favorable Subsets of Germplasm Collections</td>
</tr>
<tr>
<td>Version:</td>
<td>0.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-10-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Ryan C. Graebner and Alfonso Cuesta-Marcos</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ryan C. Graebner &lt;ryan.graebner@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Finds subsets of sets of genotypes with a high Heterozygosity, and Mean of Transformed Kinships (MTK), measures that can indicate a subset would be beneficial for rare-trait discovery and genome-wide association scanning, respectively.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-10-25 19:10:33 UTC; ryan</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-10-25 23:03:53</td>
</tr>
</table>
<hr>
<h2 id='CoreSetOptimizer'>
Subset Optimization
</h2><span id='topic+CoreSetOptimizer'></span>

<h3>Description</h3>

<p>This function works to systematically improves a subset via single-genotype replacements from a larger population.  This function will continue to work until no more single-genotype replacements can be made to increase the subset's value. Criteria that can be used to judge the value of subsets are Expected Heterozygosity (HET; for rare-trait discovery; called PIC in earlier versions and in the paper describing this package), and the Mean of Transformed Kinships (MTK; for GWAS). A complete comparison of these two criteria is presented in Graebner et al. (2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CoreSetOptimizer(genos=NULL, subset=NULL, criterion = "HET", 
    mat = NULL, save = NULL, power = 10, print = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CoreSetOptimizer_+3A_genos">genos</code></td>
<td>

<p>A matrix of genotypes, where each column is one individual, each row is one marker, and marker values are 1, 0, or -1, or NA, where 0 represents a heterozygous marker, and NA represents missing data. Note that this coding is different from the earlier SubsetOptimizerPIC and SubsetOptimizerMTK, which cannot handle heterozygous markers.  All data in this matrix must be numeric.
</p>
</td></tr>
<tr><td><code id="CoreSetOptimizer_+3A_subset">subset</code></td>
<td>

<p>The names of the genotypes in the starting subset.
</p>
</td></tr>
<tr><td><code id="CoreSetOptimizer_+3A_criterion">criterion</code></td>
<td>

<p>The criterion to be used for comparing subsets (HET or MTK).
</p>
</td></tr>
<tr><td><code id="CoreSetOptimizer_+3A_mat">mat</code></td>
<td>

<p>A kinship matrix, if one has already been computed for the population. If a kinship matrix is included, the &quot;genos&quot; argument may be left empty.
</p>
</td></tr>
<tr><td><code id="CoreSetOptimizer_+3A_save">save</code></td>
<td>

<p>A list of genotype names, corresponding to the column names in the genotype matrix, that will not be eliminated.
</p>
</td></tr>
<tr><td><code id="CoreSetOptimizer_+3A_power">power</code></td>
<td>

<p>The transformation that should be made to the kinship matrix, if the MTK criterion is used. If power=1, the kinship matrix is not transformed, if power=2, the kinship matrix is squared, etc.  When the power is higher, this function preferentially eliminates genotypes that are closely related to other genotypes in the population.
</p>
</td></tr>
<tr><td><code id="CoreSetOptimizer_+3A_print">print</code></td>
<td>

<p>Whether to the value of intermediate subsets.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of the genotype names included in the best subset found.
</p>


<h3>Note</h3>

<p>The ability to recogize heterozygous markers was included in CoreSetOptimizer, resulting in a slightly different genotype coding scheme than the depreciated functions SubsetOptimizerPIC and SubsetOptimizerMTK.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Graebner
</p>


<h3>References</h3>

<p>Graebner RC, Hayes PM, Hagerty CH, Cuesta-Marcos A (2016) A comparison of polymorphism information content and mean of transformed kinships as criteria for selection informative subsets of barley (Hordeum vulgare L. s. l) from the USDA Barley Core Collection. Genet Resour Crop Evol 63:477-482.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genotypes")
CoreSetOptimizer(genotypes,subset=colnames(genotypes)[c(1,3,5,7,8,9)],
    criterion="HET",save=colnames(genotypes)[c(1,5,9)])
CoreSetOptimizer(genotypes,subset=colnames(genotypes)[c(1,3,5,7,8,9)],
    criterion="MTK",save=colnames(genotypes)[c(1,5,9)])
</code></pre>

<hr>
<h2 id='CoreSetter'>
Genotype Subsetting
</h2><span id='topic+CoreSetter'></span>

<h3>Description</h3>

<p>This function systematically eliminates genotypes from a large population to arrive at a favorable subset.  This method will typically return less favorable subsets than the method used by the CoreSetterCombined function if sufficient permutations are used for the later, but CoreSetter is quicker, and will rank all genotypes, as opposed to returning a single, static subset. Criteria that can be used to judge the value of subsets are Expected Heterozygosity (HET; for rare-trait discovery; called PIC in earlier versions and in the paper describing this package), and the Mean of Transformed Kinships (MTK; for GWAS). A complete comparison of these two criteria is presented in Graebner et al. (2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CoreSetter(genos=NULL, criterion = "HET", save = NULL,
    power = 10, mat = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CoreSetter_+3A_genos">genos</code></td>
<td>

<p>A matrix of genotypes, where each column is one individual, each row is one marker, and marker values are 1, 0, or -1, or NA, where 0 represents a heterozygous marker, and NA represents missing data. Note that this coding is different from the earlier SubsetterPIC and SubsetterMTK, which cannot handle heterozygous markers.  All data in this matrix must be numeric.
</p>
</td></tr>
<tr><td><code id="CoreSetter_+3A_criterion">criterion</code></td>
<td>

<p>The criterion to be used for comparing subsets (HET or MTK).
</p>
</td></tr>
<tr><td><code id="CoreSetter_+3A_save">save</code></td>
<td>

<p>A list of genotype names, corresponding to the column names in the genotype matrix, that will not be eliminated.
</p>
</td></tr>
<tr><td><code id="CoreSetter_+3A_power">power</code></td>
<td>

<p>The transformation that should be made to the kinship matrix, if the MTK criterion is used. If power=1, the kinship matrix is not transformed, if power=2, the kinship matrix is squared, etc.  When the power is higher, this function preferentially eliminates genotypes that are closely related to other genotypes in the population.
</p>
</td></tr>
<tr><td><code id="CoreSetter_+3A_mat">mat</code></td>
<td>

<p>A kinship matrix, if one has already been computed for the population. If a kinship matrix is included, the &quot;genos&quot; argument may be left empty.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with three columns.  The first column is the rank of a particular genotype to the population's MTK, based on the order in which genotypes were eliminated (genotypes with lower rank were retained longer, genotypes with rank of 1 were not eliminated).  The second column is the name of the genotype. The third column shows the value of the subset including that genotype and all genotypes with a lower rank, as judged by the specified criterion.
</p>


<h3>Note</h3>

<p>In Graebner et al. (2015), and in their earlier functions SubsetterPIC and SubsetterMTK, heterozygous markers were counted as missing data, due to previous limitations if GeneticSubsetter. Please note that this has changed the required coding scheme for input genotype data. When using the HET criterion, this function uses the same method and criteria described in Munoz-Amatrain et al. (2014), but with a more computationally efficient approach.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Graebner and Alfonso Cuesta-Marcos
</p>


<h3>References</h3>

<p>Graebner RC, Hayes PM, Hagerty CH, Cuesta-Marcos A (2016) A comparison of polymorphism information content and mean of transformed kinships as criteria for selection informative subsets of barley (Hordeum vulgare L. s. l) from the USDA Barley Core Collection. Genet Resour Crop Evol 64:477-482.
Munoz-Amatrain M, Cuesta-Marcos A, Endelman JB, Comadran J, Bonman JM (2014) The USDA barley core collection: genetic diversity, population structure, and potential for genome-wide association studies. PloS One 9:e94688.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genotypes")
CoreSetter(genotypes,criterion="HET",save=colnames(genotypes)[c(1,5,9)])
CoreSetter(genotypes,criterion="MTK",save=colnames(genotypes)[c(1,5,9)])
</code></pre>

<hr>
<h2 id='CoreSetterCombined'>
Genotype Subsetting-Alternative Method
</h2><span id='topic+CoreSetterCombined'></span>

<h3>Description</h3>

<p>This function creates a series of random subsets.  Then, each of these subsets is improved using the CoreSetOptimizer function using a series of single-genotype replacements that result in a higher value for the subset, until no more single-genotype replacements can be made to improve the subset. This process is similar to a Local Search. Criteria that can be used to judge the value of subsets are Expected Heterozygosity (HET; for rare-trait discovery; called PIC in earlier versions and in the paper describing this package), and the Mean of Transformed Kinships (MTK; for GWAS). A complete comparison of these two criteria is presented in Graebner et al. (2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CoreSetterCombined(genos=NULL, size=NULL, criterion =  "HET",
    save = NULL, power = 10, permutations = 100, print = TRUE,
    mat = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CoreSetterCombined_+3A_genos">genos</code></td>
<td>

<p>A matrix of genotypes, where each column is one individual, each row is one marker, and marker values are 1, 0, or -1, or NA, where 0 represents a heterozygous marker, and NA represents missing data. Note that this coding is different from the earlier SubsetterCombinedPIC and SubsetterCombinedMTK, which cannot handle heterozygous markers.  All data in this matrix must be numeric.
</p>
</td></tr>
<tr><td><code id="CoreSetterCombined_+3A_size">size</code></td>
<td>

<p>The desired subset size.
</p>
</td></tr>
<tr><td><code id="CoreSetterCombined_+3A_criterion">criterion</code></td>
<td>

<p>The criterion to be used for comparing subsets (HET or MTK).
</p>
</td></tr>
<tr><td><code id="CoreSetterCombined_+3A_save">save</code></td>
<td>

<p>A list of genotype names, corresponding to the column names in the genotype matrix, that will not be eliminated.
</p>
</td></tr>
<tr><td><code id="CoreSetterCombined_+3A_power">power</code></td>
<td>

<p>The transformation that should be made to the kinship matrix, if the MTK criterion is used. If power=1, the kinship matrix is not transformed, if power=2, the kinship matrix is squared, etc.  When the power is higher, this function preferentially eliminates genotypes that are closely related to other genotypes in the population.
</p>
</td></tr>
<tr><td><code id="CoreSetterCombined_+3A_permutations">permutations</code></td>
<td>

<p>The number of random subsets to improve.
</p>
</td></tr>
<tr><td><code id="CoreSetterCombined_+3A_print">print</code></td>
<td>

<p>If TRUE, this function prints the Heterozygosity or MTK of the best subset identified.
</p>
</td></tr>
<tr><td><code id="CoreSetterCombined_+3A_mat">mat</code></td>
<td>

<p>A kinship matrix, if one has already been computed for the population. If a kinship matrix is included, the &quot;genos&quot; argument may be left empty.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of the genotype names included in the best subset found.
</p>


<h3>Note</h3>

<p>The ability to recognize heterozygous markers was included in CoreSetterCombined, resulting in a slightly different genotype coding scheme than the depreciated functions SubsetterCombinedPIC and SubsetterCombinedMTK.
</p>


<h3>Author(s)</h3>

<p>Ryan C Graebner
</p>


<h3>References</h3>

<p>Graebner RC, Hayes PM, Hagerty CH, Cuesta-Marcos A (2016) A comparison of polymorphism information content and mean of transformed kinships as criteria for selection informative subsets of barley (Hordeum vulgare L. s. l) from the USDA Barley Core Collection. Genet Resour Crop Evol 63:477-482.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genotypes")
CoreSetterCombined(genotypes,size=6,criterion="HET",permutations=10,
    save=colnames(genotypes)[c(1,5,9)])
CoreSetterCombined(genotypes,size=6,criterion="MTK",permutations=10,
    save=colnames(genotypes)[c(1,5,9)])
</code></pre>

<hr>
<h2 id='CoreSetterPoly'>
Genotype Subsetting for Autopolyploids and Polymorphic Markers
</h2><span id='topic+CoreSetterPoly'></span>

<h3>Description</h3>

<p>This function systematically eliminates genotypes from a large population to arrive at a favorable subset, and can accommodate datasets with autopolyploids and polymorphic markers. At this time, CoreSetterPoly can only use the Expected Heterozygosity criterion to quantify the value of subsets, and Sequential Backward Selection to arrive at favorable subsets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CoreSetterPoly(genos, ploidy, save = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CoreSetterPoly_+3A_genos">genos</code></td>
<td>

<p>A matrix of genotypes, where each row is one individual, and each set of X columns (where X is the ploidy) is one locus. At each locus, any number of alleles can be included, where each allele is referred to by a different integer. Missing data should be represented by NA. The X cells for any genotype-locus combination are the alleles known to be present at that locus for that genotype, in the frequency that they are present. If at least one but not all of the cells for a genotype-locus combination are listed as NA, that data point is imputed based on the other alleles at that locus.
</p>
</td></tr>
<tr><td><code id="CoreSetterPoly_+3A_ploidy">ploidy</code></td>
<td>

<p>The ploidy of the organizm to be subsetted, in respect to the number of alleles that can be present at one locus.
</p>
</td></tr>
<tr><td><code id="CoreSetterPoly_+3A_save">save</code></td>
<td>

<p>A list of genotype names, corresponding to the column names in the genotype matrix, that will not be eliminated.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with three columns.  The first column is the rank of a particular genotype to the population's MTK, based on the order in which genotypes were eliminated (genotypes with lower rank were retained longer, genotypes with rank of 1 were not eliminated).  The second column is the name of the genotype. The third column shows the value of the subset including that genotype and all genotypes with a lower rank, as judged by the Expected Heterozygosity criterion.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Graebner and Alfonso Cuesta-Marcos
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genotypes")
CoreSetterPoly(genotypes.poly,ploidy=2,save=rownames(genotypes.poly)[c(1,5,9)])
</code></pre>

<hr>
<h2 id='GeneticSubsetter'>Genetic Subsetter</h2><span id='topic+GeneticSubsetter'></span>

<h3>Description</h3>

<p>This package contains a set of tools that can be used to select a subset from a larger population, using genetic data. Two criteria are used to identify subsets, in seperate functions: Expected Heterozygosity (HET; called PIC in earlier versions and in the paper describing this package) and the Mean of Transformed Kinships (MTK).</p>


<h3>Details</h3>

<p>When selecting subsets of genotypes, two factors are important to consider: the criteria by which to judge subsets, and the method used to identfy the set of genotypes that best fit that criteria. Two criteria are Expected Heterozygosity (HET) and the Mean of Transformed Kinships (MTK). Tests suggest that of these two criteria, Expected Heterozygosity is better if the resulting subset will be used for rare-trait discovery, while MTK is better if the resulting subset will be used for genome-wide association scanning (Graebner et al. 2015). To reach subsets with a high Expected Heterozygosity or MTK, CoreSetter systematically removes genotypes from the full set, creating a full ranking of genotype's contributions to their respective criteria. When the HET criterion is selected, CoreSetter uses the same method and criteria described in Munoz-Amatrain et al. (2014), except CoreSetter uses a more computationally efficient approach, and CoreSetter can consider heterozygous markers. Alternatively, CoreSetterCombined works to systematically improve a user-defined number of random subsets via single-genotype replacements, until no replacement can increase the selected criteria.  This later method generally returns subsets with a higher Heterozyosity or MTK, but are subset-size specific, take more time to compute, and will not always return identical results in subsequent runs.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Graebner &lt;ryan.graebner@gmail.com&gt; and Alfonso Cuesta-Marcos
</p>


<h3>References</h3>

<p>Graebner RC, Hayes PM, Hagerty CH, Cuesta-Marcos A (2016) A comparison of polymorphism information content and mean of transformed kinships as criteria for selection informative subsets of barley (Hordeum vulgare L. s. l) from the USDA Barley Core Collection. Genet Resour Crop Evol 63:477-482.
Munoz-Amatrain M, Cuesta-Marcos A, Endelman JB, Comadran J, Bonman JM (2014) The USDA barley core collection: genetic diversity, population structure, and potential for genome-wide association studies. PloS One 9:e94688.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genotypes")
CoreSetter(genotypes,criterion="HET",save=colnames(genotypes)[c(1,5,9)])
</code></pre>

<hr>
<h2 id='genotypes'>
Sample Barley Genotypes
</h2><span id='topic+genotypes'></span>

<h3>Description</h3>

<p>Twenty diploid barley genotypes, with twenty markers each, for to demonstrate functions in the GeneticSubsetter package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(genotypes)</code></pre>


<h3>Format</h3>

<p>Columns are gentoypes, and rows are markers, formatted for the CoreSetter, CoreSetterCombined, CoreSetOptimizer, HET, MTK.
</p>


<h3>Source</h3>

<p>Triticeae Coordinated Agricultural Project (T-CAP) (http://triticeaetoolbox.org)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genotypes")
str(genotypes)
</code></pre>

<hr>
<h2 id='genotypes.poly'>
Sample Barley Genotypes
</h2><span id='topic+genotypes.poly'></span>

<h3>Description</h3>

<p>Twenty diploid barley genotypes, with twenty markers each, for to demonstrate functions in the GeneticSubsetter package, formatted for CoreSetterPoly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(genotypes)</code></pre>


<h3>Format</h3>

<p>Rows are genotypes, and each set of two columns is one locus.
</p>


<h3>Source</h3>

<p>Triticeae Coordinated Agricultural Project (T-CAP) (http://triticeaetoolbox.org)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genotypes")
str(genotypes.poly)
</code></pre>

<hr>
<h2 id='HET'>
Heterozygosity Calculator
</h2><span id='topic+HET'></span>

<h3>Description</h3>

<p>This function calculates the Expected Heterozygosity (HET; called PIC in earlier versions and in the paper describing this package) of a set of genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HET(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HET_+3A_data">data</code></td>
<td>

<p>A matrix of genotypes, where each column is one individual, each row is one marker, and marker values are 1, 0, or -1, or NA, where 0 represents a heterozygous marker, and NA represents missing data. Note that this coding is different from the earlier PicCalc, which cannot handle heterozygous markers.  All data in this matrix must be numeric.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The mean Heterozygosity of all markers for the given set of genotypes.
</p>


<h3>Note</h3>

<p>The ability to recogize heterozygous markers was included in HET, resulting in a slightly different genotype coding scheme than the earlier PicCalc.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Graebner
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genotypes")
HET(genotypes)
</code></pre>

<hr>
<h2 id='Mat'>
Kinship Matrix Creator
</h2><span id='topic+Mat'></span>

<h3>Description</h3>

<p>This function creates a kinship matrix for a set of genotypes. This function is a simplified version of the function &quot;A.mat&quot; in the R package rrBLUP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Mat(genos)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Mat_+3A_genos">genos</code></td>
<td>

<p>A matrix of genotypes, where each column is one individual, each row is one marker, and marker values are 1, 0, or -1, or NA, where 0 represents a heterozygous marker, and NA represents missing data.  All data in this matrix must be numeric.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of kinship values between genotypes.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Graebner
</p>


<h3>References</h3>

<p>Endelman JB (2011) Ridge regression and other kernels for genomic selection with R package rrBLUP. The Plant Genome 4:250-255.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genotypes)
Mat(genotypes)
</code></pre>

<hr>
<h2 id='MTK'>
MTK calculator
</h2><span id='topic+MTK'></span>

<h3>Description</h3>

<p>This function calculates the Mean of Transformed Kinships (MTK) of a set of genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MTK(genos, subset, mat = NULL, power = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MTK_+3A_genos">genos</code></td>
<td>

<p>A matrix of genotypes, where each column is one individual, each row is one marker, and marker values are 1, 0, or -1, or NA, where 0 represents a heterozygous marker, and NA represents missing data. Note that this coding is different from the earlier MtkCalc, which cannot handle heterozygous markers.  All data in this matrix must be numeric.
</p>
</td></tr>
<tr><td><code id="MTK_+3A_subset">subset</code></td>
<td>

<p>A vector of genotype names for which to calculate MTK.
</p>
</td></tr>
<tr><td><code id="MTK_+3A_mat">mat</code></td>
<td>

<p>A kinship matrix, if one has already been computed for the population.
</p>
</td></tr>
<tr><td><code id="MTK_+3A_power">power</code></td>
<td>

<p>The transformation that should be made to the kinship matrix, if the MTK criterion is used. If power=1, the kinship matrix is not transformed, if power=2, the kinship matrix is squared, etc.  When the power is higher, this function preferentially eliminates genotypes that are closely related to other genotypes in the population.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The MTK of the specified subset.
</p>


<h3>Note</h3>

<p>The ability to recognize heterozygous markers was included in MTK, resulting in a slightly different genotype coding scheme than the earlier MtkCalc.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Graebner
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genotypes)
MTK(genotypes,subset=colnames(genotypes[1:5]))
</code></pre>

<hr>
<h2 id='MtkCalc'>
MTK calculator (Depreciated)
</h2><span id='topic+MtkCalc'></span>

<h3>Description</h3>

<p>***This function has been superseded by MTK.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MtkCalc(genos, subset, power = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MtkCalc_+3A_genos">genos</code></td>
<td>

<p>A matrix of genotypes, that includes all genotypes that should be used to create the kinship matrix, where each column is one individual, each row is one marker, and marker values are 1, 0, or -1, where 0 represents missing data.  All data in this matrix must be numeric.
</p>
</td></tr>
<tr><td><code id="MtkCalc_+3A_subset">subset</code></td>
<td>

<p>A vector of genotype names for which to calculate MTK.
</p>
</td></tr>
<tr><td><code id="MtkCalc_+3A_power">power</code></td>
<td>

<p>The transformation that should be made to the kinship matrix. If power=1, the kinship matrix is not transformed, if power=2, the kinship matrix is squared, etc.  When the power is higher, this function preferentially eliminates genotypes that are closely related to other specific genotypes.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The MTK of the specified subset.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Graebner
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genotypes)
MtkCalc(genotypes,subset=colnames(genotypes[1:5]))
</code></pre>

<hr>
<h2 id='PicCalc'>
PIC Calculator (Depreciated)
</h2><span id='topic+PicCalc'></span>

<h3>Description</h3>

<p>***This function has been superseded by PIC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PicCalc(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PicCalc_+3A_data">data</code></td>
<td>

<p>A matrix of genotypes, where each column is one individual, each row is one marker, and marker values are 1, 0, or -1, where 0 represents missing data.  All data in this matrix must be numeric.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The mean PIC of all markers included for the given set of genotypes.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Graebner and Alfonso Cuesta-Marcos
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genotypes")
PicCalc(genotypes)
</code></pre>

<hr>
<h2 id='SubsetterMTK'>Genotype Subsetting with PIC - Method One (Depreciated)</h2><span id='topic+SubsetterMTK'></span>

<h3>Description</h3>

<p>***This function has been superseded by CoreSetter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubsetterMTK(genos, save = NULL, power = 10, mat = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SubsetterMTK_+3A_genos">genos</code></td>
<td>

<p>A matrix of genotypes, where each column is one individual, each row is one marker, and marker values are 1, 0, or -1, where 0 represents missing data.  All data in this matrix must be numeric.
</p>
</td></tr>
<tr><td><code id="SubsetterMTK_+3A_save">save</code></td>
<td>

<p>A list of genotype names, corresponding to the column names in the genotype matrix, that will not be eliminated.
</p>
</td></tr>
<tr><td><code id="SubsetterMTK_+3A_power">power</code></td>
<td>

<p>The transformation that should be made to the kinship matrix. If power=1, the kinship matrix is not transformed, if power=2, the kinship matrix is squared, etc.  When the power is higher, this function preferentially eliminates genotypes that are closely related to other specific genotypes.
</p>
</td></tr>
<tr><td><code id="SubsetterMTK_+3A_mat">mat</code></td>
<td>

<p>A list of genotype names, corresponding to the column names in the genotype matrix, that will not be eliminated.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with four columns.  The first column is the importance of a particular genotype to the population's MTK, based on the order in which genotypes were eliminated.  The second column is the name of the genotype. The third and fourth columns are the MTK, and the mean of untransformed kinship values, respectively, of a population that includes the corresponding genotype, plus all genotypes that are more important.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Graebner
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genotypes")
SubsetterMTK(genotypes,save=colnames(genotypes)[c(1,5,9)])
</code></pre>

<hr>
<h2 id='SubsetterPIC'>
Genotype Subsetting with PIC - Method One (Depreciated)
</h2><span id='topic+SubsetterPIC'></span>

<h3>Description</h3>

<p>***This function has been superseded by CoreSetter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubsetterPIC(genos, save = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SubsetterPIC_+3A_genos">genos</code></td>
<td>

<p>A matrix of genotypes, where each column is one individual, each row is one marker, and marker values are 1, 0, or -1, where 0 represents missing data.  All data in this matrix must be numeric.
</p>
</td></tr>
<tr><td><code id="SubsetterPIC_+3A_save">save</code></td>
<td>

<p>A list of genotype names, corresponding to the column names in the genotype matrix, that will not be eliminated.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with three columns.  The first column is the importance of a particular genotype to the population's genetic diversity, based on the order in which genotypes were eliminated.  The second column is the name of the genotype, and the third column is the mean PIC of a population that includes the corresponding genotype, plus all genotypes that are more important.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Graebner and Alfonso Cuesta-Marcos
</p>


<h3>References</h3>

<p>Munoz-Amatrain M, Cuesta-Marcos A, Endelman JB, Comadran J, Bonman JM (2014) The USDA barley core collection: genetic diversity, population structure, and potential for genome-wide association studies. PloS One 9:e94688.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("genotypes")
SubsetterPIC(genotypes,save=colnames(genotypes)[c(1,5,9)])
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
