<!DOCTYPE html><html><head><title>Help for package GeneScape</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GeneScape}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fcsim'><p>fcsim</p></a></li>
<li><a href='#GeneScape'><p>GeneScape</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulation of Single Cell RNA-Seq Data with Complex Structure</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-6-4</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Qi Gao &lt;qi.gao@duke.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulating single cell RNA-seq data with complicated structure. This package is developed based on the Splat method (Zappia, Phipson and Oshlack (2017) &lt;<a href="https://doi.org/10.1186%2Fs13059-017-1305-0">doi:10.1186/s13059-017-1305-0</a>&gt;). 'GeneScape' incorporates additional features to simulate single cell RNA-seq data with complicated differential expression and correlation structures, such as sub-cell-types, correlated genes (pathway genes) and hub genes. </td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS (&ge; 7.3-53.1), corpcor (&ge; 1.6.10), stats</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-13 01:46:17 UTC; qg</td>
</tr>
<tr>
<td>Author:</td>
<td>Qi Gao [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-13 09:00:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='fcsim'>fcsim</h2><span id='topic+fcsim'></span>

<h3>Description</h3>

<p>This function similate differential expression fold change level
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fcsim(n.gene, de.id, fc.loc, fc.scale)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fcsim_+3A_n.gene">n.gene</code></td>
<td>
<p>total number of genes</p>
</td></tr>
<tr><td><code id="fcsim_+3A_de.id">de.id</code></td>
<td>
<p>index of differentially expressed genes</p>
</td></tr>
<tr><td><code id="fcsim_+3A_fc.loc">fc.loc</code></td>
<td>
<p>location parameter for fold change (log-normal distribution)</p>
</td></tr>
<tr><td><code id="fcsim_+3A_fc.scale">fc.scale</code></td>
<td>
<p>scale parameter for fold change (log-normal distribution)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Zappia, L., Phipson, B., &amp; Oshlack, A. (2017). Splatter: Simulation of single-cell RNA sequencing data. Genome Biology, 18(1). https://doi.org/10.1186/s13059-017-1305-0
</p>

<hr>
<h2 id='GeneScape'>GeneScape</h2><span id='topic+GeneScape'></span>

<h3>Description</h3>

<p>This function simulate single cell RNAseq data with complicated differential expression and correlation structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GeneScape(
  nCells = 6000,
  nGroups = NULL,
  groups = NULL,
  lib.size.loc = 9.3,
  lib.size.scale = 0.25,
  de.fc.mat = NULL,
  nGenes = 5000,
  gene.mean.shape = 0.3,
  gene.mean.rate = 0.15,
  gene.means = NULL,
  de.n = 50,
  de.share = NULL,
  de.id = NULL,
  de.fc.loc = 0.7,
  de.fc.scale = 0.2,
  add.sub = FALSE,
  sub.major = NULL,
  sub.prop = 0.1,
  sub.group = NULL,
  sub.de.n = 20,
  sub.de.id = NULL,
  sub.de.common = FALSE,
  sub.de.fc.loc = 0.7,
  sub.de.fc.scale = 0.2,
  add.cor = FALSE,
  cor.n = 4,
  cor.size = 20,
  cor.cor = 0.7,
  cor.id = NULL,
  band.width = 10,
  add.hub = FALSE,
  hub.n = 10,
  hub.size = 20,
  hub.cor = 0.4,
  hub.id = NULL,
  hub.fix = NULL,
  drop = FALSE,
  dropout.location = -2,
  dropout.slope = -1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GeneScape_+3A_ncells">nCells</code></td>
<td>
<p>number of cells</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_ngroups">nGroups</code></td>
<td>
<p>number of cell groups</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_groups">groups</code></td>
<td>
<p>group information for cells</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_lib.size.loc">lib.size.loc</code></td>
<td>
<p>location parameter for library size (log-normal distribution)</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_lib.size.scale">lib.size.scale</code></td>
<td>
<p>scale parameter for library size (log-normal distribution)</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_de.fc.mat">de.fc.mat</code></td>
<td>
<p>differential expression fold change matrix, could be generated by this function</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_ngenes">nGenes</code></td>
<td>
<p>number of genes</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_gene.mean.shape">gene.mean.shape</code></td>
<td>
<p>shape parameter for mean expression level (Gamma distribution)</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_gene.mean.rate">gene.mean.rate</code></td>
<td>
<p>rate parameter for mean expression level (Gamma distribution)</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_gene.means">gene.means</code></td>
<td>
<p>mean gene expression levels</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_de.n">de.n</code></td>
<td>
<p>number of differentially expressed genes in each cell type. Should be a integer or a vector of length nGroups</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_de.share">de.share</code></td>
<td>
<p>number of shared DE genes between neighbor cell types. Should be a vector of length (nGroups - 1)</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_de.id">de.id</code></td>
<td>
<p>the index of genes that are DE across cell types. Should be a list of vectors. 
Each vector corresponds to a cell type. With non-null value of de.id, de.n and de.share would be ignored.</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_de.fc.loc">de.fc.loc</code></td>
<td>
<p>the location parameter for the fold change of DE genes. Should be a number, a vector of length nGroups</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_de.fc.scale">de.fc.scale</code></td>
<td>
<p>the scale parameter for fold change (log-normal distribution). Should be a number or a vector of length nGroups</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_add.sub">add.sub</code></td>
<td>
<p>whether to add sub-cell-types</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_sub.major">sub.major</code></td>
<td>
<p>the major cell types correspond to the sub-cell-types</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_sub.prop">sub.prop</code></td>
<td>
<p>proportion of sub-cell-types in the corresponding major cell type</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_sub.group">sub.group</code></td>
<td>
<p>cell index for sub-cell-types. With non-null sub.group specified, sub.prop would be ignored.</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_sub.de.n">sub.de.n</code></td>
<td>
<p>number of differentially expressed genes in each sub-cell-type compared to the corresponding major cell type. Should be a integer or a vector of length sub.major</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_sub.de.id">sub.de.id</code></td>
<td>
<p>the index of additional differentially expressed genes between sub-cell-types and the corresponding major cell types</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_sub.de.common">sub.de.common</code></td>
<td>
<p>whether the additional differential expression structure should be same for all sub-cell-types</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_sub.de.fc.loc">sub.de.fc.loc</code></td>
<td>
<p>similar to de.fc.loc, but for addtional differentially expressed genes in sub-cell-types</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_sub.de.fc.scale">sub.de.fc.scale</code></td>
<td>
<p>similar to de.fc.scale, but for addtional differentially expressed genes in sub-cell-types</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_add.cor">add.cor</code></td>
<td>
<p>whether to add pathways (correlated genes)</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_cor.n">cor.n</code></td>
<td>
<p>number of pathways included. Should be a integer.</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_cor.size">cor.size</code></td>
<td>
<p>number of correlated genes (length of pathway). Should be a number or a vector of length cor.n</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_cor.cor">cor.cor</code></td>
<td>
<p>correlation parameters</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_cor.id">cor.id</code></td>
<td>
<p>gene index of correlated (pathway) genes. Should be a list of vectors, with each vector represents a pathway. With non-null value of cor.id, cor.n would be ignored.</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_band.width">band.width</code></td>
<td>
<p>No correlation exists if distance of 2 genes are further than band_width in a pathway</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_add.hub">add.hub</code></td>
<td>
<p>whether to add hub genes</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_hub.n">hub.n</code></td>
<td>
<p>number of hub genes included. Should be a integer.</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_hub.size">hub.size</code></td>
<td>
<p>number of genes correlated to the hub gene. Should be a number or a vector of length hub.n</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_hub.cor">hub.cor</code></td>
<td>
<p>correlation parameters between hub genes and their correlated genes</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_hub.id">hub.id</code></td>
<td>
<p>gene index of hub genes. Should be a list of vectors. With non-null value of hub.id, hub.n would be ignored.</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_hub.fix">hub.fix</code></td>
<td>
<p>user defined genes correlated to hub genes (others are randomly selected). Should be a list of vectors of length hub.n or same as hub.id.</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_drop">drop</code></td>
<td>
<p>whether to add dropout</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_dropout.location">dropout.location</code></td>
<td>
<p>dropout mid point (the mean expression level at which the probability is equal to 0.5, same as splat. Could be negative)</p>
</td></tr>
<tr><td><code id="GeneScape_+3A_dropout.slope">dropout.slope</code></td>
<td>
<p>how dropout proportion changes with increasing expression</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Compared to splat method in Splatter R package, this function can fix the number and position of differentially expressed genes, 
have more complicated differential expression structure, add sub-cell-types, correlated genes (AR(1) correlation structure with bound, mimicking pathways) and hub genes.
</p>


<h3>Value</h3>

<p>A list of observed data, true data (without dropout), differential expression rate and hub gene indices.
</p>


<h3>References</h3>

<p>Zappia, L., Phipson, B., &amp; Oshlack, A. (2017). Splatter: Simulation of single-cell RNA sequencing data. Genome Biology, 18(1). https://doi.org/10.1186/s13059-017-1305-0
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
data &lt;- GeneScape()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
