<!DOCTYPE html><html><head><title>Help for package geneHapR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {geneHapR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#geneHapR-package'><p>Gene Haplotype Statistics, Pheno Associations, and Visualization</p></a></li>
<li><a href='#addINFO'><p>Add Infomation to Haplotype Results</p></a></li>
<li><a href='#addPromoter'><p>add promoter to annotation</p></a></li>
<li><a href='#ashaplotype'><p>as.haplotype</p></a></li>
<li><a href='#DataSet'><p>Datasets</p>
<code>gff</code> contains a example of gff file used for test of
visualization mutations on gene model.</a></li>
<li><a href='#displayVarOnGeneModel'><p>Display Variants on Gene Model</p></a></li>
<li><a href='#filter_hap'><p>Filter hap</p></a></li>
<li><a href='#filter_hmp'><p>filter variants in hapmap format</p></a></li>
<li><a href='#filter_plink.pedmap'><p>filter_plink.pedmap</p></a></li>
<li><a href='#filter_table'><p>filter variants stored in table</p></a></li>
<li><a href='#filter_vcf'><p>Filter VCF</p></a></li>
<li><a href='#filterLargep.link'><p>Pre-process of Large VCF File(s)</p></a></li>
<li><a href='#filterLargeVCF'><p>Pre-process of Large VCF File(s)</p></a></li>
<li><a href='#getGenePOS'><p>Get Gene Position</p></a></li>
<li><a href='#getGeneRanges'><p>Get Gene Ranges</p></a></li>
<li><a href='#hap_summary'><p>Summary Hap Results</p></a></li>
<li><a href='#hap2hmp'><p>Convert hapResult object to hapmap (hmp) format,</p>
for interact with other packages</a></li>
<li><a href='#hapDistribution'><p>Display of Geography Distribution</p></a></li>
<li><a href='#hapVsPheno'><p>hapVsPheno</p></a></li>
<li><a href='#hapVsPhenoPerSite'><p>hapVsPhenoPerSite</p></a></li>
<li><a href='#hapVsPhenos'><p>hapVsPhenos</p></a></li>
<li><a href='#import_AccINFO'><p>Import Accession Information from File</p></a></li>
<li><a href='#import_bed'><p>import annotation files in BED format</p></a></li>
<li><a href='#import_gff'><p>Import Annotations from GFF Format File</p></a></li>
<li><a href='#import_hap'><p>Import hapResult/hapSummary</p></a></li>
<li><a href='#import_MultipleAlignment'><p>Import MultipleAlignment Result</p></a></li>
<li><a href='#import_plink.pedmap'><p>import_plink.pedmap</p></a></li>
<li><a href='#import_seqs'><p>Import Sequences</p></a></li>
<li><a href='#import_vcf'><p>Import VCF from File</p></a></li>
<li><a href='#LDheatmap'><p>This function produces a pairwise LD plot.</p></a></li>
<li><a href='#network'><p>Generate Haplotype Net Relationshop with Haplotype Result</p></a></li>
<li><a href='#plink.pedmap2hap'><p>plink.pedmap2hap</p></a></li>
<li><a href='#plotEFF'><p>plotEFF</p></a></li>
<li><a href='#plotHapNet'><p>plotHapNet</p></a></li>
<li><a href='#plotHapTable'><p>plotHapTable</p></a></li>
<li><a href='#plotHapTable2'><p>plotHapTable2</p></a></li>
<li><a href='#seqs2hap'><p>Generate Hap Results from Seqs</p></a></li>
<li><a href='#SetATGas0'><p>Set Position of ATG as Zero</p></a></li>
<li><a href='#siteEFF'><p>Calculation of Sites Effective</p></a></li>
<li><a href='#sites_compar'><p>sites comparison</p></a></li>
<li><a href='#table2hap'><p>table2hap</p></a></li>
<li><a href='#vcf2hap'><p>Generat Haps from VCF</p></a></li>
<li><a href='#write.hap'><p>Save Haplotype Results to Disk</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Haplotype Statistics, Phenotype Association and
Visualization</td>
</tr>
<tr>
<td>Description:</td>
<td>Import genome variants data and perform gene haplotype Statistics, 
  visualization and phenotype association with 'R'. </td>
</tr>
<tr>
<td>biocViews:</td>
<td>NucleosomePositioning, DataImport</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zhang Renliang &lt;zhang_renliang@163.com&gt;</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.4</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>LazyData:</td>
<td>True</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, Biostrings, ggpubr, genetics, GenomicRanges, lolliplot,
maps, methods, IRanges, pegas, reshape2, rlang, rtracklayer,
shiny, shinyjs, stats, stringdist, stringr, tibble, sf, tidyr,
utils, vcfR</td>
</tr>
<tr>
<td>Depends:</td>
<td>ggplot2, R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>mapdata, knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-01 12:29:13 UTC; zhang</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhang Renliang [aut, cre],
  Jia Guanqing [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-01 14:32:40 UTC</td>
</tr>
</table>
<hr>
<h2 id='geneHapR-package'>Gene Haplotype Statistics, Pheno Associations, and Visualization</h2><span id='topic+geneHapR'></span><span id='topic+geneHapR-package'></span>

<h3>Description</h3>

<p>This packages provides a suite for gene(s) haplotype analysis,
phenotypes association and visualization of geography distribution.
</p>


<h3>Details</h3>

<p>The haplotype analysis requires sequences such as fasta which could be
import by <a href="#topic+import_vcf">import_vcf</a>
</p>
<p><strong>A complete list of functions</strong> can be displayed with:
<code>library(help = 'geneHapR')</code>.
</p>
<p><strong>Vignettes</strong> (documentation) can be listed with:
<code>browseVignettes('geneHapR')</code>.
</p>
<p>Several small example <strong>datasets</strong> are included in 'geneHapR'.
<strong>vcf</strong> and <strong>seqs</strong> provide a data set for test of <code>seq2hap()</code> and
<code>seqs2hap()</code>.
<strong>gff</strong> provide a data set for visualization mutations on gene model.
<strong>pheno</strong> provide a data set for phenotype comparison between haplotypes.
<strong>AccINFO</strong> provide a data provide accessions information for
pie plot in hapNet visualization, and information for geography distrbution.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Zhang Renliang <a href="mailto:zhang_renliang@163.com">zhang_renliang@163.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Jia Guanqing <a href="mailto:jiaguanqing@caas.cn">jiaguanqing@caas.cn</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>More documentation for 'geneHapR' could be found at repos under
<a href="https://gitee.com/zhangrenl">gitee</a>
</p>

<hr>
<h2 id='addINFO'>Add Infomation to Haplotype Results</h2><span id='topic+addINFO'></span><span id='topic+sites'></span>

<h3>Description</h3>

<p>add annotations to INFO fields used for <code>plotHapTable()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addINFO(hap,
        tag = "", values = values,
        replace = FALSE, sep = ";")

sites(hap)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addINFO_+3A_hap">hap</code></td>
<td>
<p>object of <code>hapResult</code> or <code>hapSummary</code> class</p>
</td></tr>
<tr><td><code id="addINFO_+3A_tag">tag</code></td>
<td>
<p>tag names, usually is a single word used before &quot;=&quot;</p>
</td></tr>
<tr><td><code id="addINFO_+3A_values">values</code></td>
<td>
<p>annotation for each site.
Length of values must be equal with sites in hapResult</p>
</td></tr>
<tr><td><code id="addINFO_+3A_replace">replace</code></td>
<td>
<p>whether replace origin INFOs in hapResult or not.
Default as FALSE</p>
</td></tr>
<tr><td><code id="addINFO_+3A_sep">sep</code></td>
<td>
<p>a character string to separate the terms.  Not
<code><a href="base.html#topic+NA_character_">NA_character_</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of hapSummary or hapResult class with added/replaced INFOs
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotHapTable">plotHapTable()</a></code>
</p>
<p><code><a href="#topic+plotHapTable">plotHapTable()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("geneHapR_test")

# length of values must be equal with number of sites in hap result
values &lt;- paste0("newInfo",c(1:9))
hapResult &lt;- addINFO(hapResult, tag = "new", values = values, replace = TRUE)

data("geneHapR_test")

# check how many sites were concluded in hapResult/hapSummary
sites(hapResult)

</code></pre>

<hr>
<h2 id='addPromoter'>add promoter to annotation</h2><span id='topic+addPromoter'></span>

<h3>Description</h3>

<p>add promoter to annotation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addPromoter(anno, PromoterLength = 1500, bedFile = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addPromoter_+3A_anno">anno</code></td>
<td>
<p>anotation, imported gff/bed</p>
</td></tr>
<tr><td><code id="addPromoter_+3A_promoterlength">PromoterLength</code></td>
<td>
<p>the length of promoter region, default as 1500</p>
</td></tr>
<tr><td><code id="addPromoter_+3A_bedfile">bedFile</code></td>
<td>
<p>the output bed file name</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data("geneHapR_test")
bed &lt;- addPromoter(gff)
</code></pre>

<hr>
<h2 id='ashaplotype'>as.haplotype</h2><span id='topic+ashaplotype'></span><span id='topic+as.haplotype'></span>

<h3>Description</h3>

<p>convert <code>hapSummary</code> or <code>hapResult</code> class into <code>haplotype</code> class (pegas)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.haplotype(hap)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ashaplotype_+3A_hap">hap</code></td>
<td>
<p>object of <code>hapSummary</code> or <code>hapResult</code> class</p>
</td></tr>
</table>


<h3>Value</h3>

<p>haplotype class
</p>


<h3>Note</h3>

<p>It's not advised for <code>hapSummary</code> or <code>hapResult</code> with indels, due to indels will
convert to SNPs with equal length of each indel.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("geneHapR_test")
hap &lt;- as.haplotype(hapResult)
hapSummary &lt;- hap_summary(hapResult)
hap &lt;- as.haplotype(hapSummary)
</code></pre>

<hr>
<h2 id='DataSet'>Datasets
<code>gff</code> contains a example of gff file used for test of
visualization mutations on gene model.</h2><span id='topic+DataSet'></span><span id='topic+gff'></span><span id='topic+seqs'></span><span id='topic+hapResult'></span><span id='topic+gt.geno'></span><span id='topic+china'></span><span id='topic+pheno'></span><span id='topic+vcf'></span><span id='topic+AccINFO'></span>

<h3>Description</h3>

<p><code>pheno</code> contains a simulated test pheno data used for test of comparison
between different haps
</p>
<p><code>vcf</code>, a vcfR object provide a data set for test of <code>seq2hap()</code>.
<code>vcf</code> contains indels, snps, biallelic sites and multiallelic sites.
</p>
<p><code>AccINFO</code> a data.frame provide addtional information of accessions,
including accession type, source and location.
</p>

<hr>
<h2 id='displayVarOnGeneModel'>Display Variants on Gene Model</h2><span id='topic+displayVarOnGeneModel'></span>

<h3>Description</h3>

<p>show variants on gene model using hapSummary and gene annotations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>displayVarOnGeneModel(
  hapSummary,
  gff,
  Chr,
  startPOS,
  endPOS,
  type = "pin",
  cex = 0.7,
  CDS_h = 0.05,
  fiveUTR_h = 0.02,
  threeUTR_h = 0.01,
  geneElement = geneElement,
  hap
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="displayVarOnGeneModel_+3A_hapsummary">hapSummary</code>, <code id="displayVarOnGeneModel_+3A_hap">hap</code></td>
<td>
<p>haplotype result</p>
</td></tr>
<tr><td><code id="displayVarOnGeneModel_+3A_gff">gff</code></td>
<td>
<p>gff</p>
</td></tr>
<tr><td><code id="displayVarOnGeneModel_+3A_chr">Chr</code></td>
<td>
<p>the chromosome name.
If missing, the first element in the hapSummary will be used</p>
</td></tr>
<tr><td><code id="displayVarOnGeneModel_+3A_startpos">startPOS</code></td>
<td>
<p>If missing, will use the min position in hapSummary</p>
</td></tr>
<tr><td><code id="displayVarOnGeneModel_+3A_endpos">endPOS</code></td>
<td>
<p>If missing, will use the max position in hapSummary</p>
</td></tr>
<tr><td><code id="displayVarOnGeneModel_+3A_type">type</code></td>
<td>
<p>character. Could be &quot;circle&quot;, &quot;pie&quot;, &quot;pin&quot;,
&quot;pie.stack&quot; or &quot;flag&quot;</p>
</td></tr>
<tr><td><code id="displayVarOnGeneModel_+3A_cex">cex</code></td>
<td>
<p>a numeric control the size of circle</p>
</td></tr>
<tr><td><code id="displayVarOnGeneModel_+3A_cds_h">CDS_h</code>, <code id="displayVarOnGeneModel_+3A_fiveutr_h">fiveUTR_h</code>, <code id="displayVarOnGeneModel_+3A_threeutr_h">threeUTR_h</code></td>
<td>
<p>The height of CDS 5'UTR and 3'UTR
in gene model</p>
</td></tr>
<tr><td><code id="displayVarOnGeneModel_+3A_geneelement">geneElement</code></td>
<td>
<p>ploted elements, eg.: c(&quot;CDS&quot;,&quot;five_prime_UTR&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("geneHapR_test")
hapSummary &lt;- hap_summary(hapResult)
displayVarOnGeneModel(hapSummary, gff,
                      startPOS = 4100,
                      endPOS = 8210,
                      cex = 0.75)

</code></pre>

<hr>
<h2 id='filter_hap'>Filter hap</h2><span id='topic+filter_hap'></span>

<h3>Description</h3>

<p>filter hapResult or hapSummary by remove
positions or accessions or haplotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_hap(hap,
           rm.mode = c("position", "accession", "haplotype", "freq"),
           position.rm = position.rm,
           accession.rm = accession.rm,
           haplotype.rm = haplotype.rm,
           freq.min = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_hap_+3A_hap">hap</code></td>
<td>
<p>object of hapSummary or hapResult class</p>
</td></tr>
<tr><td><code id="filter_hap_+3A_rm.mode">rm.mode</code></td>
<td>
<p>filter mode, one of &quot;position&quot;, &quot;accession&quot;, &quot;haplotype&quot;</p>
</td></tr>
<tr><td><code id="filter_hap_+3A_position.rm">position.rm</code></td>
<td>
<p>numeric vector contains positions need to be removed</p>
</td></tr>
<tr><td><code id="filter_hap_+3A_accession.rm">accession.rm</code></td>
<td>
<p>character vector contains accessions need to be removed,
only hapResult can be filtered by accessions</p>
</td></tr>
<tr><td><code id="filter_hap_+3A_haplotype.rm">haplotype.rm</code></td>
<td>
<p>character vector contains haplotypes need to be removed</p>
</td></tr>
<tr><td><code id="filter_hap_+3A_freq.min">freq.min</code></td>
<td>
<p>numeric, hapltypes with accessions number less than freq.min
will be removed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>hapSummary or hapResult depend input
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("geneHapR_test")
hap &lt;- filter_hap(hapResult,
                  rm.mode = c("position", "accession", "haplotype", "freq"),
                  position.rm = c(4879, 4950),
                  accession.rm = c("C1", "C9"),
                  haplotype.rm = c("H009", "H008"),
                  freq.min = 5)
</code></pre>

<hr>
<h2 id='filter_hmp'>filter variants in hapmap format</h2><span id='topic+filter_hmp'></span>

<h3>Description</h3>

<p>filter variants in hapmap format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_hmp(
  x,
  mode = c("POS", "type", "both"),
  Chr = Chr,
  start = start,
  end = end,
  gff = gff,
  type = type,
  cusTyp = cusTyp,
  geneID = geneID
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_hmp_+3A_x">x</code></td>
<td>
<p>genotype dataset in hapmap format, object of data.frame class</p>
</td></tr>
<tr><td><code id="filter_hmp_+3A_mode">mode</code></td>
<td>
<p>filter mode, one of &quot;POS&quot;, &quot;type&quot;, &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_hmp_+3A_chr">Chr</code></td>
<td>
<p>chromosome name, needed if mode set to &quot;POS&quot; or &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_hmp_+3A_start">start</code></td>
<td>
<p>start position, needed if mode set to &quot;POS&quot; or &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_hmp_+3A_end">end</code></td>
<td>
<p>end position, needed if mode set to &quot;POS&quot; or &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_hmp_+3A_gff">gff</code></td>
<td>
<p>object of GRanges class, genome annotations imported by
<code>import_gff()</code></p>
</td></tr>
<tr><td><code id="filter_hmp_+3A_type">type</code></td>
<td>
<p>filter type, needed if mode set to &quot;type&quot; or &quot;both&quot;,
one of &quot;CDS&quot;, &quot;exon&quot;, &quot;gene&quot;, &quot;genome&quot;, &quot;custom&quot;,
if <code>type</code> was set to &quot;custom&quot;, then <code>custom_type</code> is needed.</p>
</td></tr>
<tr><td><code id="filter_hmp_+3A_custyp">cusTyp</code></td>
<td>
<p>character vector, custom filter type,
needed if <code>type</code> set to &quot;custom&quot;</p>
</td></tr>
<tr><td><code id="filter_hmp_+3A_geneid">geneID</code></td>
<td>
<p>gene ID</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># create a dataset of genotype in hapmap format
hmp &lt;- hap2hmp(hapResult);

# example
hmp &lt;- filter_hmp(hmp, mode = "POS",
                      Chr = "scaffold_1", start = 4100, end = 5000)

</code></pre>

<hr>
<h2 id='filter_plink.pedmap'>filter_plink.pedmap</h2><span id='topic+filter_plink.pedmap'></span>

<h3>Description</h3>

<p>used for filtration of p.link
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_plink.pedmap(x,
                    mode = c("POS", "type", "both"),
                    Chr = Chr, start = start, end = end,
                    gff = gff, type = type, cusTyp = cusTyp,
                    geneID = geneID)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_plink.pedmap_+3A_x">x</code></td>
<td>
<p>a list stored the p.link information</p>
</td></tr>
<tr><td><code id="filter_plink.pedmap_+3A_mode">mode</code></td>
<td>
<p>filtration mode, one of c(&quot;POS&quot;, &quot;type&quot;, &quot;both&quot;)</p>
</td></tr>
<tr><td><code id="filter_plink.pedmap_+3A_chr">Chr</code></td>
<td>
<p>the chromosome name, need if mode set as POS or both</p>
</td></tr>
<tr><td><code id="filter_plink.pedmap_+3A_start">start</code>, <code id="filter_plink.pedmap_+3A_end">end</code></td>
<td>
<p>numeric, the range of filtration, and the start should smaller than end</p>
</td></tr>
<tr><td><code id="filter_plink.pedmap_+3A_gff">gff</code></td>
<td>
<p>the imported gff object</p>
</td></tr>
<tr><td><code id="filter_plink.pedmap_+3A_type">type</code></td>
<td>
<p>should be in <code>unique(gff$type)</code>, usually as &quot;CDS&quot;, &quot;genome&quot;.</p>
</td></tr>
<tr><td><code id="filter_plink.pedmap_+3A_custyp">cusTyp</code></td>
<td>
<p>if <code>type</code> set as custom, then <code>cusTyp</code> is needed</p>
</td></tr>
<tr><td><code id="filter_plink.pedmap_+3A_geneid">geneID</code></td>
<td>
<p>geneID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list, similar with <code>x</code>, but filtered
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
   pedfile &lt;- system.file("extdata",
                          "snp3kvars-CHR8-25947258-25951166-plink.ped",
                          package = "geneHapR")
   mapfile &lt;- system.file("extdata",
                          "snp3kvars-CHR8-25947258-25951166-plink.map",
                          package = "geneHapR")
   p.link &lt;- import_plink.pedmap(pedfile = pedfile, mapfile = mapfile,
                                 sep_map = "\t", sep_ped = "\t")
   p.link &lt;- filter_plink.pedmap(p.link, mode = "POS",
                                 Chr = "chr08", start = 25948004,
                                 end = 25949944)
   hapResult &lt;- plink.pedmap2hap(p.link, hapPrefix = "H",
                                 hetero_remove = TRUE,
                                 na_drop = TRUE)

</code></pre>

<hr>
<h2 id='filter_table'>filter variants stored in table</h2><span id='topic+filter_table'></span>

<h3>Description</h3>

<p>filter variants stored in table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_table(
  x,
  mode = c("POS", "type", "both"),
  Chr = Chr,
  start = start,
  end = end,
  gff = gff,
  type = type,
  cusTyp = cusTyp,
  geneID = geneID
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_table_+3A_x">x</code></td>
<td>
<p>genotype dataset in hapmap format, object of data.frame class</p>
</td></tr>
<tr><td><code id="filter_table_+3A_mode">mode</code></td>
<td>
<p>filter mode, one of &quot;POS&quot;, &quot;type&quot;, &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_table_+3A_chr">Chr</code></td>
<td>
<p>chromosome name, needed if mode set to &quot;POS&quot; or &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_table_+3A_start">start</code></td>
<td>
<p>start position, needed if mode set to &quot;POS&quot; or &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_table_+3A_end">end</code></td>
<td>
<p>end position, needed if mode set to &quot;POS&quot; or &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_table_+3A_gff">gff</code></td>
<td>
<p>object of GRanges class, genome annotations imported by
<code>import_gff()</code></p>
</td></tr>
<tr><td><code id="filter_table_+3A_type">type</code></td>
<td>
<p>filter type, needed if mode set to &quot;type&quot; or &quot;both&quot;,
one of &quot;CDS&quot;, &quot;exon&quot;, &quot;gene&quot;, &quot;genome&quot;, &quot;custom&quot;,
if <code>type</code> was set to &quot;custom&quot;, then <code>custom_type</code> is needed.</p>
</td></tr>
<tr><td><code id="filter_table_+3A_custyp">cusTyp</code></td>
<td>
<p>character vector, custom filter type,
needed if <code>type</code> set to &quot;custom&quot;</p>
</td></tr>
<tr><td><code id="filter_table_+3A_geneid">geneID</code></td>
<td>
<p>gene ID</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># example
data("geneHapR_test")
table &lt;- filter_table(gt.geno, mode = "POS",
                      Chr = "scaffold_1", start = 4100, end = 5000)
</code></pre>

<hr>
<h2 id='filter_vcf'>Filter VCF</h2><span id='topic+filter_vcf'></span>

<h3>Description</h3>

<p>filter VCF by GFF annotation or by position or both
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_vcf(vcf, gff = gff,
           mode = c("POS", "type", "both"),
           Chr = Chr, start = start, end = end,
           type = c("CDS", "exon", "gene", "genome", "custom"),
           cusTyp = c("CDS", "five_prime_UTR", "three_prime_UTR"),
           geneID = geneID)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_vcf_+3A_vcf">vcf</code></td>
<td>
<p>object of vcfR class, VCF file imported by <code>import_vcf()</code></p>
</td></tr>
<tr><td><code id="filter_vcf_+3A_gff">gff</code></td>
<td>
<p>object of GRanges class, genome annotations imported by
<code>import_gff()</code></p>
</td></tr>
<tr><td><code id="filter_vcf_+3A_mode">mode</code></td>
<td>
<p>filter mode, one of &quot;POS&quot;, &quot;type&quot;, &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_vcf_+3A_chr">Chr</code></td>
<td>
<p>chromosome name, needed if mode set to &quot;POS&quot; or &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_vcf_+3A_start">start</code></td>
<td>
<p>start position, needed if mode set to &quot;POS&quot; or &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_vcf_+3A_end">end</code></td>
<td>
<p>end position, needed if mode set to &quot;POS&quot; or &quot;both&quot;</p>
</td></tr>
<tr><td><code id="filter_vcf_+3A_type">type</code></td>
<td>
<p>filter type, needed if mode set to &quot;type&quot; or &quot;both&quot;,
one of &quot;CDS&quot;, &quot;exon&quot;, &quot;gene&quot;, &quot;genome&quot;, &quot;custom&quot;,
if <code>type</code> was set to &quot;custom&quot;, then <code>custom_type</code> is needed.</p>
</td></tr>
<tr><td><code id="filter_vcf_+3A_custyp">cusTyp</code></td>
<td>
<p>character vector, custom filter type,
needed if <code>type</code> set to &quot;custom&quot;</p>
</td></tr>
<tr><td><code id="filter_vcf_+3A_geneid">geneID</code></td>
<td>
<p>gene ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vcfR
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# filtet vcf
data("geneHapR_test")
vcf_f1 &lt;- filter_vcf(vcf, mode = "POS",
                    Chr = "scaffold_1",
                    start = 4300, end = 5890)

vcf_f2 &lt;- filter_vcf(vcf, mode = "type",
                    gff = gff,
                    type = "CDS")
vcf_f3 &lt;- filter_vcf(vcf, mode = "both",
                    Chr = "scaffold_1",
                    start = 4300, end = 5890,
                    gff = gff,
                    type = "CDS")

</code></pre>

<hr>
<h2 id='filterLargep.link'>Pre-process of Large VCF File(s)</h2><span id='topic+filterLargep.link'></span><span id='topic+filterLargeP.link'></span>

<h3>Description</h3>

<p>Filter/extract one or multiple gene(s)/range(s) from a large
<code>p.link</code> file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterLargeP.link(
  root,
  rootOut = rootOut,
  Chr = Chr,
  POS = NULL,
  start = start,
  end = end,
  override = TRUE,
  sep = "\t"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterLargep.link_+3A_root">root</code></td>
<td>
<p>The file name without suffix. This function only support p.link file
format stored in &quot;map&quot; and &quot;ped&quot; format, the file names after removed suffix
should be same with each other.</p>
</td></tr>
<tr><td><code id="filterLargep.link_+3A_rootout">rootOut</code></td>
<td>
<p>Path(s) of output <code>p.link</code> file stored in &quot;ped&amp;map&quot; format.</p>
</td></tr>
<tr><td><code id="filterLargep.link_+3A_chr">Chr</code></td>
<td>
<p>a single CHROM name or CHROM names vector.</p>
</td></tr>
<tr><td><code id="filterLargep.link_+3A_pos">POS</code>, <code id="filterLargep.link_+3A_start">start</code>, <code id="filterLargep.link_+3A_end">end</code></td>
<td>
<p>provide the chromosome name should be extract from orignal p.link dataset.
<code>POS</code>: a vector consist with start and end position, eg.: <code>c(1,200)</code>
indicates 3 ranges (1~200, 300~500 and 300~400).
if <code>POS</code> is <code>NULL</code>, <code>start</code> and <code>end</code> are needed.</p>
</td></tr>
<tr><td><code id="filterLargep.link_+3A_override">override</code></td>
<td>
<p>whether override existed file or not, default as <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="filterLargep.link_+3A_sep">sep</code></td>
<td>
<p>a character indicate the separation of map and ped file, default is <code style="white-space: pre;">&#8288;\t&#8288;</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This package import P.link files. However, import a large P.link file is time and
memory consuming. It's suggested that extract variants in target
range with <code>filterLargeP.link()</code> before identification of haplotype.
</p>
<p>When filter/extract multi genes/ranges, the parameter of <code>Chr</code> and <code>POS</code>
must have equal length. Results will save to a single file if the user
provide a single file path or save to multiple P.link file(s) when a equal length
vector consist with file paths is provided.
</p>


<h3>Value</h3>

<p>No return value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # The filteration of P.link of regular size should be done with `filter_plink.pedmap()`.
 # however, here, we use a mini vcf instead just for example and test

 pedfile &lt;- system.file("extdata",
                        "snp3kvars-CHR8-25947258-25951166-plink.ped",
                        package = "geneHapR")
 mapfile &lt;- system.file("extdata",
                        "snp3kvars-CHR8-25947258-25951166-plink.map",
                        package = "geneHapR")
 oldDir &lt;- getwd()
 temp_dir &lt;- tempdir()
 if(! dir.exists(temp_dir))
   dir.create(temp_dir)
 setwd(temp_dir)
 file.copy(pedfile, "test.ped")
 file.copy(mapfile, "test.map")

 # extract a single gene/range from large vcf
 filterLargeP.link(root = "test",
                   rootOut = "filtered_test",
                   Chr = "scaffold_1", POS = c(4300,5000), override = TRUE)

setwd(oldDir)

# delete temp_dir
unlink(temp_dir, recursive = TRUE)

</code></pre>

<hr>
<h2 id='filterLargeVCF'>Pre-process of Large VCF File(s)</h2><span id='topic+filterLargeVCF'></span>

<h3>Description</h3>

<p>Filter/extract one or multiple gene(s)/range(s) from a large
<code style="white-space: pre;">&#8288;*.vcf/*.vcf.gz&#8288;</code> file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterLargeVCF(VCFin = VCFin, VCFout = VCFout,
                Chr = Chr,
                POS = NULL,
                start = start,
                end = end,
                override = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterLargeVCF_+3A_vcfin">VCFin</code></td>
<td>
<p>Path of input <code style="white-space: pre;">&#8288;*.vcf/*.vcf.gz&#8288;</code> file.</p>
</td></tr>
<tr><td><code id="filterLargeVCF_+3A_vcfout">VCFout</code></td>
<td>
<p>Path(s) of output <code style="white-space: pre;">&#8288;*.vcf/*.vcf.gz&#8288;</code> file.</p>
</td></tr>
<tr><td><code id="filterLargeVCF_+3A_chr">Chr</code></td>
<td>
<p>a single CHROM name or CHROM names vector.</p>
</td></tr>
<tr><td><code id="filterLargeVCF_+3A_pos">POS</code>, <code id="filterLargeVCF_+3A_start">start</code>, <code id="filterLargeVCF_+3A_end">end</code></td>
<td>
<p>provide the range should be extract from orignal vcf.
<code>POS</code>: a vector consist with start and end position or a list
with length equal to <code>Chr</code>, eg.: <code>list(c(1,200), c(300,500), c(300,400))</code>
indicates 3 ranges (1~200, 300~500 and 300~400).
if <code>POS</code> is <code>NULL</code>, <code>start</code> and <code>end</code> are needed, eg.:
<code>start = c(1, 30)</code> and <code>end = c(200, 150)</code> indicates 2 ranges
(1~200 and 30~150)</p>
</td></tr>
<tr><td><code id="filterLargeVCF_+3A_override">override</code></td>
<td>
<p>whether override existed file or not, default as <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This package import VCF files with 'vcfR' which is more efficient to
import/manipulate VCF files in 'R'. However, import a large VCF file is time and
memory consuming. It's suggested that filter/extract variants in target
range with <code>filterLargeVCF()</code>.
</p>
<p>When filter/extract multi genes/ranges, the parameter of <code>Chr</code> and <code>POS</code>
must have equal length. Results will save to a single file if the user
provide a single file path or save to multiple VCF file(s) when a equal length
vector consist with file paths is provided.
</p>
<p>However, if you have hundreds gene/ranges need to extract from very
large VCF file(s), it's prefer to process with other linux tools in a script
on server, such as: 'vcftools' and 'bcftools'.
</p>


<h3>Value</h3>

<p>No return value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # The filteration of small vcf should be done with `filter_vcf()`.
 # however, here, we use a mini vcf instead just for example and test.

 vcfPath &lt;- system.file("extdata", "var.vcf.gz", package = "geneHapR")

 oldDir &lt;- getwd()
 temp_dir &lt;- tempdir()
 if(! dir.exists(temp_dir))
   dir.create(temp_dir)
 setwd(temp_dir)
 # extract a single gene/range from large vcf
 filterLargeVCF(VCFin = vcfPath, VCFout = "filtered.vcf.gz",
                Chr = "scaffold_1", POS = c(4300,5000), override = TRUE)

 # extract multi genes/ranges from large vcf
 filterLargeVCF(VCFin = vcfPath,
                VCFout = c("filtered1.vcf.gz",
                           "filtered2.vcf.gz",
                           "filtered3.vcf.gz"),
                Chr = rep("scaffold_1", 3),
                POS = list(c(4300, 5000),
                           c(5000, 6000),
                           c(5000, 7000)),
                override = TRUE)

setwd(oldDir)

</code></pre>

<hr>
<h2 id='getGenePOS'>Get Gene Position</h2><span id='topic+getGenePOS'></span>

<h3>Description</h3>

<p>Get Gene Position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGenePOS(gff= gff,
           geneID = geneID,
           type = type,
           gffTermContaingeneID = "Parent")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGenePOS_+3A_gff">gff</code></td>
<td>
<p>imported gff</p>
</td></tr>
<tr><td><code id="getGenePOS_+3A_geneid">geneID</code></td>
<td>
<p>target geneID</p>
</td></tr>
<tr><td><code id="getGenePOS_+3A_type">type</code></td>
<td>
<p>vector consist with one or more types in gff</p>
</td></tr>
<tr><td><code id="getGenePOS_+3A_gfftermcontaingeneid">gffTermContaingeneID</code></td>
<td>
<p>which term contains the geneID in your gff,
defalt is Parent</p>
</td></tr>
</table>


<h3>Value</h3>

<p>named vectors contains start, end and strand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("geneHapR_test")
genePOS &lt;- getGenePOS(gff = gff,
           geneID = "test1G0387",
           type = "CDS",
           gffTermContaingeneID = "Parent")

</code></pre>

<hr>
<h2 id='getGeneRanges'>Get Gene Ranges</h2><span id='topic+getGeneRanges'></span>

<h3>Description</h3>

<p>Get Gene Ranges
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGeneRanges(gff= gff,
             geneID = geneID,
             type = type,
             gffTermContaingeneID = "Parent")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGeneRanges_+3A_gff">gff</code></td>
<td>
<p>imported gff</p>
</td></tr>
<tr><td><code id="getGeneRanges_+3A_geneid">geneID</code></td>
<td>
<p>target geneID</p>
</td></tr>
<tr><td><code id="getGeneRanges_+3A_type">type</code></td>
<td>
<p>vector consist with one or more types in gff</p>
</td></tr>
<tr><td><code id="getGeneRanges_+3A_gfftermcontaingeneid">gffTermContaingeneID</code></td>
<td>
<p>which term contains the geneID in your gff,
defalt is Parent</p>
</td></tr>
</table>


<h3>Value</h3>

<p>GRanges
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("geneHapR_test")
geneRanges &lt;- getGeneRanges(gff = gff,
                            geneID = "test1G0387",
                            type = "CDS",
                            gffTermContaingeneID = "Parent")
</code></pre>

<hr>
<h2 id='hap_summary'>Summary Hap Results</h2><span id='topic+hap_summary'></span>

<h3>Description</h3>

<p>A function used for summarize hapResult to visualization and calculation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hap_summary(hap,
            hapPrefix = "H",
            file = file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hap_summary_+3A_hap">hap</code></td>
<td>
<p>object of hapResult class, generated by <code>vcf2hap()</code>
or <code>seqs2hap</code> or <code>import_hap()</code></p>
</td></tr>
<tr><td><code id="hap_summary_+3A_happrefix">hapPrefix</code></td>
<td>
<p>prefix of hap names, default as &quot;H&quot;</p>
</td></tr>
<tr><td><code id="hap_summary_+3A_file">file</code></td>
<td>
<p>file path where to save the hap summary result.
If missing, nothing will be saved to disk.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is suggested to use the result of <code>vcf2hap()</code> or <code>seqs2hap()</code>
as input directly.
However the user can import previously hap result from local file
with <code>import_hap()</code>
</p>


<h3>Value</h3>

<p>hapSummary, first four rows are fixed to meta information: CHR, POS, INFO, ALLELE
Hap names were placed in first column, Accessions and freqs were placed at the last two columns.
</p>


<h3>Note</h3>

<p>If the user have changed the default <code>hapPrefix</code> in <code>vcf2hap()</code> or <code>seqs2hap()</code>,
then the parameter <code>hapPrefix</code> is needed.
Furthermore, a multi-letter prefix of hap names is possible.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("geneHapR_test")
hapSummary &lt;- hap_summary(hapResult, hapPrefix = "H")
</code></pre>

<hr>
<h2 id='hap2hmp'>Convert hapResult object to hapmap (hmp) format,
for interact with other packages</h2><span id='topic+hap2hmp'></span><span id='topic+hmp2hap'></span>

<h3>Description</h3>

<p>Convert hapResult object to hapmap (hmp) format,
for interact with other packages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hap2hmp(hap)

hmp2hap(hmp, hapPrefix = "H", hetero_remove = TRUE, na_drop = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hap2hmp_+3A_hap">hap</code></td>
<td>
<p>object of &quot;hapResult&quot; class</p>
</td></tr>
<tr><td><code id="hap2hmp_+3A_hmp">hmp</code></td>
<td>
<p>object of &quot;data.frame&quot; class in hapmap format</p>
</td></tr>
<tr><td><code id="hap2hmp_+3A_happrefix">hapPrefix</code></td>
<td>
<p>prefix of haplotype names</p>
</td></tr>
<tr><td><code id="hap2hmp_+3A_hetero_remove">hetero_remove</code></td>
<td>
<p>whether remove accessions contains hyb-sites, Character not A T C G</p>
</td></tr>
<tr><td><code id="hap2hmp_+3A_na_drop">na_drop</code></td>
<td>
<p>whether drop accessions contains missing data (&quot;N&quot;, &quot;NA&quot;, &quot;.&quot;)</p>
</td></tr>
<tr><td><code id="hap2hmp_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="#topic+table2hap">table2hap</a></code>
</p>

<dl>
<dt><code>x</code></dt><dd><p>a data.frame contains variants information.
The first file column are fix as Chrome name, position, reference nuclieotide,
alter nuclieotide and INFO. Accession genotype should be in followed columns.
&quot;-&quot; will be treated as Indel. &quot;.&quot; and &quot;N&quot; will be treated as missing data.
Heterozygotes should be &quot;A/T&quot;, &quot;AAA/A&quot;</p>
</dd>
<dt><code>pad</code></dt><dd><p>The number length in haplotype names should be extend to.</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame in hapmap format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("geneHapR_test")
hmp &lt;- hap2hmp(hapResult)
hap &lt;- hmp2hap(hmp)

</code></pre>

<hr>
<h2 id='hapDistribution'>Display of Geography Distribution</h2><span id='topic+hapDistribution'></span>

<h3>Description</h3>

<p>show distribution of intereted haplotypes on maps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapDistribution(
  hap,
  AccINFO,
  LON.col,
  LAT.col,
  hapNames,
  database = "world",
  regions = ".",
  hap.color = hap.color,
  zColours = zColours,
  legend = TRUE,
  symbolSize = 1,
  symbol.lim = c(1, 10),
  ratio = 1,
  cex.legend = 0.8,
  lwd.pie = 1,
  borderCol.pie = NA,
  lty.pie = 1,
  showlabel = TRUE,
  label.col = "black",
  label.cex = 0.8,
  label.font = 1,
  label.adj = c(0.5, 0.5),
  map.fill.color = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hapDistribution_+3A_hap">hap</code></td>
<td>
<p>an object of hapResult class</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_accinfo">AccINFO</code></td>
<td>
<p>a data.frame contains accession information</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_lon.col">LON.col</code>, <code id="hapDistribution_+3A_lat.col">LAT.col</code></td>
<td>
<p>column names of
longitude(<code>LON.col</code>) and latitude(<code>LAT.col</code>)</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_hapnames">hapNames</code></td>
<td>
<p>haplotype names used for display</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_database">database</code></td>
<td>

<p>character string naming a geographical database, a list of
<code>x</code>, <code>y</code>, and <code>names</code> obtained from a previous call to
<code>map</code> or a spatial object of class <code>SpatialPolygons</code> or <code>SpatialLines</code>.
The string choices include a <code><a href="maps.html#topic+world">world</a></code> map,
three USA databases (<code><a href="maps.html#topic+usa">usa</a></code>, <code><a href="maps.html#topic+state">state</a></code>,
<code><a href="maps.html#topic+county">county</a></code>), and more (type <code>help(package='maps')</code> to see the package index). If the requied database is in a different package that has not been attached, the string may be started with &quot;packagename::&quot;. The
location of the map databases may be overridden by setting the
<code>R_MAP_DATA_DIR</code> environment variable.
</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_regions">regions</code></td>
<td>

<p>character vector that names the polygons to draw.
Each database is composed of a collection of polygons, and each polygon has
a unique name.
When a region is composed of more than one polygon, the individual polygons
have the name of the region, followed by a colon and a qualifier,
as in <code>michigan:north</code> and <code>michigan:south</code>.
Each element of <code>regions</code> is matched
against the polygon names in the database and, according to
<code>exact</code>, a subset is selected for drawing. The regions may also be 
defined using (perl) regular expressions. This makes it possible to use 
'negative' expressions like <code>"Norway(?!:Svalbard)"</code>, which means Norway
and all islands except Svalbard. All entries are case insensitive.
The default selects all polygons in the database.
</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_hap.color">hap.color</code>, <code id="hapDistribution_+3A_zcolours">zColours</code></td>
<td>
<p>colors to apply to the pie section for each attribute column,
&quot;zColours&quot; will be detached in future.</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_legend">legend</code></td>
<td>
<p>a keyword specified the position of legend, one of
&quot;bottomright&quot;, &quot;bottom&quot;, &quot;bottomleft&quot;, &quot;left&quot;, &quot;topleft&quot;, &quot;top&quot;, &quot;topright&quot;, &quot;right&quot; and &quot;center&quot;;
or a numeric vector of length two contains x,y coordinate of the legend</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_symbolsize">symbolSize</code></td>
<td>
<p>a numeric specified the symbol size.
It will be detached in future. Please use &quot;symbol.lim&quot; instead.</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_symbol.lim">symbol.lim</code></td>
<td>
<p>a numeric vector give the maximum and minimum size  of each symbol</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_ratio">ratio</code></td>
<td>
<p>the ratio of Y to N in the output map, set to 1 as default</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_cex.legend">cex.legend</code></td>
<td>
<p>character expansion factor for legend relative to current <code>par("cex")</code></p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_lwd.pie">lwd.pie</code></td>
<td>
<p>line width of the pies</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_bordercol.pie">borderCol.pie</code></td>
<td>
<p>The color of pie's border, default is NA, which means no border will be plotted</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_lty.pie">lty.pie</code></td>
<td>
<p>the line type of pie border</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_showlabel">showlabel</code></td>
<td>
<p>a bool vector indicates whether show the labels
which represens number of individuals. Default as TRUE.</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_label.col">label.col</code></td>
<td>
<p>color of the labels, default as &quot;black&quot;</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_label.cex">label.cex</code></td>
<td>
<p>a number indicates the text size in label, default as 0.8</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_label.font">label.font</code></td>
<td>
<p>Font of label, 1 for normal, 2 for bold, 3 for italica, 4 for bold-italica</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_label.adj">label.adj</code></td>
<td>
<p>the position of label, default as c(0.5, 0.5)</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_map.fill.color">map.fill.color</code></td>
<td>
<p>vector of colors. If fill is FALSE, the first color is
used for plotting all lines, and any other colors are ignored. Otherwise,
the colors are matched one-one with the polygons that get selected by the
region argument (and are reused cyclically, if necessary). If fill = TRUE,
the default boundary line colour is given by par(&quot;fg&quot;). To change this,
you can use the border argument (see '...'). A color of NA causes the
corresponding region to be deleted from the list of polygons to be drawn.
Polygon colors are assigned after polygons are deleted due to values of the
xlim and ylim arguments</p>
</td></tr>
<tr><td><code id="hapDistribution_+3A_...">...</code></td>
<td>
<p>Extra arguments passed to <code>polygon</code> or <code>lines</code>. Of particular interest may be the options <code>border</code> and<code>lty</code> that control the color and line type of the polygon borders when <code>fill = TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("geneHapR_test")
hapDistribution(hapResult,
                AccINFO = AccINFO,
                LON.col = "longitude",
                LAT.col = "latitude",
                hapNames = c("H001", "H002", "H003"))

</code></pre>

<hr>
<h2 id='hapVsPheno'>hapVsPheno</h2><span id='topic+hapVsPheno'></span>

<h3>Description</h3>

<p>hapVsPheno
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapVsPheno(
  hap,
  pheno,
  phenoName,
  hapPrefix = "H",
  title = "",
  comparisons = comparisons,
  method = "t.test",
  method.args = list(),
  symnum.args = list(),
  mergeFigs = FALSE,
  angle = angle,
  hjust = hjust,
  vjust = vjust,
  minAcc = minAcc,
  freq.min = freq.min,
  outlier.rm = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hapVsPheno_+3A_hap">hap</code></td>
<td>
<p>object of hapResult class, generate with<code>vcf2hap()</code> or
<code>seqs2hap()</code></p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_pheno">pheno</code></td>
<td>
<p>object of data.frame class, imported by <code>import_pheno()</code></p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_phenoname">phenoName</code></td>
<td>
<p>pheno name for plot, should be one column name of pheno</p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_happrefix">hapPrefix</code></td>
<td>
<p>prefix of hapotypes, default as &quot;H&quot;</p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_title">title</code></td>
<td>
<p>a charater which will used for figure title</p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_comparisons">comparisons</code></td>
<td>
<p>a list contains comparison pairs
eg. <code>list(c("H001", "H002"), c("H001", "H004"))</code>,
or a character vector contains haplotype names for comparison,
or &quot;none&quot; indicates do not add comparisons.</p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_method">method</code></td>
<td>
<p>a character string indicating which method to be used for
comparing means.</p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_method.args">method.args</code></td>
<td>
<p>a list of additional arguments used for the test method.
For example one might use <code>method.args = list(alternative = "greater")</code>
for wilcoxon test.</p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_symnum.args">symnum.args</code></td>
<td>
<p>a list of arguments to pass to the function
<code><a href="stats.html#topic+symnum">symnum</a></code> for symbolic number coding of p-values. For
example, <code>symnum.args &lt;- list(cutpoints = c(0, 0.0001, 0.001,
 0.01, 0.05, Inf), symbols = c("****", "***", "**", "*",  "ns"))</code>.
</p>
<p>In other words, we use the following convention for symbols indicating
statistical significance: </p>
 <ul>
<li> <p><code>ns</code>: p &gt; 0.05 </p>
</li>
<li>
<p><code>*</code>: p &lt;= 0.05 </p>
</li>
<li> <p><code>**</code>: p &lt;= 0.01 </p>
</li>
<li> <p><code>***</code>: p &lt;= 0.001 </p>
</li>
<li> <p><code>****</code>:  p &lt;= 0.0001 </p>
</li></ul>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_mergefigs">mergeFigs</code></td>
<td>
<p>bool type, indicate whether merge the heat map and box
plot or not. Default as <code>FALSE</code></p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_angle">angle</code></td>
<td>
<p>the angle of x labels</p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_hjust">hjust</code>, <code id="hapVsPheno_+3A_vjust">vjust</code></td>
<td>
<p>hjust and vjust of x labels</p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_minacc">minAcc</code>, <code id="hapVsPheno_+3A_freq.min">freq.min</code></td>
<td>
<p>If observations number of a Hap less than this number will
not be compared with others or be ploted. Should not less than 3 due to the
t-test will meaninglessly. Default as 5</p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_outlier.rm">outlier.rm</code></td>
<td>
<p>whether remove ouliers, default as TRUE</p>
</td></tr>
<tr><td><code id="hapVsPheno_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="ggpubr.html#topic+ggviolin">ggpubr::ggviolin</a></code>
</p>

<dl>
<dt><code>data</code></dt><dd><p>a data frame</p>
</dd>
<dt><code>x</code></dt><dd><p>character string containing the name of x variable.</p>
</dd>
<dt><code>y</code></dt><dd><p>character vector containing one or more variables to plot</p>
</dd>
<dt><code>combine</code></dt><dd><p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</dd>
<dt><code>merge</code></dt><dd><p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</dd>
<dt><code>color</code></dt><dd><p>outline color.</p>
</dd>
<dt><code>fill</code></dt><dd><p>fill color.</p>
</dd>
<dt><code>palette</code></dt><dd><p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</dd>
<dt><code>alpha</code></dt><dd><p>color transparency. Values should be between 0 and 1.</p>
</dd>
<dt><code>xlab</code></dt><dd><p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</dd>
<dt><code>ylab</code></dt><dd><p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</dd>
<dt><code>facet.by</code></dt><dd><p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</dd>
<dt><code>panel.labs</code></dt><dd><p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</dd>
<dt><code>short.panel.labs</code></dt><dd><p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</dd>
<dt><code>linetype</code></dt><dd><p>line types.</p>
</dd>
<dt><code>trim</code></dt><dd><p>If <code>TRUE</code> (default), trim the tails of the violins
to the range of the data. If <code>FALSE</code>, don't trim the tails.</p>
</dd>
<dt><code>size</code></dt><dd><p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</dd>
<dt><code>width</code></dt><dd><p>violin width.</p>
</dd>
<dt><code>draw_quantiles</code></dt><dd><p>If <code>not(NULL)</code> (default), draw horizontal lines
at the given quantiles of the density estimate.</p>
</dd>
<dt><code>select</code></dt><dd><p>character vector specifying which items to display.</p>
</dd>
<dt><code>remove</code></dt><dd><p>character vector specifying which items to remove from the plot.</p>
</dd>
<dt><code>order</code></dt><dd><p>character vector specifying the order of items.</p>
</dd>
<dt><code>add</code></dt><dd><p>character vector for adding another plot element (e.g.: dot plot or
error bars). Allowed values are one or the combination of: &quot;none&quot;,
&quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;,
&quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;,
&quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;; see ?desc_statby for more
details.</p>
</dd>
<dt><code>add.params</code></dt><dd><p>parameters (color, shape, size, fill, linetype) for the
argument 'add'; e.g.: add.params = list(color = &quot;red&quot;).</p>
</dd>
<dt><code>error.plot</code></dt><dd><p>plot type used to visualize error. Allowed values are one of
c(&quot;pointrange&quot;, &quot;linerange&quot;, &quot;crossbar&quot;, &quot;errorbar&quot;, &quot;upper_errorbar&quot;,
&quot;lower_errorbar&quot;, &quot;upper_pointrange&quot;, &quot;lower_pointrange&quot;, &quot;upper_linerange&quot;,
&quot;lower_linerange&quot;). Default value is &quot;pointrange&quot; or &quot;errorbar&quot;. Used only
when add != &quot;none&quot; and add contains one &quot;mean_*&quot; or &quot;med_*&quot; where &quot;*&quot; = sd,
se, ....</p>
</dd>
<dt><code>label</code></dt><dd><p>the name of the column containing point labels. Can be also a
character vector with length = nrow(data).</p>
</dd>
<dt><code>font.label</code></dt><dd><p>a list which can contain the combination of the following
elements: the size (e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;,
&quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of labels. For example font.label
= list(size = 14, face = &quot;bold&quot;, color =&quot;red&quot;). To specify only the size and
the style, use font.label = list(size = 14, face = &quot;plain&quot;).</p>
</dd>
<dt><code>label.select</code></dt><dd><p>can be of two formats: </p>
 <ul>
<li><p> a character vector
specifying some labels to show. </p>
</li>
<li><p> a list containing one or the
combination of the following components: </p>
 <ul>
<li> <p><code>top.up</code> and
<code>top.down</code>: to display the labels  of the top up/down points. For
example, <code>label.select = list(top.up = 10, top.down = 4)</code>. </p>
</li>
<li>
<p><code>criteria</code>: to filter, for example, by x and y variabes values, use
this: <code>label.select = list(criteria = "`y` &gt; 2 &amp; `y` &lt; 5 &amp; `x` %in%
c('A', 'B')")</code>. </p>
</li></ul>
 </li></ul>
</dd>
<dt><code>repel</code></dt><dd><p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</dd>
<dt><code>label.rectangle</code></dt><dd><p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</dd>
<dt><code>position</code></dt><dd><p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</dd>
<dt><code>ggtheme</code></dt><dd><p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Value</h3>

<p>list. A list contains a character vector with Haps were applied
student test, a mattrix contains p-value of each compare of Haps and a
ggplot2 object named as figs if mergeFigs set as <code>TRUE</code>, or two ggplot2
objects names as fig_pvalue and fig_Violin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data("geneHapR_test")
# plot the figs directly
hapVsPheno(hap = hapResult,
           pheno = pheno,
           phenoName = "GrainWeight.2021",
           minAcc = 3)

# do not merge the files
results &lt;- hapVsPheno(hap = hapResult,
                      pheno = pheno,
                      phenoName = "GrainWeight.2021",
                      minAcc = 3,
                      mergeFigs = FALSE)
plot(results$fig_pvalue)
plot(results$fig_Violin)

</code></pre>

<hr>
<h2 id='hapVsPhenoPerSite'>hapVsPhenoPerSite</h2><span id='topic+hapVsPhenoPerSite'></span>

<h3>Description</h3>

<p>Comparie phenotype site by site.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapVsPhenoPerSite(
  hap,
  pheno,
  phenoName,
  sitePOS,
  fileName,
  fileType = NULL,
  freq.min = 5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hapVsPhenoPerSite_+3A_hap">hap</code></td>
<td>
<p>an R object of hapresult class</p>
</td></tr>
<tr><td><code id="hapVsPhenoPerSite_+3A_pheno">pheno</code>, <code id="hapVsPhenoPerSite_+3A_phenoname">phenoName</code></td>
<td>
<p>pheno, a data.frame contains the phenotypes;
Only one phenotype name is required.</p>
</td></tr>
<tr><td><code id="hapVsPhenoPerSite_+3A_sitepos">sitePOS</code></td>
<td>
<p>the coordinate of site</p>
</td></tr>
<tr><td><code id="hapVsPhenoPerSite_+3A_filename">fileName</code>, <code id="hapVsPhenoPerSite_+3A_filetype">fileType</code></td>
<td>
<p>file name and file type will be needed for saving result,
file type could be one of &quot;png, tiff, jpg&quot;</p>
</td></tr>
<tr><td><code id="hapVsPhenoPerSite_+3A_freq.min">freq.min</code></td>
<td>
<p>miner allies frequency less than freq.min will not be skipped</p>
</td></tr>
<tr><td><code id="hapVsPhenoPerSite_+3A_...">...</code></td>
<td>
<p>addtional params will be passed to plot saving function like <code>tiff()</code>, <code>png()</code>, <code>pdf()</code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data("geneHapR_test")
hapVsPhenoPerSite(hapResult, pheno, sitePOS = "4300")
</code></pre>

<hr>
<h2 id='hapVsPhenos'>hapVsPhenos</h2><span id='topic+hapVsPhenos'></span>

<h3>Description</h3>

<p>hapVsPhenos
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapVsPhenos(
  hap,
  pheno,
  outPutSingleFile = TRUE,
  hapPrefix = "H",
  title = "Seita.0G000000",
  width = 12,
  height = 8,
  res = 300,
  compression = "lzw",
  filename.prefix = filename.prefix,
  filename.surfix = "pdf",
  filename.sep = "_",
  outlier.rm = TRUE,
  mergeFigs = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hapVsPhenos_+3A_hap">hap</code></td>
<td>
<p>object of hapResult class, generate with<code>vcf2hap()</code> or
<code>seqs2hap()</code></p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_pheno">pheno</code></td>
<td>
<p>object of data.frame class, imported by <code>import_pheno()</code></p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_outputsinglefile">outPutSingleFile</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> indicate whether put all figs
into to each pages of single file or generate multi-files.
Only worked while file type is pdf</p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_happrefix">hapPrefix</code></td>
<td>
<p>prefix of hapotypes, default as &quot;H&quot;</p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_title">title</code></td>
<td>
<p>a charater which will used for figure title</p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_width">width</code></td>
<td>
<p>manual option for determining the output file width in inches.
(default: 12)</p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_height">height</code></td>
<td>
<p>manual option for determining the output file height in inches.
(default: 8)</p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_res">res</code></td>
<td>
<p>The nominal resolution in ppi which will be recorded in the
bitmap file, if a positive integer. Also used for units other than the
default, and to convert points to pixels</p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_compression">compression</code></td>
<td>
<p>the type of compression to be used.</p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_filename.prefix">filename.prefix</code>, <code id="hapVsPhenos_+3A_filename.surfix">filename.surfix</code>, <code id="hapVsPhenos_+3A_filename.sep">filename.sep</code></td>
<td>
<p>if multi files generated, file names will be formed by
prefix <code>filename.prefix</code>, a seperate charcter <code>filename.sep</code>,
pheno name, a dot and surfix <code>filename.surfix</code>,
and file type was decide by <code>filename.surfix</code>;
if single file was generated, file name will be formed by
prefix <code>filename.prefix</code>, a dot and surfix <code>filename.surfix</code></p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_outlier.rm">outlier.rm</code></td>
<td>
<p>whether remove ouliers, default as TRUE</p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_mergefigs">mergeFigs</code></td>
<td>
<p>bool type, indicate whether merge the heat map and box
plot or not. Default as <code>FALSE</code></p>
</td></tr>
<tr><td><code id="hapVsPhenos_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="#topic+hapVsPheno">hapVsPheno</a></code>
</p>

<dl>
<dt><code>phenoName</code></dt><dd><p>pheno name for plot, should be one column name of pheno</p>
</dd>
<dt><code>minAcc,freq.min</code></dt><dd><p>If observations number of a Hap less than this number will
not be compared with others or be ploted. Should not less than 3 due to the
t-test will meaninglessly. Default as 5</p>
</dd>
<dt><code>angle</code></dt><dd><p>the angle of x labels</p>
</dd>
<dt><code>hjust,vjust</code></dt><dd><p>hjust and vjust of x labels</p>
</dd>
<dt><code>comparisons</code></dt><dd><p>a list contains comparison pairs
eg. <code>list(c("H001", "H002"), c("H001", "H004"))</code>,
or a character vector contains haplotype names for comparison,
or &quot;none&quot; indicates do not add comparisons.</p>
</dd>
<dt><code>method</code></dt><dd><p>a character string indicating which method to be used for
comparing means.</p>
</dd>
<dt><code>method.args</code></dt><dd><p>a list of additional arguments used for the test method.
For example one might use <code>method.args = list(alternative = "greater")</code>
for wilcoxon test.</p>
</dd>
<dt><code>symnum.args</code></dt><dd><p>a list of arguments to pass to the function
<code><a href="stats.html#topic+symnum">symnum</a></code> for symbolic number coding of p-values. For
example, <code>symnum.args &lt;- list(cutpoints = c(0, 0.0001, 0.001,
 0.01, 0.05, Inf), symbols = c("****", "***", "**", "*",  "ns"))</code>.
</p>
<p>In other words, we use the following convention for symbols indicating
statistical significance: </p>
 <ul>
<li> <p><code>ns</code>: p &gt; 0.05 </p>
</li>
<li>
<p><code>*</code>: p &lt;= 0.05 </p>
</li>
<li> <p><code>**</code>: p &lt;= 0.01 </p>
</li>
<li> <p><code>***</code>: p &lt;= 0.001 </p>
</li>
<li> <p><code>****</code>:  p &lt;= 0.0001 </p>
</li></ul>
</dd>
</dl>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("geneHapR_test")

oriDir &lt;- getwd()
temp_dir &lt;- tempdir()
if(! dir.exists(temp_dir))
  dir.create(temp_dir)
setwd(temp_dir)
# analysis all pheno in the data.frame of pheno
hapVsPhenos(hapResult,
            pheno,
            outPutSingleFile = TRUE,
            hapPrefix = "H",
            title = "Seita.0G000000",
            filename.prefix = "test",
            width = 12,
            height = 8,
            res = 300)
setwd(oriDir)

</code></pre>

<hr>
<h2 id='import_AccINFO'>Import Accession Information from File</h2><span id='topic+import_AccINFO'></span>

<h3>Description</h3>

<p>import accession information including phenotype data,
accession group, location from a tab delimed table file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_AccINFO(file, comment.char = "#",
               check.names = FALSE, row.names = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_AccINFO_+3A_file">file</code></td>
<td>
<p>file path, this file should be a tab delimed table</p>
</td></tr>
<tr><td><code id="import_AccINFO_+3A_comment.char">comment.char</code></td>
<td>
<p>character: a character vector of length one
containing a single character or an empty string.  Use <code>""</code> to
turn off the interpretation of comments altogether.</p>
</td></tr>
<tr><td><code id="import_AccINFO_+3A_check.names">check.names</code></td>
<td>
<p>logical.  If <code>TRUE</code> then the names of the
variables in the data frame are checked to ensure that they are
syntactically valid variable names.  If necessary they are adjusted
(by <code><a href="base.html#topic+make.names">make.names</a></code>) so that they are, and also to ensure
that there are no duplicates.</p>
</td></tr>
<tr><td><code id="import_AccINFO_+3A_row.names">row.names</code></td>
<td>
<p>a vector of row names.  This can be a vector giving
the actual row names, or a single number giving the column of the
table which contains the row names, or character string giving the
name of the table column containing the row names.
</p>
<p>If there is a header and the first row contains one fewer field than
the number of columns, the first column in the input is used for the
row names.  Otherwise if <code>row.names</code> is missing, the rows are
numbered.
</p>
<p>Using <code>row.names = NULL</code> forces row numbering. Missing or
<code>NULL</code> <code>row.names</code> generate row names that are considered
to be &lsquo;automatic&rsquo; (and not preserved by <code><a href="base.html#topic+as.matrix">as.matrix</a></code>).
</p>
</td></tr>
<tr><td><code id="import_AccINFO_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>read.table</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>First column should be Accessions;
phenos/accession information should begin from second column,
phenoName/group/locations should located at the first row,
If a dot '.' is located in pheno name, then
the part before the dot will be set as y axis name
and the latter will be set as foot when plot figures.
</p>


<h3>Value</h3>

<p>data.frame, Accession names were set as rownames and columns were
named by pheno/info names
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
oldDir &lt;- getwd()
temp_dir &lt;- tempdir()
if(! dir.exists(temp_dir))
  dir.create(temp_dir)
setwd(temp_dir)
data("geneHapR_test")
write.table(pheno, file = "test.pheno.txt", sep = "\t")
pheno &lt;- import_AccINFO("test.pheno.txt")
pheno
setwd(oldDir)

</code></pre>

<hr>
<h2 id='import_bed'>import annotation files in BED format</h2><span id='topic+import_bed'></span>

<h3>Description</h3>

<p>import bed files contains annotations into R as GRanges object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_bed(con, quite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_bed_+3A_con">con</code></td>
<td>
<p>A path, URL, connection or <code>BEDFile</code> object. For the
functions ending in <code>.bed</code>, <code>.bedGraph</code> and <code>.bed15</code>,
the file format is indicated by the function name. For the base
<code>export</code> and <code>import</code> functions, the format must be
indicated another way. If <code>con</code> is a path, URL or connection,
either the file extension or the <code>format</code> argument needs to be
one of &ldquo;bed&rdquo;, &ldquo;bed15&rdquo;, &ldquo;bedGraph&rdquo;,
&ldquo;bedpe&rdquo;, &ldquo;narrowPeak&rdquo;, or
&ldquo;broadPeak&rdquo;. Compressed
files (&ldquo;gz&rdquo;, &ldquo;bz2&rdquo; and
&ldquo;xz&rdquo;) are handled transparently.
</p>
</td></tr>
<tr><td><code id="import_bed_+3A_quite">quite</code></td>
<td>
<p>whether show message</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If there is no genome annotation file in GFF format for your interest species,
a BED file is convenient to custom a simple annotation file for a gene.
Here we suggest two type of BED format: BED6 and BED4.
</p>
<p>As the definition of
<a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1">UCSC</a>.
The BED6 contains 6 columns, which are 1) chrom, 2) chromStart, 3) chromEnd,
4) name, 5) score and 6) strand. The BED4 format contains the first 4 column
of BED6 format.
</p>
<p>However, in gene haplotype statistics, we only care about the type of each site.
Thus we use the fourth column to definition the
transcripts name and &quot;CDS&quot; or &quot;URTs&quot;, separated by a space, eg.:
</p>
<p><code>Chr8   678   890   HD1.1 CDS   .   -</code>
</p>
<p><code>Chr8   891   989   HD1.1 five_prime_UTR   .   -</code>
</p>
<p><code>Chr8   668   759   HD1.2 CDS   .   -</code>
</p>
<p><code>Chr8   908   989   HD1.2 CDS   .   -</code>
</p>
<p>This example indicate a small gene named as HD1 have two transcripts,
named as HD1.1 and HD1.2, separately. HD1 has a CDS and a UTR region;
while HD1.2 has two CDS region.
</p>


<h3>Value</h3>

<p>GRange object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
bed.Path &lt;- system.file("extdata", "annotation.bed6", package = "geneHapR")
bed &lt;- import_bed(bed.Path)
bed

</code></pre>

<hr>
<h2 id='import_gff'>Import Annotations from GFF Format File</h2><span id='topic+import_gff'></span>

<h3>Description</h3>

<p>import genome annotations in GFF/GFF3 format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_gff(gffFile, format = "gff")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_gff_+3A_gfffile">gffFile</code></td>
<td>
<p>the gff file path</p>
</td></tr>
<tr><td><code id="import_gff_+3A_format">format</code></td>
<td>
<p>should be one of &quot;gff&quot;, &quot;gff1&quot;, &quot;gff2&quot;, &quot;gff3&quot;, &quot;gvf&quot;,
or &quot;gtf&quot;. Default as gff</p>
</td></tr>
</table>


<h3>Value</h3>

<p>GRange object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gff.Path &lt;- system.file("extdata", "annotation.gff", package = "geneHapR")
gff &lt;- import_gff(gff.Path, format = "gff")
gff

</code></pre>

<hr>
<h2 id='import_hap'>Import hapResult/hapSummary</h2><span id='topic+import_hap'></span>

<h3>Description</h3>

<p>This function could be used for import hap result or hap summary result.
The type of returned object is decided by input file, see details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_hap(file, type = "auto", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_hap_+3A_file">file</code></td>
<td>
<p>hapSummary or hapResult file path.</p>
</td></tr>
<tr><td><code id="import_hap_+3A_type">type</code></td>
<td>
<p>the content type of imported file, should be one of c(&quot;hapResult&quot;, &quot;hapSummary&quot;)</p>
</td></tr>
<tr><td><code id="import_hap_+3A_...">...</code></td>
<td>
<p>extras will pass to <code>read.delim()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The hap result and hap summary result have common features.
The common features of these two types are:
First four rows contains extra information: CHR, POS, INFO and ALLELE
Hap names were in the first column.
The differences are:
Hap summary result have a freq column while hap result not.
Rows represent haplotypes in hap summary result, while rows represent accessions in hap result.
In addtion, the accessions of each haplotype in hap summary result were separated by &quot;;&quot;.
</p>


<h3>Value</h3>

<p>hapSummary or hapResult
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
oldDir &lt;- getwd()
temp_dir &lt;- tempdir()
if(! dir.exists(temp_dir))
  dir.create(temp_dir)
setwd(temp_dir)
data("geneHapR_test")
write.hap(hapResult, file = "test.pheno.txt", sep = "\t")
hap &lt;- import_hap("test.pheno.txt")
hap
setwd(oldDir)

</code></pre>

<hr>
<h2 id='import_MultipleAlignment'>Import MultipleAlignment Result</h2><span id='topic+import_MultipleAlignment'></span>

<h3>Description</h3>

<p>import sequences algned results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_MultipleAlignment(filepath, format = "fasta", type = "DNA")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_MultipleAlignment_+3A_filepath">filepath</code></td>
<td>

<p>A character vector (of arbitrary length when reading, of length 1
when writing) containing the paths to the files to read or write.
Note that special values like <code>""</code> or <code>"|cmd"</code> (typically
supported by other I/O functions in R) are not supported here. Also
<code>filepath</code> cannot be a connection.
</p>
</td></tr>
<tr><td><code id="import_MultipleAlignment_+3A_format">format</code></td>
<td>

<p>Either <code>"fasta"</code> (the default), <code>stockholm</code>, or
<code>"clustal"</code>.
</p>
</td></tr>
<tr><td><code id="import_MultipleAlignment_+3A_type">type</code></td>
<td>
<p>one of &quot;DNA&quot; and &quot;Protein&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of DNAMultipleAlignment
</p>


<h3>Examples</h3>

<pre><code class='language-R'>aliSeqPath &lt;- system.file("extdata", "seqs.fa", package = "geneHapR")

geneSeqs &lt;- import_MultipleAlignment(filepath = aliSeqPath,
                                     format = "fasta",
                                     type = "DNA")
geneSeqs &lt;- import_MultipleAlignment(filepath = aliSeqPath,
                                     format = "fasta",
                                     type = "Protein")

</code></pre>

<hr>
<h2 id='import_plink.pedmap'>import_plink.pedmap</h2><span id='topic+import_plink.pedmap'></span>

<h3>Description</h3>

<p>used for import regular p.link file stored in map and ped format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_plink.pedmap(root = root,
                      sep_ped = "\t", sep_map = "\t",
                      pedfile = pedfile, mapfile = mapfile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_plink.pedmap_+3A_root">root</code></td>
<td>
<p>The file name without suffix. This function only support p.link file
format stored in &quot;map&quot; and &quot;ped&quot; format, the file names after removed suffix
should be same with each other.</p>
</td></tr>
<tr><td><code id="import_plink.pedmap_+3A_sep_ped">sep_ped</code></td>
<td>
<p>a character indicate the separation of ped file</p>
</td></tr>
<tr><td><code id="import_plink.pedmap_+3A_sep_map">sep_map</code></td>
<td>
<p>a character indicate the separation of map file</p>
</td></tr>
<tr><td><code id="import_plink.pedmap_+3A_pedfile">pedfile</code>, <code id="import_plink.pedmap_+3A_mapfile">mapfile</code></td>
<td>
<p>if <code>root</code> is missing then <code>pedfile</code> and <code>mapfile</code> are needed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list, contains map information stored in data.frame and ped
information stored in data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
   pedfile &lt;- system.file("extdata",
                          "snp3kvars-CHR8-25947258-25951166-plink.ped",
                          package = "geneHapR")
   mapfile &lt;- system.file("extdata",
                          "snp3kvars-CHR8-25947258-25951166-plink.map",
                          package = "geneHapR")
   p.link &lt;- import_plink.pedmap(pedfile = pedfile, mapfile = mapfile,
                                 sep_map = "\t", sep_ped = "\t")
   p.link &lt;- filter_plink.pedmap(p.link, mode = "POS",
                                 Chr = "chr08", start = 25948004,
                                 end = 25949944)
   hapResult &lt;- plink.pedmap2hap(p.link, hapPrefix = "H",
                                 hetero_remove = TRUE,
                                 na_drop = TRUE)

</code></pre>

<hr>
<h2 id='import_seqs'>Import Sequences</h2><span id='topic+import_seqs'></span>

<h3>Description</h3>

<p>import DNA sequences in FASTA format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_seqs(filepath, format = "fasta")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_seqs_+3A_filepath">filepath</code></td>
<td>
<p>A character vector containing the path to the DNA sequences file.
Reading files in gzip format (which usually have the '.gz' extension) is
supported.
<em>Note</em> that only DNA supported here.</p>
</td></tr>
<tr><td><code id="import_seqs_+3A_format">format</code></td>
<td>
<p>Either &quot;fasta&quot; (the default) or &quot;fastq&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of DNAStringSet class
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
seqPath &lt;- system.file("extdata", "seqs.fa", package = "geneHapR")
geneSeqs &lt;- import_seqs(filepath = seqPath, format = "fasta")

</code></pre>

<hr>
<h2 id='import_vcf'>Import VCF from File</h2><span id='topic+import_vcf'></span>

<h3>Description</h3>

<p>import *.vcf structured text format,
as well as the compressed <code style="white-space: pre;">&#8288;*.vcf.gz&#8288;</code> format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_vcf(file = file, ...)

import_vcf(file = file, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_vcf_+3A_file">file</code></td>
<td>
<p>file path of VCF file</p>
</td></tr>
<tr><td><code id="import_vcf_+3A_...">...</code></td>
<td>
<p>pass to <code>vcfR::read.vcfR()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>vcfR object
</p>


<h3>Author(s)</h3>

<p>Zhangrenl
</p>


<h3>See Also</h3>

<p><code><a href="vcfR.html#topic+read.vcfR">vcfR::read.vcfR()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
vcfPath &lt;- system.file("extdata", "var.vcf.gz", package = "geneHapR")
vcf &lt;- import_vcf(file = vcfPath)
vcf

</code></pre>

<hr>
<h2 id='LDheatmap'>This function produces a pairwise LD plot.</h2><span id='topic+LDheatmap'></span><span id='topic+plot_LDheatmap'></span>

<h3>Description</h3>

<p><code>LDheatmap()</code> is used to produce a graphical display, as a heat map,
of pairwise linkage disequilibrium (LD) measurements for SNPs.
The heat map is a false color image in the upper-left diagonal of a square plot.
Optionally, a line parallel to the diagonal of the image indicating
the physical or genetic map positions of the SNPs may be added, along
with text reporting the total length of the genomic region considered.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_LDheatmap(
  hap,
  gff,
  Chr,
  start,
  end,
  geneID = NULL,
  distances = "physical",
  LDmeasure = "r",
  title = "Pairwise LD",
  add.map = TRUE,
  map.height = 1,
  colorLegend = TRUE,
  geneMapLocation = 0.15,
  geneMapLabelX = NULL,
  geneMapLabelY = NULL,
  SNP.name = TRUE,
  color = NULL,
  color_gmodel = "grey",
  color_snp = "grey",
  color_snpname = "grey40",
  cex_snpname = 0.8,
  snpmarks_height = NULL,
  newpage = TRUE,
  name = "ldheatmap",
  vp.name = NULL,
  pop = FALSE,
  text = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDheatmap_+3A_hap">hap</code></td>
<td>
<p>R object of hapSummary or hapResult class.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_gff">gff</code></td>
<td>
<p>annotations</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_chr">Chr</code>, <code id="LDheatmap_+3A_start">start</code>, <code id="LDheatmap_+3A_end">end</code>, <code id="LDheatmap_+3A_geneid">geneID</code></td>
<td>
<p>chromosome, start and end position, gene ID for
extract annotation in target range.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_distances">distances</code></td>
<td>
<p>A character string to specify whether the provided map locations
are in physical or genetic distances.
If <code>distances = "physical"</code> (default), the text describing the total
length of the region will be &ldquo;Physical Length:XXkb&rdquo; where XX is the
length of the region in kilobases. If <code>distances = "genetic"</code>, the
text will be &ldquo;Genetic Map Length:YYcM&rdquo; where YY is
the length of the region in centiMorgans.
If <code>gdat</code> is an object of class LDheatmap,
<code>distances</code> is taken from <code>gdat</code>.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_ldmeasure">LDmeasure</code></td>
<td>
<p>A character string specifying the measure of LD
</p>

<ul>
<li><p> either allelic correlation <code class="reqn">r^2</code> or Lewontin's
|D<code class="reqn">'</code>|; default = <code>"r"</code> for <code class="reqn">r^2</code>;
type <code>"D'"</code> for |D<code class="reqn">'</code>|. This argument is ignored when the user has already
supplied calculated LD measurements through <code>gdat</code> (i.e., when <code>gdat</code>
is a matrix of pairwise LD measurements or an object of class <code>"LDheatmap"</code>).
</p>
</li></ul>
</td></tr>
<tr><td><code id="LDheatmap_+3A_title">title</code></td>
<td>
<p>A character string for the main title of the plot.
Default is &ldquo;Pairwise LD&rdquo;.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_add.map">add.map</code></td>
<td>
<p>If <code>TRUE</code> (default) a diagonal line indicating
the physical or genetic map positions of the SNPs will be added to
the plot, along with text indicating the total length of the
genetic region.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_map.height">map.height</code></td>
<td>
<p>the height of gene map, default is 0.02</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_colorlegend">colorLegend</code></td>
<td>
<p>If <code>TRUE</code> (default) the color legend is drawn.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_genemaplocation">geneMapLocation</code></td>
<td>
<p>A numeric value specifying the position of the line
parallel to the diagonal of the matrix; the larger the value, the
farther it lies from the matrix diagonal. Ignored when <code>add.map = FALSE</code>.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_genemaplabelx">geneMapLabelX</code></td>
<td>
<p>A numeric value specifying the x-coordinate
of the text indicating the total length of the genomic region
being considered. Ignored when <code>add.map = FALSE</code>.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_genemaplabely">geneMapLabelY</code></td>
<td>
<p>A numeric value specifying the y-coordinate
of the text indicating the total length of the genomic region
being considered. Ignored when <code>add.map = FALSE</code>.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_snp.name">SNP.name</code></td>
<td>
<p>a logical vector indicated wherther display SNP names using positions.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_color">color</code></td>
<td>
<p>A range of colors to be used for drawing the heat map. Default
is <code>grDevices::colorRampPalette(c("red", "grey"))(30)</code>.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_color_gmodel">color_gmodel</code>, <code id="LDheatmap_+3A_color_snp">color_snp</code>, <code id="LDheatmap_+3A_color_snpname">color_snpname</code></td>
<td>
<p>the color of gene model and snp and snp names respectively, default as grey80.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_cex_snpname">cex_snpname</code></td>
<td>
<p>the size of snp names/labels</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_snpmarks_height">snpmarks_height</code></td>
<td>
<p>the height of snp marks, if set as NULL, nothing will display on gene model
where the heat map is going to be drawn.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_newpage">newpage</code></td>
<td>
<p>If <code>TRUE</code> (default), the heat map will be drawn on a new page.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_name">name</code></td>
<td>
<p>A character string specifying the name of the LDheatmap
graphical object (<code>grob</code>) to be produced.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_vp.name">vp.name</code></td>
<td>
<p>A character string specifying the name of the viewport</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_pop">pop</code></td>
<td>
<p>If <code>TRUE</code>, the viewport where the heat map is drawn is
<code>pop</code>ped (i.e. removed) from the viewport tree after drawing.
Default = <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="LDheatmap_+3A_text">text</code></td>
<td>
<p>If <code>TRUE</code>, the LD measurements are printed on each cell.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input object <code>gdat</code> can be a data frame of <code>genotype</code> objects
(a data structure from the <span class="pkg">genetics</span> package), a <code>SnpMatrix</code> object
(a data structure from the <span class="pkg">snpStats</span> package), or
any square matrix with values between 0 and 1 inclusive.
LD computation is much faster for <code>SnpMatrix</code> objects than for
<code>genotype</code> objects.
In the case of a matrix of LD values between 0 and 1,
the values above the diagonal will be plotted.
In the display of LD, SNPs appear in the order supplied by the
user as the horizontal and vertical coordinates are increased
and one moves along the off-diagonal line, from the bottom-left
to the top-right corner. To achieve this, the conventions of
the <code>image()</code> function have been adopted, in which horizontal
coordinates correspond to the rows of the matrix and vertical coordinates
correspond to columns, and vertical coordinates are indexed in increasing
order from bottom to top.
See the package vignette <code>LDheatmap</code> for more examples and details
of the implementation. Examples of adding &ldquo;tracks&rdquo; of genomic
annotation above a flipped heatmap are in the package vignette
<code>addTracks</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"LDheatmap"</code> which contains the following components:
</p>
<table>
<tr><td><code>LDmatrix</code></td>
<td>
<p> The matrix of pairwise LD measurements plotted in the heat map. </p>
</td></tr>
<tr><td><code>LDheatmapGrob</code></td>
<td>
<p> A grid graphical object (grob) representing the produced heat map. </p>
</td></tr>
<tr><td><code>heatmapVP</code></td>
<td>
<p> The viewport in which the heat map is drawn. See <a href="grid.html#topic+viewport">viewport</a>.</p>
</td></tr>
<tr><td><code>genetic.distances</code></td>
<td>
<p>The vector of the supplied physical or
genetic map locations, or the vector of equispaced marker distances
when no distance vector is supplied.</p>
</td></tr>
<tr><td><code>distances</code></td>
<td>
<p> A character string specifying whether the provided map
distances are physical or genetic. </p>
</td></tr>
<tr><td><code>color</code></td>
<td>
<p> The range of colors used for drawing the heat map. </p>
</td></tr>
</table>
<p>The <code>grob</code> <code>LDheatmapGrob</code> has three <code>grob</code>s as its children (components).
They are listed below along with their own children and respectively represent
the color image with main title, genetic map and color key of the heat map:
<code>"heatMap"</code> - <code>"heatmap"</code>, <code>"title"</code>;
<code>"geneMap"</code> - <code>"diagonal"</code>, <code>"segments"</code>,
<code>"title"</code>, <code>"symbols"</code>, <code>"SNPnames"</code>; and
<code>"Key"</code> - <code>"colorKey"</code>, <code>"title"</code>, <code>"labels"</code>,
<code>"ticks"</code>, <code>"box"</code>.
</p>


<h3>Note</h3>

<p>The produced heat map can be modified in two ways.
First, it is possible to edit <em>interactively</em> the grob components of the heat map,
by using the function <code><a href="grid.html#topic+grid.edit">grid.edit</a></code>;
the function will not work if there is no
open graphical device showing the heat map.
Alternatively, the user can use the function
<code><a href="grid.html#topic+grid.edit">editGrob</a></code> and work with
the grob <code>LDheatmapGrob</code> returned by <code>LDheatmap</code>.
See Examples for usage.
<code>LDheatmap()</code> uses <code><a href="grid.html#topic+Grid">Grid</a></code>, which
does not respond to <code>par()</code> settings.
Hence modifying <code>par()</code> settings of <code>mfrow</code> and <code>mfcol</code>
will not work with <code>LDheatmap()</code>. The Examples section shows how to
display multiple heat maps on one plot without the use
of <code>par()</code>.
</p>


<h3>References</h3>

<p>Shin J-H, Blay S, McNeney B and Graham J (2006). LDheatmap:
An R Function for Graphical Display of Pairwise Linkage
Disequilibria Between Single Nucleotide Polymorphisms.
Journal of Statistical Software, <b>16</b> Code Snippet 3
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Pass LDheatmap a SnpMatrix object
data(geneHapR_test)
plot_LDheatmap(hap = hapResult,
               gff = gff,
               Chr = hapResult[1,2],
               start = 4000, end = 8200)
</code></pre>

<hr>
<h2 id='network'>Generate Haplotype Net Relationshop with Haplotype Result</h2><span id='topic+network'></span><span id='topic+get_hapNet'></span><span id='topic+getHapGroup'></span>

<h3>Description</h3>

<p>computes a haplotype network with haplotype summary result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_hapNet(hapSummary,
           AccINFO = AccINFO,
           groupName = groupName,
           na.label = "Unknown")

getHapGroup(
  hapSummary,
  AccINFO = AccINFO,
  groupName = groupName,
  na.label = na.label
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="network_+3A_hapsummary">hapSummary</code></td>
<td>
<p>object of <code>hapSummary</code> class, generated by <code>hap_summary()</code></p>
</td></tr>
<tr><td><code id="network_+3A_accinfo">AccINFO</code></td>
<td>
<p>data.frame, specified groups of each accession.
Used for pie plot. If missing, pie will not draw in plotHapNet.
Or you can supplied a hap_group mattrix with <code>plot(hapNet, pie = hap_group)</code>.</p>
</td></tr>
<tr><td><code id="network_+3A_groupname">groupName</code></td>
<td>
<p>the group name used for pie plot,
should be in <code>AccINFO</code> column names, default as the first column name</p>
</td></tr>
<tr><td><code id="network_+3A_na.label">na.label</code></td>
<td>
<p>the label of <code>NA</code>s</p>
</td></tr>
</table>


<h3>Value</h3>

<p>hapNet class
</p>


<h3>References</h3>

<p>Mark P.J. van der Loo (2014)
<a href="https://doi.org/10.32614/RJ-2014-011">doi:10.32614/RJ-2014-011</a>;
</p>
<p>E. Paradis (2010)
<a href="https://doi.org/10.1093/bioinformatics/btp696">doi:10.1093/bioinformatics/btp696</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotHapNet">plotHapNet()</a></code> and
<code><a href="#topic+hap_summary">hap_summary()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data("geneHapR_test")
hapSummary &lt;- hap_summary(hapResult)

# calculate haploNet
hapNet &lt;- get_hapNet(hapSummary,
                     AccINFO = AccINFO, # accession types
                     groupName = colnames(AccINFO)[2])

# plot haploNet
plot(hapNet)

# plot haploNet
plotHapNet(hapNet,
           size = "freq",   # circle size
           scale = "log10", # scale circle with 'log10(size + 1)'
           cex = 1, # size of hap symbol
           col.link = 2, # link colors
           link.width = 2, # link widths
           show.mutation = 2, # mutation types one of c(0,1,2,3)
           legend = FALSE) # legend position

</code></pre>

<hr>
<h2 id='plink.pedmap2hap'>plink.pedmap2hap</h2><span id='topic+plink.pedmap2hap'></span>

<h3>Description</h3>

<p>convert p.link format data into hapResult
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plink.pedmap2hap(
  p.link,
  hapPrefix = "H",
  pad = 3,
  hetero_remove = TRUE,
  na_drop = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plink.pedmap2hap_+3A_p.link">p.link</code></td>
<td>
<p>list contains p.link information</p>
</td></tr>
<tr><td><code id="plink.pedmap2hap_+3A_happrefix">hapPrefix</code></td>
<td>
<p>prefix of haplotype names</p>
</td></tr>
<tr><td><code id="plink.pedmap2hap_+3A_pad">pad</code></td>
<td>
<p>The number length in haplotype names should be extend to.</p>
</td></tr>
<tr><td><code id="plink.pedmap2hap_+3A_hetero_remove">hetero_remove</code></td>
<td>
<p>whether remove accessions contains hyb-sites</p>
</td></tr>
<tr><td><code id="plink.pedmap2hap_+3A_na_drop">na_drop</code></td>
<td>
<p>whether drop accessions contains missing data (&quot;N&quot;, NA)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of hapSummary class
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
   pedfile &lt;- system.file("extdata",
                          "snp3kvars-CHR8-25947258-25951166-plink.ped",
                          package = "geneHapR")
   mapfile &lt;- system.file("extdata",
                          "snp3kvars-CHR8-25947258-25951166-plink.map",
                          package = "geneHapR")
   p.link &lt;- import_plink.pedmap(pedfile = pedfile, mapfile = mapfile,
                                 sep_map = "\t", sep_ped = "\t")
   p.link &lt;- filter_plink.pedmap(p.link, mode = "POS",
                                 Chr = "chr08", start = 25948004,
                                 end = 25949944)
   hapResult &lt;- plink.pedmap2hap(p.link, hapPrefix = "H",
                                 hetero_remove = TRUE,
                                 na_drop = TRUE)

</code></pre>

<hr>
<h2 id='plotEFF'>plotEFF</h2><span id='topic+plotEFF'></span>

<h3>Description</h3>

<p>plotEFF
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotEFF(
  siteEFF,
  gff = gff,
  Chr = Chr,
  start = start,
  end = end,
  showType = c("five_prime_UTR", "CDS", "three_prime_UTR"),
  CDS.height = CDS.height,
  cex = 0.1,
  col = c("red", "yellow"),
  pch = 20,
  main = main,
  legend.cex = 0.8,
  gene.legend = TRUE,
  markMutants = TRUE,
  mutants.col = 1,
  mutants.type = 1,
  y = c("pvalue", "effect"),
  ylab = ylab,
  legendtitle = legendtitle,
  par.restore = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotEFF_+3A_siteeff">siteEFF</code></td>
<td>
<p>matrix, column name are pheno names and row name are site position</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_gff">gff</code></td>
<td>
<p>gff annotation</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_chr">Chr</code></td>
<td>
<p>the chromosome name</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_start">start</code></td>
<td>
<p>start position</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_end">end</code></td>
<td>
<p>end position</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_showtype">showType</code></td>
<td>
<p>character vector, eg.: &quot;CDS&quot;, &quot;five_prime_UTR&quot;,
&quot;three_prime_UTR&quot;</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_cds.height">CDS.height</code></td>
<td>
<p>numeric indicate the height of CDS in gene model,
range: <code style="white-space: pre;">&#8288;[0,1]&#8288;</code></p>
</td></tr>
<tr><td><code id="plotEFF_+3A_cex">cex</code></td>
<td>
<p>a numeric control the size of point</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_col">col</code></td>
<td>
<p>vector specified the color bar</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_pch">pch</code></td>
<td>
<p>vector controls points type, see
<code><a href="graphics.html#topic+par">par()</a></code></p>
</td></tr>
<tr><td><code id="plotEFF_+3A_main">main</code></td>
<td>
<p>main title</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_legend.cex">legend.cex</code></td>
<td>
<p>a numeric control the legend size</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_gene.legend">gene.legend</code></td>
<td>
<p>whether add legend for gene model</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_markmutants">markMutants</code></td>
<td>
<p>whether mark mutants on gene model, default as <code>TRUE</code></p>
</td></tr>
<tr><td><code id="plotEFF_+3A_mutants.col">mutants.col</code></td>
<td>
<p>color of lines which mark mutants</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_mutants.type">mutants.type</code></td>
<td>
<p>a vector of line types</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_y">y</code>, <code id="plotEFF_+3A_ylab">ylab</code>, <code id="plotEFF_+3A_legendtitle">legendtitle</code></td>
<td>
<p><em>y:</em> indicate either pvalue or effect should be used as y axix,
<strong>ylab,legendtitle:</strong>,character, if missing, the value will be decide by y.</p>
</td></tr>
<tr><td><code id="plotEFF_+3A_par.restore">par.restore</code></td>
<td>
<p>default as <code>TRUE</code>, wether restore the origin par after ploted EFF.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("geneHapR_test")

# calculate site functional effect
# siteEFF &lt;- siteEFF(hapResult, pheno, names(pheno))
# plotEFF(siteEFF, gff = gff, Chr = "scaffold_1")

</code></pre>

<hr>
<h2 id='plotHapNet'>plotHapNet</h2><span id='topic+plotHapNet'></span>

<h3>Description</h3>

<p>plotHapNet
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotHapNet(
  hapNet,
  size = "freq",
  scale = 1,
  cex = 0.8,
  cex.legend = 0.6,
  col.link = 1,
  link.width = 1,
  show.mutation = 2,
  backGround = backGround,
  hapGroup = hapGroup,
  legend = FALSE,
  show_size_legend = TRUE,
  show_color_legend = TRUE,
  pie.lim = c(0.5, 2),
  main = main,
  labels = TRUE,
  legend_version = 0,
  labels.cex = 0.8,
  labels.col = "blue",
  labels.adj = NULL,
  labels.font = 2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotHapNet_+3A_hapnet">hapNet</code></td>
<td>
<p>an object of class &quot;haploNet&quot;</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_size">size</code></td>
<td>
<p>a numeric vector giving the diameter of the circles
representing the haplotypes: this is in the same unit than the
links and eventually recycled.</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_scale">scale</code></td>
<td>
<p>a numeric indicate the ratio of the scale of the links
representing the number of steps on the scale of the circles
representing the haplotypes or a character one of <code>c('log10', 'log2')</code>
indicate the scale method by <code>log10(size)</code> or <code>log2(size)</code>, respectively.
Default as 1</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_cex">cex</code></td>
<td>
<p>character expansion factor relative to current par(&quot;cex&quot;)</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_cex.legend">cex.legend</code></td>
<td>
<p>same as <code>cex</code>, but for text in legend</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_col.link">col.link</code></td>
<td>
<p>a character vector specifying the colours of the links;
eventually recycled.</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_link.width">link.width</code></td>
<td>
<p>a numeric vector giving the width of the links;
eventually recycled.</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_show.mutation">show.mutation</code></td>
<td>
<p>an integer value:
</p>
<p>if 0, nothing is drawn on the links;
</p>
<p>if 1, the mutations are shown with small segments on the links;
</p>
<p>if 2, they are shown with small dots;
</p>
<p>if 3, the number of mutations are printed on the links.</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_background">backGround</code></td>
<td>
<p>a color vector with length equal to number of
Accession types</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_hapgroup">hapGroup</code></td>
<td>
<p>a matrix used to draw pie charts for each haplotype;
its number of rows must be equal to the number of haplotypes</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_legend">legend</code></td>
<td>
<p>a logical specifying whether to draw the legend,
or a vector of length two giving the coordinates where to draw the legend;
<code>FALSE</code> by default.
If <code>TRUE</code>, the user is asked to click where to draw the legend.</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_show_size_legend">show_size_legend</code>, <code id="plotHapNet_+3A_show_color_legend">show_color_legend</code></td>
<td>
<p>wether show size or color legend</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_pie.lim">pie.lim</code></td>
<td>
<p>A numeric vector define the maximum and minmum pie size,
which will be avoid the pie to samll or too large</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_main">main</code></td>
<td>
<p>The main title (on top) using font, size (character
expansion) and color <code>par(c("font.main", "cex.main", "col.main"))</code>.</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_labels">labels</code></td>
<td>
<p>a logical specifying whether to identify the haplotypes
with their labels (default as TRUE)</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_legend_version">legend_version</code></td>
<td>
<p>the size legened style, default as 0</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_labels.cex">labels.cex</code></td>
<td>
<p>the size of labels</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_labels.col">labels.col</code></td>
<td>
<p>the labels color</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_labels.adj">labels.adj</code></td>
<td>
<p>a named list contains two length vectors defining the adjustment of labels.
The names should be exactly matched with the haplotype names.
default as NULL.</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_labels.font">labels.font</code></td>
<td>
<p>the font of labels, default as 2</p>
</td></tr>
<tr><td><code id="plotHapNet_+3A_...">...</code></td>
<td>
<p>other parameters will pass to <code>plot</code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Additional parameters control the network features:
labels.cex = 1,
labels.font = 2,
link.color = &quot;black&quot;,
link.type = 1,
link.type.alt = 2,
link.width = 1,
link.width.alt = 1,
altlinks = TRUE,
threshold = c(1,2),
haplotype.inner.color = &quot;white&quot;,
haplotype.outer.color = &quot;black&quot;,
mutations.cex = 1,
mutations.font = 1,
mutations.frame.background = &quot;#0000FF4D&quot;,
mutations.frame.border = &quot;black&quot;,
mutations.text.color = 1,
mutations.arrow.color = &quot;black&quot;,
mutations.arrow.type = &quot;triangle&quot;,
mutations.sequence.color = &quot;#BFBFBF4D&quot;,
mutations.sequence.end = &quot;round&quot;,
mutations.sequence.length = 0.3,
mutations.sequence.width = 5,
pie.inner.segments.color = &quot;black&quot;,
pie.colors.function = rainbow,
scale.ratio = 1,
show.mutation = 2
</p>
<p>The alter links could be eliminated by set the 'threshold' to 0 or set 'altlinks' as FALSE.
</p>


<h3>Value</h3>

<p>No return value
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hap_summary">hap_summary()</a></code> and
<code><a href="#topic+get_hapNet">get_hapNet()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data("geneHapR_test")
hapSummary &lt;- hap_summary(hapResult)

# calculate haploNet
hapNet &lt;- get_hapNet(hapSummary,
                     AccINFO = AccINFO, # accession types
                     groupName = colnames(AccINFO)[2])

# plot haploNet
plot(hapNet)

# plot haploNet
plotHapNet(hapNet,
           size = "freq",   # circle size
           scale = "log10", # scale circle with 'log10(size + 1)'
           cex = 1, # size of hap symbol
           col.link = 2, # link colors
           link.width = 2, # link widths
           show.mutation = 2, # mutation types one of c(0,1,2,3)
           legend = FALSE) # legend position

</code></pre>

<hr>
<h2 id='plotHapTable'>plotHapTable</h2><span id='topic+plotHapTable'></span>

<h3>Description</h3>

<p>display hap result as a table-like figure
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotHapTable(hapSummary,
             hapPrefix = "H",
             title = "",
             geneName = geneName,
             INFO_tag = NULL,
             tag_split = tag_split,
             tag_field = tag_field,
             tag_name = tag_name,
             displayIndelSize = 0, angle = c(0,45,90),
             replaceMultiAllele = TRUE,
             ALLELE.color = "grey90")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotHapTable_+3A_hapsummary">hapSummary</code></td>
<td>
<p>object of hapSummary class</p>
</td></tr>
<tr><td><code id="plotHapTable_+3A_happrefix">hapPrefix</code></td>
<td>
<p>prefix of haplotype names. Default as &quot;H&quot;</p>
</td></tr>
<tr><td><code id="plotHapTable_+3A_title">title</code></td>
<td>
<p>the main title of the final figure</p>
</td></tr>
<tr><td><code id="plotHapTable_+3A_genename">geneName</code></td>
<td>
<p>character, will be used for filter INFO filed of ANN</p>
</td></tr>
<tr><td><code id="plotHapTable_+3A_info_tag">INFO_tag</code></td>
<td>
<p>The annotations in the INFO field are represented as tag-value
pairs, where the tag and value are separated by an equal sign, ie &quot;=&quot;, and
pairs are separated by colons, ie &quot;;&quot;.
For more information please see details.</p>
</td></tr>
<tr><td><code id="plotHapTable_+3A_tag_split">tag_split</code></td>
<td>
<p>usually, the value of tag-value contains one information.
However, if a tag contains more than one fields, eg &quot;ANN&quot;, then <code>tag_split</code> is
needed. When <code>INFO_tag</code> was set as &quot;ANN&quot; or &quot;SNPEFF&quot;, <code>tag_split</code> will be set
as &quot;|&quot; by default, see details.</p>
</td></tr>
<tr><td><code id="plotHapTable_+3A_tag_field">tag_field</code></td>
<td>
<p>integer, if a tag-value contains more than one fields,
user need to specified which field should be display. If <code>tag_field</code> set as
0, the whole contents will be displayed. Default as 0.</p>
</td></tr>
<tr><td><code id="plotHapTable_+3A_tag_name">tag_name</code></td>
<td>
<p>tag name is displayed in Hap figure. If <code>tag_name</code> is
missing, will take the value of <code>INFO_tag</code>.</p>
</td></tr>
<tr><td><code id="plotHapTable_+3A_displayindelsize">displayIndelSize</code></td>
<td>
<p>display indels with max size of <code>displayIndelSize</code>,
If set as 0, all indels will convert into &quot;i*&quot; of which &quot;i&quot; represents &quot;indel&quot;.</p>
</td></tr>
<tr><td><code id="plotHapTable_+3A_angle">angle</code></td>
<td>
<p>the angle of coordinates, should be one of 0, 45 and 90</p>
</td></tr>
<tr><td><code id="plotHapTable_+3A_replacemultiallele">replaceMultiAllele</code></td>
<td>
<p>whether to replace MultiAllele with &quot;T*&quot;,
default as <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotHapTable_+3A_allele.color">ALLELE.color</code></td>
<td>
<p>the color of ALLELE row, default as &quot;grey90&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In <strong>VCF</strong> files, the INFO field are represented as tag-value
pairs, where the tag and value are separated by an equal sign, ie &quot;=&quot;, and
pairs are separated by colons, ie &quot;;&quot;.
</p>
<p>If hapSummarys were generated from sequences, INFO row is null.
If hapSummarys were generated from VCF, INFO was take from the INFO column
in the source VCF file.
Some tag-values may contains more than one value separated by
&quot;|&quot;, eg.: &quot;ANN&quot; or &quot;snpEFF&quot; added by 'snpeff' or other software. For those
fields we need specified value of <code>tag_field = "ANN"</code> and <code>tag_split = "[\|]"</code>,
it's suggest specified the value of <code>tag_name</code> for display in figure.
</p>
<p>'snpeff', a toolbox for genetic variant annotation and functional effect
prediction, will add annotations to INFO filed in VCF file under a tag
named as &quot;ANN&quot;. The annotations contains several fields separated by &quot;|&quot;.
eg.:
</p>

<ol>
<li><p> Allele
</p>
</li>
<li><p> Annotation
</p>
</li>
<li><p> Annotation_Impact
</p>
</li>
<li><p> Gene_Name
</p>
</li>
<li><p> Gene_ID
</p>
</li>
<li><p> Feature_Type
</p>
</li>
<li><p> Feature_ID
</p>
</li>
<li><p> Transcript_BioType
</p>
</li>
<li><p> Rank
</p>
</li>
<li><p> HGVS.c
</p>
</li>
<li><p> HGVS.p
</p>
</li>
<li><p> cDNA.pos/cDNA.length
... ...
</p>
</li></ol>

<p>However, the INFO in hapResults may missing annotations that we need.
In this case, we can custom INFOs in hapSummarys with <code>addINFO()</code>.
Once the needed annotations were included in hap results, we can display
them with <code>plotHapTable()</code> by specify the value of <code>INFO_tag</code>.
</p>


<h3>Value</h3>

<p>ggplot2 object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addINFO">addINFO()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("geneHapR_test")
plotHapTable(hapResult)
</code></pre>

<hr>
<h2 id='plotHapTable2'>plotHapTable2</h2><span id='topic+plotHapTable2'></span>

<h3>Description</h3>

<p>plot the hapResult in table like style using grid system.
This function is under development and may not stable.
Some parameters may deleted or renamed in future.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotHapTable2(
  hapSummary,
  show_indel_size = 1,
  replaceMultiAllele = TRUE,
  angle = 0,
  show_INFO = FALSE,
  INFO_split = c(";", ",", "\\|"),
  INFO_tag = "ANN",
  geneID = NA,
  tag_field = -1,
  title = "",
  gff = NULL,
  show_chr_name = TRUE,
  Chr = NULL,
  start = NULL,
  end = NULL,
  model_rect_col = "black",
  model_rect_fill = "grey80",
  model_line_col = "black",
  model_anno_txt = NULL,
  model_anno_col = "black",
  model_anno_cex = 1,
  table_txt_col = "black",
  table_txt_cex = 1,
  model_anno_pos = c(-1, -1),
  model_anno_adj = c(0, 1),
  gene_model_height = 0.2,
  space_height = 0.1,
  table_height = NULL,
  CDS_height = 0.3,
  link_line_type = 3,
  headrows = 1,
  equal_col_width = FALSE,
  head_anno = 1,
  col_annots = 0,
  row_labels = 1,
  row_annots = 1,
  table_line_col = "white",
  annot_for_each_transcrips = TRUE,
  labels_fill = "white",
  annot_fill = "grey90",
  head_fill = NULL,
  cell_fill = NULL,
  style = gpar(fontfamily = "sans", fontface = 1, cex = 0.7),
  footbar = "",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotHapTable2_+3A_hapsummary">hapSummary</code></td>
<td>
<p>the hapSummary or hapResult object</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_show_indel_size">show_indel_size</code></td>
<td>
<p>the Indel length longer will be replaced by &quot;i1,i2,i3,...&quot;</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_replacemultiallele">replaceMultiAllele</code></td>
<td>
<p>replace multi-allele title by 'T1,T2,...' or not</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_angle">angle</code></td>
<td>
<p>the angle of number positions</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_show_info">show_INFO</code></td>
<td>
<p>show annotation field or not, default as <code>FALSE</code></p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_info_split">INFO_split</code>, <code id="plotHapTable2_+3A_info_tag">INFO_tag</code>, <code id="plotHapTable2_+3A_geneid">geneID</code>, <code id="plotHapTable2_+3A_tag_field">tag_field</code></td>
<td>
<p>used to set annotation in haplotype table.
And the geneID was used to fileter annotation in INFO field.</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_title">title</code></td>
<td>
<p>title of plot</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_gff">gff</code></td>
<td>
<p>gff or bed annotation</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_show_chr_name">show_chr_name</code></td>
<td>
<p>show chromosome name at left-top cell or not</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_chr">Chr</code>, <code id="plotHapTable2_+3A_start">start</code>, <code id="plotHapTable2_+3A_end">end</code></td>
<td>
<p>which range should be plotted in gene model</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_model_rect_col">model_rect_col</code>, <code id="plotHapTable2_+3A_model_rect_fill">model_rect_fill</code>, <code id="plotHapTable2_+3A_model_line_col">model_line_col</code></td>
<td>
<p>a string specified the color for line/rectangle in gene model</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_table_txt_col">table_txt_col</code>, <code id="plotHapTable2_+3A_table_txt_cex">table_txt_cex</code></td>
<td>
<p>controls the color and size of texts in genotype table</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_model_anno_pos">model_anno_pos</code>, <code id="plotHapTable2_+3A_model_anno_adj">model_anno_adj</code>, <code id="plotHapTable2_+3A_model_anno_cex">model_anno_cex</code>, <code id="plotHapTable2_+3A_model_anno_col">model_anno_col</code>, <code id="plotHapTable2_+3A_model_anno_txt">model_anno_txt</code></td>
<td>
<p>the position (x,y), just (hjust, vjust), color, size and content of annotation text in gene model</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_gene_model_height">gene_model_height</code>, <code id="plotHapTable2_+3A_table_height">table_height</code>, <code id="plotHapTable2_+3A_space_height">space_height</code></td>
<td>
<p>the plotting range height of gene model, table and spacer</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_cds_height">CDS_height</code></td>
<td>
<p>a numeric vector specified the height of CDS, and the height of utr is half of that,
only useful when gff is provided,</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_link_line_type">link_line_type</code></td>
<td>
<p>the type of link lines for mutations in gene model and genotype table</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_equal_col_width">equal_col_width</code></td>
<td>
<p>a bool or numeric vector specified whether column with should be equal</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_col_annots">col_annots</code>, <code id="plotHapTable2_+3A_head_anno">head_anno</code>, <code id="plotHapTable2_+3A_headrows">headrows</code>, <code id="plotHapTable2_+3A_row_annots">row_annots</code>, <code id="plotHapTable2_+3A_row_labels">row_labels</code></td>
<td>
<p>the column or row number of annotation or labels or heads</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_table_line_col">table_line_col</code></td>
<td>
<p>the line color in genotype table</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_annot_for_each_transcrips">annot_for_each_transcrips</code></td>
<td>
<p>mark the strand and trancripts name for each gene modle</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_annot_fill">annot_fill</code>, <code id="plotHapTable2_+3A_head_fill">head_fill</code>, <code id="plotHapTable2_+3A_labels_fill">labels_fill</code></td>
<td>
<p>the fill color of annotation, head and label row or columns</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_cell_fill">cell_fill</code></td>
<td>
<p>a color vector or function or named vector specified cell fill color</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_style">style</code></td>
<td>
<p>see help(gpar)</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_footbar">footbar</code></td>
<td>
<p>the foot notes</p>
</td></tr>
<tr><td><code id="plotHapTable2_+3A_...">...</code></td>
<td>
<p>param not used</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#
data(geneHapR_test)
plotHapTable2(hapResult)
plotHapTable2(hapResult, gff = gff)
</code></pre>

<hr>
<h2 id='seqs2hap'>Generate Hap Results from Seqs</h2><span id='topic+seqs2hap'></span><span id='topic+trimSeqs'></span>

<h3>Description</h3>

<p>generate hapResults from aligned and trimed sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seqs2hap(
  seqs,
  Ref = names(seqs)[1],
  hetero_remove = TRUE,
  na_drop = TRUE,
  maxGapsPerSeq = 0.25,
  hapPrefix = "H",
  pad = 3,
  chrName = "Chr0",
  ...
)

trimSeqs(seqs,
         minFlankFraction = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seqs2hap_+3A_seqs">seqs</code></td>
<td>
<p>object of DNAStringSet or DNAMultipleAlignment class</p>
</td></tr>
<tr><td><code id="seqs2hap_+3A_ref">Ref</code></td>
<td>
<p>the name of reference sequences.
Default as the name of the first sequence</p>
</td></tr>
<tr><td><code id="seqs2hap_+3A_hetero_remove">hetero_remove</code></td>
<td>
<p>whether remove accessions contains hybrid site or not.
Default as <code>TRUE</code></p>
</td></tr>
<tr><td><code id="seqs2hap_+3A_na_drop">na_drop</code></td>
<td>
<p>whether drop sequeces contain &quot;N&quot;
Default as <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="seqs2hap_+3A_maxgapsperseq">maxGapsPerSeq</code></td>
<td>
<p>value in <code style="white-space: pre;">&#8288;[0, 1]&#8288;</code> that indicates the maximum
fraction of gaps allowed in each seq after alignment (default as 0.25).
Seqs with gap percent exceed that will be dropped</p>
</td></tr>
<tr><td><code id="seqs2hap_+3A_happrefix">hapPrefix</code></td>
<td>
<p>prefix of hap names. Default as &quot;H&quot;</p>
</td></tr>
<tr><td><code id="seqs2hap_+3A_pad">pad</code></td>
<td>
<p>The number length in haplotype names should be extend to.</p>
</td></tr>
<tr><td><code id="seqs2hap_+3A_chrname">chrName</code></td>
<td>
<p>the Name should be used for haplotype</p>
</td></tr>
<tr><td><code id="seqs2hap_+3A_...">...</code></td>
<td>
<p>Parameters not used.</p>
</td></tr>
<tr><td><code id="seqs2hap_+3A_minflankfraction">minFlankFraction</code></td>
<td>
<p>A value in <code style="white-space: pre;">&#8288;[0, 1]&#8288;</code> that indicates the minimum
fraction needed to call a gap in the consensus string (default as 0.1).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of hapResult class
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("geneHapR_test")
seqs
seqs &lt;- trimSeqs(seqs,
         minFlankFraction = 0.1)
seqs
hapResult &lt;- seqs2hap(seqs,
                      Ref = names(seqs)[1],
                      hetero_remove = TRUE, na_drop = TRUE,
                      maxGapsPerSeq = 0.25,
                      hapPrefix = "H")

</code></pre>

<hr>
<h2 id='SetATGas0'>Set Position of ATG as Zero</h2><span id='topic+SetATGas0'></span><span id='topic+gffSetATGas0'></span><span id='topic+hapSetATGas0'></span>

<h3>Description</h3>

<p>Set position of ATG as zero in hap result and gff annotation.
The upstream was negative while the gene range and downstream was positive.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gffSetATGas0(gff = gff, hap = hap,
             geneID = geneID,
             Chr = Chr, POS = POS)

hapSetATGas0(gff = gff, hap = hap,
             geneID = geneID,
             Chr = Chr, POS = POS)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SetATGas0_+3A_gff">gff</code></td>
<td>
<p>gene annotations</p>
</td></tr>
<tr><td><code id="SetATGas0_+3A_hap">hap</code></td>
<td>
<p>object of hapResult or hapSummary class</p>
</td></tr>
<tr><td><code id="SetATGas0_+3A_geneid">geneID</code></td>
<td>
<p>geneID</p>
</td></tr>
<tr><td><code id="SetATGas0_+3A_chr">Chr</code></td>
<td>
<p>Chromsome name</p>
</td></tr>
<tr><td><code id="SetATGas0_+3A_pos">POS</code></td>
<td>
<p>vector consist with start and end position</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Filter hap result and gff annotation according to provided information.
And then set position of ATG as zero in hap result and gff annotation.
The upstream was negative while the gene range and downstream was positive.
</p>
<p><strong>Notice:</strong> the position of &quot;ATG&quot; after modified was 0, 1 and 2 separately.
The site in hap result exceed the selected range will be <strong>dropped</strong>.
</p>


<h3>Value</h3>

<p><code>gffSetATGas0</code>: filtered gff with position of ATG was as zero
</p>
<p><code>hapSetATGas0</code>: hap results with position of ATG was set as zero
</p>


<h3>See Also</h3>

<p><code><a href="#topic+displayVarOnGeneModel">displayVarOnGeneModel()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load example dataset
data("geneHapR_test")


# set position of ATG as zero in gff
newgff &lt;- gffSetATGas0(gff = gff, hap = hapResult,
                       geneID = "test1G0387",
                       Chr = "scaffold_1",
                       POS = c(4300, 7910))


data("geneHapR_test")

# set position of ATG as zero in hap results
newhapResult &lt;- hapSetATGas0(gff = gff, hap = hapResult,
                             geneID = "test1G0387",
                             Chr = "scaffold_1",
                             POS = c(4300, 7910))
</code></pre>

<hr>
<h2 id='siteEFF'>Calculation of Sites Effective</h2><span id='topic+siteEFF'></span>

<h3>Description</h3>

<p>Calculation of Sites Effective
</p>


<h3>Usage</h3>

<pre><code class='language-R'>siteEFF(hap, pheno, phenoNames, quality = FALSE, method = "auto",
        p.adj = "none")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="siteEFF_+3A_hap">hap</code></td>
<td>
<p>object of &quot;hapResult&quot; class</p>
</td></tr>
<tr><td><code id="siteEFF_+3A_pheno">pheno</code></td>
<td>
<p>phenotype data, with column names as pheno name
and row name as accessions.</p>
</td></tr>
<tr><td><code id="siteEFF_+3A_phenonames">phenoNames</code></td>
<td>
<p>pheno names used for analysis, if missing,
will use all pheno names in <code>pheno</code></p>
</td></tr>
<tr><td><code id="siteEFF_+3A_quality">quality</code></td>
<td>
<p>bool type, indicate whther the type of phenos are quality or
quantitative. Length of <code>quality</code> could be 1 or equal with length of
<code>phenoNames</code>. Default as <code>FALSE</code></p>
</td></tr>
<tr><td><code id="siteEFF_+3A_method">method</code></td>
<td>
<p>character or character vector with length equal with
<code>phenoNames</code> indicate which method should be performed towards each
phenotype. Should be one of &quot;t.test&quot;, &quot;chi.test&quot;, &quot;wilcox.test&quot; and &quot;auto&quot;.
Default as &quot;auto&quot;, see details.</p>
</td></tr>
<tr><td><code id="siteEFF_+3A_p.adj">p.adj</code></td>
<td>
<p>character, indicate correction method.
Could be &quot;BH&quot;, &quot;BY&quot;, &quot;none&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The site <strong>EFF</strong> was determinate by the phenotype difference between each
site geno-type.
</p>
<p>The <em>p</em> was calculated with statistical analysis method as designated by the
parameter <code>method</code>. If <code>method</code> set as &quot;auto&quot;, then
chi.test will be
selected for quantity phenotype, eg.: color;
for quantity phynotype, eg.: height, with at least 30 observations per
geno-type and fit Gaussian distribution t.test will be performed, otherwise
wilcox.test will be performed.
</p>


<h3>Value</h3>

<p>a list containing two matrix names as &quot;p&quot; and &quot;EFF&quot;,
with column name are pheno names and row name are site position.
The matrix names as &quot;p&quot; contains all <em>p</em>-value.
The matrix named as &quot;EFF&quot; contains scaled difference between each geno-types
per site.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("geneHapR_test")

# calculate site functional effect
# siteEFF &lt;- siteEFF(hapResult, pheno, names(pheno))
# plotEFF(siteEFF, gff = gff, Chr = "scaffold_1")

</code></pre>

<hr>
<h2 id='sites_compar'>sites comparison</h2><span id='topic+sites_compar'></span><span id='topic+compareAllSites'></span>

<h3>Description</h3>

<p>Used for all allele effect compare once
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareAllSites(
  hap,
  pheno,
  phenoName = names(pheno)[1],
  hetero_remove = TRUE,
  title = "",
  file = file
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sites_compar_+3A_hap">hap</code></td>
<td>
<p>object of hapResult class</p>
</td></tr>
<tr><td><code id="sites_compar_+3A_pheno">pheno</code></td>
<td>
<p>a data.frame contains phenotypes</p>
</td></tr>
<tr><td><code id="sites_compar_+3A_phenoname">phenoName</code></td>
<td>
<p>the name of used phenotype</p>
</td></tr>
<tr><td><code id="sites_compar_+3A_hetero_remove">hetero_remove</code></td>
<td>
<p>removing the heter-sites or not, default as TRUE</p>
</td></tr>
<tr><td><code id="sites_compar_+3A_title">title</code></td>
<td>
<p>the title of the figure</p>
</td></tr>
<tr><td><code id="sites_compar_+3A_file">file</code></td>
<td>
<p>if provieds a file path the comparing results will saved to file.</p>
</td></tr>
</table>

<hr>
<h2 id='table2hap'>table2hap</h2><span id='topic+table2hap'></span>

<h3>Description</h3>

<p>convert variants stored in table format into hapResult
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table2hap(x, hapPrefix = "H", pad = 3, hetero_remove = TRUE, na_drop = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="table2hap_+3A_x">x</code></td>
<td>
<p>a data.frame contains variants information.
The first file column are fix as Chrome name, position, reference nuclieotide,
alter nuclieotide and INFO. Accession genotype should be in followed columns.
&quot;-&quot; will be treated as Indel. &quot;.&quot; and &quot;N&quot; will be treated as missing data.
Heterozygotes should be &quot;A/T&quot;, &quot;AAA/A&quot;</p>
</td></tr>
<tr><td><code id="table2hap_+3A_happrefix">hapPrefix</code></td>
<td>
<p>prefix of haplotype names</p>
</td></tr>
<tr><td><code id="table2hap_+3A_pad">pad</code></td>
<td>
<p>The number length in haplotype names should be extend to.</p>
</td></tr>
<tr><td><code id="table2hap_+3A_hetero_remove">hetero_remove</code></td>
<td>
<p>whether remove accessions contains hyb-sites, Character not A T C G</p>
</td></tr>
<tr><td><code id="table2hap_+3A_na_drop">na_drop</code></td>
<td>
<p>whether drop accessions contains missing data (&quot;N&quot;, &quot;NA&quot;, &quot;.&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of hapSummary class
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
   data("geneHapR_test")
   hapResult &lt;- table2hap(gt.geno, hapPrefix = "H",
                          hetero_remove = TRUE,
                          na_drop = TRUE)

</code></pre>

<hr>
<h2 id='vcf2hap'>Generat Haps from VCF</h2><span id='topic+vcf2hap'></span>

<h3>Description</h3>

<p>Generate hapResult from vcfR object
A simple filter by position was provided in this function,
however it's prefer to filter VCF (vcfR object) through
<code><a href="#topic+filter_vcf">filter_vcf()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf2hap(
  vcf,
  hapPrefix = "H",
  filter_Chr = FALSE,
  filter_POS = FALSE,
  pad = 3,
  hetero_remove = TRUE,
  na_drop = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcf2hap_+3A_vcf">vcf</code></td>
<td>
<p>vcfR object imported by <code>import_vcf()</code></p>
</td></tr>
<tr><td><code id="vcf2hap_+3A_happrefix">hapPrefix</code></td>
<td>
<p>prefix of hap names, default as &quot;H&quot;</p>
</td></tr>
<tr><td><code id="vcf2hap_+3A_filter_chr">filter_Chr</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="vcf2hap_+3A_filter_pos">filter_POS</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="vcf2hap_+3A_pad">pad</code></td>
<td>
<p>The number length in haplotype names should be extend to.</p>
</td></tr>
<tr><td><code id="vcf2hap_+3A_hetero_remove">hetero_remove</code></td>
<td>
<p>whether remove accessions contains hybrid site or not.
Default as <code>TRUE</code></p>
</td></tr>
<tr><td><code id="vcf2hap_+3A_na_drop">na_drop</code></td>
<td>
<p>whether remove accessions contains unknown allele site or not
Default as <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="vcf2hap_+3A_...">...</code></td>
<td>
<p>Parameters not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of hapResult class
</p>


<h3>Author(s)</h3>

<p>Zhangrenl
</p>


<h3>See Also</h3>

<p>extract genotype from vcf:
<code><a href="vcfR.html#topic+extract_gt_tidy">vcfR::extract_gt_tidy()</a></code>,
import vcf files:
<code><a href="#topic+import_vcf">import_vcf()</a></code> (preferred) and
<code><a href="vcfR.html#topic+read.vcfR">vcfR::read.vcfR()</a></code>,
filter vcf according <strong>position</strong> and <strong>annotations</strong>:
<code><a href="#topic+filter_vcf">filter_vcf()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("geneHapR_test")
hapResult &lt;- vcf2hap(vcf)
</code></pre>

<hr>
<h2 id='write.hap'>Save Haplotype Results to Disk</h2><span id='topic+write.hap'></span>

<h3>Description</h3>

<p>This function will write hap result into a txt file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.hap(x, file = file, sep = "\t", pheno = pheno, phenoName = phenoName)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.hap_+3A_x">x</code></td>
<td>
<p>objec of hapResult or hapSummary class</p>
</td></tr>
<tr><td><code id="write.hap_+3A_file">file</code></td>
<td>
<p>file path, where to save the hap result/summary</p>
</td></tr>
<tr><td><code id="write.hap_+3A_sep">sep</code></td>
<td>
<p>the field separator string. Values within each row of x are separated by this string.
Default as &quot;<code style="white-space: pre;">&#8288;\t&#8288;</code>&quot;</p>
</td></tr>
<tr><td><code id="write.hap_+3A_pheno">pheno</code>, <code id="write.hap_+3A_phenoname">phenoName</code></td>
<td>
<p>the phenotype data.frame, only used for export hapResult object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The hap result and hap summary result have common features.
The common features of these two types are:
First four rows contains extra information: CHR, POS, INFO and ALLELE
Hap names were in the first column.
The differences are:
Hap summary result have a freq column while hap result not.
Rows represent haplotypes in hap summary result, while rows represent accessions in hap result.
In addtion, the accessions of each haplotype in hap summary result were separated by &quot;;&quot;.
</p>


<h3>Value</h3>

<p>No return value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

oriDir &lt;- getwd()
 temp_dir &lt;- tempdir()
 if(! dir.exists(temp_dir))
   dir.create(temp_dir)
 setwd(temp_dir)
data("geneHapR_test")
write.hap(hapResult, file = "hapResult.txt")
setwd(oriDir)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
