<!DOCTYPE html><html><head><title>Help for package genekitr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {genekitr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#as.enrichdat'><p>Modify dataframe for enrichment plot</p></a></li>
<li><a href='#Datasets'><p>Datasets</p>
geneList entrez gene list with decreasing fold change value</a></li>
<li><a href='#expoSheet'><p>Export list of data sets into different 'Excel' sheets</p></a></li>
<li><a href='#genGSEA'><p>Gene Set Enrichment Analysis</p></a></li>
<li><a href='#genInfo'><p>Get gene related information</p></a></li>
<li><a href='#genORA'><p>Gene Over-Representation Enrichment Analysis</p></a></li>
<li><a href='#getGEO'><p>Get GEO metadata</p></a></li>
<li><a href='#getPubmed'><p>Get 'PubMed' paper records by searching abstract</p></a></li>
<li><a href='#importCP'><p>Import 'clusterProfiler' result</p></a></li>
<li><a href='#importPanther'><p>Import 'Panther' web result</p></a></li>
<li><a href='#importShinygo'><p>Import 'shinyGO' web result</p></a></li>
<li><a href='#plot_theme'><p>Themes for all plots</p></a></li>
<li><a href='#plotEnrich'><p>Plot for gene enrichment analysis of ORA method</p></a></li>
<li><a href='#plotEnrichAdv'><p>Advanced Plot for  gene enrichment analysis of ORA method</p></a></li>
<li><a href='#plotGSEA'><p>Plot for gene enrichment analysis of GSEA method</p></a></li>
<li><a href='#plotVenn'><p>Venn plot for groups of genes</p></a></li>
<li><a href='#plotVolcano'><p>Volcano plot for differential expression analysis</p></a></li>
<li><a href='#simGO'><p>Simplify GO enrichment result</p></a></li>
<li><a href='#transId'><p>Transform id among symbol, entrezid,  ensembl and uniprot.</p></a></li>
<li><a href='#transProbe'><p>Transform probe id to symbol, entrezid,  ensembl or uniprot.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Analysis Toolkit</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.5</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yunze Liu &lt;jieandze1314@gmail.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.genekitr.fun/">https://www.genekitr.fun/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/GangLiLab/genekitr/issues">https://github.com/GangLiLab/genekitr/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>An analysis toolkit focused on genes. It mainly includes five
    features (search, convert, analysis, plot, and export).
    The user just needs to input feature id ('entrez', 'symbol', 'ensembl' or
    'uniprot') to retrieve feature information and 'PubMed'<a href="https://pubmed.ncbi.nlm.nih.gov/">https://pubmed.ncbi.nlm.nih.gov/</a> records, 
    to convert id types, to easily handle gene enrichment analysis and publication-level figures, 
    to plot group interaction and export results as sheets in one 'excel'
    file to easily share and communicate with others.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>Imports:</td>
<td>clusterProfiler, dplyr, europepmc, fst, geneset, ggplot2,
ggraph, ggvenn, igraph, magrittr, openxlsx, stringr, stringi,
tidyr, rlang</td>
</tr>
<tr>
<td>Suggests:</td>
<td>AnnotationDbi, BiocManager, cowplot, ComplexUpset,
data.table, fuzzyjoin, forcats, fgsea, futile.logger,
ggplotify, ggsci, ggupset, ggrepel, ggridges, ggnewscale,
GOplot, GOSemSim, labeling, msigdbr, pheatmap, pracma, tm,
treemap, RColorBrewer, rappdirs, RCurl, reshape2, rio, rrvgo,
scales, stats, testthat (&ge; 3.0.0), tibble, wordcloud, knitr,
rmarkdown, XML, xml2, httr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-28 00:52:49 UTC; reedliu1</td>
</tr>
<tr>
<td>Author:</td>
<td>Yunze Liu <a href="https://orcid.org/0000-0002-7414-8556"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-07 08:50:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling 'rhs(lhs)'.
</p>

<hr>
<h2 id='as.enrichdat'>Modify dataframe for enrichment plot</h2><span id='topic+as.enrichdat'></span>

<h3>Description</h3>

<p>To make sure colname contains Description, Count, FoldEnrich/GeneRatio, pvalue/qvalue/p.adjust
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.enrichdat(enrich_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.enrichdat_+3A_enrich_df">enrich_df</code></td>
<td>
<p>Enrichment analysis 'data.frame' result.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'data.frame'
</p>

<hr>
<h2 id='Datasets'>Datasets
geneList entrez gene list with decreasing fold change value</h2><span id='topic+Datasets'></span><span id='topic+geneList'></span><span id='topic+deg'></span><span id='topic+msig_org'></span><span id='topic+msig_category'></span><span id='topic+biocOrg_name'></span><span id='topic+keggOrg_name'></span><span id='topic+ensOrg_name'></span><span id='topic+hsapiens_probe_platform'></span>

<h3>Description</h3>

<p>Datasets
geneList entrez gene list with decreasing fold change value
</p>
<p>Datasets
Differential expression analysis result of GSE42872
</p>
<p>Datasets
msig_species contains msigdb species information
</p>
<p>Datasets
msig_category contains msigdb category information
</p>
<p>Datasets
biocOrg_name contains organism name of bioconductor
</p>
<p>Datasets
keggOrg_name contains organism name of KEGG
https://www.genome.jp/kegg/catalog/org_list.html
</p>
<p>Datasets
ensOrg_name contains organism name of ensembl
</p>
<p>Datasets
hsapiens_probe_platform contains human probe platforms
</p>

<hr>
<h2 id='expoSheet'>Export list of data sets into different 'Excel' sheets</h2><span id='topic+expoSheet'></span>

<h3>Description</h3>

<p>Export list of data sets into different 'Excel' sheets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expoSheet(
  data_list,
  data_name,
  filename = NULL,
  dir = tempdir(),
  overwrite = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expoSheet_+3A_data_list">data_list</code></td>
<td>
<p>List of datasets.</p>
</td></tr>
<tr><td><code id="expoSheet_+3A_data_name">data_name</code></td>
<td>
<p>Character of data names.</p>
</td></tr>
<tr><td><code id="expoSheet_+3A_filename">filename</code></td>
<td>
<p>A character string naming an xlsx file.</p>
</td></tr>
<tr><td><code id="expoSheet_+3A_dir">dir</code></td>
<td>
<p>A character string naming output directory.</p>
</td></tr>
<tr><td><code id="expoSheet_+3A_overwrite">overwrite</code></td>
<td>
<p>If TRUE, overwrite any existing file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An Excel file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(openxlsx)
expoSheet(
  data_list = list(mtcars, ToothGrowth),
  data_name = c("mtcars", "tooth"),
  filename = "test.xlsx", dir = tempfile()
)
</code></pre>

<hr>
<h2 id='genGSEA'>Gene Set Enrichment Analysis</h2><span id='topic+genGSEA'></span>

<h3>Description</h3>

<p>Gene Set Enrichment Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genGSEA(
  genelist,
  geneset,
  padj_method = "BH",
  p_cutoff = 0.05,
  q_cutoff = 0.05,
  min_gset_size = 10,
  max_gset_size = 500,
  set_seed = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genGSEA_+3A_genelist">genelist</code></td>
<td>
<p>Pre-ranked genelist with decreasing order, gene can be
entrez, ensembl or symbol.</p>
</td></tr>
<tr><td><code id="genGSEA_+3A_geneset">geneset</code></td>
<td>
<p>Gene set is a two-column data.frame with term id and gene id.
Please use package 'geneset' to select available gene set or make new one.</p>
</td></tr>
<tr><td><code id="genGSEA_+3A_padj_method">padj_method</code></td>
<td>
<p>One of &quot;BH&quot;, &quot;BY&quot;, &quot;bonferroni&quot;,&quot;fdr&quot;,&quot;hochberg&quot;,
&quot;holm&quot;, &quot;hommel&quot;, &quot;none&quot;</p>
</td></tr>
<tr><td><code id="genGSEA_+3A_p_cutoff">p_cutoff</code></td>
<td>
<p>Numeric of cutoff for both unadjusted and adjusted pvalue, default is 0.05.</p>
</td></tr>
<tr><td><code id="genGSEA_+3A_q_cutoff">q_cutoff</code></td>
<td>
<p>Numeric of cutoff for qvalue, default is 0.05.</p>
</td></tr>
<tr><td><code id="genGSEA_+3A_min_gset_size">min_gset_size</code></td>
<td>
<p>Numeric of minimal size of each geneset for analyzing,
default is 10.</p>
</td></tr>
<tr><td><code id="genGSEA_+3A_max_gset_size">max_gset_size</code></td>
<td>
<p>Numeric of maximal size of each geneset for analyzing,
default is 500.</p>
</td></tr>
<tr><td><code id="genGSEA_+3A_set_seed">set_seed</code></td>
<td>
<p>GSEA permutations are performed using random reordering,
which causes slightly difference results after every time running.
If user want to get same result every time for same input,
please set 'set_seed = TRUE' or 'set.seed()' prior to running.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'data.frame'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# only gene ids
data(geneList, package = "genekitr")
gs &lt;- geneset::getGO(org = "human",ont = "mf")
gse &lt;- genGSEA(genelist = geneList, geneset = gs)


</code></pre>

<hr>
<h2 id='genInfo'>Get gene related information</h2><span id='topic+genInfo'></span>

<h3>Description</h3>

<p>Get gene related information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genInfo(
  id = NULL,
  org = "hs",
  unique = FALSE,
  keepNA = TRUE,
  hgVersion = c("v38", "v19")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genInfo_+3A_id">id</code></td>
<td>
<p>Gene id (symbol, ensembl or entrez id) or uniprot id. If this argument is NULL, return all gene info.</p>
</td></tr>
<tr><td><code id="genInfo_+3A_org">org</code></td>
<td>
<p>Latin organism shortname from 'ensOrg_name'. Default is human.</p>
</td></tr>
<tr><td><code id="genInfo_+3A_unique">unique</code></td>
<td>
<p>Logical, if one-to-many mapping occurs, only keep one record with fewest NA. Default is FALSE.</p>
</td></tr>
<tr><td><code id="genInfo_+3A_keepna">keepNA</code></td>
<td>
<p>If some id has no match at all, keep it or not. Default is TRUE.</p>
</td></tr>
<tr><td><code id="genInfo_+3A_hgversion">hgVersion</code></td>
<td>
<p>Select human genome build version from &quot;v38&quot; (default) and &quot;v19&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'data.frame'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# example1: input list with fake id and one-to-many mapping id
x &lt;- genInfo(id = c(
  "MCM10", "CDC20", "S100A9", "MMP1", "BCC7",
  "FAKEID", "TP53", "HBD", "NUDT10"
))

# example2: statistics of human gene biotypes
genInfo(org = "hs") %&gt;%
  {
    table(.$gene_biotype)
  }

# example3: use hg19 data
x &lt;- genInfo(id = c("TP53","BCC7"), hgVersion = "v19")

# example4: search genes with case-insensitive
x &lt;- genInfo(id = c("tp53","nc886","FAke","EZh2"), org = "hs", unique = TRUE)


</code></pre>

<hr>
<h2 id='genORA'>Gene Over-Representation Enrichment Analysis</h2><span id='topic+genORA'></span>

<h3>Description</h3>

<p>Gene Over-Representation Enrichment Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genORA(
  id,
  geneset,
  group_list = NULL,
  padj_method = "BH",
  p_cutoff = 0.05,
  q_cutoff = 0.15,
  min_gset_size = 10,
  max_gset_size = 500,
  universe
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genORA_+3A_id">id</code></td>
<td>
<p>A vector of gene id which can be entrezid, ensembl, symbol or uniprot.</p>
</td></tr>
<tr><td><code id="genORA_+3A_geneset">geneset</code></td>
<td>
<p>Gene set is a two-column data.frame with term id and gene id.
Please use package 'geneset' to select available gene set or make new one.</p>
</td></tr>
<tr><td><code id="genORA_+3A_group_list">group_list</code></td>
<td>
<p>A list of gene group information, default is NULL.</p>
</td></tr>
<tr><td><code id="genORA_+3A_padj_method">padj_method</code></td>
<td>
<p>One of &quot;BH&quot;, &quot;BY&quot;, &quot;bonferroni&quot;,&quot;fdr&quot;,&quot;hochberg&quot;,
&quot;holm&quot;, &quot;hommel&quot;, &quot;none&quot;</p>
</td></tr>
<tr><td><code id="genORA_+3A_p_cutoff">p_cutoff</code></td>
<td>
<p>Numeric of cutoff for both unadjusted and adjusted pvalue, default is 0.05.</p>
</td></tr>
<tr><td><code id="genORA_+3A_q_cutoff">q_cutoff</code></td>
<td>
<p>Numeric of cutoff for qvalue, default is 0.15.</p>
</td></tr>
<tr><td><code id="genORA_+3A_min_gset_size">min_gset_size</code></td>
<td>
<p>Numeric of minimal size of each geneset for analyzing,
default is 10.</p>
</td></tr>
<tr><td><code id="genORA_+3A_max_gset_size">max_gset_size</code></td>
<td>
<p>Numeric of maximal size of each geneset for analyzing,
default is 500.</p>
</td></tr>
<tr><td><code id="genORA_+3A_universe">universe</code></td>
<td>
<p>Character of background genes. If missing, all genes in
geneset will be used as background.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'data.frame'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# only gene ids
data(geneList, package = "genekitr")
id &lt;- names(geneList)[abs(geneList) &gt; 1]
gs &lt;- geneset::getGO(org = "human",ont = "mf")
ora &lt;- genORA(id, geneset = gs)

# gene id with groups
id &lt;- c(head(names(geneList), 50), tail(names(geneList), 50))
group &lt;- list(
  group1 = c(rep("up", 50), rep("down", 50)),
  group2 = c(rep("A", 20), rep("B", 30))
)
gora &lt;- genORA(id, geneset = gs, group_list = group)


</code></pre>

<hr>
<h2 id='getGEO'>Get GEO metadata</h2><span id='topic+getGEO'></span>

<h3>Description</h3>

<p>Get GEO metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGEO(searchterm, minnum = 0, maxnum = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGEO_+3A_searchterm">searchterm</code></td>
<td>
<p>input searching terms as GEO database keywords, multiple terms are seperated by blanks</p>
</td></tr>
<tr><td><code id="getGEO_+3A_minnum">minnum</code></td>
<td>
<p>The minimum return records, default is 0</p>
</td></tr>
<tr><td><code id="getGEO_+3A_maxnum">maxnum</code></td>
<td>
<p>The maximum return records, default is 1000</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'data.frame'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
meta &lt;- getGEO('ezh2 knockout',maxnum = 5)

</code></pre>

<hr>
<h2 id='getPubmed'>Get 'PubMed' paper records by searching abstract</h2><span id='topic+getPubmed'></span>

<h3>Description</h3>

<p>PubMed&lt;https://pubmed.ncbi.nlm.nih.gov/&gt; is a free search engine accessing primarily the database of references
and abstracts on life ciences and biomedical topics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPubmed(term, add_term = NULL, num = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPubmed_+3A_term">term</code></td>
<td>
<p>query terms e.g. gene id, GO/KEGG pathway</p>
</td></tr>
<tr><td><code id="getPubmed_+3A_add_term">add_term</code></td>
<td>
<p>other searching terms Default is NULL</p>
</td></tr>
<tr><td><code id="getPubmed_+3A_num">num</code></td>
<td>
<p>limit the number of records . Default is 100.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of 'tibble' for pubmed records
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
term &lt;- c("Tp53", "Brca1", "Tet2")
add_term &lt;- c("stem cell", "mouse")
l &lt;- getPubmed(term, add_term, num = 30)
# very easy to output
expoSheet(l, data_name = term, filename = "test.xlsx", dir = tempfile())

</code></pre>

<hr>
<h2 id='importCP'>Import 'clusterProfiler' result</h2><span id='topic+importCP'></span>

<h3>Description</h3>

<p>Import 'clusterProfiler' result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importCP(object, type = c("go", "gsea", "other"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importCP_+3A_object">object</code></td>
<td>
<p>clusterProfiler object.</p>
</td></tr>
<tr><td><code id="importCP_+3A_type">type</code></td>
<td>
<p>object type from &quot;go&quot;, &quot;gsea&quot; and &quot;other&quot;. &quot;other&quot; includes ORA (over-representation analysis) of KEGG, DOSE,...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'data.frame'
</p>

<hr>
<h2 id='importPanther'>Import 'Panther' web result</h2><span id='topic+importPanther'></span>

<h3>Description</h3>

<p>Import 'Panther' web result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importPanther(panther_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importPanther_+3A_panther_file">panther_file</code></td>
<td>
<p>Panther result file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'data.frame'
</p>

<hr>
<h2 id='importShinygo'>Import 'shinyGO' web result</h2><span id='topic+importShinygo'></span>

<h3>Description</h3>

<p>Import 'shinyGO' web result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importShinygo(shinygo_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importShinygo_+3A_shinygo_file">shinygo_file</code></td>
<td>
<p>ShinyGO result file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'data.frame'
</p>

<hr>
<h2 id='plot_theme'>Themes for all plots</h2><span id='topic+plot_theme'></span>

<h3>Description</h3>

<p>Change ggplot text, font, legend and border
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_theme(
  main_text_size = 8,
  legend_text_size = 6,
  font_type = "sans",
  border_thick = 1.5,
  remove_grid = TRUE,
  remove_border = FALSE,
  remove_main_text = FALSE,
  remove_legend_text = FALSE,
  remove_legend = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_theme_+3A_main_text_size">main_text_size</code></td>
<td>
<p>Numeric, main text size</p>
</td></tr>
<tr><td><code id="plot_theme_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>Numeric, legend text size</p>
</td></tr>
<tr><td><code id="plot_theme_+3A_font_type">font_type</code></td>
<td>
<p>Character, specify the plot text font family, default is &quot;sans&quot;.</p>
</td></tr>
<tr><td><code id="plot_theme_+3A_border_thick">border_thick</code></td>
<td>
<p>Numeric, border thickness, default is 1.
If set 0, remove both border and ticks.</p>
</td></tr>
<tr><td><code id="plot_theme_+3A_remove_grid">remove_grid</code></td>
<td>
<p>Logical, remove background grid lines, default is FALSE.</p>
</td></tr>
<tr><td><code id="plot_theme_+3A_remove_border">remove_border</code></td>
<td>
<p>Logical, remove border line, default is FALSE.</p>
</td></tr>
<tr><td><code id="plot_theme_+3A_remove_main_text">remove_main_text</code></td>
<td>
<p>Logical, remove all axis text, default is FALSE.</p>
</td></tr>
<tr><td><code id="plot_theme_+3A_remove_legend_text">remove_legend_text</code></td>
<td>
<p>Logical, remove all legend text, default is FALSE.</p>
</td></tr>
<tr><td><code id="plot_theme_+3A_remove_legend">remove_legend</code></td>
<td>
<p>Logical, remove entire legend, default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot theme
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
ggplot(mtcars, aes(x = wt, y = mpg)) +
  geom_point() +
  plot_theme(font_type = "Times", border_thick = 2)
</code></pre>

<hr>
<h2 id='plotEnrich'>Plot for gene enrichment analysis of ORA method</h2><span id='topic+plotEnrich'></span>

<h3>Description</h3>

<p>Over-representation analysis (ORA) is a simple method for objectively deciding whether a set of variables of
known or suspected biological relevance, such as a gene set or pathway, is more prevalent in a set of variables
of interest than we expect by chance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotEnrich(
  enrich_df,
  fold_change = NULL,
  plot_type = c("bar", "wego", "dot", "bubble", "lollipop", "geneheat", "genechord",
    "network", "gomap", "goheat", "gotangram", "wordcloud", "upset"),
  term_metric = c("FoldEnrich", "GeneRatio", "Count", "RichFactor"),
  stats_metric = c("p.adjust", "pvalue", "qvalue"),
  sim_method = c("Resnik", "Lin", "Rel", "Jiang", "Wang", "JC"),
  up_color = "#E31A1C",
  down_color = "#1F78B4",
  show_gene = "all",
  xlim_left = 0,
  xlim_right = NA,
  wrap_length = NULL,
  org = NULL,
  ont = NULL,
  scale_ratio,
  layout,
  n_term,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotEnrich_+3A_enrich_df">enrich_df</code></td>
<td>
<p>Enrichment analysis 'data.frame' result.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_fold_change">fold_change</code></td>
<td>
<p>Fold change or logFC values with gene IDs as names. Used in &quot;heat&quot; and &quot;chord&quot;
plot.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_plot_type">plot_type</code></td>
<td>
<p>Choose from &quot;bar&quot;, &quot;wego&quot;,&quot;bubble&quot;,&quot;dot&quot;, &quot;lollipop&quot;,&quot;geneheat&quot;, &quot;genechord&quot;,
&quot;network&quot;,&quot;gomap&quot;,&quot;goheat&quot;,&quot;gotangram&quot;,&quot;wordcloud&quot;,&quot;upset&quot;.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_term_metric">term_metric</code></td>
<td>
<p>Pathway term metric from one of 'GeneRatio','Count','FoldEnrich' and
'RichFactor'.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_stats_metric">stats_metric</code></td>
<td>
<p>Statistic metric from one of &quot;pvalue&quot;, &quot;p.adjust&quot;, &quot;qvalue&quot;.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_sim_method">sim_method</code></td>
<td>
<p>Method of calculating the similarity between nodes, one of one of &quot;Resnik&quot;,
&quot;Lin&quot;, &quot;Rel&quot;, &quot;Jiang&quot; , &quot;Wang&quot; or &quot;JC&quot; (Jaccardâ€™s similarity index). Only &quot;JC&quot; supports KEGG data.
Used in &quot;map&quot;,&quot;goheat&quot;,&quot;gotangram&quot;,&quot;wordcloud&quot;.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_up_color">up_color</code></td>
<td>
<p>Color of higher statistical power (e.g. Pvalue 0.01) or higher logFC, default is &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_down_color">down_color</code></td>
<td>
<p>Color of lower statistical power (e.g. Pvalue 1) or lower logFC, default is
&quot;blue&quot;.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_show_gene">show_gene</code></td>
<td>
<p>Select genes to show. Default is &quot;all&quot;. Used in &quot;heat&quot; and &quot;chord&quot; plot.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_xlim_left">xlim_left</code></td>
<td>
<p>X-axis left limit, default is 0.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_xlim_right">xlim_right</code></td>
<td>
<p>X-axis right limit, default is NA.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_wrap_length">wrap_length</code></td>
<td>
<p>Numeric, wrap text if longer than this length. Default is NULL.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_org">org</code></td>
<td>
<p>Organism name from 'biocOrg_name'.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_ont">ont</code></td>
<td>
<p>One of &quot;BP&quot;, &quot;MF&quot;, and &quot;CC&quot;.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_scale_ratio">scale_ratio</code></td>
<td>
<p>Numeric, scale of node and line size.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_layout">layout</code></td>
<td>
<p>Grapgh layout in &quot;map&quot; plot, e,g, &quot;circle&quot;, &quot;dh&quot;, &quot;drl&quot;, &quot;fr&quot;,&quot;graphopt&quot;, &quot;grid&quot;,
&quot;lgl&quot;, &quot;kk&quot;, &quot;mds&quot;, &quot;nicely&quot; (default),&quot;randomly&quot;, &quot;star&quot;.</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_n_term">n_term</code></td>
<td>
<p>Number of terms (used in WEGO plot)</p>
</td></tr>
<tr><td><code id="plotEnrich_+3A_...">...</code></td>
<td>
<p>other arguments from 'plot_theme' function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## example data
## More examples please refer to https://www.genekitr.fun/plot-ora-1.html
library(ggplot2)
library(igraph)
library(ggraph)
data(geneList, package = "genekitr")
id &lt;- names(geneList)[abs(geneList) &gt; 1.5]
logfc &lt;- geneList[id]

gs &lt;- geneset::getGO(org = "human",ont = "bp")
ego &lt;- genORA(id, geneset = gs)
ego &lt;- ego[1:10, ]

## example plots
plotEnrich(ego, plot_type = "dot")

plotEnrich(ego, plot_type = "bubble", scale_ratio = 0.4)

plotEnrich(ego, plot_type = "bar")

plotEnrich(ego,
  plot_type = "lollipop",
  down_color = "#325CAC", up_color = "#E69056",
  wrap_length = 25, scale_ratio = 0.4
)

plotEnrich(ego, plot_type = "geneheat")



</code></pre>

<hr>
<h2 id='plotEnrichAdv'>Advanced Plot for  gene enrichment analysis of ORA method</h2><span id='topic+plotEnrichAdv'></span>

<h3>Description</h3>

<p>Over-representation analysis (ORA) is a simple method for objectively deciding whether a set of variables of
known or suspected biological relevance, such as a gene set or pathway, is more prevalent in a set of variables
of interest than we expect by chance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotEnrichAdv(
  up_enrich_df,
  down_enrich_df,
  plot_type = c("one", "two"),
  term_metric = c("FoldEnrich", "GeneRatio", "Count", "RichFactor"),
  stats_metric = c("p.adjust", "pvalue", "qvalue"),
  wrap_length = NULL,
  xlim_left = NULL,
  xlim_right = NULL,
  color,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotEnrichAdv_+3A_up_enrich_df">up_enrich_df</code></td>
<td>
<p>Enrichment analysis 'data.frame' for up-regulated genes.</p>
</td></tr>
<tr><td><code id="plotEnrichAdv_+3A_down_enrich_df">down_enrich_df</code></td>
<td>
<p>Enrichment analysis 'data.frame' for down-regulated genes.</p>
</td></tr>
<tr><td><code id="plotEnrichAdv_+3A_plot_type">plot_type</code></td>
<td>
<p>Choose from &quot;one&quot; and &quot;two&quot;. &quot;One&quot; represents both up and down pathways are plotted
together; &quot;two&quot; represents up and down are plotted seperately.</p>
</td></tr>
<tr><td><code id="plotEnrichAdv_+3A_term_metric">term_metric</code></td>
<td>
<p>Pathway term metric from one of 'GeneRatio','Count','FoldEnrich' and
'RichFactor'.</p>
</td></tr>
<tr><td><code id="plotEnrichAdv_+3A_stats_metric">stats_metric</code></td>
<td>
<p>Statistic metric from one of &quot;pvalue&quot;, &quot;p.adjust&quot;, &quot;qvalue&quot;.</p>
</td></tr>
<tr><td><code id="plotEnrichAdv_+3A_wrap_length">wrap_length</code></td>
<td>
<p>Numeric, wrap text if longer than this length. Default is NULL.</p>
</td></tr>
<tr><td><code id="plotEnrichAdv_+3A_xlim_left">xlim_left</code></td>
<td>
<p>X-axis left limit</p>
</td></tr>
<tr><td><code id="plotEnrichAdv_+3A_xlim_right">xlim_right</code></td>
<td>
<p>X-axis right limit</p>
</td></tr>
<tr><td><code id="plotEnrichAdv_+3A_color">color</code></td>
<td>
<p>Plot colors.</p>
</td></tr>
<tr><td><code id="plotEnrichAdv_+3A_...">...</code></td>
<td>
<p>other arguments from 'plot_theme' function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both up and down regulated pathways could be plotted in one figure as two-side barplot
</p>


<h3>Value</h3>

<p>A ggplot object
</p>

<hr>
<h2 id='plotGSEA'>Plot for gene enrichment analysis of GSEA method</h2><span id='topic+plotGSEA'></span>

<h3>Description</h3>

<p>Gene Set Enrichment Analysis (GSEA) is a computational method that determines whether an a priori defined set of
genes shows statistically significant, concordant differences between two biological states (e.g. phenotypes).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotGSEA(
  gsea_list,
  plot_type = c("volcano", "classic", "fgsea", "ridge", "bar"),
  stats_metric = c("p.adjust", "pvalue", "qvalue"),
  show_pathway = NULL,
  show_gene = NULL,
  colour = NULL,
  wrap_length = NULL,
  label_by = c("id", "description"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotGSEA_+3A_gsea_list">gsea_list</code></td>
<td>
<p>GSEA result from 'genGSEA' function</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_plot_type">plot_type</code></td>
<td>
<p>GSEA plot type, one of 'volcano', 'classic', 'fgsea', 'ridge' or 'bar'.</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_stats_metric">stats_metric</code></td>
<td>
<p>Statistic metric from one of &quot;pvalue&quot;, &quot;p.adjust&quot;, &quot;qvalue&quot;.</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_show_pathway">show_pathway</code></td>
<td>
<p>Select plotting pathways by number (will choose top N pathways)
or pathway name (choose from ID column).</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_show_gene">show_gene</code></td>
<td>
<p>Select genes to show. Default is &quot;all&quot;. Used in &quot;classic&quot; plot.</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_colour">colour</code></td>
<td>
<p>Colour vector. Deafault is NULL. Used in volcano, ridge and bar plot.</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_wrap_length">wrap_length</code></td>
<td>
<p>Numeric, wrap text if longer than this length. Default is NULL.</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_label_by">label_by</code></td>
<td>
<p>Select which column as the label. If user wants to modify labels in plot,
please modify the &quot;Description&quot; column and set the argument as &quot;description&quot;. Default is by 'id'.</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_...">...</code></td>
<td>
<p>other arguments transfer to 'plot_theme' function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(ggplot2)
## get GSEA result
data(geneList, package = "genekitr")
gs &lt;- geneset::getMsigdb(org = "human",category = "H")
gse &lt;- genGSEA(genelist = geneList, geneset = gs)

## volcano plot
# get top3 of up and down pathways
plotGSEA(gse, plot_type = "volcano", show_pathway = 3)
# choose pathway by character
pathways &lt;- c('HALLMARK_KRAS_SIGNALING_UP','HALLMARK_P53_PATHWAY','HALLMARK_GLYCOLYSIS')
plotGSEA(gse, plot_type = "volcano", show_pathway = pathways)

## classic pathway plot
genes &lt;- c('ENG','TP53','MET')
plotGSEA(gse, plot_type = "classic", show_pathway = pathways, show_gene = genes)

## fgsea table plot
plotGSEA(gse, plot_type = "fgsea", show_pathway = 3)

## ridgeplot
plotGSEA(gse,
  plot_type = "ridge",
  show_pathway = 10, stats_metric = "p.adjust"
)

## two-side barplot
plotGSEA(gse,
  plot_type = "bar", main_text_size = 8,
  colour = c("navyblue", "orange")
)


</code></pre>

<hr>
<h2 id='plotVenn'>Venn plot for groups of genes</h2><span id='topic+plotVenn'></span>

<h3>Description</h3>

<p>If gene group over 4, plot will be visulized using UpSet plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotVenn(
  venn_list,
  use_venn = TRUE,
  color = NULL,
  alpha_degree = 0.3,
  venn_percent = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotVenn_+3A_venn_list">venn_list</code></td>
<td>
<p>A list of gene id.</p>
</td></tr>
<tr><td><code id="plotVenn_+3A_use_venn">use_venn</code></td>
<td>
<p>Logical, use venn to plot, default is 'TRUE', the other
option is upsetplot for large list.</p>
</td></tr>
<tr><td><code id="plotVenn_+3A_color">color</code></td>
<td>
<p>Colors for gene lists, default is NULL.</p>
</td></tr>
<tr><td><code id="plotVenn_+3A_alpha_degree">alpha_degree</code></td>
<td>
<p>Alpha transparency of each circle's area, default is 0.3.</p>
</td></tr>
<tr><td><code id="plotVenn_+3A_venn_percent">venn_percent</code></td>
<td>
<p>Logical to show both number and percentage in venn plot.</p>
</td></tr>
<tr><td><code id="plotVenn_+3A_...">...</code></td>
<td>
<p>other arguments transfer to 'plot_theme' function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
set1 &lt;- paste0(rep("gene", 30), sample(1:1000, 30))
set2 &lt;- paste0(rep("gene", 40), sample(1:1000, 40))
set3 &lt;- paste0(rep("gene", 50), sample(1:1000, 50))
set4 &lt;- paste0(rep("gene", 60), sample(1:1000, 60))
set5 &lt;- paste0(rep("gene", 70), sample(1:1000, 70))
sm_gene_list &lt;- list(gset1 = set1, gset2 = set2, gset3 = set3)
la_gene_list &lt;- list(
  gset1 = set1, gset2 = set2, gset3 = set3,
  gset4 = set4, gset5 = set5
)
plotVenn(sm_gene_list,
  use_venn = TRUE,
  color = ggsci::pal_lancet()(3),
  alpha_degree = 0.5,
  main_text_size = 3,
  border_thick = 0,
  venn_percent = TRUE
)
plotVenn(la_gene_list,
  use_venn = FALSE,
  main_text_size = 15,
  legend_text_size = 8,
  legend_position = 'left'
)
</code></pre>

<hr>
<h2 id='plotVolcano'>Volcano plot for differential expression analysis</h2><span id='topic+plotVolcano'></span>

<h3>Description</h3>

<p>Volcano plot for differential expression analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotVolcano(
  deg_df,
  stat_metric = c("p.adjust", "pvalue"),
  stat_cutoff = 0.05,
  logFC_cutoff = 1,
  up_color = "#E31A1C",
  down_color = "#1F78B4",
  show_gene = NULL,
  dot_size = 1.75,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotVolcano_+3A_deg_df">deg_df</code></td>
<td>
<p>DEG dataframe with gene id, logFC and stat(e.g. pvalue/qvalue).</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_stat_metric">stat_metric</code></td>
<td>
<p>Statistic metric from &quot;pvalue&quot; or &quot;p.adjust&quot;.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_stat_cutoff">stat_cutoff</code></td>
<td>
<p>Statistic cutoff, default is 0.05.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_logfc_cutoff">logFC_cutoff</code></td>
<td>
<p>Log2 fold change cutoff, default is 1 which is actually 2 fold change.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_up_color">up_color</code></td>
<td>
<p>Color of up-regulated genes, default is &quot;dark red&quot;.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_down_color">down_color</code></td>
<td>
<p>Color of down-regulated genes, default is &quot;dark blue&quot;.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_show_gene">show_gene</code></td>
<td>
<p>Select genes to show, default is no genes to show.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_dot_size">dot_size</code></td>
<td>
<p>Volcano dot size, default is 1.75.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_...">...</code></td>
<td>
<p>other arguments from 'plot_theme' function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(ggplot2)
data(deg, package = "genekitr")
plotVolcano(deg, "p.adjust", remove_legend = TRUE, dot_size = 3)

# show some genes
plotVolcano(deg, "p.adjust",
  remove_legend = TRUE,
  show_gene = c("CD36", "DUSP6", "IER3","CDH7")
)


</code></pre>

<hr>
<h2 id='simGO'>Simplify GO enrichment result</h2><span id='topic+simGO'></span>

<h3>Description</h3>

<p>The Gene Ontology (GO) is a major bioinformatics initiative to unify the representation of gene and gene product
attributes across all species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simGO(
  enrich_df,
  sim_method = c("Resnik", "Lin", "Rel", "Jiang", "Wang"),
  org = NULL,
  ont = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simGO_+3A_enrich_df">enrich_df</code></td>
<td>
<p>GO enrichment analysis of 'genORA()' result.</p>
</td></tr>
<tr><td><code id="simGO_+3A_sim_method">sim_method</code></td>
<td>
<p>Method of calculating the similarity between nodes, one of one of &quot;Resnik&quot;,
&quot;Lin&quot;, &quot;Rel&quot;, &quot;Jiang&quot; , &quot;Wang&quot; methods.</p>
</td></tr>
<tr><td><code id="simGO_+3A_org">org</code></td>
<td>
<p>Organism name from 'biocOrg_name'.</p>
</td></tr>
<tr><td><code id="simGO_+3A_ont">ont</code></td>
<td>
<p>One of &quot;bp&quot;, &quot;mf&quot;, and &quot;cc&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'data.frame' contains simplified GO terms.
</p>

<hr>
<h2 id='transId'>Transform id among symbol, entrezid,  ensembl and uniprot.</h2><span id='topic+transId'></span>

<h3>Description</h3>

<p>Transform id among symbol, entrezid,  ensembl and uniprot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transId(
  id,
  transTo,
  org = "hs",
  unique = FALSE,
  keepNA = FALSE,
  hgVersion = c("v38", "v19")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transId_+3A_id">id</code></td>
<td>
<p>Gene ids or protein ids.</p>
</td></tr>
<tr><td><code id="transId_+3A_transto">transTo</code></td>
<td>
<p>Transform to what type. User could select one or more from
&quot;symbol&quot;, &quot;entrez&quot;, &quot;ensembl&quot; or &quot;uniprot.&quot;</p>
</td></tr>
<tr><td><code id="transId_+3A_org">org</code></td>
<td>
<p>Latin organism shortname from 'ensOrg_name'. Default is human.</p>
</td></tr>
<tr><td><code id="transId_+3A_unique">unique</code></td>
<td>
<p>Logical, if one-to-many mapping occurs, only keep one record with fewest NA. Default is FALSE.</p>
</td></tr>
<tr><td><code id="transId_+3A_keepna">keepNA</code></td>
<td>
<p>If some id has no match at all, keep it or not. Default is FALSE.</p>
</td></tr>
<tr><td><code id="transId_+3A_hgversion">hgVersion</code></td>
<td>
<p>Select human genome build version from &quot;v38&quot; (default) and &quot;v19&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame, first column is input id and others are converted id.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# example1:
transId(
  id = c("Cyp2c23", "Fhit", "Gal3st2b", "Trp53", "Tp53"),
  transTo = "ensembl", org = "mouse", keepNA = FALSE
)

## example2: input id with one-to-many mapping and fake one
transId(
  id = c("MMD2", "HBD", "RNR1", "TEC", "BCC7", "FAKEID", "TP53"),
  transTo = c("entrez", "ensembl"), keepNA = TRUE
)

# example3: auto-recognize ensembl version number
transId("ENSG00000141510.11", "symbol")

# example4: search genes with case-insensitive
transId(c('nc886','ezh2','TP53'),transTo = "ensembl",org = 'hs',unique = TRUE)


</code></pre>

<hr>
<h2 id='transProbe'>Transform probe id to symbol, entrezid,  ensembl or uniprot.</h2><span id='topic+transProbe'></span>

<h3>Description</h3>

<p>Transform probe id to symbol, entrezid,  ensembl or uniprot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transProbe(id, transTo, org = "human", platform = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transProbe_+3A_id">id</code></td>
<td>
<p>probe ids.</p>
</td></tr>
<tr><td><code id="transProbe_+3A_transto">transTo</code></td>
<td>
<p>Transform to what type. User could select one or more from
&quot;symbol&quot;, &quot;entrez&quot;, &quot;ensembl&quot; or &quot;uniprot.&quot;</p>
</td></tr>
<tr><td><code id="transProbe_+3A_org">org</code></td>
<td>
<p>'human'.</p>
</td></tr>
<tr><td><code id="transProbe_+3A_platform">platform</code></td>
<td>
<p>Probe platform. If NULL, program will detect automatically.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame, first column is probe id and others are converted id.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(deg, package = "genekitr")
id &lt;- deg$probe_id
transProbe(id, transTo = "symbol")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
