<!DOCTYPE html><html><head><title>Help for package genemodel</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {genemodel}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AT5G62640'><p>Gene model for AT5G62640</p></a></li>
<li><a href='#genemodel.plot'><p>genemodel.plot</p></a></li>
<li><a href='#mutation.plot'><p>mutation.plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Gene Model Plotting in R</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-02-20</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Using simple input, this package creates plots of gene models. Users can create plots of alternatively spliced gene variants and the positions of mutations and other gene features.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stringr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/greymonroe/genemodel">https://github.com/greymonroe/genemodel</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-02-20 17:05:28 UTC; greymonroe</td>
</tr>
<tr>
<td>Author:</td>
<td>J Grey Monroe [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>J Grey Monroe &lt;greymonroe@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-02-20 19:09:06</td>
</tr>
</table>
<hr>
<h2 id='AT5G62640'>Gene model for AT5G62640</h2><span id='topic+AT5G62640'></span>

<h3>Description</h3>

<p>Gene model for AT5G62640
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AT5G62640
</code></pre>


<h3>Format</h3>

<p>A data frame with 32 rows and 2 variables:
</p>

<dl>
<dt>type</dt><dd><p>feature of gene ie intron exon or UTR</p>
</dd>
<dt>coordinates</dt><dd><p>start and stop of gene feature separated by a &quot;-&quot;</p>
</dd>
</dl>
<p>...
</p>


<h3>Source</h3>

<p><a href="https://www.arabidopsis.org/servlets/TairObject?type=gene&amp;id=1000654517">https://www.arabidopsis.org/servlets/TairObject?type=gene&amp;id=1000654517</a>
</p>

<hr>
<h2 id='genemodel.plot'>genemodel.plot</h2><span id='topic+genemodel.plot'></span>

<h3>Description</h3>

<p>This function plots a gene model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genemodel.plot(model, start, bpstop, orientation, xaxis = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genemodel.plot_+3A_model">model</code></td>
<td>
<p>data.frame containing model information. Required columns are &quot;type&quot;, &quot;coordinates&quot;</p>
</td></tr>
<tr><td><code id="genemodel.plot_+3A_start">start</code></td>
<td>
<p>start position</p>
</td></tr>
<tr><td><code id="genemodel.plot_+3A_bpstop">bpstop</code></td>
<td>
<p>stop position</p>
</td></tr>
<tr><td><code id="genemodel.plot_+3A_orientation">orientation</code></td>
<td>
<p>either &quot;foward&quot; or &quot;reverse&quot; indicates the direction of transcription</p>
</td></tr>
<tr><td><code id="genemodel.plot_+3A_xaxis">xaxis</code></td>
<td>
<p>default is TRUE and adds axis above gene model showing position</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(AT5G62640)
genemodel.plot(AT5G62640, 25149433, 25152541, "reverse", xaxis=TRUE)
</code></pre>

<hr>
<h2 id='mutation.plot'>mutation.plot</h2><span id='topic+mutation.plot'></span>

<h3>Description</h3>

<p>This function plots mutations along genemodels created with genemodel.plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation.plot(start, stop, text = "", drop = -0.15, col = "red",
  haplotypes = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutation.plot_+3A_start">start</code></td>
<td>
<p>start position</p>
</td></tr>
<tr><td><code id="mutation.plot_+3A_stop">stop</code></td>
<td>
<p>stop position</p>
</td></tr>
<tr><td><code id="mutation.plot_+3A_text">text</code></td>
<td>
<p>any text that you want displayed on the label</p>
</td></tr>
<tr><td><code id="mutation.plot_+3A_drop">drop</code></td>
<td>
<p>how far below the gene model you want the mutation label to be placed</p>
</td></tr>
<tr><td><code id="mutation.plot_+3A_col">col</code></td>
<td>
<p>the color of the text and mutation line to be</p>
</td></tr>
<tr><td><code id="mutation.plot_+3A_haplotypes">haplotypes</code></td>
<td>
<p>the color of dots that you want to place along the mutation line to indicate some factor such as haplotype that the mutation belongs to</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(AT5G62640)
genemodel.plot(AT5G62640, 25149433, 25152541, "reverse")
mutation.plot(25149593, 25149593, text="P-&gt;S", col="red", haplotype="blue")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
