<!DOCTYPE html><html><head><title>Help for package genepop</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {genepop}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#genepop-package'>
<p>Population genetic analyses using the Genepop software</p></a></li>
<li><a href='#basic_info'><p>Allele and genotype frequencies</p></a></li>
<li><a href='#clean_workdir'><p>Removing files created by Genepop</p></a></li>
<li><a href='#Contingency-test'><p>Exact test on a single contingency table</p></a></li>
<li><a href='#conversion'><p>File conversions</p></a></li>
<li><a href='#Differentiation'><p>Tests of genic and genotypic differentiation</p></a></li>
<li><a href='#Fst'><p>Fst (or rho_ST) estimation</p></a></li>
<li><a href='#genedivFis'><p>Gene diversities and Fis (or rho_IS)</p></a></li>
<li><a href='#genepop-internals'><p>Internal genepop Functions</p></a></li>
<li><a href='#genepopExample'><p>Copy an example file from the Genepop package distribution</p></a></li>
<li><a href='#GUI'><p>Call an experimental GUI for Genepop</p></a></li>
<li><a href='#Hardy-Weinberg'><p>Tests of Hardy-Weinberg genotypic proportions</p></a></li>
<li><a href='#IBD'><p>Isolation by distance</p></a></li>
<li><a href='#Linkage'><p>Tables and exact test for genotypic linkage disequilibrium</p></a></li>
<li><a href='#manipulation'><p>Various data manipulation utilities</p></a></li>
<li><a href='#Nm_private'><p>Private allele method</p></a></li>
<li><a href='#nulls'><p>Estimation of allele frequencies under genotyping failure.</p></a></li>
<li><a href='#set_restriction'><p>Programming utilities</p></a></li>
<li><a href='#setMantelSeed'><p>Set random generator seed for Mantel test</p></a></li>
<li><a href='#setRandomSeed'><p>Set random generator seed (except for Mantel test)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Population Genetic Data Analysis Using Genepop</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-01-19</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>François Rousset &lt;francois.rousset@umontpellier.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Makes the Genepop software available in R. This software implements a mixture of traditional population genetic methods and some more focused developments: it computes exact tests for Hardy-Weinberg equilibrium, for population differentiation and for genotypic disequilibrium among pairs of loci; it computes estimates of F-statistics, null allele frequencies, allele size-based statistics for microsatellites, etc.; and it performs analyses of isolation by distance from pairwise comparisons of individuals or population samples. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.cecill.info/licences/Licence_CeCILL_V2-en.txt">CeCILL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.10), stringr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppProgress</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, shiny</td>
</tr>
<tr>
<td>Config/build/clean-inst-doc:</td>
<td>false</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.r-project.org">https://www.r-project.org</a>,
<a href="https://kimura.univ-montp2.fr/~rousset/Genepop.htm">https://kimura.univ-montp2.fr/~rousset/Genepop.htm</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-01-19 22:45:00 UTC; francois.rousset</td>
</tr>
<tr>
<td>Author:</td>
<td>François Rousset <a href="https://orcid.org/0000-0003-4670-0371"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  Jimmy Lopez [ctb],
  Alexandre Genin [ctb],
  Khalid Belkhir [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-01-19 23:30:17 UTC</td>
</tr>
</table>
<hr>
<h2 id='genepop-package'>
Population genetic analyses using the Genepop software
</h2><span id='topic+genepop-package'></span><span id='topic+genepop'></span>

<h3>Description</h3>

<p>A distribution of the Genepop software as an R package. The included C++ sources are suitable for compilation as a stand-alone executable. A shiny interface is also included.    Genepop performs three main tasks: it computes <b>exact tests</b> for Hardy-Weinberg equilibrium (<code><a href="#topic+test_HW">test_HW</a></code>), for population differentiation  (<code><a href="#topic+test_diff">test_diff</a></code>) and for genotypic disequilibrium among pairs of loci  (<code><a href="#topic+test_LD">test_LD</a></code>); it computes <b>estimates</b> of F-statistics  (<code><a href="#topic+Fst">Fst</a></code>), null allele frequencies (<code><a href="#topic+nulls">nulls</a></code>), allele size-based statistics for microsatellites, etc., and of number of immigrants by Barton &amp; Slatkin's (1986) private allele method  (<code><a href="#topic+Nm_private">Nm_private</a></code>); It performs analyses of <b>isolation by distance</b> from pairwise comparisons of individuals or groups (<code><a href="#topic+ibd">ibd</a></code>), including confidence intervals for &quot;neighborhood size&quot;. It also provides various data <code><a href="#topic+conversion">conversion</a></code> and <code><a href="#topic+manipulation">manipulation</a></code> utilities.  
</p>


<h3>Author(s)</h3>

<p>R package originally developed by Jimmy Lopez and Khalid Belkhir from the C++ sources of the Genepop executable version 4.6 (2016; Rousset 2008). 
</p>


<h3>References</h3>

<p><b>Main reference for current maintained version of Genepop</b>:
</p>
<p>Rousset, F. (2008). Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. Mol. Ecol. Res. 8: 103-106.
</p>
<p>Original Genepop publication: 
</p>
<p>Raymond, M. &amp; Rousset, F., 1995b. GENEPOP Version 1.2: population genetics software for exact tests and ecumenicism. J. Hered. 86: 248-249.
</p>
<p>Methods implemented in Genepop:
</p>
<p>Barton, N. H. &amp; Slatkin, M., 1986. A quasi-equilibrium theory of the distribution of rare alleles in a subdivided population. Heredity 56: 409-415.
</p>
<p>Brookfield, J. F. Y., 1996. A simple new method for estimating null allele frequency from heterozygote deficiency. Mol. Ecol. 5: 453-455.
</p>
<p>Goudet, J., Raymond, M., de Meeus, T. &amp; Rousset, F., 1996. Testing differentiation in diploid populations. Genetics 144: 1931-1938.
</p>
<p>Guo, S. W. &amp; Thompson, E. A., 1992. Performing the exact test of Hardy-Weinberg proportion for multiple alleles. Biometrics 48: 361-372.
</p>
<p>Kalinowski, S. T. &amp; Taper, M. L., 2006. Maximum likelihood estimation of the frequency of null alleles at microsatellite loci. Conserv. Genetics 7:991-995.
</p>
<p>Louis, E. J. &amp; Dempster, E. R., 1987. An exact test for Hardy-Weinberg and multiple alleles. Biometrics 43: 805-811.
</p>
<p>Mantel, N., 1967. The detection of disease clustering and a generalized regression approach. Cancer Research 27: 209-220.
</p>
<p>Michalakis, Y. &amp; Excoffier, L., 1996. A generic estimation of population subdivision using distances between alleles with special interest to microsatellite loci. Genetics 142: 1061-1064.
</p>
<p>Raymond, M. &amp; Rousset, F., 1995a. An exact test for population differentiation.Evolution 49: 1283-1286.
</p>
<p>Robertson, A. &amp; Hill, W. G., 1984. Deviations from Hardy-Weinberg proportions: sampling variances and use in estimation of inbreeding coefficients. Genetics 107: 703-718.
</p>
<p>Rousset, F., 1996. Equilibrium values of measures of population subdivision for stepwise mutation processes. Genetics 142: 1357-1362.
</p>
<p>Rousset, F., 1997. Genetic differentiation and estimation of gene flow from F-statistics under isolation by distance. Genetics 145: 1219-1228.
</p>
<p>Rousset, F., 2000. Genetic differentiation between individuals. J. Evol. Biol. 13:58-62.
</p>
<p>Rousset, F. &amp; Raymond, M., 1995. Testing heterozygote excess and deficiency. Genetics 140: 1413-1419.
</p>
<p>Watts, P. C., Rousset, F., Saccheri, I. J., Leblois, R., Kemp, S. J. &amp; Thompson, D. J., 2007. Compatible genetic and ecological estimates of dispersal rates in insect (Coenagrion mercuriale: Odonata: Zygoptera) populations: analysis of 'neighbourhood size' using a more precise estimator. Mol. Ecol. 16: 737-751.
</p>
<p>Weir, B. S., 1996. Genetic Data Analysis II. Sinauer, Sunderland, Mass.
</p>
<p>Weir, B. S. &amp; Cockerham, C. C., 1984. Estimating F-statistics for the analysis of population structure. Evolution 38: 1358-1370.
</p>

<hr>
<h2 id='basic_info'>Allele and genotype frequencies</h2><span id='topic+basic_info'></span>

<h3>Description</h3>

<p>Allele and genotype frequencies per locus and per sample. See <a href="../doc/all-menu-options.html#sub-option-1-allele-and-genotype-frequencies">this section</a> of the Genepop executable documentation for more information on the statistical methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basic_info(inputFile, outputFile = "", verbose = interactive())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="basic_info_+3A_inputfile">inputFile</code></td>
<td>
<p>The path of the input file, in Genepop format</p>
</td></tr>
<tr><td><code id="basic_info_+3A_outputfile">outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td></tr>
<tr><td><code id="basic_info_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>locinfile &lt;- genepopExample('sample.txt')
basic_info(locinfile,'sample.txt.INF')
if ( ! interactive()) clean_workdir(otherfiles='sample.txt')
</code></pre>

<hr>
<h2 id='clean_workdir'>Removing files created by Genepop</h2><span id='topic+clean_workdir'></span>

<h3>Description</h3>

<p>This removes &ldquo;temporary files&rdquo; created by Genepop, but also output files, so it should be used only when one no longer needs the latter files. This function asumes that the input file name contains only alphanumeric, dot, or underscore characters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean_workdir(
  otherfiles = NULL,
  path = ".",
  suffixes = c("GRA", "ISO", "MIG", "PRI", "DAT", "DG", "DIV", "D", "DIS", "FST", "NUL",
    "RHO", "2G2", "G", "GE", "GE2", "INF", "MSD", "TAB", "ST2"),
  in. = TRUE,
  cmdline = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clean_workdir_+3A_otherfiles">otherfiles</code></td>
<td>
<p>Character vector(s): one or more names of files to be removed and not matched by the other arguments (such as the input file, or some output files not identified by their suffix, as shown in the Example).</p>
</td></tr>
<tr><td><code id="clean_workdir_+3A_path">path</code></td>
<td>
<p>character vector: path from where files should be removed.</p>
</td></tr>
<tr><td><code id="clean_workdir_+3A_suffixes">suffixes</code></td>
<td>
<p>Character vector(s): suffixes of files to be removed (useful for output files with readily identifiable suffixes).</p>
</td></tr>
<tr><td><code id="clean_workdir_+3A_in.">in.</code></td>
<td>
<p>boolean: whether to remove the <code>fichier.in</code> file created by Genepop.</p>
</td></tr>
<tr><td><code id="clean_workdir_+3A_cmdline">cmdline</code></td>
<td>
<p>boolean: whether to remove the <code>cmdline.txt</code> file created by Genepop.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Removing files possibly written by other examples in the documentation:
clean_workdir(otherfiles=c("sample.txt", "Dsample.txt", "w2.txt", 
"PEL1600withCoord.txt", "Rhesus.txt", "structest.txt"))
</code></pre>

<hr>
<h2 id='Contingency-test'>Exact test on a single contingency table</h2><span id='topic+Contingency-test'></span><span id='topic+struc'></span>

<h3>Description</h3>

<p>Performs an exact conditional contingency-table test. There are many other ways of doing this in R but this function replicates the functionality of earlier genepop code analysing a contingency table provided in a file with ad hoc format. See <a href="../doc/all-menu-options.html#analyzing-a-single-contingency-table">this section</a> of the Genepop executable documentation for more information on the statistical methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>struc(
  inputFile,
  settingsFile = "",
  dememorization = 10000,
  batches = 100,
  iterations = 5000,
  verbose = interactive()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Contingency-test_+3A_inputfile">inputFile</code></td>
<td>
<p>character: The path of the input file. This file should be in an ad hoc format</p>
</td></tr>
<tr><td><code id="Contingency-test_+3A_settingsfile">settingsFile</code></td>
<td>
<p>character: The path of the settings file</p>
</td></tr>
<tr><td><code id="Contingency-test_+3A_dememorization">dememorization</code></td>
<td>
<p>integer: length of dememorization step of Markov chain algorithm</p>
</td></tr>
<tr><td><code id="Contingency-test_+3A_batches">batches</code></td>
<td>
<p>integer: Number of batches</p>
</td></tr>
<tr><td><code id="Contingency-test_+3A_iterations">iterations</code></td>
<td>
<p>integer: Iterations per batch</p>
</td></tr>
<tr><td><code id="Contingency-test_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>locinfile &lt;- genepopExample('structest.txt')
struc(locinfile)
if ( ! interactive()) clean_workdir(otherfiles='structest.txt')
</code></pre>

<hr>
<h2 id='conversion'>File conversions</h2><span id='topic+conversion'></span>

<h3>Description</h3>

<p>Converts input files from genepop format to some other formats (some maybe only of historical interest): Fstat, two Biosys formats. and linkdos. See <a href="../doc/all-menu-options.html#option-7-file-conversions">this section</a> of the Genepop executable documentation for more information on the statistical methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>conversion(inputFile, format, outputFile = "", verbose = interactive())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="conversion_+3A_inputfile">inputFile</code></td>
<td>
<p>The path of the input file, in Genepop format</p>
</td></tr>
<tr><td><code id="conversion_+3A_format">format</code></td>
<td>
<p>Character string: must be one of <code>'Fstat'</code>, <code>'BiosysL'</code>, <code>'BiosysN'</code>, or <code>'Linkdos'</code></p>
</td></tr>
<tr><td><code id="conversion_+3A_outputfile">outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td></tr>
<tr><td><code id="conversion_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>locinfile &lt;- genepopExample('sample.txt')
conversion(locinfile, format='Fstat', 'sample.txt.DAT')
if ( ! interactive()) clean_workdir(otherfiles='sample.txt')
</code></pre>

<hr>
<h2 id='Differentiation'>Tests of genic and genotypic differentiation</h2><span id='topic+Differentiation'></span><span id='topic+test_diff'></span>

<h3>Description</h3>

<p>Exact conditional contingency-table tests for genic or genotypic differentiation. A single test for all populations, or distinct tests for all pairs of populations, may be computed. See <a href="../doc/all-menu-options.html#option-3-population-differentiation">this section</a> of the Genepop executable documentation for more information on the statistical methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_diff(
  inputFile,
  genic = TRUE,
  pairs = FALSE,
  outputFile = "",
  settingsFile = "",
  dememorization = 10000,
  batches = 100,
  iterations = 5000,
  verbose = interactive()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Differentiation_+3A_inputfile">inputFile</code></td>
<td>
<p>The path of the input file, in Genepop format</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_genic">genic</code></td>
<td>
<p>logical: whether to perform genic or genotypic tests</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_pairs">pairs</code></td>
<td>
<p>logical: whether to test differentiation between all pairs of ppulation, or to perform a single global test</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_outputfile">outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_settingsfile">settingsFile</code></td>
<td>
<p>character: The path of the settings file</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_dememorization">dememorization</code></td>
<td>
<p>integer: length of dememorization step of Markov chain algorithm</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_batches">batches</code></td>
<td>
<p>integer: Number of batches</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_iterations">iterations</code></td>
<td>
<p>integer: Iterations per batch</p>
</td></tr>
<tr><td><code id="Differentiation_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>locinfile &lt;- genepopExample('sample.txt')
test_diff(locinfile,outputFile='sample.txt.GE')
if ( ! interactive()) clean_workdir(otherfiles='sample.txt')
</code></pre>

<hr>
<h2 id='Fst'>Fst (or rho_ST) estimation</h2><span id='topic+Fst'></span>

<h3>Description</h3>

<p>Evaluates Fst or related measures based on allele sizes, for all populations of for all pairs of populations. See <a href="../doc/all-menu-options.html#sub-options-14-f-statistics-and-rho-statistics">this section</a> of the Genepop executable documentation for more information on the statistical methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Fst(
  inputFile,
  sizes = FALSE,
  pairs = FALSE,
  outputFile = "",
  dataType = "Diploid",
  verbose = interactive()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Fst_+3A_inputfile">inputFile</code></td>
<td>
<p>The path of the input file, in Genepop format</p>
</td></tr>
<tr><td><code id="Fst_+3A_sizes">sizes</code></td>
<td>
<p>logical: whether to estimate allele-size based statistics, or identity-based Fst</p>
</td></tr>
<tr><td><code id="Fst_+3A_pairs">pairs</code></td>
<td>
<p>whether to estimate differentiation between all pairs of populations, or to compute a global estimate for all populations</p>
</td></tr>
<tr><td><code id="Fst_+3A_outputfile">outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td></tr>
<tr><td><code id="Fst_+3A_datatype">dataType</code></td>
<td>
<p>character: The haploid and diploid data</p>
</td></tr>
<tr><td><code id="Fst_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>locinfile &lt;- genepopExample('sample.txt')
Fst(locinfile, outputFile= 'sample.txt.DIV')
if ( ! interactive()) clean_workdir(otherfiles='sample.txt')
</code></pre>

<hr>
<h2 id='genedivFis'>Gene diversities and Fis (or rho_IS)</h2><span id='topic+genedivFis'></span>

<h3>Description</h3>

<p>Evaluates Fis and gene diversities, or related measures based on allele sizes. See <a href="../doc/all-menu-options.html#sub-option-2-identity-based-gene-diversities-and-f_mathrmis">this section</a> of the Genepop executable documentation for more information on the identity-based statistical methods, and <a href="../doc/all-menu-options.html#sub-option-3-allele-size-based-gene-diversities-and-rho_mathrmis">this one</a> for allele-size based ones.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genedivFis(
  inputFile,
  sizes = FALSE,
  outputFile = "",
  dataType = "Diploid",
  verbose = interactive()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genedivFis_+3A_inputfile">inputFile</code></td>
<td>
<p>The path of the input file, in Genepop format</p>
</td></tr>
<tr><td><code id="genedivFis_+3A_sizes">sizes</code></td>
<td>
<p>logical: whether to compute statistics based on allele size, or not.</p>
</td></tr>
<tr><td><code id="genedivFis_+3A_outputfile">outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td></tr>
<tr><td><code id="genedivFis_+3A_datatype">dataType</code></td>
<td>
<p>character: The haploid and diploid data</p>
</td></tr>
<tr><td><code id="genedivFis_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>locinfile &lt;- genepopExample('sample.txt')
genedivFis(locinfile,outputFile = 'sample.txt.DIV')
if ( ! interactive()) clean_workdir(otherfiles='sample.txt')
</code></pre>

<hr>
<h2 id='genepop-internals'>Internal genepop Functions</h2><span id='topic+genepop-internals'></span><span id='topic+summary.GP_outfile'></span>

<h3>Description</h3>

<p>These are not to be called by the user, or are waiting for documentation to be written.
</p>

<hr>
<h2 id='genepopExample'>Copy an example file from the Genepop package distribution</h2><span id='topic+genepopExample'></span>

<h3>Description</h3>

<p>This function is used to copy an example file to the user's directory. It should not be used when analysing one's own data!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genepopExample(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genepopExample_+3A_filename">filename</code></td>
<td>
<p>The name of an example file from the Genepop distribution.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the <code>filename</code>
</p>

<hr>
<h2 id='GUI'>Call an experimental GUI for Genepop</h2><span id='topic+GUI'></span>

<h3>Description</h3>

<p>Call an experimental GUI for Genepop
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GUI()
</code></pre>


<h3>Value</h3>

<p>The return value of a 'shiny::runApp()' call.
</p>

<hr>
<h2 id='Hardy-Weinberg'>Tests of Hardy-Weinberg genotypic proportions</h2><span id='topic+Hardy-Weinberg'></span><span id='topic+test_HW'></span><span id='topic+HWtable_analysis'></span>

<h3>Description</h3>

<p>Compute variants of the exact conditional test for Hardy-Weinberg genotypic proportions.  The tests differ by their test statistics. <code>HWtable_analysis</code> handles a single table of genotype counts, and <code>test_HW</code> requires a standard genepop input file. See <a href="../doc/all-menu-options.html#option-1-hardy-weinberg-hw-exact-tests">this section</a> of the Genepop-executable documentation for more information on the statistical methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_HW(
  inputFile,
  which = "Proba",
  outputFile = "",
  settingsFile = "",
  enumeration = FALSE,
  dememorization = 10000,
  batches = 20,
  iterations = 5000,
  verbose = interactive()
)

HWtable_analysis(
  inputFile,
  which = "Proba",
  settingsFile = "",
  enumeration = FALSE,
  dememorization = 10000,
  batches = 20,
  iterations = 5000,
  verbose = interactive()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Hardy-Weinberg_+3A_inputfile">inputFile</code></td>
<td>
<p>character: The path of the input file. For <code>test_HW</code>, this file should be in Genepop format. For <code>HWtable_analysis</code>, it should be in ad hoc format illustrated by sample file <code>Rhesus.txt</code> used in the Examples section, and further detailed in <a href="../doc/all-menu-options.html#analyzing-a-single-genotypic-matrix">this section</a> of the Genepop-executable documentation.</p>
</td></tr>
<tr><td><code id="Hardy-Weinberg_+3A_which">which</code></td>
<td>
<p>character: <code>'Proba'</code>, <code>'excess'</code>, and <code>'deficit'</code> to perform the probability test, score test for excess, and score tests for deficit, respectively, in each population and for each locus. <code>test_HW</code> additionally handles <code>'global excess'</code> and  <code>'global deficit'</code> for global tests for all loci and/or all populations, and <code>HWtable_analysis</code> additionally handles <code>'Fis'</code> to report basic information (allele frequencies and Fis).</p>
</td></tr>
<tr><td><code id="Hardy-Weinberg_+3A_outputfile">outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td></tr>
<tr><td><code id="Hardy-Weinberg_+3A_settingsfile">settingsFile</code></td>
<td>
<p>character: The path of the settings file</p>
</td></tr>
<tr><td><code id="Hardy-Weinberg_+3A_enumeration">enumeration</code></td>
<td>
<p>logical: whether to compute the complete enumeration test for samples with less than 5 alleles</p>
</td></tr>
<tr><td><code id="Hardy-Weinberg_+3A_dememorization">dememorization</code></td>
<td>
<p>integer: length of dememorization step of Markov chain algorithm</p>
</td></tr>
<tr><td><code id="Hardy-Weinberg_+3A_batches">batches</code></td>
<td>
<p>integer: Number of batches</p>
</td></tr>
<tr><td><code id="Hardy-Weinberg_+3A_iterations">iterations</code></td>
<td>
<p>integer: Iterations per batch</p>
</td></tr>
<tr><td><code id="Hardy-Weinberg_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>locinfile &lt;- genepopExample('sample.txt')
test_HW(locinfile, which='deficit', 'sample.txt.D')
if ( ! interactive()) clean_workdir(otherfiles='sample.txt')
# Example in Guo &amp; Thompson 1992 Table 5
locinfile &lt;- genepopExample('Rhesus.txt')
outfile &lt;- HWtable_analysis(locinfile,which='Proba',batches = 1000,iterations = 1000)
readLines(outfile)[21]
#clean_workdir(otherfiles='Rhesus.txt')
</code></pre>

<hr>
<h2 id='IBD'>Isolation by distance</h2><span id='topic+IBD'></span><span id='topic+ibd'></span>

<h3>Description</h3>

<p>Estimates isolation by distance by regression of genetic distance to geographical distance. See <a href="../doc/all-menu-options.html#sub-option-5-isolation-by-distance-between-individuals">this section</a> of the Genepop executable documentation for more information on individual-based analyses and <a href="../doc/all-menu-options.html#sub-option-6-isolation-by-distance-between-groups">this one</a> for group-based analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ibd(
  inputFile,
  outputFile = "",
  settingsFile = "",
  dataType = "Diploid",
  statistic = "F/(1-F)",
  geographicScale = "2D",
  CIcoverage = 0.95,
  testPoint = 0,
  minimalDistance = 1e-04,
  maximalDistance = 1e+09,
  mantelPermutations = 1000,
  mantelRankTest = FALSE,
  bootstrapMethod = "ABC",
  bootstrapNsim = 999,
  verbose = interactive()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IBD_+3A_inputfile">inputFile</code></td>
<td>
<p>The path of the input file, in Genepop format</p>
</td></tr>
<tr><td><code id="IBD_+3A_outputfile">outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td></tr>
<tr><td><code id="IBD_+3A_settingsfile">settingsFile</code></td>
<td>
<p>character: The path of the settings file</p>
</td></tr>
<tr><td><code id="IBD_+3A_datatype">dataType</code></td>
<td>
<p>character: <code>'haploid'</code> or <code>'diploid'</code></p>
</td></tr>
<tr><td><code id="IBD_+3A_statistic">statistic</code></td>
<td>
<p>character: The pairwise genetic distance, either <code>'a'</code> or <code>'e'</code> for diploid individual data, <code>'a-like'</code> for haploid individual data, and <code>'F/(1-F)'</code> or <code>'SingleGeneDiv'</code> for group data (haploid or diploid)</p>
</td></tr>
<tr><td><code id="IBD_+3A_geographicscale">geographicScale</code></td>
<td>
<p>character: gives either the scale transformation <code>'Log'</code> or <code>'Linear'</code>  for geographic distances, or the shape of the habitat <code>'2D'</code> or <code>'1D'</code></p>
</td></tr>
<tr><td><code id="IBD_+3A_cicoverage">CIcoverage</code></td>
<td>
<p>numeric: The coverage probability of confidence intervals</p>
</td></tr>
<tr><td><code id="IBD_+3A_testpoint">testPoint</code></td>
<td>
<p>numeric: Given value of the slope to be tested</p>
</td></tr>
<tr><td><code id="IBD_+3A_minimaldistance">minimalDistance</code></td>
<td>
<p>numeric: The minimal geographic distance</p>
</td></tr>
<tr><td><code id="IBD_+3A_maximaldistance">maximalDistance</code></td>
<td>
<p>numeric: The maximal geographic distance</p>
</td></tr>
<tr><td><code id="IBD_+3A_mantelpermutations">mantelPermutations</code></td>
<td>
<p>numeric: The number of permutations may be specified</p>
</td></tr>
<tr><td><code id="IBD_+3A_mantelranktest">mantelRankTest</code></td>
<td>
<p>logical: whether to use ranks in the Mantel test</p>
</td></tr>
<tr><td><code id="IBD_+3A_bootstrapmethod">bootstrapMethod</code></td>
<td>
<p>character: which bootstrap method to use (one of 
&quot;ABC&quot;, &quot;BC&quot; or &quot;BCa&quot;).</p>
</td></tr>
<tr><td><code id="IBD_+3A_bootstrapnsim">bootstrapNsim</code></td>
<td>
<p>integer: the number of bootstrap simulations to use 
(has no effect if method is &quot;ABC&quot;).</p>
</td></tr>
<tr><td><code id="IBD_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
locinfile &lt;- genepopExample('w2.txt')
outfile &lt;- ibd(locinfile,'w2.txt.ISO', geographicScale = 'Log', statistic='e')
if ( ! interactive()) clean_workdir(otherfiles='w2.txt')

locinfile &lt;- genepopExample('PEL1600withCoord.txt')
outfile &lt;- ibd(locinfile,'PEL1600withCoord.ISO', statistic = 'SingleGeneDiv',
               geographicScale = '1D')
if ( ! interactive()) clean_workdir(otherfiles='PEL1600withCoord.txt')

## End(Not run)
</code></pre>

<hr>
<h2 id='Linkage'>Tables and exact test for genotypic linkage disequilibrium</h2><span id='topic+Linkage'></span><span id='topic+test_LD'></span><span id='topic+write_LD_tables'></span>

<h3>Description</h3>

<p>Exact test for each pair of loci in each population. See <a href="../doc/all-menu-options.html#option-2-tests-and-tables-for-linkage-disequilibrium">this section</a> of the Genepop executable documentation for more information on the statistical methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_LD(
  inputFile,
  outputFile = "",
  settingsFile = "",
  dememorization = 10000,
  batches = 100,
  iterations = 5000,
  verbose = interactive()
)

write_LD_tables(inputFile, outputFile = "", verbose = interactive())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Linkage_+3A_inputfile">inputFile</code></td>
<td>
<p>The path of the input file, in Genepop format</p>
</td></tr>
<tr><td><code id="Linkage_+3A_outputfile">outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td></tr>
<tr><td><code id="Linkage_+3A_settingsfile">settingsFile</code></td>
<td>
<p>character: The path of the settings file</p>
</td></tr>
<tr><td><code id="Linkage_+3A_dememorization">dememorization</code></td>
<td>
<p>integer: length of dememorization step of Markov chain algorithm</p>
</td></tr>
<tr><td><code id="Linkage_+3A_batches">batches</code></td>
<td>
<p>integer: Number of batches</p>
</td></tr>
<tr><td><code id="Linkage_+3A_iterations">iterations</code></td>
<td>
<p>integer: Iterations per batch</p>
</td></tr>
<tr><td><code id="Linkage_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  # 'dontrun' only because a bit too slow for CRAN checks
locinfile &lt;- genepopExample('sample.txt')
test_LD(locinfile,'sample.txt.DIS')
if ( ! interactive()) clean_workdir(otherfiles='sample.txt')

## End(Not run)
locinfile &lt;- genepopExample('sample.txt')
write_LD_tables(locinfile,'sample.txt.TAB')
if ( ! interactive()) clean_workdir(otherfiles='sample.txt')
</code></pre>

<hr>
<h2 id='manipulation'>Various data manipulation utilities</h2><span id='topic+manipulation'></span><span id='topic+diploidize'></span><span id='topic+relabel_alleles'></span><span id='topic+pop_to_indiv'></span><span id='topic+sample_haploid'></span>

<h3>Description</h3>

<p>Various procedures described in the linked sections of the Genepop executable documentation: <a href="../doc/all-menu-options.html#sub-option-2-diploidisation-of-haploid-data">diploidize</a> haploid data, <a href="../doc/all-menu-options.html#sub-option-3-relabeling-alleles-names">relabel_alleles</a>, <a href="../doc/all-menu-options.html#sub-option-6-random-sampling-of-haploid-genotypes-from-diploid-ones">sample_haploid</a>, and <a href="../doc/all-menu-options.html#sub-options-4-and-5-conversion-of-population-data-to-individual-data">pop_to_indiv</a>. The latter procedure converts population samples (several individuals in each population) to individual data. The names given to the individuals in the new file created (names which are to be interpreted as coordinates in a spatial analysis) may be the population coordinates (given as the name of the last individual in the original data file), or each individual's coordinates (given as the name of each individual in the original data file).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diploidize(inputFile, outputFile = "", verbose = interactive())

relabel_alleles(inputFile, outputFile = "", verbose = interactive())

pop_to_indiv(inputFile, coordinates, outputFile = "", verbose = interactive())

sample_haploid(inputFile, outputFile = "", verbose = interactive())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="manipulation_+3A_inputfile">inputFile</code></td>
<td>
<p>The path of the input file, in Genepop format</p>
</td></tr>
<tr><td><code id="manipulation_+3A_outputfile">outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td></tr>
<tr><td><code id="manipulation_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
<tr><td><code id="manipulation_+3A_coordinates">coordinates</code></td>
<td>
<p>character: either <code>'population'</code> (to use population coordinates) or any other charater string (to use individual coordinates).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>locinfile &lt;- genepopExample('sample.txt')
outfile &lt;- diploidize(inputFile = locinfile,outputFile="Dsample.txt")
if ( ! interactive()) clean_workdir(c("sample.txt", "Dsample.txt"))
</code></pre>

<hr>
<h2 id='Nm_private'>Private allele method</h2><span id='topic+Nm_private'></span>

<h3>Description</h3>

<p>Estimation of Nm by private allele method of Slatkin and Barton. See <a href="../doc/all-menu-options.html#option-4-private-alleles">this section</a> of the Genepop executable documentation for more information on the statistical methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nm_private(
  inputFile,
  outputFile = "",
  dataType = "Diploid",
  verbose = interactive()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nm_private_+3A_inputfile">inputFile</code></td>
<td>
<p>The path of the input file, in Genepop format</p>
</td></tr>
<tr><td><code id="Nm_private_+3A_outputfile">outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td></tr>
<tr><td><code id="Nm_private_+3A_datatype">dataType</code></td>
<td>
<p>character: The haploid and diploid data</p>
</td></tr>
<tr><td><code id="Nm_private_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>locinfile &lt;- genepopExample('sample.txt')
Nm_private(locinfile,'sample.txt.PRI')
if ( ! interactive()) clean_workdir(otherfiles='sample.txt')
</code></pre>

<hr>
<h2 id='nulls'>Estimation of allele frequencies under genotyping failure.</h2><span id='topic+nulls'></span>

<h3>Description</h3>

<p>Estimates allele frequencies (and failure rate if relevant) under dfferent assumptions:  maximum likelihood assuming that there is null allele (default method), maximum likelihood assuming that apparent nulls are technical failures independent of genotype (<code>'ApparentNulls'</code>), and Brookfield's (1996) estimator (<code>'B96'</code>). See <a href="../doc/all-menu-options.html#sub-option-1-null-alleles">this section</a> of the Genepop executable documentation for more information on the statistical methods. Genepop takes the allele with the highest number for a given locus across all populations as the null allele. For example, if you have 4 alleles plus a null allele, a null homozygote individual should be indicated as e.g. <code>0505</code> or <code>9999</code> in the input file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nulls(
  inputFile,
  outputFile = "",
  settingsFile = "",
  nullAlleleMethod = "",
  CIcoverage = 0.95,
  verbose = interactive()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nulls_+3A_inputfile">inputFile</code></td>
<td>
<p>The path of the input file, in Genepop format</p>
</td></tr>
<tr><td><code id="nulls_+3A_outputfile">outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td></tr>
<tr><td><code id="nulls_+3A_settingsfile">settingsFile</code></td>
<td>
<p>character: The path of the settings file</p>
</td></tr>
<tr><td><code id="nulls_+3A_nullallelemethod">nullAlleleMethod</code></td>
<td>
<p>character: <code>'ApparentNulls'</code>, <code>'B96'</code> or anything else (default method).</p>
</td></tr>
<tr><td><code id="nulls_+3A_cicoverage">CIcoverage</code></td>
<td>
<p>numeric: The coverage probability of confidence interval</p>
</td></tr>
<tr><td><code id="nulls_+3A_verbose">verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>

<hr>
<h2 id='set_restriction'>Programming utilities</h2><span id='topic+set_restriction'></span><span id='topic+genepop-utils'></span><span id='topic+getVersion'></span>

<h3>Description</h3>

<p><code>getVersion</code> returns the version number of the C++ code (the same number that identifies the C++ executable). <code>set_restriction(TRUE)</code> sets the maximum number of populations and of loci to 300.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_restriction(set = FALSE)

getVersion()
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_restriction_+3A_set">set</code></td>
<td>
<p>logical: whether to set restrictions on number of populations and of loci</p>
</td></tr>
</table>

<hr>
<h2 id='setMantelSeed'>Set random generator seed for Mantel test</h2><span id='topic+setMantelSeed'></span>

<h3>Description</h3>

<p>Set random generator seed for Mantel test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setMantelSeed(seed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setMantelSeed_+3A_seed">seed</code></td>
<td>
<p>integer: the new seed</p>
</td></tr>
</table>

<hr>
<h2 id='setRandomSeed'>Set random generator seed (except for Mantel test)</h2><span id='topic+setRandomSeed'></span>

<h3>Description</h3>

<p>Set random generator seed (except for Mantel test)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setRandomSeed(seed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setRandomSeed_+3A_seed">seed</code></td>
<td>
<p>integer: the new seed</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
