<!DOCTYPE html><html><head><title>Help for package genieBPC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {genieBPC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#genieBPC-package'><p>genieBPC: Project GENIE BioPharma Collaborative Data Processing Pipeline</p></a></li>
<li><a href='#.check_download_path'><p>Check download_path user passed and create folder if needed</p></a></li>
<li><a href='#.get_and_query_file_url'><p>Get URL for a given 'Synapse' file and download to local machine</p></a></li>
<li><a href='#.get_env'><p>Get an object from the genieBPC package environment ('genieBPC_env')</p></a></li>
<li><a href='#.get_synapse_token'><p>Retrieve a synapse token using username and password</p></a></li>
<li><a href='#.get_token_by_pat'><p>Retrieve a synapse token using personal access token</p></a></li>
<li><a href='#.get_token_by_username'><p>Retrieve a synapse token using username and password</p></a></li>
<li><a href='#.is_connected_to_genie'><p>Check Synapse Login Status &amp; Ability to Access Data</p></a></li>
<li><a href='#.pull_data_by_cohort'><p>Function to retrieve data by synapse ID</p></a></li>
<li><a href='#.resolve_duplicates'><p>Select unique NGS report when multiple are available</p></a></li>
<li><a href='#check_genie_access'><p>Check Access to GENIE Data</p></a></li>
<li><a href='#create_analytic_cohort'><p>Select cohort of patients for analysis</p></a></li>
<li><a href='#drug_regimen_list'><p>List of Drug Regimen Names by Cohort</p></a></li>
<li><a href='#drug_regimen_sunburst'><p>Visualize drug regimen sequences in a sunburst plot</p></a></li>
<li><a href='#genie_panels'><p>Genomic Panels Included in GENIE BPC Data</p></a></li>
<li><a href='#nsclc_test_data'><p>Simulated fake GENIE BPC data for function examples and tests</p></a></li>
<li><a href='#pull_data_synapse'><p>Obtain clinical &amp; genomic data files for GENIE BPC Project</p></a></li>
<li><a href='#regimen_abbreviations'><p>List of Drug Regimen Abbreviations</p></a></li>
<li><a href='#select_unique_ngs'><p>Selecting corresponding unique next generation sequencing reports</p></a></li>
<li><a href='#set_synapse_credentials'><p>Connect to 'Synapse' API</p></a></li>
<li><a href='#synapse_tables'><p>'Synapse' table IDs</p></a></li>
<li><a href='#synapse_version'><p>Return list of available GENIE BPC data releases</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Project GENIE BioPharma Collaborative Data Processing Pipeline</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>The American Association Research (AACR) Project 
    Genomics Evidence Neoplasia Information Exchange (GENIE) BioPharma 
    Collaborative represents a multi-year, multi-institution effort
    to build a pan-cancer repository of linked clinico-genomic data. 
    The genomic and clinical data are provided in multiple releases (separate
    releases for each cancer cohort with updates following data corrections), 
    which are stored on the data sharing platform 
    'Synapse' <a href="https://www.synapse.org/">https://www.synapse.org/</a>. 
    The 'genieBPC' package provides a seamless way to obtain the
    data corresponding to each release from 'Synapse' and to prepare
    datasets for analysis. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/GENIE-BPC/genieBPC/issues">https://github.com/GENIE-BPC/genieBPC/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4)</td>
</tr>
<tr>
<td>Imports:</td>
<td>cli (&ge; 2.5.0), dplyr (&ge; 1.0.6), dtplyr (&ge; 1.1.0), httr,
jsonlite, purrr (&ge; 0.3.4), rlang (&ge; 1.0.0), stringr (&ge;
1.4.0), sunburstR, tibble (&ge; 3.1.2), tidyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr (&ge; 3.5.1), ggplot2 (&ge; 3.3.5), gt (&ge; 0.3.0), gtsummary
(&ge; 1.5.2), knitr (&ge; 1.33), magrittr (&ge; 2.0.1), plotly (&ge;
4.10.0), rmarkdown (&ge; 2.8), testthat (&ge; 3.0.0), markdown,
spelling</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://genie-bpc.github.io/genieBPC/">https://genie-bpc.github.io/genieBPC/</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-11 21:55:28 UTC; laveryj</td>
</tr>
<tr>
<td>Author:</td>
<td>Jessica A. Lavery <a href="https://orcid.org/0000-0002-2746-5647"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Michael A. Curry <a href="https://orcid.org/0000-0002-0261-4044"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Samantha Brown <a href="https://orcid.org/0000-0001-5352-974X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Karissa Whiting <a href="https://orcid.org/0000-0002-4683-1868"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Hannah Fuchs <a href="https://orcid.org/0000-0003-2426-0096"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Axel Martin [aut],
  Daniel D. Sjoberg <a href="https://orcid.org/0000-0003-0862-2018"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jessica A. Lavery &lt;laveryj@mskcc.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-11 22:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='genieBPC-package'>genieBPC: Project GENIE BioPharma Collaborative Data Processing Pipeline</h2><span id='topic+genieBPC'></span><span id='topic+genieBPC-package'></span>

<h3>Description</h3>

<p>The American Association Research (AACR) Project Genomics Evidence Neoplasia Information Exchange (GENIE) BioPharma Collaborative represents a multi-year, multi-institution effort to build a pan-cancer repository of linked clinico-genomic data. The genomic and clinical data are provided in multiple releases (separate releases for each cancer cohort with updates following data corrections), which are stored on the data sharing platform 'Synapse' <a href="https://www.synapse.org/">https://www.synapse.org/</a>. The 'genieBPC' package provides a seamless way to obtain the data corresponding to each release from 'Synapse' and to prepare datasets for analysis.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Jessica A. Lavery <a href="mailto:laveryj@mskcc.org">laveryj@mskcc.org</a> (<a href="https://orcid.org/0000-0002-2746-5647">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Michael A. Curry (<a href="https://orcid.org/0000-0002-0261-4044">ORCID</a>)
</p>
</li>
<li><p> Samantha Brown (<a href="https://orcid.org/0000-0001-5352-974X">ORCID</a>)
</p>
</li>
<li><p> Karissa Whiting (<a href="https://orcid.org/0000-0002-4683-1868">ORCID</a>)
</p>
</li>
<li><p> Hannah Fuchs (<a href="https://orcid.org/0000-0003-2426-0096">ORCID</a>)
</p>
</li>
<li><p> Axel Martin
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Daniel D. Sjoberg <a href="mailto:danield.sjoberg@gmail.com">danield.sjoberg@gmail.com</a> (<a href="https://orcid.org/0000-0003-0862-2018">ORCID</a>) [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://genie-bpc.github.io/genieBPC/">https://genie-bpc.github.io/genieBPC/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/GENIE-BPC/genieBPC/issues">https://github.com/GENIE-BPC/genieBPC/issues</a>
</p>
</li></ul>


<hr>
<h2 id='.check_download_path'>Check download_path user passed and create folder if needed</h2><span id='topic+.check_download_path'></span>

<h3>Description</h3>

<p>Check download_path user passed and create folder if needed
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.check_download_path(download_location, version_num)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".check_download_path_+3A_download_location">download_location</code></td>
<td>
<p>a local path or NULL</p>
</td></tr>
<tr><td><code id=".check_download_path_+3A_version_num">version_num</code></td>
<td>
<p>vector of cohort/version_number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of file paths. If download_location is NULL, will return
temporary file path
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.check_download_path(download_location = NULL, version_num = "CRC_v2.1")

</code></pre>

<hr>
<h2 id='.get_and_query_file_url'>Get URL for a given 'Synapse' file and download to local machine</h2><span id='topic+.get_and_query_file_url'></span>

<h3>Description</h3>

<p>Get URL for a given 'Synapse' file and download to local machine
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get_and_query_file_url(
  version_num,
  file_handle_id,
  synapse_id,
  df,
  name,
  download_folder,
  download_location,
  token,
  file_endpoint_url
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".get_and_query_file_url_+3A_version_num">version_num</code></td>
<td>
<p>cohort name and version</p>
</td></tr>
<tr><td><code id=".get_and_query_file_url_+3A_file_handle_id">file_handle_id</code></td>
<td>
<p>'Synapse' file handle ID</p>
</td></tr>
<tr><td><code id=".get_and_query_file_url_+3A_synapse_id">synapse_id</code></td>
<td>
<p>'Synapse' ID</p>
</td></tr>
<tr><td><code id=".get_and_query_file_url_+3A_df">df</code></td>
<td>
<p>package designated name of file</p>
</td></tr>
<tr><td><code id=".get_and_query_file_url_+3A_name">name</code></td>
<td>
<p>file name from 'Synapse'</p>
</td></tr>
<tr><td><code id=".get_and_query_file_url_+3A_download_folder">download_folder</code></td>
<td>
<p>location to download data</p>
</td></tr>
<tr><td><code id=".get_and_query_file_url_+3A_token">token</code></td>
<td>
<p>Synapse token</p>
</td></tr>
<tr><td><code id=".get_and_query_file_url_+3A_file_endpoint_url">file_endpoint_url</code></td>
<td>
<p>'Synapse' endpoint for file info</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of 'Synapse' data frames
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

file &lt;- data.frame(
  version_num = "NSCLC_v2.1",
  file_handle_id = c("79432768"),
  synapse_id = c("syn25985884"),
  df = c("pt_char"),
  name = c("patient_level_dataset.csv"),
  download_folder = file.path(tempdir(), "NSCLC_v2.1")
)

purrr::pmap(file, .get_and_query_file_url)

</code></pre>

<hr>
<h2 id='.get_env'>Get an object from the genieBPC package environment ('genieBPC_env')</h2><span id='topic+.get_env'></span>

<h3>Description</h3>

<p>Get an object from the genieBPC package environment ('genieBPC_env')
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get_env(thing_to_check)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".get_env_+3A_thing_to_check">thing_to_check</code></td>
<td>
</td></tr>
</table>


<h3>Value</h3>

<p>The object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
.get_env("username")

## End(Not run)
</code></pre>

<hr>
<h2 id='.get_synapse_token'>Retrieve a synapse token using username and password</h2><span id='topic+.get_synapse_token'></span>

<h3>Description</h3>

<p>Retrieve a synapse token using username and password
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get_synapse_token(username = NULL, password = NULL, pat = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".get_synapse_token_+3A_username">username</code></td>
<td>
<p>'Synapse' username. If NULL, package will search package
environment for &quot;username&quot;.</p>
</td></tr>
<tr><td><code id=".get_synapse_token_+3A_password">password</code></td>
<td>
<p>'Synapse' password. If NULL, package will search package
environment for &quot;password&quot;.</p>
</td></tr>
<tr><td><code id=".get_synapse_token_+3A_pat">pat</code></td>
<td>
<p>'Synapse' Personal Access Token. If NULL, package will search package
environment for &quot;pat&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 'Synapse' token. Will first look for PAT and return that if successful HTTP call, else
will try with username and password
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
.get_synapse_token(username = "test", password = "test")

## End(Not run)
</code></pre>

<hr>
<h2 id='.get_token_by_pat'>Retrieve a synapse token using personal access token</h2><span id='topic+.get_token_by_pat'></span>

<h3>Description</h3>

<p>Retrieve a synapse token using personal access token
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get_token_by_pat(pat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".get_token_by_pat_+3A_pat">pat</code></td>
<td>
<p>'Synapse' Personal Access Token. If NULL, package will search package
environment for &quot;pat&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 'Synapse' token
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
.get_token_by_pat(pat = "test")

## End(Not run)
</code></pre>

<hr>
<h2 id='.get_token_by_username'>Retrieve a synapse token using username and password</h2><span id='topic+.get_token_by_username'></span>

<h3>Description</h3>

<p>Retrieve a synapse token using username and password
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get_token_by_username(username, password)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".get_token_by_username_+3A_username">username</code></td>
<td>
<p>'Synapse' username. If NULL, package will search package
environment for &quot;username&quot;.</p>
</td></tr>
<tr><td><code id=".get_token_by_username_+3A_password">password</code></td>
<td>
<p>'Synapse' password. If NULL, package will search package
environment for &quot;password&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 'Synapse' token.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
.get_token_by_username(username = "test", password = "test")

## End(Not run)
</code></pre>

<hr>
<h2 id='.is_connected_to_genie'>Check Synapse Login Status &amp; Ability to Access Data</h2><span id='topic+.is_connected_to_genie'></span>

<h3>Description</h3>

<p>The '.is_connected_to_genie()' function assesses whether the
user is logged into 'Synapse' and confirms whether the
user has permission to access the GENIE BPC data. It returns TRUE or FALSE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.is_connected_to_genie(
  username = NULL,
  password = NULL,
  pat = NULL,
  check_consortium_access = FALSE
)
</code></pre>


<h3>Value</h3>

<p>Returns message indicating user is logged into
'Synapse' and has permission to access the GENIE BPC data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.is_connected_to_genie()
</code></pre>

<hr>
<h2 id='.pull_data_by_cohort'>Function to retrieve data by synapse ID</h2><span id='topic+.pull_data_by_cohort'></span>

<h3>Description</h3>

<p>Function to retrieve data by synapse ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pull_data_by_cohort(version_num_df, token, download_location)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pull_data_by_cohort_+3A_version_num_df">version_num_df</code></td>
<td>
<p>a dataframe of 'Synapse' IDs</p>
</td></tr>
<tr><td><code id=".pull_data_by_cohort_+3A_token">token</code></td>
<td>
<p>a 'Synapse' token</p>
</td></tr>
<tr><td><code id=".pull_data_by_cohort_+3A_download_location">download_location</code></td>
<td>
<p>if 'NULL' (default), data will be returned as a list
of dataframes with requested data as list items. Otherwise, specify a
folder path to have data automatically downloaded there.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>downloaded 'Synapse' data as a list if 'download_location'= 'NULL, or
to a local path
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

temp_directory &lt;- tempdir()

syn_df &lt;- data.frame(
  cohort = c("NSCLC", "NSCLC"),
  version = c("v2.2-consortium", "v2.0-public"),
  version_num = c("NSCLC_v2.2", "NSCLC_v2.0"),
  download_folder = c(temp_directory, temp_directory),
  df = c("pt_char", "ca_dx_index"),
  synapse_id = c("syn53470868", "syn30350575")
)

.pull_data_by_cohort(
  version_num_df = syn_df,
  token = .get_synapse_token(), download_location = NULL
)

#

</code></pre>

<hr>
<h2 id='.resolve_duplicates'>Select unique NGS report when multiple are available</h2><span id='topic+.resolve_duplicates'></span>

<h3>Description</h3>

<p>See 'select_unique_ngs' for details on selection criteria
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.resolve_duplicates(x, data_cohort, oncotree_code, sample_type, min_max_time)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".resolve_duplicates_+3A_x">x</code></td>
<td>
<p>sample ID to select unique NGS report for</p>
</td></tr>
<tr><td><code id=".resolve_duplicates_+3A_data_cohort">data_cohort</code></td>
<td>
<p>CPT (NGS) dataframe returned from the create_analytic_cohort function</p>
</td></tr>
<tr><td><code id=".resolve_duplicates_+3A_oncotree_code">oncotree_code</code></td>
<td>
<p>character vector specifying which sample
OncoTree codes to keep. See &quot;cpt_oncotree_code&quot; column
of data_cohort argument above to get options.</p>
</td></tr>
<tr><td><code id=".resolve_duplicates_+3A_sample_type">sample_type</code></td>
<td>
<p>character specifying which type of genomic sample
to prioritize, options are &quot;Primary&quot;, &quot;Local&quot; and &quot;Metastasis&quot;.
Default is to not select a NGS sample based on the sample type.</p>
</td></tr>
<tr><td><code id=".resolve_duplicates_+3A_min_max_time">min_max_time</code></td>
<td>
<p>character specifying if the first or last genomic
sample recorded should be kept.
Options are &quot;min&quot; (first) and &quot;max&quot; (last).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of samples with one observation per patient.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
nsclc_2_0 &lt;- pull_data_synapse("NSCLC", version = "v2.0-public")

ex1 &lt;- create_analytic_cohort(
  data_synapse = nsclc_2_0$NSCLC_v2.0,
  stage_dx = c("Stage IV"),
  histology = "Adenocarcinoma"
)

 samples_data1 &lt;- .resolve_duplicates(
   x = "GENIE-MSK-P-0025741",
  data_cohort = ex1$cohort_ngs,
  oncotree_code = "LUAD",
  sample_type = "Metastasis",
  min_max_time = "max"
)

samples_data2 &lt;- .resolve_duplicates(
   x = "GENIE-MSK-P-0025741",
  data_cohort = ex1$cohort_ngs,
  oncotree_code = "LUAD",
  sample_type = "Primary",
  min_max_time = "max"
)

</code></pre>

<hr>
<h2 id='check_genie_access'>Check Access to GENIE Data</h2><span id='topic+check_genie_access'></span>

<h3>Description</h3>

<p>Check Access to GENIE Data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_genie_access(
  username = NULL,
  password = NULL,
  pat = NULL,
  check_consortium_access = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_genie_access_+3A_username">username</code></td>
<td>
<p>'Synapse' username. If NULL, package will search package
environment for &quot;username&quot;.</p>
</td></tr>
<tr><td><code id="check_genie_access_+3A_password">password</code></td>
<td>
<p>'Synapse' password. If NULL, package will search package
environment for &quot;password&quot;.</p>
</td></tr>
<tr><td><code id="check_genie_access_+3A_pat">pat</code></td>
<td>
<p>'Synapse' Personal Access Token. If NULL, package will search package
environment for &quot;pat&quot;.</p>
</td></tr>
<tr><td><code id="check_genie_access_+3A_check_consortium_access">check_consortium_access</code></td>
<td>
<p>Specifies whether access to GENIE BPC consortium
data releases (vs. public data releases) is checked. Default is FALSE,
indicating that access to GENIE BPC public data releases is checked instead.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A success message if you are able to access GENIE BPC data; otherwise
an error
</p>


<h3>Author(s)</h3>

<p>Karissa Whiting
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# if credentials are saved:
check_genie_access()

## End(Not run)
</code></pre>

<hr>
<h2 id='create_analytic_cohort'>Select cohort of patients for analysis</h2><span id='topic+create_analytic_cohort'></span>

<h3>Description</h3>

<p>This function allows the user to create a cohort from the GENIE BPC data
based on cancer diagnosis information such as cancer cohort, treating
institution, histology, and stage at diagnosis, as well as cancer-directed
regimen information including regimen name and regimen order. This function
returns each of the clinical and genomic data files subset on the patients
that met criteria for the analytic cohort. Documentation regarding the
structure and contents of each file can be found in the Analytic Data Guide
corresponding to each data release, as well as in the
<a href="https://genie-bpc.github.io/genieBPC/articles/clinical_data_structure_vignette.html">Clinical Data Structure vignette</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_analytic_cohort(
  data_synapse,
  index_ca_seq = 1,
  institution,
  stage_dx,
  histology,
  regimen_drugs,
  regimen_type = "Exact",
  regimen_order,
  regimen_order_type,
  return_summary = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_analytic_cohort_+3A_data_synapse">data_synapse</code></td>
<td>
<p>The item from the nested list returned from
pull_data_synapse() that corresponds to the cancer cohort of interest.</p>
</td></tr>
<tr><td><code id="create_analytic_cohort_+3A_index_ca_seq">index_ca_seq</code></td>
<td>
<p>Index cancer sequence. Default is 1, indicating the
patient's first index cancer. The index cancer is also referred to as the
BPC Project cancer in the GENIE BPC Analytic Data Guide; this is the
cancer that met the eligibility criteria for the project and was
selected at random for PRISSMM phenomic data curation.
Specifying multiple index cancer sequences, e.g.
index_ca_seq = c(1, 2) will return index cancers to
patients with 1 index cancer and will return the first AND second index
cancers to patients with multiple.</p>
</td></tr>
<tr><td><code id="create_analytic_cohort_+3A_institution">institution</code></td>
<td>
<p>GENIE BPC participating institution. Must be one of
&quot;DFCI&quot;, &quot;MSK&quot;, &quot;UHN&quot;, or &quot;VICC&quot; for NSCLC, BLADDER, Prostate, and PANC cohorts; must be one of &quot;DFCI&quot;,
&quot;MSK&quot;, &quot;VICC&quot; for CRC and BrCa. Default selection is all institutions.
This parameter corresponds to the variable 'institution' in the
Analytic Data Guide.</p>
</td></tr>
<tr><td><code id="create_analytic_cohort_+3A_stage_dx">stage_dx</code></td>
<td>
<p>Stage at diagnosis. Must be one of &quot;Stage I&quot;, &quot;Stage II&quot;,
&quot;Stage III&quot;, &quot;Stage I-III NOS&quot;, &quot;Stage IV&quot;. The default selection is all
stages.
Note that if this parameter is specified, any cases that are missing stage
information are automatically excluded from the resulting cohort.
This parameter corresponds to the variable 'stage_dx' in the
Analytic Data Guide.</p>
</td></tr>
<tr><td><code id="create_analytic_cohort_+3A_histology">histology</code></td>
<td>
<p>Cancer histology. For all cancer cohorts except for BrCa
(breast cancer), this parameter corresponds to the variable
'ca_hist_adeno_squamous' and must be one of &quot;Adenocarcinoma&quot;,
&quot;Squamous cell&quot;, &quot;Sarcoma&quot;,
&quot;Small cell carcinoma&quot;, &quot;Carcinoma&quot;, &quot;Other histologies/mixed tumor&quot;.
For BrCa, this parameter corresponds to the variable
'ca_hist_brca' and must be one of
&quot;Invasive lobular carcinoma&quot;, &quot;Invasive ductal carcinoma&quot;, &quot;Other histology&quot;.
The default selection is all histologies. Note that if this parameter is
specified, any cases that are missing histology information are automatically
excluded from the resulting cohort.</p>
</td></tr>
<tr><td><code id="create_analytic_cohort_+3A_regimen_drugs">regimen_drugs</code></td>
<td>
<p>Vector with names of drugs in cancer-directed regimen,
separated by a comma. For example, to specify a regimen consisting of
Carboplatin and Pemetrexed, specify regimen_drugs = &quot;Carboplatin,
Pemetrexed&quot;. Acceptable values are found in the 'drug_regimen_list'
dataset provided with this package. This parameter
corresponds to the variable 'regimen_drugs' in the Analytic Data Guide.</p>
</td></tr>
<tr><td><code id="create_analytic_cohort_+3A_regimen_type">regimen_type</code></td>
<td>
<p>Indicates whether the regimen(s) specified in
'regimen_drugs' indicates the exact regimen to return, or if regimens
containing the drugs listed in 'regimen_drugs' should be returned. Must be
one of &quot;Exact&quot; or &quot;Containing&quot;. The default is &quot;Exact&quot;.</p>
</td></tr>
<tr><td><code id="create_analytic_cohort_+3A_regimen_order">regimen_order</code></td>
<td>
<p>Order of cancer-directed regimen. If multiple drugs
are specified, 'regimen_order' indicates the regimen order for all drugs;
different values of 'regimen_order' cannot be specified for different drug
regimens. If multiple values are specified, e.g. c(1, 2), then drug regimens
that met either order criteria are returned.</p>
</td></tr>
<tr><td><code id="create_analytic_cohort_+3A_regimen_order_type">regimen_order_type</code></td>
<td>
<p>Specifies whether the 'regimen_order' parameter
refers to the order of receipt of the drug regimen within the cancer
diagnosis (across all other drug regimens; &quot;within cancer&quot;) or the order of
receipt of the drug regimen within the times that that drug regimen was
administered (e.g. the first time carboplatin pemetrexed was received, out
of all times that the patient received carboplatin pemetrexed; &quot;within
regimen&quot;). Acceptable values are &quot;within cancer&quot; and &quot;within regimen&quot;.</p>
</td></tr>
<tr><td><code id="create_analytic_cohort_+3A_return_summary">return_summary</code></td>
<td>
<p>Specifies whether a summary table for the cohort is
returned. Default is FALSE. The 'gtsummary' package is required to return a
summary table.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the <a href="https://genie-bpc.github.io/genieBPC/articles/create_analytic_cohort_vignette.html">create_analytic_cohort vignette</a> for further documentation and examples.
</p>


<h3>Value</h3>

<p>A list of data frames containing clinical and next generation
sequencing information for patients that met the specified criteria.
Optionally, if return_summary = TRUE, the list also includes summary
tables for the number of records per dataset ('tbl_overall_summary')
as well as tables of key cancer diagnosis ('tbl_cohort'),
cancer-directed regimen ('tbl_drugs') and next generation sequencing
('tbl_ngs') variables.
</p>


<h3>Author(s)</h3>

<p>Jessica Lavery
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Examples using package test data
# Example 1 ----------------------------------
# Create a cohort of all patients with stage IV NSCLC adenocarcinoma and
# obtain all of their corresponding clinical and genomic data

ex1 &lt;- create_analytic_cohort(
  data_synapse = genieBPC::nsclc_test_data,
  stage_dx = "Stage IV",
  histology = "Adenocarcinoma"
)

names(ex1)

# Example 2 ----------------------------------
# Create a cohort of all NSCLC patients who received Cisplatin,
# Pemetrexed Disodium or Cisplatin, Etoposide as their first drug regimen
# for their first index NSCLC

ex2 &lt;- create_analytic_cohort(
  data_synapse = genieBPC::nsclc_test_data,
  regimen_drugs = c(
    "Cisplatin, Pemetrexed Disodium",
    "Cisplatin, Etoposide"
  ),
  regimen_order = 1,
  regimen_order_type = "within cancer"
)

# Example 3 ----------------------------------
# Create a cohort of all NSCLC patients who received Cisplatin, Pemetrexed
# Disodium at any time throughout the course of treatment for their
# cancer diagnosis,
# but in the event that the patient received the drug multiple times,
# only select the first time.

ex3 &lt;- create_analytic_cohort(
  data_synapse = genieBPC::nsclc_test_data,
  regimen_drugs = c("Cisplatin, Pemetrexed Disodium"),
  regimen_order = 1,
  regimen_order_type = "within regimen"
)


# Example 4 ----------------------------------
# Using create_analytic_cohort with pull_data_synapse
set_synapse_credentials()

nsclc_2_0 &lt;- pull_data_synapse("NSCLC", version = "v2.0-public")

ex4 &lt;- create_analytic_cohort(
  data_synapse = nsclc_2_0$NSCLC_v2.0,
  regimen_drugs = c("Cisplatin, Pemetrexed Disodium"),
  regimen_order = 1,
  regimen_order_type = "within regimen"
)

</code></pre>

<hr>
<h2 id='drug_regimen_list'>List of Drug Regimen Names by Cohort</h2><span id='topic+drug_regimen_list'></span>

<h3>Description</h3>

<p>A dataset containing the cancer-directed drug names and their synonyms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_regimen_list
</code></pre>


<h3>Format</h3>

<p>A table for cancer-directed drug names associated with each
data release:
</p>

<dl>
<dt>cohort</dt><dd><p>GENIE BPC Project cancer. One of &quot;NSCLC&quot;
(non-small cell lung cancer), &quot;CRC&quot; (colorectal cancer),
&quot;BrCa&quot; (breast cancer), &quot;PANC&quot; (pancreatic cancer),
&quot;Prostate&quot; (prostate cancer), and &quot;BLADDER&quot; (bladder cancer).</p>
</dd>
<dt>cohort_data_release</dt><dd><p>GENIE BPC data release. Occasionally, drug names
were updated across releases to include additional drug name synonyms.</p>
</dd>
<dt>drug_name</dt><dd><p>Name of generic/ingredient cancer-directed drug</p>
</dd>
<dt>drug_name_full</dt><dd><p>Name of generic/ingredient cancer-directed drug
with associated synonyms in parentheses</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='drug_regimen_sunburst'>Visualize drug regimen sequences in a sunburst plot</h2><span id='topic+drug_regimen_sunburst'></span>

<h3>Description</h3>

<p>This function allows the user to visualize the complete treatment
course for selected cancer diagnoses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_regimen_sunburst(data_synapse, data_cohort, max_n_regimens = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drug_regimen_sunburst_+3A_data_synapse">data_synapse</code></td>
<td>
<p>The item from the nested list returned from
'pull_data_synapse()'</p>
</td></tr>
<tr><td><code id="drug_regimen_sunburst_+3A_data_cohort">data_cohort</code></td>
<td>
<p>The list returned from the 'create_analytic_cohort()'
function call</p>
</td></tr>
<tr><td><code id="drug_regimen_sunburst_+3A_max_n_regimens">max_n_regimens</code></td>
<td>
<p>The maximum number of regimens displayed in the
sunburst plot</p>
</td></tr>
<tr><td><code id="drug_regimen_sunburst_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to 'sunburstR::sunburst()'</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the <a href="https://genie-bpc.github.io/genieBPC/articles/drug_regimen_sunburst_vignette.html">drug_regimen_sunburst vignette</a>
for additional details and examples.
</p>


<h3>Value</h3>

<p>Returns data frame 'treatment_history' and interactive
plot 'sunburst_plot'
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1 ----------------------------------
# Example using package test data
# get clinico-genomic files for a specific cohort
nsclc_sub &lt;- create_analytic_cohort(
  data_synapse = genieBPC::nsclc_test_data,
  stage_dx = c("Stage III", "Stage IV")
)

# create sunburst plot
ex1 &lt;- drug_regimen_sunburst(
  data_synapse = nsclc_test_data,
  data_cohort = nsclc_sub,
  max_n_regimens = 3
)


# Example 2 ----------------------------------
# using pull_data_synapse
nsclc_2_0 &lt;- pull_data_synapse("NSCLC", version = "v2.0-public")

nsclc_stg_iv &lt;- create_analytic_cohort(
  data_synapse = nsclc_2_0$NSCLC_v2.0,
  stage = "Stage IV"
)

ex2 &lt;- drug_regimen_sunburst(
  data_synapse = nsclc_2_0$NSCLC_v2.0,
  data_cohort = nsclc_stg_iv,
  max_n_regimens = 3
)

</code></pre>

<hr>
<h2 id='genie_panels'>Genomic Panels Included in GENIE BPC Data</h2><span id='topic+genie_panels'></span>

<h3>Description</h3>

<p>A dataset containing the name, assay identifier, and number of genes
in each next-generation sequencing targeted panel included in GENIE BPC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genie_panels
</code></pre>


<h3>Format</h3>

<p>A data frame with 12 rows and 3 variables:
</p>

<dl>
<dt>Sequence.Assay.ID</dt><dd><p>Next-generation
sequencing targeted panel assay identifier</p>
</dd>
<dt>Panel</dt><dd><p>Panel name</p>
</dd>
<dt>Genes</dt><dd><p>Number of genes included</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='nsclc_test_data'>Simulated fake GENIE BPC data for function examples and tests</h2><span id='topic+nsclc_test_data'></span>

<h3>Description</h3>

<p>A named list of simulated NSCLC clinical and genomic data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nsclc_test_data
</code></pre>


<h3>Format</h3>

<p>A list of data frames
</p>

<dl>
<dt>pt_char</dt><dd><p>Patient characteristic data.frame</p>
</dd>
<dt>ca_dx_index</dt><dd><p>Index cancer diagnosis data.frame</p>
</dd>
<dt>ca_dx_non_index</dt><dd><p>Non-index cancer diagnosis data.frame</p>
</dd>
<dt>ca_drugs</dt><dd><p>Cancer directed-regimen data.frame</p>
</dd>
<dt>prissmm_imaging</dt><dd><p>PRISSMM Imaging report data.frame</p>
</dd>
<dt>prissmm_pathology</dt><dd><p>PRISSMM Pathology report data.frame</p>
</dd>
<dt>prissmm_md</dt><dd><p>PRISSMM medical oncologist report data.frame</p>
</dd>
<dt>cpt</dt><dd><p>Cancer Panel Test (CPT)/Next Generation Sequencing (NGS) data.frame</p>
</dd>
<dt>mutations_extended</dt><dd><p>Mutations data.frame</p>
</dd>
<dt>fusions</dt><dd><p>Fusions data.frame</p>
</dd>
<dt>cpt</dt><dd><p>Copy Number Alteration (CNA) data.frame</p>
</dd>
</dl>


<hr>
<h2 id='pull_data_synapse'>Obtain clinical &amp; genomic data files for GENIE BPC Project</h2><span id='topic+pull_data_synapse'></span>

<h3>Description</h3>

<p>Function to access specified
versions of clinical and genomic GENIE BPC data from
<a href="https://www.synapse.org/#!Synapse:syn21226493/wiki/599164">Synapse</a>
and read them into the R environment. See the <a href="https://genie-bpc.github.io/genieBPC/articles/pull_data_synapse_vignette.html">pull_data_synapse vignette</a>
for further documentation and examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pull_data_synapse(
  cohort = NULL,
  version = NULL,
  download_location = NULL,
  username = NULL,
  password = NULL,
  pat = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pull_data_synapse_+3A_cohort">cohort</code></td>
<td>
<p>Vector or list specifying the cohort(s) of interest. Must be
one of &quot;NSCLC&quot; (Non-Small Cell Lung Cancer), &quot;CRC&quot; (Colorectal Cancer), or
&quot;BrCa&quot; (Breast Cancer), &quot;PANC&quot; (Pancreatic Cancer), &quot;Prostate&quot; (Prostate Cancer),
and &quot;BLADDER&quot; (Bladder Cancer). This is not case sensitive.</p>
</td></tr>
<tr><td><code id="pull_data_synapse_+3A_version">version</code></td>
<td>
<p>Vector specifying the version of the cohort. Must match one of the
release versions available for the specified 'cohort' (see 'synapse_version()' for available cohort versions).
When entering multiple cohorts, it is inferred that the order of the version
numbers passed corresponds to the order of the cohorts passed.
Therefore, 'cohort' and 'version' must be in the same
order to ensure the correct data versions are pulled. See examples below for details.</p>
</td></tr>
<tr><td><code id="pull_data_synapse_+3A_download_location">download_location</code></td>
<td>
<p>if 'NULL' (default), data will be returned as a list
of dataframes with requested data as list items. Otherwise, specify a
folder path to have data automatically downloaded there. When a path is
specified, data are not read into the R environment.</p>
</td></tr>
<tr><td><code id="pull_data_synapse_+3A_username">username</code></td>
<td>
<p>'Synapse' username</p>
</td></tr>
<tr><td><code id="pull_data_synapse_+3A_password">password</code></td>
<td>
<p>'Synapse' password</p>
</td></tr>
<tr><td><code id="pull_data_synapse_+3A_pat">pat</code></td>
<td>
<p>'Synapse' personal access token</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a nested list of clinical and genomic data corresponding to
the specified cohort(s).
</p>


<h3>Authentication</h3>

<p>To access data, users must have a valid 'Synapse' account with permission to
access the data set and they must have accepted any necessary 'Terms of Use'.
Users must always authenticate themselves in their current R session.
(see <a href="https://genie-bpc.github.io/genieBPC/articles/pull_data_synapse_vignette.html">README: Data Access and Authentication</a>
</p>
<p>for details).
To set your 'Synapse' credentials during each session, call:
</p>
<p>'set_synapse_credentials(username = &quot;your_username&quot;, password = &quot;your_password&quot;)'
</p>
<p>In addition to passing your 'Synapse' username and password, you may choose to set
your 'Synapse' Personal Access Token (PAT) by calling:
'set_synapse_credentials(pat = &quot;your_pat&quot;)'.
</p>
<p>If your credentials are stored as environmental variables, you do not need to
call 'set_synapse_credentials()' explicitly each session. To store
authentication information in your environmental variables, add the following
to your .Renviron file, then restart your R session ' (tip: you can use
'usethis::edit_r_environ()' to easily open/edit this file):
</p>

<ul>
<li><p> 'SYNAPSE_USERNAME = &lt;your-username&gt;'
</p>
</li>
<li><p> 'SYNAPSE_PASSWORD = &lt;your-password&gt;'
</p>
</li>
<li><p> 'SYNAPSE_PAT = &lt;your-pat&gt;'
</p>
</li></ul>

<p>Alternatively, you can pass your username and password or your PAT to each individual
data pull function if preferred, although it is recommended that you manage
your passwords outside of your scripts for security purposes.
</p>


<h3>Analytic Data Guides</h3>

<p>Documentation corresponding to the clinical data files
can be found on 'Synapse' in the Analytic Data Guides:
</p>

<ul>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn23002641">NSCLC v1.1-Consortium Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn53463493">NSCLC v2.2-Consortium Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn30557304">NSCLC v2.0-Public Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn58597690">NSCLC v3.1-Consortium Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn53463650">CRC v1.3-Consortium Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn31751466">CRC v2.0-Public Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn26077313">BrCa v1.1-Consortium Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn32330194">BrCa v1.2-Consortium Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn30787692">BLADDER v1.1-Consortium Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn53018714">BLADDER v1.2-Consortium Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn29787285">PANC v1.1-Consortium Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn50612821">PANC v1.2-Consortium Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn30148714">Prostate v1.1-Consortium Analytic Data Guide</a>
</p>
</li>
<li> <p><a href="https://www.synapse.org/#!Synapse:syn50612204">Prostate v1.2-Consortium Analytic Data Guide</a>
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Karissa Whiting, Michael Curry
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Example 1 ----------------------------------
# Set up 'Synapse' credentials
set_synapse_credentials()

# Print available versions of the data
synapse_version(most_recent = TRUE)

# Pull version 2.0-public for non-small cell lung cancer
# and version 2.0-public for colorectal cancer data

 ex1 &lt;- pull_data_synapse(
   cohort = c("NSCLC", "CRC"),
   version = c("v2.0-public", "v2.0-public")
 )

 names(ex1)

</code></pre>

<hr>
<h2 id='regimen_abbreviations'>List of Drug Regimen Abbreviations</h2><span id='topic+regimen_abbreviations'></span>

<h3>Description</h3>

<p>A dataset containing the cancer-directed
drug regimens and their common abbreviations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>regimen_abbreviations
</code></pre>


<h3>Format</h3>

<p>A table for cancer-directed drug regimens and
their common abbreviations
</p>

<dl>
<dt>regimen_drugs</dt><dd><p>List of all drugs in the regimen</p>
</dd>
<dt>abbreviation</dt><dd><p>Common name of drug regimen, e.g. FOLFOX</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='select_unique_ngs'>Selecting corresponding unique next generation sequencing reports</h2><span id='topic+select_unique_ngs'></span>

<h3>Description</h3>

<p>For patients with multiple associated next generation (NGS) sequencing
reports, select one unique NGS report per patient for the purpose of creating
an analytic dataset based on user-defined criterion, including OncoTree code,
primary vs. metastatic tumor sample, and earliest vs. most recent sample. If
multiple reports for a patient remain available after the user-defined
specifications, or if no specifications are provided, the panel with the
largest number of genes is selected by default. Sample optimization is
performed in the order that the arguments are specified in the function,
regardless of the arguments’ order provided by the user. Namely, the OncoTree
code is prioritized first, sample type is prioritized second and finally the
time is prioritized last. For patients with exactly one genomic sample, that
unique genomic sample will be returned regardless of whether it meets the
user-specified parameters. Running the select_unique_ngs() function will
ensure that the resulting dataset returned by merging the next generation
sequencing report data onto the cohort_ca_dx dataset returned by
create_analytic_cohort() will maintain the structure of cohort_ca_dx (either
one record per patient or one record per diagnosis). Currently, if multiple
diagnoses per patient are returned from create_analytic_cohort(), using
select_unique_ngs() will select a single NGS report per patient. In future
iterations, this will be updated so that one NGS report per diagnosis can be
selected.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_unique_ngs(
  data_cohort,
  oncotree_code = NULL,
  sample_type = NULL,
  min_max_time = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_unique_ngs_+3A_data_cohort">data_cohort</code></td>
<td>
<p>CPT (NGS) dataframe returned from the create_analytic_cohort function</p>
</td></tr>
<tr><td><code id="select_unique_ngs_+3A_oncotree_code">oncotree_code</code></td>
<td>
<p>character vector specifying which sample
OncoTree codes to keep. See &quot;cpt_oncotree_code&quot; column
of data_cohort argument above to get options.</p>
</td></tr>
<tr><td><code id="select_unique_ngs_+3A_sample_type">sample_type</code></td>
<td>
<p>character specifying which type of genomic sample
to prioritize, options are &quot;Primary&quot;, &quot;Local&quot; and &quot;Metastasis&quot;.
Default is to not select a NGS sample based on the sample type.</p>
</td></tr>
<tr><td><code id="select_unique_ngs_+3A_min_max_time">min_max_time</code></td>
<td>
<p>character specifying if the first or last genomic
sample recorded should be kept.
Options are &quot;min&quot; (first) and &quot;max&quot; (last).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the NGS dataset serves as the link between the clinical and
genomic data, where the NGS dataset includes one record per NGS report per
patient, including the NGS sample ID that is used to link to the genomic
data files. Merging data from the NGS report onto the analytic cohort
returned from create_analytic_cohort() therefore allows users to utilize all
clinical and genomic data available.
</p>
<p>See the
<a href="https://genie-bpc.github.io/genieBPC/articles/select_unique_ngs_vignette.html">select_unique_ngs vignette</a>
for further documentation and examples.
</p>


<h3>Value</h3>

<p>returns the 'cohort_ngs' object of the create_analytic_cohort
with unique genomic samples taken from each patients.
</p>


<h3>Author(s)</h3>

<p>Karissa Whiting
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Example 1 ----------------------------------
# Create a cohort of all patients with stage IV NSCLC of
# histology adenocarcinoma
set_synapse_credentials()

nsclc_2_0 &lt;- pull_data_synapse("NSCLC", version = "v2.0-public")

ex1 &lt;- create_analytic_cohort(
  data_synapse = nsclc_2_0$NSCLC_v2.0,
  stage_dx = c("Stage IV"),
  histology = "Adenocarcinoma"
)

# select unique next generation sequencing reports for those patients
samples_data1 &lt;- select_unique_ngs(
  data_cohort = ex1$cohort_ngs,
  sample_type = "Primary"
)

# Example 2 ----------------------------------
# Create a cohort of all NSCLC patients who
# received Cisplatin, Pemetrexed Disodium or Cisplatin,
# Etoposide as their first drug regimen
ex2 &lt;- create_analytic_cohort(
  data_synapse = nsclc_2_0$NSCLC_v2.0,
  regimen_drugs = c(
    "Cisplatin, Pemetrexed Disodium",
    "Cisplatin, Etoposide"
  ),
  regimen_order = 1,
  regimen_order_type = "within regimen"
)

samples_data2 &lt;- select_unique_ngs(
  data_cohort = ex2$cohort_ngs,
  oncotree_code = "NSCLCPD",
  sample_type = "Metastasis",
  min_max_time = "max"
)

</code></pre>

<hr>
<h2 id='set_synapse_credentials'>Connect to 'Synapse' API</h2><span id='topic+set_synapse_credentials'></span>

<h3>Description</h3>

<p>This function sets 'Synapse' credentials for the user's current session.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_synapse_credentials(username = NULL, password = NULL, pat = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_synapse_credentials_+3A_username">username</code></td>
<td>
<p>'Synapse' username. If NULL, package will search
environmental variables for 'SYNAPSE_USERNAME'.</p>
</td></tr>
<tr><td><code id="set_synapse_credentials_+3A_password">password</code></td>
<td>
<p>'Synapse' password. If NULL, package will search
environmental variables for 'SYNAPSE_PASSWORD'.</p>
</td></tr>
<tr><td><code id="set_synapse_credentials_+3A_pat">pat</code></td>
<td>
<p>'Synapse' Personal Access Token. If NULL, package will search
environmental variables for 'SYNAPSE_PAT'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To access data, users must have a valid 'Synapse' account with permission to
access the data set and they must have accepted any necessary 'Terms of Use'.
Users must authenticate themselves in their current R session.
(See README 'Data Access and Authentication' at https://genie-bpc.github.io/genieBPC/ for details).
To set your 'Synapse' credentials during each session, call:
'set_synapse_credentials(username = &quot;your_username&quot;, password = &quot;your_password&quot;)'.
</p>
<p>In addition to passing your 'Synapse' username and password, you may choose to set
your 'Synapse' Personal Access Token (PAT) by calling:
'set_synapse_credentials(pat = &quot;your_pat&quot;)'.
</p>
<p>If your credentials are stored as environmental variables, you do not need to call
'set_synapse_credentials()' explicitly each session. To store authentication
information in your environmental variables, add the following to your
.Renviron file, then restart your R session
(tip: you can use 'usethis::edit_r_environ()' to easily open/edit this file):
</p>

<ul>
<li><p> 'SYNAPSE_USERNAME = &lt;your-username&gt;'
</p>
</li>
<li><p> 'SYNAPSE_PASSWORD = &lt;your-password&gt;'
</p>
</li>
<li><p> 'SYNAPSE_PAT = &lt;your-pat&gt;'
</p>
</li></ul>

<p>Alternatively, you can pass your username and password or your PAT to each individual
data pull function if preferred, although it is recommended that you manage
your passwords outside of your scripts for security purposes.
</p>


<h3>Value</h3>

<p>A success message if you credentials are valid for 'Synapse'
platform; otherwise an error
</p>


<h3>Author(s)</h3>

<p>Karissa Whiting
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set_synapse_credentials(
  username = "your-username",
  password = "your-password"
)
set_synapse_credentials(
  pat = "your-personal-access-token"
)

## End(Not run)

</code></pre>

<hr>
<h2 id='synapse_tables'>'Synapse' table IDs</h2><span id='topic+synapse_tables'></span>

<h3>Description</h3>

<p>A dataset containing the 'Synapse' table IDs for
each dataset in GENIE BPC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>synapse_tables
</code></pre>


<h3>Format</h3>

<p>A lookup table for 'Synapse' data table IDs:
</p>

<dl>
<dt>cohort</dt><dd><p>GENIE BPC Project Cohort</p>
</dd>
<dt>version</dt><dd><p>Release version</p>
</dd>
<dt>release_date</dt><dd><p>Month and year of data release</p>
</dd>
<dt>df</dt><dd><p>Clinical, Genomic, or Metadata Dataset</p>
</dd>
<dt>synapse_id</dt><dd><p>'Synapse' table ID for each dataset</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p><a href="https://www.synapse.org/#!Synapse:syn21226493/wiki/599164">https://www.synapse.org/#!Synapse:syn21226493/wiki/599164</a>
</p>

<hr>
<h2 id='synapse_version'>Return list of available GENIE BPC data releases</h2><span id='topic+synapse_version'></span>

<h3>Description</h3>

<p>GENIE BPC data are updated periodically to add variables and reflect
additional data cleaning. Each time the data are updated the data release
version number is incremented. The 'synapse_version()' function will get
available version numbers for each cohort to help the user determine what is
the most recent version for each cohort.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>synapse_version(cohort = NULL, most_recent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="synapse_version_+3A_cohort">cohort</code></td>
<td>
<p>Vector specifying the cohort(s) of interest. Cohorts must be
one of &quot;NSCLC&quot; (Non-Small Cell Lung Cancer), &quot;CRC&quot; (Colorectal Cancer), or
&quot;BrCa&quot; (Breast Cancer), &quot;PANC&quot; (Pancreatic Cancer), &quot;Prostate&quot; (Prostate Cancer),
and &quot;BLADDER&quot; (Bladder Cancer).</p>
</td></tr>
<tr><td><code id="synapse_version_+3A_most_recent">most_recent</code></td>
<td>
<p>Indicates whether the function will return only the most
recent version number for each cohort ('most_recent' = TRUE) or all
available version numbers for each cohort ('most_recent'= FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Specifies the version numbers available for each cancer cohort. Version
numbers are specified as part of the call to 'pull_data_synapse()'.
</p>


<h3>Value</h3>

<p>Returns a table containing the available versions for each cohort.
Consortium releases are restricted to GENIE BPC consortium members.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>synapse_version()
synapse_version(most_recent = TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
