<!DOCTYPE html><html lang="en"><head><title>Help for package genomicper</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {genomicper}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#genomicper-package'>
<p>Circular Genomic Permutations</p></a></li>
<li><a href='#demo'>
<p>GWAS p_values demo data</p></a></li>
<li><a href='#genes_permutation'>
<p>Gene-level Permutations</p></a></li>
<li><a href='#genome_order'>
<p>Genome Order</p></a></li>
<li><a href='#get_pathways'>
<p>Pathways</p></a></li>
<li><a href='#get_results'>
<p>Circular Permutation Results</p></a></li>
<li><a href='#hyprbg'>
<p>Hypergeometric Test (phyper)</p></a></li>
<li><a href='#plot_results'>
<p>Plot Results Circular Permutation</p></a></li>
<li><a href='#read_pvals'>
<p>Read GWAS p-values of association and Merge with SNP annotations</p></a></li>
<li><a href='#read2_paths'>
<p>Read to SNPs to sets; Map SNPs to gene-sets/pathways</p></a></li>
<li><a href='#RHSAXXXXX'>
<p>Reactome Pathway examples</p></a></li>
<li><a href='#snps_permutation'>
<p>SNP-level permutations</p></a></li>
<li><a href='#SNPsAnnotation'>
<p>SNPs-Genes annotation to Distance 0 (SNPs within a gene)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Circular Genomic Permutation using Genome Wide Association
p-Values</td>
</tr>
<tr>
<td>Version:</td>
<td>1.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-05-06</td>
</tr>
<tr>
<td>Author:</td>
<td>Claudia P Cabrera [aut, cre],
  Pau Navarro [aut],
  Chris S Haley [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Claudia P Cabrera &lt;c.cabrera@qmul.ac.uk&gt;</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats,grDevices,utils,graphics,DBI,reactome.db,AnnotationDbi</td>
</tr>
<tr>
<td>Description:</td>
<td>Circular genomic permutation approach uses genome wide association studies (GWAS) results to establish the significance of pathway/gene-set associations whilst accounting for genomic structure(Cabrera et al (2012) &lt;<a href="https://doi.org/10.1534%2Fg3.112.002618">doi:10.1534/g3.112.002618</a>&gt;). All single nucleotide polymorphisms (SNPs) in the GWAS are placed in a 'circular genome' according to their location. Then the complete set of SNP association p-values are permuted by rotation with respect to the SNPs' genomic locations. Two testing frameworks are available: permutations at the gene level, and permutations at the SNP level. The permutation at the gene level uses Fisher's combination test to calculate a single gene p-value, followed by the hypergeometric test. The SNP count methodology maps each SNP to pathways/gene-sets and calculates the proportion of SNPs for the real and the permutated datasets above a pre-defined threshold. Genomicper requires a matrix of GWAS association p-values and SNPs annotation to genes. Pathways can be obtained from within the package or can be provided by the user.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-05-06 16:06:27 UTC; PallisDell</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-05-08 08:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='genomicper-package'>
Circular Genomic Permutations
</h2><span id='topic+genomicper-package'></span><span id='topic+genomicper'></span>

<h3>Description</h3>

<p>Description: Circular genomic permutation approach uses genome wide association studies (GWAS) results to establish the significance of pathway/gene-set associations whilst accounting for genomic structure. All single nucleotide polymorphisms (SNPs) in the GWAS are placed in a 'circular genome' according to their location. Then the complete set of SNP association p-values are permuted by rotation with respect to the SNPs' genomic locations. Two testing frameworks are available: permutations at the gene level, and permutations at the SNP level. The permutation at the gene level uses Fisher's combination test to calculate a single gene p-value, followed by the hypergeometric test. The SNP count methodology maps each SNP to pathways/gene-sets and calculates the proportion of SNPs for the real and the permutated datasets above a pre-defined threshold. Genomicper requires a matrix of GWAS association p-values and SNPs annotation to genes. Pathways can be obtained from within the package or can be provided by the user.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> genomicper</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.7</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-05-06</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Claudia P. Cabrera, Pau Navarro, Chris S. Haley <br />
Maintainer: Claudia Cabrera &lt;c.cabrera@qmul.ac.uk&gt;
</p>


<h3>References</h3>

<p>SNP-level Permutations:<br />
Genomicper: genome-wide association SNP-set analysis<br />
Claudia P. Cabrera*, Pau Navarro*, Jennifer E. Huffman, Alan F. Wright, Caroline Hayward,Harry Campbell, James F. Wilson, Igor Rudan, Nicholas D. Hastie, Veronique Vitart, Chris S. Haley*<br />	
<br />
Gene-level Permutations:<br />
Uncovering Networks from Genome-Wide Association Studies via<br /> 
Circular Genomic Permutation. G3: Genes|Genomes|Genetics 2, 1067-1075.<br />
Claudia P. Cabrera*, Pau Navarro*, Jennifer E. Huffman, Alan F. Wright, Caroline Hayward,Harry Campbell, James F. Wilson, Igor Rudan, Nicholas D. Hastie, Veronique Vitart, Chris S. Haley*
</p>


<h3>See Also</h3>

<p>Genomicper functions:	
1)  <code><a href="#topic+read_pvals">read_pvals</a></code>,
2)  <code><a href="#topic+genome_order">genome_order</a></code>,
3)  <code><a href="#topic+get_pathways">get_pathways</a></code>,
4)  <code><a href="#topic+read2_paths">read2_paths</a></code>,
5A) <code><a href="#topic+snps_permutation">snps_permutation</a></code>,
5B) <code><a href="#topic+genes_permutation">genes_permutation</a></code>,
6)  <code><a href="#topic+get_results">get_results</a></code>,
7)  <code><a href="#topic+plot_results">plot_results</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#############################################################################
#  Genomicper functions                                            ##########
# 1)  read_pvals(data_name="",snps_ann="")
# 2)  genome_order(all_data="")
# 3)  get_pathways(source="reactome",all_paths="",envir="")
# 4)  read2_paths(ordered_alldata="",gs_locs="",sets_from="",sets_prefix="RHSA",level="")
# 5A) snps_permutation(ordered_alldata="",pers_ids="",ntraits="",nper="",saveto="",
#		threshold="",gs_locs=gs_locs,envir = "")	
# 5B) genes_permutation(ordered_alldata="",pers_ids="",pathways="",
#		ntraits="",nper="",threshold="",saveto="",gs_locs=gs_locs,envir = "")
# 6)  get_results(res_pattern="Permus",level="snp",from="workspace",
#		threshold=0.05, envir = "")
# 7) plot_results(results = "", by = "", plot_all = TRUE, var = "", save_plot = TRUE, 
# 						plot_name = "", bf = FALSE, save_qq = TRUE)  
#############################################################################
############## DEMO: #######################################################

#### SNP-level  #############################################################
# SNPs annotation and Pathways provided by user
# all data stored at the WORKSPACE

### Load files for analysis
data(demo,SNPsAnnotation)

# Read &amp; format GWAS pvalues
all_data &lt;- read_pvals(data_name=demo,snps_ann=SNPsAnnotation)

# Order data according to the genome
genome_results &lt;-genome_order(all_data=all_data)
	# Results from genome_order
	ordered_alldata &lt;- genome_results$ordered_alldata
	gs_locs &lt;- genome_results$gs_locs

# Create new environment to save variables (e.g. pathways, permutations):
gper.env &lt;- new.env()

# Pathways can be downloaded using the function get_pathways()  
# Load example pathways into the new environment. 
data(RHSA164843,RHSA446343,RHSA8876384,RHSA8964572,RHSA109582,RHSA1474244,envir=gper.env)

# Map SNPs to pathways 
paths_res &lt;- read2_paths(ordered_alldata=ordered_alldata,
gs_locs=gs_locs,sets_from="workspace",sets_prefix="RHSA",
level="snp",envir=gper.env)
	# Results from read2_paths:		
	pers_ids &lt;- paths_res$per_ors
	pathways&lt;- paths_res$pathways

# Perform permutations:
snps_permutation(ordered_alldata=ordered_alldata,
pers_ids=pers_ids,ntraits=c(7:13),nper=10,saveto="workspace",
threshold=0.05,gs_locs=gs_locs,envir = gper.env)		  

# Get results						
results &lt;- get_results(res_pattern="Permus",level="snp",
from="workspace",threshold=0.05,envir = gper.env)

# Plot results
## Not run: 
#saves plots to working directory
qq &lt;- plot_results(results=results,by="set",plot_all=TRUE)
qq &lt;- plot_results(results=results,by="trait",
plot_all=FALSE,var="trait1")
# Displays interactive plot. Select a trait/set to plot and 
# set arguments save_plot=FALSE, plot_all = FALSE
# IMPORTANT: to EXIT interactive plot, RIGHT CLICK on the
# plot and STOP.
qq &lt;- plot_results(results=results,by="set",plot_all=FALSE,
var="RHSA109582",save_plot=FALSE)

## End(Not run)
# -- END OF DEMO 
###############################################
</code></pre>

<hr>
<h2 id='demo'>
GWAS p_values demo data
</h2><span id='topic+demo'></span>

<h3>Description</h3>

<p>GWAS p-values (tab delimited file). 
First Column must contain the SNP ids and the column name = &quot;name&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(demo)</code></pre>


<h3>Format</h3>

<p>A data frame with SNPs identifiers and gwas p-values of association
</p>

<dl>
<dt><code>name</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Trait1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Trait2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Trait3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Trait4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Trait5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Trait6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Trait7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Trait8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Trait9</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>

<pre>
	name       Trait1      Trait2    Trait3   Trait4 	Trait5    Trait6
	rs10000010 0.9122360 0.30088096 0.2332038 0.5193068 0.1255104 0.07253145 
	rs10000023 0.8642906 0.52064064 0.9243443 0.7177759 0.9512171 0.81716250 
	rs10000030 0.2832705 0.99021664 0.8359339 0.9662707 0.8491221 0.50208681 

  </pre>


<h3>Examples</h3>

<pre><code class='language-R'>#Read input demo file for "read_pvals" function
data(demo)
</code></pre>

<hr>
<h2 id='genes_permutation'>
Gene-level Permutations 
</h2><span id='topic+genes_permutation'></span>

<h3>Description</h3>

<p>Performs gene-level circular genomic permutations. In each permutation,the complete set of<br /> 
SNP association p-values are permuted by rotation with respect to the SNPs' genomic locations.<br />
Once these 'simulated' p-values are assigned,the joint gene p-values are calculated using<br />
Fisher's combination test,and pathways' association tested using the hypergeometric test<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genes_permutation(ordered_alldata = "", pers_ids = "", pathways = "", 
ntraits = "", nper = 100, threshold = 0.05, seed=10,saveto = "workspace", 
gs_locs="", envir = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genes_permutation_+3A_ordered_alldata">ordered_alldata</code></td>
<td>

<p>Return variable from &quot;genome_order&quot;. Ordered genome and trait p-values
</p>
</td></tr>
<tr><td><code id="genes_permutation_+3A_gs_locs">gs_locs</code></td>
<td>

<p>Return variable from &quot;genome_order&quot;. SNP indexes
</p>
</td></tr>
<tr><td><code id="genes_permutation_+3A_pers_ids">pers_ids</code></td>
<td>

<p>Return variable &quot;per_ors&quot; from &quot;read2_paths&quot;. Gene indexes
</p>
</td></tr>
<tr><td><code id="genes_permutation_+3A_pathways">pathways</code></td>
<td>

<p>Return variable &quot;pathways&quot; from &quot;read2_paths&quot;
</p>
</td></tr>
<tr><td><code id="genes_permutation_+3A_ntraits">ntraits</code></td>
<td>

<p>Traits INDEX to be analysed. Index according to &quot;ordered_alldata&quot;. <br />
Trait Columns index must start at 7. Example: ntraits=c(7:9), ntraits=7
</p>
</td></tr>
<tr><td><code id="genes_permutation_+3A_nper">nper</code></td>
<td>

<p>Number of permutations.Example: nper=1000
</p>
</td></tr>
<tr><td><code id="genes_permutation_+3A_threshold">threshold</code></td>
<td>

<p>Threshold to be set by the hypergeometric test. threshold=0.05
</p>
</td></tr>
<tr><td><code id="genes_permutation_+3A_seed">seed</code></td>
<td>

<p>Set a number for random sampling
</p>
</td></tr>
<tr><td><code id="genes_permutation_+3A_saveto">saveto</code></td>
<td>

<p>Save permutation results to &quot;workspace&quot; OR &quot;directory&quot;
</p>
</td></tr>
<tr><td><code id="genes_permutation_+3A_envir">envir</code></td>
<td>

<p>R environment to save the data to when saveto is set to &quot;workspace&quot;
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns &quot;Permus_trait&quot; variables or files (permutation datasets). 
</p>


<h3>References</h3>

<p>Imports phyper (from stats)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+snps_permutation">snps_permutation</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load data
data(demo,SNPsAnnotation)
all_data &lt;- read_pvals(data_name=demo,snps_ann=SNPsAnnotation)

# Prepare Genome
genome_results &lt;-genome_order(all_data=all_data)
	# Results from genome_order
	ordered_alldata &lt;- genome_results$ordered_alldata
	gs_locs &lt;- genome_results$gs_locs

# Create new environment to save data:
gper.env &lt;- new.env()

# Get pathways
data(RHSA164843,RHSA446343,RHSA8876384,RHSA8964572,RHSA109582,RHSA1474244,envir=gper.env)

# Map Genes to pathways
paths_res &lt;- read2_paths(ordered_alldata=ordered_alldata,gs_locs=gs_locs,
sets_from="workspace",sets_prefix="RHSA",level="gene",envir=gper.env)
pers_ids &lt;- paths_res$per_ors
pathways&lt;- paths_res$pathways

# Perform Permutations:
genes_permutation(ordered_alldata=ordered_alldata,
pers_ids=pers_ids,pathways=pathways,ntraits=c(7:9),
nper=10,threshold=0.05, saveto="workspace",
gs_locs=gs_locs,envir = gper.env)

# Results
results &lt;- get_results(res_pattern="Permus",level="gene",
from="workspace",threshold=0.05,envir= gper.env)
</code></pre>

<hr>
<h2 id='genome_order'>
Genome Order
</h2><span id='topic+genome_order'></span>

<h3>Description</h3>

<p>Orders the SNPs according to their genomic location
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genome_order(all_data = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genome_order_+3A_all_data">all_data</code></td>
<td>

<p>SNPs to Genes Annotation and Trait Pvalues of Association<br />
all_data = (read_pvals output) OR matrix/dataframe.<br />
</p>
</td></tr>
</table>


<h3>Details</h3>

<pre>
Input Columns with "*" must be included for analysis
 
NOTE: Trait p-values must start at Column #7

# *Column 1: "name" (SNP_IDs - any SNP ID as character)
# *Column 2: Chromosome Location
# *Column 3: SNP Location 
# *Column 4: Gene ID 
# Column 5: Symbol (OR Annotation Field 1)
# Column 6: Annotaiton Field 2
# *Column 7: First trait pvalues of association
# Column N: Next trait pvalues of association
# Example Input Data:
name       Chromosome  Location GENE_ID Symbol Orientation abpi
rs10000010          4  21618674   80333 KCNIP4           - 0.91
rs10000023          4  95733906     658 BMPR1B           + 0.86
rs10000092          4  21895517   80333 KCNIP4           - 0.20
rs1000022          13  100461219 171425 CLYBL            + 0.26
rs10000300          4  40466547   54502 RBM47            - 0.58
</pre>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>ordered_alldata</code></td>
<td>

<p>SNPs annotated to Genes and Trait p-values
</p>
</td></tr>
<tr><td><code>gs_locs</code></td>
<td>

<p>Gene annotations, location indexes and number of observations
</p>
</td></tr>
</table>


<h3>Format</h3>

<pre>	
	SNPs annotated to Genes and Trait p-values
#ordered_alldata[1:5,1:8]
name   Chromosome	Location GENE_ID  Symbol Orientation Trait1 Trait2
rs3934834	1  		1005806  NA     &lt;NA&gt;        &lt;NA&gt;  	 0.97  0.92
rs3737728	1		1021415  54991 	C1orf159      - 	 0.91  0.69
rs6687776	1		1030565  54991  C1orf159      - 	 0.71  0.45
rs9651273	1		1031540  54991  C1orf159      - 	 0.22  0.60
rs4970405	1		1048955  54991  C1orf159      - 	 0.77  0.56

	Gene annotations, location indexes and number of observations
#gs_locs[1:5,]
#      Symbol     Chromosome Location    Gene_ID Start_Indx Observations
# [1,] "A1BG"     "19"       "58864479"  "1"     "293976"   "1"         
# [2,] "A2M"      "12"       "9232268"   "2"     "215264"   "5"         
# [3,] "NAT1"     "8"        "18077310"  "9"     "151804"   "1"         
# [4,] "NAT2"     "8"        "18257280"  "10"    "151831"   "2"         
# [5,] "SERPINA3" "14"       "95080803"  "12"    "249519"   "2"         
</pre>


<h3>See Also</h3>

<p><code><a href="#topic+read2_paths">read2_paths</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## DEMO WORKSPACE 

data(demo,SNPsAnnotation)
all_data&lt;-read_pvals(data_name=demo,snps_ann=SNPsAnnotation)
# GENOME ORDER
genome_results &lt;- genome_order(all_data=all_data)
	ordered_alldata &lt;- genome_results$ordered_alldata
	gs_locs &lt;- genome_results$gs_locs
</code></pre>

<hr>
<h2 id='get_pathways'>
Pathways
</h2><span id='topic+get_pathways'></span>

<h3>Description</h3>

<p>Helper function to download pathways and their gene identifiers.
reactome.db used for pathway annotations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pathways(source="reactome",all_paths=TRUE,envir = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_pathways_+3A_source">source</code></td>
<td>

<p>&quot;reactome&quot;
</p>
</td></tr>
<tr><td><code id="get_pathways_+3A_all_paths">all_paths</code></td>
<td>

<p>TRUE or FALSE. If FALSE a subset will be asked by the function
</p>
</td></tr>
<tr><td><code id="get_pathways_+3A_envir">envir</code></td>
<td>

<p>R environment to save Pathways to
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns &quot;Pathways description&quot; All downloaded pathways are saved in the workspace
User will be prompt to set a prefix.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read2_paths">read2_paths</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# get pathways source = "reactome"
if (!require("reactome.db")) install.packages("reactome.db") 
library(reactome.db)

# Create new environment to save data:
gper.env &lt;- new.env()

paths &lt;- get_pathways(source="reactome",all_paths=FALSE,envir=gper.env)
# when prompted introduce species as listed
Homo sapiens
# when prompted introduce prefix. Avoid characters "-" and "_" (e.g mypath, or leave blank)
# if all_paths set to TRUE. All pathways are downloaded automatically
# IF all_paths set to FALSE, select a subset of pathway identifiers from 
# list. Separated by ","
R-HSA-8964572,R-HSA-9613354,R-HSA-8876384,R-HSA-446343,R-HSA-9620244

## End(Not run)
</code></pre>

<hr>
<h2 id='get_results'>
Circular Permutation Results
</h2><span id='topic+get_results'></span>

<h3>Description</h3>

<p>Creates a summary dataframe of the genomic permutations datasets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_results(res_pattern="Permus",level="snp",from="workspace",
threshold=0.05,envir = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_results_+3A_res_pattern">res_pattern</code></td>
<td>

<p>Pattern of the Permutation files/variable. eg. res=pattern=&quot;Permus&quot;
</p>
</td></tr>
<tr><td><code id="get_results_+3A_level">level</code></td>
<td>

<p>Permutation level performed.level values &quot;snp&quot; or &quot;gene&quot;
</p>
</td></tr>
<tr><td><code id="get_results_+3A_from">from</code></td>
<td>

<p>Location of the permutation datasets.from values &quot;workspace&quot; or &quot;directory&quot; 
</p>
</td></tr>
<tr><td><code id="get_results_+3A_threshold">threshold</code></td>
<td>

<p>Threshold of significance set 
</p>
</td></tr>
<tr><td><code id="get_results_+3A_envir">envir</code></td>
<td>

<p>R environment where save the data to
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>results</code></td>
<td>

<p>Data frame with Pathway ID, Trait, Threshold set by permutations,<br /> 
Gene results include the theoretical hypergeometric p-value and the,<br /> 
observed (Empirical Hypergeometric p-values)<br />
SNP results include the count of significan SNPs and the overall score<br />
Score is the proportion of tests observed with more significant results<br />
</p>
</td></tr>
</table>


<h3>Format</h3>

<pre>
## SNP level results
     PathID    Trait Threshold RealCount Score
1  hsa00010     abpi         0         0 0.037
2  hsa00010 abpildfa         0         0 0.040
3  hsa04720     abpi         2         0 0.311	
## Gene level results	
     PathID Trait   Threshold     P-Value  Observed
1  hsa00010  abpi 0.040441176 0.058823529 1.0000000
2  hsa00020  abpi 0.000000000 0.000000000 0.1666667
3  hsa00030  abpi 0.040441176 0.058823529 1.0000000
</pre>


<h3>Examples</h3>

<pre><code class='language-R'>data(demo,SNPsAnnotation)
all_data &lt;- read_pvals(data_name=demo,snps_ann=SNPsAnnotation)
genome_results &lt;-genome_order(all_data=all_data)
	# Results from genome_order
	ordered_alldata &lt;- genome_results$ordered_alldata
	gs_locs &lt;- genome_results$gs_locs

# Create new environment to save data
gper.env &lt;- new.env()

# Get pathways
data(RHSA164843,RHSA446343,RHSA8876384,RHSA8964572,RHSA109582,RHSA1474244,envir=gper.env)

paths_res &lt;- read2_paths(ordered_alldata=ordered_alldata,gs_locs=gs_locs,
sets_from="workspace",sets_prefix="RHSA",level="snp",envir=gper.env)
		pers_ids &lt;- paths_res$per_ors
		pathways&lt;- paths_res$pathways

snps_permutation(ordered_alldata=ordered_alldata,pers_ids=pers_ids,
ntraits=c(7,9),nper=10,saveto="workspace",threshold=0.05,
gs_locs=gs_locs,envir= gper.env)

results &lt;- get_results(res_pattern="Permus",level="snp",
from="workspace",threshold=0.05,envir = gper.env)
</code></pre>

<hr>
<h2 id='hyprbg'>
Hypergeometric Test (phyper)
</h2><span id='topic+hyprbg'></span>

<h3>Description</h3>

<p>Performs Hypergeometric test (phyper() from R) 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hyprbg(Sig_in_Paths, uniSig, gns_in_Paths, universe)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hyprbg_+3A_sig_in_paths">Sig_in_Paths</code></td>
<td>

<p>Number of significant genes in the pathway
</p>
</td></tr>
<tr><td><code id="hyprbg_+3A_unisig">uniSig</code></td>
<td>

<p>Number of significant genes in the dataset
</p>
</td></tr>
<tr><td><code id="hyprbg_+3A_gns_in_paths">gns_in_Paths</code></td>
<td>

<p>Number of genes in the pathway
</p>
</td></tr>
<tr><td><code id="hyprbg_+3A_universe">universe</code></td>
<td>

<p>Number of genes in the dataset
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns hypergeometric test
</p>


<h3>References</h3>

<p>hyprbg Imports phyper() (from stats)
</p>

<hr>
<h2 id='plot_results'>
Plot Results Circular Permutation
</h2><span id='topic+plot_results'></span>

<h3>Description</h3>

<p>QQ plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_results(results="",by="",plot_all=TRUE, var = "", save_plot=TRUE, plot_name="", 
bf= FALSE , save_qq = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_results_+3A_results">results</code></td>
<td>

<p>Results datarame from &quot;get_results()&quot;
</p>
</td></tr>
<tr><td><code id="plot_results_+3A_by">by</code></td>
<td>

<p>Visualize results by &quot;trait&quot; OR by &quot;set&quot;  
</p>
</td></tr>
<tr><td><code id="plot_results_+3A_plot_all">plot_all</code></td>
<td>

<p>plot_all = TRUE plots all the variables in the results dataframe and saves a pdf file in the working directory.
Setting plot_all to FALSE plots a single variable(trait or set). The argument &quot;var&quot; must be declared.
</p>
</td></tr>
<tr><td><code id="plot_results_+3A_var">var</code></td>
<td>

<p>Variable name to plot
</p>
</td></tr>
<tr><td><code id="plot_results_+3A_save_plot">save_plot</code></td>
<td>

<p>save_plot = TRUE saves the plots in the working directory. save_plot = FALSE the plot is visualized at the console. 
save_plot = FALSE can be used only when plot_all is set to FALSE. 
The plot displayed at the console is interactive, clicking on a point displays the points name.
</p>
</td></tr>
<tr><td><code id="plot_results_+3A_plot_name">plot_name</code></td>
<td>

<p>Argument used to save the file name for the plots. 
Default value = Results_genomicper_[set/trait]
</p>
</td></tr>
<tr><td><code id="plot_results_+3A_bf">bf</code></td>
<td>

<p>Displays the bonferroni correction 
</p>
</td></tr>
<tr><td><code id="plot_results_+3A_save_qq">save_qq</code></td>
<td>

<p>TRUE returns the qq plot values
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>qq</code></td>
<td>

<p>Data frame with qq plot values
</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+get_results">get_results</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(demo,SNPsAnnotation)
all_data &lt;- read_pvals(data_name=demo,snps_ann=SNPsAnnotation)
genome_results &lt;-genome_order(all_data=all_data)
	# Results from genome_order
	ordered_alldata &lt;- genome_results$ordered_alldata
	gs_locs &lt;- genome_results$gs_locs

# Create new environment to save the data:
gper.env &lt;- new.env()

# Load Pathways
data(RHSA164843,RHSA446343,RHSA8876384,RHSA8964572,RHSA109582,RHSA1474244,envir=gper.env)


paths_res &lt;- read2_paths(ordered_alldata=ordered_alldata,gs_locs=gs_locs,
sets_from="workspace",sets_prefix="RHSA",level="snp",envir=gper.env)
		pers_ids &lt;- paths_res$per_ors
		pathways&lt;- paths_res$pathways

snps_permutation(ordered_alldata=ordered_alldata,pers_ids=pers_ids,
ntraits=c(7,9),nper=10,saveto="workspace",threshold=0.05,
gs_locs=gs_locs,envir = gper.env)

results &lt;- get_results(res_pattern="Permus",level="snp",
from="workspace",threshold=0.05,envir = gper.env)

#saves plots to working directory
## Not run: 
qq &lt;- plot_results(results=results,by="set",plot_all=TRUE)
qq &lt;- plot_results(results=results,by="trait",plot_all=FALSE,var="trait1")
qq &lt;- plot_results(results=results,by="set",
	plot_all=FALSE,var="R-HSA-8964572",
	save_plot=FALSE) ## IMPORTANT: to EXIT interactive plot 
	## right click on the plot to stop

## End(Not run)
</code></pre>

<hr>
<h2 id='read_pvals'>
Read GWAS p-values of association and Merge with SNP annotations
</h2><span id='topic+read_pvals'></span>

<h3>Description</h3>

<p>Read GWAS p-values of association and Merge with SNP annotations for analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_pvals(data_name="",snps_ann="",from="workspace")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_pvals_+3A_data_name">data_name</code></td>
<td>

<p>GWAS p_values (tab delimited file)(SNP_IDs Trait1 Trait2 ...TraitN)
</p>
</td></tr>
<tr><td><code id="read_pvals_+3A_snps_ann">snps_ann</code></td>
<td>

<p>SNPs Annotation (SNPsAnnotation). Genomicper uses entrez gene ids to annotate associate SNPs-to genes-pathways<br />
The annotation MUST match your data input (coordinates and chromosome format)<br />
Any SNP ID is valid, as long the ID is set as character<br />
The examples below show an option on how to annotate the SNPs prior the use of genomicper<br />
</p>
</td></tr>
<tr><td><code id="read_pvals_+3A_from">from</code></td>
<td>

<p>Datasets location. Values &quot;workspace&quot; OR &quot;directory&quot; <br />
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe: 
name; chromosome; Location; GeneID; Symbol; Orientation; Trait1; TraitN
</p>


<h3>Formats</h3>

<pre>
GWAS p_values (tab delimited file)(SNP_IDs Trait1 Trait2 ...TraitN)
name       Trait1      Trait2    TraitN  
rs10000010 0.9122360 0.30088096 0.2332038
rs10000023 0.8642906 0.52064064 0.9243443 
rs10000030 0.2832705 0.99021664 0.8359339

SNPs Annotation (SNPsAnnotation)
name        Chromosome  Location  GENE_ID   Symbol    Orientation
rs1000313   1           15405489  23254     KIAA1026  +
rs1000533   1           168282491 9095      TBX19     +
rs1000731   1           231963491 27185     DISC1     +

Output:
name       Chromosome  Location GENE_ID Symbol Orientation   Trait1  
rs10000010          4  21618674   80333 KCNIP4           - 0.9122360
rs10000023          4  95733906     658 BMPR1B           + 0.8642906
rs10000030          4 103374154      NA   &lt;NA&gt;        &lt;NA&gt; 0.2832705 
</pre>


<h3>See Also</h3>

<p><code><a href="#topic+genome_order">genome_order</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## DEMO // WORKSPACE
data(demo,SNPsAnnotation)
all_data &lt;- read_pvals(data_name=demo,snps_ann=SNPsAnnotation)
</code></pre>

<hr>
<h2 id='read2_paths'>
Read to SNPs to sets; Map SNPs to gene-sets/pathways
</h2><span id='topic+read2_paths'></span>

<h3>Description</h3>

<p>Reads the sets/pathways, map the SNPs and genes to the gene-sets/pathways
read2_paths uses the &quot;genome_order&quot; output(ordered_alldata, gs_locs) to assign genomic location indexes
to each element in the gene-set. The permutation method must be defined (i.e. level = &quot;snp&quot; OR level = &quot;gene&quot;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read2_paths(ordered_alldata="",gs_locs="",sets_from="workspace",
sets_prefix="RHSA",level="snp",envir="")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read2_paths_+3A_ordered_alldata">ordered_alldata</code></td>
<td>

<p>Ordered data according to the SNPs genomic location. Traits start at column 7<br />
Return variable from:<br />
genome_results &lt;-genome_order(all_data=all_data)<br />
ordered_alldata &lt;- genome_results$ordered_alldata<br />
</p>
</td></tr>
<tr><td><code id="read2_paths_+3A_gs_locs">gs_locs</code></td>
<td>

<p>Gene annotation,indexes and number of observations<br />
Return variable from genome_order():<br />
genome_results &lt;-genome_order(all_data=all_data)<br />
gs_locs &lt;- genome_results$gs_locs<br />
</p>
</td></tr>
<tr><td><code id="read2_paths_+3A_sets_from">sets_from</code></td>
<td>

<p>Location of the gene-sets. Default set to &quot;workspace&quot; <br />
sets_from=&quot;workspace&quot; OR sets_from=&quot;directory&quot;<br />
&quot;directory&quot;, only will search for information in the working directory.
</p>
</td></tr>
<tr><td><code id="read2_paths_+3A_sets_prefix">sets_prefix</code></td>
<td>

<p>Prefix of the gene-set variables or files.<br />
Default set to sets_prefix= &quot;RHSA&quot; 
e.g. Variables &quot;RHSA164843&quot;,&quot;RHSA446343&quot;,&quot;RHSA8876384&quot;<br />
each variable/file contains the list of gene identifiers part of that pathway
</p>
</td></tr>
<tr><td><code id="read2_paths_+3A_level">level</code></td>
<td>

<p>The level at which the permutations will be performed. Assigns the indexes according to snps or genes<br />
Default value &quot;snp&quot; 
level values = &quot;snp&quot; OR &quot;gene&quot;
</p>
</td></tr>
<tr><td><code id="read2_paths_+3A_envir">envir</code></td>
<td>

<p>R environment where pathway data is stored. e.g(envir=.GlobalEnv, envir=gper.env)
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pathways</code></td>
<td>

<p>Pathway Id, Description,Number of Genes in the pathway, Number of genes found in the dataset,
Number of SNPs found in the dataset</p>
</td></tr>
<tr><td><code>per_ors</code></td>
<td>

<p>A list of identifiers mapped to each pathway
</p>
</td></tr>
</table>


<h3>Format</h3>

<pre>
Input: Ordered_alldata
name     Chromosome  Location GENE_ID   Symbol Orientation Trait1 Trait2
rs1001567          1 9194614     &lt;NA&gt;     &lt;NA&gt;        &lt;NA&gt; 0.96 0.89
rs1000313          1 15405489   23254 KIAA1026           + 0.93 0.57
rs1002365          1 19797248    &lt;NA&gt;     &lt;NA&gt;        &lt;NA&gt; 0.68 0.58
rs1002706          1 25051153    &lt;NA&gt;     &lt;NA&gt;        &lt;NA&gt; 0.71 0.02
rs1002487          1 26865971    6195  RPS6KA1           + 0.98 0.78

Input:gs_locs
      Symbol   Chromosome Location    Gene_ID Start_Indx Observations
 [1,] "ACYP2"  "2"        "54399633"  "98"    "35"       "1"         
 [2,] "AMPD3"  "11"       "10514707"  "272"   "898"      "1"         
 [3,] "ANK2"   "4"        "113830885" "287"   "479"      "4"
 
Input:pathway example
RHSA8964572
[1]   1149 128486 161247  29923 345275  63924
          
Output:pathways
     ID            GenesInPath GenesFound SNPsInPath
"RHSA109582"  "681"       "8"        "11"      
"RHSA1474244" "418"       "7"        "10"      
"RHSA164843"  "11"        "0"        "0"       
"RHSA446343"  "4"         "1"        "1"       
"RHSA8876384" "32"        "1"        "1"       
"RHSA8964572" "6"         "1"        "1"       

</pre>


<h3>See Also</h3>

<p><code><a href="#topic+genes_permutation">genes_permutation</a></code>
<code><a href="#topic+snps_permutation">snps_permutation</a></code>
<code><a href="#topic+genome_order">genome_order</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## DEMO - SNP Level data stored in workspace #######################
# library(genomicper)
data(demo,SNPsAnnotation)
all_data &lt;- read_pvals(data_name=demo,snps_ann=SNPsAnnotation)
genome_results &lt;-genome_order(all_data=all_data)
	ordered_alldata &lt;- genome_results$ordered_alldata
	gs_locs &lt;- genome_results$gs_locs
	
# Create new environment to save variables (e.g. pathways, permutations):
gper.env &lt;- new.env()

data(RHSA164843,RHSA446343,RHSA8876384,RHSA8964572,RHSA109582,RHSA1474244,envir=gper.env)

paths_res &lt;- read2_paths(ordered_alldata=ordered_alldata,
gs_locs=gs_locs,sets_from="workspace",sets_prefix="RHSA",
level="snp",envir=gper.env)		
	pers_ids &lt;- paths_res$per_ors
	pathways&lt;- paths_res$pathways
####################################################################
</code></pre>

<hr>
<h2 id='RHSAXXXXX'>
Reactome Pathway examples
</h2><span id='topic+RHSAXXXXX'></span><span id='topic+RHSA109582'></span><span id='topic+RHSA164843'></span><span id='topic+RHSA446343'></span><span id='topic+RHSA1474244'></span><span id='topic+RHSA8876384'></span><span id='topic+RHSA8964572'></span>

<h3>Description</h3>

<p>Each file &quot;RHSAXXXX&quot; contains the gene identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RHSA164843)</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:6] 11168 155030 155348 155459 155908   2547...
</p>


<h3>Source</h3>

<p>reactome.db
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RHSA164843)
</code></pre>

<hr>
<h2 id='snps_permutation'>
SNP-level permutations
</h2><span id='topic+snps_permutation'></span>

<h3>Description</h3>

<p>Performs SNP-level circular genomic permutations. In each permutation,<br /> 
the complete set of SNP association p-values are permuted by rotation<br />
with respect to the SNPs' genomic locations.<br />
Once these 'simulated' p-values are assigned,the proportion of SNPs per<br />
set above a pre-defined threshold is calculated
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snps_permutation(ordered_alldata = "", pers_ids = "", ntraits = "", 
nper = 100, threshold = 0.05, seed=10,saveto = "workspace", 
gs_locs = "",envir ="")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snps_permutation_+3A_ordered_alldata">ordered_alldata</code></td>
<td>

<p>Return variable from &quot;genome_order&quot;. Ordered genome and trait p-values
</p>
</td></tr>
<tr><td><code id="snps_permutation_+3A_gs_locs">gs_locs</code></td>
<td>

<p>Return variable from &quot;genome_order&quot;. SNP indexes
</p>
</td></tr>
<tr><td><code id="snps_permutation_+3A_pers_ids">pers_ids</code></td>
<td>

<p>Return variable &quot;per_ors&quot; from &quot;read2_paths&quot;. SNP indexes
</p>
</td></tr>
<tr><td><code id="snps_permutation_+3A_ntraits">ntraits</code></td>
<td>

<p>Traits INDEX to be analysed. Index according to &quot;ordered_alldata&quot;. <br />
Trait Columns index must start at 7. Example: ntraits=c(7:9), ntraits=7
</p>
</td></tr>
<tr><td><code id="snps_permutation_+3A_nper">nper</code></td>
<td>

<p>Number of permutations.Example: nper=1000
</p>
</td></tr>
<tr><td><code id="snps_permutation_+3A_threshold">threshold</code></td>
<td>

<p>Threshold to be set by the hypergeometric test. threshold=0.05
</p>
</td></tr>
<tr><td><code id="snps_permutation_+3A_seed">seed</code></td>
<td>

<p>Set number for random sampling
</p>
</td></tr>
<tr><td><code id="snps_permutation_+3A_saveto">saveto</code></td>
<td>

<p>Save permutation results to &quot;workspace&quot; OR &quot;directory&quot;
</p>
</td></tr>
<tr><td><code id="snps_permutation_+3A_envir">envir</code></td>
<td>

<p>R environment to save the Permutations to when saveto is set to &quot;workspace&quot;
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns &quot;Permus_genesetsname&quot; variables or files (permutation datasets). 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+genes_permutation">genes_permutation</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(demo,SNPsAnnotation)
all_data &lt;- read_pvals(data_name=demo,snps_ann=SNPsAnnotation)
genome_results &lt;-genome_order(all_data=all_data)
	# Results from genome_order
	ordered_alldata &lt;- genome_results$ordered_alldata
	gs_locs &lt;- genome_results$gs_locs

# Create new environment to save the permutations to:
gper.env &lt;- new.env()

data(RHSA164843,RHSA446343,RHSA8876384,RHSA8964572,RHSA109582,RHSA1474244,envir=gper.env)

paths_res &lt;- read2_paths(ordered_alldata=ordered_alldata,gs_locs=gs_locs,
sets_from="workspace",sets_prefix="RHSA",level="snp",envir=gper.env)
		pers_ids &lt;- paths_res$per_ors
		pathways&lt;- paths_res$pathways

		
# SNP permutations
snps_permutation(ordered_alldata=ordered_alldata,pers_ids=pers_ids,
ntraits=c(7,9),nper=10,saveto="workspace",threshold=0.05,
gs_locs=gs_locs,envir = gper.env)

# Get results						
results &lt;- get_results(res_pattern="Permus",level="snp",
from="workspace",threshold=0.05,envir = gper.env)

</code></pre>

<hr>
<h2 id='SNPsAnnotation'>
SNPs-Genes annotation to Distance 0 (SNPs within a gene)
</h2><span id='topic+SNPsAnnotation'></span>

<h3>Description</h3>

<p>SNPs annotated to genes. Annotation only when the SNPs fall within start and end of transcription of the genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SNPsAnnotation)</code></pre>


<h3>Format</h3>

<p>Sample data frame with 339096 SNP observations on the following 6 variables.
</p>

<dl>
<dt><code>name</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Chromosome</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Location</code></dt><dd><p>a numeric vector of the SNP location</p>
</dd>
<dt><code>GENE_ID</code></dt><dd><p>a numeric vector with entrez geneID</p>
</dd>
<dt><code>Symbol</code></dt><dd><p>a character vector ; other annotation slot 1</p>
</dd>
<dt><code>Orientation</code></dt><dd><p>a character vector; other annotation slot 2</p>
</dd>
</dl>

<pre>
	name      Chromosome  Location  GENE_ID    Symbol     Orientation
	rs1000313          1  15405489    23254    KIAA1026             +
	rs1000533          1 168282491     9095    TBX19                +
	rs1000731          1 231963491    27185    DISC1                +
	</pre>


<h3>Source</h3>

<p>NCBI Gene database,(http://www.ncbi.nlm.nih.gov/gene ; Build.37.1).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPsAnnotation)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
