<!DOCTYPE html><html lang="en"><head><title>Help for package GenomicSig</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GenomicSig}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AminoAcidContent'>
<p>Amino Acid Content</p></a></li>
<li><a href='#AMIP'>
<p>Average Mutual Information Profile (AMIP)</p></a></li>
<li><a href='#CGR'>
<p>Chaos Game Representation</p></a></li>
<li><a href='#GC_content'>
<p>GC content of nucleic acid sequences</p></a></li>
<li><a href='#Genomicdata'>
<p>Nucleotide sequence Data</p></a></li>
<li><a href='#OFDEG'>
<p>Oligonucleotide Frequency Derived Error Gradient</p></a></li>
<li><a href='#RSCU'>
<p>Relative Synonymous Codon Usage</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Computation of Genomic Signatures</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Anu Sharma &lt;anu.sharma@icar.gov.in&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Genomic signatures represent unique features within a species' DNA, enabling the differentiation of species and offering broad applications across various fields. This package provides essential tools for calculating these specific signatures, streamlining the process for researchers and offering a comprehensive and time-saving solution for genomic analysis.The amino acid contents are identified based on the work published by Sandberg et al. (2003) &lt;<a href="https://doi.org/10.1016%2Fs0378-1119%2803%2900581-x">doi:10.1016/s0378-1119(03)00581-x</a>&gt; and Xiao et al. (2015) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtv042">doi:10.1093/bioinformatics/btv042</a>&gt;. The Average Mutual Information Profiles (AMIP) values are calculated based on the work of Bauer et al. (2008) &lt;<a href="https://doi.org/10.1186%2F1471-2105-9-48">doi:10.1186/1471-2105-9-48</a>&gt;. The Chaos Game Representation (CGR) plot visualization was done based on the work of Deschavanne et al. (1999) &lt;<a href="https://doi.org/10.1093%2Foxfordjournals.molbev.a026048">doi:10.1093/oxfordjournals.molbev.a026048</a>&gt; and Jeffrey et al. (1990) &lt;<a href="https://doi.org/10.1093%2Fnar%2F18.8.2163">doi:10.1093/nar/18.8.2163</a>&gt;. The GC content is calculated based on the work published by Nakabachi et al. (2006) &lt;<a href="https://doi.org/10.1126%2Fscience.1134196">doi:10.1126/science.1134196</a>&gt; and Barbu et al. (1956) <a href="https://pubmed.ncbi.nlm.nih.gov/13363015">https://pubmed.ncbi.nlm.nih.gov/13363015</a>. The Oligonucleotide Frequency Derived Error Gradient (OFDEG) values are computed based on the work published by Saeed et al. (2009) &lt;<a href="https://doi.org/10.1186%2F1471-2164-10-S3-S10">doi:10.1186/1471-2164-10-S3-S10</a>&gt;. The Relative Synonymous Codon Usage (RSCU) values are calculated based on the work published by Elek (2018) <a href="https://urn.nsk.hr/urn:nbn:hr:217:686131">https://urn.nsk.hr/urn:nbn:hr:217:686131</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>kaos, Biostrings, stats, entropy, seqinr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-09 07:40:03 UTC; kabil</td>
</tr>
<tr>
<td>Author:</td>
<td>Mailarlinga [aut],
  Shashi Bhushan Lal [aut],
  Anu Sharma [aut, cre],
  Dwijesh Chandra Mishra [aut],
  Sudhir Srivastava [aut],
  Sanjeev Kumar [aut],
  Girish Kumar Jha [aut],
  Sayanti Guha Majumdar [aut],
  Megha Garg [aut],
  Sharanbasappa [ctb],
  Kabilan S [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-11 16:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='AminoAcidContent'>
Amino Acid Content
</h2><span id='topic+AminoAcidContent'></span>

<h3>Description</h3>

<p>Amino acid content refers to the relative frequencies of amino acids used in a protein or a proteome with 20 different amino acids as 20 dimensional vectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AminoAcidContent(fasta_file, type = c("DNA", "protein"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AminoAcidContent_+3A_fasta_file">fasta_file</code></td>
<td>

<p>Path of a fasta file containing nucleotide or protein sequence.
</p>
</td></tr>
<tr><td><code id="AminoAcidContent_+3A_type">type</code></td>
<td>

<p>Type of the sequence, can be either &quot;DNA&quot; or &quot;protein&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Amino acid content refers to the relative frequencies of amino acids in the protein.If DNA sequence is given as input, it will be translated to a protein sequence. Then amino acid content will be calculated.
</p>


<h3>Value</h3>

<p>This function returns a data frame containing the sequence identifier fetched from the input fasta file and amino acid content of that sequence.
</p>


<h3>Author(s)</h3>

<p>Dr. Anu Sharma,
Megha Garg
</p>


<h3>References</h3>

<p>Sandberg, R., Bränden, C. I., Ernberg, I., &amp; Cöster, J. (2003). Quantifying the species-specificity in genomic signatures, synonymous codon choice, amino acid usage and G+ C content. Gene, 311, 35-42.
</p>
<p>Xiao, N., Cao, D. S., Zhu, M. F., &amp; Xu, Q. S. (2015). protr/ProtrWeb: R package and web server for generating various numerical representation schemes of protein sequences. Bioinformatics, 31(11), 1857-1859.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GenomicSig)
AminoAcidContent(fasta_file= system.file("extdata/Nuc_sequence.fasta", package = "GenomicSig"),
type = "DNA")
AminoAcidContent(fasta_file= system.file("extdata/prot_sequence.fasta", package = "GenomicSig"),
type = "protein")
</code></pre>

<hr>
<h2 id='AMIP'>
Average Mutual Information Profile (AMIP)
</h2><span id='topic+AMIP'></span>

<h3>Description</h3>

<p>The Average Mutual Information Profile (AMIP) detects long-range correlations in a given DNA sequence by estimating the shared information between nucleotides situated k bases apart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AMIP(fasta_file, n1 = 1, n2 = 4)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AMIP_+3A_fasta_file">fasta_file</code></td>
<td>

<p>Path to the input FASTA file containing the DNA sequence.
</p>
</td></tr>
<tr><td><code id="AMIP_+3A_n1">n1</code></td>
<td>

<p>The starting position (in bases) for Mutual Information calculation.
</p>
</td></tr>
<tr><td><code id="AMIP_+3A_n2">n2</code></td>
<td>

<p>The end position (in bases) for Mutual Information calculation.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Average Mutual Information (AMI) provides a statistical estimate of the shared information between nucleotides situated k bases apart in the DNA sequence, where k ranges from <code>n1</code> to <code>n2</code>. This method helps identify potential patterns or correlations in the nucleotide arrangement.
</p>


<h3>Value</h3>

<p>This function returns a data frame containing the mutual information values for the specified nucleotide positions.
</p>


<h3>Author(s)</h3>

<p>Dr. Anu Sharma,
Dr. Shashi Bhushan Lal
</p>


<h3>References</h3>

<p>Bauer, M., Schuster, S. M., &amp; Sayood, K. (2008). The average mutual information profile as a genomic signature. BMC Bioinformatics, 9(1), 48.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GenomicSig)
AMIP(fasta_file = system.file("extdata/Nuc_sequence.fasta", package = "GenomicSig"), n1 = 1, n2 = 4)
</code></pre>

<hr>
<h2 id='CGR'>
Chaos Game Representation
</h2><span id='topic+CGR'></span>

<h3>Description</h3>

<p>Chaos game representation is the depiction of sequence in graphical form. It converts long single dimensional sequence (in this case genetic sequence) into graphical form.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CGR(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CGR_+3A_data">data</code></td>
<td>

<p>Input as a nucleic acid sequence of characters from fasta file.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function produces visual image of DNA sequence different from the usual linear ordering of nucleotides.
</p>


<h3>Value</h3>

<p>This function produces a chaos game representation (CGR) plot and a data frame.
</p>


<h3>Author(s)</h3>

<p>Dr. Anu Sharma,
Dr. Dwijesh Chandra Mishra
</p>


<h3>References</h3>

<p>Deschavanne, P. J., Giron, A., Vilain, J., Fagot, G., &amp; Fertil, B. (1999). Genomic signature: characterization and classification of species assessed by chaos game representation of sequences. Molecular biology and evolution, 16(10), 1391-1399.
</p>
<p>Jeffrey, H. J. (1990). Chaos game representation of gene structure. Nucleic acids research, 18(8), 2163-2170.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GenomicSig)
data(Genomicdata)
CGR(Genomicdata)
</code></pre>

<hr>
<h2 id='GC_content'>
GC content of nucleic acid sequences
</h2><span id='topic+GC_content'></span>

<h3>Description</h3>

<p>This function calculates the percentage of guanine (G) or cytosine (C) nitrogenous bases in a DNA or RNA molecule. This measure indicates the proportion of G and C bases out of an implied four total bases, which including adenine (A) and thymine (T) in DNA. And adenine and uracil (U) in RNA along with G and C.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC_content(sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GC_content_+3A_sequence">sequence</code></td>
<td>

<p>Input as a nucleic acid sequence of characters from fasta file.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>G+C content is estimated with ambiguous bases taken into account.
</p>


<h3>Value</h3>

<p>This function returns the fraction of G+C as a numeric vector of length one for all sequences.
</p>


<h3>Author(s)</h3>

<p>Dr. Anu Sharma,
Dr. Girish Kumar Jha
</p>


<h3>References</h3>

<p>Nakabachi, A., Yamashita, A., Toh, H., Ishikawa, H., Dunbar, H. E., Moran, N. A., &amp; Hattori, M. (2006). The 160-kilobase genome of the bacterial endosymbiont Carsonella. Science, 314(5797), 267-267.
</p>
<p>Barbu, E., Lee, K. Y., &amp; Wahl, R. (1956, August). Content of purine and pyrimidine base in desoxyribonucleic acid of bacteria. In Annales de l'Institut Pasteur (Vol. 91, No. 2, p. 212).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GenomicSig)
GC_content(sequence = system.file("extdata/Nuc_sequence.fasta", package = "GenomicSig"))
</code></pre>

<hr>
<h2 id='Genomicdata'>
Nucleotide sequence Data
</h2><span id='topic+Genomicdata'></span>

<h3>Description</h3>

<p>A multifasta file of 34 sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Genomicdata")</code></pre>


<h3>Details</h3>

<p>Dataset contains 34 nucleic acid sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Genomicdata)

</code></pre>

<hr>
<h2 id='OFDEG'>
Oligonucleotide Frequency Derived Error Gradient
</h2><span id='topic+OFDEG'></span>

<h3>Description</h3>

<p>Oligonucleotide Frequency Derived Error Gradient computes approximate convergence rate of oligonucleotide frequencies with subsequent increasing sequence length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OFDEG(sequence, c, rc, d, m, t, k, norm=0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="OFDEG_+3A_sequence">sequence</code></td>
<td>

<p>Input is a fasta file nucleic acid sequence.It accepts RData object of the fasta file
</p>
</td></tr>
<tr><td><code id="OFDEG_+3A_c">c</code></td>
<td>

<p>Minimum sequence cutoff c (which corresponds to the length of the shortest sequence in the
data set). Default is 160.
</p>
</td></tr>
<tr><td><code id="OFDEG_+3A_rc">rc</code></td>
<td>

<p>Cutoff of Resampling Depth (Number of subsequence of cutoff length). Default is set to 10.
</p>
</td></tr>
<tr><td><code id="OFDEG_+3A_d">d</code></td>
<td>

<p>Sampling depth (The sampling depth refers to the number of equal length sub-sequences randomly selected from the
entire sequence). Default is set to 10. Larger sequence lengths will require greater sampling depths.
</p>
</td></tr>
<tr><td><code id="OFDEG_+3A_m">m</code></td>
<td>

<p>Word size which is initial subsequence length. Default is set to 100.
</p>
</td></tr>
<tr><td><code id="OFDEG_+3A_t">t</code></td>
<td>

<p>Step size (The step size is the change in sub-sequence length from one sampling instance to the next). Default is set to 6.
</p>
</td></tr>
<tr><td><code id="OFDEG_+3A_k">k</code></td>
<td>

<p>Size of the oligonucleotide (e.g.for tetranucleotide it is 4,for hexanucleotide it is 6 ). Default is set to 1.
</p>
</td></tr>
<tr><td><code id="OFDEG_+3A_norm">norm</code></td>
<td>

<p>normalization of oligonucleotide frequency (OF) Profile (0 - no normalization, 1 - normalize the OF profile). Default is set to norm = 0.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Oligonucleotide Frequency Derived Error Gradient (OFDEG) attempts to capture the convergence behavior by subsampling the genomic fragment and measuring the decrease in error as the length of the subsamples increases upto the fragment lenth.OFDEG, derived from the oligonucleotide frequency profile of a DNA sequence shows that it is possible to obtain a meaningful phylogenetic signal for relatively short DNA sequences.
</p>


<h3>Value</h3>

<p>This function returns a data frame containing error gradients of each nucleotide sequence.
</p>


<h3>Author(s)</h3>

<p>Dr. Anu Sharma,
Dr. Sanjeev Kumar
</p>


<h3>References</h3>

<p>Saeed, I., Halgamuge, S.K. The oligonucleotide frequency derived error gradient and its application to the binning of metagenome fragments. BMC Genomics 10, S10 (2009).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GenomicSig)

OFDEG(sequence= system.file("extdata/Nuc_sequence.fasta", package = "GenomicSig")[1], c=60, rc=10
, d=10, m=50, t=6, k=1, norm=0)

</code></pre>

<hr>
<h2 id='RSCU'>
Relative Synonymous Codon Usage
</h2><span id='topic+RSCU'></span>

<h3>Description</h3>

<p>This function computes the relative frequency of each codon coding for an amino acid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RSCU(sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RSCU_+3A_sequence">sequence</code></td>
<td>

<p>Input as a nucleic acid sequence of characters from fasta file.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>RSCU values are the number of times a particular codon is observed, relative to the number of times that the codon would be observed for a uniform synonymous codon usage.
</p>


<h3>Value</h3>

<p>RSCU returns the data frame with all indices.
</p>


<h3>Author(s)</h3>

<p>Dr. Anu Sharma,
Dr. Sudhir Srivastava
</p>


<h3>References</h3>

<p>Elek, A. (2018). coRdon: an R package for codon usage analysis and prediction of gene expressivity (Master's thesis, University of Zagreb. Faculty of Science. Department of Biology).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GenomicSig)
RSCU(sequence= system.file("extdata/Nuc_sequence.fasta", package = "GenomicSig"))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
