<!DOCTYPE html><html><head><title>Help for package GenomicTools.fileHandler</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GenomicTools.fileHandler}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GenomicTools.fileHandler-package'><p>R Package To Handle Files From Genomic Data</p>
GenomicTools.fileHandler is a loose collection of I/O Functions Needed in Genomic Data Analysis</a></li>
<li><a href='#example.bed'><p>Example Gene Annotation in Bed-Format</p></a></li>
<li><a href='#example.fasta'><p>Example Sequencing Reads in fasta-Format</p></a></li>
<li><a href='#example.fastq'><p>Example Sequencing Reads in fastq-Format</p></a></li>
<li><a href='#example.gff'><p>Example Gene Annotation in gff-Format</p></a></li>
<li><a href='#example.gtf'><p>Example Gene Annotation in gtf-Format</p></a></li>
<li><a href='#example.ped'><p>Example Variant data in ped/map-Format</p></a></li>
<li><a href='#example.vcf'><p>Example Variant data in vcf-Format</p></a></li>
<li><a href='#example2.gtf.gz'><p>Example Gene Annotation in zipped gtf-Format</p></a></li>
<li><a href='#exportBed'><p>Exporting a Bed File.</p></a></li>
<li><a href='#exportFA'><p>Exporting a Fasta File.</p></a></li>
<li><a href='#importBed'><p>Importing a Bed File.</p></a></li>
<li><a href='#importBlastTab'><p>Import a Tab Delimited Blast Output File</p></a></li>
<li><a href='#importFA'><p>Importing a Fasta File.</p></a></li>
<li><a href='#importFeatureCounts'><p>Import from FeatureCounts</p></a></li>
<li><a href='#importFQ'><p>Importing a Fastq File.</p></a></li>
<li><a href='#importGFF'><p>importGFF</p></a></li>
<li><a href='#importGFF3'><p>importGFF3</p></a></li>
<li><a href='#importGTF'><p>Import a GTF File</p></a></li>
<li><a href='#importPED'><p>importPED</p></a></li>
<li><a href='#importSTARLog'><p>importSTARLog</p></a></li>
<li><a href='#importVCF'><p>importVCF</p></a></li>
<li><a href='#importXML'><p>importXML</p></a></li>
<li><a href='#plotTotalReads'><p>plotTotalReads</p></a></li>
<li><a href='#plotUniquelyMappedReads'><p>plotUniquelyMappedReads</p></a></li>
<li><a href='#prereadGTF'><p>prereadGTF</p></a></li>
<li><a href='#print.bed'><p>Print a bed Object</p></a></li>
<li><a href='#print.fa'><p>Print a fa Object</p></a></li>
<li><a href='#print.featureCounts'><p>Print a featureCounts Object</p></a></li>
<li><a href='#print.fq'><p>Print a fq Object</p></a></li>
<li><a href='#print.gtf'><p>Print a gtf Object</p></a></li>
<li><a href='#print.pedMap'><p>Print a pedMap Object</p></a></li>
<li><a href='#print.vcf'><p>Print a vcf Object</p></a></li>
<li><a href='#summary.bed'><p>Summary of a bed Object</p></a></li>
<li><a href='#summary.fa'><p>Summary of a fa Object</p></a></li>
<li><a href='#summary.featureCounts'><p>Summary of a featureCounts Object</p></a></li>
<li><a href='#summary.fq'><p>Summary of a fq Object</p></a></li>
<li><a href='#summary.gtf'><p>Summary of a gtf Object</p></a></li>
<li><a href='#summary.STARLog'><p>Summary of a STARLog Object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>File Handlers for Genomic Data Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.5.9</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-03-05</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel Fischer</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel Fischer &lt;daniel.fischer@luke.fi&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of I/O tools for handling the most commonly used genomic datafiles, like fasta/-q, bed, gff, gtf, ped/map and vcf.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3), data.table (&ge; 1.9.6)</td>
</tr>
<tr>
<td>Imports:</td>
<td>snpStats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-03-05 07:25:36 UTC; ejo138</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-03-05 08:00:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='GenomicTools.fileHandler-package'>R Package To Handle Files From Genomic Data
GenomicTools.fileHandler is a loose collection of I/O Functions Needed in Genomic Data Analysis</h2><span id='topic+GenomicTools.fileHandler-package'></span>

<h3>Description</h3>


<table>
<tr>
 <td style="text-align: left;"> Package: </td><td style="text-align: left;"> GenomicTools.fileHandler</td>
</tr>
<tr>
 <td style="text-align: left;"> Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;"> Version: </td><td style="text-align: left;">
0.1.5.9</td>
</tr>
<tr>
 <td style="text-align: left;"> Date: </td><td style="text-align: left;"> 2020-03-05</td>
</tr>
<tr>
 <td style="text-align: left;"> License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;"> LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Daniel Fischer
</p>
<p>Maintainer: Daniel Fischer &lt;daniel.fischer@luke.fi&gt;
</p>

<hr>
<h2 id='example.bed'>Example Gene Annotation in Bed-Format</h2><span id='topic+example.bed'></span>

<h3>Description</h3>

<p>This file contains some example lines to represent a typical bed file that can be used
to try the corresponding functions.
</p>


<h3>Format</h3>

<p>A file with three column Chr, Start and End.</p>


<h3>Details</h3>

<p>The file is locate din the /extdata folder of the package and is accessible after installation via
<code>system.file("extdata","example.bed", package="GenomicTools.fileHandler")</code>
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='example.fasta'>Example Sequencing Reads in fasta-Format</h2><span id='topic+example.fasta'></span>

<h3>Description</h3>

<p>This file contains some example reads to represent a typical fasta file that can be used
to try the corresponding functions.
</p>


<h3>Details</h3>

<p>The file is locate din the /extdata folder of the package and is accessible after installation via
<code>system.file("extdata","example.fasta", package="GenomicTools.fileHandler")</code>
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='example.fastq'>Example Sequencing Reads in fastq-Format</h2><span id='topic+example.fastq'></span>

<h3>Description</h3>

<p>This file contains some example reads to represent a typical fastq file that can be used
to try the corresponding functions.
</p>


<h3>Details</h3>

<p>The file is locate din the /extdata folder of the package and is accessible after installation via
<code>system.file("extdata","example.fastq", package="GenomicTools.fileHandler")</code>
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='example.gff'>Example Gene Annotation in gff-Format</h2><span id='topic+example.gff'></span>

<h3>Description</h3>

<p>This file contains some example gene annotations to represent a typical gff file that can be used
to try the corresponding functions.
</p>


<h3>Details</h3>

<p>The file is locate din the /extdata folder of the package and is accessible after installation via
<code>system.file("extdata","example.gff", package="GenomicTools.fileHandler")</code>
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='example.gtf'>Example Gene Annotation in gtf-Format</h2><span id='topic+example.gtf'></span>

<h3>Description</h3>

<p>This file contains some example gene annotations to represent a typical gtf file that can be used
to try the corresponding functions.
</p>


<h3>Details</h3>

<p>The file is locate din the /extdata folder of the package and is accessible after installation via
<code>system.file("extdata","example.gtf", package="GenomicTools.fileHandler")</code>
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='example.ped'>Example Variant data in ped/map-Format</h2><span id='topic+example.ped'></span>

<h3>Description</h3>

<p>This file contains some example variants to represent a typical ped/map file pair that can be used
to try the corresponding functions.
</p>


<h3>Details</h3>

<p>The file is locate din the /extdata folder of the package and is accessible after installation via
<code>system.file("extdata","example.ped", package="GenomicTools.fileHandler")</code>
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='example.vcf'>Example Variant data in vcf-Format</h2><span id='topic+example.vcf'></span>

<h3>Description</h3>

<p>This file contains some example variants to represent a typical vcf file that can be used
to try the corresponding functions.
</p>


<h3>Details</h3>

<p>The file is locate din the /extdata folder of the package and is accessible after installation via
<code>system.file("extdata","example.vcf", package="GenomicTools.fileHandler")</code>
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='example2.gtf.gz'>Example Gene Annotation in zipped gtf-Format</h2><span id='topic+example2.gtf.gz'></span>

<h3>Description</h3>

<p>This file contains some example gene annotations to represent a typical zipped gtf file that can be used
to try the corresponding functions.
</p>


<h3>Details</h3>

<p>The file is locate din the /extdata folder of the package and is accessible after installation via
<code>system.file("extdata","example2.gtf.gz", package="GenomicTools.fileHandler")</code>
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='exportBed'>Exporting a Bed File.</h2><span id='topic+exportBed'></span>

<h3>Description</h3>

<p>This function exports a standard bed file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exportBed(x, file = NULL, header = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exportBed_+3A_x">x</code></td>
<td>
<p>data.frame</p>
</td></tr>
<tr><td><code id="exportBed_+3A_file">file</code></td>
<td>
<p>Character, specifies filename/path</p>
</td></tr>
<tr><td><code id="exportBed_+3A_header">header</code></td>
<td>
<p>Logical, shall a header be written</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function exports a data.frame to a standard bed file. If no file name is given, the variable name will be used instead.
</p>


<h3>Value</h3>

<p>A bed file
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>novelBed &lt;- data.frame(Chr=c(11,18,3),
                      Start=c(72554673, 62550696, 18148822),
                      End=c(72555273, 62551296, 18149422),
                      Gene=c("LOC1", "LOC2", "LOC3"))

# Create a temporary file to where the output of the function is stored
myfile &lt;- file.path(tempdir(), "myLocs.bed")

exportBed(novelBed, file=myfile)
exportBed(novelBed, file=myfile, header=TRUE)

</code></pre>

<hr>
<h2 id='exportFA'>Exporting a Fasta File.</h2><span id='topic+exportFA'></span>

<h3>Description</h3>

<p>This function exports a standard fasta file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exportFA(fa, file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exportFA_+3A_fa">fa</code></td>
<td>
<p>fasta object</p>
</td></tr>
<tr><td><code id="exportFA_+3A_file">file</code></td>
<td>
<p>Character, specifies filename/path</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function exports a fasta object to a standard fasta file. If no file name is given, the variable name will be used instead.
</p>


<h3>Value</h3>

<p>A fasta file
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # Define here the location on HDD for the example file
   fpath &lt;- system.file("extdata","example.fasta", package="GenomicTools.fileHandler")
 # Import the example fasta file  
   fastaFile &lt;- importFA(file=fpath)
   newFasta &lt;- fastaFile[1:5]
   
myfile &lt;- file.path(tempdir(), "myLocs.fa")
   
  exportFA(newFasta, file=myfile)


</code></pre>

<hr>
<h2 id='importBed'>Importing a Bed File.</h2><span id='topic+importBed'></span>

<h3>Description</h3>

<p>This function imports a standard bed file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importBed(file, header = FALSE, sep = "\t")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importBed_+3A_file">file</code></td>
<td>
<p>Specifies the filename/path</p>
</td></tr>
<tr><td><code id="importBed_+3A_header">header</code></td>
<td>
<p>Logical, is a header present</p>
</td></tr>
<tr><td><code id="importBed_+3A_sep">sep</code></td>
<td>
<p>Column separator</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function imports a standard bed-file into a data.frame. It is basically a convenience wrapper around <code>read.table</code>. However,
if no header lines is given, this function automatically assigns the column names, as they are given in the bed-specification on the
Ensembl page here: https://www.ensembl.org/info/website/upload/bed.html
</p>


<h3>Value</h3>

<p>A <code>data.frame</code>
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>See Also</h3>

<p>[exportBed], [read.table]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   
 # Define here the location on HDD for the example file
   fpath &lt;- system.file("extdata","example.bed", package="GenomicTools.fileHandler")
 # Import the example bed file  
   bedFile &lt;- importBed(file=fpath)
 
</code></pre>

<hr>
<h2 id='importBlastTab'>Import a Tab Delimited Blast Output File</h2><span id='topic+importBlastTab'></span>

<h3>Description</h3>

<p>This function imports a tab delimited blast output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importBlastTab(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importBlastTab_+3A_file">file</code></td>
<td>
<p>Filename</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function imports a tab delimited blast output file, currently the same as <code>read.table</code>
</p>


<h3>Value</h3>

<p>A data.frame
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='importFA'>Importing a Fasta File.</h2><span id='topic+importFA'></span>

<h3>Description</h3>

<p>This function imports a standard fasta file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importFA(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importFA_+3A_file">file</code></td>
<td>
<p>Specifies the filename/path</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function imports a standard fasta file. Hereby, it does not matter if the identifier and sequence are alternating or not,
as the rows starting with '&gt;' are used as identifer.
</p>
<p>The example file was downloaded from here and was then further truncated respective transformed to fasta format:
</p>
<p>ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/data/HG00096/sequence_read/
</p>


<h3>Value</h3>

<p>An object of class <code>fa</code> containing the sequences. The names correspond to the sequence names given in the fasta file.
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>See Also</h3>

<p>print.fa, summary.fa
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # Define here the location on HDD for the example file
   fpath &lt;- system.file("extdata","example.fasta", package="GenomicTools.fileHandler")
 # Import the example fasta file  
   fastaFile &lt;- importFA(file=fpath)
 
</code></pre>

<hr>
<h2 id='importFeatureCounts'>Import from FeatureCounts</h2><span id='topic+importFeatureCounts'></span>

<h3>Description</h3>

<p>This functions imports the output from FeatureCounts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importFeatureCounts(file, skip = 0, headerLine = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importFeatureCounts_+3A_file">file</code></td>
<td>
<p>Character, file name</p>
</td></tr>
<tr><td><code id="importFeatureCounts_+3A_skip">skip</code></td>
<td>
<p>Number of lines to skip from txt file</p>
</td></tr>
<tr><td><code id="importFeatureCounts_+3A_headerline">headerLine</code></td>
<td>
<p>Linenumber that contains the header information</p>
</td></tr>
</table>


<h3>Details</h3>

<p>FeatureCounts produces two files, the txt that contain the expression values and then the summary that containts
all the information about the mapping statistics. This function imports both and stores them in a corresponding list.
</p>


<h3>Value</h3>

<p>A list with expValues, geneInfo and summary
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # Define here the location on HDD for the example file
   fpath &lt;- system.file("extdata","featureCountsExample.txt", package="GenomicTools.fileHandler")
 # Import the example featureCounts file  
   fcFile &lt;- importFeatureCounts(file=fpath)
 
</code></pre>

<hr>
<h2 id='importFQ'>Importing a Fastq File.</h2><span id='topic+importFQ'></span>

<h3>Description</h3>

<p>This function imports a standard fastq file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importFQ(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importFQ_+3A_file">file</code></td>
<td>
<p>Specifies the filename/path</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function imports a standard fastq file that consists out of blocks of four lines per entry
</p>


<h3>Value</h3>

<p>An object of class <code>fq</code> containing the sequences and the quality meausure. The names correspond to the sequence names given in the fasta file.
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>See Also</h3>

<p>print.fq, summary.fq
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # Define here the location on HDD for the example file
   fpath &lt;- system.file("extdata","example.fastq", package="GenomicTools.fileHandler")
 # Import the example fastq file  
   fastqFile &lt;- importFQ(file=fpath)
 
</code></pre>

<hr>
<h2 id='importGFF'>importGFF</h2><span id='topic+importGFF'></span>

<h3>Description</h3>

<p>Import a GFF file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importGFF(file, skip = "auto", nrow = -1, use.data.table = TRUE,
  level = "gene", features = NULL, num.features = c("FPKM", "TPM"),
  print.features = FALSE, merge.feature = NULL, merge.all = TRUE,
  class.names = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importGFF_+3A_file">file</code></td>
<td>
<p>file or folder</p>
</td></tr>
<tr><td><code id="importGFF_+3A_skip">skip</code></td>
<td>
<p>numeric, lines to skip</p>
</td></tr>
<tr><td><code id="importGFF_+3A_nrow">nrow</code></td>
<td>
<p>numeric, lines to read</p>
</td></tr>
<tr><td><code id="importGFF_+3A_use.data.table">use.data.table</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="importGFF_+3A_level">level</code></td>
<td>
<p>Character, read level, default: &quot;gene&quot;</p>
</td></tr>
<tr><td><code id="importGFF_+3A_features">features</code></td>
<td>
<p>features to import</p>
</td></tr>
<tr><td><code id="importGFF_+3A_num.features">num.features</code></td>
<td>
<p>names of the numeric features</p>
</td></tr>
<tr><td><code id="importGFF_+3A_print.features">print.features</code></td>
<td>
<p>Logical, print available features</p>
</td></tr>
<tr><td><code id="importGFF_+3A_merge.feature">merge.feature</code></td>
<td>
<p>Character, merge multiple samples to dataset</p>
</td></tr>
<tr><td><code id="importGFF_+3A_merge.all">merge.all</code></td>
<td>
<p>Logical, shall all samples be merged together</p>
</td></tr>
<tr><td><code id="importGFF_+3A_class.names">class.names</code></td>
<td>
<p>Definition of class name sin V9</p>
</td></tr>
<tr><td><code id="importGFF_+3A_verbose">verbose</code></td>
<td>
<p>Logical, verbose function output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function imports a standard gff file.
</p>


<h3>Value</h3>

<p>A gff object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # Define here the location on HDD for the example file
   fpath &lt;- system.file("extdata","example.gff", package="GenomicTools.fileHandler")
 # Import the example gff file  
   importGFF(fpath)
   
</code></pre>

<hr>
<h2 id='importGFF3'>importGFF3</h2><span id='topic+importGFF3'></span>

<h3>Description</h3>

<p>Import a GFF3 file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importGFF3(gff, chromosomes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importGFF3_+3A_gff">gff</code></td>
<td>
<p>file or folder</p>
</td></tr>
<tr><td><code id="importGFF3_+3A_chromosomes">chromosomes</code></td>
<td>
<p>The chromosome to import</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function imports a standard gff3 file.
</p>


<h3>Value</h3>

<p>A gff object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='importGTF'>Import a GTF File</h2><span id='topic+importGTF'></span>

<h3>Description</h3>

<p>This function imports a gtf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importGTF(file, skip = "auto", nrow = -1, use.data.table = TRUE,
  level = "gene", features = NULL, num.features = c("FPKM", "TPM"),
  print.features = FALSE, merge.feature = NULL, merge.all = TRUE,
  class.names = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importGTF_+3A_file">file</code></td>
<td>
<p>file or folder</p>
</td></tr>
<tr><td><code id="importGTF_+3A_skip">skip</code></td>
<td>
<p>numeric, lines to skip</p>
</td></tr>
<tr><td><code id="importGTF_+3A_nrow">nrow</code></td>
<td>
<p>numeric, lines to read</p>
</td></tr>
<tr><td><code id="importGTF_+3A_use.data.table">use.data.table</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="importGTF_+3A_level">level</code></td>
<td>
<p>Character, read level, default: &quot;gene&quot;</p>
</td></tr>
<tr><td><code id="importGTF_+3A_features">features</code></td>
<td>
<p>features to import</p>
</td></tr>
<tr><td><code id="importGTF_+3A_num.features">num.features</code></td>
<td>
<p>names of the numeric features</p>
</td></tr>
<tr><td><code id="importGTF_+3A_print.features">print.features</code></td>
<td>
<p>Logical, print available features</p>
</td></tr>
<tr><td><code id="importGTF_+3A_merge.feature">merge.feature</code></td>
<td>
<p>Character, merge multiple samples to dataset</p>
</td></tr>
<tr><td><code id="importGTF_+3A_merge.all">merge.all</code></td>
<td>
<p>Logial, shall all samples be merged</p>
</td></tr>
<tr><td><code id="importGTF_+3A_class.names">class.names</code></td>
<td>
<p>Vector with class names</p>
</td></tr>
<tr><td><code id="importGTF_+3A_verbose">verbose</code></td>
<td>
<p>Logical, verbose function output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function imports a gtf file. The features names to be imported are defined in <code>features</code>, several features are then
provided as vector. A list of available feature can beprinted, by setting <code>print.features=TRUE</code>.
</p>
<p>The <code>skip</code> option allows to skip a given number of rows, the default is, however, <code>auto</code>. In that case, all rows that
start with the <code>#</code> symbol are skipped.
</p>
<p>In case a set of expression values given in gtf format should be imported and to be merged into a single data table, the feature
that should be used for merging can be provided to the <code>merge.feature</code> option. In that case the function expects a folder
in <code>file</code> and it will import all gtfs located in that folder and merges them according to the <code>merge.feature</code> option.
With the option <code>class.names</code> a vector of prefixes for the merged features can be provided. If this is kept empty, then the 
filenames of the gtf will be used instead (without gtf extension).
</p>
<p>By default the function imprts all features in column 9 as string character. However, for common labels (FPKM and TPM) the class
type is set automatically to numeric. Additional numerical feature names can be defined with the <code>num.feature</code> option.
</p>


<h3>Value</h3>

<p>A gtf object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # Define here the location on HDD for the example file
   fpath &lt;- system.file("extdata","example.gtf", package="GenomicTools.fileHandler")
 # Same file, but this time as gzipped version
   fpath.gz &lt;- system.file("extdata","example2.gtf.gz", package="GenomicTools.fileHandler")
   
 # Import the example gtf file  
   importGTF(fpath, level="transcript", features=c("gene_id","FPKM"))
   
 ## Not run: 
 # For the current you need to have zcat installed (should be standard on a Linux system)
   importGTF(fpath.gz, level="transcript", features=c("gene_id","FPKM"))
   
## End(Not run)
 
</code></pre>

<hr>
<h2 id='importPED'>importPED</h2><span id='topic+importPED'></span>

<h3>Description</h3>

<p>Import a PED/MAP file pair
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importPED(file, n, snps = NULL, which, split = "\t| +", sep = ".",
  na.strings = "0", lex.order = FALSE, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importPED_+3A_file">file</code></td>
<td>
<p>ped filename</p>
</td></tr>
<tr><td><code id="importPED_+3A_n">n</code></td>
<td>
<p>Number of samples to read</p>
</td></tr>
<tr><td><code id="importPED_+3A_snps">snps</code></td>
<td>
<p>map filename</p>
</td></tr>
<tr><td><code id="importPED_+3A_which">which</code></td>
<td>
<p>Names of SNPS to import</p>
</td></tr>
<tr><td><code id="importPED_+3A_split">split</code></td>
<td>
<p>Columns separator in ped file</p>
</td></tr>
<tr><td><code id="importPED_+3A_sep">sep</code></td>
<td>
<p>Character that separates Alleles</p>
</td></tr>
<tr><td><code id="importPED_+3A_na.strings">na.strings</code></td>
<td>
<p>Definition for missing values</p>
</td></tr>
<tr><td><code id="importPED_+3A_lex.order">lex.order</code></td>
<td>
<p>Logical, lexicographical order</p>
</td></tr>
<tr><td><code id="importPED_+3A_verbose">verbose</code></td>
<td>
<p>Logical, verbose output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is to a large extend taken from <code>snpStat::read.pedmap</code>, but here is internally the <code>data.table::fread</code> function used
that resulted in much faster file processing.
</p>
<p>To import the data, the ped file can be provided to the <code>file</code> option and the map file to the <code>snps</code> option. If no option is given to
<code>snps</code> and the <code>file</code> option is provided without any file extension, then the ped/map extension are automaticall added
</p>


<h3>Value</h3>

<p>a pedmap object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # Define here the location on HDD for the example file
   pedPath &lt;- system.file("extdata","example.ped", package="GenomicTools.fileHandler")
   mapPath &lt;- system.file("extdata","example.map", package="GenomicTools.fileHandler")
 # Import the example ped/map files  
   importPED(file=pedPath, snps=mapPath)


</code></pre>

<hr>
<h2 id='importSTARLog'>importSTARLog</h2><span id='topic+importSTARLog'></span>

<h3>Description</h3>

<p>Import the Log-File from STAR
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importSTARLog(dir, recursive = TRUE, log = FALSE, finalLog = TRUE,
  verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importSTARLog_+3A_dir">dir</code></td>
<td>
<p>The directory name</p>
</td></tr>
<tr><td><code id="importSTARLog_+3A_recursive">recursive</code></td>
<td>
<p>Logical, check for sub-directories</p>
</td></tr>
<tr><td><code id="importSTARLog_+3A_log">log</code></td>
<td>
<p>boolean, import also log file</p>
</td></tr>
<tr><td><code id="importSTARLog_+3A_finallog">finalLog</code></td>
<td>
<p>boolean, import also final_log file</p>
</td></tr>
<tr><td><code id="importSTARLog_+3A_verbose">verbose</code></td>
<td>
<p>Logical, talkactive function feedback</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function imports the Log file from STAR
</p>


<h3>Value</h3>

<p>a data frame
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='importVCF'>importVCF</h2><span id='topic+importVCF'></span>

<h3>Description</h3>

<p>Import a VCF function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importVCF(file, na.seq = "./.")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importVCF_+3A_file">file</code></td>
<td>
<p>The file name</p>
</td></tr>
<tr><td><code id="importVCF_+3A_na.seq">na.seq</code></td>
<td>
<p>The missing value definition</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function imports a VCF file.
</p>
<p>In case the logicl flag 'phased' is set to TRUE then the genotypes are expected to be in the format 0|0, otherwise they are expected
to be like 0/1 .
</p>
<p>The example file was downloaded from here:
</p>
<p>ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/pilot_data/release/2010_07/exon/snps/
</p>


<h3>Value</h3>

<p>A vcf object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # Define here the location on HDD for the example file
   fpath &lt;- system.file("extdata","example.vcf", package="GenomicTools.fileHandler")
 # Import the example vcf file  
   importVCF(fpath)

</code></pre>

<hr>
<h2 id='importXML'>importXML</h2><span id='topic+importXML'></span>

<h3>Description</h3>

<p>Import an Blast XML file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importXML(folder, seqNames = NULL, which = NULL, idTH = 0.8,
  verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importXML_+3A_folder">folder</code></td>
<td>
<p>Character, folder path</p>
</td></tr>
<tr><td><code id="importXML_+3A_seqnames">seqNames</code></td>
<td>
<p>Names of sequences</p>
</td></tr>
<tr><td><code id="importXML_+3A_which">which</code></td>
<td>
<p>Which sequences to import</p>
</td></tr>
<tr><td><code id="importXML_+3A_idth">idTH</code></td>
<td>
<p>Use the threshold as cut-off</p>
</td></tr>
<tr><td><code id="importXML_+3A_verbose">verbose</code></td>
<td>
<p>Logical, verbose output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function imports XML files as provided as Blast output, it is mainly aimied to import the output of the hoardeR package
</p>


<h3>Value</h3>

<p>An XML object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='plotTotalReads'>plotTotalReads</h2><span id='topic+plotTotalReads'></span>

<h3>Description</h3>

<p>Plot the total reads
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTotalReads(STARLog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTotalReads_+3A_starlog">STARLog</code></td>
<td>
<p>A STARLog object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function plots the total reads from a STARlog object
</p>
<p>Part of the diagnostic plot series for of the STARLog. The function accepts also a list of STARLogs and creates then comparative boxplots
</p>


<h3>Value</h3>

<p>A plot
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='plotUniquelyMappedReads'>plotUniquelyMappedReads</h2><span id='topic+plotUniquelyMappedReads'></span>

<h3>Description</h3>

<p>Plot the uniquely mapped reads
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotUniquelyMappedReads(STARLog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotUniquelyMappedReads_+3A_starlog">STARLog</code></td>
<td>
<p>A STARLog object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function plots the percenage of uniquely reads from a STARlog object
</p>
<p>Part of the diagnostic plot series for of the STARLog. The function accepts also a list of STARLogs and creates then comparative boxplots
</p>


<h3>Value</h3>

<p>A plot
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='prereadGTF'>prereadGTF</h2><span id='topic+prereadGTF'></span>

<h3>Description</h3>

<p>Preread a gtf file and prints features of it for importing it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prereadGTF(file, nrow = 1000, skip = "auto")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prereadGTF_+3A_file">file</code></td>
<td>
<p>Filename</p>
</td></tr>
<tr><td><code id="prereadGTF_+3A_nrow">nrow</code></td>
<td>
<p>Number of rows to read</p>
</td></tr>
<tr><td><code id="prereadGTF_+3A_skip">skip</code></td>
<td>
<p>Rows to skip from top</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function reads in a gtf file and prints its features for the import step.
</p>
<p>By default this function only imports the first 1000 rows, in case all rows should be imported set <code>nrow=-1</code>.
</p>
<p>The number to skip in the beginning can be adjusted by the <code>skip</code> option. The default is here <code>auto</code> so that
the function can identify the correct amount of header rows. Hence, this option should be changed only, if there is a
good reason.
</p>


<h3>Value</h3>

<p>A list of available features
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='print.bed'>Print a bed Object</h2><span id='topic+print.bed'></span>

<h3>Description</h3>

<p>Prints a <code>bed</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bed'
print(x, n = 6, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.bed_+3A_x">x</code></td>
<td>
<p>Object of class <code>bed</code>.</p>
</td></tr>
<tr><td><code id="print.bed_+3A_n">n</code></td>
<td>
<p>Number of lines to print</p>
</td></tr>
<tr><td><code id="print.bed_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The print function displays a bed object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='print.fa'>Print a fa Object</h2><span id='topic+print.fa'></span>

<h3>Description</h3>

<p>Prints a <code>fa</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fa'
print(x, n = 2, seq.out = 50, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.fa_+3A_x">x</code></td>
<td>
<p>Object of class <code>fa</code>.</p>
</td></tr>
<tr><td><code id="print.fa_+3A_n">n</code></td>
<td>
<p>Number of sequences to display</p>
</td></tr>
<tr><td><code id="print.fa_+3A_seq.out">seq.out</code></td>
<td>
<p>Length of the subsequence to display</p>
</td></tr>
<tr><td><code id="print.fa_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The print function displays a fa object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='print.featureCounts'>Print a featureCounts Object</h2><span id='topic+print.featureCounts'></span>

<h3>Description</h3>

<p>Prints an <code>featureCounts</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'featureCounts'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.featureCounts_+3A_x">x</code></td>
<td>
<p>Object of class <code>featureCounts</code>.</p>
</td></tr>
<tr><td><code id="print.featureCounts_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The print function displays a featureCounts object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='print.fq'>Print a fq Object</h2><span id='topic+print.fq'></span>

<h3>Description</h3>

<p>Prints a <code>fq</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fq'
print(x, n = 2, seq.out = 50, print.qual = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.fq_+3A_x">x</code></td>
<td>
<p>Object of class <code>fq</code>.</p>
</td></tr>
<tr><td><code id="print.fq_+3A_n">n</code></td>
<td>
<p>Number of sequences to display</p>
</td></tr>
<tr><td><code id="print.fq_+3A_seq.out">seq.out</code></td>
<td>
<p>Length of the subsequence to display</p>
</td></tr>
<tr><td><code id="print.fq_+3A_print.qual">print.qual</code></td>
<td>
<p>Logical, shall the quality measures also be printed</p>
</td></tr>
<tr><td><code id="print.fq_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The print function displays a fa object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='print.gtf'>Print a gtf Object</h2><span id='topic+print.gtf'></span>

<h3>Description</h3>

<p>Prints a <code>gtf</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gtf'
print(x, n = 6, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.gtf_+3A_x">x</code></td>
<td>
<p>Object of class <code>gtf</code>.</p>
</td></tr>
<tr><td><code id="print.gtf_+3A_n">n</code></td>
<td>
<p>Number of lines to print</p>
</td></tr>
<tr><td><code id="print.gtf_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The print function displays a bed object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='print.pedMap'>Print a pedMap Object</h2><span id='topic+print.pedMap'></span>

<h3>Description</h3>

<p>Prints an <code>pedMap</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pedMap'
print(x, n = 6, m = 6, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.pedMap_+3A_x">x</code></td>
<td>
<p>Object of class <code>pedMap</code>.</p>
</td></tr>
<tr><td><code id="print.pedMap_+3A_n">n</code></td>
<td>
<p>Number of samples to display</p>
</td></tr>
<tr><td><code id="print.pedMap_+3A_m">m</code></td>
<td>
<p>Number of columns to display</p>
</td></tr>
<tr><td><code id="print.pedMap_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The print function displays a pedMap object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='print.vcf'>Print a vcf Object</h2><span id='topic+print.vcf'></span>

<h3>Description</h3>

<p>Prints an <code>vcf</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'vcf'
print(x, n = 6, m = 6, fullHeader = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.vcf_+3A_x">x</code></td>
<td>
<p>Object of class <code>vcf</code>.</p>
</td></tr>
<tr><td><code id="print.vcf_+3A_n">n</code></td>
<td>
<p>Number of samples to display</p>
</td></tr>
<tr><td><code id="print.vcf_+3A_m">m</code></td>
<td>
<p>Number of columns to display</p>
</td></tr>
<tr><td><code id="print.vcf_+3A_fullheader">fullHeader</code></td>
<td>
<p>Logical, shall the whole header be printed</p>
</td></tr>
<tr><td><code id="print.vcf_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The print function displays a vcf object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='summary.bed'>Summary of a bed Object</h2><span id='topic+summary.bed'></span>

<h3>Description</h3>

<p>Summarizes a <code>bed</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bed'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.bed_+3A_object">object</code></td>
<td>
<p>Object of class <code>bed</code>.</p>
</td></tr>
<tr><td><code id="summary.bed_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The summary function displays an informative summary of a bed object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='summary.fa'>Summary of a fa Object</h2><span id='topic+summary.fa'></span>

<h3>Description</h3>

<p>Summarizes a <code>fa</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fa'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.fa_+3A_object">object</code></td>
<td>
<p>Object of class <code>fa</code>.</p>
</td></tr>
<tr><td><code id="summary.fa_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The summary function displays an informative summary of a fa object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='summary.featureCounts'>Summary of a featureCounts Object</h2><span id='topic+summary.featureCounts'></span>

<h3>Description</h3>

<p>Summarizes a <code>featureCounts</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'featureCounts'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.featureCounts_+3A_object">object</code></td>
<td>
<p>Object of class <code>featureCounts</code>.</p>
</td></tr>
<tr><td><code id="summary.featureCounts_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The summary function displays an informative summary of a featureCounts object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='summary.fq'>Summary of a fq Object</h2><span id='topic+summary.fq'></span>

<h3>Description</h3>

<p>Summarizes a <code>fq</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fq'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.fq_+3A_object">object</code></td>
<td>
<p>Object of class <code>fq</code>.</p>
</td></tr>
<tr><td><code id="summary.fq_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The summary function displays an informative summary of a fq object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='summary.gtf'>Summary of a gtf Object</h2><span id='topic+summary.gtf'></span>

<h3>Description</h3>

<p>Summarizes a <code>gtf</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gtf'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.gtf_+3A_object">object</code></td>
<td>
<p>Object of class <code>gtf</code>.</p>
</td></tr>
<tr><td><code id="summary.gtf_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The summary function displays an informative summary of a gtf object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

<hr>
<h2 id='summary.STARLog'>Summary of a STARLog Object</h2><span id='topic+summary.STARLog'></span>

<h3>Description</h3>

<p>Summarizes a <code>STARLog</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'STARLog'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.STARLog_+3A_object">object</code></td>
<td>
<p>Object of class <code>STARLog</code>.</p>
</td></tr>
<tr><td><code id="summary.STARLog_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The summary function displays an informative summary of a STARLog object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
