<!DOCTYPE html><html><head><title>Help for package geno2proteo</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {geno2proteo}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#geno2proteo-package'>
<p>Finding the DNA and Protein Sequences of Any Genomic or</p>
Proteomic Loci</a></li>
<li><a href='#generatingCDSaaFile'>
<p>Generating a file containing the DNA and AA sequences of all the protein</p>
coding regions (CDSs) in a genome</a></li>
<li><a href='#genomicLocsToProteinSequence'><p>Obtaining the protein sequences and DNA sequences of the coding regions</p>
within a list of loci in genome</a></li>
<li><a href='#genomicLocsToWholeDNASequence'>
<p>Obtaining the DNA sequences of a list of genomic loci</p></a></li>
<li><a href='#proteinLocsToGenomic'>
<p>Obtaining the genomic coordinates for a list of protein sections</p></a></li>
<li><a href='#proteinLocsToProteinSeq'>
<p>Obtaining the amino acid sequences of a list of protein sections</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Finding the DNA and Protein Sequences of Any Genomic or
Proteomic Loci</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-06-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Yaoyong Li </td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yaoyong Li&lt;liyaoyong85@gmail.com&gt;</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Genetics, Proteomics, Sequencing, Annotation,
GenomeAnnotation, GenomeAssembly</td>
</tr>
<tr>
<td>Description:</td>
<td>Using the DNA sequence and gene annotation files provided in 
            'ENSEMBL' <a href="https://www.ensembl.org/index.html">https://www.ensembl.org/index.html</a>, 
            the functions implemented in the package try to find the DNA sequences and 
            protein sequences of any given genomic loci, and to find the genomic coordinates 
            and protein sequences of any given protein locations, which are the frequent
            tasks in the analysis of genomic and proteomic data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>S4Vectors, BiocGenerics, GenomicRanges, IRanges, R.utils,
RUnit</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>Perl (&gt;= 2.0.0)</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-13 10:12:17 UTC; mdehsyl2</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-13 10:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='geno2proteo-package'>
Finding the DNA and Protein Sequences of Any Genomic or
Proteomic Loci
</h2><span id='topic+geno2proteo-package'></span><span id='topic+geno2proteo'></span>

<h3>Description</h3>

<p>Using the DNA sequence and gene annotation files provided in 
            'ENSEMBL' &lt;https://www.ensembl.org/index.html&gt;, 
            the functions implemented in the package try to find the DNA sequences and 
            protein sequences of any given genomic loci, and to find the genomic coordinates 
            and protein sequences of any given protein locations, which are the frequent
            tasks in the analysis of genomic and proteomic data.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> geno2proteo</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Finding the DNA and Protein Sequences of Any Genomic or
Proteomic Loci</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.0.6</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2022-06-10</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Yaoyong Li </td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Yaoyong Li&lt;liyaoyong85@gmail.com&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
biocViews: </td><td style="text-align: left;"> Genetics, Proteomics, Sequencing, Annotation,
GenomeAnnotation, GenomeAssembly</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Using the DNA sequence and gene annotation files provided in 
            'ENSEMBL' &lt;https://www.ensembl.org/index.html&gt;, 
            the functions implemented in the package try to find the DNA sequences and 
            protein sequences of any given genomic loci, and to find the genomic coordinates 
            and protein sequences of any given protein locations, which are the frequent
            tasks in the analysis of genomic and proteomic data.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> Artistic-2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R (&gt;= 3.0)</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> S4Vectors, BiocGenerics, GenomicRanges, IRanges, R.utils,
RUnit</td>
</tr>
<tr>
 <td style="text-align: left;">
SystemRequirements: </td><td style="text-align: left;"> Perl (&gt;= 2.0.0)</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
NeedsCompilation: </td><td style="text-align: left;"> no</td>
</tr>
<tr>
 <td style="text-align: left;">
Packaged: </td><td style="text-align: left;"> 2022-06-10 15:10:25 UTC; mdehsyl2</td>
</tr>
<tr>
 <td style="text-align: left;">
Repository: </td><td style="text-align: left;"> CRAN</td>
</tr>
<tr>
 <td style="text-align: left;">
Date/Publication: </td><td style="text-align: left;"> 2022-06-10 15:50:02 UTC</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<p>Index of help topics:
</p>
<pre>
generatingCDSaaFile                                   Generating a file containing the DNA and AA sequences of all the protein coding regions (CDSs) in a genome
geno2proteo-package                                   Finding the DNA and Protein Sequences of Any Genomic or Proteomic Loci
genomicLocsToProteinSequence                          Obtaining the protein sequences and DNA sequences of the coding regions within a list of loci in genome
genomicLocsToWholeDNASequence                         Obtaining the DNA sequences of a list of genomic loci
proteinLocsToGenomic                                  Obtaining the genomic coordinates for a list of protein sections
proteinLocsToProteinSeq                               Obtaining the amino acid sequences of a list of protein sections
</pre>

<p>Further information is available in the following vignettes:<br /><br />
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>geno2proteoIntroduction</code> </td><td style="text-align: left;"> An Introduction to the geno2proteo package (source, pdf)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>~~ An overview of how to use the package and the most important functions ~~
</p>
<p>The package needs three data files. One contains the genetic table, another 
one contains the DNA sequences of a certain genome, and the third one contains
the gene annotations in the same format as the GTF file used in ENSEMBL.
The standard genetic table was provided in this package. The other two data 
files can be downloaded from ENSEMBL web site. For details about those data 
files and how to obtain them, see the introduction document of this package. 
Of course you can create your own data files, as long as you observe the format 
of the files which are specified in the introduction document as well. 
</p>
<p>The package also needs Perl installed and being available for use in order to 
run some of the functions. 
</p>
<p>Once you have the three data files, you need to run the function 
<code>generatingCDSaaFile</code> to generate another data file, which 
will be used by some of the functions in this package.
</p>
<p>Four main functions were implemented in this package so far.
The function <code>genomicLocToProteinSequence</code> will find the protein 
sequences and DNA sequences of the coding regions within a list of genomic 
loci given as input. <code>genomicLocToWholeDNASequence</code> will obtain the 
whole DNA sequences of any genomic loci given in the input data. 
<code>proteinLocsToGenomic</code> will find the genomic coordinates for a list of
sections in proteins as input. <code>proteinLocsToProteinSeq</code> will find the
the protein sequences of a list of the protein sections.
</p>


<h3>Author(s)</h3>

<p>Yaoyong Li 
</p>
<p>Maintainer: Yaoyong Li&lt;liyaoyong85@gmail.com&gt;
</p>

<hr>
<h2 id='generatingCDSaaFile'>
Generating a file containing the DNA and AA sequences of all the protein 
coding regions (CDSs) in a genome
</h2><span id='topic+generatingCDSaaFile'></span>

<h3>Description</h3>

<p>This function will find the DNA and protein sequences for each CDS region 
listed in the ENSEMBL gene annotation file (gtf file) provided, and store the
CDS regions and the corresponding DNA and protein sequences in an output data 
file. The output data file will be needed by some of the functions in this 
package.   
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generatingCDSaaFile(geneticCodeFile_line, gtfFile, DNAfastaFile, 
outputFolder = "./", perlExec='perl')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generatingCDSaaFile_+3A_geneticcodefile_line">geneticCodeFile_line</code></td>
<td>

<p>A text file containing a genetic coding table of the codons and the amino acids
coded by those codons. A file containing the standard genetic coding table is 
provided in this package, which is the file
&lsquo;geneticCode_standardTable_lines.txt&rsquo; 
in the folder &quot;geno2proteo/extdata/&quot; which will be located in the folder that
you install the package. This will be 
available after you have installed the package. 
</p>
<p>Alternatively you can create your own genetic table if needed. The format of 
the genetic table used in this package 
is one line for one codon. The first column is the codon (namely 3 DNA letters)
and the second column 
is the name (in a single letter) of the amino acid coded by the codon. You may
have more columns 
if you want but only the first two columns are used by the package, and the
columns are separated by a tab.
</p>
</td></tr>
<tr><td><code id="generatingCDSaaFile_+3A_gtffile">gtfFile</code></td>
<td>

<p>A text file in GTF format containing the gene annotations of the species
that you are interested in. You may obtain this file from the ENSEMBL 
web site. The text file can be compressed by GNU Zip (e.g. gzip). For the 
details about how to get the 
data file from ENSEMBL web site, see the documentation of this package.
</p>
</td></tr>
<tr><td><code id="generatingCDSaaFile_+3A_dnafastafile">DNAfastaFile</code></td>
<td>

<p>A text file in fasta format containing the DNA sequence for the genome that 
you want to use 
for analysing the data. The text file can be compressed by GNU Zip (e.g. gzip).
You may download the file directly 
from the ENSEMBL web site. For the details about how to get the 
data file from ENSEMBL web site, see the documentation of this package.
</p>
</td></tr>
<tr><td><code id="generatingCDSaaFile_+3A_outputfolder">outputFolder</code></td>
<td>

<p>A folder where the result file will be stored. The default value is the 
current folder &quot;./&quot;.
</p>
</td></tr>
<tr><td><code id="generatingCDSaaFile_+3A_perlexec">perlExec</code></td>
<td>

<p>Its value should be the full path of the executable file which can be used 
to run Perl scripts (e.g. &quot;/usr/bin/perl&quot; in a linux computer
or &quot;C:/Strawberry/perl/bin/perl&quot; in a Windows computer). 
The default value is &quot;perl&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates a data file containing the genomic locations, 
DNA sequences and protein sequences of all of the coding regions (CDSs), which
will be used by some of other functions in this packages.
</p>


<h3>Value</h3>

<p>This function does not return any specific value but generates a file in the 
output folder containing the DNA and AA sequences of the CDS regions. The first
part of the name of the result file is same as the gene annotation file 
(namely the gtf file), and the last part is &lsquo;_AAseq.txt.gz&rsquo;. 
</p>


<h3>Author(s)</h3>

<p>Yaoyong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # the data folder in this package
    dataFolder = system.file("extdata", package="geno2proteo") 

    geneticCodeFile_line = file.path(dataFolder, 
                                    "geneticCode_standardTable_lines.txt")
    gtfFile = file.path(dataFolder, 
                "Homo_sapiens.GRCh37.74_chromosome16_35Mlong.gtf.gz")
    DNAfastaFile =  file.path(dataFolder, 
            "Homo_sapiens.GRCh37.74.dna.chromosome.16.fa_theFirst3p5M.txt.gz")
    
    outputFolder = tempdir(); # using the current folder as output folder
    # calling the function.
    generatingCDSaaFile(geneticCodeFile_line=geneticCodeFile_line, 
        gtfFile=gtfFile, DNAfastaFile=DNAfastaFile, outputFolder=outputFolder)
    
    filename00 = sub(".*/", "", gtfFile)
    # get the output file's name.
    outputFile = paste(outputFolder, "/", filename00, "_AAseq.txt.gz", sep="")
    # read the content of the output file into a data frame.
    aaSeq = read.table(outputFile, sep="\t", stringsAsFactors=FALSE)

</code></pre>

<hr>
<h2 id='genomicLocsToProteinSequence'>Obtaining the protein sequences and DNA sequences of the coding regions
within a list of loci in genome</h2><span id='topic+genomicLocsToProteinSequence'></span>

<h3>Description</h3>

<p><code>genomicsLocToProteinSequence</code> takes a list of genomic loci given in the 
input and tries to find the protein sequences and DNA sequences of the coding 
regions of genome which are within those genomic loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomicLocsToProteinSequence(inputLoci, CDSaaFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genomicLocsToProteinSequence_+3A_inputloci">inputLoci</code></td>
<td>

<p>A data frame containing the genomic loci as the input. Each row is for one 
genomic locus. The first column is for the chromosome, the 2nd and 3rd columns 
are for the start and end coordinates of the locus in the chromosome, and the 
4th column is for the strand (&quot;+&quot; or &quot;-&quot; for forward and reverse strand, 
respectively). Other columns are optional and will not be used by the function.
Note that the chromosome name can be either in the ENSEMBL style, e.g. 1, 2, 3,
..., and X, Y and MT, or in another popular style, namely chr1, chr2, chr3, 
..., and chrX, chrY and chrM. But they cannot be mixed in the input of one 
function call.
</p>
</td></tr>
<tr><td><code id="genomicLocsToProteinSequence_+3A_cdsaafile">CDSaaFile</code></td>
<td>

<p>The data file generated by the package's function <code>generatingCDSaaFile</code>,
containing the genomic locations, DNA sequences and protein sequences of all 
coding regions in a specific genome which is used in your analysis.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the original genomic loci specified in the input and 
the protein sequence and the DNA sequence of the coding regions within each of 
the loci. In detail, the returned data frame contains the original genomic loci
specified in the input and after them, the five added columns:
</p>

<ul>
<li><p> Column &quot;transId&quot; lists the ENSEMBL IDs of the transcripts whose coding 
regions overlap with locus specified and the overlapping coding regions are 
exactly the same among those transcripts.
</p>
</li>
<li><p> Column &quot;dnaSeq&quot; contains the DNA sequence in the overlapping coding 
regions.
</p>
</li>
<li><p> Column &quot;dnaBefore&quot; contains the DNA letters which are in the same codon 
as the first letter in the DNA sequence in the column &quot;dnaSeq&quot;.
</p>
</li>
<li><p> Column &quot;dnaAfter&quot; contains the DNA letters which are in the same codon 
as the last letter in the DNA sequence in the previous column 'dnaSeq'.
</p>
</li>
<li><p> Column &quot;pepSeq&quot; contains the protein sequence translated from the DNA 
sequences in the three preceding columns, &quot;dnaBefore&quot;, &quot;dnaSeq&quot; and &quot;dnaAfter&quot;.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yaoyong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
    dataFolder = system.file("extdata", package="geno2proteo")
    inputFile_loci=file.path(dataFolder, 
        "transId_pfamDomainStartEnd_chr16_Zdomains_22examples_genomicPos.txt")
    CDSaaFile=file.path(dataFolder, 
        "Homo_sapiens.GRCh37.74_chromosome16_35Mlong.gtf.gz_AAseq.txt.gz")

    inputLoci = read.table(inputFile_loci, sep="\t", stringsAsFactors=FALSE)

    proteinSeq = genomicLocsToProteinSequence(inputLoci=inputLoci, 
                                            CDSaaFile=CDSaaFile)

</code></pre>

<hr>
<h2 id='genomicLocsToWholeDNASequence'>
Obtaining the DNA sequences of a list of genomic loci
</h2><span id='topic+genomicLocsToWholeDNASequence'></span>

<h3>Description</h3>

<p>The function takes a list of genomic loci and tries to find the whole DNA 
sequences within each of the loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomicLocsToWholeDNASequence(inputLoci, DNAfastaFile, 
                                tempFolder = "./", perlExec='perl')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genomicLocsToWholeDNASequence_+3A_inputloci">inputLoci</code></td>
<td>

<p>A data frame containing the genomic loci as the input. Each row is for one 
genomic locus. The first column is the chromosome name, the 2nd and 3rd columns
are the start and end coordinates of the locus in the chromosome, and the 4th 
column specifies the strand of chromosome (&quot;+&quot; and &quot;-&quot; for forward and reverse
strand, respectively). Other columns are optional and will not be used by the 
function. Note that the chromosome name can be either in the ENSEMBL style, 
e.g. 1, 2, 3, ..., and X, Y and MT, or in another popular style, namely 
chr1, chr2, chr3, ..., and chrX, chrY and chrM. But they cannot be mixed in 
the input of one function call.
</p>
</td></tr>
<tr><td><code id="genomicLocsToWholeDNASequence_+3A_dnafastafile">DNAfastaFile</code></td>
<td>

<p>The name of a fasta file containing the whole DNA sequence of the genome used.
For details about this data file see the documentation of this package.
</p>
</td></tr>
<tr><td><code id="genomicLocsToWholeDNASequence_+3A_tempfolder">tempFolder</code></td>
<td>

<p>A temporary folder into which the program can write some temporary files 
which will be deleted when the function running is finished. The default value
is the current folder.
</p>
</td></tr>
<tr><td><code id="genomicLocsToWholeDNASequence_+3A_perlexec">perlExec</code></td>
<td>

<p>Its value should be the full path of the executable file which can be used 
to run Perl scripts (e.g. &quot;/usr/bin/perl&quot; in a linux computer or 
&quot;C:/Strawberry/perl/bin/perl&quot; in a Windows computer). The default value is 
&quot;perl&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function obtains the whole DNA sequences of a list of genomic loci. 
Note that, in contrast, another function <code>genomicLocToProteinSequence</code> in
this package can return the DNA sequences of the coding regions within the 
given genomic loci. 
</p>


<h3>Value</h3>

<p>The function returns a data frame containing the original genomic loci as in 
the input and after them, one additional column for the DNA sequence of the 
corresponding genomic locus.
</p>


<h3>Author(s)</h3>

<p>Yaoyong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
    dataFolder = system.file("extdata", package="geno2proteo")
    inputFile_loci=file.path(dataFolder, 
        "transId_pfamDomainStartEnd_chr16_Zdomains_22examples_genomicPos.txt")
    DNAfastaFile =  file.path(dataFolder, 
        "Homo_sapiens.GRCh37.74.dna.chromosome.16.fa_theFirst3p5M.txt.gz")

    inputLoci = read.table(inputFile_loci, sep="\t", stringsAsFactors=FALSE)
    
    tmpFolder = tempdir()

    DNASeqNow = genomicLocsToWholeDNASequence(inputLoci=inputLoci, 
                            DNAfastaFile=DNAfastaFile, tempFolder=tmpFolder)

</code></pre>

<hr>
<h2 id='proteinLocsToGenomic'>
Obtaining the genomic coordinates for a list of protein sections
</h2><span id='topic+proteinLocsToGenomic'></span>

<h3>Description</h3>

<p>The function takes a list of protein sections and the corresponding ENSEMBL ID
of these proteins, and tries to find the genomic coordinates of these protein 
sections.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>proteinLocsToGenomic(inputLoci, CDSaaFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="proteinLocsToGenomic_+3A_inputloci">inputLoci</code></td>
<td>

<p>A data frame containing the protein sections as the input. The 1st column must
be the ENSEMBL ID of either the protein or the transcript encoding the protein
(or the equivalent of 
ENSEMBL ID if you have created your own gene annotation GTF file). But you have
to use only one of two formats (namely either protein ID or transcript ID), and
cannot use both of them in the input of one function call. The 2nd and 3rd 
columns give the coordinate of the first and last amino acids of the section 
along the protein sequence. Other columns are optional and will not be used by
the function.  
</p>
</td></tr>
<tr><td><code id="proteinLocsToGenomic_+3A_cdsaafile">CDSaaFile</code></td>
<td>

<p>The data file generated by the package's function <code>generatingCDSaaFile</code>, 
containing the genomic locations, DNA sequences and protein sequences of all 
coding regions in a specific genome which is used in your analysis.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a data frame containing the original protein locations 
specified in the input and before them, the six added columns for the 
corresponding genomic coordinates of the protein sections:
</p>

<ul>
<li><p> The 1st, 2nd, 3rd and 4th columns give the chromosome name, the 
coordinates of the start and end positions, and the strand in the chromosome, 
which specify the genomic locus corresponding to the protein section.
</p>
</li>
<li><p> The 5th and 6th columns give the first and last coding exons in the given
transcript which correspond to the given protein section. 
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Yaoyong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
    dataFolder = system.file("extdata", package="geno2proteo")
    inputFile_loci=file.path(dataFolder, 
        "transId_pfamDomainStartEnd_chr16_Zdomains_22examples.txt")
    CDSaaFile=file.path(dataFolder, 
        "Homo_sapiens.GRCh37.74_chromosome16_35Mlong.gtf.gz_AAseq.txt.gz")

    inputLoci = read.table(inputFile_loci, sep="\t", stringsAsFactors=FALSE)

    genomicLoci = proteinLocsToGenomic(inputLoci=inputLoci, CDSaaFile=CDSaaFile)
</code></pre>

<hr>
<h2 id='proteinLocsToProteinSeq'>
Obtaining the amino acid sequences of a list of protein sections
</h2><span id='topic+proteinLocsToProteinSeq'></span>

<h3>Description</h3>

<p>Given a list of sections in proteins defined by the ENSEMBL IDs of those
proteins and thestart and end coordinates of those sections along the amino
acid sequences of the proteins, the function returns the amino acid sequences
of those sections.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>proteinLocsToProteinSeq(inputLoci, CDSaaFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="proteinLocsToProteinSeq_+3A_inputloci">inputLoci</code></td>
<td>

<p>A data frame containing the coordinates of the protein sections in the protein
sequences. The 1st column must be the ENSEMBL ID of either the protein or the 
transcript that the protein corresponds to (or the equivalent of ENSEMBL ID if
you have created your own gene annotation GTF file). But you have to use onnly
one of two formats (namely protein ID or transcript ID), and cannot use both 
of them in the input of one function call. The 2nd and 3rd columns give the 
coordinate of the first and last amino acids of the section in the protein 
sequence. Other columns are optional and will not be used by the function.
</p>
</td></tr>
<tr><td><code id="proteinLocsToProteinSeq_+3A_cdsaafile">CDSaaFile</code></td>
<td>

<p>The data file generated by the package's function <code>generatingCDSaaFile</code>, 
containing the genomic locations, DNA sequences and protein sequences of all 
coding regions in a specific genome which is used in your analysis.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a data frame containing the original protein locations 
specified in the input and after them, one added columnfor the amino acid 
sequences of the protein sections.
</p>


<h3>Author(s)</h3>

<p>Yaoyong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    dataFolder = system.file("extdata", package="geno2proteo")
    inputFile_loci=file.path(dataFolder, 
        "transId_pfamDomainStartEnd_chr16_Zdomains_22examples.txt")
    CDSaaFile=file.path(dataFolder, 
        "Homo_sapiens.GRCh37.74_chromosome16_35Mlong.gtf.gz_AAseq.txt.gz")

    inputLoci = read.table(inputFile_loci, sep="\t", stringsAsFactors=FALSE)

    ProtSeqNow = proteinLocsToProteinSeq(inputLoci=inputLoci, 
                                            CDSaaFile=CDSaaFile)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
