<!DOCTYPE html><html><head><title>Help for package genogeographer</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {genogeographer}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#app_genogeo'><p>Shiny application for GenoGeoGrapher</p></a></li>
<li><a href='#bar_colour'><p>bar_colour</p></a></li>
<li><a href='#error_bar_plot'><p>Plot log likelihoods of profiles with approximate confidence intervals</p></a></li>
<li><a href='#exponent_tilt'><p>P-values from Importing Sampling using Exponential tilting</p></a></li>
<li><a href='#genogeo'><p>Likelihood ratio tests for AIMs</p></a></li>
<li><a href='#genogeographer'><p>genogeographer: Methods for analysing forensic Ancestry Informative Markers</p></a></li>
<li><a href='#kidd_loci'><p>Kenn Kidd Lab markers</p></a></li>
<li><a href='#LR_table'><p>Compute pairwise likelihood ratios</p></a></li>
<li><a href='#main_alleles'><p>AIMs markers in Precision ID Ancestry Panel (Thermo Fisher Scientific)</p></a></li>
<li><a href='#map_plot'><p>Plot LTR z-scores on map</p></a></li>
<li><a href='#pops_to_DB'><p>Pre-compute the scores for a given reference database</p></a></li>
<li><a href='#profile_AA_x0'><p>Function that compute the genotype probability for each population (rows in df)</p></a></li>
<li><a href='#profile_admixture'><p>Compute the z-score (and more) for admixed hypotheses</p></a></li>
<li><a href='#random_AIMs_profile'><p>Simulate a random AIMs profile</p></a></li>
<li><a href='#seldin_loci'><p>Seldin Lab markers</p></a></li>
<li><a href='#simulate_pops'><p>Simulate random populations</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Methods for Analysing Forensic Ancestry Informative Markers</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.19</td>
</tr>
<tr>
<td>Author:</td>
<td>Torben Tvedebrink</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Torben Tvedebrink &lt;tvede@math.aau.dk&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>leaflet, shiny, shinyjs, knitr, DT, shinycssloaders, purrr,
dplyr, magrittr, tidyr, ggplot2, tibble, forcats, readr,
rmarkdown, rio, maps, shinyWidgets, rlang</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tidyverse</td>
</tr>
<tr>
<td>Description:</td>
<td>Evaluates likelihood ratio tests for alleged ancestry. Implements the methods of Tvedebrink et al (2018) &lt;<a href="https://doi.org/10.1016%2Fj.tpb.2017.12.004">doi:10.1016/j.tpb.2017.12.004</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-09-27 09:41:48 UTC; tvede</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-09-27 10:20:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='app_genogeo'>Shiny application for GenoGeoGrapher</h2><span id='topic+app_genogeo'></span>

<h3>Description</h3>

<p>Shiny application for GenoGeoGrapher
</p>


<h3>Usage</h3>

<pre><code class='language-R'>app_genogeo(db_list = NULL, reporting_panel = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="app_genogeo_+3A_db_list">db_list</code></td>
<td>
<p>A named list of databases of reference populations. 
Each component is expected to be returned from <code>pops_to_DB</code>.</p>
</td></tr>
<tr><td><code id="app_genogeo_+3A_reporting_panel">reporting_panel</code></td>
<td>
<p>Logical. Should report generate and download be available after sample analysis.</p>
</td></tr>
</table>

<hr>
<h2 id='bar_colour'>bar_colour</h2><span id='topic+bar_colour'></span>

<h3>Description</h3>

<p>Creates the colour scale for the accepted and rejected populations based on z-score and the log likelihood (log P).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bar_colour(df, alpha = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bar_colour_+3A_df">df</code></td>
<td>
<p>A data.frame with at least three coloums. The first column is the logP, the second logical (z_score accept/reject), the third a unique naming column.</p>
</td></tr>
<tr><td><code id="bar_colour_+3A_alpha">alpha</code></td>
<td>
<p>Should the alpha opacity be applied? And what value, 1 = solid, 0 = transparent.</p>
</td></tr>
</table>

<hr>
<h2 id='error_bar_plot'>Plot log likelihoods of profiles with approximate confidence intervals</h2><span id='topic+error_bar_plot'></span>

<h3>Description</h3>

<p>Plots the estimated profile probabilities in each population.
The colour depends on the profiles likelihood and rejection/acceptance (blue/red) based on z-score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>error_bar_plot(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="error_bar_plot_+3A_data">data</code></td>
<td>
<p>The output from the <code>genogeo</code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A barplot of the log likelihoods for each population with confidence limits
</p>


<h3>Author(s)</h3>

<p>Torben Tvedebrink, <a href="mailto:tvede@math.aau.dk">tvede@math.aau.dk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_ &lt;- simulate_pops(pop_n = 20, aims_n = 50)
df_db &lt;- pops_to_DB(df_)
profile &lt;- random_AIMs_profile(df_db, keep_pop = TRUE)
profile$pop[1] # The true population
result &lt;- genogeo(profile[,c("locus","x0")], df = df_db)
error_bar_plot(result)
</code></pre>

<hr>
<h2 id='exponent_tilt'>P-values from Importing Sampling using Exponential tilting</h2><span id='topic+exponent_tilt'></span>

<h3>Description</h3>

<p>P-values from Importing Sampling using Exponential tilting
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exponent_tilt(x0, x1, n, p_limit = 0.1, B = 500, return_all = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exponent_tilt_+3A_x0">x0</code></td>
<td>
<p>Allele count of profile</p>
</td></tr>
<tr><td><code id="exponent_tilt_+3A_x1">x1</code></td>
<td>
<p>Population allele count</p>
</td></tr>
<tr><td><code id="exponent_tilt_+3A_n">n</code></td>
<td>
<p>Sampled alleles in total in population</p>
</td></tr>
<tr><td><code id="exponent_tilt_+3A_p_limit">p_limit</code></td>
<td>
<p>Upper limit to which we use the normal approximation</p>
</td></tr>
<tr><td><code id="exponent_tilt_+3A_b">B</code></td>
<td>
<p>An integer specifying the number of importance samples.</p>
</td></tr>
<tr><td><code id="exponent_tilt_+3A_return_all">return_all</code></td>
<td>
<p>Default is FALSE. If TRUE: Returns p-value, standard deviation, and method (and diagnostics).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The method of importance sampling described in Tvedebrink et al (2018), Section 2.3 is implemented.
It relies on exponential tilting of the proposal distribution using in the importance sampling.
</p>


<h3>Value</h3>

<p>If <code>return_all= FALSE</code> the p-value is returned. Otherwise list of elements (see <code>return_all</code>) is returned.
</p>

<hr>
<h2 id='genogeo'>Likelihood ratio tests for AIMs</h2><span id='topic+genogeo'></span>

<h3>Description</h3>

<p>Computes the likelihood ratio test statistics for each population in a database of reference populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genogeo(profile, df, CI = 0.95, min_n = 75, grouping = "pop",
  tilt = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genogeo_+3A_profile">profile</code></td>
<td>
<p>The AIMs profile encoded as returned by the <code>profile_AA_x0</code> function.</p>
</td></tr>
<tr><td><code id="genogeo_+3A_df">df</code></td>
<td>
<p>The database of reference populations as returned by the <code>pops_to_DB</code> function.</p>
</td></tr>
<tr><td><code id="genogeo_+3A_ci">CI</code></td>
<td>
<p>The confidence level used to reject or accept the various hypotheses (between 0 and 1).</p>
</td></tr>
<tr><td><code id="genogeo_+3A_min_n">min_n</code></td>
<td>
<p>Minimum number of individuals in each database sample</p>
</td></tr>
<tr><td><code id="genogeo_+3A_grouping">grouping</code></td>
<td>
<p>should <code>"pop"</code> (the default) or <code>"meta"</code> be used for aggregating the results.
Can also be <code>"cluster"</code> if this variable is defined in the input database.</p>
</td></tr>
<tr><td><code id="genogeo_+3A_tilt">tilt</code></td>
<td>
<p>Should exponential titling be used to obtain more accurate $p$-values in the distribution's tail
(currently not implemented)</p>
</td></tr>
<tr><td><code id="genogeo_+3A_...">...</code></td>
<td>
<p>Further arguments that are passed to other functions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble containing the $z$-scores, $p$-values etc for each population.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_ &lt;- simulate_pops(pop_n = 20, aims_n = 50)
df_db &lt;- pops_to_DB(df_)
profile &lt;- random_AIMs_profile(df_db, keep_pop = TRUE)
profile$pop[1] # The true population
result &lt;- genogeo(profile[,c("locus","x0")], df = df_db)
</code></pre>

<hr>
<h2 id='genogeographer'>genogeographer: Methods for analysing forensic Ancestry Informative Markers</h2><span id='topic+genogeographer'></span><span id='topic+genogeographer-package'></span>

<h3>Description</h3>

<p>The genogeographer package provides:
genogeo()
</p>


<h3>genogeo functions</h3>

<p>See ?genogeo
</p>

<hr>
<h2 id='kidd_loci'>Kenn Kidd Lab markers</h2><span id='topic+kidd_loci'></span>

<h3>Description</h3>

<p>List of markers identified by Kenn Kidd lab.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kidd_loci
</code></pre>


<h3>Format</h3>

<p>List of 55 markers
</p>

<dl>
<dt>locus</dt><dd><p>Locus/Marker names</p>
</dd>
</dl>


<h3>Source</h3>

<p>K.K. Kidd et al.Progress toward an efficient panel of SNPs for ancestry inference.
Forensic Science International: Genetics 10 (2014) 23–32
</p>

<hr>
<h2 id='LR_table'>Compute pairwise likelihood ratios</h2><span id='topic+LR_table'></span>

<h3>Description</h3>

<p>For each pair of a specified vector of profiles the likelihood ratios are computed.
The list can include all populations in the data or only a subset.
We may for inferral purposes restrict to ratios including at least one &quot;accepted&quot; population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LR_table(result_df, lr_populations = NULL, only_accepted = TRUE,
  CI = 0.95, digits = NULL, keep_logP = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LR_table_+3A_result_df">result_df</code></td>
<td>
<p>The output from <code>genogeo</code></p>
</td></tr>
<tr><td><code id="LR_table_+3A_lr_populations">lr_populations</code></td>
<td>
<p>A vector of population names (<code>pop</code> in <code>result_df</code>). If NULL all populations are used.</p>
</td></tr>
<tr><td><code id="LR_table_+3A_only_accepted">only_accepted</code></td>
<td>
<p>Restrict the ratios to include minimum one accepted population.</p>
</td></tr>
<tr><td><code id="LR_table_+3A_ci">CI</code></td>
<td>
<p>The level of confidence interval to be computed</p>
</td></tr>
<tr><td><code id="LR_table_+3A_digits">digits</code></td>
<td>
<p>If rounding of the output should be performed.</p>
</td></tr>
<tr><td><code id="LR_table_+3A_keep_logp">keep_logP</code></td>
<td>
<p>Logical. Should the logP's be returned in output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with numerator and denominator populations with their log10 LR and uncertainty.
</p>


<h3>Author(s)</h3>

<p>Torben Tvedebrink <a href="mailto:tvede@math.aau.dk">tvede@math.aau.dk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_ &lt;- simulate_pops(pop_n = 4, aims_n = 50)
df_db &lt;- pops_to_DB(df_)
profile &lt;- random_AIMs_profile(df_db, keep_pop = TRUE)
profile$pop[1] # The true population
result &lt;- genogeo(profile[,c("locus","x0")], df = df_db)
LR_table(result)
</code></pre>

<hr>
<h2 id='main_alleles'>AIMs markers in Precision ID Ancestry Panel (Thermo Fisher Scientific)</h2><span id='topic+main_alleles'></span>

<h3>Description</h3>

<p>List of markers with their main and alternative allele. 
The markers is the union of Seldin's and Kidd's markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>main_alleles
</code></pre>


<h3>Format</h3>

<p>List of 164 markers
</p>

<dl>
<dt>locus</dt><dd><p>Locus/Marker names</p>
</dd>
<dt>main_allele</dt><dd><p>The main allele (alleles are in lexicographic order)</p>
</dd>
<dt>other_allele</dt><dd><p>The other variant</p>
</dd>
</dl>

<hr>
<h2 id='map_plot'>Plot LTR z-scores on map</h2><span id='topic+map_plot'></span>

<h3>Description</h3>

<p>Plots the results from LRT on a map based on lat/lon info in the database.
If no location is found in the data (e.g. using <code>simulte_pops</code>) nothing is plotted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map_plot(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="map_plot_+3A_data">data</code></td>
<td>
<p>The output from the <code>genogeo</code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A map with population z-scores at their geographic origin
</p>


<h3>Author(s)</h3>

<p>Torben Tvedebrink, <a href="mailto:tvede@math.aau.dk">tvede@math.aau.dk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_ &lt;- simulate_pops(pop_n = 4, aims_n = 50)
df_db &lt;- pops_to_DB(df_)
profile &lt;- random_AIMs_profile(df_db, keep_pop = TRUE)
profile$pop[1] # The true population
result &lt;- genogeo(profile[,c("locus","x0")], df = df_db, min_n = 0) 
result$lon &lt;- runif(n = 4, min = -125, max = 125)
result$lat &lt;- runif(n = 4, min = -50, max = 80)
## Not run: map_plot(result)
</code></pre>

<hr>
<h2 id='pops_to_DB'>Pre-compute the scores for a given reference database</h2><span id='topic+pops_to_DB'></span>

<h3>Description</h3>

<p>Convert the counts from each population over a range of AIMs SNPs q
to observed likelihood ratio test, its mean and variance.
Based on these pre-computed the evaluation of a specific profile is done
using <code>genogeo</code> with the resulting dataframe as <code>df</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pops_to_DB(db, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pops_to_DB_+3A_db">db</code></td>
<td>
<p>A dataframe with columns similar to those of <code>simulate_pops()</code>.
If <code>db</code> contains information (recommended!) about &quot;meta&quot; (meta population)
and &quot;lat&quot;/&quot;lon&quot; (location) these are carried over into the calculations</p>
</td></tr>
<tr><td><code id="pops_to_DB_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>score_add_df</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with population and locus specific score information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_ &lt;- simulate_pops(pop_n = 4, aims_n = 50)
df_db &lt;- pops_to_DB(df_)
</code></pre>

<hr>
<h2 id='profile_AA_x0'>Function that compute the genotype probability for each population (rows in df)</h2><span id='topic+profile_AA_x0'></span>

<h3>Description</h3>

<p>Function that compute the genotype probability for each population (rows in df)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>profile_AA_x0(AA_profile, df, select = c("locus", "x0"),
  keep_dropped = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="profile_AA_x0_+3A_aa_profile">AA_profile</code></td>
<td>
<p>A tibble/data.frame with columns 'locus', 'A1' and 'A2' holding the separated version of a genotype, eg. AG -&gt; A1: A, A2: G</p>
</td></tr>
<tr><td><code id="profile_AA_x0_+3A_df">df</code></td>
<td>
<p>The database with main alleles per locus</p>
</td></tr>
<tr><td><code id="profile_AA_x0_+3A_select">select</code></td>
<td>
<p>Which columns to return</p>
</td></tr>
<tr><td><code id="profile_AA_x0_+3A_keep_dropped">keep_dropped</code></td>
<td>
<p>Logical. Keep the non-matching alleles (compared to 'db') and those with genotype 'NN'</p>
</td></tr>
</table>

<hr>
<h2 id='profile_admixture'>Compute the z-score (and more) for admixed hypotheses</h2><span id='topic+profile_admixture'></span>

<h3>Description</h3>

<p>Compute the z-score (and more) for admixed hypotheses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>profile_admixture(x0, df, hyp = NULL, grouping = "meta",
  return_all = FALSE, calc_logP = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="profile_admixture_+3A_x0">x0</code></td>
<td>
<p>A data frame/tibble with two columns: 'locus' and 'x0'</p>
</td></tr>
<tr><td><code id="profile_admixture_+3A_df">df</code></td>
<td>
<p>A tibble of reference profiles (as for 'genogeo')</p>
</td></tr>
<tr><td><code id="profile_admixture_+3A_hyp">hyp</code></td>
<td>
<p>If NULL all levels of 'grouping' is crossed and looped over as pairwise hypotheses. 
If a single level of 'grouping', this value is crossed with the remaining levels.
If vector of two levels this is the only tested hypothesis.</p>
</td></tr>
<tr><td><code id="profile_admixture_+3A_grouping">grouping</code></td>
<td>
<p>Should the calculations be for meta populations (&quot;meta&quot;) or sample populations (&quot;pop&quot;)?</p>
</td></tr>
<tr><td><code id="profile_admixture_+3A_return_all">return_all</code></td>
<td>
<p>Should z-score be returned (FALSE) or all locus results (TRUE)?</p>
</td></tr>
<tr><td><code id="profile_admixture_+3A_calc_logp">calc_logP</code></td>
<td>
<p>Should log P(Geno|Hyp) be calculated (TRUE) or not (FALSE)?</p>
</td></tr>
<tr><td><code id="profile_admixture_+3A_...">...</code></td>
<td>
<p>additional arguments passed on to other functions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble of z-scores, or a list of pairwise results if 'return_all = TRUE'
</p>

<hr>
<h2 id='random_AIMs_profile'>Simulate a random AIMs profile</h2><span id='topic+random_AIMs_profile'></span>

<h3>Description</h3>

<p>Use the information from <code>pops_to_DB</code> to simulate a profile from a random or given population.
The sampling is done with respect to the null hypothesis, such that the total count is adjusted accordingly.
For further details see Tvedebrink et al (2018), Section 3.1 (Simulations).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_AIMs_profile(df, grouping = "pop", population = NULL,
  n = FALSE, keep_pop = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="random_AIMs_profile_+3A_df">df</code></td>
<td>
<p>Database of reference profiles as returned by <code>pops_to_DB</code></p>
</td></tr>
<tr><td><code id="random_AIMs_profile_+3A_grouping">grouping</code></td>
<td>
<p>Simualte from <code>pop</code> (default) or <code>meta</code>.</p>
</td></tr>
<tr><td><code id="random_AIMs_profile_+3A_population">population</code></td>
<td>
<p>The population to sample from. If NULL chosen at random.</p>
</td></tr>
<tr><td><code id="random_AIMs_profile_+3A_n">n</code></td>
<td>
<p>Use numbers of samples as weights to choose the population randomly</p>
</td></tr>
<tr><td><code id="random_AIMs_profile_+3A_keep_pop">keep_pop</code></td>
<td>
<p>Keep information on population</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Torben Tvedebrink <a href="mailto:tvede@math.aau.dk">tvede@math.aau.dk</a>
</p>

<hr>
<h2 id='seldin_loci'>Seldin Lab markers</h2><span id='topic+seldin_loci'></span>

<h3>Description</h3>

<p>List of markers identified by Seldin lab.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seldin_loci
</code></pre>


<h3>Format</h3>

<p>List of 122 markers
</p>

<dl>
<dt>locus</dt><dd><p>Locus/Marker names</p>
</dd>
</dl>


<h3>Source</h3>

<p>Kosoy et al.  
Ancestry Informative Marker Sets for Determining Continental Origin and Admixture Proportions in Common Populations in America. 
HUMAN MUTATION, Vol. 30, No. 1, 69–78, 2009.
</p>

<hr>
<h2 id='simulate_pops'>Simulate random populations</h2><span id='topic+simulate_pops'></span>

<h3>Description</h3>

<p>Simulate random populations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_pops(pop_n = 100, pop_names = NULL, pop_totals = NULL,
  aims_n = 50, aims_names = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_pops_+3A_pop_n">pop_n</code></td>
<td>
<p>Number of populations to simulate</p>
</td></tr>
<tr><td><code id="simulate_pops_+3A_pop_names">pop_names</code></td>
<td>
<p>Their names. If NULL: The names are &quot;pop_001&quot; through &quot;pop_pop_n&quot;</p>
</td></tr>
<tr><td><code id="simulate_pops_+3A_pop_totals">pop_totals</code></td>
<td>
<p>How many observations/sampled individuals per population. If one number this is used as parameter in a Poisson distribution</p>
</td></tr>
<tr><td><code id="simulate_pops_+3A_aims_n">aims_n</code></td>
<td>
<p>Number of AIMs</p>
</td></tr>
<tr><td><code id="simulate_pops_+3A_aims_names">aims_names</code></td>
<td>
<p>Their names. If NULL: The names are &quot;rs_001&quot; through &quot;rs_aims_n&quot;</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Torben Tvedebrink <a href="mailto:tvede@math.aau.dk">tvede@math.aau.dk</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
