<!DOCTYPE html><html><head><title>Help for package GenoScan</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GenoScan}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GenoScan.example'>
<p>Data example for GenoScan (A Genome-Wide Scan Statistic Framework For Whole-Genome Sequence Data Analysis)</p></a></li>
<li><a href='#GenoScan.info'>
<p>hg19 chromosome sizes</p></a></li>
<li><a href='#GenoScan.prelim'>
<p>The preliminary data management for GenoScan</p></a></li>
<li><a href='#GenoScan.Region'>
<p>Scan the association between an quantitative/dichotomous outcome variable and a region by score type statistics allowing for multiple functional annotation scores.</p></a></li>
<li><a href='#GenoScan.SingleWindow'>
<p>Test the association between an quantitative/dichotomous outcome variable and a single window by dispersion or burden test allowing for multiple functional annotation scores.</p></a></li>
<li><a href='#GenoScan.VCF.chr'>
<p>Scan a VCF file to study the association between an quantitative/dichotomous outcome variable and a region or whole chromosome by score type statistics allowing for multiple functional annotation scores.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Genome-Wide Scan Statistic Framework for Whole-Genome Sequence
Data Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-11-13</td>
</tr>
<tr>
<td>Author:</td>
<td>Zihuai He</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zihuai He &lt;zihuai@stanford.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for whole-genome sequencing studies, including genome-wide scan, candidate region scan and single window test.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.10.0), SKAT, Matrix, MASS, seqminer, data.table</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-12-11 18:20:45 UTC; zhe</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-12-21 15:20:16 UTC</td>
</tr>
</table>
<hr>
<h2 id='GenoScan.example'>
Data example for GenoScan (A Genome-Wide Scan Statistic Framework For Whole-Genome Sequence Data Analysis)
</h2><span id='topic+GenoScan.example'></span>

<h3>Description</h3>

<p>The dataset contains outcome variable Y, covariate X, genotype data G, positions of genetic variants pos, weight matrix for functional annotations Z.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GenoScan.example)</code></pre>

<hr>
<h2 id='GenoScan.info'>
hg19 chromosome sizes
</h2><span id='topic+GenoScan.info'></span>

<h3>Description</h3>

<p>The dataset contains hg19 chromosome sizes from:
</p>
<p>http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GenoScan.info)</code></pre>

<hr>
<h2 id='GenoScan.prelim'>
The preliminary data management for GenoScan
</h2><span id='topic+GenoScan.prelim'></span>

<h3>Description</h3>

<p>This function does the preliminary data management and fit the model under null hypothesis. The output will be used in the other GenoScan functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenoScan.prelim(Y, X=NULL, id=NULL, out_type="C", B=5000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenoScan.prelim_+3A_y">Y</code></td>
<td>

<p>The outcome variable, an n*1 matrix where n is the total number of observations
</p>
</td></tr>
<tr><td><code id="GenoScan.prelim_+3A_x">X</code></td>
<td>

<p>An n*d covariates matrix where d is the total number of covariates.
</p>
</td></tr>
<tr><td><code id="GenoScan.prelim_+3A_id">id</code></td>
<td>

<p>The subject id. This is used to match phenotype with genotype. The default is NULL, where the matched phenotype and genotype matrices are assumed.
</p>
</td></tr>
<tr><td><code id="GenoScan.prelim_+3A_out_type">out_type</code></td>
<td>

<p>Type of outcome variable. Can be either &quot;C&quot; for continuous or &quot;D&quot; for dichotomous. The default is &quot;C&quot;.
</p>
</td></tr>
<tr><td><code id="GenoScan.prelim_+3A_b">B</code></td>
<td>

<p>Number of resampling replicates. The default is 5000. A larger value leads to more accurate and stable p-value calculation, but requires more computing time.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list used for function GenoScan.Region(), GenoScan.SingleWindow() and GenoScan.VCF.chr().





</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GenoScan)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by d matrix
# G: genotype matrix, n by p matrix where n is the total number of subjects
# Z: functional annotation matrix, p by q matrix

data(GenoScan.example)
Y&lt;-GenoScan.example$Y;X&lt;-GenoScan.example$X;G&lt;-GenoScan.example$G;Z&lt;-GenoScan.example$Z

# Preliminary data management
result.prelim&lt;-GenoScan.prelim(Y,X=X,out_type="C",B=5000)

</code></pre>

<hr>
<h2 id='GenoScan.Region'>
Scan the association between an quantitative/dichotomous outcome variable and a region by score type statistics allowing for multiple functional annotation scores.
</h2><span id='topic+GenoScan.Region'></span>

<h3>Description</h3>

<p>Once the preliminary work is done by &quot;GenoScan.prelim()&quot;, this function scan a target region. This function is often used for candidate region analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenoScan.Region(result.prelim,G,pos,Gsub.id=NULL,Z=NULL,MAF.weights='beta',
test='combined',window.size=c(5000,10000,15000,20000,25000,50000),MAF.threshold=1,
impute.method='fixed')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenoScan.Region_+3A_result.prelim">result.prelim</code></td>
<td>

<p>The output of function &quot;GenoScan.prelim()&quot;
</p>
</td></tr>
<tr><td><code id="GenoScan.Region_+3A_g">G</code></td>
<td>

<p>Genetic variants in the target region, an n*p matrix where n is the subject ID and p is the total number of genetic variants.
</p>
</td></tr>
<tr><td><code id="GenoScan.Region_+3A_pos">pos</code></td>
<td>

<p>The positions of genetic variants, an p dimensional vector. Each position corresponds to a column in the genotype matrix.
</p>
</td></tr>
<tr><td><code id="GenoScan.Region_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an n dimensional vector. Each ID corresponds to a row in the genotype matrix. This is used to match phenotype with genotype. The default is NULL, where the matched phenotype and genotype matrices are assumed.
</p>
</td></tr>
<tr><td><code id="GenoScan.Region_+3A_z">Z</code></td>
<td>

<p>Weight matrix for functional annotations, an p*q matrix where p is the total number of genetic variables and q is the number of weights. This is used to incorperate functional annotations. The default is NULL, where minor allele frequency weighted (see MAF.weights) dispersion and/or burden tests are applied.
</p>
</td></tr>
<tr><td><code id="GenoScan.Region_+3A_maf.weights">MAF.weights</code></td>
<td>

<p>Minor allele frequency based weight. Can be 'beta' to up-weight rare variants or 'equal' for a flat weight. The default is 'beta'.
</p>
</td></tr>
<tr><td><code id="GenoScan.Region_+3A_test">test</code></td>
<td>

<p>Can be 'dispersion', 'burden' or 'combined'. The test is 'combined', both dispersion and burden tests are applied. The default is 'combined'.
</p>
</td></tr>
<tr><td><code id="GenoScan.Region_+3A_window.size">window.size</code></td>
<td>

<p>Candidate window sizes in base pairs. The default is c(5000,10000,15000,20000,25000,50000). Note that extemely small window size (e.g. 1) requires large sample size.
</p>
</td></tr>
<tr><td><code id="GenoScan.Region_+3A_maf.threshold">MAF.threshold</code></td>
<td>

<p>Threshold for minor allele frequency. Variants above MAF.threshold are ignored. The default is 1.
</p>
</td></tr>
<tr><td><code id="GenoScan.Region_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;bestguess&quot; uses the genotype with highest probability.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>n.marker</code></td>
<td>

<p>Number of tested variants in the window (heterozygous variants below MAF threshold).
</p>
</td></tr>
<tr><td><code>window.summary</code></td>
<td>

<p>Results for all windows. Each row presents a window, including chromosome number, start position, end position,dispersion p-value(s), burden p-values(s).
</p>
</td></tr>
<tr><td><code>M</code></td>
<td>

<p>Estimated number of effective tests.
</p>
</td></tr>
<tr><td><code>threshold</code></td>
<td>

<p>Estimated threshold, 0.05/M. This threshold is for windows tested in this particular region.
</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>

<p>P-value of entire region.
</p>
</td></tr>





</table>


<h3>Examples</h3>

<pre><code class='language-R'>## GenoScan.prelim does the preliminary data management.
# Input: Y, X (covariates)
## GenoScan.Region scans a region.
# Input: G (genetic variants), pos (position) Z (weights) and result of GenoScan.prelim

library(GenoScan)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by d matrix
# G: genotype matrix, n by p matrix where n is the total number of subjects
# pos: positions of genetic variants, p dimention vector
# Z: functional annotation matrix, p by q matrix

data(GenoScan.example)
Y&lt;-GenoScan.example$Y;X&lt;-GenoScan.example$X
G&lt;-GenoScan.example$G;pos&lt;-GenoScan.example$pos
Z&lt;-GenoScan.example$Z

# Preliminary data management
result.prelim&lt;-GenoScan.prelim(Y,X=X,out_type="C",B=5000)

# Scan the region with functional annotations defined in Z
result&lt;-GenoScan.Region(result.prelim,G,pos,Z=Z)
</code></pre>

<hr>
<h2 id='GenoScan.SingleWindow'>
Test the association between an quantitative/dichotomous outcome variable and a single window by dispersion or burden test allowing for multiple functional annotation scores.
</h2><span id='topic+GenoScan.SingleWindow'></span>

<h3>Description</h3>

<p>Once the preliminary work is done by &quot;GenoScan.prelim()&quot;, this function tests a single window. This is often used to double-check significant windows identified by GenoScan.Region or GenoScan.VCF.chr, with an increased number of resampling replicates in GenoScan.prelim.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenoScan.SingleWindow(result.prelim,G,Gsub.id=NULL,Z=NULL,MAF.weights='beta',
test='combined',MAF.threshold=1,impute.method='fixed')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenoScan.SingleWindow_+3A_result.prelim">result.prelim</code></td>
<td>

<p>The output of function &quot;GenoScan.prelim()&quot;
</p>
</td></tr>
<tr><td><code id="GenoScan.SingleWindow_+3A_g">G</code></td>
<td>

<p>Genetic variants in the target region, an n*p matrix where n is the subject ID and p is the total number of genetic variants.
</p>
</td></tr>
<tr><td><code id="GenoScan.SingleWindow_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an n dimensional vector. Each ID corresponds to a row in the genotype matrix. This is used to match phenotype with genotype. The default is NULL, where the matched phenotype and genotype matrices are assumed.
</p>
</td></tr>
<tr><td><code id="GenoScan.SingleWindow_+3A_z">Z</code></td>
<td>

<p>Weight matrix for functional annotations, an p*q matrix where p is the total number of genetic variables and q is the number of weights. This is used to incorperate functional annotations. The default is NULL, where minor allele frequency weighted (see MAF.weights) dispersion and/or burden tests are applied.
</p>
</td></tr>
<tr><td><code id="GenoScan.SingleWindow_+3A_maf.weights">MAF.weights</code></td>
<td>

<p>Minor allele frequency based weight. Can be 'beta' to up-weight rare variants or 'equal' for a flat weight. The default is 'beta'.
</p>
</td></tr>
<tr><td><code id="GenoScan.SingleWindow_+3A_test">test</code></td>
<td>

<p>Can be 'dispersion', 'burden' or 'combined'. The test is 'combined', both dispersion and burden tests are applied. The default is 'combined'.
</p>
</td></tr>
<tr><td><code id="GenoScan.SingleWindow_+3A_maf.threshold">MAF.threshold</code></td>
<td>

<p>Threshold for minor allele frequency. Variants above MAF.threshold are ignored. The default is 1.
</p>
</td></tr>
<tr><td><code id="GenoScan.SingleWindow_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;bestguess&quot; uses the genotype with highest probability.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>n.marker</code></td>
<td>

<p>Number of tested variants in the window (heterozygous variants below MAF threshold).
</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>

<p>P-value(s) of the window (dispersion p-value(s), then burden p-values(s))
</p>
</td></tr>





</table>


<h3>Examples</h3>

<pre><code class='language-R'>## GenoScan.prelim does the preliminary data management.
# Input: Y, X (covariates)
## GenoScan.Region scans a region.
# Input: G (genetic variants), pos (position) Z (weights) and result of GenoScan.prelim

library(GenoScan)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by d matrix
# G: genotype matrix, n by p matrix where n is the total number of subjects
# pos: positions of genetic variants, p dimention vector
# Z: functional annotation matrix, p by q matrix

data(GenoScan.example)
Y&lt;-GenoScan.example$Y;X&lt;-GenoScan.example$X
G&lt;-GenoScan.example$G;pos&lt;-GenoScan.example$pos
Z&lt;-GenoScan.example$Z

# Preliminary data management
result.prelim&lt;-GenoScan.prelim(Y,X=X,out_type="C",B=5000)

# Scan the region with functional annotations defined in Z
result&lt;-GenoScan.SingleWindow(result.prelim,G,Z=Z)
</code></pre>

<hr>
<h2 id='GenoScan.VCF.chr'>
Scan a VCF file to study the association between an quantitative/dichotomous outcome variable and a region or whole chromosome by score type statistics allowing for multiple functional annotation scores.
</h2><span id='topic+GenoScan.VCF.chr'></span>

<h3>Description</h3>

<p>Once the preliminary work is done by &quot;GenoScan.prelim()&quot;, this function scan a target region or chromosome, and output results for all windows as well as an estimated significance threshold. For genome-wide scan, users can scan each chromosome individually, then the genome-wide significance threshold can be obtained by combining chromosome-wise thresholds:
</p>
<p>alpha=1/(1/alpha_1+1/alpha_2+...+1/alpha_22).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenoScan.VCF.chr(result.prelim,vcf.filename,chr,pos.min=NULL,pos.max=NULL,
Gsub.id=NULL,annot.filename=NULL,cell.type=NULL,MAF.weights='beta',
test='combined',window.size=c(5000,10000,15000,20000,25000,50000),
MAF.threshold=1,impute.method='fixed')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenoScan.VCF.chr_+3A_result.prelim">result.prelim</code></td>
<td>

<p>The output of function &quot;GenoScan.prelim()&quot;
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_vcf.filename">vcf.filename</code></td>
<td>

<p>A character specifying the directory (including the file name) of the vcf file.
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_chr">chr</code></td>
<td>

<p>Chromosome number.
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_pos.min">pos.min</code></td>
<td>

<p>Minimum position of the scan. The default is NULL, where the scan starts at the first base pair.
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_pos.max">pos.max</code></td>
<td>

<p>Maximum position of the scan. The default is NULL, where the scan ends at the last base pair, according to the chromosome sizes at:
</p>
<p>http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes.
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an n dimensional vector. This is used to match phenotype with genotype. The default is NULL, where the subject id in the vcf file is used.
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_annot.filename">annot.filename</code></td>
<td>

<p>A character specifying the directory (including the file name) of functional annotations. Currently GenoScan supports GenoNet scores across 127 tissues/cell types, which can be downloaded at:
</p>
<p>http://www.openbioinformatics.org/annovar/download/GenoNetScores/
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_cell.type">cell.type</code></td>
<td>

<p>A character specifying the tissue/cell type integrated in the analysis, in addition to standard dispersion and/or burden tests. The default is NULL, where no functional annotation is included. If cell.type='all', GenoNet scores across all 127 tissues/cell types are incorperated.
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_maf.weights">MAF.weights</code></td>
<td>

<p>Minor allele frequency based weight. Can be 'beta' to up-weight rare variants or 'equal' for a flat weight. The default is 'beta'.
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_test">test</code></td>
<td>

<p>Can be 'dispersion', 'burden' or 'combined'. The test is 'combined', both dispersion and burden tests are applied. The default is 'combined'.
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_window.size">window.size</code></td>
<td>

<p>Candidate window sizes in base pairs. The default is c(5000,10000,15000,20000,25000,50000). Note that extemely small window size (e.g. 1) requires large sample size.
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_maf.threshold">MAF.threshold</code></td>
<td>

<p>Threshold for minor allele frequency. Variants above MAF.threshold are ignored. The default is 1.
</p>
</td></tr>
<tr><td><code id="GenoScan.VCF.chr_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;bestguess&quot; uses the genotype with highest probability.
</p>
</td></tr>



</table>


<h3>Value</h3>

<table>
<tr><td><code>window.summary</code></td>
<td>

<p>Results for all windows. Each row presents a window.
</p>
</td></tr>
<tr><td><code>M</code></td>
<td>

<p>Estimated number of effective tests.
</p>
</td></tr>
<tr><td><code>threshold</code></td>
<td>

<p>Estimated threshold, 0.05/M.
</p>
</td></tr>





</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# load example vcf file from package "seqminer"
vcf.filename = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")

# simulated outcomes, covariates and inidividual id.
Y&lt;-as.matrix(rnorm(3,0,1))
X&lt;-as.matrix(rnorm(3,0,1))
id&lt;-c("NA12286", "NA12341", "NA12342")

# fit null model
result.prelim&lt;-GenoScan.prelim(Y,X=X,id=id,out_type="C",B=5000)

# scan the vcf file
result&lt;-GenoScan.VCF.chr(result.prelim,vcf.filename,chr=1,pos.min=196621007,pos.max=196716634)


## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
