<!DOCTYPE html><html><head><title>Help for package genpwr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {genpwr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add.fun.t'><p>Function to Calculate t matrix for logistic outcome with binary environment interaction in additive model</p></a></li>
<li><a href='#add.or.function'><p>Additive Model Function</p></a></li>
<li><a href='#additive.ll'><p>Function to Calculate Additive Log Likelihood for a Logistic Regression Model</p></a></li>
<li><a href='#additive.ll.linear'><p>Function to Calculate Additive Log Likelihood for a Linear Regression Model</p></a></li>
<li><a href='#as.numeric2'><p>Function to convert to numeric with scientific notation containing the &quot;.&quot; character</p></a></li>
<li><a href='#calc.like'><p>Function to Calculate Log Likelihood for a Logistic Regression Model</p></a></li>
<li><a href='#calc.like.linear'><p>Function to Calculate Log Likelihood for a Linear Regression Model</p></a></li>
<li><a href='#calc.like.linear.log.envir.interaction'><p>Function to calculate the standard deviation of y given x for linear models with logistic environment interaction</p></a></li>
<li><a href='#df2.ll'><p>Function to Calculate 2df Log Likelihood for a Logistic Regression Model</p></a></li>
<li><a href='#df2.ll.linear'><p>Function to Calculate 2 Degree of Freedom Log Likelihood for a Linear Regression Model</p></a></li>
<li><a href='#dom.fun.t'><p>Function to Calculate t matrix for logistic outcome with binary environment interaction in dominant model</p></a></li>
<li><a href='#dom.or.function'><p>Dominant Model Function</p></a></li>
<li><a href='#dominant.ll'><p>Function to Calculate Dominant Log Likelihood for a Logistic Regression Model</p></a></li>
<li><a href='#dominant.ll.linear'><p>Function to Calculate Dominant Log Likelihood for a Linear Regression Model</p></a></li>
<li><a href='#es.calc.linear'><p>Function to Calculate Effect Size for Linear Models</p></a></li>
<li><a href='#expected.linear.ll'><p>Function to Calculate Expected Log Likelihood for a Single Genotype</p></a></li>
<li><a href='#expected.linear.ll.lin.env'><p>Function to Calculate Expected Log Likelihood for a Single Genotype with linear environment interaction</p></a></li>
<li><a href='#find.prob.dom'><p>Dominant probability finding function</p></a></li>
<li><a href='#find.prob.rec'><p>Recessive probability finding function</p></a></li>
<li><a href='#genpwr.calc'><p>Function to Calculate Power for Linear Models with logistic environment interaction</p></a></li>
<li><a href='#integrand_funct_case'><p>Function to generate integrand for mle for cases</p></a></li>
<li><a href='#integrand_funct_control'><p>Function to generate integrand for mle for controls</p></a></li>
<li><a href='#linear.mles'><p>Function to calculate MLE's for linear models</p></a></li>
<li><a href='#linear.mles.lin.envir.interaction'><p>Function to calculate the standard deviation of y given x for linear models with linear environment interaction</p></a></li>
<li><a href='#linear.mles.lin.envir.interaction_reduced'><p>Function to calculate the standard deviation of y given x for linear models with linear environment interaction</p>
for the reduced model without GxE interaction</a></li>
<li><a href='#linear.mles.log.envir.interaction'><p>Function to calculate the standard deviation of y given x for linear models with logistic environment interaction</p></a></li>
<li><a href='#linear.outcome.lin.envir.interaction.sds'><p>Function to calculate the standard deviation of y given x for linear models with linear environment interaction</p></a></li>
<li><a href='#linear.outcome.lin.envir.interaction.sds_reduced'><p>Function to calculate the standard deviation of y given x for linear models with linear environment interaction</p></a></li>
<li><a href='#linear.outcome.log.envir.interaction.sds'><p>Function to calculate the standard deviation of y given x for linear models with logistic environment interaction</p></a></li>
<li><a href='#linear.sds'><p>Functions to Calculate Residual SD for Normal/Continuous Outcomes</p>
Function to calculate the standard deviation of y given x for linear models</a></li>
<li><a href='#ll_zero_finder2'><p>Zero finding function</p></a></li>
<li><a href='#ll.ge.logistic'><p>Function to calculate MLE's for logistic models with logistic environment interaction</p></a></li>
<li><a href='#ll.ge.logistic.lin.envir'><p>Function to output log likelihood for logistic outcome with linear environment variables</p></a></li>
<li><a href='#ll.linear.selector'><p>Function to return log likelihood function for specified model type</p></a></li>
<li><a href='#logistic.mles'><p>Function to calculate MLE's for logistic models</p></a></li>
<li><a href='#logit'><p>Logit Function</p></a></li>
<li><a href='#ncp.search'><p>Function to Determine Non-Centrality Parameter of the Chi-squared distribution</p></a></li>
<li><a href='#null.ll'><p>Function to Calculate Null Log Likelihood for a Logistic Regression Model</p></a></li>
<li><a href='#null.ll.linear'><p>Function to Calculate Expected Null Log Likelihood for a Linear Regression Model</p></a></li>
<li><a href='#odds_ratio_function'><p>Odds Ratio Function</p></a></li>
<li><a href='#or_calc'><p>Odds ratio calculation</p></a></li>
<li><a href='#or.function.2df'><p>2df Model Function</p></a></li>
<li><a href='#or.plot'><p>Function to Plot Odds Ratio Results</p></a></li>
<li><a href='#p_vec_returner'><p>Function to output probability vector used in calculation of MLE's for linear outcome with logistic environment interaction</p></a></li>
<li><a href='#p_vec_returner_lin_env'><p>Function to output probability vector used in calculation of MLE's for linear outcome with linear environment interaction</p></a></li>
<li><a href='#power_envir.calc'><p>Function to Calculate Power for Logistic Models with Environment Interaction</p></a></li>
<li><a href='#power_envir.calc.linear_outcome'><p>Function to Calculate Power for Linear Models with logistic environment interaction</p></a></li>
<li><a href='#power_linear_envir.calc.linear_outcome'><p>Function to Calculate Power for Linear Models with linear environment interaction</p></a></li>
<li><a href='#power_linear_envir.calc.logistic_outcome'><p>Function to Calculate Power for Linear Models with logistic environment interaction</p></a></li>
<li><a href='#power.calc'><p>Function to Calculate Power</p></a></li>
<li><a href='#power.calc.linear'><p>Function to Calculate Power for Linear Models</p></a></li>
<li><a href='#power.plot'><p>Function to Plot Power Results</p></a></li>
<li><a href='#quad_roots'><p>Function to Solve Quadratic Equations</p></a></li>
<li><a href='#rec.fun.t'><p>Function to Calculate t matrix for logistic outcome with binary environment interaction in recessive model</p></a></li>
<li><a href='#rec.or.function'><p>Recessive Model Function</p></a></li>
<li><a href='#recessive.ll'><p>Function to Calculate Recessive Log Likelihood for a Logistic Regression Model</p></a></li>
<li><a href='#recessive.ll.linear'><p>Function to Calculate Recessive Log Likelihood for a Linear Regression Model</p></a></li>
<li><a href='#solve_a'><p>Binomial coefficient calculation</p></a></li>
<li><a href='#ss_envir.calc'><p>Function to Calculate Power for Logistic Models with Environment Interaction</p></a></li>
<li><a href='#ss_envir.calc.linear_outcome'><p>Function to Calculate Power for Linear Models with logistic environment interaction</p></a></li>
<li><a href='#ss_linear_envir.calc.linear_outcome'><p>Function to Calculate Power for Linear Models with linear environment interaction</p></a></li>
<li><a href='#ss_linear_envir.calc.logistic_outcome'><p>Function to Calculate Sample Size for Linear Models with logistic environment interaction</p></a></li>
<li><a href='#ss.calc'><p>Function to Calculate Sample Size</p></a></li>
<li><a href='#ss.calc.linear'><p>Function to Calculate Sample Size in Linear Models</p></a></li>
<li><a href='#ss.plot'><p>Function to Plot Sample Size Results</p></a></li>
<li><a href='#X_mat_returner'><p>Function to output X matrices used in calculation of MLE's for linear outcome with logistic environment interaction</p></a></li>
<li><a href='#X_mat_returner_lle'><p>Function to output X matrices used in calculation of MLE's for linear outcome with linear environment interaction</p></a></li>
<li><a href='#zero_finder_nleqslv'><p>Zero finder</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Power Calculations Under Genetic Model Misspecification</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Power and sample size calculations for genetic association studies allowing 
    for misspecification of the model of genetic susceptibility.
    "Hum Hered. 2019;84(6):256-271.&lt;<a href="https://doi.org/10.1159%2F000508558">doi:10.1159/000508558</a>&gt;. Epub 2020 Jul 28."
    Power and/or sample size can be calculated for logistic (case/control study design) 
    and linear (continuous phenotype) regression models, using additive, dominant, 
    recessive or degree of freedom coding of the genetic covariate while assuming 
    a true dominant, recessive or additive genetic effect. In addition, power and 
    sample size calculations can be performed for gene by environment interactions.  
    These methods are extensions of Gauderman (2002) 
    &lt;<a href="https://doi.org/10.1093%2Faje%2F155.5.478">doi:10.1093/aje/155.5.478</a>&gt; and Gauderman (2002) &lt;<a href="https://doi.org/10.1002%2Fsim.973">doi:10.1002/sim.973</a>&gt;
    and are described in: 
    Moore CM, Jacobson S, Fingerlin TE. Power and Sample Size Calculations 
    for Genetic Association Studies in the Presence of Genetic Model Misspecification. 
    American Society of Human Genetics. 
    October 2018, San Diego. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, nleqslv, MASS, stats, utils</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-03-30 14:57:52 UTC; jacobsons</td>
</tr>
<tr>
<td>Author:</td>
<td>Camille Moore [aut, cre],
  Sean Jacobson [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Camille Moore &lt;moorec@njhealth.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-03-31 00:00:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='add.fun.t'>Function to Calculate t matrix for logistic outcome with binary environment interaction in additive model</h2><span id='topic+add.fun.t'></span>

<h3>Description</h3>

<p>Calculates the t matrix for logistic outcome with binary environment interaction in additive model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.fun.t(MAF, P_e, OR_E, OR_G, OR_GE, Case.Rate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.fun.t_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="add.fun.t_+3A_p_e">P_e</code></td>
<td>
<p>Vector of proportions of the population with exposure to the environmental effect</p>
</td></tr>
<tr><td><code id="add.fun.t_+3A_or_e">OR_E</code></td>
<td>
<p>Vector of environmental odds ratios to detect</p>
</td></tr>
<tr><td><code id="add.fun.t_+3A_or_g">OR_G</code></td>
<td>
<p>Vector of genetic odds ratios to detect</p>
</td></tr>
<tr><td><code id="add.fun.t_+3A_or_ge">OR_GE</code></td>
<td>
<p>Vector of genetic/environmental interaction odds ratios to detect</p>
</td></tr>
<tr><td><code id="add.fun.t_+3A_case.rate">Case.Rate</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>t matrix for all combinations of environment/outcome
</p>


<h3>Examples</h3>

<pre><code class='language-R'>add.fun.t(MAF = 0.1, P_e = 0.2, Case.Rate = 0.5,
	OR_G = 1.5, OR_E = 2, OR_GE = 1.8)

</code></pre>

<hr>
<h2 id='add.or.function'>Additive Model Function</h2><span id='topic+add.or.function'></span>

<h3>Description</h3>

<p>Operates within odds_ratio_function to calculate odds ratios for a Test.Model of &quot;Additive&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.or.function(like, Case.Rate, P_AA, P_AB, P_BB, True.Model, risk_allele)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add.or.function_+3A_like">like</code></td>
<td>
<p>Expected log likelihood</p>
</td></tr>
<tr><td><code id="add.or.function_+3A_case.rate">Case.Rate</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
<tr><td><code id="add.or.function_+3A_p_aa">P_AA</code></td>
<td>
<p>Probability the allele is homozygous for the major allele</p>
</td></tr>
<tr><td><code id="add.or.function_+3A_p_ab">P_AB</code></td>
<td>
<p>Probability the allele is heterozygous</p>
</td></tr>
<tr><td><code id="add.or.function_+3A_p_bb">P_BB</code></td>
<td>
<p>Probability the allele is homozygous for the minor allele</p>
</td></tr>
<tr><td><code id="add.or.function_+3A_true.model">True.Model</code></td>
<td>
<p>A vector object specifying the true underlying genetic model(s): 'Dominant', 'Additive', or 'Recessive'</p>
</td></tr>
<tr><td><code id="add.or.function_+3A_risk_allele">risk_allele</code></td>
<td>
<p>Logical: If OR &gt; 1, the allele is classified as a &quot;risk allele&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>: The odds ratios and their corresponding genetic model(s)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>add.or.function(like=-0.57162, Case.Rate=0.3, P_AA=0.5625, P_AB=0.375, 
 P_BB=0.0625, True.Model="Additive", risk_allele=TRUE)

</code></pre>

<hr>
<h2 id='additive.ll'>Function to Calculate Additive Log Likelihood for a Logistic Regression Model</h2><span id='topic+additive.ll'></span>

<h3>Description</h3>

<p>Calculates the log likelihood for a given set of logistic regression coefficients under an additive genetic model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>additive.ll(beta, t)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="additive.ll_+3A_beta">beta</code></td>
<td>
<p>Vector of logistic regression coefficients.</p>
</td></tr>
<tr><td><code id="additive.ll_+3A_t">t</code></td>
<td>
<p>A 2x3 table of joint probabilities of disease and genotype.  Rows = case vs. control, columns=genotype.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>additive.ll(c(-0.3793525, -1.1395417), 
 rbind(c(0.2339079, 0.05665039, 0.009441731), 
 c(0.3285921, 0.31834961, 0.053058269)))

</code></pre>

<hr>
<h2 id='additive.ll.linear'>Function to Calculate Additive Log Likelihood for a Linear Regression Model</h2><span id='topic+additive.ll.linear'></span>

<h3>Description</h3>

<p>Calculates the log likelihood for a given set of linear regression coefficients under an additive genetic model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>additive.ll.linear(beta, m, es, sd_y_x_model, sd_y_x_truth)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="additive.ll.linear_+3A_beta">beta</code></td>
<td>
<p>Vector of linear regression coefficients.</p>
</td></tr>
<tr><td><code id="additive.ll.linear_+3A_m">m</code></td>
<td>
<p>Minor allele frequency.</p>
</td></tr>
<tr><td><code id="additive.ll.linear_+3A_es">es</code></td>
<td>
<p>Vector of effect sizes with two elements, (mean AB - mean AA) and (mean BB - mean AA).</p>
</td></tr>
<tr><td><code id="additive.ll.linear_+3A_sd_y_x_model">sd_y_x_model</code></td>
<td>
<p>The standard deviation of Y (the outcome) given X (predictors/genotype) under the test model.</p>
</td></tr>
<tr><td><code id="additive.ll.linear_+3A_sd_y_x_truth">sd_y_x_truth</code></td>
<td>
<p>The standard deviation of Y given X (predictors/genotype) given genotype under the true model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>additive.ll.linear(beta = c(-0.03, 0.3), m = 0.1, es = c(0,3), 
 sd_y_x_model = 0.9918669, sd_y_x_truth = 0.9544108)

</code></pre>

<hr>
<h2 id='as.numeric2'>Function to convert to numeric with scientific notation containing the &quot;.&quot; character</h2><span id='topic+as.numeric2'></span>

<h3>Description</h3>

<p>convert to numeric with scientific notation containing the &quot;.&quot; character
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.numeric2(char)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.numeric2_+3A_char">char</code></td>
<td>
<p>string to be converted to numeric</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a number
</p>


<h3>Examples</h3>

<pre><code class='language-R'>as.numeric2("2e.2")

</code></pre>

<hr>
<h2 id='calc.like'>Function to Calculate Log Likelihood for a Logistic Regression Model</h2><span id='topic+calc.like'></span>

<h3>Description</h3>

<p>Convenience function to calculate the log likelihood of a specified model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.like(beta, t, model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc.like_+3A_beta">beta</code></td>
<td>
<p>Vector of logistic regression coefficients.</p>
</td></tr>
<tr><td><code id="calc.like_+3A_t">t</code></td>
<td>
<p>A 2x3 table of joint probabilities of disease and genotype.  Rows = case vs. control, columns=genotype.</p>
</td></tr>
<tr><td><code id="calc.like_+3A_model">model</code></td>
<td>
<p>The genetic model in the logistic regression: &quot;Dominant&quot;, &quot;Additive&quot;, &quot;Recessive&quot;, &quot;2df&quot; or &quot;null&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>t &lt;- rbind(c(0.2967437, 0.1806723, 0.02258404), 
 c(0.3432563, 0.1393277, 0.01741596))
calc.like(logistic.mles(t, "Dominant"), t, model="Dominant")

</code></pre>

<hr>
<h2 id='calc.like.linear'>Function to Calculate Log Likelihood for a Linear Regression Model</h2><span id='topic+calc.like.linear'></span>

<h3>Description</h3>

<p>Convenience function to calculate the log likelihood of a specified model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.like.linear(beta, m, es_ab, es_bb, sd_y_x_model, sd_y_x_truth, model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc.like.linear_+3A_beta">beta</code></td>
<td>
<p>Vector of linear regression coefficients.</p>
</td></tr>
<tr><td><code id="calc.like.linear_+3A_m">m</code></td>
<td>
<p>Minor allele frequency.</p>
</td></tr>
<tr><td><code id="calc.like.linear_+3A_es_ab">es_ab</code></td>
<td>
<p>effect size for mean AB - mean AA</p>
</td></tr>
<tr><td><code id="calc.like.linear_+3A_es_bb">es_bb</code></td>
<td>
<p>effect size for mean BB - mean AA</p>
</td></tr>
<tr><td><code id="calc.like.linear_+3A_sd_y_x_model">sd_y_x_model</code></td>
<td>
<p>The standard deviation of Y (the outcome) given X (predictors/genotype) under the test model.</p>
</td></tr>
<tr><td><code id="calc.like.linear_+3A_sd_y_x_truth">sd_y_x_truth</code></td>
<td>
<p>The standard deviation of Y given X (predictors/genotype) given genotype under the true model.</p>
</td></tr>
<tr><td><code id="calc.like.linear_+3A_model">model</code></td>
<td>
<p>The genetic model in the linear regression: &quot;Dominant&quot;, &quot;Additive&quot;, &quot;Recessive&quot;, &quot;2df&quot; or &quot;null&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>calc.like.linear(beta = c(0.0000000, 0.1578947), m = 0.1, es_ab = 0, es_bb = 3, 
 sd_y_x_model = 0.9980797, sd_y_x_truth = 0.9544108, model = "Dominant")

</code></pre>

<hr>
<h2 id='calc.like.linear.log.envir.interaction'>Function to calculate the standard deviation of y given x for linear models with logistic environment interaction</h2><span id='topic+calc.like.linear.log.envir.interaction'></span>

<h3>Description</h3>

<p>Returns the standard deviation of y given x for linear models with logistic environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.like.linear.log.envir.interaction(
  beta_hat,
  MAF,
  P_e,
  ES_G,
  ES_E,
  ES_GE,
  sd_y_x_truth,
  sd_y_x_model,
  Test.Model,
  True.Model,
  reduced = F
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc.like.linear.log.envir.interaction_+3A_beta_hat">beta_hat</code></td>
<td>
<p>Effect sizes from MLE</p>
</td></tr>
<tr><td><code id="calc.like.linear.log.envir.interaction_+3A_maf">MAF</code></td>
<td>
<p>Minor allele Frequency</p>
</td></tr>
<tr><td><code id="calc.like.linear.log.envir.interaction_+3A_p_e">P_e</code></td>
<td>
<p>Population prevalence of logistic environmental factor</p>
</td></tr>
<tr><td><code id="calc.like.linear.log.envir.interaction_+3A_es_g">ES_G</code></td>
<td>
<p>Genetic Effect size</p>
</td></tr>
<tr><td><code id="calc.like.linear.log.envir.interaction_+3A_es_e">ES_E</code></td>
<td>
<p>Environment Effect size</p>
</td></tr>
<tr><td><code id="calc.like.linear.log.envir.interaction_+3A_es_ge">ES_GE</code></td>
<td>
<p>Environment x Genetic interaction Effect size</p>
</td></tr>
<tr><td><code id="calc.like.linear.log.envir.interaction_+3A_sd_y_x_truth">sd_y_x_truth</code></td>
<td>
<p>Standard deviation of y for the true model</p>
</td></tr>
<tr><td><code id="calc.like.linear.log.envir.interaction_+3A_sd_y_x_model">sd_y_x_model</code></td>
<td>
<p>Standard deviation of y for the test model</p>
</td></tr>
<tr><td><code id="calc.like.linear.log.envir.interaction_+3A_test.model">Test.Model</code></td>
<td>
<p>Test model</p>
</td></tr>
<tr><td><code id="calc.like.linear.log.envir.interaction_+3A_true.model">True.Model</code></td>
<td>
<p>True model</p>
</td></tr>
<tr><td><code id="calc.like.linear.log.envir.interaction_+3A_reduced">reduced</code></td>
<td>
<p>logical, indicates whether the X matrix will be used for a reduced model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The standard deviation of y given x for linear models with logistic environment interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>beta_hat = linear.mles.log.envir.interaction(MAF = 0.1, P_e = 0.2, 
	ES_G = 1.2, ES_E = 1.3, ES_GE = 2, 
	Test.Model = "Dominant", True.Model = "Additive")
calc.like.linear.log.envir.interaction(beta_hat = beta_hat,
	MAF = 0.1, P_e = 0.2, ES_G = 1.2, ES_E = 1.3,
	ES_GE = 2, sd_y_x_truth = 9.947945, sd_y_x_model = 9.949468, 
	True.Model = "Additive", Test.Model="Dominant")

</code></pre>

<hr>
<h2 id='df2.ll'>Function to Calculate 2df Log Likelihood for a Logistic Regression Model</h2><span id='topic+df2.ll'></span>

<h3>Description</h3>

<p>Calculates the log likelihood for a given set of logistic regression coefficients under an unspecificed/2df genetic model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>df2.ll(beta, t)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="df2.ll_+3A_beta">beta</code></td>
<td>
<p>Vector of logistic regression coefficients.</p>
</td></tr>
<tr><td><code id="df2.ll_+3A_t">t</code></td>
<td>
<p>A 2x3 table of joint probabilities of disease and genotype.  Rows = case vs. control, columns=genotype.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df2.ll(c(-0.3793525, -1.1395417), 
 rbind(c(0.2339079, 0.05665039, 0.009441731), 
 c(0.3285921, 0.31834961, 0.053058269)))

</code></pre>

<hr>
<h2 id='df2.ll.linear'>Function to Calculate 2 Degree of Freedom Log Likelihood for a Linear Regression Model</h2><span id='topic+df2.ll.linear'></span>

<h3>Description</h3>

<p>Calculates the log likelihood for a given set of linear regression coefficients under a the 2df model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>df2.ll.linear(beta, m, es, sd_y_x_model, sd_y_x_truth)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="df2.ll.linear_+3A_beta">beta</code></td>
<td>
<p>Vector of linear regression coefficients.</p>
</td></tr>
<tr><td><code id="df2.ll.linear_+3A_m">m</code></td>
<td>
<p>Minor allele frequency.</p>
</td></tr>
<tr><td><code id="df2.ll.linear_+3A_es">es</code></td>
<td>
<p>Vector of effect sizes with two elements, (mean AB - mean AA) and (mean BB - mean AA).</p>
</td></tr>
<tr><td><code id="df2.ll.linear_+3A_sd_y_x_model">sd_y_x_model</code></td>
<td>
<p>The standard deviation of Y (the outcome) given X (predictors/genotype) under the test model.</p>
</td></tr>
<tr><td><code id="df2.ll.linear_+3A_sd_y_x_truth">sd_y_x_truth</code></td>
<td>
<p>The standard deviation of Y given X (predictors/genotype) given genotype under the true model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df2.ll.linear(beta = c(0, 0, 3), m = 0.1, es = c(0,3), 
 sd_y_x_model = 0.9544108, sd_y_x_truth = 0.9544108)

</code></pre>

<hr>
<h2 id='dom.fun.t'>Function to Calculate t matrix for logistic outcome with binary environment interaction in dominant model</h2><span id='topic+dom.fun.t'></span>

<h3>Description</h3>

<p>Calculates the t matrix for logistic outcome with binary environment interaction in dominant model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dom.fun.t(MAF, P_e, OR_E, OR_G, OR_GE, Case.Rate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dom.fun.t_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="dom.fun.t_+3A_p_e">P_e</code></td>
<td>
<p>Vector of proportions of the population with exposure to the environmental effect</p>
</td></tr>
<tr><td><code id="dom.fun.t_+3A_or_e">OR_E</code></td>
<td>
<p>Vector of environmental odds ratios to detect</p>
</td></tr>
<tr><td><code id="dom.fun.t_+3A_or_g">OR_G</code></td>
<td>
<p>Vector of genetic odds ratios to detect</p>
</td></tr>
<tr><td><code id="dom.fun.t_+3A_or_ge">OR_GE</code></td>
<td>
<p>Vector of genetic/environmental interaction odds ratios to detect</p>
</td></tr>
<tr><td><code id="dom.fun.t_+3A_case.rate">Case.Rate</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>t matrix for all combinations of environment/outcome
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dom.fun.t(MAF = 0.1, P_e = 0.2, Case.Rate = 0.5,
	OR_G = 1.5, OR_E = 2, OR_GE = 1.8)

</code></pre>

<hr>
<h2 id='dom.or.function'>Dominant Model Function</h2><span id='topic+dom.or.function'></span>

<h3>Description</h3>

<p>Operates within odds_ratio_function to calculate odds ratios for a Test.Model of &quot;Dominant&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dom.or.function(like, Case.Rate, P_AA, P_AB, P_BB, True.Model, risk_allele)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dom.or.function_+3A_like">like</code></td>
<td>
<p>Expected log likelihood</p>
</td></tr>
<tr><td><code id="dom.or.function_+3A_case.rate">Case.Rate</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
<tr><td><code id="dom.or.function_+3A_p_aa">P_AA</code></td>
<td>
<p>Probability the allele is homozygous for the major allele</p>
</td></tr>
<tr><td><code id="dom.or.function_+3A_p_ab">P_AB</code></td>
<td>
<p>Probability the allele is heterozygous</p>
</td></tr>
<tr><td><code id="dom.or.function_+3A_p_bb">P_BB</code></td>
<td>
<p>Probability the allele is homozygous for the minor allele</p>
</td></tr>
<tr><td><code id="dom.or.function_+3A_true.model">True.Model</code></td>
<td>
<p>A vector object specifying the true underlying genetic model(s): 'Dominant', 'Additive', or 'Recessive'</p>
</td></tr>
<tr><td><code id="dom.or.function_+3A_risk_allele">risk_allele</code></td>
<td>
<p>Logical: If OR &gt; 1, the allele is classified as a &quot;risk allele&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>: The odds ratios and their corresponding genetic model(s)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dom.or.function(like=-0.57162, Case.Rate=0.3, P_AA=0.5625, P_AB=0.375, 
 P_BB=0.0625, True.Model="Dominant", risk_allele=TRUE)

</code></pre>

<hr>
<h2 id='dominant.ll'>Function to Calculate Dominant Log Likelihood for a Logistic Regression Model</h2><span id='topic+dominant.ll'></span>

<h3>Description</h3>

<p>Calculates the log likelihood for a given set of logistic regression coefficients under a dominant genetic model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dominant.ll(beta, t)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dominant.ll_+3A_beta">beta</code></td>
<td>
<p>Vector of logistic regression coefficients.</p>
</td></tr>
<tr><td><code id="dominant.ll_+3A_t">t</code></td>
<td>
<p>A 2x3 table of joint probabilities of disease and genotype.  Rows = case vs. control, columns=genotype.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dominant.ll(c(-0.3793525, -1.1395417), 
 rbind(c(0.2339079, 0.05665039, 0.009441731), 
 c(0.3285921, 0.31834961, 0.053058269)))

</code></pre>

<hr>
<h2 id='dominant.ll.linear'>Function to Calculate Dominant Log Likelihood for a Linear Regression Model</h2><span id='topic+dominant.ll.linear'></span>

<h3>Description</h3>

<p>Calculates the log likelihood for a given set of linear regression coefficients under a dominant genetic model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dominant.ll.linear(beta, m, es, sd_y_x_model, sd_y_x_truth)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dominant.ll.linear_+3A_beta">beta</code></td>
<td>
<p>Vector of linear regression coefficients.</p>
</td></tr>
<tr><td><code id="dominant.ll.linear_+3A_m">m</code></td>
<td>
<p>Minor allele frequency.</p>
</td></tr>
<tr><td><code id="dominant.ll.linear_+3A_es">es</code></td>
<td>
<p>Vector of effect sizes with two elements, (mean AB - mean AA) and (mean BB - mean AA).</p>
</td></tr>
<tr><td><code id="dominant.ll.linear_+3A_sd_y_x_model">sd_y_x_model</code></td>
<td>
<p>The standard deviation of Y (the outcome) given X (predictors/genotype) under the test model.</p>
</td></tr>
<tr><td><code id="dominant.ll.linear_+3A_sd_y_x_truth">sd_y_x_truth</code></td>
<td>
<p>The standard deviation of Y given X (predictors/genotype) given genotype under the true model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dominant.ll.linear(beta = c(0.0000000, 0.1578947), m = 0.1, es = c(0,3), 
 sd_y_x_model = 0.9980797, sd_y_x_truth = 0.9544108)

</code></pre>

<hr>
<h2 id='es.calc.linear'>Function to Calculate Effect Size for Linear Models</h2><span id='topic+es.calc.linear'></span>

<h3>Description</h3>

<p>Calculates the detectable effect size/regression coefficient, at a given sample size, N, and power, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>es.calc.linear(
  power = NULL,
  N = NULL,
  MAF = NULL,
  sd_y = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="es.calc.linear_+3A_power">power</code></td>
<td>
<p>Vector of the desired power(s)</p>
</td></tr>
<tr><td><code id="es.calc.linear_+3A_n">N</code></td>
<td>
<p>Vector of the desired sample size(s)</p>
</td></tr>
<tr><td><code id="es.calc.linear_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="es.calc.linear_+3A_sd_y">sd_y</code></td>
<td>
<p>Standard deviation of the outcome in the population (ignoring genotype). Either sd_y_x or sd_y must be specified.</p>
</td></tr>
<tr><td><code id="es.calc.linear_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="es.calc.linear_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="es.calc.linear_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, ES, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>es &lt;- es.calc.linear(N=1000,power=0.8,
    MAF=0.1, sd_y = 1, Alpha=0.05,
    True.Model='All', Test.Model='All')

</code></pre>

<hr>
<h2 id='expected.linear.ll'>Function to Calculate Expected Log Likelihood for a Single Genotype</h2><span id='topic+expected.linear.ll'></span>

<h3>Description</h3>

<p>Calculates the expected log likelihood for a single genotype given the true and estimated mean and standard deviation for the outcome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expected.linear.ll(mean_truth, mean_model, sd_y_x_truth, sd_y_x_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expected.linear.ll_+3A_mean_truth">mean_truth</code></td>
<td>
<p>Mean of the outcome given X(predictors/genotype) under the true model.</p>
</td></tr>
<tr><td><code id="expected.linear.ll_+3A_mean_model">mean_model</code></td>
<td>
<p>Mean of the outcome given X(predictors/genotype) under the test model.</p>
</td></tr>
<tr><td><code id="expected.linear.ll_+3A_sd_y_x_truth">sd_y_x_truth</code></td>
<td>
<p>The standard deviation of Y given X (predictors/genotype) given genotype under the true model.</p>
</td></tr>
<tr><td><code id="expected.linear.ll_+3A_sd_y_x_model">sd_y_x_model</code></td>
<td>
<p>The standard deviation of Y (the outcome) given X (predictors/genotype) under the test model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>expected.linear.ll(mean_truth = 0, mean_model = 0.03,
 sd_y_x_model = 1, sd_y_x_truth = 0.9544108)

</code></pre>

<hr>
<h2 id='expected.linear.ll.lin.env'>Function to Calculate Expected Log Likelihood for a Single Genotype with linear environment interaction</h2><span id='topic+expected.linear.ll.lin.env'></span>

<h3>Description</h3>

<p>Calculates the expected log likelihood for a single genotype with linear environment interaction 
given the true and estimated mean and standard deviation for the outcome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expected.linear.ll.lin.env(sd_y_x_model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expected.linear.ll.lin.env_+3A_sd_y_x_model">sd_y_x_model</code></td>
<td>
<p>The standard deviation of Y (the outcome) given X (predictors/genotype) under the test model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>expected.linear.ll.lin.env(4.309354)

</code></pre>

<hr>
<h2 id='find.prob.dom'>Dominant probability finding function</h2><span id='topic+find.prob.dom'></span>

<h3>Description</h3>

<p>Operates within add.or.function to find probability of disease in a dominant truth given AB or BB, additive test model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.prob.dom(x, P_AA, P_AB, P_BB, cr, like)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find.prob.dom_+3A_x">x</code></td>
<td>
<p>Probability of disease given AB or BB</p>
</td></tr>
<tr><td><code id="find.prob.dom_+3A_p_aa">P_AA</code></td>
<td>
<p>Probability the allele is homozygous for the major allele</p>
</td></tr>
<tr><td><code id="find.prob.dom_+3A_p_ab">P_AB</code></td>
<td>
<p>Probability the allele is heterozygous</p>
</td></tr>
<tr><td><code id="find.prob.dom_+3A_p_bb">P_BB</code></td>
<td>
<p>Probability the allele is homozygous for the minor allele</p>
</td></tr>
<tr><td><code id="find.prob.dom_+3A_cr">cr</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
<tr><td><code id="find.prob.dom_+3A_like">like</code></td>
<td>
<p>Expected log likelihood</p>
</td></tr>
</table>


<h3>Value</h3>

<p>: The &quot;a&quot; in the binomial function ax^2 + bx + c that arises in solution for the additive OR functions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>find.prob.dom(0.1510677, 0.5625, 0.375, 0.0625, 0.3, -0.57162)

</code></pre>

<hr>
<h2 id='find.prob.rec'>Recessive probability finding function</h2><span id='topic+find.prob.rec'></span>

<h3>Description</h3>

<p>Operates within add.or.function to find probability of disease in a recessive truth given AB or BB, additive test model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.prob.rec(x, P_AA, P_AB, P_BB, cr, like)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find.prob.rec_+3A_x">x</code></td>
<td>
<p>Probability of disease given AB or BB</p>
</td></tr>
<tr><td><code id="find.prob.rec_+3A_p_aa">P_AA</code></td>
<td>
<p>Probability the allele is homozygous for the major allele</p>
</td></tr>
<tr><td><code id="find.prob.rec_+3A_p_ab">P_AB</code></td>
<td>
<p>Probability the allele is heterozygous</p>
</td></tr>
<tr><td><code id="find.prob.rec_+3A_p_bb">P_BB</code></td>
<td>
<p>Probability the allele is homozygous for the minor allele</p>
</td></tr>
<tr><td><code id="find.prob.rec_+3A_cr">cr</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
<tr><td><code id="find.prob.rec_+3A_like">like</code></td>
<td>
<p>Expected log likelihood</p>
</td></tr>
</table>


<h3>Value</h3>

<p>: The &quot;a&quot; in the binomial function ax^2 + bx + c that arises in solution for the additive OR functions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>find.prob.rec(0.7072381, 0.5625, 0.375, 0.0625, 0.3, -0.6005743)

</code></pre>

<hr>
<h2 id='genpwr.calc'>Function to Calculate Power for Linear Models with logistic environment interaction</h2><span id='topic+genpwr.calc'></span>

<h3>Description</h3>

<p>Calculates the power to detect an difference in means/effect size/regression coefficient, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genpwr.calc(
  calc,
  model,
  ge.interaction = NULL,
  N = NULL,
  Power = NULL,
  MAF = NULL,
  Alpha = 0.05,
  P_e = NULL,
  sd_e = NULL,
  sd_y = NULL,
  Case.Rate = NULL,
  k = NULL,
  OR = NULL,
  OR_G = NULL,
  OR_E = NULL,
  OR_GE = NULL,
  risk_allele = TRUE,
  ES = NULL,
  ES_G = NULL,
  ES_E = NULL,
  ES_GE = NULL,
  R2 = NULL,
  R2_G = NULL,
  R2_E = NULL,
  R2_GE = NULL,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genpwr.calc_+3A_calc">calc</code></td>
<td>
<p>What kind of calculation to perform? sample size (&quot;ss&quot;), power (&quot;power&quot;), or effect size (&quot;es&quot;)</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_model">model</code></td>
<td>
<p>Distribution of the outcome variable? (&quot;logistic&quot; or &quot;linear&quot;)</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_ge.interaction">ge.interaction</code></td>
<td>
<p>If no environment interaction, should be NULL, otherwise should be &quot;logistic&quot; or &quot;linear&quot;</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_n">N</code></td>
<td>
<p>Vector of the desired sample size(s)</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_power">Power</code></td>
<td>
<p>Vector of the desired power(s)</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_p_e">P_e</code></td>
<td>
<p>Vector of proportions of the population with exposure to the environmental effect</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_sd_e">sd_e</code></td>
<td>
<p>Standard deviation of the environmental variable</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_sd_y">sd_y</code></td>
<td>
<p>Standard deviation of the outcome in the population (ignoring genotype). Either sd_y_x or sd_y must be specified.</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_case.rate">Case.Rate</code></td>
<td>
<p>Case Rate of the outcome in the population (ignoring genotype). Either Case.Rate_x or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_k">k</code></td>
<td>
<p>Vector of the number of controls per case. Either k or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_or">OR</code></td>
<td>
<p>Vector of genetic odds ratios to detect in absence of environmental odds ratios</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_or_g">OR_G</code></td>
<td>
<p>Vector of genetic odds ratios to detect</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_or_e">OR_E</code></td>
<td>
<p>Vector of environmental odds ratios to detect</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_or_ge">OR_GE</code></td>
<td>
<p>Vector of genetic/environmental interaction odds ratios to detect</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_risk_allele">risk_allele</code></td>
<td>
<p>Logical: If OR &gt; 1, the allele is classified as a &quot;risk allele&quot;</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_es">ES</code></td>
<td>
<p>Vector of effect sizes (difference in means) to detect. Either ES or R2 must be specified.</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_es_g">ES_G</code></td>
<td>
<p>Vector of genetic effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_es_e">ES_E</code></td>
<td>
<p>Vector of environmental effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_es_ge">ES_GE</code></td>
<td>
<p>Vector of genetic/environment interaction effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_r2">R2</code></td>
<td>
<p>Vector of R-squared values to detect. Either ES or R2 must be specified.</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_r2_g">R2_G</code></td>
<td>
<p>Vector of genetic R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_r2_e">R2_E</code></td>
<td>
<p>Vector of environmental R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_r2_ge">R2_GE</code></td>
<td>
<p>Vector of genetic/environment interaction R-squared values Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="genpwr.calc_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, ES, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pw &lt;- genpwr.calc(calc = "power", model = "logistic", ge.interaction = "continuous",
	N=100, OR_G=2, OR_E=1.4, OR_GE=c(1.5, 2), 
	sd_e = 1.1, MAF=0.1, Case.Rate = 0.3, Alpha=0.05,
	True.Model="All", Test.Model=c("Dominant", "Recessive"))


</code></pre>

<hr>
<h2 id='integrand_funct_case'>Function to generate integrand for mle for cases</h2><span id='topic+integrand_funct_case'></span>

<h3>Description</h3>

<p>Returns the standard deviation of y given x for linear models with linear environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>integrand_funct_case(x1, x2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="integrand_funct_case_+3A_x1">x1</code></td>
<td>
<p>&quot;true&quot; part of model</p>
</td></tr>
<tr><td><code id="integrand_funct_case_+3A_x2">x2</code></td>
<td>
<p>&quot;test&quot; part of model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a function to be used as the integrand for the mle
</p>


<h3>Examples</h3>

<pre><code class='language-R'>integrand_funct_case(-1.462531 + 1*0.1823216,
	-1.462531 + 1*0.1823216) 

</code></pre>

<hr>
<h2 id='integrand_funct_control'>Function to generate integrand for mle for controls</h2><span id='topic+integrand_funct_control'></span>

<h3>Description</h3>

<p>Returns the standard deviation of y given x for linear models with linear environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>integrand_funct_control(x1, x2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="integrand_funct_control_+3A_x1">x1</code></td>
<td>
<p>&quot;true&quot; part of model</p>
</td></tr>
<tr><td><code id="integrand_funct_control_+3A_x2">x2</code></td>
<td>
<p>&quot;test&quot; part of model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a function to be used as the integrand for the mle
</p>


<h3>Examples</h3>

<pre><code class='language-R'>integrand_funct_control(-1.462531 + 1*0.1823216,
	-1.462531 + 1*0.1823216) 

</code></pre>

<hr>
<h2 id='linear.mles'>Function to calculate MLE's for linear models</h2><span id='topic+linear.mles'></span>

<h3>Description</h3>

<p>Finds the maximum likelihood estimates for a given MAF under the specified genetic model and effect size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linear.mles(m, es_ab, es_bb, model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linear.mles_+3A_m">m</code></td>
<td>
<p>minor allele frequency</p>
</td></tr>
<tr><td><code id="linear.mles_+3A_es_ab">es_ab</code></td>
<td>
<p>effect size for mean AB - mean AA</p>
</td></tr>
<tr><td><code id="linear.mles_+3A_es_bb">es_bb</code></td>
<td>
<p>effect size for mean BB - mean AA</p>
</td></tr>
<tr><td><code id="linear.mles_+3A_model">model</code></td>
<td>
<p>The assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive', '2df'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of linear regression model coefficients.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>linear.mles(m = 0.1, es_ab = 0, es_bb = 3, model = "Dominant")

</code></pre>

<hr>
<h2 id='linear.mles.lin.envir.interaction'>Function to calculate the standard deviation of y given x for linear models with linear environment interaction</h2><span id='topic+linear.mles.lin.envir.interaction'></span>

<h3>Description</h3>

<p>Returns the standard deviation of y given x for linear models with linear environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linear.mles.lin.envir.interaction(
  MAF,
  beta0,
  ES_G,
  ES_E,
  ES_GE,
  Test.Model,
  True.Model
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linear.mles.lin.envir.interaction_+3A_maf">MAF</code></td>
<td>
<p>Minor allele Frequency</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_+3A_beta0">beta0</code></td>
<td>
<p>baseline value for the outcome</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_+3A_es_g">ES_G</code></td>
<td>
<p>Genetic Effect size</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_+3A_es_e">ES_E</code></td>
<td>
<p>Environment Effect size</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_+3A_es_ge">ES_GE</code></td>
<td>
<p>Environment x Genetic interaction Effect size</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_+3A_test.model">Test.Model</code></td>
<td>
<p>Test Model</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_+3A_true.model">True.Model</code></td>
<td>
<p>True Model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The standard deviation of y given x for linear models with linear environment interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>linear.mles.lin.envir.interaction(MAF = 0.28, ES_G = 0.5, beta0 = -0.28,
	ES_E = 1.6, ES_GE = 1.4, Test.Model = "Dominant", True.Model = "Additive")

</code></pre>

<hr>
<h2 id='linear.mles.lin.envir.interaction_reduced'>Function to calculate the standard deviation of y given x for linear models with linear environment interaction 
for the reduced model without GxE interaction</h2><span id='topic+linear.mles.lin.envir.interaction_reduced'></span>

<h3>Description</h3>

<p>Returns the standard deviation of y given x for linear models with linear environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linear.mles.lin.envir.interaction_reduced(
  MAF,
  beta0,
  ES_G,
  ES_E,
  ES_GE,
  Test.Model,
  True.Model
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linear.mles.lin.envir.interaction_reduced_+3A_maf">MAF</code></td>
<td>
<p>Minor allele Frequency</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_reduced_+3A_beta0">beta0</code></td>
<td>
<p>baseline value for the outcome</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_reduced_+3A_es_g">ES_G</code></td>
<td>
<p>Genetic Effect size</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_reduced_+3A_es_e">ES_E</code></td>
<td>
<p>Environment Effect size</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_reduced_+3A_es_ge">ES_GE</code></td>
<td>
<p>Environment x Genetic interaction Effect size</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_reduced_+3A_test.model">Test.Model</code></td>
<td>
<p>Test Model</p>
</td></tr>
<tr><td><code id="linear.mles.lin.envir.interaction_reduced_+3A_true.model">True.Model</code></td>
<td>
<p>True Model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The standard deviation of y given x for linear models with linear environment interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>linear.mles.lin.envir.interaction_reduced(MAF = 0.28, ES_G = 0.5, beta0 = -0.28,
	ES_E = 1.6, ES_GE = 1.4, Test.Model = "Dominant", True.Model = "Additive")

</code></pre>

<hr>
<h2 id='linear.mles.log.envir.interaction'>Function to calculate the standard deviation of y given x for linear models with logistic environment interaction</h2><span id='topic+linear.mles.log.envir.interaction'></span>

<h3>Description</h3>

<p>Returns the standard deviation of y given x for linear models with logistic environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linear.mles.log.envir.interaction(
  MAF,
  P_e,
  ES_G,
  ES_E,
  ES_GE,
  Test.Model,
  True.Model,
  reduced = F
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linear.mles.log.envir.interaction_+3A_maf">MAF</code></td>
<td>
<p>Minor allele Frequency</p>
</td></tr>
<tr><td><code id="linear.mles.log.envir.interaction_+3A_p_e">P_e</code></td>
<td>
<p>Population prevalence of logistic environmental factor</p>
</td></tr>
<tr><td><code id="linear.mles.log.envir.interaction_+3A_es_g">ES_G</code></td>
<td>
<p>Genetic Effect size</p>
</td></tr>
<tr><td><code id="linear.mles.log.envir.interaction_+3A_es_e">ES_E</code></td>
<td>
<p>Environment Effect size</p>
</td></tr>
<tr><td><code id="linear.mles.log.envir.interaction_+3A_es_ge">ES_GE</code></td>
<td>
<p>Environment x Genetic interaction Effect size</p>
</td></tr>
<tr><td><code id="linear.mles.log.envir.interaction_+3A_test.model">Test.Model</code></td>
<td>
<p>Test model</p>
</td></tr>
<tr><td><code id="linear.mles.log.envir.interaction_+3A_true.model">True.Model</code></td>
<td>
<p>True model</p>
</td></tr>
<tr><td><code id="linear.mles.log.envir.interaction_+3A_reduced">reduced</code></td>
<td>
<p>logical, indicates whether the X matrix will be used for a reduced model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The standard deviation of y given x for linear models with logistic environment interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>linear.mles.log.envir.interaction(MAF = 0.1, P_e = 0.2, 
	ES_G = 1.2, ES_E = 1.3, ES_GE = 2, 
	Test.Model = "Dominant", True.Model = "Additive")

</code></pre>

<hr>
<h2 id='linear.outcome.lin.envir.interaction.sds'>Function to calculate the standard deviation of y given x for linear models with linear environment interaction</h2><span id='topic+linear.outcome.lin.envir.interaction.sds'></span>

<h3>Description</h3>

<p>Returns the standard deviation of y given x for linear models with linear environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linear.outcome.lin.envir.interaction.sds(
  MAF,
  sd_e,
  beta0,
  ES_G,
  ES_E,
  ES_GE,
  mod,
  True.Model,
  sd_y
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_+3A_maf">MAF</code></td>
<td>
<p>Minor allele Frequency</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_+3A_sd_e">sd_e</code></td>
<td>
<p>Standard deviation of linear environmental factor</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_+3A_beta0">beta0</code></td>
<td>
<p>baseline value for the outcome</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_+3A_es_g">ES_G</code></td>
<td>
<p>Genetic Effect size</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_+3A_es_e">ES_E</code></td>
<td>
<p>Environment Effect size</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_+3A_es_ge">ES_GE</code></td>
<td>
<p>Environment x Genetic interaction Effect size</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_+3A_mod">mod</code></td>
<td>
<p>Test model</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_+3A_true.model">True.Model</code></td>
<td>
<p>True model</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_+3A_sd_y">sd_y</code></td>
<td>
<p>Standard deviation of y</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The standard deviation of y given x for linear models with linear environment interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>linear.outcome.lin.envir.interaction.sds(MAF = 0.28, beta0 = -0.28, 
	sd_y = 5, sd_e = 1, ES_G = 0.5, ES_E = 1.6, ES_GE = 1.4, 
	mod = "Dominant", True.Model = "Additive")

</code></pre>

<hr>
<h2 id='linear.outcome.lin.envir.interaction.sds_reduced'>Function to calculate the standard deviation of y given x for linear models with linear environment interaction</h2><span id='topic+linear.outcome.lin.envir.interaction.sds_reduced'></span>

<h3>Description</h3>

<p>Returns the standard deviation of y given x for linear models with linear environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linear.outcome.lin.envir.interaction.sds_reduced(
  MAF,
  sd_e,
  beta0,
  ES_G,
  ES_E,
  ES_GE,
  mod,
  True.Model,
  sd_y
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_reduced_+3A_maf">MAF</code></td>
<td>
<p>Minor allele Frequency</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_reduced_+3A_sd_e">sd_e</code></td>
<td>
<p>Standard deviation of linear environmental factor</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_reduced_+3A_beta0">beta0</code></td>
<td>
<p>baseline value for the outcome</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_reduced_+3A_es_g">ES_G</code></td>
<td>
<p>Genetic Effect size</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_reduced_+3A_es_e">ES_E</code></td>
<td>
<p>Environment Effect size</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_reduced_+3A_es_ge">ES_GE</code></td>
<td>
<p>Environment x Genetic interaction Effect size</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_reduced_+3A_mod">mod</code></td>
<td>
<p>Test model</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_reduced_+3A_true.model">True.Model</code></td>
<td>
<p>True model</p>
</td></tr>
<tr><td><code id="linear.outcome.lin.envir.interaction.sds_reduced_+3A_sd_y">sd_y</code></td>
<td>
<p>Standard deviation of y</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The standard deviation of y given x for linear models with linear environment interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>linear.outcome.lin.envir.interaction.sds_reduced(MAF = 0.28, beta0 = -0.28, 
	sd_y = 5, sd_e = 1, ES_G = 0.5, ES_E = 1.6, 
	ES_GE = 1.4, mod = "Dominant", True.Model = "Additive")

</code></pre>

<hr>
<h2 id='linear.outcome.log.envir.interaction.sds'>Function to calculate the standard deviation of y given x for linear models with logistic environment interaction</h2><span id='topic+linear.outcome.log.envir.interaction.sds'></span>

<h3>Description</h3>

<p>Returns the standard deviation of y given x for linear models with logistic environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linear.outcome.log.envir.interaction.sds(
  MAF,
  P_e,
  ES_G,
  ES_E,
  ES_GE,
  mod,
  True.Model,
  sd_y,
  reduced = F
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linear.outcome.log.envir.interaction.sds_+3A_maf">MAF</code></td>
<td>
<p>Minor allele Frequency</p>
</td></tr>
<tr><td><code id="linear.outcome.log.envir.interaction.sds_+3A_p_e">P_e</code></td>
<td>
<p>Population prevalence of logistic environmental factor</p>
</td></tr>
<tr><td><code id="linear.outcome.log.envir.interaction.sds_+3A_es_g">ES_G</code></td>
<td>
<p>Genetic Effect size</p>
</td></tr>
<tr><td><code id="linear.outcome.log.envir.interaction.sds_+3A_es_e">ES_E</code></td>
<td>
<p>Environment Effect size</p>
</td></tr>
<tr><td><code id="linear.outcome.log.envir.interaction.sds_+3A_es_ge">ES_GE</code></td>
<td>
<p>Environment x Genetic interaction Effect size</p>
</td></tr>
<tr><td><code id="linear.outcome.log.envir.interaction.sds_+3A_mod">mod</code></td>
<td>
<p>Test model</p>
</td></tr>
<tr><td><code id="linear.outcome.log.envir.interaction.sds_+3A_true.model">True.Model</code></td>
<td>
<p>True model</p>
</td></tr>
<tr><td><code id="linear.outcome.log.envir.interaction.sds_+3A_sd_y">sd_y</code></td>
<td>
<p>Standard deviation of y</p>
</td></tr>
<tr><td><code id="linear.outcome.log.envir.interaction.sds_+3A_reduced">reduced</code></td>
<td>
<p>logical, indicates whether the X matrix will be used for a reduced model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The standard deviation of y given x for linear models with logistic environment interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>linear.outcome.log.envir.interaction.sds(MAF = 0.1, P_e = 0.2, sd_y = 10,
	ES_G = 1.2, ES_E = 1.3, ES_GE = 2, mod = "Dominant", True.Model = "Additive")

</code></pre>

<hr>
<h2 id='linear.sds'>Functions to Calculate Residual SD for Normal/Continuous Outcomes
Function to calculate the standard deviation of y given x for linear models</h2><span id='topic+linear.sds'></span>

<h3>Description</h3>

<p>Functions to Calculate Residual SD for Normal/Continuous Outcomes
Function to calculate the standard deviation of y given x for linear models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linear.sds(m, es_ab, es_bb, sd_y, model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linear.sds_+3A_m">m</code></td>
<td>
<p>minor allele frequency</p>
</td></tr>
<tr><td><code id="linear.sds_+3A_es_ab">es_ab</code></td>
<td>
<p>effect size for mean AB - mean AA</p>
</td></tr>
<tr><td><code id="linear.sds_+3A_es_bb">es_bb</code></td>
<td>
<p>effect size for mean BB - mean AA</p>
</td></tr>
<tr><td><code id="linear.sds_+3A_sd_y">sd_y</code></td>
<td>
<p>the standard deviation of y in the overall population.</p>
</td></tr>
<tr><td><code id="linear.sds_+3A_model">model</code></td>
<td>
<p>The assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive', '2df'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of linear regression model coefficients.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>linear.sds(m = 0.1, es_ab = 0, es_bb = 3, sd_y = 1, model = "Dominant")

</code></pre>

<hr>
<h2 id='ll_zero_finder2'>Zero finding function</h2><span id='topic+ll_zero_finder2'></span>

<h3>Description</h3>

<p>Finds the zeros of a function af. Alternative to uniroot, designed specifically to work with the genpwr package.
Finds multiple zeros if a function has more than one in the given range.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ll_zero_finder2(af, ii = 6, lower = 0, upper = 1, qdelta = 27)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ll_zero_finder2_+3A_af">af</code></td>
<td>
<p>The function for which to find the zero(s)</p>
</td></tr>
<tr><td><code id="ll_zero_finder2_+3A_ii">ii</code></td>
<td>
<p>Number of iterations. The more iterations, the more accuracy. It is recommended that ii be at least 4.</p>
</td></tr>
<tr><td><code id="ll_zero_finder2_+3A_lower">lower</code></td>
<td>
<p>Lower limit of region in which to find the zero</p>
</td></tr>
<tr><td><code id="ll_zero_finder2_+3A_upper">upper</code></td>
<td>
<p>Upper limit of region in which to find the zero</p>
</td></tr>
<tr><td><code id="ll_zero_finder2_+3A_qdelta">qdelta</code></td>
<td>
<p>Factor for finding intervals over which the function is close to zero</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Points over the given interval at which the given function is approximately equal to zero
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ll_zero_finder2(function(x) (x-0.5)^2 - 0.1)
ll_zero_finder2(function(x) 8*x^3 - 11.2*x^2 + 4.56*x - 0.476)

</code></pre>

<hr>
<h2 id='ll.ge.logistic'>Function to calculate MLE's for logistic models with logistic environment interaction</h2><span id='topic+ll.ge.logistic'></span>

<h3>Description</h3>

<p>Finds the maximum likelihood estimates for a given 2x3 table under the specified genetic model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ll.ge.logistic(t, N = NULL, power = NULL, Alpha, mod)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ll.ge.logistic_+3A_t">t</code></td>
<td>
<p>A 2x6 table of the joint probabilities of disease, genotype, and environment. Rows are case vs. control and columns are genotypes.</p>
</td></tr>
<tr><td><code id="ll.ge.logistic_+3A_n">N</code></td>
<td>
<p>Sample size</p>
</td></tr>
<tr><td><code id="ll.ge.logistic_+3A_power">power</code></td>
<td>
<p>Power</p>
</td></tr>
<tr><td><code id="ll.ge.logistic_+3A_alpha">Alpha</code></td>
<td>
<p>Alpha</p>
</td></tr>
<tr><td><code id="ll.ge.logistic_+3A_mod">mod</code></td>
<td>
<p>Test model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of logistic regression model coefficients.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>t &lt;- rbind(c(0.2870353, 0.07833006, 0.00435167, 0.09946088, 0.029199878, 0.0016222154),
	c(0.3609647, 0.06566994, 0.00364833, 0.06253912, 0.006800122, 0.0003777846))
ll.ge.logistic(t, N = 200, Alpha = 0.05, mod = "Dominant")

</code></pre>

<hr>
<h2 id='ll.ge.logistic.lin.envir'>Function to output log likelihood for logistic outcome with linear environment variables</h2><span id='topic+ll.ge.logistic.lin.envir'></span>

<h3>Description</h3>

<p>Returns the standard deviation of y given x for linear models with linear environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ll.ge.logistic.lin.envir(
  sd_e,
  N = NULL,
  MAF,
  power = NULL,
  beta0,
  OR_G,
  OR_E,
  OR_GE,
  Alpha,
  True.Model,
  Test.Model
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ll.ge.logistic.lin.envir_+3A_sd_e">sd_e</code></td>
<td>
<p>Standard deviation of the environmental variable</p>
</td></tr>
<tr><td><code id="ll.ge.logistic.lin.envir_+3A_n">N</code></td>
<td>
<p>desired sample size</p>
</td></tr>
<tr><td><code id="ll.ge.logistic.lin.envir_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="ll.ge.logistic.lin.envir_+3A_power">power</code></td>
<td>
<p>desired power</p>
</td></tr>
<tr><td><code id="ll.ge.logistic.lin.envir_+3A_beta0">beta0</code></td>
<td>
<p>the beta0 coefficient in the logistic model</p>
</td></tr>
<tr><td><code id="ll.ge.logistic.lin.envir_+3A_or_g">OR_G</code></td>
<td>
<p>Vector of genetic odds ratios to detect</p>
</td></tr>
<tr><td><code id="ll.ge.logistic.lin.envir_+3A_or_e">OR_E</code></td>
<td>
<p>Vector of environmental odds ratios to detect</p>
</td></tr>
<tr><td><code id="ll.ge.logistic.lin.envir_+3A_or_ge">OR_GE</code></td>
<td>
<p>Vector of genetic/environmental interaction odds ratios to detect</p>
</td></tr>
<tr><td><code id="ll.ge.logistic.lin.envir_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="ll.ge.logistic.lin.envir_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="ll.ge.logistic.lin.envir_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a function to be used as the integrand for the mle
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ll.ge.logistic.lin.envir(sd_e = 1, MAF = 0.2, N = 30, beta0 = -1.462531, OR_G = 1.1, 
	OR_E = 1.2, OR_GE = 1.5, Alpha = 0.05, True.Model = "Dominant", Test.Model = "Dominant")
</code></pre>

<hr>
<h2 id='ll.linear.selector'>Function to return log likelihood function for specified model type</h2><span id='topic+ll.linear.selector'></span>

<h3>Description</h3>

<p>Convenience function to return log likelihood function for specified model type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ll.linear.selector(model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ll.linear.selector_+3A_model">model</code></td>
<td>
<p>The genetic model in the linear regression: &quot;Dominant&quot;, &quot;Additive&quot;, &quot;Recessive&quot;, &quot;2df&quot; or &quot;null&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Log likelihood function for specified model type
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ll.linear.selector("Dominant")

</code></pre>

<hr>
<h2 id='logistic.mles'>Function to calculate MLE's for logistic models</h2><span id='topic+logistic.mles'></span>

<h3>Description</h3>

<p>Finds the maximum likelihood estimates for a given 2x3 table under the specified genetic model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logistic.mles(t, model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logistic.mles_+3A_t">t</code></td>
<td>
<p>A 2x3 table of the joint probabilities of disease and genotype.  Rows are case vs. control and columns are genotypes.</p>
</td></tr>
<tr><td><code id="logistic.mles_+3A_model">model</code></td>
<td>
<p>The assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of logistic regression model coefficients.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>logistic.mles(rbind(c(0.2967437, 0.1806723, 0.02258404), 
 c(0.3432563, 0.1393277, 0.01741596)), "Dominant")

</code></pre>

<hr>
<h2 id='logit'>Logit Function</h2><span id='topic+logit'></span>

<h3>Description</h3>

<p>Calculates the logit of a specified value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logit(x, min = 0, max = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logit_+3A_x">x</code></td>
<td>
<p>a number between 0 and 1.</p>
</td></tr>
<tr><td><code id="logit_+3A_min">min</code></td>
<td>
<p>minimum</p>
</td></tr>
<tr><td><code id="logit_+3A_max">max</code></td>
<td>
<p>maximum</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The logit of x.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>logit(0.5)

</code></pre>

<hr>
<h2 id='ncp.search'>Function to Determine Non-Centrality Parameter of the Chi-squared distribution</h2><span id='topic+ncp.search'></span>

<h3>Description</h3>

<p>This function is set to 0 and solved for x, the non-centrality parameter to determine the sample size in <code><a href="#topic+ss.calc">ss.calc</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncp.search(x, power, Alpha, df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ncp.search_+3A_x">x</code></td>
<td>
<p>the non-centrality parameter</p>
</td></tr>
<tr><td><code id="ncp.search_+3A_power">power</code></td>
<td>
<p>the desired power</p>
</td></tr>
<tr><td><code id="ncp.search_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate</p>
</td></tr>
<tr><td><code id="ncp.search_+3A_df">df</code></td>
<td>
<p>the degrees of freedom for the likelihood ratio test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric value of the function
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ncp.search(x = 7.848861, pow = 0.8, Alpha = 0.05, df=1)

</code></pre>

<hr>
<h2 id='null.ll'>Function to Calculate Null Log Likelihood for a Logistic Regression Model</h2><span id='topic+null.ll'></span>

<h3>Description</h3>

<p>Calculates the log likelihood for a given set of logistic regression coefficients under the null.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>null.ll(t)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="null.ll_+3A_t">t</code></td>
<td>
<p>A 2x3 table of joint probabilities of disease and genotype.  Rows = case vs. control, columns=genotype.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>null.ll(rbind(c(0.2339079, 0.05665039, 0.009441731), 
 c(0.3285921, 0.31834961, 0.053058269)))

</code></pre>

<hr>
<h2 id='null.ll.linear'>Function to Calculate Expected Null Log Likelihood for a Linear Regression Model</h2><span id='topic+null.ll.linear'></span>

<h3>Description</h3>

<p>Calculates the expected log likelihood for a given set of linear regression coefficients under the null.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>null.ll.linear(beta, m, es, sd_y_x_model, sd_y_x_truth)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="null.ll.linear_+3A_beta">beta</code></td>
<td>
<p>Vector of linear regression coefficients.</p>
</td></tr>
<tr><td><code id="null.ll.linear_+3A_m">m</code></td>
<td>
<p>Minor allele frequency.</p>
</td></tr>
<tr><td><code id="null.ll.linear_+3A_es">es</code></td>
<td>
<p>Vector of effect sizes with two elements, (mean AB - mean AA) and (mean BB - mean AA).</p>
</td></tr>
<tr><td><code id="null.ll.linear_+3A_sd_y_x_model">sd_y_x_model</code></td>
<td>
<p>The standard deviation of Y (the outcome) given X (predictors/genotype) under the test model.</p>
</td></tr>
<tr><td><code id="null.ll.linear_+3A_sd_y_x_truth">sd_y_x_truth</code></td>
<td>
<p>The standard deviation of Y given X (predictors/genotype) given genotype under the true model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>null.ll.linear(beta = 0.03, m = 0.1, es = c(0,3), 
 sd_y_x_model = 1, sd_y_x_truth = 0.9544108)

</code></pre>

<hr>
<h2 id='odds_ratio_function'>Odds Ratio Function</h2><span id='topic+odds_ratio_function'></span>

<h3>Description</h3>

<p>Calculates the odds ratio for a given power, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>odds_ratio_function(
  N = NULL,
  Case.Rate = NULL,
  k = NULL,
  MAF = NULL,
  power = NULL,
  risk_allele = TRUE,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="odds_ratio_function_+3A_n">N</code></td>
<td>
<p>Vector of the desired sample size(s)</p>
</td></tr>
<tr><td><code id="odds_ratio_function_+3A_case.rate">Case.Rate</code></td>
<td>
<p>Vector of the proportion(s) of cases in the sample (cases/(cases + controls)). Either k or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="odds_ratio_function_+3A_k">k</code></td>
<td>
<p>Vector of the number of controls per case. Either k or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="odds_ratio_function_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="odds_ratio_function_+3A_power">power</code></td>
<td>
<p>Vector of powers to detect</p>
</td></tr>
<tr><td><code id="odds_ratio_function_+3A_risk_allele">risk_allele</code></td>
<td>
<p>Logical: If OR &gt; 1, the allele is classified as a &quot;risk allele&quot;</p>
</td></tr>
<tr><td><code id="odds_ratio_function_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="odds_ratio_function_+3A_true.model">True.Model</code></td>
<td>
<p>A vector vector the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="odds_ratio_function_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the odds ratios for all combinations of the specified parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'>or &lt;- odds_ratio_function(N=c(100), Case.Rate=0.3, 
 k=NULL, MAF= 0.25, power=0.8, 
 Alpha = 0.05, risk_allele = TRUE, True.Model = 'All', Test.Model = 'All')

</code></pre>

<hr>
<h2 id='or_calc'>Odds ratio calculation</h2><span id='topic+or_calc'></span>

<h3>Description</h3>

<p>Calculates odds ratio for given parameters. Used by the function odds_ratio_function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>or_calc(a, b, c, d, e, f, mod, risk_allele)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="or_calc_+3A_a">a</code></td>
<td>
<p>The probability of a case given homogeneity for the major allele</p>
</td></tr>
<tr><td><code id="or_calc_+3A_b">b</code></td>
<td>
<p>The probability of a case given heterozygosity</p>
</td></tr>
<tr><td><code id="or_calc_+3A_c">c</code></td>
<td>
<p>The probability of a case given homogeneity for the minor allele</p>
</td></tr>
<tr><td><code id="or_calc_+3A_d">d</code></td>
<td>
<p>The probability of a control given homogeneity for the major allele</p>
</td></tr>
<tr><td><code id="or_calc_+3A_e">e</code></td>
<td>
<p>The probability of a control given heterozygosity</p>
</td></tr>
<tr><td><code id="or_calc_+3A_f">f</code></td>
<td>
<p>The probability of a control given homogeneity for the minor allele#'</p>
</td></tr>
<tr><td><code id="or_calc_+3A_mod">mod</code></td>
<td>
<p>The model to be used (eg &quot;Dominant&quot;, &quot;Recessive&quot;, etc)</p>
</td></tr>
<tr><td><code id="or_calc_+3A_risk_allele">risk_allele</code></td>
<td>
<p>Is this allele a risk allele? use T or F</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Odds ratio
</p>


<h3>Examples</h3>

<pre><code class='language-R'>or_calc(a = 0.3649185, b = 0.12797197, c = 0.007109554, 
 d= 0.4450815, e= 0.05202803, f = 0.002890446, 
 mod = "Dominant", risk_allele = TRUE)

</code></pre>

<hr>
<h2 id='or.function.2df'>2df Model Function</h2><span id='topic+or.function.2df'></span>

<h3>Description</h3>

<p>Operates within odds_ratio_function to calculate odds ratios for a Test.Model of &quot;2df&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>or.function.2df(like, Case.Rate, P_AA, P_AB, P_BB, True.Model, risk_allele)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="or.function.2df_+3A_like">like</code></td>
<td>
<p>Expected log likelihood</p>
</td></tr>
<tr><td><code id="or.function.2df_+3A_case.rate">Case.Rate</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
<tr><td><code id="or.function.2df_+3A_p_aa">P_AA</code></td>
<td>
<p>Probability the allele is homozygous for the major allele</p>
</td></tr>
<tr><td><code id="or.function.2df_+3A_p_ab">P_AB</code></td>
<td>
<p>Probability the allele is heterozygous</p>
</td></tr>
<tr><td><code id="or.function.2df_+3A_p_bb">P_BB</code></td>
<td>
<p>Probability the allele is homozygous for the minor allele</p>
</td></tr>
<tr><td><code id="or.function.2df_+3A_true.model">True.Model</code></td>
<td>
<p>A vector object specifying the true underlying genetic model(s): 'Dominant', 'Additive', or 'Recessive'</p>
</td></tr>
<tr><td><code id="or.function.2df_+3A_risk_allele">risk_allele</code></td>
<td>
<p>Logical: If OR &gt; 1, the allele is classified as a &quot;risk allele&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>: The odds ratios and their corresponding genetic model(s)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>or.function.2df(like=-0.5626909, Case.Rate=0.3, P_AA=0.5625, 
 P_AB=0.375, P_BB=0.0625, True.Model="Recessive", risk_allele=TRUE)

</code></pre>

<hr>
<h2 id='or.plot'>Function to Plot Odds Ratio Results</h2><span id='topic+or.plot'></span>

<h3>Description</h3>

<p>Plot the power results by MAF, Power, Alpha or N
</p>


<h3>Usage</h3>

<pre><code class='language-R'>or.plot(
  data = NULL,
  x = "MAF",
  panel.by = "True.Model",
  y_limit = NULL,
  y_log = F,
  return_gg = F,
  linear.effect.measure = "ES",
  select.Alpha = NULL,
  select.power = NULL,
  select.ES = NULL,
  select.N = NULL,
  select.MAF = NULL,
  select.Case.Rate = NULL,
  select.SD = NULL,
  select.True.Model = NULL,
  select.Test.Model = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="or.plot_+3A_data">data</code></td>
<td>
<p>The data frame result from <code><a href="#topic+power.calc">power.calc</a></code></p>
</td></tr>
<tr><td><code id="or.plot_+3A_x">x</code></td>
<td>
<p>The desired variable on the y axis: &quot;MAF&quot;, &quot;OR&quot;, &quot;Alpha&quot;, or &quot;N_total&quot;</p>
</td></tr>
<tr><td><code id="or.plot_+3A_panel.by">panel.by</code></td>
<td>
<p>A grouping variable to panel the graphs by: &quot;True.Model&quot;, &quot;MAF&quot;, &quot;Power&quot;, &quot;Alpha&quot;, or &quot;N_total&quot;</p>
</td></tr>
<tr><td><code id="or.plot_+3A_y_limit">y_limit</code></td>
<td>
<p>An object specifying the minimum and maximum of the y-axis (eg c(0,4)) default is NULL, which allows the limits to be picked automatically</p>
</td></tr>
<tr><td><code id="or.plot_+3A_y_log">y_log</code></td>
<td>
<p>Logical, specifying whether the y axis should be logarithmic. Default is F</p>
</td></tr>
<tr><td><code id="or.plot_+3A_return_gg">return_gg</code></td>
<td>
<p>Logical, specifying whether to return the ggplot object instead of printing out the plot</p>
</td></tr>
<tr><td><code id="or.plot_+3A_linear.effect.measure">linear.effect.measure</code></td>
<td>
<p>Should the graphs indicate ES values, or R2 values? (default ES)</p>
</td></tr>
<tr><td><code id="or.plot_+3A_select.alpha">select.Alpha</code></td>
<td>
<p>Only produce graphs for the specified Alpha level(s).</p>
</td></tr>
<tr><td><code id="or.plot_+3A_select.power">select.power</code></td>
<td>
<p>Only produce graphs for the specified Power(s).</p>
</td></tr>
<tr><td><code id="or.plot_+3A_select.es">select.ES</code></td>
<td>
<p>Only produce graphs for the specified effect sizes(s).</p>
</td></tr>
<tr><td><code id="or.plot_+3A_select.n">select.N</code></td>
<td>
<p>Only produce graphs for the specified sample size(s).</p>
</td></tr>
<tr><td><code id="or.plot_+3A_select.maf">select.MAF</code></td>
<td>
<p>Only produce graphs for the specified minor allele frequency(ies).</p>
</td></tr>
<tr><td><code id="or.plot_+3A_select.case.rate">select.Case.Rate</code></td>
<td>
<p>Only produce graphs for the specified case rate(s).</p>
</td></tr>
<tr><td><code id="or.plot_+3A_select.sd">select.SD</code></td>
<td>
<p>Only produce graphs for the specified standard deviation(s).</p>
</td></tr>
<tr><td><code id="or.plot_+3A_select.true.model">select.True.Model</code></td>
<td>
<p>Only produce graphs for the specified true genetic model(s): &quot;Additive&quot;, &quot;Dominant&quot;, &quot;Recessive&quot;.</p>
</td></tr>
<tr><td><code id="or.plot_+3A_select.test.model">select.Test.Model</code></td>
<td>
<p>Only produce graphs for the specified testing model(s): &quot;Additive&quot;, &quot;Dominant&quot;, &quot;Recessive&quot;, &quot;2df&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A series of plots with power on the Y axis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>or &lt;- odds_ratio_function(N=1000, Case.Rate=0.5, k=NULL,
    MAF=seq(0.3, 0.32, 0.01), power=0.8,Alpha=0.05,
    True.Model=c("Dominant", "Recessive"), Test.Model=c("Dominant", "Recessive"))
or.plot(data=or, x='MAF')

</code></pre>

<hr>
<h2 id='p_vec_returner'>Function to output probability vector used in calculation of MLE's for linear outcome with logistic environment interaction</h2><span id='topic+p_vec_returner'></span>

<h3>Description</h3>

<p>Returns probability vector used in calculation of MLE's for linear outcome with logistic environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_vec_returner(MAF, P_e)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="p_vec_returner_+3A_maf">MAF</code></td>
<td>
<p>Minor allele frequency</p>
</td></tr>
<tr><td><code id="p_vec_returner_+3A_p_e">P_e</code></td>
<td>
<p>Population prevalence of logistic environmental factor</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A probability vector to be used in MLE calculation for linear outcome with logistic environment interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p_vec_returner(MAF = 0.1, P_e = 0.2)

</code></pre>

<hr>
<h2 id='p_vec_returner_lin_env'>Function to output probability vector used in calculation of MLE's for linear outcome with linear environment interaction</h2><span id='topic+p_vec_returner_lin_env'></span>

<h3>Description</h3>

<p>Returns probability vector used in calculation of MLE's for linear outcome with linear environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_vec_returner_lin_env(MAF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="p_vec_returner_lin_env_+3A_maf">MAF</code></td>
<td>
<p>Minor Allele Frequency</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A probability vector to be used in MLE calculation for linear outcome with linear environment interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p_vec_returner_lin_env(0.1)

</code></pre>

<hr>
<h2 id='power_envir.calc'>Function to Calculate Power for Logistic Models with Environment Interaction</h2><span id='topic+power_envir.calc'></span>

<h3>Description</h3>

<p>Calculates the power to detect an difference in means/effect size/regression coefficient, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power_envir.calc(
  N = NULL,
  Case.Rate = NULL,
  k = NULL,
  MAF = NULL,
  OR_G = NULL,
  OR_E = NULL,
  OR_GE = NULL,
  P_e = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power_envir.calc_+3A_n">N</code></td>
<td>
<p>Vector of the desired sample size(s)</p>
</td></tr>
<tr><td><code id="power_envir.calc_+3A_case.rate">Case.Rate</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
<tr><td><code id="power_envir.calc_+3A_k">k</code></td>
<td>
<p>Vector of the number of controls per case. Either k or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="power_envir.calc_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="power_envir.calc_+3A_or_g">OR_G</code></td>
<td>
<p>Vector of genetic odds ratios to detect</p>
</td></tr>
<tr><td><code id="power_envir.calc_+3A_or_e">OR_E</code></td>
<td>
<p>Vector of environmental odds ratios to detect</p>
</td></tr>
<tr><td><code id="power_envir.calc_+3A_or_ge">OR_GE</code></td>
<td>
<p>Vector of genetic/environmental interaction odds ratios to detect</p>
</td></tr>
<tr><td><code id="power_envir.calc_+3A_p_e">P_e</code></td>
<td>
<p>Vector of proportions of the population with exposure to the environmental effect</p>
</td></tr>
<tr><td><code id="power_envir.calc_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="power_envir.calc_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive1', 'Additive2', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="power_envir.calc_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, ES, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pw &lt;- power_envir.calc(P_e = 0.2, MAF = 0.1, N = 200, Case.Rate = 0.5, Alpha = 0.05, 
	OR_G = 1.5, OR_E = 2, OR_GE = 1.8, Test.Model = "All", True.Model = "All")


</code></pre>

<hr>
<h2 id='power_envir.calc.linear_outcome'>Function to Calculate Power for Linear Models with logistic environment interaction</h2><span id='topic+power_envir.calc.linear_outcome'></span>

<h3>Description</h3>

<p>Calculates the power to detect an difference in means/effect size/regression coefficient, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power_envir.calc.linear_outcome(
  N = NULL,
  MAF = NULL,
  ES_G = NULL,
  ES_E = NULL,
  ES_GE = NULL,
  P_e = NULL,
  R2_G = NULL,
  R2_E = NULL,
  R2_GE = NULL,
  sd_y = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_n">N</code></td>
<td>
<p>Vector of the desired sample size(s)</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_es_g">ES_G</code></td>
<td>
<p>Vector of genetic effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_es_e">ES_E</code></td>
<td>
<p>Vector of environmental effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_es_ge">ES_GE</code></td>
<td>
<p>Vector of genetic/environment interaction effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_p_e">P_e</code></td>
<td>
<p>Vector of proportions of the population with exposure to the environmental effect</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_r2_g">R2_G</code></td>
<td>
<p>Vector of genetic R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_r2_e">R2_E</code></td>
<td>
<p>Vector of environmental R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_r2_ge">R2_GE</code></td>
<td>
<p>Vector of genetic/environment interaction R-squared valuesEither ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_sd_y">sd_y</code></td>
<td>
<p>Standard deviation of the outcome in the population (ignoring genotype). Either sd_y_x or sd_y must be specified.</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="power_envir.calc.linear_outcome_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, ES, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pw &lt;- power_envir.calc.linear_outcome(N=100, ES_G = 1.2, ES_E = 1.3, 
	ES_GE = 2, Alpha = 0.05, MAF = 0.2, P_e = 0.2, 
	sd_y = 10, True.Model = "All", Test.Model = "All")

</code></pre>

<hr>
<h2 id='power_linear_envir.calc.linear_outcome'>Function to Calculate Power for Linear Models with linear environment interaction</h2><span id='topic+power_linear_envir.calc.linear_outcome'></span>

<h3>Description</h3>

<p>Calculates the power to detect an difference in means/effect size/regression coefficient, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power_linear_envir.calc.linear_outcome(
  N = NULL,
  MAF = NULL,
  ES_G = NULL,
  ES_E = NULL,
  ES_GE = NULL,
  sd_e = NULL,
  R2_G = NULL,
  R2_E = NULL,
  R2_GE = NULL,
  sd_y = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_n">N</code></td>
<td>
<p>Vector of the desired sample size(s)</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_es_g">ES_G</code></td>
<td>
<p>Vector of genetic effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_es_e">ES_E</code></td>
<td>
<p>Vector of environmental effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_es_ge">ES_GE</code></td>
<td>
<p>Vector of genetic/environment interaction effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_sd_e">sd_e</code></td>
<td>
<p>Standard deviation of the environmental variable</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_r2_g">R2_G</code></td>
<td>
<p>Vector of genetic R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_r2_e">R2_E</code></td>
<td>
<p>Vector of environmental R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_r2_ge">R2_GE</code></td>
<td>
<p>Vector of genetic/environment interaction R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_sd_y">sd_y</code></td>
<td>
<p>Standard deviation of the outcome in the population (ignoring genotype). Either sd_y_x or sd_y must be specified.</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.linear_outcome_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, ES, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pw &lt;- power_linear_envir.calc.linear_outcome(N=1000, 
	ES_G=0.5, ES_E=1.6, ES_GE=1.4, 
	sd_e = 1, MAF=0.28, 
	sd_y = 5,Alpha=0.05,
	True.Model='All', Test.Model='All')

</code></pre>

<hr>
<h2 id='power_linear_envir.calc.logistic_outcome'>Function to Calculate Power for Linear Models with logistic environment interaction</h2><span id='topic+power_linear_envir.calc.logistic_outcome'></span>

<h3>Description</h3>

<p>Calculates the power to detect an difference in means/effect size/regression coefficient, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power_linear_envir.calc.logistic_outcome(
  N = NULL,
  MAF = NULL,
  OR_G = NULL,
  OR_E = NULL,
  OR_GE = NULL,
  sd_e = NULL,
  Case.Rate = NULL,
  k = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power_linear_envir.calc.logistic_outcome_+3A_n">N</code></td>
<td>
<p>Vector of the desired sample size(s)</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.logistic_outcome_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.logistic_outcome_+3A_or_g">OR_G</code></td>
<td>
<p>Vector of genetic odds ratios to detect</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.logistic_outcome_+3A_or_e">OR_E</code></td>
<td>
<p>Vector of environmental odds ratios to detect</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.logistic_outcome_+3A_or_ge">OR_GE</code></td>
<td>
<p>Vector of genetic/environmental interaction odds ratios to detect</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.logistic_outcome_+3A_sd_e">sd_e</code></td>
<td>
<p>Standard deviation of the environmental variable</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.logistic_outcome_+3A_case.rate">Case.Rate</code></td>
<td>
<p>Standard deviation of the outcome in the population (ignoring genotype). Either Case.Rate_x or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.logistic_outcome_+3A_k">k</code></td>
<td>
<p>Vector of the number of controls per case. Either k or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.logistic_outcome_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.logistic_outcome_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="power_linear_envir.calc.logistic_outcome_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, ES, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pw &lt;- power_linear_envir.calc.logistic_outcome(N=30, 
	OR_G=1.1, OR_E=1.2, OR_GE=1.5, 
	sd_e = 1, MAF=0.2, Case.Rate = 0.2,
	Alpha=0.05, True.Model="All", Test.Model="All")


</code></pre>

<hr>
<h2 id='power.calc'>Function to Calculate Power</h2><span id='topic+power.calc'></span>

<h3>Description</h3>

<p>Calculates the power to detect an odds ratio, OR, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.calc(
  N = NULL,
  Case.Rate = NULL,
  k = NULL,
  MAF = NULL,
  OR = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.calc_+3A_n">N</code></td>
<td>
<p>Vector of the desired sample size(s)</p>
</td></tr>
<tr><td><code id="power.calc_+3A_case.rate">Case.Rate</code></td>
<td>
<p>Vector of the proportion(s) of cases in the sample (cases/(cases + controls)). Either k or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="power.calc_+3A_k">k</code></td>
<td>
<p>Vector of the number of controls per case. Either k or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="power.calc_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="power.calc_+3A_or">OR</code></td>
<td>
<p>Vector of odds ratios to detect</p>
</td></tr>
<tr><td><code id="power.calc_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="power.calc_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="power.calc_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, OR, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pw &lt;- power.calc(N=2000, Case.Rate=0.5, k=NULL,
    MAF=0.2, OR=1.5,Alpha=0.05,
    True.Model='All', Test.Model='All')

</code></pre>

<hr>
<h2 id='power.calc.linear'>Function to Calculate Power for Linear Models</h2><span id='topic+power.calc.linear'></span>

<h3>Description</h3>

<p>Calculates the power to detect an difference in means/effect size/regression coefficient, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.calc.linear(
  N = NULL,
  MAF = NULL,
  ES = NULL,
  R2 = NULL,
  sd_y = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.calc.linear_+3A_n">N</code></td>
<td>
<p>Vector of the desired sample size(s)</p>
</td></tr>
<tr><td><code id="power.calc.linear_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="power.calc.linear_+3A_es">ES</code></td>
<td>
<p>Vector of effect sizes (difference in means) to detect. Either ES or R2 must be specified.</p>
</td></tr>
<tr><td><code id="power.calc.linear_+3A_r2">R2</code></td>
<td>
<p>Vector of R-squared values to detect. Either ES or R2 must be specified.</p>
</td></tr>
<tr><td><code id="power.calc.linear_+3A_sd_y">sd_y</code></td>
<td>
<p>Standard deviation of the outcome in the population (ignoring genotype). Either sd_y_x or sd_y must be specified.</p>
</td></tr>
<tr><td><code id="power.calc.linear_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="power.calc.linear_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive1', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="power.calc.linear_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, ES, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pw &lt;- power.calc.linear(N=1000,
    MAF=0.1, ES=3,sd_y = 1,Alpha=0.05,
    True.Model='All', Test.Model='All')

</code></pre>

<hr>
<h2 id='power.plot'>Function to Plot Power Results</h2><span id='topic+power.plot'></span>

<h3>Description</h3>

<p>Plot the power results by MAF, OR, Alpha or N
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.plot(
  data = NULL,
  x = "MAF",
  panel.by = "True.Model",
  y_limit = NULL,
  y_log = F,
  return_gg = F,
  linear.effect.measure = "ES",
  select.Alpha = NULL,
  select.OR = NULL,
  select.ES = NULL,
  select.N = NULL,
  select.MAF = NULL,
  select.Case.Rate = NULL,
  select.SD = NULL,
  select.True.Model = NULL,
  select.Test.Model = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.plot_+3A_data">data</code></td>
<td>
<p>The data frame result from <code><a href="#topic+power.calc">power.calc</a></code></p>
</td></tr>
<tr><td><code id="power.plot_+3A_x">x</code></td>
<td>
<p>The desired variable on the y axis: &quot;MAF&quot;, &quot;OR&quot;, &quot;Alpha&quot;, or &quot;N_total&quot;</p>
</td></tr>
<tr><td><code id="power.plot_+3A_panel.by">panel.by</code></td>
<td>
<p>A grouping variable to panel the graphs by: &quot;True.Model&quot;, &quot;MAF&quot;, &quot;OR&quot;, &quot;Alpha&quot;, or &quot;N_total&quot;</p>
</td></tr>
<tr><td><code id="power.plot_+3A_y_limit">y_limit</code></td>
<td>
<p>An object specifying the minimum and maximum of the y-axis (eg c(0,4)) default is NULL, which allows the limits to be picked automatically</p>
</td></tr>
<tr><td><code id="power.plot_+3A_y_log">y_log</code></td>
<td>
<p>Logical, specifying whether the y axis should be logarithmic. Default is F</p>
</td></tr>
<tr><td><code id="power.plot_+3A_return_gg">return_gg</code></td>
<td>
<p>Logical, specifying whether to return the ggplot object instead of printing out the plot</p>
</td></tr>
<tr><td><code id="power.plot_+3A_linear.effect.measure">linear.effect.measure</code></td>
<td>
<p>Should the graphs indicate ES values, or R2 values? (default ES)</p>
</td></tr>
<tr><td><code id="power.plot_+3A_select.alpha">select.Alpha</code></td>
<td>
<p>Only produce graphs for the specified Alpha level(s).</p>
</td></tr>
<tr><td><code id="power.plot_+3A_select.or">select.OR</code></td>
<td>
<p>Only produce graphs for the specified odds ratio(s).</p>
</td></tr>
<tr><td><code id="power.plot_+3A_select.es">select.ES</code></td>
<td>
<p>Only produce graphs for the specified effect sizes(s).</p>
</td></tr>
<tr><td><code id="power.plot_+3A_select.n">select.N</code></td>
<td>
<p>Only produce graphs for the specified sample size(s).</p>
</td></tr>
<tr><td><code id="power.plot_+3A_select.maf">select.MAF</code></td>
<td>
<p>Only produce graphs for the specified minor allele frequency(ies).</p>
</td></tr>
<tr><td><code id="power.plot_+3A_select.case.rate">select.Case.Rate</code></td>
<td>
<p>Only produce graphs for the specified case rate(s).</p>
</td></tr>
<tr><td><code id="power.plot_+3A_select.sd">select.SD</code></td>
<td>
<p>Only produce graphs for the specified standard deviation(s).</p>
</td></tr>
<tr><td><code id="power.plot_+3A_select.true.model">select.True.Model</code></td>
<td>
<p>Only produce graphs for the specified true genetic model(s): &quot;Additive&quot;, &quot;Dominant&quot;, &quot;Recessive&quot;.</p>
</td></tr>
<tr><td><code id="power.plot_+3A_select.test.model">select.Test.Model</code></td>
<td>
<p>Only produce graphs for the specified testing model(s): &quot;Additive&quot;, &quot;Dominant&quot;, &quot;Recessive&quot;, &quot;2df&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A series of plots with power on the Y axis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pw &lt;- power.calc(N=1000, Case.Rate=c(0.5), k=NULL,
    MAF=seq(0.15, 0.2, 0.01), OR=1.5,Alpha=c(0.05),
    True.Model='All', Test.Model='All')

</code></pre>

<hr>
<h2 id='quad_roots'>Function to Solve Quadratic Equations</h2><span id='topic+quad_roots'></span>

<h3>Description</h3>

<p>Finds the positive root of a quadratic equation ax^2 + bx + c .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quad_roots(a, b, c)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="quad_roots_+3A_a">a</code></td>
<td>
<p>the coefficient for x^2</p>
</td></tr>
<tr><td><code id="quad_roots_+3A_b">b</code></td>
<td>
<p>the coefficient for x</p>
</td></tr>
<tr><td><code id="quad_roots_+3A_c">c</code></td>
<td>
<p>the constant</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The positive root of the quadratic equation ax^2 + bx + c
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pw&lt;-power.calc(N=c(1000,2000), Case.Rate=c(0.5), 
	k=NULL, MAF=seq(0.05, 0.1, 0.01), OR=c(3,4),
	Alpha=c(0.05), True.Model='All', Test.Model='All')

</code></pre>

<hr>
<h2 id='rec.fun.t'>Function to Calculate t matrix for logistic outcome with binary environment interaction in recessive model</h2><span id='topic+rec.fun.t'></span>

<h3>Description</h3>

<p>Calculates the t matrix for logistic outcome with binary environment interaction in recessive model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rec.fun.t(MAF, P_e, OR_E, OR_G, OR_GE, Case.Rate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rec.fun.t_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="rec.fun.t_+3A_p_e">P_e</code></td>
<td>
<p>Vector of proportions of the population with exposure to the environmental effect</p>
</td></tr>
<tr><td><code id="rec.fun.t_+3A_or_e">OR_E</code></td>
<td>
<p>Vector of environmental odds ratios to detect</p>
</td></tr>
<tr><td><code id="rec.fun.t_+3A_or_g">OR_G</code></td>
<td>
<p>Vector of genetic odds ratios to detect</p>
</td></tr>
<tr><td><code id="rec.fun.t_+3A_or_ge">OR_GE</code></td>
<td>
<p>Vector of genetic/environmental interaction odds ratios to detect</p>
</td></tr>
<tr><td><code id="rec.fun.t_+3A_case.rate">Case.Rate</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>t matrix for all combinations of environment/outcome
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rec.fun.t(MAF = 0.1, P_e = 0.2, Case.Rate = 0.5,
	OR_G = 1.5, OR_E = 2, OR_GE = 1.8)

</code></pre>

<hr>
<h2 id='rec.or.function'>Recessive Model Function</h2><span id='topic+rec.or.function'></span>

<h3>Description</h3>

<p>Operates within odds_ratio_function to calculate odds ratios for a Test.Model of &quot;Recessive&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rec.or.function(like, Case.Rate, P_AA, P_AB, P_BB, True.Model, risk_allele)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rec.or.function_+3A_like">like</code></td>
<td>
<p>Expected log likelihood</p>
</td></tr>
<tr><td><code id="rec.or.function_+3A_case.rate">Case.Rate</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
<tr><td><code id="rec.or.function_+3A_p_aa">P_AA</code></td>
<td>
<p>Probability the allele is homozygous for the major allele</p>
</td></tr>
<tr><td><code id="rec.or.function_+3A_p_ab">P_AB</code></td>
<td>
<p>Probability the allele is heterozygous</p>
</td></tr>
<tr><td><code id="rec.or.function_+3A_p_bb">P_BB</code></td>
<td>
<p>Probability the allele is homozygous for the minor allele</p>
</td></tr>
<tr><td><code id="rec.or.function_+3A_true.model">True.Model</code></td>
<td>
<p>A vector object specifying the true underlying genetic model(s): 'Dominant', 'Additive', or 'Recessive'</p>
</td></tr>
<tr><td><code id="rec.or.function_+3A_risk_allele">risk_allele</code></td>
<td>
<p>Logical: If OR &gt; 1, the allele is classified as a &quot;risk allele&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>: The odds ratios and their corresponding genetic model(s)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rec.or.function(like=-0.57162, Case.Rate=0.3, P_AA=0.5625, P_AB=0.375, 
 P_BB=0.0625, True.Model="Recessive", risk_allele=TRUE)

</code></pre>

<hr>
<h2 id='recessive.ll'>Function to Calculate Recessive Log Likelihood for a Logistic Regression Model</h2><span id='topic+recessive.ll'></span>

<h3>Description</h3>

<p>Calculates the log likelihood for a given set of logistic regression coefficients under a recessive genetic model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recessive.ll(beta, t)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="recessive.ll_+3A_beta">beta</code></td>
<td>
<p>Vector of logistic regression coefficients.</p>
</td></tr>
<tr><td><code id="recessive.ll_+3A_t">t</code></td>
<td>
<p>A 2x3 table of joint probabilities of disease and genotype.  Rows = case vs. control, columns=genotype.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>recessive.ll(c(-0.3793525, -1.1395417), 
 rbind(c(0.2339079, 0.05665039, 0.009441731), 
 c(0.3285921, 0.31834961, 0.053058269)))

</code></pre>

<hr>
<h2 id='recessive.ll.linear'>Function to Calculate Recessive Log Likelihood for a Linear Regression Model</h2><span id='topic+recessive.ll.linear'></span>

<h3>Description</h3>

<p>Calculates the log likelihood for a given set of linear regression coefficients under a recessive genetic model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recessive.ll.linear(beta, m, es, sd_y_x_model, sd_y_x_truth)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="recessive.ll.linear_+3A_beta">beta</code></td>
<td>
<p>Vector of linear regression coefficients.</p>
</td></tr>
<tr><td><code id="recessive.ll.linear_+3A_m">m</code></td>
<td>
<p>Minor allele frequency.</p>
</td></tr>
<tr><td><code id="recessive.ll.linear_+3A_es">es</code></td>
<td>
<p>Vector of effect sizes with two elements, (mean AB - mean AA) and (mean BB - mean AA).</p>
</td></tr>
<tr><td><code id="recessive.ll.linear_+3A_sd_y_x_model">sd_y_x_model</code></td>
<td>
<p>The standard deviation of Y (the outcome) given X (predictors/genotype) under the test model.</p>
</td></tr>
<tr><td><code id="recessive.ll.linear_+3A_sd_y_x_truth">sd_y_x_truth</code></td>
<td>
<p>The standard deviation of Y given X (predictors/genotype) given genotype under the true model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>recessive.ll.linear(beta = c(0, 3), m = 0.1, es = c(0,3), 
 sd_y_x_model = 0.9544108, sd_y_x_truth = 0.9544108)

</code></pre>

<hr>
<h2 id='solve_a'>Binomial coefficient calculation</h2><span id='topic+solve_a'></span>

<h3>Description</h3>

<p>Operates within add.or.function to solve for 'a' when 'b' is known in an additive model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solve_a(b, cr, P_AA, P_AB, P_BB)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="solve_a_+3A_b">b</code></td>
<td>
<p>The &quot;b&quot; in the binomial function ax^2 + bx + c that arises in solution for the additive OR functions</p>
</td></tr>
<tr><td><code id="solve_a_+3A_cr">cr</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
<tr><td><code id="solve_a_+3A_p_aa">P_AA</code></td>
<td>
<p>Probability the allele is homozygous for the major allele</p>
</td></tr>
<tr><td><code id="solve_a_+3A_p_ab">P_AB</code></td>
<td>
<p>Probability the allele is heterozygous</p>
</td></tr>
<tr><td><code id="solve_a_+3A_p_bb">P_BB</code></td>
<td>
<p>Probability the allele is homozygous for the minor allele</p>
</td></tr>
</table>


<h3>Value</h3>

<p>: The &quot;a&quot; in the binomial function ax^2 + bx + c that arises in solution for the additive OR functions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>solve_a(0.1493558, 0.3, 0.5625, 0.375, 0.062)

</code></pre>

<hr>
<h2 id='ss_envir.calc'>Function to Calculate Power for Logistic Models with Environment Interaction</h2><span id='topic+ss_envir.calc'></span>

<h3>Description</h3>

<p>Calculates the power to detect an difference in means/effect size/regression coefficient, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ss_envir.calc(
  power = 0.8,
  Case.Rate = NULL,
  k = NULL,
  MAF = NULL,
  OR_G = NULL,
  OR_E = NULL,
  OR_GE = NULL,
  P_e = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ss_envir.calc_+3A_power">power</code></td>
<td>
<p>Vector of the desired power(s)</p>
</td></tr>
<tr><td><code id="ss_envir.calc_+3A_case.rate">Case.Rate</code></td>
<td>
<p>proportion of cases in the sample (cases/(cases + controls)).</p>
</td></tr>
<tr><td><code id="ss_envir.calc_+3A_k">k</code></td>
<td>
<p>Vector of the number of controls per case. Either k or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="ss_envir.calc_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="ss_envir.calc_+3A_or_g">OR_G</code></td>
<td>
<p>Vector of genetic odds ratios to detect</p>
</td></tr>
<tr><td><code id="ss_envir.calc_+3A_or_e">OR_E</code></td>
<td>
<p>Vector of environmental odds ratios to detect</p>
</td></tr>
<tr><td><code id="ss_envir.calc_+3A_or_ge">OR_GE</code></td>
<td>
<p>Vector of genetic/environmental interaction odds ratios to detect</p>
</td></tr>
<tr><td><code id="ss_envir.calc_+3A_p_e">P_e</code></td>
<td>
<p>Vector of proportions of the population with exposure to the environmental effect</p>
</td></tr>
<tr><td><code id="ss_envir.calc_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="ss_envir.calc_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive1', 'Additive2', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="ss_envir.calc_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, ES, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ssc &lt;- ss_envir.calc(P_e = 0.2, MAF = 0.1, power = 0.6, Case.Rate = 0.5, Alpha = 0.05, 
	OR_G = 1.5, OR_E = 2, OR_GE = 1.8, Test.Model = "All", True.Model = "All")


</code></pre>

<hr>
<h2 id='ss_envir.calc.linear_outcome'>Function to Calculate Power for Linear Models with logistic environment interaction</h2><span id='topic+ss_envir.calc.linear_outcome'></span>

<h3>Description</h3>

<p>Calculates the power to detect an difference in means/effect size/regression coefficient, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ss_envir.calc.linear_outcome(
  pow = NULL,
  MAF = NULL,
  ES_G = NULL,
  ES_E = NULL,
  ES_GE = NULL,
  P_e = NULL,
  R2_G = NULL,
  R2_E = NULL,
  R2_GE = NULL,
  sd_y = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_pow">pow</code></td>
<td>
<p>Vector of the desired power(s)</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_es_g">ES_G</code></td>
<td>
<p>Vector of genetic effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_es_e">ES_E</code></td>
<td>
<p>Vector of environmental effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_es_ge">ES_GE</code></td>
<td>
<p>Vector of genetic/environment interaction effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_p_e">P_e</code></td>
<td>
<p>Vector of proportions of the population with exposure to the environmental effect</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_r2_g">R2_G</code></td>
<td>
<p>Vector of genetic R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_r2_e">R2_E</code></td>
<td>
<p>Vector of environmental R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_r2_ge">R2_GE</code></td>
<td>
<p>Vector of genetic/environment interaction R-squared values Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_sd_y">sd_y</code></td>
<td>
<p>Standard deviation of the outcome in the population (ignoring genotype). Either sd_y_x or sd_y must be specified.</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="ss_envir.calc.linear_outcome_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, ES, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ss_envir.calc.linear_outcome(pow=0.8, ES_G = 1.2, ES_E = 1.3, 
	ES_GE = 2, Alpha = 0.05, MAF = 0.1, P_e = 0.2, 
	sd_y = 10, True.Model = "All", Test.Model = "All")

</code></pre>

<hr>
<h2 id='ss_linear_envir.calc.linear_outcome'>Function to Calculate Power for Linear Models with linear environment interaction</h2><span id='topic+ss_linear_envir.calc.linear_outcome'></span>

<h3>Description</h3>

<p>Calculates the power to detect an difference in means/effect size/regression coefficient, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ss_linear_envir.calc.linear_outcome(
  pow = NULL,
  MAF = NULL,
  ES_G = NULL,
  ES_E = NULL,
  ES_GE = NULL,
  sd_e = NULL,
  R2_G = NULL,
  R2_E = NULL,
  R2_GE = NULL,
  sd_y = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_pow">pow</code></td>
<td>
<p>Vector of the desired power(s)</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_es_g">ES_G</code></td>
<td>
<p>Vector of genetic effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_es_e">ES_E</code></td>
<td>
<p>Vector of environmental effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_es_ge">ES_GE</code></td>
<td>
<p>Vector of genetic/environment interaction effect sizes (difference in means) to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_sd_e">sd_e</code></td>
<td>
<p>Standard deviation of the environmental variable</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_r2_g">R2_G</code></td>
<td>
<p>Vector of genetic R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_r2_e">R2_E</code></td>
<td>
<p>Vector of environmental R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_r2_ge">R2_GE</code></td>
<td>
<p>Vector of genetic/environment interaction R-squared values to detect. Either ES_G, ES_E, and ES_EG or R2_G, R2_E, and R2_EG must be specified.</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_sd_y">sd_y</code></td>
<td>
<p>Standard deviation of the outcome in the population (ignoring genotype). Either sd_y_x or sd_y must be specified.</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.linear_outcome_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, ES, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ss_linear_envir.calc.linear_outcome(pow = 0.8, 
	ES_G=0.5, ES_E=1.6, ES_GE=1.4, 
	sd_e = 1, MAF=0.28, 
	sd_y = 5,Alpha=0.05,
	True.Model='All', Test.Model='All')

</code></pre>

<hr>
<h2 id='ss_linear_envir.calc.logistic_outcome'>Function to Calculate Sample Size for Linear Models with logistic environment interaction</h2><span id='topic+ss_linear_envir.calc.logistic_outcome'></span>

<h3>Description</h3>

<p>Calculates the power to detect an difference in means/effect size/regression coefficient, at a given sample size, N, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ss_linear_envir.calc.logistic_outcome(
  power = NULL,
  MAF = NULL,
  OR_G = NULL,
  OR_E = NULL,
  OR_GE = NULL,
  sd_e = NULL,
  Case.Rate = NULL,
  k = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ss_linear_envir.calc.logistic_outcome_+3A_power">power</code></td>
<td>
<p>Vector of the desired power(s)</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.logistic_outcome_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.logistic_outcome_+3A_or_g">OR_G</code></td>
<td>
<p>Vector of genetic odds ratios to detect</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.logistic_outcome_+3A_or_e">OR_E</code></td>
<td>
<p>Vector of environmental odds ratios to detect</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.logistic_outcome_+3A_or_ge">OR_GE</code></td>
<td>
<p>Vector of genetic/environmental interaction odds ratios to detect</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.logistic_outcome_+3A_sd_e">sd_e</code></td>
<td>
<p>Standard deviation of the environmental variable</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.logistic_outcome_+3A_case.rate">Case.Rate</code></td>
<td>
<p>Standard deviation of the outcome in the population (ignoring genotype). Either Case.Rate_x or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.logistic_outcome_+3A_k">k</code></td>
<td>
<p>Vector of the number of controls per case. Either k or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.logistic_outcome_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.logistic_outcome_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="ss_linear_envir.calc.logistic_outcome_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the power for all combinations of the specified parameters (Case.Rate, ES, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ss &lt;- ss_linear_envir.calc.logistic_outcome(power=0.8, 
	OR_G=1.1, OR_E=1.2, OR_GE=1.5, 
	sd_e = 1, MAF=0.2, Case.Rate = 0.2,
	Alpha=0.05, True.Model="All", Test.Model="All")


</code></pre>

<hr>
<h2 id='ss.calc'>Function to Calculate Sample Size</h2><span id='topic+ss.calc'></span>

<h3>Description</h3>

<p>Calculates the necessary sample size to achieve the specified level of power to detect an odds ratio, OR, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ss.calc(
  power = 0.8,
  Case.Rate = NULL,
  k = NULL,
  MAF = NULL,
  OR = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ss.calc_+3A_power">power</code></td>
<td>
<p>Vector of the desired power(s)</p>
</td></tr>
<tr><td><code id="ss.calc_+3A_case.rate">Case.Rate</code></td>
<td>
<p>Vector of the proportion(s) of cases in the sample (cases/(cases + controls)).  Either k or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="ss.calc_+3A_k">k</code></td>
<td>
<p>Vector of the number of controls per case. Either k or Case.Rate must be specified.</p>
</td></tr>
<tr><td><code id="ss.calc_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="ss.calc_+3A_or">OR</code></td>
<td>
<p>Vector of odds ratios to detect</p>
</td></tr>
<tr><td><code id="ss.calc_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="ss.calc_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="ss.calc_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the total number of subjects required for all combinations of the specified parameters (Case.Rate, OR, Power, etc)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ss &lt;- ss.calc(power=0.8, Case.Rate=0.5, k=NULL,
   MAF=0.1, OR=3,Alpha=0.05,
   True.Model='All', Test.Model='All')

</code></pre>

<hr>
<h2 id='ss.calc.linear'>Function to Calculate Sample Size in Linear Models</h2><span id='topic+ss.calc.linear'></span>

<h3>Description</h3>

<p>Calculates the necessary sample size to acheive the specified level of power to detect an effect size, ES or R2 value, with type 1 error rate, Alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ss.calc.linear(
  power = 0.8,
  MAF = NULL,
  ES = NULL,
  R2 = NULL,
  sd_y = NULL,
  Alpha = 0.05,
  True.Model = "All",
  Test.Model = "All"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ss.calc.linear_+3A_power">power</code></td>
<td>
<p>Vector of the desired power(s)</p>
</td></tr>
<tr><td><code id="ss.calc.linear_+3A_maf">MAF</code></td>
<td>
<p>Vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="ss.calc.linear_+3A_es">ES</code></td>
<td>
<p>Vector of effect sizes (difference in means) to detect. Either ES or R2 must be specified.</p>
</td></tr>
<tr><td><code id="ss.calc.linear_+3A_r2">R2</code></td>
<td>
<p>Vector of R-squared values to detect. Either ES or R2 must be specified.</p>
</td></tr>
<tr><td><code id="ss.calc.linear_+3A_sd_y">sd_y</code></td>
<td>
<p>Standard deviation of the outcome in the population (ignoring genotype). Either sd_y_x or sd_y must be specified.</p>
</td></tr>
<tr><td><code id="ss.calc.linear_+3A_alpha">Alpha</code></td>
<td>
<p>the desired type 1 error rate(s)</p>
</td></tr>
<tr><td><code id="ss.calc.linear_+3A_true.model">True.Model</code></td>
<td>
<p>A vector specifying the true underlying genetic model(s): 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
<tr><td><code id="ss.calc.linear_+3A_test.model">Test.Model</code></td>
<td>
<p>A vector specifying the assumed genetic model(s) used in testing: 'Dominant', 'Additive', 'Recessive' or 'All'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame including the total number of subjects required for all combinations of the specified parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ss &lt;- ss.calc.linear(power=0.8,MAF=0.1,
    ES=3, R2=NULL, sd_y = 1,Alpha=0.05,
    True.Model='All', Test.Model='All')

</code></pre>

<hr>
<h2 id='ss.plot'>Function to Plot Sample Size Results</h2><span id='topic+ss.plot'></span>

<h3>Description</h3>

<p>Plot the sample size results by MAF, OR, Alpha or Power
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ss.plot(
  data = NULL,
  x = "MAF",
  panel.by = "True.Model",
  y_limit = NULL,
  y_log = F,
  return_gg = F,
  linear.effect.measure = "ES",
  select.Alpha = NULL,
  select.OR = NULL,
  select.ES = NULL,
  select.Power = NULL,
  select.MAF = NULL,
  select.Case.Rate = NULL,
  select.SD = NULL,
  select.True.Model = NULL,
  select.Test.Model = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ss.plot_+3A_data">data</code></td>
<td>
<p>The data frame result from <code><a href="#topic+ss.calc">ss.calc</a></code></p>
</td></tr>
<tr><td><code id="ss.plot_+3A_x">x</code></td>
<td>
<p>The desired variable on the y axis: &quot;MAF&quot;, &quot;OR&quot;, &quot;ES&quot;,&quot;Alpha&quot;, or &quot;Power&quot;</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_panel.by">panel.by</code></td>
<td>
<p>A grouping variable to panel the graphs by: &quot;True.Model&quot;, &quot;MAF&quot;, &quot;OR&quot;, &quot;Alpha&quot;, or &quot;Power&quot;</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_y_limit">y_limit</code></td>
<td>
<p>An object specifying the minimum and maximum of the y-axis (eg c(0,4)) default is NULL, which allows the limits to be picked automatically</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_y_log">y_log</code></td>
<td>
<p>Logical, specifying whether the y axis should be logarithmic. Default is F</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_return_gg">return_gg</code></td>
<td>
<p>Logical, specifying whether to return the ggplot object instead of printing out the plot</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_linear.effect.measure">linear.effect.measure</code></td>
<td>
<p>Should the graphs indicate ES values, or R2 values? (default ES)</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_select.alpha">select.Alpha</code></td>
<td>
<p>Only produce graphs for the specified Alpha level(s).</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_select.or">select.OR</code></td>
<td>
<p>Only produce graphs for the specified odds ratio(s).</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_select.es">select.ES</code></td>
<td>
<p>Only produce graphs for the specified effect size(s).</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_select.power">select.Power</code></td>
<td>
<p>Only produce graphs for the specified power(s).</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_select.maf">select.MAF</code></td>
<td>
<p>Only produce graphs for the specified minor allele frequency(ies).</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_select.case.rate">select.Case.Rate</code></td>
<td>
<p>Only produce graphs for the specified case rate(s).</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_select.sd">select.SD</code></td>
<td>
<p>Only produce graphs for the specified standard deviation(s).</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_select.true.model">select.True.Model</code></td>
<td>
<p>Only produce graphs for the specified true genetic model(s): &quot;Additive&quot;, &quot;Dominant&quot;, &quot;Recessive&quot;.</p>
</td></tr>
<tr><td><code id="ss.plot_+3A_select.test.model">select.Test.Model</code></td>
<td>
<p>Only produce graphs for the specified testing model(s): &quot;Additive&quot;, &quot;Dominant&quot;, &quot;Recessive&quot;, &quot;2df&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A series of plots with sample size on the Y axis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ss &lt;- ss.calc(power=0.8, Case.Rate=c(0.5), k=NULL,
    MAF=seq(0.01, 0.05, 0.01), OR=c(4),Alpha=c(0.05),
    True.Model='All', Test.Model='All')
ss.plot(data=ss, x='MAF',panel.by='OR')

</code></pre>

<hr>
<h2 id='X_mat_returner'>Function to output X matrices used in calculation of MLE's for linear outcome with logistic environment interaction</h2><span id='topic+X_mat_returner'></span>

<h3>Description</h3>

<p>Returns X matrices used in calculation of MLE's for linear outcome with logistic environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>X_mat_returner(mod, reduced = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="X_mat_returner_+3A_mod">mod</code></td>
<td>
<p>type of model</p>
</td></tr>
<tr><td><code id="X_mat_returner_+3A_reduced">reduced</code></td>
<td>
<p>logical, indicates whether the X matrix will be used for a reduced model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix to be used in MLE calculation for linear outcome with logistic environment interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X_mat_returner(mod = "Dominant")

</code></pre>

<hr>
<h2 id='X_mat_returner_lle'>Function to output X matrices used in calculation of MLE's for linear outcome with linear environment interaction</h2><span id='topic+X_mat_returner_lle'></span>

<h3>Description</h3>

<p>Function to output X matrices used in calculation of MLE's for linear outcome with linear environment interaction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>X_mat_returner_lle(mod)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="X_mat_returner_lle_+3A_mod">mod</code></td>
<td>
<p>type of model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A probability vector to be used in MLE calculation for linear outcome with linear environment interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X_mat_returner_lle("Dominant")

</code></pre>

<hr>
<h2 id='zero_finder_nleqslv'>Zero finder</h2><span id='topic+zero_finder_nleqslv'></span>

<h3>Description</h3>

<p>Finds zeros of multinomial functions using the nleqslv package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zero_finder_nleqslv(
  afun,
  veclength,
  tol = 0.4,
  x.start.vals = NULL,
  upper.lim = Inf
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zero_finder_nleqslv_+3A_afun">afun</code></td>
<td>
<p>The function to find zeros</p>
</td></tr>
<tr><td><code id="zero_finder_nleqslv_+3A_veclength">veclength</code></td>
<td>
<p>The dimension of the system of equations</p>
</td></tr>
<tr><td><code id="zero_finder_nleqslv_+3A_tol">tol</code></td>
<td>
<p>The range within which to set start values for the function to use to find zeros</p>
</td></tr>
<tr><td><code id="zero_finder_nleqslv_+3A_x.start.vals">x.start.vals</code></td>
<td>
<p>Optional user defined start values</p>
</td></tr>
<tr><td><code id="zero_finder_nleqslv_+3A_upper.lim">upper.lim</code></td>
<td>
<p>to be used if there is to be an upper limit to the solution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Predicted zeros of the given equation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>afun &lt;- function(x) {
	y &lt;- numeric(2)
	y[1] &lt;- x[1]^2 + x[2]^2 - 1
	y[2] &lt;- exp(x[1]-1) + x[2]^3 - 1.1
	y
}

zero_finder_nleqslv(afun, veclength = 2)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
