<!DOCTYPE html><html><head><title>Help for package GenWin</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GenWin}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#chr6'><p>Maize Fst Data</p></a></li>
<li><a href='#splineAnalyze'><p>Spline-based window analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Spline Based Window Boundaries for Genomic Analyses</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-11-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Timothy M. Beissinger &lt;timbeissinger@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Timothy M. Beissinger &lt;timbeissinger@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Defines window or bin boundaries for the analysis of genomic data.
    Boundaries are based on the inflection points of a cubic smoothing spline
    fitted to the raw data. Along with defining boundaries, a technique to
    evaluate results obtained from unequally-sized windows is provided.
    Applications are particularly pertinent for, though not limited to, genome
    scans for selection based on variability between populations (e.g. using
    Wright's fixations index, Fst, which measures variability in subpopulations
    relative to the total population).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>pspline, stats, graphics</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-25 08:04:04 UTC; beissinger</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-25 12:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='chr6'>Maize Fst Data</h2><span id='topic+chr6'></span>

<h3>Description</h3>

<p>Data from 85,388 SNPs on chromosome 6 of a maize, in a selection experiment
involving 30 generations of selection for increased number of ears per plant
The variables are:
</p>

<ul>
<li><p> Chromosome. Chromosome 6, always.
</p>
</li>
<li><p> Position. SNP position on maize reference version 2.
</p>
</li>
<li><p> C0Freq. Frequency of minor allele in cycle 0.
</p>
</li>
<li><p> C30Freq. Frequency of same allele in C30 (not necessarily minor)
</p>
</li>
<li><p> Fst. Fst value based on C0 and C30 frequencies.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(chr6)
</code></pre>


<h3>Format</h3>

<p>A data frame with 85,388 rows and 5 columns.
</p>


<h3>References</h3>

<p>Beissinger et al. (2014) Genetics 196(3):829-840
(http://www.ncbi.nlm.nih.gov/pubmed/24381334)
</p>

<hr>
<h2 id='splineAnalyze'>Spline-based window analysis</h2><span id='topic+splineAnalyze'></span>

<h3>Description</h3>

<p>Defines window boundaries for analyzing genetic data and evaluates the specified windows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splineAnalyze(
  Y,
  map,
  smoothness = 100,
  s2 = NA,
  mean = NA,
  plotRaw = FALSE,
  plotWindows = FALSE,
  method = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="splineAnalyze_+3A_y">Y</code></td>
<td>
<p>A vector of estimates of some parameter, such as Fst, computed at individual markers. One chromosome should be analyzed at a time.</p>
</td></tr>
<tr><td><code id="splineAnalyze_+3A_map">map</code></td>
<td>
<p>A vector of positions for each marker with a corresponding parameter estimate.</p>
</td></tr>
<tr><td><code id="splineAnalyze_+3A_smoothness">smoothness</code></td>
<td>
<p>The level of resolution (in base pairs) for computing the spline and its derivatives</p>
</td></tr>
<tr><td><code id="splineAnalyze_+3A_s2">s2</code></td>
<td>
<p>The variance of parameter estimates, to be used for computing Wstats. Default is to compute this automatically, but it may be manually specified, e.g. so that the value across chromosomes may be utilized.</p>
</td></tr>
<tr><td><code id="splineAnalyze_+3A_mean">mean</code></td>
<td>
<p>The mean of parameter estimates, to be used for computing Wstats. Default is to comput this automatically, but it may be manually specified, e.g. so that the value across chromosomes may be utilized.</p>
</td></tr>
<tr><td><code id="splineAnalyze_+3A_plotraw">plotRaw</code></td>
<td>
<p>Whether or not to produce a plot of raw data, with the fitted spline</p>
</td></tr>
<tr><td><code id="splineAnalyze_+3A_plotwindows">plotWindows</code></td>
<td>
<p>Whether or not to include a plot of Wstat values over the computed windows</p>
</td></tr>
<tr><td><code id="splineAnalyze_+3A_method">method</code></td>
<td>
<p>The method for controlling amount of smoothing: 1, 2, 3, or 4. See documentation of smooth.Pspline for description. Usual choices are either 3 for generalized cross validation or 4 for ordinary cross validation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>rawSpline The fitted spline object
</p>
<p>breaks The spline-suggested window breaks
</p>
<p>windowData A table of mean parameter estimates and Wstats computed over spline-suggested windows
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(chr6)
sub6 &lt;- chr6[55000:63000,]
chr6Spline &lt;- splineAnalyze(Y=sub6$Fst,map=sub6$Position,smoothness=100,
plotRaw=TRUE,plotWindows=TRUE,method=4)

## Not run: 
chr6Spline &lt;- splineAnalyze(Y=chr6$Fst,map=chr6$Position,smoothness=100,
plotRaw=TRUE,plotWindows=TRUE,method=4)

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
