<!DOCTYPE html><html lang="en"><head><title>Help for package GeRnika</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GeRnika}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.distribute_freqs'><p>Distribute frequencies among clone and its children clones</p></a></li>
<li><a href='#add_noise'><p>Add noise to the VAF values in an F matrix</p></a></li>
<li><a href='#B_mats'><p>A set of 10 trios of B matrices for experimenting with the methods of <code>GeRnika</code></p></a></li>
<li><a href='#B_to_phylotree'><p>Create a <code>Phylotree</code> object from a <code>B</code> matrix.</p></a></li>
<li><a href='#calc_clone_proportions'><p>Calculate clone proportions for a tumor</p></a></li>
<li><a href='#combine_trees'><p>Get consensus tree between two phylogenetic trees</p></a></li>
<li><a href='#create_B'><p>Create tumor phylogenetic tree topology</p></a></li>
<li><a href='#create_instance'><p>Create a tumor phylogenetic tree instance</p></a></li>
<li><a href='#create_phylotree'><p>Create a <code>Phylotree</code> object</p></a></li>
<li><a href='#create_U'><p>Calculate tumor clone frequencies in samples</p></a></li>
<li><a href='#equals'><p>Check if two phylogenetic trees are equal</p></a></li>
<li><a href='#find_common_subtrees'><p>Find the set of common subtrees between two phylogenetic trees</p></a></li>
<li><a href='#get_clones'><p>Retrieve the clone indices for a set of gene indices</p></a></li>
<li><a href='#get_genes'><p>Retrieve the gene indices for the clones in a phylogenetic tree</p></a></li>
<li><a href='#get_parents'><p>Get parent nodes in a phylogenetic tree.</p></a></li>
<li><a href='#hyperparameters'><p>Hyperparameters for the methods of <code>GeRnika</code></p></a></li>
<li><a href='#Node-class'><p>S4 class to represent a node in a phylogenetic tree.</p></a></li>
<li><a href='#palettes'><p>Palettes for the methods of <code>GeRnika</code></p></a></li>
<li><a href='#phylotree_to_B'><p>Get B matrix from <code>Phylotree</code> object.</p></a></li>
<li><a href='#phylotree_to_tree'><p>Extract tree from <code>Phylotree</code> object.</p></a></li>
<li><a href='#Phylotree-class'><p>S4 class to represent phylogenetic trees.</p></a></li>
<li><a href='#place_clones_space'><p>Create a model for the spatial distribution of the clones in a tumor</p></a></li>
<li><a href='#plot'><p>Plot a Phylotree object.</p></a></li>
<li><a href='#plot_p'><p>Plot a phylogenetic tree with proportional node sizes and colors.</p></a></li>
<li><a href='#plot_phylotree'><p>Plot a phylogenetic tree</p></a></li>
<li><a href='#plot_proportions'><p>Plot a phylogenetic tree with proportional node sizes and colors</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulation, Visualization and Comparison of Tumor Evolution Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulating, visualizing and comparing tumor clonal data by using simple commands. This aims at providing a tool to help researchers to easily simulate tumor data and analyze the results of their approaches for studying the composition and the evolutionary history of tumors.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.tree, purrr, reshape2, DiagrammeR, colorspace, dplyr,
methods, vctrs, magrittr, stats</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.00)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown, knitcitations, ggpubr, ggplot2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, knitcitations, ggpubr, ggplot2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-08-31 17:15:39 UTC; aitor</td>
</tr>
<tr>
<td>Author:</td>
<td>Aitor Sánchez-Ferrera
    <a href="https://orcid.org/0000-0001-6127-0686"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre, aut],
  Maitena Tellaetxe-Abete
    <a href="https://orcid.org/0000-0003-1894-4547"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Borja Calvo <a href="https://orcid.org/0000-0001-9969-9664"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Aitor Sánchez-Ferrera &lt;aitor.sanchezf@ehu.eus&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-04 13:30:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='.distribute_freqs'>Distribute frequencies among clone and its children clones</h2><span id='topic+.distribute_freqs'></span>

<h3>Description</h3>

<p>This function distributes frequencies among the clones in a multifurcation of a phylogenetic tree. It uses a Dirichlet distribution to generate random proportions for a clone and its children, and then normalizes the children's proportions to the parent's proportion.
This is an internal function used by <code>calc_clone_proportions</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.distribute_freqs(B, clone_idx, clone_proportions, selection)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id=".distribute_freqs_+3A_b">B</code></td>
<td>
<p>A matrix representing the mutation relationships between the nodes in the phylogenetic tree (B matrix).</p>
</td></tr>
<tr><td><code id=".distribute_freqs_+3A_clone_idx">clone_idx</code></td>
<td>
<p>An integer representing the index of the clone whose own proportion and its children's proportions are going to be updated.</p>
</td></tr>
<tr><td><code id=".distribute_freqs_+3A_clone_proportions">clone_proportions</code></td>
<td>
<p>A numeric vector representing the proportions of each clone in the phylogenetic tree.</p>
</td></tr>
<tr><td><code id=".distribute_freqs_+3A_selection">selection</code></td>
<td>
<p>A character string representing the evolutionary mode the tumor follows. This should be either &quot;positive&quot; or &quot;neutral&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector representing the updated proportions of each clone in the phylogenetic tree.
</p>

<hr>
<h2 id='add_noise'>Add noise to the VAF values in an F matrix</h2><span id='topic+add_noise'></span>

<h3>Description</h3>

<p>This function adds noise to the variant allele frequency (VAF) values in an F matrix, simulating the effect of sequencing errors. The noise is modeled as a negative binomial distribution for the depth of the reads and a binomial distribution for both the variant allele counts and the mismatch counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_noise(F_matrix, depth, overdispersion)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_noise_+3A_f_matrix">F_matrix</code></td>
<td>
<p>A matrix representing the true VAF values of a series of mutations in a set of samples (F matrix).</p>
</td></tr>
<tr><td><code id="add_noise_+3A_depth">depth</code></td>
<td>
<p>A numeric value representing the mean depth of sequencing.</p>
</td></tr>
<tr><td><code id="add_noise_+3A_overdispersion">overdispersion</code></td>
<td>
<p>A numeric value representing the overdispersion parameter for the negative binomial distribution used to simulate the depth of sequencing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing noisy VAF values of a series of mutations in a set of samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Calculate the noisy VAF values of a series of mutations in a set of samples, given the true 
# VAF values in the F matrix F_true, a depth of 30 and an overdispersion of 5

# Create an instance of a tumor with 50 clones,
# 10 samples, k = 5, positive selection without noise
F_true &lt;- create_instance(
  n = 50,
  m = 10,
  k = 5,
  selection = "positive", 
  noisy = FALSE)$F

# Then we add the noise using a depth of 30 and an overdispersion of 5.
noisy_F &lt;- add_noise(F_true, 30, 5)

</code></pre>

<hr>
<h2 id='B_mats'>A set of 10 trios of B matrices for experimenting with the methods of <code>GeRnika</code></h2><span id='topic+B_mats'></span>

<h3>Description</h3>

<p>A list of lists composed by 10 trios of B matrices; a real B matrix, a B matrix got by using the GRASP method and another one as a result of an ILS. These matrices can be used as examples for the methods of <code>GeRnika</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>B_mats
</code></pre>


<h3>Format</h3>

<p>A list of lists composed by 10 trios of B matrices.
</p>

<dl>
<dt>Trio 1</dt><dd><p>B_real, B_grasp and B_opt (matrices composed by 5 clones)</p>
</dd>
<dt>Trio 2</dt><dd><p>B_real, B_grasp and B_opt (matrices composed by 5 clones)</p>
</dd>
<dt>Trio 3</dt><dd><p>B_real, B_grasp and B_opt (matrices composed by 5 clones)</p>
</dd>
<dt>Trio 4</dt><dd><p>B_real, B_grasp and B_opt (matrices composed by 5 clones)</p>
</dd>
<dt>Trio 5</dt><dd><p>B_real, B_grasp and B_opt (matrices composed by 5 clones)</p>
</dd>
<dt>Trio 6</dt><dd><p>B_real, B_grasp and B_opt (matrices composed by 10 clones)</p>
</dd>
<dt>Trio 7</dt><dd><p>B_real, B_grasp and B_opt (matrices composed by 10 clones)</p>
</dd>
<dt>Trio 8</dt><dd><p>B_real, B_grasp and B_opt (matrices composed by 10 clones)</p>
</dd>
<dt>Trio 9</dt><dd><p>B_real, B_grasp and B_opt (matrices composed by 10 clones)</p>
</dd>
<dt>Trio 10</dt><dd><p>B_real, B_grasp and B_opt (matrices composed by 10 clones)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Local source; as a result of the Grasp and the ILS methods used for solving the Clonal Deconvolution and Evolution Problem (CDEP).
</p>

<hr>
<h2 id='B_to_phylotree'>Create a <code>Phylotree</code> object from a <code>B</code> matrix.</h2><span id='topic+B_to_phylotree'></span>

<h3>Description</h3>

<p>This function creates a <code>Phylotree</code> class object from a <code>B</code> matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>B_to_phylotree(B, labels = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="B_to_phylotree_+3A_b">B</code></td>
<td>
<p>A square matrix that represents the phylogenetic tree.</p>
</td></tr>
<tr><td><code id="B_to_phylotree_+3A_labels">labels</code></td>
<td>
<p>An optional vector containing the tags of the genes in the phylogenetic tree. <code>NA</code> by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>Phylotree</code> class object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a B matrix instance
# composed by 10 subpopulations of
# clones
B &lt;- create_instance(
       n = 10, 
       m = 4, 
       k = 1, 
       selection = "neutral")$B

# Create a new 'Phylotree' object
# on the basis of the B matrix
phylotree &lt;- B_to_phylotree(B = B)

# Generate the tags for the genes of
# the phyogenetic tree
tags &lt;- LETTERS[1:nrow(B)]

# Create a new 'Phylotree' object
# on the basis of the B matrix and
# the list of tags
phylotree_tags &lt;- B_to_phylotree(
                    B = B, 
                    labels = tags)
</code></pre>

<hr>
<h2 id='calc_clone_proportions'>Calculate clone proportions for a tumor</h2><span id='topic+calc_clone_proportions'></span>

<h3>Description</h3>

<p>This function calculates the proportions of each clone in a phylogenetic tree, following a given evolutionary mode (positive selection or neutral evolution).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_clone_proportions(B, selection)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_clone_proportions_+3A_b">B</code></td>
<td>
<p>A matrix representing the mutation relationships between the nodes in the phylogenetic tree (B matrix).</p>
</td></tr>
<tr><td><code id="calc_clone_proportions_+3A_selection">selection</code></td>
<td>
<p>A character string representing the evolutionary mode the tumor follows. This should be either &quot;positive&quot; or &quot;neutral&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with two columns: 'clone_idx', which contains the clone identifiers, and 'proportion', which contains the calculated proportions of each clone.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Calculate clone proportions for a tumor phylogenetic tree represented by a B matrix 
# and following a positive selection model

# Create a mutation matrix for a phylogenetic tree with 10 nodes and k = 2
B_mat &lt;- create_B(10, 2)

# Calculate the clone proportions following a positive selection model
clone_proportions &lt;- calc_clone_proportions(B_mat, "positive")

</code></pre>

<hr>
<h2 id='combine_trees'>Get consensus tree between two phylogenetic trees</h2><span id='topic+combine_trees'></span>

<h3>Description</h3>

<p>Returns a graph representing the consensus tree between two phylogenetic trees.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_trees(
  phylotree_1,
  phylotree_2,
  palette = GeRnika::palettes$Simpsons,
  labels = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combine_trees_+3A_phylotree_1">phylotree_1</code></td>
<td>
<p>A <code>Phylotree</code> class object.</p>
</td></tr>
<tr><td><code id="combine_trees_+3A_phylotree_2">phylotree_2</code></td>
<td>
<p>A <code>Phylotree</code> class object.</p>
</td></tr>
<tr><td><code id="combine_trees_+3A_palette">palette</code></td>
<td>
<p>A vector composed by the hexadecimal code of three colors. &quot;The Simpsons&quot; palette used as default.</p>
</td></tr>
<tr><td><code id="combine_trees_+3A_labels">labels</code></td>
<td>
<p>A boolean, if <code>TRUE</code> the resulting graph will be plotted with the tags of the genes in the phylogenetic trees instead of their mutation index. <code>FALSE</code> by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>dgr_graph</code> object representing the consensus graph between <code>phylotree_1</code> <code>phylotree_2</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load the predefined B matrices of the package
B_mats &lt;- GeRnika::B_mats


B_real &lt;- B_mats[[2]]$B_real
B_opt &lt;- B_mats[[2]]$B_opt


# Generate the tags for the genes of
# the phyogenetic tree
tags &lt;- LETTERS[1:nrow(B_real)]


# Instantiate two \code{Phylotree} class objects on 
# the basis of the B matrices
phylotree_real &lt;- B_to_phylotree(
                    B = B_real, 
                    labels = tags)
                    
phylotree_opt &lt;- B_to_phylotree(
                    B = B_opt, 
                    labels = tags)


# Create the consensus tree between phylotree_real
# and phylotree_opt
consensus &lt;- combine_trees(
               phylotree_1 = phylotree_real,
               phylotree_2 = phylotree_opt)
               
               
# Render the consensus tree
DiagrammeR::render_graph(consensus)


# Load another palette
palette_1 &lt;- GeRnika::palettes$Lancet


# Create the consensus tree between phylotree_real
# and phylotree_opt using tags and another palette
consensus_tag &lt;- combine_trees(
                   phylotree_1 = phylotree_real, 
                   phylotree_2 = phylotree_opt,
                   palette = palette_1,
                   labels = TRUE)


# Render the consensus tree using tags and the
# selected palette
DiagrammeR::render_graph(consensus_tag)
</code></pre>

<hr>
<h2 id='create_B'>Create tumor phylogenetic tree topology</h2><span id='topic+create_B'></span>

<h3>Description</h3>

<p>This function generates a mutation matrix (B matrix) for a tumor phylogenetic tree with a given number of nodes. This matrix represents the topology and it is created randomly, with the probability of a node to be chosen as a parent of a new node being proportional to the number of its ascendants raised to the power of a constant 'k'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_B(n, k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_B_+3A_n">n</code></td>
<td>
<p>An integer representing the number of nodes in the phylogenetic tree.</p>
</td></tr>
<tr><td><code id="create_B_+3A_k">k</code></td>
<td>
<p>A numeric value representing the constant used to calculate the probability of a node to be chosen as a parent of a new node.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A square matrix representing the mutation relationships between the nodes in the phylogenetic tree. Each row corresponds to a node, and each column corresponds to a mutation. The value at the i-th row and j-th column is 1 if the i-th node has the j-th mutation, and 0 otherwise.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Create a mutation matrix for a phylogenetic tree with 10 nodes and k = 2
B &lt;- create_B(10, 2)

</code></pre>

<hr>
<h2 id='create_instance'>Create a tumor phylogenetic tree instance</h2><span id='topic+create_instance'></span>

<h3>Description</h3>

<p>This function generates a tumor phylogenetic tree instance, composed by a mutation matrix (B matrix), a matrix of true variant allele frequencies (F_true), a matrix of noisy variant allele frequencies (F), and a matrix of clone frequencies in samples (U).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_instance(
  n,
  m,
  k,
  selection,
  noisy = TRUE,
  depth = 30,
  seed = Sys.time()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_instance_+3A_n">n</code></td>
<td>
<p>An integer representing the number of clones.</p>
</td></tr>
<tr><td><code id="create_instance_+3A_m">m</code></td>
<td>
<p>An integer representing the number of samples.</p>
</td></tr>
<tr><td><code id="create_instance_+3A_k">k</code></td>
<td>
<p>A numeric value that determines the linearity of the tree topology. Also referred to as the topology parameter. Increasing values of this parameter increase the linearity of the topology. When 'k' is set to 1, all nodes have equal probabilities of being chosen as parents, resulting in a completely random topology.</p>
</td></tr>
<tr><td><code id="create_instance_+3A_selection">selection</code></td>
<td>
<p>A character string representing the evolutionary mode the tumor follows. This should be either &quot;positive&quot; or &quot;neutral&quot;.</p>
</td></tr>
<tr><td><code id="create_instance_+3A_noisy">noisy</code></td>
<td>
<p>A logical value indicating whether to add noise to the frequency matrix. If 'TRUE', noise is added to the frequency matrix. If 'FALSE', no noise is added. 'TRUE' by default.</p>
</td></tr>
<tr><td><code id="create_instance_+3A_depth">depth</code></td>
<td>
<p>A numeric value representing the mean depth of sequencing. 30 by default.</p>
</td></tr>
<tr><td><code id="create_instance_+3A_seed">seed</code></td>
<td>
<p>A numeric value used to set the seed for the random number generator. Sys.time() by default.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The B matrix is a square matrix representing the mutation relationships between the clones in the tumor, or, in other words, it represents the topology of the phylogenetic tree. The F_true matrix represents the true variant allele frequencies of the mutations present in the tumor in a set of samples. The F matrix represents the noisy variant allele frequencies of the mutations in the same set of samples. The U matrix represents the frequencies of the clones in the tumor in the set of samples.
</p>


<h3>Value</h3>

<p>A list containing four elements: 'F', a matrix representing the noisy frequencies of each mutation in each sample; 'B', a matrix representing the mutation relationships between the clones in the tumor; 'U', a matrix that represents the frequencies of the clones in the tumor in the set of samples; and 'F_true', a matrix representing the true frequencies of each mutation in each sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create an instance of a tumor with 10 clones,
# 4 samples, k = 1, neutral evolution and
# added noise with depth = 500
I1 &lt;- create_instance(
  n = 10,
  m = 4,
  k = 1,
  selection = "neutral",
  depth = 500)
  

# Create an instance of a tumor with 50 clones,
# 10 samples, k = 5, positive selection and
# added noise with depth = 500
I2 &lt;- create_instance(
  n = 50,
  m = 10,
  k = 5,
  selection = "positive", 
  noisy = TRUE,
  depth = 500)
  
  
# Create an instance of a tumor with 100 clones,
# 25 samples, k = 0, positive selection without 
# noise
I3 &lt;- create_instance(
  n = 100,
  m = 25,
  k = 0,
  selection = "positive", 
  noisy = FALSE)
</code></pre>

<hr>
<h2 id='create_phylotree'>Create a <code>Phylotree</code> object</h2><span id='topic+create_phylotree'></span>

<h3>Description</h3>

<p>This is the general constructor of the <code>Phylotree</code> S4 class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_phylotree(B, clones, genes, parents, tree, labels = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_phylotree_+3A_b">B</code></td>
<td>
<p>A square matrix that represents the phylogenetic tree.</p>
</td></tr>
<tr><td><code id="create_phylotree_+3A_clones">clones</code></td>
<td>
<p>A numeric vector representing the clones in the phylogenetic tree.</p>
</td></tr>
<tr><td><code id="create_phylotree_+3A_genes">genes</code></td>
<td>
<p>A numeric vector representing the genes in the phylogenetic tree.</p>
</td></tr>
<tr><td><code id="create_phylotree_+3A_parents">parents</code></td>
<td>
<p>A numeric vector representing the parents of the clones in the phylogenetic tree.</p>
</td></tr>
<tr><td><code id="create_phylotree_+3A_tree">tree</code></td>
<td>
<p>A <code>data.tree</code> object containing the tree structure of the phylogenetic tree.</p>
</td></tr>
<tr><td><code id="create_phylotree_+3A_labels">labels</code></td>
<td>
<p>An optional vector containing the tags of the genes in the phylogenetic tree. <code>NA</code> by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>Phylotree</code> class object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a B matrix instance
# composed by 10 subpopulations of
# clones
B &lt;- create_instance(
       n = 10, 
       m = 4, 
       k = 1, 
       selection = "neutral")$B


# Create a new 'Phylotree' object
# on the basis of the B matrix
phylotree1 &lt;- B_to_phylotree(B = B)


# Create a new 'Phylotree' object
# with the general constructor of
# the class
phylotree2 &lt;- create_phylotree(
                B = B, 
                clones = phylotree1@clones, 
                genes = phylotree1@genes, 
                parents = phylotree1@parents, 
                tree = phylotree1@tree)


# Generate the tags for the genes of
# the phyogenetic tree
tags &lt;- LETTERS[1:nrow(B)]

 
# Create a new 'Phylotree' object
# with the general constructor of
# the class using tags
phylotree_tags &lt;- create_phylotree(
                    B = B, 
                    clones = phylotree1@clones, 
                    genes = phylotree1@genes, 
                    parents = phylotree1@parents, 
                    tree = phylotree1@tree, 
                    labels = tags)
</code></pre>

<hr>
<h2 id='create_U'>Calculate tumor clone frequencies in samples</h2><span id='topic+create_U'></span>

<h3>Description</h3>

<p>This function calculates the frequencies of each clone in a set of samples, given the global clone proportions in the tumor and their spatial distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_U(B, clone_proportions, density_coords, m, x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_U_+3A_b">B</code></td>
<td>
<p>A matrix representing the mutation relationships between the nodes in the phylogenetic tree (B matrix).</p>
</td></tr>
<tr><td><code id="create_U_+3A_clone_proportions">clone_proportions</code></td>
<td>
<p>A data frame with two columns: 'clone_idx', which contains the clone identifiers, and 'proportion', which contains the proportions of each clone in the tumor.</p>
</td></tr>
<tr><td><code id="create_U_+3A_density_coords">density_coords</code></td>
<td>
<p>A data frame where each column represents the density of a clone at different spatial coordinates.</p>
</td></tr>
<tr><td><code id="create_U_+3A_m">m</code></td>
<td>
<p>An integer representing the number of samples taken from the tumor.</p>
</td></tr>
<tr><td><code id="create_U_+3A_x">x</code></td>
<td>
<p>A numeric vector representing the spatial coordinates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix where each row corresponds to a sample, and each column corresponds to a clone. The value at the i-th row and j-th column is the frequency of the j-th clone in the i-th sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Calculate the frequencies of each clone in 10 samples taken from a tumor represented by the B 
# matrix B_mat, with global clone proportions clone_proportions, spatial distribution 
# density_coords, and spatial coordinates x

# Create random topology
B &lt;- create_B(20, 3)

# Assign proportions to each clone following a neutral evolution model
clone_proportions &lt;- calc_clone_proportions(B, "neutral")

# Place clones in 1D space
clones_space &lt;- place_clones_space(B)
density_coords &lt;- clones_space$spatial_coords
domain &lt;- clones_space$x

# Create U matrix with parameter m=4
U &lt;- create_U(B = B, clone_proportions = clone_proportions, 
              density_coords = density_coords, m = 4, x = domain)
</code></pre>

<hr>
<h2 id='equals'>Check if two phylogenetic trees are equal</h2><span id='topic+equals'></span>

<h3>Description</h3>

<p>Checks wether two phylogenetic trees are equivalent or not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>equals(phylotree_1, phylotree_2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="equals_+3A_phylotree_1">phylotree_1</code></td>
<td>
<p>A <code>Phylotree</code> class object.</p>
</td></tr>
<tr><td><code id="equals_+3A_phylotree_2">phylotree_2</code></td>
<td>
<p>A <code>Phylotree</code> class object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A boolean, <code>TRUE</code> if they are equal and <code>FALSE</code> if not.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load the predefined B matrices of the package
B_mats &lt;- GeRnika::B_mats


B_real &lt;- B_mats[[2]]$B_real
B_opt &lt;- B_mats[[2]]$B_opt


# Instantiate two \code{Phylotree} class objects on 
# the basis of the B matrices
phylotree_real &lt;- B_to_phylotree(
                    B = B_real)
                    
phylotree_opt &lt;- B_to_phylotree(
                    B = B_opt)


equals(phylotree_real, phylotree_opt)
</code></pre>

<hr>
<h2 id='find_common_subtrees'>Find the set of common subtrees between two phylogenetic trees</h2><span id='topic+find_common_subtrees'></span>

<h3>Description</h3>

<p>Plots the common subtrees between two phylogenetic trees and prints the information about their similarities and their differences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_common_subtrees(phylotree_1, phylotree_2, labels = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find_common_subtrees_+3A_phylotree_1">phylotree_1</code></td>
<td>
<p>A <code>Phylotree</code> class object.</p>
</td></tr>
<tr><td><code id="find_common_subtrees_+3A_phylotree_2">phylotree_2</code></td>
<td>
<p>A <code>Phylotree</code> class object.</p>
</td></tr>
<tr><td><code id="find_common_subtrees_+3A_labels">labels</code></td>
<td>
<p>A boolean, if <code>TRUE</code> the rendered graph will be plotted with the tags of the genes in the phylogenetic trees instead of their gene index. <code>FALSE</code> by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot of the common subtrees between two phylogenetic trees and the information about the distance between them based on their independent and common edges.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load the predefined B matrices of the package
B_mats &lt;- GeRnika::B_mats


B_real &lt;- B_mats[[2]]$B_real
B_opt &lt;- B_mats[[2]]$B_opt


# Generate the tags for the genes of
# the phyogenetic tree
tags &lt;- LETTERS[1:nrow(B_real)]


# Instantiate two Phylotree class objects on 
# the basis of the B matrices using tags
phylotree_real &lt;- B_to_phylotree(
                    B = B_real, 
                    labels = tags)
                    
phylotree_opt &lt;- B_to_phylotree(
                    B = B_opt, 
                    labels = tags)


# find the set of common subtrees between both 
# phylogenetic trees
find_common_subtrees(
  phylotree_1 = phylotree_real, 
  phylotree_2 = phylotree_opt)


# find the set of common subtrees between both
# phylogenetic trees using tags
find_common_subtrees(
  phylotree_1 = phylotree_real, 
  phylotree_2 = phylotree_opt, 
  labels = TRUE)
</code></pre>

<hr>
<h2 id='get_clones'>Retrieve the clone indices for a set of gene indices</h2><span id='topic+get_clones'></span>

<h3>Description</h3>

<p>Retrieve the clone indices for a set of gene indices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clones(genes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_clones_+3A_genes">genes</code></td>
<td>
<p>A vector of gene indices.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of clone indices.
</p>

<hr>
<h2 id='get_genes'>Retrieve the gene indices for the clones in a phylogenetic tree</h2><span id='topic+get_genes'></span>

<h3>Description</h3>

<p>Retrieve the gene indices for the clones in a phylogenetic tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genes(B)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_genes_+3A_b">B</code></td>
<td>
<p>A square matrix that represents the phylogenetic tree.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of gene indices.
</p>

<hr>
<h2 id='get_parents'>Get parent nodes in a phylogenetic tree.</h2><span id='topic+get_parents'></span>

<h3>Description</h3>

<p>This function retrieves the parent nodes for each node in a given phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_parents(phylotree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_parents_+3A_phylotree">phylotree</code></td>
<td>
<p>An object of the <code>Phylotree</code> class representing the phylogenetic tree.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of parent nodes.
</p>

<hr>
<h2 id='hyperparameters'>Hyperparameters for the methods of <code>GeRnika</code></h2><span id='topic+hyperparameters'></span>

<h3>Description</h3>

<p>A data.frame containing the static values for the parameters used in the methods of <code>GeRnika</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hyperparameters
</code></pre>


<h3>Format</h3>

<p>A data.frame containing different static values.
</p>

<dl>
<dt>Overdispersion</dt><dd><p>value = 0.5</p>
</dd>
<dt>Depth_sequencing</dt><dd><p>value = 30</p>
</dd>
</dl>



<h3>Source</h3>

<p>local source; inspired on the optimal parameters for the methods of <code>GeRnika</code>.
</p>

<hr>
<h2 id='Node-class'>S4 class to represent a node in a phylogenetic tree.</h2><span id='topic+Node-class'></span>

<h3>Description</h3>

<p>S4 class to represent a node in a phylogenetic tree.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Node</code></dt><dd><p>A node object</p>
</dd>
</dl>

<hr>
<h2 id='palettes'>Palettes for the methods of <code>GeRnika</code></h2><span id='topic+palettes'></span>

<h3>Description</h3>

<p>A data.frame containing 3 default palettes for the parameters used in the methods of <code>GeRnika</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>palettes
</code></pre>


<h3>Format</h3>

<p>A data.frame containing 3 palettes.
</p>

<dl>
<dt>Lancet</dt><dd><p>#0099B444, #AD002A77, #42B540FF</p>
</dd>
<dt>NEJM</dt><dd><p>#FFDC9177, #7876B188, #EE4C97FF</p>
</dd>
<dt>Simpsons</dt><dd><p>#FED43966, #FD744688, #197EC0FF</p>
</dd>
</dl>



<h3>Source</h3>

<p>Lancet, NEJM and The Simpsons palettes; inspired by the plots in Lancet journals, the plots in the New England Journal of Medicine and the colors used in the TV show The Simpsons, respectively.
</p>

<hr>
<h2 id='phylotree_to_B'>Get B matrix from <code>Phylotree</code> object.</h2><span id='topic+phylotree_to_B'></span>

<h3>Description</h3>

<p>This function retrieves the B matrix of a <code>Phylotree</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylotree_to_B(phylotree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phylotree_to_B_+3A_phylotree">phylotree</code></td>
<td>
<p>A <code>Phylotree</code> class object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> representing the B matrix of the phylogenetic tree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the B matrix of a tumor instance
# composed by 10 subpopulations of
# clones
B &lt;- create_instance(
       n = 10, 
       m = 4, 
       k = 1, 
       selection = "neutral")$B

# Create a new 'Phylotree' object
# on the basis of the B matrix
phylotree &lt;- B_to_phylotree(B)

# Get the B matrix of the phyotree
b1 &lt;- phylotree_to_B(phylotree)
</code></pre>

<hr>
<h2 id='phylotree_to_tree'>Extract tree from <code>Phylotree</code> object.</h2><span id='topic+phylotree_to_tree'></span>

<h3>Description</h3>

<p>This function extracts the tree structure from a given Phylotree object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylotree_to_tree(phylotree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phylotree_to_tree_+3A_phylotree">phylotree</code></td>
<td>
<p>An object of the <code>Phylotree</code> class.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The tree structure of the input <code>Phylotree</code> object.
</p>

<hr>
<h2 id='Phylotree-class'>S4 class to represent phylogenetic trees.</h2><span id='topic+Phylotree-class'></span>

<h3>Description</h3>

<p>S4 class to represent phylogenetic trees.
</p>


<h3>Slots</h3>


<dl>
<dt><code>B</code></dt><dd><p>A data.frame containing the square matrix that represents the ancestral relations among the clones of the phylogenetic tree.</p>
</dd>
<dt><code>clones</code></dt><dd><p>A vector representing the equivalence table of the clones in the phylogenetic tree.</p>
</dd>
<dt><code>genes</code></dt><dd><p>A vector representing the equivalence table of the genes in the phylogenetic tree.</p>
</dd>
<dt><code>parents</code></dt><dd><p>A vector representing the parents of the clones in the phylogenetic tree.</p>
</dd>
<dt><code>tree</code></dt><dd><p>A <code>Node</code> class object representing the phylogenetic tree.</p>
</dd>
<dt><code>labels</code></dt><dd><p>A vector representing the tags of the genes in the phylogenetic tree.</p>
</dd>
</dl>

<hr>
<h2 id='place_clones_space'>Create a model for the spatial distribution of the clones in a tumor</h2><span id='topic+place_clones_space'></span>

<h3>Description</h3>

<p>This function creates a Gaussian mixture model for the spatial distribution of the clones in a tumor in a 1D space. In the model, each component represents a clone, and the mean of the component represents the position of the clone in the space. The standard deviation of the components is fixed to 1, while the mean values are random variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>place_clones_space(B)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="place_clones_space_+3A_b">B</code></td>
<td>
<p>A matrix representing the mutation relationships between the nodes in the phylogenetic tree (B matrix).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two elements: 'spatial_coords', a data frame where each column represents the density of a clone at different spatial coordinates, and 'x', a numeric vector representing the spatial coordinates.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a model for the spatial distribution of the clones in a tumor represented by the 
# B matrix B_mat

# Create a mutation matrix for a phylogenetic tree with 10 nodes and k = 2
B_mat &lt;- create_B(10, 2)

clone_placement &lt;- place_clones_space(B_mat)

</code></pre>

<hr>
<h2 id='plot'>Plot a Phylotree object.</h2><span id='topic+plot'></span><span id='topic+plot+2CPhylotree-method'></span>

<h3>Description</h3>

<p>Plot a Phylotree object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot(object, labels = FALSE)

## S4 method for signature 'Phylotree'
plot(object, labels = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_+3A_object">object</code></td>
<td>
<p>A <code>Phylotree</code> object.</p>
</td></tr>
<tr><td><code id="plot_+3A_labels">labels</code></td>
<td>
<p>A label vector.</p>
</td></tr>
</table>

<hr>
<h2 id='plot_p'>Plot a phylogenetic tree with proportional node sizes and colors.</h2><span id='topic+plot_p'></span>

<h3>Description</h3>

<p>This function plots a phylogenetic tree with nodes sized and colored according to the proportions of each clone.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_p(phylotree, proportions)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_p_+3A_phylotree">phylotree</code></td>
<td>
<p>An object of the <code>Phylotree</code> class representing the phylogenetic tree to be plotted.</p>
</td></tr>
<tr><td><code id="plot_p_+3A_proportions">proportions</code></td>
<td>
<p>A numeric vector representing the proportions of each clone in the phylogenetic tree. The length of this vector should be equal to the number of clones in the tree.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A graph representing the phylogenetic tree, with node sizes and colors reflecting clone proportions.
</p>

<hr>
<h2 id='plot_phylotree'>Plot a phylogenetic tree</h2><span id='topic+plot_phylotree'></span>

<h3>Description</h3>

<p>This function generates a plot of a phylogenetic tree. If the 'labels' parameter is set to TRUE, the nodes of the tree will be labeled with the labels stored in the <code>Phylotree</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_phylotree(phylotree, labels = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_phylotree_+3A_phylotree">phylotree</code></td>
<td>
<p>An object of the <code>Phylotree</code> class representing the phylogenetic tree to be plotted.</p>
</td></tr>
<tr><td><code id="plot_phylotree_+3A_labels">labels</code></td>
<td>
<p>A logical value indicating whether to label the nodes with the labels stored in the <code>Phylotree</code> object. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot of the phylogenetic tree.
</p>

<hr>
<h2 id='plot_proportions'>Plot a phylogenetic tree with proportional node sizes and colors</h2><span id='topic+plot_proportions'></span>

<h3>Description</h3>

<p>This function plots a phylogenetic tree with nodes sized and colored according to the proportions of each clone. If a matrix of proportions is provided, multiple phylogenetic trees will be plotted, each corresponding to a row of proportions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_proportions(phylotree, proportions, labels = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_proportions_+3A_phylotree">phylotree</code></td>
<td>
<p>A <code>Phylotree</code> class object representing the phylogenetic tree to be plotted.</p>
</td></tr>
<tr><td><code id="plot_proportions_+3A_proportions">proportions</code></td>
<td>
<p>A numeric vector or matrix representing the proportions of each clone in the phylogenetic tree. If a matrix is provided, each row should represent the proportions for a separate tree.</p>
</td></tr>
<tr><td><code id="plot_proportions_+3A_labels">labels</code></td>
<td>
<p>A logical value indicating whether to label the nodes with gene tags (if <code>TRUE</code>) or gene indices (if <code>FALSE</code>). Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A graph representing the phylogenetic tree, with node sizes and colors reflecting clone proportions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create an instance
# composed by 5 subpopulations of clones
# and 4 samples
instance &lt;- create_instance(
       n = 5, 
       m = 4, 
       k = 1, 
       selection = "neutral")
       
# Extract its associated B matrix
B &lt;- instance$B

# Create a new 'Phylotree' object
# on the basis of the B matrix
phylotree &lt;- B_to_phylotree(B = B)

# Generate the tags for the genes of
# the phyogenetic tree
tags &lt;- LETTERS[1:nrow(B)]

# Plot the phylogenetic tree taking
# into account the proportions of the
# previously generated instance
plot_proportions(phylotree, instance$U, labels=TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
