<!DOCTYPE html><html lang="en"><head><title>Help for package GESE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GESE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GESE-package'>
<p>Gene-Based Segregation Test</p></a></li>
<li><a href='#condSegProbF'>
<p>Computes conditional segregation probability for the family</p></a></li>
<li><a href='#database'>
<p>database file in example</p></a></li>
<li><a href='#dataRaw'>
<p>dataRaw - a data frame containing the pedigree, phenotype and genotype information</p></a></li>
<li><a href='#GESE'>
<p>Gene-Based Segregation Test</p></a></li>
<li><a href='#GESE-internal'>
<p>GESE package internal functions</p></a></li>
<li><a href='#getSegInfo'>
<p>Computes segregation information for different mode of inheritance.</p></a></li>
<li><a href='#mapInfo'>
<p>mafInfo - example data</p></a></li>
<li><a href='#pednew'>
<p>pednew - an example pedigree structure</p></a></li>
<li><a href='#trim_oneLineage'>
<p>Trims the pedigree structure to include one lineage only.</p></a></li>
<li><a href='#trim_unrelated'>
<p>Trims the pedigree structure to exclude multiple founder cases</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene-Based Segregation Test</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-05-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Dandi Qiao, Michael H. Cho</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dandi Qiao &lt;redaq@channing.harvard.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the gene-based segregation test(GESE) and the weighted GESE test for identifying genes with causal variants of large effects for family-based sequencing data. The methods are described in Qiao, D. Lange, C., Laird, N.M., Won, S., Hersh, C.P., et al. (2017). &lt;<a href="https://doi.org/10.1002%2Fgepi.22037">doi:10.1002/gepi.22037</a>&gt;. Gene-based segregation method for identifying rare variants for family-based sequencing studies. Genet Epidemiol 41(4):309-319. More details can be found at <a href="http://scholar.harvard.edu/dqiao/gese">http://scholar.harvard.edu/dqiao/gese</a>.</td>
</tr>
<tr>
<td>Depends:</td>
<td>kinship2</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-05-18 16:31:07 UTC; redaq</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-05-18 18:16:22 UTC</td>
</tr>
</table>
<hr>
<h2 id='GESE-package'>
Gene-Based Segregation Test
</h2><span id='topic+GESE-package'></span>

<h3>Description</h3>

<p>Implements the gene-based segregation test(GESE) and the weighted GESE test for identifying genes with causal variants of large effects for family-based sequencing data. The methods are described in Qiao, D. Lange, C., Laird, N.M., Won, S., Hersh, C.P., et al. (2017). &lt;DOI:10.1002/gepi.22037&gt;. Gene-based segregation method for identifying rare variants for family-based sequencing studies. Genet Epidemiol 41(4):309-319. More details can be found at &lt;http://scholar.harvard.edu/dqiao/gese&gt;.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> GESE</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Gene-Based Segregation Test</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2017-05-17</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Dandi Qiao, Michael H. Cho</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Dandi Qiao &lt;redaq@channing.harvard.edu&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Implements the gene-based segregation test(GESE) and the weighted GESE test for identifying genes with causal variants of large effects for family-based sequencing data. The methods are described in Qiao, D. Lange, C., Laird, N.M., Won, S., Hersh, C.P., et al. (2017). &lt;DOI:10.1002/gepi.22037&gt;. Gene-based segregation method for identifying rare variants for family-based sequencing studies. Genet Epidemiol 41(4):309-319. More details can be found at &lt;http://scholar.harvard.edu/dqiao/gese&gt;.</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> kinship2</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
NeedsCompilation: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
Suggests: </td><td style="text-align: left;"> knitr,
rmarkdown</td>
</tr>
<tr>
 <td style="text-align: left;">
VignetteBuilder: </td><td style="text-align: left;"> knitr</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Index of help topics:
</p>
<pre>
GESE                    Gene-Based Segregation Test
GESE-internal functions
                        GESE package internal functions
GESE-package            Gene-Based Segregation Test
condSegProbF            Computes conditional segregation probability
                        for the family
dataRaw                 dataRaw - a data frame containing the pedigree,
                        phenotype and genotype information
database                database file in example
getSegInfo              Computes segregation information for different
                        mode of inheritance.
mapInfo                 mafInfo - example data
pednew                  pednew - an example pedigree structure
trim_oneLineage         Trims the pedigree structure to include one
                        lineage only.
trim_unrelated          Trims the pedigree structure to exclude
                        multiple founder cases
</pre>
<p>computes gene-based segregation tests(GESE and weighted GESE) for family-based sequencing data. The main functions are:
<code>GESE</code>: computes gene-based segregation information and GESE test p-values (unweighted and weighted version).
<code>trim_oneLineage</code>: trims the pedigree so that for any subject, either the paternal family or the maternal family is included. Minimal set of sequenced subjects may be removed to ensure one lineage per pedigree only.
<code>trim_unrelated</code>: trims the pedigree so that only one founder case is kept for each pedigree, and pedigrees with no cases are removed.
<code>condSegProbF</code>: computes the conditional probability that a variant in the gene is segregating in the family specified, conditional on that the variant is present in the family.
</p>


<h3>Author(s)</h3>

<p>Dandi Qiao, Michael H. Cho
</p>
<p>Maintainer: Dandi Qiao &lt;redaq@channing.harvard.edu&gt;
</p>


<h3>References</h3>

<p>Qiao, D. Lange, C., Laird, N.M., Won, S.,  Hersh, C.P., et al. (2017). <em>Gene-based segregation method for identifying rare variants for family-based sequencing studies.</em> Genet Epidemiol 41(4):309-319. DOI:10.1002/gepi.22037.
</p>
<p><a href="http://scholar.harvard.edu/dqiao/gese">http://scholar.harvard.edu/dqiao/gese</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GESE-package">GESE</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pednew)
data(mapInfo)
data(dataRaw)
data(database)
results &lt;- GESE(pednew, database, 1000000, dataRaw, mapInfo, threshold=1e-2)
results

</code></pre>

<hr>
<h2 id='condSegProbF'>
Computes conditional segregation probability for the family
</h2><span id='topic+condSegProbF'></span>

<h3>Description</h3>

<p>Computes the conditional probability that a variant is segregating in the family conditional on that the variant is present in one of the founders in the family.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>condSegProbF(pedTemp, subjInfo)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="condSegProbF_+3A_pedtemp">pedTemp</code></td>
<td>

<p>The data frame that includes the complete pedigree structure for the family</p>
</td></tr>
<tr><td><code id="condSegProbF_+3A_subjinfo">subjInfo</code></td>
<td>

<p>A data frame that contains the subject phenotype information for the sequenced subjects. it should include the columns FID, IID, and PHENOTYPE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns the conditional segregating probability of a variant in the family
</p>


<h3>Author(s)</h3>

<p>Dandi Qiao
</p>


<h3>References</h3>

<p>Qiao, D. Lange, C., Laird, N.M., Won, S., Hersh, C.P., et al. (2017). <em>Gene-based segregation method for identifying rare variants for family-based sequencing studies.</em> Genet Epidemiol 41(4):309-319. DOI:10.1002/gepi.22037.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GESE">GESE</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pednew)
data(mapInfo)
data(dataRaw)
data(database)
library(kinship2)
pedigrees = kinship2::pedigree(pednew$IID, pednew$faID, pednew$moID,pednew$sex,famid=pednew$FID)
subjects= dataRaw[,c(1,2,6)]
condSegProbF(pedigrees['93'], subjects)
condSegProbF(pedigrees['412'], subjects)
results2 &lt;- GESE(pednew, database, 1000000, dataRaw, mapInfo, threshold=1e-2)
results2$condSegProb
</code></pre>

<hr>
<h2 id='database'>
database file in example
</h2><span id='topic+database'></span>

<h3>Description</h3>

<p>a data frame containing the GENE and MAF information for the variants under consideration in the public reference database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("database")</code></pre>


<h3>Format</h3>

<p>A data frame of 20 observations on the following 3 variables.
</p>

<dl>
<dt><code>SNP</code></dt><dd><p>an unique identifier for variant</p>
</dd>
<dt><code>GENE</code></dt><dd><p>a character vector: Gene name</p>
</dd>
<dt><code>MAF</code></dt><dd><p>a numeric vector: minor allele frequency of the variants in the referecne database</p>
</dd>
</dl>



<h3>Details</h3>

<p>A data frame containing the information for all the variants satisfying the same filtering criteria in the chosen reference genome. It should include at least three columns with these names: SNP (unique SNP ID), GENE (gene name), MAF (minor allele frequency for the variant in reference database for the corresponding population).
</p>


<h3>Source</h3>

<p>Randomly simulated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(database)
</code></pre>

<hr>
<h2 id='dataRaw'>
dataRaw - a data frame containing the pedigree, phenotype and genotype information
</h2><span id='topic+dataRaw'></span>

<h3>Description</h3>

<p>A data frame that can be created from the <code>.raw</code> formatted filed generated by PLINK. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("dataRaw")</code></pre>


<h3>Format</h3>

<p>A data frame with 198 observations on the following 26 variables.
</p>

<dl>
<dt><code>FID</code></dt><dd><p>Family iD</p>
</dd>
<dt><code>IID</code></dt><dd><p>Individual ID</p>
</dd>
<dt><code>PAT</code></dt><dd><p>Father ID</p>
</dd>
<dt><code>MAT</code></dt><dd><p>Mother ID</p>
</dd>
<dt><code>SEX</code></dt><dd><p>sex</p>
</dd>
<dt><code>PHENOTYPE</code></dt><dd><p>Affection status</p>
</dd>
<dt><code>X1</code></dt><dd><p>Genotype for variant 1</p>
</dd>
<dt><code>X2</code></dt><dd><p>Genotype for variant 2</p>
</dd>
<dt><code>X3</code></dt><dd><p>Genotype for variant 3</p>
</dd>
<dt><code>X4</code></dt><dd><p>Genotype for variant 4</p>
</dd>
<dt><code>X5</code></dt><dd><p>Genotype for variant 5</p>
</dd>
<dt><code>X6</code></dt><dd><p>Genotype for variant 6</p>
</dd>
<dt><code>X7</code></dt><dd><p>Genotype for variant 7</p>
</dd>
<dt><code>X8</code></dt><dd><p>Genotype for variant 8</p>
</dd>
<dt><code>X9</code></dt><dd><p>Genotype for variant 9</p>
</dd>
<dt><code>X10</code></dt><dd><p>Genotype for variant 10</p>
</dd>
</dl>



<h3>Details</h3>

<p>The number of rows equal the number of subjects in the data and the number of columns equas the number of markers M + 6. The first six columns with specific column names include: the Family ID (FID), Individual ID (IID), father ID(PAT), mother ID (MAT), sex (SEX) and affection status (PHENOTYPE). The rest of the columns containing the genotypes for the variants listed in the coreesponding <code>mapInfo</code> file. It is also important to make sure that the recoding is with respect to the minor allele in the population. The affection status of this file will be used as the phenotype.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataRaw)
</code></pre>

<hr>
<h2 id='GESE'>
Gene-Based Segregation Test
</h2><span id='topic+GESE'></span>

<h3>Description</h3>

<p>Computes the gene-based segregation information and tests for family-based sequencing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GESE(pednew, variantInformation, dbSize, dataPed, mapInfo, 
threshold = 1e-7, onlySeg = FALSE, familyWeight = NA )

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GESE_+3A_pednew">pednew</code></td>
<td>

<p>A data frame of the complete pedigree information for all families in the dataset. The required column names of this data frame include: FID (family ID), IID (individual ID, must be of class character), faID (father ID, NA if unavailable), moID (mother ID, NA if unavailable), and sex.
</p>
</td></tr>
<tr><td><code id="GESE_+3A_variantinformation">variantInformation</code></td>
<td>

<p>A data frame containing the information for all the variants satisfying the same filtering criteria in the chosen reference genome. It should include at least three columns with these names: SNP (unique SNP ID), GENE (gene name), MAF (minor allele frequency for the variant in reference database for the corresponding population).
</p>
</td></tr>
<tr><td><code id="GESE_+3A_dbsize">dbSize</code></td>
<td>

<p>An integer indicating the sample size of the reference database used.
</p>
</td></tr>
<tr><td><code id="GESE_+3A_dataped">dataPed</code></td>
<td>

<p>A data frame in the <code>raw</code> file format generated by PLINK. The number of rows equal the number of subjects in the data and the number of columns equas the number of markers M + 6. The first six columns with specific column names include: the Family ID (FID), Individual ID (IID), father ID(PAT), mother ID (MAT), sex (SEX) and affection status (PHENOTYPE). The rest of the columns containing the genotypes for the variants listed in the coreesponding <code>mapInfo</code> file. It is also important to make sure that the recoding is with respect to the minor allele in the population. The affection status of this file will be used as the phenotype.
</p>
</td></tr>
<tr><td><code id="GESE_+3A_mapinfo">mapInfo</code></td>
<td>

<p>A data frame that contains at least two columns (required column names): variant ID (SNP) and Gene name (GENE). The number of rows equal to the number of SNPs/markers to be considered (M).
</p>
</td></tr>
<tr><td><code id="GESE_+3A_threshold">threshold</code></td>
<td>

<p>Specifies the precision needed to be reached for significant p-values. Default value is 1e-7.
</p>
</td></tr>
<tr><td><code id="GESE_+3A_onlyseg">onlySeg</code></td>
<td>

<p>True if only the segregation information (number of pedigrees segregating in each gene) is needed, else FALSE (DEFUALT), which computes the GESE p-values too.
</p>
</td></tr>
<tr><td><code id="GESE_+3A_familyweight">familyWeight</code></td>
<td>
<p>An optional data frame. It gives the weight for the families. If it is NA, no weighting scheme is used. Otherwise, its dimenstion could be (number of families)x(number of genes+1) or (number of families)x2. The first column should be family name (column name FID). If the weights for the families are the same for all the genes, the second column should just be weight (columns name &quot;weight&quot;), otherwise the second column and above should be the gene names (columns names are corresponding GENE names). 	
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the main function in the GESE package. The gene-based segregation tests (GESE) described in Qiao et al (2016) is a segregation-based test extending the work of Bureau et al (2014) by computing the marginal probability of segregation events within a gene. 
The first step in this function is to trim the families such that only one lineage (with the most possible number of cases) is included (i.e. for any subject, only the information of either the parental pedigree or the maternal pedigree would be included). In addition, if multiple founder cases are present, remove the (smallest set of) founder(s) that are unrelated most other sequenced subjects. 
Then this function computes the gene-based segregating information and p-values for multiple families. If only the segregation information (number of families segregating in each gene) is needed, set onlySeg = TRUE. If different family weights will be used to boost the power, assign the weights to familyWeight parameter.</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>segregation</code></td>
<td>
<p> a data frame containing the information about whether each gene is segregating in each family. The number of columns equals the number of families +3. The last column is the number of families the gene is segregating in. The number of rows equals the number of genes. Only this data frame and <code>varSeg</code> will be returned if onlySeg is set to TRUE.</p>
</td></tr>
<tr><td><code>varSeg</code></td>
<td>
<p>a data frame containing the information about whether each variant is segregating in each family. The number of columns equals the number of families +3. The last column is the number of families the variant is segregating in. The number of rows equals the number of variants. Only this data frame and <code>segregation</code> will be returned if onlySeg is set to TRUE.</p>
</td></tr>
<tr><td><code>results</code></td>
<td>
<p>This is available when onlySeg = FALSE. The datat frame contains the columns: GENE (gene name), obs_prob (the observed segregating probability for the gene), pvalue (gene-based p-value for GESE), numSim (The number of simulations used to compute the p-value if resampling-based method is used), N_seg (the number of families that are segregating in the gene). If familyWeight is not NA, obs_weight_stat (the observed weighted test statistic) and pvalue_weighted (the p-value for the weighted test statistic) will also be returned.</p>
</td></tr>
<tr><td><code>condSegProb</code></td>
<td>
<p>A vector of length equals the number of families. The conditional probability of at least one variant in the gene is segregating in the family condition on at least one variant (among the set of variants to be considered) is present in the familiy.</p>
</td></tr>
<tr><td><code>segProbGene</code></td>
<td>
<p>A matrix of the segregating probability for the gene and for each family. This is a working matrix that could be used in other functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dandi Qiao
</p>


<h3>References</h3>

	
<p>Qiao, D. Lange, C., Laird, N.M., Won, S.,  Hersh, C.P., et al. (2017). <em>Gene-based segregation method for identifying rare variants for family-based sequencing studies.</em> Genet Epidemiol 41(4):309-319. DOI:10.1002/gepi.22037.
</p>
<p><a href="http://scholar.harvard.edu/dqiao/gese">http://scholar.harvard.edu/dqiao/gese</a>
</p>
<p>Bureau, A., Younkin, S.G., Parker, M.M., Bailey-Wilson, J.E., Marazita, M.L., et al. (2014). <em>Inferring rare disease risk variants based on exact probabilities of sharing by multiple affected relatives</em>. Bioinformatics 30, 2189-2196. DOI:10.1093/bioinformatics/btu198.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pednew)
data(mapInfo)
data(dataRaw)
data(database)
results &lt;- GESE(pednew, database, 1000000, dataRaw, mapInfo, threshold=1e-3)
results

</code></pre>

<hr>
<h2 id='GESE-internal'>
GESE package internal functions
</h2><span id='topic+GESE-internal+20functions'></span>

<h3>Description</h3>

<p>GESE package internal functions.
</p>


<h3>Details</h3>

<p><code>computeP_resampling</code>
<code>findIntermediateFounder</code>
<code>findMostRecentCommonFounder</code>
<code>findMostRecentCommonFounderControl</code>
<code>getFounder</code>
<code>getProb</code>
<code>getPvalue_resampling</code>
<code>getTranProb_dv</code>
<code>isRelated</code>
<code>oneSetSim</code>
<code>segProb</code>
<code>getProb_weight</code>
</p>


<h3>Author(s)</h3>

<p>Dandi Qiao
</p>
<p>Maintainer: Dandi Qiao &lt;redaq@channing.harvard.edu&gt;
</p>


<h3>References</h3>

<p>Qiao, D. Lange, C., Laird, N.M., Won, S.,  Hersh, C.P., et al. (2017). <em>Gene-based segregation method for identifying rare variants for family-based sequencing studies.</em> Genet Epidemiol 41(4):309-319. DOI:10.1002/gepi.22037.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GESE-package">GESE</a></code>
</p>

<hr>
<h2 id='getSegInfo'>
Computes segregation information for different mode of inheritance.
</h2><span id='topic+getSegInfo'></span>

<h3>Description</h3>

<p>Computes variant-based and gene-based segregation information for different mode of inheritance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSegInfo(pednew, dataPed, mapInfo, mode="recessive")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getSegInfo_+3A_pednew">pednew</code></td>
<td>

<p>A data frame of the complete pedigree information for all families in the dataset. The required column names of this data frame include: FID (family ID), IID (individual ID, must be of class character), faID (father ID, NA if unavailable), moID (mother ID, NA if unavailable), and sex.
</p>
</td></tr>
<tr><td><code id="getSegInfo_+3A_dataped">dataPed</code></td>
<td>

<p>A data frame in the <code>raw</code> file format generated by PLINK. The number of rows equal the number of subjects in the data and the number of columns equas the number of markers M + 6. The first six columns with specific column names include: the Family ID (FID), Individual ID (IID), father ID(PAT), mother ID (MAT), sex (SEX) and affection status (PHENOTYPE). The rest of the columns containing the genotypes for the variants listed in the coreesponding <code>mapInfo</code> file. It is also important to make sure that the recoding is with respect to the minor allele in the population. The affection status of this file will be used as the phenotype.
</p>
</td></tr>
<tr><td><code id="getSegInfo_+3A_mapinfo">mapInfo</code></td>
<td>

<p>A data frame of at least two columns (required column names): variant ID (SNP) and Gene name (GENE). The number of rows equal to the number of SNPs/markers to be considered (M).
</p>
</td></tr>
<tr><td><code id="getSegInfo_+3A_mode">mode</code></td>
<td>

<p>The mode of inheriance assumed to compute the segregation information. The options are &quot;dominant&quot;, &quot;recessive&quot;, and &quot;CH&quot; (compound heterozygous). The default value is &quot;recessive&quot;.  	
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to compute the segregation information for different mode of inheritance without computing the GESE test. The mode of inheritance supported here are: dominant, recessive and compound heterozygous (CH). For dominant mode of inheritance, a variant is segregating if all the cases in the family carry at least one alternative allele (genotype X&gt;0), and all the controls in the family do not carry any alternative allele (X=0). For recessive mode of inheritance, a variant is segregating if all the cases in the family carry two alternative alleles (X=2), and all the controls in the family carry less than 2 alternative alleles (X=0 or X=1). For compound heterozygous mode of inheritance, a variant is segregating at two variant position if all the cases in the family carry at least one alternative allele at the two positions (X1&gt;0 and X2&gt;0), and all the controls in the family do not carry any alternative allele at either of the two positions (X1 = 0 or X2 = 0).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>varSeg</code></td>
<td>
<p>For dominant and recessive mode of inheriancce, this is a data frame containing the information about whether each variant is segregating in each family. The number of columns equals the number of families +3. The last column is the number of families the variant is segregating in. The number of rows equals the number of variants. For compound heterozygous mode of inheritance, this is a data frame containing the information of whether each pair of variants is segregating in each of the families. We consider all pairs in the dataset, if the pair of variants are not included in this data frame, they are not segregating in any families.</p>
</td></tr>
<tr><td><code>geneSeg</code></td>
<td>
<p>For dominant and recessive mode of inheriancce, this is a data frame containing the information about whether each gene is segregating in each family. The number of columns equals the number of families +3. The last column is the number of families the gene is segregating in. The number of rows equals the number of genes. For compound heterozygous mode of inheritance, this is a data frame containing the information of whether any pair of variants in this gene are segregating in each of the families. The last columns is the number of families with the presence of any pair of variants segregating in the gene.</p>
</td></tr>
<tr><td><code>genePairSeg</code></td>
<td>
<p>This data frame is returned only for compound heterozygous mode of inheritance. This considers any pair of genes in the data. It returns a data frame containing the information of whether any pair of variants, each in a different gene, is segregating in each of the families considered. Each row represents the information for each gene pair, summed over all possible pairs of variants in the two genes, one in each gene.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dandi Qiao
</p>


<h3>References</h3>

<p>Qiao, D. Lange, C., Laird, N.M., Won, S.,  Hersh, C.P., et al. (2017). <em>Gene-based segregation method for identifying rare variants for family-based sequencing studies.</em> Genet Epidemiol 41(4):309-319. DOI:10.1002/gepi.22037.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GESE">GESE</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pednew)
data(mapInfo)
data(dataRaw)
data(database)
result &lt;- getSegInfo(pednew, dataRaw, mapInfo)
result$varSeg
result$geneSeg

result &lt;- getSegInfo(pednew, dataRaw, mapInfo, mode="recessive")
result$varSeg
result$geneSeg

result &lt;- getSegInfo(pednew, dataRaw, mapInfo, mode="CH")
result$varSeg
result$geneSeg
result$genePairSeg


</code></pre>

<hr>
<h2 id='mapInfo'>
mafInfo - example data</h2><span id='topic+mapInfo'></span>

<h3>Description</h3>

<p>a data frame containing the gene information for the variants in the study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("mapInfo")</code></pre>


<h3>Format</h3>

<p>A data frame of 20 observations on the following 2 variables.
</p>

<dl>
<dt><code>GENE</code></dt><dd><p>The gene name</p>
</dd>
<dt><code>SNP</code></dt><dd><p>An unique SNP identifier</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(mapInfo)
</code></pre>

<hr>
<h2 id='pednew'>
pednew - an example pedigree structure
</h2><span id='topic+pednew'></span>

<h3>Description</h3>

<p>A data frame of the complete pedigree strucutre for the families included
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("pednew")</code></pre>


<h3>Format</h3>

<p>A data frame of 1700 observations on the following 26 variables.
</p>

<dl>
<dt><code>FID</code></dt><dd><p>Family ID of class character</p>
</dd>
<dt><code>IID</code></dt><dd><p>Individual ID of class character</p>
</dd>
<dt><code>faID</code></dt><dd><p>Father ID, NA if missing</p>
</dd>
<dt><code>moID</code></dt><dd><p>Mother ID, NA if missing</p>
</dd>
<dt><code>sex</code></dt><dd><p>Sex, 1 for male, 2 for female and NA if missing.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(pednew)
</code></pre>

<hr>
<h2 id='trim_oneLineage'>
Trims the pedigree structure to include one lineage only.
</h2><span id='topic+trim_oneLineage'></span>

<h3>Description</h3>

<p>Trims the families to include only one lineage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trim_oneLineage(seqSub, pednew)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trim_oneLineage_+3A_seqsub">seqSub</code></td>
<td>

<p>A data frame that should include three columns FID (family ID), IID (individual ID), and PHENOTYPE (affection status) for the sequenced subjects in the data. One example is the 1st, 2nd and 6th columns from the plink raw format.
</p>
</td></tr>
<tr><td><code id="trim_oneLineage_+3A_pednew">pednew</code></td>
<td>

<p>A data frame includes the complete pedigree structure information for all sequenced families in the dataset. The required column names of this data frame include: FID (family ID), IID (individual ID, must be of class character), faID (father ID, NA if unavailable), moID (mother ID, NA if unavailable), and sex.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each subject, only the maternal or the paternal family is included, since the rare variant should be present in only the related subjects. The lineage with the maximal set of sequenced cases will be used as the final pedigree.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pedInfoUpdate</code></td>
<td>
<p>the complete pedigrees with only the paternal or maternal lineage </p>
</td></tr>
<tr><td><code>seqSubjUpdate</code></td>
<td>
<p>The sequenced subjects that are in the selected lineage are returned for the rest of the analysis.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function can be used for other analysis of family-based data processing. For example, the pre-processing step for PVAAST analysis.
</p>


<h3>Author(s)</h3>

<p>Dandi Qiao
</p>


<h3>References</h3>

<p>Qiao, D. Lange, C., Laird, N.M., Won, S.,  Hersh, C.P., et al. (2017). <em>Gene-based segregation method for identifying rare variants for family-based sequencing studies.</em> Genet Epidemiol 41(4):309-319. DOI:10.1002/gepi.22037.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GESE">GESE</a></code>, <code><a href="#topic+trim_unrelated">trim_unrelated</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pednew)
data(mapInfo)
data(dataRaw)
data(database)
subjects &lt;- dataRaw[,c(1:2, 6)]
cat("Trimming the families...\n")
cat("Trimming step 1: keep only one lineage \n")
trim &lt;- trim_oneLineage(seqSub=subjects, pednew)
	
</code></pre>

<hr>
<h2 id='trim_unrelated'>
Trims the pedigree structure to exclude multiple founder cases
</h2><span id='topic+trim_unrelated'></span>

<h3>Description</h3>

<p>Trims the families to include only one founder case in each pedigree. It also exclude families with only one control subject.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trim_unrelated(seqSub, pednew2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trim_unrelated_+3A_seqsub">seqSub</code></td>
<td>
 
<p>A data frame that includes at least three columns: the Family ID (FID), Individual ID (IID), and affection status (PHENOTYPE). This input values should be output from the <code>trim_oneLineage</code> function.
</p>
</td></tr>
<tr><td><code id="trim_unrelated_+3A_pednew2">pednew2</code></td>
<td>

<p>A data frame includes the complete pedigree structure information for all sequenced families in the dataset. The required column names of this data frame include: FID (family ID), IID (individual ID, must be of class character), faID (father ID, NA if unavailable), moID (mother ID, NA if unavailable), and sex.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each pedigree,each there are multiple case founders in the pedigree, to satisfy the assumption that the causal rare variant must be introduced by one founder only, we need to keep only one case fonder that is related to most of the sequened subjects in the pedigree. We also want to remove families with no case.
</p>


<h3>Value</h3>

<p>The modified dataPed2 file after the trimming.
</p>


<h3>Author(s)</h3>

<p>Dandi Qiao
</p>


<h3>References</h3>

<p>Qiao, D. Lange, C., Laird, N.M., Won, S.,  Hersh, C.P., et al. (2017). <em>Gene-based segregation method for identifying rare variants for family-based sequencing studies.</em> Genet Epidemiol 41(4):309-319. DOI:10.1002/gepi.22037.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GESE">GESE</a></code>, <code><a href="#topic+trim_oneLineage">trim_oneLineage</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pednew)
data(mapInfo)
data(dataRaw)
data(database)
subjects &lt;- dataRaw[,c(1:2, 6)]
cat("Trimming the families...\n")
cat("Trimming step 1: keep only one lineage \n")
trim &lt;- trim_oneLineage(seqSub=subjects, pednew)
subjects2 &lt;- trim_unrelated(trim$seqSubjUpdate, trim$pedInfoUpdate)

	
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
