<!DOCTYPE html><html lang="en"><head><title>Help for package getmstatistic</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {getmstatistic}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#draw_table'><p>Helper function to draw table grobs.</p></a></li>
<li><a href='#getmstatistic'><p>Quantifying Systematic Heterogeneity in Meta-Analysis.</p></a></li>
<li><a href='#heartgenes214'><p>heartgenes214.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Quantifying Systematic Heterogeneity in Meta-Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Quantifying systematic heterogeneity in meta-analysis using R.
    The M statistic aggregates heterogeneity information across multiple
    variants to, identify systematic heterogeneity patterns and their direction
    of effect in meta-analysis. It's primary use is to identify outlier studies,
    which either show "null" effects or consistently show stronger or weaker
    genetic effects than average across, the panel of variants examined in a
    GWAS meta-analysis. In contrast to conventional heterogeneity metrics
    (Q-statistic, I-squared and tau-squared) which measure random heterogeneity
    at individual variants, M measures systematic (non-random)
    heterogeneity across multiple independently associated variants. Systematic
    heterogeneity can arise in a meta-analysis due to differences in the study
    characteristics of participating studies. Some of the differences may
    include: ancestry, allele frequencies, phenotype definition, age-of-disease
    onset, family-history, gender, linkage disequilibrium and quality control
    thresholds. See <a href="https://magosil86.github.io/getmstatistic/">https://magosil86.github.io/getmstatistic/</a> for statistical
    statistical theory, documentation and examples.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://magosil86.github.io/getmstatistic/">https://magosil86.github.io/getmstatistic/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/magosil86/getmstatistic/issues">https://github.com/magosil86/getmstatistic/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2 (&ge; 1.0.1), gridExtra (&ge; 0.9.1), gtable (&ge; 0.1.2),
metafor (&ge; 1.9-6), psych (&ge; 1.5.1), stargazer (&ge; 5.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>foreign (&ge; 0.8-62), knitr (&ge; 1.10.5), testthat, covr,
rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-05-09 03:56:42 UTC; lmagosi</td>
</tr>
<tr>
<td>Author:</td>
<td>Lerato E Magosi [aut],
  Jemma C Hopewell [aut],
  Martin Farrall [aut],
  Lerato E Magosi [cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lerato E Magosi &lt;magosil86@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-05-09 05:10:15 UTC</td>
</tr>
</table>
<hr>
<h2 id='draw_table'>Helper function to draw table grobs.</h2><span id='topic+draw_table'></span>

<h3>Description</h3>

<p><code>draw_table()</code> Pre and post version: 2.0.0 gridExtra packages
handle drawing tables differently. <code>draw_table()</code> determines
the installed version of gridExtra and applies the appropriate
syntax. If gridExtra version &lt; 2.0.0 then it uses old gridExtra 
syntax to build table Grob(graphical object) else uses new syntax. 
<code>draw_table()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_table(body, heading, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="draw_table_+3A_body">body</code></td>
<td>
<p>A dataframe. Table body.</p>
</td></tr>
<tr><td><code id="draw_table_+3A_heading">heading</code></td>
<td>
<p>A string. Table title.</p>
</td></tr>
<tr><td><code id="draw_table_+3A_...">...</code></td>
<td>
<p>Further arguments to control the gtable.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>prints tables without rownames.
</p>


<h3>Acknowledgements</h3>

<p>Thanks to Ryan Welch, 
https://github.com/welchr/LocusZoom/issues/16
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(gridExtra)

## Not run: 

# Table of iris values
iris_dframe &lt;- head(iris)
title_iris_dframe &lt;- paste("Table: Length and width measurements (cm) of sepals and petals,",
                            "for 50 flowers from 3 species of iris (setosa, versicolor,", 
                            "and virginica).\n", sep = " ")
# Wrap title text at column 60
title_iris_dframe &lt;- sapply(strwrap(title_iris_dframe, width = 60, simplify = FALSE), 
                            paste, collapse = "\n")
# Draw table
table_influential_studies &lt;- draw_table(body = iris_dframe, heading = title_iris_dframe)

# Table of mtcars values
mtcars_dframe &lt;- head(mtcars)
title_mtcars_dframe &lt;- paste("Table: Motor Trend US magazine (1974) automobile statistics", 
                             "for fuel consumption, \nautomobile design and performance.\n", 
                             sep = " ")
# Wrap title text at column 60
title_mtcars_dframe &lt;- sapply(strwrap(title_mtcars_dframe, width = 60, simplify = FALSE), 
                              paste, collapse = "\n")
# Draw table
table_influential_studies &lt;- draw_table(body = mtcars_dframe, heading = title_mtcars_dframe)

## End(Not run)
</code></pre>

<hr>
<h2 id='getmstatistic'>Quantifying Systematic Heterogeneity in Meta-Analysis.</h2><span id='topic+getmstatistic'></span><span id='topic+getm'></span><span id='topic+getmstat'></span><span id='topic+Rgetmstatistic'></span><span id='topic+getmstatistic.default'></span>

<h3>Description</h3>

<p><code>getmstatistic</code> computes <em>M</em> statistics to assess the contribution
of each participating study in a meta-analysis. The <em>M</em> statistic 
aggregates heterogeneity information across multiple variants to, identify 
systematic heterogeneity patterns and their direction of effect in 
meta-analysis. It's primary use is to identify outlier studies, which either 
show &quot;null&quot; effects or consistently show stronger or weaker genetic effects 
than average, across the panel of variants examined in a GWAS meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getmstatistic(beta_in, lambda_se_in, study_names_in, variant_names_in, ...)

## Default S3 method:
getmstatistic(
  beta_in,
  lambda_se_in,
  study_names_in,
  variant_names_in,
  save_dir = getwd(),
  tau2_method = "DL",
  x_axis_increment_in = 0.02,
  x_axis_round_in = 2,
  produce_plots = TRUE,
  verbose_output = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getmstatistic_+3A_beta_in">beta_in</code></td>
<td>
<p>A numeric vector of study effect-sizes e.g. log odds-ratios.</p>
</td></tr>
<tr><td><code id="getmstatistic_+3A_lambda_se_in">lambda_se_in</code></td>
<td>
<p>A numeric vector of standard errors, genomically 
corrected at study-level.</p>
</td></tr>
<tr><td><code id="getmstatistic_+3A_study_names_in">study_names_in</code></td>
<td>
<p>A character vector of study names.</p>
</td></tr>
<tr><td><code id="getmstatistic_+3A_variant_names_in">variant_names_in</code></td>
<td>
<p>A character vector of variant names e.g. rsIDs.</p>
</td></tr>
<tr><td><code id="getmstatistic_+3A_...">...</code></td>
<td>
<p>Further arguments.</p>
</td></tr>
<tr><td><code id="getmstatistic_+3A_save_dir">save_dir</code></td>
<td>
<p>A character scalar specifying a path to the directory where
plots should be stored (optional). Required if produce_plots = TRUE.</p>
</td></tr>
<tr><td><code id="getmstatistic_+3A_tau2_method">tau2_method</code></td>
<td>
<p>A character scalar, method to estimate heterogeneity: 
either &quot;DL&quot; or &quot;REML&quot; (Optional). Note: The REML method uses the 
iterative Fisher scoring algorithm (step length = 0.5, maximum 
iterations = 10000) to estimate tau2.</p>
</td></tr>
<tr><td><code id="getmstatistic_+3A_x_axis_increment_in">x_axis_increment_in</code></td>
<td>
<p>A numeric scalar, value by which x-axis 
of M scatterplot will be incremented (Optional).</p>
</td></tr>
<tr><td><code id="getmstatistic_+3A_x_axis_round_in">x_axis_round_in</code></td>
<td>
<p>A numeric scalar, value to which x-axis labels of 
M scatterplot will be rounded (Optional).</p>
</td></tr>
<tr><td><code id="getmstatistic_+3A_produce_plots">produce_plots</code></td>
<td>
<p>A boolean to generate plots (optional).</p>
</td></tr>
<tr><td><code id="getmstatistic_+3A_verbose_output">verbose_output</code></td>
<td>
<p>An optional boolean to display intermediate output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In contrast to conventional heterogeneity metrics (Q-statistic, I-squared and 
tau-squared) which measure random heterogeneity at individual variants, 
<em>M</em> measures systematic (non-random) heterogeneity across multiple 
independently associated variants. 
</p>
<p>Systematic heterogeneity can arise in a meta-analysis due to differences in 
the study characteristics of participating studies. Some of the differences 
may include: ancestry, allele frequencies, phenotype definition, 
age-of-disease onset, family-history, gender, linkage disequilibrium and 
quality control thresholds. See the getmstatistic website for statistical 
theory, documentation and examples.
</p>
<p><code>getmstatistic</code> uses summary data i.e. study effect-sizes and their 
corresponding standard errors to calculate <em>M</em> statistics (One <em>M</em> 
for each study in the meta-analysis).
</p>
<p>In particular, <code>getmstatistic</code> employs the inverse-variance weighted 
random effects regression model provided in the <code>metafor</code> R package 
to extract SPREs (standardized predicted random effects) which are then 
aggregated to formulate <em>M</em> statistics.
</p>


<h3>Value</h3>

<p>Returns a list containing:
</p>

<ul>
<li><p> Mstatistic_expected_mean   , A numeric scalar for the expected mean for M
</p>
</li>
<li><p> Mstatistic_expected_sd     , A numeric scalar for the expected standard deviation 
for M
</p>
</li>
<li><p> number_studies             , A numeric scalar for the number of studies
</p>
</li>
<li><p> number_variants            , A numeric scalar for the number of variants
</p>
</li>
<li><p> Mstatistic_crit_alpha_0_05 , A numeric scalar of the critical M value at the 5
percent significance level.
</p>
</li>
<li><p> M_dataset (dataframe) A dataset of the computed M statistics, which
includes the following fields:
</p>

<ul>
<li><p> M           , Mstatistic
</p>
</li>
<li><p> M_sd        , standard deviation of M
</p>
</li>
<li><p> M_se        , standard error of M
</p>
</li>
<li><p> lowerbound  , lowerbound of M 95
</p>
</li>
<li><p> upperbound  , upperbound of M 95
</p>
</li>
<li><p> bonfpvalue  , 2-sided bonferroni pvalues of M
</p>
</li>
<li><p> qvalue      , false discovery rate adjusted pvalues of M
</p>
</li>
<li><p> tau2        , tau_squared, DL estimates of between-study 
heterogeneity
</p>
</li>
<li><p> I2          , I_squared, proportion of total variation due to 
between study variance
</p>
</li>
<li><p> Q           , Cochran's Q
</p>
</li>
<li><p> xb          , fitted values excluding random effects
</p>
</li>
<li><p> usta        , standardized predicted random effect (SPRE)
</p>
</li>
<li><p> xbu         , fitted values including random effects
</p>
</li>
<li><p> stdxbu      , standard error of prediction (fitted values) 
including random effects
</p>
</li>
<li><p> hat         , diagonal elements of the projection hat matrix
</p>
</li>
<li><p> study       , study numbers    
</p>
</li>
<li><p> snp         , variant numbers
</p>
</li>
<li><p> beta_mean   , average variant effect size
</p>
</li>
<li><p> oddsratio   , average variant effect size as oddsratio
</p>
</li>
<li><p> beta_n   , number of variants in each study
</p>
</li></ul>

</li>
<li><p> influential_studies_0_05 (dataframe) A dataset of influential studies 
significant at the 5 percent level.
</p>
</li>
<li><p> weaker_studies_0_05 (dataframe) A dataset of under-performing studies 
significant at the 5 percent level.
</p>
</li></ul>



<h3>Methods (by class)</h3>


<ul>
<li> <p><code>default</code>: Computes M statistics
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="metafor.html#topic+rma.uni">rma.uni</a></code> function in <code>metafor</code> for random
effects model, and <a href="https://magosil86.github.io/getmstatistic/">https://magosil86.github.io/getmstatistic/</a> for 
getmstatistic website.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(getmstatistic)
library(gridExtra)


# Basic M analysis using the heartgenes214 dataset.
# heartgenes214 is a multi-ethnic GWAS meta-analysis dataset for coronary artery disease.
# To learn more about the heartgenes214 dataset ?heartgenes214

# Running an M analysis on 20 GWAS significant variants (p &lt; 5e-08) in the first 10 studies

heartgenes44_10studies &lt;- subset(heartgenes214, studies &lt;= 10 &amp; fdr214_gwas46 == 2) 
heartgenes20_10studies &lt;- subset(heartgenes44_10studies, 
    variants %in% unique(heartgenes44_10studies$variants)[1:20])

# Set directory to store plots, this can be a temporary directory
# or a path to a directory of choice e.g. plots_dir &lt;- "~/Downloads"
plots_dir &lt;- tempdir()

getmstatistic_results &lt;- getmstatistic(heartgenes20_10studies$beta_flipped, 
                                        heartgenes20_10studies$gcse, 
                                        heartgenes20_10studies$variants, 
                                        heartgenes20_10studies$studies,
                                        save_dir = plots_dir)
getmstatistic_results

# Explore results generated by getmstatistic function

# Retrieve dataset of M statistics
dframe &lt;- getmstatistic_results$M_dataset



str(dframe)


# Retrieve dataset of stronger than average studies (significant at 5% level)
getmstatistic_results$influential_studies_0_05

# Retrieve dataset of weaker than average studies (significant at 5% level)
getmstatistic_results$weaker_studies_0_05

# Retrieve number of studies and variants
getmstatistic_results$number_studies
getmstatistic_results$number_variants

# Retrieve expected mean, sd and critical M value at 5% significance level
getmstatistic_results$M_expected_mean
getmstatistic_results$M_expected_sd
getmstatistic_results$M_crit_alpha_0_05

# To view plots stored in a temporary directory, call `tempdir()` to view the directory path 
tempdir()


# Additional examples: These take a little bit longer to run

## Not run: 

# Set directory to store plots, this can be a temporary directory
# or a path to a directory of choice e.g. plots_dir &lt;- "~/Downloads"
plots_dir &lt;- tempdir()

# Run M analysis on all 214 lead variants
# heartgenes214 is a multi-ethnic GWAS meta-analysis dataset for coronary artery disease.
getmstatistic_results &lt;- getmstatistic(heartgenes214$beta_flipped, 
                                        heartgenes214$gcse, 
                                        heartgenes214$variants, 
                                        heartgenes214$studies,
                                        save_dir = plots_dir)
getmstatistic_results


# Subset the GWAS significant variants (p &lt; 5e-08) in heartgenes214
heartgenes44 &lt;- subset(heartgenes214, heartgenes214$fdr214_gwas46 == 2)

# Exploring getmstatistic options:
#     Estimate heterogeneity using "REML", default is "DL"
#     Modify x-axis of M scatterplot
#     Run M analysis verbosely
getmstatistic_results &lt;- getmstatistic(heartgenes44$beta_flipped, 
                                        heartgenes44$gcse, 
                                        heartgenes44$variants, 
                                        heartgenes44$studies,
                                        save_dir = plots_dir,
                                        tau2_method = "REML",
                                        x_axis_increment_in = 0.03, 
                                        x_axis_round_in = 3,
                                        produce_plots = TRUE,
                                        verbose_output = TRUE)
getmstatistic_results



## End(Not run)

</code></pre>

<hr>
<h2 id='heartgenes214'>heartgenes214.</h2><span id='topic+heartgenes214'></span>

<h3>Description</h3>

<p>heartgenes214 is a multi-ethnic GWAS meta-analysis dataset for coronary
artery disease.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heartgenes214
</code></pre>


<h3>Format</h3>

<p>A data frame with seven variables:
</p>

<dl>
<dt><code>beta_flipped</code></dt><dd><p>Effect-sizes expressed as log odds ratios. Numeric</p>
</dd>
<dt><code>gcse</code></dt><dd><p>Standard errors</p>
</dd>
<dt><code>studies</code></dt><dd><p>Names of participating studies</p>
</dd>
<dt><code>variants</code></dt><dd><p>Names of genetic variants/SNPs</p>
</dd>
<dt><code>cases</code></dt><dd><p>Number of cases in each participating study</p>
</dd>
<dt><code>controls</code></dt><dd><p>Number of controls in each participating study</p>
</dd>
<dt><code>fdr214_gwas46</code></dt><dd><p>Flag indicating GWAS significant variants, 
1: Not GWAS-significant, 2: GWAS-significant</p>
</dd>
</dl>



<h3>Details</h3>

<p>It comprises summary data (effect-sizes and their corresponding standard 
errors) for 48 studies (68,801 cases and 123,504 controls), at 214 lead 
variants independently associated with coronary artery disease 
(P &lt; 0.00005, FDR &lt; 5%). Of the 214 lead variants, 44 are genome-wide 
significant (p &lt; 5e-08). The meta-analysis dataset is based on 
individuals of: African American, Hispanic American, East Asian, 
South Asian, Middle Eastern and European ancestry. 
</p>
<p>The study effect-sizes have been flipped to ensure alignment of the 
effect alleles.
</p>
<p>Standard errors were genomically corrected at the study-level.
</p>


<h3>Source</h3>

<p>Magosi LE, Goel A, Hopewell JC, Farrall M, on behalf of the 
CARDIoGRAMplusC4D Consortium (2017) Identifying systematic 
heterogeneity patterns in genetic association meta-analysis studies. 
PLoS Genet 13(5): e1006755. https://doi.org/10.1371/journal.pgen.1006755.
</p>
<p><a href="https://magosil86.github.io/getmstatistic/">https://magosil86.github.io/getmstatistic/</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
