<!DOCTYPE html><html lang="en"><head><title>Help for package getspres</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {getspres}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#getspres'><p>Exploring Heterogeneity in Meta-Analysis with <em>SPRE</em> Statistics.</p></a></li>
<li><a href='#heartgenes214'><p>heartgenes214.</p></a></li>
<li><a href='#plotspres'><p><code>plotspres</code> generates forest plots showing <em>SPRE</em> statistics.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>SPRE Statistics for Exploring Heterogeneity in Meta-Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of SPRE (standardised predicted random-effects)
    statistics in R to explore heterogeneity in genetic association meta-
    analyses, as described by Magosi et al. (2019) 
    &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtz590">doi:10.1093/bioinformatics/btz590</a>&gt;. SPRE statistics are precision 
    weighted residuals that indicate the direction and extent with which 
    individual study-effects in a meta-analysis deviate from the average 
    genetic effect. Overly influential positive outliers have the potential 
    to inflate average genetic effects in a meta-analysis whilst negative 
    outliers might lower or change the direction of effect. See the 'getspres' 
    website for documentation and examples 
    <a href="https://magosil86.github.io/getspres/">https://magosil86.github.io/getspres/</a>.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://magosil86.github.io/getspres/">https://magosil86.github.io/getspres/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/magosil86/getspres/issues">https://github.com/magosil86/getspres/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr (&ge; 1.10.5), testthat, covr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>metafor (&ge; 1.9-6), dplyr (&ge; 0.4.1), plotrix (&ge; 3.5-12),
colorspace (&ge; 1.2-6), RColorBrewer (&ge; 1.1-2), colorRamps (&ge;
2.3)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-05-09 05:03:54 UTC; lmagosi</td>
</tr>
<tr>
<td>Author:</td>
<td>Lerato E Magosi [aut],
  Jemma C Hopewell [aut],
  Martin Farrall [aut],
  Lerato E Magosi [cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lerato E Magosi &lt;magosil86@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-05-09 05:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='getspres'>Exploring Heterogeneity in Meta-Analysis with <em>SPRE</em> Statistics.</h2><span id='topic+getspres'></span><span id='topic+getspre'></span><span id='topic+spre'></span><span id='topic+spres'></span><span id='topic+getspres.default'></span>

<h3>Description</h3>

<p><code>getspres</code> computes <em>SPRE</em> (standardised predicted random-effects)
statistics to identify outlier studies in genetic association meta-analyses
which might have undue influence on the average genetic effect leading to 
inflated genetic signals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getspres(beta_in, se_in, study_names_in, variant_names_in, ...)

## Default S3 method:
getspres(
  beta_in,
  se_in,
  study_names_in,
  variant_names_in,
  tau2_method = "DL",
  verbose_output = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getspres_+3A_beta_in">beta_in</code></td>
<td>
<p>A numeric vector of study effect-sizes e.g. log odds-ratios.</p>
</td></tr>
<tr><td><code id="getspres_+3A_se_in">se_in</code></td>
<td>
<p>A numeric vector of standard errors, genomically corrected at study-level.</p>
</td></tr>
<tr><td><code id="getspres_+3A_study_names_in">study_names_in</code></td>
<td>
<p>A character vector of study names.</p>
</td></tr>
<tr><td><code id="getspres_+3A_variant_names_in">variant_names_in</code></td>
<td>
<p>A character vector of variant names e.g. rsIDs.</p>
</td></tr>
<tr><td><code id="getspres_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
<tr><td><code id="getspres_+3A_tau2_method">tau2_method</code></td>
<td>
<p>A character scalar, specifying the method that should be used to estimate heterogeneity either through DerSimonian and Laird's moment-based estimate &quot;DL&quot; or restricted maximum likelihood &quot;REML&quot;. Note: The REML method uses the iterative Fisher scoring algorithm (step length = 0.5, maximum iterations = 10000) to estimate tau2. Default is &quot;DL&quot;.</p>
</td></tr>
<tr><td><code id="getspres_+3A_verbose_output">verbose_output</code></td>
<td>
<p>An optional boolean to display intermediate output. (Default is <code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>SPRE</em> statistics are precision-weighted residuals that summarise the 
direction and extent with which observed study effects in a meta-analysis
differ from the summary (or average genetic) effect. See the getspres 
website for more information, documentation and examples.
</p>
<p><code>getspres</code> takes as input study effect-size estimates and their 
corresponding standard errors (i.e. summary data). Study effect estimates 
could be in the form of linear regression coefficients or log-transformed
regression coefficients (per-allele log odds ratios) from logistic 
regression. 
</p>
<p><code>getspres</code> uses inverse-variance weighted meta-analysis models in the
<code>metafor</code> R package to calculate <em>SPRE</em> statistics.
</p>


<h3>Value</h3>

<p>Returns a list containing:
</p>

<ul>
<li><p> number_variants A numeric scalar for the number of variants 
</p>
</li>
<li><p> number_studies  A numeric scalar for the number of studies
</p>
</li>
<li><p> spre_dataset    A dataframe that is a dataset of computed SPRE statistics and contains the following fields:
</p>

<ul>
<li><p> beta          , study effect-size estimates
</p>
</li>
<li><p> se            , corresponding standard errors of the study effect-size estimates
</p>
</li>
<li><p> variant_names , variant names
</p>
</li>
<li><p> study_names   , study names
</p>
</li>
<li><p> study         , study numbers
</p>
</li>
<li><p> snp           , snp numbers
</p>
</li>
<li><p> tau2          , tau_squared, estimate of amount of between-study variance
</p>
</li>
<li><p> I2            , I_squared, heterogeneity index (Higgins inconsistency metric)
representing proportion of total observed variation due to 
between-study variance
</p>
</li>
<li><p> Q             , Q-statistic (Cochran's Q)
</p>
</li>
<li><p> xb            , prediction excluding random effects
</p>
</li>
<li><p> xbse          , standard error of prediction excluding random effects
</p>
</li>
<li><p> xbu           , predictions including random effects
</p>
</li>
<li><p> stdxbu        , standard error of prediction (fitted values) 
including random effects
</p>
</li>
<li><p> hat           , leverage a.k.a diagonal elements of the projection hat matrix
</p>
</li>
<li><p> rawresid      , raw residuals
</p>
</li>
<li><p> uncondse      , unconditional standard errors
</p>
</li>
<li><p> spre          , <em>SPRE</em> statistics (standardised predicted random effects) i.e. 
raw residuals divided by the unconditional standard errors
</p>
</li></ul>

</li></ul>



<h3>Methods (by class)</h3>


<ul>
<li> <p><code>default</code>: Computes <em>SPRE</em> statistics in genetic association meta-analyses
</p>
</li></ul>


<h3>See Also</h3>

<p><a href="https://magosil86.github.io/getspres/">https://magosil86.github.io/getspres/</a> to the visit <code>getspres</code> website.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(getspres)


# Calculate SPRE statistics for a subset of variants in the heartgenes214 dataset.
# heartgenes214 is a case-control GWAS meta-analysis of coronary artery disease.
# To learn more about the heartgenes214 dataset ?heartgenes214

# Calculating SPRE statistics for 3 variants in heartgenes214

heartgenes3 &lt;- subset(heartgenes214, 
    variants %in% c("rs10139550", "rs10168194", "rs11191416")) 

getspres_results &lt;- getspres(beta_in = heartgenes3$beta_flipped, 
                               se_in = heartgenes3$gcse, 
                      study_names_in = heartgenes3$studies, 
                    variant_names_in = heartgenes3$variants)


# Explore results generated by the getspres function
str(getspres_results)

# Retrieve number of studies and variants
getspres_results$number_variants
getspres_results$number_studies

# Retrieve SPRE dataset
df_spres &lt;- getspres_results$spre_dataset
head(df_spres)

# Extract SPREs from SPRE dataset
head(spres &lt;- df_spres[, "spre"])



# Exploring available options in the getspres function:
#     1. Estimate heterogeneity using "REML", default is "DL"
#     2. Calculate SPRE statistics verbosely
getspres_results &lt;- getspres(beta_in = heartgenes3$beta_flipped, 
                               se_in = heartgenes3$gcse, 
                      study_names_in = heartgenes3$studies, 
                    variant_names_in = heartgenes3$variants,
                         tau2_method = "REML",
                      verbose_output = TRUE)

str(getspres_results)



</code></pre>

<hr>
<h2 id='heartgenes214'>heartgenes214.</h2><span id='topic+heartgenes214'></span>

<h3>Description</h3>

<p>heartgenes214 is a multi-ethnic GWAS meta-analysis dataset for coronary
artery disease.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heartgenes214
</code></pre>


<h3>Format</h3>

<p>A data frame with seven variables:
</p>

<dl>
<dt><code>beta_flipped</code></dt><dd><p>Effect-sizes expressed as log odds ratios. Numeric</p>
</dd>
<dt><code>gcse</code></dt><dd><p>Standard errors</p>
</dd>
<dt><code>studies</code></dt><dd><p>Names of participating studies</p>
</dd>
<dt><code>variants</code></dt><dd><p>Names of genetic variants/SNPs</p>
</dd>
<dt><code>cases</code></dt><dd><p>Number of cases in each participating study</p>
</dd>
<dt><code>controls</code></dt><dd><p>Number of controls in each participating study</p>
</dd>
<dt><code>fdr214_gwas46</code></dt><dd><p>Flag indicating GWAS significant variants, 
1: Not GWAS-significant, 2: GWAS-significant</p>
</dd>
</dl>



<h3>Details</h3>

<p>It comprises summary data (effect-sizes and their corresponding standard 
errors) for 48 studies (68,801 cases and 123,504 controls), at 214 lead 
variants independently associated with coronary artery disease 
(P &lt; 0.00005, FDR &lt; 5%). Of the 214 lead variants, 44 are genome-wide 
significant (p &lt; 5e-08). The meta-analysis dataset is based on 
individuals of: African American, Hispanic American, East Asian, 
South Asian, Middle Eastern and European ancestry. 
</p>
<p>The study effect-sizes have been flipped to ensure alignment of the 
effect alleles.
</p>
<p>Standard errors were genomically corrected at the study-level.
</p>


<h3>Source</h3>

<p>Magosi LE, Goel A, Hopewell JC, Farrall M, on behalf of the 
CARDIoGRAMplusC4D Consortium (2017) Identifying systematic 
heterogeneity patterns in genetic association meta-analysis studies. 
PLoS Genet 13(5): e1006755. https://doi.org/10.1371/journal.pgen.1006755.
</p>
<p><a href="https://magosil86.github.io/getmstatistic/">https://magosil86.github.io/getmstatistic/</a>
</p>

<hr>
<h2 id='plotspres'><code>plotspres</code> generates forest plots showing <em>SPRE</em> statistics.</h2><span id='topic+plotspres'></span><span id='topic+plotspre'></span><span id='topic+forestspre'></span><span id='topic+spreforest'></span><span id='topic+plotspres.default'></span>

<h3>Description</h3>

<p>Forest plots showing <em>SPRE</em> (standardised predicted random-effects) statistics 
can be useful in highlighting overly influential outlier studies with the potential
to inflate summary effect estimates in genetic association meta-analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotspres(beta_in, se_in, study_names_in, variant_names_in, spres_in, ...)

## Default S3 method:
plotspres(
  beta_in,
  se_in,
  study_names_in,
  variant_names_in,
  spres_in,
  spre_colour_palette = c("mono_colour", "black"),
  set_studyNOs_as_studyIDs = FALSE,
  set_study_field_width = "%02.0f",
  set_cex = 0.66,
  set_xlim,
  set_ylim,
  set_at,
  tau2_method = "DL",
  adjust_labels = 1,
  save_plot = TRUE,
  verbose_output = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotspres_+3A_beta_in">beta_in</code></td>
<td>
<p>A numeric vector of observed study effects e.g. log odds-ratios.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_se_in">se_in</code></td>
<td>
<p>A numeric vector of standard errors, genomically corrected at study-level.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_study_names_in">study_names_in</code></td>
<td>
<p>A character vector of study names.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_variant_names_in">variant_names_in</code></td>
<td>
<p>A character vector of variant names e.g. rsIDs.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_spres_in">spres_in</code></td>
<td>
<p>A numeric vector of <em>SPRE</em> statistics.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_spre_colour_palette">spre_colour_palette</code></td>
<td>
<p>An optional character vector specifying the colour palette that should be used for observed study effects. There are 3 types of colour palettes available, namely: &quot;mono_colour&quot;, &quot;dual_colour&quot; and &quot;multi_colour&quot;; with the &quot;dual_colour&quot; palette, observed study effects with negative <em>SPRE</em> statistics are coloured differently from those with positive <em>SPRE</em> statistics, and with the &quot;multi_colour&quot; palette observed study effects are colored in a gradient according to the <em>SPRE</em> statistic values. Default palette option is <code>spre_colour_palette = c("mono_colour", "black")</code>.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_set_studynos_as_studyids">set_studyNOs_as_studyIDs</code></td>
<td>
<p>An optional boolean specifying whether study numbers should be used as study IDs in the forest plot. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_set_study_field_width">set_study_field_width</code></td>
<td>
<p>An optional character vector of format strings, akin to the fmt character vector in the sprintf function. (Default is <code>set_study_field_width = "%02.0f"</code>).</p>
</td></tr>
<tr><td><code id="plotspres_+3A_set_cex">set_cex</code></td>
<td>
<p>An optional character scalar and symbol expansion factor indicating the percentage by which text and symbols should be scaled relative to the reference; e.g. 1=reference, 1.3 is 30% larger, 0.3 is 30% smaller. (Default is <code>cex = 0.66</code>).</p>
</td></tr>
<tr><td><code id="plotspres_+3A_set_xlim">set_xlim</code></td>
<td>
<p>An optional numeric vector of length 2 indicating the horizontal limits of the plot region.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_set_ylim">set_ylim</code></td>
<td>
<p>An optional numeric vector of length 2 indicating the y-axis limits of the plot.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_set_at">set_at</code></td>
<td>
<p>An optional numeric vector indicating position of the x-axis tick marks and corresponding labels.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_tau2_method">tau2_method</code></td>
<td>
<p>An optional character scalar, specifying the method that should be used to estimate heterogeneity either through DerSimonian and Laird's moment-based estimate &quot;DL&quot; or restricted maximum likelihood &quot;REML&quot;. Note: The REML method uses the iterative Fisher scoring algorithm (step length = 0.5, maximum iterations = 10000) to estimate tau2. Default is &quot;DL&quot;.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_adjust_labels">adjust_labels</code></td>
<td>
<p>An optional numeric scalar value that tweaks label (column header) positions. (Default is <code>adjust_labels = 1</code>).</p>
</td></tr>
<tr><td><code id="plotspres_+3A_save_plot">save_plot</code></td>
<td>
<p>An optional boolean to save forestplot as a tiff file. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotspres_+3A_verbose_output">verbose_output</code></td>
<td>
<p>An optional boolean to display intermediate output. (Default is <code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>plotspres</code> takes as input <em>SPRE</em> statistics, observed study effects 
and corresponding standard errors (i.e. summary data). The observed study effects 
(i.e. study effect-size estimates) could be association statistics from either 
quantitative or binary trait meta-analyses, for instance, linear regression coefficients 
might be employed for quantitative traits and log-transformed logistic regression 
coefficients (per-allele log odds ratios) used for case-control meta-analyses. 
<em>SPRE</em> statistics can be calculated using the <code><a href="#topic+getspres">getspres</a></code> function.
</p>
<p><code>plotspres</code> uses inverse-variance weighted fixed and random-effects 
meta-analysis models in the <code>metafor</code> R package to generate forestplots.
</p>


<h3>Value</h3>

<p>Returns a list containing:
</p>

<ul>
<li><p> number_variants         A numeric scalar indicating the number of variants 
</p>
</li>
<li><p> number_studies          A numeric scalar indicating the number of studies
</p>
</li>
<li><p> fixed_effect_results    A list of fixed-effect meta-analysis results for each variant examined
</p>
</li>
<li><p> random_effects_results  A list of random-effects meta-analysis results for each variant examined
</p>
</li>
<li><p> spre_forestplot_dataset A dataframe of the data provided by the user for analysis which contains the following fields:
</p>

<ul>
<li><p> beta            , study effect-size estimates
</p>
</li>
<li><p> se              , corresponding standard errors of study effect-size estimates
</p>
</li>
<li><p> variant_names   , variant names
</p>
</li>
<li><p> study_names     , study names
</p>
</li>
<li><p> spre            , <em>SPRE</em> (standardised predicted random-effects) statistics
</p>
</li>
<li><p> study_numbers   , study numbers
</p>
</li>
<li><p> variant_numbers , variant numbers
</p>
</li></ul>

</li></ul>



<h3>Methods (by class)</h3>


<ul>
<li> <p><code>default</code>: Generates forest plots showing <em>SPRE</em> statistics
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+getspres">getspres</a></code> to calculate <em>SPRE</em> statistics and the 
<code><a href="metafor.html#topic+rma.uni">metafor</a></code> package to explore implementations of fixed and 
random-effects meta-analysis models in R. To access more information and examples 
visit the getspres website at: <a href="https://magosil86.github.io/getspres/">https://magosil86.github.io/getspres/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(getspres)


# Generate a forest plot showing SPRE statistics for variants in heartgenes214.
# heartgenes214 is a case-control GWAS meta-analysis of coronary artery disease.
# To learn more about the heartgenes214 dataset ?heartgenes214

# Calculating SPRE statistics for 3 variants in heartgenes214

heartgenes3 &lt;- subset(heartgenes214, 
    variants %in% c("rs10139550", "rs10168194", "rs11191416")) 

getspres_results &lt;- getspres(beta_in = heartgenes3$beta_flipped, 
                               se_in = heartgenes3$gcse, 
                      study_names_in = heartgenes3$studies, 
                    variant_names_in = heartgenes3$variants)

# Explore results generated by the getspres function
str(getspres_results)

# Retrieve number of studies and variants
getspres_results$number_variants
getspres_results$number_studies

# Retrieve SPRE dataset
df_spres &lt;- getspres_results$spre_dataset
head(df_spres)

# Extract SPREs from SPRE dataset
head(spres &lt;- df_spres[, "spre"])


# Generating forest plots showing SPREs for variants in heartgenes3

# Forest plot with default settings
# Tip: To store plots set save_plot = TRUE (useful when generating multiple plots)
plotspres_res &lt;- plotspres(beta_in = df_spres$beta, 
                             se_in = df_spres$se, 
                    study_names_in = as.character(df_spres$study_names), 
                  variant_names_in = as.character(df_spres$variant_names),
                          spres_in = df_spres$spre,
                         save_plot = FALSE)

# Explore results generated by the plotspres function

# Retrieve number of studies and variants
plotspres_res$number_variants
plotspres_res$number_studies

# Retrieve fixed and random-effects meta-analysis results
fixed_effect_res &lt;- plotspres_res$fixed_effect_results
random_effects_res &lt;- plotspres_res$random_effects_results

# Retrieve dataset that was used to generate forest plots
df_plotspres &lt;- plotspres_res$spre_forestplot_dataset



# Retrieve more detailed meta-analysis output
str(plotspres_res)



# Explore available options for plotspres forest plots: 
#   1. Colorize study-effect estimates according to SPRE statistic values
#   2. Label studies by study number instead of study names
#   3. Format study labels (useful when using study numbers as study labels)
#   4. Change text size
#   5. Adjust x and y axes limits
#   6. Change method used to estimate amount of heterogeneity from "DL" to "REML"
#   7. Run verbosely to show intermediate results
#   8. Adjust label (i.e. column header) positions
#   9. Save plot as a tiff file (useful when generating multiple plots)

# Colorize study-effect estimates according to SPRE statistic values

# Use a dual colour palette for observed study effects so that study effect estimates 
#   with negative SPRE statistics are coloured differently from those with positive 
#   SPRE statistics.
plotspres_res &lt;- plotspres(beta_in = df_spres$beta, 
                             se_in = df_spres$se, 
                    study_names_in = as.character(df_spres$study_names), 
                  variant_names_in = as.character(df_spres$variant_names),
                          spres_in = df_spres$spre,
               spre_colour_palette = c("dual_colour", c("blue","black")),
                         save_plot = FALSE)


# Use a multi-colour palette for observed study effects so that study effects estimates
#   are colored in a gradient according to SPRE statistic values.
#   Available multi-colour palettes:
#
#       gr_devices_palettes: "rainbow", "cm.colors", "topo.colors", "terrain.colors" 
#                            and "heat.colors" 
#
#       colorspace_hcl_hsv_palettes: "rainbow_hcl", "diverge_hcl", "terrain_hcl", 
#                                    "sequential_hcl" and "diverge_hsl"
#
#       color_ramps_palettes: "matlab.like", "matlab.like2", "magenta2green", 
#                             "cyan2yellow", "blue2yellow", "green2red", 
#                             "blue2green" and "blue2red"

plotspres_res &lt;- plotspres(beta_in = df_spres$beta, 
                             se_in = df_spres$se, 
                    study_names_in = as.character(df_spres$study_names), 
                  variant_names_in = as.character(df_spres$variant_names),
                          spres_in = df_spres$spre,
               spre_colour_palette = c("multi_colour", "rainbow"),
                         save_plot = FALSE)

# Exploring other options in the plotspres function.
#     Label studies by study number instead of study names (option: set_studyNOs_as_studyIDs)
#     Format study labels (option: set_study_field_width)
#     Adjust text size (option: set_cex)
#     Adjust x and y axes limits (options: set_xlim, set_ylim)
#     Change method used to estimate heterogeneity from "DL" to "REML" (option: tau2_method)
#     Adjust position of x-axis tick marks (option: set_at)
#     Run verbosely (option: verbose_output)

df_rs10139550 &lt;- subset(df_spres, variant_names == "rs10139550")
plotspres_res &lt;- plotspres(beta_in = df_rs10139550$beta, 
                             se_in = df_rs10139550$se, 
                    study_names_in = as.character(df_rs10139550$study_names), 
                  variant_names_in = as.character(df_rs10139550$variant_names),
                          spres_in = df_rs10139550$spre,
               spre_colour_palette = c("multi_colour", "matlab.like"),
          set_studyNOs_as_studyIDs = TRUE,
             set_study_field_width = "%03.0f",
                           set_cex = 0.75, set_xlim = c(-2,2), set_ylim = c(-1.5,51),
                            set_at = c(-0.6, -0.4, -0.2,  0.0,  0.2,  0.4,  0.6),
                       tau2_method = "REML", verbose_output = TRUE,
                         save_plot = FALSE)

# Adjust label (i.e. column header) position, also keep plot in graphics window rather
#     than save as tiff file
df_rs10139550_3studies &lt;- subset(df_rs10139550, as.numeric(df_rs10139550$study_names) &lt;= 3)

# Before adjusting label positions
plotspres_res &lt;- plotspres(beta_in = df_rs10139550_3studies$beta, 
                             se_in = df_rs10139550_3studies$se, 
                    study_names_in = as.character(df_rs10139550_3studies$study_names), 
                  variant_names_in = as.character(df_rs10139550_3studies$variant_names),
                          spres_in = df_rs10139550_3studies$spre,
               spre_colour_palette = c("dual_colour", c("blue","black")),
                         save_plot = FALSE)

# After adjusting label positions
plotspres_res &lt;- plotspres(beta_in = df_rs10139550_3studies$beta, 
                             se_in = df_rs10139550_3studies$se, 
                    study_names_in = as.character(df_rs10139550_3studies$study_names), 
                  variant_names_in = as.character(df_rs10139550_3studies$variant_names),
                          spres_in = df_rs10139550_3studies$spre,
               spre_colour_palette = c("dual_colour", c("blue","black")),
                     adjust_labels = 1.7, save_plot = FALSE)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
