<!DOCTYPE html><html lang="en"><head><title>Help for package GEVACO</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GEVACO}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cov_example'><p>Covariate Information</p></a></li>
<li><a href='#geno_example'><p>Genotypic Information</p></a></li>
<li><a href='#GxEscreen'><p>Gene-Environment Interaction: Genome-wide Screen</p></a></li>
<li><a href='#GxEtest'><p>Joint test for genetic and gene-environment interaction effects for a single SNP</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Joint Test of Gene and GxE Interactions via Varying Coefficients</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhengyang Zhou [aut]
        Sydney Manning [aut, cre]
        Chao Xing [aut]
        Hung-Chih Ku [aut]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>Description:</td>
<td>A novel statistical model to detect the joint genetic and dynamic gene-environment (GxE) interaction with continuous traits in genetic association studies. It uses varying-coefficient models to account for different GxE trajectories, regardless whether the relationship is linear or not. The package includes one function, GxEtest(), to test a single genetic variant (e.g., a single nucleotide polymorphism or SNP), and another function, GxEscreen(), to test for a set of genetic variants. The method involves a likelihood ratio test described in Crainiceanu, C. M., and Ruppert, D. (2004) &lt;<a href="https://doi.org/10.1111%2Fj.1467-9868.2004.00438.x">doi:10.1111/j.1467-9868.2004.00438.x</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>nlme, RLRsim, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sydney Manning &lt;sydney.manning@unthsc.edu&gt;</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-12-08 17:43:36 UTC; Student</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-12-10 12:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='cov_example'>Covariate Information</h2><span id='topic+cov_example'></span>

<h3>Description</h3>

<p>Dataset containing example covariate information for analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cov_example
</code></pre>


<h3>Format</h3>

<p>A dataframe with 500 rows and 4 variables:
</p>

<dl>
<dt>y</dt><dd><p>phenotypic factor of interest</p>
</dd>
<dt>BMI</dt><dd><p>BMI in kg/m^2</p>
</dd>
<dt>age</dt><dd><p>age in years</p>
</dd>
<dt>sex</dt><dd><p>sex coded as 1 &amp; 2</p>
</dd>
</dl>


<hr>
<h2 id='geno_example'>Genotypic Information</h2><span id='topic+geno_example'></span>

<h3>Description</h3>

<p>Dataset containing example genotypic information for analysis, coded as 0 1 2. Has 500 observations
and assumes a minor allele frequency threshold of 0.05. We included the first 20 SNPs to meet this
threshold in this example. Each column holds the information for an individual SNP, and each row
contains the genotypic information for that observation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geno_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 500 rows and 20 columns.
</p>

<hr>
<h2 id='GxEscreen'>Gene-Environment Interaction: Genome-wide Screen</h2><span id='topic+GxEscreen'></span>

<h3>Description</h3>

<p>Function to test for the joint genetic and gene-environment
interaction effects for a set of variant using a LRT model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GxEscreen(dat, geno, nsim = 1e+05, K = 7)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GxEscreen_+3A_dat">dat</code></td>
<td>
<p>a data frame with covariate information. Col 1 should be phenotype,
col 2 should be environmental factor, col 3 and later should be additional covariates</p>
</td></tr>
<tr><td><code id="GxEscreen_+3A_geno">geno</code></td>
<td>
<p>a genotype matrix with 0-1-2 coding</p>
</td></tr>
<tr><td><code id="GxEscreen_+3A_nsim">nsim</code></td>
<td>
<p>the number of replicates in obtaining the p-value (standard 1e5)</p>
</td></tr>
<tr><td><code id="GxEscreen_+3A_k">K</code></td>
<td>
<p>the number of knots used to control the flexibility in modeling GxE interaction</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector containing the p-value from the LRT associated with each SNP
</p>


<h3>Examples</h3>

<pre><code class='language-R'>GxEscreen(cov_example, geno_example, nsim=1e5, K=7)
</code></pre>

<hr>
<h2 id='GxEtest'>Joint test for genetic and gene-environment interaction effects for a single SNP</h2><span id='topic+GxEtest'></span>

<h3>Description</h3>

<p>Function to test for the joint genetic and gene-environment
interaction effects for a single variant using a LRT model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GxEtest(dat, snp_geno, nsim = 1e+05, K)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GxEtest_+3A_dat">dat</code></td>
<td>
<p>a data frame with covariate information. Column 1 should be phenotype,
column 2 should be the environmental factor of interest, columns 3 and later should be
additional covariates</p>
</td></tr>
<tr><td><code id="GxEtest_+3A_snp_geno">snp_geno</code></td>
<td>
<p>a vector containing genotypic information of SNP of interest to be tested</p>
</td></tr>
<tr><td><code id="GxEtest_+3A_nsim">nsim</code></td>
<td>
<p>the number of replicates in obtaining the p-value (standard 1e5)</p>
</td></tr>
<tr><td><code id="GxEtest_+3A_k">K</code></td>
<td>
<p>the number of knots used to control the flexibility in modeling GxE interaction</p>
</td></tr>
</table>


<h3>Value</h3>

<p>empirical p-value obtained as the proportion of T0 that are greater than
the observed test statistic T
</p>


<h3>References</h3>

<p>Crainiceanu, C. M., &amp; Ruppert, D. (2004). Likelihood ratio tests in linear mixed models with one variance component.
Journal of the Royal Statistical Society Series B-Statistical Methodology, 66, 165-185. doi: 10.1111/j.1467-9868.2004.00438.x
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
