<!DOCTYPE html><html><head><title>Help for package GGally</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GGally}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GGally-package'><p>GGally: Extension to 'ggplot2'</p></a></li>
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#+.gg'><p>Modify a <code>ggmatrix</code> object by adding an <span class="pkg">ggplot2</span> object to all plots</p></a></li>
<li><a href='#add_and_overwrite_aes'><p>Add new aes</p></a></li>
<li><a href='#add_ref_boxes'><p>Add reference boxes around each cell of the glyphmap.</p></a></li>
<li><a href='#add_ref_lines'><p>Add reference lines for each cell of the glyphmap.</p></a></li>
<li><a href='#australia_PISA2012'><p>Programme for International Student Assessment (PISA) 2012 Data for Australia</p></a></li>
<li><a href='#baseball'><p>Yearly batting records for all major league baseball players</p></a></li>
<li><a href='#brew_colors'><p>RColorBrewer Set1 colors</p></a></li>
<li><a href='#broomify'><p>Broomify a model</p></a></li>
<li><a href='#column_is_character'><p>Get vector of variable types from data frame</p></a></li>
<li><a href='#eval_data_col'><p>Evaluate data column</p></a></li>
<li><a href='#find_plot_type'><p>Find plot types</p></a></li>
<li><a href='#flea'><p>Historical data used for classification examples.</p></a></li>
<li><a href='#fn_switch'><p>Function switch</p></a></li>
<li><a href='#get_x_axis_labels'><p>Get x axis labels</p></a></li>
<li><a href='#getPlot'><p>Subset a <code>ggmatrix</code> object</p></a></li>
<li><a href='#ggally_autopoint'><p>Scatterplot for continuous and categorical variables</p></a></li>
<li><a href='#ggally_barDiag'><p>Bar plot</p></a></li>
<li><a href='#ggally_blank'><p>Blank plot</p></a></li>
<li><a href='#ggally_box'><p>Box plot</p></a></li>
<li><a href='#ggally_colbar'><p>Column and row bar plots</p></a></li>
<li><a href='#ggally_cor'><p>Correlation value plot</p></a></li>
<li><a href='#ggally_cor_v1_5'><p>Correlation value plot</p></a></li>
<li><a href='#ggally_count'><p>Display counts of observations</p></a></li>
<li><a href='#ggally_cross'><p>Plots the number of observations</p></a></li>
<li><a href='#ggally_crosstable'><p>Display a cross-tabulated table</p></a></li>
<li><a href='#ggally_density'><p>Bivariate density plot</p></a></li>
<li><a href='#ggally_densityDiag'><p>Univariate density plot</p></a></li>
<li><a href='#ggally_denstrip'><p>Tile plot with facets</p></a></li>
<li><a href='#ggally_diagAxis'><p>Internal axis labels for ggpairs</p></a></li>
<li><a href='#ggally_dot'><p>Grouped dot plot</p></a></li>
<li><a href='#ggally_dot_and_box'><p>Box and dot plot</p></a></li>
<li><a href='#ggally_facetbar'><p>Faceted bar plot</p></a></li>
<li><a href='#ggally_facetdensity'><p>Faceted density plot</p></a></li>
<li><a href='#ggally_facetdensitystrip'><p>Density or tiles plot with facets</p></a></li>
<li><a href='#ggally_facethist'><p>Faceted histogram</p></a></li>
<li><a href='#ggally_na'><p>NA plot</p></a></li>
<li><a href='#ggally_nostic_cooksd'><p><code>ggnostic</code> Cook's distance</p></a></li>
<li><a href='#ggally_nostic_hat'><p><code>ggnostic</code> leverage points</p></a></li>
<li><a href='#ggally_nostic_line'><p><code>ggnostic</code> background line with geom</p></a></li>
<li><a href='#ggally_nostic_resid'><p><code>ggnostic</code> residuals</p></a></li>
<li><a href='#ggally_nostic_se_fit'><p><code>ggnostic</code> fitted value's standard error</p></a></li>
<li><a href='#ggally_nostic_sigma'><p><code>ggnostic</code> leave one out model sigma</p></a></li>
<li><a href='#ggally_nostic_std_resid'><p><code>ggnostic</code> standardized residuals</p></a></li>
<li><a href='#ggally_points'><p>Scatter plot</p></a></li>
<li><a href='#ggally_ratio'><p>Mosaic plot</p></a></li>
<li><a href='#ggally_smooth'><p>Scatter plot with a smoothed line</p></a></li>
<li><a href='#ggally_statistic'><p>Generalized text display</p></a></li>
<li><a href='#ggally_summarise_by'><p>Summarize a continuous variable by each value of a discrete variable</p></a></li>
<li><a href='#ggally_table'><p>Display a table of the number of observations</p></a></li>
<li><a href='#ggally_text'><p>Text plot</p></a></li>
<li><a href='#ggally_trends'><p>Trends line plot</p></a></li>
<li><a href='#ggbivariate'><p>Display an outcome using several potential explanatory variables</p></a></li>
<li><a href='#ggcoef'><p>Model coefficients with <span class="pkg">broom</span> and <span class="pkg">ggplot2</span></p></a></li>
<li><a href='#ggcorr'><p>Correlation matrix</p></a></li>
<li><a href='#ggduo'><p><span class="pkg">ggplot2</span> generalized pairs plot for two columns sets of data</p></a></li>
<li><a href='#ggfacet'><p>Single <span class="pkg">ggplot2</span> plot matrix with <code>facet_grid</code></p></a></li>
<li><a href='#gglegend'><p>Plot only legend of plot function</p></a></li>
<li><a href='#ggmatrix'><p><span class="pkg">ggplot2</span> plot matrix</p></a></li>
<li><a href='#ggmatrix_gtable'><p><code>ggmatrix</code> <span class="pkg">gtable</span> object</p></a></li>
<li><a href='#ggmatrix_location'><p><code>ggmatrix</code> plot locations</p></a></li>
<li><a href='#ggmatrix_progress'><p><code>ggmatrix</code> default progress bar</p></a></li>
<li><a href='#ggnet'><p>Network plot</p></a></li>
<li><a href='#ggnet2'><p>Network plot</p></a></li>
<li><a href='#ggnetworkmap'><p>Network plot map overlay</p></a></li>
<li><a href='#ggnostic'><p>Plot matrix of statistical model diagnostics</p></a></li>
<li><a href='#ggpairs'><p>ggplot2 generalized pairs plot</p></a></li>
<li><a href='#ggparcoord'><p>Parallel coordinate plot</p></a></li>
<li><a href='#ggscatmat'><p>Traditional scatterplot matrix for purely quantitative variables</p></a></li>
<li><a href='#ggsurv'><p>Survival curves</p></a></li>
<li><a href='#ggtable'><p>Cross-tabulated tables of discrete variables</p></a></li>
<li><a href='#ggts'><p>Multiple time series</p></a></li>
<li><a href='#glyphplot'><p>Glyph plot class</p></a></li>
<li><a href='#glyphs'><p>Create <code>glyphplot</code> data</p></a></li>
<li><a href='#grab_legend'><p>Grab the legend and print it as a plot</p></a></li>
<li><a href='#happy'><p>Data related to happiness from the General Social Survey, 1972-2006.</p></a></li>
<li><a href='#is_blank_plot'><p>Is Blank Plot?</p>
Find out if the plot equals a blank plot</a></li>
<li><a href='#is_date'><p>Check if object is a date</p></a></li>
<li><a href='#is_horizontal'><p>Check if plot is horizontal</p></a></li>
<li><a href='#lowertriangle'><p>lowertriangle - rearrange dataset as the preparation of <code>ggscatmat</code> function</p></a></li>
<li><a href='#mapping_color_to_fill'><p>Aesthetic mapping color fill</p></a></li>
<li><a href='#mapping_string'><p>Aes name</p></a></li>
<li><a href='#mapping_swap_x_y'><p>Swap x and y mapping</p></a></li>
<li><a href='#model_response_variables'><p>Model term names</p></a></li>
<li><a href='#nasa'><p>Data from the Data Expo JSM 2006.</p></a></li>
<li><a href='#pigs'><p>United Kingdom Pig Production</p></a></li>
<li><a href='#plot_types'><p>Plot Types</p></a></li>
<li><a href='#plotting_data_type'><p>Get plotting data type</p></a></li>
<li><a href='#print_if_interactive'><p>Print if not CRAN</p></a></li>
<li><a href='#print.ggmatrix'><p>Print <code>ggmatrix</code> object</p></a></li>
<li><a href='#psychademic'><p>UCLA canonical correlation analysis data</p></a></li>
<li><a href='#putPlot'><p>Insert a plot into a <code>ggmatrix</code> object</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#remove_color_unless_equal'><p>Remove colour mapping unless found in select mapping keys</p></a></li>
<li><a href='#require_namespaces'><p>Loads package namespaces</p></a></li>
<li><a href='#rescale01'><p>Rescaling functions</p></a></li>
<li><a href='#scag_order'><p>Find order of variables</p></a></li>
<li><a href='#scatmat'><p>Plots the lowertriangle and density plots of the scatter plot matrix.</p></a></li>
<li><a href='#singleClassOrder'><p>Order axis variables</p></a></li>
<li><a href='#skewness'><p>Sample skewness</p></a></li>
<li><a href='#str.ggmatrix'><p><code>ggmatrix</code> structure</p></a></li>
<li><a href='#tips'><p>Tipping data</p></a></li>
<li><a href='#twitter_spambots'><p>Twitter spambots</p></a></li>
<li><a href='#uppertriangle'><p>Rearrange dataset as the preparation of <code>ggscatmat</code> function</p></a></li>
<li><a href='#v1_ggmatrix_theme'><p>Modify a <code>ggmatrix</code> object by adding an <span class="pkg">ggplot2</span> object to all</p></a></li>
<li><a href='#vig_ggally'><p>View GGally vignettes</p></a></li>
<li><a href='#wrap_fn_with_param_arg'><p>Wrap a function with different parameter values</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.2.1</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2.0)]</td>
</tr>
<tr>
<td>Title:</td>
<td>Extension to 'ggplot2'</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://ggobi.github.io/ggally/">https://ggobi.github.io/ggally/</a>, <a href="https://github.com/ggobi/ggally">https://github.com/ggobi/ggally</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ggobi/ggally/issues">https://github.com/ggobi/ggally/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>
    The R package 'ggplot2' is a plotting system based on the grammar of graphics.
    'GGally' extends 'ggplot2' by adding several functions
    to reduce the complexity of combining geometric objects with transformed data.
    Some of these functions include a pairwise plot matrix, a two group pairwise plot
    matrix, a parallel coordinates plot, a survival plot, and several functions to
    plot networks.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1), ggplot2 (&ge; 3.4.4)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr (&ge; 1.0.0), tidyr (&ge; 1.3.0), grDevices, grid, ggstats,
gtable (&ge; 0.2.0), lifecycle, plyr (&ge; 1.8.3), progress,
RColorBrewer, rlang, scales (&ge; 1.1.0), utils, magrittr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>broom (&ge; 0.7.0), broom.helpers (&ge; 1.3.0), chemometrics,
geosphere (&ge; 1.5-1), ggforce, Hmisc, igraph (&ge; 1.0.1),
intergraph (&ge; 2.0-2), labelled, maps (&ge; 3.1.0), mapproj,
nnet, network (&ge; 1.17.1), scagnostics, sna (&ge; 2.3-2),
survival, rmarkdown, roxygen2, testthat, crosstalk, knitr,
spelling, emmeans, vdiffr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>openssl</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>lifecycle</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-13 19:02:36 UTC; barret</td>
</tr>
<tr>
<td>Author:</td>
<td>Barret Schloerke [aut, cre],
  Di Cook [aut, ths],
  Joseph Larmarange [aut],
  Francois Briatte [aut],
  Moritz Marbach [aut],
  Edwin Thoen [aut],
  Amos Elberg [aut],
  Ott Toomet [ctb],
  Jason Crowley [aut],
  Heike Hofmann [ths],
  Hadley Wickham [ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Barret Schloerke &lt;schloerke@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-14 00:53:32 UTC</td>
</tr>
</table>
<hr>
<h2 id='GGally-package'>GGally: Extension to 'ggplot2'</h2><span id='topic+GGally'></span><span id='topic+GGally-package'></span>

<h3>Description</h3>

<p>The R package 'ggplot2' is a plotting system based on the grammar of graphics. 'GGally' extends 'ggplot2' by adding several functions to reduce the complexity of combining geometric objects with transformed data. Some of these functions include a pairwise plot matrix, a two group pairwise plot matrix, a parallel coordinates plot, a survival plot, and several functions to plot networks.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Barret Schloerke <a href="mailto:schloerke@gmail.com">schloerke@gmail.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Di Cook <a href="mailto:dicook@monash.edu">dicook@monash.edu</a> [thesis advisor]
</p>
</li>
<li><p> Joseph Larmarange <a href="mailto:joseph@larmarange.net">joseph@larmarange.net</a>
</p>
</li>
<li><p> Francois Briatte <a href="mailto:f.briatte@gmail.com">f.briatte@gmail.com</a>
</p>
</li>
<li><p> Moritz Marbach <a href="mailto:mmarbach@mail.uni-mannheim.de">mmarbach@mail.uni-mannheim.de</a>
</p>
</li>
<li><p> Edwin Thoen <a href="mailto:edwinthoen@gmail.com">edwinthoen@gmail.com</a>
</p>
</li>
<li><p> Amos Elberg <a href="mailto:amos.elberg@gmail.com">amos.elberg@gmail.com</a>
</p>
</li>
<li><p> Jason Crowley <a href="mailto:crowley.jason.s@gmail.com">crowley.jason.s@gmail.com</a>
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Ott Toomet <a href="mailto:otoomet@gmail.com">otoomet@gmail.com</a> [contributor]
</p>
</li>
<li><p> Heike Hofmann <a href="mailto:hofmann@iastate.edu">hofmann@iastate.edu</a> [thesis advisor]
</p>
</li>
<li><p> Hadley Wickham <a href="mailto:h.wickham@gmail.com">h.wickham@gmail.com</a> [thesis advisor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://ggobi.github.io/ggally/">https://ggobi.github.io/ggally/</a>
</p>
</li>
<li> <p><a href="https://github.com/ggobi/ggally">https://github.com/ggobi/ggally</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/ggobi/ggally/issues">https://github.com/ggobi/ggally/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='+2B.gg'>Modify a <code><a href="#topic+ggmatrix">ggmatrix</a></code> object by adding an <span class="pkg">ggplot2</span> object to all plots</h2><span id='topic++2B.gg'></span><span id='topic+add_to_ggmatrix'></span>

<h3>Description</h3>

<p>This operator allows you to add <span class="pkg">ggplot2</span> objects to a <code><a href="#topic+ggmatrix">ggmatrix</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gg'
e1 + e2

add_to_ggmatrix(e1, e2, location = NULL, rows = NULL, cols = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B2B.gg_+3A_e1">e1</code></td>
<td>
<p>An object of class <code><a href="#topic+ggnostic">ggnostic</a></code> or <code>ggplot</code></p>
</td></tr>
<tr><td><code id="+2B2B.gg_+3A_e2">e2</code></td>
<td>
<p>A component to add to <code>e1</code></p>
</td></tr>
<tr><td><code id="+2B2B.gg_+3A_location">location</code></td>
<td>

<dl>
<dt><code>"all"</code>, <code>TRUE</code></dt><dd><p>All row and col combinations</p>
</dd>
<dt><code>"none"</code></dt><dd><p>No row and column combinations</p>
</dd>
<dt><code>"upper"</code></dt><dd><p>Locations where the column value is higher than the row value</p>
</dd>
<dt><code>"lower"</code></dt><dd><p>Locations where the row value is higher than the column value</p>
</dd>
<dt><code>"diag"</code></dt><dd><p>Locations where the column value is equal to the row value</p>
</dd>
<dt><code>matrix</code> or <code>data.frame</code></dt><dd>
<p><code>matrix</code> values will be converted into <code>data.frame</code>s.
</p>

<ul>
<li><p> A <code>data.frame</code> with the exact column names <code>c("row", "col")</code>
</p>
</li>
<li><p> A <code>data.frame</code> with the number of rows and columns matching the plot matrix object provided.  Each cell will be tested for a &quot;truthy&quot; value to determine if the location should be kept.
</p>
</li></ul>

</dd>
</dl>
</td></tr>
<tr><td><code id="+2B2B.gg_+3A_rows">rows</code></td>
<td>
<p>numeric vector of the rows to be used. Will be used with  <code>cols</code> if <code>location</code> is <code>NULL</code></p>
</td></tr>
<tr><td><code id="+2B2B.gg_+3A_cols">cols</code></td>
<td>
<p>numeric vector of the cols to be used. Will be used with <code>rows</code> if <code>location</code> is <code>NULL</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the first object is an object of class <code><a href="#topic+ggmatrix">ggmatrix</a></code>, you can add
the following types of objects, and it will return a modified <span class="pkg">ggplot2</span>
object.
</p>

<ul>
<li> <p><code>theme</code>: update plot theme
</p>
</li>
<li> <p><code>scale</code>: replace current scale
</p>
</li>
<li> <p><code>coord</code>: override current coordinate system
</p>
</li></ul>

<p>The <code>+</code> operator completely replaces elements
with elements from e2.
</p>
<p><code>add_to_ggmatrix</code> gives you more control to modify
only some subplots.  This function may be replaced and/or removed in the future. <a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>See Also</h3>

<p><a href="ggplot2.html#topic+gg-add">ggplot2::+.gg</a> and <code><a href="ggplot2.html#topic+theme">ggplot2::theme()</a></code>
</p>
<p><code><a href="#topic+ggmatrix_location">ggmatrix_location</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive
data(tips)

pm &lt;- ggpairs(tips[, 2:4], ggplot2::aes(color = sex))
## change to black and white theme
pm + ggplot2::theme_bw()
## change to linedraw theme
p_(pm + ggplot2::theme_linedraw())
## change to custom theme
p_(pm + ggplot2::theme(panel.background = ggplot2::element_rect(fill = "lightblue")))
## add a list of information
extra &lt;- list(ggplot2::theme_bw(), ggplot2::labs(caption = "My caption!"))
p_(pm + extra)
## modify scale
p_(pm + scale_fill_brewer(type = "qual"))
## only first row
p_(add_to_ggmatrix(pm, scale_fill_brewer(type = "qual"), rows = 1:2))
## only second col
p_(add_to_ggmatrix(pm, scale_fill_brewer(type = "qual"), cols = 2:3))
## only to upper triangle of plot matrix
p_(add_to_ggmatrix(
  pm,
  scale_fill_brewer(type = "qual"),
  location = "upper"
))
</code></pre>

<hr>
<h2 id='add_and_overwrite_aes'>Add new aes</h2><span id='topic+add_and_overwrite_aes'></span>

<h3>Description</h3>

<p>Add new aesthetics to a previous aes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_and_overwrite_aes(current, new)
</code></pre>


<h3>Value</h3>

<p>aes_ output
</p>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(diamonds, package = "ggplot2")
diamonds.samp &lt;- diamonds[sample(1:dim(diamonds)[1], 1000), ]
pm &lt;- ggpairs(diamonds.samp,
  columns = 5:7,
  mapping = ggplot2::aes(color = color),
  upper = list(continuous = "cor", mapping = ggplot2::aes(color = clarity)),
  lower = list(continuous = "cor", mapping = ggplot2::aes(color = cut)),
  title = "Diamonds Sample"
)
str(pm)

</code></pre>

<hr>
<h2 id='add_ref_boxes'>Add reference boxes around each cell of the glyphmap.</h2><span id='topic+add_ref_boxes'></span>

<h3>Description</h3>

<p>Add reference boxes around each cell of the glyphmap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_ref_boxes(
  data,
  var_fill = NULL,
  color = "white",
  size = 0.5,
  fill = NA,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_ref_boxes_+3A_data">data</code></td>
<td>
<p>A glyphmap structure.</p>
</td></tr>
<tr><td><code id="add_ref_boxes_+3A_var_fill">var_fill</code></td>
<td>
<p>Variable name to use to set the fill color</p>
</td></tr>
<tr><td><code id="add_ref_boxes_+3A_color">color</code></td>
<td>
<p>Set the color to draw in, default is &quot;white&quot;</p>
</td></tr>
<tr><td><code id="add_ref_boxes_+3A_size">size</code></td>
<td>
<p>Set the line size, default is 0.5</p>
</td></tr>
<tr><td><code id="add_ref_boxes_+3A_fill">fill</code></td>
<td>
<p>fill value used if <code>var_fill</code> is <code>NULL</code></p>
</td></tr>
<tr><td><code id="add_ref_boxes_+3A_...">...</code></td>
<td>
<p>other arguments passed onto <code><a href="ggplot2.html#topic+geom_tile">ggplot2::geom_rect()</a></code></p>
</td></tr>
</table>

<hr>
<h2 id='add_ref_lines'>Add reference lines for each cell of the glyphmap.</h2><span id='topic+add_ref_lines'></span>

<h3>Description</h3>

<p>Add reference lines for each cell of the glyphmap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_ref_lines(data, color = "white", size = 1.5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_ref_lines_+3A_data">data</code></td>
<td>
<p>A glyphmap structure.</p>
</td></tr>
<tr><td><code id="add_ref_lines_+3A_color">color</code></td>
<td>
<p>Set the color to draw in, default is &quot;white&quot;</p>
</td></tr>
<tr><td><code id="add_ref_lines_+3A_size">size</code></td>
<td>
<p>Set the line size, default is 1.5</p>
</td></tr>
<tr><td><code id="add_ref_lines_+3A_...">...</code></td>
<td>
<p>other arguments passed onto <code><a href="ggplot2.html#topic+geom_path">ggplot2::geom_line()</a></code></p>
</td></tr>
</table>

<hr>
<h2 id='australia_PISA2012'>Programme for International Student Assessment (PISA) 2012 Data for Australia</h2><span id='topic+australia_PISA2012'></span>

<h3>Description</h3>

<p>About PISA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(australia_PISA2012)
</code></pre>


<h3>Format</h3>

<p>A data frame with 8247 rows and 32 variables
</p>


<h3>Details</h3>

<p>The Programme for International Student Assessment (PISA) is a triennial international survey which aims to evaluate education systems worldwide by testing the skills and knowledge of 15-year-old students. To date, students representing more than 70 economies have participated in the assessment.
</p>
<p>While 65 economies took part in the 2012 study, this data set only contains information from the country of Australia.
</p>

<ul>
<li><p> gender     : Factor w/ 2 levels &quot;female&quot;,&quot;male&quot;: 1 1 2 2 2 1 1 1 2 1 ...
</p>
</li>
<li><p> age        : Factor w/ 4 levels &quot;4&quot;,&quot;5&quot;,&quot;6&quot;,&quot;7&quot;: 2 2 2 4 3 1 2 2 2 2 ...
</p>
</li>
<li><p> homework   : num  5 5 9 3 2 3 4 3 5 1 ...
</p>
</li>
<li><p> desk       : num  1 0 1 1 1 1 1 1 1 1 ...
</p>
</li>
<li><p> room       : num  1 1 1 1 1 1 1 1 1 1 ...
</p>
</li>
<li><p> study      : num  1 1 1 1 1 1 1 1 1 1 ...
</p>
</li>
<li><p> computer   : num  1 1 1 1 1 1 1 1 1 1 ...
</p>
</li>
<li><p> software   : num  1 1 1 1 1 1 1 1 1 1 ...
</p>
</li>
<li><p> internet   : num  1 1 1 1 1 1 1 1 1 1 ...
</p>
</li>
<li><p> literature : num  0 0 1 0 1 1 1 1 1 0 ...
</p>
</li>
<li><p> poetry     : num  0 0 1 0 1 1 0 1 1 1 ...
</p>
</li>
<li><p> art        : num  1 0 1 0 1 1 0 1 1 1 ...
</p>
</li>
<li><p> textbook   : num  1 1 1 1 1 0 1 1 1 1 ...
</p>
</li>
<li><p> dictionary : num  1 1 1 1 1 1 1 1 1 1 ...
</p>
</li>
<li><p> dishwasher : num  1 1 1 1 0 1 1 1 1 1 ...
</p>
</li>
<li><p> PV1MATH    : num  562 565 602 520 613 ...
</p>
</li>
<li><p> PV2MATH    : num  569 557 594 507 567 ...
</p>
</li>
<li><p> PV3MATH    : num  555 553 552 501 585 ...
</p>
</li>
<li><p> PV4MATH    : num  579 538 526 521 596 ...
</p>
</li>
<li><p> PV5MATH    : num  548 573 619 547 603 ...
</p>
</li>
<li><p> PV1READ    : num  582 617 650 554 605 ...
</p>
</li>
<li><p> PV2READ    : num  571 572 608 560 557 ...
</p>
</li>
<li><p> PV3READ    : num  602 560 594 517 627 ...
</p>
</li>
<li><p> PV4READ    : num  572 564 575 564 597 ...
</p>
</li>
<li><p> PV5READ    : num  585 565 620 572 598 ...
</p>
</li>
<li><p> PV1SCIE    : num  583 627 668 574 639 ...
</p>
</li>
<li><p> PV2SCIE    : num  579 600 665 612 635 ...
</p>
</li>
<li><p> PV3SCIE    : num  593 574 620 571 666 ...
</p>
</li>
<li><p> PV4SCIE    : num  567 582 592 598 700 ...
</p>
</li>
<li><p> PV5SCIE    : num  587 625 656 662 670 ...
</p>
</li>
<li><p> SENWGT_STU : num  0.133 0.133 0.141 0.141 0.141 ...
</p>
</li>
<li><p> possessions: num  10 8 12 9 11 11 10 12 12 11 ...
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://www.oecd.org/pisa/pisaproducts/database-cbapisa2012.htm">https://www.oecd.org/pisa/pisaproducts/database-cbapisa2012.htm</a>
</p>

<hr>
<h2 id='baseball'>Yearly batting records for all major league baseball players</h2><span id='topic+baseball'></span>

<h3>Description</h3>

<p>This data frame contains batting statistics for a subset of players
collected from <a href="http://www.baseball-databank.org/">http://www.baseball-databank.org/</a>. There are a total
of 21,699 records, covering 1,228 players from 1871 to 2007. Only players
with more 15 seasons of play are included.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>baseball
</code></pre>


<h3>Format</h3>

<p>A 21699 x 22 data frame
</p>


<h3>Variables</h3>

<p>Variables:
</p>

<ul>
<li><p> id, unique player id
</p>
</li>
<li><p> year, year of data
</p>
</li>
<li><p> stint
</p>
</li>
<li><p> team, team played for
</p>
</li>
<li><p> lg, league
</p>
</li>
<li><p> g, number of games
</p>
</li>
<li><p> ab, number of times at bat
</p>
</li>
<li><p> r, number of runs
</p>
</li>
<li><p> h, hits, times reached base because of a batted, fair ball without
error by the defense
</p>
</li>
<li><p> X2b, hits on which the batter reached second base safely
</p>
</li>
<li><p> X3b, hits on which the batter reached third base safely
</p>
</li>
<li><p> hr, number of home runs
</p>
</li>
<li><p> rbi, runs batted in
</p>
</li>
<li><p> sb, stolen bases
</p>
</li>
<li><p> cs, caught stealing
</p>
</li>
<li><p> bb, base on balls (walk)
</p>
</li>
<li><p> so, strike outs
</p>
</li>
<li><p> ibb, intentional base on balls
</p>
</li>
<li><p> hbp, hits by pitch
</p>
</li>
<li><p> sh, sacrifice hits
</p>
</li>
<li><p> sf, sacrifice flies
</p>
</li>
<li><p> gidp, ground into double play
</p>
</li></ul>



<h3>References</h3>

<p><a href="http://www.baseball-databank.org/">http://www.baseball-databank.org/</a>
</p>

<hr>
<h2 id='brew_colors'>RColorBrewer Set1 colors</h2><span id='topic+brew_colors'></span>

<h3>Description</h3>

<p>RColorBrewer Set1 colors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brew_colors(col)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="brew_colors_+3A_col">col</code></td>
<td>
<p>standard color name used to retrieve hex color value</p>
</td></tr>
</table>

<hr>
<h2 id='broomify'>Broomify a model</h2><span id='topic+broomify'></span>

<h3>Description</h3>

<p>broom::augment a model and add broom::glance and broom::tidy output as attributes. X and Y variables are also added.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>broomify(model, lmStars = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="broomify_+3A_model">model</code></td>
<td>
<p>model to be sent to <code><a href="broom.html#topic+reexports">broom::augment()</a></code>, <code><a href="broom.html#topic+reexports">broom::glance()</a></code>, and <code><a href="broom.html#topic+reexports">broom::tidy()</a></code></p>
</td></tr>
<tr><td><code id="broomify_+3A_lmstars">lmStars</code></td>
<td>
<p>boolean that determines if stars are added to labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>broom::augmented data frame with the broom::glance data.frame and broom::tidy data.frame as 'broom_glance' and 'broom_tidy' attributes respectively.  <code>var_x</code> and <code>var_y</code> variables are also added as attributes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mtcars)
model &lt;- stats::lm(mpg ~ wt + qsec + am, data = mtcars)
broomified_model &lt;- broomify(model)
str(broomified_model)
</code></pre>

<hr>
<h2 id='column_is_character'>Get vector of variable types from data frame</h2><span id='topic+column_is_character'></span><span id='topic+column_is_factor'></span>

<h3>Description</h3>

<p>Get vector of variable types from data frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>column_is_character(df)

column_is_factor(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="column_is_character_+3A_df">df</code></td>
<td>
<p>data frame to extract variable types from</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector with variable types, with names corresponding to
the variable names from df
</p>


<h3>Author(s)</h3>

<p>Jason Crowley
</p>

<hr>
<h2 id='eval_data_col'>Evaluate data column</h2><span id='topic+eval_data_col'></span>

<h3>Description</h3>

<p>Evaluate data column
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eval_data_col(data, aes_col)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eval_data_col_+3A_data">data</code></td>
<td>
<p>data set to evaluate the data with</p>
</td></tr>
<tr><td><code id="eval_data_col_+3A_aes_col">aes_col</code></td>
<td>
<p>Single value from an <code>ggplot2::<a href="ggplot2.html#topic+aes">aes</a>(...)</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Aes mapping with the x and y values switched
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mapping &lt;- ggplot2::aes(Petal.Length)
eval_data_col(iris, mapping$x)
</code></pre>

<hr>
<h2 id='find_plot_type'>Find plot types</h2><span id='topic+find_plot_type'></span>

<h3>Description</h3>

<p>Retrieves the type of plot for the specific columns
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_plot_type(col1Name, col2Name, type1, type2, isAllNa, allowDiag)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_plot_type_+3A_col1name">col1Name</code></td>
<td>
<p>x column name</p>
</td></tr>
<tr><td><code id="find_plot_type_+3A_col2name">col2Name</code></td>
<td>
<p>y column name</p>
</td></tr>
<tr><td><code id="find_plot_type_+3A_type1">type1</code></td>
<td>
<p>x column type</p>
</td></tr>
<tr><td><code id="find_plot_type_+3A_type2">type2</code></td>
<td>
<p>y column type</p>
</td></tr>
<tr><td><code id="find_plot_type_+3A_isallna">isAllNa</code></td>
<td>
<p>is.na(data)</p>
</td></tr>
<tr><td><code id="find_plot_type_+3A_allowdiag">allowDiag</code></td>
<td>
<p>allow for diag values to be returned</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>

<hr>
<h2 id='flea'>Historical data used for classification examples.</h2><span id='topic+flea'></span>

<h3>Description</h3>

<p>This data contains physical measurements on three species of flea beetles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(flea)
</code></pre>


<h3>Format</h3>

<p>A data frame with 74 rows and 7 variables
</p>


<h3>Details</h3>


<ul>
<li><p> species Ch. concinna, Ch. heptapotamica, Ch. heikertingeri
</p>
</li>
<li><p> tars1 width of the first joint of the first tarsus in microns
</p>
</li>
<li><p> tars2 width of the second joint of the first tarsus in microns
</p>
</li>
<li><p> head the maximal width of the head between the external edges of the eyes in 0.01 mm
</p>
</li>
<li><p> aede1 the maximal width of the aedeagus in the fore-part in microns
</p>
</li>
<li><p> aede2 the front angle of the aedeagus (1 unit = 7.5 degrees)
</p>
</li>
<li><p> aede3 the aedeagus width from the side in microns
</p>
</li></ul>



<h3>References</h3>

<p>Lubischew, A. A. (1962), On the Use of Discriminant Functions in
Taxonomy, Biometrics 18:455-477.
</p>

<hr>
<h2 id='fn_switch'>Function switch</h2><span id='topic+fn_switch'></span>

<h3>Description</h3>

<p>Function that allows you to call different functions based upon an aesthetic variable value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fn_switch(types, mapping_val = "y")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fn_switch_+3A_types">types</code></td>
<td>
<p>list of functions that follow the <code><a href="#topic+ggmatrix">ggmatrix</a></code> function standard: <code>function(data, mapping, ...){ #make ggplot2 object }</code>.  One key should be a 'default' key for a default switch case.</p>
</td></tr>
<tr><td><code id="fn_switch_+3A_mapping_val">mapping_val</code></td>
<td>
<p>mapping value to switch on.  Defaults to the 'y' variable of the aesthetics list.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>ggnostic_continuous_fn &lt;- fn_switch(list(
  default = ggally_points,
  .fitted = ggally_points,
  .se.fit = ggally_nostic_se_fit,
  .resid = ggally_nostic_resid,
  .hat = ggally_nostic_hat,
  .sigma = ggally_nostic_sigma,
  .cooksd = ggally_nostic_cooksd,
  .std.resid = ggally_nostic_std_resid
))

ggnostic_combo_fn &lt;- fn_switch(list(
  default = ggally_box_no_facet,
  fitted = ggally_box_no_facet,
  .se.fit = ggally_nostic_se_fit,
  .resid = ggally_nostic_resid,
  .hat = ggally_nostic_hat,
  .sigma = ggally_nostic_sigma,
  .cooksd = ggally_nostic_cooksd,
  .std.resid = ggally_nostic_std_resid
))
</code></pre>

<hr>
<h2 id='get_x_axis_labels'>Get x axis labels</h2><span id='topic+get_x_axis_labels'></span>

<h3>Description</h3>

<p>Retrieves x axis labels from the plot object directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_x_axis_labels(p, xRange)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_x_axis_labels_+3A_p">p</code></td>
<td>
<p>plot object</p>
</td></tr>
<tr><td><code id="get_x_axis_labels_+3A_xrange">xRange</code></td>
<td>
<p>range of x values</p>
</td></tr>
</table>

<hr>
<h2 id='getPlot'>Subset a <code><a href="#topic+ggmatrix">ggmatrix</a></code> object</h2><span id='topic+getPlot'></span><span id='topic++5B.ggmatrix'></span>

<h3>Description</h3>

<p>Retrieves the ggplot object at the desired location.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPlot(pm, i, j)

## S3 method for class 'ggmatrix'
pm[i, j, ...]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPlot_+3A_pm">pm</code></td>
<td>
<p><code><a href="#topic+ggmatrix">ggmatrix</a></code> object to select from</p>
</td></tr>
<tr><td><code id="getPlot_+3A_i">i</code></td>
<td>
<p>row from the top</p>
</td></tr>
<tr><td><code id="getPlot_+3A_j">j</code></td>
<td>
<p>column from the left</p>
</td></tr>
<tr><td><code id="getPlot_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>See Also</h3>

<p><code><a href="#topic+putPlot">putPlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
plotMatrix2 &lt;- ggpairs(tips[, 3:2], upper = list(combo = "denstrip"))
p_(plotMatrix2[1, 2])
</code></pre>

<hr>
<h2 id='ggally_autopoint'>Scatterplot for continuous and categorical variables</h2><span id='topic+ggally_autopoint'></span><span id='topic+ggally_autopointDiag'></span>

<h3>Description</h3>

<p>Make scatterplots compatible with both continuous and categorical variables
using <code><a href="ggforce.html#topic+geom_autopoint">geom_autopoint</a></code> from package <span class="pkg">ggforce</span>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_autopoint(data, mapping, ...)

ggally_autopointDiag(data, mapping, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_autopoint_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_autopoint_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_autopoint_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="ggforce.html#topic+geom_autopoint">geom_autopoint</a>(...)</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Joseph Larmarange
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_autopoint(tips, mapping = aes(x = tip, y = total_bill)))
p_(ggally_autopoint(tips, mapping = aes(x = tip, y = sex)))
p_(ggally_autopoint(tips, mapping = aes(x = smoker, y = sex)))
p_(ggally_autopoint(tips, mapping = aes(x = smoker, y = sex, color = day)))
p_(ggally_autopoint(tips, mapping = aes(x = smoker, y = sex), size = 8))
p_(ggally_autopoint(tips, mapping = aes(x = smoker, y = sex), alpha = .9))

p_(ggpairs(
  tips,
  mapping = aes(colour = sex),
  upper = list(discrete = "autopoint", combo = "autopoint", continuous = "autopoint"),
  diag = list(discrete = "autopointDiag", continuous = "autopointDiag")
))
</code></pre>

<hr>
<h2 id='ggally_barDiag'>Bar plot</h2><span id='topic+ggally_barDiag'></span>

<h3>Description</h3>

<p>Displays a bar plot for the diagonal of a <code><a href="#topic+ggpairs">ggpairs</a></code> plot matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_barDiag(data, mapping, ..., rescale = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_barDiag_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_barDiag_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_barDiag_+3A_...">...</code></td>
<td>
<p>other arguments are sent to geom_bar</p>
</td></tr>
<tr><td><code id="ggally_barDiag_+3A_rescale">rescale</code></td>
<td>
<p>boolean to decide whether or not to rescale the count output. Only applies to numeric data</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_barDiag(tips, mapping = ggplot2::aes(x = day)))
p_(ggally_barDiag(tips, mapping = ggplot2::aes(x = tip), binwidth = 0.25))
</code></pre>

<hr>
<h2 id='ggally_blank'>Blank plot</h2><span id='topic+ggally_blank'></span><span id='topic+ggally_blankDiag'></span>

<h3>Description</h3>

<p>Draws nothing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_blank(...)

ggally_blankDiag(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_blank_+3A_...">...</code></td>
<td>
<p>other arguments ignored</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Makes a &quot;blank&quot; ggplot object that will only draw white space
</p>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+element">ggplot2::element_blank()</a></code>
</p>

<hr>
<h2 id='ggally_box'>Box plot</h2><span id='topic+ggally_box'></span><span id='topic+ggally_box_no_facet'></span>

<h3>Description</h3>

<p>Make a box plot with a given data set. <code>ggally_box_no_facet</code> will be a single panel plot, while <code>ggally_box</code> will be a faceted plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_box(data, mapping, ...)

ggally_box_no_facet(data, mapping, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_box_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_box_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_box_+3A_...">...</code></td>
<td>
<p>other arguments being supplied to geom_boxplot</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_box(tips, mapping = ggplot2::aes(x = total_bill, y = sex)))
p_(ggally_box(
  tips,
  mapping        = ggplot2::aes(sex, total_bill, color = sex),
  outlier.colour = "red",
  outlier.shape  = 13,
  outlier.size   = 8
))
</code></pre>

<hr>
<h2 id='ggally_colbar'>Column and row bar plots</h2><span id='topic+ggally_colbar'></span><span id='topic+ggally_rowbar'></span>

<h3>Description</h3>

<p>Plot column or row percentage using bar plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_colbar(
  data,
  mapping,
  label_format = scales::label_percent(accuracy = 0.1),
  ...,
  remove_background = FALSE,
  remove_percentage_axis = FALSE,
  reverse_fill_levels = FALSE,
  geom_bar_args = NULL
)

ggally_rowbar(
  data,
  mapping,
  label_format = scales::label_percent(accuracy = 0.1),
  ...,
  remove_background = FALSE,
  remove_percentage_axis = FALSE,
  reverse_fill_levels = TRUE,
  geom_bar_args = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_colbar_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_colbar_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_colbar_+3A_label_format">label_format</code></td>
<td>
<p>formatter function for displaying proportions, not taken into account if a label aesthetic is provided in <code>mapping</code></p>
</td></tr>
<tr><td><code id="ggally_colbar_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="ggplot2.html#topic+geom_text">geom_text</a>(...)</code></p>
</td></tr>
<tr><td><code id="ggally_colbar_+3A_remove_background">remove_background</code></td>
<td>
<p>should the <code>panel.background</code> be removed?</p>
</td></tr>
<tr><td><code id="ggally_colbar_+3A_remove_percentage_axis">remove_percentage_axis</code></td>
<td>
<p>should percentage axis be removed? Removes the y-axis for <code>ggally_colbar()</code> and x-axis for <code>ggally_rowbar()</code></p>
</td></tr>
<tr><td><code id="ggally_colbar_+3A_reverse_fill_levels">reverse_fill_levels</code></td>
<td>
<p>should the levels of the fill variable be reversed?</p>
</td></tr>
<tr><td><code id="ggally_colbar_+3A_geom_bar_args">geom_bar_args</code></td>
<td>
<p>other arguments passed to <code><a href="ggplot2.html#topic+geom_bar">geom_bar</a>(...)</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Joseph Larmarange
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_colbar(tips, mapping = aes(x = smoker, y = sex)))
p_(ggally_rowbar(tips, mapping = aes(x = smoker, y = sex)))

# change labels' size
p_(ggally_colbar(tips, mapping = aes(x = smoker, y = sex), size = 8))

# change labels' colour and use bold
p_(ggally_colbar(tips,
  mapping = aes(x = smoker, y = sex),
  colour = "white", fontface = "bold"
))

# display number of observations instead of proportions
p_(ggally_colbar(tips, mapping = aes(x = smoker, y = sex, label = after_stat(count))))

# custom bar width
p_(ggally_colbar(tips, mapping = aes(x = smoker, y = sex), geom_bar_args = list(width = .5)))

# change format of labels
p_(ggally_colbar(tips,
  mapping = aes(x = smoker, y = sex),
  label_format = scales::label_percent(accuracy = .01, decimal.mark = ",")
))

p_(ggduo(
  data = as.data.frame(Titanic),
  mapping = aes(weight = Freq),
  columnsX = "Survived",
  columnsY = c("Sex", "Class", "Age"),
  types = list(discrete = "rowbar"),
  legend = 1
))
</code></pre>

<hr>
<h2 id='ggally_cor'>Correlation value plot</h2><span id='topic+ggally_cor'></span>

<h3>Description</h3>

<p>Estimate correlation from the given data. If a color variable is supplied, the correlation will also be calculated per group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_cor(
  data,
  mapping,
  ...,
  stars = TRUE,
  method = "pearson",
  use = "complete.obs",
  display_grid = FALSE,
  digits = 3,
  title_args = list(...),
  group_args = list(...),
  justify_labels = "right",
  align_percent = 0.5,
  title = "Corr",
  alignPercent = warning("deprecated. Use `align_percent`"),
  displayGrid = warning("deprecated. Use `display_grid`")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_cor_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_...">...</code></td>
<td>
<p>other arguments being supplied to <code><a href="ggplot2.html#topic+geom_text">geom_text</a>()</code> for the title and groups</p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_stars">stars</code></td>
<td>
<p>logical value which determines if the significance stars should be displayed.  Given the <code><a href="stats.html#topic+cor.test">cor.test</a></code> p-values, display </p>

<dl>
<dt><code>"***"</code></dt><dd><p>if the p-value is <code style="white-space: pre;">&#8288;&lt; 0.001&#8288;</code></p>
</dd>
<dt><code>"**"</code></dt><dd><p>if the p-value is <code style="white-space: pre;">&#8288;&lt; 0.01&#8288;</code></p>
</dd>
<dt><code>"*"</code></dt><dd><p>if the p-value is <code style="white-space: pre;">&#8288;&lt; 0.05&#8288;</code></p>
</dd>
<dt><code>"."</code></dt><dd><p>if the p-value is <code style="white-space: pre;">&#8288;&lt; 0.10&#8288;</code></p>
</dd>
<dt><code>""</code></dt><dd><p>otherwise</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="ggally_cor_+3A_method">method</code></td>
<td>
<p><code>method</code> supplied to cor function</p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_use">use</code></td>
<td>
<p><code>use</code> supplied to <code><a href="stats.html#topic+cor">cor</a></code> function</p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_display_grid">display_grid</code></td>
<td>
<p>if <code>TRUE</code>, display aligned panel grid lines. If <code>FALSE</code> (default), display a thin panel border.</p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_digits">digits</code></td>
<td>
<p>number of digits to be displayed after the decimal point. See <code><a href="base.html#topic+formatC">formatC</a></code> for how numbers are calculated.</p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_title_args">title_args</code></td>
<td>
<p>arguments being supplied to the title's <code><a href="ggplot2.html#topic+geom_text">geom_text</a>()</code></p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_group_args">group_args</code></td>
<td>
<p>arguments being supplied to the split-by-color group's <code><a href="ggplot2.html#topic+geom_text">geom_text</a>()</code></p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_justify_labels">justify_labels</code></td>
<td>
<p><code>justify</code> argument supplied when <code><a href="base.html#topic+format">format</a></code>ting the labels</p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_align_percent">align_percent</code></td>
<td>
<p>relative align position of the text. When <code>justify_labels = 0.5</code>, this should not be needed to be set.</p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_title">title</code></td>
<td>
<p>title text to be displayed</p>
</td></tr>
<tr><td><code id="ggally_cor_+3A_alignpercent">alignPercent</code>, <code id="ggally_cor_+3A_displaygrid">displayGrid</code></td>
<td>
<p>deprecated. Please use their snake-case counterparts.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggally_statistic">ggally_statistic</a></code>, <code><a href="#topic+ggally_cor_v1_5">ggally_cor_v1_5</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_cor(tips, mapping = ggplot2::aes(total_bill, tip)))
# display with grid
p_(ggally_cor(
  tips,
  mapping = ggplot2::aes(total_bill, tip),
  display_grid = TRUE
))
# change text attributes
p_(ggally_cor(
  tips,
  mapping = ggplot2::aes(x = total_bill, y = tip),
  size = 15,
  colour = I("red"),
  title = "Correlation"
))
# split by a variable
p_(ggally_cor(
  tips,
  mapping = ggplot2::aes(total_bill, tip, color = sex),
  size = 5
))
</code></pre>

<hr>
<h2 id='ggally_cor_v1_5'>Correlation value plot</h2><span id='topic+ggally_cor_v1_5'></span>

<h3>Description</h3>

<p>(Deprecated. See <code><a href="#topic+ggally_cor">ggally_cor</a></code>.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_cor_v1_5(
  data,
  mapping,
  alignPercent = 0.6,
  method = "pearson",
  use = "complete.obs",
  corAlignPercent = NULL,
  corMethod = NULL,
  corUse = NULL,
  displayGrid = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_cor_v1_5_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_cor_v1_5_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_cor_v1_5_+3A_alignpercent">alignPercent</code></td>
<td>
<p>right align position of numbers. Default is 60 percent across the horizontal</p>
</td></tr>
<tr><td><code id="ggally_cor_v1_5_+3A_method">method</code></td>
<td>
<p><code>method</code> supplied to cor function</p>
</td></tr>
<tr><td><code id="ggally_cor_v1_5_+3A_use">use</code></td>
<td>
<p><code>use</code> supplied to cor function</p>
</td></tr>
<tr><td><code id="ggally_cor_v1_5_+3A_coralignpercent">corAlignPercent</code></td>
<td>
<p>deprecated. Use parameter <code>alignPercent</code></p>
</td></tr>
<tr><td><code id="ggally_cor_v1_5_+3A_cormethod">corMethod</code></td>
<td>
<p>deprecated. Use parameter <code>method</code></p>
</td></tr>
<tr><td><code id="ggally_cor_v1_5_+3A_coruse">corUse</code></td>
<td>
<p>deprecated. Use parameter <code>use</code></p>
</td></tr>
<tr><td><code id="ggally_cor_v1_5_+3A_displaygrid">displayGrid</code></td>
<td>
<p>if TRUE, display aligned panel gridlines</p>
</td></tr>
<tr><td><code id="ggally_cor_v1_5_+3A_...">...</code></td>
<td>
<p>other arguments being supplied to geom_text</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Estimate correlation from the given data.
</p>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggally_cor">ggally_cor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_cor_v1_5(tips, mapping = ggplot2::aes(total_bill, tip)))

# display with no grid
p_(ggally_cor_v1_5(
  tips,
  mapping = ggplot2::aes(total_bill, tip),
  displayGrid = FALSE
))

# change text attributes
p_(ggally_cor_v1_5(
  tips,
  mapping = ggplot2::aes(x = total_bill, y = tip),
  size = 15,
  colour = I("red")
))

# split by a variable
p_(ggally_cor_v1_5(
  tips,
  mapping = ggplot2::aes(total_bill, tip, color = sex),
  size = 5
))
</code></pre>

<hr>
<h2 id='ggally_count'>Display counts of observations</h2><span id='topic+ggally_count'></span><span id='topic+stat_ggally_count'></span><span id='topic+StatGGallyCount'></span><span id='topic+ggally_countDiag'></span>

<h3>Description</h3>

<p>Plot the number of observations by using rectangles
with proportional areas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_count(data, mapping, ...)

ggally_countDiag(data, mapping, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_count_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_count_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_count_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="ggplot2.html#topic+geom_tile">geom_tile</a>(...)</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can adjust the size of rectangles with the <code>x.width</code> argument.
</p>


<h3>Author(s)</h3>

<p>Joseph Larmarange
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_count(tips, mapping = ggplot2::aes(x = smoker, y = sex)))
p_(ggally_count(tips, mapping = ggplot2::aes(x = smoker, y = sex, fill = day)))

p_(ggally_count(
  as.data.frame(Titanic),
  mapping = ggplot2::aes(x = Class, y = Survived, weight = Freq)
))
p_(ggally_count(
  as.data.frame(Titanic),
  mapping = ggplot2::aes(x = Class, y = Survived, weight = Freq),
  x.width = 0.5
))
# Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

p_(ggally_countDiag(tips, mapping = ggplot2::aes(x = smoker)))
p_(ggally_countDiag(tips, mapping = ggplot2::aes(x = smoker, fill = sex)))
</code></pre>

<hr>
<h2 id='ggally_cross'>Plots the number of observations</h2><span id='topic+ggally_cross'></span>

<h3>Description</h3>

<p>Plot the number of observations by using square points
with proportional areas. Could be filled according to chi-squared
statistics computed by <code><a href="#topic+stat_cross">stat_cross()</a></code>. Labels could also
be added (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_cross(data, mapping, ..., scale_max_size = 20, geom_text_args = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_cross_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_cross_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_cross_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="ggplot2.html#topic+geom_point">ggplot2::geom_point()</a></code></p>
</td></tr>
<tr><td><code id="ggally_cross_+3A_scale_max_size">scale_max_size</code></td>
<td>
<p><code>max_size</code> argument supplied to <code><a href="ggplot2.html#topic+scale_size">ggplot2::scale_size_area()</a></code></p>
</td></tr>
<tr><td><code id="ggally_cross_+3A_geom_text_args">geom_text_args</code></td>
<td>
<p>other arguments passed to <code><a href="ggplot2.html#topic+geom_text">ggplot2::geom_text()</a></code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Joseph Larmarange
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_cross(tips, mapping = aes(x = smoker, y = sex)))
p_(ggally_cross(tips, mapping = aes(x = day, y = time)))

# Custom max size
p_(ggally_cross(tips, mapping = aes(x = smoker, y = sex)) +
  scale_size_area(max_size = 40))

# Custom fill
p_(ggally_cross(tips, mapping = aes(x = smoker, y = sex), fill = "red"))

# Custom shape
p_(ggally_cross(tips, mapping = aes(x = smoker, y = sex), shape = 21))

# Fill squares according to standardized residuals
d &lt;- as.data.frame(Titanic)
p_(ggally_cross(
  d,
  mapping = aes(x = Class, y = Survived, weight = Freq, fill = after_stat(std.resid))
) +
  scale_fill_steps2(breaks = c(-3, -2, 2, 3), show.limits = TRUE))

# Add labels
p_(ggally_cross(
  tips,
  mapping = aes(
    x = smoker, y = sex, colour = smoker,
    label = scales::percent(after_stat(prop))
  )
))

# Customize labels' appearance and same size for all squares
p_(ggally_cross(
  tips,
  mapping = aes(
    x = smoker, y = sex,
    size = NULL, # do not map size to a variable
    label = scales::percent(after_stat(prop))
  ),
  size = 40, # fix value for points size
  fill = "darkblue",
  geom_text_args = list(colour = "white", fontface = "bold", size = 6)
))
</code></pre>

<hr>
<h2 id='ggally_crosstable'>Display a cross-tabulated table</h2><span id='topic+ggally_crosstable'></span>

<h3>Description</h3>

<p><code>ggally_crosstable</code> is a variation of <code><a href="#topic+ggally_table">ggally_table</a></code> with few modifications: (i) table cells are drawn; (ii) x and y axis are not expanded (and therefore are not aligned with other <code>ggally_*</code> plots); (iii) content and fill of cells can be easily controlled with dedicated arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_crosstable(
  data,
  mapping,
  cells = c("observed", "prop", "row.prop", "col.prop", "expected", "resid", "std.resid"),
  fill = c("none", "std.resid", "resid"),
  ...,
  geom_tile_args = list(colour = "grey50")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_crosstable_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_crosstable_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_crosstable_+3A_cells">cells</code></td>
<td>
<p>Which statistic should be displayed in table cells?</p>
</td></tr>
<tr><td><code id="ggally_crosstable_+3A_fill">fill</code></td>
<td>
<p>Which statistic should be used for filling table cells?</p>
</td></tr>
<tr><td><code id="ggally_crosstable_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="ggplot2.html#topic+geom_text">geom_text</a>(...)</code></p>
</td></tr>
<tr><td><code id="ggally_crosstable_+3A_geom_tile_args">geom_tile_args</code></td>
<td>
<p>other arguments passed to <code><a href="ggplot2.html#topic+geom_tile">geom_tile</a>(...)</code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)

# differences with ggally_table()
p_(ggally_table(tips, mapping = aes(x = day, y = time)))
p_(ggally_crosstable(tips, mapping = aes(x = day, y = time)))

# display column proportions
p_(ggally_crosstable(tips, mapping = aes(x = day, y = sex), cells = "col.prop"))

# display row proportions
p_(ggally_crosstable(tips, mapping = aes(x = day, y = sex), cells = "row.prop"))

# change size of text
p_(ggally_crosstable(tips, mapping = aes(x = day, y = sex), size = 8))

# fill cells with standardized residuals
p_(ggally_crosstable(tips, mapping = aes(x = day, y = sex), fill = "std.resid"))

# change scale for fill
p_(ggally_crosstable(tips, mapping = aes(x = day, y = sex), fill = "std.resid") +
  scale_fill_steps2(breaks = c(-2, 0, 2), show.limits = TRUE))
</code></pre>

<hr>
<h2 id='ggally_density'>Bivariate density plot</h2><span id='topic+ggally_density'></span>

<h3>Description</h3>

<p>Make a 2D density plot from a given data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_density(data, mapping, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_density_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_density_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_density_+3A_...">...</code></td>
<td>
<p>parameters sent to either stat_density2d or geom_density2d</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The aesthetic &quot;fill&quot; determines whether or not <code>stat_density2d</code> (filled) or <code>geom_density2d</code> (lines) is used.
</p>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_density(tips, mapping = ggplot2::aes(x = total_bill, y = tip)))
p_(ggally_density(
  tips,
  mapping = ggplot2::aes(total_bill, tip, fill = after_stat(level))
))
p_(ggally_density(
  tips,
  mapping = ggplot2::aes(total_bill, tip, fill = after_stat(level))
) + ggplot2::scale_fill_gradient(breaks = c(0.05, 0.1, 0.15, 0.2)))
</code></pre>

<hr>
<h2 id='ggally_densityDiag'>Univariate density plot</h2><span id='topic+ggally_densityDiag'></span>

<h3>Description</h3>

<p>Displays a density plot for the diagonal of a <code><a href="#topic+ggpairs">ggpairs</a></code> plot matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_densityDiag(data, mapping, ..., rescale = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_densityDiag_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_densityDiag_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used.</p>
</td></tr>
<tr><td><code id="ggally_densityDiag_+3A_...">...</code></td>
<td>
<p>other arguments sent to stat_density</p>
</td></tr>
<tr><td><code id="ggally_densityDiag_+3A_rescale">rescale</code></td>
<td>
<p>boolean to decide whether or not to rescale the count output</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_densityDiag(tips, mapping = ggplot2::aes(x = total_bill)))
p_(ggally_densityDiag(tips, mapping = ggplot2::aes(x = total_bill, color = day)))
</code></pre>

<hr>
<h2 id='ggally_denstrip'>Tile plot with facets</h2><span id='topic+ggally_denstrip'></span>

<h3>Description</h3>

<p>Displays a Tile Plot as densely as possible.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_denstrip(data, mapping, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_denstrip_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_denstrip_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_denstrip_+3A_...">...</code></td>
<td>
<p>other arguments being sent to stat_bin</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_denstrip(tips, mapping = ggplot2::aes(x = total_bill, y = sex)))
p_(ggally_denstrip(
  tips,
  mapping = ggplot2::aes(sex, tip), binwidth = 0.2
) + ggplot2::scale_fill_gradient(low = "grey80", high = "black"))
</code></pre>

<hr>
<h2 id='ggally_diagAxis'>Internal axis labels for ggpairs</h2><span id='topic+ggally_diagAxis'></span>

<h3>Description</h3>

<p>This function is used when <code>axisLabels == "internal"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_diagAxis(
  data,
  mapping,
  label = mapping$x,
  labelSize = 5,
  labelXPercent = 0.5,
  labelYPercent = 0.55,
  labelHJust = 0.5,
  labelVJust = 0.5,
  gridLabelSize = 4,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_diagAxis_+3A_data">data</code></td>
<td>
<p>dataset being plotted</p>
</td></tr>
<tr><td><code id="ggally_diagAxis_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used (x is the variable the plot will be made for)</p>
</td></tr>
<tr><td><code id="ggally_diagAxis_+3A_label">label</code></td>
<td>
<p>title to be displayed in the middle.  Defaults to <code>mapping$x</code></p>
</td></tr>
<tr><td><code id="ggally_diagAxis_+3A_labelsize">labelSize</code></td>
<td>
<p>size of variable label</p>
</td></tr>
<tr><td><code id="ggally_diagAxis_+3A_labelxpercent">labelXPercent</code></td>
<td>
<p>percent of horizontal range</p>
</td></tr>
<tr><td><code id="ggally_diagAxis_+3A_labelypercent">labelYPercent</code></td>
<td>
<p>percent of vertical range</p>
</td></tr>
<tr><td><code id="ggally_diagAxis_+3A_labelhjust">labelHJust</code></td>
<td>
<p>hjust supplied to label</p>
</td></tr>
<tr><td><code id="ggally_diagAxis_+3A_labelvjust">labelVJust</code></td>
<td>
<p>vjust supplied to label</p>
</td></tr>
<tr><td><code id="ggally_diagAxis_+3A_gridlabelsize">gridLabelSize</code></td>
<td>
<p>size of grid labels</p>
</td></tr>
<tr><td><code id="ggally_diagAxis_+3A_...">...</code></td>
<td>
<p>other arguments for geom_text</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jason Crowley and Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_diagAxis(tips, ggplot2::aes(x = tip)))
p_(ggally_diagAxis(tips, ggplot2::aes(x = sex)))
</code></pre>

<hr>
<h2 id='ggally_dot'>Grouped dot plot</h2><span id='topic+ggally_dot'></span><span id='topic+ggally_dot_no_facet'></span>

<h3>Description</h3>

<p>Add jittering with the box plot. <code>ggally_dot_no_facet</code> will be a single panel plot, while <code>ggally_dot</code> will be a faceted plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_dot(data, mapping, ...)

ggally_dot_no_facet(data, mapping, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_dot_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_dot_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_dot_+3A_...">...</code></td>
<td>
<p>other arguments being supplied to geom_jitter</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_dot(tips, mapping = ggplot2::aes(x = total_bill, y = sex)))
p_(ggally_dot(
  tips,
  mapping = ggplot2::aes(sex, total_bill, color = sex)
))
p_(ggally_dot(
  tips,
  mapping = ggplot2::aes(sex, total_bill, color = sex, shape = sex)
) + ggplot2::scale_shape(solid = FALSE))
</code></pre>

<hr>
<h2 id='ggally_dot_and_box'>Box and dot plot</h2><span id='topic+ggally_dot_and_box'></span>

<h3>Description</h3>

<p>Place box plots or dot plots on the graph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_dot_and_box(data, mapping, ..., boxPlot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_dot_and_box_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_dot_and_box_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_dot_and_box_+3A_...">...</code></td>
<td>
<p>parameters passed to either geom_jitter or geom_boxplot</p>
</td></tr>
<tr><td><code id="ggally_dot_and_box_+3A_boxplot">boxPlot</code></td>
<td>
<p>boolean to decide to plot either box plots (TRUE) or dot plots (FALSE)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_dot_and_box(
  tips,
  mapping = ggplot2::aes(x = total_bill, y = sex, color = sex),
  boxPlot = TRUE
))
p_(ggally_dot_and_box(
  tips,
  mapping = ggplot2::aes(x = total_bill, y = sex, color = sex),
  boxPlot = FALSE
))
</code></pre>

<hr>
<h2 id='ggally_facetbar'>Faceted bar plot</h2><span id='topic+ggally_facetbar'></span>

<h3>Description</h3>

<p>X variables are plotted using <code>geom_bar</code> and are faceted by the Y variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_facetbar(data, mapping, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_facetbar_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_facetbar_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_facetbar_+3A_...">...</code></td>
<td>
<p>other arguments are sent to geom_bar</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_facetbar(tips, ggplot2::aes(x = sex, y = smoker, fill = time)))
p_(ggally_facetbar(tips, ggplot2::aes(x = smoker, y = sex, fill = time)))
</code></pre>

<hr>
<h2 id='ggally_facetdensity'>Faceted density plot</h2><span id='topic+ggally_facetdensity'></span>

<h3>Description</h3>

<p>Make density plots by displaying subsets of the data in different panels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_facetdensity(data, mapping, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_facetdensity_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_facetdensity_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_facetdensity_+3A_...">...</code></td>
<td>
<p>other arguments being sent to stat_density</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_facetdensity(tips, mapping = ggplot2::aes(x = total_bill, y = sex)))
p_(ggally_facetdensity(
  tips,
  mapping = ggplot2::aes(sex, total_bill, color = sex)
))
</code></pre>

<hr>
<h2 id='ggally_facetdensitystrip'>Density or tiles plot with facets</h2><span id='topic+ggally_facetdensitystrip'></span>

<h3>Description</h3>

<p>Make tile plot or density plot as compact as possible.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_facetdensitystrip(data, mapping, ..., den_strip = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_facetdensitystrip_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_facetdensitystrip_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_facetdensitystrip_+3A_...">...</code></td>
<td>
<p>other arguments being sent to either geom_histogram or stat_density</p>
</td></tr>
<tr><td><code id="ggally_facetdensitystrip_+3A_den_strip">den_strip</code></td>
<td>
<p>boolean to decide whether or not to plot a density strip(TRUE) or a facet density(FALSE) plot.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example(ggally_facetdensity)
example(ggally_denstrip)
</code></pre>

<hr>
<h2 id='ggally_facethist'>Faceted histogram</h2><span id='topic+ggally_facethist'></span>

<h3>Description</h3>

<p>Display subsetted histograms of the data in different panels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_facethist(data, mapping, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_facethist_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_facethist_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_facethist_+3A_...">...</code></td>
<td>
<p>parameters sent to stat_bin()</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_facethist(tips, mapping = ggplot2::aes(x = tip, y = sex)))
p_(ggally_facethist(tips, mapping = ggplot2::aes(x = tip, y = sex), binwidth = 0.1))
</code></pre>

<hr>
<h2 id='ggally_na'>NA plot</h2><span id='topic+ggally_na'></span><span id='topic+ggally_naDiag'></span>

<h3>Description</h3>

<p>Draws a large <code>NA</code> in the middle of the plotting area.  This plot is useful when all X or Y data is <code>NA</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_na(data = NULL, mapping = NULL, size = 10, color = "grey20", ...)

ggally_naDiag(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_na_+3A_data">data</code></td>
<td>
<p>ignored</p>
</td></tr>
<tr><td><code id="ggally_na_+3A_mapping">mapping</code></td>
<td>
<p>ignored</p>
</td></tr>
<tr><td><code id="ggally_na_+3A_size">size</code></td>
<td>
<p>size of the geom_text 'NA'</p>
</td></tr>
<tr><td><code id="ggally_na_+3A_color">color</code></td>
<td>
<p>color of the geom_text 'NA'</p>
</td></tr>
<tr><td><code id="ggally_na_+3A_...">...</code></td>
<td>
<p>other arguments sent to geom_text</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>

<hr>
<h2 id='ggally_nostic_cooksd'><code><a href="#topic+ggnostic">ggnostic</a></code> Cook's distance</h2><span id='topic+ggally_nostic_cooksd'></span>

<h3>Description</h3>

<p>A function to display <code><a href="stats.html#topic+influence.measures">stats::cooks.distance()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_nostic_cooksd(
  data,
  mapping,
  ...,
  linePosition = pf(0.5, length(attr(data, "var_x")), nrow(data) - length(attr(data,
    "var_x"))),
  lineColor = brew_colors("grey"),
  lineType = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_nostic_cooksd_+3A_data">data</code>, <code id="ggally_nostic_cooksd_+3A_mapping">mapping</code>, <code id="ggally_nostic_cooksd_+3A_...">...</code>, <code id="ggally_nostic_cooksd_+3A_linecolor">lineColor</code>, <code id="ggally_nostic_cooksd_+3A_linetype">lineType</code></td>
<td>
<p>parameters supplied to <code><a href="#topic+ggally_nostic_line">ggally_nostic_line</a></code></p>
</td></tr>
<tr><td><code id="ggally_nostic_cooksd_+3A_lineposition">linePosition</code></td>
<td>
<p>4 / n is the general cutoff point for Cook's Distance</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A line is added at <code class="reqn">F_{p,n-p}(0.5)</code> to display the general cutoff point for Cook's Distance.
</p>
<p>Reference: Michael H. Kutner, Christopher J. Nachtsheim, John Neter, and William Li. Applied linear statistical models. The McGraw-Hill / Irwin series operations and decision sciences. McGraw-Hill Irwin, 2005, p. 403
</p>


<h3>Value</h3>

<p><span class="pkg">ggplot2</span> plot object
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+influence.measures">stats::cooks.distance()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

dt &lt;- broomify(stats::lm(mpg ~ wt + qsec + am, data = mtcars))
p_(ggally_nostic_cooksd(dt, ggplot2::aes(wt, .cooksd)))
</code></pre>

<hr>
<h2 id='ggally_nostic_hat'><code><a href="#topic+ggnostic">ggnostic</a></code> leverage points</h2><span id='topic+ggally_nostic_hat'></span>

<h3>Description</h3>

<p>A function to display stats::influence's hat information against a given explanatory variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_nostic_hat(
  data,
  mapping,
  ...,
  linePosition = 2 * sum(eval_data_col(data, mapping$y))/nrow(data),
  lineColor = brew_colors("grey"),
  lineSize = 0.5,
  lineAlpha = 1,
  lineType = 2,
  avgLinePosition = sum(eval_data_col(data, mapping$y))/nrow(data),
  avgLineColor = brew_colors("grey"),
  avgLineSize = lineSize,
  avgLineAlpha = lineAlpha,
  avgLineType = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_nostic_hat_+3A_data">data</code>, <code id="ggally_nostic_hat_+3A_mapping">mapping</code>, <code id="ggally_nostic_hat_+3A_...">...</code></td>
<td>
<p>supplied directly to <code><a href="#topic+ggally_nostic_line">ggally_nostic_line</a></code></p>
</td></tr>
<tr><td><code id="ggally_nostic_hat_+3A_lineposition">linePosition</code>, <code id="ggally_nostic_hat_+3A_linecolor">lineColor</code>, <code id="ggally_nostic_hat_+3A_linesize">lineSize</code>, <code id="ggally_nostic_hat_+3A_linealpha">lineAlpha</code>, <code id="ggally_nostic_hat_+3A_linetype">lineType</code></td>
<td>
<p>parameters supplied to
<code><a href="ggplot2.html#topic+geom_path">ggplot2::geom_line()</a></code> for the cutoff line</p>
</td></tr>
<tr><td><code id="ggally_nostic_hat_+3A_avglineposition">avgLinePosition</code>, <code id="ggally_nostic_hat_+3A_avglinecolor">avgLineColor</code>, <code id="ggally_nostic_hat_+3A_avglinesize">avgLineSize</code>, <code id="ggally_nostic_hat_+3A_avglinealpha">avgLineAlpha</code>, <code id="ggally_nostic_hat_+3A_avglinetype">avgLineType</code></td>
<td>
<p>parameters supplied
to <code><a href="ggplot2.html#topic+geom_path">ggplot2::geom_line()</a></code> for the average line</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As stated in <code><a href="stats.html#topic+lm.influence">stats::influence()</a></code> documentation:
</p>
<p>hat: a vector containing the diagonal of the 'hat' matrix.
</p>
<p>The diagonal elements of the 'hat' matrix describe the influence each response value has on the fitted value for that same observation.
</p>
<p>A suggested &quot;cutoff&quot; line is added to the plot at a height of 2 * p / n and an expected line at a height of p / n.
If either <code>linePosition</code> or <code>avgLinePosition</code> is <code>NULL</code>, the respective line will not be drawn.
</p>


<h3>Value</h3>

<p><span class="pkg">ggplot2</span> plot object
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm.influence">stats::influence()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

dt &lt;- broomify(stats::lm(mpg ~ wt + qsec + am, data = mtcars))
p_(ggally_nostic_hat(dt, ggplot2::aes(wt, .hat)))
</code></pre>

<hr>
<h2 id='ggally_nostic_line'><code><a href="#topic+ggnostic">ggnostic</a></code> background line with geom</h2><span id='topic+ggally_nostic_line'></span>

<h3>Description</h3>

<p>If a non-null <code>linePosition</code> value is given, a line will be drawn before the given <code>continuous_geom</code> or <code>combo_geom</code> is added to the plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_nostic_line(
  data,
  mapping,
  ...,
  linePosition = NULL,
  lineColor = "red",
  lineSize = 0.5,
  lineAlpha = 1,
  lineType = 1,
  continuous_geom = ggplot2::geom_point,
  combo_geom = ggplot2::geom_boxplot,
  mapColorToFill = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_nostic_line_+3A_data">data</code>, <code id="ggally_nostic_line_+3A_mapping">mapping</code></td>
<td>
<p>supplied directly to <code><a href="ggplot2.html#topic+ggplot">ggplot2::ggplot()</a></code></p>
</td></tr>
<tr><td><code id="ggally_nostic_line_+3A_...">...</code></td>
<td>
<p>parameters supplied to <code>continuous_geom</code> or <code>combo_geom</code></p>
</td></tr>
<tr><td><code id="ggally_nostic_line_+3A_lineposition">linePosition</code>, <code id="ggally_nostic_line_+3A_linecolor">lineColor</code>, <code id="ggally_nostic_line_+3A_linesize">lineSize</code>, <code id="ggally_nostic_line_+3A_linealpha">lineAlpha</code>, <code id="ggally_nostic_line_+3A_linetype">lineType</code></td>
<td>
<p>parameters supplied to
<code><a href="ggplot2.html#topic+geom_path">ggplot2::geom_line()</a></code></p>
</td></tr>
<tr><td><code id="ggally_nostic_line_+3A_continuous_geom">continuous_geom</code></td>
<td>
<p><span class="pkg">ggplot2</span> geom that is executed after the line is (possibly)
added and if the x data is continuous</p>
</td></tr>
<tr><td><code id="ggally_nostic_line_+3A_combo_geom">combo_geom</code></td>
<td>
<p><span class="pkg">ggplot2</span> geom that is executed after the line is (possibly) added and
if the x data is discrete</p>
</td></tr>
<tr><td><code id="ggally_nostic_line_+3A_mapcolortofill">mapColorToFill</code></td>
<td>
<p>boolean to determine if combo plots should cut the color mapping to the fill mapping</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Functions with a color in their name have different default color behavior.
</p>


<h3>Value</h3>

<p><span class="pkg">ggplot2</span> plot object
</p>

<hr>
<h2 id='ggally_nostic_resid'><code><a href="#topic+ggnostic">ggnostic</a></code> residuals</h2><span id='topic+ggally_nostic_resid'></span>

<h3>Description</h3>

<p>If non-null <code>pVal</code> and <code>sigma</code> values are given, confidence interval lines will be added to the plot at the specified <code>pVal</code> percentiles of a N(0, sigma) distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_nostic_resid(
  data,
  mapping,
  ...,
  linePosition = 0,
  lineColor = brew_colors("grey"),
  lineSize = 0.5,
  lineAlpha = 1,
  lineType = 1,
  lineConfColor = brew_colors("grey"),
  lineConfSize = lineSize,
  lineConfAlpha = lineAlpha,
  lineConfType = 2,
  pVal = c(0.025, 0.975),
  sigma = attr(data, "broom_glance")$sigma,
  se = TRUE,
  method = "auto",
  formula = y ~ x
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_nostic_resid_+3A_data">data</code>, <code id="ggally_nostic_resid_+3A_mapping">mapping</code>, <code id="ggally_nostic_resid_+3A_...">...</code></td>
<td>
<p>parameters supplied to <code><a href="#topic+ggally_nostic_line">ggally_nostic_line</a></code></p>
</td></tr>
<tr><td><code id="ggally_nostic_resid_+3A_lineposition">linePosition</code>, <code id="ggally_nostic_resid_+3A_linecolor">lineColor</code>, <code id="ggally_nostic_resid_+3A_linesize">lineSize</code>, <code id="ggally_nostic_resid_+3A_linealpha">lineAlpha</code>, <code id="ggally_nostic_resid_+3A_linetype">lineType</code></td>
<td>
<p>parameters supplied to
<code><a href="ggplot2.html#topic+geom_path">ggplot2::geom_line()</a></code></p>
</td></tr>
<tr><td><code id="ggally_nostic_resid_+3A_lineconfcolor">lineConfColor</code>, <code id="ggally_nostic_resid_+3A_lineconfsize">lineConfSize</code>, <code id="ggally_nostic_resid_+3A_lineconfalpha">lineConfAlpha</code>, <code id="ggally_nostic_resid_+3A_lineconftype">lineConfType</code></td>
<td>
<p>parameters supplied to the
confidence interval lines</p>
</td></tr>
<tr><td><code id="ggally_nostic_resid_+3A_pval">pVal</code></td>
<td>
<p>percentiles of a N(0, sigma) distribution to be drawn</p>
</td></tr>
<tr><td><code id="ggally_nostic_resid_+3A_sigma">sigma</code></td>
<td>
<p>sigma value for the <code>pVal</code> percentiles</p>
</td></tr>
<tr><td><code id="ggally_nostic_resid_+3A_se">se</code></td>
<td>
<p>boolean to determine if the confidence intervals should be displayed</p>
</td></tr>
<tr><td><code id="ggally_nostic_resid_+3A_method">method</code>, <code id="ggally_nostic_resid_+3A_formula">formula</code></td>
<td>
<p>parameters supplied to <code><a href="ggplot2.html#topic+geom_smooth">ggplot2::geom_smooth()</a></code>.
Defaults to <code>"auto"</code> and <code>"y ~ x"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><span class="pkg">ggplot2</span> plot object
</p>


<h3>See Also</h3>

<p><code>stats::<a href="stats.html#topic+residuals">residuals</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

dt &lt;- broomify(stats::lm(mpg ~ wt + qsec + am, data = mtcars))
p_(ggally_nostic_resid(dt, ggplot2::aes(wt, .resid)))
</code></pre>

<hr>
<h2 id='ggally_nostic_se_fit'><code><a href="#topic+ggnostic">ggnostic</a></code> fitted value's standard error</h2><span id='topic+ggally_nostic_se_fit'></span>

<h3>Description</h3>

<p>A function to display <code>stats::<a href="stats.html#topic+predict">predict</a></code>'s standard errors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_nostic_se_fit(
  data,
  mapping,
  ...,
  lineColor = brew_colors("grey"),
  linePosition = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_nostic_se_fit_+3A_data">data</code>, <code id="ggally_nostic_se_fit_+3A_mapping">mapping</code>, <code id="ggally_nostic_se_fit_+3A_...">...</code>, <code id="ggally_nostic_se_fit_+3A_linecolor">lineColor</code></td>
<td>
<p>parameters supplied to <code><a href="#topic+ggally_nostic_line">ggally_nostic_line</a></code></p>
</td></tr>
<tr><td><code id="ggally_nostic_se_fit_+3A_lineposition">linePosition</code></td>
<td>
<p>base comparison for a perfect fit</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As stated in <code>stats::<a href="stats.html#topic+predict">predict</a></code> documentation:
</p>
<p>If the logical 'se.fit' is 'TRUE', standard errors of the predictions are calculated.  If the numeric argument 'scale' is set (with optional &rdquo;df'), it is used as the residual standard deviation in the computation of the standard errors, otherwise this is extracted from the model fit.
</p>
<p>Since the se.fit is <code>TRUE</code> and scale is unset by default, the standard errors are extracted from the model fit.
</p>
<p>A base line of 0 is added to give reference to a perfect fit.
</p>


<h3>Value</h3>

<p><span class="pkg">ggplot2</span> plot object
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm.influence">stats::influence()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

dt &lt;- broomify(stats::lm(mpg ~ wt + qsec + am, data = mtcars))
p_(ggally_nostic_se_fit(dt, ggplot2::aes(wt, .se.fit)))
</code></pre>

<hr>
<h2 id='ggally_nostic_sigma'><code><a href="#topic+ggnostic">ggnostic</a></code> leave one out model sigma</h2><span id='topic+ggally_nostic_sigma'></span>

<h3>Description</h3>

<p>A function to display <code><a href="stats.html#topic+lm.influence">stats::influence()</a></code>'s sigma value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_nostic_sigma(
  data,
  mapping,
  ...,
  lineColor = brew_colors("grey"),
  linePosition = attr(data, "broom_glance")$sigma
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_nostic_sigma_+3A_data">data</code>, <code id="ggally_nostic_sigma_+3A_mapping">mapping</code>, <code id="ggally_nostic_sigma_+3A_...">...</code>, <code id="ggally_nostic_sigma_+3A_linecolor">lineColor</code></td>
<td>
<p>parameters supplied to <code><a href="#topic+ggally_nostic_line">ggally_nostic_line</a></code></p>
</td></tr>
<tr><td><code id="ggally_nostic_sigma_+3A_lineposition">linePosition</code></td>
<td>
<p>line that is drawn in the background of the plot. Defaults to the overall model's sigma value.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As stated in <code><a href="stats.html#topic+lm.influence">stats::influence()</a></code> documentation:
</p>
<p>sigma: a vector whose i-th element contains the estimate of the residual standard deviation obtained when the i-th case is dropped from the regression.  (The approximations needed for GLMs can result in this being 'NaN'.)
</p>
<p>A line is added to display the overall model's sigma value. This gives a baseline for comparison
</p>


<h3>Value</h3>

<p><span class="pkg">ggplot2</span> plot object
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm.influence">stats::influence()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

dt &lt;- broomify(stats::lm(mpg ~ wt + qsec + am, data = mtcars))
p_(ggally_nostic_sigma(dt, ggplot2::aes(wt, .sigma)))
</code></pre>

<hr>
<h2 id='ggally_nostic_std_resid'><code><a href="#topic+ggnostic">ggnostic</a></code> standardized residuals</h2><span id='topic+ggally_nostic_std_resid'></span>

<h3>Description</h3>

<p>If non-null <code>pVal</code> and <code>sigma</code> values are given, confidence interval lines will be added to the plot at the specified <code>pVal</code> locations of a N(0, 1) distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_nostic_std_resid(data, mapping, ..., sigma = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_nostic_std_resid_+3A_data">data</code>, <code id="ggally_nostic_std_resid_+3A_mapping">mapping</code>, <code id="ggally_nostic_std_resid_+3A_...">...</code></td>
<td>
<p>parameters supplied to <code><a href="#topic+ggally_nostic_resid">ggally_nostic_resid</a></code></p>
</td></tr>
<tr><td><code id="ggally_nostic_std_resid_+3A_sigma">sigma</code></td>
<td>
<p>sigma value for the <code>pVal</code> percentiles. Set to 1 for standardized residuals</p>
</td></tr>
</table>


<h3>Value</h3>

<p><span class="pkg">ggplot2</span> plot object
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+influence.measures">stats::rstandard()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

dt &lt;- broomify(stats::lm(mpg ~ wt + qsec + am, data = mtcars))
p_(ggally_nostic_std_resid(dt, ggplot2::aes(wt, .std.resid)))
</code></pre>

<hr>
<h2 id='ggally_points'>Scatter plot</h2><span id='topic+ggally_points'></span>

<h3>Description</h3>

<p>Make a scatter plot with a given data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_points(data, mapping, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_points_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_points_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_points_+3A_...">...</code></td>
<td>
<p>other arguments are sent to geom_point</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(mtcars)
p_(ggally_points(mtcars, mapping = ggplot2::aes(disp, hp)))
p_(ggally_points(mtcars, mapping = ggplot2::aes(disp, hp)))
p_(ggally_points(
  mtcars,
  mapping = ggplot2::aes(
    x     = disp,
    y     = hp,
    color = as.factor(cyl),
    size  = gear
  )
))
</code></pre>

<hr>
<h2 id='ggally_ratio'>Mosaic plot</h2><span id='topic+ggally_ratio'></span>

<h3>Description</h3>

<p>Plots the mosaic plot by using fluctuation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_ratio(
  data,
  mapping = ggplot2::aes(!!!stats::setNames(lapply(colnames(data)[1:2], as.name), c("x",
    "y"))),
  ...,
  floor = 0,
  ceiling = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_ratio_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_ratio_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used. Only x and y will used and both are required</p>
</td></tr>
<tr><td><code id="ggally_ratio_+3A_...">...</code></td>
<td>
<p>passed to <code><a href="ggplot2.html#topic+geom_tile">geom_tile</a>(...)</code></p>
</td></tr>
<tr><td><code id="ggally_ratio_+3A_floor">floor</code></td>
<td>
<p>don't display cells smaller than this value</p>
</td></tr>
<tr><td><code id="ggally_ratio_+3A_ceiling">ceiling</code></td>
<td>
<p>max value to scale frequencies.  If any frequency is larger than the ceiling, the fill color is displayed darker than other rectangles</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_ratio(tips, ggplot2::aes(sex, day)))
p_(ggally_ratio(tips, ggplot2::aes(sex, day)) + ggplot2::coord_equal())
# only plot tiles greater or equal to 20 and scale to a max of 50
p_(ggally_ratio(
  tips, ggplot2::aes(sex, day),
  floor = 20, ceiling = 50
) + ggplot2::theme(aspect.ratio = 4 / 2))
</code></pre>

<hr>
<h2 id='ggally_smooth'>Scatter plot with a smoothed line</h2><span id='topic+ggally_smooth'></span><span id='topic+ggally_smooth_loess'></span><span id='topic+ggally_smooth_lm'></span>

<h3>Description</h3>

<p>Add a smoothed condition mean with a given scatter plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_smooth(
  data,
  mapping,
  ...,
  method = "lm",
  formula = y ~ x,
  se = TRUE,
  shrink = TRUE
)

ggally_smooth_loess(data, mapping, ...)

ggally_smooth_lm(data, mapping, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_smooth_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_smooth_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_smooth_+3A_method">method</code>, <code id="ggally_smooth_+3A_se">se</code></td>
<td>
<p>parameters supplied to <code><a href="ggplot2.html#topic+geom_smooth">geom_smooth</a></code></p>
</td></tr>
<tr><td><code id="ggally_smooth_+3A_formula">formula</code>, <code id="ggally_smooth_+3A_...">...</code></td>
<td>
<p>other arguments to add to geom_smooth</p>
</td></tr>
<tr><td><code id="ggally_smooth_+3A_shrink">shrink</code></td>
<td>
<p>boolean to determine if y range is reduced to range of points or points and error ribbon</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Y limits are reduced to match original Y range with the goal of keeping the Y axis the same across plots.
</p>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_smooth(tips, mapping = ggplot2::aes(x = total_bill, y = tip)))
p_(ggally_smooth(tips, mapping = ggplot2::aes(total_bill, tip, color = sex)))
</code></pre>

<hr>
<h2 id='ggally_statistic'>Generalized text display</h2><span id='topic+ggally_statistic'></span>

<h3>Description</h3>

<p>Generalized text display
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_statistic(
  data,
  mapping,
  text_fn,
  title,
  na.rm = NA,
  display_grid = FALSE,
  justify_labels = "right",
  justify_text = "left",
  sep = ": ",
  family = "mono",
  title_args = list(),
  group_args = list(),
  align_percent = 0.5,
  title_hjust = 0.5,
  group_hjust = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_statistic_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_text_fn">text_fn</code></td>
<td>
<p>function that takes in <code>x</code> and <code>y</code> and returns a text string</p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_title">title</code></td>
<td>
<p>title text to be displayed</p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_na.rm">na.rm</code></td>
<td>
<p>logical value which determines if <code>NA</code> values are removed. If <code>TRUE</code>, no warning message will be displayed.</p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_display_grid">display_grid</code></td>
<td>
<p>if <code>TRUE</code>, display aligned panel grid lines. If <code>FALSE</code> (default), display a thin panel border.</p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_justify_labels">justify_labels</code></td>
<td>
<p><code>justify</code> argument supplied when <code><a href="base.html#topic+format">format</a></code>ting the labels</p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_justify_text">justify_text</code></td>
<td>
<p><code>justify</code> argument supplied when <code><a href="base.html#topic+format">format</a></code>ting the returned <code>text_fn(x, y)</code> values</p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_sep">sep</code></td>
<td>
<p>separation value to be placed between the labels and text</p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_family">family</code></td>
<td>
<p>font family used when displaying all text.  This value will be set in <code>title_args</code> or <code>group_args</code> if no <code>family</code> value exists.  By using <code>"mono"</code>, groups will align with each other.</p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_title_args">title_args</code></td>
<td>
<p>arguments being supplied to the title's <code><a href="ggplot2.html#topic+geom_text">geom_text</a>()</code></p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_group_args">group_args</code></td>
<td>
<p>arguments being supplied to the split-by-color group's <code><a href="ggplot2.html#topic+geom_text">geom_text</a>()</code></p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_align_percent">align_percent</code></td>
<td>
<p>relative align position of the text. When <code>title_hjust = 0.5</code> and <code>group_hjust = 0.5</code>, this should not be needed to be set.</p>
</td></tr>
<tr><td><code id="ggally_statistic_+3A_title_hjust">title_hjust</code>, <code id="ggally_statistic_+3A_group_hjust">group_hjust</code></td>
<td>
<p><code>hjust</code> sent to <code><a href="ggplot2.html#topic+geom_text">geom_text</a>()</code> for the title and group values respectively. Any <code>hjust</code> value supplied in <code>title_args</code> or <code>group_args</code> will take precedence.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ggally_cor">ggally_cor</a></code>
</p>

<hr>
<h2 id='ggally_summarise_by'>Summarize a continuous variable by each value of a discrete variable</h2><span id='topic+ggally_summarise_by'></span><span id='topic+weighted_median_iqr'></span><span id='topic+weighted_mean_sd'></span>

<h3>Description</h3>

<p>Display summary statistics of a continuous variable for each value of a discrete variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_summarise_by(
  data,
  mapping,
  text_fn = weighted_median_iqr,
  text_fn_vertical = NULL,
  ...
)

weighted_median_iqr(x, weights = NULL)

weighted_mean_sd(x, weights = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_summarise_by_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_summarise_by_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_summarise_by_+3A_text_fn">text_fn</code></td>
<td>
<p>function that takes an x and weights and returns a text string</p>
</td></tr>
<tr><td><code id="ggally_summarise_by_+3A_text_fn_vertical">text_fn_vertical</code></td>
<td>
<p>function that takes an x and weights and returns a text string, used when <code>x</code> is discrete and <code>y</code> is continuous. If not provided, will use <code>text_fn</code>, replacing spaces by carriage returns.</p>
</td></tr>
<tr><td><code id="ggally_summarise_by_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="ggplot2.html#topic+geom_text">geom_text</a>(...)</code></p>
</td></tr>
<tr><td><code id="ggally_summarise_by_+3A_x">x</code></td>
<td>
<p>a numeric vector</p>
</td></tr>
<tr><td><code id="ggally_summarise_by_+3A_weights">weights</code></td>
<td>
<p>an optional numeric vectors of weights. If <code>NULL</code>, equal weights of 1 will be taken into account.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>weighted_median_iqr</code> computes weighted median and interquartile range.
</p>
<p><code>weighted_mean_sd</code> computes weighted mean and standard deviation.
</p>


<h3>Author(s)</h3>

<p>Joseph Larmarange
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

if (require(Hmisc)) {
  data(tips)
  p_(ggally_summarise_by(tips, mapping = aes(x = total_bill, y = day)))
  p_(ggally_summarise_by(tips, mapping = aes(x = day, y = total_bill)))

  # colour is kept only if equal to the discrete variable
  p_(ggally_summarise_by(tips, mapping = aes(x = total_bill, y = day, color = day)))
  p_(ggally_summarise_by(tips, mapping = aes(x = total_bill, y = day, color = sex)))
  p_(ggally_summarise_by(tips, mapping = aes(x = day, y = total_bill, color = day)))

  # custom text size
  p_(ggally_summarise_by(tips, mapping = aes(x = total_bill, y = day), size = 6))

  # change statistic to display
  p_(ggally_summarise_by(tips, mapping = aes(x = total_bill, y = day), text_fn = weighted_mean_sd))

  # custom stat function
  weighted_sum &lt;- function(x, weights = NULL) {
    if (is.null(weights)) weights &lt;- 1
    paste0("Total : ", round(sum(x * weights, na.rm = TRUE), digits = 1))
  }
  p_(ggally_summarise_by(tips, mapping = aes(x = total_bill, y = day), text_fn = weighted_sum))
}
</code></pre>

<hr>
<h2 id='ggally_table'>Display a table of the number of observations</h2><span id='topic+ggally_table'></span><span id='topic+ggally_tableDiag'></span>

<h3>Description</h3>

<p>Plot the number of observations as a table. Other statistics computed
by <code><a href="#topic+stat_cross">stat_cross</a></code>  could be used (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_table(
  data,
  mapping,
  keep.zero.cells = FALSE,
  ...,
  geom_tile_args = NULL
)

ggally_tableDiag(
  data,
  mapping,
  keep.zero.cells = FALSE,
  ...,
  geom_tile_args = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_table_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_table_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_table_+3A_keep.zero.cells">keep.zero.cells</code></td>
<td>
<p>If <code>TRUE</code>, display cells with no observation.</p>
</td></tr>
<tr><td><code id="ggally_table_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="ggplot2.html#topic+geom_text">geom_text</a>(...)</code></p>
</td></tr>
<tr><td><code id="ggally_table_+3A_geom_tile_args">geom_tile_args</code></td>
<td>
<p>other arguments passed to <code><a href="ggplot2.html#topic+geom_tile">geom_tile</a>(...)</code></p>
</td></tr>
</table>


<h3>Note</h3>

<p>The <strong>colour</strong> aesthetic is taken into account only if equal to
<strong>x</strong> or <strong>y</strong>.
</p>


<h3>Author(s)</h3>

<p>Joseph Larmarange
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggally_table(tips, mapping = aes(x = smoker, y = sex)))
p_(ggally_table(tips, mapping = aes(x = day, y = time)))
p_(ggally_table(tips, mapping = aes(x = smoker, y = sex, colour = smoker)))

# colour is kept only if equal to x or y
p_(ggally_table(tips, mapping = aes(x = smoker, y = sex, colour = day)))

# diagonal version
p_(ggally_tableDiag(tips, mapping = aes(x = smoker)))

# custom label size and color
p_(ggally_table(tips, mapping = aes(x = smoker, y = sex), size = 16, color = "red"))

# display column proportions
p_(ggally_table(
  tips,
  mapping = aes(x = day, y = sex, label = scales::percent(after_stat(col.prop)))
))

# draw table cells
p_(ggally_table(
  tips,
  mapping = aes(x = smoker, y = sex),
  geom_tile_args = list(colour = "black", fill = "white")
))

# Use standardized residuals to fill table cells
p_(ggally_table(
  as.data.frame(Titanic),
  mapping = aes(
    x = Class, y = Survived, weight = Freq,
    fill = after_stat(std.resid),
    label = scales::percent(after_stat(col.prop), accuracy = .1)
  ),
  geom_tile_args = list(colour = "black")
) +
  scale_fill_steps2(breaks = c(-3, -2, 2, 3), show.limits = TRUE))
</code></pre>

<hr>
<h2 id='ggally_text'>Text plot</h2><span id='topic+ggally_text'></span>

<h3>Description</h3>

<p>Plot text for a plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_text(
  label,
  mapping = ggplot2::aes(color = I("black")),
  xP = 0.5,
  yP = 0.5,
  xrange = c(0, 1),
  yrange = c(0, 1),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_text_+3A_label">label</code></td>
<td>
<p>text that you want to appear</p>
</td></tr>
<tr><td><code id="ggally_text_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics that don't relate to position (such as color)</p>
</td></tr>
<tr><td><code id="ggally_text_+3A_xp">xP</code></td>
<td>
<p>horizontal position percentage</p>
</td></tr>
<tr><td><code id="ggally_text_+3A_yp">yP</code></td>
<td>
<p>vertical position percentage</p>
</td></tr>
<tr><td><code id="ggally_text_+3A_xrange">xrange</code></td>
<td>
<p>range of the data around it.  Only nice to have if plotting in a matrix</p>
</td></tr>
<tr><td><code id="ggally_text_+3A_yrange">yrange</code></td>
<td>
<p>range of the data around it.  Only nice to have if plotting in a matrix</p>
</td></tr>
<tr><td><code id="ggally_text_+3A_...">...</code></td>
<td>
<p>other arguments for geom_text</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

p_(ggally_text("Example 1"))
p_(ggally_text("Example\nTwo", mapping = ggplot2::aes(size = 15), color = I("red")))
</code></pre>

<hr>
<h2 id='ggally_trends'>Trends line plot</h2><span id='topic+ggally_trends'></span>

<h3>Description</h3>

<p>Plot trends using line plots.
For continuous y variables, plot the evolution of the mean.
For binary y variables, plot the evolution of the proportion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggally_trends(data, mapping, ..., include_zero = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggally_trends_+3A_data">data</code></td>
<td>
<p>data set using</p>
</td></tr>
<tr><td><code id="ggally_trends_+3A_mapping">mapping</code></td>
<td>
<p>aesthetics being used</p>
</td></tr>
<tr><td><code id="ggally_trends_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="ggplot2.html#topic+geom_path">ggplot2::geom_line()</a></code></p>
</td></tr>
<tr><td><code id="ggally_trends_+3A_include_zero">include_zero</code></td>
<td>
<p>Should 0 be included on the y-axis?</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Joseph Larmarange
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
tips_f &lt;- tips
tips_f$day &lt;- factor(tips$day, c("Thur", "Fri", "Sat", "Sun"))

# Numeric variable
p_(ggally_trends(tips_f, mapping = aes(x = day, y = total_bill)))
p_(ggally_trends(tips_f, mapping = aes(x = day, y = total_bill, colour = time)))

# Binary variable
p_(ggally_trends(tips_f, mapping = aes(x = day, y = smoker)))
p_(ggally_trends(tips_f, mapping = aes(x = day, y = smoker, colour = sex)))

# Discrete variable with 3 or more categories
p_(ggally_trends(tips_f, mapping = aes(x = smoker, y = day)))
p_(ggally_trends(tips_f, mapping = aes(x = smoker, y = day, color = sex)))

# Include zero on Y axis
p_(ggally_trends(tips_f, mapping = aes(x = day, y = total_bill), include_zero = TRUE))
p_(ggally_trends(tips_f, mapping = aes(x = day, y = smoker), include_zero = TRUE))

# Change line size
p_(ggally_trends(tips_f, mapping = aes(x = day, y = smoker, colour = sex), size = 3))

# Define weights with the appropriate aesthetic
d &lt;- as.data.frame(Titanic)
p_(ggally_trends(
  d,
  mapping = aes(x = Class, y = Survived, weight = Freq, color = Sex),
  include_zero = TRUE
))
</code></pre>

<hr>
<h2 id='ggbivariate'>Display an outcome using several potential explanatory variables</h2><span id='topic+ggbivariate'></span>

<h3>Description</h3>

<p><code>ggbivariate</code> is a variant of <code><a href="#topic+ggduo">ggduo</a></code> for plotting
an outcome variable with several potential explanatory variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggbivariate(
  data,
  outcome,
  explanatory = NULL,
  mapping = NULL,
  types = NULL,
  ...,
  rowbar_args = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggbivariate_+3A_data">data</code></td>
<td>
<p>dataset to be used, can have both categorical and
numerical variables</p>
</td></tr>
<tr><td><code id="ggbivariate_+3A_outcome">outcome</code></td>
<td>
<p>name or position of the outcome variable (one variable only)</p>
</td></tr>
<tr><td><code id="ggbivariate_+3A_explanatory">explanatory</code></td>
<td>
<p>names or positions of the explanatory variables (if <code>NULL</code>,
will take all variables other than <code>outcome</code>)</p>
</td></tr>
<tr><td><code id="ggbivariate_+3A_mapping">mapping</code></td>
<td>
<p>additional aesthetic to be used, for example to indicate
weights (see examples)</p>
</td></tr>
<tr><td><code id="ggbivariate_+3A_types">types</code></td>
<td>
<p>custom types of plots to use, see <code><a href="#topic+ggduo">ggduo</a></code></p>
</td></tr>
<tr><td><code id="ggbivariate_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code><a href="#topic+ggduo">ggduo</a></code> (see examples)</p>
</td></tr>
<tr><td><code id="ggbivariate_+3A_rowbar_args">rowbar_args</code></td>
<td>
<p>additional arguments passed to <code><a href="#topic+ggally_rowbar">ggally_rowbar</a></code>
(see examples)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Joseph Larmarange
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggbivariate(tips, "smoker", c("day", "time", "sex", "tip")))

# Personalize plot title and legend title
p_(ggbivariate(
  tips, "smoker", c("day", "time", "sex", "tip"),
  title = "Custom title"
) +
  labs(fill = "Smoker ?"))

# Customize fill colour scale
p_(ggbivariate(tips, "smoker", c("day", "time", "sex", "tip")) +
  scale_fill_brewer(type = "qual"))

# Customize labels
p_(ggbivariate(
  tips, "smoker", c("day", "time", "sex", "tip"),
  rowbar_args = list(
    colour = "white",
    size = 4,
    fontface = "bold",
    label_format = scales::label_percent(accurary = 1)
  )
))

# Choose the sub-plot from which get legend
p_(ggbivariate(tips, "smoker"))
p_(ggbivariate(tips, "smoker", legend = 3))

# Use mapping to indicate weights
d &lt;- as.data.frame(Titanic)
p_(ggbivariate(d, "Survived", mapping = aes(weight = Freq)))

# outcome can be numerical
p_(ggbivariate(tips, outcome = "tip", title = "tip"))
</code></pre>

<hr>
<h2 id='ggcoef'>Model coefficients with <span class="pkg">broom</span> and <span class="pkg">ggplot2</span></h2><span id='topic+ggcoef'></span>

<h3>Description</h3>

<p>Plot the coefficients of a model with <span class="pkg">broom</span> and <span class="pkg">ggplot2</span>.
For an updated and improved version, see <code><a href="#topic+ggcoef_model">ggcoef_model()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggcoef(
  x,
  mapping = aes(!!as.name("estimate"), !!as.name("term")),
  conf.int = TRUE,
  conf.level = 0.95,
  exponentiate = FALSE,
  exclude_intercept = FALSE,
  vline = TRUE,
  vline_intercept = "auto",
  vline_color = "gray50",
  vline_linetype = "dotted",
  vline_size = 1,
  errorbar_color = "gray25",
  errorbar_height = 0,
  errorbar_linetype = "solid",
  errorbar_size = 0.5,
  sort = c("none", "ascending", "descending"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggcoef_+3A_x">x</code></td>
<td>
<p>a model object to be tidied with <code><a href="broom.html#topic+reexports">broom::tidy()</a></code> or a data frame (see Details)</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_mapping">mapping</code></td>
<td>
<p>default aesthetic mapping</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_conf.int">conf.int</code></td>
<td>
<p>display confidence intervals as error bars?</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_conf.level">conf.level</code></td>
<td>
<p>level of confidence intervals (passed to <code><a href="broom.html#topic+reexports">broom::tidy()</a></code>
if <code>x</code> is not a data frame)</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_exponentiate">exponentiate</code></td>
<td>
<p>if <code>TRUE</code>, x-axis will be logarithmic (also passed to <code><a href="broom.html#topic+reexports">broom::tidy()</a></code>
if <code>x</code> is not a data frame)</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_exclude_intercept">exclude_intercept</code></td>
<td>
<p>should the intercept be excluded from the plot?</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_vline">vline</code></td>
<td>
<p>print a vertical line?</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_vline_intercept">vline_intercept</code></td>
<td>
<p><code>xintercept</code> for the vertical line.
<code>"auto"</code> for <code>x = 0</code> (or <code>x = 1</code> if <code>exponentiate</code> is <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_vline_color">vline_color</code></td>
<td>
<p>color of the vertical line</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_vline_linetype">vline_linetype</code></td>
<td>
<p>line type of the vertical line</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_vline_size">vline_size</code></td>
<td>
<p>size of the vertical line</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_errorbar_color">errorbar_color</code></td>
<td>
<p>color of the error bars</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_errorbar_height">errorbar_height</code></td>
<td>
<p>height of the error bars</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_errorbar_linetype">errorbar_linetype</code></td>
<td>
<p>line type of the error bars</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_errorbar_size">errorbar_size</code></td>
<td>
<p>size of the error bars</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_sort">sort</code></td>
<td>
<p><code>"none"</code> (default) do not sort, <code>"ascending"</code> sort by increasing coefficient value, or <code>"descending"</code> sort by decreasing coefficient value</p>
</td></tr>
<tr><td><code id="ggcoef_+3A_...">...</code></td>
<td>
<p>additional arguments sent to <code><a href="ggplot2.html#topic+geom_point">ggplot2::geom_point()</a></code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

library(broom)
reg &lt;- lm(Sepal.Length ~ Sepal.Width + Petal.Length + Petal.Width, data = iris)
p_(ggcoef(reg))

d &lt;- as.data.frame(Titanic)
reg2 &lt;- glm(Survived ~ Sex + Age + Class, family = binomial, data = d, weights = d$Freq)
ggcoef(reg2, exponentiate = TRUE)
ggcoef(
  reg2,
  exponentiate = TRUE, exclude_intercept = TRUE,
  errorbar_height = .2, color = "blue", sort = "ascending"
)

</code></pre>

<hr>
<h2 id='ggcorr'>Correlation matrix</h2><span id='topic+ggcorr'></span>

<h3>Description</h3>

<p>Function for making a correlation matrix plot, using <span class="pkg">ggplot2</span>.
The function is directly inspired by Tian Zheng and Yu-Sung Su's
<code>corrplot</code> function in the 'arm' package.
Please visit <a href="https://github.com/briatte/ggcorr">https://github.com/briatte/ggcorr</a> for the latest version
of <code>ggcorr</code>, and see the vignette at
<a href="https://briatte.github.io/ggcorr/">https://briatte.github.io/ggcorr/</a> for many examples of how to use it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggcorr(
  data,
  method = c("pairwise", "pearson"),
  cor_matrix = NULL,
  nbreaks = NULL,
  digits = 2,
  name = "",
  low = "#3B9AB2",
  mid = "#EEEEEE",
  high = "#F21A00",
  midpoint = 0,
  palette = NULL,
  geom = "tile",
  min_size = 2,
  max_size = 6,
  label = FALSE,
  label_alpha = FALSE,
  label_color = "black",
  label_round = 1,
  label_size = 4,
  limits = c(-1, 1),
  drop = is.null(limits) || identical(limits, FALSE),
  layout.exp = 0,
  legend.position = "right",
  legend.size = 9,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggcorr_+3A_data">data</code></td>
<td>
<p>a data frame or matrix containing numeric (continuous) data. If
any of the columns contain non-numeric data, they will be dropped with a
warning.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_method">method</code></td>
<td>
<p>a vector of two character strings. The first value gives the
method for computing covariances in the presence of missing values, and must
be (an abbreviation of) one of <code>"everything"</code>, <code>"all.obs"</code>,
<code>"complete.obs"</code>, <code>"na.or.complete"</code> or
<code>"pairwise.complete.obs"</code>. The second value gives the type of
correlation coefficient to compute, and must be one of <code>"pearson"</code>,
<code>"kendall"</code> or <code>"spearman"</code>.
See <code><a href="stats.html#topic+cor">cor</a></code> for details.
Defaults to <code>c("pairwise", "pearson")</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_cor_matrix">cor_matrix</code></td>
<td>
<p>the named correlation matrix to use for calculations.
Defaults to the correlation matrix of <code>data</code> when <code>data</code> is
supplied.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_nbreaks">nbreaks</code></td>
<td>
<p>the number of breaks to apply to the correlation coefficients,
which results in a categorical color scale. See 'Note'.
Defaults to <code>NULL</code> (no breaks, continuous scaling).</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_digits">digits</code></td>
<td>
<p>the number of digits to show in the breaks of the correlation
coefficients: see <code><a href="base.html#topic+cut">cut</a></code> for details.
Defaults to <code>2</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_name">name</code></td>
<td>
<p>a character string for the legend that shows the colors of the
correlation coefficients.
Defaults to <code>""</code> (no legend name).</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_low">low</code></td>
<td>
<p>the lower color of the gradient for continuous scaling of the
correlation coefficients.
Defaults to <code>"#3B9AB2"</code> (blue).</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_mid">mid</code></td>
<td>
<p>the midpoint color of the gradient for continuous scaling of the
correlation coefficients.
Defaults to <code>"#EEEEEE"</code> (very light grey).</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_high">high</code></td>
<td>
<p>the upper color of the gradient for continuous scaling of the
correlation coefficients.
Defaults to <code>"#F21A00"</code> (red).</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_midpoint">midpoint</code></td>
<td>
<p>the midpoint value for continuous scaling of the
correlation coefficients.
Defaults to <code>0</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_palette">palette</code></td>
<td>
<p>if <code>nbreaks</code> is used, a ColorBrewer palette to use
instead of the colors specified by <code>low</code>, <code>mid</code> and <code>high</code>.
Defaults to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_geom">geom</code></td>
<td>
<p>the geom object to use. Accepts either <code>"tile"</code>,
<code>"circle"</code>, <code>"text"</code> or <code>"blank"</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_min_size">min_size</code></td>
<td>
<p>when <code>geom</code> has been set to <code>"circle"</code>, the minimum
size of the circles.
Defaults to <code>2</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_max_size">max_size</code></td>
<td>
<p>when <code>geom</code> has been set to <code>"circle"</code>, the maximum
size of the circles.
Defaults to <code>6</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_label">label</code></td>
<td>
<p>whether to add correlation coefficients to the plot.
Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_label_alpha">label_alpha</code></td>
<td>
<p>whether to make the correlation coefficients increasingly
transparent as they come close to 0. Also accepts any numeric value between
<code>0</code> and <code>1</code>, in which case the level of transparency is set to that
fixed value.
Defaults to <code>FALSE</code> (no transparency).</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_label_color">label_color</code></td>
<td>
<p>the color of the correlation coefficients.
Defaults to <code>"grey75"</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_label_round">label_round</code></td>
<td>
<p>the decimal rounding of the correlation coefficients.
Defaults to <code>1</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_label_size">label_size</code></td>
<td>
<p>the size of the correlation coefficients.
Defaults to <code>4</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_limits">limits</code></td>
<td>
<p>bounding of color scaling for correlations, set <code>limits = NULL</code> or <code>FALSE</code> to remove</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_drop">drop</code></td>
<td>
<p>if using <code>nbreaks</code>, whether to drop unused breaks from the
color scale.
Defaults to <code>FALSE</code> (recommended).</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_layout.exp">layout.exp</code></td>
<td>
<p>a multiplier to expand the horizontal axis to the left if
variable names get clipped.
Defaults to <code>0</code> (no expansion).</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_legend.position">legend.position</code></td>
<td>
<p>where to put the legend of the correlation
coefficients: see <code><a href="ggplot2.html#topic+theme">theme</a></code> for details.
Defaults to <code>"bottom"</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_legend.size">legend.size</code></td>
<td>
<p>the size of the legend title and labels, in points: see
<code><a href="ggplot2.html#topic+theme">theme</a></code> for details.
Defaults to <code>9</code>.</p>
</td></tr>
<tr><td><code id="ggcorr_+3A_...">...</code></td>
<td>
<p>other arguments supplied to <code><a href="ggplot2.html#topic+geom_text">geom_text</a></code> for
the diagonal labels.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Recommended values for the <code>nbreaks</code> argument are <code>3</code> to
<code>11</code>, as values above 11 are visually difficult to separate and are not
supported by diverging ColorBrewer palettes.
</p>


<h3>Author(s)</h3>

<p>Francois Briatte, with contributions from Amos B. Elberg and
Barret Schloerke
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+cor">cor</a></code> and <code>corrplot</code> in the
<code>arm</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

# Basketball statistics provided by Nathan Yau at Flowing Data.
dt &lt;- read.csv("http://datasets.flowingdata.com/ppg2008.csv")

# Default output.
p_(ggcorr(dt[, -1]))

# Labeled output, with coefficient transparency.
p_(ggcorr(dt[, -1],
  label = TRUE,
  label_alpha = TRUE
))

# Custom options.
p_(ggcorr(
  dt[, -1],
  name = expression(rho),
  geom = "circle",
  max_size = 10,
  min_size = 2,
  size = 3,
  hjust = 0.75,
  nbreaks = 6,
  angle = -45,
  palette = "PuOr" # colorblind safe, photocopy-able
))

# Supply your own correlation matrix
p_(ggcorr(
  data = NULL,
  cor_matrix = cor(dt[, -1], use = "pairwise")
))
</code></pre>

<hr>
<h2 id='ggduo'><span class="pkg">ggplot2</span> generalized pairs plot for two columns sets of data</h2><span id='topic+ggduo'></span>

<h3>Description</h3>

<p>Make a matrix of plots with a given data set with two different column sets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggduo(
  data,
  mapping = NULL,
  columnsX = 1:ncol(data),
  columnsY = 1:ncol(data),
  title = NULL,
  types = list(continuous = "smooth_loess", comboVertical = "box_no_facet",
    comboHorizontal = "facethist", discrete = "count"),
  axisLabels = c("show", "none"),
  columnLabelsX = colnames(data[columnsX]),
  columnLabelsY = colnames(data[columnsY]),
  labeller = "label_value",
  switch = NULL,
  xlab = NULL,
  ylab = NULL,
  showStrips = NULL,
  legend = NULL,
  cardinality_threshold = 15,
  progress = NULL,
  xProportions = NULL,
  yProportions = NULL,
  legends = stop("deprecated")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggduo_+3A_data">data</code></td>
<td>
<p>data set using.  Can have both numerical and categorical data.</p>
</td></tr>
<tr><td><code id="ggduo_+3A_mapping">mapping</code></td>
<td>
<p>aesthetic mapping (besides <code>x</code> and <code>y</code>).  See <code><a href="ggplot2.html#topic+aes">aes</a>()</code>.  If <code>mapping</code> is numeric, <code>columns</code> will be set to the <code>mapping</code> value and <code>mapping</code> will be set to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="ggduo_+3A_columnsx">columnsX</code>, <code id="ggduo_+3A_columnsy">columnsY</code></td>
<td>
<p>which columns are used to make plots.  Defaults to all columns.</p>
</td></tr>
<tr><td><code id="ggduo_+3A_title">title</code>, <code id="ggduo_+3A_xlab">xlab</code>, <code id="ggduo_+3A_ylab">ylab</code></td>
<td>
<p>title, x label, and y label for the graph</p>
</td></tr>
<tr><td><code id="ggduo_+3A_types">types</code></td>
<td>
<p>see Details</p>
</td></tr>
<tr><td><code id="ggduo_+3A_axislabels">axisLabels</code></td>
<td>
<p>either &quot;show&quot; to display axisLabels or &quot;none&quot; for no axis labels</p>
</td></tr>
<tr><td><code id="ggduo_+3A_columnlabelsx">columnLabelsX</code>, <code id="ggduo_+3A_columnlabelsy">columnLabelsY</code></td>
<td>
<p>label names to be displayed.  Defaults to names of columns being used.</p>
</td></tr>
<tr><td><code id="ggduo_+3A_labeller">labeller</code></td>
<td>
<p>labeller for facets. See <code><a href="ggplot2.html#topic+labellers">labellers</a></code>. Common values are <code>"label_value"</code> (default) and <code>"label_parsed"</code>.</p>
</td></tr>
<tr><td><code id="ggduo_+3A_switch">switch</code></td>
<td>
<p>switch parameter for facet_grid. See <code>ggplot2::<a href="ggplot2.html#topic+facet_grid">facet_grid</a></code>.  By default, the labels are displayed on the top and right of the plot. If <code>"x"</code>, the top labels will be displayed to the bottom. If <code>"y"</code>, the right-hand side labels will be displayed to the left. Can also be set to <code>"both"</code></p>
</td></tr>
<tr><td><code id="ggduo_+3A_showstrips">showStrips</code></td>
<td>
<p>boolean to determine if each plot's strips should be displayed. <code>NULL</code> will default to the top and right side plots only. <code>TRUE</code> or <code>FALSE</code> will turn all strips on or off respectively.</p>
</td></tr>
<tr><td><code id="ggduo_+3A_legend">legend</code></td>
<td>
<p>May be the two objects described below or the default <code>NULL</code> value.  The legend position can be moved by using ggplot2's theme element <code>pm + theme(legend.position = "bottom")</code> </p>
<dl>
<dt>a numeric vector of length 2</dt><dd><p>provides the location of the plot to use the legend for the plot matrix's legend. Such as <code>legend = c(3,5)</code> which will use the legend from the plot in the third row and fifth column</p>
</dd><dt>a single numeric value</dt><dd><p>provides the location of a plot according to the display order. Such as <code>legend = 3</code> in a plot matrix with 2 rows and 5 columns displayed by column will return the plot in position <code>c(1,2)</code></p>
</dd><dt>a object from <code><a href="#topic+grab_legend">grab_legend</a>()</code></dt><dd><p>a predetermined plot legend that will be displayed directly</p>
</dd></dl>
</td></tr>
<tr><td><code id="ggduo_+3A_cardinality_threshold">cardinality_threshold</code></td>
<td>
<p>maximum number of levels allowed in a character / factor column.  Set this value to NULL to not check factor columns. Defaults to 15</p>
</td></tr>
<tr><td><code id="ggduo_+3A_progress">progress</code></td>
<td>
<p><code>NULL</code> (default) for a progress bar in interactive sessions with more than 15 plots, <code>TRUE</code> for a progress bar, <code>FALSE</code> for no progress bar, or a function that accepts at least a plot matrix and returns a new <code>progress::<a href="progress.html#topic+progress_bar">progress_bar</a></code>.  See <code><a href="#topic+ggmatrix_progress">ggmatrix_progress</a></code>.</p>
</td></tr>
<tr><td><code id="ggduo_+3A_xproportions">xProportions</code>, <code id="ggduo_+3A_yproportions">yProportions</code></td>
<td>
<p>Value to change how much area is given for each plot. Either <code>NULL</code> (default), numeric value matching respective length, <code>grid::<a href="grid.html#topic+unit">unit</a></code> object with matching respective length or <code>"auto"</code> for automatic relative proportions based on the number of levels for categorical variables.</p>
</td></tr>
<tr><td><code id="ggduo_+3A_legends">legends</code></td>
<td>
<p>deprecated</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>types</code> is a list that may contain the variables
'continuous', 'combo', 'discrete', and 'na'. Each element of the list may be a function or a string.  If a string is supplied, If a string is supplied, it must be a character string representing the tail end of a <code>ggally_NAME</code> function. The list of current valid <code>ggally_NAME</code> functions is visible in a dedicated vignette.
</p>

<dl>
<dt>continuous</dt><dd><p>This option is used for continuous X and Y data.</p>
</dd>
<dt>comboHorizontal</dt><dd><p>This option is used for either continuous X and categorical Y data or categorical X and continuous Y data.</p>
</dd>
<dt>comboVertical</dt><dd><p>This option is used for either continuous X and categorical Y data or categorical X and continuous Y data.</p>
</dd>
<dt>discrete</dt><dd><p>This option is used for categorical X and Y data.</p>
</dd>
<dt>na</dt><dd><p>This option is used when all X data is <code>NA</code>, all Y data is <code>NA</code>, or either all X or Y data is <code>NA</code>.</p>
</dd>
</dl>

<p>If 'blank' is ever chosen as an option, then ggduo will produce an empty plot.
</p>
<p>If a function is supplied as an option, it should implement the function api of <code>function(data, mapping, ...){#make ggplot2 plot}</code>.  If a specific function needs its parameters set, <code><a href="#topic+wrap">wrap</a>(fn, param1 = val1, param2 = val2)</code> the function with its parameters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(baseball)

# Keep players from 1990-1995 with at least one at bat
# Add how many singles a player hit
# (must do in two steps as X1b is used in calculations)
dt &lt;- transform(
  subset(baseball, year &gt;= 1990 &amp; year &lt;= 1995 &amp; ab &gt; 0),
  X1b = h - X2b - X3b - hr
)
# Add
#  the player's batting average,
#  the player's slugging percentage,
#  and the player's on base percentage
# Make factor a year, as each season is discrete
dt &lt;- transform(
  dt,
  batting_avg = h / ab,
  slug = (X1b + 2 * X2b + 3 * X3b + 4 * hr) / ab,
  on_base = (h + bb + hbp) / (ab + bb + hbp),
  year = as.factor(year)
)


pm &lt;- ggduo(
  dt,
  c("year", "g", "ab", "lg"),
  c("batting_avg", "slug", "on_base"),
  mapping = ggplot2::aes(color = lg)
)
# Prints, but
#   there is severe over plotting in the continuous plots
#   the labels could be better
#   want to add more hitting information
p_(pm)

# address overplotting issues and add a title
pm &lt;- ggduo(
  dt,
  c("year", "g", "ab", "lg"),
  c("batting_avg", "slug", "on_base"),
  columnLabelsX = c("year", "player game count", "player at bat count", "league"),
  columnLabelsY = c("batting avg", "slug %", "on base %"),
  title = "Baseball Hitting Stats from 1990-1995",
  mapping = ggplot2::aes(color = lg),
  types = list(
    # change the shape and add some transparency to the points
    continuous = wrap("smooth_loess", alpha = 0.50, shape = "+")
  ),
  showStrips = FALSE
)

p_(pm)

# Use "auto" to adapt width of the sub-plots
pm &lt;- ggduo(
  dt,
  c("year", "g", "ab", "lg"),
  c("batting_avg", "slug", "on_base"),
  mapping = ggplot2::aes(color = lg),
  xProportions = "auto"
)

p_(pm)

# Custom widths &amp; heights of the sub-plots
pm &lt;- ggduo(
  dt,
  c("year", "g", "ab", "lg"),
  c("batting_avg", "slug", "on_base"),
  mapping = ggplot2::aes(color = lg),
  xProportions = c(6, 4, 3, 2),
  yProportions = c(1, 2, 1)
)

p_(pm)

# Example derived from:
## R Data Analysis Examples | Canonical Correlation Analysis.  UCLA: Institute for Digital
##   Research and Education.
##   from http://www.stats.idre.ucla.edu/r/dae/canonical-correlation-analysis
##   (accessed May 22, 2017).
# "Example 1. A researcher has collected data on three psychological variables, four
#  academic variables (standardized test scores) and gender for 600 college freshman.
#  She is interested in how the set of psychological variables relates to the academic
#  variables and gender. In particular, the researcher is interested in how many
#  dimensions (canonical variables) are necessary to understand the association between
#  the two sets of variables."
data(psychademic)
summary(psychademic)

(psych_variables &lt;- attr(psychademic, "psychology"))
(academic_variables &lt;- attr(psychademic, "academic"))

## Within correlation
p_(ggpairs(psychademic, columns = psych_variables))
p_(ggpairs(psychademic, columns = academic_variables))

## Between correlation
loess_with_cor &lt;- function(data, mapping, ..., method = "pearson") {
  x &lt;- eval_data_col(data, mapping$x)
  y &lt;- eval_data_col(data, mapping$y)
  cor &lt;- cor(x, y, method = method)
  ggally_smooth_loess(data, mapping, ...) +
    ggplot2::geom_label(
      data = data.frame(
        x = min(x, na.rm = TRUE),
        y = max(y, na.rm = TRUE),
        lab = round(cor, digits = 3)
      ),
      mapping = ggplot2::aes(x = x, y = y, label = lab),
      hjust = 0, vjust = 1,
      size = 5, fontface = "bold",
      inherit.aes = FALSE # do not inherit anything from the ...
    )
}
pm &lt;- ggduo(
  psychademic,
  rev(psych_variables), academic_variables,
  types = list(continuous = loess_with_cor),
  showStrips = FALSE
)
suppressWarnings(p_(pm)) # ignore warnings from loess

# add color according to sex
pm &lt;- ggduo(
  psychademic,
  mapping = ggplot2::aes(color = sex),
  rev(psych_variables), academic_variables,
  types = list(continuous = loess_with_cor),
  showStrips = FALSE,
  legend = c(5, 2)
)
suppressWarnings(p_(pm))


# add color according to sex
pm &lt;- ggduo(
  psychademic,
  mapping = ggplot2::aes(color = motivation),
  rev(psych_variables), academic_variables,
  types = list(continuous = loess_with_cor),
  showStrips = FALSE,
  legend = c(5, 2)
) +
  ggplot2::theme(legend.position = "bottom")
suppressWarnings(p_(pm))
#  dt,
#  c("year", "g", "ab", "lg", "lg"),
#  c("batting_avg", "slug", "on_base", "hit_type"),
#  columnLabelsX = c("year", "player game count", "player at bat count", "league", ""),
#  columnLabelsY = c("batting avg", "slug %", "on base %", "hit type"),
#  title = "Baseball Hitting Stats from 1990-1995 (player strike in 1994)",
#  mapping = aes(color = year),
#  types = list(
#    continuous = wrap("smooth_loess", alpha = 0.50, shape = "+"),
#    comboHorizontal = wrap(display_hit_type_combo, binwidth = 15),
#    discrete = wrap(display_hit_type_discrete, color = "black", size = 0.15)
#  ),
#  showStrips = FALSE
## make the 5th column blank, except for the legend
#  australia_PISA2012,
#  c("gender", "age", "homework", "possessions"),
#  c("PV1MATH", "PV2MATH", "PV3MATH", "PV4MATH", "PV5MATH"),
#  types = list(
#    continuous = "points",
#    combo = "box",
#    discrete = "ratio"
#  )
#  australia_PISA2012,
#  c("gender", "age", "homework", "possessions"),
#  c("PV1MATH", "PV2MATH", "PV3MATH", "PV4MATH", "PV5MATH"),
#  mapping = ggplot2::aes(color = gender),
#  types = list(
#    continuous = wrap("smooth", alpha = 0.25, method = "loess"),
#    combo = "box",
#    discrete = "ratio"
#  )
</code></pre>

<hr>
<h2 id='ggfacet'>Single <span class="pkg">ggplot2</span> plot matrix with <code><a href="ggplot2.html#topic+facet_grid">facet_grid</a></code></h2><span id='topic+ggfacet'></span>

<h3>Description</h3>

<p>Single <span class="pkg">ggplot2</span> plot matrix with <code><a href="ggplot2.html#topic+facet_grid">facet_grid</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggfacet(
  data,
  mapping = NULL,
  columnsX = 1:ncol(data),
  columnsY = 1:ncol(data),
  fn = ggally_points,
  ...,
  columnLabelsX = names(data[columnsX]),
  columnLabelsY = names(data[columnsY]),
  xlab = NULL,
  ylab = NULL,
  title = NULL,
  scales = "free"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggfacet_+3A_data">data</code></td>
<td>
<p>data.frame that contains all columns to be displayed.  This data will be melted before being passed into the function <code>fn</code></p>
</td></tr>
<tr><td><code id="ggfacet_+3A_mapping">mapping</code></td>
<td>
<p>aesthetic mapping (besides <code>x</code> and <code>y</code>).  See <code><a href="ggplot2.html#topic+aes">aes</a>()</code></p>
</td></tr>
<tr><td><code id="ggfacet_+3A_columnsx">columnsX</code></td>
<td>
<p>columns to be displayed in the plot matrix</p>
</td></tr>
<tr><td><code id="ggfacet_+3A_columnsy">columnsY</code></td>
<td>
<p>rows to be displayed in the plot matrix</p>
</td></tr>
<tr><td><code id="ggfacet_+3A_fn">fn</code></td>
<td>
<p>function to be executed. Similar to <code><a href="#topic+ggpairs">ggpairs</a></code> and <code><a href="#topic+ggduo">ggduo</a></code>, the function may either be a string identifier or a real function that <code><a href="#topic+wrap">wrap</a></code> understands.</p>
</td></tr>
<tr><td><code id="ggfacet_+3A_...">...</code></td>
<td>
<p>extra arguments passed directly to <code>fn</code></p>
</td></tr>
<tr><td><code id="ggfacet_+3A_columnlabelsx">columnLabelsX</code>, <code id="ggfacet_+3A_columnlabelsy">columnLabelsY</code></td>
<td>
<p>column and row labels to display in the plot matrix</p>
</td></tr>
<tr><td><code id="ggfacet_+3A_xlab">xlab</code>, <code id="ggfacet_+3A_ylab">ylab</code>, <code id="ggfacet_+3A_title">title</code></td>
<td>
<p>plot matrix labels</p>
</td></tr>
<tr><td><code id="ggfacet_+3A_scales">scales</code></td>
<td>
<p>parameter supplied to <code>ggplot2::<a href="ggplot2.html#topic+facet_grid">facet_grid</a></code>. Default behavior is <code>"free"</code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive
if (requireNamespace("chemometrics", quietly = TRUE)) {
  data(NIR, package = "chemometrics")
  NIR_sub &lt;- data.frame(NIR$yGlcEtOH, NIR$xNIR[, 1:3])
  str(NIR_sub)
  x_cols &lt;- c("X1115.0", "X1120.0", "X1125.0")
  y_cols &lt;- c("Glucose", "Ethanol")

  # using ggduo directly
  p &lt;- ggduo(NIR_sub, x_cols, y_cols, types = list(continuous = "points"))
  p_(p)

  # using ggfacet
  p &lt;- ggfacet(NIR_sub, x_cols, y_cols)
  p_(p)

  # add a smoother
  p &lt;- ggfacet(NIR_sub, x_cols, y_cols, fn = "smooth_loess")
  p_(p)
  # same output
  p &lt;- ggfacet(NIR_sub, x_cols, y_cols, fn = ggally_smooth_loess)
  p_(p)

  # Change scales to be the same in for every row and for every column
  p &lt;- ggfacet(NIR_sub, x_cols, y_cols, scales = "fixed")
  p_(p)
}
</code></pre>

<hr>
<h2 id='gglegend'>Plot only legend of plot function</h2><span id='topic+gglegend'></span>

<h3>Description</h3>

<p>Plot only legend of plot function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gglegend(fn)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gglegend_+3A_fn">fn</code></td>
<td>
<p>this value is passed directly to an empty <code><a href="#topic+wrap">wrap</a></code> call.  Please see <code>?<a href="#topic+wrap">wrap</a></code> for more details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a function that when called with arguments will produce the legend of the plotting function supplied.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

# display regular plot
p_(ggally_points(iris, ggplot2::aes(Sepal.Length, Sepal.Width, color = Species)))

# Make a function that will only print the legend
points_legend &lt;- gglegend(ggally_points)
p_(points_legend(iris, ggplot2::aes(Sepal.Length, Sepal.Width, color = Species)))

# produce the sample legend plot, but supply a string that 'wrap' understands
same_points_legend &lt;- gglegend("points")
identical(
  attr(attr(points_legend, "fn"), "original_fn"),
  attr(attr(same_points_legend, "fn"), "original_fn")
)

# Complicated examples
custom_legend &lt;- wrap(gglegend("points"), size = 6)
p_(custom_legend(iris, ggplot2::aes(Sepal.Length, Sepal.Width, color = Species)))

# Use within ggpairs
pm &lt;- ggpairs(
  iris, 1:2,
  mapping = ggplot2::aes(color = Species),
  upper = list(continuous = gglegend("points"))
)
p_(pm)

# Place a legend in a specific location
pm &lt;- ggpairs(iris, 1:2, mapping = ggplot2::aes(color = Species))
# Make the legend
pm[1, 2] &lt;- points_legend(iris, ggplot2::aes(Sepal.Width, Sepal.Length, color = Species))
p_(pm)
</code></pre>

<hr>
<h2 id='ggmatrix'><span class="pkg">ggplot2</span> plot matrix</h2><span id='topic+ggmatrix'></span>

<h3>Description</h3>

<p>Make a generic matrix of <span class="pkg">ggplot2</span> plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggmatrix(
  plots,
  nrow,
  ncol,
  xAxisLabels = NULL,
  yAxisLabels = NULL,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  byrow = TRUE,
  showStrips = NULL,
  showAxisPlotLabels = TRUE,
  showXAxisPlotLabels = TRUE,
  showYAxisPlotLabels = TRUE,
  labeller = NULL,
  switch = NULL,
  xProportions = NULL,
  yProportions = NULL,
  progress = NULL,
  data = NULL,
  gg = NULL,
  legend = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggmatrix_+3A_plots">plots</code></td>
<td>
<p>list of plots to be put into matrix</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_nrow">nrow</code>, <code id="ggmatrix_+3A_ncol">ncol</code></td>
<td>
<p>number of rows and columns</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_xaxislabels">xAxisLabels</code>, <code id="ggmatrix_+3A_yaxislabels">yAxisLabels</code></td>
<td>
<p>strip titles for the x and y axis respectively. Set to <code>NULL</code> to not be displayed</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_title">title</code>, <code id="ggmatrix_+3A_xlab">xlab</code>, <code id="ggmatrix_+3A_ylab">ylab</code></td>
<td>
<p>title, x label, and y label for the graph. Set to <code>NULL</code> to not be displayed</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_byrow">byrow</code></td>
<td>
<p>boolean that determines whether the plots should be ordered by row or by column</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_showstrips">showStrips</code></td>
<td>
<p>boolean to determine if each plot's strips should be displayed. <code>NULL</code> will default to the top and right side plots only. <code>TRUE</code> or <code>FALSE</code> will turn all strips on or off respectively.</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_showaxisplotlabels">showAxisPlotLabels</code>, <code id="ggmatrix_+3A_showxaxisplotlabels">showXAxisPlotLabels</code>, <code id="ggmatrix_+3A_showyaxisplotlabels">showYAxisPlotLabels</code></td>
<td>
<p>booleans that determine if the plots axis labels are printed on the X (bottom) or Y (left) part of the plot matrix. If <code>showAxisPlotLabels</code> is set, both <code>showXAxisPlotLabels</code> and <code>showYAxisPlotLabels</code> will be set to the given value.</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_labeller">labeller</code></td>
<td>
<p>labeller for facets. See <code><a href="ggplot2.html#topic+labellers">labellers</a></code>. Common values are <code>"label_value"</code> (default) and <code>"label_parsed"</code>.</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_switch">switch</code></td>
<td>
<p>switch parameter for facet_grid. See <code>ggplot2::<a href="ggplot2.html#topic+facet_grid">facet_grid</a></code>.  By default, the labels are displayed on the top and right of the plot. If <code>"x"</code>, the top labels will be displayed to the bottom. If <code>"y"</code>, the right-hand side labels will be displayed to the left. Can also be set to <code>"both"</code></p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_xproportions">xProportions</code>, <code id="ggmatrix_+3A_yproportions">yProportions</code></td>
<td>
<p>Value to change how much area is given for each plot. Either <code>NULL</code> (default), numeric value matching respective length, or <code>grid::<a href="grid.html#topic+unit">unit</a></code> object with matching respective length</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_progress">progress</code></td>
<td>
<p><code>NULL</code> (default) for a progress bar in interactive sessions with more than 15 plots, <code>TRUE</code> for a progress bar, <code>FALSE</code> for no progress bar, or a function that accepts at least a plot matrix and returns a new <code>progress::<a href="progress.html#topic+progress_bar">progress_bar</a></code>.  See <code><a href="#topic+ggmatrix_progress">ggmatrix_progress</a></code>.</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_data">data</code></td>
<td>
<p>data set using. This is the data to be used in place of 'ggally_data' if the plot is a string to be evaluated at print time</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_gg">gg</code></td>
<td>
<p><span class="pkg">ggplot2</span> theme objects to be applied to every plot</p>
</td></tr>
<tr><td><code id="ggmatrix_+3A_legend">legend</code></td>
<td>
<p>May be the two objects described below or the default <code>NULL</code> value.  The legend position can be moved by using ggplot2's theme element <code>pm + theme(legend.position = "bottom")</code> </p>
<dl>
<dt>a numeric vector of length 2</dt><dd><p>provides the location of the plot to use the legend for the plot matrix's legend. Such as <code>legend = c(3,5)</code> which will use the legend from the plot in the third row and fifth column</p>
</dd><dt>a single numeric value</dt><dd><p>provides the location of a plot according to the display order. Such as <code>legend = 3</code> in a plot matrix with 2 rows and 5 columns displayed by column will return the plot in position <code>c(1,2)</code></p>
</dd><dt>a object from <code><a href="#topic+grab_legend">grab_legend</a>()</code></dt><dd><p>a predetermined plot legend that will be displayed directly</p>
</dd></dl>
</td></tr>
</table>


<h3>Memory usage</h3>

<p>Now that the <code><a href="#topic+print.ggmatrix">print.ggmatrix</a></code> method uses a large <span class="pkg">gtable</span> object, rather than print each plot independently, memory usage may be of concern.  From small tests, memory usage flutters around <code>object.size(data) * 0.3 * length(plots)</code>.  So, for a 80Mb random noise dataset with 100 plots, about 2.4 Gb of memory needed to print. For the 3.46 Mb diamonds dataset with 100 plots, about 100 Mb of memory was needed to print.  The benefits of using the <span class="pkg">ggplot2</span> format greatly outweigh the price of about 20% increase in memory usage from the prior ad-hoc print method.
</p>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

plotList &lt;- list()
for (i in 1:6) {
  plotList[[i]] &lt;- ggally_text(paste("Plot #", i, sep = ""))
}
pm &lt;- ggmatrix(
  plotList,
  2, 3,
  c("A", "B", "C"),
  c("D", "E"),
  byrow = TRUE
)
p_(pm)

pm &lt;- ggmatrix(
  plotList,
  2, 3,
  xAxisLabels = c("A", "B", "C"),
  yAxisLabels = NULL,
  byrow = FALSE,
  showXAxisPlotLabels = FALSE
)
p_(pm)
</code></pre>

<hr>
<h2 id='ggmatrix_gtable'><code><a href="#topic+ggmatrix">ggmatrix</a></code> <span class="pkg">gtable</span> object</h2><span id='topic+ggmatrix_gtable'></span>

<h3>Description</h3>

<p>Specialized method to print the <code><a href="#topic+ggmatrix">ggmatrix</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggmatrix_gtable(
  pm,
  ...,
  progress = NULL,
  progress_format = formals(ggmatrix_progress)$format
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggmatrix_gtable_+3A_pm">pm</code></td>
<td>
<p><code><a href="#topic+ggmatrix">ggmatrix</a></code> object to be plotted</p>
</td></tr>
<tr><td><code id="ggmatrix_gtable_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
<tr><td><code id="ggmatrix_gtable_+3A_progress">progress</code>, <code id="ggmatrix_gtable_+3A_progress_format">progress_format</code></td>
<td>
<p>Please use the 'progress' parameter in your <code><a href="#topic+ggmatrix">ggmatrix</a></code>-like function.  See <code><a href="#topic+ggmatrix_progress">ggmatrix_progress</a></code> for a few examples.  These parameters will soon be deprecated.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tips)
pm &lt;- ggpairs(tips, c(1, 3, 2), mapping = ggplot2::aes(color = sex))
ggmatrix_gtable(pm)
</code></pre>

<hr>
<h2 id='ggmatrix_location'><code><a href="#topic+ggmatrix">ggmatrix</a></code> plot locations</h2><span id='topic+ggmatrix_location'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggmatrix_location(pm, location = NULL, rows = NULL, cols = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggmatrix_location_+3A_pm">pm</code></td>
<td>
<p><code><a href="#topic+ggmatrix">ggmatrix</a></code> plot object</p>
</td></tr>
<tr><td><code id="ggmatrix_location_+3A_location">location</code></td>
<td>

<dl>
<dt><code>"all"</code>, <code>TRUE</code></dt><dd><p>All row and col combinations</p>
</dd>
<dt><code>"none"</code></dt><dd><p>No row and column combinations</p>
</dd>
<dt><code>"upper"</code></dt><dd><p>Locations where the column value is higher than the row value</p>
</dd>
<dt><code>"lower"</code></dt><dd><p>Locations where the row value is higher than the column value</p>
</dd>
<dt><code>"diag"</code></dt><dd><p>Locations where the column value is equal to the row value</p>
</dd>
<dt><code>matrix</code> or <code>data.frame</code></dt><dd>
<p><code>matrix</code> values will be converted into <code>data.frame</code>s.
</p>

<ul>
<li><p> A <code>data.frame</code> with the exact column names <code>c("row", "col")</code>
</p>
</li>
<li><p> A <code>data.frame</code> with the number of rows and columns matching the plot matrix object provided.  Each cell will be tested for a &quot;truthy&quot; value to determine if the location should be kept.
</p>
</li></ul>

</dd>
</dl>
</td></tr>
<tr><td><code id="ggmatrix_location_+3A_rows">rows</code></td>
<td>
<p>numeric vector of the rows to be used. Will be used with  <code>cols</code> if <code>location</code> is <code>NULL</code></p>
</td></tr>
<tr><td><code id="ggmatrix_location_+3A_cols">cols</code></td>
<td>
<p>numeric vector of the cols to be used. Will be used with <code>rows</code> if <code>location</code> is <code>NULL</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Convert many types of location values to a consistent <code>data.frame</code> of <code>row</code> and <code>col</code> values.
</p>


<h3>Value</h3>

<p>Data frame with columns <code>c("row", "col")</code> containing locations for the plot matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pm &lt;- ggpairs(tips, 1:3)

# All locations
ggmatrix_location(pm, location = "all")
ggmatrix_location(pm, location = TRUE)

# No locations
ggmatrix_location(pm, location = "none")

# "upper" triangle locations
ggmatrix_location(pm, location = "upper")

# "lower" triangle locations
ggmatrix_location(pm, location = "lower")

# "diag" locations
ggmatrix_location(pm, location = "diag")

# specific rows
ggmatrix_location(pm, rows = 2)

# specific columns
ggmatrix_location(pm, cols = 2)

# row and column combinations
ggmatrix_location(pm, rows = c(1, 2), cols = c(1, 3))

# matrix locations
mat &lt;- matrix(TRUE, ncol = 3, nrow = 3)
mat[1, 1] &lt;- FALSE
locs &lt;- ggmatrix_location(pm, location = mat)
## does not contain the 1, 1 cell
locs

# Use the output of a prior ggmatrix_location
ggmatrix_location(pm, location = locs)
</code></pre>

<hr>
<h2 id='ggmatrix_progress'><code><a href="#topic+ggmatrix">ggmatrix</a></code> default progress bar</h2><span id='topic+ggmatrix_progress'></span>

<h3>Description</h3>

<p><code><a href="#topic+ggmatrix">ggmatrix</a></code> default progress bar
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggmatrix_progress(
  format = " plot: [:plot_i, :plot_j] [:bar]:percent est::eta ",
  clear = TRUE,
  show_after = 0,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggmatrix_progress_+3A_format">format</code>, <code id="ggmatrix_progress_+3A_clear">clear</code>, <code id="ggmatrix_progress_+3A_show_after">show_after</code>, <code id="ggmatrix_progress_+3A_...">...</code></td>
<td>
<p>parameters supplied directly to <code>progress::<a href="progress.html#topic+progress_bar">progress_bar</a>$new()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>function that accepts a plot matrix as the first argument and <code>...</code> for future expansion. Internally, the plot matrix is used to determine the total number of plots for the progress bar.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p_ &lt;- GGally::print_if_interactive

pm &lt;- ggpairs(iris, 1:2, progress = ggmatrix_progress())
p_(pm)

# does not clear after finishing
pm &lt;- ggpairs(iris, 1:2, progress = ggmatrix_progress(clear = FALSE))
p_(pm)
</code></pre>

<hr>
<h2 id='ggnet'>Network plot</h2><span id='topic+ggnet'></span>

<h3>Description</h3>

<p>Function for plotting network objects using <span class="pkg">ggplot2</span>, now replaced by the
<code><a href="#topic+ggnet2">ggnet2</a></code> function, which provides additional control over
plotting parameters. Please visit <a href="https://github.com/briatte/ggnet">https://github.com/briatte/ggnet</a> for
the latest version of ggnet2, and <a href="https://briatte.github.io/ggnet/">https://briatte.github.io/ggnet/</a> for a
vignette that contains many examples and explanations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggnet(
  net,
  mode = "fruchtermanreingold",
  layout.par = NULL,
  layout.exp = 0,
  size = 9,
  alpha = 1,
  weight = "none",
  weight.legend = NA,
  weight.method = weight,
  weight.min = NA,
  weight.max = NA,
  weight.cut = FALSE,
  group = NULL,
  group.legend = NA,
  node.group = group,
  node.color = NULL,
  node.alpha = alpha,
  segment.alpha = alpha,
  segment.color = "grey50",
  segment.label = NULL,
  segment.size = 0.25,
  arrow.size = 0,
  arrow.gap = 0,
  arrow.type = "closed",
  label = FALSE,
  label.nodes = label,
  label.size = size/2,
  label.trim = FALSE,
  legend.size = 9,
  legend.position = "right",
  names = c("", ""),
  quantize.weights = FALSE,
  subset.threshold = 0,
  top8.nodes = FALSE,
  trim.labels = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggnet_+3A_net">net</code></td>
<td>
<p>an object of class <code><a href="network.html#topic+network">network</a></code>, or any object
that can be coerced to this class, such as an adjacency or incidence matrix,
or an edge list: see <a href="network.html#topic+edgeset.constructors">edgeset.constructors</a> and
<a href="network.html#topic+network">network</a> for details. If the object is of class
<a href="igraph.html#topic+aaa-igraph-package">igraph</a> and the
<a href="intergraph.html#topic+intergraph-package">intergraph</a> package is installed,
it will be used to convert the object: see
<code><a href="intergraph.html#topic+asNetwork">asNetwork</a></code> for details.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_mode">mode</code></td>
<td>
<p>a placement method from those provided in the
<code><a href="sna.html#topic+sna">sna</a></code> package: see <a href="sna.html#topic+gplot.layout">gplot.layout</a> for
details. Also accepts the names of two numeric vertex attributes of
<code>net</code>, or a matrix of numeric coordinates, in which case the first two
columns of the matrix are used.
Defaults to the Fruchterman-Reingold force-directed algorithm.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_layout.par">layout.par</code></td>
<td>
<p>options to be passed to the placement method, as listed in
<a href="sna.html#topic+gplot.layout">gplot.layout</a>.
Defaults to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_layout.exp">layout.exp</code></td>
<td>
<p>a multiplier to expand the horizontal axis if node labels
get clipped: see <a href="scales.html#topic+expand_range">expand_range</a> for details.
Defaults to <code>0</code> (no expansion).</p>
</td></tr>
<tr><td><code id="ggnet_+3A_size">size</code></td>
<td>
<p>size of the network nodes. If the nodes are weighted, their area is proportionally scaled up to the size set by <code>size</code>.
Defaults to <code>9</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_alpha">alpha</code></td>
<td>
<p>a level of transparency for nodes, vertices and arrows.
Defaults to <code>1</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_weight">weight</code></td>
<td>
<p>the weighting method for the nodes, which might be a vertex
attribute or a vector of size values. Also accepts <code>"indegree"</code>,
<code>"outdegree"</code>, <code>"degree"</code> or <code>"freeman"</code> to size the nodes by
their unweighted degree centrality (<code>"degree"</code> and <code>"freeman"</code> are
equivalent): see <code><a href="sna.html#topic+degree">degree</a></code> for details. All node weights must
be positive.
Defaults to <code>"none"</code> (no weighting).</p>
</td></tr>
<tr><td><code id="ggnet_+3A_weight.legend">weight.legend</code></td>
<td>
<p>the name to assign to the legend created by
<code>weight</code>.
Defaults to <code>NA</code> (no name).</p>
</td></tr>
<tr><td><code id="ggnet_+3A_weight.method">weight.method</code></td>
<td>
<p>see <code>weight</code></p>
</td></tr>
<tr><td><code id="ggnet_+3A_weight.min">weight.min</code></td>
<td>
<p>whether to subset the network to nodes with a minimum size,
based on the values of <code>weight</code>.
Defaults to <code>NA</code> (preserves all nodes).</p>
</td></tr>
<tr><td><code id="ggnet_+3A_weight.max">weight.max</code></td>
<td>
<p>whether to subset the network to nodes with a maximum size,
based on the values of <code>weight</code>.
Defaults to <code>NA</code> (preserves all nodes).</p>
</td></tr>
<tr><td><code id="ggnet_+3A_weight.cut">weight.cut</code></td>
<td>
<p>whether to cut the size of the nodes into a certain number
of quantiles. Accepts <code>TRUE</code>, which tries to cut the sizes into
quartiles, or any positive numeric value, which tries to cut the sizes into
that many quantiles. If the size of the nodes do not contain the specified
number of distinct quantiles, the largest possible number is used.
See <code><a href="stats.html#topic+quantile">quantile</a></code> and <code><a href="base.html#topic+cut">cut</a></code> for details.
Defaults to <code>FALSE</code> (does nothing).</p>
</td></tr>
<tr><td><code id="ggnet_+3A_group">group</code></td>
<td>
<p>the groups of the nodes, either as a vector of values or as a
vertex attribute. If set to <code>mode</code> on a bipartite network, the nodes
will be grouped as <code>"actor"</code> if they belong to the primary mode and
<code>"event"</code> if they belong to the secondary mode.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_group.legend">group.legend</code></td>
<td>
<p>the name to assign to the legend created by
<code>group</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_node.group">node.group</code></td>
<td>
<p>see <code>group</code></p>
</td></tr>
<tr><td><code id="ggnet_+3A_node.color">node.color</code></td>
<td>
<p>a vector of character strings to color the nodes with,
holding as many colors as there are levels in <code>node.group</code>.
Defaults to <code>NULL</code>, which will assign grayscale colors to each group.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_node.alpha">node.alpha</code></td>
<td>
<p>transparency of the nodes. Inherits from <code>alpha</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_segment.alpha">segment.alpha</code></td>
<td>
<p>the level of transparency of the edges.
Defaults to <code>alpha</code>, which defaults to <code>1</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_segment.color">segment.color</code></td>
<td>
<p>the color of the edges, as a color value, a vector of
color values, or as an edge attribute containing color values.
Defaults to <code>"grey50"</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_segment.label">segment.label</code></td>
<td>
<p>the labels to plot at the middle of the edges, as a
single value, a vector of values, or as an edge attribute.
Defaults to <code>NULL</code> (no edge labels).</p>
</td></tr>
<tr><td><code id="ggnet_+3A_segment.size">segment.size</code></td>
<td>
<p>the size of the edges, in points, as a single numeric
value, a vector of values, or as an edge attribute.
Defaults to <code>0.25</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_arrow.size">arrow.size</code></td>
<td>
<p>the size of the arrows for directed network edges, in
points. See <code><a href="grid.html#topic+arrow">arrow</a></code> for details.
Defaults to <code>0</code> (no arrows).</p>
</td></tr>
<tr><td><code id="ggnet_+3A_arrow.gap">arrow.gap</code></td>
<td>
<p>a setting aimed at improving the display of edge arrows by
plotting slightly shorter edges. Accepts any value between <code>0</code> and
<code>1</code>, where a value of <code>0.05</code> will generally achieve good results
when the size of the nodes is reasonably small.
Defaults to <code>0</code> (no shortening).</p>
</td></tr>
<tr><td><code id="ggnet_+3A_arrow.type">arrow.type</code></td>
<td>
<p>the type of the arrows for directed network edges. See
<code><a href="grid.html#topic+arrow">arrow</a></code> for details.
Defaults to <code>"closed"</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_label">label</code></td>
<td>
<p>whether to label the nodes. If set to <code>TRUE</code>, nodes are
labeled with their vertex names. If set to a vector that contains as many
elements as there are nodes in <code>net</code>, nodes are labeled with these. If
set to any other vector of values, the nodes are labeled only when their
vertex name matches one of these values.
Defaults to <code>FALSE</code> (no labels).</p>
</td></tr>
<tr><td><code id="ggnet_+3A_label.nodes">label.nodes</code></td>
<td>
<p>see <code>label</code></p>
</td></tr>
<tr><td><code id="ggnet_+3A_label.size">label.size</code></td>
<td>
<p>the size of the node labels, in points, as a numeric value,
a vector of numeric values, or as a vertex attribute containing numeric
values.
Defaults to <code>size / 2</code> (half the maximum node size), which defaults to
<code>6</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_label.trim">label.trim</code></td>
<td>
<p>whether to apply some trimming to the node labels. Accepts
any function that can process a character vector, or a strictly positive
numeric value, in which case the labels are trimmed to a fixed-length
substring of that length: see <code><a href="base.html#topic+substr">substr</a></code> for details.
Defaults to <code>FALSE</code> (does nothing).</p>
</td></tr>
<tr><td><code id="ggnet_+3A_legend.size">legend.size</code></td>
<td>
<p>the size of the legend symbols and text, in points.
Defaults to <code>9</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_legend.position">legend.position</code></td>
<td>
<p>the location of the plot legend(s). Accepts all
<code>legend.position</code> values supported by <code><a href="ggplot2.html#topic+theme">theme</a></code>.
Defaults to <code>"right"</code>.</p>
</td></tr>
<tr><td><code id="ggnet_+3A_names">names</code></td>
<td>
<p>deprecated: see <code>group.legend</code> and <code>size.legend</code></p>
</td></tr>
<tr><td><code id="ggnet_+3A_quantize.weights">quantize.weights</code></td>
<td>
<p>deprecated: see <code>weight.cut</code></p>
</td></tr>
<tr><td><code id="ggnet_+3A_subset.threshold">subset.threshold</code></td>
<td>
<p>deprecated: see <code>weight.min</code></p>
</td></tr>
<tr><td><code id="ggnet_+3A_top8.nodes">top8.nodes</code></td>
<td>
<p>deprecated: this functionality was experimental and has
been removed entirely from <code>ggnet</code></p>
</td></tr>
<tr><td><code id="ggnet_+3A_trim.labels">trim.labels</code></td>
<td>
<p>deprecated: see <code>label.trim</code></p>
</td></tr>
<tr><td><code id="ggnet_+3A_...">...</code></td>
<td>
<p>other arguments passed to the <code>geom_text</code> object that sets
the node labels: see <code><a href="ggplot2.html#topic+geom_text">geom_text</a></code> for details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The degree centrality measures that can be produced through the
<code>weight</code> argument will take the directedness of the network into account,
but will be unweighted. To compute weighted network measures, see the
<code>tnet</code> package by Tore Opsahl (<code>help("tnet", package = "tnet")</code>).
</p>


<h3>Author(s)</h3>

<p>Moritz Marbach and Francois Briatte, with help from Heike Hofmann,
Pedro Jordano and Ming-Yu Liu
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggnet2">ggnet2</a></code> in this package,
<code><a href="sna.html#topic+gplot">gplot</a></code> in the <code><a href="sna.html#topic+sna">sna</a></code> package, and
<code><a href="network.html#topic+plot.network">plot.network</a></code> in the <code><a href="network.html#topic+network">network</a></code>
package
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

library(network)

# random adjacency matrix
x           &lt;- 10
ndyads      &lt;- x * (x - 1)
density     &lt;- x / ndyads
m           &lt;- matrix(0, nrow = x, ncol = x)
dimnames(m) &lt;- list(letters[1:x], letters[1:x])
m[row(m) != col(m)] &lt;- runif(ndyads) &lt; density
m

# random undirected network
n &lt;- network::network(m, directed = FALSE)
n

ggnet(n, label = TRUE, alpha = 1, color = "white", segment.color = "black")

# random groups
g &lt;- sample(letters[1:3], 10, replace = TRUE)
g

# color palette
p &lt;- c("a" = "steelblue", "b" = "forestgreen", "c" = "tomato")

p_(ggnet(n, node.group = g, node.color = p, label = TRUE, color = "white"))

# edge arrows on a directed network
p_(ggnet(network(m, directed = TRUE), arrow.gap = 0.05, arrow.size = 10))
</code></pre>

<hr>
<h2 id='ggnet2'>Network plot</h2><span id='topic+ggnet2'></span>

<h3>Description</h3>

<p>Function for plotting network objects using <span class="pkg">ggplot2</span>, with additional control
over graphical parameters that are not supported by the <code><a href="#topic+ggnet">ggnet</a></code>
function. Please visit <a href="https://github.com/briatte/ggnet">https://github.com/briatte/ggnet</a> for the latest
version of ggnet2, and <a href="https://briatte.github.io/ggnet/">https://briatte.github.io/ggnet/</a> for a vignette
that contains many examples and explanations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggnet2(
  net,
  mode = "fruchtermanreingold",
  layout.par = NULL,
  layout.exp = 0,
  alpha = 1,
  color = "grey75",
  shape = 19,
  size = 9,
  max_size = 9,
  na.rm = NA,
  palette = NULL,
  alpha.palette = NULL,
  alpha.legend = NA,
  color.palette = palette,
  color.legend = NA,
  shape.palette = NULL,
  shape.legend = NA,
  size.palette = NULL,
  size.legend = NA,
  size.zero = FALSE,
  size.cut = FALSE,
  size.min = NA,
  size.max = NA,
  label = FALSE,
  label.alpha = 1,
  label.color = "black",
  label.size = max_size/2,
  label.trim = FALSE,
  node.alpha = alpha,
  node.color = color,
  node.label = label,
  node.shape = shape,
  node.size = size,
  edge.alpha = 1,
  edge.color = "grey50",
  edge.lty = "solid",
  edge.size = 0.25,
  edge.label = NULL,
  edge.label.alpha = 1,
  edge.label.color = label.color,
  edge.label.fill = "white",
  edge.label.size = max_size/2,
  arrow.size = 0,
  arrow.gap = 0,
  arrow.type = "closed",
  legend.size = 9,
  legend.position = "right",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggnet2_+3A_net">net</code></td>
<td>
<p>an object of class <code><a href="network.html#topic+network">network</a></code>, or any object
that can be coerced to this class, such as an adjacency or incidence matrix,
or an edge list: see <a href="network.html#topic+edgeset.constructors">edgeset.constructors</a> and
<a href="network.html#topic+network">network</a> for details. If the object is of class
<a href="igraph.html#topic+aaa-igraph-package">igraph</a> and the
<a href="intergraph.html#topic+intergraph-package">intergraph</a> package is installed,
it will be used to convert the object: see
<code><a href="intergraph.html#topic+asNetwork">asNetwork</a></code> for details.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_mode">mode</code></td>
<td>
<p>a placement method from those provided in the
<code><a href="sna.html#topic+sna">sna</a></code> package: see <a href="sna.html#topic+gplot.layout">gplot.layout</a> for
details. Also accepts the names of two numeric vertex attributes of
<code>net</code>, or a matrix of numeric coordinates, in which case the first two
columns of the matrix are used.
Defaults to the Fruchterman-Reingold force-directed algorithm.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_layout.par">layout.par</code></td>
<td>
<p>options to be passed to the placement method, as listed in
<a href="sna.html#topic+gplot.layout">gplot.layout</a>.
Defaults to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_layout.exp">layout.exp</code></td>
<td>
<p>a multiplier to expand the horizontal axis if node labels
get clipped: see <a href="scales.html#topic+expand_range">expand_range</a> for details.
Defaults to <code>0</code> (no expansion).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_alpha">alpha</code></td>
<td>
<p>the level of transparency of the edges and nodes, which might be
a single value, a vertex attribute, or a vector of values.
Also accepts <code>"mode"</code> on bipartite networks (see 'Details').
Defaults to <code>1</code> (no transparency).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_color">color</code></td>
<td>
<p>the color of the nodes, which might be a single value, a vertex
attribute, or a vector of values.
Also accepts <code>"mode"</code> on bipartite networks (see 'Details').
Defaults to <code>grey75</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_shape">shape</code></td>
<td>
<p>the shape of the nodes, which might be a single value, a vertex
attribute, or a vector of values.
Also accepts <code>"mode"</code> on bipartite networks (see 'Details').
Defaults to <code>19</code> (solid circle).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size">size</code></td>
<td>
<p>the size of the nodes, in points, which might be a single value,
a vertex attribute, or a vector of values. Also accepts <code>"indegree"</code>,
<code>"outdegree"</code>, <code>"degree"</code> or <code>"freeman"</code> to size the nodes by
their unweighted degree centrality (<code>"degree"</code> and <code>"freeman"</code> are
equivalent): see <code><a href="sna.html#topic+degree">degree</a></code> for details. All node sizes must
be strictly positive.
Also accepts <code>"mode"</code> on bipartite networks (see 'Details').
Defaults to <code>9</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_max_size">max_size</code></td>
<td>
<p>the <em>maximum</em> size of the node when <code>size</code> produces
nodes of different sizes, in points.
Defaults to <code>9</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_na.rm">na.rm</code></td>
<td>
<p>whether to subset the network to nodes that are <em>not</em>
missing a given vertex attribute. If set to any vertex attribute of
<code>net</code>, the nodes for which this attribute is <code>NA</code> will be removed.
Defaults to <code>NA</code> (does nothing).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_palette">palette</code></td>
<td>
<p>the palette to color the nodes, when <code>color</code> is not a
color value or a vector of color values. Accepts named vectors of color
values, or if <a href="RColorBrewer.html#topic+ColorBrewer">RColorBrewer</a> is installed, any
ColorBrewer palette name: see <code><a href="RColorBrewer.html#topic+ColorBrewer">RColorBrewer::brewer.pal()</a></code> and
<a href="https://colorbrewer2.org/">https://colorbrewer2.org/</a> for details.
Defaults to <code>NULL</code>, which will create an array of grayscale color values
if <code>color</code> is not a color value or a vector of color values.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_alpha.palette">alpha.palette</code></td>
<td>
<p>the palette to control the transparency levels of the
nodes set by <code>alpha</code> when the levels are not numeric values.
Defaults to <code>NULL</code>, which will create an array of alpha transparency
values if <code>alpha</code> is not a numeric value or a vector of numeric values.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_alpha.legend">alpha.legend</code></td>
<td>
<p>the name to assign to the legend created by
<code>alpha</code> when its levels are not numeric values.
Defaults to <code>NA</code> (no name).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_color.palette">color.palette</code></td>
<td>
<p>see <code>palette</code></p>
</td></tr>
<tr><td><code id="ggnet2_+3A_color.legend">color.legend</code></td>
<td>
<p>the name to assign to the legend created by
<code>palette</code>.
Defaults to <code>NA</code> (no name).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_shape.palette">shape.palette</code></td>
<td>
<p>the palette to control the shapes of the nodes set by
<code>shape</code> when the shapes are not numeric values.
Defaults to <code>NULL</code>, which will create an array of shape values if
<code>shape</code> is not a numeric value or a vector of numeric values.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_shape.legend">shape.legend</code></td>
<td>
<p>the name to assign to the legend created by
<code>shape</code> when its levels are not numeric values.
Defaults to <code>NA</code> (no name).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.palette">size.palette</code></td>
<td>
<p>the palette to control the sizes of the nodes set by
<code>size</code> when the sizes are not numeric values.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.legend">size.legend</code></td>
<td>
<p>the name to assign to the legend created by
<code>size</code>.
Defaults to <code>NA</code> (no name).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.zero">size.zero</code></td>
<td>
<p>whether to accept zero-sized nodes based on the value(s) of
<code>size</code>.
Defaults to <code>FALSE</code>, which ensures that zero-sized nodes are still
shown in the plot and its size legend.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.cut">size.cut</code></td>
<td>
<p>whether to cut the size of the nodes into a certain number of
quantiles. Accepts <code>TRUE</code>, which tries to cut the sizes into quartiles,
or any positive numeric value, which tries to cut the sizes into that many
quantiles. If the size of the nodes do not contain the specified number of
distinct quantiles, the largest possible number is used.
See <code><a href="stats.html#topic+quantile">quantile</a></code> and <code><a href="base.html#topic+cut">cut</a></code> for details.
Defaults to <code>FALSE</code> (does nothing).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.min">size.min</code></td>
<td>
<p>whether to subset the network to nodes with a minimum size,
based on the values of <code>size</code>.
Defaults to <code>NA</code> (preserves all nodes).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.max">size.max</code></td>
<td>
<p>whether to subset the network to nodes with a maximum size,
based on the values of <code>size</code>.
Defaults to <code>NA</code> (preserves all nodes).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_label">label</code></td>
<td>
<p>whether to label the nodes. If set to <code>TRUE</code>, nodes are
labeled with their vertex names. If set to a vector that contains as many
elements as there are nodes in <code>net</code>, nodes are labeled with these. If
set to any other vector of values, the nodes are labeled only when their
vertex name matches one of these values.
Defaults to <code>FALSE</code> (no labels).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_label.alpha">label.alpha</code></td>
<td>
<p>the level of transparency of the node labels, as a
numeric value, a vector of numeric values, or as a vertex attribute
containing numeric values.
Defaults to <code>1</code> (no transparency).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_label.color">label.color</code></td>
<td>
<p>the color of the node labels, as a color value, a vector
of color values, or as a vertex attribute containing color values.
Defaults to <code>"black"</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_label.size">label.size</code></td>
<td>
<p>the size of the node labels, in points, as a numeric value,
a vector of numeric values, or as a vertex attribute containing numeric
values.
Defaults to <code>max_size / 2</code> (half the maximum node size), which defaults
to <code>4.5</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_label.trim">label.trim</code></td>
<td>
<p>whether to apply some trimming to the node labels. Accepts
any function that can process a character vector, or a strictly positive
numeric value, in which case the labels are trimmed to a fixed-length
substring of that length: see <code><a href="base.html#topic+substr">substr</a></code> for details.
Defaults to <code>FALSE</code> (does nothing).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_node.alpha">node.alpha</code></td>
<td>
<p>see <code>alpha</code></p>
</td></tr>
<tr><td><code id="ggnet2_+3A_node.color">node.color</code></td>
<td>
<p>see <code>color</code></p>
</td></tr>
<tr><td><code id="ggnet2_+3A_node.label">node.label</code></td>
<td>
<p>see <code>label</code></p>
</td></tr>
<tr><td><code id="ggnet2_+3A_node.shape">node.shape</code></td>
<td>
<p>see <code>shape</code></p>
</td></tr>
<tr><td><code id="ggnet2_+3A_node.size">node.size</code></td>
<td>
<p>see <code>size</code></p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.alpha">edge.alpha</code></td>
<td>
<p>the level of transparency of the edges.
Defaults to the value of <code>alpha</code>, which defaults to <code>1</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.color">edge.color</code></td>
<td>
<p>the color of the edges, as a color value, a vector of color
values, or as an edge attribute containing color values.
Defaults to <code>"grey50"</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.lty">edge.lty</code></td>
<td>
<p>the linetype of the edges, as a linetype value, a vector of
linetype values, or as an edge attribute containing linetype values.
Defaults to <code>"solid"</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.size">edge.size</code></td>
<td>
<p>the size of the edges, in points, as a numeric value, a
vector of numeric values, or as an edge attribute containing numeric values.
All edge sizes must be strictly positive.
Defaults to <code>0.25</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.label">edge.label</code></td>
<td>
<p>the labels to plot at the middle of the edges, as a single
value, a vector of values, or as an edge attribute.
Defaults to <code>NULL</code> (no edge labels).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.label.alpha">edge.label.alpha</code></td>
<td>
<p>the level of transparency of the edge labels, as a
numeric value, a vector of numeric values, or as an edge attribute
containing numeric values.
Defaults to <code>1</code> (no transparency).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.label.color">edge.label.color</code></td>
<td>
<p>the color of the edge labels, as a color value, a
vector of color values, or as an edge attribute containing color values.
Defaults to <code>label.color</code>, which defaults to <code>"black"</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.label.fill">edge.label.fill</code></td>
<td>
<p>the background color of the edge labels.
Defaults to <code>"white"</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.label.size">edge.label.size</code></td>
<td>
<p>the size of the edge labels, in points, as a numeric
value, a vector of numeric values, or as an edge attribute containing numeric
values. All edge label sizes must be strictly positive.
Defaults to <code>max_size / 2</code> (half the maximum node size), which defaults
to <code>4.5</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_arrow.size">arrow.size</code></td>
<td>
<p>the size of the arrows for directed network edges, in
points. See <code><a href="grid.html#topic+arrow">arrow</a></code> for details.
Defaults to <code>0</code> (no arrows).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_arrow.gap">arrow.gap</code></td>
<td>
<p>a setting aimed at improving the display of edge arrows by
plotting slightly shorter edges. Accepts any value between <code>0</code> and
<code>1</code>, where a value of <code>0.05</code> will generally achieve good results
when the size of the nodes is reasonably small.
Defaults to <code>0</code> (no shortening).</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_arrow.type">arrow.type</code></td>
<td>
<p>the type of the arrows for directed network edges. See
<code><a href="grid.html#topic+arrow">arrow</a></code> for details.
Defaults to <code>"closed"</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_legend.size">legend.size</code></td>
<td>
<p>the size of the legend symbols and text, in points.
Defaults to <code>9</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_legend.position">legend.position</code></td>
<td>
<p>the location of the plot legend(s). Accepts all
<code>legend.position</code> values supported by <code><a href="ggplot2.html#topic+theme">theme</a></code>.
Defaults to <code>"right"</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_...">...</code></td>
<td>
<p>other arguments passed to the <code>geom_text</code> object that sets
the node labels: see <code><a href="ggplot2.html#topic+geom_text">geom_text</a></code> for details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The degree centrality measures that can be produced through the
<code>size</code> argument will take the directedness of the network into account,
but will be unweighted. To compute weighted network measures, see the
<code>tnet</code> package by Tore Opsahl (<code>help("tnet", package = "tnet")</code>).
</p>
<p>The nodes of bipartite networks can be mapped to their mode by passing the
<code>"mode"</code> argument to any of <code>alpha</code>, <code>color</code>, <code>shape</code> and
<code>size</code>, in which case the nodes of the primary mode will be mapped as
<code>"actor"</code>, and the nodes of the secondary mode will be mapped as
<code>"event"</code>.
</p>


<h3>Author(s)</h3>

<p>Moritz Marbach and Francois Briatte, with help from Heike Hofmann,
Pedro Jordano and Ming-Yu Liu
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggnet">ggnet</a></code> in this package,
<code><a href="sna.html#topic+gplot">gplot</a></code> in the <code><a href="sna.html#topic+sna">sna</a></code> package, and
<code><a href="network.html#topic+plot.network">plot.network</a></code> in the <code><a href="network.html#topic+network">network</a></code>
package
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

library(network)

# random adjacency matrix
x           &lt;- 10
ndyads      &lt;- x * (x - 1)
density     &lt;- x / ndyads
m           &lt;- matrix(0, nrow = x, ncol = x)
dimnames(m) &lt;- list(letters[1:x], letters[1:x])
m[row(m) != col(m)] &lt;- runif(ndyads) &lt; density
m

# random undirected network
n &lt;- network::network(m, directed = FALSE)
n

p_(ggnet2(n, label = TRUE))
p_(ggnet2(n, label = TRUE, shape = 15))
p_(ggnet2(n, label = TRUE, shape = 15, color = "black", label.color = "white"))

# add vertex attribute
x = network.vertex.names(n)
x = ifelse(x %in% c("a", "e", "i"), "vowel", "consonant")
n %v% "phono" = x

p_(ggnet2(n, color = "phono"))
p_(ggnet2(n, color = "phono", palette = c("vowel" = "gold", "consonant" = "grey")))
p_(ggnet2(n, shape = "phono", color = "phono"))

if (require(RColorBrewer)) {

  # random groups
  n %v% "group" &lt;- sample(LETTERS[1:3], 10, replace = TRUE)

  p_(ggnet2(n, color = "group", palette = "Set2"))

}

# random weights
n %e% "weight" &lt;- sample(1:3, network.edgecount(n), replace = TRUE)
p_(ggnet2(n, edge.size = "weight", edge.label = "weight"))

# edge arrows on a directed network
p_(ggnet2(network(m, directed = TRUE), arrow.gap = 0.05, arrow.size = 10))

# Padgett's Florentine wedding data
data(flo, package = "network")
flo

p_(ggnet2(flo, label = TRUE))
p_(ggnet2(flo, label = TRUE, label.trim = 4, vjust = -1, size = 3, color = 1))
p_(ggnet2(flo, label = TRUE, size = 12, color = "white"))
</code></pre>

<hr>
<h2 id='ggnetworkmap'>Network plot map overlay</h2><span id='topic+ggnetworkmap'></span>

<h3>Description</h3>

<p>Plots a network with <span class="pkg">ggplot2</span> suitable for overlay on a <span class="pkg">ggmap</span> plot or <span class="pkg">ggplot2</span>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggnetworkmap(
  gg,
  net,
  size = 3,
  alpha = 0.75,
  weight,
  node.group,
  node.color = NULL,
  node.alpha = NULL,
  ring.group,
  segment.alpha = NULL,
  segment.color = "grey",
  great.circles = FALSE,
  segment.size = 0.25,
  arrow.size = 0,
  label.nodes = FALSE,
  label.size = size/2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggnetworkmap_+3A_gg">gg</code></td>
<td>
<p>an object of class <code>ggplot</code>.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_net">net</code></td>
<td>
<p>an object of class <code><a href="network.html#topic+network">network</a></code>, or any object
that can be coerced to this class, such as an adjacency or incidence matrix,
or an edge list: see <a href="network.html#topic+edgeset.constructors">edgeset.constructors</a> and
<a href="network.html#topic+network">network</a> for details. If the object is of class
<a href="igraph.html#topic+aaa-igraph-package">igraph</a> and the
<a href="intergraph.html#topic+intergraph-package">intergraph</a> package is installed,
it will be used to convert the object: see
<code><a href="intergraph.html#topic+asNetwork">asNetwork</a></code> for details.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_size">size</code></td>
<td>
<p>size of the network nodes. Defaults to 3. If the nodes are weighted, their area is proportionally scaled up to the size set by <code>size</code>.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_alpha">alpha</code></td>
<td>
<p>a level of transparency for nodes, vertices and arrows. Defaults to 0.75.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_weight">weight</code></td>
<td>
<p>if present, the unquoted name of a vertex attribute in <code>data</code>.  Otherwise nodes are unweighted.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_node.group">node.group</code></td>
<td>
<p><code>NULL</code>, the default, or the unquoted name of a vertex attribute that will be used to determine the color of each node.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_node.color">node.color</code></td>
<td>
<p>If <code>node.group</code> is null, a character string specifying a color.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_node.alpha">node.alpha</code></td>
<td>
<p>transparency of the nodes. Inherits from <code>alpha</code>.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_ring.group">ring.group</code></td>
<td>
<p>if not <code>NULL</code>, the default, the unquoted name of a vertex attribute that will be used to determine the color of each node border.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_segment.alpha">segment.alpha</code></td>
<td>
<p>transparency of the vertex links. Inherits from <code>alpha</code></p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_segment.color">segment.color</code></td>
<td>
<p>color of the vertex links. Defaults to <code>"grey"</code>.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_great.circles">great.circles</code></td>
<td>
<p>whether to draw edges as great circles using the <code>geosphere</code> package.  Defaults to <code>FALSE</code></p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_segment.size">segment.size</code></td>
<td>
<p>size of the vertex links, as a vector of values or as a single value. Defaults to 0.25.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_arrow.size">arrow.size</code></td>
<td>
<p>size of the vertex arrows for directed network plotting, in centimeters. Defaults to 0.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_label.nodes">label.nodes</code></td>
<td>
<p>label nodes with their vertex names attribute. If set to <code>TRUE</code>, all nodes are labelled. Also accepts a vector of character strings to match with vertex names.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_label.size">label.size</code></td>
<td>
<p>size of the labels.  Defaults to <code>size / 2</code>.</p>
</td></tr>
<tr><td><code id="ggnetworkmap_+3A_...">...</code></td>
<td>
<p>other arguments supplied to geom_text for the node labels. Arguments pertaining to the title or other items can be achieved through <span class="pkg">ggplot2</span> methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a descendant of the original <code>ggnet</code> function.  <code>ggnet</code> added the innovation of plotting the network geographically.
However, <code>ggnet</code> needed to be the first object in the ggplot chain.  <code>ggnetworkmap</code> does not.  If passed a <code>ggplot</code> object as its first argument,
such as output from <code>ggmap</code>, <code>ggnetworkmap</code> will plot on top of that chart, looking for vertex attributes <code>lon</code> and <code>lat</code> as coordinates.
Otherwise, <code>ggnetworkmap</code> will generate coordinates using the Fruchterman-Reingold algorithm.
</p>
<p>This is a function for plotting graphs generated by <code>network</code> or <code>igraph</code> in a more flexible and elegant manner than permitted by ggnet.  The function does not need to be the first plot in the ggplot chain, so the graph can be plotted on top of a map or other chart.  Segments can be straight lines, or plotted as great circles.  Note that the great circles feature can produce odd results with arrows and with vertices beyond the plot edges; this is a <span class="pkg">ggplot2</span> limitation and cannot yet be fixed.  Nodes can have two color schemes, which are then plotted as the center and ring around the node.  The color schemes are selected by adding scale_fill_ or scale_color_ just like any other <span class="pkg">ggplot2</span> plot. If there are no rings, scale_color sets the color of the nodes.  If there are rings, scale_color sets the color of the rings, and scale_fill sets the color of the centers.  Note that additional arguments in the ... are passed to geom_text for plotting labels.
</p>


<h3>Author(s)</h3>

<p>Amos Elberg. Original by Moritz Marbach, Francois Briatte
</p>


<h3>Examples</h3>

<pre><code class='language-R'># small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

invisible(lapply(c("ggplot2", "maps", "network", "sna"), base::library, character.only = TRUE))

## Example showing great circles on a simple map of the USA
## http://flowingdata.com/2011/05/11/how-to-map-connections-with-great-circles/

airports &lt;- read.csv("http://datasets.flowingdata.com/tuts/maparcs/airports.csv", header = TRUE)
rownames(airports) &lt;- airports$iata

# select some random flights
set.seed(123)
flights &lt;- data.frame(
  origin = sample(airports[200:400, ]$iata, 200, replace = TRUE),
  destination = sample(airports[200:400, ]$iata, 200, replace = TRUE)
)

# convert to network
flights &lt;- network(flights, directed = TRUE)

# add geographic coordinates
flights %v% "lat" &lt;- airports[network.vertex.names(flights), "lat"]
flights %v% "lon" &lt;- airports[network.vertex.names(flights), "long"]

# drop isolated airports
delete.vertices(flights, which(degree(flights) &lt; 2))

# compute degree centrality
flights %v% "degree" &lt;- degree(flights, gmode = "digraph")

# add random groups
flights %v% "mygroup" &lt;- sample(letters[1:4], network.size(flights), replace = TRUE)

# create a map of the USA
usa &lt;- ggplot(map_data("usa"), aes(x = long, y = lat)) +
  geom_polygon(aes(group = group),
    color = "grey65",
    fill = "#f9f9f9", linewidth = 0.2
  )

# overlay network data to map
p &lt;- ggnetworkmap(
  usa, flights,
  size = 4, great.circles = TRUE,
  node.group = mygroup, segment.color = "steelblue",
  ring.group = degree, weight = degree
)
p_(p)

## Exploring a community of spambots found on Twitter
## Data by Amos Elberg: see ?twitter_spambots for details

data(twitter_spambots)

# create a world map
world &lt;- fortify(map("world", plot = FALSE, fill = TRUE))
world &lt;- ggplot(world, aes(x = long, y = lat)) +
  geom_polygon(aes(group = group),
    color = "grey65",
    fill = "#f9f9f9", linewidth = 0.2
  )

# view global structure
p &lt;- ggnetworkmap(world, twitter_spambots)
p_(p)

# domestic distribution
p &lt;- ggnetworkmap(net = twitter_spambots)
p_(p)

# topology
p &lt;- ggnetworkmap(net = twitter_spambots, arrow.size = 0.5)
p_(p)

# compute indegree and outdegree centrality
twitter_spambots %v% "indegree" &lt;- degree(twitter_spambots, cmode = "indegree")
twitter_spambots %v% "outdegree" &lt;- degree(twitter_spambots, cmode = "outdegree")

p &lt;- ggnetworkmap(
  net = twitter_spambots,
  arrow.size = 0.5,
  node.group = indegree,
  ring.group = outdegree, size = 4
) +
  scale_fill_continuous("Indegree", high = "red", low = "yellow") +
  labs(color = "Outdegree")
p_(p)

# show some vertex attributes associated with each account
p &lt;- ggnetworkmap(
  net = twitter_spambots,
  arrow.size = 0.5,
  node.group = followers,
  ring.group = friends,
  size = 4,
  weight = indegree,
  label.nodes = TRUE, vjust = -1.5
) +
  scale_fill_continuous("Followers", high = "red", low = "yellow") +
  labs(color = "Friends") +
  scale_color_continuous(low = "lightgreen", high = "darkgreen")
p_(p)


</code></pre>

<hr>
<h2 id='ggnostic'>Plot matrix of statistical model diagnostics</h2><span id='topic+ggnostic'></span>

<h3>Description</h3>

<p>Plot matrix of statistical model diagnostics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggnostic(
  model,
  ...,
  columnsX = attr(data, "var_x"),
  columnsY = c(".resid", ".sigma", ".hat", ".cooksd"),
  columnLabelsX = attr(data, "var_x_label"),
  columnLabelsY = gsub("\\.", " ", gsub("^\\.", "", columnsY)),
  xlab = "explanatory variables",
  ylab = "diagnostics",
  title = paste(deparse(model$call, width.cutoff = 500L), collapse = "\n"),
  continuous = list(default = ggally_points, .fitted = ggally_points, .se.fit =
    ggally_nostic_se_fit, .resid = ggally_nostic_resid, .hat = ggally_nostic_hat, .sigma
    = ggally_nostic_sigma, .cooksd = ggally_nostic_cooksd, .std.resid =
    ggally_nostic_std_resid),
  combo = list(default = ggally_box_no_facet, .fitted = ggally_box_no_facet, .se.fit =
    ggally_nostic_se_fit, .resid = ggally_nostic_resid, .hat = ggally_nostic_hat, .sigma
    = ggally_nostic_sigma, .cooksd = ggally_nostic_cooksd, .std.resid =
    ggally_nostic_std_resid),
  discrete = list(default = ggally_ratio, .fitted = ggally_ratio, .se.fit = ggally_ratio,
    .resid = ggally_ratio, .hat = ggally_ratio, .sigma = ggally_ratio, .cooksd =
    ggally_ratio, .std.resid = ggally_ratio),
  progress = NULL,
  data = broomify(model)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggnostic_+3A_model">model</code></td>
<td>
<p>statistical model object such as output from <code>stats::<a href="stats.html#topic+lm">lm</a></code> or <code>stats::<a href="stats.html#topic+glm">glm</a></code></p>
</td></tr>
<tr><td><code id="ggnostic_+3A_...">...</code></td>
<td>
<p>arguments passed directly to <code><a href="#topic+ggduo">ggduo</a></code></p>
</td></tr>
<tr><td><code id="ggnostic_+3A_columnsx">columnsX</code></td>
<td>
<p>columns to be displayed in the plot matrix. Defaults to the predictor columns of the <code>model</code></p>
</td></tr>
<tr><td><code id="ggnostic_+3A_columnsy">columnsY</code></td>
<td>
<p>rows to be displayed in the plot matrix. Defaults to residuals, leave one out sigma value, diagonal of the hat matrix, and Cook's Distance. The possible values are the response variables in the model and the added columns provided by <code><a href="broom.html#topic+reexports">broom::augment()</a></code>. See details for more information.</p>
</td></tr>
<tr><td><code id="ggnostic_+3A_columnlabelsx">columnLabelsX</code>, <code id="ggnostic_+3A_columnlabelsy">columnLabelsY</code></td>
<td>
<p>column and row labels to display in the plot matrix</p>
</td></tr>
<tr><td><code id="ggnostic_+3A_xlab">xlab</code>, <code id="ggnostic_+3A_ylab">ylab</code>, <code id="ggnostic_+3A_title">title</code></td>
<td>
<p>plot matrix labels passed directly to <code><a href="#topic+ggmatrix">ggmatrix</a></code></p>
</td></tr>
<tr><td><code id="ggnostic_+3A_continuous">continuous</code>, <code id="ggnostic_+3A_combo">combo</code>, <code id="ggnostic_+3A_discrete">discrete</code></td>
<td>
<p>list of functions for each y variable.  See details for more information.</p>
</td></tr>
<tr><td><code id="ggnostic_+3A_progress">progress</code></td>
<td>
<p><code>NULL</code> (default) for a progress bar in interactive sessions with more than 15 plots, <code>TRUE</code> for a progress bar, <code>FALSE</code> for no progress bar, or a function that accepts at least a plot matrix and returns a new <code>progress::<a href="progress.html#topic+progress_bar">progress_bar</a></code>.  See <code><a href="#topic+ggmatrix_progress">ggmatrix_progress</a></code>.</p>
</td></tr>
<tr><td><code id="ggnostic_+3A_data">data</code></td>
<td>
<p>data defaults to a 'broomify'ed model object.  This object will contain information about the X variables, Y variables, and multiple broom outputs. See <code><a href="#topic+broomify">broomify</a>(model)</code> for more information</p>
</td></tr>
</table>


<h3><code>columnsY</code></h3>

<p><code><a href="broom.html#topic+reexports">broom::augment()</a></code> collects data from the supplied model and returns a data.frame with the following columns (taken directly from broom documentation).  These columns are the only allowed values in the <code>columnsY</code> parameter to <code><a href="#topic+ggnostic">ggnostic</a></code>.
</p>

<dl>
<dt>.resid</dt><dd><p>Residuals</p>
</dd>
<dt>.hat</dt><dd><p>Diagonal of the hat matrix</p>
</dd>
<dt>.sigma</dt><dd><p>Estimate of residual standard deviation when
corresponding observation is dropped from model</p>
</dd>
<dt>.cooksd</dt><dd><p>Cooks distance, <code><a href="stats.html#topic+influence.measures">stats::cooks.distance()</a></code></p>
</dd>
<dt>.fitted</dt><dd><p>Fitted values of model</p>
</dd>
<dt>.se.fit</dt><dd><p>Standard errors of fitted values</p>
</dd>
<dt>.std.resid</dt><dd><p>Standardized residuals</p>
</dd>
<dt>response variable name</dt><dd><p>The response variable in the model may be added. Such as <code>"mpg"</code> in the model <code>lm(mpg ~ ., data = mtcars)</code></p>
</dd>
</dl>



<h3><code>continuous</code>, <code>combo</code>, <code>discrete</code> types</h3>

<p>Similar to <code><a href="#topic+ggduo">ggduo</a></code> and <code><a href="#topic+ggpairs">ggpairs</a></code>, functions may be supplied to display the different column types.  However, since the Y rows are fixed, each row has it's own corresponding function in each of the plot types: continuous, combo, and discrete.  Each plot type list can have keys that correspond to the <code><a href="broom.html#topic+reexports">broom::augment()</a></code> output: <code>".fitted"</code>, <code>".resid"</code>, <code>".std.resid"</code>, <code>".sigma"</code>, <code>".se.fit"</code>, <code>".hat"</code>, <code>".cooksd"</code>. An extra key, <code>"default"</code>, is used to plot the response variables of the model if they are included.  Having a function for each diagnostic allows for very fine control over the diagnostics plot matrix.  The functions for each type list are wrapped into a switch function that calls the function corresponding to the y variable being plotted.  These switch functions are then passed directly to the <code>types</code> parameter in <code><a href="#topic+ggduo">ggduo</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive
data(mtcars)

# use mtcars dataset and alter the 'am' column to display actual name values
mtc &lt;- mtcars
mtc$am &lt;- c("0" = "automatic", "1" = "manual")[as.character(mtc$am)]

# step the complete model down to a smaller model
mod &lt;- stats::step(stats::lm(mpg ~ ., data = mtc), trace = FALSE)

# display using defaults
pm &lt;- ggnostic(mod)
p_(pm)

# color by am value
pm &lt;- ggnostic(mod, mapping = ggplot2::aes(color = am))
p_(pm)

# turn resid smooth error ribbon off
pm &lt;- ggnostic(mod, continuous = list(.resid = wrap("nostic_resid", se = FALSE)))
p_(pm)


## plot residuals vs fitted in a ggpairs plot matrix
dt &lt;- broomify(mod)
pm &lt;- ggpairs(
  dt, c(".fitted", ".resid"),
  columnLabels = c("fitted", "residuals"),
  lower = list(continuous = ggally_nostic_resid)
)
p_(pm)
</code></pre>

<hr>
<h2 id='ggpairs'>ggplot2 generalized pairs plot</h2><span id='topic+ggpairs'></span>

<h3>Description</h3>

<p>Make a matrix of plots with a given data set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggpairs(
  data,
  mapping = NULL,
  columns = 1:ncol(data),
  title = NULL,
  upper = list(continuous = "cor", combo = "box_no_facet", discrete = "count", na = "na"),
  lower = list(continuous = "points", combo = "facethist", discrete = "facetbar", na =
    "na"),
  diag = list(continuous = "densityDiag", discrete = "barDiag", na = "naDiag"),
  params = NULL,
  ...,
  xlab = NULL,
  ylab = NULL,
  axisLabels = c("show", "internal", "none"),
  columnLabels = colnames(data[columns]),
  labeller = "label_value",
  switch = NULL,
  showStrips = NULL,
  legend = NULL,
  cardinality_threshold = 15,
  progress = NULL,
  proportions = NULL,
  legends = stop("deprecated")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggpairs_+3A_data">data</code></td>
<td>
<p>data set using.  Can have both numerical and categorical data.</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_mapping">mapping</code></td>
<td>
<p>aesthetic mapping (besides <code>x</code> and <code>y</code>).  See <code><a href="ggplot2.html#topic+aes">aes</a>()</code>.  If <code>mapping</code> is numeric, <code>columns</code> will be set to the <code>mapping</code> value and <code>mapping</code> will be set to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_columns">columns</code></td>
<td>
<p>which columns are used to make plots.  Defaults to all columns.</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_title">title</code>, <code id="ggpairs_+3A_xlab">xlab</code>, <code id="ggpairs_+3A_ylab">ylab</code></td>
<td>
<p>title, x label, and y label for the graph</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_upper">upper</code></td>
<td>
<p>see Details</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_lower">lower</code></td>
<td>
<p>see Details</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_diag">diag</code></td>
<td>
<p>see Details</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_params">params</code></td>
<td>
<p>deprecated.  Please see <code><a href="#topic+wrap_fn_with_param_arg">wrap_fn_with_param_arg</a></code></p>
</td></tr>
<tr><td><code id="ggpairs_+3A_...">...</code></td>
<td>
<p>deprecated. Please use <code>mapping</code></p>
</td></tr>
<tr><td><code id="ggpairs_+3A_axislabels">axisLabels</code></td>
<td>
<p>either &quot;show&quot; to display axisLabels, &quot;internal&quot; for labels in the diagonal plots, or &quot;none&quot; for no axis labels</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_columnlabels">columnLabels</code></td>
<td>
<p>label names to be displayed.  Defaults to names of columns being used.</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_labeller">labeller</code></td>
<td>
<p>labeller for facets. See <code><a href="ggplot2.html#topic+labellers">labellers</a></code>. Common values are <code>"label_value"</code> (default) and <code>"label_parsed"</code>.</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_switch">switch</code></td>
<td>
<p>switch parameter for facet_grid. See <code>ggplot2::<a href="ggplot2.html#topic+facet_grid">facet_grid</a></code>.  By default, the labels are displayed on the top and right of the plot. If <code>"x"</code>, the top labels will be displayed to the bottom. If <code>"y"</code>, the right-hand side labels will be displayed to the left. Can also be set to <code>"both"</code></p>
</td></tr>
<tr><td><code id="ggpairs_+3A_showstrips">showStrips</code></td>
<td>
<p>boolean to determine if each plot's strips should be displayed. <code>NULL</code> will default to the top and right side plots only. <code>TRUE</code> or <code>FALSE</code> will turn all strips on or off respectively.</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_legend">legend</code></td>
<td>
<p>May be the two objects described below or the default <code>NULL</code> value.  The legend position can be moved by using ggplot2's theme element <code>pm + theme(legend.position = "bottom")</code> </p>
<dl>
<dt>a numeric vector of length 2</dt><dd><p>provides the location of the plot to use the legend for the plot matrix's legend. Such as <code>legend = c(3,5)</code> which will use the legend from the plot in the third row and fifth column</p>
</dd><dt>a single numeric value</dt><dd><p>provides the location of a plot according to the display order. Such as <code>legend = 3</code> in a plot matrix with 2 rows and 5 columns displayed by column will return the plot in position <code>c(1,2)</code></p>
</dd><dt>a object from <code><a href="#topic+grab_legend">grab_legend</a>()</code></dt><dd><p>a predetermined plot legend that will be displayed directly</p>
</dd></dl>
</td></tr>
<tr><td><code id="ggpairs_+3A_cardinality_threshold">cardinality_threshold</code></td>
<td>
<p>maximum number of levels allowed in a character / factor column.  Set this value to NULL to not check factor columns. Defaults to 15</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_progress">progress</code></td>
<td>
<p><code>NULL</code> (default) for a progress bar in interactive sessions with more than 15 plots, <code>TRUE</code> for a progress bar, <code>FALSE</code> for no progress bar, or a function that accepts at least a plot matrix and returns a new <code>progress::<a href="progress.html#topic+progress_bar">progress_bar</a></code>.  See <code><a href="#topic+ggmatrix_progress">ggmatrix_progress</a></code>.</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_proportions">proportions</code></td>
<td>
<p>Value to change how much area is given for each plot. Either <code>NULL</code> (default), numeric value matching respective length, <code>grid::<a href="grid.html#topic+unit">unit</a></code> object with matching respective length or <code>"auto"</code> for automatic relative proportions based on the number of levels for categorical variables.</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_legends">legends</code></td>
<td>
<p>deprecated</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>upper</code> and <code>lower</code> are lists that may contain the variables
'continuous', 'combo', 'discrete', and 'na'. Each element of the list may be a function or a string.  If a string is supplied, it must be a character string representing the tail end of a <code>ggally_NAME</code> function. The list of current valid <code>ggally_NAME</code> functions is visible in a dedicated vignette.
</p>

<dl>
<dt>continuous</dt><dd><p>This option is used for continuous X and Y data.</p>
</dd>
<dt>combo</dt><dd><p>This option is used for either continuous X and categorical Y data or categorical X and continuous Y data.</p>
</dd>
<dt>discrete</dt><dd><p>This option is used for categorical X and Y data.</p>
</dd>
<dt>na</dt><dd><p>This option is used when all X data is <code>NA</code>, all Y data is <code>NA</code>, or either all X or Y data is <code>NA</code>.</p>
</dd>
</dl>

<p><code>diag</code> is a list that may only contain the variables 'continuous', 'discrete', and 'na'. Each element of the diag list is a string implementing the following options:
</p>

<dl>
<dt>continuous</dt><dd><p>exactly one of ('densityDiag', 'barDiag', 'blankDiag'). This option is used for continuous X data.</p>
</dd>
<dt>discrete</dt><dd><p>exactly one of ('barDiag', 'blankDiag'). This option is used for categorical X and Y data.</p>
</dd>
<dt>na</dt><dd><p>exactly one of ('naDiag', 'blankDiag').  This option is used when all X data is <code>NA</code>.</p>
</dd>
</dl>

<p>If 'blank' is ever chosen as an option, then ggpairs will produce an empty plot.
</p>
<p>If a function is supplied as an option to <code>upper</code>, <code>lower</code>, or <code>diag</code>, it should implement the function api of <code>function(data, mapping, ...){#make ggplot2 plot}</code>.  If a specific function needs its parameters set, <code><a href="#topic+wrap">wrap</a>(fn, param1 = val1, param2 = val2)</code> the function with its parameters.
</p>


<h3>Value</h3>

<p><code><a href="#topic+ggmatrix">ggmatrix</a></code> object that if called, will print
</p>


<h3>Author(s)</h3>

<p>Barret Schloerke, Jason Crowley, Di Cook, Heike Hofmann, Hadley Wickham
</p>


<h3>References</h3>

<p>John W Emerson, Walton A Green, Barret Schloerke, Jason Crowley, Dianne Cook, Heike Hofmann, Hadley Wickham. The Generalized Pairs Plot. Journal of Computational and Graphical Statistics, vol. 22, no. 1, pp. 79-91, 2012.
</p>


<h3>See Also</h3>

<p>wrap v1_ggmatrix_theme
</p>


<h3>Examples</h3>

<pre><code class='language-R'># small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive


## Quick example, with and without colour
data(flea)
ggpairs(flea, columns = 2:4)
pm &lt;- ggpairs(flea, columns = 2:4, ggplot2::aes(colour = species))
p_(pm)
# Note: colour should be categorical, else you will need to reset
# the upper triangle to use points instead of trying to compute corr

data(tips)
pm &lt;- ggpairs(tips[, 1:3])
p_(pm)
pm &lt;- ggpairs(tips, 1:3, columnLabels = c("Total Bill", "Tip", "Sex"))
p_(pm)
pm &lt;- ggpairs(tips, upper = "blank")
p_(pm)

## Plot Types
# Change default plot behavior
pm &lt;- ggpairs(
  tips[, c(1, 3, 4, 2)],
  upper = list(continuous = "density", combo = "box_no_facet"),
  lower = list(continuous = "points", combo = "dot_no_facet")
)
p_(pm)
# Supply Raw Functions (may be user defined functions!)
pm &lt;- ggpairs(
  tips[, c(1, 3, 4, 2)],
  upper = list(continuous = ggally_density, combo = ggally_box_no_facet),
  lower = list(continuous = ggally_points, combo = ggally_dot_no_facet)
)
p_(pm)

# Use sample of the diamonds data
data(diamonds, package = "ggplot2")
diamonds.samp &lt;- diamonds[sample(1:dim(diamonds)[1], 1000), ]

# Different aesthetics for different plot sections and plot types
pm &lt;- ggpairs(
  diamonds.samp[, 1:5],
  mapping = ggplot2::aes(color = cut),
  upper = list(continuous = wrap("density", alpha = 0.5), combo = "box_no_facet"),
  lower = list(continuous = wrap("points", alpha = 0.3), combo = wrap("dot_no_facet", alpha = 0.4)),
  title = "Diamonds"
)
p_(pm)

## Axis Label Variations
# Only Variable Labels on the diagonal (no axis labels)
pm &lt;- ggpairs(tips[, 1:3], axisLabels = "internal")
p_(pm)
# Only Variable Labels on the outside (no axis labels)
pm &lt;- ggpairs(tips[, 1:3], axisLabels = "none")
p_(pm)

## Facet Label Variations
#  Default:
df_x &lt;- rnorm(100)
df_y &lt;- df_x + rnorm(100, 0, 0.1)
df &lt;- data.frame(x = df_x, y = df_y, c = sqrt(df_x^2 + df_y^2))
pm &lt;- ggpairs(
  df,
  columnLabels = c("alpha[foo]", "alpha[bar]", "sqrt(alpha[foo]^2 + alpha[bar]^2)")
)
p_(pm)
#  Parsed labels:
pm &lt;- ggpairs(
  df,
  columnLabels = c("alpha[foo]", "alpha[bar]", "sqrt(alpha[foo]^2 + alpha[bar]^2)"),
  labeller = "label_parsed"
)
p_(pm)

## Plot Insertion Example
custom_car &lt;- ggpairs(mtcars[, c("mpg", "wt", "cyl")], upper = "blank", title = "Custom Example")
# ggplot example taken from example(geom_text)
plot &lt;- ggplot2::ggplot(mtcars, ggplot2::aes(x = wt, y = mpg, label = rownames(mtcars)))
plot &lt;- plot +
  ggplot2::geom_text(ggplot2::aes(colour = factor(cyl)), size = 3) +
  ggplot2::scale_colour_discrete(l = 40)
custom_car[1, 2] &lt;- plot
personal_plot &lt;- ggally_text(
  "ggpairs allows you\nto put in your\nown plot.\nLike that one.\n &lt;---"
)
custom_car[1, 3] &lt;- personal_plot
p_(custom_car)

## Remove binwidth warning from ggplot2
# displays warning about picking a better binwidth
pm &lt;- ggpairs(tips, 2:3)
p_(pm)
# no warning displayed
pm &lt;- ggpairs(tips, 2:3, lower = list(combo = wrap("facethist", binwidth = 0.5)))
p_(pm)
# no warning displayed with user supplied function
pm &lt;- ggpairs(tips, 2:3, lower = list(combo = wrap(ggally_facethist, binwidth = 0.5)))
p_(pm)

## Remove panel grid lines from correlation plots
pm &lt;- ggpairs(
  flea,
  columns = 2:4,
  upper = list(continuous = wrap(ggally_cor, displayGrid = FALSE))
)
p_(pm)

## Custom with/height of subplots
pm &lt;- ggpairs(tips, columns = c(2, 3, 5))
p_(pm)

pm &lt;- ggpairs(tips, columns = c(2, 3, 5), proportions = "auto")
p_(pm)

pm &lt;- ggpairs(tips, columns = c(2, 3, 5), proportions = c(1, 3, 2))
p_(pm)

</code></pre>

<hr>
<h2 id='ggparcoord'>Parallel coordinate plot</h2><span id='topic+ggparcoord'></span>

<h3>Description</h3>

<p>A function for plotting static parallel coordinate plots, utilizing
the <code>ggplot2</code> graphics package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggparcoord(
  data,
  columns = 1:ncol(data),
  groupColumn = NULL,
  scale = "std",
  scaleSummary = "mean",
  centerObsID = 1,
  missing = "exclude",
  order = columns,
  showPoints = FALSE,
  splineFactor = FALSE,
  alphaLines = 1,
  boxplot = FALSE,
  shadeBox = NULL,
  mapping = NULL,
  title = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggparcoord_+3A_data">data</code></td>
<td>
<p>the dataset to plot</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_columns">columns</code></td>
<td>
<p>a vector of variables (either names or indices) to be axes in the plot</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_groupcolumn">groupColumn</code></td>
<td>
<p>a single variable to group (color) by</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_scale">scale</code></td>
<td>
<p>method used to scale the variables (see Details)</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_scalesummary">scaleSummary</code></td>
<td>
<p>if scale==&quot;center&quot;, summary statistic to univariately
center each variable by</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_centerobsid">centerObsID</code></td>
<td>
<p>if scale==&quot;centerObs&quot;, row number of case plot should
univariately be centered on</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_missing">missing</code></td>
<td>
<p>method used to handle missing values (see Details)</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_order">order</code></td>
<td>
<p>method used to order the axes (see Details)</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_showpoints">showPoints</code></td>
<td>
<p>logical operator indicating whether points should be
plotted or not</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_splinefactor">splineFactor</code></td>
<td>
<p>logical or numeric operator indicating whether spline interpolation should be used.  Numeric values will multiplied by the number of columns, <code>TRUE</code> will default to cubic interpolation, <code><a href="base.html#topic+AsIs">AsIs</a></code> to set the knot count directly and <code>0</code>, <code>FALSE</code>, or non-numeric values will not use spline interpolation.</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_alphalines">alphaLines</code></td>
<td>
<p>value of alpha scaler for the lines of the parcoord plot or a column name of the data</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_boxplot">boxplot</code></td>
<td>
<p>logical operator indicating whether or not boxplots should
underlay the distribution of each variable</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_shadebox">shadeBox</code></td>
<td>
<p>color of underlying box which extends from the min to the
max for each variable (no box is plotted if <code>shadeBox == NULL</code>)</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_mapping">mapping</code></td>
<td>
<p>aes string to pass to ggplot object</p>
</td></tr>
<tr><td><code id="ggparcoord_+3A_title">title</code></td>
<td>
<p>character string denoting the title of the plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>scale</code> is a character string that denotes how to scale the variables
in the parallel coordinate plot. Options:
</p>

<dl>
<dt><code>std</code></dt><dd><p>: univariately, subtract mean and divide by standard deviation</p>
</dd>
<dt><code>robust</code></dt><dd><p>: univariately, subtract median and divide by median absolute deviation</p>
</dd>
<dt><code>uniminmax</code></dt><dd><p>: univariately, scale so the minimum of the variable is zero, and the maximum is one</p>
</dd>
<dt><code>globalminmax</code></dt><dd><p>: no scaling is done; the range of the graphs is defined
by the global minimum and the global maximum</p>
</dd>
<dt><code>center</code></dt><dd><p>: use <code>uniminmax</code> to standardize vertical height, then
center each variable at a value specified by the <code>scaleSummary</code> param</p>
</dd>
<dt><code>centerObs</code></dt><dd><p>: use <code>uniminmax</code> to standardize vertical height, then
center each variable at the value of the observation specified by the <code>centerObsID</code> param</p>
</dd>
</dl>

<p><code>missing</code> is a character string that denotes how to handle missing
missing values. Options:
</p>

<dl>
<dt><code>exclude</code></dt><dd><p>: remove all cases with missing values</p>
</dd>
<dt><code>mean</code></dt><dd><p>: set missing values to the mean of the variable</p>
</dd>
<dt><code>median</code></dt><dd><p>: set missing values to the median of the variable</p>
</dd>
<dt><code>min10</code></dt><dd><p>: set missing values to 10% below the minimum of the variable</p>
</dd>
<dt><code>random</code></dt><dd><p>: set missing values to value of randomly chosen observation on that variable</p>
</dd>
</dl>

<p><code>order</code> is either a vector of indices or a character string that denotes how to
order the axes (variables) of the parallel coordinate plot. Options:
</p>

<dl>
<dt><code>(default)</code></dt><dd><p>: order by the vector denoted by <code>columns</code></p>
</dd>
<dt><code>(given vector)</code></dt><dd><p>: order by the vector specified</p>
</dd>
<dt><code>anyClass</code></dt><dd><p>: order variables by their separation between any one class and
the rest (as opposed to their overall variation between classes). This is accomplished
by calculating the F-statistic for each class vs. the rest, for each axis variable.
The axis variables are then ordered (decreasing) by their maximum of k F-statistics,
where k is the number of classes.</p>
</dd>
<dt><code>allClass</code></dt><dd><p>: order variables by their overall F statistic (decreasing) from
an ANOVA with <code>groupColumn</code> as the explanatory variable (note: it is required
to specify a <code>groupColumn</code> with this ordering method). Basically, this method
orders the variables by their variation between classes (most to least).</p>
</dd>
<dt><code>skewness</code></dt><dd><p>: order variables by their sample skewness (most skewed to
least skewed)</p>
</dd>
<dt><code>Outlying</code></dt><dd><p>: order by the scagnostic measure, Outlying, as calculated
by the package <code>scagnostics</code>. Other scagnostic measures available to order
by are <code>Skewed</code>, <code>Clumpy</code>, <code>Sparse</code>, <code>Striated</code>, <code>Convex</code>, <code>Skinny</code>, <code>Stringy</code>, and
<code>Monotonic</code>. Note: To use these methods of ordering, you must have the <code>scagnostics</code>
package loaded.</p>
</dd>
</dl>



<h3>Value</h3>

<p>ggplot object that if called, will print
</p>


<h3>Author(s)</h3>

<p>Jason Crowley, Barret Schloerke, Dianne Cook, Heike Hofmann, Hadley Wickham
</p>


<h3>Examples</h3>

<pre><code class='language-R'># small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

# use sample of the diamonds data for illustrative purposes
data(diamonds, package = "ggplot2")
diamonds.samp &lt;- diamonds[sample(1:dim(diamonds)[1], 100), ]

# basic parallel coordinate plot, using default settings
p &lt;- ggparcoord(data = diamonds.samp, columns = c(1, 5:10))
p_(p)

# this time, color by diamond cut
p &lt;- ggparcoord(data = diamonds.samp, columns = c(1, 5:10), groupColumn = 2)
p_(p)

# underlay univariate boxplots, add title, use uniminmax scaling
p &lt;- ggparcoord(
  data = diamonds.samp, columns = c(1, 5:10), groupColumn = 2,
  scale = "uniminmax", boxplot = TRUE, title = "Parallel Coord. Plot of Diamonds Data"
)
p_(p)

# utilize ggplot2 aes to switch to thicker lines
p &lt;- ggparcoord(
  data = diamonds.samp, columns = c(1, 5:10), groupColumn = 2,
  title = "Parallel Coord. Plot of Diamonds Data", mapping = ggplot2::aes(linewidth = 1)
) +
  ggplot2::scale_linewidth_identity()
p_(p)

# basic parallel coord plot of the msleep data, using 'random' imputation and
# coloring by diet (can also use variable names in the columns and groupColumn
# arguments)
data(msleep, package = "ggplot2")
p &lt;- ggparcoord(
  data = msleep, columns = 6:11, groupColumn = "vore", missing =
    "random", scale = "uniminmax"
)
p_(p)

# center each variable by its median, using the default missing value handler,
# 'exclude'
p &lt;- ggparcoord(
  data = msleep, columns = 6:11, groupColumn = "vore", scale =
    "center", scaleSummary = "median"
)
p_(p)

# with the iris data, order the axes by overall class (Species) separation using
# the anyClass option
p &lt;- ggparcoord(data = iris, columns = 1:4, groupColumn = 5, order = "anyClass")
p_(p)

# add points to the plot, add a title, and use an alpha scalar to make the lines
# transparent
p &lt;- ggparcoord(
  data = iris, columns = 1:4, groupColumn = 5, order = "anyClass",
  showPoints = TRUE, title = "Parallel Coordinate Plot for the Iris Data",
  alphaLines = 0.3
)
p_(p)

# color according to a column
iris2 &lt;- iris
iris2$alphaLevel &lt;- c("setosa" = 0.2, "versicolor" = 0.3, "virginica" = 0)[iris2$Species]
p &lt;- ggparcoord(
  data = iris2, columns = 1:4, groupColumn = 5, order = "anyClass",
  showPoints = TRUE, title = "Parallel Coordinate Plot for the Iris Data",
  alphaLines = "alphaLevel"
)
p_(p)

## Use splines on values, rather than lines (all produce the same result)
columns &lt;- c(1, 5:10)
p &lt;- ggparcoord(diamonds.samp, columns, groupColumn = 2, splineFactor = TRUE)
p_(p)
p &lt;- ggparcoord(diamonds.samp, columns, groupColumn = 2, splineFactor = 3)
p_(p)
</code></pre>

<hr>
<h2 id='ggscatmat'>Traditional scatterplot matrix for purely quantitative variables</h2><span id='topic+ggscatmat'></span>

<h3>Description</h3>

<p>This function makes a scatterplot matrix for quantitative variables with density plots on the diagonal
and correlation printed in the upper triangle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggscatmat(
  data,
  columns = 1:ncol(data),
  color = NULL,
  alpha = 1,
  corMethod = "pearson"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggscatmat_+3A_data">data</code></td>
<td>
<p>a data matrix. Should contain numerical (continuous) data.</p>
</td></tr>
<tr><td><code id="ggscatmat_+3A_columns">columns</code></td>
<td>
<p>an option to choose the column to be used in the raw dataset. Defaults to <code>1:ncol(data)</code>.</p>
</td></tr>
<tr><td><code id="ggscatmat_+3A_color">color</code></td>
<td>
<p>an option to group the dataset by the factor variable and color them by different colors.
Defaults to <code>NULL</code>, i.e. no coloring. If supplied, it will be converted to a factor.</p>
</td></tr>
<tr><td><code id="ggscatmat_+3A_alpha">alpha</code></td>
<td>
<p>an option to set the transparency in scatterplots for large data. Defaults to <code>1</code>.</p>
</td></tr>
<tr><td><code id="ggscatmat_+3A_cormethod">corMethod</code></td>
<td>
<p>method argument supplied to <code><a href="stats.html#topic+cor">cor</a></code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mengjia Ni, Di Cook
</p>


<h3>Examples</h3>

<pre><code class='language-R'># small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(flea)

p_(ggscatmat(flea, columns = 2:4))
p_(ggscatmat(flea, columns = 2:4, color = "species"))
</code></pre>

<hr>
<h2 id='ggsurv'>Survival curves</h2><span id='topic+ggsurv'></span>

<h3>Description</h3>

<p>This function produces Kaplan-Meier plots using <span class="pkg">ggplot2</span>.
As a first argument it needs a <code>survfit</code> object, created by the
<code>survival</code> package. Default settings differ for single stratum and
multiple strata objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggsurv(
  s,
  CI = "def",
  plot.cens = TRUE,
  surv.col = "gg.def",
  cens.col = "gg.def",
  lty.est = 1,
  lty.ci = 2,
  size.est = 0.5,
  size.ci = size.est,
  cens.size = 2,
  cens.shape = 3,
  back.white = FALSE,
  xlab = "Time",
  ylab = "Survival",
  main = "",
  order.legend = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggsurv_+3A_s">s</code></td>
<td>
<p>an object of class <code>survfit</code></p>
</td></tr>
<tr><td><code id="ggsurv_+3A_ci">CI</code></td>
<td>
<p>should a confidence interval be plotted? Defaults to <code>TRUE</code>
for single stratum objects and <code>FALSE</code> for multiple strata objects.</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_plot.cens">plot.cens</code></td>
<td>
<p>mark the censored observations?</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_surv.col">surv.col</code></td>
<td>
<p>colour of the survival estimate. Defaults to black for
one stratum, and to the default <span class="pkg">ggplot2</span> colours for multiple
strata. Length of vector with colour names should be either 1 or equal
to the number of strata.</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_cens.col">cens.col</code></td>
<td>
<p>colour of the points that mark censored observations.</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_lty.est">lty.est</code></td>
<td>
<p>linetype of the survival curve(s). Vector length should be
either 1 or equal to the number of strata.</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_lty.ci">lty.ci</code></td>
<td>
<p>linetype of the bounds that mark the 95% CI.</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_size.est">size.est</code></td>
<td>
<p>line width of the survival curve</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_size.ci">size.ci</code></td>
<td>
<p>line width of the 95% CI</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_cens.size">cens.size</code></td>
<td>
<p>point size of the censoring points</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_cens.shape">cens.shape</code></td>
<td>
<p>shape of the points that mark censored observations.</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_back.white">back.white</code></td>
<td>
<p>if TRUE the background will not be the default
grey of <code>ggplot2</code> but will be white with borders around the plot.</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_xlab">xlab</code></td>
<td>
<p>the label of the x-axis.</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_ylab">ylab</code></td>
<td>
<p>the label of the y-axis.</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_main">main</code></td>
<td>
<p>the plot label.</p>
</td></tr>
<tr><td><code id="ggsurv_+3A_order.legend">order.legend</code></td>
<td>
<p>boolean to determine if the legend display should be ordered by final survival time</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>ggplot</code>
</p>


<h3>Author(s)</h3>

<p>Edwin Thoen
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

if (require(survival) &amp;&amp; require(scales)) {
  lung &lt;- survival::lung
  sf.lung &lt;- survival::survfit(Surv(time, status) ~ 1, data = lung)
  p_(ggsurv(sf.lung))

  # Multiple strata examples
  sf.sex &lt;- survival::survfit(Surv(time, status) ~ sex, data = lung)
  pl.sex &lt;- ggsurv(sf.sex)
  p_(pl.sex)

  # Adjusting the legend of the ggsurv fit
  p_(pl.sex +
    ggplot2::guides(linetype = "none") +
    ggplot2::scale_colour_discrete(
      name   = "Sex",
      breaks = c(1, 2),
      labels = c("Male", "Female")
    ))

  # Multiple factors
  lung2 &lt;- dplyr::mutate(lung, older = as.factor(age &gt; 60))
  sf.sex2 &lt;- survival::survfit(Surv(time, status) ~ sex + older, data = lung2)
  pl.sex2 &lt;- ggsurv(sf.sex2)
  p_(pl.sex2)

  # Change legend title
  p_(pl.sex2 + labs(color = "New Title", linetype = "New Title"))

  # We can still adjust the plot after fitting
  kidney &lt;- survival::kidney
  sf.kid &lt;- survival::survfit(Surv(time, status) ~ disease, data = kidney)
  pl.kid &lt;- ggsurv(sf.kid, plot.cens = FALSE)
  p_(pl.kid)

  # Zoom in to first 80 days
  p_(pl.kid + ggplot2::coord_cartesian(xlim = c(0, 80), ylim = c(0.45, 1)))

  # Add the diseases names to the plot and remove legend
  p_(pl.kid +
    ggplot2::annotate(
      "text",
      label = c("PKD", "Other", "GN", "AN"),
      x = c(90, 125, 5, 60),
      y = c(0.8, 0.65, 0.55, 0.30),
      size = 5,
      colour = scales::hue_pal(
        h         = c(0, 360) + 15,
        c         = 100,
        l         = 65,
        h.start   = 0,
        direction = 1
      )(4)
    ) +
    ggplot2::guides(color = "none", linetype = "none"))
}
</code></pre>

<hr>
<h2 id='ggtable'>Cross-tabulated tables of discrete variables</h2><span id='topic+ggtable'></span>

<h3>Description</h3>

<p><code>ggtable</code> is a variant of <code><a href="#topic+ggduo">ggduo</a></code> for quick
cross-tabulated tables of discrete variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggtable(
  data,
  columnsX = 1:ncol(data),
  columnsY = 1:ncol(data),
  cells = c("observed", "prop", "row.prop", "col.prop", "expected", "resid", "std.resid"),
  fill = c("none", "std.resid", "resid"),
  mapping = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggtable_+3A_data">data</code></td>
<td>
<p>dataset to be used, can have both categorical and
numerical variables</p>
</td></tr>
<tr><td><code id="ggtable_+3A_columnsx">columnsX</code>, <code id="ggtable_+3A_columnsy">columnsY</code></td>
<td>
<p>names or positions of which columns are used to make plots. Defaults to all columns.</p>
</td></tr>
<tr><td><code id="ggtable_+3A_cells">cells</code></td>
<td>
<p>Which statistic should be displayed in table cells?</p>
</td></tr>
<tr><td><code id="ggtable_+3A_fill">fill</code></td>
<td>
<p>Which statistic should be used for filling table cells?</p>
</td></tr>
<tr><td><code id="ggtable_+3A_mapping">mapping</code></td>
<td>
<p>additional aesthetic to be used, for example to indicate
weights (see examples)</p>
</td></tr>
<tr><td><code id="ggtable_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code><a href="#topic+ggduo">ggduo</a></code> (see examples)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Joseph Larmarange
</p>


<h3>Examples</h3>

<pre><code class='language-R'># small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(tips)
p_(ggtable(tips, "smoker", c("day", "time", "sex")))

# displaying row proportions
p_(ggtable(tips, "smoker", c("day", "time", "sex"), cells = "row.prop"))

# filling cells with standardized residuals
p_(ggtable(tips, "smoker", c("day", "time", "sex"), fill = "std.resid", legend = 1))

# if continuous variables are provided, just displaying some summary statistics
p_(ggtable(tips, c("smoker", "total_bill"), c("day", "time", "sex", "tip")))

# specifying weights
d &lt;- as.data.frame(Titanic)
p_(ggtable(
  d,
  "Survived",
  c("Class", "Sex", "Age"),
  mapping = aes(weight = Freq),
  cells = "row.prop",
  fill = "std.resid"
))
</code></pre>

<hr>
<h2 id='ggts'>Multiple time series</h2><span id='topic+ggts'></span>

<h3>Description</h3>

<p>GGally implementation of ts.plot. Wraps around the ggduo function and removes the column strips
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggts(..., columnLabelsX = NULL, xlab = "time")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggts_+3A_...">...</code></td>
<td>
<p>supplied directly to <code><a href="#topic+ggduo">ggduo</a></code></p>
</td></tr>
<tr><td><code id="ggts_+3A_columnlabelsx">columnLabelsX</code></td>
<td>
<p>remove top strips for the X axis by default</p>
</td></tr>
<tr><td><code id="ggts_+3A_xlab">xlab</code></td>
<td>
<p>defaults to &quot;time&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code><a href="#topic+ggmatrix">ggmatrix</a></code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

p_(ggts(pigs, "time", c("gilts", "profit", "s_per_herdsz", "production", "herdsz")))
</code></pre>

<hr>
<h2 id='glyphplot'>Glyph plot class</h2><span id='topic+glyphplot'></span><span id='topic+is.glyphplot'></span><span id='topic++5B.glyphplot'></span><span id='topic+print.glyphplot'></span>

<h3>Description</h3>

<p>Glyph plot class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glyphplot(data, width, height, polar, x_major, y_major)

is.glyphplot(x)

## S3 method for class 'glyphplot'
x[...]

## S3 method for class 'glyphplot'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glyphplot_+3A_data">data</code></td>
<td>
<p>A data frame containing variables named in <code>x_major</code>,
<code>x_minor</code>, <code>y_major</code> and <code>y_minor</code>.</p>
</td></tr>
<tr><td><code id="glyphplot_+3A_height">height</code>, <code id="glyphplot_+3A_width">width</code></td>
<td>
<p>The height and width of each glyph. Defaults to 95% of
the <code><a href="ggplot2.html#topic+resolution">resolution</a></code> of the data. Specify the width
absolutely by supplying a numeric vector of length 1, or relative to the</p>
</td></tr>
<tr><td><code id="glyphplot_+3A_polar">polar</code></td>
<td>
<p>A logical of length 1, specifying whether the glyphs should
be drawn in polar coordinates.  Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="glyphplot_+3A_x_major">x_major</code>, <code id="glyphplot_+3A_y_major">y_major</code></td>
<td>
<p>The name of the variable (as a
string) for the major x and y axes.  Together, the</p>
</td></tr>
<tr><td><code id="glyphplot_+3A_x">x</code></td>
<td>
<p>glyphplot to be printed</p>
</td></tr>
<tr><td><code id="glyphplot_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Di Cook, Heike Hofmann, Hadley Wickham
</p>

<hr>
<h2 id='glyphs'>Create <code><a href="#topic+glyphplot">glyphplot</a></code> data</h2><span id='topic+glyphs'></span>

<h3>Description</h3>

<p>Create the data needed to generate a glyph plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glyphs(
  data,
  x_major,
  x_minor,
  y_major,
  y_minor,
  polar = FALSE,
  height = ggplot2::rel(0.95),
  width = ggplot2::rel(0.95),
  y_scale = identity,
  x_scale = identity
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glyphs_+3A_data">data</code></td>
<td>
<p>A data frame containing variables named in <code>x_major</code>,
<code>x_minor</code>, <code>y_major</code> and <code>y_minor</code>.</p>
</td></tr>
<tr><td><code id="glyphs_+3A_x_major">x_major</code>, <code id="glyphs_+3A_x_minor">x_minor</code>, <code id="glyphs_+3A_y_major">y_major</code>, <code id="glyphs_+3A_y_minor">y_minor</code></td>
<td>
<p>The name of the variable (as a
string) for the major and minor x and y axes.  Together, each unique</p>
</td></tr>
<tr><td><code id="glyphs_+3A_polar">polar</code></td>
<td>
<p>A logical of length 1, specifying whether the glyphs should
be drawn in polar coordinates.  Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="glyphs_+3A_height">height</code>, <code id="glyphs_+3A_width">width</code></td>
<td>
<p>The height and width of each glyph. Defaults to 95% of
the <code><a href="ggplot2.html#topic+resolution">resolution</a></code> of the data. Specify the width
absolutely by supplying a numeric vector of length 1, or relative to the</p>
</td></tr>
<tr><td><code id="glyphs_+3A_y_scale">y_scale</code>, <code id="glyphs_+3A_x_scale">x_scale</code></td>
<td>
<p>The scaling function to be applied to each set of
minor values within a grid cell.  Defaults to <code><a href="base.html#topic+identity">identity</a></code> so
that no scaling is performed.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Di Cook, Heike Hofmann, Hadley Wickham
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(nasa)
nasaLate &lt;- nasa[
  nasa$date &gt;= as.POSIXct("1998-01-01") &amp;
    nasa$lat &gt;= 20 &amp;
    nasa$lat &lt;= 40 &amp;
    nasa$long &gt;= -80 &amp;
    nasa$long &lt;= -60,
]
temp.gly &lt;- glyphs(nasaLate, "long", "day", "lat", "surftemp", height = 2.5)
p_(ggplot2::ggplot(temp.gly, ggplot2::aes(gx, gy, group = gid)) +
  add_ref_lines(temp.gly, color = "grey90") +
  add_ref_boxes(temp.gly, color = "grey90") +
  ggplot2::geom_path() +
  ggplot2::theme_bw() +
  ggplot2::labs(x = "", y = ""))
</code></pre>

<hr>
<h2 id='grab_legend'>Grab the legend and print it as a plot</h2><span id='topic+grab_legend'></span><span id='topic+print.legend_guide_box'></span>

<h3>Description</h3>

<p>Grab the legend and print it as a plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grab_legend(p)

## S3 method for class 'legend_guide_box'
print(x, ..., plotNew = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grab_legend_+3A_p">p</code></td>
<td>
<p>ggplot2 plot object</p>
</td></tr>
<tr><td><code id="grab_legend_+3A_x">x</code></td>
<td>
<p>legend object that has been grabbed from a ggplot2 object</p>
</td></tr>
<tr><td><code id="grab_legend_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
<tr><td><code id="grab_legend_+3A_plotnew">plotNew</code></td>
<td>
<p>boolean to determine if the <code>grid.newpage()</code> command and a new blank rectangle should be printed</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

library(ggplot2)
histPlot &lt;-
  ggplot(iris, aes(Sepal.Length, fill = Species)) +
  geom_histogram(binwidth = 1 / 4)
(right &lt;- histPlot)
(bottom &lt;- histPlot + theme(legend.position = "bottom"))
(top &lt;- histPlot + theme(legend.position = "top"))
(left &lt;- histPlot + theme(legend.position = "left"))

p_(grab_legend(right))
p_(grab_legend(bottom))
p_(grab_legend(top))
p_(grab_legend(left))
</code></pre>

<hr>
<h2 id='happy'>Data related to happiness from the General Social Survey, 1972-2006.</h2><span id='topic+happy'></span>

<h3>Description</h3>

<p>This data extract is taken from Hadley Wickham's <code>productplots</code> package.
The original description follows, with minor edits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(happy)
</code></pre>


<h3>Format</h3>

<p>A data frame with 51020 rows and 10 variables
</p>


<h3>Details</h3>

<p>The data is a small sample of variables related to
happiness from the General Social Survey (GSS). The GSS
is a yearly cross-sectional survey of Americans, run from
1972. We combine data for 25 years to yield 51,020
observations, and of the over 5,000 variables, we select
nine related to happiness:
</p>

<ul>
<li><p> age. age in years: 18&ndash;89.
</p>
</li>
<li><p> degree. highest education: lt high school, high school, junior college, bachelor, graduate.
</p>
</li>
<li><p> finrela. relative financial status: far above, above average, average, below average, far below.
</p>
</li>
<li><p> happy. happiness: very happy, pretty happy, not too happy.
</p>
</li>
<li><p> health. health: excellent, good, fair, poor.
</p>
</li>
<li><p> marital. marital status: married, never married, divorced, widowed, separated.
</p>
</li>
<li><p> sex. sex: female, male.
</p>
</li>
<li><p> wtsall. probability weight. 0.43&ndash;6.43.
</p>
</li></ul>



<h3>References</h3>

<p>Smith, Tom W., Peter V. Marsden, Michael Hout, Jibum Kim. <em>General Social Surveys, 1972-2006</em>.
[machine-readable data file]. Principal Investigator, Tom W. Smith; Co-Principal Investigators,
Peter V. Marsden and Michael Hout, NORC ed.
Chicago: National Opinion Research Center, producer, 2005;
Storrs, CT: The Roper Center for Public Opinion Research, University of Connecticut, distributor.
1 data file (57,061 logical records) and 1 codebook (3,422 pp).
</p>

<hr>
<h2 id='is_blank_plot'>Is Blank Plot?
Find out if the plot equals a blank plot</h2><span id='topic+is_blank_plot'></span>

<h3>Description</h3>

<p>Is Blank Plot?
Find out if the plot equals a blank plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_blank_plot(p)
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>GGally:::is_blank_plot(ggally_blank())
GGally:::is_blank_plot(ggally_points(mtcars, ggplot2::aes(disp, hp)))

</code></pre>

<hr>
<h2 id='is_date'>Check if object is a date</h2><span id='topic+is_date'></span>

<h3>Description</h3>

<p>Check if object is a date
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_date(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_date_+3A_x">x</code></td>
<td>
<p>vector</p>
</td></tr>
</table>

<hr>
<h2 id='is_horizontal'>Check if plot is horizontal</h2><span id='topic+is_horizontal'></span><span id='topic+is_character_column'></span>

<h3>Description</h3>

<p>Check if plot is horizontal
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_horizontal(data, mapping, val = "y")

is_character_column(data, mapping, val = "y")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_horizontal_+3A_data">data</code></td>
<td>
<p>data used in ggplot2 plot</p>
</td></tr>
<tr><td><code id="is_horizontal_+3A_mapping">mapping</code></td>
<td>
<p>ggplot2 <code>aes()</code> mapping</p>
</td></tr>
<tr><td><code id="is_horizontal_+3A_val">val</code></td>
<td>
<p>key to retrieve from <code>mapping</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Boolean determining if the data is a character-like data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_horizontal(iris, ggplot2::aes(Sepal.Length, Species)) # TRUE
is_horizontal(iris, ggplot2::aes(Sepal.Length, Species), "x") # FALSE
is_horizontal(iris, ggplot2::aes(Sepal.Length, Sepal.Width)) # FALSE
</code></pre>

<hr>
<h2 id='lowertriangle'>lowertriangle - rearrange dataset as the preparation of <code><a href="#topic+ggscatmat">ggscatmat</a></code> function</h2><span id='topic+lowertriangle'></span>

<h3>Description</h3>

<p>function for making the melted dataset used to plot the lowertriangle scatterplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lowertriangle(data, columns = 1:ncol(data), color = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lowertriangle_+3A_data">data</code></td>
<td>
<p>a data matrix. Should contain numerical (continuous) data.</p>
</td></tr>
<tr><td><code id="lowertriangle_+3A_columns">columns</code></td>
<td>
<p>an option to choose the column to be used in the raw dataset. Defaults to <code>1:ncol(data)</code></p>
</td></tr>
<tr><td><code id="lowertriangle_+3A_color">color</code></td>
<td>
<p>an option to choose a factor variable to be grouped with. Defaults to <code>(NULL)</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mengjia Ni, Di Cook
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flea)
head(lowertriangle(flea, columns = 2:4))
head(lowertriangle(flea))
head(lowertriangle(flea, color = "species"))
</code></pre>

<hr>
<h2 id='mapping_color_to_fill'>Aesthetic mapping color fill</h2><span id='topic+mapping_color_to_fill'></span>

<h3>Description</h3>

<p>Replace the fill with the color and make color NULL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapping_color_to_fill(current)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapping_color_to_fill_+3A_current">current</code></td>
<td>
<p>the current aesthetics</p>
</td></tr>
</table>

<hr>
<h2 id='mapping_string'>Aes name</h2><span id='topic+mapping_string'></span>

<h3>Description</h3>

<p>Aes name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapping_string(aes_col)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapping_string_+3A_aes_col">aes_col</code></td>
<td>
<p>Single value from <code>ggplot2::<a href="ggplot2.html#topic+aes">aes</a>(...)</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>character string
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mapping &lt;- ggplot2::aes(Petal.Length)
mapping_string(mapping$x)
</code></pre>

<hr>
<h2 id='mapping_swap_x_y'>Swap x and y mapping</h2><span id='topic+mapping_swap_x_y'></span>

<h3>Description</h3>

<p>Swap x and y mapping
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapping_swap_x_y(mapping)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapping_swap_x_y_+3A_mapping">mapping</code></td>
<td>
<p>output of <code>ggplot2::<a href="ggplot2.html#topic+aes">aes</a>(...)</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Aes mapping with the x and y values switched
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mapping &lt;- ggplot2::aes(Petal.Length, Sepal.Width)
mapping
mapping_swap_x_y(mapping)
</code></pre>

<hr>
<h2 id='model_response_variables'>Model term names</h2><span id='topic+model_response_variables'></span><span id='topic+model_beta_variables'></span><span id='topic+model_beta_label'></span>

<h3>Description</h3>

<p>Retrieve either the response variable names, the beta variable names, or beta variable names.  If the model is an object of class 'lm', by default, the beta variable names will include anova significance stars.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>model_response_variables(model, data = broom::augment(model))

model_beta_variables(model, data = broom::augment(model))

model_beta_label(model, data = broom::augment(model), lmStars = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="model_response_variables_+3A_model">model</code></td>
<td>
<p>model in question</p>
</td></tr>
<tr><td><code id="model_response_variables_+3A_data">data</code></td>
<td>
<p>equivalent to <code>broom::augment(model)</code></p>
</td></tr>
<tr><td><code id="model_response_variables_+3A_lmstars">lmStars</code></td>
<td>
<p>boolean that determines if stars are added to labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector of names
</p>

<hr>
<h2 id='nasa'>Data from the Data Expo JSM 2006.</h2><span id='topic+nasa'></span>

<h3>Description</h3>

<p>This data was provided by NASA for the competition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nasa)
</code></pre>


<h3>Format</h3>

<p>A data frame with 41472 rows and 17 variables
</p>


<h3>Details</h3>

<p>The data shows 6 years of monthly measurements of a 24x24 spatial grid
from Central America:
</p>

<ul>
<li><p> time integer specifying temporal order of measurements
</p>
</li>
<li><p> x, y, lat, long spatial location of measurements.
</p>
</li>
<li><p> cloudhigh, cloudlow, cloudmid, ozone, pressure, surftemp, temperature
are the various satellite measurements.
</p>
</li>
<li><p> date, day, month, year specifying the time of measurements.
</p>
</li>
<li><p> id unique ide for each spatial position.
</p>
</li></ul>



<h3>References</h3>

<p>Murrell, P. (2010) The 2006 Data Expo of the American Statistical Association.
Computational Statistics, 25:551-554.
</p>

<hr>
<h2 id='pigs'>United Kingdom Pig Production</h2><span id='topic+pigs'></span>

<h3>Description</h3>

<p>This data contains about the United Kingdom Pig Production from the book 'Data' by Andrews and Herzberg. The original data can be on Statlib: http://lib.stat.cmu.edu/datasets/Andrews/T62.1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pigs)
</code></pre>


<h3>Format</h3>

<p>A data frame with 48 rows and 8 variables
</p>


<h3>Details</h3>

<p>The time variable has been added from a combination of year and quarter
</p>

<ul>
<li><p> time year + (quarter - 1) / 4
</p>
</li>
<li><p> year year of production
</p>
</li>
<li><p> quarter quarter of the year of production
</p>
</li>
<li><p> gilts number of sows giving birth for the first time
</p>
</li>
<li><p> profit ratio of price to an index of feed price
</p>
</li>
<li><p> s_per_herdsz ratio of the number of breeding pigs slaughtered to the total breeding herd size
</p>
</li>
<li><p> production number of pigs slaughtered that were reared for meat
</p>
</li>
<li><p> herdsz breeding herd size
</p>
</li></ul>



<h3>References</h3>

<p>Andrews, David F., and Agnes M. Herzberg. Data: a collection of problems from many fields for the student and research worker. Springer Science &amp; Business Media, 2012.
</p>

<hr>
<h2 id='plot_types'>Plot Types</h2><span id='topic+plot_types'></span>

<h3>Description</h3>

<p>Retrieves the type of plot that should be used for all combinations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_types(data, columnsX, columnsY, allowDiag = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_types_+3A_data">data</code></td>
<td>
<p>data set to be used</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>

<hr>
<h2 id='plotting_data_type'>Get plotting data type</h2><span id='topic+plotting_data_type'></span>

<h3>Description</h3>

<p>Get plotting data type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotting_data_type(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotting_data_type_+3A_x">x</code></td>
<td>
<p>vector</p>
</td></tr>
</table>

<hr>
<h2 id='print_if_interactive'>Print if not CRAN</h2><span id='topic+print_if_interactive'></span>

<h3>Description</h3>

<p>Small function to print a plot if the R session is interactive or in a CI build
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_if_interactive(p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_if_interactive_+3A_p">p</code></td>
<td>
<p>plot to be displayed</p>
</td></tr>
</table>

<hr>
<h2 id='print.ggmatrix'>Print <code><a href="#topic+ggmatrix">ggmatrix</a></code> object</h2><span id='topic+print.ggmatrix'></span>

<h3>Description</h3>

<p>Print method taken from <code>ggplot2:::print.ggplot</code> and altered for a <code><a href="#topic+ggmatrix">ggmatrix</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ggmatrix'
print(x, newpage = is.null(vp), vp = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.ggmatrix_+3A_x">x</code></td>
<td>
<p>plot to display</p>
</td></tr>
<tr><td><code id="print.ggmatrix_+3A_newpage">newpage</code></td>
<td>
<p>draw new (empty) page first?</p>
</td></tr>
<tr><td><code id="print.ggmatrix_+3A_vp">vp</code></td>
<td>
<p>viewport to draw plot in</p>
</td></tr>
<tr><td><code id="print.ggmatrix_+3A_...">...</code></td>
<td>
<p>arguments passed onto <code><a href="#topic+ggmatrix_gtable">ggmatrix_gtable</a></code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tips)
pMat &lt;- ggpairs(tips, c(1, 3, 2), mapping = ggplot2::aes(color = sex))
pMat # calls print(pMat), which calls print.ggmatrix(pMat)
</code></pre>

<hr>
<h2 id='psychademic'>UCLA canonical correlation analysis data</h2><span id='topic+psychademic'></span>

<h3>Description</h3>

<p>This data contains 600 observations on eight variables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(psychademic)
</code></pre>


<h3>Format</h3>

<p>A data frame with 600 rows and 8 variables
</p>


<h3>Details</h3>


<ul>
<li><p> locus_of_control - psychological
</p>
</li>
<li><p> self_concept - psychological
</p>
</li>
<li><p> motivation - psychological. Converted to four character groups
</p>
</li>
<li><p> read - academic
</p>
</li>
<li><p> write - academic
</p>
</li>
<li><p> math - academic
</p>
</li>
<li><p> science - academic
</p>
</li>
<li><p> female - academic. Dropped from original source
</p>
</li>
<li><p> sex - academic. Added as a character version of female column
</p>
</li></ul>



<h3>References</h3>

<p>R Data Analysis Examples | Canonical Correlation Analysis.  UCLA: Institute for Digital Research and Education. from http://www.stats.idre.ucla.edu/r/dae/canonical-correlation-analysis (accessed May 22, 2017).
</p>

<hr>
<h2 id='putPlot'>Insert a plot into a <code><a href="#topic+ggmatrix">ggmatrix</a></code> object</h2><span id='topic+putPlot'></span><span id='topic++5B+3C-.ggmatrix'></span>

<h3>Description</h3>

<p>Function to place your own plot in the layout.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>putPlot(pm, value, i, j)

## S3 replacement method for class 'ggmatrix'
pm[i, j, ...] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="putPlot_+3A_pm">pm</code></td>
<td>
<p>ggally object to be altered</p>
</td></tr>
<tr><td><code id="putPlot_+3A_value">value</code></td>
<td>
<p>ggplot object to be placed</p>
</td></tr>
<tr><td><code id="putPlot_+3A_i">i</code></td>
<td>
<p>row from the top</p>
</td></tr>
<tr><td><code id="putPlot_+3A_j">j</code></td>
<td>
<p>column from the left</p>
</td></tr>
<tr><td><code id="putPlot_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getPlot">getPlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

custom_car &lt;- ggpairs(mtcars[, c("mpg", "wt", "cyl")], upper = "blank", title = "Custom Example")
# ggplot example taken from example(geom_text)
plot &lt;- ggplot2::ggplot(mtcars, ggplot2::aes(x = wt, y = mpg, label = rownames(mtcars)))
plot &lt;- plot +
  ggplot2::geom_text(ggplot2::aes(colour = factor(cyl)), size = 3) +
  ggplot2::scale_colour_discrete(l = 40)
custom_car[1, 2] &lt;- plot
personal_plot &lt;- ggally_text(
  "ggpairs allows you\nto put in your\nown plot.\nLike that one.\n &lt;---"
)
custom_car[1, 3] &lt;- personal_plot
# custom_car

# remove plots after creating a plot matrix
custom_car[2, 1] &lt;- NULL
custom_car[3, 1] &lt;- "blank" # the same as storing null
custom_car[3, 2] &lt;- NULL
p_(custom_car)
</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+ggcoef_model'></span><span id='topic+ggcoef_compare'></span><span id='topic+ggcoef_multinom'></span><span id='topic+ggcoef_plot'></span><span id='topic+signif_stars'></span><span id='topic+geom_stripped_cols'></span><span id='topic+geom_stripped_rows'></span><span id='topic+stat_cross'></span><span id='topic+StatCross'></span><span id='topic+stat_prop'></span><span id='topic+StatProp'></span><span id='topic+stat_weighted_mean'></span><span id='topic+StatWeightedMean'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>ggstats</dt><dd><p><code><a href="ggstats.html#topic+geom_stripped_rows">geom_stripped_cols</a></code>, <code><a href="ggstats.html#topic+geom_stripped_rows">geom_stripped_rows</a></code>, <code><a href="ggstats.html#topic+ggcoef_model">ggcoef_compare</a></code>, <code><a href="ggstats.html#topic+ggcoef_model">ggcoef_model</a></code>, <code><a href="ggstats.html#topic+ggcoef_model">ggcoef_multinom</a></code>, <code><a href="ggstats.html#topic+ggcoef_model">ggcoef_plot</a></code>, <code><a href="ggstats.html#topic+signif_stars">signif_stars</a></code>, <code><a href="ggstats.html#topic+stat_cross">stat_cross</a></code>, <code><a href="ggstats.html#topic+stat_prop">stat_prop</a></code>, <code><a href="ggstats.html#topic+stat_weighted_mean">stat_weighted_mean</a></code>, <code><a href="ggstats.html#topic+stat_cross">StatCross</a></code>, <code><a href="ggstats.html#topic+stat_prop">StatProp</a></code>, <code><a href="ggstats.html#topic+stat_weighted_mean">StatWeightedMean</a></code></p>
</dd>
</dl>

<hr>
<h2 id='remove_color_unless_equal'>Remove colour mapping unless found in select mapping keys</h2><span id='topic+remove_color_unless_equal'></span>

<h3>Description</h3>

<p>Remove colour mapping unless found in select mapping keys
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_color_unless_equal(mapping, to = c("x", "y"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_color_unless_equal_+3A_mapping">mapping</code></td>
<td>
<p>output of <code>ggplot2::<a href="ggplot2.html#topic+aes">aes</a>(...)</code></p>
</td></tr>
<tr><td><code id="remove_color_unless_equal_+3A_to">to</code></td>
<td>
<p>set of mapping keys to check</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Aes mapping with colour mapping kept only if found in selected mapping keys.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mapping &lt;- aes(x = sex, y = age, colour = sex)

mapping &lt;- aes(x = sex, y = age, colour = region)
remove_color_unless_equal(mapping)
</code></pre>

<hr>
<h2 id='require_namespaces'>Loads package namespaces</h2><span id='topic+require_namespaces'></span>

<h3>Description</h3>

<p>Loads package namespaces or yells at user... loudly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>require_namespaces(pkgs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="require_namespaces_+3A_pkgs">pkgs</code></td>
<td>
<p>vector of character values</p>
</td></tr>
</table>

<hr>
<h2 id='rescale01'>Rescaling functions</h2><span id='topic+rescale01'></span><span id='topic+range01'></span><span id='topic+max1'></span><span id='topic+mean0'></span><span id='topic+min0'></span><span id='topic+rescale11'></span>

<h3>Description</h3>

<p>Rescaling functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>range01(x)

max1(x)

mean0(x)

min0(x)

rescale01(x, xlim = NULL)

rescale11(x, xlim = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rescale01_+3A_x">x</code></td>
<td>
<p>numeric vector</p>
</td></tr>
<tr><td><code id="rescale01_+3A_xlim">xlim</code></td>
<td>
<p>value used in <code>range</code></p>
</td></tr>
</table>

<hr>
<h2 id='scag_order'>Find order of variables</h2><span id='topic+scag_order'></span>

<h3>Description</h3>

<p>Find order of variables based on a specified scagnostic measure
by maximizing the index values of that measure along the path.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scag_order(scag, vars, measure)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scag_order_+3A_scag">scag</code></td>
<td>
<p><code>scagnostics</code> object</p>
</td></tr>
<tr><td><code id="scag_order_+3A_vars">vars</code></td>
<td>
<p>character vector of the variables to be ordered</p>
</td></tr>
<tr><td><code id="scag_order_+3A_measure">measure</code></td>
<td>
<p>scagnostics measure to order according to</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector of variable ordered according to the given
scagnostic measure
</p>


<h3>Author(s)</h3>

<p>Barret Schloerke
</p>

<hr>
<h2 id='scatmat'>Plots the lowertriangle and density plots of the scatter plot matrix.</h2><span id='topic+scatmat'></span>

<h3>Description</h3>

<p>Function for making scatterplots in the lower triangle and diagonal density plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scatmat(data, columns = 1:ncol(data), color = NULL, alpha = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scatmat_+3A_data">data</code></td>
<td>
<p>a data matrix. Should contain numerical (continuous) data.</p>
</td></tr>
<tr><td><code id="scatmat_+3A_columns">columns</code></td>
<td>
<p>an option to choose the column to be used in the raw dataset. Defaults to <code>1:ncol(data)</code></p>
</td></tr>
<tr><td><code id="scatmat_+3A_color">color</code></td>
<td>
<p>an option to group the dataset by the factor variable and color them by different colors. Defaults to <code>NULL</code></p>
</td></tr>
<tr><td><code id="scatmat_+3A_alpha">alpha</code></td>
<td>
<p>an option to set the transparency in scatterplots for large data. Defaults to <code>1</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mengjia Ni, Di Cook
</p>


<h3>Examples</h3>

<pre><code class='language-R'># small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

data(flea)

p_(scatmat(flea, columns = 2:4))
p_(scatmat(flea, columns = 2:4, color = "species"))
</code></pre>

<hr>
<h2 id='singleClassOrder'>Order axis variables</h2><span id='topic+singleClassOrder'></span>

<h3>Description</h3>

<p>Order axis variables by separation between one class and the rest
(most separation to least).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>singleClassOrder(classVar, axisVars, specClass = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="singleClassOrder_+3A_classvar">classVar</code></td>
<td>
<p>class variable (vector from original dataset)</p>
</td></tr>
<tr><td><code id="singleClassOrder_+3A_axisvars">axisVars</code></td>
<td>
<p>variables to be plotted as axes (data frame)</p>
</td></tr>
<tr><td><code id="singleClassOrder_+3A_specclass">specClass</code></td>
<td>
<p>character string matching to level of <code>classVar</code>; instead
of looking for separation between any class and the rest, will only look for
separation between this class and the rest</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector of names of axisVars ordered such that the first
variable has the most separation between one of the classes and the rest, and
the last variable has the least (as measured by F-statistics from an ANOVA)
</p>


<h3>Author(s)</h3>

<p>Jason Crowley
</p>

<hr>
<h2 id='skewness'>Sample skewness</h2><span id='topic+skewness'></span>

<h3>Description</h3>

<p>Calculate the sample skewness of a vector
while ignoring missing values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>skewness(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="skewness_+3A_x">x</code></td>
<td>
<p>numeric vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sample skewness of <code>x</code>
</p>


<h3>Author(s)</h3>

<p>Jason Crowley
</p>

<hr>
<h2 id='str.ggmatrix'><code><a href="#topic+ggmatrix">ggmatrix</a></code> structure</h2><span id='topic+str.ggmatrix'></span>

<h3>Description</h3>

<p>View the condensed version of the <code><a href="#topic+ggmatrix">ggmatrix</a></code> object. The attribute &quot;class&quot; is ALWAYS altered to &quot;_class&quot; to avoid recursion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ggmatrix'
str(object, ..., raw = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="str.ggmatrix_+3A_object">object</code></td>
<td>
<p><code><a href="#topic+ggmatrix">ggmatrix</a></code> object to be viewed</p>
</td></tr>
<tr><td><code id="str.ggmatrix_+3A_...">...</code></td>
<td>
<p>passed on to the default <code>str</code> method</p>
</td></tr>
<tr><td><code id="str.ggmatrix_+3A_raw">raw</code></td>
<td>
<p>boolean to determine if the plots should be converted to text or kept as original objects</p>
</td></tr>
</table>

<hr>
<h2 id='tips'>Tipping data</h2><span id='topic+tips'></span>

<h3>Description</h3>

<p>One waiter recorded information about each tip he received over a
period of a few months working in one restaurant. He collected several
variables:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tips
</code></pre>


<h3>Format</h3>

<p>A data frame with 244 rows and 7 variables
</p>


<h3>Details</h3>


<ul>
<li><p> tip in dollars,
</p>
</li>
<li><p> bill in dollars,
</p>
</li>
<li><p> sex of the bill payer,
</p>
</li>
<li><p> whether there were smokers in the party,
</p>
</li>
<li><p> day of the week,
</p>
</li>
<li><p> time of day,
</p>
</li>
<li><p> size of the party.
</p>
</li></ul>

<p>In all he recorded 244 tips. The data was reported in a collection of
case studies for business statistics (Bryant &amp; Smith 1995).
</p>


<h3>References</h3>

<p>Bryant, P. G. and Smith, M (1995) <em>Practical Data
Analysis: Case Studies in Business Statistics</em>.  Homewood, IL: Richard D.
Irwin Publishing:
</p>

<hr>
<h2 id='twitter_spambots'>Twitter spambots</h2><span id='topic+twitter_spambots'></span>

<h3>Description</h3>

<p>A network of spambots found on Twitter as part of a data mining project.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(twitter_spambots)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>network</code> with 120 edges and 94 vertices.
</p>


<h3>Details</h3>

<p>Each node of the network is identified by the Twitter screen name of the
account and further carries five vertex attributes:
</p>

<ul>
<li><p> location user's location, as provided by the user
</p>
</li>
<li><p> lat latitude, based on the user's location
</p>
</li>
<li><p> lon longitude, based on the user's location
</p>
</li>
<li><p> followers number of Twitter accounts that follow this account
</p>
</li>
<li><p> friends number of Twitter accounts followed by the account
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Amos Elberg
</p>

<hr>
<h2 id='uppertriangle'>Rearrange dataset as the preparation of <code><a href="#topic+ggscatmat">ggscatmat</a></code> function</h2><span id='topic+uppertriangle'></span>

<h3>Description</h3>

<p>Function for making the dataset used to plot the uppertriangle plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uppertriangle(
  data,
  columns = 1:ncol(data),
  color = NULL,
  corMethod = "pearson"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="uppertriangle_+3A_data">data</code></td>
<td>
<p>a data matrix. Should contain numerical (continuous) data.</p>
</td></tr>
<tr><td><code id="uppertriangle_+3A_columns">columns</code></td>
<td>
<p>an option to choose the column to be used in the raw dataset. Defaults to <code>1:ncol(data)</code></p>
</td></tr>
<tr><td><code id="uppertriangle_+3A_color">color</code></td>
<td>
<p>an option to choose a factor variable to be grouped with. Defaults to <code>(NULL)</code></p>
</td></tr>
<tr><td><code id="uppertriangle_+3A_cormethod">corMethod</code></td>
<td>
<p>method argument supplied to <code><a href="stats.html#topic+cor">cor</a></code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mengjia Ni, Di Cook
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flea)
head(uppertriangle(flea, columns = 2:4))
head(uppertriangle(flea))
head(uppertriangle(flea, color = "species"))
</code></pre>

<hr>
<h2 id='v1_ggmatrix_theme'>Modify a <code><a href="#topic+ggmatrix">ggmatrix</a></code> object by adding an <span class="pkg">ggplot2</span> object to all</h2><span id='topic+v1_ggmatrix_theme'></span>

<h3>Description</h3>

<p>Modify a <code><a href="#topic+ggmatrix">ggmatrix</a></code> object by adding an <span class="pkg">ggplot2</span> object to all
</p>


<h3>Usage</h3>

<pre><code class='language-R'>v1_ggmatrix_theme()
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'># Small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

p_(ggpairs(iris, 1:2) + v1_ggmatrix_theme())
# move the column names to the left and bottom
p_(ggpairs(iris, 1:2, switch = "both") + v1_ggmatrix_theme())
</code></pre>

<hr>
<h2 id='vig_ggally'>View GGally vignettes</h2><span id='topic+vig_ggally'></span>

<h3>Description</h3>

<p>This function will open the directly to the vignette requested. If no <code>name</code> is provided, the index of all <span class="pkg">GGally</span> vignettes will be opened.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vig_ggally(name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vig_ggally_+3A_name">name</code></td>
<td>
<p>Vignette name to open. If no name is provided, the vignette index will be opened</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method allows for vignettes to be hosted remotely, reducing <span class="pkg">GGally</span>'s package size, and installation time.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# View `ggnostic` vignette
vig_ggally("ggnostic")

# View all vignettes by GGally
vig_ggally()

</code></pre>

<hr>
<h2 id='wrap_fn_with_param_arg'>Wrap a function with different parameter values</h2><span id='topic+wrap_fn_with_param_arg'></span><span id='topic+wrapp'></span><span id='topic+wrap'></span><span id='topic+wrap_fn_with_params'></span>

<h3>Description</h3>

<p>Wraps a function with the supplied parameters to force different default behavior.  This is useful for functions that are supplied to ggpairs.  It allows you to change the behavior of one function, rather than creating multiple functions with different parameter settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wrap_fn_with_param_arg(
  funcVal,
  params = NULL,
  funcArgName = deparse(substitute(funcVal))
)

wrapp(funcVal, params = NULL, funcArgName = deparse(substitute(funcVal)))

wrap(funcVal, ..., funcArgName = deparse(substitute(funcVal)))

wrap_fn_with_params(funcVal, ..., funcArgName = deparse(substitute(funcVal)))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wrap_fn_with_param_arg_+3A_funcval">funcVal</code></td>
<td>
<p>function that the <code>params</code> will be applied to.  The function should follow the api of <code>function(data, mapping, ...){}</code>. <code>funcVal</code> is allowed to be a string of one of the <code>ggally_NAME</code> functions, such as <code>"points"</code> for <code>ggally_points</code> or <code>"facetdensity"</code> for <code>ggally_facetdensity</code>.</p>
</td></tr>
<tr><td><code id="wrap_fn_with_param_arg_+3A_params">params</code></td>
<td>
<p>named vector or list of parameters to be applied to the <code>funcVal</code></p>
</td></tr>
<tr><td><code id="wrap_fn_with_param_arg_+3A_funcargname">funcArgName</code></td>
<td>
<p>name of function to be displayed</p>
</td></tr>
<tr><td><code id="wrap_fn_with_param_arg_+3A_...">...</code></td>
<td>
<p>named parameters to be supplied to <code>wrap_fn_with_param_arg</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>wrap</code> is identical to <code>wrap_fn_with_params</code>.  These function take the new parameters as arguments.
</p>
<p><code>wrapp</code> is identical to <code>wrap_fn_with_param_arg</code>.  These functions take the new parameters as a single list.
</p>
<p>The <code>params</code> and <code>fn</code> attributes are there for debugging purposes.  If either attribute is altered, the function must be re-wrapped to have the changes take effect.
</p>


<h3>Value</h3>

<p>a <code>function(data, mapping, ...){}</code> that will wrap the original function with the parameters applied as arguments
</p>


<h3>Examples</h3>

<pre><code class='language-R'># small function to display plots only if it's interactive
p_ &lt;- GGally::print_if_interactive

# example function that prints 'val'
fn &lt;- function(data, mapping, val = 2) {
  print(val)
}
fn(data = NULL, mapping = NULL) # 2

# wrap function to change default value 'val' to 5 instead of 2
wrapped_fn1 &lt;- wrap(fn, val = 5)
wrapped_fn1(data = NULL, mapping = NULL) # 5
# you may still supply regular values
wrapped_fn1(data = NULL, mapping = NULL, val = 3) # 3

# wrap function to change 'val' to 5 using the arg list
wrapped_fn2 &lt;- wrap_fn_with_param_arg(fn, params = list(val = 5))
wrapped_fn2(data = NULL, mapping = NULL) # 5

# change parameter settings in ggpairs for a particular function
## Goal output:
regularPlot &lt;- ggally_points(
  iris,
  ggplot2::aes(Sepal.Length, Sepal.Width),
  size = 5, color = "red"
)
p_(regularPlot)

# Wrap ggally_points to have parameter values size = 5 and color = 'red'
w_ggally_points &lt;- wrap(ggally_points, size = 5, color = "red")
wrappedPlot &lt;- w_ggally_points(
  iris,
  ggplot2::aes(Sepal.Length, Sepal.Width)
)
p_(wrappedPlot)

# Double check the aes parameters are the same for the geom_point layer
identical(regularPlot$layers[[1]]$aes_params, wrappedPlot$layers[[1]]$aes_params)

# Use a wrapped function in ggpairs
pm &lt;- ggpairs(iris, 1:3, lower = list(continuous = wrap(ggally_points, size = 5, color = "red")))
p_(pm)
pm &lt;- ggpairs(iris, 1:3, lower = list(continuous = w_ggally_points))
p_(pm)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
