<!DOCTYPE html><html lang="en-GB"><head><title>Help for package ggalign</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ggalign}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ggalign-package'><p>ggalign: A 'ggplot2' Extension for Consistent Axis Alignment</p></a></li>
<li><a href='#.link_draw'><p>Define the links to connect a pair of observations</p></a></li>
<li><a href='#.mark_draw'><p>Define the links to connect the marked observations</p></a></li>
<li><a href='#+2B+2Cggalign_plot+2CANY-method'><p>Add custom objects to ggalign plot</p></a></li>
<li><a href='#+24+2CLayoutProto-method'><p>Subset a <code>Layout</code> object</p></a></li>
<li><a href='#active'><p>Plot Adding Context Settings</p></a></li>
<li><a href='#align'><p>Create a New <code>align</code> Object</p></a></li>
<li><a href='#align_dendro'><p>Plot dendrogram tree</p></a></li>
<li><a href='#align_group'><p>Group and align observations based on a group vector</p></a></li>
<li><a href='#align_hclust'><p>Reorder or Group observations based on hierarchical clustering</p></a></li>
<li><a href='#align_kmeans'><p>Split observations by k-means clustering groups.</p></a></li>
<li><a href='#align_order'><p>Order observations based on weights</p></a></li>
<li><a href='#align_order2'><p>Reorders layout observations based on specific statistics.</p></a></li>
<li><a href='#align_panel'><p>Create ggplot object with layout panel data</p></a></li>
<li><a href='#align_phylo'><p>Plot Phylogenetics tree</p></a></li>
<li><a href='#align_plots'><p>Arrange multiple plots into a grid</p></a></li>
<li><a href='#alignpatch'><p>Prepare plots to be aligned with <code>align_plots</code></p></a></li>
<li><a href='#area'><p>Define the plotting areas in <code>align_plots</code></p></a></li>
<li><a href='#circle_layout'><p>Arrange plots in a circular layout</p></a></li>
<li><a href='#circle_switch'><p>Determine the active context of circle layout</p></a></li>
<li><a href='#continuous_limits'><p>Set continuous limits for the layout</p></a></li>
<li><a href='#coord_circle'><p>Polar Coordinates with Enhanced Controls</p></a></li>
<li><a href='#cross_link'><p>Add a plot to connect selected observations</p></a></li>
<li><a href='#cross_mark'><p>Add a plot to annotate observations</p></a></li>
<li><a href='#cross_none'><p>Reset layout ordering and panel group</p></a></li>
<li><a href='#draw_key_gshape'><p>Key glyphs for legends</p></a></li>
<li><a href='#element_curve'><p>Theme curve elements</p></a></li>
<li><a href='#element_polygon'><p>Theme Polygon elements</p></a></li>
<li><a href='#element_vec'><p>Apply a function to the fields of an element object</p></a></li>
<li><a href='#facet_sector'><p>Polar coordinates with Facet support</p></a></li>
<li><a href='#fortify_data_frame'><p>Build a data frame</p></a></li>
<li><a href='#fortify_data_frame.character'><p>Build a data frame</p></a></li>
<li><a href='#fortify_data_frame.default'><p>Build a data frame</p></a></li>
<li><a href='#fortify_data_frame.dendrogram'><p>Build a data frame</p></a></li>
<li><a href='#fortify_data_frame.matrix'><p>Build a data frame</p></a></li>
<li><a href='#fortify_data_frame.phylo'><p>Build a data frame</p></a></li>
<li><a href='#fortify_matrix'><p>Build a Matrix</p></a></li>
<li><a href='#fortify_matrix.default'><p>Build a Matrix</p></a></li>
<li><a href='#fortify_matrix.GISTIC'><p>Build a matrix from a maftools object</p></a></li>
<li><a href='#fortify_matrix.list_upset'><p>Build a Matrix for UpSet plot</p></a></li>
<li><a href='#fortify_matrix.MAF'><p>Build a Matrix for OncoPrint</p></a></li>
<li><a href='#fortify_matrix.matrix'><p>Build a matrix</p></a></li>
<li><a href='#fortify_matrix.matrix_oncoplot'><p>Build a Matrix for OncoPrint</p></a></li>
<li><a href='#fortify_matrix.matrix_upset'><p>Build a Matrix for UpSet plot</p></a></li>
<li><a href='#fortify_matrix.phylo'><p>Build a matrix from <code>phylo</code> object</p></a></li>
<li><a href='#free_align'><p>Free from alignment</p></a></li>
<li><a href='#geom_draw'><p>Layer with Grid or Function</p></a></li>
<li><a href='#geom_gshape'><p>Layer with a customized shape graphic using grid functions.</p></a></li>
<li><a href='#geom_pie'><p>Pie charts</p></a></li>
<li><a href='#geom_rect3d'><p>Add z-aesthetic for geom_tile</p></a></li>
<li><a href='#geom_subrect'><p>Subdivide Rectangles</p></a></li>
<li><a href='#ggalign'><p>Add ggplot by Aligning discrete or continuous variable</p></a></li>
<li><a href='#ggalign_attr'><p>Get Data from the Attribute Attached by ggalign</p></a></li>
<li><a href='#ggalign_data_set'><p>Attach supplementary data and levels for ggalign</p></a></li>
<li><a href='#ggalign_stat'><p>Get the statistics from the layout</p></a></li>
<li><a href='#ggalignGrob'><p>Generate a plot grob.</p></a></li>
<li><a href='#ggcross'><p>Connect two layout crosswise</p></a></li>
<li><a href='#ggfree'><p>Add ggplot to layout without alignment</p></a></li>
<li><a href='#ggmark'><p>Add a plot to annotate selected observations</p></a></li>
<li><a href='#ggoncoplot'><p>Create an OncoPrint</p></a></li>
<li><a href='#ggupset'><p>Create an UpSet plot</p></a></li>
<li><a href='#ggwrap'><p>Wrap Arbitrary Graphics to ggplot</p></a></li>
<li><a href='#hclust2'><p>Generate Tree Structures with Hierarchical Clustering</p></a></li>
<li><a href='#heatmap_layout'><p>Create a heatmap</p></a></li>
<li><a href='#inset'><p>Create a ggplot inset</p></a></li>
<li><a href='#is_layout'><p>Reports whether <code>x</code> is layout object</p></a></li>
<li><a href='#layer_order'><p>Change the layer adding order</p></a></li>
<li><a href='#layout_annotation'><p>Modify components of the layout</p></a></li>
<li><a href='#layout_design'><p>Define the grid to compose plots in</p></a></li>
<li><a href='#layout_expand'><p>Set Expansion for the Layout</p></a></li>
<li><a href='#layout_title'><p>Annotate the whole layout</p></a></li>
<li><a href='#layout-operator'><p>Layout operator</p></a></li>
<li><a href='#LayoutProto-class'><p>A <code>Layout</code> object</p></a></li>
<li><a href='#link_draw'><p>Define the links to connect a pair of observations</p></a></li>
<li><a href='#link_line'><p>Link the paired observations with a line</p></a></li>
<li><a href='#link_tetragon'><p>Link the paired observations with a quadrilateral</p></a></li>
<li><a href='#mark_draw'><p>Define the links to connect the marked observations</p></a></li>
<li><a href='#mark_line'><p>Link the observations and the panel with a line</p></a></li>
<li><a href='#mark_tetragon'><p>Link the observations and the panel with a quadrilateral</p></a></li>
<li><a href='#mark_triangle'><p>Link the observations and the panel with a triangle</p></a></li>
<li><a href='#memo_order'><p>Sort matrix for better visualization</p></a></li>
<li><a href='#new_tune'><p>Change the shape of the input object</p></a></li>
<li><a href='#no_expansion'><p>Remove scale expansion</p></a></li>
<li><a href='#order2'><p>Ordering Permutation</p></a></li>
<li><a href='#pair_links'><p>Helper function to create pairs of observation groups</p></a></li>
<li><a href='#patch'><p>Convert Object into a Grob</p></a></li>
<li><a href='#patch_titles'><p>Add patch titles to plot borders</p></a></li>
<li><a href='#patch.alignpatches'><p>Convert Object into a Grob</p></a></li>
<li><a href='#patch.formula'><p>Convert Object into a Grob</p></a></li>
<li><a href='#patch.ggplot'><p>Convert Object into a Grob</p></a></li>
<li><a href='#patch.grob'><p>Convert Object into a Grob</p></a></li>
<li><a href='#patch.Heatmap'><p>Convert Object into a Grob</p></a></li>
<li><a href='#patch.patch'><p>Convert Object into a Grob</p></a></li>
<li><a href='#patch.patch_ggplot'><p>Convert Object into a Grob</p></a></li>
<li><a href='#patch.patchwork'><p>Convert Object into a Grob</p></a></li>
<li><a href='#patch.pheatmap'><p>Convert Object into a Grob</p></a></li>
<li><a href='#patch.recordedplot'><p>Convert Object into a Grob</p></a></li>
<li><a href='#patch.trellis'><p>Convert Object into a Grob</p></a></li>
<li><a href='#quad_active'><p>Determine the Active Context of Quad-Layout</p></a></li>
<li><a href='#quad_init'><p>Initialize Quad-Layout Annotation</p></a></li>
<li><a href='#quad_layout'><p>Arrange plots in the quad-side of a main plot</p></a></li>
<li><a href='#quad_switch'><p>Determine the Active Context of Quad-Layout</p></a></li>
<li><a href='#raster_magick'><p>Rasterize the input object</p></a></li>
<li><a href='#raster_magick_interal'><p>Rasterize the input object</p></a></li>
<li><a href='#read_example'><p>Read Example Data</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#scale_gshape_manual'><p>Scale for <code>gshape</code> aesthetic</p></a></li>
<li><a href='#scheme_align'><p>Align Specifications in the Layout</p></a></li>
<li><a href='#scheme_data'><p>Plot data Specifications</p></a></li>
<li><a href='#scheme_theme'><p>Plot default theme</p></a></li>
<li><a href='#show+2Cggalign_plot-method'><p>Show <code>ggalign_plot</code> information</p></a></li>
<li><a href='#show+2CLayoutProto-method'><p>Print Layout object</p></a></li>
<li><a href='#stack_cross'><p>Arrange plots crosswise horizontally or vertically</p></a></li>
<li><a href='#stack_layout'><p>Arrange plots horizontally or vertically</p></a></li>
<li><a href='#stack_switch'><p>Determine the active context of stack layout</p></a></li>
<li><a href='#theme_no_axes'><p>Remove axis elements</p></a></li>
<li><a href='#tune'><p>Change the shape of the input object</p></a></li>
<li><a href='#tune.list'><p>Convert the shape of a list for fortify method</p></a></li>
<li><a href='#tune.MAF'><p>Convert the shape of a MAF for fortify method</p></a></li>
<li><a href='#tune.matrix'><p>Convert the shape of a matrix for fortify method</p></a></li>
<li><a href='#with_quad'><p>Modify operated Context in <code>quad_layout()</code></p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>A 'ggplot2' Extension for Consistent Axis Alignment</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A 'ggplot2' extension offers various tools the creation of complex,
    multi-plot visualizations. Built on the familiar grammar of graphics, it
    provides intuitive tools to align and organize plots, making it ideal for
    complex visualizations. It excels in multi-omics research—such as genomics
    and microbiomes—by simplifying the visualization of intricate relationships
    between datasets, for example, linking genes to pathways. Whether you need
    to stack plots, arrange them around a central figure, or use a circular
    layout, 'ggalign' delivers flexibility and accuracy with minimal effort.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Yunuuuu/ggalign">https://github.com/Yunuuuu/ggalign</a>,
<a href="https://yunuuuu.github.io/ggalign/">https://yunuuuu.github.io/ggalign/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Yunuuuu/ggalign/issues">https://github.com/Yunuuuu/ggalign/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>ggplot2 (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>cli, grDevices, grid, gtable, lifecycle, methods, rlang,
stats, utils, vctrs (&ge; 0.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggrastr, gridGraphics, magick, patchwork, ragg, knitr,
rmarkdown, scales, testthat (&ge; 3.0.0), vdiffr</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>Collate:</td>
<td>'active.R' 'plot-.R' 'align-.R' 'align-hclust.R'
'align-dendrogram.R' 'align-group.R' 'align-kmeans.R'
'align-order.R' 'align-order2.R' 'align-phylo.R'
'alignpatch-.R' 'alignpatch-align_plots.R'
'alignpatch-alignpatches.R' 'alignpatch-area.R'
'alignpatch-build.R' 'alignpatch-free-align.R'
'alignpatch-free-border.R' 'alignpatch-free-guide.R'
'alignpatch-free-lab.R' 'alignpatch-free-space.R'
'alignpatch-free-vp.R' 'alignpatch-ggplot2.R'
'alignpatch-guides.R' 'alignpatch-inset.R' 'alignpatch-patch.R'
'alignpatch-patchwork.R' 'alignpatch-title.R'
'alignpatch-wrap.R' 'attributes.R' 'cross-.R' 'cross-link.R'
'cross-mark.R' 'cross-none.R' 'fortify-data-frame-.R'
'fortify-data-frame-dendrogram.R' 'fortify-matrix-.R'
'fortify-matrix-list.R' 'fortify-matrix-maftools.R'
'fortify-matrix-matrix.R' 'geom-draw.R' 'geom-gshape.R'
'geom-pie.R' 'geom-subrect.R' 'geom-tile3d.R'
'ggalign-package.R' 'ggalign.R' 'ggcross.R' 'ggfree.R'
'ggmark.R' 'ggplot-coord-circle.R' 'ggplot-facet-sector.R'
'ggplot-helper.R' 'ggplot-theme.R' 'ggplot-utils.R'
'import-standalone-assert.R' 'import-standalone-obj-type.R'
'import-standalone-pkg.R' 'import-standalone-purrr.R'
'import-standalone-tibble.R' 'layer-order.R' 'layout-.R'
'layout-align.R' 'layout-chain-.R' 'layout-chain-circle-.R'
'layout-chain-circle-build.R' 'layout-chain-circle-switch.R'
'layout-chain-operator.R' 'layout-chain-stack-.R'
'layout-chain-stack-build.R' 'layout-chain-stack-composer.R'
'layout-chain-stack-cross.R' 'layout-chain-stack-switch.R'
'layout-quad-.R' 'layout-heatmap-.R' 'layout-heatmap-build.R'
'layout-heatmap-oncoplot.R' 'layout-quad-operator.R'
'layout-operator.R' 'layout-quad-add.R' 'layout-quad-build.R'
'layout-quad-switch.R' 'layout-quad-upset.R' 'link.R' 'mark.R'
'object-name.R' 'pair-links.R' 'raster-magick.R' 'rasterise.R'
'scheme-.R' 'scheme-align.R' 'scheme-data.R' 'scheme-theme.R'
'tune.R' 'utils-assert.R' 'utils-grid.R' 'utils-rd.R' 'utils.R'
'with_quad.R' 'zzz.R'</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-14 17:34:56 UTC; yun</td>
</tr>
<tr>
<td>Author:</td>
<td>Yun Peng <a href="https://orcid.org/0000-0003-2801-3332"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Shixiang Wang <a href="https://orcid.org/0000-0001-9855-7357"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Guangchuang Yu <a href="https://orcid.org/0000-0002-6485-8781"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yun Peng &lt;yunyunp96@163.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-15 23:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ggalign-package'>ggalign: A 'ggplot2' Extension for Consistent Axis Alignment</h2><span id='topic+ggalign-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>A 'ggplot2' extension offers various tools the creation of complex, multi-plot visualizations. Built on the familiar grammar of graphics, it provides intuitive tools to align and organize plots, making it ideal for complex visualizations. It excels in multi-omics research—such as genomics and microbiomes—by simplifying the visualization of intricate relationships between datasets, for example, linking genes to pathways. Whether you need to stack plots, arrange them around a central figure, or use a circular layout, 'ggalign' delivers flexibility and accuracy with minimal effort.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Yun Peng <a href="mailto:yunyunp96@163.com">yunyunp96@163.com</a> (<a href="https://orcid.org/0000-0003-2801-3332">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a> (<a href="https://orcid.org/0000-0001-9855-7357">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Guangchuang Yu <a href="mailto:guangchuangyu@gmail.com">guangchuangyu@gmail.com</a> (<a href="https://orcid.org/0000-0002-6485-8781">ORCID</a>) [thesis advisor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/Yunuuuu/ggalign">https://github.com/Yunuuuu/ggalign</a>
</p>
</li>
<li> <p><a href="https://yunuuuu.github.io/ggalign/">https://yunuuuu.github.io/ggalign/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/Yunuuuu/ggalign/issues">https://github.com/Yunuuuu/ggalign/issues</a>
</p>
</li></ul>


<hr>
<h2 id='.link_draw'>Define the links to connect a pair of observations</h2><span id='topic+.link_draw'></span>

<h3>Description</h3>

<p>A base version of <code><a href="#topic+link_draw">link_draw()</a></code>, optimized for performance. This function
serves as the foundation for building other <code style="white-space: pre;">&#8288;link_*&#8288;</code> functions that manage
the drawing of links between pairs of observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.link_draw(.draw, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id=".link_draw_+3A_.draw">.draw</code></td>
<td>
<p>A function used to draw the links. The function must return a
<code><a href="grid.html#topic+grid.grob">grob()</a></code> object. If the function does not return a valid
<code>grob</code>, no drawing will occur. The input data for the function should be
a list, where each item is a data frame containing the coordinates of
the pair of observations.</p>
</td></tr>
<tr><td><code id=".link_draw_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; A list of formulas, where each side
of the formula should be an <code>integer</code> or <code>character</code> index of the original
data, or a <code>range_link()</code> object defining the linked observations. Use
<code>NULL</code> to indicate no link on that side. You can also combine these by
wrapping them into a single <code>list()</code>. If only the left-hand side of the
formula exists, you can input it directly. For integer indices, wrap them
with <code><a href="base.html#topic+I">I()</a></code> to use the ordering from the layout. You can also use
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code> to inherit values from the other group.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+link_draw">link_draw()</a></code>
</p>

<hr>
<h2 id='.mark_draw'>Define the links to connect the marked observations</h2><span id='topic+.mark_draw'></span>

<h3>Description</h3>

<p>A base version of <code><a href="#topic+mark_draw">mark_draw</a></code>, designed for performance optimization. This
function is used to build other <code style="white-space: pre;">&#8288;mark_*&#8288;</code> functions that manage the drawing of
links between marked observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.mark_draw(.draw, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id=".mark_draw_+3A_.draw">.draw</code></td>
<td>
<p>A function used to draw the links. The function must return a
<code><a href="grid.html#topic+grid.grob">grob()</a></code> object. If the function does not return a valid
<code>grob</code>, nothing will be drawn. The input data for the function contains a
list, where each item is a list of two data frames: one for the panel side
coordinates (<code>"panel"</code>) and one for the marked observations coordinates
(<code>"link"</code>).</p>
</td></tr>
<tr><td><code id=".mark_draw_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; A list of formulas, where each side
of the formula should be an <code>integer</code> or <code>character</code> index of the original
data, or a <code>range_link()</code> object defining the linked observations. Use
<code>NULL</code> to indicate no link on that side. You can also combine these by
wrapping them into a single <code>list()</code>. If only the left-hand side of the
formula exists, you can input it directly. For integer indices, wrap them
with <code><a href="base.html#topic+I">I()</a></code> to use the ordering from the layout. You can also use
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code> to inherit values from the other group.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+mark_draw">mark_draw()</a></code>
</p>

<hr>
<h2 id='+2B+2Cggalign_plot+2CANY-method'>Add custom objects to ggalign plot</h2><span id='topic++2B+2Cggalign_plot+2CANY-method'></span>

<h3>Description</h3>

<p>Add custom objects to ggalign plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ggalign_plot,ANY'
e1 + e2
</code></pre>

<hr>
<h2 id='+24+2CLayoutProto-method'>Subset a <code>Layout</code> object</h2><span id='topic++24+2CLayoutProto-method'></span>

<h3>Description</h3>

<p>Used by <code><a href="ggplot2.html#topic+ggplot_build">ggplot_build</a></code> and
<code><a href="ggplot2.html#topic+ggsave">ggsave</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'LayoutProto'
x$name
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B24+2B2CLayoutProto-method_+3A_x">x</code></td>
<td>
<p>A <code>Layout</code> object</p>
</td></tr>
<tr><td><code id="+2B24+2B2CLayoutProto-method_+3A_name">name</code></td>
<td>
<p>A string of slot name in <code>Layout</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The slot value.
</p>

<hr>
<h2 id='active'>Plot Adding Context Settings</h2><span id='topic+active'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>These settings control the behavior of the plot when added to a layout, as
well as the arrangement of individual plot areas within the layout.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>active(order = waiver(), use = waiver(), name = waiver())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="active_+3A_order">order</code></td>
<td>
<p>An integer specifying the order of the plot area within the
layout.</p>
</td></tr>
<tr><td><code id="active_+3A_use">use</code></td>
<td>
<p>A logical (<code>TRUE</code>/<code>FALSE</code>) indicating whether to set the
active context to the current plot when added to a layout. If <code>TRUE</code>,
any subsequent <code>ggplot</code> elements will be applied to this plot.</p>
</td></tr>
<tr><td><code id="active_+3A_name">name</code></td>
<td>
<p>A string specifying the plot's name, useful for switching active
contexts through the <code>what</code> argument in functions like
<code><a href="#topic+quad_anno">quad_anno()</a></code>/<code><a href="#topic+stack_switch">stack_switch()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, the active context is set only for functions that add plot areas.
This allows other <code>ggplot2</code> elements-such as <code>geoms</code>, <code>stats</code>, <code>scales</code>, or
<code>themes</code>- to be seamlessly added to the current plot area.
</p>
<p>The default ordering of the plot areas is from top to bottom or from left to
right, depending on the layout orientation. However, users can customize this
order using the <code>order</code> argument.
</p>

<hr>
<h2 id='align'>Create a New <code>align</code> Object</h2><span id='topic+align'></span><span id='topic+Align'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>An <code>Align</code> object interacts with the <code>Layout</code> object to reorder or
split observations and, in some cases, add plot components to the <code>Layout</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align(
  align,
  data = NULL,
  ...,
  plot = NULL,
  size = NULL,
  schemes = NULL,
  no_axes = NULL,
  active = NULL,
  call = caller_call()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_+3A_align">align</code></td>
<td>
<p>An <code>Align</code> object.</p>
</td></tr>
<tr><td><code id="align_+3A_data">data</code></td>
<td>
<p>The following options can be used:
</p>

<ul>
<li> <p><code>NULL</code>: No data is set.
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+waiver">waiver()</a></code>: Inherits the data from the layout matrix.
</p>
</li>
<li><p> A <code>function</code> (including purrr-like lambda syntax): Applied to the layout
matrix to transform the data before use. To transform the final plot
data, please use <code><a href="#topic+scheme_data">scheme_data()</a></code>.
</p>
</li>
<li><p> A <code>matrix</code>, <code>data.frame</code>, or atomic vector.
</p>
</li></ul>
</td></tr>
<tr><td><code id="align_+3A_...">...</code></td>
<td>
<p>Additional fields passed to the <code>align</code> object.</p>
</td></tr>
<tr><td><code id="align_+3A_plot">plot</code></td>
<td>
<p>A ggplot object.</p>
</td></tr>
<tr><td><code id="align_+3A_size">size</code></td>
<td>
<p>The relative size of the plot, can be specified as a
<code><a href="grid.html#topic+unit">unit()</a></code>. Note that for <code><a href="#topic+circle_layout">circle_layout()</a></code>, all size values
will be interpreted as relative sizes, as this layout type adjusts based on
the available space in the circular arrangement.</p>
</td></tr>
<tr><td><code id="align_+3A_schemes">schemes</code></td>
<td>
<p>Options for <code>schemes</code>:
</p>

<ul>
<li> <p><code>NULL</code>: Used when <code>align</code> do not add a plot.
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+waiver">waiver()</a></code>: Try to infer <code>schemes</code> based on <code>data</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="align_+3A_no_axes">no_axes</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a> Logical; if <code>TRUE</code>,
removes axes elements for the alignment axis using <code><a href="#topic+theme_no_axes">theme_no_axes()</a></code>. By
default, will use the option-
<code>"ggalign.align_no_axes"</code>.</p>
</td></tr>
<tr><td><code id="align_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
<tr><td><code id="align_+3A_call">call</code></td>
<td>
<p>The <code>call</code> used to construct the <code>Align</code> object, for reporting
messages.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each of the <code style="white-space: pre;">&#8288;Align*&#8288;</code> objects is just a <code><a href="ggplot2.html#topic+ggproto">ggproto()</a></code>
object, descended from the top-level <code>Align</code>, and each implements
various methods and fields.
</p>
<p>To create a new type of <code style="white-space: pre;">&#8288;Align*&#8288;</code> object, you typically will want to
override one or more of the following:
</p>

<ul>
<li> <p><code>setup_params</code>: Prepare parameter or check parameters used by this plot.
</p>
</li>
<li> <p><code>setup_data</code>: Prepare data used by this plot.
</p>
</li>
<li> <p><code>compute</code>: A method used to compute statistics.
</p>
</li>
<li> <p><code>align</code>: A method used to group observations into panel or reorder
observations.
</p>
</li>
<li> <p><code>draw</code>: A method used to draw the plot. Must return a <code>ggplot</code> object.
</p>
</li></ul>



<h3>Value</h3>

<p>A new <code>ggalign_plot</code> object.
</p>


<h3>Discrete Axis Alignment</h3>

<p>It is important to note that we consider rows as observations, meaning
<code>vec_size(data)</code>/<code>NROW(data)</code> must match the number of observations along the
axis used for alignment (x-axis for a vertical stack layout, y-axis for a
horizontal stack layout).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>align_dendro()
</code></pre>

<hr>
<h2 id='align_dendro'>Plot dendrogram tree</h2><span id='topic+align_dendro'></span>

<h3>Description</h3>

<p>Plot dendrogram tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_dendro(
  mapping = aes(),
  ...,
  distance = "euclidean",
  method = "complete",
  use_missing = "pairwise.complete.obs",
  reorder_dendrogram = FALSE,
  merge_dendrogram = FALSE,
  reorder_group = FALSE,
  k = NULL,
  h = NULL,
  cutree = NULL,
  plot_dendrogram = TRUE,
  plot_cut_height = NULL,
  root = NULL,
  center = FALSE,
  type = "rectangle",
  size = NULL,
  data = NULL,
  no_axes = NULL,
  active = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_dendro_+3A_mapping">mapping</code></td>
<td>
<p>Default list of aesthetic mappings to use for plot.
If not specified, must be supplied in each layer added to the plot.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; Additional arguments passed to
<code><a href="ggplot2.html#topic+geom_segment">geom_segment()</a></code>.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_distance">distance</code></td>
<td>
<p>A string of distance measure to be used. This must be one of
<code>"euclidean"</code>, <code>"maximum"</code>, <code>"manhattan"</code>, <code>"canberra"</code>, <code>"binary"</code> or
<code>"minkowski"</code>.  Correlation coefficient can be also used, including
<code>"pearson"</code>, <code>"spearman"</code> or <code>"kendall"</code>. In this way, <code>1 - cor</code> will be used
as the distance. In addition, you can also provide a <code><a href="stats.html#topic+dist">dist</a></code>
object directly or a function return a <code><a href="stats.html#topic+dist">dist</a></code> object. Use
<code>NULL</code>, if you don't want to calculate the distance.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_method">method</code></td>
<td>
<p>A string of the agglomeration method to be used. This should be
(an unambiguous abbreviation of) one of <code>"ward.D"</code>, <code>"ward.D2"</code>, <code>"single"</code>,
<code>"complete"</code>, <code>"average"</code> (= UPGMA), <code>"mcquitty"</code> (= WPGMA), <code>"median"</code> (=
WPGMC) or <code>"centroid"</code> (= UPGMC). You can also provide a function which
accepts the calculated distance (or the input matrix if <code>distance</code> is <code>NULL</code>)
and returns a <code><a href="stats.html#topic+hclust">hclust</a></code> object. Alternative, you can supply
an object which can be coerced to <code><a href="stats.html#topic+hclust">hclust</a></code>.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_use_missing">use_missing</code></td>
<td>
<p>An optional character string giving a method for computing
covariances in the presence of missing values. This must be (an abbreviation
of) one of the strings <code>"everything"</code>, <code>"all.obs"</code>, <code>"complete.obs"</code>,
<code>"na.or.complete"</code>, or <code>"pairwise.complete.obs"</code>. Only used when <code>distance</code>
is a correlation coefficient string.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_reorder_dendrogram">reorder_dendrogram</code></td>
<td>
<p>A single boolean value indicating whether to
reorder the dendrogram based on the means. Alternatively, you can provide a
custom function that accepts an <code><a href="stats.html#topic+hclust">hclust</a></code> object and the data
used to generate the tree, returning either an <code><a href="stats.html#topic+hclust">hclust</a></code> or
<code><a href="stats.html#topic+dendrogram">dendrogram</a></code> object. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_merge_dendrogram">merge_dendrogram</code></td>
<td>
<p>A single boolean value, indicates whether we should
merge multiple dendrograms, only used when previous groups have been
established. Default: <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_reorder_group">reorder_group</code></td>
<td>
<p>A single boolean value, indicates whether we should do
Hierarchical Clustering between groups, only used when previous groups have
been established. Default: <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_k">k</code></td>
<td>
<p>An integer scalar indicates the desired number of groups.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_h">h</code></td>
<td>
<p>A numeric scalar indicates heights where the tree should be cut.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_cutree">cutree</code></td>
<td>
<p>A function used to cut the <code><a href="stats.html#topic+hclust">hclust</a></code> tree. It
should accept four arguments: the <code><a href="stats.html#topic+hclust">hclust</a></code> tree object,
<code>distance</code> (only applicable when <code>method</code> is a string or a function for
performing hierarchical clustering), <code>k</code> (the number of clusters), and <code>h</code>
(the height at which to cut the tree). By default,
<code><a href="stats.html#topic+cutree">cutree()</a></code> is used.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_plot_dendrogram">plot_dendrogram</code></td>
<td>
<p>A boolean value indicates whether plot the dendrogram
tree.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_plot_cut_height">plot_cut_height</code></td>
<td>
<p>A boolean value indicates whether plot the cut height.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_root">root</code></td>
<td>
<p>A length one string or numeric indicates the root branch.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_center">center</code></td>
<td>
<p>A boolean value. if <code>TRUE</code>, nodes are plotted centered with
respect to all leaves/tips in the branch. Otherwise (default), plot them in
the middle of the direct child nodes.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_type">type</code></td>
<td>
<p>A string indicates the plot type, <code>"rectangle"</code> or <code>"triangle"</code>.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_size">size</code></td>
<td>
<p>The relative size of the plot, can be specified as a
<code><a href="grid.html#topic+unit">unit()</a></code>. Note that for <code><a href="#topic+circle_layout">circle_layout()</a></code>, all size values
will be interpreted as relative sizes, as this layout type adjusts based on
the available space in the circular arrangement.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_data">data</code></td>
<td>
<p>A matrix-like object. By default, it inherits from the layout
<code>matrix</code>.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_no_axes">no_axes</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a> Logical; if <code>TRUE</code>,
removes axes elements for the alignment axis using <code><a href="#topic+theme_no_axes">theme_no_axes()</a></code>. By
default, will use the option-
<code>"ggalign.align_no_axes"</code>.</p>
</td></tr>
<tr><td><code id="align_dendro_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>ggplot2 specification</h3>

<p><code>align_dendro</code> initializes a ggplot <code>data</code> and <code>mapping</code>.
</p>
<p>The internal <code>ggplot</code> object will always use a default mapping of
<code>aes(x = .data$x, y = .data$y)</code>.
</p>
<p>The default ggplot data is the <code>node</code> coordinates with <code>edge</code> data attached
in <code><a href="#topic+ggalign_attr">ggalign</a></code> attribute, in addition, a
<code><a href="ggplot2.html#topic+geom_segment">geom_segment</a></code> layer with a data frame of the <code>edge</code>
coordinates will be added when <code>plot_dendrogram = TRUE</code>.
</p>
<p>See <code><a href="#topic+fortify_data_frame.dendrogram">fortify_data_frame.dendrogram()</a></code> for details.
</p>


<h3>Discrete Axis Alignment</h3>

<p>It is important to note that we consider rows as observations, meaning
<code>vec_size(data)</code>/<code>NROW(data)</code> must match the number of observations along the
axis used for alignment (x-axis for a vertical stack layout, y-axis for a
horizontal stack layout).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># align_dendro will always add a plot area
ggheatmap(matrix(rnorm(81), nrow = 9)) +
    anno_top() +
    align_dendro()
ggheatmap(matrix(rnorm(81), nrow = 9)) +
    anno_top() +
    align_dendro(k = 3L)

</code></pre>

<hr>
<h2 id='align_group'>Group and align observations based on a group vector</h2><span id='topic+align_group'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Splits observations into groups, with slice ordering based on group levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_group(group, active = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_group_+3A_group">group</code></td>
<td>
<p>A character define the groups of the observations.</p>
</td></tr>
<tr><td><code id="align_group_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1L)
small_mat &lt;- matrix(rnorm(81), nrow = 9)
ggheatmap(small_mat) +
    anno_top() +
    align_group(sample(letters[1:4], ncol(small_mat), replace = TRUE))
</code></pre>

<hr>
<h2 id='align_hclust'>Reorder or Group observations based on hierarchical clustering</h2><span id='topic+align_hclust'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function aligns observations within the layout according to a
hierarchical clustering tree, enabling reordering or grouping of elements
based on clustering results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_hclust(
  distance = "euclidean",
  method = "complete",
  use_missing = "pairwise.complete.obs",
  reorder_dendrogram = FALSE,
  reorder_group = FALSE,
  k = NULL,
  h = NULL,
  cutree = NULL,
  data = NULL,
  active = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_hclust_+3A_distance">distance</code></td>
<td>
<p>A string of distance measure to be used. This must be one of
<code>"euclidean"</code>, <code>"maximum"</code>, <code>"manhattan"</code>, <code>"canberra"</code>, <code>"binary"</code> or
<code>"minkowski"</code>.  Correlation coefficient can be also used, including
<code>"pearson"</code>, <code>"spearman"</code> or <code>"kendall"</code>. In this way, <code>1 - cor</code> will be used
as the distance. In addition, you can also provide a <code><a href="stats.html#topic+dist">dist</a></code>
object directly or a function return a <code><a href="stats.html#topic+dist">dist</a></code> object. Use
<code>NULL</code>, if you don't want to calculate the distance.</p>
</td></tr>
<tr><td><code id="align_hclust_+3A_method">method</code></td>
<td>
<p>A string of the agglomeration method to be used. This should be
(an unambiguous abbreviation of) one of <code>"ward.D"</code>, <code>"ward.D2"</code>, <code>"single"</code>,
<code>"complete"</code>, <code>"average"</code> (= UPGMA), <code>"mcquitty"</code> (= WPGMA), <code>"median"</code> (=
WPGMC) or <code>"centroid"</code> (= UPGMC). You can also provide a function which
accepts the calculated distance (or the input matrix if <code>distance</code> is <code>NULL</code>)
and returns a <code><a href="stats.html#topic+hclust">hclust</a></code> object. Alternative, you can supply
an object which can be coerced to <code><a href="stats.html#topic+hclust">hclust</a></code>.</p>
</td></tr>
<tr><td><code id="align_hclust_+3A_use_missing">use_missing</code></td>
<td>
<p>An optional character string giving a method for computing
covariances in the presence of missing values. This must be (an abbreviation
of) one of the strings <code>"everything"</code>, <code>"all.obs"</code>, <code>"complete.obs"</code>,
<code>"na.or.complete"</code>, or <code>"pairwise.complete.obs"</code>. Only used when <code>distance</code>
is a correlation coefficient string.</p>
</td></tr>
<tr><td><code id="align_hclust_+3A_reorder_dendrogram">reorder_dendrogram</code></td>
<td>
<p>A single boolean value indicating whether to
reorder the dendrogram based on the means. Alternatively, you can provide a
custom function that accepts an <code><a href="stats.html#topic+hclust">hclust</a></code> object and the data
used to generate the tree, returning either an <code><a href="stats.html#topic+hclust">hclust</a></code> or
<code><a href="stats.html#topic+dendrogram">dendrogram</a></code> object. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="align_hclust_+3A_reorder_group">reorder_group</code></td>
<td>
<p>A single boolean value, indicates whether we should do
Hierarchical Clustering between groups, only used when previous groups have
been established. Default: <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="align_hclust_+3A_k">k</code></td>
<td>
<p>An integer scalar indicates the desired number of groups.</p>
</td></tr>
<tr><td><code id="align_hclust_+3A_h">h</code></td>
<td>
<p>A numeric scalar indicates heights where the tree should be cut.</p>
</td></tr>
<tr><td><code id="align_hclust_+3A_cutree">cutree</code></td>
<td>
<p>A function used to cut the <code><a href="stats.html#topic+hclust">hclust</a></code> tree. It
should accept four arguments: the <code><a href="stats.html#topic+hclust">hclust</a></code> tree object,
<code>distance</code> (only applicable when <code>method</code> is a string or a function for
performing hierarchical clustering), <code>k</code> (the number of clusters), and <code>h</code>
(the height at which to cut the tree). By default,
<code><a href="stats.html#topic+cutree">cutree()</a></code> is used.</p>
</td></tr>
<tr><td><code id="align_hclust_+3A_data">data</code></td>
<td>
<p>A matrix-like object. By default, it inherits from the layout
<code>matrix</code>.</p>
</td></tr>
<tr><td><code id="align_hclust_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>Discrete Axis Alignment</h3>

<p>It is important to note that we consider rows as observations, meaning
<code>vec_size(data)</code>/<code>NROW(data)</code> must match the number of observations along the
axis used for alignment (x-axis for a vertical stack layout, y-axis for a
horizontal stack layout).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hclust2">hclust2()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># align_hclust won't add a dendrogram
ggheatmap(matrix(rnorm(81), nrow = 9)) +
    anno_top() +
    align_hclust(k = 3L)
</code></pre>

<hr>
<h2 id='align_kmeans'>Split observations by k-means clustering groups.</h2><span id='topic+align_kmeans'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Aligns and groups observations based on k-means clustering, enabling
observation splits by cluster groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_kmeans(..., data = NULL, active = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_kmeans_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="stats.html#topic+kmeans">stats::kmeans</a></code>
</p>

<dl>
<dt><code>iter.max</code></dt><dd><p>the maximum number of iterations allowed.</p>
</dd>
<dt><code>nstart</code></dt><dd><p>if <code>centers</code> is a number, how many random sets
should be chosen?</p>
</dd>
<dt><code>algorithm</code></dt><dd><p>character: may be abbreviated.  Note that
<code>"Lloyd"</code> and <code>"Forgy"</code> are alternative names for one
algorithm.</p>
</dd>
<dt><code>trace</code></dt><dd><p>logical or integer number, currently only used in the
default method (<code>"Hartigan-Wong"</code>): if positive (or true),
tracing information on the progress of the algorithm is
produced.  Higher values may produce more tracing information.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="align_kmeans_+3A_data">data</code></td>
<td>
<p>A numeric matrix to be used by k-means. By default, it will
inherit from the layout matrix.</p>
</td></tr>
<tr><td><code id="align_kmeans_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>Discrete Axis Alignment</h3>

<p>It is important to note that we consider rows as observations, meaning
<code>vec_size(data)</code>/<code>NROW(data)</code> must match the number of observations along the
axis used for alignment (x-axis for a vertical stack layout, y-axis for a
horizontal stack layout).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ggheatmap(matrix(rnorm(81), nrow = 9)) +
    anno_top() +
    align_kmeans(3L)
</code></pre>

<hr>
<h2 id='align_order'>Order observations based on weights</h2><span id='topic+align_order'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Ordering observations based on summary weights or a specified ordering
character or integer index.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_order(
  weights = rowMeans,
  ...,
  reverse = FALSE,
  strict = TRUE,
  data = NULL,
  active = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_order_+3A_weights">weights</code></td>
<td>
<p>A summary function which accepts a data and returns the
weights for each observations. Alternatively, you can provide an ordering
index as either an integer or a character. Since characters have been
designated as character indices, if you wish to specify a function name as a
string, you must enclose it with <code><a href="base.html#topic+I">I()</a></code>.</p>
</td></tr>
<tr><td><code id="align_order_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; Additional arguments passed to
function provided in <code>weights</code> argument.</p>
</td></tr>
<tr><td><code id="align_order_+3A_reverse">reverse</code></td>
<td>
<p>A boolean value. Should the sort order be in reverse?</p>
</td></tr>
<tr><td><code id="align_order_+3A_strict">strict</code></td>
<td>
<p>A boolean value indicates whether the order should be strict.
If previous groups has been established, and strict is <code>FALSE</code>, this will
reorder the observations in each group.</p>
</td></tr>
<tr><td><code id="align_order_+3A_data">data</code></td>
<td>
<p>A <code>matrix</code>, <code style="white-space: pre;">&#8288;data frame&#8288;</code>, or atomic vector used as the input for
the <code>weights</code> function. Alternatively, you can specify a <code>function</code>
(including purrr-like lambda syntax) that will be applied to the layout
matrix, transforming it as necessary for weight calculations. By default, it
will inherit from the layout matrix.</p>
</td></tr>
<tr><td><code id="align_order_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>Discrete Axis Alignment</h3>

<p>It is important to note that we consider rows as observations, meaning
<code>vec_size(data)</code>/<code>NROW(data)</code> must match the number of observations along the
axis used for alignment (x-axis for a vertical stack layout, y-axis for a
horizontal stack layout).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ggheatmap(matrix(rnorm(81), nrow = 9)) +
    anno_left() +
    align_order(I("rowMeans"))
</code></pre>

<hr>
<h2 id='align_order2'>Reorders layout observations based on specific statistics.</h2><span id='topic+align_order2'></span>

<h3>Description</h3>

<p>Reorders layout observations based on specific statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_order2(
  stat,
  ...,
  reverse = FALSE,
  strict = TRUE,
  data = NULL,
  active = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_order2_+3A_stat">stat</code></td>
<td>
<p>A statistical function which accepts a data and returns the
statistic, which we'll call <code><a href="#topic+order2">order2()</a></code> to extract the ordering information.</p>
</td></tr>
<tr><td><code id="align_order2_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; Additional arguments passed to
function provided in <code>stat</code> argument.</p>
</td></tr>
<tr><td><code id="align_order2_+3A_reverse">reverse</code></td>
<td>
<p>A boolean value. Should the sort order be in reverse?</p>
</td></tr>
<tr><td><code id="align_order2_+3A_strict">strict</code></td>
<td>
<p>A boolean value indicates whether the order should be strict.
If previous groups has been established, and strict is <code>FALSE</code>, this will
reorder the observations in each group.</p>
</td></tr>
<tr><td><code id="align_order2_+3A_data">data</code></td>
<td>
<p>A <code>matrix</code>, <code style="white-space: pre;">&#8288;data frame&#8288;</code>, or atomic vector used as the input for
the <code>stat</code> function. Alternatively, you can specify a <code>function</code> (including
purrr-like lambda syntax) that will be applied to the layout matrix,
transforming it as necessary for statistic calculations. By default, it will
inherit from the layout matrix.</p>
</td></tr>
<tr><td><code id="align_order2_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>The <code>align_order2()</code> function differs from <code>align_order()</code> in that the
<code>weights</code> argument in <code>align_order()</code> must return atomic weights for each
observation. In contrast, the <code>stat</code> argument in <code>align_order2()</code> can
return more complex structures, such as <a href="stats.html#topic+hclust">hclust</a> or
<a href="stats.html#topic+dendrogram">dendrogram</a>, among others.
</p>
<p>Typically, you can achieve the functionality of <code>align_order2()</code> using
<code>align_order()</code> by manually extracting the ordering information from
the statistic.
</p>


<h3>Discrete Axis Alignment</h3>

<p>It is important to note that we consider rows as observations, meaning
<code>vec_size(data)</code>/<code>NROW(data)</code> must match the number of observations along the
axis used for alignment (x-axis for a vertical stack layout, y-axis for a
horizontal stack layout).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+order2">order2()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ggheatmap(matrix(rnorm(81), nrow = 9)) +
    anno_left() +
    align_order2(hclust2)
</code></pre>

<hr>
<h2 id='align_panel'>Create ggplot object with layout panel data</h2><span id='topic+align_panel'></span><span id='topic+ggpanel'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
</p>
<p>This function was deprecated because we realised that it's a special case of
the <code><a href="#topic+ggalign">ggalign()</a></code> function. Please use <code>ggalign(data = NULL)</code> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_panel(...)

ggpanel(...)
</code></pre>

<hr>
<h2 id='align_phylo'>Plot Phylogenetics tree</h2><span id='topic+align_phylo'></span>

<h3>Description</h3>

<p>Plot Phylogenetics tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_phylo(
  phylo,
  ...,
  ladderize = NULL,
  type = "rectangle",
  center = FALSE,
  tree_type = NULL,
  no_axes = NULL,
  active = NULL,
  size = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_phylo_+3A_phylo">phylo</code></td>
<td>
<p>A <code><a href="ape.html#topic+as.phylo">phylo</a></code> object.</p>
</td></tr>
<tr><td><code id="align_phylo_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; Additional arguments passed to
<code><a href="ggplot2.html#topic+geom_segment">geom_segment()</a></code>.</p>
</td></tr>
<tr><td><code id="align_phylo_+3A_ladderize">ladderize</code></td>
<td>
<p>A single string of <code>"left"</code> or <code>"right"</code>,
indicating whether to ladderize the tree. Ladderizing arranges the tree so
that the smallest clade is positioned on the <code>"right"</code> or the <code>"left"</code>. By
default, <code>NULL</code> means the tree will not be ladderized.</p>
</td></tr>
<tr><td><code id="align_phylo_+3A_type">type</code></td>
<td>
<p>A string indicates the plot type, <code>"rectangle"</code> or <code>"triangle"</code>.</p>
</td></tr>
<tr><td><code id="align_phylo_+3A_center">center</code></td>
<td>
<p>A boolean value. if <code>TRUE</code>, nodes are plotted centered with
respect to all leaves/tips in the branch. Otherwise (default), plot them in
the middle of the direct child nodes.</p>
</td></tr>
<tr><td><code id="align_phylo_+3A_tree_type">tree_type</code></td>
<td>
<p>A single string, one of
<code>"phylogram"</code> or <code>"cladogram"</code>, indicating the type of tree.
</p>

<ul>
<li> <p><code>phylogram</code>: Represents a phylogenetic tree where branch lengths indicate
evolutionary distance or time.
</p>
</li>
<li> <p><code>cladogram</code>: Represents a tree where branch lengths are not used, or the
branches do not reflect evolutionary time.
</p>
</li></ul>

<p>Usually, you don't need to modify this.</p>
</td></tr>
<tr><td><code id="align_phylo_+3A_no_axes">no_axes</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a> Logical; if <code>TRUE</code>,
removes axes elements for the alignment axis using <code><a href="#topic+theme_no_axes">theme_no_axes()</a></code>. By
default, will use the option-
<code>"ggalign.align_no_axes"</code>.</p>
</td></tr>
<tr><td><code id="align_phylo_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
<tr><td><code id="align_phylo_+3A_size">size</code></td>
<td>
<p>The relative size of the plot, can be specified as a
<code><a href="grid.html#topic+unit">unit()</a></code>. Note that for <code><a href="#topic+circle_layout">circle_layout()</a></code>, all size values
will be interpreted as relative sizes, as this layout type adjusts based on
the available space in the circular arrangement.</p>
</td></tr>
</table>

<hr>
<h2 id='align_plots'>Arrange multiple plots into a grid</h2><span id='topic+align_plots'></span>

<h3>Description</h3>

<p>Arrange multiple plots into a grid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_plots(
  ...,
  ncol = NULL,
  nrow = NULL,
  byrow = TRUE,
  widths = NA,
  heights = NA,
  design = NULL,
  guides = waiver(),
  theme = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_plots_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; A list of plots, ususally the
ggplot object. Use <code>NULL</code> to indicate an empty spacer.</p>
</td></tr>
<tr><td><code id="align_plots_+3A_ncol">ncol</code>, <code id="align_plots_+3A_nrow">nrow</code></td>
<td>
<p>The dimensions of the grid to create - if both are <code>NULL</code> it
will use the same logic as <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap()</a></code> to set the
dimensions</p>
</td></tr>
<tr><td><code id="align_plots_+3A_byrow">byrow</code></td>
<td>
<p>If <code>FALSE</code> the plots will be filled in in column-major order.</p>
</td></tr>
<tr><td><code id="align_plots_+3A_widths">widths</code>, <code id="align_plots_+3A_heights">heights</code></td>
<td>
<p>The relative widths and heights of each column and row
in the grid. Will get repeated to match the dimensions of the grid. The
special value of <code>NA</code> will behave as <code style="white-space: pre;">&#8288;1null&#8288;</code> unit unless a fixed aspect plot
is inserted in which case it will allow the dimension to expand or contract
to match the aspect ratio of the content.</p>
</td></tr>
<tr><td><code id="align_plots_+3A_design">design</code></td>
<td>
<p>Specification of the location of areas in the layout. Can
either be specified as a text string or by concatenating calls to
<code><a href="#topic+area">area()</a></code> together.</p>
</td></tr>
<tr><td><code id="align_plots_+3A_guides">guides</code></td>
<td>
<p>A string with one or more of <code>"t"</code>, <code>"l"</code>, <code>"b"</code>, and <code>"r"</code> indicating
which side of guide legends should be collected. Defaults to
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code>, which inherits from the parent layout. If
there is no parent layout, or if <code>NULL</code> is provided, no guides will be
collected.</p>
</td></tr>
<tr><td><code id="align_plots_+3A_theme">theme</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+theme">theme()</a></code> object used to customize various
elements of the plot, including <code>guides</code>, <code>title</code>, <code>subtitle</code>, <code>caption</code>,
<code>margins</code>, <code>panel.border</code>, and <code>background</code>. By default, the
theme will inherit from the parent <code>layout</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>alignpatches</code> object.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+layout_design">layout_design()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+layout_title">layout_title()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+layout_annotation">layout_annotation()</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># directly copied from patchwork
p1 &lt;- ggplot(mtcars) +
    geom_point(aes(mpg, disp))
p2 &lt;- ggplot(mtcars) +
    geom_boxplot(aes(gear, disp, group = gear))
p3 &lt;- ggplot(mtcars) +
    geom_bar(aes(gear)) +
    facet_wrap(~cyl)
p4 &lt;- ggplot(mtcars) +
    geom_bar(aes(carb))
p5 &lt;- ggplot(mtcars) +
    geom_violin(aes(cyl, mpg, group = cyl))

# Either add the plots as single arguments
align_plots(p1, p2, p3, p4, p5)

# Or use bang-bang-bang to add a list
align_plots(!!!list(p1, p2, p3), p4, p5)

# Match plots to areas by name
design &lt;- "#BB
           AA#"
align_plots(B = p1, A = p2, design = design)

# Compare to not using named plot arguments
align_plots(p1, p2, design = design)
</code></pre>

<hr>
<h2 id='alignpatch'>Prepare plots to be aligned with <code>align_plots</code></h2><span id='topic+alignpatch'></span>

<h3>Description</h3>

<p>Prepare plots to be aligned with <code>align_plots</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alignpatch(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="alignpatch_+3A_x">x</code></td>
<td>
<p>A plot object to be prepared for alignment.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ggalign</code> has implement <code>alignpatch</code> method for following objects:
</p>

<ul>
<li> <p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>
</p>
</li>
<li> <p><code><a href="#topic+align_plots">alignpatches</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ggwrap">wrapped_plot</a></code>
</p>
</li>
<li> <p><code><a href="patchwork.html#topic+patchGrob">patch</a></code>
</p>
</li>
<li> <p><code><a href="patchwork.html#topic+wrap_elements">wrapped_patch</a></code>
</p>
</li>
<li> <p><code><a href="patchwork.html#topic+plot_spacer">spacer</a></code>
</p>
</li></ul>



<h3>Value</h3>

<p>A <code>Patch</code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+align_plots">align_plots()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alignpatch(ggplot())
</code></pre>

<hr>
<h2 id='area'>Define the plotting areas in <code>align_plots</code></h2><span id='topic+area'></span>

<h3>Description</h3>

<p>This is a small helper used to specify a single area in a rectangular grid
that should contain a plot. Objects constructed with <code>area()</code> can be
concatenated together with <code>c()</code> in order to specify multiple areas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>area(t, l, b = t, r = l)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="area_+3A_t">t</code>, <code id="area_+3A_b">b</code></td>
<td>
<p>The top and bottom bounds of the area in the grid</p>
</td></tr>
<tr><td><code id="area_+3A_l">l</code>, <code id="area_+3A_r">r</code></td>
<td>
<p>The left and right bounds of the area int the grid</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The grid that the areas are specified in reference to enumerate rows from top
to bottom, and coloumns from left to right. This means that <code>t</code> and <code>l</code>
should always be less or equal to <code>b</code> and <code>r</code> respectively. Instead of
specifying area placement with a combination of <code>area()</code> calls, it is
possible to instead pass in a single string
</p>
<div class="sourceCode"><pre>areas &lt;- c(area(1, 1, 2, 1),
           area(2, 3, 3, 3))
</pre></div>
<p>is equivalent to
</p>
<div class="sourceCode"><pre>areas &lt; -"A##
          A#B
          ##B"
</pre></div>


<h3>Value</h3>

<p>A <code>ggalign_area</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p1 &lt;- ggplot(mtcars) +
    geom_point(aes(mpg, disp))
p2 &lt;- ggplot(mtcars) +
    geom_boxplot(aes(gear, disp, group = gear))
p3 &lt;- ggplot(mtcars) +
    geom_bar(aes(gear)) +
    facet_wrap(~cyl)

layout &lt;- c(
    area(1, 1),
    area(1, 3, 3),
    area(3, 1, 3, 2)
)

# Show the layout to make sure it looks as it should
plot(layout)

# Apply it to a alignpatches
align_plots(p1, p2, p3, design = layout)
</code></pre>

<hr>
<h2 id='circle_layout'>Arrange plots in a circular layout</h2><span id='topic+circle_layout'></span><span id='topic+circle_discrete'></span><span id='topic+circle_continuous'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>If <code>limits</code> is provided, a continuous variable will be required and aligned
in the direction specified (<code>circle_continuous</code>). Otherwise, a discrete
variable will be required and aligned (<code>circle_discrete</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circle_layout(
  data = NULL,
  ...,
  radial = NULL,
  direction = "outward",
  limits = waiver(),
  theme = NULL
)

circle_discrete(
  data = NULL,
  ...,
  radial = NULL,
  direction = "outward",
  spacing_theta = NULL,
  theme = NULL
)

circle_continuous(
  data = NULL,
  ...,
  radial = NULL,
  direction = "outward",
  limits = NULL,
  theme = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circle_layout_+3A_data">data</code></td>
<td>
<p>Default dataset to use for the layout. If not specified, it must be supplied in each plot added to the layout:
</p>

<ul>
<li><p> If <code>limits</code> is not provided, <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> will be used to get a
matrix.
</p>
</li>
<li><p> If <code>limits</code> is specified, <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> will be used to get a
data frame.
</p>
</li></ul>
</td></tr>
<tr><td><code id="circle_layout_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> or
<code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>.</p>
</td></tr>
<tr><td><code id="circle_layout_+3A_radial">radial</code></td>
<td>
<p>A <code><a href="#topic+coord_circle">coord_circle()</a></code>/<code><a href="ggplot2.html#topic+coord_polar">coord_radial()</a></code>
object that defines the global parameters for coordinate across all plots
in the layout. The parameters <code>start</code>, <code>end</code>, <code>direction</code>, and <code>expand</code> will
be inherited and applied uniformly to all plots within the layout. The
parameters <code>theta</code> and <code>r.axis.inside</code> will always be ignored and will be set
to <code>"x"</code> and <code>TRUE</code>, respectively, for all plots.</p>
</td></tr>
<tr><td><code id="circle_layout_+3A_direction">direction</code></td>
<td>
<p>A single string of <code>"inward"</code> or <code>"outward"</code>,
indicating the direction in which the plot is added.
</p>

<ul>
<li> <p><code>outward</code>: The plot is added from the inner to the outer.
</p>
</li>
<li> <p><code>inward</code>: The plot is added from the outer to the inner.
</p>
</li></ul>
</td></tr>
<tr><td><code id="circle_layout_+3A_limits">limits</code></td>
<td>
<p>A <code><a href="#topic+continuous_limits">continuous_limits()</a></code> object specifying the left/lower
limit and the right/upper limit of the scale. Used to align the continuous
axis.</p>
</td></tr>
<tr><td><code id="circle_layout_+3A_theme">theme</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+theme">theme()</a></code> object used to customize various
elements of the layout, including <code>guides</code>, <code>title</code>, <code>subtitle</code>, <code>caption</code>,
<code>margins</code>, <code>panel.border</code>, and <code>background</code>. By default, the theme will
inherit from the parent <code>layout</code>. It also controls the panel spacing for all
plots in the layout.</p>
</td></tr>
<tr><td><code id="circle_layout_+3A_spacing_theta">spacing_theta</code></td>
<td>
<p>The size of spacing between different panel. A numeric
of the radians or a <code><a href="ggplot2.html#topic+element">rel()</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>CircleLayout</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)

small_mat &lt;- matrix(rnorm(56), nrow = 7)
rownames(small_mat) &lt;- paste0("row", seq_len(nrow(small_mat)))
colnames(small_mat) &lt;- paste0("column", seq_len(ncol(small_mat)))

# circle_layout
# same for circle_discrete()
circle_layout(small_mat) +
    ggalign() +
    geom_tile(aes(y = .column_index, fill = value)) +
    scale_fill_viridis_c() +
    align_dendro(aes(color = branch), k = 3L) +
    scale_color_brewer(palette = "Dark2")

# same for circle_continuous()
circle_layout(mpg, limits = continuous_limits(c(3, 5))) +
    ggalign(mapping = aes(displ, hwy, colour = class)) +
    geom_point(size = 2) +
    ggalign(mapping = aes(displ, hwy, colour = class)) +
    geom_point(size = 2) &amp;
    scale_color_brewer(palette = "Dark2") &amp;
    theme_bw()

# circle_discrete()
# direction outward
circle_discrete(small_mat) +
    align_dendro(aes(color = branch), k = 3L) +
    scale_color_brewer(palette = "Dark2") +
    ggalign() +
    geom_tile(aes(y = .column_index, fill = value)) +
    scale_fill_viridis_c()

# direction inward
circle_discrete(small_mat, direction = "inward") +
    ggalign() +
    geom_tile(aes(y = .column_index, fill = value)) +
    scale_fill_viridis_c() +
    align_dendro(aes(color = branch), k = 3L) +
    scale_color_brewer(palette = "Dark2")

# circle_continuous()
circle_continuous(mpg, limits = continuous_limits(c(3, 5))) +
    ggalign(mapping = aes(displ, hwy, colour = class)) +
    geom_point(size = 2) +
    ggalign(mapping = aes(displ, hwy, colour = class)) +
    geom_point(size = 2) &amp;
    scale_color_brewer(palette = "Dark2") &amp;
    theme_bw()
</code></pre>

<hr>
<h2 id='circle_switch'>Determine the active context of circle layout</h2><span id='topic+circle_switch'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circle_switch(radial = waiver(), direction = NULL, what = waiver(), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circle_switch_+3A_radial">radial</code></td>
<td>
<p>A <code><a href="#topic+coord_circle">coord_circle()</a></code>/<code><a href="ggplot2.html#topic+coord_polar">coord_radial()</a></code>
object that defines the global parameters for coordinate across all plots
in the layout. The parameters <code>start</code>, <code>end</code>, <code>direction</code>, and <code>expand</code> will
be inherited and applied uniformly to all plots within the layout. The
parameters <code>theta</code> and <code>r.axis.inside</code> will always be ignored and will be set
to <code>"x"</code> and <code>TRUE</code>, respectively, for all plots.</p>
</td></tr>
<tr><td><code id="circle_switch_+3A_direction">direction</code></td>
<td>
<p>A single string of <code>"inward"</code> or <code>"outward"</code>,
indicating the direction in which the plot is added.
</p>

<ul>
<li> <p><code>outward</code>: The plot is added from the inner to the outer.
</p>
</li>
<li> <p><code>inward</code>: The plot is added from the outer to the inner.
</p>
</li></ul>
</td></tr>
<tr><td><code id="circle_switch_+3A_what">what</code></td>
<td>
<p>What should get activated for the <code><a href="#topic+circle_layout">circle_layout()</a></code>?
A single number or string of the plot elements in the layout. If <code>NULL</code>, will remove any active context.</p>
</td></tr>
<tr><td><code id="circle_switch_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>circle_switch</code> object which can be added to <code><a href="#topic+circle_layout">circle_layout()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
small_mat &lt;- matrix(rnorm(56), nrow = 7)
rownames(small_mat) &lt;- paste0("row", seq_len(nrow(small_mat)))
colnames(small_mat) &lt;- paste0("column", seq_len(ncol(small_mat)))
circle_discrete(small_mat) +
    ggalign() +
    geom_tile(aes(y = .column_index, fill = value)) +
    scale_fill_viridis_c() +
    align_dendro(aes(color = branch), k = 3L) +
    scale_color_brewer(palette = "Dark2")
</code></pre>

<hr>
<h2 id='continuous_limits'>Set continuous limits for the layout</h2><span id='topic+continuous_limits'></span>

<h3>Description</h3>

<p>To align continuous axes, it is important to keep the limits consistent
across all plots in the layout. You can set the limits by passing a function
directly to the <code>limits</code> or <code>xlim</code>/<code>ylim</code> argument, using <code>...</code> only.
Alternatively, you can add a <code>continuous_limits()</code> object to the layout. For
the <code>quad_layout()</code> function, you must specify <code>x</code>/<code>y</code> arguments. For other
layouts, you should pass the limits using <code>...</code> directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>continuous_limits(..., x = waiver(), y = waiver())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="continuous_limits_+3A_...">...</code></td>
<td>
<p>A list of two numeric values, specifying the left/lower limit and
the right/upper limit of the scale.</p>
</td></tr>
<tr><td><code id="continuous_limits_+3A_x">x</code>, <code id="continuous_limits_+3A_y">y</code></td>
<td>
<p>Same as <code>...</code>, but specifically for <code>quad_layout()</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='coord_circle'>Polar Coordinates with Enhanced Controls</h2><span id='topic+coord_circle'></span>

<h3>Description</h3>

<p>An extended version of <code><a href="ggplot2.html#topic+coord_polar">coord_radial()</a></code>, providing
additional customization options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coord_circle(
  theta = "x",
  start = 0,
  end = NULL,
  xlim = NULL,
  ylim = NULL,
  expand = FALSE,
  direction = 1,
  clip = "off",
  r.axis.inside = NULL,
  rotate.angle = FALSE,
  inner.radius = 0,
  outer.radius = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coord_circle_+3A_theta">theta</code></td>
<td>
<p>variable to map angle to (<code>x</code> or <code>y</code>)</p>
</td></tr>
<tr><td><code id="coord_circle_+3A_start">start</code></td>
<td>
<p>Offset of starting point from 12 o'clock in radians. Offset
is applied clockwise or anticlockwise depending on value of <code>direction</code>.</p>
</td></tr>
<tr><td><code id="coord_circle_+3A_end">end</code></td>
<td>
<p>Position from 12 o'clock in radians where plot ends, to allow
for partial polar coordinates. The default, <code>NULL</code>, is set to
<code>start + 2 * pi</code>.</p>
</td></tr>
<tr><td><code id="coord_circle_+3A_xlim">xlim</code>, <code id="coord_circle_+3A_ylim">ylim</code></td>
<td>
<p>Limits for the x and y axes.</p>
</td></tr>
<tr><td><code id="coord_circle_+3A_expand">expand</code></td>
<td>
<p>If <code>TRUE</code>, the default, adds a small expansion factor the
the limits to prevent overlap between data and axes. If <code>FALSE</code>, limits
are taken directly from the scale.</p>
</td></tr>
<tr><td><code id="coord_circle_+3A_direction">direction</code></td>
<td>
<p>1, clockwise; -1, anticlockwise</p>
</td></tr>
<tr><td><code id="coord_circle_+3A_clip">clip</code></td>
<td>
<p>Should drawing be clipped to the extent of the plot panel? A
setting of <code>"on"</code> (the default) means yes, and a setting of <code>"off"</code>
means no. For details, please see <code><a href="ggplot2.html#topic+coord_cartesian">coord_cartesian()</a></code>.</p>
</td></tr>
<tr><td><code id="coord_circle_+3A_r.axis.inside">r.axis.inside</code></td>
<td>
<p>If <code>TRUE</code>, places the radius axis inside the
panel. If <code>FALSE</code>, places the radius axis next to the panel. The default,
<code>NULL</code>, places the radius axis outside if the <code>start</code> and <code>end</code> arguments
form a full circle.</p>
</td></tr>
<tr><td><code id="coord_circle_+3A_rotate.angle">rotate.angle</code></td>
<td>
<p>If <code>TRUE</code>, transforms the <code>angle</code> aesthetic in data
in accordance with the computed <code>theta</code> position. If <code>FALSE</code> (default),
no such transformation is performed. Can be useful to rotate text geoms in
alignment with the coordinates.</p>
</td></tr>
<tr><td><code id="coord_circle_+3A_inner.radius">inner.radius</code></td>
<td>
<p>A numeric in <code style="white-space: pre;">&#8288;[0, 1)&#8288;</code> indicates the inner radius.</p>
</td></tr>
<tr><td><code id="coord_circle_+3A_outer.radius">outer.radius</code></td>
<td>
<p>A numeric in <code style="white-space: pre;">&#8288;(0, 1]&#8288;</code> indicates the outer radius.
<code><a href="ggplot2.html#topic+coord_polar">coord_radial()</a></code> by default uses <code>0.8</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>ggplot(mtcars, aes(disp, mpg)) +
    geom_point() +
    coord_circle(
        start = -0.4 * pi, end = 0.4 * pi,
        inner.radius = 0.3, outer.radius = 1
    )
ggplot(mtcars, aes(disp, mpg)) +
    geom_point() +
    coord_circle(
        start = -0.4 * pi, end = 0.4 * pi,
        inner.radius = 0.3, outer.radius = 0.5
    )
</code></pre>

<hr>
<h2 id='cross_link'>Add a plot to connect selected observations</h2><span id='topic+cross_link'></span>

<h3>Description</h3>

<p>Add a plot to connect selected observations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cross_link(
  link,
  data = waiver(),
  ...,
  on_top = TRUE,
  obs_size = 1,
  inherit_index = NULL,
  inherit_panel = NULL,
  inherit_nobs = NULL,
  size = NULL,
  active = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cross_link_+3A_link">link</code></td>
<td>
<p>A <code><a href="#topic+link_draw">link_draw()</a></code> object that defines how to draw the links,
such as <code><a href="#topic+link_line">link_line()</a></code>.</p>
</td></tr>
<tr><td><code id="cross_link_+3A_data">data</code></td>
<td>
<p>The dataset to use for the layout. By default,
<code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> will convert the data to a matrix. This argument
allows you to change the layout data. If not specified, the original data
will be used.</p>
</td></tr>
<tr><td><code id="cross_link_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; Additional arguments passed to
<code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>.</p>
</td></tr>
<tr><td><code id="cross_link_+3A_on_top">on_top</code></td>
<td>
<p>A boolean value indicating whether to draw the link on top of
the plot panel (<code>TRUE</code>) or below (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="cross_link_+3A_obs_size">obs_size</code></td>
<td>
<p>A single numeric value that indicates the size of a single
observation, ranging from <code style="white-space: pre;">&#8288;(0, 1]&#8288;</code>.</p>
</td></tr>
<tr><td><code id="cross_link_+3A_inherit_index">inherit_index</code></td>
<td>
<p>A boolean value indicating whether to inherit the
ordering index. If <code>TRUE</code>, will match the layout ordering index with the
data names.</p>
</td></tr>
<tr><td><code id="cross_link_+3A_inherit_panel">inherit_panel</code></td>
<td>
<p>A boolean value indicating whether to inherit the
panel group. If <code>TRUE</code>, will match the layout panel with the data names.</p>
</td></tr>
<tr><td><code id="cross_link_+3A_inherit_nobs">inherit_nobs</code></td>
<td>
<p>A boolean value indicating whether to inherit the
number of observations (nobs). If <code>TRUE</code>, the <code>data</code> input must be
compatible with the layout data.</p>
</td></tr>
<tr><td><code id="cross_link_+3A_size">size</code></td>
<td>
<p>The relative size of the plot, can be specified as a
<code><a href="grid.html#topic+unit">unit()</a></code>. Note that for <code><a href="#topic+circle_layout">circle_layout()</a></code>, all size values
will be interpreted as relative sizes, as this layout type adjusts based on
the available space in the circular arrangement.</p>
</td></tr>
<tr><td><code id="cross_link_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>ggplot2 Specification</h3>

<p>The <code>cross_link</code> function initializes a <code>ggplot</code> object but does not
initialize any data. Using <code><a href="#topic+scheme_data">scheme_data()</a></code> to change the internal data if
needed.
</p>

<hr>
<h2 id='cross_mark'>Add a plot to annotate observations</h2><span id='topic+cross_mark'></span>

<h3>Description</h3>

<p>Add a plot to annotate observations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cross_mark(
  mark,
  data = waiver(),
  ...,
  obs_size = 1,
  inherit_index = NULL,
  inherit_panel = NULL,
  inherit_nobs = NULL,
  size = NULL,
  active = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cross_mark_+3A_mark">mark</code></td>
<td>
<p>A <code><a href="#topic+mark_draw">mark_draw()</a></code> object to define how to draw the links. Like
<code><a href="#topic+mark_line">mark_line()</a></code>, <code><a href="#topic+mark_tetragon">mark_tetragon()</a></code>. Note the names of the pair links will
be used to define the panel names so must be unique.</p>
</td></tr>
<tr><td><code id="cross_mark_+3A_data">data</code></td>
<td>
<p>The dataset to use for the layout. By default,
<code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> will convert the data to a matrix. This argument
allows you to change the layout data. If not specified, the original data
will be used.</p>
</td></tr>
<tr><td><code id="cross_mark_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; Additional arguments passed to
<code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>.</p>
</td></tr>
<tr><td><code id="cross_mark_+3A_obs_size">obs_size</code></td>
<td>
<p>A single numeric value that indicates the size of a single
observation, ranging from <code style="white-space: pre;">&#8288;(0, 1]&#8288;</code>.</p>
</td></tr>
<tr><td><code id="cross_mark_+3A_inherit_index">inherit_index</code></td>
<td>
<p>A boolean value indicating whether to inherit the
ordering index. If <code>TRUE</code>, will match the layout ordering index with the
data names.</p>
</td></tr>
<tr><td><code id="cross_mark_+3A_inherit_panel">inherit_panel</code></td>
<td>
<p>A boolean value indicating whether to inherit the
panel group. If <code>TRUE</code>, will match the layout panel with the data names.</p>
</td></tr>
<tr><td><code id="cross_mark_+3A_inherit_nobs">inherit_nobs</code></td>
<td>
<p>A boolean value indicating whether to inherit the
number of observations (nobs). If <code>TRUE</code>, the <code>data</code> input must be
compatible with the layout data.</p>
</td></tr>
<tr><td><code id="cross_mark_+3A_size">size</code></td>
<td>
<p>The relative size of the plot, can be specified as a
<code><a href="grid.html#topic+unit">unit()</a></code>. Note that for <code><a href="#topic+circle_layout">circle_layout()</a></code>, all size values
will be interpreted as relative sizes, as this layout type adjusts based on
the available space in the circular arrangement.</p>
</td></tr>
<tr><td><code id="cross_mark_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>ggplot2 Specification</h3>

<p>The <code>cross_mark</code> function initializes a <code>ggplot</code> object. The underlying data
contains following columns:
</p>

<ul>
<li> <p><code>.panel</code>: the panel for the aligned axis. It means <code>x-axis</code> for vertical
stack layout (including top and bottom annotation), <code>y-axis</code> for
horizontal stack layout (including left and right annotation).
</p>
</li>
<li> <p><code>.names</code> (<code><a href="vctrs.html#topic+vec_names">vec_names()</a></code>) and <code>.index</code>
(<code><a href="vctrs.html#topic+vec_size">vec_size()</a></code>/<code><a href="base.html#topic+NROW">NROW()</a></code>): a character names (only
applicable when names exists) and an integer of index of the original
data.
</p>
</li>
<li> <p><code>.hand</code>: A factor with levels <code>c("left", "right")</code> for horizontal stack
layouts, or <code>c("top", "bottom")</code> for vertical stack layouts, indicating
the position of the linked observations.
</p>
</li></ul>

<p>You can use <code><a href="#topic+scheme_data">scheme_data()</a></code> to modify the internal data if needed.
</p>

<hr>
<h2 id='cross_none'>Reset layout ordering and panel group</h2><span id='topic+cross_none'></span>

<h3>Description</h3>

<p>Reset layout ordering and panel group
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cross_none(
  data = waiver(),
  ...,
  inherit_index = NULL,
  inherit_panel = NULL,
  inherit_nobs = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cross_none_+3A_data">data</code></td>
<td>
<p>The dataset to use for the layout. By default,
<code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> will convert the data to a matrix. This argument
allows you to change the layout data. If not specified, the original data
will be used.</p>
</td></tr>
<tr><td><code id="cross_none_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; Additional arguments passed to
<code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>.</p>
</td></tr>
<tr><td><code id="cross_none_+3A_inherit_index">inherit_index</code></td>
<td>
<p>A boolean value indicating whether to inherit the
ordering index. If <code>TRUE</code>, will match the layout ordering index with the
data names.</p>
</td></tr>
<tr><td><code id="cross_none_+3A_inherit_panel">inherit_panel</code></td>
<td>
<p>A boolean value indicating whether to inherit the
panel group. If <code>TRUE</code>, will match the layout panel with the data names.</p>
</td></tr>
<tr><td><code id="cross_none_+3A_inherit_nobs">inherit_nobs</code></td>
<td>
<p>A boolean value indicating whether to inherit the
number of observations (nobs). If <code>TRUE</code>, the <code>data</code> input must be
compatible with the layout data.</p>
</td></tr>
</table>

<hr>
<h2 id='draw_key_gshape'>Key glyphs for legends</h2><span id='topic+draw_key_gshape'></span>

<h3>Description</h3>

<p>Each geom has an associated function that draws the key when the geom needs
to be displayed in a legend. These functions are called <code style="white-space: pre;">&#8288;draw_key_*()&#8288;</code>, where
<code>*</code> stands for the name of the respective key glyph. The key glyphs can be
customized for individual geoms by providing a geom with the <code>key_glyph</code>
argument. The <code>draw_key_gshape</code> function provides this interface for custom
key glyphs used with <code><a href="#topic+geom_gshape">geom_gshape()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_key_gshape(data, params, size)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="draw_key_gshape_+3A_data">data</code></td>
<td>
<p>A single row data frame containing the scaled aesthetics to
display in this key</p>
</td></tr>
<tr><td><code id="draw_key_gshape_+3A_params">params</code></td>
<td>
<p>A list of additional parameters supplied to the geom.</p>
</td></tr>
<tr><td><code id="draw_key_gshape_+3A_size">size</code></td>
<td>
<p>Width and height of key in mm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A grid grob.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p &lt;- ggplot(economics, aes(date, psavert, color = "savings rate"))
# key glyphs can be specified by their name
p + geom_line(key_glyph = "timeseries")

# key glyphs can be specified via their drawing function
p + geom_line(key_glyph = draw_key_rect)
</code></pre>

<hr>
<h2 id='element_curve'>Theme curve elements</h2><span id='topic+element_curve'></span>

<h3>Description</h3>

<p>Draw curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>element_curve(
  colour = NULL,
  linewidth = NULL,
  linetype = NULL,
  lineend = NULL,
  color = NULL,
  curvature = NULL,
  angle = NULL,
  ncp = NULL,
  shape = NULL,
  arrow = NULL,
  arrow.fill = NULL,
  inherit.blank = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="element_curve_+3A_colour">colour</code>, <code id="element_curve_+3A_color">color</code></td>
<td>
<p>Line/border colour. Color is an alias for colour.</p>
</td></tr>
<tr><td><code id="element_curve_+3A_linewidth">linewidth</code></td>
<td>
<p>Line/border size in mm.</p>
</td></tr>
<tr><td><code id="element_curve_+3A_linetype">linetype</code></td>
<td>
<p>Line type. An integer (0:8), a name (blank, solid,
dashed, dotted, dotdash, longdash, twodash), or a string with
an even number (up to eight) of hexadecimal digits which give the
lengths in consecutive positions in the string.</p>
</td></tr>
<tr><td><code id="element_curve_+3A_lineend">lineend</code></td>
<td>
<p>Line end Line end style (round, butt, square)</p>
</td></tr>
<tr><td><code id="element_curve_+3A_curvature">curvature</code></td>
<td>
<p>A numeric value giving the amount of curvature.
Negative values produce left-hand curves, positive values
produce right-hand curves, and zero produces a straight line.</p>
</td></tr>
<tr><td><code id="element_curve_+3A_angle">angle</code></td>
<td>
<p>A numeric value between 0 and 180,
giving an amount to skew the control
points of the curve.  Values less than 90 skew the curve towards
the start point and values greater than 90 skew the curve
towards the end point.</p>
</td></tr>
<tr><td><code id="element_curve_+3A_ncp">ncp</code></td>
<td>
<p>The number of control points used to draw the curve.
More control points creates a smoother curve.</p>
</td></tr>
<tr><td><code id="element_curve_+3A_shape">shape</code></td>
<td>
<p>A numeric vector of values between -1 and 1, which
control the shape of the curve relative to its control points.
See <code>grid.xspline</code> for more details.</p>
</td></tr>
<tr><td><code id="element_curve_+3A_arrow">arrow</code></td>
<td>
<p>A list describing arrow heads to place at either end
of the curve, as produced by the <code>arrow</code> function.</p>
</td></tr>
<tr><td><code id="element_curve_+3A_arrow.fill">arrow.fill</code></td>
<td>
<p>Fill colour for arrows.</p>
</td></tr>
<tr><td><code id="element_curve_+3A_inherit.blank">inherit.blank</code></td>
<td>
<p>Should this element inherit the existence of an
<code>element_blank</code> among its parents? If <code>TRUE</code> the existence of
a blank element among its parents will cause this element to be blank as
well. If <code>FALSE</code> any blank parent element will be ignored when
calculating final element state.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>element_curve</code> object
</p>

<hr>
<h2 id='element_polygon'>Theme Polygon elements</h2><span id='topic+element_polygon'></span>

<h3>Description</h3>

<p>Draw polygon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>element_polygon(
  fill = NULL,
  colour = NULL,
  linewidth = NULL,
  linetype = NULL,
  alpha = NULL,
  lineend = NULL,
  linejoin = NULL,
  linemitre = NULL,
  color = NULL,
  inherit.blank = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="element_polygon_+3A_fill">fill</code></td>
<td>
<p>Fill colour.</p>
</td></tr>
<tr><td><code id="element_polygon_+3A_colour">colour</code>, <code id="element_polygon_+3A_color">color</code></td>
<td>
<p>Line/border colour. Color is an alias for colour.</p>
</td></tr>
<tr><td><code id="element_polygon_+3A_linewidth">linewidth</code></td>
<td>
<p>Line/border size in mm.</p>
</td></tr>
<tr><td><code id="element_polygon_+3A_linetype">linetype</code></td>
<td>
<p>Line type. An integer (0:8), a name (blank, solid,
dashed, dotted, dotdash, longdash, twodash), or a string with
an even number (up to eight) of hexadecimal digits which give the
lengths in consecutive positions in the string.</p>
</td></tr>
<tr><td><code id="element_polygon_+3A_alpha">alpha</code></td>
<td>
<p>A transparency value between 0 (transparent) and 1 (opaque),
parallel to <code>fill</code>.</p>
</td></tr>
<tr><td><code id="element_polygon_+3A_lineend">lineend</code></td>
<td>
<p>Line end Line end style (round, butt, square)</p>
</td></tr>
<tr><td><code id="element_polygon_+3A_linejoin">linejoin</code></td>
<td>
<p>Line join style (round, mitre, bevel).</p>
</td></tr>
<tr><td><code id="element_polygon_+3A_linemitre">linemitre</code></td>
<td>
<p>Line mitre limit (number greater than 1).</p>
</td></tr>
<tr><td><code id="element_polygon_+3A_inherit.blank">inherit.blank</code></td>
<td>
<p>Should this element inherit the existence of an
<code>element_blank</code> among its parents? If <code>TRUE</code> the existence of
a blank element among its parents will cause this element to be blank as
well. If <code>FALSE</code> any blank parent element will be ignored when
calculating final element state.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>element_polygon</code> object
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+element">element_rect</a></code>
</p>

<hr>
<h2 id='element_vec'>Apply a function to the fields of an element object</h2><span id='topic+element_vec'></span><span id='topic+element_rep'></span><span id='topic+element_rep_len'></span><span id='topic+element_vec_recycle'></span><span id='topic+element_vec_rep'></span><span id='topic+element_vec_rep_each'></span><span id='topic+element_vec_slice'></span>

<h3>Description</h3>

<p>For an <code><a href="ggplot2.html#topic+element">element</a></code> object, some fields are
vectorized, while others are not. This function allows you to apply a
function to the vectorized fields.
</p>
<p>The following helper functions are available:
</p>

<ul>
<li> <p><code>element_rep</code>: Applies <code><a href="base.html#topic+rep">rep()</a></code>.
</p>
</li>
<li> <p><code>element_rep_len</code>: Applies <code><a href="base.html#topic+rep_len">rep_len()</a></code>.
</p>
</li>
<li> <p><code>element_vec_recycle</code>: Applies <code><a href="vctrs.html#topic+vec_recycle">vec_recycle()</a></code>.
</p>
</li>
<li> <p><code>element_vec_rep</code>: Applies <code><a href="vctrs.html#topic+vec-rep">vec_rep()</a></code>.
</p>
</li>
<li> <p><code>element_vec_rep_each</code>: Applies <code><a href="vctrs.html#topic+vec-rep">vec_rep_each()</a></code>.
</p>
</li>
<li> <p><code>element_vec_slice</code>: Applies <code><a href="vctrs.html#topic+vec_slice">vec_slice()</a></code>.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>element_vec(.el, .fn, ...)

element_rep(.el, ...)

element_rep_len(.el, length.out, ...)

element_vec_recycle(.el, size, ...)

element_vec_rep(.el, times, ...)

element_vec_rep_each(.el, times, ...)

element_vec_slice(.el, i, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="element_vec_+3A_.el">.el</code></td>
<td>
<p>An <code><a href="ggplot2.html#topic+element">element</a></code> object.</p>
</td></tr>
<tr><td><code id="element_vec_+3A_.fn">.fn</code></td>
<td>
<p>The function to be applied to the vectorized fields of the element
object.</p>
</td></tr>
<tr><td><code id="element_vec_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to <code>fn</code>.</p>
</td></tr>
<tr><td><code id="element_vec_+3A_length.out">length.out</code></td>
<td>
<p>Non-negative integer. The desired length of the output
vector. Other inputs will be coerced to a double vector and the first element
taken. Ignored if <code>NA</code> or invalid.</p>
</td></tr>
<tr><td><code id="element_vec_+3A_size">size</code></td>
<td>
<p>Desired output size.</p>
</td></tr>
<tr><td><code id="element_vec_+3A_times">times</code></td>
<td>
<p>For <code>vec_rep()</code>, a single integer for the number of times to repeat
the entire vector.
</p>
<p>For <code>vec_rep_each()</code>, an integer vector of the number of times to repeat
each element of <code>x</code>. <code>times</code> will be <a href="vctrs.html#topic+theory-faq-recycling">recycled</a> to
the size of <code>x</code>.</p>
</td></tr>
<tr><td><code id="element_vec_+3A_i">i</code></td>
<td>
<p>An integer, character or logical vector specifying the
locations or names of the observations to get/set. Specify
<code>TRUE</code> to index all elements (as in <code>x[]</code>), or <code>NULL</code>, <code>FALSE</code> or
<code>integer()</code> to index none (as in <code>x[NULL]</code>).</p>
</td></tr>
</table>

<hr>
<h2 id='facet_sector'>Polar coordinates with Facet support</h2><span id='topic+facet_sector'></span>

<h3>Description</h3>

<p>Draw each panel in a sector of the polar coordinate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>facet_sector(facets, radial = NULL, spacing_theta = pi/180, drop = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="facet_sector_+3A_facets">facets</code></td>
<td>
<p>A set of variables or expressions quoted by <code><a href="ggplot2.html#topic+vars">vars()</a></code>
and defining faceting groups on the rows or columns dimension.
The variables can be named (the names are passed to <code>labeller</code>).
</p>
<p>For compatibility with the classic interface, can also be a
formula or character vector. Use either a one sided formula, <code>~a + b</code>,
or a character vector, <code>c("a", "b")</code>.</p>
</td></tr>
<tr><td><code id="facet_sector_+3A_radial">radial</code></td>
<td>
<p>A Polar <a href="ggplot2.html#topic+ggplot2-ggproto">Coord</a> created by <code><a href="#topic+coord_circle">coord_circle()</a></code>.</p>
</td></tr>
<tr><td><code id="facet_sector_+3A_spacing_theta">spacing_theta</code></td>
<td>
<p>The size of spacing between different panel. A numeric
of the radians or a <code><a href="ggplot2.html#topic+element">rel()</a></code> object.</p>
</td></tr>
<tr><td><code id="facet_sector_+3A_drop">drop</code></td>
<td>
<p>If <code>TRUE</code>, the default, all factor levels not used in the
data will automatically be dropped. If <code>FALSE</code>, all factor levels
will be shown, regardless of whether or not they appear in the data.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>ggplot(mtcars, aes(disp, mpg)) +
    geom_point() +
    facet_sector(
        vars(cyl),
        coord_circle(
            start = -0.4 * pi, end = 0.4 * pi, inner.radius = 0.3,
            outer.radius = 0.8, expand = TRUE
        )
    )
</code></pre>

<hr>
<h2 id='fortify_data_frame'>Build a data frame</h2><span id='topic+fortify_data_frame'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function converts various objects to a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fortify_data_frame(data, ..., data_arg = NULL, call = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_data_frame_+3A_data">data</code></td>
<td>
<p>An object to be converted to a data frame.</p>
</td></tr>
<tr><td><code id="fortify_data_frame_+3A_...">...</code></td>
<td>
<p>Arguments passed to methods.</p>
</td></tr>
<tr><td><code id="fortify_data_frame_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_data_frame_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame.
</p>


<h3><code>fortify_data_frame</code> method collections</h3>


<ul>
<li> <p><code><a href="#topic+fortify_data_frame.phylo">fortify_data_frame.phylo()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_data_frame.default">fortify_data_frame.default()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_data_frame.character">fortify_data_frame.character()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_data_frame.matrix">fortify_data_frame.matrix()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_data_frame.dendrogram">fortify_data_frame.dendrogram()</a></code>
</p>
</li></ul>


<hr>
<h2 id='fortify_data_frame.character'>Build a data frame</h2><span id='topic+fortify_data_frame.character'></span><span id='topic+fortify_data_frame.numeric'></span><span id='topic+fortify_data_frame.logical'></span><span id='topic+fortify_data_frame.complex'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function converts various objects to a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'character'
fortify_data_frame(data, ..., data_arg = NULL, call = NULL)

## S3 method for class 'numeric'
fortify_data_frame(data, ..., data_arg = NULL, call = NULL)

## S3 method for class 'logical'
fortify_data_frame(data, ..., data_arg = NULL, call = NULL)

## S3 method for class 'complex'
fortify_data_frame(data, ..., data_arg = NULL, call = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_data_frame.character_+3A_data">data</code></td>
<td>
<p>An object to be converted to a data frame.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.character_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.character_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.character_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with following columns:
</p>

<ul>
<li> <p><code>.names</code>: the names for the vector (only applicable if names exist).
</p>
</li>
<li> <p><code>value</code>: the actual value of the vector.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> methods:
<code><a href="#topic+fortify_data_frame.default">fortify_data_frame.default</a>()</code>,
<code><a href="#topic+fortify_data_frame.dendrogram">fortify_data_frame.dendrogram</a>()</code>,
<code><a href="#topic+fortify_data_frame.matrix">fortify_data_frame.matrix</a>()</code>,
<code><a href="#topic+fortify_data_frame.phylo">fortify_data_frame.phylo</a>()</code>
</p>

<hr>
<h2 id='fortify_data_frame.default'>Build a data frame</h2><span id='topic+fortify_data_frame.default'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function converts various objects to a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
fortify_data_frame(data, ..., data_arg = NULL, call = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_data_frame.default_+3A_data">data</code></td>
<td>
<p>An object to be converted to a data frame.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.default_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="ggplot2.html#topic+fortify">fortify()</a></code>.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.default_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.default_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, it calls <code><a href="ggplot2.html#topic+fortify">fortify()</a></code> to build the
data frame.
</p>


<h3>See Also</h3>

<p>Other <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> methods:
<code><a href="#topic+fortify_data_frame.character">fortify_data_frame.character</a>()</code>,
<code><a href="#topic+fortify_data_frame.dendrogram">fortify_data_frame.dendrogram</a>()</code>,
<code><a href="#topic+fortify_data_frame.matrix">fortify_data_frame.matrix</a>()</code>,
<code><a href="#topic+fortify_data_frame.phylo">fortify_data_frame.phylo</a>()</code>
</p>

<hr>
<h2 id='fortify_data_frame.dendrogram'>Build a data frame</h2><span id='topic+fortify_data_frame.dendrogram'></span><span id='topic+fortify_data_frame.hclust'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function converts various objects to a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dendrogram'
fortify_data_frame(
  data,
  ...,
  priority = "right",
  center = FALSE,
  type = "rectangle",
  leaf_pos = NULL,
  leaf_braches = NULL,
  reorder_branches = TRUE,
  branch_gap = NULL,
  root = NULL,
  double = TRUE,
  data_arg = NULL,
  call = NULL
)

## S3 method for class 'hclust'
fortify_data_frame(data, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_data_frame.dendrogram_+3A_data">data</code></td>
<td>
<p>A <code><a href="stats.html#topic+hclust">hclust</a></code> or a
<code><a href="stats.html#topic+dendrogram">dendrogram</a></code> object.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code>dendrogram</code> method.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_priority">priority</code></td>
<td>
<p>A string of &quot;left&quot; or &quot;right&quot;. if we draw from <code>right</code> to
<code>left</code>, the left will override the right, so we take the <code>"left"</code> as the
priority. If we draw from <code>left</code> to <code>right</code>, the right will override the
left, so we take the <code>"right"</code> as priority. This is used by
<code><a href="#topic+align_dendro">align_dendro()</a></code> to provide support of facet operation in ggplot2.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_center">center</code></td>
<td>
<p>A boolean value. if <code>TRUE</code>, nodes are plotted centered with
respect to all leaves/tips in the branch. Otherwise (default), plot them in
the middle of the direct child nodes.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_type">type</code></td>
<td>
<p>A string indicates the plot type, <code>"rectangle"</code> or <code>"triangle"</code>.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_leaf_pos">leaf_pos</code></td>
<td>
<p>The x-coordinates of the leaf node. Must be the same length
of the number of observations in <code>tree</code>.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_leaf_braches">leaf_braches</code></td>
<td>
<p>Branches of the leaf node. Must be the same length of the
number of observations in <code>tree</code>. Usually come from <a href="stats.html#topic+cutree">cutree</a>.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_reorder_branches">reorder_branches</code></td>
<td>
<p>A single boolean value, indicates whether reorder the
provided <code>leaf_braches</code> based on the actual index.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_branch_gap">branch_gap</code></td>
<td>
<p>A single numeric value indicates the gap between different
branches.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_root">root</code></td>
<td>
<p>A length one string or numeric indicates the root branch.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_double">double</code></td>
<td>
<p>A single logical value indicating whether horizontal lines
should be doubled when segments span multiple branches. If <code>TRUE</code>, the
horizontal lines will be repeated for each branch that the segment spans. If
<code>FALSE</code>, only one horizontal line will be drawn. This is used by
<code><a href="#topic+align_dendro">align_dendro()</a></code> to provide support of facet operation in ggplot2.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.dendrogram_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code style="white-space: pre;">&#8288;data frame&#8288;</code> with the node coordinates:
</p>

<ul>
<li> <p><code>.panel</code>: Similar with <code>panel</code> column, but always give the correct
branch for usage of the ggplot facet.
</p>
</li>
<li> <p><code>.index</code>: the original index in the tree for the the node
</p>
</li>
<li> <p><code>label</code>: node label text
</p>
</li>
<li> <p><code>x</code> and <code>y</code>: x-axis and y-axis coordinates for the node
</p>
</li>
<li> <p><code>branch</code>: which branch the node is. You can use this column to color
different groups.
</p>
</li>
<li> <p><code>panel</code>: which panel the node is, if we split the plot into panel
using <a href="ggplot2.html#topic+facet_grid">facet_grid</a>, this column will show
which panel the node is from. Note: some nodes may
fall outside panel (between two panels), so there are possible
<code>NA</code> values in this column.
</p>
</li>
<li> <p><code>leaf</code>: A logical value indicates whether the node is a leaf.
</p>
</li></ul>



<h3>ggalign attributes</h3>

<p><code>edge</code>: A <code style="white-space: pre;">&#8288;data frame&#8288;</code> for edge coordinates:
</p>

<ul>
<li> <p><code>.panel</code>: Similar with <code>panel</code> column, but always give the correct
branch for usage of the ggplot facet.
</p>
</li>
<li> <p><code>x</code> and <code>y</code>: x-axis and y-axis coordinates for the start node of the edge.
</p>
</li>
<li> <p><code>xend</code> and <code>yend</code>: the x-axis and y-axis coordinates of the terminal node
for edge.
</p>
</li>
<li> <p><code>branch</code>: which branch the edge is. You can use this column to color
different groups.
</p>
</li>
<li> <p><code>panel1</code> and <code>panel2</code>: The panel1 and panel2 columns have the same
functionality as <code>panel</code>, but they are specifically for the <code>edge</code> data
and correspond to both nodes of each edge.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> methods:
<code><a href="#topic+fortify_data_frame.character">fortify_data_frame.character</a>()</code>,
<code><a href="#topic+fortify_data_frame.default">fortify_data_frame.default</a>()</code>,
<code><a href="#topic+fortify_data_frame.matrix">fortify_data_frame.matrix</a>()</code>,
<code><a href="#topic+fortify_data_frame.phylo">fortify_data_frame.phylo</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fortify_data_frame(hclust(dist(USArrests), "ave"))
</code></pre>

<hr>
<h2 id='fortify_data_frame.matrix'>Build a data frame</h2><span id='topic+fortify_data_frame.matrix'></span><span id='topic+fortify_data_frame.DelayedMatrix'></span><span id='topic+fortify_data_frame.Matrix'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function converts various objects to a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
fortify_data_frame(data, ..., data_arg = NULL, call = NULL)

## S3 method for class 'DelayedMatrix'
fortify_data_frame(data, ...)

## S3 method for class 'Matrix'
fortify_data_frame(data, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_data_frame.matrix_+3A_data">data</code></td>
<td>
<p>A matrix-like object.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.matrix_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.matrix_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.matrix_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix will be transformed into a long-form data frame, where each row
represents a unique combination of matrix indices and their corresponding
values. The resulting data frame will contain the following columns:
</p>

<ul>
<li> <p><code>.row_names</code> and <code>.row_index</code>: the row names (only applicable when names
exist) and an integer representing the row index of the original matrix.
</p>
</li>
<li> <p><code>.column_names</code> and <code>.column_index</code>: the column names (only applicable
when names exist) and column index of the original matrix.
</p>
</li>
<li> <p><code>value</code>: the actual value.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> methods:
<code><a href="#topic+fortify_data_frame.character">fortify_data_frame.character</a>()</code>,
<code><a href="#topic+fortify_data_frame.default">fortify_data_frame.default</a>()</code>,
<code><a href="#topic+fortify_data_frame.dendrogram">fortify_data_frame.dendrogram</a>()</code>,
<code><a href="#topic+fortify_data_frame.phylo">fortify_data_frame.phylo</a>()</code>
</p>

<hr>
<h2 id='fortify_data_frame.phylo'>Build a data frame</h2><span id='topic+fortify_data_frame.phylo'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function converts various objects to a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylo'
fortify_data_frame(
  data,
  ...,
  type = "rectangle",
  center = FALSE,
  tree_type = NULL,
  tip_pos = NULL,
  data_arg = NULL,
  call = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_data_frame.phylo_+3A_data">data</code></td>
<td>
<p>A <code><a href="stats.html#topic+hclust">hclust</a></code> or a
<code><a href="stats.html#topic+dendrogram">dendrogram</a></code> object.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.phylo_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.phylo_+3A_type">type</code></td>
<td>
<p>A string indicates the plot type, <code>"rectangle"</code> or <code>"triangle"</code>.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.phylo_+3A_center">center</code></td>
<td>
<p>A boolean value. if <code>TRUE</code>, nodes are plotted centered with
respect to all leaves/tips in the branch. Otherwise (default), plot them in
the middle of the direct child nodes.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.phylo_+3A_tree_type">tree_type</code></td>
<td>
<p>A single string, one of
<code>"phylogram"</code> or <code>"cladogram"</code>, indicating the type of tree.
</p>

<ul>
<li> <p><code>phylogram</code>: Represents a phylogenetic tree where branch lengths indicate
evolutionary distance or time.
</p>
</li>
<li> <p><code>cladogram</code>: Represents a tree where branch lengths are not used, or the
branches do not reflect evolutionary time.
</p>
</li></ul>

<p>Usually, you don't need to modify this.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.phylo_+3A_tip_pos">tip_pos</code></td>
<td>
<p>The x-coordinates of the tip. Must be the same length
of the number of tips in <code>tree</code>.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.phylo_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_data_frame.phylo_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code style="white-space: pre;">&#8288;data frame&#8288;</code> with the node coordinates:
</p>

<ul>
<li> <p><code>.index</code>: the original index in the tree for the the tip/node.
</p>
</li>
<li> <p><code>label</code>: the tip/node label text.
</p>
</li>
<li> <p><code>x</code> and <code>y</code>: x-axis and y-axis coordinates for the tip/node.
</p>
</li>
<li> <p><code>tip</code>: A logical value indicates whether current node is a tip.
</p>
</li></ul>



<h3>ggalign attributes</h3>

<p><code>edge</code>: A <code style="white-space: pre;">&#8288;data frame&#8288;</code> for edge coordinates:
</p>

<ul>
<li> <p><code>x</code> and <code>y</code>: x-axis and y-axis coordinates for the start node of the edge.
</p>
</li>
<li> <p><code>xend</code> and <code>yend</code>: the x-axis and y-axis coordinates of the terminal node
for edge.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> methods:
<code><a href="#topic+fortify_data_frame.character">fortify_data_frame.character</a>()</code>,
<code><a href="#topic+fortify_data_frame.default">fortify_data_frame.default</a>()</code>,
<code><a href="#topic+fortify_data_frame.dendrogram">fortify_data_frame.dendrogram</a>()</code>,
<code><a href="#topic+fortify_data_frame.matrix">fortify_data_frame.matrix</a>()</code>
</p>

<hr>
<h2 id='fortify_matrix'>Build a Matrix</h2><span id='topic+fortify_matrix'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function converts various objects into a matrix format. By default, it
calls <code><a href="base.html#topic+as.matrix">as.matrix()</a></code> to build a matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fortify_matrix(data, ..., data_arg = NULL, call = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_matrix_+3A_data">data</code></td>
<td>
<p>An object to be converted into a matrix.</p>
</td></tr>
<tr><td><code id="fortify_matrix_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to methods.</p>
</td></tr>
<tr><td><code id="fortify_matrix_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_matrix_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix.
</p>


<h3><code>fortify_matrix</code> method collections</h3>


<ul>
<li> <p><code><a href="#topic+fortify_matrix.phylo">fortify_matrix.phylo()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_matrix.default">fortify_matrix.default()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_matrix.list_upset">fortify_matrix.list_upset()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_matrix.MAF">fortify_matrix.MAF()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_matrix.GISTIC">fortify_matrix.GISTIC()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_matrix.matrix">fortify_matrix.matrix()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_matrix.matrix_upset">fortify_matrix.matrix_upset()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_matrix.matrix_oncoplot">fortify_matrix.matrix_oncoplot()</a></code>
</p>
</li></ul>


<hr>
<h2 id='fortify_matrix.default'>Build a Matrix</h2><span id='topic+fortify_matrix.default'></span>

<h3>Description</h3>

<p>By default, it calls <code><a href="base.html#topic+as.matrix">as.matrix()</a></code> to build a matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
fortify_matrix(data, ..., data_arg = NULL, call = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_matrix.default_+3A_data">data</code></td>
<td>
<p>An object to be converted into a matrix.</p>
</td></tr>
<tr><td><code id="fortify_matrix.default_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
<tr><td><code id="fortify_matrix.default_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_matrix.default_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix.
</p>


<h3>See Also</h3>

<p>Other <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> methods:
<code><a href="#topic+fortify_matrix.GISTIC">fortify_matrix.GISTIC</a>()</code>,
<code><a href="#topic+fortify_matrix.MAF">fortify_matrix.MAF</a>()</code>,
<code><a href="#topic+fortify_matrix.list_upset">fortify_matrix.list_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix">fortify_matrix.matrix</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_oncoplot">fortify_matrix.matrix_oncoplot</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_upset">fortify_matrix.matrix_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.phylo">fortify_matrix.phylo</a>()</code>
</p>

<hr>
<h2 id='fortify_matrix.GISTIC'>Build a matrix from a maftools object</h2><span id='topic+fortify_matrix.GISTIC'></span>

<h3>Description</h3>

<p>Build a matrix from a maftools object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GISTIC'
fortify_matrix(
  data,
  ...,
  n_top = NULL,
  bands = NULL,
  ignored_bands = NULL,
  sample_anno = NULL,
  remove_empty_samples = TRUE,
  data_arg = NULL,
  call = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_matrix.GISTIC_+3A_data">data</code></td>
<td>
<p>A <code><a href="maftools.html#topic+readGistic">GISTIC</a></code> object.</p>
</td></tr>
<tr><td><code id="fortify_matrix.GISTIC_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
<tr><td><code id="fortify_matrix.GISTIC_+3A_n_top">n_top</code></td>
<td>
<p>A single number indicates how many top bands to be drawn.</p>
</td></tr>
<tr><td><code id="fortify_matrix.GISTIC_+3A_bands">bands</code></td>
<td>
<p>An atomic character defines the bands to draw.</p>
</td></tr>
<tr><td><code id="fortify_matrix.GISTIC_+3A_ignored_bands">ignored_bands</code></td>
<td>
<p>An atomic character defines the bands to be ignored.</p>
</td></tr>
<tr><td><code id="fortify_matrix.GISTIC_+3A_sample_anno">sample_anno</code></td>
<td>
<p>A data frame of sample clinical features to be added.</p>
</td></tr>
<tr><td><code id="fortify_matrix.GISTIC_+3A_remove_empty_samples">remove_empty_samples</code></td>
<td>
<p>A single boolean value indicating whether to drop
samples without any genomic alterations.</p>
</td></tr>
<tr><td><code id="fortify_matrix.GISTIC_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_matrix.GISTIC_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>ggalign attributes</h3>


<ul>
<li> <p><code>sample_anno</code>: sample clinical informations provided in <code>sample_anno</code>.
</p>
</li>
<li> <p><code>sample_summary</code>: sample copy number summary informations. See
<code>data@cnv.summary</code> for details.
</p>
</li>
<li> <p><code>cytoband_summary</code>: cytoband summary informations. See
<code>data@cytoband.summary</code> for details.
</p>
</li>
<li> <p><code>gene_summary</code>: gene summary informations. See
<code>data@gene.summary</code> for details.
</p>
</li>
<li> <p><code>summary</code>: A data frame of summary information. See <code>data@summary</code> for
details.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> methods:
<code><a href="#topic+fortify_matrix.MAF">fortify_matrix.MAF</a>()</code>,
<code><a href="#topic+fortify_matrix.default">fortify_matrix.default</a>()</code>,
<code><a href="#topic+fortify_matrix.list_upset">fortify_matrix.list_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix">fortify_matrix.matrix</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_oncoplot">fortify_matrix.matrix_oncoplot</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_upset">fortify_matrix.matrix_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.phylo">fortify_matrix.phylo</a>()</code>
</p>

<hr>
<h2 id='fortify_matrix.list_upset'>Build a Matrix for UpSet plot</h2><span id='topic+fortify_matrix.list_upset'></span><span id='topic+fortify_matrix.list'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>This function converts a list into a matrix format suitable for creating an
UpSet plot. It always returns a matrix for a <code>horizontal</code> UpSet plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'list_upset'
fortify_matrix(data, mode = "distinct", ..., data_arg = NULL, call = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_matrix.list_upset_+3A_data">data</code></td>
<td>
<p>A list of sets.</p>
</td></tr>
<tr><td><code id="fortify_matrix.list_upset_+3A_mode">mode</code></td>
<td>
<p>A string of <code>"distinct"</code>, <code>"intersect"</code>, or <code>"union"</code>
indicates the mode to define the set intersections. Check
<a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/upset-plot.html#upset-mode">https://jokergoo.github.io/ComplexHeatmap-reference/book/upset-plot.html#upset-mode</a>
for details.</p>
</td></tr>
<tr><td><code id="fortify_matrix.list_upset_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
<tr><td><code id="fortify_matrix.list_upset_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_matrix.list_upset_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>ggalign attributes</h3>


<ul>
<li> <p><code>intersection_sizes</code>: An integer vector indicating the size of each
intersection.
</p>
</li>
<li> <p><code>set_sizes</code>: An integer vector indicating the size of each set.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+tune.list">tune.list()</a></code>
</p>
<p>Other <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> methods:
<code><a href="#topic+fortify_matrix.GISTIC">fortify_matrix.GISTIC</a>()</code>,
<code><a href="#topic+fortify_matrix.MAF">fortify_matrix.MAF</a>()</code>,
<code><a href="#topic+fortify_matrix.default">fortify_matrix.default</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix">fortify_matrix.matrix</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_oncoplot">fortify_matrix.matrix_oncoplot</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_upset">fortify_matrix.matrix_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.phylo">fortify_matrix.phylo</a>()</code>
</p>

<hr>
<h2 id='fortify_matrix.MAF'>Build a Matrix for OncoPrint</h2><span id='topic+fortify_matrix.MAF'></span><span id='topic+fortify_matrix.MAF_pathways'></span>

<h3>Description</h3>

<p>Convert <code>MAF</code> object to a matrix:
</p>

<ul>
<li> <p><code>fortify_matrix.MAF</code>: Extract genomic alterations for genes.
</p>
</li>
<li> <p><code>fortify_matrix.MAF_pathways</code>: Extract genomic alterations for pathways.
<code><a href="#topic+tune.MAF">tune.MAF()</a></code> helps convert <code>MAF</code> object to a <code>MAF_pathways</code> object.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MAF'
fortify_matrix(
  data,
  ...,
  genes = NULL,
  n_top = NULL,
  remove_empty_genes = TRUE,
  remove_empty_samples = TRUE,
  collapse_vars = TRUE,
  use_syn = TRUE,
  missing_genes = "error",
  data_arg = NULL,
  call = NULL
)

## S3 method for class 'MAF_pathways'
fortify_matrix(
  data,
  ...,
  pathdb = "smgbp",
  remove_empty_pathways = TRUE,
  remove_empty_samples = TRUE,
  data_arg = NULL,
  call = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_matrix.MAF_+3A_data">data</code></td>
<td>
<p>A <code><a href="maftools.html#topic+read.maf">MAF</a></code> object.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_genes">genes</code></td>
<td>
<p>An atomic character defines the genes to draw.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_n_top">n_top</code></td>
<td>
<p>A single number indicates how many top genes to be drawn.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_remove_empty_genes">remove_empty_genes</code></td>
<td>
<p>A single boolean value indicats whether to drop
genes without any genomic alterations.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_remove_empty_samples">remove_empty_samples</code></td>
<td>
<p>A single boolean value indicats whether to drop
samples without any genomic alterations.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_collapse_vars">collapse_vars</code></td>
<td>
<p>A single boolean value indicating whether to collapse
multiple alterations in the same sample and gene into a single value
<code>"Multi_Hit"</code>. Alternatively, you can provide a single string indicates the
collapsed values.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_use_syn">use_syn</code></td>
<td>
<p>A single boolean value indicates whether to include synonymous
variants when Classifies SNPs into transitions and transversions.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_missing_genes">missing_genes</code></td>
<td>
<p>A string, either <code>"error"</code> or <code>"remove"</code>, specifying the
action for handling missing genes.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_pathdb">pathdb</code></td>
<td>
<p>A string of <code>"smgbp"</code> or <code>"sigpw"</code>, or a named list
of genes to define the pathways.</p>
</td></tr>
<tr><td><code id="fortify_matrix.MAF_+3A_remove_empty_pathways">remove_empty_pathways</code></td>
<td>
<p>A single boolean value indicats whether to drop
pathways without any genomic alterations.</p>
</td></tr>
</table>


<h3>ggalign attributes</h3>

<p>For <code>fortify_matrix.MAF</code>:
</p>

<ul>
<li> <p><code>gene_summary</code>: gene summary informations. See
<code>maftools::getGeneSummary()</code> for details.
</p>
</li>
<li> <p><code>sample_summary</code>: sample summary informations. See
<code>maftools::getSampleSummary()</code> for details.
</p>
</li>
<li> <p><code>sample_anno</code>: sample clinical informations. See
<code>maftools::getClinicalData()</code> for details.
</p>
</li>
<li> <p><code>n_genes</code>: Total number of genes.
</p>
</li>
<li> <p><code>n_samples</code>: Total number of samples.
</p>
</li>
<li> <p><code>titv</code>: A list of <code>data.frames</code> with Transitions and Transversions
summary. See <code>maftools::titv()</code> for details.
</p>
</li></ul>

<p>For <code>fortify_matrix.MAF_pathways</code>:
</p>

<ul>
<li> <p><code>gene_list</code>: the pathway contents.
</p>
</li>
<li> <p><code>pathway_summary</code>: pathway summary informations. See
<code>maftools::pathways()</code> for details.
</p>
</li>
<li> <p><code>sample_summary</code>: sample summary informations. See
<code>maftools::getSampleSummary()</code> for details.
</p>
</li>
<li> <p><code>sample_anno</code>: sample clinical informations. See
<code>maftools::getClinicalData()</code> for details.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> methods:
<code><a href="#topic+fortify_matrix.GISTIC">fortify_matrix.GISTIC</a>()</code>,
<code><a href="#topic+fortify_matrix.default">fortify_matrix.default</a>()</code>,
<code><a href="#topic+fortify_matrix.list_upset">fortify_matrix.list_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix">fortify_matrix.matrix</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_oncoplot">fortify_matrix.matrix_oncoplot</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_upset">fortify_matrix.matrix_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.phylo">fortify_matrix.phylo</a>()</code>
</p>

<hr>
<h2 id='fortify_matrix.matrix'>Build a matrix</h2><span id='topic+fortify_matrix.matrix'></span>

<h3>Description</h3>

<p>Build a matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
fortify_matrix(data, ..., data_arg = NULL, call = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_matrix.matrix_+3A_data">data</code></td>
<td>
<p>A matrix object.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>shape</h3>


<ul>
<li> <p><code>upset</code>: <code><a href="#topic+fortify_matrix.matrix_upset">fortify_matrix.matrix_upset()</a></code>
</p>
</li>
<li> <p><code>oncoplot</code>: <code><a href="#topic+fortify_matrix.matrix_oncoplot">fortify_matrix.matrix_oncoplot()</a></code>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> methods:
<code><a href="#topic+fortify_matrix.GISTIC">fortify_matrix.GISTIC</a>()</code>,
<code><a href="#topic+fortify_matrix.MAF">fortify_matrix.MAF</a>()</code>,
<code><a href="#topic+fortify_matrix.default">fortify_matrix.default</a>()</code>,
<code><a href="#topic+fortify_matrix.list_upset">fortify_matrix.list_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_oncoplot">fortify_matrix.matrix_oncoplot</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_upset">fortify_matrix.matrix_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.phylo">fortify_matrix.phylo</a>()</code>
</p>

<hr>
<h2 id='fortify_matrix.matrix_oncoplot'>Build a Matrix for OncoPrint</h2><span id='topic+fortify_matrix.matrix_oncoplot'></span>

<h3>Description</h3>

<p>Converts a matrix suitable for creating an OncoPrint. <code><a href="#topic+tune.matrix">tune.matrix()</a></code>
helps convert <code>matrix</code> object to a <code>matrix_oncoplot</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix_oncoplot'
fortify_matrix(
  data,
  ...,
  genes = NULL,
  n_top = NULL,
  remove_empty_genes = TRUE,
  remove_empty_samples = TRUE,
  missing_genes = "error",
  data_arg = NULL,
  call = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_matrix.matrix_oncoplot_+3A_data">data</code></td>
<td>
<p>A matrix where each row represents an genes, and each column
represents samples. The values in the matrix indicate whether the element is
part of the set.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_oncoplot_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_oncoplot_+3A_genes">genes</code></td>
<td>
<p>An atomic character defines the genes to draw.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_oncoplot_+3A_n_top">n_top</code></td>
<td>
<p>A single number indicates how many top genes to be drawn.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_oncoplot_+3A_remove_empty_genes">remove_empty_genes</code></td>
<td>
<p>A single boolean value indicats whether to drop
genes without any genomic alterations.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_oncoplot_+3A_remove_empty_samples">remove_empty_samples</code></td>
<td>
<p>A single boolean value indicats whether to drop
samples without any genomic alterations.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_oncoplot_+3A_missing_genes">missing_genes</code></td>
<td>
<p>A string, either <code>"error"</code> or <code>"remove"</code>, specifying the
action for handling missing genes.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_oncoplot_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_oncoplot_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>ggalign attributes</h3>


<ul>
<li> <p><code>gene_summary</code>: An integer vector of the altered samples for each gene.
</p>
</li>
<li> <p><code>sample_summary</code>: An integer vector of the altered genes for each sample.
</p>
</li>
<li> <p><code>n_genes</code>: Total number of genes.
</p>
</li>
<li> <p><code>n_samples</code>: Total number of samples.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+tune.matrix">tune.matrix()</a></code>
</p>
<p>Other <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> methods:
<code><a href="#topic+fortify_matrix.GISTIC">fortify_matrix.GISTIC</a>()</code>,
<code><a href="#topic+fortify_matrix.MAF">fortify_matrix.MAF</a>()</code>,
<code><a href="#topic+fortify_matrix.default">fortify_matrix.default</a>()</code>,
<code><a href="#topic+fortify_matrix.list_upset">fortify_matrix.list_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix">fortify_matrix.matrix</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_upset">fortify_matrix.matrix_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.phylo">fortify_matrix.phylo</a>()</code>
</p>

<hr>
<h2 id='fortify_matrix.matrix_upset'>Build a Matrix for UpSet plot</h2><span id='topic+fortify_matrix.matrix_upset'></span>

<h3>Description</h3>

<p>Converts a matrix suitable for creating an UpSet plot. <code><a href="#topic+tune.matrix">tune.matrix()</a></code>
helps convert <code>matrix</code> object to a <code>matrix_upset</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix_upset'
fortify_matrix(data, ..., data_arg = NULL, call = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_matrix.matrix_upset_+3A_data">data</code></td>
<td>
<p>A matrix where each row represents an element, and each column
defines a set. The values in the matrix indicate whether the element is part
of the set. Any non-missing value signifies that the element exists in the
set.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_upset_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="#topic+fortify_matrix.list_upset">fortify_matrix.list_upset</a></code>
</p>

<dl>
<dt><code>mode</code></dt><dd><p>A string of <code>"distinct"</code>, <code>"intersect"</code>, or <code>"union"</code>
indicates the mode to define the set intersections. Check
<a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/upset-plot.html#upset-mode">https://jokergoo.github.io/ComplexHeatmap-reference/book/upset-plot.html#upset-mode</a>
for details.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_upset_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_matrix.matrix_upset_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>ggalign attributes</h3>


<ul>
<li> <p><code>intersection_sizes</code>: An integer vector indicating the size of each
intersection.
</p>
</li>
<li> <p><code>set_sizes</code>: An integer vector indicating the size of each set.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+tune.matrix">tune.matrix()</a></code>
</p>
<p>Other <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> methods:
<code><a href="#topic+fortify_matrix.GISTIC">fortify_matrix.GISTIC</a>()</code>,
<code><a href="#topic+fortify_matrix.MAF">fortify_matrix.MAF</a>()</code>,
<code><a href="#topic+fortify_matrix.default">fortify_matrix.default</a>()</code>,
<code><a href="#topic+fortify_matrix.list_upset">fortify_matrix.list_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix">fortify_matrix.matrix</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_oncoplot">fortify_matrix.matrix_oncoplot</a>()</code>,
<code><a href="#topic+fortify_matrix.phylo">fortify_matrix.phylo</a>()</code>
</p>

<hr>
<h2 id='fortify_matrix.phylo'>Build a matrix from <code>phylo</code> object</h2><span id='topic+fortify_matrix.phylo'></span>

<h3>Description</h3>

<p>This method allows a <code><a href="ape.html#topic+as.phylo">phylo</a></code> object to be
directly input into <code>stack_discrete()</code> or <code>circle_discrete()</code>. This makes it
possible to add <code><a href="#topic+align_phylo">align_phylo()</a></code> to the stack independently, as
<code><a href="#topic+align_phylo">align_phylo()</a></code> requires the layout to have labels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylo'
fortify_matrix(data, ..., data_arg = caller_arg(data), call = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_matrix.phylo_+3A_data">data</code></td>
<td>
<p>A <code><a href="ape.html#topic+as.phylo">phylo</a></code> object.</p>
</td></tr>
<tr><td><code id="fortify_matrix.phylo_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
<tr><td><code id="fortify_matrix.phylo_+3A_data_arg">data_arg</code></td>
<td>
<p>The argument name for <code>data</code>. Developers can use it to
improve messages. Not used by the user.</p>
</td></tr>
<tr><td><code id="fortify_matrix.phylo_+3A_call">call</code></td>
<td>
<p>The execution environment where <code>data</code> and other arguments for
the method are collected. Developers can use it to improve messages. Not
used by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A one-column matrix where the tip labels are the values, and the row
names will also be the tip labels.
</p>


<h3>See Also</h3>

<p>Other <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> methods:
<code><a href="#topic+fortify_matrix.GISTIC">fortify_matrix.GISTIC</a>()</code>,
<code><a href="#topic+fortify_matrix.MAF">fortify_matrix.MAF</a>()</code>,
<code><a href="#topic+fortify_matrix.default">fortify_matrix.default</a>()</code>,
<code><a href="#topic+fortify_matrix.list_upset">fortify_matrix.list_upset</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix">fortify_matrix.matrix</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_oncoplot">fortify_matrix.matrix_oncoplot</a>()</code>,
<code><a href="#topic+fortify_matrix.matrix_upset">fortify_matrix.matrix_upset</a>()</code>
</p>

<hr>
<h2 id='free_align'>Free from alignment</h2><span id='topic+free_align'></span><span id='topic+free_border'></span><span id='topic+free_guide'></span><span id='topic+free_lab'></span><span id='topic+free_space'></span><span id='topic+free_vp'></span>

<h3>Description</h3>

<p><a href="#topic+align_plots">align_plots</a> will try to align plot panels, and every elements of the plot,
following functions romove these restrictions:
</p>

<ul>
<li> <p><code>free_align</code>: if we want to compose plots without alignment of some panel
axes (panel won't be aligned). we can wrap the plot with <code>free_align</code>.
</p>
</li>
<li> <p><code>free_border</code>: If we want to compose plots without alignment of the panel
borders (but still align the panels themselves), we can wrap the plot with
<code>free_border</code>.
</p>
</li>
<li> <p><code>free_lab</code>: If we want to compose plots without alignment of the axis
title, we can wrap the plot with <code>free_lab</code>.
</p>
</li>
<li> <p><code>free_space</code>: Removing the ggplot element sizes when aligning.
</p>
</li>
<li> <p><code>free_vp</code>: Customize the <a href="grid.html#topic+viewport">viewport</a> when aligning.
</p>
</li>
<li> <p><code>free_guide</code>: If we want to override the behaviour of the overall guides
behaviour, we can wrap the plot with <code>free_guide</code>.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>free_align(plot, axes = "tlbr")

free_border(plot, borders = "tlbr")

free_guide(plot, guides = "tlbr")

free_lab(plot, labs = "tlbr")

free_space(plot, spaces = "tlbr")

free_vp(plot, x = 0.5, y = 0.5, width = NA, height = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="free_align_+3A_plot">plot</code></td>
<td>
<p>A <a href="ggplot2.html#topic+ggplot">ggplot</a> or <a href="#topic+align_plots">alignpatches</a>
object.</p>
</td></tr>
<tr><td><code id="free_align_+3A_axes">axes</code></td>
<td>
<p>Which axes shouldn't be aligned? A string containing
one or more of <code>"t"</code>, <code>"l"</code>, <code>"b"</code>, and <code>"r"</code>.</p>
</td></tr>
<tr><td><code id="free_align_+3A_borders">borders</code></td>
<td>
<p>Which border shouldn't be aligned? A string containing one or
more of <code>"t"</code>, <code>"l"</code>, <code>"b"</code>, and <code>"r"</code>.</p>
</td></tr>
<tr><td><code id="free_align_+3A_guides">guides</code></td>
<td>
<p>A string containing one or more of <code>"t"</code>, <code>"l"</code>, <code>"b"</code>, and <code>"r"</code>
indicates which side of guide legends should be collected for the plot. If
<code>NULL</code>, no guide legends will be collected.</p>
</td></tr>
<tr><td><code id="free_align_+3A_labs">labs</code></td>
<td>
<p>Which axis labs to be free? A string containing one or more of
<code>"t"</code>, <code>"l"</code>, <code>"b"</code>, and <code>"r"</code>.</p>
</td></tr>
<tr><td><code id="free_align_+3A_spaces">spaces</code></td>
<td>
<p>Which border spaces should be removed? A string containing one
or more of <code>"t"</code>, <code>"l"</code>, <code>"b"</code>, and <code>"r"</code>.</p>
</td></tr>
<tr><td><code id="free_align_+3A_x">x</code></td>
<td>
<p>A numeric vector or unit object specifying x-location.</p>
</td></tr>
<tr><td><code id="free_align_+3A_y">y</code></td>
<td>
<p>A numeric vector or unit object specifying y-location.</p>
</td></tr>
<tr><td><code id="free_align_+3A_width">width</code></td>
<td>
<p>A numeric vector or unit object specifying width.</p>
</td></tr>
<tr><td><code id="free_align_+3A_height">height</code></td>
<td>
<p>A numeric vector or unit object specifying height.</p>
</td></tr>
<tr><td><code id="free_align_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="grid.html#topic+viewport">grid::viewport</a></code>
</p>

<dl>
<dt><code>default.units</code></dt><dd><p>A string indicating the default units to use
if <code>x</code>, <code>y</code>, <code>width</code>, or <code>height</code>
are only given as numeric vectors.</p>
</dd>
<dt><code>just</code></dt><dd><p>A string or numeric
vector specifying the justification of the viewport
relative to its (x, y) location.  If there are two values, the first
value specifies horizontal justification and the second value specifies
vertical justification.  Possible string values are: <code>"left"</code>,
<code>"right"</code>, <code>"centre"</code>, <code>"center"</code>, <code>"bottom"</code>,
and <code>"top"</code>.  For numeric values, 0 means left alignment
and 1 means right alignment.
</p>
</dd>
<dt><code>gp</code></dt><dd><p>An object of class <code>"gpar"</code>, typically the output
from a call to the function <code><a href="grid.html#topic+gpar">gpar</a></code>.  This is basically
a list of graphical parameter settings.</p>
</dd>
<dt><code>clip</code></dt><dd><p>One of <code>"on"</code>, <code>"inherit"</code>, or
<code>"off"</code>, indicating whether to
clip to the extent of this viewport, inherit the clipping region
from the parent viewport, or turn clipping off altogether.
For back-compatibility, a logical value of <code>TRUE</code> corresponds
to <code>"on"</code> and <code>FALSE</code> corresponds to <code>"inherit"</code>.
</p>
<p>May also be a grob (or a gTree) that describes a clipping path
or the result of a call to <code><a href="grid.html#topic+as.path">as.path</a></code>.</p>
</dd>
<dt><code>mask</code></dt><dd><p>One of <code>"none"</code> (or <code>FALSE</code>)
or <code>"inherit"</code> (or <code>TRUE</code>) 
or a grob (or a gTree) or the result of call to
<code><a href="grid.html#topic+as.mask">as.mask</a></code>.  This specifies that the viewport
should have no mask, or it should inherit the mask of its parent,
or it should have its own mask, as described by the grob.</p>
</dd>
<dt><code>xscale</code></dt><dd><p>A numeric vector of length two indicating the minimum and
maximum on the x-scale.  The limits may not be identical. </p>
</dd>
<dt><code>yscale</code></dt><dd><p>A numeric vector of length two indicating the minimum
and maximum on the y-scale. The limits may not be identical. </p>
</dd>
<dt><code>angle</code></dt><dd><p>A numeric value indicating the angle of rotation of the
viewport.  Positive values indicate the amount of rotation, in
degrees, anticlockwise from the positive x-axis.</p>
</dd>
<dt><code>layout</code></dt><dd><p>A Grid layout object which splits the viewport into
subregions.</p>
</dd>
<dt><code>layout.pos.row</code></dt><dd><p>A numeric vector giving the
rows occupied by this viewport in its
parent's layout.</p>
</dd>
<dt><code>layout.pos.col</code></dt><dd><p>A numeric vector giving the
columns occupied by this viewport in its
parent's layout.</p>
</dd>
<dt><code>name</code></dt><dd><p>A character value to uniquely identify the viewport
once it has been pushed onto the viewport tree.  </p>
</dd>
</dl>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code>free_align</code>: A modified version of <code>plot</code> with a <code>free_align</code> class.
</p>
</li></ul>


<ul>
<li> <p><code>free_border</code>: A modified version of <code>plot</code> with a <code>free_border</code> class.
</p>
</li></ul>


<ul>
<li> <p><code>free_guide</code>: A modified version of <code>plot</code> with a <code>free_guide</code> class.
</p>
</li></ul>


<ul>
<li> <p><code>free_lab</code>: A modified version of <code>plot</code> with a <code>free_lab</code> class.
</p>
</li></ul>


<ul>
<li> <p><code>free_space</code>: A modified version of <code>plot</code> with a <code>free_space</code> class.
</p>
</li></ul>


<ul>
<li> <p><code>free_vp</code>: A modified version of <code>plot</code> with a <code>free_vp</code> class.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># directly copied from `patchwork`
# Sometimes you have a plot that defies good composition alginment, e.g. due
# to long axis labels
p1 &lt;- ggplot(mtcars) +
    geom_bar(aes(y = factor(gear), fill = factor(gear))) +
    scale_y_discrete(
        "",
        labels = c(
            "3 gears are often enough",
            "But, you know, 4 is a nice number",
            "I would def go with 5 gears in a modern car"
        )
    )

# When combined with other plots it ends up looking bad
p2 &lt;- ggplot(mtcars) +
    geom_point(aes(mpg, disp))

align_plots(p1, p2, ncol = 1L)

# We can fix this be using `free_align`
align_plots(free_align(p1), p2, ncol = 1L)

# If we still want the panels to be aligned to the right, we can choose to
# free only the left side
align_plots(free_align(p1, axes = "l"), p2, ncol = 1L)

# We could use `free_lab` to fix the layout in a different way
align_plots(p1, free_lab(p2), ncol = 1L)

# `free_border` is similar with `free_lab`, they have a distinction in terms
# of placement on either the top or bottom side of the panel. Specifically,
# the top side contains the `title` and `subtitle`, while the bottom side
# contains the `caption`. free_lab() does not attach these elements in the
# panel area.
p3 &lt;- ggplot(mtcars) +
    geom_point(aes(hp, wt, colour = mpg)) +
    ggtitle("Plot 3")
p_axis_top &lt;- ggplot(mtcars) +
    geom_point(aes(mpg, disp)) +
    ggtitle("Plot axis in top") +
    scale_x_continuous(position = "top")
align_plots(p_axis_top, free_lab(p3))
align_plots(p_axis_top, free_border(p3))

# Another issue is that long labels can occupy much spaces
align_plots(NULL, p1, p2, p2)

# This can be fixed with `free_space`
align_plots(NULL, free_space(p1, "l"), p2, p2)

</code></pre>

<hr>
<h2 id='geom_draw'>Layer with Grid or Function</h2><span id='topic+geom_draw'></span>

<h3>Description</h3>

<p>Draw a ggplot2 layer using a grob or a function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_draw(
  draw,
  mapping = NULL,
  data = NULL,
  type = "group",
  stat = "identity",
  position = "identity",
  ...,
  na.rm = FALSE,
  show.legend = FALSE,
  inherit.aes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_draw_+3A_draw">draw</code></td>
<td>
<p>Either a <a href="grid.html#topic+grid.grob">grob</a> object or a function (can be
purrr-style) that accepts at least one argument (a data frame of
transformed coordinates) and returns a <a href="grid.html#topic+grid.grob">grob</a>.</p>
</td></tr>
<tr><td><code id="geom_draw_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="geom_draw_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="geom_draw_+3A_type">type</code></td>
<td>
<p>A single string of <code>"group"</code> or <code>"panel"</code>, <code>"group"</code>
draws geoms with <code>draw_group</code>, which displays multiple observations as one
geometric object, and <code>"panel"</code> draws geoms with <code>draw_panel</code>, displaying
individual graphical objects for each observation (row). Default:
<code>"group"</code>.</p>
</td></tr>
<tr><td><code id="geom_draw_+3A_stat">stat</code></td>
<td>
<p>The statistical transformation to use on the data for this layer.
When using a <code style="white-space: pre;">&#8288;geom_*()&#8288;</code> function to construct a layer, the <code>stat</code>
argument can be used the override the default coupling between geoms and
stats. The <code>stat</code> argument accepts the following:
</p>

<ul>
<li><p> A <code>Stat</code> ggproto subclass, for example <code>StatCount</code>.
</p>
</li>
<li><p> A string naming the stat. To give the stat as a string, strip the
function name of the <code>stat_</code> prefix. For example, to use <code>stat_count()</code>,
give the stat as <code>"count"</code>.
</p>
</li>
<li><p> For more information and other ways to specify the stat, see the
<a href="ggplot2.html#topic+layer_stats">layer stat</a> documentation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_draw_+3A_position">position</code></td>
<td>
<p>A position adjustment to use on the data for this layer. This
can be used in various ways, including to prevent overplotting and
improving the display. The <code>position</code> argument accepts the following:
</p>

<ul>
<li><p> The result of calling a position function, such as <code>position_jitter()</code>.
This method allows for passing extra arguments to the position.
</p>
</li>
<li><p> A string naming the position adjustment. To give the position as a
string, strip the function name of the <code>position_</code> prefix. For example,
to use <code>position_jitter()</code>, give the position as <code>"jitter"</code>.
</p>
</li>
<li><p> For more information and other ways to specify the position, see the
<a href="ggplot2.html#topic+layer_positions">layer position</a> documentation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_draw_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggplot2.html#topic+layer">layer()</a></code>'s <code>params</code> argument. These
arguments broadly fall into one of 4 categories below. Notably, further
arguments to the <code>position</code> argument, or aesthetics that are required
can <em>not</em> be passed through <code>...</code>. Unknown arguments that are not part
of the 4 categories below are ignored.
</p>

<ul>
<li><p> Static aesthetics that are not mapped to a scale, but are at a fixed
value and apply to the layer as a whole. For example, <code>colour = "red"</code>
or <code>linewidth = 3</code>. The geom's documentation has an <strong>Aesthetics</strong>
section that lists the available options. The 'required' aesthetics
cannot be passed on to the <code>params</code>. Please note that while passing
unmapped aesthetics as vectors is technically possible, the order and
required length is not guaranteed to be parallel to the input data.
</p>
</li>
<li><p> When constructing a layer using
a <code style="white-space: pre;">&#8288;stat_*()&#8288;</code> function, the <code>...</code> argument can be used to pass on
parameters to the <code>geom</code> part of the layer. An example of this is
<code>stat_density(geom = "area", outline.type = "both")</code>. The geom's
documentation lists which parameters it can accept.
</p>
</li>
<li><p> Inversely, when constructing a layer using a
<code style="white-space: pre;">&#8288;geom_*()&#8288;</code> function, the <code>...</code> argument can be used to pass on parameters
to the <code>stat</code> part of the layer. An example of this is
<code>geom_area(stat = "density", adjust = 0.5)</code>. The stat's documentation
lists which parameters it can accept.
</p>
</li>
<li><p> The <code>key_glyph</code> argument of <code><a href="ggplot2.html#topic+layer">layer()</a></code> may also be passed on through
<code>...</code>. This can be one of the functions described as
<a href="ggplot2.html#topic+draw_key">key glyphs</a>, to change the display of the layer in the legend.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_draw_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p>
</td></tr>
<tr><td><code id="geom_draw_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="geom_draw_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you want to combine the functionality of multiple geoms, it can
typically be achieved by preparing the data for each geom inside the
<code style="white-space: pre;">&#8288;draw_*()&#8288;</code> call and sending it off to the different geoms, collecting the
output in a <code><a href="grid.html#topic+grid.grob">grid::gList</a></code> (a list of grobs) for <code>draw_group()</code> or a
<code><a href="grid.html#topic+grid.grob">grid::gTree</a></code> (a grob containing multiple child grobs) for
<code>draw_panel()</code>.
</p>


<h3>See Also</h3>

<p><a href="https://ggplot2.tidyverse.org/reference/ggplot2-ggproto.html">https://ggplot2.tidyverse.org/reference/ggplot2-ggproto.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>text &lt;- grid::textGrob(
    "ggdraw",
    x = c(0, 0, 0.5, 1, 1),
    y = c(0, 1, 0.5, 0, 1),
    hjust = c(0, 0, 0.5, 1, 1),
    vjust = c(0, 1, 0.5, 0, 1)
)
ggplot(data.frame(x = 1, y = 2)) +
    geom_draw(text)
</code></pre>

<hr>
<h2 id='geom_gshape'>Layer with a customized shape graphic using grid functions.</h2><span id='topic+geom_gshape'></span>

<h3>Description</h3>

<p>Layer with a customized shape graphic using grid functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_gshape(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  ...,
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_gshape_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="geom_gshape_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="geom_gshape_+3A_stat">stat</code></td>
<td>
<p>The statistical transformation to use on the data for this layer.
When using a <code style="white-space: pre;">&#8288;geom_*()&#8288;</code> function to construct a layer, the <code>stat</code>
argument can be used the override the default coupling between geoms and
stats. The <code>stat</code> argument accepts the following:
</p>

<ul>
<li><p> A <code>Stat</code> ggproto subclass, for example <code>StatCount</code>.
</p>
</li>
<li><p> A string naming the stat. To give the stat as a string, strip the
function name of the <code>stat_</code> prefix. For example, to use <code>stat_count()</code>,
give the stat as <code>"count"</code>.
</p>
</li>
<li><p> For more information and other ways to specify the stat, see the
<a href="ggplot2.html#topic+layer_stats">layer stat</a> documentation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_gshape_+3A_position">position</code></td>
<td>
<p>A position adjustment to use on the data for this layer. This
can be used in various ways, including to prevent overplotting and
improving the display. The <code>position</code> argument accepts the following:
</p>

<ul>
<li><p> The result of calling a position function, such as <code>position_jitter()</code>.
This method allows for passing extra arguments to the position.
</p>
</li>
<li><p> A string naming the position adjustment. To give the position as a
string, strip the function name of the <code>position_</code> prefix. For example,
to use <code>position_jitter()</code>, give the position as <code>"jitter"</code>.
</p>
</li>
<li><p> For more information and other ways to specify the position, see the
<a href="ggplot2.html#topic+layer_positions">layer position</a> documentation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_gshape_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggplot2.html#topic+layer">layer()</a></code>'s <code>params</code> argument. These
arguments broadly fall into one of 4 categories below. Notably, further
arguments to the <code>position</code> argument, or aesthetics that are required
can <em>not</em> be passed through <code>...</code>. Unknown arguments that are not part
of the 4 categories below are ignored.
</p>

<ul>
<li><p> Static aesthetics that are not mapped to a scale, but are at a fixed
value and apply to the layer as a whole. For example, <code>colour = "red"</code>
or <code>linewidth = 3</code>. The geom's documentation has an <strong>Aesthetics</strong>
section that lists the available options. The 'required' aesthetics
cannot be passed on to the <code>params</code>. Please note that while passing
unmapped aesthetics as vectors is technically possible, the order and
required length is not guaranteed to be parallel to the input data.
</p>
</li>
<li><p> When constructing a layer using
a <code style="white-space: pre;">&#8288;stat_*()&#8288;</code> function, the <code>...</code> argument can be used to pass on
parameters to the <code>geom</code> part of the layer. An example of this is
<code>stat_density(geom = "area", outline.type = "both")</code>. The geom's
documentation lists which parameters it can accept.
</p>
</li>
<li><p> Inversely, when constructing a layer using a
<code style="white-space: pre;">&#8288;geom_*()&#8288;</code> function, the <code>...</code> argument can be used to pass on parameters
to the <code>stat</code> part of the layer. An example of this is
<code>geom_area(stat = "density", adjust = 0.5)</code>. The stat's documentation
lists which parameters it can accept.
</p>
</li>
<li><p> The <code>key_glyph</code> argument of <code><a href="ggplot2.html#topic+layer">layer()</a></code> may also be passed on through
<code>...</code>. This can be one of the functions described as
<a href="ggplot2.html#topic+draw_key">key glyphs</a>, to change the display of the layer in the legend.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_gshape_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p>
</td></tr>
<tr><td><code id="geom_gshape_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="geom_gshape_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>geom_gshape</code> depends on the new aesthetics <code>gshape</code> (shape with grid
functions), which should always be provided with <code><a href="#topic+scale_gshape_manual">scale_gshape_manual()</a></code>,
in which, we can provide a list of grobs or functions that define how each
value should be drawn. Any ggplot2 aesthetics can be used as the arguments.
</p>


<h3>Aesthetics</h3>

<p><code>geom_gshape()</code> understands the following aesthetics (required aesthetics are in bold):
</p>

<ul>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">x</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">y</a></code></strong>
</p>
</li>
<li> <p><strong><code>gshape</code></strong>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">alpha</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">colour</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">fill</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_group_order">group</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linetype</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linewidth</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">shape</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">size</a></code>
</p>
</li>
<li> <p><code>stroke</code>
</p>
</li></ul>

<p>Learn more about setting these aesthetics in <code>vignette("ggplot2-specs", package = "ggplot2")</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(grid)
ggplot(data.frame(value = letters[seq_len(5)], y = seq_len(5))) +
    geom_gshape(aes(x = 1, y = y, gshape = value, fill = value)) +
    scale_gshape_manual(values = list(
        a = function(x, y, width, height, fill) {
            rectGrob(x, y,
                width = width, height = height,
                gp = gpar(fill = fill),
                default.units = "native"
            )
        },
        b = function(x, y, width, height, fill) {
            rectGrob(x, y,
                width = width, height = height,
                gp = gpar(fill = fill),
                default.units = "native"
            )
        },
        c = function(x, y, width, height, fill) {
            rectGrob(x, y,
                width = width, height = height,
                gp = gpar(fill = fill),
                default.units = "native"
            )
        },
        d = function(x, y, width, height, shape) {
            gList(
                pointsGrob(x, y, pch = shape),
                # To ensure the rectangle color is shown in the legends, you
                # must explicitly provide a color argument and include it in
                # the `gpar()` of the graphical object
                rectGrob(x, y, width, height,
                    gp = gpar(col = "black", fill = NA)
                )
            )
        },
        e = function(xmin, xmax, ymin, ymax) {
            segmentsGrob(
                xmin, ymin,
                xmax, ymax,
                gp = gpar(lwd = 2)
            )
        }
    )) +
    scale_fill_brewer(palette = "Dark2") +
    theme_void()

</code></pre>

<hr>
<h2 id='geom_pie'>Pie charts</h2><span id='topic+geom_pie'></span>

<h3>Description</h3>

<p>Pie charts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_pie(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  ...,
  clockwise = TRUE,
  steps = 100,
  lineend = "butt",
  linejoin = "round",
  linemitre = 10,
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_pie_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="geom_pie_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="geom_pie_+3A_stat">stat</code></td>
<td>
<p>The statistical transformation to use on the data for this layer.
When using a <code style="white-space: pre;">&#8288;geom_*()&#8288;</code> function to construct a layer, the <code>stat</code>
argument can be used the override the default coupling between geoms and
stats. The <code>stat</code> argument accepts the following:
</p>

<ul>
<li><p> A <code>Stat</code> ggproto subclass, for example <code>StatCount</code>.
</p>
</li>
<li><p> A string naming the stat. To give the stat as a string, strip the
function name of the <code>stat_</code> prefix. For example, to use <code>stat_count()</code>,
give the stat as <code>"count"</code>.
</p>
</li>
<li><p> For more information and other ways to specify the stat, see the
<a href="ggplot2.html#topic+layer_stats">layer stat</a> documentation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_pie_+3A_position">position</code></td>
<td>
<p>A position adjustment to use on the data for this layer. This
can be used in various ways, including to prevent overplotting and
improving the display. The <code>position</code> argument accepts the following:
</p>

<ul>
<li><p> The result of calling a position function, such as <code>position_jitter()</code>.
This method allows for passing extra arguments to the position.
</p>
</li>
<li><p> A string naming the position adjustment. To give the position as a
string, strip the function name of the <code>position_</code> prefix. For example,
to use <code>position_jitter()</code>, give the position as <code>"jitter"</code>.
</p>
</li>
<li><p> For more information and other ways to specify the position, see the
<a href="ggplot2.html#topic+layer_positions">layer position</a> documentation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_pie_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggplot2.html#topic+layer">layer()</a></code>'s <code>params</code> argument. These
arguments broadly fall into one of 4 categories below. Notably, further
arguments to the <code>position</code> argument, or aesthetics that are required
can <em>not</em> be passed through <code>...</code>. Unknown arguments that are not part
of the 4 categories below are ignored.
</p>

<ul>
<li><p> Static aesthetics that are not mapped to a scale, but are at a fixed
value and apply to the layer as a whole. For example, <code>colour = "red"</code>
or <code>linewidth = 3</code>. The geom's documentation has an <strong>Aesthetics</strong>
section that lists the available options. The 'required' aesthetics
cannot be passed on to the <code>params</code>. Please note that while passing
unmapped aesthetics as vectors is technically possible, the order and
required length is not guaranteed to be parallel to the input data.
</p>
</li>
<li><p> When constructing a layer using
a <code style="white-space: pre;">&#8288;stat_*()&#8288;</code> function, the <code>...</code> argument can be used to pass on
parameters to the <code>geom</code> part of the layer. An example of this is
<code>stat_density(geom = "area", outline.type = "both")</code>. The geom's
documentation lists which parameters it can accept.
</p>
</li>
<li><p> Inversely, when constructing a layer using a
<code style="white-space: pre;">&#8288;geom_*()&#8288;</code> function, the <code>...</code> argument can be used to pass on parameters
to the <code>stat</code> part of the layer. An example of this is
<code>geom_area(stat = "density", adjust = 0.5)</code>. The stat's documentation
lists which parameters it can accept.
</p>
</li>
<li><p> The <code>key_glyph</code> argument of <code><a href="ggplot2.html#topic+layer">layer()</a></code> may also be passed on through
<code>...</code>. This can be one of the functions described as
<a href="ggplot2.html#topic+draw_key">key glyphs</a>, to change the display of the layer in the legend.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_pie_+3A_clockwise">clockwise</code></td>
<td>
<p>A single boolean value indicates clockwise or not.</p>
</td></tr>
<tr><td><code id="geom_pie_+3A_steps">steps</code></td>
<td>
<p>An integer indicating the number of steps to generate the pie
chart radian. Increasing this value results in a smoother pie circular.</p>
</td></tr>
<tr><td><code id="geom_pie_+3A_lineend">lineend</code></td>
<td>
<p>Line end style (round, butt, square).</p>
</td></tr>
<tr><td><code id="geom_pie_+3A_linejoin">linejoin</code></td>
<td>
<p>Line join style (round, mitre, bevel).</p>
</td></tr>
<tr><td><code id="geom_pie_+3A_linemitre">linemitre</code></td>
<td>
<p>Line mitre limit (number greater than 1).</p>
</td></tr>
<tr><td><code id="geom_pie_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p>
</td></tr>
<tr><td><code id="geom_pie_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="geom_pie_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
</table>


<h3>new aesthetics</h3>


<ul>
<li> <p><code>angle</code>: the pie circle angle.
</p>
</li>
<li> <p><code>angle0</code>: the initial pie circle angle.
</p>
</li>
<li> <p><code>radius</code>: the circle radius.
</p>
</li></ul>



<h3>Aesthetics</h3>

<p><code>geom_pie()</code> understands the following aesthetics (required aesthetics are in bold):
</p>

<ul>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">x</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">y</a></code></strong>
</p>
</li>
<li> <p><strong><code>angle</code></strong>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">alpha</a></code>
</p>
</li>
<li> <p><code>angle0</code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">colour</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">fill</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_group_order">group</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linetype</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linewidth</a></code>
</p>
</li>
<li> <p><code>radius</code>
</p>
</li></ul>

<p>Learn more about setting these aesthetics in <code>vignette("ggplot2-specs", package = "ggplot2")</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ggplot(data.frame(x = 1:10, y = 1:10, value = 1:10 / sum(1:10))) +
    geom_pie(aes(x, y, angle = value * 360))
</code></pre>

<hr>
<h2 id='geom_rect3d'>Add z-aesthetic for geom_tile</h2><span id='topic+geom_rect3d'></span><span id='topic+geom_tile3d'></span>

<h3>Description</h3>

<p>Add z-aesthetic for geom_tile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_rect3d(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  ...,
  lineend = "butt",
  linejoin = "round",
  linemitre = 10,
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE
)

geom_tile3d(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  ...,
  lineend = "butt",
  linejoin = "round",
  linemitre = 10,
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_rect3d_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="geom_rect3d_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="geom_rect3d_+3A_stat">stat</code></td>
<td>
<p>The statistical transformation to use on the data for this layer.
When using a <code style="white-space: pre;">&#8288;geom_*()&#8288;</code> function to construct a layer, the <code>stat</code>
argument can be used the override the default coupling between geoms and
stats. The <code>stat</code> argument accepts the following:
</p>

<ul>
<li><p> A <code>Stat</code> ggproto subclass, for example <code>StatCount</code>.
</p>
</li>
<li><p> A string naming the stat. To give the stat as a string, strip the
function name of the <code>stat_</code> prefix. For example, to use <code>stat_count()</code>,
give the stat as <code>"count"</code>.
</p>
</li>
<li><p> For more information and other ways to specify the stat, see the
<a href="ggplot2.html#topic+layer_stats">layer stat</a> documentation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_rect3d_+3A_position">position</code></td>
<td>
<p>A position adjustment to use on the data for this layer. This
can be used in various ways, including to prevent overplotting and
improving the display. The <code>position</code> argument accepts the following:
</p>

<ul>
<li><p> The result of calling a position function, such as <code>position_jitter()</code>.
This method allows for passing extra arguments to the position.
</p>
</li>
<li><p> A string naming the position adjustment. To give the position as a
string, strip the function name of the <code>position_</code> prefix. For example,
to use <code>position_jitter()</code>, give the position as <code>"jitter"</code>.
</p>
</li>
<li><p> For more information and other ways to specify the position, see the
<a href="ggplot2.html#topic+layer_positions">layer position</a> documentation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_rect3d_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggplot2.html#topic+layer">layer()</a></code>'s <code>params</code> argument. These
arguments broadly fall into one of 4 categories below. Notably, further
arguments to the <code>position</code> argument, or aesthetics that are required
can <em>not</em> be passed through <code>...</code>. Unknown arguments that are not part
of the 4 categories below are ignored.
</p>

<ul>
<li><p> Static aesthetics that are not mapped to a scale, but are at a fixed
value and apply to the layer as a whole. For example, <code>colour = "red"</code>
or <code>linewidth = 3</code>. The geom's documentation has an <strong>Aesthetics</strong>
section that lists the available options. The 'required' aesthetics
cannot be passed on to the <code>params</code>. Please note that while passing
unmapped aesthetics as vectors is technically possible, the order and
required length is not guaranteed to be parallel to the input data.
</p>
</li>
<li><p> When constructing a layer using
a <code style="white-space: pre;">&#8288;stat_*()&#8288;</code> function, the <code>...</code> argument can be used to pass on
parameters to the <code>geom</code> part of the layer. An example of this is
<code>stat_density(geom = "area", outline.type = "both")</code>. The geom's
documentation lists which parameters it can accept.
</p>
</li>
<li><p> Inversely, when constructing a layer using a
<code style="white-space: pre;">&#8288;geom_*()&#8288;</code> function, the <code>...</code> argument can be used to pass on parameters
to the <code>stat</code> part of the layer. An example of this is
<code>geom_area(stat = "density", adjust = 0.5)</code>. The stat's documentation
lists which parameters it can accept.
</p>
</li>
<li><p> The <code>key_glyph</code> argument of <code><a href="ggplot2.html#topic+layer">layer()</a></code> may also be passed on through
<code>...</code>. This can be one of the functions described as
<a href="ggplot2.html#topic+draw_key">key glyphs</a>, to change the display of the layer in the legend.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_rect3d_+3A_lineend">lineend</code></td>
<td>
<p>Line end style (round, butt, square).</p>
</td></tr>
<tr><td><code id="geom_rect3d_+3A_linejoin">linejoin</code></td>
<td>
<p>Line join style (round, mitre, bevel).</p>
</td></tr>
<tr><td><code id="geom_rect3d_+3A_linemitre">linemitre</code></td>
<td>
<p>Line mitre limit (number greater than 1).</p>
</td></tr>
<tr><td><code id="geom_rect3d_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p>
</td></tr>
<tr><td><code id="geom_rect3d_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="geom_rect3d_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
</table>


<h3>new aesthetics</h3>


<ul>
<li> <p><code>z</code>: the third dimention (in the z direction).
</p>
</li>
<li> <p><code>theta</code>: Angle between x-axis and z-axis.
</p>
</li></ul>



<h3>Aesthetics</h3>

<p><code>geom_rect3d()</code> understands the following aesthetics (required aesthetics are in bold):
</p>

<ul>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">xmin</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">xmax</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">ymin</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">ymax</a></code></strong>
</p>
</li>
<li> <p><strong><code>z</code></strong>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">alpha</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">colour</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">fill</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_group_order">group</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linetype</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linewidth</a></code>
</p>
</li></ul>

<p>Learn more about setting these aesthetics in <code>vignette("ggplot2-specs", package = "ggplot2")</code>.
</p>
<p><code>geom_tile3d()</code> understands the following aesthetics (required aesthetics are in bold):
</p>

<ul>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">x</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">y</a></code></strong>
</p>
</li>
<li> <p><strong><code>z</code></strong>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">alpha</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">colour</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">fill</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_group_order">group</a></code>
</p>
</li>
<li> <p><code>height</code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linetype</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linewidth</a></code>
</p>
</li>
<li> <p><code>width</code>
</p>
</li></ul>

<p>Learn more about setting these aesthetics in <code>vignette("ggplot2-specs", package = "ggplot2")</code>.
</p>

<hr>
<h2 id='geom_subrect'>Subdivide Rectangles</h2><span id='topic+geom_subrect'></span><span id='topic+geom_subtile'></span>

<h3>Description</h3>

<p>These geoms subdivide rectangles with shared borders into a grid. Both geoms
achieve the same result but differ in how the rectangles are parameterized:
</p>

<ul>
<li> <p><strong><code>geom_subrect()</code></strong>: Defines rectangles using their four corners (<code>xmin</code>,
<code>xmax</code>, <code>ymin</code>, <code>ymax</code>).
</p>
</li>
<li> <p><strong><code>geom_subtile()</code></strong>: Defines rectangles using the center (<code>x</code>, <code>y</code>) and
dimensions (<code>width</code>, <code>height</code>).
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>geom_subrect(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  ...,
  byrow = TRUE,
  direction = NULL,
  lineend = "butt",
  linejoin = "mitre",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE
)

geom_subtile(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  ...,
  byrow = TRUE,
  direction = NULL,
  lineend = "butt",
  linejoin = "mitre",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_subrect_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="geom_subrect_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="geom_subrect_+3A_stat">stat</code></td>
<td>
<p>The statistical transformation to use on the data for this layer.
When using a <code style="white-space: pre;">&#8288;geom_*()&#8288;</code> function to construct a layer, the <code>stat</code>
argument can be used the override the default coupling between geoms and
stats. The <code>stat</code> argument accepts the following:
</p>

<ul>
<li><p> A <code>Stat</code> ggproto subclass, for example <code>StatCount</code>.
</p>
</li>
<li><p> A string naming the stat. To give the stat as a string, strip the
function name of the <code>stat_</code> prefix. For example, to use <code>stat_count()</code>,
give the stat as <code>"count"</code>.
</p>
</li>
<li><p> For more information and other ways to specify the stat, see the
<a href="ggplot2.html#topic+layer_stats">layer stat</a> documentation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_subrect_+3A_position">position</code></td>
<td>
<p>A position adjustment to use on the data for this layer. This
can be used in various ways, including to prevent overplotting and
improving the display. The <code>position</code> argument accepts the following:
</p>

<ul>
<li><p> The result of calling a position function, such as <code>position_jitter()</code>.
This method allows for passing extra arguments to the position.
</p>
</li>
<li><p> A string naming the position adjustment. To give the position as a
string, strip the function name of the <code>position_</code> prefix. For example,
to use <code>position_jitter()</code>, give the position as <code>"jitter"</code>.
</p>
</li>
<li><p> For more information and other ways to specify the position, see the
<a href="ggplot2.html#topic+layer_positions">layer position</a> documentation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_subrect_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggplot2.html#topic+layer">layer()</a></code>'s <code>params</code> argument. These
arguments broadly fall into one of 4 categories below. Notably, further
arguments to the <code>position</code> argument, or aesthetics that are required
can <em>not</em> be passed through <code>...</code>. Unknown arguments that are not part
of the 4 categories below are ignored.
</p>

<ul>
<li><p> Static aesthetics that are not mapped to a scale, but are at a fixed
value and apply to the layer as a whole. For example, <code>colour = "red"</code>
or <code>linewidth = 3</code>. The geom's documentation has an <strong>Aesthetics</strong>
section that lists the available options. The 'required' aesthetics
cannot be passed on to the <code>params</code>. Please note that while passing
unmapped aesthetics as vectors is technically possible, the order and
required length is not guaranteed to be parallel to the input data.
</p>
</li>
<li><p> When constructing a layer using
a <code style="white-space: pre;">&#8288;stat_*()&#8288;</code> function, the <code>...</code> argument can be used to pass on
parameters to the <code>geom</code> part of the layer. An example of this is
<code>stat_density(geom = "area", outline.type = "both")</code>. The geom's
documentation lists which parameters it can accept.
</p>
</li>
<li><p> Inversely, when constructing a layer using a
<code style="white-space: pre;">&#8288;geom_*()&#8288;</code> function, the <code>...</code> argument can be used to pass on parameters
to the <code>stat</code> part of the layer. An example of this is
<code>geom_area(stat = "density", adjust = 0.5)</code>. The stat's documentation
lists which parameters it can accept.
</p>
</li>
<li><p> The <code>key_glyph</code> argument of <code><a href="ggplot2.html#topic+layer">layer()</a></code> may also be passed on through
<code>...</code>. This can be one of the functions described as
<a href="ggplot2.html#topic+draw_key">key glyphs</a>, to change the display of the layer in the legend.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_subrect_+3A_byrow">byrow</code></td>
<td>
<p>A single boolean value indicates whether we should arrange the
divided rectangles in the row-major order.</p>
</td></tr>
<tr><td><code id="geom_subrect_+3A_direction">direction</code></td>
<td>
<p>A string specifying the arrangement direction:
</p>

<ul>
<li> <p><code>"h"</code>(<code>horizontal</code>): Creates a single row (one-row layout).
</p>
</li>
<li> <p><code>"v"</code>(<code>vertical</code>): Creates a single column (one-column layout).
</p>
</li>
<li> <p><code>NULL</code>: Automatically determines the layout dimensions using logic similar
to <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap()</a></code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_subrect_+3A_lineend">lineend</code></td>
<td>
<p>Line end style (round, butt, square).</p>
</td></tr>
<tr><td><code id="geom_subrect_+3A_linejoin">linejoin</code></td>
<td>
<p>Line join style (round, mitre, bevel).</p>
</td></tr>
<tr><td><code id="geom_subrect_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p>
</td></tr>
<tr><td><code id="geom_subrect_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="geom_subrect_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
</table>


<h3>Aesthetics</h3>

<p><code>geom_subrect()</code> understands the following aesthetics (required aesthetics are in bold):
</p>

<ul>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">xmin</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">xmax</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">ymin</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">ymax</a></code></strong>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">alpha</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">colour</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">fill</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_group_order">group</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linetype</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linewidth</a></code>
</p>
</li></ul>

<p>Learn more about setting these aesthetics in <code>vignette("ggplot2-specs", package = "ggplot2")</code>.
</p>
<p><code>geom_subtile()</code> understands the following aesthetics (required aesthetics are in bold):
</p>

<ul>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">x</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">y</a></code></strong>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">alpha</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">colour</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">fill</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_group_order">group</a></code>
</p>
</li>
<li> <p><code>height</code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linetype</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linewidth</a></code>
</p>
</li>
<li> <p><code>width</code>
</p>
</li></ul>

<p>Learn more about setting these aesthetics in <code>vignette("ggplot2-specs", package = "ggplot2")</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># arranges by row
ggplot(data.frame(value = letters[seq_len(5)])) +
    geom_subtile(aes(x = 1, y = 1, fill = value))

# arranges by column
ggplot(data.frame(value = letters[seq_len(9)])) +
    geom_subtile(aes(x = 1, y = 1, fill = value), byrow = FALSE)

# one-row
ggplot(data.frame(value = letters[seq_len(4)])) +
    geom_subtile(aes(x = 1, y = 1, fill = value), direction = "h")

# one-column
ggplot(data.frame(value = letters[seq_len(4)])) +
    geom_subtile(aes(x = 1, y = 1, fill = value), direction = "v")

</code></pre>

<hr>
<h2 id='ggalign'>Add ggplot by Aligning discrete or continuous variable</h2><span id='topic+ggalign'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p><code>ggalign()</code> is similar to <code>ggplot</code> in that it initializes a <code>ggplot</code> data and
<code>mapping</code>. <code>ggalign()</code> allowing you to provide data in various formats,
including matrices, data frames, or simple vectors. By default, it will
inherit from the layout. If a function, it will apply with the layout matrix.
<code>ggalign()</code> focuses on integrating plots into a layout by aligning the axes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggalign(
  data = waiver(),
  mapping = aes(),
  ...,
  size = NULL,
  no_axes = NULL,
  active = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggalign_+3A_data">data</code></td>
<td>
<p>The following options can be used:
</p>

<ul>
<li> <p><code>NULL</code>: No data is set.
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+waiver">waiver()</a></code>: Inherits the data from the layout matrix.
</p>
</li>
<li><p> A <code>function</code> (including purrr-like lambda syntax): Applied to the layout
matrix to transform the data before use. To transform the final plot
data, please use <code><a href="#topic+scheme_data">scheme_data()</a></code>.
</p>
</li>
<li><p> A <code>matrix</code>, <code>data.frame</code>, or atomic vector.
</p>
</li></ul>
</td></tr>
<tr><td><code id="ggalign_+3A_mapping">mapping</code></td>
<td>
<p>Default list of aesthetic mappings to use for plot.
If not specified, must be supplied in each layer added to the plot.</p>
</td></tr>
<tr><td><code id="ggalign_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; Additional arguments passed to
<code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code>.</p>
</td></tr>
<tr><td><code id="ggalign_+3A_size">size</code></td>
<td>
<p>The relative size of the plot, can be specified as a
<code><a href="grid.html#topic+unit">unit()</a></code>. Note that for <code><a href="#topic+circle_layout">circle_layout()</a></code>, all size values
will be interpreted as relative sizes, as this layout type adjusts based on
the available space in the circular arrangement.</p>
</td></tr>
<tr><td><code id="ggalign_+3A_no_axes">no_axes</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a> Logical; if <code>TRUE</code>,
removes axes elements for the alignment axis using <code><a href="#topic+theme_no_axes">theme_no_axes()</a></code>. By
default, will use the option-
<code>"ggalign.align_no_axes"</code>.</p>
</td></tr>
<tr><td><code id="ggalign_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>ggplot2 specification</h3>

<p><code>ggalign</code> initializes a ggplot object. The underlying data is created using
<code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code>. Please refer to it for more details.
</p>
<p>When aligning discrete variables, <code>ggalign()</code> always applies a default
mapping for the axis of the data index in the layout. Specifically:
</p>

<ul>
<li> <p><code>aes(y = .data$.y)</code> is used for the horizontal <code>stack_layout()</code> (including
left and right annotations).
</p>
</li>
<li> <p><code>aes(x = .data$.x)</code> is used for the vertical <code>stack_layout()</code> (including
top and bottom annotations) and <code>circle_layout()</code>.
</p>
</li></ul>

<p>The following columns will be added to the data frame to align discrete
variables:
</p>

<ul>
<li> <p><code>.panel</code>: The panel for the aligned axis. Refers to the <code>x-axis</code> for
vertical <code>stack_layout()</code> (including top and bottom annotations), and the
<code>y-axis</code> for horizontal <code>stack_layout()</code> (including left and right
annotations).
</p>
</li>
<li> <p><code>.names</code> (<code><a href="vctrs.html#topic+vec_names">vec_names()</a></code>) and <code>.index</code>
(<code><a href="vctrs.html#topic+vec_size">vec_size()</a></code>/<code><a href="base.html#topic+NROW">NROW()</a></code>): Character names (if
available) and the integer index of the original data.
</p>
</li>
<li> <p><code>.x</code>/<code>.y</code> and <code>.discrete_x</code>/<code>.discrete_y</code>: Integer indices for <code>x</code>/<code>y</code>
coordinates, and a factor of the data labels (only applicable when names
exist).
</p>
</li></ul>

<p>It is recommended to use <code>.x</code>/<code>.y</code>, or <code>.discrete_x</code>/<code>.discrete_y</code> as the
<code>x</code>/<code>y</code> mapping.
</p>
<p>If the data inherits from <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code>, additional
columns will be added:
</p>

<ul>
<li> <p><code>.extra_panel</code>: Provides the panel information for the column (left or
right annotation) or row (top or bottom annotation).
</p>
</li>
<li> <p><code>.extra_index</code>: The index information for the column (left or right
annotation) or row (top or bottom annotation).
</p>
</li></ul>



<h3>Discrete Axis Alignment</h3>

<p>It is important to note that we consider rows as observations, meaning
<code>vec_size(data)</code>/<code>NROW(data)</code> must match the number of observations along the
axis used for alignment (x-axis for a vertical stack layout, y-axis for a
horizontal stack layout).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ggheatmap(matrix(rnorm(81), nrow = 9)) +
    anno_top() +
    ggalign() +
    geom_point(aes(y = value))

ggheatmap(matrix(rnorm(81), nrow = 9)) +
    anno_top(size = 0.5) +
    align_dendro(k = 3L) +
    ggalign(data = NULL, size = 0.2) +
    geom_tile(aes(y = 1L, fill = .panel))

</code></pre>

<hr>
<h2 id='ggalign_attr'>Get Data from the Attribute Attached by ggalign</h2><span id='topic+ggalign_attr'></span><span id='topic+ggalign_lvls'></span>

<h3>Description</h3>

<p><code>ggalign_attr</code> provides access to supplementary information stored as
attributes during the layout rendering process. These attributes, commonly
attached during data transformation by functions like <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> or
<code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code>, can include essential details such as filtered or
supplementary data that inform downstream operations.
</p>
<p>An additional attribute, which stores the factor levels, can be accessed with
<code>ggalign_lvls</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggalign_attr(x, field = NULL, check = TRUE)

ggalign_lvls(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggalign_attr_+3A_x">x</code></td>
<td>
<p>Data used, typically inherited from the layout <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code> or <code><a href="#topic+stack_layout">stack_layout()</a></code> object.</p>
</td></tr>
<tr><td><code id="ggalign_attr_+3A_field">field</code></td>
<td>
<p>A string specifying the particular data to retrieve from the
attached attribute. If <code>NULL</code>, the entire attached attribute list will be
returned.</p>
</td></tr>
<tr><td><code id="ggalign_attr_+3A_check">check</code></td>
<td>
<p>A boolean indicating whether to check if the <code>field</code> exists. If
<code>TRUE</code>, an error will be raised if the specified <code>field</code> does not exist.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Attributes attached to the data are especially useful when the input data is
transformed in ways that limit access to the complete dataset. For example,
<code><a href="#topic+fortify_matrix.MAF">fortify_matrix.MAF()</a></code> might filter mutation data while adding attributes
that retain important context, such as the total number of observations, for
detailed or aggregated analyses. Additionally, it stores the levels of
<code>Variant_Classification</code> for further usage.
</p>


<h3>Value</h3>


<ul>
<li> <p><code>ggalign_attr</code>: The specified data from the attached supplementary data or
<code>NULL</code> if it is unavailable.
</p>
</li>
<li> <p><code>ggalign_lvls</code>: The attached supplementary levels or <code>NULL</code> if it is
unavailable.
</p>
</li></ul>


<hr>
<h2 id='ggalign_data_set'>Attach supplementary data and levels for ggalign</h2><span id='topic+ggalign_data_set'></span>

<h3>Description</h3>

<p>Attach supplementary data and levels for ggalign
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggalign_data_set(.data, ..., .lvls = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggalign_data_set_+3A_.data">.data</code></td>
<td>
<p>Input data for the layout.</p>
</td></tr>
<tr><td><code id="ggalign_data_set_+3A_...">...</code></td>
<td>
<p>A list of data to be attached.</p>
</td></tr>
<tr><td><code id="ggalign_data_set_+3A_.lvls">.lvls</code></td>
<td>
<p>A character vector representing the attached levels.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Used by developers in <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>, <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code>,
and other related methods.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggalign_attr">ggalign_attr()</a></code>/<code><a href="#topic+ggalign_lvls">ggalign_lvls()</a></code>
</p>

<hr>
<h2 id='ggalign_stat'>Get the statistics from the layout</h2><span id='topic+ggalign_stat'></span><span id='topic+ggalign_stat.QuadLayout'></span><span id='topic+ggalign_stat.StackLayout'></span>

<h3>Description</h3>

<p>Get the statistics from the layout
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggalign_stat(x, ...)

## S3 method for class 'QuadLayout'
ggalign_stat(x, position, ...)

## S3 method for class 'StackLayout'
ggalign_stat(x, what, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggalign_stat_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code> or <code><a href="#topic+stack_layout">stack_layout()</a></code> object.</p>
</td></tr>
<tr><td><code id="ggalign_stat_+3A_...">...</code></td>
<td>
<p>Arguments passed to methods.</p>
</td></tr>
<tr><td><code id="ggalign_stat_+3A_position">position</code></td>
<td>
<p>A string of <code>"top"</code>, <code>"left"</code>, <code>"bottom"</code>, or <code>"right"</code>.</p>
</td></tr>
<tr><td><code id="ggalign_stat_+3A_what">what</code></td>
<td>
<p>A single number or string of the plot elements in the stack
layout.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The statistics
</p>

<hr>
<h2 id='ggalignGrob'>Generate a plot grob.</h2><span id='topic+ggalignGrob'></span>

<h3>Description</h3>

<p>Generate a plot grob.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggalignGrob(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggalignGrob_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <a href="grid.html#topic+grid.grob">grob</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob()</a></code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ggalignGrob(ggplot())
</code></pre>

<hr>
<h2 id='ggcross'>Connect two layout crosswise</h2><span id='topic+ggcross'></span>

<h3>Description</h3>

<p><code>ggcross</code> resets the layout ordering index of a <code><a href="#topic+stack_cross">stack_cross()</a></code>. This
allows you to add other <code style="white-space: pre;">&#8288;align_*&#8288;</code> objects to define a new layout ordering
index. Any objects added after <code>ggcross</code> will use this updated layout
ordering index. This feature is particularly useful for creating <code>tanglegram</code>
visualizations. <code>ggcross()</code> is an alias of <code>ggcross()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggcross(mapping = aes(), size = NULL, no_axes = NULL, active = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggcross_+3A_mapping">mapping</code></td>
<td>
<p>Default list of aesthetic mappings to use for plot.
If not specified, must be supplied in each layer added to the plot.</p>
</td></tr>
<tr><td><code id="ggcross_+3A_size">size</code></td>
<td>
<p>The relative size of the plot, can be specified as a
<code><a href="grid.html#topic+unit">unit()</a></code>. Note that for <code><a href="#topic+circle_layout">circle_layout()</a></code>, all size values
will be interpreted as relative sizes, as this layout type adjusts based on
the available space in the circular arrangement.</p>
</td></tr>
<tr><td><code id="ggcross_+3A_no_axes">no_axes</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a> Logical; if <code>TRUE</code>,
removes axes elements for the alignment axis using <code><a href="#topic+theme_no_axes">theme_no_axes()</a></code>. By
default, will use the option-
<code>"ggalign.align_no_axes"</code>.</p>
</td></tr>
<tr><td><code id="ggcross_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>ggplot2 specification</h3>

<p><code>ggcross()</code> initializes a ggplot <code>data</code> and <code>mapping</code>.
</p>
<p><code>ggcross()</code> always applies a default mapping for the axis of the data index
in the layout. This mapping is <code>aes(y = .data$.y)</code> for horizontal stack
layout (including left and right annotation) and <code>aes(x = .data$.x)</code> for
vertical stack layout (including top and bottom annotation).
</p>
<p>The data in the underlying <code>ggplot</code> object will contain following columns:
</p>

<ul>
<li> <p><code>.panel</code>: The panel for the aligned axis. Refers to the <code>x-axis</code> for
vertical <code>stack_layout()</code> (including top and bottom annotations), and the
<code>y-axis</code> for horizontal <code>stack_layout()</code> (including left and right
annotations).
</p>
</li>
<li> <p><code>.names</code> (<code><a href="vctrs.html#topic+vec_names">vec_names()</a></code>) and <code>.index</code>
(<code><a href="vctrs.html#topic+vec_size">vec_size()</a></code>/<code><a href="base.html#topic+NROW">NROW()</a></code>): Character names (if
available) and the integer index of the original data.
</p>
</li>
<li> <p><code>.hand</code>: a factor indicates the index groups.
</p>
</li>
<li> <p><code>.x</code>/<code>.y</code> and <code>.discrete_x</code>/<code>.discrete_y</code>: Integer indices for <code>x</code>/<code>y</code>
coordinates, and a factor of the data labels (only applicable when names
exist).
</p>
</li></ul>

<p>It is recommended to use <code>.x</code>/<code>.y</code>, or <code>.discrete_x</code>/<code>.discrete_y</code> as the
<code>x</code>/<code>y</code> mapping.
</p>

<hr>
<h2 id='ggfree'>Add ggplot to layout without alignment</h2><span id='topic+ggfree'></span><span id='topic+ggfree.default'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>The <code>ggfree()</code> function allows you to incorporate a ggplot object into your
layout. Unlike <code>ggalign()</code>, which aligns every axis value precisely,
<code>ggfree()</code> focuses on integrating plots into the layout without enforcing
strict axis alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggfree(data = waiver(), ..., size = NULL, active = NULL)

## Default S3 method:
ggfree(data = waiver(), mapping = aes(), ..., size = NULL, active = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggfree_+3A_data">data</code></td>
<td>
<p>The following options can be used:
</p>

<ul>
<li> <p><code>NULL</code>: No data is set.
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+waiver">waiver()</a></code>: Inherits the data from the layout matrix.
</p>
</li>
<li><p> A <code>function</code> (including purrr-like lambda syntax): Applied to the layout
matrix to transform the data before use. To transform the final plot
data, please use <code><a href="#topic+scheme_data">scheme_data()</a></code>.
</p>
</li>
<li><p> A <code>matrix</code>, <code>data.frame</code>, or atomic vector.
</p>
</li></ul>
</td></tr>
<tr><td><code id="ggfree_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; Additional arguments passed to
<code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code>.</p>
</td></tr>
<tr><td><code id="ggfree_+3A_size">size</code></td>
<td>
<p>The relative size of the plot, can be specified as a
<code><a href="grid.html#topic+unit">unit()</a></code>. Note that for <code><a href="#topic+circle_layout">circle_layout()</a></code>, all size values
will be interpreted as relative sizes, as this layout type adjusts based on
the available space in the circular arrangement.</p>
</td></tr>
<tr><td><code id="ggfree_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
<tr><td><code id="ggfree_+3A_mapping">mapping</code></td>
<td>
<p>Default list of aesthetic mappings to use for plot.
If not specified, must be supplied in each layer added to the plot.</p>
</td></tr>
</table>


<h3>ggplot2 specification</h3>

<p><code>ggalign</code> initializes a ggplot object. The underlying data is created using
<code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code>. Please refer to this method for more details.
</p>
<p>When used in <code>quad_layout()</code>/<code>ggheatmap()</code>, if the data is inherited from the
<code>quad_layout()</code> and the other direction aligns discrete variables, following
columns will be added:
</p>

<ul>
<li> <p><code>.extra_panel</code>: Provides the panel information for the column (left or
right annotation) or row (top or bottom annotation).
</p>
</li>
<li> <p><code>.extra_index</code>: The index information for the column (left or right
annotation) or row (top or bottom annotation).
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>ggheatmap(matrix(rnorm(56), nrow = 7)) +
    anno_top() +
    align_dendro() +
    ggfree(mtcars, aes(wt, mpg)) +
    geom_point()
</code></pre>

<hr>
<h2 id='ggmark'>Add a plot to annotate selected observations</h2><span id='topic+ggmark'></span>

<h3>Description</h3>

<p>Add a plot to annotate selected observations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggmark(
  mark,
  data = waiver(),
  mapping = aes(),
  ...,
  group1 = NULL,
  group2 = NULL,
  obs_size = 1,
  size = NULL,
  active = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggmark_+3A_mark">mark</code></td>
<td>
<p>A <code><a href="#topic+mark_draw">mark_draw()</a></code> object to define how to draw the links. Like
<code><a href="#topic+mark_line">mark_line()</a></code>, <code><a href="#topic+mark_tetragon">mark_tetragon()</a></code>. Note the names of the pair links will
be used to define the panel names so must be unique.</p>
</td></tr>
<tr><td><code id="ggmark_+3A_data">data</code></td>
<td>
<p>The following options can be used:
</p>

<ul>
<li> <p><code>NULL</code>: No data is set.
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+waiver">waiver()</a></code>: Inherits the data from the layout matrix.
</p>
</li>
<li><p> A <code>function</code> (including purrr-like lambda syntax): Applied to the layout
matrix to transform the data before use. To transform the final plot
data, please use <code><a href="#topic+scheme_data">scheme_data()</a></code>.
</p>
</li>
<li><p> A <code>matrix</code>, <code>data.frame</code>, or atomic vector.
</p>
</li></ul>
</td></tr>
<tr><td><code id="ggmark_+3A_mapping">mapping</code></td>
<td>
<p>Default list of aesthetic mappings to use for plot.
If not specified, must be supplied in each layer added to the plot.</p>
</td></tr>
<tr><td><code id="ggmark_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; Additional arguments passed to
<code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code>.</p>
</td></tr>
<tr><td><code id="ggmark_+3A_group1">group1</code>, <code id="ggmark_+3A_group2">group2</code></td>
<td>
<p>A single boolean value indicating whether to use the
panel group information from the layout as the paired groups. By default,
if no specific observations are selected in <code>mark</code>, <code>ggmark()</code> will
automatically connect all observations and group them according to the
layout's defined groups.</p>
</td></tr>
<tr><td><code id="ggmark_+3A_obs_size">obs_size</code></td>
<td>
<p>A single numeric value that indicates the size of a single
observation, ranging from <code style="white-space: pre;">&#8288;(0, 1]&#8288;</code>.</p>
</td></tr>
<tr><td><code id="ggmark_+3A_size">size</code></td>
<td>
<p>The relative size of the plot, can be specified as a
<code><a href="grid.html#topic+unit">unit()</a></code>. Note that for <code><a href="#topic+circle_layout">circle_layout()</a></code>, all size values
will be interpreted as relative sizes, as this layout type adjusts based on
the available space in the circular arrangement.</p>
</td></tr>
<tr><td><code id="ggmark_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>ggplot2 specification</h3>

<p><code>ggmark</code> initializes a ggplot object. The underlying data is created using
<code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code>. Please refer to it for more details.
</p>
<p>In addition, the following columns will be added to the data frame:
</p>

<ul>
<li> <p><code>.panel</code>: the panel for the aligned axis. It means <code>x-axis</code> for vertical
stack layout (including top and bottom annotation), <code>y-axis</code> for
horizontal stack layout (including left and right annotation).
</p>
</li>
<li> <p><code>.names</code> (<code><a href="vctrs.html#topic+vec_names">vec_names()</a></code>) and <code>.index</code>
(<code><a href="vctrs.html#topic+vec_size">vec_size()</a></code>/<code><a href="base.html#topic+NROW">NROW()</a></code>): a character names (only
applicable when names exists) and an integer of index of the original
data.
</p>
</li>
<li> <p><code>.hand</code>: A factor with levels <code>c("left", "right")</code> for horizontal stack
layouts, or <code>c("top", "bottom")</code> for vertical stack layouts, indicating
the position of the linked observations.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
small_mat &lt;- matrix(rnorm(56), nrow = 7)
rownames(small_mat) &lt;- paste0("row", seq_len(nrow(small_mat)))
colnames(small_mat) &lt;- paste0("column", seq_len(ncol(small_mat)))

# mark_line
ggheatmap(small_mat) +
    theme(axis.text.x = element_text(hjust = 0, angle = -60)) +
    anno_right() +
    align_kmeans(3L) +
    ggmark(mark_line(I(1:3) ~ NULL)) +
    geom_boxplot(aes(.names, value)) +
    theme(plot.margin = margin(l = 0.1, t = 0.1, unit = "npc"))

# mark_tetragon
ggheatmap(small_mat) +
    theme(axis.text.x = element_text(hjust = 0, angle = -60)) +
    anno_right() +
    align_kmeans(3L) +
    ggmark(mark_tetragon(I(1:3) ~ NULL)) +
    geom_boxplot(aes(.names, value)) +
    theme(plot.margin = margin(l = 0.1, t = 0.1, unit = "npc"))
</code></pre>

<hr>
<h2 id='ggoncoplot'>Create an OncoPrint</h2><span id='topic+ggoncoplot'></span><span id='topic+ggoncoplot.default'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>The <code>ggoncoplot()</code> function generates <code>oncoPrint</code> visualizations that display
genetic alterations in a matrix format. This function is especially useful
for visualizing complex genomic data, such as mutations, copy number
variations, and other genomic alterations in cancer research.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggoncoplot(
  data = NULL,
  mapping = aes(),
  ...,
  map_width = NULL,
  map_height = NULL,
  reorder_row = reorder_column,
  reorder_column = TRUE,
  width = NA,
  height = NA,
  filling = waiver(),
  theme = NULL,
  active = NULL
)

## Default S3 method:
ggoncoplot(
  data = NULL,
  mapping = aes(),
  ...,
  map_width = NULL,
  map_height = NULL,
  reorder_row = reorder_column,
  reorder_column = TRUE,
  width = NA,
  height = NA,
  filling = waiver(),
  theme = NULL,
  active = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggoncoplot_+3A_data">data</code></td>
<td>
<p>A character matrix which encodes the alterations, you can use
<code>";"</code>, <code>":"</code>, <code>","</code>, or <code>"|"</code> to separate multiple alterations.</p>
</td></tr>
<tr><td><code id="ggoncoplot_+3A_mapping">mapping</code></td>
<td>
<p>Default list of aesthetic mappings to use for main plot in the
layout. If not specified, must be supplied in each layer added to the main
plot.</p>
</td></tr>
<tr><td><code id="ggoncoplot_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>.</p>
</td></tr>
<tr><td><code id="ggoncoplot_+3A_map_width">map_width</code>, <code id="ggoncoplot_+3A_map_height">map_height</code></td>
<td>
<p>A named numeric value defines the width/height of
each alterations.</p>
</td></tr>
<tr><td><code id="ggoncoplot_+3A_reorder_row">reorder_row</code></td>
<td>
<p>A boolean value indicating whether to reorder the rows
based on the frequency of alterations. You can set this to <code>FALSE</code>, then add
<code>align_order(~rowSums(!is.na(.x)), reverse = TRUE)</code> to achieve the same
result. You may also need to set <code>strit = FALSE</code> in <code><a href="#topic+align_order">align_order()</a></code> if
there are already groups.</p>
</td></tr>
<tr><td><code id="ggoncoplot_+3A_reorder_column">reorder_column</code></td>
<td>
<p>A boolean value indicating whether to reorder the
columns based on the characteristics of the alterations. You can set this to
<code>FALSE</code>, then add <code>align_order2(memo_order)</code> to achieve the same result. You
may also need to set <code>strit = FALSE</code> in <code><a href="#topic+align_order2">align_order2()</a></code> if there are
already groups.</p>
</td></tr>
<tr><td><code id="ggoncoplot_+3A_width">width</code>, <code id="ggoncoplot_+3A_height">height</code></td>
<td>
<p>The relative width/height of the main plot, can be a
<code><a href="grid.html#topic+unit">unit</a></code> object.</p>
</td></tr>
<tr><td><code id="ggoncoplot_+3A_filling">filling</code></td>
<td>
<p>Same as <code><a href="#topic+ggheatmap">ggheatmap()</a></code>, but only <code>"tile"</code> can be used.</p>
</td></tr>
<tr><td><code id="ggoncoplot_+3A_theme">theme</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+theme">theme()</a></code> object used to customize various
elements of the layout, including <code>guides</code>, <code>title</code>, <code>subtitle</code>, <code>caption</code>,
<code>margins</code>, <code>panel.border</code>, and <code>background</code>. By default, the theme will
inherit from the parent <code>layout</code>. It also controls the panel spacing for all
plots in the layout.</p>
</td></tr>
<tr><td><code id="ggoncoplot_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ggoncoplot()</code> is a wrapper around the <code><a href="#topic+ggheatmap">ggheatmap()</a></code> function, designed to
simplify the creation of <code>OncoPrint</code>-style visualizations. The function
automatically processes the input character matrix by splitting the encoded
alterations (delimited by <code>";"</code>, <code>":"</code>, <code>","</code>, or <code>"|"</code>) into
individual genomic events and unnesting the columns for visualization.
</p>


<h3>Value</h3>

<p>A <code>HeatmapLayout</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A simple example from `ComplexHeatmap`
mat &lt;- read.table(textConnection(
    "s1,s2,s3
g1,snv;indel,snv,indel
g2,,snv;indel,snv
g3,snv,,indel;snv"
), row.names = 1, header = TRUE, sep = ",", stringsAsFactors = FALSE)

ggoncoplot(mat, map_width = c(snv = 0.5), map_height = c(indel = 0.9)) +
    # Note that guide legends from `geom_tile` and `geom_bar` are different.
    # Although they appear similar, the internal mechanisms won't collapse
    # the guide legends. Therefore, we remove the guide legends from
    # `geom_tile`.
    guides(fill = "none") +
    anno_top(size = 0.5) +
    ggalign() +
    geom_bar(aes(fill = value), data = function(x) {
        subset(x, !is.na(value))
    }) +
    anno_right(size = 0.5) +
    ggalign() +
    geom_bar(aes(fill = value), orientation = "y", data = function(x) {
        subset(x, !is.na(value))
    }) &amp;
    scale_fill_brewer(palette = "Dark2", na.translate = FALSE)
</code></pre>

<hr>
<h2 id='ggupset'>Create an UpSet plot</h2><span id='topic+ggupset'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p><code>ggupset</code> is a specialized version of <code><a href="#topic+quad_discrete">quad_discrete()</a></code>, which simplifies
the creation of Upset plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggupset(
  data = NULL,
  mapping = aes(),
  ...,
  direction = "h",
  point = NULL,
  line = NULL,
  rect = NULL,
  width = NA,
  height = NA,
  theme = NULL,
  active = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggupset_+3A_data">data</code></td>
<td>
<p>Data used to create the UpSet plot. <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> will be
used to convert the data to a matrix. Currently, only
<code><a href="#topic+fortify_matrix.list_upset">fortify_matrix.list_upset</a></code> and <code><a href="#topic+fortify_matrix.matrix_upset">fortify_matrix.matrix_upset</a></code> are
suitable for creating an UpSet plot.</p>
</td></tr>
<tr><td><code id="ggupset_+3A_mapping">mapping</code></td>
<td>
<p>Default list of aesthetic mappings to use for main plot in the
layout. If not specified, must be supplied in each layer added to the main
plot.</p>
</td></tr>
<tr><td><code id="ggupset_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>.</p>
</td></tr>
<tr><td><code id="ggupset_+3A_direction">direction</code></td>
<td>
<p>A string indicating the direction of the UpSet plot,
<code>"h"</code>(<code>horizontal</code>) or <code>"v"</code>(<code>vertical</code>). In a vertical UpSet plot, the
columns of the matrix correspond to the sets, and the rows correspond to the
intersections. By default, the horizontal UpSet plot is used, where the rows
of the matrix correspond to the sets and the columns correspond to the
intersections.</p>
</td></tr>
<tr><td><code id="ggupset_+3A_point">point</code></td>
<td>
<p>A list of parameters passed to
<code><a href="ggplot2.html#topic+geom_point">geom_point()</a></code>.</p>
</td></tr>
<tr><td><code id="ggupset_+3A_line">line</code></td>
<td>
<p>A list of parameters passed to
<code><a href="ggplot2.html#topic+geom_path">geom_line()</a></code>.</p>
</td></tr>
<tr><td><code id="ggupset_+3A_rect">rect</code></td>
<td>
<p>A list of parameters passed to
<code><a href="ggplot2.html#topic+geom_tile">geom_rect()</a></code>.</p>
</td></tr>
<tr><td><code id="ggupset_+3A_width">width</code>, <code id="ggupset_+3A_height">height</code></td>
<td>
<p>The relative width/height of the main plot, can be a
<code><a href="grid.html#topic+unit">unit</a></code> object.</p>
</td></tr>
<tr><td><code id="ggupset_+3A_theme">theme</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+theme">theme()</a></code> object used to customize various
elements of the layout, including <code>guides</code>, <code>title</code>, <code>subtitle</code>, <code>caption</code>,
<code>margins</code>, <code>panel.border</code>, and <code>background</code>. By default, the theme will
inherit from the parent <code>layout</code>. It also controls the panel spacing for all
plots in the layout.</p>
</td></tr>
<tr><td><code id="ggupset_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>ggplot2 specification</h3>

<p>The data input will be converted to a matrix using <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>, and
the data in the underlying main plot will contain the following columns:
</p>

<ul>
<li> <p><code>.panel_x</code> and <code>.panel_y</code>: the column and row panel groups.
</p>
</li>
<li> <p><code>.x</code> and <code>.y</code>: an integer index of <code>x</code> and <code>y</code> coordinates
</p>
</li>
<li> <p><code>.discrete_x</code> and <code>.discrete_y</code>: a factor of the data labels (only
applicable when <code>.row_names</code> and <code>.column_names</code> exists).
</p>
</li>
<li> <p><code>.row_names</code> and <code>.column_names</code>: A character of the row and column names
of the original matrix (only applicable when names exist).
</p>
</li>
<li> <p><code>.row_index</code> and <code>.column_index</code>: the row and column index of the original
matrix.
</p>
</li>
<li> <p><code>value</code>: the actual matrix value.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
lt &lt;- list(
    a = sample(letters, 5),
    b = sample(letters, 10),
    c = sample(letters, 15)
)
ggupset(tune(lt)) +
    scale_fill_manual(values = c("#F0F0F0", "white"), guide = "none") +
    scale_color_manual(values = c("grey", "black"), guide = "none") +
    anno_top() +
    ggalign(data = function(d) ggalign_attr(d, "intersection_sizes")) +
    ggplot2::geom_bar(aes(y = .data$value), stat = "identity") +
    anno_right() +
    ggalign(data = function(d) ggalign_attr(d, "set_sizes")) +
    ggplot2::geom_bar(aes(x = .data$value),
        stat = "identity",
        orientation = "y"
    )
</code></pre>

<hr>
<h2 id='ggwrap'>Wrap Arbitrary Graphics to ggplot</h2><span id='topic+ggwrap'></span>

<h3>Description</h3>

<p>The <code>ggwrap()</code> function allows non-ggplot2 elements to be converted into a
compliant representation for use with <code><a href="#topic+align_plots">align_plots()</a></code>. This is useful for
adding any graphics that can be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code> with
the <code><a href="#topic+patch">patch()</a></code> method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggwrap(plot, ..., align = "panel", on_top = FALSE, clip = TRUE, vp = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggwrap_+3A_plot">plot</code></td>
<td>
<p>Any graphic that can be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>
using <code><a href="#topic+patch">patch()</a></code>.</p>
</td></tr>
<tr><td><code id="ggwrap_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code><a href="#topic+patch">patch()</a></code> method.</p>
</td></tr>
<tr><td><code id="ggwrap_+3A_align">align</code></td>
<td>
<p>A string specifying the area to place the plot: <code>"full"</code> for the
full area, <code>"plot"</code> for the full plotting area (including the axis label), or
<code>"panel"</code> for only the actual area where data is drawn.</p>
</td></tr>
<tr><td><code id="ggwrap_+3A_on_top">on_top</code></td>
<td>
<p>A single boolean value indicates whether the graphic plot
should be put frontmost. Note: the graphic plot will always put above the
background.</p>
</td></tr>
<tr><td><code id="ggwrap_+3A_clip">clip</code></td>
<td>
<p>A single boolean value indicating whether the grob should be
clipped if they expand outside their designated area.</p>
</td></tr>
<tr><td><code id="ggwrap_+3A_vp">vp</code></td>
<td>
<p>A <code><a href="grid.html#topic+viewport">viewport</a></code> object, you can use this to define
the plot area.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>wrapped_plot</code> object that can be directly placed into
<code><a href="#topic+align_plots">align_plots()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(grid)
ggwrap(rectGrob(gp = gpar(fill = "goldenrod")), align = "full") +
    inset(rectGrob(gp = gpar(fill = "steelblue")), align = "panel") +
    inset(textGrob("Here are some text", gp = gpar(color = "black")),
        align = "panel"
    )
p1 &lt;- ggplot(mtcars) +
    geom_point(aes(mpg, disp)) +
    ggtitle("Plot 1")
align_plots(p1, ggwrap(
    ~ plot(mtcars$mpg, mtcars$disp),
    mar = c(0, 2, 0, 0), bg = NA
))

</code></pre>

<hr>
<h2 id='hclust2'>Generate Tree Structures with Hierarchical Clustering</h2><span id='topic+hclust2'></span>

<h3>Description</h3>

<p>Generate Tree Structures with Hierarchical Clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hclust2(
  matrix,
  distance = "euclidean",
  method = "complete",
  use_missing = "pairwise.complete.obs"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hclust2_+3A_matrix">matrix</code></td>
<td>
<p>A numeric matrix, or data frame.</p>
</td></tr>
<tr><td><code id="hclust2_+3A_distance">distance</code></td>
<td>
<p>A string of distance measure to be used. This must be one of
<code>"euclidean"</code>, <code>"maximum"</code>, <code>"manhattan"</code>, <code>"canberra"</code>, <code>"binary"</code> or
<code>"minkowski"</code>.  Correlation coefficient can be also used, including
<code>"pearson"</code>, <code>"spearman"</code> or <code>"kendall"</code>. In this way, <code>1 - cor</code> will be used
as the distance. In addition, you can also provide a <code><a href="stats.html#topic+dist">dist</a></code>
object directly or a function return a <code><a href="stats.html#topic+dist">dist</a></code> object. Use
<code>NULL</code>, if you don't want to calculate the distance.</p>
</td></tr>
<tr><td><code id="hclust2_+3A_method">method</code></td>
<td>
<p>A string of the agglomeration method to be used. This should be
(an unambiguous abbreviation of) one of <code>"ward.D"</code>, <code>"ward.D2"</code>, <code>"single"</code>,
<code>"complete"</code>, <code>"average"</code> (= UPGMA), <code>"mcquitty"</code> (= WPGMA), <code>"median"</code> (=
WPGMC) or <code>"centroid"</code> (= UPGMC). You can also provide a function which
accepts the calculated distance (or the input matrix if <code>distance</code> is <code>NULL</code>)
and returns a <code><a href="stats.html#topic+hclust">hclust</a></code> object. Alternative, you can supply
an object which can be coerced to <code><a href="stats.html#topic+hclust">hclust</a></code>.</p>
</td></tr>
<tr><td><code id="hclust2_+3A_use_missing">use_missing</code></td>
<td>
<p>An optional character string giving a method for computing
covariances in the presence of missing values. This must be (an abbreviation
of) one of the strings <code>"everything"</code>, <code>"all.obs"</code>, <code>"complete.obs"</code>,
<code>"na.or.complete"</code>, or <code>"pairwise.complete.obs"</code>. Only used when <code>distance</code>
is a correlation coefficient string.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="stats.html#topic+hclust">hclust</a> object.
</p>


<h3>See Also</h3>


<ul>
<li> <p><a href="stats.html#topic+cor">cor()</a>
</p>
</li>
<li> <p><a href="stats.html#topic+dist">dist()</a>
</p>
</li>
<li> <p><a href="stats.html#topic+hclust">hclust()</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>hclust2(dist(USArrests), method = "ward.D")
</code></pre>

<hr>
<h2 id='heatmap_layout'>Create a heatmap</h2><span id='topic+heatmap_layout'></span><span id='topic+ggheatmap'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p><code>heatmap_layout</code> is a specialized version of <code><a href="#topic+quad_discrete">quad_discrete()</a></code>, which
simplifies the creation of heatmap plots by integrating essential elements
for a standard heatmap layout, ensuring that the appropriate data mapping and
visualization layers are automatically applied. <code>ggheatmap</code> is an alias for
<code>heatmap_layout</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_layout(
  data = NULL,
  mapping = aes(),
  ...,
  width = NA,
  height = NA,
  filling = waiver(),
  theme = NULL,
  active = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="heatmap_layout_+3A_data">data</code></td>
<td>
<p>Default dataset to use for the layout. If not specified, it must be supplied in each plot added to the layout. By default, it will try to inherit from
parent layout. <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> will be used to convert data to a
matrix.</p>
</td></tr>
<tr><td><code id="heatmap_layout_+3A_mapping">mapping</code></td>
<td>
<p>Default list of aesthetic mappings to use for main plot in the
layout. If not specified, must be supplied in each layer added to the main
plot.</p>
</td></tr>
<tr><td><code id="heatmap_layout_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>.</p>
</td></tr>
<tr><td><code id="heatmap_layout_+3A_width">width</code>, <code id="heatmap_layout_+3A_height">height</code></td>
<td>
<p>The relative width/height of the main plot, can be a
<code><a href="grid.html#topic+unit">unit</a></code> object.</p>
</td></tr>
<tr><td><code id="heatmap_layout_+3A_filling">filling</code></td>
<td>
<p>A single string of <code>"raster"</code> or <code>"tile"</code> to
indicate the filling style. By default, <code>waiver()</code> is used, which means that
if the input matrix has more than 20,000 cells (<code>nrow * ncol &gt; 20000</code>),
<code><a href="ggplot2.html#topic+geom_tile">geom_raster()</a></code> will be used for performance
efficiency; for smaller matrices, <code><a href="ggplot2.html#topic+geom_tile">geom_tile()</a></code> will be
used. To customize the filling style, set this to <code>NULL</code>.
</p>
<p>For backward compatibility, a single boolean value is acceptable: <code>TRUE</code>
means <code>waiver()</code>, and <code>FALSE</code> means <code>NULL</code>.
</p>
<p>By default, the classic heatmap color scheme
<code><a href="ggplot2.html#topic+scale_gradient">scale_fill_gradient2(low = &quot;blue&quot;, high = &quot;red&quot;)</a></code>
is utilized for continuous values.
</p>
<p>You can use the options
<code>"ggalign.heatmap_continuous_fill"</code> or
<code>"ggalign.heatmap_discrete_fill"</code> to modify the
default heatmap body filling color scale. See
<code><a href="ggplot2.html#topic+scale_colour_continuous">scale_fill_continuous()</a></code> or
<code><a href="ggplot2.html#topic+scale_colour_discrete">scale_fill_discrete()</a></code> for details on
option settings.</p>
</td></tr>
<tr><td><code id="heatmap_layout_+3A_theme">theme</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+theme">theme()</a></code> object used to customize various
elements of the layout, including <code>guides</code>, <code>title</code>, <code>subtitle</code>, <code>caption</code>,
<code>margins</code>, <code>panel.border</code>, and <code>background</code>. By default, the theme will
inherit from the parent <code>layout</code>. It also controls the panel spacing for all
plots in the layout.</p>
</td></tr>
<tr><td><code id="heatmap_layout_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>HeatmapLayout</code> object.
</p>


<h3>ggplot2 specification</h3>

<p>The data input will be converted to a matrix using <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>, and
the data in the underlying main plot will contain the following columns:
</p>

<ul>
<li> <p><code>.panel_x</code> and <code>.panel_y</code>: the column and row panel groups.
</p>
</li>
<li> <p><code>.x</code> and <code>.y</code>: an integer index of <code>x</code> and <code>y</code> coordinates
</p>
</li>
<li> <p><code>.discrete_x</code> and <code>.discrete_y</code>: a factor of the data labels (only
applicable when <code>.row_names</code> and <code>.column_names</code> exists).
</p>
</li>
<li> <p><code>.row_names</code> and <code>.column_names</code>: A character of the row and column names
of the original matrix (only applicable when names exist).
</p>
</li>
<li> <p><code>.row_index</code> and <code>.column_index</code>: the row and column index of the original
matrix.
</p>
</li>
<li> <p><code>value</code>: the actual matrix value.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>ggheatmap(1:10)
ggheatmap(letters)
ggheatmap(matrix(rnorm(81), nrow = 9L))
</code></pre>

<hr>
<h2 id='inset'>Create a ggplot inset</h2><span id='topic+inset'></span>

<h3>Description</h3>

<p>Create a ggplot inset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inset(plot, ..., align = "panel", on_top = TRUE, clip = TRUE, vp = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inset_+3A_plot">plot</code></td>
<td>
<p>Any graphic that can be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>
using <code><a href="#topic+patch">patch()</a></code>.</p>
</td></tr>
<tr><td><code id="inset_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the <code><a href="#topic+patch">patch()</a></code> method.</p>
</td></tr>
<tr><td><code id="inset_+3A_align">align</code></td>
<td>
<p>A string specifying the area to place the plot: <code>"full"</code> for the
full area, <code>"plot"</code> for the full plotting area (including the axis label), or
<code>"panel"</code> for only the actual area where data is drawn.</p>
</td></tr>
<tr><td><code id="inset_+3A_on_top">on_top</code></td>
<td>
<p>A single boolean value indicates whether the graphic plot
should be put frontmost. Note: the graphic plot will always put above the
background.</p>
</td></tr>
<tr><td><code id="inset_+3A_clip">clip</code></td>
<td>
<p>A single boolean value indicating whether the grob should be
clipped if they expand outside their designated area.</p>
</td></tr>
<tr><td><code id="inset_+3A_vp">vp</code></td>
<td>
<p>A <code><a href="grid.html#topic+viewport">viewport</a></code> object, you can use this to define
the plot area.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>patch_inset</code> object, which can be added in ggplot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(grid)
p1 &lt;- ggplot(mtcars) +
    geom_point(aes(mpg, disp))
p2 &lt;- ggplot(mtcars) +
    geom_boxplot(aes(gear, disp, group = gear))
p1 + inset(p2, vp = viewport(0.6, 0.6,
    just = c(0, 0), width = 0.4, height = 0.4
))
</code></pre>

<hr>
<h2 id='is_layout'>Reports whether <code>x</code> is layout object</h2><span id='topic+is_layout'></span><span id='topic+is_quad_layout'></span><span id='topic+is_stack_layout'></span><span id='topic+is_stack_cross'></span><span id='topic+is_circle_layout'></span><span id='topic+is_heatmap_layout'></span><span id='topic+is_ggheatmap'></span>

<h3>Description</h3>

<p>Reports whether <code>x</code> is layout object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_layout(x)

is_quad_layout(x)

is_stack_layout(x)

is_stack_cross(x)

is_circle_layout(x)

is_heatmap_layout(x)

is_ggheatmap(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_layout_+3A_x">x</code></td>
<td>
<p>An object to test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single boolean value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_layout(ggheatmap(1:10))

# for quad_layout()
is_quad_layout(quad_alignb(1:10))
is_quad_layout(quad_alignh(1:10))
is_quad_layout(quad_alignv(1:10))
is_quad_layout(quad_free(mtcars))

# for stack_layout()
is_stack_layout(stack_discrete("h", 1:10))
is_stack_layout(stack_continuous("h", 1:10))

# for heatmap_layout()
is_heatmap_layout(ggheatmap(1:10))
is_ggheatmap(ggheatmap(1:10))
</code></pre>

<hr>
<h2 id='layer_order'>Change the layer adding order</h2><span id='topic+layer_order'></span>

<h3>Description</h3>

<p>This function allows you to change the order in which layers are added to a
ggplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>layer_order(layer, order = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="layer_order_+3A_layer">layer</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+layer_geoms">layer geometry</a></code> object to be added.</p>
</td></tr>
<tr><td><code id="layer_order_+3A_order">order</code></td>
<td>
<p>An integer indicating the position at which the layer should be
added. If <code style="white-space: pre;">&#8288;&lt;= 0&#8288;</code>, the layer will be added at the beginning. If greater than
the number of plot layers, it will be added at the end.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>layer_order</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ggplot(faithfuld, aes(waiting, eruptions)) +
    geom_raster(aes(fill = density)) +
    geom_point(color = "red", size = 1)
ggplot(faithfuld, aes(waiting, eruptions)) +
    geom_raster(aes(fill = density)) +
    layer_order(geom_point(color = "red", size = 1))
</code></pre>

<hr>
<h2 id='layout_annotation'>Modify components of the layout</h2><span id='topic+layout_annotation'></span>

<h3>Description</h3>


<ul>
<li><p> modify the theme of the layout
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>layout_annotation(theme = waiver(), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="layout_annotation_+3A_theme">theme</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+theme">theme()</a></code> object used to customize various
elements of the plot, including <code>guides</code>, <code>title</code>, <code>subtitle</code>, <code>caption</code>,
<code>margins</code>, <code>panel.border</code>, and <code>background</code>. By default, the
theme will inherit from the parent <code>layout</code>.</p>
</td></tr>
<tr><td><code id="layout_annotation_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li> <p><code>guides</code>, <code>panel.border</code>, and <code>background</code> will always be used even for the
nested <code>alignpatches</code> object.
</p>
</li>
<li> <p><code>title</code>, <code>subtitle</code>, <code>caption</code>, and <code>margins</code> will be added for the
top-level <code>alignpatches</code> object only.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>p1 &lt;- ggplot(mtcars) +
    geom_point(aes(mpg, disp))
p2 &lt;- ggplot(mtcars) +
    geom_boxplot(aes(gear, disp, group = gear))
p3 &lt;- ggplot(mtcars) +
    geom_bar(aes(gear)) +
    facet_wrap(~cyl)
align_plots(
    p1 + theme(plot.background = element_blank()),
    p2 + theme(plot.background = element_blank()),
    p3 + theme(plot.background = element_blank())
) +
    layout_annotation(
        theme = theme(plot.background = element_rect(fill = "red"))
    )
</code></pre>

<hr>
<h2 id='layout_design'>Define the grid to compose plots in</h2><span id='topic+layout_design'></span>

<h3>Description</h3>

<p>To control how different plots are laid out, you need to add a layout design
specification. If you are nesting grids, the layout is scoped to the current
nesting level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>layout_design(
  ncol = waiver(),
  nrow = waiver(),
  byrow = waiver(),
  widths = waiver(),
  heights = waiver(),
  design = waiver(),
  guides = NA
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="layout_design_+3A_ncol">ncol</code>, <code id="layout_design_+3A_nrow">nrow</code></td>
<td>
<p>The dimensions of the grid to create - if both are <code>NULL</code> it
will use the same logic as <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap()</a></code> to set the
dimensions</p>
</td></tr>
<tr><td><code id="layout_design_+3A_byrow">byrow</code></td>
<td>
<p>If <code>FALSE</code> the plots will be filled in in column-major order.</p>
</td></tr>
<tr><td><code id="layout_design_+3A_widths">widths</code>, <code id="layout_design_+3A_heights">heights</code></td>
<td>
<p>The relative widths and heights of each column and row
in the grid. Will get repeated to match the dimensions of the grid. The
special value of <code>NA</code> will behave as <code style="white-space: pre;">&#8288;1null&#8288;</code> unit unless a fixed aspect plot
is inserted in which case it will allow the dimension to expand or contract
to match the aspect ratio of the content.</p>
</td></tr>
<tr><td><code id="layout_design_+3A_design">design</code></td>
<td>
<p>Specification of the location of areas in the layout. Can
either be specified as a text string or by concatenating calls to
<code><a href="#topic+area">area()</a></code> together.</p>
</td></tr>
<tr><td><code id="layout_design_+3A_guides">guides</code></td>
<td>
<p>A string with one or more of <code>"t"</code>, <code>"l"</code>, <code>"b"</code>, and <code>"r"</code> indicating
which side of guide legends should be collected. Defaults to
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code>, which inherits from the parent layout. If
there is no parent layout, or if <code>NULL</code> is provided, no guides will be
collected.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>layout_design</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p1 &lt;- ggplot(mtcars) +
    geom_point(aes(mpg, disp))
p2 &lt;- ggplot(mtcars) +
    geom_boxplot(aes(gear, disp, group = gear))
p3 &lt;- ggplot(mtcars) +
    geom_bar(aes(gear)) +
    facet_wrap(~cyl)
align_plots(p1, p2, p3) +
    layout_design(nrow = 1L)
align_plots(p1, p2, p3) +
    layout_design(ncol = 1L)
</code></pre>

<hr>
<h2 id='layout_expand'>Set Expansion for the Layout</h2><span id='topic+layout_expand'></span>

<h3>Description</h3>

<p>To align axes, it is important to keep the expansion consistent across all
plots in the layout. You can add a <code>layout_expand</code> object to the layout. For
the <code>quad_layout()</code> function, you must specify <code>x</code> and <code>y</code> arguments. For
other layouts, you can pass the expansion values using <code>...</code> directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>layout_expand(..., x = waiver(), y = waiver())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="layout_expand_+3A_...">...</code></td>
<td>
<p>A list of range expansion constants, used to add padding around
the data to ensure they are placed some distance away from the axes. Use the
convenience function <code><a href="ggplot2.html#topic+expansion">expansion()</a></code> to generate the
values.</p>
</td></tr>
<tr><td><code id="layout_expand_+3A_x">x</code>, <code id="layout_expand_+3A_y">y</code></td>
<td>
<p>Same as <code>...</code>, but specifically for <code>quad_layout()</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='layout_title'>Annotate the whole layout</h2><span id='topic+layout_title'></span>

<h3>Description</h3>

<p>Annotate the whole layout
</p>


<h3>Usage</h3>

<pre><code class='language-R'>layout_title(title = waiver(), subtitle = waiver(), caption = waiver())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="layout_title_+3A_title">title</code></td>
<td>
<p>The text for the title.</p>
</td></tr>
<tr><td><code id="layout_title_+3A_subtitle">subtitle</code></td>
<td>
<p>The text for the subtitle for the plot which will be
displayed below the title.</p>
</td></tr>
<tr><td><code id="layout_title_+3A_caption">caption</code></td>
<td>
<p>The text for the caption which will be displayed in the
bottom-right of the plot by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>layout_title</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p1 &lt;- ggplot(mtcars) +
    geom_point(aes(mpg, disp))
p2 &lt;- ggplot(mtcars) +
    geom_boxplot(aes(gear, disp, group = gear))
p3 &lt;- ggplot(mtcars) +
    geom_bar(aes(gear)) +
    facet_wrap(~cyl)
align_plots(p1, p2, p3) +
    layout_title(title = "I'm title")
</code></pre>

<hr>
<h2 id='layout-operator'>Layout operator</h2><span id='topic+layout-operator'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>

<ul>
<li> <p><code>+</code>: Adds elements to the active plot in the active layout.
</p>
</li>
<li> <p><code>&amp;</code>: Applies elements to all plots in the layout.
</p>
</li>
<li> <p><code>-</code>: Adds elements to multiple plots in the layout.
</p>
</li></ul>



<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="layout-operator_+3A_e1">e1</code></td>
<td>
<p>A <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code> or <code><a href="#topic+stack_layout">stack_layout()</a></code> object.</p>
</td></tr>
<tr><td><code id="layout-operator_+3A_e2">e2</code></td>
<td>
<p>An object to be added to the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>+</code> operator is straightforward and should be used as needed.
</p>
<p>In order to reduce code repetition <code>ggalign</code> provides two operators for
adding ggplot elements (geoms, themes, facets, etc.) to multiple/all plots in
<code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code> or <code><a href="#topic+stack_layout">stack_layout()</a></code> object: <code>-</code> and <code>&amp;</code>.
</p>


<h3>Value</h3>

<p>A modified <code>Layout</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
small_mat &lt;- matrix(rnorm(56), nrow = 7)
ggheatmap(small_mat) +
    anno_top() +
    ggalign() +
    geom_point(aes(y = value))

# `&amp;` operator apply it to all plots
ggheatmap(small_mat) +
    anno_top() +
    align_dendro() &amp;
    theme(panel.border = element_rect(
        colour = "red", fill = NA, linewidth = unit(2, "mm")
    ))

# If the active layout is the annotation stack, the `-` operator will only
# add the elements to all plots in the active annotation stack:
ggheatmap(small_mat) +
    anno_left(size = 0.2) +
    align_dendro(aes(color = branch), k = 3L) +
    align_dendro(aes(color = branch), k = 3L) -
    # Modify the the color scales of all plots in the left annotation
    scale_color_brewer(palette = "Dark2")

# If the active layout is the `stack_layout()` itself, `-`
# applies the elements to all plots in the layout except the nested
# `ggheatmap()`/`quad_layout()`.
stack_alignv(small_mat) +
    align_dendro() +
    ggtitle("I'm from the parent stack") +
    ggheatmap() +
    # remove any active context
    stack_active() +
    align_dendro() +
    ggtitle("I'm from the parent stack") -
    # Modify the the color scales of all plots in the stack layout except the
    # heatmap layout
    scale_color_brewer(palette = "Dark2") -
    # set the background of all plots in the stack layout except the heatmap
    # layout
    theme(plot.background = element_rect(fill = "red"))

</code></pre>

<hr>
<h2 id='LayoutProto-class'>A <code>Layout</code> object</h2><span id='topic+LayoutProto-class'></span>

<h3>Description</h3>

<p>A <code>Layout</code> object defines how to place the plots.
</p>

<hr>
<h2 id='link_draw'>Define the links to connect a pair of observations</h2><span id='topic+link_draw'></span>

<h3>Description</h3>

<p>This function allows users to define links between a pair of observations,
facilitating the visualization of connections between related data points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>link_draw(.draw, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="link_draw_+3A_.draw">.draw</code></td>
<td>
<p>A function used to draw the links. The function must return a
<code><a href="grid.html#topic+grid.grob">grob()</a></code> object. If the function does not return a valid
<code>grob</code>, no drawing will occur. The input data for the function should
include a data frame with the coordinates of the pair of observations to
be linked.</p>
</td></tr>
<tr><td><code id="link_draw_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; A list of formulas, where each side
of the formula should be an <code>integer</code> or <code>character</code> index of the original
data, or a <code>range_link()</code> object defining the linked observations. Use
<code>NULL</code> to indicate no link on that side. You can also combine these by
wrapping them into a single <code>list()</code>. If only the left-hand side of the
formula exists, you can input it directly. For integer indices, wrap them
with <code><a href="base.html#topic+I">I()</a></code> to use the ordering from the layout. You can also use
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code> to inherit values from the other group.</p>
</td></tr>
</table>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+link_line">link_line()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+.link_draw">.link_draw()</a></code>
</p>
</li></ul>


<hr>
<h2 id='link_line'>Link the paired observations with a line</h2><span id='topic+link_line'></span>

<h3>Description</h3>

<p>Link the paired observations with a line
</p>


<h3>Usage</h3>

<pre><code class='language-R'>link_line(..., .element = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="link_line_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; A list of formulas, where each side
of the formula should be an <code>integer</code> or <code>character</code> index of the original
data, or a <code>range_link()</code> object defining the linked observations. Use
<code>NULL</code> to indicate no link on that side. You can also combine these by
wrapping them into a single <code>list()</code>. If only the left-hand side of the
formula exists, you can input it directly. For integer indices, wrap them
with <code><a href="base.html#topic+I">I()</a></code> to use the ordering from the layout. You can also use
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code> to inherit values from the other group.</p>
</td></tr>
<tr><td><code id="link_line_+3A_.element">.element</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+element">element_line()</a></code> object.
Vectorized fields will be recycled to match the total number of groups, or
you can wrap the element with <code><a href="base.html#topic+I">I()</a></code> to recycle to match the drawing
groups. The drawing groups typically correspond to the product of the
number of observations from both sides, as each pair of observations will
be linked with a single line.</p>
</td></tr>
</table>

<hr>
<h2 id='link_tetragon'>Link the paired observations with a quadrilateral</h2><span id='topic+link_tetragon'></span>

<h3>Description</h3>

<p>Link the paired observations with a quadrilateral
</p>


<h3>Usage</h3>

<pre><code class='language-R'>link_tetragon(..., .element = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="link_tetragon_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; A list of formulas, where each side
of the formula should be an <code>integer</code> or <code>character</code> index of the original
data, or a <code>range_link()</code> object defining the linked observations. Use
<code>NULL</code> to indicate no link on that side. You can also combine these by
wrapping them into a single <code>list()</code>. If only the left-hand side of the
formula exists, you can input it directly. For integer indices, wrap them
with <code><a href="base.html#topic+I">I()</a></code> to use the ordering from the layout. You can also use
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code> to inherit values from the other group.</p>
</td></tr>
<tr><td><code id="link_tetragon_+3A_.element">.element</code></td>
<td>
<p>A <code><a href="#topic+element_polygon">element_polygon()</a></code> object. Vectorized fields will be
recycled to match the total number of groups, or you can wrap the element
with <code><a href="base.html#topic+I">I()</a></code> to recycle to match the drawing groups. The drawing groups
are usually the same as the defined groups, but they will differ when the
defined group of observations is separated and cannot be linked with a
single quadrilateral. In such cases, the number of drawing groups will be
larger than the number of defined groups.</p>
</td></tr>
</table>

<hr>
<h2 id='mark_draw'>Define the links to connect the marked observations</h2><span id='topic+mark_draw'></span>

<h3>Description</h3>

<p>This function allows users to define links between marked observations and
plot panel (e.g., for creating visual connections for related data), which
could help explain the observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mark_draw(.draw, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mark_draw_+3A_.draw">.draw</code></td>
<td>
<p>A function used to draw the links. The function must return a
<code><a href="grid.html#topic+grid.grob">grob()</a></code> object. If the function does not return a valid
<code>grob</code>, nothing will be drawn. The input data for the function must contain
two arguments: a data frame for the panel side coordinates and a data frame
for the marked observation coordinates.</p>
</td></tr>
<tr><td><code id="mark_draw_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; A list of formulas, where each side
of the formula should be an <code>integer</code> or <code>character</code> index of the original
data, or a <code>range_link()</code> object defining the linked observations. Use
<code>NULL</code> to indicate no link on that side. You can also combine these by
wrapping them into a single <code>list()</code>. If only the left-hand side of the
formula exists, you can input it directly. For integer indices, wrap them
with <code><a href="base.html#topic+I">I()</a></code> to use the ordering from the layout. You can also use
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code> to inherit values from the other group.</p>
</td></tr>
</table>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+mark_line">mark_line()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+mark_tetragon">mark_tetragon()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+mark_triangle">mark_triangle()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+.mark_draw">.mark_draw()</a></code>
</p>
</li></ul>


<hr>
<h2 id='mark_line'>Link the observations and the panel with a line</h2><span id='topic+mark_line'></span>

<h3>Description</h3>

<p>Link the observations and the panel with a line
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mark_line(..., .element = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mark_line_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; A list of formulas, where each side
of the formula should be an <code>integer</code> or <code>character</code> index of the original
data, or a <code>range_link()</code> object defining the linked observations. Use
<code>NULL</code> to indicate no link on that side. You can also combine these by
wrapping them into a single <code>list()</code>. If only the left-hand side of the
formula exists, you can input it directly. For integer indices, wrap them
with <code><a href="base.html#topic+I">I()</a></code> to use the ordering from the layout. You can also use
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code> to inherit values from the other group.</p>
</td></tr>
<tr><td><code id="mark_line_+3A_.element">.element</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+element">element_line()</a></code> object.
Vectorized fields will be recycled to match the total number of groups, or
you can wrap the element with <code><a href="base.html#topic+I">I()</a></code> to recycle to match the drawing
groups. The drawing groups typically correspond to the number of
observations, as each observation will be linked with the plot panel.</p>
</td></tr>
</table>

<hr>
<h2 id='mark_tetragon'>Link the observations and the panel with a quadrilateral</h2><span id='topic+mark_tetragon'></span>

<h3>Description</h3>

<p>Link the observations and the panel with a quadrilateral
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mark_tetragon(..., .element = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mark_tetragon_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; A list of formulas, where each side
of the formula should be an <code>integer</code> or <code>character</code> index of the original
data, or a <code>range_link()</code> object defining the linked observations. Use
<code>NULL</code> to indicate no link on that side. You can also combine these by
wrapping them into a single <code>list()</code>. If only the left-hand side of the
formula exists, you can input it directly. For integer indices, wrap them
with <code><a href="base.html#topic+I">I()</a></code> to use the ordering from the layout. You can also use
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code> to inherit values from the other group.</p>
</td></tr>
<tr><td><code id="mark_tetragon_+3A_.element">.element</code></td>
<td>
<p>A <code><a href="#topic+element_polygon">element_polygon()</a></code> object. Vectorized fields will be
recycled to match the total number of groups, or you can wrap the element
with <code><a href="base.html#topic+I">I()</a></code> to recycle to match the drawing groups. The drawing groups
are usually the same as the defined groups, but they will differ when the
defined group of observations is separated and cannot be linked with a
single quadrilateral. In such cases, the number of drawing groups will be
larger than the number of defined groups.</p>
</td></tr>
</table>

<hr>
<h2 id='mark_triangle'>Link the observations and the panel with a triangle</h2><span id='topic+mark_triangle'></span>

<h3>Description</h3>

<p>Link the observations and the panel with a triangle
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mark_triangle(..., orientation = "plot", .element = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mark_triangle_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; A list of formulas, where each side
of the formula should be an <code>integer</code> or <code>character</code> index of the original
data, or a <code>range_link()</code> object defining the linked observations. Use
<code>NULL</code> to indicate no link on that side. You can also combine these by
wrapping them into a single <code>list()</code>. If only the left-hand side of the
formula exists, you can input it directly. For integer indices, wrap them
with <code><a href="base.html#topic+I">I()</a></code> to use the ordering from the layout. You can also use
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code> to inherit values from the other group.</p>
</td></tr>
<tr><td><code id="mark_triangle_+3A_orientation">orientation</code></td>
<td>
<p>A single string, either <code>"plot"</code> or <code>"observation"</code>,
indicating the base of the triangle.</p>
</td></tr>
<tr><td><code id="mark_triangle_+3A_.element">.element</code></td>
<td>
<p>An <code><a href="#topic+element_polygon">element_polygon()</a></code> object. Vectorized fields will be
recycled to match the total number of groups, or you can wrap the element
with <code><a href="base.html#topic+I">I()</a></code> to recycle to match the drawing groups.
</p>

<ul>
<li><p> When <code>orientation</code> is <code>"plot"</code>, the drawing groups typically correspond
to the number of observations.
</p>
</li>
<li><p> When <code>orientation</code> is <code>"observation"</code>, the drawing groups usually match
the defined groups, but will differ if the defined group of observations
is separated and cannot be linked with a single triangle. In this case,
the number of drawing groups will be larger than the number of defined
groups.
</p>
</li></ul>
</td></tr>
</table>

<hr>
<h2 id='memo_order'>Sort matrix for better visualization</h2><span id='topic+memo_order'></span>

<h3>Description</h3>

<p>Helper function used to order the Oncoplot samples. Typically, you would use
this in combination with <code><a href="#topic+align_order2">align_order2()</a></code>, e.g.,
<code>align_order2(memo_order)</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>memo_order(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="memo_order_+3A_x">x</code></td>
<td>
<p>A matrix, where <code>NA</code> values will be treated as empty.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of ordering weights.
</p>

<hr>
<h2 id='new_tune'>Change the shape of the input object</h2><span id='topic+new_tune'></span><span id='topic+tune_data'></span>

<h3>Description</h3>


<ul>
<li> <p><code>new_tune</code>: Creates a new object by wrapping it in a scalar list with
the specified attributes and class.
</p>
</li>
<li> <p><code>tune_data</code>: Retrieves the original input data.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>new_tune(x, ..., class = character())

tune_data(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="new_tune_+3A_x">x</code></td>
<td>
<p>An R object.</p>
</td></tr>
<tr><td><code id="new_tune_+3A_...">...</code></td>
<td>
<p>Additional attributes passed to <code><a href="base.html#topic+structure">structure()</a></code>.</p>
</td></tr>
<tr><td><code id="new_tune_+3A_class">class</code></td>
<td>
<p>A character vector specifying the class name to be added.</p>
</td></tr>
</table>

<hr>
<h2 id='no_expansion'>Remove scale expansion</h2><span id='topic+no_expansion'></span>

<h3>Description</h3>

<p>Remove scale expansion
</p>


<h3>Usage</h3>

<pre><code class='language-R'>no_expansion(borders = "tlbr")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="no_expansion_+3A_borders">borders</code></td>
<td>
<p>Which border should be removed? A string containing
one or more of <code>"t"</code>, <code>"l"</code>, <code>"b"</code>, <code>"r"</code>, <code>"x"</code>, and <code>"y"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object which can be added to ggplot.
</p>

<hr>
<h2 id='order2'>Ordering Permutation</h2><span id='topic+order2'></span><span id='topic+order2.hclust'></span><span id='topic+order2.dendrogram'></span><span id='topic+order2.ser_permutation_vector'></span><span id='topic+order2.ser_permutation'></span><span id='topic+order2.phylo'></span><span id='topic+order2.memo_weights'></span>

<h3>Description</h3>

<p><code>order2</code> returns a permutation which rearranges its first argument into
ascending order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>order2(x)

## S3 method for class 'hclust'
order2(x)

## S3 method for class 'dendrogram'
order2(x)

## S3 method for class 'ser_permutation_vector'
order2(x)

## S3 method for class 'ser_permutation'
order2(x)

## S3 method for class 'phylo'
order2(x)

## S3 method for class 'memo_weights'
order2(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="order2_+3A_x">x</code></td>
<td>
<p>Any objects can be extracting ordering.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer vector unless any of the inputs has <code>2^31</code> or more
elements, when it is a double vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>order2(hclust2(matrix(rnorm(100L), nrow = 10L)))
</code></pre>

<hr>
<h2 id='pair_links'>Helper function to create pairs of observation groups</h2><span id='topic+pair_links'></span><span id='topic+range_link'></span>

<h3>Description</h3>

<p><code><a href="#topic+ggmark">ggmark()</a></code> and <code><a href="#topic+cross_link">cross_link()</a></code> allow users to add links between
observations. These functions help define the linked observations. The
selected pairs will either be linked together, or each group in the pair will
be linked separately to the same plot area.
</p>

<ul>
<li> <p><code>pair_links</code>: Helper function to create pairs of observation groups.
</p>
</li>
<li> <p><code>range_link</code>: Helper function to create a range of observations.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>pair_links(..., .handle_missing = "error", .reorder = NULL)

range_link(point1, point2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pair_links_+3A_...">...</code></td>
<td>
<p>&lt;<a href="rlang.html#topic+dyn-dots">dyn-dots</a>&gt; A list of formulas, where each side
of the formula should be an <code>integer</code> or <code>character</code> index of the original
data, or a <code>range_link()</code> object defining the linked observations. Use
<code>NULL</code> to indicate no link on that side. You can also combine these by
wrapping them into a single <code>list()</code>. If only the left-hand side of the
formula exists, you can input it directly. For integer indices, wrap them
with <code><a href="base.html#topic+I">I()</a></code> to use the ordering from the layout. You can also use
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code> to inherit values from the other group.</p>
</td></tr>
<tr><td><code id="pair_links_+3A_.handle_missing">.handle_missing</code></td>
<td>
<p>A string of <code>"error"</code> or <code>"remove"</code>
indicates the action for handling missing observations.</p>
</td></tr>
<tr><td><code id="pair_links_+3A_.reorder">.reorder</code></td>
<td>
<p>A string of <code>"hand1"</code> or <code>"hand2"</code> indicating
whether to reorder the input links to follow the specified layout ordering.</p>
</td></tr>
<tr><td><code id="pair_links_+3A_point1">point1</code>, <code id="pair_links_+3A_point2">point2</code></td>
<td>
<p>A single integer or character index, defining the lower
and higher bounds of the range. For integer indices, wrap them with <code><a href="base.html#topic+I">I()</a></code>
to indicate the ordered index by the layout.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- pair_links(
    # group on the left hand only
    c("a", "b"),
    # normally, integer index will be interpreted as the index of the
    # origianl data
    1:2,
    # wrapped with `I()` indicate` the integer index is ordering of the
    # layout
    I(1:2),
    range_link(1, 6),
    range_link("a", "b"),
    # group on the right hand only
    ~ 1:2,
    ~ c("a", "b"),
    ~ range_link(1, 6),
    # group on the both side
    range_link(1, 6) ~ c("a", "b"),
    # waiver() indicates the right hand is the same of the left hand
    range_link(1, 6) ~ waiver(),
    # the same for the left hand
    waiver() ~ 1:2,
    ~NULL # an empty link
)
x

# we can modify it as usual list
x[[1]] &lt;- NULL # remove the first link
x$a &lt;- ~LETTERS
x

# modify with a list
x[1:2] &lt;- list(~ c("a", "b"), ~ range_link("a", "b"))
x
</code></pre>

<hr>
<h2 id='patch'>Convert Object into a Grob</h2><span id='topic+patch'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>patch(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to specific methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3><code>patch</code> method collections</h3>


<ul>
<li> <p><code><a href="#topic+patch.grob">patch.grob()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+patch.ggplot">patch.ggplot()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+patch.patch_ggplot">patch.patch_ggplot()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+patch.alignpatches">patch.alignpatches()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+patch.patchwork">patch.patchwork()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+patch.patch">patch.patch()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+patch.formula">patch.formula()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+patch.recordedplot">patch.recordedplot()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+patch.trellis">patch.trellis()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+patch.Heatmap">patch.Heatmap()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+patch.pheatmap">patch.pheatmap()</a></code>
</p>
</li></ul>


<hr>
<h2 id='patch_titles'>Add patch titles to plot borders</h2><span id='topic+patch_titles'></span>

<h3>Description</h3>

<p>This function extends ggplot2's title functionality, allowing you to add
titles to each border of the plot: top, left, bottom, and right.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>patch_titles(
  top = waiver(),
  left = waiver(),
  bottom = waiver(),
  right = waiver()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch_titles_+3A_top">top</code>, <code id="patch_titles_+3A_left">left</code>, <code id="patch_titles_+3A_bottom">bottom</code>, <code id="patch_titles_+3A_right">right</code></td>
<td>
<p>A string specifying the title to be added to the
top, left, bottom, and right border of the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The appearance and alignment of these patch titles can be customized using
<a href="ggplot2.html#topic+theme">theme()</a>:
</p>

<ul>
<li> <p><code>plot.patch_title</code>/<code style="white-space: pre;">&#8288;plot.patch_title.*&#8288;</code>: Controls the text appearance of
patch titles. By default, <code>plot.patch_title</code> inherit from <code>plot.title</code>, and
settings for each border will inherit from <code>plot.patch_title</code>, with the
exception of the <code>angle</code> property, which is not inherited.
</p>
</li>
<li> <p><code>plot.patch_title.position</code>/<code style="white-space: pre;">&#8288;plot.patch_title.position.*&#8288;</code>: Determines the
alignment of the patch titles. By default, <code>plot.patch_title.position</code>
inherit from <code>plot.title.position</code>, and settings for each border will
inherit from <code>plot.patch_title</code>. The value <code>"panel"</code> aligns the patch
titles with the plot panels. Setting this to <code>"plot"</code> aligns the patch
title with the entire plot (excluding margins and plot tags).
</p>
</li></ul>



<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+labs">labels</a></code> object to be added to ggplot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ggplot(mtcars) +
    geom_point(aes(mpg, disp)) +
    patch_titles(
        top = "I'm top patch title",
        left = "I'm left patch title",
        bottom = "I'm bottom patch title",
        right = "I'm right patch title"
    )
</code></pre>

<hr>
<h2 id='patch.alignpatches'>Convert Object into a Grob</h2><span id='topic+patch.alignpatches'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'alignpatches'
patch(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch.alignpatches_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch.alignpatches_+3A_...">...</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+align_plots">alignpatches</a></code>
</p>
<p>Other <code><a href="#topic+patch">patch()</a></code> methods:
<code><a href="#topic+patch.Heatmap">patch.Heatmap</a>()</code>,
<code><a href="#topic+patch.formula">patch.formula</a>()</code>,
<code><a href="#topic+patch.ggplot">patch.ggplot</a>()</code>,
<code><a href="#topic+patch.grob">patch.grob</a>()</code>,
<code><a href="#topic+patch.patch">patch.patch</a>()</code>,
<code><a href="#topic+patch.patch_ggplot">patch.patch_ggplot</a>()</code>,
<code><a href="#topic+patch.patchwork">patch.patchwork</a>()</code>,
<code><a href="#topic+patch.pheatmap">patch.pheatmap</a>()</code>,
<code><a href="#topic+patch.recordedplot">patch.recordedplot</a>()</code>,
<code><a href="#topic+patch.trellis">patch.trellis</a>()</code>
</p>

<hr>
<h2 id='patch.formula'>Convert Object into a Grob</h2><span id='topic+patch.formula'></span><span id='topic+patch.function'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'formula'
patch(x, ..., device = NULL, name = NULL)

## S3 method for class ''function''
patch(x, ..., device = NULL, name = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch.formula_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch.formula_+3A_...">...</code></td>
<td>
<p>Graphical Parameters passed on to <a href="graphics.html#topic+par">par()</a>.</p>
</td></tr>
<tr><td><code id="patch.formula_+3A_device">device</code></td>
<td>

<p>A function that opens a graphics device for <code>grid.echo()</code> to
work on.  By default this is an off-screen, in-memory device
based on the <code>pdf</code> device.  This default device may not be
satisfactory when using custom fonts.
</p>
</td></tr>
<tr><td><code id="patch.formula_+3A_name">name</code></td>
<td>

<p>A character identifier.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+plot">plot()</a></code>
</p>
<p>Other <code><a href="#topic+patch">patch()</a></code> methods:
<code><a href="#topic+patch.Heatmap">patch.Heatmap</a>()</code>,
<code><a href="#topic+patch.alignpatches">patch.alignpatches</a>()</code>,
<code><a href="#topic+patch.ggplot">patch.ggplot</a>()</code>,
<code><a href="#topic+patch.grob">patch.grob</a>()</code>,
<code><a href="#topic+patch.patch">patch.patch</a>()</code>,
<code><a href="#topic+patch.patch_ggplot">patch.patch_ggplot</a>()</code>,
<code><a href="#topic+patch.patchwork">patch.patchwork</a>()</code>,
<code><a href="#topic+patch.pheatmap">patch.pheatmap</a>()</code>,
<code><a href="#topic+patch.recordedplot">patch.recordedplot</a>()</code>,
<code><a href="#topic+patch.trellis">patch.trellis</a>()</code>
</p>

<hr>
<h2 id='patch.ggplot'>Convert Object into a Grob</h2><span id='topic+patch.ggplot'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ggplot'
patch(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch.ggplot_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch.ggplot_+3A_...">...</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3>See Also</h3>

<p><a href="ggplot2.html#topic+ggplot">ggplot</a>
</p>
<p>Other <code><a href="#topic+patch">patch()</a></code> methods:
<code><a href="#topic+patch.Heatmap">patch.Heatmap</a>()</code>,
<code><a href="#topic+patch.alignpatches">patch.alignpatches</a>()</code>,
<code><a href="#topic+patch.formula">patch.formula</a>()</code>,
<code><a href="#topic+patch.grob">patch.grob</a>()</code>,
<code><a href="#topic+patch.patch">patch.patch</a>()</code>,
<code><a href="#topic+patch.patch_ggplot">patch.patch_ggplot</a>()</code>,
<code><a href="#topic+patch.patchwork">patch.patchwork</a>()</code>,
<code><a href="#topic+patch.pheatmap">patch.pheatmap</a>()</code>,
<code><a href="#topic+patch.recordedplot">patch.recordedplot</a>()</code>,
<code><a href="#topic+patch.trellis">patch.trellis</a>()</code>
</p>

<hr>
<h2 id='patch.grob'>Convert Object into a Grob</h2><span id='topic+patch.grob'></span><span id='topic+patch.gList'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'grob'
patch(x, ...)

## S3 method for class 'gList'
patch(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch.grob_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch.grob_+3A_...">...</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3>See Also</h3>

<p>Other <code><a href="#topic+patch">patch()</a></code> methods:
<code><a href="#topic+patch.Heatmap">patch.Heatmap</a>()</code>,
<code><a href="#topic+patch.alignpatches">patch.alignpatches</a>()</code>,
<code><a href="#topic+patch.formula">patch.formula</a>()</code>,
<code><a href="#topic+patch.ggplot">patch.ggplot</a>()</code>,
<code><a href="#topic+patch.patch">patch.patch</a>()</code>,
<code><a href="#topic+patch.patch_ggplot">patch.patch_ggplot</a>()</code>,
<code><a href="#topic+patch.patchwork">patch.patchwork</a>()</code>,
<code><a href="#topic+patch.pheatmap">patch.pheatmap</a>()</code>,
<code><a href="#topic+patch.recordedplot">patch.recordedplot</a>()</code>,
<code><a href="#topic+patch.trellis">patch.trellis</a>()</code>
</p>

<hr>
<h2 id='patch.Heatmap'>Convert Object into a Grob</h2><span id='topic+patch.Heatmap'></span><span id='topic+patch.HeatmapList'></span><span id='topic+patch.HeatmapAnnotation'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Heatmap'
patch(x, ..., device = NULL)

## S3 method for class 'HeatmapList'
patch(x, ..., device = NULL)

## S3 method for class 'HeatmapAnnotation'
patch(x, ..., device = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch.Heatmap_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch.Heatmap_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <a href="ComplexHeatmap.html#topic+draw-dispatch">draw()</a>.</p>
</td></tr>
<tr><td><code id="patch.Heatmap_+3A_device">device</code></td>
<td>
<p> A function that opens a graphics device for temporary
rendering. By default this is an off-screen, in-memory device
based on the <code>pdf</code> device, but this default device may not be
satisfactory when using custom fonts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="ComplexHeatmap.html#topic+Heatmap">Heatmap()</a></code>
</p>
</li>
<li> <p><code><a href="ComplexHeatmap.html#topic+HeatmapAnnotation">HeatmapAnnotation()</a></code>
</p>
</li></ul>

<p>Other <code><a href="#topic+patch">patch()</a></code> methods:
<code><a href="#topic+patch.alignpatches">patch.alignpatches</a>()</code>,
<code><a href="#topic+patch.formula">patch.formula</a>()</code>,
<code><a href="#topic+patch.ggplot">patch.ggplot</a>()</code>,
<code><a href="#topic+patch.grob">patch.grob</a>()</code>,
<code><a href="#topic+patch.patch">patch.patch</a>()</code>,
<code><a href="#topic+patch.patch_ggplot">patch.patch_ggplot</a>()</code>,
<code><a href="#topic+patch.patchwork">patch.patchwork</a>()</code>,
<code><a href="#topic+patch.pheatmap">patch.pheatmap</a>()</code>,
<code><a href="#topic+patch.recordedplot">patch.recordedplot</a>()</code>,
<code><a href="#topic+patch.trellis">patch.trellis</a>()</code>
</p>

<hr>
<h2 id='patch.patch'>Convert Object into a Grob</h2><span id='topic+patch.patch'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'patch'
patch(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch.patch_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch.patch_+3A_...">...</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="patchwork.html#topic+patchGrob">patch</a></code>
</p>
<p>Other <code><a href="#topic+patch">patch()</a></code> methods:
<code><a href="#topic+patch.Heatmap">patch.Heatmap</a>()</code>,
<code><a href="#topic+patch.alignpatches">patch.alignpatches</a>()</code>,
<code><a href="#topic+patch.formula">patch.formula</a>()</code>,
<code><a href="#topic+patch.ggplot">patch.ggplot</a>()</code>,
<code><a href="#topic+patch.grob">patch.grob</a>()</code>,
<code><a href="#topic+patch.patch_ggplot">patch.patch_ggplot</a>()</code>,
<code><a href="#topic+patch.patchwork">patch.patchwork</a>()</code>,
<code><a href="#topic+patch.pheatmap">patch.pheatmap</a>()</code>,
<code><a href="#topic+patch.recordedplot">patch.recordedplot</a>()</code>,
<code><a href="#topic+patch.trellis">patch.trellis</a>()</code>
</p>

<hr>
<h2 id='patch.patch_ggplot'>Convert Object into a Grob</h2><span id='topic+patch.patch_ggplot'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'patch_ggplot'
patch(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch.patch_ggplot_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch.patch_ggplot_+3A_...">...</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+patch_titles">patch_titles()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+inset">inset()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ggwrap">ggwrap()</a></code>
</p>
</li></ul>

<p>Other <code><a href="#topic+patch">patch()</a></code> methods:
<code><a href="#topic+patch.Heatmap">patch.Heatmap</a>()</code>,
<code><a href="#topic+patch.alignpatches">patch.alignpatches</a>()</code>,
<code><a href="#topic+patch.formula">patch.formula</a>()</code>,
<code><a href="#topic+patch.ggplot">patch.ggplot</a>()</code>,
<code><a href="#topic+patch.grob">patch.grob</a>()</code>,
<code><a href="#topic+patch.patch">patch.patch</a>()</code>,
<code><a href="#topic+patch.patchwork">patch.patchwork</a>()</code>,
<code><a href="#topic+patch.pheatmap">patch.pheatmap</a>()</code>,
<code><a href="#topic+patch.recordedplot">patch.recordedplot</a>()</code>,
<code><a href="#topic+patch.trellis">patch.trellis</a>()</code>
</p>

<hr>
<h2 id='patch.patchwork'>Convert Object into a Grob</h2><span id='topic+patch.patchwork'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'patchwork'
patch(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch.patchwork_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch.patchwork_+3A_...">...</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="patchwork.html#topic+patchworkGrob">patchwork</a></code>
</p>
<p>Other <code><a href="#topic+patch">patch()</a></code> methods:
<code><a href="#topic+patch.Heatmap">patch.Heatmap</a>()</code>,
<code><a href="#topic+patch.alignpatches">patch.alignpatches</a>()</code>,
<code><a href="#topic+patch.formula">patch.formula</a>()</code>,
<code><a href="#topic+patch.ggplot">patch.ggplot</a>()</code>,
<code><a href="#topic+patch.grob">patch.grob</a>()</code>,
<code><a href="#topic+patch.patch">patch.patch</a>()</code>,
<code><a href="#topic+patch.patch_ggplot">patch.patch_ggplot</a>()</code>,
<code><a href="#topic+patch.pheatmap">patch.pheatmap</a>()</code>,
<code><a href="#topic+patch.recordedplot">patch.recordedplot</a>()</code>,
<code><a href="#topic+patch.trellis">patch.trellis</a>()</code>
</p>

<hr>
<h2 id='patch.pheatmap'>Convert Object into a Grob</h2><span id='topic+patch.pheatmap'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pheatmap'
patch(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch.pheatmap_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch.pheatmap_+3A_...">...</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="pheatmap.html#topic+pheatmap">pheatmap()</a></code>
</p>
<p>Other <code><a href="#topic+patch">patch()</a></code> methods:
<code><a href="#topic+patch.Heatmap">patch.Heatmap</a>()</code>,
<code><a href="#topic+patch.alignpatches">patch.alignpatches</a>()</code>,
<code><a href="#topic+patch.formula">patch.formula</a>()</code>,
<code><a href="#topic+patch.ggplot">patch.ggplot</a>()</code>,
<code><a href="#topic+patch.grob">patch.grob</a>()</code>,
<code><a href="#topic+patch.patch">patch.patch</a>()</code>,
<code><a href="#topic+patch.patch_ggplot">patch.patch_ggplot</a>()</code>,
<code><a href="#topic+patch.patchwork">patch.patchwork</a>()</code>,
<code><a href="#topic+patch.recordedplot">patch.recordedplot</a>()</code>,
<code><a href="#topic+patch.trellis">patch.trellis</a>()</code>
</p>

<hr>
<h2 id='patch.recordedplot'>Convert Object into a Grob</h2><span id='topic+patch.recordedplot'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'recordedplot'
patch(x, ..., device = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch.recordedplot_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch.recordedplot_+3A_...">...</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
<tr><td><code id="patch.recordedplot_+3A_device">device</code></td>
<td>

<p>A function that opens a graphics device for <code>grid.echo()</code> to
work on.  By default this is an off-screen, in-memory device
based on the <code>pdf</code> device.  This default device may not be
satisfactory when using custom fonts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="grDevices.html#topic+recordplot">recordPlot()</a></code>
</p>
<p>Other <code><a href="#topic+patch">patch()</a></code> methods:
<code><a href="#topic+patch.Heatmap">patch.Heatmap</a>()</code>,
<code><a href="#topic+patch.alignpatches">patch.alignpatches</a>()</code>,
<code><a href="#topic+patch.formula">patch.formula</a>()</code>,
<code><a href="#topic+patch.ggplot">patch.ggplot</a>()</code>,
<code><a href="#topic+patch.grob">patch.grob</a>()</code>,
<code><a href="#topic+patch.patch">patch.patch</a>()</code>,
<code><a href="#topic+patch.patch_ggplot">patch.patch_ggplot</a>()</code>,
<code><a href="#topic+patch.patchwork">patch.patchwork</a>()</code>,
<code><a href="#topic+patch.pheatmap">patch.pheatmap</a>()</code>,
<code><a href="#topic+patch.trellis">patch.trellis</a>()</code>
</p>

<hr>
<h2 id='patch.trellis'>Convert Object into a Grob</h2><span id='topic+patch.trellis'></span>

<h3>Description</h3>

<p>The <code>patch()</code> function is used by <code><a href="#topic+ggwrap">ggwrap()</a></code> and <code><a href="#topic+inset">inset()</a></code> to convert
objects into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'trellis'
patch(x, ..., device = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="patch.trellis_+3A_x">x</code></td>
<td>
<p>An object to be converted into a <code><a href="grid.html#topic+grid.grob">grob</a></code>.</p>
</td></tr>
<tr><td><code id="patch.trellis_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="grid.html#topic+grid.grab">grid::grid.grabExpr</a></code>
</p>

<dl>
<dt><code>warn</code></dt><dd><p> An integer specifying the amount of warnings
to emit.  0 means no warnings, 1 means warn when it is
certain that the grab will not faithfully represent the
original scene. 2 means warn if there's any possibility
that the grab will not faithfully represent the
original scene.
</p>
</dd>
<dt><code>wrap</code></dt><dd><p> A logical indicating how the output should
be captured. If <code>TRUE</code>, each non-grob element on the
display list is captured by wrapping it in a grob.
</p>
</dd>
<dt><code>wrap.grobs</code></dt><dd><p> A logical indicating whether, if we are wrapping
elements (<code>wrap=TRUE</code>), we should wrap grobs (or just
wrap viewports).
</p>
</dd>
<dt><code>width,height</code></dt><dd><p> Size of the device used for temporary
rendering. </p>
</dd>
</dl>
</td></tr>
<tr><td><code id="patch.trellis_+3A_device">device</code></td>
<td>
<p> A function that opens a graphics device for temporary
rendering. By default this is an off-screen, in-memory device
based on the <code>pdf</code> device, but this default device may not be
satisfactory when using custom fonts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="grid.html#topic+grid.grob">grob</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="lattice.html#topic+trellis.object">trellis</a></code>
</p>
<p>Other <code><a href="#topic+patch">patch()</a></code> methods:
<code><a href="#topic+patch.Heatmap">patch.Heatmap</a>()</code>,
<code><a href="#topic+patch.alignpatches">patch.alignpatches</a>()</code>,
<code><a href="#topic+patch.formula">patch.formula</a>()</code>,
<code><a href="#topic+patch.ggplot">patch.ggplot</a>()</code>,
<code><a href="#topic+patch.grob">patch.grob</a>()</code>,
<code><a href="#topic+patch.patch">patch.patch</a>()</code>,
<code><a href="#topic+patch.patch_ggplot">patch.patch_ggplot</a>()</code>,
<code><a href="#topic+patch.patchwork">patch.patchwork</a>()</code>,
<code><a href="#topic+patch.pheatmap">patch.pheatmap</a>()</code>,
<code><a href="#topic+patch.recordedplot">patch.recordedplot</a>()</code>
</p>

<hr>
<h2 id='quad_active'>Determine the Active Context of Quad-Layout</h2><span id='topic+quad_active'></span><span id='topic+quad_anno'></span><span id='topic+anno_top'></span><span id='topic+anno_left'></span><span id='topic+anno_bottom'></span><span id='topic+anno_right'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>

<ul>
<li> <p><code>quad_active</code>: Sets the active context to the <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code> itself.
</p>
</li>
<li> <p><code>quad_anno</code>: Sets the active context to the specified annotation stack
based on the <code>position</code> argument.
</p>
</li>
<li> <p><code>anno_top</code>: A special case of <code>quad_anno</code> with <code>position = "top"</code>.
</p>
</li>
<li> <p><code>anno_left</code>: A special case of <code>quad_anno</code> with <code>position = "left"</code>.
</p>
</li>
<li> <p><code>anno_bottom</code>: A special case of <code>quad_anno</code> with <code>position = "bottom"</code>.
</p>
</li>
<li> <p><code>anno_right</code>: A special case of <code>quad_anno</code> with <code>position = "right"</code>.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>quad_active(width = NULL, height = NULL)

quad_anno(
  position,
  size = NULL,
  free_guides = waiver(),
  initialize = NULL,
  what = waiver()
)

anno_top(
  size = NULL,
  free_guides = waiver(),
  initialize = NULL,
  what = waiver()
)

anno_left(
  size = NULL,
  free_guides = waiver(),
  initialize = NULL,
  what = waiver()
)

anno_bottom(
  size = NULL,
  free_guides = waiver(),
  initialize = NULL,
  what = waiver()
)

anno_right(
  size = NULL,
  free_guides = waiver(),
  initialize = NULL,
  what = waiver()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="quad_active_+3A_width">width</code>, <code id="quad_active_+3A_height">height</code></td>
<td>
<p>The relative width/height of the main plot, can be a
<code><a href="grid.html#topic+unit">unit</a></code> object.</p>
</td></tr>
<tr><td><code id="quad_active_+3A_position">position</code></td>
<td>
<p>A string of <code>"top"</code>, <code>"left"</code>, <code>"bottom"</code>, or <code>"right"</code> indicates which annotation stack should be activated.</p>
</td></tr>
<tr><td><code id="quad_active_+3A_size">size</code></td>
<td>
<p>A numeric value or an <code><a href="grid.html#topic+unit">unit</a></code> object to set the
total <code>height</code>/<code>width</code> of the annotation stack.
</p>

<ul>
<li><p> If <code>position</code> is <code>"top"</code> or <code>"bottom"</code>, <code>size</code> sets the total height of
the annotation.
</p>
</li>
<li><p> If <code>position</code> is <code>"left"</code> or <code>"right"</code>, <code>size</code> sets the total width of the
annotation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="quad_active_+3A_free_guides">free_guides</code></td>
<td>
<p>Override the <code>guides</code> collection behavior specified in the
<code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code> for the annotation stack.</p>
</td></tr>
<tr><td><code id="quad_active_+3A_initialize">initialize</code></td>
<td>
<p>A boolean indicating whether the annotation stack should be
initialized if it is not already. By default, the annotation stack layout
will attempt to initialize when the data is compatible. If set to <code>TRUE</code>, and
the data in <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code> is incompatible with the annotation stack, no
data will be used in the stack.</p>
</td></tr>
<tr><td><code id="quad_active_+3A_what">what</code></td>
<td>
<p>What should get activated in the annotation stack?
A single number or string of the plot elements in the layout. If <code>NULL</code>, will remove any active context.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, <code>quad_anno()</code> attempts to initialize the annotation stack layout
using data from <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code>. However, in situations where you want to use
different data for the annotation stack, you can set <code>initialize = FALSE</code>
and then provide a custom <code>stack_layout()</code>.
</p>


<h3>Value</h3>

<p>An object that can be added to <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+quad_switch">quad_switch()</a></code>
</p>

<hr>
<h2 id='quad_init'>Initialize Quad-Layout Annotation</h2><span id='topic+quad_init'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
</p>
<p>This function was deprecated, you can add <code>stack_layout()</code> directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quad_init(position, data = waiver(), ...)
</code></pre>

<hr>
<h2 id='quad_layout'>Arrange plots in the quad-side of a main plot</h2><span id='topic+quad_layout'></span><span id='topic+quad_alignh'></span><span id='topic+quad_alignv'></span><span id='topic+quad_discrete'></span><span id='topic+quad_alignb'></span><span id='topic+quad_continuous'></span><span id='topic+ggside'></span><span id='topic+quad_free'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>This function arranges plots around the quad-sides of a main plot, aligning
both horizontal and vertical axes, and can handle either discrete or
continuous variables.
</p>

<ul>
<li><p> If <code>xlim</code> is provided, a continuous variable will be required and aligned
in the vertical direction. Otherwise, a discrete variable will be required
and aligned.
</p>
</li>
<li><p> If <code>ylim</code> is provided, a continuous variable will be required and aligned
in the horizontal direction. Otherwise, a discrete variable will be
required and aligned.
</p>
</li></ul>

<p>The <code>quad_discrete</code> is a special case where both <code>xlim</code> and <code>ylim</code> are not
provided.
</p>
<p>The <code>quad_continuous</code> is a special case where both <code>xlim</code> and <code>ylim</code> are
provided.
</p>
<p>For historical reasons, the following aliases are available:
</p>

<ul>
<li> <p><code>quad_alignh</code>: Align discrete variables in the horizontal direction and
continuous variables in vertical direction.
</p>
</li>
<li> <p><code>quad_alignv</code>: Align discrete variables in the vertical direction and
continuous variables in horizontal direction.
</p>
</li>
<li> <p><code>quad_alignb</code> is an alias for <code>quad_discrete</code>.
</p>
</li>
<li> <p><code>quad_free</code> is an alias for <code>quad_continuous</code>.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>quad_layout(
  data = waiver(),
  mapping = aes(),
  xlim = waiver(),
  ylim = waiver(),
  ...,
  theme = NULL,
  active = NULL,
  width = NA,
  height = NA
)

quad_alignh(..., ylim = waiver())

quad_alignv(..., xlim = waiver())

quad_discrete(
  data = waiver(),
  mapping = aes(),
  ...,
  theme = NULL,
  active = NULL,
  width = NA,
  height = NA
)

quad_continuous(
  data = waiver(),
  mapping = aes(),
  xlim = NULL,
  ylim = NULL,
  ...,
  theme = NULL,
  active = NULL,
  width = NA,
  height = NA
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="quad_layout_+3A_data">data</code></td>
<td>
<p>Default dataset to use for the layout. If not specified, it must be supplied in each plot added to the layout. By default, this will attempt
to inherit from the parent layout.
</p>
<p>If both <code>xlim</code> and <code>ylim</code> are provided, a <code style="white-space: pre;">&#8288;data frame&#8288;</code> is required, and
<code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> will be used to convert the data to a data frame.
When inherited by an annotation stack, no transposition will be applied.
</p>
<p>Otherwise, a <code>matrix</code> is required, and <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> will be used to
convert the data to a matrix. When inherited by the column annotation
stack, the data will be transposed.</p>
</td></tr>
<tr><td><code id="quad_layout_+3A_mapping">mapping</code></td>
<td>
<p>Default list of aesthetic mappings to use for main plot in the
layout. If not specified, must be supplied in each layer added to the main
plot.</p>
</td></tr>
<tr><td><code id="quad_layout_+3A_xlim">xlim</code>, <code id="quad_layout_+3A_ylim">ylim</code></td>
<td>
<p>A <code><a href="#topic+continuous_limits">continuous_limits()</a></code> object specifying the left/lower
limit and the right/upper limit of the scale. Used to align the continuous
axis.</p>
</td></tr>
<tr><td><code id="quad_layout_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> or
<code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>.</p>
</td></tr>
<tr><td><code id="quad_layout_+3A_theme">theme</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+theme">theme()</a></code> object used to customize various
elements of the layout, including <code>guides</code>, <code>title</code>, <code>subtitle</code>, <code>caption</code>,
<code>margins</code>, <code>panel.border</code>, and <code>background</code>. By default, the theme will
inherit from the parent <code>layout</code>. It also controls the panel spacing for all
plots in the layout.</p>
</td></tr>
<tr><td><code id="quad_layout_+3A_active">active</code></td>
<td>
<p>A <code><a href="#topic+active">active()</a></code> object that defines the context settings when
added to a layout.</p>
</td></tr>
<tr><td><code id="quad_layout_+3A_width">width</code>, <code id="quad_layout_+3A_height">height</code></td>
<td>
<p>The relative width/height of the main plot, can be a
<code><a href="grid.html#topic+unit">unit</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>QuadLayout</code> object.
</p>


<h3>ggplot2 specification</h3>

<p>If either <code>xlim</code> or <code>ylim</code> is not provided, the data input will be converted
to a matrix using <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>, and the data in the underlying main
plot will contain the following columns:
</p>

<ul>
<li> <p><code>.panel_x</code> and <code>.panel_y</code>: the column and row panel groups.
</p>
</li>
<li> <p><code>.x</code> and <code>.y</code>: an integer index of <code>x</code> and <code>y</code> coordinates
</p>
</li>
<li> <p><code>.discrete_x</code> and <code>.discrete_y</code>: a factor of the data labels (only
applicable when <code>.row_names</code> and <code>.column_names</code> exists).
</p>
</li>
<li> <p><code>.row_names</code> and <code>.column_names</code>: A character of the row and column names
of the original matrix (only applicable when names exist).
</p>
</li>
<li> <p><code>.row_index</code> and <code>.column_index</code>: the row and column index of the original
matrix.
</p>
</li>
<li> <p><code>value</code>: the actual matrix value.
</p>
</li></ul>

<p>Otherwise, the data input will be used for the main plot.
</p>

<hr>
<h2 id='quad_switch'>Determine the Active Context of Quad-Layout</h2><span id='topic+quad_switch'></span><span id='topic+hmanno'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p><code>quad_switch()</code> integrates <code><a href="#topic+quad_active">quad_active()</a></code> and <code><a href="#topic+quad_anno">quad_anno()</a></code> into one
function for ease of use. This function allows you to quickly change the
active context of the <code><a href="#topic+quad_layout">quad_layout()</a></code> and its annotations.
</p>
<p><code>hmanno</code> is an alias for <code>quad_switch</code>, with additional arguments for
backward compatibility
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quad_switch(
  position = NULL,
  size = NULL,
  width = NULL,
  height = NULL,
  free_guides = waiver(),
  initialize = NULL,
  what = waiver()
)

hmanno(
  position = NULL,
  size = NULL,
  width = NULL,
  height = NULL,
  free_guides = waiver(),
  initialize = NULL,
  what = waiver()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="quad_switch_+3A_position">position</code></td>
<td>
<p>A string of <code>"top"</code>, <code>"left"</code>, <code>"bottom"</code>, or <code>"right"</code> indicates which annotation stack should be activated. If <code>NULL</code>, it sets the
active context to the <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code> itself.</p>
</td></tr>
<tr><td><code id="quad_switch_+3A_size">size</code></td>
<td>
<p>A numeric value or an <code><a href="grid.html#topic+unit">unit</a></code> object to set the
total <code>height</code>/<code>width</code> of the annotation stack.
</p>

<ul>
<li><p> If <code>position</code> is <code>"top"</code> or <code>"bottom"</code>, <code>size</code> sets the total height of
the annotation.
</p>
</li>
<li><p> If <code>position</code> is <code>"left"</code> or <code>"right"</code>, <code>size</code> sets the total width of the
annotation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="quad_switch_+3A_width">width</code>, <code id="quad_switch_+3A_height">height</code></td>
<td>
<p>The relative width/height of the main plot, can be a
<code><a href="grid.html#topic+unit">unit</a></code> object.</p>
</td></tr>
<tr><td><code id="quad_switch_+3A_free_guides">free_guides</code></td>
<td>
<p>Override the <code>guides</code> collection behavior specified in the
<code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code> for the annotation stack.</p>
</td></tr>
<tr><td><code id="quad_switch_+3A_initialize">initialize</code></td>
<td>
<p>A boolean indicating whether the annotation stack should be
initialized if it is not already. By default, the annotation stack layout
will attempt to initialize when the data is compatible. If set to <code>TRUE</code>, and
the data in <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code> is incompatible with the annotation stack, no
data will be used in the stack.</p>
</td></tr>
<tr><td><code id="quad_switch_+3A_what">what</code></td>
<td>
<p>What should get activated in the annotation stack?
A single number or string of the plot elements in the layout. If <code>NULL</code>, will remove any active context.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object that can be added to <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+quad_active">quad_active()</a></code>/<code><a href="#topic+quad_anno">quad_anno()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ggheatmap(matrix(rnorm(81), nrow = 9)) +
    anno_top() +
    align_dendro()
</code></pre>

<hr>
<h2 id='raster_magick'>Rasterize the input object</h2><span id='topic+raster_magick'></span>

<h3>Description</h3>

<p>The function rasterizes input graphical objects (e.g., grob, layer, ggplot)
and optionally processes the resulting raster using magick, a powerful image
manipulation library. This allows for advanced graphical transformations
directly within the plotting pipeline.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>raster_magick(x, magick = NULL, ..., res = NULL, interpolate = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="raster_magick_+3A_x">x</code></td>
<td>
<p>An object to rasterize, can be a <code><a href="grid.html#topic+grid.grob">grob()</a></code>,
<code><a href="ggplot2.html#topic+layer">layer()</a></code>, <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>, or a list of such
objects.</p>
</td></tr>
<tr><td><code id="raster_magick_+3A_magick">magick</code></td>
<td>
<p>A function (purrr-style formula is accepted) that takes an
<code><a href="magick.html#topic+editing">image_read()</a></code> object as input and returns an object
compatible with <code><a href="grDevices.html#topic+as.raster">as.raster()</a></code>. You can use any of
the <code style="white-space: pre;">&#8288;image_*()&#8288;</code> functions from the <strong>magick</strong> package to process the raster
image.</p>
</td></tr>
<tr><td><code id="raster_magick_+3A_...">...</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
<tr><td><code id="raster_magick_+3A_res">res</code></td>
<td>
<p>An integer sets the desired resolution in pixels.</p>
</td></tr>
<tr><td><code id="raster_magick_+3A_interpolate">interpolate</code></td>
<td>

<p>A logical value indicating whether to linearly interpolate the
image (the alternative is to use nearest-neighbour interpolation,
which gives a more blocky result).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with the same class of the input.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># data generated code was copied from `ComplexHeatmap`
set.seed(123)
small_mat &lt;- matrix(rnorm(56), nrow = 7)
rownames(small_mat) &lt;- paste0("row", seq_len(nrow(small_mat)))
colnames(small_mat) &lt;- paste0("column", seq_len(ncol(small_mat)))
ggheatmap(small_mat, aes(.x, .y), filling = NULL) +
    raster_magick(geom_tile(aes(fill = value)), res = 20)

ggheatmap(small_mat, aes(.x, .y), filling = NULL) +
    # Use `magick::filter_types()` to check available `filter` arguments
    raster_magick(geom_tile(aes(fill = value)),
        magick = function(image) {
            magick::image_resize(image,
                geometry = "50%x", filter = "Lanczos"
            )
        }
    )
</code></pre>

<hr>
<h2 id='raster_magick_interal'>Rasterize the input object</h2><span id='topic+raster_magick_interal'></span><span id='topic+.raster_magick'></span><span id='topic+.raster_magick.Layer'></span><span id='topic+.raster_magick.list'></span><span id='topic+.raster_magick.ggplot'></span><span id='topic+.raster_magick.grob'></span><span id='topic+.raster_magick.gList'></span><span id='topic+.raster_magick.ggalignRasterMagick'></span><span id='topic+.raster_magick.zeroGrob'></span><span id='topic+.raster_magick.default'></span>

<h3>Description</h3>

<p>An internal function designed to implement the functionality of
<code>raster_magick()</code>. It assumes the input arguments are valid and does not
perform any additional checks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.raster_magick(x, ...)

## S3 method for class 'Layer'
.raster_magick(x, ...)

## S3 method for class 'list'
.raster_magick(x, ...)

## S3 method for class 'ggplot'
.raster_magick(x, ...)

## S3 method for class 'grob'
.raster_magick(x, magick = NULL, ..., res = NULL, interpolate = FALSE)

## S3 method for class 'gList'
.raster_magick(x, magick = NULL, ..., res = NULL, interpolate = FALSE)

## S3 method for class 'ggalignRasterMagick'
.raster_magick(x, magick = NULL, ..., res = NULL, interpolate = FALSE)

## S3 method for class 'zeroGrob'
.raster_magick(x, ...)

## Default S3 method:
.raster_magick(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="raster_magick_interal_+3A_x">x</code></td>
<td>
<p>An object to rasterize, can be a <code><a href="grid.html#topic+grid.grob">grob()</a></code>,
<code><a href="ggplot2.html#topic+layer">layer()</a></code>, <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>, or a list of such
objects.</p>
</td></tr>
<tr><td><code id="raster_magick_interal_+3A_...">...</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
<tr><td><code id="raster_magick_interal_+3A_magick">magick</code></td>
<td>
<p>A function (purrr-style formula is accepted) that takes an
<code><a href="magick.html#topic+editing">image_read()</a></code> object as input and returns an object
compatible with <code><a href="grDevices.html#topic+as.raster">as.raster()</a></code>. You can use any of
the <code style="white-space: pre;">&#8288;image_*()&#8288;</code> functions from the <strong>magick</strong> package to process the raster
image.</p>
</td></tr>
<tr><td><code id="raster_magick_interal_+3A_res">res</code></td>
<td>
<p>An integer sets the desired resolution in pixels.</p>
</td></tr>
<tr><td><code id="raster_magick_interal_+3A_interpolate">interpolate</code></td>
<td>

<p>A logical value indicating whether to linearly interpolate the
image (the alternative is to use nearest-neighbour interpolation,
which gives a more blocky result).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with the same class of the input.
</p>

<hr>
<h2 id='read_example'>Read Example Data</h2><span id='topic+read_example'></span>

<h3>Description</h3>

<p>This function reads example data from the file. If no file is specified, it
returns a list of available example files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_example(file = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_example_+3A_file">file</code></td>
<td>
<p>A string representing the name of the example file to be read. If
<code>NULL</code>, the function will return a list of available example file names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>file</code> is <code>NULL</code>, returns a character vector of available example
file names. Otherwise, returns the contents of the specified example file,
read as an R object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>read_example()
</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+gpar'></span><span id='topic+unit'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>grid</dt><dd><p><code><a href="grid.html#topic+gpar">gpar</a></code>, <code><a href="grid.html#topic+unit">unit</a></code></p>
</dd>
</dl>

<hr>
<h2 id='scale_gshape_manual'>Scale for <code>gshape</code> aesthetic</h2><span id='topic+scale_gshape_manual'></span>

<h3>Description</h3>

<p>Scale for <code>gshape</code> aesthetic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_gshape_manual(..., values, breaks = waiver(), na.value = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scale_gshape_manual_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="ggplot2.html#topic+discrete_scale">ggplot2::discrete_scale</a></code>
</p>

<dl>
<dt><code>name</code></dt><dd><p>The name of the scale. Used as the axis or legend title. If
<code>waiver()</code>, the default, the name of the scale is taken from the first
mapping used for that aesthetic. If <code>NULL</code>, the legend title will be
omitted.</p>
</dd>
<dt><code>labels</code></dt><dd><p>One of:
</p>

<ul>
<li> <p><code>NULL</code> for no labels
</p>
</li>
<li> <p><code>waiver()</code> for the default labels computed by the
transformation object
</p>
</li>
<li><p> A character vector giving labels (must be same length as <code>breaks</code>)
</p>
</li>
<li><p> An expression vector (must be the same length as breaks). See ?plotmath for details.
</p>
</li>
<li><p> A function that takes the breaks as input and returns labels
as output. Also accepts rlang <a href="rlang.html#topic+as_function">lambda</a> function
notation.
</p>
</li></ul>
</dd>
<dt><code>limits</code></dt><dd><p>One of:
</p>

<ul>
<li> <p><code>NULL</code> to use the default scale values
</p>
</li>
<li><p> A character vector that defines possible values of the scale and their
order
</p>
</li>
<li><p> A function that accepts the existing (automatic) values and returns
new ones. Also accepts rlang <a href="rlang.html#topic+as_function">lambda</a> function
notation.
</p>
</li></ul>
</dd>
<dt><code>na.translate</code></dt><dd><p>Unlike continuous scales, discrete scales can easily show
missing values, and do so by default. If you want to remove missing values
from a discrete scale, specify <code>na.translate = FALSE</code>.</p>
</dd>
<dt><code>drop</code></dt><dd><p>Should unused factor levels be omitted from the scale?
The default, <code>TRUE</code>, uses the levels that appear in the data;
<code>FALSE</code> includes the levels in the factor. Please note that to display
every level in a legend, the layer should use <code>show.legend = TRUE</code>.</p>
</dd>
<dt><code>guide</code></dt><dd><p>A function used to create a guide or its name. See
<code><a href="ggplot2.html#topic+guides">guides()</a></code> for more information.</p>
</dd>
<dt><code>call</code></dt><dd><p>The <code>call</code> used to construct the scale for reporting messages.</p>
</dd>
<dt><code>super</code></dt><dd><p>The super class to use for the constructed scale</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="scale_gshape_manual_+3A_values">values</code></td>
<td>
<p>A list of grobs or functions (including purrr-like lambda
syntax) that define how each cell's grob (graphical object) should be drawn.</p>
</td></tr>
<tr><td><code id="scale_gshape_manual_+3A_breaks">breaks</code></td>
<td>
<p>One of:
</p>

<ul>
<li> <p><code>NULL</code> for no breaks
</p>
</li>
<li> <p><code>waiver()</code> for the default breaks (the scale limits)
</p>
</li>
<li><p> A character vector of breaks
</p>
</li>
<li><p> A function that takes the limits as input and returns breaks
as output
</p>
</li></ul>
</td></tr>
<tr><td><code id="scale_gshape_manual_+3A_na.value">na.value</code></td>
<td>
<p>The aesthetic value to use for missing (<code>NA</code>) values</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>geom_gshape</code> depends on the new aesthetics <code>gshape</code> (shape with grid
functions), which should always be provided with <code><a href="#topic+scale_gshape_manual">scale_gshape_manual()</a></code>,
in which, we can provide a list of grobs or functions that define how each
value should be drawn. Any ggplot2 aesthetics can be used as the arguments.
</p>


<h3>Aesthetics</h3>

<p><code>geom_gshape()</code> understands the following aesthetics (required aesthetics are in bold):
</p>

<ul>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">x</a></code></strong>
</p>
</li>
<li> <p><strong><code><a href="ggplot2.html#topic+aes_position">y</a></code></strong>
</p>
</li>
<li> <p><strong><code>gshape</code></strong>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">alpha</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">colour</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_colour_fill_alpha">fill</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_group_order">group</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linetype</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">linewidth</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">shape</a></code>
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+aes_linetype_size_shape">size</a></code>
</p>
</li>
<li> <p><code>stroke</code>
</p>
</li></ul>

<p>Learn more about setting these aesthetics in <code>vignette("ggplot2-specs", package = "ggplot2")</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(grid)
ggplot(data.frame(value = letters[seq_len(5)], y = seq_len(5))) +
    geom_gshape(aes(x = 1, y = y, gshape = value, fill = value)) +
    scale_gshape_manual(values = list(
        a = function(x, y, width, height, fill) {
            rectGrob(x, y,
                width = width, height = height,
                gp = gpar(fill = fill),
                default.units = "native"
            )
        },
        b = function(x, y, width, height, fill) {
            rectGrob(x, y,
                width = width, height = height,
                gp = gpar(fill = fill),
                default.units = "native"
            )
        },
        c = function(x, y, width, height, fill) {
            rectGrob(x, y,
                width = width, height = height,
                gp = gpar(fill = fill),
                default.units = "native"
            )
        },
        d = function(x, y, width, height, shape) {
            gList(
                pointsGrob(x, y, pch = shape),
                # To ensure the rectangle color is shown in the legends, you
                # must explicitly provide a color argument and include it in
                # the `gpar()` of the graphical object
                rectGrob(x, y, width, height,
                    gp = gpar(col = "black", fill = NA)
                )
            )
        },
        e = function(xmin, xmax, ymin, ymax) {
            segmentsGrob(
                xmin, ymin,
                xmax, ymax,
                gp = gpar(lwd = 2)
            )
        }
    )) +
    scale_fill_brewer(palette = "Dark2") +
    theme_void()

</code></pre>

<hr>
<h2 id='scheme_align'>Align Specifications in the Layout</h2><span id='topic+scheme_align'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>The <code>scheme_align()</code> function defines the align Specifications for plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scheme_align(guides = NA, free_spaces = NA, free_labs = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scheme_align_+3A_guides">guides</code></td>
<td>
<p>A string with one or more of <code>"t"</code>, <code>"l"</code>, <code>"b"</code>, and <code>"r"</code> indicating
which side of guide legends should be collected. Defaults to
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code>, which inherits from the parent layout. If no
parent layout, all guides will be collected. If <code>NULL</code>, no guides will be
collected.</p>
</td></tr>
<tr><td><code id="scheme_align_+3A_free_spaces">free_spaces</code></td>
<td>
<p>A string with one or more of <code>"t"</code>, <code>"l"</code>, <code>"b"</code>, and <code>"r"</code>
indicating which border spaces should be removed. Defaults to
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code>, which inherits from the parent layout. If no
parent, the default is <code>NULL</code>, meaning no spaces are removed.
</p>
<p>Usually you want to apply this with the whole layout, instead of individual
plots.</p>
</td></tr>
<tr><td><code id="scheme_align_+3A_free_labs">free_labs</code></td>
<td>
<p>A string with one or more of <code>"t"</code>, <code>"l"</code>, <code>"b"</code>, and <code>"r"</code>
indicating which axis titles should be free from alignment. Defaults to
<code><a href="ggplot2.html#topic+waiver">waiver()</a></code>, which inherits from the parent layout. If no
parent layout, no axis titles will be aligned. If <code>NULL</code>, all axis titles
will be aligned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>scheme_align</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
mat &lt;- matrix(rnorm(72), nrow = 8)
# used in the layout, define the default action for all plots in the layout
ggheatmap(mat) -
    scheme_align(guides = NULL) +
    anno_right() +
    align_dendro(aes(color = branch), k = 3)

# You can also add it for a single plot
ggheatmap(mat) -
    # for all plots in the layout, we default won't collect any guide legends
    scheme_align(guides = NULL) +
    # for the heatmap body, we collect guide legends in the right
    # note, the guide legends will be collected to the right side of the
    # layout which will overlap the legends in the right annotation
    scheme_align(guides = "r") +
    anno_right() +
    align_dendro(aes(color = branch), k = 3)

# to avoid overlapping, we can also collect the guide legends in the
# right annotation
ggheatmap(mat) -
    scheme_align(guides = NULL) +
    scheme_align(guides = "r") +
    anno_right() +
    align_dendro(aes(color = branch), k = 3) +
    scheme_align(guides = "r")
</code></pre>

<hr>
<h2 id='scheme_data'>Plot data Specifications</h2><span id='topic+scheme_data'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Transforms the plot data. Many functions in this package require a specific
data format to align observations, <code>scheme_data()</code> helps reformat data frames
as needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scheme_data(data, inherit = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scheme_data_+3A_data">data</code></td>
<td>
<p>A function to transform the plot data before rendering.
Acceptable values include:
</p>

<ul>
<li> <p><code>NULL</code>: No action taken.
</p>
</li>
<li> <p><code><a href="ggplot2.html#topic+waiver">waiver()</a></code>: Inherits from the parent layout.
</p>
</li>
<li><p> A <code>function</code> or purrr-style <code>formula</code>: Used to transform the plot data,
which should accept a data frame and return a data frame. You can apply
this after the parent layout <code>scheme_data</code> function, using the <code>inherit</code>
argument.
</p>
</li></ul>

<p>Use this hook to modify the data for all <code>geoms</code> after the layout is created
(for matrix data, it has been melted to a long format data frame) but before
rendering by <code>ggplot2</code>. The returned data must be a data frame for ggplot.</p>
</td></tr>
<tr><td><code id="scheme_data_+3A_inherit">inherit</code></td>
<td>
<p>A single boolean value indicates whether to apply the parent
<code>scheme_data</code> first and then apply the specified <code>scheme_data</code> for the plot.
Defaults to <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Defaults will attempt to inherit from the parent layout if the actual data is
inherited from the parent layout, with one exception: <code>align_dendro()</code>, which
will not inherit the <code>scheme_data</code> by default.
</p>

<hr>
<h2 id='scheme_theme'>Plot default theme</h2><span id='topic+scheme_theme'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p><code>scheme_theme()</code> serves as the default theme and will always be overridden by
any <code>theme()</code> settings applied directly to the plot. The default theme
(<code>scheme_theme()</code>) is applied first, followed by any specific <code>theme()</code>
settings, even if <code>theme()</code> is added before <code>scheme_theme()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scheme_theme(
  ...,
  line,
  rect,
  text,
  title,
  aspect.ratio,
  axis.title,
  axis.title.x,
  axis.title.x.top,
  axis.title.x.bottom,
  axis.title.y,
  axis.title.y.left,
  axis.title.y.right,
  axis.text,
  axis.text.x,
  axis.text.x.top,
  axis.text.x.bottom,
  axis.text.y,
  axis.text.y.left,
  axis.text.y.right,
  axis.text.theta,
  axis.text.r,
  axis.ticks,
  axis.ticks.x,
  axis.ticks.x.top,
  axis.ticks.x.bottom,
  axis.ticks.y,
  axis.ticks.y.left,
  axis.ticks.y.right,
  axis.ticks.theta,
  axis.ticks.r,
  axis.minor.ticks.x.top,
  axis.minor.ticks.x.bottom,
  axis.minor.ticks.y.left,
  axis.minor.ticks.y.right,
  axis.minor.ticks.theta,
  axis.minor.ticks.r,
  axis.ticks.length,
  axis.ticks.length.x,
  axis.ticks.length.x.top,
  axis.ticks.length.x.bottom,
  axis.ticks.length.y,
  axis.ticks.length.y.left,
  axis.ticks.length.y.right,
  axis.ticks.length.theta,
  axis.ticks.length.r,
  axis.minor.ticks.length,
  axis.minor.ticks.length.x,
  axis.minor.ticks.length.x.top,
  axis.minor.ticks.length.x.bottom,
  axis.minor.ticks.length.y,
  axis.minor.ticks.length.y.left,
  axis.minor.ticks.length.y.right,
  axis.minor.ticks.length.theta,
  axis.minor.ticks.length.r,
  axis.line,
  axis.line.x,
  axis.line.x.top,
  axis.line.x.bottom,
  axis.line.y,
  axis.line.y.left,
  axis.line.y.right,
  axis.line.theta,
  axis.line.r,
  legend.background,
  legend.margin,
  legend.spacing,
  legend.spacing.x,
  legend.spacing.y,
  legend.key,
  legend.key.size,
  legend.key.height,
  legend.key.width,
  legend.key.spacing,
  legend.key.spacing.x,
  legend.key.spacing.y,
  legend.frame,
  legend.ticks,
  legend.ticks.length,
  legend.axis.line,
  legend.text,
  legend.text.position,
  legend.title,
  legend.title.position,
  legend.position,
  legend.position.inside,
  legend.direction,
  legend.byrow,
  legend.justification,
  legend.justification.top,
  legend.justification.bottom,
  legend.justification.left,
  legend.justification.right,
  legend.justification.inside,
  legend.location,
  legend.box,
  legend.box.just,
  legend.box.margin,
  legend.box.background,
  legend.box.spacing,
  panel.background,
  panel.border,
  panel.spacing,
  panel.spacing.x,
  panel.spacing.y,
  panel.grid,
  panel.grid.major,
  panel.grid.minor,
  panel.grid.major.x,
  panel.grid.major.y,
  panel.grid.minor.x,
  panel.grid.minor.y,
  panel.ontop,
  plot.background,
  plot.title,
  plot.title.position,
  plot.subtitle,
  plot.caption,
  plot.caption.position,
  plot.tag,
  plot.tag.position,
  plot.tag.location,
  plot.margin,
  strip.background,
  strip.background.x,
  strip.background.y,
  strip.clip,
  strip.placement,
  strip.text,
  strip.text.x,
  strip.text.x.bottom,
  strip.text.x.top,
  strip.text.y,
  strip.text.y.left,
  strip.text.y.right,
  strip.switch.pad.grid,
  strip.switch.pad.wrap,
  complete = FALSE,
  validate = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scheme_theme_+3A_...">...</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+theme">theme()</a></code> object or additional element
specifications not part of base ggplot2. In general, these should also be
defined in the <code style="white-space: pre;">&#8288;element tree&#8288;</code> argument. <code><a href="rlang.html#topic+splice">Splicing</a></code> a list
is also supported.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_line">line</code></td>
<td>
<p>all line elements (<code><a href="ggplot2.html#topic+element_line">element_line()</a></code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_rect">rect</code></td>
<td>
<p>all rectangular elements (<code><a href="ggplot2.html#topic+element_rect">element_rect()</a></code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_text">text</code></td>
<td>
<p>all text elements (<code><a href="ggplot2.html#topic+element_text">element_text()</a></code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_title">title</code></td>
<td>
<p>all title elements: plot, axes, legends (<code><a href="ggplot2.html#topic+element_text">element_text()</a></code>;
inherits from <code>text</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>aspect ratio of the panel</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_axis.title">axis.title</code>, <code id="scheme_theme_+3A_axis.title.x">axis.title.x</code>, <code id="scheme_theme_+3A_axis.title.y">axis.title.y</code>, <code id="scheme_theme_+3A_axis.title.x.top">axis.title.x.top</code>, <code id="scheme_theme_+3A_axis.title.x.bottom">axis.title.x.bottom</code>, <code id="scheme_theme_+3A_axis.title.y.left">axis.title.y.left</code>, <code id="scheme_theme_+3A_axis.title.y.right">axis.title.y.right</code></td>
<td>
<p>labels of axes (<code><a href="ggplot2.html#topic+element_text">element_text()</a></code>). Specify all axes' labels (<code>axis.title</code>),
labels by plane (using <code>axis.title.x</code> or <code>axis.title.y</code>), or individually
for each axis (using <code>axis.title.x.bottom</code>, <code>axis.title.x.top</code>,
<code>axis.title.y.left</code>, <code>axis.title.y.right</code>). <code style="white-space: pre;">&#8288;axis.title.*.*&#8288;</code> inherits from
<code style="white-space: pre;">&#8288;axis.title.*&#8288;</code> which inherits from <code>axis.title</code>, which in turn inherits
from <code>text</code></p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_axis.text">axis.text</code>, <code id="scheme_theme_+3A_axis.text.x">axis.text.x</code>, <code id="scheme_theme_+3A_axis.text.y">axis.text.y</code>, <code id="scheme_theme_+3A_axis.text.x.top">axis.text.x.top</code>, <code id="scheme_theme_+3A_axis.text.x.bottom">axis.text.x.bottom</code>, <code id="scheme_theme_+3A_axis.text.y.left">axis.text.y.left</code>, <code id="scheme_theme_+3A_axis.text.y.right">axis.text.y.right</code>, <code id="scheme_theme_+3A_axis.text.theta">axis.text.theta</code>, <code id="scheme_theme_+3A_axis.text.r">axis.text.r</code></td>
<td>
<p>tick labels along axes (<code><a href="ggplot2.html#topic+element_text">element_text()</a></code>). Specify all axis tick labels (<code>axis.text</code>),
tick labels by plane (using <code>axis.text.x</code> or <code>axis.text.y</code>), or individually
for each axis (using <code>axis.text.x.bottom</code>, <code>axis.text.x.top</code>,
<code>axis.text.y.left</code>, <code>axis.text.y.right</code>). <code style="white-space: pre;">&#8288;axis.text.*.*&#8288;</code> inherits from
<code style="white-space: pre;">&#8288;axis.text.*&#8288;</code> which inherits from <code>axis.text</code>, which in turn inherits
from <code>text</code></p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_axis.ticks">axis.ticks</code>, <code id="scheme_theme_+3A_axis.ticks.x">axis.ticks.x</code>, <code id="scheme_theme_+3A_axis.ticks.x.top">axis.ticks.x.top</code>, <code id="scheme_theme_+3A_axis.ticks.x.bottom">axis.ticks.x.bottom</code>, <code id="scheme_theme_+3A_axis.ticks.y">axis.ticks.y</code>, <code id="scheme_theme_+3A_axis.ticks.y.left">axis.ticks.y.left</code>, <code id="scheme_theme_+3A_axis.ticks.y.right">axis.ticks.y.right</code>, <code id="scheme_theme_+3A_axis.ticks.theta">axis.ticks.theta</code>, <code id="scheme_theme_+3A_axis.ticks.r">axis.ticks.r</code></td>
<td>
<p>tick marks along axes (<code><a href="ggplot2.html#topic+element_line">element_line()</a></code>). Specify all tick marks (<code>axis.ticks</code>),
ticks by plane (using <code>axis.ticks.x</code> or <code>axis.ticks.y</code>), or individually
for each axis (using <code>axis.ticks.x.bottom</code>, <code>axis.ticks.x.top</code>,
<code>axis.ticks.y.left</code>, <code>axis.ticks.y.right</code>). <code style="white-space: pre;">&#8288;axis.ticks.*.*&#8288;</code> inherits from
<code style="white-space: pre;">&#8288;axis.ticks.*&#8288;</code> which inherits from <code>axis.ticks</code>, which in turn inherits
from <code>line</code></p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_axis.minor.ticks.x.top">axis.minor.ticks.x.top</code>, <code id="scheme_theme_+3A_axis.minor.ticks.x.bottom">axis.minor.ticks.x.bottom</code>, <code id="scheme_theme_+3A_axis.minor.ticks.y.left">axis.minor.ticks.y.left</code>, <code id="scheme_theme_+3A_axis.minor.ticks.y.right">axis.minor.ticks.y.right</code>, <code id="scheme_theme_+3A_axis.minor.ticks.theta">axis.minor.ticks.theta</code>, <code id="scheme_theme_+3A_axis.minor.ticks.r">axis.minor.ticks.r</code></td>
<td>
<p>minor tick marks along axes (<code><a href="ggplot2.html#topic+element_line">element_line()</a></code>). <code style="white-space: pre;">&#8288;axis.minor.ticks.*.*&#8288;</code>
inherit from the corresponding major ticks <code style="white-space: pre;">&#8288;axis.ticks.*.*&#8288;</code>.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_axis.ticks.length">axis.ticks.length</code>, <code id="scheme_theme_+3A_axis.ticks.length.x">axis.ticks.length.x</code>, <code id="scheme_theme_+3A_axis.ticks.length.x.top">axis.ticks.length.x.top</code>, <code id="scheme_theme_+3A_axis.ticks.length.x.bottom">axis.ticks.length.x.bottom</code>, <code id="scheme_theme_+3A_axis.ticks.length.y">axis.ticks.length.y</code>, <code id="scheme_theme_+3A_axis.ticks.length.y.left">axis.ticks.length.y.left</code>, <code id="scheme_theme_+3A_axis.ticks.length.y.right">axis.ticks.length.y.right</code>, <code id="scheme_theme_+3A_axis.ticks.length.theta">axis.ticks.length.theta</code>, <code id="scheme_theme_+3A_axis.ticks.length.r">axis.ticks.length.r</code></td>
<td>
<p>length of tick marks (<code>unit</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_axis.minor.ticks.length">axis.minor.ticks.length</code>, <code id="scheme_theme_+3A_axis.minor.ticks.length.x">axis.minor.ticks.length.x</code>, <code id="scheme_theme_+3A_axis.minor.ticks.length.x.top">axis.minor.ticks.length.x.top</code>, <code id="scheme_theme_+3A_axis.minor.ticks.length.x.bottom">axis.minor.ticks.length.x.bottom</code>, <code id="scheme_theme_+3A_axis.minor.ticks.length.y">axis.minor.ticks.length.y</code>, <code id="scheme_theme_+3A_axis.minor.ticks.length.y.left">axis.minor.ticks.length.y.left</code>, <code id="scheme_theme_+3A_axis.minor.ticks.length.y.right">axis.minor.ticks.length.y.right</code>, <code id="scheme_theme_+3A_axis.minor.ticks.length.theta">axis.minor.ticks.length.theta</code>, <code id="scheme_theme_+3A_axis.minor.ticks.length.r">axis.minor.ticks.length.r</code></td>
<td>
<p>length of minor tick marks (<code>unit</code>), or relative to <code>axis.ticks.length</code> when provided with <code>rel()</code>.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_axis.line">axis.line</code>, <code id="scheme_theme_+3A_axis.line.x">axis.line.x</code>, <code id="scheme_theme_+3A_axis.line.x.top">axis.line.x.top</code>, <code id="scheme_theme_+3A_axis.line.x.bottom">axis.line.x.bottom</code>, <code id="scheme_theme_+3A_axis.line.y">axis.line.y</code>, <code id="scheme_theme_+3A_axis.line.y.left">axis.line.y.left</code>, <code id="scheme_theme_+3A_axis.line.y.right">axis.line.y.right</code>, <code id="scheme_theme_+3A_axis.line.theta">axis.line.theta</code>, <code id="scheme_theme_+3A_axis.line.r">axis.line.r</code></td>
<td>
<p>lines along axes (<code><a href="ggplot2.html#topic+element_line">element_line()</a></code>). Specify lines along all axes (<code>axis.line</code>),
lines for each plane (using <code>axis.line.x</code> or <code>axis.line.y</code>), or individually
for each axis (using <code>axis.line.x.bottom</code>, <code>axis.line.x.top</code>,
<code>axis.line.y.left</code>, <code>axis.line.y.right</code>). <code style="white-space: pre;">&#8288;axis.line.*.*&#8288;</code> inherits from
<code style="white-space: pre;">&#8288;axis.line.*&#8288;</code> which inherits from <code>axis.line</code>, which in turn inherits
from <code>line</code></p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.background">legend.background</code></td>
<td>
<p>background of legend (<code><a href="ggplot2.html#topic+element_rect">element_rect()</a></code>; inherits
from <code>rect</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.margin">legend.margin</code></td>
<td>
<p>the margin around each legend (<code><a href="ggplot2.html#topic+margin">margin()</a></code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.spacing">legend.spacing</code>, <code id="scheme_theme_+3A_legend.spacing.x">legend.spacing.x</code>, <code id="scheme_theme_+3A_legend.spacing.y">legend.spacing.y</code></td>
<td>
<p>the spacing between legends (<code>unit</code>). <code>legend.spacing.x</code> &amp; <code>legend.spacing.y</code>
inherit from <code>legend.spacing</code> or can be specified separately</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.key">legend.key</code></td>
<td>
<p>background underneath legend keys (<code><a href="ggplot2.html#topic+element_rect">element_rect()</a></code>;
inherits from <code>rect</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.key.size">legend.key.size</code>, <code id="scheme_theme_+3A_legend.key.height">legend.key.height</code>, <code id="scheme_theme_+3A_legend.key.width">legend.key.width</code></td>
<td>
<p>size of legend keys (<code>unit</code>); key background height &amp; width inherit from
<code>legend.key.size</code> or can be specified separately</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.key.spacing">legend.key.spacing</code>, <code id="scheme_theme_+3A_legend.key.spacing.x">legend.key.spacing.x</code>, <code id="scheme_theme_+3A_legend.key.spacing.y">legend.key.spacing.y</code></td>
<td>
<p>spacing
between legend keys given as a <code>unit</code>. Spacing in the horizontal (x) and
vertical (y) direction inherit from <code>legend.key.spacing</code> or can be
specified separately.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.frame">legend.frame</code></td>
<td>
<p>frame drawn around the bar (<code><a href="ggplot2.html#topic+element_rect">element_rect()</a></code>).</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.ticks">legend.ticks</code></td>
<td>
<p>tick marks shown along bars or axes (<code><a href="ggplot2.html#topic+element_line">element_line()</a></code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.ticks.length">legend.ticks.length</code></td>
<td>
<p>length of tick marks in legend (<code>unit</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.axis.line">legend.axis.line</code></td>
<td>
<p>lines along axes in legends (<code><a href="ggplot2.html#topic+element_line">element_line()</a></code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.text">legend.text</code></td>
<td>
<p>legend item labels (<code><a href="ggplot2.html#topic+element_text">element_text()</a></code>; inherits from
<code>text</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.text.position">legend.text.position</code></td>
<td>
<p>placement of legend text relative to legend keys
or bars (&quot;top&quot;, &quot;right&quot;, &quot;bottom&quot; or &quot;left&quot;). The legend text placement
might be incompatible with the legend's direction for some guides.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.title">legend.title</code></td>
<td>
<p>title of legend (<code><a href="ggplot2.html#topic+element_text">element_text()</a></code>; inherits from
<code>title</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.title.position">legend.title.position</code></td>
<td>
<p>placement of legend title relative to the main
legend (&quot;top&quot;, &quot;right&quot;, &quot;bottom&quot; or &quot;left&quot;).</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.position">legend.position</code></td>
<td>
<p>the default position of legends (&quot;none&quot;, &quot;left&quot;,
&quot;right&quot;, &quot;bottom&quot;, &quot;top&quot;, &quot;inside&quot;)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.position.inside">legend.position.inside</code></td>
<td>
<p>A numeric vector of length two setting the
placement of legends that have the <code>"inside"</code> position.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.direction">legend.direction</code></td>
<td>
<p>layout of items in legends (&quot;horizontal&quot; or
&quot;vertical&quot;)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.byrow">legend.byrow</code></td>
<td>
<p>whether the legend-matrix is filled by columns
(<code>FALSE</code>, the default) or by rows (<code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.justification">legend.justification</code></td>
<td>
<p>anchor point for positioning legend inside plot
(&quot;center&quot; or two-element numeric vector) or the justification according to
the plot area when positioned outside the plot</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.justification.top">legend.justification.top</code>, <code id="scheme_theme_+3A_legend.justification.bottom">legend.justification.bottom</code>, <code id="scheme_theme_+3A_legend.justification.left">legend.justification.left</code>, <code id="scheme_theme_+3A_legend.justification.right">legend.justification.right</code>, <code id="scheme_theme_+3A_legend.justification.inside">legend.justification.inside</code></td>
<td>
<p>Same as <code>legend.justification</code> but specified per <code>legend.position</code> option.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.location">legend.location</code></td>
<td>
<p>Relative placement of legends outside the plot as a
string. Can be <code>"panel"</code> (default) to align legends to the panels or
<code>"plot"</code> to align legends to the plot as a whole.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.box">legend.box</code></td>
<td>
<p>arrangement of multiple legends (&quot;horizontal&quot; or
&quot;vertical&quot;)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.box.just">legend.box.just</code></td>
<td>
<p>justification of each legend within the overall
bounding box, when there are multiple legends (&quot;top&quot;, &quot;bottom&quot;, &quot;left&quot;, or
&quot;right&quot;)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.box.margin">legend.box.margin</code></td>
<td>
<p>margins around the full legend area, as specified
using <code><a href="ggplot2.html#topic+margin">margin()</a></code></p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.box.background">legend.box.background</code></td>
<td>
<p>background of legend area (<code><a href="ggplot2.html#topic+element_rect">element_rect()</a></code>;
inherits from <code>rect</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_legend.box.spacing">legend.box.spacing</code></td>
<td>
<p>The spacing between the plotting area and the
legend box (<code>unit</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_panel.background">panel.background</code></td>
<td>
<p>background of plotting area, drawn underneath plot
(<code><a href="ggplot2.html#topic+element_rect">element_rect()</a></code>; inherits from <code>rect</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_panel.border">panel.border</code></td>
<td>
<p>border around plotting area, drawn on top of plot so that
it covers tick marks and grid lines. This should be used with
<code>fill = NA</code>
(<code><a href="ggplot2.html#topic+element_rect">element_rect()</a></code>; inherits from <code>rect</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_panel.spacing">panel.spacing</code>, <code id="scheme_theme_+3A_panel.spacing.x">panel.spacing.x</code>, <code id="scheme_theme_+3A_panel.spacing.y">panel.spacing.y</code></td>
<td>
<p>spacing between facet
panels (<code>unit</code>). <code>panel.spacing.x</code> &amp; <code>panel.spacing.y</code> inherit from <code>panel.spacing</code>
or can be specified separately.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_panel.grid">panel.grid</code>, <code id="scheme_theme_+3A_panel.grid.major">panel.grid.major</code>, <code id="scheme_theme_+3A_panel.grid.minor">panel.grid.minor</code>, <code id="scheme_theme_+3A_panel.grid.major.x">panel.grid.major.x</code>, <code id="scheme_theme_+3A_panel.grid.major.y">panel.grid.major.y</code>, <code id="scheme_theme_+3A_panel.grid.minor.x">panel.grid.minor.x</code>, <code id="scheme_theme_+3A_panel.grid.minor.y">panel.grid.minor.y</code></td>
<td>
<p>grid lines (<code><a href="ggplot2.html#topic+element_line">element_line()</a></code>). Specify major grid lines,
or minor grid lines separately (using <code>panel.grid.major</code> or <code>panel.grid.minor</code>)
or individually for each axis (using <code>panel.grid.major.x</code>, <code>panel.grid.minor.x</code>,
<code>panel.grid.major.y</code>, <code>panel.grid.minor.y</code>).  Y axis grid lines are horizontal
and x axis grid lines are vertical. <code style="white-space: pre;">&#8288;panel.grid.*.*&#8288;</code> inherits from
<code style="white-space: pre;">&#8288;panel.grid.*&#8288;</code> which inherits from <code>panel.grid</code>, which in turn inherits
from <code>line</code></p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_panel.ontop">panel.ontop</code></td>
<td>
<p>option to place the panel (background, gridlines) over
the data layers (<code>logical</code>). Usually used with a transparent or blank
<code>panel.background</code>.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_plot.background">plot.background</code></td>
<td>
<p>background of the entire plot (<code><a href="ggplot2.html#topic+element_rect">element_rect()</a></code>;
inherits from <code>rect</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_plot.title">plot.title</code></td>
<td>
<p>plot title (text appearance) (<code><a href="ggplot2.html#topic+element_text">element_text()</a></code>; inherits
from <code>title</code>) left-aligned by default</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_plot.title.position">plot.title.position</code>, <code id="scheme_theme_+3A_plot.caption.position">plot.caption.position</code></td>
<td>
<p>Alignment of the plot title/subtitle
and caption.  The setting for <code>plot.title.position</code> applies to both
the title and the subtitle. A value of &quot;panel&quot; (the default) means that
titles and/or caption are aligned to the plot panels. A value of &quot;plot&quot; means
that titles and/or caption are aligned to the entire plot (minus any space
for margins and plot tag).</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_plot.subtitle">plot.subtitle</code></td>
<td>
<p>plot subtitle (text appearance) (<code><a href="ggplot2.html#topic+element_text">element_text()</a></code>;
inherits from <code>title</code>) left-aligned by default</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_plot.caption">plot.caption</code></td>
<td>
<p>caption below the plot (text appearance)
(<code><a href="ggplot2.html#topic+element_text">element_text()</a></code>; inherits from <code>title</code>) right-aligned by default</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_plot.tag">plot.tag</code></td>
<td>
<p>upper-left label to identify a plot (text appearance)
(<code><a href="ggplot2.html#topic+element_text">element_text()</a></code>; inherits from <code>title</code>) left-aligned by default</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_plot.tag.position">plot.tag.position</code></td>
<td>
<p>The position of the tag as a string (&quot;topleft&quot;,
&quot;top&quot;, &quot;topright&quot;, &quot;left&quot;, &quot;right&quot;, &quot;bottomleft&quot;, &quot;bottom&quot;, &quot;bottomright&quot;)
or a coordinate. If a coordinate, can be a numeric vector of length 2 to
set the x,y-coordinate relative to the whole plot. The coordinate option
is unavailable for <code>plot.tag.location = "margin"</code>.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_plot.tag.location">plot.tag.location</code></td>
<td>
<p>The placement of the tag as a string, one of
<code>"panel"</code>, <code>"plot"</code> or <code>"margin"</code>. Respectively, these will place the tag
inside the panel space, anywhere in the plot as a whole, or in the margin
around the panel space.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_plot.margin">plot.margin</code></td>
<td>
<p>margin around entire plot (<code>unit</code> with the sizes of
the top, right, bottom, and left margins)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_strip.background">strip.background</code>, <code id="scheme_theme_+3A_strip.background.x">strip.background.x</code>, <code id="scheme_theme_+3A_strip.background.y">strip.background.y</code></td>
<td>
<p>background of facet labels (<code><a href="ggplot2.html#topic+element_rect">element_rect()</a></code>;
inherits from <code>rect</code>). Horizontal facet background (<code>strip.background.x</code>)
&amp; vertical facet background (<code>strip.background.y</code>) inherit from
<code>strip.background</code> or can be specified separately</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_strip.clip">strip.clip</code></td>
<td>
<p>should strip background edges and strip labels be clipped
to the extend of the strip background? Options are <code>"on"</code> to clip, <code>"off"</code>
to disable clipping or <code>"inherit"</code> (default) to take the clipping setting
from the parent viewport.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_strip.placement">strip.placement</code></td>
<td>
<p>placement of strip with respect to axes,
either &quot;inside&quot; or &quot;outside&quot;. Only important when axes and strips are
on the same side of the plot.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_strip.text">strip.text</code>, <code id="scheme_theme_+3A_strip.text.x">strip.text.x</code>, <code id="scheme_theme_+3A_strip.text.y">strip.text.y</code>, <code id="scheme_theme_+3A_strip.text.x.top">strip.text.x.top</code>, <code id="scheme_theme_+3A_strip.text.x.bottom">strip.text.x.bottom</code>, <code id="scheme_theme_+3A_strip.text.y.left">strip.text.y.left</code>, <code id="scheme_theme_+3A_strip.text.y.right">strip.text.y.right</code></td>
<td>
<p>facet labels (<code><a href="ggplot2.html#topic+element_text">element_text()</a></code>; inherits from  <code>text</code>). Horizontal facet labels (<code>strip.text.x</code>) &amp; vertical
facet labels (<code>strip.text.y</code>) inherit from <code>strip.text</code> or can be specified
separately. Facet strips have dedicated position-dependent theme elements
(<code>strip.text.x.top</code>, <code>strip.text.x.bottom</code>, <code>strip.text.y.left</code>, <code>strip.text.y.right</code>)
that inherit from <code>strip.text.x</code> and <code>strip.text.y</code>, respectively.
As a consequence, some theme stylings need to be applied to
the position-dependent elements rather than to the parent elements</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_strip.switch.pad.grid">strip.switch.pad.grid</code></td>
<td>
<p>space between strips and axes when strips are
switched (<code>unit</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_strip.switch.pad.wrap">strip.switch.pad.wrap</code></td>
<td>
<p>space between strips and axes when strips are
switched (<code>unit</code>)</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_complete">complete</code></td>
<td>
<p>set this to <code>TRUE</code> if this is a complete theme, such as
the one returned by <code><a href="ggplot2.html#topic+theme_grey">theme_grey()</a></code>. Complete themes behave
differently when added to a ggplot object. Also, when setting
<code>complete = TRUE</code> all elements will be set to inherit from blank
elements.</p>
</td></tr>
<tr><td><code id="scheme_theme_+3A_validate">validate</code></td>
<td>
<p><code>TRUE</code> to run <code>validate_element()</code>, <code>FALSE</code> to bypass checks.</p>
</td></tr>
</table>


<h3>Theme inheritance</h3>

<p>Theme elements inherit properties from other theme elements hierarchically.
For example, <code>axis.title.x.bottom</code> inherits from <code>axis.title.x</code> which inherits
from <code>axis.title</code>, which in turn inherits from <code>text</code>. All text elements inherit
directly or indirectly from <code>text</code>; all lines inherit from
<code>line</code>, and all rectangular objects inherit from <code>rect</code>.
This means that you can modify the appearance of multiple elements by
setting a single high-level component.
</p>
<p>Learn more about setting these aesthetics in <code>vignette("ggplot2-specs")</code>.
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic++2B.gg">+.gg()</a></code> and <a href="ggplot2.html#topic++25+2Breplace+25">%+replace%</a>,
<code><a href="ggplot2.html#topic+element_blank">element_blank()</a></code>, <code><a href="ggplot2.html#topic+element_line">element_line()</a></code>,
<code><a href="ggplot2.html#topic+element_rect">element_rect()</a></code>, and <code><a href="ggplot2.html#topic+element_text">element_text()</a></code> for
details of the specific theme elements.
</p>
<p>The <a href="https://ggplot2-book.org/themes#modifying-theme-components">modifying theme components</a> and <a href="https://ggplot2-book.org/themes#sec-theme-elements">theme elements sections</a> of the online ggplot2 book.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
small_mat &lt;- matrix(rnorm(56), nrow = 8)
ggheatmap(small_mat) +
    scheme_theme(plot.background = element_rect(fill = "red"))

# `scheme_theme()` serves as the default theme and will always be
# overridden by any `theme()` settings applied directly to the plot
ggheatmap(small_mat) +
    theme(plot.background = element_rect(fill = "blue")) +
    scheme_theme(plot.background = element_rect(fill = "red"))

</code></pre>

<hr>
<h2 id='show+2Cggalign_plot-method'>Show <code>ggalign_plot</code> information</h2><span id='topic+show+2Cggalign_plot-method'></span>

<h3>Description</h3>

<p>Show <code>ggalign_plot</code> information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ggalign_plot'
show(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="show+2B2Cggalign_plot-method_+3A_object">object</code></td>
<td>
<p>A <code>ggalign_plot</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input invisiblely.
</p>

<hr>
<h2 id='show+2CLayoutProto-method'>Print Layout object</h2><span id='topic+show+2CLayoutProto-method'></span>

<h3>Description</h3>

<p>Print Layout object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'LayoutProto'
show(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="show+2B2CLayoutProto-method_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+quad_layout">quad_layout()</a></code>/<code><a href="#topic+ggheatmap">ggheatmap()</a></code> or <code><a href="#topic+stack_layout">stack_layout()</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input invisiblely.
</p>

<hr>
<h2 id='stack_cross'>Arrange plots crosswise horizontally or vertically</h2><span id='topic+stack_cross'></span><span id='topic+stack_crossv'></span><span id='topic+stack_crossh'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>The <code>stack_cross</code> function is derived from <code><a href="#topic+stack_discrete">stack_discrete()</a></code> and allows
for different layout ordering indices within a single layout.
</p>
<p>Two aliases are provided for convenience:
</p>

<ul>
<li> <p><code>stack_crossv</code>: A special case of <code>stack_cross</code> that sets
<code>direction = "v"</code> for vertical alignment.
</p>
</li>
<li> <p><code>stack_crossh</code>: A special case of <code>stack_cross</code> that sets
<code>direction = "h"</code> for horizontal alignment.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>stack_cross(direction, data = NULL, ..., theme = NULL, sizes = NA)

stack_crossv(data = NULL, ...)

stack_crossh(data = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stack_cross_+3A_direction">direction</code></td>
<td>
<p>A string indicating the direction of the stack layout,
either <code>"h"</code>(<code>horizontal</code>) or <code>"v"</code>(<code>vertical</code>).</p>
</td></tr>
<tr><td><code id="stack_cross_+3A_data">data</code></td>
<td>
<p>Default dataset to use for the layout. If not specified, it must be supplied in each plot added to the layout, <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> will be used to
convert the data to a matrix.</p>
</td></tr>
<tr><td><code id="stack_cross_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>.</p>
</td></tr>
<tr><td><code id="stack_cross_+3A_theme">theme</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+theme">theme()</a></code> object used to customize various
elements of the layout, including <code>guides</code>, <code>title</code>, <code>subtitle</code>, <code>caption</code>,
<code>margins</code>, <code>panel.border</code>, and <code>background</code>. By default, the theme will
inherit from the parent <code>layout</code>. It also controls the panel spacing for all
plots in the layout.</p>
</td></tr>
<tr><td><code id="stack_cross_+3A_sizes">sizes</code></td>
<td>
<p>A numeric value or a <code><a href="grid.html#topic+unit">unit</a></code> object. When used for
the <code><a href="#topic+quad_layout">quad_layout()</a></code> annotation, it must be of length <code>1</code>. When used in the
<code><a href="#topic+stack_layout">stack_layout()</a></code> with a nested <code><a href="#topic+quad_layout">quad_layout()</a></code>, it should be of length
<code>3</code>, specifying the relative heights (for <code>direction = "h"</code>) or widths (for
<code>direction = "v"</code>) to be applied to the layout.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ggcross">ggcross()</a></code>
</p>

<hr>
<h2 id='stack_layout'>Arrange plots horizontally or vertically</h2><span id='topic+stack_layout'></span><span id='topic+stack_horizontal'></span><span id='topic+stack_vertical'></span><span id='topic+stack_discrete'></span><span id='topic+stack_align'></span><span id='topic+stack_discretev'></span><span id='topic+stack_alignv'></span><span id='topic+stack_discreteh'></span><span id='topic+stack_alignh'></span><span id='topic+stack_continuous'></span><span id='topic+stack_free'></span><span id='topic+stack_continuousv'></span><span id='topic+stack_freev'></span><span id='topic+stack_continuoush'></span><span id='topic+stack_freeh'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>If <code>limits</code> is provided, a continuous variable will be required and aligned
in the direction specified (<code>stack_continuous</code>). Otherwise, a discrete
variable will be required and aligned (<code>stack_discrete</code>).
</p>
<p>Several aliases are provided for convenience:
</p>

<ul>
<li> <p><code>stack_vertical</code>: A special case of <code>stack_layout</code> that sets <code>direction = "v"</code>.
</p>
</li>
<li> <p><code>stack_horizontal</code>: A special case of <code>stack_layout</code> that sets <code>direction = "h"</code>.
</p>
</li>
<li> <p><code>stack_discretev</code>: A special case of <code>stack_discrete</code> that sets <code>direction = "v"</code>.
</p>
</li>
<li> <p><code>stack_discreteh</code>: A special case of <code>stack_discrete</code> that sets <code>direction = "h"</code>.
</p>
</li>
<li> <p><code>stack_continuousv()</code>: A special case of <code>stack_free</code> that sets <code>direction = "v"</code>.
</p>
</li>
<li> <p><code>stack_continuoush()</code>: A special case of <code>stack_free</code> that sets <code>direction = "h"</code>.
</p>
</li></ul>

<p>For historical reasons, the following aliases are available:
</p>

<ul>
<li> <p><code>stack_align</code> is an alias for <code>stack_discrete</code>.
</p>
</li>
<li> <p><code>stack_alignv</code> is an alias for <code>stack_discretev</code>.
</p>
</li>
<li> <p><code>stack_alignh</code> is an alias for <code>stack_discreteh</code>.
</p>
</li>
<li> <p><code>stack_free</code> is an alias for <code>stack_continuous</code>.
</p>
</li>
<li> <p><code>stack_freev</code> is an alias for <code>stack_continuousv</code>.
</p>
</li>
<li> <p><code>stack_freeh</code> is an alias for <code>stack_continuoush</code>.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>stack_layout(
  direction,
  data = NULL,
  ...,
  theme = NULL,
  sizes = NA,
  limits = waiver()
)

stack_horizontal(data = NULL, ..., limits = waiver())

stack_vertical(data = NULL, ..., limits = waiver())

stack_discrete(direction, data = NULL, ..., theme = NULL, sizes = NA)

stack_discretev(data = NULL, ...)

stack_discreteh(data = NULL, ...)

stack_continuous(
  direction,
  data = NULL,
  ...,
  limits = NULL,
  theme = NULL,
  sizes = NA
)

stack_continuousv(data = NULL, ...)

stack_continuoush(data = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stack_layout_+3A_direction">direction</code></td>
<td>
<p>A string indicating the direction of the stack layout,
either <code>"h"</code>(<code>horizontal</code>) or <code>"v"</code>(<code>vertical</code>).</p>
</td></tr>
<tr><td><code id="stack_layout_+3A_data">data</code></td>
<td>
<p>Default dataset to use for the layout. If not specified, it must be supplied in each plot added to the layout:
</p>

<ul>
<li><p> If <code>limits</code> is not provided, <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> will be used to get a
matrix.
</p>
</li>
<li><p> If <code>limits</code> is specified, <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> will be used to get a
data frame.
</p>
</li></ul>
</td></tr>
<tr><td><code id="stack_layout_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> or
<code><a href="#topic+fortify_matrix">fortify_matrix()</a></code>.</p>
</td></tr>
<tr><td><code id="stack_layout_+3A_theme">theme</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+theme">theme()</a></code> object used to customize various
elements of the layout, including <code>guides</code>, <code>title</code>, <code>subtitle</code>, <code>caption</code>,
<code>margins</code>, <code>panel.border</code>, and <code>background</code>. By default, the theme will
inherit from the parent <code>layout</code>. It also controls the panel spacing for all
plots in the layout.</p>
</td></tr>
<tr><td><code id="stack_layout_+3A_sizes">sizes</code></td>
<td>
<p>A numeric value or a <code><a href="grid.html#topic+unit">unit</a></code> object. When used for
the <code><a href="#topic+quad_layout">quad_layout()</a></code> annotation, it must be of length <code>1</code>. When used in the
<code><a href="#topic+stack_layout">stack_layout()</a></code> with a nested <code><a href="#topic+quad_layout">quad_layout()</a></code>, it should be of length
<code>3</code>, specifying the relative heights (for <code>direction = "h"</code>) or widths (for
<code>direction = "v"</code>) to be applied to the layout.</p>
</td></tr>
<tr><td><code id="stack_layout_+3A_limits">limits</code></td>
<td>
<p>A <code><a href="#topic+continuous_limits">continuous_limits()</a></code> object specifying the left/lower
limit and the right/upper limit of the scale. Used to align the continuous
axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>StackLayout</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
small_mat &lt;- matrix(rnorm(56), nrow = 7L)

stack_horizontal(small_mat) + align_dendro()

# this is the same with:
stack_discrete("h", small_mat) + align_dendro()

stack_discreteh(small_mat) + align_dendro()

# For vertical layout:
stack_vertical(small_mat) + align_dendro()

</code></pre>

<hr>
<h2 id='stack_switch'>Determine the active context of stack layout</h2><span id='topic+stack_switch'></span><span id='topic+stack_active'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p><code>stack_active</code> is an alias for <code>stack_switch()</code>, which sets <code>what = NULL</code> by
default.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stack_switch(sizes = NULL, what = waiver(), ...)

stack_active(sizes = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stack_switch_+3A_sizes">sizes</code></td>
<td>
<p>A numeric value or a <code><a href="grid.html#topic+unit">unit</a></code> object. When used for
the <code><a href="#topic+quad_layout">quad_layout()</a></code> annotation, it must be of length <code>1</code>. When used in the
<code><a href="#topic+stack_layout">stack_layout()</a></code> with a nested <code><a href="#topic+quad_layout">quad_layout()</a></code>, it should be of length
<code>3</code>, specifying the relative heights (for <code>direction = "h"</code>) or widths (for
<code>direction = "v"</code>) to be applied to the layout.</p>
</td></tr>
<tr><td><code id="stack_switch_+3A_what">what</code></td>
<td>
<p>What should get activated for the stack layout?
A single number or string of the plot elements in the layout. If <code>NULL</code>, will remove any active context, this is useful when the active context is a
<code><a href="#topic+quad_layout">quad_layout()</a></code> object, where any <code style="white-space: pre;">&#8288;align_*()&#8288;</code> will be added to the
<code><a href="#topic+quad_layout">quad_layout()</a></code>. By removing the active context, we can add <code style="white-space: pre;">&#8288;align_*()&#8288;</code>
into the <code><a href="#topic+stack_layout">stack_layout()</a></code>.</p>
</td></tr>
<tr><td><code id="stack_switch_+3A_...">...</code></td>
<td>
<p>These dots are for future extensions and must be empty.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>stack_switch</code> object which can be added to <code><a href="#topic+stack_layout">stack_layout()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stack_discrete("h", matrix(1:9, nrow = 3L)) +
    ggheatmap() +
    # ggheamtap will set the active context, directing following addition
    # into the heatmap plot area. To remove the heatmap active context,
    # we can use `stack_active()` which will direct subsequent addition into
    # the stack
    stack_active() +
    # here we add a dendrogram to the stack.
    align_dendro()
</code></pre>

<hr>
<h2 id='theme_no_axes'>Remove axis elements</h2><span id='topic+theme_no_axes'></span>

<h3>Description</h3>

<p>Remove axis elements
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_no_axes(
  axes = "xy",
  text = TRUE,
  ticks = TRUE,
  title = TRUE,
  line = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="theme_no_axes_+3A_axes">axes</code></td>
<td>
<p>Which axes elements should be removed? A string containing
one or more of <code>"t"</code>, <code>"l"</code>, <code>"b"</code>, <code>"r"</code>, <code>"x"</code>, and <code>"y"</code>.</p>
</td></tr>
<tr><td><code id="theme_no_axes_+3A_text">text</code></td>
<td>
<p>If <code>TRUE</code>, will remove the axis labels.</p>
</td></tr>
<tr><td><code id="theme_no_axes_+3A_ticks">ticks</code></td>
<td>
<p>If <code>TRUE</code>, will remove the axis ticks.</p>
</td></tr>
<tr><td><code id="theme_no_axes_+3A_title">title</code></td>
<td>
<p>If <code>TRUE</code>, will remove the axis title.</p>
</td></tr>
<tr><td><code id="theme_no_axes_+3A_line">line</code></td>
<td>
<p>If <code>TRUE</code>, will remove the axis line.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+theme">theme()</a></code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p &lt;- ggplot() +
    geom_point(aes(x = wt, y = qsec), data = mtcars)
p + theme_no_axes()
p + theme_no_axes("b")
p + theme_no_axes("l")
</code></pre>

<hr>
<h2 id='tune'>Change the shape of the input object</h2><span id='topic+tune'></span>

<h3>Description</h3>

<p>Change the shape of the input object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tune(data, shape = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tune_+3A_data">data</code></td>
<td>
<p>An R object.</p>
</td></tr>
<tr><td><code id="tune_+3A_shape">shape</code></td>
<td>
<p>Usually <code>NULL</code> or a string, specifying the new shape for the
object. Refer to the detailed method for allowed values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In most cases, <code><a href="#topic+fortify_matrix">fortify_matrix()</a></code> or <code><a href="#topic+fortify_data_frame">fortify_data_frame()</a></code> provide full
support for transforming objects. However, some objects may require two
completely different approaches to be fortified. The <code>tune</code> function acts as
a helper to create a new class tailored for these objects.
</p>


<h3><code>tune</code> method collections</h3>


<ul>
<li> <p><code><a href="#topic+tune.list">tune.list()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+tune.MAF">tune.MAF()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+tune.matrix">tune.matrix()</a></code>
</p>
</li></ul>


<hr>
<h2 id='tune.list'>Convert the shape of a list for fortify method</h2><span id='topic+tune.list'></span>

<h3>Description</h3>

<p>Convert the shape of a list for fortify method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'list'
tune(data, shape = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tune.list_+3A_data">data</code></td>
<td>
<p>A list</p>
</td></tr>
<tr><td><code id="tune.list_+3A_shape">shape</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+fortify_matrix.list_upset">fortify_matrix.list_upset()</a></code>
</p>
<p>Other tune: 
<code><a href="#topic+tune.MAF">tune.MAF</a>()</code>,
<code><a href="#topic+tune.matrix">tune.matrix</a>()</code>
</p>

<hr>
<h2 id='tune.MAF'>Convert the shape of a MAF for fortify method</h2><span id='topic+tune.MAF'></span>

<h3>Description</h3>

<p>Convert the shape of a MAF for fortify method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MAF'
tune(data, shape = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tune.MAF_+3A_data">data</code></td>
<td>
<p>A <code><a href="maftools.html#topic+read.maf">MAF</a></code> object.</p>
</td></tr>
<tr><td><code id="tune.MAF_+3A_shape">shape</code></td>
<td>
<p>Not used currently.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+fortify_matrix.MAF_pathways">fortify_matrix.MAF_pathways()</a></code>
</p>
<p>Other tune: 
<code><a href="#topic+tune.list">tune.list</a>()</code>,
<code><a href="#topic+tune.matrix">tune.matrix</a>()</code>
</p>

<hr>
<h2 id='tune.matrix'>Convert the shape of a matrix for fortify method</h2><span id='topic+tune.matrix'></span>

<h3>Description</h3>

<p>Convert the shape of a matrix for fortify method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'matrix'
tune(data, shape)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tune.matrix_+3A_data">data</code></td>
<td>
<p>A matrix.</p>
</td></tr>
<tr><td><code id="tune.matrix_+3A_shape">shape</code></td>
<td>
<p>A string of <code>"upset"</code> or <code>"oncoplot"</code>.</p>
</td></tr>
</table>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+fortify_matrix.matrix">fortify_matrix.matrix()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_matrix.matrix_upset">fortify_matrix.matrix_upset()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fortify_matrix.matrix_oncoplot">fortify_matrix.matrix_oncoplot()</a></code>
</p>
</li></ul>

<p>Other tune: 
<code><a href="#topic+tune.MAF">tune.MAF</a>()</code>,
<code><a href="#topic+tune.list">tune.list</a>()</code>
</p>

<hr>
<h2 id='with_quad'>Modify operated Context in <code>quad_layout()</code></h2><span id='topic+with_quad'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>The <code>with_quad()</code> function modifies the application context of elements in
<code>ggheatmap()</code>/<code>quad_layout()</code>. It controls how objects like themes, scales,
or other plot modifications apply to specific annotation stacks or the main
plot without altering the currently active layout or plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>with_quad(x, position = waiver(), main = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="with_quad_+3A_x">x</code></td>
<td>
<p>An object which can be added to the ggplot, including
<strong>schemes</strong>. See <code><a href="#topic+scheme_align">scheme_align()</a></code>, <code><a href="#topic+scheme_data">scheme_data()</a></code>, and
<code><a href="#topic+scheme_theme">scheme_theme()</a></code></p>
</td></tr>
<tr><td><code id="with_quad_+3A_position">position</code></td>
<td>
<p>A string specifying one or more positions-
<code>"t"</code>, <code>"l"</code>, <code>"b"</code>, and <code>"r"</code>- to indicate the annotation stack context for <code>x</code>. If
<code>NULL</code>, will change the operated context to the <code>quad_layout()</code> itself. For
default behaivours, see <code>details</code> section.</p>
</td></tr>
<tr><td><code id="with_quad_+3A_main">main</code></td>
<td>
<p>A single boolean value indicating whether <code>x</code> should apply to the
main plot, used only when <code>position</code> is not <code>NULL</code>. By default, if <code>position</code>
is <code>waiver()</code> and the active context of <code>quad_layout()</code> is an annotation
stack or the active context of <code>stack_layout()</code> is itself, <code>main</code> will be set
to <code>TRUE</code>; otherwise, it defaults to <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Default Behavior when adding object wrapped with <code>with_quad()</code>:
</p>
<p>For <code>quad_layout()</code> object:
</p>

<ul>
<li><p> When <code>ggheatmap()</code>/<code>quad_layout()</code> has no active annotation stack, objects
added via <code>+</code> or <code>-</code> operate normally without <code>with_quad()</code>.
</p>
</li>
<li><p> When the active annotation stack is set, <code>with_quad()</code> ensures the applied
object also modifies:
</p>

<ul>
<li><p> The main plot (by default).
</p>
</li>
<li><p> Opposite annotation stacks when using <code>-</code>.
</p>
</li></ul>

</li></ul>

<p>For <code>stack_layout()</code> object:
</p>

<ul>
<li><p> When the active layout is the <code>stack_layout()</code> itself:
</p>

<ul>
<li> <p><code>-</code> operator will apply changes to all plots along the
<code>stack_layout()</code>, which means if the stack layout is in <code>horizontal</code>,
<code>-</code> operator will also add the element to the <code>left</code> and <code>right</code>
annotation, if the stack layout is in <code>vertical</code>, <code>-</code> operator will
also add element to the <code>top</code> and <code>bottom</code> annotation.
</p>
</li>
<li> <p><code>+</code> operator won't do anything special.
</p>
</li></ul>

</li>
<li><p> When the active layout is the nested <code>ggheatmap()</code>/<code>quad_layout()</code>, the
<code>+</code>/<code>-</code> operator applies the elements to this nested layout, following the
same principles as for <code>ggheatmap()</code>/<code>quad_layout()</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>The original object with an added attribute that sets the specified
context.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
small_mat &lt;- matrix(rnorm(56), nrow = 7)

# By wrapping object with `with_quad()`, the `+` operator will apply the
# object not only to the active plot in the annotation stack, but also to
# the main plot unless specified by `main` argument otherwise.
ggheatmap(small_mat) +
    # initialize the left annotation
    anno_left(size = 0.2) +
    align_dendro() +
    # apply the object not only to the active plot in the annotation stack,
    # but also to the main plot
    with_quad(theme(plot.background = element_rect(fill = "red")))

# the `-` operator will apply changes not only to the active annotation
# stack but also to the opposite one (i.e., bottom if top is active, and
# vice versa). The same principle applies to the left and right annotation.
ggheatmap(small_mat) +
    anno_left(size = 0.2) +
    align_dendro(aes(color = branch), k = 3L) +
    # Change the active layout to the left annotation
    anno_top(size = 0.2) +
    align_dendro(aes(color = branch), k = 3L) +
    anno_bottom(size = 0.2) +
    align_dendro(aes(color = branch), k = 3L) -
    # Modify the color scale of all plots in the bottom and the opposite
    # annotation, in this way, the `main` argument by default would be `TRUE`
    with_quad(scale_color_brewer(palette = "Dark2", name = "Top and bottom"))

# When the `position` argument is manually set, the
# default value of the `main` argument will be `FALSE`.
ggheatmap(small_mat) +
    anno_left(size = 0.2) +
    align_dendro(aes(color = branch), k = 3L) +
    anno_top(size = 0.2) +
    align_dendro(aes(color = branch), k = 3L) +
    anno_bottom(size = 0.2) +
    align_dendro(aes(color = branch), k = 3L) -
    # Modify the background of all plots in the left and top annotation
    with_quad(theme(plot.background = element_rect(fill = "red")), "tl")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
