<!DOCTYPE html><html lang="en"><head><title>Help for package ggenealogy</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ggenealogy}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#buildAncDesCoordDF'><p>Returns the coordinate positions of all ancestors and descendants of a variety.</p></a></li>
<li><a href='#buildAncDesTotalDF'><p>Returns data frame with plot coordinates of all ancestors and descendants of a variety.</p></a></li>
<li><a href='#buildAncList'><p>Returns the ancestors of a particular variety (if they exist).</p></a></li>
<li><a href='#buildDesList'><p>Returns the descendants of a particular variety (if they exist).</p></a></li>
<li><a href='#buildEdgeTotalDF'><p>Build the edges in the genealogy graph.</p></a></li>
<li><a href='#buildMinusPathDF'><p>Process the genealogy graph</p></a></li>
<li><a href='#buildPathDF'><p>Build data frame for path representation</p></a></li>
<li><a href='#buildPlotTotalDF'><p>Build all labels in the graph</p></a></li>
<li><a href='#buildSpreadTotalDF'><p>Build a data frame where the varieties are spread so they do not overlap</p></a></li>
<li><a href='#dfToIG'><p>Process the genealogy graph</p></a></li>
<li><a href='#getAncestors'><p>Returns a list of the ancestors of a particular variety (if they exist)</p></a></li>
<li><a href='#getBasicStatistics'><p>Determine basic statistics of the graph object</p></a></li>
<li><a href='#getBranchQual'><p>Descendant branch calculations for quantitative variable</p></a></li>
<li><a href='#getBranchQuant'><p>Descendant branch calculations for quantitative variable</p></a></li>
<li><a href='#getChild'><p>Returns the children of a particular variety (if they exist)</p></a></li>
<li><a href='#getDegree'><p>Determine the degree between two varieties</p></a></li>
<li><a href='#getDescendants'><p>Returns a list of the descendants of a particular variety (if they exist)</p></a></li>
<li><a href='#getEdges'><p>Returns edges (vertex names and edge weights) for the full genealogy</p></a></li>
<li><a href='#getNodes'><p>Returns the nodes for a full genealogy</p></a></li>
<li><a href='#getParent'><p>Returns the parents of a particular variety (if they exist)</p></a></li>
<li><a href='#getPath'><p>Determine the path between two varieties</p></a></li>
<li><a href='#getPathOnly'><p>Determine the path between two varieties</p></a></li>
<li><a href='#getVariable'><p>Determine the date of a variety</p></a></li>
<li><a href='#isChild'><p>Determine if a variety is a child of another</p></a></li>
<li><a href='#isParent'><p>Determine if a variety is a parent of another</p></a></li>
<li><a href='#nodeToDF'><p>Returns the data frame representation of all ancestors and descendants of a variety</p></a></li>
<li><a href='#plotAncDes'><p>Returns the image object to show the ancestors and descendants of a variety</p></a></li>
<li><a href='#plotDegMatrix'><p>Returns the image object to show the heat map of degrees between the inputted set of vertices</p></a></li>
<li><a href='#plotPath'><p>Construct the graphic object of the path</p></a></li>
<li><a href='#plotPathOnAll'><p>Plot a path between two vertices over the full genealogy</p></a></li>
<li><a href='#plotVariableMatrix'><p>Returns the image object to show the heat map of dates between the inputted set of vertices</p></a></li>
<li><a href='#sbGeneal'><p>Soybean data</p></a></li>
<li><a href='#statGeneal'><p>Academic statistics data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Title:</td>
<td>Visualization Tools for Genealogical Data</td>
</tr>
<tr>
<td>Description:</td>
<td>Methods for searching through genealogical data and displaying the results. Plotting algorithms assist with data exploration and publication-quality image generation. Includes interactive genealogy visualization tools. Provides parsing and calculation methods for variables in descendant branches of interest. Uses the Grammar of Graphics.</td>
</tr>
<tr>
<td>Author:</td>
<td>Lindsay Rutter, Susan Vanderplas, Di Cook</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lindsay Rutter &lt;ggenealogypackage@gmail.com&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2 (&ge; 2.2.0), igraph (&ge; 0.7.1), plyr (&ge; 1.8.1),
reshape2 (&ge; 1.4), plotly (&ge; 4.5.6)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>stringr (&ge; 0.6.2), knitr (&ge; 1.13), roxygen2 (&ge; 3.0.0),
dplyr (&ge; 0.5.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-21 15:39:35 UTC; suzu</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-21 16:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='buildAncDesCoordDF'>Returns the coordinate positions of all ancestors and descendants of a variety.</h2><span id='topic+buildAncDesCoordDF'></span>

<h3>Description</h3>

<p>Calculates coordinates to plot each ancestors and descendant of a variety in a lineage. The x and y values describe the coordinates of the label, while the xstart, ystart, xend, and yend values describe the edges of the label.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildAncDesCoordDF(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildAncDesCoordDF_+3A_df">df</code></td>
<td>
<p>the data frame of the ancestors and descendants of a variety (from function buildAncDesTotalDF)</p>
</td></tr>
</table>


<h3>References</h3>

<p><code><a href="utils.html#topic+CITATION">CITATION</a></code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+buildAncList">buildAncList</a></code> for information on determining ancestors
</p>
<p><code><a href="#topic+buildDesList">buildDesList</a></code> for information on determining descendants
</p>

<hr>
<h2 id='buildAncDesTotalDF'>Returns data frame with plot coordinates of all ancestors and descendants of a variety.</h2><span id='topic+buildAncDesTotalDF'></span>

<h3>Description</h3>

<p>Returns the data frame that includes labels and plot coordinates of all ancestors and descendants of a variety. Users can specify the maximum number of ancestors and descendants to display.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildAncDesTotalDF(v1, geneal, mAnc = 3, mDes = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildAncDesTotalDF_+3A_v1">v1</code></td>
<td>
<p>the label of the vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="buildAncDesTotalDF_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy (in data frame format)</p>
</td></tr>
<tr><td><code id="buildAncDesTotalDF_+3A_manc">mAnc</code></td>
<td>
<p>the maximum number of generations of ancestors of v1 to be displayed (in numeric format)</p>
</td></tr>
<tr><td><code id="buildAncDesTotalDF_+3A_mdes">mDes</code></td>
<td>
<p>the maximum number of generations of descendants of v1 to be displayed (in numeric format)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+buildAncList">buildAncList</a></code> for information on determining ancestors
</p>
<p><code><a href="#topic+buildDesList">buildDesList</a></code> for information on determining descendants
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
v1 &lt;- "Essex"
buildAncDesTotalDF(v1, sbGeneal)
</code></pre>

<hr>
<h2 id='buildAncList'>Returns the ancestors of a particular variety (if they exist).</h2><span id='topic+buildAncList'></span>

<h3>Description</h3>

<p>This function returns a nested list of the ancestors of the inputted variety.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildAncList(v1, geneal, gen = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildAncList_+3A_v1">v1</code></td>
<td>
<p>the label of the vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="buildAncList_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="buildAncList_+3A_gen">gen</code></td>
<td>
<p>the generation (note: This should be left as default, as any other input will not affect results anyway)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getParent">getParent</a></code> for information on determining parents
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
getParent("Essex", sbGeneal)
buildAncList("Essex", sbGeneal)
</code></pre>

<hr>
<h2 id='buildDesList'>Returns the descendants of a particular variety (if they exist).</h2><span id='topic+buildDesList'></span>

<h3>Description</h3>

<p>This function returns a nested list of the descendants of the inputted variety.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildDesList(v1, geneal, gen = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildDesList_+3A_v1">v1</code></td>
<td>
<p>the label of the vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="buildDesList_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="buildDesList_+3A_gen">gen</code></td>
<td>
<p>the generation (note: This should be left as default, as any other input will not affect results)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getChild">getChild</a></code> for information on determining children
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
getParent("Essex", sbGeneal)
buildDesList("Essex", sbGeneal, 3)
</code></pre>

<hr>
<h2 id='buildEdgeTotalDF'>Build the edges in the genealogy graph.</h2><span id='topic+buildEdgeTotalDF'></span>

<h3>Description</h3>

<p>This function takes the graph object and creates a data frame object of the edges between all parent-child relationships in the graph.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildEdgeTotalDF(geneal, ig, colName, bin = 12)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildEdgeTotalDF_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="buildEdgeTotalDF_+3A_ig">ig</code></td>
<td>
<p>the graph representation of the data genealogy (in igraph format)</p>
</td></tr>
<tr><td><code id="buildEdgeTotalDF_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the quantitative variable of interest (in character string format)</p>
</td></tr>
<tr><td><code id="buildEdgeTotalDF_+3A_bin">bin</code></td>
<td>
<p>the number of bins to determine the vertical positions of nodes (default is 12). For more information on choosing bin size, please visit the ggenealogy vignette.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+dfToIG">dfToIG</a></code> for information on producing ig from the genealogy
</p>
<p><a href="https://www.r-project.org">https://www.r-project.org</a> for iGraph information
</p>

<hr>
<h2 id='buildMinusPathDF'>Process the genealogy graph</h2><span id='topic+buildMinusPathDF'></span>

<h3>Description</h3>

<p>This function takes the spreadTotalDF object (from the buildSpreadTotalDF function) and the path object
as inputs. From these objects, it creates a data frame object of the label, x, and y values of all nodes
in the ful genealogy. However, the data frame object does not include the labels of the path varieties, as they
will be treated differently.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildMinusPathDF(path, geneal, ig, colName, colNameY, bin = 12)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildMinusPathDF_+3A_path">path</code></td>
<td>
<p>path as returned from getPath() or a vector of two variety names which exist in the ig object</p>
</td></tr>
<tr><td><code id="buildMinusPathDF_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="buildMinusPathDF_+3A_ig">ig</code></td>
<td>
<p>the graph representation of the data genealogy (in igraph format)</p>
</td></tr>
<tr><td><code id="buildMinusPathDF_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the quantitative variable of interest (in character string format)</p>
</td></tr>
<tr><td><code id="buildMinusPathDF_+3A_colnamey">colNameY</code></td>
<td>
<p>the name of the second optional column of the data frame that contains the second optional quantitative variable of interest (in character string format). This optional quantitative variable will be plotted on the vertical axis.</p>
</td></tr>
<tr><td><code id="buildMinusPathDF_+3A_bin">bin</code></td>
<td>
<p>the number of bins to determine the vertical positions of nodes (default is 12). For more information on choosing bin size, please visit the ggenealogy vignette.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="https://www.r-project.org">https://www.r-project.org</a> for iGraph information
</p>
<p><code><a href="#topic+getPath">getPath</a></code> for information on input path building
</p>

<hr>
<h2 id='buildPathDF'>Build data frame for path representation</h2><span id='topic+buildPathDF'></span>

<h3>Description</h3>

<p>This function builds a dataframe of information about the path object that can later be used
for visualization. The dataframe includes &quot;label&quot; (name of each variety) of each node,
&quot;x&quot; (the date of the variety, the x-axis value for which the label and
incoming/outgoing edges are centered), &quot;y&quot; (the y-axis value, which is the index
of the path, incremented by unity), &quot;xstart&quot; (the x-axis position of the 
outgoing edge (leaving to connect to the node at the next largest y-value)),
&quot;xend&quot; (the x-axis position of the outgoing edge (connected to the node at the
next largest y-value)), &quot;ystart&quot; (the y-axis position of the outgoing edge (leaving
to connect to the node at the next largest y-value), &quot;yend&quot; (the y-axis position
of the outgoing edge (connected to the node at the next largest y-value))).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildPathDF(path, geneal, colName, colNameY = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildPathDF_+3A_path">path</code></td>
<td>
<p>path object representing the path between two vertices</p>
</td></tr>
<tr><td><code id="buildPathDF_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="buildPathDF_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the quantitative variable of interest (in character string format)</p>
</td></tr>
<tr><td><code id="buildPathDF_+3A_colnamey">colNameY</code></td>
<td>
<p>the name of the second optional column of the data frame that contains the second optional quantitative variable of interest (in character string format). This optional quantitative variable will be plotted on the vertical axis.</p>
</td></tr>
</table>

<hr>
<h2 id='buildPlotTotalDF'>Build all labels in the graph</h2><span id='topic+buildPlotTotalDF'></span>

<h3>Description</h3>

<p>This function takes the spreadTotalDF object (from the buildSpreadTotalDF function) and the path object
as inputs. From these objects, it creates a data frame object of the text label positions for the
varieties in the path, as well as the edges only in the varieties in the path.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildPlotTotalDF(path, geneal, ig, colName, colNameY = "", bin = 12)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildPlotTotalDF_+3A_path">path</code></td>
<td>
<p>path as returned from getPath() or a vector of two variety names which exist in ig</p>
</td></tr>
<tr><td><code id="buildPlotTotalDF_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="buildPlotTotalDF_+3A_ig">ig</code></td>
<td>
<p>the graph representation of the data genealogy (in igraph format)</p>
</td></tr>
<tr><td><code id="buildPlotTotalDF_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the quantitative variable of interest (in character string format)</p>
</td></tr>
<tr><td><code id="buildPlotTotalDF_+3A_colnamey">colNameY</code></td>
<td>
<p>the name of the second optional column of the data frame that contains the second optional quantitative variable of interest (in character string format). This optional quantitative variable will be plotted on the vertical axis.</p>
</td></tr>
<tr><td><code id="buildPlotTotalDF_+3A_bin">bin</code></td>
<td>
<p>the number of bins to determine the vertical positions of nodes (default is 12). For more information on choosing bin size, please visit the ggenealogy vignette</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="https://www.r-project.org">https://www.r-project.org</a> for iGraph information
</p>
<p><a href="https://www.r-project.org">https://www.r-project.org</a> for iGraph information
</p>
<p><code><a href="#topic+getPath">getPath</a></code> for information on input path building
</p>

<hr>
<h2 id='buildSpreadTotalDF'>Build a data frame where the varieties are spread so they do not overlap</h2><span id='topic+buildSpreadTotalDF'></span>

<h3>Description</h3>

<p>Constructs a data frame object so that varieties are spread such that they do not overlap, even
though the x-axis position will represent dates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildSpreadTotalDF(geneal, ig, colName, bin = 12)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildSpreadTotalDF_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="buildSpreadTotalDF_+3A_ig">ig</code></td>
<td>
<p>the graph representation of the data genealogy (in igraph format)</p>
</td></tr>
<tr><td><code id="buildSpreadTotalDF_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the quantitative variable of interest (in character string format)</p>
</td></tr>
<tr><td><code id="buildSpreadTotalDF_+3A_bin">bin</code></td>
<td>
<p>the number of bins to determine the vertical positions of nodes (default is 12). For more information on choosing bin size, please visit the ggenealogy vignette</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="https://www.r-project.org">https://www.r-project.org</a> for iGraph information
</p>

<hr>
<h2 id='dfToIG'>Process the genealogy graph</h2><span id='topic+dfToIG'></span>

<h3>Description</h3>

<p>Processes the genealogy into an igraph object with appropriate vertex information, graph type, and edge weights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfToIG(geneal, vertexinfo = NULL, edgeweights = 1, isDirected = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dfToIG_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="dfToIG_+3A_vertexinfo">vertexinfo</code></td>
<td>
<p>(default NULL) either names of columns in the genealogy which should be added to the database as vertex information or a data frame with information for all vertices such that the first column contains vertex names.</p>
</td></tr>
<tr><td><code id="dfToIG_+3A_edgeweights">edgeweights</code></td>
<td>
<p>(default 1) name of a column which contains edge weights</p>
</td></tr>
<tr><td><code id="dfToIG_+3A_isdirected">isDirected</code></td>
<td>
<p>(default FALSE) should the graph be a directed graph?</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="https://www.r-project.org">https://www.r-project.org</a> for iGraph information
</p>

<hr>
<h2 id='getAncestors'>Returns a list of the ancestors of a particular variety (if they exist)</h2><span id='topic+getAncestors'></span>

<h3>Description</h3>

<p>This function returns a list of the ancestors of the inputted variety within and including a given number of generations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAncestors(v1, geneal, gen = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAncestors_+3A_v1">v1</code></td>
<td>
<p>the label of the vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getAncestors_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="getAncestors_+3A_gen">gen</code></td>
<td>
<p>the number of generations back to include as ancestors</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
getParent("Essex", sbGeneal)
getAncestors("Essex", sbGeneal, 1)
getAncestors("Essex", sbGeneal, 5)
</code></pre>

<hr>
<h2 id='getBasicStatistics'>Determine basic statistics of the graph object</h2><span id='topic+getBasicStatistics'></span>

<h3>Description</h3>

<p>Returns basic statistics of the graph object (number of nodes, number of edges, whether or not the
whole graph is connected, number of components, average path length, graph diameter, etc.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBasicStatistics(ig)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBasicStatistics_+3A_ig">ig</code></td>
<td>
<p>the graph representation of the data genealogy (in igraph format)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
ig &lt;- dfToIG(sbGeneal)
getBasicStatistics(ig)
</code></pre>

<hr>
<h2 id='getBranchQual'>Descendant branch calculations for quantitative variable</h2><span id='topic+getBranchQual'></span>

<h3>Description</h3>

<p>Returns a data frame containing the names of all children of an individual of interest (&quot;Name&quot;). The mean and standard deviation (&quot;Mean&quot; and &quot;SD&quot;) of a quantitative variable across all descendents of each child is reported. In addition, for each child, the number of its descendants is reported (&quot;Count&quot;), the number of its descendants who do not have a value for the quantitative variable (&quot;NACount&quot;) is reported, and the names of all of its descendants is reported (&quot;DesNames&quot;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBranchQual(v1, geneal, colName, rExpr, gen = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBranchQual_+3A_v1">v1</code></td>
<td>
<p>the label of the vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getBranchQual_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="getBranchQual_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the qualitative variable of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getBranchQual_+3A_rexpr">rExpr</code></td>
<td>
<p>regular expression to be applied to the column that contains the qualitative variable of interest (in character string format). The regular expression syntax must work on a data frame column of type character. The term geneal$colName must be used in the regular expression.</p>
</td></tr>
<tr><td><code id="getBranchQual_+3A_gen">gen</code></td>
<td>
<p>the number of generations back to include as ancestors</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(statGeneal)
rExpr = "geneal$colName=='The Johns Hopkins University'"
DC_JHU = getBranchQual("David Cox", statGeneal, "school", rExpr, 15)
rExpr = "geneal$colName=='UnitedKingdom'"
DC_UK = getBranchQual("David Cox", statGeneal, "country", rExpr, 15)
rExpr = "grepl('(?i)Stochastic', geneal$colName)"
DC_Stochastic = getBranchQual("David Cox", statGeneal, "thesis", rExpr, 15)
</code></pre>

<hr>
<h2 id='getBranchQuant'>Descendant branch calculations for quantitative variable</h2><span id='topic+getBranchQuant'></span>

<h3>Description</h3>

<p>Returns a data frame containing the names of all children of an individual of interest (&quot;Name&quot;). The mean and standard deviation (&quot;Mean&quot; and &quot;SD&quot;) of a quantitative variable across all descendents of each child is reported. In addition, for each child, the number of its descendants is reported (&quot;Count&quot;), the number of its descendants who do not have a value for the quantitative variable (&quot;NACount&quot;) is reported, and the names of all of its descendants is reported (&quot;DesNames&quot;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBranchQuant(v1, geneal, colName, gen = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBranchQuant_+3A_v1">v1</code></td>
<td>
<p>the label of the vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getBranchQuant_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="getBranchQuant_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the quantitative variable of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getBranchQuant_+3A_gen">gen</code></td>
<td>
<p>the number of generations back to include as ancestors</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(statGeneal)
DC_Year &lt;- getBranchQuant("David Cox", statGeneal, "gradYear", 15)
</code></pre>

<hr>
<h2 id='getChild'>Returns the children of a particular variety (if they exist)</h2><span id='topic+getChild'></span>

<h3>Description</h3>

<p>This function returns zero or more values that indicate the children of the inputted variety.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getChild(v1, geneal)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getChild_+3A_v1">v1</code></td>
<td>
<p>the label of the vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getChild_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
getChild("Tokyo", sbGeneal)
getChild("Essex", sbGeneal)
</code></pre>

<hr>
<h2 id='getDegree'>Determine the degree between two varieties</h2><span id='topic+getDegree'></span>

<h3>Description</h3>

<p>Returns the degree (distance between unweighted edges) between two varieties, where an edge
represents a parent-child relationship
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDegree(v1, v2, ig, geneal)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDegree_+3A_v1">v1</code></td>
<td>
<p>the label of the first vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getDegree_+3A_v2">v2</code></td>
<td>
<p>the label of the second vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getDegree_+3A_ig">ig</code></td>
<td>
<p>the graph representation of the data genealogy (in igraph format)</p>
</td></tr>
<tr><td><code id="getDegree_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
ig &lt;- dfToIG(sbGeneal)
getDegree("Brim", "Bedford", ig, sbGeneal)
</code></pre>

<hr>
<h2 id='getDescendants'>Returns a list of the descendants of a particular variety (if they exist)</h2><span id='topic+getDescendants'></span>

<h3>Description</h3>

<p>This function returns a list of the descendants of the inputted variety within and including a given number of generations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDescendants(v1, geneal, gen = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDescendants_+3A_v1">v1</code></td>
<td>
<p>the label of the vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getDescendants_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="getDescendants_+3A_gen">gen</code></td>
<td>
<p>the number of generations back to include as descendants</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
getChild("Essex", sbGeneal)
getDescendants("Essex", sbGeneal, 1)
getDescendants("Essex", sbGeneal, 3)
</code></pre>

<hr>
<h2 id='getEdges'>Returns edges (vertex names and edge weights) for the full genealogy</h2><span id='topic+getEdges'></span>

<h3>Description</h3>

<p>Returns a matrix, where each row contains information about an edge (two vertex names and edge weight, if present) of the full genealogy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getEdges(ig, geneal)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getEdges_+3A_ig">ig</code></td>
<td>
<p>the graph representation of the data genealogy (in igraph format)</p>
</td></tr>
<tr><td><code id="getEdges_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
ig &lt;- dfToIG(sbGeneal)
getEdges(ig, sbGeneal)
</code></pre>

<hr>
<h2 id='getNodes'>Returns the nodes for a full genealogy</h2><span id='topic+getNodes'></span>

<h3>Description</h3>

<p>Returns a character list, where rows contains names of the unique nodes in the full genealogy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNodes(geneal)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getNodes_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy (in data frame format)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
getNodes(sbGeneal)
</code></pre>

<hr>
<h2 id='getParent'>Returns the parents of a particular variety (if they exist)</h2><span id='topic+getParent'></span>

<h3>Description</h3>

<p>This function returns up to two values that indicate the parents of the inputted variety.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getParent(v1, geneal)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getParent_+3A_v1">v1</code></td>
<td>
<p>the label of the vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getParent_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
getParent("Tokyo", sbGeneal)
getParent("Essex", sbGeneal)
</code></pre>

<hr>
<h2 id='getPath'>Determine the path between two varieties</h2><span id='topic+getPath'></span>

<h3>Description</h3>

<p>Determines the shortest path between the two inputted vertices, and takes into
account whether or not the graph is directed. If there is a path, the list of vertices of the
path will be returned. If there is not a path, a list of character(0) will be returned. Note:
For a directed graph, the direction matters. However, this function will check both directions
and return the path if it exists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPath(v1, v2, ig, geneal, colName, silent = FALSE, isDirected = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getPath_+3A_v1">v1</code></td>
<td>
<p>the label of the first vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getPath_+3A_v2">v2</code></td>
<td>
<p>the label of the second vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getPath_+3A_ig">ig</code></td>
<td>
<p>the graph representation of the data genealogy (in igraph format)</p>
</td></tr>
<tr><td><code id="getPath_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="getPath_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the quantitative variable of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getPath_+3A_silent">silent</code></td>
<td>
<p>whether or not to print output (defaults to false)</p>
</td></tr>
<tr><td><code id="getPath_+3A_isdirected">isDirected</code></td>
<td>
<p>whether or not the graph is directed (defaults to false)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
ig &lt;- dfToIG(sbGeneal)
getPath("Brim", "Bedford", ig, sbGeneal, "devYear")
getPath("Tokyo", "Volstate", ig, sbGeneal, "yield")
</code></pre>

<hr>
<h2 id='getPathOnly'>Determine the path between two varieties</h2><span id='topic+getPathOnly'></span>

<h3>Description</h3>

<p>Determines the shortest path between the two inputted vertices, and takes into
account whether or not the graph is directed. If there is a path, the list of vertices of the
path will be returned. If there is not a path, a list of character(0) will be returned. Note:
For a directed graph, the direction matters. However, this function will check both directions
and return the path if it exists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPathOnly(v1, v2, ig, geneal, silent = FALSE, isDirected = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getPathOnly_+3A_v1">v1</code></td>
<td>
<p>the label of the first vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getPathOnly_+3A_v2">v2</code></td>
<td>
<p>the label of the second vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getPathOnly_+3A_ig">ig</code></td>
<td>
<p>the graph representation of the data genealogy (in igraph format)</p>
</td></tr>
<tr><td><code id="getPathOnly_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="getPathOnly_+3A_silent">silent</code></td>
<td>
<p>whether or not to print output (defaults to false)</p>
</td></tr>
<tr><td><code id="getPathOnly_+3A_isdirected">isDirected</code></td>
<td>
<p>whether or not the graph is directed (defaults to false)</p>
</td></tr>
</table>

<hr>
<h2 id='getVariable'>Determine the date of a variety</h2><span id='topic+getVariable'></span>

<h3>Description</h3>

<p>Returns the documented date of the inputted variety
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getVariable(v1, geneal, colName)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getVariable_+3A_v1">v1</code></td>
<td>
<p>the label of the vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="getVariable_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="getVariable_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the quantitative variable of interest (in character string format)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
getVariable("Essex", sbGeneal, "devYear")
getVariable("Tokyo", sbGeneal, "yield")
</code></pre>

<hr>
<h2 id='isChild'>Determine if a variety is a child of another</h2><span id='topic+isChild'></span>

<h3>Description</h3>

<p>Returns a boolean variable for whether the first variety is a child of the second variety
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isChild(child, parent, geneal)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isChild_+3A_child">child</code></td>
<td>
<p>possible child variety</p>
</td></tr>
<tr><td><code id="isChild_+3A_parent">parent</code></td>
<td>
<p>possible parent variety</p>
</td></tr>
<tr><td><code id="isChild_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
isChild("Essex", "Young", sbGeneal)
isChild("Young", "Essex", sbGeneal)
</code></pre>

<hr>
<h2 id='isParent'>Determine if a variety is a parent of another</h2><span id='topic+isParent'></span>

<h3>Description</h3>

<p>Returns a boolean variable for whether the second variety is a parent of the first variety
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isParent(child, parent, geneal)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isParent_+3A_child">child</code></td>
<td>
<p>possible child variety</p>
</td></tr>
<tr><td><code id="isParent_+3A_parent">parent</code></td>
<td>
<p>possible parent variety</p>
</td></tr>
<tr><td><code id="isParent_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
isParent("Essex", "Young", sbGeneal)
isParent("Young", "Essex", sbGeneal)
</code></pre>

<hr>
<h2 id='nodeToDF'>Returns the data frame representation of all ancestors and descendants of a variety</h2><span id='topic+nodeToDF'></span>

<h3>Description</h3>

<p>Converts the list-style-genealogy to a data frame, where each variety has an id value
and references its' parent's id value. ID value ranges correspond to generation.
It is possible that with more complex genealogical structures the range of id values may need to
expand to reduce the probability of two varieties being assigned the same id value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nodeToDF(tlist, branch = 0, par.id = NA, id = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nodeToDF_+3A_tlist">tlist</code></td>
<td>
<p>list of varieties</p>
</td></tr>
<tr><td><code id="nodeToDF_+3A_branch">branch</code></td>
<td>
<p>of particular variety in the genealogy</p>
</td></tr>
<tr><td><code id="nodeToDF_+3A_par.id">par.id</code></td>
<td>
<p>the id of the parent</p>
</td></tr>
<tr><td><code id="nodeToDF_+3A_id">id</code></td>
<td>
<p>id offset</p>
</td></tr>
</table>

<hr>
<h2 id='plotAncDes'>Returns the image object to show the ancestors and descendants of a variety</h2><span id='topic+plotAncDes'></span>

<h3>Description</h3>

<p>Returns the image object to show the ancestors and descendants of a variety, with the variety highlighted, if desired
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotAncDes(v1, geneal, mAnc = 3, mDes = 3, vColor = "#D35C79")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotAncDes_+3A_v1">v1</code></td>
<td>
<p>the label of the vertex of interest (in character string format)</p>
</td></tr>
<tr><td><code id="plotAncDes_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="plotAncDes_+3A_manc">mAnc</code></td>
<td>
<p>the maximum number of generations of ancestors of v1 to be displayed (in numeric format)</p>
</td></tr>
<tr><td><code id="plotAncDes_+3A_mdes">mDes</code></td>
<td>
<p>the maximum number of generations of descendants of v1 to be displayed (in numeric format)</p>
</td></tr>
<tr><td><code id="plotAncDes_+3A_vcolor">vColor</code></td>
<td>
<p>the color of the text of the main variety</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
plotAncDes("Tokyo", sbGeneal, vColor = "red")
plotAncDes("Essex", sbGeneal, 2, 3, "blue") + ggplot2::labs(x = "Generation index", y = "")
</code></pre>

<hr>
<h2 id='plotDegMatrix'>Returns the image object to show the heat map of degrees between the inputted set of vertices</h2><span id='topic+plotDegMatrix'></span>

<h3>Description</h3>

<p>Returns the image object to show the heat map of degrees between the inputted set of vertices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDegMatrix(varieties, ig, geneal)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotDegMatrix_+3A_varieties">varieties</code></td>
<td>
<p>subset of varieties used to generate the heat map</p>
</td></tr>
<tr><td><code id="plotDegMatrix_+3A_ig">ig</code></td>
<td>
<p>the graph representation of the data genealogy (in igraph format)</p>
</td></tr>
<tr><td><code id="plotDegMatrix_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>See Also</h3>

<p><a href="https://www.r-project.org">https://www.r-project.org</a> for iGraph information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
ig &lt;- dfToIG(sbGeneal)
varieties &lt;- c("Bedford", "Calland", "Narow", "Pella", "Tokyo", "Young", "Zane")
p &lt;- plotDegMatrix(varieties, ig, sbGeneal)
p + ggplot2::scale_fill_continuous(low = "white", high = "darkgreen")
</code></pre>

<hr>
<h2 id='plotPath'>Construct the graphic object of the path</h2><span id='topic+plotPath'></span>

<h3>Description</h3>

<p>This function takes the path as input and outputs an ggplot2 object. The
image will correctly position the node labels with x-axis representing the node
date, and y-axis representing the node path index. Edges between two nodes represent
parent-child relationships between those nodes. For visual appeal, there is a grey
box that outlines the node label, as well as an underline and overline for each label.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPath(path, geneal, colName, colNameY = "", fontFace = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotPath_+3A_path">path</code></td>
<td>
<p>object created from function getPath</p>
</td></tr>
<tr><td><code id="plotPath_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="plotPath_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the quantitative variable of interest (in character string format)</p>
</td></tr>
<tr><td><code id="plotPath_+3A_colnamey">colNameY</code></td>
<td>
<p>the name of the second optional column of the data frame that contains the second optional quantitative variable of interest (in character string format). This optional quantitative variable will be plotted on the vertical axis.</p>
</td></tr>
<tr><td><code id="plotPath_+3A_fontface">fontFace</code></td>
<td>
<p>fontface for the two nodes of interest (1=plain, 2=bold, 3=italic, 4=bold-italic), DEFAULT is 1</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getPath">getPath</a></code> for information on input path building
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
ig &lt;- dfToIG(sbGeneal)
pathTN &lt;- getPath("Tokyo", "Narow", sbIG, sbGeneal, "devYear")
plotPath(pathTN, sbGeneal, "devYear")

sbFilt &lt;- sbGeneal[complete.cases(sbGeneal[1:3]),]
sbFiltIG &lt;- dfToIG(sbFilt)
pathCL &lt;- getPath("Clark", "Lawrence", sbFiltIG, sbFilt, "yield")
plotPath(pathCL, sbFilt, "devYear", "yield") + ggplot2::xlab("Dev Year") + ggplot2::ylab("Yield")
</code></pre>

<hr>
<h2 id='plotPathOnAll'>Plot a path between two vertices over the full genealogy</h2><span id='topic+plotPathOnAll'></span>

<h3>Description</h3>

<p>This function requires a path and the ig object, and plots the full genealogy 
with the path highlighted.
The image will correctly position the node labels with x-axis representing the node
date, and y-axis representing the node path index. Light grey edges between two nodes
represent parent-child relationships between those nodes. To enhance the visual
understanding of how the path-of-interest fits into the entire graph structure, the
nodes within the path are labelled in boldface, and connected with light-green
boldfaced edges.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPathOnAll(
  path,
  geneal,
  ig,
  colName,
  colNameY = "",
  bin = 12,
  edgeCol = "gray84",
  pathEdgeCol = "seagreen",
  nodeSize = 3,
  pathNodeSize = 3,
  pathNodeFont = "bold",
  nodeCol = "black",
  animate = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotPathOnAll_+3A_path">path</code></td>
<td>
<p>path as returned from getPath() or a vector of two variety names which exist in ig</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_ig">ig</code></td>
<td>
<p>the graph representation of the data genealogy (in igraph format)</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the quantitative variable of interest (in character string format)</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_colnamey">colNameY</code></td>
<td>
<p>the name of the second optional column of the data frame that contains the second optional quantitative variable of interest (in character string format). This optional quantitative variable will be plotted on the vertical axis.</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_bin">bin</code></td>
<td>
<p>the number of bins to determine the vertical positions of nodes (default is 12). For more information on choosing bin size, please visit the ggenealogy vignette</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_edgecol">edgeCol</code></td>
<td>
<p>color of the non-path edges, default is &quot;gray84&quot;</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_pathedgecol">pathEdgeCol</code></td>
<td>
<p>color of the path edges, default is &quot;seagreen&quot;</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_nodesize">nodeSize</code></td>
<td>
<p>text size of the non-path node labels, default is 3</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_pathnodesize">pathNodeSize</code></td>
<td>
<p>text size of the path node labels, default is 3</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_pathnodefont">pathNodeFont</code></td>
<td>
<p>font face of text of the path node labels (&quot;plain&quot;, &quot;italic&quot;, &quot;bold&quot;, &quot;bold.italic&quot;), default is &quot;bold&quot;</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_nodecol">nodeCol</code></td>
<td>
<p>color of the non-path node labels, default is black</p>
</td></tr>
<tr><td><code id="plotPathOnAll_+3A_animate">animate</code></td>
<td>
<p>if the plot will have interactive capabilities, default is FALSE</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>See Also</h3>

<p><a href="https://www.r-project.org">https://www.r-project.org</a> for iGraph information
</p>
<p><code><a href="#topic+getPath">getPath</a></code> for information on input path building
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
sb &lt;- sbGeneal[complete.cases(sbGeneal[1:3]),]
ig &lt;- dfToIG(sb)
pathCL &lt;- getPath("Clark", "Lawrence", ig, sb, "yield")
plotPathOnAll(pathCL, sb, ig, "yield", bin = 3, pathEdgeCol = "red") + ggplot2::xlab("Yield")
plotPathOnAll(pathCL, sb, ig, "yield", "devYear") + ggplot2::xlab("Yield") + ggplot2::ylab("Year")
</code></pre>

<hr>
<h2 id='plotVariableMatrix'>Returns the image object to show the heat map of dates between the inputted set of vertices</h2><span id='topic+plotVariableMatrix'></span>

<h3>Description</h3>

<p>Returns the image object to show the heat map of dates between the inputted set of vertices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotVariableMatrix(
  varieties,
  geneal,
  colName,
  xLab = "Variety",
  yLab = "Variety",
  legendLab = "Difference in variable"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotVariableMatrix_+3A_varieties">varieties</code></td>
<td>
<p>subset of varieties used to generate the heat map</p>
</td></tr>
<tr><td><code id="plotVariableMatrix_+3A_geneal">geneal</code></td>
<td>
<p>the full genealogy  (in data frame format)</p>
</td></tr>
<tr><td><code id="plotVariableMatrix_+3A_colname">colName</code></td>
<td>
<p>the name of the column of the data frame that contains the quantitative variable of interest (in character string format)</p>
</td></tr>
<tr><td><code id="plotVariableMatrix_+3A_xlab">xLab</code></td>
<td>
<p>string label on the x axis (default is &quot;Variety&quot;)</p>
</td></tr>
<tr><td><code id="plotVariableMatrix_+3A_ylab">yLab</code></td>
<td>
<p>string label on the y axis (default is &quot;Variety&quot;)</p>
</td></tr>
<tr><td><code id="plotVariableMatrix_+3A_legendlab">legendLab</code></td>
<td>
<p>string label on the legend (default is &quot;Degree&quot;)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rutter L, VanderPlas S, Cook D, Graham MA (2019).
ggenealogy: An R Package for Visualizing Genealogical Data.
<em>Journal of Statistical Software</em>,
<b>89</b>(13), 1&ndash;31.
<a href="https://doi.org/10.18637/jss.v089.i13">doi:10.18637/jss.v089.i13</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sbGeneal)
varieties &lt;- c("Bedford", "Calland", "Narow", "Pella", "Tokyo", "Young", "Zane")
p &lt;- plotVariableMatrix(varieties, sbGeneal, "devYear", "Variety", "Variety", "Difference")
p + ggplot2::scale_fill_continuous(low = "white", high = "darkgreen")
</code></pre>

<hr>
<h2 id='sbGeneal'>Soybean data</h2><span id='topic+sbGeneal'></span>

<h3>Description</h3>

<p>This data set contains soy bean genealogical information maintained by the United States Department of Agriculture to be used by plant breeders, geneticists, bioinformaticians, pathologists,and many other research workers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sbGeneal)
</code></pre>


<h3>Format</h3>

<p>a <code>RData</code> instance, 1 row per each child-parent relationship between soybean varieties
</p>
<p>A data frame with 412 rows and 7 variables
</p>


<h3>Details</h3>

<p>Soybean genealogical data
</p>
<p>This data contains information on copy number variants, single nucleotide polymorphisms, protein content, and yield, of soybeans. The available data consists of a data frame structure that contains 412 direct child-parent relationships between pairs of soybean varieties. These data were collected from field trials, genetic studies, and United States Department of Agriculture (USDA) bulletins, and date as early as the first decade of the 1900s.
</p>

<ul>
<li><p> child name of child soybean variety
</p>
</li>
<li><p> devYear year child variety was introduced
</p>
</li>
<li><p> yield protein yield
</p>
</li>
<li><p> yearImputed whether or not the introduced year of the child variety was imputed
</p>
</li>
<li><p> parent name of parent soybean variety
</p>
</li></ul>



<h3>References</h3>

<p>&quot;Pedigrees of soybean cultivars released in the United States and Canada.&quot; Theodore Hyivitz, C.A. Newell, S.G. Carmer. College of Agriculture, University of Illinois at Urbana-Champaign (1977).
</p>

<hr>
<h2 id='statGeneal'>Academic statistics data</h2><span id='topic+statGeneal'></span>

<h3>Description</h3>

<p>This data set contains academic genealogical information from the Mathematics Genealogy Project, a web-based database serviced by the North Dakota State University Department of Mathematics and the American Mathematical Society. Specifically, this data set represents a subset of the Mathematical Genealogy Project that contains all the parent-child relationships where both parent and child received an advanced degree of statistics as of June 6, 2015.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(statGeneal)
</code></pre>


<h3>Format</h3>

<p>a <code>RData</code> instance, 1 row per each individual in the Mathematics Genealogy Project with an advanced degree in statistics. If the child had a parent who was also listed as having received an advanced degree in statistics in the Mathematics Genealogy project, then the row also contains the parental information.
</p>
<p>A data frame with 8165 rows and 6 variables
</p>


<h3>Details</h3>

<p>Academic statistics genealogical data
</p>
<p>This data contains information on copy number variants, single nucleotide polymorphisms, protein content, and yield, of soybeans. The available data consists of a data frame structure that contains 412 direct child-parent relationships between pairs of soybean varieties. These data were collected from field trials, genetic studies, and United States Department of Agriculture (USDA) bulletins, and date as early as the first decade of the 1900s.
</p>

<ul>
<li><p> child name of the individual who received an advanced degree in statistics
</p>
</li>
<li><p> parent name of the individual who mentored the child and also received an advanced degree in statistics. If the child has no such parent, then this field is an empty string
</p>
</li>
<li><p> gradYear year the child received their advanced degree in statistics
</p>
</li>
<li><p> country country from which the child received their advanced degree in statistics
</p>
</li>
<li><p> school school from which the child received their advanced degree in statistics
</p>
</li>
<li><p> thesis title of the thesis the child submitted to receive their advanced degree in statistics. If this information is not available, then this field is an empty string
</p>
</li></ul>



<h3>References</h3>

<p>North Dakota State University and American Mathematical Society (2010). The Mathematics Genealogy Project. Archived Web Site. Retrieved from the Library of Congress, Accessed on March 6, 2015, URL http://www.genealogy.math.ndsu.nodak.edu.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
