<!DOCTYPE html><html lang="en-US"><head><title>Help for package gggenes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gggenes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gggenes-package'><p>gggenes: Draw Gene Arrow Maps in 'ggplot2'</p></a></li>
<li><a href='#example_dummies'><p>A set of example dummy alignment genes.</p></a></li>
<li><a href='#example_features'><p>A set of example genetic features.</p></a></li>
<li><a href='#example_genes'><p>A set of example genes.</p></a></li>
<li><a href='#geom_feature'><p>A 'ggplot2' geom to draw point genetic features</p></a></li>
<li><a href='#geom_feature_label'><p>A 'ggplot2' geom to add text labels to point genetic features</p></a></li>
<li><a href='#geom_gene_arrow'><p>A 'ggplot2' geom to draw genes as arrows</p></a></li>
<li><a href='#geom_gene_label'><p>A 'ggplot2' geom to add text labels to gene arrows</p></a></li>
<li><a href='#geom_subgene_arrow'><p>A 'ggplot2' geom to draw subgene segments of gene arrows</p></a></li>
<li><a href='#geom_subgene_label'><p>A 'ggplot2' geom to add text labels to subgenes</p></a></li>
<li><a href='#make_alignment_dummies'><p>Prepare dummy data to visually align a single gene across faceted molecules</p></a></li>
<li><a href='#theme_genes'><p>A 'ggplot2' theme for drawing gene maps</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Draw Gene Arrow Maps in 'ggplot2'</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A 'ggplot2' extension for drawing gene arrow maps.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>Imports:</td>
<td>grid (&ge; 3.3.0), ggplot2 (&ge; 2.2.1), ggfittext (&ge; 0.8.0),
rlang (&ge; 0.2.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://wilkox.org/gggenes/">https://wilkox.org/gggenes/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/wilkox/gggenes/issues/">https://github.com/wilkox/gggenes/issues/</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, vdiffr, spelling</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-05 05:34:29 UTC; wilkox</td>
</tr>
<tr>
<td>Author:</td>
<td>David Wilkins [aut, cre],
  Zachary Kurtz [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>David Wilkins &lt;david@wilkox.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-05 11:50:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='gggenes-package'>gggenes: Draw Gene Arrow Maps in 'ggplot2'</h2><span id='topic+gggenes'></span><span id='topic+gggenes-package'></span>

<h3>Description</h3>

<p>A 'ggplot2' extension for drawing gene arrow maps.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: David Wilkins <a href="mailto:david@wilkox.org">david@wilkox.org</a>
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Zachary Kurtz <a href="mailto:zdkurtz@gmail.com">zdkurtz@gmail.com</a> [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://wilkox.org/gggenes/">https://wilkox.org/gggenes/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/wilkox/gggenes/issues/">https://github.com/wilkox/gggenes/issues/</a>
</p>
</li></ul>


<hr>
<h2 id='example_dummies'>A set of example dummy alignment genes.</h2><span id='topic+example_dummies'></span>

<h3>Description</h3>

<p>Dummy genes, similar to those generated with <code>make_alignment_dummies()</code>, for
example purposes only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_dummies
</code></pre>


<h3>Format</h3>

<p>A data frame with eight rows and three variables:
</p>

<dl>
<dt>molecule</dt><dd><p>the genome</p>
</dd>
<dt>start</dt><dd><p>the start position of the dummy</p>
</dd>
<dt>end</dt><dd><p>the end position of the dummy</p>
</dd>
<dt>gene</dt><dd><p>the name of the dummy gene</p>
</dd>
</dl>


<hr>
<h2 id='example_features'>A set of example genetic features.</h2><span id='topic+example_features'></span>

<h3>Description</h3>

<p>Genetic features for example purposes only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_features
</code></pre>


<h3>Format</h3>

<p>A data frame with 23 rows and five variables:
</p>

<dl>
<dt>molecule</dt><dd><p>the genome</p>
</dd>
<dt>name</dt><dd><p>the name of the feature</p>
</dd>
<dt>type</dt><dd><p>the type of the feature</p>
</dd>
<dt>position</dt><dd><p>the position of the feature</p>
</dd>
<dt>forward</dt><dd><p>is the feature oriented, and if so in the forward direction?</p>
</dd>
</dl>


<hr>
<h2 id='example_genes'>A set of example genes.</h2><span id='topic+example_genes'></span><span id='topic+example_subgenes'></span>

<h3>Description</h3>

<p>Genes for example purposes only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_genes

example_subgenes
</code></pre>


<h3>Format</h3>

<p>A data frame with 72 rows and six variables:
</p>

<dl>
<dt>molecule</dt><dd><p>the genome</p>
</dd>
<dt>gene</dt><dd><p>the name of the gene</p>
</dd>
<dt>start</dt><dd><p>the start position of the gene</p>
</dd>
<dt>end</dt><dd><p>the end position of the gene</p>
</dd>
<dt>strand</dt><dd><p>the strand of the gene</p>
</dd>
<dt>orientation</dt><dd><p>the orientation of the gene</p>
</dd>
</dl>

<p>example_subgenes (143 rows) also contains:
</p>

<dl>
<dt>subgene</dt><dd><p>the name of the subgene</p>
</dd>
<dt>from</dt><dd><p>the start position of the subgene segment</p>
</dd>
<dt>to</dt><dd><p>the end position of the subgene segment</p>
</dd>
</dl>

<p>An object of class <code>data.frame</code> with 143 rows and 9 columns.
</p>

<hr>
<h2 id='geom_feature'>A 'ggplot2' geom to draw point genetic features</h2><span id='topic+geom_feature'></span>

<h3>Description</h3>

<p><code>geom_feature()</code> draws lines to indicate the positions of point genetic
features, for example restriction sites, origins of replication or
transcription start sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_feature(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = FALSE,
  feature_height = unit(3, "mm"),
  feature_width = unit(3, "mm"),
  arrowhead_width = unit(2, "mm"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_feature_+3A_mapping">mapping</code>, <code id="geom_feature_+3A_data">data</code>, <code id="geom_feature_+3A_stat">stat</code>, <code id="geom_feature_+3A_position">position</code>, <code id="geom_feature_+3A_na.rm">na.rm</code>, <code id="geom_feature_+3A_show.legend">show.legend</code>, <code id="geom_feature_+3A_inherit.aes">inherit.aes</code>, <code id="geom_feature_+3A_...">...</code></td>
<td>
<p>As
standard for ggplot2. inherit.aes is set to FALSE by default, as features
are not likely to share any plot aesthetics other than y.</p>
</td></tr>
<tr><td><code id="geom_feature_+3A_feature_height">feature_height</code></td>
<td>
<p><code>grid::unit()</code> object giving the height of a feature
above the molecule line. Can be set as a negative value to draw features
below the line. Defaults to 3 mm.</p>
</td></tr>
<tr><td><code id="geom_feature_+3A_feature_width">feature_width</code></td>
<td>
<p><code>grid::unit()</code> object giving the width of a feature
(distance from the elbow to the tip of the arrow). Only relevant for
oriented features. Defaults to 3 mm.</p>
</td></tr>
<tr><td><code id="geom_feature_+3A_arrowhead_width">arrowhead_width</code></td>
<td>
<p><code>grid::unit()</code> object giving the width of the
arrowhead indicating the direction of an oriented feature. Only relevant for
oriented features. Defaults to 2 mm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Features are drawn as vertical lines extending from the horizontal line
representing the molecule. The position of the feature is expressed with the
<code>x</code> aesthetic. Optionally, the <code>forward</code> aesthetic can be used to specific
an orientation for the feature (e.g. the direction of transcription), in
which case an angled arrowhead will be added. The <code>forward</code> aesthetic
assumes that the x-axis is oriented in the normal direction, i.e. increasing
from left to right; if it is not, the values in <code>forward</code> will need to be
inverted manually.
</p>


<h3>Aesthetics</h3>


<ul>
<li><p> x (required; position of the feature)
</p>
</li>
<li><p> y (required; molecule)
</p>
</li>
<li><p> forward (optional; if TRUE, or a value coercible to TRUE, the feature will
be drawn with an arrowhead pointing right, if FALSE, pointing left, if NA,
the feature will be drawn as a vertical line)
</p>
</li>
<li><p> alpha
</p>
</li>
<li><p> colour
</p>
</li>
<li><p> linetype
</p>
</li>
<li><p> size
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+geom_feature_label">geom_feature_label()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ggplot2::ggplot(example_genes, ggplot2::aes(xmin = start, xmax = end,
                                            y = molecule, fill = gene)) +
  geom_gene_arrow() +
  geom_feature(data = example_features, ggplot2::aes(x = position, y = molecule, 
                                                     forward = forward)) +
  ggplot2::facet_wrap(~ molecule, scales = "free")

</code></pre>

<hr>
<h2 id='geom_feature_label'>A 'ggplot2' geom to add text labels to point genetic features</h2><span id='topic+geom_feature_label'></span>

<h3>Description</h3>

<p><code>geom_feature_label()</code> adds text labels to features drawn with
<code>geom_feature</code>().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_feature_label(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  na.rm = FALSE,
  show.legend = FALSE,
  inherit.aes = FALSE,
  feature_height = unit(4, "mm"),
  label_height = unit(3, "mm"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_feature_label_+3A_mapping">mapping</code>, <code id="geom_feature_label_+3A_data">data</code>, <code id="geom_feature_label_+3A_stat">stat</code>, <code id="geom_feature_label_+3A_position">position</code>, <code id="geom_feature_label_+3A_na.rm">na.rm</code>, <code id="geom_feature_label_+3A_show.legend">show.legend</code>, <code id="geom_feature_label_+3A_inherit.aes">inherit.aes</code>, <code id="geom_feature_label_+3A_...">...</code></td>
<td>
<p>As
standard for ggplot2. inherit.aes is set to FALSE by default, as features
are not likely to share any plot aesthetics other than y.</p>
</td></tr>
<tr><td><code id="geom_feature_label_+3A_feature_height">feature_height</code></td>
<td>
<p><code>grid::unit()</code> object giving the height of the feature
being labelled, and hence the distance of the label above or below the
molecule line. Can be set as a negative value for features drawn below the
line. Defaults to 4 mm, to align labels with the default height of
<code>geom_feature()</code>.</p>
</td></tr>
<tr><td><code id="geom_feature_label_+3A_label_height">label_height</code></td>
<td>
<p><code>grid::unit()</code> object giving the height of the label
text. Defaults to 3 mm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Standard 'ggplot2' aesthetics for text are supported (see Aesthetics).
</p>


<h3>Aesthetics</h3>


<ul>
<li><p> x (required; position of the feature)
</p>
</li>
<li><p> y (required; molecule)
</p>
</li>
<li><p> label (required; the label text)
</p>
</li>
<li><p> forward (optional; will draw text in the appropriate location for features
with angled arrowheads)
</p>
</li>
<li><p> colour
</p>
</li>
<li><p> size
</p>
</li>
<li><p> alpha
</p>
</li>
<li><p> family
</p>
</li>
<li><p> fontface
</p>
</li>
<li><p> angle
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+geom_feature">geom_feature()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ggplot2::ggplot(example_genes, ggplot2::aes(xmin = start, xmax = end,
                                            y = molecule, fill = gene)) +
  geom_gene_arrow() +
  geom_feature(data = example_features, ggplot2::aes(x = position, y = molecule, 
                                                     forward = forward)) +
  geom_feature_label(data = example_features,
                     ggplot2::aes(x = position, y = molecule, label = name, 
                                  forward = forward)) + 
  ggplot2::facet_wrap(~ molecule, scales = "free")

</code></pre>

<hr>
<h2 id='geom_gene_arrow'>A 'ggplot2' geom to draw genes as arrows</h2><span id='topic+geom_gene_arrow'></span>

<h3>Description</h3>

<p><code>geom_gene_arrow()</code> draws genes as arrows, allowing gene maps to be drawn.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_gene_arrow(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  arrowhead_width = grid::unit(4, "mm"),
  arrowhead_height = grid::unit(4, "mm"),
  arrow_body_height = grid::unit(3, "mm"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_gene_arrow_+3A_mapping">mapping</code>, <code id="geom_gene_arrow_+3A_data">data</code>, <code id="geom_gene_arrow_+3A_stat">stat</code>, <code id="geom_gene_arrow_+3A_position">position</code>, <code id="geom_gene_arrow_+3A_na.rm">na.rm</code>, <code id="geom_gene_arrow_+3A_show.legend">show.legend</code>, <code id="geom_gene_arrow_+3A_inherit.aes">inherit.aes</code>, <code id="geom_gene_arrow_+3A_...">...</code></td>
<td>
<p>As
standard for ggplot2.</p>
</td></tr>
<tr><td><code id="geom_gene_arrow_+3A_arrowhead_width">arrowhead_width</code></td>
<td>
<p><code>grid::unit()</code> object giving the width of the
arrowhead.  Defaults to 4 mm. If the gene is drawn smaller than this width,
only the arrowhead will be drawn, compressed to the length of the gene.</p>
</td></tr>
<tr><td><code id="geom_gene_arrow_+3A_arrowhead_height">arrowhead_height</code></td>
<td>
<p><code>grid::unit()</code> object giving the height of the
arrowhead.  Defaults to 4 mm.</p>
</td></tr>
<tr><td><code id="geom_gene_arrow_+3A_arrow_body_height">arrow_body_height</code></td>
<td>
<p><code>grid::unit()</code> object giving the height of the body
of the arrow. Defaults to 3 mm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This geom draws genes as arrows along a horizontal line representing the
molecule. The start and end locations of the gene are expressed with the
<code>xmin</code> and <code>xmax</code> aesthetics, while the molecule can be specified with the
<code>y</code> aesthetic. Optionally, an additional <code>forward</code> aesthetic can be used to
reverse the orientation of some or all genes from that implied by <code>xmin</code> and
<code>xmax</code>.
</p>
<p>Unless the plot is faceted with a free x scale, all the molecules will share
a common x axis. This means that if the locations are very different across
different molecules, the genes might appear very small and squished together
with a lot of unnecessary empty space. To get around this, either facet the
plot with <code>scales = "free_x"</code>, or normalise the gene locations if their
exact locations are not important.
</p>
<p>See <code>make_alignment_dummies()</code> for a method to align genes between molecules.
</p>


<h3>Aesthetics</h3>


<ul>
<li><p> xmin,xmax (start and end of the gene; will be used to determine gene
orientation)
</p>
</li>
<li><p> y (molecule)
</p>
</li>
<li><p> forward (if any value that is not TRUE, or coercible to TRUE, the gene
arrow will be drawn in the opposite direction to that determined by <code>xmin</code>
and <code>xmax</code>)
</p>
</li>
<li><p> alpha
</p>
</li>
<li><p> colour
</p>
</li>
<li><p> fill
</p>
</li>
<li><p> linetype
</p>
</li>
<li><p> size
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+theme_genes">theme_genes()</a></code>, <code><a href="#topic+make_alignment_dummies">make_alignment_dummies()</a></code>, <code><a href="#topic+geom_gene_label">geom_gene_label()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ggplot2::ggplot(example_genes, ggplot2::aes(xmin = start, xmax = end,
                                            y = molecule, fill = gene)) +
geom_gene_arrow() +
ggplot2::facet_wrap(~ molecule, scales = "free")

</code></pre>

<hr>
<h2 id='geom_gene_label'>A 'ggplot2' geom to add text labels to gene arrows</h2><span id='topic+geom_gene_label'></span>

<h3>Description</h3>

<p><code>geom_gene_label()</code> can be used to add a text label to genes drawn with
<code>geom_gene_arrow()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_gene_label(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  na.rm = FALSE,
  show.legend = FALSE,
  inherit.aes = TRUE,
  padding.x = grid::unit(1, "mm"),
  padding.y = grid::unit(0.1, "lines"),
  align = "centre",
  min.size = 4,
  grow = F,
  reflow = F,
  height = grid::unit(3, "mm"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_gene_label_+3A_mapping">mapping</code>, <code id="geom_gene_label_+3A_data">data</code>, <code id="geom_gene_label_+3A_stat">stat</code>, <code id="geom_gene_label_+3A_position">position</code>, <code id="geom_gene_label_+3A_na.rm">na.rm</code>, <code id="geom_gene_label_+3A_show.legend">show.legend</code>, <code id="geom_gene_label_+3A_inherit.aes">inherit.aes</code>, <code id="geom_gene_label_+3A_...">...</code></td>
<td>
<p>Standard
geom arguments as for <code>ggplot2::geom_text()</code>.</p>
</td></tr>
<tr><td><code id="geom_gene_label_+3A_padding.x">padding.x</code>, <code id="geom_gene_label_+3A_padding.y">padding.y</code></td>
<td>
<p><code>grid::unit()</code> object, giving horizontal or vertical
padding around the text. Defaults to 1 mm and 0.1 lines respectively.</p>
</td></tr>
<tr><td><code id="geom_gene_label_+3A_align">align</code></td>
<td>
<p>Where inside the gene to place the text label. Default is
'centre'; other options are 'left' and 'right'.</p>
</td></tr>
<tr><td><code id="geom_gene_label_+3A_min.size">min.size</code></td>
<td>
<p>Minimum font size, in points. If provided, text that would
need to be shrunk below this size to fit inside the gene arrow will not be
drawn. Defaults to 4 pt.</p>
</td></tr>
<tr><td><code id="geom_gene_label_+3A_grow">grow</code></td>
<td>
<p>If <code>TRUE</code>, text will be grown as well as shrunk to fill the
arrow.</p>
</td></tr>
<tr><td><code id="geom_gene_label_+3A_reflow">reflow</code></td>
<td>
<p>If <code>TRUE</code>, text will be reflowed (wrapped) to better fit the
arrow.</p>
</td></tr>
<tr><td><code id="geom_gene_label_+3A_height">height</code></td>
<td>
<p><code>grid::unit()</code> object giving the maximum height of the text.
Defaults to 3 mm, which is the default height of gene arrows drawn with
<code>geom_gene_arrow()</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>geom_gene_label()</code> uses the 'ggfittext' package to fit text to genes. All
text drawing options available in <code>ggfittext::geom_fit_text()</code> (growing,
reflowing, etc.) are also available here. For full details on how these
options work, see the documentation for <code>ggfittext::geom_fit_text()</code>.
</p>
<p>Standard 'ggplot2' aesthetics for text are supported (see Aesthetics).
</p>


<h3>Aesthetics</h3>


<ul>
<li><p> xmin,xmax (start and end of the gene; required)
</p>
</li>
<li><p> y (molecule; required)
</p>
</li>
<li><p> label (the label text; required)
</p>
</li>
<li><p> colour
</p>
</li>
<li><p> size
</p>
</li>
<li><p> alpha
</p>
</li>
<li><p> family
</p>
</li>
<li><p> fontface
</p>
</li>
<li><p> angle
</p>
</li></ul>



<h3>See Also</h3>

<p>geom_gene_arrow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ggplot2::ggplot(example_genes, ggplot2::aes(xmin = start, xmax = end,
  y = molecule, fill = gene, label = gene)) +
  geom_gene_arrow() +
  geom_gene_label() +
  ggplot2::facet_wrap(~ molecule, ncol = 1, scales = "free") +
  theme_genes()
</code></pre>

<hr>
<h2 id='geom_subgene_arrow'>A 'ggplot2' geom to draw subgene segments of gene arrows</h2><span id='topic+geom_subgene_arrow'></span>

<h3>Description</h3>

<p><code>geom_subgene_arrow()</code> draws subgenes segments within gene arrows drawn with
<code>geom_gene_arrow()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_subgene_arrow(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  arrowhead_width = grid::unit(4, "mm"),
  arrowhead_height = grid::unit(4, "mm"),
  arrow_body_height = grid::unit(3, "mm"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_subgene_arrow_+3A_mapping">mapping</code>, <code id="geom_subgene_arrow_+3A_data">data</code>, <code id="geom_subgene_arrow_+3A_stat">stat</code>, <code id="geom_subgene_arrow_+3A_position">position</code>, <code id="geom_subgene_arrow_+3A_na.rm">na.rm</code>, <code id="geom_subgene_arrow_+3A_show.legend">show.legend</code>, <code id="geom_subgene_arrow_+3A_inherit.aes">inherit.aes</code>, <code id="geom_subgene_arrow_+3A_...">...</code></td>
<td>
<p>As
standard for 'ggplot2'.</p>
</td></tr>
<tr><td><code id="geom_subgene_arrow_+3A_arrowhead_width">arrowhead_width</code></td>
<td>
<p><code>grid::unit()</code> object giving the width of the
arrowhead.  Defaults to 4 mm. If the gene is drawn smaller than this width,
only the arrowhead will be drawn, compressed to the length of the gene.</p>
</td></tr>
<tr><td><code id="geom_subgene_arrow_+3A_arrowhead_height">arrowhead_height</code></td>
<td>
<p><code>grid::unit()</code> object giving the height of the
arrowhead.  Defaults to 4 mm.</p>
</td></tr>
<tr><td><code id="geom_subgene_arrow_+3A_arrow_body_height">arrow_body_height</code></td>
<td>
<p><code>grid::unit()</code> object giving the height of the body
of the arrow. Defaults to 3 mm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The start and end locations of the subgene are given with the <code>xsubmin</code> and
<code>xsubmax</code> aesthetics. <code>geom_subgene_arrow()</code> requires some information about
the 'parent' gene, provided with the same aesthetics used for
<code>geom_gene_arrow()</code>: start and end locations of the 'parent' gene with the
<code>xmin</code> and <code>xmax</code> aesthetics, the molecule with the <code>y</code> aesthetic, and
optionally the direction with the <code>forward</code> aesthetic. If the geometry of
the parent gene has been changed with <code>arrowhead_width</code>, <code>arrowhead_height</code>
or <code>arrow_body_height</code>, identical parameters should be given to
<code>geom_subgene_arrow()</code>.
</p>


<h3>Aesthetics</h3>


<ul>
<li><p> xmin,xmax (start and end of the gene; will be used to determine gene
orientation)
</p>
</li>
<li><p> xsubmin,xsubmax (start and end of subgene segment). Should be consistent
with <code>xmin</code>/<code>xmax</code>
</p>
</li>
<li><p> y (molecule)
</p>
</li>
<li><p> forward (if FALSE, or coercible to FALSE, the gene arrow will be drawn in
the opposite direction to that determined by <code>xmin</code> and <code>xmax</code>)
</p>
</li>
<li><p> alpha
</p>
</li>
<li><p> colour
</p>
</li>
<li><p> fill
</p>
</li>
<li><p> linetype
</p>
</li>
<li><p> size
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+geom_gene_arrow">geom_gene_arrow()</a></code>, <code><a href="#topic+geom_subgene_label">geom_subgene_label()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ggplot2::ggplot(example_genes, ggplot2::aes(xmin = start, xmax = end,
                                            y = molecule)) +
geom_gene_arrow() +
geom_subgene_arrow(data = example_subgenes,
      ggplot2::aes(xmin = start, xmax = end, xsubmin = from, xsubmax = to,
                   y = molecule, fill = gene)) +
ggplot2::facet_wrap(~ molecule, scales = "free")

</code></pre>

<hr>
<h2 id='geom_subgene_label'>A 'ggplot2' geom to add text labels to subgenes</h2><span id='topic+geom_subgene_label'></span>

<h3>Description</h3>

<p><code>geom_subgene_label()</code> can be used to add a text label to subgenes drawn
with <code>geom_subgene_arrow()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_subgene_label(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  na.rm = FALSE,
  show.legend = FALSE,
  inherit.aes = TRUE,
  padding.x = grid::unit(1, "mm"),
  padding.y = grid::unit(0.1, "lines"),
  align = "centre",
  min.size = 4,
  grow = F,
  reflow = F,
  height = grid::unit(3, "mm"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_subgene_label_+3A_mapping">mapping</code>, <code id="geom_subgene_label_+3A_data">data</code>, <code id="geom_subgene_label_+3A_stat">stat</code>, <code id="geom_subgene_label_+3A_position">position</code>, <code id="geom_subgene_label_+3A_na.rm">na.rm</code>, <code id="geom_subgene_label_+3A_show.legend">show.legend</code>, <code id="geom_subgene_label_+3A_inherit.aes">inherit.aes</code>, <code id="geom_subgene_label_+3A_...">...</code></td>
<td>
<p>Standard
geom arguments as for <code>ggplot2::geom_text()</code>.</p>
</td></tr>
<tr><td><code id="geom_subgene_label_+3A_padding.x">padding.x</code>, <code id="geom_subgene_label_+3A_padding.y">padding.y</code></td>
<td>
<p><code>grid::unit()</code> object, giving horizontal or
vertical padding around the text. Defaults to 1 mm and 0.1 lines
respectively.</p>
</td></tr>
<tr><td><code id="geom_subgene_label_+3A_align">align</code></td>
<td>
<p>Where inside the subgene to place the text label. Default is
'centre'; other options are 'left' and 'right'.</p>
</td></tr>
<tr><td><code id="geom_subgene_label_+3A_min.size">min.size</code></td>
<td>
<p>Minimum font size, in points. If provided, text that would
need to be shrunk below this size to fit inside the subgene will not be
drawn. Defaults to 4 pt.</p>
</td></tr>
<tr><td><code id="geom_subgene_label_+3A_grow">grow</code></td>
<td>
<p>If <code>TRUE</code>, text will be grown as well as shrunk to fill the
subgene.</p>
</td></tr>
<tr><td><code id="geom_subgene_label_+3A_reflow">reflow</code></td>
<td>
<p>If <code>TRUE</code>, text will be reflowed (wrapped) to better fit the
subgene.</p>
</td></tr>
<tr><td><code id="geom_subgene_label_+3A_height">height</code></td>
<td>
<p><code>grid::unit()</code> object giving the maximum height of the text.
Defaults to 3 mm, which is the default height of gene arrows (and therefore
of subgenes) drawn with <code>geom_gene_arrow()</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>geom_subgene_label()</code> uses the 'ggfittext' package to fit text to genes.
All text drawing options available in <code>ggfittext::geom_fit_text()</code> (growing,
reflowing, etc.) are also available here. For full details on how these
options work, see the documentation for <code>ggfittext::geom_fit_text()</code>.
</p>
<p>Standard 'ggplot2' aesthetics for text are supported (see Aesthetics.)
</p>


<h3>Aesthetics</h3>


<ul>
<li><p> xsubmin,xsubmax (start and end of the subgene; required)
</p>
</li>
<li><p> y (molecule; required)
</p>
</li>
<li><p> colour
</p>
</li>
<li><p> size
</p>
</li>
<li><p> alpha
</p>
</li>
<li><p> family
</p>
</li>
<li><p> fontface
</p>
</li>
<li><p> angle
</p>
</li></ul>


<hr>
<h2 id='make_alignment_dummies'>Prepare dummy data to visually align a single gene across faceted molecules</h2><span id='topic+make_alignment_dummies'></span>

<h3>Description</h3>

<p><code>make_alignment_dummies()</code> helps you to visually align genes across
molecules that have been faceted with a free x scale. The output of this
function is a data frame of dummy genes. If these dummy genes are added to a
'ggplot2' plot with <code>ggplot::geom_blank()</code>, they will extend the x axis
range in such a way that the start or end of a selected gene is visually
aligned across the facets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_alignment_dummies(data, mapping, on, side = "left")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_alignment_dummies_+3A_data">data</code></td>
<td>
<p>Data frame of genes. This is almost certainly the same data frame
that will later be passed to <code>ggplot2::ggplot()</code>.</p>
</td></tr>
<tr><td><code id="make_alignment_dummies_+3A_mapping">mapping</code></td>
<td>
<p>Aesthetic mapping, created with <code>ggplot2::aes()</code>. Must
contain the following aesthetics: <code>xmin</code>, <code>xmax</code>, <code>y</code>, and <code>id</code> (a unique
identifier for each gene).</p>
</td></tr>
<tr><td><code id="make_alignment_dummies_+3A_on">on</code></td>
<td>
<p>Name of gene to be visually aligned across facets. This gene must
be present in 'data', in the column mapped to the <code>id</code> aesthetic.</p>
</td></tr>
<tr><td><code id="make_alignment_dummies_+3A_side">side</code></td>
<td>
<p>Should the visual alignment be of the 'left' (default) or
'right' side of the gene?</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
dummies &lt;- make_alignment_dummies(example_genes, ggplot2::aes(xmin = start,
  xmax = end, y = molecule, id = gene), on = "genE")

ggplot2::ggplot(example_genes, ggplot2::aes(xmin = start, xmax = end,
    y = molecule, fill = gene)) +
  geom_gene_arrow() +
  ggplot2::geom_blank(data = dummies) +
  ggplot2::facet_wrap(~ molecule, scales = "free", ncol = 1)

</code></pre>

<hr>
<h2 id='theme_genes'>A 'ggplot2' theme for drawing gene maps</h2><span id='topic+theme_genes'></span><span id='topic+theme_genes_flipped'></span>

<h3>Description</h3>

<p>This theme removes extraneous plot elements for drawing an
'arrows-on-a-string' style gene map in 'ggplot2'.<code>theme_genes_flipped()</code> is
like <code>theme_genes()</code>, but for flipped coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_genes()

theme_genes_flipped()
</code></pre>


<h3>Details</h3>

<p>This theme removes strip text (the text that labels facets when you use
<code>ggplot2::facet_wrap()</code> or <code>ggplot::facet_grid()</code>). This makes it easier to
draw molecules on different x scales by setting the y aesthetic to the
molecule, then faceting with <code>facet_grid( ~ molecule, scales = "free")</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geom_gene_arrow">geom_gene_arrow()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ggplot2::ggplot(example_genes, ggplot2::aes(xmin = start, xmax = end,
                                            y = molecule, fill = gene)) +
geom_gene_arrow() +
ggplot2::facet_wrap(~ molecule, scales = "free") +
theme_genes()

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
