<!DOCTYPE html><html><head><title>Help for package ggmugs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ggmugs}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ggmugs'><p>Visualizing Multiple Genetic Studies</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Visualization of Multiple Genome-Wide Association Study Summary
Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.6.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A 'grammar of graphics' approach for visualizing summary statistics from multiple Genome-wide Association Studies (GWAS). It offers geneticists, bioinformaticians, and researchers a powerful yet flexible tool for illustrating complex genetic associations using data from various GWAS datasets. The visualizations can be extensively customized, facilitating detailed comparative analysis across different genetic studies. Reference: Uffelmann, E. et al. (2021) &lt;<a href="https://doi.org/10.1038%2Fs43586-021-00056-9">doi:10.1038/s43586-021-00056-9</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Suggests:</td>
<td>spelling, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, dplyr, ggplot2, purrr, tibble, tidyr</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-06 19:14:04 UTC; wanjun</td>
</tr>
<tr>
<td>Author:</td>
<td>Wanjun Gu <a href="https://orcid.org/0000-0002-7342-7000"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wanjun Gu &lt;wanjun.gu@ucsf.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-07 08:00:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='ggmugs'>Visualizing Multiple Genetic Studies</h2><span id='topic+ggmugs'></span>

<h3>Description</h3>

<p>This function processes and combines summary statistics from multiple genetic studies and
creates a visualization for all studies. The genetic loci are colored based on three significance
thresholds to facilitate the visualization of highly significant genomic regions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggmugs(
  study_name = c("sumstat1", "sumstat2", "sumstat3", "sumstat4", "sumstat5"),
  summary_stat = c("data/sumstat1.txt", "data/sumstat2.txt", "data/sumstat3.txt",
    "data/sumstat4.txt", "data/sumstat5.txt"),
  p1 = 0.001,
  p2 = 5e-05,
  p3 = 1e-08,
  color1 = "#FFFFE0",
  color2 = "#FFC300",
  color3 = "#FF5733"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggmugs_+3A_study_name">study_name</code></td>
<td>
<p>A character vector of names for the studies.</p>
</td></tr>
<tr><td><code id="ggmugs_+3A_summary_stat">summary_stat</code></td>
<td>
<p>A character vector of file paths where each path points to the summary
statistics data file for the corresponding study. Files should be in a tabular format
readable by 'fread' from the 'data.table' package.
The files should contain 3 fields: 'chr' (Chromosome), 'pos' (chromosome position), and 'p' (association p-value).
The positions of multiple GWAS summary statistics should have consistent genome builds.</p>
</td></tr>
<tr><td><code id="ggmugs_+3A_p1">p1</code></td>
<td>
<p>The first significance level threshold for p-values (default is 1e-3).</p>
</td></tr>
<tr><td><code id="ggmugs_+3A_p2">p2</code></td>
<td>
<p>The second, more stringent significance level threshold for p-values (default is 5e-5).</p>
</td></tr>
<tr><td><code id="ggmugs_+3A_p3">p3</code></td>
<td>
<p>The most stringent significance level threshold for p-values (default is 1e-8).</p>
</td></tr>
<tr><td><code id="ggmugs_+3A_color1">color1</code></td>
<td>
<p>The color for points below the first significance level (default is &quot;#FFFFE0&quot;).</p>
</td></tr>
<tr><td><code id="ggmugs_+3A_color2">color2</code></td>
<td>
<p>The color for points between the first and second significance levels (default is &quot;#FFC300&quot;).</p>
</td></tr>
<tr><td><code id="ggmugs_+3A_color3">color3</code></td>
<td>
<p>The color for points above the second significance level (default is &quot;#FF5733&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggplot' object representing the visualization with the specified data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### NOT RUN
# ggmugs(
#   study_name = c("study1", "study2", "study3", "study4", "study5"),
#   summary_stat = c("https://raw.githubusercontent.com/Broccolito/ggmugs_data/main/sumstat1.txt",
#                    "https://raw.githubusercontent.com/Broccolito/ggmugs_data/main/sumstat2.txt",
#                    "https://raw.githubusercontent.com/Broccolito/ggmugs_data/main/sumstat3.txt",
#                    "https://raw.githubusercontent.com/Broccolito/ggmugs_data/main/sumstat4.txt",
#                    "https://raw.githubusercontent.com/Broccolito/ggmugs_data/main/sumstat5.txt"),
#   p1 = 1e-4,
#   p2 = 1e-6,
#   p3 = 1e-8,
#   color1 = "#FFFFE0",
#   color2 = "#FFC300",
#   color3 = "#FF5733"
# )
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
