<!DOCTYPE html><html><head><title>Help for package ggpicrust2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ggpicrust2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#compare_daa_results'><p>Compare the Consistency of Statistically Significant Features</p></a></li>
<li><a href='#compare_metagenome_results'><p>Compare Metagenome Results</p></a></li>
<li><a href='#daa_annotated_results_df'><p>Differentially Abundant Analysis Results with Annotation</p></a></li>
<li><a href='#daa_results_df'><p>DAA Results Dataset</p></a></li>
<li><a href='#ggpicrust2'><p>This function integrates pathway name/description annotations, ten of the most advanced differential abundance (DA) methods, and visualization of DA results.</p></a></li>
<li><a href='#import_MicrobiomeAnalyst_daa_results'><p>Import Differential Abundance Analysis (DAA) results from MicrobiomeAnalyst</p></a></li>
<li><a href='#kegg_abundance'><p>KEGG Abundance Dataset</p></a></li>
<li><a href='#ko_abundance'><p>KO Abundance Dataset</p></a></li>
<li><a href='#ko2kegg_abundance'><p>Convert KO abundance in picrust2 export files to KEGG pathway abundance</p></a></li>
<li><a href='#metacyc_abundance'><p>MetaCyc Abundance Dataset</p></a></li>
<li><a href='#metadata'><p>Metadata for ggpicrust2 Demonstration</p></a></li>
<li><a href='#pathway_annotation'><p>Pathway information annotation of &quot;EC&quot;, &quot;KO&quot;, &quot;MetaCyc&quot; pathway</p></a></li>
<li><a href='#pathway_daa'><p>Predictional functional patwhay differential abundance (DA)</p></a></li>
<li><a href='#pathway_errorbar'><p>The function pathway_errorbar() is used to visualize the results of functional pathway differential abundance analysis as error bar plots.</p></a></li>
<li><a href='#pathway_heatmap'><p>Create pathway heatmap</p></a></li>
<li><a href='#pathway_pca'><p>Perform Principal Component Analysis (PCA) on functional pathway abundance data and create visualizations of the PCA results.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Make 'PICRUSt2' Output Analysis and Visualization Easier</td>
</tr>
<tr>
<td>Version:</td>
<td>1.7.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a convenient way to analyze and visualize 'PICRUSt2' output with pre-defined plots and functions. Allows for generating statistical plots about microbiome functional predictions and offers customization options. Features a one-click option for creating publication-level plots, saving time and effort in producing professional-grade figures. Streamlines the 'PICRUSt2' analysis and visualization process. For more details, see Yang et al. (2023) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtad470">doi:10.1093/bioinformatics/btad470</a>&gt;.</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/cafferychen777/ggpicrust2/issues">https://github.com/cafferychen777/ggpicrust2/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/cafferychen777/ggpicrust2">https://github.com/cafferychen777/ggpicrust2</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>ALDEx2, aplot, DESeq2, dplyr, edgeR, GGally, ggplot2, grid,
ggh4x, lefser, limma, Maaslin2, metagenomeSeq, MicrobiomeStat,
readr, SummarizedExperiment, tibble, tidyr, ggprism, patchwork,
circlize</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Biobase, KEGGREST, ComplexHeatmap, ggplotify, BiocGenerics,
magrittr, utils, knitr, rmarkdown</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-08 11:38:19 UTC; apple</td>
</tr>
<tr>
<td>Author:</td>
<td>Chen Yang [aut, cre],
  Liangliang Zhang [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chen Yang &lt;cafferychen7850@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-08 16:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='compare_daa_results'>Compare the Consistency of Statistically Significant Features</h2><span id='topic+compare_daa_results'></span>

<h3>Description</h3>

<p>This function compares the consistency and inconsistency of statistically significant features obtained
using different methods in 'pathway_daa' from the 'ggpicrust2' package. It creates a report showing the number of common and
different features identified by each method, and the features themselves.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare_daa_results_+3A_daa_results_list">daa_results_list</code></td>
<td>
<p>A list of data frames containing statistically significant features obtained using different methods.</p>
</td></tr>
<tr><td><code id="compare_daa_results_+3A_method_names">method_names</code></td>
<td>
<p>A character vector of names for each method used.</p>
</td></tr>
<tr><td><code id="compare_daa_results_+3A_p_values_threshold">p_values_threshold</code></td>
<td>
<p>A numeric value representing the threshold for the p-values. Features with p-values less than this
threshold are considered statistically significant. Default is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the comparison results. The data frame has the following columns:
</p>

<ul>
<li> <p><code>method</code>: The name of the method.
</p>
</li>
<li> <p><code>num_features</code>: The total number of statistically significant features obtained by the method.
</p>
</li>
<li> <p><code>num_common_features</code>: The number of features that are common to other methods.
</p>
</li>
<li> <p><code>num_diff_features</code>: The number of features that are different from other methods.
</p>
</li>
<li> <p><code>diff_features</code>: The names of the features that are different from other methods.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
library(magrittr)
library(ggpicrust2)
library(tibble)
data("metacyc_abundance")
data("metadata")

# Run pathway_daa function for multiple methods
methods &lt;- c("DESeq2", "edgeR","Maaslin2")
daa_results_list &lt;- lapply(methods, function(method) {
pathway_daa(abundance = metacyc_abundance %&gt;% column_to_rownames("pathway"),
metadata = metadata, group = "Environment", daa_method = method)
})

names(daa_results_list) &lt;- methods
# Correct Maaslin2 feature names by replacing dots with hyphens.
# Note: When using Maaslin2 as the differential abundance analysis method,
# it modifies the original feature names by replacing hyphens (-) with dots (.).
# This replacement can cause inconsistencies when trying to compare results from Maaslin2
# with those from other methods that do not modify feature names.
# Therefore, this line of code reverses that replacement, converting the dots back into
# hyphens for accurate and consistent comparisons across different methods.
daa_results_list[["Maaslin2"]]$feature &lt;- gsub("\\.", "-", daa_results_list[["Maaslin2"]]$feature)

# Compare results across different methods
comparison_results &lt;- compare_daa_results(daa_results_list = daa_results_list,
method_names = c("DESeq2", "edgeR", "Maaslin2"))

</code></pre>

<hr>
<h2 id='compare_metagenome_results'>Compare Metagenome Results</h2><span id='topic+compare_metagenome_results'></span>

<h3>Description</h3>

<p>Compare Metagenome Results
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare_metagenome_results_+3A_metagenomes">metagenomes</code></td>
<td>
<p>A list of metagenomes matrices with rows as KOs and columns as samples.
Each matrix in the list should correspond to a different metagenome.</p>
</td></tr>
<tr><td><code id="compare_metagenome_results_+3A_names">names</code></td>
<td>
<p>A vector of names for the metagenomes in the same order as in the 'metagenomes' list.</p>
</td></tr>
<tr><td><code id="compare_metagenome_results_+3A_daa_method">daa_method</code></td>
<td>
<p>A character specifying the method for differential abundance analysis (DAA).
Possible choices are: &quot;ALDEx2&quot;, &quot;DESeq2&quot;, &quot;edgeR&quot;, &quot;limma voom&quot;, &quot;metagenomeSeq&quot;, &quot;LinDA&quot;,
&quot;Maaslin2&quot;, and &quot;Lefse&quot;. The default is &quot;ALDEx2&quot;.</p>
</td></tr>
<tr><td><code id="compare_metagenome_results_+3A_p.adjust">p.adjust</code></td>
<td>
<p>A character specifying the method for p-value adjustment.
Possible choices are: &quot;BH&quot; (Benjamini-Hochberg), &quot;holm&quot;, &quot;bonferroni&quot;, &quot;hochberg&quot;, &quot;fdr&quot;, and &quot;none&quot;.
The default is &quot;BH&quot;.</p>
</td></tr>
<tr><td><code id="compare_metagenome_results_+3A_reference">reference</code></td>
<td>
<p>A character specifying the reference group level for DAA.
This parameter is used when there are more than two groups. The default is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two elements:
</p>

<ul>
<li><p> &quot;daa&quot;: a list of results from the 'pathway_daa' function. Each result is a data frame
containing the differential abundance analysis results with columns for the feature ID,
the test statistic, the raw p-value, and the adjusted p-value.
</p>
</li>
<li><p> &quot;correlation&quot;: a list with two elements: &quot;cor_matrix&quot; and &quot;p_matrix&quot;, which are
matrices of Spearman correlation coefficients and their corresponding p-values, respectively,
between every pair of metagenomes.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)
library(ComplexHeatmap)
# Generate example data
set.seed(123)
# First metagenome
metagenome1 &lt;- abs(matrix(rnorm(1000), nrow = 100, ncol = 10))
rownames(metagenome1) &lt;- paste0("KO", 1:100)
colnames(metagenome1) &lt;- paste0("sample", 1:10)
# Second metagenome
metagenome2 &lt;- abs(matrix(rnorm(1000), nrow = 100, ncol = 10))
rownames(metagenome2) &lt;- paste0("KO", 1:100)
colnames(metagenome2) &lt;- paste0("sample", 1:10)
# Put the metagenomes into a list
metagenomes &lt;- list(metagenome1, metagenome2)
# Define names
names &lt;- c("metagenome1", "metagenome2")
# Call the function
results &lt;- compare_metagenome_results(metagenomes, names, daa_method = "LinDA")
# Print the correlation matrix
print(results$correlation$cor_matrix)
# Print the p-value matrix
print(results$correlation$p_matrix)

</code></pre>

<hr>
<h2 id='daa_annotated_results_df'>Differentially Abundant Analysis Results with Annotation</h2><span id='topic+daa_annotated_results_df'></span>

<h3>Description</h3>

<p>This is a result dataset after processing 'kegg_abundance' through
the 'pathway_daa' with the LinDA method and further annotation with 'pathway_annotation'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>daa_annotated_results_df
</code></pre>


<h3>Format</h3>

<p>A data frame with 10 variables:
</p>

<dl>
<dt>adj_method</dt><dd><p>Method used for adjusting p-values.</p>
</dd>
<dt>feature</dt><dd><p>Feature being tested.</p>
</dd>
<dt>group1</dt><dd><p>One group in the comparison.</p>
</dd>
<dt>group2</dt><dd><p>The other group in the comparison.</p>
</dd>
<dt>method</dt><dd><p>Statistical test used.</p>
</dd>
<dt>p_adjust</dt><dd><p>Adjusted p-value.</p>
</dd>
<dt>p_values</dt><dd><p>P-values from the statistical test.</p>
</dd>
<dt>pathway_class</dt><dd><p>Class of the pathway.</p>
</dd>
<dt>pathway_description</dt><dd><p>Description of the pathway.</p>
</dd>
<dt>pathway_map</dt><dd><p>Map of the pathway.</p>
</dd>
<dt>pathway_name</dt><dd><p>Name of the pathway.</p>
</dd>
</dl>



<h3>Source</h3>

<p>From ggpicrust2 package demonstration.
</p>


<h3>References</h3>

<p>Douglas GM, Maffei VJ, Zaneveld J, Yurgel SN, Brown JR, Taylor CM, Huttenhower C, Langille MGI. PICRUSt2 for prediction of metagenome functions. Nat Biotechnol. 2020.
</p>

<hr>
<h2 id='daa_results_df'>DAA Results Dataset</h2><span id='topic+daa_results_df'></span>

<h3>Description</h3>

<p>This dataset is the result of processing 'kegg_abundance' through the 'LinDA' method in the 'pathway_daa' function.
It includes information about the feature, groups compared, p values, and method used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>daa_results_df
</code></pre>


<h3>Format</h3>

<p>A data frame with columns:
</p>

<dl>
<dt>adj_method</dt><dd><p>Method used for p-value adjustment.</p>
</dd>
<dt>feature</dt><dd><p>The feature (pathway) being compared.</p>
</dd>
<dt>group1</dt><dd><p>The first group in the comparison.</p>
</dd>
<dt>group2</dt><dd><p>The second group in the comparison.</p>
</dd>
<dt>method</dt><dd><p>The method used for the comparison.</p>
</dd>
<dt>p_adjust</dt><dd><p>The adjusted p-value from the comparison.</p>
</dd>
<dt>p_values</dt><dd><p>The raw p-value from the comparison.</p>
</dd>
</dl>



<h3>Source</h3>

<p>From ggpicrust2 package demonstration.
</p>


<h3>References</h3>

<p>Douglas GM, Maffei VJ, Zaneveld J, Yurgel SN, Brown JR, Taylor CM, Huttenhower C, Langille MGI. PICRUSt2 for prediction of metagenome functions. Nat Biotechnol. 2020.
</p>

<hr>
<h2 id='ggpicrust2'>This function integrates pathway name/description annotations, ten of the most advanced differential abundance (DA) methods, and visualization of DA results.</h2><span id='topic+ggpicrust2'></span>

<h3>Description</h3>

<p>This function integrates pathway name/description annotations, ten of the most advanced differential abundance (DA) methods, and visualization of DA results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggpicrust2(
  file = NULL,
  data = NULL,
  metadata,
  group,
  pathway,
  daa_method = "ALDEx2",
  ko_to_kegg = FALSE,
  p.adjust = "BH",
  order = "group",
  p_values_bar = TRUE,
  x_lab = NULL,
  select = NULL,
  reference = NULL,
  colors = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggpicrust2_+3A_file">file</code></td>
<td>
<p>A character string representing the file path of the input file containing KO abundance data in picrust2 export format. The input file should have KO identifiers in the first column and sample identifiers in the first row. The remaining cells should contain the abundance values for each KO-sample pair.</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_data">data</code></td>
<td>
<p>An optional data.frame containing KO abundance data in the same format as the input file. If provided, the function will use this data instead of reading from the file. By default, this parameter is set to NULL.</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_metadata">metadata</code></td>
<td>
<p>A tibble, consisting of sample information</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_group">group</code></td>
<td>
<p>A character, name of the group</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_pathway">pathway</code></td>
<td>
<p>A character, consisting of &quot;EC&quot;, &quot;KO&quot;, &quot;MetaCyc&quot;</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_daa_method">daa_method</code></td>
<td>
<p>a character specifying the method for differential abundance analysis, default is &quot;ALDEx2&quot;, choices are:
- &quot;ALDEx2&quot;: ANOVA-Like Differential Expression tool for high throughput sequencing data
- &quot;DESeq2&quot;: Differential expression analysis based on the negative binomial distribution using DESeq2
- &quot;edgeR&quot;: Exact test for differences between two groups of negative-binomially distributed counts using edgeR
- &quot;limma voom&quot;: Limma-voom framework for the analysis of RNA-seq data
- &quot;metagenomeSeq&quot;: Fit logistic regression models to test for differential abundance between groups using metagenomeSeq
- &quot;LinDA&quot;: Linear models for differential abundance analysis of microbiome compositional data
- &quot;Maaslin2&quot;: Multivariate Association with Linear Models (MaAsLin2) for differential abundance analysis</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_ko_to_kegg">ko_to_kegg</code></td>
<td>
<p>A character to control the conversion of KO abundance to KEGG abundance</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_p.adjust">p.adjust</code></td>
<td>
<p>a character specifying the method for p-value adjustment, default is &quot;BH&quot;, choices are:
- &quot;BH&quot;: Benjamini-Hochberg correction
- &quot;holm&quot;: Holm's correction
- &quot;bonferroni&quot;: Bonferroni correction
- &quot;hochberg&quot;: Hochberg's correction
- &quot;fdr&quot;: False discovery rate correction
- &quot;none&quot;: No p-value adjustment.</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_order">order</code></td>
<td>
<p>A character to control the order of the main plot rows</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_p_values_bar">p_values_bar</code></td>
<td>
<p>A character to control if the main plot has the p_values bar</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_x_lab">x_lab</code></td>
<td>
<p>A character to control the x-axis label name, you can choose from &quot;feature&quot;,&quot;pathway_name&quot; and &quot;description&quot;</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_select">select</code></td>
<td>
<p>A vector consisting of pathway names to be selected</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_reference">reference</code></td>
<td>
<p>A character, a reference group level for several DA methods</p>
</td></tr>
<tr><td><code id="ggpicrust2_+3A_colors">colors</code></td>
<td>
<p>A vector consisting of colors number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>daa.results.df, a dataframe of DA results
A list of sub-lists, each containing a ggplot2 plot ('plot') and a dataframe of differential abundance results ('results') for a specific DA method.
Each plot visualizes the differential abundance results of a specific DA method, and the corresponding dataframe contains the results used to create the plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load necessary data: abundance data and metadata
abundance_file &lt;- "path/to/your/abundance_file.tsv"
metadata &lt;- read.csv("path/to/your/metadata.csv")

# Run ggpicrust2 with input file path
results_file_input &lt;- ggpicrust2(file = abundance_file,
                                 metadata = metadata,
                                 group = "your_group_column",
                                 pathway = "KO",
                                 daa_method = "LinDA",
                                 ko_to_kegg = "TRUE",
                                 order = "pathway_class",
                                 p_values_bar = TRUE,
                                 x_lab = "pathway_name")

# Run ggpicrust2 with imported data.frame
abundance_data &lt;- read_delim(abundance_file, delim="\t", col_names=TRUE, trim_ws=TRUE)

# Run ggpicrust2 with input data
results_data_input &lt;- ggpicrust2(data = abundance_data,
                                 metadata = metadata,
                                 group = "your_group_column",
                                 pathway = "KO",
                                 daa_method = "LinDA",
                                 ko_to_kegg = "TRUE",
                                 order = "pathway_class",
                                 p_values_bar = TRUE,
                                 x_lab = "pathway_name")

# Access the plot and results dataframe for the first DA method
example_plot &lt;- results_file_input[[1]]$plot
example_results &lt;- results_file_input[[1]]$results

# Use the example data in ggpicrust2 package
data(ko_abundance)
data(metadata)
results_file_input &lt;- ggpicrust2(data = ko_abundance,
                                 metadata = metadata,
                                 group = "Environment",
                                 pathway = "KO",
                                 daa_method = "LinDA",
                                 ko_to_kegg = TRUE,
                                 order = "pathway_class",
                                 p_values_bar = TRUE,
                                 x_lab = "pathway_name")
# Analyze the EC or MetaCyc pathway
data(metacyc_abundance)
results_file_input &lt;- ggpicrust2(data = metacyc_abundance,
                                 metadata = metadata,
                                 group = "Environment",
                                 pathway = "MetaCyc",
                                 daa_method = "LinDA",
                                 ko_to_kegg = FALSE,
                                 order = "group",
                                 p_values_bar = TRUE,
                                 x_lab = "description")

## End(Not run)
</code></pre>

<hr>
<h2 id='import_MicrobiomeAnalyst_daa_results'>Import Differential Abundance Analysis (DAA) results from MicrobiomeAnalyst</h2><span id='topic+import_MicrobiomeAnalyst_daa_results'></span>

<h3>Description</h3>

<p>This function imports DAA results from an external platform such as MicrobiomeAnalyst. It can be used to compare the results obtained from different platforms.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_MicrobiomeAnalyst_daa_results_+3A_file_path">file_path</code></td>
<td>
<p>a character string specifying the path to the CSV file containing the DAA results from MicrobiomeAnalyst. If this parameter is NULL and no data frame is provided, an error will be thrown. Default is NULL.</p>
</td></tr>
<tr><td><code id="import_MicrobiomeAnalyst_daa_results_+3A_data">data</code></td>
<td>
<p>a data frame containing the DAA results from MicrobiomeAnalyst. If this parameter is NULL and no file path is provided, an error will be thrown. Default is NULL.</p>
</td></tr>
<tr><td><code id="import_MicrobiomeAnalyst_daa_results_+3A_method">method</code></td>
<td>
<p>a character string specifying the method used for the DAA. This will be added as a new column in the returned data frame. Default is &quot;MicrobiomeAnalyst&quot;.</p>
</td></tr>
<tr><td><code id="import_MicrobiomeAnalyst_daa_results_+3A_group_levels">group_levels</code></td>
<td>
<p>a character vector specifying the group levels for the DAA. This will be added as new columns in the returned data frame. Default is c(&quot;control&quot;, &quot;treatment&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing the DAA results from MicrobiomeAnalyst with additional columns for the method and group levels.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Assuming you have a CSV file named "DAA_results.csv" in your current directory
daa_results &lt;- import_MicrobiomeAnalyst_daa_results(file_path = "DAA_results.csv")

## End(Not run)

</code></pre>

<hr>
<h2 id='kegg_abundance'>KEGG Abundance Dataset</h2><span id='topic+kegg_abundance'></span>

<h3>Description</h3>

<p>A dataset derived from 'ko_abundance' by the function 'ko2kegg_abundance' in the ggpicrust2 package.
Each row corresponds to a KEGG pathway, and each column corresponds to a sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kegg_abundance
</code></pre>


<h3>Format</h3>

<p>A data frame where rownames are KEGG pathways and column names are individual sample names, including:
&quot;SRR11393730&quot;, &quot;SRR11393731&quot;, &quot;SRR11393732&quot;, &quot;SRR11393733&quot;, &quot;SRR11393734&quot;, &quot;SRR11393735&quot;, &quot;SRR11393736&quot;,
&quot;SRR11393737&quot;, &quot;SRR11393738&quot;, &quot;SRR11393739&quot;, &quot;SRR11393740&quot;, &quot;SRR11393741&quot;, &quot;SRR11393742&quot;, &quot;SRR11393743&quot;,
&quot;SRR11393744&quot;, &quot;SRR11393745&quot;, &quot;SRR11393746&quot;, &quot;SRR11393747&quot;, &quot;SRR11393748&quot;, &quot;SRR11393749&quot;, &quot;SRR11393750&quot;,
&quot;SRR11393751&quot;, &quot;SRR11393752&quot;, &quot;SRR11393753&quot;, &quot;SRR11393754&quot;, &quot;SRR11393755&quot;, &quot;SRR11393756&quot;, &quot;SRR11393757&quot;,
&quot;SRR11393758&quot;, &quot;SRR11393759&quot;, &quot;SRR11393760&quot;, &quot;SRR11393761&quot;, &quot;SRR11393762&quot;, &quot;SRR11393763&quot;, &quot;SRR11393764&quot;,
&quot;SRR11393765&quot;, &quot;SRR11393766&quot;, &quot;SRR11393767&quot;, &quot;SRR11393768&quot;, &quot;SRR11393769&quot;, &quot;SRR11393770&quot;, &quot;SRR11393771&quot;,
&quot;SRR11393772&quot;, &quot;SRR11393773&quot;, &quot;SRR11393774&quot;, &quot;SRR11393775&quot;, &quot;SRR11393776&quot;, &quot;SRR11393777&quot;, &quot;SRR11393778&quot;, &quot;SRR11393779&quot;
</p>


<h3>Source</h3>

<p>From ggpicrust2 package demonstration.
</p>


<h3>References</h3>

<p>Douglas GM, Maffei VJ, Zaneveld J, Yurgel SN, Brown JR, Taylor CM, Huttenhower C, Langille MGI. PICRUSt2 for prediction of metagenome functions. Nat Biotechnol. 2020.
</p>

<hr>
<h2 id='ko_abundance'>KO Abundance Dataset</h2><span id='topic+ko_abundance'></span>

<h3>Description</h3>

<p>This is a demonstration dataset from the ggpicrust2 package, representing the output of PICRUSt2.
Each row represents a KO (KEGG Orthology) group, and each column corresponds to a sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ko_abundance
</code></pre>


<h3>Format</h3>

<p>A data frame where rownames are KO groups and column names include #NAME and individual sample names, such as:
&quot;#NAME&quot;, &quot;SRR11393730&quot;, &quot;SRR11393731&quot;, &quot;SRR11393732&quot;, &quot;SRR11393733&quot;, &quot;SRR11393734&quot;, &quot;SRR11393735&quot;, &quot;SRR11393736&quot;,
&quot;SRR11393737&quot;, &quot;SRR11393738&quot;, &quot;SRR11393739&quot;, &quot;SRR11393740&quot;, &quot;SRR11393741&quot;, &quot;SRR11393742&quot;, &quot;SRR11393743&quot;,
&quot;SRR11393744&quot;, &quot;SRR11393745&quot;, &quot;SRR11393746&quot;, &quot;SRR11393747&quot;, &quot;SRR11393748&quot;, &quot;SRR11393749&quot;, &quot;SRR11393750&quot;,
&quot;SRR11393751&quot;, &quot;SRR11393752&quot;, &quot;SRR11393753&quot;, &quot;SRR11393754&quot;, &quot;SRR11393755&quot;, &quot;SRR11393756&quot;, &quot;SRR11393757&quot;,
&quot;SRR11393758&quot;, &quot;SRR11393759&quot;, &quot;SRR11393760&quot;, &quot;SRR11393761&quot;, &quot;SRR11393762&quot;, &quot;SRR11393763&quot;, &quot;SRR11393764&quot;,
&quot;SRR11393765&quot;, &quot;SRR11393766&quot;, &quot;SRR11393767&quot;, &quot;SRR11393768&quot;, &quot;SRR11393769&quot;, &quot;SRR11393770&quot;, &quot;SRR11393771&quot;,
&quot;SRR11393772&quot;, &quot;SRR11393773&quot;, &quot;SRR11393774&quot;, &quot;SRR11393775&quot;, &quot;SRR11393776&quot;, &quot;SRR11393777&quot;, &quot;SRR11393778&quot;, &quot;SRR11393779&quot;
</p>


<h3>Source</h3>

<p>From ggpicrust2 package demonstration.
</p>


<h3>References</h3>

<p>Douglas GM, Maffei VJ, Zaneveld J, Yurgel SN, Brown JR, Taylor CM, Huttenhower C, Langille MGI. PICRUSt2 for prediction of metagenome functions. Nat Biotechnol. 2020.
</p>

<hr>
<h2 id='ko2kegg_abundance'>Convert KO abundance in picrust2 export files to KEGG pathway abundance</h2><span id='topic+ko2kegg_abundance'></span>

<h3>Description</h3>

<p>This function takes a file containing KO (KEGG Orthology) abundance data in picrust2 export format and converts it to KEGG pathway abundance data.
The input file should be in .tsv, .txt, or .csv format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ko2kegg_abundance(file = NULL, data = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ko2kegg_abundance_+3A_file">file</code></td>
<td>
<p>A character string representing the file path of the input file containing KO abundance data in picrust2 export format. The input file should have KO identifiers in the first column and sample identifiers in the first row. The remaining cells should contain the abundance values for each KO-sample pair.</p>
</td></tr>
<tr><td><code id="ko2kegg_abundance_+3A_data">data</code></td>
<td>
<p>An optional data.frame containing KO abundance data in the same format as the input file. If provided, the function will use this data instead of reading from the file. By default, this parameter is set to NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with KEGG pathway abundance values. Rows represent KEGG pathways, identified by their KEGG pathway IDs. Columns represent samples, identified by their sample IDs from the input file. Each cell contains the abundance of a specific KEGG pathway in a given sample, calculated by summing the abundances of the corresponding KOs in the input file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ggpicrust2)
library(readr)

# Example 1: Demonstration with a hypothetical input file

# Prepare an input file path
input_file &lt;- "path/to/your/picrust2/results/pred_metagenome_unstrat.tsv"

# Run ko2kegg_abundance function
kegg_abundance &lt;- ko2kegg_abundance(file = input_file)

# Alternatively, read the data from a file and use the data argument
file_path &lt;- "path/to/your/picrust2/results/pred_metagenome_unstrat.tsv"
ko_abundance &lt;- read_delim(file_path, delim = "\t")
kegg_abundance &lt;- ko2kegg_abundance(data = ko_abundance)

# Example 2: Working with real data
# In this case, we're using an existing dataset from the ggpicrust2 package.

# Load the data
data(ko_abundance)

# Apply the ko2kegg_abundance function to our real dataset
kegg_abundance &lt;- ko2kegg_abundance(data = ko_abundance)

## End(Not run)
</code></pre>

<hr>
<h2 id='metacyc_abundance'>MetaCyc Abundance Dataset</h2><span id='topic+metacyc_abundance'></span>

<h3>Description</h3>

<p>This is a demonstration dataset from the ggpicrust2 package, representing the output of PICRUSt2.
Each row represents a MetaCyc pathway, and each column corresponds to a sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metacyc_abundance
</code></pre>


<h3>Format</h3>

<p>A data frame where rownames are MetaCyc pathways and column names include &quot;pathway&quot; and individual sample names, such as:
&quot;pathway&quot;, &quot;SRR11393730&quot;, &quot;SRR11393731&quot;, &quot;SRR11393732&quot;, &quot;SRR11393733&quot;, &quot;SRR11393734&quot;, &quot;SRR11393735&quot;, &quot;SRR11393736&quot;,
&quot;SRR11393737&quot;, &quot;SRR11393738&quot;, &quot;SRR11393739&quot;, &quot;SRR11393740&quot;, &quot;SRR11393741&quot;, &quot;SRR11393742&quot;, &quot;SRR11393743&quot;,
&quot;SRR11393744&quot;, &quot;SRR11393745&quot;, &quot;SRR11393746&quot;, &quot;SRR11393747&quot;, &quot;SRR11393748&quot;, &quot;SRR11393749&quot;, &quot;SRR11393750&quot;,
&quot;SRR11393751&quot;, &quot;SRR11393752&quot;, &quot;SRR11393753&quot;, &quot;SRR11393754&quot;, &quot;SRR11393755&quot;, &quot;SRR11393756&quot;, &quot;SRR11393757&quot;,
&quot;SRR11393758&quot;, &quot;SRR11393759&quot;, &quot;SRR11393760&quot;, &quot;SRR11393761&quot;, &quot;SRR11393762&quot;, &quot;SRR11393763&quot;, &quot;SRR11393764&quot;,
&quot;SRR11393765&quot;, &quot;SRR11393766&quot;, &quot;SRR11393767&quot;, &quot;SRR11393768&quot;, &quot;SRR11393769&quot;, &quot;SRR11393770&quot;, &quot;SRR11393771&quot;,
&quot;SRR11393772&quot;, &quot;SRR11393773&quot;, &quot;SRR11393774&quot;, &quot;SRR11393775&quot;, &quot;SRR11393776&quot;, &quot;SRR11393777&quot;, &quot;SRR11393778&quot;, &quot;SRR11393779&quot;
</p>


<h3>Source</h3>

<p>From ggpicrust2 package demonstration.
</p>


<h3>References</h3>

<p>Douglas GM, Maffei VJ, Zaneveld J, Yurgel SN, Brown JR, Taylor CM, Huttenhower C, Langille MGI. PICRUSt2 for prediction of metagenome functions. Nat Biotechnol. 2020.
</p>

<hr>
<h2 id='metadata'>Metadata for ggpicrust2 Demonstration</h2><span id='topic+metadata'></span>

<h3>Description</h3>

<p>This is a demonstration dataset from the ggpicrust2 package. It provides the metadata
required for the demonstration functions in the package. The dataset includes
environmental information for each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metadata
</code></pre>


<h3>Format</h3>

<p>A tibble with each row representing metadata for a sample.
</p>

<dl>
<dt>Sample1</dt><dd><p>Metadata for Sample1, including Environment</p>
</dd>
<dt>Sample2</dt><dd><p>Metadata for Sample2, including Environment</p>
</dd>
<dt>...</dt><dd><p>...</p>
</dd>
</dl>



<h3>Source</h3>

<p>ggpicrust2 package demonstration.
</p>


<h3>References</h3>

<p>Douglas GM, Maffei VJ, Zaneveld J, Yurgel SN, Brown JR, Taylor CM, Huttenhower C, Langille MGI. PICRUSt2 for prediction of metagenome functions. Nat Biotechnol. 2020.
</p>

<hr>
<h2 id='pathway_annotation'>Pathway information annotation of &quot;EC&quot;, &quot;KO&quot;, &quot;MetaCyc&quot; pathway</h2><span id='topic+pathway_annotation'></span>

<h3>Description</h3>

<p>This function has two primary use cases:
1. Annotating pathway information using the output file from PICRUSt2.
2. Annotating pathway information from the output of 'pathway_daa' function, and converting KO abundance to KEGG pathway abundance when 'ko_to_kegg' is set to TRUE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway_annotation(
  file = NULL,
  pathway = NULL,
  daa_results_df = NULL,
  ko_to_kegg = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway_annotation_+3A_file">file</code></td>
<td>
<p>A character, address to store PICRUSt2 export files. Provide this parameter when using the function for the first use case.</p>
</td></tr>
<tr><td><code id="pathway_annotation_+3A_pathway">pathway</code></td>
<td>
<p>A character, consisting of &quot;EC&quot;, &quot;KO&quot;, &quot;MetaCyc&quot;</p>
</td></tr>
<tr><td><code id="pathway_annotation_+3A_daa_results_df">daa_results_df</code></td>
<td>
<p>A data frame, output of pathway_daa. Provide this parameter when using the function for the second use case.</p>
</td></tr>
<tr><td><code id="pathway_annotation_+3A_ko_to_kegg">ko_to_kegg</code></td>
<td>
<p>A logical, decide if convert KO abundance to KEGG pathway abundance. Default is FALSE. Set to TRUE when using the function for the second use case.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing pathway annotation information. The data frame has the following columns:
</p>

<ul>
<li> <p><code>feature</code>: The feature ID of the pathway (e.g., KO, EC, or MetaCyc ID).
</p>
</li>
<li> <p><code>description</code>: The description or name of the pathway.
</p>
</li>
<li><p> Other columns depending on the input parameters and type of pathway.
</p>
</li></ul>

<p>If <code>ko_to_kegg</code> is set to TRUE, the output data frame will also include the following columns:
</p>

<ul>
<li> <p><code>pathway_name</code>: The name of the KEGG pathway.
</p>
</li>
<li> <p><code>pathway_description</code>: The description of the KEGG pathway.
</p>
</li>
<li> <p><code>pathway_class</code>: The class of the KEGG pathway.
</p>
</li>
<li> <p><code>pathway_map</code>: The KEGG pathway map ID.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Prepare the required input files and data frames
# Then, you can use the pathway_annotation function as follows:

# Use case 1: Annotating pathway information using the output file from PICRUSt2
result1 &lt;- pathway_annotation(file = "path/to/picrust2/export/file.txt",
                              pathway = "KO",
                              daa_results_df = NULL,
                              ko_to_kegg = FALSE)

# Use case 2: Annotating pathway information from the output of pathway_daa function
# and converting KO abundance to KEGG pathway abundance
# This use case will be demonstrated using both a hypothetical example, and a real dataset.

## Hypothetical example
hypothetical_daa_results_df &lt;- data.frame() # Replace this with your actual data frame
result2 &lt;- pathway_annotation(file = NULL,
                              pathway = "KO",
                              daa_results_df = hypothetical_daa_results_df,
                              ko_to_kegg = TRUE)

## Real dataset example
# Load the real dataset
data(daa_results_df)
result3 &lt;- pathway_annotation(file = NULL,
                              pathway = "KO",
                              daa_results_df = daa_results_df,
                              ko_to_kegg = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='pathway_daa'>Predictional functional patwhay differential abundance (DA)</h2><span id='topic+pathway_daa'></span>

<h3>Description</h3>

<p>Predictional functional patwhay differential abundance (DA)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway_daa(
  abundance,
  metadata,
  group,
  daa_method = "ALDEx2",
  select = NULL,
  p.adjust = "BH",
  reference = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway_daa_+3A_abundance">abundance</code></td>
<td>
<p>a data frame containing predicted functional pathway abundance, with pathways/features as rows and samples as columns. The column names of abundance should match the sample names in metadata. Pathway abundance values should be counts</p>
</td></tr>
<tr><td><code id="pathway_daa_+3A_metadata">metadata</code></td>
<td>
<p>a tibble containing samples information</p>
</td></tr>
<tr><td><code id="pathway_daa_+3A_group">group</code></td>
<td>
<p>a character specifying the group name for differential abundance analysis</p>
</td></tr>
<tr><td><code id="pathway_daa_+3A_daa_method">daa_method</code></td>
<td>
<p>a character specifying the method for differential abundance analysis, choices are:
- &quot;ALDEx2&quot;: ANOVA-Like Differential Expression tool for high throughput sequencing data
- &quot;DESeq2&quot;: Differential expression analysis based on the negative binomial distribution using DESeq2
- &quot;edgeR&quot;: Exact test for differences between two groups of negative-binomially distributed counts using edgeR
- &quot;limma voom&quot;: Limma-voom framework for the analysis of RNA-seq data
- &quot;metagenomeSeq&quot;: Fit logistic regression models to test for differential abundance between groups using metagenomeSeq
- &quot;LinDA&quot;: Linear models for differential abundance analysis of microbiome compositional data
- &quot;Maaslin2&quot;: Multivariate Association with Linear Models (MaAsLin2) for differential abundance analysis
- &quot;Lefser&quot;: Linear discriminant analysis (LDA) effect size algorithm for high-dimensional microbiome data</p>
</td></tr>
<tr><td><code id="pathway_daa_+3A_select">select</code></td>
<td>
<p>a vector containing sample names for analysis, if NULL all samples are included. This parameter can be used to specify which samples are included in the differential abundance analysis. Default is NULL.</p>
</td></tr>
<tr><td><code id="pathway_daa_+3A_p.adjust">p.adjust</code></td>
<td>
<p>a character specifying the method for p-value adjustment, choices are:
- &quot;BH&quot;: Benjamini-Hochberg correction
- &quot;holm&quot;: Holm's correction
- &quot;bonferroni&quot;: Bonferroni correction
- &quot;hochberg&quot;: Hochberg's correction
- &quot;fdr&quot;: False discovery rate correction
- &quot;none&quot;: No p-value adjustment.</p>
</td></tr>
<tr><td><code id="pathway_daa_+3A_reference">reference</code></td>
<td>
<p>a character specifying the reference group level, required for several differential abundance analysis methods such as LinDA, limme voom and Maaslin2. This parameter is used to specify the reference group when there are more than two groups. Default is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing the differential abundance analysis results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(ggpicrust2)
library(MicrobiomeStat)
library(tibble)
library(magrittr)
abundance &lt;- data.frame(sample1 = c(10, 20, 30),
sample2 = c(20, 30, 40),
sample3 = c(30, 40, 50),
row.names = c("pathway1", "pathway2", "pathway3"))

metadata &lt;- tibble::tibble(sample = paste0("sample", 1:3),
group = c("control", "control", "treatment"))

#Run pathway_daa function
result &lt;- pathway_daa(abundance = abundance, metadata = metadata, group = "group",
daa_method = "LinDA")

data(metacyc_abundance)
data(metadata)
daa_results_df &lt;- pathway_daa(metacyc_abundance %&gt;%
column_to_rownames("pathway"), metadata, "Environment", daa_method = "Lefser")

</code></pre>

<hr>
<h2 id='pathway_errorbar'>The function pathway_errorbar() is used to visualize the results of functional pathway differential abundance analysis as error bar plots.</h2><span id='topic+pathway_errorbar'></span>

<h3>Description</h3>

<p>The function pathway_errorbar() is used to visualize the results of functional pathway differential abundance analysis as error bar plots.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway_errorbar_+3A_abundance">abundance</code></td>
<td>
<p>A data frame with row names representing pathways and column names representing samples. Each element represents the relative abundance of the corresponding pathway in the corresponding sample.</p>
</td></tr>
<tr><td><code id="pathway_errorbar_+3A_daa_results_df">daa_results_df</code></td>
<td>
<p>A data frame containing the results of the differential abundance analysis of the pathways, generated by the pathway_daa function. x_lab should be a column name of daa_results_df.</p>
</td></tr>
<tr><td><code id="pathway_errorbar_+3A_group">Group</code></td>
<td>
<p>A data frame or a vector that assigns each sample to a group. The groups are used to color the samples in the figure.</p>
</td></tr>
<tr><td><code id="pathway_errorbar_+3A_ko_to_kegg">ko_to_kegg</code></td>
<td>
<p>A logical parameter indicating whether there was a convertion that convert ko abundance to kegg abundance.</p>
</td></tr>
<tr><td><code id="pathway_errorbar_+3A_p_values_threshold">p_values_threshold</code></td>
<td>
<p>A numeric parameter specifying the threshold for statistical significance of differential abundance. Pathways with p-values below this threshold will be considered significant.</p>
</td></tr>
<tr><td><code id="pathway_errorbar_+3A_order">order</code></td>
<td>
<p>A parameter controlling the ordering of the rows in the figure. The options are: &quot;p_values&quot; (order by p-values), &quot;name&quot; (order by pathway name), &quot;group&quot; (order by the group with the highest mean relative abundance), or &quot;pathway_class&quot; (order by the pathway category).</p>
</td></tr>
<tr><td><code id="pathway_errorbar_+3A_select">select</code></td>
<td>
<p>A vector of pathway names to be included in the figure. This can be used to limit the number of pathways displayed. If NULL, all pathways will be displayed.</p>
</td></tr>
<tr><td><code id="pathway_errorbar_+3A_p_value_bar">p_value_bar</code></td>
<td>
<p>A logical parameter indicating whether to display a bar showing the p-value threshold for significance. If TRUE, the bar will be displayed.</p>
</td></tr>
<tr><td><code id="pathway_errorbar_+3A_colors">colors</code></td>
<td>
<p>A vector of colors to be used to represent the groups in the figure. Each color corresponds to a group.</p>
</td></tr>
<tr><td><code id="pathway_errorbar_+3A_x_lab">x_lab</code></td>
<td>
<p>A character string to be used as the x-axis label in the figure. The default value is &quot;description&quot; for KOs'descriptions and &quot;pathway_name&quot; for KEGG pathway names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A figure showing the error bar plot of the differential abundance analysis results for the functional pathways.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Example 1: Analyzing KEGG pathway abundance
metadata &lt;- read_delim(
  "path/to/your/metadata.txt",
  delim = "\t",
  escape_double = FALSE,
  trim_ws = TRUE
)

# data(metadata)

kegg_abundance &lt;- ko2kegg_abundance(
  "path/to/your/pred_metagenome_unstrat.tsv"
)

# data(kegg_abundance)

# Please change group to "your_group_column" if you are not using example dataset
group &lt;- "Environment"

daa_results_df &lt;- pathway_daa(
  abundance = kegg_abundance,
  metadata = metadata,
  group = group,
  daa_method = "ALDEx2",
  select = NULL,
  reference = NULL
)

# Please check the unique(daa_results_df$method) and choose one
daa_sub_method_results_df &lt;- daa_results_df[daa_results_df$method
== "ALDEx2_Welch's t test", ]

daa_annotated_sub_method_results_df &lt;- pathway_annotation(
  pathway = "KO",
  daa_results_df = daa_sub_method_results_df,
  ko_to_kegg = TRUE
)

# Please change Group to metadata$your_group_column if you are not using example dataset
Group &lt;- metadata$Environment

p &lt;- pathway_errorbar(
  abundance = kegg_abundance,
  daa_results_df = daa_annotated_sub_method_results_df,
  Group = Group,
  p_values_threshold = 0.05,
  order = "pathway_class",
  select = daa_annotated_sub_method_results_df %&gt;%
  arrange(p_adjust) %&gt;%
  slice(1:20) %&gt;%
  select("feature") %&gt;% pull(),
  ko_to_kegg = TRUE,
  p_value_bar = TRUE,
  colors = NULL,
  x_lab = "pathway_name"
)

# Example 2: Analyzing EC, MetaCyc, KO without conversions
metadata &lt;- read_delim(
  "path/to/your/metadata.txt",
  delim = "\t",
  escape_double = FALSE,
  trim_ws = TRUE
)
# data(metadata)

ko_abundance &lt;- read.delim("path/to/your/metacyc_abundance.tsv")

# data(metacyc_abundance)

group &lt;- "Environment"

daa_results_df &lt;- pathway_daa(
  abundance = metacyc_abundance %&gt;% column_to_rownames("pathway"),
  metadata = metadata,
  group = group,
  daa_method = "LinDA",
  select = NULL,
  reference = NULL
)


daa_annotated_results_df &lt;- pathway_annotation(
  pathway = "MetaCyc",
  daa_results_df = daa_results_df,
  ko_to_kegg = FALSE
)

Group &lt;- metadata$Environment

p &lt;- pathway_errorbar(
  abundance = metacyc_abundance %&gt;% column_to_rownames("pathway"),
  daa_results_df = daa_annotated_results_df,
  Group = Group,
  p_values_threshold = 0.05,
  order = "group",
  select = NULL,
  ko_to_kegg = FALSE,
  p_value_bar = TRUE,
  colors = NULL,
  x_lab = "description"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='pathway_heatmap'>Create pathway heatmap</h2><span id='topic+pathway_heatmap'></span>

<h3>Description</h3>

<p>This function creates a heatmap of the predicted functional pathway abundance data. The function first makes the abundance data relative, then converts the abundance data to a long format and orders the samples based on the environment information. The heatmap is then created using the 'ggplot2' library. The color palette, appearance and the color bar of the heatmap can be customized using the 'scale_fill_gradientn', 'theme' and 'guides' functions respectively.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway_heatmap_+3A_abundance">abundance</code></td>
<td>
<p>A matrix or data frame of pathway abundance data, where columns correspond to samples and rows correspond to pathways.</p>
</td></tr>
<tr><td><code id="pathway_heatmap_+3A_metadata">metadata</code></td>
<td>
<p>A data frame of metadata, where each row corresponds to a sample and each column corresponds to a metadata variable.</p>
</td></tr>
<tr><td><code id="pathway_heatmap_+3A_group">group</code></td>
<td>
<p>A character string specifying the column name in the metadata data frame that contains the group variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot heatmap object. The output is a ggplot object representing the heatmap of the predicted functional pathway abundance data. The heatmap visualizes the z score of pathways in different samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(ggpicrust2)
library(ggh4x)
library(dplyr)
library(tidyr)
library(tibble)
library(magrittr)
# Create example functional pathway abundance data
kegg_abundance_example &lt;- matrix(rnorm(30), nrow = 3, ncol = 10)
colnames(kegg_abundance_example) &lt;- paste0("Sample", 1:10)
rownames(kegg_abundance_example) &lt;- c("PathwayA", "PathwayB", "PathwayC")

# Create example metadata
# Please ensure the sample IDs in the metadata have the column name "sample_name"
metadata_example &lt;- data.frame(sample_name = colnames(kegg_abundance_example),
                               group = factor(rep(c("Control", "Treatment"), each = 5)))

# Create a heatmap
pathway_heatmap(kegg_abundance_example, metadata_example, "group")
# Use real dataset
data("metacyc_abundance")
data("metadata")
metacyc_daa_results_df &lt;- pathway_daa(abundance = metacyc_abundance %&gt;%
column_to_rownames("pathway"),
metadata = metadata, group = "Environment", daa_method = "LinDA")
annotated_metacyc_daa_results_df &lt;- pathway_annotation(pathway = "MetaCyc",
daa_results_df = metacyc_daa_results_df, ko_to_kegg = FALSE)
feature_with_p_0.05 &lt;- metacyc_daa_results_df %&gt;% filter(p_adjust &lt; 0.05)
pathway_heatmap(abundance = metacyc_abundance %&gt;%
right_join(annotated_metacyc_daa_results_df %&gt;%
select(all_of(c("feature","description"))), by = c("pathway" = "feature")) %&gt;%
filter(pathway %in% feature_with_p_0.05$feature) %&gt;%
select(-"pathway") %&gt;%
column_to_rownames("description"), metadata = metadata, group = "Environment")

</code></pre>

<hr>
<h2 id='pathway_pca'>Perform Principal Component Analysis (PCA) on functional pathway abundance data and create visualizations of the PCA results.</h2><span id='topic+pathway_pca'></span>

<h3>Description</h3>

<p>Perform Principal Component Analysis (PCA) on functional pathway abundance data and create visualizations of the PCA results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway_pca(abundance, metadata, group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway_pca_+3A_abundance">abundance</code></td>
<td>
<p>A data frame, predicted functional pathway abundance.</p>
</td></tr>
<tr><td><code id="pathway_pca_+3A_metadata">metadata</code></td>
<td>
<p>A tibble, consisting of sample information.</p>
</td></tr>
<tr><td><code id="pathway_pca_+3A_group">group</code></td>
<td>
<p>A character, group name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object showing the PCA results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(magrittr)
library(dplyr)
library(tibble)
# Create example functional pathway abundance data
kegg_abundance_example &lt;- matrix(rnorm(30), nrow = 3, ncol = 10)
colnames(kegg_abundance_example) &lt;- paste0("Sample", 1:10)
rownames(kegg_abundance_example) &lt;- c("PathwayA", "PathwayB", "PathwayC")

# Create example metadata
# Please ensure the sample IDs in the metadata have the column name "sample_name"
metadata_example &lt;- data.frame(sample_name = colnames(kegg_abundance_example),
                               group = factor(rep(c("Control", "Treatment"), each = 5)))

pca_plot &lt;- pathway_pca(kegg_abundance_example, metadata_example, "group")
print(pca_plot)


data("metacyc_abundance")
data("metadata")
pathway_pca(metacyc_abundance %&gt;% column_to_rownames("pathway"), metadata, "Environment")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
