<!DOCTYPE html><html lang="en"><head><title>Help for package ggpubr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ggpubr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ggpubr-package'><p>ggpubr: 'ggplot2' Based Publication Ready Plots</p></a></li>
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#add_summary'><p>Add Summary Statistics onto a ggplot.</p></a></li>
<li><a href='#annotate_figure'><p>Annotate Arranged Figure</p></a></li>
<li><a href='#as_ggplot'><p>Storing grid.arrange() arrangeGrob() and plots</p></a></li>
<li><a href='#as_npc'><p>Convert Character Coordinates into Normalized Parent Coordinates (NPC)</p></a></li>
<li><a href='#axis_scale'><p>Change Axis Scale: log2, log10 and more</p></a></li>
<li><a href='#background_image'><p>Add Background Image to ggplot2</p></a></li>
<li><a href='#bgcolor'><p>Change ggplot Panel Background Color</p></a></li>
<li><a href='#border'><p>Set ggplot Panel Border Line</p></a></li>
<li><a href='#compare_means'><p>Comparison of Means</p></a></li>
<li><a href='#create_aes'><p>Create Aes Mapping from a List</p></a></li>
<li><a href='#desc_statby'><p>Descriptive statistics by groups</p></a></li>
<li><a href='#diff_express'><p>Differential gene expression analysis results</p></a></li>
<li><a href='#facet'><p>Facet a ggplot into Multiple Panels</p></a></li>
<li><a href='#font'><p>Change the Appearance of Titles and Axis Labels</p></a></li>
<li><a href='#gene_citation'><p>Gene Citation Index</p></a></li>
<li><a href='#gene_expression'><p>Gene Expression Data</p></a></li>
<li><a href='#geom_exec'><p>Execute ggplot2 functions</p></a></li>
<li><a href='#geom_signif'><p>geom_signif exported from ggsignif package</p></a></li>
<li><a href='#get_breaks'><p>Easy Break Creation for Numeric Axes</p></a></li>
<li><a href='#get_coord'><p>Checks and Returns Data Coordinates from Multiple Input Options</p></a></li>
<li><a href='#get_legend'><p>Extract Legends from a ggplot object</p></a></li>
<li><a href='#get_palette'><p>Generate Color Palettes</p></a></li>
<li><a href='#ggadd'><p>Add Summary Statistics or a Geom onto a ggplot</p></a></li>
<li><a href='#ggadjust_pvalue'><p>Adjust p-values Displayed on a GGPlot</p></a></li>
<li><a href='#ggarrange'><p>Arrange Multiple ggplots</p></a></li>
<li><a href='#ggballoonplot'><p>Ballon plot</p></a></li>
<li><a href='#ggbarplot'><p>Bar plot</p></a></li>
<li><a href='#ggboxplot'><p>Box plot</p></a></li>
<li><a href='#ggdensity'><p>Density plot</p></a></li>
<li><a href='#ggdonutchart'><p>Donut chart</p></a></li>
<li><a href='#ggdotchart'><p>Cleveland's Dot Plots</p></a></li>
<li><a href='#ggdotplot'><p>Dot plot</p></a></li>
<li><a href='#ggecdf'><p>Empirical cumulative density function</p></a></li>
<li><a href='#ggerrorplot'><p>Visualizing Error</p></a></li>
<li><a href='#ggexport'><p>Export ggplots</p></a></li>
<li><a href='#gghistogram'><p>Histogram plot</p></a></li>
<li><a href='#ggline'><p>Line plot</p></a></li>
<li><a href='#ggmaplot'><p>MA-plot from means and log fold changes</p></a></li>
<li><a href='#ggpaired'><p>Plot Paired Data</p></a></li>
<li><a href='#ggpar'><p>Graphical parameters</p></a></li>
<li><a href='#ggparagraph'><p>Draw a Paragraph of Text</p></a></li>
<li><a href='#ggpie'><p>Pie chart</p></a></li>
<li><a href='#ggpubr_args'><p>ggpubr General Arguments Description</p></a></li>
<li><a href='#ggpubr_options'><p>Global Options for GGPubr</p></a></li>
<li><a href='#ggqqplot'><p>QQ Plots</p></a></li>
<li><a href='#ggscatter'><p>Scatter plot</p></a></li>
<li><a href='#ggscatterhist'><p>Scatter Plot with Marginal Histograms</p></a></li>
<li><a href='#ggstripchart'><p>Stripcharts</p></a></li>
<li><a href='#ggsummarytable'><p>GGPLOT with Summary Stats Table Under the Plot</p></a></li>
<li><a href='#ggtext'><p>Text</p></a></li>
<li><a href='#ggtexttable'><p>Draw a Textual Table</p></a></li>
<li><a href='#ggviolin'><p>Violin plot</p></a></li>
<li><a href='#gradient_color'><p>Set Gradient Color</p></a></li>
<li><a href='#grids'><p>Add Grids to a ggplot</p></a></li>
<li><a href='#npc_to_data_coord'><p>Convert NPC to Data Coordinates</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#rotate'><p>Rotate a ggplot Horizontally</p></a></li>
<li><a href='#rotate_axis_text'><p>Rotate Axes Text</p></a></li>
<li><a href='#rremove'><p>Remove a ggplot Component</p></a></li>
<li><a href='#set_palette'><p>Set Color Palette</p></a></li>
<li><a href='#show_line_types'><p>Line types available in R</p></a></li>
<li><a href='#show_point_shapes'><p>Point shapes available in R</p></a></li>
<li><a href='#stat_anova_test'><p>Add Anova Test P-values to a GGPlot</p></a></li>
<li><a href='#stat_bracket'><p>Add Brackets with Labels to a GGPlot</p></a></li>
<li><a href='#stat_central_tendency'><p>Add Central Tendency Measures to a GGPLot</p></a></li>
<li><a href='#stat_chull'><p>Plot convex hull of a set of points</p></a></li>
<li><a href='#stat_compare_means'><p>Add Mean Comparison P-values to a ggplot</p></a></li>
<li><a href='#stat_conf_ellipse'><p>Plot confidence ellipses.</p></a></li>
<li><a href='#stat_cor'><p>Add Correlation Coefficients with P-values to a Scatter Plot</p></a></li>
<li><a href='#stat_friedman_test'><p>Add Friedman Test P-values to a GGPlot</p></a></li>
<li><a href='#stat_kruskal_test'><p>Add Kruskal-Wallis Test P-values to a GGPlot</p></a></li>
<li><a href='#stat_mean'><p>Draw group mean points</p></a></li>
<li><a href='#stat_overlay_normal_density'><p>Overlay Normal Density Plot</p></a></li>
<li><a href='#stat_pvalue_manual'><p>Add Manually P-values to a ggplot</p></a></li>
<li><a href='#stat_pwc'><p>Add Pairwise Comparisons P-values to a GGPlot</p></a></li>
<li><a href='#stat_regline_equation'><p>Add Regression Line Equation and R-Square to a GGPLOT.</p></a></li>
<li><a href='#stat_stars'><p>Add Stars to a Scatter Plot</p></a></li>
<li><a href='#stat_welch_anova_test'><p>Add Welch One-Way ANOVA Test P-values to a GGPlot</p></a></li>
<li><a href='#text_grob'><p>Create a Text Graphical object</p></a></li>
<li><a href='#theme_pubr'><p>Publication ready theme</p></a></li>
<li><a href='#theme_transparent'><p>Create a ggplot with Transparent Background</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>'ggplot2' Based Publication Ready Plots</td>
</tr>
<tr>
<td>Version:</td>
<td>0.6.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-02-05</td>
</tr>
<tr>
<td>Description:</td>
<td>The 'ggplot2' package is excellent and flexible for elegant data
    visualization in R. However the default generated plots requires some formatting
    before we can send them for publication. Furthermore, to customize a 'ggplot',
    the syntax is opaque and this raises the level of difficulty for researchers
    with no advanced R programming skills. 'ggpubr' provides some easy-to-use
    functions for creating and customizing 'ggplot2'- based publication ready plots.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0), ggplot2 (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggrepel (&ge; 0.9.2), grid, ggsci, stats, utils, tidyr (&ge;
1.3.0), purrr, dplyr (&ge; 0.7.1), cowplot (&ge; 1.1.1), ggsignif,
scales, gridExtra, glue, polynom, rlang (&ge; 0.4.6), rstatix (&ge;
0.7.2), tibble, magrittr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>grDevices, knitr, RColorBrewer, gtable, testthat</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://rpkgs.datanovia.com/ggpubr/">https://rpkgs.datanovia.com/ggpubr/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kassambara/ggpubr/issues">https://github.com/kassambara/ggpubr/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Collate:</td>
<td>'utilities_color.R' 'utilities_base.R' 'desc_statby.R'
'utilities.R' 'add_summary.R' 'annotate_figure.R' 'as_ggplot.R'
'as_npc.R' 'axis_scale.R' 'background_image.R' 'bgcolor.R'
'border.R' 'compare_means.R' 'create_aes.R' 'diff_express.R'
'facet.R' 'font.R' 'gene_citation.R' 'gene_expression.R'
'geom_bracket.R' 'geom_exec.R' 'utils-aes.R'
'utils_stat_test_label.R' 'geom_pwc.R' 'get_breaks.R'
'get_coord.R' 'get_legend.R' 'get_palette.R' 'ggadd.R'
'ggadjust_pvalue.R' 'ggarrange.R' 'ggballoonplot.R' 'ggpar.R'
'ggbarplot.R' 'ggboxplot.R' 'ggdensity.R' 'ggpie.R'
'ggdonutchart.R' 'stat_conf_ellipse.R' 'stat_chull.R'
'ggdotchart.R' 'ggdotplot.R' 'ggecdf.R' 'ggerrorplot.R'
'ggexport.R' 'gghistogram.R' 'ggline.R' 'ggmaplot.R'
'ggpaired.R' 'ggparagraph.R' 'ggpubr-package.R' 'ggpubr_args.R'
'ggpubr_options.R' 'ggqqplot.R' 'utilities_label.R'
'stat_cor.R' 'stat_stars.R' 'ggscatter.R' 'ggscatterhist.R'
'ggstripchart.R' 'ggsummarystats.R' 'ggtext.R' 'ggtexttable.R'
'ggviolin.R' 'gradient_color.R' 'grids.R' 'npc_to_data_coord.R'
'reexports.R' 'rotate.R' 'rotate_axis_text.R' 'rremove.R'
'set_palette.R' 'show_line_types.R' 'show_point_shapes.R'
'stat_anova_test.R' 'stat_central_tendency.R'
'stat_compare_means.R' 'stat_friedman_test.R'
'stat_kruskal_test.R' 'stat_mean.R'
'stat_overlay_normal_density.R' 'stat_pvalue_manual.R'
'stat_regline_equation.R' 'stat_welch_anova_test.R'
'text_grob.R' 'theme_pubr.R' 'theme_transparent.R'
'utils-geom-signif.R' 'utils-pipe.R' 'utils-tidyr.R'</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-06 02:39:47 UTC; kassambara</td>
</tr>
<tr>
<td>Author:</td>
<td>Alboukadel Kassambara [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alboukadel Kassambara &lt;alboukadel.kassambara@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-10 16:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ggpubr-package'>ggpubr: 'ggplot2' Based Publication Ready Plots</h2><span id='topic+ggpubr'></span><span id='topic+ggpubr-package'></span>

<h3>Description</h3>

<p>The 'ggplot2' package is excellent and flexible for elegant data visualization in R. However the default generated plots requires some formatting before we can send them for publication. Furthermore, to customize a 'ggplot', the syntax is opaque and this raises the level of difficulty for researchers with no advanced R programming skills. 'ggpubr' provides some easy-to-use functions for creating and customizing 'ggplot2'- based publication ready plots.
</p>


<h3>Details</h3>

<p>General resources:
</p>

<ul>
<li> <p><a href="https://rpkgs.datanovia.com/ggpubr/">ggpubr documentation</a>
</p>
</li>
<li> <p><a href="https://www.datanovia.com/en/blog/tag/ggpubr/">ggpubr tutorials</a>
</p>
</li></ul>



<h3>Package options</h3>


<dl>
<dt>ggpubr.parse_aes</dt><dd><p>logical indicating whether to parse or not aesthetics variables names.
Default is <code>TRUE</code>. For example, if you want <code>ggpubr</code> to handle non-standard column names, like <code>A-A</code>,
without parsing, then set this option to <code>FALSE</code> using <code>options(ggpubr.parse_aes = FALSE)</code>.</p>
</dd>
<dt>ggpubr.null_device</dt><dd><p>A function that creates an appropriate null device.
These include: <code><a href="cowplot.html#topic+pdf_null_device">cowplot::pdf_null_device</a></code>,
<code><a href="cowplot.html#topic+png_null_device">cowplot::png_null_device</a></code>,
<code><a href="cowplot.html#topic+cairo_null_device">cowplot::cairo_null_device</a></code> and
<code><a href="cowplot.html#topic+agg_null_device">cowplot::agg_null_device</a></code>. Default is
<code><a href="cowplot.html#topic+pdf_null_device">cowplot::pdf_null_device</a></code>. This is used in
function like <code><a href="#topic+as_ggplot">as_ggplot</a>()</code>, which needs to open a graphics
device to render ggplot objects into grid graphics objects. This function is
used to open null device for avoiding the display of unnecessary blank page
when calling <code><a href="#topic+ggarrange">ggarrange</a>()</code> or <code><a href="#topic+as_ggplot">as_ggplot</a>()</code></p>
</dd> </dl>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Alboukadel Kassambara <a href="mailto:alboukadel.kassambara@gmail.com">alboukadel.kassambara@gmail.com</a>
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://rpkgs.datanovia.com/ggpubr/">https://rpkgs.datanovia.com/ggpubr/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/kassambara/ggpubr/issues">https://github.com/kassambara/ggpubr/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>

<hr>
<h2 id='add_summary'>Add Summary Statistics onto a ggplot.</h2><span id='topic+add_summary'></span><span id='topic+mean_se_'></span><span id='topic+mean_sd'></span><span id='topic+mean_ci'></span><span id='topic+mean_range'></span><span id='topic+median_iqr'></span><span id='topic+median_hilow_'></span><span id='topic+median_q1q3'></span><span id='topic+median_mad'></span><span id='topic+median_range'></span>

<h3>Description</h3>

<p>add summary statistics onto a ggplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_summary(
  p,
  fun = "mean_se",
  error.plot = "pointrange",
  color = "black",
  fill = "white",
  group = 1,
  width = NULL,
  shape = 19,
  size = 1,
  linetype = 1,
  show.legend = NA,
  ci = 0.95,
  data = NULL,
  position = position_dodge(0.8)
)

mean_se_(x, error.limit = "both")

mean_sd(x, error.limit = "both")

mean_ci(x, ci = 0.95, error.limit = "both")

mean_range(x, error.limit = "both")

median_iqr(x, error.limit = "both")

median_hilow_(x, ci = 0.95, error.limit = "both")

median_q1q3(x, error.limit = "both")

median_mad(x, error.limit = "both")

median_range(x, error.limit = "both")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_summary_+3A_p">p</code></td>
<td>
<p>a ggplot on which you want to add summary statistics.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_fun">fun</code></td>
<td>
<p>a function that is given the complete data and should return a data
frame with variables ymin, y, and ymax. Allowed values are one of: &quot;mean&quot;,
&quot;mean_se&quot;, &quot;mean_sd&quot;, &quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;,
&quot;median_hilow&quot;, &quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_error.plot">error.plot</code></td>
<td>
<p>plot type used to visualize error. Allowed values are one of
<code>c("pointrange", "linerange", "crossbar", "errorbar", "upper_errorbar",
"lower_errorbar", "upper_pointrange", "lower_pointrange", "upper_linerange",
"lower_linerange")</code>. Default value is &quot;pointrange&quot;.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_color">color</code></td>
<td>
<p>point or outline color.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_fill">fill</code></td>
<td>
<p>fill color. Used only whne <code>error.plot = "crossbar"</code>.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_group">group</code></td>
<td>
<p>grouping variable. Allowed values are 1 (for one group) or a
character vector specifying the name of the grouping variable. Used only for
adding statistical summary per group.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_width">width</code></td>
<td>
<p>numeric value between 0 and 1 specifying bar or box width.
Example width = 0.8. Used only when <code>error.plot</code> is one of
c(&quot;crossbar&quot;, &quot;errorbar&quot;).</p>
</td></tr>
<tr><td><code id="add_summary_+3A_shape">shape</code></td>
<td>
<p>point shape. Allowed values can be displayed using the function
<code><a href="#topic+show_point_shapes">show_point_shapes</a>()</code>.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_size">size</code></td>
<td>
<p>numeric value in [0-1] specifying point and line size.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_linetype">linetype</code></td>
<td>
<p>line type.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends? NA,
the default, includes if any aesthetics are mapped. <code>FALSE</code> never includes,
and TRUE always includes. It can also be a named logical vector to finely
select the aesthetics to display.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_ci">ci</code></td>
<td>
<p>the percent range of the confidence interval (default is 0.95).</p>
</td></tr>
<tr><td><code id="add_summary_+3A_data">data</code></td>
<td>
<p>a <code>data.frame</code> to be displayed. If <code>NULL</code>, the default,
the data is inherited from the plot data as specified in the call to
<a href="ggplot2.html#topic+ggplot">ggplot</a>.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_position">position</code></td>
<td>
<p>position adjustment, either as a string, or the result of a
call to a position adjustment function. Used to adjust position for multiple
groups.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_x">x</code></td>
<td>
<p>a numeric vector.</p>
</td></tr>
<tr><td><code id="add_summary_+3A_error.limit">error.limit</code></td>
<td>
<p>allowed values are one of (&quot;both&quot;, &quot;lower&quot;, &quot;upper&quot;,
&quot;none&quot;) specifying whether to plot the lower and/or the upper limits of
error interval.</p>
</td></tr>
</table>


<h3>Functions</h3>


<ul>
<li> <p><code>add_summary()</code>: add summary statistics onto a ggplot.
</p>
</li>
<li> <p><code>mean_se_()</code>: returns the <code>mean</code> and the error limits defined by the
<code>standard error</code>. We used the name <code>mean_se_</code>() to avoid masking <code><a href="ggplot2.html#topic+mean_se">mean_se</a></code>().
</p>
</li>
<li> <p><code>mean_sd()</code>: returns the <code>mean</code> and the error limits defined by the
<code>standard deviation</code>.
</p>
</li>
<li> <p><code>mean_ci()</code>: returns the <code>mean</code> and the error limits defined by the
<code>confidence interval</code>.
</p>
</li>
<li> <p><code>mean_range()</code>: returns the <code>mean</code> and the error limits defined by the
<code>range = max - min</code>.
</p>
</li>
<li> <p><code>median_iqr()</code>: returns the <code>median</code> and the error limits
defined by the <code>interquartile range</code>.
</p>
</li>
<li> <p><code>median_hilow_()</code>: computes the sample median and a selected pair of
outer quantiles having equal tail areas. This function is a reformatted
version of <code>Hmisc::smedian.hilow()</code>. The confidence limits are computed
as follow: <code>lower.limits = (1-ci)/2</code> percentiles; <code>upper.limits =
(1+ci)/2</code> percentiles. By default (<code>ci = 0.95</code>), the 2.5th and the
97.5th percentiles are used as the lower and the upper confidence limits,
respectively. If you want to use the 25th and the 75th percentiles as the
confidence limits, then specify <code>ci = 0.5</code> or use the function
<code>median_q1q3()</code>.
</p>
</li>
<li> <p><code>median_q1q3()</code>: computes the sample median and, the 25th and 75th
percentiles. Wrapper around the function <code>median_hilow_()</code> using
<code>ci = 0.5</code>.
</p>
</li>
<li> <p><code>median_mad()</code>: returns the <code>median</code> and the error limits
defined by the <code>median absolute deviation</code>.
</p>
</li>
<li> <p><code>median_range()</code>: returns the <code>median</code> and the error limits
defined by the <code>range = max - min</code>.
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>
# Basic violin plot
p &lt;- ggviolin(ToothGrowth, x = "dose", y = "len", add = "none")
p

# Add mean_sd
add_summary(p, "mean_sd")


</code></pre>

<hr>
<h2 id='annotate_figure'>Annotate Arranged Figure</h2><span id='topic+annotate_figure'></span>

<h3>Description</h3>

<p>Annotate figures including: i) ggplots, ii) arranged ggplots from
<code><a href="#topic+ggarrange">ggarrange</a>()</code>, <code><a href="gridExtra.html#topic+arrangeGrob">grid.arrange</a>()</code> and
<code><a href="cowplot.html#topic+plot_grid">plot_grid</a>()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotate_figure(
  p,
  top = NULL,
  bottom = NULL,
  left = NULL,
  right = NULL,
  fig.lab = NULL,
  fig.lab.pos = c("top.left", "top", "top.right", "bottom.left", "bottom",
    "bottom.right"),
  fig.lab.size,
  fig.lab.face
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="annotate_figure_+3A_p">p</code></td>
<td>
<p>(arranged) ggplots.</p>
</td></tr>
<tr><td><code id="annotate_figure_+3A_top">top</code>, <code id="annotate_figure_+3A_bottom">bottom</code>, <code id="annotate_figure_+3A_left">left</code>, <code id="annotate_figure_+3A_right">right</code></td>
<td>
<p>optional string, or grob.</p>
</td></tr>
<tr><td><code id="annotate_figure_+3A_fig.lab">fig.lab</code></td>
<td>
<p>figure label (e.g.: &quot;Figure 1&quot;).</p>
</td></tr>
<tr><td><code id="annotate_figure_+3A_fig.lab.pos">fig.lab.pos</code></td>
<td>
<p>position of the figure label, can be one of &quot;top.left&quot;,
&quot;top&quot;, &quot;top.right&quot;, &quot;bottom.left&quot;, &quot;bottom&quot;, &quot;bottom.right&quot;. Default is
&quot;top.left&quot;.</p>
</td></tr>
<tr><td><code id="annotate_figure_+3A_fig.lab.size">fig.lab.size</code></td>
<td>
<p>optional size of the figure label.</p>
</td></tr>
<tr><td><code id="annotate_figure_+3A_fig.lab.face">fig.lab.face</code></td>
<td>
<p>optional font face of the figure label. Allowed values
include: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara <a href="mailto:alboukadel.kassambara@gmail.com">alboukadel.kassambara@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggarrange">ggarrange</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("ToothGrowth")
df &lt;- ToothGrowth
df$dose &lt;- as.factor(df$dose)

# Create some plots
# ::::::::::::::::::::::::::::::::::::::::::::::::::
# Box plot
bxp &lt;- ggboxplot(df, x = "dose", y = "len",
    color = "dose", palette = "jco")
# Dot plot
dp &lt;- ggdotplot(df, x = "dose", y = "len",
    color = "dose", palette = "jco")
# Density plot
dens &lt;- ggdensity(df, x = "len", fill = "dose", palette = "jco")

# Arrange and annotate
# ::::::::::::::::::::::::::::::::::::::::::::::::::
figure &lt;- ggarrange(bxp, dp, dens, ncol = 2, nrow = 2)
annotate_figure(figure,
               top = text_grob("Visualizing Tooth Growth", color = "red", face = "bold", size = 14),
               bottom = text_grob("Data source: \n ToothGrowth data set", color = "blue",
                                  hjust = 1, x = 1, face = "italic", size = 10),
               left = text_grob("Figure arranged using ggpubr", color = "green", rot = 90),
               right = text_grob(bquote("Superscript: ("*kg~NH[3]~ha^-1~yr^-1*")"), rot = 90),
               fig.lab = "Figure 1", fig.lab.face = "bold"
)


</code></pre>

<hr>
<h2 id='as_ggplot'>Storing grid.arrange() arrangeGrob() and plots</h2><span id='topic+as_ggplot'></span>

<h3>Description</h3>

<p>Transform the output of
<code><a href="gridExtra.html#topic+arrangeGrob">arrangeGrob</a>()</code> and
<code><a href="gridExtra.html#topic+arrangeGrob">grid.arrange</a>()</code> to a an object of class
ggplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_ggplot(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as_ggplot_+3A_x">x</code></td>
<td>
<p>an object of class gtable or grob as returned by the functions
<code><a href="gridExtra.html#topic+arrangeGrob">arrangeGrob</a>()</code> and
<code><a href="gridExtra.html#topic+arrangeGrob">grid.arrange</a>()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class ggplot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Creat some plots
bxp &lt;- ggboxplot(iris, x = "Species", y = "Sepal.Length")
vp &lt;- ggviolin(iris, x = "Species", y = "Sepal.Length",
              add = "mean_sd")

# Arrange the plots in one page
# Returns a gtable (grob) object
library(gridExtra)
gt &lt;- arrangeGrob(bxp, vp, ncol = 2)

# Transform to a ggplot and print
as_ggplot(gt)

</code></pre>

<hr>
<h2 id='as_npc'>Convert Character Coordinates into Normalized Parent Coordinates (NPC)</h2><span id='topic+as_npc'></span><span id='topic+as_npcx'></span><span id='topic+as_npcy'></span>

<h3>Description</h3>

<p>Convert character coordinates to npc units and shift postions to avoid
overlaps when grouping is active. If numeric validate npc values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_npc(
  value,
  group = 1L,
  step = 0.1,
  margin.npc = 0.05,
  axis = c("xy", "x", "y")
)

as_npcx(value, group = 1L, step = 0.1, margin.npc = 0.05)

as_npcy(value, group = 1L, step = 0.1, margin.npc = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as_npc_+3A_value">value</code></td>
<td>
<p>numeric (in [0-1]) or character vector of coordinates. If
character, should be one of <code>c('right', 'left', 'bottom', 'top',
'center', 'centre', 'middle')</code>.</p>
</td></tr>
<tr><td><code id="as_npc_+3A_group">group</code></td>
<td>
<p>integer ggplot's group id. Used to shift coordinates to avoid
overlaps.</p>
</td></tr>
<tr><td><code id="as_npc_+3A_step">step</code></td>
<td>
<p>numeric value in [0-1]. The step size for shifting coordinates
in npc units. Considered as horizontal step for x-axis and vertical step
for y-axis. For y-axis, the step value can be negative to reverse the order of groups.</p>
</td></tr>
<tr><td><code id="as_npc_+3A_margin.npc">margin.npc</code></td>
<td>
<p>numeric [0-1] The margin added towards the nearest
plotting area edge when converting character coordinates into npc.</p>
</td></tr>
<tr><td><code id="as_npc_+3A_axis">axis</code></td>
<td>
<p>the concerned axis . Should be one of <code>c("xy", "x", "y")</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>the <code>as_npc()</code> function is an adaptation from
<code>ggpmisc::compute_npc()</code>.
</p>


<h3>Value</h3>

<p>A numeric vector with values in the range [0-1] representing npc
coordinates.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>as_npc()</code>: converts x or y coordinate values into npc. Input values
should be numeric or one of the following values <code>c('right', 'left',
'bottom', 'top', 'center', 'centre', 'middle')</code>.
</p>
</li>
<li> <p><code>as_npcx()</code>: converts x coordinate values into npc. Input values should
be numeric or one of the following values <code>c('right', 'left',
'center', 'centre', 'middle')</code>. Wrapper around <code>as_npc(axis = "x")</code>.
</p>
</li>
<li> <p><code>as_npcy()</code>: converts y coordinate values into npc. Input values should
be numeric or one of the following values <code>c( 'bottom', 'top',
'center', 'centre', 'middle')</code>. Wrapper around <code>as_npc(axis = "y")</code>.
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+npc_to_data_coord">npc_to_data_coord</a></code>, <code><a href="#topic+get_coord">get_coord</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>as_npc(c("left", "right"))
as_npc(c("top", "right"))

</code></pre>

<hr>
<h2 id='axis_scale'>Change Axis Scale: log2, log10 and more</h2><span id='topic+axis_scale'></span><span id='topic+xscale'></span><span id='topic+yscale'></span>

<h3>Description</h3>

<p>Change axis scale.
</p>

<ul>
<li> <p><code>xscale</code>: change x axis scale.
</p>
</li>
<li> <p><code>yscale</code>: change y axis scale.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>xscale(.scale, .format = FALSE)

yscale(.scale, .format = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="axis_scale_+3A_.scale">.scale</code></td>
<td>
<p>axis scale. Allowed values are one of c(&quot;none&quot;, &quot;log2&quot;, &quot;log10&quot;,
&quot;sqrt&quot;, &quot;percent&quot;, &quot;dollar&quot;, &quot;scientific&quot;); e.g.: .scale=&quot;log2&quot;.</p>
</td></tr>
<tr><td><code id="axis_scale_+3A_.format">.format</code></td>
<td>
<p>ogical value. If TRUE, axis tick mark labels will be formatted
when .scale  = &quot;log2&quot; or &quot;log10&quot;.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Basic scatter plots
data(cars)
p &lt;- ggscatter(cars, x = "speed", y = "dist")
p

# Set log scale
p + yscale("log2", .format = TRUE)
</code></pre>

<hr>
<h2 id='background_image'>Add Background Image to ggplot2</h2><span id='topic+background_image'></span>

<h3>Description</h3>

<p>Add background image to ggplot2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>background_image(raster.img)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="background_image_+3A_raster.img">raster.img</code></td>
<td>
<p>raster object to display, as returned by the function
<code>readPNG()</code>[in <code>png</code> package] and <code>readJPEG()</code> [in <code>jpeg</code> package].</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara &lt;alboukadel.kassambara@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
install.packages("png")

# Import the image
img.file &lt;- system.file(file.path("images", "background-image.png"),
                       package = "ggpubr")
img &lt;- png::readPNG(img.file)

# Plot with background image
ggplot(iris, aes(Species, Sepal.Length))+
 background_image(img)+
 geom_boxplot(aes(fill = Species), color = "white")+
 fill_palette("jco")
 
## End(Not run)

</code></pre>

<hr>
<h2 id='bgcolor'>Change ggplot Panel Background Color</h2><span id='topic+bgcolor'></span>

<h3>Description</h3>

<p>Change ggplot panel background color.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bgcolor(color)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bgcolor_+3A_color">color</code></td>
<td>
<p>background color.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="#topic+border">border</a>().
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")
df &lt;- ToothGrowth

# Basic plot
p &lt;- ggboxplot(df, x = "dose", y = "len")
p

# Change panel background color
p +
  bgcolor("#BFD5E3")+
  border("#BFD5E3")
</code></pre>

<hr>
<h2 id='border'>Set ggplot Panel Border Line</h2><span id='topic+border'></span>

<h3>Description</h3>

<p>Change or set ggplot panel border.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>border(color = "black", size = 0.8, linetype = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="border_+3A_color">color</code></td>
<td>
<p>border line color.</p>
</td></tr>
<tr><td><code id="border_+3A_size">size</code></td>
<td>
<p>numeric value specifying border line size.</p>
</td></tr>
<tr><td><code id="border_+3A_linetype">linetype</code></td>
<td>
<p>line type. An integer (0:8), a name (blank, solid, dashed,
dotted, dotdash, longdash, twodash). Sess <code><a href="#topic+show_line_types">show_line_types</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")
df &lt;- ToothGrowth

# Basic plot
p &lt;- ggboxplot(df, x = "dose", y = "len")
p

# Add border
p + border()
</code></pre>

<hr>
<h2 id='compare_means'>Comparison of Means</h2><span id='topic+compare_means'></span>

<h3>Description</h3>

<p>Performs one or multiple mean comparisons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare_means(
  formula,
  data,
  method = "wilcox.test",
  paired = FALSE,
  group.by = NULL,
  ref.group = NULL,
  symnum.args = list(),
  p.adjust.method = "holm",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compare_means_+3A_formula">formula</code></td>
<td>
<p>a formula of the form <code>x ~ group</code> where <code>x</code> is a
numeric variable giving the data values and <code>group</code> is a factor with
one or multiple levels giving the corresponding groups. For example,
<code>formula = TP53 ~ cancer_group</code>.
</p>
<p>It's also possible to perform the test for multiple response variables at
the same time. For example, <code>formula = c(TP53, PTEN) ~ cancer_group</code>.</p>
</td></tr>
<tr><td><code id="compare_means_+3A_data">data</code></td>
<td>
<p>a data.frame containing the variables in the formula.</p>
</td></tr>
<tr><td><code id="compare_means_+3A_method">method</code></td>
<td>
<p>the type of test. Default is <a href="stats.html#topic+wilcox.test">wilcox.test</a>. Allowed
values include: </p>
 <ul>
<li> <p><code><a href="stats.html#topic+t.test">t.test</a></code> (parametric) and
<code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code> (non-parametric). Perform comparison
between two groups of samples. If the grouping variable contains more than
two levels, then a pairwise comparison is performed. </p>
</li>
<li>
<p><code><a href="stats.html#topic+anova">anova</a></code> (parametric) and
<code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code> (non-parametric). Perform one-way ANOVA
test comparing multiple groups. </p>
</li></ul>
</td></tr>
<tr><td><code id="compare_means_+3A_paired">paired</code></td>
<td>
<p>a logical indicating whether you want a paired test. Used only
in <code><a href="stats.html#topic+t.test">t.test</a></code> and in <a href="stats.html#topic+wilcox.test">wilcox.test</a>.</p>
</td></tr>
<tr><td><code id="compare_means_+3A_group.by">group.by</code></td>
<td>
<p>a character vector containing the name of grouping variables.</p>
</td></tr>
<tr><td><code id="compare_means_+3A_ref.group">ref.group</code></td>
<td>
<p>a character string specifying the reference group. If
specified, for a given grouping variable, each of the group levels will be
compared to the reference group (i.e. control group).
</p>
<p><code>ref.group</code> can be also <code>".all."</code>. In this case, each of the
grouping variable levels is compared to all (i.e. basemean).</p>
</td></tr>
<tr><td><code id="compare_means_+3A_symnum.args">symnum.args</code></td>
<td>
<p>a list of arguments to pass to the function
<code><a href="stats.html#topic+symnum">symnum</a></code> for symbolic number coding of p-values. For
example, <code>symnum.args &lt;- list(cutpoints = c(0, 0.0001, 0.001,
 0.01, 0.05, Inf), symbols = c("****", "***", "**", "*",  "ns"))</code>.
</p>
<p>In other words, we use the following convention for symbols indicating
statistical significance: </p>
 <ul>
<li> <p><code>ns</code>: p &gt; 0.05 </p>
</li>
<li>
<p><code>*</code>: p &lt;= 0.05 </p>
</li>
<li> <p><code>**</code>: p &lt;= 0.01 </p>
</li>
<li> <p><code>***</code>: p &lt;= 0.001 </p>
</li>
<li> <p><code>****</code>:  p &lt;= 0.0001 </p>
</li></ul>
</td></tr>
<tr><td><code id="compare_means_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for adjusting p values (see
<code><a href="stats.html#topic+p.adjust">p.adjust</a></code>). Has impact only in a situation, where
multiple pairwise tests are performed; or when there are multiple grouping
variables. Allowed values include &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;,
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;. If you don't want to adjust the p
value (not recommended), use p.adjust.method = &quot;none&quot;.
</p>
<p>Note that, when the <code>formula</code> contains multiple variables, the p-value
adjustment is done independently for each variable.</p>
</td></tr>
<tr><td><code id="compare_means_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed to the test function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return a data frame with the following columns:
</p>

<ul>
<li> <p><code>.y.</code>: the y variable used in the test.
</p>
</li>
<li> <p><code>group1,group2</code>: the compared groups in the pairwise tests.
Available only when <code>method = "t.test"</code> or <code>method = "wilcox.test"</code>.
</p>
</li>
<li> <p><code>p</code>: the p-value.
</p>
</li>
<li> <p><code>p.adj</code>: the adjusted p-value. Default for <code>p.adjust.method = "holm"</code>.
</p>
</li>
<li> <p><code>p.format</code>: the formatted p-value.
</p>
</li>
<li> <p><code>p.signif</code>: the significance level.
</p>
</li>
<li> <p><code>method</code>: the statistical test used to compare groups.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Load data
#:::::::::::::::::::::::::::::::::::::::
data("ToothGrowth")
df &lt;- ToothGrowth

# One-sample test
#:::::::::::::::::::::::::::::::::::::::::
compare_means(len ~ 1, df, mu = 0)

# Two-samples unpaired test
#:::::::::::::::::::::::::::::::::::::::::
compare_means(len ~ supp, df)

# Two-samples paired test
#:::::::::::::::::::::::::::::::::::::::::
compare_means(len ~ supp, df, paired = TRUE)

# Compare supp levels after grouping the data by "dose"
#::::::::::::::::::::::::::::::::::::::::
compare_means(len ~ supp, df, group.by = "dose")

# pairwise comparisons
#::::::::::::::::::::::::::::::::::::::::
# As dose contains more thant two levels ==&gt;
# pairwise test is automatically performed.
compare_means(len ~ dose, df)

# Comparison against reference group
#::::::::::::::::::::::::::::::::::::::::
compare_means(len ~ dose, df, ref.group = "0.5")

# Comparison against all
#::::::::::::::::::::::::::::::::::::::::
compare_means(len ~ dose, df, ref.group = ".all.")

# Anova and kruskal.test
#::::::::::::::::::::::::::::::::::::::::
compare_means(len ~ dose, df, method = "anova")
compare_means(len ~ dose, df, method = "kruskal.test")
</code></pre>

<hr>
<h2 id='create_aes'>Create Aes Mapping from a List</h2><span id='topic+create_aes'></span>

<h3>Description</h3>

<p>Create aes mapping to make programming easy with ggplot2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_aes(.list, parse = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_aes_+3A_.list">.list</code></td>
<td>
<p>a list of aesthetic arguments; for example .list = list(x = &quot;dose&quot;, y = &quot;len&quot;, color = &quot;dose&quot;).</p>
</td></tr>
<tr><td><code id="create_aes_+3A_parse">parse</code></td>
<td>
<p>logical. If TRUE, parse the input as an expression.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Simple aes creation
create_aes(list(x = "Sepal.Length", y = "Petal.Length" ))

# Parse an expression
x &lt;- "log2(Sepal.Length)"
y &lt;- "log2(Petal.Length)"
create_aes(list(x = x, y = y ), parse = TRUE)

# Create a ggplot
mapping &lt;- create_aes(list(x = x, y = y ), parse = TRUE)
ggplot(iris, mapping) +
 geom_point()

</code></pre>

<hr>
<h2 id='desc_statby'>Descriptive statistics by groups</h2><span id='topic+desc_statby'></span>

<h3>Description</h3>

<p>Computes descriptive statistics by groups for a measure variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>desc_statby(data, measure.var, grps, ci = 0.95)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="desc_statby_+3A_data">data</code></td>
<td>
<p>a data frame.</p>
</td></tr>
<tr><td><code id="desc_statby_+3A_measure.var">measure.var</code></td>
<td>
<p>the name of a column containing the variable to be summarized.</p>
</td></tr>
<tr><td><code id="desc_statby_+3A_grps">grps</code></td>
<td>
<p>a character vector containing grouping variables; e.g.: grps = c(&quot;grp1&quot;, &quot;grp2&quot;)</p>
</td></tr>
<tr><td><code id="desc_statby_+3A_ci">ci</code></td>
<td>
<p>the percent range of the confidence interval (default is 0.95).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing descriptive statistics, such as:
</p>

<ul>
<li> <p><strong>length</strong>: the number of elements in each group
</p>
</li>
<li> <p><strong>min</strong>: minimum
</p>
</li>
<li> <p><strong>max</strong>: maximum
</p>
</li>
<li> <p><strong>median</strong>: median
</p>
</li>
<li> <p><strong>mean</strong>: mean
</p>
</li>
<li> <p><strong>iqr</strong>: interquartile range
</p>
</li>
<li> <p><strong>mad</strong>: median absolute deviation (see ?MAD)
</p>
</li>
<li> <p><strong>sd</strong>: standard deviation of the sample
</p>
</li>
<li> <p><strong>se</strong>: standard error of the mean. It's calculated as the sample standard deviation divided by the root of the sample size.
</p>
</li>
<li> <p><strong>ci</strong>: confidence interval of the mean
</p>
</li>
<li> <p><strong>range</strong>: the range = max - min
</p>
</li>
<li> <p><strong>cv</strong>: coefficient of variation, sd/mean
</p>
</li>
<li> <p><strong>var</strong>: variance, sd^2
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")

# Descriptive statistics
res &lt;- desc_statby(ToothGrowth, measure.var = "len",
   grps = c("dose", "supp"))
head(res[, 1:10])

</code></pre>

<hr>
<h2 id='diff_express'>Differential gene expression analysis results</h2><span id='topic+diff_express'></span>

<h3>Description</h3>

<p>Differential gene expression analysis results obtained from
comparing the RNAseq data of two different cell populations using DESeq2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("diff_express")
</code></pre>


<h3>Format</h3>

<p>A data frame with 36028 rows and 5 columns. </p>

<dl>
<dt><code>name</code></dt><dd><p>gene names</p>
</dd> <dt><code>baseMean</code></dt><dd><p>mean expression
signal across all samples</p>
</dd> <dt><code>log2FoldChange</code></dt><dd><p>log2 fold change</p>
</dd>
<dt><code>padj</code></dt><dd><p>Adjusted p-value</p>
</dd><dt><code>detection_call</code></dt><dd><p>a numeric
vector specifying whether the genes is expressed (value = 1) or not (value = 0).</p>
</dd></dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(diff_express)

# Default plot
ggmaplot(diff_express, main = expression("Group 1" %-&gt;% "Group 2"),
   fdr = 0.05, fc = 2, size = 0.4,
   palette = c("#B31B21", "#1465AC", "darkgray"),
   genenames = as.vector(diff_express$name),
   legend = "top", top = 20,
   font.label = c("bold", 11),
   font.legend = "bold",
   font.main = "bold",
   ggtheme = ggplot2::theme_minimal())

# Add rectangle around labesl
ggmaplot(diff_express, main = expression("Group 1" %-&gt;% "Group 2"),
   fdr = 0.05, fc = 2, size = 0.4,
   palette = c("#B31B21", "#1465AC", "darkgray"),
   genenames = as.vector(diff_express$name),
   legend = "top", top = 20,
   font.label = c("bold", 11), label.rectangle = TRUE,
   font.legend = "bold",
   font.main = "bold",
   ggtheme = ggplot2::theme_minimal())

</code></pre>

<hr>
<h2 id='facet'>Facet a ggplot into Multiple Panels</h2><span id='topic+facet'></span>

<h3>Description</h3>

<p>Create multi-panel plots of a data set grouped by one or two
grouping variables. Wrapper around <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>facet(
  p,
  facet.by,
  nrow = NULL,
  ncol = NULL,
  scales = "fixed",
  short.panel.labs = TRUE,
  labeller = "label_value",
  panel.labs = NULL,
  panel.labs.background = list(color = NULL, fill = NULL),
  panel.labs.font = list(face = NULL, color = NULL, size = NULL, angle = NULL),
  panel.labs.font.x = panel.labs.font,
  panel.labs.font.y = panel.labs.font,
  strip.position = "top",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="facet_+3A_p">p</code></td>
<td>
<p>a ggplot</p>
</td></tr>
<tr><td><code id="facet_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="facet_+3A_nrow">nrow</code>, <code id="facet_+3A_ncol">ncol</code></td>
<td>
<p>Number of rows and columns in the panel. Used only when the
data is faceted by one grouping variable.</p>
</td></tr>
<tr><td><code id="facet_+3A_scales">scales</code></td>
<td>
<p>should axis scales of panels be fixed (&quot;fixed&quot;, the default),
free (&quot;free&quot;), or free in one dimension (&quot;free_x&quot;, &quot;free_y&quot;).</p>
</td></tr>
<tr><td><code id="facet_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="facet_+3A_labeller">labeller</code></td>
<td>
<p>Character vector. An alternative to the argument
<code>short.panel.labs</code>. Possible values are one of &quot;label_both&quot; (panel
labelled by both grouping variable names and levels) and &quot;label_value&quot;
(panel labelled with only grouping levels).</p>
</td></tr>
<tr><td><code id="facet_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="facet_+3A_panel.labs.background">panel.labs.background</code></td>
<td>
<p>a list to customize the background of panel
labels. Should contain the combination of the following elements: </p>

<ul>
<li> <p><code>color, linetype, size</code>: background line color, type and size
</p>
</li>
<li> <p><code>fill</code>: background fill color. </p>
</li></ul>
<p> For example,
panel.labs.background = list(color = &quot;blue&quot;, fill = &quot;pink&quot;, linetype =
&quot;dashed&quot;, size = 0.5).</p>
</td></tr>
<tr><td><code id="facet_+3A_panel.labs.font">panel.labs.font</code></td>
<td>
<p>a list of aestheics indicating the size (e.g.: 14), the
face/style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and the color
(e.g.: &quot;red&quot;) and the orientation angle (e.g.: 45) of panel labels.</p>
</td></tr>
<tr><td><code id="facet_+3A_panel.labs.font.x">panel.labs.font.x</code>, <code id="facet_+3A_panel.labs.font.y">panel.labs.font.y</code></td>
<td>
<p>same as panel.labs.font but for
only x and y direction, respectively.</p>
</td></tr>
<tr><td><code id="facet_+3A_strip.position">strip.position</code></td>
<td>
<p>(used only in <code>facet_wrap()</code>). By default, the
labels are displayed on the top of the plot. Using <code>strip.position</code> it
is possible to place the labels on either of the four sides by setting
<code>strip.position = c("top", "bottom", "left", "right")</code></p>
</td></tr>
<tr><td><code id="facet_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>p &lt;- ggboxplot(ToothGrowth, x = "dose", y = "len",
      color = "supp")
print(p)

facet(p, facet.by = "supp")

# Customize
facet(p + theme_bw(), facet.by = "supp",
  short.panel.labs = FALSE,   # Allow long labels in panels
  panel.labs.background = list(fill = "steelblue", color = "steelblue")
)
</code></pre>

<hr>
<h2 id='font'>Change the Appearance of Titles and Axis Labels</h2><span id='topic+font'></span>

<h3>Description</h3>

<p>Change the appearance of the main title, subtitle, caption, axis
labels and text, as well as the legend title and texts. Wrapper around
<code><a href="ggplot2.html#topic+element">element_text</a>()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>font(object, size = NULL, color = NULL, face = NULL, family = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="font_+3A_object">object</code></td>
<td>
<p>character string specifying the plot components. Allowed values
include: </p>
 <ul>
<li> <p><code>"title"</code> for the main title </p>
</li>
<li>
<p><code>"subtitle"</code> for the plot subtitle </p>
</li>
<li> <p><code>"caption"</code> for the plot
caption </p>
</li>
<li> <p><code>"legend.title"</code> for the legend title </p>
</li>
<li>
<p><code>"legend.text"</code> for the legend text </p>
</li>
<li> <p><code>"x", "xlab", or "x.title"</code>
for x axis label </p>
</li>
<li> <p><code>"y", "ylab", or "y.title"</code> for y axis label </p>
</li>
<li>
<p><code>"xy", "xylab", "xy.title" or "axis.title"</code> for both x and y axis
labels </p>
</li>
<li> <p><code>"x.text"</code> for x axis texts (x axis tick labels) </p>
</li>
<li>
<p><code>"y.text"</code> for y axis texts (y axis tick labels) </p>
</li>
<li> <p><code>"xy.text"</code>
or <code>"axis.text"</code> for both x and y axis texts </p>
</li></ul>
</td></tr>
<tr><td><code id="font_+3A_size">size</code></td>
<td>
<p>numeric value specifying the font size, (e.g.: <code>size = 12</code>).</p>
</td></tr>
<tr><td><code id="font_+3A_color">color</code></td>
<td>
<p>character string specifying the font color, (e.g.: <code>color =
"red"</code>).</p>
</td></tr>
<tr><td><code id="font_+3A_face">face</code></td>
<td>
<p>the font face or style. Allowed values include one of
<code>"plain", "bold", "italic", "bold.italic"</code>, (e.g.: <code>face =
"bold.italic"</code>).</p>
</td></tr>
<tr><td><code id="font_+3A_family">family</code></td>
<td>
<p>the font family.</p>
</td></tr>
<tr><td><code id="font_+3A_...">...</code></td>
<td>
<p>other arguments to pass to the function
<code><a href="ggplot2.html#topic+element">element_text</a>()</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")

# Basic plot
p &lt;- ggboxplot(ToothGrowth, x = "dose", y = "len", color = "dose",
              title = "Box Plot created with ggpubr",
              subtitle = "Length by dose",
              caption = "Source: ggpubr",
              xlab ="Dose (mg)", ylab = "Teeth length")
p

# Change the appearance of titles and labels
p +
 font("title", size = 14, color = "red", face = "bold.italic")+
 font("subtitle", size = 10, color = "orange")+
 font("caption", size = 10, color = "orange")+
 font("xlab", size = 12, color = "blue")+
 font("ylab", size = 12, color = "#993333")+
 font("xy.text", size = 12, color = "gray", face = "bold")

# Change the appearance of legend title and texts
p +
 font("legend.title", color = "blue", face = "bold")+
 font("legend.text", color = "red")

</code></pre>

<hr>
<h2 id='gene_citation'>Gene Citation Index</h2><span id='topic+gene_citation'></span>

<h3>Description</h3>

<p>Contains the mean citation index of 66 genes obtained by
assessing PubMed abstracts and annotations using two key words i) Gene name
+ b cell differentiation and ii) Gene name + plasma cell differentiation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("gene_citation")
</code></pre>


<h3>Format</h3>

<p>A data frame with 66 rows and 2 columns. </p>

<dl>
<dt><code>gene</code></dt><dd><p>gene names</p>
</dd> <dt><code>citation_index</code></dt><dd><p>mean citation index</p>
</dd> </dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(gene_citation)

# Some key genes of interest to be highlighted
key.gns &lt;- c("MYC", "PRDM1", "CD69", "IRF4", "CASP3", "BCL2L1", "MYB",  "BACH2", "BIM1",  "PTEN",
            "KRAS", "FOXP1", "IGF1R", "KLF4", "CDK6", "CCND2", "IGF1", "TNFAIP3", "SMAD3", "SMAD7",
            "BMPR2", "RB1", "IGF2R", "ARNT")
# Density distribution
ggdensity(gene_citation, x = "citation_index", y = "..count..",
  xlab = "Number of citation",
  ylab = "Number of genes",
  fill = "lightgray", color = "black",
  label = "gene", label.select = key.gns, repel = TRUE,
  font.label = list(color= "citation_index"),
  xticks.by = 20, # Break x ticks by 20
  gradient.cols = c("blue", "red"),
  legend = "bottom",
  legend.title = ""                                     # Hide legend title
  )

</code></pre>

<hr>
<h2 id='gene_expression'>Gene Expression Data</h2><span id='topic+gene_expression'></span>

<h3>Description</h3>

<p>Gene expression data extracted from TCGA using the 'RTCGA' and
'RTCGA.mRNA' R packages. It contains the mRNA expression for 3 genes -
GATA3, PTEN and XBP1- from 3 different datasets: Breast invasive carcinoma
(BRCA), Ovarian serous cystadenocarcinoma (OV) and Lung squamous cell
carcinoma (LUSC)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("gene_expression")
</code></pre>


<h3>Format</h3>

<p>A data frame with 1305 rows and 5 columns. </p>

<dl>
<dt><code>bcr_patient_barcode</code></dt><dd><p>sample ID</p>
</dd> <dt><code>dataset</code></dt><dd><p>cance type</p>
</dd> <dt><code>GATA3</code></dt><dd><p>GATA3 gene expression</p>
</dd>
<dt><code>PTEN</code></dt><dd><p>PTEN gene expression</p>
</dd><dt><code>XBP1</code></dt><dd><p>XBP1 gene expression.</p>
</dd></dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(gene_expression)

ggboxplot(gene_expression, x = "dataset",
y = c("GATA3", "PTEN", "XBP1"),
combine = TRUE,
ylab = "Expression",
color = "dataset", palette = "jco")

</code></pre>

<hr>
<h2 id='geom_exec'>Execute ggplot2 functions</h2><span id='topic+geom_exec'></span>

<h3>Description</h3>

<p>A helper function used by ggpubr functions to execute any geom_*
functions in ggplot2. Useful only when you want to call a geom_* function
without carrying about the arguments to put in aes(). Basic users of ggpubr
don't need this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_exec(geomfunc = NULL, data = NULL, position = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_exec_+3A_geomfunc">geomfunc</code></td>
<td>
<p>a ggplot2 function (e.g.: geom_point)</p>
</td></tr>
<tr><td><code id="geom_exec_+3A_data">data</code></td>
<td>
<p>a data frame to be used for mapping</p>
</td></tr>
<tr><td><code id="geom_exec_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string, or the result of a
call to a position adjustment function.</p>
</td></tr>
<tr><td><code id="geom_exec_+3A_...">...</code></td>
<td>
<p>arguments accepted by the function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return a plot if geomfunc!=Null or a list(option, mapping) if
geomfunc = NULL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ggplot() + geom_exec(geom_point, data = mtcars,
    x = "mpg", y = "wt", size = "cyl", color = "cyl")

## End(Not run)
</code></pre>

<hr>
<h2 id='geom_signif'>geom_signif exported from ggsignif package</h2><span id='topic+geom_signif'></span>

<h3>Description</h3>

<p>See <code>ggsignif::<a href="ggsignif.html#topic+stat_signif">geom_signif</a></code> for details.
</p>

<hr>
<h2 id='get_breaks'>Easy Break Creation for Numeric Axes</h2><span id='topic+get_breaks'></span>

<h3>Description</h3>

<p>Creates breaks for numeric axes to be used in the functions
<code><a href="ggplot2.html#topic+scale_continuous">scale_x_continuous</a>()</code> and
<code><a href="ggplot2.html#topic+scale_continuous">scale_y_continuous</a>()</code>. Can be used to increase the
number of x and y ticks by specifying the option <code>n</code>. It's also
possible to control axis breaks by specifying a step between ticks.  For
example, if <code>by = 5</code>, a tick mark is shown on every 5.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_breaks(n = NULL, by = NULL, from = NULL, to = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_breaks_+3A_n">n</code></td>
<td>
<p>number of breaks.</p>
</td></tr>
<tr><td><code id="get_breaks_+3A_by">by</code></td>
<td>
<p>number: the step between breaks.</p>
</td></tr>
<tr><td><code id="get_breaks_+3A_from">from</code></td>
<td>
<p>the starting value of breaks. By default, 0 is used for positive
variables</p>
</td></tr>
<tr><td><code id="get_breaks_+3A_to">to</code></td>
<td>
<p>the end values of breaks. This corresponds generally to the maximum
limit of the axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a break function
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate 5 breaks for a variable x
get_breaks(n = 5)(x = 1:100)

# Generate breaks using an increasing step
get_breaks(by = 10)(x = 1:100)

# Combine with ggplot scale_xx functions
library(ggplot2)

# Create a basic plot
p &lt;- ggscatter(mtcars, x = "wt", y = "mpg")
p

# Increase the number of ticks
p +
 scale_x_continuous(breaks = get_breaks(n = 10)) +
 scale_y_continuous(breaks = get_breaks(n = 10))

# Set ticks according to a specific step, starting from 0
p + scale_x_continuous(
  breaks = get_breaks(by = 1.5, from = 0),
  limits =  c(0, 6)
) +
 scale_y_continuous(
  breaks = get_breaks(by = 10, from = 0),
  limits = c(0, 40)
  )

</code></pre>

<hr>
<h2 id='get_coord'>Checks and Returns Data Coordinates from Multiple Input Options</h2><span id='topic+get_coord'></span>

<h3>Description</h3>

<p>Checks and returns selected coordinates from multiple input
options, which can be either data (x-y) coordinates or npc (normalized
parent coordinates).
</p>
<p>Helper function internally used in <code>ggpubr</code> function to guess the type
of coordinates specified by the user. For example, in the function
<code>stat_cor()</code>, users can specify either the option <code>label.x</code> (data
coordinates) or <code>label.x.npc</code> (npc coordinates); those coordinates are
passed to <code>get_coord()</code>, which will make some checking and then return
a unique coordinates for the label position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_coord(
  group = 1L,
  data.ranges = NULL,
  coord = NULL,
  npc = "left",
  step = 0.1,
  margin.npc = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_coord_+3A_group">group</code></td>
<td>
<p>integer ggplot's group id. Used to shift coordinates to avoid
overlaps.</p>
</td></tr>
<tr><td><code id="get_coord_+3A_data.ranges">data.ranges</code></td>
<td>
<p>a numeric vector of length 2 containing the data ranges
(minimum and the maximum). Should be specified only when <code>coord =
NULL</code> and <code>npc</code> is specified. Used to convert <code>npc</code> to data
coordinates. Considered only when the argument <code>npc</code> is specified.</p>
</td></tr>
<tr><td><code id="get_coord_+3A_coord">coord</code></td>
<td>
<p>data coordinates (i.e., either x or y coordinates).</p>
</td></tr>
<tr><td><code id="get_coord_+3A_npc">npc</code></td>
<td>
<p>numeric (in [0-1]) or character vector of coordinates. If
character, should be one of c('right', 'left', 'bottom', 'top', 'center',
'centre', 'middle'). Note that, the <code>data.ranges</code>, <code>step</code> and
<code>margin.npc</code>, arguments are considered only when <code>npc</code> is
specified. The option <code>npc</code> is ignored when the argument <code>coord</code> is specified.</p>
</td></tr>
<tr><td><code id="get_coord_+3A_step">step</code></td>
<td>
<p>numeric value in [0-1]. The step size for shifting coordinates
in npc units. Considered as horizontal step for x-axis and vertical step
for y-axis. For y-axis, the step value can be negative to reverse the order of groups.</p>
</td></tr>
<tr><td><code id="get_coord_+3A_margin.npc">margin.npc</code></td>
<td>
<p>numeric [0-1] The margin added towards the nearest
plotting area edge when converting character coordinates into npc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric vector representing data coordinates.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as_npc">as_npc</a></code>, <code><a href="#topic+npc_to_data_coord">npc_to_data_coord</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># If npc is specified, it is converted into data coordinates
get_coord(data.ranges = c(2, 20), npc = "left")
get_coord(data.ranges = c(2, 20), npc = 0.1)

# When coord is specified, no transformation is performed
# because this is assumed to be a data coordinate
get_coord(coord = 5)

# For grouped plots
res_top &lt;- get_coord(
  data.ranges = c(4.2, 36.4), group = c(1, 2, 3),
  npc = "top", step = -0.1, margin.npc = 0
)
res_top

</code></pre>

<hr>
<h2 id='get_legend'>Extract Legends from a ggplot object</h2><span id='topic+get_legend'></span>

<h3>Description</h3>

<p>Extract the legend labels from a ggplot object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_legend(p, position = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_legend_+3A_p">p</code></td>
<td>
<p>an object of class ggplot or a list of ggplots. If p is a list, only the first legend is returned.</p>
</td></tr>
<tr><td><code id="get_legend_+3A_position">position</code></td>
<td>
<p>character specifying legend position. Allowed values are one of
c(&quot;top&quot;, &quot;bottom&quot;, &quot;left&quot;, &quot;right&quot;, &quot;none&quot;). To remove the legend use
legend = &quot;none&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class gtable.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a scatter plot
p &lt;- ggscatter(iris, x = "Sepal.Length", y = "Sepal.Width",
        color = "Species", palette = "jco",
        ggtheme = theme_minimal())
p

# Extract the legend. Returns a gtable
leg &lt;- get_legend(p)

# Convert to a ggplot and print
as_ggplot(leg)

</code></pre>

<hr>
<h2 id='get_palette'>Generate Color Palettes</h2><span id='topic+get_palette'></span>

<h3>Description</h3>

<p>Generate a palette of k colors from ggsci palettes, RColorbrewer
palettes and custom color palettes. Useful to extend RColorBrewer and ggsci to support more colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_palette(palette = "default", k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_palette_+3A_palette">palette</code></td>
<td>
<p>Color palette. Allowed values include: </p>
 <ul>
<li>
<p><strong>Grey color palettes</strong>: &quot;grey&quot; or &quot;gray&quot;; </p>
</li>
<li> <p><strong>RColorBrewer
palettes</strong>, see <code><a href="RColorBrewer.html#topic+ColorBrewer">brewer.pal</a></code> and details section.
Examples of palette names include: &quot;RdBu&quot;, &quot;Blues&quot;, &quot;Dark2&quot;, &quot;Set2&quot;, ...;
</p>
</li>
<li> <p><strong>Custom color palettes</strong>. For example, palette = c(&quot;#00AFBB&quot;,
&quot;#E7B800&quot;, &quot;#FC4E07&quot;); </p>
</li>
<li> <p><strong>ggsci scientific journal palettes</strong>,
e.g.: &quot;npg&quot;, &quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and
&quot;rickandmorty&quot;. </p>
</li></ul>
</td></tr>
<tr><td><code id="get_palette_+3A_k">k</code></td>
<td>
<p>the number of colors to generate.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>RColorBrewer palettes</strong>: To display all available color
palettes, type this in R:RColorBrewer::display.brewer.all(). Color palette
names include:
</p>
 <ul>
<li> <p><strong>Sequential palettes</strong>, suited to ordered data that
progress from low to high. Palette names include: Blues BuGn BuPu GnBu
Greens Greys Oranges OrRd PuBu PuBuGn PuRd Purples RdPu Reds YlGn YlGnBu
YlOrBr YlOrRd. </p>
</li>
<li> <p><strong>Diverging palettes</strong>:Gradient colors. Names
include: BrBG PiYG PRGn PuOr RdBu RdGy RdYlBu RdYlGn Spectral. </p>
</li>
<li>
<p><strong>Qualitative palettes</strong>:  Best suited to representing nominal or
categorical data. Names include: Accent, Dark2, Paired, Pastel1, Pastel2,
Set1, Set2, Set3.
</p>
</li></ul>



<h3>Value</h3>

<p>Returns a vector of color palettes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("iris")
iris$Species2 &lt;- factor(rep(c(1:10), each = 15))

# Generate a gradient of 10 colors
ggscatter(iris, x = "Sepal.Length", y = "Petal.Length",
 color = "Species2",
 palette = get_palette(c("#00AFBB", "#E7B800", "#FC4E07"), 10))

# Scatter plot with default color palette
ggscatter(iris, x = "Sepal.Length", y = "Petal.Length",
 color = "Species")

# RColorBrewer color palettes
ggscatter(iris, x = "Sepal.Length", y = "Petal.Length",
 color = "Species", palette = get_palette("Dark2", 3))

# ggsci color palettes
ggscatter(iris, x = "Sepal.Length", y = "Petal.Length",
 color = "Species", palette = get_palette("npg", 3))

# Custom color palette
ggscatter(iris, x = "Sepal.Length", y = "Petal.Length",
 color = "Species",
 palette = c("#00AFBB", "#E7B800", "#FC4E07"))

# Or use this
ggscatter(iris, x = "Sepal.Length", y = "Petal.Length",
 color = "Species",
 palette = get_palette(c("#00AFBB", "#FC4E07"), 3))





</code></pre>

<hr>
<h2 id='ggadd'>Add Summary Statistics or a Geom onto a ggplot</h2><span id='topic+ggadd'></span>

<h3>Description</h3>

<p>Add summary statistics or a geometry onto a ggplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggadd(
  p,
  add = NULL,
  color = "black",
  fill = "white",
  group = 1,
  width = 1,
  shape = 19,
  size = NULL,
  alpha = 1,
  jitter = 0.2,
  seed = 123,
  binwidth = NULL,
  dotsize = size,
  linetype = 1,
  show.legend = NA,
  error.plot = "pointrange",
  ci = 0.95,
  data = NULL,
  position = position_dodge(0.8),
  p_geom = ""
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggadd_+3A_p">p</code></td>
<td>
<p>a ggplot</p>
</td></tr>
<tr><td><code id="ggadd_+3A_add">add</code></td>
<td>
<p>character vector specifying other plot elements to be added.
Allowed values are one or the combination of: &quot;none&quot;, &quot;dotplot&quot;, &quot;jitter&quot;,
&quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;, &quot;mean_ci&quot;, &quot;mean_range&quot;,
&quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;, &quot;median_q1q3&quot;, &quot;median_mad&quot;,
&quot;median_range&quot;.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_color">color</code></td>
<td>
<p>point or outline color.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_fill">fill</code></td>
<td>
<p>fill color. Used only when <code>error.plot = "crossbar"</code>.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_group">group</code></td>
<td>
<p>grouping variable. Allowed values are 1 (for one group) or a
character vector specifying the name of the grouping variable. Used only for
adding statistical summary per group.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_width">width</code></td>
<td>
<p>numeric value between 0 and 1 specifying bar or box width.
Example width = 0.8. Used only when <code>error.plot</code> is one of
c(&quot;crossbar&quot;, &quot;errorbar&quot;).</p>
</td></tr>
<tr><td><code id="ggadd_+3A_shape">shape</code></td>
<td>
<p>point shape. Allowed values can be displayed using the function
<code><a href="#topic+show_point_shapes">show_point_shapes</a>()</code>.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_size">size</code></td>
<td>
<p>numeric value in [0-1] specifying point and line size.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_alpha">alpha</code></td>
<td>
<p>numeric value specifying fill color transparency. Value should be
in [0, 1], where 0 is full transparency and 1 is no transparency.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_jitter">jitter</code></td>
<td>
<p>a numeric value specifying the amount of jittering. Used only
when <code>add</code> contains &quot;jitter&quot;.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_seed">seed</code></td>
<td>
<p>A random seed to make the jitter reproducible. Default is '123'. Useful if you need
to apply the same jitter twice, e.g., for a point and a corresponding label.
The random seed is reset after jittering. If 'NA', the
seed is initialized with a random value; this makes sure that two subsequent
calls start with a different seed. Use NULL to use the current random seed
and also avoid resetting (the behaviour of ggplot 2.2.1 and earlier).</p>
</td></tr>
<tr><td><code id="ggadd_+3A_binwidth">binwidth</code></td>
<td>
<p>numeric value specifying bin width. use value between 0 and 1
when you have a strong dense dotplot. For example binwidth = 0.2. Used only
when <code>add</code> contains &quot;dotplot&quot;.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_dotsize">dotsize</code></td>
<td>
<p>as <code>size</code> but applied only to dotplot.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_linetype">linetype</code></td>
<td>
<p>line type.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends? NA,
the default, includes if any aesthetics are mapped. <code>FALSE</code> never
includes, and TRUE always includes. It can also be a named logical vector to
finely select the aesthetics to display.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_error.plot">error.plot</code></td>
<td>
<p>plot type used to visualize error. Allowed values are one of
<code>c("pointrange", "linerange", "crossbar", "errorbar", "upper_errorbar",
"lower_errorbar", "upper_pointrange", "lower_pointrange", "upper_linerange",
"lower_linerange")</code>. Default value is &quot;pointrange&quot;.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_ci">ci</code></td>
<td>
<p>the percent range of the confidence interval (default is 0.95).</p>
</td></tr>
<tr><td><code id="ggadd_+3A_data">data</code></td>
<td>
<p>a <code>data.frame</code> to be displayed. If <code>NULL</code>, the default,
the data is inherited from the plot data as specified in the call to
<a href="ggplot2.html#topic+ggplot">ggplot</a>.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_position">position</code></td>
<td>
<p>position adjustment, either as a string, or the result of a
call to a position adjustment function. Used to adjust position for multiple
groups.</p>
</td></tr>
<tr><td><code id="ggadd_+3A_p_geom">p_geom</code></td>
<td>
<p>the geometry of the main plot. Ex: p_geom = &quot;geom_line&quot;. If
NULL, the geometry is extracted from p. Used only by <a href="#topic+ggline">ggline</a>().</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Basic violin plot
data("ToothGrowth")
p &lt;- ggviolin(ToothGrowth, x = "dose", y = "len", add = "none")

# Add mean +/- SD and jitter points
p %&gt;% ggadd(c("mean_sd", "jitter"), color = "dose")

# Add box plot
p %&gt;% ggadd(c("boxplot", "jitter"), color = "dose")

</code></pre>

<hr>
<h2 id='ggadjust_pvalue'>Adjust p-values Displayed on a GGPlot</h2><span id='topic+ggadjust_pvalue'></span>

<h3>Description</h3>

<p>Adjust p-values produced by <code><a href="#topic+geom_pwc">geom_pwc</a>()</code> on a ggplot.
This is mainly useful when using facet, where p-values are generally
computed and adjusted by panel without taking into account the other panels.
In this case, one might want to adjust after the p-values of all panels together.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggadjust_pvalue(
  p,
  layer = NULL,
  p.adjust.method = "holm",
  label = "p.adj",
  hide.ns = NULL,
  symnum.args = list(),
  output = c("plot", "stat_test")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggadjust_pvalue_+3A_p">p</code></td>
<td>
<p>a ggplot</p>
</td></tr>
<tr><td><code id="ggadjust_pvalue_+3A_layer">layer</code></td>
<td>
<p>An integer indicating the statistical layer rank in the ggplot
(in the order added to the plot).</p>
</td></tr>
<tr><td><code id="ggadjust_pvalue_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for adjusting p values (see
<code><a href="stats.html#topic+p.adjust">p.adjust</a></code>).  Has impact only in a situation, where
multiple pairwise tests are performed; or when there are multiple grouping
variables. Ignored when the specified method is <code>"tukey_hsd"</code> or
<code>"games_howell_test"</code> because they come with internal p adjustment
method. Allowed values include &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;,
&quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;. If you don't want to adjust the p value (not
recommended), use p.adjust.method = &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="ggadjust_pvalue_+3A_label">label</code></td>
<td>
<p>character string specifying label. Can be: </p>
 <ul>
<li><p> the
column containing the label (e.g.: <code>label = "p"</code> or <code>label =
"p.adj"</code>), where <code>p</code> is the p-value. Other possible values are
<code>"p.signif", "p.adj.signif", "p.format", "p.adj.format"</code>. </p>
</li>
<li><p> an
expression that can be formatted by the <code><a href="glue.html#topic+glue">glue</a>()</code> package.
For example, when specifying <code>label = "Wilcoxon, p = \{p\}"</code>, the
expression {p} will be replaced by its value. </p>
</li>
<li><p> a combination of
plotmath expressions and glue expressions. You may want some of the
statistical parameter in italic; for example:<code>label = "Wilcoxon,
italic(p)= {p}"</code></p>
</li></ul>
<p>.</p>
</td></tr>
<tr><td><code id="ggadjust_pvalue_+3A_hide.ns">hide.ns</code></td>
<td>
<p>can be logical value (<code>TRUE</code> or <code>FALSE</code>) or a character vector (<code>"p.adj"</code> or <code>"p"</code>).</p>
</td></tr>
<tr><td><code id="ggadjust_pvalue_+3A_symnum.args">symnum.args</code></td>
<td>
<p>a list of arguments to pass to the function
<code><a href="stats.html#topic+symnum">symnum</a></code> for symbolic number coding of p-values. For
example, <code>symnum.args &lt;- list(cutpoints = c(0, 0.0001, 0.001, 0.01,
 0.05, Inf), symbols = c("****", "***", "**", "*",  "ns"))</code>.
</p>
<p>In other words, we use the following convention for symbols indicating
statistical significance: </p>
 <ul>
<li> <p><code>ns</code>: p &gt; 0.05 </p>
</li>
<li>
<p><code>*</code>: p &lt;= 0.05 </p>
</li>
<li> <p><code>**</code>: p &lt;= 0.01 </p>
</li>
<li> <p><code>***</code>: p &lt;= 0.001
</p>
</li>
<li> <p><code>****</code>:  p &lt;= 0.0001 </p>
</li></ul>
</td></tr>
<tr><td><code id="ggadjust_pvalue_+3A_output">output</code></td>
<td>
<p>character. Possible values are one of <code>c("plot",
"stat_test")</code>. Default is &quot;plot&quot;.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Data preparation
#:::::::::::::::::::::::::::::::::::::::
df &lt;- ToothGrowth
df$dose &lt;- as.factor(df$dose)
# Add a random grouping variable
df$group &lt;- factor(rep(c("grp1", "grp2"), 30))
head(df, 3)

# Boxplot: Two groups by panel
#:::::::::::::::::::::::::::::::::::::::
# Create a box plot
bxp &lt;- ggboxplot(
  df, x = "supp", y = "len", fill = "#00AFBB",
  facet.by = "dose"
)
# Make facet and add p-values
bxp &lt;- bxp + geom_pwc(method = "t_test")
bxp
# Adjust all p-values together after
ggadjust_pvalue(
  bxp, p.adjust.method = "bonferroni",
  label = "{p.adj.format}{p.adj.signif}", hide.ns = TRUE
)


# Boxplot: Three groups by panel
#:::::::::::::::::::::::::::::::::::::::
# Create a box plot
bxp &lt;- ggboxplot(
  df, x = "dose", y = "len", fill = "#00AFBB",
  facet.by = "supp"
)
# Make facet and add p-values
bxp &lt;- bxp + geom_pwc(method = "t_test")
bxp
# Adjust all p-values together after
ggadjust_pvalue(
  bxp, p.adjust.method = "bonferroni",
  label = "{p.adj.format}{p.adj.signif}"
)
</code></pre>

<hr>
<h2 id='ggarrange'>Arrange Multiple ggplots</h2><span id='topic+ggarrange'></span>

<h3>Description</h3>

<p>Arrange multiple ggplots on the same page. Wrapper around
<code><a href="cowplot.html#topic+plot_grid">plot_grid</a>()</code>. Can arrange multiple ggplots over
multiple pages, compared to the standard
<code><a href="cowplot.html#topic+plot_grid">plot_grid</a>()</code>. Can also create a common unique legend
for multiple plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggarrange(
  ...,
  plotlist = NULL,
  ncol = NULL,
  nrow = NULL,
  labels = NULL,
  label.x = 0,
  label.y = 1,
  hjust = -0.5,
  vjust = 1.5,
  font.label = list(size = 14, color = "black", face = "bold", family = NULL),
  align = c("none", "h", "v", "hv"),
  widths = 1,
  heights = 1,
  legend = NULL,
  common.legend = FALSE,
  legend.grob = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggarrange_+3A_...">...</code></td>
<td>
<p>list of plots to be arranged into the grid. The plots can be
either ggplot2 plot objects or arbitrary gtables.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_plotlist">plotlist</code></td>
<td>
<p>(optional) list of plots to display.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_ncol">ncol</code></td>
<td>
<p>(optional) number of columns in the plot grid.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_nrow">nrow</code></td>
<td>
<p>(optional) number of rows in the plot grid.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_labels">labels</code></td>
<td>
<p>(optional) list of labels to be added to the plots. You can
also set labels=&quot;AUTO&quot; to auto-generate upper-case labels or labels=&quot;auto&quot;
to auto-generate lower-case labels.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_label.x">label.x</code></td>
<td>
<p>(optional) Single value or vector of x positions for plot
labels, relative to each subplot. Defaults to 0 for all labels. (Each label
is placed all the way to the left of each plot.)</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_label.y">label.y</code></td>
<td>
<p>(optional) Single value or vector of y positions for plot
labels, relative to each subplot. Defaults to 1 for all labels. (Each label
is placed all the way to the top of each plot.)</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_hjust">hjust</code></td>
<td>
<p>Adjusts the horizontal position of each label. More negative values move the label further
to the right on the plot canvas. Can be a single value (applied to all labels) or a vector of values
(one for each label). Default is -0.5.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_vjust">vjust</code></td>
<td>
<p>Adjusts the vertical position of each label. More positive values move the label further
down on the plot canvas. Can be a single value (applied to all labels) or a vector of values
(one for each label). Default is 1.5.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_font.label">font.label</code></td>
<td>
<p>a list of arguments for customizing labels. Allowed values
are the combination of the following elements: size (e.g.: 14), face (e.g.:
&quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;), color (e.g.: &quot;red&quot;) and family.
For example font.label = list(size = 14, face = &quot;bold&quot;, color =&quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_align">align</code></td>
<td>
<p>(optional) Specifies whether graphs in the grid should be horizontally (&quot;h&quot;) or
vertically (&quot;v&quot;) aligned. Options are &quot;none&quot; (default), &quot;hv&quot; (align in both directions), &quot;h&quot;, and &quot;v&quot;.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_widths">widths</code></td>
<td>
<p>(optional) numerical vector of relative columns widths. For
example, in a two-column grid, widths = c(2, 1) would make the first column
twice as wide as the second column.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_heights">heights</code></td>
<td>
<p>same as <code>widths</code> but for column heights.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_legend">legend</code></td>
<td>
<p>character specifying legend position. Allowed values are one of
c(&quot;top&quot;, &quot;bottom&quot;, &quot;left&quot;, &quot;right&quot;, &quot;none&quot;). To remove the legend use
legend = &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_common.legend">common.legend</code></td>
<td>
<p>logical value. Default is FALSE. If TRUE, a common
unique legend will be created for arranged plots.</p>
</td></tr>
<tr><td><code id="ggarrange_+3A_legend.grob">legend.grob</code></td>
<td>
<p>a legend grob as returned by the function
<code><a href="#topic+get_legend">get_legend</a>()</code>. If provided, it will be used as the common
legend.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return an object of class <code>ggarrange</code>, which is a ggplot or a
list of ggplot.
</p>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara <a href="mailto:alboukadel.kassambara@gmail.com">alboukadel.kassambara@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+annotate_figure">annotate_figure</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("ToothGrowth")
df &lt;- ToothGrowth
df$dose &lt;- as.factor(df$dose)

# Create some plots
# ::::::::::::::::::::::::::::::::::::::::::::::::::
# Box plot
bxp &lt;- ggboxplot(df, x = "dose", y = "len",
    color = "dose", palette = "jco")
# Dot plot
dp &lt;- ggdotplot(df, x = "dose", y = "len",
    color = "dose", palette = "jco")
# Density plot
dens &lt;- ggdensity(df, x = "len", fill = "dose", palette = "jco")

# Arrange
# ::::::::::::::::::::::::::::::::::::::::::::::::::
ggarrange(bxp, dp, dens, ncol = 2, nrow = 2)
# Use a common legend for multiple plots
ggarrange(bxp, dp,  common.legend = TRUE)


</code></pre>

<hr>
<h2 id='ggballoonplot'>Ballon plot</h2><span id='topic+ggballoonplot'></span>

<h3>Description</h3>

<p>Plot a graphical matrix where each cell contains a dot whose
size reflects the relative magnitude of the corresponding component. Useful
to visualize contingency table formed by two categorical variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggballoonplot(
  data,
  x = NULL,
  y = NULL,
  size = "value",
  facet.by = NULL,
  size.range = c(1, 10),
  shape = 21,
  color = "black",
  fill = "gray",
  show.label = FALSE,
  font.label = list(size = 12, color = "black"),
  rotate.x.text = TRUE,
  ggtheme = theme_minimal(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggballoonplot_+3A_data">data</code></td>
<td>
<p>a data frame. Can be: </p>
 <ul>
<li> <p><b>a standard
contingency table</b> formed by two categorical variables: a data frame with
row names and column names. The categories of the first variable are
columns and the categories of the second variable are rows. </p>
</li>
<li> <p><b>a
streched contingency table</b>: a data frame containing at least three columns
corresponding, respectively, to (1) the categories of the first variable,
(2) the categories of the second varible, (3) the frequency value. In this
case, you should specify the argument x and y in the function
<code>ggballoonplot()</code></p>
</li></ul>
<p>.</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_x">x</code>, <code id="ggballoonplot_+3A_y">y</code></td>
<td>
<p>the column names specifying, respectively, the first and the
second variable forming the contingency table. Required only when the data
is a stretched contingency table.</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_size">size</code></td>
<td>
<p>point size. By default, the points size reflects the relative
magnitude of the value of the corresponding cell (<code>size = "value"</code>).
Can be also numeric (<code>size = 4</code>).</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping variables for
faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_size.range">size.range</code></td>
<td>
<p>a numeric vector of length 2 that specifies the minimum and
maximum size of the plotting symbol. Default values are <code>size.range =
c(1, 10)</code>.</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_shape">shape</code></td>
<td>
<p>points shape. The default value is 21. Alternaive values include
22, 23, 24, 25.</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_color">color</code></td>
<td>
<p>point border line color.</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_fill">fill</code></td>
<td>
<p>point fill color. Default is &quot;lightgray&quot;. Considered only for
points 21 to 25.</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_show.label">show.label</code></td>
<td>
<p>logical. If TRUE, show the data cell values as point
labels.</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_font.label">font.label</code></td>
<td>
<p>a vector of length 3 indicating respectively the size
(e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and
the color (e.g.: &quot;red&quot;) of point labels. For example font.label = c(14,
&quot;bold&quot;, &quot;red&quot;). To specify only the size and the style, use font.label =
c(14, &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_rotate.x.text">rotate.x.text</code></td>
<td>
<p>logica. If TRUE (default), rotate the x axis text.</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggballoonplot_+3A_...">...</code></td>
<td>
<p>other arguments passed to the function <code><a href="#topic+ggpar">ggpar</a></code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Define color palette
my_cols &lt;- c("#0D0887FF", "#6A00A8FF", "#B12A90FF",
"#E16462FF", "#FCA636FF", "#F0F921FF")

# Standard contingency table
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# Read a contingency table: housetasks
# Repartition of 13 housetasks in the couple
data &lt;- read.delim(
  system.file("demo-data/housetasks.txt", package = "ggpubr"),
  row.names = 1
  )
data

# Basic ballon plot
ggballoonplot(data)

# Change color and fill
ggballoonplot(data, color = "#0073C2FF", fill = "#0073C2FF")


# Change color according to the value of table cells
ggballoonplot(data, fill = "value")+
   scale_fill_gradientn(colors = my_cols)

# Change the plotting symbol shape
ggballoonplot(data, fill = "value",  shape = 23)+
  gradient_fill(c("blue", "white", "red"))


# Set points size to 8, but change fill color by values
# Sow labels
ggballoonplot(data, fill = "value", color = "lightgray",
              size = 10, show.label = TRUE)+
  gradient_fill(c("blue", "white", "red"))

# Streched contingency table
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::

# Create an Example Data Frame Containing Car x Color data
carnames &lt;- c("bmw","renault","mercedes","seat")
carcolors &lt;- c("red","white","silver","green")
datavals &lt;- round(rnorm(16, mean=100, sd=60),1)
car_data &lt;- data.frame(Car = rep(carnames,4),
                   Color = rep(carcolors, c(4,4,4,4) ),
                   Value=datavals )

car_data

ggballoonplot(car_data, x = "Car", y = "Color",
              size = "Value", fill = "Value") +
   scale_fill_gradientn(colors = my_cols) +
  guides(size = FALSE)


# Grouped frequency table
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::
data("Titanic")
dframe &lt;- as.data.frame(Titanic)
head(dframe)
ggballoonplot(
 dframe, x = "Class", y = "Sex",
 size = "Freq", fill = "Freq",
 facet.by = c("Survived", "Age"),
 ggtheme = theme_bw()
)+
  scale_fill_gradientn(colors = my_cols)

# Hair and Eye Color of Statistics Students
data(HairEyeColor)
ggballoonplot( as.data.frame(HairEyeColor),
              x = "Hair", y = "Eye", size = "Freq",
              ggtheme = theme_gray()) %&gt;%
 facet("Sex")


</code></pre>

<hr>
<h2 id='ggbarplot'>Bar plot</h2><span id='topic+ggbarplot'></span>

<h3>Description</h3>

<p>Create a bar plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggbarplot(
  data,
  x,
  y,
  combine = FALSE,
  merge = FALSE,
  color = "black",
  fill = "white",
  palette = NULL,
  size = NULL,
  width = NULL,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  select = NULL,
  remove = NULL,
  order = NULL,
  add = "none",
  add.params = list(),
  error.plot = "errorbar",
  label = FALSE,
  lab.col = "black",
  lab.size = 4,
  lab.pos = c("out", "in"),
  lab.vjust = NULL,
  lab.hjust = NULL,
  lab.nb.digits = NULL,
  sort.val = c("none", "desc", "asc"),
  sort.by.groups = TRUE,
  top = Inf,
  position = position_stack(),
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggbarplot_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_x">x</code>, <code id="ggbarplot_+3A_y">y</code></td>
<td>
<p>x and y variables for drawing.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_color">color</code>, <code id="ggbarplot_+3A_fill">fill</code></td>
<td>
<p>outline and fill colors.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_width">width</code></td>
<td>
<p>numeric value between 0 and 1 specifying box width.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_select">select</code></td>
<td>
<p>character vector specifying which items to display.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_remove">remove</code></td>
<td>
<p>character vector specifying which items to remove from the plot.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_order">order</code></td>
<td>
<p>character vector specifying the order of items.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_add">add</code></td>
<td>
<p>character vector for adding another plot element (e.g.: dot plot or
error bars). Allowed values are one or the combination of: &quot;none&quot;,
&quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;,
&quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;,
&quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;; see ?desc_statby for more
details.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, shape, size, fill, linetype) for the
argument 'add'; e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_error.plot">error.plot</code></td>
<td>
<p>plot type used to visualize error. Allowed values are one of
c(&quot;pointrange&quot;, &quot;linerange&quot;, &quot;crossbar&quot;, &quot;errorbar&quot;, &quot;upper_errorbar&quot;,
&quot;lower_errorbar&quot;, &quot;upper_pointrange&quot;, &quot;lower_pointrange&quot;, &quot;upper_linerange&quot;,
&quot;lower_linerange&quot;). Default value is &quot;pointrange&quot; or &quot;errorbar&quot;. Used only
when add != &quot;none&quot; and add contains one &quot;mean_*&quot; or &quot;med_*&quot; where &quot;*&quot; = sd,
se, ....</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_label">label</code></td>
<td>
<p>specify whether to add labels on the bar plot. Allowed values
are: </p>
 <ul>
<li> <p><strong>logical value</strong>: If TRUE, y values is added as
labels on the bar plot </p>
</li>
<li> <p><strong>character vector</strong>: Used as text
labels; must be the same length as y. </p>
</li></ul>
</td></tr>
<tr><td><code id="ggbarplot_+3A_lab.col">lab.col</code>, <code id="ggbarplot_+3A_lab.size">lab.size</code></td>
<td>
<p>text color and size for labels.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_lab.pos">lab.pos</code></td>
<td>
<p>character specifying the position for labels. Allowed values
are &quot;out&quot; (for outside) or &quot;in&quot; (for inside). Ignored when lab.vjust !=
NULL.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_lab.vjust">lab.vjust</code></td>
<td>
<p>numeric, vertical justification of labels. Provide negative
value (e.g.: -0.4) to put labels outside the bars or positive value to put
labels inside (e.g.: 2).</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_lab.hjust">lab.hjust</code></td>
<td>
<p>numeric, horizontal justification of labels.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_lab.nb.digits">lab.nb.digits</code></td>
<td>
<p>integer indicating the number of decimal places (round) to be used.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_sort.val">sort.val</code></td>
<td>
<p>a string specifying whether the value should be sorted.
Allowed values are &quot;none&quot; (no sorting), &quot;asc&quot; (for ascending) or &quot;desc&quot; (for descending).</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_sort.by.groups">sort.by.groups</code></td>
<td>
<p>logical value. If TRUE the data are sorted by groups.
Used only when sort.val != &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_top">top</code></td>
<td>
<p>a numeric value specifying the number of top elements to be shown.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggbarplot_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to be passed to ggpar().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation
= c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>, <code><a href="#topic+ggline">ggline</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Data
df &lt;- data.frame(dose=c("D0.5", "D1", "D2"),
   len=c(4.2, 10, 29.5))
print(df)

# Basic plot with label outsite
# +++++++++++++++++++++++++++
ggbarplot(df, x = "dose", y = "len",
  label = TRUE, label.pos = "out")

# Change width
ggbarplot(df, x = "dose", y = "len", width = 0.5)

# Change the plot orientation: horizontal
ggbarplot(df, "dose", "len", orientation = "horiz")

# Change the default order of items
ggbarplot(df, "dose", "len",
   order = c("D2", "D1", "D0.5"))


# Change colors
# +++++++++++++++++++++++++++

# Change fill and outline color
# add labels inside bars
ggbarplot(df, "dose", "len",
 fill = "steelblue", color = "steelblue",
 label = TRUE, lab.pos = "in", lab.col = "white")

# Change colors by groups: dose
# Use custom color palette
 ggbarplot(df, "dose", "len", color = "dose",
   palette = c("#00AFBB", "#E7B800", "#FC4E07"))

# Change fill and outline colors by groups
 ggbarplot(df, "dose", "len",
   fill = "dose", color = "dose",
   palette = c("#00AFBB", "#E7B800", "#FC4E07"))


# Plot with multiple groups
# +++++++++++++++++++++

# Create some data
df2 &lt;- data.frame(supp=rep(c("VC", "OJ"), each=3),
   dose=rep(c("D0.5", "D1", "D2"),2),
   len=c(6.8, 15, 33, 4.2, 10, 29.5))
print(df2)

# Plot "len" by "dose" and change color by a second group: "supp"
# Add labels inside bars
ggbarplot(df2, "dose", "len",
  fill = "supp", color = "supp", palette = "Paired",
  label = TRUE, lab.col = "white", lab.pos = "in")

# Change position: Interleaved (dodged) bar plot
ggbarplot(df2, "dose", "len",
  fill = "supp", color = "supp", palette = "Paired",
  label = TRUE,
  position = position_dodge(0.9))

# Add points and errors
# ++++++++++++++++++++++++++

# Data: ToothGrowth data set we'll be used.
df3 &lt;- ToothGrowth
head(df3, 10)

# It can be seen that for each group we have
# different values
ggbarplot(df3, x = "dose", y = "len")

# Visualize the mean of each group
ggbarplot(df3, x = "dose", y = "len",
 add = "mean")

# Add error bars: mean_se
# (other values include: mean_sd, mean_ci, median_iqr, ....)
# Add labels
ggbarplot(df3, x = "dose", y = "len",
 add = "mean_se", label = TRUE, lab.vjust = -1.6)

# Use only "upper_errorbar"
ggbarplot(df3, x = "dose", y = "len",
 add = "mean_se", error.plot = "upper_errorbar")

# Change error.plot to "pointrange"
ggbarplot(df3, x = "dose", y = "len",
 add = "mean_se", error.plot = "pointrange")

# Add jitter points and errors (mean_se)
ggbarplot(df3, x = "dose", y = "len",
 add = c("mean_se", "jitter"))

# Add dot and errors (mean_se)
ggbarplot(df3, x = "dose", y = "len",
 add = c("mean_se", "dotplot"))

# Multiple groups with error bars and jitter point
ggbarplot(df3, x = "dose", y = "len", color = "supp",
 add = "mean_se", palette = c("#00AFBB", "#E7B800"),
 position = position_dodge())


</code></pre>

<hr>
<h2 id='ggboxplot'>Box plot</h2><span id='topic+ggboxplot'></span>

<h3>Description</h3>

<p>Create a box plot with points. Box plots display a group of
numerical data through their quartiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggboxplot(
  data,
  x,
  y,
  combine = FALSE,
  merge = FALSE,
  color = "black",
  fill = "white",
  palette = NULL,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  bxp.errorbar = FALSE,
  bxp.errorbar.width = 0.4,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  linetype = "solid",
  size = NULL,
  width = 0.7,
  notch = FALSE,
  outlier.shape = 19,
  select = NULL,
  remove = NULL,
  order = NULL,
  add = "none",
  add.params = list(),
  error.plot = "pointrange",
  label = NULL,
  font.label = list(size = 11, color = "black"),
  label.select = NULL,
  repel = FALSE,
  label.rectangle = FALSE,
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggboxplot_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_x">x</code></td>
<td>
<p>character string containing the name of x variable.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_y">y</code></td>
<td>
<p>character vector containing one or more variables to plot</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_color">color</code></td>
<td>
<p>outline color.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_fill">fill</code></td>
<td>
<p>fill color.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_bxp.errorbar">bxp.errorbar</code></td>
<td>
<p>logical value. If TRUE, shows error bars of box plots.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_bxp.errorbar.width">bxp.errorbar.width</code></td>
<td>
<p>numeric value specifying the width of box plot error
bars. Default is 0.4.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_linetype">linetype</code></td>
<td>
<p>line types.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_width">width</code></td>
<td>
<p>numeric value between 0 and 1 specifying box width.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_notch">notch</code></td>
<td>
<p>If <code>FALSE</code> (default) make a standard box plot. If
<code>TRUE</code>, make a notched box plot. Notches are used to compare groups;
if the notches of two boxes do not overlap, this suggests that the medians
are significantly different.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_outlier.shape">outlier.shape</code></td>
<td>
<p>point shape of outlier. Default is 19. To hide outlier,
specify <code>outlier.shape = NA</code>. When jitter is added, then outliers will
be automatically hidden.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_select">select</code></td>
<td>
<p>character vector specifying which items to display.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_remove">remove</code></td>
<td>
<p>character vector specifying which items to remove from the plot.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_order">order</code></td>
<td>
<p>character vector specifying the order of items.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_add">add</code></td>
<td>
<p>character vector for adding another plot element (e.g.: dot plot or
error bars). Allowed values are one or the combination of: &quot;none&quot;,
&quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;,
&quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;,
&quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;; see ?desc_statby for more
details.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, shape, size, fill, linetype) for the
argument 'add'; e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_error.plot">error.plot</code></td>
<td>
<p>plot type used to visualize error. Allowed values are one of
c(&quot;pointrange&quot;, &quot;linerange&quot;, &quot;crossbar&quot;, &quot;errorbar&quot;, &quot;upper_errorbar&quot;,
&quot;lower_errorbar&quot;, &quot;upper_pointrange&quot;, &quot;lower_pointrange&quot;, &quot;upper_linerange&quot;,
&quot;lower_linerange&quot;). Default value is &quot;pointrange&quot; or &quot;errorbar&quot;. Used only
when add != &quot;none&quot; and add contains one &quot;mean_*&quot; or &quot;med_*&quot; where &quot;*&quot; = sd,
se, ....</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_label">label</code></td>
<td>
<p>the name of the column containing point labels. Can be also a
character vector with length = nrow(data).</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_font.label">font.label</code></td>
<td>
<p>a list which can contain the combination of the following
elements: the size (e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;,
&quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of labels. For example font.label
= list(size = 14, face = &quot;bold&quot;, color =&quot;red&quot;). To specify only the size and
the style, use font.label = list(size = 14, face = &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_label.select">label.select</code></td>
<td>
<p>can be of two formats: </p>
 <ul>
<li><p> a character vector
specifying some labels to show. </p>
</li>
<li><p> a list containing one or the
combination of the following components: </p>
 <ul>
<li> <p><code>top.up</code> and
<code>top.down</code>: to display the labels  of the top up/down points. For
example, <code>label.select = list(top.up = 10, top.down = 4)</code>. </p>
</li>
<li>
<p><code>criteria</code>: to filter, for example, by x and y variabes values, use
this: <code>label.select = list(criteria = "`y` &gt; 2 &amp; `y` &lt; 5 &amp; `x` %in%
c('A', 'B')")</code>. </p>
</li></ul>
 </li></ul>
</td></tr>
<tr><td><code id="ggboxplot_+3A_repel">repel</code></td>
<td>
<p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggboxplot_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to
<code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code>, <code><a href="#topic+ggpar">ggpar</a></code> and
<code><a href="#topic+facet">facet</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main, xlab,
ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes: palette
= &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title, labels
and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation =
c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>Suggestions for the argument &quot;add&quot;</h3>

<p>Suggested values are one of
c(&quot;dotplot&quot;, &quot;jitter&quot;).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>, <code><a href="#topic+ggviolin">ggviolin</a></code>, <code><a href="#topic+ggdotplot">ggdotplot</a></code>
and <code><a href="#topic+ggstripchart">ggstripchart</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")
df &lt;- ToothGrowth

# Basic plot
# +++++++++++++++++++++++++++
# width: change box plots width
ggboxplot(df, x = "dose", y = "len", width = 0.8)

# Change orientation: horizontal
ggboxplot(df, "dose", "len", orientation = "horizontal")

# Notched box plot
ggboxplot(df, x = "dose", y = "len",
   notch = TRUE)

# Add dots
# ++++++++++++++++++++++++++
ggboxplot(df, x = "dose", y = "len",
   add = "dotplot")

# Add jitter points and change the shape by groups
ggboxplot(df, x = "dose", y = "len",
   add = "jitter", shape = "dose")


# Select and order items
# ++++++++++++++++++++++++++++++

# Select which items to display: "0.5" and "2"
ggboxplot(df, "dose", "len",
   select = c("0.5", "2"))

# Change the default order of items
ggboxplot(df, "dose", "len",
   order = c("2", "1", "0.5"))


# Change colors
# +++++++++++++++++++++++++++
# Change outline and fill colors
 ggboxplot(df, "dose", "len",
   color = "black", fill = "gray")

# Change outline colors by groups: dose
# Use custom color palette
# Add jitter points and change the shape by groups
 ggboxplot(df, "dose", "len",
    color = "dose", palette =c("#00AFBB", "#E7B800", "#FC4E07"),
    add = "jitter", shape = "dose")

# Change fill color by groups: dose
 ggboxplot(df, "dose", "len",
     fill = "dose", palette = c("#00AFBB", "#E7B800", "#FC4E07"))


# Box plot with multiple groups
# +++++++++++++++++++++
# fill or color box plot by a second group : "supp"
ggboxplot(df, "dose", "len", color = "supp",
 palette = c("#00AFBB", "#E7B800"))

</code></pre>

<hr>
<h2 id='ggdensity'>Density plot</h2><span id='topic+ggdensity'></span>

<h3>Description</h3>

<p>Create a density plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggdensity(
  data,
  x,
  y = "density",
  combine = FALSE,
  merge = FALSE,
  color = "black",
  fill = NA,
  palette = NULL,
  size = NULL,
  linetype = "solid",
  alpha = 0.5,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  add = c("none", "mean", "median"),
  add.params = list(linetype = "dashed"),
  rug = FALSE,
  label = NULL,
  font.label = list(size = 11, color = "black"),
  label.select = NULL,
  repel = FALSE,
  label.rectangle = FALSE,
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggdensity_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_x">x</code></td>
<td>
<p>variable to be drawn.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_y">y</code></td>
<td>
<p>one of &quot;density&quot; or &quot;count&quot;.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_color">color</code>, <code id="ggdensity_+3A_fill">fill</code></td>
<td>
<p>density line color and fill color.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_linetype">linetype</code></td>
<td>
<p>line type. See <code><a href="#topic+show_line_types">show_line_types</a></code>.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_alpha">alpha</code></td>
<td>
<p>numeric value specifying fill color transparency. Value should
be in [0, 1], where 0 is full transparency and 1 is no transparency.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_add">add</code></td>
<td>
<p>allowed values are one of &quot;mean&quot; or &quot;median&quot; (for adding mean or
median line, respectively).</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, size, linetype) for the argument 'add';
e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_rug">rug</code></td>
<td>
<p>logical value. If TRUE, add marginal rug.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_label">label</code></td>
<td>
<p>the name of the column containing point labels. Can be also a
character vector with length = nrow(data).</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_font.label">font.label</code></td>
<td>
<p>a list which can contain the combination of the following
elements: the size (e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;,
&quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of labels. For example font.label
= list(size = 14, face = &quot;bold&quot;, color =&quot;red&quot;). To specify only the size and
the style, use font.label = list(size = 14, face = &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_label.select">label.select</code></td>
<td>
<p>can be of two formats: </p>
 <ul>
<li><p> a character vector
specifying some labels to show. </p>
</li>
<li><p> a list containing one or the
combination of the following components: </p>
 <ul>
<li> <p><code>top.up</code> and
<code>top.down</code>: to display the labels  of the top up/down points. For
example, <code>label.select = list(top.up = 10, top.down = 4)</code>. </p>
</li>
<li>
<p><code>criteria</code>: to filter, for example, by x and y variabes values, use
this: <code>label.select = list(criteria = "`y` &gt; 2 &amp; `y` &lt; 5 &amp; `x` %in%
c('A', 'B')")</code>. </p>
</li></ul>
 </li></ul>
</td></tr>
<tr><td><code id="ggdensity_+3A_repel">repel</code></td>
<td>
<p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggdensity_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to
<code><a href="ggplot2.html#topic+geom_density">geom_density</a></code> and <code><a href="#topic+ggpar">ggpar</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation
= c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gghistogram">gghistogram</a></code> and <code><a href="#topic+ggpar">ggpar</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create some data format
set.seed(1234)
wdata = data.frame(
   sex = factor(rep(c("F", "M"), each=200)),
   weight = c(rnorm(200, 55), rnorm(200, 58)))

head(wdata, 4)

# Basic density plot
 # Add mean line and marginal rug
ggdensity(wdata, x = "weight", fill = "lightgray",
   add = "mean", rug = TRUE)

# Change outline colors by groups ("sex")
# Use custom palette
ggdensity(wdata, x = "weight",
   add = "mean", rug = TRUE,
   color = "sex", palette = c("#00AFBB", "#E7B800"))


# Change outline and fill colors by groups ("sex")
# Use custom palette
ggdensity(wdata, x = "weight",
   add = "mean", rug = TRUE,
   color = "sex", fill = "sex",
   palette = c("#00AFBB", "#E7B800"))


</code></pre>

<hr>
<h2 id='ggdonutchart'>Donut chart</h2><span id='topic+ggdonutchart'></span>

<h3>Description</h3>

<p>Create a donut chart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggdonutchart(
  data,
  x,
  label = x,
  lab.pos = c("out", "in"),
  lab.adjust = 0,
  lab.font = c(4, "plain", "black"),
  font.family = "",
  color = "black",
  fill = "white",
  palette = NULL,
  size = NULL,
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggdonutchart_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggdonutchart_+3A_x">x</code></td>
<td>
<p>variable containing values for drawing.</p>
</td></tr>
<tr><td><code id="ggdonutchart_+3A_label">label</code></td>
<td>
<p>variable specifying the label of each slice.</p>
</td></tr>
<tr><td><code id="ggdonutchart_+3A_lab.pos">lab.pos</code></td>
<td>
<p>character specifying the position for labels. Allowed values
are &quot;out&quot; (for outside) or &quot;in&quot; (for inside).</p>
</td></tr>
<tr><td><code id="ggdonutchart_+3A_lab.adjust">lab.adjust</code></td>
<td>
<p>numeric value, used to adjust label position when lab.pos =
&quot;in&quot;. Increase or decrease this value to see the effect.</p>
</td></tr>
<tr><td><code id="ggdonutchart_+3A_lab.font">lab.font</code></td>
<td>
<p>a vector of length 3 indicating respectively the size (e.g.:
14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and the
color (e.g.: &quot;red&quot;) of label font. For example <em>lab.font= c(4, &quot;bold&quot;,
&quot;red&quot;)</em>.</p>
</td></tr>
<tr><td><code id="ggdonutchart_+3A_font.family">font.family</code></td>
<td>
<p>character vector specifying font family.</p>
</td></tr>
<tr><td><code id="ggdonutchart_+3A_color">color</code>, <code id="ggdonutchart_+3A_fill">fill</code></td>
<td>
<p>outline and fill colors.</p>
</td></tr>
<tr><td><code id="ggdonutchart_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggdonutchart_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggdonutchart_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggdonutchart_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to be passed to ggpar().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main, xlab,
ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes: palette
= &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title, labels
and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation =
c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>, <code><a href="#topic+ggpie">ggpie</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Data: Create some data
# +++++++++++++++++++++++++++++++

df &lt;- data.frame(
 group = c("Male", "Female", "Child"),
  value = c(25, 25, 50))

head(df)


# Basic pie charts
# ++++++++++++++++++++++++++++++++

ggdonutchart(df, "value", label = "group")


# Change color
# ++++++++++++++++++++++++++++++++

# Change fill color by group
# set line color to white
# Use custom color palette
 ggdonutchart(df, "value", label = "group",
      fill = "group", color = "white",
       palette = c("#00AFBB", "#E7B800", "#FC4E07") )


# Change label
# ++++++++++++++++++++++++++++++++

# Show group names and value as labels
labs &lt;- paste0(df$group, " (", df$value, "%)")
ggdonutchart(df, "value", label = labs,
   fill = "group", color = "white",
   palette = c("#00AFBB", "#E7B800", "#FC4E07"))

# Change the position and font color of labels
ggdonutchart(df, "value", label = labs,
   lab.pos = "in", lab.font = "white",
   fill = "group", color = "white",
   palette = c("#00AFBB", "#E7B800", "#FC4E07"))



</code></pre>

<hr>
<h2 id='ggdotchart'>Cleveland's Dot Plots</h2><span id='topic+ggdotchart'></span><span id='topic+theme_cleveland'></span>

<h3>Description</h3>

<p>Draw a Cleveland dot plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggdotchart(
  data,
  x,
  y,
  group = NULL,
  combine = FALSE,
  color = "black",
  palette = NULL,
  shape = 19,
  size = NULL,
  dot.size = size,
  sorting = c("ascending", "descending", "none"),
  add = c("none", "segment"),
  add.params = list(),
  x.text.col = TRUE,
  rotate = FALSE,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  select = NULL,
  remove = NULL,
  order = NULL,
  label = NULL,
  font.label = list(size = 11, color = "black"),
  label.select = NULL,
  repel = FALSE,
  label.rectangle = FALSE,
  position = "identity",
  ggtheme = theme_pubr(),
  ...
)

theme_cleveland(rotate = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggdotchart_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_x">x</code>, <code id="ggdotchart_+3A_y">y</code></td>
<td>
<p>x and y variables for drawing.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_group">group</code></td>
<td>
<p>an optional column name indicating how the elements of x are
grouped.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_color">color</code>, <code id="ggdotchart_+3A_size">size</code></td>
<td>
<p>points color and size.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_shape">shape</code></td>
<td>
<p>point shape. See <code><a href="#topic+show_point_shapes">show_point_shapes</a></code>.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_dot.size">dot.size</code></td>
<td>
<p>numeric value specifying the dot size.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_sorting">sorting</code></td>
<td>
<p>a character vector for sorting into ascending or descending
order. Allowed values are one of &quot;descending&quot;, &quot;ascending&quot; and &quot;none&quot;. Partial
match are allowed (e.g. sorting = &quot;desc&quot; or &quot;asc&quot;). Default is
&quot;descending&quot;.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_add">add</code></td>
<td>
<p>character vector for adding another plot element (e.g.: dot plot or
error bars). Allowed values are one or the combination of: &quot;none&quot;,
&quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;,
&quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;,
&quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;; see ?desc_statby for more
details.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, shape, size, fill, linetype) for the
argument 'add'; e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_x.text.col">x.text.col</code></td>
<td>
<p>logical. If TRUE (default), x axis texts are colored by
groups.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_rotate">rotate</code></td>
<td>
<p>logical value. If TRUE, rotate the graph by setting the plot
orientation to horizontal.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_select">select</code></td>
<td>
<p>character vector specifying which items to display.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_remove">remove</code></td>
<td>
<p>character vector specifying which items to remove from the plot.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_order">order</code></td>
<td>
<p>character vector specifying the order of items.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_label">label</code></td>
<td>
<p>the name of the column containing point labels.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_font.label">font.label</code></td>
<td>
<p>a list which can contain the combination of the following
elements: the size (e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;,
&quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of labels. For example font.label
= list(size = 14, face = &quot;bold&quot;, color =&quot;red&quot;). To specify only the size and
the style, use font.label = list(size = 14, face = &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_label.select">label.select</code></td>
<td>
<p>can be of two formats: </p>
 <ul>
<li><p> a character vector
specifying some labels to show. </p>
</li>
<li><p> a list containing one or the
combination of the following components: </p>
 <ul>
<li> <p><code>top.up</code> and
<code>top.down</code>: to display the labels  of the top up/down points. For
example, <code>label.select = list(top.up = 10, top.down = 4)</code>. </p>
</li>
<li>
<p><code>criteria</code>: to filter, for example, by x and y variabes values, use
this: <code>label.select = list(criteria = "`y` &gt; 2 &amp; `y` &lt; 5 &amp; `x` %in%
c('A', 'B')")</code>. </p>
</li></ul>
 </li></ul>
</td></tr>
<tr><td><code id="ggdotchart_+3A_repel">repel</code></td>
<td>
<p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string, or the result of a
call to a position adjustment function.</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggdotchart_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>
and <code><a href="#topic+ggpar">ggpar</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation
= c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("mtcars")
df &lt;- mtcars
df$cyl &lt;- as.factor(df$cyl)
df$name &lt;- rownames(df)
head(df[, c("wt", "mpg", "cyl")], 3)

# Basic plot
ggdotchart(df, x = "name", y ="mpg",
  ggtheme = theme_bw())

# Change colors by  group cyl
ggdotchart(df, x = "name", y = "mpg",
   group = "cyl", color = "cyl",
   palette = c('#999999','#E69F00','#56B4E9'),
   rotate = TRUE,
   sorting = "descending",
   ggtheme = theme_bw(),
   y.text.col = TRUE )


# Plot with multiple groups
# +++++++++++++++++++++
# Create some data
df2 &lt;- data.frame(supp=rep(c("VC", "OJ"), each=3),
                 dose=rep(c("D0.5", "D1", "D2"),2),
                 len=c(6.8, 15, 33, 4.2, 10, 29.5))
print(df2)

ggdotchart(df2, x = "dose", y = "len",
          color = "supp", size = 3,
          add = "segment",
          add.params = list(color = "lightgray", size = 1.5),
          position = position_dodge(0.3),
          palette = "jco",
          ggtheme = theme_pubclean()
)


</code></pre>

<hr>
<h2 id='ggdotplot'>Dot plot</h2><span id='topic+ggdotplot'></span>

<h3>Description</h3>

<p>Create a dot plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggdotplot(
  data,
  x,
  y,
  combine = FALSE,
  merge = FALSE,
  color = "black",
  fill = "lightgray",
  palette = NULL,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  size = NULL,
  binwidth = NULL,
  select = NULL,
  remove = NULL,
  order = NULL,
  add = "mean_se",
  add.params = list(),
  error.plot = "pointrange",
  label = NULL,
  font.label = list(size = 11, color = "black"),
  label.select = NULL,
  repel = FALSE,
  label.rectangle = FALSE,
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggdotplot_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_x">x</code></td>
<td>
<p>character string containing the name of x variable.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_y">y</code></td>
<td>
<p>character vector containing one or more variables to plot</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_color">color</code></td>
<td>
<p>outline color.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_fill">fill</code></td>
<td>
<p>fill color.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_binwidth">binwidth</code></td>
<td>
<p>numeric value specifying bin width. use value between 0 and 1
when you have a strong dense dotplot. For example binwidth = 0.2.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_select">select</code></td>
<td>
<p>character vector specifying which items to display.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_remove">remove</code></td>
<td>
<p>character vector specifying which items to remove from the plot.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_order">order</code></td>
<td>
<p>character vector specifying the order of items.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_add">add</code></td>
<td>
<p>character vector for adding another plot element (e.g.: dot plot or
error bars). Allowed values are one or the combination of: &quot;none&quot;,
&quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;,
&quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;,
&quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;; see ?desc_statby for more
details.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, shape, size, fill, linetype) for the
argument 'add'; e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_error.plot">error.plot</code></td>
<td>
<p>plot type used to visualize error. Allowed values are one of
c(&quot;pointrange&quot;, &quot;linerange&quot;, &quot;crossbar&quot;, &quot;errorbar&quot;, &quot;upper_errorbar&quot;,
&quot;lower_errorbar&quot;, &quot;upper_pointrange&quot;, &quot;lower_pointrange&quot;, &quot;upper_linerange&quot;,
&quot;lower_linerange&quot;). Default value is &quot;pointrange&quot; or &quot;errorbar&quot;. Used only
when add != &quot;none&quot; and add contains one &quot;mean_*&quot; or &quot;med_*&quot; where &quot;*&quot; = sd,
se, ....</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_label">label</code></td>
<td>
<p>the name of the column containing point labels. Can be also a
character vector with length = nrow(data).</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_font.label">font.label</code></td>
<td>
<p>a list which can contain the combination of the following
elements: the size (e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;,
&quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of labels. For example font.label
= list(size = 14, face = &quot;bold&quot;, color =&quot;red&quot;). To specify only the size and
the style, use font.label = list(size = 14, face = &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_label.select">label.select</code></td>
<td>
<p>can be of two formats: </p>
 <ul>
<li><p> a character vector
specifying some labels to show. </p>
</li>
<li><p> a list containing one or the
combination of the following components: </p>
 <ul>
<li> <p><code>top.up</code> and
<code>top.down</code>: to display the labels  of the top up/down points. For
example, <code>label.select = list(top.up = 10, top.down = 4)</code>. </p>
</li>
<li>
<p><code>criteria</code>: to filter, for example, by x and y variabes values, use
this: <code>label.select = list(criteria = "`y` &gt; 2 &amp; `y` &lt; 5 &amp; `x` %in%
c('A', 'B')")</code>. </p>
</li></ul>
 </li></ul>
</td></tr>
<tr><td><code id="ggdotplot_+3A_repel">repel</code></td>
<td>
<p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggdotplot_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to
<code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code>, <code><a href="#topic+ggpar">ggpar</a></code> and
<code><a href="#topic+facet">facet</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation
= c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>, <code><a href="#topic+ggviolin">ggviolin</a></code>, <code><a href="#topic+ggboxplot">ggboxplot</a></code>
and <code><a href="#topic+ggstripchart">ggstripchart</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")
df &lt;- ToothGrowth

# Basic plot with summary statistics : mean_sd
# +++++++++++++++++++++++++++
ggdotplot(df, x = "dose", y = "len",
   add = "mean_sd")

# Change error.plot to "crossbar"
ggdotplot(df, x = "dose", y = "len",
 add = "mean_sd", add.params = list(width = 0.5),
 error.plot = "crossbar")


# Add box plot
ggdotplot(df, x = "dose", y = "len",
 add = "boxplot")

# Add violin + mean_sd
ggdotplot(df, x = "dose", y = "len",
 add = c("violin", "mean_sd"))


# Change colors
# +++++++++++++++++++++++++++
# Change fill and outline colors by groups: dose
# Use custom color palette
 ggdotplot(df, "dose", "len",
     add = "boxplot",
      color = "dose", fill = "dose",
      palette = c("#00AFBB", "#E7B800", "#FC4E07"))


# Plot with multiple groups
# +++++++++++++++++++++
# Change color by a second group : "supp"
ggdotplot(df, "dose", "len", fill = "supp", color = "supp",
    palette = c("#00AFBB", "#E7B800"))


</code></pre>

<hr>
<h2 id='ggecdf'>Empirical cumulative density function</h2><span id='topic+ggecdf'></span>

<h3>Description</h3>

<p>Empirical Cumulative Density Function (ECDF).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggecdf(
  data,
  x,
  combine = FALSE,
  merge = FALSE,
  color = "black",
  palette = NULL,
  size = NULL,
  linetype = "solid",
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggecdf_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_x">x</code></td>
<td>
<p>variable to be drawn.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_color">color</code></td>
<td>
<p>line and point color.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_size">size</code></td>
<td>
<p>line and point size.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_linetype">linetype</code></td>
<td>
<p>line type. See <code><a href="#topic+show_line_types">show_line_types</a></code>.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggecdf_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to
<code><a href="ggplot2.html#topic+stat_ecdf">stat_ecdf</a></code> and <code><a href="#topic+ggpar">ggpar</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation
= c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create some data format
set.seed(1234)
wdata = data.frame(
   sex = factor(rep(c("F", "M"), each=200)),
   weight = c(rnorm(200, 55), rnorm(200, 58)))

head(wdata, 4)

# Basic ECDF plot
ggecdf(wdata, x = "weight")

# Change colors and linetype by groups ("sex")
# Use custom palette
ggecdf(wdata, x = "weight",
   color = "sex", linetype = "sex",
   palette = c("#00AFBB", "#E7B800"))

</code></pre>

<hr>
<h2 id='ggerrorplot'>Visualizing Error</h2><span id='topic+ggerrorplot'></span>

<h3>Description</h3>

<p>Visualizing error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggerrorplot(
  data,
  x,
  y,
  desc_stat = "mean_se",
  numeric.x.axis = FALSE,
  combine = FALSE,
  merge = FALSE,
  color = "black",
  fill = "white",
  palette = NULL,
  size = NULL,
  width = NULL,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  select = NULL,
  remove = NULL,
  order = NULL,
  add = "none",
  add.params = list(),
  error.plot = "pointrange",
  ci = 0.95,
  position = position_dodge(),
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggerrorplot_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_x">x</code>, <code id="ggerrorplot_+3A_y">y</code></td>
<td>
<p>x and y variables for drawing.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_desc_stat">desc_stat</code></td>
<td>
<p>descriptive statistics to be used for visualizing errors. Default value is &quot;mean_se&quot;.
Allowed values are one of , &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;, &quot;mean_ci&quot;, &quot;mean_range&quot;,
&quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;, &quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;; see <code><a href="#topic+desc_statby">desc_statby</a></code> for more details.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_numeric.x.axis">numeric.x.axis</code></td>
<td>
<p>logical. If TRUE, x axis will be treated as numeric. Default is FALSE.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_color">color</code>, <code id="ggerrorplot_+3A_fill">fill</code></td>
<td>
<p>outline and fill colors.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_width">width</code></td>
<td>
<p>numeric value between 0 and 1 specifying box width.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_select">select</code></td>
<td>
<p>character vector specifying which items to display.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_remove">remove</code></td>
<td>
<p>character vector specifying which items to remove from the plot.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_order">order</code></td>
<td>
<p>character vector specifying the order of items. Considered only when x axis is a factor variable.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_add">add</code></td>
<td>
<p>character vector for adding another plot element (e.g.: dot plot or
error bars). Allowed values are one or the combination of: &quot;none&quot;,
&quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;,
&quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;,
&quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;; see ?desc_statby for more
details.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, shape, size, fill, linetype) for the
argument 'add'; e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_error.plot">error.plot</code></td>
<td>
<p>plot type used to visualize error. Allowed values are one of
c(&quot;pointrange&quot;, &quot;linerange&quot;, &quot;crossbar&quot;, &quot;errorbar&quot;, &quot;upper_errorbar&quot;,
&quot;lower_errorbar&quot;, &quot;upper_pointrange&quot;, &quot;lower_pointrange&quot;, &quot;upper_linerange&quot;,
&quot;lower_linerange&quot;). Default value is &quot;pointrange&quot; or &quot;errorbar&quot;. Used only
when add != &quot;none&quot; and add contains one &quot;mean_*&quot; or &quot;med_*&quot; where &quot;*&quot; = sd,
se, ....</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_ci">ci</code></td>
<td>
<p>the percent range of the confidence interval (default is 0.95).</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggerrorplot_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to be passed to ggpar().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation
= c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>, <code><a href="#topic+ggline">ggline</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Data: ToothGrowth data set we'll be used.
df&lt;- ToothGrowth
head(df, 10)

# Plot mean_se
ggerrorplot(df, x = "dose", y = "len")


# Change desc_stat to mean_sd
# (other values include: mean_sd, mean_ci, median_iqr, ....)
# Add labels
ggerrorplot(df, x = "dose", y = "len",
 desc_stat = "mean_sd")

# Change error.plot to "errorbar" and add mean point
# Visualize the mean of each group
ggerrorplot(df, x = "dose", y = "len",
 add = "mean", error.plot = "errorbar")

# Horizontal plot
ggerrorplot(df, x = "dose", y = "len",
 add = "mean", error.plot = "errorbar",
 orientation = "horizontal")


# Change error.plot to "crossbar"
ggerrorplot(df, x = "dose", y = "len",
 error.plot = "crossbar", width = 0.5)


# Add jitter points and errors (mean_se)
ggerrorplot(df, x = "dose", y = "len",
 add = "jitter")

# Add dot and errors (mean_se)
ggerrorplot(df, x = "dose", y = "len",
 add = "dotplot")

# Multiple groups with error bars and jitter point
ggerrorplot(df, x = "dose", y = "len",
 color = "supp", palette = "Paired",
 error.plot = "pointrange",
 position = position_dodge(0.5))


</code></pre>

<hr>
<h2 id='ggexport'>Export ggplots</h2><span id='topic+ggexport'></span>

<h3>Description</h3>

<p>Export ggplots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggexport(
  ...,
  plotlist = NULL,
  filename = NULL,
  ncol = NULL,
  nrow = NULL,
  width = 480,
  height = 480,
  pointsize = 12,
  res = NA,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggexport_+3A_...">...</code></td>
<td>
<p>list of plots to be arranged into the grid. The plots can be either
ggplot2 plot objects, arbitrary gtables or an object of class
<code><a href="#topic+ggarrange">ggarrange</a></code>.</p>
</td></tr>
<tr><td><code id="ggexport_+3A_plotlist">plotlist</code></td>
<td>
<p>(optional) list of plots to display.</p>
</td></tr>
<tr><td><code id="ggexport_+3A_filename">filename</code></td>
<td>
<p>File name to create on disk.</p>
</td></tr>
<tr><td><code id="ggexport_+3A_ncol">ncol</code></td>
<td>
<p>(optional) number of columns in the plot grid.</p>
</td></tr>
<tr><td><code id="ggexport_+3A_nrow">nrow</code></td>
<td>
<p>(optional) number of rows in the plot grid.</p>
</td></tr>
<tr><td><code id="ggexport_+3A_width">width</code>, <code id="ggexport_+3A_height">height</code></td>
<td>
<p>plot width and height, respectively (example, width = 800,
height = 800). Applied only to raster plots: &quot;png&quot;, &quot;jpeg&quot;, &quot;jpg&quot;, &quot;bmp&quot; and
&quot;tiff&quot;.</p>
</td></tr>
<tr><td><code id="ggexport_+3A_pointsize">pointsize</code></td>
<td>
<p>the default pointsize of plotted text (example, pointsize =
8). Used only for raster plots.</p>
</td></tr>
<tr><td><code id="ggexport_+3A_res">res</code></td>
<td>
<p>the resolution in ppi (example, res = 250). Used only for raster
plots.</p>
</td></tr>
<tr><td><code id="ggexport_+3A_verbose">verbose</code></td>
<td>
<p>logical. If TRUE, show message.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara &lt;alboukadel.kassambara@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require("magrittr")
# Load data
data("ToothGrowth")
df &lt;- ToothGrowth
df$dose &lt;- as.factor(df$dose)

# Box plot
bxp &lt;- ggboxplot(df, x = "dose", y = "len",
    color = "dose", palette = "jco")
# Dot plot
dp &lt;- ggdotplot(df, x = "dose", y = "len",
    color = "dose", palette = "jco")
# Density plot
dens &lt;- ggdensity(df, x = "len", fill = "dose", palette = "jco")

# Export to pdf
ggarrange(bxp, dp, dens, ncol = 2) %&gt;%
  ggexport(filename = "test.pdf")

# Export to png
ggarrange(bxp, dp, dens, ncol = 2) %&gt;%
  ggexport(filename = "test.png")
 
## End(Not run)

</code></pre>

<hr>
<h2 id='gghistogram'>Histogram plot</h2><span id='topic+gghistogram'></span>

<h3>Description</h3>

<p>Create a histogram plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gghistogram(
  data,
  x,
  y = "count",
  combine = FALSE,
  merge = FALSE,
  weight = NULL,
  color = "black",
  fill = NA,
  palette = NULL,
  size = NULL,
  linetype = "solid",
  alpha = 0.5,
  bins = NULL,
  binwidth = NULL,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  add = c("none", "mean", "median"),
  add.params = list(linetype = "dashed"),
  rug = FALSE,
  add_density = FALSE,
  label = NULL,
  font.label = list(size = 11, color = "black"),
  label.select = NULL,
  repel = FALSE,
  label.rectangle = FALSE,
  position = position_identity(),
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gghistogram_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_x">x</code></td>
<td>
<p>variable to be drawn.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_y">y</code></td>
<td>
<p>one of &quot;density&quot; or &quot;count&quot;.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_weight">weight</code></td>
<td>
<p>a variable name available in the input data for creating a weighted histogram.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_color">color</code>, <code id="gghistogram_+3A_fill">fill</code></td>
<td>
<p>histogram line color and fill color.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_linetype">linetype</code></td>
<td>
<p>line type. See <code><a href="#topic+show_line_types">show_line_types</a></code>.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_alpha">alpha</code></td>
<td>
<p>numeric value specifying fill color transparency. Value should
be in [0, 1], where 0 is full transparency and 1 is no transparency.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_bins">bins</code></td>
<td>
<p>Number of bins. Defaults to 30.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_binwidth">binwidth</code></td>
<td>
<p>numeric value specifying bin width. use value between 0 and 1
when you have a strong dense dotplot. For example binwidth = 0.2.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_add">add</code></td>
<td>
<p>allowed values are one of &quot;mean&quot; or &quot;median&quot; (for adding mean or
median line, respectively).</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, size, linetype) for the argument 'add';
e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_rug">rug</code></td>
<td>
<p>logical value. If TRUE, add marginal rug.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_add_density">add_density</code></td>
<td>
<p>logical value. If TRUE, add density curves.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_label">label</code></td>
<td>
<p>the name of the column containing point labels. Can be also a
character vector with length = nrow(data).</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_font.label">font.label</code></td>
<td>
<p>a list which can contain the combination of the following
elements: the size (e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;,
&quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of labels. For example font.label
= list(size = 14, face = &quot;bold&quot;, color =&quot;red&quot;). To specify only the size and
the style, use font.label = list(size = 14, face = &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_label.select">label.select</code></td>
<td>
<p>can be of two formats: </p>
 <ul>
<li><p> a character vector
specifying some labels to show. </p>
</li>
<li><p> a list containing one or the
combination of the following components: </p>
 <ul>
<li> <p><code>top.up</code> and
<code>top.down</code>: to display the labels  of the top up/down points. For
example, <code>label.select = list(top.up = 10, top.down = 4)</code>. </p>
</li>
<li>
<p><code>criteria</code>: to filter, for example, by x and y variabes values, use
this: <code>label.select = list(criteria = "`y` &gt; 2 &amp; `y` &lt; 5 &amp; `x` %in%
c('A', 'B')")</code>. </p>
</li></ul>
 </li></ul>
</td></tr>
<tr><td><code id="gghistogram_+3A_repel">repel</code></td>
<td>
<p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string, or the result of a
call to a position adjustment function. Allowed values include &quot;identity&quot;,
&quot;stack&quot;, &quot;dodge&quot;.</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="gghistogram_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to
<code><a href="ggplot2.html#topic+geom_histogram">geom_histogram</a></code> and <code><a href="#topic+ggpar">ggpar</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation
= c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggdensity">ggdensity</a></code> and <code><a href="#topic+ggpar">ggpar</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create some data format
set.seed(1234)
wdata = data.frame(
   sex = factor(rep(c("F", "M"), each=200)),
   weight = c(rnorm(200, 55), rnorm(200, 58)))

head(wdata, 4)

# Basic density plot
# Add mean line and marginal rug
gghistogram(wdata, x = "weight", fill = "lightgray",
   add = "mean", rug = TRUE)

# Change outline colors by groups ("sex")
# Use custom color palette
gghistogram(wdata, x = "weight",
   add = "mean", rug = TRUE,
   color = "sex", palette = c("#00AFBB", "#E7B800"))

# Change outline and fill colors by groups ("sex")
# Use custom color palette
gghistogram(wdata, x = "weight",
   add = "mean", rug = TRUE,
   color = "sex", fill = "sex",
   palette = c("#00AFBB", "#E7B800"))



# Combine histogram and density plots
gghistogram(wdata, x = "weight",
   add = "mean", rug = TRUE,
   fill = "sex", palette = c("#00AFBB", "#E7B800"),
   add_density = TRUE)

# Weighted histogram
gghistogram(iris, x = "Sepal.Length", weight = "Petal.Length")
</code></pre>

<hr>
<h2 id='ggline'>Line plot</h2><span id='topic+ggline'></span>

<h3>Description</h3>

<p>Create a line plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggline(
  data,
  x,
  y,
  group = 1,
  numeric.x.axis = FALSE,
  combine = FALSE,
  merge = FALSE,
  color = "black",
  palette = NULL,
  linetype = "solid",
  plot_type = c("b", "l", "p"),
  size = 0.5,
  shape = 19,
  stroke = NULL,
  point.size = size,
  point.color = color,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  select = NULL,
  remove = NULL,
  order = NULL,
  add = "none",
  add.params = list(),
  error.plot = "errorbar",
  label = NULL,
  font.label = list(size = 11, color = "black"),
  label.select = NULL,
  repel = FALSE,
  label.rectangle = FALSE,
  show.line.label = FALSE,
  position = "identity",
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggline_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggline_+3A_x">x</code>, <code id="ggline_+3A_y">y</code></td>
<td>
<p>x and y variables for drawing.</p>
</td></tr>
<tr><td><code id="ggline_+3A_group">group</code></td>
<td>
<p>grouping variable to connect points by line.
Allowed values are 1 (for one line, one group) or a character vector specifying
the name of the grouping variable (case of multiple lines).</p>
</td></tr>
<tr><td><code id="ggline_+3A_numeric.x.axis">numeric.x.axis</code></td>
<td>
<p>logical. If TRUE, x axis will be treated as numeric. Default is FALSE.</p>
</td></tr>
<tr><td><code id="ggline_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggline_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggline_+3A_color">color</code></td>
<td>
<p>line colors.</p>
</td></tr>
<tr><td><code id="ggline_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggline_+3A_linetype">linetype</code></td>
<td>
<p>line type.</p>
</td></tr>
<tr><td><code id="ggline_+3A_plot_type">plot_type</code></td>
<td>
<p>plot type. Allowed values are one of &quot;b&quot; for both line and point;
&quot;l&quot; for line only; and &quot;p&quot; for point only. Default is &quot;b&quot;.</p>
</td></tr>
<tr><td><code id="ggline_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggline_+3A_shape">shape</code></td>
<td>
<p>point shapes.</p>
</td></tr>
<tr><td><code id="ggline_+3A_stroke">stroke</code></td>
<td>
<p>point stroke. Used only for shapes 21-24 to control the thickness of points border.</p>
</td></tr>
<tr><td><code id="ggline_+3A_point.size">point.size</code></td>
<td>
<p>point size.</p>
</td></tr>
<tr><td><code id="ggline_+3A_point.color">point.color</code></td>
<td>
<p>point color.</p>
</td></tr>
<tr><td><code id="ggline_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggline_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggline_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggline_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggline_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggline_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggline_+3A_select">select</code></td>
<td>
<p>character vector specifying which items to display.</p>
</td></tr>
<tr><td><code id="ggline_+3A_remove">remove</code></td>
<td>
<p>character vector specifying which items to remove from the plot.</p>
</td></tr>
<tr><td><code id="ggline_+3A_order">order</code></td>
<td>
<p>character vector specifying the order of items.</p>
</td></tr>
<tr><td><code id="ggline_+3A_add">add</code></td>
<td>
<p>character vector for adding another plot element (e.g.: dot plot or
error bars). Allowed values are one or the combination of: &quot;none&quot;,
&quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;,
&quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;,
&quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;; see ?desc_statby for more
details.</p>
</td></tr>
<tr><td><code id="ggline_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, shape, size, fill, linetype) for the
argument 'add'; e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggline_+3A_error.plot">error.plot</code></td>
<td>
<p>plot type used to visualize error. Allowed values are one of
c(&quot;pointrange&quot;, &quot;linerange&quot;, &quot;crossbar&quot;, &quot;errorbar&quot;, &quot;upper_errorbar&quot;,
&quot;lower_errorbar&quot;, &quot;upper_pointrange&quot;, &quot;lower_pointrange&quot;, &quot;upper_linerange&quot;,
&quot;lower_linerange&quot;). Default value is &quot;pointrange&quot; or &quot;errorbar&quot;. Used only
when add != &quot;none&quot; and add contains one &quot;mean_*&quot; or &quot;med_*&quot; where &quot;*&quot; = sd,
se, ....</p>
</td></tr>
<tr><td><code id="ggline_+3A_label">label</code></td>
<td>
<p>the name of the column containing point labels. Can be also a
character vector with length = nrow(data).</p>
</td></tr>
<tr><td><code id="ggline_+3A_font.label">font.label</code></td>
<td>
<p>a list which can contain the combination of the following
elements: the size (e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;,
&quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of labels. For example font.label
= list(size = 14, face = &quot;bold&quot;, color =&quot;red&quot;). To specify only the size and
the style, use font.label = list(size = 14, face = &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="ggline_+3A_label.select">label.select</code></td>
<td>
<p>can be of two formats: </p>
 <ul>
<li><p> a character vector
specifying some labels to show. </p>
</li>
<li><p> a list containing one or the
combination of the following components: </p>
 <ul>
<li> <p><code>top.up</code> and
<code>top.down</code>: to display the labels  of the top up/down points. For
example, <code>label.select = list(top.up = 10, top.down = 4)</code>. </p>
</li>
<li>
<p><code>criteria</code>: to filter, for example, by x and y variabes values, use
this: <code>label.select = list(criteria = "`y` &gt; 2 &amp; `y` &lt; 5 &amp; `x` %in%
c('A', 'B')")</code>. </p>
</li></ul>
 </li></ul>
</td></tr>
<tr><td><code id="ggline_+3A_repel">repel</code></td>
<td>
<p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</td></tr>
<tr><td><code id="ggline_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="ggline_+3A_show.line.label">show.line.label</code></td>
<td>
<p>logical value. If TRUE, shows line labels.</p>
</td></tr>
<tr><td><code id="ggline_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="ggline_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggline_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to geom_dotplot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation
= c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>, <code><a href="#topic+ggbarplot">ggbarplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Data
df &lt;- data.frame(dose=c("D0.5", "D1", "D2"),
   len=c(4.2, 10, 29.5))
print(df)

# Basic plot
# +++++++++++++++++++++++++++
ggline(df, x = "dose", y = "len")


# Plot with multiple groups
# +++++++++++++++++++++

# Create some data
df2 &lt;- data.frame(supp=rep(c("VC", "OJ"), each=3),
   dose=rep(c("D0.5", "D1", "D2"),2),
   len=c(6.8, 15, 33, 4.2, 10, 29.5))
print(df2)

# Plot "len" by "dose" and
# Change line types and point shapes by a second groups: "supp"
ggline(df2, "dose", "len",
  linetype = "supp", shape = "supp")


# Change colors
# +++++++++++++++++++++

# Change color by group: "supp"
# Use custom color palette
ggline(df2, "dose", "len",
   linetype = "supp", shape = "supp",
   color = "supp", palette = c("#00AFBB", "#E7B800"))


# Add points and errors
# ++++++++++++++++++++++++++

# Data: ToothGrowth data set we'll be used.
df3 &lt;- ToothGrowth
head(df3, 10)

# It can be seen that for each group we have
# different values
ggline(df3, x = "dose", y = "len")

# Visualize the mean of each group
ggline(df3, x = "dose", y = "len",
 add = "mean")

# Add error bars: mean_se
# (other values include: mean_sd, mean_ci, median_iqr, ....)
# Add labels
ggline(df3, x = "dose", y = "len", add = "mean_se")

# Change error.plot to "pointrange"
ggline(df3, x = "dose", y = "len",
 add = "mean_se", error.plot = "pointrange")

# Add jitter points and errors (mean_se)
ggline(df3, x = "dose", y = "len",
 add = c("mean_se", "jitter"))

# Add dot and errors (mean_se)
ggline(df3, x = "dose", y = "len",
 add = c("mean_se", "dotplot"), color = "steelblue")

# Add violin and errors (mean_se)
ggline(df3, x = "dose", y = "len",
 add = c("mean_se", "violin"), color = "steelblue")

# Multiple groups with error bars
# ++++++++++++++++++++++

ggline(df3, x = "dose", y = "len", color = "supp",
 add = "mean_se", palette = c("#00AFBB", "#E7B800"))

# Add jitter
ggline(df3, x = "dose", y = "len", color = "supp",
 add = c("mean_se", "jitter"), palette = c("#00AFBB", "#E7B800"))

# Add dot plot
ggline(df3, x = "dose", y = "len", color = "supp",
 add = c("mean_se", "dotplot"), palette = c("#00AFBB", "#E7B800"))


</code></pre>

<hr>
<h2 id='ggmaplot'>MA-plot from means and log fold changes</h2><span id='topic+ggmaplot'></span>

<h3>Description</h3>

<p>Make MA-plot which is a scatter plot of log2 fold changes (M, on
the y-axis) versus the average expression signal (A, on the x-axis). <code>M
 = log2(x/y)</code> and <code>A = (log2(x) + log2(y))/2 = log2(xy)*1/2</code>, where x
and y are respectively the mean of the two groups being compared.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggmaplot(
  data,
  fdr = 0.05,
  fc = 1.5,
  genenames = NULL,
  detection_call = NULL,
  size = NULL,
  alpha = 1,
  seed = 42,
  font.label = c(12, "plain", "black"),
  label.rectangle = FALSE,
  palette = c("#B31B21", "#1465AC", "darkgray"),
  top = 15,
  select.top.method = c("padj", "fc"),
  label.select = NULL,
  main = NULL,
  xlab = "Log2 mean expression",
  ylab = "Log2 fold change",
  ggtheme = theme_classic(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggmaplot_+3A_data">data</code></td>
<td>
<p>an object of class DESeqResults, get_diff, DE_Results, matrix or
data frame containing the columns baseMean (or baseMeanLog2),
log2FoldChange, and padj. Rows are genes.
</p>
<p>Two possible formats are accepted for the input data: </p>
 <ul>
<li><p> 1/
<code>baseMean | log2FoldChange | padj</code>. This is a typical output from
DESeq2 pipeline. Here, we'll use log2(baseMean) as the x-axis variable.
</p>
</li>
<li><p> 2/ <code>baseMeanLog2 | log2FoldChange | padj</code>. Here, baseMeanLog2 is
assumed to be the mean of logged values; so we'll use it as the x-axis
variable without any transformation. This is the real A in MA plot. In other
words, it is the average of two log-scales values: <code>A = (log2(x) +
 log2(y))/2 = log2(xy)*1/2</code> </p>
</li></ul>

<p>Terminology:
</p>
 <ul>
<li><p> baseMean: the mean expression of genes in the two groups.
</p>
</li>
<li><p> log2FoldChange: the log2 fold changes of group 2 compared to group 1
</p>
</li>
<li><p> padj: the adjusted p-value of the used statiscal test. </p>
</li></ul>
</td></tr>
<tr><td><code id="ggmaplot_+3A_fdr">fdr</code></td>
<td>
<p>Accepted false discovery rate for considering genes as
differentially expressed.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_fc">fc</code></td>
<td>
<p>the fold change threshold. Only genes with a fold change &gt;= fc and
padj &lt;= fdr are considered as significantly differentially expressed.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_genenames">genenames</code></td>
<td>
<p>a character vector of length nrow(data) specifying gene names
corresponding to each row. Used for point labels.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_detection_call">detection_call</code></td>
<td>
<p>a numeric vector with length = nrow(data), specifying if
the genes is expressed (value = 1) or not (value = 0). For example
detection_call = c(1, 1, 0, 1, 0, 1). Default is NULL. If detection_call
column is available in data, it will be used.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_size">size</code></td>
<td>
<p>points size.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_alpha">alpha</code></td>
<td>
<p>numeric value betwenn 0 an 1 specifying point alpha for
controlling transparency. For example, use alpha = 0.5.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_seed">seed</code></td>
<td>
<p>Random seed passed to <code>set.seed</code>. if
<code>NA</code>, set.seed will not be called. Default is 42 for reproducibility.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_font.label">font.label</code></td>
<td>
<p>a vector of length 3 indicating respectively the size (e.g.:
14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and the
color (e.g.: &quot;red&quot;) of point labels. For example <em>font.label = c(14,
&quot;bold&quot;, &quot;red&quot;)</em>.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_top">top</code></td>
<td>
<p>the number of top genes to be shown on the plot. Use top = 0 to
hide to gene labels.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_select.top.method">select.top.method</code></td>
<td>
<p>methods to be used for selecting top genes. Allowed
values include &quot;padj&quot; and &quot;fc&quot; for selecting by adjusted p values or fold
changes, respectively.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_label.select">label.select</code></td>
<td>
<p>character vector specifying some labels to show.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_main">main</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggmaplot_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <code><a href="#topic+ggpar">ggpar</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a ggplot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(diff_express)

# Default plot
ggmaplot(diff_express, main = expression("Group 1" %-&gt;% "Group 2"),
   fdr = 0.05, fc = 2, size = 0.4,
   palette = c("#B31B21", "#1465AC", "darkgray"),
   genenames = as.vector(diff_express$name),
   legend = "top", top = 20,
   font.label = c("bold", 11),
   font.legend = "bold",
   font.main = "bold",
   ggtheme = ggplot2::theme_minimal())

# Add rectangle around labels
ggmaplot(diff_express, main = expression("Group 1" %-&gt;% "Group 2"),
   fdr = 0.05, fc = 2, size = 0.4,
   palette = c("#B31B21", "#1465AC", "darkgray"),
   genenames = as.vector(diff_express$name),
   legend = "top", top = 20,
   font.label = c("bold", 11), label.rectangle = TRUE,
   font.legend = "bold",
   font.main = "bold",
   ggtheme = ggplot2::theme_minimal())

# Select specific genes to show
# set top = 0, then specify genes using label.select argument
ggmaplot(diff_express, main = expression("Group 1" %-&gt;% "Group 2"),
         fdr = 0.05, fc = 2, size = 0.4,
         genenames = as.vector(diff_express$name),
         ggtheme = ggplot2::theme_minimal(),
         top = 0, label.select = c("BUB1", "CD83")
)

</code></pre>

<hr>
<h2 id='ggpaired'>Plot Paired Data</h2><span id='topic+ggpaired'></span>

<h3>Description</h3>

<p>Plot paired data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggpaired(
  data,
  cond1,
  cond2,
  x = NULL,
  y = NULL,
  id = NULL,
  color = "black",
  fill = "white",
  palette = NULL,
  width = 0.5,
  point.size = 1.2,
  line.size = 0.5,
  line.color = "black",
  linetype = "solid",
  title = NULL,
  xlab = "Condition",
  ylab = "Value",
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  label = NULL,
  font.label = list(size = 11, color = "black"),
  label.select = NULL,
  repel = FALSE,
  label.rectangle = FALSE,
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggpaired_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_cond1">cond1</code></td>
<td>
<p>variable name corresponding to the first condition.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_cond2">cond2</code></td>
<td>
<p>variable name corresponding to the second condition.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_x">x</code>, <code id="ggpaired_+3A_y">y</code></td>
<td>
<p>x and y variables, where x is a grouping variable and y contains
values for each group. Considered only when <code>cond1</code> and <code>cond2</code>
are missing.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_id">id</code></td>
<td>
<p>variable name corresponding to paired samples' id. Used to connect
paired points with lines.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_color">color</code></td>
<td>
<p>points and box plot colors. To color by conditions, use color =
&quot;condition&quot;.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_fill">fill</code></td>
<td>
<p>box plot fill color. To change fill color by conditions, use fill
= &quot;condition&quot;.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_width">width</code></td>
<td>
<p>box plot width.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_point.size">point.size</code>, <code id="ggpaired_+3A_line.size">line.size</code></td>
<td>
<p>point and line size, respectively.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_line.color">line.color</code></td>
<td>
<p>line color.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_linetype">linetype</code></td>
<td>
<p>line type.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_label">label</code></td>
<td>
<p>the name of the column containing point labels. Can be also a
character vector with length = nrow(data).</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_font.label">font.label</code></td>
<td>
<p>a list which can contain the combination of the following
elements: the size (e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;,
&quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of labels. For example font.label
= list(size = 14, face = &quot;bold&quot;, color =&quot;red&quot;). To specify only the size and
the style, use font.label = list(size = 14, face = &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_label.select">label.select</code></td>
<td>
<p>can be of two formats: </p>
 <ul>
<li><p> a character vector
specifying some labels to show. </p>
</li>
<li><p> a list containing one or the
combination of the following components: </p>
 <ul>
<li> <p><code>top.up</code> and
<code>top.down</code>: to display the labels  of the top up/down points. For
example, <code>label.select = list(top.up = 10, top.down = 4)</code>. </p>
</li>
<li>
<p><code>criteria</code>: to filter, for example, by x and y variabes values, use
this: <code>label.select = list(criteria = "`y` &gt; 2 &amp; `y` &lt; 5 &amp; `x` %in%
c('A', 'B')")</code>. </p>
</li></ul>
 </li></ul>
</td></tr>
<tr><td><code id="ggpaired_+3A_repel">repel</code></td>
<td>
<p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggpaired_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to be passed to <a href="#topic+ggpar">ggpar</a>().</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# Example 1
#::::::::::::::::::::::::::::::::::::::::::
before &lt;-c(200.1, 190.9, 192.7, 213, 241.4, 196.9, 172.2, 185.5, 205.2, 193.7)
after &lt;-c(392.9, 393.2, 345.1, 393, 434, 427.9, 422, 383.9, 392.3, 352.2)

d &lt;- data.frame(before = before, after = after)
ggpaired(d, cond1 = "before", cond2 = "after",
    fill = "condition", palette = "jco")

# Example 2
#::::::::::::::::::::::::::::::::::::::::::
ggpaired(ToothGrowth, x = "supp", y = "len",
 color = "supp", line.color = "gray", line.size = 0.4,
 palette = "npg")

</code></pre>

<hr>
<h2 id='ggpar'>Graphical parameters</h2><span id='topic+ggpar'></span>

<h3>Description</h3>

<p>Graphical parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggpar(
  p,
  palette = NULL,
  gradient.cols = NULL,
  main = NULL,
  submain = NULL,
  caption = NULL,
  xlab = NULL,
  ylab = NULL,
  title = NULL,
  subtitle = NULL,
  font.main = NULL,
  font.submain = NULL,
  font.x = NULL,
  font.y = NULL,
  font.caption = NULL,
  font.title = NULL,
  font.subtitle = NULL,
  font.family = "",
  xlim = NULL,
  ylim = NULL,
  xscale = c("none", "log2", "log10", "sqrt"),
  yscale = c("none", "log2", "log10", "sqrt"),
  format.scale = FALSE,
  legend = NULL,
  legend.title = NULL,
  font.legend = NULL,
  ticks = TRUE,
  tickslab = TRUE,
  font.tickslab = NULL,
  font.xtickslab = font.tickslab,
  font.ytickslab = font.tickslab,
  x.text.angle = NULL,
  y.text.angle = NULL,
  xtickslab.rt = x.text.angle,
  ytickslab.rt = y.text.angle,
  xticks.by = NULL,
  yticks.by = NULL,
  rotate = FALSE,
  orientation = c("vertical", "horizontal", "reverse"),
  ggtheme = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggpar_+3A_p">p</code></td>
<td>
<p>an object of class ggplot or a list of ggplots</p>
</td></tr>
<tr><td><code id="ggpar_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by
groups. Allowed values include &quot;grey&quot; for grey color palettes; brewer
palettes e.g. &quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;,
&quot;red&quot;); and scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;,
&quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and
&quot;rickandmorty&quot;. Can be also a numeric vector of length(groups); in this
case a basic color palette is created using the function
<a href="grDevices.html#topic+palette">palette</a>.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_gradient.cols">gradient.cols</code></td>
<td>
<p>vector of colors to use for n-colour gradient. Allowed
values include brewer and ggsci color palettes.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_main">main</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_submain">submain</code>, <code id="ggpar_+3A_subtitle">subtitle</code></td>
<td>
<p>plot subtitle.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_caption">caption</code></td>
<td>
<p>plot caption.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_font.main">font.main</code>, <code id="ggpar_+3A_font.submain">font.submain</code>, <code id="ggpar_+3A_font.caption">font.caption</code>, <code id="ggpar_+3A_font.x">font.x</code>, <code id="ggpar_+3A_font.y">font.y</code></td>
<td>
<p>a vector of length 3
indicating respectively the size (e.g.: 14), the style (e.g.: &quot;plain&quot;,
&quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of main title,
subtitle, caption, xlab and ylab, respectively. For example <em>font.x =
c(14, &quot;bold&quot;, &quot;red&quot;)</em>. Use font.x = 14, to change only font size; or use
font.x = &quot;bold&quot;, to change only font face.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_font.title">font.title</code>, <code id="ggpar_+3A_font.subtitle">font.subtitle</code></td>
<td>
<p>alias of font.submain and font.submain,
respectively.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_font.family">font.family</code></td>
<td>
<p>character vector specifying font family.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_xlim">xlim</code>, <code id="ggpar_+3A_ylim">ylim</code></td>
<td>
<p>a numeric vector of length 2, specifying  x and y axis
limits (minimum and maximum), respectively. e.g.: ylim = c(0, 50).</p>
</td></tr>
<tr><td><code id="ggpar_+3A_xscale">xscale</code>, <code id="ggpar_+3A_yscale">yscale</code></td>
<td>
<p>x and y axis scale, respectively. Allowed values are one
of c(&quot;none&quot;, &quot;log2&quot;, &quot;log10&quot;, &quot;sqrt&quot;); e.g.: yscale=&quot;log2&quot;.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_format.scale">format.scale</code></td>
<td>
<p>logical value. If TRUE, axis tick mark labels will be
formatted when xscale or yscale = &quot;log2&quot; or &quot;log10&quot;.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_legend">legend</code></td>
<td>
<p>character specifying legend position. Allowed values are one of
c(&quot;top&quot;, &quot;bottom&quot;, &quot;left&quot;, &quot;right&quot;, &quot;none&quot;). To remove the legend use
legend = &quot;none&quot;. Legend position can be also specified using a numeric
vector c(x, y); see details section.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title, e.g.: <code>legend.title = "Species"</code>. Can
be also a list, <code>legend.title = list(color = "Species", linetype = "Species",
shape = "Species")</code>.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_font.legend">font.legend</code></td>
<td>
<p>legend text font style; e.g.: font.legend = c(10, &quot;plain&quot;,
&quot;black&quot;).</p>
</td></tr>
<tr><td><code id="ggpar_+3A_ticks">ticks</code></td>
<td>
<p>logical value. Default is TRUE. If FALSE, hide axis tick marks.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_tickslab">tickslab</code></td>
<td>
<p>logical value. Default is TRUE. If FALSE, hide axis tick
labels.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_font.tickslab">font.tickslab</code>, <code id="ggpar_+3A_font.xtickslab">font.xtickslab</code>, <code id="ggpar_+3A_font.ytickslab">font.ytickslab</code></td>
<td>
<p>Font style (size, face,
color) for tick labels, e.g.: c(14, &quot;bold&quot;, &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggpar_+3A_x.text.angle">x.text.angle</code>, <code id="ggpar_+3A_y.text.angle">y.text.angle</code></td>
<td>
<p>Numeric value specifying the rotation angle
of x and y axis tick labels, respectively. Default value is NULL. For
vertical x axis texts use x.text.angle = 90.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_xtickslab.rt">xtickslab.rt</code>, <code id="ggpar_+3A_ytickslab.rt">ytickslab.rt</code></td>
<td>
<p>Same as x.text.angle and y.text.angle,
respectively. Will be deprecated in the near future.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_xticks.by">xticks.by</code>, <code id="ggpar_+3A_yticks.by">yticks.by</code></td>
<td>
<p>numeric value controlling x and y axis breaks,
respectively. For example, if yticks.by = 5, a tick mark is shown on every
5. Default value is NULL.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_rotate">rotate</code></td>
<td>
<p>logical value. If TRUE, rotate the graph by setting the plot
orientation to horizontal.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_orientation">orientation</code></td>
<td>
<p>change the orientation of the plot. Allowed values are one
of c( &quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;). Partial match is allowed.</p>
</td></tr>
<tr><td><code id="ggpar_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggpar_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")
df &lt;- ToothGrowth

# Basic box plot
# +++++++++++++++++++++++++++

p &lt;- ggboxplot(df, x = "dose", y = "len")

# Change the plot orientation: horizontal
ggpar(p, orientation = "horiz")


 # Change main title and axis labels
 # ++++++++++++++++++++++++++++

 ggpar(p,
   main = "Plot of length \n by dose",
   xlab = "Dose (mg)", ylab = "Length")

 # Title font styles: 'plain', 'italic', 'bold', 'bold.italic'
 ggpar(p,
   main = "Length by dose",
   font.main = c(14,"bold.italic", "red"),
   font.x = c(14, "bold", "#2E9FDF"),
   font.y = c(14, "bold", "#E7B800"))

 # Hide axis labels
 ggpar(p, xlab = FALSE, ylab = FALSE)


# Change colors
# ++++++++++++++++++++++

# Change outline colors by groups: dose
 p2 &lt;- ggboxplot(df, "dose", "len", color = "dose")
 p2

# Use custom color palette
ggpar(p2, palette = c("#00AFBB", "#E7B800", "#FC4E07"))

# Use brewer palette
ggpar(p2, palette = "Dark2" )

# Use grey palette
ggpar(p2, palette = "grey")

# Use scientific journal palette from ggsci package
ggpar(p2, palette = "npg") # nature

# Axis ticks, limits, scales
# +++++++++++++++++++++++++

# Axis ticks labels and rotation
ggpar(p,
 font.tickslab = c(14,"bold", "#993333"),
 xtickslab.rt = 45, ytickslab.rt = 45)
# Hide axis ticks and tick labels
ggpar(p, ticks = FALSE, tickslab = FALSE)

# Axis limits
ggpar(p, ylim = c(0, 50))

# Axis scale
ggpar(p, yscale = "log2")

# Format axis scale
ggpar(p, yscale = "log2", format.scale = TRUE)

# Legends
# ++++++++++++++++++
# Change legend position and title
ggpar(p2,
 legend = "right", legend.title = "Dose (mg)",
 font.legend = c(10, "bold", "red"))

</code></pre>

<hr>
<h2 id='ggparagraph'>Draw a Paragraph of Text</h2><span id='topic+ggparagraph'></span><span id='topic+drawDetails.splitText'></span>

<h3>Description</h3>

<p>Draw a paragraph o text. Splits a long text into multiple lines
(by inserting line breaks) so that the output will fit within the current
viewport.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggparagraph(
  text,
  color = NULL,
  size = NULL,
  face = NULL,
  family = NULL,
  lineheight = NULL
)

## S3 method for class 'splitText'
drawDetails(x, recording)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggparagraph_+3A_text">text</code></td>
<td>
<p>the text to plot.</p>
</td></tr>
<tr><td><code id="ggparagraph_+3A_color">color</code></td>
<td>
<p>font color, example: color = &quot;black&quot;</p>
</td></tr>
<tr><td><code id="ggparagraph_+3A_size">size</code></td>
<td>
<p>font size, example: size = 12</p>
</td></tr>
<tr><td><code id="ggparagraph_+3A_face">face</code></td>
<td>
<p>font face. Allowed values are one of &quot;plain&quot;, &quot;italic&quot;, &quot;bold&quot;,
&quot;bold.italic&quot;.</p>
</td></tr>
<tr><td><code id="ggparagraph_+3A_family">family</code></td>
<td>
<p>font family</p>
</td></tr>
<tr><td><code id="ggparagraph_+3A_lineheight">lineheight</code></td>
<td>
<p>Line height, example: lineheight = 2.</p>
</td></tr>
<tr><td><code id="ggparagraph_+3A_x">x</code></td>
<td>
<p>a grid grob</p>
</td></tr>
<tr><td><code id="ggparagraph_+3A_recording">recording</code></td>
<td>
<p>a logical value indicating whether a grob is being added to
the display list or redrawn from the display list.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara &lt;alboukadel.kassambara@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Density plot
density.p &lt;- ggdensity(iris, x = "Sepal.Length",
                      fill = "Species", palette = "jco")

# Text plot
text &lt;- paste("iris data set gives the measurements in cm",
             "of the variables sepal length and width",
             "and petal length and width, respectively,",
             "for 50 flowers from each of 3 species of iris.",
             "The species are Iris setosa, versicolor, and virginica.", sep = " ")
text.p &lt;- ggparagraph(text, face = "italic", size = 12)

# Arrange the plots on the same page
ggarrange(density.p, text.p,
         ncol = 1, nrow = 2,
         heights = c(1, 0.3))

</code></pre>

<hr>
<h2 id='ggpie'>Pie chart</h2><span id='topic+ggpie'></span>

<h3>Description</h3>

<p>Create a pie chart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggpie(
  data,
  x,
  label = x,
  lab.pos = c("out", "in"),
  lab.adjust = 0,
  lab.font = c(4, "plain", "black"),
  font.family = "",
  color = "black",
  fill = "white",
  palette = NULL,
  size = NULL,
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggpie_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggpie_+3A_x">x</code></td>
<td>
<p>variable containing values for drawing.</p>
</td></tr>
<tr><td><code id="ggpie_+3A_label">label</code></td>
<td>
<p>variable specifying the label of each slice.</p>
</td></tr>
<tr><td><code id="ggpie_+3A_lab.pos">lab.pos</code></td>
<td>
<p>character specifying the position for labels. Allowed values
are &quot;out&quot; (for outside) or &quot;in&quot; (for inside).</p>
</td></tr>
<tr><td><code id="ggpie_+3A_lab.adjust">lab.adjust</code></td>
<td>
<p>numeric value, used to adjust label position when lab.pos =
&quot;in&quot;. Increase or decrease this value to see the effect.</p>
</td></tr>
<tr><td><code id="ggpie_+3A_lab.font">lab.font</code></td>
<td>
<p>a vector of length 3 indicating respectively the size (e.g.:
14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and the
color (e.g.: &quot;red&quot;) of label font. For example <em>lab.font= c(4, &quot;bold&quot;,
&quot;red&quot;)</em>.</p>
</td></tr>
<tr><td><code id="ggpie_+3A_font.family">font.family</code></td>
<td>
<p>character vector specifying font family.</p>
</td></tr>
<tr><td><code id="ggpie_+3A_color">color</code>, <code id="ggpie_+3A_fill">fill</code></td>
<td>
<p>outline and fill colors.</p>
</td></tr>
<tr><td><code id="ggpie_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggpie_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggpie_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggpie_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to be passed to ggpar().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main, xlab,
ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes: palette
= &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title, labels
and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation =
c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>, <code><a href="#topic+ggline">ggline</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Data: Create some data
# +++++++++++++++++++++++++++++++

df &lt;- data.frame(
 group = c("Male", "Female", "Child"),
  value = c(25, 25, 50))

head(df)


# Basic pie charts
# ++++++++++++++++++++++++++++++++

ggpie(df, "value", label = "group")

# Reducing margins around the pie chart
ggpie(df, "value", label = "group") +
 theme( plot.margin = unit(c(-.75,-.75,-.75,-.75),"cm"))


# Change color
# ++++++++++++++++++++++++++++++++

# Change fill color by group
# set line color to white
# Use custom color palette
 ggpie(df, "value", label = "group",
      fill = "group", color = "white",
       palette = c("#00AFBB", "#E7B800", "#FC4E07") )


# Change label
# ++++++++++++++++++++++++++++++++

# Show group names and value as labels
labs &lt;- paste0(df$group, " (", df$value, "%)")
ggpie(df, "value", label = labs,
   fill = "group", color = "white",
   palette = c("#00AFBB", "#E7B800", "#FC4E07"))

# Change the position and font color of labels
ggpie(df, "value", label = labs,
   lab.pos = "in", lab.font = "white",
   fill = "group", color = "white",
   palette = c("#00AFBB", "#E7B800", "#FC4E07"))



</code></pre>

<hr>
<h2 id='ggpubr_args'>ggpubr General Arguments Description</h2><span id='topic+ggpubr_args'></span>

<h3>Description</h3>

<p>ggpubr General Arguments Description
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggpubr_args_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_x">x</code></td>
<td>
<p>character string containing the name of x variable.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_y">y</code></td>
<td>
<p>character vector containing one or more variables to plot</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_color">color</code></td>
<td>
<p>outline color.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_fill">fill</code></td>
<td>
<p>fill color.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_linetype">linetype</code></td>
<td>
<p>line types.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_select">select</code></td>
<td>
<p>character vector specifying which items to display.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_remove">remove</code></td>
<td>
<p>character vector specifying which items to remove from the plot.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_order">order</code></td>
<td>
<p>character vector specifying the order of items.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_add">add</code></td>
<td>
<p>character vector for adding another plot element (e.g.: dot plot or
error bars). Allowed values are one or the combination of: &quot;none&quot;,
&quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;,
&quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_mad&quot;,
&quot;median_range&quot;; see ?desc_statby for more details.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, shape, size, fill, linetype) for the
argument 'add'; e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_error.plot">error.plot</code></td>
<td>
<p>plot type used to visualize error. Allowed values are one of
c(&quot;pointrange&quot;, &quot;linerange&quot;, &quot;crossbar&quot;, &quot;errorbar&quot;, &quot;upper_errorbar&quot;,
&quot;lower_errorbar&quot;, &quot;upper_pointrange&quot;, &quot;lower_pointrange&quot;, &quot;upper_linerange&quot;,
&quot;lower_linerange&quot;). Default value is &quot;pointrange&quot; or &quot;errorbar&quot;. Used only
when add != &quot;none&quot; and add contains one &quot;mean_*&quot; or &quot;med_*&quot; where &quot;*&quot; = sd,
se, ....</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_font.label">font.label</code></td>
<td>
<p>a list which can contain the combination of the following elements: the size
(e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and
the color (e.g.: &quot;red&quot;) of labels. For example font.label = list(size = 14,
face = &quot;bold&quot;, color =&quot;red&quot;). To specify only the size and the style, use font.label =
list(size = 14, face = &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab
= FALSE to hide xlab.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggpubr_args_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
</table>

<hr>
<h2 id='ggpubr_options'>Global Options for GGPubr</h2><span id='topic+ggpubr_options'></span>

<h3>Description</h3>

<p>Displays allowed global options in ggpubr.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggpubr_options()
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>
ggpubr_options()

</code></pre>

<hr>
<h2 id='ggqqplot'>QQ Plots</h2><span id='topic+ggqqplot'></span>

<h3>Description</h3>

<p>Quantile-Quantile plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggqqplot(
  data,
  x,
  combine = FALSE,
  merge = FALSE,
  color = "black",
  palette = NULL,
  size = NULL,
  shape = NULL,
  add = c("qqline", "none"),
  add.params = list(linetype = "solid"),
  conf.int = TRUE,
  conf.int.level = 0.95,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggqqplot_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_x">x</code></td>
<td>
<p>variable to be drawn.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_color">color</code></td>
<td>
<p>point color.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_size">size</code></td>
<td>
<p>point size.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_shape">shape</code></td>
<td>
<p>point shape.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_add">add</code></td>
<td>
<p>character vector. Allowed values are one of &quot;none&quot; and &quot;qqline&quot;
(for adding qqline).</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, size,  linetype) for the
argument 'add'; e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_conf.int">conf.int</code></td>
<td>
<p>logical value. If TRUE, confidence interval is added.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_conf.int.level">conf.int.level</code></td>
<td>
<p>the confidence level. Default value is 0.95.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggqqplot_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <code><a href="#topic+ggpar">ggpar</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation
= c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create some data format
set.seed(1234)
wdata = data.frame(
   sex = factor(rep(c("F", "M"), each=200)),
   weight = c(rnorm(200, 55), rnorm(200, 58)))

head(wdata, 4)

# Basic QQ plot
ggqqplot(wdata, x = "weight")

# Change colors and shape by groups ("sex")
# Use custom palette
ggqqplot(wdata, x = "weight",
   color = "sex", palette = c("#00AFBB", "#E7B800"))

</code></pre>

<hr>
<h2 id='ggscatter'>Scatter plot</h2><span id='topic+ggscatter'></span>

<h3>Description</h3>

<p>Create a scatter plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggscatter(
  data,
  x,
  y,
  combine = FALSE,
  merge = FALSE,
  color = "black",
  fill = "lightgray",
  palette = NULL,
  shape = 19,
  size = 2,
  point = TRUE,
  rug = FALSE,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  add = c("none", "reg.line", "loess"),
  add.params = list(),
  conf.int = FALSE,
  conf.int.level = 0.95,
  fullrange = FALSE,
  ellipse = FALSE,
  ellipse.level = 0.95,
  ellipse.type = "norm",
  ellipse.alpha = 0.1,
  ellipse.border.remove = FALSE,
  mean.point = FALSE,
  mean.point.size = ifelse(is.numeric(size), 2 * size, size),
  star.plot = FALSE,
  star.plot.lty = 1,
  star.plot.lwd = NULL,
  label = NULL,
  font.label = c(12, "plain"),
  font.family = "",
  label.select = NULL,
  repel = FALSE,
  label.rectangle = FALSE,
  parse = FALSE,
  cor.coef = FALSE,
  cor.coeff.args = list(),
  cor.method = "pearson",
  cor.coef.coord = c(NULL, NULL),
  cor.coef.size = 4,
  ggp = NULL,
  show.legend.text = NA,
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggscatter_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_x">x</code></td>
<td>
<p>x variables for drawing.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_y">y</code></td>
<td>
<p>y variables for drawing.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_color">color</code>, <code id="ggscatter_+3A_fill">fill</code></td>
<td>
<p>point colors.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_shape">shape</code></td>
<td>
<p>point shape. See <code><a href="#topic+show_point_shapes">show_point_shapes</a></code>.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_point">point</code></td>
<td>
<p>logical value. If TRUE, show points.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_rug">rug</code></td>
<td>
<p>logical value. If TRUE, add marginal rug.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_add">add</code></td>
<td>
<p>allowed values are one of &quot;none&quot;, &quot;reg.line&quot; (for adding linear
regression line) or &quot;loess&quot; (for adding local regression fitting).</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, size, linetype) for the argument 'add';
e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_conf.int">conf.int</code></td>
<td>
<p>logical value. If TRUE, adds confidence interval.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_conf.int.level">conf.int.level</code></td>
<td>
<p>Level controlling confidence region. Default is 95%.
Used only when add != &quot;none&quot; and conf.int = TRUE.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_fullrange">fullrange</code></td>
<td>
<p>should the fit span the full range of the plot, or just the
data. Used only when add != &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_ellipse">ellipse</code></td>
<td>
<p>logical value. If TRUE, draws ellipses around points.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_ellipse.level">ellipse.level</code></td>
<td>
<p>the size of the concentration ellipse in normal
probability.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_ellipse.type">ellipse.type</code></td>
<td>
<p>Character specifying frame type. Possible values are
<code>"convex"</code>, <code>"confidence"</code> or types supported by
<code><a href="ggplot2.html#topic+stat_ellipse">stat_ellipse</a>()</code> including one of <code>c("t", "norm",
 "euclid")</code> for plotting concentration ellipses.
</p>
 <ul>
<li> <p><code>"convex"</code>: plot convex hull of a set o points. </p>
</li>
<li>
<p><code>"confidence"</code>: plot confidence ellipses arround group mean points as
<code>FactoMineR::coord.ellipse()</code>. </p>
</li>
<li> <p><code>"t"</code>:
assumes a multivariate t-distribution. </p>
</li>
<li> <p><code>"norm"</code>: assumes a
multivariate normal distribution. </p>
</li>
<li> <p><code>"euclid"</code>: draws a circle with
the radius equal to level, representing the euclidean distance from the
center. This ellipse probably won't appear circular unless
<code><a href="ggplot2.html#topic+coord_fixed">coord_fixed</a>()</code> is applied.</p>
</li></ul>
</td></tr>
<tr><td><code id="ggscatter_+3A_ellipse.alpha">ellipse.alpha</code></td>
<td>
<p>Alpha for ellipse specifying the transparency level of
fill color. Use alpha = 0 for no fill color.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_ellipse.border.remove">ellipse.border.remove</code></td>
<td>
<p>logical value. If TRUE, remove ellipse border lines.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_mean.point">mean.point</code></td>
<td>
<p>logical value. If TRUE, group mean points are added to the
plot.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_mean.point.size">mean.point.size</code></td>
<td>
<p>numeric value specifying the size of mean points.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_star.plot">star.plot</code></td>
<td>
<p>logical value. If TRUE, a star plot is generated.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_star.plot.lty">star.plot.lty</code>, <code id="ggscatter_+3A_star.plot.lwd">star.plot.lwd</code></td>
<td>
<p>line type and line width (size) for star
plot, respectively.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_label">label</code></td>
<td>
<p>the name of the column containing point labels. Can be also a
character vector with length = nrow(data).</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_font.label">font.label</code></td>
<td>
<p>a vector of length 3 indicating respectively the size
(e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and
the color (e.g.: &quot;red&quot;) of point labels. For example <em>font.label =
c(14, &quot;bold&quot;, &quot;red&quot;)</em>. To specify only the size and the style, use
font.label = c(14, &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_font.family">font.family</code></td>
<td>
<p>character vector specifying font family.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_label.select">label.select</code></td>
<td>
<p>character vector specifying some labels to show.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_repel">repel</code></td>
<td>
<p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_parse">parse</code></td>
<td>
<p>If <code>TRUE</code>, the labels will be parsed into expressions and
displayed as described in <code>?plotmath</code>.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_cor.coef">cor.coef</code></td>
<td>
<p>logical value. If TRUE, correlation coefficient with the
p-value will be added to the plot.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_cor.coeff.args">cor.coeff.args</code></td>
<td>
<p>a list of arguments to pass to the function
<code><a href="#topic+stat_cor">stat_cor</a></code> for customizing the displayed correlation
coefficients. For example: <code>cor.coeff.args = list(method = "pearson",
label.x.npc = "right", label.y.npc = "top")</code>.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_cor.method">cor.method</code></td>
<td>
<p>method for computing correlation coefficient. Allowed
values are one of &quot;pearson&quot;, &quot;kendall&quot;, or &quot;spearman&quot;.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_cor.coef.coord">cor.coef.coord</code></td>
<td>
<p>numeric vector, of length 2, specifying the x and y
coordinates of the correlation coefficient. Default values are NULL.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_cor.coef.size">cor.coef.size</code></td>
<td>
<p>correlation coefficient text font size.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_ggp">ggp</code></td>
<td>
<p>a ggplot. If not NULL, points are added to an existing plot.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_show.legend.text">show.legend.text</code></td>
<td>
<p>logical. Should text be included in the legends? NA,
the default, includes if any aesthetics are mapped. FALSE never includes,
and TRUE always includes.</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggscatter_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>
and <code><a href="#topic+ggpar">ggpar</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation
= c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+stat_cor">stat_cor</a></code>, <code><a href="#topic+stat_stars">stat_stars</a></code>, <code><a href="#topic+stat_conf_ellipse">stat_conf_ellipse</a></code> and <code><a href="#topic+ggpar">ggpar</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("mtcars")
df &lt;- mtcars
df$cyl &lt;- as.factor(df$cyl)
head(df[, c("wt", "mpg", "cyl")], 3)

# Basic plot
# +++++++++++++++++++++++++++
ggscatter(df, x = "wt", y = "mpg",
   color = "black", shape = 21, size = 3, # Points color, shape and size
   add = "reg.line",  # Add regressin line
   add.params = list(color = "blue", fill = "lightgray"), # Customize reg. line
   conf.int = TRUE, # Add confidence interval
   cor.coef = TRUE, # Add correlation coefficient. see ?stat_cor
   cor.coeff.args = list(method = "pearson", label.x = 3, label.sep = "\n")
   )

# loess method: local regression fitting
ggscatter(df, x = "wt", y = "mpg",
   add = "loess", conf.int = TRUE)


# Control point size by continuous variable values ("qsec")
ggscatter(df, x = "wt", y = "mpg",
   color = "#00AFBB", size = "qsec")


# Change colors
# +++++++++++++++++++++++++++
# Use custom color palette
# Add marginal rug
ggscatter(df, x = "wt", y = "mpg", color = "cyl",
   palette = c("#00AFBB", "#E7B800", "#FC4E07") )




# Add group ellipses and mean points
# Add stars
# +++++++++++++++++++
ggscatter(df, x = "wt", y = "mpg",
   color = "cyl", shape = "cyl",
   palette = c("#00AFBB", "#E7B800", "#FC4E07"),
   ellipse = TRUE, mean.point = TRUE,
   star.plot = TRUE)


# Textual annotation
# +++++++++++++++++
df$name &lt;- rownames(df)
ggscatter(df, x = "wt", y = "mpg",
   color = "cyl", palette = c("#00AFBB", "#E7B800", "#FC4E07"),
   label = "name", repel = TRUE)


</code></pre>

<hr>
<h2 id='ggscatterhist'>Scatter Plot with Marginal Histograms</h2><span id='topic+ggscatterhist'></span><span id='topic+print.ggscatterhist'></span>

<h3>Description</h3>

<p>Create a scatter plot with marginal histograms, density plots or
box plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggscatterhist(
  data,
  x,
  y,
  group = NULL,
  color = "black",
  fill = NA,
  palette = NULL,
  shape = 19,
  size = 2,
  linetype = "solid",
  bins = 30,
  margin.plot = c("density", "histogram", "boxplot"),
  margin.params = list(),
  margin.ggtheme = theme_void(),
  margin.space = FALSE,
  main.plot.size = 2,
  margin.plot.size = 1,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  legend = "top",
  ggtheme = theme_pubr(),
  print = TRUE,
  ...
)

## S3 method for class 'ggscatterhist'
print(
  x,
  margin.space = FALSE,
  main.plot.size = 2,
  margin.plot.size = 1,
  title = NULL,
  legend = "top",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggscatterhist_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_x">x</code></td>
<td>
<p>an object of class <code>ggscatterhist</code>.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_y">y</code></td>
<td>
<p>y variables for drawing.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_group">group</code></td>
<td>
<p>a grouping variable. Change points color and shape by groups if
the options <code>color</code> and <code>shape</code> are missing. Should be also
specified when you want to create a marginal box plot that is grouped.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_color">color</code>, <code id="ggscatterhist_+3A_fill">fill</code></td>
<td>
<p>point colors.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_shape">shape</code></td>
<td>
<p>point shape. See <code><a href="#topic+show_point_shapes">show_point_shapes</a></code>.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_linetype">linetype</code></td>
<td>
<p>line type (&quot;solid&quot;, &quot;dashed&quot;, ...)</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_bins">bins</code></td>
<td>
<p>Number of histogram bins. Defaults to 30. Pick a better value that
fit to your data.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_margin.plot">margin.plot</code></td>
<td>
<p>the type of the marginal plot. Default is &quot;hist&quot;.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_margin.params">margin.params</code></td>
<td>
<p>parameters to be applied to the marginal plots.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_margin.ggtheme">margin.ggtheme</code></td>
<td>
<p>the theme of the marginal plot. Default is
<code><a href="ggplot2.html#topic+ggtheme">theme_void</a>()</code>.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_margin.space">margin.space</code></td>
<td>
<p>logical value. If TRUE, adds space between the main plot
and the marginal plot.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_main.plot.size">main.plot.size</code></td>
<td>
<p>the width of the main plot. Default is 2.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_margin.plot.size">margin.plot.size</code></td>
<td>
<p>the width of the marginal plot. Default is 1.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_legend">legend</code></td>
<td>
<p>specify the legend position. Allowed values include: &quot;top&quot;,
&quot;bottom&quot;, &quot;left&quot;, &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_ggtheme">ggtheme</code></td>
<td>
<p>the theme to be used for the scatter plot. Default is
<code><a href="#topic+theme_pubr">theme_pubr</a>()</code>.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_print">print</code></td>
<td>
<p>logical value. If <code>TRUE</code> (default), print the plot.</p>
</td></tr>
<tr><td><code id="ggscatterhist_+3A_...">...</code></td>
<td>
<p>other arguments passed to the function <code><a href="#topic+ggscatter">ggscatter</a>()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code>ggscatterhist</code>, which is list of ggplots,
including the following elements: </p>
<ul>
<li><p> sp: main scatter plot;
</p>
</li>
<li><p> xplot: marginal x-axis plot; </p>
</li>
<li><p> yplot: marginal y-axis plot. </p>
</li></ul>
<p>.
</p>
<p>User can modify each of plot before printing.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Basic scatter plot with marginal density plot
ggscatterhist(iris, x = "Sepal.Length", y = "Sepal.Width",
              color = "#00AFBB",
              margin.params = list(fill = "lightgray"))


# Grouped data
ggscatterhist(
 iris, x = "Sepal.Length", y = "Sepal.Width",
 color = "Species", size = 3, alpha = 0.6,
 palette = c("#00AFBB", "#E7B800", "#FC4E07"),
 margin.params = list(fill = "Species", color = "black", size = 0.2)
)

# Use boxplot as marginal
ggscatterhist(
 iris, x = "Sepal.Length", y = "Sepal.Width",
 color = "Species", size = 3, alpha = 0.6,
 palette = c("#00AFBB", "#E7B800", "#FC4E07"),
 margin.plot = "boxplot",
 ggtheme = theme_bw()
)

# Add vertical and horizontal line to a ggscatterhist
plots &lt;- ggscatterhist(iris, x = "Sepal.Length", y = "Sepal.Width", print = FALSE)
plots$sp &lt;- plots$sp +
 geom_hline(yintercept = 3, linetype = "dashed", color = "blue") +
 geom_vline(xintercept = 6, linetype = "dashed", color = "red")
plots

</code></pre>

<hr>
<h2 id='ggstripchart'>Stripcharts</h2><span id='topic+ggstripchart'></span>

<h3>Description</h3>

<p>Create a stripchart, also known as one dimensional scatter
plots. These plots are suitable compared to box plots when sample sizes are
small.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggstripchart(
  data,
  x,
  y,
  combine = FALSE,
  merge = FALSE,
  color = "black",
  fill = "white",
  palette = NULL,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  shape = 19,
  size = NULL,
  select = NULL,
  remove = NULL,
  order = NULL,
  add = "mean_se",
  add.params = list(),
  error.plot = "pointrange",
  label = NULL,
  font.label = list(size = 11, color = "black"),
  label.select = NULL,
  repel = FALSE,
  label.rectangle = FALSE,
  jitter = 0.2,
  position = position_jitter(jitter, seed = 123),
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggstripchart_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_x">x</code></td>
<td>
<p>character string containing the name of x variable.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_y">y</code></td>
<td>
<p>character vector containing one or more variables to plot</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_color">color</code></td>
<td>
<p>outline color.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_fill">fill</code></td>
<td>
<p>fill color.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_shape">shape</code></td>
<td>
<p>point shape</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_select">select</code></td>
<td>
<p>character vector specifying which items to display.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_remove">remove</code></td>
<td>
<p>character vector specifying which items to remove from the plot.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_order">order</code></td>
<td>
<p>character vector specifying the order of items.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_add">add</code></td>
<td>
<p>character vector for adding another plot element (e.g.: dot plot or
error bars). Allowed values are one or the combination of: &quot;none&quot;,
&quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;,
&quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;,
&quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;; see ?desc_statby for more
details.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, shape, size, fill, linetype) for the
argument 'add'; e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_error.plot">error.plot</code></td>
<td>
<p>plot type used to visualize error. Allowed values are one of
c(&quot;pointrange&quot;, &quot;linerange&quot;, &quot;crossbar&quot;, &quot;errorbar&quot;, &quot;upper_errorbar&quot;,
&quot;lower_errorbar&quot;, &quot;upper_pointrange&quot;, &quot;lower_pointrange&quot;, &quot;upper_linerange&quot;,
&quot;lower_linerange&quot;). Default value is &quot;pointrange&quot; or &quot;errorbar&quot;. Used only
when add != &quot;none&quot; and add contains one &quot;mean_*&quot; or &quot;med_*&quot; where &quot;*&quot; = sd,
se, ....</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_label">label</code></td>
<td>
<p>the name of the column containing point labels. Can be also a
character vector with length = nrow(data).</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_font.label">font.label</code></td>
<td>
<p>a list which can contain the combination of the following
elements: the size (e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;,
&quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of labels. For example font.label
= list(size = 14, face = &quot;bold&quot;, color =&quot;red&quot;). To specify only the size and
the style, use font.label = list(size = 14, face = &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_label.select">label.select</code></td>
<td>
<p>can be of two formats: </p>
 <ul>
<li><p> a character vector
specifying some labels to show. </p>
</li>
<li><p> a list containing one or the
combination of the following components: </p>
 <ul>
<li> <p><code>top.up</code> and
<code>top.down</code>: to display the labels  of the top up/down points. For
example, <code>label.select = list(top.up = 10, top.down = 4)</code>. </p>
</li>
<li>
<p><code>criteria</code>: to filter, for example, by x and y variabes values, use
this: <code>label.select = list(criteria = "`y` &gt; 2 &amp; `y` &lt; 5 &amp; `x` %in%
c('A', 'B')")</code>. </p>
</li></ul>
 </li></ul>
</td></tr>
<tr><td><code id="ggstripchart_+3A_repel">repel</code></td>
<td>
<p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_jitter">jitter</code></td>
<td>
<p>the amount of jitter.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_position">position</code></td>
<td>
<p>position adjustment, either as a string, or the result of a
call to a position adjustment function. Used to adjust position for
multiple groups.</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggstripchart_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to
<code><a href="ggplot2.html#topic+geom_jitter">geom_jitter</a></code>, <code><a href="#topic+ggpar">ggpar</a></code> and
<code><a href="#topic+facet">facet</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation
= c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>, <code><a href="#topic+ggviolin">ggviolin</a></code>, <code><a href="#topic+ggdotplot">ggdotplot</a></code>
and <code><a href="#topic+ggboxplot">ggboxplot</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")
df &lt;- ToothGrowth

# Basic plot with summary statistics: mean_se
# +++++++++++++++++++++++++++
# Change point shapes by groups: "dose"
ggstripchart(df, x = "dose", y = "len",
   shape = "dose", size = 3,
   add = "mean_se")

# Use mean_sd
# Change error.plot to "crossbar"
ggstripchart(df, x = "dose", y = "len",
   shape = "dose", size = 3,
   add = "mean_sd", add.params = list(width = 0.5),
   error.plot = "crossbar")



# Add summary statistics
# ++++++++++++++++++++++++++

# Add box plot
ggstripchart(df, x = "dose", y = "len",
 shape = "dose", add = "boxplot")

# Add violin + mean_sd
ggstripchart(df, x = "dose", y = "len",
 shape = "dose", add = c("violin", "mean_sd"))


# Change colors
# +++++++++++++++++++++++++++
# Change colors by groups: dose
# Use custom color palette
 ggstripchart(df, "dose", "len",  shape = "dose",
   color = "dose", palette = c("#00AFBB", "#E7B800", "#FC4E07"),
   add = "mean_sd")



# Plot with multiple groups
# +++++++++++++++++++++
# Change shape and color by a second group : "supp"
ggstripchart(df, "dose", "len", shape = "supp",
  color = "supp", palette = c("#00AFBB", "#E7B800"))

# Adjust point position
ggstripchart(df, "dose", "len", shape = "supp",
  color = "supp", palette = c("#00AFBB", "#E7B800"),
  position = position_dodge(0.8) )

# You can also use position_jitterdodge()
# but fill aesthetic is required
ggstripchart(df, "dose", "len",  shape = "supp",
   color = "supp", palette = c("#00AFBB", "#E7B800"),
   position = position_jitterdodge() )

# Add boxplot
ggstripchart(df, "dose", "len", shape = "supp",
 color = "supp", palette = c("#00AFBB", "#E7B800"),
 add = "boxplot", add.params = list(color = "black") )

</code></pre>

<hr>
<h2 id='ggsummarytable'>GGPLOT with Summary Stats Table Under the Plot</h2><span id='topic+ggsummarytable'></span><span id='topic+ggsummarystats'></span><span id='topic+print.ggsummarystats'></span><span id='topic+print.ggsummarystats_list'></span>

<h3>Description</h3>

<p>Create a ggplot with summary stats (n, median, mean, iqr) table
under the plot. Read more: <a href="https://www.datanovia.com/en/blog/how-to-create-a-beautiful-plots-in-r-with-summary-statistics-labels/">How to Create a Beautiful Plots in R with Summary Statistics Labels</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggsummarytable(
  data,
  x,
  y,
  digits = 0,
  size = 3,
  color = "black",
  palette = NULL,
  facet.by = NULL,
  labeller = "label_value",
  position = "identity",
  ggtheme = theme_pubr(),
  ...
)

ggsummarystats(
  data,
  x,
  y,
  summaries = c("n", "median", "iqr"),
  ggfunc = ggboxplot,
  color = "black",
  fill = "white",
  palette = NULL,
  facet.by = NULL,
  free.panels = FALSE,
  labeller = "label_value",
  heights = c(0.8, 0.2),
  digits = 0,
  table.font.size = 3,
  ggtheme = theme_pubr(),
  ...
)

## S3 method for class 'ggsummarystats'
print(x, heights = c(0.8, 0.2), ...)

## S3 method for class 'ggsummarystats_list'
print(x, heights = c(0.8, 0.2), legend = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggsummarytable_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_x">x</code></td>
<td>
<p>a list of <code>ggsummarystats</code>.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_y">y</code></td>
<td>
<p>character vector containing one or more variables to plot</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_digits">digits</code></td>
<td>
<p>integer indicating the number of decimal places (round) to be
used.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_color">color</code></td>
<td>
<p>outline color.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_labeller">labeller</code></td>
<td>
<p>Character vector. An alternative to the argument
<code>short.panel.labs</code>. Possible values are one of &quot;label_both&quot; (panel
labelled by both grouping variable names and levels) and &quot;label_value&quot;
(panel labelled with only grouping levels).</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string, or the result of a
call to a position adjustment function.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_...">...</code></td>
<td>
<p>other arguments passed to the function <code><a href="#topic+ggpar">ggpar</a>()</code>,
<code><a href="#topic+facet">facet</a>()</code> or <code><a href="#topic+ggarrange">ggarrange</a>()</code> when printing the plot.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_summaries">summaries</code></td>
<td>
<p>summary stats to display in the table. Possible values are
those returned by the function <code><a href="rstatix.html#topic+get_summary_stats">get_summary_stats</a>()</code>,
including: <code>"n", "min", "max",  "median",  "q1", "q2", "q3", "mad",
"mean", "sd", "se", "ci"</code>.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_ggfunc">ggfunc</code></td>
<td>
<p>a ggpubr function, including: ggboxplot, ggviolin, ggdotplot,
ggbarplot, ggline, etc. Can be any other ggplot function that accepts the
following arguments <code>data, x, color, fill, palette, ggtheme,
facet.by</code>.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_fill">fill</code></td>
<td>
<p>fill color.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_free.panels">free.panels</code></td>
<td>
<p>logical. If TRUE, create free plot panels when the
argument <code>facet.by</code> is specified.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_heights">heights</code></td>
<td>
<p>a numeric vector of length 2, specifying the heights of the
main and the summary table, respectively.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_table.font.size">table.font.size</code></td>
<td>
<p>the summary table font size.</p>
</td></tr>
<tr><td><code id="ggsummarytable_+3A_legend">legend</code></td>
<td>
<p>character specifying legend position. Allowed values are one of
c(&quot;top&quot;, &quot;bottom&quot;, &quot;left&quot;, &quot;right&quot;, &quot;none&quot;). To remove the legend use
legend = &quot;none&quot;.</p>
</td></tr>
</table>


<h3>Functions</h3>


<ul>
<li> <p><code>ggsummarytable()</code>: Create a table of summary stats
</p>
</li>
<li> <p><code>ggsummarystats()</code>: Create a ggplot with a summary stat table under the plot.
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'># Data preparation
#::::::::::::::::::::::::::::::::::::::::::::::::
data("ToothGrowth")
df &lt;- ToothGrowth
df$dose &lt;- as.factor(df$dose)
# Add random QC column
set.seed(123)
qc &lt;- rep(c("pass", "fail"), 30)
df$qc &lt;- as.factor(sample(qc, 60))
# Inspect the data
head(df)


# Basic summary stats
#::::::::::::::::::::::::::::::::::::::::::::::::
# Compute summary statistics
summary.stats &lt;- df %&gt;%
  group_by(dose) %&gt;%
  get_summary_stats(type = "common")
summary.stats

# Visualize summary table
ggsummarytable(
  summary.stats, x = "dose", y = c("n", "median", "iqr"),
  ggtheme = theme_bw()
)


# Create plots with summary table under the plot
#::::::::::::::::::::::::::::::::::::::::::::::::
# Basic plot
ggsummarystats(
  df, x = "dose", y = "len",
  ggfunc = ggboxplot, add = "jitter"
)

# Color by groups
ggsummarystats(
  df, x = "dose", y = "len",
  ggfunc = ggboxplot, add = "jitter",
  color = "dose", palette = "npg"
)

# Create a barplot
ggsummarystats(
  df, x = "dose", y = "len",
  ggfunc = ggbarplot, add = c("jitter", "median_iqr"),
  color = "dose", palette = "npg"
)

# Facet
#::::::::::::::::::::::::::::::::::::::::::::::::
# Specify free.panels = TRUE for free panels
ggsummarystats(
  df, x = "dose", y = "len",
  ggfunc = ggboxplot, add = "jitter",
  color = "dose", palette = "npg",
  facet.by = c("supp", "qc"),
  labeller = "label_both"
)

</code></pre>

<hr>
<h2 id='ggtext'>Text</h2><span id='topic+ggtext'></span>

<h3>Description</h3>

<p>Add text to a plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggtext(
  data,
  x = NULL,
  y = NULL,
  label = NULL,
  color = "black",
  palette = NULL,
  size = 11,
  face = "plain",
  family = "",
  show.legend = NA,
  label.select = NULL,
  repel = FALSE,
  label.rectangle = FALSE,
  parse = FALSE,
  grouping.vars = NULL,
  position = "identity",
  ggp = NULL,
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggtext_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggtext_+3A_x">x</code>, <code id="ggtext_+3A_y">y</code></td>
<td>
<p>x and y variables for drawing.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_label">label</code></td>
<td>
<p>the name of the column containing point labels. Can be also a
character vector with length = nrow(data).</p>
</td></tr>
<tr><td><code id="ggtext_+3A_color">color</code></td>
<td>
<p>text font color.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_size">size</code></td>
<td>
<p>text font size.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_face">face</code></td>
<td>
<p>text font style. Allowed values are one of c(&quot;plain&quot;, &quot;bold&quot;,
&quot;italic&quot;, &quot;bold.italic&quot;).</p>
</td></tr>
<tr><td><code id="ggtext_+3A_family">family</code></td>
<td>
<p>character vector specifying font family.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should text be included in the legends? NA, the
default, includes if any aesthetics are mapped. FALSE never includes, and
TRUE always includes.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_label.select">label.select</code></td>
<td>
<p>can be of two formats: </p>
 <ul>
<li><p> a character vector
specifying some labels to show. </p>
</li>
<li><p> a list containing one or the
combination of the following components: </p>
 <ul>
<li> <p><code>top.up</code> and
<code>top.down</code>: to display the labels  of the top up/down points. For
example, <code>label.select = list(top.up = 10, top.down = 4)</code>. </p>
</li>
<li>
<p><code>criteria</code>: to filter, for example, by x and y variabes values, use
this: <code>label.select = list(criteria = "`y` &gt; 2 &amp; `y` &lt; 5 &amp; `x` %in%
c('A', 'B')")</code>. </p>
</li></ul>
 </li></ul>
</td></tr>
<tr><td><code id="ggtext_+3A_repel">repel</code></td>
<td>
<p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_parse">parse</code></td>
<td>
<p>If <code>TRUE</code>, the labels will be parsed into expressions and
displayed as described in <code>?plotmath</code>.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_grouping.vars">grouping.vars</code></td>
<td>
<p>grouping variables to sort the data by, when the user
wants to display the top n up/down labels.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string, or the result of a
call to a position adjustment function.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_ggp">ggp</code></td>
<td>
<p>a ggplot. If not NULL, points are added to an existing plot.</p>
</td></tr>
<tr><td><code id="ggtext_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggtext_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <code><a href="#topic+ggpar">ggpar</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main,
xlab, ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes:
palette = &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title,
labels and position: legend = &quot;right&quot;  </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("mtcars")
df &lt;- mtcars
df$cyl &lt;- as.factor(df$cyl)
df$name &lt;- rownames(df)
head(df[, c("wt", "mpg", "cyl")], 3)

# Textual annotation
# +++++++++++++++++
ggtext(df, x = "wt", y = "mpg",
   color = "cyl", palette = c("#00AFBB", "#E7B800", "#FC4E07"),
   label = "name", repel = TRUE)

# Add rectangle around label
ggtext(df, x = "wt", y = "mpg",
   color = "cyl", palette = c("#00AFBB", "#E7B800", "#FC4E07"),
   label = "name", repel = TRUE,  label.rectangle = TRUE)


</code></pre>

<hr>
<h2 id='ggtexttable'>Draw a Textual Table</h2><span id='topic+ggtexttable'></span><span id='topic+ttheme'></span><span id='topic+colnames_style'></span><span id='topic+rownames_style'></span><span id='topic+tbody_style'></span><span id='topic+table_cell_font'></span><span id='topic+table_cell_bg'></span><span id='topic+tab_cell_crossout'></span><span id='topic+tab_ncol'></span><span id='topic+tab_nrow'></span><span id='topic+tab_add_hline'></span><span id='topic+tab_add_vline'></span><span id='topic+tab_add_border'></span><span id='topic+tbody_add_border'></span><span id='topic+thead_add_border'></span><span id='topic+tab_add_title'></span><span id='topic+tab_add_footnote'></span>

<h3>Description</h3>

<p>Draw a textual table. </p>
 <ul>
<li> <p><code>ggtexttable()</code>: draw
a textual table. </p>
</li>
<li> <p><code>ttheme()</code>: customize table theme. </p>
</li>
<li>
<p><code>rownames_style(), colnames_style(), tbody_style()</code>: helper functions
to customize the table row names, column names and body.
</p>
</li>
<li> <p><code>table_cell_font()</code>: access to a table cell for changing the text font (size and face).
</p>
</li>
<li> <p><code>table_cell_bg()</code>: access to a table cell for changing the background (fill, color, linewidth).
</p>
</li>
<li> <p><code>tab_cell_crossout()</code>: cross out a table cell.
</p>
</li>
<li> <p><code>tab_ncol(), tab_nrow()</code>: returns, respectively, the number of columns and rows in a ggtexttable.
</p>
</li>
<li> <p><code>tab_add_hline()</code>: Creates horizontal lines or separators at the top or the bottom side of a given specified row.
</p>
</li>
<li> <p><code>tab_add_vline()</code>: Creates vertical lines or separators at the right or the left side of a given specified column.
</p>
</li>
<li> <p><code>tab_add_border(), tbody_add_border(), thead_add_border()</code>: Add borders to table; tbody is for table body and thead is for table head.
</p>
</li>
<li> <p><code>tab_add_title(),tab_add_footnote()</code>: Add title, subtitle and footnote to a table.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>ggtexttable(
  x,
  rows = rownames(x),
  cols = colnames(x),
  vp = NULL,
  theme = ttheme(),
  ...
)

ttheme(
  base_style = "default",
  base_size = 11,
  base_colour = "black",
  padding = unit(c(4, 4), "mm"),
  colnames.style = colnames_style(size = base_size),
  rownames.style = rownames_style(size = base_size),
  tbody.style = tbody_style(size = base_size)
)

colnames_style(
  color = "black",
  face = "bold",
  size = 12,
  fill = "grey80",
  linewidth = 1,
  linecolor = "white",
  parse = FALSE,
  ...
)

rownames_style(
  color = "black",
  face = "italic",
  size = 12,
  fill = NA,
  linewidth = 1,
  linecolor = "white",
  parse = FALSE,
  ...
)

tbody_style(
  color = "black",
  face = "plain",
  size = 12,
  fill = c("grey95", "grey90"),
  linewidth = 1,
  linecolor = "white",
  parse = FALSE,
  ...
)

table_cell_font(tab, row, column, face = NULL, size = NULL, color = NULL)

table_cell_bg(
  tab,
  row,
  column,
  fill = NULL,
  color = NULL,
  linewidth = NULL,
  alpha = NULL
)

tab_cell_crossout(
  tab,
  row,
  column,
  linetype = 1,
  linewidth = 1,
  linecolor = "black",
  reduce.size.by = 0
)

tab_ncol(tab)

tab_nrow(tab)

tab_add_hline(
  tab,
  at.row = 2:tab_nrow(tab),
  row.side = c("bottom", "top"),
  from.column = 1,
  to.column = tab_ncol(tab),
  linetype = 1,
  linewidth = 1,
  linecolor = "black"
)

tab_add_vline(
  tab,
  at.column = 2:tab_ncol(tab),
  column.side = c("left", "right"),
  from.row = 1,
  to.row = tab_nrow(tab),
  linetype = 1,
  linewidth = 1,
  linecolor = "black"
)

tab_add_border(
  tab,
  from.row = 2,
  to.row = tab_nrow(tab),
  from.column = 1,
  to.column = tab_ncol(tab),
  linetype = 1,
  linewidth = 1,
  linecolor = "black"
)

tbody_add_border(
  tab,
  from.row = 2,
  to.row = tab_nrow(tab),
  from.column = 1,
  to.column = tab_ncol(tab),
  linetype = 1,
  linewidth = 1,
  linecolor = "black"
)

thead_add_border(
  tab,
  from.row = 1,
  to.row = 1,
  from.column = 1,
  to.column = tab_ncol(tab),
  linetype = 1,
  linewidth = 1,
  linecolor = "black"
)

tab_add_title(
  tab,
  text,
  face = NULL,
  size = NULL,
  color = NULL,
  family = NULL,
  padding = unit(1.5, "line"),
  just = "left",
  hjust = NULL,
  vjust = NULL
)

tab_add_footnote(
  tab,
  text,
  face = NULL,
  size = NULL,
  color = NULL,
  family = NULL,
  padding = unit(1.5, "line"),
  just = "right",
  hjust = NULL,
  vjust = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggtexttable_+3A_x">x</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_rows">rows</code></td>
<td>
<p>optional vector to specify row names</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_cols">cols</code></td>
<td>
<p>optional vector to specify column names</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_vp">vp</code></td>
<td>
<p>optional viewport</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_theme">theme</code></td>
<td>
<p>a list, as returned by the function <code>ttheme()</code>, defining the
parameters of the table theme. Allowed values include one of <code>ttheme()</code>
and <code>ttheme_clean()</code>.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_...">...</code></td>
<td>
<p>extra parameters for text justification, e.g.: hjust and x. Default
is &quot;centre&quot; for the body and header, and &quot;right&quot; for the row names. Left
justification: <code>hjust = 0, x = 0.1</code>. Right justification: <code>hjust = 1, x = 0.9</code>.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_base_style">base_style</code></td>
<td>
<p>character string the table style/theme. The available themes
are illustrated in the
<a href="https://rpkgs.datanovia.com/ggpubr/files/ggtexttable-theme.pdf">ggtexttable-theme.pdf</a>
file. Allowed values include one of <code>c("default", "blank", "classic",
"minimal", "light", "lBlack", "lBlue", "lRed", "lGreen", "lViolet", "lCyan",
"lOrange", "lBlackWhite", "lBlueWhite", "lRedWhite", "lGreenWhite",
"lVioletWhite", "lCyanWhite", "lOrangeWhite", "mBlack", "mBlue", "mRed",
"mGreen", "mViolet", "mCyan", "mOrange", "mBlackWhite", "mBlueWhite",
"mRedWhite", "mGreenWhite", "mVioletWhite", "mCyanWhite", "mOrangeWhite" )</code>.
Note that, l = &quot;light&quot;; m = &quot;medium&quot;.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_base_size">base_size</code></td>
<td>
<p>default font size</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_base_colour">base_colour</code></td>
<td>
<p>default font colour</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_padding">padding</code></td>
<td>
<p>length-2 unit vector specifying the horizontal and vertical padding of text within each cell</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_colnames.style">colnames.style</code></td>
<td>
<p>a list, as returned by the function
<code>colnames_style()</code>, defining the style of the table column names.
Considered only when <code>base_size = "default"</code>.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_rownames.style">rownames.style</code></td>
<td>
<p>a list, as returned by the function
<code>rownames_style()</code>, defining the style of the table row names.
Considered only when <code>base_size = "default"</code>.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_tbody.style">tbody.style</code></td>
<td>
<p>a list, as returned by the function <code>tbody_style()</code>,
defining the style of the table body. Considered only when <code>base_size =
"default"</code>.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_color">color</code>, <code id="ggtexttable_+3A_face">face</code>, <code id="ggtexttable_+3A_size">size</code></td>
<td>
<p>text font color, face and size, respectively. Allowed
values for face include c(&quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;).</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_fill">fill</code></td>
<td>
<p>background color.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_linewidth">linewidth</code>, <code id="ggtexttable_+3A_linecolor">linecolor</code></td>
<td>
<p>line width and color, respectively.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_parse">parse</code></td>
<td>
<p>logical, default behaviour for parsing text as plotmath</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_tab">tab</code></td>
<td>
<p>an object from <code>ggtexttable</code> or from <code>gridExtra::tableGrob()</code>.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_row">row</code>, <code id="ggtexttable_+3A_column">column</code></td>
<td>
<p>an integer specifying the row and the column numbers for the cell of interest.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_alpha">alpha</code></td>
<td>
<p>numeric value specifying fill color transparency.
Value should be in [0, 1], where 0 is full transparency and 1 is no transparency.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_linetype">linetype</code></td>
<td>
<p>line type</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_reduce.size.by">reduce.size.by</code></td>
<td>
<p>Numeric value in [0, 1] to reduce the size by.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_at.row">at.row</code></td>
<td>
<p>a numeric vector of row indexes; for example <code>at.row = c(1, 2)</code>.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_row.side">row.side</code></td>
<td>
<p>row side to which the horinzotal line should be added. Can be one of <code>c("bottom", "top")</code>.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_from.column">from.column</code></td>
<td>
<p>integer indicating the column from which to start drawing the horizontal line.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_to.column">to.column</code></td>
<td>
<p>integer indicating the column to which the horizontal line should end.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_at.column">at.column</code></td>
<td>
<p>a numeric vector of column indexes; for example <code>at.column = c(1, 2)</code>.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_column.side">column.side</code></td>
<td>
<p>column side to which the vertical line should be added. Can be one of <code>c("left", "right")</code>.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_from.row">from.row</code></td>
<td>
<p>integer indicating the row from which to start drawing the horizontal line.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_to.row">to.row</code></td>
<td>
<p>integer indicating the row to which the vertical line should end.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_text">text</code></td>
<td>
<p>text to be added as title or footnote.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_family">family</code></td>
<td>
<p>font family</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_just">just</code></td>
<td>
<p>The justification of the text relative to its (x, y) location. If
there are two values, the first value specifies horizontal justification
and the second value specifies vertical justification. Possible string
values are: &quot;left&quot;, &quot;right&quot;, &quot;centre&quot;, &quot;center&quot;, &quot;bottom&quot;, and &quot;top&quot;. For
numeric values, 0 means left (bottom) alignment and 1 means right (top)
alignment.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_hjust">hjust</code></td>
<td>
<p>A numeric vector specifying horizontal justification. If
specified, overrides the just setting.</p>
</td></tr>
<tr><td><code id="ggtexttable_+3A_vjust">vjust</code></td>
<td>
<p>A numeric vector specifying vertical justification. If
specified, overrides the just setting.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class ggplot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># data
df &lt;- head(iris)

# Default table
# Remove row names using rows = NULL
ggtexttable(df, rows = NULL)

# Text justification for individual cells/rows/columns (#335)
# First column is left justified i.e., hjust = 0 , x = 0.1
# Remaining columns are right justified i.e., hjust = 1 , x = 0.9
table_theme &lt;- ttheme(
  tbody.style = tbody_style(
   hjust = as.vector(matrix(c(0, 1, 1, 1, 1), ncol = 5, nrow = nrow(df), byrow = TRUE)),
   x = as.vector(matrix(c(.1, .9, .9,.9, .9), ncol = 5, nrow = nrow(df), byrow = TRUE))
 )
)
ggtexttable(df, rows = NULL, theme = table_theme)

# Blank theme
ggtexttable(df, rows = NULL, theme = ttheme("blank"))

# light theme
ggtexttable(df, rows = NULL, theme = ttheme("light"))

# Column names border only
ggtexttable(df, rows = NULL, theme = ttheme("blank")) %&gt;%
 tab_add_hline(at.row = 1:2, row.side = "top", linewidth = 2)

# classic theme
ggtexttable(df, rows = NULL, theme = ttheme("classic"))

# minimal theme
ggtexttable(df, rows = NULL, theme = ttheme("minimal"))

# Medium blue (mBlue) theme
ggtexttable(df, rows = NULL, theme = ttheme("mBlue"))


# Customize the table as you want
ggtexttable(df, rows = NULL,
           theme = ttheme(
             colnames.style = colnames_style(color = "white", fill = "#8cc257"),
             tbody.style = tbody_style(color = "black", fill = c("#e8f3de", "#d3e8bb"))
           )
)

# Use RColorBrewer palette
# Provide as many fill color as there are rows in the table body, here nrow = 6
ggtexttable(df,
           theme = ttheme(
             colnames.style = colnames_style(fill = "white"),
             tbody.style = tbody_style(fill = get_palette("RdBu", 6))
           )
)

# Text justification
#::::::::::::::::::::::::::::::::::::::::::::::
# Default is "centre" for the body and header, and "right" for the row names.
# Left justification: hjust=0, x=0.1
# Right justification: hjust=1, x=0.9
tbody.style = tbody_style(color = "black",
   fill = c("#e8f3de", "#d3e8bb"), hjust=1, x=0.9)
ggtexttable(head(iris), rows = NULL,
           theme = ttheme(
             colnames.style = colnames_style(color = "white", fill = "#8cc257"),
             tbody.style = tbody.style
           )
)

# Access and modify the font and
# the background of table cells
# :::::::::::::::::::::::::::::::::::::::::::::
tab &lt;- ggtexttable(head(iris), rows = NULL,
                  theme = ttheme("classic"))
tab &lt;- table_cell_font(tab, row = 3, column = 2,
                      face = "bold")
tab &lt;- table_cell_bg(tab, row = 4, column = 3, linewidth = 5,
                    fill="darkolivegreen1", color = "darkolivegreen4")
tab

# Change table cells background and font for column 3,
# Spaning from row 2 to the last row in the data
tab &lt;- ggtexttable(df, rows = NULL, theme = ttheme("classic"))
tab %&gt;%
 table_cell_bg(row = 2:tab_nrow(tab), column = 3, fill = "darkblue") %&gt;%
 table_cell_font(row = 2:tab_nrow(tab), column = 3, face = "italic", color = "white")

# Add separators and borders
# :::::::::::::::::::::::::::::::::::::::::::::::::::
# Table with blank theme
tab &lt;- ggtexttable(df, theme = ttheme("blank"), rows = NULL)
# Add horizontal and vertical lines
tab %&gt;%
 tab_add_hline(at.row = c(1, 2), row.side = "top", linewidth = 3, linetype = 1) %&gt;%
 tab_add_hline(at.row = c(7), row.side = "bottom", linewidth = 3, linetype = 1) %&gt;%
 tab_add_vline(at.column = 2:tab_ncol(tab), column.side = "left", from.row = 2, linetype = 2)

# Add borders to table body and header
# Cross out some cells
tab %&gt;%
 tbody_add_border() %&gt;%
 thead_add_border() %&gt;%
 tab_cell_crossout(
   row = c(2, 4), column = 3, linecolor = "red",
   reduce.size.by = 0.6
 )

# Add titles andd footnote
# :::::::::::::::::::::::::::::::::::::::::::::::::::
# Add titles and footnote
# Wrap subtitle into multiple lines using strwrap()
main.title &lt;- "Edgar Anderson's Iris Data"
subtitle &lt;- paste0(
"This famous (Fisher's or Anderson's) iris data set gives the measurements",
" in centimeters of the variables sepal length and width and petal length and width,",
 " respectively, for 50 flowers from each of 3 species of iris.",
 " The species are Iris setosa, versicolor, and virginica."
) %&gt;%
 strwrap(width = 80) %&gt;%
 paste(collapse = "\n")

tab &lt;- ggtexttable(head(iris), theme = ttheme("light"))
tab %&gt;%
 tab_add_title(text = subtitle, face = "plain", size = 10) %&gt;%
 tab_add_title(text = main.title, face = "bold", padding = unit(0.1, "line")) %&gt;%
 tab_add_footnote(text = "*Table created using ggpubr", size = 10, face = "italic")


# Combine density plot and summary table
#:::::::::::::::::::::::::::::::::::::
# Density plot of "Sepal.Length"
density.p &lt;- ggdensity(iris, x = "Sepal.Length",
                      fill = "Species", palette = "jco")

# Draw the summary table of Sepal.Length
# Descriptive statistics by groups
stable &lt;- desc_statby(iris, measure.var = "Sepal.Length",
                     grps = "Species")
stable &lt;- stable[, c("Species", "length", "mean", "sd")]
stable.p &lt;- ggtexttable(stable, rows = NULL,
                       theme = ttheme("mOrange"))

# Arrange the plots on the same page
ggarrange(density.p, stable.p,
         ncol = 1, nrow = 2,
         heights = c(1, 0.5))

</code></pre>

<hr>
<h2 id='ggviolin'>Violin plot</h2><span id='topic+ggviolin'></span>

<h3>Description</h3>

<p>Create a violin plot with error bars. Violin plots are similar to
box plots, except that they also show the kernel probability density of the
data at different values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggviolin(
  data,
  x,
  y,
  combine = FALSE,
  merge = FALSE,
  color = "black",
  fill = "white",
  palette = NULL,
  alpha = 1,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  facet.by = NULL,
  panel.labs = NULL,
  short.panel.labs = TRUE,
  linetype = "solid",
  trim = FALSE,
  size = NULL,
  width = 1,
  draw_quantiles = NULL,
  select = NULL,
  remove = NULL,
  order = NULL,
  add = "mean_se",
  add.params = list(),
  error.plot = "pointrange",
  label = NULL,
  font.label = list(size = 11, color = "black"),
  label.select = NULL,
  repel = FALSE,
  label.rectangle = FALSE,
  position = position_dodge(0.8),
  ggtheme = theme_pubr(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggviolin_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_x">x</code></td>
<td>
<p>character string containing the name of x variable.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_y">y</code></td>
<td>
<p>character vector containing one or more variables to plot</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_combine">combine</code></td>
<td>
<p>logical value. Default is FALSE. Used only when y is a vector
containing multiple variables to plot. If TRUE, create a multi-panel plot by
combining the plot of y variables.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_merge">merge</code></td>
<td>
<p>logical or character value. Default is FALSE. Used only when y is
a vector containing multiple variables to plot. If TRUE, merge multiple y
variables in the same plotting area. Allowed values include also &quot;asis&quot;
(TRUE) and &quot;flip&quot;. If merge = &quot;flip&quot;, then y variables are used as x tick
labels and the x variable is used as grouping variable.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_color">color</code></td>
<td>
<p>outline color.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_fill">fill</code></td>
<td>
<p>fill color.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by groups.
Allowed values include &quot;grey&quot; for grey color palettes; brewer palettes e.g.
&quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and
scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;,
&quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_alpha">alpha</code></td>
<td>
<p>color transparency. Values should be between 0 and 1.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_title">title</code></td>
<td>
<p>plot main title.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_xlab">xlab</code></td>
<td>
<p>character vector specifying x axis labels. Use xlab = FALSE to
hide xlab.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_ylab">ylab</code></td>
<td>
<p>character vector specifying y axis labels. Use ylab = FALSE to
hide ylab.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot into multiple panels. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet panel
labels. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is TRUE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_linetype">linetype</code></td>
<td>
<p>line types.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_trim">trim</code></td>
<td>
<p>If <code>TRUE</code> (default), trim the tails of the violins
to the range of the data. If <code>FALSE</code>, don't trim the tails.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_size">size</code></td>
<td>
<p>Numeric value (e.g.: size = 1). change the size of points and
outlines.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_width">width</code></td>
<td>
<p>violin width.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_draw_quantiles">draw_quantiles</code></td>
<td>
<p>If <code>not(NULL)</code> (default), draw horizontal lines
at the given quantiles of the density estimate.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_select">select</code></td>
<td>
<p>character vector specifying which items to display.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_remove">remove</code></td>
<td>
<p>character vector specifying which items to remove from the plot.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_order">order</code></td>
<td>
<p>character vector specifying the order of items.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_add">add</code></td>
<td>
<p>character vector for adding another plot element (e.g.: dot plot or
error bars). Allowed values are one or the combination of: &quot;none&quot;,
&quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;,
&quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;,
&quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;; see ?desc_statby for more
details.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_add.params">add.params</code></td>
<td>
<p>parameters (color, shape, size, fill, linetype) for the
argument 'add'; e.g.: add.params = list(color = &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_error.plot">error.plot</code></td>
<td>
<p>plot type used to visualize error. Allowed values are one of
c(&quot;pointrange&quot;, &quot;linerange&quot;, &quot;crossbar&quot;, &quot;errorbar&quot;, &quot;upper_errorbar&quot;,
&quot;lower_errorbar&quot;, &quot;upper_pointrange&quot;, &quot;lower_pointrange&quot;, &quot;upper_linerange&quot;,
&quot;lower_linerange&quot;). Default value is &quot;pointrange&quot; or &quot;errorbar&quot;. Used only
when add != &quot;none&quot; and add contains one &quot;mean_*&quot; or &quot;med_*&quot; where &quot;*&quot; = sd,
se, ....</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_label">label</code></td>
<td>
<p>the name of the column containing point labels. Can be also a
character vector with length = nrow(data).</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_font.label">font.label</code></td>
<td>
<p>a list which can contain the combination of the following
elements: the size (e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;,
&quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of labels. For example font.label
= list(size = 14, face = &quot;bold&quot;, color =&quot;red&quot;). To specify only the size and
the style, use font.label = list(size = 14, face = &quot;plain&quot;).</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_label.select">label.select</code></td>
<td>
<p>can be of two formats: </p>
 <ul>
<li><p> a character vector
specifying some labels to show. </p>
</li>
<li><p> a list containing one or the
combination of the following components: </p>
 <ul>
<li> <p><code>top.up</code> and
<code>top.down</code>: to display the labels  of the top up/down points. For
example, <code>label.select = list(top.up = 10, top.down = 4)</code>. </p>
</li>
<li>
<p><code>criteria</code>: to filter, for example, by x and y variabes values, use
this: <code>label.select = list(criteria = "`y` &gt; 2 &amp; `y` &lt; 5 &amp; `x` %in%
c('A', 'B')")</code>. </p>
</li></ul>
 </li></ul>
</td></tr>
<tr><td><code id="ggviolin_+3A_repel">repel</code></td>
<td>
<p>a logical value, whether to use ggrepel to avoid overplotting
text labels or not.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_label.rectangle">label.rectangle</code></td>
<td>
<p>logical value. If TRUE, add rectangle underneath the
text, making it easier to read.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_pubr().
Allowed values include ggplot2 official themes: theme_gray(), theme_bw(),
theme_minimal(), theme_classic(), theme_void(), ....</p>
</td></tr>
<tr><td><code id="ggviolin_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to
<code><a href="ggplot2.html#topic+geom_violin">geom_violin</a></code>, <code><a href="#topic+ggpar">ggpar</a></code> and
<code><a href="#topic+facet">facet</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot can be easily customized using the function ggpar(). Read
?ggpar for changing: </p>
 <ul>
<li><p> main title and axis labels: main, xlab,
ylab </p>
</li>
<li><p> axis limits: xlim, ylim (e.g.: ylim = c(0, 30)) </p>
</li>
<li><p> axis
scales: xscale, yscale (e.g.: yscale = &quot;log2&quot;) </p>
</li>
<li><p> color palettes: palette
= &quot;Dark2&quot; or palette = c(&quot;gray&quot;, &quot;blue&quot;, &quot;red&quot;) </p>
</li>
<li><p> legend title, labels
and position: legend = &quot;right&quot; </p>
</li>
<li><p> plot orientation : orientation =
c(&quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;) </p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")
df &lt;- ToothGrowth

# Basic plot
# +++++++++++++++++++++++++++
ggviolin(df, x = "dose", y = "len")
# Change the plot orientation: horizontal
ggviolin(df, "dose", "len", orientation = "horiz")

# Add summary statistics
# ++++++++++++++++++++++++++
# Draw quantiles
ggviolin(df, "dose", "len", add = "none",
   draw_quantiles = 0.5)

# Add box plot
ggviolin(df, x = "dose", y = "len",
 add = "boxplot")

ggviolin(df, x = "dose", y = "len",
 add = "dotplot")

# Add jitter points and
# change point shape by groups ("dose")
ggviolin(df, x = "dose", y = "len",
add = "jitter", shape = "dose")


# Add mean_sd + jittered points
ggviolin(df, x = "dose", y = "len",
 add = c("jitter", "mean_sd"))

# Change error.plot to "crossbar"
ggviolin(df, x = "dose", y = "len",
 add = "mean_sd", error.plot = "crossbar")


# Change colors
# +++++++++++++++++++++++++++
# Change outline and fill colors
ggviolin(df, "dose", "len",
   color = "black", fill = "gray")

# Change outline colors by groups: dose
# Use custom color palette and add boxplot
ggviolin(df, "dose", "len",  color = "dose",
   palette = c("#00AFBB", "#E7B800", "#FC4E07"),
   add = "boxplot")

# Change fill color by groups: dose
# add boxplot with white fill color
ggviolin(df, "dose", "len", fill = "dose",
   palette = c("#00AFBB", "#E7B800", "#FC4E07"),
   add = "boxplot", add.params = list(fill = "white"))


# Plot with multiple groups
# +++++++++++++++++++++
# fill or color box plot by a second group : "supp"
ggviolin(df, "dose", "len", color = "supp",
 palette = c("#00AFBB", "#E7B800"), add = "boxplot")

</code></pre>

<hr>
<h2 id='gradient_color'>Set Gradient Color</h2><span id='topic+gradient_color'></span><span id='topic+gradient_fill'></span>

<h3>Description</h3>

<p>Change gradient color.
</p>

<ul>
<li> <p><code>gradient_color()</code>: Change gradient color.
</p>
</li>
<li> <p><code>gradient_fill()</code>: Change gradient fill.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>gradient_color(palette)

gradient_fill(palette)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gradient_color_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring or filling by
groups. Allowed values include &quot;grey&quot; for grey color palettes; brewer
palettes e.g. &quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;,
&quot;red&quot;); and scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;,
&quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and
&quot;rickandmorty&quot;. Can be also a numeric vector; in this
case a basic color palette is created using the function
<a href="grDevices.html#topic+palette">palette</a>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="#topic+set_palette">set_palette</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- mtcars
p &lt;- ggscatter(df, x = "wt", y = "mpg",
               color = "mpg")

# Change gradient color
# Use one custom color
p + gradient_color("red")

# Two colors
p + gradient_color(c("blue",  "red"))

# Three colors
p + gradient_color(c("blue", "white", "red"))

# Use RColorBrewer palette
p + gradient_color("RdYlBu")

# Use ggsci color palette
p + gradient_color("npg")
</code></pre>

<hr>
<h2 id='grids'>Add Grids to a ggplot</h2><span id='topic+grids'></span>

<h3>Description</h3>

<p>Add grids to ggplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grids(axis = c("xy", "x", "y"), color = "grey92", size = NULL, linetype = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="grids_+3A_axis">axis</code></td>
<td>
<p>axis for which grid should be added. Allowed values include <code>c("xy", "x", "y")</code>.</p>
</td></tr>
<tr><td><code id="grids_+3A_color">color</code></td>
<td>
<p>grid line color.</p>
</td></tr>
<tr><td><code id="grids_+3A_size">size</code></td>
<td>
<p>numeric value specifying grid line size.</p>
</td></tr>
<tr><td><code id="grids_+3A_linetype">linetype</code></td>
<td>
<p>line type. An integer (0:8), a name (blank, solid, dashed,
dotted, dotdash, longdash, twodash). Sess <code><a href="#topic+show_line_types">show_line_types</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")

# Basic plot
p &lt;- ggboxplot(ToothGrowth, x = "dose", y = "len")
p

# Add border
p + grids(linetype = "dashed")
</code></pre>

<hr>
<h2 id='npc_to_data_coord'>Convert NPC to Data Coordinates</h2><span id='topic+npc_to_data_coord'></span>

<h3>Description</h3>

<p>Convert NPC (Normalized Parent Coordinates) into data coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>npc_to_data_coord(npc, data.ranges)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="npc_to_data_coord_+3A_npc">npc</code></td>
<td>
<p>a numeric vector. Each value should be in [0-1]</p>
</td></tr>
<tr><td><code id="npc_to_data_coord_+3A_data.ranges">data.ranges</code></td>
<td>
<p>a numeric vector of length 2 containing the data ranges (minimum and the maximum)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric vector representing data coordinates.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as_npc">as_npc</a></code>, <code><a href="#topic+get_coord">get_coord</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>npc_to_data_coord(npc = c(0.2, 0.95), data.ranges = c(1, 20))
as_npc(c("top", "right")) %&gt;%
   npc_to_data_coord(data.ranges = c(1, 20))

</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+mutate'></span><span id='topic+group_by'></span><span id='topic+get_summary_stats'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>dplyr</dt><dd><p><code><a href="dplyr.html#topic+group_by">group_by</a></code>, <code><a href="dplyr.html#topic+mutate">mutate</a></code></p>
</dd>
<dt>rstatix</dt><dd><p><code><a href="rstatix.html#topic+get_summary_stats">get_summary_stats</a></code></p>
</dd>
</dl>

<hr>
<h2 id='rotate'>Rotate a ggplot Horizontally</h2><span id='topic+rotate'></span>

<h3>Description</h3>

<p>Rotate a ggplot to create horizontal plots. Wrapper around
<code><a href="ggplot2.html#topic+coord_flip">coord_flip</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rotate(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rotate_+3A_...">...</code></td>
<td>
<p>other arguments to pass to <code><a href="ggplot2.html#topic+coord_flip">coord_flip</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")
df &lt;- ToothGrowth

# Basic plot
p &lt;- ggboxplot(df, x = "dose", y = "len",
   color = "dose", palette = "jco")
p
# Create horizontal plots
p + rotate()
</code></pre>

<hr>
<h2 id='rotate_axis_text'>Rotate Axes Text</h2><span id='topic+rotate_axis_text'></span><span id='topic+rotate_x_text'></span><span id='topic+rotate_y_text'></span>

<h3>Description</h3>

<p>Rotate the x-axis text (tick mark labels).
</p>

<ul>
<li> <p><code>rotate_x_text</code>(): Rotate x axis text.
</p>
</li>
<li> <p><code>rotate_y_text</code>(): Rotate y axis text.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>rotate_x_text(angle = 90, hjust = NULL, vjust = NULL, ...)

rotate_y_text(angle = 90, hjust = NULL, vjust = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rotate_axis_text_+3A_angle">angle</code></td>
<td>
<p>numeric value specifying the rotation angle. Default is 90 for vertical x-axis text.</p>
</td></tr>
<tr><td><code id="rotate_axis_text_+3A_hjust">hjust</code></td>
<td>
<p>horizontal justification (in [0, 1]).</p>
</td></tr>
<tr><td><code id="rotate_axis_text_+3A_vjust">vjust</code></td>
<td>
<p>vertical justification (in [0, 1]).</p>
</td></tr>
<tr><td><code id="rotate_axis_text_+3A_...">...</code></td>
<td>
<p>other arguments to pass to the function
<code><a href="ggplot2.html#topic+element">element_text</a>()</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")
df &lt;- ToothGrowth

# Basic plot
p &lt;- ggboxplot(df, x = "dose", y = "len")
p
# Vertical x axis text
p + rotate_x_text()
# Set rotation angle to 45
p + rotate_x_text(45)
p + rotate_y_text(45)

</code></pre>

<hr>
<h2 id='rremove'>Remove a ggplot Component</h2><span id='topic+rremove'></span>

<h3>Description</h3>

<p>Remove a specific component from a ggplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rremove(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rremove_+3A_object">object</code></td>
<td>
<p>character string specifying the plot components. Allowed values
include: </p>

<ul>
<li> <p><code>"grid"</code> for both x and y grids
</p>
</li>
<li> <p><code>"x.grid"</code> for x axis grids
</p>
</li>
<li> <p><code>"y.grid"</code> for y axis grids
</p>
</li>
<li> <p><code>"axis"</code> for both x and y axes
</p>
</li>
<li> <p><code>"x.axis"</code> for x axis
</p>
</li>
<li> <p><code>"y.axis"</code> for y axis
</p>
</li>
<li> <p><code>"xlab", or "x.title"</code>
for x axis label
</p>
</li>
<li> <p><code>"ylab", or "y.title"</code> for y axis label
</p>
</li>
<li> <p><code>"xylab", "xy.title" or "axis.title"</code> for both x and y axis
labels
</p>
</li>
<li> <p><code>"x.text"</code> for x axis texts (x axis tick labels)
</p>
</li>
<li> <p><code>"y.text"</code> for y axis texts (y axis tick labels)
</p>
</li>
<li> <p><code>"xy.text"</code> or <code>"axis.text"</code> for both x and y axis texts
</p>
</li>
<li> <p><code>"ticks"</code> for both x and y ticks
</p>
</li>
<li> <p><code>"x.ticks"</code> for x ticks
</p>
</li>
<li> <p><code>"y.ticks"</code> for y ticks
</p>
</li>
<li> <p><code>"legend.title"</code> for the legend title
</p>
</li>
<li> <p><code>"legend"</code> for the legend
</p>
</li></ul>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")

# Basic plot
p &lt;- ggboxplot(ToothGrowth, x = "dose", y = "len",
  ggtheme = theme_gray())
p

# Remove all grids
p + rremove("grid")

# Remove only x grids
p + rremove("x.grid")
</code></pre>

<hr>
<h2 id='set_palette'>Set Color Palette</h2><span id='topic+set_palette'></span><span id='topic+change_palette'></span><span id='topic+color_palette'></span><span id='topic+fill_palette'></span>

<h3>Description</h3>

 <ul>
<li> <p><code>change_palette(), set_palette()</code>: Change
both color and fill palettes. </p>
</li>
<li> <p><code>color_palette()</code>: change color
palette only. </p>
</li>
<li> <p><code>fill_palette()</code>: change fill palette only.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>set_palette(p, palette)

change_palette(p, palette)

color_palette(palette = NULL, ...)

fill_palette(palette = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_palette_+3A_p">p</code></td>
<td>
<p>a ggplot</p>
</td></tr>
<tr><td><code id="set_palette_+3A_palette">palette</code></td>
<td>
<p>Color palette. Allowed values include: </p>
 <ul>
<li>
<p><strong>Grey color palettes</strong>: &quot;grey&quot; or &quot;gray&quot;; </p>
</li>
<li> <p><strong>RColorBrewer
palettes</strong>, see <code><a href="RColorBrewer.html#topic+ColorBrewer">brewer.pal</a></code> and details section.
Examples of palette names include: &quot;RdBu&quot;, &quot;Blues&quot;, &quot;Dark2&quot;, &quot;Set2&quot;, ...;
</p>
</li>
<li> <p><strong>Custom color palettes</strong>. For example, palette = c(&quot;#00AFBB&quot;,
&quot;#E7B800&quot;, &quot;#FC4E07&quot;); </p>
</li>
<li> <p><strong>ggsci scientific journal palettes</strong>,
e.g.: &quot;npg&quot;, &quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and
&quot;rickandmorty&quot;. </p>
</li></ul>
</td></tr>
<tr><td><code id="set_palette_+3A_...">...</code></td>
<td>
<p>other arguments passed to ggplot2 scale_color_xxx() and
scale_fill_xxx() functions.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="#topic+get_palette">get_palette</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")
df &lt;- ToothGrowth

# Basic plot
p &lt;- ggboxplot(df, x = "dose", y = "len",
   color = "dose")
p

# Change the color palette
set_palette(p, "jco")
</code></pre>

<hr>
<h2 id='show_line_types'>Line types available in R</h2><span id='topic+show_line_types'></span>

<h3>Description</h3>

<p>Show line types available in R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show_line_types()
</code></pre>


<h3>Value</h3>

<p>a ggplot.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code> and <code><a href="#topic+ggline">ggline</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>show_line_types()+
 theme_minimal()
</code></pre>

<hr>
<h2 id='show_point_shapes'>Point shapes available in R</h2><span id='topic+show_point_shapes'></span>

<h3>Description</h3>

<p>Show point shapes available in R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show_point_shapes()
</code></pre>


<h3>Value</h3>

<p>a ggplot.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code> and <code><a href="#topic+ggline">ggline</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>show_point_shapes()+
 theme_minimal()
</code></pre>

<hr>
<h2 id='stat_anova_test'>Add Anova Test P-values to a GGPlot</h2><span id='topic+stat_anova_test'></span>

<h3>Description</h3>

<p>Adds automatically one-way and two-way ANOVA test p-values to a
ggplot, such as box blots, dot plots and stripcharts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_anova_test(
  mapping = NULL,
  data = NULL,
  method = c("one_way", "one_way_repeated", "two_way", "two_way_repeated",
    "two_way_mixed"),
  wid = NULL,
  group.by = NULL,
  type = NULL,
  effect.size = "ges",
  error = NULL,
  correction = c("auto", "GG", "HF", "none"),
  label = "{method}, p = {p.format}",
  label.x.npc = "left",
  label.y.npc = "top",
  label.x = NULL,
  label.y = NULL,
  step.increase = 0.1,
  p.adjust.method = "holm",
  significance = list(),
  geom = "text",
  position = "identity",
  na.rm = FALSE,
  show.legend = FALSE,
  inherit.aes = TRUE,
  parse = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_anova_test_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_method">method</code></td>
<td>
<p>ANOVA test methods. Possible values are one of
<code>c("one_way", "one_way_repeated", "two_way", "two_way_repeated",
"two_way_mixed")</code>.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_wid">wid</code></td>
<td>
<p>(factor) column name containing individuals/subjects identifier.
Should be unique per individual. Required only for repeated measure tests
(<code>"one_way_repeated", "two_way_repeated", "friedman_test", etc</code>).</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_group.by">group.by</code></td>
<td>
<p>(optional) character vector specifying the grouping variable;
it should be used only for grouped plots. Possible values are : </p>

<ul>
<li> <p><code>"x.var"</code>: Group by the x-axis variable and perform the test
between legend groups. In other words, the p-value is compute between legend
groups at each x position </p>
</li>
<li> <p><code>"legend.var"</code>: Group by the legend
variable and perform the test between x-axis groups. In other words, the
test is performed between the x-groups for each legend level. </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_type">type</code></td>
<td>
<p>the type of sums of squares for ANOVA. Allowed values are either
1, 2 or 3. <code>type = 2</code> is the default because this will yield identical
ANOVA results as type = 1 when data are balanced but type = 2 will
additionally yield various assumption tests where appropriate. When the data
are unbalanced the <code>type = 3</code> is used by popular commercial softwares
including SPSS.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_effect.size">effect.size</code></td>
<td>
<p>the effect size to compute and to show in the ANOVA
results. Allowed values can be either &quot;ges&quot; (generalized eta squared) or
&quot;pes&quot; (partial eta squared) or both. Default is &quot;ges&quot;.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_error">error</code></td>
<td>
<p>(optional) for a linear model, an lm model object from which the
overall error sum of squares and degrees of freedom are to be calculated.
Read more in <code><a href="car.html#topic+Anova">Anova</a>()</code> documentation.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_correction">correction</code></td>
<td>
<p>character. Used only in repeated measures ANOVA test to
specify which correction of the degrees of freedom should be reported for
the within-subject factors. Possible values are: </p>
 <ul>
<li><p>&quot;GG&quot;:
applies Greenhouse-Geisser correction to all within-subjects factors even if
the assumption of sphericity is met (i.e., Mauchly's test is not
significant, p &gt; 0.05). </p>
</li>
<li><p>&quot;HF&quot;: applies Hyunh-Feldt correction to all
within-subjects factors even if the assumption of sphericity is met,
</p>
</li>
<li><p>&quot;none&quot;: returns the ANOVA table without any correction and
</p>
</li>
<li><p>&quot;auto&quot;: apply automatically GG correction to only within-subjects
factors violating the sphericity assumption (i.e., Mauchly's test p-value is
significant, p &lt;= 0.05). </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_label">label</code></td>
<td>
<p>character string specifying label. Can be: </p>
 <ul>
<li><p> the
column containing the label (e.g.: <code>label = "p"</code> or <code>label =
"p.adj"</code>), where <code>p</code> is the p-value. Other possible values are
<code>"p.signif", "p.adj.signif", "p.format", "p.adj.format"</code>. </p>
</li>
<li><p> an
expression that can be formatted by the <code><a href="glue.html#topic+glue">glue</a>()</code> package.
For example, when specifying <code>label = "Anova, p = \{p\}"</code>, the
expression {p} will be replaced by its value. </p>
</li>
<li><p> a combination of
plotmath expressions and glue expressions. You may want some of the
statistical parameter in italic; for example:<code>label = "Anova, italic(p)
= {p}"</code>. </p>
</li>
<li><p> a constant: <code>label = "as_italic"</code>: display statistical
parameters in italic; <code>label = "as_detailed"</code>: detailed plain text;
<code>label = "as_detailed_expression"</code> or <code>label =
"as_detailed_italic"</code>: detailed plotmath expression. Statistical parameters
will be displayed in italic.</p>
</li></ul>
<p>.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_label.x.npc">label.x.npc</code>, <code id="stat_anova_test_+3A_label.y.npc">label.y.npc</code></td>
<td>
<p>can be <code>numeric</code> or <code>character</code>
vector of the same length as the number of groups and/or panels. If too
short they will be recycled. </p>
 <ul>
<li><p> If <code>numeric</code>, value should
be between 0 and 1. Coordinates to be used for positioning the label,
expressed in &quot;normalized parent coordinates&quot;. </p>
</li>
<li><p> If <code>character</code>,
allowed values include: i) one of c('right', 'left', 'center', 'centre',
'middle') for x-axis; ii) and one of c( 'bottom', 'top', 'center', 'centre',
'middle') for y-axis.</p>
</li></ul>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_label.x">label.x</code>, <code id="stat_anova_test_+3A_label.y">label.y</code></td>
<td>
<p><code>numeric</code> Coordinates (in data units) to be used
for absolute positioning of the label. If too short they will be recycled.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_step.increase">step.increase</code></td>
<td>
<p>numeric value in with the increase in fraction of total
height for every additional comparison to minimize overlap. The step value
can be negative to reverse the order of groups.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for adjusting p values (see
<code><a href="stats.html#topic+p.adjust">p.adjust</a></code>).  Has impact only in a situation, where
multiple pairwise tests are performed; or when there are multiple grouping
variables. Allowed values include &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;,
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;. If you don't want to adjust the p
value (not recommended), use p.adjust.method = &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_significance">significance</code></td>
<td>
<p>a list of arguments specifying the signifcance cutpoints
and symbols. For example, <code>significance &lt;- list(cutpoints = c(0,
 0.0001, 0.001, 0.01, 0.05, Inf), symbols = c("****", "***", "**", "*",
 "ns"))</code>.
</p>
<p>In other words, we use the following convention for symbols indicating
statistical significance: </p>
 <ul>
<li> <p><code>ns</code>: p &gt; 0.05 </p>
</li>
<li>
<p><code>*</code>: p &lt;= 0.05 </p>
</li>
<li> <p><code>**</code>: p &lt;= 0.01 </p>
</li>
<li> <p><code>***</code>: p &lt;= 0.001
</p>
</li>
<li> <p><code>****</code>:  p &lt;= 0.0001 </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_na.rm">na.rm</code></td>
<td>
<p>If FALSE (the default), removes missing values with a warning. If
TRUE silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_parse">parse</code></td>
<td>
<p>If TRUE, the labels will be parsed into expressions and displayed
as described in <code>?plotmath</code>.</p>
</td></tr>
<tr><td><code id="stat_anova_test_+3A_...">...</code></td>
<td>
<p>other arguments to pass to
<code><a href="ggplot2.html#topic+geom_text">geom_text</a></code>, such as:</p>
 <ul>
<li>
<p><code>hjust</code>: horizontal justification of the text. Move the text left or
right and </p>
</li>
<li> <p><code>vjust</code>: vertical justification of the text. Move the
text up or down. </p>
</li></ul>
</td></tr>
</table>


<h3>Computed variables</h3>

 <ul>
<li><p>DFn: Degrees of Freedom in the
numerator (i.e. DF effect). </p>
</li>
<li><p>DFd:	Degrees of Freedom in the
denominator (i.e., DF error). </p>
</li>
<li><p>ges:	Generalized Eta-Squared measure of
effect size. Computed only when the option <code>effect.size = "ges"</code>.
</p>
</li>
<li><p>pes:	Partial Eta-Squared measure of effect size. Computed only when
the option <code>effect.size = "pes"</code>. </p>
</li>
<li><p>F:	F-value. </p>
</li>
<li><p>p:	p-value.
</p>
</li>
<li><p>p.adj: Adjusted p-values. </p>
</li>
<li><p>p.signif: P-value significance.
</p>
</li>
<li><p>p.adj.signif: Adjusted p-value significance. </p>
</li>
<li><p>p.format:
Formated p-value. </p>
</li>
<li><p>p.adj.format: Formated adjusted p-value. </p>
</li>
<li><p>n:
number of samples. </p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Data preparation
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Transform `dose` into factor variable
df &lt;- ToothGrowth
df$dose &lt;- as.factor(df$dose)
# Add individuals id
df$id &lt;- rep(1:10, 6)
# Add a random grouping variable
set.seed(123)
df$group &lt;- sample(factor(rep(c("grp1", "grp2", "grp3"), 20)))
df$len &lt;- ifelse(df$group == "grp2", df$len+2, df$len)
df$len &lt;- ifelse(df$group == "grp3", df$len+7, df$len)
head(df, 3)


# Basic boxplot
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Create a basic boxplot
# Add 5% and 10% space to the plot bottom and the top, respectively
bxp &lt;- ggboxplot(df, x = "dose", y = "len") +
  scale_y_continuous(expand = expansion(mult = c(0.05, 0.1)))

# Add the p-value to the boxplot
bxp + stat_anova_test()

## Not run: 
# Change the label position
# Using coordinates in data units
bxp + stat_anova_test(label.x = "1", label.y = 10, hjust = 0)

## End(Not run)

# Format the p-value differently
custom_p_format &lt;- function(p) {
  rstatix::p_format(p, accuracy = 0.0001, digits = 3, leading.zero = FALSE)
}
bxp + stat_anova_test(
  label = "Anova, italic(p) = {custom_p_format(p)}{p.signif}"
)

# Show a detailed label in italic
bxp + stat_anova_test(label = "as_detailed_italic")


# Faceted plots
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Create a ggplot facet
bxp &lt;- ggboxplot(df, x = "dose", y = "len", facet.by = "supp") +
  scale_y_continuous(expand = expansion(mult = c(0.05, 0.1)))
# Add p-values
bxp + stat_anova_test()


# Grouped plots
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
bxp2 &lt;- ggboxplot(df, x = "group", y = "len", color = "dose", palette = "npg")

# For each x-position, computes tests between legend groups
bxp2 + stat_anova_test(aes(group = dose), label = "p = {p.format}{p.signif}")

#  For each legend group, computes tests between x variable groups
bxp2 + stat_anova_test(aes(group = dose, color = dose), group.by = "legend.var")


## Not run: 
# Two-way ANOVA: Independent measures
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Visualization: box plots with p-values
# Two-way interaction p-values between x and legend (group) variables
bxp3 &lt;- ggboxplot(
  df, x = "supp", y = "len",
  color = "dose", palette = "jco"
)
bxp3 + stat_anova_test(aes(group = dose),  method = "two_way")

# One-way repeatead measures ANOVA
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
df$id &lt;- as.factor(c(rep(1:10, 3), rep(11:20, 3)))
ggboxplot(df, x = "dose", y = "len") +
  stat_anova_test(method = "one_way_repeated", wid = "id")

# Two-way repeatead measures ANOVA
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
df$id &lt;- as.factor(rep(1:10, 6))
ggboxplot(df, x = "dose", y = "len", color = "supp", palette = "jco") +
  stat_anova_test(aes(group = supp), method = "two_way_repeated", wid = "id")

# Grouped one-way repeated measures ANOVA
ggboxplot(df, x = "dose", y = "len", color = "supp", palette = "jco") +
  stat_anova_test(aes(group = supp, color = supp),
  method = "one_way_repeated", wid = "id", group.by = "legend.var")
 
## End(Not run)
</code></pre>

<hr>
<h2 id='stat_bracket'>Add Brackets with Labels to a GGPlot</h2><span id='topic+stat_bracket'></span><span id='topic+geom_bracket'></span>

<h3>Description</h3>

<p>add brackets with label annotation to a ggplot. Helpers for
adding p-value or significance levels to a plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_bracket(
  mapping = NULL,
  data = NULL,
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  label = NULL,
  type = c("text", "expression"),
  y.position = NULL,
  xmin = NULL,
  xmax = NULL,
  step.increase = 0,
  step.group.by = NULL,
  tip.length = 0.03,
  bracket.nudge.y = 0,
  bracket.shorten = 0,
  size = 0.3,
  label.size = 3.88,
  family = "",
  vjust = 0,
  ...
)

geom_bracket(
  mapping = NULL,
  data = NULL,
  stat = "bracket",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  label = NULL,
  type = c("text", "expression"),
  y.position = NULL,
  xmin = NULL,
  xmax = NULL,
  step.increase = 0,
  step.group.by = NULL,
  tip.length = 0.03,
  bracket.nudge.y = 0,
  bracket.shorten = 0,
  size = 0.3,
  label.size = 3.88,
  family = "",
  vjust = 0,
  coord.flip = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_bracket_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code> (the default), removes missing values with a
warning.  If <code>TRUE</code> silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_label">label</code></td>
<td>
<p>character vector with alternative label, if not null test is
ignored</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_type">type</code></td>
<td>
<p>the label type. Can be one of &quot;text&quot; and &quot;expression&quot; (for
parsing plotmath expression).</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_y.position">y.position</code></td>
<td>
<p>numeric vector with the y positions of the brackets</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_xmin">xmin</code></td>
<td>
<p>numeric vector with the positions of the left sides of the
brackets</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_xmax">xmax</code></td>
<td>
<p>numeric vector with the positions of the right sides of the
brackets</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_step.increase">step.increase</code></td>
<td>
<p>numeric vector with the increase in fraction of total
height for every additional comparison to minimize overlap.</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_step.group.by">step.group.by</code></td>
<td>
<p>a variable name for grouping brackets before adding
step.increase. Useful to group bracket by facet panel.</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_tip.length">tip.length</code></td>
<td>
<p>numeric vector with the fraction of total height that the
bar goes down to indicate the precise column</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_bracket.nudge.y">bracket.nudge.y</code></td>
<td>
<p>Vertical adjustment to nudge brackets by. Useful to
move up or move down the bracket. If positive value, brackets will be moved
up; if negative value, brackets are moved down.</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_bracket.shorten">bracket.shorten</code></td>
<td>
<p>a small numeric value in [0-1] for shortening the with
of bracket.</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_size">size</code></td>
<td>
<p>change the width of the lines of the bracket</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_label.size">label.size</code></td>
<td>
<p>change the size of the label text</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_family">family</code></td>
<td>
<p>change the font used for the text</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_vjust">vjust</code></td>
<td>
<p>move the text up or down relative to the bracket</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_...">...</code></td>
<td>
<p>other arguments passed on to <code>layer</code>. These are often
aesthetics, used to set an aesthetic to a fixed value, like <code>color =
"red"</code> or <code>size = 3</code>. They may also be parameters to the paired
geom/stat.</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_stat">stat</code></td>
<td>
<p>The statistical transformation to use on the data for this
layer, either as a <code>ggproto</code> <code>Geom</code> subclass or as a string naming the
stat stripped of the <code>stat_</code> prefix (e.g. <code>"count"</code> rather than
<code>"stat_count"</code>)</p>
</td></tr>
<tr><td><code id="stat_bracket_+3A_coord.flip">coord.flip</code></td>
<td>
<p>logical. If <code>TRUE</code>, flip x and y coordinates so that
horizontal becomes vertical, and vertical, horizontal. When adding the
p-values to a horizontal ggplot (generated using
<code><a href="ggplot2.html#topic+coord_flip">coord_flip</a>()</code>), you need to specify the option
<code>coord.flip = TRUE</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- ToothGrowth
df$dose &lt;- factor(df$dose)

# Add bracket with labels
ggboxplot(df, x = "dose", y = "len") +
  geom_bracket(
    xmin = "0.5", xmax = "1", y.position = 30,
    label = "t-test, p &lt; 0.05"
  )

# Customize bracket tip.length tip.length
ggboxplot(df, x = "dose", y = "len") +
  geom_bracket(
    xmin = "0.5", xmax = "1", y.position = 30,
    label = "t-test, p &lt; 0.05", tip.length = c(0.2, 0.02)
  )

#Using plotmath expression
ggboxplot(df, x = "dose", y = "len") +
 geom_bracket(
   xmin = "0.5", xmax = "1", y.position = 30,
   label = "list(~italic(p)&lt;=0.001)", type = "expression",
   tip.length = c(0.2, 0.02)
 )

# Specify multiple brackets manually
ggboxplot(df, x = "dose", y = "len") +
  geom_bracket(
    xmin = c("0.5", "1"), xmax = c("1", "2"),
    y.position = c(30, 35), label = c("***", "**"),
    tip.length = 0.01
  )

# Compute statistical tests and add p-values
stat.test &lt;- compare_means(len ~ dose, ToothGrowth, method = "t.test")
ggboxplot(df, x = "dose", y = "len") +
  geom_bracket(
    aes(xmin = group1, xmax = group2, label = signif(p, 2)),
    data = stat.test, y.position = 35
  )

# Increase step length between brackets
ggboxplot(df, x = "dose", y = "len") +
  geom_bracket(
    aes(xmin = group1, xmax = group2, label = signif(p, 2)),
    data = stat.test, y.position = 35, step.increase = 0.1
  )

# Or specify the positions of each comparison
ggboxplot(df, x = "dose", y = "len") +
  geom_bracket(
    aes(xmin = group1, xmax = group2, label = signif(p, 2)),
    data = stat.test, y.position = c(32, 35, 38)
   )
</code></pre>

<hr>
<h2 id='stat_central_tendency'>Add Central Tendency Measures to a GGPLot</h2><span id='topic+stat_central_tendency'></span>

<h3>Description</h3>

<p>Add central tendency measures (mean, median, mode) to density
and histogram plots created using ggplots.
</p>
<p>Note that, normally, the mode is used for categorical data where we wish to
know which is the most common category. Therefore, we can have have two or
more values that share the highest frequency. This might be problematic for
continuous variable.
</p>
<p>For continuous variable, we can consider using mean or median as the
measures of the central tendency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_central_tendency(
  mapping = NULL,
  data = NULL,
  geom = c("line", "point"),
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  type = c("mean", "median", "mode"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_central_tendency_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_central_tendency_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_central_tendency_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_central_tendency_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_central_tendency_+3A_na.rm">na.rm</code></td>
<td>
<p>If FALSE (the default), removes missing values with a warning.
If TRUE silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_central_tendency_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_central_tendency_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_central_tendency_+3A_type">type</code></td>
<td>
<p>the type of central tendency measure to be used. Possible values
include: <code>"mean", "median", "mode"</code>.</p>
</td></tr>
<tr><td><code id="stat_central_tendency_+3A_...">...</code></td>
<td>
<p>other arguments to pass to <code><a href="ggplot2.html#topic+geom_path">geom_line</a></code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ggdensity">ggdensity</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simple density plot
data("mtcars")
ggdensity(mtcars, x = "mpg", fill = "red") +
  scale_x_continuous(limits = c(-1, 50)) +
  stat_central_tendency(type = "mean", linetype = "dashed")

# Color by groups
data(iris)
ggdensity(iris, "Sepal.Length", color = "Species") +
  stat_central_tendency(aes(color = Species), type = "median", linetype = 2)

# Use geom = "point" for central tendency
data(iris)
ggdensity(iris, "Sepal.Length", color = "Species") +
  stat_central_tendency(
     aes(color = Species), type = "median",
     geom = "point", size = 4
     )

# Facet
ggdensity(iris, "Sepal.Length", facet.by = "Species") +
  stat_central_tendency(type = "mean", color = "red", linetype = 2) +
  stat_central_tendency(type = "median", color = "blue", linetype = 2)

</code></pre>

<hr>
<h2 id='stat_chull'>Plot convex hull of a set of points</h2><span id='topic+stat_chull'></span>

<h3>Description</h3>

<p>Plot convex hull of a set of points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_chull(
  mapping = NULL,
  data = NULL,
  geom = "path",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_chull_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_chull_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_chull_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_chull_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_chull_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p>
</td></tr>
<tr><td><code id="stat_chull_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_chull_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_chull_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggplot2.html#topic+layer">layer()</a></code>. These are
often aesthetics, used to set an aesthetic to a fixed value, like
<code>colour = "red"</code> or <code>size = 3</code>. They may also be parameters
to the paired geom/stat.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ggpar">ggpar</a></code>, <code><a href="#topic+ggscatter">ggscatter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("mtcars")
df &lt;- mtcars
df$cyl &lt;- as.factor(df$cyl)

# scatter plot with convex hull
ggscatter(df, x = "wt", y = "mpg", color = "cyl")+
 stat_chull(aes(color = cyl))

ggscatter(df, x = "wt", y = "mpg", color = "cyl")+
 stat_chull(aes(color = cyl, fill = cyl), alpha = 0.1, geom = "polygon")

</code></pre>

<hr>
<h2 id='stat_compare_means'>Add Mean Comparison P-values to a ggplot</h2><span id='topic+stat_compare_means'></span>

<h3>Description</h3>

<p>Add mean comparison p-values to a ggplot, such as box blots, dot
plots and stripcharts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_compare_means(
  mapping = NULL,
  data = NULL,
  method = NULL,
  paired = FALSE,
  method.args = list(),
  ref.group = NULL,
  comparisons = NULL,
  hide.ns = FALSE,
  label.sep = ", ",
  label = NULL,
  label.x.npc = "left",
  label.y.npc = "top",
  label.x = NULL,
  label.y = NULL,
  vjust = 0,
  tip.length = 0.03,
  bracket.size = 0.3,
  step.increase = 0,
  symnum.args = list(),
  geom = "text",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_compare_means_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_method">method</code></td>
<td>
<p>a character string indicating which method to be used for
comparing means.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_paired">paired</code></td>
<td>
<p>a logical indicating whether you want a paired test. Used only
in <code><a href="stats.html#topic+t.test">t.test</a></code> and in <a href="stats.html#topic+wilcox.test">wilcox.test</a>.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_method.args">method.args</code></td>
<td>
<p>a list of additional arguments used for the test method.
For example one might use <code>method.args = list(alternative = "greater")</code>
for wilcoxon test.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_ref.group">ref.group</code></td>
<td>
<p>a character string specifying the reference group. If
specified, for a given grouping variable, each of the group levels will be
compared to the reference group (i.e. control group).
</p>
<p><code>ref.group</code> can be also <code>".all."</code>. In this case, each of the
grouping variable levels is compared to all (i.e. basemean).</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_comparisons">comparisons</code></td>
<td>
<p>A list of length-2 vectors. The entries in the vector are
either the names of 2 values on the x-axis or the 2 integers that correspond
to the index of the groups of interest, to be compared.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_hide.ns">hide.ns</code></td>
<td>
<p>logical value. If TRUE, hide ns symbol when displaying
significance levels.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_label.sep">label.sep</code></td>
<td>
<p>a character string to separate the terms. Default is &quot;, &quot;, to
separate the correlation coefficient and the p.value.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_label">label</code></td>
<td>
<p>character string specifying label type. Allowed values include
&quot;p.signif&quot; (shows the significance levels), &quot;p.format&quot; (shows the formatted
p value).</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_label.x.npc">label.x.npc</code>, <code id="stat_compare_means_+3A_label.y.npc">label.y.npc</code></td>
<td>
<p>can be <code>numeric</code> or <code>character</code>
vector of the same length as the number of groups and/or panels. If too
short they will be recycled. </p>
 <ul>
<li><p> If <code>numeric</code>, value should
be between 0 and 1. Coordinates to be used for positioning the label,
expressed in &quot;normalized parent coordinates&quot;. </p>
</li>
<li><p> If <code>character</code>,
allowed values include: i) one of c('right', 'left', 'center', 'centre',
'middle') for x-axis; ii) and one of c( 'bottom', 'top', 'center', 'centre',
'middle') for y-axis.</p>
</li></ul>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_label.x">label.x</code>, <code id="stat_compare_means_+3A_label.y">label.y</code></td>
<td>
<p><code>numeric</code> Coordinates (in data units) to be used
for absolute positioning of the label. If too short they will be recycled.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_vjust">vjust</code></td>
<td>
<p>move the text up or down relative to the bracket.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_tip.length">tip.length</code></td>
<td>
<p>numeric vector with the fraction of total height that the
bar goes down to indicate the precise column. Default is 0.03. Can be of
same length as the number of comparisons to adjust specifically the tip
lenth of each comparison. For example tip.length = c(0.01, 0.03).
</p>
<p>If too short they will be recycled.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_bracket.size">bracket.size</code></td>
<td>
<p>Width of the lines of the bracket.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_step.increase">step.increase</code></td>
<td>
<p>numeric vector with the increase in fraction of total
height for every additional comparison to minimize overlap.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_symnum.args">symnum.args</code></td>
<td>
<p>a list of arguments to pass to the function
<code><a href="stats.html#topic+symnum">symnum</a></code> for symbolic number coding of p-values. For
example, <code>symnum.args &lt;- list(cutpoints = c(0, 0.0001, 0.001,
 0.01, 0.05, Inf), symbols = c("****", "***", "**", "*",  "ns"))</code>.
</p>
<p>In other words, we use the following convention for symbols indicating
statistical significance: </p>
 <ul>
<li> <p><code>ns</code>: p &gt; 0.05 </p>
</li>
<li>
<p><code>*</code>: p &lt;= 0.05 </p>
</li>
<li> <p><code>**</code>: p &lt;= 0.01 </p>
</li>
<li> <p><code>***</code>: p &lt;= 0.001 </p>
</li>
<li> <p><code>****</code>:  p &lt;= 0.0001 </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_na.rm">na.rm</code></td>
<td>
<p>If FALSE (the default), removes missing values with a warning. If
TRUE silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_compare_means_+3A_...">...</code></td>
<td>
<p>other arguments to pass to <code><a href="ggplot2.html#topic+geom_text">geom_text</a></code> or
<code><a href="ggplot2.html#topic+geom_text">geom_label</a></code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+compare_means">compare_means</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("ToothGrowth")
head(ToothGrowth)

# Two independent groups
#:::::::::::::::::::::::::::::::::::::::::::::::::
p &lt;- ggboxplot(ToothGrowth, x = "supp", y = "len",
  color = "supp", palette = "npg", add = "jitter")

#  Add p-value
p + stat_compare_means()
# Change method
p + stat_compare_means(method = "t.test")

 # Paired samples
 #:::::::::::::::::::::::::::::::::::::::::::::::::
 ggpaired(ToothGrowth, x = "supp", y = "len",
   color = "supp", line.color = "gray", line.size = 0.4,
   palette = "npg")+
 stat_compare_means(paired = TRUE)

# More than two groups
#:::::::::::::::::::::::::::::::::::::::::::::::::
# Pairwise comparisons: Specify the comparisons you want
my_comparisons &lt;- list( c("0.5", "1"), c("1", "2"), c("0.5", "2") )
ggboxplot(ToothGrowth, x = "dose", y = "len",
          color = "dose", palette = "npg")+
# Add pairwise comparisons p-value
stat_compare_means(comparisons = my_comparisons, label.y = c(29, 35, 40))+
stat_compare_means(label.y = 45)     # Add global Anova p-value

# Multiple pairwise test against a reference group
ggboxplot(ToothGrowth, x = "dose", y = "len",
    color = "dose", palette = "npg")+
stat_compare_means(method = "anova", label.y = 40)+ # Add global p-value
stat_compare_means(aes(label = after_stat(p.signif)),
                  method = "t.test", ref.group = "0.5")

# Multiple grouping variables
#:::::::::::::::::::::::::::::::::::::::::::::::::
# Box plot facetted by "dose"
p &lt;- ggboxplot(ToothGrowth, x = "supp", y = "len",
              color = "supp", palette = "npg",
              add = "jitter",
              facet.by = "dose", short.panel.labs = FALSE)
# Use only p.format as label. Remove method name.
p + stat_compare_means(
 aes(label = paste0("p = ", after_stat(p.format)))
)

</code></pre>

<hr>
<h2 id='stat_conf_ellipse'>Plot confidence ellipses.</h2><span id='topic+stat_conf_ellipse'></span>

<h3>Description</h3>

<p>Plot confidence ellipses around barycenters. The method for
computing confidence ellipses has been modified from <code>FactoMineR::coord.ellipse()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_conf_ellipse(
  mapping = NULL,
  data = NULL,
  geom = "path",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  level = 0.95,
  npoint = 100,
  bary = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_conf_ellipse_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_conf_ellipse_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_conf_ellipse_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_conf_ellipse_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_conf_ellipse_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p>
</td></tr>
<tr><td><code id="stat_conf_ellipse_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_conf_ellipse_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_conf_ellipse_+3A_level">level</code></td>
<td>
<p>confidence level used to construct the ellipses. By
default, 0.95.</p>
</td></tr>
<tr><td><code id="stat_conf_ellipse_+3A_npoint">npoint</code></td>
<td>
<p>number of points used to draw the ellipses.</p>
</td></tr>
<tr><td><code id="stat_conf_ellipse_+3A_bary">bary</code></td>
<td>
<p>logical value. If TRUE, the coordinates of the ellipse around the
barycentre of individuals are calculated.</p>
</td></tr>
<tr><td><code id="stat_conf_ellipse_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggplot2.html#topic+layer">layer()</a></code>. These are
often aesthetics, used to set an aesthetic to a fixed value, like
<code>colour = "red"</code> or <code>size = 3</code>. They may also be parameters
to the paired geom/stat.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+stat_conf_ellipse">stat_conf_ellipse</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("mtcars")
df &lt;- mtcars
df$cyl &lt;- as.factor(df$cyl)

# scatter plot with confidence ellipses
ggscatter(df, x = "wt", y = "mpg", color = "cyl")+
 stat_conf_ellipse(aes(color = cyl))

ggscatter(df, x = "wt", y = "mpg", color = "cyl")+
 stat_conf_ellipse(aes(color = cyl, fill = cyl), alpha = 0.1, geom = "polygon")

</code></pre>

<hr>
<h2 id='stat_cor'>Add Correlation Coefficients with P-values to a Scatter Plot</h2><span id='topic+stat_cor'></span>

<h3>Description</h3>

<p>Add correlation coefficients with p-values to a scatter plot. Can
be also used to add 'R2'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_cor(
  mapping = NULL,
  data = NULL,
  method = "pearson",
  alternative = "two.sided",
  cor.coef.name = c("R", "rho", "tau"),
  label.sep = ", ",
  label.x.npc = "left",
  label.y.npc = "top",
  label.x = NULL,
  label.y = NULL,
  output.type = "expression",
  digits = 2,
  r.digits = digits,
  p.digits = digits,
  r.accuracy = NULL,
  p.accuracy = NULL,
  geom = "text",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_cor_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_method">method</code></td>
<td>
<p>a character string indicating which correlation coefficient (or
covariance) is to be computed. One of &quot;pearson&quot; (default), &quot;kendall&quot;, or
&quot;spearman&quot;.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_alternative">alternative</code></td>
<td>
<p>a character string specifying the alternative hypothesis,
must be one of &quot;two.sided&quot; (default), &quot;greater&quot; or &quot;less&quot;. You can specify
just the initial letter.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_cor.coef.name">cor.coef.name</code></td>
<td>
<p>character. Can be one of <code>"R"</code> (pearson coef),
<code>"rho"</code> (spearman coef) and <code>"tau"</code> (kendall coef). Uppercase and
lowercase are allowed.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_label.sep">label.sep</code></td>
<td>
<p>a character string to separate the terms. Default is &quot;, &quot;, to
separate the correlation coefficient and the p.value.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_label.x.npc">label.x.npc</code>, <code id="stat_cor_+3A_label.y.npc">label.y.npc</code></td>
<td>
<p>can be <code>numeric</code> or <code>character</code>
vector of the same length as the number of groups and/or panels. If too
short they will be recycled. </p>
 <ul>
<li><p> If <code>numeric</code>, value should
be between 0 and 1. Coordinates to be used for positioning the label,
expressed in &quot;normalized parent coordinates&quot;. </p>
</li>
<li><p> If <code>character</code>,
allowed values include: i) one of c('right', 'left', 'center', 'centre',
'middle') for x-axis; ii) and one of c( 'bottom', 'top', 'center', 'centre',
'middle') for y-axis.</p>
</li></ul>

<p>If too short they will be recycled.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_label.x">label.x</code>, <code id="stat_cor_+3A_label.y">label.y</code></td>
<td>
<p><code>numeric</code> Coordinates (in data units) to be used
for absolute positioning of the label. If too short they will be recycled.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_output.type">output.type</code></td>
<td>
<p>character One of &quot;expression&quot;, &quot;latex&quot;, &quot;tex&quot; or &quot;text&quot;.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_digits">digits</code>, <code id="stat_cor_+3A_r.digits">r.digits</code>, <code id="stat_cor_+3A_p.digits">p.digits</code></td>
<td>
<p>integer indicating the number of decimal
places (round) or significant digits (signif) to be used for the correlation
coefficient and the p-value, respectively..</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_r.accuracy">r.accuracy</code></td>
<td>
<p>a real value specifying the number of decimal places of
precision for the correlation coefficient. Default is NULL. Use (e.g.) 0.01
to show 2 decimal places of precision. If specified, then <code>r.digits</code> is
ignored.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_p.accuracy">p.accuracy</code></td>
<td>
<p>a real value specifying the number of decimal places of
precision for the p-value. Default is NULL. Use (e.g.) 0.0001 to show 4
decimal places of precision. If specified, then <code>p.digits</code> is ignored.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_na.rm">na.rm</code></td>
<td>
<p>If FALSE (the default), removes missing values with a warning. If
TRUE silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_cor_+3A_...">...</code></td>
<td>
<p>other arguments to pass to <code><a href="ggplot2.html#topic+geom_text">geom_text</a></code> or
<code><a href="ggplot2.html#topic+geom_text">geom_label</a></code>.</p>
</td></tr>
</table>


<h3>Computed variables</h3>

 <dl>
<dt>r</dt><dd><p>correlation coefficient</p>
</dd>
<dt>rr</dt><dd><p>correlation coefficient squared</p>
</dd> <dt>r.label</dt><dd><p>formatted label
for the correlation coefficient</p>
</dd> <dt>rr.label</dt><dd><p>formatted label for the
squared correlation coefficient</p>
</dd> <dt>p.label</dt><dd><p>label for the p-value</p>
</dd>
<dt>label</dt><dd><p>default labeldisplayed by <code>stat_cor()</code></p>
</dd> </dl>



<h3>See Also</h3>

<p><code><a href="#topic+ggscatter">ggscatter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("mtcars")
df &lt;- mtcars
df$cyl &lt;- as.factor(df$cyl)

# Scatter plot with correlation coefficient
#:::::::::::::::::::::::::::::::::::::::::::::::::
sp &lt;- ggscatter(df, x = "wt", y = "mpg",
   add = "reg.line",  # Add regressin line
   add.params = list(color = "blue", fill = "lightgray"), # Customize reg. line
   conf.int = TRUE # Add confidence interval
   )
# Add correlation coefficient
sp + stat_cor(method = "pearson", label.x = 3, label.y = 30)

# Specify the number of decimal places of precision for p and r
# Using 3 decimal places for the p-value and
# 2 decimal places for the correlation coefficient (r)
sp + stat_cor(p.accuracy = 0.001, r.accuracy = 0.01)

# Show only the r.label but not the p.label
sp + stat_cor(aes(label = ..r.label..), label.x = 3)

# Use R2 instead of R
ggscatter(df, x = "wt", y = "mpg", add = "reg.line") +
 stat_cor(
   aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~")),
  label.x = 3
)

# Color by groups and facet
#::::::::::::::::::::::::::::::::::::::::::::::::::::
sp &lt;- ggscatter(df, x = "wt", y = "mpg",
   color = "cyl", palette = "jco",
   add = "reg.line", conf.int = TRUE)
sp + stat_cor(aes(color = cyl), label.x = 3)

</code></pre>

<hr>
<h2 id='stat_friedman_test'>Add Friedman Test P-values to a GGPlot</h2><span id='topic+stat_friedman_test'></span>

<h3>Description</h3>

<p>Add automatically Friedman test p-values to a ggplot, such as
box blots, dot plots and stripcharts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_friedman_test(
  mapping = NULL,
  data = NULL,
  wid = NULL,
  group.by = NULL,
  label = "{method}, p = {p.format}",
  label.x.npc = "left",
  label.y.npc = "top",
  label.x = NULL,
  label.y = NULL,
  step.increase = 0.1,
  p.adjust.method = "holm",
  significance = list(),
  geom = "text",
  position = "identity",
  na.rm = FALSE,
  show.legend = FALSE,
  inherit.aes = TRUE,
  parse = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_friedman_test_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_wid">wid</code></td>
<td>
<p>(factor) column name containing individuals/subjects identifier.
Should be unique per individual. Required only for repeated measure tests
(<code>"one_way_repeated", "two_way_repeated", "friedman_test", etc</code>).</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_group.by">group.by</code></td>
<td>
<p>(optional) character vector specifying the grouping variable;
it should be used only for grouped plots. Possible values are : </p>

<ul>
<li> <p><code>"x.var"</code>: Group by the x-axis variable and perform the test
between legend groups. In other words, the p-value is compute between legend
groups at each x position </p>
</li>
<li> <p><code>"legend.var"</code>: Group by the legend
variable and perform the test between x-axis groups. In other words, the
test is performed between the x-groups for each legend level. </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_label">label</code></td>
<td>
<p>the column containing the label (e.g.: label = &quot;p&quot; or label =
&quot;p.adj&quot;), where <code>p</code> is the p-value. Can be also an expression that can
be formatted by the <code><a href="glue.html#topic+glue">glue</a>()</code> package. For example, when
specifying label = &quot;t-test, p = {p}&quot;, the expression {p} will be
replaced by its value.</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_label.x.npc">label.x.npc</code>, <code id="stat_friedman_test_+3A_label.y.npc">label.y.npc</code></td>
<td>
<p>can be <code>numeric</code> or <code>character</code>
vector of the same length as the number of groups and/or panels. If too
short they will be recycled. </p>
 <ul>
<li><p> If <code>numeric</code>, value should
be between 0 and 1. Coordinates to be used for positioning the label,
expressed in &quot;normalized parent coordinates&quot;. </p>
</li>
<li><p> If <code>character</code>,
allowed values include: i) one of c('right', 'left', 'center', 'centre',
'middle') for x-axis; ii) and one of c( 'bottom', 'top', 'center', 'centre',
'middle') for y-axis.</p>
</li></ul>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_label.x">label.x</code>, <code id="stat_friedman_test_+3A_label.y">label.y</code></td>
<td>
<p><code>numeric</code> Coordinates (in data units) to be used
for absolute positioning of the label. If too short they will be recycled.</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_step.increase">step.increase</code></td>
<td>
<p>numeric vector with the increase in fraction of total
height for every additional comparison to minimize overlap.</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for adjusting p values (see
<code><a href="stats.html#topic+p.adjust">p.adjust</a></code>).  Has impact only in a situation, where
multiple pairwise tests are performed; or when there are multiple grouping
variables. Allowed values include &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;,
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;. If you don't want to adjust the p
value (not recommended), use p.adjust.method = &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_significance">significance</code></td>
<td>
<p>a list of arguments specifying the signifcance cutpoints
and symbols. For example, <code>significance &lt;- list(cutpoints = c(0,
 0.0001, 0.001, 0.01, 0.05, Inf), symbols = c("****", "***", "**", "*",
 "ns"))</code>.
</p>
<p>In other words, we use the following convention for symbols indicating
statistical significance: </p>
 <ul>
<li> <p><code>ns</code>: p &gt; 0.05 </p>
</li>
<li>
<p><code>*</code>: p &lt;= 0.05 </p>
</li>
<li> <p><code>**</code>: p &lt;= 0.01 </p>
</li>
<li> <p><code>***</code>: p &lt;= 0.001
</p>
</li>
<li> <p><code>****</code>:  p &lt;= 0.0001 </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_na.rm">na.rm</code></td>
<td>
<p>If FALSE (the default), removes missing values with a warning. If
TRUE silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_parse">parse</code></td>
<td>
<p>If TRUE, the labels will be parsed into expressions and displayed
as described in <code>?plotmath</code>.</p>
</td></tr>
<tr><td><code id="stat_friedman_test_+3A_...">...</code></td>
<td>
<p>other arguments passed to the function <code>geom_bracket()</code> or
<code>geom_text()</code></p>
</td></tr>
</table>


<h3>Computed variables</h3>


<ul>
<li><p>statistic:	the value of the test statistic (Chi-squared).
</p>
</li>
<li><p>df: the degrees of freedom of the approximate chi-squared distribution of the test statistic.
</p>
</li>
<li><p>p:	p-value.
</p>
</li>
<li><p>p.adj: Adjusted p-values.
</p>
</li>
<li><p>p.signif: P-value significance.
</p>
</li>
<li><p>p.adj.signif: Adjusted p-value significance.
</p>
</li>
<li><p>p.format: Formated p-value.
</p>
</li>
<li><p>p.adj.format: Formated adjusted p-value.
</p>
</li>
<li><p>n: number of samples.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Data preparation
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Transform `dose` into factor variable
df &lt;- ToothGrowth
df$dose &lt;- as.factor(df$dose)
df$id &lt;- as.factor(c(rep(1:10, 3), rep(11:20, 3)))
# Add a random grouping variable
set.seed(123)
df$group &lt;- sample(factor(rep(c("grp1", "grp2", "grp3"), 20)))
df$len &lt;- ifelse(df$group == "grp2", df$len+2, df$len)
df$len &lt;- ifelse(df$group == "grp3", df$len+7, df$len)
head(df, 3)


# Basic boxplot
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Create a basic boxplot
# Add 5% and 10% space to the plot bottom and the top, respectively
bxp &lt;- ggboxplot(df, x = "dose", y = "len") +
  scale_y_continuous(expand = expansion(mult = c(0.05, 0.1)))

# Add the p-value to the boxplot
bxp + stat_friedman_test(aes(wid = id))

# Change the label position
# Using coordinates in data units
bxp + stat_friedman_test(aes(wid = id), label.x = "1", label.y = 10, hjust = 0)

# Format the p-value differently
custom_p_format &lt;- function(p) {
  rstatix::p_format(p, accuracy = 0.0001, digits = 3, leading.zero = FALSE)
}
bxp + stat_friedman_test(
  aes(wid = id),
  label = "Friedman test, italic(p) = {custom_p_format(p)}{p.signif}"
)

# Show a detailed label in italic
bxp + stat_friedman_test(aes(wid = id), label = "as_detailed_italic")


# Faceted plots
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Create a ggplot facet
df$id &lt;- rep(1:10,6)
bxp &lt;- ggboxplot(df, x = "dose", y = "len", facet.by = "supp") +
 scale_y_continuous(expand = expansion(mult = c(0.05, 0.1)))
# Add p-values
bxp + stat_friedman_test(aes(wid = id))


# Grouped plots
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
df$id &lt;- rep(1:10,6)
bxp &lt;- ggboxplot(df, x = "dose", y = "len", color = "supp", palette = "jco")

# For each legend group, computes tests within x variable groups
bxp + stat_friedman_test(aes(wid = id, group = supp, color = supp), within = "x")

# For each x-position, computes tests within legend variable groups
bxp + stat_friedman_test(
  aes(wid = id, group = supp, color = supp),
  within =  "group", label = "p = {p.format}"
)

</code></pre>

<hr>
<h2 id='stat_kruskal_test'>Add Kruskal-Wallis Test P-values to a GGPlot</h2><span id='topic+stat_kruskal_test'></span>

<h3>Description</h3>

<p>Add Kruskal-Wallis test p-values to a ggplot, such as
box blots, dot plots and stripcharts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_kruskal_test(
  mapping = NULL,
  data = NULL,
  group.by = NULL,
  label = "{method}, p = {p.format}",
  label.x.npc = "left",
  label.y.npc = "top",
  label.x = NULL,
  label.y = NULL,
  step.increase = 0.1,
  p.adjust.method = "holm",
  significance = list(),
  geom = "text",
  position = "identity",
  na.rm = FALSE,
  show.legend = FALSE,
  inherit.aes = TRUE,
  parse = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_kruskal_test_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_group.by">group.by</code></td>
<td>
<p>(optional) character vector specifying the grouping variable;
it should be used only for grouped plots. Possible values are : </p>

<ul>
<li> <p><code>"x.var"</code>: Group by the x-axis variable and perform the test
between legend groups. In other words, the p-value is compute between legend
groups at each x position </p>
</li>
<li> <p><code>"legend.var"</code>: Group by the legend
variable and perform the test between x-axis groups. In other words, the
test is performed between the x-groups for each legend level. </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_label">label</code></td>
<td>
<p>the column containing the label (e.g.: label = &quot;p&quot; or label =
&quot;p.adj&quot;), where <code>p</code> is the p-value. Can be also an expression that can
be formatted by the <code><a href="glue.html#topic+glue">glue</a>()</code> package. For example, when
specifying label = &quot;t-test, p = {p}&quot;, the expression {p} will be
replaced by its value.</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_label.x.npc">label.x.npc</code>, <code id="stat_kruskal_test_+3A_label.y.npc">label.y.npc</code></td>
<td>
<p>can be <code>numeric</code> or <code>character</code>
vector of the same length as the number of groups and/or panels. If too
short they will be recycled. </p>
 <ul>
<li><p> If <code>numeric</code>, value should
be between 0 and 1. Coordinates to be used for positioning the label,
expressed in &quot;normalized parent coordinates&quot;. </p>
</li>
<li><p> If <code>character</code>,
allowed values include: i) one of c('right', 'left', 'center', 'centre',
'middle') for x-axis; ii) and one of c( 'bottom', 'top', 'center', 'centre',
'middle') for y-axis.</p>
</li></ul>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_label.x">label.x</code>, <code id="stat_kruskal_test_+3A_label.y">label.y</code></td>
<td>
<p><code>numeric</code> Coordinates (in data units) to be used
for absolute positioning of the label. If too short they will be recycled.</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_step.increase">step.increase</code></td>
<td>
<p>numeric vector with the increase in fraction of total
height for every additional comparison to minimize overlap.</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for adjusting p values (see
<code><a href="stats.html#topic+p.adjust">p.adjust</a></code>).  Has impact only in a situation, where
multiple pairwise tests are performed; or when there are multiple grouping
variables. Allowed values include &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;,
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;. If you don't want to adjust the p
value (not recommended), use p.adjust.method = &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_significance">significance</code></td>
<td>
<p>a list of arguments specifying the signifcance cutpoints
and symbols. For example, <code>significance &lt;- list(cutpoints = c(0,
 0.0001, 0.001, 0.01, 0.05, Inf), symbols = c("****", "***", "**", "*",
 "ns"))</code>.
</p>
<p>In other words, we use the following convention for symbols indicating
statistical significance: </p>
 <ul>
<li> <p><code>ns</code>: p &gt; 0.05 </p>
</li>
<li>
<p><code>*</code>: p &lt;= 0.05 </p>
</li>
<li> <p><code>**</code>: p &lt;= 0.01 </p>
</li>
<li> <p><code>***</code>: p &lt;= 0.001
</p>
</li>
<li> <p><code>****</code>:  p &lt;= 0.0001 </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_na.rm">na.rm</code></td>
<td>
<p>If FALSE (the default), removes missing values with a warning. If
TRUE silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_parse">parse</code></td>
<td>
<p>If TRUE, the labels will be parsed into expressions and displayed
as described in <code>?plotmath</code>.</p>
</td></tr>
<tr><td><code id="stat_kruskal_test_+3A_...">...</code></td>
<td>
<p>other arguments passed to the function <code>geom_bracket()</code> or
<code>geom_text()</code></p>
</td></tr>
</table>


<h3>Computed variables</h3>


<ul>
<li><p>statistic:	the Kruskal-Wallis rank sum chi-squared statistic used to compute the p-value.
</p>
</li>
<li><p>p:	p-value.
</p>
</li>
<li><p>p.adj: Adjusted p-values.
</p>
</li>
<li><p>p.signif: P-value significance.
</p>
</li>
<li><p>p.adj.signif: Adjusted p-value significance.
</p>
</li>
<li><p>p.format: Formated p-value.
</p>
</li>
<li><p>p.adj.format: Formated adjusted p-value.
</p>
</li>
<li><p>n: number of samples.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Data preparation
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Transform `dose` into factor variable
df &lt;- ToothGrowth
df$dose &lt;- as.factor(df$dose)
# Add a random grouping variable
set.seed(123)
df$group &lt;- sample(factor(rep(c("grp1", "grp2", "grp3"), 20)))
df$len &lt;- ifelse(df$group == "grp2", df$len+2, df$len)
df$len &lt;- ifelse(df$group == "grp3", df$len+7, df$len)
head(df, 3)


# Basic boxplot
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Create a basic boxplot
# Add 5% and 10% space to the plot bottom and the top, respectively
bxp &lt;- ggboxplot(df, x = "dose", y = "len") +
  scale_y_continuous(expand = expansion(mult = c(0.05, 0.1)))

# Add the p-value to the boxplot
bxp + stat_kruskal_test()

# Change the label position
# Using coordinates in data units
bxp + stat_kruskal_test(label.x = "1", label.y = 10, hjust = 0)

# Format the p-value differently
custom_p_format &lt;- function(p) {
  rstatix::p_format(p, accuracy = 0.0001, digits = 3, leading.zero = FALSE)
}
bxp + stat_kruskal_test(
  label = "Kruskal-Wallis, italic(p) = {custom_p_format(p)}{p.signif}"
)

# Show a detailed label in italic
bxp + stat_kruskal_test(label = "as_detailed_italic")


# Faceted plots
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Create a ggplot facet
bxp &lt;- ggboxplot(df, x = "dose", y = "len", facet.by = "supp") +
 scale_y_continuous(expand = expansion(mult = c(0.05, 0.1)))
# Add p-values
bxp + stat_kruskal_test()


# Grouped plots
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
bxp2 &lt;- ggboxplot(df, x = "group", y = "len", color = "dose", palette = "npg")

# For each x-position, computes tests between legend groups
bxp2 + stat_kruskal_test(aes(group = dose), label = "p = {p.format}{p.signif}")

#  For each legend group, computes tests between x variable groups
bxp2 + stat_kruskal_test(aes(group = dose, color = dose), group.by = "legend.var")

</code></pre>

<hr>
<h2 id='stat_mean'>Draw group mean points</h2><span id='topic+stat_mean'></span>

<h3>Description</h3>

<p>Draw the mean point of each group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_mean(
  mapping = NULL,
  data = NULL,
  geom = "point",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_mean_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_mean_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_mean_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_mean_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_mean_+3A_na.rm">na.rm</code></td>
<td>
<p>If FALSE (the default), removes missing values with a warning.
If TRUE silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_mean_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_mean_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_mean_+3A_...">...</code></td>
<td>
<p>other arguments to pass to <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+stat_conf_ellipse">stat_conf_ellipse</a></code>, <code><a href="#topic+stat_chull">stat_chull</a></code> and
<code><a href="#topic+ggscatter">ggscatter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("mtcars")
df &lt;- mtcars
df$cyl &lt;- as.factor(df$cyl)

# Scatter plot with ellipses and group mean points
ggscatter(df, x = "wt", y = "mpg",
   color = "cyl", shape = "cyl", ellipse = TRUE)+
 stat_mean(aes(color = cyl, shape = cyl), size = 4)

</code></pre>

<hr>
<h2 id='stat_overlay_normal_density'>Overlay Normal Density Plot</h2><span id='topic+stat_overlay_normal_density'></span>

<h3>Description</h3>

<p>Overlay normal density plot (with the same mean and SD) to the
density distribution of 'x'. This is useful for visually inspecting the
degree of deviance from normality.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_overlay_normal_density(
  mapping = NULL,
  data = NULL,
  geom = "line",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_overlay_normal_density_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_overlay_normal_density_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_overlay_normal_density_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_overlay_normal_density_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_overlay_normal_density_+3A_na.rm">na.rm</code></td>
<td>
<p>If FALSE (the default), removes missing values with a warning.
If TRUE silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_overlay_normal_density_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_overlay_normal_density_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_overlay_normal_density_+3A_...">...</code></td>
<td>
<p>other arguments to pass to <code><a href="ggplot2.html#topic+geom_path">geom_line</a></code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ggdensity">ggdensity</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simpledensity plot
data("mtcars")
ggdensity(mtcars, x = "mpg", fill = "red") +
  scale_x_continuous(limits = c(-1, 50)) +
  stat_overlay_normal_density(color = "red", linetype = "dashed")

# Color by groups
data(iris)
ggdensity(iris, "Sepal.Length", color = "Species") +
 stat_overlay_normal_density(aes(color = Species), linetype = "dashed")


# Facet
ggdensity(iris, "Sepal.Length", facet.by = "Species") +
 stat_overlay_normal_density(color = "red", linetype = "dashed")

</code></pre>

<hr>
<h2 id='stat_pvalue_manual'>Add Manually P-values to a ggplot</h2><span id='topic+stat_pvalue_manual'></span>

<h3>Description</h3>

<p>Add manually p-values to a ggplot, such as box blots, dot plots
and stripcharts. Frequently asked questions are available on <a href="https://www.datanovia.com/en/blog/tag/ggpubr/">Datanovia ggpubr FAQ page</a>, for example:
</p>

<ul>
<li> <p><a href="https://www.datanovia.com/en/blog/how-to-add-p-values-onto-basic-ggplots/">How to Add P-Values onto Basic GGPLOTS</a>
</p>
</li>
<li> <p><a href="https://www.datanovia.com/en/blog/ggpubr-how-to-add-adjusted-p-values-to-a-multi-panel-ggplot/">How to Add Adjusted P-values to a Multi-Panel GGPlot</a>
</p>
</li>
<li> <p><a href="https://www.datanovia.com/en/blog/how-to-add-p-values-to-ggplot-facets/">How to Add P-values to GGPLOT Facets</a>
</p>
</li>
<li> <p><a href="https://www.datanovia.com/en/blog/ggpubr-how-to-add-p-values-generated-elsewhere-to-a-ggplot/">How to Add P-Values Generated Elsewhere to a GGPLOT</a>
</p>
</li>
<li> <p><a href="https://www.datanovia.com/en/blog/how-to-add-p-values-onto-a-grouped-ggplot-using-the-ggpubr-r-package/">How to Add P-Values onto a Grouped GGPLOT using the GGPUBR R Package</a>
</p>
</li>
<li> <p><a href="https://www.datanovia.com/en/blog/how-to-create-stacked-bar-plots-with-error-bars-and-p-values/">How to Create Stacked Bar Plots with Error Bars and P-values</a>
</p>
</li>
<li> <p><a href="https://www.datanovia.com/en/blog/how-to-add-p-values-onto-horizontal-ggplots/">How to Add P-Values onto Horizontal GGPLOTS</a>
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>stat_pvalue_manual(
  data,
  label = NULL,
  y.position = "y.position",
  xmin = "group1",
  xmax = "group2",
  x = NULL,
  size = 3.88,
  label.size = size,
  bracket.size = 0.3,
  bracket.nudge.y = 0,
  bracket.shorten = 0,
  color = "black",
  linetype = 1,
  tip.length = 0.03,
  remove.bracket = FALSE,
  step.increase = 0,
  step.group.by = NULL,
  hide.ns = FALSE,
  vjust = 0,
  coord.flip = FALSE,
  position = "identity",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_pvalue_manual_+3A_data">data</code></td>
<td>
<p>a data frame containing statitistical test results. The expected
default format should contain the following columns: <code>group1 | group2 |
p | y.position | etc</code>. <code>group1</code> and <code>group2</code> are the groups that
have been compared. <code>p</code> is the resulting p-value. <code>y.position</code> is
the y coordinates of the p-values in the plot.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_label">label</code></td>
<td>
<p>the column containing the label (e.g.: label = &quot;p&quot; or label =
&quot;p.adj&quot;), where <code>p</code> is the p-value. Can be also an expression that can
be formatted by the <code><a href="glue.html#topic+glue">glue</a>()</code> package. For example, when
specifying label = &quot;t-test, p = {p}&quot;, the expression {p} will be
replaced by its value.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_y.position">y.position</code></td>
<td>
<p>column containing the coordinates (in data units) to be used
for absolute positioning of the label. Default value is &quot;y.position&quot;. Can be
also a numeric vector.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_xmin">xmin</code></td>
<td>
<p>column containing the position of the left sides of the brackets.
Default value is &quot;group1&quot;.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_xmax">xmax</code></td>
<td>
<p>(optional) column containing the position of the right sides of
the brackets. Default value is &quot;group2&quot;. If NULL, the p-values are plotted
as a simple text.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_x">x</code></td>
<td>
<p>x position of the p-value. Should be used only when you want plot the
p-value as text (without brackets).</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_size">size</code>, <code id="stat_pvalue_manual_+3A_label.size">label.size</code></td>
<td>
<p>size of label text.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_bracket.size">bracket.size</code></td>
<td>
<p>Width of the lines of the bracket.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_bracket.nudge.y">bracket.nudge.y</code></td>
<td>
<p>Vertical adjustment to nudge brackets by. Useful to
move up or move down the bracket. If positive value, brackets will be moved
up; if negative value, brackets are moved down.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_bracket.shorten">bracket.shorten</code></td>
<td>
<p>a small numeric value in [0-1] for shortening the with
of bracket.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_color">color</code></td>
<td>
<p>text and line color. Can be variable name in the data for coloring by groups.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_linetype">linetype</code></td>
<td>
<p>linetype. Can be variable name in the data for changing linetype by groups.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_tip.length">tip.length</code></td>
<td>
<p>numeric vector with the fraction of total height that the
bar goes down to indicate the precise column. Default is 0.03.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_remove.bracket">remove.bracket</code></td>
<td>
<p>logical, if <code>TRUE</code>, brackets are removed from the
plot. Considered only in the situation, where comparisons are performed
against reference group or against &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_step.increase">step.increase</code></td>
<td>
<p>numeric vector with the increase in fraction of total
height for every additional comparison to minimize overlap.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_step.group.by">step.group.by</code></td>
<td>
<p>a variable name for grouping brackets before adding
step.increase. Useful to group bracket by facet panel.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_hide.ns">hide.ns</code></td>
<td>
<p>can be logical value or a character vector.
</p>

<ul>
<li><p> Case when logical value. If TRUE, hide ns symbol when displaying
significance levels. Filter is done by checking the column
<code>p.adj.signif</code>, <code>p.signif</code>, <code>p.adj</code> and <code>p</code>.
</p>
</li>
<li><p> Case when character value. Possible values are &quot;p&quot; or &quot;p.adj&quot;, for filtering out non significant.
</p>
</li></ul>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_vjust">vjust</code></td>
<td>
<p>move the text up or down relative to the bracket. Can be also a
column name available in the data.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_coord.flip">coord.flip</code></td>
<td>
<p>logical. If <code>TRUE</code>, flip x and y coordinates so that
horizontal becomes vertical, and vertical, horizontal. When adding the
p-values to a horizontal ggplot (generated using
<code><a href="ggplot2.html#topic+coord_flip">coord_flip</a>()</code>), you need to specify the option
<code>coord.flip = TRUE</code>.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_position">position</code></td>
<td>
<p>position adjustment, either as a string, or the result of a
call to a position adjustment function.</p>
</td></tr>
<tr><td><code id="stat_pvalue_manual_+3A_...">...</code></td>
<td>
<p>other arguments passed to the function <code>geom_bracket()</code> or
<code>geom_text()</code></p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+stat_compare_means">stat_compare_means</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# T-test
stat.test &lt;- compare_means(
 len ~ dose, data = ToothGrowth,
 method = "t.test"
)
stat.test

# Create a simple box plot
p &lt;- ggboxplot(ToothGrowth, x = "dose", y = "len")
p

# Perform a t-test between groups
stat.test &lt;- compare_means(
 len ~ dose, data = ToothGrowth,
 method = "t.test"
)
stat.test

# Add manually p-values from stat.test data
# First specify the y.position of each comparison
stat.test &lt;- stat.test %&gt;%
 mutate(y.position = c(29, 35, 39))
p + stat_pvalue_manual(stat.test, label = "p.adj")

# Customize the label with glue expression
# (https://github.com/tidyverse/glue)
p + stat_pvalue_manual(stat.test, label = "p = {p.adj}")


# Grouped bar plots
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
ToothGrowth$dose &lt;- as.factor(ToothGrowth$dose)
# Comparisons against reference
stat.test &lt;- compare_means(
  len ~ dose, data = ToothGrowth, group.by = "supp",
  method = "t.test", ref.group = "0.5"
)
stat.test
# Plot
bp &lt;- ggbarplot(ToothGrowth, x = "supp", y = "len",
                fill = "dose", palette = "jco",
                add = "mean_sd", add.params = list(group = "dose"),
                position = position_dodge(0.8))
bp + stat_pvalue_manual(
  stat.test, x = "supp", y.position = 33,
  label = "p.signif",
  position = position_dodge(0.8)
)

</code></pre>

<hr>
<h2 id='stat_pwc'>Add Pairwise Comparisons P-values to a GGPlot</h2><span id='topic+stat_pwc'></span><span id='topic+geom_pwc'></span>

<h3>Description</h3>

<p>add pairwise comparison p-values to a ggplot such as box plots,
dot plots and stripcharts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_pwc(
  mapping = NULL,
  data = NULL,
  method = "wilcox_test",
  method.args = list(),
  ref.group = NULL,
  label = "p.format",
  y.position = NULL,
  group.by = NULL,
  dodge = 0.8,
  bracket.nudge.y = 0.05,
  bracket.shorten = 0,
  bracket.group.by = c("x.var", "legend.var"),
  step.increase = 0.12,
  tip.length = 0.03,
  size = 0.3,
  label.size = 3.88,
  family = "",
  vjust = 0,
  hjust = 0.5,
  p.adjust.method = "holm",
  p.adjust.by = c("group", "panel"),
  symnum.args = list(),
  hide.ns = FALSE,
  remove.bracket = FALSE,
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  parse = FALSE,
  ...
)

geom_pwc(
  mapping = NULL,
  data = NULL,
  stat = "pwc",
  method = "wilcox_test",
  method.args = list(),
  ref.group = NULL,
  label = "p.format",
  y.position = NULL,
  group.by = NULL,
  dodge = 0.8,
  stack = FALSE,
  step.increase = 0.12,
  tip.length = 0.03,
  bracket.nudge.y = 0.05,
  bracket.shorten = 0,
  bracket.group.by = c("x.var", "legend.var"),
  size = 0.3,
  label.size = 3.88,
  family = "",
  vjust = 0,
  hjust = 0.5,
  p.adjust.method = "holm",
  p.adjust.by = c("group", "panel"),
  symnum.args = list(),
  hide.ns = FALSE,
  remove.bracket = FALSE,
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  parse = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_pwc_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_method">method</code></td>
<td>
<p>a character string indicating which method to be used for
pairwise comparisons. Default is <code>"wilcox_test"</code>. Allowed methods
include pairwise comparisons methods implemented in the <code>rstatix</code> R
package. These methods are: <code>"wilcox_test", "t_test", "sign_test",
"dunn_test", "emmeans_test", "tukey_hsd", "games_howell_test"</code>.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_method.args">method.args</code></td>
<td>
<p>a list of additional arguments used for the test method.
For example one might use <code>method.args = list(alternative = "greater")</code>
for wilcoxon test.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_ref.group">ref.group</code></td>
<td>
<p>a character string or a numeric value specifying the
reference group. If specified, for a given grouping variable, each of the
group levels will be compared to the reference group (i.e. control group).
</p>
<p><code>ref.group</code> can be also <code>"all"</code>. In this case, each of the
grouping variable levels is compared to all (i.e. basemean).
</p>
<p>Allowed values can be: </p>
 <ul>
<li> <p><strong>numeric value</strong>: specifying
the rank of the reference group. For example, use <code>ref.group = 1</code> when
the first group is the reference; use <code>ref.group = 2</code> when the second
group is the reference, and so on. This works for all situations, including
i) when comparisons are performed between x-axis groups and ii) when
comparisons are performed between legend groups. </p>
</li>
<li> <p><strong>character
value</strong>: For example, you can use <code>ref.group = "ctrl"</code> instead of using
the numeric rank value of the &quot;ctrl&quot; group. </p>
</li>
<li> <p><strong>&quot;all&quot;</strong>: In this
case, each of the grouping variable levels is compared to all (i.e.
basemean). </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_pwc_+3A_label">label</code></td>
<td>
<p>character string specifying label. Can be: </p>
 <ul>
<li><p> the
column containing the label (e.g.: <code>label = "p"</code> or <code>label =
"p.adj"</code>), where <code>p</code> is the p-value. Other possible values are
<code>"p.signif", "p.adj.signif", "p.format", "p.adj.format"</code>. </p>
</li>
<li><p> an
expression that can be formatted by the <code><a href="glue.html#topic+glue">glue</a>()</code> package.
For example, when specifying <code>label = "Wilcoxon, p = \{p\}"</code>, the
expression {p} will be replaced by its value. </p>
</li>
<li><p> a combination of
plotmath expressions and glue expressions. You may want some of the
statistical parameter in italic; for example:<code>label = "Wilcoxon,
italic(p)= {p}"</code></p>
</li></ul>
<p>.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_y.position">y.position</code></td>
<td>
<p>numeric vector with the y positions of the brackets</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_group.by">group.by</code></td>
<td>
<p>(optional) character vector specifying the grouping variable;
it should be used only for grouped plots. Possible values are : </p>

<ul>
<li> <p><code>"x.var"</code>: Group by the x-axis variable and perform the test
between legend groups. In other words, the p-value is compute between legend
groups at each x position </p>
</li>
<li> <p><code>"legend.var"</code>: Group by the legend
variable and perform the test between x-axis groups. In other words, the
test is performed between the x-groups for each legend level. </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_pwc_+3A_dodge">dodge</code></td>
<td>
<p>dodge width for grouped ggplot/test. Default is 0.8. It's used to
dodge the brackets position when <code>group.by = "legend.var"</code>.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_bracket.nudge.y">bracket.nudge.y</code></td>
<td>
<p>Vertical adjustment to nudge brackets by (in fraction
of the total height). Useful to move up or move down the bracket. If
positive value, brackets will be moved up; if negative value, brackets are
moved down.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_bracket.shorten">bracket.shorten</code></td>
<td>
<p>a small numeric value in [0-1] for shortening the width
of bracket.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_bracket.group.by">bracket.group.by</code></td>
<td>
<p>(optional); a variable name for grouping brackets
before adding step.increase. Useful for grouped plots. Possible values
include <code>"x.var"</code> and <code>"legend.var"</code>.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_step.increase">step.increase</code></td>
<td>
<p>numeric vector with the increase in fraction of total
height for every additional comparison to minimize overlap.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_tip.length">tip.length</code></td>
<td>
<p>numeric vector with the fraction of total height that the
bar goes down to indicate the precise column/</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_size">size</code></td>
<td>
<p>change the width of the lines of the bracket</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_label.size">label.size</code></td>
<td>
<p>change the size of the label text</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_family">family</code></td>
<td>
<p>change the font used for the text</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_vjust">vjust</code></td>
<td>
<p>move the text up or down relative to the bracket.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_hjust">hjust</code></td>
<td>
<p>move the text left or right relative to the bracket.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for adjusting p values (see
<code><a href="stats.html#topic+p.adjust">p.adjust</a></code>).  Has impact only in a situation, where
multiple pairwise tests are performed; or when there are multiple grouping
variables. Ignored when the specified method is <code>"tukey_hsd"</code> or
<code>"games_howell_test"</code> because they come with internal p adjustment
method. Allowed values include &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;,
&quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;. If you don't want to adjust the p value (not
recommended), use p.adjust.method = &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_p.adjust.by">p.adjust.by</code></td>
<td>
<p>possible value is one of <code>c("group", "panel")</code>.
Default is <code>"group"</code>: for a grouped data, if pairwise test is
performed, then the p-values are adjusted for each group level
independently. P-values are adjusted by panel when <code>p.adjust.by =
"panel"</code>.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_symnum.args">symnum.args</code></td>
<td>
<p>a list of arguments to pass to the function
<code><a href="stats.html#topic+symnum">symnum</a></code> for symbolic number coding of p-values. For
example, <code>symnum.args &lt;- list(cutpoints = c(0, 0.0001, 0.001, 0.01,
 0.05, Inf), symbols = c("****", "***", "**", "*",  "ns"))</code>.
</p>
<p>In other words, we use the following convention for symbols indicating
statistical significance: </p>
 <ul>
<li> <p><code>ns</code>: p &gt; 0.05 </p>
</li>
<li>
<p><code>*</code>: p &lt;= 0.05 </p>
</li>
<li> <p><code>**</code>: p &lt;= 0.01 </p>
</li>
<li> <p><code>***</code>: p &lt;= 0.001
</p>
</li>
<li> <p><code>****</code>:  p &lt;= 0.0001 </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_pwc_+3A_hide.ns">hide.ns</code></td>
<td>
<p>can be logical value (<code>TRUE</code> or <code>FALSE</code>) or a character vector (<code>"p.adj"</code> or <code>"p"</code>).</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_remove.bracket">remove.bracket</code></td>
<td>
<p>logical, if <code>TRUE</code>, brackets are removed from the
plot. </p>
 <ul>
<li><p> Case when logical value. If TRUE, hide ns symbol when
displaying significance levels. Filter is done by checking the column
<code>p.adj.signif</code>, <code>p.signif</code>, <code>p.adj</code> and <code>p</code>. </p>
</li>
<li><p> Case
when character value. Possible values are &quot;p&quot; or &quot;p.adj&quot;, for filtering out
non significant. </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_pwc_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code> (the default), removes missing values with a
warning.  If <code>TRUE</code> silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_parse">parse</code></td>
<td>
<p>logical for parsing plotmath expression.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_...">...</code></td>
<td>
<p>other arguments passed on to <code>layer</code>. These are often
aesthetics, used to set an aesthetic to a fixed value, like <code>color =
"red"</code> or <code>size = 3</code>. They may also be parameters to the paired
geom/stat.</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_stat">stat</code></td>
<td>
<p>The statistical transformation to use on the data for this
layer, either as a <code>ggproto</code> <code>Geom</code> subclass or as a string naming the
stat stripped of the <code>stat_</code> prefix (e.g. <code>"count"</code> rather than
<code>"stat_count"</code>)</p>
</td></tr>
<tr><td><code id="stat_pwc_+3A_stack">stack</code></td>
<td>
<p>logical value. Default is FALSE; should be set to TRUE for
stacked bar plots or line plots. If TRUE, then the brackets are
automatically removed and the <code>dodge</code> value is set to zero.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><b>Notes on adjusted p-values and facet</b>. When using the ggplot facet functions, the p-values are computed and adjusted by panel, without taking into account the other panels. This is by design in ggplot2.
</p>
<p>In this case, when there is only one computed p-value by panel, then using 'label = &quot;p&quot;' or 'label = &quot;p.adj&quot;' will give the same results using 'geom_pwc()'. Again, p-value computation and adjustment in a given facet panel is done independently to the other panels.
</p>
<p>One might want to adjust the p-values of all the facet panels together. There are two solutions for that:
</p>

<ul>
<li><p> Using <code><a href="#topic+ggadjust_pvalue">ggadjust_pvalue</a>(p)</code> after creating the plot <code>p</code>
</p>
</li>
<li><p> or adding the adjusted p-value manually using <code><a href="#topic+stat_pvalue_manual">stat_pvalue_manual</a>()</code>. Read more at:
</p>

<ul>
<li> <p><a href="https://www.datanovia.com/en/blog/how-to-add-p-values-to-ggplot-facets/">How to Add P-values to GGPLOT Facets</a>
</p>
</li>
<li> <p><a href="https://www.datanovia.com/en/blog/add-p-values-to-ggplot-facets-with-different-scales/">Add P-values to GGPLOT Facets with Different Scales</a>
</p>
</li></ul>

</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+ggadjust_pvalue">ggadjust_pvalue</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- ToothGrowth
df$dose &lt;- factor(df$dose)

# Data preparation
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Transform `dose` into factor variable
df &lt;- ToothGrowth
df$dose &lt;- as.factor(df$dose)
# Add a random grouping variable
df$group &lt;- factor(rep(c("grp1", "grp2"), 30))
head(df, 3)


# Two groups by x position
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

# Create a box plot
# Add 10% spaces between the p-value labels and the plot border
bxp &lt;- ggboxplot(
  df, x = "dose", y = "len",
  color = "supp", palette = c("#00AFBB", "#E7B800")
) +
 scale_y_continuous(expand = expansion(mult = c(0.05, 0.10)))


# Add p-values onto the box plots
# label can be "p.format"  or "p.adj.format"
bxp + geom_pwc(
  aes(group = supp), tip.length = 0,
  method = "t_test", label = "p.format"
)

# Show adjusted p-values and significance levels
# Hide ns (non-significant)
bxp + geom_pwc(
  aes(group = supp), tip.length = 0,
  method = "t_test", label = "{p.adj.format}{p.adj.signif}",
  p.adjust.method = "bonferroni", p.adjust.by = "panel",
  hide.ns = TRUE
)

# Complex cases
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# 1. Add p-values of OJ vs VC at each dose group
bxp.complex &lt;- bxp +
  geom_pwc(
    aes(group = supp), tip.length = 0,
    method = "t_test", label = "p.adj.format",
    p.adjust.method = "bonferroni", p.adjust.by = "panel"
  )
# 2. Add pairwise comparisons between dose levels
# Nudge up the brackets by 20% of the total height
bxp.complex &lt;- bxp.complex +
  geom_pwc(
    method = "t_test", label = "p.adj.format",
    p.adjust.method = "bonferroni",
    bracket.nudge.y = 0.2
  )
# 3. Display the plot
bxp.complex


# Three groups by x position
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

# Simple plots
#_____________________________________

# Box plots with p-values
bxp &lt;- ggboxplot(
  df, x = "supp", y = "len", fill = "dose",
  palette = "npg"
)
bxp +
  geom_pwc(
    aes(group = dose), tip.length = 0,
    method = "t_test", label = "p.adj.format",
    bracket.nudge.y = -0.08
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

# Bar plots with p-values
bp &lt;- ggbarplot(
  df, x = "supp", y = "len", fill = "dose",
  palette = "npg", add = "mean_sd",
  position = position_dodge(0.8)
)
bp +
  geom_pwc(
    aes(group = dose), tip.length = 0,
    method = "t_test", label = "p.adj.format",
    bracket.nudge.y = -0.08
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

</code></pre>

<hr>
<h2 id='stat_regline_equation'>Add Regression Line Equation and R-Square to a GGPLOT.</h2><span id='topic+stat_regline_equation'></span>

<h3>Description</h3>

<p>Add regression line equation and R^2 to a ggplot. Regression
model is fitted using the function <code><a href="stats.html#topic+lm">lm</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_regline_equation(
  mapping = NULL,
  data = NULL,
  formula = y ~ x,
  label.x.npc = "left",
  label.y.npc = "top",
  label.x = NULL,
  label.y = NULL,
  output.type = "expression",
  geom = "text",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_regline_equation_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_regline_equation_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_regline_equation_+3A_formula">formula</code></td>
<td>
<p>a formula object</p>
</td></tr>
<tr><td><code id="stat_regline_equation_+3A_label.x.npc">label.x.npc</code>, <code id="stat_regline_equation_+3A_label.y.npc">label.y.npc</code></td>
<td>
<p>can be <code>numeric</code> or <code>character</code>
vector of the same length as the number of groups and/or panels. If too
short they will be recycled. </p>
 <ul>
<li><p> If <code>numeric</code>, value should
be between 0 and 1. Coordinates to be used for positioning the label,
expressed in &quot;normalized parent coordinates&quot;. </p>
</li>
<li><p> If <code>character</code>,
allowed values include: i) one of c('right', 'left', 'center', 'centre',
'middle') for x-axis; ii) and one of c( 'bottom', 'top', 'center', 'centre',
'middle') for y-axis.</p>
</li></ul>

<p>If too short they will be recycled.</p>
</td></tr>
<tr><td><code id="stat_regline_equation_+3A_label.x">label.x</code>, <code id="stat_regline_equation_+3A_label.y">label.y</code></td>
<td>
<p><code>numeric</code> Coordinates (in data units) to be used
for absolute positioning of the label. If too short they will be recycled.</p>
</td></tr>
<tr><td><code id="stat_regline_equation_+3A_output.type">output.type</code></td>
<td>
<p>character One of &quot;expression&quot;, &quot;latex&quot; or &quot;text&quot;.</p>
</td></tr>
<tr><td><code id="stat_regline_equation_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_regline_equation_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_regline_equation_+3A_na.rm">na.rm</code></td>
<td>
<p>If FALSE (the default), removes missing values with a warning. If
TRUE silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_regline_equation_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_regline_equation_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_regline_equation_+3A_...">...</code></td>
<td>
<p>other arguments to pass to <code><a href="ggplot2.html#topic+geom_text">geom_text</a></code> or
<code><a href="ggplot2.html#topic+geom_text">geom_label</a></code>.</p>
</td></tr>
</table>


<h3>Computed variables</h3>

 <dl>
<dt>x</dt><dd><p>x position for left edge</p>
</dd>
<dt>y</dt><dd><p>y position near upper edge</p>
</dd>
<dt>eq.label</dt><dd><p>equation for the
fitted polynomial as a character string to be parsed</p>
</dd>
<dt>rr.label</dt><dd><p><code class="reqn">R^2</code> of the fitted model as a character string to be parsed</p>
</dd>
<dt>adj.rr.label</dt><dd><p>Adjusted <code class="reqn">R^2</code> of the fitted model as a character string
to be parsed</p>
</dd>
<dt>AIC.label</dt><dd><p>AIC for the fitted model.</p>
</dd>
<dt>BIC.label</dt><dd><p>BIC for the fitted model.</p>
</dd>
<dt>hjust</dt><dd><p>Set to zero to override the default of the &quot;text&quot; geom.</p>
</dd></dl>



<h3>References</h3>

<p>the source code of the function <code>stat_regline_equation()</code> is
inspired from the code of the function <code>stat_poly_eq()</code> (in ggpmisc
package).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggscatter">ggscatter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Simple scatter plot with correlation coefficient and
# regression line
#::::::::::::::::::::::::::::::::::::::::::::::::::::
ggscatter(mtcars, x = "wt", y = "mpg", add = "reg.line") +
  stat_cor(label.x = 3, label.y = 34) +
  stat_regline_equation(label.x = 3, label.y = 32)


# Groupped scatter plot
#::::::::::::::::::::::::::::::::::::::::::::::::::::
ggscatter(
  iris, x = "Sepal.Length", y = "Sepal.Width",
  color = "Species", palette = "jco",
  add = "reg.line"
  ) +
  facet_wrap(~Species) +
  stat_cor(label.y = 4.4) +
  stat_regline_equation(label.y = 4.2)

# Polynomial equation
#::::::::::::::::::::::::::::::::::::::::::::::::::::

# Demo data
set.seed(4321)
x &lt;- 1:100
y &lt;- (x + x^2 + x^3) + rnorm(length(x), mean = 0, sd = mean(x^3) / 4)
my.data &lt;- data.frame(x, y, group = c("A", "B"),
                      y2 = y * c(0.5,2), block = c("a", "a", "b", "b"))

# Fit polynomial regression line and add labels
formula &lt;- y ~ poly(x, 3, raw = TRUE)
p &lt;- ggplot(my.data, aes(x, y2, color = group)) +
  geom_point() +
  stat_smooth(aes(fill = group, color = group), method = "lm", formula = formula) +
  stat_regline_equation(
    aes(label =  paste(..eq.label.., ..adj.rr.label.., sep = "~~~~")),
    formula = formula
  ) +
  theme_bw()
ggpar(p, palette = "jco")

</code></pre>

<hr>
<h2 id='stat_stars'>Add Stars to a Scatter Plot</h2><span id='topic+stat_stars'></span>

<h3>Description</h3>

<p>Create a star plot by drawing segments from group centroid to each points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_stars(
  mapping = NULL,
  data = NULL,
  geom = "segment",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_stars_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_stars_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_stars_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_stars_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_stars_+3A_na.rm">na.rm</code></td>
<td>
<p>If FALSE (the default), removes missing values with a warning.
If TRUE silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_stars_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_stars_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_stars_+3A_...">...</code></td>
<td>
<p>other arguments to pass to <code><a href="ggplot2.html#topic+geom_segment">geom_segment</a></code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ggscatter">ggscatter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("mtcars")
df &lt;- mtcars
df$cyl &lt;- as.factor(df$cyl)

# Scatter plot with ellipses and group mean points
ggscatter(df, x = "wt", y = "mpg",
   color = "cyl", shape = "cyl",
   mean.point = TRUE, ellipse = TRUE)+
 stat_stars(aes(color = cyl))

</code></pre>

<hr>
<h2 id='stat_welch_anova_test'>Add Welch One-Way ANOVA Test P-values to a GGPlot</h2><span id='topic+stat_welch_anova_test'></span>

<h3>Description</h3>

<p>Add Welch one-way ANOVA test p-values to a ggplot, such as
box blots, dot plots and stripcharts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_welch_anova_test(
  mapping = NULL,
  data = NULL,
  group.by = NULL,
  label = "{method}, p = {p.format}",
  label.x.npc = "left",
  label.y.npc = "top",
  label.x = NULL,
  label.y = NULL,
  step.increase = 0.1,
  p.adjust.method = "holm",
  significance = list(),
  geom = "text",
  position = "identity",
  na.rm = FALSE,
  show.legend = FALSE,
  inherit.aes = TRUE,
  parse = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat_welch_anova_test_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_group.by">group.by</code></td>
<td>
<p>(optional) character vector specifying the grouping variable;
it should be used only for grouped plots. Possible values are : </p>

<ul>
<li> <p><code>"x.var"</code>: Group by the x-axis variable and perform the test
between legend groups. In other words, the p-value is compute between legend
groups at each x position </p>
</li>
<li> <p><code>"legend.var"</code>: Group by the legend
variable and perform the test between x-axis groups. In other words, the
test is performed between the x-groups for each legend level. </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_label">label</code></td>
<td>
<p>the column containing the label (e.g.: label = &quot;p&quot; or label =
&quot;p.adj&quot;), where <code>p</code> is the p-value. Can be also an expression that can
be formatted by the <code><a href="glue.html#topic+glue">glue</a>()</code> package. For example, when
specifying label = &quot;t-test, p = {p}&quot;, the expression {p} will be
replaced by its value.</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_label.x.npc">label.x.npc</code>, <code id="stat_welch_anova_test_+3A_label.y.npc">label.y.npc</code></td>
<td>
<p>can be <code>numeric</code> or <code>character</code>
vector of the same length as the number of groups and/or panels. If too
short they will be recycled. </p>
 <ul>
<li><p> If <code>numeric</code>, value should
be between 0 and 1. Coordinates to be used for positioning the label,
expressed in &quot;normalized parent coordinates&quot;. </p>
</li>
<li><p> If <code>character</code>,
allowed values include: i) one of c('right', 'left', 'center', 'centre',
'middle') for x-axis; ii) and one of c( 'bottom', 'top', 'center', 'centre',
'middle') for y-axis.</p>
</li></ul>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_label.x">label.x</code>, <code id="stat_welch_anova_test_+3A_label.y">label.y</code></td>
<td>
<p><code>numeric</code> Coordinates (in data units) to be used
for absolute positioning of the label. If too short they will be recycled.</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_step.increase">step.increase</code></td>
<td>
<p>numeric vector with the increase in fraction of total
height for every additional comparison to minimize overlap.</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for adjusting p values (see
<code><a href="stats.html#topic+p.adjust">p.adjust</a></code>).  Has impact only in a situation, where
multiple pairwise tests are performed; or when there are multiple grouping
variables. Allowed values include &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;,
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;. If you don't want to adjust the p
value (not recommended), use p.adjust.method = &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_significance">significance</code></td>
<td>
<p>a list of arguments specifying the signifcance cutpoints
and symbols. For example, <code>significance &lt;- list(cutpoints = c(0,
 0.0001, 0.001, 0.01, 0.05, Inf), symbols = c("****", "***", "**", "*",
 "ns"))</code>.
</p>
<p>In other words, we use the following convention for symbols indicating
statistical significance: </p>
 <ul>
<li> <p><code>ns</code>: p &gt; 0.05 </p>
</li>
<li>
<p><code>*</code>: p &lt;= 0.05 </p>
</li>
<li> <p><code>**</code>: p &lt;= 0.01 </p>
</li>
<li> <p><code>***</code>: p &lt;= 0.001
</p>
</li>
<li> <p><code>****</code>:  p &lt;= 0.0001 </p>
</li></ul>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data, either as a
<code>ggproto</code> <code>Geom</code> subclass or as a string naming the geom stripped of the
<code>geom_</code> prefix (e.g. <code>"point"</code> rather than <code>"geom_point"</code>)</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string naming the adjustment
(e.g. <code>"jitter"</code> to use <code>position_jitter</code>), or the result of a call to a
position adjustment function. Use the latter if you need to change the
settings of the adjustment.</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_na.rm">na.rm</code></td>
<td>
<p>If FALSE (the default), removes missing values with a warning. If
TRUE silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_parse">parse</code></td>
<td>
<p>If TRUE, the labels will be parsed into expressions and displayed
as described in <code>?plotmath</code>.</p>
</td></tr>
<tr><td><code id="stat_welch_anova_test_+3A_...">...</code></td>
<td>
<p>other arguments passed to the function <code>geom_bracket()</code> or
<code>geom_text()</code></p>
</td></tr>
</table>


<h3>Computed variables</h3>


<ul>
<li><p>statistic:	the value of the test statistic (F-value)
</p>
</li>
<li><p>DFn: Degrees of Freedom in the numerator (i.e. DF effect)
</p>
</li>
<li><p>DFd:	Degrees of Freedom in the denominator (i.e., DF error)
</p>
</li>
<li><p>p:	p-value.
</p>
</li>
<li><p>p.adj: Adjusted p-values.
</p>
</li>
<li><p>p.signif: P-value significance.
</p>
</li>
<li><p>p.adj.signif: Adjusted p-value significance.
</p>
</li>
<li><p>p.format: Formated p-value.
</p>
</li>
<li><p>p.adj.format: Formated adjusted p-value.
</p>
</li>
<li><p>n: number of samples.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Data preparation
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Transform `dose` into factor variable
df &lt;- ToothGrowth
df$dose &lt;- as.factor(df$dose)
# Add a random grouping variable
set.seed(123)
df$group &lt;- sample(factor(rep(c("grp1", "grp2", "grp3"), 20)))
df$len &lt;- ifelse(df$group == "grp2", df$len+2, df$len)
df$len &lt;- ifelse(df$group == "grp3", df$len+7, df$len)
head(df, 3)


# Basic boxplot
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Create a basic boxplot
# Add 5% and 10% space to the plot bottom and the top, respectively
bxp &lt;- ggboxplot(df, x = "dose", y = "len") +
  scale_y_continuous(expand = expansion(mult = c(0.05, 0.1)))

# Add the p-value to the boxplot
bxp + stat_welch_anova_test()

# Change the label position
# Using coordinates in data units
bxp + stat_welch_anova_test(label.x = "1", label.y = 10, hjust = 0)

# Format the p-value differently
custom_p_format &lt;- function(p) {
  rstatix::p_format(p, accuracy = 0.0001, digits = 3, leading.zero = FALSE)
}
bxp + stat_welch_anova_test(
  label = "Welch Anova, italic(p) = {custom_p_format(p)}{p.signif}"
)

# Show a detailed label in italic
bxp + stat_welch_anova_test(label = "as_detailed_italic")


# Faceted plots
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Create a ggplot facet
bxp &lt;- ggboxplot(df, x = "dose", y = "len", facet.by = "supp") +
 scale_y_continuous(expand = expansion(mult = c(0.05, 0.1)))
# Add p-values
bxp + stat_welch_anova_test()


# Grouped plots
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
bxp2 &lt;- ggboxplot(df, x = "group", y = "len", color = "dose", palette = "npg")

# For each x-position, computes tests between legend groups
bxp2 + stat_welch_anova_test(aes(group = dose), label = "p = {p.format}{p.signif}")

#  For each legend group, computes tests between x variable groups
bxp2 + stat_welch_anova_test(aes(group = dose, color = dose), group.by = "legend.var")

</code></pre>

<hr>
<h2 id='text_grob'>Create a Text Graphical object</h2><span id='topic+text_grob'></span>

<h3>Description</h3>

<p>Create easily a customized text grob (graphical object). Wrapper around
<code><a href="grid.html#topic+grid.text">textGrob</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>text_grob(
  label,
  just = "centre",
  hjust = NULL,
  vjust = NULL,
  rot = 0,
  color = "black",
  face = "plain",
  size = NULL,
  lineheight = NULL,
  family = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="text_grob_+3A_label">label</code></td>
<td>
<p>A character or <a href="base.html#topic+expression">expression</a> vector.  Other
objects are coerced by <code><a href="grDevices.html#topic+as.graphicsAnnot">as.graphicsAnnot</a></code>.</p>
</td></tr>
<tr><td><code id="text_grob_+3A_just">just</code></td>
<td>
<p>The justification of the text
relative to its (x, y) location.  If there are two values, the first
value specifies horizontal justification and the second value specifies
vertical justification.  Possible string values are: <code>"left"</code>,
<code>"right"</code>, <code>"centre"</code>, <code>"center"</code>, <code>"bottom"</code>,
and <code>"top"</code>.  For numeric values, 0 means left (bottom) alignment
and 1 means right (top) alignment.
</p>
</td></tr>
<tr><td><code id="text_grob_+3A_hjust">hjust</code></td>
<td>
<p>A numeric vector specifying horizontal justification.
If specified, overrides the <code>just</code> setting.</p>
</td></tr>
<tr><td><code id="text_grob_+3A_vjust">vjust</code></td>
<td>
<p>A numeric vector specifying vertical justification.
If specified, overrides the <code>just</code> setting.</p>
</td></tr>
<tr><td><code id="text_grob_+3A_rot">rot</code></td>
<td>
<p>The angle to rotate the text.</p>
</td></tr>
<tr><td><code id="text_grob_+3A_color">color</code></td>
<td>
<p>text font color.</p>
</td></tr>
<tr><td><code id="text_grob_+3A_face">face</code></td>
<td>
<p>font face. Allowed values include one of <code>"plain", "bold",
"italic", "bold.italic"</code>.</p>
</td></tr>
<tr><td><code id="text_grob_+3A_size">size</code></td>
<td>
<p>font size (e.g.: size = 12)</p>
</td></tr>
<tr><td><code id="text_grob_+3A_lineheight">lineheight</code></td>
<td>
<p>line height (e.g.: lineheight = 2).</p>
</td></tr>
<tr><td><code id="text_grob_+3A_family">family</code></td>
<td>
<p>font family.</p>
</td></tr>
<tr><td><code id="text_grob_+3A_...">...</code></td>
<td>
<p>other arguments passed to <a href="grid.html#topic+grid.text">textGrob</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a text grob.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>text &lt;- paste("iris data set gives the measurements in cm",
             "of the variables sepal length and width",
             "and petal length and width, respectively,",
             "for 50 flowers from each of 3 species of iris.",
             "The species are Iris setosa, versicolor, and virginica.", sep = "\n")

# Create a text grob
tgrob &lt;- text_grob(text, face = "italic", color = "steelblue")
# Draw the text
as_ggplot(tgrob)

</code></pre>

<hr>
<h2 id='theme_pubr'>Publication ready theme</h2><span id='topic+theme_pubr'></span><span id='topic+theme_pubclean'></span><span id='topic+labs_pubr'></span><span id='topic+theme_classic2'></span><span id='topic+clean_theme'></span><span id='topic+clean_table_theme'></span>

<h3>Description</h3>

 <ul>
<li> <p><strong>theme_pubr()</strong>: Create a publication ready
theme </p>
</li>
<li> <p><strong>theme_pubclean()</strong>: a clean theme without axis lines, to
direct more attention to the data.  </p>
</li>
<li> <p><strong>labs_pubr()</strong>: Format only
plot labels to a publication ready style </p>
</li>
<li> <p><strong>theme_classic2()</strong>:
Create a classic theme with axis lines. </p>
</li>
<li> <p><strong>clean_theme()</strong>: Remove
axis lines, ticks, texts and titles. </p>
</li>
<li> <p><strong>clean_table_theme()</strong>:
Clean the the theme of a table, such as those created by
<code><a href="#topic+ggsummarytable">ggsummarytable</a>()</code></p>
</li></ul>
<p>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_pubr(
  base_size = 12,
  base_family = "",
  border = FALSE,
  margin = TRUE,
  legend = c("top", "bottom", "left", "right", "none"),
  x.text.angle = 0
)

theme_pubclean(base_size = 12, base_family = "", flip = FALSE)

labs_pubr(base_size = 14, base_family = "")

theme_classic2(base_size = 12, base_family = "")

clean_theme()

clean_table_theme()
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="theme_pubr_+3A_base_size">base_size</code></td>
<td>
<p>base font size</p>
</td></tr>
<tr><td><code id="theme_pubr_+3A_base_family">base_family</code></td>
<td>
<p>base font family</p>
</td></tr>
<tr><td><code id="theme_pubr_+3A_border">border</code></td>
<td>
<p>logical value. Default is FALSE. If TRUE, add panel border.</p>
</td></tr>
<tr><td><code id="theme_pubr_+3A_margin">margin</code></td>
<td>
<p>logical value. Default is TRUE. If FALSE, reduce plot margin.</p>
</td></tr>
<tr><td><code id="theme_pubr_+3A_legend">legend</code></td>
<td>
<p>character specifying legend position. Allowed values are one of
c(&quot;top&quot;, &quot;bottom&quot;, &quot;left&quot;, &quot;right&quot;, &quot;none&quot;). Default is &quot;top&quot; side position.
to remove the legend use legend = &quot;none&quot;. Legend position can be also
specified using a numeric vector c(x, y).  In this case it is possible to
position the legend inside the plotting area. x and y are the coordinates of
the legend box. Their values should be between 0 and 1. c(0,0) corresponds
to the &quot;bottom left&quot; and c(1,1) corresponds to the &quot;top right&quot; position. For
instance use legend = c(0.8, 0.2).</p>
</td></tr>
<tr><td><code id="theme_pubr_+3A_x.text.angle">x.text.angle</code></td>
<td>
<p>Rotation angle of x axis tick labels. Default value is 0.
Use 90 for vertical text.</p>
</td></tr>
<tr><td><code id="theme_pubr_+3A_flip">flip</code></td>
<td>
<p>logical. If TRUE, grid lines are added to y axis instead of x
axis.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>p &lt;- ggplot(mtcars, aes(x = wt, y = mpg)) +
   geom_point(aes(color = gear))

# Default plot
p

# Use theme_pubr()
p + theme_pubr()

# Format labels
p + labs_pubr()

</code></pre>

<hr>
<h2 id='theme_transparent'>Create a ggplot with Transparent Background</h2><span id='topic+theme_transparent'></span>

<h3>Description</h3>

<p>Create a ggplot with transparent background.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_transparent(base_size = 12, base_family = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="theme_transparent_+3A_base_size">base_size</code></td>
<td>
<p>base font size</p>
</td></tr>
<tr><td><code id="theme_transparent_+3A_base_family">base_family</code></td>
<td>
<p>base font family</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+theme_pubr">theme_pubr</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a scatter plot
sp &lt;- ggscatter(iris, x = "Sepal.Length", y = "Sepal.Width",
               color = "Species", palette = "jco",
               size = 3, alpha = 0.6)
sp

# Transparent theme
sp + theme_transparent()

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
