<!DOCTYPE html><html><head><title>Help for package ggrasp</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ggrasp}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ggrasp-class'><p>An S4 class representing the GGRaSP data and output</p></a></li>
<li><a href='#ggrasp.addRanks'><p>ggrasp.addRanks</p></a></li>
<li><a href='#ggrasp.cluster'><p>ggrasp.cluster</p></a></li>
<li><a href='#ggrasp.load'><p>ggrasp.load</p></a></li>
<li><a href='#ggrasp.recluster'><p>ggrasp.recluster</p></a></li>
<li><a href='#ggrasp.write'><p>ggrasp.write</p></a></li>
<li><a href='#plot.ggrasp'><p>plot.ggrasp</p></a></li>
<li><a href='#print.ggrasp'><p>print.ggrasp</p></a></li>
<li><a href='#summary.ggrasp'><p>summary.ggrasp</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gaussian-Based Genome Representative Selector with
Prioritization</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Given a group of genomes and their relationship with each other, the package clusters the genomes and selects the most representative members of each cluster. Additional data can be provided to the prioritize certain genomes. The results can be printed out as a list or a new phylogeny with graphs of the trees and distance distributions also available. For detailed introduction see: Thomas H Clarke, Lauren M Brinkac, Granger Sutton, and Derrick E Fouts (2018), GGRaSP: a R-package for selecting representative genomes using Gaussian mixture models, Bioinformatics, bty300, &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbty300">doi:10.1093/bioinformatics/bty300</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, mixtools, ape, bgmm, colorspace, methods</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>False</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-21 15:10:43 UTC; tclarke</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas Clarke [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomas Clarke &lt;tclarke@jcvi.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-21 15:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ggrasp-class'>An S4 class representing the GGRaSP data and output</h2><span id='topic+ggrasp-class'></span>

<h3>Description</h3>

<p>An S4 class representing the GGRaSP data and output
</p>


<h3>Slots</h3>


<dl>
<dt><code>dist.mst</code></dt><dd><p>The distance matrix showing the distances between different genomes</p>
</dd>
<dt><code>phy</code></dt><dd><p>The phylogenetic tree in newick format</p>
</dd>
<dt><code>rank</code></dt><dd><p>The ranks of the respective genomes with lower getting higher priority in being called as a medoid</p>
</dd>
<dt><code>cluster</code></dt><dd><p>A vector giving the numeric cluster ID for each genome</p>
</dd>
<dt><code>h</code></dt><dd><p>The threshold variable used to make the clusters</p>
</dd>
<dt><code>medoids</code></dt><dd><p>A vector giving the medoid for each cluster</p>
</dd>
<dt><code>gmm</code></dt><dd><p>A data.frame containing all the gaussian distributions used to find the threshold when available</p>
</dd>
<dt><code>gmm.orig</code></dt><dd><p>A data.frame containing all the gaussian distributions prior to cleaning. Used to recalculate the threshold when needed</p>
</dd>
</dl>

<hr>
<h2 id='ggrasp.addRanks'>ggrasp.addRanks</h2><span id='topic+ggrasp.addRanks'></span>

<h3>Description</h3>

<p>adds a rank file to a GGRaSP object. If clusters have been defined, the medoids will be re-defined
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggrasp.addRanks(x, rank.file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggrasp.addRanks_+3A_x">x</code></td>
<td>
<p>the GGRaSP object for which the ranks will be added.</p>
</td></tr>
<tr><td><code id="ggrasp.addRanks_+3A_rank.file">rank.file</code></td>
<td>
<p>string pointing to a file containing the ranks</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A GGRaSP object where the ranks have been entirely redefined with the ranks in rank.file
</p>

<hr>
<h2 id='ggrasp.cluster'>ggrasp.cluster</h2><span id='topic+ggrasp.cluster'></span>

<h3>Description</h3>

<p>ggrasp.cluster() clusters the genomes in a GGRaSP class variable and assigns the most representative genome in each cluster after accounting for rank as a medoid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggrasp.cluster(ggrasp.data, threshold, num.clusters, z.limit = 1,
  gmm.start = 2, gmm.max = 10, min.lambda = 0.005, run.type = "bgmm",
  left.dist = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggrasp.cluster_+3A_ggrasp.data">ggrasp.data</code></td>
<td>
<p>Required. If neither a threshold or a num.cluster is given, a mixed model of Gaussian distributions is used to estimate a threshold to use the cluster.</p>
</td></tr>
<tr><td><code id="ggrasp.cluster_+3A_threshold">threshold</code></td>
<td>
<p>The threshold used to cluster together all genomes within this distance.</p>
</td></tr>
<tr><td><code id="ggrasp.cluster_+3A_num.clusters">num.clusters</code></td>
<td>
<p>Create this number of clusters independent of the cluster.</p>
</td></tr>
<tr><td><code id="ggrasp.cluster_+3A_z.limit">z.limit</code></td>
<td>
<p>All Gaussian distributions with means within this number of standard deviations will be reduced to only the larger distribution. Defaults to 1. Set to 0 to keep all non-overlapping distributions.</p>
</td></tr>
<tr><td><code id="ggrasp.cluster_+3A_gmm.start">gmm.start</code></td>
<td>
<p>Number of Gaussian distributions to start the examination. Must be at least 2 and not greater than the gmm.max.</p>
</td></tr>
<tr><td><code id="ggrasp.cluster_+3A_gmm.max">gmm.max</code></td>
<td>
<p>Maximum number of Gaussian distributions to examine. Has to be at least 2. 10 is the default</p>
</td></tr>
<tr><td><code id="ggrasp.cluster_+3A_min.lambda">min.lambda</code></td>
<td>
<p>All Gaussian distributions with lambda value (proportion of the total distribution) below this value are removed before calculating the threshold. Default is 0.005. Set to 0 to keep all.</p>
</td></tr>
<tr><td><code id="ggrasp.cluster_+3A_run.type">run.type</code></td>
<td>
<p>String giving the package to use to get the mixture model. Currently &quot;bgmm&quot; (default) and mixtools&quot; are implemented.</p>
</td></tr>
<tr><td><code id="ggrasp.cluster_+3A_left.dist">left.dist</code></td>
<td>
<p>Number giving the number Gaussian distribution model immediately to the left of the threshold used. 1 is the default. Only value between 1 and k-1 where k is the total number of number of Gaussian distributions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a class GGRaSP variable with the clusters and medoids assigned. In cases where the Gaussian Mixture Model was used to estimate the cutoff threshold, the descriptive values of the different distributions is also stored
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The following data is from Chavda et al 2016 which phylotyped Enterobacter genomes
# Our example uses the data underpinning the tree shown in Figure 2
# Also included is a ranking file to prioritize closed Enterobactor genomes

#Loading the tree 
library(ggrasp);
tree.file &lt;- system.file("extdata", "Enter.kSNP.tree", package="ggrasp")
rank.file.in &lt;- system.file("extdata", "Enter.kSNP.ranks", package="ggrasp")
Enter.tree &lt;- ggrasp.load(tree.file, file.format = "tree", rank.file = rank.file.in)

#Clustering the tree using a threshold estimated by Gaussian Mixture Models (GMMs)
Enter.tree.cluster &lt;- ggrasp.cluster(Enter.tree)


#Use print to get a list of the medoids selected
print(Enter.tree.cluster)

#Re-clustering the tree using a threshold estimated by the GMMs but without the distribution
#cleaning (i.e. removing the overlapping and low count distributions)
Enter.tree.reclust &lt;- ggrasp.recluster(Enter.tree.cluster, z.limit=0, min.lambda = 0)

#Use plot to examine the tree with the clusters highlighted and the medoid genome names on the edge
plot(Enter.tree.cluster)

#Additional printing and plotting options are availible with plot() and print(). 
#For more information refer to ?plot.ggrasp and ?print.ggrasp
</code></pre>

<hr>
<h2 id='ggrasp.load'>ggrasp.load</h2><span id='topic+ggrasp.load'></span>

<h3>Description</h3>

<p>ggrasp.load() initializes a class GGRaSP object from a file containing either a tree, a distance matrix or a multi-fasta alignment. The returned object can subsequently be clustered using ggrasp.cluster().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggrasp.load(file, file.format, rank.file, offset, tree.method = "complete")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggrasp.load_+3A_file">file</code></td>
<td>
<p>File containing the tree, matrix or sequence alignment used to initialize the ggrasp object. Required.</p>
</td></tr>
<tr><td><code id="ggrasp.load_+3A_file.format">file.format</code></td>
<td>
<p>The format the file is in, with tree, fasta and matrix accepted. If not given the program makes a guess.</p>
</td></tr>
<tr><td><code id="ggrasp.load_+3A_rank.file">rank.file</code></td>
<td>
<p>File containing the ranks of genomes in a tab-delineated file with the genome in column 1 and the rank in column 2. The rank is a non-negative number.</p>
</td></tr>
<tr><td><code id="ggrasp.load_+3A_offset">offset</code></td>
<td>
<p>Numeric representing a perfect match. Default is 0.</p>
</td></tr>
<tr><td><code id="ggrasp.load_+3A_tree.method">tree.method</code></td>
<td>
<p>The method used to make the tree from a distance matrix. &quot;Complete&quot; (Default), &quot;Average&quot;, &quot;Single&quot;, and &quot;nj&quot; (Neighbor Joining) are currently available.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a class GGRaSP variable
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The following data is from Chavda et al 2016 which phylotyped Enterobacter genomes
# Our example uses the data underpinning the tree shown in Figure 2
# Also included is a ranking file to prioritize closed Enterobactor genomes

library(ggrasp);
tree.file &lt;- system.file("extdata", "Enter.kSNP.tree", package="ggrasp")
rank.file.in &lt;- system.file("extdata", "Enter.kSNP.ranks", package="ggrasp")
Enter.tree &lt;- ggrasp.load(tree.file, file.format = "tree", rank.file = rank.file.in);

# Other options include loading by fasta file:
fasta.file &lt;- system.file("extdata", "Enter.kSNP2.fasta", package="ggrasp")
rank.file.in &lt;- system.file("extdata", "Enter.kSNP.ranks", package="ggrasp")
Enter.tree &lt;- ggrasp.load(fasta.file, file.format = "fasta", rank.file =rank.file.in)

# and by distance matrix. Since this distance matrix is actually percent identity,
# we will us an offset of 100
mat.file &lt;- system.file("extdata", "Enter.ANI.mat", package="ggrasp")
rank.file.in &lt;- system.file("extdata", "Enter.kSNP.ranks", package="ggrasp")
Enter.in &lt;- ggrasp.load(mat.file, file.format = "matrix", rank.file =rank.file.in, offset = 100)

# Use summary() to examine the data loaded
summary(Enter.in)

#Use plot() to see the tree
plot(Enter.in)
</code></pre>

<hr>
<h2 id='ggrasp.recluster'>ggrasp.recluster</h2><span id='topic+ggrasp.recluster'></span>

<h3>Description</h3>

<p>recalculates a threshold and the resulting cluster using the previously defined Gaussian Mixture Model and provided threshold-determining factors. Requires the ggrasp.cluster to already have run
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggrasp.recluster(x, z.limit = 1, min.lambda = 0.005, left.dist = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggrasp.recluster_+3A_x">x</code></td>
<td>
<p>the GGRaSP object for which the ranks will be added.</p>
</td></tr>
<tr><td><code id="ggrasp.recluster_+3A_z.limit">z.limit</code></td>
<td>
<p>All Gaussian distributions with means within this number of standard deviations will be reduced to only the larger distribution. Defaults to 1. Set to 0 to keep all non-overlapping distributions.</p>
</td></tr>
<tr><td><code id="ggrasp.recluster_+3A_min.lambda">min.lambda</code></td>
<td>
<p>All Gaussian distributions with lambda value (proportion of the total distribution) below this value are removed before calculating the threshold. Default is 0.005. Set to 0 to keep all.</p>
</td></tr>
<tr><td><code id="ggrasp.recluster_+3A_left.dist">left.dist</code></td>
<td>
<p>Number giving the number Gaussian distribution model immediately to the left of the threshold used. 1 is the default. Only value between 1 and k-1 where k is the total number of number of Gaussian distributions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A GGRaSP object with the recalculated thresholds and the medoids using a previously generated GMM
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The following data is from Chavda et al 2016 which phylotyped Enterobacter genomes
# Our example uses the data underpinning the tree shown in Figure 2

#Loading the tree 
library(ggrasp);
tree.file &lt;- system.file("extdata", "Enter.kSNP.tree", package="ggrasp")
Enter.tree &lt;- ggrasp.load(tree.file, file.format = "tree");

#Clustering the tree using a threshold estimated by Gaussian Mixture Models (GMMs)
Enter.tree.cluster &lt;- ggrasp.cluster(Enter.tree)


#Use print to get a list of the medoids selected
print(Enter.tree.cluster)

#Re-clustering the tree using a threshold estimated by the GMMs but without the distribution
#cleaning (i.e. removing the overlapping and low count distributions)
Enter.tree.reclust &lt;- ggrasp.recluster(Enter.tree.cluster, z.limit=0, min.lambda = 0)

</code></pre>

<hr>
<h2 id='ggrasp.write'>ggrasp.write</h2><span id='topic+ggrasp.write'></span>

<h3>Description</h3>

<p>writes formatted information from a class GGRaSP object to a file. Multiple output options are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggrasp.write(x, file, type, rank.level)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggrasp.write_+3A_x">x</code></td>
<td>
<p>ggrasp-class object to be written</p>
</td></tr>
<tr><td><code id="ggrasp.write_+3A_file">file</code></td>
<td>
<p>String pointing to file where the data will be saved. If no file is given, the result will be printed out on the screen.</p>
</td></tr>
<tr><td><code id="ggrasp.write_+3A_type">type</code></td>
<td>
<p>Format of the data printed, either &quot;tree&quot; (New Hampshire extended style), &quot;table&quot; where the medoids or representative are shown in a table format, &quot;list&quot; where the information is shown in a pseudo-fasta format, or &quot;itol&quot; which prints out a file that can be loaded into the itol phylogeny viewer (http://itol.embl.de) which will color the clades of the different clusters</p>
</td></tr>
<tr><td><code id="ggrasp.write_+3A_rank.level">rank.level</code></td>
<td>
<p>Maximum level of the rank to show. Ignored pre-clustering. After clustering, 0 will show only the medoids, -1 will show all values independent of rank, and any value &gt;= 1 will show all the genomes less than or equal to that rank (including medoids). Default is 0 (only the medoids)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#Getting the ggrasp object
Enter.tree &lt;- ggrasp.load(system.file("extdata", "Enter.kSNP.tree", package="ggrasp"), 
file.format = "tree", rank.file =system.file("extdata", "Enter.kSNP.ranks", package="ggrasp"));
Enter.tree.cluster &lt;- ggrasp.cluster(Enter.tree)

#Default examples: using the initizalized ggrasp object will 
#write the newick tree string to "tree.nwk"
ggrasp.write(Enter.tree, type="tree", file=file.path(tempdir(), "tree.nwk"));

# Using the clustered ggrasp object will write a text file with the clusters saved as an ITOL clade
# In conjecture with the phylogeny, this is readable by 
# ITOL web phylogeny visualizer (http://itol.embl.de/) 
ggrasp.write(Enter.tree.cluster, type="itol", file=file.path(tempdir(), "tree.itol.clade.txt"));
</code></pre>

<hr>
<h2 id='plot.ggrasp'>plot.ggrasp</h2><span id='topic+plot.ggrasp'></span>

<h3>Description</h3>

<p>plots a class GGRaSP variable either the full tree, a reduced tree, or the gmm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ggrasp'
plot(x, type, layout = "circular", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.ggrasp_+3A_x">x</code></td>
<td>
<p>ggrasp-class object to be plotted</p>
</td></tr>
<tr><td><code id="plot.ggrasp_+3A_type">type</code></td>
<td>
<p>Type of plot generated, either &quot;tree&quot; (the full tree with the clusters shown as grouped leaves), &quot;reduced&quot; (tree with only the medoids shown), &quot;hist&quot; (histogram of the distribution of the distances) or &quot;gmm&quot; (histogram of the distances overlayed with the Gaussian distributions)</p>
</td></tr>
<tr><td><code id="plot.ggrasp_+3A_layout">layout</code></td>
<td>
<p>The layout style of the tree, either &quot;circular&quot; (default), &quot;radial&quot;, &quot;slanted&quot;, &quot;linear&quot; or &quot;rectangular&quot; (&quot;linear&quot; or &quot;rectangular&quot; are the same).</p>
</td></tr>
<tr><td><code id="plot.ggrasp_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object containing the plot. It can be printed to standard output or saved using ggsave.
</p>

<hr>
<h2 id='print.ggrasp'>print.ggrasp</h2><span id='topic+print.ggrasp'></span>

<h3>Description</h3>

<p>prints formatted information from a class GGRaSP object. Multiple output options are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ggrasp'
print(x, type, rank.level, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.ggrasp_+3A_x">x</code></td>
<td>
<p>ggrasp-class object to be printed</p>
</td></tr>
<tr><td><code id="print.ggrasp_+3A_type">type</code></td>
<td>
<p>Format of the data printed, either &quot;tree&quot; (new hampshire extended style), &quot;table&quot; where the medoids or representative are shown in a table format, or &quot;list&quot; where the information is shown in a pseudo-fasta format</p>
</td></tr>
<tr><td><code id="print.ggrasp_+3A_rank.level">rank.level</code></td>
<td>
<p>Maximum level of the rank to show. Ignored pre-clustering. After clustering, 0 will show only the medoids, -1 will show all values independent of rank, and any value &gt;= 1 will show all the genomes less than or equal to that rank (including medoids). Default is 0 (only the medoids)</p>
</td></tr>
<tr><td><code id="print.ggrasp_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#Getting the ggrasp object
Enter.tree &lt;- ggrasp.load(system.file("extdata", "Enter.kSNP.tree", package="ggrasp"), 
file.format = "tree", rank.file =system.file("extdata", "Enter.kSNP.ranks", package="ggrasp"));
Enter.tree.cluster &lt;- ggrasp.cluster(Enter.tree)

#Default examples: using the initizalized ggrasp object will print the newick tree string 
print(Enter.tree);

# Using the clustered ggrasp object will print the medoids and their respective clusters
print(Enter.tree.cluster)
#Below are examples of using different output formats and rank levels
print(Enter.tree.cluster, "tree")
print(Enter.tree.cluster, "table", 1)
print(Enter.tree.cluster, "table", 0)
</code></pre>

<hr>
<h2 id='summary.ggrasp'>summary.ggrasp</h2><span id='topic+summary.ggrasp'></span>

<h3>Description</h3>

<p>prints a summary of the class GGRaSP variable. Output includes medoids and cutoff value after the clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ggrasp'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.ggrasp_+3A_object">object</code></td>
<td>
<p>ggrasp-class object</p>
</td></tr>
<tr><td><code id="summary.ggrasp_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
