<!DOCTYPE html><html lang="en"><head><title>Help for package Ghat</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Ghat}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Ghat'><p>Quantifying evolution and selection on complex traits</p></a></li>
<li><a href='#ld_decay'><p>Evaluation of Linkage Disequilibrium Decay</p></a></li>
<li><a href='#Maize_wqs'><p>The Wisconsin Quality Synthetic (WQS) maize population datasets.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Quantifying Evolution and Selection on Complex Traits</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions are provided for quantifying evolution and selection on complex traits. 
              The package implements effective handling and analysis algorithms scaled for 
              genome-wide data and calculates a composite statistic, denoted Ghat, which is used 
              to test for selection on a trait. The package provides a number of simple examples 
              for handling and analysing the genome data and visualising the output and results. 
              Beissinger et al., (2018) &lt;<a href="https://doi.org/10.1534%2Fgenetics.118.300857">doi:10.1534/genetics.118.300857</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://academic.oup.com/genetics/article/209/1/321/5931021">https://academic.oup.com/genetics/article/209/1/321/5931021</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Medhat86/Ghat/issues">https://github.com/Medhat86/Ghat/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Imports:</td>
<td>rrBLUP</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-25 10:46:34 UTC; beissinger</td>
</tr>
<tr>
<td>Author:</td>
<td>Medhat Mahmoud [aut],
  Ngoc-Thuy Ha [aut],
  Tim Beissinger [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tim Beissinger &lt;timbeissinger@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-14 12:00:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='Ghat'>Quantifying evolution and selection on complex traits</h2><span id='topic+Ghat'></span>

<h3>Description</h3>

<p>G-hat: R function to estimate G-hat from allele frequency and effect size data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ghat(effects = effects, change = change, method = "scale",
  perms = 1000, plot = "Both", blockSize = 1000, num_eff = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Ghat_+3A_effects">effects</code></td>
<td>
<p>Vector of allele effects.</p>
</td></tr>
<tr><td><code id="Ghat_+3A_change">change</code></td>
<td>
<p>Vector of changes in allele frequency (could be positive, negative or zero).</p>
</td></tr>
<tr><td><code id="Ghat_+3A_method">method</code></td>
<td>
<p>&quot;vanilla&quot; (assumes complete linkage equilibrium between markers), &quot;trim&quot; (excludes markers to approximate linkage equilibrium some of the extreme values) or &quot;scale&quot; (scales results to reflect underlying levels of linkage LD)</p>
</td></tr>
<tr><td><code id="Ghat_+3A_perms">perms</code></td>
<td>
<p>Number of permutations to run.</p>
</td></tr>
<tr><td><code id="Ghat_+3A_plot">plot</code></td>
<td>
<p>&quot;Ghat&quot;, &quot;Cor&quot;, or &quot;Both&quot;, Should a plot of the Ghat or correlation test be returned?</p>
</td></tr>
<tr><td><code id="Ghat_+3A_blocksize">blockSize</code></td>
<td>
<p>How large should blocks for trimming be? Only required if method = &quot;trim&quot;.</p>
</td></tr>
<tr><td><code id="Ghat_+3A_num_eff">num_eff</code></td>
<td>
<p>The effective number of independent markers, to be used only in conjunction with the “scale” method, above (see “ld_decay” function or use help (?ld_decay).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Ghat Ghat-value
</p>
<p>Cor  Correlation between alleles frequencies and their effects
</p>
<p>p.val two-sided P-value of Evidence of selection
</p>
<p>plot relationship between estimated allelic effects at individual SNPs and the change in allele frequency over generations
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Example-1 Both SNP effects and change in allele frequency are known
maize		&lt;- Maize_wqs[[1]]
result.adf	&lt;- Ghat(effects =maize[,1], change=maize[,2], method="scale",
                     perms=1000, plot="Ghat", num_eff=54.74819)
mtext(paste("WQS ADF test for selection, pval = ", round(result.adf$p.val,4)))
message (c(result.adf$Ghat , result.adf$Cor , result.adf$p.va))


## Not run: 
#Example-2 Both SNP effects and change in allele frequency are known
##################################################################
## step 1: #run rrBLUP and estimating allels effects            ##
##################################################################

library(Ghat)
library(parallel)
library(rrBLUP)
phe                 &lt;- Maize_wqs[[2]]
map                 &lt;- Maize_wqs[[3]]
gen                 &lt;- Maize_wqs[[4]]
phe                 &lt;-phe[which(is.na(phe[,2])==FALSE),]
gen                 &lt;-gen[which(is.na(phe[,2])==FALSE),]
result              &lt;- mixed.solve(phe[,2],
                                   Z= as.matrix(gen[,2:ncol(gen)]),
                                   X= model.matrix(phe[,2]~phe[,3]),
                                   K=NULL, SE=FALSE, return.Hinv=FALSE,
                                   method="ML")
                                   
##################################################################
## step 2: is to calculate the allele frequency at Cycle 1 and 3##
##################################################################
CycleIndicator      &lt;- as.numeric(unlist(strsplit(gen$X,
                       split="_C")) [seq(2,2*nrow(gen),2)])
Cycle1              &lt;- gen[which(CycleIndicator == 1),]
Cycle3              &lt;- gen[which(CycleIndicator == 3),]
CycleList           &lt;- list(Cycle1,Cycle3)
frequencies         &lt;- matrix(nrow=ncol(gen)-1,ncol=2)
for(i in 1:2){
  frequencies[,i]   &lt;- colMeans(CycleList[[i]][,-1],na.rm=TRUE)/2
}
frequencies         &lt;- as.data.frame(frequencies)
names(frequencies)  &lt;- c("Cycle1","Cycle3")
change&lt;-frequencies$Cycle3-frequencies$Cycle1

################################################################
## step 3: Calculate LD Decay                                   ##
################################################################
ld                  &lt;- ld_decay (gen=gen, map=map,
                                 max_win_snp=2000, max.chr=10,
                                 cores=1, max_r2=0.03)

##################################################################
## step 4: Calculate Ghat                                       ##
##################################################################
Ghat.adf    &lt;- Ghat(effects=result$u, change=change, method = "scale",
                    perms=1000,plot="Ghat", num_eff = 54.74819)

message (paste("Ghat=" , Ghat.adf$Ghat,
            "Cor="  , Ghat.adf$Cor ,
            "P-val=", Ghat.adf$p.va, sep = " "))

## End(Not run)

</code></pre>

<hr>
<h2 id='ld_decay'>Evaluation of Linkage Disequilibrium Decay</h2><span id='topic+ld_decay'></span>

<h3>Description</h3>

<p>ld_decay: R function for calculating the effective number of independent markers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ld_decay(gen = gen, map = map, max_win_snp = 2000,
  max.chr = max.chr, cores = 1, max_r2 = max_r2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ld_decay_+3A_gen">gen</code></td>
<td>
<p>Matrix of genotype data. Individuals in rows, genotypes (0, 1, 2) in columns.</p>
</td></tr>
<tr><td><code id="ld_decay_+3A_map">map</code></td>
<td>
<p>Dataframe inculding the name for each marker with a corresponding chromosome number and physical position.</p>
</td></tr>
<tr><td><code id="ld_decay_+3A_max_win_snp">max_win_snp</code></td>
<td>
<p>The maximum number of markers in each window. Sets the maximum number of markers allowed per window within a chromosome before estimating the LD. Default is 2000.</p>
</td></tr>
<tr><td><code id="ld_decay_+3A_max.chr">max.chr</code></td>
<td>
<p>Chromosomes above this number will be excluded from the analysis.</p>
</td></tr>
<tr><td><code id="ld_decay_+3A_cores">cores</code></td>
<td>
<p>Numer of cores for using parallelized calculation, Default is 1 for windows machine.</p>
</td></tr>
<tr><td><code id="ld_decay_+3A_max_r2">max_r2</code></td>
<td>
<p>the threshold of r^2 to calculate the effective number of independent markers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>cor: Correlation matrix
</p>
<p>ch_eff_nmark: The Number of independent marker per chromosome
</p>
<p>eff_nmark: The effective number of independent markers
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library("parallel")
gen         &lt;- Maize_wqs[[4]]
map         &lt;- Maize_wqs[[3]]
Res_ld &lt;- ld_decay (gen=gen, map=map, max_win_snp=2000,
                    max.chr=10, cores=1, max_r2=0.03)

## End(Not run)                     
                    
</code></pre>

<hr>
<h2 id='Maize_wqs'>The Wisconsin Quality Synthetic (WQS) maize population datasets.</h2><span id='topic+Maize_wqs'></span>

<h3>Description</h3>

<p>The Wisconsin Quality Synthetic (WQS) maize population datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Maize_wqs
</code></pre>


<h3>Format</h3>

<p>A list of 4 data sets:
</p>

<dl>
<dt>Dataset-1</dt><dd><p>Data frame including all SNP effects and changes in allele frequencies between cycle 2 and 5.</p>
</dd>
<dt>Dataset-2</dt><dd><p>BLUP breeding values for Acid Detergent Fiber (ADF) involving 5 generations of selection.</p>
</dd>
<dt>Dataset-3</dt><dd><p>Map file: Each line of the Map file describes a single marker and must contain at least three columns. 1: chromosome number; 2: SNP (snp id); 3: SNP position (in base-pairs (bp)).</p>
</dd>
<dt>Dataset-4</dt><dd><p>Maize Genotype (Illumina MaizeSNP50 BeadChip); an Infinium HD assay (Illumina, Inc. San Diego, CA). 
10,017 SNP markers (0,1 and 2) after filtration, distributed across the maize genome (Ganal et al.2011).</p>
</dd>
</dl>


<h3>Source</h3>

<p>Lorenz, A. J., Beissinger, T.M., Rodrigues, R., de Leon, N. 2015. 
Selection for silage yield and composition did not affect genomic diversity 
within the Wisconsin Quality Synthetic maize population. 
Genes Genomes Genetics. DOI: 10.1534/g3.114.015263.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>maize &lt;- Maize_wqs[[1]]
phe   &lt;- Maize_wqs[[2]]
map   &lt;- Maize_wqs[[3]]
gen   &lt;- Maize_wqs[[4]]
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
