<!DOCTYPE html><html lang="en"><head><title>Help for package GhostKnockoff</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GhostKnockoff}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GhostKnockoff.filter'>
<p>GhostKnockoff filter</p></a></li>
<li><a href='#GhostKnockoff.fit'>
<p>Feature importance score generator</p></a></li>
<li><a href='#GhostKnockoff.GetCorStudy'>
<p>Calculate study correlation</p></a></li>
<li><a href='#GhostKnockoff.prelim'>
<p>Preliminary data management for GhostKnockoff</p></a></li>
<li><a href='#GhostKnockoff.prelim.Meta'>
<p>Additional preliminary data management for GhostKnockoff if multiple studies are involved</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>The Knockoff Inference Using Summary Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-11-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Zihuai He &lt;zihuai@stanford.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zihuai He &lt;zihuai@stanford.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for multiple knockoff inference using summary statistics, e.g. Z-scores. The knockoff inference is a general procedure for controlling the false discovery rate (FDR) when performing variable selection. This package provides a procedure which performs knockoff inference without ever constructing individual knockoffs (GhostKnockoff). It additionally supports multiple knockoff inference for improved stability and reproducibility. Moreover, it supports meta-analysis of multiple overlapping studies. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.5.0), Matrix, CVXR, Rdsdp, gtools, seqminer</td>
</tr>
<tr>
<td>Imports:</td>
<td>RSpectra, corpcor</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-22 18:09:19 UTC; zhe</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-23 08:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='GhostKnockoff.filter'>
GhostKnockoff filter
</h2><span id='topic+GhostKnockoff.filter'></span>

<h3>Description</h3>

<p>This function calculates Q-values to select variants associated with the outcome, given the feature importance scores
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GhostKnockoff.filter(T_0,T_k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GhostKnockoff.filter_+3A_t_0">T_0</code></td>
<td>

<p>A p*1 matrix. The feature importance score for the original variants.
</p>
</td></tr>
<tr><td><code id="GhostKnockoff.filter_+3A_t_k">T_k</code></td>
<td>

<p>A p*M matrix where M is the number of hypothetical knockoffs. The knockoff feature importance scores.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>q</code></td>
<td>

<p>A vector of length p, where each element is a Q-value. Variants with q &lt;= target FDR will be selected.
</p>
</td></tr>
<tr><td><code>kappa</code></td>
<td>

<p>A vector of length p. Multiple knockoff statistics kappa.
</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>

<p>A vector of length p. Multiple knockoff statistics tau.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# We use genetic data as an example
library(GhostKnockoff)

# load example vcf file from package "seqminer", this serves as the reference panel
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])
example.G &lt;- example.G[,apply(example.G,2,sd)!=0]
example.G &lt;- example.G[,1:100]

# compute correlation among variants
cor.G&lt;-matrix(as.numeric(corpcor::cor.shrink(example.G)), nrow=ncol(example.G))

# fit null model
fit.prelim&lt;-GhostKnockoff.prelim(cor.G,M=5,method='asdp',max.size=500)

### if only one study is involved
Zscore_0&lt;-as.matrix(rnorm(nrow(cor.G))) # hypothetical Z-scores
Zscore_0&lt;-Zscore_0+rbinom(nrow(cor.G),size=2,0.1) # set causal
n.study&lt;-c(5000)

# knockoff scores for each block, this can be run in parallel too
GK.stat&lt;-GhostKnockoff.fit(Zscore_0,n.study,fit.prelim,gamma=1,weight.study=NULL)

# knockoff filter
GK.filter&lt;-GhostKnockoff.filter(GK.stat$T_0,GK.stat$T_k)
GK.filter$q

### if multiple studies are involved, for a meta-analysis

# compute study correlation
Zscore_0&lt;-cbind(rnorm(nrow(cor.G)),rnorm(nrow(cor.G))) # hypothetical Z-scores
Zscore_0&lt;-Zscore_0+rbinom(nrow(cor.G),size=2,0.1) # set causal
cor.study&lt;-GhostKnockoff.GetCorStudy(Zscore_0,fit.prelim)

# note that all steps above can be run in parallel for nonoverlapping blocks of the genome.
# Then the overall study correlation can be computed by averaging cor.study of all blocks.

# compute optimal weights and study dependency using overall study correlaton
n.study&lt;-c(5000,7500)
Meta.prelim&lt;-GhostKnockoff.prelim.Meta(cor.study, n.study)
gamma&lt;-Meta.prelim$gamma
weight.study&lt;-Meta.prelim$w_opt

# knockoff scores for each block, this can be run in parallel too
GK.stat&lt;-GhostKnockoff.fit(Zscore_0,n.study,fit.prelim,gamma=gamma,weight.study=weight.study)

# knockoff filter
GK.filter&lt;-GhostKnockoff.filter(GK.stat$T_0,GK.stat$T_k)
GK.filter$q

</code></pre>

<hr>
<h2 id='GhostKnockoff.fit'>
Feature importance score generator
</h2><span id='topic+GhostKnockoff.fit'></span>

<h3>Description</h3>

<p>Generate original and knockoff feature importance scores given original Z-scores from multiple studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GhostKnockoff.fit(Zscore_0,n.study,fit.prelim,gamma=1,weight.study=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GhostKnockoff.fit_+3A_zscore_0">Zscore_0</code></td>
<td>

<p>A p*K Z-score matrix, where p is the number of variants and K is the number of studies. Variants not observed in the study should be coded as NA.
</p>
</td></tr>
<tr><td><code id="GhostKnockoff.fit_+3A_n.study">n.study</code></td>
<td>

<p>A vector of length K, where each element is the study sample size.
</p>
</td></tr>
<tr><td><code id="GhostKnockoff.fit_+3A_fit.prelim">fit.prelim</code></td>
<td>

<p>The output of function &quot;GhostKnockoff.prelim&quot;.
</p>
</td></tr>
<tr><td><code id="GhostKnockoff.fit_+3A_gamma">gamma</code></td>
<td>

<p>The estimated study dependency. See functon &quot;GhostKnockoff.GetGamma&quot;.
</p>
</td></tr>
<tr><td><code id="GhostKnockoff.fit_+3A_weight.study">weight.study</code></td>
<td>

<p>The weights to combine multiple studies. The default is based on sample size. The optimal weights can be estimated using function &quot;GhostKnockoff.prelim.Meta&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>GK.Zscore_0</code></td>
<td>

<p>A vector of length p, where each element is weighted combination of original Z-scores
</p>
</td></tr>
<tr><td><code>GK.Zscore_k</code></td>
<td>

<p>A p*M matrix, where each column is a knockoff copy of GK.Zscore_0.
</p>
</td></tr>
<tr><td><code>T_0</code></td>
<td>

<p>Feature importance score of original data: square of GK.Zscore_0.
</p>
</td></tr>
<tr><td><code>T_k</code></td>
<td>

<p>Feature importance score of knockoff copies: square of GK.Zscore_k.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# We use genetic data as an example
library(GhostKnockoff)

# load example vcf file from package "seqminer", this serves as the reference panel
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])
example.G &lt;- example.G[,apply(example.G,2,sd)!=0]
example.G &lt;- example.G[,1:100]

# compute correlation among variants
cor.G&lt;-matrix(as.numeric(corpcor::cor.shrink(example.G)), nrow=ncol(example.G))

# fit null model
fit.prelim&lt;-GhostKnockoff.prelim(cor.G,M=5,method='asdp',max.size=500)

### if only one study is involved
Zscore_0&lt;-as.matrix(rnorm(nrow(cor.G))) # hypothetical Z-scores
Zscore_0&lt;-Zscore_0+rbinom(nrow(cor.G),size=2,0.1) # set causal
n.study&lt;-c(5000)

# knockoff scores for each block, this can be run in parallel too
GK.stat&lt;-GhostKnockoff.fit(Zscore_0,n.study,fit.prelim,gamma=1,weight.study=NULL)

### if multiple studies are involved, for a meta-analysis

# compute study correlation
Zscore_0&lt;-cbind(rnorm(nrow(cor.G)),rnorm(nrow(cor.G))) # hypothetical Z-scores
Zscore_0&lt;-Zscore_0+rbinom(nrow(cor.G),size=2,0.1) # set causal
cor.study&lt;-GhostKnockoff.GetCorStudy(Zscore_0,fit.prelim)

# note that all steps above can be run in parallel for nonoverlapping blocks of the genome.
# Then the overall study correlation can be computed by averaging cor.study of all blocks.

# compute optimal weights and study dependency using overall study correlaton
n.study&lt;-c(5000,7500)
Meta.prelim&lt;-GhostKnockoff.prelim.Meta(cor.study, n.study)
gamma&lt;-Meta.prelim$gamma
weight.study&lt;-Meta.prelim$w_opt

# knockoff scores for each block, this can be run in parallel too
GK.stat&lt;-GhostKnockoff.fit(Zscore_0,n.study,fit.prelim,gamma=gamma,weight.study=weight.study)

</code></pre>

<hr>
<h2 id='GhostKnockoff.GetCorStudy'>
Calculate study correlation
</h2><span id='topic+GhostKnockoff.GetCorStudy'></span>

<h3>Description</h3>

<p>This function computes correlation among studies given Z-scores and the output of GhostKnockoff.prelim.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GhostKnockoff.GetCorStudy(Zscore_0, fit.prelim)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GhostKnockoff.GetCorStudy_+3A_zscore_0">Zscore_0</code></td>
<td>

<p>A p*K Z-score matrix, where p is the number of variants and K is the number of studies. Variants not observed in the study should be coded as NA.
</p>
</td></tr>
<tr><td><code id="GhostKnockoff.GetCorStudy_+3A_fit.prelim">fit.prelim</code></td>
<td>

<p>The output of function &quot;GhostKnockoff.prelim&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>cor.study</code></td>
<td>

<p>The correlation among studies.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# We use genetic data as an example
library(GhostKnockoff)

# load example vcf file from package "seqminer", this serves as the reference panel
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])
example.G &lt;- example.G[,apply(example.G,2,sd)!=0]
example.G &lt;- example.G[,1:100]

# compute correlation among variants
cor.G&lt;-matrix(as.numeric(corpcor::cor.shrink(example.G)), nrow=ncol(example.G))

# fit null model
fit.prelim&lt;-GhostKnockoff.prelim(cor.G,M=5,method='asdp',max.size=500)

# compute study correlation
Zscore_0&lt;-cbind(rnorm(nrow(cor.G)),rnorm(nrow(cor.G))) # hypothetical Z-scores
Zscore_0&lt;-Zscore_0+rbinom(nrow(cor.G),size=2,0.1) # set causal
cor.study&lt;-GhostKnockoff.GetCorStudy(Zscore_0,fit.prelim)

</code></pre>

<hr>
<h2 id='GhostKnockoff.prelim'>
Preliminary data management for GhostKnockoff
</h2><span id='topic+GhostKnockoff.prelim'></span>

<h3>Description</h3>

<p>This function does the preliminary data management and pre-computes matrices for GhostKnockoff inference, given pre-computed correlation matrix of the variants (e.g. using reference panel in genetic studies). The output will be passed to the other functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GhostKnockoff.prelim(cor.G,M=5,method='asdp',max.size=500)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GhostKnockoff.prelim_+3A_cor.g">cor.G</code></td>
<td>

<p>The pre-computed correlation matrix of the variants.
</p>
</td></tr>
<tr><td><code id="GhostKnockoff.prelim_+3A_m">M</code></td>
<td>

<p>Hypothetical number of knockoffs per variant. The default is 5 for multiple knockoff inference.
</p>
</td></tr>
<tr><td><code id="GhostKnockoff.prelim_+3A_method">method</code></td>
<td>

<p>Either &quot;sdp&quot; or &quot;asdp&quot; (default: &quot;asdp&quot;). This determines the method that will be used to minimize the correlation between the original variables and the knockoffs.
</p>
</td></tr>
<tr><td><code id="GhostKnockoff.prelim_+3A_max.size">max.size</code></td>
<td>

<p>The maximum number in each cluster in the &quot;asdp&quot; method. The default is 500. It will be ignored for &quot;sdp&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list that will be passed to function GhostKnockoff.fit().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# We use genetic data as an example
library(GhostKnockoff)

# load example vcf file from package "seqminer", this serves as the reference panel
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])
example.G &lt;- example.G[,apply(example.G,2,sd)!=0]
example.G &lt;- example.G[,1:100]

# compute correlation among variants
cor.G&lt;-matrix(as.numeric(corpcor::cor.shrink(example.G)), nrow=ncol(example.G))

# fit null model
GhostKnockoff.prelim(cor.G,M=5,method='asdp',max.size=500)
</code></pre>

<hr>
<h2 id='GhostKnockoff.prelim.Meta'>
Additional preliminary data management for GhostKnockoff if multiple studies are involved
</h2><span id='topic+GhostKnockoff.prelim.Meta'></span>

<h3>Description</h3>

<p>This function compute study dependency gamma and the optimal weights to combine multiple studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GhostKnockoff.prelim.Meta(cor.study, n.study)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GhostKnockoff.prelim.Meta_+3A_cor.study">cor.study</code></td>
<td>

<p>The correlation among studies.
</p>
</td></tr>
<tr><td><code id="GhostKnockoff.prelim.Meta_+3A_n.study">n.study</code></td>
<td>

<p>A vector of length K, where each element is the study sample size.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>w_opt</code></td>
<td>

<p>Optimal weights to combine multiple studies.
</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>

<p>study dependency.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# We use genetic data as an example
library(GhostKnockoff)

# load example vcf file from package "seqminer", this serves as the reference panel
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])
example.G &lt;- example.G[,apply(example.G,2,sd)!=0]
example.G &lt;- example.G[,1:100]

# compute correlation among variants
cor.G&lt;-matrix(as.numeric(corpcor::cor.shrink(example.G)), nrow=ncol(example.G))

# fit null model
fit.prelim&lt;-GhostKnockoff.prelim(cor.G,M=5,method='asdp',max.size=500)

# compute study correlation
Zscore_0&lt;-cbind(rnorm(nrow(cor.G)),rnorm(nrow(cor.G))) # hypothetical Z-scores
Zscore_0&lt;-Zscore_0+rbinom(nrow(cor.G),size=2,0.1) # set causal
cor.study&lt;-GhostKnockoff.GetCorStudy(Zscore_0,fit.prelim)

# compute optimal weights and study dependency
n.study&lt;-c(5000,7500)
Meta.prelim&lt;-GhostKnockoff.prelim.Meta(cor.study, n.study)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
