<!DOCTYPE html><html lang="en"><head><title>Help for package GiANT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GiANT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#countdata'>
<p>Example data</p></a></li>
<li><a href='#createSummaryTable'>
<p>Create an overview table for an analysis</p></a></li>
<li><a href='#evaluateGeneSetUncertainty'>
<p>Quantify gene set uncertainty</p></a></li>
<li><a href='#Filter+20gene+20sets'>
<p>Filtering of gene sets</p></a></li>
<li><a href='#Gene+20set+20analysis'>
<p>Main interface for enrichment analyses.</p></a></li>
<li><a href='#GeneLevelStatistics'>
<p>Gene-level statistics</p></a></li>
<li><a href='#GeneSetStatistics'>
<p>Gene set statistics</p></a></li>
<li><a href='#GiANT-package'>
<p>Enrichment analysis</p></a></li>
<li><a href='#GlobalAnalysis'>
<p>Global analyses</p></a></li>
<li><a href='#gsAnalysis'>
<p>Gene set analysis.</p></a></li>
<li><a href='#hist'>
<p>Null distribution histogram and statistic of the input set for enrichment analyses.</p></a></li>
<li><a href='#labels'>
<p>Example data to label samples in countdata (rows) in two classes 0 and 1</p></a></li>
<li><a href='#mergeProbesForGenes'>
<p>Merge multiple probes for one gene</p></a></li>
<li><a href='#parse+20GMT+20files'>
<p>Venn Euler Diagramm</p></a></li>
<li><a href='#pathways'>
<p>Example data for pathways to to gene set enrichtment analyses</p></a></li>
<li><a href='#plot'>
<p>Plots the results of an uncertainty analysis.</p></a></li>
<li><a href='#plotOverrepresentation'>
<p>Plot overlap of gene sets and core set</p></a></li>
<li><a href='#predefinedAnalyses'>
<p>Predefined enrichment analyses</p></a></li>
<li><a href='#preprocessGeneSets'>
<p>Eliminate unknown genes from gene sets</p></a></li>
<li><a href='#SignificanceAssessment'>
<p>Significance assessment</p></a></li>
<li><a href='#summary.gsaResult'>
<p>Summarize gene set analysis results</p></a></li>
<li><a href='#Transformations'>
<p>Transformations</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Set Uncertainty in Enrichment Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-09-02</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hans A. Kestler &lt;hans.kestler@uni-ulm.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Toolbox for various enrichment analysis methods and quantification of uncertainty of gene sets, Schmid et al. (2016) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw030">doi:10.1093/bioinformatics/btw030</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>parallel, graphics, grDevices, methods, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>fdrtool, st, limma, globaltest, DESeq2, GlobalAncova</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-03 12:20:46 UTC; julian_schwab</td>
</tr>
<tr>
<td>Author:</td>
<td>Florian Schmid [aut],
  Christoph Muessel [aut],
  Johann M. Kraus [aut],
  Hans A. Kestler [aut],
  Hans A. Kestler [cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-05 16:50:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='countdata'>
Example data 
</h2><span id='topic+countdata'></span>

<h3>Description</h3>

<p>Randomly generated example data representing a count matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exampleData)</code></pre>


<h3>Details</h3>

<p>The data consists of a variable <code>countdata</code> with a matrix with 1113 genes and 96 samples. The count values are randomly generated numbers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exampleData)

print(countdata)

</code></pre>

<hr>
<h2 id='createSummaryTable'>
Create an overview table for an analysis
</h2><span id='topic+createSummaryTable'></span>

<h3>Description</h3>

<p>Creates a data frame summarizing an analysis. This table has one row per gene set, each comprising the adjusted and unadjusted p-values and the number of genes for the set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createSummaryTable(object,
		orderBy = c("adjustedPValues", "rawPValues", "geneSetName"),
		significantOnly = FALSE,
		signLevel = object$signLevel)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createSummaryTable_+3A_object">object</code></td>
<td>

<p>A result object as returned by <code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>.
</p>
</td></tr>
<tr><td><code id="createSummaryTable_+3A_orderby">orderBy</code></td>
<td>

<p>Specifies which field should be used for the row ordering. By default, rows are ordered according to the adjusted p-values.
</p>
</td></tr>
<tr><td><code id="createSummaryTable_+3A_significantonly">significantOnly</code></td>
<td>

<p>Specifies whether all gene sets (<code>significantOnly=FALSE</code>) or only the statistically significant gene sets (<code>significantOnly=TRUE</code>) should be included in the table. 
</p>
</td></tr>
<tr><td><code id="createSummaryTable_+3A_signlevel">signLevel</code></td>
<td>

<p>If <code>significantOnly=TRUE</code>, this specifies the significance level for the results that should be included in the table. By default, the original significance level of the analysis is used.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with one row for each included gene set and the columns &quot;adjustedPValues&quot;, &quot;rawPValues&quot;, &quot;geneSetName&quot; and &quot;geneSetSize&quot;. For overrepresentation analyses, there is an additional column &quot;intersectSize&quot; specifying the size of the intersection of the core set and the corresponding gene set.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+hist.gsaResult">hist.gsaResult</a></code>, <code><a href="#topic+summary.gsaResult">summary</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load data
data(exampleData)
  # perform gene set analyses for several pathways
  res &lt;- geneSetAnalysis(
  	# global parameters
	  dat = countdata,
	  geneSets = pathways, 
	  analysis = analysis.averageCorrelation(), 
	  # additional parameters for analysis.averageCorrelation
	  labs = labels,
	  numSamples = 10)

  tab &lt;- createSummaryTable(res)
</code></pre>

<hr>
<h2 id='evaluateGeneSetUncertainty'>
Quantify gene set uncertainty
</h2><span id='topic+evaluateGeneSetUncertainty'></span>

<h3>Description</h3>

<p>A robustness measure that quantifies the uncertainty of a gene set  by performing a resampling experiment and can be used in the <code>robustness</code> parameter of <code>gsAnalysis</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evaluateGeneSetUncertainty(
	...,
	dat,
	geneSet,
	analysis,
	numSamplesUncertainty,
	blockSize = 1,
	k = seq(0.01, 0.99, by=0.01),
	signLevel = 0.05,
	preprocessGeneSet = FALSE,
	cluster = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="evaluateGeneSetUncertainty_+3A_...">...</code></td>
<td>

<p>Additional parameters for the different steps of the analysis pipeline, depending on the concrete configuration supplied in <code>analysis</code>.
</p>
</td></tr>
<tr><td><code id="evaluateGeneSetUncertainty_+3A_dat">dat</code></td>
<td>

<p>A numeric matrix of gene expression values for all analyzed genes. Here, each row corresponds to one gene, and each column corresponds to one sample. The rows must be named with the gene names used in the gene sets.
</p>
</td></tr>
<tr><td><code id="evaluateGeneSetUncertainty_+3A_geneset">geneSet</code></td>
<td>

<p>A vector containing the names of genes in a gene set. All genes set must correspond to the row names of <code>dat</code>.
</p>
</td></tr>
<tr><td><code id="evaluateGeneSetUncertainty_+3A_analysis">analysis</code></td>
<td>

<p>The parameters of the analysis that is applied to the perturbed copies of the gene set. These parameters are described by an object of class <code>gsAnalysis</code> as returned by the function <code><a href="#topic+gsAnalysis">gsAnalysis</a></code> or the predefined analysis descriptors in <code><a href="#topic+predefinedAnalyses">predefinedAnalyses</a></code>.
</p>
</td></tr>
<tr><td><code id="evaluateGeneSetUncertainty_+3A_numsamplesuncertainty">numSamplesUncertainty</code></td>
<td>

<p>The number of resampling experiments which should be applied to estimate the robustness of <code>geneSet</code>.
</p>
</td></tr>
<tr><td><code id="evaluateGeneSetUncertainty_+3A_blocksize">blockSize</code></td>
<td>

<p>Number of genes in one resampled block.
</p>
</td></tr>
<tr><td><code id="evaluateGeneSetUncertainty_+3A_k">k</code></td>
<td>

<p>A <code>vector</code> of percentages of genes in the randomized gene sets that should be taken from the original gene set. The remaining genes are chosen randomly. For each value a resampling experiment is performed.
</p>
</td></tr>
<tr><td><code id="evaluateGeneSetUncertainty_+3A_signlevel">signLevel</code></td>
<td>

<p>The significance level for the significance assessment of the gene sets (defaults to <code>0.05</code>). 
</p>
</td></tr>
<tr><td><code id="evaluateGeneSetUncertainty_+3A_preprocessgeneset">preprocessGeneSet</code></td>
<td>

<p>Specifies whether the gene sets in <code>geneSets</code> should be preprocessed or not. If set to <code>TRUE</code>, all genes that are not part of the data set (i.e. not in <code>rownames(dat)</code>) are removed from the gene sets.
</p>
</td></tr>
<tr><td><code id="evaluateGeneSetUncertainty_+3A_cluster">cluster</code></td>
<td>

<p>If the analyses should be applied in parallel for the different values of <code>k</code>, this parameter must hold an initialized cluster as returned by <code><a href="parallel.html#topic+makeCluster">makeCluster</a></code>. If this parameter is <code>NULL</code>, the analyses are performed sequentially.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The uncertainty analysis repeatedly replaces parts of the original gene sets by random genes and calculating the gene set statistics for these randomized gene sets. This yields a distribution of gene set statistic values for slightly modified variants of the original gene set.
</p>


<h3>Value</h3>

<p>Returns a list (of class <code>uncertaintyResult</code>) with the following elements:
</p>
<table role = "presentation">
<tr><td><code>uncertainty</code></td>
<td>

<p>The calculated stability of the original gene set.
</p>
</td></tr>
<tr><td><code>confidenceValues</code></td>
<td>

<p>A matrix of quantiles of <code>gssValues</code> (signLevel, 0.5, 1-signLevel). One row for each value in <code>k</code>.
</p>
</td></tr>
<tr><td><code>uncertaintyEvaluations</code></td>
<td>

<p>A list with one entry per value in <code>k</code> containing the following elements: 
</p>

<ul>
<li><p> Quantiles of <code>gssValues</code>: signLevel, 0.5, 1-signLevel.
</p>
</li>
<li><p> gssValues: A vector of gene set statistic values, one for each randomly sampled gene set.
</p>
</li>
<li><p> uncertainGeneSets: A matrix containing all partially random gene sets.
</p>
</li>
<li><p> k: The percentage of genes in the randomized gene sets taken from the original gene set.
</p>
</li></ul>

</td></tr>
<tr><td><code>signLevel</code></td>
<td>

<p>The significance level used for this analysis.
</p>
</td></tr>
<tr><td><code>originalGeneSetValues</code></td>
<td>

<p>Result of <code>geneSetAnalysis</code> for the original <code>geneSet</code>.
</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>, <code><a href="#topic+gls">gls</a></code>, <code><a href="#topic+transformation">transformation</a></code>, <code><a href="#topic+gss">gss</a></code>, <code><a href="#topic+plot.uncertaintyResult">plot.uncertaintyResult</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exampleData)

  res &lt;- evaluateGeneSetUncertainty(
  	# parameters for evaluateGeneSetUncertainty
  	dat = countdata,
  	geneSet = pathways[[1]],
  	analysis = analysis.averageCorrelation(),
  	numSamplesUncertainty = 10,
  	k = seq(0.1,0.9, by=0.1),
  	# additional parameters for analysis.averageCorrelation
  	labs = labels,
  	numSamples = 10)
</code></pre>

<hr>
<h2 id='Filter+20gene+20sets'>
Filtering of gene sets
</h2><span id='topic+filterGeneSets'></span>

<h3>Description</h3>

<p>Filters gene sets according to different criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterGeneSets(
	geneSets,
	includedGenes = NULL,
	minIntersectSize = length(includedGenes),
	adjMatrix = NULL,
	steps = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Filter+2B20gene+2B20sets_+3A_genesets">geneSets</code></td>
<td>

<p>A list of gene sets, where each gene set is a vector of gene names corresponding to the row names of <code>dat</code>.
</p>
</td></tr>
<tr><td><code id="Filter+2B20gene+2B20sets_+3A_includedgenes">includedGenes</code></td>
<td>

<p>A vector of gene names whose occurence in each of the gene sets is checked. The further parameters how these genes are used to filter gene sets
</p>
</td></tr>
<tr><td><code id="Filter+2B20gene+2B20sets_+3A_minintersectsize">minIntersectSize</code></td>
<td>

<p>If this parameter is not <code>NULL</code>, only gene sets with an intersection of at least <code>minIntersectSize</code> genes with respect to <code>includedGenes</code> (or <code>includedGenes</code> expanded by its interactions if <code>adjMatrix</code> and <code>steps</code> are supplied) are included in the result set. By default, this is the size of <code>includedGenes</code>, requiring <code>includedGenes</code> to be a subset of each gene set.
</p>
</td></tr>
<tr><td><code id="Filter+2B20gene+2B20sets_+3A_adjmatrix">adjMatrix</code></td>
<td>

<p>An optional adjacency matrix in which an entry is <code>1</code> if there is a direct interaction between the corresponding genes and <code>0</code> otherwise. If this is non-null, the set of genes in <code>includedGenes</code> is expanded by adding all genes whose distance in the adjacency graph is at most <code>steps</code>.
</p>
</td></tr>
<tr><td><code id="Filter+2B20gene+2B20sets_+3A_steps">steps</code></td>
<td>

<p>The maximum distance of interacting genes to the genes in <code>includedGenes</code> according to <code>adjMatrix</code> to be added to the expanded gene list. E.g., <code>steps = 1</code> means that all genes which are direct interaction partners of the initial genes in <code>includedGenes</code> are added to <code>includedGenes</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a filtered list of gene sets with the same structure as <code>geneSets</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+preprocessGs">preprocessGs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geneSets &lt;- list(
		gs1 = paste("gene",1:20,sep=""),
		gs2 = paste("gene",50:60,sep=""),
		gs3 = paste("gene",90:92,sep=""),
		gs4 = paste("gene",55:65,sep="")
	)

newGeneSets1 &lt;- filterGeneSets(
	geneSets = geneSets,
	includedGenes = c("gene55","gene60"))

newGeneSets2 &lt;- filterGeneSets(
	geneSets = geneSets,
	includedGenes = c("gene1","gene55","gene20","gene100"),
	minIntersectSize = 2)

examplePathway &lt;- c("gene1","gene2","gene3","gene4")
pathwayAdjMatrix &lt;- matrix(0,100,100)
rownames(pathwayAdjMatrix) &lt;- paste("gene",1:100,sep="")
colnames(pathwayAdjMatrix) &lt;- paste("gene",1:100,sep="")

# gene1 interacts with gene2 and gene3
# -&gt; step 1 if gene1 is starting point
pathwayAdjMatrix[1,2:3] &lt;- 1
pathwayAdjMatrix[2:3,1] &lt;- 1

# gene3 interacts with gene4
# -&gt; step 2 if gene1 is starting point
pathwayAdjMatrix[3,4] &lt;- 1
pathwayAdjMatrix[4,3] &lt;- 1

newGeneSets3 &lt;- filterGeneSets(
	geneSets = geneSets,
	includedGenes = "gene1",
	adjMatrix = pathwayAdjMatrix,
	steps = 2)
</code></pre>

<hr>
<h2 id='Gene+20set+20analysis'>
Main interface for enrichment analyses.
</h2><span id='topic+geneSetAnalysis'></span>

<h3>Description</h3>

<p>The main function of the package that performs a gene set analysis for a list of gene sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneSetAnalysis(
	...,
	dat,
	geneSets,
	analysis,
	signLevel = 0.05,
	preprocessGeneSets = FALSE,
	adjustmentMethod = p.adjust.methods,
	cluster = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Gene+2B20set+2B20analysis_+3A_...">...</code></td>
<td>

<p>Additional parameters for the different steps of the analysis pipeline, depending on the concrete configuration supplied in <code>analysis</code>.
</p>
</td></tr>
<tr><td><code id="Gene+2B20set+2B20analysis_+3A_dat">dat</code></td>
<td>

<p>A numeric matrix of gene expression values for all analyzed genes. Here, each row corresponds to one gene, and each column corresponds to one sample. The rows must be named with the gene names used in the gene sets.
</p>
</td></tr>
<tr><td><code id="Gene+2B20set+2B20analysis_+3A_genesets">geneSets</code></td>
<td>

<p>A list of gene sets, where each gene set is a vector of gene names corresponding to the row names of <code>dat</code>.
</p>
</td></tr>
<tr><td><code id="Gene+2B20set+2B20analysis_+3A_analysis">analysis</code></td>
<td>

<p>An object of type <code>gsAnalysis</code> as returned by <code><a href="#topic+gsAnalysis">gsAnalysis</a></code> or by the predefined configurations (see <code><a href="#topic+predefinedAnalyses">predefinedAnalyses</a></code>).
</p>
</td></tr>
<tr><td><code id="Gene+2B20set+2B20analysis_+3A_signlevel">signLevel</code></td>
<td>

<p>The significance level for the significance assessment of the gene sets (defaults to <code>0.05</code>). 
</p>
</td></tr>
<tr><td><code id="Gene+2B20set+2B20analysis_+3A_preprocessgenesets">preprocessGeneSets</code></td>
<td>

<p>Specifies whether the gene sets in <code>geneSets</code> should be preprocessed or not. If set to <code>TRUE</code>, all genes that are not part of the data set (i.e. not in <code>rownames(dat)</code>) are removed from the gene sets.
</p>
</td></tr>
<tr><td><code id="Gene+2B20set+2B20analysis_+3A_adjustmentmethod">adjustmentMethod</code></td>
<td>

<p>The method to use for the adjustment for multiple testing (see <code>method</code> parameter of <code><a href="stats.html#topic+p.adjust">p.adjust</a></code> for possible values).
</p>
</td></tr>
<tr><td><code id="Gene+2B20set+2B20analysis_+3A_cluster">cluster</code></td>
<td>

<p>If the analyses should be applied in parallel for the gene sets, this parameter must hold an initialized cluster as returned by <code><a href="parallel.html#topic+makeCluster">makeCluster</a></code>. If this parameter is <code>NULL</code>, the analyses are performed sequentially.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the main interface function of the package for gene set enrichment analyses. Analyses usually consist of a pipeline of steps. Often, the first step is the calculation of a summary statistic for the relation of each gene to the class labels. These values or transformations thereof are employed to calculate a gene set statistic for each of the supplied gene sets. The significance of gene set enrichments can be determined according to different methods, and the robustness of gene sets can be evaluated by slightly modifying the gene sets. To provide a flexible mechanism for the plethora of different approaches arising from the different choices, basic pipeline configurations are encapsulated in <code>gsAnalysis</code> objects which can be created using the <code><a href="#topic+gsAnalysis">gsAnalysis</a></code> function. Ready-to-use configuration objects for certain well-known methods are included in the package (see <code><a href="#topic+predefinedAnalyses">predefinedAnalyses</a></code>). Parameters of the chosen analysis pipeline can be set in the <code>...</code> parameter.
</p>


<h3>Value</h3>

<p>An object of the type <code>gsaResult</code> with the following elements:
</p>
<table role = "presentation">
<tr><td><code>adjustedPValues</code></td>
<td>
<p>A vector of p-values, one for each gene set. These values are already adjusted for multiple testing according to the <code>adjustmentMethod</code> parameter.
</p>
</td></tr>
<tr><td><code>rawPValues</code></td>
<td>
<p>The raw unadjusted p-values, one for each gene set.
</p>
</td></tr>
<tr><td><code>res.all</code></td>
<td>

<p>A list comprising the detailed results for each gene set. Each element of this list is another list with the following components:
</p>
<p><code>pValue</code>: The raw (unadjusted) p-value for the gene set.
</p>
<p><code>geneSetValues</code>: If <code>analysis</code> is a global analysis, this is the object returned by the method for the corresponding gene set. For an analysis pipeline, this holds the values of the gene-level statistic, the transformed values and the values of the gene set statistic (see also <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>).
</p>
<p><code>significanceValues</code>: Gene set statistics for each randomly drawn gene set for significance assessment and a <code>list</code> of this gene sets. Only set for analysis of type 'geneSetAnalysis'. <code>NULL</code> for 'global' analysis.
</p>
<p><code>geneSet</code>: The supplied gene set.
</p>
</td></tr>
<tr><td><code>signLevel</code></td>
<td>
<p>The significance level used for this analysis.</p>
</td></tr>
<tr><td><code>analysis</code></td>
<td>
<p>The performed analysis (of type <code>gsAnalysis</code>).</p>
</td></tr>
<tr><td><code>analysisType</code></td>
<td>
<p>A character string identifying the analysis as an enrichment analysis pipeline (<code>"geneSetAnalysis"</code>) or as a global analysis (<code>"global"</code>).</p>
</td></tr>
<tr><td><code>adjustmentMethod</code></td>
<td>
<p>The method used to adjust the p-values in <code>adjustedPValues</code></p>
</td></tr>
</table>


<h3>References</h3>

<p>Ackermann, M., Strimmer, K. (2009) A general modular framework for gene set enrichment analysis. <em>BMC Bioinformatics</em>, <b>10</b>(1), 47.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gsAnalysis">gsAnalysis</a></code>, <code><a href="#topic+gls">gls</a></code>, <code><a href="#topic+transformation">transformation</a></code>, <code><a href="#topic+gss">gss</a></code>, <code><a href="#topic+global">global</a></code>, <code><a href="#topic+significance">significance</a></code>, <code><a href="#topic+evaluateGeneSetUncertainty">evaluateGeneSetUncertainty</a></code>, <code><a href="#topic+hist.gsaResult">hist.gsaResult</a></code>, <code><a href="#topic+preprocessGs">preprocessGs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load data
data(exampleData)
  
  # apply predefined analysis for gene set enrichment analysis
  res &lt;- geneSetAnalysis(	
  	# parameters for geneSetAnalysis
  	dat = countdata,
  	geneSets = pathways[1],
  	analysis = analysis.averageCorrelation(),
  	adjustmentMethod = "fdr",
  	# additional parameters for analysis.averageCorrelation
  	labs = labels,
  	method = "pearson",
  	numSamples = 10)
</code></pre>

<hr>
<h2 id='GeneLevelStatistics'>
Gene-level statistics
</h2><span id='topic+gls'></span><span id='topic+gls.cor'></span><span id='topic+gls.regression'></span><span id='topic+gls.foldChange'></span><span id='topic+gls.tStatistic'></span><span id='topic+gls.moderateTStatistic'></span><span id='topic+gls.nBinomTest'></span>

<h3>Description</h3>

<p>Functions to calculate the gene-level statistic, as used in the <code>gls</code> parameter of <code>gsAnalysis</code>.
A gene-level statistic calculates a measure of correlation between the expression of a gene and the class labels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gls.cor(dat, labs, method = "pearson")

gls.regression(dat, labs)

gls.foldChange(dat, labs, logMeasurements = TRUE)

gls.tStatistic(dat, labs, pValue = FALSE, alternative = "two.sided")

gls.moderateTStatistic(dat,labs)

gls.nBinomTest(dat, labs,
	returnValue = c("pval", "qval", "foldChange", "log2FoldChange"),
	dispersionMethod = "blind",
	dispersionSharingMode = "fit-only",
	dispersionFitType = "local")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GeneLevelStatistics_+3A_dat">dat</code></td>
<td>

<p>A numeric matrix of gene expression values for all analyzed genes. Here, each row corresponds to one gene, and each column corresponds to one sample. The rows must be named with the gene names used in the gene sets.
</p>
</td></tr>
<tr><td><code id="GeneLevelStatistics_+3A_labs">labs</code></td>
<td>

<p>A vector of class labels for the samples in <code>dat</code>. 
</p>
</td></tr>
<tr><td><code id="GeneLevelStatistics_+3A_logmeasurements">logMeasurements</code></td>
<td>

<p>For <code>gls.foldChange</code>, whether the values in <code>dat</code> are logarithmized (<code>logMeasurements=TRUE</code>) or not (<code>logMeasurements=FALSE</code>).
</p>
</td></tr>
<tr><td><code id="GeneLevelStatistics_+3A_method">method</code></td>
<td>

<p>For <code>gls.cor</code>, the correlation method to be used (see <code><a href="stats.html#topic+cor">cor</a></code>).
</p>
</td></tr>
<tr><td><code id="GeneLevelStatistics_+3A_pvalue">pValue</code></td>
<td>

<p>For <code>gls.tStatistic</code>, this specifies whether the p-value (<code>pValue=TRUE</code>) or the test statistic (<code>pValue=FALSE</code>) of the t test should be returned.
</p>
</td></tr>
<tr><td><code id="GeneLevelStatistics_+3A_alternative">alternative</code></td>
<td>

<p>For <code>gls.tStatistic</code>, this specifies the alternative of the t-test. See also <code><a href="stats.html#topic+t.test">t.test</a></code>.
</p>
</td></tr>
<tr><td><code id="GeneLevelStatistics_+3A_returnvalue">returnValue</code></td>
<td>

<p>For <code>gls.nBinomTest</code>, this determines the type of values values that should be returned. <code>"pval"</code> returns the raw p-values, <code>"qval"</code> returns the p-values adjusted by the FDR, <code>"foldChange"</code> returns the fold changes, and <code>"log2FoldChange"</code> returns the log2 fold changes. For more details, see <code><a href="DESeq2.html#topic+results">results</a></code>.
</p>
</td></tr>
<tr><td><code id="GeneLevelStatistics_+3A_dispersionmethod">dispersionMethod</code></td>
<td>

<p>For <code>gls.nBinomTest</code>, this specifies how the empirical dispersion is computed (see <code><a href="DESeq2.html#topic+estimateDispersions">estimateDispersions</a></code>).
</p>
</td></tr>
<tr><td><code id="GeneLevelStatistics_+3A_dispersionsharingmode">dispersionSharingMode</code></td>
<td>

<p>For <code>gls.nBinomTest</code>, this specifies which values should be used by <code><a href="DESeq2.html#topic+results">results</a></code> (fitted values or empirical values, see <code><a href="DESeq2.html#topic+estimateDispersions">estimateDispersions</a></code> for more details).
</p>
</td></tr>
<tr><td><code id="GeneLevelStatistics_+3A_dispersionfittype">dispersionFitType</code></td>
<td>

<p>For <code>gls.nBinomTest</code>, this determines the method for fitting the dispersion-mean relation (see <code><a href="DESeq2.html#topic+estimateDispersions">estimateDispersions</a></code>).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Standard functions for the calculation of gene-level statistics (to be used in an analysis pipeline defined by <code>gsAnalysis</code>): 
</p>

<ul>
<li> <p><code>gls.cor</code>: Calculates the correlation of the gene expression values to the class labels.
</p>
</li>
<li> <p><code>gls.regression</code>: Calculates the slope of a linear regression of the gene expression values and the class labels.
</p>
</li>
<li> <p><code>gls.foldChange</code>: Calculates the (standard or log2) fold change between the measurements for the two classes.
</p>
</li>
<li> <p><code>gls.tStatistic</code>: Calculates the p-value or the statistic of a two-sample t test for the measurements of the two classes
</p>
</li>
<li> <p><code>gls.moderateTStatistic</code>: Calculates the moderate t statistic for the measurements of the two classes
</p>
</li>
<li> <p><code>gls.nBinomTest</code>: Applies the negative binomial test for sequencing data based on the <span class="pkg">DESeq2</span> package to test for differences between two classes (see <code><a href="DESeq2.html#topic+results">results</a></code>).
</p>
</li></ul>



<h3>Value</h3>

<p>Each of these function returns a numeric vector of gene-level statistics (one entry per gene).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>, <code><a href="#topic+gss">gss</a></code>, <code><a href="#topic+transformation">transformation</a></code>
</p>

<hr>
<h2 id='GeneSetStatistics'>
Gene set statistics
</h2><span id='topic+gss'></span><span id='topic+gss.mean'></span><span id='topic+gss.sum'></span><span id='topic+gss.wilcoxonRankTest'></span><span id='topic+gss.maxmean'></span><span id='topic+gss.median'></span><span id='topic+gss.enrichmentScore'></span><span id='topic+gss.fisherExactTest'></span><span id='topic+gss.gsz'></span>

<h3>Description</h3>

<p>Functions to calculate a gene set statistic, as used in the <code>gss</code> parameter of <code>gsAnalysis</code>. A gene set statistic summarizes a single gene set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gss.mean(x, geneSetIndices)

gss.sum(x, geneSetIndices)

gss.wilcoxonRankTest(x, geneSetIndices)

gss.maxmean(x, geneSetIndices)

gss.median(x, geneSetIndices)

gss.enrichmentScore(x, geneSetIndices, p = 1)

gss.fisherExactTest(x, geneSetIndices)

gss.gsz(x, geneSetIndices, w1 = 0.2, w2 = 0.5, preVar = 0, varConstant = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GeneSetStatistics_+3A_x">x</code></td>
<td>

<p>A vector comprising one numeric value for each gene in the data set. This vector is usually obtained from the previous step, the gene-level statistic (see <code><a href="#topic+gls">gls</a></code>) or the transformed gene-level statistic (see <code><a href="#topic+transformation">transformation</a></code>).
</p>
</td></tr>
<tr><td><code id="GeneSetStatistics_+3A_genesetindices">geneSetIndices</code></td>
<td>

<p>A vector containing the indices of the genes in the gene set with respect to the full gene set (i.e., the indices of the rows containing the measurements for these genes in <code>dat</code>).
</p>
</td></tr>
<tr><td><code id="GeneSetStatistics_+3A_p">p</code></td>
<td>

<p>Factor for <code>gss.enrichmentScore</code> that specifies the way hits are weighted. For <code>p = 0</code>, the enrichment score is a Kolmogorov-Smirnov statistic. For <code>p = 1</code> (the default), hits are weighted by their correlation.
</p>
</td></tr>
<tr><td><code id="GeneSetStatistics_+3A_w1">w1</code></td>
<td>

<p>Weight for the median of the variance estimates for a gene set of size <code>varConstant</code>. Should be between 0 and 1. Default is <code>w1 = 0.2</code>.
</p>
</td></tr>
<tr><td><code id="GeneSetStatistics_+3A_w2">w2</code></td>
<td>

<p>Weight for the median of the variance estimates for a gene set across the whole gene list. Should be between 0 and 1. Default is <code>w2 = 0.5</code>.
</p>
</td></tr>
<tr><td><code id="GeneSetStatistics_+3A_prevar">preVar</code></td>
<td>

<p>Parameter for incorporating the uncertainty of the observations. This is omitted by default (<code>preVar = 0</code>).
</p>
</td></tr>
<tr><td><code id="GeneSetStatistics_+3A_varconstant">varConstant</code></td>
<td>

<p>Reference gene set size used for variance estimates. Default is <code>varConstant = 10</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Standard functions for the calculation of gene set statistics (to be used in an analysis pipeline defined by <code>gsAnalysis</code>):
</p>

<ul>
<li> <p><code>gss.mean</code>: Calculates the mean of the (transformed) gene-level statistic values for the genes in the set.
</p>
</li>
<li> <p><code>gss.sum</code>: Calculates the sum of the (transformed) gene-level statistic values for the genes in the set.
</p>
</li>
<li> <p><code>gss.wilcoxonRankTest</code>: Calculates a Wilcoxon test comparing the (transformed) gene-level statistic values for the genes in the set versus those of the genes not in the set.
</p>
</li>
<li> <p><code>gss.maxmean</code>: Calculates the maximum of the means of positive and negative statistic values, weighted by the overall proportion of positive/negative values (e.g. for correlation scores where the sign denotes the direction). Described in Efron and Tibshirani.
</p>
</li>
<li> <p><code>gss.median</code>: Calculates the median of the (transformed) gene-level statistic values for the genes in the set.
</p>
</li>
<li> <p><code>gss.enrichmentScore</code>: Calculates the enrichment score of the (transformed) gene-level statistic values for the genes in the set, as described in Subramanian et al.
</p>
</li>
<li> <p><code>gss.fisherExactTest</code>: Performs Fisher's exact test to check gene sets for overrepresentation in the differential genes. This should be used used in combination with the transformation <code><a href="#topic+transformation">transformation.adjustAndBinarize</a></code>.
</p>
</li>
<li> <p><code>gss.gsz</code>: Calculates the Gene Set Z-score of the (transformed) gene-level statistic values for the genes in the set. Described in Toronen et al.
</p>
</li></ul>



<h3>Value</h3>

<p>Each method returns a single numeric value, the gene set statistic for the supplied gene set.
</p>


<h3>References</h3>

<p>Efron, B., Tibshirani, R. (2007) On testing the significance of sets of genes. <em>Annals of Applied Statistics</em>, <b>1</b>, 107-129.
</p>
<p>Subramanian, A., Tamayo, P., Mootha, V. K., Mukherjee, S., Ebert, B. L., Gillette, M. A., Paulovich, A., Pomeroy, S. L., Golub, T. R., Lander, E. S., Mesirov, J. P. (2005) Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. <em>Proceedings of the National Academy of Science of the United States of America</em>, <b>102</b>, 15545&ndash;15550.
</p>
<p>Toronen, P., Ojala, P. J., Marttinen, P., Holm L. (2009) Robust extraction of functional signals from gene set analysis using a generalized threshold free scoring function. <em>BMC Bioinformatics</em>, <b>10</b>(1), 307.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>, <code><a href="#topic+gls">gls</a></code>, <code><a href="#topic+transformation">transformation</a></code>
</p>

<hr>
<h2 id='GiANT-package'>
Enrichment analysis
</h2><span id='topic+GiANT'></span>

<h3>Description</h3>

<p>Toolbox for gene set analysis of uncertain gene sets.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> GiANT</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.3</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-04-29</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> Artistic-2.0 </td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>This package provides an approach for evaluating the fuzziness of a gene set. This is done by repeatedly performing gene set analyses on slightly modified versions of the gene set and comparing their enrichment scores. A utility for such uncertainty tests is provided in the <code><a href="#topic+evaluateGeneSetUncertainty">evaluateGeneSetUncertainty</a></code> function.
</p>
<p>The package also comprises a generic framework for different types of enrichment analyses (Ackermann and Strimmer). It establishes a customizeable pipeline that typically consists of the following steps:
</p>

<ul>
<li><p> Calculation of gene-level statistics: A gene-level statistic scores the relationship between the measurements for a specific gene and the class labels. Typical measures include correlation coefficients, the t statistic or the fold change between the groups (see <code><a href="#topic+gls">gls</a></code> for gene-level statistics included in the package). 
</p>
</li>
<li><p> Transformation of gene-level statistic values: Optionally, the gene-level statistic values can be postprocessed, e.g. by taking the absolute value or the square for correlation values or by binarizing or ranking values. See <code><a href="#topic+transformation">transformation</a></code> for transformations included in the package.
</p>
</li>
<li><p> Calculation of gene set statistics: Based on the (possibly transformed) gene-level statistics, the gene set(s) of interest is/are scored. Examples are the median, the mean or the enrichment score of the gene-level statistic values in the gene set(s). See <code><a href="#topic+gss">gss</a></code> for gene set statistics included in the package.
</p>
</li>
<li><p> Significance assessment: To assess the significance of the gene set statistic value(s) with respect to a null distribution, computer-intensive tests are performed. These tests repeatedly sample random label vectors or gene sets and calculate their gene set statistic values. These values can then be compared to the true gene set statistics. See <code><a href="#topic+significance">significance</a></code> for significance assessment methods included in the package.
</p>
</li></ul>

<p>The package represents such analysis pipelines as configuration objects that can be created using the function <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>. For predefined state-of-the-art methods, such as Gene Set Enrichment Analysis (Subramanian et al), Overrepresentation Analysis or Global Ancova (Hummel et al), it provides predefined configurations (see <code><a href="#topic+predefinedAnalyses">predefinedAnalyses</a></code>).
</p>
<p>The main function for standard gene set analyses, <code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, performs enrichment analyses based on pipeline configuration objects.
</p>


<h3>Author(s)</h3>

<p>Florian Schmid, Christoph Müssel, Johann M. Kraus, Hans A. Kestler
</p>
<p>Maintainer: Hans A. Kestler &lt;hans.kestler@uni-ulm.de&gt;
</p>


<h3>References</h3>

<p>Ackermann, M., Strimmer, K. (2009) A general modular framework for gene set enrichment analysis. <em>BMC Bioinformatics</em>, <b>10</b>(1), 47.
</p>
<p>Subramanian, A., Tamayo, P., Mootha, V. K., Mukherjee, S., Ebert, B. L., Gillette, M. A., Paulovich, A., Pomeroy, S. L., Golub, T. R., Lander, E. S., Mesirov, J. P. (2005) Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. <em>Proceedings of the National Academy of Science of the United States of America</em>, <b>102</b>, 15545-15550.
</p>
<p>Hummel, M., Meister, R., Mansmann, U. (2008) Globalancova: exploration and assessment of gene group effects. <em>Bioinformatics</em>, <b>24</b>(1), 78&ndash;85.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  data(exampleData)
  ##################################
  # Example 1: gene set analysis   #
  ##################################
  res &lt;- geneSetAnalysis(  
    # parameters for geneSetAnalysis
    dat = countdata,
    geneSets = pathways[1],
    analysis = analysis.averageCorrelation(),
    adjustmentMethod = "fdr",
    # additional parameters for analysis.averageCorrelation
    labs = labels,
    method = "pearson",
    numSamples = 50)
  
  summary(res, mode="table")
  
  ####################################
  # Example 2: uncertainty analysis  #
  ####################################
  resUncertainty &lt;- evaluateGeneSetUncertainty(  
    # parameters for evaluateGeneSetUncertainty
    dat = countdata,
    geneSet = pathways[[3]],
    analysis = analysis.averageCorrelation(),
    numSamplesUncertainty = 5,
    blockSize = 1,
    k = seq(0.1,0.9,by=0.1),
    # additional parameters for analysis.averageCorrelation
    labs = labels,
    numSamples = 5)
  
  plot(resUncertainty, main = names(pathways[3]))

</code></pre>

<hr>
<h2 id='GlobalAnalysis'>
Global analyses
</h2><span id='topic+global'></span><span id='topic+global.overrepresentation'></span><span id='topic+global.ancova'></span><span id='topic+global.test'></span>

<h3>Description</h3>

<p>Functions to perform global gene set analyses, as used in the <code>globalStat</code> parameter of <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>global.overrepresentation(dat,
	geneSet,
	coreSet)

global.ancova(dat,
	geneSet,
	labs,
	...)

global.test(dat,
	geneSet,
	labs,
	...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GlobalAnalysis_+3A_dat">dat</code></td>
<td>

<p>A numeric matrix of gene expression values for all analyzed genes. Here, each row corresponds to one gene, and each column corresponds to one sample. The rows must be named with the gene names used in the gene sets.
</p>
</td></tr>
<tr><td><code id="GlobalAnalysis_+3A_geneset">geneSet</code></td>
<td>

<p>A gene set in form of a vector of gene names corresponding to the row names of <code>dat</code>.
</p>
</td></tr>
<tr><td><code id="GlobalAnalysis_+3A_coreset">coreSet</code></td>
<td>

<p>A gene set of interest resulting from an analysis of <code>dat</code> that should be compared to <code>geneSet</code> in the overrepresentation analysis. This is also a vector of gene names corresponding to the row names of <code>dat</code>.
</p>
</td></tr>
<tr><td><code id="GlobalAnalysis_+3A_labs">labs</code></td>
<td>
 
<p>A vector of class labels for the samples in <code>dat</code>. 
</p>
</td></tr>
<tr><td><code id="GlobalAnalysis_+3A_...">...</code></td>
<td>
 
<p>Further parameters for  <code><a href="GlobalAncova.html#topic+GlobalAncova">GlobalAncova</a></code> and <code><a href="globaltest.html#topic+gt">gt</a></code>, as defined in the corresponding manual pages. The parameters <code>xx</code>, <code>test.genes</code> and <code>group</code> are set automatically by <code>global.ancova</code>, and the parameters <code>alternative</code>, <code>subsets</code> and <code>response</code> are set automatically by <code>global.test</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Wrapper functions for global gene set analyses.
</p>

<ul>
<li> <p><code>global.overrepresentation</code>:	This function performs an overrepresentation analysis by rating the overlap of <code>geneSet</code> and <code>coreSet</code> with respect to the set of all genes using Fisher's exact test.
</p>
</li>
<li> <p><code>global.ancova</code>:	This function performs a global gene set enrichment analysis using the global ANCOVA method by Hummel et al. It wraps the <code><a href="GlobalAncova.html#topic+GlobalAncova">GlobalAncova</a></code> function in the <span class="pkg">GlobalAncova</span> package.
</p>
</li>
<li> <p><code>global.test</code>: This function performs a global gene set enrichment analysis using a global test by Goeman et al. It wraps the <code><a href="globaltest.html#topic+gt">gt</a></code> function in the <span class="pkg">globaltest</span> package.
</p>
</li></ul>



<h3>Value</h3>

<p>A list containing the following items:
</p>
<table role = "presentation">
<tr><td><code>pValue</code></td>
<td>

<p>The p-value for the significance of <code>geneSet</code>.
</p>
</td></tr>
<tr><td><code>intersectGeneSetCoreSet</code></td>
<td>

<p>This element is only returned in case of an overrepresentation analysis and consists of a vector of genes included in both sets (<code>geneSet</code> and <code>coreSet</code>).
</p>
</td></tr>
<tr><td><code>res.all</code></td>
<td>

<p>The full result object returned by <code><a href="stats.html#topic+fisher.test">fisher.test</a></code>, <code><a href="GlobalAncova.html#topic+GlobalAncova">GlobalAncova</a></code> or <code><a href="globaltest.html#topic+gt">gt</a></code> respectively.
</p>
</td></tr>
</table>


<h3>References</h3>

<p>Hummel, M., Meister, R., Mansmann, U. (2008) GlobalANCOVA: exploration and assessment of gene group effects. <em>Bioinformatics</em>, <b>24</b>(1), 78&ndash;85.
</p>
<p>Goeman, J. J., van de Geer, S. A., de Kort, F., van Houwelingen, H. C. (2004) A global test for groups of genes: testing association with a clinical outcome. <em>Bioinformatics</em>, <b>20</b>(1), 93&ndash;99.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>
</p>

<hr>
<h2 id='gsAnalysis'>
Gene set analysis.
</h2><span id='topic+gsAnalysis'></span>

<h3>Description</h3>

<p>Defines the configuration of an analysis that can be performed using <code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, and returns it as a wrapper object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsAnalysis(name,
	gls = NULL,
	glsParameterNames = NULL,
	transformation = NULL,
	transformationParameterNames = NULL,
	gss = NULL,
	gssParameterNames = NULL,
	globalStat = NULL,
	globalStatParameterNames = NULL,
	significance = NULL,
	significanceParameterNames = NULL,
	testAlternative = c("greater", "less"))

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gsAnalysis_+3A_name">name</code></td>
<td>

<p>A character string describing the analysis.
</p>
</td></tr>
<tr><td><code id="gsAnalysis_+3A_gls">gls</code></td>
<td>

<p>The name of the function that calculates the gene-level statistic for a given dataset. If set to <code>NULL</code>, it is assumed that the input data already comprises gene-level statistic values, and the input is directly supplied to <code>transformation</code>. The first (fixed) parameter of a gls function is the dataset.
</p>
</td></tr>
<tr><td><code id="gsAnalysis_+3A_glsparameternames">glsParameterNames</code></td>
<td>

<p>A character vector of names of the parameters used by the gene-level statistic defined in <code>gls</code>.
</p>
</td></tr>
<tr><td><code id="gsAnalysis_+3A_transformation">transformation</code></td>
<td>

<p>The name of the function that transforms the gene-level statistics values. If set to <code>NULL</code>, the values supplied by <code>gls</code> are directly handed over to <code>gss</code>. The only fixed parameter for transformation is the gene-level statistic (supplied as first parameter).
</p>
</td></tr>
<tr><td><code id="gsAnalysis_+3A_transformationparameternames">transformationParameterNames</code></td>
<td>

<p>A character vector of names of the parameters used by the transformation defined in <code>transformation</code>.
</p>
</td></tr>
<tr><td><code id="gsAnalysis_+3A_gss">gss</code></td>
<td>

<p>The name of the function that calculates the gene set statistics from untransformed or transformed gene-level statistic values. If set to <code>NULL</code>, the values supplied by <code>transformation</code> are directly handed over to <code>significance</code>. Fixed parameters are the transformed values (first parameter) and <code>geneSetIndices</code> containing the (row-) indices of the current gene set genes in the dataset.
</p>
</td></tr>
<tr><td><code id="gsAnalysis_+3A_gssparameternames">gssParameterNames</code></td>
<td>

<p>A character vector vector of names of the parameters used by the gene set statistic defined in <code>gss</code>.
</p>
</td></tr>
<tr><td><code id="gsAnalysis_+3A_globalstat">globalStat</code></td>
<td>

<p>If the gene set analysis consists of a global test and cannot be divided into the calculation of a gene-level statistic, a transformation and the calculation of the gene set statistics, this parameter defines the name of the function that performs this global analysis. In this case, the parameters <code>gls</code>, <code>transformation</code> and <code>gss</code> are ignored. Fixed parameters are <code>dat</code> containing the whole dataset and <code>geneSet</code> containing the current gene set. 
</p>
</td></tr>
<tr><td><code id="gsAnalysis_+3A_globalstatparameternames">globalStatParameterNames</code></td>
<td>

<p>A character vector names of the parameters used by the global analysis defined in <code>globalStat</code>.
</p>
</td></tr>
<tr><td><code id="gsAnalysis_+3A_significance">significance</code></td>
<td>

<p>The name of a method that performs a significance assessment for the gene set statistic values. If set to <code>NULL</code>, <code>geneSetAnalysis</code> does not return p-values, but returns the statistics supplied by <code>gss</code> or <code>globalStat</code>. Fixed parameters are <code>dat</code> containing the whole dataset, <code>geneSet</code> containing the current gene set, <code>analysis</code> with the supplied <code>gsAnalysis</code> and <code>glsValues</code> with (depending on whether a transformation is supplied or not) transformed gene-level statistics for each gene in the dataset.
</p>
</td></tr>
<tr><td><code id="gsAnalysis_+3A_significanceparameternames">significanceParameterNames</code></td>
<td>

<p>A character vector of names of the parameters used by the significance assessment method defined in <code>significance</code>.
</p>
</td></tr>
<tr><td><code id="gsAnalysis_+3A_testalternative">testAlternative</code></td>
<td>

<p>Specifies the alternative hypothesis of the significance test for the gene set enrichment, which may be dependent on the chosen gene set statistic. Must be one of <code>"greater"</code> or <code>"less"</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function provides a way of flexibly defining the steps of the gene set analysis pipeline. This pipeline consists of a subset of the following steps, each of which may have specific parameters:
</p>

<ul>
<li><p> Gene-level analysis: A gene-level statistic scores the relationship between the measurements for a specific gene and the class labels. Typical measures include correlation coefficients, the t statistic or the fold change between the groups (see <code><a href="#topic+gls">gls</a></code> for gene-level statistics included in the package). 
</p>
</li>
<li><p> Transformation of gene-level statistics: Optionally, the gene-level statistic values can be postprocessed, e.g. by taking the absolute value or the square for correlation values or by binarizing or ranking values. See <code><a href="#topic+transformation">transformation</a></code> for transformations included in the package.
</p>
</li>
<li><p> Gene set analysis: Based on the (possibly transformed) gene-level statistics, the gene set(s) of interest is/are scored. Examples are the median, the mean or the enrichment score of the gene-level statistic values in the gene set(s). See <code><a href="#topic+gss">gss</a></code> for gene set statistics included in the package.
</p>
</li>
<li><p> Significance assessment: To assess the significance of the gene set statistic value(s) with respect to a null distribution, computer-intensive tests are performed. These tests repeatedly sample random label vectors or gene sets and calculate their gene set statistic values. These values can then be compared to the true gene set statistics. See <code><a href="#topic+significance">significance</a></code> for significance assessment methods included in the package.
</p>
</li>
<li><p> Global analysis: As an alternative to the above pipeline steps, it is possible to define a single, global method that directly calculates an enrichment p-value for a supplied data set and gene set. See <code><a href="#topic+global">global</a></code> for the global analysis tests included in the package.
</p>
</li></ul>

<p>Several state-of-the-art analyses have predefined configuration objects in which the above steps are defined accordingly (see <code><a href="#topic+predefinedAnalyses">predefinedAnalyses</a></code>).
</p>


<h3>Value</h3>

<p>An object of class <code>gsAnalysis</code> with components corresponding to the above parameters.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+predefinedAnalyses">predefinedAnalyses</a></code>, <code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+evaluateGeneSetUncertainty">evaluateGeneSetUncertainty</a></code>, <code><a href="#topic+gls">gls</a></code>, <code><a href="#topic+transformation">transformation</a></code>, <code><a href="#topic+gss">gss</a></code>, <code><a href="#topic+global">global</a></code>, <code><a href="#topic+significance">significance</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(exampleData)
# defines an analysis that corresponds to gsAna1()
gsa &lt;- gsAnalysis(
	name = "averageCorrelation",
	gls = "gls.cor",
	glsParameterNames = c("labs","method"),
	transformation = "transformation.abs",
	transformationParameterNames = NULL,
	gss = "gss.mean",
	gssParameterNames = NULL,
	globalStat = NULL,
	globalStatParameterNames = NULL,
	significance = "significance.sampling",
	significanceParameterNames = c("numSamples"),
	testAlternative = "greater")
print(gsa)


  # apply the previously defined analysis
  res &lt;- geneSetAnalysis(
  	# global parameters
  	dat = countdata,
  	geneSets = pathways[1], 
  	analysis = gsa, 
  	# parameters for the specific analysis gsAna1
  	labs = labels,
  	numSamples = 10)
</code></pre>

<hr>
<h2 id='hist'>
Null distribution histogram and statistic of the input set for enrichment analyses.
</h2><span id='topic+hist.gsaResult'></span>

<h3>Description</h3>

<p>Plots the distribution of gene set statistic values obtained in different resampling settings of an enrichment analysis, and draws the statistic value of the input set as a vertical line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gsaResult'
hist(x,
	signLevel = x$signLevel,
	subset = NULL,
	ask = FALSE,
	addLegend = TRUE,
	...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hist_+3A_x">x</code></td>
<td>

<p>A result of a call to <code>geneSetAnalysis</code> (see also Details). 
</p>
</td></tr>
<tr><td><code id="hist_+3A_signlevel">signLevel</code></td>
<td>

<p>The significance level that should be applied for the plots. Default is the significance level used for the analysis in <code>x</code>.
</p>
</td></tr>
<tr><td><code id="hist_+3A_subset">subset</code></td>
<td>

<p>Indices for the results that should be included in the diagram. 
</p>
</td></tr>
<tr><td><code id="hist_+3A_ask">ask</code></td>
<td>

<p>If set to true, the plot function will prompt for a user input for each new plot that is shown on an interactive device (see <code><a href="graphics.html#topic+par">par(&quot;ask&quot;)</a></code>). 
</p>
</td></tr>
<tr><td><code id="hist_+3A_addlegend">addLegend</code></td>
<td>

<p>If set to true (default), a <code><a href="graphics.html#topic+legend">legend</a></code> is added to the plot.
</p>
</td></tr>
<tr><td><code id="hist_+3A_...">...</code></td>
<td>

<p>Other parameters which can be used for histograms (see <code><a href="graphics.html#topic+hist">hist</a></code>).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function plots the distribution of gene set statistic values under the null hypothesis. It requires the 
significance assessment step of the enrichment analysis configuration (parameter <code>significance</code> or <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>) to be a computer-intensive testing procedure that yields a distribution of gene set statistic p-values under the null hypothesis. Predefined configurations for which this plot works are <code><a href="#topic+analysis.gsea">analysis.gsea</a></code>, <code><a href="#topic+analysis.averageCorrelation">analysis.averageCorrelation</a></code> and <code><a href="#topic+analysis.averageTStatistic">analysis.averageTStatistic</a></code>.
</p>
<p>A histogram is plotted for the analysis in <code>x</code>. If <code>x</code> includes the analyses for several gene sets, one histogram is plotted for each of the gene sets.
</p>
<p>The statistic value of the input set is depicted as a vertical line.
</p>
<p>The most common graphical parameters can be supplied as vectors (one entry per analyzed gene set) to vary them between the different analyses. These parameters are: <code>main</code>, <code>xlab</code>, <code>ylab</code>.
</p>


<h3>Value</h3>

<p>Returns a list with all the underlying data for the plotted histograms as invisible object.</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+predefinedAnalyses">predefinedAnalyses</a></code>, <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>, <code><a href="#topic+evaluateGeneSetUncertainty">evaluateGeneSetUncertainty</a></code>, <code><a href="#topic+plot.uncertaintyResult">plot.uncertaintyResult</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load data
data(exampleData)
  res &lt;- geneSetAnalysis(
  	# global parameters
  	dat = countdata,
  	geneSets = pathways[3], 
  	analysis = analysis.averageCorrelation(), 
  	# additional parameters for analysis.averageCorrelation
  	labs = labels,
  	p = 1,
  	numSamples = 10)
  
  # plot the histogram for the cell cycle control gene set
  hist(res, main = names(pathways[3]))
</code></pre>

<hr>
<h2 id='labels'>
Example data to label samples in countdata (rows) in two classes 0 and 1
</h2><span id='topic+labels'></span>

<h3>Description</h3>

<p>Randomly generated example data to give samples in countdata a class label 0 or 1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exampleData)</code></pre>


<h3>Details</h3>

<p>The data consists of a variable <code>labels</code> with a vector of 96 values. The values are a random sequence of 0's and 1's.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exampleData)

print(labels)

</code></pre>

<hr>
<h2 id='mergeProbesForGenes'>
Merge multiple probes for one gene
</h2><span id='topic+mergeProbesForGenes'></span>

<h3>Description</h3>

<p>Merges all probes belonging to the same gene by identifying duplicate row names in a data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeProbesForGenes(dat,
	method = c("mean", "max", "min", "median"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mergeProbesForGenes_+3A_dat">dat</code></td>
<td>

<p>A numeric matrix of gene expression values for all analyzed genes. Here, each row corresponds to one probe, and each column corresponds to one sample. The rows must be named with the gene names and may contain duplicates if multiple probes correspond to the same gene.
</p>
</td></tr>
<tr><td><code id="mergeProbesForGenes_+3A_method">method</code></td>
<td>
  
<p>The method which should be used to merge probes entries in <code>dat</code>. Depending on the chosen method, the merged value for a gene and a specific sample is defined as the mean value, the maximum value, the minimum value or the median of all probes of this sample belonging to the gene.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of the same structure as <code>dat</code>, but possibly with fewer rows if probes were merged.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- matrix(1:6, nrow=3, ncol=2)
rownames(dat) &lt;- c("g1", "g2", "g1")

newDat &lt;- mergeProbesForGenes(dat, method = "mean")
</code></pre>

<hr>
<h2 id='parse+20GMT+20files'>
Venn Euler Diagramm
</h2><span id='topic+parseGmt'></span>

<h3>Description</h3>

<p>Parses a GMT file as downloadable from MSigDB (presented in Subramanian et al.) and returns a list of gene sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	parseGmt(file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parse+2B20GMT+2B20files_+3A_file">file</code></td>
<td>

<p>A file name.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Parses a GMT file and returns a list of gene sets. Each list element named according to the included gene set. The gene set files can be downloaded from http://www.broadinstitute.org/gsea/msigdb.
</p>


<h3>Value</h3>

<p>A named <code>list</code> of gene sets.
</p>


<h3>References</h3>

<p>Subramanian, A., Tamayo, P., Mootha, V. K., Mukherjee, S., Ebert, B. L., Gillette, M. A., Paulovich, A., Pomeroy, S. L., Golub, T. R., Lander, E. S., Mesirov, J. P. (2005) Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. <em>Proceedings of the National Academy of Science of the United States of America</em>, <b>102</b>, 15545&ndash;15550.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+predefinedAnalyses">predefinedAnalyses</a></code>, <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>
</p>

<hr>
<h2 id='pathways'>
Example data for pathways to to gene set enrichtment analyses
</h2><span id='topic+pathways'></span>

<h3>Description</h3>

<p>A list comprising exemplary pathways. Each pathway is represented by a vectro of random genes from countdata which are associated to this pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exampleData)</code></pre>


<h3>Details</h3>

<p>The data consists of a variable <code>pathways</code> of class <code>list</code> with 9 exemplary pathways. Each element of the list corresponds to one pathway - represented by a vector of gene names which are involved in this pathway. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exampleData)

print(pathways)

</code></pre>

<hr>
<h2 id='plot'>
Plots the results of an uncertainty analysis.
</h2><span id='topic+plot.uncertaintyResult'></span>

<h3>Description</h3>

<p>For each percentage of original gene set genes, the quantiles of the distribution obtained by a resampling simulation are plotted. Significance threshold (quantile of the Null distribution) and the test statistic of the original gene set are drawn as horizontal lines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'uncertaintyResult'
plot(x,
	signLevel = x$signLevel,
	addLegend = TRUE,
	addMinimalStability = FALSE,
	...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_+3A_x">x</code></td>
<td>

<p>A result of a call to <code>evaluateGeneSetUncertainty</code> (see also Details). 
</p>
</td></tr>
<tr><td><code id="plot_+3A_signlevel">signLevel</code></td>
<td>

<p>Only results with significance level smaller than the given value are plotted.
</p>
</td></tr>
<tr><td><code id="plot_+3A_addlegend">addLegend</code></td>
<td>

<p>If set to true (default), a <code><a href="graphics.html#topic+legend">legend</a></code> is added to the plot.
</p>
</td></tr>
<tr><td><code id="plot_+3A_addminimalstability">addMinimalStability</code></td>
<td>

<p>If set to true, a line is added to the plot giving the minimal stability.
</p>
</td></tr>
<tr><td><code id="plot_+3A_...">...</code></td>
<td>

<p>Other parameters which can be used for histograms (see <code><a href="graphics.html#topic+plot">plot</a></code>).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function plots the quantiles of the resampling distributions for evaluated degrees of fuzziness. It requires the 
significance assessment step of the enrichment analysis configuration (parameter <code>significance</code> or <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>) to be a computer-intensive testing procedure that yields a distribution of gene set statistic values under the null hypothesis. Predefined configurations for which this plot works are <code><a href="#topic+analysis.gsea">analysis.gsea</a></code>, <code><a href="#topic+analysis.averageCorrelation">analysis.averageCorrelation</a></code> and <code><a href="#topic+analysis.averageTStatistic">analysis.averageTStatistic</a></code>.
</p>
<p>Three lines, corresponding to the different qunatiles with one dot per fuzziness evaluation (<code>k</code>) are plotted for the analysis in <code>x</code>. The significance threshold is shown as a green horizontal line. The statistic value of the original input set is depicted as a red horizontal line.
</p>
<p>If <code>addMinimalStability</code> is <code>TRUE</code>, the lower bound of the stability is ploted as a dotted line.
</p>


<h3>Value</h3>

<p>No return value, called for creating a plot.</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+predefinedAnalyses">predefinedAnalyses</a></code>, <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>, <code><a href="#topic+evaluateGeneSetUncertainty">evaluateGeneSetUncertainty</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load data
data(exampleData)

  res &lt;- evaluateGeneSetUncertainty(
  	# parameters for evaluateGeneSetUncertainty
  	dat = countdata,
  	geneSet = pathways[[1]],
  	analysis = analysis.averageCorrelation(),
  	numSamplesUncertainty = 10,
  	N = seq(0.1,0.9, by=0.1),
  	# additional parameters for analysis.averageCorrelation
  	labs = labels,
  	numSamples = 10)
  
  # plot the results for the cell cycle control gene set
  plot(res, addMinimalStability = TRUE)
</code></pre>

<hr>
<h2 id='plotOverrepresentation'>
Plot overlap of gene sets and core set
</h2><span id='topic+plotOverrepresentation'></span>

<h3>Description</h3>

<p>Plots a Venn diagramm of the overlaps of the core set and gene sets in an overrepresentation analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotOverrepresentation(
	object,
	signLevel = object$signLevel,
	subset = NULL,
	aggregate = FALSE,
	ask = FALSE,
	...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotOverrepresentation_+3A_object">object</code></td>
<td>

<p>A result of a call to <code>geneSetAnalysis</code> using the predefined analysis <code><a href="#topic+predefinedAnalyses">analysis.customOverrepresentation</a></code> or <code><a href="#topic+predefinedAnalyses">analysis.overrepresentation</a></code>.
</p>
</td></tr>
<tr><td><code id="plotOverrepresentation_+3A_signlevel">signLevel</code></td>
<td>

<p>Only results with significance level smaller than the given value are included in the venn diagram.
</p>
</td></tr>
<tr><td><code id="plotOverrepresentation_+3A_subset">subset</code></td>
<td>

<p>Indices for the results that should be included in the diagram. 
</p>
</td></tr>
<tr><td><code id="plotOverrepresentation_+3A_aggregate">aggregate</code></td>
<td>

<p>Specifies whether all gene sets should be plotted in a single Venn diagram (which is possible for at most four gene sets) or whether there should be one Venn diagram for each gene set.
</p>
</td></tr>
<tr><td><code id="plotOverrepresentation_+3A_ask">ask</code></td>
<td>

<p>If set to true, the plot function will prompt for a user input for each new plot that is shown on an interactive device (see <code><a href="graphics.html#topic+par">par(&quot;ask&quot;)</a></code>). If <code>aggregate = TRUE</code>, <code>ask</code> is ignored.
</p>
</td></tr>
<tr><td><code id="plotOverrepresentation_+3A_...">...</code></td>
<td>

<p>Further parameters to be passed to <code><a href="limma.html#topic+vennDiagram">vennDiagram</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for creating a plot.</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+predefinedAnalyses">predefinedAnalyses</a></code>, <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(exampleData)
  # use the absolute correlation as a gene-level statistic
  stat &lt;- abs(apply(countdata,1,cor,y = labels))
  # define the core set as the 25% genes with the highest correlation
  coreSet &lt;- rownames(countdata)[tail(order(stat), 25)]
  
  # perform an overrepresentation analysis
  resOverrep &lt;- geneSetAnalysis(
  	dat = countdata,
  	geneSets = pathways,
  	analysis = analysis.customOverrepresentation(),
  	coreSet = coreSet,
  	adjustmentMethod = "fdr")
  
  # plot a Venn diagram
  plotOverrepresentation(resOverrep, subset = 1:3, aggregate = TRUE)
</code></pre>

<hr>
<h2 id='predefinedAnalyses'>
Predefined enrichment analyses
</h2><span id='topic+predefinedAnalyses'></span><span id='topic+analysis.gsea'></span><span id='topic+analysis.overrepresentation'></span><span id='topic+analysis.customOverrepresentation'></span><span id='topic+analysis.averageCorrelation'></span><span id='topic+analysis.averageTStatistic'></span><span id='topic+analysis.globalTest'></span><span id='topic+analysis.globalAncova'></span>

<h3>Description</h3>

<p>Predefined analysis configurations that can be used in <code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analysis.gsea()
analysis.overrepresentation()
analysis.customOverrepresentation()
analysis.averageCorrelation()
analysis.averageTStatistic()
analysis.globalTest()
analysis.globalAncova()
</code></pre>


<h3>Details</h3>

<p>The above functions return configurations for state-of-the-art analysis pipelines that can be used in <code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>. All configurations are preconfigured collections of standard methods for the different pipeline steps. The following lists the methods chosen for the different steps and their parameters. For more detailed descriptions of these methods, please refer to the linked manual pages.
</p>

<ul>
<li> <p><code>analysis.gsea</code> defines the Gene Set Enrichment Analysis (GSEA) method by Subramanain et al. 
Here, the gene-level statistic the absolute correlation calculated by <code><a href="#topic+gls.cor">gls.cor</a></code> with the associated parameters <code>labs, method</code> and a preprocessing by <code><a href="#topic+transformation.abs">transformation.abs</a></code>. As a gene set statistic, the enrichment score (function <code><a href="#topic+gss.enrichmentScore">gss.enrichmentScore</a></code> with parameter <code>p</code>) is calculated. The significance is assessed in a permutation test using <code><a href="#topic+significance.permutation">significance.permutation</a></code> with <code>testAlternative = "greater"</code> and free parameter <code>numSamples, labs</code>.
</p>
</li>
<li> <p><code>analysis.overrepresentation</code> calculates an overrepresentation analysis using the gene-level statistic <code><a href="#topic+gls.tStatistic">gls.tStatistic</a></code> with parameters <code>pValue</code>(should be <code>TRUE</code>), <code>alternative</code> and <code>labs</code>. The resulting values are then transformed via <code>transformation.adjustAndBinarize</code> (parameters are the <code>adjMethod</code> and <code>threshold</code>). Finally <code>gss.fisherExactTest</code> is used as gene set statistic.
</p>
</li>
<li> <p><code>analysis.customOverrepresentation</code> calculates an overrepresentation analysis using a user-defined core set <code>coreSet</code>. That is, instead of calculating this core set internally based on differential expression as the standard overrepresentation analysis, this function allows for defining custom core sets. It internally uses the global analysis <code><a href="#topic+global.overrepresentation">global.overrepresentation</a></code>.
</p>
</li>
<li> <p><code>analysis.averageCorrelation</code> calculates the gene-level statistic as the absolute correlation using <code><a href="#topic+gls.cor">gls.cor</a></code> (with parameters <code>labs, method</code>) and <code><a href="#topic+transformation.abs">transformation.abs</a></code>. The gene set statistic is the mean correlation calculated by <code><a href="#topic+gss.mean">gss.mean</a></code>. The significance is assessed by comparing the gene set statistic to randomly sampled gene sets using <code><a href="#topic+significance.sampling">significance.sampling</a></code> (with the parameter <code>numSamples</code> and the preset parameter <code>testAlternative = "greater"</code>).
</p>
</li>
<li> <p><code>analysis.averageTStatistic</code> uses the absolute t statistic as the gene-level statistic by applying <code><a href="#topic+gls.tStatistic">gls.tStatistic</a></code> (with parameters <code>labs, pValue, alternative</code>) and <code><a href="#topic+transformation.abs">transformation.abs</a></code>. The gene set statistic is the mean t statistic in the gene set as returned by <code><a href="#topic+gss.mean">gss.mean</a></code>. The significance is assessed by comparing the gene set statistic to randomly sampled gene sets using <code><a href="#topic+significance.sampling">significance.sampling</a></code> (with the parameter <code>numSamples</code> and the preset parameter <code>testAlternative = "greater"</code>).
</p>
</li>
<li> <p><code>analysis.globalTest</code> performs a global gene set enrichment analysis by Goeman et al. by applying the <code><a href="#topic+global.test">global.test</a></code> function which in turn wraps the <code><a href="globaltest.html#topic+gt">gt</a></code> function in the <span class="pkg">globaltest</span> package.
</p>
</li>
<li> <p><code>analysis.globalAncova</code> applies the global ANCOVA method by Hummel et al. using the global method <code><a href="#topic+global.ancova">global.ancova</a></code> which wraps the <code><a href="GlobalAncova.html#topic+GlobalAncova">GlobalAncova</a></code> function in the <span class="pkg">GlobalAncova</span> package.
</p>
</li></ul>



<h3>Value</h3>

<p>All functions return an object of class <code>gsAnalysis</code> that specifies the corresponding analysis parameters for <code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>.
</p>


<h3>References</h3>

<p>Subramanian, A., Tamayo, P., Mootha, V. K., Mukherjee, S., Ebert, B. L., Gillette, M. A., Paulovich, A., Pomeroy, S. L., Golub, T. R., Lander, E. S., Mesirov, J. P. (2005) Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. <em>Proceedings of the National Academy of Science of the United States of America</em>, <b>102</b>, 15545&ndash;15550.
</p>
<p>Hummel, M., Meister, R., Mansmann, U. (2008) GlobalANCOVA: exploration and assessment of gene group effects. <em>Bioinformatics</em>, <b>24</b>(1), 78&ndash;85.
</p>
<p>Goeman, J. J., van de Geer, S. A., de Kort, F., van Houwelingen, H. C. (2004) A global test for groups of genes: testing association with a clinical outcome. <em>Bioinformatics</em>, <b>20</b>(1), 93&ndash;99.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
	data(exampleData)
  # apply a gene set analysis based on the average absolute correlation
  resAvCor &lt;- geneSetAnalysis(
  	# parameters for geneSetAnalysis
  	dat = countdata,
  	geneSets = pathways[1],
  	analysis = analysis.averageCorrelation(),
  	adjustmentMethod = "fdr",
  	# additional parameters for analysis.averageCorrelation
    labs = labels,
  	method = "pearson",
  	numSamples = 10)
  
  # apply an overrepresentation analysis
  resOverrep &lt;- geneSetAnalysis(
  	# parameters for geneSetAnalysis
  	dat = countdata,
  	geneSets = pathways,
  	analysis = analysis.overrepresentation(),
  	adjustmentMethod = "fdr",
  	# additional parameters for analysis.overrepresentation
  	pValue = TRUE,
  	threshold = 0.1,
  	labs = labels
  )
  
  # apply a global analysis using GlobalAncova
  resGA &lt;- geneSetAnalysis(
  	# parameters for geneSetAnalysis
  	dat = countdata,
  	geneSets = pathways[1],
  	analysis = analysis.globalAncova(),
  	adjustmentMethod = "fdr",
  	# additional parameters for analysis.globalAncova
  	labs = labels,
  	method = "approx")

</code></pre>

<hr>
<h2 id='preprocessGeneSets'>
Eliminate unknown genes from gene sets
</h2><span id='topic+preprocessGeneSets'></span><span id='topic+preprocessGs'></span>

<h3>Description</h3>

<p>This function removes all genes that are not part of the experiment (not in <code>rownames(dat)</code>) from the specified gene sets which. All names are set to lower case.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preprocessGs(
	dat,
	geneSets)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="preprocessGeneSets_+3A_dat">dat</code></td>
<td>

<p>A numeric matrix of gene expression values for all analyzed genes. Here, each row corresponds to one gene, and each column corresponds to one sample. The rows must be named with the gene names used in the gene sets. Here, only the row names (i.e. the gene names) are used by the function.
</p>
</td></tr>
<tr><td><code id="preprocessGeneSets_+3A_genesets">geneSets</code></td>
<td>

<p>A list of gene sets to be processed, where each gene set is a vector of gene names corresponding to the row names of <code>dat</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of preprocessed gene sets, where each gene set only contains those genes that are also present in <code>dat</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#values are not important, only the row names are used
dat &lt;- matrix(0,100,10)
rownames(dat) &lt;- paste("gene",1:100,sep="")

geneSets &lt;- list(
		gs1 = paste("GENE",1:20,sep=""),# all genes in the analyzed data
		gs2 = paste("Gene",101:110,sep=""),#no gene in the analyzed data
		gs3 = paste("gene",90:110,sep="")#some genes in the analyzed data
	)

newGeneSets &lt;- preprocessGs(dat = dat, geneSets = geneSets)
</code></pre>

<hr>
<h2 id='SignificanceAssessment'>
Significance assessment
</h2><span id='topic+significance'></span><span id='topic+significance.sampling'></span><span id='topic+significance.permutation'></span><span id='topic+significance.restandardization'></span>

<h3>Description</h3>

<p>Functions to assess the significance of the gene-level statistics, as used in the <code>significance</code> parameter of <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>. These functions are based on applying the same analysis to randomly modified data sets or gene sets and comparing their statistic values to the original gene set statistic value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>significance.sampling(
	...,
	dat,
	geneSet,
	analysis,
	glsValues,
	numSamples = 1000)

significance.permutation(
	...,
	dat,
	geneSet,
	analysis,
	glsValues,
	numSamples = 1000,
	labs)

significance.restandardization(
	...,
	dat,
	geneSet,
	analysis,
	glsValues,
	numSamples = 1000,
	labs)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SignificanceAssessment_+3A_...">...</code></td>
<td>

<p>Additional parameters for the different steps of the analysis pipeline, depending on the concrete configuration supplied in <code>analysis</code>.
</p>
</td></tr>
<tr><td><code id="SignificanceAssessment_+3A_dat">dat</code></td>
<td>

<p>The original data set as a numeric matrix of gene expression values for all analyzed genes. Here, each row corresponds to one gene, and each column corresponds to one sample. The rows must be named with the gene names used in the gene sets.
</p>
</td></tr>
<tr><td><code id="SignificanceAssessment_+3A_geneset">geneSet</code></td>
<td>

<p>The original gene set in form of a vector of gene names corresponding to the row names of <code>dat</code>.
</p>
</td></tr>
<tr><td><code id="SignificanceAssessment_+3A_analysis">analysis</code></td>
<td>

<p>The analysis applied to the original gene set (that should also be applied to the modified gene sets). This is an object of type <code>gsAnalysis</code> as produced by the function <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>.
</p>
</td></tr>
<tr><td><code id="SignificanceAssessment_+3A_glsvalues">glsValues</code></td>
<td>

<p>A vector containing the (possibly transformed) gene-level statistic values for each gene in the original data set <code>dat</code>.
</p>
</td></tr>
<tr><td><code id="SignificanceAssessment_+3A_numsamples">numSamples</code></td>
<td>

<p>The number of random samples that should be taked to calculate the null distribution for the significance assessment. Default is <code>1000</code> for each test.
</p>
</td></tr>
<tr><td><code id="SignificanceAssessment_+3A_labs">labs</code></td>
<td>

<p>A vector of class labels for the samples in <code>dat</code> for <code>significance.permutation</code> and <code>significance.restandardization</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Standard methods for the significance assessment of a gene set statistic (to be used in an analysis pipeline defined by <code>gsAnalysis</code>):
</p>

<ul>
<li> <p><code>significance.sampling</code>: This function repeatedly draws random gene sets. Their gene set statistic values form the null distribution.
</p>
</li>
<li> <p><code>significance.permutation</code>: This function repeatedly permutes the labels of the data set. The gene set statistic values for the original gene set on the permuted data set form the null distribution.
</p>
</li>
<li> <p><code>significance.restandardization</code>: This function applies both a gene set sampling and a label permutation. The permutation statistic values are standardized by their mean and standard deviation and then restandardized based on the gene set sampling statistic values. These restandardized values form the null distribution (Efron and Tibshirani).
</p>
</li></ul>



<h3>Value</h3>

<p><code>significance.sampling</code> returns a list with the following elements:
</p>
<table role = "presentation">
<tr><td><code>gssValues</code></td>
<td>
<p>A vector of gene set statistic values, one entry per sample.</p>
</td></tr>
<tr><td><code>randomGeneSets</code></td>
<td>
<p>A matrix containing the gene sets which were sampled randomly from the set of all genes.</p>
</td></tr>
</table>
<p><code>significance.permutation</code> returns a list with the following elements:
</p>
<table role = "presentation">
<tr><td><code>gssValues</code></td>
<td>
<p>A vector of gene set statistics, one entry per sample.</p>
</td></tr>
<tr><td><code>permutations</code></td>
<td>
<p>A matrix, where each column contains the indices of one permutation.</p>
</td></tr>
</table>
<p><code>significance.restandardization</code> returns a list with the following elements:
</p>
<table role = "presentation">
<tr><td><code>gssValues</code></td>
<td>
<p>A vector of gene set statistics, one entry per sample.</p>
</td></tr>
<tr><td><code>samplingValues</code></td>
<td>
<p>A list of sub-lists, each containing one sampling result as defined above.</p>
</td></tr>
<tr><td><code>permutationValues</code></td>
<td>
<p>A list of sub-lists, each containing one permutation result as defined above.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Efron, B., Tibshirani, R. (2007) On testing the significance of sets of genes. <em>Annals of Applied Statistics</em>, <b>1</b>, 107-129.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>, <code><a href="#topic+hist.gsaResult">hist.gsaResult</a></code>
</p>

<hr>
<h2 id='summary.gsaResult'>
Summarize gene set analysis results
</h2><span id='topic+summary.gsaResult'></span>

<h3>Description</h3>

<p>Prints a summary of a gene set analysis result object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gsaResult'
summary(object,
		mode = c("summary", "table"),
		orderBy = c("adjustedPValues", "rawPValues", "geneSetName"),
		significantOnly = FALSE,
		signLevel = object$signLevel,
		...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.gsaResult_+3A_object">object</code></td>
<td>

<p>A result object as returned by <code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>.
</p>
</td></tr>
<tr><td><code id="summary.gsaResult_+3A_mode">mode</code></td>
<td>

<p>Specifies the type of information that is displayed: By default (<code>mode="summary"</code>), a brief summary of the number of significant and insignificant gene sets is printed. For <code>mode="table"</code>, <code><a href="#topic+createSummaryTable">createSummaryTable</a></code> is called, and a detailed table of adjusted and unadjusted p-values and the number of genes for each gene set is printed.
</p>
</td></tr>
<tr><td><code id="summary.gsaResult_+3A_orderby">orderBy</code></td>
<td>

<p>If <code>mode="table"</code>, this specifies which field should be used for the row ordering. By default, rows are ordered according to the adjusted p-values.
</p>
</td></tr>
<tr><td><code id="summary.gsaResult_+3A_significantonly">significantOnly</code></td>
<td>

<p>If <code>mode="table"</code>, this specifies whether all gene sets (<code>significantOnly=FALSE</code>) or only the statistically significant gene sets (<code>significantOnly=TRUE</code>) should be included in the table. 
</p>
</td></tr>
<tr><td><code id="summary.gsaResult_+3A_signlevel">signLevel</code></td>
<td>

<p>If <code>mode="table"</code> and <code>significantOnly=TRUE</code>, this specifies the significance level for the results that should be included in the table. By default, the original significance level of the analysis is used.
</p>
</td></tr>
<tr><td><code id="summary.gsaResult_+3A_...">...</code></td>
<td>

<p>Currently unused
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame with a summary of the produced results such as data set name, p-values, size of the given gene set.</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+hist.gsaResult">hist.gsaResult</a></code>, <code><a href="#topic+createSummaryTable">createSummaryTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exampleData)
  # perform gene set analyses for several pathways
  res &lt;- geneSetAnalysis(
  	# global parameters
  	dat = countdata,
  	geneSets = pathways, 
  	analysis = analysis.averageCorrelation(), 
  	# additional parameters for analysis.averageCorrelation
  	labs = labels,
  	numSamples = 10)
  
  #summarize the analyses
  summary(res, mode = "summary")
  
  summary(res, mode = "table", orderBy = "rawPValues")
</code></pre>

<hr>
<h2 id='Transformations'>
Transformations
</h2><span id='topic+transformation'></span><span id='topic+transformation.abs'></span><span id='topic+transformation.square'></span><span id='topic+transformation.localFdr'></span><span id='topic+transformation.binarize'></span><span id='topic+transformation.rank'></span><span id='topic+transformation.adjust'></span><span id='topic+transformation.adjustAndBinarize'></span>

<h3>Description</h3>

<p>Functions to transform the gene-level statistic values prior to the calculation of the gene set statistics, as used in the <code>transformation</code> parameter of <code>gsAnalysis</code>. Most of the functions wrap existing R functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transformation.abs(x)

transformation.square(x)

transformation.localFdr(x,
	statistic="pvalue",
	cutoff.method="fndr",
	pct0=0.75)

transformation.binarize(x, quant)

transformation.rank(x)

transformation.adjust(x, adjMethod = "fdr")

transformation.adjustAndBinarize(x, adjMethod = "fdr", threshold = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Transformations_+3A_x">x</code></td>
<td>

<p>A numeric vector of gene-level statistic values, one per gene. These values are calculated by the previous step (see <code><a href="#topic+gls">gls</a></code>).
</p>
</td></tr>
<tr><td><code id="Transformations_+3A_statistic">statistic</code></td>
<td>

<p>Specifies the null model for <code>transformation.localFdr</code> (see <code>statistic</code> parameter of <code><a href="fdrtool.html#topic+fdrtool">fdrtool</a></code> for possible values).
</p>
</td></tr>
<tr><td><code id="Transformations_+3A_cutoff.method">cutoff.method</code></td>
<td>

<p>Type of cut-off method used in <code>transformation.localFdr</code> (see <code>cutoff.method</code> parameter of <code><a href="fdrtool.html#topic+fdrtool">fdrtool</a></code> for possible values).
</p>
</td></tr>
<tr><td><code id="Transformations_+3A_pct0">pct0</code></td>
<td>

<p>Fraction of data used by <code>transformation.localFdr</code> if <code>cutoff.method="pct0"</code> (see <code><a href="fdrtool.html#topic+fdrtool">fdrtool</a></code> for a detailed description).
</p>
</td></tr>
<tr><td><code id="Transformations_+3A_quant">quant</code></td>
<td>

<p>For <code>transformation.binarize</code>, this numeric value in the interval [0,1] defines the percentage of gene-level statistic values which should be set to zero. The remaining values are set to one.
</p>
</td></tr>
<tr><td><code id="Transformations_+3A_adjmethod">adjMethod</code></td>
<td>

<p>The method to use for the adjustment for multiple testing (see <code>method</code> parameter of <code><a href="stats.html#topic+p.adjust">p.adjust</a></code> for possible values).
</p>
</td></tr>
<tr><td><code id="Transformations_+3A_threshold">threshold</code></td>
<td>

<p>The threshold for differential expression of a gene (defaults to <code>0.05</code>). Values smaller than these threshold are set to 1 , others to 0.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Standard transformation functions for gene-level statistics (to be used in an analysis pipeline defined by <code>gsAnalysis</code>):
</p>

<ul>
<li> <p><code>transformation.abs</code>: Calculates the absolute values of the elements in <code>x</code> (a wrapper for <code><a href="base.html#topic+abs">abs</a></code>).
</p>
</li>
<li> <p><code>transformation.square</code>: Squares all elements in <code>x</code>.
</p>
</li>
<li> <p><code>transformation.localFdr</code>: Calculates the local fdr for the elements in <code>x</code>. This is a wrapper for <code><a href="fdrtool.html#topic+fdrtool">fdrtool</a></code>.
</p>
</li>
<li> <p><code>transformation.binarize</code>: Binarizes the values in <code>x</code> by using the <code>quant</code> quantile as a threshold.
</p>
</li>
<li> <p><code>transformation.rank</code>: Ranks the values in <code>x</code> and returns the rank vector.
</p>
</li>
<li> <p><code>transformation.adjust</code>: Adjusts for multiple testing according to the adjustment method specified in <code>adjMethod</code>.
</p>
</li>
<li> <p><code>transformation.adjustAndBinarize</code>: Adjusts for multiple testing according to the adjustment method specified in <code>adjMethod</code> and binarizes the resulting p-values according to <code>threshold</code> (values smaller than the threshold become 1 others 0).</p>
</li></ul>



<h3>Value</h3>

<p>All functions return a vector of transformed values having the same length as <code>x</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geneSetAnalysis">geneSetAnalysis</a></code>, <code><a href="#topic+gsAnalysis">gsAnalysis</a></code>, <code><a href="#topic+gss">gss</a></code>, <code><a href="#topic+gls">gls</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
