<!DOCTYPE html><html lang="en"><head><title>Help for package gimap</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gimap}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#calc_gi'><p>Calculate Genetic Interaction scores</p></a></li>
<li><a href='#cn_setup'><p>Download and set up DepMap CN</p></a></li>
<li><a href='#ctrl_genes'><p>Download and set up control genes</p></a></li>
<li><a href='#delete_annotation'><p>Refresh the annotation files by redownloading them</p></a></li>
<li><a href='#delete_example_data'><p>Refresh the example data files by redownloading them</p></a></li>
<li><a href='#encrypt_creds_path'><p>Default creds path</p></a></li>
<li><a href='#example_data_folder'><p>Get file path to an default credentials RDS</p></a></li>
<li><a href='#get_example_data'><p>Returns example data for package</p></a></li>
<li><a href='#get_figshare'><p>Handler function for GET requests from Figshare</p></a></li>
<li><a href='#gimap_annotate'><p>Annotate gimap data</p></a></li>
<li><a href='#gimap_filter'><p>A function to run filtering</p></a></li>
<li><a href='#gimap_normalize'><p>Normalize Log fold changes</p></a></li>
<li><a href='#gimap_object'><p>Make an empty gimap dataset object</p></a></li>
<li><a href='#gimap_stats'><p>Do tests &ndash;an internal function used by calc_gi() function</p></a></li>
<li><a href='#key_encrypt_creds_path'><p>Get file path to an key encryption RDS</p></a></li>
<li><a href='#plot_crispr'><p>Plot CRISPR scores after normalization</p></a></li>
<li><a href='#plot_exp_v_obs_scatter'><p>Expected vs Observed CRISPR Scatterplot</p></a></li>
<li><a href='#plot_rank_scatter'><p>Rank plot for target-level GI scores</p></a></li>
<li><a href='#plot_targets_bar'><p>Target bar plot for CRISPR scores</p></a></li>
<li><a href='#plot_theme'><p>Standardized plot theme</p></a></li>
<li><a href='#plot_volcano'><p>Volcano plot for GI scores</p></a></li>
<li><a href='#qc_cdf'><p>Create a CDF for the pgRNA normalized counts</p></a></li>
<li><a href='#qc_constructs_countzero_bar'><p>Create a bar graph that shows the number of replicates with a zero count for pgRNA constructs flagged by the zero count filter</p></a></li>
<li><a href='#qc_cor_heatmap'><p>Create a correlation heatmap for the pgRNA CPMs</p></a></li>
<li><a href='#qc_filter_plasmid'><p>Create a filter for pgRNAs which have a low log2 CPM value for the</p>
plasmid/Day 0 sample/time point</a></li>
<li><a href='#qc_filter_zerocounts'><p>Filter out samples of zero counts</p>
Create a filter for pgRNAs which have a raw count of 0 for any sample/time
# point</a></li>
<li><a href='#qc_plasmid_histogram'><p>Create a histogram with plasmid log2 CPM values and ascertain a cutoff for low values</p></a></li>
<li><a href='#qc_sample_hist'><p>Create a histogram for the pgRNA log2 CPMs, faceted by sample</p></a></li>
<li><a href='#qc_variance_hist'><p>Create a histogram for the variance within replicates for each pgRNA</p></a></li>
<li><a href='#run_qc'><p>Run Quality Control Checks</p></a></li>
<li><a href='#setup_data'><p>Making a new gimap dataset</p></a></li>
<li><a href='#supported_cell_lines'><p>List the supported cell lines</p></a></li>
<li><a href='#tpm_setup'><p>Download and set up DepMap TPM data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Calculate Genetic Interactions for Paired CRISPR Targets</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Helps find meaningful patterns in complex genetic experiments. First gimap takes data from paired CRISPR (Clustered regularly interspaced short palindromic repeats) screens that has been pre-processed to counts table of paired gRNA (guide Ribonucleic Acid) reads. The input data will have cell counts for how well cells grow (or don't grow) when different genes or pairs of genes are disabled. The output of the 'gimap' package is genetic interaction scores which are  the distance between the observed CRISPR score and the expected CRISPR score. The expected CRISPR scores are what we expect for the CRISPR values to be for two unrelated genes. The further away an observed CRISPR score is from its expected score the more we suspect genetic interaction. The work in this package is based off of original research from the Alice Berger lab at Fred Hutchinson Cancer Center (2021) &lt;<a href="https://doi.org/10.1016%2Fj.celrep.2021.109597">doi:10.1016/j.celrep.2021.109597</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/FredHutch/gimap">https://github.com/FredHutch/gimap</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/FredHutch/gimap/issues">https://github.com/FredHutch/gimap/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>readr, dplyr, tidyr, rmarkdown, vroom, ggplot2, magrittr,
pheatmap, purrr, janitor, broom, utils, stats, stringr, httr,
jsonlite, openssl</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), roxygen2, kableExtra, knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-07 18:54:22 UTC; candacesavonen</td>
</tr>
<tr>
<td>Author:</td>
<td>Candace Savonen [aut, cre],
  Phoebe Parrish [aut],
  Kate Isaac [aut],
  Howard Baek [aut],
  Daniel Grosso [aut],
  Siobhan O'Brien [aut],
  Alice Berger [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Candace Savonen &lt;cansav09@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-07 20:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling 'rhs(lhs)'.
</p>

<hr>
<h2 id='calc_gi'>Calculate Genetic Interaction scores</h2><span id='topic+calc_gi'></span>

<h3>Description</h3>

<p>Create results table that has CRISPR scores, Wilcoxon rank-sum
test and t tests.
The output of the 'gimap' package is genetic interaction scores which _is the
distance between the observed CRISPR score and the expected CRISPR score._
The expected CRISPR scores are what we expect for the CRISPR values should
two genes be unrelated to each other. The further away an observed CRISPR
scoreis from its expected the more we suspect genetic interaction.
This can be true in a positive way (a CRISPR knockout pair caused more cell
proliferation than expected) or in a negative way (a CRISPR knockout pair
caused more cell lethality than expected).
</p>
<p>The genetic interaction scores are based on a linear model calculated for
each sample where 'observed_crispr_single' is the outcome variable and
'expected_crispr_single' is the predictor variable.
For each sample: lm(observed_crispr_single ~ expected_crispr_single)
</p>
<p>Using 'y = mx+b', we can fill in the following values:
* 'y' = observed CRISPR score
* 'x' = expected CRISPR score
* 'm' = slope from linear model for this sample
* 'b' = intercept from linear model for this sample
</p>
<p>The intercept and slope from this linear model are used to adjust the CRISPR
scores for each sample:
single target gi score =
observed single crispr - (intercept + slope * expected single crispr)
double_target_gi_score =
double crispr score - (intercept + slope * expected double crispr)
These single and double target genetic interaction scores are calculated at
the construct level and are then summarized using a t-test to see if the the
distribution of the set of double targeting constructs is significantly
different than the overall distribution single targeting constructs.
After multiple testing correction, FDR values are reported.
Low FDR value for a double construct means high suspicion of paralogs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_gi(.data = NULL, gimap_dataset, use_lfc = FALSE, stats_by_rep = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_gi_+3A_.data">.data</code></td>
<td>
<p>Data can be piped in with tidyverse pipes from function to
function. But the data must still be a gimap_dataset</p>
</td></tr>
<tr><td><code id="calc_gi_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>A special dataset structure that is setup using the
'setup_data()' function.</p>
</td></tr>
<tr><td><code id="calc_gi_+3A_use_lfc">use_lfc</code></td>
<td>
<p>Should Log fold change be used to calculate GI scores instead
of CRISPR scores? If you do not have negative controls or CRISPR scores you
will need to set this to TRUE.</p>
</td></tr>
<tr><td><code id="calc_gi_+3A_stats_by_rep">stats_by_rep</code></td>
<td>
<p>Should statistics be calculated per rep or should replicates
be collapsed?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A gimap dataset with statistics and genetic interaction scores
calculated. Overall results in the returned object can be obtained using
gimap_dataset$overall_results Whereas target level genetic interaction
scores can be retrieved using 'gimap_dataset$gi_scores'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

gimap_dataset &lt;- get_example_data("gimap",
                                  data_dir = tempdir()) %&gt;%
  gimap_filter() %&gt;%
  gimap_annotate(cell_line = "HELA",
                 annot_dir = tempdir()) %&gt;%
  gimap_normalize(
    timepoints = "day",
    missing_ids_file = tempfile()
  ) %&gt;%
  calc_gi()

saveRDS(gimap_dataset, file.path(tempdir(), "gimap_dataset_final.RDS"))

</code></pre>

<hr>
<h2 id='cn_setup'>Download and set up DepMap CN</h2><span id='topic+cn_setup'></span>

<h3>Description</h3>

<p>This function sets up the tpm data from DepMap is called
by the 'gimap_annotate()' function if the cn_annotate = TRUE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cn_setup(
  overwrite = TRUE,
  data_dir = system.file("extdata", package = "gimap")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cn_setup_+3A_overwrite">overwrite</code></td>
<td>
<p>Should the files be redownloaded?</p>
</td></tr>
<tr><td><code id="cn_setup_+3A_data_dir">data_dir</code></td>
<td>
<p>What directory should this be saved to? Default is with package
files</p>
</td></tr>
</table>

<hr>
<h2 id='ctrl_genes'>Download and set up control genes</h2><span id='topic+ctrl_genes'></span>

<h3>Description</h3>

<p>This function sets up the control genes file from DepMap is
called by the 'gimap_annotate()'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctrl_genes(
  overwrite = TRUE,
  data_dir = system.file("extdata", package = "gimap")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ctrl_genes_+3A_overwrite">overwrite</code></td>
<td>
<p>Should the file be redownloaded and reset up?</p>
</td></tr>
<tr><td><code id="ctrl_genes_+3A_data_dir">data_dir</code></td>
<td>
<p>What directory should this be saved to? Default is with package
files</p>
</td></tr>
</table>

<hr>
<h2 id='delete_annotation'>Refresh the annotation files by redownloading them</h2><span id='topic+delete_annotation'></span>

<h3>Description</h3>

<p>This function will set annotation file options to NULL so files
will be re-downloaded
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delete_annotation()
</code></pre>


<h3>Value</h3>

<p>options for tpm_file, cn_file, and ctrl_genes_file are set to NULL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
delete_annotation()

</code></pre>

<hr>
<h2 id='delete_example_data'>Refresh the example data files by redownloading them</h2><span id='topic+delete_example_data'></span>

<h3>Description</h3>

<p>This function will set example data file options to NULL so files
will be re-downloaded
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delete_example_data()
</code></pre>


<h3>Value</h3>

<p>options for example data are are set to NULL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
delete_example_data()

</code></pre>

<hr>
<h2 id='encrypt_creds_path'>Default creds path</h2><span id='topic+encrypt_creds_path'></span>

<h3>Description</h3>

<p>Default creds path
</p>


<h3>Usage</h3>

<pre><code class='language-R'>encrypt_creds_path(app_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="encrypt_creds_path_+3A_app_name">app_name</code></td>
<td>
<p>What app set up are you looking for? Supported apps are 'google' 'calendly' and 'github'</p>
</td></tr>
</table>

<hr>
<h2 id='example_data_folder'>Get file path to an default credentials RDS</h2><span id='topic+example_data_folder'></span>

<h3>Description</h3>

<p>Get file path to an default credentials RDS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_data_folder()
</code></pre>


<h3>Value</h3>

<p>Returns the file path to folder where the example data is stored
</p>

<hr>
<h2 id='get_example_data'>Returns example data for package</h2><span id='topic+get_example_data'></span>

<h3>Description</h3>

<p>This function loads and returns example data for the package.
Which dataset is returned must be specified. Data will be downloaded from Figshare
the first time it is used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_example_data(
  which_data,
  data_dir = system.file("extdata", package = "gimap"),
  refresh_data = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_example_data_+3A_which_data">which_data</code></td>
<td>
<p>options are &quot;count&quot; or &quot;meta&quot;; specifies which example dataset should be returned</p>
</td></tr>
<tr><td><code id="get_example_data_+3A_data_dir">data_dir</code></td>
<td>
<p>Where should the data be saved if applicable?</p>
</td></tr>
<tr><td><code id="get_example_data_+3A_refresh_data">refresh_data</code></td>
<td>
<p>should the example data that's been downloaded be deleted
and redownloaded?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the respective example data either as a data frame or a specialized
gimap_dataset depending on what was requested.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

counts_timepoint &lt;- get_example_data("count")
counts_treatment &lt;- get_example_data("count_treatment")
gimap_timepoint_dataset &lt;- get_example_data("gimap")
gimap_treatment_dataset &lt;- get_example_data("gimap_treatment")
metadata &lt;- get_example_data("meta")
annotation &lt;- get_example_data("annotation")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_figshare'>Handler function for GET requests from Figshare</h2><span id='topic+get_figshare'></span>

<h3>Description</h3>

<p>Handler function for GET requests from Figshare
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_figshare(
  file_name = NA,
  item = "19700056",
  output_dir = tempdir(),
  return_list = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_figshare_+3A_file_name">file_name</code></td>
<td>
<p>Which item are we downloading?</p>
</td></tr>
<tr><td><code id="get_figshare_+3A_item">item</code></td>
<td>
<p>What is the item we are retrieving?</p>
</td></tr>
<tr><td><code id="get_figshare_+3A_output_dir">output_dir</code></td>
<td>
<p>Where should the file be saved?</p>
</td></tr>
<tr><td><code id="get_figshare_+3A_return_list">return_list</code></td>
<td>
<p>Should the list of files be returned instead of the file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Downloads necessary annotation files from Figshare and reads them
in as data frames.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

get_figshare(
  return_list = TRUE,
  output_dir = tempdir()
)

get_figshare(
  file_name = "Achilles_common_essentials.csv",
  output_dir = tempdir()
)

</code></pre>

<hr>
<h2 id='gimap_annotate'>Annotate gimap data</h2><span id='topic+gimap_annotate'></span>

<h3>Description</h3>

<p>In this function, a 'gimap_dataset' is annotated as far as which
genes should be used as controls.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gimap_annotate(
  .data = NULL,
  gimap_dataset,
  annotation_file = NULL,
  control_genes = NULL,
  cell_line_annotate = TRUE,
  custom_tpm = NULL,
  cell_line = NULL,
  annot_dir = system.file("extdata", package = "gimap"),
  refresh_annot = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gimap_annotate_+3A_.data">.data</code></td>
<td>
<p>Data can be piped in with tidyverse pipes from function to
function. But the data must still be a gimap_dataset</p>
</td></tr>
<tr><td><code id="gimap_annotate_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>A special dataset structure that is setup using the
'setup_data()' function.</p>
</td></tr>
<tr><td><code id="gimap_annotate_+3A_annotation_file">annotation_file</code></td>
<td>
<p>If no file is given, will attempt to use the design
file from
https://media.addgene.org/cms/filer_public/a9/9a/
a99a9328-324b-42ff-8ccc-30c544b899e4/pgrna_library.xlsx</p>
</td></tr>
<tr><td><code id="gimap_annotate_+3A_control_genes">control_genes</code></td>
<td>
<p>A vector of gene symbols (e.g. AAMP) that should be
labeled as control genes. These will be used for log fold change
calculations. If no list is given then DepMap Public 23Q4
Achilles_common_essentials.csv is used
https://depmap.org/portal/download/all/</p>
</td></tr>
<tr><td><code id="gimap_annotate_+3A_cell_line_annotate">cell_line_annotate</code></td>
<td>
<p>(Optional) TRUE or FALSE you'd also like to have
cell_line_annotation from DepMap.</p>
</td></tr>
<tr><td><code id="gimap_annotate_+3A_custom_tpm">custom_tpm</code></td>
<td>
<p>(Optional) You may supply your own data frame of transcript
per million expression to be used for this calculation if you can't or don't
want to use DepMap data annotation for your cell_line. This data frame needs
to have two columns: 'log2_tpm' that has the log2 tpm expression data for
this cell line and and 'genes' which needs to be gene symbols that match
those in the data. eg. &quot;NDL1&quot;.
Note that you can use custom_tpm with cell_line_annotate but your custom_tpm
will be used instead of the tpm data from DepMap. However other data from
DepMap like CN will be added.</p>
</td></tr>
<tr><td><code id="gimap_annotate_+3A_cell_line">cell_line</code></td>
<td>
<p>which cell line are you using? (e.g., HELA, PC9, etc.).
Required argument if cell_line_annotate is TRUE.</p>
</td></tr>
<tr><td><code id="gimap_annotate_+3A_annot_dir">annot_dir</code></td>
<td>
<p>Where should the annotation files be saved? default is a
temporary directory.</p>
</td></tr>
<tr><td><code id="gimap_annotate_+3A_refresh_annot">refresh_annot</code></td>
<td>
<p>Should the annotation file paths saved as options be reset
and the files redownloaded? TRUE or FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A gimap_dataset with annotation data frame that can be retrieve by using
gimap_dataset$annotation. This will contain information about your included
genes in the set.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# By default DepMap annotation will be used to determine genes which are
# unexpressed. In the `gimap_normalize` this will by default be used to
# normalize to.
gimap_dataset &lt;- get_example_data("gimap",
                                  data_dir = tempdir()) %&gt;%
  gimap_filter() %&gt;%
  gimap_annotate(
    cell_line = "HELA",
    missing_ids_file =  tempfile(),
    annot_dir = tempdir())
  )


# You can also say cell_line_annotate = false if you don't want to use DepMap
# annotation BUT if you don't also specify that you say you are
# `normalize_by_unexpressed = FALSE` in the normalize step you will get a
# warning.
gimap_dataset &lt;- get_example_data("gimap",
                                  data_dir = tempdir()) %&gt;%
  gimap_filter() %&gt;%
  gimap_annotate(
    cell_line_annotate = FALSE,
    annot_dir = tempdir()) %&gt;%
  gimap_normalize(
    timepoints = "day",
    normalize_by_unexpressed = FALSE,
    missing_ids_file = tempfile()
  )

### CUSTOM TPM example
# Lastly, this is also an option:
# where custom data is provided to `custom_tpm` is a data frame with
# `genes` and `log2_tpm` as the columns.
gimap_dataset &lt;- get_example_data("gimap",
                                  data_dir = tempdir()) %&gt;%
  gimap_filter() %&gt;%
  gimap_annotate(
    cell_line = "HELA",
    custom_tpm = custom_tpm,
    annot_dir = tempdir()
  ) %&gt;%
  gimap_normalize(
    timepoints = "day",
    missing_ids_file = tempfile()
  )

## End(Not run)
</code></pre>

<hr>
<h2 id='gimap_filter'>A function to run filtering</h2><span id='topic+gimap_filter'></span>

<h3>Description</h3>

<p>This function applies filters to the gimap data. By default it
runs both the zero count (across all samples) and the low plasmid cpm
filters, but users can select a subset of these filters or even adjust the
behavior of each filter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gimap_filter(
  .data = NULL,
  gimap_dataset,
  filter_type = "both",
  cutoff = NULL,
  filter_zerocount_target_col = NULL,
  filter_plasmid_target_col = NULL,
  filter_replicates_target_col = NULL,
  min_n_filters = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gimap_filter_+3A_.data">.data</code></td>
<td>
<p>Data can be piped in with tidyverse pipes from function to
function. But the data must still be a gimap_dataset</p>
</td></tr>
<tr><td><code id="gimap_filter_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>A special dataset structure that is setup using the
'setup_data()' function.</p>
</td></tr>
<tr><td><code id="gimap_filter_+3A_filter_type">filter_type</code></td>
<td>
<p>Can be one of the following: 'zero_count_only',
'low_plasmid_cpm_only' or 'both'.
Potentially in the future also 'rep_variation', 'zero_in_last_time_point'
or a vector that includes multiple of these filters.</p>
</td></tr>
<tr><td><code id="gimap_filter_+3A_cutoff">cutoff</code></td>
<td>
<p>default is NULL, relates to the low_plasmid_cpm filter; the
cutoff for low log2 CPM values for the plasmid time period; if not specified,
The lower outlier (defined by taking the difference of the lower quartile and
1.5 * interquartile range) is used</p>
</td></tr>
<tr><td><code id="gimap_filter_+3A_filter_zerocount_target_col">filter_zerocount_target_col</code></td>
<td>
<p>default is NULL; Which sample column(s)
should be used to check for counts of 0? If NULL and not specified,
downstream analysis will select all sample columns</p>
</td></tr>
<tr><td><code id="gimap_filter_+3A_filter_plasmid_target_col">filter_plasmid_target_col</code></td>
<td>
<p>default is NULL, and if NULL, will select
the first column only; this parameter specifically should be used to specify
the plasmid column(s) that will be selected</p>
</td></tr>
<tr><td><code id="gimap_filter_+3A_filter_replicates_target_col">filter_replicates_target_col</code></td>
<td>
<p>default is NULL, Which sample columns
are the final time point replicates; If NULL, the last 3 sample columns are
used. This is only used by this function to save a list of which pgRNA IDs
have a zero count for all of these samples.</p>
</td></tr>
<tr><td><code id="gimap_filter_+3A_min_n_filters">min_n_filters</code></td>
<td>
<p>default is 1; this parameter defines at least
how many/the minimum number of independent filters have to flag a pgRNA
construct before the construct is filtered when using a combination of
filters You should decide on the appropriate filter based on the results of
your QC report.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a filtered version of the gimap_dataset returned in the
$filtered_data section filter_step_run is a boolean reporting if the filter
step was run or not (since it's optional)
metadata_pg_ids is a subset the pgRNA IDs such that these are the ones that
remain in the dataset following completion of filtering
transformed_log2_cpm is a subset the log2_cpm data such that these are the
ones that remain in the dataset following completion of filtering
removed_pg_ids is a record of which pgRNAs are filtered out once filtering
is complete
all_reps_zerocount_ids is not actually filtered data necessarily.
Instead it's just a record of which pgRNAs have a zero count in all final
timepoint replicates
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

gimap_dataset &lt;- get_example_data("gimap", data_dir = tempdir()) %&gt;%
  gimap_filter()

# To see filtered data
# gimap_dataset$filtered_data

# If you want to only use a single filter or some subset,
# specify which using the filter_type parameter
gimap_dataset &lt;- get_example_data("gimap") %&gt;%
  gimap_filter(filter_type = "zero_count_only")
# or
gimap_dataset &lt;- get_example_data("gimap") %&gt;%
  gimap_filter(filter_type = "low_plasmid_cpm_only")

# If you want to use multiple filters and more than one to flag a pgRNA
# construct before it's filtered out, use the `min_n_filters` argument
gimap_dataset &lt;- get_example_data("gimap") %&gt;%
gimap_filter(
  filter_type = "both",
  min_n_filters = 2
)

# You can also specify which columns the filters will be applied to
gimap_dataset &lt;- get_example_data("gimap") %&gt;%
 gimap_filter(
  filter_type = "zero_count_only",
  filter_zerocount_target_col = c(1, 2)
)

</code></pre>

<hr>
<h2 id='gimap_normalize'>Normalize Log fold changes</h2><span id='topic+gimap_normalize'></span>

<h3>Description</h3>

<p>This calculates the log fold change for a gimap dataset based on
the annotation and metadata provided.
gimap takes in a counts matrix that represents the number of cells that have
each type of pgRNA this data needs some normalization before CRISPR scores
and Genetic Interaction scores can be calculated.
</p>
<p>There are three steps of normalization.
1. 'Calculate log2CPM' - First we account for different read depths across
samples and transforms data to log2 counts per million reads.
'log2((counts / total counts for sample)) * 1 million) + 1)'
2. 'Calculate log2 fold change' - This is done by subtracting the log2CPM for
the pre-treatment from each sample.  control is what is highlighted.
The pretreatment is the day 0 of CRISPR treatment, before CRISPR pgRNAs
have taken effect.
'log2FC = log2CPM for each sample - pretreament log2CPM'
</p>
<p>3. 'Normalize by negative and positive controls' - Calculate a negative
control median for each sample and a positive control median for each sample
and divide each log2FC by this value.
log2FC adjusted = log2FC /
(median negative control for a sample - median positive control for a sample)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gimap_normalize(
  .data = NULL,
  gimap_dataset,
  normalize_by_unexpressed = TRUE,
  timepoints = NULL,
  treatments = NULL,
  control_name = NULL,
  adj_method = "negative_control_adj",
  num_ids_wo_annot = 20,
  rm_ids_wo_annot = TRUE,
  missing_ids_file = "missing_ids_file.csv",
  overwrite = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gimap_normalize_+3A_.data">.data</code></td>
<td>
<p>Data can be piped in with a tidyverse pipe from function to
function. But the data must still be a gimap_dataset</p>
</td></tr>
<tr><td><code id="gimap_normalize_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>A special dataset structure that is setup using the
'setup_data()' function.</p>
</td></tr>
<tr><td><code id="gimap_normalize_+3A_normalize_by_unexpressed">normalize_by_unexpressed</code></td>
<td>
<p>TRUE/FALSE crispr data should be normalized
so that the median of unexpressed controls is 0. For this to happen set this
to TRUE but you need to have added TPM data in the gimap_annotate step using
cell_line_annotation or custom_tpm.</p>
</td></tr>
<tr><td><code id="gimap_normalize_+3A_timepoints">timepoints</code></td>
<td>
<p>Specifies the column name of the metadata set up in
'$metadata$sample_metadata'
that has a factor that represents the timepoints.
Timepoints will be made into three categories:
plasmid for the earliest time point, early for all middle timepoints and
late for the latest timepoints.
The late timepoints will be the focus for the calculations. The column used
for timepoints must be numeric or at least ordinal.</p>
</td></tr>
<tr><td><code id="gimap_normalize_+3A_treatments">treatments</code></td>
<td>
<p>Specifies the column name of the metadata set up in
'$metadata$sample_metadata'
that has a factor that represents column that specifies the treatment applied
to each. The replicates will be kept collapsed to an average.</p>
</td></tr>
<tr><td><code id="gimap_normalize_+3A_control_name">control_name</code></td>
<td>
<p>A name that specifies the data either in the treatments
column that should be used as the control. This could be the Day 0 of
treatment or an untreated sample.
For timepoints testing it will be assumed that the mininmum timepoint
is the control.</p>
</td></tr>
<tr><td><code id="gimap_normalize_+3A_adj_method">adj_method</code></td>
<td>
<p>Must be one of three methods as stated by a character string
&quot;negative_control_adj&quot; or &quot;no_adjustment&quot;. Default is &quot;negative_control_adj&quot;
&quot;negative_control_adj&quot; where CRISPR scores will be used for the GI scores
&quot;no_adjustment&quot; is where LFC adjusted will be used for the GI scores</p>
</td></tr>
<tr><td><code id="gimap_normalize_+3A_num_ids_wo_annot">num_ids_wo_annot</code></td>
<td>
<p>default is 20; the number of pgRNA IDs to display to
console if they don't have corresponding annotation data;
ff there are more IDs without annotation data than this number, the output
will be sent to a file rather than the console.</p>
</td></tr>
<tr><td><code id="gimap_normalize_+3A_rm_ids_wo_annot">rm_ids_wo_annot</code></td>
<td>
<p>default is TRUE; whether or not to filter out pgRNA
IDs from the input dataset that don't have corresponding annotation data
available</p>
</td></tr>
<tr><td><code id="gimap_normalize_+3A_missing_ids_file">missing_ids_file</code></td>
<td>
<p>If there are missing IDs and a file is saved, where
do you want this file to be saved? Provide a file path.</p>
</td></tr>
<tr><td><code id="gimap_normalize_+3A_overwrite">overwrite</code></td>
<td>
<p>Should existing normalized_log_fc data in the gimap_dataset
be overwritten?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A gimap_dataset with normalized log FC as a data frame that can be
retrieve by using gimap_dataset$normalized_log_fc. This will contain
log2FC adjusted stored in a column named 'log_adj' and the CRISPR scores
stored in a column named 'crispr_score'.
genes in the set.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

gimap_dataset &lt;- get_example_data("gimap",
                                  data_dir = tempdir()) %&gt;%
  gimap_filter() %&gt;%
  gimap_annotate(cell_line = "HELA",
                 annot_dir = tempdir()) %&gt;%
  gimap_normalize(
    timepoints = "day",
    missing_ids_file = tempfile()
  )

</code></pre>

<hr>
<h2 id='gimap_object'>Make an empty gimap dataset object</h2><span id='topic+gimap_object'></span>

<h3>Description</h3>

<p>This function makes an empty gimap data object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gimap_object()
</code></pre>


<h3>Value</h3>

<p>an empty 'gimap_dataset' which is a named list which will
be filled by various 'gimap' functions.
</p>

<hr>
<h2 id='gimap_stats'>Do tests &ndash;an internal function used by calc_gi() function</h2><span id='topic+gimap_stats'></span>

<h3>Description</h3>

<p>Create results table that has t test p values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gimap_stats(gi_calc_double, gi_calc_single, replicate = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gimap_stats_+3A_gi_calc_double">gi_calc_double</code></td>
<td>
<p>a data.frame with adjusted double gi scores</p>
</td></tr>
<tr><td><code id="gimap_stats_+3A_gi_calc_single">gi_calc_single</code></td>
<td>
<p>a data.frame with adjusted single gi scores</p>
</td></tr>
<tr><td><code id="gimap_stats_+3A_replicate">replicate</code></td>
<td>
<p>a name of a replicate to filter out from gi_calc_adj Optional</p>
</td></tr>
</table>

<hr>
<h2 id='key_encrypt_creds_path'>Get file path to an key encryption RDS</h2><span id='topic+key_encrypt_creds_path'></span>

<h3>Description</h3>

<p>Get file path to an key encryption RDS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>key_encrypt_creds_path()
</code></pre>

<hr>
<h2 id='plot_crispr'>Plot CRISPR scores after normalization</h2><span id='topic+plot_crispr'></span>

<h3>Description</h3>

<p>This plots normalization after CRISPR scores have been calculated
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_crispr(.data = NULL, gimap_dataset, output_file = "crispr_norm_plot.png")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_crispr_+3A_.data">.data</code></td>
<td>
<p>Data can be piped in with tidyverse pipes from function to
function. But the data must still be a gimap_dataset</p>
</td></tr>
<tr><td><code id="plot_crispr_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>A special dataset structure that is setup using the
'setup_data()' function.</p>
</td></tr>
<tr><td><code id="plot_crispr_+3A_output_file">output_file</code></td>
<td>
<p>A file for the output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 boxplot of the CRISPR scores separated by the type of
target. Can be used to determine the normalization has proceeded properly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

gimap_dataset &lt;- get_example_data("gimap") %&gt;%
  gimap_filter() %&gt;%
  gimap_annotate(cell_line = "HELA") %&gt;%
  gimap_normalize(
    timepoints = "day",
    missing_ids_file = tempfile()
  )

# Plot:
plot_crispr(gimap_dataset)

</code></pre>

<hr>
<h2 id='plot_exp_v_obs_scatter'>Expected vs Observed CRISPR Scatterplot</h2><span id='topic+plot_exp_v_obs_scatter'></span>

<h3>Description</h3>

<p>This plot is meant to be functionally equivalent to Fig S5K (for HeLa, equivalent of Fig 3a for PC9).
Scatter plot of target-level observed versus expected CRISPR scores in the screen.
The solid line is the linear regression line for the negative control (single KO) pgRNAs,
while dashed lines indicate the lower and upper quartile residuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_exp_v_obs_scatter(gimap_dataset, facet_rep = FALSE, reps_to_drop = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_exp_v_obs_scatter_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>A special dataset structure that is originally setup using 'setup_data()' and has had gi scores calculated with 'calc_gi()'.</p>
</td></tr>
<tr><td><code id="plot_exp_v_obs_scatter_+3A_facet_rep">facet_rep</code></td>
<td>
<p>Should the replicates be wrapped with facet_wrap()?</p>
</td></tr>
<tr><td><code id="plot_exp_v_obs_scatter_+3A_reps_to_drop">reps_to_drop</code></td>
<td>
<p>Names of replicates that should be not plotted (Optional)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 scatterplot of the target level observed vs expected
CRISPR scores.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

gimap_dataset &lt;- get_example_data("gimap") %&gt;%
  gimap_filter() %&gt;%
  gimap_annotate(cell_line = "HELA") %&gt;%
  gimap_normalize(
    timepoints = "day",
    missing_ids_file = tempfile()
  ) %&gt;%
  calc_gi()

# To plot results
plot_exp_v_obs_scatter(gimap_dataset)
plot_rank_scatter(gimap_dataset)
plot_volcano(gimap_dataset)

</code></pre>

<hr>
<h2 id='plot_rank_scatter'>Rank plot for target-level GI scores</h2><span id='topic+plot_rank_scatter'></span>

<h3>Description</h3>

<p>This plot is meant to be functionally equivalent to Fig 5a (for HeLa, equivalent of Fig 3c for PC9).
Rank plot of target-level GI scores.
Dashed horizontal lines are for GI scores of 0.25 and -0.5
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_rank_scatter(gimap_dataset, reps_to_drop = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_rank_scatter_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>A special dataset structure that is originally setup using 'setup_data()' and has had gi scores calculated with 'calc_gi()'.</p>
</td></tr>
<tr><td><code id="plot_rank_scatter_+3A_reps_to_drop">reps_to_drop</code></td>
<td>
<p>Names of replicates that should be not plotted (Optional)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 rankplot of the target level genetic interaction scores.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

gimap_dataset &lt;- get_example_data("gimap") %&gt;%
  gimap_filter() %&gt;%
  gimap_annotate(cell_line = "HELA") %&gt;%
  gimap_normalize(
    timepoints = "day"
  ) %&gt;%
  calc_gi()

# To plot results
plot_exp_v_obs_scatter(gimap_dataset)
plot_rank_scatter(gimap_dataset)
plot_volcano(gimap_dataset)

</code></pre>

<hr>
<h2 id='plot_targets_bar'>Target bar plot for CRISPR scores</h2><span id='topic+plot_targets_bar'></span>

<h3>Description</h3>

<p>This plot is for when you'd like to examine a target pair
specifically &ndash; meant to be functionally equivalent to Fig 3b
CRISPR scores for representative synthetic lethal paralog pairs.
Data shown are the mean CRISPR score for each single KO or DKO target across
three biological replicates with replicate data shown in overlaid points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_targets_bar(gimap_dataset, target1, target2, reps_to_drop = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_targets_bar_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>A special dataset structure that is originally setup using
'setup_data()' and has had gi scores calculated with 'calc_gi()'.</p>
</td></tr>
<tr><td><code id="plot_targets_bar_+3A_target1">target1</code></td>
<td>
<p>Name of the first target to be plotted e.g.</p>
</td></tr>
<tr><td><code id="plot_targets_bar_+3A_target2">target2</code></td>
<td>
<p>Name of the second target to be plotted e.g.</p>
</td></tr>
<tr><td><code id="plot_targets_bar_+3A_reps_to_drop">reps_to_drop</code></td>
<td>
<p>Names of replicates that should be not plotted (Optional)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 bar plot of the specific target's genetic interaction scores.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

gimap_dataset &lt;- get_example_data("gimap") %&gt;%
  gimap_filter() %&gt;%
  gimap_annotate(cell_line = "HELA") %&gt;%
  gimap_normalize(
    timepoints = "day"
  ) %&gt;%
  calc_gi()

# To plot results, pick out two targets from the gi_score table
head(dplyr::arrange(gimap_dataset$gi_score, fdr))

# "TIAL1_TIA1" is top result so let's plot that
plot_targets_bar(gimap_dataset, target1 = "TIAL1", target2 = "TIA1")

</code></pre>

<hr>
<h2 id='plot_theme'>Standardized plot theme</h2><span id='topic+plot_theme'></span>

<h3>Description</h3>

<p>this is a ggplot2 standardized plot theme for this package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_theme()
</code></pre>


<h3>Value</h3>

<p>A ggplot2 theme that can be used on the plots.
</p>

<hr>
<h2 id='plot_volcano'>Volcano plot for GI scores</h2><span id='topic+plot_volcano'></span>

<h3>Description</h3>

<p>This plot is meant to be functionally equivalent to Fig 5b (for HeLa, equivalent of Fig 3d for PC9).
Volcano plot of target-level GI scores
Blue points are synthetic lethal paralog GIs with GI &lt; 0.5 and FDR &lt; 0.1; red points are buffering paralog GIs with GI &gt; 0.25 and FDR &lt; 0.1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_volcano(gimap_dataset, facet_rep = FALSE, reps_to_drop = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_volcano_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>A special dataset structure that is originally setup using 'setup_data()' and has had gi scores calculated with 'calc_gi()'.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_facet_rep">facet_rep</code></td>
<td>
<p>Should the replicates be wrapped with facet_wrap()?</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_reps_to_drop">reps_to_drop</code></td>
<td>
<p>Names of replicates that should be not plotted (Optional)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 volcano plot of the target level genetic interaction scores.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

gimap_dataset &lt;- get_example_data("gimap") %&gt;%
  gimap_filter() %&gt;%
  gimap_annotate(cell_line = "HELA") %&gt;%
  gimap_normalize(
    timepoints = "day"
  ) %&gt;%
  calc_gi()

# To plot results
plot_exp_v_obs_scatter(gimap_dataset)
plot_rank_scatter(gimap_dataset)
plot_volcano(gimap_dataset)

</code></pre>

<hr>
<h2 id='qc_cdf'>Create a CDF for the pgRNA normalized counts</h2><span id='topic+qc_cdf'></span>

<h3>Description</h3>

<p>This function uses pivot_longer to rearrange the data for plotting
and then plots a CDF of the normalized counts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_cdf(gimap_dataset, wide_ar = 0.75)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qc_cdf_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>The special gimap_dataset from the 'setup_data' function
which contains the transformed data</p>
</td></tr>
<tr><td><code id="qc_cdf_+3A_wide_ar">wide_ar</code></td>
<td>
<p>aspect ratio, default is 0.75</p>
</td></tr>
</table>


<h3>Value</h3>

<p>counts_cdf a ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gimap_dataset &lt;- get_example_data("gimap")
qc_cdf(gimap_dataset)

</code></pre>

<hr>
<h2 id='qc_constructs_countzero_bar'>Create a bar graph that shows the number of replicates with a zero count for pgRNA constructs flagged by the zero count filter</h2><span id='topic+qc_constructs_countzero_bar'></span>

<h3>Description</h3>

<p>This bar graph first uses the specified 'filter_zerocount_target_col' columns to flag pgRNA constructs
that have a raw count of 0 in any one of those columns/samples of interest.
Then, it looks at the specified columns for the final day/sample replicates
('filter_replicates_target_col') to see for pgRNAs that were flagged by the filter,
how many of those replicate samples had raw counts of zeros. And it produces a bar plot reporting on this.
Note, if you select samples/columns to check with the filter that don't have the
replicate samples, this graph won't be informative. So you want there to be overlap
between the columns for the two target_col parameters to have an informative graph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_constructs_countzero_bar(
  gimap_dataset,
  filter_zerocount_target_col = NULL,
  filter_replicates_target_col = NULL,
  wide_ar = 0.75
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qc_constructs_countzero_bar_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>The special gimap_dataset from the 'setup_data' function
which contains the transformed data</p>
</td></tr>
<tr><td><code id="qc_constructs_countzero_bar_+3A_filter_zerocount_target_col">filter_zerocount_target_col</code></td>
<td>
<p>default is NULL; Which sample column(s)
should be used to check for counts of 0? If NULL and not specified, downstream analysis will select all sample columns</p>
</td></tr>
<tr><td><code id="qc_constructs_countzero_bar_+3A_filter_replicates_target_col">filter_replicates_target_col</code></td>
<td>
<p>default is NULL; Which sample columns are
replicates whose variation you'd like to analyze; If NULL, the last 3 sample columns are used</p>
</td></tr>
<tr><td><code id="qc_constructs_countzero_bar_+3A_wide_ar">wide_ar</code></td>
<td>
<p>aspect ratio, default is 0.75</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot barplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gimap_dataset &lt;- get_example_data("gimap")
qc_constructs_countzero_bar(gimap_dataset)

# or if you want to select a specific column(s) for
# looking at where/which samples zero counts are present for
qc_constructs_countzero_bar(gimap_dataset, filter_zerocount_target_col = 3:5)

# or if you want to select a specific column(s) for the final day/sample replicates
qc_constructs_countzero_bar(gimap_dataset, filter_replicates_target_col = 3:5)

# or some combination of those
qc_constructs_countzero_bar(gimap_dataset,
  filter_zerocount_target_col = 3:5,
  filter_replicates_target_col = 3:5
)

</code></pre>

<hr>
<h2 id='qc_cor_heatmap'>Create a correlation heatmap for the pgRNA CPMs</h2><span id='topic+qc_cor_heatmap'></span>

<h3>Description</h3>

<p>This function uses the 'cor' function to find correlations between the
sample CPM's and then plots a heatmap of these
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_cor_heatmap(gimap_dataset)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qc_cor_heatmap_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>The special gimap_dataset from the 'setup_data' function which contains the transformed data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'sample_cor_heatmap' a pheatmap
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gimap_dataset &lt;- get_example_data("gimap")
qc_cor_heatmap(gimap_dataset)

</code></pre>

<hr>
<h2 id='qc_filter_plasmid'>Create a filter for pgRNAs which have a low log2 CPM value for the
plasmid/Day 0 sample/time point</h2><span id='topic+qc_filter_plasmid'></span>

<h3>Description</h3>

<p>This function flags and reports which and how many pgRNAs have
low log2 CPM values for the plasmid/Day 0 sample/time point. If more than one
column is specified as the plasmid sample,
we pool all the replicate samples to find the lower outlier and flag
constructs for which any plasmid replicate has a log2 CPM value below the
cutoff
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_filter_plasmid(
  gimap_dataset,
  cutoff = NULL,
  filter_plasmid_target_col = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qc_filter_plasmid_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>The special gimap_dataset from the 'setup_data' function
which contains the log2 CPM transformed data</p>
</td></tr>
<tr><td><code id="qc_filter_plasmid_+3A_cutoff">cutoff</code></td>
<td>
<p>default is NULL, the cutoff for low log2 CPM values for the
plasmid time period; if not specified, The lower outlier (defined by taking
the difference of the lower quartile and 1.5 * interquartile range) is used</p>
</td></tr>
<tr><td><code id="qc_filter_plasmid_+3A_filter_plasmid_target_col">filter_plasmid_target_col</code></td>
<td>
<p>default is NULL, and if NULL, will select
the first column only; this parameter specifically should be used to specify
the plasmid column(s) that will be selected</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list with the filter 'filter' specifying which pgRNAs have
low plasmid log2 CPM (column of interest is 'plasmid_cpm_filter') and a
report df 'reportdf' for the number and percent of pgRNA which have a low
plasmid log2 CPM
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gimap_dataset &lt;- get_example_data("gimap", data_dir = tempdir())

qc_filter_plasmid(gimap_dataset)

# or to specify a cutoff value to be used in the filter rather than the lower
# outlier default
qc_filter_plasmid(gimap_dataset, cutoff = 2)

# or to specify a different column (or set of columns to select)
qc_filter_plasmid(gimap_dataset, filter_plasmid_target_col = 1:2)

# or to specify a cutoff value that will be used in the filter rather than
# the lower outlier default as well as to specify a different column (or set
# of columns) to select
qc_filter_plasmid(gimap_dataset,
  cutoff = 1.75,
  filter_plasmid_target_col = 1:2
)


</code></pre>

<hr>
<h2 id='qc_filter_zerocounts'>Filter out samples of zero counts
Create a filter for pgRNAs which have a raw count of 0 for any sample/time
# point</h2><span id='topic+qc_filter_zerocounts'></span>

<h3>Description</h3>

<p>This function flags and reports which and how many pgRNAs have a
raw count of 0 for any sample/time point
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_filter_zerocounts(gimap_dataset, filter_zerocount_target_col = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qc_filter_zerocounts_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>The special gimap_dataset from the 'setup_data' function
which contains the raw count data</p>
</td></tr>
<tr><td><code id="qc_filter_zerocounts_+3A_filter_zerocount_target_col">filter_zerocount_target_col</code></td>
<td>
<p>default is NULL; Which sample column(s)
should be used to check for counts of 0? If NULL and not specified,
downstream analysis will select all sample columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list with the filter 'filter' specifying which pgRNA have a
count zero for at least one sample/time point and a report df 'reportdf' for
the number and percent of pgRNA which have a count zero for at least one
sample/time point
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gimap_dataset &lt;- get_example_data("gimap", data_dir = tempdir())
qc_filter_zerocounts(gimap_dataset)

# or to specify a different column (or set of columns to select)
qc_filter_zerocounts(gimap_dataset, filter_zerocount_target_col = 1:2)

</code></pre>

<hr>
<h2 id='qc_plasmid_histogram'>Create a histogram with plasmid log2 CPM values and ascertain a cutoff for low values</h2><span id='topic+qc_plasmid_histogram'></span>

<h3>Description</h3>

<p>Find the distribution of plasmid (day0 data) pgRNA log2 CPM values, and ascertain a cutoff or filter for low log2 CPM values.
Assumes the first column of the dataset is the day0 data; do I need a better
method to tell, especially if there are reps?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_plasmid_histogram(
  gimap_dataset,
  cutoff = NULL,
  filter_plasmid_target_col = NULL,
  wide_ar = 0.75
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qc_plasmid_histogram_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>The special gimap_dataset from the 'setup_data' function which contains the transformed data</p>
</td></tr>
<tr><td><code id="qc_plasmid_histogram_+3A_cutoff">cutoff</code></td>
<td>
<p>default is NULL, the cutoff for low log2 CPM values for the plasmid time period; if not specified, The lower outlier (defined by taking the difference of the lower quartile and 1.5 * interquartile range) is used</p>
</td></tr>
<tr><td><code id="qc_plasmid_histogram_+3A_filter_plasmid_target_col">filter_plasmid_target_col</code></td>
<td>
<p>default is NULL, and if NULL, will select the first column only; this parameter specifically should be used to specify the plasmid column(s) that will be selected</p>
</td></tr>
<tr><td><code id="qc_plasmid_histogram_+3A_wide_ar">wide_ar</code></td>
<td>
<p>aspect ratio, default is 0.75</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot histogram
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gimap_dataset &lt;- get_example_data("gimap")

qc_plasmid_histogram(gimap_dataset)

# or to specify a "cutoff" value that will be displayed as a dashed vertical line
qc_plasmid_histogram(gimap_dataset, cutoff = 1.75)

# or to specify a different column (or set of columns) to select
qc_plasmid_histogram(gimap_dataset, filter_plasmid_target_col = 1:2)

# or to specify a "cutoff" value that will be displayed as a dashed vertical
# line as well as to specify a different column (or set of columns) to select
qc_plasmid_histogram(gimap_dataset, cutoff = 2, filter_plasmid_target_col = 1:2)

</code></pre>

<hr>
<h2 id='qc_sample_hist'>Create a histogram for the pgRNA log2 CPMs, faceted by sample</h2><span id='topic+qc_sample_hist'></span>

<h3>Description</h3>

<p>This function uses pivot_longer to rearrange the data for plotting and then
plots sample specific histograms of the pgRNA cpm's
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_sample_hist(gimap_dataset, wide_ar = 0.75)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qc_sample_hist_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>The special gimap_dataset from the 'setup_data' function
which contains the transformed data</p>
</td></tr>
<tr><td><code id="qc_sample_hist_+3A_wide_ar">wide_ar</code></td>
<td>
<p>aspect ratio, default is 0.75</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sample_cpm_histogram a ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gimap_dataset &lt;- get_example_data("gimap")
qc_sample_hist(gimap_dataset)

</code></pre>

<hr>
<h2 id='qc_variance_hist'>Create a histogram for the variance within replicates for each pgRNA</h2><span id='topic+qc_variance_hist'></span>

<h3>Description</h3>

<p>This function uses pivot_longer to rearrange the data for plotting,
finds the variance for each pgRNA construct (using row number as a proxy) and then
plots a histogram of these variances
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_variance_hist(
  gimap_dataset,
  filter_replicates_target_col = NULL,
  wide_ar = 0.75
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qc_variance_hist_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>The special gimap_dataset from the 'setup_data' function
which contains the transformed data</p>
</td></tr>
<tr><td><code id="qc_variance_hist_+3A_filter_replicates_target_col">filter_replicates_target_col</code></td>
<td>
<p>default is NULL; Which sample columns are
replicates whose variation you'd like to analyze; If NULL, the last 3 sample columns are used</p>
</td></tr>
<tr><td><code id="qc_variance_hist_+3A_wide_ar">wide_ar</code></td>
<td>
<p>aspect ratio, default is 0.75</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot histogram
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gimap_dataset &lt;- get_example_data("gimap")
qc_variance_hist(gimap_dataset)

</code></pre>

<hr>
<h2 id='run_qc'>Run Quality Control Checks</h2><span id='topic+run_qc'></span>

<h3>Description</h3>

<p>This function takes a 'gimap_dataset' and creates a QC report
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_qc(
  gimap_dataset,
  output_file,
  plots_dir,
  overwrite = FALSE,
  filter_zerocount_target_col = NULL,
  filter_plasmid_target_col = NULL,
  filter_replicates_target_col = NULL,
  open_results = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_qc_+3A_gimap_dataset">gimap_dataset</code></td>
<td>
<p>A special dataset structure that is setup using the
'setup_data()' function.</p>
</td></tr>
<tr><td><code id="run_qc_+3A_output_file">output_file</code></td>
<td>
<p>Needs to be a string that ends with &quot;.Rmd&quot; What the
name of the output QC report file should be.</p>
</td></tr>
<tr><td><code id="run_qc_+3A_plots_dir">plots_dir</code></td>
<td>
<p>directory to save plots created with this function,
if it doesn't exist already it will be created</p>
</td></tr>
<tr><td><code id="run_qc_+3A_overwrite">overwrite</code></td>
<td>
<p>default is FALSE; whether to overwrite the QC Report file</p>
</td></tr>
<tr><td><code id="run_qc_+3A_filter_zerocount_target_col">filter_zerocount_target_col</code></td>
<td>
<p>default is NULL; Which sample column(s)
should be
used to check for counts of 0?  If NULL and not specified, downstream
analysis will select all sample columns</p>
</td></tr>
<tr><td><code id="run_qc_+3A_filter_plasmid_target_col">filter_plasmid_target_col</code></td>
<td>
<p>default is NULL; Which sample columns(s)
should be used to look at log2
CPM expression for plasmid pgRNA constructs? If NULL and not specified,
downstream analysis
will select the first sample column only</p>
</td></tr>
<tr><td><code id="run_qc_+3A_filter_replicates_target_col">filter_replicates_target_col</code></td>
<td>
<p>default is NULL; Which sample columns
are replicates whose variation you'd like to analyze; If NULL, the last
3 sample columns are used</p>
</td></tr>
<tr><td><code id="run_qc_+3A_open_results">open_results</code></td>
<td>
<p>default is TRUE but if you don't want the report
automatically opened, choose FALSE.</p>
</td></tr>
<tr><td><code id="run_qc_+3A_...">...</code></td>
<td>
<p>additional parameters are sent to 'rmarkdown::render()'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a QC report saved locally
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gimap_dataset &lt;- get_example_data("gimap")

run_qc(
  gimap_dataset,
  plots_dir = tempdir(),
  output_file = paste0(tempfile(), "_QC_Report.Rmd")
)

</code></pre>

<hr>
<h2 id='setup_data'>Making a new gimap dataset</h2><span id='topic+setup_data'></span>

<h3>Description</h3>

<p>This function allows people to have their data ready to be
processed by gimap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setup_data(counts = NULL, pg_ids = NULL, sample_metadata = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setup_data_+3A_counts">counts</code></td>
<td>
<p>a matrix of data that contains the counts where rows are each
paired_guide target and columns are each sample</p>
</td></tr>
<tr><td><code id="setup_data_+3A_pg_ids">pg_ids</code></td>
<td>
<p>the pgRNA IDs: metadata associated with the pgRNA constructs
that correspond to the rows of the counts data</p>
</td></tr>
<tr><td><code id="setup_data_+3A_sample_metadata">sample_metadata</code></td>
<td>
<p>metadata associated with the samples of the dataset
that correspond to the columns of the counts data.
Should include a column that has replicate information as well as a column
that contains timepoint information respectively (this will be used for log
fold calculations). These columns should be factors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A special gimap_dataset to be used with the other functions in this
package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
counts &lt;- get_example_data("count", data_dir = tempdir()) %&gt;%
  dplyr::select(c(
    "Day00_RepA", "Day05_RepA", "Day22_RepA", "Day22_RepB",
    "Day22_RepC"
  )) %&gt;%
  as.matrix()

pg_ids &lt;- get_example_data("count", data_dir = tempdir()) %&gt;%
  dplyr::select("id")

sample_metadata &lt;- data.frame(
  col_names = c("Day00_RepA", "Day05_RepA", "Day22_RepA", "Day22_RepB", "Day22_RepC"),
  day = as.numeric(c("0", "5", "22", "22", "22")),
  rep = as.factor(c("RepA", "RepA", "RepA", "RepB", "RepC"))
)

gimap_dataset &lt;- setup_data(
  counts = counts,
  pg_ids = pg_ids,
  sample_metadata = sample_metadata
)

</code></pre>

<hr>
<h2 id='supported_cell_lines'>List the supported cell lines</h2><span id='topic+supported_cell_lines'></span>

<h3>Description</h3>

<p>This function downloads the metadata for DepMap and lists which
cell lines are supported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>supported_cell_lines()
</code></pre>


<h3>Value</h3>

<p>A list of the cell line names that are available in DepMap for use
for annotation in this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
cell_lines &lt;- supported_cell_lines()

</code></pre>

<hr>
<h2 id='tpm_setup'>Download and set up DepMap TPM data</h2><span id='topic+tpm_setup'></span>

<h3>Description</h3>

<p>This function sets up the tpm data from DepMap is
called by the 'gimap_annotate()' function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tpm_setup(
  overwrite = TRUE,
  data_dir = system.file("extdata", package = "gimap")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tpm_setup_+3A_overwrite">overwrite</code></td>
<td>
<p>should the files be re downloaded</p>
</td></tr>
<tr><td><code id="tpm_setup_+3A_data_dir">data_dir</code></td>
<td>
<p>What directory should this be saved to? Default is with package
files</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
