<!DOCTYPE html><html><head><title>Help for package GiNA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GiNA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#brewer.pal'><p>Generate a sequence of colors for groups.</p></a></li>
<li><a href='#color.plot'><p>Analyzing the digital color of objects</p></a></li>
<li><a href='#cutoff'><p>Finding the optimal cutoff value for segmentation</p></a></li>
<li><a href='#GiNA-package'><p>Software for high throughput phenotyping of fruit crops</p></a></li>
<li><a href='#GINA.sample'><p>cranberry image</p></a></li>
<li><a href='#jet.scan'><p>Assessing variability for measurements using jet plot</p></a></li>
<li><a href='#pixArea'>
<p>Finding the average area in pixels for objects</p></a></li>
<li><a href='#scanCRAN'><p>High throughput  phenotyping using scanCRAN</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>High Throughput Phenotyping</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-03-29</td>
</tr>
<tr>
<td>Author:</td>
<td>Luis Diaz-Garcia and Giovanny Covarrubias-Pazaran, with collaborations of Brandon Schlautman, Walter Salazar and Juan Zalapa.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Giovanny Covarrubias-Pazaran &lt;covarrubiasp@wisc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs image segmentation in fruit or seeds pictures in order to measure physical features in a high-throughput manner for genome-wide association (GWAS) and genomic selection programs.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.1), methods, stats, EBImage, png, parallel,
doParallel, foreach</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://www.wisc.edu">http://www.wisc.edu</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-04-09 15:29:30 UTC; cova_ruber</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-04-10 00:27:50</td>
</tr>
</table>
<hr>
<h2 id='brewer.pal'>Generate a sequence of colors for groups.</h2><span id='topic+brewer.pal'></span>

<h3>Description</h3>

<p>wrapper of brewer.pal function from RColorBrewer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  brewer.pal(n, name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="brewer.pal_+3A_n">n</code></td>
<td>
<p>Number of different colors in the palette, minimum 3, maximum depending on palette.</p>
</td></tr>
<tr><td><code id="brewer.pal_+3A_name">name</code></td>
<td>
<p>A palette name from the lists below.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of colors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# Plot a colorbar with brewer.pal
mypalette&lt;-brewer.pal(7,"Greens")
}
</code></pre>

<hr>
<h2 id='color.plot'>Analyzing the digital color of objects</h2><span id='topic+color.plot'></span>

<h3>Description</h3>

<p>This function was design to assess the color output of the <code><a href="#topic+scanCRAN">scanCRAN</a></code> function by drawing the digital color detected by the program in a way that the user can assess the colors that the computer is detecting for each object in the picture.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>color.plot(x, cex.axis = 0.7, cex = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="color.plot_+3A_x">x</code></td>
<td>
<p>the data to be used for plotting, should be the output from the 'scanCRAN function.</p>
</td></tr>
<tr><td><code id="color.plot_+3A_cex.axis">cex.axis</code></td>
<td>
<p>the size of the dots in the plot</p>
</td></tr>
<tr><td><code id="color.plot_+3A_cex">cex</code></td>
<td>
<p>the size of the font of the x labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If parameters are indicated correctly the function returns:
</p>

<dl>
<dt>$plot</dt><dd><p>a plot showing the digital color detected by the program for each of the objects found in the picture.</p>
</dd>
</dl>



<h3>References</h3>

<p>We have spent valuable time developing this package, please cite it in your publication:
</p>
<p>Diaz-Garcia L, Covarrubias-Pazaran G, Schlautman B, Zalapa J. GiNA: A flexible high throughput phenotyping tool. http://horticulture.wisc.edu/cggl/ZalapaLab/People.html. 2015.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## x &lt;- ScanCran(folder=0.8,minArea=6000)
## color.plot(x, cex.axis = 0.7, cex = 5)
</code></pre>

<hr>
<h2 id='cutoff'>Finding the optimal cutoff value for segmentation</h2><span id='topic+cutoff'></span>

<h3>Description</h3>

<p>This function reads the picture provided in a path and performs image segmentation in a gray scale version of the picture at different values of color between 0 and 1 depending of the background color. If the background is black the function will segment the picture at values of 0.01 to 0.12 and if a white background is used values 0.5 to 0.8 will be used in order to determine the optimal value of segmentation, such values can be modified by the user in the argument 'prefs'. The resulting images at different cutoff values will be stored in a folder created with the name 'cutoff_vals' in the same folder provided by the user.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cutoff(folder, cores = 1, prefs = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cutoff_+3A_folder">folder</code></td>
<td>
<p>A path where the sample pictures are located.</p>
</td></tr>
<tr><td><code id="cutoff_+3A_cores">cores</code></td>
<td>
<p>A scalar value indicating the number of cores that you want to use for parallelizing the function, default is 1</p>
</td></tr>
<tr><td><code id="cutoff_+3A_prefs">prefs</code></td>
<td>
<p>a vector containing the values of the cutoffvalues to be used in the analysis.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In image analysis segmentation can be done based on different parameters, the most common way to do it is by maximizing the contrast of a background with the object of study (i.e. fruits). Most software uses white or black backgrounds to maximize the contrast with the objects, but a cuttoff value needs to be set in order to get accurate estimates of the objects in the pictures. GiNA has been optimized to be used with any background but an initial cutoff value needs to be provided to the core function <code><a href="#topic+scanCRAN">scanCRAN</a></code>. The purpose of this function is to try different cutoff values and decide the optimal.
</p>


<h3>Value</h3>

<p>If parameters are indicated correctly the function returns:
</p>

<dl>
<dt>$pics</dt><dd><p>pictures in a folder named &quot;cutoff_vals&quot; with different values for segmentation to help the user define the optimum cutoff value for the crop.</p>
</dd>
</dl>



<h3>References</h3>

<p>We have spent valuable time developing this package, please cite it in your publication:
</p>
<p>Diaz-Garcia L, Covarrubias-Pazaran G, Schlautman B, Zalapa J. GiNA: A flexible high throughput phenotyping tool. http://horticulture.wisc.edu/cggl/ZalapaLab/People.html. 2015.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(GiNA)

data(GINA.sample) # RUN
display(GINA.sample)
writeImage(GINA.sample, "gina_cran.JPG") # RUN
folder &lt;- getwd() # RUN
?cutoff
cutoff(folder, cores=1) # RUN

</code></pre>

<hr>
<h2 id='GiNA-package'>Software for high throughput phenotyping of fruit crops</h2><span id='topic+GiNA'></span>

<h3>Description</h3>

<p>GiNA is a package designed for automatic highthoroughput phenotyping analysis of images. The program was conceived to read files with .JPG and .TIFF extensions (which stands for Joint Picture Experts Group) and uses image segmentation based on the <code>EBImage</code> package in order to recognize fruits and compute its features such as area, perimeter, color (in 3 channels), shape and volume to be used for GWAS, QTL or descriptive studies. The core of the package relays in 3 functions; 1) <code><a href="#topic+cutoff">cutoff</a></code> is the function in charge of determining the best cut off value for segmentation, the color scale in R goes from 0 to 1 and in other siftwares from 0 to 265, the function detects the color of the background (black or white) and returns several pictures of segmentation at different cutoff values in order to pick the most suitable for your crop, 2) <code><a href="#topic+pixArea">pixArea</a></code> reads the pictures and returns a plot with the pixel area for each object in the picture in order to provide an estimate of the minimum value to use in the real analysis, the value obtained from this function should be passed to the <code><a href="#topic+scanCRAN">scanCRAN</a></code> function as the minArea to discard all noise from the picture, 3) <code><a href="#topic+scanCRAN">scanCRAN</a></code> is the core of the package and performs image analysis for all pictures stored in the folder path provided at the cutoff value selected. The output from the <code><a href="#topic+scanCRAN">scanCRAN</a></code> function is a list structure with each element of the list being a dataframe with all measurements for the fruits or objects found in the picture.
</p>
<p>Additional plotting functions to visualize the data have been provided such as <code><a href="#topic+color.plot">color.plot</a></code> which displays the digital color detected by the computer for each picture, the input of this function is the output of the scanCRAN function. The function <code><a href="#topic+jet.scan">jet.scan</a></code> returns a heat.map of the variability found in the pictures and takes the same input than 'color.plot' function. For additional information such as tutorials and most recent releases please visit our website http://cggl.horticulture.wisc.edu/software/. 
</p>
<p>We have spent valuable time developing this package, please cite it in your publication:
</p>
<p>Diaz-Garcia L, Covarrubias-Pazaran G, Schlautman B, Zalapa J. GiNA: A flexible high throughput phenotyping tool. http://horticulture.wisc.edu/cggl/ZalapaLab/People.html. 2015.
</p>


<h3>Author(s)</h3>

<p>Luis Diaz-Garcia, Giovanny Covarrubias-Pazaran, Brandon Schlautman, Juan Zalapa.
</p>


<h3>References</h3>

<p>Diaz-Garcia L, Covarrubias-Pazaran G, Schlautman B, Salazar W, Zalapa J. (2015) GiNA: A flexible high throughput phenotyping tool. http://horticulture.wisc.edu/cggl/ZalapaLab/People.html. 2015.
</p>
<p>Ben Hui Liu. 1998. Statistical Genomics. CRC Press LLC. ISBN 0-8493-3166-8.
</p>


<h3>See Also</h3>

<p>http://cggl.horticulture.wisc.edu/software/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>####=========================================####
#### For CRAN time limitations most lines in the 
#### examples are silenced with one '#' mark, 
#### remove them and run the examples
####=========================================####

###########################################
###            GiNA package            ####
###########################################

# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX #
###########################################
### call library 
library(GiNA)

### XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX #
###########################################
### EXPLORE YOUR DATA FIRST TO DETERMINE THE 'minArea' AND 'cutoff' VALUES
### indicate folder where sample pictures are, i.e.
### folder &lt;- "C:/Users/zalapalab/Desktop/examp"

### for demonstration purposes we will call a sample picture
### included in the package, run the following 4 lines

data(GINA.sample) # RUN
display(GINA.sample)
writeImage(GINA.sample, "gina_cran.JPG") # RUN
folder &lt;- getwd() # RUN

### XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX #
############################################
### check the best cutoff values for your pictures

?cutoff
cutoff(folder, cores=1) # RUN

### go to your working directory and analyze the segmentation
### results, you will agree with us than 0.5 is the best value
### for segmentation in the 'cutoff_vals' folder FOR THIS SAMPLE PICTURE

### XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX #
############################################
### check the best minimum area (in pixels) 
### to get rid of little trashes in the picture
### can be provided to the scanCRAN function if 
### you don't like the default

?pixArea
#area &lt;- pixArea(folder,cutoffvalue=0.5,cores=1) # RUN!!!

### XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX #
###########################################
### RUN the scanCRAN function providing your
### cutoff and minimum area values found previously

?scanCRAN
#folder &lt;- getwd()
#mydata &lt;- scanCRAN(folder,cutoffvalue=0.5,cores=1, fact = 1) # RUN!!!
#str(mydata) # RUN!!!

### XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX #
###########################################
### see colors of your fruits in each picture

#color.plot(mydata) # RUN!!!
#jet.scan(mydata, var=3) # RUN!!!

### XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX #
###########################################

</code></pre>

<hr>
<h2 id='GINA.sample'>cranberry image</h2><span id='topic+GINA.sample'></span>

<h3>Description</h3>

<p>Sample image from the cranberry genomics program. The image displays 25 cranberry fruits from a single genotyped captured by a camera in a white background with black reference circles for standarization purposes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(GINA.sample)
</code></pre>


<h3>Format</h3>

<p>An image with 3 layers for green, red and blue. 
</p>


<h3>Source</h3>

<p>Cranberry Genomics Program, Madison, WI, USA.
</p>


<h3>References</h3>

<p>Diaz-Garcia L, Covarrubias-Pazaran G, Schlautman B, Salazar W, Zalapa J. (2015) GINA: A flexible high throughput phenotyping tool. http://horticulture.wisc.edu/cggl/ZalapaLab/People.html. 2015.
</p>
<p>Ben Hui Liu. 1998. Statistical Genomics. CRC Press LLC. ISBN 0-8493-3166-8.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>####=========================================####
#### For CRAN time limitations most lines in the 
#### examples are silenced with one '#' mark, 
#### remove them and run the examples
####=========================================####
data(GINA.sample)
#display(GINA.sample)
#str(GINA.sample)
</code></pre>

<hr>
<h2 id='jet.scan'>Assessing variability for measurements using jet plot</h2><span id='topic+jet.scan'></span>

<h3>Description</h3>

<p>Same than <code><a href="#topic+color.plot">color.plot</a></code> function, this function was designed to assess the results from the <code><a href="#topic+scanCRAN">scanCRAN</a></code> function by providing the output of such function and draw plots showing the variation found in the objects measured in each object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jet.scan(mydata, var = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jet.scan_+3A_mydata">mydata</code></td>
<td>
<p>the data to be used for drawing the jet plot, is the output from the <code><a href="#topic+scanCRAN">scanCRAN</a></code> function</p>
</td></tr>
<tr><td><code id="jet.scan_+3A_var">var</code></td>
<td>
<p>the variable to be plotted from several variables measured by the scanCRAN function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If parameters are indicated correctly the function returns:
</p>

<dl>
<dt>$plot</dt><dd><p>a plot showing the digital representation for the variable selected using a plot with jet colors to emphasize the variability within pictures.</p>
</dd>
</dl>



<h3>References</h3>

<p>We have spent valuable time developing this package, please cite it in your publication:
</p>
<p>Diaz-Garcia L, Covarrubias-Pazaran G, Schlautman B, Zalapa J. GiNA: A flexible high throughput phenotyping tool. http://horticulture.wisc.edu/cggl/ZalapaLab/People.html. 2015.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## x &lt;- scanCran(folder=0.8,minArea=6000)
## jet.scan(x, var = 1)
</code></pre>

<hr>
<h2 id='pixArea'>
Finding the average area in pixels for objects
</h2><span id='topic+pixArea'></span>

<h3>Description</h3>

<p>Even not required, the average value area in pixels for the objects analyzed is a parameter of interest when the objects to be analyzed are particularly small such as seeds or extremely small fruits. Therefore, the use of this function provides a plot of the are in pixels for all the objects in the picture for a given cutoff value provided. Once the value has determined this can be provided to the <code><a href="#topic+scanCRAN">scanCRAN</a></code> function in the argument 'minArea' to make the function perform faster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pixArea(folder, cutoffvalue = 0.5, cores = 1, square = 10, fact = 0.25)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pixArea_+3A_folder">folder</code></td>
<td>
<p>A path where the pictures are located</p>
</td></tr>
<tr><td><code id="pixArea_+3A_cutoffvalue">cutoffvalue</code></td>
<td>
<p>A scalar value where the fruits are distinguishable from the background in the color scale handled by R which is 0-1</p>
</td></tr>
<tr><td><code id="pixArea_+3A_cores">cores</code></td>
<td>
<p>A scalar value indicating the number of cores that you want to use for parallelizing the function</p>
</td></tr>
<tr><td><code id="pixArea_+3A_square">square</code></td>
<td>
<p>size of the square to brush smaller objects from the picture</p>
</td></tr>
<tr><td><code id="pixArea_+3A_fact">fact</code></td>
<td>
<p>a scalar value between 0-1 indicating at what extent the picture should be reduced in size in order to make computations faster, by default the pictures are sized at the 25 percent of their original size. If no reduction is required the user should set the argument to 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If parameters are indicated correctly the function returns:
</p>

<dl>
<dt>$plot</dt><dd><p>a plot showing the size of different objects in the picture in number of pixels.</p>
</dd>
</dl>



<h3>References</h3>

<p>We have spent valuable time developing this package, please cite it in your publication:
</p>
<p>Diaz-Garcia L, Covarrubias-Pazaran G, Schlautman B, Zalapa J. GiNA: A flexible high throughput phenotyping tool. http://horticulture.wisc.edu/cggl/ZalapaLab/People.html. 2015.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GiNA)
data(GINA.sample) # RUN
display(GINA.sample)
writeImage(GINA.sample, "gina_cran.JPG") # RUN
folder &lt;- getwd() # RUN
area &lt;- pixArea(folder,cutoffvalue=0.5,cores=1) # RUN!!!
</code></pre>

<hr>
<h2 id='scanCRAN'>High throughput  phenotyping using scanCRAN</h2><span id='topic+scanCRAN'></span>

<h3>Description</h3>

<p>This function is the core of the GiNA package which is in charge of reading the pictures find in the folder provided and perform the image segmentation in order to find the objects in the picture (i.e. fruits) and measure parameters such as area, perimeter, shape, volume, gray color, red, green and blue color. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scanCRAN(folder, cutoffvalue=NULL, minArea = NULL, 
        cores = 1, gray = TRUE, stand = c(0, 0), 
        fact = 0.25)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scanCRAN_+3A_folder">folder</code></td>
<td>
<p>A path where the pictures are located</p>
</td></tr>
<tr><td><code id="scanCRAN_+3A_cutoffvalue">cutoffvalue</code></td>
<td>
<p>A scalar value (0-1) where the fruits are distinguishable from the background in the color scale handled by R which is 0-1. if not passed the function will determine the color of the background and assign the value of 0.08 if black and 0.6 if white. We highly recommend to use the <code><a href="#topic+cutoff">cutoff</a></code> function to find the optimum segmentation value which will determine the success of your experiment.</p>
</td></tr>
<tr><td><code id="scanCRAN_+3A_minarea">minArea</code></td>
<td>
<p>A scalar number indicating the minimum size in pixels for a real object in the picture to avoid the program to confuse some little trash with an object. The function <code><a href="#topic+cutoff">cutoff</a></code> can provide an idea of the sizes of real objects and noise. If not passed the argument uses minArea=(length/40)*(width/40), where length and width refers to the dimensions of the picture</p>
</td></tr>
<tr><td><code id="scanCRAN_+3A_cores">cores</code></td>
<td>
<p>A scalar value indicating the number of cores that you want to use for parallelizing the function</p>
</td></tr>
<tr><td><code id="scanCRAN_+3A_gray">gray</code></td>
<td>
<p>a TRUE/FALSE value indicating if the user want the color to bereturned in grayscale besides the RGB color</p>
</td></tr>
<tr><td><code id="scanCRAN_+3A_stand">stand</code></td>
<td>
<p>a vector with 2 scalar values for width and length of your references</p>
</td></tr>
<tr><td><code id="scanCRAN_+3A_fact">fact</code></td>
<td>
<p>a scalar value between 0-1 indicating at what extent the picture should be reduced in size in order to make computations faster, by default the pictures are sized at the 25 percent of their original size. If no reduction is required the user should set the argument to 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In order to exploit the advantages of this function we highly recommend the users to determine the best cutoff value with the use of the <code><a href="#topic+cutoff">cutoff</a></code> function to achieve a good segmentation of the objects. The package is sensitive to the prescense of shades in the picture, for that reason we highly recommend the use of black background and reference circles in a clear color (i.e. white).
</p>
<p>This is a rough idea of how you should take the pictures, where the circles are your references and the dots are your fruits. Your fruits shouldn't block the references and YOUR REFERENCES SHOULD BE ALWAYS ON THE SIDES OF YOUR PICTURE. 
</p>
<p>O    . .  .    .  O
</p>
<p>O  .  .  .  .  . O
</p>
<p>O   .  .  . .  . O
</p>
<p>For additional information such as tutorials and most recent releases 
please visit our website http://cggl.horticulture.wisc.edu/software/. 
</p>


<h3>Value</h3>

<p>If parameters are indicated correctly the function returns:
</p>

<dl>
<dt>$x3</dt><dd><p>a list of dataframes, one for each picture provided. The dataframe contains the measurements for each object (rows) found in the puncture for the variables (columns) area, perimeter, length, width, shape (ratio of length/width), volume and red, green, blue, grey color.</p>
</dd>
</dl>



<h3>References</h3>

<p>We have spent valuable time developing this package, please cite it in your publication:
</p>
<p>Diaz-Garcia L, Covarrubias-Pazaran G, Schlautman B, Zalapa J. GiNA: A flexible high throughput phenotyping tool. http://horticulture.wisc.edu/cggl/ZalapaLab/People.html. 2015.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(GiNA)
data(GINA.sample) # RUN
writeImage(GINA.sample, "gina_cran.JPG") # RUN
folder &lt;- getwd()
mydata &lt;- scanCRAN(folder,cutoffvalue=0.5,cores=1, fact = 1) # RUN!!!

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
