<!DOCTYPE html><html lang="en"><head><title>Help for package gJLS2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gJLS2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#chrXdat'><p>X-chromosomal example using the 1000 Genomes Project data</p></a></li>
<li><a href='#gJLS2'><p>A Generalized Joint-Location-Scale (gJLS) Test</p></a></li>
<li><a href='#gJLS2s'><p>generalized Joint-Location-Scale (gJLS) test with summary statistics</p></a></li>
<li><a href='#leveneRegA_per_SNP'><p>The generalized Levene's test via a two-stage regression for variance homogeneity by SNP genotype (autosomes)</p></a></li>
<li><a href='#leveneRegX_per_SNP'><p>Levene's regression tests for variance homogeneity by SNP genotype (X-chromosome specific)</p></a></li>
<li><a href='#leveneTests_per_SNP'><p>Levene's test for variance homogeneity by SNP genotypes (sex-specific p-values)</p></a></li>
<li><a href='#locReg'><p>Location (mean-based association) test</p></a></li>
<li><a href='#scaleReg'><p>Scale (variance-based association) test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>A Generalized Joint Location and Scale Framework for Association
Testing</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>An update to the Joint Location-Scale (JLS) testing framework that identifies associated SNPs, gene-sets and pathways with main and/or interaction effects on quantitative traits (Soave et al., 2015; &lt;<a href="https://doi.org/10.1016%2Fj.ajhg.2015.05.015">doi:10.1016/j.ajhg.2015.05.015</a>&gt;). The JLS method simultaneously tests the null hypothesis of equal mean and equal variance across genotypes, by aggregating association evidence from the individual location/mean-only and scale/variance-only tests using Fisher's method. The generalized joint location-scale (gJLS) framework has been developed to deal specifically with sample correlation and group uncertainty (Soave and Sun, 2017; &lt;<a href="https://doi.org/10.1111%2Fbiom.12651">doi:10.1111/biom.12651</a>&gt;). The current release: gJLS2, include additional functionalities that enable analyses of X-chromosome genotype data through novel methods for location (Chen et al., 2021; &lt;<a href="https://doi.org/10.1002%2Fgepi.22422">doi:10.1002/gepi.22422</a>&gt;) and scale (Deng et al., 2019; &lt;<a href="https://doi.org/10.1002%2Fgepi.22247">doi:10.1002/gepi.22247</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, markdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, nlme, quantreg, MCMCpack, MASS, plyr, ggplot2,
moments</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-29 16:03:28 UTC; weideng</td>
</tr>
<tr>
<td>Author:</td>
<td>Wei Deng [aut, cre],
  Lei Sun [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wei Deng &lt;deng@utstat.toronto.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-30 09:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='chrXdat'>X-chromosomal example using the 1000 Genomes Project data</h2><span id='topic+chrXdat'></span>

<h3>Description</h3>

<p>A dataset containing real genotypes of 5 X-chromosomal SNPs and simulated phenotype of 473 unrelated samples from the 1000 Genomes Project.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chrXdat
</code></pre>


<h3>Format</h3>

<p>A data frame with 473 rows and 11 variables:
</p>

<dl>
<dt>FID</dt><dd><p>Family ID.</p>
</dd>
<dt>IID</dt><dd><p>Individual ID.</p>
</dd>
<dt>PAT</dt><dd><p>Paternal individual ID.</p>
</dd>
<dt>MAT</dt><dd><p>Maternal individual ID.</p>
</dd>
<dt>SEX</dt><dd><p>The genetic sex of a individual.</p>
</dd>
<dt>PHENOTYPE</dt><dd><p>The quantitative trait value, simulated from a standard normal distribution.</p>
</dd>
<dt>rs5983012_A</dt><dd><p>The genotype of SNP with MAF roughly equal to 0.1 in females.</p>
</dd>
<dt>rs4119090_G</dt><dd><p>The genotype of SNP with MAF roughly equal to 0.2 in females.</p>
</dd>
<dt>rs5911042_T</dt><dd><p>The genotype of SNP with MAF roughly equal to 0.3 in females.</p>
</dd>
<dt>rs986810_C</dt><dd><p>The genotype of SNP with MAF roughly equal to 0.4 in females.</p>
</dd>
<dt>rs180495_G</dt><dd><p>The genotype of SNP with MAF roughly equal to 0.45 in females.</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.internationalgenome.org/data/">https://www.internationalgenome.org/data/</a>
</p>

<hr>
<h2 id='gJLS2'>A Generalized Joint-Location-Scale (gJLS) Test</h2><span id='topic+gJLS2'></span>

<h3>Description</h3>

<p>This function takes as input the genotype of a SNP (<code>GENO</code>), the SEX (<code>SEX</code>), and a quantitative trait (<code>Y</code>) in a sample population, and possibly additional covariates, such as principal components. The function returns the location and scale association <em>p</em>-values for each SNP, as well as the gJLS p-value, which provides the combined evidence via Fisher's method (Soave et al., 2015, 2017). To perform this analysis genome-wide, we recommend to use the R-plugin written for PLINK, see <code>gJLSPLINK</code> for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gJLS2(
  GENO,
  Y,
  COVAR = NULL,
  SEX = NULL,
  Xchr = FALSE,
  transformed = TRUE,
  loc_alg = "LAD",
  related = FALSE,
  cov.structure = "corCompSymm",
  clust = NULL,
  genotypic = FALSE,
  origLev = FALSE,
  centre = "median",
  XchrMethod = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gJLS2_+3A_geno">GENO</code></td>
<td>
<p>a list of a genotype matrix/vector of SNPs, must contain values 0, 1, 2's coded for the number of reference allele. Alternatively, for imputed genotypes, it could either be a vector of dosage values between 0 and 2, or a list of matrix of genotype probabilities, numerically between 0 and 1 for each genotype. The length/dimension of <code>GENO</code> should match that of <code>Y</code>, and/or <code>SEX</code> and <code>COVAR</code>.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_y">Y</code></td>
<td>
<p>a vector of quantitative traits, such as human height.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_covar">COVAR</code></td>
<td>
<p>optional: a vector or matrix of covariates that are used to reduce bias due to confounding, such as age.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_sex">SEX</code></td>
<td>
<p>optional: the genetic sex of individuals in the sample population, must be a vector of 1 and 2 following the default sex code is 1 for males and 2 for females in PLINK.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_xchr">Xchr</code></td>
<td>
<p>a logical indicator for whether the analysis is for X-chromosome SNPs.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_transformed">transformed</code></td>
<td>
<p>a logical indicating whether the quantitative response <code>Y</code> should be transformed using a rank-based method to resemble a normal distribution; recommended for traits with non-symmetric distribution. The default option is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_loc_alg">loc_alg</code></td>
<td>
<p>a character indicating the type of algorithm to compute the centre in stage 1; the value is either &quot;OLS&quot;, corresponding to an ordinary linear regression under Gaussian assumptions to compute the mean, or &quot;LAD&quot;, corresponding to a quantile regression to compute the median. The recommended default option is &quot;LAD&quot;. For the quantile regression, the function calls <code>quantreg::rq</code> and the median is estimated using either the &quot;br&quot; (smaller samples) or &quot;sfn&quot; (larger samples and sparse problems) algorithm depending the sample size, for more details see <code>?quantreg::rq</code>.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_related">related</code></td>
<td>
<p>optional: a logical indicating whether the samples should be treated as related; if <code>TRUE</code> while no relatedness covariance information is given, it is then estimated under a <code>cov.structure</code> and assumes this structure among all within-group errors pertaining to the same pair/cluster if specified using <code>clust</code>. This option currently only applies to autosomal SNPs.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_cov.structure">cov.structure</code></td>
<td>
<p>optional: should be one of standard classes of correlation structures listed in <code>corClasses</code> from <span class="pkg">R</span> package <span class="pkg">nlme</span>. See <code>?corClasses</code>. The most commonly used option is <code>corCompSymm</code> for a compound symmetric correlation structure. This option currently only applies to autosomal SNPs.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_clust">clust</code></td>
<td>
<p>optional: a factor indicating the grouping of samples; it should have at least two distinct values. It could be the family ID (FID) for family studies. This option currently only applies to autosomal SNPs.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_genotypic">genotypic</code></td>
<td>
<p>a logical indicating whether the variance homogeneity should be tested with respect to an additively (linearly) coded or non-additively coded <code>geno_one</code>. The former has one less degree of freedom than the latter and is the default option. For dosage genotypes without genotypic probabilities, <code>genotypic</code> is forced to be <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_origlev">origLev</code></td>
<td>
<p>a logical indicator for whether the reported p-values should also include original Levene's test.</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_centre">centre</code></td>
<td>
<p>a character indicating whether the absolute deviation should be calculated with respect to &quot;median&quot; or &quot;mean&quot; in the traditional sex-specific and Fisher combined Levene's test p-values (three tests) for X-chromosome. The default value is &quot;median&quot;. This option applies to sex-specific analysis using original Levene's test (i.e. when <code>regression</code>$$=$$<code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="gJLS2_+3A_xchrmethod">XchrMethod</code></td>
<td>
<p>an integer taking values 0 (reports all models), 1.1, 1.2, 2, 3, for the choice of X-chromosome location association testing models; for more details, see <code><a href="#topic+locReg">locReg</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of location, scale and combined gJLS p-values for each SNP.
</p>


<h3>Note</h3>

<p>For a genome-scan, we recommend to run this in PLINK via the plugin function <code>gJLSPLINK</code>, especially for large datasets and those with more than 20 covariates.
</p>
<p>We highly recommend to quantile-normally transform <code>Y</code> for non-symmetrically distributed traits. This is typically done to avoid ‘scale-effect’ when the variance values tend to be proportional to mean values when stratified by <code>GENO</code>, as observed by Pare et al. (2010) and Yang et al. (2011).
</p>
<p>For the moment, only quantitative trait <code>Y</code> is accepted as the subsequent generalized joint location scale (gJLS) analyses require the variance be calculated on quantitative traits. However, we are working on to include binary response for the generalized JLS analyses in the next update of gJLS.
</p>


<h3>Author(s)</h3>

<p>Wei Q. Deng <a href="mailto:deng@utstat.toronto.edu">deng@utstat.toronto.edu</a>, Lei Sun <a href="mailto:sun@utstat.toronto.edu">sun@utstat.toronto.edu</a>
</p>


<h3>References</h3>

<p>Soave D, Corvol H, Panjwani N, Gong J, Li W, Boëlle PY, Durie PR, Paterson AD, Rommens JM, Strug LJ, Sun L. (2015). A Joint Location-Scale Test Improves Power to Detect Associated SNPs, Gene Sets, and Pathways. <em>American Journal of Human Genetics</em>. 2015 Jul 2;<strong>97</strong>(1):125-38. doi: <a href="https://doi.org/10.1016/j.ajhg.2015.05.015">10.1016/j.ajhg.2015.05.015</a>. PMID: 26140448; PMCID: PMC4572492.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>N &lt;- 1000
genDAT &lt;- rbinom(N, 2, 0.3)
sex &lt;- rbinom(N, 1, 0.5)+1
y &lt;- rnorm(N)
covar &lt;- matrix(rnorm(N*10), ncol=10)

gJLS2(GENO=data.frame("SNP1" = genDAT, "aSNP1" = genDAT), SEX=sex, Y=y, COVAR=covar)

gJLS2(GENO=genDAT, SEX=sex, Y=y, COVAR=covar , Xchr=TRUE)


</code></pre>

<hr>
<h2 id='gJLS2s'>generalized Joint-Location-Scale (gJLS) test with summary statistics</h2><span id='topic+gJLS2s'></span>

<h3>Description</h3>

<p>This function takes as input the gL and gS p-values for each SNP and combine to produce the gJLS p-values. It is used for genome-wide analysis where only the gL or gS p-values are available, caution should be exercised when combing gL and gS p-values obtained from separate datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gJLS2s(gL, gS)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gJLS2s_+3A_gl">gL</code></td>
<td>
<p>a vector of location p-values or a data.frame containing column names &quot;SNP&quot; and &quot;gL&quot;.</p>
</td></tr>
<tr><td><code id="gJLS2s_+3A_gs">gS</code></td>
<td>
<p>a vector of scale p-values or a data.frame containing column names &quot;SNP&quot; and &quot;gS&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of combined gJLS p-values for each SNP.
</p>


<h3>Note</h3>

<p>For a genome-scan, we recommend to run this in PLINK via the plugin function <code>gJLSPLINK</code>, especially for large datasets and those with more than 20 covariates.
</p>
<p>We highly recommend to quantile-normally transform <code>Y</code> for non-symmetrically distributed traits. This is typically done to avoid ‘scale-effect’ when the variance values tend to be proportional to mean values when stratified by <code>GENO</code>, as observed by Pare et al. (2010) and Yang et al. (2011).
</p>
<p>For the moment, only quantitative trait <code>Y</code> is accepted as the subsequent generalized joint location scale (gJLS) analyses require the variance be calculated on quantitative traits. However, we are working on to include binary response for the generalized JLS analyses in the next update of gJLS.
</p>


<h3>Author(s)</h3>

<p>Wei Q. Deng <a href="mailto:deng@utstat.toronto.edu">deng@utstat.toronto.edu</a>, Lei Sun <a href="mailto:sun@utstat.toronto.edu">sun@utstat.toronto.edu</a>
</p>


<h3>References</h3>

<p>Soave D, Corvol H, Panjwani N, Gong J, Li W, Boëlle PY, Durie PR, Paterson AD, Rommens JM, Strug LJ, Sun L. (2015). A Joint Location-Scale Test Improves Power to Detect Associated SNPs, Gene Sets, and Pathways. <em>American Journal of Human Genetics</em>. 2015 Jul 2;<strong>97</strong>(1):125-38. doi: <a href="https://doi.org/10.1016/j.ajhg.2015.05.015">10.1016/j.ajhg.2015.05.015</a>. PMID: 26140448; PMCID: PMC4572492.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gL &lt;- data.frame("SNP" = paste("rs", 1:100, sep=""), "gL"=runif(100))
gS &lt;- runif(100)

gJLS2s(gL = gL, gS=gS)


</code></pre>

<hr>
<h2 id='leveneRegA_per_SNP'>The generalized Levene's test via a two-stage regression for variance homogeneity by SNP genotype (autosomes)</h2><span id='topic+leveneRegA_per_SNP'></span>

<h3>Description</h3>

<p>This function takes as input the genotype of a SNP (<code>GENO</code>), a quantitative trait (<code>Y</code>) in a sample population, and possibly additional covariates, such as principal components. The function returns the scale association <em>p</em>-values for each autosomal SNP using the generalized Levene's test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>leveneRegA_per_SNP(
  geno_one,
  Y,
  COVAR = NULL,
  transformed = TRUE,
  loc_alg = "LAD",
  related = FALSE,
  cov.structure = "corCompSymm",
  clust = NULL,
  genotypic = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="leveneRegA_per_SNP_+3A_geno_one">geno_one</code></td>
<td>
<p>the genotype of a biallelic SNP, must be a vector of 0, 1, 2's coded for the number of reference allele. Alternatively, for imputed genotypes, it could be a matrix/vector of dosage values, numerically between 0 and 2. The length/dimension of <code>geno_one</code> should match that of <code>Y</code>, and/or <code>SEX</code> and <code>COVAR</code>.</p>
</td></tr>
<tr><td><code id="leveneRegA_per_SNP_+3A_y">Y</code></td>
<td>
<p>a vector of quantitative traits, such as human height.</p>
</td></tr>
<tr><td><code id="leveneRegA_per_SNP_+3A_covar">COVAR</code></td>
<td>
<p>optional: a vector or matrix of covariates that are used to reduce bias due to confounding, such as age.</p>
</td></tr>
<tr><td><code id="leveneRegA_per_SNP_+3A_transformed">transformed</code></td>
<td>
<p>a logical indicating whether the quantitative response <code>Y</code> should be transformed using a rank-based method to resemble a normal distribution; recommended for traits with non-symmetric distribution. The default option is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="leveneRegA_per_SNP_+3A_loc_alg">loc_alg</code></td>
<td>
<p>a character indicating the type of algorithm to compute the centre in stage 1; the value is either &quot;OLS&quot;, corresponding to an ordinary linear regression under Gaussian assumptions to compute the mean, or &quot;LAD&quot;, corresponding to a quantile regression to compute the median. The recommended default option is &quot;LAD&quot;. For the quantile regression, the function calls <code>quantreg::rq</code> and the median is estimated using either the &quot;fn&quot; (smaller samples) or &quot;sfn&quot; (larger samples and sparse problems) algorithm depending the sample size, for more details see <code>?quantreg::rq</code>.</p>
</td></tr>
<tr><td><code id="leveneRegA_per_SNP_+3A_related">related</code></td>
<td>
<p>optional: a logical indicating whether the samples should be treated as related; if <code>TRUE</code> while no relatedness covariance information is given, it is then estimated under a <code>cov.structure</code> and assumes this structure among all within-group errors pertaining to the same pair/cluster if specified using <code>clust</code>. This option currently only applies to autosomal SNPs.</p>
</td></tr>
<tr><td><code id="leveneRegA_per_SNP_+3A_cov.structure">cov.structure</code></td>
<td>
<p>optional: should be one of standard classes of correlation structures listed in <code>corClasses</code> from <span class="pkg">R</span> package <span class="pkg">nlme</span>. See <code>?corClasses</code>. The most commonly used option is <code>corCompSymm</code> for a compound symmetric correlation structure. This option currently only applies to autosomal SNPs.</p>
</td></tr>
<tr><td><code id="leveneRegA_per_SNP_+3A_clust">clust</code></td>
<td>
<p>optional: a factor indicating the grouping of samples; it should have at least two distinct values. It could be the family ID (FID) for family studies. This option currently only applies to autosomal SNPs.</p>
</td></tr>
<tr><td><code id="leveneRegA_per_SNP_+3A_genotypic">genotypic</code></td>
<td>
<p>optional: a logical indicating whether the variance homogeneity should be tested with respect to an additively (linearly) coded or non-additively coded <code>geno_one</code>. The former has one less degree of freedom than the latter and is the default option. For dosage data without genotypic probabilities, <code>genotypic</code> is forced to be <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Levene's test regression p-values for autosomal SNPs according to the model specified.
</p>


<h3>Note</h3>

<p>We recommend to quantile-normally transform <code>Y</code> to avoid ‘scale-effect’ where
the variance values tend to be proportional to mean values when stratified by <code>geno_one</code>.
</p>
<p>When the relatedness option is used, the computational time is expected to be longer for larger sample size ($$n &gt; 1000$$), thus we recommend this option for smaller studies rather than large population based studies.
</p>
<p>There is no explicit argument to supply <code>SEX</code> for autosomal SNPs, the user can choose to include the genetic sex of individuals as a column of the <code>COVAR</code> argument.
</p>


<h3>Author(s)</h3>

<p>Wei Q. Deng <a href="mailto:deng@utstat.toronto.edu">deng@utstat.toronto.edu</a>, Lei Sun <a href="mailto:sun@utstat.toronto.edu">sun@utstat.toronto.edu</a>
</p>


<h3>References</h3>

<p>Soave D, Sun L. (2017). A generalized Levene's scale test for variance heterogeneity in the presence of sample correlation and group uncertainty. <em>Biometrics</em>. <strong>73</strong>(3):960-971. doi: <a href="https://doi.org/10.1111/biom.12651">10.1111/biom.12651</a>. PMID: 28099998.
</p>
<p>Gastwirth JL, Gel YR, Miao W. (2009). The Impact of Levene's Test of Equality of Variances on Statistical Theory and Practice.&quot; <em>Statistical Science</em>. <strong>24</strong>(3) 343 - 360, doi: <a href="https://doi.org/10.1214/09-STS301">10.1214/09-STS301</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>N &lt;- 100
genDAT &lt;- rbinom(N, 2, 0.3)
Y &lt;- rnorm(N)
covar &lt;- matrix(rnorm(N*10), ncol=10)

# vanilla example:
leveneRegA_per_SNP(geno_one=genDAT, Y=Y, COVAR=covar)

# relatedness samples:
leveneRegA_per_SNP(geno_one=genDAT, Y=Y, COVAR=covar,
related=TRUE)
leveneRegA_per_SNP(geno_one=genDAT, Y=Y, COVAR=covar,
related=TRUE, clust = factor(rbinom(N, 2, 0.6)))


# dosage genotypes example:
library("MCMCpack")
a &lt;- 0.3
geno &lt;- rbinom(N, 2, 0.3)
a &lt;- 0.3 ## uncertainty
genPP &lt;- rbind(rdirichlet(sum(geno==0),c(a,(1-a)/2,(1-a)/2)),
               rdirichlet(sum(geno==1),c((1-a)/2,a,(1-a)/2)),
               rdirichlet(sum(geno==2),c((1-a)/2,(1-a)/2,a)))

leveneRegA_per_SNP(geno_one=genPP, Y=Y, COVAR=covar)
leveneRegA_per_SNP(geno_one=genPP, Y=Y, COVAR=covar,
genotypic=TRUE)

# dosage and related samples:
leveneRegA_per_SNP(geno_one=genPP, Y=Y, COVAR=covar,
related=TRUE, clust = factor(rbinom(N, 1, 0.3)))
leveneRegA_per_SNP(geno_one=genPP, Y=Y, COVAR=covar,
related=TRUE, clust = factor(rbinom(N, 1, 0.3)), genotypic=TRUE)



</code></pre>

<hr>
<h2 id='leveneRegX_per_SNP'>Levene's regression tests for variance homogeneity by SNP genotype (X-chromosome specific)</h2><span id='topic+leveneRegX_per_SNP'></span>

<h3>Description</h3>

<p>This function takes as input the genotype of a SNP (<code>geno_one</code>), the genetic sex (<code>SEX</code>), a quantitative trait (<code>Y</code>) in a sample population, and possibly additional covariates, such as principal components. The function returns the scale association <em>p</em>-values for each X-chromosome SNP using the generalized Levene's test designed for X-chromosome biallelic markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>leveneRegX_per_SNP(
  geno_one,
  SEX,
  Y,
  COVAR = NULL,
  genotypic = FALSE,
  transformed = TRUE,
  loc_alg = "LAD"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="leveneRegX_per_SNP_+3A_geno_one">geno_one</code></td>
<td>
<p>the genotype of a biallelic SNP, must be a vector of 0, 1, 2's coded for the number of reference allele. Alternatively, for imputed genotypes, it could be a matrix/vector of dosage values, numerically between 0 and 2. The length/dimension of <code>geno_one</code> should match that of <code>Y</code>, and/or <code>SEX</code> and <code>COVAR</code>.</p>
</td></tr>
<tr><td><code id="leveneRegX_per_SNP_+3A_sex">SEX</code></td>
<td>
<p>optional: the genetic sex of individuals in the sample population, must be a vector of 1 and 2 following the default sex code is 1 for males and 2 for females in PLINK.</p>
</td></tr>
<tr><td><code id="leveneRegX_per_SNP_+3A_y">Y</code></td>
<td>
<p>a vector of quantitative traits, such as human height.</p>
</td></tr>
<tr><td><code id="leveneRegX_per_SNP_+3A_covar">COVAR</code></td>
<td>
<p>optional: a vector or matrix of covariates that are used to reduce bias due to confounding, such as age.</p>
</td></tr>
<tr><td><code id="leveneRegX_per_SNP_+3A_genotypic">genotypic</code></td>
<td>
<p>optional: a logical indicating whether the variance homogeneity should be tested with respect to an additively (linearly) coded or non-additively coded <code>geno_one</code>. The former has one less degree of freedom than the latter and is the default option. For dosage genotypes without genotypic probabilities, <code>genotypic</code> is forced to be <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="leveneRegX_per_SNP_+3A_transformed">transformed</code></td>
<td>
<p>a logical indicating whether the quantitative response <code>Y</code> should be transformed using a rank-based method to resemble a normal distribution; recommended for traits with non-symmetric distribution. The default option is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="leveneRegX_per_SNP_+3A_loc_alg">loc_alg</code></td>
<td>
<p>a character indicating the type of algorithm to compute the centre in stage 1; the value is either &quot;OLS&quot;, corresponding to an ordinary linear regression under Gaussian assumptions to compute the mean, or &quot;LAD&quot;, corresponding to a quantile regression to compute the median. The recommended default option is &quot;LAD&quot;. For the quantile regression, the function calls <code>quantreg::rq</code> and the median is estimated using either the &quot;fn&quot; (smaller samples) or &quot;sfn&quot; (larger samples and sparse problems) algorithm depending the sample size, for more details see <code>?quantreg::rq</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the Levene's test regression p-value according to the model specified.
</p>


<h3>Note</h3>

<p>We recommend to quantile-normally transform <code>Y</code> to avoid ‘scale-effect’ where
the variance values tend to be proportional to mean values when stratified by <code>geno_one</code>.
</p>


<h3>Author(s)</h3>

<p>Wei Q. Deng <a href="mailto:deng@utstat.toronto.edu">deng@utstat.toronto.edu</a>, Lei Sun <a href="mailto:sun@utstat.toronto.edu">sun@utstat.toronto.edu</a>
</p>


<h3>References</h3>

<p>Deng WQ, Mao S, Kalnapenkis A, Esko T, Magi R, Pare G, Sun L. (2019) Analytical strategies to include the X-chromosome in variance heterogeneity analyses: Evidence for trait-specific polygenic variance structure. <em>Genet Epidemiol</em>. <strong>43</strong>(7):815-830. doi: <a href="https://doi.org/10.1002/gepi.22247">10.1002/gepi.22247</a>. PMID:31332826.
</p>
<p>Gastwirth JL, Gel YR, Miao W. (2009). The Impact of Levene's Test of Equality of Variances on Statistical Theory and Practice. <em>Statistical Science</em>. <strong>24</strong>(3) 343-360, doi: <a href="https://doi.org/10.1214/09-STS301">10.1214/09-STS301</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>N &lt;- 1000
sex &lt;- rbinom(N, 1, 0.5)+1
Y &lt;- rnorm(N)
genDAT &lt;- NA
genDAT[sex==2] &lt;- rbinom(sum(sex==2), 2, 0.3)
table(genDAT, sex)
genDAT[sex==1] &lt;- rbinom(sum(sex==1), 1, 0.3)
table(genDAT, sex)

leveneRegX_per_SNP(geno_one=genDAT, SEX=sex, Y=Y)
leveneRegX_per_SNP(geno_one=genDAT, SEX=sex, Y=Y, genotypic=TRUE)
leveneRegX_per_SNP(geno_one=genDAT, SEX=sex, Y=Y, loc_alg="OLS")

</code></pre>

<hr>
<h2 id='leveneTests_per_SNP'>Levene's test for variance homogeneity by SNP genotypes (sex-specific p-values)</h2><span id='topic+leveneTests_per_SNP'></span>

<h3>Description</h3>

<p>This function takes as input the genotype of a SNP (<code>geno_one</code>), the genetic sex (<code>SEX</code>), a quantitative trait (<code>Y</code>) in a sample population. The function then returns the variance heterogeneity <em>p</em>-values for each sex and the overall variance heterogeneity signal using Fisher's method by combining the sex-specific results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>leveneTests_per_SNP(
  geno_one,
  SEX = NULL,
  Y,
  centre = "median",
  transformed = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="leveneTests_per_SNP_+3A_geno_one">geno_one</code></td>
<td>
<p>the genotype of a bi-allelic SNP, must be a vector of 0, 1, 2's coded for the number of reference allele. Alternatively, for imputed genotypes, it could be a matrix/vector of dosage values, numerically between 0 and 2. The length/dimension of <code>geno_one</code> should match that of <code>Y</code>, and/or <code>SEX</code> and <code>COVAR</code>.</p>
</td></tr>
<tr><td><code id="leveneTests_per_SNP_+3A_sex">SEX</code></td>
<td>
<p>optional: the genetic sex of individuals in the sample population, must be a vector of 1 and 2 following the default sex code is 1 for males and 2 for females in PLINK.</p>
</td></tr>
<tr><td><code id="leveneTests_per_SNP_+3A_y">Y</code></td>
<td>
<p>a vector of quantitative traits, such as human height.</p>
</td></tr>
<tr><td><code id="leveneTests_per_SNP_+3A_centre">centre</code></td>
<td>
<p>a character indicating whether the absolute deviation should be calculated with respect to &quot;median&quot; or &quot;mean&quot;, the default option is &quot;median&quot;.</p>
</td></tr>
<tr><td><code id="leveneTests_per_SNP_+3A_transformed">transformed</code></td>
<td>
<p>a logical indicating whether the quantitative response <code>Y</code> should be transformed using a rank-based method to resemble a normal distribution; recommended for traits with non-symmetric distribution. The default option is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of Levene's test p-values according to levels specified by <code>geno_one</code> in each sex and the Fisher's method to combine the sex-specific Levene's test <em>p</em>-values.
</p>


<h3>Note</h3>

<p>We recommend to quantile-normally transform <code>Y</code> to avoid ‘scale-effect’ where
the variance values tend to be proportional to mean values when stratified by <code>G</code>.
</p>


<h3>Author(s)</h3>

<p>Wei Q. Deng <a href="mailto:deng@utstat.toronto.edu">deng@utstat.toronto.edu</a>, Lei Sun <a href="mailto:sun@utstat.toronto.edu">sun@utstat.toronto.edu</a>
</p>


<h3>References</h3>

<p>Levene H. (1960) Robust tests for equality of variances. In <em>Contributions to Probability and Statistics: Essays in Honor of Harold Hotelling</em> eds:I. Olkin, S.G. Ghurye, W. Hoeffding, W.G. Madow &amp; H.B.Mann, pp.278-292. Stanford: Stanford University Press.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>N &lt;- 5000
sex &lt;- rbinom(N, 1, 0.5)+1
genDAT &lt;- rbinom(N, 2, 0.3)
y &lt;- rnorm(N);

genDAT[sex==2] &lt;- rbinom(sum(sex==2), 1, 0.3)
table(genDAT, sex)
leveneTests_per_SNP(geno_one=genDAT, SEX=sex, Y=y^2, transform=TRUE)

genDAT[sex==2] &lt;- rbinom(sum(sex==2), 1, 0.01)
table(genDAT, sex)
leveneTests_per_SNP(geno_one=genDAT, SEX=sex, Y=y^2, transform=FALSE)

leveneTests_per_SNP(geno_one=rep(0, N), SEX=sex, Y=y^2, transform=TRUE)
leveneTests_per_SNP(geno_one=rep(0, N), Y=y^2, transform=TRUE)


</code></pre>

<hr>
<h2 id='locReg'>Location (mean-based association) test</h2><span id='topic+locReg'></span>

<h3>Description</h3>

<p>This function takes as input the genotype of SNPs (<code>GENO</code>), the SEX (<code>SEX</code>), and a quantitative trait (<code>Y</code>) in a sample population, and possibly additional covariates, such as principal components. The function returns the location association <em>p</em>-values for each SNP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locReg(
  GENO,
  Y,
  SEX = NULL,
  COVAR = NULL,
  Xchr = FALSE,
  XchrMethod = 3,
  transformed = FALSE,
  related = FALSE,
  cov.structure = "corCompSymm",
  clust = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="locReg_+3A_geno">GENO</code></td>
<td>
<p>a list of a genotype matrix/vector of SNPs, must contain values 0, 1, 2's coded for the number of reference allele. Alternatively, for imputed genotypes, it could either be a vector of dosage values between 0 and 2, or a list of matrix of genotype probabilities, numerically between 0 and 1 for each genotype. The length/dimension of <code>GENO</code> should match that of <code>Y</code>, and/or <code>SEX</code> and <code>COVAR</code>.</p>
</td></tr>
<tr><td><code id="locReg_+3A_y">Y</code></td>
<td>
<p>a numeric vector of quantitative trait, such as human height.</p>
</td></tr>
<tr><td><code id="locReg_+3A_sex">SEX</code></td>
<td>
<p>the genetic sex of individuals in the sample population, must be a vector of 1's and 2's following PLINK default coding, where males are coded as 1 and females 2. <strong>Optional for analysis of autosomal SNPs, but required for X-chromosome.</strong></p>
</td></tr>
<tr><td><code id="locReg_+3A_covar">COVAR</code></td>
<td>
<p>optional: a vector or a matrix of covariates, such as age or principal components.</p>
</td></tr>
<tr><td><code id="locReg_+3A_xchr">Xchr</code></td>
<td>
<p>a logical indicator for whether the analysis is for X-chromosome SNPs, if <code>TRUE</code> then the following association testing model is used: Y~G+G_D+S+GxS; with p-value given by comparing Y~G+S+GxS vs. Y~S (G is the additive coded genotype; G_D is an indicator for female heterozygotes).</p>
</td></tr>
<tr><td><code id="locReg_+3A_xchrmethod">XchrMethod</code></td>
<td>
<p>an integer taking values 0 (reports all models), 1.1, 1.2, 2, 3, for the choice of X-chromosome association testing models:
model 1,1: Y~G (females only)
model 1.2: Y~G (males only)
model 2: Y~G+S+GxSex; with p-value given by comparing Y~G+Sex+GxSex vs. Y~Sex (the additively coded G is robust to X-chromosome inactivation uncertainty). This is also the option for dosage genotypes.
model 3 (recommended): Y~G+G_D+S+GxSex; with p-value given by comparing Y ~ G+G_D+Sex+GxSex vs. Y ~ Sex (G_D is an indicator for female heterozygotes, this model is robust to X-chromosome inactivation uncertainty and skewed inactivation). For imputed data in the form of genotypic probabilities, the model becomes: Y ~ G1 + G2 + G1xSex + Sex, where G1 and G2 are the genotypic probabilities for the heterozygote and alternative allele homozygote.</p>
</td></tr>
<tr><td><code id="locReg_+3A_transformed">transformed</code></td>
<td>
<p>a logical indicating whether the quantitative response <code>Y</code> should be transformed using a rank-based method to resemble a normal distribution; recommended for traits with non-symmetric distribution. The default option is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="locReg_+3A_related">related</code></td>
<td>
<p>optional: a logical indicating whether the samples should be treated as related; if <code>TRUE</code> while no relatedness covariance information is given, it is then estimated under a <code>cov.structure</code> and assumes this structure among all within-group errors pertaining to the same pair/cluster if specified using <code>clust</code>. This option currently only applies to autosomal SNPs.</p>
</td></tr>
<tr><td><code id="locReg_+3A_cov.structure">cov.structure</code></td>
<td>
<p>optional: should be one of standard classes of correlation structures listed in <code>corClasses</code> from <span class="pkg">R</span> package <span class="pkg">nlme</span>. See <code>?corClasses</code>. The most commonly used option is <code>corCompSymm</code> for a compound symmetric correlation structure. This option currently only applies to autosomal SNPs.</p>
</td></tr>
<tr><td><code id="locReg_+3A_clust">clust</code></td>
<td>
<p>optional: a factor indicating the grouping of samples; it should have at least two distinct values. It could be the family ID (FID) for family studies. This option currently only applies to autosomal SNPs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of location association <em>p</em>-values for each SNP.
</p>


<h3>Note</h3>

<p>The choice to use a rank-based inverse normal transformation is left to the user's discretion. See XXX for a discussion on the pros and cons of quantile transformation with respect to location association.
</p>
<p>For X-chromosome markers, when the samples consist entirely of females or males, we report only results from model 1, regardless of the <code>XchrMethod</code> option.
</p>


<h3>Author(s)</h3>

<p>Wei Q. Deng <a href="mailto:deng@utstat.toronto.edu">deng@utstat.toronto.edu</a>, Lei Sun <a href="mailto:sun@utstat.toronto.edu">sun@utstat.toronto.edu</a>
</p>


<h3>References</h3>

<p>Chen B, Craiu RV, Sun L. (2020) Bayesian model averaging for the X-chromosome inactivation dilemma in genetic association study. <em>Biostatistics</em>. <strong>21</strong>(2):319-335. doi: <a href="https://doi.org/10.1093/biostatistics/kxy049">10.1093/biostatistics/kxy049</a>. PMID: 30247537.
</p>
<p>Chen B, Craiu RV, Strug LJ, Sun L. (2021) The X factor: A robust and powerful approach to X-chromosome-inclusive whole-genome association studies. <em>Genetic Epidemiology</em>. doi: <a href="https://doi.org/10.1002/gepi.22422">10.1002/gepi.22422</a>. PMID: 34224641.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>N &lt;- 100
genDAT &lt;- rbinom(N, 2, 0.3)
sex &lt;- rbinom(N, 1, 0.5)+1
y &lt;- rnorm(N)
COVAR &lt;- matrix(rnorm(N*10), ncol=10)

locReg(GENO=genDAT, SEX=sex, Y=y, COVAR=COVAR)

# correlated example:
library("MASS")
yy &lt;- mvrnorm(1, mu= rep(0, N), Sigma = matrix(0.3, N, N) + diag(0.7, N))
locReg(GENO=list("SNP1"= genDAT, "SNP2" = genDAT[sample(1:100)]),
SEX=sex, Y=as.numeric(yy), COVAR=COVAR, related = TRUE,
clust = rep(1, 100))

# sibpair example:
pairedY &lt;- mvrnorm(N/2,rep(0,2),matrix(c(1,0.2,0.2,1), 2))
yy &lt;- c(pairedY[,1], pairedY[,2])
locReg(GENO=list("SNP1"= genDAT, "SNP2" = genDAT[sample(1:100)]),
SEX=sex, Y=as.numeric(yy), COVAR=COVAR, related = TRUE,
clust = rep(c(1:50), 2))

# Xchr data example:
genDAT1 &lt;- rep(NA, N)
genDAT1[sex==1] &lt;- rbinom(sum(sex==1), 1, 0.5)
genDAT1[sex==2] &lt;-rbinom(sum(sex==2), 2, 0.5)
locReg(GENO=genDAT1, SEX=sex, Y=y, COVAR=COVAR, Xchr=TRUE)


</code></pre>

<hr>
<h2 id='scaleReg'>Scale (variance-based association) test</h2><span id='topic+scaleReg'></span>

<h3>Description</h3>

<p>This function takes as input the genotype of SNPs (<code>GENO</code>), the SEX (<code>SEX</code>), and a quantitative trait (<code>Y</code>) in a sample population, and possibly additional covariates, such as principal components. The function returns the scale association <em>p</em>-values for each SNP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scaleReg(
  GENO,
  Y,
  COVAR = NULL,
  SEX = NULL,
  Xchr = FALSE,
  transformed = FALSE,
  loc_alg = "LAD",
  related = FALSE,
  cov.structure = "corCompSymm",
  clust = NULL,
  genotypic = FALSE,
  origLev = FALSE,
  centre = "median"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scaleReg_+3A_geno">GENO</code></td>
<td>
<p>a list of a genotype matrix/vector of SNPs, must contain values 0, 1, 2's coded for the number of reference allele. Alternatively, for imputed genotypes, it could either be a vector of dosage values between 0 and 2, or a list of matrix of genotype probabilities, numerically between 0 and 1 for each genotype. The length/dimension of <code>GENO</code> should match that of <code>Y</code>, and/or <code>SEX</code> and <code>COVAR</code>.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_y">Y</code></td>
<td>
<p>a vector of quantitative traits, such as human height.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_covar">COVAR</code></td>
<td>
<p>optional: a vector or matrix of covariates that are used to reduce bias due to confounding, such as age.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_sex">SEX</code></td>
<td>
<p>optional: the genetic sex of individuals in the sample population, must be a vector of 1 and 2 following the default sex code is 1 for males and 2 for females in PLINK.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_xchr">Xchr</code></td>
<td>
<p>a logical indicator for whether the analysis is for X-chromosome SNPs.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_transformed">transformed</code></td>
<td>
<p>a logical indicating whether the quantitative response <code>Y</code> should be transformed using a rank-based method to resemble a normal distribution; recommended for traits with non-symmetric distribution. The default option is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_loc_alg">loc_alg</code></td>
<td>
<p>a character indicating the type of algorithm to compute the centre in stage 1; the value is either &quot;OLS&quot;, corresponding to an ordinary linear regression under Gaussian assumptions to compute the mean, or &quot;LAD&quot;, corresponding to a quantile regression to compute the median. The recommended default option is &quot;LAD&quot;. For the quantile regression, the function calls <code>quantreg::rq</code> and the median is estimated using either the &quot;br&quot; (smaller samples) or &quot;sfn&quot; (larger samples and sparse problems) algorithm depending the sample size, for more details see <code>?quantreg::rq</code>.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_related">related</code></td>
<td>
<p>optional: a logical indicating whether the samples should be treated as related; if <code>TRUE</code> while no relatedness covariance information is given, it is then estimated under a <code>cov.structure</code> and assumes this structure among all within-group errors pertaining to the same pair/cluster if specified using <code>clust</code>. This option currently only applies to autosomal SNPs.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_cov.structure">cov.structure</code></td>
<td>
<p>optional: should be one of standard classes of correlation structures listed in <code>corClasses</code> from <span class="pkg">R</span> package <span class="pkg">nlme</span>. See <code>?corClasses</code>. The most commonly used option is <code>corCompSymm</code> for a compound symmetric correlation structure. This option currently only applies to autosomal SNPs.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_clust">clust</code></td>
<td>
<p>optional: a factor indicating the grouping of samples; it should have at least two distinct values. It could be the family ID (FID) for family studies. This option currently only applies to autosomal SNPs.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_genotypic">genotypic</code></td>
<td>
<p>a logical indicating whether the variance homogeneity should be tested with respect to an additively (linearly) coded or non-additively coded <code>geno_one</code>. The former has one less degree of freedom than the latter and is the default option. For dosage genotypes without genotypic probabilities, <code>genotypic</code> is forced to be <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_origlev">origLev</code></td>
<td>
<p>a logical indicator for whether the reported p-values should also include original Levene's test.</p>
</td></tr>
<tr><td><code id="scaleReg_+3A_centre">centre</code></td>
<td>
<p>a character indicating whether the absolute deviation should be calculated with respect to &quot;median&quot; or &quot;mean&quot; in the traditional sex-specific and Fisher combined Levene's test p-values (three tests) for X-chromosome. The default value is &quot;median&quot;. This option applies to sex-specific analysis using original Levene's test (i.e. when <code>regression</code>$$=$$<code>TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of Levene's test regression p-values according to the models
specified.
</p>


<h3>Note</h3>

<p>We recommend to quantile-normally transform <code>Y</code> to avoid ‘scale-effect’ where
the variance values tend to be proportional to mean values when stratified by <code>GENO</code>.
</p>


<h3>Author(s)</h3>

<p>Wei Q. Deng <a href="mailto:deng@utstat.toronto.edu">deng@utstat.toronto.edu</a>, Lei Sun <a href="mailto:sun@utstat.toronto.edu">sun@utstat.toronto.edu</a>
</p>


<h3>References</h3>

<p>Deng WQ, Mao S, Kalnapenkis A, Esko T, Magi R, Pare G, Sun L. (2019) Analytical strategies to include the X-chromosome in variance heterogeneity analyses: Evidence for trait-specific polygenic variance structure. <em>Genet Epidemiol</em>. <strong>43</strong>(7):815-830. doi: <a href="https://doi.org/10.1002/gepi.22247">10.1002/gepi.22247</a>. PMID:31332826.
</p>
<p>Gastwirth JL, Gel YR, Miao W. (2009). The Impact of Levene's Test of Equality of Variances on Statistical Theory and Practice.&quot; <em>Statistical Science</em>. <strong>24</strong>(3) 343-360, doi: <a href="https://doi.org/10.1214/09-STS301">10.1214/09-STS301</a>.
</p>
<p>Soave D, Sun L. (2017). A generalized Levene's scale test for variance heterogeneity in the presence of sample correlation and group uncertainty. <em>Biometrics</em>. <strong>73</strong>(3):960-971. doi: <a href="https://doi.org/10.1111/biom.12651">10.1111/biom.12651</a>. PMID:28099998.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>N &lt;- 1000
genoDAT &lt;- rbinom(N, 2, 0.3)
sex &lt;- rbinom(N, 1, 0.5)+1
Y &lt;- rnorm(N)
covar &lt;- matrix(rnorm(N*10), ncol=10)

# vanilla example:
scaleReg(GENO=list(genoDAT, genoDAT), Y=Y, COVAR=covar)
scaleReg(GENO=list(genoDAT, genoDAT), Y=Y, COVAR=covar, genotypic=TRUE)
scaleReg(GENO=list(genoDAT, genoDAT), Y=Y, COVAR=covar, origLev = TRUE)
scaleReg(GENO=list(genoDAT, genoDAT), Y=Y, COVAR=covar, origLev = TRUE, SEX=sex)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
