<!DOCTYPE html><html lang="en"><head><title>Help for package gkmSVM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gkmSVM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gkmSVM-package'>
<p>Gapped-Kmer Support Vector Machine</p></a></li>
<li><a href='#genNullSeqs'>
<p>Generating GC/repeat matched randomly selected genomic sequences for the negative set</p></a></li>
<li><a href='#gkmsvm_classify'><p>Classifying(/scoring) new sequences using the gkmSVM model</p></a></li>
<li><a href='#gkmsvm_delta'>
<p>Calculating deltaSVM scores</p></a></li>
<li><a href='#gkmsvm_kernel'><p>Computing the kernel matrix</p></a></li>
<li><a href='#gkmsvm_train'><p>Training the SVM model</p></a></li>
<li><a href='#gkmsvm_trainCV'><p>Training the SVM model, using repeated CV to tune parameter C and plot ROC curves</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gapped-Kmer Support Vector Machine</td>
</tr>
<tr>
<td>Version:</td>
<td>0.83.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Mahmoud Ghandi</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mike Beer &lt;mbeer@jhu.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Imports the 'gkmSVM' v2.0 functionalities into R <a href="https://www.beerlab.org/gkmsvm/">https://www.beerlab.org/gkmsvm/</a>
    It also uses the 'kernlab' library (separate R package by different authors) for various SVM algorithms.
    Users should note that the suggested packages 'rtracklayer', 'GenomicRanges', 'BSgenome', 'BiocGenerics', 
    'Biostrings', 'GenomeInfoDb', 'IRanges', and 'S4Vectors' are all BioConductor packages <a href="https://bioconductor.org">https://bioconductor.org</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>kernlab, seqinr, utils, ROCR, Rcpp, grDevices, graphics, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rtracklayer, GenomicRanges, BSgenome,
BSgenome.Hsapiens.UCSC.hg18.masked,
BSgenome.Hsapiens.UCSC.hg19.masked, BiocGenerics, Biostrings,
GenomeInfoDb, IRanges, S4Vectors</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-20 16:51:57 UTC; mbeer3</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-20 17:12:32 UTC</td>
</tr>
</table>
<hr>
<h2 id='gkmSVM-package'>
Gapped-Kmer Support Vector Machine
</h2><span id='topic+gkmSVM-package'></span><span id='topic+gkmSVM'></span>

<h3>Description</h3>

<p>Imports the 'gkmSVM' v2.0 functionalities into R &lt;http://www.beerlab.org/gkmsvm/&gt; . It also uses the 'kernlab' library (separate R package by different authors) for various SVM algorithms.   
</p>


<h3>Details</h3>

<p>The gkm-SVM provides implementation of a new SVM kernel method using gapped
k-mers as features for DNA or Protein sequences.
</p>
<p>There are three main functions in the gkmSVM package: 
</p>
<p>gkmsvm_kernel: computes the kernel matrix
</p>
<p>gkmsvm_train:  computes the SVM coefficients
</p>
<p>gkmsvm_classify: scores new sequences using the SVM model  
</p>
<p>Tutorial
</p>
<p>========
</p>
<p>We introduce the users to the basic workflow of our gkmSVM step-by-step.
Please refer to help messages for more detailed information of each function.
</p>
<p>1) making a kernel matrix
</p>
<p>First of all, we should calculate a full kernel matrix before training SVM
classifiers. In this tutorial, we are going to use test_positives.fa
as a positive set, and test_negatives.fa as a negative set. 
</p>
<p>#Input file names:  
</p>
<p>posfn= 'test_positives.fa'   #positive set (FASTA format)
</p>
<p>negfn= 'test_negatives.fa'   #negative set (FASTA format)
</p>
<p>testfn= 'test_testset.fa'    #test set (FASTA format)
</p>
<p>Alternatively if the negative set is not available, and positive set is provided as a bed file, genNullSeqs function could be used to generate the negative set and positive set sequences. 
</p>
<p>#Output file names:  
</p>
<p>kernelfn= 'test_kernel.txt' #kernel matrix
</p>
<p>svmfnprfx= 'test_svmtrain'  #SVM files 
</p>
<p>outfn =   'output.txt'      #output scores for sequences in the test set       
</p>
<p>gkmsvm_kernel(posfn, negfn, kernelfn);                #computes kernel 
</p>
<p>2) training SVM
</p>
<p>We can now train a SVM classifier using the kernel matrix generated above. For that we use gkmsvm_train function It takes four arguments; kernel file, positive sequences file, negative sequences file, and prefix of output file names for the svm model.
</p>
<p>gkmsvm_train(kernelfn, posfn, negfn, svmfnprfx);        #trains SVM
</p>
<p>It will generate two files, test_svmtrain_svalpha.out and
test_svmtrain_svseq.fa, which will then be used for classification/scoring
of test sequences as described below.
</p>
<p>3) classification using SVM
</p>
<p>gkmsvm_classify can be used to score any set of sequences. Here, we will 
score the test sequences which are given in test_testset.fa. Note that the same
set of parameters used in the gkmsvm_kernel should always be specified for
optimal classification (here we used default parameters).
</p>
<p>gkmsvm_classify(testfn, svmfnprfx, outfn);            #scores test sequences 
</p>
<p>In a more advanced example, we set the word length L=18, and the number of non-gapped positions K=7, and maximum number of mismatches maxnmm=4:
</p>
<p>gkmsvm_kernel(posfn, negfn, kernelfn, L=18, K=7, maxnmm=4);     #computes kernel 
</p>
<p>gkmsvm_train(kernelfn,posfn, negfn, svmfnprfx);                 #trains SVM
</p>
<p>gkmsvm_classify(testfn, svmfnprfx, outfn, L=18, K=7, maxnmm=4); #scores test sequences 
</p>
<p>In another example, we run a 5-fold cross validation to plot the ROC curves: 
</p>
<p>gkmsvm_kernel(posfn, negfn, kernelfn);     #computes kernel 
</p>
<p>cvres = gkmsvm_trainCV(kernelfn, posfn, negfn, svmfnprfx, outputPDFfn='ROC.pdf', outputCVpredfn='cvpred.out');       #trains SVM, plots ROC and PRC curves, and outputs model predictions.
</p>


<h3>Author(s)</h3>

<p>Mahmoud Ghandi
</p>
<p>Maintainer: Mike Beer &lt;mbeer@jhu.edu&gt;
</p>


<h3>References</h3>

<p>Ghandi M, Lee D, Mohammad-Noori M, Beer MA. 2014. Enhanced Regulatory Sequence Prediction Using Gapped k-mer Features. PLoS Comput Biol 10: e1003711. 
</p>
<p>Ghandi M, Mohammad-Noori M, Ghareghani N, Lee D, Garraway LA, and Beer MA. 2016. gkmSVM an R package for gapped-kmer SVM, Bioinformatics 32 (14), 2205-2207.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #Input file names:  
  posfn= 'test_positives.fa'   #positive set (FASTA format)
  negfn= 'test_negatives.fa'   #negative set (FASTA format)
  testfn= 'test_testset.fa'    #test set (FASTA format)
  
  #Output file names:  
  kernelfn= 'test_kernel.txt' #kernel matrix
  svmfnprfx= 'test_svmtrain'  #SVM files 
  outfn =   'output.txt'      #output scores for sequences in the test set       

#  gkmsvm_kernel(posfn, negfn, kernelfn);                #computes kernel 
#  gkmsvm_train(kernelfn, posfn, negfn, svmfnprfx);       #trains SVM
#  gkmsvm_classify(testfn, svmfnprfx, outfn);            #scores test sequences 

#  using L=18, K=7, maxnmm=4

#  gkmsvm_kernel(posfn, negfn, kernelfn, L=18, K=7, maxnmm=4);     #computes kernel 
#  gkmsvm_train(kernelfn, posfn, negfn, svmfnprfx);                 #trains SVM
#  gkmsvm_classify(testfn, svmfnprfx, outfn, L=18, K=7, maxnmm=4); #scores test sequences 

</code></pre>

<hr>
<h2 id='genNullSeqs'>
Generating GC/repeat matched randomly selected genomic sequences for the negative set
</h2><span id='topic+genNullSeqs'></span>

<h3>Description</h3>

<p>Generates null sequences (negative set) with matching repeat and GC content as the input bed file for positive set regions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genNullSeqs(
  inputBedFN, 
  genomeVersion='hg19', 
  outputBedFN = 'negSet.bed', 
  outputPosFastaFN = 'posSet.fa',
  outputNegFastaFN = 'negSet.fa', 
  xfold = 1, 
  repeat_match_tol = 0.02, 
  GC_match_tol = 0.02, 
  length_match_tol = 0.02, 
  batchsize = 5000, 
  nMaxTrials = 20, 
  genome = NULL)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genNullSeqs_+3A_inputbedfn">inputBedFN</code></td>
<td>

<p>positive set regions 
</p>
</td></tr>
<tr><td><code id="genNullSeqs_+3A_genomeversion">genomeVersion</code></td>
<td>

<p>genome version: 'hg19' and 'hg18' are supported. Default='hg19'. For other genomes, provide the BSgenome object using parameter 'genome'
</p>
</td></tr>
<tr><td><code id="genNullSeqs_+3A_outputbedfn">outputBedFN</code></td>
<td>

<p>output file name for the null sequences genomic regions. Default='negSet.bed'
</p>
</td></tr>
<tr><td><code id="genNullSeqs_+3A_outputposfastafn">outputPosFastaFN</code></td>
<td>

<p>output file name for the positive set sequences. Default='posSet.fa'
</p>
</td></tr>
<tr><td><code id="genNullSeqs_+3A_outputnegfastafn">outputNegFastaFN</code></td>
<td>

<p>output file name for the negative set sequences. Default='negSet.fa'
</p>
</td></tr>
<tr><td><code id="genNullSeqs_+3A_xfold">xfold</code></td>
<td>

<p>controls the desired number of sequences in the negative set. Default=1 (same number as in positive set)
</p>
</td></tr>
<tr><td><code id="genNullSeqs_+3A_repeat_match_tol">repeat_match_tol</code></td>
<td>

<p>tolerance for difference in repeat ratio. Default=0.02 (repeat content difference of 0.02 or less is acceptable)      </p>
</td></tr>
<tr><td><code id="genNullSeqs_+3A_gc_match_tol">GC_match_tol</code></td>
<td>

<p>tolerance for difference in GC content. Default=0.02
</p>
</td></tr>
<tr><td><code id="genNullSeqs_+3A_length_match_tol">length_match_tol</code></td>
<td>

<p>tolerance for difference in relative sequence length. Default=0.02
</p>
</td></tr>
<tr><td><code id="genNullSeqs_+3A_batchsize">batchsize</code></td>
<td>

<p>number of candidate random sequences tested in each trial. Default=5000
</p>
</td></tr>
<tr><td><code id="genNullSeqs_+3A_nmaxtrials">nMaxTrials</code></td>
<td>

<p>maximum number of trials. Default=20. 
</p>
</td></tr>
<tr><td><code id="genNullSeqs_+3A_genome">genome</code></td>
<td>

<p>BSgenome object. Default=NULL. If this parameter is used, parameter genomeVersion is ignored. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Writes the null sequences to files with the provided filenames. Outputs the filename for the output negative sequences file. 
</p>


<h3>Author(s)</h3>

<p>Mahmoud Ghandi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Example 1: 
# genNullSeqs('ctcfpos.bed' );

#Example 2:
# genNullSeqs('ctcfpos.bed', nMaxTrials=3, xfold=2, genomeVersion = 'hg18' );

#Example 3:
# genNullSeqs('ctcfpos.bed', xfold=2, genomeVersion = 'hg18', outputBedFN = 'ctcf_negSet.bed',
# outputPosFastaFN = 'ctcf_posSet.fa',outputNegFastaFN = 'ctcf_negSet.fa' );

#Example 4:
# Input file names:  
  
  posBedFN = 'test_positives.bed' # positive set genomic ranges (bed format)
  genomeVer = 'hg19' #genome version 
  testfn= 'test_testset.fa'    #test set (FASTA format)
  
# output file names:  
  posfn= 'test_positives.fa'   #positive set (FASTA format)
  negfn= 'test_negatives.fa'   #negative set (FASTA format)
  kernelfn= 'test_kernel.txt' #kernel matrix
  svmfnprfx= 'test_svmtrain'  #SVM files 
  outfn =   'output.txt'      #output scores for sequences in the test set       

#  genNullSeqs(posBedFN, genomeVersion = genomeVer, 
#    outputPosFastaFN = posfn, outputNegFastaFN = negfn );

#  gkmsvm_kernel(posfn, negfn, kernelfn);                #computes kernel 
#  gkmsvm_train(kernelfn, posfn, negfn, svmfnprfx);       #trains SVM
#  gkmsvm_classify(testfn, svmfnprfx, outfn);            #scores test sequences 

#  using L=18, K=7, maxnmm=4

#  gkmsvm_kernel(posfn, negfn, kernelfn, L=18, K=7, maxnmm=4);     #computes kernel 
#  gkmsvm_train(kernelfn, posfn, negfn, svmfnprfx);                 #trains SVM
#  gkmsvm_classify(testfn, svmfnprfx, outfn, L=18, K=7, maxnmm=4); #scores test sequences 


</code></pre>

<hr>
<h2 id='gkmsvm_classify'>Classifying(/scoring) new sequences using the gkmSVM model</h2><span id='topic+gkmsvm_classify'></span>

<h3>Description</h3>

<p>Given support vectors SVs and corresponding coefficients alphas and a set of 
sequences, calculates the SVM scores for the sequences.</p>


<h3>Usage</h3>

<pre><code class='language-R'>gkmsvm_classify(seqfile, svmfnprfx, outfile, L=10, K=6, maxnmm=3, 
maxseqlen=10000, maxnumseq=1000000, useTgkm=1, alg=0, addRC=TRUE, usePseudocnt=FALSE, 
batchSize=100000, wildcardLambda=1.0, wildcardMismatchM=2, alphabetFN="NULL", 
svseqfile=NA, alphafile=NA)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gkmsvm_classify_+3A_seqfile">seqfile</code></td>
<td>
<p>input sequences file name (FASTA format)</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_svmfnprfx">svmfnprfx</code></td>
<td>
<p>SVM model file name prefix</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_outfile">outfile</code></td>
<td>
<p>output file name</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_l">L</code></td>
<td>
<p>word length, default=10</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_k">K</code></td>
<td>
<p>number of informative columns, default=6</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_maxnmm">maxnmm</code></td>
<td>
<p>maximum number of mismatches to consider, default=3</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_maxseqlen">maxseqlen</code></td>
<td>
<p>maximum sequence length in the sequence files, default=10000</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_maxnumseq">maxnumseq</code></td>
<td>
<p>maximum number of sequences in the sequence files, default=1000000</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_usetgkm">useTgkm</code></td>
<td>
<p>filter type: 0(use full filter), 1(use truncated filter: this gaurantees non-negative counts for all L-mers), 2(use h[m], gkm count vector), 3(wildcard), 4(mismatch), default=1</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_alg">alg</code></td>
<td>
<p>algorithm type: 0(auto), 1(XOR Hashtable), 2(tree), default=0</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_addrc">addRC</code></td>
<td>
<p>adds reverse complement sequences, default=TRUE</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_usepseudocnt">usePseudocnt</code></td>
<td>
<p>adds a constant to count estimates, default=FALSE</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_batchsize">batchSize</code></td>
<td>
<p>number of sequences to compute scores for in batch, default=100000</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_wildcardlambda">wildcardLambda</code></td>
<td>
<p>lambda for wildcard kernel, defaul=0.9</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_wildcardmismatchm">wildcardMismatchM</code></td>
<td>
<p>max mismatch for Mismatch kernel or wildcard kernel, default=2</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_alphabetfn">alphabetFN</code></td>
<td>
<p>alphabets file name, if not specified, it is assumed the inputs are DNA sequences</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_svseqfile">svseqfile</code></td>
<td>
<p>SVM support vectors sequence file name (not needed if svmfnprfx is provided)</p>
</td></tr>
<tr><td><code id="gkmsvm_classify_+3A_alphafile">alphafile</code></td>
<td>
<p>SVM support vectors weights file name (not needed if svmfnprfx is provided)</p>
</td></tr>
</table>


<h3>Details</h3>

<p> classification using SVM:
gkmsvm_classify can be used to score any set of sequences. Note that the same
set of parameters (L, K, maxnmm) used in the gkmsvm_kernel should be specified for
optimal classification.
</p>
<p>gkmsvm_classify(testfn, svmfnprfx, outfn);            #scores test sequences  </p>


<h3>Author(s)</h3>

<p>Mahmoud Ghandi</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #Input file names:  
  posfn= 'test_positives.fa'   #positive set (FASTA format)
  negfn= 'test_negatives.fa'   #negative set (FASTA format)
  testfn= 'test_testset.fa'    #test set (FASTA format)
  
  #Output file names:  
  kernelfn= 'test_kernel.txt' #kernel matrix
  svmfnprfx= 'test_svmtrain'  #SVM files 
  outfn =   'output.txt'      #output scores for sequences in the test set       

#  gkmsvm_kernel(posfn, negfn, kernelfn);                #computes kernel 
#  gkmsvm_train(kernelfn,posfn, negfn, svmfnprfx);       #trains SVM
#  gkmsvm_classify(testfn, svmfnprfx, outfn);            #scores test sequences 
</code></pre>

<hr>
<h2 id='gkmsvm_delta'>
Calculating deltaSVM scores 
</h2><span id='topic+gkmsvm_delta'></span>

<h3>Description</h3>

<p>Given support vectors SVs and corresponding coefficients alphas and a pair of file test sequence files (one for reference allele, and one for alternate allele), calculates the deltaSVM scores for the sequences.</p>


<h3>Usage</h3>

<pre><code class='language-R'>gkmsvm_delta(seqfile_allele1, seqfile_allele2, svmfnprfx, outfile, L = 10, 
K = 6, maxnmm = 3, maxseqlen = 10000, maxnumseq = 1e+06, useTgkm = 1, alg = 2, 
addRC = TRUE, usePseudocnt = FALSE, batchSize = 1e+05, wildcardLambda = 1, 
wildcardMismatchM = 2, alphabetFN = "NULL", svseqfile = NA,alphafile = NA, 
outfile_allele1 = NA, outfile_allele2 = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gkmsvm_delta_+3A_seqfile_allele1">seqfile_allele1</code></td>
<td>

<p>fasta file containing the test sequences (reference allele)</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_seqfile_allele2">seqfile_allele2</code></td>
<td>

<p>fasta file containing the test sequences (alternate allele). The sequences in this file should be in the exact same order as in seqfile_allele1.</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_svmfnprfx">svmfnprfx</code></td>
<td>
<p>SVM model file name prefix</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_outfile">outfile</code></td>
<td>
<p>output file name</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_l">L</code></td>
<td>
<p>word length, default=10</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_k">K</code></td>
<td>
<p>number of informative columns, default=6</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_maxnmm">maxnmm</code></td>
<td>
<p>maximum number of mismatches to consider, default=3</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_maxseqlen">maxseqlen</code></td>
<td>
<p>maximum sequence length in the sequence files, default=10000</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_maxnumseq">maxnumseq</code></td>
<td>
<p>maximum number of sequences in the sequence files, default=1000000</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_usetgkm">useTgkm</code></td>
<td>
<p>filter type: 0(use full filter), 1(use truncated filter: this gaurantees non-negative counts for all L-mers), 2(use h[m], gkm count vector), 3(wildcard), 4(mismatch), default=1</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_alg">alg</code></td>
<td>
<p>algorithm type: 0(auto), 1(XOR Hashtable), 2(tree), default=0</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_addrc">addRC</code></td>
<td>
<p>adds reverse complement sequences, default=TRUE</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_usepseudocnt">usePseudocnt</code></td>
<td>
<p>adds a constant to count estimates, default=FALSE</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_batchsize">batchSize</code></td>
<td>
<p>number of sequences to compute scores for in batch, default=100000</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_wildcardlambda">wildcardLambda</code></td>
<td>
<p>lambda for wildcard kernel, defaul=0.9</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_wildcardmismatchm">wildcardMismatchM</code></td>
<td>
<p>max mismatch for Mismatch kernel or wildcard kernel, default=2</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_alphabetfn">alphabetFN</code></td>
<td>
<p>alphabets file name, if not specified, it is assumed the inputs are DNA sequences</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_svseqfile">svseqfile</code></td>
<td>
<p>SVM support vectors sequence file name (not needed if svmfnprfx is provided)</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_alphafile">alphafile</code></td>
<td>
<p>SVM support vectors weights file name (not needed if svmfnprfx is provided)</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_outfile_allele1">outfile_allele1</code></td>
<td>

<p>output filename for gkmSVM scores for the reference sequences (optional)
</p>
</td></tr>
<tr><td><code id="gkmsvm_delta_+3A_outfile_allele2">outfile_allele2</code></td>
<td>

<p>output filename for gkmSVM scores for the alternate sequences (optional)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>predicting the effect of variants using gkmSVM model:
gkmsvm_delta can be used to predict the effect of sequence variants. The sequences corresponding to reference allele and alternate alleles are given in two separate files. gkmSVM is used to score each set of sequences, and the difference in the gkmSVM score for the reference and alternate allele is reported. 
Note that the same set of parameters (L, K, maxnmm) used in the gkmsvm_kernel should be specified for
optimal scoring.
</p>
<p>gkmsvm_kernel(seqfile_allele1, seqfile_allele2, svmfnprfx, outfn);            #scores test sequences </p>


<h3>Value</h3>

<p>deltaSVM scores
</p>


<h3>Author(s)</h3>

<p>Mahmoud Ghandi
</p>


<h3>References</h3>

<p>Ghandi M, Mohammad-Noori M, Ghareghani N, Lee D, Garraway LA, and Beer MA. gkmSVM: an R package for gapped-kmer SVM, Bioinformatics 2016.
</p>
<p>Ghandi M, Lee D, Mohammad-Noori M, Beer MA. 2014. Enhanced Regulatory Sequence Prediction Using Gapped k-mer Features. PLoS Comput Biol 10: e1003711. 
</p>
<p>Lee D, Gorkin DU, Baker M, Strober BJ, Asoni AL, McCallion AS, and Beer MA. A method to predict the impact of regulatory variants from DNA sequence. Nature Genetics 2015. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  #Input file names:  
  posfn= 'test_positives.fa'   #positive set (FASTA format)
  negfn= 'test_negatives.fa'   #negative set (FASTA format)
  testfn_ref= 'test_testsetRef.fa'    #test set (reference allele) (FASTA format)
  testfn_alt= 'test_testsetAlt.fa'    #test set (alternate allele) (FASTA format)
  
  #Output file names:  
  kernelfn= 'test_kernel.txt' #kernel matrix
  svmfnprfx= 'test_svmtrain'  #SVM files 
  outfn =   'output.txt'      #output delta svm scores for sequences in the test set       

#  gkmsvm_kernel(posfn, negfn, kernelfn);                #computes kernel 
#  gkmsvm_train(kernelfn,posfn, negfn, svmfnprfx);       #trains SVM
#  gkmsvm_delta(testfn_ref,testfn_alt, svmfnprfx, outfn);            #scores test sequences 

</code></pre>

<hr>
<h2 id='gkmsvm_kernel'>Computing the kernel matrix</h2><span id='topic+gkmsvm_kernel'></span>

<h3>Description</h3>

<p>Generates a lower triangle of kernel matrix (i.e. pairwise similarities)
between the sequences.</p>


<h3>Usage</h3>

<pre><code class='language-R'>gkmsvm_kernel(posfile, negfile, outfile, L=10, K=6, maxnmm=3, maxseqlen=10000, 
maxnumseq=1000000, useTgkm=1, alg=0, addRC=TRUE, usePseudocnt=FALSE, wildcardLambda=1.0,
wildcardMismatchM=2, alphabetFN="NULL")</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gkmsvm_kernel_+3A_posfile">posfile</code></td>
<td>
<p>positive sequences file name (FASTA format)</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_negfile">negfile</code></td>
<td>
<p>negative sequences file name (FASTA format)</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_outfile">outfile</code></td>
<td>
<p>output file name</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_l">L</code></td>
<td>
<p>word length, default=10</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_k">K</code></td>
<td>
<p>number of informative columns, default=6</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_maxnmm">maxnmm</code></td>
<td>
<p>maximum number of mismatches to consider, default=3</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_maxseqlen">maxseqlen</code></td>
<td>
<p>maximum sequence length in the sequence files, default=10000</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_maxnumseq">maxnumseq</code></td>
<td>
<p>maximum number of sequences in the sequence files, default=1000000</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_usetgkm">useTgkm</code></td>
<td>
<p>filter type: 0(use full filter), 1(use truncated filter: this gaurantees non-negative counts for all L-mers), 2(use h[m], gkm count vector), 3(wildcard), 4(mismatch), default=1</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_alg">alg</code></td>
<td>
<p>algorithm type: 0(auto), 1(XOR Hashtable), 2(tree), default=0</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_addrc">addRC</code></td>
<td>
<p>adds reverse complement sequences, default=TRUE</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_usepseudocnt">usePseudocnt</code></td>
<td>
<p>adds a constant to count estimates, default=FALSE</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_wildcardlambda">wildcardLambda</code></td>
<td>
<p>lambda for wildcard kernel, defaul=0.9</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_wildcardmismatchm">wildcardMismatchM</code></td>
<td>
<p>max mismatch for Mismatch kernel or wildcard kernel, default=2</p>
</td></tr>
<tr><td><code id="gkmsvm_kernel_+3A_alphabetfn">alphabetFN</code></td>
<td>
<p>alphabets file name, if not specified, it is assumed the inputs are DNA sequences</p>
</td></tr>
</table>


<h3>Details</h3>

<p>  It calculates the full kernel matrix that can be then used to train an SVM classifier. 
gkmsvm_kernel(posfn, negfn, kernelfn);
</p>


<h3>Author(s)</h3>

<p>Mahmoud Ghandi</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #Input file names:  
  posfn= 'test_positives.fa'   #positive set (FASTA format)
  negfn= 'test_negatives.fa'   #negative set (FASTA format)
  testfn= 'test_testset.fa'    #test set (FASTA format)
  
  #Output file names:  
  kernelfn= 'test_kernel.txt' #kernel matrix
  svmfnprfx= 'test_svmtrain'  #SVM files 
  outfn =   'output.txt'      #output scores for sequences in the test set       

#  gkmsvm_kernel(posfn, negfn, kernelfn);                #computes kernel 
#  gkmsvm_train(kernelfn,posfn, negfn, svmfnprfx);       #trains SVM
#  gkmsvm_classify(testfn, svmfnprfx, outfn);            #scores test sequences 
</code></pre>

<hr>
<h2 id='gkmsvm_train'>Training the SVM model</h2><span id='topic+gkmsvm_train'></span>

<h3>Description</h3>

<p>Using the kernel matrix created by 'gkmsvm_kernel', this function trains the SVM classifier. Here we rely on the 'kernlab' package, and merely provide a wrapper function.</p>


<h3>Usage</h3>

<pre><code class='language-R'>gkmsvm_train(kernelfn, posfn, negfn, svmfnprfx,  Type="C-svc", C=1, shrinking=FALSE, ...)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gkmsvm_train_+3A_kernelfn">kernelfn</code></td>
<td>
<p>kernel matrix file name</p>
</td></tr>
<tr><td><code id="gkmsvm_train_+3A_posfn">posfn</code></td>
<td>
<p>positive sequences file name</p>
</td></tr>
<tr><td><code id="gkmsvm_train_+3A_negfn">negfn</code></td>
<td>
<p>negative sequences file name</p>
</td></tr>
<tr><td><code id="gkmsvm_train_+3A_svmfnprfx">svmfnprfx</code></td>
<td>
<p>output SVM model file name prefix</p>
</td></tr>
<tr><td><code id="gkmsvm_train_+3A_type">Type</code></td>
<td>
<p>optional: SVM type (default='C-svc'), see 'kernlab' documentation for more details.</p>
</td></tr>
<tr><td><code id="gkmsvm_train_+3A_c">C</code></td>
<td>
<p>optional: SVM parameter C (default=1), see 'kernlab' documentation for more details.</p>
</td></tr>
<tr><td><code id="gkmsvm_train_+3A_shrinking">shrinking</code></td>
<td>
<p>optional: shrinking parameter for kernlab (default=FALSE), see 'kernlab' documentation for more details.</p>
</td></tr>
<tr><td><code id="gkmsvm_train_+3A_...">...</code></td>
<td>
<p>optional: additional SVM parameters, see 'kernlab' documentation for more details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Trains SVM classifier and generates two files: [svmfnprfx]_svalpha.out for SVM alphas and the other for the corresponding SV sequences ([svmfnprfx]_svseq.fa) </p>


<h3>Author(s)</h3>

<p>Mahmoud Ghandi</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #Input file names:  
  posfn= 'test_positives.fa'   #positive set (FASTA format)
  negfn= 'test_negatives.fa'   #negative set (FASTA format)
  testfn= 'test_testset.fa'    #test set (FASTA format)
  
  #Output file names:  
  kernelfn= 'test_kernel.txt' #kernel matrix
  svmfnprfx= 'test_svmtrain'  #SVM files 
  outfn =   'output.txt'      #output scores for sequences in the test set       

#  gkmsvm_kernel(posfn, negfn, kernelfn);                #computes kernel 
#  gkmsvm_train(kernelfn,posfn, negfn, svmfnprfx);       #trains SVM
#  gkmsvm_classify(testfn, svmfnprfx, outfn);            #scores test sequences 
</code></pre>

<hr>
<h2 id='gkmsvm_trainCV'>Training the SVM model, using repeated CV to tune parameter C and plot ROC curves</h2><span id='topic+gkmsvm_trainCV'></span>

<h3>Description</h3>

<p>Using the kernel matrix created by 'gkmsvm_kernel', this function trains the SVM classifier. 
It uses repeated CV to find optimum SVM parameter C. Also generates ROC and PRC curves.</p>


<h3>Usage</h3>

<pre><code class='language-R'>gkmsvm_trainCV(kernelfn, posfn, negfn, svmfnprfx=NA, 
  nCV=5, nrepeat=1, cv=NA, Type="C-svc", C=1, shrinking=FALSE, 
  showPlots=TRUE, outputPDFfn=NA,  outputCVpredfn=NA, outputROCfn=NA, ...)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gkmsvm_trainCV_+3A_kernelfn">kernelfn</code></td>
<td>
<p>kernel matrix file name</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_posfn">posfn</code></td>
<td>
<p>positive sequences file name</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_negfn">negfn</code></td>
<td>
<p>negative sequences file name</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_svmfnprfx">svmfnprfx</code></td>
<td>
<p>(optional) output SVM model file name prefix </p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_ncv">nCV</code></td>
<td>
<p>(optional) number of CV folds</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_nrepeat">nrepeat</code></td>
<td>
<p>(optional) number of repeated CVs</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_cv">cv</code></td>
<td>
<p>(optional) CV group label. An array of length (npos+nneg), containing CV group number (between 1 an nCV) for each sequence</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_type">Type</code></td>
<td>
<p>(optional) SVM type (default='C-svc'), see 'kernlab' documentation for more details.</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_c">C</code></td>
<td>
<p>(optional)a vector of all values of C (SVM parameter) to be tested. (default=1), see 'kernlab' documentation for more details.</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_shrinking">shrinking</code></td>
<td>
<p>optional: shrinking parameter for kernlab (default=FALSE), see 'kernlab' documentation for more details.</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_showplots">showPlots</code></td>
<td>
<p>generate plots (default==TRUE)</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_outputpdffn">outputPDFfn</code></td>
<td>
<p>filename for output PDF, default=NA (no PDF output)</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_outputcvpredfn">outputCVpredfn</code></td>
<td>
<p>filename for output cvpred (predicted CV values), default=NA (no output)</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_outputrocfn">outputROCfn</code></td>
<td>
<p>filename for output auROC (Area Under an ROC Curve) and auPRC (Area Under the Precision Recall Curve) values, default=NA (no output)</p>
</td></tr>
<tr><td><code id="gkmsvm_trainCV_+3A_...">...</code></td>
<td>
<p>optional: additional SVM parameters, see 'kernlab' documentation for more details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Trains SVM classifier and generates two files: [svmfnprfx]_svalpha.out for SVM alphas and the other for the corresponding SV sequences ([svmfnprfx]_svseq.fa) </p>


<h3>Author(s)</h3>

<p>Mahmoud Ghandi</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #Input file names:  
  posfn= 'test_positives.fa'   #positive set (FASTA format)
  negfn= 'test_negatives.fa'   #negative set (FASTA format)
  testfn= 'test_testset.fa'    #test set (FASTA format)
  
  #Output file names:  
  kernelfn= 'test_kernel.txt' #kernel matrix
  svmfnprfx= 'test_svmtrain'  #SVM files 
  outfn =   'output.txt'      #output scores for sequences in the test set       

#  gkmsvm_kernel(posfn, negfn, kernelfn);                #computes kernel 
#  cvres = gkmsvm_trainCV(kernelfn,posfn, negfn, svmfnprfx, 
#      outputPDFfn='ROC.pdf', outputCVpredfn='cvpred.out');    
#      #trains SVM, plots ROC and PRC curves, and outputs model predictions.
#  gkmsvm_classify(testfn, svmfnprfx, outfn);            #scores test sequences 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
