<!DOCTYPE html><html><head><title>Help for package GLIDE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GLIDE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#checkdata'>
<p>A function used to check the core inputs of function glide</p></a></li>
<li><a href='#glide'>
<p>Evaluate individual and global pleiotropy among genotypes being used as instrumental variables</p>
in Mendelian randomization studies.</a></li>
<li><a href='#plot.egger'>
<p>A function used to draw Egger plot</p></a></li>
<li><a href='#plot.glide'>
<p>A function used to draw the q-q plot</p></a></li>
<li><a href='#remove_missingdata'>
<p>A function used to remove missing data</p></a></li>
<li><a href='#simdata'><p>An example dataset to demonstrate the usage of GLIDE</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Global and Individual Tests for Direct Effects</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-20</td>
</tr>
<tr>
<td>Author:</td>
<td>James Dai [aut, cre],
  Xiaoyu Wang [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>James Dai &lt;jdai@fredhutch.org&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Global and individual tests for pleiotropy and direct effects in Mendelian randomization studies. Refer to J. Y. Dai, U. Peters, X. Wang, J. Kocarnik et al. AJE (2018) &lt;<a href="https://doi.org/10.1093%2Faje%2Fkwy177">doi:10.1093/aje/kwy177</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS,foreach,parallel,doParallel</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>no</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-25 21:53:18 UTC; xwang234</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-25 23:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='checkdata'>
A function used to check the core inputs of function glide
</h2><span id='topic+checkdata'></span>

<h3>Description</h3>

<p>It is used to check the core inputs of function glide
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkdata(formula,exposure_coeff,genotype_columns,data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkdata_+3A_formula">formula</code></td>
<td>
<p>A an object of class &quot;formula&quot;: outcome ~ adjusting covariates.</p>
</td></tr>
<tr><td><code id="checkdata_+3A_exposure_coeff">exposure_coeff</code></td>
<td>

<p>A named mumeric vector or a one-column dataframe for the external coefficients 
for genotype-exposure associations.</p>
</td></tr>
<tr><td><code id="checkdata_+3A_genotype_columns">genotype_columns</code></td>
<td>
<p>An integer vector, column indices of data in which genotype data are stored.</p>
</td></tr>
<tr><td><code id="checkdata_+3A_data">data</code></td>
<td>
<p>A data frame composed of formula-related components and snp genotype columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, would generate error messages if input data are not right.</p>


<h3>Author(s)</h3>

<p>James Y. Dai and X. Wang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load an example data; it includes a dataframe "simdat" and a one-column dataframe  "coeff"
data(simdata)
simdat=simdata$simdat
coeff=simdata$coeff
formula=as.formula("outcome~age+sex+pc1+pc2+pc3")
genotype_columns=which(grepl("^SNP",colnames(simdat)))
checkdata(formula=formula,exposure_coeff=coeff,genotype_columns,data=simdat)
</code></pre>

<hr>
<h2 id='glide'>
Evaluate individual and global pleiotropy among genotypes being used as instrumental variables
in Mendelian randomization studies.
</h2><span id='topic+glide'></span>

<h3>Description</h3>

<p>Glide is a function to perform global and individual tests for assessing direct effects in
Mendelian randomization studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glide(formula,exposure_coeff=NULL,genotype_columns=NULL,data,
np=100000,qcutoff=0.2,parallel=TRUE,corenumber=1,verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glide_+3A_formula">formula</code></td>
<td>

<p>An object of class &quot;formula&quot;: outcome ~ adjusting covariates, specifying the logistic regression
of a binary study outcome with respect to covariates being adjusted for in addition to genotype
data.
</p>
</td></tr>
<tr><td><code id="glide_+3A_exposure_coeff">exposure_coeff</code></td>
<td>

<p>A named numeric vector or a one-column dataframe for the external regression coefficients
for genotype-exposure associations. These coefficients are typically derived from previous
genome-wide association studies.
All the names or rownames must be found in &quot;genotype_columns&quot; (specified below) columns of the
input data.
</p>
</td></tr>
<tr><td><code id="glide_+3A_genotype_columns">genotype_columns</code></td>
<td>

<p>An integer vector that  contains column indices of data columns in which genotype data are stored.  
</p>
</td></tr>
<tr><td><code id="glide_+3A_data">data</code></td>
<td>

<p>A data frame composed of formula-related variables and SNP genotype columns.
</p>
</td></tr>
<tr><td><code id="glide_+3A_np">np</code></td>
<td>

<p>An integer for the number of observations in the parametric simulation being generated 
to compute expected p-values and q-values.
</p>
</td></tr>
<tr><td><code id="glide_+3A_qcutoff">qcutoff</code></td>
<td>

<p>The q-value cutoff to declare significance of the direct effects of individual variants. 
The default value is set to be 0.2.
</p>
</td></tr>
<tr><td><code id="glide_+3A_parallel">parallel</code></td>
<td>

<p>If TRUE, use parallel foreach to speed up computation.
</p>
</td></tr>
<tr><td><code id="glide_+3A_corenumber">corenumber</code></td>
<td>

<p>An integer to set the number of computer cores used in parallel computation. If the requested
number of cores is greater than number of cores available, GLIDE will user the latter number.
</p>
</td></tr>
<tr><td><code id="glide_+3A_verbose">verbose</code></td>
<td>

<p>If TRUE, print on screen the computing time spent in each step of GLIDE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a logistic regression for each SNP being evaluated for instrumental variables, 
regressing outcome on one genotype at a time and adjusting covariates as specified in the
formula (outcome ~ adjusting covariates). The p-values for the genotypes derived from these
regression models were compared to the distribution of simulation-based null p-values to
declare significance of individual variant and global pleiotropy. See Dai et al. (2017) for
details of the algorithm.
</p>


<h3>Value</h3>

<p>A data frame containing observed p-value, expected null p-value, family wise error rate (FWER),
q-value, genetic association with exposure, and variance of the estimated association of 
each SNP.
</p>


<h3>Author(s)</h3>

<p>James Y. Dai and X. Wang
</p>


<h3>References</h3>

<p>J. Y. Dai, U. Peters, X. Wang, J. Kocarnik, J. et al. Diagnostics of pleiotropy in Mendelian
randomization studies: Global and individual tests for direct effects. Am J Epidemiol 
187(12):2672-80, 2018.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load an example data.
data(simdata)
#The example dataset is a list composed of two dataframes. 
#simat stores 20,000 observations of 81 variables, including outcome, 5 ajusting covariates, 
#and 75 SNPs.
simdat=simdata$simdat
#coeff stores the 75 external regression cofficients.
coeff=simdata$coeff
#We define the regression formula for outcome and adjusting covariates
formula=as.formula("outcome~age+sex+pc1+pc2+pc3")
#We next define the columns in simdat that contain genotype data
genotype_columns=which(grepl("^SNP",colnames(simdat)))


##run glide
out=glide(formula=formula,exposure_coeff=coeff,genotype_columns,data=simdat,np=100000,
qcutoff=0.2,parallel=TRUE,corenumber=1,verbose=TRUE)

##use more cores
out=glide(formula=formula,exposure_coeff=coeff,genotype_columns,data=simdat,np=100000,
qcutoff=0.2,parallel=TRUE,corenumber=2,verbose=TRUE)

</code></pre>

<hr>
<h2 id='plot.egger'>
A function used to draw Egger plot
</h2><span id='topic+plot.egger'></span>

<h3>Description</h3>

<p>It is used to draw MR-Egger regression plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'egger'
plot(x,exposure_coeff,qcutoff=0.2,xlab="Genetic association with the exposure",
                       ylab="Genetic association with the outcome",...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.egger_+3A_x">x</code></td>
<td>

<p>x is a dataframe, it is the outcome of the glide function.
</p>
</td></tr>
<tr><td><code id="plot.egger_+3A_exposure_coeff">exposure_coeff</code></td>
<td>

<p>A named numeric vector or a one-column dataframe for the external regression coefficients
for genotype-exposure associations. These coefficients are typically derived from previous
genome-wide association studies.
All the names or rownames must be found in &quot;genotype_columns&quot; (specified below) columns of the
input data.
</p>
</td></tr>
<tr><td><code id="plot.egger_+3A_qcutoff">qcutoff</code></td>
<td>

<p>qcutoff is the q-value cutoff.
</p>
</td></tr>
<tr><td><code id="plot.egger_+3A_xlab">xlab</code></td>
<td>

<p>xlab is the default x label.
</p>
</td></tr>
<tr><td><code id="plot.egger_+3A_ylab">ylab</code></td>
<td>

<p>ylab is the default y label.
</p>
</td></tr>
<tr><td><code id="plot.egger_+3A_...">...</code></td>
<td>

<p>the rest of plot arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, the Egger plot will be generated.</p>


<h3>Author(s)</h3>

<p>James Y. Dai and X. Wang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##first generate output using the glide function
#load an example data
data(simdata)
simdat=simdata$simdat
coeff=simdata$coeff

formula=as.formula("outcome~age+sex+pc1+pc2+pc3")
genotype_columns=which(grepl("^SNP",colnames(simdat)))

#run glide
out=glide(formula=formula,exposure_coeff=coeff,genotype_columns,data=simdat,np=100000,
qcutoff=0.2,parallel=TRUE,corenumber=1,verbose=TRUE)

##draw the Egger plot
plot.egger(out,exposure_coeff=coeff)

</code></pre>

<hr>
<h2 id='plot.glide'>
A function used to draw the q-q plot 
</h2><span id='topic+plot.glide'></span>

<h3>Description</h3>

<p>It is used to draw the q-q plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'glide'
plot(x,qcutoff=0.2,xlab="Expected null p-values (log base 10)",
                       ylab="Observed p-values (log base 10)",...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.glide_+3A_x">x</code></td>
<td>

<p>x is a dataframe, it is the outcome of the glide function.
</p>
</td></tr>
<tr><td><code id="plot.glide_+3A_qcutoff">qcutoff</code></td>
<td>

<p>qcutoff is the q-value cutoff.
</p>
</td></tr>
<tr><td><code id="plot.glide_+3A_xlab">xlab</code></td>
<td>

<p>xlab is the default x label.
</p>
</td></tr>
<tr><td><code id="plot.glide_+3A_ylab">ylab</code></td>
<td>

<p>ylab is the default y label.
</p>
</td></tr>
<tr><td><code id="plot.glide_+3A_...">...</code></td>
<td>

<p>the rest plot arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, q-q plot will be generated.</p>


<h3>Author(s)</h3>

<p>James Y. Dai and X. Wang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##first generate output using the glide function
#load an example data
data(simdata)
simdat=simdata$simdat
coeff=simdata$coeff

formula=as.formula("outcome~age+sex+pc1+pc2+pc3")
genotype_columns=which(grepl("^SNP",colnames(simdat)))

#run glide
out=glide(formula=formula,exposure_coeff=coeff,genotype_columns,data=simdat,np=100000,
qcutoff=0.2,parallel=TRUE,corenumber=1,verbose=TRUE)

##draw the plot
plot.glide(out)

</code></pre>

<hr>
<h2 id='remove_missingdata'>
A function used to remove missing data
</h2><span id='topic+remove_missingdata'></span>

<h3>Description</h3>

<p>It is used to remove samples which have NA/missing data in covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_missingdata(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_missingdata_+3A_data">data</code></td>
<td>

<p>data is a dataframe.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>row index of the input data which don't have missing values. 
</p>

<hr>
<h2 id='simdata'>An example dataset to demonstrate the usage of GLIDE
</h2><span id='topic+simdata'></span>

<h3>Description</h3>

<p>An example simulation dataset used to demonstrate the usage of GLIDE. 
It is a list composed of 
a dataframe &quot;simdat&quot; and a dataframe &quot;coeff&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("simdata")</code></pre>


<h3>Format</h3>

<p>coeff is a one-column dataframe. It consists of exposure coefficients of genotypes of 75 SNPs.
simdat is a data frame consisting of 20000 observations. It has 81 columns, with 75 columns for 
genotypes of 75 SNPs and 6 columns of following covariates:
</p>

<dl>
<dt><code>age</code></dt><dd><p>an integer vector to indicate age</p>
</dd>
<dt><code>sex</code></dt><dd><p>a factor vector to indicate gender</p>
</dd>
<dt><code>pc1</code></dt><dd><p>a numeric vector of the first principal component of genotypes</p>
</dd>
<dt><code>pc2</code></dt><dd><p>a numeric vector of the second principal component of genotypes</p>
</dd>
<dt><code>pc3</code></dt><dd><p>a numeric vector of the third principal component of genotypes</p>
</dd>
<dt><code>outcome</code></dt><dd><p>a factor vector to indicate the outcome, 1/0</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>  data("simdata")
  ls()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
