<!DOCTYPE html><html><head><title>Help for package glmmSeq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {glmmSeq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fcPlot'><p>Plotly or ggplot fold change plots</p></a></li>
<li><a href='#ggmodelPlot'><p>Mixed model effects plot using ggplot2</p></a></li>
<li><a href='#glmmQvals'><p>Glmm Sequencing qvalues</p></a></li>
<li><a href='#glmmRefit'><p>Refit mixed effects model</p></a></li>
<li><a href='#glmmSeq'><p>GLMM with negative binomial distribution for sequencing count data</p></a></li>
<li><a href='#GlmmSeq-class'><p>An S4 class to define the glmmSeq output</p></a></li>
<li><a href='#lmmSeq'><p>Linear mixed models for data matrix</p></a></li>
<li><a href='#lmmSeq-class'><p>An S4 class to define the lmmSeq output</p></a></li>
<li><a href='#maPlot'><p>MA plots</p></a></li>
<li><a href='#metadata'><p>Minimal metadata from PEAC</p></a></li>
<li><a href='#modelPlot'><p>Mixed model effects plot</p></a></li>
<li><a href='#summary.lmmSeq'><p>Summarise a 'glmmSeq'/'lmmSeq' object</p></a></li>
<li><a href='#tpm'><p>TPM count data from PEAC</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>General Linear Mixed Models for Gene-Level Differential
Expression</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.5</td>
</tr>
<tr>
<td>Description:</td>
<td>Using mixed effects models to analyse longitudinal gene expression can highlight differences between sample groups over time. The most widely used differential gene expression tools are unable to fit linear mixed effect models, and are less optimal for analysing longitudinal data. This package provides negative binomial and Gaussian mixed effects models to fit gene expression and other biological data across repeated samples. This is particularly useful for investigating changes in RNA-Sequencing gene expression between groups of individuals over time, as described in: Rivellese, F., Surace, A. E., Goldmann, K., Sciacca, E., Cubuk, C., Giorli, G., ... Lewis, M. J., &amp; Pitzalis, C. (2022) Nature medicine &lt;<a href="https://doi.org/10.1038%2Fs41591-022-01789-0">doi:10.1038/s41591-022-01789-0</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Language:</td>
<td>en-gb</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://myles-lewis.github.io/glmmSeq/">https://myles-lewis.github.io/glmmSeq/</a>,
<a href="https://github.com/myles-lewis/glmmSeq">https://github.com/myles-lewis/glmmSeq</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/myles-lewis/glmmSeq/issues">https://github.com/myles-lewis/glmmSeq/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, kableExtra, DESeq2, edgeR, emmeans</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, car, stats, ggplot2, ggpubr, glmmTMB, graphics, lme4,
lmerTest, methods, plotly, qvalue, pbapply, pbmcapply</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-08 13:31:14 UTC; myles</td>
</tr>
<tr>
<td>Author:</td>
<td>Myles Lewis <a href="https://orcid.org/0000-0001-9365-5345"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Katriona Goldmann <a href="https://orcid.org/0000-0002-9073-6323"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Elisabetta Sciacca
    <a href="https://orcid.org/0000-0001-7525-1558"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Cankut Cubuk <a href="https://orcid.org/0000-0003-4646-0849"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Anna Surace <a href="https://orcid.org/0000-0001-9589-3005"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Myles Lewis &lt;myles.lewis@qmul.ac.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-08 13:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='fcPlot'>Plotly or ggplot fold change plots</h2><span id='topic+fcPlot'></span>

<h3>Description</h3>

<p>Plotly or ggplot fold change plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fcPlot(
  object,
  x1var,
  x2var,
  x1Values = NULL,
  x2Values = NULL,
  pCutoff = 0.01,
  labels = c(),
  useAdjusted = FALSE,
  plotCutoff = 1,
  graphics = "ggplot",
  fontSize = 12,
  labelFontSize = 4,
  colours = c("grey", "goldenrod1", "red", "blue"),
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fcPlot_+3A_object">object</code></td>
<td>
<p>A glmmSeq object created by
<code><a href="#topic+glmmSeq">glmmSeq::glmmSeq()</a></code>.</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_x1var">x1var</code></td>
<td>
<p>The name of the first (inner) x parameter</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_x2var">x2var</code></td>
<td>
<p>The name of the second (outer) x parameter</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_x1values">x1Values</code></td>
<td>
<p>Timepoints or categories in <code>x1var</code> used to calculate fold
change. If <code>NULL</code> the first two levels in <code>x1var</code> are used.</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_x2values">x2Values</code></td>
<td>
<p>Categories in <code>x2var</code> to be compared on x and y axis.</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_pcutoff">pCutoff</code></td>
<td>
<p>The significance cut-off for colour-coding
(default = 0.01)</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_labels">labels</code></td>
<td>
<p>Row names or indices to label on plot</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_useadjusted">useAdjusted</code></td>
<td>
<p>whether to use adjusted p-values (must have q-values in
<code>object</code>). Default = FALSE</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_plotcutoff">plotCutoff</code></td>
<td>
<p>Which probes to include on plot by significance cut-off
(default = 1, for all markers)</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_graphics">graphics</code></td>
<td>
<p>Graphics system to use: &quot;ggplot&quot; or &quot;plotly&quot;</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_fontsize">fontSize</code></td>
<td>
<p>Font size</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_labelfontsize">labelFontSize</code></td>
<td>
<p>Font size for labels</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_colours">colours</code></td>
<td>
<p>Vector of colours to use for significance groups</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print statistics</p>
</td></tr>
<tr><td><code id="fcPlot_+3A_...">...</code></td>
<td>
<p>Other parameters to pass to plotly or ggplot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a plot for fold change between x1Values in one x2Value
subset on x axis and fold change in the other x2Value on the y axis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PEAC_minimal_load)

disp &lt;- apply(tpm, 1, function(x) {
(var(x, na.rm = TRUE)-mean(x, na.rm = TRUE))/(mean(x, na.rm = TRUE)**2)
})

glmmFit &lt;- glmmSeq(~ Timepoint * EULAR_6m + (1 | PATID),
                     countdata = tpm[1:5, ],
                     metadata = metadata,
                     dispersion = disp,
                     verbose = FALSE)

fcPlot(object = glmmFit,
      x1var = "Timepoint",
      x2var = "EULAR_6m",
      x2Values = c("Good", "Non-response"),
      pCutoff = 0.05,
      useAdjusted = FALSE,
      plotCutoff = 1,
      graphics = "plotly")
</code></pre>

<hr>
<h2 id='ggmodelPlot'>Mixed model effects plot using ggplot2</h2><span id='topic+ggmodelPlot'></span>

<h3>Description</h3>

<p>Plot to show differences between groups and over time using ggplot2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggmodelPlot(
  object,
  geneName = NULL,
  x1var = NULL,
  x2var = NULL,
  x2shift = NULL,
  xlab = NULL,
  ylab = geneName,
  plab = NULL,
  title = geneName,
  logTransform = is(object, "GlmmSeq"),
  shapes = 19,
  colours = "grey60",
  lineColours = "grey60",
  markerSize = 1,
  fontSize = 12,
  alpha = 0.7,
  x2Offset = 5,
  addPoints = TRUE,
  addModel = TRUE,
  modelSize = 4,
  modelColours = "blue",
  modelLineSize = 1,
  modelLineColours = modelColours,
  addBox = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggmodelPlot_+3A_object">object</code></td>
<td>
<p>A glmmSeq/lmmSeq object created by
<code><a href="#topic+glmmSeq">glmmSeq::glmmSeq()</a></code> or
<code><a href="#topic+lmmSeq">glmmSeq::lmmSeq()</a></code></p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_genename">geneName</code></td>
<td>
<p>The gene/row name to be plotted</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_x1var">x1var</code></td>
<td>
<p>The name of the first (inner) x parameter, typically 'time'.
This is anticipated to have different values when matched by ID.</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_x2var">x2var</code></td>
<td>
<p>The name of an optional second (outer) x parameter, which should be a
factor.</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_x2shift">x2shift</code></td>
<td>
<p>Amount to shift along x axis for each level of <code>x2var</code>. By
default the function will arrange each level of <code>x2var</code> side by side.</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_xlab">xlab</code></td>
<td>
<p>Title for the x axis</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_ylab">ylab</code></td>
<td>
<p>Title for the y axis</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_plab">plab</code></td>
<td>
<p>Optional character vector of labels for p-values. These must
align with column names in <code>object@stats$pvals</code>.</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_title">title</code></td>
<td>
<p>Plot title. If NULL gene name is used</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_logtransform">logTransform</code></td>
<td>
<p>Whether to perform a log10 transform on the y axis</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_shapes">shapes</code></td>
<td>
<p>The marker shapes (default=19)</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_colours">colours</code></td>
<td>
<p>The marker colours as vector</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_linecolours">lineColours</code></td>
<td>
<p>The line colours (default='grey60') as vector</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_markersize">markerSize</code></td>
<td>
<p>Size of markers (default=1)</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_fontsize">fontSize</code></td>
<td>
<p>Plot font size</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_alpha">alpha</code></td>
<td>
<p>Line and marker opacity (default=0.7)</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_x2offset">x2Offset</code></td>
<td>
<p>Vertical adjustment to secondary x-axis labels (default=5)</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_addpoints">addPoints</code></td>
<td>
<p>Whether to add underlying data points (default=TRUE)</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_addmodel">addModel</code></td>
<td>
<p>Whether to add the fit model with markers (default=TRUE)</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_modelsize">modelSize</code></td>
<td>
<p>Size of model points (default=4)</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_modelcolours">modelColours</code></td>
<td>
<p>Colour of model fit markers (default=&quot;blue&quot;) as vector</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_modellinesize">modelLineSize</code></td>
<td>
<p>Size of model points (default=1) as vector</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_modellinecolours">modelLineColours</code></td>
<td>
<p>Colour of model fit lines</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_addbox">addBox</code></td>
<td>
<p>Logical whether to add boxplots for mean and IQR</p>
</td></tr>
<tr><td><code id="ggmodelPlot_+3A_...">...</code></td>
<td>
<p>Other parameters to pass to
<code><a href="ggplot2.html#topic+theme">ggplot2::theme()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a paired plot for matched samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PEAC_minimal_load)

disp &lt;- apply(tpm, 1, function(x){
  (var(x, na.rm=TRUE)-mean(x, na.rm=TRUE))/(mean(x, na.rm=TRUE)**2)
})

MS4A1glmm &lt;- glmmSeq(~ Timepoint * EULAR_6m + (1 | PATID),
                     countdata = tpm['MS4A1', , drop = FALSE],
                     metadata = metadata,
                     dispersion = disp,
                     verbose = FALSE)

ggmodelPlot(object = MS4A1glmm,
           geneName = 'MS4A1',
           x1var = 'Timepoint',
           x2var = 'EULAR_6m',
           colours = c('skyblue', 'goldenrod1', 'mediumvioletred'))
</code></pre>

<hr>
<h2 id='glmmQvals'>Glmm Sequencing qvalues</h2><span id='topic+glmmQvals'></span>

<h3>Description</h3>

<p>Add qvalue columns to the glmmSeq dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmmQvals(object, cutoff = 0.05, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmmQvals_+3A_object">object</code></td>
<td>
<p>A glmmSeq/lmmSeq object created by
<code><a href="#topic+glmmSeq">glmmSeq::glmmSeq()</a></code>.</p>
</td></tr>
<tr><td><code id="glmmQvals_+3A_cutoff">cutoff</code></td>
<td>
<p>Prints a table showing the number of probes considered
significant by the pvalue cut-off (default=0.05)</p>
</td></tr>
<tr><td><code id="glmmQvals_+3A_verbose">verbose</code></td>
<td>
<p>Logical whether to print the number of significant probes
(default=TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a GlmmSeq object with results for gene-wise general linear
mixed models with adjusted p-values using the qvalue function
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PEAC_minimal_load)
disp &lt;- apply(tpm, 1, function(x) {
(var(x, na.rm=TRUE)-mean(x, na.rm = TRUE))/(mean(x, na.rm = TRUE)**2)
})
MS4A1glmm &lt;- glmmSeq(~ Timepoint * EULAR_6m + (1 | PATID),
                     countdata = tpm[1:5, ],
                     metadata = metadata,
                     dispersion = disp[1:5],
                     verbose=FALSE)
MS4A1glmm &lt;- glmmQvals(MS4A1glmm)
</code></pre>

<hr>
<h2 id='glmmRefit'>Refit mixed effects model</h2><span id='topic+glmmRefit'></span><span id='topic+glmmRefit.lmmSeq'></span><span id='topic+glmmRefit.GlmmSeq'></span>

<h3>Description</h3>

<p>Based on a 'GlmmSeq' or 'lmmSeq' class result object, this function attempts
to refit an identical model for a specific gene based on the data and fitting
parameters stored in the results object and refitting using either
<code><a href="lme4.html#topic+glmer">lme4::glmer()</a></code> for <code>GlmmSeq</code> objects or <code>lmer()</code> for <code>lmmSeq</code> objects. The
fitted model can then be passed on to other packages such as <code>emmeans</code> to
look at estimated marginal means for the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmmRefit(object, gene, ...)

## S3 method for class 'lmmSeq'
glmmRefit(object, gene, formula = object@formula, ...)

## S3 method for class 'GlmmSeq'
glmmRefit(
  object,
  gene,
  formula = object@formula,
  control = object@info$control,
  family = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmmRefit_+3A_object">object</code></td>
<td>
<p>A fitted results object of class <code>GlmmSeq</code> or <code>lmmSeq</code></p>
</td></tr>
<tr><td><code id="glmmRefit_+3A_gene">gene</code></td>
<td>
<p>A character value specifying a single gene to extract a fitted
model for</p>
</td></tr>
<tr><td><code id="glmmRefit_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to either <code><a href="lme4.html#topic+glmer">lme4::glmer()</a></code> or
<code><a href="lme4.html#topic+lmer">lme4::lmer()</a></code></p>
</td></tr>
<tr><td><code id="glmmRefit_+3A_formula">formula</code></td>
<td>
<p>Optional formula to use when refitting model</p>
</td></tr>
<tr><td><code id="glmmRefit_+3A_control">control</code></td>
<td>
<p>Optional control parameters, see <code><a href="lme4.html#topic+lmerControl">lme4::lmerControl()</a></code> or
<code><a href="lme4.html#topic+lmerControl">lme4::glmerControl()</a></code></p>
</td></tr>
<tr><td><code id="glmmRefit_+3A_family">family</code></td>
<td>
<p>Optional GLM family when refitting GLMM using <code><a href="lme4.html#topic+glmer">lme4::glmer()</a></code>
or <code><a href="glmmTMB.html#topic+glmmTMB">glmmTMB()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fitted model of class <code>lmerMod</code> in the case of LMM, or <code>glmerMod</code> or
<code>glmmTMB</code> for a GLMM dependent on the original method.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PEAC_minimal_load)
disp &lt;- apply(tpm, 1, function(x) {
  (var(x, na.rm = TRUE)-mean(x, na.rm = TRUE))/(mean(x, na.rm = TRUE)**2)
})
glmmtest &lt;- glmmSeq(~ Timepoint * EULAR_6m + (1 | PATID),
                     countdata = tpm[1:2, ],
                     metadata = metadata,
                     dispersion = disp,
                     verbose = FALSE)

# show summary for single gene
summary(glmmtest, "MS4A1")

# refit a single model using lme4::glmer()
fit &lt;- glmmRefit(glmmtest, "MS4A1")

# refit model with reduced formula
fit2 &lt;- glmmRefit(glmmtest, "MS4A1",
                  formula = count ~ Timepoint + EULAR_6m + (1 | PATID))

# LRT
anova(fit, fit2)

</code></pre>

<hr>
<h2 id='glmmSeq'>GLMM with negative binomial distribution for sequencing count data</h2><span id='topic+glmmSeq'></span>

<h3>Description</h3>

<p>Fits many generalised linear mixed effects models (GLMM) with negative
binomial distribution for analysis of overdispersed count data with random
effects. Designed for longitudinal analysis of RNA-Sequencing count data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmmSeq(
  modelFormula,
  countdata,
  metadata,
  id = NULL,
  dispersion = NA,
  sizeFactors = NULL,
  reduced = NULL,
  modelData = NULL,
  designMatrix = NULL,
  method = c("lme4", "glmmTMB"),
  control = NULL,
  family = nbinom2,
  cores = 1,
  removeSingles = FALSE,
  zeroCount = 0.125,
  verbose = TRUE,
  returnList = FALSE,
  progress = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmmSeq_+3A_modelformula">modelFormula</code></td>
<td>
<p>the model formula. This must be of the form <code>"~ ..."</code>
where the structure is assumed to be <code>"counts ~ ..."</code>. The formula must
include a random effects term. For more information on formula structure
for random effects see <code><a href="lme4.html#topic+glmer">lme4::glmer()</a></code></p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_countdata">countdata</code></td>
<td>
<p>the sequencing count data matrix with genes in rows and
samples in columns</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_metadata">metadata</code></td>
<td>
<p>a dataframe of sample information with variables in columns
and samples in rows</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_id">id</code></td>
<td>
<p>Optional. Used to specify the column in metadata which contains the
sample IDs to be used in repeated samples for random effects. If not
specified, the function defaults to using the variable after the &quot;|&quot; in the
random effects term in the formula.</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_dispersion">dispersion</code></td>
<td>
<p>a numeric vector of gene dispersion. Not required for
<code>glmmTMB</code> models, as these determine and fit dispersion for each gene.</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_sizefactors">sizeFactors</code></td>
<td>
<p>size factors (default = NULL). If provided the <code>glmer</code>
offset is set to log(sizeFactors). For more information see&ldquo;
<code><a href="lme4.html#topic+glmer">lme4::glmer()</a></code></p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_reduced">reduced</code></td>
<td>
<p>Optional reduced model formula. If this is chosen, a
likelihood ratio test is used to calculate p-values instead of the default
Wald type 2 Chi-squared test.</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_modeldata">modelData</code></td>
<td>
<p>Optional dataframe. Default is generated by call to
<code>expand.grid</code> using levels of variables in the formula. Used to calculate
model predictions (estimated means &amp; 95% CI) for plotting via <a href="#topic+modelPlot">modelPlot</a>.
It can therefore be used to add/remove points in <a href="#topic+modelPlot">modelPlot</a>.</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_designmatrix">designMatrix</code></td>
<td>
<p>custom design matrix, used only for prediction</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_method">method</code></td>
<td>
<p>Specifies which package to use for fitting GLMM models. Either
&quot;lme4&quot; or &quot;glmmTMB&quot; depending on whether to use <a href="lme4.html#topic+glmer">lme4::glmer</a> or
<a href="glmmTMB.html#topic+glmmTMB">glmmTMB::glmmTMB</a> to fit GLMM models.</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_control">control</code></td>
<td>
<p>the <code>glmer</code> optimizer control. If <code>method = "lme4"</code> default is
<code>glmerControl(optimizer = "bobyqa")</code>). If
<code>method = "glmmTMB"</code> default is <code>glmmTMBControl()</code></p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_family">family</code></td>
<td>
<p>Only used with <code>glmmTMB</code> models. Default is <code>nbinom2</code>. See
<a href="glmmTMB.html#topic+nbinom2">glmmTMB::nbinom2</a></p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_cores">cores</code></td>
<td>
<p>number of cores to use. Default = 1.</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_removesingles">removeSingles</code></td>
<td>
<p>whether to remove individuals without repeated measures
(default = FALSE)</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_zerocount">zeroCount</code></td>
<td>
<p>numerical value to offset zeroes for the purpose of log
(default = 0.125)</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_verbose">verbose</code></td>
<td>
<p>Logical whether to display messaging (default = TRUE)</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_returnlist">returnList</code></td>
<td>
<p>Logical whether to return results as a list or <code>glmmSeq</code>
object (default = FALSE). Useful for debugging.</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_progress">progress</code></td>
<td>
<p>Logical whether to display a progress bar</p>
</td></tr>
<tr><td><code id="glmmSeq_+3A_...">...</code></td>
<td>
<p>Other parameters to pass to
<code><a href="lme4.html#topic+glmer">lme4::glmer()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a wrapper for <code><a href="lme4.html#topic+glmer">lme4::glmer()</a></code>. By default, p-values for each
model term are computed using Wald type 2 Chi-squared test as per
<code><a href="car.html#topic+Anova">car::Anova()</a></code>. The underlying code for this has been optimised for speed.
However, if a reduced model formula is specified by setting <code>reduced</code>, then a
likelihood ratio test is performed instead using <a href="stats.html#topic+anova">stats::anova</a>. This will
double computation time since two GLMM have to be fitted.
</p>
<p>Parallelisation is provided using <a href="parallel.html#topic+mclapply">parallel::mclapply</a> on Unix/Mac or
<a href="parallel.html#topic+clusterApply">parallel::parLapply</a> on PC.
</p>
<p>Setting <code>method = "glmmTMB"</code> enables an alternative method of fitting GLMM
using the <code>glmmTMB</code> package. This gives access to a variety of alternative
GLM family functions. Note, <code>glmmTMB</code> negative binomial models are
substantially slower to fit than <code>glmer</code> models with known dispersion due to
the extra time taken by <code>glmmTMB</code> to optimise the dispersion parameter.
</p>
<p>The <code>id</code> argument is usually optional. By default the <code>id</code> column in the
metadata is determined as the term after the bar in the random effects term
of the model. Note that <code>id</code> is not passed to <code>glmer</code> or <code>glmmTMB</code>. It is
only really used to remove singletons from the <code>countdata</code> matrix and
<code>metadata</code> dataframe. The <code>id</code> is also stored in the output from <code>glmmSeq</code>
and used by plotting function <code><a href="#topic+modelPlot">modelPlot()</a></code>. However, due to its flexible
nature, in theory <code>glmmSeq</code> should allow for more than one random effect
term, although this has not been tested formally. In this case, it is
probably prudent to specify a value for <code>id</code>.
</p>


<h3>Value</h3>

<p>Returns an S4 class <code>GlmmSeq</code> object with results for gene-wise
general linear mixed models. A list of results is returned if <code>returnList</code>
is <code>TRUE</code> which is useful for debugging. If all genes return errors from
<code>glmer</code>, then an error message is shown and a character vector containing
error messages for all genes is returned.
</p>


<h3>See Also</h3>

<p><a href="lme4.html#topic+glmer">lme4::glmer</a> <a href="lme4.html#topic+lmerControl">lme4::glmerControl</a> <a href="glmmTMB.html#topic+glmmTMB">glmmTMB::glmmTMB</a>
<a href="glmmTMB.html#topic+nbinom2">glmmTMB::nbinom2</a> <a href="glmmTMB.html#topic+glmmTMBControl">glmmTMB::glmmTMBControl</a> <a href="car.html#topic+Anova">car::Anova</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PEAC_minimal_load)
disp &lt;- apply(tpm, 1, function(x) {
(var(x, na.rm = TRUE)-mean(x, na.rm = TRUE))/(mean(x, na.rm = TRUE)**2)
})
MS4A1glmm &lt;- glmmSeq(~ Timepoint * EULAR_6m + (1 | PATID),
                     countdata = tpm[1:2, ],
                     metadata = metadata,
                     dispersion = disp,
                     verbose = FALSE)
names(attributes(MS4A1glmm))

</code></pre>

<hr>
<h2 id='GlmmSeq-class'>An S4 class to define the glmmSeq output</h2><span id='topic+GlmmSeq-class'></span>

<h3>Description</h3>

<p>An S4 class to define the glmmSeq output
</p>


<h3>Slots</h3>


<dl>
<dt><code>info</code></dt><dd><p>List including the matched call, dispersions, offset, designMatrix</p>
</dd>
<dt><code>formula</code></dt><dd><p>The model formula</p>
</dd>
<dt><code>stats</code></dt><dd><p>Statistics from fitted models</p>
</dd>
<dt><code>predict</code></dt><dd><p>Predicted values</p>
</dd>
<dt><code>reduced</code></dt><dd><p>Optional reduced formula for LRT</p>
</dd>
<dt><code>countdata</code></dt><dd><p>The input expression data with count data in rows</p>
</dd>
<dt><code>metadata</code></dt><dd><p>The input metadata</p>
</dd>
<dt><code>modelData</code></dt><dd><p>Model data for predictions</p>
</dd>
<dt><code>optInfo</code></dt><dd><p>Information on whether the model was singular or converged</p>
</dd>
<dt><code>errors</code></dt><dd><p>Any errors</p>
</dd>
<dt><code>vars</code></dt><dd><p>List of variables stored from the original call, including the
<code>id</code> variable (by default automatically identified from the random effect
term in the model) and <code>removeSingles</code> argument</p>
</dd>
</dl>

<hr>
<h2 id='lmmSeq'>Linear mixed models for data matrix</h2><span id='topic+lmmSeq'></span>

<h3>Description</h3>

<p>Fits many linear mixed effects models for analysis of gaussian data with
random effects, with parallelisation and optimisation for speed. It is
suitable for longitudinal analysis of high dimensional data. Wald type 2
Chi-squared test is used to calculate p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lmmSeq(
  modelFormula,
  maindata,
  metadata,
  id = NULL,
  offset = NULL,
  test.stat = c("Wald", "F", "LRT"),
  reduced = NULL,
  modelData = NULL,
  designMatrix = NULL,
  control = lmerControl(),
  cores = 1,
  removeSingles = FALSE,
  verbose = TRUE,
  returnList = FALSE,
  progress = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lmmSeq_+3A_modelformula">modelFormula</code></td>
<td>
<p>the model formula. This must be of the form <code>"~ ..."</code>
where the structure is assumed to be <code>"gene ~ ..."</code>. The formula must
include a random effects term. See formula structure for random effects in
<code><a href="lme4.html#topic+lmer">lme4::lmer()</a></code></p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_maindata">maindata</code></td>
<td>
<p>data matrix with genes in rows and samples in columns</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_metadata">metadata</code></td>
<td>
<p>a dataframe of sample information with variables in columns
and samples in rows</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_id">id</code></td>
<td>
<p>Optional. Used to specify the column in metadata which contains the
sample IDs to be used in repeated samples for random effects. If not
specified, the function defaults to using the variable after the &quot;|&quot; in the
random effects term in the formula.</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_offset">offset</code></td>
<td>
<p>Vector containing model offsets (default = NULL). If provided
the <code>lmer()</code> offset is set to <code>offset</code>. See
<code><a href="lme4.html#topic+lmer">lme4::lmer()</a></code></p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_test.stat">test.stat</code></td>
<td>
<p>Character value specifying test statistic. Current options
are &quot;Wald&quot; for type 2 Wald Chi square test using code derived and modified
from <a href="car.html#topic+Anova">car::Anova</a> to improve speed for matrix tests. Or &quot;F&quot; for conditional
F tests using Saiterthwaite's method of approximated Df. This uses
<a href="lmerTest.html#topic+lmer">lmerTest::lmer</a> and is somewhat slower.</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_reduced">reduced</code></td>
<td>
<p>Optional reduced model formula. If this is chosen, a
likelihood ratio test is used to calculate p-values instead of the default
Wald type 2 Chi-squared test.</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_modeldata">modelData</code></td>
<td>
<p>Optional dataframe. Default is generated by call to
<code>expand.grid</code> using levels of variables in the formula. Used to calculate
model predictions (estimated means &amp; 95% CI) for plotting via <a href="#topic+modelPlot">modelPlot</a>.
It can therefore be used to add/remove points in <a href="#topic+modelPlot">modelPlot</a>.</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_designmatrix">designMatrix</code></td>
<td>
<p>Optional custom design matrix generated by call to
<code>model.matrix</code> using <code>modelData</code> and <code>FEformula</code>. Used to calculate
model predictions for plotting.</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_control">control</code></td>
<td>
<p>the <code>lmer</code> optimizer control (default = <code>lmerControl()</code>). See
<code><a href="lme4.html#topic+lmerControl">lme4::lmerControl()</a></code>.</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_cores">cores</code></td>
<td>
<p>number of cores to use for parallelisation. Default = 1.</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_removesingles">removeSingles</code></td>
<td>
<p>whether to remove individuals with no repeated measures
(default = FALSE)</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_verbose">verbose</code></td>
<td>
<p>Logical whether to display messaging (default = TRUE)</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_returnlist">returnList</code></td>
<td>
<p>Logical whether to return results as a list or lmmSeq
object (default = FALSE). Helpful for debugging.</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_progress">progress</code></td>
<td>
<p>Logical whether to display a progress bar</p>
</td></tr>
<tr><td><code id="lmmSeq_+3A_...">...</code></td>
<td>
<p>Other parameters passed to
<code><a href="lmerTest.html#topic+lmer">lmerTest::lmer()</a></code>. Only available if
<code>test.stat = "F"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, p-values for each model term are computed using Wald type 2
Chi-squared test as per <code><a href="car.html#topic+Anova">car::Anova()</a></code>. The underlying code for this has been
optimised for speed. However, if a reduced model formula is specified by
setting <code>reduced</code>, then a likelihood ratio test (LRT) is performed instead
using <code><a href="lme4.html#topic+anova.merMod">anova</a></code>. This will double computation
time since two LMM have to be fitted for each gene. For LRT, models being
compared are optimised by maximum likelihood and not REML (<code>REML=FALSE</code>).
</p>
<p>Two key methods are used to speed up computation above and beyond simple
parallelisation. The first is to speed up <code><a href="lme4.html#topic+lmer">lme4::lmer()</a></code> by calling
<code><a href="lme4.html#topic+modular">lme4::lFormula()</a></code> once at the start and then updating the <code>lFormula</code> output
with new data. The 2nd speed up is through optimised code for repeated type 2
Wald Chi-squared tests (original code was derived from <a href="car.html#topic+Anova">car::Anova</a>). For
example, elements such as the hypothesis matrices are generated only once to
reduce unnecessarily repetitive computation, and the generation of p-values
from Chi-squared values is vectorised and performed at the end. F-tests using
the <code>lmerTest</code> package have not been optimised and are therefore slower.
</p>
<p>Parallelisation is performed using <a href="parallel.html#topic+mclapply">parallel::mclapply</a> on unix/mac and
<a href="parallel.html#topic+clusterApply">parallel::parLapply</a> on windows. Progress bars use <a href="pbmcapply.html#topic+pbmclapply">pbmcapply::pbmclapply</a>
on unix/mac and <a href="pbapply.html#topic+pbapply">pbapply::pblapply</a> on windows.
</p>
<p>The <code>id</code> argument is usually optional. By default the <code>id</code> column in the
metadata is determined as the term after the bar in the random effects term
of the model. Note that <code>id</code> is not passed to <code>lmer</code>. It is only really used
to remove singletons from the <code>maindata</code> matrix and <code>metadata</code> dataframe. The
<code>id</code> is also stored in the output from <code>lmmSeq</code> and used by plotting function
<code><a href="#topic+modelPlot">modelPlot()</a></code>. However, due to its flexible nature, in theory <code>lmmSeq</code> should
allow for more than one random effect term, although this has not been tested
formally. In this case, it is probably prudent to specify a value for <code>id</code>.
</p>


<h3>Value</h3>

<p>Returns an S4 class <code>lmmSeq</code> object with results for gene-wise linear
mixed models; or a list of results if <code>returnList</code> is <code>TRUE</code>, which is
useful for debugging. If all genes return errors from <code>lmer</code>, then an error
message is shown and a character vector containing error messages for all
genes is returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PEAC_minimal_load)
logtpm &lt;- log2(tpm +1)
lmmtest &lt;- lmmSeq(~ Timepoint * EULAR_6m + (1 | PATID),
                     maindata = logtpm[1:2, ],
                     metadata = metadata,
                     verbose = FALSE)
names(attributes(lmmtest))
</code></pre>

<hr>
<h2 id='lmmSeq-class'>An S4 class to define the lmmSeq output</h2><span id='topic+lmmSeq-class'></span>

<h3>Description</h3>

<p>An S4 class to define the lmmSeq output
</p>


<h3>Slots</h3>


<dl>
<dt><code>info</code></dt><dd><p>List including matched call, offset, designMatrix</p>
</dd>
<dt><code>formula</code></dt><dd><p>The model formula</p>
</dd>
<dt><code>stats</code></dt><dd><p>Statistics from fitted models</p>
</dd>
<dt><code>predict</code></dt><dd><p>Predicted values</p>
</dd>
<dt><code>reduced</code></dt><dd><p>Optional reduced formula for LRT</p>
</dd>
<dt><code>maindata</code></dt><dd><p>The input expression data with variables in rows</p>
</dd>
<dt><code>metadata</code></dt><dd><p>The input metadata</p>
</dd>
<dt><code>modelData</code></dt><dd><p>Model data for predictions</p>
</dd>
<dt><code>optInfo</code></dt><dd><p>Information on whether the model was singular or converged</p>
</dd>
<dt><code>errors</code></dt><dd><p>Any errors</p>
</dd>
<dt><code>vars</code></dt><dd><p>List of variables stored from the original call</p>
</dd>
</dl>

<hr>
<h2 id='maPlot'>MA plots</h2><span id='topic+maPlot'></span>

<h3>Description</h3>

<p>MA plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maPlot(
  object,
  x1var,
  x2var,
  x1Values = NULL,
  x2Values = NULL,
  pCutoff = 0.01,
  plotCutoff = 1,
  zeroCountCutoff = 50,
  colours = c("grey", "midnightblue", "mediumvioletred", "goldenrod"),
  labels = c(),
  fontSize = 12,
  labelFontSize = 4,
  useAdjusted = FALSE,
  graphics = "ggplot",
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maPlot_+3A_object">object</code></td>
<td>
<p>A glmmSeq object created by
<code><a href="#topic+glmmSeq">glmmSeq::glmmSeq()</a></code>.</p>
</td></tr>
<tr><td><code id="maPlot_+3A_x1var">x1var</code></td>
<td>
<p>The name of the first (inner) x parameter</p>
</td></tr>
<tr><td><code id="maPlot_+3A_x2var">x2var</code></td>
<td>
<p>The name of the second (outer) x parameter</p>
</td></tr>
<tr><td><code id="maPlot_+3A_x1values">x1Values</code></td>
<td>
<p>Timepoints or categories in <code>x1var</code> to be used to calculate fold
change. If <code>NULL</code> the first two levels in <code>x1var</code> are used.</p>
</td></tr>
<tr><td><code id="maPlot_+3A_x2values">x2Values</code></td>
<td>
<p>Categories in <code>x2var</code> to be compared on x and y axis.</p>
</td></tr>
<tr><td><code id="maPlot_+3A_pcutoff">pCutoff</code></td>
<td>
<p>The significance cut-off for colour-coding (default=0.01)</p>
</td></tr>
<tr><td><code id="maPlot_+3A_plotcutoff">plotCutoff</code></td>
<td>
<p>Which probes to include by significance cut-off
(default=1 for all markers)</p>
</td></tr>
<tr><td><code id="maPlot_+3A_zerocountcutoff">zeroCountCutoff</code></td>
<td>
<p>Which probes to include by minimum counts cut-off
(default=50)</p>
</td></tr>
<tr><td><code id="maPlot_+3A_colours">colours</code></td>
<td>
<p>Vector of colours to use for significance groups</p>
</td></tr>
<tr><td><code id="maPlot_+3A_labels">labels</code></td>
<td>
<p>Row names or indices to label on plot</p>
</td></tr>
<tr><td><code id="maPlot_+3A_fontsize">fontSize</code></td>
<td>
<p>Font size</p>
</td></tr>
<tr><td><code id="maPlot_+3A_labelfontsize">labelFontSize</code></td>
<td>
<p>Font size for labels</p>
</td></tr>
<tr><td><code id="maPlot_+3A_useadjusted">useAdjusted</code></td>
<td>
<p>whether to use adjusted p-values
(must have q-values in <code>object</code>)</p>
</td></tr>
<tr><td><code id="maPlot_+3A_graphics">graphics</code></td>
<td>
<p>Either &quot;ggplot&quot; or &quot;plotly&quot;</p>
</td></tr>
<tr><td><code id="maPlot_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print statistics</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of three plots. One plot for each <code>x2Value</code> and one combined
figure
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PEAC_minimal_load)

disp &lt;- apply(tpm, 1, function(x){
(var(x, na.rm=TRUE)-mean(x, na.rm=TRUE))/(mean(x, na.rm=TRUE)**2)
})

resultTable &lt;- glmmSeq(~ Timepoint * EULAR_6m + (1 | PATID),
                       countdata = tpm[1:5, ],
                       metadata = metadata,
                       dispersion = disp)

plots &lt;- maPlot(resultTable,
                x1var='Timepoint',
                x2var='EULAR_6m',
                x2Values=c('Good', 'Non-response'),
                graphics="plotly")

plots$combined
</code></pre>

<hr>
<h2 id='metadata'>Minimal metadata from PEAC</h2><span id='topic+metadata'></span>

<h3>Description</h3>

<p>Minimal metadata for paired longitudinal response analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metadata
</code></pre>


<h3>Format</h3>

<p>A data frame
</p>

<dl>
<dt>PATID</dt><dd><p>Id for matching patients</p>
</dd>
<dt>Timepoint</dt><dd><p>timepoints</p>
</dd>
<dt>EULAR_6m</dt><dd><p>response data</p>
</dd>
</dl>


<hr>
<h2 id='modelPlot'>Mixed model effects plot</h2><span id='topic+modelPlot'></span>

<h3>Description</h3>

<p>Plot to show differences between groups over time using base graphics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modelPlot(
  object,
  geneName = NULL,
  x1var = NULL,
  x2var = NULL,
  x2shift = NULL,
  xlab = NA,
  ylab = geneName,
  plab = NULL,
  title = geneName,
  logTransform = is(object, "GlmmSeq"),
  shapes = 21,
  colours = "grey60",
  lineColours = "grey60",
  markerSize = 0.5,
  fontSize = NULL,
  alpha = 0.7,
  addModel = TRUE,
  addPoints = TRUE,
  modelSize = 2,
  modelColours = "royalblue",
  modelLineSize = 1,
  modelLineColours = modelColours,
  errorBarLwd = 2.5,
  errorBarLength = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="modelPlot_+3A_object">object</code></td>
<td>
<p>A glmmSeq/lmmSeq object created by
<code><a href="#topic+glmmSeq">glmmSeq::glmmSeq()</a></code> or
<code><a href="#topic+lmmSeq">glmmSeq::lmmSeq()</a></code></p>
</td></tr>
<tr><td><code id="modelPlot_+3A_genename">geneName</code></td>
<td>
<p>The gene/row name to be plotted</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_x1var">x1var</code></td>
<td>
<p>The name of the first (inner) x parameter, typically 'time'.
This is anticipated to have different values when matched by ID.</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_x2var">x2var</code></td>
<td>
<p>The name of an optional second (outer) x parameter, which should
be a factor.</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_x2shift">x2shift</code></td>
<td>
<p>Amount to shift along x axis for each level of <code>x2var</code>. By
default the function will arrange each level of <code>x2var</code> side by side. Lower
values of <code>x2shift</code> or <code>x2shift = 0</code> can be used to overlap plots similar
to 'dodge' or stagger them.</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_xlab">xlab</code></td>
<td>
<p>Title for the x axis</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_ylab">ylab</code></td>
<td>
<p>Title for the y axis</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_plab">plab</code></td>
<td>
<p>Optional character vector of labels for p-values. These must
align with column names in <code>object@stats$pvals</code>.</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_title">title</code></td>
<td>
<p>Plot title. If NULL gene name is used</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_logtransform">logTransform</code></td>
<td>
<p>Whether to perform a log10 transform on the y axis</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_shapes">shapes</code></td>
<td>
<p>The marker shapes (default=19)</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_colours">colours</code></td>
<td>
<p>The marker colours (default='red') as vector or named vector</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_linecolours">lineColours</code></td>
<td>
<p>The line colours (default='grey60') as vector or named
vector</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_markersize">markerSize</code></td>
<td>
<p>Size of markers (default=2)</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_fontsize">fontSize</code></td>
<td>
<p>Plot font size</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_alpha">alpha</code></td>
<td>
<p>Line and marker opacity (default=0.7)</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_addmodel">addModel</code></td>
<td>
<p>Whether to add the fit model with markers (default=TRUE)</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_addpoints">addPoints</code></td>
<td>
<p>Whether to add underlying data points (default=TRUE)</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_modelsize">modelSize</code></td>
<td>
<p>Size of model points (default=2)</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_modelcolours">modelColours</code></td>
<td>
<p>Colour of model fit markers (default=&quot;black&quot;) as vector
or named vector</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_modellinesize">modelLineSize</code></td>
<td>
<p>Size of model points (default=1) as vector or named
vector</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_modellinecolours">modelLineColours</code></td>
<td>
<p>Colour of model fit lines.</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_errorbarlwd">errorBarLwd</code></td>
<td>
<p>Line width of error bars</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_errorbarlength">errorBarLength</code></td>
<td>
<p>Head width of error bars</p>
</td></tr>
<tr><td><code id="modelPlot_+3A_...">...</code></td>
<td>
<p>Other parameters to pass to
<code><a href="graphics.html#topic+plot">graphics::plot()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a paired plot for matched samples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PEAC_minimal_load)

disp &lt;- apply(tpm, 1, function(x){
  (var(x, na.rm=TRUE)-mean(x, na.rm=TRUE))/(mean(x, na.rm=TRUE)**2)
})

MS4A1glmm &lt;- glmmSeq(~ Timepoint * EULAR_6m + (1 | PATID),
                     countdata = tpm[1:2, ],
                     metadata = metadata,
                     dispersion = disp)

modelPlot(object=MS4A1glmm,
          geneName = 'MS4A1',
          x1var = 'Timepoint',
          x2var='EULAR_6m')
</code></pre>

<hr>
<h2 id='summary.lmmSeq'>Summarise a 'glmmSeq'/'lmmSeq' object</h2><span id='topic+summary.lmmSeq'></span><span id='topic+summary.GlmmSeq'></span>

<h3>Description</h3>

<p>Summarise results from <a href="#topic+glmmSeq">glmmSeq</a> or <a href="#topic+lmmSeq">lmmSeq</a> analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'lmmSeq'
summary(object, gene = NULL, digits = max(3L, getOption("digits") - 3L), ...)

## S3 method for class 'GlmmSeq'
summary(object, gene = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.lmmSeq_+3A_object">object</code></td>
<td>
<p>an object of class <code>"GlmmSeq"</code> or <code>"lmmSeq"</code></p>
</td></tr>
<tr><td><code id="summary.lmmSeq_+3A_gene">gene</code></td>
<td>
<p>an optional character value specifying a single gene whose
results are summarised</p>
</td></tr>
<tr><td><code id="summary.lmmSeq_+3A_digits">digits</code></td>
<td>
<p>integer, used for number formatting</p>
</td></tr>
<tr><td><code id="summary.lmmSeq_+3A_...">...</code></td>
<td>
<p>arguments to be passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>gene=NULL</code> a dataframe of results for all genes is returned. Otherwise
the output of GLMM or LMM model results for a single gene including
coefficients, test statistics, p-values is printed and the dataframe for all
genes is returned invisibly.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+glmmSeq">glmmSeq()</a></code>, <code><a href="#topic+lmmSeq">lmmSeq()</a></code>
</p>

<hr>
<h2 id='tpm'>TPM count data from PEAC</h2><span id='topic+tpm'></span>

<h3>Description</h3>

<p>Transcripts Per Million (TPM) count data for PEAC synovial
biopsies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tpm
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 50 rows and 123 columns.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
