<!DOCTYPE html><html><head><title>Help for package gLRTH</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gLRTH}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gLRTH_A'><p>The function for the likelihood ratio test for genome-wide association under genetic</p>
heterogeneity with genotype frequencies as input values</a></li>
<li><a href='#gLRTH_L'><p>The function for the likelihood ratio test for genetic linkage under transmission heterogeneity</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genome-Wide Association and Linkage Analysis under Heterogeneity</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Likelihood ratio tests for genome-wide association and genome-wide linkage analysis under heterogeneity. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-01-22 13:19:33 UTC; Xiaoxia Han</td>
</tr>
<tr>
<td>Author:</td>
<td>Xiaoxia Han [aut, cre],
  Yongzhao Shao [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xiaoxia Han &lt;xh414@nyu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-01-22 13:34:44 UTC</td>
</tr>
</table>
<hr>
<h2 id='gLRTH_A'>The function for the likelihood ratio test for genome-wide association under genetic
heterogeneity with genotype frequencies as input values</h2><span id='topic+gLRTH_A'></span>

<h3>Description</h3>

<p>We consider a binary trait and focus on detecting association with disease at a single locus with two
alleles <code class="reqn">A</code> and <code class="reqn">a</code>. The likelihood ratio test is based
on a binomial mixture model of <code class="reqn">J</code> components (<code class="reqn">J \ge 2</code>) for diseased cases:
</p>
<p style="text-align: center;"><code class="reqn">P_{\eta}(X_D=g)=\sum_{j=1}^J \alpha_j B_2(g, \theta_j), \; g=0, 1, 2, \; J \geq 2, \; \sum_{j=1}^J \alpha_j=1, \; \theta_j, \alpha_j \in (0, 1),</code>
</p>

<p>where <code class="reqn">\eta=(\eta_j)_{j \leq J}, \eta_j=(\theta_j, \alpha_j)^T, j=1, \ldots, J</code>,
<code class="reqn">B_2(g, \theta_j)</code> is the probability mass function for a binomial distribution <code class="reqn">X \sim Bin(2, \theta_j)</code>,
and <code class="reqn">\theta_i=\theta_j</code> if and only if <code class="reqn">i=j</code>. <code class="reqn">\theta_j</code> is the probability
of having the allele of interest on one chromosome for a subgroup of case <code class="reqn">j</code>.
In particular, <code class="reqn">J</code> is likely to be quite
large for many of the complex disease with genetic heterogeneity. Note that the LRT-H can
be applied to association studies without the need to know the exact value of <code class="reqn">J</code> while allowing <code class="reqn">J \ge 2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gLRTH_A(n0, n1, n2, m0, m1, m2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gLRTH_A_+3A_n0">n0</code></td>
<td>
<p>AA genotype frequency in case</p>
</td></tr>
<tr><td><code id="gLRTH_A_+3A_n1">n1</code></td>
<td>
<p>Aa genotype frequency in case</p>
</td></tr>
<tr><td><code id="gLRTH_A_+3A_n2">n2</code></td>
<td>
<p>aa genotype frequency in case</p>
</td></tr>
<tr><td><code id="gLRTH_A_+3A_m0">m0</code></td>
<td>
<p>AA genotype frequency in control</p>
</td></tr>
<tr><td><code id="gLRTH_A_+3A_m1">m1</code></td>
<td>
<p>Aa genotype frequency in control</p>
</td></tr>
<tr><td><code id="gLRTH_A_+3A_m2">m2</code></td>
<td>
<p>aa genotype frequency in control</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The test statistic and asymptotic p-value for the likelihood ratio test for GWAS under genetic heterogeneity
</p>


<h3>Author(s)</h3>

<p>Xiaoxia Han and Yongzhao Shao
</p>


<h3>References</h3>

<p>Qian M., Shao Y. (2013) A Likelihood Ratio Test for Genome-Wide Association under Genetic Heterogeneity.
Annals of Human Genetics, 77(2): 174-182.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gLRTH_A(n0=2940, n1=738, n2=53, m0=3601, m1=1173, m2=117)
</code></pre>

<hr>
<h2 id='gLRTH_L'>The function for the likelihood ratio test for genetic linkage under transmission heterogeneity</h2><span id='topic+gLRTH_L'></span>

<h3>Description</h3>

<p>We consider a binary trait and focus on detecting a transmission heterogeneity at a single locus with two
alleles <code class="reqn">A</code> and <code class="reqn">a</code>. We consider independent families each with one marker homozygous (<code class="reqn">AA</code>) parent, one marker
heterozygous parent (<code class="reqn">Aa</code>) and two diseased children. This likelihood ratio test is to test transmission
heterogeneity of preferential transmission of marker allele &quot;a&quot; to an affected child based on a binomial
mixture model with <code class="reqn">J</code> components (<code class="reqn">J \ge 2</code>),
</p>
<p style="text-align: center;"><code class="reqn">P_{\eta}(X_D=g)=\sum_{j=1}^J \alpha_j B_2(g, \theta_j), \; g=0, 1, 2, \; J \geq 2, \; \sum_{j=1}^J \alpha_j=1, \; \theta_j, \alpha_j \in (0, 1),</code>
</p>

<p>where <code class="reqn">\eta=(\eta_j)_{j \leq J}, \eta_j=(\theta_j, \alpha_j)^T, j=1, \ldots, J</code>,
<code class="reqn">B_2(g, \theta_j)</code> is the probability mass function for a binomial distribution <code class="reqn">X \sim Bin(2, \theta_j)</code>,
and <code class="reqn">\theta_i=\theta_j</code> if and only if <code class="reqn">i=j</code>. <code class="reqn">\theta_j</code> is the probability
of transmission of the allele of interest in a subgroup of families <code class="reqn">j</code>.
In particular, <code class="reqn">J</code> is likely to be quite
large for many of the complex disease under transmission heterogeneity. Note that this LRT can
be applied to genome-wide linkage analysis without the need to know the exact value of <code class="reqn">J</code> while allowing <code class="reqn">J \ge 2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gLRTH_L(n0, n1, n2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gLRTH_L_+3A_n0">n0</code></td>
<td>
<p>Number of affected sibling pairs both of which inherited A from their heterozygous parent Aa</p>
</td></tr>
<tr><td><code id="gLRTH_L_+3A_n1">n1</code></td>
<td>
<p>Number of affected sibling pairs which one inherited A and the other inherited a from their heterozygous parent Aa</p>
</td></tr>
<tr><td><code id="gLRTH_L_+3A_n2">n2</code></td>
<td>
<p>Number of affected sibling pairs both of which inherited a from their heterozygous parent Aa</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The test statistic and asymptotic p-value for the likelihood ratio test for linkage analysis under genetic heterogeneity
</p>


<h3>Author(s)</h3>

<p>Xiaoxia Han and Yongzhao Shao
</p>


<h3>References</h3>

<p>Shao Y. (2014) Linkage analysis, originally published
in Encyclopedia of Quantitative Risk Analysis and Assessment, John Wiley &amp; Sons, Ltd, USA, 2008, and
republished in Wiley StatsRef: Statistics Reference Online 2014.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gLRTH_L(n0=100, n1=70, n2=30)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
