<!DOCTYPE html><html><head><title>Help for package GMMAT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GMMAT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GMMAT-package'>
<p>Generalized Linear Mixed Model Association Tests</p></a></li>
<li><a href='#example'><p>Example dataset</p></a></li>
<li><a href='#glmm.score'>
<p>Performing GLMM based score tests</p></a></li>
<li><a href='#glmm.score.meta'>
<p>Performing meta-analysis for GLMM based score test results</p></a></li>
<li><a href='#glmm.wald'>
<p>Performing GLMM based Wald tests</p></a></li>
<li><a href='#glmmkin'>
<p>Fit generalized linear mixed model with known relationship matrices</p></a></li>
<li><a href='#SMMAT'>
<p>Variant Set Mixed Model Association Tests (SMMAT)</p></a></li>
<li><a href='#SMMAT.meta'>
<p>Meta-analysis for variant Set Mixed Model Association Tests (SMMAT)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.4.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-17</td>
</tr>
<tr>
<td>Title:</td>
<td>Generalized Linear Mixed Model Association Tests</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform association tests using generalized linear mixed models (GLMMs) in genome-wide association studies (GWAS) and sequencing association studies. First, GMMAT fits a GLMM with covariate adjustment and random effects to account for population structure and familial or cryptic relatedness. For GWAS, GMMAT performs score tests for each genetic variant as proposed in Chen et al. (2016) &lt;<a href="https://doi.org/10.1016%2Fj.ajhg.2016.02.012">doi:10.1016/j.ajhg.2016.02.012</a>&gt;. For candidate gene studies, GMMAT can also perform Wald tests to get the effect size estimate for each genetic variant. For rare variant analysis from sequencing association studies, GMMAT performs the variant Set Mixed Model Association Tests (SMMAT) as proposed in Chen et al. (2019) &lt;<a href="https://doi.org/10.1016%2Fj.ajhg.2018.12.012">doi:10.1016/j.ajhg.2018.12.012</a>&gt;, including the burden test, the sequence kernel association test (SKAT), SKAT-O and an efficient hybrid test of the burden test and SKAT, based on user-defined variant sets.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Copyright:</td>
<td>See COPYRIGHTS for details.</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, CompQuadForm, foreach, parallel, Matrix, methods,
data.table</td>
</tr>
<tr>
<td>Suggests:</td>
<td>doMC, SeqArray, SeqVarTools, testthat</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>Author:</td>
<td>Han Chen [aut, cre],
  Matthew Conomos [aut],
  Duy Pham [aut],
  Arthor Gilly [ctb],
  Robert Gentleman [ctb, cph] (Author and copyright holder of the C
    function Brent_fmin),
  Ross Ihaka [ctb, cph] (Author and copyright holder of the C function
    Brent_fmin),
  The R Core Team [ctb, cph] (Author and copyright holder of the C
    function Brent_fmin),
  The R Foundation [cph] (Copyright holder of the C function Brent_fmin),
  Eric Biggers [ctb, cph] (Author and copyright holder of included
    libdeflate library),
  Tino Reichardt [ctb, cph] (Author and copyright holder of threading
    code used in the included Zstandard (zstd) library),
  Meta Platforms, Inc. and affiliates [cph] (Copyright holder of included
    Zstandard (zstd) library)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Han Chen &lt;han.chen.2@uth.tmc.edu&gt;</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-17 08:56:15 UTC; hchen</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-17 10:20:33 UTC</td>
</tr>
</table>
<hr>
<h2 id='GMMAT-package'>
Generalized Linear Mixed Model Association Tests
</h2><span id='topic+GMMAT-package'></span><span id='topic+GMMAT'></span>

<h3>Description</h3>

<p>An R package for performing association tests using generalized linear mixed models (GLMMs) in genome-wide association studies (GWAS) and sequencing association studies. First, GMMAT fits a GLMM with covariate adjustment and random effects to account for population structure and familial or cryptic relatedness. For GWAS, GMMAT performs score tests for each genetic variant. For candidate gene studies, GMMAT can also perform Wald tests to get the effect size estimate for each genetic variant. For rare variant analysis from sequencing association studies, GMMAT performs the variant Set Mixed Model Association Tests (SMMAT), including the burden test, the sequence kernel association test (SKAT), SKAT-O and an efficient hybrid test of the burden test and SKAT, based on user-defined variant sets.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> GMMAT</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.4.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2023-11-17</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 3)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Han Chen, Matthew P. Conomos, Duy T. Pham
</p>
<p>Maintainer: Han Chen &lt;Han.Chen.2@uth.tmc.edu&gt;
</p>


<h3>References</h3>

<p>Brent, R.P. (1973) &quot;Chapter 4: An Algorithm with Guaranteed Convergence for Finding a Zero of a Function&quot;, Algorithms for Minimization without Derivatives, Englewood Cliffs, NJ: Prentice-Hall, ISBN 0-13-022335-2.
</p>
<p>Breslow, N.E. and Clayton, D.G. (1993) Approximate Inference in Generalized Linear Mixed Models. Journal of the American Statistical Association 88, 9-25.
</p>
<p>Chen, H., Huffman, J.E., Brody, J.A., Wang, C., Lee, S., Li, Z., Gogarten, S.M., Sofer, T., Bielak, L.F., Bis, J.C., et al. (2019) Efficient variant set mixed model association tests for continuous and binary traits in large-scale whole-genome sequencing studies. The American Journal of Human Genetics 104, 260-274.
</p>
<p>Chen, H., Wang, C., Conomos, M.P., Stilp, A.M., Li, Z., Sofer, T., Szpiro, A.A., Chen, W., Brehm, J.M., Celed√≥n, J.C., Redline, S., Papanicolaou, G.J., Thornton, T.A., Laurie, C.C., Rice, K. and Lin, X. (2016) Control forpopulation structure and relatedness for binary traits in genetic association studies via logistic mixed models. The American Journal of Human Genetics 98, 653-666.
</p>
<p>Gilmour, A.R., Thompson, R. and Cullis, B.R. (1995) Average Information REML: An Efficient Algorithm for Variance Parameter Estimation in Linear Mixed Models. Biometrics 51, 1440-1450.
</p>
<p>Lee, S., Teslovich, T., Boehnke, M., Lin, X. (2013) General framework for meta-analysis of rare variants in sequencing association studies. The American Journal of Human Genetics 93, 42-53.
</p>
<p>Lee, S., Wu, M.C., Lin, X. (2012) Optimal tests for rare variant effects in sequencing association studies. Biostatistics 13, 762-775.
</p>
<p>Nelder, J.A. and Mead, R. (1965) A simplex algorithm for function minimization. Computer Journal 7, 308-313.
</p>
<p>Sun, J., Zheng, Y., Hsu, L. (2013) A unified mixed-effects model for rare-variant association in sequencing studies. Genetic Epidemiology 37, 334-344.
</p>
<p>Wu, M.C., Lee, S., Cai, T., Li, Y., Boehnke, M., Lin, X. (2011) Rare-variant association testing for sequencing data with the sequence kernel association test. The American Journal of Human Genetics 89, 82-93.
</p>
<p>Yang, J., Lee, S.H., Goddard, M.E. and Visscher, P.M. (2011) GCTA: A Tool for Genome-wide Complex Trait Analysis. The American Journal of Human Genetics 88, 76-82.
</p>
<p>Zhou, X. and Stephens, M. (2012) Genome-wide efficient mixed-model analysis for association studies. Nature Genetics 44, 821-824.
</p>

<hr>
<h2 id='example'>Example dataset</h2><span id='topic+example'></span>

<h3>Description</h3>

<p>Example dataset for GMMAT.
</p>


<h3>Format</h3>

<p>Contains the following objects:
</p>

<dl>
<dt>pheno</dt><dd><p>a data frame of 400 observations from a cross-sectional study with 5 variables: id, disease, trait, age and sex.</p>
</dd>
<dt>pheno2</dt><dd><p>a data frame of 2,000 observations from a longitudinal study with 400 individuals and 5 variables: id, y.repeated, y.trend, time and sex.</p>
</dd>
<dt>GRM</dt><dd><p>a genetic relationship matrix for 400 observations.</p>
</dd>
</dl>


<hr>
<h2 id='glmm.score'>
Performing GLMM based score tests
</h2><span id='topic+glmm.score'></span>

<h3>Description</h3>

<p>Use a glmmkin class object from the null GLMM to perform score tests for association with genotypes in a plink .bed file (binary genotypes), a GDS file .gds, or a plain text file (or compressed .gz or .bz2 file).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmm.score(obj, infile, outfile, BGEN.samplefile = NULL, center = T, select = NULL, 
	MAF.range = c(1e-7, 0.5), miss.cutoff = 1, 
	missing.method = "impute2mean", nperbatch = 100, tol = 1e-5,
	infile.nrow = NULL, infile.nrow.skip = 0, infile.sep = "\t",
	infile.na = "NA", infile.ncol.skip = 1, infile.ncol.print = 1,
	infile.header.print = "SNP", is.dosage = FALSE, ncores = 1, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmm.score_+3A_obj">obj</code></td>
<td>

<p>a class glmmkin or class glmmkin.multi object, returned by fitting the null GLMM using <code>glmmkin</code>.
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_infile">infile</code></td>
<td>

<p>the input file name or an object of class SeqVarGDSClass. Note that for plink binary genotype files only the prefix without .bed, .bim or .fam should be used. Only SNP major mode recognized in the binary file. Alternatively, it can be the full name of a BGEN file (including the suffix .bgen), a GDS file (including the suffix .gds), or a plain text file with some delimiters (comma, space, tab or something else), with one row for each SNP and one column for each individual. In that case, SNPs should be coded as numeric values (0/1/2 or dosages allowed, A/C/G/T coding is not recognized). There can be additional rows and columns to skip at the beginning. The order of individuals can be different from <code>obj</code> in the null GLMM (see the argument <code>select</code>). Some compressed files (.gz and .bz2) also allowed. If <code>infile</code> is an object of class SeqVarGDSClass, the .gds file will be closed upon successful completion of the function.
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_outfile">outfile</code></td>
<td>

<p>the output file name.
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_bgen.samplefile">BGEN.samplefile</code></td>
<td>

<p>path to the BGEN sample file. Required when the BGEN file does not contain sample identifiers or the <code>select</code> parameter is NULL (default = NULL).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_center">center</code></td>
<td>

<p>a logical switch for centering genotypes before tests. If TRUE, genotypes will be centered to have mean 0 before tests, otherwise raw values will be directly used in tests (default = TRUE).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_select">select</code></td>
<td>

<p>an optional vector indicating the order of individuals in <code>infile</code>. If supplied, the length must match the number of individuals in <code>infile</code> (default = NULL). Individuals to be excluded should be coded 0. For example, <code>select = c(2, 3, 1, 0)</code> means the 1st individual in <code>infile</code> corresponds to the 2nd individual in <code>obj</code>, the 2nd individual in <code>infile</code> corresponds to the 3rd individual in <code>obj</code>, the 3rd individual in <code>infile</code> corresponds to the 1st individual in <code>obj</code>, the 4th individual in <code>infile</code> is not included in <code>obj</code>. If there are any duplicated <code>id_include</code> in <code>obj</code> (longitudinal data analysis), indices in <code>select</code> should match the order of individuals with unique <code>id_include</code> in <code>obj</code>. For plink binary genotype files and GDS files, this argument is not required and the sample ID's are automatically matched.
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_maf.range">MAF.range</code></td>
<td>

<p>a numeric vector of length 2 defining the minimum and maximum minor allele frequencies of variants that should be included in the analysis (default = c(1e-7, 0.5)).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_miss.cutoff">miss.cutoff</code></td>
<td>

<p>the maximum missing rate allowed for a variant to be included (default = 1, including all variants).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_missing.method">missing.method</code></td>
<td>

<p>method of handling missing genotypes. Either &quot;impute2mean&quot; or &quot;omit&quot; (default = &quot;impute2mean&quot;).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_nperbatch">nperbatch</code></td>
<td>

<p>an integer for how many SNPs should be tested in a batch (default = 100). The computational time can increase dramatically if this value is either small or large. The optimal value for best performance depends on the user's system.
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_tol">tol</code></td>
<td>

<p>the threshold for determining monomorphism. If a SNP has value range less than the tolerance, it will be considered monomorphic and its association test p-value will be NA (default = 1e-5). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_infile.nrow">infile.nrow</code></td>
<td>

<p>number of rows to read in <code>infile</code>, including number of rows to skip at the beginning. If NULL, the program will determine how many rows there are in <code>infile</code> automatically and read all rows (default = NULL). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_infile.nrow.skip">infile.nrow.skip</code></td>
<td>

<p>number of rows to skip at the beginning of <code>infile</code>. Must be nonnegative integers. Useful when header or comment lines are present (default = 0). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_infile.sep">infile.sep</code></td>
<td>

<p>delimiter in <code>infile</code> (default = &quot;\t&quot;). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_infile.na">infile.na</code></td>
<td>

<p>symbol in <code>infile</code> to denote missing genotypes (default = &quot;NA&quot;). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_infile.ncol.skip">infile.ncol.skip</code></td>
<td>

<p>number of columns to skip before genotype data in <code>infile</code>. These columns can be SNP name, alleles and/or quality measures and should be placed at the beginning in each line. After skipping these columns, the program will read in genotype data and perform score tests. Must be nonnegative integers. It is recommended that SNP name should be included as the first column in <code>infile</code> and genotype data should start from the second column or later (default = 1). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_infile.ncol.print">infile.ncol.print</code></td>
<td>

<p>a vector indicating which column(s) in <code>infile</code> should be printed to the output directly. These columns can be SNP name, alleles and/or quality measures placed at the beginning in each line. Must be nonnegative integers, no greater than <code>infile.ncol.skip</code> and sorted numerically in ascending order. By default, it is assumed that the first column is SNP name and genotype data start from the second column, and SNP name should be carried over to the output (default = 1). Should be set to NULL if <code>infile.ncol.skip</code> is 0. Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_infile.header.print">infile.header.print</code></td>
<td>

<p>a character vector indicating column name(s) of column(s) selected to print by <code>infile.ncol.print</code> (default = &quot;SNP&quot;). Should be set to NULL if <code>infile.ncol.skip</code> is 0. Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_is.dosage">is.dosage</code></td>
<td>

<p>a logical switch for whether imputed dosage should be used from a GDS <code>infile</code> (default = FALSE).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_ncores">ncores</code></td>
<td>

<p>a positive integer indicating the number of cores to be used in parallel computing (default = 1).
</p>
</td></tr>
<tr><td><code id="glmm.score_+3A_verbose">verbose</code></td>
<td>

<p>a logical switch for whether a progress bar should be shown for a GDS <code>infile</code> (default = FALSE).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL if <code>infile</code> is a BGEN file (.bgen) or a GDS file (.gds), otherwise computational time in seconds, excluding I/O time.
</p>


<h3>Author(s)</h3>

<p>Han Chen, Duy T. Pham
</p>


<h3>References</h3>

<p>Chen, H., Wang, C., Conomos, M.P., Stilp, A.M., Li, Z., Sofer, T., Szpiro, A.A., Chen, W., Brehm, J.M., Celed√≥n, J.C., Redline, S., Papanicolaou, G.J., Thornton, T.A., Laurie, C.C., Rice, K. and Lin, X. (2016) Control forpopulation structure and relatedness for binary traits in genetic association studies via logistic mixed models. The American Journal of Human Genetics 98, 653-666.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+glmmkin">glmmkin</a></code>, <code><a href="#topic+glmm.wald">glmm.wald</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(example)
attach(example)
model0 &lt;- glmmkin(disease ~ age + sex, data = pheno, kins = GRM, id = "id",
       family = binomial(link = "logit"))
plinkfiles &lt;- strsplit(system.file("extdata", "geno.bed", package = "GMMAT"), 
       ".bed", fixed = TRUE)[[1]]
outfile.bed &lt;- tempfile()
glmm.score(model0, infile = plinkfiles, outfile = outfile.bed)
if(requireNamespace("SeqArray", quietly = TRUE) &amp;&amp; requireNamespace("SeqVarTools",
        quietly = TRUE)) {
	infile &lt;- system.file("extdata", "geno.gds", package = "GMMAT")
	outfile.gds &lt;- tempfile()
	glmm.score(model0, infile = infile, outfile = outfile.gds)
	unlink(outfile.gds)
}
infile &lt;- system.file("extdata", "geno.txt", package = "GMMAT")
outfile.text &lt;- tempfile()
glmm.score(model0, infile = infile, outfile = outfile.text, infile.nrow.skip = 5, 
	infile.ncol.skip = 3, infile.ncol.print = 1:3, 
	infile.header.print = c("SNP", "Allele1", "Allele2"))
infile &lt;- system.file("extdata", "geno.bgen", package = "GMMAT")
samplefile &lt;- system.file("extdata", "geno.sample", package = "GMMAT")
outfile.bgen &lt;- tempfile()
glmm.score(model0, infile = infile, BGEN.samplefile = samplefile,
        outfile = outfile.bgen)
unlink(c(outfile.bed, outfile.text, outfile.bgen))

</code></pre>

<hr>
<h2 id='glmm.score.meta'>
Performing meta-analysis for GLMM based score test results
</h2><span id='topic+glmm.score.meta'></span>

<h3>Description</h3>

<p>Use output files from GLMM based score tests to perform meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmm.score.meta(files, outfile, SNP = rep("SNP", length(files)), 
	A1 = rep("A1", length(files)), A2 = rep("A2", length(files)))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmm.score.meta_+3A_files">files</code></td>
<td>

<p>a vector of input file names. The input files should be the output files of <code>glmm.score( )</code>, or customized tab or space delimited files that include at least 8 columns: SNP, effect allele, noneffect allele, N, AF, SCORE, VAR and PVAL. The column names of SNP, effect allele and noneffect allele can be customized and provided in <code>SNP</code>, <code>A1</code> and <code>A2</code>.
</p>
</td></tr>
<tr><td><code id="glmm.score.meta_+3A_outfile">outfile</code></td>
<td>

<p>the output file name.
</p>
</td></tr>
<tr><td><code id="glmm.score.meta_+3A_snp">SNP</code></td>
<td>

<p>a character vector of SNP column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep(&quot;SNP&quot;, length(files))).
</p>
</td></tr>
<tr><td><code id="glmm.score.meta_+3A_a1">A1</code></td>
<td>

<p>a character vector of allele 1 column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep(&quot;A1&quot;, length(files))). Note that <code>glmm.score.meta( )</code> does not define A1 as the effect allele or noneffect allele: it is the user's choice. However, the choice should be consistent across different studies, if A1 column is the effect allele in one study but the noneffect allele in another, meta-analysis results will be incorrect.
</p>
</td></tr>
<tr><td><code id="glmm.score.meta_+3A_a2">A2</code></td>
<td>

<p>a character vector of allele 2 column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep(&quot;A2&quot;, length(files))). Note that <code>glmm.score.meta( )</code> does not define A2 as the effect allele or noneffect allele: it is the user's choice. However, the choice should be consistent across different studies, if A2 column is the effect allele in one study but the noneffect allele in another, meta-analysis results will be incorrect.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing the following:
</p>
<table>
<tr><td><code>SNP</code></td>
<td>
<p>SNP name.</p>
</td></tr>
<tr><td><code>A1</code></td>
<td>
<p>allele 1.</p>
</td></tr>
<tr><td><code>A2</code></td>
<td>
<p>allele 2.</p>
</td></tr>
<tr><td><code>N</code></td>
<td>
<p>total sample size.</p>
</td></tr>
<tr><td><code>AF</code></td>
<td>
<p>effect allele frequency (user-defined: can be either allele 1 or allele 2).</p>
</td></tr>
<tr><td><code>SCORE</code></td>
<td>
<p>the summary score of the effect allele.</p>
</td></tr>
<tr><td><code>VAR</code></td>
<td>
<p>the variance of the summary score.</p>
</td></tr>
<tr><td><code>PVAL</code></td>
<td>
<p>meta-analysis p-value.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Han Chen
</p>


<h3>See Also</h3>

<p><code><a href="#topic+glmm.score">glmm.score</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
infile1 &lt;- system.file("extdata", "meta1.txt", package = "GMMAT")
infile2 &lt;- system.file("extdata", "meta2.txt", package = "GMMAT")
infile3 &lt;- system.file("extdata", "meta3.txt", package = "GMMAT")
outfile &lt;- tempfile()
glmm.score.meta(files = c(infile1, infile2, infile3), outfile = outfile,
	SNP = rep("SNP", 3), A1 = rep("A1", 3), A2 = rep("A2", 3))
unlink(outfile)

</code></pre>

<hr>
<h2 id='glmm.wald'>
Performing GLMM based Wald tests
</h2><span id='topic+glmm.wald'></span>

<h3>Description</h3>

<p>Fit a GLMM under the alternative hypothesis to perform Wald tests for association with genotypes in a plink .bed file (binary genotypes), a GDS file .gds, or a plain text file (or compressed .gz or .bz2 file).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmm.wald(fixed, data = parent.frame(), kins = NULL, id, random.slope = NULL, 
	groups = NULL, family = binomial(link = "logit"), infile, snps, 
	method = "REML", method.optim = "AI", maxiter = 500, tol = 1e-5, 
	taumin = 1e-5, taumax = 1e5, tauregion = 10, center = T, 
	select = NULL, missing.method = "impute2mean", infile.nrow = NULL, 
	infile.nrow.skip = 0, infile.sep = "\t", infile.na = "NA", 
	snp.col = 1, infile.ncol.skip = 1, infile.ncol.print = 1, 
	infile.header.print = "SNP", is.dosage = FALSE, verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmm.wald_+3A_fixed">fixed</code></td>
<td>

<p>an object of class <code><a href="stats.html#topic+formula">formula</a></code> (or one that can be coerced to that class): a symbolic description of the fixed effects model (without including any <code>snps</code> to be tested) to be fitted.
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_data">data</code></td>
<td>

<p>a data frame or list (or object coercible by <code><a href="base.html#topic+as.data.frame">as.data.frame</a></code> to a data frame) containing the variables in the model.
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_kins">kins</code></td>
<td>

<p>a known positive semi-definite relationship matrix (e.g. kinship matrix in genetic association studies) or a list of known positive semi-definite relationship matrices. The rownames and colnames of these matrices must at least include all samples as specified in the <code>id</code> column of the data frame <code>data</code>. If not provided, glmmkin will switch to the generalized linear model with no random effects (default = NULL).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_id">id</code></td>
<td>

<p>a column in the data frame <code>data</code>, indicating the id of samples. When there are duplicates in id, the data is assumed to be longitudinal with repeated measures.
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_random.slope">random.slope</code></td>
<td>

<p>an optional column indicating the random slope for time effect used in a mixed effects model for cross-sectional data with related individuals, and longitudinal data. It must be included in the names of <code>data</code>. There must be duplicates in <code>id</code> and <code>method.optim</code> must be &quot;AI&quot; (default = NULL).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_groups">groups</code></td>
<td>

<p>an optional categorical variable indicating the groups used in a heteroscedastic linear mixed model (allowing residual variances in different groups to be different). This variable must be included in the names of <code>data</code>, and <code>family</code> must be &quot;gaussian&quot; and <code>method.optim</code> must be &quot;AI&quot; (default = NULL).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_family">family</code></td>
<td>

<p>a description of the error distribution and link function to be used in the model. This can be a character string naming a family function, a family function or the result of a call to a family function. (See <code><a href="stats.html#topic+family">family</a></code> for details of family functions.)
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_infile">infile</code></td>
<td>

<p>the input file name. Note that for plink binary genotype files only the prefix without .bed, .bim or .fam should be used. Only SNP major mode recognized in the binary file. Alternatively, it can be the full name of a GDS file (including the suffix .gds) or a plain text file with some delimiters (comma, space, tab or something else), with one row for each SNP and one column for each individual. In that case, SNPs should be coded as numeric values (0/1/2 or dosages allowed, A/C/G/T coding is not recognized). There can be additional rows and columns to skip at the beginning. The order of individuals can be different from <code>obj</code> in the null GLMM (see the argument <code>select</code>). Some compressed files (.gz and .bz2) also allowed.
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_snps">snps</code></td>
<td>

<p>a vector of SNP names to be tested.
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_method">method</code></td>
<td>

<p>method of fitting the generalized linear mixed model. Either &quot;REML&quot; or &quot;ML&quot; (default = &quot;REML&quot;).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_method.optim">method.optim</code></td>
<td>

<p>optimization method of fitting the generalized linear mixed model. Either &quot;AI&quot;, &quot;Brent&quot; or &quot;Nelder-Mead&quot; (default = &quot;AI&quot;).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_maxiter">maxiter</code></td>
<td>

<p>a positive integer specifying the maximum number of iterations when fitting the generalized linear mixed model (default = 500).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_tol">tol</code></td>
<td>

<p>a positive number specifying tolerance, the difference threshold for parameter estimates below which iterations should be stopped. Also the threshold for determining monomorphism. If a SNP has value range less than the tolerance, it will be considered monomorphic and its association test p-value will be NA (default = 1e-5).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_taumin">taumin</code></td>
<td>

<p>the lower bound of search space for the variance component parameter <code class="reqn">\tau</code> (default = 1e-5), used when <code>method.optim = "Brent"</code>. See <code><a href="#topic+glmmkin">glmmkin</a></code>.
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_taumax">taumax</code></td>
<td>

<p>the upper bound of search space for the variance component parameter <code class="reqn">\tau</code> (default = 1e5), used when <code>method.optim = "Brent"</code>. See <code><a href="#topic+glmmkin">glmmkin</a></code>.
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_tauregion">tauregion</code></td>
<td>

<p>the number of search intervals for the REML or ML estimate of the variance component parameter <code class="reqn">\tau</code> (default = 10), used when <code>method.optim = "Brent"</code>. See <code><a href="#topic+glmmkin">glmmkin</a></code>.
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_center">center</code></td>
<td>

<p>a logical switch for centering genotypes before tests. If TRUE, genotypes will be centered to have mean 0 before tests, otherwise raw values will be directly used in tests (default = TRUE).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_select">select</code></td>
<td>

<p>an optional vector indicating the order of individuals in <code>infile</code>. If supplied, the length must match the number of individuals in <code>infile</code> (default = NULL). Individuals to be excluded should be coded 0. For example, <code>select = c(2, 3, 1, 0)</code> means the 1st individual in <code>infile</code> corresponds to the 2nd individual in <code>data</code>, the 2nd individual in <code>infile</code> corresponds to the 3rd individual in <code>data</code>, the 3rd individual in <code>infile</code> corresponds to the 1st individual in <code>data</code>, the 4th individual in <code>infile</code> is not included in <code>data</code>. If there are any duplicated <code>id</code> in <code>data</code> (longitudinal data analysis), indices in <code>select</code> should match the order of individuals with unique <code>id</code> in <code>data</code>. For plink binary genotype files and GDS files, this argument is not required and the sample ID's are automatically matched.
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_missing.method">missing.method</code></td>
<td>

<p>method of handling missing genotypes. Either &quot;impute2mean&quot; or &quot;omit&quot; (default = &quot;impute2mean&quot;).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_infile.nrow">infile.nrow</code></td>
<td>

<p>number of rows to read in <code>infile</code>, including number of rows to skip at the beginning. If NULL, the program will determine how many rows there are in <code>infile</code> automatically and read all rows (default = NULL). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_infile.nrow.skip">infile.nrow.skip</code></td>
<td>

<p>number of rows to skip at the beginning of <code>infile</code>. Must be nonnegative integers. Useful when header or comment lines are present (default = 0). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_infile.sep">infile.sep</code></td>
<td>

<p>delimiter in <code>infile</code> (default = &quot;\t&quot;). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_infile.na">infile.na</code></td>
<td>

<p>symbol in <code>infile</code> to denote missing genotypes (default = &quot;NA&quot;). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_snp.col">snp.col</code></td>
<td>

<p>a positive integer specifying which column in <code>infile</code> is SNP names. Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_infile.ncol.skip">infile.ncol.skip</code></td>
<td>

<p>number of columns to skip before genotype data in <code>infile</code>. These columns can be SNP name, alleles and/or quality measures and should be placed at the beginning in each line. After skipping these columns, the program will read in genotype data and perform Wald tests. Must be positive integers. It is recommended that SNP name should be included as the first column in <code>infile</code> and genotype data should start from the second column or later (default = 1). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_infile.ncol.print">infile.ncol.print</code></td>
<td>

<p>a vector indicating which column(s) in <code>infile</code> should be shown in the results. These columns can be SNP name, alleles and/or quality measures placed at the beginning in each line. Must be positive integers, no greater than <code>infile.ncol.skip</code> and sorted numerically in ascending order. By default, it is assumed that the first column is SNP name and genotype data start from the second column, and SNP name should be carried over to the results (default = 1). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_infile.header.print">infile.header.print</code></td>
<td>

<p>a character vector indicating column name(s) of column(s) selected to print by <code>infile.ncol.print</code> (default = &quot;SNP&quot;). Only used when <code>infile</code> is a plain text file (or compressed .gz or .bz2 file).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_is.dosage">is.dosage</code></td>
<td>

<p>a logical switch for whether imputed dosage should be used from a GDS <code>infile</code> (default = FALSE).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_verbose">verbose</code></td>
<td>

<p>a logical switch for printing a progress bar and detailed information (parameter estimates in each iteration) for testing and debugging purpose (default = FALSE).
</p>
</td></tr>
<tr><td><code id="glmm.wald_+3A_...">...</code></td>
<td>

<p>additional arguments that could be passed to <code><a href="stats.html#topic+glm">glm</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if <code>infile</code> is a plain text file, a data frame containing variables included in <code>infile.header.print</code> and the following:
</p>
<table>
<tr><td><code>N</code></td>
<td>
<p>number of individuals with non-missing genotypes for each SNP.</p>
</td></tr>
<tr><td><code>AF</code></td>
<td>
<p>effect allele frequency for each SNP.</p>
</td></tr>
<tr><td><code>BETA</code></td>
<td>
<p>effect size estimate for each SNP from the GLMM under the alternative hypothesis.</p>
</td></tr>
<tr><td><code>SE</code></td>
<td>
<p>standard error of the effect size estimate for each SNP.</p>
</td></tr>
<tr><td><code>PVAL</code></td>
<td>
<p>Wald test p-value for each SNP.</p>
</td></tr>
<tr><td><code>converged</code></td>
<td>
<p>a logical indicator for convergence for each SNP.</p>
</td></tr>
</table>
<p>if <code>infile</code> is the prefix of plink binary files (.bed, .bim and .fam), a data frame containing the following:
</p>
<table>
<tr><td><code>CHR</code></td>
<td>
<p>Chromosome, copied from .bim file.</p>
</td></tr>
<tr><td><code>SNP</code></td>
<td>
<p>SNP name, as supplied in <code>snps</code>.</p>
</td></tr>
<tr><td><code>cM</code></td>
<td>
<p>genetic location in centi Morgans, copied from .bim file.</p>
</td></tr>
<tr><td><code>POS</code></td>
<td>
<p>physical position in base pairs, copied from .bim file.</p>
</td></tr>
<tr><td><code>A1</code></td>
<td>
<p>allele 1, copied from .bim file.</p>
</td></tr>
<tr><td><code>A2</code></td>
<td>
<p>allele 2, copied from .bim file.</p>
</td></tr>
<tr><td><code>N</code></td>
<td>
<p>number of individuals with non-missing genotypes for each SNP.</p>
</td></tr>
<tr><td><code>AF</code></td>
<td>
<p>effect allele frequency for each SNP.</p>
</td></tr>
<tr><td><code>BETA</code></td>
<td>
<p>effect size estimate for each SNP from the GLMM under the alternative hypothesis.</p>
</td></tr>
<tr><td><code>SE</code></td>
<td>
<p>standard error of the effect size estimate for each SNP.</p>
</td></tr>
<tr><td><code>PVAL</code></td>
<td>
<p>Wald test p-value for each SNP.</p>
</td></tr>
<tr><td><code>converged</code></td>
<td>
<p>a logical indicator for convergence for each SNP.</p>
</td></tr>
</table>
<p>if <code>infile</code> is a GDS file (.gds), a data frame containing the following:
</p>
<table>
<tr><td><code>SNP</code></td>
<td>
<p>SNP name, as supplied in <code>snps</code>.</p>
</td></tr>
<tr><td><code>CHR</code></td>
<td>
<p>Chromosome, copied from .gds file.</p>
</td></tr>
<tr><td><code>POS</code></td>
<td>
<p>physical position in base pairs, copied from .gds file.</p>
</td></tr>
<tr><td><code>REF</code></td>
<td>
<p>reference allele, copied from .gds file.</p>
</td></tr>
<tr><td><code>ALT</code></td>
<td>
<p>alternate allele, copied from .gds file.</p>
</td></tr>
<tr><td><code>N</code></td>
<td>
<p>number of individuals with non-missing genotypes for each SNP.</p>
</td></tr>
<tr><td><code>AF</code></td>
<td>
<p>ALT allele frequency for each SNP.</p>
</td></tr>
<tr><td><code>BETA</code></td>
<td>
<p>effect size estimate for each SNP from the GLMM under the alternative hypothesis.</p>
</td></tr>
<tr><td><code>SE</code></td>
<td>
<p>standard error of the effect size estimate for each SNP.</p>
</td></tr>
<tr><td><code>PVAL</code></td>
<td>
<p>Wald test p-value for each SNP.</p>
</td></tr>
<tr><td><code>converged</code></td>
<td>
<p>a logical indicator for convergence for each SNP.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Han Chen, Matthew P. Conomos
</p>


<h3>References</h3>

<p>Brent, R.P. (1973) &quot;Chapter 4: An Algorithm with Guaranteed Convergence for Finding a Zero of a Function&quot;, Algorithms for Minimization without Derivatives, Englewood Cliffs, NJ: Prentice-Hall, ISBN 0-13-022335-2.
</p>
<p>Breslow, N.E. and Clayton, D.G. (1993) Approximate Inference in Generalized Linear Mixed Models. Journal of the American Statistical Association 88, 9-25.
</p>
<p>Chen, H., Wang, C., Conomos, M.P., Stilp, A.M., Li, Z., Sofer, T., Szpiro, A.A., Chen, W., Brehm, J.M., Celed√≥n, J.C., Redline, S., Papanicolaou, G.J., Thornton, T.A., Laurie, C.C., Rice, K. and Lin, X. (2016) Control forpopulation structure and relatedness for binary traits in genetic association studies via logistic mixed models. The American Journal of Human Genetics 98, 653-666.
</p>
<p>Gilmour, A.R., Thompson, R. and Cullis, B.R. (1995) Average Information REML: An Efficient Algorithm for Variance Parameter Estimation in Linear Mixed Models. Biometrics 51, 1440-1450.
</p>
<p>Nelder, J.A. and Mead, R. (1965) A simplex algorithm for function minimization. Computer Journal 7, 308-313.
</p>
<p>Yang, J., Lee, S.H., Goddard, M.E. and Visscher, P.M. (2011) GCTA: A Tool for Genome-wide Complex Trait Analysis. The American Journal of Human Genetics 88, 76-82.
</p>
<p>Zhou, X. and Stephens, M. (2012) Genome-wide efficient mixed-model analysis for association studies. Nature Genetics 44, 821-824.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+glmmkin">glmmkin</a></code>, <code><a href="#topic+glmm.score">glmm.score</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(example)
attach(example)
snps &lt;- c("SNP10", "SNP25", "SNP1", "SNP0")
plinkfiles &lt;- strsplit(system.file("extdata", "geno.bed", package = "GMMAT"), 
       ".bed", fixed = TRUE)[[1]]
glmm.wald(disease ~ age + sex, data = pheno, kins = GRM, id = "id",
	family = binomial(link = "logit"), infile = plinkfiles, snps = snps) 
if(requireNamespace("SeqArray", quietly = TRUE) &amp;&amp; requireNamespace("SeqVarTools",
        quietly = TRUE)) {
	infile &lt;- system.file("extdata", "geno.gds", package = "GMMAT")
	glmm.wald(disease ~ age + sex, data = pheno, kins = GRM, id = "id",
	        family = binomial(link = "logit"), infile = infile, snps = snps)
}
infile &lt;- system.file("extdata", "geno.txt", package = "GMMAT")
glmm.wald(disease ~ age + sex, data = pheno, kins = GRM, id = "id",
	family = binomial(link = "logit"), infile = infile, snps = snps, 
	infile.nrow.skip = 5, infile.ncol.skip = 3, infile.ncol.print = 1:3, 
	infile.header.print = c("SNP", "Allele1", "Allele2"))

</code></pre>

<hr>
<h2 id='glmmkin'>
Fit generalized linear mixed model with known relationship matrices
</h2><span id='topic+glmmkin'></span>

<h3>Description</h3>

<p>Fit a generalized linear mixed model with a random intercept, or a random intercept and an optional random slope of time effect for longitudinal data. The covariance matrix of the random intercept is proportional to a known relationship matrix (e.g. kinship matrix in genetic association studies). Alternatively, it can be a variance components model with multiple random effects, and each component has a known relationship matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmmkin(fixed, data = parent.frame(), kins = NULL, id, random.slope = NULL, 
	groups = NULL, family = binomial(link = "logit"), method = "REML", 
	method.optim = "AI", maxiter = 500, tol = 1e-5, taumin = 1e-5, 
	taumax = 1e5, tauregion = 10, verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmmkin_+3A_fixed">fixed</code></td>
<td>

<p>an object of class <code><a href="stats.html#topic+formula">formula</a></code> (or one that can be coerced to that class): a symbolic description of the fixed effects model to be fitted.
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_data">data</code></td>
<td>

<p>a data frame or list (or object coercible by <code><a href="base.html#topic+as.data.frame">as.data.frame</a></code> to a data frame) containing the variables in the model.
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_kins">kins</code></td>
<td>

<p>a known positive semi-definite relationship matrix (e.g. kinship matrix in genetic association studies) or a list of known positive semi-definite relationship matrices. The rownames and colnames of these matrices must at least include all samples as specified in the <code>id</code> column of the data frame <code>data</code>. If not provided, glmmkin will switch to the generalized linear model with no random effects (default = NULL).
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_id">id</code></td>
<td>

<p>a column in the data frame <code>data</code>, indicating the id of samples. When there are duplicates in <code>id</code>, the data is assumed to be longitudinal with repeated measures.
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_random.slope">random.slope</code></td>
<td>

<p>an optional column indicating the random slope for time effect used in a mixed effects model for cross-sectional data with related individuals, and longitudinal data. It must be included in the names of <code>data</code>. There must be duplicates in <code>id</code> and <code>method.optim</code> must be &quot;AI&quot; (default = NULL).
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_groups">groups</code></td>
<td>

<p>an optional categorical variable indicating the groups used in a heteroscedastic linear mixed model (allowing residual variances in different groups to be different). This variable must be included in the names of <code>data</code>, and <code>family</code> must be &quot;gaussian&quot; and <code>method.optim</code> must be &quot;AI&quot; (default = NULL).
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_family">family</code></td>
<td>

<p>a description of the error distribution and link function to be used in the model. This can be a character string naming a family function, a family function or the result of a call to a family function. (See <code><a href="stats.html#topic+family">family</a></code> for details of family functions.)
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_method">method</code></td>
<td>

<p>method of fitting the generalized linear mixed model. Either &quot;REML&quot; or &quot;ML&quot; (default = &quot;REML&quot;).
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_method.optim">method.optim</code></td>
<td>

<p>optimization method of fitting the generalized linear mixed model. Either &quot;AI&quot;, &quot;Brent&quot; or &quot;Nelder-Mead&quot; (default = &quot;AI&quot;).
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_maxiter">maxiter</code></td>
<td>

<p>a positive integer specifying the maximum number of iterations when fitting the generalized linear mixed model (default = 500).
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_tol">tol</code></td>
<td>

<p>a positive number specifying tolerance, the difference threshold for parameter estimates below which iterations should be stopped (default = 1e-5).
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_taumin">taumin</code></td>
<td>

<p>the lower bound of search space for the variance component parameter <code class="reqn">\tau</code> (default = 1e-5), used when <code>method.optim = "Brent"</code>. See Details.
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_taumax">taumax</code></td>
<td>

<p>the upper bound of search space for the variance component parameter <code class="reqn">\tau</code> (default = 1e5), used when <code>method.optim = "Brent"</code>. See Details.
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_tauregion">tauregion</code></td>
<td>

<p>the number of search intervals for the REML or ML estimate of the variance component parameter <code class="reqn">\tau</code> (default = 10), used when <code>method.optim = "Brent"</code>. See Details.
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_verbose">verbose</code></td>
<td>

<p>a logical switch for printing detailed information (parameter estimates in each iteration) for testing and debugging purpose (default = FALSE).
</p>
</td></tr>
<tr><td><code id="glmmkin_+3A_...">...</code></td>
<td>

<p>additional arguments that could be passed to <code><a href="stats.html#topic+glm">glm</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Generalized linear mixed models (GLMM) are fitted using the penalized quasi-likelihood (PQL) method proposed by Breslow and Clayton (1993). Generally, fitting a GLMM is computationally expensive, and by default we use the Average Information REML algorithm (Gilmour, Thompson and Cullis, 1995; Yang et al., 2011) to fit the model. If only one relationship matrix is specified (<code>kins</code> is a matrix), iterations may be accelerated using the algorithm proposed by Zhou and Stephens (2012) for linear mixed models. An eigendecomposition is performed in each outer iteration and the estimate of the variance component parameter <code class="reqn">\tau</code> is obtained by maximizing the profiled log restricted likelihood (or likelihood) in a search space from <code>taumin</code> to <code>taumax</code>, equally divided into <code>tauregion</code> intervals on the log scale, using Brent's method (1973). If <code>kins</code> is a list of matrices and <code>method = "Nelder-Mead"</code>, iterations are performed as a multi-dimensional maximization problem solved by Nelder and Mead's method (1965). It can be very slow, and we do not recommend using this method unless the likelihood function is badly behaved. Both Brent's method and Nelder and Mead's method are derivative-free. When the Average Information REML algorithm fails to converge, a warning message is given and the algorithm is default to derivative-free approaches: Brent's method if only one relationship matrix is specified, Nelder and Mead's method if more than one relationship matrix is specified.
</p>
<p>For longitudinal data (with duplicated <code>id</code>), two types of models can be applied: random intercept only models, and random intercept and random slope models. The random intercept only model is appropriate for analyzing repeated measures with no time trends, and observations for the same individual are assumed to be exchangeable. The random intercept and random slope model is appropriate for analyzing longitudinal data with individual-specific time trends (therefore, a random slope for time effect). Typically, the time effect should be included in the model as a fixed effect covariate as well. Covariances of the random intercept and the random slope are estimated.
</p>
<p>For multiple phenotype analysis, <code><a href="stats.html#topic+formula">formula</a></code> recognized by <code><a href="stats.html#topic+lm">lm</a></code>, such as <code>cbind(y1, y2, y3) ~ x1 + x2</code>, can be used in <code>fixed</code> as fixed effects. For each matrix in <code>kins</code>, variance components corresponding to each phenotype, as well as their covariance components, will be estimated. Currently, <code>family</code> must be &quot;gaussian&quot; and <code>method.optim</code> must be &quot;AI&quot;.
</p>


<h3>Value</h3>

<table>
<tr><td><code>theta</code></td>
<td>
<p>a vector or a list of variance component parameter estimates. See below.
</p>
<p>For cross-sectional data, if <code>kins</code> is not provided (unrelated individuals), <code>theta</code> is the dispersion parameter estimate from the generalized linear model; if <code>kins</code> is a matrix and <code>groups</code> is not provided, <code>theta</code> is a length 2 vector, with <code>theta[1]</code> being the dispersion parameter estimate and <code>theta[2]</code> being the variance component parameter estimate for <code>kins</code>; if <code>kins</code> is a list and <code>groups</code> is not provided, <code>theta</code> is a length <code>1 + length(kins)</code> vector, with <code>theta[1]</code> being the dispersion parameter estimate and <code>theta[2:(1 + length(kins))]</code> being the variance component parameter estimates, corresponding to the order of matrices in the list <code>kins</code>; if <code>kins</code> is a matrix and <code>groups</code> is provided (a heteroscedastic linear mixed model with <code>n.groups</code> residual variance groups), <code>theta</code> is a length <code>1 + n.groups</code> vector, with <code>theta[1:n.groups]</code> being the residual variance estimates for each group and <code>theta[1 + n.groups]</code> being the variance component parameter estimate for <code>kins</code>; if <code>kins</code> is a list and <code>groups</code> is provided (a heteroscedastic linear mixed model with <code>n.groups</code> residual variance groups), <code>theta</code> is a length <code>length(kins) + n.groups</code> vector, with <code>theta[1:n.groups]</code> being the residual variance estimates for each group and <code>theta[(1 + n.groups):(length(kins) + n.groups)]</code> being the variance component parameter estimates, corresponding to the order of matrices in the list <code>kins</code>. 
</p>
<p>For longitudinal data (with duplicated <code>id</code>) in a random intercept only model, if <code>kins</code> is not provided (unrelated individuals) and <code>groups</code> is not provided, <code>theta</code> is a length 2 vector, with <code>theta[1]</code> being the dispersion parameter estimate and <code>theta[2]</code> being the variance component parameter estimate for the random individual effects; if <code>kins</code> is a matrix and <code>groups</code> is not provided, <code>theta</code> is a length 3 vector, with <code>theta[1]</code> being the dispersion parameter estimate, <code>theta[2]</code> being the variance component parameter estimate for the random individual effects attributable to relatedness from <code>kins</code>, and <code>theta[3]</code> being the variance component parameter estimate for the random individual effects not attributable to relatedness from <code>kins</code>; if <code>kins</code> is a list and <code>groups</code> is not provided, <code>theta</code> is a length <code>2 + length(kins)</code> vector, with <code>theta[1]</code> being the dispersion parameter estimate, <code>theta[2:(1 + length(kins))]</code> being the variance component parameter estimates for the random individual effects attributable to relatedness from <code>kins</code>, corresponding to the order of matrices in the list <code>kins</code>, and <code>theta[2 + length(kins)]</code> being the variance component parameter estimate for the random individual effects not attributable to relatedness from <code>kins</code>; if <code>kins</code> is not provided (unrelated individuals) and <code>groups</code> is provided (a heteroscedastic linear mixed model with <code>n.groups</code> residual variance groups), <code>theta</code> is a length <code>1 + n.groups</code> vector, with <code>theta[1:n.groups]</code> being the residual variance estimates for each group and <code>theta[1 + n.groups]</code> being the variance component parameter estimate for the random individual effects; if <code>kins</code> is a matrix and <code>groups</code> is provided (a heteroscedastic linear mixed model with <code>n.groups</code> residual variance groups), <code>theta</code> is a length <code>2 + n.groups</code> vector, with <code>theta[1:n.groups]</code> being the residual variance estimates for each group, <code>theta[1 + n.groups]</code> being the variance component parameter estimate for the random individual effects attributable to relatedness from <code>kins</code>, and <code>theta[2 + n.groups]</code> being the variance component parameter estimate for the random individual effects not attributable to relatedness from <code>kins</code>; if <code>kins</code> is a list and <code>groups</code> is provided (a heteroscedastic linear mixed model with <code>n.groups</code> residual variance groups), <code>theta</code> is a length <code>1 + length(kins) + n.groups</code> vector, with <code>theta[1:n.groups]</code> being the residual variance estimates for each group, <code>theta[(1 + n.groups):(length(kins) + n.groups)]</code> being the variance component parameter estimates for the random individual effects attributable to relatedness from <code>kins</code>, corresponding to the order of matrices in the list <code>kins</code>, and <code>theta[1 + length(kins) + n.groups]</code> being the variance component parameter estimate for the random individual effects not attributable to relatedness from <code>kins</code>.
</p>
<p>For longitudinal data (with duplicated <code>id</code>) in a random intercept and random slope (for time effect) model, if <code>kins</code> is not provided (unrelated individuals) and <code>groups</code> is not provided, <code>theta</code> is a length 4 vector, with <code>theta[1]</code> being the dispersion parameter estimate, <code>theta[2]</code> being the variance component parameter estimate for the random individual effects of the intercept, <code>theta[3]</code> being the covariance estimate for the random individual effects of the intercept and the random individual effects of the time slope, and <code>theta[4]</code> being the variance component parameter estimate for the random individual effects of the time slope; if <code>kins</code> is a matrix and <code>groups</code> is not provided, <code>theta</code> is a length 7 vector, with <code>theta[1]</code> being the dispersion parameter estimate, <code>theta[2]</code> being the variance component parameter estimate for the random individual effects of the intercept attributable to relatedness from <code>kins</code>, <code>theta[3]</code> being the variance component parameter estimate for the random individual effects of the intercept not attributable to relatedness from <code>kins</code>, <code>theta[4]</code> being the covariance estimate for the random individual effects of the intercept and the random individual effects of the time slope attributable to relatedness from <code>kins</code>, <code>theta[5]</code> being the covariance estimate for the random individual effects of the intercept and the random individual effects of the time slope not attributable to relatedness from <code>kins</code>, <code>theta[6]</code> being the variance component parameter estimate for the random individual effects of the time slope attributable to relatedness from <code>kins</code>, and <code>theta[7]</code> being the variance component parameter estimate for the random individual effects of the time slope not attributable to relatedness from <code>kins</code>; if <code>kins</code> is a list and <code>groups</code> is not provided, <code>theta</code> is a length <code>4 + 3 * length(kins)</code> vector, with <code>theta[1]</code> being the dispersion parameter estimate, <code>theta[2:(1 + length(kins))]</code> being the variance component parameter estimates for the random individual effects of the intercept attributable to relatedness from <code>kins</code>, corresponding to the order of matrices in the list <code>kins</code>, <code>theta[2 + length(kins)]</code> being the variance component parameter estimate for the random individual effects of the intercept not attributable to relatedness from <code>kins</code>, <code>theta[(3 + length(kins)):(2 + 2 * length(kins))]</code> being the covariance estimates for the random individual effects of the intercept and the random individual effects of the time slope attributable to relatedness from <code>kins</code>, corresponding to the order of matrices in the list <code>kins</code>, <code>theta[3 + 2 * length(kins)]</code> being the covariance estimate for the random individual effects of the intercept and the random individual effects of the time slope not attributable to relatedness from <code>kins</code>, <code>theta[(4 + 2 * length(kins)):(3 + 3 * length(kins))]</code> being the variance component parameter estimates for the random individual effects of the time slope attributable to relatedness from <code>kins</code>, corresponding to the order of matrices in the list <code>kins</code>, <code>theta[4 + 3 * length(kins)]</code> being the variance component parameter estimate for the random individual effects of the time slope not attributable to relatedness from <code>kins</code>; if <code>kins</code> is not provided (unrelated individuals) and <code>groups</code> is provided (a heteroscedastic linear mixed model with <code>n.groups</code> residual variance groups), <code>theta</code> is a length <code>3 + n.groups</code> vector, with <code>theta[1:n.groups]</code> being the residual variance estimates for each group, <code>theta[1 + n.groups]</code> being the variance component parameter estimate for the random individual effects of the intercept, <code>theta[2 + n.groups]</code> being the covariance estimate for the random individual effect of the intercept and the random individual effects of the time slope, and <code>theta[3 + n.groups]</code> being the variance component parameter estimate for the random individual effects of the time slope; if <code>kins</code> is a matrix and <code>groups</code> is provided (a heteroscedastic linear mixed model with <code>n.groups</code> residual variance groups), <code>theta</code> is a length <code>6 + n.groups</code> vector, with <code>theta[1:n.groups]</code> being the residual variance estimates for each group, <code>theta[1 + n.groups]</code> being the variance component parameter estimate for the random individual effects of the intercept attributable to relatedness from <code>kins</code>, <code>theta[2 + n.groups]</code> being the variance component parameter estimate for the random individual effects of the intercept not attributable to relatedness from <code>kins</code>, <code>theta[3 + n.groups]</code> being the covariance estimate for the random individual effects of the intercept and the random individual effects of the time slope attributable to relatedness from <code>kins</code>, <code>theta[4 + n.groups]</code> being the covariance estimate for the random individual effects of the intercept and the random individual effects of the time slope not attributable to relatedness from <code>kins</code>, <code>theta[5 + n.groups]</code> being the variance component parameter estimate for the random individual effects of the time slope attributable to relatedness from <code>kins</code>, and <code>theta[6 + n.groups]</code> being the variance component parameter estimate for the random individual effects of the time slope not attributable to relatedness from <code>kins</code>; if <code>kins</code> is a list and <code>groups</code> is provided (a heteroscedastic linear mixed model with <code>n.groups</code> residual variance groups), <code>theta</code> is a length <code>3 + 3 * length(kins) + n.groups</code> vector, with <code>theta[1:n.groups]</code> being the residual variance estimates for each group, <code>theta[(1 + n.groups):(length(kins) + n.groups)]</code> being the variance component parameter estimates for the random individual effects of the intercept attributable to relatedness from <code>kins</code>, corresponding to the order of matrices in the list <code>kins</code>, <code>theta[1 + length(kins) + n.groups]</code> being the variance component parameter estimate for the random individual effects of the intercept not attributable to relatedness from <code>kins</code>, <code>theta[(2 + length(kins) + n.groups):(1 + 2 * length(kins) + n.groups)]</code> being the covariance estimates for the random individual effects of the intercept and the random individual effects of the time slope attributable to relatedness from <code>kins</code>, corresponding to the order of matrices in the list <code>kins</code>, <code>theta[2 + 2 * length(kins) + n.groups]</code> being the covariance estimate for the random individual effects of the intercept and the random individual effects of the time slope not attributable to relatedness from <code>kins</code>, <code>theta[(3 + 2 * length(kins) + n.groups):(2 + 3 * length(kins) + n.groups)]</code> being the variance component parameter estimates for the random individual effects of the time slope attributable to relatedness from <code>kins</code>, corresponding to the order of matrices in the list <code>kins</code>, and <code>theta[3 + 3 * length(kins) + n.groups]</code> being the variance component parameter estimate for the random individual effects of the time slope not attributable to relatedness from <code>kins</code>.
</p>
<p>For multiple phenotype analysis, <code>theta</code> is a list of variance-covariance matrices. If <code>kins</code> is not provided (unrelated individuals), <code>theta</code> is an <code>n.pheno</code> by <code>n.pheno</code> variance-covariance matrix for the residuals of the multiple phenotypes from the linear model; if <code>kins</code> is a matrix and <code>groups</code> is not provided, <code>theta</code> is a length 2 list, with <code>theta[[1]]</code> being the variance-covariance matrix for the residuals and <code>theta[[2]]</code> being the variance-covariance matrix for <code>kins</code>; if <code>kins</code> is a list and <code>groups</code> is not provided, <code>theta</code> is a length <code>1 + length(kins)</code> list, with <code>theta[[1]]</code> being the variance-covariance matrix for the residuals and <code>theta[[2]]</code> to <code>theta[[1 + length(kins)]]</code> being the variance-covariance matrices, corresponding to the order of matrices in the list <code>kins</code>; if <code>kins</code> is a matrix and <code>groups</code> is provided (a heteroscedastic linear mixed model with <code>n.groups</code> residual variance groups), <code>theta</code> is a length <code>1 + n.groups</code> list, with <code>theta[[1]]</code> to <code>theta[[n.groups]]</code> being the variance-covariance matrices for the residuals in each group and <code>theta[[1 + n.groups]]</code> being the variance-covariance matrix for <code>kins</code>; if <code>kins</code> is a list and <code>groups</code> is provided (a heteroscedastic linear mixed model with <code>n.groups</code> residual variance groups), <code>theta</code> is a length <code>length(kins) + n.groups</code> list, with <code>theta[[1]]</code> to <code>theta[[n.groups]]</code> being the variance-covariance matrices for the residuals in each group and <code>theta[[1 + n.groups]]</code> to <code>theta[[length(kins) + n.groups]]</code> being the variance-covariance matrices, corresponding to the order of matrices in the list <code>kins</code>.</p>
</td></tr>
<tr><td><code>n.pheno</code></td>
<td>
<p>an integer indicating the number of phenotypes in multiple phenotype analysis (for single phenotype analysis, <code>n.pheno = 1</code>).</p>
</td></tr>
<tr><td><code>n.groups</code></td>
<td>
<p>an integer indicating the number of distinct residual variance groups in heteroscedastic linear mixed models (for other models, <code>n.groups = 1</code>).</p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p>a vector or a matrix for the fixed effects parameter estimates (including the intercept).</p>
</td></tr>
<tr><td><code>linear.predictors</code></td>
<td>
<p>a vector or a matrix for the linear predictors.</p>
</td></tr>
<tr><td><code>fitted.values</code></td>
<td>
<p>a vector or a matrix for the fitted mean values on the original scale.</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>a vector or a matrix for the final working vector.</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>model matrix for the fixed effects.</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>the projection matrix with dimensions equal to the sample size multiplied by <code>n.pheno</code>. Used in <code><a href="#topic+glmm.score">glmm.score</a></code> and <code><a href="#topic+SMMAT">SMMAT</a></code> for dense matrices.</p>
</td></tr>
<tr><td><code>residuals</code></td>
<td>
<p>a vector or a matrix for the residuals on the original scale. NOT rescaled by the dispersion parameter.</p>
</td></tr>
<tr><td><code>scaled.residuals</code></td>
<td>
<p>a vector or a matrix for the scaled residuals, calculated as the original residuals divided by the dispersion parameter (in heteroscedastic linear mixed models, corresponding residual variance estimates by each group).</p>
</td></tr>
<tr><td><code>cov</code></td>
<td>
<p>covariance matrix for the fixed effects (including the intercept).</p>
</td></tr>
<tr><td><code>Sigma_i</code></td>
<td>
<p>the inverse of the estimated covariance matrix for samples, with dimensions equal to the sample size multiplied by <code>n.pheno</code>. Used in <code><a href="#topic+glmm.score">glmm.score</a></code> and <code><a href="#topic+SMMAT">SMMAT</a></code> for sparse matrices.</p>
</td></tr>
<tr><td><code>Sigma_iX</code></td>
<td>
<p>Sigma_i multiplied by X. Used in <code><a href="#topic+glmm.score">glmm.score</a></code> and <code><a href="#topic+SMMAT">SMMAT</a></code> for sparse matrices.</p>
</td></tr>
<tr><td><code>converged</code></td>
<td>
<p>a logical indicator for convergence.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>the matched call.</p>
</td></tr>
<tr><td><code>id_include</code></td>
<td>
<p>a vector indicating the <code>id</code> of rows in <code>data</code> with nonmissing outcome and covariates, thus are included in the model fit.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Han Chen, Matthew P. Conomos
</p>


<h3>References</h3>

<p>Brent, R.P. (1973) &quot;Chapter 4: An Algorithm with Guaranteed Convergence for Finding a Zero of a Function&quot;, Algorithms for Minimization without Derivatives, Englewood Cliffs, NJ: Prentice-Hall, ISBN 0-13-022335-2.
</p>
<p>Breslow, N.E. and Clayton, D.G. (1993) Approximate Inference in Generalized Linear Mixed Models. Journal of the American Statistical Association 88, 9-25.
</p>
<p>Chen, H., Wang, C., Conomos, M.P., Stilp, A.M., Li, Z., Sofer, T., Szpiro, A.A., Chen, W., Brehm, J.M., Celed√≥n, J.C., Redline, S., Papanicolaou, G.J., Thornton, T.A., Laurie, C.C., Rice, K. and Lin, X. (2016) Control for population structure and relatedness for binary traits in genetic association studies via logistic mixed models. The American Journal of Human Genetics 98, 653-666.
</p>
<p>Gilmour, A.R., Thompson, R. and Cullis, B.R. (1995) Average Information REML: An Efficient Algorithm for Variance Parameter Estimation in Linear Mixed Models. Biometrics 51, 1440-1450.
</p>
<p>Nelder, J.A. and Mead, R. (1965) A simplex algorithm for function minimization. Computer Journal 7, 308-313.
</p>
<p>Yang, J., Lee, S.H., Goddard, M.E. and Visscher, P.M. (2011) GCTA: A Tool for Genome-wide Complex Trait Analysis. The American Journal of Human Genetics 88, 76-82.
</p>
<p>Zhou, X. and Stephens, M. (2012) Genome-wide efficient mixed-model analysis for association studies. Nature Genetics 44, 821-824.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example)
attach(example)
model0 &lt;- glmmkin(disease ~ age + sex, data = pheno, kins = GRM, id = "id",
       family = binomial(link = "logit"))
model0$theta
model0$coefficients
model0$cov

model1 &lt;- glmmkin(y.repeated ~ sex, data = pheno2, kins = GRM, id = "id", 
       family = gaussian(link = "identity"))
model1$theta
model1$coefficients
model1$cov
model2 &lt;- glmmkin(y.trend ~ sex + time, data = pheno2, kins = GRM, id = "id", 
       random.slope = "time", family = gaussian(link = "identity"))
model2$theta
model2$coefficients
model2$cov

</code></pre>

<hr>
<h2 id='SMMAT'>
Variant Set Mixed Model Association Tests (SMMAT)
</h2><span id='topic+SMMAT'></span><span id='topic+SMMAT.prep'></span><span id='topic+SMMAT.lowmem'></span>

<h3>Description</h3>

<p>Variant Set Mixed Model Association Tests (SMMAT-B, SMMAT-S, SMMAT-O and SMMAT-E) for multiple user-defined test units and a null generalized linear mixed model.
<code>SMMAT.prep</code> and <code>SMMAT.lowmem</code> are the two-step low-memory version of <code>SMMAT</code>. <code>SMMAT.lowmem</code> takes the returned R object from <code>SMMAT.prep</code> and uses less memory (if the returned R object from <code>SMMAT.prep</code> is saved to an R data file, the R session is terminated, and this R object is loaded into a new R session for running <code>SMMAT.lowmem</code>), especially when <code>group.file</code> contains only a subset of variants from <code>geno.file</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SMMAT(null.obj, geno.file, group.file, group.file.sep = "\t", 
	meta.file.prefix = NULL, MAF.range = c(1e-7, 0.5), 
	MAF.weights.beta = c(1, 25), miss.cutoff = 1, 
	missing.method = "impute2mean", method = "davies", 
	tests = "E", rho = c(0, 0.1^2, 0.2^2, 0.3^2, 0.4^2, 
	0.5^2, 0.5, 1), use.minor.allele = FALSE, 
	auto.flip = FALSE, Garbage.Collection = FALSE,
	is.dosage = FALSE, ncores = 1, verbose = FALSE)
SMMAT.prep(null.obj, geno.file, group.file, group.file.sep = "\t", 
	auto.flip = FALSE)
SMMAT.lowmem(SMMAT.prep.obj, geno.file = NULL, meta.file.prefix = NULL,
        MAF.range = c(1e-7, 0.5), MAF.weights.beta = c(1, 25),
	miss.cutoff = 1, missing.method = "impute2mean",
	method = "davies", tests = "E", rho = c(0, 0.1^2,
	0.2^2, 0.3^2, 0.4^2, 0.5^2, 0.5, 1),
	use.minor.allele = FALSE, Garbage.Collection = FALSE,
	is.dosage = FALSE, ncores = 1, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SMMAT_+3A_null.obj">null.obj</code></td>
<td>

<p>a class glmmkin or a class glmmkin.multi object, returned by fitting the null GLMM using <code>glmmkin</code>.
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_geno.file">geno.file</code></td>
<td>

<p>the .gds file name or an object of class SeqVarGDSClass for the full genotypes. The <code>sample.id</code> in <code>geno.file</code> should overlap <code>id_include</code> in <code>null.obj</code>. It is recommended that <code>sample.id</code> in <code>geno.file</code> include the full samples (at least all samples as specified in <code>id_include</code> of <code>null.obj</code>). It is not necessary for the user to take a subset of <code>geno.file</code> before running the analysis. If <code>geno.file</code> is an object of class SeqVarGDSClass, the .gds file will be closed upon successful completion of the function.
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_group.file">group.file</code></td>
<td>

<p>a plain text file with 6 columns defining the test units. There should be no headers in the file, and the columns are group name, chromosome, position, reference allele, alternative allele and weight, respectively.
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_group.file.sep">group.file.sep</code></td>
<td>

<p>the delimiter in group.file (default = &quot;\t&quot;).
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_meta.file.prefix">meta.file.prefix</code></td>
<td>

<p>prefix of intermediate files (.score.* and .var.*) required in a meta-analysis. If NULL, such intermediate files are not generated (default = NULL).
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_maf.range">MAF.range</code></td>
<td>

<p>a numeric vector of length 2 defining the minimum and maximum minor allele frequencies of variants that should be included in the analysis (default = c(1e-7, 0.5)).
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_maf.weights.beta">MAF.weights.beta</code></td>
<td>

<p>a numeric vector of length 2 defining the beta probability density function parameters on the minor allele frequencies. This internal minor allele frequency weight is multiplied by the external weight given by the group.file. To turn off internal minor allele frequency weight and only use the external weight given by the group.file, use c(1, 1) to assign flat weights (default = c(1, 25)).
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_miss.cutoff">miss.cutoff</code></td>
<td>

<p>the maximum missing rate allowed for a variant to be included (default = 1, including all variants).
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_missing.method">missing.method</code></td>
<td>

<p>method of handling missing genotypes. Either &quot;impute2mean&quot; or &quot;impute2zero&quot; (default = &quot;impute2mean&quot;).
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_method">method</code></td>
<td>

<p>a method to compute p-values for SKAT-type test statistics (default = &quot;davies&quot;). &quot;davies&quot; represents an exact method that computes a p-value by inverting the characteristic function of the mixture chisq distribution, with an accuracy of 1e-6. When &quot;davies&quot; p-value is less than 1e-5, it defaults to method &quot;kuonen&quot;. &quot;kuonen&quot; represents a saddlepoint approximation method that computes the tail probabilities of the mixture chisq distribution. When &quot;kuonen&quot; fails to compute a p-value, it defaults to method &quot;liu&quot;. &quot;liu&quot; is a moment-matching approximation method for the mixture chisq distribution.
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_tests">tests</code></td>
<td>

<p>a character vector indicating which SMMAT tests should be performed (&quot;B&quot; for the burden test, &quot;S&quot; for SKAT, &quot;O&quot; for SKAT-O and &quot;E&quot; for the efficient hybrid test of the burden test and SKAT). The burden test and SKAT are automatically included when performing &quot;O&quot;, and the burden test is automatically included when performing &quot;E&quot; (default = &quot;E&quot;).
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_rho">rho</code></td>
<td>

<p>a numeric vector defining the search grid used in SMMAT-O for SKAT-O (see the SKAT-O paper for details). Not used for SMMAT-B for the burden test, SMMAT-S for SKAT or SMMAT-E for the efficient hybrid test of the burden test and SKAT (default = c(0, 0.1^2, 0.2^2, 0.3^2, 0.4^2, 0.5^2, 0.5, 1)).
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_use.minor.allele">use.minor.allele</code></td>
<td>

<p>a logical switch for whether to use the minor allele (instead of the alt allele) as the coding allele (default = FALSE). It does not change SMMAT-S results, but SMMAT-B (as well as SMMAT-O and SMMAT-E) will be affected. Along with the MAF filter, this option is useful for combining rare mutations, assuming rare allele effects are in the same direction.
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_auto.flip">auto.flip</code></td>
<td>

<p>a logical switch for whether to enable automatic allele flipping if a variant with alleles ref/alt is not found at a position, but a variant at the same position with alleles alt/ref is found (default = FALSE). Use with caution for whole genome sequence data, as both ref/alt and alt/ref variants at the same position are not uncommon, and they are likely two different variants, rather than allele flipping.
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_garbage.collection">Garbage.Collection</code></td>
<td>

<p>a logical switch for whether to enable garbage collection in each test (default = FALSE). Pay for memory efficiency with slower computation speed.
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_is.dosage">is.dosage</code></td>
<td>

<p>a logical switch for whether imputed dosage should be used from <code>geno.file</code> (default = FALSE).
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_ncores">ncores</code></td>
<td>

<p>a positive integer indicating the number of cores to be used in parallel computing (default = 1).
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_verbose">verbose</code></td>
<td>

<p>a logical switch for whether a progress bar should be shown (default = FALSE).
</p>
</td></tr>
<tr><td><code id="SMMAT_+3A_smmat.prep.obj">SMMAT.prep.obj</code></td>
<td>

<p>a class SMMAT.prep object, returned by <code>SMMAT.prep</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>SMMAT</code> and <code>SMMAT.lowmem</code> return a data frame with the following components:
</p>
<table>
<tr><td><code>group</code></td>
<td>

<p>name of the test unit group.
</p>
</td></tr>
<tr><td><code>n.variants</code></td>
<td>

<p>number of variants in the test unit group that pass the missing rate and allele frequency filters.
</p>
</td></tr>
<tr><td><code>miss.min</code></td>
<td>

<p>minimum missing rate for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>miss.mean</code></td>
<td>

<p>mean missing rate for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>miss.max</code></td>
<td>

<p>maximum missing rate for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>freq.min</code></td>
<td>

<p>minimum coding allele frequency for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>freq.mean</code></td>
<td>

<p>mean coding allele frequency for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>freq.max</code></td>
<td>

<p>maximum coding allele frequency for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>B.score</code></td>
<td>

<p>burden test score statistic.
</p>
</td></tr>
<tr><td><code>B.var</code></td>
<td>

<p>variance of burden test score statistic.
</p>
</td></tr>
<tr><td><code>B.pval</code></td>
<td>

<p>burden test p-value.
</p>
</td></tr>
<tr><td><code>S.pval</code></td>
<td>

<p>SKAT p-value.
</p>
</td></tr>
<tr><td><code>O.pval</code></td>
<td>

<p>SKAT-O p-value.
</p>
</td></tr>
<tr><td><code>O.minp</code></td>
<td>

<p>minimum p-value in the SKAT-O search grid.
</p>
</td></tr>
<tr><td><code>O.minp.rho</code></td>
<td>

<p>rho value at the minimum p-value in the SKAT-O search grid.
</p>
</td></tr>
<tr><td><code>E.pval</code></td>
<td>

<p>SMMAT efficient hybrid test of the burden test and SKAT p-value.
</p>
</td></tr>
</table>
<p><code>SMMAT.prep</code> return a list with the following components:
</p>
<table>
<tr><td><code>null.obj</code></td>
<td>

<p>a class glmmkin or a class glmmkin.multi object from the null model, after pre-processing.
</p>
</td></tr>
<tr><td><code>geno.file</code></td>
<td>

<p>the name of the .gds file for the full genotypes.
</p>
</td></tr>
<tr><td><code>group.file</code></td>
<td>

<p>the name of the plain text file with 6 columns defining the test units.
</p>
</td></tr>
<tr><td><code>group.file.sep</code></td>
<td>

<p>the delimiter in group.file.
</p>
</td></tr>
<tr><td><code>auto.flip</code></td>
<td>

<p>a logical indicator showing whether automatic allele flipping is enabled in pre-processing if a variant with alleles ref/alt is not found at a position, but a variant at the same position with alleles alt/ref is found.
</p>
</td></tr>
<tr><td><code>residuals</code></td>
<td>

<p>residuals from the null model, after pre-processing.
</p>
</td></tr>
<tr><td><code>sample.id</code></td>
<td>

<p>sample.id from geno.file, after pre-processing.
</p>
</td></tr>
<tr><td><code>group.info</code></td>
<td>

<p>group.info read from group.file, after pre-processing.
</p>
</td></tr>
<tr><td><code>groups</code></td>
<td>

<p>unique groups in group.info, after pre-processing.
</p>
</td></tr>
<tr><td><code>group.idx.start</code></td>
<td>

<p>a vector of the start variant index for each group, after pre-processing.
</p>
</td></tr>
<tr><td><code>group.idx.end</code></td>
<td>

<p>a vector of the end variant index for each group, after pre-processing.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Han Chen
</p>


<h3>References</h3>

<p>Wu, M.C., Lee, S., Cai, T., Li, Y., Boehnke, M., Lin, X. (2011) Rare-variant association testing for sequencing data with the sequence kernel association test. The American Journal of Human Genetics 89, 82-93.
</p>
<p>Lee, S., Wu, M.C., Lin, X. (2012) Optimal tests for rare variant effects in sequencing association studies. Biostatistics 13, 762-775.
</p>
<p>Sun, J., Zheng, Y., Hsu, L. (2013) A unified mixed-effects model for rare-variant association in sequencing studies. Genetic Epidemiology 37, 334-344.
</p>
<p>Chen, H., Huffman, J.E., Brody, J.A., Wang, C., Lee, S., Li, Z., Gogarten, S.M., Sofer, T., Bielak, L.F., Bis, J.C., et al. (2019) Efficient variant set mixed model association tests for continuous and binary traits in large-scale whole-genome sequencing studies. The American Journal of Human Genetics 104, 260-274.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+glmmkin">glmmkin</a></code>, <code><a href="#topic+SMMAT.meta">SMMAT.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("SeqArray", quietly = TRUE) &amp;&amp; requireNamespace("SeqVarTools",
        quietly = TRUE)) {
data(example)
attach(example)
model0 &lt;- glmmkin(disease ~ age + sex, data = pheno, kins = GRM, id = "id",
        family = binomial(link = "logit"))
geno.file &lt;- system.file("extdata", "geno.gds", package = "GMMAT")
group.file &lt;- system.file("extdata", "SetID.withweights.txt",
	package = "GMMAT")
out &lt;- SMMAT(model0, geno.file, group.file, MAF.range = c(0, 0.5), 
       		miss.cutoff = 1, method = "davies")
print(out)
}

## Not run: 
obj &lt;- SMMAT.prep(model0, geno.file, group.file)
save(obj, file = "SMMAT.prep.tmp.Rdata")
# quit R session
# open a new R session
obj &lt;- get(load("SMMAT.prep.tmp.Rdata"))
out &lt;- SMMAT.lowmem(obj, MAF.range = c(0, 0.5), miss.cutoff = 1,
        method = "davies")
print(out)
unlink("SMMAT.prep.tmp.Rdata")

## End(Not run)
</code></pre>

<hr>
<h2 id='SMMAT.meta'>
Meta-analysis for variant Set Mixed Model Association Tests (SMMAT)
</h2><span id='topic+SMMAT.meta'></span>

<h3>Description</h3>

<p>Variant Set Mixed Model Association Tests (SMMAT-B, SMMAT-S, SMMAT-O and SMMAT-E) in the meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SMMAT.meta(meta.files.prefix, n.files = rep(1, length(meta.files.prefix)), 
	cohort.group.idx = NULL, group.file, group.file.sep = "\t", 
	MAF.range = c(1e-7, 0.5), MAF.weights.beta = c(1, 25), 
	miss.cutoff = 1, method = "davies", tests = "E", rho = c(0, 0.1^2, 
	0.2^2, 0.3^2, 0.4^2, 0.5^2, 0.5, 1), use.minor.allele = FALSE,
	verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SMMAT.meta_+3A_meta.files.prefix">meta.files.prefix</code></td>
<td>

<p>a character vector for prefix of intermediate files (.score.* and .var.*) required in a meta-analysis. Each element represents the prefix of .score.* and .var.* from one cohort. The length of vector should be equal to the number of cohorts.
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_n.files">n.files</code></td>
<td>

<p>an integer vector with the same length as meta.files.prefix, indicating how many sets of intermediate files (.score.* and .var.*) are expected from each cohort, usually as the result of multi-threading in creating the intermediate files (default = rep(1, length(meta.files.prefix))).
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_cohort.group.idx">cohort.group.idx</code></td>
<td>

<p>a vector with the same length as meta.files.prefix, indicating which cohorts should be grouped together in the meta-analysis assuming homogeneous genetic effects. For example, c(&quot;a&quot;,&quot;b&quot;,&quot;a&quot;,&quot;a&quot;,&quot;b&quot;) means cohorts 1, 3, 4 are assumed to have homogeneous genetic effects, and cohorts 2, 5 are in another group with homogeneous genetic effects (but possibly heterogeneous with group &quot;a&quot;). If NULL, all cohorts are in the same group (default = NULL).
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_group.file">group.file</code></td>
<td>

<p>a plain text file with 6 columns defining the test units. There should be no headers in the file, and the columns are group name, chromosome, position, reference allele, alternative allele and weight, respectively.
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_group.file.sep">group.file.sep</code></td>
<td>

<p>the delimiter in group.file (default = &quot;\t&quot;).
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_maf.range">MAF.range</code></td>
<td>

<p>a numeric vector of length 2 defining the minimum and maximum minor allele frequencies of variants that should be included in the analysis (default = c(1e-7, 0.5)). Filter applied to the combined samples.
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_maf.weights.beta">MAF.weights.beta</code></td>
<td>

<p>a numeric vector of length 2 defining the beta probability density function parameters on the minor allele frequencies. This internal minor allele frequency weight is multiplied by the external weight given by the group.file. To turn off internal minor allele frequency weight and only use the external weight given by the group.file, use c(1, 1) to assign flat weights (default = c(1, 25)). Applied to the combined samples.
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_miss.cutoff">miss.cutoff</code></td>
<td>

<p>the maximum missing rate allowed for a variant to be included (default = 1, including all variants). Filter applied to the combined samples.
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_method">method</code></td>
<td>

<p>a method to compute p-values for SKAT-type test statistics (default = &quot;davies&quot;). &quot;davies&quot; represents an exact method that computes a p-value by inverting the characteristic function of the mixture chisq distribution, with an accuracy of 1e-6. When &quot;davies&quot; p-value is less than 1e-5, it defaults to method &quot;kuonen&quot;. &quot;kuonen&quot; represents a saddlepoint approximation method that computes the tail probabilities of the mixture chisq distribution. When &quot;kuonen&quot; fails to compute a p-value, it defaults to method &quot;liu&quot;. &quot;liu&quot; is a moment-matching approximation method for the mixture chisq distribution.
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_tests">tests</code></td>
<td>

<p>a character vector indicating which SMMAT tests should be performed (&quot;B&quot; for the burden test, &quot;S&quot; for SKAT, &quot;O&quot; for SKAT-O and &quot;E&quot; for the efficient hybrid test of the burden test and SKAT). The burden test and SKAT are automatically included when performing &quot;O&quot;, and the burden test is automatically included when performing &quot;E&quot; (default = &quot;E&quot;).
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_rho">rho</code></td>
<td>

<p>a numeric vector defining the search grid used in SMMAT-O for SKAT-O (see the SKAT-O paper for details). Not used for SMMAT-B for the burden test, SMMAT-S for SKAT or SMMAT-E for the efficient hybrid test of the burden test and SKAT (default = c(0, 0.1^2, 0.2^2, 0.3^2, 0.4^2, 0.5^2, 0.5, 1)).
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_use.minor.allele">use.minor.allele</code></td>
<td>

<p>a logical switch for whether to use the minor allele (instead of the alt allele) as the coding allele (default = FALSE). It does not change SMMAT-S results, but SMMAT-B (as well as SMMAT-O and SMMAT-E) will be affected. Along with the MAF filter, this option is useful for combining rare mutations, assuming rare allele effects are in the same direction. Use with caution, as major/minor alleles may flip in different cohorts. In that case, minor allele will be determined based on the allele frequency in the combined samples.
</p>
</td></tr>
<tr><td><code id="SMMAT.meta_+3A_verbose">verbose</code></td>
<td>

<p>a logical switch for whether a progress bar should be shown (default = FALSE).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with the following components:
</p>
<table>
<tr><td><code>group</code></td>
<td>

<p>name of the test unit group.
</p>
</td></tr>
<tr><td><code>n.variants</code></td>
<td>

<p>number of variants in the test unit group that pass the missing rate and allele frequency filters.
</p>
</td></tr>
<tr><td><code>B.score</code></td>
<td>

<p>burden test score statistic.
</p>
</td></tr>
<tr><td><code>B.var</code></td>
<td>

<p>variance of burden test score statistic.
</p>
</td></tr>
<tr><td><code>B.pval</code></td>
<td>

<p>burden test p-value.
</p>
</td></tr>
<tr><td><code>S.pval</code></td>
<td>

<p>SKAT p-value.
</p>
</td></tr>
<tr><td><code>O.pval</code></td>
<td>

<p>SKAT-O p-value.
</p>
</td></tr>
<tr><td><code>O.minp</code></td>
<td>

<p>minimum p-value in the SKAT-O search grid.
</p>
</td></tr>
<tr><td><code>O.minp.rho</code></td>
<td>

<p>rho value at the minimum p-value in the SKAT-O search grid.
</p>
</td></tr>
<tr><td><code>E.pval</code></td>
<td>

<p>SMMAT efficient hybrid test of the burden test and SKAT p-value.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Han Chen
</p>


<h3>References</h3>

<p>Wu, M.C., Lee, S., Cai, T., Li, Y., Boehnke, M., Lin, X. (2011) Rare-variant association testing for sequencing data with the sequence kernel association test. The American Journal of Human Genetics 89, 82-93.
</p>
<p>Lee, S., Wu, M.C., Lin, X. (2012) Optimal tests for rare variant effects in sequencing association studies. Biostatistics 13, 762-775.
</p>
<p>Sun, J., Zheng, Y., Hsu, L. (2013) A unified mixed-effects model for rare-variant association in sequencing studies. Genetic Epidemiology 37, 334-344.
</p>
<p>Chen, H., Huffman, J.E., Brody, J.A., Wang, C., Lee, S., Li, Z., Gogarten, S.M., Sofer, T., Bielak, L.F., Bis, J.C., et al. (2019) Efficient variant set mixed model association tests for continuous and binary traits in large-scale whole-genome sequencing studies. The American Journal of Human Genetics 104, 260-274.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+glmmkin">glmmkin</a></code>, <code><a href="#topic+SMMAT">SMMAT</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("SeqArray", quietly = TRUE) &amp;&amp; requireNamespace("SeqVarTools",
        quietly = TRUE)) {
data(example)
attach(example)
model0 &lt;- glmmkin(disease ~ age + sex, data = pheno, kins = GRM, id = "id",
        family = binomial(link = "logit"))
geno.file &lt;- system.file("extdata", "geno.gds", package = "GMMAT")
group.file &lt;- system.file("extdata", "SetID.withweights.txt",
	package = "GMMAT")
metafile &lt;- tempfile()
out &lt;- SMMAT(model0, geno.file, group.file, meta.file.prefix = metafile, 
       	MAF.range = c(0, 0.5), miss.cutoff = 1, method = "davies")
print(out)
out1 &lt;- SMMAT.meta(metafile, group.file = group.file)
print(out1)
unlink(paste0(metafile, c(".score", ".var"), ".1"))
}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
