<!DOCTYPE html><html><head><title>Help for package gMWT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gMWT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#createGroups'>
<p>Renumbering the Group Labels</p></a></li>
<li><a href='#estPI'>
<p>Calculate Probabilistic Index</p></a></li>
<li><a href='#getSigTests'>
<p>Extract Significant Test Results for a Given Alpha or Different Multiple Testing Corrections.</p></a></li>
<li><a href='#gmw'>
<p>Generalized Mann-Whitney Type Tests</p></a></li>
<li><a href='#gMWT-package'>
<p>Implementations of Generalized Mann-Whitney Type Tests</p></a></li>
<li><a href='#plot.estPI'><p>Plot an estPI Object</p></a></li>
<li><a href='#plot.re'><p>Plot an re Object</p></a></li>
<li><a href='#plotPI'>
<p>Creates Scatterplot Matrices for Probabilistic Indices.</p></a></li>
<li><a href='#print.estPI'><p>Print an estPI Object</p></a></li>
<li><a href='#print.re'><p>Print an re Object</p></a></li>
<li><a href='#rejectionPlot'>
<p>Creates a Rejection Plot.</p></a></li>
<li><a href='#summary.re'><p>Summarize an re Object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Generalized Mann-Whitney Type Tests</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-04-14</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel Fischer &lt;daniel.fischer@luke.fi&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2), clinfun, Rcpp (&ge; 0.9.13)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Description:</td>
<td>Generalized Mann-Whitney type tests based on probabilistic
        indices and new diagnostic plots, for the underlying manuscript see Fischer, Oja (2015) &lt;<a href="https://doi.org/10.18637%2Fjss.v065.i09">doi:10.18637/jss.v065.i09</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-04-18 11:10:28 UTC; ejo138</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel Fischer <a href="https://orcid.org/0000-0002-7513-683X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Hannu Oja [aut, ths]</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-19 10:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='createGroups'>
Renumbering the Group Labels 
</h2><span id='topic+createGroups'></span>

<h3>Description</h3>

<p>This function transforms the k group labels to integers 1,...,k in a specified order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  createGroups(g, desOrder)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createGroups_+3A_g">g</code></td>
<td>
<p>Vector with group labels.</p>
</td></tr>
<tr><td><code id="createGroups_+3A_desorder">desOrder</code></td>
<td>
<p>Specified order for the groups.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function helps to renumber the group labels in a specified way. 
The vector <code>desOrder</code> gives the old numbers as the function of the new numbers.
The first element of <code>desOrder</code> gives the old number that is transformed to be one, and so on.
The renumbering is used to specify alternative hypotheses.
</p>
<p>The output is then a transformed vector  <code>g</code>  with new group numbers.
</p>
<p>For non-numeric old group labels, the old number is the rank in the lexicographical order.
</p>


<h3>Value</h3>

<p>A vector containing the new group numbers.
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g &lt;- c(rep(1,5),rep(2,6),rep(3,4))
createGroups(g,c(2,3,1))

# 'Smallest' letter should become 3, second smallest 2 and largest 1:
g &lt;- sample(c("A","B","C"),10,replace=TRUE)
createGroups(g,c(3,2,1))
</code></pre>

<hr>
<h2 id='estPI'>
Calculate Probabilistic Index
</h2><span id='topic+estPI'></span>

<h3>Description</h3>

<p>This function calculates the probabilistic indices <code class="reqn">\hat{P}_t</code> , <code class="reqn">\hat{P}_{tt'}</code> and <code class="reqn">\hat{P}_{tt't''}</code> 
to compare the groups of observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  estPI(X, g, type="pair", goi=NULL, mc=1, order=TRUE, alg="Cnaive")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estPI_+3A_x">X</code></td>
<td>
<p>Matrix or vector with observations. The rows refer to individuals, the columns to variables.</p>
</td></tr>
<tr><td><code id="estPI_+3A_g">g</code></td>
<td>
<p>Vector of group numbers for observations in <code>X</code>. Its length has to be the number of observations in <code>X</code>.</p>
</td></tr>
<tr><td><code id="estPI_+3A_type">type</code></td>
<td>
<p>Type of probabilistic index, see details.</p>
</td></tr>
<tr><td><code id="estPI_+3A_goi">goi</code></td>
<td>
<p>Groups Of Interest, see details.</p>
</td></tr>
<tr><td><code id="estPI_+3A_mc">mc</code></td>
<td>
<p>Multiple Cores, set the amount of cores to use for parallel calculation (only available for Linux).</p>
</td></tr>
<tr><td><code id="estPI_+3A_order">order</code></td>
<td>
<p>Boolean, calculate probabilistic index only for a specified order of groups or for all possible orders.</p>
</td></tr>
<tr><td><code id="estPI_+3A_alg">alg</code></td>
<td>
<p>Internal function, which algorithm should be used to calculate the probabilistic index.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The matrix <code>X</code> contains the data. Each column refers to a variable, each row to an observation. 
The group memberships of the observations are given in <code>g</code>. In the case of one dimensional data,  
<code>X</code> is a vector.
</p>
<p>There is also an option to calculate the probabilistic indices (PI) only for a subset of the groups. 
In that case those group labels can be specified with the <code>goi</code> option by giving the corresponding
group labels. 
</p>
<p>Different types of PI can be calculated: <code>"single"</code> calculates the probability <code class="reqn">\hat{P}_t</code> for each group,
<code>"pair"</code> produces the probabilistic indices <code class="reqn">\hat{P}_{tt'}</code> for all pairs of groups <code class="reqn">t&lt;t'</code>, and
<code>"triple"</code> provides the probabilities <code class="reqn">\hat{P}_{tt't''}</code> for all triples of groups <code class="reqn">t&lt;t'&lt;t''</code>.
See Fischer et al. (2013) for more details.
</p>
<p>Note that the PIs are calculated using the group numbering as given in <code>g</code>. See also the function
<code><a href="#topic+createGroups">createGroups</a></code> for renumbering the group labels. By specifying the option <code>order=FALSE</code>
the PIs for all possible group orders will be calculated. The default is that the PI is only calculated for the
natural order given by <code>g</code>.
</p>
<p>In case this code is executed on a Linux machine and <code>X</code> is a data matrix, the calculation can be parallelized
using the option <code>mc</code> to specify the amount of used calculation cores. 
</p>
<p>Different algorithms are available for the  calculation of the PIs. The default is the fastest
possible option and the user should not change this option. Different algorithms are only provided for validation and testing
purposes. Options here are currently <code>Cnaive</code>, <code>Rnaive</code>, <code>Rgrid</code>, <code>Rsub</code>, <code>Csub</code> but not all combinations
of <code>type</code>/<code>alg</code> are available.
</p>


<h3>Value</h3>

<p>A list with class 'estPI' containing the following components:
</p>
<table>
<tr><td><code>probs</code></td>
<td>
<p>Matrix or vector of the  PIs.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>String, storing the type of PI.</p>
</td></tr>
<tr><td><code>goi</code></td>
<td>
<p>Vector, the Groups Of Interest, as given in the function call.</p>
</td></tr>
<tr><td><code>order</code></td>
<td>
<p>Boolean, PI just for the specified order or for all orders.</p>
</td></tr>
<tr><td><code>obs</code></td>
<td>
<p>Matrix, the original data matrix.</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>Vector, the original group vector.</p>
</td></tr>
<tr><td><code>alg</code></td>
<td>
<p>String, the requested algorithm.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>References</h3>

<p>Fischer, D., Oja, H., Schleutker, Sen, P.K., J., Wahlfors, T. (2013): Generalized Mann-Whitney Type Tests for Microarray Experiments, Scandinavian Journal of Statistics, to appear.<br />
</p>
<p>Fischer, D., Oja, H. (2013): Mann-Whitney Type Tests for Microarray Experiments: The R Package gMWT, submitted article.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createGroups">createGroups</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
X &lt;- c(sample(15))
g &lt;- c(1,1,1,2,2,2,2,3,3,3,4,4,4,4,4)
estPI(X,g,type="single")

X &lt;- matrix(c(rnorm(5000,1.5,2),rnorm(6000,2,2),rnorm(4000,3.5,1)),byrow=TRUE, ncol=10)
colnames(X) &lt;- letters[1:10]
g &lt;- c(rep(1,500),rep(2,600),rep(3,400))

estPI(X,g,type="single",mc=1)

</code></pre>

<hr>
<h2 id='getSigTests'>
Extract Significant Test Results for a Given Alpha or Different Multiple Testing Corrections.
</h2><span id='topic+getSigTests'></span>

<h3>Description</h3>

<p>This function finds the p-values less than or equal to <code>alpha</code>. Alpha is as given or corresponds to an adjusted alpha.
Different multiple testing correction methods to adjust alpha are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  getSigTests(pvalues, alpha=0.05, method="plain")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSigTests_+3A_pvalues">pvalues</code></td>
<td>
<p>A matrix, vector or an object of class <code>gmw</code>.</p>
</td></tr>
<tr><td><code id="getSigTests_+3A_alpha">alpha</code></td>
<td>
<p>Significance level.</p>
</td></tr>
<tr><td><code id="getSigTests_+3A_method">method</code></td>
<td>
<p>Method of multiple testing adjustment.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument <code>pvalues</code> is the result of a <code>gmw</code> test run or a matrix/vector of p-values. If <code>pvalues</code> is in matrix shape (several test
methods applied to the same data), then each row corresponds to a different test method and the columns to different variables.  The option <code>method</code>
specifies the method of multiple testing correction. Typical options are <code>"plain"</code> for no correction, <code>"bonferroni"</code> for a Bonferroni correction,
<code>"simes"</code> for an improved Bonferroni correction and <code>"BH"</code> for a Benjamini-Hochberg correction. Please note that the Simes method and the
Benjamini-Hochberg correction lead to the same results.
For permutation test results there is also the Westfall and Young method <code>"maxT"</code> available. In order to perform this correction the option <code>keepPM=TRUE</code>
has to be set in the <code>gmw</code> call in order to keep the required permutation matrix.
</p>
<p>Additional there are the two options <code>"csD"</code> and <code>"csR"</code>. Those calculate for each alpha between 0 and <code>alpha</code> the difference (=<code>"csD"</code>)
or the ratio (=<code>"csR"</code>) between observed and expected rejections and report an optimal alpha (and corresponding test rejections) for which these
criteria are maximal. Please keep in mind that this method might return unreasonable large 'optimal' cut-off points.
</p>
<p>See also the function <code><a href="#topic+rejectionPlot">rejectionPlot</a></code> for more details. 
</p>


<h3>Value</h3>

<p>A list object of class 're' with the values:
(In case <code>pvalues</code> is a matrix the output is a list with length <code>nrow(X)</code> and each list item contains a single <code>re</code> object)
</p>
<table>
<tr><td><code>sigTests</code></td>
<td>
<p>Position of the significant tests.</p>
</td></tr>
<tr><td><code>sigPvalues</code></td>
<td>
<p>P-values of the significant tests.</p>
</td></tr>
<tr><td><code>pvalues</code></td>
<td>
<p>The original <code>pvalues</code> object.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>Chosen method.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>Chosen alpha.</p>
</td></tr>
<tr><td><code>multAlpha</code></td>
<td>
<p>Adjusted / Optimal alpha.</p>
</td></tr> 
<tr><td><code>inputN</code></td>
<td>
<p>Rows of <code>pvalues</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>References</h3>

<p>Benjamini, Y. and Hochberg, Y. (1995): Controlling the false discovery rate: a practical and powerful approach to multiple testing. J.R. Statist. Soc. B, 57(1):289 - 300.
</p>
<p>Simes, R. J. (1986): An improved bonferroni procedure for multiple tests of significance. Biometrika, 73:751 - 754.
</p>
<p>Westfall, P.H. and Young, S.S. (1993): Resampling-Based Multiple Testing: Examples and Methods for p-Value Adjustment. Wiley, New York.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rejectionPlot">rejectionPlot</a></code>, <code><a href="#topic+gmw">gmw</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  X &lt;- matrix(c(rnorm(500,2,1),rnorm(600,2,1),rnorm(400,2.2,1)),byrow=TRUE, ncol=10)
  colnames(X) &lt;- letters[1:10]
  g &lt;- c(rep(1,50),rep(2,60),rep(3,40))
  test &lt;- gmw(X,g,test="kw",type="external")
  cs1 &lt;- getSigTests(test)
  cs1

  simData &lt;- runif(1000,0,1)
  simData &lt;- c(simData,runif(200,0,0.01))
  simData2 &lt;- runif(1000,0,1)
  simData2 &lt;- c(simData2,runif(200,0,0.01))

  simDataMat &lt;- rbind(simData,simData2)

  getSigTests(simDataMat, method="bon")
  getSigTests(simData, method="bon")
  getSigTests(simData2, method="bon")
  getSigTests(simData, method="sim")
  getSigTests(pvalues=simData, method="bh",alpha=0.05)
  getSigTests(pvalues=simData, method="csD")
  getSigTests(pvalues=simData, method="csR")

  set.seed(731)
  X &lt;- matrix(c(rnorm(50,2,1),rnorm(60,2,1),rnorm(40,2.2,1)),byrow=TRUE, ncol=10)
  colnames(X) &lt;- letters[1:10]
  g &lt;- c(rep(1,5),rep(2,6),rep(3,4))
  X[12:15,1] &lt;- X[12:15,1] + 5

  # Keep the permutation matrix in order to perform the W&amp;Y adjustment
    testPM &lt;- gmw(X,g,test="kw",type="perm",keepPM=TRUE)
  # Apply the Westfall&amp; Young adjustment
    getSigTests(testPM,method="maxT")

</code></pre>

<hr>
<h2 id='gmw'>
Generalized Mann-Whitney Type Tests
</h2><span id='topic+gmw'></span>

<h3>Description</h3>

<p>The function calculates p-values for different tests as presented in the paper &quot;Generalized Mann-Whitney Type Tests for Microarray Experiments&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  gmw(X,g,goi=NULL,test="mw",type="permutation",prob="pair",nper=2000,
  alternative="greater",mc=1,output="min", cluster=NULL, order = TRUE,
  keepPM= FALSE, mwAkw=FALSE, alg=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gmw_+3A_x">X</code></td>
<td>
<p>Data matrix, each column corresponds to a variable, each row to an individual. Can also be a vector (one variable).</p>
</td></tr>
<tr><td><code id="gmw_+3A_g">g</code></td>
<td>
<p>Vector of length <code>nrow(X)</code> (respective <code>length(X)</code>), assigning treatment groups (numbers)
to observations, see details.</p>
</td></tr>
<tr><td><code id="gmw_+3A_goi">goi</code></td>
<td>
<p>Vector with elements of <code>g</code>, defining for which treatment groups the test should be performed.</p>
</td></tr>
<tr><td><code id="gmw_+3A_test">test</code></td>
<td>
<p>Specifies the test statistic.</p>
</td></tr>
<tr><td><code id="gmw_+3A_type">type</code></td>
<td>
<p>Permutation test (<code>"permutation"</code>) or asymptotic tests(<code>"asymptotic"</code>) for the calculation of the p-values. Tests implemented
in R-base <code>"external"</code> are also accessible, see details.</p>
</td></tr>
<tr><td><code id="gmw_+3A_prob">prob</code></td>
<td>
<p>This option is only for the Mann-Whitney test, see details.</p>
</td></tr>
<tr><td><code id="gmw_+3A_nper">nper</code></td>
<td>
<p>If type is <code>"permutation"</code> this option specifies how many permutations are used to calculate the p-value.</p>
</td></tr>
<tr><td><code id="gmw_+3A_alternative">alternative</code></td>
<td>
<p>Specifies the alternative, the options are <code>"smaller"</code>,<code>"greater"</code> and <code>"two.sided"</code>, see details.</p>
</td></tr>
<tr><td><code id="gmw_+3A_mc">mc</code></td>
<td>
<p>Multiple Cores, determines how many tests will be performed parallel (only available under Linux), see details.</p>
</td></tr>
<tr><td><code id="gmw_+3A_output">output</code></td>
<td>
<p>Determines the level of the details in the output.</p>
</td></tr>
<tr><td><code id="gmw_+3A_cluster">cluster</code></td>
<td>
<p>A vector of same length as <code>g</code>, giving possible cluster information for performing the permutation test.</p>
</td></tr>
<tr><td><code id="gmw_+3A_order">order</code></td>
<td>
<p>Boolean, shall all orders be calculated or only increasing orders?</p>
</td></tr>
<tr><td><code id="gmw_+3A_keeppm">keepPM</code></td>
<td>
<p>Boolean, keep the permutation matrices, required for Westfall &amp; Young multiple testing adjustment.</p>
</td></tr>
<tr><td><code id="gmw_+3A_mwakw">mwAkw</code></td>
<td>
<p>Boolean or numeric, if <code>TRUE</code> pairwise Mann-Whitney tests are performed after the Kruskal-Wallis test. If <code>numeric</code> MW tests are only performed for KW tests with smaller p-value than that value.</p>
</td></tr>
<tr><td><code id="gmw_+3A_alg">alg</code></td>
<td>
<p>Internal function, what permutation algorithm should be used. Shouln't be changed by the user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The object <code>X</code> is the data vector (one variable) or the data matrix.
Each row refers to  an observation, and  each column to a variable. The tests are performed separately for all variables.
</p>
<p>The vector <code>g</code> gives the group number. The directional tests are based on this numbering of the group.
</p>
<p>The <code>goi</code> option defines, which treatment groups are used in the test constructions.
If no groups are specified (default), all groups are used.
</p>
<p>The <code>test</code> option specifies the test statistic. Possible options are <code>'uit'</code>(union intersection test), <code>'triple'</code> (test based on triple indicator functions),
<code>'jt'</code> for Jonckheere-Terpstra test, <code>'jt*'</code> for a modified Jonckheere-Terpstra test, 
<code>'mw'</code> for the Mann-Whitney / Wilcoxon test and 
<code>'kw'</code> for the Kruskal-Wallis test. See also reference [1] for further details.
</p>
<p>The option <code>type</code> is used to decide how the p-values are computed. For all tests are permutation type tests available
and the option for that is <code>type="permutation"</code>. In addition for <code>test='mw'</code>, <code>test='kw'</code> or <code>test='jt'</code> 
also the option <code>type='external'</code> is available. This calls then the code from the base system or other, imported packages.
For <code>test='uit'</code> there is also an asymptotic test (<code>type="asymptotic"</code>) available. For <code>test='triple'</code> or <code>test='jt*'</code>
asymptotic implementations are currently under development.
</p>
<p>The <code>prob</code> option is only for the Mann-Whitney test. For the option <code>"single"</code>, the tests are to
compare a single group versus all the other groups. The option <code>"pair"</code> makes all pair-wise comparisons
between the groups. 
</p>
<p>The option <code>alternative</code> is used to specify whether one-sided or two-sided alternatives are used. 
If the test is based on the PIs, the option  <code>"greater"</code>  for example means that, according to the alternative,
the groups with larger group numbers tend to have larger observations as well. The function <code>createGroups</code>
may be used to renumber the groups, if needed. 
</p>
<p>The <code>mc</code> option is only valid if <code>X</code> is a matrix and the used OS is Linux, because the parallelisation is based
on the package parallel, and that again is based on the concept of forking, which is currently only supported under Linux. 
</p>
<p>The option <code>output</code> can be used to control how detailed the output is. The default <code>"min"</code>  reports just 
the matrix of p-values in a matrix (columns=variables, rows=tests). If <code>output="full"</code>,
a list will be returned with items containing  full test objects of class <code>htest</code>.
</p>
<p>The option <code>cluster</code> is an additional object for the Kruskal-Wallis permutation test. For cluster-dependent
observation, only the permutations within clusters are acceptable for the p-value calculation.
</p>
<p>In the <code>getSigTests</code> function it is possible to apply the Westfall &amp; Young multiple testing method. In this approach the
permutation matrix is used to adjust for multiple testing, hence if one wishes to apply this method, the only option for
<code>type</code> is <code>"permutation"</code>. In addition the boolean flag <code>keepPM</code> has to be set to <code>TRUE</code>. Is default is
to drop the permutation matrix after each run in order to save memory.
</p>
<p>If a Kruskal-Wallis test is performed, there is also the option to perform afterwards paiwise Mann-Whitney tests to identify concrete, deviating groups.
If one wishes to do that just for significant variables one can set the option <code>mwAkw</code> to the corresponding significance level. If <code>mwAkw</code> is
set to <code>TRUE</code> (or respective <code>1</code>) the Mann-Whitney tests are performed for all variables. 
</p>
<p>There is also a function to choose the used calculation algorithm, options here are <code>"Rsubmat"</code>, <code>"Rnaive"</code>, <code>"Csubmat"</code>, <code>"Cnaive"</code>. The
purpose is just for validation.
</p>


<h3>Value</h3>

<p>A matrix or vector of p-values of the underlying hypothesis test(s). In case of <code>output="full"</code> we give a list, and each list
item contains the <code>htest</code> object for the column-wise performed test.
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>References</h3>

<p>Fischer, D., Oja, H., Schleutker, J., Sen, P.K., Wahlfors, T. (2013): Generalized Mann-Whitney Type Tests for Microarray Experiments, Scandinavian Journal of Statistic, doi: 10.1111/sjos.12055.<br />
</p>
<p>Daniel Fischer, Hannu Oja (2015). Mann-Whitney Type Tests for Microarray Experiments: The R Package gMWT. Journal of Statistical Software, 65(9), 1-19. URL http://www.jstatsoft.org/v65/i01/.
</p>
<p>Westfall, P.H. and Young, S.S. (1993): Resampling-Based Multiple Testing: Examples and Methods for p-Value Adjustment. Wiley, New York.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  X &lt;- c(sample(15))
  X &lt;- c(X,101,102,103)
  g &lt;- c(1,1,1,2,2,2,2,3,3,3,4,4,4,4,4,5,5,5)
  cluster=c(rep(c(1,2),9))

  gmw(X,g,test="kw",type="external")
  gmw(X,g,test="kw",type="permutation")
  gmw(X,g,test="kw",type="permutation",cluster=cluster)

  gmw(X,g,test="jt",type="permutation")

</code></pre>

<hr>
<h2 id='gMWT-package'>
Implementations of Generalized Mann-Whitney Type Tests
</h2><span id='topic+gMWT-package'></span><span id='topic+gMWT'></span>

<h3>Description</h3>

<p>The package provides nonparametric tools for the comparison of several groups/treatments when 
the number of variables is large. The tools are the following.
</p>
<p>(i) Computation of the so-called probabilistic index (PI) estimates for the group comparisons.
The probabilistic indices here are (a) the probability <code class="reqn">\hat P_t</code> that a random observation from group <code class="reqn">t</code>
is smaller than a random observation from other groups, (b) the probability <code class="reqn">\hat P_{tt'}</code> that 
a random observation from group <code class="reqn">t</code> is smaller than a random observation from group <code class="reqn">t'</code>,
and (c) the probability <code class="reqn">\hat P_{tt't''}</code> that random observations from groups <code class="reqn">t,t',t''</code> 
appear in this same order. The tools are also given to produce the plots of PIs for set of variables.
</p>
<p>(ii) Computation of the p-values of some classical  and some new nonparametric tests for the comparison of
several groups/treatments.  The tests are based on the use of the  probabilistic 
indices (PI) <code class="reqn">\hat P_t</code>, <code class="reqn">\hat P_{tt'}</code> and <code class="reqn">\hat P_{tt't''}</code>. Classical Wilcoxon-Mann-Whitney test, 
Kruskal-Wallis and Jonckheere-Terpstra tests are included.
</p>
<p>(iii) Tools for the simultaneous testing problem. As the package is meant for the analysis of gene expression data, 
for example, the plots of the numbers of observed and expected rejected null hypotheses and their ratio as
a function of alpha are provided. Different multiple testing adjustment methods are implemented, i.a. Bonferroni,
Bejamini &amp; Hochberg and Westfall &amp; Young.  
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
    Package: </td><td style="text-align: left;"> GMW</td>
</tr>
<tr>
 <td style="text-align: left;">
    Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
    Version: </td><td style="text-align: left;"> 1.4</td>
</tr>
<tr>
 <td style="text-align: left;">
    Date: </td><td style="text-align: left;"> 2023-04-14</td>
</tr>
<tr>
 <td style="text-align: left;">
    License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;">
    LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Daniel Fischer, Hannu Oja
</p>
<p>Maintainer: Daniel Fischer &lt;daniel.fischer@luke.fi&gt;
</p>


<h3>References</h3>

<p>Daniel Fischer, Hannu Oja (2015). Mann-Whitney Type Tests for Microarray Experiments: The R Package gMWT. Journal of Statistical Software, 65(9), 1-19. URL http://www.jstatsoft.org/v65/i01/.
</p>
<p>Fischer, D., Oja, H., Schleutker, J., Sen, P.K., Wahlfors, T. (2013): Generalized Mann-Whitney Type Tests for Microarray Experiments, Scandinavian Journal of Statistic. doi: 10.1111/sjos.12055.
</p>

<hr>
<h2 id='plot.estPI'>Plot an estPI Object</h2><span id='topic+plot.estPI'></span><span id='topic+plot-method'></span><span id='topic+plot+2CestPI-method'></span>

<h3>Description</h3>

<p>Plots an <code>estPI</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> ## S3 method for class 'estPI'
plot(x,col="black",highlight,hlCol="red",pch=20,zoom=FALSE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.estPI_+3A_x">x</code></td>
<td>
<p>Object of class <code>estPI</code>.</p>
</td></tr>
<tr><td><code id="plot.estPI_+3A_col">col</code></td>
<td>
<p>Vector of colors of the scatterplot.</p>
</td></tr>
<tr><td><code id="plot.estPI_+3A_highlight">highlight</code></td>
<td>
<p>Vector with positions, which shall be marked in special color.</p>
</td></tr>
<tr><td><code id="plot.estPI_+3A_hlcol">hlCol</code></td>
<td>
<p>Color of highlighted spots.</p>
</td></tr>
<tr><td><code id="plot.estPI_+3A_pch">pch</code></td>
<td>
<p>Dot type of the plot.</p>
</td></tr>
<tr><td><code id="plot.estPI_+3A_zoom">zoom</code></td>
<td>
<p>Logical, shall the plots be zoomed to interesting areas?</p>
</td></tr>
<tr><td><code id="plot.estPI_+3A_...">...</code></td>
<td>
<p>Additional parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function plots the probabilistic indices from an <code>estPI</code> object. Additional graphical
parameters are <code>col</code>, that specifies the colors of the points used in the plot. Furthermore there
is the option to highlight the probabilistic indices for specified variables. Those are defined with
the <code>highlight</code> option and the <code>hlCol</code> color. One possible application of this is to highlight
only the significant test results, or e.g. in the genetic context only those genes, which are located
on a certain chromosome.
</p>
<p>The boolean option <code>zoom</code> is meant for plotting the graph in a different scale and it adjusts the
limits of the x and y axes to the minimum and maximum of the corresponding probabilistic index. 
</p>


<h3>Value</h3>

<p>A plot for <code>estPI</code> object</p>


<h3>Author(s)</h3>

<p>Daniel Fischer</p>


<h3>Examples</h3>

<pre><code class='language-R'>  X &lt;- c(sample(15))
  g &lt;- c(1,1,1,2,2,2,2,3,3,3,4,4,4,4,4)
  res &lt;- estPI(X,g,type="single")
  plot(res)

  X &lt;- matrix(c(rnorm(5000,1.5,2),rnorm(6000,2,2),rnorm(4000,3.5,1),rnorm(3000,2.5,1)),
           byrow=TRUE, ncol=100)
  colnames(X) &lt;- letters[1:100]
  g &lt;- c(rep(1,50),rep(2,60),rep(3,40),rep(4,30))

  res &lt;- estPI(X,g,type="single")
  plot(res)
</code></pre>

<hr>
<h2 id='plot.re'>Plot an re Object</h2><span id='topic+plot.re'></span><span id='topic+plot+2Cre-method'></span>

<h3>Description</h3>

<p>Plots an <code>re</code> object returned from the function <code>getSigTests</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> ## S3 method for class 're'
plot(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.re_+3A_x">x</code></td>
<td>
<p>Object of class <code>re</code>.</p>
</td></tr>
<tr><td><code id="plot.re_+3A_...">...</code></td>
<td>
<p>Additional parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function plots a rejection plot for an existing <code>re</code> object.
</p>


<h3>Value</h3>

<p>A plot from a <code>re</code> object</p>


<h3>Author(s)</h3>

<p>Daniel Fischer</p>


<h3>See Also</h3>

<p><code><a href="#topic+getSigTests">getSigTests</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  X &lt;- matrix(c(rnorm(500,2,1),rnorm(600,2,1),rnorm(400,2.2,1)),byrow=TRUE, ncol=10)
  colnames(X) &lt;- letters[1:10]
  g &lt;- c(rep(1,50),rep(2,60),rep(3,40))

  test &lt;- gmw(X,g,test="kw",type="external")

  re1 &lt;- getSigTests(test)
  
  plot(re1)
</code></pre>

<hr>
<h2 id='plotPI'>
Creates Scatterplot Matrices for Probabilistic Indices.
</h2><span id='topic+plotPI'></span>

<h3>Description</h3>

<p>This function creates scatterplot matrices for the different probabilistic indices (PI) <code class="reqn">P_t</code> , <code class="reqn">P_{tt'}</code> and <code class="reqn">P_{tt't''}</code>
computed for different variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  plotPI(X,g,type="pair",goi=NULL,mc=1,alg="Cnaive",col="black",
           highlight=NULL,hlCol="red",pch=20,zoom=FALSE,order=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPI_+3A_x">X</code></td>
<td>
<p>Matrix or vector with observations. Each column is a variable, each row an individual.</p>
</td></tr>
<tr><td><code id="plotPI_+3A_g">g</code></td>
<td>
<p>Vector of group labels for observations in <code>X</code>. Has to be the same length as <code>X</code> has observations.</p>
</td></tr>
<tr><td><code id="plotPI_+3A_type">type</code></td>
<td>
<p>Type of probabilistic index, see details.</p>
</td></tr>
<tr><td><code id="plotPI_+3A_goi">goi</code></td>
<td>
<p>Groups of Interest, see details.</p>
</td></tr>
<tr><td><code id="plotPI_+3A_mc">mc</code></td>
<td>
<p>Set the amount of cores to use for parallel calculation (only available for Linux).</p>
</td></tr>
<tr><td><code id="plotPI_+3A_order">order</code></td>
<td>
<p>Boolean, calculate PI only for natural order or for all combinations.</p>
</td></tr>
<tr><td><code id="plotPI_+3A_alg">alg</code></td>
<td>
<p>Internal function, which implementation should be used to calculate the PI.</p>
</td></tr>
<tr><td><code id="plotPI_+3A_col">col</code></td>
<td>
<p>Vector of colors of the scatterplot.</p>
</td></tr>
<tr><td><code id="plotPI_+3A_highlight">highlight</code></td>
<td>
<p>Vector with positions, which are marked in special color.</p>
</td></tr>
<tr><td><code id="plotPI_+3A_hlcol">hlCol</code></td>
<td>
<p>Color of highlighted spots.</p>
</td></tr>
<tr><td><code id="plotPI_+3A_pch">pch</code></td>
<td>
<p>Dot type of the plot.</p>
</td></tr>
<tr><td><code id="plotPI_+3A_zoom">zoom</code></td>
<td>
<p>Logical, shall the plots be zoomed to interesting areas?</p>
</td></tr>
<tr><td><code id="plotPI_+3A_...">...</code></td>
<td>
<p>Additional plotting arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function creates the scatterplot matrices for the PI, in case that they haven't
been calculated previoulsy. This means that all arguments of the <code><a href="#topic+estPI">estPI</a></code> are valid here, since
this function is called first and the results will then be passed to the plot function of the <code>estPI</code>
object. 
</p>
<p>See also <code><a href="#topic+plot.estPI">plot.estPI</a></code> for further details on the specific plot parameters.
</p>


<h3>Value</h3>

<p>A plot of probalistic indices</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>References</h3>

<p>Fischer, D., Oja, H., Schleutker, J., Sen, P.K., Wahlfors, T. (2013): Generalized Mann-Whitney Type Tests for Microarray Experiments, Scandinavian Journal of Statistic, to appear.<br />
</p>
<p>Fischer, D., Oja, H. (2013): Mann-Whitney Type Tests for Microarray Experiments: The R Package gMWT, submitted article.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+estPI">estPI</a></code>, <code><a href="#topic+plot.estPI">plot.estPI</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
X &lt;- c(sample(15))
g &lt;- c(1,1,1,2,2,2,2,3,3,3,4,4,4,4,4)
estPI(X,g,type="single")

X &lt;- matrix(c(rnorm(5000,1.5,2),rnorm(6000,2,2),rnorm(4000,3.5,1)),byrow=TRUE, ncol=10)
colnames(X) &lt;- letters[1:10]
g &lt;- c(rep(1,500),rep(2,600),rep(3,400))

plotPI(X,g,type="single",mc=1)

</code></pre>

<hr>
<h2 id='print.estPI'>Print an estPI Object</h2><span id='topic+print.estPI'></span><span id='topic+print-method'></span><span id='topic+print+2CestPI-method'></span>

<h3>Description</h3>

<p>Prints an <code>estPI</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> ## S3 method for class 'estPI'
print(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.estPI_+3A_x">x</code></td>
<td>
<p>Object of class <code>estPI</code>.</p>
</td></tr>
<tr><td><code id="print.estPI_+3A_...">...</code></td>
<td>
<p>Additional parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The print function displays the matrix of probabilistic indices.
</p>


<h3>Value</h3>

<p>Content of an <code>estPI</code> object</p>


<h3>Author(s)</h3>

<p>Daniel Fischer</p>


<h3>Examples</h3>

<pre><code class='language-R'>  X &lt;- c(sample(15))
  g &lt;- c(1,1,1,2,2,2,2,3,3,3,4,4,4,4,4)
  res &lt;- estPI(X,g,type="single")
  print(res)
</code></pre>

<hr>
<h2 id='print.re'>Print an re Object</h2><span id='topic+print.re'></span><span id='topic+print+2Cre-method'></span>

<h3>Description</h3>

<p>Prints an <code>re</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> ## S3 method for class 're'
print(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.re_+3A_x">x</code></td>
<td>
<p>Object of class <code>re</code>.</p>
</td></tr>
<tr><td><code id="print.re_+3A_...">...</code></td>
<td>
<p>Additional parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The print function displays in a list the significant variables for a chosen alpha and
the optimal alpha in terms of the best ratio between expected and observed significant
tests. For more details see also <code><a href="#topic+rejectionPlot">rejectionPlot</a></code>.
</p>


<h3>Value</h3>

<p>Content of an <code>re</code> object</p>


<h3>Author(s)</h3>

<p>Daniel Fischer</p>


<h3>See Also</h3>

<p><code><a href="#topic+rejectionPlot">rejectionPlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  X &lt;- matrix(c(rnorm(500,2,1),rnorm(600,2,1),rnorm(400,2.2,1)),byrow=TRUE, ncol=10)
  colnames(X) &lt;- letters[1:10]
  g &lt;- c(rep(1,50),rep(2,60),rep(3,40))

  test &lt;- gmw(X,g,test="kw",type="external")

  re1 &lt;- getSigTests(test)
  
  re1
</code></pre>

<hr>
<h2 id='rejectionPlot'>
Creates a Rejection Plot. 
</h2><span id='topic+rejectionPlot'></span>

<h3>Description</h3>

<p>This function plots the observed and expected numbers of rejected tests and their ratio
as a function of alpha.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  rejectionPlot(X,lCol="red",xlim=NULL,crit=NULL,rejLine = NULL,
                alpha = 0.01, rej.lty = c("dotted"), rej.col = "black",
		incLegend=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rejectionPlot_+3A_x">X</code></td>
<td>
<p>Matrix or vector of p-values, as obtained from <code>gmw</code> function.</p>
</td></tr>
<tr><td><code id="rejectionPlot_+3A_lcol">lCol</code></td>
<td>
<p>Vector of of line colors</p>
</td></tr>
<tr><td><code id="rejectionPlot_+3A_xlim">xlim</code></td>
<td>
<p>An interval for alpha values (for the plotting). Default is [0,1].</p>
</td></tr>
<tr><td><code id="rejectionPlot_+3A_crit">crit</code></td>
<td>
<p>Shall the distance or the ratio be plotted, see details.</p>
</td></tr>
<tr><td><code id="rejectionPlot_+3A_rejline">rejLine</code></td>
<td>
<p>Type of rejection line. Options are <code>"bh","bonferroni"</code> and <code>"simes"</code>.</p>
</td></tr>
<tr><td><code id="rejectionPlot_+3A_alpha">alpha</code></td>
<td>
<p>Alpha for the multiple testing adjusted p-value- determines the slope of the rejection line.</p>
</td></tr>
<tr><td><code id="rejectionPlot_+3A_rej.lty">rej.lty</code></td>
<td>
<p>Line type of the rejection line.</p>
</td></tr>
<tr><td><code id="rejectionPlot_+3A_rej.col">rej.col</code></td>
<td>
<p>Colour of the rejection line.</p>
</td></tr>
<tr><td><code id="rejectionPlot_+3A_inclegend">incLegend</code></td>
<td>
<p>Position of a possible legend.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The matrix <code>X</code> contains p-values as given from the <code>gmw</code> function. The rows correspond to different tests,
and the columns to different variables. 
</p>
<p>The function plots, for each test separately, the observed and expected numbers of rejected tests and their ratio (<code>crit="ratio"</code>)
or distance (<code>crit="distance"</code>)  as a function of alpha.
</p>


<h3>Value</h3>

<p>A rejection plot</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>References</h3>

<p>Fischer, D., Oja, H., Schleutker, J., Sen, P.K., Wahlfors, T. (2013): Generalized Mann-Whitney Type Tests for Microarray Experiments, Scandinavian Journal of Statistic, to appear.<br />
</p>
<p>Fischer, D., Oja, H. (2013): Mann-Whitney Type Tests for Microarray Experiments: The R Package gMWT, submitted article.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gmw">gmw</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- matrix(c(rnorm(50,2,1),rnorm(60,2,1),rnorm(40,2.5,1)),byrow=TRUE, ncol=10)
colnames(X) &lt;- letters[1:10]
g &lt;- c(rep(1,5),rep(2,6),rep(3,4))


test1 &lt;- gmw(X,g,test="mw",type="external",alternative="greater",goi=c(2,3))
test2 &lt;- gmw(X,g,test="kw",type="external")
test3 &lt;- gmw(X,g,test="jt",type="external",alternative="greater")

tests &lt;- rbind(test1$p.values,test2$p.values,test3$p.values)

rejectionPlot(tests,lCol=c("green","red","blue"))
</code></pre>

<hr>
<h2 id='summary.re'>Summarize an re Object</h2><span id='topic+summary.re'></span><span id='topic+summary+2Cre-method'></span>

<h3>Description</h3>

<p>Summarizes and prints an <code>re</code> object in an informative way.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> ## S3 method for class 're'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.re_+3A_object">object</code></td>
<td>
<p>Object of class <code>re</code>.</p>
</td></tr>
<tr><td><code id="summary.re_+3A_...">...</code></td>
<td>
<p>Additional parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summary for a <code>re</code> object.
</p>


<h3>Value</h3>

<p>Summary for an <code>re</code> object</p>


<h3>Author(s)</h3>

<p>Daniel Fischer</p>


<h3>Examples</h3>

<pre><code class='language-R'>  X &lt;- matrix(c(rnorm(500,2,1),rnorm(600,2,1),rnorm(400,2.2,1)),byrow=TRUE, ncol=10)
  colnames(X) &lt;- letters[1:10]
  g &lt;- c(rep(1,50),rep(2,60),rep(3,40))

  test &lt;- gmw(X,g,test="kw",type="external")

  re1 &lt;- getSigTests(test)
  
  summary(re1)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
