<!DOCTYPE html><html lang="en-US"><head><title>Help for package goat</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {goat}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#goat'><p>goat: Gene Set Analysis Using the Gene Set Ordinal Association Test</p></a></li>
<li><a href='#available_genesets_goatrepo'><p>Discover available geneset collections from the GOAT GitHub repository</p></a></li>
<li><a href='#cluster_genesets'><p>cluster significant genesets from <code>test_genesets()</code> by geneset similarity (separately for each 'geneset source')</p></a></li>
<li><a href='#darken_color'><p>naively darken a color by mixing in black</p></a></li>
<li><a href='#download_genesets_goatrepo'><p>Download and parse geneset collections from the GOAT GitHub repository</p></a></li>
<li><a href='#download_goat_manuscript_data'><p>Download the datasets that were used in the GOAT manuscript</p></a></li>
<li><a href='#filter_genesets'><p>filter a geneset table; intersect with an array of genes-of-interest then apply cutoffs on min/max genes per geneset</p></a></li>
<li><a href='#gg_color_hue'><p>generate colours analogous to ggplot's default palette</p></a></li>
<li><a href='#go_gene2go'><p>parse gene2go file</p></a></li>
<li><a href='#go_obo'><p>simple vectorized parsing of GO OBO file without any dependencies (beyond dplyr/tibble/tidyr)</p></a></li>
<li><a href='#goat_logo'><p>ASCII logo for this package</p></a></li>
<li><a href='#goat_nulldistributions'><p>Precomputed parameters used by the GOAT algorithm</p></a></li>
<li><a href='#goat_print_version'><p>Print package version and logo to console</p></a></li>
<li><a href='#goat_version'><p>Return goat package version as a string</p></a></li>
<li><a href='#hgnc_idmap_table'><p>Parse HGNC gene identifier lookup table that was downloaded from genenames.org into a table with HGNC ID, symbol, synonym (NA if unavailable), entrez ID</p></a></li>
<li><a href='#lighten_color'><p>naively lighten a color by mixing in white</p></a></li>
<li><a href='#load_genesets_gmtfile'><p>parse genesets in GMT format where gene identifiers are numeric Entrez gene IDs</p></a></li>
<li><a href='#load_genesets_go_bioconductor'><p>Load GO annotations via Bioconductor packages (e.g. org.Hs.eg.db for Human)</p></a></li>
<li><a href='#load_genesets_go_fromfile'><p>construct a geneset table from gene2go and OBO files</p></a></li>
<li><a href='#load_genesets_syngo'><p>parse genesets from the SynGO database</p></a></li>
<li><a href='#minlog10_fixzero'><p>-log10 transform a vector of p-values, replacing zeros with some limit/threshold</p></a></li>
<li><a href='#padjust_genesets'><p>Adjust p-values for all genesets, grouped by 'source' then adjust for the number of 'sources'</p></a></li>
<li><a href='#partition_genes'><p>Classify genes into 2 groups, e.g. to define significant or topN genes, resulting in a 'signif' column with boolean values</p></a></li>
<li><a href='#plot_heatmap'><p>plot the geneset similarity matrix as a heatmap</p></a></li>
<li><a href='#plot_lollipop'><p>Lollipop chart or barplot visualization of geneset enrichment testing results</p></a></li>
<li><a href='#plot_network'><p>plot geneset distance matrix as a network</p></a></li>
<li><a href='#plot_volcano'><p>For each provided geneset, a volcano plot of all genelist log2fc and p-values with respective geneset constituents highlighted</p></a></li>
<li><a href='#rankscore'><p>compute rank^2 scores and rescale these between 0~1000 (with further precision captured by decimals)</p></a></li>
<li><a href='#rankscore_fixed_order'><p>Gene score array, from low to high scores</p></a></li>
<li><a href='#reduce_genesets'><p>Reduce the set of significant genesets to a minimum</p></a></li>
<li><a href='#save_genesets'><p>Write a geneset table to file.</p></a></li>
<li><a href='#score_geneset_directionality'><p>Compute a score between -1 and 1 representing the proportion of up- or down-regulated genes for each geneset, weighted by gene effectsizes</p></a></li>
<li><a href='#score_geneset_oddsratio'><p>Compute odds-ratio for each geneset</p></a></li>
<li><a href='#string_trunc_right'><p>simple string truncation</p></a></li>
<li><a href='#symbol_to_entrez'><p>Map the the symbol column in a table to HGNC human gene IDs by matching official gene symbols and synonyms</p></a></li>
<li><a href='#taxonomy_identifiers'><p>Lookup table for taxonomy identifiers, their names and respective Bioconductor packages</p></a></li>
<li><a href='#test_genesets'><p>Perform geneset enrichment testing using any supported method</p></a></li>
<li><a href='#test_genesets_fisherexact'><p>Geneset ORA using Fisher-exact test</p></a></li>
<li><a href='#test_genesets_goat_bootstrap'><p>Naive GOAT variant where we estimate null parameters for each geneset size independently</p></a></li>
<li><a href='#test_genesets_goat_fitfunction'><p>Variant of the main GOAT function <code>test_genesets_goat_precomputed</code> that does not use previously prepared parameters</p></a></li>
<li><a href='#test_genesets_goat_precomputed'><p>Test geneset enrichment with the Geneset Ordinal Association Test (GOAT) algorithm</p></a></li>
<li><a href='#test_genesets_gsea'><p>GSEA as implemented in the fgsea R package</p></a></li>
<li><a href='#test_genesets_hypergeometric'><p>Geneset ORA using hypergeometric test</p></a></li>
<li><a href='#treemap_data'><p>Construct tree and treemap data structures from geneset parent/child relations</p></a></li>
<li><a href='#treemap_plot'><p>Plot a treemap</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Set Analysis Using the Gene Set Ordinal Association Test</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>
    Perform gene set enrichment analyses using the Gene set Ordinal
    Association Test (GOAT) algorithm and visualize your
    results. Koopmans, F. (2024) &lt;<a href="https://doi.org/10.1038%2Fs42003-024-06454-5">doi:10.1038/s42003-024-06454-5</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ftwkoopmans/goat/">https://github.com/ftwkoopmans/goat/</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (&ge; 2)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0), dplyr (&ge; 1.0.3)</td>
</tr>
<tr>
<td>Imports:</td>
<td>tibble (&ge; 3.0.0), tidyselect (&ge; 1.2.0), tidyr (&ge; 1.1.2),
data.table (&ge; 1.14.0), Matrix (&ge; 1.4-0), readxl (&ge; 1.4.1),
writexl (&ge; 1.4.1), Rcpp (&ge; 1.0.9), vctrs (&ge; 0.3.8), MonoPoly
(&ge; 0.3-10), ggplot2 (&ge; 3.3.0), pheatmap (&ge; 1.0.8), treemap
(&ge; 2.4), igraph (&ge; 1.2.5), ggraph (&ge; 2.0.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>AnnotationDbi, GO.db, org.Hs.eg.db, org.Pt.eg.db,
org.Mmu.eg.db, org.Mm.eg.db, org.Rn.eg.db, org.Dr.eg.db,
org.Dm.eg.db, org.Ce.eg.db, fgsea, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-22 12:53:46 UTC; frank</td>
</tr>
<tr>
<td>Author:</td>
<td>Frank Koopmans <a href="https://orcid.org/0000-0002-4973-5732"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Frank Koopmans &lt;ftwkoopmans@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-22 13:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='goat'>goat: Gene Set Analysis Using the Gene Set Ordinal Association Test</h2><span id='topic+goat-package'></span><span id='topic+goat'></span>

<h3>Description</h3>

<p>Perform gene set enrichment analyses using the
Gene set Ordinal Association Test (GOAT) algorithm
and visualize your results.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Frank Koopmans <a href="mailto:ftwkoopmans@gmail.com">ftwkoopmans@gmail.com</a> (<a href="https://orcid.org/0000-0002-4973-5732">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/ftwkoopmans/goat/">https://github.com/ftwkoopmans/goat/</a>
</p>
</li></ul>


<hr>
<h2 id='available_genesets_goatrepo'>Discover available geneset collections from the GOAT GitHub repository</h2><span id='topic+available_genesets_goatrepo'></span>

<h3>Description</h3>

<p>Yields a table that describes available gene sets. You can download the actual data for the described versions with <code>download_genesets_goatrepo()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_genesets_goatrepo()
</code></pre>

<hr>
<h2 id='cluster_genesets'>cluster significant genesets from <code>test_genesets()</code> by geneset similarity (separately for each 'geneset source')</h2><span id='topic+cluster_genesets'></span>

<h3>Description</h3>

<p>cluster significant genesets from <code>test_genesets()</code> by geneset similarity (separately for each 'geneset source')
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluster_genesets(x, genelist, hclust_method = "ward.D2")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cluster_genesets_+3A_x">x</code></td>
<td>
<p>results from <code>test_genesets()</code></p>
</td></tr>
<tr><td><code id="cluster_genesets_+3A_genelist">genelist</code></td>
<td>
<p>should be the same as provided to <code>test_genesets()</code></p>
</td></tr>
<tr><td><code id="cluster_genesets_+3A_hclust_method">hclust_method</code></td>
<td>
<p>hierarchical clustering method, any of; 'ward.D', 'ward.D2' (default), 'single', 'complete', 'average'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with elements genesets (param <code>x</code>), similarity, hc_row, hc_col
</p>

<hr>
<h2 id='darken_color'>naively darken a color by mixing in black</h2><span id='topic+darken_color'></span>

<h3>Description</h3>

<p>naively darken a color by mixing in black
</p>


<h3>Usage</h3>

<pre><code class='language-R'>darken_color(color, frac = 0.1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="darken_color_+3A_color">color</code></td>
<td>
<p>input colors</p>
</td></tr>
<tr><td><code id="darken_color_+3A_frac">frac</code></td>
<td>
<p>fraction of black; &gt;0 and &lt;1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>adjusted value for input <code>color</code>
</p>

<hr>
<h2 id='download_genesets_goatrepo'>Download and parse geneset collections from the GOAT GitHub repository</h2><span id='topic+download_genesets_goatrepo'></span>

<h3>Description</h3>

<p>while the Bioconductor respository is extensive, contains data for many species and is a part of
a larger infrastructure, it might contain outdated GO data when the user is not using the latest R version.
If users are on an R version that is a few years old, so will the GO data from Bioconductor be.
</p>
<p>As an alternative, we store gene2go data from NCBI (for Human genes only!) at the GOAT GitHub repository.
This function allows for a convenient way to download this data and then parse the genesets.
</p>
<p>Alternatively you can browse the files in the data branch of the GOAT GitHub repository and download these files manually,
then load them via the GOAT R function <code>load_genesets_go_fromfile()</code>.
To view these data, open this URL in a browser; https://github.com/ftwkoopmans/goat/tree/data
You can also use this R package to see all available data via <code>available_genesets_goatrepo()</code>
</p>
<p>By default (empty <code>version</code> parameter), this function will first check the online GOAT GitHub repository to find the
most recent version/date, then download the respective data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download_genesets_goatrepo(
  output_dir,
  type = "GO",
  version = "",
  ignore_cache = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="download_genesets_goatrepo_+3A_output_dir">output_dir</code></td>
<td>
<p>full path to the directory where the downloaded files should be stored. Directory is created if it does not exist.
e.g. <code>output_dir="~/data"</code> on unix systems, <code>output_dir="C:/data"</code> on Windows, or set to <code>output_dir=getwd()</code> to write output to the current working directory</p>
</td></tr>
<tr><td><code id="download_genesets_goatrepo_+3A_type">type</code></td>
<td>
<p>the type of genesets to download. Currently, only &quot;GO&quot; is supported (default)</p>
</td></tr>
<tr><td><code id="download_genesets_goatrepo_+3A_version">version</code></td>
<td>
<p>the dataset version. This must be a date in format YYYY-MM-DD (for example; &quot;2024-01-01&quot;) OR be left empty (NA or empty string, the default) to automatically download the latest version.</p>
</td></tr>
<tr><td><code id="download_genesets_goatrepo_+3A_ignore_cache">ignore_cache</code></td>
<td>
<p>boolean, set to TRUE to force re-download and ignore cached data, if any. Default: FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>result from respective geneset parser function. e.g. if parameter <code>type</code> was set to&quot;GO&quot; (default), this function returns the result of <code>load_genesets_go_fromfile()</code>. These data returned by this function is typically used as input for <code>filter_genesets()</code>, c.f. full example at documentation for test_genesets()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# note: this example will download 2 files of approx 10MB in total

# store the downloaded files in the following directory. Here, the temporary file
# directory is used. Alternatively, consider storing this data in a more permanent location.
# e.g. output_dir="~/data/go" on unix systems or output_dir="C:/data/go" on Windows
output_dir = tempdir()

# download data files with GO annotations (note that the release/date is printed to console)
# these are then parsed with the load_genesets_go_fromfile() function
# if the files are already available at output_dir, these are used and download is skipped
genesets_asis = download_genesets_goatrepo(output_dir)

### for a basic example on how to use the data obtain here,
### refer to the example included at function documentation of: test_genesets()

</code></pre>

<hr>
<h2 id='download_goat_manuscript_data'>Download the datasets that were used in the GOAT manuscript</h2><span id='topic+download_goat_manuscript_data'></span>

<h3>Description</h3>

<p>Downloads OMICs-based datasets that were used in the GOAT manuscript from the GOAT GitHub page.
This file is cached in the output directory and only needs to be downloaded once. Multiple datasets
are included and their names include the respective PubMed identifiers (PMID).
</p>
<p>If you encounter technical difficulties, try to;
</p>

<ol>
<li><p> download the file by copy/pasting this URL into your browser: https://github.com/ftwkoopmans/goat/raw/main/analyses/goat_manuscript_datasets.rda
</p>
</li>
<li><p> load the data in R using the following 2 lines of code, here assuming you stored the downloaded file at C:/data/goat_manuscript_datasets.rda
</p>
</li></ol>

<p><code>load("C:/data/goat_manuscript_datasets.rda")</code>
</p>
<p><code>genelist = goat_manuscript_datasets.rda[["Wingo 2020:mass-spec:PMID32424284"]]</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download_goat_manuscript_data(output_dir, ignore_cache = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="download_goat_manuscript_data_+3A_output_dir">output_dir</code></td>
<td>
<p>full path to the directory where the downloaded files should be stored. Directory is created if it does not exist.
e.g. <code>output_dir="~/data"</code> on unix systems, <code>output_dir="C:/data"</code> on Windows, or set to <code>output_dir=getwd()</code> to write output to the current working directory</p>
</td></tr>
<tr><td><code id="download_goat_manuscript_data_+3A_ignore_cache">ignore_cache</code></td>
<td>
<p>boolean, set to TRUE to force re-download and ignore cached data, if any. Default: FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of genelist data tables. The names of the list represent the datasets,
values in the list are data tables that can be used as a &quot;genelist&quot; in the GOAT R package
</p>

<hr>
<h2 id='filter_genesets'>filter a geneset table; intersect with an array of genes-of-interest then apply cutoffs on min/max genes per geneset</h2><span id='topic+filter_genesets'></span>

<h3>Description</h3>

<p>filter a geneset table; intersect with an array of genes-of-interest then apply cutoffs on min/max genes per geneset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_genesets(
  genesets,
  genelist,
  min_overlap = 10L,
  max_overlap = 1500L,
  max_overlap_fraction = 0.5,
  min_signif = NA,
  max_size = NA,
  dedupe = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_genesets_+3A_genesets">genesets</code></td>
<td>
<p>tibble with genesets, must contain columns 'id', 'genes' and 'ngenes'</p>
</td></tr>
<tr><td><code id="filter_genesets_+3A_genelist">genelist</code></td>
<td>
<p>tibble with genes, must contain column 'gene' and 'signif'. gene = character column, which are matched against list column 'genes' in genesets tibble. signif = boolean column (you can set all to FALSE if not performing Fisher-exact or hypergeometric test downstream)</p>
</td></tr>
<tr><td><code id="filter_genesets_+3A_min_overlap">min_overlap</code></td>
<td>
<p>integer, minimum number of genes in the <code>genelist</code> table that must match a geneset. Must be at least 1 but when using the GOAT algorithm downstream, this should be set to at least 10 (default=10). e.g. when set to 10, this will only retain genesets that contain at least 10 genes that are also in your genelist.</p>
</td></tr>
<tr><td><code id="filter_genesets_+3A_max_overlap">max_overlap</code></td>
<td>
<p>integer, maximum number of genes in the <code>genelist</code> table that must match a geneset. Set to NA to disable</p>
</td></tr>
<tr><td><code id="filter_genesets_+3A_max_overlap_fraction">max_overlap_fraction</code></td>
<td>
<p>analogous to <code>max_overlap</code>, which limits the max geneset size to a given N, this parameter defines the maximum geneset size that is to be retained as a fraction of the input genelist length. For example, setting this to 0.5 will remove all genesets that contain more than half the genes in the input genelist (i.e. testing enrichment of a geneset that contains 1000 out of a total 1200 genes from your input genelist is probably meaningless). Defaults to 50%</p>
</td></tr>
<tr><td><code id="filter_genesets_+3A_min_signif">min_signif</code></td>
<td>
<p>expert setting for debugging and algorithm evaluation/benchmarking, NOT for regular geneset analyses. integer, minimum number of genes in the <code>genelist</code> table that are <code>signif==TRUE</code> and match a geneset. Be careful, this is &quot;prefiltering&quot; and will affect the correctness / calibration of estimated geneset p-values. For GOAT and GSEA, this is NOT RECOMMENDED and will cause bias in your dataset! Set to NA to disable (default)</p>
</td></tr>
<tr><td><code id="filter_genesets_+3A_max_size">max_size</code></td>
<td>
<p>integer, maximum number of genes in the geneset (i.e. prior to intersect with user's gene list provided as <code>genelist</code>). Optionally, use this to remove highly generic terms. Set to NA to disable</p>
</td></tr>
<tr><td><code id="filter_genesets_+3A_dedupe">dedupe</code></td>
<td>
<p>boolean, remove duplicate genesets (as determined after intersection with <code>genelist</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the input <code>genesets</code> filtered for the subset of rows that match user's filter parameters
</p>

<hr>
<h2 id='gg_color_hue'>generate colours analogous to ggplot's default palette</h2><span id='topic+gg_color_hue'></span>

<h3>Description</h3>

<p>https://stackoverflow.com/a/8197703
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gg_color_hue(n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gg_color_hue_+3A_n">n</code></td>
<td>
<p>number of colors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a color code (string)
</p>

<hr>
<h2 id='go_gene2go'>parse gene2go file</h2><span id='topic+go_gene2go'></span>

<h3>Description</h3>

<p>note that this file lacks parent/child relations, so we only learn 'direct annotations'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>go_gene2go(f, taxid_filter = 9606)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="go_gene2go_+3A_f">f</code></td>
<td>
<p>full path to gene2go file stored on the computer, e.g. previously downloaded from https://ftp.ncbi.nih.gov/gene/DATA/gene2go.gz</p>
</td></tr>
<tr><td><code id="go_gene2go_+3A_taxid_filter">taxid_filter</code></td>
<td>
<p>return gene annotations only for the specified taxonomy identifier (integer value). Typical options are 9606 (Human, the default) or 10090 (mouse). Importantly, select the taxonomy/organism that is also in your input gene list (which typically contains human Entrez gene identifiers)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble with columns; source, source_version, id, name, genes, ngenes
</p>

<hr>
<h2 id='go_obo'>simple vectorized parsing of GO OBO file without any dependencies (beyond dplyr/tibble/tidyr)</h2><span id='topic+go_obo'></span>

<h3>Description</h3>

<p>note that we remove links between GO terms that are across GO domains (e.g. no CC to MF relations)
The only supported relations are those that match this regex;
<code>"^(is_a:|relationship: part_of|relationship: regulates|relationship: positively_regulates|relationship: negatively_regulates)"</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>go_obo(f, rename_namespace = TRUE, remove_obsolete = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="go_obo_+3A_f">f</code></td>
<td>
<p>full path to go.obo file stored on the computer, e.g. previously downloaded from http://current.geneontology.org/ontology/go.obo . Also works with a gzipped file; obo.gz</p>
</td></tr>
<tr><td><code id="go_obo_+3A_rename_namespace">rename_namespace</code></td>
<td>
<p>boolean; rename official namespace values like 'cellular_component' to CC? (analogous for BP and MF)</p>
</td></tr>
<tr><td><code id="go_obo_+3A_remove_obsolete">remove_obsolete</code></td>
<td>
<p>boolean; remove obsoleted terms?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tibble with ontology terms and their relations
</p>

<hr>
<h2 id='goat_logo'>ASCII logo for this package</h2><span id='topic+goat_logo'></span>

<h3>Description</h3>

<p>ASCII logo for this package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>goat_logo()
</code></pre>


<h3>Value</h3>

<p>package logo as a string
</p>

<hr>
<h2 id='goat_nulldistributions'>Precomputed parameters used by the GOAT algorithm</h2><span id='topic+goat_nulldistributions'></span>

<h3>Description</h3>

<p>there parameters are used by goat to efficiently perform geneset testing without bootstrapping
</p>


<h3>Usage</h3>

<pre><code class='language-R'>goat_nulldistributions
</code></pre>


<h3>Format</h3>



<h4><code>goat_nulldistributions</code></h4>

<p>a data.frame with precomputed GOAT null distribution parameters
</p>


<hr>
<h2 id='goat_print_version'>Print package version and logo to console</h2><span id='topic+goat_print_version'></span>

<h3>Description</h3>

<p>Print package version and logo to console
</p>


<h3>Usage</h3>

<pre><code class='language-R'>goat_print_version()
</code></pre>


<h3>Value</h3>

<p>prints to console without returning a value
</p>

<hr>
<h2 id='goat_version'>Return goat package version as a string</h2><span id='topic+goat_version'></span>

<h3>Description</h3>

<p>simple wrapper around utils::packageVersion()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>goat_version()
</code></pre>


<h3>Value</h3>

<p>package version as a string
</p>

<hr>
<h2 id='hgnc_idmap_table'>Parse HGNC gene identifier lookup table that was downloaded from genenames.org into a table with HGNC ID, symbol, synonym (NA if unavailable), entrez ID</h2><span id='topic+hgnc_idmap_table'></span>

<h3>Description</h3>

<p>download link: https://www.genenames.org/download/statistics-and-files/
table: &quot;Complete dataset download links&quot; &ndash;&gt;&gt; &quot;Complete HGNC approved dataset&quot; &ndash;&gt;&gt; download the &quot;TXT&quot; table
filename is typically something like hgnc_complete_set.txt
URL; https://ftp.ebi.ac.uk/pub/databases/genenames/hgnc/tsv/hgnc_complete_set.txt
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hgnc_idmap_table(filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hgnc_idmap_table_+3A_filename">filename</code></td>
<td>
<p>full path to the downloaded table (expected to be tsv format, typically has .txt or .tsv file extension)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a long-format table with columns; hgnc_id, hgnc_symbol, type, value
</p>


<h3>alternatively;</h3>

<p>table: &quot;Total Approved Symbols&quot; &ndash;&gt;&gt; &quot;TXT&quot; / &quot;text file in TSV format&quot;
filename is typically something like non_alt_loci_set.txt
</p>

<hr>
<h2 id='lighten_color'>naively lighten a color by mixing in white</h2><span id='topic+lighten_color'></span>

<h3>Description</h3>

<p>naively lighten a color by mixing in white
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lighten_color(color, frac = 0.1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lighten_color_+3A_color">color</code></td>
<td>
<p>input colors</p>
</td></tr>
<tr><td><code id="lighten_color_+3A_frac">frac</code></td>
<td>
<p>fraction of white; &gt;0 and &lt;1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>adjusted value for input <code>color</code>
</p>

<hr>
<h2 id='load_genesets_gmtfile'>parse genesets in GMT format where gene identifiers are numeric Entrez gene IDs</h2><span id='topic+load_genesets_gmtfile'></span>

<h3>Description</h3>

<p>parse genesets in GMT format where gene identifiers are numeric Entrez gene IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_genesets_gmtfile(filename, label)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="load_genesets_gmtfile_+3A_filename">filename</code></td>
<td>
<p>input file for this function should be the full path to genesets defined in GMT format</p>
</td></tr>
<tr><td><code id="load_genesets_gmtfile_+3A_label">label</code></td>
<td>
<p>a shortname for the genesets in this file, for example &quot;GO_CC&quot;, &quot;KEGG&quot;, &quot;MY_DB_V1&quot;. This will be stored in the 'source' column of the resulting table. Importantly, multiple testing correction in GOAT is grouped by this 'source' column so you probably want to use a different label for each collection-of-genesets that you load. Must not be empty, only allowed characters are; upper/lower-case letter, numbers 0-9 and underscore</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tibble with columns; source (character), source_version (character), id (character), name (character), genes (list), ngenes (int)
</p>


<h3>Example data;</h3>

<p>URL: https://www.gsea-msigdb.org/gsea/msigdb/human/collections.jsp#C5
download this data: KEGG subset of curated pathways &ndash;&gt;&gt; NCBI (Entrez) Gene IDs
filename should be something like &quot;c2.cp.kegg.v2023.1.Hs.entrez.gmt&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # TODO: update the filename to your downloaded file
  f = "C:/DATA/c2.cp.kegg.v2023.1.Hs.entrez.gmt"
  if(file.exists(f)) genesets_asis = load_genesets_gmtfile(f, label = "KEGG")
</code></pre>

<hr>
<h2 id='load_genesets_go_bioconductor'>Load GO annotations via Bioconductor packages (e.g. org.Hs.eg.db for Human)</h2><span id='topic+load_genesets_go_bioconductor'></span>

<h3>Description</h3>

<p>Download and import genesets from the GO database using the Bioconductor infrastructure.
Use the <code>goat::load_genesets_go_fromfile</code> function for more fine-grained control over the GO database version that you use; it allows you to import NCBI gene2go files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_genesets_go_bioconductor(include_child_annotations = TRUE, taxid = 9606)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="load_genesets_go_bioconductor_+3A_include_child_annotations">include_child_annotations</code></td>
<td>
<p>boolean; include annotations against child terms? In most situations, TRUE (default) is the desired setting</p>
</td></tr>
<tr><td><code id="load_genesets_go_bioconductor_+3A_taxid">taxid</code></td>
<td>
<p>integer value that indicates the taxonomy id. Default: 9606 (Human, Homo sapiens). Currently supported options:
</p>

<ul>
<li><p> 9606 = Human (Homo sapiens)
</p>
</li>
<li><p> 9598 = Chimpanzee (Pan troglodytes)
</p>
</li>
<li><p> 9544 = Rhesus monkey (Macaca mulatta)
</p>
</li>
<li><p> 10090 = Mouse (Mus musculus)
</p>
</li>
<li><p> 10116 = Rat (Rattus norvegicus)
</p>
</li>
<li><p> 7955 = Zebrafish (Danio rerio)
</p>
</li>
<li><p> 7227 = Fruit fly (Drosophila melanogaster)
</p>
</li>
<li><p> 6239 = Worm (Caenorhabditis elegans)
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that org.Hs.eg.db pulls data semi-annually from NCBI gene2go (as do similar databases for other species),
but the GO database version returned by this function is tied to the version of the org.Hs.eg.db on your computer (this is controlled by the Bioconductor infrastructure).
E.g. in an outdated R installation you may get outdated GO annotations as well.
</p>
<p>The actual GO database version that is retrieved is returned by this function in the <code>source_version</code> column.
</p>


<h3>Value</h3>

<p>table with columns; source (character), source_version (character), id (character), name (character), genes (list), ngenes (int)
</p>

<hr>
<h2 id='load_genesets_go_fromfile'>construct a geneset table from gene2go and OBO files</h2><span id='topic+load_genesets_go_fromfile'></span>

<h3>Description</h3>

<p>This function is used to load Gene Ontology (GO) genesets from files that you
manually downloaded from the links below. This enables the use of the latest data
from GO (in contrast,  Bioconductor GO data may lag behind current data considerably).
To construct genesets from available raw data, download the &quot;gene2go&quot; file
(the gene annotations) from below NCBI link and download the GO OBO
(ontology terms and relations to respective parent/child terms)  from below
geneontology.org link. Provide the full path to the downloaded file to this function.
Both &quot;gzipped&quot; and &quot;uncompressed&quot; files are supported.
</p>
<p>We encourage you to rename the files after your downloaded them such that
the date of download in incorporated; this ensures you can always keep track of
the GO database version that was used! For example, rename the downloaded
&quot;gene2go.gz&quot; file to &quot;gene2go_2024-01-31.gz&quot;.
</p>
<p>Download link for gene2go file; https://ftp.ncbi.nih.gov/gene/DATA/gene2go.gz
</p>
<p>Download link for gene ontology OBO file; http://current.geneontology.org/ontology/go.obo
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_genesets_go_fromfile(
  file_gene2go,
  file_goobo,
  include_child_annotations = TRUE,
  taxid_filter = 9606
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="load_genesets_go_fromfile_+3A_file_gene2go">file_gene2go</code></td>
<td>
<p>full path to the gene2go file from NCBI. Also works with a gzipped file; gene2go.gz</p>
</td></tr>
<tr><td><code id="load_genesets_go_fromfile_+3A_file_goobo">file_goobo</code></td>
<td>
<p>full path to the OBO file from geneontology.org. Also works with a gzipped file; obo.gz</p>
</td></tr>
<tr><td><code id="load_genesets_go_fromfile_+3A_include_child_annotations">include_child_annotations</code></td>
<td>
<p>boolean; include annotations against child terms? In most situations, TRUE (default) is the desired setting</p>
</td></tr>
<tr><td><code id="load_genesets_go_fromfile_+3A_taxid_filter">taxid_filter</code></td>
<td>
<p>return gene annotations only for the specified taxonomy identifier (integer value). Typical options are 9606 (Human, the default) or 10090 (mouse). Importantly, select the taxonomy/organism that is also in your input gene list (which typically contains human Entrez gene identifiers)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>table with columns; source (character), source_version (character), id (character), name (character), genes (list), ngenes (int)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # TODO: update the filenames to your downloaded files
  file_gene2go = "C:/DATA/gene2go_2024-01-01.gz"
  file_goobo = "C:/DATA/go_2024-01-01.obo"
  if(file.exists(file_gene2go) &amp;&amp; file.exists(file_goobo)) {
    genesets_asis = load_genesets_go_fromfile(file_gene2go, file_goobo)
  }
</code></pre>

<hr>
<h2 id='load_genesets_syngo'>parse genesets from the SynGO database</h2><span id='topic+load_genesets_syngo'></span>

<h3>Description</h3>

<p>Workflow;
</p>

<ul>
<li><p> obtain the input file from; https://www.syngoportal.org
</p>
</li>
<li><p> click &quot;bulk download SynGO release ...&quot; for SynGO release of interest
</p>
</li>
<li><p> unzip
</p>
</li>
<li><p> call this function with the full file path to the 'syngo_ontologies.xlsx' file
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>load_genesets_syngo(filename, gene_database = "entrez")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="load_genesets_syngo_+3A_filename">filename</code></td>
<td>
<p>full path to the &quot;syngo_ontologies.xlsx&quot; file that was extracted from a SynGO bulk download ZIP archive</p>
</td></tr>
<tr><td><code id="load_genesets_syngo_+3A_gene_database">gene_database</code></td>
<td>
<p>gene IDs to return. must be any of; &quot;entrez&quot; (default), &quot;hgnc&quot;, &quot;ensembl&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>table with columns; source (character), source_version (character), id (character), name (character), genes (list), ngenes (int)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # TODO: update the filename to your downloaded file
  f = "C:/DATA/SynGO_bulk_download_release_20231201/syngo_ontologies.xlsx"
  if(file.exists(f)) genesets_asis = load_genesets_syngo(f)
</code></pre>

<hr>
<h2 id='minlog10_fixzero'>-log10 transform a vector of p-values, replacing zeros with some limit/threshold</h2><span id='topic+minlog10_fixzero'></span>

<h3>Description</h3>

<p>-log10 transform a vector of p-values, replacing zeros with some limit/threshold
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minlog10_fixzero(x, limit = 2.22e-16)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="minlog10_fixzero_+3A_x">x</code></td>
<td>
<p>p-value vector to transform to -log10</p>
</td></tr>
<tr><td><code id="minlog10_fixzero_+3A_limit">limit</code></td>
<td>
<p>value to replace zero's in <code>x</code> with. Set NA to replace zero's in <code>x</code> with the smallest finite value in <code>x</code> (if there is none, defaults to 2.22e-16)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>input parameter <code>x</code> transformed to -log10
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  pval = c(0, 10^-6, 0.001, 0.01, 1, NA, -Inf, Inf, NaN)
  cbind(
    input = pval,
    # default; replace zeros with typical R machine precision for doubles
    minlog10_default = minlog10_fixzero(pval),
    # alternatively, replace zero with lowest non-zero pvalue in input
    minlog10_limit_from_data = minlog10_fixzero(pval, limit = NA)
  )
</code></pre>

<hr>
<h2 id='padjust_genesets'>Adjust p-values for all genesets, grouped by 'source' then adjust for the number of 'sources'</h2><span id='topic+padjust_genesets'></span>

<h3>Description</h3>

<p>Adjust p-values for all genesets, grouped by 'source' then adjust for the number of 'sources'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>padjust_genesets(
  genesets,
  method = "BH",
  cutoff = 0.01,
  correct_sources = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="padjust_genesets_+3A_genesets">genesets</code></td>
<td>
<p>tibble with genesets, must contain column 'pvalue'</p>
</td></tr>
<tr><td><code id="padjust_genesets_+3A_method">method</code></td>
<td>
<p>method for multiple testing correction, must be any of <code>stats::p.adjust.methods</code>, e.g. &quot;BH&quot; or &quot;bonferroni&quot;</p>
</td></tr>
<tr><td><code id="padjust_genesets_+3A_cutoff">cutoff</code></td>
<td>
<p>numeric cutoff value for adjusted p-value, <code>signif</code> column is set to TRUE for all values lesser-equals</p>
</td></tr>
<tr><td><code id="padjust_genesets_+3A_correct_sources">correct_sources</code></td>
<td>
<p>apply Bonferroni adjustment to all p-values according to the number of geneset sources that were tested. Boolean parameter, set TRUE to enable (default) or FALSE to disable</p>
</td></tr>
</table>


<h3>Value</h3>

<p>updated <code>genesets</code> table
</p>

<hr>
<h2 id='partition_genes'>Classify genes into 2 groups, e.g. to define significant or topN genes, resulting in a 'signif' column with boolean values</h2><span id='topic+partition_genes'></span>

<h3>Description</h3>

<p>This can be convenient to prepare the significant/test/foreground set for classical ORA,
e.g. <code>test_genesets()</code> with parameter <code>method = "fisherexact"</code>. Note that the GOAT geneset
enrichment algorithm does not use data in the 'signif' column of the input genelist.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>partition_genes(
  genes,
  col,
  decreasing = FALSE,
  use_abs = FALSE,
  cutoff = NULL,
  fraction = NULL,
  topn = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="partition_genes_+3A_genes">genes</code></td>
<td>
<p>gene tibble where each row is a unique gene, must contain column name <code>col</code></p>
</td></tr>
<tr><td><code id="partition_genes_+3A_col">col</code></td>
<td>
<p>column name in <code>genes</code></p>
</td></tr>
<tr><td><code id="partition_genes_+3A_decreasing">decreasing</code></td>
<td>
<p>order <code>col</code> in descending (set TRUE) or ascending order (set FALSE, default) prior to partitioning?</p>
</td></tr>
<tr><td><code id="partition_genes_+3A_use_abs">use_abs</code></td>
<td>
<p>use absolute values (default FALSE), e.g. when setting a threshold on effect-sizes</p>
</td></tr>
<tr><td><code id="partition_genes_+3A_cutoff">cutoff</code></td>
<td>
<p>threshold for values in <code>col</code> to select  (must provide exactly 1 parameter for filtering, either <code>cutoff</code>, <code>fraction</code> or <code>topn</code>)</p>
</td></tr>
<tr><td><code id="partition_genes_+3A_fraction">fraction</code></td>
<td>
<p>fraction of rows in <code>genes</code> tibble to select  (must provide exactly 1 parameter for filtering, either <code>cutoff</code>, <code>fraction</code> or <code>topn</code>)</p>
</td></tr>
<tr><td><code id="partition_genes_+3A_topn">topn</code></td>
<td>
<p>number of rows in <code>genes</code> tibble to select  (must provide exactly 1 parameter for filtering, either <code>cutoff</code>, <code>fraction</code> or <code>topn</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>input table <code>genes</code> with results in the &quot;signif&quot; column
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# note: this example will download 1 files of approx 4MB

# store the downloaded files in the following directory. Here, the temporary file
# directory is used. Alternatively, consider storing this data in a more permanent location.
# e.g. output_dir="~/data/goat" on unix systems or output_dir="C:/data/goat" on Windows
output_dir = tempdir()

# Download an example gene list, i.e. one of the datasets analyzed in the GOAT manuscript.
datasets = download_goat_manuscript_data(output_dir)
genelist = datasets$`Wingo 2020:mass-spec:PMID32424284`

# example 1: significant hits
genelist = partition_genes(genelist, col="pvalue_adjust", decreasing=FALSE, cutoff=0.01)
cat(sum(genelist$signif), "/", nrow(genelist), "are signif\n")

# example 2: abs(effectsize) &gt;= 5
genelist = partition_genes(genelist, col="effectsize", decreasing=TRUE, use_abs=TRUE, cutoff=5)
cat(sum(genelist$signif), "/", nrow(genelist), "are signif\n")

# example 3: top 10% 'best' p-values
genelist = partition_genes(genelist, col="pvalue", decreasing=FALSE, fraction = 0.1)
cat(sum(genelist$signif), "/", nrow(genelist), "are signif\n")

</code></pre>

<hr>
<h2 id='plot_heatmap'>plot the geneset similarity matrix as a heatmap</h2><span id='topic+plot_heatmap'></span>

<h3>Description</h3>

<p>plot the geneset similarity matrix as a heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heatmap(
  x,
  output_dir,
  colors = grDevices::hcl.colors(100, "Viridis", rev = FALSE),
  fontsize = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heatmap_+3A_x">x</code></td>
<td>
<p>result from <code>cluster_genesets()</code></p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_output_dir">output_dir</code></td>
<td>
<p>set to NA to directly show the figures instead of writing them to file.
Otherwise, this is the full path to the directory where the downloaded files should be stored. Directory is created if it does not exist.
e.g. <code>output_dir="~/data"</code> on unix systems, <code>output_dir="C:/data"</code> on Windows, or set to <code>output_dir=getwd()</code> to write output to the current working directory</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_colors">colors</code></td>
<td>
<p>a vector of 100 colors to be used for the heatmap (101 breaks are computed between 0 and the max value in the distance matrix)</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter sent to pheatmap::pheatmap(); control the size of labels in the plot, defaults to 10. Note that you can also change the plot device size, see examples</p>
</td></tr>
</table>


<h3>Value</h3>

<p>does not return a value, plots are printed to device or files depending on <code>output_dir</code> parameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# note; this example downloads data when first run, and typically takes ~60seconds

# store the downloaded files in the following directory. Here, the temporary file
# directory is used. Alternatively, consider storing this data in a more permanent location.
# e.g. output_dir="~/data/goat" on unix systems or output_dir="C:/data/goat" on Windows
output_dir = tempdir()

## first run the default example from test_genesets() to obtain geneset results
datasets = download_goat_manuscript_data(output_dir)
genelist = datasets$`Wingo 2020:mass-spec:PMID32424284`
genesets_asis = download_genesets_goatrepo(output_dir)
genesets_filtered = filter_genesets(genesets_asis, genelist)
result = test_genesets(genesets_filtered, genelist, method = "goat",
  score_type = "effectsize", padj_method = "bonferroni", padj_cutoff = 0.05)

# prior to running this function, cluster the genesets
clusters = cluster_genesets(result, genelist)

# use the plot heatmap function and try various color palettes
plot_heatmap(clusters, output_dir, colors = hcl.colors(100, "Viridis", rev = FALSE))
plot_heatmap(clusters, output_dir, colors = hcl.colors(100, "Inferno", rev = FALSE))
plot_heatmap(clusters, output_dir, colors = hcl.colors(100, "Lajolla", rev = TRUE))
plot_heatmap(clusters, output_dir, colors = hcl.colors(100, "Mako", rev = FALSE))
plot_heatmap(clusters, output_dir, colors = hcl.colors(100, "Turku", rev = TRUE))
plot_heatmap(clusters, output_dir, colors = hcl.colors(100, "Grays", rev = TRUE))

</code></pre>

<hr>
<h2 id='plot_lollipop'>Lollipop chart or barplot visualization of geneset enrichment testing results</h2><span id='topic+plot_lollipop'></span>

<h3>Description</h3>

<p>Lollipop chart or barplot visualization of geneset enrichment testing results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_lollipop(
  x,
  output_dir,
  only_reduced = FALSE,
  plot_type = "lollipop",
  show_pvalue = FALSE,
  score_xaxis = "minlogp",
  score_color = ifelse(is.data.frame(x) &amp;&amp; "score_type" %in% colnames(x) &amp;&amp;
    is.character(x$score_type) &amp;&amp; any(x$score_type %in% c("effectsize_up",
    "effectsize_down")), "updown", "minlogp"),
  score_color_limits = "source",
  score_color_updown = c("#E57373", "#5E7ABC"),
  max_ngenes = NA,
  topn = NA,
  padj_cutoff = NA
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_lollipop_+3A_x">x</code></td>
<td>
<p>results from function <code>test_genesets</code></p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_output_dir">output_dir</code></td>
<td>
<p>set to NA to directly show the figures instead of writing them to file.
Otherwise, this is the full path to the directory where the downloaded files should be stored. Directory is created if it does not exist.
e.g. <code>output_dir="~/data"</code> on unix systems, <code>output_dir="C:/data"</code> on Windows, or set to <code>output_dir=getwd()</code> to write output to the current working directory</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_only_reduced">only_reduced</code></td>
<td>
<p>only show the reduced/summarized set of significant genesets. This requires that you first applied the <code>reduce_genesets</code> function</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_plot_type">plot_type</code></td>
<td>
<p>Options: &quot;barplot&quot;, &quot;lollipop&quot; (default)</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_show_pvalue">show_pvalue</code></td>
<td>
<p>boolean parameter that indicates whether adjusted p-values should be shown next to each data point</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_score_xaxis">score_xaxis</code></td>
<td>
<p>type of score to show on the x-axis. Options: &quot;minlogp&quot; to show -log10(adjusted pvalue), which is default. Use &quot;oddsratio&quot; to show the enrichment of significant genes.
For further details on this score and its computation, see the <code>score_geneset_oddsratio</code> function documentation.
Basically, the genesets in this plot are sorted by their proportion of foreground/significant genes
(and this ratio is standardized against the overall ratio of significant genes as to make this statistic comparable across analyses/datasets).</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_score_color">score_color</code></td>
<td>
<p>analogous to <code>score_xaxis</code>, here you can specify the data used for color-coding the plot. Options: &quot;minlogp&quot;, &quot;oddsratio&quot;, &quot;updown&quot;.
The former 2 options are the same as for <code>score_xaxis</code>, the latter enables color-coding by up-/down-regulation as encoded in the &quot;score_type&quot; column.
Note that this only works when geneset testing based on &quot;effectsize&quot; was performed and thus the &quot;score_type&quot; column has values &quot;effectsize_up&quot; or &quot;effectsize_down&quot; (encoding directionality).
Rows with other values are assumed NA and colored as grey.</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_score_color_limits">score_color_limits</code></td>
<td>
<p>defines the limits for the color scales. options; <code>score_color_limits = "source"</code> use the range of values per 'source' to compute colors scales (default). Set to &quot;overall&quot; in order to have a unified color scale across 'source'  (e.g. same color bar across GO_CC/GO_MF/GO_BP). Alternatively, provide a numeric vector of 2 values to manually define lower/upper-limits.</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_score_color_updown">score_color_updown</code></td>
<td>
<p>array of 2 strings that describe the colors for up- and down-regulation (used when <code>score_color</code> is set to &quot;updown&quot;). Default color-coding; up = red, down = blue. Use hex color codes, e.g. &quot;#ff0000&quot; for red.</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_max_ngenes">max_ngenes</code></td>
<td>
<p>only plot terms with less than N genes (quick way to get rid of large/unspecific terms)</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_topn">topn</code></td>
<td>
<p>topn terms to show after sorting genesets by p-value. For example, this makes it easy to plot the top10 GO terms. Set to NA to ignore this filter (default)</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_padj_cutoff">padj_cutoff</code></td>
<td>
<p>adjusted pvalue cutoff for terms shown in the plot. If set to NA (default), all significant genesets are used (i.e. 'signif' column in the input geneset table)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if <code>output_dir</code> is <code>NA</code>, a list of ggplot2 objects. Otherwise, write plots to file and do not return any value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# note; this example downloads data when first run, and typically takes ~60seconds

# store the downloaded files in the following directory. Here, the temporary file
# directory is used. Alternatively, consider storing this data in a more permanent location.
# e.g. output_dir="~/data/go" on unix systems or output_dir="C:/data/go" on Windows
output_dir = tempdir()

## first run the default example from test_genesets() to obtain geneset results
datasets = download_goat_manuscript_data(output_dir)
genelist = datasets$`Wingo 2020:mass-spec:PMID32424284`
genesets_asis = download_genesets_goatrepo(output_dir)
genesets_filtered = filter_genesets(genesets_asis, genelist)
result = test_genesets(genesets_filtered, genelist, method = "goat",
  score_type = "effectsize", padj_method = "bonferroni", padj_cutoff = 0.05)

# generate lollipop charts for each GO domain (CC/BP/MF), with geneset -log10
# adjusted p-value on the x-axis and color-coding by geneset up/down-regulation
plot_lollipop(
  result, output_dir, plot_type = "lollipop", topn = 50,
  score_xaxis = "minlogp", score_color = "updown"
)

# alternatively, as a barplot
plot_lollipop(
  result, output_dir, plot_type = "barplot", topn = 50,
  score_xaxis = "minlogp", score_color = "updown"
)

# alternatively, color-code genesets by enrichment of significant genes using
# parameter `score_color="oddsratio"` . See further `score_geneset_oddsratio`
# function documentation for definition/computation of this score.
plot_lollipop(
  result, output_dir, plot_type = "lollipop", topn = 50,
  score_xaxis = "minlogp", score_color = "oddsratio"
)

</code></pre>

<hr>
<h2 id='plot_network'>plot geneset distance matrix as a network</h2><span id='topic+plot_network'></span>

<h3>Description</h3>

<p>plot geneset distance matrix as a network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_network(
  clusters,
  src,
  show_clusters = TRUE,
  show_text = FALSE,
  topn_edges = 5,
  clr_default = "#29b6f6",
  node_color_palette = goat::gg_color_hue
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_network_+3A_clusters">clusters</code></td>
<td>
<p>result from <code>cluster_genesets</code></p>
</td></tr>
<tr><td><code id="plot_network_+3A_src">src</code></td>
<td>
<p>source property (e.g. &quot;GO_CC&quot;)</p>
</td></tr>
<tr><td><code id="plot_network_+3A_show_clusters">show_clusters</code></td>
<td>
<p>boolean value</p>
</td></tr>
<tr><td><code id="plot_network_+3A_show_text">show_text</code></td>
<td>
<p>boolean value</p>
</td></tr>
<tr><td><code id="plot_network_+3A_topn_edges">topn_edges</code></td>
<td>
<p>topN edges to retain per geneset (typically 5~8)</p>
</td></tr>
<tr><td><code id="plot_network_+3A_clr_default">clr_default</code></td>
<td>
<p>default color for the network, used only when <code>show_clusters</code> is set to <code>FALSE</code></p>
</td></tr>
<tr><td><code id="plot_network_+3A_node_color_palette">node_color_palette</code></td>
<td>
<p>function with 1 parameter, N, that returns N colors (default is goat function <code>gg_color_hue</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 object
</p>

<hr>
<h2 id='plot_volcano'>For each provided geneset, a volcano plot of all genelist log2fc and p-values with respective geneset constituents highlighted</h2><span id='topic+plot_volcano'></span>

<h3>Description</h3>

<p>For each provided geneset, a volcano plot of all genelist log2fc and p-values with respective geneset constituents highlighted
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_volcano(
  x,
  genelist,
  plot = TRUE,
  topn_labels = 0,
  color_default = "#B0B0B0",
  color_highlight = "#ef5350",
  color_label = "#000000",
  pointsize = 2,
  pointalpha = 0.75,
  labelsize = 7
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_volcano_+3A_x">x</code></td>
<td>
<p>a subset of the results from <code>test_genesets</code> function, see example</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_genelist">genelist</code></td>
<td>
<p>input genelist, must contain columns 'gene', 'log2fc' and 'pvalue_adjust' (not! log transformed). If parameter topn_labels is provided, also include a character column 'symbol' that contains gene names/symbols/labels</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_plot">plot</code></td>
<td>
<p>if <code>TRUE</code>, will directly show the plots. if <code>FALSE</code>, returns a list of ggplot objects corresponding to rows in the input <code>result</code> parameter</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_topn_labels">topn_labels</code></td>
<td>
<p>for how many genes that overlap between genelist and a geneset should we plot the gene symbol? This requires a column 'symbol' in the genelist parameter (default: 0)</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_color_default">color_default</code></td>
<td>
<p>color for genes that are not part of a geneset (default: grey)</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_color_highlight">color_highlight</code></td>
<td>
<p>color used to highlight geneset constituents (default: red)</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_color_label">color_label</code></td>
<td>
<p>provided that topn_labels is set, this is the color of the text labels (default: black)</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_pointsize">pointsize</code></td>
<td>
<p>size of the dots, this parameter is passed to geom_point (default: 2)</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_pointalpha">pointalpha</code></td>
<td>
<p>alpha of the dots, this parameter is passed to geom_point (default: 0.75)</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_labelsize">labelsize</code></td>
<td>
<p>provided that topn_labels is set, this is the text size (in pt) for the labels (default: 7)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if <code>plot==FALSE</code>, a list of ggplot2 objects. Otherwise, does not return any value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# note; this example downloads data when first run, and typically takes ~60seconds

# store the downloaded files in the following directory. Here, the temporary file
# directory is used. Alternatively, consider storing this data in a more permanent location.
# e.g. output_dir="~/data/goat" on unix systems or output_dir="C:/data/goat" on Windows
output_dir = tempdir()

## first run the default example from test_genesets() to obtain geneset results
datasets = download_goat_manuscript_data(output_dir)
genelist = datasets$`Wingo 2020:mass-spec:PMID32424284`
genesets_asis = download_genesets_goatrepo(output_dir)
genesets_filtered = filter_genesets(genesets_asis, genelist)
result = test_genesets(genesets_filtered, genelist, method = "goat",
  score_type = "effectsize", padj_method = "bonferroni", padj_cutoff = 0.05)

## example 1; select top10 GO CC terms from the geneset testing results
result_subset = result |&gt; filter(source == "GO_CC") |&gt; arrange(pvalue) |&gt; head(n = 10)
pdf(paste0(output_dir, "/volcano_CC_top10.pdf"), width = 4, height = 4)
plot_volcano(result_subset, genelist)
dev.off()

## example 2;, select small genesets that are significant and have
## near-exclusive enrichment in either up up/down-regulated genes
# first, add geneset directionality scores to our results
result = score_geneset_directionality(result, genelist)
# next, subset the geneset result table
result_subset = result |&gt;
  filter(signif &amp; ngenes &lt;= 50 &amp; abs(score_directionality_rank) &gt; 0.6) |&gt;
  arrange(pvalue_adjust)
# finally, create plots. Note that the genelist contains a column 'symbol'
# which we use here to print labels for the topN genes per plotted geneset
pdf(paste0(output_dir, "/volcano_signif_ngenes50_directionality06.pdf"), width = 4, height = 4)
plot_volcano(result_subset, genelist, topn_labels = 10)
dev.off()

</code></pre>

<hr>
<h2 id='rankscore'>compute rank^2 scores and rescale these between 0~1000 (with further precision captured by decimals)</h2><span id='topic+rankscore'></span>

<h3>Description</h3>

<p>rank^2 can yield huge values, e.g. a large genelist of N=50000 genes would imply max gene score = 50000^2 = 2.5e+09.
Thus we rescale these scores between 0~1000 so downstream applications (e.g. sum of 10000 gene scores) don't explode to huge numbers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rankscore(x, sort1, sort2, sort3, colname)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rankscore_+3A_x">x</code></td>
<td>
<p>data.frame to sort (i.e. the genelist)</p>
</td></tr>
<tr><td><code id="rankscore_+3A_sort1">sort1</code></td>
<td>
<p>numeric vector of length <code>nrow(x)</code> to sort by first/primarily (descending order, higher value gets better result score)</p>
</td></tr>
<tr><td><code id="rankscore_+3A_sort2">sort2</code></td>
<td>
<p>numeric vector of length <code>nrow(x)</code> to sort by for breaking ties (descending order, higher value gets better result score)</p>
</td></tr>
<tr><td><code id="rankscore_+3A_sort3">sort3</code></td>
<td>
<p>numeric vector of length <code>nrow(x)</code> to sort by for breaking ties (descending order, higher value gets better result score)</p>
</td></tr>
<tr><td><code id="rankscore_+3A_colname">colname</code></td>
<td>
<p>name for result column in <code>x</code>  (overwritten if already exists)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame <code>x</code> with added column <code>colname</code>, containing gene scores between 0 and 1000
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  x = data.frame(gene=c(1, 2, 3, 4, 5),
                 pvalue=c(0.01, 1, 1, 0.1, 1),
                 effectsize=c(-2, 0.25, 0.5, 1, 0.25))
  print(x, row.names = FALSE)
  print(rankscore(x, sort1 = -1*x$pvalue, sort2 = abs(x$effectsize),
        sort3 = x$gene, colname="score") |&gt;
    arrange(desc(score)), row.names = FALSE)
</code></pre>

<hr>
<h2 id='rankscore_fixed_order'>Gene score array, from low to high scores</h2><span id='topic+rankscore_fixed_order'></span>

<h3>Description</h3>

<p>Gene score array, from low to high scores
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rankscore_fixed_order(n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rankscore_fixed_order_+3A_n">n</code></td>
<td>
<p>genelist length</p>
</td></tr>
</table>


<h3>Value</h3>

<p>result of <code>(1:n)^2 / (n^2 / 1000)</code>
</p>

<hr>
<h2 id='reduce_genesets'>Reduce the set of significant genesets to a minimum</h2><span id='topic+reduce_genesets'></span>

<h3>Description</h3>

<p>Analyses are performed independently per 'source' of genesets. The result of this function is the geneset table with a newly appended column 'signif_and_reduced'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reduce_genesets(
  clusters,
  simscore_threshold = 0.9,
  universe_fraction = 0.25,
  signifgenes_fraction = 0.9
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reduce_genesets_+3A_clusters">clusters</code></td>
<td>
<p>results from <code>cluster_genesets()</code></p>
</td></tr>
<tr><td><code id="reduce_genesets_+3A_simscore_threshold">simscore_threshold</code></td>
<td>
<p>similarity score (0~1) that is required to consider one geneset to be a &quot;parent term&quot; of another.
Setting a lower value will yield fewer genesets / stronger summarization.
Typical settings for this parameter are 0.8~0.99 (0.9 is default)</p>
</td></tr>
<tr><td><code id="reduce_genesets_+3A_universe_fraction">universe_fraction</code></td>
<td>
<p>discard genesets that cover more than X fraction of all genes in the universe (unique set of genes covered by all significant genesets).
Setting this to 0.25 will deprioritize genesets that cover 25% of all genes (in significant genesets).
This prevents very generic GO terms like &quot;protein-containing complex&quot; to be included in results.
Typical settings for this parameter are 0.1~0.5  (0.25 is default)</p>
</td></tr>
<tr><td><code id="reduce_genesets_+3A_signifgenes_fraction">signifgenes_fraction</code></td>
<td>
<p>the minimum fraction of &quot;foreground genes&quot; ('genes_signif' column) found across all significant genesets that should be covered by the reduced geneset collection.
This parameter doesn't do anything if there are fewer than 5 &quot;foreground genes&quot; alltogether.
Typical settings for this parameter are 0.75~0.95 (0.9 is default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the genesets table from the <code>clusters</code> parameter, with results in column &quot;signif_and_reduced&quot;
</p>

<hr>
<h2 id='save_genesets'>Write a geneset table to file.</h2><span id='topic+save_genesets'></span>

<h3>Description</h3>

<p>Works for any filtered geneset table generated by this package, e.g. results from any of these functions;
<code>filter_genesets()</code>, <code>test_genesets()</code> or <code>simplify_genesets()</code>.
</p>
<p>The genelist table is required to 1) lookup gene symbols and 2) determine the ordering of genes within-geneset.
The latter makes it such that the output table shows the most important genes (in context of user's data) first.
</p>
<p>All output columns that list gene identifiers or symbols are trimmed to 25000 characters. Depending on your input
gene identifier type and dataset filtering settings (e.g. max geneset size), this might lead to some truncation.
For example, 15 characters per ensembl gene ID (if these are used in provided genesets instead of default NCBI
Entrez) and 1 char as delimiter, only the top ~1500 ensembl gene identifiers are included. If your geneset
filtering allows for larger genesets and you want to check if any gene identifiers are lost upon saving,
keep an eye out for the trailing ellipsis ('...'). Because genes in each geneset of the output table are sorted
by order of importance as indicated in the genelist table, this shouldn't be an issue in typical use-cases
where the output table is to check the topN most important/significant genes of a geneset of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save_genesets(x, genelist, filename, arrange_genes = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="save_genesets_+3A_x">x</code></td>
<td>
<p>result from <code>filter_genesets()</code>, <code>test_genesets()</code> or <code>simplify_genesets()</code>. When saving the latter, output will include clusters assigned to each significant geneset</p>
</td></tr>
<tr><td><code id="save_genesets_+3A_genelist">genelist</code></td>
<td>
<p>same as provided for <code>test_genesets()</code>. A column named 'symbol' is required, it'll be used to pretty-print gene symbols per geneset in the output table</p>
</td></tr>
<tr><td><code id="save_genesets_+3A_filename">filename</code></td>
<td>
<p>full path to the output file. Supported file extensions; csv, tsv, xlsx. Optionally, set to NA to not write to disk and return the result table instead</p>
</td></tr>
<tr><td><code id="save_genesets_+3A_arrange_genes">arrange_genes</code></td>
<td>
<p>set to TRUE (default) to arrange the genelist table by best p-value on top (if column 'pvalue' exists), alternatively by descending absolute effectsize (if no pvalue but effectize is available). Set FALSE to use sorting of the genelist table as-is</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if <code>filename</code> is <code>NA</code>, returns the validated and formatted geneset result table. Otherwise, writes the table to file and does not return a value
</p>

<hr>
<h2 id='score_geneset_directionality'>Compute a score between -1 and 1 representing the proportion of up- or down-regulated genes for each geneset, weighted by gene effectsizes</h2><span id='topic+score_geneset_directionality'></span>

<h3>Description</h3>

<p>Importantly, the scope/utility of this score is limited to help users post-hoc filter for genesets that contain mostly up/down-regulated genes.
However, this might not coincide with the geneset pvalues / significance. For example,
genesets may exclusively contain genes with a positive effectsize but at the same time these can all be minor effects/values and thus the geneset
is not significant or less significant than other genesets with the exact same &quot;directionality score&quot;.
For example, genesets may contain both up- and down-regulated genes but still be significant when testing with GOAT and using <code>score_type='effectsize'</code>
</p>
<p>The scores computed with this function can help in post-hoc interpretation of GOAT results to further narrow down all significant genesets to a
subset with strong directionality. For example, after <code>test_genesets()</code> we can filter the results for
A) significant genesets and B) that contain at most N genes and C) that are near-exclusively up/down-regulated.
Bringing this all together (also useful for other types of geneset testing, like ORA, score_type=&quot;pvalue&quot;, etc);
<code>result = test_genesets(genesets_filtered, genelist, method = "goat", score_type = "effectsize", padj_method = "bonferroni", padj_cutoff = 0.05)</code>
<code>result = score_geneset_directionality(result, genelist)</code>
<code>result |&gt; filter(signif &amp; ngenes &lt;= 100 &amp; abs(score_directionality_rank) &gt;= 0.95)</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score_geneset_directionality(genesets, genelist)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="score_geneset_directionality_+3A_genesets">genesets</code></td>
<td>
<p>tibble with genesets, must contain columns 'source', 'id', 'genes'</p>
</td></tr>
<tr><td><code id="score_geneset_directionality_+3A_genelist">genelist</code></td>
<td>
<p>tibble with genes, must contain columns 'gene', 'effectsize'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>input <code>genesets</code> with results in 3 columns;
<code>score_directionality_rank</code> is the weighted gene score where gene values are the sign of their effectsize and weights are linearly proportional to their inverse ranks in the input genelist.
<code>score_directionality_rank2</code> is similar, but now using rank^2 gene weights to boost the influence of most-important genes in the input genelist.
<code>score_directionality_value</code> uses the absolute gene effectsizes as gene weights
Note that the latter is least robust as it depends on the scaling of input data!
</p>


<h3>score definitions;</h3>

<p>geneset directionality score = weighted mean of respective genes,
where gene weights are 1 minus their relative rank in up/down-regulation (depending on negative/positive effectsize)
and the value for each gene is -1 or 1 depending on up/down-regulation (sign of effectsize).
</p>


<h3>pseudocode;</h3>


<ol>
<li><p> gene values and weights
A) gene weight between 0 and 1 for the subset of upregulated genes / positive effectsizes;
</p>
</li></ol>


<ul>
<li><p> r = for the subset of genes with effectsize &gt; 0, compute gene rank (1 = highest effectsize, N = smallest effectsize that is greater than zero)
</p>
</li>
<li><p> weight = 1 - r/max(r)
B) analogous to (A) for the subset of genes with negative effectsize
C) result per gene: value = sign of its effectsize, weight = 0 if effectsize 0, otherwise respective weights from (A) or (B)
</p>
</li></ul>


<ol>
<li><p> geneset score_directionality = weighted mean over values/weights of respective genes
</p>
</li></ol>


<hr>
<h2 id='score_geneset_oddsratio'>Compute odds-ratio for each geneset</h2><span id='topic+score_geneset_oddsratio'></span>

<h3>Description</h3>

<p>gs_signif = number of significant genes in geneset G that intersect with user's genelist (i.e. foreground genes in G)
gs_all  = number of genes in geneset G that intersect with user's genelist (i.e. foreground+background genes in G)
k_signif = total number of <em>significant</em> genes in user's genelist
k_all  = total number of genes in user's genelist
</p>
<p>gs_signif/gs_all = ratio of foreground genes in geneset G
k_signif/k_all = ratio of overall foreground genes (i.e. expected value for a random geneset)
</p>
<p>oddsratio = (gs_signif/gs_all) / (k_signif/k_all)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score_geneset_oddsratio(genesets, genelist)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="score_geneset_oddsratio_+3A_genesets">genesets</code></td>
<td>
<p>tibble with genesets, must contain columns 'source', 'id', 'ngenes', 'ngenes_signif'</p>
</td></tr>
<tr><td><code id="score_geneset_oddsratio_+3A_genelist">genelist</code></td>
<td>
<p>tibble with genes, must contain columns 'gene', 'signif'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>input <code>genesets</code> with results in column &quot;score_oddsratio&quot;
</p>

<hr>
<h2 id='string_trunc_right'>simple string truncation</h2><span id='topic+string_trunc_right'></span>

<h3>Description</h3>

<p>replacement for stringr::trunc() so we don't need a package dependency for just 1 function (our code was adapter therefrom)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>string_trunc_right(string, width, trim_left = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="string_trunc_right_+3A_string">string</code></td>
<td>
<p>string that should be truncated</p>
</td></tr>
<tr><td><code id="string_trunc_right_+3A_width">width</code></td>
<td>
<p>desired max length</p>
</td></tr>
<tr><td><code id="string_trunc_right_+3A_trim_left">trim_left</code></td>
<td>
<p>instead of right trunc (default), do left instead</p>
</td></tr>
</table>


<h3>Value</h3>

<p>truncated variant of input <code>string</code>
</p>

<hr>
<h2 id='symbol_to_entrez'>Map the the symbol column in a table to HGNC human gene IDs by matching official gene symbols and synonyms</h2><span id='topic+symbol_to_entrez'></span>

<h3>Description</h3>

<p>Map the the symbol column in a table to HGNC human gene IDs by matching official gene symbols and synonyms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symbol_to_entrez(x, hgnc)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="symbol_to_entrez_+3A_x">x</code></td>
<td>
<p>a data.table with a column symbol</p>
</td></tr>
<tr><td><code id="symbol_to_entrez_+3A_hgnc">hgnc</code></td>
<td>
<p>HGNC lookup table from <code>hgnc_idmap_table()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>entrez gene IDs are returned in the &quot;gene&quot; column of table <code>x</code>. Additionally, columns &quot;entrez_id&quot;, &quot;hgnc_id&quot; and &quot;hgnc_symbol&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # TODO: update the filename to your downloaded file
  # download instructions in the documentation of `hgnc_idmap_table()`
  f = "C:/DATA/HGNC/hgnc_complete_set.txt"

  if(file.exists(f)) {
    df = data.frame(symbol = c("vamp2", "STXBP1", "UNC18", NA, "PSD95", "NOT-A-GENE"))
    hgnc = hgnc_idmap_table(f)
    df = symbol_to_entrez(df, hgnc)
    print(df)
  }
</code></pre>

<hr>
<h2 id='taxonomy_identifiers'>Lookup table for taxonomy identifiers, their names and respective Bioconductor packages</h2><span id='topic+taxonomy_identifiers'></span>

<h3>Description</h3>

<p>To prevent adding a huge number of dependencies to this R package,
as the respective bioconductor_package also has to be added to the DESCRIPTION &gt; Suggests,
we've only included the limited set of commonly used organisms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonomy_identifiers()
</code></pre>

<hr>
<h2 id='test_genesets'>Perform geneset enrichment testing using any supported method</h2><span id='topic+test_genesets'></span>

<h3>Description</h3>

<p>Perform geneset enrichment testing using any supported method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_genesets(
  genesets,
  genelist,
  method,
  padj_method = "BH",
  padj_sources = TRUE,
  padj_cutoff = 0.01,
  padj_min_signifgenes = 0L,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="test_genesets_+3A_genesets">genesets</code></td>
<td>
<p>tibble with genesets, must contain columns 'source', 'source_version', 'id', 'name', 'genes', 'ngenes', 'ngenes_signif'</p>
</td></tr>
<tr><td><code id="test_genesets_+3A_genelist">genelist</code></td>
<td>
<p>tibble with genes, must contain column 'gene' and 'test'. gene = character column, which are matched against list column 'genes' in genesets tibble. test = boolean column (you can set all to FALSE if not performing Fisher-exact or hypergeometric test downstream)</p>
</td></tr>
<tr><td><code id="test_genesets_+3A_method">method</code></td>
<td>
<p>method for overrepresentation analysis. Options: &quot;goat&quot;, &quot;hypergeometric&quot;, &quot;fisherexact&quot;, &quot;fisherexact_ease&quot;, &quot;gsea&quot;, &quot;idea&quot;</p>
</td></tr>
<tr><td><code id="test_genesets_+3A_padj_method">padj_method</code></td>
<td>
<p>first step of multiple testing correction; method for p-value adjustment, passed to <code>stats::p.adjust()</code> via <code>padjust_genesets()</code>, e.g. set &quot;BH&quot; to compute FDR adjusted p-values (default) or &quot;bonferroni&quot; for a more stringent procedure</p>
</td></tr>
<tr><td><code id="test_genesets_+3A_padj_sources">padj_sources</code></td>
<td>
<p>second step of multiple testing correction; apply Bonferroni adjustment to all p-values according to the number of geneset sources that were tested. Boolean parameter, set TRUE to enable (default) or FALSE to disable</p>
</td></tr>
<tr><td><code id="test_genesets_+3A_padj_cutoff">padj_cutoff</code></td>
<td>
<p>cutoff for adjusted p-value, <code>signif</code> column is set to TRUE for all values lesser-equals</p>
</td></tr>
<tr><td><code id="test_genesets_+3A_padj_min_signifgenes">padj_min_signifgenes</code></td>
<td>
<p>if a value larger than zero is provided, this will perform additional post-hoc filtering; after p-value adjustment, set the <code>pvalue_adjust</code> to <code>NA</code> and <code>signif</code> to <code>FALSE</code> for all genesets with fewer than <code>padj_min_signifgenes</code> 'input genes that were significant' (<code>ngenes_signif</code> column in genesets table).
So this does not affect the accuracy of estimated p-values, in contrast to prefiltering genesets prior to p-value computation or adjusting p-values</p>
</td></tr>
<tr><td><code id="test_genesets_+3A_...">...</code></td>
<td>
<p>further parameters are passed to the respective stats method</p>
</td></tr>
</table>


<h3>Details</h3>

<p>After application of the enrichment testing algorithm (e.g. GOAT, ORA or GSEA), multiple testing correction is applied to obtain adjusted p-values using <code>padjust_genesets</code>.
That function will first apply the specified pvalue adjustment procedure in the <code>padj_method</code> parameter within each 'source' in the genesets table. Second, it applies Bonferroni adjustment to all p-values according to the number of different geneset sources that were tested (or set <code>padj_sources = FALSE</code> to disable).
</p>
<p>For example, if the input is a genesets table that contains GO_CC, GO_BP and GO_MF genesets, first multiple testing correction is applied within each source (e.g. using FDR if so desired) and afterwards a Bonferroni correction is applied based on 3 repeated analyses.
</p>
<p>Note that this is more rigorous than typical GO tools; hypothetically, one could split all GO_CC pathways into 1000 different databases/'sources' and then run enrichment testing. Consequently, the multiple testing burden is reduced if one doesn't adjust p-values for the number of 'sources' as we do here.
</p>


<h3>Value</h3>

<p>the input <code>genesets</code>, with results stored in columns 'pvalue', 'pvalue_adjust' and 'signif'
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#' # note; this example downloads data when first run, and typically takes ~60seconds

## Basic example for a complete GOAT workflow
# Downloads test data to your computer and stores it at current working directory
# Refer to the GitHub documentation for elaborate documentation and a worked example

# store the downloaded files in the following directory. Here, the temporary file
# directory is used. Alternatively, consider storing this data in a more permanent location.
# e.g. output_dir="~/data/go" on unix systems or output_dir="C:/data/go" on Windows
output_dir = tempdir()

# download an example gene list
datasets = download_goat_manuscript_data(output_dir)
genelist = datasets$`Wingo 2020:mass-spec:PMID32424284`

# download GO genesets
genesets_asis = download_genesets_goatrepo(output_dir)

# filter genesets for sufficient overlap with the genelist, then apply GOAT
genesets_filtered = filter_genesets(genesets_asis, genelist)
result = test_genesets(genesets_filtered, genelist, method = "goat",
  score_type = "effectsize", padj_method = "bonferroni", padj_cutoff = 0.05)

# print first 10 rows of the result table
print(result |&gt; select(source, name, ngenes, pvalue_adjust) |&gt; utils::head(n=10))

</code></pre>

<hr>
<h2 id='test_genesets_fisherexact'>Geneset ORA using Fisher-exact test</h2><span id='topic+test_genesets_fisherexact'></span>

<h3>Description</h3>

<p>In most cases, it's more convenient to call the more generic <code>test_genesets</code> function which also deals with multiple-testing correction (per geneset source)
</p>
<p>It is assumed that the <code>genesets</code> and <code>genelist</code> parameters are in sync, i.e. <code>genesets</code> provided
here is the result of the <code>filter_genesets()</code> function (using the same <code>genelist</code> table)
</p>
<p>Same as hypergeometric for non-EASE, but slower because stats::fisher.test isn't vectorized
</p>
<p>Only genesets with at least 1 significant gene are subjected to statistical testing (e.g. NA is returned for genesets without significant genes)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_genesets_fisherexact(
  genesets,
  genelist,
  require_nsignif = 1L,
  use_ease = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="test_genesets_fisherexact_+3A_genesets">genesets</code></td>
<td>
<p>tibble with genesets, must contain columns 'id', 'ngenes' and 'ngenes_signif'</p>
</td></tr>
<tr><td><code id="test_genesets_fisherexact_+3A_genelist">genelist</code></td>
<td>
<p>tibble with genes, must contain column 'signif'. The number of rows in this table (where signif is not NA)
is assumed to be the total number of tested genes, the number of rows where signif==TRUE is assumed the total number of significant genes.</p>
</td></tr>
<tr><td><code id="test_genesets_fisherexact_+3A_require_nsignif">require_nsignif</code></td>
<td>
<p>minimum number of 'signif genes' that overlap with a geneset; <code>NA</code> pvalues are returned for genesets with <code>ngenes_signif &lt;= require_nsignif</code>.
This function 'prefilters' genesets, so beware that this will influence downstream multiple testing correction. Default is 1</p>
</td></tr>
<tr><td><code id="test_genesets_fisherexact_+3A_use_ease">use_ease</code></td>
<td>
<p>use a more conservative score coined by DAVID online tools @ https://david.ncifcrf.gov/helps/functional_annotation.html#fisher</p>
</td></tr>
</table>


<h3>Value</h3>

<p>input <code>genesets</code> table with results in the &quot;pvalue&quot; column
</p>


<h3>See Also</h3>

<p><code>test_genesets</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# note; this example downloads data when first run, and typically takes ~60seconds

# store the downloaded files in the following directory. Here, the temporary file
# directory is used. Alternatively, consider storing this data in a more permanent location.
# e.g. output_dir="~/data/goat" on unix systems or output_dir="C:/data/goat" on Windows
output_dir = tempdir()

## first run the default example from test_genesets() to obtain input data
datasets = download_goat_manuscript_data(output_dir)
genelist = datasets$`Wingo 2020:mass-spec:PMID32424284`
genesets_asis = download_genesets_goatrepo(output_dir)
genesets_filtered = filter_genesets(genesets_asis, genelist)

## example: same results between Fisher-exact and hypergeometric tests
result_hg = test_genesets_hypergeometric(genesets_filtered, genelist, require_nsignif = 3L)
result_fe = test_genesets_fisherexact(genesets_filtered, genelist, require_nsignif = 3L)
all.equal(result_hg$pvalue, result_fe$pvalue)

</code></pre>

<hr>
<h2 id='test_genesets_goat_bootstrap'>Naive GOAT variant where we estimate null parameters for each geneset size independently</h2><span id='topic+test_genesets_goat_bootstrap'></span>

<h3>Description</h3>

<p>In typical use-cases, one applies <code>test_genesets()</code> instead with parameter <code>method="goat"</code> ,
which in turn will use <code>test_genesets_goat_precomputed</code> (and not this function).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_genesets_goat_bootstrap(
  genesets,
  genelist,
  score_type,
  niter = 5e+05,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="test_genesets_goat_bootstrap_+3A_genesets">genesets</code></td>
<td>
<p>see <code>test_genesets_goat_precomputed</code></p>
</td></tr>
<tr><td><code id="test_genesets_goat_bootstrap_+3A_genelist">genelist</code></td>
<td>
<p>see <code>test_genesets_goat_precomputed</code></p>
</td></tr>
<tr><td><code id="test_genesets_goat_bootstrap_+3A_score_type">score_type</code></td>
<td>
<p>see <code>test_genesets_goat_precomputed</code></p>
</td></tr>
<tr><td><code id="test_genesets_goat_bootstrap_+3A_niter">niter</code></td>
<td>
<p>integer number of bootstrap iterations; at least 10000, at most 5000000</p>
</td></tr>
<tr><td><code id="test_genesets_goat_bootstrap_+3A_verbose">verbose</code></td>
<td>
<p>boolean, create debug plots</p>
</td></tr>
</table>


<h3>Value</h3>

<p>see <code>test_genesets_goat_precomputed</code>
</p>

<hr>
<h2 id='test_genesets_goat_fitfunction'>Variant of the main GOAT function <code>test_genesets_goat_precomputed</code> that does not use previously prepared parameters</h2><span id='topic+test_genesets_goat_fitfunction'></span>

<h3>Description</h3>

<p>In typical use-cases, one applies <code>test_genesets()</code> instead with parameter <code>method="goat"</code> ,
which in turn will use <code>test_genesets_goat_precomputed</code> (and not this function).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_genesets_goat_fitfunction(
  genesets,
  genelist,
  score_type,
  niter = 5e+05,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="test_genesets_goat_fitfunction_+3A_genesets">genesets</code></td>
<td>
<p>see <code>test_genesets_goat_precomputed</code></p>
</td></tr>
<tr><td><code id="test_genesets_goat_fitfunction_+3A_genelist">genelist</code></td>
<td>
<p>see <code>test_genesets_goat_precomputed</code></p>
</td></tr>
<tr><td><code id="test_genesets_goat_fitfunction_+3A_score_type">score_type</code></td>
<td>
<p>see <code>test_genesets_goat_precomputed</code></p>
</td></tr>
<tr><td><code id="test_genesets_goat_fitfunction_+3A_niter">niter</code></td>
<td>
<p>integer number of bootstrap iterations; at least 10000, at most 5000000</p>
</td></tr>
<tr><td><code id="test_genesets_goat_fitfunction_+3A_verbose">verbose</code></td>
<td>
<p>boolean, create debug plots</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Optionally, use this function to ignore precomputed/bundled null distribution estimates and perform new permutations and function fitting
(e.g. because you want to test the effect of huge niter parameter, but beware of RAM requirements)
</p>


<h3>Value</h3>

<p>see <code>test_genesets_goat_precomputed</code>
</p>

<hr>
<h2 id='test_genesets_goat_precomputed'>Test geneset enrichment with the Geneset Ordinal Association Test (GOAT) algorithm</h2><span id='topic+test_genesets_goat_precomputed'></span>

<h3>Description</h3>

<p>In most cases, it's more convenient to call the more generic <code>test_genesets</code> function which also applies multiple-testing correction (per geneset source) to the geneset p-values computed by this function.
</p>
<p>This is the canonical geneset test function for GOAT that uses precomputed null distributions that are bundled with the GOAT package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_genesets_goat_precomputed(genesets, genelist, score_type)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="test_genesets_goat_precomputed_+3A_genesets">genesets</code></td>
<td>
<p>genesets data.frame, must contain columns; &quot;source&quot;, &quot;id&quot;, &quot;genes&quot;, &quot;ngenes&quot;</p>
</td></tr>
<tr><td><code id="test_genesets_goat_precomputed_+3A_genelist">genelist</code></td>
<td>
<p>genelist data.frame, must contain columns &quot;gene&quot; and &quot;pvalue&quot;/&quot;effectsize&quot; (depending on parameter <code>score_type</code>)</p>
</td></tr>
<tr><td><code id="test_genesets_goat_precomputed_+3A_score_type">score_type</code></td>
<td>
<p>how to compute gene scores?
Option &quot;pvalue&quot; uses values from the pvalue column in <code>genelist</code> in a one-way test for enrichment; lower p-value is better
Option &quot;effectsize&quot; uses values from the effectsize column in <code>genelist</code> in a two-way test for enrichment; is a geneset enriched in either down- or up-regulated genes?
Option &quot;effectsize_abs&quot; uses values from the effectsize column in <code>genelist</code> in a one-way test for enrichment; is a geneset enriched when testing absolute effectsizes?
Option &quot;effectsize_up&quot; uses values from the effectsize column in <code>genelist</code> in a one-way test for enrichment; is a geneset enriched in up-regulated genes?  (i.e. positive effectsize)
Option &quot;effectsize_down&quot; uses values from the effectsize column in <code>genelist</code> in a one-way test for enrichment; is a geneset enriched in down-regulated genes?  (i.e. negative effectsize)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>input <code>genesets</code> table with results in the &quot;pvalue&quot;, &quot;score_type&quot; columns.
&quot;zscore&quot; column:
A standardized z-score is computed from geneset p-values + effectsize direction (up/down) if tested.
Importantly, we here return standardized z-scores because the GOAT geneset score (mean of gene scores) is relative to the respective geneset-size-matched null distributions (a skewed normal)!
In contrast, the standardized z-scores are comparable between genesets (as are the pvalues).
</p>
<p>Only if either (or both) the effectsize-up/down was tested, the direction of regulation has been tested (effectsize_abs and pvalue score types are agnostic to up/down regulation).
So when score_type was set to any of effectsize/effectsize_down/effectsize_up, the z-scores are negative values in case the &quot;score_type&quot; output column is &quot;effectsize_down&quot;.
</p>


<h3>See Also</h3>

<p><code>test_genesets</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# note; this example downloads data when first run, and typically takes ~60seconds

# store the downloaded files in the following directory. Here, the temporary file
# directory is used. Alternatively, consider storing this data in a more permanent location.
# e.g. output_dir="~/data/goat" on unix systems or output_dir="C:/data/goat" on Windows
output_dir = tempdir()

## first run the default example from test_genesets() to obtain input data
datasets = download_goat_manuscript_data(output_dir)
genelist = datasets$`Wingo 2020:mass-spec:PMID32424284`
genesets_asis = download_genesets_goatrepo(output_dir)
genesets_filtered = filter_genesets(genesets_asis, genelist)

### we here compare GOAT with precomputed null distributions against
### a GOAT function that performs bootstrapping to compute null distributions on-demand

# apply goat with precomputed null (default) and goat with on-demand bootstrapping
result_precomputed = test_genesets(genesets_filtered, genelist, method = "goat",
  score_type = "effectsize", padj_method = "bonferroni", padj_cutoff = 0.05) |&gt;
  # undo sorting by p-value @ test_genesets(), instead sort by stable IDs
  arrange(source, id)
result_bootstrapped = test_genesets(genesets_filtered, genelist, method = "goat_bootstrap",
  score_type = "effectsize", padj_method = "bonferroni", padj_cutoff = 0.05, verbose = TRUE) |&gt;
  arrange(source, id)

# tables should align
stopifnot(result_precomputed$id == result_bootstrapped$id)
# no missing values
stopifnot(is.finite(result_precomputed$pvalue) &amp;
          is.finite(is.finite(result_bootstrapped$pvalue)))

# compare results
plot(result_precomputed$pvalue, result_bootstrapped$pvalue)
abline(0, 1, col=2)

plot(minlog10_fixzero(result_precomputed$pvalue),
     minlog10_fixzero(result_bootstrapped$pvalue))
abline(0, 1, col=2)

summary(minlog10_fixzero(result_precomputed$pvalue) -
        minlog10_fixzero(result_bootstrapped$pvalue))

</code></pre>

<hr>
<h2 id='test_genesets_gsea'>GSEA as implemented in the fgsea R package</h2><span id='topic+test_genesets_gsea'></span>

<h3>Description</h3>

<p>In most cases, it's more convenient to call the more generic <code>test_genesets</code> function which also deals with multiple-testing correction (per geneset source)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_genesets_gsea(
  genesets,
  genelist,
  score_type = NULL,
  parallel_threads = 1L,
  gseaParam = 1,
  nPermSimple = 50000,
  gsea_genelist_col = NULL,
  gsea_scoretype = NULL,
  random_seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="test_genesets_gsea_+3A_genesets">genesets</code></td>
<td>
<p>data.frame/tibble with geneset and gene columns</p>
</td></tr>
<tr><td><code id="test_genesets_gsea_+3A_genelist">genelist</code></td>
<td>
<p>data.frame/tibble with gene and score columns. Should contain columns gene and either pvalue or effectsize, depending on <code>score_type</code> parameter</p>
</td></tr>
<tr><td><code id="test_genesets_gsea_+3A_score_type">score_type</code></td>
<td>
<p>how to compute gene scores? options: &quot;pvalue&quot;, &quot;effectsize&quot;, &quot;custom&quot;.
Option &quot;pvalue&quot; uses -log10 transformed values from the pvalue column in <code>genelist</code>.
Option &quot;effectsize&quot; uses values from the effectsize column in <code>genelist</code> as-is.
Option &quot;custom&quot; expects 2 additional parameters; <code>gsea_genelist_col</code> should be a column name in <code>genelist</code> to be used for fGSEA (values used as-is), <code>gsea_scoretype</code> should be the respective value for the fGSEA scoreType parameter ('pos', 'neg' or 'std)</p>
</td></tr>
<tr><td><code id="test_genesets_gsea_+3A_parallel_threads">parallel_threads</code></td>
<td>
<p>number of threads to use for parallel processing. Set to 0 to automatically select all available processors/cores, set to 1 to disable (default) or to N to use N processes.
Note that multiprocessing sometimes breaks on RStudio + Windows, hence this parameter is set to 1 to disable multiprocessing by default for now</p>
</td></tr>
<tr><td><code id="test_genesets_gsea_+3A_gseaparam">gseaParam</code></td>
<td>
<p>passed to <code>fgsea::fgsea()</code>, from manual: &quot;GSEA parameter value, all gene-level statis are raised to the power of 'gseaParam' before calculation of GSEA enrichment scores.&quot;. default = 1. Further comments by fGSEA author at https://github.com/ctlab/fgsea/issues/45</p>
</td></tr>
<tr><td><code id="test_genesets_gsea_+3A_npermsimple">nPermSimple</code></td>
<td>
<p>passed to <code>fgsea::fgsea()</code>, from manual: &quot;Number of permutations in the simple fgsea implementation for preliminary estimation of P-values.&quot;. default = 50000 in this R package but 1000 by default in fGSEA v1.22.0; we observed much better accuracy in null simulations when increasing this from default 1k to 10k and further minor improvement towards 50k, hence the latter is our default</p>
</td></tr>
<tr><td><code id="test_genesets_gsea_+3A_gsea_genelist_col">gsea_genelist_col</code></td>
<td>
<p>optional, only used for <code>score_type</code> &quot;custom&quot;</p>
</td></tr>
<tr><td><code id="test_genesets_gsea_+3A_gsea_scoretype">gsea_scoretype</code></td>
<td>
<p>optional, only used for <code>score_type</code> &quot;custom&quot;</p>
</td></tr>
<tr><td><code id="test_genesets_gsea_+3A_random_seed">random_seed</code></td>
<td>
<p>the random seed that is passed to <code>set.seed()</code> in order to ensure fgsea results are reproducible. default: 123</p>
</td></tr>
</table>


<h3>Value</h3>

<p>input <code>genesets</code> table with results in the &quot;pvalue&quot;, &quot;score_type&quot; and &quot;gsea_nes&quot; columns
</p>


<h3>See Also</h3>

<p><code>test_genesets</code>
</p>

<hr>
<h2 id='test_genesets_hypergeometric'>Geneset ORA using hypergeometric test</h2><span id='topic+test_genesets_hypergeometric'></span>

<h3>Description</h3>

<p>In most cases, it's more convenient to call the more generic <code>test_genesets</code> function which also deals with multiple-testing correction (per geneset source)
</p>
<p>It is assumed that the <code>genesets</code> and <code>genelist</code> parameters are in sync, i.e. <code>genesets</code> provided
here is the result of the <code>filter_genesets()</code> function (using the same <code>genelist</code> table)
</p>
<p>Only genesets with at least 1 significant gene are subjected to statistical testing (e.g. NA is returned for genesets without significant genes)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_genesets_hypergeometric(genesets, genelist, require_nsignif = 1L)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="test_genesets_hypergeometric_+3A_genesets">genesets</code></td>
<td>
<p>tibble with genesets, must contain columns 'id', 'ngenes' and 'ngenes_signif'</p>
</td></tr>
<tr><td><code id="test_genesets_hypergeometric_+3A_genelist">genelist</code></td>
<td>
<p>tibble with genes, must contain column 'signif'. The number of rows in this table (where signif is not NA)
is assumed to be the total number of tested genes, the number of rows where signif==TRUE is assumed the total number of significant genes.</p>
</td></tr>
<tr><td><code id="test_genesets_hypergeometric_+3A_require_nsignif">require_nsignif</code></td>
<td>
<p>minimum number of 'signif genes' that overlap with a geneset; <code>NA</code> pvalues are returned for genesets with <code>ngenes_signif &lt;= require_nsignif</code>.
This function 'prefilters' genesets, so beware that this will influence downstream multiple testing correction. Default is 1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>input <code>genesets</code> table with results in the &quot;pvalue&quot; column
</p>


<h3>See Also</h3>

<p><code>test_genesets</code>
</p>

<hr>
<h2 id='treemap_data'>Construct tree and treemap data structures from geneset parent/child relations</h2><span id='topic+treemap_data'></span>

<h3>Description</h3>

<p>refer to the goat::treemap_plot() function for a complete example
</p>


<h3>Usage</h3>

<pre><code class='language-R'>treemap_data(
  geneset_ids,
  genesets,
  genesets_test_result,
  simplify = "none",
  toplevel_max_ngenes = NA
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="treemap_data_+3A_geneset_ids">geneset_ids</code></td>
<td>
<p>vector of geneset identifiers</p>
</td></tr>
<tr><td><code id="treemap_data_+3A_genesets">genesets</code></td>
<td>
<p>entire geneset table; typically the complete GO database</p>
</td></tr>
<tr><td><code id="treemap_data_+3A_genesets_test_result">genesets_test_result</code></td>
<td>
<p>geneset testing results; the output from <code>test_genesets()</code></p>
</td></tr>
<tr><td><code id="treemap_data_+3A_simplify">simplify</code></td>
<td>
<p>strategy for reducing the genesets returned in the treemap. Options;
&quot;leaf_only&quot; (most stringent, returns only leaves in the tree structure)
&quot;prune_singletons&quot; (remove parent terms that have exactly 1 child)
&quot;pvalue&quot; (remove parent terms where the child term p-value is at least 4 times better)
&quot;none&quot; (default; return all significant genesets that are not a &quot;grouping term&quot; in the treemap)</p>
</td></tr>
<tr><td><code id="treemap_data_+3A_toplevel_max_ngenes">toplevel_max_ngenes</code></td>
<td>
<p>groups in the treemap should not have more than this many genes ('ngenes' in geneset test results). If not set, this defaults to 50% of the total number of unique genes in the geneset test results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data structure needed for <code>treemap_plot()</code>
</p>

<hr>
<h2 id='treemap_plot'>Plot a treemap</h2><span id='topic+treemap_plot'></span>

<h3>Description</h3>

<p>simple wrapper around the treemap R package. To customize this plot, copy/paste its code and tweak parameters as desired
</p>


<h3>Usage</h3>

<pre><code class='language-R'>treemap_plot(x, label_group = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="treemap_plot_+3A_x">x</code></td>
<td>
<p><code>treemap_plotdata</code> data table that was computed by the <code>treemap_data()</code> function</p>
</td></tr>
<tr><td><code id="treemap_plot_+3A_label_group">label_group</code></td>
<td>
<p>set TRUE to show only group-level labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 object constructed by <code>treemap::treemap()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# note; this example downloads data when first run, and typically takes ~60seconds

# store the downloaded files in the following directory. Here, the temporary file
# directory is used. Alternatively, consider storing this data in a more permanent location.
# e.g. output_dir="~/data/goat" on unix systems or output_dir="C:/data/goat" on Windows
output_dir = tempdir()

## first run the default example from test_genesets() to obtain geneset results
datasets = download_goat_manuscript_data(output_dir)
genelist = datasets$`Wingo 2020:mass-spec:PMID32424284`
genesets_asis = download_genesets_goatrepo(output_dir)
genesets_filtered = filter_genesets(genesets_asis, genelist)
result = test_genesets(genesets_filtered, genelist, method = "goat",
  score_type = "effectsize", padj_method = "bonferroni", padj_cutoff = 0.05)

# subset GO CC results
x = result |&gt; filter(signif &amp; source == "GO_CC")
tm = treemap_data(
  geneset_ids = x$id,
  genesets = genesets_filtered,
  genesets_test_result = x,
  simplify = "leaf_only" # options: none/leaf_only/prune_singletons/pvalue
)
treemap_plot(tm$treemap_plotdata)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
