<!DOCTYPE html><html lang="en"><head><title>Help for package GOCompare</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<script type="text/javascript" src="mathjax-config.js"></script>
<script type="text/javascript" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GOCompare}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GOCompare-package'><p>GOCompare: An R package to compare GO terms of gene lists (categories)  and their orthologs</p></a></li>
<li><a href='#A_thaliana'><p>A thaliana functional enrichment analysis  of 2224 ortholog genes related</p>
to cancer-hallmarks</a></li>
<li><a href='#A_thaliana_compress'><p>A thaliana functional enrichment analysis results for</p>
&quot;AID&quot;,&quot;DCE&quot;,&quot;RCD&quot;,&quot;SPS&quot; cancer-hallmarks</a></li>
<li><a href='#compareGOspecies'><p>Visual representation for the results of functional</p>
enrichment analysis to compare two species and a series of categories</a></li>
<li><a href='#comparison_ex_compress'><p>Functional enrichment analysis comparison between H. sapiens and A. thaliana</p>
for &quot;AID&quot;,&quot;DCE&quot;,&quot;RCD&quot;,&quot;SPS&quot; cancer-hallmarks</a></li>
<li><a href='#comparison_ex_compress_CH'><p>Functional enrichment analysis comparison between H. sapiens and A. thaliana</p>
for &quot;DCE&quot;, and &quot;RCD&quot; cancer-hallmarks. This dataset contains 10 GO terms per category
to allow a fast run of the function graph_two_GOspecies.</a></li>
<li><a href='#evaluateCAT_species'><p>Comprehensive comparison between species using categories and Pearson's</p>
Chi-squared Tests</a></li>
<li><a href='#evaluateGO_species'><p>Comprehensive comparison between species using GO terms and Pearson's</p>
Chi-squared Tests</a></li>
<li><a href='#graph_two_GOspecies'><p>Undirected network representation  for the results of functional</p>
enrichment analysis to compare two species and a series of categories</a></li>
<li><a href='#graphGOspecies'><p>Undirected network representation  for the results of functional</p>
enrichment analysis for one species</a></li>
<li><a href='#H_sapiens'><p>H. sapiens functional enrichment analysis of 5494 genes related to cancer-hallmarks</p></a></li>
<li><a href='#H_sapiens_compress'><p>H. sapiens functional enrichment analysis results for</p>
&quot;AID&quot;,&quot;DCE&quot;,&quot;RCD&quot;,&quot;SPS&quot; cancer-hallmarks</a></li>
<li><a href='#mostFrequentGOs'><p>Most frequent GO terms among groups for a data.frame</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Comprehensive GO Terms Comparison Between Species</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Supports the assessment of functional enrichment analyses obtained for several lists of genes and provides a workflow to analyze them between two species via weighted graphs. Methods are described in Sosa et al. (2023) &lt;<a href="https://doi.org/10.1016%2Fj.ygeno.2022.110528">doi:10.1016/j.ygeno.2022.110528</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ccsosa/GOCompare">https://github.com/ccsosa/GOCompare</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ccsosa/GOCompare/issues">https://github.com/ccsosa/GOCompare/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>base (&ge; 3.5), utils (&ge; 3.5), methods (&ge; 3.5), stats,
grDevices, ape, vegan, ggplot2, ggrepel, igraph, parallel,
stringr, mathjaxr,</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>mathjaxr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-12-02 20:19:51 UTC; cami_</td>
</tr>
<tr>
<td>Author:</td>
<td>Chrystian Camilo Sosa
    <a href="https://orcid.org/0000-0002-3734-3248"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre,
    cph],
  Diana Carolina Clavijo-Buriticá [aut],
  Mauricio Alberto Quimbaya [aut],
  Maria Victoria Diaz [ctb],
  Camila Riccio Rengifo [ctb],
  Nicolas López-Rozo [ctb],
  Victor Hugo García Merchán [aut, ctb],
  Arlen James Mosquera [ctb],
  Andrés Álvarez [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chrystian Camilo Sosa &lt;ccsosaa@javerianacali.edu.co&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-09 23:40:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='GOCompare-package'>GOCompare: An R package to compare GO terms of gene lists (categories)  and their orthologs</h2><span id='topic+GOCompare-package'></span><span id='topic+GOCompare'></span>

<h3>Description</h3>

<p>GOCompare is a an R package used to compare a GO terms list between
two species
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> GOCompare</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.2.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2022-12-02</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<hr>
<h2 id='A_thaliana'>A thaliana functional enrichment analysis  of 2224 ortholog genes related
to cancer-hallmarks</h2><span id='topic+A_thaliana'></span>

<h3>Description</h3>

<p>This dataset is the original dataset obtained for Clavijo-Buriticá (In preparation)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>A_thaliana
</code></pre>


<h3>Format</h3>

<p>A data frame with 4063 rows and 6 variables:
</p>

<dl>
<dt>Enrichment_FDR</dt><dd><p>Numeric: False discovery rate values for the GO term</p>
</dd>
<dt>Genes_in_list</dt><dd><p>numeric: Number of genes in the list of genes for a given GO term</p>
</dd>
<dt>Total_genes</dt><dd><p>numeric: Number of genes in the genome of a species for a given GO term</p>
</dd>
<dt>Functional_Category</dt><dd><p>character: GO term name or GO term id</p>
</dd>
<dt>Genes</dt><dd><p>character: Genes found fot a given GO term</p>
</dd>
<dt>feature</dt><dd><p>character: A column representing the belonging of a group of comparison</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://data.mendeley.com/datasets/myyy2wxd59/1">https://data.mendeley.com/datasets/myyy2wxd59/1</a>
</p>


<h3>References</h3>

<p>Clavijo-Buriticá, Sosa, C.C., Mosquera, A.J. Álvarez, A., Medina, J. Quimbaya,
M.A. A systematic comparison of the molecular machinery associated with Cancer-Hallmarks
between plants and humans reveals Arabidopsis thaliana as a useful model to understand
specific carcinogenic events (to be submitted, Target journal: Plos Biology)
</p>

<hr>
<h2 id='A_thaliana_compress'>A thaliana functional enrichment analysis results for
&quot;AID&quot;,&quot;DCE&quot;,&quot;RCD&quot;,&quot;SPS&quot; cancer-hallmarks</h2><span id='topic+A_thaliana_compress'></span>

<h3>Description</h3>

<p>This dataset is a subset of the original dataset obtained
for Clavijo-Buriticá (In preparation)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>A_thaliana_compress
</code></pre>


<h3>Format</h3>

<p>A data frame with 120 rows and 6 variables  (30 GO terms per cancer hallmark):
</p>

<dl>
<dt>Enrichment_FDR</dt><dd><p>Numeric: False discovery rate values for the GO term</p>
</dd>
<dt>Genes_in_list</dt><dd><p>numeric: Number of genes in the list of genes for a given GO term</p>
</dd>
<dt>Total_genes</dt><dd><p>numeric: Number of genes in the genome of a species for a given GO term</p>
</dd>
<dt>Functional_Category</dt><dd><p>character: GO term name or GO term id</p>
</dd>
<dt>Genes</dt><dd><p>character: Genes found fot a given GO term</p>
</dd>
<dt>feature</dt><dd><p>character: A column representing the belonging of a group of comparison</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://data.mendeley.com/datasets/myyy2wxd59/1">https://data.mendeley.com/datasets/myyy2wxd59/1</a>
</p>


<h3>References</h3>

<p>Clavijo-Buriticá, Sosa, C.C., Mosquera, A.J. Álvarez, A., Medina, J. Quimbaya,
M.A. A systematic comparison of the molecular machinery associated with Cancer-Hallmarks
between plants and humans reveals Arabidopsis thaliana as a useful model to understand
specific carcinogenic events (to be submitted, Target journal: Plos Biology)
</p>

<hr>
<h2 id='compareGOspecies'>Visual representation for the results of functional
enrichment analysis to compare two species and a series of categories</h2><span id='topic+compareGOspecies'></span>

<h3>Description</h3>

<p>compareGOspecies function provides a simple workflow to compare results
of functional enrichment analysis for two species.
</p>
<p>To use this function you will need two matrices with a column which, represents the features to be compared (e.g.feature).
This function will extract the unique GO terms for two matrices and it will generate a presence-absence matrix
where rows will represent a combination of categories and species (e.g H.sapiens AID) and columns will represent
the GO terms analyzed. Further, this function  will calculate Jaccard distances and it will provide as outputs a list with four slots:
1.) A principal coordinates analysis (PCoA)
2.) The Jaccard distance matrix
3.) A list of shared GO terms between species
4.) Finally, a list of the unique GO terms and the belonging to the respective species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareGOspecies(
  df1,
  df2,
  GOterm_field,
  species1,
  species2,
  skipPCoA = FALSE,
  paired_lists = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compareGOspecies_+3A_df1">df1</code></td>
<td>
<p>A data frame with the results of a functional enrichment analysis for the species 1
with an extra column &quot;feature&quot; with the features to be compared</p>
</td></tr>
<tr><td><code id="compareGOspecies_+3A_df2">df2</code></td>
<td>
<p>A data frame with the results of a functional enrichment analysis for the species 2
with an extra column &quot;feature&quot; with the features to be compared</p>
</td></tr>
<tr><td><code id="compareGOspecies_+3A_goterm_field">GOterm_field</code></td>
<td>
<p>This is a string with the column name of the GO terms (e.g; &quot;Functional_Category&quot;)</p>
</td></tr>
<tr><td><code id="compareGOspecies_+3A_species1">species1</code></td>
<td>
<p>This is a string with the species name for species 1 (e.g; &quot;H. sapiens&quot;)</p>
</td></tr>
<tr><td><code id="compareGOspecies_+3A_species2">species2</code></td>
<td>
<p>This is a string with the species name for species 2 (e.g; &quot;A. thaliana&quot;)</p>
</td></tr>
<tr><td><code id="compareGOspecies_+3A_skippcoa">skipPCoA</code></td>
<td>
<p>This is a boolean to indicate if the PCoA graphics can be skipped</p>
</td></tr>
<tr><td><code id="compareGOspecies_+3A_paired_lists">paired_lists</code></td>
<td>
<p>This is a boolean to indicate if both species have same comparable categories (gene lists).
If the paired_lists is FALSE the counts will be done only for species and categories will be kept in
the outcomes. Please use carefully when paired_lists = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will return a list with four slots: graphics, distance shared_GO_list, and unique_GO_list
</p>


<h3>Note</h3>

<p>Do not use &quot;-&quot; in the feature column. This will lead to wrong results!
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Loading example datasets
data(H_sapiens_compress)
data(A_thaliana_compress)
#Defining the column with the GO terms to be compared
GOterm_field &lt;- "Functional_Category"
#Defining the species names
species1 &lt;- "H. sapiens"
species2 &lt;- "A. thaliana"

#Running function
x &lt;- compareGOspecies(df1=H_sapiens_compress,
                      df2=A_thaliana_compress,
                      GOterm_field=GOterm_field,
                      species1=species1,
                      species2=species2,
                      skipPCoA=FALSE,
                      paired_lists=TRUE)

## Not run: 
#Displaying PCoA results
 x$graphics
# Checking shared GO terms between species
 print(tapply(x$shared_GO_list$feature,x$shared_GO_list$feature,length))
 
## End(Not run)

</code></pre>

<hr>
<h2 id='comparison_ex_compress'>Functional enrichment analysis comparison between H. sapiens and A. thaliana
for &quot;AID&quot;,&quot;DCE&quot;,&quot;RCD&quot;,&quot;SPS&quot; cancer-hallmarks</h2><span id='topic+comparison_ex_compress'></span>

<h3>Description</h3>

<p>This dataset is the results of running the compareGOspecies species and
it is composed of four slots:
</p>

<dl>
<dt>graphics</dt><dd><p>PCoA graphics</p>
</dd>
<dt>distance</dt><dd><p>numeric: Jaccard distance matrix </p>
</dd>
<dt>shared_GO_list</dt><dd><p>data.frame with shared GO terms between species</p>
</dd>
<dt>unique_GO_list</dt><dd><p>data.frame with unique GO terms and their belonging two each species</p>
</dd>
</dl>



<h3>Usage</h3>

<pre><code class='language-R'>comparison_ex_compress
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 4.
</p>


<h3>Source</h3>

<p><a href="https://data.mendeley.com/datasets/myyy2wxd59/1">https://data.mendeley.com/datasets/myyy2wxd59/1</a>
</p>


<h3>References</h3>

<p>Clavijo-Buriticá, Sosa, C.C., Mosquera, A.J. Álvarez, A., Medina, J. Quimbaya,
M.A. A systematic comparison of the molecular machinery associated with Cancer-Hallmarks
between plants and humans reveals Arabidopsis thaliana as a useful model to understand
specific carcinogenic events (to be submitted, Target journal: Plos Biology)
</p>

<hr>
<h2 id='comparison_ex_compress_CH'>Functional enrichment analysis comparison between H. sapiens and A. thaliana
for &quot;DCE&quot;, and &quot;RCD&quot; cancer-hallmarks. This dataset contains 10 GO terms per category
to allow a fast run of the function graph_two_GOspecies.</h2><span id='topic+comparison_ex_compress_CH'></span>

<h3>Description</h3>

<p>This dataset is the results of running the compareGOspecies species and
it is composed of three slots:
</p>

<dl>
<dt>distance</dt><dd><p>numeric: Jaccard distance matrix </p>
</dd>
<dt>shared_GO_list</dt><dd><p>data.frame with shared GO terms between species</p>
</dd>
<dt>unique_GO_list</dt><dd><p>data.frame with unique GO terms and their belonging two each species</p>
</dd>
</dl>



<h3>Usage</h3>

<pre><code class='language-R'>comparison_ex_compress_CH
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 3.
</p>


<h3>Source</h3>

<p><a href="https://data.mendeley.com/datasets/myyy2wxd59/1">https://data.mendeley.com/datasets/myyy2wxd59/1</a>
</p>


<h3>References</h3>

<p>Clavijo-Buriticá, Sosa, C.C., Mosquera, A.J. Álvarez, A., Medina, J. Quimbaya,
M.A. A systematic comparison of the molecular machinery associated with Cancer-Hallmarks
between plants and humans reveals Arabidopsis thaliana as a useful model to understand
specific carcinogenic events (to be submitted, Target journal: Plos Biology)
</p>

<hr>
<h2 id='evaluateCAT_species'>Comprehensive comparison between species using categories and Pearson's
Chi-squared Tests</h2><span id='topic+evaluateCAT_species'></span>

<h3>Description</h3>

<p>evaluateGO_species provides a simple function  to compare results
of functional enrichment analysis for two species through the use of proportion tests or Pearson's
Chi-squared Tests and a False discovery rate correction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evaluateCAT_species(df1, df2, species1, species2, GOterm_field, test = "prop")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="evaluateCAT_species_+3A_df1">df1</code></td>
<td>
<p>A data frame with the results of a functional enrichment analysis
for the species 1 with an extra column &quot;feature&quot; with the features to be compared</p>
</td></tr>
<tr><td><code id="evaluateCAT_species_+3A_df2">df2</code></td>
<td>
<p>A data frame with the results of a functional enrichment analysis
for the species 2 with an extra column &quot;feature&quot; with
the features to be compared</p>
</td></tr>
<tr><td><code id="evaluateCAT_species_+3A_species1">species1</code></td>
<td>
<p>This is a string with the species name for the species 1 (e.g; &quot;H. sapiens&quot;)</p>
</td></tr>
<tr><td><code id="evaluateCAT_species_+3A_species2">species2</code></td>
<td>
<p>This is a string with the species name for the species 2 (e.g; &quot;A. thaliana&quot;)</p>
</td></tr>
<tr><td><code id="evaluateCAT_species_+3A_goterm_field">GOterm_field</code></td>
<td>
<p>This is a string with the column name of the GO terms
(e.g; &quot;Functional_Category&quot;)</p>
</td></tr>
<tr><td><code id="evaluateCAT_species_+3A_test">test</code></td>
<td>
<p>This is a string with the hypothesis test to be performed. Two options are provided,
&quot;prop&quot; and &quot;chi-squared&quot; (default value=&quot;prop&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will return a data.frame with the following fields:
</p>

<table>
<tr>
 <td style="text-align: left;">
CAT </td><td style="text-align: center;"> Category  </td>
</tr>
<tr>
 <td style="text-align: left;">
pvalue </td><td style="text-align: center;"> p-value obtained through the use of Pearson's Chi-squared Test</td>
</tr>
<tr>
 <td style="text-align: left;">
FDR </td><td style="text-align: center;"> Multiple comparison correction for the p-value column </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>
#Loading example datasets
data(H_sapiens)
data(A_thaliana)
#Defining the column with the GO terms to be compared
GOterm_field &lt;- "Functional_Category"
#Defining the species names
species1 &lt;- "H. sapiens"
species2 &lt;- "A. thaliana"
#Running function
x &lt;- evaluateCAT_species(df1= H_sapiens,
                       df2=A_thaliana,
                       species1=species1,
                       species2=species2,
                       GOterm_field=GOterm_field,
                         test="prop")
print(x)
</code></pre>

<hr>
<h2 id='evaluateGO_species'>Comprehensive comparison between species using GO terms and Pearson's
Chi-squared Tests</h2><span id='topic+evaluateGO_species'></span>

<h3>Description</h3>

<p>evaluateGO_species provides a simple function  to compare results
of functional enrichment analysis for two species through the use of proportion tests or Pearson's
Chi-squared Tests and a False discovery rate correction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evaluateGO_species(df1, df2, species1, species2, GOterm_field, test = "prop")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="evaluateGO_species_+3A_df1">df1</code></td>
<td>
<p>A data frame with the results of a functional enrichment analysis
for the species 1 with an extra column &quot;feature&quot; with the features to be compared</p>
</td></tr>
<tr><td><code id="evaluateGO_species_+3A_df2">df2</code></td>
<td>
<p>A data frame with the results of a functional enrichment analysis
for the species 2 with an extra column &quot;feature&quot; with
the features to be compared</p>
</td></tr>
<tr><td><code id="evaluateGO_species_+3A_species1">species1</code></td>
<td>
<p>This is a string with the species name for the species 1 (e.g; &quot;H. sapiens&quot;)</p>
</td></tr>
<tr><td><code id="evaluateGO_species_+3A_species2">species2</code></td>
<td>
<p>This is a string with the species name for the species 2 (e.g; &quot;A. thaliana&quot;)</p>
</td></tr>
<tr><td><code id="evaluateGO_species_+3A_goterm_field">GOterm_field</code></td>
<td>
<p>This is a string with the column name of the GO terms
(e.g; &quot;Functional_Category&quot;)</p>
</td></tr>
<tr><td><code id="evaluateGO_species_+3A_test">test</code></td>
<td>
<p>This is a string with the hypothesis test to be performed. Two options are provided,
&quot;prop&quot; and &quot;chi-squared&quot; (default value=&quot;prop&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will return a data.frame with the following fields:
</p>

<table>
<tr>
 <td style="text-align: left;">
GO </td><td style="text-align: center;"> GO term analyzed </td>
</tr>
<tr>
 <td style="text-align: left;">
pvalue </td><td style="text-align: center;"> p-value obtained through the use of Pearson's Chi-squared Test</td>
</tr>
<tr>
 <td style="text-align: left;">
FDR </td><td style="text-align: center;"> Multiple comparison correction for the p-value column </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>
#Loading example datasets
data(H_sapiens)
data(A_thaliana)
#Defining the column with the GO terms to be compared
GOterm_field &lt;- "Functional_Category"
#Defining the species names
species1 &lt;- "H. sapiens"
species2 &lt;- "A. thaliana"
#Running function
x &lt;- evaluateGO_species(df1= H_sapiens,
                        df2=A_thaliana,
                        species1=species1,
                        species2=species2,
                        GOterm_field=GOterm_field,
                        test="prop")
print(x)
</code></pre>

<hr>
<h2 id='graph_two_GOspecies'>Undirected network representation  for the results of functional
enrichment analysis to compare two species and a series of categories</h2><span id='topic+graph_two_GOspecies'></span>

<h3>Description</h3>

<p>graph_two_GOspecies is a function to create undirected graphs
<script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>
<p>The graph_two_GOspecies is an analog of the graphGOspecies function, and it has the same options (&quot; Categories &quot; and &quot; GO &quot;).
Nevertheless, the way in which the edge and node weights are calculated is slightly different. Since two species are compared,
three possible graphs are available \({G}_1,\, {G}_2\), and \({G}_3 \). \({G}_1\), and \({G}_2 \) represent each of the species analyzed
and \({G}_3\) is a subgraph of \({G}_1,\, {G}_2\), which contains the GO terms or Categories co-ocurring between both species.
</p>
<p>Categories option:
(Weight): The nodes \((V)\) represent groups of gene lists (categories), and the edges \((E)\) represent GO terms co-occurring between pairs of categories
and the weight of the nodes provides a measure of how a GO term is conserved between two species and a series of categories but it is biased
to categories.
</p>
\[\widehat{K}_w(u)=\sum_{v \epsilon V_1}^{}w(u,v) + \sum_{v \epsilon V_2}^{}w(u,v)\]<p> (5)
</p>
<p>(shared weight): The nodes \((V)\) represent groups of gene lists (categories), and the edges \((E)\) represent GO terms co-occurring between pairs of categories that are only
shared between species. This node weight  \({K}_s\) is computed from a shared weight of edges \({s}\), where \({N}1\) and \({N}2\) are the set of GO terms associated
with the edge \(e = (u,v) \) for species 1 and 2, respectively. Therefore the node shared weight \({K}_s(u)\)  is the sum of \({s}\).
</p>
\[s(e) = \frac{\mid {N1} \ n \ {N2} \mid}{\mid {N1} \bigcup {N2} \mid}\]<p> (6)
</p>
\[{K}_s(u)=\sum_{v \epsilon (V_1  \bigcup V_2) }^{}{s(u,v)}\]<p> (7)
</p>
<p>(combined weight): This node weight  \({K}_c(u)\) is a combination of the weight and the shared weight. The idea of this combined weight is
to find categories with more frequent GO terms co-ocurring in order to observe functional similarities between two species with a balance of GO terms co-occurring among
gene lists (categories) and the two species. This node weight varies from -1 (categories with GO terms found only in one species and few categories) to 1
(categories with GO terms shared widely between species and among other categories). the combined node weight \({K}_c\) is defined as the sum of the min-max normalized weights
\(\widehat{K}_w\) and \({K}_s\) minus 1.
</p>
\[minmax(y)=\frac{y-min(y)}{max(y)-min(y)}\]<p> (8)
\[{K}_c(u)= minmax(\widehat{K}_w(u)) + minmax({K}_s(u)) - 1 \] (9)
</p>
<p>GO option:
Given there are three possible graphs are available \({G}_1,\, {G}_2\), and \({G}_3\).  \({G}_1\), and \({G}_2\) represent each of the species analyzed
and \({G}_3\) is a subgraph of \({G}_1,\, {G}_2\), which contains the GO terms or Categories co-ocurring between both species. For this case, Nodes are GO terms and edges are
categories where a GO terms is co-ocurring. This weight is similar to the GO weight calculated for graphGOspecies function. it is calculated as the equation 5.
</p>
\[\widehat{K}_w(u)=\sum_{v \epsilon V_1}^{}w(u,v) + \sum_{v \epsilon V_2}^{}w(u,v)\]<p> (5)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graph_two_GOspecies(
  x,
  species1,
  species2,
  GOterm_field,
  saveGraph = FALSE,
  option = "Categories",
  numCores = 2,
  outdir = NULL,
  filename = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="graph_two_GOspecies_+3A_x">x</code></td>
<td>
<p>is a list obtained as output of the comparegOspecies function</p>
</td></tr>
<tr><td><code id="graph_two_GOspecies_+3A_species1">species1</code></td>
<td>
<p>This is a string with the species name for species 1 (e.g; &quot;H. sapiens&quot;)</p>
</td></tr>
<tr><td><code id="graph_two_GOspecies_+3A_species2">species2</code></td>
<td>
<p>This is a string with the species name for species 2 (e.g; &quot;A. thaliana&quot;)</p>
</td></tr>
<tr><td><code id="graph_two_GOspecies_+3A_goterm_field">GOterm_field</code></td>
<td>
<p>This is a string with the column name of the GO terms (e.g; &quot;Functional_Category&quot;)</p>
</td></tr>
<tr><td><code id="graph_two_GOspecies_+3A_savegraph">saveGraph</code></td>
<td>
<p>logical, if <code>TRUE</code> the function will allow save the graph in graphml format</p>
</td></tr>
<tr><td><code id="graph_two_GOspecies_+3A_option">option</code></td>
<td>
<p>(values: &quot;Categories or &quot;GO&quot;). This option allows create either a graph
where nodes are GO terms and edges are features and GO as well as species belonging are edges attributes or
a graph where nodes are GO terms and edges are species belonging  (default value=&quot;Categories&quot;)</p>
</td></tr>
<tr><td><code id="graph_two_GOspecies_+3A_numcores">numCores</code></td>
<td>
<p>numeric, Number of cores to use for the process (default value numCores=2). For the example below, only one core will be used</p>
</td></tr>
<tr><td><code id="graph_two_GOspecies_+3A_outdir">outdir</code></td>
<td>
<p>This parameter will allow save the graph file in a folder described here (e.g: &quot;D:&quot;).This parameter only
works when saveGraph=TRUE</p>
</td></tr>
<tr><td><code id="graph_two_GOspecies_+3A_filename">filename</code></td>
<td>
<p>The name of the graph filename to be saved in the outdir detailed by the user.This parameter only
works when saveGraph=TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will return a list with two slots: edges and nodes.
(Categories):
Edges list columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
  Column </td><td style="text-align: right;"> Description </td>
</tr>
<tr>
 <td style="text-align: right;">
  SOURCE and TARGET </td><td style="text-align: right;"> The source and target categories (Nodes in the edge) </td>
</tr>
<tr>
 <td style="text-align: right;">
  GO_N </td><td style="text-align: right;"> The number of GO terms between the categories </td>
</tr>
<tr>
 <td style="text-align: right;">
  WEIGHT </td><td style="text-align: right;"> Edge weight </td>
</tr>
<tr>
 <td style="text-align: right;">
  GO </td><td style="text-align: right;"> GO terms available for both nodes </td>
</tr>
<tr>
 <td style="text-align: right;">
  SP1 </td><td style="text-align: right;"> Number of GO terms for the species 1 </td>
</tr>
<tr>
 <td style="text-align: right;">
  SP2 </td><td style="text-align: right;"> Number of GO terms for the species 2 </td>
</tr>
<tr>
 <td style="text-align: right;">
  SHARED </td><td style="text-align: right;"> Number of GO terms shared or co-ocurring between the categories </td>
</tr>
<tr>
 <td style="text-align: right;">
  SHARED_WEIGHT </td><td style="text-align: right;"> Shared weight for the edge </td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>

<p>Node list columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
  Column </td><td style="text-align: right;"> Description </td>
</tr>
<tr>
 <td style="text-align: right;">
  CAT</td><td style="text-align: right;"> Category name </td>
</tr>
<tr>
 <td style="text-align: right;">
  CAT_WEIGHT  </td><td style="text-align: right;"> Node weight </td>
</tr>
<tr>
 <td style="text-align: right;">
  SHARED_WEIGHT  </td><td style="text-align: right;"> Shared weight for the node </td>
</tr>
<tr>
 <td style="text-align: right;">
  COMBINED_WEIGHT </td><td style="text-align: right;"> Combined weight for the node </td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>

<p>(GO):
</p>
<p>Edges list columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
  Column </td><td style="text-align: right;"> Description </td>
</tr>
<tr>
 <td style="text-align: right;">
  SOURCE and TARGET </td><td style="text-align: right;"> The source and target GO terms (Nodes in the edge) </td>
</tr>
<tr>
 <td style="text-align: right;">
  FEATURE </td><td style="text-align: right;"> The number of Categories where both GO Terms were found </td>
</tr>
<tr>
 <td style="text-align: right;">
  SP </td><td style="text-align: right;"> Species where the GO terms was found (Species 1, Species 2 or Shared) </td>
</tr>
<tr>
 <td style="text-align: right;">
  WEIGHT </td><td style="text-align: right;"> Edge weight </td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>

<p>Node list columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
  Column </td><td style="text-align: right;"> Description </td>
</tr>
<tr>
 <td style="text-align: right;">
  GO </td><td style="text-align: right;"> GO term node name </td>
</tr>
<tr>
 <td style="text-align: right;">
  GO_WEIGHT </td><td style="text-align: right;"> Node weight </td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>
GOterm_field &lt;- "Functional_Category"
data(comparison_ex_compress_CH)
#Defining the species names
species1 &lt;- "H. sapiens"
species2 &lt;- "A. thaliana"
x_graph &lt;- graph_two_GOspecies(x=comparison_ex_compress_CH,
          species1=species1,
          species2=species2,
          GOterm_field=GOterm_field,
          numCores=1,
          saveGraph = FALSE,
          option= "Categories",
          outdir = NULL,
          filename= NULL)

</code></pre>

<hr>
<h2 id='graphGOspecies'>Undirected network representation  for the results of functional
enrichment analysis for one species</h2><span id='topic+graphGOspecies'></span>

<h3>Description</h3>

<p>graphGOspecies is a function to create undirected graphs using two options:
<script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>
<p>Categories option:
</p>
<p>The nodes \((V)\) represent groups of gene lists (categories), and the edges \((E)\) represent GO terms co-occurring between pairs of categories. More specifically,
Two categories: \(u,v \epsilon V \) are connected by an edge \(e=(u,v)\).the edge weights \(w(e)\) are defined as the ratio of the number of GO terms co-occurring
between two categories. Edge weights w(e) are defined as the ratio of the number of GO terms (e.g. biological processes) co-occurring between two categories
\(BP_{u} \ n  BP_{v}\) compared to the total number of GO terms available.
A node weight \(K_{w}(u)\) is defined as the sum of the edge weights where the node u is a participant. Thus, the node weight represents how frequently
GO terms are reported and expressed in a biological phenomenon.
</p>
\[w(e) = \frac{\mid BP_{u} n {BP_{v}}\mid}{\mid BP\mid}\]<p> (1)
</p>
\[K_{w} = \sum_{{v} \epsilon {V}}{w(u,v)}\]<p> (2)
</p>
<p>GO option:
</p>
<p>The nodes \({V}\) represent GO terms and the edges \({E}'\) represent categories where a pair of GO terms co-occur. More specifically,
two GO terms are connected by an edge \({e}'=({u},{v}')\). the edge weight \({w}'({e}')\) corresponds to the number of categories co-occurring
the GO terms \({u}\) and \({v}'\),compared with the total number of GO terms (Equation 3). A node weight \({K}'_w({u}')\) is defined,in this case the weight
represents the importance of a GO term (more frequent co-occurring).(Please be patient, it requires a long time to finish).
</p>
\[{w}'({e}')=\frac{\mid{Cu}'\cap  {Cv}'\mid}{\mid BP \mid}\]<p> (3)
</p>
\[{K}'_w({u}')=\sum_{{v}'\epsilon {V}'}{{w}'({u}',{v}')}\]<p> (4)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graphGOspecies(
  df,
  GOterm_field,
  option = "Categories",
  numCores = 2,
  saveGraph = FALSE,
  outdir = NULL,
  filename = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="graphGOspecies_+3A_df">df</code></td>
<td>
<p>A data frame with the results of a functional enrichment analysis for
a species with an extra column &quot;feature&quot; with the features to be compared</p>
</td></tr>
<tr><td><code id="graphGOspecies_+3A_goterm_field">GOterm_field</code></td>
<td>
<p>This is a string with the column name of the GO terms (e.g: &quot;Functional.Category&quot;)</p>
</td></tr>
<tr><td><code id="graphGOspecies_+3A_option">option</code></td>
<td>
<p>(values: &quot;GO&quot; or &quot;Categories&quot;). This option allows create either a graph where nodes are GO terms and edges are features or alternatively
a graph where nodes are features and edges are GO terms (default value=&quot;Categories&quot;)</p>
</td></tr>
<tr><td><code id="graphGOspecies_+3A_numcores">numCores</code></td>
<td>
<p>numeric, Number of cores to use for the process (default value numCores=2). For the example below, only one core will be used</p>
</td></tr>
<tr><td><code id="graphGOspecies_+3A_savegraph">saveGraph</code></td>
<td>
<p>logical, if <code>TRUE</code> the function will allow save the graph in graphml format</p>
</td></tr>
<tr><td><code id="graphGOspecies_+3A_outdir">outdir</code></td>
<td>
<p>This parameter will allow save the graph file in a folder described here (e.g: &quot;D:&quot;).This parameter only
works when saveGraph=TRUE</p>
</td></tr>
<tr><td><code id="graphGOspecies_+3A_filename">filename</code></td>
<td>
<p>The name of the graph filename to be saved in the outdir detailed by the user.This parameter only
works when saveGraph=TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will return a list with two slots: edges and nodes.
</p>
<p>(Categories):
Edges list columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
  Column </td><td style="text-align: right;"> Description </td>
</tr>
<tr>
 <td style="text-align: right;">
  SOURCE and TARGET </td><td style="text-align: right;"> The source and target categories (Nodes in the edge) </td>
</tr>
<tr>
 <td style="text-align: right;">
  FEATURES_N </td><td style="text-align: right;"> The number of GO terms between the categories </td>
</tr>
<tr>
 <td style="text-align: right;">
  WEIGHT </td><td style="text-align: right;"> Edge weight </td>
</tr>
<tr>
 <td style="text-align: right;">
  FEATURES </td><td style="text-align: right;"> GO terms available for both nodes </td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>

<p>Node list columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
  Column </td><td style="text-align: right;"> Description </td>
</tr>
<tr>
 <td style="text-align: right;">
  feature</td><td style="text-align: right;"> Category name </td>
</tr>
<tr>
 <td style="text-align: right;">
  GO_count  </td><td style="text-align: right;"> GO terms counts for the node </td>
</tr>
<tr>
 <td style="text-align: right;">
  WEIGHT  </td><td style="text-align: right;"> Node weight </td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>

<p>(GO):
</p>
<p>Edges list columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
  Column </td><td style="text-align: right;"> Description </td>
</tr>
<tr>
 <td style="text-align: right;">
  SOURCE and TARGET </td><td style="text-align: right;"> The source and target GO terms (Nodes in the edge) </td>
</tr>
<tr>
 <td style="text-align: right;">
  FEATURE </td><td style="text-align: right;"> The number of Categories where both GO Terms were found </td>
</tr>
<tr>
 <td style="text-align: right;">
  WEIGHT </td><td style="text-align: right;"> Edge weight </td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>

<p>Node list columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
  Column </td><td style="text-align: right;"> Description </td>
</tr>
<tr>
 <td style="text-align: right;">
  GO </td><td style="text-align: right;"> GO term node name </td>
</tr>
<tr>
 <td style="text-align: right;">
  GO_WEIGHT </td><td style="text-align: right;"> Node weight </td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>
#Loading example datasets
data(H_sapiens_compress)

GOterm_field &lt;- "Functional_Category"

#Running function
x &lt;- graphGOspecies(df=H_sapiens_compress,
                     GOterm_field=GOterm_field,
                     option = "Categories",
                     numCores=1,
                     saveGraph=FALSE,
                     outdir = NULL,
                     filename=NULL)

</code></pre>

<hr>
<h2 id='H_sapiens'>H. sapiens functional enrichment analysis of 5494 genes related to cancer-hallmarks</h2><span id='topic+H_sapiens'></span>

<h3>Description</h3>

<p>This dataset is a subset of the original dataset obtained for Clavijo-Buriticá (In preparation)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>H_sapiens
</code></pre>


<h3>Format</h3>

<p>A data frame with 5000 rows and 6 variables:
</p>

<dl>
<dt>Enrichment_FDR</dt><dd><p>Numeric: False discovery rate values for the GO term</p>
</dd>
<dt>Genes_in_list</dt><dd><p>numeric: Number of genes in the list of genes for a given GO term</p>
</dd>
<dt>Total_genes</dt><dd><p>numeric: Number of genes in the genome of a species for a given GO term</p>
</dd>
<dt>Functional_Category</dt><dd><p>character: GO term name or GO term id</p>
</dd>
<dt>Genes</dt><dd><p>character: Genes found fot a given GO term</p>
</dd>
<dt>feature</dt><dd><p>character: A column representing the belonging of a group of comparison</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://data.mendeley.com/datasets/myyy2wxd59/1">https://data.mendeley.com/datasets/myyy2wxd59/1</a>
</p>


<h3>References</h3>

<p>Clavijo-Buriticá, Sosa, C.C., Mosquera, A.J. Álvarez, A., Medina, J. Quimbaya,
M.A. A systematic comparison of the molecular machinery associated with Cancer-Hallmarks
between plants and humans reveals Arabidopsis thaliana as a useful model to understand
specific carcinogenic events (to be submitted, Target journal: Plos Biology)
</p>

<hr>
<h2 id='H_sapiens_compress'>H. sapiens functional enrichment analysis results for
&quot;AID&quot;,&quot;DCE&quot;,&quot;RCD&quot;,&quot;SPS&quot; cancer-hallmarks</h2><span id='topic+H_sapiens_compress'></span>

<h3>Description</h3>

<p>This dataset is a subset of the original dataset obtained
for Clavijo-Buriticá (In preparation)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>H_sapiens_compress
</code></pre>


<h3>Format</h3>

<p>A data frame with 120 rows and 6 variables (30 GO terms per cancer hallmark):
</p>

<dl>
<dt>Enrichment_FDR</dt><dd><p>Numeric: False discovery rate values for the GO term</p>
</dd>
<dt>Genes_in_list</dt><dd><p>numeric: Number of genes in the list of genes for a given GO term</p>
</dd>
<dt>Total_genes</dt><dd><p>numeric: Number of genes in the genome of a species for a given GO term</p>
</dd>
<dt>Functional_Category</dt><dd><p>character: GO term name or GO term id</p>
</dd>
<dt>Genes</dt><dd><p>character: Genes found fot a given GO term</p>
</dd>
<dt>feature</dt><dd><p>character: A column representing the belonging of a group of comparison</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://data.mendeley.com/datasets/myyy2wxd59/1">https://data.mendeley.com/datasets/myyy2wxd59/1</a>
</p>


<h3>References</h3>

<p>Clavijo-Buriticá, Sosa, C.C., Mosquera, A.J. Álvarez, A., Medina, J. Quimbaya,
M.A. A systematic comparison of the molecular machinery associated with Cancer-Hallmarks
between plants and humans reveals Arabidopsis thaliana as a useful model to understand
specific carcinogenic events (to be submitted, Target journal: Plos Biology)
</p>

<hr>
<h2 id='mostFrequentGOs'>Most frequent GO terms among groups for a data.frame</h2><span id='topic+mostFrequentGOs'></span>

<h3>Description</h3>

<p>Provides an easy way to get the frequency of GO terms
such as biological processes for a data frame and a series of features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mostFrequentGOs(df, GOterm_field)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mostFrequentGOs_+3A_df">df</code></td>
<td>
<p>A data frame with the results of a functional enrichment analysis
for a species with an extra column &quot;feature&quot; with the features to be compared</p>
</td></tr>
<tr><td><code id="mostFrequentGOs_+3A_goterm_field">GOterm_field</code></td>
<td>
<p>This is a string with the column name of the GO terms (e.g; &quot;Functional.Category&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will return a table with the frequency of GO terms per feature
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Loading example datasets
data(H_sapiens)
#Defining the column with the GO terms to be compared
GOterm_field &lt;- "Functional_Category"
#Running function
x &lt;- mostFrequentGOs(df=H_sapiens, GOterm_field=GOterm_field)
#Displaying results
head(x)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
