<!DOCTYPE html><html><head><title>Help for package GOplot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GOplot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#chord_dat'><p>Creates a binary matrix.</p></a></li>
<li><a href='#circle_dat'><p>Creates a plotting object.</p></a></li>
<li><a href='#EC'><p>Transcriptomic information of endothelial cells.</p></a></li>
<li><a href='#GOBar'><p>Z-score coloured barplot.</p></a></li>
<li><a href='#GOBubble'><p>Bubble plot.</p></a></li>
<li><a href='#GOChord'><p>Displays the relationship between genes and terms.</p></a></li>
<li><a href='#GOCircle'><p>Circular visualization of the results of a functional analysis.</p></a></li>
<li><a href='#GOCluster'><p>Circular dendrogram.</p></a></li>
<li><a href='#GOHeat'><p>Displays heatmap of the relationship between genes and terms.</p></a></li>
<li><a href='#GOVenn'><p>Venn diagram of differentially expressed genes.</p></a></li>
<li><a href='#reduce_overlap'><p>Eliminates redundant terms.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Visualization of Functional Analysis Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-03-30</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/wencke/wencke.github.io">https://github.com/wencke/wencke.github.io</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/wencke/wencke.github.io/issues">https://github.com/wencke/wencke.github.io/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of multilayered visualizations for enhanced
    graphical representation of functional analysis data. It combines and integrates
    omics data derived from expression and functional annotation enrichment
    analyses. Its plotting functions have been developed with an hierarchical
    structure in mind: starting from a general overview to identify the most
    enriched categories (modified bar plot, bubble plot) to a more detailed one
    displaying different types of relevant information for the molecules in a given
    set of categories (circle plot, chord plot, cluster plot, Venn diagram, heatmap).</td>
</tr>
<tr>
<td>Depends:</td>
<td>ggplot2 (&ge; 2.0.0), ggdendro (&ge; 0.1-17), gridExtra (&ge;
2.0.0), RColorBrewer (&ge; 1.1.2), R (&ge; 3.2.3)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-03-30 08:24:21 UTC; BioinfoNerd</td>
</tr>
<tr>
<td>Author:</td>
<td>Wencke Walter [aut, cre],
  Fatima Sanchez-Cabo [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wencke Walter &lt;wencke.walter@arcor.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-03-30 20:35:02</td>
</tr>
</table>
<hr>
<h2 id='chord_dat'>Creates a binary matrix.</h2><span id='topic+chord_dat'></span>

<h3>Description</h3>

<p>The function creates a matrix which represents the binary 
relation (1= is related to, 0= is not related to) between selected genes 
(row) and processes (column). The resulting matrix can be visualized with 
the <code><a href="#topic+GOChord">GOChord</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chord_dat(data, genes, process)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chord_dat_+3A_data">data</code></td>
<td>
<p>A data frame with at least two coloumns: GO ID|term and genes. 
Each row contains exactly one GO ID|term and one gene. A column containing
logFC values is optional and might be used if <code>genes</code> is missing.</p>
</td></tr>
<tr><td><code id="chord_dat_+3A_genes">genes</code></td>
<td>
<p>A character vector of selected genes OR data frame with coloumns
for gene ID and logFC.</p>
</td></tr>
<tr><td><code id="chord_dat_+3A_process">process</code></td>
<td>
<p>A character vector of selected processes</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If more than one logFC value for each gene is at disposal, only one 
should be used to create the binary matrix. The other values have to be 
added manually later.
</p>


<h3>Value</h3>

<p>A binary matrix
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GOChord">GOChord</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load the included dataset
data(EC)

# Building the circ object
circ &lt;- circle_dat(EC$david, EC$genelist)

# Building the binary matrix
chord &lt;- chord_dat(circ, EC$genes, EC$process)


## End(Not run)
</code></pre>

<hr>
<h2 id='circle_dat'>Creates a plotting object.</h2><span id='topic+circle_dat'></span>

<h3>Description</h3>

<p>The function takes the results from a functional analysis (for 
example DAVID) and combines it with a list of selected genes and their 
logFC. The resulting data frame can be used as an input for various ploting
functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circle_dat(terms, genes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="circle_dat_+3A_terms">terms</code></td>
<td>
<p>A data frame with columns for 'category', 'ID', 'term', adjusted
p-value ('adj_pval') and 'genes'</p>
</td></tr>
<tr><td><code id="circle_dat_+3A_genes">genes</code></td>
<td>
<p>A data frame with columns for 'ID', 'logFC'</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Since most of the gene- annotation enrichment analysis are based on 
the gene ontology database the package was build with this structure in 
mind, but is not restricted to it. Gene ontology is structured as an 
acyclic graph and it provides terms covering different areas. These terms 
are grouped into three independent <code>categories</code>: BP (biological 
process), CC (cellular component) or MF (molecular function).
</p>
<p>The &quot;ID&quot; and &quot;term&quot; columns of the <code>terms</code> data frame refer to the ID 
and term description, whereas the ID is optional.
</p>
<p>The &quot;ID&quot; column of the <code>genes</code> data frame can contain any unique 
identifier. Nevertheless, the identifier has to be the same as in &quot;genes&quot; 
from <code>terms</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Load the included dataset
data(EC)

#Building the circ object
circ&lt;-circular_dat(EC$david, EC$genelist)

## End(Not run)
</code></pre>

<hr>
<h2 id='EC'>Transcriptomic information of endothelial cells.</h2><span id='topic+EC'></span>

<h3>Description</h3>

<p>The data set contains the transcriptomic information of endothelial cells
from two steady state tissues (brain and heart). More detailed information
can be found in the paper by Nolan et al. 2013. The data was normalized and a
statistical analysis was performed to determine differentially expressed
genes. DAVID functional annotation tool was used to perform a gene-
annotation enrichment analysis of the set of differentially expressed genes
(adjusted p-value &lt; 0.05).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(EC)
</code></pre>


<h3>Format</h3>

<p>A list containing 5 items</p>


<h3>Source</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE47067">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE47067</a>
</p>

<hr>
<h2 id='GOBar'>Z-score coloured barplot.</h2><span id='topic+GOBar'></span>

<h3>Description</h3>

<p>Z-score coloured barplot of terms ordered alternatively by 
z-score or the negative logarithm of the adjusted p-value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GOBar(data, display, order.by.zscore = T, title, zsc.col)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GOBar_+3A_data">data</code></td>
<td>
<p>A data frame containing at least the term ID and/or term, the 
adjusted p-value and the z-score. A possible input can be generated with 
the <code>circle_dat</code> function</p>
</td></tr>
<tr><td><code id="GOBar_+3A_display">display</code></td>
<td>
<p>A character vector indicating whether a single plot ('single')
or a facet plot with panels for each category should be drawn 
(default='single')</p>
</td></tr>
<tr><td><code id="GOBar_+3A_order.by.zscore">order.by.zscore</code></td>
<td>
<p>Defines the order of the bars. If TRUE the bars are 
ordered according to the z-scores of the processes. Otherwise the bars are 
ordered by the negative logarithm of the adjusted p-value</p>
</td></tr>
<tr><td><code id="GOBar_+3A_title">title</code></td>
<td>
<p>The title of the plot</p>
</td></tr>
<tr><td><code id="GOBar_+3A_zsc.col">zsc.col</code></td>
<td>
<p>Character vector to define the colour scale for the z-score of 
the form c(high, midpoint,low)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>display</code> is used to facet the plot the width of the panels 
will be proportional to the length of the x scale.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Load the included dataset
data(EC)

#Building the circ object
circ&lt;-circular_dat(EC$david, EC$genelist)

#Creating the bar plot
GOBar(circ)

#Faceting the plot
GOBar(circ, display='multiple')

## End(Not run)
</code></pre>

<hr>
<h2 id='GOBubble'>Bubble plot.</h2><span id='topic+GOBubble'></span>

<h3>Description</h3>

<p>The function creates a bubble plot of the input <code>data</code>. The
input <code>data</code> can be created with the help of the 
<code><a href="#topic+circle_dat">circle_dat</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GOBubble(data, display, title, colour, labels, ID = T, table.legend = T,
  table.col = T, bg.col = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GOBubble_+3A_data">data</code></td>
<td>
<p>A data frame with coloumns for category, GO ID, term, adjusted 
p-value, z-score, count(num of genes)</p>
</td></tr>
<tr><td><code id="GOBubble_+3A_display">display</code></td>
<td>
<p>A character vector. Indicates whether it should be a single 
plot ('single') or a facet plot with panels for each category 
(default='single')</p>
</td></tr>
<tr><td><code id="GOBubble_+3A_title">title</code></td>
<td>
<p>The title (on top) of the plot</p>
</td></tr>
<tr><td><code id="GOBubble_+3A_colour">colour</code></td>
<td>
<p>A character vector which defines the colour of the bubbles for 
each category</p>
</td></tr>
<tr><td><code id="GOBubble_+3A_labels">labels</code></td>
<td>
<p>Sets a threshold for the displayed labels. The threshold refers
to the -log(adjusted p-value) (default=5)</p>
</td></tr>
<tr><td><code id="GOBubble_+3A_id">ID</code></td>
<td>
<p>If TRUE then labels are IDs else terms</p>
</td></tr>
<tr><td><code id="GOBubble_+3A_table.legend">table.legend</code></td>
<td>
<p>Defines whether a table of GO ID and GO term should be 
displayed on the right side of the plot or not (default = TRUE)</p>
</td></tr>
<tr><td><code id="GOBubble_+3A_table.col">table.col</code></td>
<td>
<p>If TRUE then the table entries are coloured according to 
their category, if FALSE then entries are black</p>
</td></tr>
<tr><td><code id="GOBubble_+3A_bg.col">bg.col</code></td>
<td>
<p>Should only be used in case of a facet plot. If TRUE then the
panel backgrounds are coloured according to the displayed category</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The x- axis of the plot represents the z-score. The negative 
logarithm of the adjusted p-value (corresponding to the significance of the
term) is displayed on the y-axis. The area of the plotted circles is 
proportional to the number of genes assigned to the term. Each circle is 
coloured according to its category and labeled alternatively with the ID or 
term name.If static is set to FALSE the mouse hover effect will be enabled.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Load the included dataset
data(EC)

#Building the circ object
circ &lt;- circular_dat(EC$david, EC$genelist)

#Creating the bubble plot colouring the table entries according to the category
GOBubble(circ, table.col = T)

#Creating the bubble plot displaying the term instead of the ID and without the table
GOBubble(circ, ID = F, table.legend = F)

#Faceting the plot
GOBubble(circ, display = 'multiple')

## End(Not run)
</code></pre>

<hr>
<h2 id='GOChord'>Displays the relationship between genes and terms.</h2><span id='topic+GOChord'></span>

<h3>Description</h3>

<p>The GOChord function generates a circularly composited overview 
of selected/specific genes and their assigned processes or terms. More 
generally, it joins genes and processes via ribbons in an intersection-like
graph. The input can be generated with the <code><a href="#topic+chord_dat">chord_dat</a></code> 
function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GOChord(data, title, space, gene.order, gene.size, gene.space, nlfc = 1,
  lfc.col, lfc.min, lfc.max, ribbon.col, border.size, process.label, limit)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GOChord_+3A_data">data</code></td>
<td>
<p>The matrix represents the binary relation (1= is related to, 0= 
is not related to) between a set of genes (rows) and processes (columns); a
column for the logFC of the genes is optional</p>
</td></tr>
<tr><td><code id="GOChord_+3A_title">title</code></td>
<td>
<p>The title (on top) of the plot</p>
</td></tr>
<tr><td><code id="GOChord_+3A_space">space</code></td>
<td>
<p>The space between the chord segments of the plot</p>
</td></tr>
<tr><td><code id="GOChord_+3A_gene.order">gene.order</code></td>
<td>
<p>A character vector defining the order of the displayed gene
labels</p>
</td></tr>
<tr><td><code id="GOChord_+3A_gene.size">gene.size</code></td>
<td>
<p>The size of the gene labels</p>
</td></tr>
<tr><td><code id="GOChord_+3A_gene.space">gene.space</code></td>
<td>
<p>The space between the gene labels and the segement of the 
logFC</p>
</td></tr>
<tr><td><code id="GOChord_+3A_nlfc">nlfc</code></td>
<td>
<p>Defines the number of logFC columns (default=1)</p>
</td></tr>
<tr><td><code id="GOChord_+3A_lfc.col">lfc.col</code></td>
<td>
<p>The fill color for the logFC specified in the following form: 
c(color for low values, color for the mid point, color for the high values)</p>
</td></tr>
<tr><td><code id="GOChord_+3A_lfc.min">lfc.min</code></td>
<td>
<p>Specifies the minimium value of the logFC scale (default = -3)</p>
</td></tr>
<tr><td><code id="GOChord_+3A_lfc.max">lfc.max</code></td>
<td>
<p>Specifies the maximum value of the logFC scale (default = 3)</p>
</td></tr>
<tr><td><code id="GOChord_+3A_ribbon.col">ribbon.col</code></td>
<td>
<p>The background color of the ribbons</p>
</td></tr>
<tr><td><code id="GOChord_+3A_border.size">border.size</code></td>
<td>
<p>Defines the size of the ribbon borders</p>
</td></tr>
<tr><td><code id="GOChord_+3A_process.label">process.label</code></td>
<td>
<p>The size of the legend entries</p>
</td></tr>
<tr><td><code id="GOChord_+3A_limit">limit</code></td>
<td>
<p>A vector with two cutoff values (default= c(0,0)). The first 
value defines the minimum number of terms a gene has to be assigned to. The 
second the minimum number of genes assigned to a selected term.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>gene.order</code> argument has three possible options: &quot;logFC&quot;, 
&quot;alphabetical&quot;, &quot;none&quot;, which are quite self- explanatory.
</p>
<p>Maybe the most important argument of the function is <code>nlfc</code>.If your 
<code>data</code> does not contain a column of logFC values you have to set
<code>nlfc = 0</code>. Differential expression analysis can be performed for
multiple conditions and/or batches. Therefore, the data frame might contain
more than one logFC value per gene. To adjust to this situation the
<code>nlfc</code> argument is used as well. It is a numeric value and it defines
the number of logFC columns of your <code>data</code>. The default is &quot;1&quot;
assuming that most of the time only one contrast is considered.
</p>
<p>To represent the data more useful it might be necessary to reduce the 
dimension of <code>data</code>. This can be achieved with <code>limit</code>. The first
value of the vector defines the threshold for the minimum number of terms a
gene has to be assigned to in order to be represented in the plot. Most of
the time it is more meaningful to represent genes with various functions. A
value of 3 excludes all genes with less than three term assignments. 
Whereas the second value of the parameter restricts the number of terms 
according to the number of assigned genes. All terms with a count smaller 
or equal to the threshold are excluded.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+chord_dat">chord_dat</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load the included dataset
data(EC)

# Generating the binary matrix
chord&lt;-chord_dat(circ,EC$genes,EC$process)

# Creating the chord plot
GOChord(chord)

# Excluding process with less than 5 assigned genes
GOChord(chord, limit = c(0,5))

# Creating the chord plot genes ordered by logFC and a different logFC color scale
GOChord(chord,space=0.02,gene.order='logFC',lfc.col=c('red','black','cyan'))

## End(Not run)
</code></pre>

<hr>
<h2 id='GOCircle'>Circular visualization of the results of a functional analysis.</h2><span id='topic+GOCircle'></span>

<h3>Description</h3>

<p>The circular plot combines gene expression and gene- annotation 
enrichment data. A subset of terms is displayed like the <code>GOBar</code> plot 
in combination with a scatterplot of the gene expression data. The whole 
plot is drawn on a specific coordinate system to achieve the circular 
layout.The segments are labeled with the term ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GOCircle(data, title, nsub, rad1, rad2, table.legend = T, zsc.col, lfc.col,
  label.size, label.fontface)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GOCircle_+3A_data">data</code></td>
<td>
<p>A special data frame which should be the result of 
<code>circle_dat</code></p>
</td></tr>
<tr><td><code id="GOCircle_+3A_title">title</code></td>
<td>
<p>The title of the plot</p>
</td></tr>
<tr><td><code id="GOCircle_+3A_nsub">nsub</code></td>
<td>
<p>A numeric or character vector. If it's numeric then the number 
defines how many processes are displayed (starting from the first row of 
<code>data</code>). If it's a character string of processes then these processes 
are displayed</p>
</td></tr>
<tr><td><code id="GOCircle_+3A_rad1">rad1</code></td>
<td>
<p>The radius of the inner circle (default=2)</p>
</td></tr>
<tr><td><code id="GOCircle_+3A_rad2">rad2</code></td>
<td>
<p>The radius of the outer circle (default=3)</p>
</td></tr>
<tr><td><code id="GOCircle_+3A_table.legend">table.legend</code></td>
<td>
<p>Shall a table be displayd or not? (default=TRUE)</p>
</td></tr>
<tr><td><code id="GOCircle_+3A_zsc.col">zsc.col</code></td>
<td>
<p>Character vector to define the colour scale for the z-score of 
the form c(high, midpoint,low)</p>
</td></tr>
<tr><td><code id="GOCircle_+3A_lfc.col">lfc.col</code></td>
<td>
<p>A character vector specifying the colour for up- and 
down-regulated genes</p>
</td></tr>
<tr><td><code id="GOCircle_+3A_label.size">label.size</code></td>
<td>
<p>Size of the segment labels (default=5)</p>
</td></tr>
<tr><td><code id="GOCircle_+3A_label.fontface">label.fontface</code></td>
<td>
<p>Font style of the segment labels (default='bold')</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The outer circle shows a scatter plot for each term of the logFC of 
the assigned genes. The colours can be changed with the argument 
<code>lfc.col</code>.
</p>
<p>The <code>nsub</code> argument needs a bit more explanation to be used wisely. First of 
all, it can be a numeric or a character vector. If it is a character vector
then it contains the IDs or term descriptions of the displayed processes.If
<code>nsub</code> is a numeric vector then the number defines how many terms are 
displayed. It starts with the first row of the input data frame.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+circle_dat">circle_dat</a></code>, <code><a href="#topic+GOBar">GOBar</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load the included dataset
data(EC)

# Building the circ object
circ &lt;- circle_dat(EC$david, EC$genelist)

# Creating the circular plot
GOCircle(circ)

# Creating the circular plot with a different colour scale for the logFC
GOCircle(circ, lfc.col = c('purple', 'orange'))

# Creating the circular plot with a different colour scale for the z-score
GOCircle(circ, zsc.col = c('yellow', 'black', 'cyan'))

# Creating the circular plot with different font style
GOCircle(circ, label.size = 5, label.fontface = 'italic')

## End(Not run)
</code></pre>

<hr>
<h2 id='GOCluster'>Circular dendrogram.</h2><span id='topic+GOCluster'></span>

<h3>Description</h3>

<p>GOCluster generates a circular dendrogram of the <code>data</code> 
clustering using by default euclidean distance and average linkage.The 
inner ring displays the color coded logFC while the outside one encodes the
assigned terms to each gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GOCluster(data, process, metric, clust, clust.by, nlfc, lfc.col, lfc.min,
  lfc.max, lfc.space, lfc.width, term.col, term.space, term.width)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GOCluster_+3A_data">data</code></td>
<td>
<p>A data frame which should be the result of 
<code><a href="#topic+circle_dat">circle_dat</a></code> in case the data contains only one logFC column. 
Otherwise <code>data</code> is a data frame whereas the first column contains the
genes, the second the term and the following columns the logFCs of the 
different contrasts.</p>
</td></tr>
<tr><td><code id="GOCluster_+3A_process">process</code></td>
<td>
<p>A character vector of selected processes (ID or term
description)</p>
</td></tr>
<tr><td><code id="GOCluster_+3A_metric">metric</code></td>
<td>
<p>A character vector specifying the distance measure to be used 
(default='euclidean'), see <code>dist</code></p>
</td></tr>
<tr><td><code id="GOCluster_+3A_clust">clust</code></td>
<td>
<p>A character vector specifying the agglomeration method to be 
used (default='average'), see <code>hclust</code></p>
</td></tr>
<tr><td><code id="GOCluster_+3A_clust.by">clust.by</code></td>
<td>
<p>A character vector specifying if the clustering should be 
done for gene expression pattern or functional categories. By default the 
clustering is done based on the functional categories.</p>
</td></tr>
<tr><td><code id="GOCluster_+3A_nlfc">nlfc</code></td>
<td>
<p>If TRUE <code>data</code> contains multiple logFC columns (default= 
FALSE)</p>
</td></tr>
<tr><td><code id="GOCluster_+3A_lfc.col">lfc.col</code></td>
<td>
<p>Character vector to define the color scale for the logFC of 
the form c(high, midpoint,low)</p>
</td></tr>
<tr><td><code id="GOCluster_+3A_lfc.min">lfc.min</code></td>
<td>
<p>Specifies the minimium value of the logFC scale (default = -3)</p>
</td></tr>
<tr><td><code id="GOCluster_+3A_lfc.max">lfc.max</code></td>
<td>
<p>Specifies the maximum value of the logFC scale (default = 3)</p>
</td></tr>
<tr><td><code id="GOCluster_+3A_lfc.space">lfc.space</code></td>
<td>
<p>The space between the leafs of the dendrogram and the ring 
for the logFC</p>
</td></tr>
<tr><td><code id="GOCluster_+3A_lfc.width">lfc.width</code></td>
<td>
<p>The width of the logFC ring</p>
</td></tr>
<tr><td><code id="GOCluster_+3A_term.col">term.col</code></td>
<td>
<p>A character vector specifying the colors of the term bands</p>
</td></tr>
<tr><td><code id="GOCluster_+3A_term.space">term.space</code></td>
<td>
<p>The space between the logFC ring and the term ring</p>
</td></tr>
<tr><td><code id="GOCluster_+3A_term.width">term.width</code></td>
<td>
<p>The width of the term ring</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The inner ring can be split into smaller rings to display multiply
logFC values resulting from various comparisons.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Load the included dataset
data(EC)

#Generating the circ object
circ&lt;-circular_dat(EC$david, EC$genelist)

#Creating the cluster plot
GOCluster(circ, EC$process)

#Cluster the data according to gene expression and assigning a different color scale for the logFC
GOCluster(circ,EC$process,clust.by='logFC',lfc.col=c('darkgoldenrod1','black','cyan1'))

## End(Not run)
</code></pre>

<hr>
<h2 id='GOHeat'>Displays heatmap of the relationship between genes and terms.</h2><span id='topic+GOHeat'></span>

<h3>Description</h3>

<p>The GOHeat function generates a heatmap of the relationship 
between genes and terms. Biological processes are displayed in rows and
genes in columns. In addition genes are clustered to highlight groups of
genes with similar annotated functions. The input can be generated with the
<code><a href="#topic+chord_dat">chord_dat</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GOHeat(data, nlfc, fill.col)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GOHeat_+3A_data">data</code></td>
<td>
<p>The matrix represents the binary relation (1= is related to, 0= 
is not related to) between a set of genes (rows) and processes (columns)</p>
</td></tr>
<tr><td><code id="GOHeat_+3A_nlfc">nlfc</code></td>
<td>
<p>Defines the number of logFC columns (default = 0)</p>
</td></tr>
<tr><td><code id="GOHeat_+3A_fill.col">fill.col</code></td>
<td>
<p>Defines the color scale break points</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The heatmap has in general two modes which depend on the <code>nlfc</code>
argument. If <code>nlfc = 0</code>, so no logFC values are available, the 
coloring encodes for the overall number of processes the respective gene is
assigned to. In case of <code>nlfc = 1</code> the color corresponds to the logFC 
of the gene.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load the included dataset
data(EC)

# Generate the circ object
circ &lt;- circle_dat(EC$david, EC$genelist)

# Generate the chord object
chord &lt;- chord_dat(circ, EC$genes, EC$process)

# Create the plot with user-defined colors
GOHeat(chord, nlfc = 1, fill.col = c('red', 'yellow', 'green'))

## End(Not run)
</code></pre>

<hr>
<h2 id='GOVenn'>Venn diagram of differentially expressed genes.</h2><span id='topic+GOVenn'></span>

<h3>Description</h3>

<p>The function compares lists of differentially expressed genes 
and illustrates possible relations.Additionally it represents the variety 
of gene expression patterns within the intersection in small pie charts 
with three segements. Clockwise are shown the number of commonly up- 
regulated, commonly down- regulated and contra- regulated genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GOVenn(data1, data2, data3, title, label, lfc.col, circle.col, plot = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GOVenn_+3A_data1">data1</code></td>
<td>
<p>A data frame consisting of two columns: ID, logFC</p>
</td></tr>
<tr><td><code id="GOVenn_+3A_data2">data2</code></td>
<td>
<p>A data frame consisting of two columns: ID, logFC</p>
</td></tr>
<tr><td><code id="GOVenn_+3A_data3">data3</code></td>
<td>
<p>A data frame consisting of two columns: ID, logFC</p>
</td></tr>
<tr><td><code id="GOVenn_+3A_title">title</code></td>
<td>
<p>The title of the plot</p>
</td></tr>
<tr><td><code id="GOVenn_+3A_label">label</code></td>
<td>
<p>A character vector to define the legend keys</p>
</td></tr>
<tr><td><code id="GOVenn_+3A_lfc.col">lfc.col</code></td>
<td>
<p>A character vector determining the background colors of the 
pie segments representing up- and down- regulated genes</p>
</td></tr>
<tr><td><code id="GOVenn_+3A_circle.col">circle.col</code></td>
<td>
<p>A character vector to assign clockwise colors for the 
circles</p>
</td></tr>
<tr><td><code id="GOVenn_+3A_plot">plot</code></td>
<td>
<p>If TRUE only the venn diagram is plotted. Otherwise the function 
returns a list with two items: the actual plot and a list containing the 
overlap entries (default= TRUE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>plot</code> argument can be used to adjust the amount of 
information that is returned by calling the function. If you are only 
interested in the actual plot of the venn diagram, <code>plot</code> should be 
set to TRUE. Sometimes you also want to know the elements of the 
intersections. In this case <code>plot</code> should be set to FALSE and the 
function call will return a list of two items. The first item, that can be 
accessed by $plot, contains the plotting information. Additionally, a list
($table) will be returned containing the elements of the various overlaps.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Load the included dataset
data(EC)

#Generating the circ object
circ&lt;-circular_dat(EC$david, EC$genelist)

#Selecting terms of interest
l1&lt;-subset(circ,term=='heart development',c(genes,logFC))
l2&lt;-subset(circ,term=='plasma membrane',c(genes,logFC))
l3&lt;-subset(circ,term=='tissue morphogenesis',c(genes,logFC))

GOVenn(l1,l2,l3, label=c('heart development','plasma membrane','tissue morphogenesis'))

## End(Not run)
</code></pre>

<hr>
<h2 id='reduce_overlap'>Eliminates redundant terms.</h2><span id='topic+reduce_overlap'></span>

<h3>Description</h3>

<p>The function eliminates all terms with a gene overlap &gt;= set
threshold (<code>overlap</code>) The reduced dataset can be used to improve the
readability of plots such as <code>GOBubble</code> and <code>GOBar</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reduce_overlap(data, overlap)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reduce_overlap_+3A_data">data</code></td>
<td>
<p>A data frame created with <code>circle_dat</code>.</p>
</td></tr>
<tr><td><code id="reduce_overlap_+3A_overlap">overlap</code></td>
<td>
<p>Skalar indicating the threshold for gene overlap (default = 0.75).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is currently very slow.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load the included dataset
data(EC)

# Building the circ object
circ &lt;- circle_dat(EC$david, EC$genelist)

# Eliminate redundant terms
reduced_circ &lt;- reduce_overlap(circ)

# Plot reduced data
GOBubble(reduced_circ)


## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
