<!DOCTYPE html><html lang="en"><head><title>Help for package gpcp</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gpcp}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#phenotypeFile'><p>Example Phenotype Data</p></a></li>
<li><a href='#runGPCP'><p>Genomic Prediction of Cross Performance</p>
This function performs genomic prediction of cross performance using genotype and phenotype data.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genomic Prediction of Cross Performance</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Christine Nyaga &lt;cmn92@cornell.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>This function performs genomic prediction of cross performance using genotype and phenotype data.
  It processes data in several steps including loading necessary software, converting genotype data, processing phenotype data,
  fitting mixed models, and predicting cross performance based on weighted marker effects.
  For more information, see Labroo et al. (2023) &lt;<a href="https://doi.org/10.1007%2Fs00122-023-04377-z">doi:10.1007/s00122-023-04377-z</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Imports:</td>
<td>BiocManager, Rcpp, dplyr, sommer, AGHmatrix, snpStats,
VariantAnnotation, tools, magrittr, methods</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-04 13:31:44 UTC; cmn92</td>
</tr>
<tr>
<td>Author:</td>
<td>Marlee Labroo [aut],
  Christine Nyaga [cre, aut],
  Lukas Mueller [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-06 15:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='phenotypeFile'>Example Phenotype Data</h2><span id='topic+phenotypeFile'></span>

<h3>Description</h3>

<p>This is a sample phenotype dataset used for genomic prediction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phenotypeFile
</code></pre>


<h3>Format</h3>

<p>A data frame with 24 columns:
</p>

<dl>
<dt>ATW</dt><dd><p>Description of ATW</p>
</dd>
<dt>AUDPC_YAD</dt><dd><p>Area Under Disease Progress Curve for YAD</p>
</dd>
<dt>AUDPC_YMV</dt><dd><p>Area Under Disease Progress Curve for YMV</p>
</dd>
<dt>Accession</dt><dd><p>Genotype IDs for each individual</p>
</dd>
<dt>Block</dt><dd><p>Block information</p>
</dd>
<dt>DMC</dt><dd><p>Dry Matter Content values</p>
</dd>
<dt>Design</dt><dd><p>Experimental design</p>
</dd>
<dt>LOC</dt><dd><p>Location of the trials</p>
</dd>
<dt>NPH</dt><dd><p>Number of Plants Harvested</p>
</dd>
<dt>OXBI</dt><dd><p>Oxidation Index</p>
</dd>
<dt>Oxint180Minutes</dt><dd><p>Oxidation intensity after 180 minutes</p>
</dd>
<dt>PLOT</dt><dd><p>Plot number</p>
</dd>
<dt>REP</dt><dd><p>Replication number</p>
</dd>
<dt>Settweight</dt><dd><p>Weight of the planting setts</p>
</dd>
<dt>TTNPL</dt><dd><p>Total Tuber Number per Plant</p>
</dd>
<dt>TTWPL</dt><dd><p>Total Tuber Weight per Plant</p>
</dd>
<dt>Trial</dt><dd><p>Trial name or ID</p>
</dd>
<dt>Vigor</dt><dd><p>Plant vigor score</p>
</dd>
<dt>YIELD</dt><dd><p>Yield values</p>
</dd>
<dt>Year</dt><dd><p>Year of the experiment</p>
</dd>
<dt>Yield.per.plot..kg.</dt><dd><p>Yield per plot in kilograms</p>
</dd>
<dt>Yield_udj</dt><dd><p>Unadjusted Yield</p>
</dd>
<dt>rAUDPC_YAD</dt><dd><p>Relative AUDPC for YAD</p>
</dd>
<dt>rAUDPC_YMV</dt><dd><p>Relative AUDPC for YMV</p>
</dd>
</dl>



<h3>Source</h3>

<p>Generated for the gpcp package example
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phenotypeFile)
head(phenotypeFile)
</code></pre>

<hr>
<h2 id='runGPCP'>Genomic Prediction of Cross Performance
This function performs genomic prediction of cross performance using genotype and phenotype data.</h2><span id='topic+runGPCP'></span>

<h3>Description</h3>

<p>Genomic Prediction of Cross Performance
This function performs genomic prediction of cross performance using genotype and phenotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runGPCP(
  phenotypeFile,
  genotypeFile,
  genotypes,
  traits,
  weights = NA,
  userSexes = "",
  userFixed = NA,
  userRandom = NA,
  Ploidy = NA,
  NCrosses = NA
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="runGPCP_+3A_phenotypefile">phenotypeFile</code></td>
<td>
<p>A data frame containing phenotypic data, typically read from a CSV file.</p>
</td></tr>
<tr><td><code id="runGPCP_+3A_genotypefile">genotypeFile</code></td>
<td>
<p>Path to the genotypic data, either in VCF or HapMap format.</p>
</td></tr>
<tr><td><code id="runGPCP_+3A_genotypes">genotypes</code></td>
<td>
<p>A character string representing the column name in the phenotype file for the genotype IDs.</p>
</td></tr>
<tr><td><code id="runGPCP_+3A_traits">traits</code></td>
<td>
<p>A string of comma-separated trait names from the phenotype file.</p>
</td></tr>
<tr><td><code id="runGPCP_+3A_weights">weights</code></td>
<td>
<p>A numeric vector specifying weights for the traits.</p>
</td></tr>
<tr><td><code id="runGPCP_+3A_usersexes">userSexes</code></td>
<td>
<p>A string representing the column name corresponding to the individuals' sexes.</p>
</td></tr>
<tr><td><code id="runGPCP_+3A_userfixed">userFixed</code></td>
<td>
<p>A string of comma-separated fixed effect variables.</p>
</td></tr>
<tr><td><code id="runGPCP_+3A_userrandom">userRandom</code></td>
<td>
<p>A string of comma-separated random effect variables.</p>
</td></tr>
<tr><td><code id="runGPCP_+3A_ploidy">Ploidy</code></td>
<td>
<p>An integer representing the ploidy level of the organism.</p>
</td></tr>
<tr><td><code id="runGPCP_+3A_ncrosses">NCrosses</code></td>
<td>
<p>An integer specifying the number of top crosses to output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing predicted cross performance.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load phenotype data from CSV
phenotypeFile &lt;- read.csv(system.file("extdata", "phenotypeFile.csv", package = "gpcp"))
genotypeFile &lt;- system.file("extdata", "genotypeFile_Chr9and11.vcf", package = "gpcp")
finalcrosses &lt;- runGPCP(
    phenotypeFile = phenotypeFile,
    genotypeFile = genotypeFile,
    genotypes = "Accession",
    traits = "YIELD,DMC",
    weights = c(3, 1),
    userFixed = "LOC,REP",
    Ploidy = 2,
    NCrosses = 150
)
print(finalcrosses)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
