<!DOCTYPE html><html><head><title>Help for package gpmap</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gpmap}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#decompose_monotone'><p>Decompose genotype-phenotype map(s) using monotone regression</p></a></li>
<li><a href='#degree_of_monotonicity'><p>Degree of monotonicity of GP map</p></a></li>
<li><a href='#enumerate_genotypes'><p>Function for enumerating genotypes for <code class="reqn">N</code> biallelic loci</p></a></li>
<li><a href='#generate_gpmap'><p> Function for creating genotype-phenotype (GP) maps</p></a></li>
<li><a href='#gpmap-package'>
<p>Tools for analysing and plotting genotype-phenotype maps</p></a></li>
<li><a href='#GPmaps'><p>Dataset containing example GP maps</p></a></li>
<li><a href='#monotone_regression'><p> Perform monotone regression on a genotype-phenotype (GP) map</p></a></li>
<li><a href='#partial_genotype_order'><p>Generate partial ordering of genotype space based from allele content of genotypes</p></a></li>
<li><a href='#plot.gpmap'><p> Functions for creating lineplots of genotype-phenotype (GP) maps</p></a></li>
<li><a href='#print.gpmap'><p> Print function for <code>gpmap</code> objects</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysing and Plotting Genotype-Phenotype Maps</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-01-28</td>
</tr>
<tr>
<td>Author:</td>
<td>
    Arne B. Gjuvsland &lt;arne.gjuvsland@nmbu.no&gt; and Yunpeng Wang &lt;yunpeng.wng@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Arne B. Gjuvsland &lt;arne.gjuvsland@nmbu.no&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>isotone, plyr, ggplot2, foreach</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for studying genotype-phenotype maps for bi-allelic loci underlying quantitative phenotypes. The 0.1 version is released in connection with the publication of Gjuvsland et al (2013) and implements basic line plots and the monotonicity measures for GP maps presented in the paper. Reference: Gjuvsland AB, Wang Y, Plahte E and Omholt SW (2013) Monotonicity is a key feature of genotype-phenotype maps. Frontier in Genetics 4:216 &lt;<a href="https://doi.org/10.3389%2Ffgene.2013.00216">doi:10.3389/fgene.2013.00216</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-02-16 00:10:02 UTC</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-02-15 21:04:13 UTC; arne</td>
</tr>
</table>
<hr>
<h2 id='decompose_monotone'>Decompose genotype-phenotype map(s) using monotone regression </h2><span id='topic+decompose_monotone'></span><span id='topic+decompose_monotone_single'></span>

<h3>Description</h3>

<p>Functions for decomposing genotype-phenotype (GP) maps with <code class="reqn">N</code> biallelic loci using monotone regression from the <a href="https://cran.r-project.org/package=isotone"><code>isotone</code></a> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decompose_monotone(gpmap)
decompose_monotone_single(gpmap)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="decompose_monotone_+3A_gpmap">gpmap</code></td>
<td>
<p> A <code><a href="#topic+gpmap">gpmap</a></code> object </p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>decompose_monotone</code> works for any <code><a href="#topic+gpmap">gpmap</a></code> object (<code>values</code> is <code class="reqn">(3^N)xK</code> matrix of genotypic values) 
and calls the internal function <code>decompose_monotone_single</code> for each column. <code>decompose_monotone_single</code> takes a 
<code><a href="#topic+gpmap">gpmap</a></code> object with a single set of genotypic values (<code class="reqn">K=1</code>), loops through all <code class="reqn">2^N</code> possible
combinations of plusalleles, calls <code><a href="#topic+monotone_regression">monotone_regression</a></code> and identifies the best fit. The code uses the 
<a href="https://cran.r-project.org/package=foreach"><code>foreach</code></a> package and will run in parallel if a <em>parallel backend</em> is registered (see 
<a href="https://cran.r-project.org/package=foreach"><code>foreach</code></a> documentation).
</p>


<h3>Value</h3>

<p>The input <code><a href="#topic+gpmap">gpmap</a></code> is returned with two added elements
</p>
<table>
<tr><td><code>monoR2</code></td>
<td>
<p>The coefficient of determination of the monotone regression</p>
</td></tr>
<tr><td><code>values.mono</code></td>
<td>
<p>A matrix of genotypic values for the monotone component of genotype-phenotype map(s)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Arne B. Gjuvsland &lt;arne.gjuvsland@nmbu.no&gt; and Yunpeng Wang &lt;yunpeng.wng@gmail.com&gt;
</p>


<h3>References</h3>

<p>Leeuw J, Hornik K and Mair P (2009) Isotone Optimization in R: Pool-Adjacent-Violators Algorithm (PAVA) and Active Set Methods. Journal of Statistical Software 32(5) [<a href="https://www.jstatsoft.org/v32/i05">link</a>]
</p>
<p>Gjuvsland AB, Wang Y, Plahte E and Omholt SW (2013) Monotonicity is a key feature of genotype-phenotype maps. 
Front. Genet. 4:216. doi: 10.3389/fgene.2013.00216 [<a href="https://www.frontiersin.org/articles/10.3389/fgene.2013.00216/full">link</a>]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GPmaps)

#Additive GP map is monotone so monoR2=1 and values.mono=values
decompose_monotone(A)

#Pure AxA epistasis map
decompose_monotone(AA)

#two-locus example in Cheverud &amp; Routman (1995)
decompose_monotone(mouseweight)

#decompose four random 3-locus GP maps
set.seed(0)
randomGP &lt;- rnorm(3^2*4)
dim(randomGP) &lt;- c(9,4)
decompose_monotone(generate_gpmap(randomGP))

</code></pre>

<hr>
<h2 id='degree_of_monotonicity'>Degree of monotonicity of GP map</h2><span id='topic+degree_of_monotonicity'></span><span id='topic+degree_of_monotonicity_single'></span>

<h3>Description</h3>

<p> Functions for computing degree of monotonicity <code class="reqn">m</code> for <code><a href="#topic+gpmap">gpmap</a></code> objects. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> 
  degree_of_monotonicity(gpmap) 
  degree_of_monotonicity_single(gpmap) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="degree_of_monotonicity_+3A_gpmap">gpmap</code></td>
<td>
<p> A <code><a href="#topic+gpmap">gpmap</a></code> object </p>
</td></tr>
</table>


<h3>Details</h3>

 
<p><code>degree_of_monotonicity</code> works for any <code><a href="#topic+gpmap">gpmap</a></code> object (<code>values</code> is <code class="reqn">(3^N)xK</code> matrix of genotypic values) 
and calls the internal function <code>degree_of_monotonicity_single</code> for each column. <code>degree_of_monotonicity_single</code> computes
substitution effect, locus weights and per-locus and overall degree of monotonicity as decribed in Gjuvsland <em>et al.</em> (2013).
</p>


<h3>Value</h3>

 
<p><code>degree_of_monotonicity</code> returns the input <code>gpmap</code> with the following added fields:
</p>
<table>
<tr><td><code>degree.monotonicity</code></td>
<td>
<p>Overall degree of monotonicity for the <code class="reqn">K</code> GP maps</p>
</td></tr>
<tr><td><code>degree.monotonicity.locus</code></td>
<td>
<p>Data frame with per locus degree of monotonicity for the <code class="reqn">K</code> GP maps</p>
</td></tr>
<tr><td><code>locus.weight</code></td>
<td>
<p> Data frame with locus weigths</p>
</td></tr>
</table>


<h3>Author(s)</h3>

 
<p>Arne B. Gjuvsland &lt;arne.gjuvsland@nmbu.no&gt; and Yunpeng Wang &lt;yunpeng.wng@gmail.com&gt;
</p>


<h3>References</h3>

 
<p>Gjuvsland AB, Wang Y, Plahte E and Omholt SW (2013) Monotonicity is a key feature of genotype-phenotype maps. 
Front. Genet. 4:216. doi: 10.3389/fgene.2013.00216 [<a href="https://www.frontiersin.org/articles/10.3389/fgene.2013.00216/full">link</a>]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GPmaps)

#Additive GP map is monotone 
degree_of_monotonicity(A)

#Pure AxA epistasis map
degree_of_monotonicity(AA)

#two-locus example in Cheverud &amp; Routman (1995)
degree_of_monotonicity(mouseweight)
</code></pre>

<hr>
<h2 id='enumerate_genotypes'>Function for enumerating genotypes for <code class="reqn">N</code> biallelic loci</h2><span id='topic+enumerate_genotypes'></span>

<h3>Description</h3>

<p> Function for enumerating all <code class="reqn">3^N</code> genotypes for <code class="reqn">N</code> biallelic loci. Optional specification of names of loci and alleles. 
genotypes. Generates a data frame of multilocus genotypes in the sequence used for objects of class <code><a href="#topic+gpmap">gpmap</a></code>.   </p>


<h3>Usage</h3>

<pre><code class='language-R'> enumerate_genotypes(nloci=1, locinames=NULL, allelenames=NULL) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="enumerate_genotypes_+3A_nloci">nloci</code></td>
<td>
<p> The number of loci <code class="reqn">N</code> </p>
</td></tr>
<tr><td><code id="enumerate_genotypes_+3A_locinames">locinames</code></td>
<td>
<p> An optional character vector with <code class="reqn">N</code> names of loci </p>
</td></tr>
<tr><td><code id="enumerate_genotypes_+3A_allelenames">allelenames</code></td>
<td>
<p> An optional character object specifying allele names </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Unless specified locinames default to &quot;Locus 1&quot;, &quot;Locus 2&quot;,..,&quot;Locus N&quot;. 
</p>
<p>If <code>allelenames</code> is not specified then the alleles will be named &quot;1&quot; and &quot;2&quot;.
</p>


<h3>Value</h3>

<p>Returns a data frame with locinames as colnames, and with <code class="reqn">3^N</code> rows specifying all possible genotypes in the sequence 
used for all GP maps in the package (the same sequence as used in Gjuvsland <em>et al.</em>( 2011)), where the genotype at the first locus varies fastest,
then the second locus, and so on:
</p>
<pre>
  Locus_1 Locus_2
1      11      11
2      12      11
3      22      11
4      11      12
5      12      12
6      22      12
7      11      22
8      12      22
9      22      22
</pre>


<h3>Author(s)</h3>

<p>Arne B. Gjuvsland &lt;arne.gjuvsland@nmbu.no&gt; and Yunpeng Wang &lt;yunpeng.wng@gmail.com&gt;
</p>


<h3>References</h3>

<p>Gjuvsland AB, Vik JO, Woolliams JA, Omholt SW (2011) Order-preserving principles 
underlying genotype-phenotype maps ensure high additive proportions of genetic 
variance. Journal of Evolutionary Biology 24(10):2269-2279 [<a href="https://onlinelibrary.wiley.com/doi/full/10.1111/j.1420-9101.2011.02358.x">link</a>]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#genotypes for a single locus
enumerate_genotypes()

#genotypes for two loci "A" and "B", with alleles named "H" and "L"
enumerate_genotypes(2,c("A","B"),c("H","L"))

#genotypes for the two-locus example in Cheverud &amp; Routman (1995)
enumerate_genotypes(2,c("D7Mit17","D1Mit7"),rbind(c('A1','A2'),c('B1','B2')))
</code></pre>

<hr>
<h2 id='generate_gpmap'> Function for creating genotype-phenotype (GP) maps </h2><span id='topic+generate_gpmap'></span><span id='topic+gpmap'></span>

<h3>Description</h3>

<p> Function for creating a <code><a href="#topic+gpmap">gpmap</a></code> object representing a genotype-phenotype (GP) map 
for <code class="reqn">N</code> biallelic loci or more generally <code class="reqn">K</code> such maps, from a matrix of genotypic values. </p>


<h3>Usage</h3>

<pre><code class='language-R'> generate_gpmap(y, locinames = NULL, allelenames = NULL, mapnames = NULL) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_gpmap_+3A_y">y</code></td>
<td>
<p> A <code class="reqn">(3^N\times K)</code> matrix or numeric with each column specifying the <code class="reqn">3^N</code> genotypic values for <code class="reqn">K</code> GP maps </p>
</td></tr>
<tr><td><code id="generate_gpmap_+3A_locinames">locinames</code></td>
<td>
<p> An optional character vector with <code class="reqn">N</code> names of loci </p>
</td></tr>
<tr><td><code id="generate_gpmap_+3A_allelenames">allelenames</code></td>
<td>
<p> An optional character object specifying allele names </p>
</td></tr>
<tr><td><code id="generate_gpmap_+3A_mapnames">mapnames</code></td>
<td>
<p> An optional character vector with <code class="reqn">K</code> names of GP maps / phenotypes </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Arguments <code>locinames</code> and <code>allelenames</code> are passed on to <code><a href="#topic+enumerate_genotypes">enumerate_genotypes</a></code>, and the genotypic values in <code>y</code> should be given 
in the same sequence as the sequence of genotypes returned by <code><a href="#topic+enumerate_genotypes">enumerate_genotypes</a></code>. If <code>mapnames</code> is not specified then the GP maps 
will be named &quot;GPmap_1&quot;, &quot;GPmap_2&quot;,..,&quot;GPmap_K&quot;.
</p>


<h3>Value</h3>

<p>The function returns an object of class <code><a href="#topic+gpmap">gpmap</a></code> containing the following components
</p>
<table>
<tr><td><code>values</code></td>
<td>
<p> The vector or matrix of genotypic values </p>
</td></tr>
<tr><td><code>nloci</code></td>
<td>
<p> The number of loci in the map </p>
</td></tr>
<tr><td><code>genotypes</code></td>
<td>
<p> Data frame with enumeration of genotypes </p>
</td></tr>
<tr><td><code>locinames</code></td>
<td>
<p> Character vector with names for all loci </p>
</td></tr>
<tr><td><code>mapname</code></td>
<td>
<p> The name(s) of the GP map </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Arne B. Gjuvsland &lt;arne.gjuvsland@nmbu.no&gt; and Yunpeng Wang &lt;yunpeng.wng@gmail.com&gt;
</p>


<h3>References</h3>

<p>Gjuvsland AB, Vik JO, Woolliams JA, Omholt SW (2011) Order-preserving principles underlying 
genotype-phenotype maps ensure high additive proportions of genetic variance.
Journal of Evolutionary Biology 24(10):2269-2279 [<a href="https://onlinelibrary.wiley.com/doi/full/10.1111/j.1420-9101.2011.02358.x">link</a>]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#inter- and intra-locus additive GPmap with two loci 
generate_gpmap(c(-2,1,0,-1,0,1,0,1,2),mapnames="Additive") 

#two-locus GP map with AxA epistasis for loci named A and B
generate_gpmap(c(-1,0,1,0,0,0,1,0,-1),locinames=c("A","B"),mapnames="AxA")

#random GP map with 3 loci
set.seed(0)
generate_gpmap(rnorm(27))
</code></pre>

<hr>
<h2 id='gpmap-package'>
Tools for analysing and plotting genotype-phenotype maps</h2><span id='topic+gpmap-package'></span>

<h3>Description</h3>

<p>The <span class="pkg">gpmap</span> package deals with genotype-phenotype maps for biallelic loci underlying 
quantitative phenotypes. The package provides a class <code>gpmaps</code>, analysis functions and basic
lineplots. The package is designed for studying the properties of GP maps without
reference to any particular population, i.e. the physiological (Cheverud &amp; Routman, 1995)
or functional (Hansen, 2001) properties of the GP map. This is opposed to statistical effects underlying 
most of quantitative genetics, where the GP-map is analysed togehter with genotype frequencies in a given 
population (e.g. Lynch &amp; Walsh, 1998).
</p>
<p>In version 0.1 which is released as part of the publication of Gjuvsland <em>et al.</em> (2013) 
we have implemented functionality for studying monotonicity Gjuvsland <em>et al.</em> (2011) of GP maps. 
The package utilizes the <a href="https://cran.r-project.org/package=isotone"><span class="pkg">isotone</span></a> package for monotone regression, 
and the <a href="https://cran.r-project.org/package=foreach"><span class="pkg">foreach</span></a> package for parallel computation.
</p>
<p>The package consists of the following high-level functions :
<code><a href="#topic+enumerate_genotypes">enumerate_genotypes</a></code>, <code><a href="#topic+generate_gpmap">generate_gpmap</a></code>,  
<code><a href="#topic+degree_of_monotonicity">degree_of_monotonicity</a></code>, <code><a href="#topic+decompose_monotone">decompose_monotone</a></code> and
<code><a href="#topic+plot.gpmap">plot.gpmap</a></code>
</p>


<h3>Author(s)</h3>

<p>Arne B. Gjuvsland &lt;arne.gjuvsland@nmbu.no&gt; and Yunpeng Wang &lt;yunpeng.wng@gmail.com&gt;
</p>


<h3>References</h3>

<p>Cheverud JM &amp; Routman EJ (1995) Epistasis and Its Controbution to Genetic Variance Components. 
Genetics 139:1455-1461 [<a href="https://www.genetics.org/content/139/3/1455.abstract">link</a>]
</p>
<p>Gjuvsland AB, Vik JO, Woolliams JA, Omholt SW (2011) Order-preserving principles underlying 
genotype-phenotype maps ensure high additive proportions of genetic variance.
Journal of Evolutionary Biology 24(10):2269-2279 [<a href="https://onlinelibrary.wiley.com/doi/full/10.1111/j.1420-9101.2011.02358.x">link</a>]
</p>
<p>Gjuvsland AB, Wang Y, Plahte E and Omholt SW (2013) Monotonicity is a key feature of genotype-phenotype maps. 
Front. Genet. 4:216. doi: 10.3389/fgene.2013.00216 [<a href="https://www.frontiersin.org/articles/10.3389/fgene.2013.00216/full">link</a>]
</p>
<p>Hansen T &amp; Wagner GP (2001) Modeling genetic Architecture: A Multilinear Theory of gene Interaction. 
Theoretical Population Biology 59:61-86 [<a href="https://www.sciencedirect.com/science/article/pii/S0040580900915089">link</a>]
</p>
<p>Leeuw J, Hornik K and Mair P (2009) Isotone Optimization in R: Pool-Adjacent-Violators 
Algorithm (PAVA) and Active Set Methods. Journal of Statistical Software 32(5) [<a href="https://www.jstatsoft.org/v32/i05">link</a>]
</p>
<p>Lynch M &amp; Walsh B (1998) Genetics and Analysis of Quantitative Traits, Sunderland, MA: Sinauer Associates
</p>

<hr>
<h2 id='GPmaps'>Dataset containing example GP maps</h2><span id='topic+A'></span><span id='topic+D'></span><span id='topic+AA'></span><span id='topic+AD'></span><span id='topic+DA'></span><span id='topic+DD'></span><span id='topic+mouseweight'></span>

<h3>Description</h3>

<p> Example GP maps with two loci including: The orthogonal GP maps <code>A</code>, <code>D</code>, <code>AA</code>, <code>AD</code>, <code>DA</code> and <code>DD</code>  used in 
decomposition of genetic variance (Zeng <em>et al.</em> 2005). The GP map <code>mouseweight</code> for body weight studied by Cheverud <em>et al.</em> (1995). </p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GPmaps)</code></pre>


<h3>Format</h3>

<p>Objects of class <code><a href="#topic+gpmap">gpmap</a></code></p>


<h3>References</h3>

<p>Cheverud JM &amp; Routman EJ (1995) Epistasis and Its Controbution to Genetic Variance Components. 
Genetics 139:1455-1461 [<a href="https://www.genetics.org/content/139/3/1455.abstract">link</a>]
</p>
<p>Zeng ZB, Wang T, Zou W. (2005). Modelling quantitative trait loci and interpretation of models. 
Genetics 169: 1711-1725. [<a href="https://www.genetics.org/content/169/3/1711.full">link</a>]
</p>

<hr>
<h2 id='monotone_regression'> Perform monotone regression on a genotype-phenotype (GP) map </h2><span id='topic+monotone_regression'></span>

<h3>Description</h3>

<p>The function uses <code><a href="#topic+partial_genotype_order">partial_genotype_order</a></code> and <code><a href="isotone.html#topic+activeSet">activeSet</a></code> from the <a href="https://cran.r-project.org/package=isotone">isotone</a> package to do 
monotone regression (Leeuw <em>et al.</em>, 2009) on a GP map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>monotone_regression(gpmap, plusallele)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="monotone_regression_+3A_gpmap">gpmap</code></td>
<td>
<p> An object of class <code><a href="#topic+gpmap">gpmap</a></code> </p>
</td></tr>
<tr><td><code id="monotone_regression_+3A_plusallele">plusallele</code></td>
<td>
<p>An <code class="reqn">N</code> vector of allele indexes (1 or 2)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Element <em>i</em> in <code>plusallele</code> specifies the ordering of the genotypes at locus <em>i</em>, if the element is 1 then 11 &lt; 12 &lt; 22 and conversely if it is 2 then
22 &lt; 12 &lt; 11. <code>monotone_regression</code> calls <code><a href="#topic+partial_genotype_order">partial_genotype_order</a></code> to obtain the partial ordering of genotypic values for the given <code>plusalleles</code>.
This partial ordering is then used together with the GP map itself as input to the <code><a href="isotone.html#topic+activeSet">activeSet</a></code> function from the package 
<a href="https://cran.r-project.org/package=isotone">isotone</a>.
</p>


<h3>Value</h3>

<p><code>monotone_regression</code> returns the output from <code><a href="isotone.html#topic+activeSet">activeSet</a></code> directly. 
</p>


<h3>Author(s)</h3>

<p>Arne B. Gjuvsland &lt;arne.gjuvsland@nmbu.no&gt; and Yunpeng Wang &lt;yunpeng.wng@gmail.com&gt;
</p>


<h3>References</h3>

<p>Leeuw J, Hornik K and Mair P (2009) Isotone Optimization in R: Pool-Adjacent-Violators 
Algorithm (PAVA) and Active Set Methods. Journal of Statistical Software 32(5) [<a href="https://www.jstatsoft.org/v32/i05">link</a>]
</p>
<p>Gjuvsland AB, Wang Y, Plahte E and Omholt SW (2013) Monotonicity is a key feature of genotype-phenotype maps. 
Front. Genet. 4:216. doi: 10.3389/fgene.2013.00216 [<a href="https://www.frontiersin.org/articles/10.3389/fgene.2013.00216/full">link</a>]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GPmaps)

#Additive GP map is monotone 
monotone_regression(A,c(2,2))

#Pure AxA epistasis map
monotone_regression(AA,c(2,2))

#two-locus example in Cheverud &amp; Routman (1995)
monotone_regression(mouseweight,c(1,1))
</code></pre>

<hr>
<h2 id='partial_genotype_order'>Generate partial ordering of genotype space based from allele content of genotypes</h2><span id='topic+partial_genotype_order'></span>

<h3>Description</h3>

<p> Generate the strict partial order on genotype space specified in eq. (13) in Gjuvsland <em>et al.</em> (2011). 
For a genotype space with <code class="reqn">N</code> biallelic loci a minimal description of this partial order is given by <code class="reqn">2N*3^{(N-1)}</code>
inequalities. </p>


<h3>Usage</h3>

<pre><code class='language-R'> partial_genotype_order(plusallele) </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="partial_genotype_order_+3A_plusallele">plusallele</code></td>
<td>
<p> A <code class="reqn">N</code>-vector with indexes (1 or 2) for the +allele for each locus </p>
</td></tr>
</table>


<h3>Details</h3>

<p>In short a partial order on a set is a binary relation defining a parwise ordering of some pairs of elements in the set, for a formal definiton see <a href="https://en.wikipedia.org/wiki/Partial_order">https://en.wikipedia.org/wiki/Partial_order</a>. 
In the partial order on the set of genotypes defined in Gjuvsland et al. (2011) the comparable pairs of genotypes are equal at every locus except one, 
while all other pairs of genotypes are incomparable. This partial ordering of genotype space is implicit in the regression on gene content 
(the number of alleles with a given index in each genotype) used for decomposition of tne genotypic value in quantitative genetics (see e.g. Lynch and Walsh page 65).
</p>


<h3>Value</h3>

<p>Returns a <code class="reqn">(2N*3^{(N-1)}) \times 2</code> matrix of genotype indexes. The genotype indexes refer to row number in the genotype sequence set up in <code><a href="#topic+enumerate_genotypes">enumerate_genotypes</a></code>.  
Each row vector in the matrix contains the genotype indexes of one comparable pairs, and if the first index is <code class="reqn">k</code> and the second is <code class="reqn">l</code> then genotype[<code class="reqn">k</code>] &lt; genotype[<code class="reqn">l</code>]. 
</p>


<h3>Author(s)</h3>

<p>Arne B. Gjuvsland &lt;arne.gjuvsland@nmbu.no&gt; and Yunpeng Wang &lt;yunpeng.wng@gmail.com&gt;
</p>


<h3>References</h3>

<p>Gjuvsland AB, Vik JO, Woolliams JA, Omholt SW (2011) Order-preserving principles underlying 
genotype-phenotype maps ensure high additive proportions of genetic variance.
Journal of Evolutionary Biology 24(10):2269-2279 [<a href="https://onlinelibrary.wiley.com/doi/full/10.1111/j.1420-9101.2011.02358.x">link</a>]
</p>
<p>Lynch M &amp; Walsh B (1998) Genetics and Analysis of Quantitative Traits, Sunderland, MA: Sinauer Associates
</p>

<hr>
<h2 id='plot.gpmap'> Functions for creating lineplots of genotype-phenotype (GP) maps </h2><span id='topic+plot.gpmap'></span><span id='topic+plot.gpmap.decomp.mono'></span>

<h3>Description</h3>

<p> Function for creating lineplots for genotype-phenotype (GP) map (an object of class <code><a href="#topic+gpmap">gpmap</a></code>) 
with 1-3 biallelic loci.</p>


<h3>Usage</h3>

<pre><code class='language-R'> 
## S3 method for class 'gpmap'
plot(x, show=1, decomposed=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.gpmap_+3A_x">x</code></td>
<td>
<p> A <code>gpmap</code> object </p>
</td></tr>
<tr><td><code id="plot.gpmap_+3A_show">show</code></td>
<td>
<p> Which map (only used if &gt;1 map in <code>gpmap</code> object) to plot</p>
</td></tr>
<tr><td><code id="plot.gpmap_+3A_decomposed">decomposed</code></td>
<td>
<p> Decomposition into monotone and non-monotone component plotted if TRUE</p>
</td></tr>
<tr><td><code id="plot.gpmap_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Arne B. Gjuvsland &lt;arne.gjuvsland@nmbu.no&gt; and Yunpeng Wang &lt;yunpeng.wng@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GPmaps)

#plot additive gpmap
plot(A)

#plot decomposition of GPmap from Cheverud &amp; Routman (1995)
decomp &lt;- decompose_monotone(mouseweight)
plot(decomp, decomposed=TRUE)
</code></pre>

<hr>
<h2 id='print.gpmap'> Print function for <code><a href="#topic+gpmap">gpmap</a></code> objects </h2><span id='topic+print.gpmap'></span>

<h3>Description</h3>

<p> Print a summary of a genotype-phenotype (GP) map (an object of class <code><a href="#topic+gpmap">gpmap</a></code>) 
with 1-3 <code class="reqn">N</code> biallelic loci.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gpmap'
print(x, ...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.gpmap_+3A_x">x</code></td>
<td>
<p> A <code><a href="#topic+gpmap">gpmap</a></code> object </p>
</td></tr>
<tr><td><code id="print.gpmap_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Prints name(s) of GP map(s) and loci, a summary of genotypic values. 
Monotonicity measures are printed if available.
</p>


<h3>Author(s)</h3>

<p>Arne B. Gjuvsland &lt;arne.gjuvsland@nmbu.no&gt; and Yunpeng Wang &lt;yunpeng.wng@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GPmaps)
print(A)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
