<!DOCTYPE html><html><head><title>Help for package gprofiler2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gprofiler2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gconvert'><p>Gene ID conversion.</p></a></li>
<li><a href='#get_base_url'><p>Get the current base URL.</p></a></li>
<li><a href='#get_tls_version'><p>Get the TLS version for SSL</p></a></li>
<li><a href='#get_user_agent'><p>Get current user agent string.</p></a></li>
<li><a href='#get_version_info'><p>Get version info of g:Profiler data sources</p></a></li>
<li><a href='#gorth'><p>Orthology search.</p></a></li>
<li><a href='#gost'><p>Gene list functional enrichment.</p></a></li>
<li><a href='#gostplot'><p>Manhattan plot of functional enrichment results.</p></a></li>
<li><a href='#gsnpense'><p>Convert SNP rs identifiers to genes.</p></a></li>
<li><a href='#mapViridis'><p>Map vector of numeric values to Viridis color scale.</p></a></li>
<li><a href='#publish_gostplot'><p>Create and save an annotated Manhattan plot of enrichment results.</p></a></li>
<li><a href='#publish_gosttable'><p>Create and save a table with the functional enrichment analysis results.</p></a></li>
<li><a href='#random_query'><p>Generate a random gene list for testing.</p></a></li>
<li><a href='#set_base_url'><p>Set the base URL.</p></a></li>
<li><a href='#set_tls_version'><p>Set the TLS version to use for SSL</p></a></li>
<li><a href='#set_user_agent'><p>Set custom user agent string.</p></a></li>
<li><a href='#upload_GMT_file'><p>Upload custom annotations for functional enrichment analysis in g:GOSt.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Interface to the 'g:Profiler' Toolset</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;, Uku Raudvere &lt;uku.raudvere@ut.ee&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A toolset for functional enrichment analysis and visualization, gene/protein/SNP identifier conversion and mapping orthologous genes across species via 'g:Profiler' (<a href="https://biit.cs.ut.ee/gprofiler/">https://biit.cs.ut.ee/gprofiler/</a>). 
    The main tools are: 
    (1) 'g:GOSt' - functional enrichment analysis and visualization of gene lists; 
    (2) 'g:Convert' - gene/protein/transcript identifier conversion across various namespaces;
    (3) 'g:Orth' - orthology search across species;
    (4) 'g:SNPense' - mapping SNP rs identifiers to chromosome positions, genes and variant effects.
    This package is an R interface corresponding to the 2019 update of 'g:Profiler' and provides access to 'g:Profiler' for versions 'e94_eg41_p11' and higher. See the package 'gProfileR' for accessing older versions from the 'g:Profiler' toolset.  </td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://biit.cs.ut.ee/gprofiler/page/contact">https://biit.cs.ut.ee/gprofiler/page/contact</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>jsonlite, RCurl, ggplot2, plotly, tidyr (&ge; 1.0.0), crosstalk,
grDevices, gridExtra, grid, viridisLite, dplyr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, prettydoc</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-23 20:09:31 UTC; Liis</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-23 21:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='gconvert'>Gene ID conversion.</h2><span id='topic+gconvert'></span>

<h3>Description</h3>

<p>Interface to the g:Profiler tool g:Convert (<a href="https://biit.cs.ut.ee/gprofiler/convert">https://biit.cs.ut.ee/gprofiler/convert</a>) that uses the information in Ensembl databases to handle hundreds of types of identifiers for genes, proteins, transcripts, microarray probesets, etc, for many species,
experimental platforms and biological databases.
The input is flexible: it accepts a mixed list of IDs and recognises their types automatically.
It can also serve as a service to get all genes belonging to a particular functional category.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gconvert(
  query,
  organism = "hsapiens",
  target = "ENSG",
  numeric_ns = "",
  mthreshold = Inf,
  filter_na = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gconvert_+3A_query">query</code></td>
<td>
<p>character vector that can consist of mixed types of gene IDs (proteins, transcripts, microarray IDs, etc), SNP IDs, chromosomal intervals or term IDs.</p>
</td></tr>
<tr><td><code id="gconvert_+3A_organism">organism</code></td>
<td>
<p>organism name. Organism names are constructed by
concatenating the first letter of the name and the family name. Example: human
- 'hsapiens', mouse - 'mmusculus'.</p>
</td></tr>
<tr><td><code id="gconvert_+3A_target">target</code></td>
<td>
<p>target namespace.</p>
</td></tr>
<tr><td><code id="gconvert_+3A_numeric_ns">numeric_ns</code></td>
<td>
<p>namespace to use for fully numeric IDs (<a href="https://biit.cs.ut.ee/gprofiler/page/namespaces-list">list of available namespaces</a>).</p>
</td></tr>
<tr><td><code id="gconvert_+3A_mthreshold">mthreshold</code></td>
<td>
<p>maximum number of results per initial alias to show. Shows all by default.</p>
</td></tr>
<tr><td><code id="gconvert_+3A_filter_na">filter_na</code></td>
<td>
<p>logical indicating whether to filter out results without a
corresponding target.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a data.frame which is a table closely corresponding to the
web interface output.
</p>
<p>The result fields are further described in the <a href="https://cran.r-project.org/package=gprofiler2/vignettes/gprofiler2.html">vignette</a>.
</p>


<h3>Author(s)</h3>

<p>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;, Uku Raudvere &lt;uku.raudvere@ut.ee&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gconvert(c("POU5F1", "SOX2", "NANOG"), organism = "hsapiens", target="AFFY_HG_U133_PLUS_2")
</code></pre>

<hr>
<h2 id='get_base_url'>Get the current base URL.</h2><span id='topic+get_base_url'></span>

<h3>Description</h3>

<p>Get the current base URL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_base_url()
</code></pre>

<hr>
<h2 id='get_tls_version'>Get the TLS version for SSL</h2><span id='topic+get_tls_version'></span>

<h3>Description</h3>

<p>Get the TLS version for SSL
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tls_version()
</code></pre>

<hr>
<h2 id='get_user_agent'>Get current user agent string.</h2><span id='topic+get_user_agent'></span>

<h3>Description</h3>

<p>Get the HTTP User-Agent string.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_user_agent()
</code></pre>

<hr>
<h2 id='get_version_info'>Get version info of g:Profiler data sources</h2><span id='topic+get_version_info'></span>

<h3>Description</h3>

<p>Get version info of g:Profiler data sources
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_version_info(organism = "hsapiens")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_version_info_+3A_organism">organism</code></td>
<td>
<p>organism name. Organism names are constructed by concatenating the first letter of the name and the family name. Example: human - 'hsapiens', mouse - 'mmusculus'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named nested list that includes the versions for all the data sources (GO, KEGG, Reactome, WP, etc) at the time of the data extraction for the given organism.
The versions correspond to the g:Profiler version embedded in the base_url which is also returned by this function under the name 'gprofiler_version'.
</p>


<h3>Author(s)</h3>

<p>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: version_info &lt;- get_version_info(organism = "hsapiens")

</code></pre>

<hr>
<h2 id='gorth'>Orthology search.</h2><span id='topic+gorth'></span>

<h3>Description</h3>

<p>Interface to the g:Profiler tool g:Orth (<a href="https://biit.cs.ut.ee/gprofiler/orth">https://biit.cs.ut.ee/gprofiler/orth</a>) that, given a target organism, retrieves the genes of the target organism that are similar in sequence to the source organism genes in the input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gorth(
  query,
  source_organism = "hsapiens",
  target_organism = "mmusculus",
  numeric_ns = "",
  mthreshold = Inf,
  filter_na = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gorth_+3A_query">query</code></td>
<td>
<p>character vector of gene IDs to be translated.</p>
</td></tr>
<tr><td><code id="gorth_+3A_source_organism">source_organism</code></td>
<td>
<p>name of the source organism. Organism names are constructed by concatenating
the first letter of the name and the family name. Example: human - 'hsapiens',
mouse - 'mmusculus'.</p>
</td></tr>
<tr><td><code id="gorth_+3A_target_organism">target_organism</code></td>
<td>
<p>name of the target organism. Organism names are constructed by concatenating
the first letter of the name and the family name. Example: human - 'hsapiens',
mouse - 'mmusculus'.</p>
</td></tr>
<tr><td><code id="gorth_+3A_numeric_ns">numeric_ns</code></td>
<td>
<p>namespace to use for fully numeric IDs (<a href="https://biit.cs.ut.ee/gprofiler/page/namespaces-list">list of available namespaces</a>).</p>
</td></tr>
<tr><td><code id="gorth_+3A_mthreshold">mthreshold</code></td>
<td>
<p>maximum number of ortholog names per gene to show.</p>
</td></tr>
<tr><td><code id="gorth_+3A_filter_na">filter_na</code></td>
<td>
<p>logical indicating whether to filter out results without a
corresponding target name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a data.frame which is a table closely corresponding to the
web interface output.
</p>
<p>The result fields are further described in the <a href="https://cran.r-project.org/package=gprofiler2/vignettes/gprofiler2.html">vignette</a>.
</p>


<h3>Author(s)</h3>

<p>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;, Uku Raudvere &lt;uku.raudvere@ut.ee&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gorth(c("Klf4","Pax5","Sox2","Nanog"), source_organism="mmusculus", target_organism="hsapiens")
</code></pre>

<hr>
<h2 id='gost'>Gene list functional enrichment.</h2><span id='topic+gost'></span>

<h3>Description</h3>

<p>Interface to the g:Profiler tool g:GOSt (<a href="https://biit.cs.ut.ee/gprofiler/gost">https://biit.cs.ut.ee/gprofiler/gost</a>) for functional enrichments analysis of gene lists.
In case the input 'query' is a list of gene vectors, results for multiple queries will be returned in the same data frame with column 'query' indicating the corresponding query name.
If 'multi_query' is selected, the result is a data frame for comparing multiple input lists,
just as in the web tool.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gost(
  query,
  organism = "hsapiens",
  ordered_query = FALSE,
  multi_query = FALSE,
  significant = TRUE,
  exclude_iea = FALSE,
  measure_underrepresentation = FALSE,
  evcodes = FALSE,
  user_threshold = 0.05,
  correction_method = c("g_SCS", "bonferroni", "fdr", "false_discovery_rate", "gSCS",
    "analytical"),
  domain_scope = c("annotated", "known", "custom", "custom_annotated"),
  custom_bg = NULL,
  numeric_ns = "",
  sources = NULL,
  as_short_link = FALSE,
  highlight = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gost_+3A_query">query</code></td>
<td>
<p>character vector, or a (named) list of character vectors for multiple queries, that can consist of mixed types of gene IDs (proteins, transcripts, microarray IDs, etc), SNP IDs, chromosomal intervals or term IDs.</p>
</td></tr>
<tr><td><code id="gost_+3A_organism">organism</code></td>
<td>
<p>organism name. Organism names are constructed by concatenating the first letter of the name and the
family name. Example: human - 'hsapiens', mouse - 'mmusculus'.</p>
</td></tr>
<tr><td><code id="gost_+3A_ordered_query">ordered_query</code></td>
<td>
<p>in case input gene lists are ranked this option may be
used to get GSEA style p-values.</p>
</td></tr>
<tr><td><code id="gost_+3A_multi_query">multi_query</code></td>
<td>
<p>in case of multiple gene lists, returns comparison table of these lists.
If enabled, the result data frame has columns named 'p_values', 'gconvert_sizes', 'intersection_sizes' with vectors showing values in the order of input queries.
Set 'multi_gconvert' to FALSE and simply input query as list of multiple gene vectors to get the results in a long format.</p>
</td></tr>
<tr><td><code id="gost_+3A_significant">significant</code></td>
<td>
<p>whether all or only statistically significant results should
be returned.</p>
</td></tr>
<tr><td><code id="gost_+3A_exclude_iea">exclude_iea</code></td>
<td>
<p>exclude GO electronic annotations (IEA).</p>
</td></tr>
<tr><td><code id="gost_+3A_measure_underrepresentation">measure_underrepresentation</code></td>
<td>
<p>measure underrepresentation.</p>
</td></tr>
<tr><td><code id="gost_+3A_evcodes">evcodes</code></td>
<td>
<p>include evidence codes to the results. Note
that this can decrease performance and make the query slower.
In addition, a column 'intersection' is created that contains the gene id-s that intersect between the query and term.
This parameter does not work if 'multi_query' is set to TRUE.</p>
</td></tr>
<tr><td><code id="gost_+3A_user_threshold">user_threshold</code></td>
<td>
<p>custom p-value threshold for significance, results with smaller p-value are tagged as significant. We don't recommend to set it higher than 0.05.</p>
</td></tr>
<tr><td><code id="gost_+3A_correction_method">correction_method</code></td>
<td>
<p>the algorithm used for multiple testing correction, one of &quot;gSCS&quot; (synonyms: &quot;analytical&quot;, &quot;g_SCS&quot;), &quot;fdr&quot; (synonyms: &quot;false_discovery_rate&quot;), &quot;bonferroni&quot;.</p>
</td></tr>
<tr><td><code id="gost_+3A_domain_scope">domain_scope</code></td>
<td>
<p>how to define statistical domain, one of &quot;annotated&quot;, &quot;known&quot;, &quot;custom&quot; or &quot;custom_annotated&quot;.</p>
</td></tr>
<tr><td><code id="gost_+3A_custom_bg">custom_bg</code></td>
<td>
<p>vector of gene names to use as a statistical background. If given, the domain_scope is by default set to &quot;custom&quot;, if domain_scope is set to &quot;custom_annotated&quot;, then this is used instead.</p>
</td></tr>
<tr><td><code id="gost_+3A_numeric_ns">numeric_ns</code></td>
<td>
<p>namespace to use for fully numeric IDs (<a href="https://biit.cs.ut.ee/gprofiler/page/namespaces-list">list of available namespaces</a>).</p>
</td></tr>
<tr><td><code id="gost_+3A_sources">sources</code></td>
<td>
<p>a vector of data sources to use. Currently, these include
GO (GO:BP, GO:MF, GO:CC to select a particular GO branch), KEGG, REAC, TF,
MIRNA, CORUM, HP, HPA, WP. Please see the g:GOSt web tool for the comprehensive
list and details on incorporated data sources.</p>
</td></tr>
<tr><td><code id="gost_+3A_as_short_link">as_short_link</code></td>
<td>
<p>indicator to return results as short-link to the g:Profiler web tool. If set to TRUE, then the function returns the results URL as a character string instead of the data.frame.</p>
</td></tr>
<tr><td><code id="gost_+3A_highlight">highlight</code></td>
<td>
<p>indicator to return a TRUE-FALSE column called 'highlighted' to indicate driver terms in GO.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input gene lists are not stored in g:Profiler unless the option 'as_short_link' is set to TRUE.
</p>


<h3>Value</h3>

<p>A named list where 'result' contains data.frame with the enrichment analysis results and 'meta' contains metadata needed for Manhattan plot. If the input
consisted of several lists the corresponding list is indicated with a variable
'query'. The 'result' data.frame is ordered first by the query name, data source (such as GO:BP, GO:CC, GO:MF, REAC, etc), and then by the adjusted p-value.
When requesting a 'multi_query', either TRUE or FALSE, the columns of the resulting data frame differ.
If 'evcodes' is set, the return value includes columns 'evidence_codes' and 'intersection'.
The latter conveys info about the intersecting genes between the corresponding query and term.
</p>
<p>The result fields are further described in the <a href="https://cran.r-project.org/package=gprofiler2/vignettes/gprofiler2.html">vignette</a>.
</p>
<p>If 'as_short_link' is set to TRUE, then the result is a character short-link to see and share corresponding results via the g:Profiler web tool. In this case, the input gene lists will be stored in a database.
</p>


<h3>Author(s)</h3>

<p>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;, Uku Raudvere &lt;uku.raudvere@ut.ee&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gostres &lt;- gost(c("X:1000:1000000", "rs17396340", "GO:0005005", "ENSG00000156103", "NLRP1"))

</code></pre>

<hr>
<h2 id='gostplot'>Manhattan plot of functional enrichment results.</h2><span id='topic+gostplot'></span>

<h3>Description</h3>

<p>This function creates a Manhattan plot out of the results from gprofiler2::gost().
The plot is very similar to the one shown in the g:GOSt web tool.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gostplot(
  gostres,
  capped = TRUE,
  interactive = TRUE,
  pal = c(`GO:MF` = "#dc3912", `GO:BP` = "#ff9900", `GO:CC` = "#109618", KEGG =
    "#dd4477", REAC = "#3366cc", WP = "#0099c6", TF = "#5574a6", MIRNA = "#22aa99", HPA =
    "#6633cc", CORUM = "#66aa00", HP = "#990099")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gostplot_+3A_gostres">gostres</code></td>
<td>
<p>named list from gost() function (with names 'result' and 'meta')</p>
</td></tr>
<tr><td><code id="gostplot_+3A_capped">capped</code></td>
<td>
<p>whether the -log10(p-values) would be capped if &gt;= 16, just as in the web options.</p>
</td></tr>
<tr><td><code id="gostplot_+3A_interactive">interactive</code></td>
<td>
<p>if enabled, returns interactive plot using 'plotly'. If disabled, static 'ggplot()' object is returned.</p>
</td></tr>
<tr><td><code id="gostplot_+3A_pal">pal</code></td>
<td>
<p>values mapped to relevant colors for data sources.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is either a plotly object (if interactive = TRUE) or a ggplot object (if interactive = FALSE).
</p>


<h3>Author(s)</h3>

<p>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'> gostres &lt;- gost(c("Klf4", "Pax5", "Sox2", "Nanog"), organism = "mmusculus")
 gostplot(gostres)
</code></pre>

<hr>
<h2 id='gsnpense'>Convert SNP rs identifiers to genes.</h2><span id='topic+gsnpense'></span>

<h3>Description</h3>

<p>Interface to the g:Profiler tool g:SNPense (<a href="https://biit.cs.ut.ee/gprofiler/snpense">https://biit.cs.ut.ee/gprofiler/snpense</a>) that maps SNP rs identifiers to chromosome positions, genes and variant effects.
Available only for human variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsnpense(query, filter_na = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gsnpense_+3A_query">query</code></td>
<td>
<p>vector of SNP IDs to be translated (should start with prefix 'rs').</p>
</td></tr>
<tr><td><code id="gsnpense_+3A_filter_na">filter_na</code></td>
<td>
<p>logical indicating whether to filter out results without a
corresponding target name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a data.frame which is a table closely corresponding to the
web interface output. Columns 'ensgs' and 'gene_names' can contain list of multiple values.
</p>
<p>The result fields are further described in the <a href="https://cran.r-project.org/package=gprofiler2/vignettes/gprofiler2.html">vignette</a>.
</p>


<h3>Author(s)</h3>

<p>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;, Uku Raudvere &lt;uku.raudvere@ut.ee&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gsnpense(c("rs11734132", "rs7961894", "rs4305276", "rs17396340", "rs3184504"))
</code></pre>

<hr>
<h2 id='mapViridis'>Map vector of numeric values to Viridis color scale.</h2><span id='topic+mapViridis'></span>

<h3>Description</h3>

<p>Map vector of numeric values to Viridis color scale.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapViridis(values, domain_min = 0, domain_max = 50, n = 256)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapViridis_+3A_values">values</code></td>
<td>
<p>vector of numeric values (mostly -log10(p-values))</p>
</td></tr>
<tr><td><code id="mapViridis_+3A_domain_min">domain_min</code></td>
<td>
<p>numeric value that corresponds to the 'yellow' in the color scale</p>
</td></tr>
<tr><td><code id="mapViridis_+3A_domain_max">domain_max</code></td>
<td>
<p>numeric value that corresponds to the 'dark blue' in the color scale</p>
</td></tr>
<tr><td><code id="mapViridis_+3A_n">n</code></td>
<td>
<p>number of bins to generate from the color scale</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a corresponding vector of colors from the Viridis color scale with domain in range(domain_min, domain_max).
</p>


<h3>Author(s)</h3>

<p>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;
</p>

<hr>
<h2 id='publish_gostplot'>Create and save an annotated Manhattan plot of enrichment results.</h2><span id='topic+publish_gostplot'></span>

<h3>Description</h3>

<p>This function allows to highlight a list of selected terms on the Manhattan plot created with the gprofiler2::gostplot() function.
The resulting plot is saved to a publication ready image if 'filename' is specified.
The plot is very similar to the one shown in the g:GOSt web tool after clicking on circles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>publish_gostplot(
  p,
  highlight_terms = NULL,
  filename = NULL,
  width = NA,
  height = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="publish_gostplot_+3A_p">p</code></td>
<td>
<p>ggplot object from gostplot(gostres, interactive = FALSE) function</p>
</td></tr>
<tr><td><code id="publish_gostplot_+3A_highlight_terms">highlight_terms</code></td>
<td>
<p>vector of selected term IDs from the analysis or a (subset) data.frame that has a column called 'term_id'. No annotation is added if set to NULL.</p>
</td></tr>
<tr><td><code id="publish_gostplot_+3A_filename">filename</code></td>
<td>
<p>file name to create on disk and save the annotated plot. Filename extension should be from c(&quot;png&quot;, &quot;pdf&quot;, &quot;jpeg&quot;, &quot;tiff&quot;, &quot;bmp&quot;).</p>
</td></tr>
<tr><td><code id="publish_gostplot_+3A_width">width</code></td>
<td>
<p>plot width in inches. If not supplied, the size of current graphics device is used.</p>
</td></tr>
<tr><td><code id="publish_gostplot_+3A_height">height</code></td>
<td>
<p>plot height in inches. If not supplied, the size of current graphics device is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a ggplot object.
</p>


<h3>Author(s)</h3>

<p>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'> gostres &lt;- gost(c("Klf4", "Pax5", "Sox2", "Nanog"), organism = "mmusculus")
 p &lt;- gostplot(gostres, interactive = FALSE)
 publish_gostplot(p, highlight_terms = c("GO:0001010", "REAC:R-MMU-8939245"))
</code></pre>

<hr>
<h2 id='publish_gosttable'>Create and save a table with the functional enrichment analysis results.</h2><span id='topic+publish_gosttable'></span>

<h3>Description</h3>

<p>This function creates a table mainly for the results from gost() function.
However, if the input at least contains columns named 'term_id' and 'p_value' then any enrichment results data frame can be visualised in a table with this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>publish_gosttable(
  gostres,
  highlight_terms = NULL,
  use_colors = TRUE,
  show_columns = c("source", "term_name", "term_size", "intersection_size"),
  filename = NULL,
  ggplot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="publish_gosttable_+3A_gostres">gostres</code></td>
<td>
<p>named list from gost() function (with names 'result' and 'meta') or a data frame that has columns named &quot;term_id&quot; and &quot;p_value(s)&quot;.</p>
</td></tr>
<tr><td><code id="publish_gosttable_+3A_highlight_terms">highlight_terms</code></td>
<td>
<p>vector of selected term IDs from the analysis or a (subset) data.frame that has a column called 'term_id'. All data is shown if set to NULL.</p>
</td></tr>
<tr><td><code id="publish_gosttable_+3A_use_colors">use_colors</code></td>
<td>
<p>if enabled, the p-values are highlighted in the viridis colorscale just as in g:Profiler, otherwise the table has no background colors.</p>
</td></tr>
<tr><td><code id="publish_gosttable_+3A_show_columns">show_columns</code></td>
<td>
<p>names of additional columns to show besides term_id and p_value. By default the output table shows additional columns named &quot;source&quot;, &quot;term_name&quot;, &quot;term_size&quot;, &quot;intersection_size&quot;</p>
</td></tr>
<tr><td><code id="publish_gosttable_+3A_filename">filename</code></td>
<td>
<p>file name to create on disk and save the annotated plot. Filename extension should be from c(&quot;png&quot;, &quot;pdf&quot;, &quot;jpeg&quot;, &quot;tiff&quot;, &quot;bmp&quot;).</p>
</td></tr>
<tr><td><code id="publish_gosttable_+3A_ggplot">ggplot</code></td>
<td>
<p>if FALSE, then the function returns a gtable object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output table is very similar to the one shown under the Manhattan plot.
</p>


<h3>Value</h3>

<p>The output is a ggplot object.
</p>


<h3>Author(s)</h3>

<p>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'> gostres &lt;- gost(c("Klf4", "Pax5", "Sox2", "Nanog"), organism = "mmusculus")
 publish_gosttable(gostres, highlight_terms = c("GO:0001010", "REAC:R-MMU-8939245"))
</code></pre>

<hr>
<h2 id='random_query'>Generate a random gene list for testing.</h2><span id='topic+random_query'></span>

<h3>Description</h3>

<p>This function returns a vector of randomly selected genes from the selected organism.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_query(organism = "hsapiens")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="random_query_+3A_organism">organism</code></td>
<td>
<p>organism name. Organism names are constructed by concatenating the first letter of the name and the
family name. Example: human - 'hsapiens', mouse - 'mmusculus'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector containing randomly selected gene IDs from the selected organism.
</p>


<h3>Author(s)</h3>

<p>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>random_genes &lt;- random_query()
</code></pre>

<hr>
<h2 id='set_base_url'>Set the base URL.</h2><span id='topic+set_base_url'></span>

<h3>Description</h3>

<p>Function to change the g:Profiler base URL. Useful for overriding the default URL (http://biit.cs.ut.ee/gprofiler)
with the beta (http://biit.cs.ut.ee/gprofiler_beta) or an archived version (available starting from the version e94_eg41_p11, e.g. http://biit.cs.ut.ee/gprofiler_archive3/e94_eg41_p11).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_base_url(url)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_base_url_+3A_url">url</code></td>
<td>
<p>the base URL.</p>
</td></tr>
</table>

<hr>
<h2 id='set_tls_version'>Set the TLS version to use for SSL</h2><span id='topic+set_tls_version'></span>

<h3>Description</h3>

<p>Set the TLS version. Could be useful at environments where SSL was built without TLS 1.2 support
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_tls_version(v)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_tls_version_+3A_v">v</code></td>
<td>
<p>version: &quot;1.2&quot; (default), &quot;1.1&quot; (fallback)</p>
</td></tr>
</table>

<hr>
<h2 id='set_user_agent'>Set custom user agent string.</h2><span id='topic+set_user_agent'></span>

<h3>Description</h3>

<p>Set the HTTP User-Agent string. Useful for overriding the default user agent for
packages that depend on gprofiler2 functionality.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_user_agent(ua, append = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_user_agent_+3A_ua">ua</code></td>
<td>
<p>the user agent string.</p>
</td></tr>
<tr><td><code id="set_user_agent_+3A_append">append</code></td>
<td>
<p>logical indicating whether to append the passed string to the default user agent string.</p>
</td></tr>
</table>

<hr>
<h2 id='upload_GMT_file'>Upload custom annotations for functional enrichment analysis in g:GOSt.</h2><span id='topic+upload_GMT_file'></span>

<h3>Description</h3>

<p>Upload your own annotation data using files in the Gene Matrix Transposed file format (GMT) for functional enrichment analysis in g:GOSt.
The accepted file is either a single annotations file (with the extension .gmt) or a compressed directory of multiple annotation GMT files (with the extension .zip).
The GMT format is a tab-separated list of gene annotation sets where every line represents a separate gene set/functional term. The first column defines the function ID, second defines a short name/description of the function and the following columns
are the list of genes related to the specific function in that row.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>upload_GMT_file(gmtfile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="upload_GMT_file_+3A_gmtfile">gmtfile</code></td>
<td>
<p>the filepath of the GMT file to be uploaded. The file extension should be .gmt or .zip in case of multiple GMT files.
If the filepath does not contain an absolute path, the filename is relative to the current working directory.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The uploaded filename is used to define 'source' name in the g:GOSt results.
</p>


<h3>Value</h3>

<p>A string that denotes the ID of the uploaded custom annotations in the g:Profiler database.
After the GMT file upload this unique ID can be used as a value for the argument 'organism' in the <code>gost()</code> function to perform
functional enrichment analysis based on these custom data.
</p>
<p>No need to repeatedly upload the same custom GMT file(s) every time you want to do the enrichment analysis.
The custom ID can also be used in the web tool as a token under the Custom GMT options.
</p>


<h3>Author(s)</h3>

<p>Liis Kolberg &lt;liis.kolberg@ut.ee&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: custom_id &lt;- upload_GMT_file("path/to/file.gmt")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
