<!DOCTYPE html><html lang="en"><head><title>Help for package gProfileR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gProfileR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gconvert'><p>Convert gene IDs.</p></a></li>
<li><a href='#get_base_url'><p>Get the base URL.</p></a></li>
<li><a href='#get_tls_version'><p>Get the TLS version for SSL</p></a></li>
<li><a href='#get_user_agent'><p>Get current user agent string.</p></a></li>
<li><a href='#gorth'><p>Find orthologs.</p></a></li>
<li><a href='#gprofiler'><p>Annotate gene list functionally.</p></a></li>
<li><a href='#set_base_url'><p>Set the base URL.</p></a></li>
<li><a href='#set_tls_version'><p>Set the TLS version to use for SSL</p></a></li>
<li><a href='#set_user_agent'><p>Set custom user agent string.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.7.0</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Description:</td>
<td>This package has been deprecated and will not be updated. 
    New users should use the package 'gprofiler2' (<a href="https://CRAN.R-project.org/package=gprofiler2">https://CRAN.R-project.org/package=gprofiler2</a>)
    for up-to-date data and improved functionality.
    Functional enrichment analysis, gene identifier conversion and
    mapping homologous genes across related organisms via the 'g:Profiler' toolkit
    (<a href="https://biit.cs.ut.ee/gprofiler/">https://biit.cs.ut.ee/gprofiler/</a>).</td>
</tr>
<tr>
<td>Title:</td>
<td>Interface to the 'g:Profiler' Toolkit</td>
</tr>
<tr>
<td>Author:</td>
<td>Juri Reimand &lt;juri.reimand@ut.ee&gt;, Raivo Kolde
    &lt;rkolde@gmail.com&gt;, Tambet Arak &lt;tambet.arak@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ivan Kuzmin &lt;ivan.kuzmin@ut.ee&gt;</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="http://biit.cs.ut.ee/gprofiler/contact.cgi">http://biit.cs.ut.ee/gprofiler/contact.cgi</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>RCurl, plyr, utils</td>
</tr>
<tr>
<td>Collate:</td>
<td>'gProfileR.R'</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-11-04 16:04:57 UTC; inkuzmin</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-11-04 16:30:17 UTC</td>
</tr>
</table>
<hr>
<h2 id='gconvert'>Convert gene IDs.</h2><span id='topic+gconvert'></span>

<h3>Description</h3>

<p>Interface to the g:Convert tool. Organism names are constructed by
concatenating the first letter of the name and the family name. Example: human
- 'hsapiens', mouse - 'mmusculus'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gconvert(query, organism = "hsapiens", target = "ENSG",
  region_query = F, numeric_ns = "", mthreshold = Inf,
  filter_na = T, df = T)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gconvert_+3A_query">query</code></td>
<td>
<p>list of gene IDs.</p>
</td></tr>
<tr><td><code id="gconvert_+3A_organism">organism</code></td>
<td>
<p>organism name.</p>
</td></tr>
<tr><td><code id="gconvert_+3A_target">target</code></td>
<td>
<p>target namespace.</p>
</td></tr>
<tr><td><code id="gconvert_+3A_region_query">region_query</code></td>
<td>
<p>interpret query as chromosomal ranges.</p>
</td></tr>
<tr><td><code id="gconvert_+3A_numeric_ns">numeric_ns</code></td>
<td>
<p>namespace to use for fully numeric IDs.</p>
</td></tr>
<tr><td><code id="gconvert_+3A_mthreshold">mthreshold</code></td>
<td>
<p>maximum number of results per initial alias to show.</p>
</td></tr>
<tr><td><code id="gconvert_+3A_filter_na">filter_na</code></td>
<td>
<p>logical indicating whether to filter out results without a
corresponding target.</p>
</td></tr>
<tr><td><code id="gconvert_+3A_df">df</code></td>
<td>
<p>logical indicating whether the output will be a data.frame or list.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output can be either a list or a data.frame. The list has an entry
for every input gene. The data frame is a table closely corresponding to the
web interface output.
</p>


<h3>Author(s)</h3>

<p>Juri Reimand &lt;jyri.reimand@ut.ee&gt;, Raivo Kolde &lt;rkolde@gmail.com&gt;,
Tambet Arak &lt;tambet.arak@gmail.com&gt;
</p>


<h3>References</h3>

<p>J. Reimand, M. Kull, H. Peterson, J.  Hansen, J. Vilo: g:Profiler
- a web-based toolset for functional profiling of gene lists from large-scale
experiments (2007) NAR 35 W193-W200
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 gconvert(c("POU5F1", "SOX2", "NANOG"), organism = "hsapiens", target="AFFY_HG_U133_PLUS_2")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_base_url'>Get the base URL.</h2><span id='topic+get_base_url'></span>

<h3>Description</h3>

<p>Get the base URL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_base_url()
</code></pre>

<hr>
<h2 id='get_tls_version'>Get the TLS version for SSL</h2><span id='topic+get_tls_version'></span>

<h3>Description</h3>

<p>Get the TLS version for SSL
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tls_version()
</code></pre>

<hr>
<h2 id='get_user_agent'>Get current user agent string.</h2><span id='topic+get_user_agent'></span>

<h3>Description</h3>

<p>Get the HTTP User-Agent string.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_user_agent()
</code></pre>

<hr>
<h2 id='gorth'>Find orthologs.</h2><span id='topic+gorth'></span>

<h3>Description</h3>

<p>Interface to the g:Orth tool. Organism names are constructed by concatenating
the first letter of the name and the family name. Example: human - 'hsapiens',
mouse - 'mmusculus'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gorth(query, source_organism = "hsapiens",
  target_organism = "mmusculus", region_query = F, numeric_ns = "",
  mthreshold = Inf, filter_na = T, df = T)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gorth_+3A_query">query</code></td>
<td>
<p>list of gene IDs to be translated.</p>
</td></tr>
<tr><td><code id="gorth_+3A_source_organism">source_organism</code></td>
<td>
<p>name of the source organism.</p>
</td></tr>
<tr><td><code id="gorth_+3A_target_organism">target_organism</code></td>
<td>
<p>name of the target organism.</p>
</td></tr>
<tr><td><code id="gorth_+3A_region_query">region_query</code></td>
<td>
<p>interpret query as chromosomal ranges.</p>
</td></tr>
<tr><td><code id="gorth_+3A_numeric_ns">numeric_ns</code></td>
<td>
<p>namespace to use for fully numeric IDs.</p>
</td></tr>
<tr><td><code id="gorth_+3A_mthreshold">mthreshold</code></td>
<td>
<p>maximum number of ortholog names per gene to show.</p>
</td></tr>
<tr><td><code id="gorth_+3A_filter_na">filter_na</code></td>
<td>
<p>logical indicating whether to filter out results without a
corresponding target name.</p>
</td></tr>
<tr><td><code id="gorth_+3A_df">df</code></td>
<td>
<p>logical indicating whether the output will be a data.frame or list.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To alleviate the problem of having many orthologs per gene (most of them
uninformative) one can set a threshold for the number of results. The program
tries to find the most informative by selecting the most popular ones.
</p>


<h3>Value</h3>

<p>The output can be either a list or a data.frame. The list has an entry
for every input gene. The data frame is a table closely corresponding to the
web interface output.
</p>


<h3>Author(s)</h3>

<p>Raivo Kolde &lt;rkolde@gmail.com&gt;, Juri Reimand &lt;juri.reimand@ut.ee&gt;,
Tambet Arak &lt;tambet.arak@gmail.com&gt;
</p>


<h3>References</h3>

<p>J. Reimand, M. Kull, H. Peterson, J.  Hansen, J. Vilo: g:Profiler
&ndash; a web-based toolset for functional profiling of gene lists from large-scale
experiments (2007) NAR 35 W193-W200
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 gorth(c("Klf4","Pax5","Sox2","Nanog"), source_organism="mmusculus", target_organism="hsapiens")

## End(Not run)
</code></pre>

<hr>
<h2 id='gprofiler'>Annotate gene list functionally.</h2><span id='topic+gprofiler'></span>

<h3>Description</h3>

<p>Interface to the g:Profiler tool for finding enrichments in gene lists. Organism
names are constructed by concatenating the first letter of the name and the
family name. Example: human - 'hsapiens', mouse - 'mmusculus'. If requesting PNG
output, the request is directed to the g:GOSt tool in case 'query' is a vector
and the g:Cocoa (compact view of multiple queries) tool in case 'query' is a
list. PNG output can fail (return FALSE) in case the input query is too large.
In such case, it is advisable to fall back to a non-image request.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gprofiler(query, organism = "hsapiens", sort_by_structure = T,
  ordered_query = F, significant = T, exclude_iea = F,
  underrep = F, evcodes = F, region_query = F, max_p_value = 1,
  min_set_size = 0, max_set_size = 0, min_isect_size = 0,
  correction_method = "analytical", hier_filtering = "none",
  domain_size = "annotated", custom_bg = "", numeric_ns = "",
  png_fn = NULL, include_graph = F, src_filter = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gprofiler_+3A_query">query</code></td>
<td>
<p>vector of gene IDs or a list of such vectors. In the latter case,
the query is directed to g:Cocoa, which yields a different graphical output 
if requested with the <code>png_fn</code> parameter.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_organism">organism</code></td>
<td>
<p>organism name.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_sort_by_structure">sort_by_structure</code></td>
<td>
<p>whether hierarchical sorting is enabled or disabled.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_ordered_query">ordered_query</code></td>
<td>
<p>in case output gene lists are ranked this option may be
used to get GSEA style p-values.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_significant">significant</code></td>
<td>
<p>whether all or only statistically significant results should
be returned.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_exclude_iea">exclude_iea</code></td>
<td>
<p>exclude electronic annotations (IEA).</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_underrep">underrep</code></td>
<td>
<p>measure underrepresentation.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_evcodes">evcodes</code></td>
<td>
<p>include GO evidence codes as the final column of output. Note 
that this can decrease performance and make the query slower.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_region_query">region_query</code></td>
<td>
<p>interpret query as chromosomal ranges.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_max_p_value">max_p_value</code></td>
<td>
<p>custom p-value threshold, results with a larger p-value are
excluded.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_min_set_size">min_set_size</code></td>
<td>
<p>minimum size of functional category, smaller categories are
excluded.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_max_set_size">max_set_size</code></td>
<td>
<p>maximum size of functional category, larger categories are
excluded.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_min_isect_size">min_isect_size</code></td>
<td>
<p>minimum size of the overlap (intersection) between query 
and functional category, smaller intersections are excluded.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_correction_method">correction_method</code></td>
<td>
<p>the algorithm used for determining the significance
threshold, one of &quot;gSCS&quot;, &quot;fdr&quot;, &quot;bonferroni&quot;.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_hier_filtering">hier_filtering</code></td>
<td>
<p>hierarchical filtering strength, one of &quot;none&quot;,
&quot;moderate&quot;, &quot;strong&quot;.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_domain_size">domain_size</code></td>
<td>
<p>statistical domain size, one of &quot;annotated&quot;, &quot;known&quot;.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_custom_bg">custom_bg</code></td>
<td>
<p>vector of gene names to use as a statistical background.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_numeric_ns">numeric_ns</code></td>
<td>
<p>namespace to use for fully numeric IDs.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_png_fn">png_fn</code></td>
<td>
<p>request the result as PNG image and write it to png_fn.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_include_graph">include_graph</code></td>
<td>
<p>request inclusion of network data with the result.</p>
</td></tr>
<tr><td><code id="gprofiler_+3A_src_filter">src_filter</code></td>
<td>
<p>a vector of data sources to use. Currently, these include
GO (GO:BP, GO:MF, GO:CC to select a particular GO branch), KEGG, REAC, TF,
MI, CORUM, HP, HPA, OMIM. Please see the g:GOSt web tool for the comprehensive 
list and details on incorporated data sources.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the enrichment analysis results. If the input
consisted of several lists the corresponding list is indicated with a variable
'query number'.  When requesting a PNG image, either TRUE or FALSE, depending on
whether a non-empty result was received and a file written or not, respectively.
If 'include_graph' is set, the return value may include the attribute 'networks',
containing a list of all network sources, each in turn containing a list of graph edges.
The edge structure is a list containing the two interacting symbols and two boolean
values (in that order), indicating whether the first or second interactor is part of
the input query (core nodes).
</p>


<h3>Author(s)</h3>

<p>Juri Reimand &lt;jyri.reimand@ut.ee&gt;, Raivo Kolde &lt;rkolde@gmail.com&gt;,
Tambet Arak &lt;tambet.arak@gmail.com&gt;
</p>


<h3>References</h3>

<p>J. Reimand, M. Kull, H. Peterson, J. Hansen, J. Vilo: g:Profiler -
a web-based toolset for functional profiling of gene lists from large-scale
experiments (2007) NAR 35 W193-W200
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 gprofiler(c("Klf4", "Pax5", "Sox2", "Nanog"), organism = "mmusculus")

## End(Not run)
</code></pre>

<hr>
<h2 id='set_base_url'>Set the base URL.</h2><span id='topic+set_base_url'></span>

<h3>Description</h3>

<p>Set the base URL. Useful for overriding the default URL (http://biit.cs.ut.ee/gprofiler)
with the bleeding-edge beta or an archived version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_base_url(url)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_base_url_+3A_url">url</code></td>
<td>
<p>the base URL.</p>
</td></tr>
</table>

<hr>
<h2 id='set_tls_version'>Set the TLS version to use for SSL</h2><span id='topic+set_tls_version'></span>

<h3>Description</h3>

<p>Set the TLS version. Could be useful at environments where SSL was built without TLS 1.2 support
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_tls_version(v)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_tls_version_+3A_v">v</code></td>
<td>
<p>version: &quot;1.2&quot; (default), &quot;1.1&quot; (fallback)</p>
</td></tr>
</table>

<hr>
<h2 id='set_user_agent'>Set custom user agent string.</h2><span id='topic+set_user_agent'></span>

<h3>Description</h3>

<p>Set the HTTP User-Agent string. Useful for overriding the default user agent for
packages that depend on gProfileR functionality.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_user_agent(ua, append = F)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_user_agent_+3A_ua">ua</code></td>
<td>
<p>the user agent string.</p>
</td></tr>
<tr><td><code id="set_user_agent_+3A_append">append</code></td>
<td>
<p>logical indicating whether to append the passed string to the default user agent string.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
