<!DOCTYPE html><html><head><title>Help for package gquad</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gquad}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aphased'><p>Predicting A-phased DNA repeat(s)</p></a></li>
<li><a href='#gquad'><p>Predicting G quadruplexes</p></a></li>
<li><a href='#gquadO'><p>Predicting G quadruplexes including overlaps</p></a></li>
<li><a href='#hdna'><p>Predicting intramolecular triplexes (H-DNA)</p></a></li>
<li><a href='#hdnaO'><p>Predicting intramolecular triplexes (H-DNA) including overlaps</p></a></li>
<li><a href='#slipped'><p>Predicting slipped motif(s)</p></a></li>
<li><a href='#str'><p>Predicting short tandem repeats</p></a></li>
<li><a href='#tfo'><p>Predicting triplex forming oligonucleotide(s)</p></a></li>
<li><a href='#zdna'><p>Predicting Z-DNA motif(s)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Prediction of G Quadruplexes and Other Non-B DNA Motifs</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1-2</td>
</tr>
<tr>
<td>Author:</td>
<td>Hannah O. Ajoge</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hannah O. Ajoge &lt;ohuajo@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Genomic biology is not limited to the confines of the canonical B-forming DNA duplex, but includes over ten different types of other secondary structures that are collectively termed non-B DNA structures. Of these non-B DNA structures, the G-quadruplexes are highly stable four-stranded structures that are recognized by distinct subsets of nuclear factors. This package provide functions for predicting intramolecular G quadruplexes. In addition, functions for predicting other intramolecular nonB DNA structures are included.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape (&ge; 5.6-2), seqinr (&ge; 4.2-23)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-29 08:18:01 UTC; Hannah.Polytomella</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-29 08:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='aphased'>Predicting A-phased DNA repeat(s)</h2><span id='topic+aphased'></span>

<h3>Description</h3>

<p>This function predicts A-phased DNA repeat(s)
in 'x' (DNA). DNA sequence can be provided in raw or fasta format or as GenBank accession number(s).
Internet is needed to connect to GenBank database, if accession number(s) is given as argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aphased(x, xformat = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aphased_+3A_x">x</code></td>
<td>
<p>DNA sequence(s) in raw format or a fasta file or a GenBank accession number(s); from which A-phased DNA repeat(s) will be predicted.
If the fasta file name does not contain an absolute path, the file name is relative to the current working directory.</p>
</td></tr>
<tr><td><code id="aphased_+3A_xformat">xformat</code></td>
<td>
<p>a character string specifying the format of x : default (raw), fasta, GenBank (GenBank accession number(s)).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function predicts A-phased DNA repeat(s) in DNA sequences and provide the position, sequence and length of the predicted repeat(s), if any.
</p>


<h3>Value</h3>

<p>A dataframe of A-phased DNA repeats' position, sequence and length. If more than one DNA sequence is provided as argument, an input ID is returned for repeat(s) predicted from each input sequence.
</p>


<h3>Author(s)</h3>

<p>Hannah O. Ajoge
</p>


<h3>References</h3>

<p>Paper on gquad and the web application (Non-B DNA Predictor) is under review, see draft in vignettes
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## Predicting A-phased DNA repeat(s) from raw DNA sequences
E1 &lt;- "TCTTGTTTTAAAACGTTTTAAAACGTTTTAAAACGTTTTAAAACGAAT"
aphased(E1)

## Predicting A-phased DNA repeat(s) from DNA sequences in fasta file
## Not run: aphased(x="Example.fasta", xformat = "fasta")

## Predicting A-phased DNA repeat(s) from DNA sequences,
## using GenBank accession numbers.
## Internet connectivity is needed for this to work.
## Not run: aphased(c("BH114913", "AY611035"), xformat = "GenBank")
</code></pre>

<hr>
<h2 id='gquad'>Predicting G quadruplexes</h2><span id='topic+gquad'></span>

<h3>Description</h3>

<p>This function predicts G quadruplexes
in 'x' (nucleotide sequence(s)).
Nucleotide sequence can be provided in raw or fasta format or as GenBank accession number(s).
Internet is needed to connect to GenBank database, if accession number(s) is given as argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gquad(x, xformat = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gquad_+3A_x">x</code></td>
<td>
<p>nucleotide sequence(s) in raw format or a fasta file or a GenBank accession number(s); from which G quadruplexes will be predicted.
If the fasta file name does not contain an absolute path, the file name is relative to the current working directory.</p>
</td></tr>
<tr><td><code id="gquad_+3A_xformat">xformat</code></td>
<td>
<p>a character string specifying the format of x : default (raw), fasta, GenBank (GenBank accession number(s)).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function predicts G quadruplexes in nucleic (both DNA and RNA) sequences and provide the position, sequence and length of the predicted motif(s). If any motif is predicted, the degree of likeliness for the motif to be formed is computed and scored as ** (more likely) or as * (less likely).
</p>


<h3>Value</h3>

<p>A dataframe of G quadruplexes' position, sequence, length and likeliness. If more than one nucleotide sequence is provided as argument, an input ID is returned for motif(s) predicted from each input sequence.
</p>


<h3>Author(s)</h3>

<p>Hannah O. Ajoge
</p>


<h3>References</h3>

<p>Paper on gquad and the web application (Non-B DNA Predictor) is under review, see draft in vignettes
</p>


<h3>See Also</h3>

<p>gquadO
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Predicting G quadruplexes from raw nucleotide sequences
E1 &lt;- c("TCTTGGGCATCTGGAGGCCGGAAT", "taggtgctgggaggtagagacaggatatcct")
gquad(E1)

## Predicting G quadruplexes from nucleotide sequences in fasta file
## Not run: gquad(x="Example.fasta", xformat = "fasta")

## Predicting G quadruplexes from nucleotide sequences,
## using GenBank accession numbers.
## Internet connectivity is needed for this to work.
## Not run: gquad(c("BH114913", "AY611035"), xformat = "GenBank")
</code></pre>

<hr>
<h2 id='gquadO'>Predicting G quadruplexes including overlaps</h2><span id='topic+gquadO'></span>

<h3>Description</h3>

<p>This function predicts G quadruplexes
in 'x' (nucleotide sequence(s)) like the gquad function, but includes overlaps.
Nucleotide sequence can be provided in raw or fasta format or as GenBank accession number(s).
Internet is needed to connect to GenBank database, if accession number(s) is given as argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gquadO(x, xformat = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gquadO_+3A_x">x</code></td>
<td>
<p>nucleotide sequence(s) in raw format or a fasta file or a GenBank accession number(s); from which G quadruplexes (including overlaps) will be predicted.
If the fasta file name does not contain an absolute path, the file name is relative to the current working directory.</p>
</td></tr>
<tr><td><code id="gquadO_+3A_xformat">xformat</code></td>
<td>
<p>a character string specifying the format of x : default (raw), fasta, GenBank (GenBank accession number(s)).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function predicts G quadruplexes in nucleic (both DNA and RNA) sequences, including overlaps and provide the position, sequence and length of the predicted motif(s). If any motif is predicted, the degree of likeliness for the motif to be formed is computed and scored as ** (more likely) or as * (less likely).
</p>


<h3>Value</h3>

<p>A dataframe of G quadruplexes' position, sequence, length and likeliness. If more than one nucleotide sequence is provided as argument, an input ID is returned for motif(s) predicted from each input sequence.
</p>


<h3>Author(s)</h3>

<p>Hannah O. Ajoge
</p>


<h3>References</h3>

<p>Paper on gquad and the web application (Non-B DNA Predictor) is under review, see draft in vignettes
</p>


<h3>See Also</h3>

<p>gquad
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Predicting G quadruplexes (including overlaps) from raw nucleotide sequences
E1 &lt;- c("TCTTGGGCATCTGGAGGCCGGAAT", "taggtgctgggaggtagagacaggatatcct")
gquadO(E1)

## Predicting G quadruplexes (including overlaps) from nucleotide sequences in fasta file
## Not run: gquadO(x="Example.fasta", xformat = "fasta")

## Predicting G quadruplexes (including overlaps) from nucleotide sequences,
## using GenBank accession numbers.
## Internet connectivity is needed for this to work.
## Not run: gquadO(c("BH114913", "AY611035"), xformat = "GenBank")
</code></pre>

<hr>
<h2 id='hdna'>Predicting intramolecular triplexes (H-DNA)</h2><span id='topic+hdna'></span>

<h3>Description</h3>

<p>This function predicts H-DNA
in 'x' (DNA). DNA can be provided in raw or fasta format or as GenBank accession number(s).
Internet is needed to connect to GenBank database, if accession number(s) is given as argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hdna(x, xformat = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hdna_+3A_x">x</code></td>
<td>
<p>DNA sequence(s) in raw format or a fasta file or a GenBank accession number(s); from which H-DNA will be predicted.
If the fasta file name does not contain an absolute path, the file name is relative to the current working directory.</p>
</td></tr>
<tr><td><code id="hdna_+3A_xformat">xformat</code></td>
<td>
<p>a character string specifying the format of x : default (raw), fasta, GenBank (GenBank accession number(s)).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function predicts H-DNA in DNA sequences and provide the position, sequence and length of the predicted motif(s), if any.
</p>


<h3>Value</h3>

<p>A dataframe of H-DNA' position, sequence and length. If more than one DNA sequence is provided as argument, an input ID is returned for motif(s) predicted from each input sequence.
</p>


<h3>Author(s)</h3>

<p>Hannah O. Ajoge
</p>


<h3>References</h3>

<p>Paper on gquad and the web application (Non-B DNA Predictor) is under review, see draft in vignettes
</p>


<h3>See Also</h3>

<p>hdnaO
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Predicting H-DNA from raw DNA sequences
E1 &lt;- c("TCTTCCCCCCTTTTTYYYYYGCTYYYYYTTTTTCCCCCCGAAT", "taggtgctgggaggtagagacaggatatcct")
hdna(E1)

## Predicting H-DNA from DNA sequences in fasta file
## Not run: hdna(x="Example.fasta", xformat = "fasta")

## Predicting H-DNA from DNA sequences,
## using GenBank accession numbers.
## Internet connectivity is needed for this to work.
## Not run: hdna(c("BH114913", "AY611035"), xformat = "GenBank")
</code></pre>

<hr>
<h2 id='hdnaO'>Predicting intramolecular triplexes (H-DNA) including overlaps</h2><span id='topic+hdnaO'></span>

<h3>Description</h3>

<p>This function predicts H-DNA
in 'x' DNA sequence like the hdna function, but includes overlaps.
DNA sequence can be provided in raw or fasta format or as GenBank accession number(s).
Internet is needed to connect to GenBank database, if accession number(s) is given as argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hdnaO(x, xformat = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hdnaO_+3A_x">x</code></td>
<td>
<p>DNA sequence(s) in raw format or a fasta file or a GenBank accession number(s); from which H-DNA (including overlaps) will be predicted.
If the fasta file name does not contain an absolute path, the file name is relative to the current working directory.</p>
</td></tr>
<tr><td><code id="hdnaO_+3A_xformat">xformat</code></td>
<td>
<p>a character string specifying the format of x : default (raw), fasta, GenBank (GenBank accession number(s)).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function predicts H-DNA in DNA sequences, including overlaps and provide the position, sequence and length of the predicted motif(s), if any.
</p>


<h3>Value</h3>

<p>A dataframe of H-DNA' position, sequence and length. If more than one DNA sequence is provided as argument, an input ID is returned for motif(s) predicted from each input sequence.
</p>


<h3>Author(s)</h3>

<p>Hannah O. Ajoge
</p>


<h3>References</h3>

<p>Paper on gquad and the web application (Non-B DNA Predictor) is under review, see draft in vignettes
</p>


<h3>See Also</h3>

<p>hdna
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Predicting H-DNA (including overlaps) from raw DNA sequences
E1 &lt;- c("TCTTCCCCCCTTTTTYYYYYGCTYYYYYTTTTTCCCCCCGAAT", "taggtgctgggaggtagagacaggatatcct")
hdnaO(E1)

## Predicting H-DNA (including overlaps) from DNA sequences in fasta file
## Not run: hdnaO(x="Example.fasta", xformat = "fasta")

## Predicting H-DNA (including overlaps) from DNA sequences,
## using GenBank accession numbers.
## Internet connectivity is needed for this to work.
## Not run: hdnaO(c("BH114913", "AY611035"), xformat = "GenBank")
</code></pre>

<hr>
<h2 id='slipped'>Predicting slipped motif(s)</h2><span id='topic+slipped'></span>

<h3>Description</h3>

<p>This function predicts slipped motif(s)
in 'x' in DNA. DNA sequence can be provided in raw or fasta format or as GenBank accession number(s).
Internet is needed to connect to GenBank database, if accession number(s) is given as argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slipped(x, xformat = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="slipped_+3A_x">x</code></td>
<td>
<p>DNA sequence(s) in raw format or a fasta file or a GenBank accession number(s); from which slipped motif(s) will be predicted.
If the fasta file name does not contain an absolute path, the file name is relative to the current working directory.</p>
</td></tr>
<tr><td><code id="slipped_+3A_xformat">xformat</code></td>
<td>
<p>a character string specifying the format of x : default (raw), fasta, GenBank (GenBank accession number(s)).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function predicts slipped motif(s) in DNA sequences and provide the position, sequence and length of the predicted motif(s). If any motif is predicted, the degree of likeliness for the motif to be formed is computed and scored as ** (more likely) or as * (less likely).
</p>


<h3>Value</h3>

<p>A dataframe of slipped motif(s) position, sequence, length and likeliness. If more than one DNA sequence is provided as argument, an input ID is returned for motif(s) predicted from each input sequence.
</p>


<h3>Author(s)</h3>

<p>Hannah O. Ajoge
</p>


<h3>References</h3>

<p>Paper on gquad and the web application (Non-B DNA Predictor) is under review, see draft in vignettes
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## Predicting slipped motif(s) from raw DNA sequences
E1 &lt;- c("TCTTACTGTGACTGTGGAAT", "taggtgctgggaggtagagacaggatatcct")
slipped(E1)

## Predicting slipped motif(s) from DNA sequences in fasta file
## Not run: slipped(x="Example.fasta", xformat = "fasta")

## Predicting slipped motif(s) from DNA sequences,
## using GenBank accession numbers.
## Internet connectivity is needed for this to work.
## Not run: slipped(c("BH114913", "AY611035"), xformat = "GenBank")
</code></pre>

<hr>
<h2 id='str'>Predicting short tandem repeats</h2><span id='topic+str'></span>

<h3>Description</h3>

<p>This function predicts short tandem repeats
in 'x' in nucleotides. Nucleotide sequence can be provided in raw or fasta format or as GenBank accession number(s).
Internet is needed to connect to GenBank database, if accession number(s) is given as argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>str(x, xformat = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="str_+3A_x">x</code></td>
<td>
<p>Nucleotide sequence(s) in raw format or a fasta file or a GenBank accession number(s); from which short tandem repeats will be predicted.
If the fasta file name does not contain an absolute path, the file name is relative to the current working directory.</p>
</td></tr>
<tr><td><code id="str_+3A_xformat">xformat</code></td>
<td>
<p>a character string specifying the format of x : default (raw), fasta, GenBank (GenBank accession number(s)).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function predicts short tandem repeats in nucleotide sequences and provide the position, sequence and length of the predicted repeats, if any.
</p>


<h3>Value</h3>

<p>A dataframe of short tandem repeats' position, sequence and length. If more than one DNA sequence is provided as argument, an input ID is returned for repeats predicted from each input sequence.
</p>


<h3>Author(s)</h3>

<p>Hannah O. Ajoge
</p>


<h3>References</h3>

<p>Paper on gquad and the web application (Non-B DNA Predictor) is under review, see draft in vignettes
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## Predicting short tandem repeats from raw nucleotide sequences
E1 &lt;- c("TCTACACACACACACACACACGAAT", "tagggugugugugugugugugugutcct")
str(E1)

## Predicting short tandem repeats from nucleotide sequences in fasta file
## Not run: str(x="Example.fasta", xformat = "fasta")

## Predicting short tandem repeats from nucleotide sequences,
## using GenBank accession numbers.
## Internet connectivity is needed for this to work.
## Not run: str(c("BH114913", "AY611035"), xformat = "GenBank")
</code></pre>

<hr>
<h2 id='tfo'>Predicting triplex forming oligonucleotide(s)</h2><span id='topic+tfo'></span>

<h3>Description</h3>

<p>This function predicts triplex forming oligonucleotide(s)
in 'x' in DNA. DNA sequence can be provided in raw or fasta format or as GenBank accession number(s).
Internet is needed to connect to GenBank database, if accession number(s) is given as argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tfo(x, xformat = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tfo_+3A_x">x</code></td>
<td>
<p>DNA sequence(s) in raw format or a fasta file or a GenBank accession number(s); from which triplex forming oligonucleotide(s) will be predicted.
If the fasta file name does not contain an absolute path, the file name is relative to the current working directory.</p>
</td></tr>
<tr><td><code id="tfo_+3A_xformat">xformat</code></td>
<td>
<p>a character string specifying the format of x : default (raw), fasta, GenBank (GenBank accession number(s)).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function predicts triplex forming oligonucleotide(s) in DNA sequences and provide the position, sequence and length of the predicted motif(s), if any.
</p>


<h3>Value</h3>

<p>A dataframe of triplex forming oligonucleotide(s) position, sequence and length. If more than one DNA sequence is provided as argument, an input ID is returned for motif(s) predicted from each input sequence.
</p>


<h3>Author(s)</h3>

<p>Hannah O. Ajoge
</p>


<h3>References</h3>

<p>Paper on gquad and the web application (Non-B DNA Predictor) is under review, see draft in vignettes
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## Predicting triplex forming oligonucleotide(s) from raw DNA sequences
E1 &lt;- c("TCTTGGGAGGGAGAGAGAGAAAGAGATCTGGAGGCCGGAAT", "taggtgctgggaggtagagacaggatatcct")
tfo(E1)

## Predicting triplex forming oligonucleotide(s) from DNA sequences in fasta file
## Not run: tfo(x="Example.fasta", xformat = "fasta")

## Predicting triplex forming oligonucleotide(s) from DNA sequences,
## using GenBank accession numbers.
## Internet connectivity is needed for this to work.
## Not run: tfo(c("BH114913", "AY611035"), xformat = "GenBank")
</code></pre>

<hr>
<h2 id='zdna'>Predicting Z-DNA motif(s)</h2><span id='topic+zdna'></span>

<h3>Description</h3>

<p>This function predicts Z-DNA motif(s)
in 'x' in DNA. DNA sequence can be provided in raw or fasta format or as GenBank accession number(s).
Internet is needed to connect to GenBank database, if accession number(s) is given as argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zdna(x, xformat = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zdna_+3A_x">x</code></td>
<td>
<p>DNA sequence(s) in raw format or a fasta file or a GenBank accession number(s); from which Z-DNA motif(s) will be predicted.
If the fasta file name does not contain an absolute path, the file name is relative to the current working directory.</p>
</td></tr>
<tr><td><code id="zdna_+3A_xformat">xformat</code></td>
<td>
<p>a character string specifying the format of x : default (raw), fasta, GenBank (GenBank accession number(s)).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function predicts Z-DNA motif(s) in DNA sequences and provide the position, sequence and length of the predicted motif(s). If any motif is predicted, the degree of likeliness for the motif to be formed is computed and scored as ** (more likely) or as * (less likely).
</p>


<h3>Value</h3>

<p>A dataframe of Z-DNA motif(s) position, sequence, length and likeliness. If more than one DNA sequence is provided as argument, an input ID is returned for motif(s) predicted from each input sequence.
</p>


<h3>Author(s)</h3>

<p>Hannah O. Ajoge
</p>


<h3>References</h3>

<p>Paper on gquad and the web application (Non-B DNA Predictor) is under review, see draft in vignettes
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## Predicting Z-DNA motif(s) from raw DNA sequences
E1 &lt;- c("TCTTGCGCGCGCGCGCGCGCGCGCGCAAT", "taggtgctgggaggtagagacaggatatcct")
zdna(E1)

## Predicting Z-DNA motif(s) from DNA sequences in fasta file
## Not run: zdna(x="Example.fasta", xformat = "fasta")

## Predicting Z-DNA motif(s) from DNA sequences,
## using GenBank accession numbers.
## Internet connectivity is needed for this to work.
## Not run: zdna(c("BH114913", "AY611035"), xformat = "GenBank")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
