<!DOCTYPE html><html lang="en-GB"><head><title>Help for package grafify</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {grafify}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#data_1w_death'><p>In vitro experiments measuring percentage cell death in three genotypes of cells.</p></a></li>
<li><a href='#data_2w_Festing'><p>Data from two-way ANOVA with randomised block design of treatments of strains of mice.</p></a></li>
<li><a href='#data_2w_Tdeath'><p>In vitro measurement of percentage cell death - two-way ANOVA design with repeated measures, and randomised blocks.</p></a></li>
<li><a href='#data_cholesterol'><p>Hierarchical data from 25 subjects either treated or not at 5 hospitals - two-way ANOVA design with repeated measures.</p></a></li>
<li><a href='#data_doubling_time'><p>Doubling time of E.coli measured by 10 students three independent times.</p></a></li>
<li><a href='#data_t_pdiff'><p>Matched data from two groups where difference between them is consistent.</p></a></li>
<li><a href='#data_t_pratio'><p>Matched data from two groups where ratio between them is consistent.</p></a></li>
<li><a href='#data_zooplankton'><p>Time-series data on zooplankton in lake Menon.</p></a></li>
<li><a href='#ga_anova'><p>ANOVA table from a generalised additive model (<code>gam</code>)</p></a></li>
<li><a href='#ga_model'><p>Fit a generalised additive model (<code>gam</code>)</p></a></li>
<li><a href='#get_graf_colours'><p>Get graf internal</p></a></li>
<li><a href='#graf_col_palette'><p>Call <code>grafify</code> palettes for scale &amp; fill functions</p></a></li>
<li><a href='#graf_col_palette_default'><p>Call <code>grafify</code> palettes for scale &amp; fill functions</p></a></li>
<li><a href='#graf_colours'><p>List of hexcodes of colours in grafify palettes</p></a></li>
<li><a href='#graf_palettes'><p>List of palettes available in grafify package</p></a></li>
<li><a href='#make_1way_data'><p>Make one-way or two-way independent group or randomised block design data.</p></a></li>
<li><a href='#make_1way_rb_data'><p>Make one-way or two-way independent group or randomised block design data.</p></a></li>
<li><a href='#make_2way_data'><p>Make one-way or two-way independent group or randomised block design data.</p></a></li>
<li><a href='#make_2way_rb_data'><p>Make one-way or two-way independent group or randomised block design data.</p></a></li>
<li><a href='#mixed_anova'><p>ANOVA table from linear mixed effects analysis.</p></a></li>
<li><a href='#mixed_anova_slopes'><p>ANOVA table from linear mixed effects analysis.</p></a></li>
<li><a href='#mixed_model'><p>Model from a linear mixed effects model</p></a></li>
<li><a href='#mixed_model_slopes'><p>Model from a linear mixed effects model with varying slopes</p></a></li>
<li><a href='#plot_3d_point_sd'><p>Plot of mean &amp; error bars for 1-way ANOVAs with matched shapes mapped to blocking factor.</p></a></li>
<li><a href='#plot_3d_scatterbar'><p>Plot a bar graph for 1-way ANOVAs with matched shapes mapped to blocking factor.</p></a></li>
<li><a href='#plot_3d_scatterbox'><p>Plot a scatter and box plot for 1-way ANOVAs with matched shapes mapped to blocking factor.</p></a></li>
<li><a href='#plot_3d_scatterviolin'><p>Plot a scatter with violin &amp; box plot for 1-way ANOVAs with matched shapes mapped to blocking factor.</p></a></li>
<li><a href='#plot_4d_point_sd'><p>Plot mean &amp; error bars for 2-way ANOVAs with or without a blocking factor.</p></a></li>
<li><a href='#plot_4d_scatterbar'><p>Plot scatter plot with bar &amp; error bars for 2-way ANOVAs with or without a blocking factor.</p></a></li>
<li><a href='#plot_4d_scatterbox'><p>Plot scatter, box &amp; whiskers for 2-way ANOVAs with or without a blocking factor.</p></a></li>
<li><a href='#plot_4d_scatterviolin'><p>Plot scatter, box &amp; violin for 2-way ANOVAs with or without a blocking factor.</p></a></li>
<li><a href='#plot_befafter_box'><p>Before-after style graph with a boxplot</p></a></li>
<li><a href='#plot_befafter_colours'><p>Plot a before-after plot with lines joining colour-matched symbols.</p></a></li>
<li><a href='#plot_befafter_shapes'><p>Plot a before-after plot with lines joining shape-matched symbols.</p></a></li>
<li><a href='#plot_density'><p>Plot density distribution of data.</p></a></li>
<li><a href='#plot_dotbar_sd'><p>Plot a dotplot on a bar graph with SD error bars with two variables.</p></a></li>
<li><a href='#plot_dotbox'><p>Plot a dotplot on a boxplot with two variables.</p></a></li>
<li><a href='#plot_dotviolin'><p>Plot a dotplot on a violin plot with two variables.</p></a></li>
<li><a href='#plot_gam_predict'><p>Plot prediction of <code>gam</code> model</p></a></li>
<li><a href='#plot_grafify_palette'><p>See grafify colour palettes</p></a></li>
<li><a href='#plot_histogram'><p>Plot data distribution as histograms.</p></a></li>
<li><a href='#plot_lm_predict'><p>Plot data and predictions from linear model</p></a></li>
<li><a href='#plot_logscale'><p>Add log transformations to graphs</p></a></li>
<li><a href='#plot_point_sd'><p>Plot a point as mean with SD error bars using two variables.</p></a></li>
<li><a href='#plot_qq_gam'><p>Plot model diagnostics for generalised additive models</p></a></li>
<li><a href='#plot_qqline'><p>Plot quantile-quantile (QQ) graphs from data.</p></a></li>
<li><a href='#plot_qqmodel'><p>Plot quantile-quantile (QQ) graphs from residuals of linear models.</p></a></li>
<li><a href='#plot_scatterbar_sd'><p>Plot scatter dots on a bar graph with SD error bars with two variables.</p></a></li>
<li><a href='#plot_scatterbox'><p>Plot a scatter plot on a boxplot with two variables.</p></a></li>
<li><a href='#plot_scatterviolin'><p>Plot a scatter plot on a violin plot with two variables.</p></a></li>
<li><a href='#plot_xy_CatGroup'><p>Plot points on a quantitative X - Y plot &amp; a categorical grouping variable.</p></a></li>
<li><a href='#plot_xy_NumGroup'><p>Plot points on a quantitative X - Y plot &amp; a numeric grouping variable.</p></a></li>
<li><a href='#posthoc_Levelwise'><p>Level-wise post-hoc comparisons from a linear or linear mixed effects model.</p></a></li>
<li><a href='#posthoc_Pairwise'><p>Pairwise post-hoc comparisons from a linear or linear mixed effects model.</p></a></li>
<li><a href='#posthoc_Trends_Levelwise'><p>Use emtrends to get level-wise comparison of slopes from a linear model.</p></a></li>
<li><a href='#posthoc_Trends_Pairwise'><p>Use emtrends to get pairwise comparison of slopes from a linear model.</p></a></li>
<li><a href='#posthoc_Trends_vsRef'><p>Use emtrends to get level-wise comparison of slopes from a linear model.</p></a></li>
<li><a href='#posthoc_vsRef'><p>Post-hoc comparisons to a control or reference group.</p></a></li>
<li><a href='#scale_colour_grafify'><p><code>scale_colour_</code> and <code>scale_fill_</code> functions</p></a></li>
<li><a href='#scale_fill_grafify'><p><code>scale_colour_</code> and <code>scale_fill_</code> functions</p></a></li>
<li><a href='#simple_anova'><p>ANOVA table from a linear model fit to data.</p></a></li>
<li><a href='#simple_model'><p>Model from a linear model fit to data.</p></a></li>
<li><a href='#table_summary'><p>Get numeric summary grouped by factors</p></a></li>
<li><a href='#table_x_reorder'><p>Reordering groups along X-axis</p></a></li>
<li><a href='#theme_grafify'><p>A modified <code>theme_classic()</code> for <code>grafify</code>-like graphs.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Easy Graphs for Data Visualisation and Linear Models for ANOVA</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-03-10</td>
</tr>
<tr>
<td>Version:</td>
<td>5.0.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Easily explore data by plotting graphs with a few lines of code. Use these ggplot() wrappers to quickly draw graphs of scatter/dots with box-whiskers, violins or SD error bars, data distributions, before-after graphs, factorial ANOVA and more. Customise graphs in many ways, for example, by choosing from colour blind-friendly palettes (12 discreet, 3 continuous and 2 divergent palettes). Use the simple code for ANOVA as ordinary (lm()) or mixed-effects linear models (lmer()), including randomised-block or repeated-measures designs, and fit non-linear outcomes as a generalised additive model (gam) using mgcv(). Obtain estimated marginal means and perform post-hoc comparisons on fitted models (via emmeans()). Also includes small datasets for practising code and teaching basics before users move on to more complex designs. See vignettes for details on usage <a href="https://grafify.shenoylab.com/">https://grafify.shenoylab.com/</a>. Citation: &lt;<a href="https://doi.org/10.5281%2Fzenodo.5136508">doi:10.5281/zenodo.5136508</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>car, dplyr, emmeans, Hmisc, lme4, lmerTest, magrittr, mgcv,
patchwork, purrr, stats, tidyr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0), ggplot2 (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, Matrix (&ge; 1.6-5), rlang, rmarkdown, pbkrtest,
testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ashenoy-cmbi/grafify">https://github.com/ashenoy-cmbi/grafify</a></td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-10 20:35:59 UTC; avish</td>
</tr>
<tr>
<td>Author:</td>
<td>Avinash R Shenoy <a href="https://orcid.org/0000-0001-6228-9303"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Avinash R Shenoy &lt;a.shenoy@imperial.ac.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-10 22:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='data_1w_death'>In vitro experiments measuring percentage cell death in three genotypes of cells.</h2><span id='topic+data_1w_death'></span>

<h3>Description</h3>

<p>These data are from in vitro measurements of death of host cells (measured as percentage of total cells) after infection with three different strains of a pathogenic bacterium, from five independent experiments. The three strains are three levels within the fixed factor Genotype. The five independent experiments are levels within the random variable Experiment. These data can be analysed using linear mixed effects modelling. These data are from Goddard <em>et al</em>, Cell Rep, 2019 (<a href="https://doi.org/10.1016/j.celrep.2019.03.100">doi:10.1016/j.celrep.2019.03.100</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_1w_death
</code></pre>


<h3>Format</h3>

<p>data.frame: 15 obs. of  3 variables.
</p>

<dl>
<dt>Experiment</dt><dd><p>Experiment - a random factor with 5 levels &quot;Exp_1&quot;,&quot;Exp_2&quot;...</p>
</dd>
<dt>Genotype</dt><dd><p>Genotypes - a fixed factor with 3 levels: &quot;WT&quot;,&quot;KO_1&quot;,&quot;KO_2&quot;.</p>
</dd>
<dt>Death</dt><dd><p>Numerical dependent variable indicating percentage cell death.</p>
</dd>
</dl>


<hr>
<h2 id='data_2w_Festing'>Data from two-way ANOVA with randomised block design of treatments of strains of mice.</h2><span id='topic+data_2w_Festing'></span>

<h3>Description</h3>

<p>Data from Festing, ILAR Journal (2014) 55, 472&ndash;476 (<a href="https://doi.org/10.1093/ilar/ilu045">doi:10.1093/ilar/ilu045</a>). These data are suitable for two-way linear mixed effects modelling.
The activity of GST (numerical dependent variable) was measured in 4 strains of mice (levels with the fixed factor Strain) either treated or controls (levels within the fixed factor Treatment). Once mouse each was used in two randomised blocks, which is the random factor (Block).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_2w_Festing
</code></pre>


<h3>Format</h3>

<p>data.frame:	16 obs. of  4 variables:
</p>

<dl>
<dt>Block</dt><dd><p>A random factor with 2 levels &quot;A&quot; and &quot;B&quot;.</p>
</dd>
<dt>Treatment</dt><dd><p>A fixed factor with 2 levels: &quot;Control&quot; &amp; &quot;Treated&quot;</p>
</dd>
<dt>Strain</dt><dd><p>A fixed factor with 4 levels: &quot;129Ola&quot;, &quot;A/J&quot;, &quot;NIH&quot; &amp; &quot;BALB/C&quot;</p>
</dd>
<dt>GST</dt><dd><p>Numerical dependent variable indicating GST activity measurement</p>
</dd>
</dl>


<hr>
<h2 id='data_2w_Tdeath'>In vitro measurement of percentage cell death - two-way ANOVA design with repeated measures, and randomised blocks.</h2><span id='topic+data_2w_Tdeath'></span>

<h3>Description</h3>

<p>These are measurements of death of infected host cells (as percentage of total cells) upon infection with two strains of bacteria, measured at two time points, in 6 independent experiments.  These data repeated-measures data suitable for two-way linear mixed effects modelling with experiment and subjects as random factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_2w_Tdeath
</code></pre>


<h3>Format</h3>

<p>data.frame:	24 obs. of  6 variables:
</p>

<dl>
<dt>Experiment</dt><dd><p>A random factor with 6 levels &quot;e1&quot;, &quot;e2&quot;...</p>
</dd>
<dt>Time</dt><dd><p>A fixed factor with 2 levels: &quot;t100&quot; &amp; &quot;t300&quot;.</p>
</dd>
<dt>Time2</dt><dd><p>A numeric column that allows plotting data on a quantitative &quot;Time&quot; axis. The &quot;Time&quot; column has &quot;factor&quot; type values that should be used for the ANOVA..</p>
</dd>
<dt>Genotype</dt><dd><p>A fixed factor with 2 levels that we want to compare &quot;WT&quot; &amp; &quot;KO&quot;.</p>
</dd>
<dt>Subject</dt><dd><p>A random factor with 12 levels: &quot;s1&quot;, &quot;s2&quot;... These are cell culture wells that were measured at two time points, and indicate &quot;subjects&quot; that underwent repeated-measures within each of 6 experiments. Subject IDs for WT and KO are unique and clearly indicate different wells.</p>
</dd>
<dt>PI</dt><dd><p>Numerical dependent variable indicating propidium iodide dye uptake as a measure of cell death. These are percentage of dead cells out of total cells plated.</p>
</dd>
</dl>


<hr>
<h2 id='data_cholesterol'>Hierarchical data from 25 subjects either treated or not at 5 hospitals - two-way ANOVA design with repeated measures.</h2><span id='topic+data_cholesterol'></span>

<h3>Description</h3>

<p>An example dataset on measurements of blood cholesterol levels measured in 5 subjects measured before and after receiving a Drug. Five patients each were recruited at 5 hospitals (a-e), so that there are 25 different subjects (1-25) measured twice.  Data are from <a href="https://biostats.shenoylab.com/">Micro/Immuno Stats</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_cholesterol
</code></pre>


<h3>Format</h3>

<p>tibble:	30 obs. of  3 variables:
</p>

<dl>
<dt>Hospital</dt><dd><p>Factor with 5 levels (a-e), representing different hospitals where subjects were recruited.</p>
</dd>
<dt>Subject</dt><dd><p>A factor with 25 levels denoting individuals on whom measurements were made twice.</p>
</dd>
<dt>Treatment</dt><dd><p>A factor with 2 levels indicating when measurements were made, i.e. before and after drug.</p>
</dd>
<dt>Cholesterol</dt><dd><p>Numerical dependent variable indicating measured doubling time in min.</p>
</dd>
</dl>


<hr>
<h2 id='data_doubling_time'>Doubling time of E.coli measured by 10 students three independent times.</h2><span id='topic+data_doubling_time'></span>

<h3>Description</h3>

<p>An example dataset showing measurements of <em>E. coli</em> doubling times (in min) measured by 10 different students in 3 independent experiments each. Note that Experiments are just called Exp1-Exp3 even though Exp1 of any of the students are not connected in anyway - this will confuse R! Data are from <a href="https://biostats.shenoylab.com/">Micro/Immuno Stats</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_doubling_time
</code></pre>


<h3>Format</h3>

<p>tibble:	30 obs. of  3 variables:
</p>

<dl>
<dt>Student</dt><dd><p>Factor with 10 levels, representing different students.</p>
</dd>
<dt>Experiment</dt><dd><p>A factor with 3 levels representing independent experiments.</p>
</dd>
<dt>Doubling_time</dt><dd><p>Numerical dependent variable indicating measured doubling time in min.</p>
</dd>
</dl>


<hr>
<h2 id='data_t_pdiff'>Matched data from two groups where difference between them is consistent.</h2><span id='topic+data_t_pdiff'></span>

<h3>Description</h3>

<p>An example dataset for paired difference Student's <em>t</em> test. These are bodyweight (Mass) in grams of same mice left untreated or treated, which are two groups to be compared. The data are in a longtable format, and the two groups are levels within the factor &quot;Condition&quot;. The Subject column lists ID of matched mice that were measured without and with treatment. These data are from Sanchez-Garrido <em>et al</em>, Sci Signal, 2018 (<a href="https://doi.org/10.1126/scisignal.aat6903">doi:10.1126/scisignal.aat6903</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_t_pdiff
</code></pre>


<h3>Format</h3>

<p>data.frame:	20 obs. of  3 variables:
</p>

<dl>
<dt>Subject</dt><dd><p>Factor with 10 levels, denoted by capital letters, representing individuals or subjects.</p>
</dd>
<dt>Condition</dt><dd><p>A fixed factor with 2 levels: &quot;Untreated&quot; &amp; &quot;Treated&quot;.</p>
</dd>
<dt>Mass</dt><dd><p>Numerical dependent variable indicating body mass of mice</p>
</dd>
</dl>


<hr>
<h2 id='data_t_pratio'>Matched data from two groups where ratio between them is consistent.</h2><span id='topic+data_t_pratio'></span>

<h3>Description</h3>

<p>An example dataset for paired ratio Student's <em>t</em> test. These are Cytokine measurements by ELISA (in ng/ml) from 33 independent in vitro experiments performed on two Genotypes that we want to compare. The data are in a longtable format, and the two groups are levels within the factor &quot;Genotype&quot;. The Experiment column lists ID of matched experiments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_t_pratio
</code></pre>


<h3>Format</h3>

<p>data.frame:	66 obs. of  3 variables:
</p>

<dl>
<dt>Genotype</dt><dd><p>Factor with 2 levels, representing genotypes to be compared (&quot;WT&quot; &amp; &quot;KO&quot;).</p>
</dd>
<dt>Experiment</dt><dd><p>A random factor with 33 levels representing independent experiments, denoted as &quot;Exp_1&quot;, &quot;Exp_2&quot;...</p>
</dd>
<dt>Cytokine</dt><dd><p>Numerical dependent variable indicating cytokine measured by ELISA.</p>
</dd>
</dl>


<hr>
<h2 id='data_zooplankton'>Time-series data on zooplankton in lake Menon.</h2><span id='topic+data_zooplankton'></span>

<h3>Description</h3>

<p>A subset of data from (Lathro RC, 2000) (<a href="https://doi.org/10.6073/pasta/ec3d0186753985147d4f283252388e05">doi:10.6073/pasta/ec3d0186753985147d4f283252388e05</a>) provided by the Wisconsin Department of Natural Resources
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_zooplankton
</code></pre>


<h3>Format</h3>

<p>tibble:	1127 obs. of  8 variables:
</p>

<dl>
<dt>day</dt><dd><p>Numeric integer variable.</p>
</dd>
<dt>year</dt><dd><p>Numeric integer variable of years during which data were collected.</p>
</dd>
<dt>lake</dt><dd><p>This data is for lake Menon; data for other others not included in this subset.</p>
</dd>
<dt>taxon</dt><dd><p>Names of zooplankton taxa as factor of 8 levels.</p>
</dd>
<dt>density</dt><dd><p>Numeric values of density of measurements.</p>
</dd>
<dt>density_adj</dt><dd><p>Numeric values of adjusted density .</p>
</dd>
<dt>min_density</dt><dd><p>Numeric values of minimum densities.</p>
</dd>
<dt>desnsity_scaled</dt><dd><p>Numeric value of scaled density.</p>
</dd>
</dl>


<hr>
<h2 id='ga_anova'>ANOVA table from a generalised additive model (<code>gam</code>)</h2><span id='topic+ga_anova'></span>

<h3>Description</h3>

<p>One of two functions for fitting generalised additive models (<code>gam</code>) with the <a href="https://CRAN.R-project.org/package=mgcv"><code>mgcv</code> package</a>. It will use the <code>gam()</code> function in <code>mgcv</code> for ANOVA designs with <strong>up to two categorical fixed factors</strong> (with two or more levels; <code>Fixed_Factor</code>), and <strong>exactly one factor is a continuous variable</strong> (e.g. time), which is called <code>Smooth_Factor</code>.
</p>

<ol>
<li> <p><code><a href="#topic+ga_model">ga_model</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ga_anova">ga_anova</a></code>
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>ga_anova(
  data,
  Y_value,
  Fixed_Factor,
  Smooth_Factor,
  Random_Factor = NULL,
  Nodes = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ga_anova_+3A_data">data</code></td>
<td>
<p>a data frame where categorical independent variables are converted to factors using <code>as.factor()</code> first. The function will throw errors without this.</p>
</td></tr>
<tr><td><code id="ga_anova_+3A_y_value">Y_value</code></td>
<td>
<p>name of column containing quantitative (dependent) variable, provided within &quot;quotes&quot;.</p>
</td></tr>
<tr><td><code id="ga_anova_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>name(s) of categorical fixed factors (independent variables) provided within quotes (e.g., &quot;A&quot;) or as a vector if more than one (e.g., c(&quot;A&quot;, &quot;B&quot;). Convert to factors first with <code>as.factor</code>.</p>
</td></tr>
<tr><td><code id="ga_anova_+3A_smooth_factor">Smooth_Factor</code></td>
<td>
<p>the continuous variable to fit smoothly with a basis function, provided within &quot;quotes&quot; (only 1 Smooth_Factor allowed).</p>
</td></tr>
<tr><td><code id="ga_anova_+3A_random_factor">Random_Factor</code></td>
<td>
<p>name(s) of random factors to be provided in &quot;quotes&quot; (only 1 Random_Factor allowed). Convert to factor with <code>as.factor</code> first.</p>
</td></tr>
<tr><td><code id="ga_anova_+3A_nodes">Nodes</code></td>
<td>
<p>number of nodes (the parameter <code>k</code> in <code>gam</code>).</p>
</td></tr>
<tr><td><code id="ga_anova_+3A_...">...</code></td>
<td>
<p>any additional variables to pass on to <code>gam</code> or <code>anova</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>A smooth function is fitted with factor-wise smooth basis function (<code style="white-space: pre;">&#8288;by = &#8288;</code>). A default value for number of nodes (the argument <code>k</code> in <code>gam</code>) may work, but a specific number can be provided using the <code>Nodes</code> argument. The model is fit using the <code>REML</code> method. When two categorical fixed factors are provided, an interaction term is included for main effects and smooth basis functions.
</p>
<p>If a <code>Random_Factor</code> is also provided, it is fitted using <code>bs = "re"</code> smooth.
</p>


<h3>Value</h3>

<p>ANOVA table of class &quot;anova&quot; and &quot;data.frame&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#with zooplankton data
ga_anova(data = data_zooplankton,
Y_value = "log(density_adj)",
Fixed_Factor = "taxon",
Smooth_Factor = "day")


</code></pre>

<hr>
<h2 id='ga_model'>Fit a generalised additive model (<code>gam</code>)</h2><span id='topic+ga_model'></span>

<h3>Description</h3>

<p>One of two functions for fitting generalised additive models (<code>gam</code>) with the <a href="https://CRAN.R-project.org/package=mgcv"><code>mgcv</code> package</a>. It will use the <code>gam()</code> function in <code>mgcv</code> for ANOVA designs with <strong>up to two categorical fixed factors</strong> (with two or more levels; <code>Fixed_Factor</code>), and <strong>exactly one factor is a continuous variable</strong> (e.g. time), which is called <code>Smooth_Factor</code>.
</p>

<ol>
<li> <p><code><a href="#topic+ga_model">ga_model</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ga_anova">ga_anova</a></code>
</p>
</li></ol>

<p>A smooth function is fitted with factor-wise smooth basis function (<code style="white-space: pre;">&#8288;by = &#8288;</code>). A default value for number of nodes (the argument <code>k</code> in <code>gam</code>) may work, but a specific number can be provided using the <code>Nodes</code> argument. The model is fit using the <code>REML</code> method. When two categorical fixed factors are provided, an interaction term is included for main effects and smooth basis functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ga_model(
  data,
  Y_value,
  Fixed_Factor,
  Smooth_Factor,
  Random_Factor = NULL,
  Nodes = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ga_model_+3A_data">data</code></td>
<td>
<p>a data frame where categorical independent variables are converted to factors using <code>as.factor()</code> first. The function will throw errors without this.</p>
</td></tr>
<tr><td><code id="ga_model_+3A_y_value">Y_value</code></td>
<td>
<p>name of column containing quantitative (dependent) variable, provided within &quot;quotes&quot;.</p>
</td></tr>
<tr><td><code id="ga_model_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>name(s) of categorical fixed factors (independent variables) provided within quotes (e.g., &quot;A&quot;) or as a vector if more than one (e.g., c(&quot;A&quot;, &quot;B&quot;). Convert to factors first with <code>as.factor</code>.</p>
</td></tr>
<tr><td><code id="ga_model_+3A_smooth_factor">Smooth_Factor</code></td>
<td>
<p>the continuous variable to fit smoothly with a basis function, provided within &quot;quotes&quot; (only 1 Smooth_Factor allowed).</p>
</td></tr>
<tr><td><code id="ga_model_+3A_random_factor">Random_Factor</code></td>
<td>
<p>name(s) of random factors to be provided in &quot;quotes&quot; (only 1 Random_Factor allowed). Convert to factor with <code>as.factor</code> first.</p>
</td></tr>
<tr><td><code id="ga_model_+3A_nodes">Nodes</code></td>
<td>
<p>number of nodes (the parameter <code>k</code> in <code>gam</code>).</p>
</td></tr>
<tr><td><code id="ga_model_+3A_...">...</code></td>
<td>
<p>any additional variables to pass on to <code>gam</code> or <code>anova</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a <code>Random_Factor</code> is also provided, it is fitted using <code>bs = "re"</code> smooth.
</p>


<h3>Value</h3>

<p>This function gives a generalised additive model object of class &quot;gam&quot;, &quot;lm&quot; and &quot;glm&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#fit a model with zooplankton data
z1 &lt;- ga_model(data = data_zooplankton,
Y_value = "log(density_adj)",
Fixed_Factor = "taxon",
Smooth_Factor = "day")

</code></pre>

<hr>
<h2 id='get_graf_colours'>Get graf internal</h2><span id='topic+get_graf_colours'></span>

<h3>Description</h3>

<p>Function to make grafify colour scheme. <a href="https://drsimonj.svbtle.com/creating-corporate-colour-palettes-for-ggplot2">Thank you Dr Simon</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_graf_colours(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_graf_colours_+3A_...">...</code></td>
<td>
<p>internal</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To visualise grafify colours use <code>plot_grafify_palette</code>.
</p>


<h3>Value</h3>

<p>This function returns names and hexcodes of colours in grafify as a character vector.
</p>

<hr>
<h2 id='graf_col_palette'>Call <code>grafify</code> palettes for scale &amp; fill functions</h2><span id='topic+graf_col_palette'></span>

<h3>Description</h3>

<p><code>graf_col_palette</code> and <code>graf_col_palette_default</code> functions generate colours for <code>grafify</code> scale functions.
<code>graf_col_palette</code> picks sequential colours when the number of discrete colours needed is less than that in the palette. This is the default for <code>grafify</code> with <code>ColoSeq = TRUE</code>. If the number of colours required is more than that in the discrete palette, it fills intervening colours using the <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a>[grDevices]</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graf_col_palette(palette = "okabe_ito", reverse = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="graf_col_palette_+3A_palette">palette</code></td>
<td>
<p>internal</p>
</td></tr>
<tr><td><code id="graf_col_palette_+3A_reverse">reverse</code></td>
<td>
<p>internal</p>
</td></tr>
<tr><td><code id="graf_col_palette_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>graf_col_palette_default</code> picks the most distant colours within the palette, rather than in the sequence they are in the palette, when the number of colours required is less than that in the palette.
</p>
<p>Colour order can be reversed in both functions.
</p>
<p>When only one colour discreet is required, and you want to <code>reverse</code> the colour palette, <code>ColSeq</code> should be set to <code>FALSE</code>.
</p>


<h3>Value</h3>

<p>This generates required number of sequential colours from the chosen grafify palette when called by scale functions of  ggplot2.
</p>

<hr>
<h2 id='graf_col_palette_default'>Call <code>grafify</code> palettes for scale &amp; fill functions</h2><span id='topic+graf_col_palette_default'></span>

<h3>Description</h3>

<p><code>graf_col_palette</code> and <code>graf_col_palette_default</code> functions generate colours for <code>grafify</code> scale functions.
<code>graf_col_palette</code> picks sequential colours when the number of discrete colours needed is less than that in the palette. This is the default for <code>grafify</code> with <code>ColoSeq = TRUE</code>. If the number of colours required is more than that in the discrete palette, it fills intervening colours using the <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a>[grDevices]</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graf_col_palette_default(palette = "okabe_ito", reverse = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="graf_col_palette_default_+3A_palette">palette</code></td>
<td>
<p>internal</p>
</td></tr>
<tr><td><code id="graf_col_palette_default_+3A_reverse">reverse</code></td>
<td>
<p>internal</p>
</td></tr>
<tr><td><code id="graf_col_palette_default_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>graf_col_palette_default</code> picks the most distant colours within the palette, rather than in the sequence they are in the palette, when the number of colours required is less than that in the palette.
</p>
<p>Colour order can be reversed in both functions.
</p>
<p>When only one colour discreet is required, and you want to <code>reverse</code> the colour palette, <code>ColSeq</code> should be set to <code>FALSE</code>.
</p>


<h3>Value</h3>

<p>This generates required number of distant colours from the chosen grafify palette when called by scale functions of  ggplot2.
</p>

<hr>
<h2 id='graf_colours'>List of hexcodes of colours in grafify palettes</h2><span id='topic+graf_colours'></span>

<h3>Description</h3>

<p>To visualise these colours use <code>plot_grafify_palette</code>. <code>okabe_ito</code>, <code>bright</code>, <code>contrast</code>, <code>dark</code>, <code>light</code>, <code>muted</code>, <code>pale</code>, <code>vibrant</code>, <code>yello_conti</code> from Paul Tol's <a href="https://personal.sron.nl/~pault/">post</a>. Zesty, Pastel, Elegant from this <a href="https://venngage.com/blog/color-blind-friendly-palette/">link</a>. Colour hexcodes for <code>fishy</code>, <code>kelly</code>, <code>r4</code>, <code>safe</code>, <code>OrBl_div</code>, <code>PrGn_div</code>, <code>blue_conti</code>, <code>grey_conti</code> taken from <a href="https://github.com/cols4all/cols4all-R">cols4all:c4a_gui</a> package. All schemes are colour blind-friendly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graf_colours
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 154.
</p>


<h3>Value</h3>

<p>This is a character vector with names and hexcodes of colours used by palette functions. It is used by <code>get_graf_colours</code> to generate palettes.
</p>

<hr>
<h2 id='graf_palettes'>List of palettes available in grafify package</h2><span id='topic+graf_palettes'></span>

<h3>Description</h3>

<p>To visualise these colours use <code>plot_grafify_palette</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graf_palettes
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 18.
</p>


<h3>Value</h3>

<p>This function returns a list of palettes in grafify with names and hexcodes of colours in those  palettes.
Names of palettes available are as follows:
</p>
<p>Categorical/discreet palettes:
</p>

<ul>
<li> <p><code>okabe_ito</code>
</p>
</li>
<li> <p><code>bright</code>
</p>
</li>
<li> <p><code>contrast</code>
</p>
</li>
<li> <p><code>dark</code>
</p>
</li>
<li> <p><code>kelly</code>
</p>
</li>
<li> <p><code>light</code>
</p>
</li>
<li> <p><code>muted</code>
</p>
</li>
<li> <p><code>pale</code>
</p>
</li>
<li> <p><code>r4</code>
</p>
</li>
<li> <p><code>safe</code>
</p>
</li>
<li> <p><code>vibrant</code>
</p>
</li></ul>

<p>Sequential quantitative palettes:
</p>

<ul>
<li> <p><code>grey_conti</code>
</p>
</li>
<li> <p><code>blue_conti</code>
</p>
</li>
<li> <p><code>yellow_conti</code>
</p>
</li></ul>

<p>Divergent quantitative palettes:
</p>

<ul>
<li> <p><code>OrBl_div</code>
</p>
</li>
<li> <p><code>PrGn_div</code>
</p>
</li></ul>


<hr>
<h2 id='make_1way_data'>Make one-way or two-way independent group or randomised block design data.</h2><span id='topic+make_1way_data'></span>

<h3>Description</h3>

<p>The <code><a href="#topic+make_1way_data">make_1way_data</a></code>, <code><a href="#topic+make_1way_rb_data">make_1way_rb_data</a></code>, <code><a href="#topic+make_2way_data">make_2way_data</a></code> and <code><a href="#topic+make_2way_rb_data">make_2way_rb_data</a></code> functions generate independent or randomised block (rb) design data of one-way or two-way designs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_1way_data(Group_means, Num_obs, Residual_SD)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_1way_data_+3A_group_means">Group_means</code></td>
<td>
<p>a vector with means of each level of the first fixed factor (FixFac_X1) measured within Group 1.</p>
</td></tr>
<tr><td><code id="make_1way_data_+3A_num_obs">Num_obs</code></td>
<td>
<p>a single numeric value indicating the number of independent measurements, i.e. levels within the random factor Experiment.</p>
</td></tr>
<tr><td><code id="make_1way_data_+3A_residual_sd">Residual_SD</code></td>
<td>
<p>a single numeric value indicating residual SD in the model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Random variates from the normal distribution based on user provided mean and SD provided are generated. For independent designs, the <code>Residual_SD</code> argument is used to set expected residual SD from the linear model. Exp_SD is used to set experiment-to-experiment SD, that will be assigned to the random factor for rb designs.
</p>
<p>Num_exp sets the number of independent measurements per group.
</p>
<p>For one-way designs, the user provides Group_means as a vector. Number of levels are recognised based on number of means.
For two-way designs, two vectors are to be provided by the user containing means of levels of a second factor. Number of means in both vectors should be the same. These functions can only handle balanced designs, i.e. same number of observations in all groups.
</p>
<p>The output is a data frame with one or two columns denoting the fixed factor with levels that match the number of means entered. For rb data, the column for RandFac denotes levels of the blocking factor.
The quantitative response variables are in the numeric Values column.
</p>


<h3>Value</h3>

<p>This function produces a <code>data.frame</code> object containing simulated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Basic usage with three levels within Factor_X, 
#20 observations in each group, with residual SD 15

one_independent_tab &lt;- make_1way_data(c(350, 250, 100), 15, 20)

str(one_independent_tab)
head(one_independent_tab)
</code></pre>

<hr>
<h2 id='make_1way_rb_data'>Make one-way or two-way independent group or randomised block design data.</h2><span id='topic+make_1way_rb_data'></span>

<h3>Description</h3>

<p>The <code><a href="#topic+make_1way_data">make_1way_data</a></code>, <code><a href="#topic+make_1way_rb_data">make_1way_rb_data</a></code>, <code><a href="#topic+make_2way_data">make_2way_data</a></code> and <code><a href="#topic+make_2way_rb_data">make_2way_rb_data</a></code> functions generate independent or randomised block (rb) design data of one-way or two-way designs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_1way_rb_data(Group_means, Num_exp, Exp_SD, Residual_SD)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_1way_rb_data_+3A_group_means">Group_means</code></td>
<td>
<p>a vector with means of each level of the first fixed factor (FixFac_X1) measured within Group 1.</p>
</td></tr>
<tr><td><code id="make_1way_rb_data_+3A_num_exp">Num_exp</code></td>
<td>
<p>a single numeric value. indicating the number of independent measurements, i.e. levels within the random factor RandFac.</p>
</td></tr>
<tr><td><code id="make_1way_rb_data_+3A_exp_sd">Exp_SD</code></td>
<td>
<p>a single numeric value indicating the standard deviation (SD) between experiments, i.e. within RandFac.</p>
</td></tr>
<tr><td><code id="make_1way_rb_data_+3A_residual_sd">Residual_SD</code></td>
<td>
<p>a single numeric value indicating residual SD in the model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Random variates from the normal distribution based on user provided mean and SD provided are generated. For independent designs, the <code>Residual_SD</code> argument is used to set expected residual SD from the linear model. Exp_SD is used to set experiment-to-experiment SD, that will be assigned to the random factor for rb designs.
</p>
<p>Num_exp sets the number of independent measurements per group.
</p>
<p>For one-way designs, the user provides Group_means as a vector. Number of levels are recognised based on number of means.
For two-way designs, two vectors are to be provided by the user containing means of levels of a second factor. Number of means in both vectors should be the same. These functions can only handle balanced designs, i.e. same number of observations in all groups.
</p>
<p>The output is a data frame with one or two columns denoting the fixed factor with levels that match the number of means entered. For rb data, the column for RandFac denotes levels of the blocking factor.
The quantitative response variables are in the numeric Values column.
</p>


<h3>Value</h3>

<p>This function produces a <code>data.frame</code> object containing simulated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Basic usage with two levels within FactorX2, 
#20 experiments with inter-experiment SD 20, and residual SD 15

two_rb_tab &lt;- make_2way_rb_data(c(100, 20), c(200, 300), 20, 20, 15)

str(two_rb_tab)
head(two_rb_tab)
</code></pre>

<hr>
<h2 id='make_2way_data'>Make one-way or two-way independent group or randomised block design data.</h2><span id='topic+make_2way_data'></span>

<h3>Description</h3>

<p>The <code><a href="#topic+make_1way_data">make_1way_data</a></code>, <code><a href="#topic+make_1way_rb_data">make_1way_rb_data</a></code>, <code><a href="#topic+make_2way_data">make_2way_data</a></code> and <code><a href="#topic+make_2way_rb_data">make_2way_rb_data</a></code> functions generate independent or randomised block (rb) design data of one-way or two-way designs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_2way_data(Group_1_means, Group_2_means, Num_obs, Residual_SD)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_2way_data_+3A_group_1_means">Group_1_means</code></td>
<td>
<p>a vector with means of each level of the first fixed factor (FixFac_X1) measured within Group 1.</p>
</td></tr>
<tr><td><code id="make_2way_data_+3A_group_2_means">Group_2_means</code></td>
<td>
<p>only for <code>make_2way_data</code> and <code>make_2way_rb_data</code>: a vector with mean(s) of each level of FactorX2 measured within Group 2.</p>
</td></tr>
<tr><td><code id="make_2way_data_+3A_num_obs">Num_obs</code></td>
<td>
<p>a single numeric value indicating the number of independent measurements, i.e. levels within the random factor Experiment.</p>
</td></tr>
<tr><td><code id="make_2way_data_+3A_residual_sd">Residual_SD</code></td>
<td>
<p>a single numeric value indicating residual SD in the model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Random variates from the normal distribution based on user provided mean and SD provided are generated. For independent designs, the <code>Residual_SD</code> argument is used to set expected residual SD from the linear model. Exp_SD is used to set experiment-to-experiment SD, that will be assigned to the random factor for rb designs.
</p>
<p>Num_obs sets the number of independent measurements per group.
</p>
<p>For one-way designs, the user provides Group_means as a vector. Number of levels are recognised based on number of means.
For two-way designs, two vectors are to be provided by the user containing means of levels of a second factor. Number of means in both vectors should be the same. These functions can only handle balanced designs, i.e. same number of observations in all groups.
</p>
<p>The output is a data frame with one or two columns denoting the fixed factor with levels that match the number of means entered. For rb data, the column for RandFac denotes levels of the blocking factor.
The quantitative response variables are in the numeric Values column.
</p>


<h3>Value</h3>

<p>This function produces a <code>data.frame</code> object containing simulated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Basic usage with two levels within FactorX2, 20 observations in each group, with residual SD 15

two_independent_tab &lt;- make_2way_data(c(100, 20), c(200, 300), 20, 15)

#Four levels with 5 observations and residual SD 5
two_independent_tab &lt;- make_2way_data(c(100, 20, 1500, 20), c(150, 5, 1450, 25), 5, 5)
</code></pre>

<hr>
<h2 id='make_2way_rb_data'>Make one-way or two-way independent group or randomised block design data.</h2><span id='topic+make_2way_rb_data'></span>

<h3>Description</h3>

<p>The <code><a href="#topic+make_1way_data">make_1way_data</a></code>, <code><a href="#topic+make_1way_rb_data">make_1way_rb_data</a></code>, <code><a href="#topic+make_2way_data">make_2way_data</a></code> and <code><a href="#topic+make_2way_rb_data">make_2way_rb_data</a></code> functions generate independent or randomised block (rb) design data of one-way or two-way designs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_2way_rb_data(Group1_means, Group2_means, Num_exp, Exp_SD, Residual_SD)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_2way_rb_data_+3A_group1_means">Group1_means</code></td>
<td>
<p>a vector with means of each level of the first fixed factor (FixFac_X1) measured within Group 1.</p>
</td></tr>
<tr><td><code id="make_2way_rb_data_+3A_group2_means">Group2_means</code></td>
<td>
<p>only for <code>make_2way_data</code> and <code>make_2way_rb_data</code>: a vector with mean(s) of each level of FactorX2 measured within Group 2.</p>
</td></tr>
<tr><td><code id="make_2way_rb_data_+3A_num_exp">Num_exp</code></td>
<td>
<p>a single numeric value indicating the number of independent measurements, i.e. levels within the random factor RandFac.</p>
</td></tr>
<tr><td><code id="make_2way_rb_data_+3A_exp_sd">Exp_SD</code></td>
<td>
<p>a single numeric value indicating the standard deviation (SD) between experiment, i.e. within RandFac.</p>
</td></tr>
<tr><td><code id="make_2way_rb_data_+3A_residual_sd">Residual_SD</code></td>
<td>
<p>a single numeric value indicating residual SD in the model.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Random variates from the normal distribution based on user provided mean and SD provided are generated. For independent designs, the <code>Residual_SD</code> argument is used to set expected residual SD from the linear model. Exp_SD is used to set experiment-to-experiment SD, that will be assigned to the random factor (RandFac) for rb designs.
</p>
<p>Num_exp sets the number of independent measurements per group.
</p>
<p>For one-way designs, the user provides Group_means as a vector. Number of levels are recognised based on number of means.
For two-way designs, two vectors are to be provided by the user containing means of levels of a second factor. Number of means in both vectors should be the same. These functions can only handle balanced designs, i.e. same number of observations in all groups.
</p>
<p>The output is a data frame with one or two columns denoting the fixed factor with levels that match the number of means entered. For rb data, the column for RandFac denotes levels of the blocking factor.
The quantitative response variables are in the numeric Values column.
</p>


<h3>Value</h3>

<p>This function produces a <code>data.frame</code> object containing simulated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Basic usage with two levels within FactorX2, 
#20 experiments with inter-experiment SD 20, and residual SD 15

two_rb_tab &lt;- make_2way_rb_data(c(100, 20), c(200, 300), 20, 20, 15)

str(two_rb_tab)
head(two_rb_tab)
</code></pre>

<hr>
<h2 id='mixed_anova'>ANOVA table from linear mixed effects analysis.</h2><span id='topic+mixed_anova'></span>

<h3>Description</h3>

<p>One of four related functions for mixed effects analyses (based on <code><a href="lme4.html#topic+lmer">lmer</a></code> and <code><a href="lmerTest.html#topic+as_lmerModLmerTest">as_lmerModLmerTest</a></code>) to get a linear model for downstream steps, or an ANOVA table.
</p>

<ol>
<li> <p><code>mixed_model</code>
</p>
</li>
<li> <p><code>mixed_anova</code>
</p>
</li>
<li> <p><code>mixed_model_slopes</code>
</p>
</li>
<li> <p><code>mixed_anova_slopes</code>.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>mixed_anova(
  data,
  Y_value,
  Fixed_Factor,
  Random_Factor,
  Df_method = "Kenward-Roger",
  SS_method = "II",
  AvgRF = TRUE,
  Formula = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mixed_anova_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="mixed_anova_+3A_y_value">Y_value</code></td>
<td>
<p>name of column containing quantitative (dependent) variable, provided within &quot;quotes&quot;. The following transformations are permitted: &quot;log(Y_value)&quot;, &quot;log(Y_value + c)&quot; where c a positive number, &quot;logit(Y_value)&quot; or &quot;logit(Y_value/100)&quot; which may be useful when <code>Y_value</code> are percentages  (note quotes outside the log or logit calls); &quot;sqrt(Y_value)&quot; or &quot;(Y_value)^2&quot; should also work. During posthoc-comparisons, log and logit transformations will be back-transformed to the original scale. Other transformations, e.g., &quot;sqrt(Y_value)&quot; will not be back-transformed. Check out the <code><a href="emmeans.html#topic+regrid">regrid</a></code> and <code><a href="emmeans.html#topic+ref_grid">ref_grid</a></code> for details if you need back-transformation to the response scale.</p>
</td></tr>
<tr><td><code id="mixed_anova_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>name(s) of categorical fixed factors (independent variables) provided within quotes (e.g., &quot;A&quot;) or as a vector if more than one (e.g., c(&quot;A&quot;, &quot;B&quot;). If a numeric variable is used, transformations similar to <code>Y_value</code> are permitted.</p>
</td></tr>
<tr><td><code id="mixed_anova_+3A_random_factor">Random_Factor</code></td>
<td>
<p>name(s) of random factors to allow random intercepts; to be provided within quotes (e.g., &quot;R&quot;) or as a vector when more than one (e.g., c(&quot;R1&quot;, &quot;R2&quot;)).</p>
</td></tr>
<tr><td><code id="mixed_anova_+3A_df_method">Df_method</code></td>
<td>
<p>method for calculating degrees of freedom. Default is Kenward-Roger, can be changed to &quot;Satterthwaite&quot;.</p>
</td></tr>
<tr><td><code id="mixed_anova_+3A_ss_method">SS_method</code></td>
<td>
<p>type of sum of square, default is type II, can be changed to &quot;I&quot;, &quot;III&quot;, &quot;1&quot; or &quot;2&quot;, or others.</p>
</td></tr>
<tr><td><code id="mixed_anova_+3A_avgrf">AvgRF</code></td>
<td>
<p>this is a new argument since v5.0.0. The default <code>AvgRF = TRUE</code> will use the mean of <code>Y_value</code> (the response variable) grouped by levels of the <code>Fixed_Factor</code> and <code>Random_Factor</code>  (using <code><a href="#topic+table_summary">table_summary</a></code>). This ensures that replicates within <code>Random_Factor</code> (or any other unused variable) are averaged (e.g., technical replicates nested within experimental blocks) before fitting a linear model and the denominator Df values are sensible. The name of the data frame in the model object will have <code>(AvgRF)</code> appended to it to indicate the averaging within levels of the <code>Random_Factor</code>. Using <code>AvgRF = FALSE</code> will lead to behaviour like versions &lt;5.0.0.</p>
</td></tr>
<tr><td><code id="mixed_anova_+3A_formula">Formula</code></td>
<td>
<p>directly provide an a formula (within quotes) as you would if you were using <code><a href="lme4.html#topic+lmer">lmer</a></code>. If <code>Y_value</code>, <code>Fixed_Factor</code> and <code>Random_Factor</code> are provided, they will be ignored. This is basically a wrapper, which may be useful if fitting more complex random factor structures.</p>
</td></tr>
<tr><td><code id="mixed_anova_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass on to <code><a href="lme4.html#topic+lmer">lmer</a></code> if required.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions require a data table, one dependent variable (Y_value), one or more independent variables (Fixed_Factor), and at least one random factor (Random_Factor). These should match names of variables in the long-format data table exactly. Since v5.0.0, if <code>AvgRF = TRUE</code>, the response variable is averaged over levels of the fixed and random factors (to collapse replicate observations) and reduce the number of denominator degrees of freedom. If you do not want to do this, set <code>AvgRF = FALSE</code>.
</p>
<p>Outputs of <code>mixed_model</code> and <code>mixed_model_slopes</code> can be used for post-hoc comparisons with <code><a href="#topic+posthoc_Pairwise">posthoc_Pairwise</a></code>, <code><a href="#topic+posthoc_Levelwise">posthoc_Levelwise</a></code>, <code><a href="#topic+posthoc_vsRef">posthoc_vsRef</a></code>, <code><a href="#topic+posthoc_Trends_Pairwise">posthoc_Trends_Pairwise</a></code>, <code><a href="#topic+posthoc_Trends_Levelwise">posthoc_Trends_Levelwise</a></code> and <code><a href="#topic+posthoc_Trends_vsRef">posthoc_Trends_vsRef</a></code>or with <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>.
</p>
<p>More than one fixed factors can be provided as a vector (e.g. c(&quot;A&quot;, &quot;B&quot;)). A full model with interaction term is fitted.
This means when <code>Y_value = Y, Fixed_factor = c("A", "B"), Random_factor = "R"</code> are entered as arguments, these are passed on as <code>Y ~ A*B + (1|R)</code> (which is equivalent to <code>Y ~ A + B + A:B + (1|R)</code>).
</p>
<p>In <code>mixed_model_slopes</code> and <code>mixed_anova_slopes</code>, the following kind of formula is used: <code>Y ~ A*B + (S|R)</code> (which is equivalent to <code>Y ~ A + B + A:B + (S|R)</code>).
In this experimental implementation, random slopes and intercepts are fitted (<code>(Slopes_Factor|Random_Factor)</code>). Only one term each is allowed for <code>Slopes_Factor</code> and <code>Random_Factor</code>.
</p>


<h3>Value</h3>

<p>ANOVA table of class &quot;anova&quot; and &quot;data.frame&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Usage with one fixed (Student) and random factor (Experiment)
mixed_anova(data = data_doubling_time, 
Y_value = "Doubling_time", 
Fixed_Factor = "Student", 
Random_Factor = "Experiment")

#with formula
mixed_anova(data = data_doubling_time, 
Formula = "Doubling_time ~ Student +(1|Experiment)")
</code></pre>

<hr>
<h2 id='mixed_anova_slopes'>ANOVA table from linear mixed effects analysis.</h2><span id='topic+mixed_anova_slopes'></span>

<h3>Description</h3>

<p>One of four related functions for mixed effects analyses (based on <code><a href="lme4.html#topic+lmer">lmer</a></code> and <code><a href="lmerTest.html#topic+as_lmerModLmerTest">as_lmerModLmerTest</a></code>) to get a linear model for downstream steps, or an ANOVA table.
</p>

<ol>
<li> <p><code>mixed_model</code>
</p>
</li>
<li> <p><code>mixed_anova</code>
</p>
</li>
<li> <p><code>mixed_model_slopes</code>
</p>
</li>
<li> <p><code>mixed_anova_slopes</code>.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>mixed_anova_slopes(
  data,
  Y_value,
  Fixed_Factor,
  Slopes_Factor,
  Random_Factor,
  Df_method = "Kenward-Roger",
  SS_method = "II",
  AvgRF = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mixed_anova_slopes_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="mixed_anova_slopes_+3A_y_value">Y_value</code></td>
<td>
<p>name of column containing quantitative (dependent) variable, provided within &quot;quotes&quot;. The following transformations are permitted: &quot;log(Y_value)&quot;, &quot;log(Y_value + c)&quot; where c a positive number, &quot;logit(Y_value)&quot; or &quot;logit(Y_value/100)&quot; which may be useful when <code>Y_value</code> are percentages  (note quotes outside the log or logit calls); &quot;sqrt(Y_value)&quot; or &quot;(Y_value)^2&quot; should also work. During posthoc-comparisons, log and logit transformations will be back-transformed to the original scale. Other transformations, e.g., &quot;sqrt(Y_value)&quot; will not be back-transformed. Check out the <code><a href="emmeans.html#topic+regrid">regrid</a></code> and <code><a href="emmeans.html#topic+ref_grid">ref_grid</a></code> for details if you need back-transformation to the response scale.</p>
</td></tr>
<tr><td><code id="mixed_anova_slopes_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>name(s) of categorical fixed factors (independent variables) provided within quotes (e.g., &quot;A&quot;) or as a vector if more than one (e.g., c(&quot;A&quot;, &quot;B&quot;). If a numeric variable is used, transformations similar to <code>Y_value</code> are permitted.</p>
</td></tr>
<tr><td><code id="mixed_anova_slopes_+3A_slopes_factor">Slopes_Factor</code></td>
<td>
<p>name of factor to allow varying slopes on. Only one variable is allowed.</p>
</td></tr>
<tr><td><code id="mixed_anova_slopes_+3A_random_factor">Random_Factor</code></td>
<td>
<p>name(s) of random factors to allow random intercepts; to be provided within quotes (e.g., &quot;R&quot;) or as a vector when more than one (e.g., c(&quot;R1&quot;, &quot;R2&quot;)). Only one variable is allowed.</p>
</td></tr>
<tr><td><code id="mixed_anova_slopes_+3A_df_method">Df_method</code></td>
<td>
<p>method for calculating degrees of freedom. Default is Kenward-Roger, can be changed to &quot;Satterthwaite&quot;.</p>
</td></tr>
<tr><td><code id="mixed_anova_slopes_+3A_ss_method">SS_method</code></td>
<td>
<p>type of sum of square, default is type II, can be changed to &quot;I&quot;, &quot;III&quot;, &quot;1&quot; or &quot;2&quot;, or others.</p>
</td></tr>
<tr><td><code id="mixed_anova_slopes_+3A_avgrf">AvgRF</code></td>
<td>
<p>this is a new argument since v5.0.0. The default <code>AvgRF = TRUE</code> will use the mean of <code>Y_value</code> (the response variable) grouped by levels of the <code>Fixed_Factor</code> and <code>Random_Factor</code>  (using <code><a href="#topic+table_summary">table_summary</a></code>). This ensures that replicates within <code>Random_Factor</code> (or any other unused variable) are averaged (e.g., technical replicates nested within experimental blocks) before fitting a linear model and the denominator Df values are sensible. The name of the data frame in the model object will have <code>(AvgRF)</code> appended to it to indicate the averaging within levels of the <code>Random_Factor</code>. Using <code>AvgRF = FALSE</code> will lead to behaviour like versions &lt;5.0.0.</p>
</td></tr>
<tr><td><code id="mixed_anova_slopes_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass on to <code><a href="lme4.html#topic+lmer">lmer</a></code> if required.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions require a data table, one dependent variable (Y_value), one or more independent variables (Fixed_Factor), and at least one random factor (Random_Factor). These should match names of variables in the long-format data table exactly. Since v5.0.0, if <code>AvgRF = TRUE</code>, the response variable is averaged over levels of the fixed and random factors (to collapse replicate observations) and reduce the number of denominator degrees of freedom. If you do not want to do this, set <code>AvgRF = FALSE</code>. If you do not want to do this, set <code>AvgRF = FALSE</code>.
</p>
<p>For more advanced models with slopes and intercept, use <code><a href="#topic+mixed_model">mixed_model</a></code> or <code><a href="#topic+mixed_anova">mixed_anova</a></code> using the <code>Formula</code> argument.
</p>
<p>Outputs of <code>mixed_model</code> and <code>mixed_model_slopes</code> can be used for post-hoc comparisons with <code><a href="#topic+posthoc_Pairwise">posthoc_Pairwise</a></code>, <code><a href="#topic+posthoc_Levelwise">posthoc_Levelwise</a></code>, <code><a href="#topic+posthoc_vsRef">posthoc_vsRef</a></code>, <code><a href="#topic+posthoc_Trends_Pairwise">posthoc_Trends_Pairwise</a></code>, <code><a href="#topic+posthoc_Trends_Levelwise">posthoc_Trends_Levelwise</a></code> and <code><a href="#topic+posthoc_Trends_vsRef">posthoc_Trends_vsRef</a></code>or with <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>.
</p>
<p>More than one fixed factors can be provided as a vector (e.g. c(&quot;A&quot;, &quot;B&quot;)). A full model with interaction term is fitted.
This means when <code>Y_value = Y, Fixed_factor = c("A", "B"), Random_factor = "R"</code> are entered as arguments, these are passed on as <code>Y ~ A*B + (1|R)</code> (which is equivalent to <code>Y ~ A + B + A:B + (1|R)</code>).
</p>
<p>In <code>mixed_model_slopes</code> and <code>mixed_anova_slopes</code>, the following kind of formula is used: <code>Y ~ A*B + (S|R)</code> (which is equivalent to <code>Y ~ A + B + A:B + (S|R)</code>).
In this experimental implementation, random slopes and intercepts are fitted (<code>(Slopes_Factor|Random_Factor)</code>). Only one term each is allowed for <code>Slopes_Factor</code> and <code>Random_Factor</code>.
</p>


<h3>Value</h3>

<p>ANOVA table of class &quot;anova&quot; and &quot;data.frame&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mixed_anova_slopes(data = data_2w_Tdeath,
Y_value = "PI",
Fixed_Factor = c("Genotype", "Time"),
Slopes_Factor = "Time",
Random_Factor = "Experiment")

</code></pre>

<hr>
<h2 id='mixed_model'>Model from a linear mixed effects model</h2><span id='topic+mixed_model'></span>

<h3>Description</h3>

<p>One of four related functions for mixed effects analyses (based on <code><a href="lme4.html#topic+lmer">lmer</a></code> and <code><a href="lmerTest.html#topic+as_lmerModLmerTest">as_lmerModLmerTest</a></code>) to get a linear model for downstream steps, or an ANOVA table.
</p>

<ol>
<li> <p><code>mixed_model</code>
</p>
</li>
<li> <p><code>mixed_anova</code>
</p>
</li>
<li> <p><code>mixed_model_slopes</code>
</p>
</li>
<li> <p><code>mixed_anova_slopes</code>.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>mixed_model(
  data,
  Y_value,
  Fixed_Factor,
  Random_Factor,
  AvgRF = TRUE,
  Formula = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mixed_model_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="mixed_model_+3A_y_value">Y_value</code></td>
<td>
<p>name of column containing quantitative (dependent) variable, provided within &quot;quotes&quot;. The following transformations are permitted: &quot;log(Y_value)&quot;, &quot;log(Y_value + c)&quot; where c a positive number, &quot;logit(Y_value)&quot; or &quot;logit(Y_value/100)&quot; which may be useful when <code>Y_value</code> are percentages  (note quotes outside the log or logit calls); &quot;sqrt(Y_value)&quot; or &quot;(Y_value)^2&quot; should also work. During posthoc-comparisons, log and logit transformations will be back-transformed to the original scale. Other transformations, e.g., &quot;sqrt(Y_value)&quot; will not be back-transformed. Check out the <code><a href="emmeans.html#topic+regrid">regrid</a></code> and <code><a href="emmeans.html#topic+ref_grid">ref_grid</a></code> for details if you need back-transformation to the response scale.</p>
</td></tr>
<tr><td><code id="mixed_model_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>name(s) of categorical fixed factors (independent variables) provided within quotes (e.g., &quot;A&quot;) or as a vector if more than one (e.g., c(&quot;A&quot;, &quot;B&quot;). If a numeric variable is used, transformations similar to <code>Y_value</code> are permitted.</p>
</td></tr>
<tr><td><code id="mixed_model_+3A_random_factor">Random_Factor</code></td>
<td>
<p>name(s) of random factors to allow random intercepts; to be provided within quotes (e.g., &quot;R&quot;) or as a vector when more than one (e.g., c(&quot;R1&quot;, &quot;R2&quot;)).</p>
</td></tr>
<tr><td><code id="mixed_model_+3A_avgrf">AvgRF</code></td>
<td>
<p>this is a new argument since v5.0.0. The default <code>AvgRF = TRUE</code> will use the mean of <code>Y_value</code> (the response variable) grouped by levels of the <code>Fixed_Factor</code> and <code>Random_Factor</code>  (using <code><a href="#topic+table_summary">table_summary</a></code>). This ensures that replicates within <code>Random_Factor</code> (or any other unused variable) are averaged (e.g., technical replicates nested within experimental blocks) before fitting a linear model and the denominator Df values are sensible. The name of the data frame in the model object will have <code>(AvgRF)</code> appended to it to indicate the averaging within levels of the <code>Random_Factor</code>. Using <code>AvgRF = FALSE</code> will lead to behaviour like versions &lt;5.0.0.</p>
</td></tr>
<tr><td><code id="mixed_model_+3A_formula">Formula</code></td>
<td>
<p>directly provide an a formula (within quotes) as you would if you were using <code><a href="lme4.html#topic+lmer">lmer</a></code>. If <code>Y_value</code>, <code>Fixed_Factor</code> and <code>Random_Factor</code> are provided, they will be ignored. This is basically a wrapper, which may be useful if fitting more complex random factor structures.</p>
</td></tr>
<tr><td><code id="mixed_model_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass on to <code><a href="lme4.html#topic+lmer">lmer</a></code> if required.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions require a data table, one dependent variable (Y_value), one or more independent variables (Fixed_Factor), and at least one random factor (Random_Factor). These should match names of variables in the long-format data table exactly. Since v5.0.0, if <code>AvgRF = TRUE</code>, the response variable is averaged over levels of the fixed and random factors (to collapse replicate observations) and reduce the number of denominator degrees of freedom. If you do not want to do this, set <code>AvgRF = FALSE</code>.
</p>
<p>Outputs of <code>mixed_model</code> and <code>mixed_model_slopes</code> can be used for post-hoc comparisons with <code><a href="#topic+posthoc_Pairwise">posthoc_Pairwise</a></code>, <code><a href="#topic+posthoc_Levelwise">posthoc_Levelwise</a></code>, <code><a href="#topic+posthoc_vsRef">posthoc_vsRef</a></code>, <code><a href="#topic+posthoc_Trends_Pairwise">posthoc_Trends_Pairwise</a></code>, <code><a href="#topic+posthoc_Trends_Levelwise">posthoc_Trends_Levelwise</a></code> and <code><a href="#topic+posthoc_Trends_vsRef">posthoc_Trends_vsRef</a></code>or with <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>.
</p>
<p>More than one fixed factors can be provided as a vector (e.g. c(&quot;A&quot;, &quot;B&quot;)). A full model with interaction term is fitted.
This means when <code>Y_value = Y, Fixed_factor = c("A", "B"), Random_factor = "R"</code> are entered as arguments, these are passed on as <code>Y ~ A*B + (1|R)</code> (which is equivalent to <code>Y ~ A + B + A:B + (1|R)</code>).
</p>
<p>In <code>mixed_model_slopes</code> and <code>mixed_anova_slopes</code>, the following kind of formula is used: <code>Y ~ A*B + (S|R)</code> (which is equivalent to <code>Y ~ A + B + A:B + (S|R)</code>).
In this experimental implementation, random slopes and intercepts are fitted (<code>(Slopes_Factor|Random_Factor)</code>). Only one term each is allowed for <code>Slopes_Factor</code> and <code>Random_Factor</code>.
</p>


<h3>Value</h3>

<p>This function returns an S4 object of class &quot;lmerModLmerTest&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#one fixed factor and random factor
mixed_model(data = data_doubling_time, 
Y_value = "Doubling_time", 
Fixed_Factor = "Student", 
Random_Factor = "Experiment")

#with formula
mixed_anova(data = data_doubling_time, 
Formula = "Doubling_time ~ Student +(1|Experiment)")

#' #save model
model &lt;- mixed_model(data = data_doubling_time, 
Y_value =  "Doubling_time", 
Fixed_Factor = "Student", 
Random_Factor = "Experiment")

#get model summary
summary(model)
</code></pre>

<hr>
<h2 id='mixed_model_slopes'>Model from a linear mixed effects model with varying slopes</h2><span id='topic+mixed_model_slopes'></span>

<h3>Description</h3>

<p>One of four related functions for mixed effects analyses (based on <code><a href="lme4.html#topic+lmer">lmer</a></code> and <code><a href="lmerTest.html#topic+as_lmerModLmerTest">as_lmerModLmerTest</a></code>) to get a linear model for downstream steps, or an ANOVA table.
</p>

<ol>
<li> <p><code>mixed_model</code>
</p>
</li>
<li> <p><code>mixed_anova</code>
</p>
</li>
<li> <p><code>mixed_model_slopes</code>
</p>
</li>
<li> <p><code>mixed_anova_slopes</code>.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>mixed_model_slopes(
  data,
  Y_value,
  Fixed_Factor,
  Slopes_Factor,
  Random_Factor,
  AvgRF = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mixed_model_slopes_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="mixed_model_slopes_+3A_y_value">Y_value</code></td>
<td>
<p>name of column containing quantitative (dependent) variable, provided within &quot;quotes&quot;. The following transformations are permitted: &quot;log(Y_value)&quot;, &quot;log(Y_value + c)&quot; where c a positive number, &quot;logit(Y_value)&quot; or &quot;logit(Y_value/100)&quot; which may be useful when <code>Y_value</code> are percentages  (note quotes outside the log or logit calls); &quot;sqrt(Y_value)&quot; or &quot;(Y_value)^2&quot; should also work. During posthoc-comparisons, log and logit transformations will be back-transformed to the original scale. Other transformations, e.g., &quot;sqrt(Y_value)&quot; will not be back-transformed. Check out the <code><a href="emmeans.html#topic+regrid">regrid</a></code> and <code><a href="emmeans.html#topic+ref_grid">ref_grid</a></code> for details if you need back-transformation to the response scale.</p>
</td></tr>
<tr><td><code id="mixed_model_slopes_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>name(s) of categorical fixed factors (independent variables) provided within quotes (e.g., &quot;A&quot;) or as a vector if more than one (e.g., c(&quot;A&quot;, &quot;B&quot;). If a numeric variable is used, transformations similar to <code>Y_value</code> are permitted.</p>
</td></tr>
<tr><td><code id="mixed_model_slopes_+3A_slopes_factor">Slopes_Factor</code></td>
<td>
<p>name of factor to allow varying slopes on. One one variable is allowed.</p>
</td></tr>
<tr><td><code id="mixed_model_slopes_+3A_random_factor">Random_Factor</code></td>
<td>
<p>name(s) of random factors to allow random intercepts; to be provided within quotes (e.g., &quot;R&quot;) or as a vector when more than one (e.g., c(&quot;R1&quot;, &quot;R2&quot;)). Only one variable is allowed.</p>
</td></tr>
<tr><td><code id="mixed_model_slopes_+3A_avgrf">AvgRF</code></td>
<td>
<p>this is a new argument since v5.0.0. The default <code>AvgRF = TRUE</code> will use the mean of <code>Y_value</code> (the response variable) grouped by levels of the <code>Fixed_Factor</code> and <code>Random_Factor</code>  (using <code><a href="#topic+table_summary">table_summary</a></code>). This ensures that replicates within <code>Random_Factor</code> (or any other unused variable) are averaged (e.g., technical replicates nested within experimental blocks) before fitting a linear model and the denominator Df values are sensible. The name of the data frame in the model object will have <code>(AvgRF)</code> appended to it to indicate the averaging within levels of the <code>Random_Factor</code>. Using <code>AvgRF = FALSE</code> will lead to behaviour like versions &lt;5.0.0.</p>
</td></tr>
<tr><td><code id="mixed_model_slopes_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass on to <code><a href="lme4.html#topic+lmer">lmer</a></code> if required.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions require a data table, one dependent variable (Y_value), one or more independent variables (Fixed_Factor), and at least one random factor (Random_Factor). These should match names of variables in the long-format data table exactly. Since v5.0.0, if <code>AvgRF = TRUE</code>, the response variable is averaged over levels of the fixed and random factors (to collapse replicate observations) and reduce the number of denominator degrees of freedom. If you do not want to do this, set <code>AvgRF = FALSE</code>.
</p>
<p>For more advanced models with slopes and intercept, use <code><a href="#topic+mixed_model">mixed_model</a></code> or <code><a href="#topic+mixed_anova">mixed_anova</a></code> using the <code>Formula</code> argument.
</p>
<p>Outputs of <code>mixed_model</code> and <code>mixed_model_slopes</code> can be used for post-hoc comparisons with <code><a href="#topic+posthoc_Pairwise">posthoc_Pairwise</a></code>, <code><a href="#topic+posthoc_Levelwise">posthoc_Levelwise</a></code>, <code><a href="#topic+posthoc_vsRef">posthoc_vsRef</a></code>, <code><a href="#topic+posthoc_Trends_Pairwise">posthoc_Trends_Pairwise</a></code>, <code><a href="#topic+posthoc_Trends_Levelwise">posthoc_Trends_Levelwise</a></code> and <code><a href="#topic+posthoc_Trends_vsRef">posthoc_Trends_vsRef</a></code>or with <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>.
</p>
<p>More than one fixed factors can be provided as a vector (e.g. c(&quot;A&quot;, &quot;B&quot;)). A full model with interaction term is fitted.
This means when <code>Y_value = Y, Fixed_factor = c("A", "B"), Random_factor = "R"</code> are entered as arguments, these are passed on as <code>Y ~ A*B + (1|R)</code> (which is equivalent to <code>Y ~ A + B + A:B + (1|R)</code>).
</p>
<p>In <code>mixed_model_slopes</code> and <code>mixed_anova_slopes</code>, the following kind of formula is used: <code>Y ~ A*B + (S|R)</code> (which is equivalent to <code>Y ~ A + B + A:B + (S|R)</code>).
In this experimental implementation, random slopes and intercepts are fitted (<code>(Slopes_Factor|Random_Factor)</code>). Only one term each is allowed for <code>Slopes_Factor</code> and <code>Random_Factor</code>.
</p>


<h3>Value</h3>

<p>This function returns an S4 object of class &quot;lmerModLmerTest&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#two fixed factors as a vector, 
#exactly one slope factor and random factor
mod &lt;- mixed_model_slopes(data = data_2w_Tdeath,
Y_value = "PI",
Fixed_Factor = c("Genotype", "Time"),
Slopes_Factor = "Time",
Random_Factor = "Experiment")
#get summary
summary(mod)
</code></pre>

<hr>
<h2 id='plot_3d_point_sd'>Plot of mean &amp; error bars for 1-way ANOVAs with matched shapes mapped to blocking factor.</h2><span id='topic+plot_3d_point_sd'></span>

<h3>Description</h3>

<p>One of 4 related functions for plotting 1-way ANOVA designs with a blocking factor.
</p>

<ol>
<li> <p><code><a href="#topic+plot_3d_point_sd">plot_3d_point_sd</a></code> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterbar">plot_3d_scatterbar</a></code> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterbox">plot_3d_scatterbox</a></code> (box &amp; whiskers)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterviolin">plot_3d_scatterviolin</a></code> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_3d_point_sd(
  data,
  xcol,
  ycol,
  shapes,
  facet,
  ErrorType = "SD",
  symsize = 3.5,
  s_alpha = 1,
  symshape = 22,
  all_alpha = 0.3,
  all_size = 2.5,
  all_shape = 0,
  all_jitter = 0,
  ewid = 0.2,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  ethick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_3d_point_sd_+3A_data">data</code></td>
<td>
<p>a data table, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) with the categorical factor to be plotted on X axis.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with quantitative variable to plot on the Y axis.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_shapes">shapes</code></td>
<td>
<p>name of the column (without quotes) with the blocking factor or another categorical variable.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_errortype">ErrorType</code></td>
<td>
<p>select the type of error bars to display. Default is &quot;SD&quot; (standard deviation). Other options are &quot;SEM&quot; (standard error of the mean) and &quot;CI95&quot; (95% confidence interval based on t distributions).</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e. 80% opacity). Set <code>s_alpha = 0</code> to not show scatter plot.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_symshape">symshape</code></td>
<td>
<p>The mean is shown with symbol of the shape number 21 (default, filled circle). Pick a number between 0-25 to pick a different type of symbol from ggplot2.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_all_alpha">all_alpha</code></td>
<td>
<p>fractional opacity of all data points (default = 0.3). Set to non-zero value if you would like all data points plotted in addition to the mean.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_all_size">all_size</code></td>
<td>
<p>size of symbols of all data points, if shown (default = 2.5).</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_all_shape">all_shape</code></td>
<td>
<p>all data points are shown with symbols of the shape number 0 (default, open square). Pick a number between 0-25 to pick a different type of symbol from ggplot2.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_all_jitter">all_jitter</code></td>
<td>
<p>reduce overlap of all data points, if shown, by setting a value between 0-1 (default = 0).</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_ewid">ewid</code></td>
<td>
<p>width of error bars, default set to 0.2.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_ethick">ethick</code></td>
<td>
<p>thickness of error bar lines; default <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> palettes or base R to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_3d_point_sd_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The blocking factor (or any other categorical variable) can be mapped to the <code>shapes</code> argument (up to 25 levels allowed). Variables passed to <code>xcol</code> and <code>shapes</code> are internally converted to factors even if they are numeric or other type of variables.
</p>
<p>In <code>plot_3d_point_sd</code> and <code>plot_3d_scatterbar</code>, the default error bar is SD (can be changed to SEM or CI95). In <code>plot_3d_point_sd</code>, a large coloured symbol is plotted at the mean, all other data are shown as smaller symbols. Boxplot uses <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code> to depict median (thicker line), box (interquartile range (IQR)) and the whiskers (1.5*IQR).
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> (logical TRUE/FALSE) decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>The resulting <code>ggplot2</code> graph can take additional geometries or other layers.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#3d version for 1-way data with blocking
#use plot_point_sd when no a blocking factor is not used
plot_3d_point_sd(data = data_1w_death, 
xcol = Genotype, ycol = Death, 
shapes = Experiment)
</code></pre>

<hr>
<h2 id='plot_3d_scatterbar'>Plot a bar graph for 1-way ANOVAs with matched shapes mapped to blocking factor.</h2><span id='topic+plot_3d_scatterbar'></span>

<h3>Description</h3>

<p>One of 4 related functions for plotting 1-way ANOVA designs with a blocking factor.
</p>

<ol>
<li> <p><code><a href="#topic+plot_3d_point_sd">plot_3d_point_sd</a></code> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterbar">plot_3d_scatterbar</a></code> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterbox">plot_3d_scatterbox</a></code> (box &amp; whiskers)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterviolin">plot_3d_scatterviolin</a></code> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_3d_scatterbar(
  data,
  xcol,
  ycol,
  shapes,
  facet,
  ErrorType = "SD",
  symsize = 3,
  s_alpha = 0.8,
  b_alpha = 1,
  jitter = 0.1,
  ewid = 0.2,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  bthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_3d_scatterbar_+3A_data">data</code></td>
<td>
<p>a data table, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) with the categorical factor to be plotted on X axis.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with quantitative variable to plot on the Y axis.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_shapes">shapes</code></td>
<td>
<p>name of the column (without quotes) with the second categorical factor, for example from a two-way ANOVA design.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_errortype">ErrorType</code></td>
<td>
<p>select the type of error bars to display. Default is &quot;SD&quot; (standard deviation). Other options are &quot;SEM&quot; (standard error of the mean) and &quot;CI95&quot; (95% confidence interval based on t distributions).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e. 80% opacity). Set <code>s_alpha = 0</code> to not show scatter plot.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes.  Default is set to 0, which results in white boxes inside violins. Change to any value &gt;0 up to 1 for different levels of transparency.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0.1. Increase to reduce symbol overlap, set to 0 for aligned symbols.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_ewid">ewid</code></td>
<td>
<p>width of error bars, default set to 0.2.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of lines of boxes; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> palettes or base R to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbar_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The blocking factor (or any other categorical variable) can be mapped to the <code>shapes</code> argument (up to 25 levels allowed). Variables passed to <code>xcol</code> and <code>shapes</code> are internally converted to factors even if they are numeric or other type of variables.
</p>
<p>In <code>plot_3d_point_sd</code> and <code>plot_3d_scatterbar</code>, the default error bar is SD (can be changed to SEM or CI95). In <code>plot_3d_point_sd</code>, a large coloured symbol is plotted at the mean, all other data are shown as smaller symbols. Boxplot uses <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code> to depict median (thicker line), box (interquartile range (IQR)) and the whiskers (1.5*IQR).
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> (logical TRUE/FALSE) decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>The resulting <code>ggplot2</code> graph can take additional geometries or other layers.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#3d version for 1-way data with blocking
#use plot_scatterbar_sd without blocking factor
plot_3d_scatterbar(data = data_1w_death, 
xcol = Genotype, ycol = Death, 
shapes = Experiment)
</code></pre>

<hr>
<h2 id='plot_3d_scatterbox'>Plot a scatter and box plot for 1-way ANOVAs with matched shapes mapped to blocking factor.</h2><span id='topic+plot_3d_scatterbox'></span>

<h3>Description</h3>

<p>One of 4 related functions for plotting 1-way ANOVA designs with a blocking factor.
</p>

<ol>
<li> <p><code><a href="#topic+plot_3d_point_sd">plot_3d_point_sd</a></code> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterbar">plot_3d_scatterbar</a></code> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterbox">plot_3d_scatterbox</a></code> (box &amp; whiskers)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterviolin">plot_3d_scatterviolin</a></code> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_3d_scatterbox(
  data,
  xcol,
  ycol,
  shapes,
  facet,
  symsize = 3,
  s_alpha = 0.8,
  b_alpha = 1,
  bwid = 0.5,
  jitter = 0.1,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  bthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_3d_scatterbox_+3A_data">data</code></td>
<td>
<p>a data table, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) with the categorical factor to be plotted on X axis. If your table has numeric X, enter <code>xcol = factor(name of colum)</code>.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with quantitative variable to plot on the Y axis.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_shapes">shapes</code></td>
<td>
<p>name of the column (without quotes) with the second categorical factor in a two-way ANOVA design.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e. 80% opacity). Set <code>s_alpha = 0</code> to not show scatter plot.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes.  Default is set to 0, which results in white boxes inside violins. Change to any value &gt;0 up to 1 for different levels of transparency.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_bwid">bwid</code></td>
<td>
<p>width of boxes; default 0.5.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0.1. Increase to reduce symbol overlap, set to 0 for aligned symbols.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of lines of boxes; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> or base R palettes to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterbox_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The blocking factor (or any other categorical variable) can be mapped to the <code>shapes</code> argument (up to 25 levels allowed). Variables passed to <code>xcol</code> and <code>shapes</code> are internally converted to factors even if they are numeric or other type of variables.
</p>
<p>In <code>plot_3d_point_sd</code> and <code>plot_3d_scatterbar</code>, the default error bar is SD (can be changed to SEM or CI95). In <code>plot_3d_point_sd</code>, a large coloured symbol is plotted at the mean, all other data are shown as smaller symbols. Boxplot uses <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code> to depict median (thicker line), box (interquartile range (IQR)) and the whiskers (1.5*IQR).
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> (logical TRUE/FALSE) decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>The resulting <code>ggplot2</code> graph can take additional geometries or other layers.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#3d version for 1-way data with blocking
plot_3d_scatterbox(data = data_1w_death, 
xcol = Genotype, ycol = Death, 
shapes = Experiment)
#use plot_scatterbox without a blocking factor

</code></pre>

<hr>
<h2 id='plot_3d_scatterviolin'>Plot a scatter with violin &amp; box plot for 1-way ANOVAs with matched shapes mapped to blocking factor.</h2><span id='topic+plot_3d_scatterviolin'></span>

<h3>Description</h3>

<p>One of 4 related functions for plotting 1-way ANOVA designs with a blocking factor.
</p>

<ol>
<li> <p><code><a href="#topic+plot_3d_point_sd">plot_3d_point_sd</a></code> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterbar">plot_3d_scatterbar</a></code> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterbox">plot_3d_scatterbox</a></code> (box &amp; whiskers)
</p>
</li>
<li> <p><code><a href="#topic+plot_3d_scatterviolin">plot_3d_scatterviolin</a></code> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_3d_scatterviolin(
  data,
  xcol,
  ycol,
  shapes,
  facet,
  symsize = 3,
  s_alpha = 0.8,
  b_alpha = 0,
  v_alpha = 1,
  bwid = 0.3,
  vadjust = 1,
  jitter = 0.1,
  TextXAngle = 0,
  scale = "width",
  trim = TRUE,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  bthick,
  vthick,
  bvthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_3d_scatterviolin_+3A_data">data</code></td>
<td>
<p>a data table, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) with the categorical factor to be plotted on X axis. If your table has numeric X, enter <code>xcol = factor(name of colum)</code>.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with quantitative variable to plot on the Y axis.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_shapes">shapes</code></td>
<td>
<p>name of the column (without quotes) with the second categorical factor in a two-way ANOVA design.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e. 80% opacity). Set <code>s_alpha = 0</code> to not show scatter plot.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes.  Default is set to 0, which results in white boxes inside violins. Change to any value &gt;0 up to 1 for different levels of transparency.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_v_alpha">v_alpha</code></td>
<td>
<p>fractional opacity of violins, default set to 1.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_bwid">bwid</code></td>
<td>
<p>width of boxes (default 0.3).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_vadjust">vadjust</code></td>
<td>
<p>number to adjust the smooth/wigglyness of violin plot (default is 1).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0.1. Increase to reduce symbol overlap, set to 0 for aligned symbols.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_scale">scale</code></td>
<td>
<p>set to &quot;area&quot; by default, can be changed to &quot;count&quot; or &quot;width&quot;.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_trim">trim</code></td>
<td>
<p>set whether tips of violin plot should be trimmed at high/low data. Default <code>trim = T</code>, can be changed to F.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of boxplots; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_vthick">vthick</code></td>
<td>
<p>thickness (in 'pt' units) of violins; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_bvthick">bvthick</code></td>
<td>
<p>thickness (in 'pt' units) of both violins and boxplots; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> or base R palettes to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_3d_scatterviolin_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code> or <code><a href="ggplot2.html#topic+geom_violin">geom_violin</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The blocking factor (or any other categorical variable) can be mapped to the <code>shapes</code> argument (up to 25 levels allowed). Variables passed to <code>xcol</code> and <code>shapes</code> are internally converted to factors even if they are numeric or other type of variables.
</p>
<p>In <code>plot_3d_point_sd</code> and <code>plot_3d_scatterbar</code>, the default error bar is SD (can be changed to SEM or CI95). In <code>plot_3d_point_sd</code>, a large coloured symbol is plotted at the mean, all other data are shown as smaller symbols. Boxplot uses <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code> to depict median (thicker line), box (interquartile range (IQR)) and the whiskers (1.5*IQR).
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> (logical TRUE/FALSE) decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>The resulting <code>ggplot2</code> graph can take additional geometries or other layers.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#3d version for 1-way data with blocking
plot_3d_scatterviolin(data = data_1w_death, 
xcol = Genotype, ycol = Death, 
shapes = Experiment)
#use plot_scatterviolin without a blocking factor

</code></pre>

<hr>
<h2 id='plot_4d_point_sd'>Plot mean &amp; error bars for 2-way ANOVAs with or without a blocking factor.</h2><span id='topic+plot_4d_point_sd'></span>

<h3>Description</h3>

<p>There are 4 related functions for 2-way ANOVA type plots. In addition to a categorical variable along the X-axis, a grouping factor is passed to either <code>points</code>, <code>bars</code> or <code>boxes</code> argument in these functions. A blocking factor (or any other categorical variable) can be optionally passed to the <code>shapes</code> argument.
</p>

<ol>
<li> <p><code><a href="#topic+plot_4d_point_sd">plot_4d_point_sd</a></code> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterbar">plot_4d_scatterbar</a></code> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterbox">plot_4d_scatterbox</a></code> (box &amp; whiskers)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterviolin">plot_4d_scatterviolin</a></code> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_4d_point_sd(
  data,
  xcol,
  ycol,
  points,
  shapes,
  facet,
  ErrorType = "SD",
  symsize = 3.5,
  s_alpha = 1,
  symshape = 22,
  all_alpha = 0.3,
  all_size = 2.5,
  all_shape = 0,
  all_jitter = 0,
  ewid = 0.2,
  group_wid = 0.8,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  ethick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_4d_point_sd_+3A_data">data</code></td>
<td>
<p>a data table, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) with the variable to plot on X axis (will be converted to a factor/categorical variable).</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_points">points</code></td>
<td>
<p>name of the column with grouping within the factor plotted on X-axis (will be converted to a factor/categorical variable).</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_shapes">shapes</code></td>
<td>
<p>name of the column (without quotes) that contains matched observations (e.g. subject IDs, experiment number) or another variable to pass on to symbol shapes (will be converted to a factor/categorical variable). If not provided, the shapes for all groups is the same, and can be changed with <code>all_shapes</code>, <code>all_alpha</code>, <code>all_size</code> etc.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_errortype">ErrorType</code></td>
<td>
<p>select the type of error bars to display. Default is &quot;SD&quot; (standard deviation). Other options are &quot;SEM&quot; (standard error of the mean) and &quot;CI95&quot; (95% confidence interval based on t distributions).</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.5.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 1 (i.e. fully opaque).</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_symshape">symshape</code></td>
<td>
<p>The mean is shown with symbol of the shape number 21 (default, filled circle). Pick a number between 0-25 to pick a different type of symbol from ggplot2.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_all_alpha">all_alpha</code></td>
<td>
<p>fractional opacity of all data points (default = 0.3).</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_all_size">all_size</code></td>
<td>
<p>size of symbols of all data points, if shown (default = 2.5).</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_all_shape">all_shape</code></td>
<td>
<p>all data points are shown with symbols of the shape number 0 (default, open square). Pick a number between 0-25 to pick a different type of symbol from ggplot2. This argument only has an effect if <code>shapes</code> argument is used.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_all_jitter">all_jitter</code></td>
<td>
<p>reduce overlap of all data points, if shown, by setting a value between 0-1 (default = 0).</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_ewid">ewid</code></td>
<td>
<p>width of error bars, default set to 0.2.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_group_wid">group_wid</code></td>
<td>
<p>space between the factors along X-axis, i.e., dodge width. Default <code>group_wid = 0.8</code> (range 0-1), which can be set to 0 if you'd like the two plotted as <code>position = position_identity()</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_ethick">ethick</code></td>
<td>
<p>thickness of error bar lines; default <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_4d_point_sd_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+stat_summary">stat_summary</a></code> or <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These can be especially useful when the fourth variable <code>shapes</code> is a random factor or blocking factor (up to 25 levels are allowed; there will be an error with more levels). The <code>shapes</code> argument can be left blank to plot ordinary 2-way ANOVAs without blocking.
</p>
<p>In <code>plot_4d_point_sd</code> and <code>plot_4d_scatterbar</code>, the default error bar is SD (can be changed to SEM or CI95). In <code>plot_4d_point_sd</code>, a large coloured symbol is plotted at the mean, all other data are shown as smaller symbols. Boxplot uses <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code> to depict median (thicker line), box (interquartile range (IQR)) and the whiskers (1.5*IQR).
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> (logical TRUE/FALSE) decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>The resulting <code>ggplot2</code> graph can take additional geometries or other layers.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#4d version for 2-way data with blocking
plot_4d_point_sd(data = data_2w_Tdeath, 
xcol = Genotype, 
ycol = PI, 
points = Time, 
shapes = Experiment)

#4d version without blocking factor
#`shapes` can be left blank
plot_4d_point_sd(data = data_2w_Festing, 
xcol = Strain, 
ycol = GST, 
points = Treatment)

</code></pre>

<hr>
<h2 id='plot_4d_scatterbar'>Plot scatter plot with bar &amp; error bars for 2-way ANOVAs with or without a blocking factor.</h2><span id='topic+plot_4d_scatterbar'></span>

<h3>Description</h3>

<p>There are 4 related functions for 2-way ANOVA type plots. In addition to a categorical variable along the X-axis, a grouping factor is passed to either <code>points</code>, <code>bars</code> or <code>boxes</code> argument in these functions. A blocking factor (or any other categorical variable) can be optionally passed to the <code>shapes</code> argument.
</p>

<ol>
<li> <p><code><a href="#topic+plot_4d_point_sd">plot_4d_point_sd</a></code> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterbar">plot_4d_scatterbar</a></code> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterbox">plot_4d_scatterbox</a></code> (box &amp; whiskers)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterviolin">plot_4d_scatterviolin</a></code> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_4d_scatterbar(
  data,
  xcol,
  ycol,
  bars,
  shapes,
  facet,
  ErrorType = "SD",
  symsize = 3,
  s_alpha = 0.8,
  b_alpha = 1,
  bwid = 0.7,
  jitter = 0.1,
  ewid = 0.2,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  group_wid = 0.8,
  symthick,
  bthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColRev = FALSE,
  ColSeq = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_4d_scatterbar_+3A_data">data</code></td>
<td>
<p>a data table, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) with the categorical factor to plot on X axis. If column is numeric, enter as <code>factor(col)</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_bars">bars</code></td>
<td>
<p>name of the column containing grouping within the factor plotted on X axis. Can be categorical or numeric X. If your table has numeric X and you want to plot as factor, enter <code>xcol = factor(name of colum)</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_shapes">shapes</code></td>
<td>
<p>name of the column (without quotes) that contains matched observations (e.g. subject IDs, experiment number) or another variable to pass on to symbol shapes. If not provided, the shapes for all groups is the same.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_errortype">ErrorType</code></td>
<td>
<p>select the type of error bars to display. Default is &quot;SD&quot; (standard deviation). Other options are &quot;SEM&quot; (standard error of the mean) and &quot;CI95&quot; (95% confidence interval based on t distributions).</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e. 80% opacity). Set <code>s_alpha = 0</code> to not show scatter plot.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes.  Default is set to 0, which results in white boxes inside violins. Change to any value &gt;0 up to 1 for different levels of transparency.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_bwid">bwid</code></td>
<td>
<p>width of boxes; default 0.7.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0.1. Increase to reduce symbol overlap, set to 0 for aligned symbols.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_ewid">ewid</code></td>
<td>
<p>width of error bars, default set to 0.2.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_group_wid">group_wid</code></td>
<td>
<p>space between the factors along X-axis, i.e., dodge width. Default <code>group_wid = 0.8</code> (range 0-1), which can be set to 0 if you'd like the two plotted as <code>position = position_identity()</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of bar and error bar lines; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbar_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+stat_summary">stat_summary</a></code> or <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These can be especially useful when the fourth variable <code>shapes</code> is a random factor or blocking factor (up to 25 levels are allowed; there will be an error with more levels). The <code>shapes</code> argument can be left blank to plot ordinary 2-way ANOVAs without blocking.
</p>
<p>In <code>plot_4d_point_sd</code> and <code>plot_4d_scatterbar</code>, the default error bar is SD (can be changed to SEM or CI95). In <code>plot_4d_point_sd</code>, a large coloured symbol is plotted at the mean, all other data are shown as smaller symbols. Boxplot uses <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code> to depict median (thicker line), box (interquartile range (IQR)) and the whiskers (1.5*IQR).
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> (logical TRUE/FALSE) decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>The resulting <code>ggplot2</code> graph can take additional geometries or other layers.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#4d version for 2-way data with blocking
plot_4d_scatterbar(data = data_2w_Tdeath, 
xcol = Genotype, 
ycol = PI, 
bars = Time, 
shapes = Experiment)

#4d version without blocking factor
#`shapes` can be left blank
plot_4d_scatterbar(data = data_2w_Tdeath, 
xcol = Genotype, 
ycol = PI, 
bars = Time)
</code></pre>

<hr>
<h2 id='plot_4d_scatterbox'>Plot scatter, box &amp; whiskers for 2-way ANOVAs with or without a blocking factor.</h2><span id='topic+plot_4d_scatterbox'></span>

<h3>Description</h3>

<p>There are 4 related functions for 2-way ANOVA type plots. In addition to a categorical variable along the X-axis, a grouping factor is passed to either <code>points</code>, <code>bars</code> or <code>boxes</code> argument in these functions. A blocking factor (or any other categorical variable) can be optionally passed to the <code>shapes</code> argument.
</p>

<ol>
<li> <p><code><a href="#topic+plot_4d_point_sd">plot_4d_point_sd</a></code> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterbar">plot_4d_scatterbar</a></code> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterbox">plot_4d_scatterbox</a></code> (box &amp; whiskers)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterviolin">plot_4d_scatterviolin</a></code> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_4d_scatterbox(
  data,
  xcol,
  ycol,
  boxes,
  shapes,
  facet,
  symsize = 3,
  s_alpha = 0.8,
  b_alpha = 1,
  bwid = 0.7,
  jitter = 0.1,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  group_wid = 0.8,
  symthick,
  bthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_4d_scatterbox_+3A_data">data</code></td>
<td>
<p>a data table, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) with the categorical factor to plot on X axis. If column is numeric, enter as <code>factor(col)</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_boxes">boxes</code></td>
<td>
<p>name of the column containing grouping within the factor plotted on X axis. Can be categorical or numeric X. If your table has numeric X and you want to plot as factor, enter <code>xcol = factor(name of colum)</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_shapes">shapes</code></td>
<td>
<p>name of the column (without quotes) that contains matched observations (e.g. subject IDs, experiment number) or another variable to pass on to symbol shapes. If not provided, the shapes for all groups is the same.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e. 80% opacity). Set <code>s_alpha = 0</code> to not show scatter plot.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes.  Default is set to 0, which results in white boxes inside violins. Change to any value &gt;0 up to 1 for different levels of transparency.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_bwid">bwid</code></td>
<td>
<p>width of boxes; default 0.7.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0.1. Increase to reduce symbol overlap, set to 0 for aligned symbols.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_group_wid">group_wid</code></td>
<td>
<p>space between the factors along X-axis, i.e., dodge width. Default <code>group_wid = 0.8</code> (range 0-1), which can be set to 0 if you'd like the two plotted as <code>position = position_identity()</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of boxplot lines; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_4d_scatterbox_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These can be especially useful when the fourth variable <code>shapes</code> is a random factor or blocking factor (up to 25 levels are allowed; there will be an error with more levels). The <code>shapes</code> argument can be left blank to plot ordinary 2-way ANOVAs without blocking.
</p>
<p>In <code>plot_4d_point_sd</code> and <code>plot_4d_scatterbar</code>, the default error bar is SD (can be changed to SEM or CI95). In <code>plot_4d_point_sd</code>, a large coloured symbol is plotted at the mean, all other data are shown as smaller symbols. Boxplot uses <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code> to depict median (thicker line), box (interquartile range (IQR)) and the whiskers (1.5*IQR).
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> (logical TRUE/FALSE) decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>The resulting <code>ggplot2</code> graph can take additional geometries or other layers.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#4d version for 2-way data with blocking
plot_4d_scatterbox(data = data_2w_Tdeath, 
xcol = Genotype, 
ycol = PI, 
boxes = Time, 
shapes = Experiment)

#4d version without blocking factor
#`shapes` can be left blank
plot_4d_scatterbox(data = data_2w_Tdeath, 
xcol = Genotype, 
ycol = PI, 
boxes = Time)
</code></pre>

<hr>
<h2 id='plot_4d_scatterviolin'>Plot scatter, box &amp; violin for 2-way ANOVAs with or without a blocking factor.</h2><span id='topic+plot_4d_scatterviolin'></span>

<h3>Description</h3>

<p>There are 4 related functions for 2-way ANOVA type plots. In addition to a categorical variable along the X-axis, a grouping factor is passed to either <code>points</code>, <code>bars</code> or <code>boxes</code> argument in these functions. A blocking factor (or any other categorical variable) can be optionally passed to the <code>shapes</code> argument.
</p>

<ol>
<li> <p><code><a href="#topic+plot_4d_point_sd">plot_4d_point_sd</a></code> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterbar">plot_4d_scatterbar</a></code> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterbox">plot_4d_scatterbox</a></code> (box &amp; whiskers)
</p>
</li>
<li> <p><code><a href="#topic+plot_4d_scatterviolin">plot_4d_scatterviolin</a></code> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_4d_scatterviolin(
  data,
  xcol,
  ycol,
  boxes,
  shapes,
  facet,
  symsize = 3,
  s_alpha = 0.8,
  v_alpha = 1,
  b_alpha = 0,
  bwid = 0.3,
  vadjust = 1,
  jitter = 0.1,
  TextXAngle = 0,
  scale = "width",
  trim = TRUE,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  group_wid = 0.8,
  symthick,
  bthick,
  vthick,
  bvthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_4d_scatterviolin_+3A_data">data</code></td>
<td>
<p>a data table, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) with the categorical factor to plot on X axis. If column is numeric, enter as <code>factor(col)</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_boxes">boxes</code></td>
<td>
<p>name of the column containing grouping within the factor plotted on X axis. Can be categorical or numeric X. If your table has numeric X and you want to plot as factor, enter <code>xcol = factor(name of colum)</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_shapes">shapes</code></td>
<td>
<p>name of the column (without quotes) that contains matched observations (e.g. subject IDs, experiment number) or another variable to pass on to symbol shapes. If not provided, the shapes for all groups is the same.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e. 80% opacity). Set <code>s_alpha = 0</code> to not show scatter plot.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_v_alpha">v_alpha</code></td>
<td>
<p>fractional opacity of violins, default set to 1.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes.  Default is set to 0, which results in white boxes inside violins. Change to any value &gt;0 up to 1 for different levels of transparency.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_bwid">bwid</code></td>
<td>
<p>width of boxes; default 0.3.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_vadjust">vadjust</code></td>
<td>
<p>number to adjust the smooth/wigglyness of violin plot (default set to 1).</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0.1. Increase to reduce symbol overlap, set to 0 for aligned symbols.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_scale">scale</code></td>
<td>
<p>set to &quot;area&quot; by default, can be changed to &quot;count&quot; or &quot;width&quot;.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_trim">trim</code></td>
<td>
<p>set whether tips of violin plot should be trimmed at high/low data. Default <code>trim = T</code>, can be changed to F.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_group_wid">group_wid</code></td>
<td>
<p>space between the factors along X-axis, i.e., dodge width. Default <code>group_wid = 0.8</code> (range 0-1), which can be set to 0 if you'd like the two plotted as <code>position = position_identity()</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of boxplots; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_vthick">vthick</code></td>
<td>
<p>thickness (in 'pt' units) of violins; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_bvthick">bvthick</code></td>
<td>
<p>thickness (in 'pt' units) of both violins and boxplots; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_4d_scatterviolin_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code> or <code><a href="ggplot2.html#topic+geom_violin">geom_violin</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These can be especially useful when the fourth variable <code>shapes</code> is a random factor or blocking factor (up to 25 levels are allowed; there will be an error with more levels). The <code>shapes</code> argument can be left blank to plot ordinary 2-way ANOVAs without blocking.
</p>
<p>In <code>plot_4d_point_sd</code> and <code>plot_4d_scatterbar</code>, the default error bar is SD (can be changed to SEM or CI95). In <code>plot_4d_point_sd</code>, a large coloured symbol is plotted at the mean, all other data are shown as smaller symbols. Boxplot uses <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code> to depict median (thicker line), box (interquartile range (IQR)) and the whiskers (1.5*IQR).
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> (logical TRUE/FALSE) decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>The resulting <code>ggplot2</code> graph can take additional geometries or other layers.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#4d version for 2-way data with blocking
plot_4d_scatterviolin(data = data_2w_Tdeath, 
xcol = Genotype, 
ycol = PI, 
boxes = Time, 
shapes = Experiment)

#4d version without blocking factor
#`shapes` can be left blank
plot_4d_scatterviolin(data = data_2w_Tdeath, 
xcol = Genotype, 
ycol = PI, 
boxes = Time)

</code></pre>

<hr>
<h2 id='plot_befafter_box'>Before-after style graph with a boxplot</h2><span id='topic+plot_befafter_box'></span>

<h3>Description</h3>

<p>One of 3 related functions to plot matching data joined by lines. The variable containing information for matching (e.g. matched subjects or experiments etc.) is passed to the <code>match</code> argument.
</p>

<ol>
<li> <p><code><a href="#topic+plot_befafter_colours">plot_befafter_colours</a></code> or <code><a href="#topic+plot_befafter_colors">plot_befafter_colors</a></code>,
</p>
</li>
<li> <p><code><a href="#topic+plot_befafter_shapes">plot_befafter_shapes</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot_befafter_box">plot_befafter_box</a></code>
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_befafter_box(
  data,
  xcol,
  ycol,
  match,
  facet,
  PlotShapes = FALSE,
  symsize = 3,
  s_alpha = 0.8,
  b_alpha = 1,
  bwid = 0.4,
  jitter = 0.1,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  bthick,
  lthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_befafter_box_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) containing the categorical variable to be plotted on the X axis.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_match">match</code></td>
<td>
<p>name of the column (without quotes) with the grouping variable to pass on to <code><a href="ggplot2.html#topic+geom_line">geom_line</a></code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_plotshapes">PlotShapes</code></td>
<td>
<p>logical TRUE or FALSE (default = FALSE) if the shape of the symbol is to be mapped to the <code>match</code> variable. Note that only 25 shapes allowed.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e., 80% opacity).</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes, default set to 1.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_bwid">bwid</code></td>
<td>
<p>width of boxplots; default 0.4.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0.1. Increase to reduce symbol overlap, set to 0 for aligned symbols.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of boxes; default = <code>(fontsize)/22</code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_lthick">lthick</code></td>
<td>
<p>thickness (in 'pt' units) of lines; default = <code>(fontsize/1.2)/22</code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> or base R palettes to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_befafter_box_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_line">geom_line</a></code>,  <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>, or <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In <code>plot_befafter_colours</code>/<code>plot_befafter_colors</code> and <code>plot_befafter_shapes</code> setting <code>Boxplot = TRUE</code> will also plot a box and whiskers plot.
</p>
<p>Note that only 25 shapes are available, and there will be errors with <code><a href="#topic+plot_befafter_shapes">plot_befafter_shapes</a></code> when there are fewer than 25 matched observations; instead use <code><a href="#topic+plot_befafter_colours">plot_befafter_colours</a></code> instead.
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>To plot a graph with a single colour along the X axis variable, use the <code>SingleColour</code> argument.
</p>
<p>The resulting <code>ggplot2</code> graph can take additional geometries or other layers.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_befafter_box(data = data_t_pdiff, 
xcol = Condition, ycol = Mass, 
match = Subject)

#with PlotShapes = TRUE
plot_befafter_box(data = data_t_pdiff, 
xcol = Condition, ycol = Mass, 
match = Subject, PlotShapes = TRUE)
</code></pre>

<hr>
<h2 id='plot_befafter_colours'>Plot a before-after plot with lines joining colour-matched symbols.</h2><span id='topic+plot_befafter_colours'></span><span id='topic+plot_befafter_colors'></span>

<h3>Description</h3>

<p>One of 3 related functions to plot matching data joined by lines. The variable containing information for matching (e.g. matched subjects or experiments etc.) is passed to the <code>match</code> argument.
</p>

<ol>
<li> <p><code><a href="#topic+plot_befafter_colours">plot_befafter_colours</a></code> or <code><a href="#topic+plot_befafter_colors">plot_befafter_colors</a></code>,
</p>
</li>
<li> <p><code><a href="#topic+plot_befafter_shapes">plot_befafter_shapes</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot_befafter_box">plot_befafter_box</a></code>
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_befafter_colours(
  data,
  xcol,
  ycol,
  match,
  facet,
  Boxplot = FALSE,
  symsize = 3,
  s_alpha = 1,
  jitter = 0.1,
  bwid = 0.4,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  bthick,
  lthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)

plot_befafter_colors(
  data,
  xcol,
  ycol,
  match,
  facet,
  Boxplot = FALSE,
  symsize = 3,
  s_alpha = 1,
  jitter = 0.1,
  bwid = 0.4,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  bthick,
  lthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_befafter_colours_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) containing the categorical variable to be plotted on the X axis.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_match">match</code></td>
<td>
<p>name of the column (without quotes) with the matching variable to pass on to <code><a href="ggplot2.html#topic+geom_line">geom_line</a></code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_boxplot">Boxplot</code></td>
<td>
<p>logical TRUE/FALSE, whether to show box and whisker plot or not (default is FALSE)</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 1 (i.e. maximum opacity &amp; zero transparency).</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0.1. Increase to reduce symbol overlap, set to 0 for aligned symbols.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_bwid">bwid</code></td>
<td>
<p>width of boxplots; default 0.4.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of boxes; default = <code>(fontsize)/22</code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_lthick">lthick</code></td>
<td>
<p>thickness (in 'pt' units) of lines; default = <code>(fontsize/1.2)/22</code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> or base R palettes to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_befafter_colours_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_line">geom_line</a></code> or <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In <code>plot_befafter_colours</code>/<code>plot_befafter_colors</code> and <code>plot_befafter_shapes</code> setting <code>Boxplot = TRUE</code> will also plot a box and whiskers plot.
</p>
<p>Note that only 25 shapes are available, and there will be errors with <code><a href="#topic+plot_befafter_shapes">plot_befafter_shapes</a></code> when there are fewer than 25 matched observations; instead use <code><a href="#topic+plot_befafter_colours">plot_befafter_colours</a></code> instead.
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>To plot a graph with a single colour along the X axis variable, use the <code>SingleColour</code> argument.
</p>
<p>The resulting <code>ggplot2</code> graph can take additional geometries or other layers.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#plot without legends if necessary
plot_befafter_colours(data = data_t_pdiff, 
xcol = Condition, ycol = Mass, 
match = Subject, s_alpha = .9, ColSeq = FALSE)+
guides(fill = "none", 
colour = "none") #remove guides
#plot with boxplot
plot_befafter_colours(data = data_t_pdiff, 
xcol = Condition, ycol = Mass, 
match = Subject, s_alpha = .9, ColSeq = FALSE,
Boxplot = TRUE)+
guides(fill = "none", 
colour = "none") #remove guides

</code></pre>

<hr>
<h2 id='plot_befafter_shapes'>Plot a before-after plot with lines joining shape-matched symbols.</h2><span id='topic+plot_befafter_shapes'></span>

<h3>Description</h3>

<p>One of 3 related functions to plot matching data joined by lines. The variable containing information for matching (e.g. matched subjects or experiments etc.) is passed to the <code>match</code> argument.
</p>

<ol>
<li> <p><code><a href="#topic+plot_befafter_colours">plot_befafter_colours</a></code> or <code><a href="#topic+plot_befafter_colors">plot_befafter_colors</a></code>,
</p>
</li>
<li> <p><code><a href="#topic+plot_befafter_shapes">plot_befafter_shapes</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot_befafter_box">plot_befafter_box</a></code>
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_befafter_shapes(
  data,
  xcol,
  ycol,
  match,
  facet,
  Boxplot = FALSE,
  symsize = 3,
  s_alpha = 1,
  bwid = 0.4,
  jitter = 0.1,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  bthick,
  lthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_befafter_shapes_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) containing the categorical variable to be plotted on the X axis.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_match">match</code></td>
<td>
<p>name of the column (without quotes) with the matching variable to pass on to <code><a href="ggplot2.html#topic+geom_line">geom_line</a></code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_boxplot">Boxplot</code></td>
<td>
<p>logical TRUE/FALSE, whether to show box and whisker plot or not (default is FALSE)</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 1.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_bwid">bwid</code></td>
<td>
<p>width of boxplots; default 0.4.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0.1. Increase to reduce symbol overlap, set to 0 for aligned symbols.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of boxes; default = <code>(fontsize)/22</code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_lthick">lthick</code></td>
<td>
<p>thickness (in 'pt' units) of lines; default = <code>(fontsize/1.2)/22</code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> or base R palettes to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_befafter_shapes_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_line">geom_line</a></code> or <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In <code>plot_befafter_colours</code>/<code>plot_befafter_colors</code> and <code>plot_befafter_shapes</code> setting <code>Boxplot = TRUE</code> will also plot a box and whiskers plot.
</p>
<p>Note that only 25 shapes are available, and there will be errors with <code><a href="#topic+plot_befafter_shapes">plot_befafter_shapes</a></code> when there are fewer than 25 matched observations; instead use <code><a href="#topic+plot_befafter_colours">plot_befafter_colours</a></code> instead.
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>To plot a graph with a single colour along the X axis variable, use the <code>SingleColour</code> argument.
</p>
<p>The resulting <code>ggplot2</code> graph can take additional geometries or other layers.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#plot without legends if necessary
plot_befafter_colors(data = data_t_pdiff, 
xcol = Condition, ycol = Mass, 
match = Subject, s_alpha = .9, ColSeq = FALSE)+
guides(fill = "none", 
colour = "none") #remove guides
#plot with boxplot
plot_befafter_colors(data = data_t_pdiff, 
xcol = Condition, ycol = Mass, 
match = Subject, s_alpha = .9, ColSeq = FALSE,
Boxplot = TRUE)+
guides(fill = "none", 
colour = "none") #remove guides

</code></pre>

<hr>
<h2 id='plot_density'>Plot density distribution of data.</h2><span id='topic+plot_density'></span>

<h3>Description</h3>

<p>This function takes a data table, <code>ycol</code> of quantitative variable and a categorical grouping variable (<code>group</code>), if available, and plots a density graph using <code><a href="ggplot2.html#topic+geom_density">geom_density</a></code>). Alternatives are <code><a href="#topic+plot_histogram">plot_histogram</a></code>, or <code><a href="#topic+plot_qqline">plot_qqline</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_density(
  data,
  ycol,
  group,
  facet,
  PlotType = c("Density", "Counts", "Normalised counts"),
  c_alpha = 0.2,
  TextXAngle = 0,
  facet_scales = "fixed",
  fontsize = 20,
  linethick,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_density_+3A_data">data</code></td>
<td>
<p>a data table e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable whose density distribution is to be plotted.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_group">group</code></td>
<td>
<p>name of the column containing a categorical grouping variable (optional Since v5.0.0).</p>
</td></tr>
<tr><td><code id="plot_density_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_plottype">PlotType</code></td>
<td>
<p>the default (<code>Density</code>) plot will be the probability density curve, which can be changed to <code>Counts</code> or <code style="white-space: pre;">&#8288;Normalised counts&#8288;</code>.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_c_alpha">c_alpha</code></td>
<td>
<p>fractional opacity of filled colours under the curve, default set to 0.2 (i.e. 20% opacity).</p>
</td></tr>
<tr><td><code id="plot_density_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_density_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_linethick">linethick</code></td>
<td>
<p>thickness of symbol border, default set to <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_density_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_density_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> palettes or base R to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_density_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_density">geom_density</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the function requires the quantitative Y variable first, and groups them based on an X variable. The group variable is mapped to the <code>fill</code> and <code>colour</code> aesthetics in <code><a href="ggplot2.html#topic+geom_density">geom_density</a></code>.
Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments. Colours available can be seen quickly with <code><a href="#topic+plot_grafify_palette">plot_grafify_palette</a></code>.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_density(data = data_t_pratio, 
ycol = log(Cytokine), group = Genotype)

#with faceting
plot_density(data = data_cholesterol, 
ycol = Cholesterol, group = Treatment, 
facet = Treatment, fontsize = 12)

#Counts
plot_density(data = data_cholesterol, 
ycol = Cholesterol, group = Treatment,
PlotType = "Counts", 
facet = Treatment, fontsize = 12)

</code></pre>

<hr>
<h2 id='plot_dotbar_sd'>Plot a dotplot on a bar graph with SD error bars with two variables.</h2><span id='topic+plot_dotbar_sd'></span>

<h3>Description</h3>

<p>There are three types of <code>plot_dot_</code> functions that plot data as &quot;dots&quot; using the <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code> geometry. They all take a data table, a categorical X variable and a numeric Y variable.
</p>

<ol>
<li> <p><a href="#topic+plot_dotbar_sd">plot_dotbar_sd</a> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><a href="#topic+plot_dotbox">plot_dotbox</a> (box &amp; whiskers)
</p>
</li>
<li> <p><a href="#topic+plot_dotviolin">plot_dotviolin</a> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_dotbar_sd(
  data,
  xcol,
  ycol,
  facet,
  ErrorType = "SD",
  dotsize = 1.5,
  d_alpha = 0.8,
  b_alpha = 1,
  bwid = 0.5,
  ewid = 0.2,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  dotthick,
  bthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_dotbar_sd_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) to plot on X axis. This should be a categorical variable.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis. This should be a quantitative variable.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_errortype">ErrorType</code></td>
<td>
<p>select the type of error bars to display. Default is &quot;SD&quot; (standard deviation). Other options are &quot;SEM&quot; (standard error of the mean) and &quot;CI95&quot; (95% confidence interval based on t distributions).</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_dotsize">dotsize</code></td>
<td>
<p>size of dots relative to binwidth used by <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code>. Default set to 1.5, increase/decrease as needed.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_d_alpha">d_alpha</code></td>
<td>
<p>fractional opacity of dots, default set to 0.8 (i.e., 80% opacity).</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of bars, default set to 1.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_bwid">bwid</code></td>
<td>
<p>width of bars; default 0.5.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_ewid">ewid</code></td>
<td>
<p>width of error bars, default set to 0.2.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_dotthick">dotthick</code></td>
<td>
<p>thickness of dot border (<code>stroke</code> parameter of <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code>), default set to <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of bar and error bar lines; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> or base R palettes to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_dotbar_sd_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Related <code>plot_scatter_</code> variants show data symbols using the <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> geometry. These are <a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a> (or SEM or CI95 error bars), <a href="#topic+plot_scatterbox">plot_scatterbox</a> and <a href="#topic+plot_scatterviolin">plot_scatterviolin</a>. Over plotting in <code>plot_scatter</code> variants can be reduced with the <code>jitter</code> argument.
</p>
<p>The X variable is mapped to the <code>fill</code> aesthetic of dots, symbols, bars, boxes and violins.
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments. Colours available can be seen quickly with <code><a href="#topic+plot_grafify_palette">plot_grafify_palette</a></code>.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>If you prefer a single colour for the graph, use the <code>SingleColour</code> argument.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_dotbar_sd(data = data_cholesterol, 
xcol = Treatment,
ycol = Cholesterol)

plot_dotbar_sd(data = data_1w_death, 
xcol = Genotype, ycol = Death, 
ColPal = "pale", ColSeq = FALSE, ColRev = TRUE)

#single colour along X
plot_dotbar_sd(data = data_1w_death, 
xcol = Genotype, ycol = Death, 
SingleColour = "light_orange")
</code></pre>

<hr>
<h2 id='plot_dotbox'>Plot a dotplot on a boxplot with two variables.</h2><span id='topic+plot_dotbox'></span>

<h3>Description</h3>

<p>There are three types of <code>plot_dot_</code> functions that plot data as &quot;dots&quot; using the <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code> geometry. They all take a data table, a categorical X variable and a numeric Y variable.
</p>

<ol>
<li> <p><a href="#topic+plot_dotbar_sd">plot_dotbar_sd</a> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><a href="#topic+plot_dotbox">plot_dotbox</a> (box &amp; whiskers)
</p>
</li>
<li> <p><a href="#topic+plot_dotviolin">plot_dotviolin</a> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_dotbox(
  data,
  xcol,
  ycol,
  facet,
  dotsize = 1.5,
  d_alpha = 0.8,
  b_alpha = 1,
  bwid = 0.5,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  dotthick,
  bthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_dotbox_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) to plot on X axis. This should be a categorical variable.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis. This should be a quantitative variable.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_dotsize">dotsize</code></td>
<td>
<p>size of dots relative to binwidth used by <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code>. Default set to 1.5, increase/decrease as needed.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_d_alpha">d_alpha</code></td>
<td>
<p>fractional opacity of dots, default set to 0.8 (i.e., 80% opacity).</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes, default set to 1.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_bwid">bwid</code></td>
<td>
<p>width of boxplots; default 0.5.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_dotthick">dotthick</code></td>
<td>
<p>thickness of dot border (<code>stroke</code> parameter of <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code>), default set to <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of boxplot lines; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> or base R palettes to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_dotbox_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code> or <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Related <code>plot_scatter_</code> variants show data symbols using the <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> geometry. These are <a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a> (or SEM or CI95 error bars), <a href="#topic+plot_scatterbox">plot_scatterbox</a> and <a href="#topic+plot_scatterviolin">plot_scatterviolin</a>. Over plotting in <code>plot_scatter</code> variants can be reduced with the <code>jitter</code> argument.
</p>
<p>The X variable is mapped to the <code>fill</code> aesthetic of dots, symbols, bars, boxes and violins.
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments. Colours available can be seen quickly with <code><a href="#topic+plot_grafify_palette">plot_grafify_palette</a></code>.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>If you prefer a single colour for the graph, use the <code>SingleColour</code> argument.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_dotbox(data = data_1w_death, 
xcol = Genotype, ycol = Death)

plot_dotbox(data = data_1w_death, 
xcol = Genotype, ycol = Death, 
ColPal = "vibrant", b_alpha = 0.5)
plot_dotbox(data = data_1w_death, 
xcol = Genotype, ycol = Death, 
SingleColour = "safe_bluegreen", b_alpha = 0.5) 
</code></pre>

<hr>
<h2 id='plot_dotviolin'>Plot a dotplot on a violin plot with two variables.</h2><span id='topic+plot_dotviolin'></span>

<h3>Description</h3>

<p>There are three types of <code>plot_dot_</code> functions that plot data as &quot;dots&quot; using the <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code> geometry. They all take a data table, a categorical X variable and a numeric Y variable.
</p>

<ol>
<li> <p><a href="#topic+plot_dotbar_sd">plot_dotbar_sd</a> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><a href="#topic+plot_dotbox">plot_dotbox</a> (box &amp; whiskers)
</p>
</li>
<li> <p><a href="#topic+plot_dotviolin">plot_dotviolin</a> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_dotviolin(
  data,
  xcol,
  ycol,
  facet,
  dotsize = 1.5,
  d_alpha = 0.8,
  b_alpha = 0,
  v_alpha = 1,
  bwid = 0.3,
  vadjust = 1,
  trim = TRUE,
  scale = "width",
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  dotthick,
  bthick,
  vthick,
  bvthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_dotviolin_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) to plot on X axis. This should be a categorical variable.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis. This should be a quantitative variable.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_dotsize">dotsize</code></td>
<td>
<p>size of dots relative to <code>binwidth</code> used by <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code>. Default set to 1.5, increase/decrease as needed.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_d_alpha">d_alpha</code></td>
<td>
<p>fractional opacity of dots, default set to 0.8 (i.e., 80% opacity).</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxplots.  Default is set to 0, which results in white boxes inside violins. Change to any value &gt;0 up to 1 for different levels of transparency.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_v_alpha">v_alpha</code></td>
<td>
<p>fractional opacity of violins, default set to 1.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_bwid">bwid</code></td>
<td>
<p>width of boxplots; default 0.3.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_vadjust">vadjust</code></td>
<td>
<p>number to adjust the smooth/wigglyness of violin plot (default set to 1).</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_trim">trim</code></td>
<td>
<p>set whether tips of violin plot should be trimmed at high/low data. Default <code>trim = TRUE</code>, can be changed to FALSE.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_scale">scale</code></td>
<td>
<p>set to &quot;area&quot; by default, can be changed to &quot;count&quot; or &quot;width&quot;.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_dotthick">dotthick</code></td>
<td>
<p>thickness of dot border (<code>stroke</code> parameter of <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code>), default set to <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of boxplots; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_vthick">vthick</code></td>
<td>
<p>thickness (in 'pt' units) of violins; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_bvthick">bvthick</code></td>
<td>
<p>thickness (in 'pt' units) of both violins and boxplots; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> or base R palettes to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_dotviolin_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code>, <code><a href="ggplot2.html#topic+geom_dotplot">geom_dotplot</a></code> or <code><a href="ggplot2.html#topic+geom_violin">geom_violin</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Related <code>plot_scatter_</code> variants show data symbols using the <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> geometry. These are <a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a> (or SEM or CI95 error bars), <a href="#topic+plot_scatterbox">plot_scatterbox</a> and <a href="#topic+plot_scatterviolin">plot_scatterviolin</a>. Over plotting in <code>plot_scatter</code> variants can be reduced with the <code>jitter</code> argument.
</p>
<p>The X variable is mapped to the <code>fill</code> aesthetic of dots, symbols, bars, boxes and violins.
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments. Colours available can be seen quickly with <code><a href="#topic+plot_grafify_palette">plot_grafify_palette</a></code>.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>If you prefer a single colour for the graph, use the <code>SingleColour</code> argument.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#plot with trim = FALSE
plot_dotviolin(data = data_t_pdiff, 
xcol = Condition, ycol = Mass, 
dotsize = 2, trim = FALSE)

plot_dotviolin(data = data_t_pdiff, 
xcol = Condition, ycol = Mass,
trim = FALSE, b_alpha = 0.5, 
ColPal = "pale", ColSeq = FALSE)

#single colour along X
plot_dotviolin(data = data_t_pdiff, 
xcol = Condition, ycol = Mass,
trim = FALSE, b_alpha = 0.5, 
SingleColour = "pale_cyan")
</code></pre>

<hr>
<h2 id='plot_gam_predict'>Plot prediction of <code>gam</code> model</h2><span id='topic+plot_gam_predict'></span>

<h3>Description</h3>

<p>Plot prediction of <code>gam</code> model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gam_predict(
  Model,
  xcol,
  ycol,
  ByFactor,
  symsize = 1,
  s_alpha = 0.1,
  smooth_alpha = 0.7,
  linethick,
  fontsize = 20,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_gam_predict_+3A_model">Model</code></td>
<td>
<p>a generalised additive model (<code>gam</code>) fitted with <code>ga_model</code> or <code>mgcv</code></p>
</td></tr>
<tr><td><code id="plot_gam_predict_+3A_xcol">xcol</code></td>
<td>
<p>the smooth in the <code>gam</code> (should match variable in the model exactly)</p>
</td></tr>
<tr><td><code id="plot_gam_predict_+3A_ycol">ycol</code></td>
<td>
<p>the dependent variable in <code>gam</code> (should match variable in the model exactly)</p>
</td></tr>
<tr><td><code id="plot_gam_predict_+3A_byfactor">ByFactor</code></td>
<td>
<p>the <code>by</code> factor used in <code>gam</code> (should match variable in the model exactly)</p>
</td></tr>
<tr><td><code id="plot_gam_predict_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols (default = 1)</p>
</td></tr>
<tr><td><code id="plot_gam_predict_+3A_s_alpha">s_alpha</code></td>
<td>
<p>opacity of symbols (default = 0.1)</p>
</td></tr>
<tr><td><code id="plot_gam_predict_+3A_smooth_alpha">smooth_alpha</code></td>
<td>
<p>opacity of the predicted CI interval (default = 0.7)</p>
</td></tr>
<tr><td><code id="plot_gam_predict_+3A_linethick">linethick</code></td>
<td>
<p>thickness of symbol lines (default = fontsize/22)</p>
</td></tr>
<tr><td><code id="plot_gam_predict_+3A_fontsize">fontsize</code></td>
<td>
<p>base font size for graph</p>
</td></tr>
<tr><td><code id="plot_gam_predict_+3A_...">...</code></td>
<td>
<p>additional arguments to pass to <code><a href="#topic+plot_xy_CatGroup">plot_xy_CatGroup</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#fit zooplankton data
z1 &lt;- ga_model(data = data_zooplankton,
Y_value = "log(density_adj)",
Fixed_Factor = "taxon",
Smooth_Factor = "day")

#plot fitted data
plot_gam_predict(Model = z1,
xcol = day, 
ycol = `log(density_adj)`,
ByFactor = taxon)


</code></pre>

<hr>
<h2 id='plot_grafify_palette'>See grafify colour palettes</h2><span id='topic+plot_grafify_palette'></span>

<h3>Description</h3>

<p>This simple function allows quick visualisation of colours in grafify palettes and their hex codes. It uses <code>plot_scatterbar_sd</code> and some arguments are similar and can be adjusted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_grafify_palette(
  palette = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant", "OrBl_div", "PrGn_div",
    "blue_conti", "grey_conti", "yellow_conti"),
  fontsize = 14,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_grafify_palette_+3A_palette">palette</code></td>
<td>
<p>name of grafify palettes: &quot;okabe_ito&quot;, &quot;vibrant, &quot;bright&quot;, &quot;pale&quot;, &quot;muted&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;contrast&quot; or &quot;all_grafify&quot;.</p>
</td></tr>
<tr><td><code id="plot_grafify_palette_+3A_fontsize">fontsize</code></td>
<td>
<p>font size.</p>
</td></tr>
<tr><td><code id="plot_grafify_palette_+3A_...">...</code></td>
<td>
<p>any additional parameters to pass to <code><a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_grafify_palette("pale")
plot_grafify_palette("contrast")

</code></pre>

<hr>
<h2 id='plot_histogram'>Plot data distribution as histograms.</h2><span id='topic+plot_histogram'></span>

<h3>Description</h3>

<p>This function takes a data table, a quantitative variable (<code>ycol</code>) and a grouping variable (<code>group</code>), if available, and plots a histogram graph using <code><a href="ggplot2.html#topic+geom_histogram">geom_histogram</a></code>).  Alternatives are <code><a href="#topic+plot_density">plot_density</a></code>, or <code><a href="#topic+plot_qqline">plot_qqline</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_histogram(
  data,
  ycol,
  group,
  facet,
  PlotType = c("Counts", "Normalised counts"),
  BinSize = 30,
  c_alpha = 0.8,
  TextXAngle = 0,
  facet_scales = "fixed",
  fontsize = 20,
  linethick,
  alpha,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_histogram_+3A_data">data</code></td>
<td>
<p>a data table e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable whose histogram distribution is to be plotted.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_group">group</code></td>
<td>
<p>name of the column containing a categorical grouping variable.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_plottype">PlotType</code></td>
<td>
<p>the default (<code>Counts</code>) plot will plot counts in the bins, which can be changed to <code style="white-space: pre;">&#8288;Normalised counts&#8288;</code>.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_binsize">BinSize</code></td>
<td>
<p>number of distinct bins to use on X-axis, default set to 30.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_c_alpha">c_alpha</code></td>
<td>
<p>fractional opacity of colour filled within histograms, default set to 0.8 (i.e. 80% opacity).</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_linethick">linethick</code></td>
<td>
<p>thickness of symbol border, default set to <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_alpha">alpha</code></td>
<td>
<p>deprecated old argument for <code>c_alpha</code>; retained for backward compatibility.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> palettes or base R to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_histogram_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_histogram">geom_histogram</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the function requires the quantitative Y variable first, and groups them based on a categorical variable passed on via the <code>group</code> argument. The grouping variable is mapped to the <code>fill</code> aesthetics in <code><a href="ggplot2.html#topic+geom_histogram">geom_histogram</a></code>.
</p>
<p>ColPal &amp; ColRev options are applied to both <code>fill</code> and <code>colour</code> scales. Colours available can be seen quickly with <code><a href="#topic+plot_grafify_palette">plot_grafify_palette</a></code>.
Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Basic usage
plot_histogram(data = data_t_pratio, 
ycol = Cytokine, group = Genotype, 
BinSize = 10)
#with log transformation
plot_histogram(data = data_t_pratio, 
ycol = log(Cytokine), group = Genotype, 
BinSize = 10)
#Normalised counts
plot_histogram(data = data_t_pratio, 
ycol = log(Cytokine), group = Genotype, 
PlotType = "Normalised counts", 
BinSize = 10)
</code></pre>

<hr>
<h2 id='plot_lm_predict'>Plot data and predictions from linear model</h2><span id='topic+plot_lm_predict'></span>

<h3>Description</h3>

<p>This function takes a linear model, and up to three variables and plots observe data (circles) and model predictions (squares). If the X-variable is categorical, a box and whiskers plot is overlaid. A variable (<code>ByFactor</code>) can be used for faceting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_lm_predict(
  Model,
  xcol,
  ycol,
  ByFactor,
  obs_size = 2,
  obs_alpha = 0.3,
  pred_size = 2,
  pred_alpha = 0.8,
  linethick,
  base_size = 15,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_lm_predict_+3A_model">Model</code></td>
<td>
<p>a linear model saved with <code>simple_model</code>, <code>mixed_model</code> or <code>ga_model</code>.</p>
</td></tr>
<tr><td><code id="plot_lm_predict_+3A_xcol">xcol</code></td>
<td>
<p>variable along the X axis (should match one of the dependent variables in model exactly).</p>
</td></tr>
<tr><td><code id="plot_lm_predict_+3A_ycol">ycol</code></td>
<td>
<p>independent variable along the Y axis (should match independent variable in model exactly).</p>
</td></tr>
<tr><td><code id="plot_lm_predict_+3A_byfactor">ByFactor</code></td>
<td>
<p>optional faceting variable (should match one of the variables in model exactly).</p>
</td></tr>
<tr><td><code id="plot_lm_predict_+3A_obs_size">obs_size</code></td>
<td>
<p>size of symbols for observed data (default = 2).</p>
</td></tr>
<tr><td><code id="plot_lm_predict_+3A_obs_alpha">obs_alpha</code></td>
<td>
<p>opacity of symbols for observed data (default = 0.3).</p>
</td></tr>
<tr><td><code id="plot_lm_predict_+3A_pred_size">pred_size</code></td>
<td>
<p>size of symbols for predicted data (default = 2).</p>
</td></tr>
<tr><td><code id="plot_lm_predict_+3A_pred_alpha">pred_alpha</code></td>
<td>
<p>opacity of symbols for predicted data (default = 0.8).</p>
</td></tr>
<tr><td><code id="plot_lm_predict_+3A_linethick">linethick</code></td>
<td>
<p>thickness of border lines for boxes and symbols (default is base_size/20).</p>
</td></tr>
<tr><td><code id="plot_lm_predict_+3A_base_size">base_size</code></td>
<td>
<p>base fontsize for <code>theme_grafify</code></p>
</td></tr>
<tr><td><code id="plot_lm_predict_+3A_...">...</code></td>
<td>
<p>any other parameters to be passed to <code>theme_grafify</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#fit a model
deathm1 &lt;- mixed_model(data_2w_Tdeath, 
"PI", c("Genotype", "Time"), 
"Experiment")
#plot model
plot_lm_predict(deathm1,
Genotype, PI, Time)
#fit zooplankton data
z1 &lt;- ga_model(data = data_zooplankton,
Y_value = "log(density_adj)",
Fixed_Factor = "taxon",
Smooth_Factor = "day")

#plot fitted data
plot_lm_predict(Model = z1,
xcol = day, 
ycol = `log(density_adj)`,
ByFactor = taxon)

</code></pre>

<hr>
<h2 id='plot_logscale'>Add log transformations to graphs</h2><span id='topic+plot_logscale'></span>

<h3>Description</h3>

<p>This function allows &quot;log10&quot; or &quot;log2&quot; transformation of X or Y axes. With &quot;log10&quot; transformation, log10 ticks are also added on the outside.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_logscale(
  Plot,
  LogYTrans = "log10",
  LogXTrans,
  LogYBreaks = waiver(),
  LogXBreaks = waiver(),
  LogYLimits = NULL,
  LogXLimits = NULL,
  LogYLabels = waiver(),
  LogXLabels = waiver(),
  fontsize = 22,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_logscale_+3A_plot">Plot</code></td>
<td>
<p>a ggplot2 object.</p>
</td></tr>
<tr><td><code id="plot_logscale_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; (default) or &quot;log2&quot;</p>
</td></tr>
<tr><td><code id="plot_logscale_+3A_logxtrans">LogXTrans</code></td>
<td>
<p>transform X axis into &quot;log10&quot; or &quot;log2&quot;</p>
</td></tr>
<tr><td><code id="plot_logscale_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_logscale_+3A_logxbreaks">LogXBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_x_continuous">scale_x_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_logscale_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_logscale_+3A_logxlimits">LogXLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the X axis.</p>
</td></tr>
<tr><td><code id="plot_logscale_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_logscale_+3A_logxlabels">LogXLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_x_continuous">scale_x_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_logscale_+3A_fontsize">fontsize</code></td>
<td>
<p>this parameter sets the linewidth of the <code>log10</code> tickmarks (<code>8*fontsize/22</code> for long ticks and <code>4*fontsize/22</code> for middle ticks). It is set to 20 as default to be consistent with rest of <code>grafify</code>. It will need to be changed to 12, which is the default fontsize for graphs produced natively with <code>ggplot2</code>.</p>
</td></tr>
<tr><td><code id="plot_logscale_+3A_...">...</code></td>
<td>
<p>any other arguments to pass to <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> or <code><a href="ggplot2.html#topic+scale_x_continuous">scale_x_continuous</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Arguments allow for axes limits, breaks and labels to passed on.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#save a ggplot object
P &lt;- ggplot(data_t_pratio, 
aes(Genotype,Cytokine))+
geom_jitter(shape = 21, 
size = 5, width = .2, 
aes(fill = Genotype), 
alpha = .7)
#transform Y axis
plot_logscale(Plot = P)

#or in one go 
plot_logscale(ggplot(data_t_pratio, 
aes(Genotype,Cytokine))+
geom_jitter(shape = 21, 
size = 5, width = .2, 
aes(fill = Genotype), 
alpha = .7))

</code></pre>

<hr>
<h2 id='plot_point_sd'>Plot a point as mean with SD error bars using two variables.</h2><span id='topic+plot_point_sd'></span>

<h3>Description</h3>

<p>There are 4 related functions that use <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> to plot a categorical variable along the X axis.
</p>

<ol>
<li> <p><a href="#topic+plot_point_sd">plot_point_sd</a> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><a href="#topic+plot_scatterbox">plot_scatterbox</a> (box &amp; whiskers)
</p>
</li>
<li> <p><a href="#topic+plot_scatterviolin">plot_scatterviolin</a> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_point_sd(
  data,
  xcol,
  ycol,
  facet,
  ErrorType = "SD",
  symsize = 3.5,
  s_alpha = 1,
  symshape = 22,
  all_alpha = 0.3,
  all_size = 2.5,
  all_shape = 1,
  all_jitter = 0,
  ewid = 0.2,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  ethick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_point_sd_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) with a X variable (will be forced to be a factor/categorical variable).</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with quantitative Y variable.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_errortype">ErrorType</code></td>
<td>
<p>select the type of error bars to display. Default is &quot;SD&quot; (standard deviation). Other options are &quot;SEM&quot; (standard error of the mean) and &quot;CI95&quot; (95% confidence interval based on t distributions).</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_symsize">symsize</code></td>
<td>
<p>size of point symbols, default set to 3.5.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 1 (i.e. maximum opacity &amp; zero transparency).</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_symshape">symshape</code></td>
<td>
<p>The mean is shown with symbol of the shape number 22 (default, filled square). Pick a number between 21-25 to pick a different type of symbol from ggplot2.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_all_alpha">all_alpha</code></td>
<td>
<p>fractional opacity of all data points (default = 0.3). Set to non-zero value if you would like all data points plotted in addition to the mean.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_all_size">all_size</code></td>
<td>
<p>size of symbols of all data points, if shown (default = 2.5).</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_all_shape">all_shape</code></td>
<td>
<p>all data points are shown with symbols of the shape number 1 (default, transparent circle). Pick a number between 0-25 to pick a different type of symbol from ggplot2.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_all_jitter">all_jitter</code></td>
<td>
<p>reduce overlap of all data points, if shown, by setting a value between 0-1 (default = 0).</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_ewid">ewid</code></td>
<td>
<p>width of error bars, default set to 0.2.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_symthick">symthick</code></td>
<td>
<p>thickness of symbol border, default set to <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_ethick">ethick</code></td>
<td>
<p>thickness of error bar lines; default <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> or base R palettes to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_point_sd_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+stat_summary">stat_summary</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions take a data table, categorical X and numeric Y variables, and plot various geometries. The X variable is mapped to the <code>fill</code> aesthetic of symbols.
</p>
<p>In <a href="#topic+plot_point_sd">plot_point_sd</a> and <a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a>, default error bars are SD, which can be changed to SEM or CI95.
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments. Colours available can be seen quickly with <code><a href="#topic+plot_grafify_palette">plot_grafify_palette</a></code>.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> (logical TRUE/FALSE) decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>If there are many groups along the X axis and you prefer a single colour for the graph,use the <code>SingleColour</code> argument.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Basic usage
plot_point_sd(data = data_doubling_time, 
xcol = Student, ycol = Doubling_time)

</code></pre>

<hr>
<h2 id='plot_qq_gam'>Plot model diagnostics for generalised additive models</h2><span id='topic+plot_qq_gam'></span>

<h3>Description</h3>

<p>This is a clone of the <code><a href="gratia.html#topic+appraise">appraise</a></code> function in the <code>gratia</code> package (rewritten to avoid depending on <code>gratia</code> package for these plots). This function will plot 4 diagnostic plots when given a generalised additive model fitted with <code><a href="#topic+ga_model">ga_model</a></code> or <code>mgcv</code>. It creates graphs that use <code>grafify</code> colours and <code>theme_grafify()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_qq_gam(
  Model,
  symsize = 2,
  s_colour = "#E69F00",
  s_alpha = 0.6,
  line_col = "black",
  base_size = 12,
  linethick,
  n_bins = c("sturges", "scott", "fd")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_qq_gam_+3A_model">Model</code></td>
<td>
<p>a model of class <code>gam</code> fitted with <code>ga_model</code> or the <code>mgcv</code> package.</p>
</td></tr>
<tr><td><code id="plot_qq_gam_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols (default = 2)</p>
</td></tr>
<tr><td><code id="plot_qq_gam_+3A_s_colour">s_colour</code></td>
<td>
<p>colour of symbols (default = <code>ok_orange</code>)</p>
</td></tr>
<tr><td><code id="plot_qq_gam_+3A_s_alpha">s_alpha</code></td>
<td>
<p>opacity of symbols (default = 0.8)</p>
</td></tr>
<tr><td><code id="plot_qq_gam_+3A_line_col">line_col</code></td>
<td>
<p>colour of lines (default = <code>black</code>)</p>
</td></tr>
<tr><td><code id="plot_qq_gam_+3A_base_size">base_size</code></td>
<td>
<p>font size for theme (default = 12)</p>
</td></tr>
<tr><td><code id="plot_qq_gam_+3A_linethick">linethick</code></td>
<td>
<p>thickness in 'pt' units of lines and symbol orders (default = base_size/22)</p>
</td></tr>
<tr><td><code id="plot_qq_gam_+3A_n_bins">n_bins</code></td>
<td>
<p>one of either &quot;sturges&quot;, &quot;scott&quot;, &quot;fd&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns an object of classes &quot;ggplot&quot; and &quot;gg&quot;.
</p>
<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>

<hr>
<h2 id='plot_qqline'>Plot quantile-quantile (QQ) graphs from data.</h2><span id='topic+plot_qqline'></span>

<h3>Description</h3>

<p>This function takes a data table, a quantitative variable (<code>ycol</code>), and a categorical grouping variable (<code>group</code>), if available, and plots a QQ graph using <code><a href="ggplot2.html#topic+stat_qq">stat_qq</a></code> and <code><a href="ggplot2.html#topic+stat_qq_line">stat_qq_line</a></code>.  Alternatives are <code><a href="#topic+plot_histogram">plot_histogram</a></code>, or <code><a href="#topic+plot_qqline">plot_qqline</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_qqline(
  data,
  ycol,
  group,
  facet,
  symsize = 3,
  s_alpha = 0.8,
  TextXAngle = 0,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  linethick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_qqline_+3A_data">data</code></td>
<td>
<p>a data table e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable whose distribution is to be plotted.</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_group">group</code></td>
<td>
<p>name of the column containing a categorical grouping variable.</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 1 (i.e. maximum opacity &amp; zero transparency).</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_symthick">symthick</code></td>
<td>
<p>thickness of symbol border, default set to <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_linethick">linethick</code></td>
<td>
<p>thickness of lines, default set to <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_qqline_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_qq">geom_qq</a></code> or <code><a href="ggplot2.html#topic+geom_qq_line">geom_qq_line</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the function requires the quantitative Y variable first, and a grouping variable as <code>group</code> if required. The graph plots sample quantiles on Y axis &amp; theoretical quantiles on X axis. The X variable is mapped to the <code>fill</code> aesthetic in<code><a href="ggplot2.html#topic+stat_qq">stat_qq</a></code> and <code>colour</code> aesthetic for the <code><a href="ggplot2.html#topic+stat_qq_line">stat_qq_line</a></code>.
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments. Colours available can be seen quickly with <code><a href="#topic+plot_grafify_palette">plot_grafify_palette</a></code>. When only one level is present within <code>group</code>, symbols will receive &quot;ok_orange&quot; colour.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_qqline(data = data_cholesterol, 
ycol = Cholesterol, group = Treatment)

</code></pre>

<hr>
<h2 id='plot_qqmodel'>Plot quantile-quantile (QQ) graphs from residuals of linear models.</h2><span id='topic+plot_qqmodel'></span>

<h3>Description</h3>

<p>This function takes a linear model (simple or mixed effects) and plots a QQ graph after running <code>rstudent</code> from <code><a href="stats.html#topic+rstudent">rstudent</a></code> to generate a table of Studentised model residuals on an ordinary (<code><a href="#topic+simple_model">simple_model</a></code>),  mixed model (<code><a href="#topic+mixed_model">mixed_model</a></code> or <code><a href="#topic+mixed_model_slopes">mixed_model_slopes</a></code>. The graph plots studentised residuals from the model (sample) on Y axis &amp; Theoretical quantiles on X axis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_qqmodel(
  Model,
  symsize = 3,
  s_alpha = 0.8,
  fontsize = 20,
  symthick,
  linethick,
  SingleColour = "#E69F00"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_qqmodel_+3A_model">Model</code></td>
<td>
<p>name of a saved model generated by <code>simple_model</code> or <code>mixed_model</code>.</p>
</td></tr>
<tr><td><code id="plot_qqmodel_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_qqmodel_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e., 80% opacity).</p>
</td></tr>
<tr><td><code id="plot_qqmodel_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_qqmodel_+3A_symthick">symthick</code></td>
<td>
<p>thickness of symbol border, default set to <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_qqmodel_+3A_linethick">linethick</code></td>
<td>
<p>thickness of line, default set to <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_qqmodel_+3A_singlecolour">SingleColour</code></td>
<td>
<p>colour of symbols (default = <code style="white-space: pre;">&#8288;#E69F00&#8288;</code>, which is <code>ok_orange</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For generalised additive models fit with <code>mgcv</code>, scaled Pearson residuals are plotted.
</p>
<p>The function uses <code><a href="ggplot2.html#topic+geom_qq">geom_qq</a></code> and <code><a href="ggplot2.html#topic+geom_qq_line">geom_qq_line</a></code> geometries. Symbols receive &quot;ok_orange&quot; colour by default.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Basic usage
m1 &lt;- simple_model(data = data_2w_Festing, 
Y_value = "GST", 
Fixed_Factor = c("Treatment", "Strain"))
plot_qqmodel(m1)

</code></pre>

<hr>
<h2 id='plot_scatterbar_sd'>Plot scatter dots on a bar graph with SD error bars with two variables.</h2><span id='topic+plot_scatterbar_sd'></span>

<h3>Description</h3>

<p>There are 4 related functions that use <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> to plot a categorical variable along the X axis.
</p>

<ol>
<li> <p><a href="#topic+plot_point_sd">plot_point_sd</a> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><a href="#topic+plot_scatterbox">plot_scatterbox</a> (box &amp; whiskers)
</p>
</li>
<li> <p><a href="#topic+plot_scatterviolin">plot_scatterviolin</a> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_scatterbar_sd(
  data,
  xcol,
  ycol,
  facet,
  ErrorType = "SD",
  symsize = 3,
  s_alpha = 0.8,
  b_alpha = 1,
  bwid = 0.5,
  ewid = 0.3,
  jitter = 0.1,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  bthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_scatterbar_sd_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) to plot on X axis. This should be a categorical variable.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis. This should be a quantitative variable.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_errortype">ErrorType</code></td>
<td>
<p>select the type of error bars to display. Default is &quot;SD&quot; (standard deviation). Other options are &quot;SEM&quot; (standard error of the mean) and &quot;CI95&quot; (95% confidence interval based on t distributions).</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_symsize">symsize</code></td>
<td>
<p>size of point symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e, 80% opacity).</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes, default set to .5 (i.e. 50% transparent).</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_bwid">bwid</code></td>
<td>
<p>width of bars, default set to 0.5.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_ewid">ewid</code></td>
<td>
<p>width of error bars, default set to 0.3.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0.1.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all facet facet graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of both bar and error bar lines; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> palettes or base R to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_scatterbar_sd_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions take a data table, categorical X and numeric Y variables, and plot various geometries. The X variable is mapped to the <code>fill</code> aesthetic of symbols.
</p>
<p>In <a href="#topic+plot_point_sd">plot_point_sd</a> and <a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a>, default error bars are SD, which can be changed to SEM or CI95.
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments. Colours available can be seen quickly with <code><a href="#topic+plot_grafify_palette">plot_grafify_palette</a></code>.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>If you prefer a single colour for the graph, use the <code>SingleColour</code> argument.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#with jitter
plot_scatterbar_sd(data = data_cholesterol, 
xcol = Treatment, ycol = Cholesterol)

plot_scatterbar_sd(data = data_doubling_time, 
xcol = Student, ycol = Doubling_time,
SingleColour = "ok_grey")

</code></pre>

<hr>
<h2 id='plot_scatterbox'>Plot a scatter plot on a boxplot with two variables.</h2><span id='topic+plot_scatterbox'></span>

<h3>Description</h3>

<p>There are 4 related functions that use <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> to plot a categorical variable along the X axis.
</p>

<ol>
<li> <p><a href="#topic+plot_point_sd">plot_point_sd</a> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><a href="#topic+plot_scatterbox">plot_scatterbox</a> (box &amp; whiskers)
</p>
</li>
<li> <p><a href="#topic+plot_scatterviolin">plot_scatterviolin</a> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_scatterbox(
  data,
  xcol,
  ycol,
  facet,
  symsize = 3,
  s_alpha = 0.8,
  b_alpha = 1,
  bwid = 0.5,
  jitter = 0.1,
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  bthick,
  Ylabels = waiver(),
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_scatterbox_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) to plot on X axis. This should be a categorical variable.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis. This should be a quantitative variable.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols, default set to 3.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e, 80% opacity).</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes, default set to 1.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_bwid">bwid</code></td>
<td>
<p>width of boxplots; default 0.5.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0.1.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of boxplot lines; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_ylabels">Ylabels</code></td>
<td>
<p>deprecated, use <code>LogYLabels</code> instead.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> or base R palettes to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_scatterbox_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions take a data table, categorical X and numeric Y variables, and plot various geometries. The X variable is mapped to the <code>fill</code> aesthetic of symbols.
</p>
<p>In <a href="#topic+plot_point_sd">plot_point_sd</a> and <a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a>, default error bars are SD, which can be changed to SEM or CI95.
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments. Colours available can be seen quickly with <code><a href="#topic+plot_grafify_palette">plot_grafify_palette</a></code>.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>If you prefer a single colour for the graph, use the <code>SingleColour</code> argument.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_scatterbox(data = data_cholesterol, 
xcol = Treatment, ycol = Cholesterol)

plot_scatterbox(data = data_doubling_time, 
xcol = Student, ycol = Doubling_time,
SingleColour = "ok_grey")

</code></pre>

<hr>
<h2 id='plot_scatterviolin'>Plot a scatter plot on a violin plot with two variables.</h2><span id='topic+plot_scatterviolin'></span>

<h3>Description</h3>

<p>There are 4 related functions that use <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> to plot a categorical variable along the X axis.
</p>

<ol>
<li> <p><a href="#topic+plot_point_sd">plot_point_sd</a> (mean &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a> (bar &amp; SD, SEM or CI95 error bars)
</p>
</li>
<li> <p><a href="#topic+plot_scatterbox">plot_scatterbox</a> (box &amp; whiskers)
</p>
</li>
<li> <p><a href="#topic+plot_scatterviolin">plot_scatterviolin</a> (box &amp; whiskers, violin)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>plot_scatterviolin(
  data,
  xcol,
  ycol,
  facet,
  symsize = 3,
  s_alpha = 0.8,
  b_alpha = 0,
  v_alpha = 1,
  bwid = 0.3,
  vadjust = 1,
  jitter = 0.1,
  trim = TRUE,
  scale = "width",
  TextXAngle = 0,
  LogYTrans,
  LogYBreaks = waiver(),
  LogYLabels = waiver(),
  LogYLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  symthick,
  bthick,
  vthick,
  bvthick,
  Ylabels = waiver(),
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  SingleColour = "NULL",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_scatterviolin_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) to plot on X axis. This should be a categorical variable.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with the quantitative variable to plot on the Y axis. This should be a quantitative variable.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_symsize">symsize</code></td>
<td>
<p>size of dots relative to <code>binwidth</code> used by <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>. Default set to 3.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to to 0.8 (i.e, 80% opacity). Set <code>s_alpha = 0</code> to not show scatter plot.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxplots.  Default is set to 0, which results in white boxes inside violins. Change to any value &gt;0 up to 1 for different levels of transparency.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_v_alpha">v_alpha</code></td>
<td>
<p>fractional opacity of violins, default set to 1.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_bwid">bwid</code></td>
<td>
<p>width of boxplots; default 0.3.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_vadjust">vadjust</code></td>
<td>
<p>number to adjust the smooth/wigglyness of violin plot (default set to 1).</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_jitter">jitter</code></td>
<td>
<p>extent of jitter (scatter) of symbols, default is 0 (i.e. aligned symbols). To reduce symbol overlap, try 0.1-0.3 or higher.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_trim">trim</code></td>
<td>
<p>set whether tips of violin plot should be trimmed at high/low data. Default <code>trim = T</code>, can be changed to F.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_scale">scale</code></td>
<td>
<p>set to &quot;area&quot; by default, can be changed to &quot;count&quot; or &quot;width&quot;.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_bthick">bthick</code></td>
<td>
<p>thickness (in 'pt' units) of boxplots; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_vthick">vthick</code></td>
<td>
<p>thickness (in 'pt' units) of violins; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_bvthick">bvthick</code></td>
<td>
<p>thickness (in 'pt' units) of both violins and boxplots; default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_ylabels">Ylabels</code></td>
<td>
<p>deprecated, use <code>LogYLabels</code> instead.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_singlecolour">SingleColour</code></td>
<td>
<p>a colour hexcode (starting with #, e.g., &quot;#E69F00&quot;), a number between 1-154, or names of colours from <code>grafify</code> or base R palettes to fill along X-axis aesthetic. Accepts any colour other than &quot;black&quot;; use <code>grey_lin11</code>, which is almost black.</p>
</td></tr>
<tr><td><code id="plot_scatterviolin_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass to <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code>, <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> or <code><a href="ggplot2.html#topic+geom_violin">geom_violin</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions take a data table, categorical X and numeric Y variables, and plot various geometries. The X variable is mapped to the <code>fill</code> aesthetic of symbols.
</p>
<p>In <a href="#topic+plot_point_sd">plot_point_sd</a> and <a href="#topic+plot_scatterbar_sd">plot_scatterbar_sd</a>, default error bars are SD, which can be changed to SEM or CI95.
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments. Colours available can be seen quickly with <code><a href="#topic+plot_grafify_palette">plot_grafify_palette</a></code>.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>If you prefer a single colour for the graph, use the <code>SingleColour</code> argument.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#plot without jitter
plot_scatterviolin(data = data_t_pdiff, 
xcol = Condition, ycol = Mass, 
symsize = 2, trim = FALSE)

#no symbols
plot_scatterviolin(data = data_t_pdiff, 
xcol = Condition, ycol = Mass, 
s_alpha = 0,
symsize = 2, trim = FALSE)

</code></pre>

<hr>
<h2 id='plot_xy_CatGroup'>Plot points on a quantitative X - Y plot &amp; a categorical grouping variable.</h2><span id='topic+plot_xy_CatGroup'></span>

<h3>Description</h3>

<p>This function takes a data table, quantitative X and Y variables along with a categorical grouping variable, and a and plots a graph with using <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>. The categorical <code>CatGroup</code> variable is mapped to the <code>fill</code> aesthetic of symbols.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_xy_CatGroup(
  data,
  xcol,
  ycol,
  CatGroup,
  facet,
  Boxplot = FALSE,
  symsize = 3,
  s_alpha = 0.8,
  TextXAngle = 0,
  LogYTrans,
  LogXTrans,
  LogYBreaks = waiver(),
  LogXBreaks = waiver(),
  LogYLabels = waiver(),
  LogXLabels = waiver(),
  LogYLimits = NULL,
  LogXLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  bwid = 0.3,
  b_alpha = 0.3,
  l_alpha = 0.8,
  symthick,
  bthick,
  ColPal = c("okabe_ito", "all_grafify", "bright", "contrast", "dark", "fishy", "kelly",
    "light", "muted", "pale", "r4", "safe", "vibrant"),
  ColSeq = TRUE,
  ColRev = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_xy_CatGroup_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) with quantitative X variable.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with quantitative Y variable.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_catgroup">CatGroup</code></td>
<td>
<p>a categorical variable as grouping factor for colour of data points, should be a categorical variable for default colours to work. Will be converted to <code>factor</code> if your column is numeric</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_boxplot">Boxplot</code></td>
<td>
<p>logical TRUE/FALSE to plot box and whiskers plot (default = FALSE).</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols used by <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>. Default set to 3.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to to 0.8 (i.e, 80% opacity).</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_logxtrans">LogXTrans</code></td>
<td>
<p>transform X axis into &quot;log10&quot; or &quot;log2&quot;</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code>ggplot2[scale_y_continuous]</code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_logxbreaks">LogXBreaks</code></td>
<td>
<p>argument for <code>ggplot2[scale_x_continuous]</code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code>ggplot2[scale_y_continuous]</code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_logxlabels">LogXLabels</code></td>
<td>
<p>argument for <code>ggplot2[scale_x_continuous]</code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_logxlimits">LogXLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the X axis.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_bwid">bwid</code></td>
<td>
<p>width of boxplot (default = 0.3).</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes, (default = 0.3).</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_l_alpha">l_alpha</code></td>
<td>
<p>fractional opacity of lines joining boxes, (default = 0.8).</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_bthick">bthick</code></td>
<td>
<p>size (in 'pt' units) of outline of boxes, whisker and joining lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours, which will be applied using  <code>scale_fill_grafify2</code>.</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_xy_CatGroup_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass on.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A box and whisker plot with lines joining the medians can be plotted with <code>Boxplot = TRUE</code>. If only box plot is needed without the line, set the opacity of the line to 0 (i.e., l_alpha = 0).
</p>
<p>Colours can be changed using <code>ColPal</code>, <code>ColRev</code> or <code>ColSeq</code> arguments. Colours available can be seen quickly with <code><a href="#topic+plot_grafify_palette">plot_grafify_palette</a></code>.
<code>ColPal</code> can be one of the following: &quot;okabe_ito&quot;, &quot;dark&quot;, &quot;light&quot;, &quot;bright&quot;, &quot;pale&quot;, &quot;vibrant,  &quot;muted&quot; or &quot;contrast&quot;.
<code>ColRev</code> (logical TRUE/FALSE) decides whether colours are chosen from first-to-last or last-to-first from within the chosen palette.
<code>ColSeq</code> (logical TRUE/FALSE) decides whether colours are picked by respecting the order in the palette or the most distant ones using <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.
</p>
<p>This plot is related to <code><a href="#topic+plot_xy_NumGroup">plot_xy_NumGroup</a></code> which requires a numeric grouping factor.
When summary statistics (mean/median) are required, use <code><a href="#topic+plot_3d_scatterbar">plot_3d_scatterbar</a></code>, <code><a href="#topic+plot_3d_scatterbox">plot_3d_scatterbox</a></code> or <code><a href="#topic+plot_4d_scatterbox">plot_4d_scatterbox</a></code>.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The grouping factor cyl  is automatically converted to categorical variable
plot_xy_CatGroup(data = mtcars,
xcol = mpg, ycol = disp, CatGroup = cyl, 
ColPal = "vibrant", ColSeq = FALSE)

#with boxplot
plot_xy_CatGroup(data = mpg, 
xcol = cyl, ycol = cty, 
CatGroup = fl, Boxplot = TRUE)

#add another variable
#with boxplot
plot_xy_CatGroup(data = mpg, 
xcol = cyl, ycol = cty, 
CatGroup = fl, facet = drv,
Boxplot = TRUE)

</code></pre>

<hr>
<h2 id='plot_xy_NumGroup'>Plot points on a quantitative X - Y plot &amp; a numeric grouping variable.</h2><span id='topic+plot_xy_NumGroup'></span>

<h3>Description</h3>

<p>This function takes a data table, quantitative X and Y variables, and a numeric grouping variable, and a and plots a graph with using <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>. The numerical <code>NumGroup</code> variable is mapped to the <code>fill</code> aesthetic of symbols, which receives the <code>scale_fill_grafify</code> default quantitative palette (<code>blue_conti</code>). Alternatives are <code>yellow_conti</code>, <code>grey_conti</code>, <code>OrBl_div</code> and <code>PrGn_div</code>. Colour order can be reversed with <code>ColRev = TRUE</code> (default is <code>FALSE</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_xy_NumGroup(
  data,
  xcol,
  ycol,
  NumGroup,
  facet,
  Boxplot = FALSE,
  symsize = 3,
  s_alpha = 0.8,
  TextXAngle = 0,
  LogYTrans,
  LogXTrans,
  LogYBreaks = waiver(),
  LogXBreaks = waiver(),
  LogYLabels = waiver(),
  LogXLabels = waiver(),
  LogYLimits = NULL,
  LogXLimits = NULL,
  facet_scales = "fixed",
  fontsize = 20,
  bwid = 0.3,
  b_alpha = 0.3,
  l_alpha = 0.8,
  symthick,
  bthick,
  ColPal = c("blue_conti", "yellow_conti", "grey_conti", "PrGn_div", "OrBl_div"),
  ColRev = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_xy_NumGroup_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_xcol">xcol</code></td>
<td>
<p>name of the column (without quotes) with quantitative X variable.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_ycol">ycol</code></td>
<td>
<p>name of the column (without quotes) with quantitative Y variable.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_numgroup">NumGroup</code></td>
<td>
<p>a numeric factor for <code>fill</code> aesthetic of data points.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_facet">facet</code></td>
<td>
<p>add another variable (without quotes) from the data table to create faceted graphs using <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_boxplot">Boxplot</code></td>
<td>
<p>logical TRUE/FALSE to plot box and whiskers plot (default = FALSE).</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_symsize">symsize</code></td>
<td>
<p>size of symbols used by <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>. Default set to 3.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_s_alpha">s_alpha</code></td>
<td>
<p>fractional opacity of symbols, default set to 0.8 (i.e, 80% opacity).</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_logytrans">LogYTrans</code></td>
<td>
<p>transform Y axis into &quot;log10&quot; or &quot;log2&quot; (in quotes).</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_logxtrans">LogXTrans</code></td>
<td>
<p>transform X axis into &quot;log10&quot; or &quot;log2&quot;</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_logybreaks">LogYBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_logxbreaks">LogXBreaks</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_x_continuous">scale_x_continuous</a></code> for Y axis breaks on log scales, default is <code>waiver()</code>, or provide a vector of desired breaks.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_logylabels">LogYLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_y_continuous">scale_y_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_logxlabels">LogXLabels</code></td>
<td>
<p>argument for <code><a href="ggplot2.html#topic+scale_x_continuous">scale_x_continuous</a></code> for Y axis labels on log scales, default is <code>waiver()</code>, or provide a vector of desired labels.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_logylimits">LogYLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the Y axis.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_logxlimits">LogXLimits</code></td>
<td>
<p>a vector of length two specifying the range (minimum and maximum) of the X axis.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_facet_scales">facet_scales</code></td>
<td>
<p>whether or not to fix scales on X &amp; Y axes for all graphs. Can be <code>fixed</code> (default), <code>free</code>, <code>free_y</code> or <code>free_x</code> (for Y and X axis one at a time, respectively).</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_fontsize">fontsize</code></td>
<td>
<p>parameter of <code>base_size</code> of fonts in <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code>, default set to size 20.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_bwid">bwid</code></td>
<td>
<p>width of boxplot (default = 0.3).</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_b_alpha">b_alpha</code></td>
<td>
<p>fractional opacity of boxes, (default = 0.3).</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_l_alpha">l_alpha</code></td>
<td>
<p>fractional opacity of lines joining boxes, (default = 0.8).</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_symthick">symthick</code></td>
<td>
<p>size (in 'pt' units) of outline of symbol lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_bthick">bthick</code></td>
<td>
<p>size (in 'pt' units) of outline of boxes, whisker and joining lines (<code>stroke</code>), default = <code>fontsize</code>/22.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_colpal">ColPal</code></td>
<td>
<p>grafify colour palette to apply (in quotes), default &quot;okabe_ito&quot;; see <code><a href="#topic+graf_palettes">graf_palettes</a></code> for available palettes.</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_colrev">ColRev</code></td>
<td>
<p>whether to reverse order of colour within the selected palette, default F (FALSE); can be set to T (TRUE).</p>
</td></tr>
<tr><td><code id="plot_xy_NumGroup_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass on.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This plot is related to <code><a href="#topic+plot_xy_CatGroup">plot_xy_CatGroup</a></code> which requires a categorical grouping factor.
When summary statistics (mean/median) are required, use <code><a href="#topic+plot_3d_scatterbar">plot_3d_scatterbar</a></code>, <code><a href="#topic+plot_3d_scatterbox">plot_3d_scatterbox</a></code> or <code><a href="#topic+plot_4d_scatterbox">plot_4d_scatterbox</a></code>.
</p>


<h3>Value</h3>

<p>This function returns a <code>ggplot2</code> object of class &quot;gg&quot; and &quot;ggplot&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The grouping factor gear is numeric 
plot_xy_NumGroup(data = mtcars,
xcol = mpg, ycol = disp, NumGroup = cyl,
s_alpha = 0.8)
#change colour palette
plot_xy_NumGroup(data = mtcars,
xcol = mpg, ycol = disp, NumGroup = cyl,
s_alpha = 0.8, 
ColPal = "grey_conti")
</code></pre>

<hr>
<h2 id='posthoc_Levelwise'>Level-wise post-hoc comparisons from a linear or linear mixed effects model.</h2><span id='topic+posthoc_Levelwise'></span>

<h3>Description</h3>

<p>This function is a wrapper based on <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>, and needs a ordinary linear model produced by <code><a href="#topic+simple_model">simple_model</a></code> or a mixed effects model produced by <code><a href="#topic+mixed_model">mixed_model</a></code> or <code><a href="#topic+mixed_model_slopes">mixed_model_slopes</a></code> (or generated directly with <code>lm</code>,  <code>lme4</code> or <code>lmerTest</code> calls). It also needs to know the fixed factor(s), which should match those in the model and data table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posthoc_Levelwise(Model, Fixed_Factor, P_Adj = "fdr", Factor, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="posthoc_Levelwise_+3A_model">Model</code></td>
<td>
<p>a model object fit using <code><a href="#topic+simple_model">simple_model</a></code> or <code><a href="#topic+mixed_model">mixed_model</a></code> or related.</p>
</td></tr>
<tr><td><code id="posthoc_Levelwise_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>one or  more categorical variables, provided as a vector (e.g., c(&quot;A&quot;, &quot;B&quot;)), whose levels you wish to compare pairwise. Names of Fixed_Factor should match Fixed_Factor used to fit the model. When more than one factor is provided e.g. <code>Fixed_factor = c("A", "B")</code>, this function passes this on as <code>specs = A|B</code> (note the vertical | between the two Fixed_Factor) to <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> to produce comparisons between each level A with each other listed separately at each level of B.</p>
</td></tr>
<tr><td><code id="posthoc_Levelwise_+3A_p_adj">P_Adj</code></td>
<td>
<p>method for correcting P values for multiple comparisons. Default is set to false discovery rate (&quot;fdr&quot;), can be changed to &quot;none&quot;, &quot;tukey&quot;, &quot;bonferroni&quot;, &quot;sidak&quot; (in quotes). See Interaction analysis in emmeans in the <a href="https://CRAN.R-project.org/package=emmeans">manual</a> for <code>emmeans</code>.</p>
</td></tr>
<tr><td><code id="posthoc_Levelwise_+3A_factor">Factor</code></td>
<td>
<p>old argument name for <code>Fixed_Factor</code>; retained for backward compatibility.</p>
</td></tr>
<tr><td><code id="posthoc_Levelwise_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> such as <code>lmer.df</code> or others. See help for sophisticated models in <a href="https://CRAN.R-project.org/package=emmeans">emmeans</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will generate <a href="https://CRAN.R-project.org/package=emmeans">level-wise comparisons</a> (as described in Comparisons and contrasts in emmeans), i.e. comparison between of every level of one factor separately at each level of the other factor.
By default, P values are corrected by the FDR method (which can be changed). If the model was fit by transforming the quantitative response variable using &quot;log&quot;, &quot;logit&quot;, &quot;sqrt&quot; etc., results will still be on the original scale, i.e. <code>type = "response"</code> is the default; data will be back-transformed (check results to confirm this), and for log or logit see Transformations and link functions in emmeans, <a href="https://CRAN.R-project.org/package=emmeans">ratios will be compared</a>.
The first part of the <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> results has the estimated marginal means, SE and CI (<code>$emmeans</code>), which are generated from the fitted model, and <strong>not</strong> the original data table. The second part has the results of the comparisons (<code>$contrasts</code>).
</p>


<h3>Value</h3>

<p>returns an &quot;emm_list&quot; object containing contrasts and emmeans through <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#make a linear model first
CholMod &lt;- mixed_model(data = data_cholesterol, 
Y_value =  "Cholesterol", 
Fixed_Factor = c("Hospital", "Treatment"),
Random_Factor =  "Subject") 

#note Fixed_Factor needs a vector or variable in quotes
#to get comparisons between different hospitals separately for each level of Treatment
posthoc_Levelwise(Model = CholMod, 
Fixed_Factor = c("Hospital", "Treatment"))

#get comparisons between treatments separately at each hospital
posthoc_Levelwise(Model = CholMod, 
Fixed_Factor = c("Treatment", "Hospital"))

</code></pre>

<hr>
<h2 id='posthoc_Pairwise'>Pairwise post-hoc comparisons from a linear or linear mixed effects model.</h2><span id='topic+posthoc_Pairwise'></span>

<h3>Description</h3>

<p>This function is a wrapper based on <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>, and needs a ordinary linear model produced by <code><a href="#topic+simple_model">simple_model</a></code> or a mixed effects model produced by <code><a href="#topic+mixed_model">mixed_model</a></code> or <code><a href="#topic+mixed_model_slopes">mixed_model_slopes</a></code> (or generated directly with <code>lm</code>,  <code>lme4</code> or <code>lmerTest</code> calls). It also needs to know the fixed factor(s), which should match those in the model and data table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posthoc_Pairwise(Model, Fixed_Factor, P_Adj = "fdr", Factor, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="posthoc_Pairwise_+3A_model">Model</code></td>
<td>
<p>a model object fit using <code><a href="#topic+simple_model">simple_model</a></code> or <code><a href="#topic+mixed_model">mixed_model</a></code> or related.</p>
</td></tr>
<tr><td><code id="posthoc_Pairwise_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>one or  more categorical variables, provided as a vector (e.g., c(&quot;A&quot;, &quot;B&quot;)), whose levels you wish to compare pairwise. Names of Fixed_Factor should match Fixed_Factor used to fit the model. When more than one factor is provided e.g. <code>Fixed_factor = c("A", "B")</code>, this function passes this on as <code>specs = A:B</code> (note the colon between the two Fixed_Factor) to <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> to produce pairwise comparisons.</p>
</td></tr>
<tr><td><code id="posthoc_Pairwise_+3A_p_adj">P_Adj</code></td>
<td>
<p>method for correcting P values for multiple comparisons. Default is set to false discovery rate (&quot;fdr&quot;), can be changed to &quot;none&quot;, &quot;tukey&quot;, &quot;bonferroni&quot;, &quot;sidak&quot; (in quotes). See Interaction analysis in emmeans in the <a href="https://CRAN.R-project.org/package=emmeans">manual</a> for <code>emmeans</code>.</p>
</td></tr>
<tr><td><code id="posthoc_Pairwise_+3A_factor">Factor</code></td>
<td>
<p>old argument name for <code>Fixed_Factor</code>; retained for backward compatibility.</p>
</td></tr>
<tr><td><code id="posthoc_Pairwise_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> such as <code>lmer.df</code> or others. See help for sophisticated models in <a href="https://CRAN.R-project.org/package=emmeans">emmeans</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will generate <a href="https://CRAN.R-project.org/package=emmeans">pairwise comparisons</a> of every level of every factor (as described in Comparisons and contrasts in emmeans). Too many comparisons will be generated and only use this when necessary.
By default, P values are corrected by the FDR method (which can be changed). If the model was fit by transforming the quantitative response variable using &quot;log&quot;, &quot;logit&quot;, &quot;sqrt&quot; etc., results will still be on the original scale, i.e. <code>type = "response"</code> is the default; data will be back-transformed (check results to confirm this), and for log or logit see Transformations and link functions in emmeans, <a href="https://CRAN.R-project.org/package=emmeans">ratios will be compared</a>.
The first part of the <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> results has the estimated marginal means, SE and CI (<code>$emmeans</code>), which are generated from the fitted model, and <strong>not</strong> the original data table. The second part has the results of the comparisons (<code>$contrasts</code>).
</p>


<h3>Value</h3>

<p>returns an &quot;emm_list&quot; object containing contrasts and emmeans through  <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#make linear models first
DoublMod &lt;- simple_model(data = data_doubling_time, 
Y_value =  "Doubling_time", Fixed_Factor = "Student") 

#mixed model
CholMod &lt;- mixed_model(data = data_cholesterol, 
Y_value = "Cholesterol",
Fixed_Factor = c("Hospital", "Treatment"), 
Random_Factor = "Subject") 

posthoc_Pairwise(Model = DoublMod, 
Fixed_Factor = "Student")

#basic use with two Fixed_Factor provided as a vector
posthoc_Pairwise(Model = CholMod, 
Fixed_Factor = c("Treatment", "Hospital"))

#same call with "tukey" adjustment
posthoc_Pairwise(Model = CholMod, 
Fixed_Factor = c("Treatment", "Hospital"), 
P_adj = "tukey")

</code></pre>

<hr>
<h2 id='posthoc_Trends_Levelwise'>Use emtrends to get level-wise comparison of slopes from a linear model.</h2><span id='topic+posthoc_Trends_Levelwise'></span>

<h3>Description</h3>

<p>This function is a wrapper based on <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>, and needs a ordinary linear model produced by <code><a href="#topic+simple_model">simple_model</a></code> or a mixed effects model produced by <code><a href="#topic+mixed_model">mixed_model</a></code> or <code><a href="#topic+mixed_model_slopes">mixed_model_slopes</a></code> (or generated directly with <code>lm</code>,  <code>lme4</code> or <code>lmerTest</code> calls). At least one of the factors should be a numeric covariate whose slopes you wish to find. It also needs to know the fixed factor(s), which should match those in the model and data table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posthoc_Trends_Levelwise(
  Model,
  Fixed_Factor,
  Trend_Factor,
  P_Adj = "sidak",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="posthoc_Trends_Levelwise_+3A_model">Model</code></td>
<td>
<p>a model object fit using <code><a href="#topic+simple_model">simple_model</a></code> or <code><a href="#topic+mixed_model">mixed_model</a></code> (or <code><a href="stats.html#topic+lm">lm</a></code> or <code><a href="lmerTest.html#topic+lmer">lmer</a></code>).</p>
</td></tr>
<tr><td><code id="posthoc_Trends_Levelwise_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>one or  more categorical variables, provided as a vector (e.g., c(&quot;A&quot;, &quot;B&quot;)), whose levels you wish to compare pairwise. Names of Fixed_Factor should match Fixed_Factor used to fit the model. When more than one factor is provided e.g. <code>Fixed_factor = c("A", "B")</code>, this function passes this on as <code>specs = A:B</code> (note the colon between the two Fixed_Factor) to <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> to produce pairwise comparisons.</p>
</td></tr>
<tr><td><code id="posthoc_Trends_Levelwise_+3A_trend_factor">Trend_Factor</code></td>
<td>
<p>a quantitative variable that interacts with a factor and whose slope (trend) is to be compared</p>
</td></tr>
<tr><td><code id="posthoc_Trends_Levelwise_+3A_p_adj">P_Adj</code></td>
<td>
<p>method for correcting P values for multiple comparisons. Default is &quot;sidak&quot;, can be changed to &quot;bonferroni&quot;. See Interaction analysis in emmeans in the <a href="https://CRAN.R-project.org/package=emmeans">manual</a> for <code>emmeans</code>.</p>
</td></tr>
<tr><td><code id="posthoc_Trends_Levelwise_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> such as <code>lmer.df</code> or others. See help for sophisticated models in <a href="https://CRAN.R-project.org/package=emmeans">emmeans</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Checkout the Interactions with covariates section in the <a href="https://CRAN.R-project.org/package=emmeans">emmeans</a> vignette for more details. One of the independent variables should  be a quantitative (e.g. time points) variable whose slope (trend) you want to find at levels of the other factor.
</p>


<h3>Value</h3>

<p>returns an &quot;emm_list&quot; object containing slopes and their contrasts calculated through  <code><a href="emmeans.html#topic+emtrends">emtrends</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#create an lm model 
#Time2 is numeric (time points)
m1 &lt;- simple_model(data = data_2w_Tdeath, 
Y_value = "PI", Fixed_Factor = c("Genotype", "Time2"))
posthoc_Trends_Levelwise(Model = m1, 
Fixed_Factor = "Genotype", 
Trend_Factor = "Time2")

</code></pre>

<hr>
<h2 id='posthoc_Trends_Pairwise'>Use emtrends to get pairwise comparison of slopes from a linear model.</h2><span id='topic+posthoc_Trends_Pairwise'></span>

<h3>Description</h3>

<p>This function is a wrapper based on <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>, and needs a ordinary linear model produced by <code><a href="#topic+simple_model">simple_model</a></code> or a mixed effects model produced by <code><a href="#topic+mixed_model">mixed_model</a></code> or <code><a href="#topic+mixed_model_slopes">mixed_model_slopes</a></code> (or generated directly with <code>lm</code>,  <code>lme4</code> or <code>lmerTest</code> calls). At least one of the factors should be a numeric covariate whose slopes you wish to find. It also needs to know the fixed factor(s), which should match those in the model and data table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posthoc_Trends_Pairwise(
  Model,
  Fixed_Factor,
  Trend_Factor,
  P_Adj = "sidak",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="posthoc_Trends_Pairwise_+3A_model">Model</code></td>
<td>
<p>a model object fit using <code><a href="#topic+simple_model">simple_model</a></code> or <code><a href="#topic+mixed_model">mixed_model</a></code> (or <code><a href="stats.html#topic+lm">lm</a></code> or <code><a href="lmerTest.html#topic+lmer">lmer</a></code>).</p>
</td></tr>
<tr><td><code id="posthoc_Trends_Pairwise_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>one or  more categorical variables, provided as a vector (e.g., c(&quot;A&quot;, &quot;B&quot;)), whose levels you wish to compare pairwise. Names of Fixed_Factor should match Fixed_Factor used to fit the model. When more than one factor is provided e.g. <code>Fixed_factor = c("A", "B")</code>, this function passes this on as <code>specs = A:B</code> (note the colon between the two Fixed_Factor) to <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> to produce pairwise comparisons.</p>
</td></tr>
<tr><td><code id="posthoc_Trends_Pairwise_+3A_trend_factor">Trend_Factor</code></td>
<td>
<p>a quantitative variable that interacts with a factor and whose slope (trend) is to be compared</p>
</td></tr>
<tr><td><code id="posthoc_Trends_Pairwise_+3A_p_adj">P_Adj</code></td>
<td>
<p>method for correcting P values for multiple comparisons. Default is &quot;sidak&quot;, can be changed to &quot;bonferroni&quot;. See Interaction analysis in emmeans in the <a href="https://CRAN.R-project.org/package=emmeans">manual</a> for <code>emmeans</code>.</p>
</td></tr>
<tr><td><code id="posthoc_Trends_Pairwise_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> such as <code>lmer.df</code> or others. See help for sophisticated models in <a href="https://CRAN.R-project.org/package=emmeans">emmeans</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Checkout the Interactions with covariates section in the <a href="https://CRAN.R-project.org/package=emmeans">emmeans</a> vignette for more details. One of the independent variables should  be a quantitative (e.g. time points) variable whose slope (trend) you want to find at levels of the other factor.
</p>


<h3>Value</h3>

<p>returns an &quot;emm_list&quot; object containing slopes and their contrasts calculated through  <code><a href="emmeans.html#topic+emtrends">emtrends</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#create an lm model 
#Time2 is numeric (time points)
m1 &lt;- simple_model(data = data_2w_Tdeath, 
Y_value = "PI", Fixed_Factor = c("Genotype", "Time2"))
posthoc_Trends_Pairwise(Model = m1, 
Fixed_Factor = "Genotype", 
Trend_Factor = "Time2")

</code></pre>

<hr>
<h2 id='posthoc_Trends_vsRef'>Use emtrends to get level-wise comparison of slopes from a linear model.</h2><span id='topic+posthoc_Trends_vsRef'></span>

<h3>Description</h3>

<p>This function is a wrapper based on <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>, and needs a ordinary linear model produced by <code><a href="#topic+simple_model">simple_model</a></code> or a mixed effects model produced by <code><a href="#topic+mixed_model">mixed_model</a></code> or <code><a href="#topic+mixed_model_slopes">mixed_model_slopes</a></code> (or generated directly with <code>lm</code>,  <code>lme4</code> or <code>lmerTest</code> calls). At least one of the factors should be a numeric covariate whose slopes you wish to find. It also needs to know the fixed factor(s), which should match those in the model and data table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posthoc_Trends_vsRef(
  Model,
  Fixed_Factor,
  Trend_Factor,
  Ref_Level = 1,
  P_Adj = "sidak",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="posthoc_Trends_vsRef_+3A_model">Model</code></td>
<td>
<p>a model object fit using <code><a href="#topic+simple_model">simple_model</a></code> or <code><a href="#topic+mixed_model">mixed_model</a></code> (or <code><a href="stats.html#topic+lm">lm</a></code> or <code><a href="lmerTest.html#topic+lmer">lmer</a></code>).</p>
</td></tr>
<tr><td><code id="posthoc_Trends_vsRef_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>one or  more categorical variables, provided as a vector (e.g., c(&quot;A&quot;, &quot;B&quot;)), whose levels you wish to compare pairwise. Names of Fixed_Factor should match Fixed_Factor used to fit the model. When more than one factor is provided e.g. <code>Fixed_factor = c("A", "B")</code>, this function passes this on as <code>specs = A:B</code> (note the colon between the two Fixed_Factor) to <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> to produce pairwise comparisons.</p>
</td></tr>
<tr><td><code id="posthoc_Trends_vsRef_+3A_trend_factor">Trend_Factor</code></td>
<td>
<p>a quantitative variable that interacts with a factor and whose slope (trend) is to be compared</p>
</td></tr>
<tr><td><code id="posthoc_Trends_vsRef_+3A_ref_level">Ref_Level</code></td>
<td>
<p>the level within that factor to be considered the reference or control to compare other levels to (to be provided as a number - by default R orders levels alphabetically); default <code>Ref_Level = 1</code>.</p>
</td></tr>
<tr><td><code id="posthoc_Trends_vsRef_+3A_p_adj">P_Adj</code></td>
<td>
<p>method for correcting P values for multiple comparisons. Default is &quot;sidak&quot;, can be changed to &quot;bonferroni&quot;. See Interaction analysis in emmeans in the <a href="https://CRAN.R-project.org/package=emmeans">manual</a> for <code>emmeans</code>.</p>
</td></tr>
<tr><td><code id="posthoc_Trends_vsRef_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> such as <code>lmer.df</code> or others. See help for sophisticated models in <a href="https://CRAN.R-project.org/package=emmeans">emmeans</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Checkout the Interactions with covariates section in the <a href="https://CRAN.R-project.org/package=emmeans">emmeans</a> vignette for more details. One of the independent variables should  be a quantitative (e.g. time points) variable whose slope (trend) you want to find at levels of the other factor.
</p>


<h3>Value</h3>

<p>returns an &quot;emm_list&quot; object containing slopes and their contrasts calculated through  <code><a href="emmeans.html#topic+emtrends">emtrends</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#create an lm model 
#Time2 is numeric (time points)
m1 &lt;- simple_model(data = data_2w_Tdeath, 
Y_value = "PI", Fixed_Factor = c("Genotype", "Time2"))
posthoc_Trends_vsRef(Model = m1, 
Fixed_Factor = "Genotype", 
Trend_Factor = "Time2", 
Ref_Level = 2)

</code></pre>

<hr>
<h2 id='posthoc_vsRef'>Post-hoc comparisons to a control or reference group.</h2><span id='topic+posthoc_vsRef'></span>

<h3>Description</h3>

<p>This function is a wrapper based on <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>, and needs a ordinary linear model produced by <code><a href="#topic+simple_model">simple_model</a></code> or a mixed effects model produced by <code><a href="#topic+mixed_model">mixed_model</a></code> or <code><a href="#topic+mixed_model_slopes">mixed_model_slopes</a></code> (or generated directly with <code>lm</code>,  <code>lme4</code> or <code>lmerTest</code> calls). It also needs to know the fixed factor(s), which should match those in the model and data table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posthoc_vsRef(Model, Fixed_Factor, Ref_Level = 1, P_Adj = "fdr", Factor, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="posthoc_vsRef_+3A_model">Model</code></td>
<td>
<p>a model object fit using <code><a href="#topic+simple_model">simple_model</a></code> or <code><a href="#topic+mixed_model">mixed_model</a></code> or related.</p>
</td></tr>
<tr><td><code id="posthoc_vsRef_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>Fixed_Factor one or  more categorical variables, provided as a vector (e.g., c(&quot;A&quot;, &quot;B&quot;)), whose levels you wish to compare pairwise. Names of Fixed_Factor should match Fixed_Factor used to fit the model. When more than one factor is provided e.g. <code>Fixed_factor = c("A", "B")</code>, this function passes this on as <code>specs = A|B</code> (note the vertical | between the two Fixed_Factor) to <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>. The specification internally is set to <code>specs = trt.vs.ctrl, Ref_Level = 1</code> to compare each group in A to the reference first group in A, separately at each level of B.</p>
</td></tr>
<tr><td><code id="posthoc_vsRef_+3A_ref_level">Ref_Level</code></td>
<td>
<p>the level within that factor to be considered the reference or control to compare other levels to (to be provided as a number - by default R orders levels alphabetically); default <code>Ref_Level = 1</code>.</p>
</td></tr>
<tr><td><code id="posthoc_vsRef_+3A_p_adj">P_Adj</code></td>
<td>
<p>method for correcting P values for multiple comparisons. Default is set to false discovery rate (&quot;fdr&quot;), can be changed to &quot;none&quot;, &quot;tukey&quot;, &quot;bonferroni&quot;, &quot;sidak&quot; (in quotes). See Interaction analysis in emmeans in the <a href="https://CRAN.R-project.org/package=emmeans">manual</a> for <code>emmeans</code>.</p>
</td></tr>
<tr><td><code id="posthoc_vsRef_+3A_factor">Factor</code></td>
<td>
<p>old argument name for <code>Fixed_Factor</code>; retained for backward compatibility.</p>
</td></tr>
<tr><td><code id="posthoc_vsRef_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> such as <code>lmer.df</code> or others. See help for sophisticated models in <a href="https://CRAN.R-project.org/package=emmeans">emmeans</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will generate <a href="https://CRAN.R-project.org/package=emmeans">treatment vs control type of comparisons</a> (as described in Comparisons and contrasts in emmeans), i.e. comparison of each level of a factor to a reference level, which is set by default to the first level in the factor (<code>Ref_Level = 1</code>).
By default, P values are corrected by the FDR method (which can be changed). If the model was fit by transforming the quantitative response variable using &quot;log&quot;, &quot;logit&quot;, &quot;sqrt&quot; etc., results will still be on the original scale, i.e. <code>type = "response"</code> is the default; data will be back-transformed (check results to confirm this), and for log or logit see Transformations and link functions in emmeans, <a href="https://CRAN.R-project.org/package=emmeans">ratios will be compared</a>.
The first part of the <code><a href="emmeans.html#topic+emmeans">emmeans</a></code> results has the estimated marginal means, SE and CI (<code>$emmeans</code>), which are generated from the fitted model, and <strong>not</strong> the original data table. The second part has the results of the comparisons (<code>$contrasts</code>).
</p>


<h3>Value</h3>

<p>returns an &quot;emm_list&quot; object containing contrasts and emmeans through  <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#make linear models first
DoublMod &lt;- simple_model(data = data_doubling_time,
Y_value =  "Doubling_time", 
Fixed_Factor =  "Student") 

CholMod &lt;- mixed_model(data = data_cholesterol, 
Y_value =  "Cholesterol", 
Fixed_Factor = c("Hospital", "Treatment"),
Random_Factor = "Subject") 

#to compare all students with student #9
posthoc_vsRef(Model = DoublMod, 
Fixed_Factor = "Student", Ref_Level = 9)

#for comparison between hospital_a to every other hospital, separately at levels of Treatment
posthoc_vsRef(Model = CholMod, 
Fixed_Factor = c("Hospital", "Treatment"), Ref_Level = 1)

</code></pre>

<hr>
<h2 id='scale_colour_grafify'><code>scale_colour_</code> and <code>scale_fill_</code> functions</h2><span id='topic+scale_colour_grafify'></span><span id='topic+scale_color_grafify'></span>

<h3>Description</h3>

<p>These let you apply <code>grafify</code> discrete or continuous palettes as <code>fill</code> or <code>colour</code> aesthetics to any <code>ggplot2</code> (<code>scale_color_</code> spelling is also accepted).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_colour_grafify(
  palette = "okabe_ito",
  ColSeq = TRUE,
  reverse = FALSE,
  discrete = TRUE,
  ...
)

scale_color_grafify(
  palette = "okabe_ito",
  ColSeq = TRUE,
  reverse = FALSE,
  discrete = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scale_colour_grafify_+3A_palette">palette</code></td>
<td>
<p>Name of the <code>grafify</code> palettes from above, provide within quotes, e.g., <code>palette = "vibrant"</code>. Default discrete palette is <code>okabe_ito</code>. For quantitative palette, set <code>discrete = FALSE</code> (which will apply <code>blue_conti</code> unless another palette is chosen).</p>
</td></tr>
<tr><td><code id="scale_colour_grafify_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours.</p>
</td></tr>
<tr><td><code id="scale_colour_grafify_+3A_reverse">reverse</code></td>
<td>
<p>Whether the colour order should be reversed.</p>
</td></tr>
<tr><td><code id="scale_colour_grafify_+3A_discrete">discrete</code></td>
<td>
<p>not used.</p>
</td></tr>
<tr><td><code id="scale_colour_grafify_+3A_...">...</code></td>
<td>
<p>Additional parameters for <code>scale_fill</code> or <code>scale_colour</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default is <code>palette = "okabe_ito"</code>. The <code>discrete</code> argument is not used at present. The following discrete and quantitative palettes can be used.
</p>
<p>Categorical/discreet palettes:
</p>

<ul>
<li> <p><code>okabe_ito</code> (default)
</p>
</li>
<li> <p><code>bright</code>
</p>
</li>
<li> <p><code>contrast</code>
</p>
</li>
<li> <p><code>dark</code>
</p>
</li>
<li> <p><code>kelly</code>
</p>
</li>
<li> <p><code>light</code>
</p>
</li>
<li> <p><code>muted</code>
</p>
</li>
<li> <p><code>pale</code>
</p>
</li>
<li> <p><code>r4</code>
</p>
</li>
<li> <p><code>safe</code>
</p>
</li>
<li> <p><code>vibrant</code>
</p>
</li></ul>

<p>By default, sequential colours from above palettes will be chosen. To choose the most distant colours set <code>ColSeq = TRUE</code>.
</p>
<p>Sequential quantitative palettes:
</p>

<ul>
<li> <p><code>grey_conti</code>
</p>
</li>
<li> <p><code>blue_conti</code>
</p>
</li>
<li> <p><code>yellow_conti</code>
</p>
</li></ul>

<p>Divergent quantitative palettes:
</p>

<ul>
<li> <p><code>OrBl_div</code>
</p>
</li>
<li> <p><code>PrGn_div</code>
</p>
</li></ul>



<h3>Value</h3>

<p>ggplot scale_fill function for discrete colours.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#add a grafify fill scheme to ggplot
ggplot(emmeans::neuralgia, aes(x = Treatment, 
                               y = Duration))+
  geom_boxplot(aes(fill = Treatment), 
               alpha = .6)+
  geom_point(aes(colour = Treatment,
                 shape = Treatment), 
             size = 3)+
  scale_fill_grafify(palette = "bright")+
  scale_colour_grafify(palette = "bright")+
  facet_wrap("Sex")+
  theme_classic()
#distant colours `ColSeq = FALSE`   
ggplot(emmeans::neuralgia, aes(x = Treatment, 
                               y = Duration))+
  geom_boxplot(aes(fill = Treatment), 
               alpha = .6)+
  geom_point(aes(colour = Treatment,
                 shape = Treatment), 
             size = 3)+
  scale_fill_grafify(palette = "bright",
                     ColSeq = FALSE)+
  scale_colour_grafify(palette = "bright",
                       ColSeq = FALSE)+
  facet_wrap("Sex")+
  theme_classic()
#quantitative colour scheme
ggplot(mtcars, aes(x = disp,
                   y = mpg))+
  geom_point(aes(colour = cyl), 
             size = 3)+
  scale_colour_grafify(palette = "blue_conti")
</code></pre>

<hr>
<h2 id='scale_fill_grafify'><code>scale_colour_</code> and <code>scale_fill_</code> functions</h2><span id='topic+scale_fill_grafify'></span>

<h3>Description</h3>

<p>These let you apply <code>grafify</code> discrete or continuous palettes as <code>fill</code> or <code>colour</code> aesthetics to any <code>ggplot2</code> (<code>scale_color_</code> spelling is also accepted).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_fill_grafify(
  palette = "okabe_ito",
  ColSeq = TRUE,
  reverse = FALSE,
  discrete = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scale_fill_grafify_+3A_palette">palette</code></td>
<td>
<p>Name of the <code>grafify</code> palettes from above, provide within quotes, e.g., <code>palette = "vibrant"</code>. Default discrete palette is <code>okabe_ito</code>. For quantitative palette, set <code>discrete = FALSE</code> (which will apply <code>blue_conti</code> unless another palette is chosen).</p>
</td></tr>
<tr><td><code id="scale_fill_grafify_+3A_colseq">ColSeq</code></td>
<td>
<p>logical TRUE or FALSE. Default TRUE for sequential colours from chosen palette. Set to FALSE for distant colours.</p>
</td></tr>
<tr><td><code id="scale_fill_grafify_+3A_reverse">reverse</code></td>
<td>
<p>Whether the colour order should be reversed.</p>
</td></tr>
<tr><td><code id="scale_fill_grafify_+3A_discrete">discrete</code></td>
<td>
<p>not used.</p>
</td></tr>
<tr><td><code id="scale_fill_grafify_+3A_...">...</code></td>
<td>
<p>Additional parameters for <code>scale_fill</code> or <code>scale_colour</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default is <code>palette = "okabe_ito"</code>. The <code>discrete</code> argument is not used at present. The following discrete and quantitative palettes can be used.
</p>
<p>Categorical/discreet palettes:
</p>

<ul>
<li> <p><code>okabe_ito</code> (default)
</p>
</li>
<li> <p><code>bright</code>
</p>
</li>
<li> <p><code>contrast</code>
</p>
</li>
<li> <p><code>dark</code>
</p>
</li>
<li> <p><code>kelly</code>
</p>
</li>
<li> <p><code>light</code>
</p>
</li>
<li> <p><code>muted</code>
</p>
</li>
<li> <p><code>pale</code>
</p>
</li>
<li> <p><code>r4</code>
</p>
</li>
<li> <p><code>safe</code>
</p>
</li>
<li> <p><code>vibrant</code>
</p>
</li></ul>

<p>By default, sequential colours from above palettes will be chosen. To choose the most distant colours set <code>ColSeq = TRUE</code>.
</p>
<p>Sequential quantitative palettes:
</p>

<ul>
<li> <p><code>grey_conti</code>
</p>
</li>
<li> <p><code>blue_conti</code>
</p>
</li>
<li> <p><code>yellow_conti</code>
</p>
</li></ul>

<p>Divergent quantitative palettes:
</p>

<ul>
<li> <p><code>OrBl_div</code>
</p>
</li>
<li> <p><code>PrGn_div</code>
</p>
</li></ul>



<h3>Value</h3>

<p>ggplot scale_fill function for discrete colours.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#add a grafify fill scheme to ggplot
ggplot(emmeans::neuralgia, aes(x = Treatment, 
                               y = Duration))+
  geom_boxplot(aes(fill = Treatment), 
               alpha = .6)+
  geom_point(aes(colour = Treatment,
                 shape = Treatment), 
             size = 3)+
  scale_fill_grafify(palette = "bright")+
  scale_colour_grafify(palette = "bright")+
  facet_wrap("Sex")+
  theme_classic()
#distant colours `ColSeq = FALSE`   
ggplot(emmeans::neuralgia, aes(x = Treatment, 
                               y = Duration))+
  geom_boxplot(aes(fill = Treatment), 
               alpha = .6)+
  geom_point(aes(colour = Treatment,
                 shape = Treatment), 
             size = 3)+
  scale_fill_grafify(palette = "bright",
                     ColSeq = FALSE)+
  scale_colour_grafify(palette = "bright",
                       ColSeq = FALSE)+
  facet_wrap("Sex")+
  theme_classic()
#quantitative colour schemes
ggplot(mtcars, aes(x = disp,
                   y = mpg))+
  geom_point(aes(colour = cyl), 
             size = 3)+
  scale_colour_grafify(palette = "blue_conti")

</code></pre>

<hr>
<h2 id='simple_anova'>ANOVA table from a linear model fit to data.</h2><span id='topic+simple_anova'></span>

<h3>Description</h3>

<p>One of two functions for simple ANOVA tables and linear models without random effects, which use <code><a href="stats.html#topic+lm">lm</a></code> to fit a linear models.
</p>

<ol>
<li> <p><code>link{simple_anova}</code>
</p>
</li>
<li> <p><code>link{simple_model}</code>
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>simple_anova(data, Y_value, Fixed_Factor, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simple_anova_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="simple_anova_+3A_y_value">Y_value</code></td>
<td>
<p>name of column containing quantitative (dependent) variable, provided within &quot;quotes&quot;. The following transformations are permitted: &quot;log(Y_value)&quot;, &quot;log(Y_value + c)&quot; where c a positive number, &quot;logit(Y_value)&quot; or &quot;logit(Y_value/100)&quot; which may be useful when <code>Y_value</code> are percentages  (note quotes outside the log or logit calls); &quot;sqrt(Y_value)&quot; or &quot;(Y_value)^2&quot; should also work. During posthoc-comparisons, log and logit transformations will be back-transformed to the original scale. Other transformations, e.g., &quot;sqrt(Y_value)&quot; will not be back-transformed. Check out the <code><a href="emmeans.html#topic+regrid">regrid</a></code> and <code><a href="emmeans.html#topic+ref_grid">ref_grid</a></code> for details if you need back-transformation to the response scale.</p>
</td></tr>
<tr><td><code id="simple_anova_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>name(s) of categorical fixed factors (independent variables) provided within quotes (e.g., &quot;A&quot;) or as a vector if more than one (e.g., c(&quot;A&quot;, &quot;B&quot;). If a numeric variable(s) is used, transformations similar to <code>Y_value</code> are permitted.</p>
</td></tr>
<tr><td><code id="simple_anova_+3A_...">...</code></td>
<td>
<p>any additional argument to pass on to <code><a href="stats.html#topic+lm">lm</a></code> if required.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Update in v0.2.1: This function uses <code><a href="stats.html#topic+lm">lm</a></code> to fit a linear model to data, passes it on to <code><a href="car.html#topic+Anova">Anova</a></code>, and outputs the ANOVA table with type II sum of squares with F statistics and <em>P</em> values.
(Previous versions produced type I sum of squares using <code><a href="stats.html#topic+anova">anova</a></code> call.)
</p>
<p>It requires a data table, one quantitative dependent variable and one or more independent variables. If your experiment design has random factors, use the related function <code><a href="#topic+mixed_anova">mixed_anova</a></code>.
</p>
<p>This function is related to <code>link{simple_model}</code>.
</p>


<h3>Value</h3>

<p>ANOVA table of class &quot;anova&quot; and &quot;data.frame&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Basic usage 
simple_anova(data = data_doubling_time, 
Y_value = "Doubling_time", 
Fixed_Factor = "Student")

</code></pre>

<hr>
<h2 id='simple_model'>Model from a linear model fit to data.</h2><span id='topic+simple_model'></span>

<h3>Description</h3>

<p>One of two functions for simple ANOVA tables and linear models without random effects, which use <code><a href="stats.html#topic+lm">lm</a></code> to fit a linear models.
</p>

<ol>
<li> <p><code>link{simple_anova}</code>
</p>
</li>
<li> <p><code>link{simple_model}</code>
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>simple_model(data, Y_value, Fixed_Factor, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simple_model_+3A_data">data</code></td>
<td>
<p>a data table object, e.g. data.frame or tibble.</p>
</td></tr>
<tr><td><code id="simple_model_+3A_y_value">Y_value</code></td>
<td>
<p>name of column containing quantitative (dependent) variable, provided within &quot;quotes&quot;. The following transformations are permitted: &quot;log(Y_value)&quot;, &quot;log(Y_value + c)&quot; where c a positive number, &quot;logit(Y_value)&quot; or &quot;logit(Y_value/100)&quot; which may be useful when <code>Y_value</code> are percentages  (note quotes outside the log or logit calls); &quot;sqrt(Y_value)&quot; or &quot;(Y_value)^2&quot; should also work. During posthoc-comparisons, log and logit transformations will be back-transformed to the original scale. Other transformations, e.g., &quot;sqrt(Y_value)&quot; will not be back-transformed. Check out the <code><a href="emmeans.html#topic+regrid">regrid</a></code> and <code><a href="emmeans.html#topic+ref_grid">ref_grid</a></code> for details if you need back-transformation to the response scale.</p>
</td></tr>
<tr><td><code id="simple_model_+3A_fixed_factor">Fixed_Factor</code></td>
<td>
<p>name(s) of categorical fixed factors (independent variables) provided within quotes (e.g., &quot;A&quot;) or as a vector if more than one (e.g., c(&quot;A&quot;, &quot;B&quot;). If a numeric variable(s) is used, transformations similar to <code>Y_value</code> are permitted.</p>
</td></tr>
<tr><td><code id="simple_model_+3A_...">...</code></td>
<td>
<p>any additional arguments to pass on to <code><a href="stats.html#topic+lm">lm</a></code> if required.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Update in v0.2.1: This function uses <code><a href="stats.html#topic+lm">lm</a></code> to fit a linear model to data, passes it on to <code><a href="car.html#topic+Anova">Anova</a></code>, and outputs the ANOVA table with type II sum of squares with F statistics and <em>P</em> values.
</p>
<p>(Previous versions produced type I sum of squares using <code><a href="stats.html#topic+anova">anova</a></code> call.)
It requires a data table, one quantitative dependent variable and one or more independent variables.
</p>
<p>The model output can be used to extract coefficients and other information, including post-hoc comparisons. If your experiment design has random factors, use the related function <code><a href="#topic+mixed_model">mixed_model</a></code>.
</p>
<p>This function is related to <code>link{simple_anova}</code>.
Output of this function can be used with <code><a href="#topic+posthoc_Pairwise">posthoc_Pairwise</a></code>, <code><a href="#topic+posthoc_Levelwise">posthoc_Levelwise</a></code> and <code><a href="#topic+posthoc_vsRef">posthoc_vsRef</a></code>, or with <code><a href="emmeans.html#topic+emmeans">emmeans</a></code>.
</p>


<h3>Value</h3>

<p>This function returns an object of class &quot;lm&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#fixed factors provided as a vector
Doubmodel &lt;- simple_model(data = data_doubling_time,
Y_value =  "Doubling_time", 
Fixed_Factor = "Student")
#get summary
summary(Doubmodel)
</code></pre>

<hr>
<h2 id='table_summary'>Get numeric summary grouped by factors</h2><span id='topic+table_summary'></span>

<h3>Description</h3>

<p>This is a wrapper around <code><a href="stats.html#topic+aggregate">aggregate</a></code> function in base R to obtain mean, median, standard deviation and count for quantitative variable(s) grouped by one or more factors. More than one column containing of quantitative variables can be passed on, and summaries for each is provided with column names with a <code>.</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table_summary(data, Ycol, ByGroup)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="table_summary_+3A_data">data</code></td>
<td>
<p>name of the data table.</p>
</td></tr>
<tr><td><code id="table_summary_+3A_ycol">Ycol</code></td>
<td>
<p>name of one column (in quotes) or a vector of column names (e.g., c(&quot;Y1&quot;, &quot;Y2&quot;)) containing the numerical variable to be summarised.</p>
</td></tr>
<tr><td><code id="table_summary_+3A_bygroup">ByGroup</code></td>
<td>
<p>name of one column (in quotes) or a vector of column names (e.g., c(&quot;A&quot;, &quot;B&quot;)) containing the grouping factors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>this function takes in a data.frame or tibble and returns a data.frame or tibble.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>table_summary(Ycol = "cty",
ByGroup = c("fl", "drv"),
data = mpg)

</code></pre>

<hr>
<h2 id='table_x_reorder'>Reordering groups along X-axis</h2><span id='topic+table_x_reorder'></span>

<h3>Description</h3>

<p>This simple function takes in a data table and reorders groups (categorical variables or factors) to be plotted along the X-axis in a user-defined order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table_x_reorder(data, xcol, OrderX, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="table_x_reorder_+3A_data">data</code></td>
<td>
<p>a data table or tibble.</p>
</td></tr>
<tr><td><code id="table_x_reorder_+3A_xcol">xcol</code></td>
<td>
<p>name of column in above data table (in quotes), e.g., &quot;A&quot;, whose levels are to be reordered.</p>
</td></tr>
<tr><td><code id="table_x_reorder_+3A_orderx">OrderX</code></td>
<td>
<p>a vector of group names within the column selected in <code>xcol</code> in the desired order, .e.g., c(&quot;D&quot;, &quot;A&quot;, &quot;C&quot;).</p>
</td></tr>
<tr><td><code id="table_x_reorder_+3A_...">...</code></td>
<td>
<p>any additional arguments for <code>factor</code> call.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It uses two base R functions: <code>as.factor</code> to first force the user-selected column into a factor, and <code>factor</code> that reorders levels based on a user-provided vector.
</p>


<h3>Value</h3>

<p>This function returns a data frame with a selected column converted into factor with reordered levels.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#reorder levels within Genotype
new_data &lt;- table_x_reorder(data_t_pratio, 
xcol = "Genotype", 
OrderX = c("KO", "WT"))
#compare 
plot_scatterbox(data_t_pratio, 
Genotype, 
Cytokine)
#with
plot_scatterbox(new_data, 
Genotype, 
Cytokine)
#also works within the plot call
plot_scatterbox(data = table_x_reorder(data_t_pratio, 
xcol = "Genotype", 
OrderX = c("KO", "WT")), 
xcol = Genotype, 
ycol = Cytokine)

</code></pre>

<hr>
<h2 id='theme_grafify'>A modified <code>theme_classic()</code> for <code>grafify</code>-like graphs.</h2><span id='topic+theme_grafify'></span>

<h3>Description</h3>

<p>This is a slightly modified <code><a href="ggplot2.html#topic+theme_classic">theme_classic</a></code> with two key differences: no border &amp; background for facet panel labels, and font size of text on axes is the same as that of the axes titles (prior to v3.2.0, this was 0.85 times the base font size). The size of text legend title is also same as base font.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_grafify(
  base_size = 20,
  base_family = "",
  base_line_size = base_size/22,
  base_rect_size = base_size/22,
  TextXAngle = 0,
  vjust = 0,
  hjust = 0,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="theme_grafify_+3A_base_size">base_size</code></td>
<td>
<p>base font size for all text (default is 20). Other text is relative to this.</p>
</td></tr>
<tr><td><code id="theme_grafify_+3A_base_family">base_family</code></td>
<td>
<p>default font family</p>
</td></tr>
<tr><td><code id="theme_grafify_+3A_base_line_size">base_line_size</code></td>
<td>
<p>default line size (default is base font size/22)</p>
</td></tr>
<tr><td><code id="theme_grafify_+3A_base_rect_size">base_rect_size</code></td>
<td>
<p>default size of rectangles (default is base font size/22)</p>
</td></tr>
<tr><td><code id="theme_grafify_+3A_textxangle">TextXAngle</code></td>
<td>
<p>orientation of text on X-axis; default 0 degrees. Change to 45 or 90 to remove overlapping text.</p>
</td></tr>
<tr><td><code id="theme_grafify_+3A_vjust">vjust</code></td>
<td>
<p>vertical adjustment of X-axis text alignment (between 0 and 1). Set <code>hjust</code> and <code>vjust</code> to 1 if <code>TextXAngle = 45</code>. Try other options if using other angles.</p>
</td></tr>
<tr><td><code id="theme_grafify_+3A_hjust">hjust</code></td>
<td>
<p>horizontal adjustment of X-axis text alignment (between 0 and 1).  Set <code>hjust</code> and <code>vjust</code> to 1 if <code>TextXAngle = 45</code>. Try other options if using other angles.</p>
</td></tr>
<tr><td><code id="theme_grafify_+3A_...">...</code></td>
<td>
<p>for any other arguments to pass to <code>theme</code>. A useful one is <code>aspect.ratio = 1</code> for square plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Since v3.2.0, <code>theme_grafify</code> produces transparent backgrounds.
</p>


<h3>Value</h3>

<p>this returns an output with class &quot;theme&quot; and &quot;gg&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ggplot(mpg, aes(drv, cty, colour = fl))+
geom_jitter(width = 0.2, 
size = 3, alpha = .7)+
theme_grafify()

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
