<!DOCTYPE html><html lang="en"><head><title>Help for package GRAPE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GRAPE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#getPathwayScores'><p>Calculate Pathway Scores</p></a></li>
<li><a href='#makeBinaryTemplateAndProbabilityTemplate'><p>Make binary template and probability template</p></a></li>
<li><a href='#makeGRAPE_psMat'><p>Calculate Pathway Space Matrix</p></a></li>
<li><a href='#makePairwiseOrder'><p>Make pairwise order representation of a sample</p></a></li>
<li><a href='#makePairwiseOrderNames'><p>Make template names from gene names</p></a></li>
<li><a href='#predictClassDIRAC'><p>DIRAC Classification</p></a></li>
<li><a href='#predictClassGRAPE'><p>GRAPE Classification</p></a></li>
<li><a href='#predictClassPC'><p>PC Classification</p></a></li>
<li><a href='#w_quad'><p>Quadratic weight function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene-Ranking Analysis of Pathway Expression</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Michael Klein &lt;michael.klein@yale.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Klein &lt;michael.klein@yale.edu&gt;</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>Description:</td>
<td>Gene-Ranking Analysis of Pathway Expression (GRAPE) is a tool for
   summarizing the consensus behavior of biological pathways in the form of a
   template, and for quantifying the extent to which individual samples deviate
   from the template. GRAPE templates are based only on the relative rankings
   of the genes within the pathway and can be used for classification of tissue
   types or disease subtypes. GRAPE can be used to represent gene-expression
   samples as vectors of pathway scores, where each pathway score indicates the
   departure from a given collection of reference samples. The resulting pathway-
   space representation can be used as the feature set for various applications,
   including survival analysis and drug-response prediction. 
   Users of GRAPE should use the following citation:
   Klein MI, Stern DF, and Zhao H. GRAPE: A pathway template method to characterize 
   tissue-specific functionality from gene expression profiles. 
   BMC Bioinformatics, 18:317 (June 2017).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-05-07 17:44:01 UTC; michaelklein</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-05-07 22:12:24 UTC</td>
</tr>
</table>
<hr>
<h2 id='getPathwayScores'>Calculate Pathway Scores</h2><span id='topic+getPathwayScores'></span>

<h3>Description</h3>

<p>Calculate pathway scores of a single pathway of a set of samples relative to a reference set of samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPathwayScores(refmat, newmat, w = w_quad)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getPathwayScores_+3A_refmat">refmat</code></td>
<td>
<p>Pathway expression matrix of reference samples. Rows are genes, columns are samples.</p>
</td></tr>
<tr><td><code id="getPathwayScores_+3A_newmat">newmat</code></td>
<td>
<p>Pathway expression matrix of new samples. Rows are genes, columns are samples.</p>
</td></tr>
<tr><td><code id="getPathwayScores_+3A_w">w</code></td>
<td>
<p>Weight function. Default is quadratic weight function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of pathway scores of each sample in newmat.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Toy example: 50 reference samples
set.seed(10);
refmat &lt;- matrix(rnorm(5*50),nrow=5,ncol=50); rownames(refmat) &lt;- paste0("g",1:5)
### make g2 and g5 larger in refmat
refmat[2,] &lt;- rnorm(50,3,2); refmat[5,] &lt;- rnorm(50,4,4)
### 15 new samples
newmat &lt;- matrix(rnorm(5*15),nrow=5,ncol=15); rownames(newmat) &lt;- paste0("g",1:5)
### make g2 and g3 larger in newmat
newmat[2,] &lt;- rnorm(15,2,3); newmat[3,] &lt;- rnorm(15,4,3)
ps_new &lt;- getPathwayScores(refmat,newmat) ### get pathway scores of new samples
ps_ref &lt;- getPathwayScores(refmat,refmat) ### get pathway scores of reference samples
ps_both &lt;- getPathwayScores(refmat,cbind(refmat,newmat)) ### get pathway scores of both
# &gt; ps_new
# [1]  6.2720  8.5696  9.9904  6.9056  3.7824  8.9344 13.0880 10.2912  3.7824
# 0.0384 13.1136  6.8032  4.8512 12.8512 10.2912
</code></pre>

<hr>
<h2 id='makeBinaryTemplateAndProbabilityTemplate'>Make binary template and probability template</h2><span id='topic+makeBinaryTemplateAndProbabilityTemplate'></span>

<h3>Description</h3>

<p>Takes in matrix, where columns are samples and rows are pathway genes, outputs the binary and probability templates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeBinaryTemplateAndProbabilityTemplate(submat)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="makeBinaryTemplateAndProbabilityTemplate_+3A_submat">submat</code></td>
<td>
<p>A matrix where columns are samples and rows are pathway genes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing binary template vector and probability template vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>submat &lt;- cbind(c(1,3,2,1.5),c(2,3,1.5,1.2),c(1.4,4.2,3.5,3.8))
rownames(submat) &lt;- c("gene_A","gene_B","gene_C","gene_D")
temp &lt;- makeBinaryTemplateAndProbabilityTemplate(submat)
bt &lt;- temp$binary_template; pt &lt;- temp$probability_template
cbind(bt,pt)
</code></pre>

<hr>
<h2 id='makeGRAPE_psMat'>Calculate Pathway Space Matrix</h2><span id='topic+makeGRAPE_psMat'></span>

<h3>Description</h3>

<p>Represents new samples as vectors of pathway scores relative to reference samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeGRAPE_psMat(refge, newge, pathway_list, w = w_quad)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="makeGRAPE_psMat_+3A_refge">refge</code></td>
<td>
<p>Gene expression matrix of reference samples. Rows are genes, columns are samples.</p>
</td></tr>
<tr><td><code id="makeGRAPE_psMat_+3A_newge">newge</code></td>
<td>
<p>Gene expression matrix of new samples. Rows are genes, columns are samples.</p>
</td></tr>
<tr><td><code id="makeGRAPE_psMat_+3A_pathway_list">pathway_list</code></td>
<td>
<p>List of pathways. Each pathway is a character vector consisting of gene names.</p>
</td></tr>
<tr><td><code id="makeGRAPE_psMat_+3A_w">w</code></td>
<td>
<p>Weight function. Default is quadratic weight function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of pathway scores of each sample in newmat.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#' ### Make pathway scores mat
set.seed(10)
### 50 reference samples
refge &lt;- matrix(rnorm(10*50),nrow=10,ncol=50); rownames(refge) &lt;- paste0("g",1:10)
refge[c(2,5,8),] &lt;- matrix(rnorm(3*50,mean=2,sd=2))
refge[c(3,4,7),] &lt;- matrix(rnorm(3*50,mean=4,sd=4))
### 6 new samples
newge &lt;- matrix(rnorm(10*6),nrow=10,ncol=6); rownames(newge) &lt;- paste0("g",1:10)
newge[c(2:7),] &lt;- matrix(rnorm(6*6,mean=3,sd=1))
newge[c(1,9),] &lt;- matrix(rnorm(2*6,mean=5,sd=3))
pathway_list &lt;- list(set1=paste0("g",1:4),set2=paste0("g",5:10),set3=paste0("g",c(1,4,8:10)))
psmat &lt;- makeGRAPE_psMat(refge,newge,pathway_list)
# &gt; psmat
# [,1]     [,2]     [,3]     [,4]     [,5]     [,6]
# set1 2.397426 1.406275 2.516492 2.358809 2.555109 2.358809
# set2 0.670354 3.245575 3.962389 2.670354 1.741150 1.579646
# set3 1.536017 2.167373 2.167373 2.167373 2.148305 1.809322
</code></pre>

<hr>
<h2 id='makePairwiseOrder'>Make pairwise order representation of a sample</h2><span id='topic+makePairwiseOrder'></span>

<h3>Description</h3>

<p>Takes in a vector of gene expression values and returns a binary vector consisting of the pairwise rankings for the sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makePairwiseOrder(samp)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="makePairwiseOrder_+3A_samp">samp</code></td>
<td>
<p>A vector of gene expression values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Binary vector of the pairwise ranking representation of the samples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>samp &lt;- c(1,3,2,1.5)
makePairwiseOrder(samp)
</code></pre>

<hr>
<h2 id='makePairwiseOrderNames'>Make template names from gene names</h2><span id='topic+makePairwiseOrderNames'></span>

<h3>Description</h3>

<p>Takes in vector of pathway gene names, returns names corresponding to the pairwise binary representation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makePairwiseOrderNames(path_genes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="makePairwiseOrderNames_+3A_path_genes">path_genes</code></td>
<td>
<p>A vector of pathway gene names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Names for the pairwise representation, of the form &quot;gA&lt;gB&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path_genes &lt;- c("gene_A","gene_B","gene_C","gene_D")
makePairwiseOrderNames(path_genes)
</code></pre>

<hr>
<h2 id='predictClassDIRAC'>DIRAC Classification</h2><span id='topic+predictClassDIRAC'></span>

<h3>Description</h3>

<p>Classification of a samples according to dirac distances from templates. Usually applied to the gene expression values for a single pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predictClassDIRAC(trainmat, testmat, train_labels)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predictClassDIRAC_+3A_trainmat">trainmat</code></td>
<td>
<p>Matrix of gene expression for set of genes accross training set samples. Each column is a sample.</p>
</td></tr>
<tr><td><code id="predictClassDIRAC_+3A_testmat">testmat</code></td>
<td>
<p>Matrix of gene expression for set of genes accross test set samples. Each column is a sample.</p>
</td></tr>
<tr><td><code id="predictClassDIRAC_+3A_train_labels">train_labels</code></td>
<td>
<p>Vector of class labels for each sample in the training set.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Predicted class labels for test set
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Toy example of two classes
set.seed(10); path_genes &lt;- c("gA","gB","gC","gD"); nsamps &lt;- 50 # Four genes, 50 samples per class
class_one_samps &lt;- matrix(NA,nrow=length(path_genes),ncol=nsamps) # Class 1
rownames(class_one_samps) &lt;- path_genes
class_one_samps[1,] &lt;- rnorm(ncol(class_one_samps),4,2)
class_one_samps[2,] &lt;- rnorm(ncol(class_one_samps),5,4)
class_one_samps[3,] &lt;- rnorm(ncol(class_one_samps),1,1)
class_one_samps[4,] &lt;- rnorm(ncol(class_one_samps),2,1)
class_two_samps &lt;- matrix(NA,nrow=length(path_genes),ncol=nsamps) # Class 2
rownames(class_two_samps) &lt;- path_genes
class_two_samps[1,] &lt;- rnorm(ncol(class_two_samps),2,3)
class_two_samps[2,] &lt;- rnorm(ncol(class_two_samps),5,2)
class_two_samps[3,] &lt;- rnorm(ncol(class_two_samps),1,1)
class_two_samps[4,] &lt;- rnorm(ncol(class_two_samps),0,1)
all_samps &lt;- cbind(class_one_samps,class_two_samps)
labels &lt;- c(rep(1,nsamps),rep(2,nsamps))
testid &lt;- sample.int(100,20)
trainmat &lt;- all_samps[,-testid]
train_labels &lt;- labels[-testid]
testmat &lt;- all_samps[,testid]
test_labels &lt;- labels[testid]
yhat &lt;- predictClassDIRAC(trainmat,testmat,train_labels)
sum(diag(table(test_labels,yhat)))/length(test_labels) # accuracy
# [1] 0.7
</code></pre>

<hr>
<h2 id='predictClassGRAPE'>GRAPE Classification</h2><span id='topic+predictClassGRAPE'></span>

<h3>Description</h3>

<p>Classification of a samples according to grape distances from templates. Usually applied to the gene expression values for a single pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predictClassGRAPE(trainmat, testmat, train_labels, w = w_quad)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predictClassGRAPE_+3A_trainmat">trainmat</code></td>
<td>
<p>Matrix of gene expression for set of genes accross training set samples. Each column is a sample.</p>
</td></tr>
<tr><td><code id="predictClassGRAPE_+3A_testmat">testmat</code></td>
<td>
<p>Matrix of gene expression for set of genes accross test set samples. Each column is a sample.</p>
</td></tr>
<tr><td><code id="predictClassGRAPE_+3A_train_labels">train_labels</code></td>
<td>
<p>Vector of class labels for each sample in the training set.</p>
</td></tr>
<tr><td><code id="predictClassGRAPE_+3A_w">w</code></td>
<td>
<p>Weight function. Default is quadratic weight function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Predicted class labels for test set
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Toy example of two classes
set.seed(10); path_genes &lt;- c("gA","gB","gC","gD"); nsamps &lt;- 50 # Four genes, 50 samples per class
class_one_samps &lt;- matrix(NA,nrow=length(path_genes),ncol=nsamps) # Class 1
rownames(class_one_samps) &lt;- path_genes
class_one_samps[1,] &lt;- rnorm(ncol(class_one_samps),4,2)
class_one_samps[2,] &lt;- rnorm(ncol(class_one_samps),5,4)
class_one_samps[3,] &lt;- rnorm(ncol(class_one_samps),1,1)
class_one_samps[4,] &lt;- rnorm(ncol(class_one_samps),2,1)
class_two_samps &lt;- matrix(NA,nrow=length(path_genes),ncol=nsamps) # Class 2
rownames(class_two_samps) &lt;- path_genes
class_two_samps[1,] &lt;- rnorm(ncol(class_two_samps),2,3)
class_two_samps[2,] &lt;- rnorm(ncol(class_two_samps),5,2)
class_two_samps[3,] &lt;- rnorm(ncol(class_two_samps),1,1)
class_two_samps[4,] &lt;- rnorm(ncol(class_two_samps),0,1)
all_samps &lt;- cbind(class_one_samps,class_two_samps)
labels &lt;- c(rep(1,nsamps),rep(2,nsamps))
testid &lt;- sample.int(100,20)
trainmat &lt;- all_samps[,-testid]
train_labels &lt;- labels[-testid]
testmat &lt;- all_samps[,testid]
test_labels &lt;- labels[testid]
yhat &lt;- predictClassGRAPE(trainmat,testmat,train_labels,w_quad)
sum(diag(table(test_labels,yhat)))/length(test_labels) # accuracy
# [1] 0.8
</code></pre>

<hr>
<h2 id='predictClassPC'>PC Classification</h2><span id='topic+predictClassPC'></span>

<h3>Description</h3>

<p>Classification of a samples according to euclidean distances from PC templates. Usually applied to the gene expression values for a single pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predictClassPC(trainmat, testmat, train_labels)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predictClassPC_+3A_trainmat">trainmat</code></td>
<td>
<p>Matrix of gene expression for set of genes accross training set samples. Each column is a sample.</p>
</td></tr>
<tr><td><code id="predictClassPC_+3A_testmat">testmat</code></td>
<td>
<p>Matrix of gene expression for set of genes accross test set samples. Each column is a sample.</p>
</td></tr>
<tr><td><code id="predictClassPC_+3A_train_labels">train_labels</code></td>
<td>
<p>Vector of class labels for each sample in the training set.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Predicted class labels for test set
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Toy example of two classes
set.seed(10); path_genes &lt;- c("gA","gB","gC","gD"); nsamps &lt;- 50 # Four genes, 50 samples per class
class_one_samps &lt;- matrix(NA,nrow=length(path_genes),ncol=nsamps) # Class 1
rownames(class_one_samps) &lt;- path_genes
class_one_samps[1,] &lt;- rnorm(ncol(class_one_samps),4,2)
class_one_samps[2,] &lt;- rnorm(ncol(class_one_samps),5,4)
class_one_samps[3,] &lt;- rnorm(ncol(class_one_samps),1,1)
class_one_samps[4,] &lt;- rnorm(ncol(class_one_samps),2,1)
class_two_samps &lt;- matrix(NA,nrow=length(path_genes),ncol=nsamps) # Class 2
rownames(class_two_samps) &lt;- path_genes
class_two_samps[1,] &lt;- rnorm(ncol(class_two_samps),2,3)
class_two_samps[2,] &lt;- rnorm(ncol(class_two_samps),5,2)
class_two_samps[3,] &lt;- rnorm(ncol(class_two_samps),1,1)
class_two_samps[4,] &lt;- rnorm(ncol(class_two_samps),0,1)
all_samps &lt;- cbind(class_one_samps,class_two_samps)
labels &lt;- c(rep(1,nsamps),rep(2,nsamps))
testid &lt;- sample.int(100,20)
trainmat &lt;- all_samps[,-testid]
train_labels &lt;- labels[-testid]
testmat &lt;- all_samps[,testid]
test_labels &lt;- labels[testid]
yhat &lt;- predictClassPC(trainmat,testmat,train_labels)
sum(diag(table(test_labels,yhat)))/length(test_labels) # accuracy
# [1] 0.55
</code></pre>

<hr>
<h2 id='w_quad'>Quadratic weight function</h2><span id='topic+w_quad'></span>

<h3>Description</h3>

<p>Calculates the weights of all input entries. All entries should take values in [0,1].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>w_quad(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="w_quad_+3A_x">x</code></td>
<td>
<p>Any number, vector of matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Weight of each element
</p>


<h3>Examples</h3>

<pre><code class='language-R'>w_quad(0.95)
w_quad(cbind(c(.7,.8),c(.9,.1)))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
