<!DOCTYPE html><html><head><title>Help for package greatR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {greatR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculate_distance'><p>Calculate distance between sample data before and after registration</p></a></li>
<li><a href='#get_approximate_stretch'><p>Get approximate stretch factor</p></a></li>
<li><a href='#plot_heatmap'><p>Visualise distances between samples from different time points</p></a></li>
<li><a href='#plot_registration_results'><p>Plot gene of interest after registration</p></a></li>
<li><a href='#register'><p>Register or synchronize different expression profiles</p></a></li>
<li><a href='#summarise_registration'><p>Summarise registration results</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Gene Registration from Expression and Time-Courses in R</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A tool for registering (aligning) gene expression profiles
    between two species (reference data and data to transform).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://ruthkr.github.io/greatR/">https://ruthkr.github.io/greatR/</a>,
<a href="https://github.com/ruthkr/greatR/">https://github.com/ruthkr/greatR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ruthkr/greatR/issues/">https://github.com/ruthkr/greatR/issues/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>cli, data.table, furrr, future, ggplot2, neldermead,
optimization, scales, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-08 20:31:57 UTC; kristiar</td>
</tr>
<tr>
<td>Author:</td>
<td>Ruth Kristianingsih
    <a href="https://orcid.org/0000-0003-1873-6203"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Alex Calderwood <a href="https://orcid.org/0000-0003-3749-035X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ruth Kristianingsih &lt;ruth.kristianingsih30@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-09 01:50:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculate_distance'>Calculate distance between sample data before and after registration</h2><span id='topic+calculate_distance'></span>

<h3>Description</h3>

<p>Calculate distance between sample data before and after registration
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_distance(results)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_distance_+3A_results">results</code></td>
<td>
<p>Result of registration process using <code><a href="#topic+register">register</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a list of data frames which includes:
</p>
<table>
<tr><td><code>registered</code></td>
<td>
<p>distance between scaled reference and query expressions using registered time points.</p>
</td></tr>
<tr><td><code>original</code></td>
<td>
<p>distance between scaled reference and query expressions using original time points.</p>
</td></tr>
</table>

<hr>
<h2 id='get_approximate_stretch'>Get approximate stretch factor</h2><span id='topic+get_approximate_stretch'></span>

<h3>Description</h3>

<p><code>get_approximate_stretch()</code> is a function to get a stretch factor
estimation given input data. This function will take the time point ranges
of both reference and query data and compare them to estimate the stretch
factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_approximate_stretch(data, reference = "ref", query = "query")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_approximate_stretch_+3A_data">data</code></td>
<td>
<p>Input data frame, either containing all replicates of gene expression or not.</p>
</td></tr>
<tr><td><code id="get_approximate_stretch_+3A_reference">reference</code></td>
<td>
<p>Accession name of reference data.</p>
</td></tr>
<tr><td><code id="get_approximate_stretch_+3A_query">query</code></td>
<td>
<p>Accession name of query data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns an estimation of a stretch factor for registering the data.
</p>

<hr>
<h2 id='plot_heatmap'>Visualise distances between samples from different time points</h2><span id='topic+plot_heatmap'></span>

<h3>Description</h3>

<p><code>plot_heatmap()</code> is a function that allows users to plot distances
between samples from different time points to investigate the similarity of
progression of gene expression states between species before or after
registration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heatmap(
  results,
  type = c("registered", "original"),
  match_timepoints = FALSE,
  title = NULL,
  axis_fontsize = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_heatmap_+3A_results">results</code></td>
<td>
<p>Results containing distances between two different reference and query data, output of <code><a href="#topic+calculate_distance">calculate_distance</a></code>.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_type">type</code></td>
<td>
<p>Type of plot, determines whether to use &quot;registered&quot; or &quot;original&quot; time points. By default, &quot;registered&quot;.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_match_timepoints">match_timepoints</code></td>
<td>
<p>If <code>TRUE</code>, will match query time points to reference time points.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_title">title</code></td>
<td>
<p>Optional plot title.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_axis_fontsize">axis_fontsize</code></td>
<td>
<p>Font size of X and Y axes labels.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Distance heatmap of gene expression profiles over time between two different species.
</p>

<hr>
<h2 id='plot_registration_results'>Plot gene of interest after registration</h2><span id='topic+plot_registration_results'></span>

<h3>Description</h3>

<p>Plot gene of interest after registration
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_registration_results(
  results,
  type = c("registered", "original"),
  genes_list = NULL,
  title = NULL,
  ncol = NULL,
  plot_mean_data = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_registration_results_+3A_results">results</code></td>
<td>
<p>Registration results, output of the <code><a href="#topic+register">register</a></code> registration process.</p>
</td></tr>
<tr><td><code id="plot_registration_results_+3A_type">type</code></td>
<td>
<p>Type of plot, determines whether to use &quot;registered&quot; or &quot;original&quot; time points. By default, &quot;registered&quot;.</p>
</td></tr>
<tr><td><code id="plot_registration_results_+3A_genes_list">genes_list</code></td>
<td>
<p>Optional vector indicating the <code>gene_id</code> values to be plotted.</p>
</td></tr>
<tr><td><code id="plot_registration_results_+3A_title">title</code></td>
<td>
<p>Optional plot title.</p>
</td></tr>
<tr><td><code id="plot_registration_results_+3A_ncol">ncol</code></td>
<td>
<p>Number of columns in the plot grid. By default this is calculated automatically.</p>
</td></tr>
<tr><td><code id="plot_registration_results_+3A_plot_mean_data">plot_mean_data</code></td>
<td>
<p>Whether mean data is displayed or not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot of genes of interest after registration process (<code>type = "registered"</code>) or showing original time points (<code>type = "original"</code>).
</p>

<hr>
<h2 id='register'>Register or synchronize different expression profiles</h2><span id='topic+register'></span>

<h3>Description</h3>

<p><code>register()</code> is a function to register expression profiles a user
wishes to compare.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>register(
  input,
  stretches = NA,
  shifts = NA,
  reference,
  query,
  scaling_method = c("none", "z-score", "min-max"),
  overlapping_percent = 50,
  optimise_registration_parameters = TRUE,
  optimisation_method = c("nm", "lbfgsb", "sa"),
  optimisation_config = NULL,
  exp_sd = NA,
  num_cores = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="register_+3A_input">input</code></td>
<td>
<p>Input data frame containing all replicates of gene expression in each genotype at each time point.</p>
</td></tr>
<tr><td><code id="register_+3A_stretches">stretches</code></td>
<td>
<p>Candidate registration stretch factors to apply to query data, only required if <code>optimise_registration_parameters = FALSE</code>.</p>
</td></tr>
<tr><td><code id="register_+3A_shifts">shifts</code></td>
<td>
<p>Candidate registration shift values to apply to query data, only required if <code>optimise_registration_parameters = FALSE</code>.</p>
</td></tr>
<tr><td><code id="register_+3A_reference">reference</code></td>
<td>
<p>Accession name of reference data.</p>
</td></tr>
<tr><td><code id="register_+3A_query">query</code></td>
<td>
<p>Accession name of query data.</p>
</td></tr>
<tr><td><code id="register_+3A_scaling_method">scaling_method</code></td>
<td>
<p>Scaling method applied to data prior to registration process. Either <code>none</code> (default), <code>z-score</code>, or <code>min-max</code>.</p>
</td></tr>
<tr><td><code id="register_+3A_overlapping_percent">overlapping_percent</code></td>
<td>
<p>Minimum percentage of overlapping time points on the reference data. Shifts will be only considered if it leaves at least this percentage of overlapping time points after applying the registration function.</p>
</td></tr>
<tr><td><code id="register_+3A_optimise_registration_parameters">optimise_registration_parameters</code></td>
<td>
<p>Whether to optimise registration parameters. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="register_+3A_optimisation_method">optimisation_method</code></td>
<td>
<p>Optimisation method to use. Either <code>"nm"</code> for Nelder-Mead (default), <code>"lbfgsb"</code> for L-BFGS-B, or <code>"sa"</code> for Simulated Annealing.</p>
</td></tr>
<tr><td><code id="register_+3A_optimisation_config">optimisation_config</code></td>
<td>
<p>Optional list with arguments to override the default optimisation configuration.</p>
</td></tr>
<tr><td><code id="register_+3A_exp_sd">exp_sd</code></td>
<td>
<p>Optional experimental standard deviation on the expression replicates.</p>
</td></tr>
<tr><td><code id="register_+3A_num_cores">num_cores</code></td>
<td>
<p>Number of cores to use if the user wants to register genes asynchronously (in parallel) in the background on the same machine. By default, <code>NA</code>, the registration will be run without parallelisation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a list of data frames, containing:
</p>
<table>
<tr><td><code>data</code></td>
<td>
<p>a table containing the scaled input data and an additional <code>timepoint_reg</code> column after applying registration parameters to the query data.</p>
</td></tr>
<tr><td><code>model_comparison</code></td>
<td>
<p>a table comparing the optimal registration function for each gene (based on <code>all_shifts_df</code> scores) to model with no registration applied.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load a data frame from the sample data
data_path &lt;- system.file("extdata/brapa_arabidopsis_all_replicates.csv", package = "greatR")
all_data &lt;- utils::read.csv(data_path)

# Running the registration
registration_results &lt;- register(
  input = all_data,
  reference = "Ro18",
  query = "Col0"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='summarise_registration'>Summarise registration results</h2><span id='topic+summarise_registration'></span>

<h3>Description</h3>

<p>Summarise registration results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarise_registration(results)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summarise_registration_+3A_results">results</code></td>
<td>
<p>Registration results, output of the <code><a href="#topic+register">register</a></code> registration process.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a list containing:
</p>
<table>
<tr><td><code>summary</code></td>
<td>
<p>contains result summaries of the registration results.</p>
</td></tr>
<tr><td><code>registered_genes</code></td>
<td>
<p>vector of gene accessions which were successfully registered.</p>
</td></tr>
<tr><td><code>non_registered_genes</code></td>
<td>
<p>vector of non-registered gene accessions.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
