<!DOCTYPE html><html><head><title>Help for package greatR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {greatR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculate_distance'><p>Calculate distance between sample data before and after registration</p></a></li>
<li><a href='#get_approximate_stretch'><p>Get approximate stretch factor</p></a></li>
<li><a href='#plot'><p>Visualise registration results</p></a></li>
<li><a href='#register'><p>Register or synchronize different expression profiles</p></a></li>
<li><a href='#summary'><p>Summarise registration results</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Gene Registration from Expression and Time-Courses in R</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A tool for registering (aligning) gene expression profiles
    between reference and query data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://ruthkr.github.io/greatR/">https://ruthkr.github.io/greatR/</a>,
<a href="https://github.com/ruthkr/greatR/">https://github.com/ruthkr/greatR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ruthkr/greatR/issues/">https://github.com/ruthkr/greatR/issues/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>cli, data.table, furrr, future, ggplot2, neldermead,
optimization, patchwork, scales, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-09 22:21:47 UTC; kristiar</td>
</tr>
<tr>
<td>Author:</td>
<td>Ruth Kristianingsih
    <a href="https://orcid.org/0000-0003-1873-6203"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ruth Kristianingsih &lt;ruth.kristianingsih30@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-09 22:40:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculate_distance'>Calculate distance between sample data before and after registration</h2><span id='topic+calculate_distance'></span>

<h3>Description</h3>

<p><code>calculate_distance()</code> is a function that allows users to calculate
pairwise distances between samples from different time points to
investigate the similarity of progression before or after registration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_distance(results, type = c("registered", "all"), genes_list = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_distance_+3A_results">results</code></td>
<td>
<p>Result of registration process using <code><a href="#topic+register">register()</a></code>.</p>
</td></tr>
<tr><td><code id="calculate_distance_+3A_type">type</code></td>
<td>
<p>Whether to calculate distance considering only &quot;registered&quot; genes (default) or &quot;all&quot; genes.</p>
</td></tr>
<tr><td><code id="calculate_distance_+3A_genes_list">genes_list</code></td>
<td>
<p>Optional vector indicating the <code>gene_id</code> values to be considered.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a <code>dist_greatR</code> object containing two data frames:
</p>
<table>
<tr><td><code>registered</code></td>
<td>
<p>pairwise distance between scaled reference and query expressions using registered time points.</p>
</td></tr>
<tr><td><code>original</code></td>
<td>
<p>pairwise distance between scaled reference and query expressions using original time points.</p>
</td></tr>
</table>

<hr>
<h2 id='get_approximate_stretch'>Get approximate stretch factor</h2><span id='topic+get_approximate_stretch'></span>

<h3>Description</h3>

<p><code>get_approximate_stretch()</code> is a function to get a stretch factor
estimation given input data. This function will take the time point ranges
of both reference and query data and compare them to estimate the stretch
factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_approximate_stretch(data, reference = "ref", query = "query")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_approximate_stretch_+3A_data">data</code></td>
<td>
<p>Input data frame, either containing all replicates of gene expression or not.</p>
</td></tr>
<tr><td><code id="get_approximate_stretch_+3A_reference">reference</code></td>
<td>
<p>Accession name of reference data.</p>
</td></tr>
<tr><td><code id="get_approximate_stretch_+3A_query">query</code></td>
<td>
<p>Accession name of query data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns an estimation of a stretch factor for registering the data.
</p>

<hr>
<h2 id='plot'>Visualise registration results</h2><span id='topic+plot'></span><span id='topic+plot.res_greatR'></span><span id='topic+plot.dist_greatR'></span><span id='topic+plot.summary.res_greatR'></span>

<h3>Description</h3>

<p>Visualise registration results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'res_greatR'
plot(
  x,
  type = c("result", "original"),
  genes_list = NULL,
  show_rep_mean = FALSE,
  ncol = NULL,
  title = NULL,
  ...
)

## S3 method for class 'dist_greatR'
plot(
  x,
  type = c("result", "original"),
  match_timepoints = TRUE,
  title = NULL,
  ...
)

## S3 method for class 'summary.res_greatR'
plot(
  x,
  type = c("all", "registered"),
  type_dist = c("histogram", "density"),
  genes_list = NULL,
  bins = 30,
  alpha = NA,
  scatterplot_size = c(4, 3),
  title = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_+3A_x">x</code></td>
<td>
<p>Input object.
</p>

<ul>
<li><p> For <code><a href="#topic+plot.res_greatR">plot.res_greatR()</a></code>: registration results, output of the <code><a href="#topic+register">register()</a></code> registration process.
</p>
</li>
<li><p> For <code><a href="#topic+plot.summary.res_greatR">plot.summary.res_greatR()</a></code>: registration results summary, output of <code><a href="#topic+summary">summary()</a></code>.
</p>
</li>
<li><p> For <code><a href="#topic+plot.dist_greatR">plot.dist_greatR()</a></code>: pairwise distances between reference and query time points, output of <code><a href="#topic+calculate_distance">calculate_distance()</a></code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot_+3A_type">type</code></td>
<td>
<p>Type of plot.
</p>

<ul>
<li><p> For both <code><a href="#topic+plot.res_greatR">plot.res_greatR()</a></code> and  <code><a href="#topic+plot.dist_greatR">plot.dist_greatR()</a></code>: whether to use registration &quot;result&quot; (default) or &quot;original&quot; time points.
</p>
</li>
<li><p> For <code><a href="#topic+plot.summary.res_greatR">plot.summary.res_greatR()</a></code>: whether to show &quot;all&quot; genes (default) or only &quot;registered&quot; ones.
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot_+3A_genes_list">genes_list</code></td>
<td>
<p>Optional vector indicating the <code>gene_id</code> values to be plotted.</p>
</td></tr>
<tr><td><code id="plot_+3A_show_rep_mean">show_rep_mean</code></td>
<td>
<p>Whether to show <code>replicate</code> mean values.</p>
</td></tr>
<tr><td><code id="plot_+3A_ncol">ncol</code></td>
<td>
<p>Number of columns in the plot grid. By default this is calculated automatically.</p>
</td></tr>
<tr><td><code id="plot_+3A_title">title</code></td>
<td>
<p>Optional plot title.</p>
</td></tr>
<tr><td><code id="plot_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to methods (ignored).</p>
</td></tr>
<tr><td><code id="plot_+3A_match_timepoints">match_timepoints</code></td>
<td>
<p>If <code>TRUE</code>, will match query time points to reference time points.</p>
</td></tr>
<tr><td><code id="plot_+3A_type_dist">type_dist</code></td>
<td>
<p>Type of marginal distribution. Can be either &quot;histogram&quot; (default), or &quot;density&quot;.</p>
</td></tr>
<tr><td><code id="plot_+3A_bins">bins</code></td>
<td>
<p>Number of bins to use when <code>type_dist</code> = &quot;histogram&quot;. By default, 30.</p>
</td></tr>
<tr><td><code id="plot_+3A_alpha">alpha</code></td>
<td>
<p>Optional opacity of the points in the scatterplot.</p>
</td></tr>
<tr><td><code id="plot_+3A_scatterplot_size">scatterplot_size</code></td>
<td>
<p>Vector <code>c(width, height)</code> specifying the ratio of width and height of the scatterplot with respect to stretch and shift distribution plots.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> For <code><a href="#topic+plot.res_greatR">plot.res_greatR()</a></code>: plot of genes of interest after registration process (<code>type = "result"</code>) or showing original time points (<code>type = "original"</code>).
</p>
</li>
<li><p> For <code><a href="#topic+plot.dist_greatR">plot.dist_greatR()</a></code>: distance heatmap of gene expression profiles over time between reference and query.
</p>
</li>
<li><p> For <code><a href="#topic+plot.summary.res_greatR">plot.summary.res_greatR()</a></code>: TODO.
</p>
</li></ul>


<hr>
<h2 id='register'>Register or synchronize different expression profiles</h2><span id='topic+register'></span>

<h3>Description</h3>

<p><code>register()</code> is a function to register expression profiles a user
wishes to compare.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>register(
  input,
  stretches = NA,
  shifts = NA,
  reference,
  query,
  scaling_method = c("none", "z-score", "min-max"),
  overlapping_percent = 50,
  use_optimisation = TRUE,
  optimisation_method = c("lbfgsb", "nm", "sa"),
  optimisation_config = NULL,
  exp_sd = NA,
  num_cores = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="register_+3A_input">input</code></td>
<td>
<p>Input data frame containing all replicates of gene expression in each genotype at each time point.</p>
</td></tr>
<tr><td><code id="register_+3A_stretches">stretches</code></td>
<td>
<p>Candidate registration stretch factors to apply to query data, only required if <code>use_optimisation = FALSE</code>.</p>
</td></tr>
<tr><td><code id="register_+3A_shifts">shifts</code></td>
<td>
<p>Candidate registration shift values to apply to query data, only required if <code>use_optimisation = FALSE</code>.</p>
</td></tr>
<tr><td><code id="register_+3A_reference">reference</code></td>
<td>
<p>Accession name of reference data.</p>
</td></tr>
<tr><td><code id="register_+3A_query">query</code></td>
<td>
<p>Accession name of query data.</p>
</td></tr>
<tr><td><code id="register_+3A_scaling_method">scaling_method</code></td>
<td>
<p>Scaling method applied to data prior to registration process. Either <code>none</code> (default), <code>z-score</code>, or <code>min-max</code>.</p>
</td></tr>
<tr><td><code id="register_+3A_overlapping_percent">overlapping_percent</code></td>
<td>
<p>Minimum percentage of overlapping time point range of the reference data. Shifts will be only considered if it leaves at least this percentage of overlapping time point range after applying the registration.</p>
</td></tr>
<tr><td><code id="register_+3A_use_optimisation">use_optimisation</code></td>
<td>
<p>Whether to optimise registration parameters. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="register_+3A_optimisation_method">optimisation_method</code></td>
<td>
<p>Optimisation method to use. Either <code>"lbfgsb"</code> for L-BFGS-B (default), <code>"nm"</code> for Nelder-Mead, or <code>"sa"</code> for Simulated Annealing.</p>
</td></tr>
<tr><td><code id="register_+3A_optimisation_config">optimisation_config</code></td>
<td>
<p>Optional list with arguments to override the default optimisation configuration.</p>
</td></tr>
<tr><td><code id="register_+3A_exp_sd">exp_sd</code></td>
<td>
<p>Optional experimental standard deviation on the expression replicates.</p>
</td></tr>
<tr><td><code id="register_+3A_num_cores">num_cores</code></td>
<td>
<p>Number of cores to use if the user wants to register genes asynchronously (in parallel) in the background on the same machine. By default, <code>NA</code>, the registration will be run without parallelisation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a <code>res_greatR</code> object containing:
</p>
<table>
<tr><td><code>data</code></td>
<td>
<p>a table containing the scaled input data and an additional <code>timepoint_reg</code> column after applying registration parameters to the query data.</p>
</td></tr>
<tr><td><code>model_comparison</code></td>
<td>
<p>a table comparing the optimal registration function for each gene (based on <code>all_shifts_df</code> scores) to model with no registration applied.</p>
</td></tr>
<tr><td><code>fun_args</code></td>
<td>
<p>a list of arguments used when calling the function.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load a data frame from the sample data
data_path &lt;- system.file("extdata/brapa_arabidopsis_data.csv", package = "greatR")
all_data &lt;- utils::read.csv(data_path)

# Running the registration
registration_results &lt;- register(
  input = all_data,
  reference = "Ro18",
  query = "Col0"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='summary'>Summarise registration results</h2><span id='topic+summary'></span><span id='topic+summary.res_greatR'></span>

<h3>Description</h3>

<p>Summarise registration results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'res_greatR'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_+3A_object">object</code></td>
<td>
<p>Registration results, output of the <code><a href="#topic+register">register()</a></code> registration process.</p>
</td></tr>
<tr><td><code id="summary_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to methods (ignored).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a list containing:
</p>
<table>
<tr><td><code>summary</code></td>
<td>
<p>table containing the summary of the registration results.</p>
</td></tr>
<tr><td><code>registered_genes</code></td>
<td>
<p>vector of gene accessions which were successfully registered.</p>
</td></tr>
<tr><td><code>non_registered_genes</code></td>
<td>
<p>vector of non-registered gene accessions.</p>
</td></tr>
<tr><td><code>reg_params</code></td>
<td>
<p>table containing distribution of registration parameters.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
