<!DOCTYPE html><html lang="en"><head><title>Help for package GRIN2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GRIN2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alex.boxplots'><p>Prepare Box Plots of Expression Data by Lesion Groups</p></a></li>
<li><a href='#alex.pathway'><p>Associate Lesions with Expression Data on the Pathway Level</p></a></li>
<li><a href='#alex.prep.lsn.expr'><p>Prepare Lesion and Expression Data for Kruskal-Wallis Test</p></a></li>
<li><a href='#alex.waterfall.plot'><p>Prepare Waterfall Plot of Lesion and Expression Data</p></a></li>
<li><a href='#alex.waterfall.prep'><p>Prepare Lesion and Expression Data for Waterfall Plots</p></a></li>
<li><a href='#clin.data'><p>Example T-ALL Dataset Clinical Data</p></a></li>
<li><a href='#compute.gw.coordinates'><p>Compute Genome-wide Coordinates</p></a></li>
<li><a href='#count.hits'><p>Count Gene Lesion Hits</p></a></li>
<li><a href='#default.grin.colors'><p>Default GRIN Colors</p></a></li>
<li><a href='#expr.data'><p>Example T-ALL Dataset Gene Expression Data</p></a></li>
<li><a href='#find.gene.lsn.overlaps'><p>Find Gene Lesion Overlaps</p></a></li>
<li><a href='#genomewide.log10q.plot'><p>Genomewide log10q Plot</p></a></li>
<li><a href='#genomewide.lsn.plot'><p>Genome-wide Lesion Plot</p></a></li>
<li><a href='#get.chrom.length'><p>Get Chromosome Length</p></a></li>
<li><a href='#get.ensembl.annotation'><p>Get Ensembl Gene and Regulatory Features Annotation Data</p></a></li>
<li><a href='#grin.assoc.lsn.outcome'><p>Associate Lesions with Clinical Outcomes</p></a></li>
<li><a href='#grin.barplt'><p>GRIN Bar Plot</p></a></li>
<li><a href='#grin.lsn.boundaries'><p>GRIN Evaluate Lesion Boundaries</p></a></li>
<li><a href='#grin.oncoprint.mtx'><p>GRIN OncoPrint Matrix</p></a></li>
<li><a href='#grin.stats'><p>GRIN Statistics Output</p></a></li>
<li><a href='#grin.stats.lsn.plot'><p>GRIN Statistics Lesions Plot</p></a></li>
<li><a href='#hg19_cytoband'><p>GRCh37 Chromosome Cytobands</p></a></li>
<li><a href='#hg19.chrom.size'><p>Chromosome Length Data File</p></a></li>
<li><a href='#hg19.gene.annotation'><p>Example Gene Annotation Data File</p></a></li>
<li><a href='#hg38_cytoband'><p>GRCh38 Chromosome Cytobands</p></a></li>
<li><a href='#KW.hit.express'><p>Associate Lesion with Expression Data Using Kruskal-Wallis Test</p></a></li>
<li><a href='#lesion.data'><p>Example T-ALL Dataset Lesion Data</p></a></li>
<li><a href='#lsn.transcripts.plot'><p>Lesions Gene Transcripts Plot</p></a></li>
<li><a href='#onco.print.props'><p>Oncoprint proportions</p></a></li>
<li><a href='#order.index.gene.data'><p>Order Index Gene Data</p></a></li>
<li><a href='#order.index.lsn.data'><p>Order Index Lesion Data</p></a></li>
<li><a href='#pathways'><p>List of Genes Annotated to a Group of Pathways</p></a></li>
<li><a href='#prep.binary.lsn.mtx'><p>Prepare Binary Lesion Matrix</p></a></li>
<li><a href='#prep.gene.lsn.data'><p>Prepare Gene and Lesion Data</p></a></li>
<li><a href='#prep.lsn.type.matrix'><p>Prepare Lesion Type Matrix</p></a></li>
<li><a href='#prob.hits'><p>Find Probablity of Locus Hit</p></a></li>
<li><a href='#top.alex.waterfall.plots'><p>Waterfall Plots for Lesion and Expression Data of Top Significant Genes</p></a></li>
<li><a href='#write.grin.xlsx'><p>Write GRIN Results</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Genomic Random Interval (GRIN)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Improved version of 'GRIN' software that streamlines its use in practice to analyze genomic lesion data, accelerate its computing, and expand its analysis capabilities to answer additional scientific questions including a rigorous evaluation of the association of genomic lesions with RNA expression. Pounds, Stan, et al. (2013) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtt372">doi:10.1093/bioinformatics/btt372</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>biomaRt, circlize, ComplexHeatmap, data.table, dplyr,
EnsDb.Hsapiens.v75, ensembldb, forcats, GenomeInfoDb, ggplot2,
graphics, grDevices, grid, gridGraphics, Gviz, magrittr, stats,
stringr, survival, tibble, tidyselect, utils, writexl</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Abdelrahman Elsayed &lt;aelsayed@stjude.org&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/abdel-elsayed87/GRIN2">https://github.com/abdel-elsayed87/GRIN2</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/abdel-elsayed87/GRIN2/issues">https://github.com/abdel-elsayed87/GRIN2/issues</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-18 22:39:09 UTC; aelsa</td>
</tr>
<tr>
<td>Author:</td>
<td>Abdelrahman Elsayed
    <a href="https://orcid.org/0000-0002-8150-6825"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre,
    cph],
  Xueyuan Cao [aut],
  Lakshmi Anuhya patibandla [aut],
  Stanley Pounds [aut, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-19 12:30:36 UTC</td>
</tr>
</table>
<hr>
<h2 id='alex.boxplots'>Prepare Box Plots of Expression Data by Lesion Groups</h2><span id='topic+alex.boxplots'></span>

<h3>Description</h3>

<p>Function return box plots for expression data by lesion groups for selected number of genes based on a specified q-value of the kruskal-wallis test results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alex.boxplots(out.dir, alex.data, alex.kw.results, q, gene.annotation)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="alex.boxplots_+3A_out.dir">out.dir</code></td>
<td>
<p>Path to the folder where the boxplots of selected genes based on the specified q value of the KW results table will be added.</p>
</td></tr>
<tr><td><code id="alex.boxplots_+3A_alex.data">alex.data</code></td>
<td>
<p>Output of the alex.prep.lsn.expr function. It's a list of three data tables that include &quot;row.mtch&quot;, &quot;alex.expr&quot; with expression data, &quot;alex.lsn&quot; with lesion data. Rows of alex.expr, and &quot;alex.lsn&quot; matrices are ordered by gene ensembl IDs and the columns are ordered by patient ID.</p>
</td></tr>
<tr><td><code id="alex.boxplots_+3A_alex.kw.results">alex.kw.results</code></td>
<td>
<p>ALEX Kruskal-Wallis test results (output of the KW.hit.express function).</p>
</td></tr>
<tr><td><code id="alex.boxplots_+3A_q">q</code></td>
<td>
<p>minimum q value for a gene to be included in output PDF file of box plots.</p>
</td></tr>
<tr><td><code id="alex.boxplots_+3A_gene.annotation">gene.annotation</code></td>
<td>
<p>Gene annotation data either provided by the user or retrieved from ensembl BioMart database using get.ensembl.annotation function included in the GRIN2.0 library. Data.frame should has four columns: &quot;gene&quot; which is the ensembl ID of annotated genes, &quot;chrom&quot; which is the chromosome on which the gene is located, &quot;loc.start&quot; which is the gene start position, and &quot;loc.end&quot; the gene end position.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Function return a PDF file with box plots for expression data by lesion groups for selected number of genes based on a specified q-value of the kruskal-wallis test results (one gene per page).
</p>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023) Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+alex.prep.lsn.expr">alex.prep.lsn.expr()</a></code>, <code><a href="#topic+KW.hit.express">KW.hit.express()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expr.data)
data(lesion.data)
data(hg19.gene.annotation)

# prepare expression, lesion data and return the set of genes with both types of data available
# ordered by gene IDs in rows and patient IDs in columns:
alex.data=alex.prep.lsn.expr(expr.data, lesion.data,
                             hg19.gene.annotation, min.expr=5, min.pts.lsn=5)

# run KW test for association between lesion groups and expression level of the same gene:
alex.kw.results=KW.hit.express(alex.data, hg19.gene.annotation, min.grp.size=5)

# return boxplots for a list of top significant genes to a pre-specified folder using 'out.dir':
dir.create(resultsFolder &lt;- file.path(tempdir(), "temp.out"))

boxplots=alex.boxplots(out.dir=resultsFolder,
                       alex.data, alex.kw.results,
                       1e-15, hg19.gene.annotation)

unlink(resultsFolder, recursive = TRUE)
</code></pre>

<hr>
<h2 id='alex.pathway'>Associate Lesions with Expression Data on the Pathway Level</h2><span id='topic+alex.pathway'></span>

<h3>Description</h3>

<p>Function compute the distance between subjects in the dataset based on the lesions that affect different genes assigned to the pathway of interest and return two panels of lesion and expression data of ordered subjects based on the computed distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alex.pathway(alex.data, lsn.data, pathways, selected.pathway)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="alex.pathway_+3A_alex.data">alex.data</code></td>
<td>
<p>output of the alex.prep.lsn.expr function. It's a list of three data tables that include &quot;row.mtch&quot;, &quot;alex.expr&quot; with expression data, &quot;alex.lsn&quot; with lesion data. Rows of alex.expr, and &quot;alex.lsn&quot; matrices are ordered by gene ensembl IDs and columns are ordered by patient ID.</p>
</td></tr>
<tr><td><code id="alex.pathway_+3A_lsn.data">lsn.data</code></td>
<td>
<p>Lesion data in a GRIN compatible format. data.frame should has five columns that include &quot;ID&quot; with patient ID, &quot;chrom&quot; which is the chromosome on which the lesion is located, &quot;loc.start&quot; which is the lesion start position, &quot;loc.end&quot; the lesion end position and &quot;lsn.type&quot; which is the lesion type for example gain, loss, mutation, fusion, etc...</p>
</td></tr>
<tr><td><code id="alex.pathway_+3A_pathways">pathways</code></td>
<td>
<p>data.frame with three columns &quot;gene.name&quot; that has gene symbols, &quot;ensembl.id&quot; with gene ensembl ID and &quot;pathway&quot; that has the pathway name.</p>
</td></tr>
<tr><td><code id="alex.pathway_+3A_selected.pathway">selected.pathway</code></td>
<td>
<p>The pathway of interest.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function compute the distance between subjects in th dataset based on lesions affecting different genes assigned to the pathway of interest and return two panels of lesion and expression data of ordered subjects based on the computed distances. Function also return a data.frame with lesion and expression data of the pathway genes ordered based on the hierarchical clustering analysis (same order of the subjects in the lesion and expression panels of the figure).
</p>


<h3>Value</h3>

<p>Function will return two panels figure of lesion and expression data of ordered subjects based on the computed distances of lesions in all genes assigned to the pathway of interest. The function will also return:
</p>
<table role = "presentation">
<tr><td><code>ordered.path.data</code></td>
<td>
<p>data.frame with lesion and expression data of the pathway genes ordered based on the hiearchial clustering analysis (same order of the subjects in the lesion and expression panels of the figure).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+alex.prep.lsn.expr">alex.prep.lsn.expr()</a></code>, <code><a href="stats.html#topic+hclust">stats::hclust()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expr.data)
data(lesion.data)
data(hg19.gene.annotation)
data(pathways)

# prepare expression, lesion data and return the set of genes with both types of data available
# ordered by gene IDs in rows and patient IDs in columns:
alex.data=alex.prep.lsn.expr(expr.data, lesion.data,
                             hg19.gene.annotation, min.expr=5,
                             min.pts.lsn=5)

# use lesions in all genes assigned to the jak_pathway as an example pathway:
alex.path=alex.pathway(alex.data, lesion.data, pathways, "Jak_Pathway")
# extract expression and lesion data (same subjects order in the figure)
alex.path
</code></pre>

<hr>
<h2 id='alex.prep.lsn.expr'>Prepare Lesion and Expression Data for Kruskal-Wallis Test</h2><span id='topic+alex.prep.lsn.expr'></span>

<h3>Description</h3>

<p>The function prepares lesion and expression data matrices for the KW.hit.express function that runs the kruskal-Wallis test for the association between lesion groups and expression level of each gene with available lesion and expression data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alex.prep.lsn.expr(
  expr.mtx,
  lsn.data,
  gene.annotation,
  min.expr = NULL,
  min.pts.lsn = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="alex.prep.lsn.expr_+3A_expr.mtx">expr.mtx</code></td>
<td>
<p>Normalized log2 transformed expression data provided by the user with genes in rows and subjects in columns (first column &quot;ensembl.ID&quot; should be gene ensembl IDs).</p>
</td></tr>
<tr><td><code id="alex.prep.lsn.expr_+3A_lsn.data">lsn.data</code></td>
<td>
<p>Lesion data in GRIN compatible format. Data frame should has five columns that include &quot;ID&quot; with patient ID, &quot;chrom&quot; which is the chromosome on which the lesion is located, &quot;loc.start&quot; which is the lesion start position, &quot;loc.end&quot; the lesion end position and &quot;lsn.type&quot; which is the lesion type for example gain, loss, mutation, fusion, etc...</p>
</td></tr>
<tr><td><code id="alex.prep.lsn.expr_+3A_gene.annotation">gene.annotation</code></td>
<td>
<p>Gene annotation data either provided by the user or retrieved from ensembl BioMart database using get.ensembl.annotation function included in the GRIN2.0 library. Data.frame should has four columns: &quot;gene&quot; which is the ensembl ID of annotated genes, &quot;chrom&quot; which is the chromosome on which the gene is located, &quot;loc.start&quot; which is the gene start position, and &quot;loc.end&quot; the gene end position.</p>
</td></tr>
<tr><td><code id="alex.prep.lsn.expr_+3A_min.expr">min.expr</code></td>
<td>
<p>Minimum allowed expression level of the gene (the sum of expression level of the gene in all patients; useful to exclude genes with very low expression)</p>
</td></tr>
<tr><td><code id="alex.prep.lsn.expr_+3A_min.pts.lsn">min.pts.lsn</code></td>
<td>
<p>Minimum number of patients with any type of lesions in a certain gene otherwise the gene will be excluded from the lesion matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function use prep.lsn.type.matrix function to prepare the lesion matrix that has each gene represented in one row with all lesion types included. Next, the function will prepare lesion and expression data matrices for the KW.hit.express function that runs the kruskal-Wallis test. It only keep genes with both lesion and expression data with rows ordered by ensembl ID and columns ordered by patient's ID.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table role = "presentation">
<tr><td><code>alex.expr</code></td>
<td>
<p>Expression data with gene ensembl IDs as row names and patient IDs as column names. Rows are ordered by ensembl ID and columns ordered by patient IDs.</p>
</td></tr>
<tr><td><code>alex.lsn</code></td>
<td>
<p>Lesion data for genes in the expression data matrix with gene ensembl IDs as row names and patient IDs as column names. Rows are ordered by ensembl ID and columns ordered by patient IDs.</p>
</td></tr>
<tr><td><code>alex.row.mtch</code></td>
<td>
<p>Data.frame of two columns with ensembl ID of genes in the expression and lesion data matrices (ID should be the same in the two columns).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+KW.hit.express">KW.hit.express()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expr.data)
data(lesion.data)
data(hg19.gene.annotation)

# prepare expression, lesion data and return the set of genes with both types of data available
# ordered by gene IDs in rows and patient IDs in columns:
alex.data=alex.prep.lsn.expr(expr.data, lesion.data,
                             hg19.gene.annotation, min.expr=1,
                             min.pts.lsn=5)
</code></pre>

<hr>
<h2 id='alex.waterfall.plot'>Prepare Waterfall Plot of Lesion and Expression Data</h2><span id='topic+alex.waterfall.plot'></span>

<h3>Description</h3>

<p>Function return a waterfall plot for expression data by lesion groups of a selected gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alex.waterfall.plot(waterfall.prep, lsn.data, lsn.clrs = NULL, delta = 0.5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="alex.waterfall.plot_+3A_waterfall.prep">waterfall.prep</code></td>
<td>
<p>Output of the alex.waterfall.prep function. It's a list of three data tables that include &quot;gene.lsn.exp&quot; that has patient ID, lesion type that affect this gene if any and expression level of the selected gene, &quot;lsns&quot; which is a data table with all lesions affecting the gene of interest in a GRIN compatible format and &quot;stats&quot; which is one row with the Kruskal-Wallis test result (output of the KW.hit.express function).</p>
</td></tr>
<tr><td><code id="alex.waterfall.plot_+3A_lsn.data">lsn.data</code></td>
<td>
<p>Lesion data in a GRIN compatible format.</p>
</td></tr>
<tr><td><code id="alex.waterfall.plot_+3A_lsn.clrs">lsn.clrs</code></td>
<td>
<p>Assigned colors per lesion types. If not specified, colors will be automatically assigned using default.grin.colors function.</p>
</td></tr>
<tr><td><code id="alex.waterfall.plot_+3A_delta">delta</code></td>
<td>
<p>Spacing argument for the waterfall plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function return a waterfall plot for expression data by lesion groups of a selected gene. This plot offers a side by side graphical representation of lesion and expression data for each patient where lesion groups are ordered alphabetically. For each lesion category, expression data is ordered from the lowest to the highest with patient with the median expression of the gene in the middle of the panel.
</p>


<h3>Value</h3>

<p>Function return a waterfall plot for expression data by lesion groups of a selected gene.
</p>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+alex.prep.lsn.expr">alex.prep.lsn.expr()</a></code>, <code><a href="#topic+KW.hit.express">KW.hit.express()</a></code>, <code><a href="#topic+alex.waterfall.prep">alex.waterfall.prep()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expr.data)
data(lesion.data)
data(hg19.gene.annotation)

# prepare expression, lesion data and return the set of genes with both types of data available
# ordered by gene IDs in rows and patient IDs in columns:
alex.data=alex.prep.lsn.expr(expr.data, lesion.data,
                             hg19.gene.annotation, min.expr=1, min.pts.lsn=5)

# run KW test for association between lesion groups and expression level of the same gene:
alex.kw.results=KW.hit.express(alex.data, hg19.gene.annotation, min.grp.size=5)

# To prepare lesion and expression data for a waterfall plot (WT1 gene):
WT1.waterfall.prep=alex.waterfall.prep(alex.data, alex.kw.results, "WT1", lesion.data)

# waterfall plot of WT1 gene:
WT1.waterfall.plot=alex.waterfall.plot(WT1.waterfall.prep, lesion.data)
</code></pre>

<hr>
<h2 id='alex.waterfall.prep'>Prepare Lesion and Expression Data for Waterfall Plots</h2><span id='topic+alex.waterfall.prep'></span>

<h3>Description</h3>

<p>Function prepares lesion and expression data of a selected gene for the alex.waterfall.plot function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alex.waterfall.prep(alex.data, alex.kw.results, gene, lsn.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="alex.waterfall.prep_+3A_alex.data">alex.data</code></td>
<td>
<p>output of the alex.prep.lsn.expr function. It's a list of three data tables that include &quot;row.mtch&quot;, &quot;alex.expr&quot; with expression data, &quot;alex.lsn&quot; with lesion data. Rows of alex.expr, and &quot;alex.lsn&quot; matrices are ordered by the gene ensembl IDs and columns are ordered by patient IDs.</p>
</td></tr>
<tr><td><code id="alex.waterfall.prep_+3A_alex.kw.results">alex.kw.results</code></td>
<td>
<p>ALEX Kruskal-Wallis test results (output of the KW.hit.express function).</p>
</td></tr>
<tr><td><code id="alex.waterfall.prep_+3A_gene">gene</code></td>
<td>
<p>Gene name or ensembl ID of the gene of interest.</p>
</td></tr>
<tr><td><code id="alex.waterfall.prep_+3A_lsn.data">lsn.data</code></td>
<td>
<p>Lesion data in a GRIN compatible format. Object should has five columns that include &quot;ID&quot; with patient ID, &quot;chrom&quot; which is the chromosome on which the lesion is located, &quot;loc.start&quot; which is the lesion start position, &quot;loc.end&quot; the lesion end position and &quot;lsn.type&quot; which is the lesion category for example gain, loss, mutation, fusion, etc...</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function prepares lesion and expression data of a selected gene for the alex.waterfall.plot function. It return a data table with patient ID, lesion types that affect each patient if any and expression level of the gene of interest. It also extract the kruskal-wallis test result and all lesions that affect the gene of interest.
</p>


<h3>Value</h3>

<p>A list of four components:
</p>
<table role = "presentation">
<tr><td><code>gene.lsn.exp</code></td>
<td>
<p>Data table with three columns(&quot;ID&quot; with patient ID, &quot;gene.name_lsn&quot; has the type of lesion affecting the patient which can be none, gain, mutation, multiple, etc.. and &quot;gene.name_expr&quot; which has the expression level of the gene of interst in this particular patient.</p>
</td></tr>
<tr><td><code>lsns</code></td>
<td>
<p>Data table with all lesions affecting the gene of interst in a GRIN compatible format extracted from the lesion data file.</p>
</td></tr>
<tr><td><code>stats</code></td>
<td>
<p>One row with the Kruskal-Wallis test result for the gene of interst (output of the KW.hit.express function).</p>
</td></tr>
<tr><td><code>gene.ID</code></td>
<td>
<p>Gene name of the gene of interst.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+alex.prep.lsn.expr">alex.prep.lsn.expr()</a></code>, <code><a href="#topic+KW.hit.express">KW.hit.express()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expr.data)
data(lesion.data)
data(hg19.gene.annotation)

# prepare expression, lesion data and return the set of genes with both types of data available
# ordered by gene IDs in rows and patient IDs in columns:
alex.data=alex.prep.lsn.expr(expr.data, lesion.data,
                             hg19.gene.annotation, min.expr=1, min.pts.lsn=5)

# run KW test for association between lesion groups and expression level of the same gene:
alex.kw.results=KW.hit.express(alex.data, hg19.gene.annotation, min.grp.size=5)

# To prepare lesion and expression data for a waterfall plot (WT1 gene):
WT1.waterfall.prep=alex.waterfall.prep(alex.data, alex.kw.results, "WT1", lesion.data)
</code></pre>

<hr>
<h2 id='clin.data'>Example T-ALL Dataset Clinical Data</h2><span id='topic+clin.data'></span>

<h3>Description</h3>

<p>Clinical data file showing demographic and clinical outcomes of 265 newly diagnosed T-cell Acute Lymphoblastic Leukemia (T-ALL) patients that was reported by Liu, Yu, et al. (2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clin.data
</code></pre>


<h3>Format</h3>



<h4><code>clin.data</code></h4>

<p>A data frame with 265 rows and 11 columns:
</p>

<dl>
<dt>ID</dt><dd><p>Patient identifier</p>
</dd>
<dt>Sex</dt><dd><p>Patient gender</p>
</dd>
<dt>Race</dt><dd><p>Patient race</p>
</dd>
<dt>Age_Days</dt><dd><p>Patient age in days</p>
</dd>
<dt>WBC</dt><dd><p>White Blood Cell (WBC) count</p>
</dd>
<dt>MRD29</dt><dd><p>Minimal Residual Disease (MRD) percentage</p>
</dd>
<dt>MRD.binary</dt><dd><p>MRD as a categorical variable (0 if MRD&lt;=0.1 or 1 if MRD&gt;0.1)</p>
</dd>
<dt>os.time</dt><dd><p>Overall survival time in years (time between diagnosis and either the last follow-up or death)</p>
</dd>
<dt>os.censor</dt><dd><p>Survival status (0=alive at the last follow-up or, 1=dead)</p>
</dd>
<dt>efs.time</dt><dd><p>Event-free survival time in years</p>
</dd>
<dt>efs.censor</dt><dd><p>Event indicator (0=censored without event or, 1=event)</p>
</dd>
</dl>




<h3>Source</h3>

<p>Data was extracted from the supplementary material tables of the published Liu, Yu, et al. (2017) manuscript <a href="https://www.nature.com/articles/ng.3909#Sec27">https://www.nature.com/articles/ng.3909#Sec27</a> and the publicly available clinical data on TARGET database. The two files were merged and selected list of variables were kept in the final clinical data file.
</p>

<hr>
<h2 id='compute.gw.coordinates'>Compute Genome-wide Coordinates</h2><span id='topic+compute.gw.coordinates'></span>

<h3>Description</h3>

<p>The function assign plotting coordinates necessary for the genome-wide lesion plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute.gw.coordinates(grin.res, scl = 1e+06)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compute.gw.coordinates_+3A_grin.res">grin.res</code></td>
<td>
<p>GRIN results (output of the grin.stats function).</p>
</td></tr>
<tr><td><code id="compute.gw.coordinates_+3A_scl">scl</code></td>
<td>
<p>length of chromosome units in base pairs. Default is 1,000,000 which means that each chromosome will be divided into multiple pieces each is 1 million base pair in length.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function divides each chromosome into multiple units based on the specified scl value. In addition, it orders and adds two columns x.start and x.end to the chromosme size file (x.start for chr2 is equal to x.end of chr1). Function also adds x.start and x.end columns to lesion and gene annotation data files (x.start is the start position of the lesion or the gene divided by scl and x.end is the end position of the lesion or the gene divided by scl taking into consideration that the start position of the chromosomes is added consecutively based on the chromosomes length).
</p>


<h3>Value</h3>

<p>Function return a list of GRIN results with the following changes to allow adding genome-wide plotting coordinates:
</p>
<table role = "presentation">
<tr><td><code>gene.hits</code></td>
<td>
<p>No changes, a data table of GRIN results that includes gene annotation, number of subjects and number of hits affecting each locus, p and FDR adjusted q-values showing the probability of each locus to be affected by one or a constellation of multiple types of lesions.</p>
</td></tr>
<tr><td><code>gene.lsn.data</code></td>
<td>
<p>No changes, each row represent a gene overlapped by a certain lesion. Column &quot;gene&quot; shows the overlapped gene ensembl ID, and ID column has the patient ID</p>
</td></tr>
<tr><td><code>lsn.data</code></td>
<td>
<p>input lesion data with two additional columns (x.start and x.end). x.start is the start position of the lesion divided by scl and x.end is the end position of the lesion divided by scl taking into consideration that the start position of the chromosomes is added consecutively based on the chromosomes length.</p>
</td></tr>
<tr><td><code>gene.data</code></td>
<td>
<p>input gene annotation data with two additional columns (x.start and x.end). x.start is the start position of the gene divided by scl and x.end is the end position of the gene divided by scl taking into consideration that the start position of the chromosomes is added consecutively based on the chromosomes length.</p>
</td></tr>
<tr><td><code>chr.size</code></td>
<td>
<p>data table showing the size of the 22 autosomes, in addition to X and Y chromosomes in base pairs with two additional columns (x.start and x.end). x.start is the start position of the chromosome divided by scl and x.end is the end position of the chromosome divided by scl taking into consideration that the start position of the chromosomes is added consecutively based on the chromosomes length.</p>
</td></tr>
<tr><td><code>gene.index</code></td>
<td>
<p>data.frame with overlapped gene-lesion data rows that belong to each chromosome in the gene.lsn.data table.</p>
</td></tr>
<tr><td><code>lsn.index</code></td>
<td>
<p>data.frame that shows the overlapped gene-lesion data rows taht belong to each lesion in the gene.lsn.data table.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+grin.stats">grin.stats()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)
data(hg19.chrom.size)

# Run GRIN model using grin.stats function
grin.results=grin.stats(lesion.data,
                        hg19.gene.annotation,
                        hg19.chrom.size)
# assign genomewide coordinates and prepare the results for the genomewide.lsn.plot function
genome.coord=compute.gw.coordinates(grin.results)
</code></pre>

<hr>
<h2 id='count.hits'>Count Gene Lesion Hits</h2><span id='topic+count.hits'></span>

<h3>Description</h3>

<p>The function computes the number of hits affecting each gene by lesion category. It also compute the number of subjects with a hit in each annotated gene by lesion category as well.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count.hits(ov.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="count.hits_+3A_ov.data">ov.data</code></td>
<td>
<p>a list of six data.frames that represent the output results of the find.gene.lsn.overlaps function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function use the output of the find.gene.lsn.overlaps function and return the number of unique subjects affected by each lesion category in the provided list of annotated genes and regulatory features (nsubj stats). It also count the number of hits affecting each loci per lesion category (nhits stats). For example, if NOTCH1 gene was found affected by three different mutations in the same subject, this patient will be considered as one subject in the nsubj stats but in the nhits stats for this event will be counted as 3 mutations that affect NOTCH1 gene.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table role = "presentation">
<tr><td><code>lsn.data</code></td>
<td>
<p>Input lesion data</p>
</td></tr>
<tr><td><code>lsn.index</code></td>
<td>
<p>data.frame that shows the overlapped gene-lesion data rows taht belong to each lesion in the gene.lsn.data table.</p>
</td></tr>
<tr><td><code>gene.data</code></td>
<td>
<p>Input gene annotation data</p>
</td></tr>
<tr><td><code>gene.index</code></td>
<td>
<p>data.frame with overlapped gene-lesion data rows that belong to each chromosome in the gene.lsn.data table.</p>
</td></tr>
<tr><td><code>nhit.mtx</code></td>
<td>
<p>A data.frame with number of hits in each gene by lesion type (number of columns will be equal to the number of lesion types in the lsn.type column).</p>
</td></tr>
<tr><td><code>nsubj.mtx</code></td>
<td>
<p>A data.frame with number of affected subjects by lesion type in each annotated gene.</p>
</td></tr>
<tr><td><code>gene.lsn.data</code></td>
<td>
<p>Each row represent a gene overlapped by a certain lesion. Column &quot;gene&quot; shows the overlapped gene and &quot;ID&quot; column has the patient ID.</p>
</td></tr>
<tr><td><code>glp.data</code></td>
<td>
<p>data.frame ordered by gene and lesions start position. Gene start position is coded as 1 in the cty column and gene end position is coded as 4. Lesion start position is coded as 2 in the cty column and lesion end position is coded as 3.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Pounds, Stan, et al. (2013) A genomic random interval model for statistical analysis of genomic lesion data.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+prep.gene.lsn.data">prep.gene.lsn.data()</a></code>, <code><a href="#topic+find.gene.lsn.overlaps">find.gene.lsn.overlaps()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)

# prepare gene and lesion data for later computations:
prep.gene.lsn=prep.gene.lsn.data(lesion.data,
                                 hg19.gene.annotation)

# determine lesions that overlap each gene (locus):
gene.lsn.overlap=find.gene.lsn.overlaps(prep.gene.lsn)

# count number of subjects affected by different types of lesions and number of hits that affect
# each locus:
count.nsubj.nhits=count.hits(gene.lsn.overlap)
</code></pre>

<hr>
<h2 id='default.grin.colors'>Default GRIN Colors</h2><span id='topic+default.grin.colors'></span>

<h3>Description</h3>

<p>Function assigns default colors for each lesion group in the whole set of GRIN plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>default.grin.colors(lsn.types)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="default.grin.colors_+3A_lsn.types">lsn.types</code></td>
<td>
<p>Unique lesion types as specified in the lesion data file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function specifies 10 colors for different lesion types. If the number of lesion types is more than 10, the user will be asked to specify the colors manually.
</p>


<h3>Value</h3>

<p>Function return a vector of colors assigned to each unique lesion type.
</p>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Pounds, Stan, et al. (2013) A genomic random interval model for statistical analysis of genomic lesion data.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)

lsn.types=unique(lesion.data$lsn.type)
# assign colors for different lesion categories using default.grin.colors function:
default.grin.colors(lsn.types)
</code></pre>

<hr>
<h2 id='expr.data'>Example T-ALL Dataset Gene Expression Data</h2><span id='topic+expr.data'></span>

<h3>Description</h3>

<p>Gene expression data file showing log2 normalized expression level of 420 genes (rows) in 265 newly diagnosed T-cell Acute Lymphoblastic Leukemia (T-ALL) patients in columns that was reported by Liu, Yu, et al. (2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expr.data
</code></pre>


<h3>Format</h3>



<h4><code>expr.data</code></h4>

<p>A data frame with 420 rows and 265 columns:
</p>

<dl>
<dt>gene</dt><dd><p>Ensembl IDs of the list of 420 genes included in the dataset</p>
</dd>
</dl>
<p>...

</p>



<h3>Source</h3>

<p>Data was extracted from the supplementary material tables of the published Liu, Yu, et al. (2017) manuscript <a href="https://www.nature.com/articles/ng.3909#Sec27">https://www.nature.com/articles/ng.3909#Sec27</a>
</p>

<hr>
<h2 id='find.gene.lsn.overlaps'>Find Gene Lesion Overlaps</h2><span id='topic+find.gene.lsn.overlaps'></span>

<h3>Description</h3>

<p>The function use the output of the prep.gene.lsn.data function to find lesion-gene overlaps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.gene.lsn.overlaps(gl.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find.gene.lsn.overlaps_+3A_gl.data">gl.data</code></td>
<td>
<p>a list of five data.frames that represent the output results of the prep.gene.lsn.data function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table role = "presentation">
<tr><td><code>lsn.data</code></td>
<td>
<p>Input lesion data</p>
</td></tr>
<tr><td><code>gene.data</code></td>
<td>
<p>Input gene annotation data</p>
</td></tr>
<tr><td><code>gene.lsn.data</code></td>
<td>
<p>data.frame ordered by gene and lesions start position. Gene start position is coded as 1 in the cty column and gene end position is coded as 4. Lesion start position is coded as 2 in the cty column and lesion end position is coded as 3.</p>
</td></tr>
<tr><td><code>gene.lsn.hits</code></td>
<td>
<p>data.frame on which each row represent a gene overlapped by a certain lesion. The data.frame has 11 columns that include &quot;gene&quot; with ensembl ID of the overlapped gene, &quot;gene.chrom&quot;, &quot;gene.loc.start&quot; and &quot;gene.loc.end&quot; with data for the chromosome on which the gene is located, start and end positions of the gene. In addition, column &quot;ID&quot; has the ID of the patient with a lesion that overlapped this gene, &quot;lsn.chrom&quot;, &quot;lsn.loc.start&quot;, &quot;lsn.loc.end&quot; and &quot;lsn.type&quot; have data for the chromosome, lesion start, lesion end positions and the lesion type respectively.</p>
</td></tr>
<tr><td><code>gene.index</code></td>
<td>
<p>data.frame that shows row start and row end for each chromosome in the gene.lsn.data table</p>
</td></tr>
<tr><td><code>lsn.index</code></td>
<td>
<p>data.frame that shows row start and row end for each lesion in the gene.lsn.data table</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Pounds, Stan, et al. (2013) A genomic random interval model for statistical analysis of genomic lesion data.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+prep.gene.lsn.data">prep.gene.lsn.data()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)

# prepare gene and lesion data for later computations:
prep.gene.lsn=prep.gene.lsn.data(lesion.data,
                                 hg19.gene.annotation)

# determine lesions that overlap each gene (locus):
gene.lsn.overlap=find.gene.lsn.overlaps(prep.gene.lsn)
</code></pre>

<hr>
<h2 id='genomewide.log10q.plot'>Genomewide log10q Plot</h2><span id='topic+genomewide.log10q.plot'></span>

<h3>Description</h3>

<p>The function return a genome-wide plot based on -log(10) q-value of each of the evaluated annotated genes or lesion boundaries on each chromosome. The plot is lesion type specific (gain, loss, mutation, etc...).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomewide.log10q.plot(
  grin.res,
  lsn.grps,
  lsn.colors = NULL,
  max.log10q = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genomewide.log10q.plot_+3A_grin.res">grin.res</code></td>
<td>
<p>GRIN results evaluating annotated genes or lesion boundaries (output of the grin.stats function using either lesion boundaries or annotated genes as a marker input file).</p>
</td></tr>
<tr><td><code id="genomewide.log10q.plot_+3A_lsn.grps">lsn.grps</code></td>
<td>
<p>Selected lesion groups to be added to the plot.</p>
</td></tr>
<tr><td><code id="genomewide.log10q.plot_+3A_lsn.colors">lsn.colors</code></td>
<td>
<p>Colors assigned to each lesion group (if NULL, lesion colors will be assigned automatically using default.grin.colors function).</p>
</td></tr>
<tr><td><code id="genomewide.log10q.plot_+3A_max.log10q">max.log10q</code></td>
<td>
<p>Maximum log10 q value to be added to the plot. All boundaries or genes with -log10 q smaller than this value will be set automatically to max.log10q.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function return a genome-wide plot based on -log(10) q-value of each of the evaluated annotated genes or lesion boundaries to be affected by a certain type of lesions (gain, loss, mutation, etc...).
</p>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+grin.lsn.boundaries">grin.lsn.boundaries()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)
data(hg19.chrom.size)

# This analysis is lesion type specific. So, user should first data extract data for a specific
# lesion group of interest for example gains from the lesion data file:
gain=lesion.data[lesion.data$lsn.type=="gain",]
# Return lesion boundaries for gains:
lsn.bound.gain=grin.lsn.boundaries(gain, hg19.chrom.size)
# Run GRIN analysis Using Lesion Boundaries as Markers Instead of the Gene Annotation File:
GRIN.results.gain.bound=grin.stats(gain, lsn.bound.gain, hg19.chrom.size)
# Return genomewide -log10q plot for association between lesion boundaries and gain:
genomewide.log10q.plot(GRIN.results.gain.bound, lsn.grps=c("gain"),
                       lsn.colors=c("gain" = "red"), max.log10q = 10)

# instead of lesion boundaries, users can also plot -log10q values for annotated genes using
# genes annotation data as a marker data file:
grin.results=grin.stats(lesion.data,
                        hg19.gene.annotation,
                        hg19.chrom.size)

genomewide.log10q.plot(grin.results, lsn.grps=c("gain"), lsn.colors=c("gain" = "red"),
                       max.log10q = 10)

# User can run this same analysis for other lesion types such as mutations and deletions.
</code></pre>

<hr>
<h2 id='genomewide.lsn.plot'>Genome-wide Lesion Plot</h2><span id='topic+genomewide.lsn.plot'></span>

<h3>Description</h3>

<p>Function return a genomewide lesion plot for all lesion types affecting different chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomewide.lsn.plot(
  grin.res,
  ordered = FALSE,
  pt.order = NULL,
  lsn.colors = NULL,
  max.log10q = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genomewide.lsn.plot_+3A_grin.res">grin.res</code></td>
<td>
<p>GRIN results (output of the grin.stats function).</p>
</td></tr>
<tr><td><code id="genomewide.lsn.plot_+3A_ordered">ordered</code></td>
<td>
<p>By default the function will order the patient IDs alphabetically. However, users can specify a certain patient's order in the genomewide lesion plot by specifying ordered=TRUE and pass a data frame with new patient's order to the pt.order argument.</p>
</td></tr>
<tr><td><code id="genomewide.lsn.plot_+3A_pt.order">pt.order</code></td>
<td>
<p>data.frame of two columns &quot;ID&quot; that has patient IDs matching the unique IDs in the lesion data file and &quot;pts.order&quot; that has the new patient's order listed as numbers that range from 1:n.patients (Should be only specified if ordered=TRUE).</p>
</td></tr>
<tr><td><code id="genomewide.lsn.plot_+3A_lsn.colors">lsn.colors</code></td>
<td>
<p>a vector of lesion colors (If not provided by the user, colors will be automatically assigned using default.grin.colors function).</p>
</td></tr>
<tr><td><code id="genomewide.lsn.plot_+3A_max.log10q">max.log10q</code></td>
<td>
<p>Maximum log10 q value for genes in the GRIN results table to be added to the plot. If max.log10q=100 for example, all -log10q values&gt;100, will be adjusted to 100 in the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function use the genome-wide plotting coordinates obtained from the compute.gw.coordinates function and plot the whole set of lesions affecting subjects included in the dataset in the middle panel of the figure. Two additional side panels show the number of affected subjects and -log10 q value of each locus to be affected by all different types of lesions.
</p>


<h3>Value</h3>

<p>The function return a genome-wide lesion plot (all chromosomes) in the middle panel. For each locus, Panel on the left shows -log10 q value and the Panel on the right show the number of subjects affected by all different types of lesions color coded by lesion category.
</p>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+compute.gw.coordinates">compute.gw.coordinates()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)
data(hg19.chrom.size)

# Run GRIN model using grin.stats function
grin.results=grin.stats(lesion.data,
                        hg19.gene.annotation,
                        hg19.chrom.size)

# prepare the genomewide lesion plot using genomewide.lsn.plot function with patient IDs ordered
# alphabetically:
genomewide.plot=genomewide.lsn.plot(grin.results, max.log10q=50)

# To pass certain patients order to the genomewide.lsn.plot function, the user should specify
# a certain patients order using the pt.order argument.
</code></pre>

<hr>
<h2 id='get.chrom.length'>Get Chromosome Length</h2><span id='topic+get.chrom.length'></span>

<h3>Description</h3>

<p>Retrieve chromosome size data from chr.info txt files available on the UCSC genome browser based on the user specified genome assembly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.chrom.length(genome.assembly)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.chrom.length_+3A_genome.assembly">genome.assembly</code></td>
<td>
<p>User can specify one of four supported genome assemblies that include &quot;Human_GRCh38&quot;, &quot;Human_GRCh37&quot;, &quot;Mouse_HGCm39&quot; and &quot;Mouse_HGCm38&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Based on the genome assembly specified by the user, the function will directly retrieve chromosome size data from chr.info txt file available on the UCSC genome browser.
</p>


<h3>Value</h3>

<p>A data table with the following two columns:
</p>
<table role = "presentation">
<tr><td><code>chrom</code></td>
<td>
<p>column has the chromosome number denoted as 1, 2, X, Y, etc..</p>
</td></tr>
<tr><td><code>size</code></td>
<td>
<p>column has the chromosome size in base pairs.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="circlize.html#topic+read.chromInfo">circlize::read.chromInfo()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># To retreive chromosome size data for hg19 genome assembly:
hg19.chrom.size=get.chrom.length("Human_GRCh37")
# "Human_GRCh38" can be used to retreive chromosome size data for hg38 genome assembly.
</code></pre>

<hr>
<h2 id='get.ensembl.annotation'>Get Ensembl Gene and Regulatory Features Annotation Data</h2><span id='topic+get.ensembl.annotation'></span>

<h3>Description</h3>

<p>Function directly retrieve gene and regulatory features annotation data from Ensembl BioMart database based on the specified genome assembly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.ensembl.annotation(genome.assembly)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.ensembl.annotation_+3A_genome.assembly">genome.assembly</code></td>
<td>
<p>User can specify one of four genome assemblies that include &quot;Human_GRCh38&quot;, &quot;Human_GRCh37&quot;, &quot;Mouse_HGCm39&quot; and &quot;Mouse_HGCm38&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Based on the genome assembly specified by the user, the function will directly retrieve gene and regulatory features annotation data from ensembl BioMart database. Annotation data include enesembl ID, the chromosome on which the gene is located, gene start and gene end position, gene name, gene description, biotype, chromosome strand and chromosome band. Gene classes (biotypes) include protein coding genes, long noncoding RNAs (lncRNAs), microRNAs (miRNAs), small nuclear RNAs (snRNA), small nucleolar RNAs (snoRNA), immunoglobulins (IGs), T-cell receptors (TCRs) and pseudogens. Regulatory features data retrieved from Ensembl regulatory build are categorized in 6 classes that include promoters, promoter flanking regions, predicted enhancers, CTCF binding sites, transcription factor (TF) binding sites and the open chromatin regions.Ensembl first imports publicly available data from different large epigenomic consortia such as ENCODE, Roadmap Epigenomics and Blueprint. All high-throughput sequencing data sets are then uniformly processed using the Ensembl Regulation Sequence Analysis (ERSA) pipeline to generate signal tracks for enriched regions also referred to as annotated features or peaks. Segmentation data provide information about promoter, promoter flanking regions, enhancers and CTCF binding sites. If TF binding probability is &gt;0 in areas outside previously mentioned regions, it takes a label of TF binding site. If any open chromatin region did not overlap with the above features, it takes a label of unannotated open chromatin. users will also have the chance to use a list of experimentally verified enhancers/transcription start sites (TSS) using the CAGE (Cap Analysis of Gene Expression) experiment on a multitude of different primary cells and tissues from the Functional Annotation of the Mouse/Mammalian Genome (FANTOM5) project.
</p>


<h3>Value</h3>

<p>A list of three components:
</p>
<table role = "presentation">
<tr><td><code>gene.annotation</code></td>
<td>
<p>A 9 columns data.frame with gene annotation data that include enesembl ID, chromosome, gene start and gene end position, gene name, gene description, biotype, chromosome strand, and chromosome band.</p>
</td></tr>
<tr><td><code>reg.annotation.predicted</code></td>
<td>
<p>A 5 columns data.frame with regulatory features annotation data directly retreived from Ensembl regulatory build that include enesembl ID, chromosome, description(promoter, enhancer, etc..), feature start and end positions.</p>
</td></tr>
<tr><td><code>reg.annotation.validated</code></td>
<td>
<p>A 5 columns data.frame with regulatory features annotation data for experimentally verified features retreived from FANTOM5 project that include feature ID, chromosome, description(enhancer, transcription start site (TSS)), feature start and end positions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>
<p>Zerbino, Daniel R., et al. (2015). The ensembl regulatory build.
</p>
<p>Kinsella, Rhoda J., et al. (2011). Ensembl BioMarts: a hub for data retrieval across taxonomic space.
</p>


<h3>See Also</h3>

<p><code><a href="biomaRt.html#topic+useEnsembl">biomaRt::useEnsembl()</a></code>, <code><a href="biomaRt.html#topic+getBM">biomaRt::getBM()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Retrieve annotation data for human hg19 genome assembly:
hg19.ann=get.ensembl.annotation("Human_GRCh37")
# gene annotation data:
hg19.gene.annotation=hg19.ann$gene.annotation
# regulatory features annotation data retrieved from Ensembl regulatory build:
hg19.reg.annotation=hg19.ann$reg.annotation.predicted
# regulatory features annotation data retrieved from FANTOM5 project:
hg19.fantom.annotation=hg19.ann$reg.annotation.validated

# "Human_GRCh38" can be used instead of "Human_GRCh37" to retrieve gene and regulatory features
# annotation data for human hg38 genome assembly.

</code></pre>

<hr>
<h2 id='grin.assoc.lsn.outcome'>Associate Lesions with Clinical Outcomes</h2><span id='topic+grin.assoc.lsn.outcome'></span>

<h3>Description</h3>

<p>The function run association analysis between the binary lesion matrix (output of prep.binary.lsn.mtx function) and clinical outcomes of interest such as Minimal Residual Disease (MRD), Event-free Survival (EFS) and Overall Survival (OS), etc...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grin.assoc.lsn.outcome(
  lsn.mtx,
  clin.data,
  annotation.data,
  clinvars,
  covariate = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="grin.assoc.lsn.outcome_+3A_lsn.mtx">lsn.mtx</code></td>
<td>
<p>Binary lesion matrix in which each type of lesions affecting certain gene is represented in a separate row for example ENSG00000148400_gain. If the gene is affected by this specific type of lesion, patient entry will be coded as 1 or 0 otherwisw. This matrix is the output of the prep.binary.lsn.mtx function.</p>
</td></tr>
<tr><td><code id="grin.assoc.lsn.outcome_+3A_clin.data">clin.data</code></td>
<td>
<p>Clinical data table in which the first column &quot;ID&quot; should has the patient ID.</p>
</td></tr>
<tr><td><code id="grin.assoc.lsn.outcome_+3A_annotation.data">annotation.data</code></td>
<td>
<p>Gene annotation data either provided by the user or retrieved from ensembl BioMart database using get.ensembl.annotation function included in the GRIN2.0 library. Data.frame should has four columns: &quot;gene&quot; which is the ensembl ID of annotated genes, &quot;chrom&quot; which is the chromosome on which the gene is located, &quot;loc.start&quot; which is the gene start position, and &quot;loc.end&quot; the gene end position.</p>
</td></tr>
<tr><td><code id="grin.assoc.lsn.outcome_+3A_clinvars">clinvars</code></td>
<td>
<p>Clinical outcome variables of interest (survival variables such as EFS and OS should be first coded as survival objects using Surv function and added as new columns to the clinical data file, binary variables such as MRD should be coded as 0, 1).</p>
</td></tr>
<tr><td><code id="grin.assoc.lsn.outcome_+3A_covariate">covariate</code></td>
<td>
<p>Covariates that the model will adjust for if any.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function run association analysis between the binary lesion matrix in which each type of lesions affecting certain gene is represented in a separate row (output of prep.binary.lsn.mtx function) and clinical outcomes.Function will run logistic regression models for association between each gene-lesion pair with numeric variables such as MRD that should be coded as 0 if the patient is MRD-negative and 1 if the patient is MRD positive. Function will also run COX-Proportional hazard models for association between lesions and survival objects such as Event-free survival (EFS) and oveall survival (OS). EFS and OS should be first coded as survival objects using Surv function and added as new columns to the clinical data file. If specified, the models can be also adjusted for one or a group of covariates such as risk group assignment, gender, age, etc...
</p>


<h3>Value</h3>

<p>Function returns a results table that has gene annotation data, and multiple columns showing results of the logistic regression model for association with binary variables such as MRD that include odds.ratio, lower and upper 95 confidence interval (CI), model p and FDR adjusted q values, in addition to the number of patients with/without lesion who experienced or did not experience the event. Results table will also include results of COXPH models for association between lesions with survival variables such as EFS, OS that include COXPH hazard ratio, lower and upper 95 CI, model p and FDR adjusted q values, in addition to the number of patients with/without the lesion who experienced or did not experience the event.
</p>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Andersen, P. and Gill, R. (1982). Cox's regression model for counting processes, a large sample study.
</p>
<p>Therneau, T., Grambsch, P. (2000) Modeling Survival Data: Extending the Cox Model.
</p>
<p>Dobson, A. J. (1990) An Introduction to Generalized Linear Models.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+glm">stats::glm()</a></code>, <code><a href="survival.html#topic+coxph">survival::coxph()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)
data(clin.data)

# prepare lesion data and find gene lesion overlaps:
gene.lsn=prep.gene.lsn.data(lesion.data, hg19.gene.annotation)
gene.lsn.overlap= find.gene.lsn.overlaps(gene.lsn)

# Prepare a binary lesion matrix for genes affected by a certain type of lesion in at least
# 5 subjects using prep.binary.lsn.mtx function:
lsn.binary.mtx=prep.binary.lsn.mtx(gene.lsn.overlap, min.ngrp=5)

# Prepare EFS and OS survival objects and add two new columns to the clinical data file:
library(survival)
clin.data$EFS &lt;- Surv(clin.data$efs.time, clin.data$efs.censor)
clin.data$OS &lt;- Surv(clin.data$os.time, clin.data$os.censor)
# define clinical outcome variables to be included in the analysis:
clinvars=c("MRD.binary", "EFS", "OS")

# Run association analysis between lesions in the binary lesion matrix and clinical variables
# in the clinvars object:
assc.outcomes=grin.assoc.lsn.outcome(lsn.binary.mtx,
                                     clin.data,
                                     hg19.gene.annotation,
                                     clinvars)

# to adjust the models for one or a group of covariates, user can specify one or a group
# of covariates using the 'covariate' argument.
</code></pre>

<hr>
<h2 id='grin.barplt'>GRIN Bar Plot</h2><span id='topic+grin.barplt'></span>

<h3>Description</h3>

<p>Function return a stacked bar plot with number of patients affected by all different types of lesions in a pre-specified list of genes of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grin.barplt(grin.res, count.genes, lsn.colors = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="grin.barplt_+3A_grin.res">grin.res</code></td>
<td>
<p>GRIN results (output of the grin.stats function).</p>
</td></tr>
<tr><td><code id="grin.barplt_+3A_count.genes">count.genes</code></td>
<td>
<p>vector with gene names of a list of genes to be added to the bar plot.</p>
</td></tr>
<tr><td><code id="grin.barplt_+3A_lsn.colors">lsn.colors</code></td>
<td>
<p>Lesion colors (If not provided by the user, colors will be automatically assigned using default.grin.colors function).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function will use the input list of gene names and extract the number of patients affected by all different types of lesions in those genes from the GRIN results table (output of the grin.stats function).
</p>


<h3>Value</h3>

<p>Function return a stacked bar plot with number of patients affected by all different types of lesions in the pre-specified list of genes of interest.
</p>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+grin.stats">grin.stats()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)
data(hg19.chrom.size)

# run GRIN analysis using grin.stats function
grin.results=grin.stats(lesion.data,
                        hg19.gene.annotation,
                        hg19.chrom.size)

# specify a list of genes to be included in the bar plot (driver genes)
count.genes=as.vector(c("TAL1", "FBXW7", "PTEN", "IRF8","NRAS",
                        "BCL11B", "MYB", "LEF1","RB1", "MLLT3", "EZH2", "ETV6", "CTCF",
                        "JAK1", "KRAS", "RUNX1", "IKZF1", "KMT2A", "RPL11", "TCF7",
                        "WT1", "JAK2", "JAK3", "FLT3"))
# return the stacked barplot
grin.barplt(grin.results, count.genes)
</code></pre>

<hr>
<h2 id='grin.lsn.boundaries'>GRIN Evaluate Lesion Boundaries</h2><span id='topic+grin.lsn.boundaries'></span>

<h3>Description</h3>

<p>The function evaluates Copy number variations that include gain and deletions as boundaries based on unique lesion start and end positions. This analysis is lesion type specific and covers the entire genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grin.lsn.boundaries(lsn.data, chrom.size)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="grin.lsn.boundaries_+3A_lsn.data">lsn.data</code></td>
<td>
<p>Lesion data file that should be limited to include either gain or deletions. If gains are splitted to gain and amplifications based on the log2Ratio value of the CNV segmentation file, the two categories can be included in the same data table, same for homozygous and heterozygous deletions.</p>
</td></tr>
<tr><td><code id="grin.lsn.boundaries_+3A_chrom.size">chrom.size</code></td>
<td>
<p>Chromosize size table that should include two columns &quot;chrom&quot; with the chromosome number and &quot;size&quot; with the chromosome size in base pairs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function evaluates Copy number variations that include gain and deletions as boundaries and return a table of ordered boundaries based on the unique start and end positions of different lesions on each chromosome. If gains are splitted to gain and amplifications based on the log2Ratio value of the CNV segmentation file, the two categories can be included in the same analysis, same for homozygous and heterozygous deletions. Boundary will be the region between each unique start and end positions where large size lesions will be splitted into multiple boundaries based on other smaller size lesions that affect the same region in other patients if any. This analysis is meant to cover the entire genome, so regions without any annotated genes or regulatory features will be incuded will be assesed in the analysis. The first boundary for each chromosome will start from the first nucleotide base on the chromosome till the start position of the first lesion that affect the chromosome. Similarly, the last boundary will start from the end position of the last lesion that affect the chromosome till the last base on the chromosome.
</p>


<h3>Value</h3>

<p>Function return a data.frame with five columns:
</p>
<table role = "presentation">
<tr><td><code>gene</code></td>
<td>
<p>Ordered boundaries by unique start and end positions of different lesions on each chromosome.</p>
</td></tr>
<tr><td><code>chrom</code></td>
<td>
<p>Chromosome on which the bounday is located.</p>
</td></tr>
<tr><td><code>loc.start</code></td>
<td>
<p>Boundary start position.</p>
</td></tr>
<tr><td><code>loc.end</code></td>
<td>
<p>Boundary end position.</p>
</td></tr>
<tr><td><code>diff</code></td>
<td>
<p>Boundary size in base pairs.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.chrom.size)

# This analysis is lesion type specific. So, user should first data extract data for a specific
# lesion group of interest for example gains from the lesion data file:
gain=lesion.data[lesion.data$lsn.type=="gain",]
# Return lesion boundaries for gains:
lsn.bound.gain=grin.lsn.boundaries(gain, hg19.chrom.size)
# Run GRIN analysis Using Lesion Boundaries markers Instead of the gene annotation file:
GRIN.results.gain.bound=grin.stats(gain, lsn.bound.gain, hg19.chrom.size)

# same analysis can be done for mutations, deletions and structural rearrangments.
</code></pre>

<hr>
<h2 id='grin.oncoprint.mtx'>GRIN OncoPrint Matrix</h2><span id='topic+grin.oncoprint.mtx'></span>

<h3>Description</h3>

<p>Function use GRIN results table and prepare the lesion matrix that the user can pass to the oncoprint function from ComplexHeatmap package to geneate an OncoPrint for a selcted list of genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grin.oncoprint.mtx(grin.res, oncoprint.genes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="grin.oncoprint.mtx_+3A_grin.res">grin.res</code></td>
<td>
<p>GRIN results (output of the grin.stats function).</p>
</td></tr>
<tr><td><code id="grin.oncoprint.mtx_+3A_oncoprint.genes">oncoprint.genes</code></td>
<td>
<p>Vector of ensembl IDs for the selected list of genes to be added to the OncoPrint.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function will use the input list of ensembl IDs to prepare a data table of lesions that affect these genes (each row is a gene and each column is a patient ID). This lesion matrix is compatible and can be passed to oncoprint function in ComplexHeatmap library to prepare an OncoPrint for lesions in the selected list of genes.
</p>


<h3>Value</h3>

<p>Function uses the output results of grin.stats function and return data table of lesions that affect a group of selected genes (each row is a gene and each column is a patient ID).
</p>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+grin.stats">grin.stats()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)
data(hg19.chrom.size)

# Run GRIN analysis using grin.stats function:
grin.results=grin.stats(lesion.data,
                        hg19.gene.annotation,
                        hg19.chrom.size)

# specify a list of genes to be included in the oncoprint (driver genes):
oncoprint.genes=as.vector(c("ENSG00000148400", "ENSG00000171862", "ENSG00000171843",
                            "ENSG00000156531", "ENSG00000162367", "ENSG00000096968",
                            "ENSG00000105639", "ENSG00000118513","ENSG00000102974",
                            "ENSG00000133703"))

# prepare the oncoprint lesion matrix:
oncoprint.mtx=grin.oncoprint.mtx(grin.results,
                                 oncoprint.genes)

# user can also specify a list of top significant genes in the GRIN constellation test:
# for example: select genes affected by two types of lesion with q2.nsubj&lt;0.01:
genes.const = grin.results$gene.hits[grin.results$gene.hits$q2.nsubj &lt; 0.01, ]
# get ensembl.ids for this list of genes
selected.genes=as.vector(genes.const$gene)
oncoprint.mtx.const=grin.oncoprint.mtx(grin.results,
                                       selected.genes)
</code></pre>

<hr>
<h2 id='grin.stats'>GRIN Statistics Output</h2><span id='topic+grin.stats'></span>

<h3>Description</h3>

<p>The function run the Genomic Random Interval (GRIN) analysis to determine whether a certain locus has an abundance of lesions or a constellation of multiple types of lesions that is statistically significant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grin.stats(lsn.data, gene.data = NULL, chr.size = NULL, genome.version = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="grin.stats_+3A_lsn.data">lsn.data</code></td>
<td>
<p>data.frame with lesion data prepared by the user in a GRIN compatible format. Object should has five columns that include &quot;ID&quot; with patient ID, &quot;chrom&quot; which is the chromosome on which the lesion is located, &quot;loc.start&quot; which is the lesion start position, &quot;loc.end&quot; the lesion end position and &quot;lsn.type&quot; which is the lesion category for example gain, loss, mutation, fusion, etc... For Single Nucleotide Variants (SNVs), loc.start will be the same as loc.end. For Copy Number Alterations (CNAs) such as gain and deletions, loc.start and loc.end should be the gain or deletion start and end positions respectively. For structural rearrangements such as inversions and translocations, each rearrangement should be coded in two different lines, one line for chromosome A involved in the translocation break-point and the second line for chromosome B break-point. For inversions on the same chromosome, the two lines will include the two breakpoints of the inversion. An example lesion data in a GRIN compatible format can be found at the GRIN2.0 package data folder (lesion.data.rda).</p>
</td></tr>
<tr><td><code id="grin.stats_+3A_gene.data">gene.data</code></td>
<td>
<p>data.frame with the gene annotation data either provided by the user or directly retreived from ensembl BioMart database using get.ensembl.annotation function included in the GRIN2.0 library if the genome.version is specified. Object should has four columns &quot;gene&quot; which is the ensembl ID of annotated genes to which the lesion data will be overlapped, &quot;chrom&quot; which is the chromosome on which the gene is located, &quot;loc.start&quot; which is the gene start position, and &quot;loc.end&quot; the gene end position.</p>
</td></tr>
<tr><td><code id="grin.stats_+3A_chr.size">chr.size</code></td>
<td>
<p>data.frame with the size of the 22 autosomes, in addition to X and Y chromosomes in base pairs. It should has two columns that include &quot;chrom&quot; with the chromosome number and &quot;size&quot; for the size of the chromosome in base pairs. Chromosome size data can be either provided by the user or directly retreived from UCSC genome browser using get.chrom.length function included in the GRIN2.0 library if genome.version is specified.</p>
</td></tr>
<tr><td><code id="grin.stats_+3A_genome.version">genome.version</code></td>
<td>
<p>Genome assembly should be only specified if the user selected not to provide gene annotation, chromosome size files, and directly retrieve those files from ensembl BioMart database, and UCSC genome browsers using get.ensembl.annotation and get.chrom.length functions respectively. Currently, the function support four genome assemblies that include &quot;Human_GRCh38&quot;, &quot;Human_GRCh37&quot;, &quot;Mouse_HGCm39&quot;, and &quot;Mouse_HGCm38&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function run the Genomic Random Interval (GRIN) analysis and evaluates the probability of each gene locus to be affected by different types of lesions based on a convolution of independent but non-identical Bernoulli distributions to determine whether this locus has an abundance of lesions that is statistically significant.In addition, FDR-adjusted q value is computed for each locus based on Pounds &amp; Cheng (2006) estimator of the proportion of tests with a true null (pi.hat). The function also evaluates if a certain locus is affected by a constellation of multiple types of lesions and return the GRIN results table.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table role = "presentation">
<tr><td><code>gene.hits</code></td>
<td>
<p>data table of GRIN results that include gene annotation, number of subjects affected by each lesion type for example gain, loss, mutation, etc.., and number of hits affecting each locus. The GRIN results table will also include P and FDR adjusted q-values showing the probability of each locus of being affected by one or a constellation of multiple types of lesions.</p>
</td></tr>
<tr><td><code>lsn.data</code></td>
<td>
<p>input lesion data</p>
</td></tr>
<tr><td><code>gene.data</code></td>
<td>
<p>input gene annotation data</p>
</td></tr>
<tr><td><code>gene.lsn.data</code></td>
<td>
<p>each row represent a gene overlapped by a certain lesion. Column &quot;gene&quot; shows the overlapped gene ensembl ID and &quot;ID&quot;&quot; column has the patient ID.</p>
</td></tr>
<tr><td><code>chr.size</code></td>
<td>
<p>data table showing the size of the 22 autosomes, in addition to X and Y chromosomes in base pairs.</p>
</td></tr>
<tr><td><code>gene.index</code></td>
<td>
<p>data.frame with overlapped gene-lesion data rows that belong to each chromosome in the gene.lsn.data table.</p>
</td></tr>
<tr><td><code>lsn.index</code></td>
<td>
<p>data.frame that shows the overlapped gene-lesion data rows taht belong to each lesion in the gene.lsn.data table.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Pounds, Stan, et al. (2013) A genomic random interval model for statistical analysis of genomic lesion data.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+prep.gene.lsn.data">prep.gene.lsn.data()</a></code>, <code><a href="#topic+find.gene.lsn.overlaps">find.gene.lsn.overlaps()</a></code>, <code><a href="#topic+count.hits">count.hits()</a></code>, <code><a href="#topic+prob.hits">prob.hits()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)
data(hg19.chrom.size)

# if gene annotation and chromosome size files will be provided by the user:
grin.results=grin.stats(lesion.data,
                        hg19.gene.annotation,
                        hg19.chrom.size)

# to directly retrieve gene annotation and chromosome size files from Ensembl BioMart database,
# and UCSC genome browsers using get.ensembl.annotation and get.chrom.length functions respectively,
# users can select to specify certain genome assembly using the 'genome.version' argument:
# "Human_GRCh37" can be used for the GRCH37 (hg19) genome assembly, and "Human_GRCh38" can be used
# for the GRCH38 (hg38) genome assembly
</code></pre>

<hr>
<h2 id='grin.stats.lsn.plot'>GRIN Statistics Lesions Plot</h2><span id='topic+grin.stats.lsn.plot'></span>

<h3>Description</h3>

<p>Function return a plot with all types of lesions that spans either a gene or regulatory feature of interest with GRIN statistics added.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grin.stats.lsn.plot(grin.res, feature = NULL, lsn.clrs = NULL, expand = 5e-04)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="grin.stats.lsn.plot_+3A_grin.res">grin.res</code></td>
<td>
<p>GRIN results for genes or regulatory features (output of the grin.stats function)</p>
</td></tr>
<tr><td><code id="grin.stats.lsn.plot_+3A_feature">feature</code></td>
<td>
<p>Feature ensembl ID from Ensembl regulatory build or FANTOM5 project. An ensembl ID of a gene can be provided as well.</p>
</td></tr>
<tr><td><code id="grin.stats.lsn.plot_+3A_lsn.clrs">lsn.clrs</code></td>
<td>
<p>Assigned colors per lesion types. If not specified, colors will be automatically assigned using default.grin.colors function.</p>
</td></tr>
<tr><td><code id="grin.stats.lsn.plot_+3A_expand">expand</code></td>
<td>
<p>Controls ratio of the feature locus (start and end position) to the whole plot with default value = 0.0005 (setting expand=0 will only plot the locus from the start to the end position without any of the upstream or downstream regions of the feature).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function return a plot with all lesions that affect either a gene regulatory feature of interest. Top panel of the plot will has all different types of lesions affecting the loci color coded according to the figure legend. Lower panel of the plot has all the GRIN statistics of the feature that include number of subjects affected by each type of lesions, -log10 p, and –log10q values showing if the feature is significantly affected by the corresponding lesion category. This plot has no panel for transcripts table as regulatory features typically do not have this kind of information.
</p>


<h3>Value</h3>

<p>Function return a plot with all types of lesions that spans either a gene or regulatory feature of interest in addition to the locus GRIN statistics without adding the transcripts panel.
</p>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+grin.stats">grin.stats()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)
data(hg19.chrom.size)

# run GRIN analysis
grin.results=grin.stats(lesion.data,
                        hg19.gene.annotation,
                        hg19.chrom.size)

# Plots showing different types of lesions and GRIN stats for a gene of interest (WT1):
grin.stats.lsn.plot(grin.results,
                     feature="ENSG00000184937")

# same function can be used to plot lesion data and GRIN statistics of regulatory features
# that typically do not have transcripts track to add to the plot.
</code></pre>

<hr>
<h2 id='hg19_cytoband'>GRCh37 Chromosome Cytobands</h2><span id='topic+hg19_cytoband'></span>

<h3>Description</h3>

<p>The dataset has the start and end positions in base pairs of all 22 autosomes in addition to X and Y chromosome cytobands for Human-GRCh37 (hg19) genome assembly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hg19_cytoband
</code></pre>


<h3>Format</h3>



<h4><code>hg19_cytoband</code></h4>

<p>A data frame with 862 rows and 5 columns:
</p>

<dl>
<dt>chrom</dt><dd><p>The chromosome number.</p>
</dd>
<dt>chromStart</dt><dd><p>The cytoband start position on the chromosome in base pairs.</p>
</dd>
<dt>chromEnd</dt><dd><p>The cytoband end position on the chromosome in base pairs.</p>
</dd>
<dt>name</dt><dd><p>The cytoband name.</p>
</dd>
<dt>gieStain</dt><dd><p>The coloring scheme of the cytobands.</p>
</dd>
</dl>




<h3>Source</h3>

<p>The Chromosome cytobands data file was downloaded from the UCSC genome browser for GRCh37 genome assembly <a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg19/database/">https://hgdownload.soe.ucsc.edu/goldenPath/hg19/database/</a>.
</p>

<hr>
<h2 id='hg19.chrom.size'>Chromosome Length Data File</h2><span id='topic+hg19.chrom.size'></span>

<h3>Description</h3>

<p>The file has the size of 22 autosomes in addition to X and Y chromosomes in base pairs directly retrieved from chr.info txt files available on the UCSC genome browser using get.chrom.length function and &quot;Human-GRCh37&quot; as a genome assembly option (hg19).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hg19.chrom.size
</code></pre>


<h3>Format</h3>



<h4><code>hg19.chrom.size</code></h4>

<p>A data frame with 24 rows and 2 columns:
</p>

<dl>
<dt>chrom</dt><dd><p>The chromosome number.</p>
</dd>
<dt>size</dt><dd><p>The chromosome length in base pairs.</p>
</dd>
</dl>




<h3>Source</h3>

<p>Chromosome size data directly retrieved from chr.info txt files available on the UCSC genome browser using get.chrom.length function and &quot;Human-GRCh37&quot; as a genome assembly option (hg19).
</p>

<hr>
<h2 id='hg19.gene.annotation'>Example Gene Annotation Data File</h2><span id='topic+hg19.gene.annotation'></span>

<h3>Description</h3>

<p>The file has an example annotation data of 420 genes (same set of genes in the gene expression data file) directly retrieved from Ensembl BioMart database using get.ensembl.annotation function and &quot;Human-GRCh37&quot; as a genome assembly option (hg19).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hg19.gene.annotation
</code></pre>


<h3>Format</h3>



<h4><code>hg19.gene.annotation</code></h4>

<p>A data frame with 420 rows and 9 columns:
</p>

<dl>
<dt>gene</dt><dd><p>Column has the gene ensembl ID</p>
</dd>
<dt>chrom</dt><dd><p>The chromosome on which the gene is located</p>
</dd>
<dt>loc.start</dt><dd><p>Gene start position in base pairs</p>
</dd>
<dt>loc.end</dt><dd><p>Gene end position in base pairs</p>
</dd>
<dt>description</dt><dd><p>Description of the gene name </p>
</dd>
<dt>gene.name</dt><dd><p>Gene symbol</p>
</dd>
<dt>biotype</dt><dd><p>Gene classes that include protein coding genes, long noncoding RNAs (lncRNAs), microRNAs (miRNAs), small nuclear RNAs (snRNA), small nucleolar RNAs (snoRNA), immunoglobulins (IGs), T-cell receptors (TCRs) and pseudogens.</p>
</dd>
<dt>chrom.strand</dt><dd><p>The chromosome strand on which the gene is located forward (1) or reverse (-1).</p>
</dd>
<dt>chrom.band</dt><dd><p>The chromosome band on which the gene is located.</p>
</dd>
</dl>




<h3>Source</h3>

<p>Data was directly retrieved from Ensembl BioMart database using get.ensembl.annotation function and &quot;Human-GRCh37&quot; as a genome assembly option (hg19).
</p>

<hr>
<h2 id='hg38_cytoband'>GRCh38 Chromosome Cytobands</h2><span id='topic+hg38_cytoband'></span>

<h3>Description</h3>

<p>The dataset has the start and end positions in base pairs of all 22 autosomes in addition to X and Y chromosome cytobands for Human-GRCh38 (hg38) genome assembly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hg38_cytoband
</code></pre>


<h3>Format</h3>



<h4><code>hg38_cytoband</code></h4>

<p>A data frame with 1,549 rows and 5 columns:
</p>

<dl>
<dt>chrom</dt><dd><p>The chromosome number.</p>
</dd>
<dt>chromStart</dt><dd><p>The cytoband start position on the chromosome in base pairs.</p>
</dd>
<dt>chromEnd</dt><dd><p>The cytoband end position on the chromosome in base pairs.</p>
</dd>
<dt>name</dt><dd><p>The cytoband name.</p>
</dd>
<dt>gieStain</dt><dd><p>The coloring scheme of the cytobands.</p>
</dd>
</dl>




<h3>Source</h3>

<p>The Chromosome cytobands data file was downloaded from the UCSC genome browser for GRCh38 genome assembly <a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/">https://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/</a>.
</p>

<hr>
<h2 id='KW.hit.express'>Associate Lesion with Expression Data Using Kruskal-Wallis Test</h2><span id='topic+KW.hit.express'></span>

<h3>Description</h3>

<p>Function uses Kruskal-Wallis test to evaluate the association between lesion groups and expression level of the same corresponding gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KW.hit.express(alex.data, gene.annotation, min.grp.size = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="KW.hit.express_+3A_alex.data">alex.data</code></td>
<td>
<p>output of the alex.prep.lsn.expr function. It's a list of three data tables that include &quot;row.mtch&quot;, &quot;alex.expr&quot; with expression data, &quot;alex.lsn&quot; with lesion data. Rows of alex.expr, and &quot;alex.lsn&quot; matrices are ordered by gene ensembl IDs and columns are ordered by patient ID.</p>
</td></tr>
<tr><td><code id="KW.hit.express_+3A_gene.annotation">gene.annotation</code></td>
<td>
<p>Gene annotation data either provided by the user or retrieved from ensembl BioMart database using get.ensembl.annotation function included in the GRIN2.0 library. Data.frame should has four columns: &quot;gene&quot; which is the ensembl ID of annotated genes, &quot;chrom&quot; which is the chromosome on which the gene is located, &quot;loc.start&quot; which is the gene start position, and &quot;loc.end&quot; the gene end position.</p>
</td></tr>
<tr><td><code id="KW.hit.express_+3A_min.grp.size">min.grp.size</code></td>
<td>
<p>Minimum number of subjects in a lesion group to be included in the KW test (there should be at least two groups with number of patients &gt; min.grp.size) to run the KW test for a certain gene.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function uses the ensembl IDs in each row of the row.mtch file and run the Kruskal-Wallis test for association between lesion groups of the gene in the &quot;hit.row&quot; column with expression level of the gene in the &quot;expr.row&quot; column. IDs in the two columns should be the same if the KW test will be used to evaluate association between lesion groups and expression level of the same corresponding gene. If the same patient is affected with multiple types of lesions in the same gene for example gain AND mutations, the entry will be denoted as &quot;multiple&quot; and patients without any type of lesions will be coded as &quot;none&quot;.
</p>


<h3>Value</h3>

<p>A data table with multiple columns that include:
</p>
<table role = "presentation">
<tr><td><code>gene</code></td>
<td>
<p>ensembl ID of the gene of interest.</p>
</td></tr>
<tr><td><code>gene.name</code></td>
<td>
<p>Gene name of the gene of interest.</p>
</td></tr>
<tr><td><code>p.KW</code></td>
<td>
<p>Kruskal-Wallis test p-value.</p>
</td></tr>
<tr><td><code>q.KW</code></td>
<td>
<p>Kruskal-Wallis test FDR adjusted q-value.</p>
</td></tr>
<tr><td><code>_n.subjects</code></td>
<td>
<p>Multiple columns with number of subjects with each type of lesion affecting the gene, number of subjects without any lesion and number of subjects with multiple types of lesions.</p>
</td></tr>
<tr><td><code>_mean</code></td>
<td>
<p>Multiple columns with mean expression level of the gene in subjects with each type of lesion, mean expression in subjects without any lesion and mean expression in subjects with multiple types of lesions.</p>
</td></tr>
<tr><td><code>_median</code></td>
<td>
<p>Multiple columns with median expression of the gene in subjects with each type of lesion, median expression in subjects without any lesion and median expression in subjects with multiple types of lesions.</p>
</td></tr>
<tr><td><code>_sd</code></td>
<td>
<p>Multiple columns with standard deviation of the expression level of the gene in subjects with each type of lesion, standard deviation in subjects without any lesion and standard deviation in subjects with multiple types of lesions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Myles Hollander and Douglas A. Wolfe (1973), Nonparametric Statistical Methods. New York: John Wiley &amp; Sons. Pages 115–120.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+alex.prep.lsn.expr">alex.prep.lsn.expr()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expr.data)
data(lesion.data)
data(hg19.gene.annotation)

# prepare expression, lesion data and return the set of genes with both types of data available
# ordered by gene IDs in rows and patient IDs in columns:
alex.data=alex.prep.lsn.expr(expr.data, lesion.data,
                             hg19.gene.annotation, min.expr=1, min.pts.lsn=5)

# run Kruskal-Wallis test for association between lesion groups and expression level of the
# same corresponding gene:
alex.kw.results=KW.hit.express(alex.data, hg19.gene.annotation, min.grp.size=5)
</code></pre>

<hr>
<h2 id='lesion.data'>Example T-ALL Dataset Lesion Data</h2><span id='topic+lesion.data'></span>

<h3>Description</h3>

<p>Lesion data file showing copy number variations, single nucleotide variations and structrual rearrangments affecting 265 newly diagnosed T-cell Acute Lymphoblastic Leukemia (T-ALL) patients that was reported by Liu, Yu, et al. (2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lesion.data
</code></pre>


<h3>Format</h3>



<h4><code>lesion.data</code></h4>

<p>A data frame with 6,887 rows and 5 columns:
</p>

<dl>
<dt>ID</dt><dd><p>patient identifier for the patient affected by the genomic lesion</p>
</dd>
<dt>chrom</dt><dd><p>the chromosome on which the lesion is located</p>
</dd>
<dt>loc.start</dt><dd><p>the lesion start position in base pairs</p>
</dd>
<dt>loc.end</dt><dd><p>the lesion end position in base pairs</p>
</dd>
<dt>lsn.type</dt><dd><p>the lesion type for example gain, loss, mutation, fusion, etc...</p>
</dd>
</dl>




<h3>Source</h3>

<p>extracted from the supplementary material tables of the published Liu, Yu, et al. (2017) manuscript <a href="https://www.nature.com/articles/ng.3909#Sec27">https://www.nature.com/articles/ng.3909#Sec27</a>
</p>

<hr>
<h2 id='lsn.transcripts.plot'>Lesions Gene Transcripts Plot</h2><span id='topic+lsn.transcripts.plot'></span>

<h3>Description</h3>

<p>Function prepare a plot with all types of lesions that spans either a gene or a region of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lsn.transcripts.plot(
  grin.res,
  genome,
  gene = NULL,
  transTrack = TRUE,
  lsn.clrs = NULL,
  chrom = NULL,
  plot.start = NULL,
  plot.end = NULL,
  lesion.grp = NULL,
  spec.lsn.clr = NULL,
  extend.left = NULL,
  extend.right = NULL,
  expand = 5e-04,
  hg38.transcripts = NULL,
  hg19.cytoband = NULL,
  hg38.cytoband = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lsn.transcripts.plot_+3A_grin.res">grin.res</code></td>
<td>
<p>GRIN results (output of the grin.stats function).</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_genome">genome</code></td>
<td>
<p>either &quot;hg19&quot; or &quot;hg38&quot; genome assemblies can be specified based on the genome assembly that has been used to prepare the lesion data.</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_gene">gene</code></td>
<td>
<p>Gene name of interest.</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_transtrack">transTrack</code></td>
<td>
<p>In case of plots that span large genomic region such as a chromosome band or the whole chromosome, this argument should be specified as 'FALSE' to exclude the transcripts track from the plot.</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_lsn.clrs">lsn.clrs</code></td>
<td>
<p>Lesion colors for the regional gene plot (If not provided by the user, colors will be automatically assigned using default.grin.colors function).</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_chrom">chrom</code></td>
<td>
<p>chromosome number (should be only specified in the locus plots where plot.start and plot.end for the locus of interest are specified).</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_plot.start">plot.start</code></td>
<td>
<p>start position of the locus of interest.</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_plot.end">plot.end</code></td>
<td>
<p>end position of the locus of interest.</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_lesion.grp">lesion.grp</code></td>
<td>
<p>lesion group of interest (should be only specified in locus plots when chrom, plot.start, plot.end are specified).</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_spec.lsn.clr">spec.lsn.clr</code></td>
<td>
<p>color assigned to the lesion of interest (should be specified when chrom, plot.start, plot.end and lesion.grp are specified).</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_extend.left">extend.left</code></td>
<td>
<p>specified number will be used to manually align the left side of the gene transcripts track directly retrieved from ensembl database with the gene lesions track if needed.</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_extend.right">extend.right</code></td>
<td>
<p>specified number will be used to manually align the right side of the gene transcripts track directly retrieved from ensembl database with the gene lesions track if needed.</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_expand">expand</code></td>
<td>
<p>Controls ratio of the gene locus (start and end position) to the whole plot with default value = 0.0005 (setting expand=0 will only plot the gene locus from the start to the end position without any of the upstream or downstream regions of the gene).</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_hg38.transcripts">hg38.transcripts</code></td>
<td>
<p>transcripts data retrieved from annotation hub for hg38 version 110 (should be only specified if genome=&quot;hg38&quot;).</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_hg19.cytoband">hg19.cytoband</code></td>
<td>
<p>hg19 chromosome bands start and end data in base pair (should be only specified if genome=&quot;hg19&quot;).</p>
</td></tr>
<tr><td><code id="lsn.transcripts.plot_+3A_hg38.cytoband">hg38.cytoband</code></td>
<td>
<p>hg38 chromosome bands start and end data in base pair (should be only specified if genome=&quot;hg38&quot;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function return a plot with all lesions that affect either a gene or a region of interest. Top panel of the regional gene plot has the transcripts track with all transcripts annotated to the gene of interest directly retrieved from ensembl database. The middle panel will has all different types of lesions affecting the gene color coded according to the figure legend. Lower panel of the plot has all the GRIN statistics of the gene that include number of subjects affected by each type of lesions, -log10 p, and –log10q values showing if the gene is significantly affected by the corresponding lesion category. If a certain locus is specified, only transcripts track and the lesion panel will be returned (GRIN results panel will not be added to the plot). In case of plots that span large genomic region such as a chromosome band or the whole chromosome and by specifying transTrack=FALSE, transcripts track will not be added to the plot as well.
</p>


<h3>Value</h3>

<p>Function will return either a gene plot with the transcripts track, lesions panel and GRIN statistic for the gene of interest, a plot with all lesions and transcripts aligned to a certain locus of interest if chrom, plot.start and plot.end were specified or a plot with all lesions affecting a region of interest without the transcripts track.
</p>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+grin.stats">grin.stats()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(lesion.data)
data(hg19.gene.annotation)
data(hg19.chrom.size)
data(hg19_cytoband)
data(hg38_cytoband)

# run GRIN analysis using grin.stats function
grin.results=grin.stats(lesion.data,
                        hg19.gene.annotation,
                        hg19.chrom.size)

# Plots showing different types of lesions affecting a gene of interest with a transcripts
# track that show all the gene transcripts retrieved from Ensembl (hg19 genome assembly):
WT1.gene.plot=lsn.transcripts.plot(grin.results, genome="hg19", gene="WT1",
                                    hg19.cytoband=hg19_cytoband)

# Plots showing different types of lesions affecting a region of interest with a transcripts
# track added to the plot:
locus.plot=lsn.transcripts.plot(grin.results, genome="hg19", hg19.cytoband=hg19_cytoband,
                                chrom=9, plot.start=21800000, plot.end=22200000,
                                lesion.grp = "loss", spec.lsn.clr = "blue")

# Plots Showing Different Types of Lesions Affecting a region of Interest without plotting the
# transcripts track (this will allow plotting a larger locus of the chromosome such as a
# chromosome band (should specify transTrack = FALSE):
noTranscripts=lsn.transcripts.plot(grin.results, genome="hg19", transTrack = FALSE,
                                   hg19.cytoband=hg19_cytoband, chrom=9, plot.start=19900000,
                                   plot.end=25600000, lesion.grp = "loss", spec.lsn.clr = "blue")

 # Plots Showing Different Types of Lesions Affecting the whole chromosome:
 chrom.plot=lsn.transcripts.plot(grin.results, genome="hg19", transTrack = FALSE,
                                 hg19.cytoband=hg19_cytoband, chrom=9, plot.start=1,
                                  plot.end=141000000)

# for GRCh38 (hg38) genome assembly, users should first call the AnnotationHub() web resource then
# specify ah[["AH113665"]] to retrieve the human hg38 gene transcripts. This formal class
# EnsDb object should be called afterwards in the 'hg38.transcripts' argument to return gene and
# regional plots.

</code></pre>

<hr>
<h2 id='onco.print.props'>Oncoprint proportions</h2><span id='topic+onco.print.props'></span>

<h3>Description</h3>

<p>The function order lesion types based on their average size and assign the proportion of the oncoprint rectangle that should be color filled based on the average size of each lesion type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>onco.print.props(lsn.data, clr = NULL, hgt = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="onco.print.props_+3A_lsn.data">lsn.data</code></td>
<td>
<p>data.frame with 5 columns including &quot;ID&quot; which is the subject identifier, &quot;chrom&quot; which is the chromosome on which the lesion is located, &quot;loc.start&quot; with lesion start position, &quot;loc.end&quot; which is the lesion end position), and &quot;lsn.type&quot; which is the lesion category for example gain, mutation, etc..).</p>
</td></tr>
<tr><td><code id="onco.print.props_+3A_clr">clr</code></td>
<td>
<p>Lesion colors (If not provided by the user, colors will be automatically assigned using default.grin.colors function).</p>
</td></tr>
<tr><td><code id="onco.print.props_+3A_hgt">hgt</code></td>
<td>
<p>Manually assign the proportion of the oncoprint rectangle that should be color filled for each lesion group.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Some patients might be affected by two or more lesion types in the same gene for example gain AND mutations. To get all lesion types represented in the same rectangle in the oncoprint, this function order lesion types based on the average size of each type and assign the proportion of the oncoprint rectangle that should be color filled based on the average size of each lesion type. Color filled proportion of the oncoprint rectangles can be also specified by the user for  each lesion type based on the hgt parameter.
</p>


<h3>Value</h3>

<p>Function return a list of three lists specifying the color assigned to each lesion type, the proportion of the rectangle that should be color filled in the oncoprint based on the average size of each lesion type, and the legend parameters.
</p>


<h3>Author(s)</h3>

<p>Lakshmi Patibandla <a href="mailto:LakshmiAnuhya.Patibandla@stjude.org">LakshmiAnuhya.Patibandla@stjude.org</a>, Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
onco.props=onco.print.props(lesion.data, hgt = c("gain"=4, "loss"=3, "mutation"=2, "fusion"=1))
# if hgt argument is not specified, the lesion category "mutation" for single point mutations will
# be assigned size=1 because it has the smallest average lesion size and will have the smallest
# proportion of the filled oncoprint rectangles 1/4=0.25
</code></pre>

<hr>
<h2 id='order.index.gene.data'>Order Index Gene Data</h2><span id='topic+order.index.gene.data'></span>

<h3>Description</h3>

<p>This function order and index gene annotation data by chromosome on which the gene is located, gene start, and end positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>order.index.gene.data(gene.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="order.index.gene.data_+3A_gene.data">gene.data</code></td>
<td>
<p>data.frame with gene annotation data either provided by the user or retrieved from ensembl BioMart database using get.ensembl.annotation function included in the GRIN2.0 library. data.frame should has four columns that include &quot;gene&quot; which is the ensembl ID of the annotated genes to which the lesion data will be overlapped, &quot;chrom&quot; which is the chromosome on which the gene is located, &quot;loc.start&quot; which is the gene start position, and &quot;loc.end&quot; the gene end position.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table role = "presentation">
<tr><td><code>gene.data</code></td>
<td>
<p>Input gene annotation data</p>
</td></tr>
<tr><td><code>gene.index</code></td>
<td>
<p>data.frame with two columns of ordered row start and row end based on the number of genes annotated to each chromosome.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Pounds, Stan, et al. (2013) A genomic random interval model for statistical analysis of genomic lesion data.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hg19.gene.annotation)

ordered.genes=order.index.gene.data(hg19.gene.annotation)
</code></pre>

<hr>
<h2 id='order.index.lsn.data'>Order Index Lesion Data</h2><span id='topic+order.index.lsn.data'></span>

<h3>Description</h3>

<p>This function order and index lesion data by lesion type, the chromosome on which the lesion is located , and subject.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>order.index.lsn.data(lsn.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="order.index.lsn.data_+3A_lsn.data">lsn.data</code></td>
<td>
<p>data.frame with lesion data prepared by the user in a GRIN compatible format. The data.frame should has five columns that include &quot;ID&quot; which is a column with id of the patient affected by the lesion, &quot;chrom&quot; which is the chromosome on which the lesion is located, &quot;loc.start&quot; which is the lesion start position, &quot;loc.end&quot; the lesion end position and &quot;lsn.type&quot; which is the lesion type for example gain, loss, mutation, fusion, etc...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table role = "presentation">
<tr><td><code>lsn.data</code></td>
<td>
<p>Input lesion data</p>
</td></tr>
<tr><td><code>lsn.index</code></td>
<td>
<p>data.frame with row start and row end for each type of lesions affecting each subject on a certain chromosome. For example, if a certain patient is affected by 1 deletion on chromosome 5, row start wil be equal to row end for loss on chromosome 5. However, if the patient is affected by 4 deletions, difference between row.start and row.end will be 3.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Pounds, Stan, et al. (2013) A genomic random interval model for statistical analysis of genomic lesion data.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)

ordered.lsn=order.index.lsn.data(lesion.data)
</code></pre>

<hr>
<h2 id='pathways'>List of Genes Annotated to a Group of Pathways</h2><span id='topic+pathways'></span>

<h3>Description</h3>

<p>The dataset has a list of genes annotated to a group of different pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathways
</code></pre>


<h3>Format</h3>



<h4><code>pathways</code></h4>

<p>A data frame with 121 rows and 3 columns:
</p>

<dl>
<dt>gene.name</dt><dd><p>Gene symbol.</p>
</dd>
<dt>ensembl.id</dt><dd><p>Gene ensembl ID.</p>
</dd>
<dt>pathway</dt><dd><p>The pathway to which the gene is annotated.</p>
</dd>
</dl>




<h3>Source</h3>

<p>Data was extracted from the supplementary material tables of the published Liu, Yu, et al. (2017) manuscript <a href="https://www.nature.com/articles/ng.3909#Sec27">https://www.nature.com/articles/ng.3909#Sec27</a>
</p>

<hr>
<h2 id='prep.binary.lsn.mtx'>Prepare Binary Lesion Matrix</h2><span id='topic+prep.binary.lsn.mtx'></span>

<h3>Description</h3>

<p>Prepares a lesion matrix with each gene affected by a certain type of lesion as a row and each patient as a column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep.binary.lsn.mtx(ov.data, min.ngrp = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep.binary.lsn.mtx_+3A_ov.data">ov.data</code></td>
<td>
<p>list of six data.frames that represent the output results of the find.gene.lsn.overlaps function.</p>
</td></tr>
<tr><td><code id="prep.binary.lsn.mtx_+3A_min.ngrp">min.ngrp</code></td>
<td>
<p>if specified, rows with number of patients affected by a specific type of lesion that's less than the specified number will be discarded (default is 0; function will return all genes affected by a lesion in at least one patient), for example if only one patient is affected by gain in MYB gene.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function uses the output results of the find.gene.lsn.overlaps function and create a binary lesion matrix with each gene affected by certain lesion type as a row and each patient as a column. Rownames are labelled as gene.ID_lesion.type (for example: ENSG00000118513_gain for gains affecting MYB gene). The entry for each patient in the table will be denoted as 1 if the patient is affected by this specific type of lesion in the gene, for example gain in MYB gene (ENSG00000118513) or 0 otherwise.
</p>


<h3>Value</h3>

<p>The function returns a binary lesion matrix with each row labelled as gene.ID_lesion.type and each column is a patient. Entry for each patient in the table will be denoted as 1 if the gene is affected by this specific type of lesion or 0 otherwise.
</p>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Pounds, Stan, et al. (2013) A genomic random interval model for statistical analysis of genomic lesion data.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+prep.gene.lsn.data">prep.gene.lsn.data()</a></code>, <code><a href="#topic+find.gene.lsn.overlaps">find.gene.lsn.overlaps()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)

# prepare gene and lesion data for later computations:
prep.gene.lsn=prep.gene.lsn.data(lesion.data,
                                 hg19.gene.annotation)

# determine lesions that overlap each gene (locus):
gene.lsn.overlap=find.gene.lsn.overlaps(prep.gene.lsn)

# prepare the lesion binary matrix with a minimum of 5 patients affected by the lesion to be
# included in the final matrix:
lsn.binary.mtx=prep.binary.lsn.mtx(gene.lsn.overlap, min.ngrp=5)
</code></pre>

<hr>
<h2 id='prep.gene.lsn.data'>Prepare Gene and Lesion Data</h2><span id='topic+prep.gene.lsn.data'></span>

<h3>Description</h3>

<p>This function prepare gene and lesion data for later GRIN computations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep.gene.lsn.data(lsn.data, gene.data, mess.freq = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep.gene.lsn.data_+3A_lsn.data">lsn.data</code></td>
<td>
<p>data.frame with lesion data prepared by the user in a GRIN compatible format. The data.frame should has five columns that include &quot;ID&quot; which is the patient ID, &quot;chrom&quot; which is the chromosome on which the lesion is located, &quot;loc.start&quot; which is the lesion start position, &quot;loc.end&quot; the lesion end position and &quot;lsn.type&quot; which is the lesion type for example gain, loss, mutation, fusion, etc...</p>
</td></tr>
<tr><td><code id="prep.gene.lsn.data_+3A_gene.data">gene.data</code></td>
<td>
<p>gene annotation data with four required columns: &quot;gene&quot; has ensembl ID, &quot;chrom&quot; which is chromosome on which the gene is located, &quot;loc.start&quot; gene start position, &quot;loc.end&quot; which is the gene end position</p>
</td></tr>
<tr><td><code id="prep.gene.lsn.data_+3A_mess.freq">mess.freq</code></td>
<td>
<p>message frequency: display message every mess.freq^th lesion block (default is 10).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function order and index gene and lesion data for later computations. Output of this function is used to ovelap gene and lesion data using find.gene.lsn.overlaps function.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table role = "presentation">
<tr><td><code>lsn.data</code></td>
<td>
<p>Input lesion data.</p>
</td></tr>
<tr><td><code>gene.data</code></td>
<td>
<p>Input gene annotation data.</p>
</td></tr>
<tr><td><code>gene.lsn.data</code></td>
<td>
<p>data.frame ordered by gene and lesions start positions. Gene start position is coded as 1 in the cty column and gene end position is coded as 4. Lesion start position is coded as 2 in the cty column and lesion end position is coded as 3.</p>
</td></tr>
<tr><td><code>gene.index</code></td>
<td>
<p>data.frame that shows row start and row end for each chromosome in the gene.lsn.data table.</p>
</td></tr>
<tr><td><code>lsn.index</code></td>
<td>
<p>data.frame that shows row start and row end for each lesion in the gene.lsn.data table.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Pounds, Stan, et al. (2013) A genomic random interval model for statistical analysis of genomic lesion data.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+order.index.gene.data">order.index.gene.data()</a></code>, <code><a href="#topic+order.index.lsn.data">order.index.lsn.data()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)

# prepare gene and lesion data for later computations:
prep.gene.lsn=prep.gene.lsn.data(lesion.data,
                                 hg19.gene.annotation)
</code></pre>

<hr>
<h2 id='prep.lsn.type.matrix'>Prepare Lesion Type Matrix</h2><span id='topic+prep.lsn.type.matrix'></span>

<h3>Description</h3>

<p>The function prepare a lesion matrix with all types of lesions affecting certain gene as a row and each patient as a column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep.lsn.type.matrix(ov.data, min.ngrp = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep.lsn.type.matrix_+3A_ov.data">ov.data</code></td>
<td>
<p>list of six data.frames that represent the output results of the find.gene.lsn.overlaps function.</p>
</td></tr>
<tr><td><code id="prep.lsn.type.matrix_+3A_min.ngrp">min.ngrp</code></td>
<td>
<p>if specified, rows with number of patients affected by all different types of lesions that's less than the specified number will be discarded (default is 0; will return all genes affected by any type of lesions in at least one patient).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns a lesion matrix with each row as a gene and each column is a patient. If a gene is affected by one type of lesions in a certain patient, the entry will be labelled by lesion type (for example: gain OR mutation). However, if the same gene is affected by more than one type of lesions in a certain patient (for example: gain AND mutation), the entry will be labelled as &quot;multiple&quot;. If the gene is not affected by any lesion, the entry for this patient will be labelled as &quot;none&quot;.
</p>


<h3>Value</h3>

<p>The function returns a lesion matrix with all types of lesions affecting certain gene as a row and each patient as a column.
</p>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Pounds, Stan, et al. (2013) A genomic random interval model for statistical analysis of genomic lesion data.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+prep.gene.lsn.data">prep.gene.lsn.data()</a></code>, <code><a href="#topic+find.gene.lsn.overlaps">find.gene.lsn.overlaps()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)

# prepare gene and lesion data for later computations:
prep.gene.lsn=prep.gene.lsn.data(lesion.data,
                                 hg19.gene.annotation)

# determine lesions that overlap each gene (locus):
gene.lsn.overlap=find.gene.lsn.overlaps(prep.gene.lsn)

# prepare the lesion matrix with a minimum of 5 patients affected by any type of lesion in the
# gene to be included in the final matrix
lsn.type.mtx=prep.lsn.type.matrix(gene.lsn.overlap, min.ngrp=5)
</code></pre>

<hr>
<h2 id='prob.hits'>Find Probablity of Locus Hit</h2><span id='topic+prob.hits'></span>

<h3>Description</h3>

<p>The function evaluates the probability of a locus to be affected by one or a constellation of multiple types of lesions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prob.hits(hit.cnt, chr.size = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prob.hits_+3A_hit.cnt">hit.cnt</code></td>
<td>
<p>output results of the count.hits function with number of subjects and number of hits affecting each locus.</p>
</td></tr>
<tr><td><code id="prob.hits_+3A_chr.size">chr.size</code></td>
<td>
<p>data.frame with the size of the 22 autosomes, in addition to X and Y chromosomes in base pairs. The data.frame should has two columns &quot;chrom&quot; with the chromosome number and &quot;size&quot; for the size of the chromosome in base pairs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function computes p-value for the probability of each locus (gene or regulatory feature) to be affected by different types of lesions based on a convolution of independent but non-identical Bernoulli distributions to determine whether a certain locus has an abundance of lesions that is statistically significant.In addition, FDR-adjusted q value is computed for each locus based on Pounds &amp; Cheng (2006) estimator of the proportion of tests with a true null (pi.hat). The function also evaluates if a certain locus is affected by a constellation of multiple types of lesions and computes a p and adjusted q values for the locus to be affected by one type of lesions (p1), two types of lesions (p2), etc...
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table role = "presentation">
<tr><td><code>gene.hits</code></td>
<td>
<p>data table of GRIN results that include gene annotation, number of subjects affected by each lesion type for example gain, loss, mutation, etc.., and number of hits affecting each locus. The GRIN results table will also include P and FDR adjusted q-values showing the probability of each locus of being affected by one or a constellation of multiple types of lesions.</p>
</td></tr>
<tr><td><code>lsn.data</code></td>
<td>
<p>input lesion data</p>
</td></tr>
<tr><td><code>gene.data</code></td>
<td>
<p>input gene annotation data</p>
</td></tr>
<tr><td><code>gene.lsn.data</code></td>
<td>
<p>each row represent a gene overlapped by a certain lesion. Column &quot;gene&quot; shows the overlapped gene ensembl ID and &quot;ID&quot;&quot; column has the patient ID.</p>
</td></tr>
<tr><td><code>chr.size</code></td>
<td>
<p>data table showing the size of the 22 autosomes, in addition to X and Y chromosomes in base pairs.</p>
</td></tr>
<tr><td><code>gene.index</code></td>
<td>
<p>data.frame with overlapped gene-lesion data rows that belong to each chromosome in the gene.lsn.data table.</p>
</td></tr>
<tr><td><code>lsn.index</code></td>
<td>
<p>data.frame that shows the overlapped gene-lesion data rows taht belong to each lesion in the gene.lsn.data table.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Pounds, Stan, et al. (2013) A genomic random interval model for statistical analysis of genomic lesion data.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+prep.gene.lsn.data">prep.gene.lsn.data()</a></code>, <code><a href="#topic+find.gene.lsn.overlaps">find.gene.lsn.overlaps()</a></code>, <code><a href="#topic+count.hits">count.hits()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)
data(hg19.chrom.size)

# prepare gene and lesion data for later computations:
prep.gene.lsn=prep.gene.lsn.data(lesion.data,
                                 hg19.gene.annotation)

# determine lesions that overlap each gene (locus):
gene.lsn.overlap=find.gene.lsn.overlaps(prep.gene.lsn)

# count number of subjects affected by different types of lesions and number of hits that affect
# each locus:
count.subj.hits=count.hits(gene.lsn.overlap)

# compute the probability of each locus to be affected by one or a constellation of multiple
# types of lesion
hits.prob=prob.hits(count.subj.hits, hg19.chrom.size)
</code></pre>

<hr>
<h2 id='top.alex.waterfall.plots'>Waterfall Plots for Lesion and Expression Data of Top Significant Genes</h2><span id='topic+top.alex.waterfall.plots'></span>

<h3>Description</h3>

<p>Function return waterfall plots for top significant genes in the KW results table based on the specified q value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>top.alex.waterfall.plots(out.dir, alex.data, alex.kw.results, q, lsn.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="top.alex.waterfall.plots_+3A_out.dir">out.dir</code></td>
<td>
<p>Path to the folder where the waterfall plots of selected genes based on the specified q value of the KW results table will be added.</p>
</td></tr>
<tr><td><code id="top.alex.waterfall.plots_+3A_alex.data">alex.data</code></td>
<td>
<p>output of the alex.prep.lsn.expr function. It's a list of three data tables that include &quot;row.mtch&quot;, &quot;alex.expr&quot; with expression data, &quot;alex.lsn&quot; with lesion data. Rows of alex.expr, and &quot;alex.lsn&quot; matrices are ordered by gene ensembl IDs and columns are ordered by patient ID.</p>
</td></tr>
<tr><td><code id="top.alex.waterfall.plots_+3A_alex.kw.results">alex.kw.results</code></td>
<td>
<p>ALEX Kruskal-Wallis test results (output of the KW.hit.express function).</p>
</td></tr>
<tr><td><code id="top.alex.waterfall.plots_+3A_q">q</code></td>
<td>
<p>Maximum allowed KW q-value threshold for a gene to be plotted based on the output of the KW.hit.express function.</p>
</td></tr>
<tr><td><code id="top.alex.waterfall.plots_+3A_lsn.data">lsn.data</code></td>
<td>
<p>Lesion data in a GRIN compatible format.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function will return waterfall plots for top significant genes in the KW results table based on the user specified q-value threshold of the KW test.The plots will be added to the user specified outdir folder.
</p>


<h3>Value</h3>

<p>Function will return waterfall plots for top significant genes in the KW results table.
</p>


<h3>Author(s)</h3>

<p>Abdelrahman Elsayed <a href="mailto:abdelrahman.elsayed@stjude.org">abdelrahman.elsayed@stjude.org</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+alex.prep.lsn.expr">alex.prep.lsn.expr()</a></code>, <code><a href="#topic+KW.hit.express">KW.hit.express()</a></code>, <code><a href="#topic+alex.waterfall.prep">alex.waterfall.prep()</a></code>, <code><a href="#topic+alex.waterfall.plot">alex.waterfall.plot()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expr.data)
data(lesion.data)
data(hg19.gene.annotation)

# prepare expression, lesion data and return the set of genes with both types of data available
# ordered by gene IDs in rows and patient IDs in columns:
alex.data=alex.prep.lsn.expr(expr.data, lesion.data,
                             hg19.gene.annotation, min.expr=5,
                              min.pts.lsn=5)

# run KW test for association between lesion groups and expression level of the same gene:
alex.kw.results=KW.hit.express(alex.data, hg19.gene.annotation, min.grp.size=5)

# return waterfall plots for a list of top significant genes to a pre-specified folder:
dir.create(resultsFolder &lt;- file.path(tempdir(), "temp.out"))

waterfall.plts=top.alex.waterfall.plots(out.dir=resultsFolder,
                                        alex.data, alex.kw.results,
                                        1e-15, lesion.data)

unlink(resultsFolder, recursive = TRUE)
</code></pre>

<hr>
<h2 id='write.grin.xlsx'>Write GRIN Results</h2><span id='topic+write.grin.xlsx'></span>

<h3>Description</h3>

<p>The function Write GRIN results to an excel file with multiple sheets that include GRIN results, lesion data, gene annotation data, chromosome size, gene-lesion overlap and methods paragraph.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.grin.xlsx(grin.result, output.file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write.grin.xlsx_+3A_grin.result">grin.result</code></td>
<td>
<p>output results of the grin.stats function.</p>
</td></tr>
<tr><td><code id="write.grin.xlsx_+3A_output.file">output.file</code></td>
<td>
<p>output file name &quot;.xlsx&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function return an excel file with seven sheets that include:
</p>
<table role = "presentation">
<tr><td><code>gene.hits</code></td>
<td>
<p>data table of GRIN results that include gene annotation, number of subjects affected by each lesion type for example gain, loss, mutation, etc.., and number of hits affecting each locus. The GRIN results table will also include P and FDR adjusted q-values showing the probability of each locus of being affected by one or a constellation of multiple types of lesions.</p>
</td></tr>
<tr><td><code>gene.lsn.data</code></td>
<td>
<p>each row represent a gene overlapped by a certain lesion. Column &quot;gene&quot; shows the overlapped gene ensembl ID and &quot;ID&quot;&quot; column has the patient ID.</p>
</td></tr>
<tr><td><code>lsn.data</code></td>
<td>
<p>input lesion data</p>
</td></tr>
<tr><td><code>gene.data</code></td>
<td>
<p>input gene annotation data</p>
</td></tr>
<tr><td><code>chr.size</code></td>
<td>
<p>data table showing the size of the 22 autosomes, in addition to X and Y chromosomes in base pairs.</p>
</td></tr>
<tr><td><code>interpretation</code></td>
<td>
<p>provides some details about  the content of each sheet in the output excel file and interpretation of each column in the &quot;gene.hits&quot; GRIN results table.</p>
</td></tr>
<tr><td><code>method.paragraph</code></td>
<td>
<p>include a paragraph that explains the GRIN model and cite some references.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Pounds, Stan, et al. (2013) A genomic random interval model for statistical analysis of genomic lesion data.
</p>
<p>Cao, X., Elsayed, A. H., &amp; Pounds, S. B. (2023). Statistical Methods Inspired by Challenges in Pediatric Cancer Multi-omics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+grin.stats">grin.stats()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lesion.data)
data(hg19.gene.annotation)
data(hg19.chrom.size)

# to directly retreive gene annotation and chromosome size files from Ensembl BioMart database,
# UCSC genome browsers and run the GRIN analysis:
grin.results=grin.stats(lesion.data,
                         hg19.gene.annotation,
                         hg19.chrom.size)

# Write GRIN results in to an excel sheet ".xlsx" using write.grin.xlsx function.
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
