<!DOCTYPE html><html><head><title>Help for package grmsem</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {grmsem}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fit.large'><p>Prefitted model: large data set</p></a></li>
<li><a href='#G.small'><p>Symmetric GRM data: small data set</p></a></li>
<li><a href='#grm.bin.input'><p>grm import function</p></a></li>
<li><a href='#grm.input'><p>grm import function</p></a></li>
<li><a href='#grmsem.biher'><p>grmsem bivariate heritability estimation function.</p></a></li>
<li><a href='#grmsem.fcoher'><p>grmsem factorial co-heritability and co-environmentality estimation function.</p></a></li>
<li><a href='#grmsem.fit'><p>grmsem model fitting function</p></a></li>
<li><a href='#grmsem.stpar'><p>grmsem standardised parameters function</p></a></li>
<li><a href='#grmsem.var'><p>grmsem variance estimation function</p></a></li>
<li><a href='#ph.large'><p>Phenotype data: large data set</p></a></li>
<li><a href='#ph.small'><p>Phenotype data: small data set</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Genetic-Relationship-Matrix Structural Equation Modelling
(GRMSEM)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Beate StPourcain &lt;Beate.StPourcain@mpi.nl&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Quantitative genetics tool supporting the modelling of multivariate 
 genetic variance structures in quantitative data. It allows fitting different 
 models through multivariate genetic-relationship-matrix (GRM) 
 structural equation modelling (SEM) in unrelated individuals, 
 using a maximum likelihood approach. Specifically, 
 it combines genome-wide genotyping information, as captured by GRMs, 
 with twin-research-based SEM techniques, 
 St Pourcain et al. (2017) &lt;<a href="https://doi.org/10.1016%2Fj.biopsych.2017.09.020">doi:10.1016/j.biopsych.2017.09.020</a>&gt;, 
 Shapland et al. (2020) &lt;<a href="https://doi.org/10.1101%2F2020.08.14.251199">doi:10.1101/2020.08.14.251199</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>msm (&ge; 1.6), numDeriv, optimParallel, stats, utils</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>Suggests:</td>
<td>bookdown, knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://CRAN.R-project.org/package=grmsem">https://CRAN.R-project.org/package=grmsem</a>,
<a href="https://gitlab.gwdg.de/beate.stpourcain/grmsem">https://gitlab.gwdg.de/beate.stpourcain/grmsem</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://gitlab.gwdg.de/beate.stpourcain/grmsem/-/issues">https://gitlab.gwdg.de/beate.stpourcain/grmsem/-/issues</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-27 11:16:48 UTC; beapou</td>
</tr>
<tr>
<td>Author:</td>
<td>Beate StPourcain <a href="https://orcid.org/0000-0002-4680-3517"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Alexander Klassmann [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-29 09:20:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='fit.large'>Prefitted model: large data set</h2><span id='topic+fit.large'></span>

<h3>Description</h3>

<p>A large quad-variate data set was simulated assuming an underlying Cholesky model, 
with 5000 observations per trait and high polygenicity (20,000 SNPs per genetic factor).
Genetic trait variances were set to 0.30, 0.60, 0.60 and 0.70 respectively, and residual variances
to 0.70, 0.40, 0.40 and 0.30. The data set is described in full, including genetic and residual
covariances, in the vignette. A 4-variate Cholesky model was fitted to the data as described in 
the vignette and the output has been saved in fit.large.RData.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit.large
</code></pre>


<h3>Format</h3>

<p>gsem.fit output (a list object)
</p>

<dl>
<dt>model.in</dt><dd><p>input parameters</p>
</dd>
<dt>formula</dt><dd><p>fitted formula</p>
</dd>
<dt>model.fit</dt><dd><p>optimisation output</p>
</dd>
<dt>model.out</dt><dd><p>fitted gsem model</p>
</dd>
<dt>VCOV</dt><dd><p>variance covariance matrix</p>
</dd>
<dt>k</dt><dd><p>Number of phenotypes</p>
</dd>
<dt>n</dt><dd><p>Number of observations across all phenotypes</p>
</dd>
<dt>n.obs</dt><dd><p>Number of observations per phenotype</p>
</dd>
<dt>n.ind</dt><dd><p>Number of individuals with at least one phenotype</p>
</dd>
<dt>model</dt><dd><p>gsem model</p>
</dd>
<dt>con</dt><dd><p>constraint</p>
</dd>
<dt>ph.nms</dt><dd><p>phenotype names</p>
</dd>   
</dl>


<hr>
<h2 id='G.small'>Symmetric GRM data: small data set</h2><span id='topic+G.small'></span>

<h3>Description</h3>

<p>A genetic relationship matrix for a small tri-variate data set was simulated assuming an underlying 
Cholesky model, with 100 observations per trait and low polygenicity (150 SNPs per genetic factor).
Genetic trait variances were set to 0.30, 0.60, 0.60 and 0.70 respectively, and residual variances
to 0.70, 0.40, 0.40 and 0.30. The data set is described in full, including genetic and residual
covariances, in the vignette. The traits are influenced by three independent genetic factors 
(A1, A2 and A3, based on a 150 SNPs each) and three independent residual factors (E1, E2 and E3).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>G.small
</code></pre>


<h3>Format</h3>

<p>Symmetric GRM data frame with 100 rows and 100 columns (observations)
</p>

<hr>
<h2 id='grm.bin.input'>grm import function</h2><span id='topic+grm.bin.input'></span>

<h3>Description</h3>

<p>This function imports genetic relationship matrices (GRMs)
in binary format as e.g. stored by the GCTA software command <code>gcta64 --grm test --make-grm-bin --out test</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grm.bin.input(file, size = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grm.bin.input_+3A_file">file</code></td>
<td>
<p>name of the binary grm file. No default.</p>
</td></tr>
<tr><td><code id="grm.bin.input_+3A_size">size</code></td>
<td>
<p>byte size used (typically gcta uses 4). See eponymous parameter of function <code><a href="base.html#topic+readBin">readBin</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of type matrix
</p>
<p><code>grmsem.bin.input</code> imports a binary GCTA GRM (.bin) file and transforms it into a symmetric matrix
</p>

<hr>
<h2 id='grm.input'>grm import function</h2><span id='topic+grm.input'></span>

<h3>Description</h3>

<p>This function imports genetic relationship matrices (GRMs) in gz format as e.g.
stored by the GCTA software command <code>gcta64 --grm test --make-grm-gz --out test</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grm.input(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grm.input_+3A_file">file</code></td>
<td>
<p>the name of the gz compressed grm file. No default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>grmsem.input</code> imports a zipped GCTA GRM (.gz) file and transforms it into a symmetric matrix
</p>

<hr>
<h2 id='grmsem.biher'>grmsem bivariate heritability estimation function.</h2><span id='topic+grmsem.biher'></span>

<h3>Description</h3>

<p>This function estimates the bivariate heritability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grmsem.biher(ph, grmsem.var.out = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grmsem.biher_+3A_ph">ph</code></td>
<td>
<p>Phenotype file as R dataframe (columns: &gt;=2 phenotypes, rows: ni individuals in the same order as G). No default.</p>
</td></tr>
<tr><td><code id="grmsem.biher_+3A_grmsem.var.out">grmsem.var.out</code></td>
<td>
<p>A grmsem.var object with unstandardised parameters (factor loadings). Default NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>grmsem.biher</code> function estimates the bivariate heritability 
(DS, Cholesky, IP and IPC models) from the observed phenotype data and a <code>grmsem.var</code> object. 
All standard errors are derived with the Delta method.
</p>


<h3>Value</h3>

<p><code>grmsem.biher</code> returns a list object consisting of the following matrices:
</p>
<table>
<tr><td><code>VPO</code></td>
<td>
<p>observed phenotypic variance/covariance matrix</p>
</td></tr>
<tr><td><code>VA</code></td>
<td>
<p>estimated genetic variance</p>
</td></tr>
<tr><td><code>BIHER</code></td>
<td>
<p>estimated bivariate heritability (off-diagonals): VA / VPO</p>
</td></tr>
<tr><td><code>BIHER.se</code></td>
<td>
<p>standard error of estimated bivariate heritability</p>
</td></tr>
<tr><td><code>BIHER.Z</code></td>
<td>
<p>Z (wald) of estimated bivariate heritability</p>
</td></tr>
<tr><td><code>BIHER.p</code></td>
<td>
<p>p (Wald) of estimated bivariate heritability</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#(runtime should be less than one minute)

out &lt;- grmsem.fit(ph.small, G.small, LogL = TRUE, estSE = TRUE)
var.out &lt;- grmsem.var(out)
grmsem.biher(ph.small, var.out)
</code></pre>

<hr>
<h2 id='grmsem.fcoher'>grmsem factorial co-heritability and co-environmentality estimation function.</h2><span id='topic+grmsem.fcoher'></span>

<h3>Description</h3>

<p>This function estimates factorial co-heritabilities and factorial co-environmentalities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grmsem.fcoher(grmsem.out = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grmsem.fcoher_+3A_grmsem.out">grmsem.out</code></td>
<td>
<p>grmsem.fit or grmsem.stpar object. Default NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>grmsem.fcoher</code> function can be used to estimate factorial co-heritabilities and factorial co-environmentalities 
for models estimating latent variables (Cholesky, IP or IPC models), based on <code>grmsem.fit</code> or <code>grmsem.stpar</code> objects.
The factorial co-heritability of a genetic factor m for trait t is the ratio of the genetic variance explained by factor m (A_mt) to the total genetic variance (A_t): A_mt / A_t. 
The factorial co-environmentality of a residual factor n for trait t is the ratio of the residual variance explained by factor n (E_nt) to the total residual variance (E_t):  E_nt / E_t.
All standard errors are derived with the Delta method.
</p>


<h3>Value</h3>

<p><code>grmsem.fcoher</code> returns an extended model.out dataframe, fcoher.model.out, with the following columns:
</p>
<table>
<tr><td><code>label</code></td>
<td>
<p>parameter label</p>
</td></tr>
<tr><td><code>estimates</code></td>
<td>
<p>estimated parameters</p>
</td></tr>
<tr><td><code>gradient</code></td>
<td>
<p>gradient</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>SE</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>Z (Wald) of factor loading</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>p (Wald) of factor loading</p>
</td></tr>
<tr><td><code>Vi</code></td>
<td>
<p>squared factor loading, explained phenotypic variation by the factor</p>
</td></tr>
<tr><td><code>Vi.se</code></td>
<td>
<p>SE of squared factor loading</p>
</td></tr>
<tr><td><code>FCOHER</code></td>
<td>
<p>factorial co-heritability</p>
</td></tr>
<tr><td><code>FCOHER.se</code></td>
<td>
<p>SE of factorial co-heritability</p>
</td></tr>
<tr><td><code>FCOHER.Z</code></td>
<td>
<p>Z (wald) of factorial co-heritability</p>
</td></tr>
<tr><td><code>FCOHER.se</code></td>
<td>
<p>p (Wald) of factorial co-heritability</p>
</td></tr>
<tr><td><code>FCOENV</code></td>
<td>
<p>factorial co-environmentality</p>
</td></tr>
<tr><td><code>FCOENV.se</code></td>
<td>
<p>SE of factorial co-environmentality</p>
</td></tr>
<tr><td><code>FCOENV.Z</code></td>
<td>
<p>Z (wald) of factorial co-environmentality</p>
</td></tr>
<tr><td><code>FCOENV.se</code></td>
<td>
<p>p (Wald) of factorial co-environmentality</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#(runtime should be less than one minute)

out &lt;- grmsem.fit(ph.small, G.small, LogL = TRUE, estSE = TRUE)
grmsem.fcoher(out)
</code></pre>

<hr>
<h2 id='grmsem.fit'>grmsem model fitting function</h2><span id='topic+grmsem.fit'></span>

<h3>Description</h3>

<p>This function fits a grmsem model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grmsem.fit(
  ph,
  G,
  A.v.free = NULL,
  E.v.free = NULL,
  A.v.startval = NULL,
  E.v.startval = NULL,
  LogL = FALSE,
  estSE = FALSE,
  cores = 1,
  model = "Cholesky",
  compl.ph = FALSE,
  printest = FALSE,
  cluster = "PSOCK",
  optim = "optim",
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grmsem.fit_+3A_ph">ph</code></td>
<td>
<p>phenotype file as R dataframe, even for single vectors (columns: k phenotypes, rows: ni individuals in same order as G). No default.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_g">G</code></td>
<td>
<p>GRM matrix as provided by the grm.input or grm.bin.input.R function. Use the same order of individuals as in ph. No default.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_a.v.free">A.v.free</code></td>
<td>
<p>vector of free parameters for genetic factor loadings (free:1, not-free:0). Default NULL, all parameters are estimated.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_e.v.free">E.v.free</code></td>
<td>
<p>vector of free parameters for residual factor loadings (free:1, not-free:0). Default NULL, all parameters are estimated.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_a.v.startval">A.v.startval</code></td>
<td>
<p>vector of starting values for genetic factor loadings. Default NULL, all starting values are generated.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_e.v.startval">E.v.startval</code></td>
<td>
<p>vector of starting values for residual factor loadings. Default NULL, all starting values are generated.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_logl">LogL</code></td>
<td>
<p>estimation of the loglikelihood using optim BFGS (TRUE/FALSE). Default FALSE.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_estse">estSE</code></td>
<td>
<p>estimation of standard errors by recalculating the Hessian matrix. Default FALSE.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_cores">cores</code></td>
<td>
<p>number of cores for multi-threaded calculations (numeric). Default 1.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_model">model</code></td>
<td>
<p>grmsem model selection. Options: &quot;Cholesky&quot;, &quot;IP&quot;, &quot;IPC&quot;, &quot;DS&quot;. Default &quot;Cholesky&quot;.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_compl.ph">compl.ph</code></td>
<td>
<p>listwise complete observations across all phenotypes (all NA are excluded). Default FALSE.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_printest">printest</code></td>
<td>
<p>print output of the model.fit function including estimates (printest.txt) that can be used as starting values. Default FALSE.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_cluster">cluster</code></td>
<td>
<p>cluster type. Options: &quot;PSOCK&quot;, &quot;FORK&quot;. Default &quot;PSOCK&quot;.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_optim">optim</code></td>
<td>
<p>optimisation function from stats or optimParallel libraries. Options: &quot;optim&quot;, &quot;optimParallel&quot;. Default &quot;optim&quot;.</p>
</td></tr>
<tr><td><code id="grmsem.fit_+3A_verbose">verbose</code></td>
<td>
<p>additional model fit information: (i) phenotype vector, (ii) n of GRM and corresponding I matrix when data are missing, (iii) Hessian if <code>estSE</code> <code>TRUE</code>. Default FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>grmsem models estimate genetic (A) and residual (E) variance/covariance of quantitative traits (AE model), where E in GRM-based methods can capture both, shared and unique residual influences. 
The estimation of parameters and their SEs is performed with the function <code>grmsem.fit</code>. 
Specifically, the loglikelihood is estimated with <code>stats::optim</code> and the BFGS (Broyden-Fletcher-Goldfarb-Shannon) approach and the variance/covariance matrix of estimated parameters 
with <code>numDeriv::hessian</code>. The statistical significance of estimated parameters is assessed using a Wald test, assuming multivariate normality. 
</p>
<p><code>grmsem.fit</code> allows fitting different models describing the underlying multivariate genetic architecture of quantitative traits, 
as captured by a genetic-relationship-matrix (GRM), using structural equation modelling (SEM) techniques and a maximum likelihood approach. 
The user can fit multiple predefined model structures to the data. A Cholesky decomposition, Independent Pathway, 
and hybrid Independent Pathway/Cholesky models can be fitted by setting the <code>model</code> option to <code>Cholesky</code>, <code>IP</code> or <code>IPC</code>, respectively. 
In addition, the Cholesky model can be re-parametrised as Direct Symmetric model, 
estimating genetic and residual covariances directly, using the <code>model</code> option <code>DS</code>. Each model can be adapted by the user by setting free parameters (<code>A.v.free</code> and <code>E.v.free</code> options) and starting values (<code>A.v.startval</code> and  <code>E.v.startval</code> 
options). 
</p>
<p>Input parameters are returned as <code>model.in</code> list object. Output from the maximum likelihood estimation is also given as 
list <code>model.fit</code> and the fitted grmsem model with estimated parameters and SEs is returned as dataframe <code>model.out</code>. 
The returned <code>grmsem.fit</code> object can be used to estimate genetic and residual covariance and correlations (<code>grmsem.var</code> function), bivariate heritabilities (<code>grmsem.biher</code> function), 
and factorial co-heritabilities and co-environmentalities (<code>grmsem.fcoher</code> function). All estimated parameters of the fitted grmsem model can also be standardised using the function <code>grmsem.stpar</code>.
</p>
<p>Listwise complete observations can be selected with the option <code>compl.ph</code>=<code>TRUE</code>. Otherwise, <code>grmsem.fit</code> fits, like GREML, all available data to the model 
with the default option <code>compl.ph</code> <code>FALSE</code>. Using the option <code>LogL</code>=<code>FALSE</code>, the user can check the model input parameters and formula without a 
maximum likelihood estimation. Using the option <code>estSE</code>=<code>FALSE</code>, the user can carry out a maximum likelihood estimation without the estimation of SEs for 
estimated parameters that require calculating the Hessian. Note that <code>grmsem.fit</code> should preferably be run in parallel, by setting the <code>cores</code> option to the required number of cores.
</p>
<p>When <code>grmsem.fit</code> is called with <code>LogL</code> <code>TRUE</code>, the user will see also the iterations of the <code>stats::optim</code> loglikelihood estimation, which are not included in the exported grmsem.fit list object.
</p>


<h3>Value</h3>

<p><code>grmsem.fit</code> returns a grmsem.fit list object consisting of:
</p>
<table>
<tr><td><code>model.in</code></td>
<td>
<p>list of input parameters</p>
</td></tr>
<tr><td><code>formula</code></td>
<td>
<p>matrix of the model specification</p>
</td></tr>
<tr><td><code>model.fit</code></td>
<td>
<p>list output of the maximum likelihood estimation, if <code>LogL</code> <code>TRUE</code></p>
</td></tr>
<tr><td><code>model.out</code></td>
<td>
<p>dataframe of fitted grmsem model with estimated parameters and SEs, if <code>estSE</code> <code>TRUE</code></p>
</td></tr>
<tr><td><code>VCOV</code></td>
<td>
<p>variance/covariance matrix</p>
</td></tr>
<tr><td><code>k</code></td>
<td>
<p>number of phenotypes</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>total number of observations across all phenotypes</p>
</td></tr>
<tr><td><code>n.obs</code></td>
<td>
<p>number of observations per phenotype</p>
</td></tr>
<tr><td><code>n.ind</code></td>
<td>
<p>number of individuals with at least one phenotype</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>type of grmsem model</p>
</td></tr>
<tr><td><code>ph.nms</code></td>
<td>
<p>vector of phenotype names</p>
</td></tr>
</table>
<p><code>model.in</code> list of input parameters:
</p>
<table>
<tr><td><code>part</code></td>
<td>
<p>a - genetic, e - residual parameters</p>
</td></tr>
<tr><td><code>label</code></td>
<td>
<p>parameter label</p>
</td></tr>
<tr><td><code>value</code></td>
<td>
<p>starting values</p>
</td></tr>
<tr><td><code>freepar</code></td>
<td>
<p>free parameters</p>
</td></tr>
</table>
<p><code>model.fit</code> list output of the maximum likelihood estimation:
</p>
<table>
<tr><td><code>optimisation</code></td>
<td>
<p>output via optim()</p>
</td></tr>
<tr><td><code>estimates</code></td>
<td>
<p>estimated parameters: factor loadings for 'Cholesky', 'IP' and 'IPC' models, but variance components for 'DS'</p>
</td></tr>
<tr><td><code>LL</code></td>
<td>
<p>loglikelihood</p>
</td></tr>
<tr><td><code>calls</code></td>
<td>
<p>optim() calls</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>optim() convergence</p>
</td></tr>
<tr><td><code>message</code></td>
<td>
<p>optim() message</p>
</td></tr>
</table>
<p><code>model.out</code> data.frame of fitted grmsem model:
</p>
<table>
<tr><td><code>label</code></td>
<td>
<p>parameter label</p>
</td></tr>
<tr><td><code>estimates</code></td>
<td>
<p>estimated parameters</p>
</td></tr>
<tr><td><code>gradient</code></td>
<td>
<p>gradient</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>SE</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>Z (Wald)</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>p (Wald)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#Set up a Cholesky model: Model formula and total number of parameters
#ph.small is a trivariate phenotype file for 100 individuals in the same order as G.small
#nrow = 100, ncol = 3
#(runtime should be less than one minute)
out &lt;- grmsem.fit(ph.small, G.small, LogL = FALSE, estSE = FALSE)

#Run a Cholesky model:
out &lt;- grmsem.fit(ph.small, G.small, LogL = TRUE, estSE = TRUE)
</code></pre>

<hr>
<h2 id='grmsem.stpar'>grmsem standardised parameters function</h2><span id='topic+grmsem.stpar'></span>

<h3>Description</h3>

<p>This function estimates standardised parameters for a grmsem.fit object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grmsem.stpar(grmsem.out = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grmsem.stpar_+3A_grmsem.out">grmsem.out</code></td>
<td>
<p>grmsem.fit object as provided by the grmsem.fit function. Default NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>grmsem.stpar</code> standardises grmsem.fit estimates so that derived or estimated A and E variance components will add up to phenotypic unit variance. 
The SEs of standardised parameters are derived using the Jacobian matrix.
</p>


<h3>Value</h3>

<p><code>grmsem.stpar</code> returns a list object consisting of:
</p>
<table>
<tr><td><code>model.in</code></td>
<td>
<p>list of input parameters</p>
</td></tr>
<tr><td><code>stand.model.out</code></td>
<td>
<p>dataframe of fitted grmsem model with standardised parameters and SEs</p>
</td></tr>
<tr><td><code>stVCOV</code></td>
<td>
<p>standardised variance/covariance matrix</p>
</td></tr>
<tr><td><code>k</code></td>
<td>
<p>number of phenotypes</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>total number of observations across all phenotypes</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>type of grmsem model</p>
</td></tr>
<tr><td><code>ph.nms</code></td>
<td>
<p>vector of phenotype names</p>
</td></tr>
</table>
<p><code>model.in</code> list of input parameters:
</p>
<table>
<tr><td><code>part</code></td>
<td>
<p>a - genetic, e - residual parameters</p>
</td></tr>
<tr><td><code>label</code></td>
<td>
<p>parameter label</p>
</td></tr>
<tr><td><code>value</code></td>
<td>
<p>starting values</p>
</td></tr>
<tr><td><code>freepar</code></td>
<td>
<p>free parameters</p>
</td></tr>
</table>
<p><code>stand.model.out</code> data.frame of fitted grmsem model with standardised parameters and SEs:
</p>
<table>
<tr><td><code>label</code></td>
<td>
<p>parameter label</p>
</td></tr>
<tr><td><code>estimates</code></td>
<td>
<p>standardised estimated parameters</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>SE</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>Z (Wald)</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>p (Wald)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#(runtime should be less than one minute)

out &lt;- grmsem.fit(ph.small, G.small, LogL = TRUE, estSE = TRUE)
stout &lt;- grmsem.stpar(out)
print(stout)
</code></pre>

<hr>
<h2 id='grmsem.var'>grmsem variance estimation function</h2><span id='topic+grmsem.var'></span>

<h3>Description</h3>

<p>This function estimates genetic and residual variances, and genetic correlations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grmsem.var(grmsem.out = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grmsem.var_+3A_grmsem.out">grmsem.out</code></td>
<td>
<p>A grmsem.fit or grmsem.stpar object. Default NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>grmsem.var</code> function can be used to estimate genetic and residual covariance and correlations for DS, Cholesky, IP and IPC models, based on 
<code>grmsem.fit</code> or <code>grmsem.stpar</code> objects. For the latter, the diagonal elements of the VA output matrix 
detail the heritabilities. Except for directly estimated variance components using the DS model, all standard errors 
are derived with the Delta method.
</p>


<h3>Value</h3>

<p><code>grmsem.var</code> returns a list object consisting of the following matrices:
</p>
<table>
<tr><td><code>VA</code></td>
<td>
<p>estimated genetic variance</p>
</td></tr>
<tr><td><code>VA.se</code></td>
<td>
<p>standard error of estimated genetic variance</p>
</td></tr>
<tr><td><code>VE</code></td>
<td>
<p>estimated residual variance</p>
</td></tr>
<tr><td><code>VE.se</code></td>
<td>
<p>standard error of estimated residual variance</p>
</td></tr>
<tr><td><code>VP</code></td>
<td>
<p>estimated total phenotypic variance</p>
</td></tr>
<tr><td><code>RG</code></td>
<td>
<p>genetic correlation</p>
</td></tr>
<tr><td><code>RG.se</code></td>
<td>
<p>standard error genetic correlation</p>
</td></tr>
<tr><td><code>RE</code></td>
<td>
<p>residual correlation</p>
</td></tr>
<tr><td><code>RG.se</code></td>
<td>
<p>standard error residual correlation</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#(runtime should be less than one minute)

out &lt;- grmsem.fit(ph.small, G.small, LogL = TRUE, estSE = TRUE)
var.out &lt;- grmsem.var(out)
print(var.out)
</code></pre>

<hr>
<h2 id='ph.large'>Phenotype data: large data set</h2><span id='topic+ph.large'></span>

<h3>Description</h3>

<p>A large quad-variate data set was simulated assuming an underlying Cholesky model, 
with 5000 observations per trait and high polygenicity (5,000 SNPs per genetic factor).
The data set is described in full in the vignette. Genetic and residual
covariances ar assumed to be influenced by four independent genetic factors
(A1, A2, A3 and A4, based on 5000 SNPs each) and four independent residual factors (E1, E2, E3 and E4), 
respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ph.large
</code></pre>


<h3>Format</h3>

<p>A data frame with 5000 rows and 4 columns
</p>

<dl>
<dt>Y1</dt><dd><p>trait one</p>
</dd>
<dt>Y2</dt><dd><p>trait two</p>
</dd>
<dt>Y3</dt><dd><p>trait three</p>
</dd>
<dt>Y4</dt><dd><p>trait four</p>
</dd>
</dl>


<hr>
<h2 id='ph.small'>Phenotype data: small data set</h2><span id='topic+ph.small'></span>

<h3>Description</h3>

<p>A small tri-variate data set was simulated assuming an underlying Cholesky model, 
with 100 observations per trait and low polygenicity (150 SNPs per genetic factor).
The data set is described in full in the vignette. Genetic and residual
covariances are assumed to be influenced by three independent genetic factors 
(A1, A2 and A3, based on a 150 SNPs each) and three independent residual factors (E1, E2 and E3), 
respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ph.small
</code></pre>


<h3>Format</h3>

<p>A data frame with 100 rows and 3 columns
</p>

<dl>
<dt>Y1</dt><dd><p>trait one</p>
</dd>
<dt>Y2</dt><dd><p>trait two</p>
</dd>
<dt>Y3</dt><dd><p>trait three</p>
</dd>
</dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
