<!DOCTYPE html><html><head><title>Help for package grpCox</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {grpCox}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cv.grpCox'>
<p>Cross-validation for grpCox</p></a></li>
<li><a href='#cv.grpCoxOverlap'>
<p>Cross-validation for grpCoxOverlap</p></a></li>
<li><a href='#grpCox'><p>Fit a penalized Cox model.</p></a></li>
<li><a href='#grpCox-package'><p>Penalized Cox Model for High-Dimensional Data with Grouped</p>
Predictors</a></li>
<li><a href='#grpCoxOverlap'><p>Fit a penalized regression path with overlapping grouped covariates.</p></a></li>
<li><a href='#plot.Coef'>
<p>Plots the coefficient paths</p></a></li>
<li><a href='#plot.gCoef'>
<p>Plots the coefficient paths with the same color for the covariates in the same group.</p></a></li>
<li><a href='#plot.llCV'>
<p>Plot the cross-validation curve produced by cv.grpCox or cv.grpCoxOverlap</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Penalized Cox Model for High-Dimensional Data with Grouped
Predictors</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-12-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Xuan Dang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xuan Dang &lt;xuandang11289@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Fit the penalized Cox models with both non-overlapping and overlapping grouped penalties including the group lasso, group smoothly clipped absolute deviation, and group minimax concave penalty. The algorithms combine the MM approach and group-wise descent with some computational tricks including the screening, active set, and warm-start. Different tuning regularization parameter methods are provided.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.11)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen</td>
</tr>
<tr>
<td>Depends:</td>
<td>Matrix (&ge; 1.6-1.1), MASS, colorspace</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-11 15:42:43 UTC; xuandang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-11 16:00:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='cv.grpCox'>
Cross-validation for grpCox
</h2><span id='topic+cv.grpCox'></span>

<h3>Description</h3>

<p>Does k-fold cross-validation for grpCox
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv.grpCox(X, y, g, m, penalty=c("glasso", "gSCAD", "gMCP"), lambda=NULL, 
nlambda=100, rlambda=NULL, gamma=switch(penalty, SCAD = 3.7, 3), 
standardize=TRUE, thresh=1e-3, maxit=1e+4, nfolds=10, foldid=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cv.grpCox_+3A_x">X</code></td>
<td>
<p>The design matrix.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_y">y</code></td>
<td>
<p>The response vector includes time corresponding to failure/censor times, and   status indicating failure (1) or censoring (0).</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_g">g</code></td>
<td>
<p>A vector indicating the group structure of the covariates. It can be unordered groups.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_m">m</code></td>
<td>
<p>Group multipliers. Default is the square root of group size.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_penalty">penalty</code></td>
<td>
<p>The penalty to be applied to the model. It is one of <code>glasso</code>, <code>gSCAD</code>, or <code>gMCP</code>.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_lambda">lambda</code></td>
<td>
<p>A user supplied sequence of <code>lambda</code> values. If it is left unspecified, and the function automatically computes a grid of lambda values.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_nlambda">nlambda</code></td>
<td>
<p>The number of <code>lambda</code> values to use in the regularization path.  Default is 100.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_rlambda">rlambda</code></td>
<td>
<p>Smallest value for lambda, as a fraction of the maximum lambda, the data derived entry value, i.e. the smallest value for which all coefficients are zero. The default depends on the sample size relative to the number of covariates. If sample size&gt;#covariates, the default is 0.001, close to zero.  If sample size&gt;#covariates, the default is 0.05.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_gamma">gamma</code></td>
<td>
<p>Tuning parameter of the group SCAD/MCP penalty.  Default is 3.7 for SCAD and 3 for MCP.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_standardize">standardize</code></td>
<td>
<p>Logical flag for variable standardization prior to fitting the model.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_thresh">thresh</code></td>
<td>
<p>Convergence threshold for one-step coordinate descent. Defaults value is 1E-7.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of passes over the data for all lambda values; default is 1E+5.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_nfolds">nfolds</code></td>
<td>
<p>The number of cross-validation folds. Default is 10.</p>
</td></tr>
<tr><td><code id="cv.grpCox_+3A_foldid">foldid</code></td>
<td>
<p>An optional vector of values between 1 and nfolds identifying what fold each observation is in.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>aBetaSTD</code></td>
<td>
<p>A standardized coefficient matrix whose columns correspond to nlambda values of lambda.</p>
</td></tr>
<tr><td><code>aBetaO</code></td>
<td>
<p>A coefficient matrix (without standardization) whose columns correspond to nlambda values of lambda.</p>
</td></tr>
<tr><td><code>mBetaSTD</code></td>
<td>
<p>The coefficient in standardized form gives maximum log-likelihood value using the first cross-validation method.</p>
</td></tr>
<tr><td><code>mBetaO</code></td>
<td>
<p>The coefficient in original form gives maximum log-likelihood value using the first cross-validation method.</p>
</td></tr>
<tr><td><code>pBetaSTD</code></td>
<td>
<p>The coefficient in standardized form gives maximum log-likelihood value using the penalized cross-validation method.</p>
</td></tr>
<tr><td><code>pBetaO</code></td>
<td>
<p>The coefficient in original formgives maximum log-likelihood value using the penalized cross-validation method.</p>
</td></tr>
<tr><td><code>fit</code></td>
<td>
<p>A matrix includes lambda value, the mean cross-validation error.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>The lambda values used.</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>A vector indicating the group structure of the covariates.</p>
</td></tr>
<tr><td><code>cvmax</code></td>
<td>
<p>The maximum value of log likelihood.</p>
</td></tr>
<tr><td><code>lambda.max</code></td>
<td>
<p>The value of lambda corresponds to the maximum value of log likelihood using the first cross-validation method.</p>
</td></tr>
<tr><td><code>lambda.pcvl</code></td>
<td>
<p>The value of lambda corresponds to the maximum value of log likelihood using the penalized cross-validation method.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xuan Dang &lt;<a href="mailto:xuandang11289@gmail.com">xuandang11289@gmail.com</a>&gt;
</p>


<h3>References</h3>

<p>Verweij PJ, Houwelingen HC. Cross-validation in survival analysis. Statistics in Medicine 1993; 12(24): 385-395.
</p>
<p>Ternes N, Rotolo F, Michiels S. Empirical extensions of the lasso penalty to reduce the false discovery rate in highdimensional Cox regression models. Statistics in Medicine 2016; 35(15): 2561-73.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(200)
N &lt;- 50
p &lt;- 9
x &lt;- matrix(rnorm(N * p), nrow = N)
beta &lt;- c(.65,.65,0,0,.65,.65,0,.65,0)
hx &lt;- exp(x %*% beta) 
ty &lt;- rexp(N,hx) 
tcens &lt;- 1 - rbinom(n=N, prob = 0.2, size = 1)
y &lt;- data.frame(illt=ty, ills=tcens)
names(y) &lt;- c("time", "status")

g &lt;- c(1,1,2,2,3,3,2,3,2)
m &lt;- c(sqrt(2),sqrt(4),sqrt(3))

cvfit &lt;- cv.grpCox(x,y,g,m,penalty="glasso")
plot.llCV(cvfit)
plot.gCoef(cvfit$aBetaO, cvfit$g, cvfit$lambda)
</code></pre>

<hr>
<h2 id='cv.grpCoxOverlap'>
Cross-validation for grpCoxOverlap
</h2><span id='topic+cv.grpCoxOverlap'></span>

<h3>Description</h3>

<p>Does k-fold cross-validation for grpCoxOverlap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv.grpCoxOverlap(X0, y, group, penalty=c("glasso", "gSCAD", "gMCP"), 
lambda=NULL, nlambda=100, rlambda=NULL,gamma=switch(penalty, SCAD = 3.7, 3), 
standardize=TRUE, thresh=1e-3, maxit=1e+4, nfolds=10, foldid=NULL, 
returnLatent=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cv.grpCoxOverlap_+3A_x0">X0</code></td>
<td>
<p>The design matrix.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_y">y</code></td>
<td>
<p>The response vector includes time corresponding to failure/censor times, and   status indicating failure (1) or censoring (0).</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_group">group</code></td>
<td>
<p>A list of groups, each includes indices of covariates in the group.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_penalty">penalty</code></td>
<td>
<p>The penalty to be applied to the model. It is one of <code>glasso</code>, <code>gSCAD</code>, or <code>gMCP</code>.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_lambda">lambda</code></td>
<td>
<p>A user supplied sequence of <code>lambda</code> values. If it is left unspecified, and the function automatically computes a grid of lambda values.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_nlambda">nlambda</code></td>
<td>
<p>The number of <code>lambda</code> values to use in the regularization path.  Default is 100.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_rlambda">rlambda</code></td>
<td>
<p>Smallest value for lambda, as a fraction of the maximum lambda, the data derived entry value, i.e. the smallest value for which all coefficients are zero. The default depends on the sample size relative to the number of covariates. If sample size&gt;#covariates, the default is 0.001, close to zero.  If sample size&gt;#covariates, the default is 0.05.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_gamma">gamma</code></td>
<td>
<p>Tuning parameter of the group SCAD/MCP penalty. Default is 3.7 for SCAD and 3 for MCP.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_standardize">standardize</code></td>
<td>
<p>Logical flag for variable standardization prior to fitting the model.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_thresh">thresh</code></td>
<td>
<p>Convergence threshold for one-step coordinate descent. Defaults value is 1E-7.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of passes over the data for all lambda values; default is 1E+5.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_nfolds">nfolds</code></td>
<td>
<p>The number of cross-validation folds. Default is 10.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_foldid">foldid</code></td>
<td>
<p>An optional vector of values between 1 and nfolds identifying what fold each observation is in.</p>
</td></tr>
<tr><td><code id="cv.grpCoxOverlap_+3A_returnlatent">returnLatent</code></td>
<td>
<p>Return the coefficient matrix in latent space. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>aBetaLatent</code></td>
<td>
<p>A coefficient matrix whose columns correspond to nlambda values of lambda in latent space.</p>
</td></tr>
<tr><td><code>aBetaOri</code></td>
<td>
<p>A coefficient matrix whose columns correspond to nlambda values of lambda in original space.</p>
</td></tr>
<tr><td><code>mBetaLatent</code></td>
<td>
<p>The coefficient in latent space gives maximum log-likelihood value using the first cross-validation method.</p>
</td></tr>
<tr><td><code>mBetaOri</code></td>
<td>
<p>The coefficient in original space gives maximum log-likelihood value using the first cross-validation method.</p>
</td></tr>
<tr><td><code>pBetaLatent</code></td>
<td>
<p>The coefficient in latent space gives maximum log-likelihood value using the penalized cross-validation method.</p>
</td></tr>
<tr><td><code>pBetaOri</code></td>
<td>
<p>The coefficient in original space gives maximum log-likelihood value using the penalized cross-validation method.</p>
</td></tr>
<tr><td><code>fit</code></td>
<td>
<p>A matrix includes lambda value, the mean cross-validation error.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>The lambda values used.</p>
</td></tr>
<tr><td><code>group</code></td>
<td>
<p>A list of groups, each includes indices of covariates in the group.</p>
</td></tr>
<tr><td><code>glatent</code></td>
<td>
<p>A vector indicating the group structure of the covariates in latent space.</p>
</td></tr>
<tr><td><code>cvmax</code></td>
<td>
<p>The maximum value of log likelihood.</p>
</td></tr>
<tr><td><code>lambda.max</code></td>
<td>
<p>The value of lambda corresponds to the maximum value of log likelihood using the first cross-validation method.</p>
</td></tr>
<tr><td><code>lambda.pcvl</code></td>
<td>
<p>The value of lambda corresponds to the maximum value of log likelihood using the penalized cross-validation method.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xuan Dang &lt;<a href="mailto:xuandang11289@gmail.com">xuandang11289@gmail.com</a>&gt;
</p>


<h3>References</h3>

<p>Verweij PJ, Houwelingen HC. Cross-validation in survival analysis. Statistics in Medicine 1993; 12(24): 385-395.
</p>
<p>Ternes N, Rotolo F, Michiels S. Empirical extensions of the lasso penalty to reduce the false discovery rate in highdimensional Cox regression models. Statistics in Medicine 2016; 35(15): 2561-73.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100001)
N &lt;- 50
p &lt;- 6
times &lt;- 1:p
rho &lt;- 0.5
H &lt;- abs(outer(times, times, "-"))
C &lt;- 1 * rho^H
C[cbind(1:p, 1:p)] &lt;- C[cbind(1:p, 1:p)] 
sigma &lt;- matrix(C,p,p)
mu &lt;- rep(0,p)
x &lt;- mvrnorm(n=N, mu, sigma)

beta &lt;- c(0, .8, 1, 2, 1, 0)
hx &lt;- exp(x %*% beta) 
ty &lt;- rexp(N,hx) 
tcens &lt;- 1 - rbinom(n=N, prob = 0.2, size = 1)
y &lt;- data.frame(illt=ty, ills=tcens)
names(y) &lt;- c("time", "status")

group &lt;- list(g1 = c(1,2,3,4), g2 = c(1,2,6), g3 = c(2,3), 
              g4 = c(4,5), g5 = c(5))
cvfit &lt;- cv.grpCoxOverlap(x, y, group, penalty="glasso", nlambda=50)
plot.llCV(cvfit)
</code></pre>

<hr>
<h2 id='grpCox'>Fit a penalized Cox model.</h2><span id='topic+grpCox'></span>

<h3>Description</h3>

<p>Fit the regularization paths for Cox models with grouped covariates.</p>


<h3>Usage</h3>

<pre><code class='language-R'>grpCox(X, y, g, m, penalty=c("glasso", "gSCAD", "gMCP"), lambda=NULL, 
nlambda=100, rlambda=NULL, gamma=switch(penalty, gSCAD = 3.7, 3), 
standardize=TRUE, thresh=1e-3, maxit=1e+4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grpCox_+3A_x">X</code></td>
<td>
<p>The design matrix.</p>
</td></tr>
<tr><td><code id="grpCox_+3A_y">y</code></td>
<td>
<p>The response vector includes time corresponding to failure/censor times, and   status indicating failure (1) or censoring (0).</p>
</td></tr>
<tr><td><code id="grpCox_+3A_g">g</code></td>
<td>
<p>A vector indicating the group structure of the covariates. It can be unordered groups.</p>
</td></tr>
<tr><td><code id="grpCox_+3A_m">m</code></td>
<td>
<p>Group multipliers. Default is the square root of group size.</p>
</td></tr>
<tr><td><code id="grpCox_+3A_penalty">penalty</code></td>
<td>
<p>The penalty to be applied to the model. It is one of <code>glasso</code>, <code>gSCAD</code>, or <code>gMCP</code>.</p>
</td></tr>
<tr><td><code id="grpCox_+3A_lambda">lambda</code></td>
<td>
<p>A user supplied sequence of <code>lambda</code> values. If it is left unspecified, and the function automatically computes a grid of lambda values.</p>
</td></tr>
<tr><td><code id="grpCox_+3A_nlambda">nlambda</code></td>
<td>
<p>The number of <code>lambda</code> values to use in the regularization path.  Default is 100.</p>
</td></tr>
<tr><td><code id="grpCox_+3A_rlambda">rlambda</code></td>
<td>
<p>Smallest value for lambda, as a fraction of the maximum lambda, the data derived entry value, i.e. the smallest value for which all coefficients are zero. The default depends on the sample size relative to the number of covariates. If sample size&gt;#covariates, the default is 0.001, close to zero.  If sample size&gt;#covariates, the default is 0.05.</p>
</td></tr>
<tr><td><code id="grpCox_+3A_gamma">gamma</code></td>
<td>
<p>Tuning parameter of the group SCAD/MCP penalty.  Default is 3.7 for SCAD and 3 for MCP.</p>
</td></tr>
<tr><td><code id="grpCox_+3A_standardize">standardize</code></td>
<td>
<p>Logical flag for variable standardization prior to fitting the model.</p>
</td></tr>
<tr><td><code id="grpCox_+3A_thresh">thresh</code></td>
<td>
<p>Convergence threshold for one-step coordinate descent. Defaults value is 1E-7.</p>
</td></tr>
<tr><td><code id="grpCox_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of passes over the data for all lambda values; default is 1E+5.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The the group SCAD (<code>gSCAD</code>) and group MCP (<code>gMCP</code>) formulations have been presented in Wang et. al 2007, Huang et. al 2012.
</p>


<h3>Value</h3>

<table>
<tr><td><code>aBetaSTD</code></td>
<td>
<p>A standardized coefficient matrix whose columns correspond to nlambda values of lambda.</p>
</td></tr>
<tr><td><code>aBetaO</code></td>
<td>
<p>A coefficient matrix (without standardization) whose columns correspond to nlambda values of lambda.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>The lambda values used.</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>The log likelihood values.</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>A vector indicating the group structure of the covariates.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xuan Dang &lt;<a href="mailto:xuandang11289@gmail.com">xuandang11289@gmail.com</a>&gt;
</p>


<h3>References</h3>

<p>Wang, L., Chen, G., and Li, H. Group SCAD regression analysis for microarray time course gene expression
data. Bioinformatics 23.12 (2007), pp. 1486-1494.
</p>
<p>Huang, J., Breheny, P., and Ma, S. A selective review of group selection in high-dimensional models. Statistical Science 27.4 (2012), pp. 481-499.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(200)
N &lt;- 50
p &lt;- 9
x &lt;- matrix(rnorm(N * p), nrow = N)
beta &lt;- c(.65,.65,0,0,.65,.65,0,.65,0)
hx &lt;- exp(x %*% beta) 
ty &lt;- rexp(N,hx) 
tcens &lt;- 1 - rbinom(n=N, prob = 0.2, size = 1)
y &lt;- data.frame(illt=ty, ills=tcens)
names(y) &lt;- c("time", "status")

g &lt;- c(1,1,2,2,3,3,2,3,2)
m &lt;- c(sqrt(2),sqrt(4),sqrt(3))

fit &lt;- grpCox(x,y,g,m,penalty="glasso")
plot.gCoef(fit$aBetaO, fit$g, fit$lambda)
</code></pre>

<hr>
<h2 id='grpCox-package'>Penalized Cox Model for High-Dimensional Data with Grouped
Predictors</h2><span id='topic+grpCox-package'></span>

<h3>Description</h3>

<p>Fit the penalized Cox models with both non-overlapping and overlapping grouped penalties including the group lasso, group smoothly clipped absolute deviation, and group minimax concave penalty. The algorithms combine the MM approach and group-wise descent with some computational tricks including the screening, active set, and warm-start. Different tuning regularization parameter methods are provided.</p>


<h3>Package Content</h3>


<p>Index of help topics:
</p>
<pre>
cv.grpCox               Cross-validation for grpCox
cv.grpCoxOverlap        Cross-validation for grpCoxOverlap
grpCox                  Fit a penalized Cox model.
grpCox-package          Penalized Cox Model for High-Dimensional Data
                        with Grouped Predictors
grpCoxOverlap           Fit a penalized regression path with
                        overlapping grouped covariates.
plot.Coef               Plots the coefficient paths
plot.gCoef              Plots the coefficient paths with the same color
                        for the covariates in the same group.
plot.llCV               Plot the cross-validation curve produced by
                        cv.grpCox or cv.grpCoxOverlap
</pre>

<h3>Maintainer</h3>

<p>Xuan Dang &lt;xuandang11289@gmail.com&gt;</p>


<h3>Author(s)</h3>

<p>Xuan Dang</p>

<hr>
<h2 id='grpCoxOverlap'>Fit a penalized regression path with overlapping grouped covariates.</h2><span id='topic+grpCoxOverlap'></span>

<h3>Description</h3>

<p>Fit the regularization paths for Cox's models with overlapping grouped covariates.</p>


<h3>Usage</h3>

<pre><code class='language-R'>grpCoxOverlap(X0, y, group, penalty=c("glasso", "gSCAD", "gMCP"), 
lambda=NULL, nlambda=100, rlambda=NULL, gamma=switch(penalty, gSCAD = 3.7, 3),
standardize = TRUE, thresh=1e-3, maxit=1e+4, returnLatent=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grpCoxOverlap_+3A_x0">X0</code></td>
<td>
<p>The design matrix.</p>
</td></tr>
<tr><td><code id="grpCoxOverlap_+3A_y">y</code></td>
<td>
<p>The response vector includes time corresponding to failure/censor times, and   status indicating failure (1) or censoring (0).</p>
</td></tr>
<tr><td><code id="grpCoxOverlap_+3A_group">group</code></td>
<td>
<p>A list of groups, each includes indices of covariates in the group.</p>
</td></tr>
<tr><td><code id="grpCoxOverlap_+3A_penalty">penalty</code></td>
<td>
<p>The penalty to be applied to the model. It is one of <code>glasso</code>, <code>gSCAD</code>, or <code>gMCP</code>.</p>
</td></tr>
<tr><td><code id="grpCoxOverlap_+3A_lambda">lambda</code></td>
<td>
<p>A user supplied sequence of <code>lambda</code> values. If it is left unspecified, and the function automatically computes a grid of lambda values.</p>
</td></tr>
<tr><td><code id="grpCoxOverlap_+3A_nlambda">nlambda</code></td>
<td>
<p>The number of <code>lambda</code> values to use in the regularization path.  Default is 100.</p>
</td></tr>
<tr><td><code id="grpCoxOverlap_+3A_rlambda">rlambda</code></td>
<td>
<p>Smallest value for lambda, as a fraction of the maximum lambda, the data derived entry value, i.e. the smallest value for which all coefficients are zero. The default depends on the sample size relative to the number of covariates. If sample size&gt;#covariates, the default is 0.001, close to zero.  If sample size&gt;#covariates, the default is 0.05.</p>
</td></tr>
<tr><td><code id="grpCoxOverlap_+3A_gamma">gamma</code></td>
<td>
<p>Tuning parameter of the group SCAD/MCP penalty. Default is 3.7 for SCAD and 3 for MCP.</p>
</td></tr>
<tr><td><code id="grpCoxOverlap_+3A_standardize">standardize</code></td>
<td>
<p>Logical flag for variable standardization prior to fitting the model.</p>
</td></tr>
<tr><td><code id="grpCoxOverlap_+3A_thresh">thresh</code></td>
<td>
<p>Convergence threshold for one-step coordinate descent. Defaults value is 1E-7.</p>
</td></tr>
<tr><td><code id="grpCoxOverlap_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of passes over the data for all lambda values; default is 1E+5.</p>
</td></tr>
<tr><td><code id="grpCoxOverlap_+3A_returnlatent">returnLatent</code></td>
<td>
<p>Return the coefficient matrix in latent space. Default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The the group SCAD (<code>gSCAD</code>) and group MCP (<code>gMCP</code>) formulations have been presented in Wang et. al 2007, Huang et. al 2012.
</p>
<p>The method based on the latent group approach (Jacob et al. 2009, Obozinski et al. 2011.)
</p>


<h3>Value</h3>

<table>
<tr><td><code>aBetaLatent</code></td>
<td>
<p>A coefficient matrix whose columns correspond to nlambda values of lambda in latent space.</p>
</td></tr>
<tr><td><code>aBetaOri</code></td>
<td>
<p>A coefficient matrix whose columns correspond to nlambda values of lambda in original space.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>The lambda values used.</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>The log likelihood values.</p>
</td></tr>
<tr><td><code>group</code></td>
<td>
<p>A list of groups, each includes indices of covariates in the group.</p>
</td></tr>
<tr><td><code>glatent</code></td>
<td>
<p>A vector indicating the group structure of the covariates in latent space.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xuan Dang &lt;<a href="mailto:xuandang11289@gmail.com">xuandang11289@gmail.com</a>&gt;
</p>


<h3>References</h3>

<p>Wang, L., Chen, G., and Li, H. Group SCAD regression analysis for microarray time course gene expression
data. Bioinformatics 23.12 (2007), pp. 1486-1494.
</p>
<p>Huang, J., Breheny, P., and Ma, S. A selective review of group selection in high-dimensional models.&quot; Statistical Science 27.4 (2012), pp. 481-499.
</p>
<p>Jacob, L., Obozinski, G., and Vert, J. P. (2009, June). Group lasso with overlap and graph lasso. In Proceedings of the 26th annual international conference on machine learning, ACM: 433-440.
</p>
<p>Obozinski, G., Jacob, L., and Vert, J. P. (2011). Group lasso with overlaps: the latent group lasso approach.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100001)
N &lt;- 50
p &lt;- 6
times &lt;- 1:p
rho &lt;- 0.5
H &lt;- abs(outer(times, times, "-"))
C &lt;- 1 * rho^H
C[cbind(1:p, 1:p)] &lt;- C[cbind(1:p, 1:p)] 
sigma &lt;- matrix(C,p,p)
mu &lt;- rep(0,p)
x &lt;- mvrnorm(n=N, mu, sigma)

beta &lt;- c(0, .8, 1, 2, 1, 0)
hx &lt;- exp(x %*% beta) 
ty &lt;- rexp(N,hx) 
tcens &lt;- 1 - rbinom(n=N, prob = 0.2, size = 1)
y &lt;- data.frame(illt=ty, ills=tcens)
names(y) &lt;- c("time", "status")

group &lt;- list(g1 = c(1,2,3,4), g2 = c(1,2,6), g3 = c(2,3), g4 = c(4,5), g5 = c(5))
fit &lt;- grpCoxOverlap(x, y, group, penalty="glasso", nlambda=50)
# plot the coefficient values in latent space
plot.gCoef(fit$aBetaLatent, fit$glatent, fit$lambda)
# plot the coefficient values in original space
plot.Coef(fit$aBetaOri, fit$lambda)
</code></pre>

<hr>
<h2 id='plot.Coef'>
Plots the coefficient paths
</h2><span id='topic+plot.Coef'></span>

<h3>Description</h3>

<p>Plots the coefficient values as a function of the lambda values used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Coef'
plot(x, lambda, label=TRUE,xlab="log(Lambda)", 
ylab="Coefficients",title=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Coef_+3A_x">x</code></td>
<td>
<p>A matrix of coefficients.</p>
</td></tr>
<tr><td><code id="plot.Coef_+3A_lambda">lambda</code></td>
<td>
<p>The lambda values used.</p>
</td></tr>
<tr><td><code id="plot.Coef_+3A_label">label</code></td>
<td>
<p>The indices of covariates. Default is TRUE.</p>
</td></tr>
<tr><td><code id="plot.Coef_+3A_xlab">xlab</code></td>
<td>
<p>The name of the x-axis.</p>
</td></tr>
<tr><td><code id="plot.Coef_+3A_ylab">ylab</code></td>
<td>
<p>The name of the y-axis.</p>
</td></tr>
<tr><td><code id="plot.Coef_+3A_title">title</code></td>
<td>
<p>The title of the plot.</p>
</td></tr>
<tr><td><code id="plot.Coef_+3A_...">...</code></td>
<td>
<p>further arguments to plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A plot is produced, and nothing is returned.
</p>


<h3>Value</h3>

<p>No return value.</p>


<h3>Author(s)</h3>

<p>Xuan Dang &lt;<a href="mailto:xuandang11289@gmail.com">xuandang11289@gmail.com</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100001)
N &lt;- 50
p &lt;- 6
times &lt;- 1:p
rho &lt;- 0.5
H &lt;- abs(outer(times, times, "-"))
C &lt;- 1 * rho^H
C[cbind(1:p, 1:p)] &lt;- C[cbind(1:p, 1:p)] 
sigma &lt;- matrix(C,p,p)
mu &lt;- rep(0,p)
x &lt;- mvrnorm(n=N, mu, sigma)

beta &lt;- c(0, .8, 1, 2, 1, 0)
hx &lt;- exp(x %*% beta) 
ty &lt;- rexp(N,hx) 
tcens &lt;- 1 - rbinom(n=N, prob = 0.2, size = 1)
y &lt;- data.frame(illt=ty, ills=tcens)
names(y) &lt;- c("time", "status")

group &lt;- list(g1 = c(1,2,3,4), g2 = c(1,2,6), g3 = c(2,3), g4 = c(4,5), g5 = c(5))
fit &lt;- grpCoxOverlap(x, y, group, penalty="glasso", nlambda=50)
# plot the coefficient values in latent space
plot.gCoef(fit$aBetaLatent, fit$glatent, fit$lambda)
# plot the coefficient values in original space
plot.Coef(fit$aBetaOri, fit$lambda)
</code></pre>

<hr>
<h2 id='plot.gCoef'>
Plots the coefficient paths with the same color for the covariates in the same group.
</h2><span id='topic+plot.gCoef'></span>

<h3>Description</h3>

<p>Plots the coefficient values as a function of the lambda values used. The covariates in the same group have the same color.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gCoef'
plot(x,g,lambda,label=TRUE,xlab="log(Lambda)",
ylab="Coefficients", title=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.gCoef_+3A_x">x</code></td>
<td>
<p>A matrix of coefficients.</p>
</td></tr>
<tr><td><code id="plot.gCoef_+3A_g">g</code></td>
<td>
<p>A vector indicating the group structure of the covariates.</p>
</td></tr>
<tr><td><code id="plot.gCoef_+3A_lambda">lambda</code></td>
<td>
<p>The lambda values used.</p>
</td></tr>
<tr><td><code id="plot.gCoef_+3A_label">label</code></td>
<td>
<p>The indices of covariates. Default is TRUE.</p>
</td></tr>
<tr><td><code id="plot.gCoef_+3A_xlab">xlab</code></td>
<td>
<p>The name of the x-axis.</p>
</td></tr>
<tr><td><code id="plot.gCoef_+3A_ylab">ylab</code></td>
<td>
<p>The name of the y-axis.</p>
</td></tr>
<tr><td><code id="plot.gCoef_+3A_title">title</code></td>
<td>
<p>The title of the plot.</p>
</td></tr>
<tr><td><code id="plot.gCoef_+3A_...">...</code></td>
<td>
<p>further arguments to plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A plot is produced, and nothing is returned.
</p>


<h3>Value</h3>

<p>No return value.</p>


<h3>Author(s)</h3>

<p>Xuan Dang &lt;<a href="mailto:xuandang11289@gmail.com">xuandang11289@gmail.com</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100001)
N &lt;- 50
p &lt;- 6
times &lt;- 1:p
rho &lt;- 0.5
H &lt;- abs(outer(times, times, "-"))
C &lt;- 1 * rho^H
C[cbind(1:p, 1:p)] &lt;- C[cbind(1:p, 1:p)] 
sigma &lt;- matrix(C,p,p)
mu &lt;- rep(0,p)
x &lt;- mvrnorm(n=N, mu, sigma)

beta &lt;- c(0, .8, 1, 2, 1, 0)
hx &lt;- exp(x %*% beta) 
ty &lt;- rexp(N,hx) 
tcens &lt;- 1 - rbinom(n=N, prob = 0.2, size = 1)
y &lt;- data.frame(illt=ty, ills=tcens)
names(y) &lt;- c("time", "status")

group &lt;- list(g1 = c(1,2,3,4), g2 = c(1,2,6), g3 = c(2,3), g4 = c(4,5), g5 = c(5))
fit &lt;- grpCoxOverlap(x, y, group, penalty="glasso", nlambda=50)
# plot the coefficient values in latent space
plot.gCoef(fit$aBetaLatent, fit$glatent, fit$lambda)
# plot the coefficient values in original space
plot.Coef(fit$aBetaOri, fit$lambda)
</code></pre>

<hr>
<h2 id='plot.llCV'>
Plot the cross-validation curve produced by cv.grpCox or cv.grpCoxOverlap
</h2><span id='topic+plot.llCV'></span>

<h3>Description</h3>

<p>Plots the cross-validation curve, and upper and lower standard deviation curves, as a function of the lambda values used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'llCV'
plot(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.llCV_+3A_x">x</code></td>
<td>
<p>fitted cv.grpCox or cv.grpCoxOverlap object</p>
</td></tr>
<tr><td><code id="plot.llCV_+3A_...">...</code></td>
<td>
<p>further arguments to plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A plot is produced, and nothing is returned.
</p>


<h3>Value</h3>

<p>No return value.</p>


<h3>Author(s)</h3>

<p>Xuan Dang &lt;<a href="mailto:xuandang11289@gmail.com">xuandang11289@gmail.com</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(200)
N &lt;- 50
p &lt;- 9
x &lt;- matrix(rnorm(N * p), nrow = N)
beta &lt;- c(.65,.65,0,0,.65,.65,0,.65,0)
hx &lt;- exp(x %*% beta) 
ty &lt;- rexp(N,hx) 
tcens &lt;- 1 - rbinom(n=N, prob = 0.2, size = 1)
y &lt;- data.frame(illt=ty, ills=tcens)
names(y) &lt;- c("time", "status")

g &lt;- c(1,1,2,2,3,3,2,3,2)
m &lt;- c(sqrt(2),sqrt(4),sqrt(3))

cvfit &lt;- cv.grpCox(x,y,g,m,penalty="glasso")
plot.llCV(cvfit)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
