<!DOCTYPE html><html><head><title>Help for package GRSxE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GRSxE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GRSxE'><p>Testing gene-environment interactions</p></a></li>
<li><a href='#GxE'><p>Testing individual gene-environment interactions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Testing Gene-Environment Interactions Through Genetic Risk
Scores</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Statistical testing procedures for detecting
  GxE (gene-environment) interactions. The main focus lies on
  GRSxE interaction tests that aim at detecting GxE interactions
  through GRS (genetic risk scores). Moreover, a novel testing
  procedure based on bagging and OOB (out-of-bag) predictions is
  implemented for incorporating all available observations at
  both GRS construction and GxE testing (Lau et al., 2023,
  &lt;<a href="https://doi.org/10.1038%2Fs41598-023-28172-4">doi:10.1038/s41598-023-28172-4</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>glmnet, ranger, stats, utils</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-30 11:08:50 UTC; Michael</td>
</tr>
<tr>
<td>Author:</td>
<td>Michael Lau <a href="https://orcid.org/0000-0002-5327-8351"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Lau &lt;michael.lau@hhu.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-30 14:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='GRSxE'>Testing gene-environment interactions</h2><span id='topic+GRSxE'></span>

<h3>Description</h3>

<p>Fitting and evaluating GRS (genetic risk scores) for testing the
presence of GxE (gene-environment) interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GRSxE(
  X,
  y,
  E,
  C = NULL,
  test.type = "bagging",
  B = 500,
  replace = TRUE,
  subsample = ifelse(replace, 1, 0.632),
  test.ind = sample(nrow(X), floor(nrow(X)/2)),
  grs.type = "rf",
  grs.args = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GRSxE_+3A_x">X</code></td>
<td>
<p>Matrix or data frame of genetic variables such as SNPs usually
coded as 0-1-2.</p>
</td></tr>
<tr><td><code id="GRSxE_+3A_y">y</code></td>
<td>
<p>Numeric vector of the outcome/phenotype. Binary outcomes such
as a disease status should be coded as 0-1 (control-case).</p>
</td></tr>
<tr><td><code id="GRSxE_+3A_e">E</code></td>
<td>
<p>Numeric vector of the environmental exposure.</p>
</td></tr>
<tr><td><code id="GRSxE_+3A_c">C</code></td>
<td>
<p>Optional data frame containing potentially confounding
variables to be adjusted for.</p>
</td></tr>
<tr><td><code id="GRSxE_+3A_test.type">test.type</code></td>
<td>
<p>Testing type. The standard setting is <code>"bagging"</code>,
which employs its OOB (out-of-bag) prediction mechanism such that the full
data can be used for both training the GRS and testing the GxE interaction.
Alternatively, this can be set to <code>"holdout"</code>, which requires
splitting the available data into a training data set and test data set.
For that, <code>test.ind</code> needs to be set to the data indices used for
testing.</p>
</td></tr>
<tr><td><code id="GRSxE_+3A_b">B</code></td>
<td>
<p>The number of bagging iterations if <code>test.type = "bagging"</code> is
used. Also used as the number of trees grown in the random forest if
<code>grs.type = "rf"</code> is set.</p>
</td></tr>
<tr><td><code id="GRSxE_+3A_replace">replace</code></td>
<td>
<p>Should sampling with or without replacement be performed?
Only used if <code>test.type = "bagging"</code> is set.</p>
</td></tr>
<tr><td><code id="GRSxE_+3A_subsample">subsample</code></td>
<td>
<p>Subsample fraction if <code>test.type = "bagging"</code> is used.</p>
</td></tr>
<tr><td><code id="GRSxE_+3A_test.ind">test.ind</code></td>
<td>
<p>Vector of indices in the supplied data for testing the GxE
interaction. Only used if <code>test.type = "holdout"</code> is set.
The standard setting corresponds to a random 50:50 training-test split.</p>
</td></tr>
<tr><td><code id="GRSxE_+3A_grs.type">grs.type</code></td>
<td>
<p>Type of GRS to be constructed. Either <code>"rf"</code> for a
random forest or <code>"elnet"</code> for an elastic net.</p>
</td></tr>
<tr><td><code id="GRSxE_+3A_grs.args">grs.args</code></td>
<td>
<p>Optional list of arguments passed to the GRS fitting
procedure.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The GRS is usually constructed through random forests for taking gene-gene
interactions into account and using its OOB (out-of-bag) prediction
mechanism. Alternatively, a classical GRS construction approach can be
employed by fitting an elastic net.
Bagging can also be applied to fit multiple elastic net models to also be
able to perform OOB predictions.
</p>
<p>The advantage of OOB predictions is that they allow the GRS model to be
constructed on the full available data, while performing unbiased
predictions also on the full available data.
Thus, both the GRS construction and the GxE interaction testing can utilize
all observations.
</p>
<p>If desired, sampling can be performed without replacement in contrast to
the classical bagging approach that utilizes bootstrap sampling.
</p>
<p>Potentially confounding variables can also be supplied that will then be
adjusted for in the GxE interaction testing.
</p>
<p>This function uses a GLM (generalized linear model) for modelling the
marginal genetic effect, marginal environmental effect, the GRSxE interaction
effect, and potential confounding effects.
The fitted GLM is returned, which can be, e.g., inspected via
<code>summary(...)</code> to retrieve the Wald test p-values for the individual
terms. The p-value corresponding to the <code>G:E</code> term is the p-value
for testing the presence of a GRSxE interaction.
</p>


<h3>Value</h3>

<p>An object of class <code>glm</code> is returned, in which <code>G:E</code>
describes the GRSxE term.
</p>


<h3>References</h3>


<ul>
<li><p> Lau, M., Kress, S., Schikowski, T. &amp; Schwender, H. (2023).
Efficient gene&ndash;environment interaction testing through
bootstrap aggregating. Scientific Reports 13:937.
doi: <a href="https://doi.org/10.1038/s41598-023-28172-4">10.1038/s41598-023-28172-4</a>
</p>
</li>
<li><p> Lau, M., Wigmann C., Kress S., Schikowski, T. &amp; Schwender, H. (2022).
Evaluation of tree-based statistical learning methods for constructing
genetic risk scores. BMC Bioinformatics 23:97.
doi: <a href="https://doi.org/10.1186/s12859-022-04634-w">10.1186/s12859-022-04634-w</a>
</p>
</li>
<li><p> Breiman, L. (1996).
Bagging predictors. Machine Learning 24:123–140.
doi: <a href="https://doi.org/10.1007/BF00058655">10.1007/BF00058655</a>
</p>
</li>
<li><p> Breiman, L. (2001).
Random Forests. Machine Learning 45:5–32.
doi: <a href="https://doi.org/10.1023/A:1010933404324">10.1023/A:1010933404324</a>
</p>
</li>
<li><p> Friedman J., Hastie T. &amp; Tibshirani R. (2010).
Regularization Paths for Generalized Linear Models via Coordinate Descent.
Journal of Statistical Software 33(1):1–22.
doi: <a href="https://doi.org/10.18637/jss.v033.i01">10.18637/jss.v033.i01</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Generate toy data
set.seed(101299)
maf &lt;- 0.25
n.snps &lt;- 10
N &lt;- 500
X &lt;- matrix(sample(0:2, n.snps * N, replace = TRUE,
                   prob = c((1-maf)^2, 1-(1-maf)^2-maf^2, maf^2)),
            ncol = n.snps)
colnames(X) &lt;- paste("SNP", 1:n.snps, sep="")
E &lt;- rnorm(N, 20, 10)
E[E &lt; 0] &lt;- 0

# Generate outcome with a GxE interaction
y.GxE &lt;- -0.75 + log(2) * (X[,"SNP1"] != 0) +
  log(4) * E/20 * (X[,"SNP2"] != 0 &amp; X[,"SNP3"] == 0) +
  rnorm(N, 0, 2)
# Test for GxE interaction (Wald test for G:E)
summary(GRSxE(X, y.GxE, E))

# Generate outcome without a GxE interaction
y.no.GxE &lt;- -0.75 + log(2) * (X[,"SNP1"] != 0) +
  log(4) * E/20 + log(4) * (X[,"SNP2"] != 0 &amp; X[,"SNP3"] == 0) +
  rnorm(N, 0, 2)
# Test for GxE interaction (Wald test for G:E)
summary(GRSxE(X, y.no.GxE, E))
</code></pre>

<hr>
<h2 id='GxE'>Testing individual gene-environment interactions</h2><span id='topic+GxE'></span>

<h3>Description</h3>

<p>Function for testing univariate GxE interactions, e.g., using single SNPs
or a GRS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GxE(G, y, E, C = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GxE_+3A_g">G</code></td>
<td>
<p>Numeric vector of a genetic variable such as a GRS (genetic risk
score) or a SNP coded as 0-1-2.</p>
</td></tr>
<tr><td><code id="GxE_+3A_y">y</code></td>
<td>
<p>Numeric vector of the outcome/phenotype. Binary outcomes such
as a disease status should be coded as 0-1 (control-case).</p>
</td></tr>
<tr><td><code id="GxE_+3A_e">E</code></td>
<td>
<p>Numeric vector of the environmental exposure.</p>
</td></tr>
<tr><td><code id="GxE_+3A_c">C</code></td>
<td>
<p>Optional data frame containing potentially confounding
variables to be adjusted for.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses a GLM (generalized linear model) for modelling the
marginal genetic effect, marginal environmental effect, the GxE interaction
effect, and potential confounding effects.
The fitted GLM is returned, which can be, e.g., inspected via
<code>summary(...)</code> to retrieve the Wald test p-values for the individual
terms. The p-value corresponding to the <code>G:E</code> term is the p-value
for testing the presence of a GxE interaction.
</p>


<h3>Value</h3>

<p>An object of class <code>glm</code> is returned, in which <code>G:E</code>
describes the GxE term.
</p>


<h3>References</h3>


<ul>
<li><p> Lau, M., Kress, S., Schikowski, T. &amp; Schwender, H. (2023).
Efficient gene&ndash;environment interaction testing through
bootstrap aggregating. Scientific Reports 13:937.
doi: <a href="https://doi.org/10.1038/s41598-023-28172-4">10.1038/s41598-023-28172-4</a>
</p>
</li></ul>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
