<!DOCTYPE html><html><head><title>Help for package GSA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GSA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GSA'><p>Gene set analysis</p></a></li>
<li><a href='#GSA-internal'><p>Internal GSA functions</p></a></li>
<li><a href='#GSA.correlate'><p>&quot;Correlates&quot; a gene set collection with a  given list of gene nams</p></a></li>
<li><a href='#GSA.func'><p>Gene set analysis without permutations</p></a></li>
<li><a href='#GSA.genescores'><p>Individual gene scores from a  gene set analysis</p></a></li>
<li><a href='#GSA.listsets'><p>List the results from a Gene set analysis</p></a></li>
<li><a href='#GSA.make.features'><p>Creates features from a GSA analysis that can be used in other procedures</p></a></li>
<li><a href='#GSA.plot'><p>Plot the results from a Gene set analysis</p></a></li>
<li><a href='#GSA.read.gmt'><p>Read in a  gene set collection from a .gmt file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Gene Set Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.03.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Brad Efron and R. Tibshirani</td>
</tr>
<tr>
<td>Description:</td>
<td>Gene Set Analysis.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rob Tibshirani &lt;tibs@stat.stanford.edu&gt;</td>
</tr>
<tr>
<td>Suggests:</td>
<td>impute</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-2">LGPL-2</a> | <a href="https://www.r-project.org/Licenses/LGPL-2.1">LGPL-2.1</a> | <a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a> [expanded from: LGPL]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://tibshirani.su.domains/GSA/">https://tibshirani.su.domains/GSA/</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-20 09:52:08 UTC; hornik</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-20 09:58:26 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
</table>
<hr>
<h2 id='GSA'>Gene set analysis</h2><span id='topic+GSA'></span>

<h3>Description</h3>

<p>Determines the significance of  pre-defined sets of genes  with respect to
an outcome
variable, such as a group indicator, a quantitative variable or a survival time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSA(x,y, genesets, genenames,
method=c("maxmean","mean","absmean"),
resp.type=c("Quantitative","Two class unpaired","Survival","Multiclass",
            "Two class paired", "tCorr", "taCorr"),
censoring.status=NULL,random.seed=NULL,  knn.neighbors=10,
s0=NULL, s0.perc=NULL,minsize=15,maxsize=500,
restand=TRUE,restand.basis=c("catalog","data"),
 nperms=200, 
xl.mode=c("regular","firsttime","next20","lasttime"), 
xl.time=NULL, xl.prevfit=NULL)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSA_+3A_x">x</code></td>
<td>
<p>Data x: p by n matrix of features (expression values),
one observation per column (missing values allowed); y: n-vector of outcome measurements</p>
</td></tr>
<tr><td><code id="GSA_+3A_y">y</code></td>
<td>
<p>Vector of response values:  1,2 for two class
problem, or 1,2,3 ... for multiclass problem, or real numbers for quantitative
or survival problems</p>
</td></tr>
<tr><td><code id="GSA_+3A_genesets">genesets</code></td>
<td>
<p>Gene set collection (a list)</p>
</td></tr>
<tr><td><code id="GSA_+3A_genenames">genenames</code></td>
<td>
<p>Vector of genenames in expression dataset</p>
</td></tr>
<tr><td><code id="GSA_+3A_method">method</code></td>
<td>
<p>Method for summarizing a gene set: &quot;maxmean&quot; (default), &quot;mean&quot; or &quot;absmean&quot;</p>
</td></tr>
<tr><td><code id="GSA_+3A_resp.type">resp.type</code></td>
<td>
<p>Problem type: &quot;quantitative&quot; for a continuous parameter;
&quot;Two class unpaired&quot; ; &quot;Survival&quot; for censored survival outcome; &quot;Multiclass&quot; :
more than 2 groups, coded 1,2,3...; &quot;Two class paired&quot; for paired outcomes, coded -1,1 (first pair), -2,2 (second pair), etc</p>
</td></tr>
<tr><td><code id="GSA_+3A_censoring.status">censoring.status</code></td>
<td>
<p>Vector of censoring status values for survival problems,
1 mean death or failure, 0 means censored</p>
</td></tr>
<tr><td><code id="GSA_+3A_random.seed">random.seed</code></td>
<td>
<p>Optional initial seed for random number generator (integer)</p>
</td></tr>
<tr><td><code id="GSA_+3A_knn.neighbors">knn.neighbors</code></td>
<td>
<p>Number of nearest neighbors to use for imputation
of missing features values</p>
</td></tr>
<tr><td><code id="GSA_+3A_s0">s0</code></td>
<td>
<p>Exchangeability factor  for denominator of test statistic; Default
is automatic choice</p>
</td></tr>
<tr><td><code id="GSA_+3A_s0.perc">s0.perc</code></td>
<td>
<p>Percentile of standard  deviation values to use for s0; default is
automatic choice; -1 means s0=0 (different from s0.perc=0, meaning
s0=zeroeth percentile of  standard  deviation values= min of sd values)</p>
</td></tr>
<tr><td><code id="GSA_+3A_minsize">minsize</code></td>
<td>
<p>Minimum number of genes in  genesets to be  considered</p>
</td></tr>
<tr><td><code id="GSA_+3A_maxsize">maxsize</code></td>
<td>
<p>Maximum number of genes in  genesets to be considered</p>
</td></tr>
<tr><td><code id="GSA_+3A_restand">restand</code></td>
<td>
<p>Should restandardization be done? Default TRUE</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code id="GSA_+3A_restand.basis">restand.basis</code></td>
<td>
<p>What should be used to do the  restandardization?
The set of genes in the genesets (&quot;catalog&quot;, the default) or the
genes in the data set (&quot;data&quot;)</p>
</td></tr>
<tr><td><code id="GSA_+3A_nperms">nperms</code></td>
<td>
<p>Number of permutations used to estimate false discovery rates</p>
</td></tr>
<tr><td><code id="GSA_+3A_xl.mode">xl.mode</code></td>
<td>
<p>Used by Excel interface</p>
</td></tr>
<tr><td><code id="GSA_+3A_xl.time">xl.time</code></td>
<td>
<p>Used by Excel interface</p>
</td></tr>
<tr><td><code id="GSA_+3A_xl.prevfit">xl.prevfit</code></td>
<td>
<p>Used by Excel interface</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Carries out a Gene set analysis, as described in the paper
by Efron and Tibshirani (2006). It differs from a Gene Set Enrichment Analysis
(Subramanian et al 2006)
in its use of the &quot;maxmean&quot; statistic: this is the mean of the positive or
negative part of gene scores in the gene set, whichever is large in absolute
values.  Efron and Tibshirani shows that this is often more powerful
than the modified KS statistic used in GSEA.
GSA also does &quot;restandardization&quot; of the genes (rows), on top of the
permutation of columns (done in GSEA).
Gene set analysis is applicable to microarray data
and other data with a large number of features. This is also the R package
that is called by the &quot;official&quot; SAM Excel package v3.0.
The format of the response vector y and the calling sequence
is illustrated in the examples below. A more complete  description
is given in the SAM manual
at http://www-stat.stanford.edu/~tibs/SAM</p>


<h3>Value</h3>

<p>A list with components
</p>
<table>
<tr><td><code>GSA.scores</code></td>
<td>
<p>Gene set scores  for each gene set</p>
</td></tr>
<tr><td><code>GSA.scores.perm</code></td>
<td>
<p>Matrix of Gene set scores from permutions,  one column per
permutation</p>
</td></tr>
<tr><td><code>fdr.lo</code></td>
<td>
<p>Estimated false discovery rates for negative gene sets
(negative means lower expression correlates with  class 2 in two sample problems,
lower expression correlates with  increased y for quantitative problems,
lower expression correlates with higher risk for survival problems)</p>
</td></tr>
<tr><td><code>fdr.hi</code></td>
<td>
<p>Estimated false discovery rates for positive gene sets; positive
is opposite of negative, as defined above</p>
</td></tr>
<tr><td><code>pvalues.lo</code></td>
<td>
<p>P-values for negative gene sets</p>
</td></tr>
<tr><td><code>pvalues.hi</code></td>
<td>
<p>P-values for positive gene sets</p>
</td></tr>
<tr><td><code>stand.info</code></td>
<td>
<p>Information from restandardization process</p>
</td></tr>
<tr><td><code>stand.info.star</code></td>
<td>
<p>Information from restandardization process in permutations</p>
</td></tr>
<tr><td><code>ngenes</code></td>
<td>
<p>Number of genes in union of gene sets</p>
</td></tr>
<tr><td><code>nperms</code></td>
<td>
<p>Number of permutations used</p>
</td></tr>
<tr><td><code>gene.scores</code></td>
<td>
<p>Individual gene scores (eg t-statistics for two class problem)</p>
</td></tr>
<tr><td><code>s0</code></td>
<td>
<p>Computed  exchangeability factor</p>
</td></tr>
<tr><td><code>s0.perc</code></td>
<td>
<p>Computed percentile of standard deviation values.
s0= s0.perc percentile of the gene standard deviations</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>The call to GSA</p>
</td></tr>
<tr><td><code>x</code></td>
<td>
<p>For internal use</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>For internal use</p>
</td></tr>
<tr><td><code>genesets</code></td>
<td>
<p>For internal use</p>
</td></tr>
<tr><td><code>genenames</code></td>
<td>
<p>For internal use</p>
</td></tr>
<tr><td><code>r.obs</code></td>
<td>
<p>For internal use</p>
</td></tr>
<tr><td><code>r.star</code></td>
<td>
<p>For internal use</p>
</td></tr>
<tr><td><code>gs.mat</code></td>
<td>
<p>For internal use</p>
</td></tr>
<tr><td><code>gs.ind</code></td>
<td>
<p>For internal use</p>
</td></tr>
<tr><td><code>catalog</code></td>
<td>
<p>For internal use</p>
</td></tr>
<tr><td><code>catalog.unique</code></td>
<td>
<p>For internal use</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Tibshirani</p>


<h3>References</h3>

<p>Efron, B. and   Tibshirani, R.  
On testing the significance of sets of genes. Stanford tech report rep 2006.
http://www-stat.stanford.edu/~tibs/ftp/GSA.pdf
</p>
<p>Subramanian, A. and Tamayo, P. Mootha, V. K. and Mukherjee, S. and Ebert, B. L. and Gillette, M. A. and Paulovich, A. and Pomeroy, S. L. and Golub, T. R. and Lander, E. S. and Mesirov, J. P. (2005)  A knowledge-based approach for interpreting genome-wide expression profiles. PNAS. 102, pg 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
######### two class unpaired comparison
# y must take values 1,2

set.seed(100)
x&lt;-matrix(rnorm(1000*20),ncol=20)
dd&lt;-sample(1:1000,size=100)

u&lt;-matrix(2*rnorm(100),ncol=10,nrow=100)
x[dd,11:20]&lt;-x[dd,11:20]+u
y&lt;-c(rep(1,10),rep(2,10))


genenames=paste("g",1:1000,sep="")

#create some random gene sets
genesets=vector("list",50)
for(i in 1:50){
 genesets[[i]]=paste("g",sample(1:1000,size=30),sep="")
}
geneset.names=paste("set",as.character(1:50),sep="")

GSA.obj&lt;-GSA(x,y, genenames=genenames, genesets=genesets,
             resp.type="Two class unpaired", nperms=100)


GSA.listsets(GSA.obj, geneset.names=geneset.names,FDRcut=.5)



#to use  "real" gene set collection, we read it in from a gmt file:
# 
# geneset.obj&lt;- GSA.read.gmt("file.gmt")
# 
# where file.gmt is a gene set collection from GSEA collection or
#  or the website http://www-stat.stanford.edu/~tibs/GSA, or one
# that you have created yourself. Then

#   GSA.obj&lt;-GSA(x,y, genenames=genenames, genesets=geneset.obj$genesets,
#                resp.type="Two class unpaired", nperms=100)
#
#



</code></pre>

<hr>
<h2 id='GSA-internal'>Internal GSA functions</h2><span id='topic+print.GSA.func'></span><span id='topic+print.GSA'></span><span id='topic+summary.GSA.genesets'></span><span id='topic+ttest.func'></span><span id='topic+cox.func'></span><span id='topic+multiclass.func'></span><span id='topic+quantitative.func'></span><span id='topic+varr'></span><span id='topic+coxvar'></span><span id='topic+coxstuff'></span><span id='topic+est.s0'></span><span id='topic+GSA.xl.genescores'></span><span id='topic+GSA.xl.correlate'></span><span id='topic+GSA.xl.summary'></span><span id='topic+GSA.xl.plot'></span><span id='topic+GSA.xl.summary.genesets'></span>

<h3>Description</h3>

<p>Internal samr functions</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GSA.func'
print(x, ...) 
## S3 method for class 'GSA'
print(x, ...) 
## S3 method for class 'GSA.genesets'
summary(object, ...)
varr(x, meanx=NULL) 
coxvar(x, y, ic, offset = rep(0., length(y)), coxstuff.obj = NULL)
coxstuff(x, y, ic, offset = rep(0., length(y))) 
est.s0(tt,sd,s0.perc=seq(0,1, by=.05))
GSA.xl.genescores(GSA.listsets.obj, genesets,  GSA.obj,  genenames)
GSA.xl.plot(GSA.obj,  fac=.10, FDRcut=1)
GSA.xl.summary.genesets(GSA.genesets.obj)
GSA.xl.correlate(GSA.genesets.obj, genenames)
ttest.func(x, y, s0 = 0, sd=NULL)
cox.func(x, y, censoring.status, s0 = 0)
multiclass.func(x, y, s0 = 0)
quantitative.func(x, y, s0 = 0)
</code></pre>


<h3>Details</h3>

<p>These are not to be called by the user.
</p>


<h3>Author(s)</h3>

<p>Robert Tibshirani</p>

<hr>
<h2 id='GSA.correlate'>&quot;Correlates&quot; a gene set collection with a  given list of gene nams</h2><span id='topic+GSA.correlate'></span>

<h3>Description</h3>

<p>&quot;Correlates&quot; a gene set collection with a  given list of gene names. Gives info
on the overlap between the collection and the list of genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSA.correlate(GSA.genesets.obj, genenames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSA.correlate_+3A_gsa.genesets.obj">GSA.genesets.obj</code></td>
<td>
<p>Gene set collection, created for example by GSA.read.gmt</p>
</td></tr>
<tr><td><code id="GSA.correlate_+3A_genenames">genenames</code></td>
<td>
<p>Vector of gene names in expression daatset</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Gives info
on the overlap between a gene set collection and the list of gene names.
This is for information purposes, to find out, for example,
how many genes in the list of genes appear in the gene set collection.</p>


<h3>Author(s)</h3>

<p>Robert Tibshirani</p>


<h3>References</h3>

<p>Efron, B. and   Tibshirani, R.  
On testing the significance of sets of genes. Stanford tech report rep 2006.
http://www-stat.stanford.edu/~tibs/ftp/GSA.pdf
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
######### two class unpaired comparison
# y must take values 1,2

set.seed(100)
x&lt;-matrix(rnorm(1000*20),ncol=20)
dd&lt;-sample(1:1000,size=100)

u&lt;-matrix(2*rnorm(100),ncol=10,nrow=100)
x[dd,11:20]&lt;-x[dd,11:20]+u
y&lt;-c(rep(1,10),rep(2,10))


genenames=paste("g",1:1000,sep="")

#create some random gene sets
genesets=vector("list",50)
for(i in 1:50){
 genesets[[i]]=paste("g",sample(1:1000,size=30),sep="")
}
geneset.names=paste("set",as.character(1:50),sep="")

GSA.correlate(genesets, genenames)





</code></pre>

<hr>
<h2 id='GSA.func'>Gene set analysis without permutations</h2><span id='topic+GSA.func'></span>

<h3>Description</h3>

<p>Determines the significance of pre-defined sets of genes  with respect to 
an outcome
variable, such as a group indicator, quantitative variable or  survival time.
This is the basic function called by GSA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSA.func(x,y, genesets, genenames,geneset.names=NULL,
 method=c("maxmean","mean","absmean"),
 resp.type=c("Quantitative","Two class unpaired","Survival","Multiclass",
             "Two class paired",  "tCorr", "taCorr" ),
censoring.status=NULL,
 first.time = TRUE, return.gene.ind = TRUE, 
ngenes = NULL, gs.mat =NULL, gs.ind = NULL,
 catalog = NULL, catalog.unique =NULL, 
s0 = NULL, s0.perc = NULL, minsize = 15, maxsize= 500, restand = TRUE,
restand.basis=c("catalog","data")) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSA.func_+3A_x">x</code></td>
<td>
<p>Data x: p by n matrix of features,
one observation per column (missing values allowed)</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_y">y</code></td>
<td>
<p>Vector of response values:  1,2 for two class
problem, or 1,2,3 ... for multiclass problem, or real numbers for quantitative
or survival problems</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_genesets">genesets</code></td>
<td>
<p>Gene set collection (a list)</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_genenames">genenames</code></td>
<td>
<p>Vector of genenames in expression dataset</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_geneset.names">geneset.names</code></td>
<td>
<p>Optional vector of gene set names</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_method">method</code></td>
<td>
<p>Method for summarizing a gene set: &quot;maxmean&quot; (default), &quot;mean&quot; or &quot;absmean&quot;</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_resp.type">resp.type</code></td>
<td>
<p>Problem type: &quot;quantitative&quot; for a continuous parameter;
&quot;Two class unpaired&quot; ; &quot;Survival&quot; for censored survival outcome; &quot;Multiclass&quot; :
more than 2 groups; &quot;Two class paired&quot; for paired outcomes, coded -1,1 (first pair), -2,2 (second pair), etc  </p>
</td></tr>
<tr><td><code id="GSA.func_+3A_censoring.status">censoring.status</code></td>
<td>
<p>Vector of censoring status values for survival problems,
1 mean death or failure, 0 means censored)</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_first.time">first.time</code></td>
<td>
<p>internal use</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_return.gene.ind">return.gene.ind</code></td>
<td>
<p>internal use</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_ngenes">ngenes</code></td>
<td>
<p>internal use</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_gs.mat">gs.mat</code></td>
<td>
<p>internal use</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_gs.ind">gs.ind</code></td>
<td>
<p>internal use</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_catalog">catalog</code></td>
<td>
<p>internal use</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_catalog.unique">catalog.unique</code></td>
<td>
<p>internal use</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_s0">s0</code></td>
<td>
<p>Exchangeability factor  for denominator of test statistic; Default
is automatic choice</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_s0.perc">s0.perc</code></td>
<td>
<p>Percentile of standard  deviation values to use for s0; default is
automatic choice; -1 means s0=0 (different from s0.perc=0, meaning
s0=zeroeth percentile of  standard  deviation values= min of sd values</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_minsize">minsize</code></td>
<td>
<p>Minimum number of genes in  genesets to be  considered</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_maxsize">maxsize</code></td>
<td>
<p>Maximum number of genes in  genesets to be considered</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_restand">restand</code></td>
<td>
<p>Should restandardization be done? Default TRUE</p>
</td></tr>
<tr><td><code id="GSA.func_+3A_restand.basis">restand.basis</code></td>
<td>
<p>What should be used to do the  restandardization?
The set of genes in the genesets (&quot;catalog&quot;, the default) or the
genes in the data set (&quot;data&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Carries out a Gene set analysis, computing the gene set scores.
This function does not do any permutations for estimation of  false discovery rates.
GSA calls this function to estimate  FDRs.
</p>


<h3>Value</h3>

<p>A list with components
</p>
<table>
<tr><td><code>scores</code></td>
<td>
<p>Gene set scores  for each gene set</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>norm.scores</code></td>
<td>
<p>Gene set scores  transformed by the inverse Gaussian cdf</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>mean</code></td>
<td>
<p>Means of gene  expression values for each sample</p>
</td></tr>
<tr><td><code>sd</code></td>
<td>
<p>Standard deviation of gene  expression values for each sample</p>
</td></tr>
<tr><td><code>gene.ind</code></td>
<td>
<p>List indicating whch genes in each positive  gene set
had positive individual scores, and similarly for negative gene sets</p>
</td></tr>
<tr><td><code>geneset.names</code></td>
<td>
<p>Names of the gene sets</p>
</td></tr>
<tr><td><code>nperms</code></td>
<td>
<p>Number of permutations used</p>
</td></tr>
<tr><td><code>gene.scores</code></td>
<td>
<p>Individual gene scores (eg t-statistics for two class problem)</p>
</td></tr>
<tr><td><code>s0</code></td>
<td>
<p>Computed  exchangeability factor</p>
</td></tr>
<tr><td><code>s0.perc</code></td>
<td>
<p>Computed percentile of standard deviation values</p>
</td></tr>
<tr><td><code>stand.info</code></td>
<td>
<p> Information computed  used in the restandardization process</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>Method used (from call to GSA.func)</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>The call to GSA</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Tibshirani</p>


<h3>References</h3>

<p>Efron, B. and   Tibshirani, R.  
On testing the significance of sets of genes. Stanford tech report rep 2006.
http://www-stat.stanford.edu/~tibs/ftp/GSA.pdf</p>


<h3>Examples</h3>

<pre><code class='language-R'>
######### two class unpaired comparison
# y must take values 1,2

set.seed(100)
x&lt;-matrix(rnorm(1000*20),ncol=20)
dd&lt;-sample(1:1000,size=100)

u&lt;-matrix(2*rnorm(100),ncol=10,nrow=100)
x[dd,11:20]&lt;-x[dd,11:20]+u
y&lt;-c(rep(1,10),rep(2,10))


genenames=paste("g",1:1000,sep="")

#create some random gene sets
genesets=vector("list",50)
for(i in 1:50){
 genesets[[i]]=paste("g",sample(1:1000,size=30),sep="")
}
geneset.names=paste("set",as.character(1:50),sep="")

GSA.func.obj&lt;-GSA.func(x,y, genenames=genenames, genesets=genesets,  resp.type="Two class unpaired")




#to use  "real" gene set collection, we read it in from a gmt file:
# 
# geneset.obj&lt;- GSA.read.gmt("file.gmt")
# 
# where file.gmt is a gene set collection from GSEA collection or
#  or the website http://www-stat.stanford.edu/~tibs/GSA, or one
# that you have created yourself. Then

#   GSA.func.obj&lt;-GSA.func(x,y, genenames=genenames,
#                          genesets=geneset.obj$genesets,
#                          resp.type="Two class unpaired")
#
#



</code></pre>

<hr>
<h2 id='GSA.genescores'>Individual gene scores from a  gene set analysis</h2><span id='topic+GSA.genescores'></span>

<h3>Description</h3>

<p>Compute individual gene scores from a  gene set analysis 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSA.genescores(geneset.number, genesets,  GSA.obj,  genenames, negfirst=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSA.genescores_+3A_geneset.number">geneset.number</code></td>
<td>
<p>Number indicating which  gene set is to examined</p>
</td></tr>
<tr><td><code id="GSA.genescores_+3A_genesets">genesets</code></td>
<td>
<p>The gene set collection</p>
</td></tr>
<tr><td><code id="GSA.genescores_+3A_gsa.obj">GSA.obj</code></td>
<td>
<p>Object returned by function GSA</p>
</td></tr>
<tr><td><code id="GSA.genescores_+3A_genenames">genenames</code></td>
<td>
<p>Vector of gene names for gene in expression dataset</p>
</td></tr>
<tr><td><code id="GSA.genescores_+3A_negfirst">negfirst</code></td>
<td>
<p>Should negative genes be listed first? Default FALSE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Compute individual gene scores from a  gene set analysis.
Useful for
looking &ldquo;inside&rdquo; a gene set that has been called significant by GSA.</p>


<h3>Value</h3>

<p>A list with components
</p>
<table>
<tr><td><code>res</code></td>
<td>
<p>Matrix of gene names and gene scores (eg t-statistics)  for each gene in the  gene set</p>
</td></tr></table>
<p>,
</p>


<h3>Author(s)</h3>

<p>Robert Tibshirani</p>


<h3>References</h3>

<p>Efron, B. and   Tibshirani, R.  
On testing the significance of sets of genes. Stanford tech report rep 2006.
http://www-stat.stanford.edu/~tibs/ftp/GSA.pdf
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
######### two class unpaired comparison
# y must take values 1,2

set.seed(100)
x&lt;-matrix(rnorm(1000*20),ncol=20)
dd&lt;-sample(1:1000,size=100)

u&lt;-matrix(2*rnorm(100),ncol=10,nrow=100)
x[dd,11:20]&lt;-x[dd,11:20]+u
y&lt;-c(rep(1,10),rep(2,10))


genenames=paste("g",1:1000,sep="")

#create some random gene sets
genesets=vector("list",50)
for(i in 1:50){
 genesets[[i]]=paste("g",sample(1:1000,size=30),sep="")
}
geneset.names=paste("set",as.character(1:50),sep="")

GSA.obj&lt;-GSA(x,y, genenames=genenames, genesets=genesets,
             resp.type="Two class unpaired", nperms=100)

# look at 10th gene set

GSA.genescores(10, genesets, GSA.obj, genenames)


</code></pre>

<hr>
<h2 id='GSA.listsets'>List the results from a Gene set analysis</h2><span id='topic+GSA.listsets'></span>

<h3>Description</h3>

<p>List the results from a call to  GSA  (Gene set analysis)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSA.listsets(GSA.obj, geneset.names = NULL, maxchar = 20, FDRcut = 0.2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSA.listsets_+3A_gsa.obj">GSA.obj</code></td>
<td>
<p> Object returned by GSA function</p>
</td></tr></table>
<p>.
</p>
<table>
<tr><td><code id="GSA.listsets_+3A_geneset.names">geneset.names</code></td>
<td>
<p>Optional vector of names for  the gene sets</p>
</td></tr>
<tr><td><code id="GSA.listsets_+3A_maxchar">maxchar</code></td>
<td>
<p>Maximum number of characters in printed output</p>
</td></tr>
<tr><td><code id="GSA.listsets_+3A_fdrcut">FDRcut</code></td>
<td>
<p>False discovery rate cutpoint for listed sets. A value of 1
will cause all sets to be listed</p>
</td></tr></table>
<p>.
</p>


<h3>Details</h3>

<p>This function  list the sigificant gene sets, based on a
call to the GSA (Gene set analysis) function. </p>


<h3>Value</h3>

<p>A list with components
</p>
<table>
<tr><td><code>FDRcut</code></td>
<td>
<p>The false discovery rate threshold used.</p>
</td></tr>
<tr><td><code>negative</code></td>
<td>
<p>A table of the negative gene sets. &quot;Negative&quot; means that
lower expression of most genes in the gene set correlates with
higher values of the phenotype y. Eg for two classes coded 1,2,
lower expression correlates with class 2. For survival data,
lower expression correlates with higher risk, i.e shorter survival
(Be careful, this can be confusing!)</p>
</td></tr>
<tr><td><code>positive</code></td>
<td>
<p>A table of the positive gene sets.
&quot;Positive&quot; means that 
higher expression of most genes in the gene set correlates with
higher values of the phenotype y. See &quot;negative&quot; above for more info.</p>
</td></tr>
<tr><td><code>nsets.neg</code></td>
<td>
<p>Number of negative gene sets</p>
</td></tr>
<tr><td><code>nsets.pos</code></td>
<td>
<p>Number of positive gene sets</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Tibshirani</p>


<h3>References</h3>

<p>Efron, B. and   Tibshirani, R.  
On testing the significance of sets of genes. Stanford tech report rep 2006.
http://www-stat.stanford.edu/~tibs/ftp/GSA.pdf
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
######### two class unpaired comparison
# y must take values 1,2

set.seed(100)
x&lt;-matrix(rnorm(1000*20),ncol=20)
dd&lt;-sample(1:1000,size=100)

u&lt;-matrix(2*rnorm(100),ncol=10,nrow=100)
x[dd,11:20]&lt;-x[dd,11:20]+u
y&lt;-c(rep(1,10),rep(2,10))


genenames=paste("g",1:1000,sep="")

#create some radnom gene sets
genesets=vector("list",50)
for(i in 1:50){
 genesets[[i]]=paste("g",sample(1:1000,size=30),sep="")
}
geneset.names=paste("set",as.character(1:50),sep="")

GSA.obj&lt;-GSA(x,y, genenames=genenames, genesets=genesets,
             resp.type="Two class unpaired", nperms=100)


GSA.listsets(GSA.obj, geneset.names=geneset.names,FDRcut=.5)


</code></pre>

<hr>
<h2 id='GSA.make.features'>Creates features from a GSA analysis that can be used in other procedures</h2><span id='topic+GSA.make.features'></span>

<h3>Description</h3>

<p>Creates features from a GSA analysis that can be used in other procedures,
for example, sample classification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSA.make.features(GSA.func.obj, x, genesets, genenames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSA.make.features_+3A_gsa.func.obj">GSA.func.obj</code></td>
<td>
<p>Object returned by GSA.func</p>
</td></tr>
<tr><td><code id="GSA.make.features_+3A_x">x</code></td>
<td>
<p>Expression dataset from which the features are to be created</p>
</td></tr>
<tr><td><code id="GSA.make.features_+3A_genesets">genesets</code></td>
<td>
<p>Gene set collection</p>
</td></tr>
<tr><td><code id="GSA.make.features_+3A_genenames">genenames</code></td>
<td>
<p>Vector of gene names in expression dataset</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates features from a GSA analysis that can be used in other procedures,
for example, sample classification.
For example, suppose the GSA analysis  computes a  maxmean score for gene set
1 that is positive, based on the  mean of the positive part of
the scores  in that
gene set. Call the  subset of genes  with positive scores &quot;A&quot;. Then  we compute a new feature for this geneset, for each sample,
by computing the mean of the scores for genes in A, setting other gene scores
to zero.
</p>


<h3>Author(s)</h3>

<p>Robert Tibshirani</p>


<h3>References</h3>

<p>Efron, B. and   Tibshirani, R.  
On testing the significance of sets of genes. Stanford tech report rep 2006.
http://www-stat.stanford.edu/~tibs/ftp/GSA.pdf
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
######### two class unpaired comparison
# y must take values 1,2

set.seed(100)
x&lt;-matrix(rnorm(1000*20),ncol=20)
dd&lt;-sample(1:1000,size=100)

u&lt;-matrix(2*rnorm(100),ncol=10,nrow=100)
x[dd,11:20]&lt;-x[dd,11:20]+u
y&lt;-c(rep(1,10),rep(2,10))


genenames=paste("g",1:1000,sep="")

#create some random gene sets
genesets=vector("list",50)
for(i in 1:50){
 genesets[[i]]=paste("g",sample(1:1000,size=30),sep="")
}
geneset.names=paste("set",as.character(1:50),sep="")

GSA.func.obj&lt;-GSA.func(x,y, genenames=genenames, genesets=genesets,  resp.type="Two class unpaired")


GSA.make.features(GSA.func.obj, x, genesets, genenames)





</code></pre>

<hr>
<h2 id='GSA.plot'>Plot the results from a Gene set analysis</h2><span id='topic+GSA.plot'></span>

<h3>Description</h3>

<p>Plots the results from a call to  GSA  (Gene set analysis)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSA.plot(GSA.obj, fac=1, FDRcut = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSA.plot_+3A_gsa.obj">GSA.obj</code></td>
<td>
<p> Object returned by GSA function</p>
</td></tr></table>
<p>.
</p>
<table>
<tr><td><code id="GSA.plot_+3A_fac">fac</code></td>
<td>
<p>value for jittering points in plot (&quot;factor&quot; in called to jitter()</p>
</td></tr>
<tr><td><code id="GSA.plot_+3A_fdrcut">FDRcut</code></td>
<td>
<p>False discovery rate cutpoint for  sets to be plotted. A value of 1
(the default) will cause all sets to be plotted</p>
</td></tr></table>
<p>.
</p>


<h3>Details</h3>

<p>This function  makes a plot of the significant gene sets, based on a
call to the GSA (Gene set analysis) function. </p>


<h3>Author(s)</h3>

<p>Robert Tibshirani</p>


<h3>References</h3>

<p>Efron, B. and   Tibshirani, R.  
On testing the significance of sets of genes. Stanford tech report rep 2006.
http://www-stat.stanford.edu/~tibs/ftp/GSA.pdf
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
######### two class unpaired comparison
# y must take values 1,2

set.seed(100)
x&lt;-matrix(rnorm(1000*20),ncol=20)
dd&lt;-sample(1:1000,size=100)

u&lt;-matrix(2*rnorm(100),ncol=10,nrow=100)
x[dd,11:20]&lt;-x[dd,11:20]+u
y&lt;-c(rep(1,10),rep(2,10))


genenames=paste("g",1:1000,sep="")

#create some radnom gene sets
genesets=vector("list",50)
for(i in 1:50){
 genesets[[i]]=paste("g",sample(1:1000,size=30),sep="")
}
geneset.names=paste("set",as.character(1:50),sep="")

GSA.obj&lt;-GSA(x,y, genenames=genenames, genesets=genesets,
             resp.type="Two class unpaired", nperms=100)


GSA.listsets(GSA.obj, geneset.names=geneset.names,FDRcut=.5)

GSA.plot(GSA.obj)

</code></pre>

<hr>
<h2 id='GSA.read.gmt'>Read in a  gene set collection from a .gmt file</h2><span id='topic+GSA.read.gmt'></span>

<h3>Description</h3>

<p>Read in a  gene set collection from a .gmt file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSA.read.gmt(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSA.read.gmt_+3A_filename">filename</code></td>
<td>
<p>The name of a file to read data values from. Should be
a tab-separated text file, with one row per gene set. Column 1 has gene set
names (identifiers), column 2 has gene set descriptions, remaining columns are gene ids for genes in that
geneset</p>
</td></tr></table>
<p>.
</p>


<h3>Details</h3>

<p>This function  reads in a geneset collection from a  .gmt text file,
and creates  an R object that can be used as input into GSA.
We use UniGene symbols for our gene set names in our .gmt files and expression datasets,
to match the  two.
However the user is free to use other identifiers, as long as the same ones
are used in the  gene set collections and expression datasets.
</p>


<h3>Value</h3>

<p>A list with components
</p>
<table>
<tr><td><code>genesets</code></td>
<td>
<p>List of gene names (identifiers) in each gene set</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>geneset.names</code></td>
<td>
<p>Vector of gene set names (identifiers)</p>
</td></tr></table>
<p>,
</p>
<table>
<tr><td><code>geneset.descriptions</code></td>
<td>
<p>Vector of gene set descriptions</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Tibshirani</p>


<h3>References</h3>

<p>Efron, B. and   Tibshirani, R.  
On testing the significance of sets of genes. Stanford tech report rep 2006.
http://www-stat.stanford.edu/~tibs/ftp/GSA.pdf
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


# read in  functional pathways gene set file from Broad institute GSEA website
# http://www.broad.mit.edu/gsea/msigdb/msigdb_index.html
# You have to register first and then download the file C2.gmt from
#   their site

#GSA.read.gmt(C2.gmt)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
