<!DOCTYPE html><html><head><title>Help for package GSAfisherCombined</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GSAfisherCombined}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GSAfisherCombined-package'>
<p>Gene Set Analysis with Fisher Combined Method</p></a></li>
<li><a href='#GSAfisher'><p>GSAfisher</p></a></li>
<li><a href='#GSAfisher.default'><p>GSAfisher.default</p></a></li>
<li><a href='#GSAfisher.multiple'><p>GSAfisher.multiple</p></a></li>
<li><a href='#GSAfisher.plot'><p>GSAfisher.plot</p></a></li>
<li><a href='#GSAfisher.print'><p>GSAfisher.print</p></a></li>
<li><a href='#GSAfisher.summary'><p>GSAfisher.summary</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Set Analysis with Fisher Combined Method</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-06-22</td>
</tr>
<tr>
<td>Author:</td>
<td>Carlos Garcia Prieto</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Carlos Garcia Prieto &lt;carlosantonio.garcia@uvic.cat&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides the p-value for a joint test of association between a phenotype and a set of genetic variants (SNPs) by combining marginal p-values using the Fisher method. See Fisher, R.A. (1925,ISBN:0-05-002170-2) Statistical Methods for Research Workers.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-06-24 13:41:53 UTC; carlosgarciaprieto</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-06-24 14:07:59 UTC</td>
</tr>
</table>
<hr>
<h2 id='GSAfisherCombined-package'>
Gene Set Analysis with Fisher Combined Method
</h2><span id='topic+GSAfisherCombined-package'></span><span id='topic+GSAfisherCombined'></span>

<h3>Description</h3>

<p>Provides the p-value for a joint test of association between a phenotype and a set of genetic variants (SNPs) by combining marginal p-values using the Fisher method. See Fisher, R.A. (1925,ISBN:0-05-002170-2) Statistical Methods for Research Workers.
</p>


<h3>Details</h3>

<p>Package: GSAfisherCombined
Type: Package
Version: 1.0
Date: 2018-02-10
License: GPL (&gt;= 2)
</p>


<h3>Author(s)</h3>

<p>Carlos Garcia Prieto
</p>
<p>Maintainer: Carlos Garcia Prieto &lt;carlosantonio.garcia@uvic.cat&gt;
</p>


<h3>References</h3>

<p>Fisher, R.A. (1925). Statistical Methods for Research Workers. ISBN 0-05-002170-2.
</p>

<hr>
<h2 id='GSAfisher'>GSAfisher</h2><span id='topic+GSAfisher'></span>

<h3>Description</h3>

<p>It is a generic function that dispatches different methods. It provides the p-value for a joint test of association between a phenotype and a set of genetic variants (SNPs) by combining marginal p-values using the Fisher method.</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSAfisher(x,...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSAfisher_+3A_x">x</code></td>
<td>
<p>Set of marginal p-values.</p>
</td></tr>
<tr><td><code id="GSAfisher_+3A_...">...</code></td>
<td>
<p>Further arguments passed to other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Use methods(&quot;GSAfisher&quot;) to get all the methods for the GSAfisher generic.</p>


<h3>Value</h3>

<p>UseMethod(&quot;GSAfisher&quot;)</p>


<h3>Author(s)</h3>

<p>Carlos Garcia Prieto</p>


<h3>References</h3>

<p>Fisher, R.A. (1925). Statistical Methods for Research Workers. ISBN 0-05-002170-2.
</p>

<hr>
<h2 id='GSAfisher.default'>GSAfisher.default</h2><span id='topic+GSAfisher.default'></span>

<h3>Description</h3>

<p>Default method of GSAfisher function that computes the p-value for a joint test of association between a phenotype and a set of genetic variants (SNPs) by combining marginal p-values using the Fisher method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
GSAfisher(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSAfisher.default_+3A_x">x</code></td>
<td>
<p>Numeric set of marginal p-values (one vector or a sinlge p-value).</p>
</td></tr>
<tr><td><code id="GSAfisher.default_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>p</code></td>
<td>
<p>Fisher method combined p-value.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Carlos Garcia Prieto</p>


<h3>References</h3>

<p>Fisher, R.A. (1925). Statistical Methods for Research Workers. ISBN 0-05-002170-2.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Generate 1000 random values from an Uniform distribution U (0, 0.25). 
x&lt;-runif(1000, 0, 0.25)
#Apply the GSAfisher.default function to this data vector.
GSAfisher(x)
</code></pre>

<hr>
<h2 id='GSAfisher.multiple'>GSAfisher.multiple</h2><span id='topic+GSAfisher.multiple'></span>

<h3>Description</h3>

<p>Multiple method of GSAfisher function that computes the p-value for a joint test of association between a phenotype and multiple gene-sets of genetic variants (SNPs) by combining multiple marginal p-values data vectors using the Fisher method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'multiple'
GSAfisher(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSAfisher.multiple_+3A_...">...</code></td>
<td>
<p>Set of multiple marginal p-values data vectors or matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>p.multiple</code></td>
<td>
<p>Set of multiple Fisher method combined p-values for each data vector.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Carlos Garcia Prieto</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Generate 3 vectors with 1000 random values from an Uniform distribution U (0, 0.25). 
data.vector1&lt;-runif(1000, 0, 0.25)
data.vector2&lt;-runif(1000, 0, 0.25)
data.vector3&lt;-runif(1000, 0, 0.25)
#Set data class to "multiple".
class(data.vector1)&lt;-"multiple"
#Apply the GSAfisher.multiple function to these data vectors.
GSAfisher(data.vector1, data.vector2, data.vector3)

#You can also try:
#Generate 5000 vectors with 1000 random values from an Uniform distribution U(0,0.25).
data.vectors&lt;-sapply(1:5000, function (x) runif(1000, 0, 0.25))
#Set data class to "multiple".
class(data.vectors)&lt;-"multiple"
#Apply the GSAfisher.multiple function to these data vectors.
GSAfisher(data.vectors)
</code></pre>

<hr>
<h2 id='GSAfisher.plot'>GSAfisher.plot</h2><span id='topic+GSAfisher.plot'></span>

<h3>Description</h3>

<p>Plot method of GSAfisher function that plots the p-values for a joint test of association between a phenotype and multiple gene-sets of genetic variants (SNPs) by combining multiple marginal p-values data vectors using the Fisher method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'plot'
GSAfisher(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSAfisher.plot_+3A_...">...</code></td>
<td>
<p>Set of multiple marginal p-values data vectors or matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>plot.p.multiple</code></td>
<td>
<p>Plot of a set of multiple Fisher method combined p-values for each data vector.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Carlos Garcia Prieto</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Generate 3 vectors with 1000 random values from an Uniform distribution U(0, 0.25). 
data.vector1&lt;-runif(1000, 0, 0.25)
data.vector2&lt;-runif(1000, 0, 0.25)
data.vector3&lt;-runif(1000, 0, 0.25)
#Set data class to "plot".
class(data.vector1)&lt;-"plot"
#Apply the GSAfisher.plot function to these data vectors.
GSAfisher(data.vector1, data.vector2, data.vector3)

#You can also try:
#Generate 5000 vectors with 1000 random values from an Uniform distribution U(0,0.25).
data.vectors&lt;-sapply(1:5000, function (x) runif(1000, 0, 0.25))
#Set data class to "plot".
class(data.vectors)&lt;-"plot"
#Apply the GSAfisher.plot function to these data vectors.
GSAfisher(data.vectors)
</code></pre>

<hr>
<h2 id='GSAfisher.print'>GSAfisher.print</h2><span id='topic+GSAfisher.print'></span>

<h3>Description</h3>

<p>Print method of GSAfisher function that computes the p-value for a joint test of association between a phenotype and a set of genetic variants (SNPs) by combining marginal p-values using the Fisher method and prints the result.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'print'
GSAfisher(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSAfisher.print_+3A_x">x</code></td>
<td>
<p>Numeric set of marginal p-values (one vector or a single p-value).</p>
</td></tr>
<tr><td><code id="GSAfisher.print_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>print.p</code></td>
<td>
<p>Displays Fisher method combined pvalue.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Carlos Garcia Prieto
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Generate 1000 random values from an Uniform distribution U(0, 0.25). 
x&lt;-runif(1000, 0, 0.25)
#Set data class to "print".
class(x)&lt;-"print"
#Apply the GSAfisher.print function to this data vector.
GSAfisher(x)
</code></pre>

<hr>
<h2 id='GSAfisher.summary'>GSAfisher.summary</h2><span id='topic+GSAfisher.summary'></span>

<h3>Description</h3>

<p>Summary method of GSAfisher function that computes a numerical summary of the marginal p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'summary'
GSAfisher(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSAfisher.summary_+3A_x">x</code></td>
<td>
<p>Numeric set of marginal p-values (one vector or a sinlge p-value).</p>
</td></tr>
<tr><td><code id="GSAfisher.summary_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>summary.p</code></td>
<td>
<p>Summary statistics (min, 1st quantile, median, 3rd quantile, max) of a numerical set of marginal p-values.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Carlos Garcia Prieto</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Generate 1000 random values from an Uniform distribution U (0, 0.25). 
x&lt;-runif(1000, 0, 0.25)
#Set data class to "summary"
class(x)&lt;-"summary"
#Apply the GSAfisher.summary function to this data vector.
GSAfisher(x)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
