<!DOCTYPE html><html><head><title>Help for package gsalib</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gsalib}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gsa.read.gatkreport'>
<p>Function to read in a GATKReport</p></a></li>
<li><a href='#gsa.read.gatkreportv0'>
<p>Function to read in an old-style GATKReport</p></a></li>
<li><a href='#gsa.read.gatkreportv1'>
<p>Function to read in a new-style GATKReport</p></a></li>
<li><a href='#gsa.reshape.concordance.table'>
<p>Reshape a Concordance Table</p></a></li>
<li><a href='#gsalib-internal'><p>Internal gsalib objects</p></a></li>
<li><a href='#gsalib-package'>
<p>Utility functions for GATK</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Utility Functions for 'GATK'</td>
</tr>
<tr>
<td>Version:</td>
<td>2.2.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-10-07</td>
</tr>
<tr>
<td>Author:</td>
<td>Kiran Garimella</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Louis Bergelson &lt;louisb@broadinstitute.org&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides utility functions used by the Genome Analysis Toolkit ('GATK') to load tables and plot data. The 'GATK' is a toolkit for variant discovery in high-throughput sequencing data.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://gatk.broadinstitute.org/hc/en-us">https://gatk.broadinstitute.org/hc/en-us</a>,
<a href="https://github.com/broadinstitute/gatk">https://github.com/broadinstitute/gatk</a>,
<a href="https://github.com/broadinstitute/gsalib/">https://github.com/broadinstitute/gsalib/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/broadinstitute/gsalib/issues">https://github.com/broadinstitute/gsalib/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-07 22:50:05 UTC</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-07 18:48:27 UTC; louisb</td>
</tr>
</table>
<hr>
<h2 id='gsa.read.gatkreport'>
Function to read in a GATKReport
</h2><span id='topic+gsa.read.gatkreport'></span>

<h3>Description</h3>

<p>This function reads in data from a GATKReport. A GATKReport is a document containing multiple tables produced by the GATK. Each table is loaded as a separate data.frame object in a list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsa.read.gatkreport(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gsa.read.gatkreport_+3A_filename">filename</code></td>
<td>

<p>The path to the GATKReport file.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The GATKReport format replaces the multi-file output format used previously by many GATK tools and provides a single, consolidated file format.  This format accommodates multiple tables and is still R-loadable through this function. 
</p>


<h3>Value</h3>

<p>Returns a LIST object, where each key is the TableName and the value is the data.frame object with the contents of the table.  If multiple tables with the same name exist, each one after the first will be given names of TableName.v1, TableName.v2, ..., TableName.vN.
</p>


<h3>Note</h3>

<p>This function accepts different versions of the GATKReport format by making internal calls to gsa.read.gatkreportv0() or gsa.read.gatkreportv1() as appropriate.
</p>


<h3>Author(s)</h3>

<p>Kiran Garimella
</p>


<h3>References</h3>

<p>https://gatk.broadinstitute.org/hc/en-us/articles/360035532172-GATKReport-and-gsalib
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test_file = system.file("extdata", "test_gatkreport.table", package = "gsalib");
report = gsa.read.gatkreport(test_file);
</code></pre>

<hr>
<h2 id='gsa.read.gatkreportv0'>
Function to read in an old-style GATKReport
</h2><span id='topic+gsa.read.gatkreportv0'></span>

<h3>Description</h3>

<p>This function reads in data from a version 0.x GATKReport. It should not be called directly; instead, use gsa.read.gatkreport()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsa.read.gatkreportv0(lines)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gsa.read.gatkreportv0_+3A_lines">lines</code></td>
<td>

<p>The lines read in from the input file.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a LIST object, where each key is the TableName and the value is the data.frame object with the contents of the table.  If multiple tables with the same name exist, each one after the first will be given names of TableName.v1, TableName.v2, ..., TableName.vN.
</p>


<h3>Author(s)</h3>

<p>Kiran Garimella
</p>


<h3>References</h3>

<p>https://gatk.broadinstitute.org/hc/en-us/articles/360035532172-GATKReport-and-gsalib
</p>

<hr>
<h2 id='gsa.read.gatkreportv1'>
Function to read in a new-style GATKReport
</h2><span id='topic+gsa.read.gatkreportv1'></span>

<h3>Description</h3>

<p>This function reads in data from a version 1.x GATKReport. It should not be called directly; instead, use gsa.read.gatkreport()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsa.read.gatkreportv1(lines)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gsa.read.gatkreportv1_+3A_lines">lines</code></td>
<td>

<p>The lines read in from the input file.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a LIST object, where each key is the TableName and the value is the data.frame object with the contents of the table.  If multiple tables with the same name exist, each one after the first will be given names of TableName.v1, TableName.v2, ..., TableName.vN.
</p>


<h3>Author(s)</h3>

<p>Kiran Garimella
</p>


<h3>References</h3>

<p>https://gatk.broadinstitute.org/hc/en-us/articles/360035532172-GATKReport-and-gsalib
</p>

<hr>
<h2 id='gsa.reshape.concordance.table'>
Reshape a Concordance Table
</h2><span id='topic+gsa.reshape.concordance.table'></span>

<h3>Description</h3>

<p>Given a GATKReport generated by GenotypeConcordance (as output by <code>gsa.read.gatkreport</code>), this function reshapes the concordance for a specified sample into a matrix with the EvalGenotypes in rows and the CompGenotypes in columns (see the documentation for GenotypeConcordance for the definition of Eval and Comp)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsa.reshape.concordance.table(
  report, 
  table.name="GenotypeConcordance_Counts", 
  sample.name="ALL")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gsa.reshape.concordance.table_+3A_report">report</code></td>
<td>

<p>A GATKReport as output by <code>gsa.read.gatkreport</code>.  If <code>table.name</code> is <code>NULL</code>, <code>report</code> is assumed to be the vector of concordance values to reshape.
</p>
</td></tr>
<tr><td><code id="gsa.reshape.concordance.table_+3A_table.name">table.name</code></td>
<td>

<p>The table name in the GATKReport to reshape.  Defaults to &quot;GenotypeConcordance_Counts&quot;, but could also be one of the proportion tables (&quot;GenotypeConcordance_EvalProportions&quot;, &quot;GenotypeConcordance_CompProportions&quot;).  This value can also be <code>NULL</code>, in which case <code>report</code> is reshaped directly.  
</p>
</td></tr>
<tr><td><code id="gsa.reshape.concordance.table_+3A_sample.name">sample.name</code></td>
<td>

<p>The sample name within <code>table.name</code> to use.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a two-dimensional matrix with Eval genotypes in the rows and Comp genotypes in the columns.  The genotypes themselves (<code>HOM_REF</code>, <code>NO_CALL</code>, etc) are specified in the row/col names of the matrix.
</p>


<h3>Author(s)</h3>

<p>Phillip Dexheimer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gsa.read.gatkreport">gsa.read.gatkreport</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test_file = system.file("extdata", "test_genconcord.table", package = "gsalib")
report = gsa.read.gatkreport(test_file)
gsa.reshape.concordance.table(report)

## Output looks like:
##              CompGenotypes
##EvalGenotypes NO_CALL HOM_REF HET HOM_VAR UNAVAILABLE MIXED
##  NO_CALL     0       0       0       0       0           0    
##  HOM_REF     0       0       0       0       0           0    
##  HET         0       0       13463   90      3901        0    
##  HOM_VAR     0       0       2935    18144   4448        0    
##  UNAVAILABLE 0       0       2053693 1326112 11290       0    
##  MIXED       0       0       0       0       0           0  
</code></pre>

<hr>
<h2 id='gsalib-internal'>Internal gsalib objects</h2><span id='topic+.gsa.assignGATKTableToEnvironment'></span><span id='topic+.gsa.splitFixedWidth'></span>

<h3>Description</h3>

<p>Internal gsalib objects.</p>


<h3>Details</h3>

<p>These are not to be called by the user.</p>

<hr>
<h2 id='gsalib-package'>
Utility functions for GATK
</h2><span id='topic+gsalib-package'></span><span id='topic+gsalib'></span>

<h3>Description</h3>

<p>Utility functions for analysis of genome sequence data with the GATK
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> gsalib</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2015-03-17</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> MIT</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>This package is primarily meant to be used programmatically by GATK tools. However the gsa.read.gatkreport() function can be used to easily read in data from a GATKReport. A GATKReport is a multi-table document generated by GATK tools. 
</p>


<h3>Author(s)</h3>

<p>Kiran Garimella
</p>
<p>Maintainer: Louis Bergelson &lt;louisb@broadinstitute.org&gt;
</p>


<h3>References</h3>

<p>https://gatk.broadinstitute.org/hc/en-us/articles/360035532172-GATKReport-and-gsalib
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test_file = system.file("inst", "extdata", "test_gatkreport.table", package = "gsalib");
report = gsa.read.gatkreport(test_file);
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
