<!DOCTYPE html><html><head><title>Help for package GSAQ</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GSAQ}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#genedist'><p>Chromosomal distribution of the genes in the selected geneset</p></a></li>
<li><a href='#genelist'>
<p>A list of genes of rice</p></a></li>
<li><a href='#geneqtl'><p>List of the selected genes along with their corresponding overlapped QTL</p></a></li>
<li><a href='#GeneSelect'><p>Selection of informative geneset</p></a></li>
<li><a href='#GSAQ'><p>Gene Set Analysis with Quantitative Trait Loci with gene sampling model</p></a></li>
<li><a href='#GSVQ'><p>Gene Set Validation with QTL using Hyper-geometric test without gene sampling model</p></a></li>
<li><a href='#qtl_salt'>
<p>A list of salt responsive Quantitative Trait Loci of rice</p></a></li>
<li><a href='#qtldist'><p>QTL wise distribution of genes in the selected geneset</p></a></li>
<li><a href='#qtlhit'><p>Computation of qtl-hit statistic for the selected gene set</p></a></li>
<li><a href='#rice_salt'>
<p>Gene expression data of rice under salinity stress</p></a></li>
<li><a href='#totqtlhit'><p>Computation of total number of qtl-hits found in the whole gene space</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Set Analysis with QTL</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-08-26</td>
</tr>
<tr>
<td>Author:</td>
<td>Samarendra Das &lt;samarendra.das@icar.gov.in&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Samarendra Das &lt;samarendra.das@icar.gov.in&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.1)</td>
</tr>
<tr>
<td>Description:</td>
<td>Computation of Quantitative Trait Loci hits in the selected gene set. Performing gene set validation with Quantitative Trait Loci information. Performing gene set enrichment analysis with available Quantitative Trait Loci data and computation of statistical significance value from gene set analysis. Obtaining the list of Quantitative Trait Loci hit genes along with their overlapped Quantitative Trait Loci names.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-09-08 09:08:07 UTC; samarendra</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-09-08 13:07:11</td>
</tr>
</table>
<hr>
<h2 id='genedist'>Chromosomal distribution of the genes in the selected geneset</h2><span id='topic+genedist'></span>

<h3>Description</h3>

<p>The function computes the chromosome wise distribution of the genes in the selected geneset and also plots the chromosomal distribution.</p>


<h3>Usage</h3>

<pre><code class='language-R'>genedist(geneset, genelist, plot)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genedist_+3A_geneset">geneset</code></td>
<td>
<p>geneset is a vector of characters representing the names of genes/ gene ids selected from the whole gene list by using a gene selection method.</p>
</td></tr>
<tr><td><code id="genedist_+3A_genelist">genelist</code></td>
<td>
<p>genelist is a N by 3 dataframe/ matrix (genes/gene ids as row names); where, N represents the number of genes in the whole gene space: first coloumn represnting the chromosomal location of genes: second coloumn representing the start position of genes in terms of basepairs: third coloumn representing the end position of genes in terms of basepairs in their respective chromosomes.</p>
</td></tr>
<tr><td><code id="genedist_+3A_plot">plot</code></td>
<td>
<p>plot is a character string indicating whether the chromosomal distribution of the genes in the selected geneset will be plotted or not. It can be either TRUE/FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns the chromosomal distribution of the genes in the selected geneset.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
data(genelist)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
geneset= GeneSelect(x, y, s=50, method="t-score")$selectgenes
genelist=as.data.frame(genelist)
genedist(geneset, genelist, plot=TRUE)
</code></pre>

<hr>
<h2 id='genelist'>
A list of genes of rice  
</h2><span id='topic+genelist'></span>

<h3>Description</h3>

<p>This data is in form of a 200 by 3 dataframe with genes/gene ids as rownames. The first column represents the chromosomal location of the genes (chromosome number). The second coloumn represents start position of the genes in terms of basepairs (bps) and the third coloumn represents end position of genes in terms of basepairs (bps) in their respective chromosomes. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("genelist")</code></pre>


<h3>Format</h3>

<p>A data frame with 200 rows as genes and the columns represent the chromosomal locations, start positions and end positions of respective genes.
</p>

<dl>
<dt><code>Chr</code></dt><dd><p>chr represents the chromosomal location of the genes</p>
</dd>
<dt><code>Start</code></dt><dd><p>start represents the start position of the genes in their respective chromosomes</p>
</dd>
<dt><code>End</code></dt><dd><p>End represents the end position of the genes in their respective chromosomes</p>
</dd>
</dl>



<h3>Details</h3>

<p>The data is created by taking 200 genes from the large number of genes from NCBI GEO database. The genomic location of the genes on the rice genome are obtained from MSU Rice Genome Annotation (Osa1).</p>


<h3>Source</h3>

<p>Gene Expression Omnibus: NCBI gene expression and hybridization array data repository.ncbi.nlm.nih.gov/geo/.
Ouyang S, Zhu W, Hamilton J, Lin H, Campbell M, et al. (2007) The TIGR Rice Genome Annotation Resource: improvements and new features. Nucleic Acids Research 35.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genelist)
</code></pre>

<hr>
<h2 id='geneqtl'>List of the selected genes along with their corresponding overlapped QTL</h2><span id='topic+geneqtl'></span>

<h3>Description</h3>

<p>The function enables to obtain list of the selected genes along with the corresponding overlapped Quantitative Trait Loci (QTL) ids/names along with their genomic positions.</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneqtl(geneset, genelist, qtl)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geneqtl_+3A_geneset">geneset</code></td>
<td>
<p>geneset is a vector of characters representing the names of genes/ gene ids selected from the whole gene list/space by using a gene selection method.</p>
</td></tr>
<tr><td><code id="geneqtl_+3A_genelist">genelist</code></td>
<td>
<p>genelist is a N by 3 dataframe/ matrix (genes/gene ids as row names); where, N represents the number of genes in the whole gene set: first coloumn represnting the chromosomal location of genes: second coloumn representing the start position of genes in terms of basepairs: third coloumn representing the end position of genes in terms of basepairs in their respective chromosomes.</p>
</td></tr>
<tr><td><code id="geneqtl_+3A_qtl">qtl</code></td>
<td>
<p>qtl is a Q by 3 dataframe/matrix (qtl names/qtl ids as row names);where, Q represents the number of qtls: first coloumn represnting the chromosomal location of qtls: second coloumn representing the start position of qtls in terms of basepairs: third coloumn representing the end position of qtls in terms of basepairs in their respective chromosomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a list with two components. First component returns the list of selected genes along with their overlapped QTL ids/names. Second component gives the list of selected genes with their overlapped QTL ids/names and their respective genomic positions.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
data(genelist)
data(qtl_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
genelist=as.data.frame(genelist)
qtl=as.data.frame(qtl_salt)
geneset= GeneSelect(x, y, s=50, method="t-score")$selectgenes
geneqtl(geneset, genelist, qtl)

</code></pre>

<hr>
<h2 id='GeneSelect'>Selection of informative geneset</h2><span id='topic+GeneSelect'></span>

<h3>Description</h3>

<p>The function returns the informative geneset from the high dimensional gene expression data using a proper statistical technique.</p>


<h3>Usage</h3>

<pre><code class='language-R'>GeneSelect(x, y, s, method)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GeneSelect_+3A_x">x</code></td>
<td>
<p>x is a N x m gene expression data matrix (must be data frame) and row names as gene names, where, N represents the number of genes in the whole gene space and m is number of samples.</p>
</td></tr>
<tr><td><code id="GeneSelect_+3A_y">y</code></td>
<td>
<p>y is a m by 1 vector representing the sample labels, is according to the different stress conditions for two class problem (must be 1: stress/-1: control)</p>
</td></tr>
<tr><td><code id="GeneSelect_+3A_s">s</code></td>
<td>
<p>s is a numeric constant representing the number of genes to be selected from the large pool of genes/ gene space.</p>
</td></tr>
<tr><td><code id="GeneSelect_+3A_method">method</code></td>
<td>
<p>method is a character string indicating which method for informative gene selection is to be used. One of method &quot;t-score&quot; (default), &quot;F-score&quot;, &quot;MRMR&quot;, &quot;BootMRMR&quot;  can be abbreviated and used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns the informative geneset using a particular method from the high dimensional gene expression data.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
GeneSelect(x, y, s=50, method="t-score")$selectgenes
</code></pre>

<hr>
<h2 id='GSAQ'>Gene Set Analysis with Quantitative Trait Loci with gene sampling model</h2><span id='topic+GSAQ'></span>

<h3>Description</h3>

<p>The function computes the statistical significance value (p-value) from gene set analysis test with QTL for the test H0: Genes in the selected geneset are at most as often overlapped with the QTL regions as the genes in not selected geneset; against H1: Genes in the geneset are more often overlapped with the QTL regions as compared to genes in not selected geneset.</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSAQ(geneset, genelist, qtl, SampleSize, K, method)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSAQ_+3A_geneset">geneset</code></td>
<td>
<p>geneset is a vector of characters representing the names of genes/ gene ids selected from the whole gene list by using a gene selection method.</p>
</td></tr>
<tr><td><code id="GSAQ_+3A_genelist">genelist</code></td>
<td>
<p>genelist is a N by 3 dataframe/ matrix (genes/gene ids as row names); where, N represents the number of genes in the whole gene space: first coloumn represnting the chromosomal location of genes: second coloumn representing the start position of genes in terms of basepairs: third coloumn representing the end position of genes in terms of basepairs in their respective chromosomes.</p>
</td></tr>
<tr><td><code id="GSAQ_+3A_qtl">qtl</code></td>
<td>
<p>qtl is a Q by 3 dataframe/matrix (qtl names/qtl ids as row names);where, Q represents the number of qtls: first coloumn represnting the chromosomal location of qtls: second coloumn representing the start position of qtls in terms of basepairs: third coloumn representing the end position of qtls in terms of basepairs in their respective chromosomes.</p>
</td></tr>
<tr><td><code id="GSAQ_+3A_samplesize">SampleSize</code></td>
<td>
<p>SampleSize is a numeric constant representing the size of the gene sample drawn from the geneset using the gene sampling model (SampleSize must be less than the size of geneset).</p>
</td></tr>
<tr><td><code id="GSAQ_+3A_k">K</code></td>
<td>
<p>K is a numeric constant representing the number of gene samples of size equal to SampleSize will be drawn by the using gene sampling model.</p>
</td></tr>
<tr><td><code id="GSAQ_+3A_method">method</code></td>
<td>
<p>method is a character string indicating which method for final p-value (combining p-values for various gene samples) is to be computed. One of &quot;meanp&quot;, &quot;sump&quot;, &quot;logit&quot;, &quot;sumz&quot;or &quot;logp&quot; (default) can be abbreviated and used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns the final statistical significance value (p-value) from Gene set Analysis with QTL test.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
data(genelist)
data(qtl_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
genelist=as.data.frame(genelist)
qtl=as.data.frame(qtl_salt)
geneset= GeneSelect(x, y, s=50, method="t-score")$selectgenes
GSAQ(geneset, genelist, qtl, SampleSize=30, K=50, method="meanp")
</code></pre>

<hr>
<h2 id='GSVQ'>Gene Set Validation with QTL using Hyper-geometric test without gene sampling model</h2><span id='topic+GSVQ'></span>

<h3>Description</h3>

<p>The function computes ths statisical significance value (p-value) for gene set validation using hypergeometric test.</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSVQ(geneset, genelist, qtl)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSVQ_+3A_geneset">geneset</code></td>
<td>
<p>geneset is a vector of characters representing the names of genes/ gene ids selected from the whole gene list by using a gene selection method.</p>
</td></tr>
<tr><td><code id="GSVQ_+3A_genelist">genelist</code></td>
<td>
<p>genelist is a N by 3 dataframe/ matrix (genes/gene ids as row names): where, N represents the number of genes in the whole gene set: first coloumn represnting the chromosomal location of genes: second coloumn representing the start position of genes in terms of basepairs: third coloumn representing the end position of genes in terms of basepairs.</p>
</td></tr>
<tr><td><code id="GSVQ_+3A_qtl">qtl</code></td>
<td>
<p>qtl is a Q by 3 dataframe/matrix (qtl names/qtl ids as row names);where, Q represents the number of qtls: first coloumn represnting the chromosomal location of qtls: second coloumn representing the start position of qtls in terms of basepairs: third coloumn representing the end position of qtls in terms of basepairs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns the statisical significance value (p-value) from Hyper-geometric test for validation of the selected gene set with qtl data.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
data(genelist)
data(qtl_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
geneset= GeneSelect(x, y, s=50, method="t-score")$selectgenes
genelist=as.data.frame(genelist)
qtl=as.data.frame(qtl_salt)
GSVQ(geneset, genelist, qtl)
</code></pre>

<hr>
<h2 id='qtl_salt'>
A list of salt responsive Quantitative Trait Loci of rice  
</h2><span id='topic+qtl_salt'></span>

<h3>Description</h3>

<p>This data is in form of a 13 by 3 dataframe with qtls/qtl ids as rownames. The first column reoresents the chromosomal location of the respective qtls (chromosome number). The second coloumn represents start position of the qtls in terms of basepairs (bps) and the third coloumn represents end position of qtls in terms of basepairs (bps) in their respective chromosomes. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("qtl_salt")</code></pre>


<h3>Format</h3>

<p>A data frame with 13 rows as qtl and the columns represent the chromosomal locations, start positions and end positions of respective qtls.
</p>

<dl>
<dt><code>Chr</code></dt><dd><p>chr represents the chromosomal location of the qtls</p>
</dd>
<dt><code>Start</code></dt><dd><p>start represents the start position of the qtls in their respective chromosomes</p>
</dd>
<dt><code>End</code></dt><dd><p>End represents the end position of the qtls in their respective chromosomes</p>
</dd>
</dl>



<h3>Details</h3>

<p>The data is created by taking 13 unique salt responsive qtls from the Gramene QTL database. The genomic locations of these QTLs on rice genome are obtained using Gramene annotation of MSU Rice Genome Annotation (Osa1).</p>


<h3>Source</h3>

<p>Gramene QTL library (http://www.gramene.org/qtl/).
Ouyang S, Zhu W, Hamilton J, Lin H, Campbell M, et al. (2007) The TIGR Rice Genome Annotation Resource: improvements and new features. Nucleic Acids Research 35: D883-D887.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(qtl_salt)
</code></pre>

<hr>
<h2 id='qtldist'>QTL wise distribution of genes in the selected geneset</h2><span id='topic+qtldist'></span>

<h3>Description</h3>

<p>Computation of number of qtl-hit genes in each QTL and also QTL wise distribution  of genes in the selected geneset </p>


<h3>Usage</h3>

<pre><code class='language-R'>qtldist(geneset, genelist, qtl, plot)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qtldist_+3A_geneset">geneset</code></td>
<td>
<p>geneset is a vector of characters representing the names of genes/ gene ids selected from the whole gene space by using a gene selection method.</p>
</td></tr>
<tr><td><code id="qtldist_+3A_genelist">genelist</code></td>
<td>
<p>genelist is a N by 3 dataframe/ matrix (genes/gene ids as row names); where, N represents the number of genes in the whole gene space: first coloumn represnting the chromosomal location of genes: second coloumn representing the start position of genes in terms of basepairs: third coloumn representing the end position of genes in terms of basepairs in their respective chromosomes.</p>
</td></tr>
<tr><td><code id="qtldist_+3A_qtl">qtl</code></td>
<td>
<p>qtl is a Q by 3 dataframe/matrix (qtl names/qtl ids as row names);where, Q represents the number of qtls: first coloumn represnting the chromosomal location of qtls: second coloumn representing the start position of qtls in terms of basepairs: third coloumn representing the end position of qtls in terms of basepairs in their respective chromosomes.</p>
</td></tr>
<tr><td><code id="qtldist_+3A_plot">plot</code></td>
<td>
<p>plot is a character string used to plot the QTL wise distribution of genes in the selected gene set. It can be either TRUE/FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns number of qtl-hit genes in each QTL and QTL wise distribution of the selected genes.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
data(genelist)
data(qtl_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
geneset= GeneSelect(x, y, s=50, method="t-score")$selectgenes
genelist=as.data.frame(genelist)
qtl=as.data.frame(qtl_salt)
qtldist(geneset, genelist, qtl, plot=TRUE)
</code></pre>

<hr>
<h2 id='qtlhit'>Computation of qtl-hit statistic for the selected gene set</h2><span id='topic+qtlhit'></span>

<h3>Description</h3>

<p>The function computes the statistic, i.e.  number of qtl-hit genes in the selected gene set.</p>


<h3>Usage</h3>

<pre><code class='language-R'>qtlhit(geneset, genelist, qtl)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qtlhit_+3A_geneset">geneset</code></td>
<td>
<p>geneset is a vector of characters representing the names of genes/ gene ids selected from the whole gene list by using a gene selection method.</p>
</td></tr>
<tr><td><code id="qtlhit_+3A_genelist">genelist</code></td>
<td>
<p>genelist is a N by 3 dataframe/ matrix (genes/gene ids as row names); where, N represents the number of genes in the whole gene set: first coloumn represnting the chromosomal location of genes: second coloumn representing the start position of genes in terms of basepairs: third coloumn representing the end position of genes in terms of basepairs.</p>
</td></tr>
<tr><td><code id="qtlhit_+3A_qtl">qtl</code></td>
<td>
<p>qtl is a Q by 3 dataframe/matrix (qtl names/qtl ids as row names);where, Q represents the number of qtls: first coloumn represnting the chromosomal location of qtls: second coloumn representing the start position of qtls in terms of basepairs: third coloumn representing the end position of qtls in terms of basepairs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a numeric value of the statistic 'qtl-hit' representing the number of qtl-hits by the genes in the selected gene set.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
data(genelist)
data(qtl_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
geneset= GeneSelect(x, y, s=50, method="t-score")$selectgenes
genelist=as.data.frame(genelist)
qtl=as.data.frame(qtl_salt)
qtlhit(geneset, genelist, qtl)
</code></pre>

<hr>
<h2 id='rice_salt'>
Gene expression data of rice under salinity stress
</h2><span id='topic+rice_salt'></span>

<h3>Description</h3>

<p>This data has gene expression values of 200 genes over 40 microarray samples/subjects for a salinity vs. control study in rice. These 40 samples belong to either of salinity stress or control condition (two class problem). This gene expression data is balanced type as the first 20 samples are under salinity stress and the later 20 are under control condition.
The first row of the data contains the samples/subjects labels with entries are 1 and -1, where the labels '1' and '-1' represent samples generated under salinity stress and control condition respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("rice_salt")</code></pre>


<h3>Format</h3>

<p>A data frame with 200 genes over 40 microarray samples/subjects.
</p>


<h3>Details</h3>

<p>The data is created by taking 200 genes from the large number of genes from NCBI GEO database. The rows are the genes and columns are the samples/subjects. The first half of the samples/subjects are generated under salinity stress condition and other half under control condition.The first row of the data contains the samples/subjects labels with entries as 1 and -1, where th label '1' and '-1' represents sample generated under salinity stress and control condition respectively.</p>


<h3>Source</h3>

<p>Gene Expression Omnibus: NCBI gene expression and hybridization array data repository.ncbi.nlm.nih.gov/geo/.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rice_salt)
</code></pre>

<hr>
<h2 id='totqtlhit'>Computation of total number of qtl-hits found in the whole gene space</h2><span id='topic+totqtlhit'></span>

<h3>Description</h3>

<p>It enable to Compute the total number qtl-hits found in the whole gene space or in the micro-array chip</p>


<h3>Usage</h3>

<pre><code class='language-R'>totqtlhit(genelist, qtl)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="totqtlhit_+3A_genelist">genelist</code></td>
<td>
<p>genelist is a N by 3 dataframe/ matrix (genes/gene ids as row names); where, N represents the number of genes in the whole gene set: first coloumn represnting the chromosomal location of genes: second coloumn representing the start position of genes in terms of basepairs: third coloumn representing the end position of genes in terms of basepairs.</p>
</td></tr>
<tr><td><code id="totqtlhit_+3A_qtl">qtl</code></td>
<td>
<p>qtl is a Q by 3 dataframe/matrix (qtl names/qtl ids as row names);where, Q represents the number of qtls: first coloumn represnting the chromosomal location of qtls: second coloumn representing the start position of qtls in terms of basepairs: third coloumn representing the end position of qtls in terms of basepairs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a numeric value representing the total number of qtl-hits found in the whole gene list or in a micro-array chip.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genelist)
data(qtl_salt)
genelist=as.data.frame(genelist)
qtl=as.data.frame(qtl_salt)
totqtlhit(genelist, qtl)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
