<!DOCTYPE html><html><head><title>Help for package GSDA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GSDA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GSDA-package'>
<p>Gene Set Distance Analysis (GSDA)</p></a></li>
<li><a href='#best.dist.corr'><p>Best Distance Correlattion</p></a></li>
<li><a href='#cat.dist'>
<p>Distance for a Categorical Variable</p></a></li>
<li><a href='#dist.corr'>
<p>Distance Correlattion</p></a></li>
<li><a href='#gsda'>
<p>Gene-Set Distance Analsysis (GSDA)</p></a></li>
<li><a href='#kegg.ml.gsets'>
<p>KEGG gene set data for the AML and CML pathways</p></a></li>
<li><a href='#prep.gsda'>
<p>GSDA Data Preparation</p></a></li>
<li><a href='#prep.msigdb'>
<p>Preparation of MSigDB for GSDA</p></a></li>
<li><a href='#print.bdc'>
<p>Print Method for Best Distance Correlation</p></a></li>
<li><a href='#print.dcor'>
<p>Print Method for Distance Correlation</p></a></li>
<li><a href='#print.gsda.result'>
<p>Print Method for GSDA Result</p></a></li>
<li><a href='#surv.dist'>
<p>Distance of a Survival Endpoint</p></a></li>
<li><a href='#target.aml.clin'><p>Clinical outcomes for AML TARGET Project</p></a></li>
<li><a href='#target.aml.expr'>
<p>RNA-seq expression from the AML TARGET project</p></a></li>
<li><a href='#U.center'><p>U Centering</p></a></li>
<li><a href='#uc.dist'>
<p>U-centered Distance Matrix</p></a></li>
<li><a href='#write.gsda.csv.file'>
<p>Write GSDA Result to a Comma Delimited File</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Set Distance Analysis (GSDA)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-01-014</td>
</tr>
<tr>
<td>Description:</td>
<td>The gene-set distance analysis of omic data is implemented by generalizing distance correlations to evaluate the association of a gene set with categorical and censored event-time variables.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0),msigdbr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Microarray, Bioinformatics, Gene expression</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-14 17:30:56 UTC; xcao12</td>
</tr>
<tr>
<td>Author:</td>
<td>Xueyuan Cao [aut, cre],
  Stanley Pounds [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xueyuan Cao &lt;xcao12@uthsc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-18 16:20:27 UTC</td>
</tr>
</table>
<hr>
<h2 id='GSDA-package'>
Gene Set Distance Analysis (GSDA)
</h2><span id='topic+GSDA-package'></span><span id='topic+GSDA'></span>

<h3>Description</h3>

<p>The gene-set distance analysis of omic data is implemented by generalizing distance correlations to evaluate the association of a gene set with categorical and censored event-time variables.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> GSDA</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Gene Set Distance Analysis (GSDA)</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2021-01-014</td>
</tr>
<tr>
 <td style="text-align: left;">
Authors@R: </td><td style="text-align: left;"> c(person("Xueyuan", "Cao", email = "xcao12@uthsc.edu", role = c("aut", "cre")),
             person("Stanley", "Pounds", email = "stanley.pounds@stjude.org", role = c("aut"))) </td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> The gene-set distance analysis of omic data is implemented by generalizing distance correlations to evaluate the association of a gene set with categorical and censored event-time variables.</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R (&gt;= 3.5.0),msigdbr</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
biocViews: </td><td style="text-align: left;"> Microarray, Bioinformatics, Gene expression</td>
</tr>
<tr>
 <td style="text-align: left;">
Suggests: </td><td style="text-align: left;"> knitr, rmarkdown</td>
</tr>
<tr>
 <td style="text-align: left;">
VignetteBuilder: </td><td style="text-align: left;"> knitr</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Xueyuan Cao [aut, cre],
  Stanley Pounds [aut]</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Xueyuan Cao &lt;xcao12@uthsc.edu&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<p>Index of help topics:
</p>
<pre>
GSDA-package            Gene Set Distance Analysis (GSDA)
U.center                U Centering
best.dist.corr          Best Distance Correlattion
cat.dist                Distance for a Categorical Variable
dist.corr               Distance Correlattion
gsda                    Gene-Set Distance Analsysis (GSDA)
kegg.ml.gsets           KEGG gene set data for the AML and CML pathways
prep.gsda               GSDA Data Preparation
prep.msigdb             Preparation of MSigDB for GSDA
print.bdc               Print Method for Best Distance Correlation
print.dcor              Print Method for Distance Correlation
print.gsda.result       Print Method for GSDA Result
surv.dist               Distance of a Survival Endpoint
target.aml.clin         Clinical outcomes for AML TARGET Project
target.aml.expr         RNA-seq expression from the AML TARGET project
uc.dist                 U-centered Distance Matrix
write.gsda.csv.file     Write GSDA Result to a Comma Delimited File
</pre>

<p>Further information is available in the following vignettes:<br /><br />
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>GSDA</code> </td><td style="text-align: left;"> An Introduction to the GSDA Package (source, pdf)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>NA
</p>
<p>Maintainer: NA
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(target.aml.clin)
data(target.aml.expr)
data(kegg.ml.gsets)
res=gsda(target.aml.expr,
         target.aml.clin,
         kegg.ml.gsets,
         "Chloroma","oe","ct")
</code></pre>

<hr>
<h2 id='best.dist.corr'>Best Distance Correlattion</h2><span id='topic+best.dist.corr'></span>

<h3>Description</h3>

<p>Use a backward elimination procedure to identify a subset of variables in X most strongly associated with Y according to the distance correlation p-value.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>best.dist.corr(X, Y, x.dist = "oe", y.dist = "oe")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="best.dist.corr_+3A_x">X</code></td>
<td>
<p>The omic numeric data matrix with subjects in rows and variables in columns.  Note that this is the TRANSPOSE of the omic data matrix for some other omic data analysis packages and for the gsda function of this package.</p>
</td></tr>
<tr><td><code id="best.dist.corr_+3A_y">Y</code></td>
<td>
<p>Numeric data matrix, vector, or data.frame with each row representing a subject.  The function assumes the same set of subjects are represent in the same order in X and Y.</p>
</td></tr>
<tr><td><code id="best.dist.corr_+3A_x.dist">x.dist</code></td>
<td>
<p>The distance metric for omic data (X), may be &quot;oe&quot; (overall Euclidean), &quot;me&quot; (marginal Euclidean), &quot;om&quot; (overall Manhattan), or &quot;mm&quot; (marginal Manhattan).</p>
</td></tr>
<tr><td><code id="best.dist.corr_+3A_y.dist">y.dist</code></td>
<td>
<p>The distance metric for clinical data, may be &quot;oe&quot; (overall Euclidean), &quot;me&quot; (marginal Euclidean), &quot;om&quot; (overall Manhattan), or &quot;mm&quot; (marginal Manhattan), same options as for X</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes dist.corr for X and Y.  It then determines which column of X may be dropped to give the smallest p-value in dist.corr.  This process is repeated until X has been reduced to only one variable.  In this way, a dist.corr p-value is obtained after dropping each X variable.  The subset of X variables giving the smallest p-value in this series of analyses is returned with additional result details.</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>rX</code></td>
<td>
<p>reduced X matrix</p>
</td></tr>
<tr><td><code>best.res</code></td>
<td>
<p>best result by backward elimination</p>
</td></tr>
<tr><td><code>all.res</code></td>
<td>
<p>all backward elimination results: the first column has the index of the column of X that was dropped; the second column has the negative log10 p-value of the resulting X matrix</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>echoes input X</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>echoes input Y</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dist.corr">dist.corr</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(target.aml.clin)
data(target.aml.expr)
target.aml.expr=sqrt(target.aml.expr)
target.aml.expr=t(target.aml.expr)

bdc.chl=best.dist.corr(target.aml.expr,
                       target.aml.clin$Chloroma,
                       "oe","ct")
</code></pre>

<hr>
<h2 id='cat.dist'>
Distance for a Categorical Variable
</h2><span id='topic+cat.dist'></span>

<h3>Description</h3>

<p>A function to calculate the distance for a categorical variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cat.dist(X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cat.dist_+3A_x">X</code></td>
<td>
<p>vector of category designations</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates distance function for a categorical variable.  The result is a square n by n matrix in which entry (i,j) has value 1 if entry i and entry j of the input vector X are not equal and entry (i,j) of the result matrix has value 0 if entry i and entry j of the input vector are equal.  The distance between subject i and subject j is zero if the two subjects have the same categorical designation.  The distance between subject i and subject j is one if the two subjects do not have the same categorical designation.
</p>


<h3>Value</h3>

<p>A square matrix with each dimension equal to the length of X.
</p>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+surv.dist">surv.dist</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(target.aml.clin)
cd=cat.dist(target.aml.clin$Chloroma)
cd[1:5,1:5]
</code></pre>

<hr>
<h2 id='dist.corr'>
Distance Correlattion
</h2><span id='topic+dist.corr'></span>

<h3>Description</h3>

<p>Calculate the distance correlation for a gene set's omic data matrix with another variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist.corr(X, Y, x.dist = "me", y.dist = "me")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist.corr_+3A_x">X</code></td>
<td>
<p>The omic numeric data matrix with subjects as rows and variables as columns.  Note this is the TRANSPOSE of how some omic data analysis packages represent omic data and how the omic data is represented in the gsda function of this package.</p>
</td></tr>
<tr><td><code id="dist.corr_+3A_y">Y</code></td>
<td>
<p>Numeric data matrix, vector, or data.frame.  The rows of X and rows of Y must represent the same set of subjects in the same order.</p>
</td></tr>
<tr><td><code id="dist.corr_+3A_x.dist">x.dist</code></td>
<td>
<p>The distance metric for omic data (X), may be &quot;oe&quot; (overall Euclidean), &quot;me&quot; (marginal Euclidean), &quot;om&quot; (overall Manhattan), or &quot;mm&quot; (marginal Manhattan).</p>
</td></tr>
<tr><td><code id="dist.corr_+3A_y.dist">y.dist</code></td>
<td>
<p>The distance metric for clinical data, may be &quot;oe&quot; (overall Euclidean), &quot;me&quot; (marginal Euclidean), &quot;om&quot; (overall Manhattan), or &quot;mm&quot; (marginal Manhattan), same options as for X</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates distance matrix for X and Y using one of the four methods &quot;oe&quot; (overall Euclidean), &quot;me&quot; (marginal Euclidean), &quot;om&quot; (overall Manhattan), or &quot;mm&quot; (marginal Manhattan). Then, the distance matrices are centered by U-centering and distance correlation is calculated as the inner product of the two U-centered distance matrices over the squared of inner product of each of the two U-centered distance matrices. The distance correlation t-statistics follows a t-distribution with n*(n-3)/2 degree of freedom according to Zhu et al.(2020).
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>odCor</code></td>
<td>
<p>overall distance correlation statistic</p>
</td></tr>
<tr><td><code>t.odCor</code></td>
<td>
<p>t-stat for overall distance correlation statistic</p>
</td></tr>
<tr><td><code>p.odCor</code></td>
<td>
<p>p-value for overall distance correlation statistic</p>
</td></tr>
<tr><td><code>dCor</code></td>
<td>
<p>distance-based correlation matrix for each pair of variables.</p>
</td></tr>
<tr><td><code>t.dCor</code></td>
<td>
<p>t-stat for distance-based correlation matrix</p>
</td></tr>
<tr><td><code>p.dCor</code></td>
<td>
<p>p-value for distance-based correlation matrix</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>echo input data matrice X</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>echo input data matrice Y</p>
</td></tr>
<tr><td><code>x.dist</code></td>
<td>
<p>echo input distance metric for X</p>
</td></tr>
<tr><td><code>y.dist</code></td>
<td>
<p>echo input distance metric for Y</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>
<p>Zhu C, Yao S, Zhang X and Shao X (2020) Distance-based and RKHS-based Dependence Metrics in High Dimension. arXiv:1902.03291
</p>


<h3>See Also</h3>

<p><code><a href="#topic+best.dist.corr">best.dist.corr</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(target.aml.clin)
data(target.aml.expr)
target.aml.expr=sqrt(target.aml.expr)
target.aml.expr=t(target.aml.expr)

dc.chl=dist.corr(target.aml.expr,
                 target.aml.clin$Chloroma,
                 "oe","ct")
</code></pre>

<hr>
<h2 id='gsda'>
Gene-Set Distance Analsysis (GSDA)
</h2><span id='topic+gsda'></span>

<h3>Description</h3>

<p>This function impletements the gene-set distance analysis (GSDA) of omic data by generalizing distance correlations to evaluate the association of each of a series gene sets with numeric, categorical, and censored event-time variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsda(omic.data, clin.data, vset.data, clin.vars, omic.dist, clin.dist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gsda_+3A_omic.data">omic.data</code></td>
<td>
<p>The genomic data matrix with features as rows and subjects as columns.  The column names of omic.data are assumed to be observation identifiers.  The gsda function calls the function prep.gsda to merge omic.data (by column name) and clin.data (by the column named &quot;ID&quot;) before performing the GSDA procedure.</p>
</td></tr>
<tr><td><code id="gsda_+3A_clin.data">clin.data</code></td>
<td>
<p>A data frame of clinical data. Each row is a subject and each column is a variable.  The &quot;ID&quot; column of clin.data includes observation identifiers. The gsda function calls the function prep.gsda to merge omic.data (by column name) and clin.data (by the column named &quot;ID&quot;) before performing the GSDA procedure.  </p>
</td></tr>
<tr><td><code id="gsda_+3A_vset.data">vset.data</code></td>
<td>
<p>Variable set data.  Each row assigns a variable (column named vID) to a variable set (column named vset).</p>
</td></tr>
<tr><td><code id="gsda_+3A_clin.vars">clin.vars</code></td>
<td>
<p>Column name(s) of clinical variable(s) to be associated with the gene-sets.</p>
</td></tr>
<tr><td><code id="gsda_+3A_omic.dist">omic.dist</code></td>
<td>
<p>The distance metric for omic data, may be &quot;oe&quot; (overall Euclidean), &quot;me&quot; (marginal Euclidean), &quot;om&quot; (overall Manhattan), or &quot;mm&quot; (marginal Manhattan)</p>
</td></tr>
<tr><td><code id="gsda_+3A_clin.dist">clin.dist</code></td>
<td>
<p>The distance metric for clinical data, may be &quot;oe&quot; (overall Euclidean), &quot;me&quot; (marginal Euclidean), &quot;om&quot; (overall Manhattan), or &quot;mm&quot; (marginal Manhattan)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function performs the GSDA method described by Cao and Pounds (2020) through generalizing distance correlations to evaluate the association of a gene set with categorical and censored event-time variables. The distance matrices are centered by U-centering and distance correlation is the inner product of the two U-centered distance matrices over the squared of inner product of each of the two U-centered distance matrices. The distance correlation t-statistics asymptotically follows a t-distribution with n*(n-3)/2 degree of freedom according to Zhu et al. (2020). 
</p>


<h3>Value</h3>

<p>A data.frame with the following columns:
</p>
<table>
<tr><td><code>vset</code></td>
<td>
<p>The name of variable set (gene-set).</p>
</td></tr>
<tr><td><code>vIDs</code></td>
<td>
<p>The list of variables in the variable set (gene-set).</p>
</td></tr>
<tr><td><code>dCor</code></td>
<td>
<p>The distance association statistics for the variable set.</p>
</td></tr>
<tr><td><code>p.vset</code></td>
<td>
<p>The p-value.</p>
</td></tr>
<tr><td><code>comp.time</code></td>
<td>
<p>Computation time for each set.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>
<p>Zhu C, Yao S, Zhang X and Shao X (2020) Distance-based and RKHS-based Dependence Metrics in High Dimension. arXiv:1902.03291
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GSDA">GSDA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(target.aml.clin)
data(target.aml.expr)
data(kegg.ml.gsets)
res=gsda(target.aml.expr,
         target.aml.clin,
         kegg.ml.gsets,
         "Chloroma","oe","ct")
</code></pre>

<hr>
<h2 id='kegg.ml.gsets'>
KEGG gene set data for the AML and CML pathways
</h2><span id='topic+kegg.ml.gsets'></span>

<h3>Description</h3>

<p>A data set with the list of ensemble gene identifiers for the acute myeloid leukemia (AML) and chronic myeloid leukemia pathways as defined in the KEGG pathway database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("kegg.ml.gsets")</code></pre>


<h3>Format</h3>

<p>A data frame with 128 rows describing the pairings between the following 2 variables.
</p>

<dl>
<dt><code>vset</code></dt><dd><p>KEGG pathway name</p>
</dd>
<dt><code>vID</code></dt><dd><p>Ensemble gene (ENSG) identifier</p>
</dd>
</dl>



<h3>Details</h3>

<p>A dataset with assignments of ensemble gene identifiers (ENSG) to KEGG pathway names
</p>


<h3>Source</h3>

<p><a href="http://software.broadinstitute.org/gsea/msigdb/cards/KEGG_CHRONIC_MYELOID_LEUKEMIA">http://software.broadinstitute.org/gsea/msigdb/cards/KEGG_CHRONIC_MYELOID_LEUKEMIA</a> <br /> <br />
<a href="http://software.broadinstitute.org/gsea/msigdb/cards/KEGG_ACUTE_MYELOID_LEUKEMIA">http://software.broadinstitute.org/gsea/msigdb/cards/KEGG_ACUTE_MYELOID_LEUKEMIA</a> <br /> <br />
Merged with information from <a href="ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz">ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz</a>
to translate gene symbols to Ensemble gene (ENSG) identifiers
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(kegg.ml.gsets)
</code></pre>

<hr>
<h2 id='prep.gsda'>
GSDA Data Preparation
</h2><span id='topic+prep.gsda'></span>

<h3>Description</h3>

<p>A function to prepare omic data, clinical data and variable set into an ordered matched format for GSDA analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep.gsda(data.mtx, clin.data, vset.data = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prep.gsda_+3A_data.mtx">data.mtx</code></td>
<td>
<p>Numeric data matrix with column names giving subject identifiers.</p>
</td></tr>
<tr><td><code id="prep.gsda_+3A_clin.data">clin.data</code></td>
<td>
<p>Data.frame with column named &quot;ID&quot; with subject identifiers matching column names of data.mtx.</p>
</td></tr>
<tr><td><code id="prep.gsda_+3A_vset.data">vset.data</code></td>
<td>
<p>data.frame of variable-set assignments with columns named &quot;vID&quot; for variable identifier and &quot;vset&quot; for name or identifier of a variable set (gene-set).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The gsda function uses prep.gsda to prepare the omic data matrix, clinical dataframe and variable set (gene set) into ordered and matched format, which is then used for GSDA analysis.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>omic.data</code></td>
<td>
<p>data matrix with columns in the same order as clin.data$ID.</p>
</td></tr>
<tr><td><code>clin.data</code></td>
<td>
<p>data.frame with ID column in same order as columns of omic.data.</p>
</td></tr>
<tr><td><code>vset.data</code></td>
<td>
<p>variable set ordered by name of variable set.</p>
</td></tr>
<tr><td><code>vset.index</code></td>
<td>
<p>simple data.frame showing first and last row of vset.data for each variable set.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gsda">gsda</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(target.aml.clin)
data(target.aml.expr)
data(kegg.ml.gsets)
gsdaprep=prep.gsda(target.aml.expr,
         target.aml.clin,
         kegg.ml.gsets)
</code></pre>

<hr>
<h2 id='prep.msigdb'>
Preparation of MSigDB for GSDA
</h2><span id='topic+prep.msigdb'></span>

<h3>Description</h3>

<p>This function prepares the gene sets of a species in MsigDB for gene-set distance analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep.msigdb(species = "Homo sapiens", vset = "gs_name", vID = "gene_symbol")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prep.msigdb_+3A_species">species</code></td>
<td>
<p>Name of species in MSigDB.</p>
</td></tr>
<tr><td><code id="prep.msigdb_+3A_vset">vset</code></td>
<td>
<p>Name of MSigDB column to use as vset in gsda, default is &quot;gs_name&quot;.</p>
</td></tr>
<tr><td><code id="prep.msigdb_+3A_vid">vID</code></td>
<td>
<p>Name of MSigDB column to use as vID in gsda, default is &quot;gene_symbol&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Take a species from MsigDB (https://www.gsea-msigdb.org/gsea/msigdb/index.jsp), extract gene set definiton and prepare a data frame with gene sets and genes to be used as vset.data in the gsda function.
</p>


<h3>Value</h3>

<p>A two-column data.frame with the columns vset and vID
</p>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gsets=prep.msigdb()
head(gsets)

</code></pre>

<hr>
<h2 id='print.bdc'>
Print Method for Best Distance Correlation
</h2><span id='topic+print.bdc'></span>

<h3>Description</h3>

<p>Print the result of the best distance correlation (best.dist.corr)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bdc'
print(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.bdc_+3A_x">x</code></td>
<td>
<p>a class of bdc</p>
</td></tr>
<tr><td><code id="print.bdc_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Print the summary of result of best distance correlation to stdout.
</p>


<h3>Value</h3>

<p>No return value, called for side effects</p>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+print.dcor">print.dcor</a></code>, <code><a href="#topic+print.gsda.result">print.gsda.result</a></code></p>

<hr>
<h2 id='print.dcor'>
Print Method for Distance Correlation
</h2><span id='topic+print.dcor'></span>

<h3>Description</h3>

<p>Print the summary of result of distance correlation (dist.corr function).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dcor'
print(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.dcor_+3A_x">x</code></td>
<td>
<p>result of dist.corr, class dcor</p>
</td></tr>
<tr><td><code id="print.dcor_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Print the summary of result of distance correlation to stdout.
</p>


<h3>Value</h3>

<p>No return value, called for side effects</p>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+print.bdc">print.bdc</a></code>, <code><a href="#topic+print.gsda.result">print.gsda.result</a></code></p>

<hr>
<h2 id='print.gsda.result'>
Print Method for GSDA Result
</h2><span id='topic+print.gsda.result'></span>

<h3>Description</h3>

<p>Print the of result of gene-set distance analysis (gsda function).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gsda.result'
print(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.gsda.result_+3A_x">x</code></td>
<td>
<p>result of gene-set distance analysis (gsda function)</p>
</td></tr>
<tr><td><code id="print.gsda.result_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Print the result of gene-set distance analysis to stdout.
</p>


<h3>Value</h3>

<p>No return value, called for side effects</p>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+print.dcor">print.dcor</a></code>, <code><a href="#topic+print.bdc">print.bdc</a></code></p>

<hr>
<h2 id='surv.dist'>
Distance of a Survival Endpoint
</h2><span id='topic+surv.dist'></span>

<h3>Description</h3>

<p>A function to calculate the distance for a survivial endpoint. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv.dist(stime.evnt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv.dist_+3A_stime.evnt">stime.evnt</code></td>
<td>

<p>A data frame with time in first column and censor in second column.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the distance matrix for a censored event-time variable. The calculation is based on the formula in Section 2.4 of Cao and Pounds (2021). The distance metric for censored event-time data is based on the rank-based association statistic for this type of data proposed by Jung et al (2005). 
</p>


<h3>Value</h3>

<p>A square matrix with nrow and ncol equal to the nrow of stime.evnt.  Entry (i,j) of the result matrix gives the survival distance between subjects represented in rows i and j of stime.evnt.
</p>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis. <br /> <br /> 
Jung SH, Owzar K, and George SL (2005) A mutiple testing procedure to associate gene expression levels with survival. Statistics in Medicine 24: 3077-88.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cat.dist">cat.dist</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(target.aml.clin)
srv.dist=surv.dist(target.aml.clin[,c("efs.time","efs.evnt")])
</code></pre>

<hr>
<h2 id='target.aml.clin'>Clinical outcomes for AML TARGET Project</h2><span id='topic+target.aml.clin'></span>

<h3>Description</h3>

<p>A dataset with subject identifier, survival time, and death indicator for 123 pediatric AML patients</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("target.aml.clin")</code></pre>


<h3>Format</h3>

<p>A data frame with 123 observations of the following 5 variables.
</p>

<dl>
<dt><code>ID</code></dt><dd><p>subject identifier, a character vector</p>
</dd>
<dt><code>Chloroma</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>logWBC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>efs.time</code></dt><dd><p>event-free survival time, a numeric vector</p>
</dd>
<dt><code>efs.evnt</code></dt><dd><p>event indicator (0 = censored, 1 = event) for efs.time, a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>A dataset with clinical data for each of 123 pediatric AML patients
</p>


<h3>Source</h3>

<p>obtained from <a href="https://target-data.nci.nih.gov/Public/AML/clinical/harmonized/">https://target-data.nci.nih.gov/Public/AML/clinical/harmonized/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(target.aml.clin)
</code></pre>

<hr>
<h2 id='target.aml.expr'>
RNA-seq expression from the AML TARGET project
</h2><span id='topic+target.aml.expr'></span>

<h3>Description</h3>

<p>A matrix of RNA-seq gene expression values for 123 pediatric AML patients from the TARGET project for genes in the KEGG AML and CML pathways
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("target.aml.expr")</code></pre>


<h3>Format</h3>

<p>Each row contains the expression values of one 94 Ensemble genes for all 123 patients.  Each column contains the expression values of all 94 Ensemble genes for one patient.  The rownames give the Ensemble identifiers for the genes. The columns give the patient identifiers.  
</p>


<h3>Details</h3>

<p>A RNA-seq dataset with expression levels in 94 ensemble gene identifiers for 123 pediatric AML patients
</p>


<h3>Source</h3>

<p><a href="https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/">https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(target.aml.expr)
</code></pre>

<hr>
<h2 id='U.center'>U Centering</h2><span id='topic+U.center'></span>

<h3>Description</h3>

<p>U-center the distance matrix in preparation of computing distance correlations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>U.center(d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="U.center_+3A_d">d</code></td>
<td>
<p>A square numeric data matrix</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This funtion centers the distance matrix according to U-centering formula on page 6 of arXiv 1902.03291 paper
</p>


<h3>Value</h3>

<p>A centered data matrix</p>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>
<p>Zhu C, Yao S, Zhang X and Shao X. Distance-based and RKHS-based Dependence Metrics in High Dimension. arXiv:1902.03291
</p>


<h3>See Also</h3>

<p><code><a href="#topic+uc.dist">uc.dist</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(target.aml.clin)
cd=cat.dist(target.aml.clin$Chloroma)
ud=U.center(cd)
ud[1:5,1:5]
</code></pre>

<hr>
<h2 id='uc.dist'>
U-centered Distance Matrix
</h2><span id='topic+uc.dist'></span>

<h3>Description</h3>

<p>The function calculates the U-centered distance matrix for a variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uc.dist(X, dmeth = "me")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="uc.dist_+3A_x">X</code></td>
<td>
<p>vector, matrix, or data.frame to compute a distance matrix</p>
</td></tr>
<tr><td><code id="uc.dist_+3A_dmeth">dmeth</code></td>
<td>
<p>Distance method to use, options include &quot;oe&quot; for overall Euclidean, &quot;me&quot; for marginal Euclidean, &quot;om&quot; for overall Manhattan, &quot;mm&quot; for marginal Manhattan, &quot;ct&quot; for categorical, and &quot;st&quot; for censored survival time.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A distance matrix is first calculated for a scale or data frame of a variable. The distance matrix is then centered according to U-centering formula on page 6 of arXiv 1902.03291 paper.
</p>


<h3>Value</h3>

<p>For distance methods &quot;oe&quot;, &quot;om&quot;, &quot;ct&quot;, and &quot;st&quot;, one matrix of overall distances computed using data from all variables.  For distance methods &quot;me&quot; and &quot;mm&quot;, an array of distance matrices, one distance matrix per variable.
</p>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>
<p>Zhu C, Yao S, Zhang X and Shao X. Distance-based and RKHS-based Dependence Metrics in High Dimension. arXiv:1902.03291
</p>


<h3>See Also</h3>

<p><code><a href="#topic+U.center">U.center</a></code>,  <code><a href="#topic+prep.gsda">prep.gsda</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(target.aml.expr)
target.aml.expr=sqrt(target.aml.expr)
target.aml.expr=t(target.aml.expr)
oe.dist=uc.dist(target.aml.expr,"oe") # overall Euclidean
</code></pre>

<hr>
<h2 id='write.gsda.csv.file'>
Write GSDA Result to a Comma Delimited File
</h2><span id='topic+write.gsda.csv.file'></span>

<h3>Description</h3>

<p>Write a gene-set distance analysis result to a comma delimited file (.csv)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.gsda.csv.file(gsda.result, out.file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.gsda.csv.file_+3A_gsda.result">gsda.result</code></td>
<td>
<p>A class of gene-set distance analysis result</p>
</td></tr>
<tr><td><code id="write.gsda.csv.file_+3A_out.file">out.file</code></td>
<td>
<p>A .csv file name with directory</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A saved .csv file.
</p>


<h3>Author(s)</h3>

<p>Xueyuan Cao <a href="mailto:xcao12@uthsc.edu">xcao12@uthsc.edu</a> and Stanley Pounds <a href="mailto:stanley.pounds@stjude.org">stanley.pounds@stjude.org</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
