<!DOCTYPE html><html lang="en"><head><title>Help for package GSelection</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GSelection}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GSelection-package'>
<p>Genomic Selection</p></a></li>
<li><a href='#feature.selection'>
<p>Genomic Feature Selection</p></a></li>
<li><a href='#genomic.prediction'>
<p>Genomic Prediction</p></a></li>
<li><a href='#GS'><p>Genotypic and phenotypic simulated dataset</p></a></li>
<li><a href='#hsic.var.ensemble'>
<p>Error Variance Estimation in Genomic Prediction</p></a></li>
<li><a href='#hsic.var.rcv'>
<p>Error Variance Estimation in Genomic Prediction</p></a></li>
<li><a href='#RED'>
<p>Redundancy Rate</p></a></li>
<li><a href='#spam.var.ensemble'>
<p>Error Variance Estimation in Genomic Prediction</p></a></li>
<li><a href='#spam.var.rcv'>
<p>Error Variance Estimation in Genomic Prediction</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genomic Selection</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Sayanti Guha Majumdar, Anil Rai, Dwijesh Chandra Mishra</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sayanti Guha Majumdar &lt;sayanti23gm@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Genomic selection is a specialized form of marker assisted selection. The package contains functions to select important genetic markers and predict phenotype on the basis of fitted training data using integrated model framework (Guha Majumdar et. al. (2019) &lt;<a href="https://doi.org/10.1089%2Fcmb.2019.0223">doi:10.1089/cmb.2019.0223</a>&gt;) developed by combining one additive (sparse additive models by Ravikumar et. al. (2009) &lt;<a href="https://doi.org/10.1111%2Fj.1467-9868.2009.00718.x">doi:10.1111/j.1467-9868.2009.00718.x</a>&gt;) and one non-additive (hsic lasso by Yamada et. al. (2014) &lt;<a href="https://doi.org/10.1162%2FNECO_a_00537">doi:10.1162/NECO_a_00537</a>&gt;) model.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>SAM, penalized, gdata, stats, utils</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-10-31 08:59:34 UTC; SAYANTI</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-11-04 16:30:27 UTC</td>
</tr>
</table>
<hr>
<h2 id='GSelection-package'>
Genomic Selection
</h2><span id='topic+GSelection-package'></span><span id='topic+GSelection'></span>

<h3>Description</h3>

<p>Genomic selection is a specialized form of marker assisted selection. The package contains functions to select important genetic markers and predict phenotype on the basis of fitted training data using integrated model framework (Guha Majumdar et. al. (2019) &lt;doi:10.1089/cmb.2019.0223&gt;) developed by combining one additive (sparse additive models by Ravikumar et. al. (2009) &lt;doi:10.1111/j.1467-9868.2009.00718.x&gt;) and one non-additive (hsic lasso by Yamada et. al. (2014) &lt;doi:10.1162/NECO_a_00537&gt;) model.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> GSelection</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Genomic Selection</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Sayanti Guha Majumdar, Anil Rai, Dwijesh Chandra Mishra</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Sayanti Guha Majumdar &lt;sayanti23gm@gmail.com&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Genomic selection is a specialized form of marker assisted selection. The package contains functions to select important genetic markers and predict phenotype on the basis of fitted training data using integrated model framework (Guha Majumdar et. al. (2019) &lt;doi:10.1089/cmb.2019.0223&gt;) developed by combining one additive (sparse additive models by Ravikumar et. al. (2009) &lt;doi:10.1111/j.1467-9868.2009.00718.x&gt;) and one non-additive (hsic lasso by Yamada et. al. (2014) &lt;doi:10.1162/NECO_a_00537&gt;) model.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
Encoding: </td><td style="text-align: left;"> UTF-8</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyData: </td><td style="text-align: left;"> true</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> SAM, penalized, gdata, stats, utils</td>
</tr>
<tr>
 <td style="text-align: left;">
RoxygenNote: </td><td style="text-align: left;"> 6.1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R (&gt;= 3.5)</td>
</tr>
<tr>
 <td style="text-align: left;">
NeedsCompilation: </td><td style="text-align: left;"> no</td>
</tr>
<tr>
 <td style="text-align: left;">
Packaged: </td><td style="text-align: left;"> 2019-10-26 10:25:25 UTC; user6</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<p>Index of help topics:
</p>
<pre>
GS                      Genotypic and phenotypic simulated dataset
GSelection-package      Genomic Selection
RED                     Redundancy Rate
feature.selection       Genomic Feature Selection
genomic.prediction      Genomic Prediction
hsic.var.ensemble       Error Variance Estimation in Genomic Prediction
hsic.var.rcv            Error Variance Estimation in Genomic Prediction
spam.var.ensemble       Error Variance Estimation in Genomic Prediction
spam.var.rcv            Error Variance Estimation in Genomic Prediction
</pre>


<h3>Author(s)</h3>

<p>Sayanti Guha Majumdar, Anil Rai, Dwijesh Chandra Mishra
</p>
<p>Maintainer: Sayanti Guha Majumdar &lt;sayanti23gm@gmail.com&gt;
</p>


<h3>References</h3>

<p>Guha Majumdar, S., Rai, A. and Mishra, D. C. (2019). Integrated framework for selection of additive and non-additive genetic markers for genomic selection. <em>Journal of Computational Biology</em>. doi:10.1089/cmb.2019.0223
<br /> Ravikumar, P., Lafferty, J., Liu, H. and Wasserman, L. (2009). Sparse additive models. <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em>, 71(5), 1009-1030. doi:10.1111/j.1467-9868.2009.00718.x
<br /> Yamada, M., Jitkrittum, W., Sigal, L., Xing, E. P. and Sugiyama, M. (2014). High-Dimensional Feature Selection by Feature-Wise Kernelized Lasso. <em>Neural Computation</em>, 26(1):185-207. doi:10.1162/NECO_a_00537
</p>

<hr>
<h2 id='feature.selection'>
Genomic Feature Selection
</h2><span id='topic+feature.selection'></span>

<h3>Description</h3>

<p>Feature (marker) selection in case of genomic prediction with integrated model framework using both additive (Sparse Additive Models) and non-additive (HSIC LASSO) statistical models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>feature.selection(x,y,d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="feature.selection_+3A_x">x</code></td>
<td>
<p>a matrix of markers or explanatory variables, each column contains one marker and each row represents an individual.</p>
</td></tr>
<tr><td><code id="feature.selection_+3A_y">y</code></td>
<td>
<p>a column vector of response variable.</p>
</td></tr>
<tr><td><code id="feature.selection_+3A_d">d</code></td>
<td>
<p>number of variables to be selected from x.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Integrated model framework was developed by combining one additive model (Sparse Additive Model) and one non-additive model (HSIC LASSO) for selection of important markers from whole genome marker data.
</p>


<h3>Value</h3>

<p>Returns a LIST containing
</p>
<table role = "presentation">
<tr><td><code>spam_selected_feature_index</code></td>
<td>
<p>returns index of selected markers from x using Sparse Additive Model</p>
</td></tr>
<tr><td><code>coefficient.spam</code></td>
<td>
<p>returns coefficient values of selected markers using Sparse Additive Model.</p>
</td></tr>
<tr><td><code>hsic_selected_feature_index</code></td>
<td>
<p>returns index of selected markers from x using HSIC LASSO.</p>
</td></tr>
<tr><td><code>coefficient.hsic</code></td>
<td>
<p>returns coefficient values of selected markers using HSIC LASSO.</p>
</td></tr>
<tr><td><code>integrated_selected_feature_index</code></td>
<td>
<p>returns index of selected markers from x using integrated model framework.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sayanti Guha Majumdar &lt;<a href="mailto:sayanti23gm@gmail.com">sayanti23gm@gmail.com</a>&gt;, Anil Rai, Dwijesh Chandra Mishra
</p>


<h3>References</h3>

<p>Guha Majumdar, S., Rai, A. and Mishra, D. C. (2019). Integrated framework for selection of additive and non-additive genetic markers for genomic selection. <em>Journal of Computational Biology</em>. doi:10.1089/cmb.2019.0223
<br /> Ravikumar, P., Lafferty, J., Liu, H. and Wasserman, L. (2009). Sparse additive models. <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em>, 71(5), 1009-1030. doi:10.1111/j.1467-9868.2009.00718.x
<br /> Yamada, M., Jitkrittum, W., Sigal, L., Xing, E. P. and Sugiyama, M. (2014). High-Dimensional Feature Selection by Feature-Wise Kernelized Lasso. <em>Neural Computation</em>, 26(1):185-207. doi:10.1162/NECO_a_00537
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GSelection)
data(GS)
x_trn &lt;- GS[1:40,1:110]
y_trn &lt;- GS[1:40,111]
x_tst &lt;- GS[41:60,1:110]
y_tst &lt;- GS[41:60,111]
fit &lt;- feature.selection(x_trn,y_trn,d=10)
</code></pre>

<hr>
<h2 id='genomic.prediction'>
Genomic Prediction
</h2><span id='topic+genomic.prediction'></span>

<h3>Description</h3>

<p>Prediction of phenotypic values based on selected markers with integrated model framework using both additive (Sparse Additive Models) and non-additive (HSIC LASSO) statistical models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomic.prediction(x,spam_error_var,hsic_error_var,
spam_selected_feature_index,hsic_selected_feature_index,
coefficient.spam,coefficient.hsic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genomic.prediction_+3A_x">x</code></td>
<td>
<p>a matrix of markers or explanatory variables for which phenotype will be predicted. Each column contains one marker and each row represents an individual.</p>
</td></tr>
<tr><td><code id="genomic.prediction_+3A_spam_error_var">spam_error_var</code></td>
<td>
<p>estimated error variance of genomic prediction by Sparse Additive Model.</p>
</td></tr>
<tr><td><code id="genomic.prediction_+3A_hsic_error_var">hsic_error_var</code></td>
<td>
<p>estimated error variance of genomic prediction by HSIC LASSO.</p>
</td></tr>
<tr><td><code id="genomic.prediction_+3A_spam_selected_feature_index">spam_selected_feature_index</code></td>
<td>
<p>index of selected markers from x using Sparse Additive Model</p>
</td></tr>
<tr><td><code id="genomic.prediction_+3A_hsic_selected_feature_index">hsic_selected_feature_index</code></td>
<td>
<p>index of selected markers from x using HSIC LASSO.</p>
</td></tr>
<tr><td><code id="genomic.prediction_+3A_coefficient.spam">coefficient.spam</code></td>
<td>
<p>coefficient values of selected markers using Sparse Additive Model.</p>
</td></tr>
<tr><td><code id="genomic.prediction_+3A_coefficient.hsic">coefficient.hsic</code></td>
<td>
<p>coefficient values of selected markers using HSIC LASSO.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Phenotypic values will be predicted for given genotype of markers by using previously fitted model object. Integrated model framework is used for this purpose which is developed by combining selected features from SpAm and HSIC LASSO.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Integrated_y</code></td>
<td>
<p>returns predicted phenotype</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sayanti Guha Majumdar &lt;<a href="mailto:sayanti23gm@gmail.com">sayanti23gm@gmail.com</a>&gt;, Anil Rai, Dwijesh Chandra Mishra
</p>


<h3>References</h3>

<p>Guha Majumdar, S., Rai, A. and Mishra, D. C. (2019). Integrated framework for selection of additive and non-additive genetic markers for genomic selection. <em>Journal of Computational Biology</em>. doi:10.1089/cmb.2019.0223
<br /> Ravikumar, P., Lafferty, J., Liu, H. and Wasserman, L. (2009). Sparse additive models. <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em>, 71(5), 1009-1030. doi:10.1111/j.1467-9868.2009.00718.x
<br /> Yamada, M., Jitkrittum, W., Sigal, L., Xing, E. P. and Sugiyama, M. (2014). High-Dimensional Feature Selection by Feature-Wise Kernelized Lasso. <em>Neural Computation</em>, 26(1):185-207. doi:10.1162/NECO_a_00537
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GSelection)
data(GS)
x_trn &lt;- GS[1:40,1:110]
y_trn &lt;- GS[1:40,111]
x_tst &lt;- GS[41:60,1:110]
y_tst &lt;- GS[41:60,111]

## estimate spam_var from function spam.var.ensemble or spam.var.rcv
spam_var &lt;- 2.681972
## estimate hsic_var from function hsic.var.ensemble or hsic.var.rcv
hsic_var &lt;- 10.36974

fit &lt;- feature.selection(x_trn,y_trn,d=10)
pred_y &lt;- genomic.prediction(x_tst,spam_var,hsic_var,
fit$spam_selected_feature_index,fit$hsic_selected_feature_index,
fit$coefficient.spam,fit$coefficient.hsic)
</code></pre>

<hr>
<h2 id='GS'>Genotypic and phenotypic simulated dataset</h2><span id='topic+GS'></span>

<h3>Description</h3>

<p>This dataset is simulated with the R package &quot;qtlbim&quot; where 10 are true features associated with the trait of study and remaining 100 are random markers. we consider 10 chromosomes each containing 10 markers. Each chromosome have 1 qtl which is the true feature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("GS")</code></pre>


<h3>Format</h3>

<p>A data frame with 60 rows as genotypes with 111 columns (i.e. contains information of genotyped markers and phenotypic traits).</p>


<h3>Details</h3>

<p>It has total 60 rows which represents 200 individuals genotypes and a total of 111 of columns, in which first 110 columns contain information of genotyped markers and last column represents value of phenotypic trait associated with genotype under study.
</p>


<h3>Source</h3>

<p>Yandell, B. S., Mehta, T., Banerjee, S., Shriner, D., Venkataraman, R. et al. (2007). R/qtlbim: QTL with Bayesian Interval Mapping in experimental crosses. <em>Bioinformatics</em>, 23, 641-643.
<br /> Yandell, B. S., Nengjun, Y., Mehta, T., Banerjee, S., Shriner, D. et al. (2012). qtlbim: QTL Bayesian Interval Mapping. R package version 2.0.5. http://CRAN.R-project.org/package=qtlbim
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GSelection)
data(GS)
X&lt;-GS[,1:110]   ## Extracting Genotype
Y&lt;-GS[,111] ## Extracting Phenotype
</code></pre>

<hr>
<h2 id='hsic.var.ensemble'>
Error Variance Estimation in Genomic Prediction
</h2><span id='topic+hsic.var.ensemble'></span>

<h3>Description</h3>

<p>Estimation of error variance using Ensemble method which combines bootstraping and sampling with srswor in HSIC LASSO.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hsic.var.ensemble(x,y,b,d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hsic.var.ensemble_+3A_x">x</code></td>
<td>
<p>a matrix of markers or explanatory variables, each column contains one marker and each row represents an individual.</p>
</td></tr>
<tr><td><code id="hsic.var.ensemble_+3A_y">y</code></td>
<td>
<p>a column vector of response variable.</p>
</td></tr>
<tr><td><code id="hsic.var.ensemble_+3A_b">b</code></td>
<td>
<p>number of bootstrap samples.</p>
</td></tr>
<tr><td><code id="hsic.var.ensemble_+3A_d">d</code></td>
<td>
<p>number of variables to be selected from x.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this method, both bootstrapping and simple random sampling without replacement are combined to estimate error variance. Variables are selected using HSIC LASSO from the original datasets and all possible samples of a particular size are taken from the selected variables set with simple random sampling without replacement. With these selected samples error variance is estimated from bootstrap samples of the original datasets using least squared regression method. Finally the average of all the estimated variances is considered as the final estimate of the error variance.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Error variance</code></td>
<td>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sayanti Guha Majumdar &lt;<a href="mailto:sayanti23gm@gmail.com">sayanti23gm@gmail.com</a>&gt;, Anil Rai, Dwijesh Chandra Mishra
</p>


<h3>References</h3>

<p>Yamada, M., Jitkrittum, W., Sigal, L., Xing, E. P. and Sugiyama, M. (2014). High-Dimensional Feature Selection by Feature-Wise Kernelized Lasso. <em>Neural Computation</em>, 26(1):185-207. doi:10.1162/NECO_a_00537
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GSelection)
data(GS)
x_trn &lt;- GS[1:40,1:110]
y_trn &lt;- GS[1:40,111]
x_tst &lt;- GS[41:60,1:110]
y_tst &lt;- GS[41:60,111]
hsic_var &lt;- hsic.var.ensemble(x_trn,y_trn,2,10)
</code></pre>

<hr>
<h2 id='hsic.var.rcv'>
Error Variance Estimation in Genomic Prediction
</h2><span id='topic+hsic.var.rcv'></span>

<h3>Description</h3>

<p>Estimation of error variance using Refitted Cross Validation in HSIC LASSO.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hsic.var.rcv(x,y,d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hsic.var.rcv_+3A_x">x</code></td>
<td>
<p>a matrix of markers or explanatory variables, each column contains one marker and each row represents an individual.</p>
</td></tr>
<tr><td><code id="hsic.var.rcv_+3A_y">y</code></td>
<td>
<p>a column vector of response variable.</p>
</td></tr>
<tr><td><code id="hsic.var.rcv_+3A_d">d</code></td>
<td>
<p>number of variables to be selected from x.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Refitted cross validation method (RCV) which is a two step method, is used to get the estimate of the error variance. In first step, dataset is divided into two sub-datasets and with the help of HSIC LASSO most significant markers(variables) are selected from the two sub-datasets. This results in two small sets of selected variables. Then using the set selected from 1st sub-dataset error variance is estimated from the 2nd sub-dataset with ordinary least square method and using the set selected from the 2nd sub-dataset error variance is estimated from the 1st sub-dataset with ordinary least square method. Finally the average of those two error variances are taken as the final estimator of error variance with RCV method.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Error variance</code></td>
<td>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sayanti Guha Majumdar &lt;<a href="mailto:sayanti23gm@gmail.com">sayanti23gm@gmail.com</a>&gt;, Anil Rai, Dwijesh Chandra Mishra
</p>


<h3>References</h3>

<p>Fan, J., Guo, S., Hao, N. (2012). Variance estimation using refitted cross-validation in ultrahigh dimensional regression. <em>Journal of the Royal Statistical Society</em>, 74(1), 37-65.
<br /> Yamada, M., Jitkrittum, W., Sigal, L., Xing, E. P. and Sugiyama, M. (2014). High-Dimensional Feature Selection by Feature-Wise Kernelized Lasso. <em>Neural Computation</em>, 26(1):185-207. doi:10.1162/NECO_a_00537
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GSelection)
data(GS)
x_trn &lt;- GS[1:40,1:110]
y_trn &lt;- GS[1:40,111]
x_tst &lt;- GS[41:60,1:110]
y_tst &lt;- GS[41:60,111]
hsic_var &lt;- hsic.var.rcv(x_trn,y_trn,10)
</code></pre>

<hr>
<h2 id='RED'>
Redundancy Rate
</h2><span id='topic+RED'></span>

<h3>Description</h3>

<p>Calculate the redundancy rate of the selected features(markers). Value will be high if many redundant features are selected.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RED(x,spam_selected_feature_index,hsic_selected_feature_index,
integrated_selected_feature_index)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RED_+3A_x">x</code></td>
<td>
<p>a matrix of markers or explanatory variables, each column contains one marker and each row represents an individual.</p>
</td></tr>
<tr><td><code id="RED_+3A_spam_selected_feature_index">spam_selected_feature_index</code></td>
<td>
<p>index of selected markers from x using Sparse Additive Model.</p>
</td></tr>
<tr><td><code id="RED_+3A_hsic_selected_feature_index">hsic_selected_feature_index</code></td>
<td>
<p>index of selected markers from x using HSIC LASSO.</p>
</td></tr>
<tr><td><code id="RED_+3A_integrated_selected_feature_index">integrated_selected_feature_index</code></td>
<td>
<p>index of selected markers from x using integrated model framework</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The RED score (Zhao et al., 2010) is determined by average of the correlation between each pair of selected markers. A large RED score signifies that selected features are more strongly correlated to each other which means many redundant features are selected. Thus, a small redundancy rate is preferable for feature selection.
</p>


<h3>Value</h3>

<p>Returns a LIST containing
</p>
<table role = "presentation">
<tr><td><code>RED_spam</code></td>
<td>
<p>returns redundancy rate of features selected by using Sparse Additive Model.</p>
</td></tr>
<tr><td><code>RED_hsic</code></td>
<td>
<p>returns redundancy rate of features selected by using HSIC LASSO.</p>
</td></tr>
<tr><td><code>RED_I</code></td>
<td>
<p>returns redundancy rate of features selected by using integrated model framework.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sayanti Guha Majumdar &lt;<a href="mailto:sayanti23gm@gmail.com">sayanti23gm@gmail.com</a>&gt;, Anil Rai, Dwijesh Chandra Mishra
</p>


<h3>References</h3>

<p>Guha Majumdar, S., Rai, A. and Mishra, D. C. (2019). Integrated framework for selection of additive and non-additive genetic markers for genomic selection. <em>Journal of Computational Biology</em>. doi:10.1089/cmb.2019.0223
<br /> Ravikumar, P., Lafferty, J., Liu, H. and Wasserman, L. (2009). Sparse additive models. <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em>, 71(5), 1009-1030. doi:10.1111/j.1467-9868.2009.00718.x
<br /> Yamada, M., Jitkrittum, W., Sigal, L., Xing, E. P. and Sugiyama, M. (2014). High-Dimensional Feature Selection by Feature-Wise Kernelized Lasso. <em>Neural Computation</em>, 26(1):185-207. doi:10.1162/NECO_a_00537
<br /> Zhao, Z., Wang, L. and Li, H. (2010). Efficient spectral feature selection with minimum redundancy. <em>In AAAI Conference on Artificial Intelligence (AAAI)</em>, pp 673-678.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GSelection)
data(GS)
x_trn &lt;- GS[1:40,1:110]
y_trn &lt;- GS[1:40,111]
x_tst &lt;- GS[41:60,1:110]
y_tst &lt;- GS[41:60,111]
fit &lt;- feature.selection(x_trn,y_trn,d=10)
red &lt;- RED(x_trn,fit$spam_selected_feature_index,fit$hsic_selected_feature_index,
fit$integrated_selected_feature_index)
</code></pre>

<hr>
<h2 id='spam.var.ensemble'>
Error Variance Estimation in Genomic Prediction
</h2><span id='topic+spam.var.ensemble'></span>

<h3>Description</h3>

<p>Estimation of error variance using Ensemble method which combines bootstraping and sampling with srswor in Sparse Additive Models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spam.var.ensemble(x,y,b,d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="spam.var.ensemble_+3A_x">x</code></td>
<td>
<p>a matrix of markers or explanatory variables, each column contains one marker and each row represents an individual.</p>
</td></tr>
<tr><td><code id="spam.var.ensemble_+3A_y">y</code></td>
<td>
<p>a column vector of response variable.</p>
</td></tr>
<tr><td><code id="spam.var.ensemble_+3A_b">b</code></td>
<td>
<p>number of bootstrap samples</p>
</td></tr>
<tr><td><code id="spam.var.ensemble_+3A_d">d</code></td>
<td>
<p>number of variables to be selected from x.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this method, both bootstrapping and simple random sampling without replacement are combined to estimate error variance. Variables are selected using Sparse Additive Models (SpAM) from the original datasets and all possible samples of a particular size are taken from the selected variables set with simple random sampling without replacement. With these selected samples error variance is estimated from bootstrap samples of the original datasets using least squared regression method. Finally the average of all the estimated variances is considered as the final estimate of the error variance.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Error variance</code></td>
<td>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sayanti Guha Majumdar &lt;<a href="mailto:sayanti23gm@gmail.com">sayanti23gm@gmail.com</a>&gt;, Anil Rai, Dwijesh Chandra Mishra
</p>


<h3>References</h3>

<p>Ravikumar, P., Lafferty, J., Liu, H. and Wasserman, L. (2009). Sparse additive models. <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em>, 71(5), 1009-1030. doi:10.1111/j.1467-9868.2009.00718.x
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GSelection)
data(GS)
x_trn &lt;- GS[1:40,1:110]
y_trn &lt;- GS[1:40,111]
x_tst &lt;- GS[41:60,1:110]
y_tst &lt;- GS[41:60,111]
spam_var &lt;- spam.var.ensemble(x_trn,y_trn,2,10)
</code></pre>

<hr>
<h2 id='spam.var.rcv'>
Error Variance Estimation in Genomic Prediction
</h2><span id='topic+spam.var.rcv'></span>

<h3>Description</h3>

<p>Estimation of error variance using Refitted cross validation in Sparse Additive Models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spam.var.rcv(x,y,d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="spam.var.rcv_+3A_x">x</code></td>
<td>
<p>a matrix of markers or explanatory variables, each column contains one marker and each row represents an individual.</p>
</td></tr>
<tr><td><code id="spam.var.rcv_+3A_y">y</code></td>
<td>
<p>a column vector of response variable.</p>
</td></tr>
<tr><td><code id="spam.var.rcv_+3A_d">d</code></td>
<td>
<p>number of variables to be selected from x.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Refitted cross validation method (RCV) which is a two step method, is used to get the estimate of the error variance. In first step, dataset is divided into two sub-datasets and with the help of Sparse Additive Models (SpAM) most significant markers(variables) are selected from the two sub-datasets. This results in two small sets of selected variables. Then using the set selected from 1st sub-dataset error variance is estimated from the 2nd sub-dataset with ordinary least square method and using the set selected from the 2nd sub-dataset error variance is estimated from the 1st sub-dataset with ordinary least square method. Finally the average of those two error variances are taken as the final estimator of error variance with RCV method.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Error variance</code></td>
<td>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sayanti Guha Majumdar &lt;<a href="mailto:sayanti23gm@gmail.com">sayanti23gm@gmail.com</a>&gt;, Anil Rai, Dwijesh Chandra Mishra
</p>


<h3>References</h3>

<p>Fan, J., Guo, S., Hao, N. (2012).Variance estimation using refitted cross-validation in ultrahigh dimensional regression. <em>Journal of the Royal Statistical Society</em>, 74(1), 37-65.
<br /> Ravikumar, P., Lafferty, J., Liu, H. and Wasserman, L. (2009). Sparse additive models. <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em>, 71(5), 1009-1030. doi:10.1111/j.1467-9868.2009.00718.x</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GSelection)
data(GS)
x_trn &lt;- GS[1:40,1:110]
y_trn &lt;- GS[1:40,111]
x_tst &lt;- GS[41:60,1:110]
y_tst &lt;- GS[41:60,111]
spam_var &lt;- spam.var.rcv(x_trn,y_trn,10)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
