<!DOCTYPE html><html lang="en"><head><title>Help for package GSEMA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GSEMA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculateESpath'><p>Calculation of Effects Sizes and their variance for the different Gene Sets</p>
and studies</a></li>
<li><a href='#createObjectMApath'><p>Creation of the object to use in gene set enrichment meta-analysis</p></a></li>
<li><a href='#filteringPaths'><p>Fuction for filtering gene sets with low expression</p></a></li>
<li><a href='#heatmapPaths'><p>Visualization of the gene set enrichment meta-analysis results</p></a></li>
<li><a href='#metaAnalysisESpath'><p>Performing Gene Set Enrichment Meta-analysis</p></a></li>
<li><a href='#simulatedData'><p>GSEMA synthetic data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Set Enrichment Meta-Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.99.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Performing the different steps of gene set enrichment meta-analysis. It provides different functions that allow the application of meta-analysis based on the combination of effect sizes from different pathways in different studies to obtain significant pathways that are common to all of them.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.4)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BiocStyle, qpdf, BiocGenerics, RUnit, knitr, rmarkdown</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biobase, limma, Rdpack, singscore, GSVA, progress, pbapply,
doParallel, parallel, BiocParallel, metafor, RColorBrewer,
pheatmap, methods, scales, plyr, grDevices, stats, impute</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>biocViews:</td>
<td>StatisticalMethod, GeneSetEnrichment, Pathways</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-09 10:50:23 UTC; juanan</td>
</tr>
<tr>
<td>Author:</td>
<td>Juan Antonio Villatoro-García [aut, cre],
  Pablo Jurado-Bascón [aut],
  Pedro Carmona-Sáez [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Juan Antonio Villatoro-García &lt;juanantoniovillatorogarcia@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-14 09:20:17 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculateESpath'>Calculation of Effects Sizes and their variance for the different Gene Sets
and studies</h2><span id='topic+calculateESpath'></span>

<h3>Description</h3>

<p>This function uses different estimators to calculate the different effects
size and their variance for each gene and for each dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateESpath(
  objectMApath,
  measure = c("limma", "SMD", "MD"),
  WithinVarCorrect = TRUE,
  missAllow = 0.3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculateESpath_+3A_objectmapath">objectMApath</code></td>
<td>
<p>A list of list. Each list contains two elements.
The first element is the Gene Set matrix (gene sets in rows and samples in
columns) and the second element is a vector of zeros and ones that represents
the state of the different samples of the Gene Sets matrix.
0 represents one group (controls) and 1 represents the other group (cases).</p>
</td></tr>
<tr><td><code id="calculateESpath_+3A_measure">measure</code></td>
<td>
<p>A character string that indicates the type of effect size to
be calculated. The options are &quot;limma&quot;, &quot;SMD&quot; and &quot;MD&quot;. The default value
is &quot;limma&quot;. See details for more information.</p>
</td></tr>
<tr><td><code id="calculateESpath_+3A_withinvarcorrect">WithinVarCorrect</code></td>
<td>
<p>A logical value that indicates if the within variance
correction should be applied. The default value is TRUE. See details for more
information.</p>
</td></tr>
<tr><td><code id="calculateESpath_+3A_missallow">missAllow</code></td>
<td>
<p>a number that indicates the maximum proportion of missing
values allowed in a sample. If the sample has more proportion of missing
values the sample will be eliminated. In the other case the missing values
will be imputed using the K-NN algorithm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The different estimator methods that can be applied are:
</p>

<ol>
<li><p> &quot;limma&quot;
</p>
</li>
<li><p> &quot;SMD&quot;
</p>
</li>
<li><p> &quot;MD&quot;
</p>
</li></ol>

<p>The <b>&quot;SMD&quot;</b> (Standardized mean different) method calculates the effect
size using the Hedges'g estimator (Hedges, 1981).
</p>
<p>The <b>&quot;MD&quot;</b> (raw mean different) calculates the effects size as the
difference between the means of the two groups (Borenstein, 2009).
</p>
<p>The <b>&quot;limma&quot;</b> method used the limma package to calculate the effect size
and the variance of the effect size. The effect size is calculated from the
moderated Student's t computed by limma. From it, the estimator of Hedges'g
and its corresponding variance are obtained based on 
(Rosenthal, R., &amp; Rosnow, R. L., 2008))
In this way, some of the false positives obtained by 
the &quot;SMD&quot; method are reduced.
</p>
<p>The <b>WithinVarCorrect</b> parameter is a logical value that indicates if
the within variance correction should be applied. In the case of applying
the correction, the variance of the gene sets in each of the studies is
calculated based on the mean of the estimators and not on the estimator of
the study itself as described in formula (21) by (Lin L and Aloe AM 2021.)
</p>


<h3>Value</h3>

<p>A list formed by two elements:
</p>

<ul>
<li><p>First element (ES) is a dataframe were columns are each of the studies
(datasets) and rows are the genes sets. Each element of the dataframe
represents the Effect Size.
</p>
</li>
<li><p>Second element (Var) is a dataframe were columns are each of the
studies (datasets) and rows are the genes sets. Each element of the dataframe
represents the variance of the Effect size.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Juan Antonio Villatoro Garcia,
<a href="mailto:juanantoniovillatorogarcia@gmail.com">juanantoniovillatorogarcia@gmail.com</a>
</p>


<h3>References</h3>

<p>Borenstein, M. (2009). Effect sizes for continuous data. In H. Cooper,
L. V. Hedges, &amp; J. C. Valentine (Eds.),
The handbook of research synthesis and meta-analysis (2nd ed., pp. 221–235).
New York: Russell Sage Foundation.
</p>
<p>Hedges, L. V. (1981). Distribution theory for Glass's estimator of effect
size and related estimators. Journal of Educational Statistics, 6(2),
107–128. <a href="https://doi.org/10.2307/1164588">doi:10.2307/1164588</a>
</p>
<p>Lin L, Aloe AM (2021). Evaluation of various estimators for standardized mean
difference in meta-analysis. Stat Med. 2021 Jan 30;40(2):403-426.
<a href="https://doi.org/10.1002/sim.8781">doi:10.1002/sim.8781</a>
</p>
<p>Rosenthal, R., &amp; Rosnow, R. L. (2008). Essentials of behavioral research:
Methods and data analysis. McGraw-Hill.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createObjectMApath">createObjectMApath</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("simulatedData")
calculateESpath(objectMApath = objectMApathSim, measure = "limma")

</code></pre>

<hr>
<h2 id='createObjectMApath'>Creation of the object to use in gene set enrichment meta-analysis</h2><span id='topic+createObjectMApath'></span>

<h3>Description</h3>

<p>It allows the creation of an object to perform gene set
enrichment meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createObjectMApath(
  listEX,
  listPheno = NULL,
  namePheno = c(rep(1, length(listEX))),
  expGroups = c(rep(1, length(listEX))),
  refGroups = c(rep(2, length(listEX))),
  geneSets,
  pathMethod = c("GSVA", "Zscore", "ssGSEA", "Singscore"),
  minSize = 7,
  kcdf = "Gaussian",
  normalize = TRUE,
  n.cores = 1,
  internal.n.cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createObjectMApath_+3A_listex">listEX</code></td>
<td>
<p>A list of dataframes or matrix (genes in rows and sample
in columns). A list of ExpressionSets can be used too</p>
</td></tr>
<tr><td><code id="createObjectMApath_+3A_listpheno">listPheno</code></td>
<td>
<p>A list of phenodatas (dataframes or matrix). If the object
listEX is a list of ExpressionSets this element can be null.</p>
</td></tr>
<tr><td><code id="createObjectMApath_+3A_namepheno">namePheno</code></td>
<td>
<p>A list or vector of the different colunm names or
positions from the phenodatas where the experimental and reference groups
are identified. Each element of namePheno correspont to its equivalent
element in  the listPheno (default a vector of 1, all the first columns of
each elements of listPheno are selected).</p>
</td></tr>
<tr><td><code id="createObjectMApath_+3A_expgroups">expGroups</code></td>
<td>
<p>A list of vectors or a vector containing the names or the
positions with which we identify the elements of the experiment groups
(cases) of the namePheno element (default a vector  of 1, all the first
groups are selected)</p>
</td></tr>
<tr><td><code id="createObjectMApath_+3A_refgroups">refGroups</code></td>
<td>
<p>A list of vectors or a vector containing the names or the
positions with which we identify the elements of the reference groups
(control) of the namePheno elements (default a vector  of 1, all the first
groups are selected)</p>
</td></tr>
<tr><td><code id="createObjectMApath_+3A_genesets">geneSets</code></td>
<td>
<p>List of gene sets to check. Object similar
to the one used in the fgsea package</p>
</td></tr>
<tr><td><code id="createObjectMApath_+3A_pathmethod">pathMethod</code></td>
<td>
<p>The single sample enrichment method used to obtain
the enrichment score of each sample and gene set.
See details for more information</p>
</td></tr>
<tr><td><code id="createObjectMApath_+3A_minsize">minSize</code></td>
<td>
<p>Minimum size of the resulting gene sets after gene
identifier mapping. By default, the minimum size is 7.</p>
</td></tr>
<tr><td><code id="createObjectMApath_+3A_kcdf">kcdf</code></td>
<td>
<p>Only neccesary for the GSVA method. Character vector of length 1
denoting the kernel to use during the non-parametric estimation of the
cumulative distribution function of expression levels across samples.
By default, kcdf=&quot;Gaussian&quot; which is suitable when input expression values
are continuous, such as microarray fluorescent units in logarithmic scale,
RNA-seq log-CPMs, log-RPKMs or log-TPMs. When input expression values are
integer counts, such as those derived from RNA-seq experiments, then this
argument should be set to kcdf=&quot;Poisson&quot;.</p>
</td></tr>
<tr><td><code id="createObjectMApath_+3A_normalize">normalize</code></td>
<td>
<p>boolean specifying if the gen set matrices should be
normalized. Default value &quot;TRUE&quot;.</p>
</td></tr>
<tr><td><code id="createObjectMApath_+3A_n.cores">n.cores</code></td>
<td>
<p>Number of cores to use in the parallelization of the datsets.
By default, n.cores=1.</p>
</td></tr>
<tr><td><code id="createObjectMApath_+3A_internal.n.cores">internal.n.cores</code></td>
<td>
<p>Number of cores to use in the parallelization of the
single sample enrichment methods. By default internal.n.cores= 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The single sample scoring methods that can be used to obtain the
enrichment score of each sample and gene set are:
</p>

<ol>
<li><p> &quot;GSVA&quot;: Gene Set Variation method (Hänzelmann S, 2013)
</p>
</li>
<li><p> &quot;Zscore&quot;: Z-score method (Lee E, 2008)
</p>
</li>
<li><p> &quot;ssGSEA&quot;: Single Sample Gene Set Enrichment Analysis method
(Barbie DA, 2009)
</p>
</li>
<li><p> &quot;Singscore&quot;: Single sample scoring of molecular phenotypes
(Foroutan M, 2018)
</p>
</li></ol>

<p>In parallelization, several aspects must be considered.
n.cores refers to the parallelization of studies or datasets.
Therefore, if we have 3 studies, the maximum number for n.cores will be 3.
internal.n.cores refers to the parallelization of single sample enrichment
methods. This is especially recommended for the ssGSEA method. For Singscore
and GSVA, it may also be advisable. The process is parallelized based on the
samples in each study. Therefore, the larger the number of samples, the
slower the process will be.
The number of cores that the computer will use is the multiplication of both
parameters n.cores * internal.n.cores = total cores.
</p>


<h3>Value</h3>

<p>The object needed to perform gene set enrichment meta-analysis.
Each list contains  two elements: The first element is the gene set matrix
(gene sets in rows
and samples in columns) The second element is a vector of zeros and ones
that represents the state of the diffenrent samples of the gene sets
matrix. 0 represents reference group (controls) and 1 represents
experimental group (cases).
</p>


<h3>References</h3>

<p>Hänzelmann S, Castelo R, Guinney J. (2013)
GSVA: gene set variation analysis for microarray and RNA-Seq data.
BMC Bioinformatics. 2013;14: 7. doi:10.1186/1471-2105-14-7
</p>
<p>Lee E, Chuang H-Y, Kim J-W, Ideker T, Lee D. (2008)
Inferring Pathway Activity toward Precise Disease Classification.
PLOS Computational Biology. 2008;4: e1000217.
doi:10.1371/journal.pcbi.1000217
</p>
<p>Barbie DA, Tamayo P, Boehm JS, Kim SY, Moody SE, Dunn IF, et al. (2009)
Systematic RNA interference reveals that oncogenic KRAS-driven cancers
require TBK1. Nature. 2009;462: 108–112. doi:10.1038/nature08460
</p>
<p>Foroutan M, Bhuva DD, Lyu R, Horan K, Cursons J, Davis MJ. (2018)
Single sample scoring of molecular phenotypes. BMC Bioinformatics.
2018;19: 404. doi:10.1186/s12859-018-2435-4
</p>
<p>Korotkevich G, Sukhov V, Budin N, Shpak B, Artyomov MN,
Sergushichev A. (2021)
Fast gene set enrichment analysis. bioRxiv; 2021. p. 060012.
doi:10.1101/060012
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("simulatedData")
listMatrices &lt;- list(study1Ex, study2Ex)
listPhenodata &lt;- list(study1Pheno, study2Pheno)
phenoGroups &lt;- c("Condition","Condition")
phenoCases &lt;- list("Case", "Case")
phenoControls &lt;- list("Healthy", "Healthy")
objectMApathSim &lt;- createObjectMApath(
   listEX = listMatrices,
   listPheno = listPhenodata, namePheno = phenoGroups,
   expGroups = phenoCases, refGroups = phenoControls,
   geneSets = GeneSets,
   pathMethod = "Zscore")



</code></pre>

<hr>
<h2 id='filteringPaths'>Fuction for filtering gene sets with low expression</h2><span id='topic+filteringPaths'></span>

<h3>Description</h3>

<p>This function eliminates gene sets with low expression in both groups in
a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filteringPaths(objectMApath, threshold = 0.65, n_cores = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filteringPaths_+3A_objectmapath">objectMApath</code></td>
<td>
<p>A list of list. Each list contains two elements.
The first element is the Gene Set matrix (gene sets in rows and samples in
columns) and the second element is a vector of zeros and ones that represents
the state of the different samples of the Gene Sets matrix.
0 represents one group (controls) and 1 represents the other group (cases).</p>
</td></tr>
<tr><td><code id="filteringPaths_+3A_threshold">threshold</code></td>
<td>
<p>A number that indicates the threshold to eliminate a gene
set. For a eliminate a gene set is necessary that the median for both groups
are less than the threshold
If  threshold = &quot;sd&quot; the threshold will be the standard deviation of the
gene set. The default value is 0.65.</p>
</td></tr>
<tr><td><code id="filteringPaths_+3A_n_cores">n_cores</code></td>
<td>
<p>A number that indicates the number of cores to use in the
parallelization. The default value is 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The same objectMApath list but with the gene sets that do not meet
the threshold eliminated.
</p>


<h3>Author(s)</h3>

<p>Juan Antonio Villatoro Garcia,
<a href="mailto:juanantoniovillatorogarcia@gmail.com">juanantoniovillatorogarcia@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createObjectMApath">createObjectMApath</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("simulatedData")
newObject &lt;- filteringPaths(objectMApathSim, threshold = "sd")

</code></pre>

<hr>
<h2 id='heatmapPaths'>Visualization of the gene set enrichment meta-analysis results</h2><span id='topic+heatmapPaths'></span>

<h3>Description</h3>

<p>It allows to see how the different significant gene sets are expressed in the
different samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmapPaths(
  objectMApath,
  resMA,
  scaling = c("zscor", "rscale", "swr", "none"),
  regulation = c("all", "up", "down"),
  breaks = c(-2, 2),
  fdrSig = 0.05,
  comES_Sig = 0.5,
  numSig = "all",
  color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
  na_col = "darkgrey",
  legend = TRUE,
  cluster_cols = FALSE,
  cluster_rows = FALSE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  fontsize = 10,
  fontsize_row = fontsize,
  fontsize_col = fontsize
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="heatmapPaths_+3A_objectmapath">objectMApath</code></td>
<td>
<p>A list of list. Each list contains two elements.
The first element is the Gene Set matrix (gene sets in rows and samples in
columns) and the second element is a vector of zeros and ones that represents
the state of the different samples of the Gene Sets matrix.
0 represents one group (controls) and 1 represents the other group (cases).</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_resma">resMA</code></td>
<td>
<p>Output generated by the  function that performs
meta-analysis (metaAnalysisESpath).</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_scaling">scaling</code></td>
<td>
<p>Character variable to choose between different scaling
approaches. See &quot;Details&quot; for more information.</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_regulation">regulation</code></td>
<td>
<p>Character variable that indicates whether we want the
heatmap to show all significant paths (&quot;all&quot;), only the up-regulated paths
(&quot;up&quot;) or only the down-regulated paths(&quot;down&quot;)</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_breaks">breaks</code></td>
<td>
<p>Numeric vector of length 2 that contains the extreme values
(minimum and maximum) of the range of values in which the heatmap
color scale will be distributed. Default a vector By default a
vector of -2 and 2 as extreme values.</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_fdrsig">fdrSig</code></td>
<td>
<p>Adjusted p-value from which a gene set is considered
significant. Default 0.05</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_comes_sig">comES_Sig</code></td>
<td>
<p>In absolute value. Combine effect size
threshold from which gene sets are considered. Default 0.5</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_numsig">numSig</code></td>
<td>
<p>The number of most significant paths to be represented.
If numSig = &quot;all&quot;, all significant paths that meet the selected parameters
will be represented.</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_color">color</code></td>
<td>
<p>vector of colors used in heatmap.</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_na_col">na_col</code></td>
<td>
<p>color of the NA cell in the heatmap.</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_legend">legend</code></td>
<td>
<p>logical to determine if legend should be drawn or not.</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_cluster_cols">cluster_cols</code></td>
<td>
<p>boolean values determining if columns should be
clustered.</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_cluster_rows">cluster_rows</code></td>
<td>
<p>boolean values determining if rows should be clustered.</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_show_rownames">show_rownames</code></td>
<td>
<p>boolean specifying if row names are be shown.</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_show_colnames">show_colnames</code></td>
<td>
<p>boolean specifying if column names are be shown.</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_fontsize">fontsize</code></td>
<td>
<p>base fontsize for the plot</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_fontsize_row">fontsize_row</code></td>
<td>
<p>fontsize for rownames (Default: fontsize)</p>
</td></tr>
<tr><td><code id="heatmapPaths_+3A_fontsize_col">fontsize_col</code></td>
<td>
<p>fontsize for colnames (Default: fontsize)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Scaling approaches that can be used are:
</p>

<ul>
<li><p> &quot;rscale&quot;: it applies rescale function of <em>scales</em> package. Values
will be between -1 and 1)
</p>
</li>
<li><p> &quot;zscor&quot;:  It calculates a z-score value for each gene, that is, the
mean gene expression from each gene is subtracted from each gene expression
value and then it is divided by the standard deviation
</p>
</li>
<li><p> &quot;swr&quot;: it applys scaling relative to reference dataset approach
</p>
</li>
<li><p> &quot;none&quot;: any scaling approach it is applied.
</p>
</li></ul>



<h3>Value</h3>

<p>The matrix represented in the heatmap
</p>


<h3>Author(s)</h3>

<p>Juan Antonio Villatoro Garcia,
<a href="mailto:juanantoniovillatorogarcia@gmail.com">juanantoniovillatorogarcia@gmail.com</a>
</p>


<h3>References</h3>

<p>Hadley Wickham and Dana Seidel (2020). scales: Scale Functions for
Visualization. R package version 1.1.1.
<a href="https://CRAN.R-project.org/package=scales">https://CRAN.R-project.org/package=scales</a>
</p>
<p>Lazar, C, Meganck, S, Taminau, J, and et al. 2013. “Batch Effect
Removal Methods for Microarray Gene Expression Data Integration:
A Survey,” 469–90.
</p>
<p>Raivo Kolde 2019. pheatmap: Pretty Heatmaps. R package version 1.0.12.
<a href="https://CRAN.R-project.org/package=pheatmap">https://CRAN.R-project.org/package=pheatmap</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createObjectMApath">createObjectMApath</a></code>, <code><a href="#topic+metaAnalysisESpath">metaAnalysisESpath</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("simulatedData")
resultsMA &lt;- metaAnalysisESpath(objectMApathSim, typeMethod="REM")
heatmapPaths(objectMA=objectMApathSim, resultsMA,
    scaling = "zscor", regulation = "all",breaks=c(-2,2),
    fdrSig = 0.05, comES_Sig = 1.5, numSig=20)

</code></pre>

<hr>
<h2 id='metaAnalysisESpath'>Performing Gene Set Enrichment Meta-analysis</h2><span id='topic+metaAnalysisESpath'></span>

<h3>Description</h3>

<p>It performs Gene Sets Enrichment meta-analysis by applying Effects size
combination methods
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaAnalysisESpath(
  objectMApath = NULL,
  effectS = NULL,
  measure = c("limma", "SMD", "MD"),
  WithinVarCorrect = TRUE,
  typeMethod = c("REM", "FEM"),
  missAllow = 0.3,
  numData = length(objectMApath)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metaAnalysisESpath_+3A_objectmapath">objectMApath</code></td>
<td>
<p>A list of list. Each list contains two elements.
The first element is the Gene Set matrix (gene sets in rows and samples in
columns) and the second element is a vector of zeros and ones that represents
the state of the different samples of the Gene Sets matrix.
0 represents one group (controls) and 1 represents the other group (cases).</p>
</td></tr>
<tr><td><code id="metaAnalysisESpath_+3A_effects">effectS</code></td>
<td>
<p>A list of two elements. The first element is a dataframe
with gene sets in rows and studies in columns.
Each component of the dataframe is the effect of a gene set in a study.
The second element of the list is also a dataframe
with the same structure, but in this case each component of the dataframe
represent the variance of the effect of a gene set in a study. This argument
should be only used in the case that objectMApath argument is null.</p>
</td></tr>
<tr><td><code id="metaAnalysisESpath_+3A_measure">measure</code></td>
<td>
<p>A character string that indicates the type of effect size to
be calculated. The options are &quot;limma&quot;, &quot;SMD&quot; and &quot;MD&quot;. The default value
is &quot;limma&quot;. See details for more information.</p>
</td></tr>
<tr><td><code id="metaAnalysisESpath_+3A_withinvarcorrect">WithinVarCorrect</code></td>
<td>
<p>A logical value that indicates if the within variance
correction should be applied. The default value is TRUE. See details for more
information.</p>
</td></tr>
<tr><td><code id="metaAnalysisESpath_+3A_typemethod">typeMethod</code></td>
<td>
<p>A character that indicates the method to be performed.
See &quot;Details&quot;for more information</p>
</td></tr>
<tr><td><code id="metaAnalysisESpath_+3A_missallow">missAllow</code></td>
<td>
<p>a number that indicates the maximum proportion of missing
values allowed in a sample. If the sample has more proportion of missing
values the sample will be eliminated. In the other case the missing values
will be imputed using the K-NN algorithm.</p>
</td></tr>
<tr><td><code id="metaAnalysisESpath_+3A_numdata">numData</code></td>
<td>
<p>The minimum number of datasets in which a gene
must be contained to be included in the emta-analysis. 
By default, the gene must be contained in all the datasets. 
If the number entered exceeds the number of studies, the total number of 
studies will be considered.&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are different ways to calculate the effect size of a gene set:
</p>

<ol>
<li><p> &quot;MD&quot;: Raw Mean Difference (Borenstein, 2009)
</p>
</li>
<li><p> &quot;SMD&quot;:  Standardized Mean Difference (Hedges, 1981)
</p>
</li>
<li><p> &quot;limma&quot;: Standardized Mean Difference calculated from the
t-statistics and degrees of freedom obtained by the limma package by
applying the transformation of Rosenthal and Rosnow, 2008). Its calculation
is similar to the one proposed by (Marot et al., 2009) but considering the
transformation of (Rosenthal and Rosnow, 2008).
</p>
</li></ol>

<p>The correction of the variance of the effect size is based on
Lin L, Aloe AM (2021) in which the variance is calculated from the
different estimators.
</p>
<p>The meta-analysis methods that can be applied are:
</p>

<ol>
<li><p> &quot;FEM&quot;: Fixed Effects model
</p>
</li>
<li><p> &quot;REM&quot;: Random Effects model (Default).
</p>
</li></ol>



<h3>Value</h3>

<p>A dataframe with the meta-analysis results. For more information
see the package vignette.
</p>


<h3>Author(s)</h3>

<p>Juan Antonio Villatoro Garcia,
<a href="mailto:juanantoniovillatorogarcia@gmail.com">juanantoniovillatorogarcia@gmail.com</a>
</p>


<h3>References</h3>

<p>Toro-Domínguez D., Villatoro-García J.A.,
Martorell-Marugán J., Román-Montoya Y., Alarcón-Riquelme M.E.,
Carmona-Sáez P. (2020).
A survey of gene expression meta-analysis: methods and applications,
Briefings in Bioinformatics, bbaa019,
<a href="https://doi.org/10.1093/bib/bbaa019">doi:10.1093/bib/bbaa019</a>
</p>
<p>Borenstein, M. (2009). Effect sizes for continuous data. In H. Cooper,
L. V. Hedges, &amp; J. C. Valentine (Eds.),
The handbook of research synthesis and meta-analysis (2nd ed., pp. 221–235).
New York: Russell Sage Foundation.
</p>
<p>Hedges, L. V. (1981). Distribution theory for Glass's estimator of effect
size and related estimators. Journal of Educational Statistics, 6(2),
107–128. <a href="https://doi.org/10.2307/1164588">doi:10.2307/1164588</a>
</p>
<p>Lin L, Aloe AM (2021). Evaluation of various estimators for standardized mean
difference in meta-analysis. Stat Med. 2021 Jan 30;40(2):403-426.
<a href="https://doi.org/10.1002/sim.8781">doi:10.1002/sim.8781</a>
</p>
<p>Marot, G., Foulley, J. L., Mayer, C. D., &amp; Jaffrézic, F. (2009).
Moderated effect size and P-value combinations for microarray meta-analyses.
Bioinformatics. 2692-2699. <a href="https://doi.org/10.1093/bioinformatics/btp444">doi:10.1093/bioinformatics/btp444</a>
</p>
<p>Rosenthal, R., &amp; Rosnow, R. L. (2008). Essentials of behavioral research:
Methods and data analysis. McGraw-Hill.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateESpath">calculateESpath</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("simulatedData")
results &lt;- metaAnalysisESpath(objectMApath = objectMApathSim,
    measure = "limma", typeMethod = "REM")

</code></pre>

<hr>
<h2 id='simulatedData'>GSEMA synthetic data</h2><span id='topic+simulatedData'></span><span id='topic+GeneSets'></span><span id='topic+objectMApathSim'></span><span id='topic+study1Ex'></span><span id='topic+study2Ex'></span><span id='topic+study1Pheno'></span><span id='topic+study2Pheno'></span>

<h3>Description</h3>


<ul>
<li><p> study1Ex, study2Ex: two expression matrices.
</p>
</li>
<li><p> study1Pheno, study2Pheno: two phenodata objects.
</p>
</li>
<li><p> GeneSets: a list of gene sets with each element are the genes that
belong to a pathway.
</p>
</li>
<li><p> objectMApathSim: the meta-analysis object created from the different
expression matrices and phenodatas.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(simulatedData)</code></pre>


<h3>Format</h3>

<p>arrays (<code>study1Ex</code>, <code>study2Ex</code>), 
data.frames (<code>study1Pheno</code>, <code>study2Pheno</code>),
list vectors (<code>GeneSets</code>) and list of nested lists
(<code>objectMApathSim</code>).
</p>


<h3>Source</h3>

<p>study1Ex, study2Ex, study1Pheno and study2Pheno are synthetic.
GeneSets was created from the information of MsigDB  database by adding
a simulated pathway with simulated genes
</p>
<p>objectMApathSim was created after using createObjectMA GSEMA function to
the different studies objects and with the information of GeneSets object
</p>
<p>maObjectDif was created after using createObjectMA DExMA function
to the listExpressionSets object.
maObject was obtained after using allSameID function to maObjectDif
function.
</p>


<h3>References</h3>

<p>Liberzon A, Birger C, Thorvaldsdóttir H, Ghandi M, Mesirov JP, Tamayo P.
The Molecular Signatures Database (MSigDB) hallmark gene set collection.
Cell Syst. 2015;1: 417–425
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
