<!DOCTYPE html><html lang="en"><head><title>Help for package gtexr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gtexr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gtexr-package'><p>gtexr: Query the GTEx Portal API</p></a></li>
<li><a href='#calculate_expression_quantitative_trait_loci'><p>Calculate Expression Quantitative Trait Loci</p></a></li>
<li><a href='#calculate_ieqtls'><p>Calculate Ieqtls</p></a></li>
<li><a href='#calculate_isqtls'><p>Calculate Isqtls</p></a></li>
<li><a href='#calculate_splicing_quantitative_trait_loci'><p>Calculate Splicing Quantitative Trait Loci</p></a></li>
<li><a href='#download'><p>Download</p></a></li>
<li><a href='#get_annotation'><p>Get Annotation</p></a></li>
<li><a href='#get_clustered_median_exon_expression'><p>Get Clustered Median Exon Expression</p></a></li>
<li><a href='#get_clustered_median_gene_expression'><p>Get Clustered Median Gene Expression</p></a></li>
<li><a href='#get_clustered_median_junction_expression'><p>Get Clustered Median Junction Expression</p></a></li>
<li><a href='#get_clustered_median_transcript_expression'><p>Get Clustered Median Transcript Expression</p></a></li>
<li><a href='#get_collapsed_gene_model_exon'><p>Get Collapsed Gene Model Exon</p></a></li>
<li><a href='#get_dataset_info'><p>Get Dataset Info</p></a></li>
<li><a href='#get_downloads_page_data'><p>Get Downloads Page Data</p></a></li>
<li><a href='#get_eqtl_genes'><p>Get Eqtl Genes</p></a></li>
<li><a href='#get_exons'><p>Get Exons</p></a></li>
<li><a href='#get_expression_pca'><p>Get Expression Pca</p></a></li>
<li><a href='#get_file_list'><p>Get File List</p></a></li>
<li><a href='#get_fine_mapping'><p>Get Fine Mapping</p></a></li>
<li><a href='#get_full_get_collapsed_gene_model_exon'><p>Get Full Get Collapsed Gene Model Exon</p></a></li>
<li><a href='#get_functional_annotation'><p>Get Functional Annotation</p></a></li>
<li><a href='#get_gene_expression'><p>Get Gene Expression</p></a></li>
<li><a href='#get_gene_search'><p>Get Gene Search</p></a></li>
<li><a href='#get_genes'><p>Get Genes</p></a></li>
<li><a href='#get_genomic_features'><p>Get Genomic Features</p></a></li>
<li><a href='#get_gwas_catalog_by_location'><p>Get Gwas Catalog By Location</p></a></li>
<li><a href='#get_image'><p>Get Image</p></a></li>
<li><a href='#get_independent_eqtl'><p>Get Independent Eqtl</p></a></li>
<li><a href='#get_linkage_disequilibrium_by_variant_data'><p>Get Linkage Disequilibrium By Variant Data</p></a></li>
<li><a href='#get_linkage_disequilibrium_data'><p>Get Linkage Disequilibrium Data</p></a></li>
<li><a href='#get_maintenance_message'><p>Get Maintenance Message</p></a></li>
<li><a href='#get_median_exon_expression'><p>Get Median Exon Expression</p></a></li>
<li><a href='#get_median_gene_expression'><p>Get Median Gene Expression</p></a></li>
<li><a href='#get_median_junction_expression'><p>Get Median Junction Expression</p></a></li>
<li><a href='#get_median_transcript_expression'><p>Get Median Transcript Expression</p></a></li>
<li><a href='#get_multi_tissue_eqtls'><p>Get Multi Tissue Eqtls</p></a></li>
<li><a href='#get_neighbor_gene'><p>Get Neighbor Gene</p></a></li>
<li><a href='#get_news_item'><p>Get News Item</p></a></li>
<li><a href='#get_sample_biobank_data'><p>Get Sample (Biobank Data)</p></a></li>
<li><a href='#get_sample_datasets_endpoints'><p>Get Sample (Datasets Endpoints)</p></a></li>
<li><a href='#get_service_info'><p>Get Service Info</p></a></li>
<li><a href='#get_significant_single_tissue_eqtls'><p>Get Significant Single Tissue Eqtls</p></a></li>
<li><a href='#get_significant_single_tissue_eqtls_by_location'><p>Get Significant Single Tissue eQTLs By Location</p></a></li>
<li><a href='#get_significant_single_tissue_ieqtls'><p>Get Significant Single Tissue Ieqtls</p></a></li>
<li><a href='#get_significant_single_tissue_isqtls'><p>et Significant Single Tissue Isqtls</p></a></li>
<li><a href='#get_significant_single_tissue_sqtls'><p>Get Significant Single Tissue Sqtls</p></a></li>
<li><a href='#get_single_nucleus_gex'><p>Get Single Nucleux Gex</p></a></li>
<li><a href='#get_single_nucleus_gex_summary'><p>Get Single Nucleus Gex Summary</p></a></li>
<li><a href='#get_sqtl_genes'><p>Get Sqtl Genes</p></a></li>
<li><a href='#get_subject'><p>Get Subject</p></a></li>
<li><a href='#get_tissue_site_detail'><p>Get Tissue Site Detail</p></a></li>
<li><a href='#get_top_expressed_genes'><p>Get Top Expressed Genes</p></a></li>
<li><a href='#get_transcripts'><p>Get Transcripts</p></a></li>
<li><a href='#get_variant'><p>Get Variant</p></a></li>
<li><a href='#get_variant_by_location'><p>Get Variant By Location</p></a></li>
<li><a href='#gtexr_arguments'><p>gtexr arguments</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Query the GTEx Portal API</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A convenient R interface to the Genotype-Tissue Expression (GTEx) 
    Portal API. For more information on the API, see 
    <a href="https://gtexportal.org/api/v2/redoc">https://gtexportal.org/api/v2/redoc</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://rmgpanw.github.io/gtexr/">https://rmgpanw.github.io/gtexr/</a>, <a href="https://github.com/rmgpanw/gtexr">https://github.com/rmgpanw/gtexr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/rmgpanw/gtexr/issues">https://github.com/rmgpanw/gtexr/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>cli, dplyr, httr2 (&ge; 1.0.0), purrr, rlang, tibble, tidyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>curl, httptest2, knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-18 23:52:25 UTC; alasdair</td>
</tr>
<tr>
<td>Author:</td>
<td>Alasdair Warwick <a href="https://orcid.org/0000-0002-0800-2890"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  Benjamin Zuckerman
    <a href="https://orcid.org/0000-0002-0077-6074"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Abraham Olvera-Barrios
    <a href="https://orcid.org/0000-0002-3305-4465"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Chuin Ying Ung <a href="https://orcid.org/0000-0001-8487-4589"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Robert Luben <a href="https://orcid.org/0000-0002-5088-6343"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alasdair Warwick &lt;alasdair.warwick.19@ucl.ac.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-19 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='gtexr-package'>gtexr: Query the GTEx Portal API</h2><span id='topic+gtexr'></span><span id='topic+gtexr-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>A convenient R interface to the Genotype-Tissue Expression (GTEx) Portal API. For more information on the API, see <a href="https://gtexportal.org/api/v2/redoc">https://gtexportal.org/api/v2/redoc</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Alasdair Warwick <a href="mailto:alasdair.warwick.19@ucl.ac.uk">alasdair.warwick.19@ucl.ac.uk</a> (<a href="https://orcid.org/0000-0002-0800-2890">ORCID</a>) [copyright holder]
</p>
<p>Authors:
</p>

<ul>
<li><p> Benjamin Zuckerman (<a href="https://orcid.org/0000-0002-0077-6074">ORCID</a>)
</p>
</li>
<li><p> Abraham Olvera-Barrios (<a href="https://orcid.org/0000-0002-3305-4465">ORCID</a>)
</p>
</li>
<li><p> Chuin Ying Ung (<a href="https://orcid.org/0000-0001-8487-4589">ORCID</a>)
</p>
</li>
<li><p> Robert Luben (<a href="https://orcid.org/0000-0002-5088-6343">ORCID</a>)
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://rmgpanw.github.io/gtexr/">https://rmgpanw.github.io/gtexr/</a>
</p>
</li>
<li> <p><a href="https://github.com/rmgpanw/gtexr">https://github.com/rmgpanw/gtexr</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/rmgpanw/gtexr/issues">https://github.com/rmgpanw/gtexr/issues</a>
</p>
</li></ul>


<hr>
<h2 id='calculate_expression_quantitative_trait_loci'>Calculate Expression Quantitative Trait Loci</h2><span id='topic+calculate_expression_quantitative_trait_loci'></span>

<h3>Description</h3>

<p>Calculate your own eQTLs
</p>

<ul>
<li><p> This service calculates the gene-variant association for any given pair of gene and variant, which may or may not be significant.
</p>
</li>
<li><p> This requires as input a GENCODE ID, GTEx variant ID, and tissue site detail ID.
</p>
</li></ul>

<p>By default, the calculation is based on the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Dynamic-Association-Endpoints/operation/calculate_expression_quantitative_trait_loci_api_v2_association_dyneqtl_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_expression_quantitative_trait_loci(
  tissueSiteDetailId,
  gencodeId,
  variantId,
  datasetId = "gtex_v8"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculate_expression_quantitative_trait_loci_+3A_tissuesitedetailid">tissueSiteDetailId</code></td>
<td>
<p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. &quot;Whole_Blood&quot;; use <code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code>
to see valid values) or an Ontology ID.</p>
</td></tr>
<tr><td><code id="calculate_expression_quantitative_trait_loci_+3A_gencodeid">gencodeId</code></td>
<td>
<p>String. A Versioned GENCODE ID of a gene, e.g.
&quot;ENSG00000065613.9&quot;.</p>
</td></tr>
<tr><td><code id="calculate_expression_quantitative_trait_loci_+3A_variantid">variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td></tr>
<tr><td><code id="calculate_expression_quantitative_trait_loci_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Notes on output:
</p>

<ul>
<li><p> Beta and standard error are recorded in columns <code>nes</code> and <code>error</code> respectively (see <a href="https://gtexportal.org/home/faq#nes_beta">GTEx FAQs</a>)
</p>
</li>
<li> <p><code>variantId</code> contains (in order) chromosome, position, reference allele, alternative allele and human genome build separated by underscores. The reference and alternative alleles for &quot;chr1_13550_G_A_b38&quot; for example are &quot;G&quot; and &quot;A&quot; respectively.
</p>
</li>
<li><p> See examples for how to calculate minor and alternative allele frequencies.
</p>
</li></ul>

<p>Notes on input:
</p>

<ul>
<li><p> Argument <code>variantId</code> also accepts RSIDs.
</p>
</li></ul>



<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Dynamic Association Endpoints: 
<code><a href="#topic+calculate_ieqtls">calculate_ieqtls</a>()</code>,
<code><a href="#topic+calculate_isqtls">calculate_isqtls</a>()</code>,
<code><a href="#topic+calculate_splicing_quantitative_trait_loci">calculate_splicing_quantitative_trait_loci</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# perform request - returns a tibble with a single row
calculate_expression_quantitative_trait_loci(tissueSiteDetailId = "Whole_Blood",
                                             gencodeId = "ENSG00000203782.5",
                                             variantId = "rs79641866")

# unnest list columns with tidyr::unnest()
calculate_expression_quantitative_trait_loci(tissueSiteDetailId = "Whole_Blood",
                                             gencodeId = "ENSG00000203782.5",
                                             variantId = "rs79641866") |&gt;
  tidyr::unnest(c("data", "genotypes"))

# to calculate minor and alternative allele frequencies
calculate_expression_quantitative_trait_loci(
  tissueSiteDetailId = "Liver",
  gencodeId = "ENSG00000237973.1",
  variantId = "rs12119111"
) |&gt;
 dplyr::bind_rows(.id = "rsid") |&gt;

 tidyr::separate(
   col = "variantId",
   into = c(
     "chromosome",
     "position",
     "reference_allele",
     "alternative_allele",
     "genome_build"
   ),
   sep = "_"
 ) |&gt;

 # ...then ascertain alternative_allele frequency
 dplyr::mutate(
   alt_allele_count = (2 * homoAltCount) + hetCount,
   total_allele_count = 2 * (homoAltCount + hetCount +  homoRefCount),
   alternative_allele_frequency = alt_allele_count / total_allele_count
 ) |&gt;

 dplyr::select(
   rsid,
   beta = nes,
   se = error,
   pValue,
   minor_allele_frequency = maf,
   alternative_allele_frequency,
   chromosome:genome_build,
   tissueSiteDetailId
 )

## End(Not run)
</code></pre>

<hr>
<h2 id='calculate_ieqtls'>Calculate Ieqtls</h2><span id='topic+calculate_ieqtls'></span>

<h3>Description</h3>

<p>Calculate your own Cell Specific eQTLs.
</p>

<ul>
<li><p> This service calculates the gene-variant association for any given pair of gene and variant, which may or may not be significant.
</p>
</li>
<li><p> This requires as input a GENCODE ID, GTEx variant ID, and tissue site detail ID.
</p>
</li></ul>

<p>By default, the calculation is based on the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Dynamic-Association-Endpoints/operation/calculate_ieqtls_api_v2_association_dynieqtl_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_ieqtls(
  cellType,
  tissueSiteDetailId,
  gencodeId,
  variantId,
  datasetId = "gtex_v8"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculate_ieqtls_+3A_celltype">cellType</code></td>
<td>
<p>String. &quot;Adipocytes&quot;, &quot;Epithelial_cells&quot;, &quot;Hepatocytes&quot;,
&quot;Keratinocytes&quot;, &quot;Myocytes&quot;, &quot;Neurons&quot;, &quot;Neutrophils&quot;.</p>
</td></tr>
<tr><td><code id="calculate_ieqtls_+3A_tissuesitedetailid">tissueSiteDetailId</code></td>
<td>
<p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. &quot;Whole_Blood&quot;; use <code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code>
to see valid values) or an Ontology ID.</p>
</td></tr>
<tr><td><code id="calculate_ieqtls_+3A_gencodeid">gencodeId</code></td>
<td>
<p>String. A Versioned GENCODE ID of a gene, e.g.
&quot;ENSG00000065613.9&quot;.</p>
</td></tr>
<tr><td><code id="calculate_ieqtls_+3A_variantid">variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td></tr>
<tr><td><code id="calculate_ieqtls_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Dynamic Association Endpoints: 
<code><a href="#topic+calculate_expression_quantitative_trait_loci">calculate_expression_quantitative_trait_loci</a>()</code>,
<code><a href="#topic+calculate_isqtls">calculate_isqtls</a>()</code>,
<code><a href="#topic+calculate_splicing_quantitative_trait_loci">calculate_splicing_quantitative_trait_loci</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# perform request
calculate_ieqtls(cellType = "Adipocytes",
                 tissueSiteDetailId = "Adipose_Subcutaneous",
                 gencodeId = "ENSG00000203782.5",
                 variantId = "chr1_1099341_T_C_b38")

## End(Not run)
</code></pre>

<hr>
<h2 id='calculate_isqtls'>Calculate Isqtls</h2><span id='topic+calculate_isqtls'></span>

<h3>Description</h3>

<p>Calculate your own Cell Specific sQTLs.
</p>

<ul>
<li><p> This service calculates the gene-variant association for any given pair of gene and variant, which may or may not be significant.
</p>
</li>
<li><p> This requires as input a GENCODE ID, GTEx variant ID, and tissue site detail ID.
</p>
</li></ul>

<p>By default, the calculation is based on the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Dynamic-Association-Endpoints/operation/calculate_ieqtls_api_v2_association_dynieqtl_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_isqtls(
  cellType,
  tissueSiteDetailId,
  phenotypeId,
  variantId,
  datasetId = "gtex_v8"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculate_isqtls_+3A_celltype">cellType</code></td>
<td>
<p>String. &quot;Adipocytes&quot;, &quot;Epithelial_cells&quot;, &quot;Hepatocytes&quot;,
&quot;Keratinocytes&quot;, &quot;Myocytes&quot;, &quot;Neurons&quot;, &quot;Neutrophils&quot;.</p>
</td></tr>
<tr><td><code id="calculate_isqtls_+3A_tissuesitedetailid">tissueSiteDetailId</code></td>
<td>
<p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. &quot;Whole_Blood&quot;; use <code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code>
to see valid values) or an Ontology ID.</p>
</td></tr>
<tr><td><code id="calculate_isqtls_+3A_phenotypeid">phenotypeId</code></td>
<td>
<p>String. See
<a href="https://www.gtexportal.org/home/faq#splicingPhenotypeId">GTEx portal FAQs</a>
for further details.</p>
</td></tr>
<tr><td><code id="calculate_isqtls_+3A_variantid">variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td></tr>
<tr><td><code id="calculate_isqtls_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Dynamic Association Endpoints: 
<code><a href="#topic+calculate_expression_quantitative_trait_loci">calculate_expression_quantitative_trait_loci</a>()</code>,
<code><a href="#topic+calculate_ieqtls">calculate_ieqtls</a>()</code>,
<code><a href="#topic+calculate_splicing_quantitative_trait_loci">calculate_splicing_quantitative_trait_loci</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# perform request
calculate_isqtls(cellType = "Neutrophils",
                 tissueSiteDetailId = "Whole_Blood",
                 phenotypeId = "chr1:15947:16607:clu_40980:ENSG00000227232.5",
                 variantId = "chr1_1099341_T_C_b38")

## End(Not run)
</code></pre>

<hr>
<h2 id='calculate_splicing_quantitative_trait_loci'>Calculate Splicing Quantitative Trait Loci</h2><span id='topic+calculate_splicing_quantitative_trait_loci'></span>

<h3>Description</h3>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Dynamic-Association-Endpoints/operation/calculate_splicing_quantitative_trait_loci_api_v2_association_dynsqtl_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_splicing_quantitative_trait_loci(
  tissueSiteDetailId,
  phenotypeId,
  variantId,
  datasetId = "gtex_v8"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculate_splicing_quantitative_trait_loci_+3A_tissuesitedetailid">tissueSiteDetailId</code></td>
<td>
<p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. &quot;Whole_Blood&quot;; use <code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code>
to see valid values) or an Ontology ID.</p>
</td></tr>
<tr><td><code id="calculate_splicing_quantitative_trait_loci_+3A_phenotypeid">phenotypeId</code></td>
<td>
<p>String. See
<a href="https://www.gtexportal.org/home/faq#splicingPhenotypeId">GTEx portal FAQs</a>
for further details.</p>
</td></tr>
<tr><td><code id="calculate_splicing_quantitative_trait_loci_+3A_variantid">variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td></tr>
<tr><td><code id="calculate_splicing_quantitative_trait_loci_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Dynamic Association Endpoints: 
<code><a href="#topic+calculate_expression_quantitative_trait_loci">calculate_expression_quantitative_trait_loci</a>()</code>,
<code><a href="#topic+calculate_ieqtls">calculate_ieqtls</a>()</code>,
<code><a href="#topic+calculate_isqtls">calculate_isqtls</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# perform request - returns a tibble with a single row
calculate_splicing_quantitative_trait_loci(
  tissueSiteDetailId = "Whole_Blood",
  phenotypeId = "chr1:15947:16607:clu_40980:ENSG00000227232.5",
  variantId = "chr1_14677_G_A_b38")

## End(Not run)
</code></pre>

<hr>
<h2 id='download'>Download</h2><span id='topic+download'></span>

<h3>Description</h3>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Biobank-Data-Endpoints/operation/download_api_v2_biobank_download_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download(
  materialTypes = NULL,
  tissueSiteDetailIds = NULL,
  pathCategory = NULL,
  tissueSampleIds = NULL,
  sex = NULL,
  sortBy = NULL,
  sortDirection = NULL,
  searchTerm = NULL,
  sampleIds = NULL,
  subjectIds = NULL,
  ageBrackets = NULL,
  hardyScales = NULL,
  hasExpressionData = NULL,
  hasGenotype = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="download_+3A_materialtypes">materialTypes</code></td>
<td>
<p>String, vector. Options: &quot;Cells:Cell Line Viable&quot;,
&quot;DNA:DNA Genomic&quot;, &quot;DNA:DNA Somatic&quot;, &quot;RNA:Total RNA&quot;, &quot;Tissue:PAXgene
Preserved&quot;, &quot;Tissue:PAXgene Preserved Paraffin-embedded&quot;, &quot;Tissue:Fresh
Frozen Tissue&quot;.</p>
</td></tr>
<tr><td><code id="download_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="download_+3A_pathcategory">pathCategory</code></td>
<td>
<p>Character vector. Options: &quot;adenoma&quot;, &quot;amylacea&quot;,
&quot;atelectasis&quot;, &quot;atherosclerosis&quot;, &quot;atherosis&quot;, &quot;atrophy&quot;, &quot;calcification&quot;,
&quot;cirrhosis&quot;, &quot;clean_specimens&quot;, &quot;congestion&quot;, &quot;corpora_albicantia&quot;, &quot;cyst&quot;,
&quot;desquamation&quot;, &quot;diabetic&quot;, &quot;dysplasia&quot;, &quot;edema&quot;, &quot;emphysema&quot;,
&quot;esophagitis&quot;, &quot;fibrosis&quot;, &quot;gastritis&quot;, &quot;glomerulosclerosis&quot;, &quot;goiter&quot;,
&quot;gynecomastoid&quot;, &quot;hashimoto&quot;, &quot;heart_failure_cells&quot;, &quot;hemorrhage&quot;,
&quot;hepatitis&quot;, &quot;hyalinization&quot;, &quot;hypereosinophilia&quot;, &quot;hyperplasia&quot;,
&quot;hypertrophy&quot;, &quot;hypoxic&quot;, &quot;infarction&quot;, &quot;inflammation&quot;, &quot;ischemic_changes&quot;,
&quot;macrophages&quot;, &quot;mastopathy&quot;, &quot;metaplasia&quot;, &quot;monckeberg&quot;, &quot;necrosis&quot;,
&quot;nephritis&quot;, &quot;nephrosclerosis&quot;, &quot;no_abnormalities&quot;, &quot;nodularity&quot;,
&quot;pancreatitis&quot;, &quot;pigment&quot;, &quot;pneumonia&quot;, &quot;post_menopausal&quot;, &quot;prostatitis&quot;,
&quot;saponification&quot;, &quot;scarring&quot;, &quot;sclerotic&quot;, &quot;solar_elastosis&quot;,
&quot;spermatogenesis&quot;, &quot;steatosis&quot;, &quot;sweat_glands&quot;, &quot;tma&quot;.</p>
</td></tr>
<tr><td><code id="download_+3A_tissuesampleids">tissueSampleIds</code></td>
<td>
<p>Array of strings. A list of Tissue Sample ID(s).</p>
</td></tr>
<tr><td><code id="download_+3A_sex">sex</code></td>
<td>
<p>String. Options: &quot;male&quot;, &quot;female&quot;.</p>
</td></tr>
<tr><td><code id="download_+3A_sortby">sortBy</code></td>
<td>
<p>String. Options: &quot;sampleId&quot;, &quot;ischemicTime&quot;, &quot;aliquotId&quot;,
&quot;tissueSampleId&quot;, &quot;hardyScale&quot;, &quot;pathologyNotes&quot;, &quot;ageBracket&quot;,
&quot;tissueSiteDetailId&quot;, &quot;sex&quot;.</p>
</td></tr>
<tr><td><code id="download_+3A_sortdirection">sortDirection</code></td>
<td>
<p>String. Options: &quot;asc&quot;, &quot;desc&quot;. Default = &quot;asc&quot;.</p>
</td></tr>
<tr><td><code id="download_+3A_searchterm">searchTerm</code></td>
<td>
<p>String.</p>
</td></tr>
<tr><td><code id="download_+3A_sampleids">sampleIds</code></td>
<td>
<p>Character vector. GTEx sample ID.</p>
</td></tr>
<tr><td><code id="download_+3A_subjectids">subjectIds</code></td>
<td>
<p>Character vector. GTEx subject ID.</p>
</td></tr>
<tr><td><code id="download_+3A_agebrackets">ageBrackets</code></td>
<td>
<p>The age bracket(s) of the donors of interest. Options:
&quot;20-29&quot;, &quot;30-39&quot;, &quot;40-49&quot;, &quot;50-59&quot;, &quot;60-69&quot;, &quot;70-79&quot;.</p>
</td></tr>
<tr><td><code id="download_+3A_hardyscales">hardyScales</code></td>
<td>
<p>Character vector. A list of Hardy Scale(s) of interest.
Options: &quot;Ventilator case&quot;, &quot;Fast death - violent&quot;, &quot;Fast death - natural
causes&quot;, &quot;Intermediate death&quot;, &quot;Slow death&quot;.</p>
</td></tr>
<tr><td><code id="download_+3A_hasexpressiondata">hasExpressionData</code></td>
<td>
<p>Logical.</p>
</td></tr>
<tr><td><code id="download_+3A_hasgenotype">hasGenotype</code></td>
<td>
<p>Logical.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: running this request with no filters (i.e. <code>download()</code>) raises an error.
</p>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Biobank Data Endpoints: 
<code><a href="#topic+get_sample_biobank_data">get_sample_biobank_data</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
download(
  materialTypes = "RNA:Total RNA",
  tissueSiteDetailIds = "Thyroid",
  pathCategory = "clean_specimens",
  sex = "male",
  ageBrackets = "50-59"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_annotation'>Get Annotation</h2><span id='topic+get_annotation'></span>

<h3>Description</h3>

<p>This service returns the list of annotations and allowed values
by which a particular dataset can be subsetted. Results may be filtered by
dataset.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_annotation_api_v2_dataset_annotation_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_annotation(datasetId = "gtex_v8", page = 0, itemsPerPage = 250)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_annotation_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_annotation_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_annotation_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: the output for this function appears to be incomplete
currently.
</p>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 get_annotation()

## End(Not run)
</code></pre>

<hr>
<h2 id='get_clustered_median_exon_expression'>Get Clustered Median Exon Expression</h2><span id='topic+get_clustered_median_exon_expression'></span>

<h3>Description</h3>

<p>Find median transcript expression data along with hierarchical
clusters.
</p>

<ul>
<li><p> Returns median normalized transcript expression in tissues of all known transcripts of a given gene along with the hierarchical clustering results of tissues and transcripts, based on exon expression, in Newick format.
</p>
</li>
<li><p> The hierarchical clustering is performed by calculating Euclidean distances and using the average linkage method.
</p>
</li>
<li> <p><strong>This endpoint is not paginated.</strong>
</p>
</li></ul>

<p>By default, this endpoint queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_clustered_median_exon_expression_api_v2_expression_clusteredMedianExonExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clustered_median_exon_expression(
  gencodeIds,
  datasetId = "gtex_v8",
  tissueSiteDetailIds = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_clustered_median_exon_expression_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_clustered_median_exon_expression_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_clustered_median_exon_expression_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble, with clustering data stored as an attribute, &quot;clusters&quot;.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_clustered_median_exon_expression(c("ENSG00000203782.5",
                                       "ENSG00000132693.12"))

# clustering data is stored as an attribute "clusters"
result &lt;- get_clustered_median_exon_expression(c("ENSG00000203782.5",
                                                 "ENSG00000132693.12"))
attr(result, "clusters")

# process clustering data with the ape package
# install.packages("ape")
# phylo_tree &lt;- ape::read.tree(text = attr(result, "clusters")$tissue)
# plot(phylo_tree)
# print(phylo_tree)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_clustered_median_gene_expression'>Get Clustered Median Gene Expression</h2><span id='topic+get_clustered_median_gene_expression'></span>

<h3>Description</h3>

<p>Find median gene expression data along with hierarchical
clusters.
</p>

<ul>
<li><p> Returns median gene expression in tissues along with The hierarchical clustering results of tissues and genes, based on gene expression, in Newick format.
</p>
</li>
<li><p> Results may be filtered by dataset, gene or tissue, but at least one gene must be provided
</p>
</li>
<li><p> The hierarchical clustering is performed by calculating Euclidean distances and using the average linkage method.
</p>
</li>
<li> <p><strong>This endpoint is not paginated.</strong>
</p>
</li></ul>

<p>By default, this service queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_clustered_median_gene_expression_api_v2_expression_clusteredMedianGeneExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clustered_median_gene_expression(
  gencodeIds,
  datasetId = "gtex_v8",
  tissueSiteDetailIds = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_clustered_median_gene_expression_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_clustered_median_gene_expression_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_clustered_median_gene_expression_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble, with clustering data stored as an attribute, &quot;clusters&quot;.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_clustered_median_gene_expression(gencodeIds = c("ENSG00000203782.5",
                                                    "ENSG00000132693.12"))

# clustering data is stored as an attribute "clusters"
result &lt;- get_clustered_median_gene_expression(c("ENSG00000203782.5",
                                                 "ENSG00000132693.12"))
attr(result, "clusters")

# process clustering data with the ape package
# install.packages("ape")
# phylo_tree &lt;- ape::read.tree(text = attr(result, "clusters")$tissue)
# plot(phylo_tree)
# print(phylo_tree)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_clustered_median_junction_expression'>Get Clustered Median Junction Expression</h2><span id='topic+get_clustered_median_junction_expression'></span>

<h3>Description</h3>

<p>Find median junction expression data along with hierarchical
clusters.
</p>

<ul>
<li><p> Returns median junction read counts in tissues of a given gene from all known transcripts along with the hierarchical clustering results of tissues and genes, based on junction expression, in Newick format.
</p>
</li>
<li><p> Results may be filtered by dataset, gene or tissue, but at least one gene must be provided.
</p>
</li>
<li><p> The hierarchical clustering is performed by calculating Euclidean distances and using the average linkage method.
</p>
</li>
<li> <p><strong>This endpoint is not paginated.</strong>
</p>
</li></ul>

<p>By default, this service queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_clustered_median_junction_expression_api_v2_expression_clusteredMedianJunctionExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clustered_median_junction_expression(
  gencodeIds,
  datasetId = "gtex_v8",
  tissueSiteDetailIds = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_clustered_median_junction_expression_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_clustered_median_junction_expression_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_clustered_median_junction_expression_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble, with clustering data stored as an attribute, &quot;clusters&quot;.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_clustered_median_junction_expression(gencodeIds = c("ENSG00000203782.5",
                                                    "ENSG00000132693.12"))

# clustering data is stored as an attribute "clusters"
result &lt;- get_clustered_median_junction_expression(c("ENSG00000203782.5",
                                                 "ENSG00000132693.12"))
attr(result, "clusters")

# process clustering data with the ape package
# install.packages("ape")
# phylo_tree &lt;- ape::read.tree(text = attr(result, "clusters")$tissue)
# plot(phylo_tree)
# print(phylo_tree)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_clustered_median_transcript_expression'>Get Clustered Median Transcript Expression</h2><span id='topic+get_clustered_median_transcript_expression'></span>

<h3>Description</h3>

<p>Find median transcript expression data of all known transcripts of a gene along with hierarchical clusters.
</p>

<ul>
<li><p> Returns median normalized expression in tissues of all known transcripts of a given gene along with the hierarchical clustering results of tissues and genes, based on expression, in Newick format.
</p>
</li>
<li><p> Results may be filtered by dataset, gene or tissue, but at least one gene must be provided.
</p>
</li>
<li><p> The hierarchical clustering is performed by calculating Euclidean distances and using the average linkage method.
</p>
</li>
<li> <p><strong>This endpoint is not paginated.</strong>
</p>
</li></ul>

<p>By default, this service queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_median_transcript_expression_api_v2_expression_medianTranscriptExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clustered_median_transcript_expression(
  gencodeIds,
  datasetId = "gtex_v8",
  tissueSiteDetailIds = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_clustered_median_transcript_expression_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_clustered_median_transcript_expression_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_clustered_median_transcript_expression_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble, with clustering data stored as an attribute, &quot;clusters&quot;.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_clustered_median_transcript_expression(gencodeIds = c("ENSG00000203782.5",
                                                    "ENSG00000132693.12"))

# clustering data is stored as an attribute "clusters"
result &lt;- get_clustered_median_transcript_expression(c("ENSG00000203782.5",
                                                 "ENSG00000132693.12"))
attr(result, "clusters")

# process clustering data with the ape package
# install.packages("ape")
# phylo_tree &lt;- ape::read.tree(text = attr(result, "clusters")$tissue)
# plot(phylo_tree)
# print(phylo_tree)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_collapsed_gene_model_exon'>Get Collapsed Gene Model Exon</h2><span id='topic+get_collapsed_gene_model_exon'></span>

<h3>Description</h3>

<p>This service returns the collapsed exons in the gene model of
the given gene. Gene-level and exon-level expression quantification were
based on the GENCODE annotation, collapsed to a single transcript model for
each gene using an algorithm developed by the GTEx analysis team.
</p>
<p>By default, this service queries the models used by the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_collapsed_gene_model_exon_api_v2_dataset_collapsedGeneModelExon_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_collapsed_gene_model_exon(
  gencodeId,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_collapsed_gene_model_exon_+3A_gencodeid">gencodeId</code></td>
<td>
<p>String. A Versioned GENCODE ID of a gene, e.g.
&quot;ENSG00000065613.9&quot;.</p>
</td></tr>
<tr><td><code id="get_collapsed_gene_model_exon_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_collapsed_gene_model_exon_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_collapsed_gene_model_exon_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 get_collapsed_gene_model_exon(gencodeId = "ENSG00000132693.12")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_dataset_info'>Get Dataset Info</h2><span id='topic+get_dataset_info'></span>

<h3>Description</h3>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Metadata-Endpoints/operation/get_dataset_info_api_v2_metadata_dataset_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_dataset_info(datasetId = "gtex_v8", organizationName = "GTEx Consortium")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_dataset_info_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_dataset_info_+3A_organizationname">organizationName</code></td>
<td>
<p>String. Options: &quot;GTEx Consortium&quot; &quot;Kid's First&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: this current only returns empty values.
</p>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_dataset_info()

## End(Not run)
</code></pre>

<hr>
<h2 id='get_downloads_page_data'>Get Downloads Page Data</h2><span id='topic+get_downloads_page_data'></span>

<h3>Description</h3>

<p>Retrieves all the files belonging to the given <code>project_id</code> for
display on the <code style="white-space: pre;">&#8288;Downloads Page&#8288;</code>
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_downloads_page_data_api_v2_dataset_openAccessFilesMetadata_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_downloads_page_data(project_id = "adult-gtex")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_downloads_page_data_+3A_project_id">project_id</code></td>
<td>
<p>String. Options: &quot;gtex&quot;, &quot;adult-gtex&quot;, &quot;egtex&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: The GTEx Portal API documentation states &quot;GTEx currently has
one project available: gtex&quot;. However, <code>project_id</code> values &quot;adult-gtex&quot; and
&quot;egtex&quot; both return results, whereas &quot;gtex&quot; does not (see examples).
</p>


<h3>Value</h3>

<p>A Tibble
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 # "adult-gtex" (default `project_id` value) and "egtex" both return results
 get_downloads_page_data()
 egtex &lt;- get_downloads_page_data("egtex")
 egtex

 # ..."gtex" does not
 get_downloads_page_data("gtex")

 # get details for whole blood methylation data, including download URL
 purrr::pluck(
   egtex$children,
   1,
   "folders",
   "Methylation - EPIC Array",
   "children",
   "folders",
   "mQTLs",
   "children",
   "files",
   "WholeBlood.mQTLs.regular.txt.gz"
 )

## End(Not run)
</code></pre>

<hr>
<h2 id='get_eqtl_genes'>Get Eqtl Genes</h2><span id='topic+get_eqtl_genes'></span>

<h3>Description</h3>

<p>Retrieve eGenes (eQTL Genes).
</p>

<ul>
<li><p> This service returns eGenes (eQTL Genes) from the specified dataset.
</p>
</li>
<li><p> eGenes are genes that have at least one significant cis-eQTL acting upon them.
</p>
</li>
<li><p> Results may be filtered by tissue. By default, the service queries the latest GTEx release.
</p>
</li></ul>

<p>For each eGene, the results include the allelic fold change
(log2AllelicFoldChange), p-value (pValue), p-value threshold
(pValueThreshold), empirical p-value (empiricalPValue), and q-value
(qValue).
</p>

<ul>
<li><p> The log2AllelicFoldChange is the allelic fold change (in log2 scale) of the most significant eQTL.
</p>
</li>
<li><p> The pValue is the nominal p-value of the most significant eQTL.
</p>
</li>
<li><p> The pValueThreshold is the p-value threshold used to determine whether a cis-eQTL for this gene is significant. For more details see https://gtexportal.org/home/documentationPage#staticTextAnalysisMethods.
</p>
</li>
<li><p> The empiricalPValue is the beta distribution-adjusted empirical p-value from FastQTL.
</p>
</li>
<li><p> The qValues were calculated based on the empirical p-values. A false discovery rate (FDR) threshold of &lt;= 0.05 was applied to identify genes with a significant eQTL.
</p>
</li></ul>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_eqtl_genes_api_v2_association_egene_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_eqtl_genes(
  tissueSiteDetailIds,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_eqtl_genes_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_eqtl_genes_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_eqtl_genes_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_eqtl_genes_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code><a href="#topic+get_fine_mapping">get_fine_mapping</a>()</code>,
<code><a href="#topic+get_independent_eqtl">get_independent_eqtl</a>()</code>,
<code><a href="#topic+get_multi_tissue_eqtls">get_multi_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls_by_location">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_ieqtls">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_isqtls">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_sqtls">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="#topic+get_sqtl_genes">get_sqtl_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_eqtl_genes(c("Whole_Blood", "Artery_Aorta"))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_exons'>Get Exons</h2><span id='topic+get_exons'></span>

<h3>Description</h3>

<p>This service returns exons from all known transcripts of the given gene.
</p>

<ul>
<li><p> A versioned GENCODE ID is required to ensure that all exons are from a single gene.
</p>
</li>
<li><p> A dataset ID or both GENCODE version and genome build must be provided.
</p>
</li>
<li><p> Although annotated exons are not dataset dependent, specifying a dataset here is equivalent to specifying the GENCODE version and genome build used by that dataset.
</p>
</li></ul>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Reference-Genome-Endpoints/operation/get_exons_api_v2_reference_exon_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_exons(
  gencodeId,
  gencodeVersion = NULL,
  genomeBuild = NULL,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_exons_+3A_gencodeid">gencodeId</code></td>
<td>
<p>String. A Versioned GENCODE ID of a gene, e.g.
&quot;ENSG00000065613.9&quot;.</p>
</td></tr>
<tr><td><code id="get_exons_+3A_gencodeversion">gencodeVersion</code></td>
<td>
<p>String (default = &quot;v26&quot;). GENCODE annotation release.
Either &quot;v26&quot; or &quot;v19&quot;.</p>
</td></tr>
<tr><td><code id="get_exons_+3A_genomebuild">genomeBuild</code></td>
<td>
<p>String. Options: &quot;GRCh38/hg38&quot;, &quot;GRCh37/hg19&quot;. Default =
&quot;GRCh38/hg38&quot;.</p>
</td></tr>
<tr><td><code id="get_exons_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_exons_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_exons_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Reference Genome Endpoints: 
<code><a href="#topic+get_gene_search">get_gene_search</a>()</code>,
<code><a href="#topic+get_genes">get_genes</a>()</code>,
<code><a href="#topic+get_genomic_features">get_genomic_features</a>()</code>,
<code><a href="#topic+get_gwas_catalog_by_location">get_gwas_catalog_by_location</a>()</code>,
<code><a href="#topic+get_neighbor_gene">get_neighbor_gene</a>()</code>,
<code><a href="#topic+get_transcripts">get_transcripts</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_exons(gencodeId = "ENSG00000203782.5")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_expression_pca'>Get Expression Pca</h2><span id='topic+get_expression_pca'></span>

<h3>Description</h3>

<p>Find gene expression PCA data.
</p>

<ul>
<li><p> Returns gene expression PCA (principal component analysis) in tissues.
</p>
</li>
<li><p> Results may be filtered by tissue, sample, or dataset.
</p>
</li></ul>

<p>By default, the service queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_expression_pca_api_v2_expression_expressionPca_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_expression_pca(
  tissueSiteDetailIds,
  datasetId = "gtex_v8",
  sampleId = NULL,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_expression_pca_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_expression_pca_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_expression_pca_+3A_sampleid">sampleId</code></td>
<td>
<p>String. <code style="white-space: pre;">&#8288;^GTEX-[A-Z0-9]{5}-[0-9]{4}-SM-[A-Z0-9]{5}$&#8288;</code></p>
</td></tr>
<tr><td><code id="get_expression_pca_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_expression_pca_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_expression_pca(tissueSiteDetailIds = c("Adipose_Subcutaneous",
                                           "Whole_Blood"))

get_expression_pca(tissueSiteDetailIds = "Adipose_Subcutaneous",
                   sampleId = "GTEX-1117F-0226-SM-5GZZ7")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_file_list'>Get File List</h2><span id='topic+get_file_list'></span>

<h3>Description</h3>

<p>Get all the files in GTEx dataset for Download page
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_file_list_api_v2_dataset_fileList_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_file_list()
</code></pre>


<h3>Details</h3>

<p>The returned tibble includes a nested list column, &quot;filesets&quot;. This
details files, sub-categorised by fileset (see examples section).
</p>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Column "filesets" is a list column
get_file_list()

# Get "GTEx Analysis V9" file list
gtex_v9_files &lt;- get_file_list() |&gt;
                   dplyr::filter(name == "GTEx Analysis V9") |&gt;
                   dplyr::pull(filesets)

# "GTEx Analysis V9" filesets
names(gtex_v9_files[[1]])

# "GTEx Analysis V9", "snRNA-Seq Data" fileset files
names(gtex_v9_files[[1]][["snRNA-Seq Data"]]$files)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_fine_mapping'>Get Fine Mapping</h2><span id='topic+get_fine_mapping'></span>

<h3>Description</h3>

<p>Retrieve Fine Mapping Data
</p>

<ul>
<li><p> Finds and returns <code style="white-space: pre;">&#8288;Fine Mapping&#8288;</code> data for the provided list of genes
</p>
</li>
<li><p> By default, this endpoint fetches data from the latest <code>GTEx</code> version
</p>
</li></ul>

<p>The retrieved data is split into pages with <code>items_per_page</code> entries per page
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_fine_mapping_api_v2_association_fineMapping_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_fine_mapping(
  gencodeIds,
  datasetId = "gtex_v8",
  variantId = NULL,
  tissueSiteDetailIds = NULL,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_fine_mapping_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_fine_mapping_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_fine_mapping_+3A_variantid">variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td></tr>
<tr><td><code id="get_fine_mapping_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_fine_mapping_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_fine_mapping_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code><a href="#topic+get_eqtl_genes">get_eqtl_genes</a>()</code>,
<code><a href="#topic+get_independent_eqtl">get_independent_eqtl</a>()</code>,
<code><a href="#topic+get_multi_tissue_eqtls">get_multi_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls_by_location">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_ieqtls">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_isqtls">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_sqtls">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="#topic+get_sqtl_genes">get_sqtl_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search by gene
get_fine_mapping(gencodeIds = c("ENSG00000132693.12",
                               "ENSG00000203782.5"))

# optionally filter for a single variant and/or one or more tissues
get_fine_mapping(gencodeIds = c("ENSG00000132693.12",
                               "ENSG00000203782.5"),
                 variantId = "chr1_153228363_A_G_b38",
                 tissueSiteDetailIds = c("Whole_Blood",
                                        "Thyroid"))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_full_get_collapsed_gene_model_exon'>Get Full Get Collapsed Gene Model Exon</h2><span id='topic+get_full_get_collapsed_gene_model_exon'></span>

<h3>Description</h3>

<p>This service allows the user to query the full Collapsed Gene
Model Exon of a specific gene by gencode ID
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_full_get_collapsed_gene_model_exon_api_v2_dataset_fullCollapsedGeneModelExon_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_full_get_collapsed_gene_model_exon(gencodeId, page = 0, itemsPerPage = 250)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_full_get_collapsed_gene_model_exon_+3A_gencodeid">gencodeId</code></td>
<td>
<p>String. A Versioned GENCODE ID of a gene, e.g.
&quot;ENSG00000065613.9&quot;.</p>
</td></tr>
<tr><td><code id="get_full_get_collapsed_gene_model_exon_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_full_get_collapsed_gene_model_exon_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  get_full_get_collapsed_gene_model_exon(gencodeId = "ENSG00000203782.5")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_functional_annotation'>Get Functional Annotation</h2><span id='topic+get_functional_annotation'></span>

<h3>Description</h3>

<p>This endpoint retrieves the functional annotation of a certain
chromosome location. Default to most recent dataset release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_full_get_collapsed_gene_model_exon_api_v2_dataset_fullCollapsedGeneModelExon_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_functional_annotation(
  datasetId = "gtex_v8",
  chromosome,
  start,
  end,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_functional_annotation_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_functional_annotation_+3A_chromosome">chromosome</code></td>
<td>
<p>String. One of &quot;chr1&quot;, &quot;chr2&quot;, &quot;chr3&quot;, &quot;chr4&quot;, &quot;chr5&quot;,
&quot;chr6&quot;, &quot;chr7&quot;, &quot;chr8&quot;, &quot;chr9&quot;, &quot;chr10&quot;, &quot;chr11&quot;, &quot;chr12&quot;, &quot;chr13&quot;,
&quot;chr14&quot;, &quot;chr15&quot;, &quot;chr16&quot;, &quot;chr17&quot;, &quot;chr18&quot;, &quot;chr19&quot;, &quot;chr20&quot;, &quot;chr21&quot;,
&quot;chr22&quot;, &quot;chrM&quot;, &quot;chrX&quot;, &quot;chrY&quot;.</p>
</td></tr>
<tr><td><code id="get_functional_annotation_+3A_start">start</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="get_functional_annotation_+3A_end">end</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="get_functional_annotation_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_functional_annotation_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  get_functional_annotation(chromosome = "chr1", start = 192168000, end = 192169000)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_gene_expression'>Get Gene Expression</h2><span id='topic+get_gene_expression'></span>

<h3>Description</h3>

<p>Find normalized gene expression data.
</p>

<ul>
<li><p> Returns normalized gene expression in tissues at the sample level.
</p>
</li>
<li><p> Results may be filtered by dataset, gene or tissue, but at least one gene must be provided.
</p>
</li></ul>

<p>By default, this service queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_gene_expression_api_v2_expression_geneExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene_expression(
  gencodeIds,
  datasetId = "gtex_v8",
  tissueSiteDetailIds = NULL,
  attributeSubset = NULL,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_gene_expression_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_gene_expression_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_gene_expression_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_gene_expression_+3A_attributesubset">attributeSubset</code></td>
<td>
<p>String. Examples include but are not limited to &quot;sex&quot;, &quot;ageBracket&quot;</p>
</td></tr>
<tr><td><code id="get_gene_expression_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_gene_expression_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  # multiple genes, selected tissues
  get_gene_expression(gencodeIds = c("ENSG00000132693.12",
                                     "ENSG00000203782.5"),
                      tissueSiteDetailIds = c("Thyroid", "Whole_Blood"))

  # single gene, selected (single) tissue
  get_gene_expression(gencodeIds = "ENSG00000132693.12",
                      tissueSiteDetailIds = "Whole_Blood")

  # subset by sex
  get_gene_expression(gencodeIds = "ENSG00000132693.12",
                      tissueSiteDetailIds = "Whole_Blood",
                      attributeSubset = "sex")

  # subset by age bracket
  get_gene_expression(gencodeIds = "ENSG00000132693.12",
                      tissueSiteDetailIds = "Whole_Blood",
                      attributeSubset = "ageBracket")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_gene_search'>Get Gene Search</h2><span id='topic+get_gene_search'></span>

<h3>Description</h3>

<p>Find genes that are partial or complete match of a gene_id
</p>

<ul>
<li><p> gene_id could be a gene symbol, a gencode ID, or an Ensemble ID
</p>
</li>
<li><p> Gencode Version and Genome Build must be specified
</p>
</li></ul>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Reference-Genome-Endpoints/operation/get_gene_search_api_v2_reference_geneSearch_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene_search(
  geneId,
  gencodeVersion = "v26",
  genomeBuild = "GRCh38/hg38",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_gene_search_+3A_geneid">geneId</code></td>
<td>
<p>String. A gene symbol, a gencode ID, or an Ensemble ID.</p>
</td></tr>
<tr><td><code id="get_gene_search_+3A_gencodeversion">gencodeVersion</code></td>
<td>
<p>String (default = &quot;v26&quot;). GENCODE annotation release.
Either &quot;v26&quot; or &quot;v19&quot;.</p>
</td></tr>
<tr><td><code id="get_gene_search_+3A_genomebuild">genomeBuild</code></td>
<td>
<p>String. Options: &quot;GRCh38/hg38&quot;, &quot;GRCh37/hg19&quot;. Default =
&quot;GRCh38/hg38&quot;.</p>
</td></tr>
<tr><td><code id="get_gene_search_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_gene_search_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Reference Genome Endpoints: 
<code><a href="#topic+get_exons">get_exons</a>()</code>,
<code><a href="#topic+get_genes">get_genes</a>()</code>,
<code><a href="#topic+get_genomic_features">get_genomic_features</a>()</code>,
<code><a href="#topic+get_gwas_catalog_by_location">get_gwas_catalog_by_location</a>()</code>,
<code><a href="#topic+get_neighbor_gene">get_neighbor_gene</a>()</code>,
<code><a href="#topic+get_transcripts">get_transcripts</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_gene_search("CRP")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_genes'>Get Genes</h2><span id='topic+get_genes'></span>

<h3>Description</h3>

<p>This service returns information about reference genes. A genome
build and GENCODE version must be provided.
</p>

<ul>
<li><p> Genes are searchable by gene symbol, GENCODE ID and versioned GENCODE ID.
</p>
</li>
<li><p> Versioned GENCODE ID is recommended to ensure unique ID matching.
</p>
</li>
<li><p> By default, this service queries the genome build and GENCODE version used by the latest GTEx release.
</p>
</li></ul>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Reference-Genome-Endpoints/operation/get_genes_api_v2_reference_gene_get">GTEx API Portal documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genes(
  geneIds,
  gencodeVersion = "v26",
  genomeBuild = "GRCh38/hg38",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_genes_+3A_geneids">geneIds</code></td>
<td>
<p>A character vector of gene symbols, versioned gencodeIds, or
unversioned gencodeIds.</p>
</td></tr>
<tr><td><code id="get_genes_+3A_gencodeversion">gencodeVersion</code></td>
<td>
<p>String (default = &quot;v26&quot;). GENCODE annotation release.
Either &quot;v26&quot; or &quot;v19&quot;.</p>
</td></tr>
<tr><td><code id="get_genes_+3A_genomebuild">genomeBuild</code></td>
<td>
<p>String. Options: &quot;GRCh38/hg38&quot;, &quot;GRCh37/hg19&quot;. Default =
&quot;GRCh38/hg38&quot;.</p>
</td></tr>
<tr><td><code id="get_genes_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_genes_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Reference Genome Endpoints: 
<code><a href="#topic+get_exons">get_exons</a>()</code>,
<code><a href="#topic+get_gene_search">get_gene_search</a>()</code>,
<code><a href="#topic+get_genomic_features">get_genomic_features</a>()</code>,
<code><a href="#topic+get_gwas_catalog_by_location">get_gwas_catalog_by_location</a>()</code>,
<code><a href="#topic+get_neighbor_gene">get_neighbor_gene</a>()</code>,
<code><a href="#topic+get_transcripts">get_transcripts</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_genes(c("CRP", "IL6R"))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_genomic_features'>Get Genomic Features</h2><span id='topic+get_genomic_features'></span>

<h3>Description</h3>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Reference-Genome-Endpoints/operation/get_genomic_features_api_v2_reference_features__featureId__get">GTEx API Portal documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genomic_features(.featureId, datasetId = "gtex_v8")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_genomic_features_+3A_.featureid">.featureId</code></td>
<td>
<p>String. A genomic feature e.g. GENCODE ID, RSID or
GTEx Variant ID.</p>
</td></tr>
<tr><td><code id="get_genomic_features_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This endpoint takes a path parameter &quot;featureId&quot;.
</p>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Reference Genome Endpoints: 
<code><a href="#topic+get_exons">get_exons</a>()</code>,
<code><a href="#topic+get_gene_search">get_gene_search</a>()</code>,
<code><a href="#topic+get_genes">get_genes</a>()</code>,
<code><a href="#topic+get_gwas_catalog_by_location">get_gwas_catalog_by_location</a>()</code>,
<code><a href="#topic+get_neighbor_gene">get_neighbor_gene</a>()</code>,
<code><a href="#topic+get_transcripts">get_transcripts</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# gene symbol
get_genomic_features("brca1")

# GENCODE ID
get_genomic_features("ENSG00000132693.12")

# RSID
get_genomic_features("rs1815739")

# GTEx variant ID
get_genomic_features("chr11_66561023_G_GTTA_b38")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_gwas_catalog_by_location'>Get Gwas Catalog By Location</h2><span id='topic+get_gwas_catalog_by_location'></span>

<h3>Description</h3>

<p>Find the GWAS Catalog on a certain chromosome between start and
end locations.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Reference-Genome-Endpoints/operation/get_gwas_catalog_by_location_api_v2_reference_gwasCatalogByLocation_get">GTEx API Portal documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gwas_catalog_by_location(
  start,
  end,
  chromosome,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_gwas_catalog_by_location_+3A_start">start</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="get_gwas_catalog_by_location_+3A_end">end</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="get_gwas_catalog_by_location_+3A_chromosome">chromosome</code></td>
<td>
<p>String. One of &quot;chr1&quot;, &quot;chr2&quot;, &quot;chr3&quot;, &quot;chr4&quot;, &quot;chr5&quot;,
&quot;chr6&quot;, &quot;chr7&quot;, &quot;chr8&quot;, &quot;chr9&quot;, &quot;chr10&quot;, &quot;chr11&quot;, &quot;chr12&quot;, &quot;chr13&quot;,
&quot;chr14&quot;, &quot;chr15&quot;, &quot;chr16&quot;, &quot;chr17&quot;, &quot;chr18&quot;, &quot;chr19&quot;, &quot;chr20&quot;, &quot;chr21&quot;,
&quot;chr22&quot;, &quot;chrM&quot;, &quot;chrX&quot;, &quot;chrY&quot;.</p>
</td></tr>
<tr><td><code id="get_gwas_catalog_by_location_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_gwas_catalog_by_location_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Reference Genome Endpoints: 
<code><a href="#topic+get_exons">get_exons</a>()</code>,
<code><a href="#topic+get_gene_search">get_gene_search</a>()</code>,
<code><a href="#topic+get_genes">get_genes</a>()</code>,
<code><a href="#topic+get_genomic_features">get_genomic_features</a>()</code>,
<code><a href="#topic+get_neighbor_gene">get_neighbor_gene</a>()</code>,
<code><a href="#topic+get_transcripts">get_transcripts</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_gwas_catalog_by_location(start = 1, end = 10000000, chromosome = "chr1")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_image'>Get Image</h2><span id='topic+get_image'></span>

<h3>Description</h3>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Histology-Endpoints/operation/get_image_api_v2_histology_image_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_image(tissueSampleIds = NULL, page = 0, itemsPerPage = 250)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_image_+3A_tissuesampleids">tissueSampleIds</code></td>
<td>
<p>Array of strings. A list of Tissue Sample ID(s).</p>
</td></tr>
<tr><td><code id="get_image_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_image_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_image()

# filter by `tissueSampleId`
result &lt;- get_image(tissueSampleIds = "GTEX-1117F-0526")
print(result)

# note that `pathologyNotesCategories` (if present) is a list column
print(result$pathologyNotesCategories)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_independent_eqtl'>Get Independent Eqtl</h2><span id='topic+get_independent_eqtl'></span>

<h3>Description</h3>

<p>Retrieve Independent eQTL Data
</p>

<ul>
<li><p> Finds and returns <code style="white-space: pre;">&#8288;Independent eQTL Data&#8288;</code> data for the provided list of genes
</p>
</li>
<li><p> By default, this endpoint fetches data from the latest <code>GTEx</code> version
</p>
</li></ul>

<p>The retrieved data is split into pages with <code>items_per_page</code> entries per page
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_independent_eqtl_api_v2_association_independentEqtl_get">GTEx portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_independent_eqtl(
  gencodeIds,
  tissueSiteDetailIds = NULL,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_independent_eqtl_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_independent_eqtl_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_independent_eqtl_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_independent_eqtl_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_independent_eqtl_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code><a href="#topic+get_eqtl_genes">get_eqtl_genes</a>()</code>,
<code><a href="#topic+get_fine_mapping">get_fine_mapping</a>()</code>,
<code><a href="#topic+get_multi_tissue_eqtls">get_multi_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls_by_location">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_ieqtls">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_isqtls">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_sqtls">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="#topic+get_sqtl_genes">get_sqtl_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search by gene
get_independent_eqtl(gencodeIds = c("ENSG00000132693.12",
                               "ENSG00000203782.5"))

# optionally filter for a single variant and/or one or more tissues
get_independent_eqtl(gencodeIds = c("ENSG00000132693.12",
                                    "ENSG00000203782.5"),
                     tissueSiteDetailIds = c("Whole_Blood",
                                             "Thyroid"))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_linkage_disequilibrium_by_variant_data'>Get Linkage Disequilibrium By Variant Data</h2><span id='topic+get_linkage_disequilibrium_by_variant_data'></span>

<h3>Description</h3>

<p>Find linkage disequilibrium (LD) data for a given variant
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_linkage_disequilibrium_by_variant_data_api_v2_dataset_ldByVariant_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_linkage_disequilibrium_by_variant_data(
  variantId,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_linkage_disequilibrium_by_variant_data_+3A_variantid">variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td></tr>
<tr><td><code id="get_linkage_disequilibrium_by_variant_data_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_linkage_disequilibrium_by_variant_data_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_linkage_disequilibrium_by_variant_data("chr1_159245536_C_T_b38")
</code></pre>

<hr>
<h2 id='get_linkage_disequilibrium_data'>Get Linkage Disequilibrium Data</h2><span id='topic+get_linkage_disequilibrium_data'></span>

<h3>Description</h3>

<p>Find linkage disequilibrium (LD) data for a given gene.
</p>
<p>This endpoint returns linkage disequilibrium data for the cis-eQTLs found
associated with the provided gene in a specified dataset. Results are queried
by gencode ID. By default, the service queries the latest GTEx release.
Specify a dataset ID to fetch results from a different dataset.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_variant_by_location_api_v2_dataset_variantByLocation_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_linkage_disequilibrium_data(
  gencodeId,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_linkage_disequilibrium_data_+3A_gencodeid">gencodeId</code></td>
<td>
<p>String. A Versioned GENCODE ID of a gene, e.g.
&quot;ENSG00000065613.9&quot;.</p>
</td></tr>
<tr><td><code id="get_linkage_disequilibrium_data_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_linkage_disequilibrium_data_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_linkage_disequilibrium_data_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_linkage_disequilibrium_data(gencodeId = "ENSG00000132693.12")
</code></pre>

<hr>
<h2 id='get_maintenance_message'>Get Maintenance Message</h2><span id='topic+get_maintenance_message'></span>

<h3>Description</h3>

<p>Getting all the maintenance messages from the database that are
enabled.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Admin-Endpoints/operation/get_maintenance_message_api_v2_admin_maintenanceMessage_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_maintenance_message(page = 0, itemsPerPage = 250)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_maintenance_message_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_maintenance_message_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note this typically returns an empty tibble.
</p>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Admin Endpoints: 
<code><a href="#topic+get_news_item">get_news_item</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_maintenance_message()

## End(Not run)
</code></pre>

<hr>
<h2 id='get_median_exon_expression'>Get Median Exon Expression</h2><span id='topic+get_median_exon_expression'></span>

<h3>Description</h3>

<p>Find median exon expression data.
</p>

<ul>
<li><p> Returns median exon read counts, in tissues, of a collapsed gene model.
</p>
</li>
<li><p> Results may be filtered by dataset, gene or tissue, but at least one gene must be provided
</p>
</li></ul>

<p>By default, this service queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_median_exon_expression_api_v2_expression_medianExonExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_median_exon_expression(
  gencodeIds,
  datasetId = "gtex_v8",
  tissueSiteDetailIds = NULL,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_median_exon_expression_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_median_exon_expression_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_median_exon_expression_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_median_exon_expression_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_median_exon_expression_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# median exon expression values for CRP, filtered for whole blood
get_median_exon_expression(gencodeIds = "ENSG00000132693.12",
                           tissueSiteDetailIds = "Whole_Blood")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_median_gene_expression'>Get Median Gene Expression</h2><span id='topic+get_median_gene_expression'></span>

<h3>Description</h3>

<p>Find median gene expression data along with hierarchical clusters.
</p>

<ul>
<li><p> Returns median gene expression in tissues.
</p>
</li>
<li><p> By default, this endpoint queries the latest GTEx release.
</p>
</li></ul>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_median_gene_expression_api_v2_expression_medianGeneExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_median_gene_expression(
  gencodeIds,
  datasetId = "gtex_v8",
  tissueSiteDetailIds = NULL,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_median_gene_expression_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_median_gene_expression_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_median_gene_expression_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_median_gene_expression_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_median_gene_expression_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_median_gene_expression(gencodeIds = "ENSG00000132693.12")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_median_junction_expression'>Get Median Junction Expression</h2><span id='topic+get_median_junction_expression'></span>

<h3>Description</h3>

<p>Find junction gene expression data.
</p>

<ul>
<li><p> Returns median junction read counts in tissues of a given gene from all known transcripts.
</p>
</li>
<li><p> Results may be filtered by dataset or tissue.
</p>
</li></ul>

<p>By default, this service queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_median_junction_expression_api_v2_expression_medianJunctionExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_median_junction_expression(
  gencodeIds,
  datasetId = "gtex_v8",
  tissueSiteDetailIds = NULL,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_median_junction_expression_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_median_junction_expression_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_median_junction_expression_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_median_junction_expression_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_median_junction_expression_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_median_junction_expression(gencodeIds = "ENSG00000132693.12")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_median_transcript_expression'>Get Median Transcript Expression</h2><span id='topic+get_median_transcript_expression'></span>

<h3>Description</h3>

<p>Find median transcript expression data of all known transcripts
of a gene.
</p>

<ul>
<li><p> Returns median normalized expression in tissues of all known transcripts of a given gene.
</p>
</li>
<li><p> Results may be filtered by dataset or tissue.
</p>
</li></ul>

<p>By default, this service queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_median_transcript_expression_api_v2_expression_medianTranscriptExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_median_transcript_expression(
  gencodeIds,
  datasetId = "gtex_v8",
  tissueSiteDetailIds = NULL,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_median_transcript_expression_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_median_transcript_expression_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_median_transcript_expression_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_median_transcript_expression_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_median_transcript_expression_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_median_transcript_expression(gencodeIds = "ENSG00000132693.12")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_multi_tissue_eqtls'>Get Multi Tissue Eqtls</h2><span id='topic+get_multi_tissue_eqtls'></span>

<h3>Description</h3>

<p>Find multi-tissue eQTL <code>Metasoft</code> results.
</p>

<ul>
<li><p> This service returns multi-tissue eQTL Metasoft results for a given gene and variant in a specified dataset.
</p>
</li>
<li><p> A Versioned GENCODE ID must be provided.
</p>
</li>
<li><p> For each tissue, the results include: m-value (mValue), normalized effect size (nes), p-value (pValue), and standard error (se).
</p>
</li>
<li><p> The m-value is the posterior probability that an eQTL effect exists in each tissue tested in the cross-tissue meta-analysis (Han and Eskin, PLoS Genetics 8(3): e1002555, 2012).
</p>
</li>
<li><p> The normalized effect size is the slope of the linear regression of normalized expression data versus the three genotype categories using single-tissue eQTL analysis, representing eQTL effect size.
</p>
</li>
<li><p> The p-value is from a t-test that compares observed NES from single-tissue eQTL analysis to a null NES of 0.
</p>
</li></ul>

<p>By default, the service queries the latest GTEx release. The retrieved data is split into pages with <code>items_per_page</code> entries per page
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_multi_tissue_eqtls_api_v2_association_metasoft_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_multi_tissue_eqtls(
  gencodeIds,
  variantId = NULL,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_multi_tissue_eqtls_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_multi_tissue_eqtls_+3A_variantid">variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td></tr>
<tr><td><code id="get_multi_tissue_eqtls_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_multi_tissue_eqtls_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_multi_tissue_eqtls_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code><a href="#topic+get_eqtl_genes">get_eqtl_genes</a>()</code>,
<code><a href="#topic+get_fine_mapping">get_fine_mapping</a>()</code>,
<code><a href="#topic+get_independent_eqtl">get_independent_eqtl</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls_by_location">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_ieqtls">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_isqtls">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_sqtls">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="#topic+get_sqtl_genes">get_sqtl_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search by gene
get_multi_tissue_eqtls(gencodeId = c("ENSG00000132693.12",
                                     "ENSG00000203782.5"))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_neighbor_gene'>Get Neighbor Gene</h2><span id='topic+get_neighbor_gene'></span>

<h3>Description</h3>

<p>Find all neighboring genes on a certain chromosome around a
position with a certain window size.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Reference-Genome-Endpoints/operation/get_neighbor_gene_api_v2_reference_neighborGene_get">GTEx API Portal documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_neighbor_gene(pos, chromosome, bp_window, page = 0, itemsPerPage = 250)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_neighbor_gene_+3A_pos">pos</code></td>
<td>
<p>Integer, vector.</p>
</td></tr>
<tr><td><code id="get_neighbor_gene_+3A_chromosome">chromosome</code></td>
<td>
<p>String. One of &quot;chr1&quot;, &quot;chr2&quot;, &quot;chr3&quot;, &quot;chr4&quot;, &quot;chr5&quot;,
&quot;chr6&quot;, &quot;chr7&quot;, &quot;chr8&quot;, &quot;chr9&quot;, &quot;chr10&quot;, &quot;chr11&quot;, &quot;chr12&quot;, &quot;chr13&quot;,
&quot;chr14&quot;, &quot;chr15&quot;, &quot;chr16&quot;, &quot;chr17&quot;, &quot;chr18&quot;, &quot;chr19&quot;, &quot;chr20&quot;, &quot;chr21&quot;,
&quot;chr22&quot;, &quot;chrM&quot;, &quot;chrX&quot;, &quot;chrY&quot;.</p>
</td></tr>
<tr><td><code id="get_neighbor_gene_+3A_bp_window">bp_window</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="get_neighbor_gene_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_neighbor_gene_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Reference Genome Endpoints: 
<code><a href="#topic+get_exons">get_exons</a>()</code>,
<code><a href="#topic+get_gene_search">get_gene_search</a>()</code>,
<code><a href="#topic+get_genes">get_genes</a>()</code>,
<code><a href="#topic+get_genomic_features">get_genomic_features</a>()</code>,
<code><a href="#topic+get_gwas_catalog_by_location">get_gwas_catalog_by_location</a>()</code>,
<code><a href="#topic+get_transcripts">get_transcripts</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_neighbor_gene(pos = 1000000, chromosome = "chr1", bp_window = 10000)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_news_item'>Get News Item</h2><span id='topic+get_news_item'></span>

<h3>Description</h3>

<p>Getting all the news items from the database that are current.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Admin-Endpoints/operation/get_news_item_api_v2_admin_newsItem_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_news_item(page = 0, itemsPerPage = 250)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_news_item_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_news_item_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Admin Endpoints: 
<code><a href="#topic+get_maintenance_message">get_maintenance_message</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_news_item()

## End(Not run)
</code></pre>

<hr>
<h2 id='get_sample_biobank_data'>Get Sample (Biobank Data)</h2><span id='topic+get_sample_biobank_data'></span>

<h3>Description</h3>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Biobank-Data-Endpoints/operation/get_sample_api_v2_biobank_sample_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sample_biobank_data(
  draw = NULL,
  materialTypes = NULL,
  tissueSiteDetailIds = NULL,
  pathCategory = NULL,
  tissueSampleIds = NULL,
  sex = NULL,
  sortBy = "sampleId",
  sortDirection = "asc",
  searchTerm = NULL,
  sampleIds = NULL,
  subjectIds = NULL,
  ageBrackets = NULL,
  hardyScales = NULL,
  hasExpressionData = NULL,
  hasGenotype = NULL,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_sample_biobank_data_+3A_draw">draw</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_materialtypes">materialTypes</code></td>
<td>
<p>String, vector. Options: &quot;Cells:Cell Line Viable&quot;,
&quot;DNA:DNA Genomic&quot;, &quot;DNA:DNA Somatic&quot;, &quot;RNA:Total RNA&quot;, &quot;Tissue:PAXgene
Preserved&quot;, &quot;Tissue:PAXgene Preserved Paraffin-embedded&quot;, &quot;Tissue:Fresh
Frozen Tissue&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_pathcategory">pathCategory</code></td>
<td>
<p>Character vector. Options: &quot;adenoma&quot;, &quot;amylacea&quot;,
&quot;atelectasis&quot;, &quot;atherosclerosis&quot;, &quot;atherosis&quot;, &quot;atrophy&quot;, &quot;calcification&quot;,
&quot;cirrhosis&quot;, &quot;clean_specimens&quot;, &quot;congestion&quot;, &quot;corpora_albicantia&quot;, &quot;cyst&quot;,
&quot;desquamation&quot;, &quot;diabetic&quot;, &quot;dysplasia&quot;, &quot;edema&quot;, &quot;emphysema&quot;,
&quot;esophagitis&quot;, &quot;fibrosis&quot;, &quot;gastritis&quot;, &quot;glomerulosclerosis&quot;, &quot;goiter&quot;,
&quot;gynecomastoid&quot;, &quot;hashimoto&quot;, &quot;heart_failure_cells&quot;, &quot;hemorrhage&quot;,
&quot;hepatitis&quot;, &quot;hyalinization&quot;, &quot;hypereosinophilia&quot;, &quot;hyperplasia&quot;,
&quot;hypertrophy&quot;, &quot;hypoxic&quot;, &quot;infarction&quot;, &quot;inflammation&quot;, &quot;ischemic_changes&quot;,
&quot;macrophages&quot;, &quot;mastopathy&quot;, &quot;metaplasia&quot;, &quot;monckeberg&quot;, &quot;necrosis&quot;,
&quot;nephritis&quot;, &quot;nephrosclerosis&quot;, &quot;no_abnormalities&quot;, &quot;nodularity&quot;,
&quot;pancreatitis&quot;, &quot;pigment&quot;, &quot;pneumonia&quot;, &quot;post_menopausal&quot;, &quot;prostatitis&quot;,
&quot;saponification&quot;, &quot;scarring&quot;, &quot;sclerotic&quot;, &quot;solar_elastosis&quot;,
&quot;spermatogenesis&quot;, &quot;steatosis&quot;, &quot;sweat_glands&quot;, &quot;tma&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_tissuesampleids">tissueSampleIds</code></td>
<td>
<p>Array of strings. A list of Tissue Sample ID(s).</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_sex">sex</code></td>
<td>
<p>String. Options: &quot;male&quot;, &quot;female&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_sortby">sortBy</code></td>
<td>
<p>String. Options: &quot;sampleId&quot;, &quot;ischemicTime&quot;, &quot;aliquotId&quot;,
&quot;tissueSampleId&quot;, &quot;hardyScale&quot;, &quot;pathologyNotes&quot;, &quot;ageBracket&quot;,
&quot;tissueSiteDetailId&quot;, &quot;sex&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_sortdirection">sortDirection</code></td>
<td>
<p>String. Options: &quot;asc&quot;, &quot;desc&quot;. Default = &quot;asc&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_searchterm">searchTerm</code></td>
<td>
<p>String.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_sampleids">sampleIds</code></td>
<td>
<p>Character vector. GTEx sample ID.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_subjectids">subjectIds</code></td>
<td>
<p>Character vector. GTEx subject ID.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_agebrackets">ageBrackets</code></td>
<td>
<p>The age bracket(s) of the donors of interest. Options:
&quot;20-29&quot;, &quot;30-39&quot;, &quot;40-49&quot;, &quot;50-59&quot;, &quot;60-69&quot;, &quot;70-79&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_hardyscales">hardyScales</code></td>
<td>
<p>Character vector. A list of Hardy Scale(s) of interest.
Options: &quot;Ventilator case&quot;, &quot;Fast death - violent&quot;, &quot;Fast death - natural
causes&quot;, &quot;Intermediate death&quot;, &quot;Slow death&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_hasexpressiondata">hasExpressionData</code></td>
<td>
<p>Logical.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_hasgenotype">hasGenotype</code></td>
<td>
<p>Logical.</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_sample_biobank_data_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Biobank Data Endpoints: 
<code><a href="#topic+download">download</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 get_sample_biobank_data(tissueSiteDetailIds = "Whole_Blood")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_sample_datasets_endpoints'>Get Sample (Datasets Endpoints)</h2><span id='topic+get_sample_datasets_endpoints'></span>

<h3>Description</h3>

<p>This service returns information of samples used in analyses
from all datasets. Results may be filtered by dataset ID, sample ID, subject
ID, sample metadata, or other provided parameters. By default, this service
queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_sample_api_v2_dataset_sample_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sample_datasets_endpoints(
  datasetId = "gtex_v8",
  sampleIds = NULL,
  tissueSampleIds = NULL,
  subjectIds = NULL,
  ageBrackets = NULL,
  sex = NULL,
  pathCategory = NULL,
  tissueSiteDetailId = NULL,
  aliquotIds = NULL,
  autolysisScores = NULL,
  hardyScales = NULL,
  ischemicTime = NULL,
  ischemicTimeGroups = NULL,
  rin = NULL,
  uberonIds = NULL,
  dataTypes = NULL,
  sortBy = NULL,
  sortDirection = NULL,
  page = NULL,
  itemsPerPage = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_sample_datasets_endpoints_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_sampleids">sampleIds</code></td>
<td>
<p>Character vector. GTEx sample ID.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_tissuesampleids">tissueSampleIds</code></td>
<td>
<p>Array of strings. A list of Tissue Sample ID(s).</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_subjectids">subjectIds</code></td>
<td>
<p>Character vector. GTEx subject ID.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_agebrackets">ageBrackets</code></td>
<td>
<p>The age bracket(s) of the donors of interest. Options:
&quot;20-29&quot;, &quot;30-39&quot;, &quot;40-49&quot;, &quot;50-59&quot;, &quot;60-69&quot;, &quot;70-79&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_sex">sex</code></td>
<td>
<p>String. Options: &quot;male&quot;, &quot;female&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_pathcategory">pathCategory</code></td>
<td>
<p>Character vector. Options: &quot;adenoma&quot;, &quot;amylacea&quot;,
&quot;atelectasis&quot;, &quot;atherosclerosis&quot;, &quot;atherosis&quot;, &quot;atrophy&quot;, &quot;calcification&quot;,
&quot;cirrhosis&quot;, &quot;clean_specimens&quot;, &quot;congestion&quot;, &quot;corpora_albicantia&quot;, &quot;cyst&quot;,
&quot;desquamation&quot;, &quot;diabetic&quot;, &quot;dysplasia&quot;, &quot;edema&quot;, &quot;emphysema&quot;,
&quot;esophagitis&quot;, &quot;fibrosis&quot;, &quot;gastritis&quot;, &quot;glomerulosclerosis&quot;, &quot;goiter&quot;,
&quot;gynecomastoid&quot;, &quot;hashimoto&quot;, &quot;heart_failure_cells&quot;, &quot;hemorrhage&quot;,
&quot;hepatitis&quot;, &quot;hyalinization&quot;, &quot;hypereosinophilia&quot;, &quot;hyperplasia&quot;,
&quot;hypertrophy&quot;, &quot;hypoxic&quot;, &quot;infarction&quot;, &quot;inflammation&quot;, &quot;ischemic_changes&quot;,
&quot;macrophages&quot;, &quot;mastopathy&quot;, &quot;metaplasia&quot;, &quot;monckeberg&quot;, &quot;necrosis&quot;,
&quot;nephritis&quot;, &quot;nephrosclerosis&quot;, &quot;no_abnormalities&quot;, &quot;nodularity&quot;,
&quot;pancreatitis&quot;, &quot;pigment&quot;, &quot;pneumonia&quot;, &quot;post_menopausal&quot;, &quot;prostatitis&quot;,
&quot;saponification&quot;, &quot;scarring&quot;, &quot;sclerotic&quot;, &quot;solar_elastosis&quot;,
&quot;spermatogenesis&quot;, &quot;steatosis&quot;, &quot;sweat_glands&quot;, &quot;tma&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_tissuesitedetailid">tissueSiteDetailId</code></td>
<td>
<p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. &quot;Whole_Blood&quot;; use <code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code>
to see valid values) or an Ontology ID.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_aliquotids">aliquotIds</code></td>
<td>
<p>Character vector.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_autolysisscores">autolysisScores</code></td>
<td>
<p>Character vector. Options: &quot;None&quot;, &quot;Mild&quot;, &quot;Moderate&quot;,
&quot;Severe&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_hardyscales">hardyScales</code></td>
<td>
<p>Character vector. A list of Hardy Scale(s) of interest.
Options: &quot;Ventilator case&quot;, &quot;Fast death - violent&quot;, &quot;Fast death - natural
causes&quot;, &quot;Intermediate death&quot;, &quot;Slow death&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_ischemictime">ischemicTime</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_ischemictimegroups">ischemicTimeGroups</code></td>
<td>
<p>Character vector. Options:
&quot;&lt;= 0&quot;, &quot;1 - 300&quot;, &quot;301 - 600&quot;, &quot;601 - 900&quot;, &quot;901 - 1200&quot;, &quot;1201 - 1500&quot;, &quot;&gt; 1500&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_rin">rin</code></td>
<td>
<p>Integer vector.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_uberonids">uberonIds</code></td>
<td>
<p>Character vector of Uberon IDs (e.g. &quot;UBERON:EFO_0000572&quot;;
use <code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values).</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_datatypes">dataTypes</code></td>
<td>
<p>Character vector. Options: &quot;RNASEQ&quot;, &quot;WGS&quot;, &quot;WES&quot;, &quot;OMNI&quot;,
&quot;EXCLUDE&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_sortby">sortBy</code></td>
<td>
<p>String. Options: &quot;sampleId&quot;, &quot;ischemicTime&quot;, &quot;aliquotId&quot;,
&quot;tissueSampleId&quot;, &quot;hardyScale&quot;, &quot;pathologyNotes&quot;, &quot;ageBracket&quot;,
&quot;tissueSiteDetailId&quot;, &quot;sex&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_sortdirection">sortDirection</code></td>
<td>
<p>String. Options: &quot;asc&quot;, &quot;desc&quot;. Default = &quot;asc&quot;.</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_sample_datasets_endpoints_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 get_sample_datasets_endpoints()

## End(Not run)
</code></pre>

<hr>
<h2 id='get_service_info'>Get Service Info</h2><span id='topic+get_service_info'></span>

<h3>Description</h3>

<p>General information about the GTEx service.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/GTEx-Portal-API-Info/operation/get_service_info_api_v2__get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_service_info()
</code></pre>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_service_info()

## End(Not run)
</code></pre>

<hr>
<h2 id='get_significant_single_tissue_eqtls'>Get Significant Single Tissue Eqtls</h2><span id='topic+get_significant_single_tissue_eqtls'></span>

<h3>Description</h3>

<p>Find significant single tissue
eQTLs.
</p>

<ul>
<li><p> This service returns precomputed significant single tissue eQTLs.
</p>
</li>
<li><p> Results may be filtered by tissue, gene, variant or dataset.
</p>
</li>
<li><p> To search by gene, use the versioned GENCODE ID.
</p>
</li>
<li><p> To search by variant, use the dbSNP rs ID (snpId).
</p>
</li></ul>

<p>By default, the service queries the latest GTEx release and the retrieved data is split into pages with <code>items_per_page</code> entries per page
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_significant_single_tissue_eqtls_api_v2_association_singleTissueEqtl_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_significant_single_tissue_eqtls(
  gencodeIds = NULL,
  variantIds = NULL,
  tissueSiteDetailIds = NULL,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_significant_single_tissue_eqtls_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_eqtls_+3A_variantids">variantIds</code></td>
<td>
<p>Character vector. Gtex variant IDs.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_eqtls_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_eqtls_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_eqtls_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_eqtls_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Note:</strong> although the GTEx Portal API documentation says to use the dbSNP rsID
when searching by variant, this returns no results. Instead use gtex variant
IDs e.g. use &quot;chr1_153209640_C_A_b38&quot; instead of &quot;rs1410858&quot;.
</p>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code><a href="#topic+get_eqtl_genes">get_eqtl_genes</a>()</code>,
<code><a href="#topic+get_fine_mapping">get_fine_mapping</a>()</code>,
<code><a href="#topic+get_independent_eqtl">get_independent_eqtl</a>()</code>,
<code><a href="#topic+get_multi_tissue_eqtls">get_multi_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls_by_location">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_ieqtls">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_isqtls">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_sqtls">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="#topic+get_sqtl_genes">get_sqtl_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search by gene
get_significant_single_tissue_eqtls(gencodeIds = c("ENSG00000132693.12",
                                                  "ENSG00000203782.5"))

# search by variant - must be variantId (not rsid)
get_significant_single_tissue_eqtls(variantIds = "chr1_153209640_C_A_b38")

# filter by gene/variant and tissue site - either `gencodeIds` or `variantIds`
# should be supplied as a minimum
get_significant_single_tissue_eqtls(gencodeIds = c("ENSG00000132693.12",
                                                  "ENSG00000203782.5"),
                                    variantIds = "chr1_153209640_C_A_b38",
                                    tissueSiteDetailIds = "Whole_Blood")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_significant_single_tissue_eqtls_by_location'>Get Significant Single Tissue eQTLs By Location</h2><span id='topic+get_significant_single_tissue_eqtls_by_location'></span>

<h3>Description</h3>

<p>Find significant single tissue eQTLs using Chromosomal Locations.
</p>

<ul>
<li><p> This service returns precomputed significant single tissue eQTLs.
</p>
</li>
<li><p> Results may be filtered by tissue, and/or dataset.
</p>
</li></ul>

<p>By default, the service queries the latest GTEx release. Since this endpoint
is used to support a third party program on the portal, the return structure
is different from other endpoints and is not paginated.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_significant_single_tissue_eqtls_api_v2_association_singleTissueEqtl_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_significant_single_tissue_eqtls_by_location(
  tissueSiteDetailId,
  start,
  end,
  chromosome,
  datasetId = "gtex_v8"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_significant_single_tissue_eqtls_by_location_+3A_tissuesitedetailid">tissueSiteDetailId</code></td>
<td>
<p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. &quot;Whole_Blood&quot;; use <code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code>
to see valid values) or an Ontology ID.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_eqtls_by_location_+3A_start">start</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_eqtls_by_location_+3A_end">end</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_eqtls_by_location_+3A_chromosome">chromosome</code></td>
<td>
<p>String. One of &quot;chr1&quot;, &quot;chr2&quot;, &quot;chr3&quot;, &quot;chr4&quot;, &quot;chr5&quot;,
&quot;chr6&quot;, &quot;chr7&quot;, &quot;chr8&quot;, &quot;chr9&quot;, &quot;chr10&quot;, &quot;chr11&quot;, &quot;chr12&quot;, &quot;chr13&quot;,
&quot;chr14&quot;, &quot;chr15&quot;, &quot;chr16&quot;, &quot;chr17&quot;, &quot;chr18&quot;, &quot;chr19&quot;, &quot;chr20&quot;, &quot;chr21&quot;,
&quot;chr22&quot;, &quot;chrM&quot;, &quot;chrX&quot;, &quot;chrY&quot;.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_eqtls_by_location_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code><a href="#topic+get_eqtl_genes">get_eqtl_genes</a>()</code>,
<code><a href="#topic+get_fine_mapping">get_fine_mapping</a>()</code>,
<code><a href="#topic+get_independent_eqtl">get_independent_eqtl</a>()</code>,
<code><a href="#topic+get_multi_tissue_eqtls">get_multi_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_ieqtls">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_isqtls">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_sqtls">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="#topic+get_sqtl_genes">get_sqtl_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_significant_single_tissue_eqtls_by_location(
  tissueSiteDetailId = "Artery_Aorta",
  start = 10000,
  end = 250000,
  chromosome = "chr11")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_significant_single_tissue_ieqtls'>Get Significant Single Tissue Ieqtls</h2><span id='topic+get_significant_single_tissue_ieqtls'></span>

<h3>Description</h3>

<p>Retrieve Interaction eQTL Data.
</p>

<ul>
<li><p> This service returns cell type interaction eQTLs (ieQTLs), from a specified dataset.
</p>
</li>
<li><p> Results may be filtered by tissue
</p>
</li>
<li><p> By default, the service queries the latest GTEx release.
</p>
</li></ul>

<p>The retrieved data is split into pages with <code>items_per_page</code> entries per page
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_significant_single_tissue_ieqtls_api_v2_association_singleTissueIEqtl_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_significant_single_tissue_ieqtls(
  gencodeIds,
  variantIds = NULL,
  tissueSiteDetailIds = NULL,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_significant_single_tissue_ieqtls_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_ieqtls_+3A_variantids">variantIds</code></td>
<td>
<p>Character vector. Gtex variant IDs.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_ieqtls_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_ieqtls_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_ieqtls_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_ieqtls_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code><a href="#topic+get_eqtl_genes">get_eqtl_genes</a>()</code>,
<code><a href="#topic+get_fine_mapping">get_fine_mapping</a>()</code>,
<code><a href="#topic+get_independent_eqtl">get_independent_eqtl</a>()</code>,
<code><a href="#topic+get_multi_tissue_eqtls">get_multi_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls_by_location">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_isqtls">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_sqtls">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="#topic+get_sqtl_genes">get_sqtl_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_significant_single_tissue_ieqtls(c("ENSG00000132693.12",
                                       "ENSG00000203782.5"))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_significant_single_tissue_isqtls'>et Significant Single Tissue Isqtls</h2><span id='topic+get_significant_single_tissue_isqtls'></span>

<h3>Description</h3>

<p>Retrieve Interaction sQTL Data.
</p>

<ul>
<li><p> This service retrieves cell type interaction
sQTLs (isQTLs), from a specified dataset.
</p>
</li>
<li><p> Results may be filtered by tissue
</p>
</li>
<li><p> By default, the service queries the latest GTEx release.
</p>
</li></ul>

<p>The retrieved data is split into pages with <code>items_per_page</code> entries per page
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_significant_single_tissue_isqtls_api_v2_association_singleTissueISqtl_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_significant_single_tissue_isqtls(
  gencodeIds,
  variantIds = NULL,
  tissueSiteDetailIds = NULL,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_significant_single_tissue_isqtls_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_isqtls_+3A_variantids">variantIds</code></td>
<td>
<p>Character vector. Gtex variant IDs.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_isqtls_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_isqtls_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_isqtls_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_isqtls_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code><a href="#topic+get_eqtl_genes">get_eqtl_genes</a>()</code>,
<code><a href="#topic+get_fine_mapping">get_fine_mapping</a>()</code>,
<code><a href="#topic+get_independent_eqtl">get_independent_eqtl</a>()</code>,
<code><a href="#topic+get_multi_tissue_eqtls">get_multi_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls_by_location">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_ieqtls">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_sqtls">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="#topic+get_sqtl_genes">get_sqtl_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_significant_single_tissue_isqtls(gencodeIds = c("ENSG00000065613.9",
                                                    "ENSG00000203782.5"))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_significant_single_tissue_sqtls'>Get Significant Single Tissue Sqtls</h2><span id='topic+get_significant_single_tissue_sqtls'></span>

<h3>Description</h3>

<p>Retrieve Single Tissue sQTL Data.
</p>

<ul>
<li><p> This service returns single tissue sQTL
data for the given genes, from a specified dataset.
</p>
</li>
<li><p> Results may be filtered
by tissue
</p>
</li>
<li><p> By default, the service queries the latest GTEx release.
</p>
</li></ul>

<p>The retrieved data is split into pages with <code>items_per_page</code> entries per page
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_significant_single_tissue_sqtls_api_v2_association_singleTissueSqtl_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_significant_single_tissue_sqtls(
  gencodeIds,
  variantIds = NULL,
  tissueSiteDetailIds = NULL,
  datasetId = "gtex_v8"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_significant_single_tissue_sqtls_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_sqtls_+3A_variantids">variantIds</code></td>
<td>
<p>Character vector. Gtex variant IDs.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_sqtls_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_significant_single_tissue_sqtls_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code><a href="#topic+get_eqtl_genes">get_eqtl_genes</a>()</code>,
<code><a href="#topic+get_fine_mapping">get_fine_mapping</a>()</code>,
<code><a href="#topic+get_independent_eqtl">get_independent_eqtl</a>()</code>,
<code><a href="#topic+get_multi_tissue_eqtls">get_multi_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls_by_location">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_ieqtls">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_isqtls">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="#topic+get_sqtl_genes">get_sqtl_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search by gene
get_significant_single_tissue_sqtls(gencodeIds = c("ENSG00000065613.9",
                                                   "ENSG00000203782.5"))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_single_nucleus_gex'>Get Single Nucleux Gex</h2><span id='topic+get_single_nucleus_gex'></span>

<h3>Description</h3>

<p>Retrieve Single Nucleus Gene Expression Data for a given Gene.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_single_nucleus_gex_api_v2_expression_singleNucleusGeneExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_single_nucleus_gex(
  gencodeIds,
  datasetId = "gtex_snrnaseq_pilot",
  tissueSiteDetailIds = NULL,
  excludeDataArray = TRUE,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_single_nucleus_gex_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="get_single_nucleus_gex_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_single_nucleus_gex_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_single_nucleus_gex_+3A_excludedataarray">excludeDataArray</code></td>
<td>
<p>String. Options are TRUE or FALSE</p>
</td></tr>
<tr><td><code id="get_single_nucleus_gex_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_single_nucleus_gex_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Search for one or more genes - returns a tibble with one row per tissue.
# Column "cellTypes" now contains a tibble of expression summary data, with
# one row for each cell type
get_single_nucleus_gex(gencodeIds = c("ENSG00000203782.5",
                                      "ENSG00000132693.12"))

# `excludeDataArray = FALSE` - expression values are stored under "celltypes"
# in an additional column called "data"
response &lt;- get_single_nucleus_gex(gencodeIds = "ENSG00000132693.12",
                                   excludeDataArray = FALSE,
                                   itemsPerPage = 2)

response

# "cellTypes" contains a tibble of data with one row for each
# cell type e.g. for Breast_Mammary_Tissue
response$cellTypes[[2]]

# when `excludeDataArray = FALSE`, expression values are stored in "data"
# e.g. for Breast_Mammary_Tissue, Epithelial cell (luminal):
response$cellTypes[[2]]$data[[1]]

## End(Not run)
</code></pre>

<hr>
<h2 id='get_single_nucleus_gex_summary'>Get Single Nucleus Gex Summary</h2><span id='topic+get_single_nucleus_gex_summary'></span>

<h3>Description</h3>

<p>Retrieve Summarized Single Nucleus Gene Expression Data.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_single_nucleus_gex_summary_api_v2_expression_singleNucleusGeneExpressionSummary_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_single_nucleus_gex_summary(
  datasetId = "gtex_snrnaseq_pilot",
  tissueSiteDetailIds = NULL,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_single_nucleus_gex_summary_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_single_nucleus_gex_summary_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="get_single_nucleus_gex_summary_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_single_nucleus_gex_summary_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_top_expressed_genes">get_top_expressed_genes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# all tissues
get_single_nucleus_gex_summary()

# filter for specific tissue
get_single_nucleus_gex_summary(tissueSiteDetailIds = c("Breast_Mammary_Tissue",
                                                       "Skin_Sun_Exposed_Lower_leg"))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_sqtl_genes'>Get Sqtl Genes</h2><span id='topic+get_sqtl_genes'></span>

<h3>Description</h3>

<p>Retrieve sGenes (sQTL Genes).
</p>

<ul>
<li><p> This service returns sGenes (sQTL Genes) from the specified dataset.
</p>
</li>
<li><p> Results may be filtered by tissue.
</p>
</li>
<li><p> By default, the service queries the latest GTEx release.
</p>
</li></ul>

<p>The retrieved data is split into pages with <code>items_per_page</code> entries per page
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_sqtl_genes_api_v2_association_sgene_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sqtl_genes(
  tissueSiteDetailId,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_sqtl_genes_+3A_tissuesitedetailid">tissueSiteDetailId</code></td>
<td>
<p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. &quot;Whole_Blood&quot;; use <code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code>
to see valid values) or an Ontology ID.</p>
</td></tr>
<tr><td><code id="get_sqtl_genes_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_sqtl_genes_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_sqtl_genes_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code><a href="#topic+get_eqtl_genes">get_eqtl_genes</a>()</code>,
<code><a href="#topic+get_fine_mapping">get_fine_mapping</a>()</code>,
<code><a href="#topic+get_independent_eqtl">get_independent_eqtl</a>()</code>,
<code><a href="#topic+get_multi_tissue_eqtls">get_multi_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_eqtls_by_location">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_ieqtls">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_isqtls">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="#topic+get_significant_single_tissue_sqtls">get_significant_single_tissue_sqtls</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_sqtl_genes("Whole_Blood")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_subject'>Get Subject</h2><span id='topic+get_subject'></span>

<h3>Description</h3>

<p>This service returns information of subjects used in analyses
from all datasets. Results may be filtered by dataset ID, subject ID, sex,
age bracket or Hardy Scale. By default, this service queries the latest GTEx
release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_tissue_site_detail_api_v2_dataset_tissueSiteDetail_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_subject(
  datasetId = "gtex_v8",
  sex = NULL,
  ageBrackets = NULL,
  hardyScales = NULL,
  subjectIds = NULL,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_subject_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_subject_+3A_sex">sex</code></td>
<td>
<p>String. Options: &quot;male&quot;, &quot;female&quot;.</p>
</td></tr>
<tr><td><code id="get_subject_+3A_agebrackets">ageBrackets</code></td>
<td>
<p>The age bracket(s) of the donors of interest. Options:
&quot;20-29&quot;, &quot;30-39&quot;, &quot;40-49&quot;, &quot;50-59&quot;, &quot;60-69&quot;, &quot;70-79&quot;.</p>
</td></tr>
<tr><td><code id="get_subject_+3A_hardyscales">hardyScales</code></td>
<td>
<p>Character vector. A list of Hardy Scale(s) of interest.
Options: &quot;Ventilator case&quot;, &quot;Fast death - violent&quot;, &quot;Fast death - natural
causes&quot;, &quot;Intermediate death&quot;, &quot;Slow death&quot;.</p>
</td></tr>
<tr><td><code id="get_subject_+3A_subjectids">subjectIds</code></td>
<td>
<p>Character vector. GTEx subject ID.</p>
</td></tr>
<tr><td><code id="get_subject_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_subject_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Tibble
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  get_subject()

## End(Not run)
</code></pre>

<hr>
<h2 id='get_tissue_site_detail'>Get Tissue Site Detail</h2><span id='topic+get_tissue_site_detail'></span>

<h3>Description</h3>

<p>Retrieve all tissue site detail information in the database
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_tissue_site_detail_api_v2_dataset_tissueSiteDetail_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tissue_site_detail(page = 0, itemsPerPage = 250)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_tissue_site_detail_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_tissue_site_detail_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  # returns a tibble with one row per tissue
  get_tissue_site_detail()

  # `eqtlSampleSummary` and `rnaSeqSampleSummary` are list columns
  bladder_site_details &lt;- get_tissue_site_detail() |&gt;
    dplyr::filter(tissueSiteDetailId == "Bladder")

  purrr::pluck(bladder_site_details, "eqtlSampleSummary", 1)

  purrr::pluck(bladder_site_details, "rnaSeqSampleSummary", 1)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_top_expressed_genes'>Get Top Expressed Genes</h2><span id='topic+get_top_expressed_genes'></span>

<h3>Description</h3>

<p>Find top expressed genes for a specified tissue.
</p>

<ul>
<li><p> Returns top expressed genes for a specified tissue in a dataset, sorted by median expression.
</p>
</li>
<li><p> When the optional parameter filterMtGene is set to true, mitochondrial genes will be excluded from the results.
By default, this service queries the latest GTEx release.
</p>
</li></ul>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_top_expressed_genes_api_v2_expression_topExpressedGene_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_top_expressed_genes(
  tissueSiteDetailId,
  datasetId = "gtex_v8",
  filterMtGene = TRUE,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_top_expressed_genes_+3A_tissuesitedetailid">tissueSiteDetailId</code></td>
<td>
<p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. &quot;Whole_Blood&quot;; use <code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code>
to see valid values) or an Ontology ID.</p>
</td></tr>
<tr><td><code id="get_top_expressed_genes_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_top_expressed_genes_+3A_filtermtgene">filterMtGene</code></td>
<td>
<p>Logical. Exclude mitochondrial genes.</p>
</td></tr>
<tr><td><code id="get_top_expressed_genes_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_top_expressed_genes_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code><a href="#topic+get_clustered_median_exon_expression">get_clustered_median_exon_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_gene_expression">get_clustered_median_gene_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_junction_expression">get_clustered_median_junction_expression</a>()</code>,
<code><a href="#topic+get_clustered_median_transcript_expression">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_expression_pca">get_expression_pca</a>()</code>,
<code><a href="#topic+get_gene_expression">get_gene_expression</a>()</code>,
<code><a href="#topic+get_median_exon_expression">get_median_exon_expression</a>()</code>,
<code><a href="#topic+get_median_gene_expression">get_median_gene_expression</a>()</code>,
<code><a href="#topic+get_median_junction_expression">get_median_junction_expression</a>()</code>,
<code><a href="#topic+get_median_transcript_expression">get_median_transcript_expression</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex">get_single_nucleus_gex</a>()</code>,
<code><a href="#topic+get_single_nucleus_gex_summary">get_single_nucleus_gex_summary</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_top_expressed_genes(tissueSiteDetailId = "Artery_Aorta")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_transcripts'>Get Transcripts</h2><span id='topic+get_transcripts'></span>

<h3>Description</h3>

<p>Find all transcripts of a reference gene.
</p>

<ul>
<li><p> This service returns information about transcripts of the given versioned GENCODE ID.
</p>
</li>
<li><p> A genome build and GENCODE version must be provided.
</p>
</li>
<li><p> By default, this service queries the genome build and GENCODE version used by the latest GTEx release.
</p>
</li></ul>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Reference-Genome-Endpoints/operation/get_transcripts_api_v2_reference_transcript_get">GTEx API Portal documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_transcripts(
  gencodeId,
  gencodeVersion = "v26",
  genomeBuild = "GRCh38/hg38",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_transcripts_+3A_gencodeid">gencodeId</code></td>
<td>
<p>String. A Versioned GENCODE ID of a gene, e.g.
&quot;ENSG00000065613.9&quot;.</p>
</td></tr>
<tr><td><code id="get_transcripts_+3A_gencodeversion">gencodeVersion</code></td>
<td>
<p>String (default = &quot;v26&quot;). GENCODE annotation release.
Either &quot;v26&quot; or &quot;v19&quot;.</p>
</td></tr>
<tr><td><code id="get_transcripts_+3A_genomebuild">genomeBuild</code></td>
<td>
<p>String. Options: &quot;GRCh38/hg38&quot;, &quot;GRCh37/hg19&quot;. Default =
&quot;GRCh38/hg38&quot;.</p>
</td></tr>
<tr><td><code id="get_transcripts_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_transcripts_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Reference Genome Endpoints: 
<code><a href="#topic+get_exons">get_exons</a>()</code>,
<code><a href="#topic+get_gene_search">get_gene_search</a>()</code>,
<code><a href="#topic+get_genes">get_genes</a>()</code>,
<code><a href="#topic+get_genomic_features">get_genomic_features</a>()</code>,
<code><a href="#topic+get_gwas_catalog_by_location">get_gwas_catalog_by_location</a>()</code>,
<code><a href="#topic+get_neighbor_gene">get_neighbor_gene</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_transcripts(gencodeId = "ENSG00000203782.5")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_variant'>Get Variant</h2><span id='topic+get_variant'></span>

<h3>Description</h3>

<p>This service returns information about a variant, including
position, dbSNP RS ID, the reference allele, the alternative allele, and
whether the minor allele frequency is &gt;= 1%. For GTEx v6p, there is also
information about whether the whole exome sequence and chip sequencing data
are available. Results may be queried by GTEx variant ID (variantId), dbSNP
RS ID (snpId) or genomic location (chromosome and pos). Variants are
identified based on the genotype data of each dataset cohort, namely, are
dataset-dependent. Each variant is assigned a unique GTEx variant ID (i.e.
the primary key). Not all variants have a mappable dbSNP RS ID. By default,
this service queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_variant_api_v2_dataset_variant_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variant(
  snpId = NULL,
  variantId = NULL,
  chromosome = NULL,
  pos = NULL,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_variant_+3A_snpid">snpId</code></td>
<td>
<p>String</p>
</td></tr>
<tr><td><code id="get_variant_+3A_variantid">variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td></tr>
<tr><td><code id="get_variant_+3A_chromosome">chromosome</code></td>
<td>
<p>String. One of &quot;chr1&quot;, &quot;chr2&quot;, &quot;chr3&quot;, &quot;chr4&quot;, &quot;chr5&quot;,
&quot;chr6&quot;, &quot;chr7&quot;, &quot;chr8&quot;, &quot;chr9&quot;, &quot;chr10&quot;, &quot;chr11&quot;, &quot;chr12&quot;, &quot;chr13&quot;,
&quot;chr14&quot;, &quot;chr15&quot;, &quot;chr16&quot;, &quot;chr17&quot;, &quot;chr18&quot;, &quot;chr19&quot;, &quot;chr20&quot;, &quot;chr21&quot;,
&quot;chr22&quot;, &quot;chrM&quot;, &quot;chrX&quot;, &quot;chrY&quot;.</p>
</td></tr>
<tr><td><code id="get_variant_+3A_pos">pos</code></td>
<td>
<p>Integer, vector.</p>
</td></tr>
<tr><td><code id="get_variant_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="get_variant_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_variant_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant_by_location">get_variant_by_location</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># search by rsid
get_variant(snpId = "rs1410858")

# search by variantId
get_variant(variantId = "chr1_153209640_C_A_b38")

# search by chromosome and position
get_variant(chromosome = "chr1",
            pos = 153209600:153209700)
</code></pre>

<hr>
<h2 id='get_variant_by_location'>Get Variant By Location</h2><span id='topic+get_variant_by_location'></span>

<h3>Description</h3>

<p>This service allows the user to query information about variants
on a certain chromosome at a certain location.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Datasets-Endpoints/operation/get_variant_by_location_api_v2_dataset_variantByLocation_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variant_by_location(
  start,
  end,
  chromosome,
  sortBy = "pos",
  sortDirection = "asc",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_variant_by_location_+3A_start">start</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="get_variant_by_location_+3A_end">end</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="get_variant_by_location_+3A_chromosome">chromosome</code></td>
<td>
<p>String. One of &quot;chr1&quot;, &quot;chr2&quot;, &quot;chr3&quot;, &quot;chr4&quot;, &quot;chr5&quot;,
&quot;chr6&quot;, &quot;chr7&quot;, &quot;chr8&quot;, &quot;chr9&quot;, &quot;chr10&quot;, &quot;chr11&quot;, &quot;chr12&quot;, &quot;chr13&quot;,
&quot;chr14&quot;, &quot;chr15&quot;, &quot;chr16&quot;, &quot;chr17&quot;, &quot;chr18&quot;, &quot;chr19&quot;, &quot;chr20&quot;, &quot;chr21&quot;,
&quot;chr22&quot;, &quot;chrM&quot;, &quot;chrX&quot;, &quot;chrY&quot;.</p>
</td></tr>
<tr><td><code id="get_variant_by_location_+3A_sortby">sortBy</code></td>
<td>
<p>String. Options: &quot;sampleId&quot;, &quot;ischemicTime&quot;, &quot;aliquotId&quot;,
&quot;tissueSampleId&quot;, &quot;hardyScale&quot;, &quot;pathologyNotes&quot;, &quot;ageBracket&quot;,
&quot;tissueSiteDetailId&quot;, &quot;sex&quot;.</p>
</td></tr>
<tr><td><code id="get_variant_by_location_+3A_sortdirection">sortDirection</code></td>
<td>
<p>String. Options: &quot;asc&quot;, &quot;desc&quot;. Default = &quot;asc&quot;.</p>
</td></tr>
<tr><td><code id="get_variant_by_location_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="get_variant_by_location_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Datasets Endpoints: 
<code><a href="#topic+get_annotation">get_annotation</a>()</code>,
<code><a href="#topic+get_collapsed_gene_model_exon">get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_downloads_page_data">get_downloads_page_data</a>()</code>,
<code><a href="#topic+get_file_list">get_file_list</a>()</code>,
<code><a href="#topic+get_full_get_collapsed_gene_model_exon">get_full_get_collapsed_gene_model_exon</a>()</code>,
<code><a href="#topic+get_functional_annotation">get_functional_annotation</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_by_variant_data">get_linkage_disequilibrium_by_variant_data</a>()</code>,
<code><a href="#topic+get_linkage_disequilibrium_data">get_linkage_disequilibrium_data</a>()</code>,
<code><a href="#topic+get_sample_datasets_endpoints">get_sample_datasets_endpoints</a>()</code>,
<code><a href="#topic+get_subject">get_subject</a>()</code>,
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail</a>()</code>,
<code><a href="#topic+get_variant">get_variant</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_variant_by_location(start = 153209600,
                        end = 153209700,
                        chromosome = "chr1")
</code></pre>

<hr>
<h2 id='gtexr_arguments'>gtexr arguments</h2><span id='topic+gtexr_arguments'></span>

<h3>Description</h3>

<p>Internal function that documents all arguments for exported gtexr functions
that wrap GTEx Portal API endpoints (using roxygen <code style="white-space: pre;">&#8288;@inheritParams&#8288;</code> tag).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtexr_arguments(
  ageBrackets = NULL,
  aliquotIds = NULL,
  attributeSubset = NULL,
  autolysisScores = NULL,
  cellType = NULL,
  bp_window = NULL,
  chromosome = NULL,
  datasetId = NULL,
  dataTypes = NULL,
  draw = NULL,
  end = NULL,
  excludeDataArray = NULL,
  .featureId = NULL,
  filterMtGene = NULL,
  gencodeId = NULL,
  gencodeIds = NULL,
  gencodeVersion = NULL,
  geneId = NULL,
  geneIds = NULL,
  genomeBuild = NULL,
  hardyScales = NULL,
  hasExpressionData = NULL,
  hasGenotype = NULL,
  itemsPerPage = NULL,
  ischemicTime = NULL,
  ischemicTimeGroups = NULL,
  materialTypes = NULL,
  organizationName = NULL,
  page = NULL,
  pathCategory = NULL,
  phenotypeId = NULL,
  pos = NULL,
  project_id = NULL,
  rin = NULL,
  sampleId = NULL,
  sampleIds = NULL,
  searchTerm = NULL,
  sex = NULL,
  snpId = NULL,
  sortBy = NULL,
  sortDirection = NULL,
  start = NULL,
  subjectIds = NULL,
  tissueSampleIds = NULL,
  tissueSiteDetailId = NULL,
  tissueSiteDetailIds = NULL,
  uberonIds = NULL,
  variantId = NULL,
  variantIds = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gtexr_arguments_+3A_agebrackets">ageBrackets</code></td>
<td>
<p>The age bracket(s) of the donors of interest. Options:
&quot;20-29&quot;, &quot;30-39&quot;, &quot;40-49&quot;, &quot;50-59&quot;, &quot;60-69&quot;, &quot;70-79&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_aliquotids">aliquotIds</code></td>
<td>
<p>Character vector.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_attributesubset">attributeSubset</code></td>
<td>
<p>String. Examples include but are not limited to &quot;sex&quot;, &quot;ageBracket&quot;</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_autolysisscores">autolysisScores</code></td>
<td>
<p>Character vector. Options: &quot;None&quot;, &quot;Mild&quot;, &quot;Moderate&quot;,
&quot;Severe&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_celltype">cellType</code></td>
<td>
<p>String. &quot;Adipocytes&quot;, &quot;Epithelial_cells&quot;, &quot;Hepatocytes&quot;,
&quot;Keratinocytes&quot;, &quot;Myocytes&quot;, &quot;Neurons&quot;, &quot;Neutrophils&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_bp_window">bp_window</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_chromosome">chromosome</code></td>
<td>
<p>String. One of &quot;chr1&quot;, &quot;chr2&quot;, &quot;chr3&quot;, &quot;chr4&quot;, &quot;chr5&quot;,
&quot;chr6&quot;, &quot;chr7&quot;, &quot;chr8&quot;, &quot;chr9&quot;, &quot;chr10&quot;, &quot;chr11&quot;, &quot;chr12&quot;, &quot;chr13&quot;,
&quot;chr14&quot;, &quot;chr15&quot;, &quot;chr16&quot;, &quot;chr17&quot;, &quot;chr18&quot;, &quot;chr19&quot;, &quot;chr20&quot;, &quot;chr21&quot;,
&quot;chr22&quot;, &quot;chrM&quot;, &quot;chrX&quot;, &quot;chrY&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_datasetid">datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: &quot;gtex_v8&quot;, &quot;gtex_snrnaseq_pilot&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_datatypes">dataTypes</code></td>
<td>
<p>Character vector. Options: &quot;RNASEQ&quot;, &quot;WGS&quot;, &quot;WES&quot;, &quot;OMNI&quot;,
&quot;EXCLUDE&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_draw">draw</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_end">end</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_excludedataarray">excludeDataArray</code></td>
<td>
<p>String. Options are TRUE or FALSE</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_.featureid">.featureId</code></td>
<td>
<p>String. A genomic feature e.g. GENCODE ID, RSID or
GTEx Variant ID.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_filtermtgene">filterMtGene</code></td>
<td>
<p>Logical. Exclude mitochondrial genes.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_gencodeid">gencodeId</code></td>
<td>
<p>String. A Versioned GENCODE ID of a gene, e.g.
&quot;ENSG00000065613.9&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_gencodeids">gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c(&quot;ENSG00000132693.12&quot;, &quot;ENSG00000203782.5&quot;).</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_gencodeversion">gencodeVersion</code></td>
<td>
<p>String (default = &quot;v26&quot;). GENCODE annotation release.
Either &quot;v26&quot; or &quot;v19&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_geneid">geneId</code></td>
<td>
<p>String. A gene symbol, a gencode ID, or an Ensemble ID.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_geneids">geneIds</code></td>
<td>
<p>A character vector of gene symbols, versioned gencodeIds, or
unversioned gencodeIds.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_genomebuild">genomeBuild</code></td>
<td>
<p>String. Options: &quot;GRCh38/hg38&quot;, &quot;GRCh37/hg19&quot;. Default =
&quot;GRCh38/hg38&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_hardyscales">hardyScales</code></td>
<td>
<p>Character vector. A list of Hardy Scale(s) of interest.
Options: &quot;Ventilator case&quot;, &quot;Fast death - violent&quot;, &quot;Fast death - natural
causes&quot;, &quot;Intermediate death&quot;, &quot;Slow death&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_hasexpressiondata">hasExpressionData</code></td>
<td>
<p>Logical.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_hasgenotype">hasGenotype</code></td>
<td>
<p>Logical.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_itemsperpage">itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_ischemictime">ischemicTime</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_ischemictimegroups">ischemicTimeGroups</code></td>
<td>
<p>Character vector. Options:
&quot;&lt;= 0&quot;, &quot;1 - 300&quot;, &quot;301 - 600&quot;, &quot;601 - 900&quot;, &quot;901 - 1200&quot;, &quot;1201 - 1500&quot;, &quot;&gt; 1500&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_materialtypes">materialTypes</code></td>
<td>
<p>String, vector. Options: &quot;Cells:Cell Line Viable&quot;,
&quot;DNA:DNA Genomic&quot;, &quot;DNA:DNA Somatic&quot;, &quot;RNA:Total RNA&quot;, &quot;Tissue:PAXgene
Preserved&quot;, &quot;Tissue:PAXgene Preserved Paraffin-embedded&quot;, &quot;Tissue:Fresh
Frozen Tissue&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_organizationname">organizationName</code></td>
<td>
<p>String. Options: &quot;GTEx Consortium&quot; &quot;Kid's First&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_page">page</code></td>
<td>
<p>Integer (default = 0).</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_pathcategory">pathCategory</code></td>
<td>
<p>Character vector. Options: &quot;adenoma&quot;, &quot;amylacea&quot;,
&quot;atelectasis&quot;, &quot;atherosclerosis&quot;, &quot;atherosis&quot;, &quot;atrophy&quot;, &quot;calcification&quot;,
&quot;cirrhosis&quot;, &quot;clean_specimens&quot;, &quot;congestion&quot;, &quot;corpora_albicantia&quot;, &quot;cyst&quot;,
&quot;desquamation&quot;, &quot;diabetic&quot;, &quot;dysplasia&quot;, &quot;edema&quot;, &quot;emphysema&quot;,
&quot;esophagitis&quot;, &quot;fibrosis&quot;, &quot;gastritis&quot;, &quot;glomerulosclerosis&quot;, &quot;goiter&quot;,
&quot;gynecomastoid&quot;, &quot;hashimoto&quot;, &quot;heart_failure_cells&quot;, &quot;hemorrhage&quot;,
&quot;hepatitis&quot;, &quot;hyalinization&quot;, &quot;hypereosinophilia&quot;, &quot;hyperplasia&quot;,
&quot;hypertrophy&quot;, &quot;hypoxic&quot;, &quot;infarction&quot;, &quot;inflammation&quot;, &quot;ischemic_changes&quot;,
&quot;macrophages&quot;, &quot;mastopathy&quot;, &quot;metaplasia&quot;, &quot;monckeberg&quot;, &quot;necrosis&quot;,
&quot;nephritis&quot;, &quot;nephrosclerosis&quot;, &quot;no_abnormalities&quot;, &quot;nodularity&quot;,
&quot;pancreatitis&quot;, &quot;pigment&quot;, &quot;pneumonia&quot;, &quot;post_menopausal&quot;, &quot;prostatitis&quot;,
&quot;saponification&quot;, &quot;scarring&quot;, &quot;sclerotic&quot;, &quot;solar_elastosis&quot;,
&quot;spermatogenesis&quot;, &quot;steatosis&quot;, &quot;sweat_glands&quot;, &quot;tma&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_phenotypeid">phenotypeId</code></td>
<td>
<p>String. See
<a href="https://www.gtexportal.org/home/faq#splicingPhenotypeId">GTEx portal FAQs</a>
for further details.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_pos">pos</code></td>
<td>
<p>Integer, vector.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_project_id">project_id</code></td>
<td>
<p>String. Options: &quot;gtex&quot;, &quot;adult-gtex&quot;, &quot;egtex&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_rin">rin</code></td>
<td>
<p>Integer vector.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_sampleid">sampleId</code></td>
<td>
<p>String. <code style="white-space: pre;">&#8288;^GTEX-[A-Z0-9]{5}-[0-9]{4}-SM-[A-Z0-9]{5}$&#8288;</code></p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_sampleids">sampleIds</code></td>
<td>
<p>Character vector. GTEx sample ID.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_searchterm">searchTerm</code></td>
<td>
<p>String.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_sex">sex</code></td>
<td>
<p>String. Options: &quot;male&quot;, &quot;female&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_snpid">snpId</code></td>
<td>
<p>String</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_sortby">sortBy</code></td>
<td>
<p>String. Options: &quot;sampleId&quot;, &quot;ischemicTime&quot;, &quot;aliquotId&quot;,
&quot;tissueSampleId&quot;, &quot;hardyScale&quot;, &quot;pathologyNotes&quot;, &quot;ageBracket&quot;,
&quot;tissueSiteDetailId&quot;, &quot;sex&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_sortdirection">sortDirection</code></td>
<td>
<p>String. Options: &quot;asc&quot;, &quot;desc&quot;. Default = &quot;asc&quot;.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_start">start</code></td>
<td>
<p>Integer.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_subjectids">subjectIds</code></td>
<td>
<p>Character vector. GTEx subject ID.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_tissuesampleids">tissueSampleIds</code></td>
<td>
<p>Array of strings. A list of Tissue Sample ID(s).</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_tissuesitedetailid">tissueSiteDetailId</code></td>
<td>
<p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. &quot;Whole_Blood&quot;; use <code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code>
to see valid values) or an Ontology ID.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_tissuesitedetailids">tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. &quot;Whole_Blood&quot;; use
<code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_uberonids">uberonIds</code></td>
<td>
<p>Character vector of Uberon IDs (e.g. &quot;UBERON:EFO_0000572&quot;;
use <code><a href="#topic+get_tissue_site_detail">get_tissue_site_detail()</a></code> to see valid values).</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_variantid">variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td></tr>
<tr><td><code id="gtexr_arguments_+3A_variantids">variantIds</code></td>
<td>
<p>Character vector. Gtex variant IDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>NULL</code> invisibly. Used for documentation only.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
