<!DOCTYPE html><html><head><title>Help for package gtWAS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gtWAS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alldata'>
<p>Data including base and expression data</p></a></li>
<li><a href='#Association'>
<p>Genome and Transcriptome Wide Association</p></a></li>
<li><a href='#gtWAS-package'>
<p>Genome and Transcriptome Wide Association Study</p></a></li>
<li><a href='#ModelFit'>
<p>Compute model fit statistics</p></a></li>
<li><a href='#StepOne'>
<p>Compute minimum p value and information criteria statistics in one step</p></a></li>
<li><a href='#stp'>
<p>stepwise regression</p></a></li>
<li><a href='#Tdata'>
<p>Phenotype data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genome and Transcriptome Wide Association Study</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-06-01</td>
</tr>
<tr>
<td>Author:</td>
<td>JunhuiLi WenxinLiu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>JunhuiLi&lt;junhuili@cau.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Quantitative trait loci mapping and genome wide association analysis are used to find candidate molecular marker or region associated with phenotype based on linkage analysis and linkage disequilibrium. Gene expression quantitative trait loci mapping is used to find candidate molecular marker or region associated with gene expression. In this package, we applied the method in Liu W. (2011) &lt;<a href="https://doi.org/10.1007%2Fs00122-011-1631-7">doi:10.1007/s00122-011-1631-7</a>&gt; and Gusev A. (2016) &lt;<a href="https://doi.org/10.1038%2Fng.3506">doi:10.1038/ng.3506</a>&gt; to genome and transcriptome wide association study, which is aimed at revealing the association relationship between phenotype and molecular markers, expression levels, molecular markers nested within different related expression effect and expression effect nested within different related molecular marker effect. F test based on full and reduced model are performed to obtain p value or likelihood ratio statistic. The best linear model can be obtained by stepwise regression analysis.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-06-01 09:35:20 UTC; lenovo</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-06-01 09:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='alldata'>
Data including base and expression data
</h2><span id='topic+alldata'></span>

<h3>Description</h3>

<p>Data including base and expression data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("alldata")</code></pre>


<h3>Format</h3>

<p>A data frame with 100 observations on the following 200 variables.
</p>

<p>The first 100th variables are SNP and the second are expression data

</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(alldata)
</code></pre>

<hr>
<h2 id='Association'>
Genome and Transcriptome Wide Association
</h2><span id='topic+Association'></span>

<h3>Description</h3>

<p>Reveal the association relationship between phenotype and molecular marker, expression effect, expression effect nested within molecular marker and molecular marker effect nested within expression effect
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Association(Tdata,alldata,independent="B(E)",Elevels=c(0.05,0.95),selection="stepwise",
      select="SL",Choose=NULL,SL=c(0.05,0.15,0.15),correct="Bonferroni")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Association_+3A_tdata">Tdata</code></td>
<td>

<p>Phenotye data
</p>
</td></tr>
<tr><td><code id="Association_+3A_alldata">alldata</code></td>
<td>

<p>Independent variables including molecular marker or corresponding expression effect related to marker on transcriptome level
</p>
</td></tr>
<tr><td><code id="Association_+3A_independent">independent</code></td>
<td>

<p>Indicator of independent variable to be used in linear model. 'B' is molecular marker effect, 'E' is expression effect, 'B(E)' is molecular marker nesting expression effect and 'E(B)' is expression effect nesting molecular marker effect
</p>
</td></tr>
<tr><td><code id="Association_+3A_elevels">Elevels</code></td>
<td>

<p>Percentage of threshold value for different expression levels
</p>
</td></tr>
<tr><td><code id="Association_+3A_selection">selection</code></td>
<td>

<p>Model selection method including &quot;forward&quot; and &quot;stepwise&quot;,forward selection starts with no effects in the model and adds effects, while stepwise regression is similar to the forward method except that effects already in the model do not necessarily stay there
</p>
</td></tr>
<tr><td><code id="Association_+3A_select">select</code></td>
<td>

<p>Specifies the criterion that uses to determine the order in which effects enter and/or leave at each step of the specified selection method including Akaike Information Criterion(AIC), the Corrected form of Akaike Information Criterion(AICc),Bayesian Information Criterion(BIC),Schwarz criterion(SBC), Significant Levels(SL) and so on
</p>
</td></tr>
<tr><td><code id="Association_+3A_choose">Choose</code></td>
<td>

<p>Chooses from the list of models at the steps of the selection process the model that yields the best value of the specified criterion. If the optimal value of the specified criterion occurs for models at more than one step, then the model with the smallest number of parameters is chosen. If you do not specify the Choose option, then the model selected is the model at the final step in the selection process
</p>
</td></tr>
<tr><td><code id="Association_+3A_sl">SL</code></td>
<td>

<p>Thresholds for significant levels of association and stepwise regression
</p>
</td></tr>
<tr><td><code id="Association_+3A_correct">correct</code></td>
<td>

<p>Bonferroni correct or the p value method for significant levels, default is bonferroni
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p value of all effect and significant ones
</p>


<h3>Author(s)</h3>

<p>JunhuiLi
</p>


<h3>References</h3>

<p>Junhui Li, Haixiao Hu, Yujie Meng, Kun Cheng, Guoliang Li, Wenxin Liu, and Shaojiang Chen.(2016)Pleiotropic QTL detection for stalk traits in maize and related R package programming. Journal of China Agricultural University. DOI 10.11841/j.issn.1007-4333.2016.06.00
</p>
<p>Gusev, A., Ko, A., Shi, H., Bhatia, G., Chung, W., &amp; Penninx, B. W., et al. (2016). Integrative approaches for large-scale transcriptome-wide association studies. Nature Genetics, 48(3), 245.
</p>
<p>Hurvich, C. M., &amp; Tsai, C. (1989). Regression and time series model selection in small samples. Biometrika, 76(2), 297-307.
</p>
<p>Judge, &amp; GeorgeG. (1985). The Theory and practice of econometrics /-2nd ed. The Theory and practice of econometrics /. Wiley.
</p>
<p>Mcquarrie, A. D. R., &amp; Tsai, C. L. (1998). Regression and Time Series Model Selection. Regression and time series model selection /. World Scientific.
</p>
<p>Sawa, T. (1978). Information criteria for discriminating among alternative regression models. Econometrica, 46(6), 1273-1291.
</p>
<p>Schwarz, G. (1978). Estimating the dimension of a model. Annals of Statistics, 6(2), pags. 15-18.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Tdata)
data(alldata)
Edata &lt;- alldata[,1:100+100]
Bdata &lt;- alldata[,1:100]
BE &lt;- "B(E)"
EB &lt;- "E(B)"
B &lt;- "B"
E &lt;- "E"

#for "B(E)"
#Association(Tdata,alldata,BE,Elevels=c(0.05,0.95),selection='stepwise',
#select="SL",Choose=NULL,SL=c(0.05,0.15,0.15),correct = "Bonferroni")

#for "E(B)" with Elevels = null
#Association(Tdata,alldata,EB,Elevels=NULL,selection='stepwise',
#select="SL",Choose=NULL,SL=c(0.05,0.15,0.15),correct = "Bonferroni")

#for "E" with Elevels = null
#Association(Tdata,Edata,E,Elevels=NULL,selection='stepwise',
#select="SL",Choose=NULL,SL=c(0.05,0.15,0.15),correct = "Bonferroni")

#for "B"
#Association(Tdata,Bdata,B,Elevels=NULL,selection='stepwise',
#select="SL",Choose=NULL,SL=c(0.05,0.15,0.15),correct = "Bonferroni")
</code></pre>

<hr>
<h2 id='gtWAS-package'>
Genome and Transcriptome Wide Association Study
</h2><span id='topic+gtWAS-package'></span>

<h3>Description</h3>

<p>Quantitative trait loci mapping and genome wide association analysis are used to find candidate molecular marker or region associated with phenotype based on linkage analysis and linkage disequilibrium. Expression quantitative trait loci mapping is used to find candidate molecular marker or region associated with gene expression. This package is aimed at revealing the association relationship between phenotype and molecular markers, expression levels, molecular markers with different related expression levels and expression levels with different related molecular marker. F test based on full and reduced model are performed to obtain p value or likelihood ratio statistic. The best linear model can be obtained by stepwise regression analysis.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> gtWAS</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2019-06-01</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>JunhuiLi WeninLiu
</p>
<p>Maintainer: JunhuiLi&lt;junhuili@cau.edu.cn&gt;
</p>


<h3>References</h3>

<p>Junhui Li, Haixiao Hu, Yujie Meng, Kun Cheng, Guoliang Li, Wenxin Liu, and Shaojiang Chen.(2016)Pleiotropic QTL detection for stalk traits in maize and related R package programming. Journal of China Agricultural University. DOI 10.11841/j.issn.1007-4333.2016.06.00
</p>
<p>Liu W., Maurer H.P., Reif J.C., Melchinger A.E., Utz F., Tucker M.R., Ranc N., Della Porta G., Wurschum T. (2013) Optimum Design of Family Structure and Allocation of Resources in Association Mapping with Lines from Multiple Crosses. Heredity 110: 71-79
</p>
<p>Gusev, A., Ko, A., Shi, H., Bhatia, G., Chung, W., &amp; Penninx, B. W., et al. (2016). Integrative approaches for large-scale transcriptome-wide association studies. Nature Genetics, 48(3), 245.
</p>
<p>Hurvich, C. M., &amp; Tsai, C. (1989). Regression and time series model selection in small samples. Biometrika, 76(2), 297-307.
</p>
<p>Judge, &amp; GeorgeG. (1985). The Theory and practice of econometrics /-2nd ed. The Theory and practice of econometrics /. Wiley.
</p>
<p>Mcquarrie, A. D. R., &amp; Tsai, C. L. (1998). Regression and Time Series Model Selection. Regression and time series model selection /. World Scientific.
</p>
<p>Sawa, T. (1978). Information criteria for discriminating among alternative regression models. Econometrica, 46(6), 1273-1291.
</p>
<p>Schwarz, G. (1978). Estimating the dimension of a model. Annals of Statistics, 6(2), pags. 15-18.
</p>

<hr>
<h2 id='ModelFit'>
Compute model fit statistics 
</h2><span id='topic+ModelFit'></span>

<h3>Description</h3>

<p>Compute model fit statistics based on a given criteria for linear model function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ModelFit(criteria, lmresult, nObs, sigma_sqr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ModelFit_+3A_criteria">criteria</code></td>
<td>

<p>The class of criteria including Akaike information criterion(AIC), the corrected form of Akaike information criterion(AICc), Bayesian information criterion(BIC), Schwarz criterion(SBC) and significant levels(SL)
</p>
</td></tr>
<tr><td><code id="ModelFit_+3A_lmresult">lmresult</code></td>
<td>

<p>Result of linear model function
</p>
</td></tr>
<tr><td><code id="ModelFit_+3A_nobs">nObs</code></td>
<td>

<p>Number of observation
</p>
</td></tr>
<tr><td><code id="ModelFit_+3A_sigma_sqr">sigma_sqr</code></td>
<td>

<p>The estimation of pure error variance for the full model in regression
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric of model fit statistics 
</p>


<h3>Author(s)</h3>

<p>JunhuiLi
</p>


<h3>References</h3>

<p>Hurvich, C. M., &amp; Tsai, C. (1989). Regression and time series model selection in small samples. Biometrika, 76(2), 297-307.
</p>
<p>Judge, &amp; GeorgeG. (1985). The Theory and practice of econometrics /-2nd ed. The Theory and practice of econometrics /. Wiley.
</p>
<p>Mcquarrie, A. D. R., &amp; Tsai, C. L. (1998). Regression and Time Series Model Selection. Regression and time series model selection /. World Scientific.
</p>
<p>R.S. Sparks, W. Zucchini, &amp; D. Coutsourides. (1985). On variable selection in multivariate regression. Communication in Statistics- Theory and Methods, 14(7), 1569-1587.
</p>
<p>Sawa, T. (1978). Information criteria for discriminating among alternative regression models. Econometrica, 46(6), 1273-1291.
</p>
<p>Schwarz, G. (1978). Estimating the dimension of a model. Annals of Statistics, 6(2), pags. 15-18.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(4)
YX &lt;- matrix(rnorm(200,20,4),20,10)
YX &lt;- as.data.frame(YX)
colnames(YX) &lt;- c("Y1","Y2",paste("X",c(1:8),sep=""))
lm_formula &lt;- as.formula("Y1~X1+X2+X3+X4+X5")
lmresult &lt;- lm(lm_formula,data=YX)
ModelFit("SBC", lmresult, nrow(YX), 0)
</code></pre>

<hr>
<h2 id='StepOne'>
Compute minimum p value and information criteria statistics in one step
</h2><span id='topic+StepOne'></span>

<h3>Description</h3>

<p>Compute minimum p value and information criteria statistics in one step by adding or removing a variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>StepOne(findIn, independent, criteria, varIn, TMdata, sigma)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="StepOne_+3A_findin">findIn</code></td>
<td>

<p>Logical value for adding or removing independent variables in regression model, the parameter is ture for removing a variable otherwise adding a variable
</p>
</td></tr>
<tr><td><code id="StepOne_+3A_independent">independent</code></td>
<td>

<p>Indicator of independent variable to be used in linear model. 'B' is molecular marker effect, 'E' is expression effect, 'B(E)' is expression effect nested within molecular marker effect and 'E(B)' is molecular marker effect nested within expression effect
</p>
</td></tr>
<tr><td><code id="StepOne_+3A_criteria">criteria</code></td>
<td>

<p>Specifies the criterion that uses to determine the order in which effects enter and/or leave at each step of the specified selection method including Akaike Information Criterion(AIC), the Corrected form of Akaike Information Criterion(AICc),Bayesian Information Criterion(BIC),Schwarz criterion(SBC),Hannan and Quinn Information Criterion(HQ), Significant Levels(SL) and so on
</p>
</td></tr>
<tr><td><code id="StepOne_+3A_varin">varIn</code></td>
<td>

<p>Sequence of vector for every independent variables, 1 indicates this independent variable stays in the regression model, and 0 is not in the model
</p>
</td></tr>
<tr><td><code id="StepOne_+3A_tmdata">TMdata</code></td>
<td>

<p>Phenotype data
</p>
</td></tr>
<tr><td><code id="StepOne_+3A_sigma">sigma</code></td>
<td>

<p>The estimation of pure error variance from the full model in regression
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of minimum p value or information criteria statistics, sequence id of independent variable staying in the model, linear model regression and rank of last step linear model
</p>


<h3>Author(s)</h3>

<p>JunhuiLi
</p>


<h3>References</h3>

<p>Hurvich, C. M., &amp; Tsai, C. (1989). Regression and time series model selection in small samples. Biometrika, 76(2), 297-307.
</p>
<p>Judge, &amp; GeorgeG. (1985). The Theory and practice of econometrics /-2nd ed. The Theory and practice of econometrics /. Wiley.
</p>
<p>Mcquarrie, A. D. R., &amp; Tsai, C. L. (1998). Regression and Time Series Model Selection. Regression and time series model selection /. World Scientific.
</p>
<p>R.S. Sparks, W. Zucchini, &amp; D. Coutsourides. (1985). On variable selection in multivariate regression. Communication in Statistics- Theory and Methods, 14(7), 1569-1587.
</p>
<p>Sawa, T. (1978). Information criteria for discriminating among alternative regression models. Econometrica, 46(6), 1273-1291.
</p>
<p>Schwarz, G. (1978). Estimating the dimension of a model. Annals of Statistics, 6(2), pags. 15-18.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Tdata)
data(alldata)
TMdata &lt;- cbind(Tdata,alldata[,1:100])
findIn = FALSE
independent = "B"
varIn &lt;- rep(0,100)
StepOne(findIn,independent,criteria="SBC",varIn,TMdata,sigma=0)
</code></pre>

<hr>
<h2 id='stp'>
stepwise regression
</h2><span id='topic+stp'></span>

<h3>Description</h3>

<p>Stepwise regression for model selection using linear model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stp(AllData, independent, selection = "stepwise", select = "SL",
sle = 0.15, sls = 0.15, Choose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stp_+3A_alldata">AllData</code></td>
<td>

<p>Data about dependent and independent variable data
</p>
</td></tr>
<tr><td><code id="stp_+3A_independent">independent</code></td>
<td>

<p>Indicator of independent variable to be used in linear model. 'B' is molecular marker effect, 'E' is expression data, 'B(E)' is expression effect nested within molecular marker effect and 'E(B)' is molecular marker effect nested within expression effect
</p>
</td></tr>
<tr><td><code id="stp_+3A_selection">selection</code></td>
<td>

<p>Model selection method including &quot;forward&quot; and &quot;stepwise&quot;,forward selection starts with no effects in the model and adds effects, while stepwise regression is similar to the forward method except that effects already in the model do not necessarily stay there
</p>
</td></tr>
<tr><td><code id="stp_+3A_select">select</code></td>
<td>

<p>Specifies the criterion that uses to determine the order in which effects enter and/or leave at each step of the specified selection method including Akaike Information Criterion(AIC), the Corrected form of Akaike Information Criterion(AICc),Bayesian Information Criterion(BIC),Schwarz criterion(SBC),Hannan and Quinn Information Criterion(HQ), Significant Levels(SL) and so on
</p>
</td></tr>
<tr><td><code id="stp_+3A_sle">sle</code></td>
<td>

<p>Specifies the significance level for entry, default is 0.15
</p>
</td></tr>
<tr><td><code id="stp_+3A_sls">sls</code></td>
<td>

<p>Specifies the significance level for staying in the model, default is 0.15
</p>
</td></tr>
<tr><td><code id="stp_+3A_choose">Choose</code></td>
<td>

<p>Chooses from the list of models at the steps of the selection process the model that yields the best value of the specified criterion. If the optimal value of the specified criterion occurs for models at more than one step, then the model with the smallest number of parameters is chosen. If you do not specify the Choose option, then the model selected is the model at the final step in the selection process
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>JunhuiLi
</p>


<h3>References</h3>

<p>Hurvich, C. M., &amp; Tsai, C. (1989). Regression and time series model selection in small samples. Biometrika, 76(2), 297-307.
</p>
<p>Judge, &amp; GeorgeG. (1985). The Theory and practice of econometrics /-2nd ed. The Theory and practice of econometrics /. Wiley.
</p>
<p>Mcquarrie, A. D. R., &amp; Tsai, C. L. (1998). Regression and Time Series Model Selection. Regression and time series model selection /. World Scientific.
</p>
<p>R.S. Sparks, W. Zucchini, &amp; D. Coutsourides. (1985). On variable selection in multivariate regression. Communication in Statistics- Theory and Methods, 14(7), 1569-1587.
</p>
<p>Sawa, T. (1978). Information criteria for discriminating among alternative regression models. Econometrica, 46(6), 1273-1291.
</p>
<p>Schwarz, G. (1978). Estimating the dimension of a model. Annals of Statistics, 6(2), pags. 15-18.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Tdata)
data(alldata)
independent &lt;- "B"
nbase &lt;- 100
AllData &lt;- cbind(Tdata[colnames(Tdata)[1]],alldata[,1:nbase])
AllData &lt;- sapply(AllData, as.numeric)
AllData &lt;- as.data.frame(AllData)
stp(AllData,independent,selection="stepwise",select="SBC",sle=0.05,sls=0.05,Choose=NULL)
</code></pre>

<hr>
<h2 id='Tdata'>
Phenotype data
</h2><span id='topic+Tdata'></span>

<h3>Description</h3>

<p>Phenotype data by rnorm function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Tdata")</code></pre>


<h3>Format</h3>

<p>A data frame with 100 observations on the following variable.
</p>

<dl>
<dt><code>Trait1</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(Tdata)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
