<!DOCTYPE html><html lang="en"><head><title>Help for package GUniFrac</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GUniFrac}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adonis3'><p>Permutational Multivariate Analysis of Variance Using Distance Matrices (Freedman-Lane permutation)</p></a></li>
<li><a href='#dICC'><p>Distance-based Intra-Class Correlation Coefficient</p></a></li>
<li><a href='#dICC.SE.asympt'><p>Asymptotic Standard Error of Distance-based Intra-Class Correlation Coefficient</p></a></li>
<li><a href='#dICC.SE.bt'><p>Bootstrap Standard Error of Distance-based Intra-Class Correlation Coefficient</p></a></li>
<li><a href='#dmanova'><p>Distance-based Multivariate Analysis of Variance (Analytical P-value Calculation)</p></a></li>
<li><a href='#GMPR'>
<p>Geometric Mean of Pairwise Ratios (GMPR) Normalization for Zero-inflated Count Data</p></a></li>
<li><a href='#GUniFrac'>
<p>Generalized UniFrac distances for comparing microbial communities.</p></a></li>
<li><a href='#PermanovaG'>
<p>Permutational Multivariate Analysis of Variance Using Multiple Distance Matrices</p></a></li>
<li><a href='#PermanovaG2'>
<p>Permutational Multivariate Analysis of Variance Using Multiple Distance Matrices(Freedman-Lane Permutation)</p>
</p></a></li>
<li><a href='#Rarefy'>
<p>Rarefy a Count Table to Equal Sequencing Depth</p></a></li>
<li><a href='#SimulateMSeq'><p>A Semiparametric Model-based Microbiome Sequencing Data Simulator for Cross-sectional and Case-control Studies</p></a></li>
<li><a href='#SimulateMSeqC'><p>A Semiparametric Model-based Microbiome Sequencing Data Simulator for Longitudinal, Matched-pair, and Replicate Sampling Designs</p></a></li>
<li><a href='#stool.otu.tab'>
<p>Stool Microbiome OTU Count Table</p></a></li>
<li><a href='#throat.meta'>
<p>Throat Microbiome Meta Data</p></a></li>
<li><a href='#throat.otu.tab'>
<p>Throat Microbiome OTU Count Table</p></a></li>
<li><a href='#throat.tree'>
<p>UPGMA Tree of Throat Microbiome OTUs</p></a></li>
<li><a href='#vaginal.otu.tab'>
<p>Vaginal Microbiome OTU Count Table</p></a></li>
<li><a href='#ZicoSeq'><p>A linear Model-based Permutation Test for Differential Abundance Analysis of Microbiome Data and Other Omics Data</p></a></li>
<li><a href='#ZicoSeq.plot'><p>A Plot Function for Visualizing the ZicoSeq Results</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Generalized UniFrac Distances, Distance-Based Multivariate
Methods and Feature-Based Univariate Methods for Microbiome
Data Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-13</td>
</tr>
<tr>
<td>Author:</td>
<td>Jun Chen, Xianyang Zhang, Lu Yang, Lujun Zhang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jun Chen &lt;chen.jun2@mayo.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A suite of methods for powerful and robust microbiome data analysis including data normalization, data simulation, community-level association testing and differential abundance analysis. It implements generalized UniFrac distances,  Geometric Mean of Pairwise Ratios (GMPR) normalization, semiparametric data simulator, distance-based statistical methods, and feature-based statistical methods. The distance-based statistical methods include three extensions of PERMANOVA: (1) PERMANOVA using the Freedman-Lane permutation scheme, (2) PERMANOVA omnibus test using multiple matrices, and  (3) analytical approach to approximating PERMANOVA p-value. Feature-based statistical methods include linear model-based methods for differential abundance analysis of zero-inflated high-dimensional compositional data. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ade4, knitr, markdown, ggpubr</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.13), vegan, ggplot2, matrixStats, Matrix, ape,
parallel, stats, utils, statmod, rmutil, dirmult, MASS,
ggrepel, foreach, modeest, inline, methods</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-13 22:22:15 UTC; m123485</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-14 00:02:32 UTC</td>
</tr>
</table>
<hr>
<h2 id='adonis3'>Permutational Multivariate Analysis of Variance Using Distance Matrices (Freedman-Lane permutation)</h2><span id='topic+adonis3'></span>

<h3>Description</h3>

<p>Analysis of variance using distance matrices &mdash; for
partitioning distance matrices among sources of variation and fitting
linear models (e.g., factors, polynomial regression) to distance 
matrices; uses a permutation test (Freedman-Lane permutation) with pseudo-<code class="reqn">F</code> ratios.</p>


<h3>Usage</h3>

<pre><code class='language-R'>adonis3(formula, data, permutations = 999, method = "bray",
    strata = NULL, contr.unordered = "contr.sum",
    contr.ordered = "contr.poly", parallel = getOption("mc.cores"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adonis3_+3A_formula">formula</code></td>
<td>
<p>model formula. The LHS must be either a community
data matrix or a dissimilarity matrix, e.g., from
<code><a href="vegan.html#topic+vegdist">vegdist</a></code> or <code><a href="stats.html#topic+dist">dist</a></code>.  If the LHS is a data
matrix, function <code><a href="vegan.html#topic+vegdist">vegdist</a></code> will be used to find the
dissimilarities. The RHS defines the independent variables. These
can be continuous variables or factors, they can be transformed
within the formula, and they can have interactions as in a typical
<code><a href="stats.html#topic+formula">formula</a></code>. </p>
</td></tr>
<tr><td><code id="adonis3_+3A_data">data</code></td>
<td>
<p> the data frame for the independent variables.</p>
</td></tr>
<tr><td><code id="adonis3_+3A_permutations">permutations</code></td>
<td>
<p>a list of control values for the permutations
as returned by the function <code><a href="permute.html#topic+how">how</a></code>, or the
number of permutations required, or a permutation matrix where each
row gives the permuted indices.</p>
</td></tr>
<tr><td><code id="adonis3_+3A_method">method</code></td>
<td>
<p> the name of any method used in <code><a href="vegan.html#topic+vegdist">vegdist</a></code> to
calculate pairwise distances if the left hand side of the
<code>formula</code> was a data frame or a matrix. </p>
</td></tr>
<tr><td><code id="adonis3_+3A_strata">strata</code></td>
<td>
<p> groups (strata) within which to constrain permutations.  </p>
</td></tr>
<tr><td><code id="adonis3_+3A_contr.unordered">contr.unordered</code>, <code id="adonis3_+3A_contr.ordered">contr.ordered</code></td>
<td>
<p>contrasts used for the design
matrix (default in R is dummy or treatment contrasts for unordered
factors). </p>
</td></tr>
<tr><td><code id="adonis3_+3A_parallel">parallel</code></td>
<td>
<p>number of parallel processes or a predefined socket
cluster.  With <code>parallel = 1</code> uses ordinary, non-parallel
processing. The parallel processing is done with <span class="pkg">parallel</span>
package.</p>
</td></tr>
<tr><td><code id="adonis3_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>vegdist</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>adonis3</code> is the re-implementation of the <code>adonis</code> function in
the <code>vegan</code> package based on the Freedman-Lane permutation scheme
(Freedman &amp; Lane (1983), Hu &amp; Satten (2020)).  The original
implementation in the <code>vegan</code> package is directly based on the algorithm of Anderson (2001) and
performs a sequential test of terms. Statistical significance is assessed
based on permuting the distance matrix. We found that such permutation
will lead to power loss in testing the effect of a covariate of interest while adjusting
for other covariates (confounders). The power loss is more evident  when the confounders' effects
are strong, the correlation between the covariate of interest and the confounders is high, and
the sample size is small. When the sample size is large than 100, the difference is usually small.
The new implementation is revised on the <code>adonis</code> function with the same interface.
</p>


<h3>Value</h3>

<p>Function <code>adonis3</code> returns an object of class <code>"adonis"</code> with
following components:
</p>
<table role = "presentation">
<tr><td><code>aov.tab</code></td>
<td>
<p>typical AOV table showing sources of variation,
degrees of freedom, sequential sums of squares, mean squares,
<code class="reqn">F</code> statistics, partial <code class="reqn">R^2</code> and <code class="reqn">P</code>
values, based on <code class="reqn">N</code> permutations. </p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p> matrix of coefficients of the linear model, with
rows representing sources of variation and columns representing
species; each column represents a fit of a species abundance to the
linear model. These are what you get when you fit one species to
your predictors. These are NOT available if you supply the distance
matrix in the formula, rather than the site x species matrix</p>
</td></tr> 
<tr><td><code>coef.sites</code></td>
<td>
<p> matrix of coefficients of the linear model, with
rows representing sources of variation and columns representing
sites; each column represents a fit of a sites distances (from all
other sites) to the  linear model. These are what you get when you
fit distances of one site to
your predictors. </p>
</td></tr>   
<tr><td><code>f.perms</code></td>
<td>
<p> an <code class="reqn">N</code> by <code class="reqn">m</code> matrix of the null <code class="reqn">F</code>
statistics for each source of variation based on <code class="reqn">N</code>
permutations of the data. The permutations can be inspected with
<code><a href="vegan.html#topic+permustats">permustats</a></code> and its support functions.</p>
</td></tr>
<tr><td><code>model.matrix</code></td>
<td>
<p>the <code><a href="stats.html#topic+model.matrix">model.matrix</a></code> for the right hand
side of the formula.</p>
</td></tr>
<tr><td><code>terms</code></td>
<td>
<p>the <code><a href="stats.html#topic+terms">terms</a></code> component of the model.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin Henry H. Stevens (<code>adonis</code>) and Jun Chen
(<code>adonis3</code>). </p>


<h3>References</h3>

<p>Anderson, M.J. 2001. A new method for non-parametric multivariate
analysis of variance. <em>Austral Ecology</em>, <strong>26</strong>: 32&ndash;46.
</p>
<p>Freedman D. &amp; Lane D. 1983. A nonstochastic interpretation of reported significance
levels. <em>Journal of Business and Economic Statistics</em>, <strong>1</strong>292&ndash;298.
</p>
<p>Hu, Y. J. &amp; Satten, G. A. 2020. Testing hypotheses about the microbiome using the 
linear decomposition model (LDM). <em>JBioinformatics</em>, <strong> 36(14)</strong> : 4106-4115.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

groups &lt;- throat.meta$SmokingStatus

# Rarefaction
otu.tab.rff &lt;- Rarefy(throat.otu.tab)$otu.tab.rff

# Calculate the UniFrac distance
unifracs &lt;- GUniFrac(otu.tab.rff, throat.tree, alpha=c(0, 0.5, 1))$unifracs

# Test the smoking effect based on unweighted UniFrac distance, adjusting sex
adonis3(as.dist(unifracs[, , 'd_UW']) ~ Sex + SmokingStatus, data = throat.meta)

## End(Not run)

</code></pre>

<hr>
<h2 id='dICC'>Distance-based Intra-Class Correlation Coefficient</h2><span id='topic+dICC'></span>

<h3>Description</h3>

<p>Distance-based Intra-Class Correlation Coefficient (ICC) is an extension of the traditional univariate ICC to multivariate case, where the relationship between
the multivariate measurements is summarized in a distance matrix. It quantifies the ability of a measurement method in reproducing the inter-sample relationship. </p>


<h3>Usage</h3>

<pre><code class='language-R'> dICC(dist.mat, strata)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dICC_+3A_dist.mat">dist.mat</code></td>
<td>
<p>a symmetrical distance matrix between all the replicates (technical and biological replicates).</p>
</td></tr>
<tr><td><code id="dICC_+3A_strata">strata</code></td>
<td>
<p>a factor with each level corresponding to a biological replicate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Function <code>dICC</code> returns a list with the following component:
</p>
<table role = "presentation">
<tr><td><code>ICC</code></td>
<td>
<p>the distance-based ICC value.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jun Chen and Xianyang Zhang</p>


<h3>References</h3>

<p>Chen, J. &amp; Zhang, X. 2022.  dICC: Distance-based Intraclass Correlation Coefficient for Metagenomic Reproducibility Studies.
submitted.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dICC.SE.asympt">dICC.SE.asympt</a></code>, <code><a href="#topic+dICC.SE.bt">dICC.SE.bt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate the error-free measurements of 20 biological replicates, each with four dimensions
y &lt;- matrix(rnorm(80), nrow = 20, ncol = 4)

# Generate two technical replicates for each biological replicate and add measurement errors
y1 &lt;- y + matrix(rnorm(80, sd = 0.5), nrow = 20, ncol = 4)
y2 &lt;- y + matrix(rnorm(80, sd = 0.5), nrow = 20, ncol = 4)
y12 &lt;- rbind(y1, y2)

# Create the design vector
strata &lt;- factor(rep(1 : 20, 2))

# Calculate the distance and distance-based ICC
dist.mat &lt;- as.matrix(dist(y12))
dICC(dist.mat, strata)


</code></pre>

<hr>
<h2 id='dICC.SE.asympt'>Asymptotic Standard Error of Distance-based Intra-Class Correlation Coefficient</h2><span id='topic+dICC.SE.asympt'></span>

<h3>Description</h3>

<p>Calculate the standard error of the distance-based intra-class correlation coefficient based on the asymptotic distribution. </p>


<h3>Usage</h3>

<pre><code class='language-R'> dICC.SE.asympt(dist.mat, strata)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dICC.SE.asympt_+3A_dist.mat">dist.mat</code></td>
<td>
<p>a symmetrical distance matrix between all the replicates (technical and biological replicates).</p>
</td></tr>
<tr><td><code id="dICC.SE.asympt_+3A_strata">strata</code></td>
<td>
<p>a factor with each level corresponding to a biological replicate. Currently only supports 
the same number of technical replicates for each biological replicate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Function <code>dICC</code> returns a list with the following component:
</p>
<table role = "presentation">
<tr><td><code>ICC</code></td>
<td>
<p>the distance-based ICC value.</p>
</td></tr>
<tr><td><code>SE</code></td>
<td>
<p>the standard error.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jun Chen and Xianyang Zhang</p>


<h3>References</h3>

<p>Chen, J. &amp; Zhang, X. 2022.  dICC: Distance-based Intraclass Correlation Coefficient for Metagenomic Reproducibility Studies.
submitted.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dICC">dICC</a></code>, <code><a href="#topic+dICC.SE.bt">dICC.SE.bt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Generate the error-free measurements of 20 biological replicates, each with four dimensions
y &lt;- matrix(rnorm(80), nrow = 20, ncol = 4)

# Generate two technical replicates for each biological replicate and add measurement errors
y1 &lt;- y + matrix(rnorm(80, sd = 0.5), nrow = 20, ncol = 4)
y2 &lt;- y + matrix(rnorm(80, sd = 0.5), nrow = 20, ncol = 4)
y12 &lt;- rbind(y1, y2)

# Create the design vector
strata &lt;- factor(rep(1 : 20, 2))

# Calculate the distance and distance-based ICC
dist.mat &lt;- as.matrix(dist(y12))
dICC.SE.asympt(dist.mat, strata)


</code></pre>

<hr>
<h2 id='dICC.SE.bt'>Bootstrap Standard Error of Distance-based Intra-Class Correlation Coefficient</h2><span id='topic+dICC.SE.bt'></span>

<h3>Description</h3>

<p>Calculate the standard error of the distance-based intra-class correlation coefficient based on bootstrapping. Biological replicates are sampled by replacement. 
More conservative than the asymptotic approach.</p>


<h3>Usage</h3>

<pre><code class='language-R'> dICC.SE.bt(dist.mat, strata, B = 199)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dICC.SE.bt_+3A_dist.mat">dist.mat</code></td>
<td>
<p>a symmetrical distance matrix between all the replicates (technical and biological replicates).</p>
</td></tr>
<tr><td><code id="dICC.SE.bt_+3A_strata">strata</code></td>
<td>
<p>a factor with each level corresponding to a biological replicate. Supports an unequal number of technical replicates for each biological replicate.</p>
</td></tr>
<tr><td><code id="dICC.SE.bt_+3A_b">B</code></td>
<td>
<p>integer, the number of bootstrap iterations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Function <code>dICC</code> returns a list with the following component:
</p>
<table role = "presentation">
<tr><td><code>ICC</code></td>
<td>
<p>the distance-based ICC value.</p>
</td></tr>
<tr><td><code>SE</code></td>
<td>
<p>the standard error.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jun Chen and Xianyang Zhang</p>


<h3>References</h3>

<p>Chen, J. &amp; Zhang, X. 2022.  dICC: Distance-based Intraclass Correlation Coefficient for Metagenomic Reproducibility Studies.
submitted.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dICC">dICC</a></code>, <code><a href="#topic+dICC.SE.asympt">dICC.SE.asympt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Generate the error-free measurements of 20 biological replicates, each with four dimensions
y &lt;- matrix(rnorm(80), nrow = 20, ncol = 4)

# Generate two technical replicates for each biological replicate and add measurement errors
y1 &lt;- y + matrix(rnorm(80, sd = 0.5), nrow = 20, ncol = 4)
y2 &lt;- y + matrix(rnorm(80, sd = 0.5), nrow = 20, ncol = 4)
y12 &lt;- rbind(y1, y2)

# Create the design vector
strata &lt;- factor(rep(1 : 20, 2))

# Calculate the distance and distance-based ICC
dist.mat &lt;- as.matrix(dist(y12))
dICC.SE.bt(dist.mat, strata)


</code></pre>

<hr>
<h2 id='dmanova'>Distance-based Multivariate Analysis of Variance (Analytical P-value Calculation)</h2><span id='topic+dmanova'></span>

<h3>Description</h3>

<p>Analysis of variance using distance matrices &mdash; for
partitioning distance matrices among sources of variation and fitting
linear models (e.g., factors, polynomial regression) to distance 
matrices; calculate the analytical p-value based on pseudo-<code class="reqn">F</code> statistic 
without permutation.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dmanova(formula, data = NULL, positify = FALSE,
		contr.unordered = "contr.sum", contr.ordered = "contr.poly", 
		returnG = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dmanova_+3A_formula">formula</code></td>
<td>
<p>model formula. The LHS must be a dissimilarity matrix (either class <code>matrix</code> or class
<code>dist</code>, e.g., from  <code><a href="vegan.html#topic+vegdist">vegdist</a></code> or <code><a href="stats.html#topic+dist">dist</a></code>.  The RHS defines the independent 
variables. These can be continuous variables or factors, they can be transformed
within the formula, and they can have interactions as in a typical
<code><a href="stats.html#topic+formula">formula</a></code>. </p>
</td></tr>
<tr><td><code id="dmanova_+3A_data">data</code></td>
<td>
<p>the data frame for the independent variables.</p>
</td></tr>
<tr><td><code id="dmanova_+3A_positify">positify</code></td>
<td>
<p>a logical value indicating whether to make the Gower's matrix  positive definite
using the <code><a href="Matrix.html#topic+nearPD">nearPD</a></code> function in <code>Matrix</code> package. This is equivalent to modifying the distance 
matrix so that it has an Euclidean embedding. </p>
</td></tr>
<tr><td><code id="dmanova_+3A_contr.unordered">contr.unordered</code>, <code id="dmanova_+3A_contr.ordered">contr.ordered</code></td>
<td>
<p>contrasts used for the design
matrix (default in R is dummy or treatment contrasts for unordered
factors). </p>
</td></tr>
<tr><td><code id="dmanova_+3A_returng">returnG</code></td>
<td>
<p>a logical value indicating whether the Gower's matrix should be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>dmanova</code> is a permutation-free method for approximating the p-value from  distance-based permutational multivariate
analysis of variance (PERMANOVA). PERMANOVA is  slow when the sample size is large. In contrast,
<code>dmanova</code> provides an analytical solution, which is several orders of magnitude faster for large sample sizes. The covariate of interest
should be put as the last term in <code>formula</code> while the variables to be adjusted are put before the covariate of interest.
</p>


<h3>Value</h3>

<p>Function <code>dmanova</code> returns a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>aov.tab</code></td>
<td>
<p>typical AOV table showing sources of variation,
degrees of freedom,  sums of squares, mean squares,
<code class="reqn">F</code> statistics, partial <code class="reqn">R^2</code> and <code class="reqn">P</code>
values. </p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>degree of freedom for the Chisquared distribution.</p>
</td></tr> 
<tr><td><code>G</code></td>
<td>
<p> The Gower's matrix if <code>returnG</code> is true.</p>
</td></tr>   
<tr><td><code>call</code></td>
<td>
<p> the call made</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jun Chen and Xianyang Zhang</p>


<h3>References</h3>

<p>Chen, J. &amp; Zhang, X. 2021.  D-MANOVA: fast distance-based multivariate analysis of variance for large-scale microbiome association studies.
Bioinformatics. https://doi.org/10.1093/bioinformatics/btab498
</p>


<h3>See Also</h3>

<p><code><a href="#topic+adonis3">adonis3</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

groups &lt;- throat.meta$SmokingStatus

# Rarefaction
otu.tab.rff &lt;- Rarefy(throat.otu.tab)$otu.tab.rff

# Calculate the UniFrac distance
unifracs &lt;- GUniFrac(otu.tab.rff, throat.tree, alpha=c(0, 0.5, 1))$unifracs

# Test the smoking effect based on unweighted UniFrac distance, adjusting sex
# 'Sex' should be put before 'SmokingStatus'
dmanova(as.dist(unifracs[, , 'd_UW']) ~ Sex + SmokingStatus, data = throat.meta)

## End(Not run)

</code></pre>

<hr>
<h2 id='GMPR'>
Geometric Mean of Pairwise Ratios (GMPR) Normalization for Zero-inflated Count Data
</h2><span id='topic+GMPR'></span>

<h3>Description</h3>

<p>A robust normalization method for zero-inflated count data such as microbiome sequencing data. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GMPR(OTUmatrix, min_ct = 2, intersect_no = 4)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GMPR_+3A_otumatrix">OTUmatrix</code></td>
<td>

<p>An OTU count table, where OTUs are arranged in rows and samples in columns.
</p>
</td></tr>
<tr><td><code id="GMPR_+3A_min_ct">min_ct</code></td>
<td>

<p>The minimal number of OTU counts. Only those OTU pairs with at least <code>min_ct</code> counts are 
considered in the ratio calculation.   The default is 2.
</p>
</td></tr>
<tr><td><code id="GMPR_+3A_intersect_no">intersect_no</code></td>
<td>

<p>The minimal number of shared OTUs between samples. Only those sample pairs sharing at least <code>intersect_no</code> OTUs 
are considered in geometric mean calculation. The default is 4.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Normalization is a critical step in microbiome sequencing data analysis to account for variable library sizes.
Microbiome data contains a vast number of zeros, which makes the traditional RNA-Seq normalization methods unstable. 
The proposed  GMPR normalization remedies this problem by switching the two steps in DESeq2 normalization:
</p>
<p>First, to calculate rij, the median count ratio of nonzero counts between samples:
rij=median(cki/ckj) (k in 1:OTU_number and cki, ckj is the non-zero count of the kth OTU)
</p>
<p>Second, to calculate the size factor si for a given sample i:
si=geometric_mean(rij)
</p>


<h3>Value</h3>

<p>A vector of GMPR size factor for each sample.
</p>


<h3>Author(s)</h3>

<p>Jun Chen and Lujun Zhang
</p>


<h3>References</h3>

<p>Li Chen, James Reeve, Lujun Zhang, Shenbing Huang, and Jun Chen. 2018. GMPR: A robust normalization method for zero-inflated count
data with application to microbiome sequencing data. PeerJ, 6, e4600.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.otu.tab)
size.factor &lt;- GMPR(t(throat.otu.tab))
</code></pre>

<hr>
<h2 id='GUniFrac'>
Generalized UniFrac distances for comparing microbial communities.
</h2><span id='topic+GUniFrac'></span>

<h3>Description</h3>

<p>A generalized version of commonly used UniFrac distances. It is defined as:
</p>
<p style="text-align: center;"><code class="reqn">d^{(\alpha)} = \frac{\sum_{i=1}^m b_i (p^A_{i} + p^B_{i})^\alpha 
\left\vert \frac{ p^A_{i} - p^B_{i} }{p^A_{i} + p^B_{i}} \right\vert } { 
 \sum_{i=1}^m b_i (p^A_{i} + p^B_{i})^\alpha},</code>
</p>

<p>where <code class="reqn">m</code> is the number of branches, <code class="reqn">b_i</code> is the length of 
<code class="reqn">i</code>th branch, <code class="reqn">p^A_{i}, p^B_{i}</code> are the branch
proportion for community A and B. 
</p>
<p>Generalized UniFrac distance contains an extra parameter <code class="reqn">\alpha</code> 
controlling the weight on abundant lineages so the distance is not dominated 
by highly abundant lineages. <code class="reqn">\alpha=0.5</code> (&quot;d_0.5&quot;) is overall very robust. 
</p>
<p>The unweighted (&quot;d_1&quot;) and weighted UniFrac (&quot;d_UW&quot;) are also implemented.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GUniFrac(otu.tab, tree, size.factor = NULL, alpha = c(0, 0.5, 1), verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GUniFrac_+3A_otu.tab">otu.tab</code></td>
<td>

<p>a matrix, the OTU count table, row - n sample, column - q OTU
</p>
</td></tr>
<tr><td><code id="GUniFrac_+3A_tree">tree</code></td>
<td>

<p>a rooted phylogenetic tree of R class &ldquo;phylo&rdquo;
</p>
</td></tr>
<tr><td><code id="GUniFrac_+3A_size.factor">size.factor</code></td>
<td>

<p>a numeric vector of the normalizing factors to divide the counts. The length
is the number of samples. This provides the flexibility to normalize data using the 
preferred normalization method (e.g. GMPR normalizing factor). If not supplied, the total sum will be used. 
</p>
</td></tr>
<tr><td><code id="GUniFrac_+3A_alpha">alpha</code></td>
<td>

<p>a numeric vector, parameters controlling the weight on abundant lineages
</p>
</td></tr>
<tr><td><code id="GUniFrac_+3A_verbose">verbose</code></td>
<td>

<p>logical value, whether to print out the messages
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a list containing
</p>
<table role = "presentation">
<tr><td><code>unifracs</code></td>
<td>
<p>a three dimensional array containing all the UniFrac
distance matrices</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The function only accepts rooted tree. To root a tree, you may consider using <code>midpoint</code> from the package <code>phangorn</code>.
</p>


<h3>Author(s)</h3>

<p>Jun Chen &lt;chen.jun2@mayo.edu&gt;
</p>


<h3>References</h3>

<p>Chen, J., Bittinger, K., Charlson, E.S., Hoffmann, C., Lewis, J., Wu, G.D., Collman, R.G., Bushman, F.D. and Li, H. (2012). Associating microbiome composition with 
environmental covariates using generalized UniFrac distances. 28(16): 2106–2113.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Rarefy">Rarefy</a></code>, <code><a href="#topic+PermanovaG">PermanovaG</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

groups &lt;- throat.meta$SmokingStatus

# Rarefaction
otu.tab.rff &lt;- Rarefy(throat.otu.tab)$otu.tab.rff

# Calculate the UniFracs
unifracs &lt;- GUniFrac(otu.tab.rff, throat.tree, alpha=c(0, 0.5, 1))$unifracs

dw &lt;- unifracs[, , "d_1"]		# Weighted UniFrac
du &lt;- unifracs[, , "d_UW"]		# Unweighted UniFrac	
d0 &lt;- unifracs[, , "d_0"]     	# GUniFrac with alpha 0  
d5 &lt;- unifracs[, , "d_0.5"]   	# GUniFrac with alpha 0.5 

# Permanova - Distance based multivariate analysis of variance
adonis3(as.dist(d5) ~ groups)

## End(Not run)

</code></pre>

<hr>
<h2 id='PermanovaG'>
Permutational Multivariate Analysis of Variance Using Multiple Distance Matrices
</h2><span id='topic+PermanovaG'></span>

<h3>Description</h3>

<p>In practice, we do not know a priori which type of change happens in 
the microbiome. Each distance measure is most powerful in detecting
only a certain scenario. When multiple distance matrices are available,
separate tests using each distance matrix will lead to loss of power 
due to multiple testing correction. Combing the distance matrices
in a single test will improve power. PermanovaG combines multiple distance
matrices  by taking the minimum of the P values for individual distance
matrices. Significance is assessed by permutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PermanovaG(formula, data = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PermanovaG_+3A_formula">formula</code></td>
<td>

<p>a formula, left side of the formula (<code>Y ~ X</code>) is a three dimensional ARRAY
containing the supplied distance matrices as produced by <code><a href="#topic+GUniFrac">GUniFrac</a></code>
function. Or it could be a list of distance matrices.
</p>
</td></tr>
<tr><td><code id="PermanovaG_+3A_data">data</code></td>
<td>

<p>a data frame containing the covariates
</p>
</td></tr>
<tr><td><code id="PermanovaG_+3A_...">...</code></td>
<td>

<p>parameter passing to <code>adonis</code> function
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a list containing:
</p>
<table role = "presentation">
<tr><td><code>p.tab</code></td>
<td>
<p>a data frame, columns: p-values for individual distance matrices and the omnibus test, 
rows: covariates. (Note: they are sequential p-values, put the variable of interest in the end)</p>
</td></tr>
<tr><td><code>aov.tab.list</code></td>
<td>
<p>a list of <code>adonis</code> AOV tables for individual distance matrices</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jun Chen &lt;chen.jun2@mayo.edu&gt;
</p>


<h3>References</h3>

<p>Chen, J., Bittinger, K., Charlson, E.S., Hoffmann, C., Lewis, J., Wu, G.D., Collman, R.G., Bushman, F.D. and Li, H.(2012). Associating microbiome composition with environmental covariates using generalized UniFrac distances. 28(16): 2106–2113.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Rarefy">Rarefy</a></code>, <code><a href="#topic+GUniFrac">GUniFrac</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

groups &lt;- throat.meta$SmokingStatus

# Rarefaction
otu.tab.rff &lt;- Rarefy(throat.otu.tab)$otu.tab.rff

# Calculate the UniFracs
unifracs &lt;- GUniFrac(otu.tab.rff, throat.tree, alpha=c(0, 0.5, 1))$unifracs

#	Combine unweighted and weighted UniFrac for testing
PermanovaG(unifracs[, , c("d_1", "d_UW")] ~ groups)
# Combine d(0), d(0.5), d(1) for testing

## End(Not run)
</code></pre>

<hr>
<h2 id='PermanovaG2'>
Permutational Multivariate Analysis of Variance Using Multiple Distance Matrices(Freedman-Lane Permutation)
</h2><span id='topic+PermanovaG2'></span>

<h3>Description</h3>

<p>In practice, we do not know a priori which type of change happens in 
the microbiome. Each distance measure is most powerful in detecting
only a certain scenario. When multiple distance matrices are available,
separate tests using each distance matrix will lead to loss of power 
due to multiple testing correction. Combing the distance matrices
in a single test will improve power. PermanovaG combines multiple distance
matrices  by taking the minimum of the P values for individual distance
matrices. Significance is assessed by permutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PermanovaG2(formula, data = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PermanovaG2_+3A_formula">formula</code></td>
<td>

<p>a formula, left side of the formula (<code>Y ~ X</code>) is a three dimensional ARRAY
containing the supplied distance matrices as produced by <code><a href="#topic+GUniFrac">GUniFrac</a></code>
function. Or it could be a list of distance matrices.
</p>
</td></tr>
<tr><td><code id="PermanovaG2_+3A_data">data</code></td>
<td>

<p>a data frame containing the covariates
</p>
</td></tr>
<tr><td><code id="PermanovaG2_+3A_...">...</code></td>
<td>

<p>parameters passing to <code>adonis</code> function
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a list containing:
</p>
<table role = "presentation">
<tr><td><code>p.tab</code></td>
<td>
<p>a data frame, columns: p-values for individual distance matrices and the omnibus test, 
rows: covariates. (Note: they are sequential p-values, put the variable of interest in the end)</p>
</td></tr>
<tr><td><code>aov.tab.list</code></td>
<td>
<p>a list of <code>adonis</code> AOV tables for individual distance matrices</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jun Chen &lt;chen.jun2@mayo.edu&gt;
</p>


<h3>References</h3>

<p>Chen, J., Bittinger, K., Charlson, E.S., Hoffmann, C., Lewis, J., Wu, G.D., Collman, R.G., Bushman, F.D. and Li, H. (2012). Associating microbiome composition with 
environmental covariates using generalized UniFrac distances. 28(16): 2106–2113.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Rarefy">Rarefy</a></code>, <code><a href="#topic+GUniFrac">GUniFrac</a></code>, <code><a href="#topic+adonis3">adonis3</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

groups &lt;- throat.meta$SmokingStatus

# Rarefaction
otu.tab.rff &lt;- Rarefy(throat.otu.tab)$otu.tab.rff

# Calculate the UniFracs
unifracs &lt;- GUniFrac(otu.tab.rff, throat.tree, alpha=c(0, 0.5, 1))$unifracs

#	Combine unweighted and weighted UniFrac for testing
PermanovaG2(unifracs[, , c("d_1", "d_UW")] ~ groups)

## End(Not run)
</code></pre>

<hr>
<h2 id='Rarefy'>
Rarefy a Count Table to Equal Sequencing Depth
</h2><span id='topic+Rarefy'></span>

<h3>Description</h3>

<p>GUniFrac is also sensitive to different sequencing depth. To compare
microbiomes on an equal basis, rarefaction might be used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Rarefy(otu.tab, depth = min(rowSums(otu.tab)))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Rarefy_+3A_otu.tab">otu.tab</code></td>
<td>

<p>OTU count table, row - n sample, column - q OTU
</p>
</td></tr>
<tr><td><code id="Rarefy_+3A_depth">depth</code></td>
<td>

<p>required sequencing depth; If not specified, the lowest sequencing
depth is used.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a list containing:
</p>
<table role = "presentation">
<tr><td><code>otu.tab.rff</code></td>
<td>
<p>rarefied OTU table</p>
</td></tr>
<tr><td><code>discard</code></td>
<td>
<p>IDs of samples that does not reach the specified sequencing depth</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jun Chen &lt;chen.jun2@mayo.edu&gt;
</p>


<h3>References</h3>

<p>Chen, J., Bittinger, K., Charlson, E.S., Hoffmann, C., Lewis, J., Wu, G.D., Collman, R.G., Bushman, F.D. and Li, H. (2012). Associating microbiome composition with 
environmental covariates using generalized UniFrac distances. 28(16): 2106–2113.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.otu.tab)
# Rarefaction
otu.tab.rff &lt;- Rarefy(throat.otu.tab, 1024)$otu.tab.rff
</code></pre>

<hr>
<h2 id='SimulateMSeq'>A Semiparametric Model-based Microbiome Sequencing Data Simulator for Cross-sectional and Case-control Studies</h2><span id='topic+SimulateMSeq'></span>

<h3>Description</h3>

<p>The function generates synthetic microbiome sequencing data for studying the performance of differential abundance analysis methods.  
It uses a user-supplied (large) reference OTU table as a template to generate a synthetic OTU table of specified size. A subset of OTUs  are affected
by a simulated covariate of interest, either binary or continuous.  Confounder effects can also be simulated.  The function allows simulating
different signal structures, i.e., the percentage of differential OTUs, their effect sizes, their direction of change,  and whether these OTUs are
relatively abundant or rare.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SimulateMSeq(
  ref.otu.tab,
  nSam = 100,
  nOTU = 500,
  diff.otu.pct = 0.1,
  diff.otu.direct = c("balanced", "unbalanced"),
  diff.otu.mode = c("abundant", "rare", "mix"),
  covariate.type = c("binary", "continuous"),
  grp.ratio = 1,
  covariate.eff.mean = 1,
  covariate.eff.sd = 0,
  confounder.type = c("none", "binary", "continuous", "both"),
  conf.cov.cor = 0.6,
  conf.diff.otu.pct = 0,
  conf.nondiff.otu.pct = 0.1,
  confounder.eff.mean = 0,
  confounder.eff.sd = 0,
  error.sd = 0,
  depth.mu = 10000,
  depth.theta = 5,
  depth.conf.factor = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SimulateMSeq_+3A_ref.otu.tab">ref.otu.tab</code></td>
<td>
<p>a matrix, the reference OTU count table (row - OTUs, column - samples), serving as the template for synthetic sample generation.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_nsam">nSam</code></td>
<td>
<p>the number of samples to be simulated.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_notu">nOTU</code></td>
<td>
<p>the number of OTUs to be simulated. </p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_diff.otu.pct">diff.otu.pct</code></td>
<td>
<p>a numeric value between 0 and 1, the percentage of differential OTUs to be simulated. If 0, global null setting is simulated. The default is 0.1.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_diff.otu.direct">diff.otu.direct</code></td>
<td>
<p>a character string of &quot;balanced&quot; or &quot;unbalanced&quot;.  &quot;balanced&quot; - the direction of change for these differential OTUs is random, &quot;unbalanced&quot; - direction of change is the same. The default is &quot;balanced&quot;.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_diff.otu.mode">diff.otu.mode</code></td>
<td>
<p>a character string of  &quot;rare&quot;, &quot;mix&quot; or &quot;abundant&quot;.  &quot;abundant&quot; - differential OTU come from the top quartile of the abundance distribution, 
&quot;rare&quot; - differential OTU come from the bottom quartile of the abundance distribution, and &quot;mix&quot; - random set. The default is &quot;abundant&quot;. </p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_covariate.type">covariate.type</code></td>
<td>
<p>a character string of &quot;binary&quot; or &quot;continuous&quot;, indicating the type of the covariate to be simulated. The default is &quot;binary&quot; (e.g., case v.s. control).</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_grp.ratio">grp.ratio</code></td>
<td>
<p>a numeric value between 0 and 1. Group size ratio. The default is 1, i.e., equal group size. Only relevant when <code>covariate.type</code> is &quot;binary&quot;. </p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_covariate.eff.mean">covariate.eff.mean</code></td>
<td>
<p>a numeric value, the mean log fold change (effect size) in response to one unit change of the covariate.  The default is 1.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_covariate.eff.sd">covariate.eff.sd</code></td>
<td>
<p>a positive numeric value, the standard deviation of the log fold change.  The default is 0, i.e., the log fold change is the same across differential OTUs.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_confounder.type">confounder.type</code></td>
<td>
<p>a character string of &quot;none&quot;, &quot;binary&quot;, &quot;continuous&quot; or &quot;both&quot;. The default is &quot;none&quot;, no confounder will be simulated.
If &quot;both&quot;, both a binary and continuous confounder will be simulated. The default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_conf.cov.cor">conf.cov.cor</code></td>
<td>
<p>a numeric value between 0 and 1.  The correlation between the covariate of interest and the confounder. The default is 0.6.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_conf.diff.otu.pct">conf.diff.otu.pct</code></td>
<td>
<p>a numeric value between 0 and 1. The percentage of OTUs affected by the confounder and the covariate of interest. The default is 0.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_conf.nondiff.otu.pct">conf.nondiff.otu.pct</code></td>
<td>
<p>a numeric value between 0 and 1. The percentage of OTUs affected by the confounder but not the covariate of interest. The default is 0.1.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_confounder.eff.mean">confounder.eff.mean</code></td>
<td>
<p>a numeric value, the mean log fold change (effect size) in response to one unit change of the confounder.  The default is 1.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_confounder.eff.sd">confounder.eff.sd</code></td>
<td>
<p>a positive numeric value, the standard deviation of the log fold change for the confounder.  The default is 0, i.e., the log fold change is the same across OTUs affected by the confounder.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_error.sd">error.sd</code></td>
<td>
<p>the sd of the log fold change unexplained by the covariate and the confounder (i.e., the error term under the log linear model).  The default is 0. </p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_depth.mu">depth.mu</code></td>
<td>
<p>the mean sequencing depth to be simulated. The default is 10,000.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_depth.theta">depth.theta</code></td>
<td>
<p>the theta value  of the negative binomial distribution controlling the variance (<code>mu + mu^2/theta</code>). The default is 5.</p>
</td></tr>
<tr><td><code id="SimulateMSeq_+3A_depth.conf.factor">depth.conf.factor</code></td>
<td>
<p>a numeric value controlling the dependence of the sequencing depth on the covariate of interest (<code>depth.mu</code> * <code>exp(scale(X)</code> * <code>depth.conf.factor</code>)). 
The default is 0, i.e., the depth is not associated with the covariate of interest. This parameter can be used to simulate depth confounding.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements a semiparametric approach for realistic independent microbiome sequencing data generation. The method draws random samples from a large reference dataset (non-parametric part) and uses
these reference samples as templates to generate new samples (parametric part). Specifically, for each drawn reference sample, it infers the underlying composition based on a Bayesian model and 
then adds covariate/confounder effects to the composition vector, based on which a new sequencing sample is generated. The method circumvents the difficulty in modeling the inter-subject variation
of the microbiome composition.   
</p>


<h3>Value</h3>

<p>Return a list with the elements:
</p>
<table role = "presentation">
<tr><td><code>otu.tab.sim</code></td>
<td>
<p>simulated OTU table</p>
</td></tr>
<tr><td><code>covariate</code></td>
<td>
<p>simulated covariate of interest</p>
</td></tr>
<tr><td><code>confounder</code></td>
<td>
<p>simulated confounder(s)</p>
</td></tr>
<tr><td><code>diff.otu.ind</code></td>
<td>
<p>indices of the differential OTUs, i.e., affected by the covariate of interest</p>
</td></tr>
<tr><td><code>otu.names</code></td>
<td>
<p>the names of the simulated OTUs</p>
</td></tr>
<tr><td><code>conf.otu.ind</code></td>
<td>
<p>indices of OTUs affected by the confounder(s)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jun Chen and Lu Yang</p>


<h3>References</h3>

<p>Yang, L. &amp; Chen, J. 2022. A comprehensive evaluation of differential abundance analysis methods: current status and potential solutions.
Microbiome. In Press.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Use throat microbiome for illustration
data(throat.otu.tab)
comm &lt;- t(throat.otu.tab)
comm &lt;- comm[rowMeans(comm != 0) &gt; 0.2, ]

# Simulate binary covariate, 10% signal density, abundant differential OTUs, unbalanced change 
# This setting simulates strong compositional effects
sim.obj &lt;- SimulateMSeq(
		ref.otu.tab = comm, nSam = 50, nOTU = 50,
		# True signal setting
		diff.otu.pct = 0.1, diff.otu.direct = c("unbalanced"), 
		diff.otu.mode = c("abundant"),
		covariate.type = c("binary"), grp.ratio = 1,
		covariate.eff.mean = 1.0, covariate.eff.sd = 0,
		# Confounder signal setting
		confounder.type = c("both"), conf.cov.cor = 0.6,
		conf.diff.otu.pct = 0.1, conf.nondiff.otu.pct = 0.1,
		confounder.eff.mean = 1.0, confounder.eff.sd = 0,
		# Depth setting
		depth.mu = 10000, depth.theta = 5, depth.conf.factor = 0
)

meta.dat &lt;- data.frame(X = sim.obj$covariate, Z1 = sim.obj$confounder[, 1],
                       Z2 = sim.obj$confounder[, 2])
otu.tab.sim &lt;- sim.obj$otu.tab.sim

# Run ZicoSeq for differential abundance analysis
zico.obj &lt;- ZicoSeq(meta.dat = meta.dat, feature.dat = otu.tab.sim, 
		grp.name = 'X', adj.name = c('Z1', 'Z2'), feature.dat.type = "count",
		# Filter to remove rare taxa
		prev.filter = 0.2, mean.abund.filter = 0,  max.abund.filter = 0.002, min.prop = 0, 
		# Winsorization to replace outliers
		is.winsor = TRUE, outlier.pct = 0.03, winsor.end = 'top',
		# Posterior sampling to impute zeros
		is.post.sample = TRUE, post.sample.no = 25, 
		# Multiple link functions to capture diverse taxon-covariate relation
		link.func = list(function (x) x^0.25, function (x) x^0.5, function (x) x^0.75), 
		stats.combine.func = max,
		# Permutation-based multiple testing correction
		perm.no = 99,  strata = NULL, 
		# Reference-based multiple stage normalization
		ref.pct = 0.5, stage.no = 6, excl.pct = 0.2,
		# Family-wise error rate control
		is.fwer = FALSE,
		verbose = TRUE, return.feature.dat = FALSE)

# Detected differential OTUs
which(zico.obj$p.adj.fdr &lt;= 0.05)

# True differential OTUs
sim.obj$otu.names[sim.obj$diff.otu.ind]
</code></pre>

<hr>
<h2 id='SimulateMSeqC'>A Semiparametric Model-based Microbiome Sequencing Data Simulator for Longitudinal, Matched-pair, and Replicate Sampling Designs</h2><span id='topic+SimulateMSeqC'></span>

<h3>Description</h3>

<p>The function generates synthetic microbiome sequencing data for studying the performance of differential abundance analysis methods for correlated microbiome data 
generated in longitudinal, matched-pair and replicate sampling study designs. 
It uses a user-supplied (large) reference OTU table as a template to generate a synthetic OTU table of specified size. A subset of OTUs can be affected by
by a binary variable (group effect) and/or a time variable (temporal effect). Time X group interaction and confounder effects can also be simulated. 
The function allows simulating different signal structures, i.e., the percentage of differential OTUs, their effect sizes, and their direction of change.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SimulateMSeqC(	
  ref.otu.tab,
  nSubject = 40,
  nOTU = 50,
  nTime = 2, 
  error.sd = 1, 
  MgX = 0.5, 
  SgX = 0, 
  X.diff.otu.pct = 0.1,
  grp.ratio = 1, 
  balanced.X = TRUE,
  MgT = 0, 
  SgT = 0, 
  SbT = 0, 
  T.diff.otu.pct = 0, 
  balanced.T = TRUE,
  MgXT = 0, 
  SgXT = 0, 
  XT.diff.otu.pct = 0, 
  balanced.XT = TRUE,
  conf.cov.cor = 0.6,
  confounder = c('X', 'T'),
  MgZ = 0.5,
  SgZ = 0, 
  Z.diff.otu.pct = 0.05,
  Z.nondiff.otu.pct = 0.1, 
  depth.mu = 10000,
  depth.theta = 5,
  depth.conf.factor = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SimulateMSeqC_+3A_ref.otu.tab">ref.otu.tab</code></td>
<td>
<p>a matrix, the reference OTU count table (row - OTUs, column - samples), serving as the template for synthetic sample generation.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_nsubject">nSubject</code></td>
<td>
<p>the number of subjects to be simulated.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_notu">nOTU</code></td>
<td>
<p>the number of OTUs to be simulated. </p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_ntime">nTime</code></td>
<td>
<p>the number of time points to be simulated. </p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_error.sd">error.sd</code></td>
<td>
<p>the standar deviation of the random error controlling the within-subject correlation strength. Large k = 1, small k = 4.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_mgx">MgX</code></td>
<td>
<p>a numeric value indicating the mean group (X) effect (log fold change ) across the associated OTUs. The default is 0.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_sgx">SgX</code></td>
<td>
<p>a numeric value indicating the variance of group (X) effect (log fold change) across the associated OTUs. The default is 0.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_x.diff.otu.pct">X.diff.otu.pct</code></td>
<td>
<p>a numeric value between 0 and 1, the percentage of differential OTUs regarding the group (X) to be simulated.  The default is 0.1.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_grp.ratio">grp.ratio</code></td>
<td>
<p>a numeric value between 0 and 1. Group size ratio. The default is 1, i.e., equal group size. </p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_balanced.x">balanced.X</code></td>
<td>
<p>a logical value.  TRUE - the direction of change for these group differential OTUs is random, FALSE - the direction of change is the same. The default is &quot;balanced&quot;.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_mgt">MgT</code></td>
<td>
<p>a numeric value indicating the population mean of the time (T) effect (log fold change) across the associated OTUs. The default is 0.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_sgt">SgT</code></td>
<td>
<p>a numeric value indicating the population variance of the time (T) effect (log fold change) across the associated OTUs. The default is 0.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_sbt">SbT</code></td>
<td>
<p>a numeric value indicating the variance of time (T) effect (log fold change) across the subjects. This parameter is to generate a subject-level random slope 
(temporal trends differ by subjects). The default is 0. </p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_t.diff.otu.pct">T.diff.otu.pct</code></td>
<td>
<p>a numeric value between 0 and 1, the percentage of time differential OTUs  to be simulated.  The default is 0.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_balanced.t">balanced.T</code></td>
<td>
<p>a logical value.  TRUE - the direction of change for these time differential OTUs is random, FALSE - the direction of change is the same. The default is &quot;balanced&quot;.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_mgxt">MgXT</code></td>
<td>
<p>a numeric value indicating the mean X:T interaction effect (log fold change) across the associated OTUs. The default is 0.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_sgxt">SgXT</code></td>
<td>
<p>a numeric value indicating the variance of X:T interaction effect (log fold change) across the associated OTUs. The default is 0.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_xt.diff.otu.pct">XT.diff.otu.pct</code></td>
<td>
<p>a numeric value between 0 and 1, the percentage of X:T interaction OTUs to be simulated.  The default is 0.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_balanced.xt">balanced.XT</code></td>
<td>
<p>a logical value.  TRUE - the direction of change for the interaction effects is random, FALSE - the direction of change is the same. The default is &quot;balanced&quot;.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_conf.cov.cor">conf.cov.cor</code></td>
<td>
<p>a numeric value between 0 and 1.  The correlation strength between the group and the confounder. The default is 0.6.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_confounder">confounder</code></td>
<td>
<p>one of 'X' or 'T', whether the confounder is correlated with 'X' or 'T'.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_mgz">MgZ</code></td>
<td>
<p>a numeric value indicating the mean confounder (Z) effect (log fold change) across the associated OTUs. The default is 0.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_sgz">SgZ</code></td>
<td>
<p>a numeric value indicating the variance of confounder (Z) effect (log fold change) across the associated OTUs. The default is 0.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_z.diff.otu.pct">Z.diff.otu.pct</code></td>
<td>
<p>a numeric value between 0 and 1, less than the percentage of group differential OTUs,
the percentage of confounder-associated OTUs, which are also group differential.  The default is 0.05.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_z.nondiff.otu.pct">Z.nondiff.otu.pct</code></td>
<td>
<p>a numeric value between 0 and 1, less than the percentage of group non-differential OTUs,
the percentage of confounder-associated OTUs, which are not group differential.  The default is 0.1.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_depth.mu">depth.mu</code></td>
<td>
<p>the mean sequencing depth to be simulated. The default is 10,000.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_depth.theta">depth.theta</code></td>
<td>
<p>the theta value  of the negative binomial distribution controlling the variance (<code>mu + mu^2/theta</code>). The default is 5.</p>
</td></tr>
<tr><td><code id="SimulateMSeqC_+3A_depth.conf.factor">depth.conf.factor</code></td>
<td>
<p>a numeric value controlling the dependence of the sequencing depth on the group variable (<code>depth.mu</code> * <code>exp(scale(X)</code> * <code>depth.conf.factor</code>)). 
The default is 0, i.e., the depth is not different between groups. This parameter can be used to simulate depth confounding.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements a semiparametric approach for realistic correlated microbiome data generation. The method draws random samples from a large reference dataset (non-parametric part) and uses
these reference samples as templates to generate new samples (parametric part). Specifically, for each drawn reference sample, it infers the underlying composition based on a Bayesian model and 
then adds group/time/group:time/confounder effects to the composition vector, based on which a new sequencing sample is generated. The method circumvents the difficulty in modeling the inter-subject variation
of the microbiome composition.   
</p>


<h3>Value</h3>

<p>Return a list with the elements:
</p>
<table role = "presentation">
<tr><td><code>otu.tab.sim</code></td>
<td>
<p>simulated OTU table</p>
</td></tr>
<tr><td><code>meta</code></td>
<td>
<p>meta data containing the simulated covariates (group, time, confounder)</p>
</td></tr>
<tr><td><code>otu.names</code></td>
<td>
<p>the names of the simulated OTUs</p>
</td></tr>
<tr><td><code>X.diff.otu.ind</code></td>
<td>
<p>indices of the group differential OTUs</p>
</td></tr>
<tr><td><code>T.diff.otu.ind</code></td>
<td>
<p>indices of the time differential OTUs</p>
</td></tr>
<tr><td><code>XT.diff.otu.ind</code></td>
<td>
<p>indices of the OTUs with group:time interaction</p>
</td></tr>
<tr><td><code>Z.diff.otu.ind</code></td>
<td>
<p>indices of OTUs affected by the confounder</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lu Yang and Jun Chen</p>


<h3>References</h3>

<p>Yang, L. &amp; Chen, J. 2022. Benchmarking Differential Abundance Analysis Methods for Correlated Microbiome Sequencing Data.
Submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Use throat microbiome for illustration
data(throat.otu.tab)
comm &lt;- t(throat.otu.tab)
comm &lt;- comm[rowMeans(comm != 0) &gt; 0.2, ]

# Example1: Simulate replicate sampling data, 40 subjects each with two replicates (nTime =2), 
# two group comparison, 10% group differential OTUs
## Not run: 
sim.obj &lt;- SimulateMSeqC(ref.otu.tab= comm, 
                         nSubject = 40, nOTU = 50, nTime = 2,
                         # Within-subject correlation setting
                         error.sd = 1, 
                         # Group effect setting, unbalanced
                         MgX = 0.5, SgX = 0, X.diff.otu.pct = 0.1, grp.ratio = 1, 
                         balanced.X = FALSE,
                         # Time effect setting (No time effect)
                         MgT = 0, SgT = 0, SbT = 0, T.diff.otu.pct = 0,
                         # Interaction effect setting (No interaction effect)
                         MgXT = 0, SgXT = 0, XT.diff.otu.pct = 0, 
                         # Confounder effect setting
                         conf.cov.cor = 0.6, confounder = 'X',
                         MgZ = 0.5, SgZ = 0,  Z.diff.otu.pct = 0.05, Z.nondiff.otu.pct = 0.1, 
                         # Sequencing Depth setting
                         depth.mu = 10000, depth.theta = 5,  depth.conf.factor = 0)

## End(Not run)
# Example2: Simulate matched-pair data, 100 subjects each with pre- and post-treatment (nTime = 2), 
# 10% differential OTUs
## Not run: 
sim.obj &lt;- SimulateMSeqC(ref.otu.tab= comm, 
                         nSubject = 100, nOTU = 50, nTime = 2,
                         # Within-subject correlation setting
                         error.sd = 1, 
                         # Group effect setting (No group effect)
                         MgX = 0, SgX = 0, X.diff.otu.pct = 0, grp.ratio = 1, 
                         # Time effect setting (No random slope, SbT=0)
                         MgT = 0.5, SgT = 0, SbT = 0, T.diff.otu.pct = 0.1, 
                         # Interaction effect setting (No interaction effect)
                         MgXT = 0, SgXT = 0, XT.diff.otu.pct = 0,
                         # Confounder effect setting (T!)
                         conf.cov.cor = 0.6, confounder = 'T',
                         MgZ = 0, SgZ = 0,  Z.diff.otu.pct = 0.05, Z.nondiff.otu.pct = 0.1, 
                         # Sequencing Depth setting
                         depth.mu = 10000, depth.theta = 5,  depth.conf.factor = 0)

## End(Not run)

# Example3: Simulate the general longitudinal data, 40 Subjects each with five time points, 
# two groups, 10% group differential OTUs, 10 % time differential OTUs and 10 % interaction OTUs.
## Not run: 
sim.obj &lt;- SimulateMSeqC(ref.otu.tab= comm, 
                         nSubject = 40, nOTU = 50, nTime = 5,
                         # Within-subject correlation setting
                         error.sd = 1, 
                         # Group effect setting, balanced
                         MgX = 0.5, SgX = 0, X.diff.otu.pct = 0.1, grp.ratio = 1, 
                         balanced.X = TRUE,
                         # Time effect setting (random slope)
                         MgT = 0.5, SgT = 0, SbT = 0.5, T.diff.otu.pct = 0.1, 
                         # Interaction effect setting 
                         MgXT = 0.5, SgXT = 0, XT.diff.otu.pct = 0.1, 
                         # Confounder effect setting
                         conf.cov.cor = 0.6, confounder = 'X',
                         MgZ = 0.5, SgZ = 0,  Z.diff.otu.pct = 0.05, Z.nondiff.otu.pct = 0.1, 
                         # Depth setting
                         depth.mu = 10000, depth.theta = 5,  depth.conf.factor = 0)

## End(Not run)
</code></pre>

<hr>
<h2 id='stool.otu.tab'>
Stool Microbiome OTU Count Table
</h2><span id='topic+stool.otu.tab'></span>

<h3>Description</h3>

<p>OTU count table from 16S V3-V5 targeted sequencing of the stool microbiome samples from the HMP project. A total of 2,094 OTUs from 295 samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(stool.otu.tab)</code></pre>


<h3>Format</h3>

<p>The format is:
chr &quot;stool.otu.tab&quot;
</p>


<h3>Details</h3>

<p>The OTU table was taken from R bioconductor &quot;HMP16SData&quot; package. OTUs with prevalence less than 10% and maximum proportion less than 0.2% were removed. 
This OTU table can be used for simulating stool microbiome sequencing data.
</p>


<h3>Source</h3>

<p>Schiffer L, Azhar R, Shepherd L, Ramos M, Geistlinger L, Huttenhower C, Dowd JB, Segata N, Waldron L (2019). 
“HMP16SData: Efficient Access to the Human Microbiome Project through Bioconductor.” American Journal of Epidemiology. doi: 10.1093/aje/kwz006.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(stool.otu.tab)
</code></pre>

<hr>
<h2 id='throat.meta'>
Throat Microbiome Meta Data
</h2><span id='topic+throat.meta'></span>

<h3>Description</h3>

<p>It is part of a microbiome data set for studying the effect of
smoking on  the upper respiratory tract microbiome. The original data 
set contains samples from both throat and  nose microbiomes, and 
from both body sides. This data set comes from the throat microbiome 
of left body side. It contains 60 subjects consisting of 32 nonsmokers
and 28 smokers. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(throat.meta)</code></pre>


<h3>Source</h3>

<p>Charlson ES, Chen J, Custers-Allen R, Bittinger K, Li H, et al. (2010) 
Disordered Microbial Communities in the Upper Respiratory Tract of 
Cigarette Smokers. PLoS ONE 5(12): e15216. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.meta)

</code></pre>

<hr>
<h2 id='throat.otu.tab'>
Throat Microbiome OTU Count Table
</h2><span id='topic+throat.otu.tab'></span>

<h3>Description</h3>

<p>It is part of a microbiome data set (16S V12-targeted 454 pyrosequencing) for studying the effect of
smoking on  the upper respiratory tract microbiome. The original data 
set contains samples from both throat and  nose microbiomes, and 
from both body sides. This data set comes from the throat microbiome 
of left body side. It contains 60 subjects consisting of 32 nonsmokers
and 28 smokers. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(throat.otu.tab)</code></pre>


<h3>Details</h3>

<p>The OTU table is produced by the QIIME software. Singleton OTUs have
been discarded.
</p>


<h3>Source</h3>

<p>Charlson ES, Chen J, Custers-Allen R, Bittinger K, Li H, et al. (2010) 
Disordered Microbial Communities in the Upper Respiratory Tract of 
Cigarette Smokers. PLoS ONE 5(12): e15216. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.otu.tab)
</code></pre>

<hr>
<h2 id='throat.tree'>
UPGMA Tree of Throat Microbiome OTUs
</h2><span id='topic+throat.tree'></span>

<h3>Description</h3>

<p>The OTU tree is constructed using UPGMA on the K80 distance matrice of
the OTUs. It is a rooted tree of class &quot;phylo&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(throat.tree)</code></pre>


<h3>Details</h3>

<p>The OTUs are produced by the QIIME software. Singleton OTUs have
been discarded.
</p>


<h3>Source</h3>

<p>Charlson ES, Chen J, Custers-Allen R, Bittinger K, Li H, et al. (2010) 
Disordered Microbial Communities in the Upper Respiratory Tract of 
Cigarette Smokers. PLoS ONE 5(12): e15216. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.tree)
</code></pre>

<hr>
<h2 id='vaginal.otu.tab'>
Vaginal Microbiome OTU Count Table
</h2><span id='topic+vaginal.otu.tab'></span>

<h3>Description</h3>

<p>OTU count table from 16S V3-V5 targeted sequencing of the vaginal microbiome samples from the HMP project. A total of 780 OTUs from 381 samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(vaginal.otu.tab)</code></pre>


<h3>Details</h3>

<p>The OTU table was taken from R bioconductor &quot;HMP16SData&quot; package. OTUs with prevalence less than 10% and maximum proportion less than 0.2% were removed. 
This OTU table can be used for simulating vaginal microbiome sequencing data.
</p>


<h3>Source</h3>

<p>Schiffer L, Azhar R, Shepherd L, Ramos M, Geistlinger L, Huttenhower C, Dowd JB, Segata N, Waldron L (2019). 
“HMP16SData: Efficient Access to the Human Microbiome Project through Bioconductor.” American Journal of Epidemiology. doi: 10.1093/aje/kwz006.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(stool.otu.tab)
</code></pre>

<hr>
<h2 id='ZicoSeq'>A linear Model-based Permutation Test for Differential Abundance Analysis of Microbiome Data and Other Omics Data</h2><span id='topic+ZicoSeq'></span>

<h3>Description</h3>

<p>ZicoSeq is a permutation test (Smith permutation) for differential abundance analysis of microbiome sequencing data. The input 
can be a count or a proportion matrix.  When a count matrix is provided, it provides an option to draw posterior samples 
of the underlying proportions to account for the sampling variability during the sequencing process. The test results are
aggregated over these posterior samples.  For both count and proportion data, a reference-based ratio approach is used
to account for compositional effects.  As a general methodology, ZicoSeq can also be applied to differential analysis of other
omics data. In this case, they are not treated as compositional data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZicoSeq(
  meta.dat,
  feature.dat,
  grp.name,
  adj.name = NULL,
  feature.dat.type = c('count', 'proportion', 'other'),
  prev.filter = 0,
  mean.abund.filter = 0,
  max.abund.filter = 0,
  min.prop = 0,
  is.winsor = TRUE,
  outlier.pct = 0.03,
  winsor.end = c('top', 'bottom', 'both'),
  is.post.sample = TRUE,
  post.sample.no = 25,
  link.func = list(function(x) sign(x) * (abs(x))^0.5),
  stats.combine.func = max,
  perm.no = 99,
  strata = NULL,
  ref.pct = 0.5,
  stage.no = 6,
  excl.pct = 0.2,
  p.max = 500,
  is.fwer = FALSE,
  verbose = TRUE,
  return.feature.dat = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZicoSeq_+3A_meta.dat">meta.dat</code></td>
<td>
<p>a data frame containing the sample meta data.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_feature.dat">feature.dat</code></td>
<td>
<p>a matrix of feature data, row - features (OTUs, genes, etc) , column - samples.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_grp.name">grp.name</code></td>
<td>
<p>the name for the variable of interest. It could be numeric or categorical; should be in <code>meta.dat</code>.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_adj.name">adj.name</code></td>
<td>
<p>the name(s) for the variable(s) to be adjusted. Multiple variables are allowed.
They could be numeric or categorical; should be in <code>meta.dat</code>.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_feature.dat.type">feature.dat.type</code></td>
<td>
<p>the type of the feature data. It could be &quot;count&quot;, &quot;proportion&quot; or &quot;other&quot;. For &quot;proportion&quot; data type, posterior sampling will not be performed,
but the reference-based ratio approach will still be used to address compositional effects. For &quot;other&quot; data type, neither posterior sampling or reference-base ratio approach
will be used.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_prev.filter">prev.filter</code></td>
<td>
<p>the prevalence (percentage of nonzeros) cutoff, under which the features will  be filtered. The default is 0. </p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_mean.abund.filter">mean.abund.filter</code></td>
<td>
<p>the mean relative abundance cutoff, under which the features will  be filtered. The default is 0.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_max.abund.filter">max.abund.filter</code></td>
<td>
<p>the max relative abundance cutoff, under which the features will  be filtered. The default is 0.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_min.prop">min.prop</code></td>
<td>
<p> proportions less than this value will be replaced with this value. Only relevant when log transformation is used.
Default is 0.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_is.winsor">is.winsor</code></td>
<td>
<p>a logical value indicating whether winsorization should be performed to replace outliers. The default is TRUE.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_outlier.pct">outlier.pct</code></td>
<td>
<p>the expected percentage of outliers. These outliers will be winsorized. The default is 0.03. For count/proportion data, 
<code>outlier.pct</code> should be less than <code>prev.filter</code>.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_winsor.end">winsor.end</code></td>
<td>
<p>a character indicating whether the outliers at the &quot;top&quot;, &quot;bottom&quot; or &quot;both&quot; will be winsorized.
The default is &quot;top&quot;. If the <code>feature.dat.type</code> is &quot;other&quot;, &quot;both&quot; may be considered.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_is.post.sample">is.post.sample</code></td>
<td>
<p>a logical value indicating whether to perform posterior sampling of the underlying proportions.
Only relevant when the feature data are counts.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_post.sample.no">post.sample.no</code></td>
<td>
<p>the number of posterior samples if posterior sampling is used. The default is 25.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_link.func">link.func</code></td>
<td>
<p>a list of transformation functions for the feature data or the ratios. Based on our experience, 
square-root transformation is a robust choice for many datasets.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_perm.no">perm.no</code></td>
<td>
<p>the number of permutations. If the raw p values are of the major interest, set <code>perm.no</code> to at least 999.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_strata">strata</code></td>
<td>
<p>a factor such as subject IDs indicating the permutation strata or characters indicating the strata variable in <code>meta.dat</code>. Permutation will be confined to each stratum. This can be used for paired or some longitudinal designs.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_stats.combine.func">stats.combine.func</code></td>
<td>
<p>function to combine the F-statistic for the omnibus test. The default is <code>max</code>.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_ref.pct">ref.pct</code></td>
<td>
<p>percentage of reference taxa. The default is 0.5.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_p.max">p.max</code></td>
<td>
<p>the maximum number of (most abundant) taxa to be considered in reference taxa selection; only relevant when the nubmer of taxa is huge. The default is 500, i.e., when the nubmer of taxa is larger than 500, only the 500 most abundant taxa will be used for reference selection. This is to reduce the computational time.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_stage.no">stage.no</code></td>
<td>
<p>the number of stages if multiple-stage normalization is used. The default is 6.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_excl.pct">excl.pct</code></td>
<td>
<p>the maximum percentage of significant features (nominal p-value &lt; 0.05) in the reference set that should be removed. 
Only relevant when multiple-stage normalization is used.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_is.fwer">is.fwer</code></td>
<td>
<p>a logical value indicating whether the family-wise error rate control (West-Young) should be performed.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_verbose">verbose</code></td>
<td>
<p>a logical value indicating whether the trace information should be printed out.</p>
</td></tr>
<tr><td><code id="ZicoSeq_+3A_return.feature.dat">return.feature.dat</code></td>
<td>
<p>a logical value indicating whether the winsorized, filtered &quot;feature.dat&quot; matrix should be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ZicoSeq</code> is a linear model-based permutation test developed for differential abundance analysis of zero-inflated compositional data. Although its development is
motivated by zero-inflated microbiome sequence count data, it can be applied to proportion (composition) data and more generally to other types of omics data. 
Currently, it has the following components: 1. Winsorization to decrease the influence of outliers; 2. Posterior sampling based on a beta mixture prior to address sampling variability 
and zero inflation; 3. Reference-based multiple-stage normalization to address compositional effects; 4. An omnibus test to address diverse feature-covariate relationships; 5. Permutation-based 
false discovery rate control / family-wise error rate control for multiple testing correction, which takes into account the correlation structure in the feature data.
</p>


<h3>Value</h3>

<p>A list with the elements
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>the call</p>
</td></tr>
<tr><td><code>feature.dat</code></td>
<td>
<p>the winsorized, filtered <code>feature.dat</code> matrix.</p>
</td></tr>
<tr><td><code>meta.dat</code></td>
<td>
<p><code>meta.dat</code> used.</p>
</td></tr>
<tr><td><code>grp.name</code></td>
<td>
<p>the name of the variable of interest.</p>
</td></tr>
<tr><td><code>filter.features</code></td>
<td>
<p>a vector of the names of the features that are filtered.</p>
</td></tr>
<tr><td><code>ref.features</code></td>
<td>
<p>a vector of the names of the reference features. Only relevant when reference approach is used.</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>a matrix of percent explained variance (number of features by number of transformation functions).</p>
</td></tr>
<tr><td><code>F0</code></td>
<td>
<p>a matrix of F-statistics (number of features by number of transformation functions).</p>
</td></tr>
<tr><td><code>RSS</code></td>
<td>
<p>a matrix of residual sum squares (number of features by number of transformation functions).</p>
</td></tr>
<tr><td><code>df.model</code>, <code>df.residual</code></td>
<td>
<p>degrees of freedom for the model and residual space.</p>
</td></tr>
<tr><td><code>coef.list</code></td>
<td>
<p>a list of the linear regression coefficients under the specified transformations.</p>
</td></tr>
<tr><td><code>p.raw</code></td>
<td>
<p>the raw p-values based on permutations (not accurate if <code>perm.no</code> is small).</p>
</td></tr>
<tr><td><code>p.adj.fdr</code></td>
<td>
<p>permutation-based FDR-adjusted p-values.</p>
</td></tr>
<tr><td><code>p.adj.fwer</code></td>
<td>
<p>permutation-based FWER-adjusted (West-Young) p-values.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jun Chen</p>


<h3>References</h3>

<p>Yang, L. &amp; Chen, J. 2022. A comprehensive evaluation of differential abundance analysis methods: current status and potential solutions.
Microbiome, 10(1), 1-23.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ZicoSeq.plot">ZicoSeq.plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

comm &lt;- t(throat.otu.tab)
meta.dat &lt;- throat.meta

set.seed(123)
# For count data
zico.obj &lt;- ZicoSeq(meta.dat = meta.dat, feature.dat = comm, 
		grp.name = 'SmokingStatus', adj.name = 'Sex', feature.dat.type = "count",
		# Filter to remove rare taxa
		prev.filter = 0.2, mean.abund.filter = 0,  max.abund.filter = 0.002, min.prop = 0, 
		# Winsorization to replace outliers
		is.winsor = TRUE, outlier.pct = 0.03, winsor.end = 'top',
		# Posterior sampling to impute zeros
		is.post.sample = TRUE, post.sample.no = 25, 
		# Multiple link functions to capture diverse taxon-covariate relation
		link.func = list(function (x) x^0.25, function (x) x^0.5, function (x) x^0.75), 
		stats.combine.func = max,
		# Permutation-based multiple testing correction
		perm.no = 99,  strata = NULL, 
		# Reference-based multiple stage normalization
		ref.pct = 0.5, stage.no = 6, excl.pct = 0.2,
		# Family-wise error rate control
		is.fwer = FALSE,
		verbose = TRUE, return.feature.dat = FALSE)

which(zico.obj$p.adj.fdr &lt;= 0.05)

# For proportion data
comm.p &lt;- t(t(comm) / colSums(comm))
zico.obj &lt;- ZicoSeq(meta.dat = meta.dat, feature.dat = comm.p, 
		grp.name = 'SmokingStatus', adj.name = 'Sex', feature.dat.type = "proportion",
		# Filter to remove rare taxa
		prev.filter = 0.2, mean.abund.filter = 0,  max.abund.filter = 0.002, min.prop = 0, 
		# Winsorization to replace outliers
		is.winsor = TRUE, outlier.pct = 0.03, winsor.end = 'top',
		# Posterior sampling will be automatically disabled
		is.post.sample = FALSE, post.sample.no = 25, 
		# Use the square-root transformation
		link.func = list(function (x) x^0.5), stats.combine.func = max,
		# Permutation-based multiple testing correction
		perm.no = 99,  strata = NULL, 
		# Reference-based multiple stage normalization
		ref.pct = 0.5, stage.no = 6, excl.pct = 0.2,
		# Family-wise error rate control
		is.fwer = FALSE,
		verbose = TRUE, return.feature.dat = FALSE)

which(zico.obj$p.adj.fdr &lt;= 0.05)


# For other type of data. The user should be responsible for the filtering.
comm.o &lt;- comm[rowMeans(comm != 0) &gt;= 0.2, ] + 1
comm.o &lt;- log(t(t(comm.o) / colSums(comm.o)))
zico.obj &lt;- ZicoSeq(meta.dat = meta.dat, feature.dat = comm.o, 
		grp.name = 'SmokingStatus', adj.name = 'Sex', feature.dat.type = "other",
		# Filter will not be applied
		prev.filter = 0, mean.abund.filter = 0,  max.abund.filter = 0, min.prop = 0, 
		# Winsorization to both ends of the distribution 
		is.winsor = TRUE, outlier.pct = 0.03, winsor.end = 'both',
		# Posterior sampling will be automatically disabled
		is.post.sample = FALSE, post.sample.no = 25, 
		# Identity function is used
		link.func = list(function (x) x), stats.combine.func = max,
		# Permutation-based multiple testing correction
		perm.no = 99,  strata = NULL, 
		# Reference-based multiple-stage normalization will not be performed
		ref.pct = 0.5, stage.no = 6, excl.pct = 0.2,
		# Family-wise error rate control
		is.fwer = TRUE,
		verbose = TRUE, return.feature.dat = FALSE)

which(zico.obj$p.adj.fdr &lt;= 0.05)

</code></pre>

<hr>
<h2 id='ZicoSeq.plot'>A Plot Function for Visualizing the ZicoSeq Results</h2><span id='topic+ZicoSeq.plot'></span>

<h3>Description</h3>

<p><code>ZicoSeq.plot</code> produces volcano plots with the y-axis being the log10 (adjusted) p-value and the x-axis being the signed  R^2^ 
with the sign indicating the association direction determined based on  the sign of the regression coefficients (for multi-categorical
variables, sign is not applicable). The names of differential taxa passing a specific cutoff will be printed on the figure.
When data types are counts and proportions, the mean abundance and prevalence will be visualized; when the data type is 'other', 
mean and standard deviation of the features will be visualized. Users need to set <code>return.feature.dat = T</code> when using the plot function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZicoSeq.plot(
  ZicoSeq.obj,
  pvalue.type = c('p.adj.fdr','p.raw','p.adj.fwer'), 
  cutoff = 0.1, 
  text.size = 10,
  out.dir = NULL, 
  file.name = 'ZicoSeq.plot.pdf', 
  width = 10, 
  height = 6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZicoSeq.plot_+3A_zicoseq.obj">ZicoSeq.obj</code></td>
<td>
<p>object from calling the function <code>ZicoSeq</code>.</p>
</td></tr>
<tr><td><code id="ZicoSeq.plot_+3A_pvalue.type">pvalue.type</code></td>
<td>
<p>character string, one of 'p.adj.fdr','p.raw' and 'p.adj.fwer'.</p>
</td></tr>
<tr><td><code id="ZicoSeq.plot_+3A_cutoff">cutoff</code></td>
<td>
<p>a cutoff between 0 and 1 for pvalue.type, below which the names of the features will be printed.</p>
</td></tr>
<tr><td><code id="ZicoSeq.plot_+3A_text.size">text.size</code></td>
<td>
<p>text size for the plots.</p>
</td></tr>
<tr><td><code id="ZicoSeq.plot_+3A_out.dir">out.dir</code></td>
<td>
<p>character string; the directory to save the figure, e.g., <code>getwd()</code>. Default is NULL. If NULL, figure will not be saved.</p>
</td></tr>
<tr><td><code id="ZicoSeq.plot_+3A_file.name">file.name</code></td>
<td>
<p>character string; name of the file to be saved. </p>
</td></tr>
<tr><td><code id="ZicoSeq.plot_+3A_width">width</code></td>
<td>
<p>the width of the graphics region in inches. See R function <code>ggsave</code>.</p>
</td></tr>
<tr><td><code id="ZicoSeq.plot_+3A_height">height</code></td>
<td>
<p>the height of the graphics region in inches. See R function <code>ggsave</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gtable of aligned plots from <code>ggarrange</code>.
</p>


<h3>Author(s)</h3>

<p>Lu Yang, Jun Chen</p>


<h3>References</h3>

<p>Yang, L. &amp; Chen, J. 2022. A comprehensive evaluation of differential abundance analysis methods: current status and potential solutions.
Microbiome. Microbiome, 10(1), 1-23.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ZicoSeq">ZicoSeq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

comm &lt;- t(throat.otu.tab)
meta.dat &lt;- throat.meta

set.seed(123)
# For count data
zico.obj &lt;- ZicoSeq(meta.dat = meta.dat, feature.dat = comm, 
		grp.name = 'SmokingStatus', adj.name = 'Sex', feature.dat.type = "count",
		# Filter to remove rare taxa
		prev.filter = 0.2, mean.abund.filter = 0,  max.abund.filter = 0.002, min.prop = 0, 
		# Winsorization to replace outliers
		is.winsor = TRUE, outlier.pct = 0.03, winsor.end = 'top',
		# Posterior sampling to impute zeros
		is.post.sample = TRUE, post.sample.no = 25, 
		# Multiple link functions to capture diverse taxon-covariate relation
		link.func = list(function (x) x^0.25, function (x) x^0.5, function (x) x^0.75), 
		stats.combine.func = max,
		# Permutation-based multiple testing correction
		perm.no = 99,  strata = NULL, 
		# Reference-based multiple stage normalization
		ref.pct = 0.5, stage.no = 6, excl.pct = 0.2, 
		# Family-wise error rate control
		is.fwer = FALSE,
		verbose = TRUE, return.feature.dat = TRUE)

which(zico.obj$p.adj.fdr &lt;= 0.1)

ZicoSeq.plot(ZicoSeq.obj = zico.obj, pvalue.type = 'p.adj.fdr', 
             cutoff = 0.1, text.size = 10, out.dir = NULL, width = 15, height = 10)    

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
