<!DOCTYPE html><html><head><title>Help for package gwaRs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gwaRs}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gwaRs'><p>Create Manhattan, Q-Q, and PCA plots for GWAS data.</p></a></li>
<li><a href='#gwasData'><p>GWAS results</p></a></li>
<li><a href='#highlightSNPS'><p>Highlight SNPs</p></a></li>
<li><a href='#karyotype_plot'><p>Karyotype Plot</p></a></li>
<li><a href='#man_plot'><p>Manhattan Plot</p></a></li>
<li><a href='#mirrored_man_plot'><p>Mirrored Manhattan Plot</p></a></li>
<li><a href='#pca_plot'><p>PCA Plot</p></a></li>
<li><a href='#pcaData'><p>PCA results</p></a></li>
<li><a href='#qq_plot'><p>QQ Plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Manhattan, Q-Q, and PCA Plots using 'ggplot2'</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Generate Manhattan, Q-Q, and PCA plots from GWAS and PCA results using 'ggplot2'.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENCE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>RColorBrewer (&ge; 1.1.2), ggplot2, ggrepel, dplyr, tidyr,
data.table, scales, stats, grDevices</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/LindoNkambule/gwaRs">https://github.com/LindoNkambule/gwaRs</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/LindoNkambule/gwaRs/issues">https://github.com/LindoNkambule/gwaRs/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-04-19 17:05:40 UTC; lindokuhle</td>
</tr>
<tr>
<td>Author:</td>
<td>Lindokuhle Nkambule
    <a href="https://orcid.org/0000-0002-5682-2834"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lindokuhle Nkambule &lt;lindonkambule116@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-04-19 17:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='gwaRs'>Create Manhattan, Q-Q, and PCA plots for GWAS data.</h2><span id='topic+gwaRs'></span><span id='topic+gwaRs-package'></span>

<h3>Description</h3>

<p>A package for creating Manhattan, Q-Q, and PCA plots for GWAS data.
</p>


<h3>Author(s)</h3>

<p>Lindokuhle Nkambule <a href="mailto:lindonkambule116@gmail.com">lindonkambule116@gmail.com</a>
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/LindoNkambule/gwaRs">https://github.com/LindoNkambule/gwaRs</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/LindoNkambule/gwaRs/issues">https://github.com/LindoNkambule/gwaRs/issues</a>
</p>
</li></ul>


<hr>
<h2 id='gwasData'>GWAS results</h2><span id='topic+gwasData'></span>

<h3>Description</h3>

<p>Example GWAS data.
</p>

<hr>
<h2 id='highlightSNPS'>Highlight SNPs</h2><span id='topic+highlightSNPS'></span>

<h3>Description</h3>

<p>A couple of SNPs to be annotated
</p>

<hr>
<h2 id='karyotype_plot'>Karyotype Plot</h2><span id='topic+karyotype_plot'></span>

<h3>Description</h3>

<p>Creates a Karyotype plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>karyotype_plot(
  data,
  density.col = c("darkgreen", "yellow", "red"),
  window.size = 1e+06,
  title = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="karyotype_plot_+3A_data">data</code></td>
<td>
<p>A data.frame with &quot;CHR&quot; and &quot;BP&quot;columns.</p>
</td></tr>
<tr><td><code id="karyotype_plot_+3A_density.col">density.col</code></td>
<td>
<p>A character vector with colors to use for gradients.</p>
</td></tr>
<tr><td><code id="karyotype_plot_+3A_window.size">window.size</code></td>
<td>
<p>A double precision numeric value indicating the window size.</p>
</td></tr>
<tr><td><code id="karyotype_plot_+3A_title">title</code></td>
<td>
<p>A string denoting the title to use for the plot. Default is 'Manhattan Plot'</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a SNP Karyotype or Density plot from an R dataframe with &quot;CHR&quot; and &quot;BP&quot; columns.
</p>


<h3>Value</h3>

<p>A SNP Karyotype plot.
</p>


<h3>Author(s)</h3>

<p>Lindokuhle Nkambule
</p>


<h3>Examples</h3>

<pre><code class='language-R'>karyotype_plot(gwasData)

</code></pre>

<hr>
<h2 id='man_plot'>Manhattan Plot</h2><span id='topic+man_plot'></span>

<h3>Description</h3>

<p>Creates a Manhattan plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>man_plot(
  data,
  chromCol = c("gray44", "black"),
  genomewideline = -log10(5e-08),
  suggestiveline = -log10(1e-05),
  chromosome = "ALL",
  annotatePval = FALSE,
  annotateSNP = NULL,
  annotateCol = "red",
  highlight = NULL,
  highlightCol = "green3",
  title = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="man_plot_+3A_data">data</code></td>
<td>
<p>PLINK assoc output, tab-delimited, or a data.frame with &quot;SNP&quot;, &quot;CHR&quot;, &quot;BP&quot;, and &quot;P&quot; columns.</p>
</td></tr>
<tr><td><code id="man_plot_+3A_chromcol">chromCol</code></td>
<td>
<p>A character vector indicating which colors to alternate for the chromosomes.</p>
</td></tr>
<tr><td><code id="man_plot_+3A_genomewideline">genomewideline</code></td>
<td>
<p>Where to draw the &quot;genome-wide significant&quot; line. Default
-log10(5e-8). Set to FALSE or F to disable</p>
</td></tr>
<tr><td><code id="man_plot_+3A_suggestiveline">suggestiveline</code></td>
<td>
<p>Where to draw the &quot;suggestive&quot; line. Default
-log10(1e-5). Set to FALSE or F to disable.</p>
</td></tr>
<tr><td><code id="man_plot_+3A_chromosome">chromosome</code></td>
<td>
<p>An integer indicating which chromosome to plot. Default is &quot;ALL&quot;.</p>
</td></tr>
<tr><td><code id="man_plot_+3A_annotatepval">annotatePval</code></td>
<td>
<p>If set, SNPs with p-value less than or equal to this p-value will be annotated on the plot.</p>
</td></tr>
<tr><td><code id="man_plot_+3A_annotatesnp">annotateSNP</code></td>
<td>
<p>A character vector of SNPs in your dataset to annotate.
If some of the SNPs are not in your dataset, gwaRs will throw a warning message.</p>
</td></tr>
<tr><td><code id="man_plot_+3A_annotatecol">annotateCol</code></td>
<td>
<p>A string denoting the color to use for the annotations.</p>
</td></tr>
<tr><td><code id="man_plot_+3A_highlight">highlight</code></td>
<td>
<p>A character vector of SNPs in the dataset to highlight.
If some of the SNPs are not in your dataset, gwaRs will throw a warning message. Default is NULL.</p>
</td></tr>
<tr><td><code id="man_plot_+3A_highlightcol">highlightCol</code></td>
<td>
<p>A string denoting the color to use to highlight the SNPs.</p>
</td></tr>
<tr><td><code id="man_plot_+3A_title">title</code></td>
<td>
<p>A string denoting the title to use for the plot. Default is 'Manhattan Plot'</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a Manhattan plot from PLINK assoc output (or any tab-delimited file or data frame with
&quot;SNP&quot;, &quot;CHR&quot;, &quot;BP&quot;, and &quot;P&quot; columns).
</p>


<h3>Value</h3>

<p>A Manhattan plot.
</p>


<h3>Author(s)</h3>

<p>Lindokuhle Nkambule
</p>


<h3>Examples</h3>

<pre><code class='language-R'>man_plot(gwasData)

</code></pre>

<hr>
<h2 id='mirrored_man_plot'>Mirrored Manhattan Plot</h2><span id='topic+mirrored_man_plot'></span>

<h3>Description</h3>

<p>Creates a Mirrored Manhattan Plot for two traits
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mirrored_man_plot(
  data,
  trait1 = NULL,
  trait2 = NULL,
  trait1_chromCols = c("gray66", "grey36"),
  trait2_chromCols = c("steelblue1", "steelblue4"),
  xlab = "Genomic Position (chromosome)",
  title = "Manhattan Plot",
  annotate_trait1_pval = FALSE,
  annotate_trait1_color = "red",
  annotate_trait2_pval = FALSE,
  annotate_trait2_color = "red",
  annotateSNP = NULL,
  annotateSNPcolor = "red",
  highlight = NULL,
  highlightcolor = "green3",
  genomewideline_trait1 = NULL,
  genomewideline_trait2 = NULL,
  genomewideline_type = "dashed",
  genomewideline_color = "red",
  suggestiveline_trait1 = NULL,
  suggestiveline_trait2 = NULL,
  suggestiveline_type = "dashed",
  suggestiveline_color = "blue"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mirrored_man_plot_+3A_data">data</code></td>
<td>
<p>A tab-delimited or data frame with the compulsory columns:
&quot;CHR&quot;, &quot;SNP&quot;, &quot;BP&quot;, &quot;P&quot;, &quot;Trait&quot;.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_trait1">trait1</code></td>
<td>
<p>A character string of the trait1 as it appears in the input data.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_trait2">trait2</code></td>
<td>
<p>A character string of the trait2 as it appears in the input data.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_trait1_chromcols">trait1_chromCols</code></td>
<td>
<p>A character vector indicating which colors to alternate
for trait1 chromosomes.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_trait2_chromcols">trait2_chromCols</code></td>
<td>
<p>A character vector indicating which colors to alternate
for trait2 chromosomes.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_xlab">xlab</code></td>
<td>
<p>A character string to be used as the x-axis label.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_title">title</code></td>
<td>
<p>A character string to be used as the plot title</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_annotate_trait1_pval">annotate_trait1_pval</code></td>
<td>
<p>If set, trait1 SNPs with p-value less than or equal to this
p-value will be annotated on the plot.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_annotate_trait1_color">annotate_trait1_color</code></td>
<td>
<p>A character string indicating the color to be used for annotating
trait1 SNPs by p-value</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_annotate_trait2_pval">annotate_trait2_pval</code></td>
<td>
<p>If set, trait2 SNPs with p-value less than or equal to this
p-value will be annotated on the plot.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_annotate_trait2_color">annotate_trait2_color</code></td>
<td>
<p>A character string indicating the color to be used for annotating
trait2 SNPs by p-value</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_annotatesnp">annotateSNP</code></td>
<td>
<p>A character vector of SNPs in your dataset to annotate.
If some of the SNPs are not in your dataset, gwaRs will throw a warning message.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_annotatesnpcolor">annotateSNPcolor</code></td>
<td>
<p>A character string denoting the color to use for the annotations.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_highlight">highlight</code></td>
<td>
<p>A character vector of SNPs in the dataset to highlight.
If some of the SNPs are not in your dataset, gwaRs will throw a warning message. Default is NULL.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_highlightcolor">highlightcolor</code></td>
<td>
<p>A character string denoting the color to use to highlight the SNPs.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_genomewideline_trait1">genomewideline_trait1</code></td>
<td>
<p>Where to draw the &quot;genome-wide significant&quot; line for trait1</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_genomewideline_trait2">genomewideline_trait2</code></td>
<td>
<p>Where to draw the &quot;genome-wide significant&quot; line for trait2</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_genomewideline_type">genomewideline_type</code></td>
<td>
<p>A character string denoting the type of line to be used for the
&quot;genome-wide significant&quot; line. This is the same for both traits. Default is dashed.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_genomewideline_color">genomewideline_color</code></td>
<td>
<p>A character string denoting the color to be used for the
&quot;genome-wide significant&quot; line. This is the same for both traits. Default is red.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_suggestiveline_trait1">suggestiveline_trait1</code></td>
<td>
<p>Where to draw the &quot;suggestive&quot; line for trait1.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_suggestiveline_trait2">suggestiveline_trait2</code></td>
<td>
<p>Where to draw the &quot;suggestive&quot; line for trait2.</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_suggestiveline_type">suggestiveline_type</code></td>
<td>
<p>A character string denoting the type of line to be used for the
&quot;suggestive&quot; line. This is the same for both traits. Default is dashed</p>
</td></tr>
<tr><td><code id="mirrored_man_plot_+3A_suggestiveline_color">suggestiveline_color</code></td>
<td>
<p>A character string denoting the color to be used for the
&quot;suggestive&quot; line. This is the same for both traits. Default is blue.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Create a Mirrored Manhattan Plot from a tab-delimited file or data frame with the compulsory
columns: &quot;CHR&quot;, &quot;SNP&quot;, &quot;BP&quot;, &quot;P&quot;, &quot;Trait&quot; .
</p>


<h3>Value</h3>

<p>A Mirrored Manhattan plot for two traits.
</p>


<h3>Author(s)</h3>

<p>Lindokuhle Nkambule
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mirrored_man_plot(inputData)

## End(Not run)

</code></pre>

<hr>
<h2 id='pca_plot'>PCA Plot</h2><span id='topic+pca_plot'></span>

<h3>Description</h3>

<p>Creates a PCA plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pca_plot(
  data,
  xComponent = "PC1",
  yComponent = "PC2",
  legendPos = "right",
  soft = "PLINK",
  colPalette = "Accent",
  title = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pca_plot_+3A_data">data</code></td>
<td>
<p>PLINK pca or EIGENSTRAT smartpca output,
(or any tab-delimited file or data.frame with the same format as PLINK pca or EIGENSTRAT smartpca output)</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_xcomponent">xComponent</code></td>
<td>
<p>A character vector indicating the principal component value to use for the x-axis.
Default is &quot;PC1&quot;</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_ycomponent">yComponent</code></td>
<td>
<p>A character vector indicating the principal component value to use for the x-axis.
Default is &quot;PC2&quot;</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_legendpos">legendPos</code></td>
<td>
<p>A character vector indicating the legend position. Default is &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_soft">soft</code></td>
<td>
<p>A character vector indicating the software output format. Default is &quot;PLINK&quot;.
If you have a tab-delimited file or data.frame with the same format as EIGENSTRAT output, use &quot;EIGENSTRAT&quot;</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_colpalette">colPalette</code></td>
<td>
<p>A character vector indicating the color palette to use. Default is &quot;Accent&quot;.</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_title">title</code></td>
<td>
<p>A string denoting the title to use for the plot. Default is 'PCA Plot'</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a PCA plot from PLINK pca output, EIGENSTRAT smartpca,
(or any tab-delimited file or data.frame with the same format as PLINK pca or EIGENSTRAT smartpca output).
</p>


<h3>Value</h3>

<p>A PCA plot.
</p>


<h3>Author(s)</h3>

<p>Lindokuhle Nkambule
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pca_plot(pcaData)

</code></pre>

<hr>
<h2 id='pcaData'>PCA results</h2><span id='topic+pcaData'></span>

<h3>Description</h3>

<p>Example PCA data.
</p>

<hr>
<h2 id='qq_plot'>QQ Plot</h2><span id='topic+qq_plot'></span>

<h3>Description</h3>

<p>Creates a Q-Q plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qq_plot(
  data,
  point_col = "black",
  diag_col = "red",
  diag_line = "solid",
  title = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qq_plot_+3A_data">data</code></td>
<td>
<p>PLINK assoc output, tab-delimited, or a data.frame with &quot;P&quot; column.</p>
</td></tr>
<tr><td><code id="qq_plot_+3A_point_col">point_col</code></td>
<td>
<p>A character vector indicating the color to use for the SNP p-values. Default is &quot;black&quot;.</p>
</td></tr>
<tr><td><code id="qq_plot_+3A_diag_col">diag_col</code></td>
<td>
<p>A character vector indicating the color to use for the diagonal line. Default is &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="qq_plot_+3A_diag_line">diag_line</code></td>
<td>
<p>A character vector indicating the line type to use for the diagonal line. Default is &quot;solid&quot;.</p>
</td></tr>
<tr><td><code id="qq_plot_+3A_title">title</code></td>
<td>
<p>A string denoting the title to use for the plot. Default is 'Q-Q Plot'</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a Q-Q plot from PLINK assoc output (or any tab-delimited file or data frame with &quot;P&quot; column).
</p>


<h3>Value</h3>

<p>A Q-Q plot.
</p>


<h3>Author(s)</h3>

<p>Lindokuhle Nkambule
</p>


<h3>Examples</h3>

<pre><code class='language-R'>qq_plot(gwasData)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
