<!DOCTYPE html><html><head><title>Help for package gwasrapidd</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gwasrapidd}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gwasrapidd-package'><p>gwasrapidd: 'REST' 'API' Client for the 'NHGRI'-'EBI' 'GWAS' Catalog</p></a></li>
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#add_object_tier'><p>Adds an element named &lt;obj_type&gt; to content sub-element.</p></a></li>
<li><a href='#allele_name'><p>Extract allele names from strings of the form rs123-G</p></a></li>
<li><a href='#ancestral_groups_tbl'><p>Creates an ancestral groups table.</p></a></li>
<li><a href='#ancestries_tbl'><p>Creates an ancestries table.</p></a></li>
<li><a href='#association_to_study'><p>Map an association id to a study id</p></a></li>
<li><a href='#association_to_trait'><p>Map an association id to an EFO trait id</p></a></li>
<li><a href='#association_to_variant'><p>Map an association id to a variant id</p></a></li>
<li><a href='#associations'><p>Constructor for the S4 associations object.</p></a></li>
<li><a href='#associations_drop_na'><p>Drop any NA associations.</p></a></li>
<li><a href='#associations_tbl'><p>Creates an associations table.</p></a></li>
<li><a href='#associations-class'><p>An S4 class to represent a set of GWAS Catalog associations</p></a></li>
<li><a href='#bind'><p>Bind GWAS Catalog objects</p></a></li>
<li><a href='#child_efo_ids'><p>Get all child terms of this trait in the EFO hierarchy</p></a></li>
<li><a href='#contains_question_mark'><p>Does a string contain a question mark?</p></a></li>
<li><a href='#countries_tbl'><p>Creates a countries table.</p></a></li>
<li><a href='#cytogenetic_band_to_genomic_range'><p>Convert a cytogenetic band string to genomic coordinates.</p></a></li>
<li><a href='#cytogenetic_bands'><p>GRCh38 human cytogenetic bands.</p></a></li>
<li><a href='#empty_to_na'><p>Converts an empty vector to a scalar NA</p></a></li>
<li><a href='#ensembl_ids_tbl'><p>Creates an Ensembl gene identifiers' table.</p></a></li>
<li><a href='#entrez_ids_tbl'><p>Creates an Entrez gene identifiers' table.</p></a></li>
<li><a href='#equal_length'><p>Grows vectors to match longest vector</p></a></li>
<li><a href='#exists_variant'><p>Check if a variant exists in the Catalog.</p></a></li>
<li><a href='#extract_association_id'><p>Extract association identifiers from URLs</p></a></li>
<li><a href='#filter_by_id'><p>Filter GWAS Catalog objects by identifier.</p></a></li>
<li><a href='#filter_variants_by_standard_chromosomes'><p>Filter variants by standard human chromosomes.</p></a></li>
<li><a href='#gc_examples'><p>gwasrapidd entities' examples</p></a></li>
<li><a href='#gc_get'><p>Get a GWAS Catalog resource</p></a></li>
<li><a href='#gc_request'><p>Request a GWAS Catalog REST API endpoint</p></a></li>
<li><a href='#gc_request_all'><p>Request a paginated GWAS Catalog REST API endpoint</p></a></li>
<li><a href='#genomic_contexts_tbl'><p>Creates a genomic contexts table.</p></a></li>
<li><a href='#genotyping_techs_tbl'><p>Creates a genotyping technologies table.</p></a></li>
<li><a href='#get_associations'><p>Get GWAS Catalog associations</p></a></li>
<li><a href='#get_associations_all'><p>Get all GWAS Catalog associations</p></a></li>
<li><a href='#get_associations_by_association_id'><p>Get GWAS Catalog associations by their association identifiers</p></a></li>
<li><a href='#get_associations_by_efo_id'><p>Get GWAS Catalog associations by EFO identifier</p></a></li>
<li><a href='#get_associations_by_efo_trait'><p>Get GWAS Catalog associations by EFO traits</p></a></li>
<li><a href='#get_associations_by_pubmed_id'><p>Get GWAS Catalog associations by PubMed identifiers</p></a></li>
<li><a href='#get_associations_by_study_id'><p>Get GWAS Catalog associations by study identifiers</p></a></li>
<li><a href='#get_associations_by_variant_id'><p>Get GWAS Catalog associations by variant identifiers</p></a></li>
<li><a href='#get_child_efo'><p>Get all child terms of this trait in the EFO hierarchy</p></a></li>
<li><a href='#get_metadata'><p>Get GWAS Catalog metadata</p></a></li>
<li><a href='#get_studies'><p>Get GWAS Catalog studies</p></a></li>
<li><a href='#get_studies_all'><p>Get all GWAS Catalog studies</p></a></li>
<li><a href='#get_studies_by_association_id'><p>Get GWAS Catalog studies by association identifiers</p></a></li>
<li><a href='#get_studies_by_efo_id'><p>Get GWAS Catalog studies by EFO identifier</p></a></li>
<li><a href='#get_studies_by_efo_trait'><p>Get GWAS Catalog studies by EFO traits</p></a></li>
<li><a href='#get_studies_by_efo_uri'><p>Get GWAS Catalog studies by EFO URIs</p></a></li>
<li><a href='#get_studies_by_full_pvalue_set'><p>Get GWAS Catalog studies by full summary statistics criterion</p></a></li>
<li><a href='#get_studies_by_pubmed_id'><p>Get GWAS Catalog studies by PubMed identifiers</p></a></li>
<li><a href='#get_studies_by_reported_trait'><p>Get GWAS Catalog studies by reported traits</p></a></li>
<li><a href='#get_studies_by_study_id'><p>Get GWAS Catalog studies by study identifiers</p></a></li>
<li><a href='#get_studies_by_user_requested'><p>Get GWAS Catalog studies that have been requested by users or not</p></a></li>
<li><a href='#get_studies_by_variant_id'><p>Get GWAS Catalog studies by variant identifiers</p></a></li>
<li><a href='#get_traits'><p>Get GWAS Catalog EFO traits</p></a></li>
<li><a href='#get_traits_all'><p>Get all GWAS Catalog EFO traits</p></a></li>
<li><a href='#get_traits_by_association_id'><p>Get GWAS Catalog traits by association identifiers</p></a></li>
<li><a href='#get_traits_by_efo_id'><p>Get GWAS Catalog traits by EFO identifier</p></a></li>
<li><a href='#get_traits_by_efo_trait'><p>Get GWAS Catalog traits by EFO traits</p></a></li>
<li><a href='#get_traits_by_efo_uri'><p>Get GWAS Catalog traits by EFO URIs</p></a></li>
<li><a href='#get_traits_by_pubmed_id'><p>Get GWAS Catalog traits by PubMed identifiers</p></a></li>
<li><a href='#get_traits_by_study_id'><p>Get GWAS Catalog traits by study identifiers</p></a></li>
<li><a href='#get_variants'><p>Get GWAS Catalog variants</p></a></li>
<li><a href='#get_variants_all'><p>Get all GWAS Catalog variants</p></a></li>
<li><a href='#get_variants_by_association_id'><p>Get GWAS Catalog variants by their association identifiers</p></a></li>
<li><a href='#get_variants_by_cytogenetic_band'><p>Get GWAS Catalog variants by cytogenetic band.</p></a></li>
<li><a href='#get_variants_by_efo_id'><p>Get GWAS Catalog studies by EFO identifier</p></a></li>
<li><a href='#get_variants_by_efo_trait'><p>Get GWAS Catalog variants by EFO traits</p></a></li>
<li><a href='#get_variants_by_gene_name'><p>Get GWAS Catalog variants by gene name.</p></a></li>
<li><a href='#get_variants_by_genomic_range'><p>Get GWAS Catalog variants by genomic range</p></a></li>
<li><a href='#get_variants_by_pubmed_id'><p>Get GWAS Catalog variants by PubMed identifiers</p></a></li>
<li><a href='#get_variants_by_reported_trait'><p>Get GWAS Catalog variants by reported traits</p></a></li>
<li><a href='#get_variants_by_study_id'><p>Get GWAS Catalog variants by study identifiers</p></a></li>
<li><a href='#get_variants_by_variant_id'><p>Get GWAS Catalog variants by variant identifiers</p></a></li>
<li><a href='#is_association_id'><p>Is a string a GWAS Catalog association accession ID?</p></a></li>
<li><a href='#is_ebi_reachable'><p>Is the GWAS Catalog REST API server reachable?</p></a></li>
<li><a href='#is_efo_id'><p>Is a string an EFO trait ID?</p></a></li>
<li><a href='#is_efo_id2'><p>Is a string an EFO trait ID in the broad sense?</p></a></li>
<li><a href='#is_embedded'><p>Is the GWAS response wrapped in an '_embedded' object?</p></a></li>
<li><a href='#is_empty_str'><p>Is string empty or an all whitespace string?</p></a></li>
<li><a href='#is_human_chromosome'><p>Is a string a human chromosome name?</p></a></li>
<li><a href='#is_paginated'><p>Is the GWAS response paginated?</p></a></li>
<li><a href='#is_pubmed_id'><p>Is a string a PubMed ID?</p></a></li>
<li><a href='#is_rs_id'><p>Is a string a valid rsID?</p></a></li>
<li><a href='#is_study_id'><p>Is a string a GWAS Catalog study accession ID?</p></a></li>
<li><a href='#list_to_s4'><p>Convert a named list to an S4 object</p></a></li>
<li><a href='#loci_tbl'><p>Creates a loci table.</p></a></li>
<li><a href='#lstjoin'><p>Joins for lists.</p></a></li>
<li><a href='#metadata_lst'><p>Generate a list to hold GWAS metadata</p></a></li>
<li><a href='#missing_to_na'><p>Recursively map all missing values to <code>NA</code>.</p></a></li>
<li><a href='#n'><p>Number of GWAS Catalog entities</p></a></li>
<li><a href='#normalise_obj'><p>Normalise a JSON-list.</p></a></li>
<li><a href='#object_type_from_url'><p>Identify the GWAS object entity from the URL endpoint</p></a></li>
<li><a href='#open_in_dbsnp'><p>Browse dbSNP from SNP identifiers.</p></a></li>
<li><a href='#open_in_gtex'><p>Browse GTEx from SNP identifiers.</p></a></li>
<li><a href='#open_in_gwas_catalog'><p>Browse GWAS Catalog entities from the GWAS Web Graphical User Interface</p></a></li>
<li><a href='#open_in_pubmed'><p>Browse PubMed from PubMed identifiers.</p></a></li>
<li><a href='#peel_off_embedded'><p>Peels off the _embedded tier from a JSON-list.</p></a></li>
<li><a href='#platforms_tbl'><p>Creates a platforms table.</p></a></li>
<li><a href='#publications_tbl'><p>Creates a publications table.</p></a></li>
<li><a href='#recursive_apply'><p>Simple rapply version that deals with NULL values.</p></a></li>
<li><a href='#reported_genes_tbl'><p>Creates an authors' reported genes table.</p></a></li>
<li><a href='#risk_alleles_tbl'><p>Creates a risk alleles table.</p></a></li>
<li><a href='#s4_to_list'><p>Convert an S4 object into a list</p></a></li>
<li><a href='#set_testing_fast'><p>Setup the environment to skip slow tests</p></a></li>
<li><a href='#set_testing_slow'><p>Setup the environment to still run slow tests</p></a></li>
<li><a href='#setop'><p>Set operations on GWAS Catalog objects.</p></a></li>
<li><a href='#skip_if_testing_is_fast'><p>Skips a test if TEST_FAST is 'true'</p></a></li>
<li><a href='#studies'><p>Constructor for the S4 studies object.</p></a></li>
<li><a href='#studies_drop_na'><p>Drop any NA studies.</p></a></li>
<li><a href='#studies_tbl'><p>Creates a studies table.</p></a></li>
<li><a href='#studies-class'><p>An S4 class to represent a set of GWAS Catalog studies</p></a></li>
<li><a href='#study_to_association'><p>Map a study id to an association id</p></a></li>
<li><a href='#study_to_trait'><p>Map a study id to a EFO trait id</p></a></li>
<li><a href='#study_to_variant'><p>Map a study id to a variant id</p></a></li>
<li><a href='#subset-associations'><p>Subset an associations object</p></a></li>
<li><a href='#subset-studies'><p>Subset a studies object</p></a></li>
<li><a href='#subset-traits'><p>Subset a traits object</p></a></li>
<li><a href='#subset-variants'><p>Subset a variants object</p></a></li>
<li><a href='#sure'><p>Are you sure?</p></a></li>
<li><a href='#trait_to_association'><p>Map an EFO trait id to an association id</p></a></li>
<li><a href='#trait_to_study'><p>Map an EFO trait id to a study id</p></a></li>
<li><a href='#trait_to_variant'><p>Map an EFO trait id to a variant id</p></a></li>
<li><a href='#traits'><p>Constructor for the S4 traits object.</p></a></li>
<li><a href='#traits_drop_na'><p>Drop any NA traits.</p></a></li>
<li><a href='#traits_tbl'><p>Creates a traits tibble</p></a></li>
<li><a href='#traits-class'><p>An S4 class to represent a set of GWAS Catalog EFO traits.</p></a></li>
<li><a href='#tws'><p>Trim whitespace.</p></a></li>
<li><a href='#v_ensembl_ids_tbl'><p>Creates a gene Ensembl identifiers table.</p></a></li>
<li><a href='#v_entrez_ids_tbl'><p>Creates a gene Entrez identifiers table.</p></a></li>
<li><a href='#variant_name'><p>Extract variant identifiers from strings of the form rs123-G</p></a></li>
<li><a href='#variant_to_association'><p>Map a variant id to an association id</p></a></li>
<li><a href='#variant_to_study'><p>Map a variant id to a study id</p></a></li>
<li><a href='#variant_to_trait'><p>Map a variant id to an EFO trait</p></a></li>
<li><a href='#variants'><p>Constructor for the S4 variants object.</p></a></li>
<li><a href='#variants_drop_na'><p>Drop any NA variants.</p></a></li>
<li><a href='#variants_tbl'><p>Creates a variants table.</p></a></li>
<li><a href='#variants-class'><p>An S4 class to represent a set of GWAS Catalog variants</p></a></li>
<li><a href='#write_xlsx'><p>Export a GWAS Catalog object to xlsx</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>'REST' 'API' Client for the 'NHGRI'-'EBI' 'GWAS' Catalog</td>
</tr>
<tr>
<td>Version:</td>
<td>0.99.17</td>
</tr>
<tr>
<td>Description:</td>
<td>'GWAS' R 'API' Data Download.
    This package provides easy access to the 'NHGRI'-'EBI' 'GWAS' Catalog data by
    accessing the 'REST' 'API' <a href="https://www.ebi.ac.uk/gwas/rest/docs/api/">https://www.ebi.ac.uk/gwas/rest/docs/api/</a>.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.3)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ramiromagno/gwasrapidd">https://github.com/ramiromagno/gwasrapidd</a>,
<a href="https://rmagno.eu/gwasrapidd/">https://rmagno.eu/gwasrapidd/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ramiromagno/gwasrapidd/issues">https://github.com/ramiromagno/gwasrapidd/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>magrittr, httr, urltools, pingr, stringr, concatenate, dplyr,
jsonlite, purrr, tibble, glue, tidyr (&gt; 0.8.99), assertthat,
rlang, methods, lubridate, plyr, testthat, utils, progress,
writexl</td>
</tr>
<tr>
<td>Suggests:</td>
<td>httptest, spelling, covr, knitr, rmarkdown, bookdown</td>
</tr>
<tr>
<td>Collate:</td>
<td>'browser.R' 'class-associations.R' 'class-studies.R'
'class-traits.R' 'class-variants.R' 'data.R' 'ebi_server.R'
'generics.R' 'get_associations.R' 'get_metadata.R'
'get_studies.R' 'get_traits.R' 'get_variants.R'
'gwasrapidd-package.R' 'id_mapping.R' 'list_joins.R'
'missing.R' 'parse-associations.R' 'parse-studies.R'
'parse-traits.R' 'parse-utils.R' 'parse-variants.R'
'post-studies.R' 'post-traits.R' 'post-variants.R'
'recursive_apply.R' 'request.R' 's4-utils.R' 'sure.R' 'tests.R'
'utils-pipe.R' 'utils.R' 'write_xlsx.R'</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>biocViews:</td>
<td>ThirdPartyClient, BiomedicalInformatics,
GenomeWideAssociation, SNP</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-15 16:42:18 UTC; rmagno</td>
</tr>
<tr>
<td>Author:</td>
<td>Ramiro Magno <a href="https://orcid.org/0000-0001-5226-3441"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Ana-Teresa Maia <a href="https://orcid.org/0000-0002-0454-9207"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  CINTESIS [cph, fnd],
  Pattern Institute [cph, fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ramiro Magno &lt;rmagno@pattern.institute&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-15 21:50:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='gwasrapidd-package'>gwasrapidd: 'REST' 'API' Client for the 'NHGRI'-'EBI' 'GWAS' Catalog</h2><span id='topic+gwasrapidd'></span><span id='topic+gwasrapidd-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>'GWAS' R 'API' Data Download. This package provides easy access to the 'NHGRI'-'EBI' 'GWAS' Catalog data by accessing the 'REST' 'API' <a href="https://www.ebi.ac.uk/gwas/rest/docs/api/">https://www.ebi.ac.uk/gwas/rest/docs/api/</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Ramiro Magno <a href="mailto:rmagno@pattern.institute">rmagno@pattern.institute</a> (<a href="https://orcid.org/0000-0001-5226-3441">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Ana-Teresa Maia <a href="mailto:maia.anateresa@gmail.com">maia.anateresa@gmail.com</a> (<a href="https://orcid.org/0000-0002-0454-9207">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> CINTESIS [copyright holder, funder]
</p>
</li>
<li><p> Pattern Institute [copyright holder, funder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/ramiromagno/gwasrapidd">https://github.com/ramiromagno/gwasrapidd</a>
</p>
</li>
<li> <p><a href="https://rmagno.eu/gwasrapidd/">https://rmagno.eu/gwasrapidd/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/ramiromagno/gwasrapidd/issues">https://github.com/ramiromagno/gwasrapidd/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Value</h3>

<p>The same as the rhs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>c(1,2,3) %&gt;% mean()
</code></pre>

<hr>
<h2 id='add_object_tier'>Adds an element named &lt;obj_type&gt; to content sub-element.</h2><span id='topic+add_object_tier'></span>

<h3>Description</h3>

<p>This function takes a list, sees if it has an element named 'content' and
then checks if content contains an element named <code>obj_type</code>: if it does
not contain an element as passed in <code>obj_type</code> this function adds it,
otherwise it leaves the list <code>obj</code> untouched. The contents of 'content'
are set as value of <code>&lt;obj&gt;$content$&lt;obj_type&gt;</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_object_tier(obj, obj_type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_object_tier_+3A_obj">obj</code></td>
<td>
<p>A list.</p>
</td></tr>
<tr><td><code id="add_object_tier_+3A_obj_type">obj_type</code></td>
<td>
<p>A non-empty string.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the <code>&lt;obj&gt;$content$&lt;obj_type&gt;</code>, whose value is
either <code>NULL</code> if it had not any value before or the value of
<code>&lt;obj&gt;$content</code>.
</p>

<hr>
<h2 id='allele_name'>Extract allele names from strings of the form rs123-G</h2><span id='topic+allele_name'></span>

<h3>Description</h3>

<p>This function parses strings of the form <code>"rs123-G"</code> and returns
the name of the allele; it uses the regex <code>([ATCG]+)$</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allele_name(risk_allele_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="allele_name_+3A_risk_allele_names">risk_allele_names</code></td>
<td>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of allele names.
</p>

<hr>
<h2 id='ancestral_groups_tbl'>Creates an ancestral groups table.</h2><span id='topic+ancestral_groups_tbl'></span>

<h3>Description</h3>

<p>Creates a ancestral groups table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ancestral_groups_tbl(
  study_id = character(),
  ancestry_id = integer(),
  ancestral_group = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ancestral_groups_tbl_+3A_study_id">study_id</code></td>
<td>
<p>GWAS Catalog study accession identifier.</p>
</td></tr>
<tr><td><code id="ancestral_groups_tbl_+3A_ancestry_id">ancestry_id</code></td>
<td>
<p>Ancestry identifier.</p>
</td></tr>
<tr><td><code id="ancestral_groups_tbl_+3A_ancestral_group">ancestral_group</code></td>
<td>
<p>Ancestral group.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='ancestries_tbl'>Creates an ancestries table.</h2><span id='topic+ancestries_tbl'></span>

<h3>Description</h3>

<p>Creates an ancestries table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ancestries_tbl(
  study_id = character(),
  ancestry_id = integer(),
  type = character(),
  number_of_individuals = integer()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ancestries_tbl_+3A_study_id">study_id</code></td>
<td>
<p>GWAS Catalog study accession identifier.</p>
</td></tr>
<tr><td><code id="ancestries_tbl_+3A_ancestry_id">ancestry_id</code></td>
<td>
<p>Ancestry identifier.</p>
</td></tr>
<tr><td><code id="ancestries_tbl_+3A_type">type</code></td>
<td>
<p>Type of cohort sample, either <code>"initial"</code> or
<code>"replication"</code>.</p>
</td></tr>
<tr><td><code id="ancestries_tbl_+3A_number_of_individuals">number_of_individuals</code></td>
<td>
<p>Number of individuals in the cohort sample.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='association_to_study'>Map an association id to a study id</h2><span id='topic+association_to_study'></span>

<h3>Description</h3>

<p>Map an association accession identifier to a study accession identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>association_to_study(association_id, verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="association_to_study_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of association accession
identifiers.</p>
</td></tr>
<tr><td><code id="association_to_study_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="association_to_study_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two identifiers. First column is the association
identifier and the second column is the study identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map GWAS association identifiers to study identifiers
association_to_study(c('24300097', '24299759'))

## End(Not run)

</code></pre>

<hr>
<h2 id='association_to_trait'>Map an association id to an EFO trait id</h2><span id='topic+association_to_trait'></span>

<h3>Description</h3>

<p>Map an association accession identifier to an EFO trait id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>association_to_trait(association_id, verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="association_to_trait_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of association accession
identifiers.</p>
</td></tr>
<tr><td><code id="association_to_trait_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="association_to_trait_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two identifiers. First column is the association
identifier and the second column is the EFO trait identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map GWAS association identifiers to EFO trait identifiers
association_to_trait(c('24300097', '24299759'))

## End(Not run)

</code></pre>

<hr>
<h2 id='association_to_variant'>Map an association id to a variant id</h2><span id='topic+association_to_variant'></span>

<h3>Description</h3>

<p>Map an association accession identifier to a variant identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>association_to_variant(association_id, verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="association_to_variant_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of association accession
identifiers.</p>
</td></tr>
<tr><td><code id="association_to_variant_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="association_to_variant_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two identifiers. First column is the association
identifier and the second column is the variant identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map GWAS association identifiers to variant identifiers
association_to_variant(c('24300097', '24299759'))

## End(Not run)

</code></pre>

<hr>
<h2 id='associations'>Constructor for the S4 associations object.</h2><span id='topic+associations'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+associations-class">associations</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>associations(
  associations = associations_tbl(),
  loci = loci_tbl(),
  risk_alleles = risk_alleles_tbl(),
  genes = reported_genes_tbl(),
  ensembl_ids = ensembl_ids_tbl(),
  entrez_ids = entrez_ids_tbl()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="associations_+3A_associations">associations</code></td>
<td>
<p>An <code><a href="#topic+associations_tbl">associations_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="associations_+3A_loci">loci</code></td>
<td>
<p>A <code><a href="#topic+loci_tbl">loci_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="associations_+3A_risk_alleles">risk_alleles</code></td>
<td>
<p>A <code><a href="#topic+risk_alleles_tbl">risk_alleles_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="associations_+3A_genes">genes</code></td>
<td>
<p>A <code><a href="#topic+reported_genes_tbl">reported_genes_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="associations_+3A_ensembl_ids">ensembl_ids</code></td>
<td>
<p>A <code><a href="#topic+ensembl_ids_tbl">ensembl_ids_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="associations_+3A_entrez_ids">entrez_ids</code></td>
<td>
<p>A <code><a href="#topic+entrez_ids_tbl">entrez_ids_tbl</a></code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+associations-class">associations</a>.
</p>

<hr>
<h2 id='associations_drop_na'>Drop any NA associations.</h2><span id='topic+associations_drop_na'></span>

<h3>Description</h3>

<p>This function takes an associations S4 object and removes any association
identifiers that might have been NA. This ensures that there is always a
non-NA <code>association_id</code> value in all tables. This is important as the
<code>association_id</code> is the primary key.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>associations_drop_na(s4_associations)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="associations_drop_na_+3A_s4_associations">s4_associations</code></td>
<td>
<p>An object of class <a href="#topic+associations-class">associations</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+associations-class">associations</a>.
</p>

<hr>
<h2 id='associations_tbl'>Creates an associations table.</h2><span id='topic+associations_tbl'></span>

<h3>Description</h3>

<p>Creates an associations table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>associations_tbl(
  association_id = character(),
  pvalue = double(),
  pvalue_description = character(),
  pvalue_mantissa = integer(),
  pvalue_exponent = integer(),
  multiple_snp_haplotype = logical(),
  snp_interaction = logical(),
  snp_type = character(),
  risk_frequency = double(),
  standard_error = double(),
  range = character(),
  or_per_copy_number = double(),
  beta_number = double(),
  beta_unit = character(),
  beta_direction = character(),
  beta_description = character(),
  last_mapping_date = lubridate::ymd_hms(),
  last_update_date = lubridate::ymd_hms()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="associations_tbl_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of association identifiers.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_pvalue">pvalue</code></td>
<td>
<p>A numeric vector of p-values.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_pvalue_description">pvalue_description</code></td>
<td>
<p>A character vector of p-value context descriptions.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_pvalue_mantissa">pvalue_mantissa</code></td>
<td>
<p>An integer vector of p-value mantissas.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_pvalue_exponent">pvalue_exponent</code></td>
<td>
<p>An integer vector of p-value exponents.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_multiple_snp_haplotype">multiple_snp_haplotype</code></td>
<td>
<p>A logical vector.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_snp_interaction">snp_interaction</code></td>
<td>
<p>A logical vector.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_snp_type">snp_type</code></td>
<td>
<p>A character vector indicating SNP novelty: 'novel' or
'known'.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_standard_error">standard_error</code></td>
<td>
<p>A numeric vector of standard errors.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_range">range</code></td>
<td>
<p>A character vector of free text descriptions of confidence
intervals.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_or_per_copy_number">or_per_copy_number</code></td>
<td>
<p>A numeric vector of odds ratios.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_beta_number">beta_number</code></td>
<td>
<p>A numeric vector of beta coefficients.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_beta_unit">beta_unit</code></td>
<td>
<p>A character vector of beta coefficient units.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_beta_direction">beta_direction</code></td>
<td>
<p>A character vector of beta coefficient directions.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_beta_description">beta_description</code></td>
<td>
<p>A character vector of beta descriptions.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_last_mapping_date">last_mapping_date</code></td>
<td>
<p>A <code><a href="base.html#topic+DateTimeClasses">POSIXct</a></code>
object indicating last time this association was mapped to Ensembl.</p>
</td></tr>
<tr><td><code id="associations_tbl_+3A_last_update_date">last_update_date</code></td>
<td>
<p>A <code><a href="base.html#topic+DateTimeClasses">POSIXct</a></code> object
indicating the last time the associations have been updated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='associations-class'>An S4 class to represent a set of GWAS Catalog associations</h2><span id='topic+associations-class'></span>

<h3>Description</h3>

<p>The association object consists of six slots, each a table
(<code><a href="tibble.html#topic+tibble">tibble</a></code>), that combined form a relational database of a
subset of GWAS Catalog associations. Each association is an observation (row)
in the <code>associations</code> table &mdash; main table. All tables have the column
<code>association_id</code> as primary key.
</p>


<h3>Slots</h3>


<dl>
<dt><code>associations</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing associations.
Columns:
</p>

<dl>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>pvalue</dt><dd><p>Reported p-value for strongest variant risk or effect allele.</p>
</dd>
<dt>pvalue_description</dt><dd><p>Information describing context of p-value.</p>
</dd>
<dt>pvalue_mantissa</dt><dd><p>Mantissa of p-value.</p>
</dd>
<dt>pvalue_exponent</dt><dd><p>Exponent of p-value.</p>
</dd>
<dt>multiple_snp_haplotype</dt><dd><p>Whether the association is for a multi-SNP
haplotype.</p>
</dd>
<dt>snp_interaction</dt><dd><p>Whether the association is for a SNP-SNP interaction.</p>
</dd>
<dt>snp_type</dt><dd><p>Whether the SNP has previously been reported. Either
<code>'known'</code> or <code>'novel'</code>.</p>
</dd>
<dt>risk_frequency</dt><dd><p>Reported risk/effect allele frequency associated with
strongest SNP in controls.</p>
</dd>
<dt>standard_error</dt><dd><p>Standard error of the effect size.</p>
</dd>
<dt>range</dt><dd><p>Reported 95% confidence interval associated with strongest SNP
risk allele, along with unit in the case of beta coefficients. If 95% CIs
have not been not reported, these are estimated using the standard error,
when available.</p>
</dd>
<dt>or_per_copy_number</dt><dd><p>Reported odds ratio (OR) associated
with strongest SNP risk allele. Note that all ORs included in the
Catalog are &gt;1.</p>
</dd>
<dt>beta_number</dt><dd><p>Beta coefficient associated with strongest SNP risk
allele.</p>
</dd>
<dt>beta_unit</dt><dd><p>Beta coefficient unit.</p>
</dd>
<dt>beta_direction</dt><dd><p>Beta coefficient direction, either <code>'decrease'</code> or
<code>'increase'</code>.</p>
</dd>
<dt>beta_description</dt><dd><p>Additional beta coefficient comment.</p>
</dd>
<dt>last_mapping_date</dt><dd><p>Last time this association was mapped to Ensembl.</p>
</dd>
<dt>last_update_date</dt><dd><p>Last time this association was updated.</p>
</dd>
</dl>
</dd>
<dt><code>loci</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing loci. Columns:
</p>

<dl>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>locus_id</dt><dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p>
</dd>
<dt>haplotype_snp_count</dt><dd><p>Number of variants per locus. Most loci are
single-SNP loci, i.e., there is a one to one relationship between a variant
and a <code>locus_id</code> (<code>haplotype_snp_count == NA</code>). There are however
cases of associations involving multiple loci at once, such as SNP-SNP
interactions and multi-SNP haplotypes. This is signalled in the columns:
<code>multiple_snp_haplotype</code> and <code>snp_interaction</code> with value
<code>TRUE</code>.</p>
</dd>
<dt>description</dt><dd><p>Description of the locus identifier, e.g.,
<code>'Single variant'</code>, <code>SNP x SNP interaction</code>, or <code>3-SNP
Haplotype</code>.</p>
</dd>
</dl>
</dd>
<dt><code>risk_alleles</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing risk alleles.
Columns:
</p>

<dl>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>locus_id</dt><dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p>
</dd>
<dt>variant_id</dt><dd><p>Variant identifier, e.g., <code>'rs1333048'</code>.</p>
</dd>
<dt>risk_allele</dt><dd><p>Risk allele or effect allele.</p>
</dd>
<dt>risk_frequency</dt><dd><p>Reported risk/effect allele frequency associated with
strongest SNP in controls (if not available among all controls, among the
control group with the largest sample size). If the associated locus is a
haplotype the haplotype frequency will be extracted.</p>
</dd>
<dt>genome_wide</dt><dd><p>Whether this variant allele has been part of a genome-wide
study or not.</p>
</dd>
<dt>limited_list</dt><dd><p>Undocumented.</p>
</dd>
</dl>
</dd>
<dt><code>genes</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing author reported genes.
Columns:
</p>

<dl>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>locus_id</dt><dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p>
</dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</dd> </dl>
</dd>
<dt><code>ensembl_ids</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing Ensembl gene
identifiers. Columns:
</p>

<dl>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>locus_id</dt><dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p>
</dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</dd>
<dt>ensembl_id</dt><dd><p>The Ensembl identifier of an Ensembl gene, see Section
<a href="https://www.ensembl.org/info/genome/genebuild/index.html">Gene
annotation in Ensembl</a> for more information.</p>
</dd> </dl>
</dd>
<dt><code>entrez_ids</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing Entrez gene
identifiers. Columns:
</p>

<dl>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>locus_id</dt><dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p>
</dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</dd>
<dt>entrez_id</dt><dd><p>The Entrez identifier of a gene, see ref.
<a href="https://doi.org/10.1093/nar/gkq1237">doi:10.1093/nar/gkq1237</a> for
more information.</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='bind'>Bind GWAS Catalog objects</h2><span id='topic+bind'></span>

<h3>Description</h3>

<p>Binds together GWAS Catalog objects of the same class. Note that
<code>bind()</code> preserves duplicates whereas
<code><a href="#topic+setop">union</a></code> does not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bind(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bind_+3A_x">x</code></td>
<td>
<p>An object of class: <a href="#topic+studies-class">studies</a>,
<a href="#topic+associations-class">associations</a>, <a href="#topic+variants-class">variants</a>, or
<a href="#topic+traits-class">traits</a>.</p>
</td></tr>
<tr><td><code id="bind_+3A_...">...</code></td>
<td>
<p>Objects of the same class as <code>x</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same class as <code>x</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Join two studies objects.
bind(studies_ex01, studies_ex02)

# Join two associations objects.
bind(associations_ex01, associations_ex02)

# Join two variants objects.
bind(variants_ex01, variants_ex02)

# Join two traits objects.
bind(traits_ex01, traits_ex02)

</code></pre>

<hr>
<h2 id='child_efo_ids'>Get all child terms of this trait in the EFO hierarchy</h2><span id='topic+child_efo_ids'></span>

<h3>Description</h3>

<p>Get all child terms of this trait in the EFO hierarchy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>child_efo_ids(
  efo_id,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="child_efo_ids_+3A_efo_id">efo_id</code></td>
<td>
<p>A <a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier.</p>
</td></tr>
<tr><td><code id="child_efo_ids_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="child_efo_ids_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="child_efo_ids_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
<tr><td><code id="child_efo_ids_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the paginated resources
are retrieved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of EFO identifiers.
</p>

<hr>
<h2 id='contains_question_mark'>Does a string contain a question mark?</h2><span id='topic+contains_question_mark'></span>

<h3>Description</h3>

<p>Find which strings contain a question mark. This function uses the following
regular expression: <code>[\?]</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contains_question_mark(str, convert_NA_to_FALSE = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="contains_question_mark_+3A_str">str</code></td>
<td>
<p>A character vector of strings.</p>
</td></tr>
<tr><td><code id="contains_question_mark_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to treat <code>NA</code> as <code>NA</code>
(<code>convert_NA_to_FALSE = FALSE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector.
</p>

<hr>
<h2 id='countries_tbl'>Creates a countries table.</h2><span id='topic+countries_tbl'></span>

<h3>Description</h3>

<p>Creates a countries table. This
function is used internally to create both the <code>countries_of_origin</code> and
<code>countries_of_recruitment</code> slots of a <a href="#topic+studies-class">studies</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countries_tbl(
  study_id = character(),
  ancestry_id = integer(),
  country_name = character(),
  major_area = character(),
  region = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="countries_tbl_+3A_study_id">study_id</code></td>
<td>
<p>GWAS Catalog study accession identifier.</p>
</td></tr>
<tr><td><code id="countries_tbl_+3A_ancestry_id">ancestry_id</code></td>
<td>
<p>Ancestry identifier.</p>
</td></tr>
<tr><td><code id="countries_tbl_+3A_country_name">country_name</code></td>
<td>
<p>Country name, according to
<a href="https://unstats.un.org/unsd/methodology/m49/overview/">The United
Nations M49 Standard of Geographic Regions.</a></p>
</td></tr>
<tr><td><code id="countries_tbl_+3A_major_area">major_area</code></td>
<td>
<p>Region name, according to
<a href="https://unstats.un.org/unsd/methodology/m49/overview/">The United
Nations M49 Standard of Geographic Regions.</a></p>
</td></tr>
<tr><td><code id="countries_tbl_+3A_region">region</code></td>
<td>
<p>Sub-region name, according to
<a href="https://unstats.un.org/unsd/methodology/m49/overview/">The United
Nations M49 Standard of Geographic Regions.</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='cytogenetic_band_to_genomic_range'>Convert a cytogenetic band string to genomic coordinates.</h2><span id='topic+cytogenetic_band_to_genomic_range'></span>

<h3>Description</h3>

<p>This function uses the provided <code><a href="#topic+cytogenetic_bands">cytogenetic_bands</a></code>
dataframe to convert cytogenetic band names to genomic coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cytogenetic_band_to_genomic_range(bands)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cytogenetic_band_to_genomic_range_+3A_bands">bands</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> vector of cytogenetic bands.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of genomic ranges. Columns are: chromosome, start and
end. Each row corresponds to a queried cytogenetic band (in the same order
as queried).
</p>

<hr>
<h2 id='cytogenetic_bands'>GRCh38 human cytogenetic bands.</h2><span id='topic+cytogenetic_bands'></span>

<h3>Description</h3>

<p>A dataset containing the GRCh38 human cytogenetic bands and their genomic
coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cytogenetic_bands
</code></pre>


<h3>Format</h3>

<p>A data frame with 862 rows and 8 variables:
</p>

<dl>
<dt>cytogenetic_band</dt><dd><p>Cytogenetic band name. See <em>Cytogenetic
Nomenclature</em> below.</p>
</dd>
<dt>chromosome</dt><dd><p>Chromosome name: 1 through 22 (the autosomes), X or
Y.</p>
</dd>
<dt>start</dt><dd><p>Genomic start position of the cytogenetic band. Starts at 1.</p>
</dd>
<dt>end</dt><dd><p>Genomic end position of the cytogenetic band. End position is
included in the band interval.</p>
</dd>
<dt>length</dt><dd><p>Length of the genomic interval of cytogenetic band.</p>
</dd>
<dt>assembly</dt><dd><p>Assembly version, should be 'GRCh38'.</p>
</dd>
<dt>stain</dt><dd><p><a href="https://en.wikipedia.org/wiki/Giemsa_stain">Giemsa
stain</a> results: Giemsa negative, <code>'gneg'</code>; Giemsa positive, of
increasing intensities, <code>'gpos25'</code>, <code>'gpos50'</code>, <code>'gpos75'</code>,
and <code>'gpos100'</code>; centromeric region, <code>'acen'</code>; heterochromatin,
either pericentric or telomeric, <code>'gvar'</code>; and short arm of
acrocentric chromosomes 13, 14, 15, 21, and 22 are coded as
<code>'stalk'</code>.</p>
</dd>
<dt>last_download_date</dt><dd><p>Time stamp of last time this dataset was
downloaded from Ensembl.</p>
</dd>
</dl>



<h3>Details</h3>

<p>Genomic coordinates are for
<a href="https://genome-blog.soe.ucsc.edu/blog/wp-content/uploads/2016/12/newInterval.png">fully
closed</a> intervals.
</p>


<h3>Cytogenetic Nomenclature</h3>

<p>Cytogenetic bands are numbered from the centromere outwards in both
directions towards the telomeres on the shorter p arm and the longer q arm.
</p>
<p>The first number or letter represents the chromosome. Chromosomes 1 through
22 (the autosomes) are designated by their chromosome number. The sex
chromosomes are designated by X or Y. The next letter represents the arm of
the chromosome: p or q.
</p>
<p>The numbers cannot be read in the normal decimal numeric system e.g. 36, but
rather 3-6 (region 3 band 6). Counting starts at the centromere as region 1
(or 1-0), to 11 (1-1) to 21 (2-1) to 22 (2-2) etc. Subbands are added in a
similar way, e.g. 21.1 to 21.2, if the bands are small or only appear at a
higher resolution.
</p>


<h3>Source</h3>

<p><a href="https://rest.ensembl.org/info/assembly/homo_sapiens?content-type=application/json&amp;bands=1">https://rest.ensembl.org/info/assembly/homo_sapiens?content-type=application/json&amp;bands=1</a>
</p>

<hr>
<h2 id='empty_to_na'>Converts an empty vector to a scalar NA</h2><span id='topic+empty_to_na'></span>

<h3>Description</h3>

<p>This function converts an empty vector to a scalar NA of the same type as the
input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>empty_to_na(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="empty_to_na_+3A_x">x</code></td>
<td>
<p>An atomic vector, of one of these types: <code>character</code>,
<code>integer</code>, <code>double</code> or <code>logical</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An atomic vector of the same type as <code>x</code>. If <code>x</code> is empty
then <code>NA</code> is returned, otherwise <code>x</code> is returned as is.
</p>

<hr>
<h2 id='ensembl_ids_tbl'>Creates an Ensembl gene identifiers' table.</h2><span id='topic+ensembl_ids_tbl'></span>

<h3>Description</h3>

<p>Creates an Ensembl gene identifiers' table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ensembl_ids_tbl(
  association_id = character(),
  locus_id = integer(),
  gene_name = character(),
  ensembl_id = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ensembl_ids_tbl_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of association identifiers.</p>
</td></tr>
<tr><td><code id="ensembl_ids_tbl_+3A_locus_id">locus_id</code></td>
<td>
<p>An integer vector of locus identifiers.</p>
</td></tr>
<tr><td><code id="ensembl_ids_tbl_+3A_gene_name">gene_name</code></td>
<td>
<p>A character vector of gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</td></tr>
<tr><td><code id="ensembl_ids_tbl_+3A_ensembl_id">ensembl_id</code></td>
<td>
<p>A character vector of Ensembl identifiers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='entrez_ids_tbl'>Creates an Entrez gene identifiers' table.</h2><span id='topic+entrez_ids_tbl'></span>

<h3>Description</h3>

<p>Creates an Entrez gene identifiers' table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_ids_tbl(
  association_id = character(),
  locus_id = integer(),
  gene_name = character(),
  entrez_id = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="entrez_ids_tbl_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of association identifiers.</p>
</td></tr>
<tr><td><code id="entrez_ids_tbl_+3A_locus_id">locus_id</code></td>
<td>
<p>An integer vector of locus identifiers.</p>
</td></tr>
<tr><td><code id="entrez_ids_tbl_+3A_gene_name">gene_name</code></td>
<td>
<p>A character vector of gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</td></tr>
<tr><td><code id="entrez_ids_tbl_+3A_entrez_id">entrez_id</code></td>
<td>
<p>A character vector of Entrez identifiers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='equal_length'>Grows vectors to match longest vector</h2><span id='topic+equal_length'></span>

<h3>Description</h3>

<p>This function determines the longest vector and pads the shorter ones by
adding NAs until they match the longest vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>equal_length(lst_of_vectors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="equal_length_+3A_lst_of_vectors">lst_of_vectors</code></td>
<td>
<p>A list of atomic vectors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of atomic vectors of the same length.
</p>

<hr>
<h2 id='exists_variant'>Check if a variant exists in the Catalog.</h2><span id='topic+exists_variant'></span>

<h3>Description</h3>

<p>This function attempts to get a variant by its variant identifier and checks
the response code. If the response code is 200 then the response has been
successful, meaning that the variant does exist in the GWAS Catalog. If the
response is 404 then the variant is not found in the Catalog database. Other
errors are mapped to NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exists_variant(variant_id = NULL, verbose = FALSE, page_size = 20L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exists_variant_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of GWAS Catalog variant identifiers.</p>
</td></tr>
<tr><td><code id="exists_variant_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="exists_variant_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named logical vector, <code>TRUE</code> indicates that the variant does
exist in the Catalog, <code>FALSE</code> otherwise. <code>NA</code> codes other types
of errors. The names of the vector are the variant identifiers passed as
<code>variant_id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exists_variant('rs12345')

exists_variant('rs11235813')

</code></pre>

<hr>
<h2 id='extract_association_id'>Extract association identifiers from URLs</h2><span id='topic+extract_association_id'></span>

<h3>Description</h3>

<p>This function extracts association identifiers from URLs of the form:
<code>".*{association_id}$"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_association_id(urls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract_association_id_+3A_urls">urls</code></td>
<td>
<p>A character vector of URLs of the form
<code>".*{association_id}$"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of association identifiers.
</p>

<hr>
<h2 id='filter_by_id'>Filter GWAS Catalog objects by identifier.</h2><span id='topic+filter_by_id'></span>

<h3>Description</h3>

<p>Use <code>filter_by_id</code> to filter GWAS Catalog objects by their respective
identifier (<code>id</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_by_id(x, id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_by_id_+3A_x">x</code></td>
<td>
<p>An object of class either <a href="#topic+studies-class">studies</a>,
<a href="#topic+associations-class">associations</a>, <a href="#topic+variants-class">variants</a>, or
<a href="#topic+traits-class">traits</a>.</p>
</td></tr>
<tr><td><code id="filter_by_id_+3A_id">id</code></td>
<td>
<p>Identifier.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class either <a href="#topic+studies-class">studies</a>,
<a href="#topic+associations-class">associations</a>, <a href="#topic+variants-class">variants</a>, or
<a href="#topic+traits-class">traits</a>.
</p>

<hr>
<h2 id='filter_variants_by_standard_chromosomes'>Filter variants by standard human chromosomes.</h2><span id='topic+filter_variants_by_standard_chromosomes'></span>

<h3>Description</h3>

<p>This function filters a <a href="#topic+variants-class">variants</a> object by standard human chromosomes, i.e.,
1&ndash;22, X and Y. In addition to these chromosomes, some variants retrieved
from the GWAS Catalog might be also mapped to non-standard locations, such as
GRC assembly patches, haplotype (HAPs) or pseudo autosomal regions (PARs).
When this happens the main table <code>variants</code> includes rows for these
cases too. This function removes these.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_variants_by_standard_chromosomes(
  s4_variants,
  chromosomes = c(seq_len(22), "X", "Y", "MT")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_variants_by_standard_chromosomes_+3A_s4_variants">s4_variants</code></td>
<td>
<p>An object of class <a href="#topic+variants-class">variants</a>.</p>
</td></tr>
<tr><td><code id="filter_variants_by_standard_chromosomes_+3A_chromosomes">chromosomes</code></td>
<td>
<p>A character vector of valid chromosome names. Default is
autosomal chromosomes 1 thru 22 and, X, Y, and MT.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+variants-class">variants</a>.
</p>

<hr>
<h2 id='gc_examples'>gwasrapidd entities' examples</h2><span id='topic+gc_examples'></span><span id='topic+studies_ex01'></span><span id='topic+studies_ex02'></span><span id='topic+associations_ex01'></span><span id='topic+associations_ex02'></span><span id='topic+variants_ex01'></span><span id='topic+variants_ex02'></span><span id='topic+traits_ex01'></span><span id='topic+traits_ex02'></span>

<h3>Description</h3>

<p>These are examples of GWAS Catalog entities shipped with gwasrapidd:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>studies_ex01

studies_ex02

associations_ex01

associations_ex02

variants_ex01

variants_ex02

traits_ex01

traits_ex02
</code></pre>


<h3>Format</h3>


<dl>
<dt>studies_ex01</dt><dd><p>An S4 <a href="#topic+studies-class">studies</a> object of 2 studies:
<code>'GCST001585'</code> and <code>'GCST003985'</code>.</p>
</dd>
<dt>studies_ex02</dt><dd><p>An S4 <a href="#topic+studies-class">studies</a> object of 2 studies:
<code>'GCST001585'</code> and <code>'GCST006655'</code>.</p>
</dd>
<dt>associations_ex01</dt><dd><p>An S4 <a href="#topic+associations-class">associations</a> object of 4
associations: <code>'22509'</code>, <code>'22505'</code>, <code>'19537565'</code> and
<code>'19537593'</code>.</p>
</dd>
<dt>associations_ex02</dt><dd><p>An S4 <a href="#topic+associations-class">associations</a> object of 3
associations: <code>'19537593'</code>, <code>'31665940'</code> and <code>'34944736'</code>.</p>
</dd>
<dt>variants_ex01</dt><dd><p>An S4 <a href="#topic+variants-class">variants</a> object of 3 variants:
<code>'rs146992477'</code>, <code>'rs56261590'</code> and <code>'rs4725504'</code>.</p>
</dd>
<dt>variants_ex02</dt><dd><p>An S4 <a href="#topic+variants-class">variants</a> object of 4 variants:
<code>'rs56261590'</code>, <code>'rs4725504'</code>, <code>'rs11099757'</code> and
<code>'rs16871509'</code>.</p>
</dd>
<dt>traits_ex01</dt><dd><p>An S4 <a href="#topic+traits-class">traits</a> object of 3 traits:
<code>'EFO_0004884'</code>, <code>'EFO_0004343'</code> and <code>'EFO_0005299'</code>.</p>
</dd>
<dt>traits_ex02</dt><dd><p>An S4 <a href="#topic+traits-class">traits</a> object of 4 traits:
<code>'EFO_0007845'</code>, <code>'EFO_0004699'</code>, <code>'EFO_0004884'</code> and
<code>'EFO_0004875'</code>.</p>
</dd>
</dl>

<p>An object of class <code>studies</code> of length 1.
</p>
<p>An object of class <code>associations</code> of length 1.
</p>
<p>An object of class <code>associations</code> of length 1.
</p>
<p>An object of class <code>variants</code> of length 1.
</p>
<p>An object of class <code>variants</code> of length 1.
</p>
<p>An object of class <code>traits</code> of length 1.
</p>
<p>An object of class <code>traits</code> of length 1.
</p>

<hr>
<h2 id='gc_get'>Get a GWAS Catalog resource</h2><span id='topic+gc_get'></span>

<h3>Description</h3>

<p>This function gets a GWAS Catalog by URL endpoint. The response must
correspond to one of the four types of entities: studies, associations,
variants or traits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gc_get(
  resource_url,
  base_url = gwas_rest_api_base_url,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gc_get_+3A_resource_url">resource_url</code></td>
<td>
<p>Endpoint URL. The endpoint is internally appended to the
<code>base_url</code>. It should start with a forward slash (<code>/</code>).</p>
</td></tr>
<tr><td><code id="gc_get_+3A_base_url">base_url</code></td>
<td>
<p>The GWAS REST API base URL (one should not need to change its
default value).</p>
</td></tr>
<tr><td><code id="gc_get_+3A_verbose">verbose</code></td>
<td>
<p>Whether to be chatty.</p>
</td></tr>
<tr><td><code id="gc_get_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="gc_get_+3A_page_size">page_size</code></td>
<td>
<p>Page parameter used in the URL endpoint.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A normalised JSON-list corresponding to either studies, associations,
variants or traits.
</p>

<hr>
<h2 id='gc_request'>Request a GWAS Catalog REST API endpoint</h2><span id='topic+gc_request'></span>

<h3>Description</h3>

<p>Performs a GET request on the specified <code>resource_url</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gc_request(
  resource_url = "/",
  base_url = gwas_rest_api_base_url,
  verbose = FALSE,
  warnings = TRUE,
  flatten = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gc_request_+3A_resource_url">resource_url</code></td>
<td>
<p>Endpoint URL. The endpoint is internally appended to the
<code>base_url</code>. It should start with a forward slash (<code>/</code>).</p>
</td></tr>
<tr><td><code id="gc_request_+3A_base_url">base_url</code></td>
<td>
<p>The GWAS REST API base URL (one should not need to change its
default value).</p>
</td></tr>
<tr><td><code id="gc_request_+3A_verbose">verbose</code></td>
<td>
<p>Whether to be chatty.</p>
</td></tr>
<tr><td><code id="gc_request_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="gc_request_+3A_flatten">flatten</code></td>
<td>
<p>Whether to flatten out the list returned by
<code>jsonlite::fromJSON</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list four named elements:
</p>

<dl>
<dt>url</dt><dd><p>The URL endpoint.</p>
</dd>
<dt>response_code</dt><dd><p><a href="https://en.wikipedia.org/wiki/List_of_HTTP_status_codes">HTTP
status code</a>.</p>
</dd>
<dt>status</dt><dd><p>A string describing the status of the response obtained. It is
&quot;OK&quot; if everything went OK or some other string describing the problem
otherwise.</p>
</dd>
<dt>content</dt><dd><p>The parsed JSON as a nested list, as returned by
<code>jsonlite::fromJSON</code>.</p>
</dd>
</dl>


<hr>
<h2 id='gc_request_all'>Request a paginated GWAS Catalog REST API endpoint</h2><span id='topic+gc_request_all'></span>

<h3>Description</h3>

<p>Performs a GET request on the specified <code>resource_url</code> and all its
pages.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gc_request_all(
  resource_url = "/",
  base_url = gwas_rest_api_base_url,
  page_size = 20L,
  verbose = FALSE,
  warnings = TRUE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gc_request_all_+3A_resource_url">resource_url</code></td>
<td>
<p>Endpoint URL. The endpoint is internally appended to the
<code>base_url</code>. It should start with a forward slash (<code>/</code>).</p>
</td></tr>
<tr><td><code id="gc_request_all_+3A_base_url">base_url</code></td>
<td>
<p>The GWAS REST API base URL (one should not need to change its
default value).</p>
</td></tr>
<tr><td><code id="gc_request_all_+3A_page_size">page_size</code></td>
<td>
<p>Page parameter used in the URL endpoint.</p>
</td></tr>
<tr><td><code id="gc_request_all_+3A_verbose">verbose</code></td>
<td>
<p>Whether to be chatty.</p>
</td></tr>
<tr><td><code id="gc_request_all_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="gc_request_all_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the paginated resources
are retrieved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list four named elements:
</p>

<dl>
<dt>url</dt><dd><p>The URL endpoint.</p>
</dd>
<dt>response_code</dt><dd><p><a href="https://tinyurl.com/8yqvhwf">HTTP
status code</a>.</p>
</dd>
<dt>status</dt><dd><p>A string describing the status of the response obtained. It is
&quot;OK&quot; if everything went OK or some other string describing the problem
otherwise.</p>
</dd>
<dt>content</dt><dd><p>The parsed JSON as a nested list, as returned by
<code>jsonlite::fromJSON</code>.</p>
</dd>
</dl>


<hr>
<h2 id='genomic_contexts_tbl'>Creates a genomic contexts table.</h2><span id='topic+genomic_contexts_tbl'></span>

<h3>Description</h3>

<p>Creates a genomic contexts table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomic_contexts_tbl(
  variant_id = character(),
  gene_name = character(),
  chromosome_name = character(),
  chromosome_position = integer(),
  distance = integer(),
  is_mapped_gene = logical(),
  is_closest_gene = logical(),
  is_intergenic = logical(),
  is_upstream = logical(),
  is_downstream = logical(),
  source = character(),
  mapping_method = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genomic_contexts_tbl_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of variant identifiers.</p>
</td></tr>
<tr><td><code id="genomic_contexts_tbl_+3A_gene_name">gene_name</code></td>
<td>
<p>A character vector of gene symbols according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</td></tr>
<tr><td><code id="genomic_contexts_tbl_+3A_chromosome_name">chromosome_name</code></td>
<td>
<p>A character vector of chromosome names.</p>
</td></tr>
<tr><td><code id="genomic_contexts_tbl_+3A_chromosome_position">chromosome_position</code></td>
<td>
<p>An integer vector of chromosome positions.</p>
</td></tr>
<tr><td><code id="genomic_contexts_tbl_+3A_distance">distance</code></td>
<td>
<p>An integer vector of genomic positions.</p>
</td></tr>
<tr><td><code id="genomic_contexts_tbl_+3A_is_closest_gene">is_closest_gene</code></td>
<td>
<p>A logical vector.</p>
</td></tr>
<tr><td><code id="genomic_contexts_tbl_+3A_is_intergenic">is_intergenic</code></td>
<td>
<p>A logical vector.</p>
</td></tr>
<tr><td><code id="genomic_contexts_tbl_+3A_is_upstream">is_upstream</code></td>
<td>
<p>A logical vector.</p>
</td></tr>
<tr><td><code id="genomic_contexts_tbl_+3A_is_downstream">is_downstream</code></td>
<td>
<p>A logical vector.</p>
</td></tr>
<tr><td><code id="genomic_contexts_tbl_+3A_source">source</code></td>
<td>
<p>A character vector of gene mapping sources.</p>
</td></tr>
<tr><td><code id="genomic_contexts_tbl_+3A_mapping_method">mapping_method</code></td>
<td>
<p>A character vector of gene mapping methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='genotyping_techs_tbl'>Creates a genotyping technologies table.</h2><span id='topic+genotyping_techs_tbl'></span>

<h3>Description</h3>

<p>Creates a genotyping technologies table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotyping_techs_tbl(
  study_id = character(),
  genotyping_technology = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genotyping_techs_tbl_+3A_study_id">study_id</code></td>
<td>
<p>GWAS Catalog study accession identifier.</p>
</td></tr>
<tr><td><code id="genotyping_techs_tbl_+3A_genotyping_technology">genotyping_technology</code></td>
<td>
<p>Genotyping technology.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='get_associations'>Get GWAS Catalog associations</h2><span id='topic+get_associations'></span>

<h3>Description</h3>

<p>Retrieves associations via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all associations that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
associations that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_associations(
  study_id = NULL,
  association_id = NULL,
  variant_id = NULL,
  efo_id = NULL,
  pubmed_id = NULL,
  efo_trait = NULL,
  set_operation = "union",
  interactive = TRUE,
  verbose = FALSE,
  warnings = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_associations_+3A_study_id">study_id</code></td>
<td>
<p>A <code>character</code> vector of GWAS Catalog study accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_associations_+3A_association_id">association_id</code></td>
<td>
<p>A <code>character</code> vector of GWAS Catalog association
identifiers.</p>
</td></tr>
<tr><td><code id="get_associations_+3A_variant_id">variant_id</code></td>
<td>
<p>A <code>character</code> vector of GWAS Catalog variant
identifiers.</p>
</td></tr>
<tr><td><code id="get_associations_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a>
identifiers.</p>
</td></tr>
<tr><td><code id="get_associations_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>An <code>integer</code> vector of
<a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a> identifiers.</p>
</td></tr>
<tr><td><code id="get_associations_+3A_efo_trait">efo_trait</code></td>
<td>
<p>A <code>character</code> vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p>
</td></tr>
<tr><td><code id="get_associations_+3A_set_operation">set_operation</code></td>
<td>
<p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how associations retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same associations found with different
criteria.</p>
</td></tr>
<tr><td><code id="get_associations_+3A_interactive">interactive</code></td>
<td>
<p>A logical. If all associations are requested, whether to ask
interactively if we really want to proceed.</p>
</td></tr>
<tr><td><code id="get_associations_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_associations_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note that all search criteria are vectorised, thus allowing for batch
mode search, e.g., one can search by multiple variant identifiers at once by
passing a vector of identifiers to <code>variant_id</code>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+associations-class">associations</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get an association by study identifier
get_associations(study_id = 'GCST001085', warnings = FALSE)

# Get an association by association identifier
get_associations(association_id = '25389945', warnings = FALSE)

# Get associations by variant identifier
get_associations(variant_id = 'rs3798440', warnings = FALSE)

# Get associations by EFO trait identifier
get_associations(efo_id = 'EFO_0005537', warnings = FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='get_associations_all'>Get all GWAS Catalog associations</h2><span id='topic+get_associations_all'></span>

<h3>Description</h3>

<p>Gets all associations. Beware this can take a few hours!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_associations_all(verbose = FALSE, warnings = TRUE, page_size = 20L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_associations_all_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_associations_all_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_associations_all_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+associations-class">associations</a> object.
</p>

<hr>
<h2 id='get_associations_by_association_id'>Get GWAS Catalog associations by their association identifiers</h2><span id='topic+get_associations_by_association_id'></span>

<h3>Description</h3>

<p>Gets associations by GWAS Catalog internal association identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_associations_by_association_id(
  association_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_associations_by_association_id_+3A_association_id">association_id</code></td>
<td>
<p>A <code>character</code> vector of GWAS Catalog association
identifiers.</p>
</td></tr>
<tr><td><code id="get_associations_by_association_id_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_associations_by_association_id_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_associations_by_association_id_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+associations-class">associations</a> object.
</p>

<hr>
<h2 id='get_associations_by_efo_id'>Get GWAS Catalog associations by EFO identifier</h2><span id='topic+get_associations_by_efo_id'></span>

<h3>Description</h3>

<p>Gets associations whose phenotypic trait is matched by EFO identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_associations_by_efo_id(
  efo_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_associations_by_efo_id_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a>
identifiers.</p>
</td></tr>
<tr><td><code id="get_associations_by_efo_id_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_associations_by_efo_id_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_associations_by_efo_id_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+associations-class">associations</a> object.
</p>

<hr>
<h2 id='get_associations_by_efo_trait'>Get GWAS Catalog associations by EFO traits</h2><span id='topic+get_associations_by_efo_trait'></span>

<h3>Description</h3>

<p>Gets associations that match <a href="https://www.ebi.ac.uk/efo/">EFO</a> trait
description.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_associations_by_efo_trait(
  efo_trait = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_associations_by_efo_trait_+3A_efo_trait">efo_trait</code></td>
<td>
<p>A <code>character</code> vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p>
</td></tr>
<tr><td><code id="get_associations_by_efo_trait_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_associations_by_efo_trait_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_associations_by_efo_trait_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+associations-class">associations</a> object.
</p>

<hr>
<h2 id='get_associations_by_pubmed_id'>Get GWAS Catalog associations by PubMed identifiers</h2><span id='topic+get_associations_by_pubmed_id'></span>

<h3>Description</h3>

<p>Gets associations whose associated publications match
<a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a> identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_associations_by_pubmed_id(
  pubmed_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_associations_by_pubmed_id_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>An <code>integer</code> vector of
<a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a> identifiers.</p>
</td></tr>
<tr><td><code id="get_associations_by_pubmed_id_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_associations_by_pubmed_id_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_associations_by_pubmed_id_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+associations-class">associations</a> object.
</p>

<hr>
<h2 id='get_associations_by_study_id'>Get GWAS Catalog associations by study identifiers</h2><span id='topic+get_associations_by_study_id'></span>

<h3>Description</h3>

<p>Gets associations by GWAS Catalog internal study identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_associations_by_study_id(
  study_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_associations_by_study_id_+3A_study_id">study_id</code></td>
<td>
<p>A <code>character</code> vector of GWAS Catalog study accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_associations_by_study_id_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_associations_by_study_id_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_associations_by_study_id_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+associations-class">associations</a> object.
</p>

<hr>
<h2 id='get_associations_by_variant_id'>Get GWAS Catalog associations by variant identifiers</h2><span id='topic+get_associations_by_variant_id'></span>

<h3>Description</h3>

<p>Gets associations by variant identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_associations_by_variant_id(
  variant_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_associations_by_variant_id_+3A_variant_id">variant_id</code></td>
<td>
<p>A <code>character</code> vector of GWAS Catalog variant identifiers.</p>
</td></tr>
<tr><td><code id="get_associations_by_variant_id_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_associations_by_variant_id_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_associations_by_variant_id_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+associations-class">associations</a> object.
</p>

<hr>
<h2 id='get_child_efo'>Get all child terms of this trait in the EFO hierarchy</h2><span id='topic+get_child_efo'></span>

<h3>Description</h3>

<p>Get all child terms of this trait in the EFO hierarchy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_child_efo(
  efo_id,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_child_efo_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a>
identifiers.</p>
</td></tr>
<tr><td><code id="get_child_efo_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_child_efo_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_child_efo_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
<tr><td><code id="get_child_efo_+3A_progress_bar">progress_bar</code></td>
<td>
<p>Whether to show a progress bar as the paginated resources
are retrieved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list whose values are character vectors of EFO identifiers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_child_efo(c('EFO_0004884', 'EFO_0004343', 'EFO_0005299'))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_metadata'>Get GWAS Catalog metadata</h2><span id='topic+get_metadata'></span>

<h3>Description</h3>

<p>Provides a list of the resources the GWAS Catalog data is currently mapped
against: <a href="https://www.ensembl.org/info/docs/api/versions.html">Ensembl
release number</a>, <a href="https://www.ncbi.nlm.nih.gov/grc/human">Genome build
version</a> and
<a href="https://www.ncbi.nlm.nih.gov/projects/SNP/snp_summary.cgi">dbSNP
version</a>.In addition, the date since this combination of resource versions
has been in use is also returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_metadata(verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_metadata_+3A_verbose">verbose</code></td>
<td>
<p>Whether to be chatty.</p>
</td></tr>
<tr><td><code id="get_metadata_+3A_warnings">warnings</code></td>
<td>
<p>Whether to trigger a warning if the request is not successful.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named <code><a href="base.html#topic+list">list</a></code> whose <code><a href="base.html#topic+names">names</a></code> are:
</p>

<ul>
<li> <p><code>ensembl_release_number</code>: <a href="https://www.ensembl.org/info/docs/api/versions.html">Ensembl release number</a>;
</p>
</li>
<li> <p><code>genome_build_version</code>: <a href="https://www.ncbi.nlm.nih.gov/grc/human">Genome build version</a>;
</p>
</li>
<li> <p><code>dbsnp_version</code>: <a href="https://www.ncbi.nlm.nih.gov/projects/SNP/snp_summary.cgi">dbSNP version</a>.
</p>
</li>
<li> <p><code>usage_start_date</code>: Date since this combination of resource versions has been in use.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_metadata(warnings = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_studies'>Get GWAS Catalog studies</h2><span id='topic+get_studies'></span>

<h3>Description</h3>

<p>Retrieves studies via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all studies that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
studies that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies(
  study_id = NULL,
  association_id = NULL,
  variant_id = NULL,
  efo_id = NULL,
  pubmed_id = NULL,
  user_requested = NULL,
  full_pvalue_set = NULL,
  efo_uri = NULL,
  efo_trait = NULL,
  reported_trait = NULL,
  set_operation = "union",
  interactive = TRUE,
  verbose = FALSE,
  warnings = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_+3A_study_id">study_id</code></td>
<td>
<p>A character vector of GWAS Catalog study accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_studies_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of GWAS Catalog association
identifiers.</p>
</td></tr>
<tr><td><code id="get_studies_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of GWAS Catalog variant identifiers.</p>
</td></tr>
<tr><td><code id="get_studies_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a>
identifiers.</p>
</td></tr>
<tr><td><code id="get_studies_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>An integer vector of
<a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a> identifiers.</p>
</td></tr>
<tr><td><code id="get_studies_+3A_user_requested">user_requested</code></td>
<td>
<p>A <code>logical</code> (scalar!) indicating to retrieve either studies
requested by users of the Catalog (<code>TRUE</code>) or otherwise
(<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="get_studies_+3A_full_pvalue_set">full_pvalue_set</code></td>
<td>
<p>A <code>logical</code> (scalar!) indicating to retrieve studies with full
summary statistics (<code>TRUE</code>) or studies without it (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="get_studies_+3A_efo_uri">efo_uri</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a> URIs.</p>
</td></tr>
<tr><td><code id="get_studies_+3A_efo_trait">efo_trait</code></td>
<td>
<p>A character vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p>
</td></tr>
<tr><td><code id="get_studies_+3A_reported_trait">reported_trait</code></td>
<td>
<p>A character vector of phenotypic traits as
reported by the original authors of the study.</p>
</td></tr>
<tr><td><code id="get_studies_+3A_set_operation">set_operation</code></td>
<td>
<p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how studies retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same studies found with different
criteria.</p>
</td></tr>
<tr><td><code id="get_studies_+3A_interactive">interactive</code></td>
<td>
<p>A logical. If all studies are requested, whether to ask
interactively if we really want to proceed.</p>
</td></tr>
<tr><td><code id="get_studies_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note that all search criteria are vectorised, thus allowing for batch
mode search, e.g., one can search by multiple variant identifiers at once by
passing a vector of identifiers to <code>variant_id</code>.
</p>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get a study by its accession identifier
get_studies(study_id = 'GCST001085', warnings = FALSE)

# Get a study by association identifier
get_studies(association_id = '25389945', warnings = FALSE)

# Get studies by variant identifier
get_studies(variant_id = 'rs3798440', warnings = FALSE)

# Get studies by EFO trait identifier
get_studies(efo_id = 'EFO_0005537', warnings = FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='get_studies_all'>Get all GWAS Catalog studies</h2><span id='topic+get_studies_all'></span>

<h3>Description</h3>

<p>Gets all studies. Beware this can take several minutes!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies_all(verbose = FALSE, warnings = TRUE, page_size = 20L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_all_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_all_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_studies_all_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>

<hr>
<h2 id='get_studies_by_association_id'>Get GWAS Catalog studies by association identifiers</h2><span id='topic+get_studies_by_association_id'></span>

<h3>Description</h3>

<p>Gets studies by GWAS Catalog internal association identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies_by_association_id(
  association_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_by_association_id_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of GWAS Catalog association
identifiers.</p>
</td></tr>
<tr><td><code id="get_studies_by_association_id_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_by_association_id_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_studies_by_association_id_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>

<hr>
<h2 id='get_studies_by_efo_id'>Get GWAS Catalog studies by EFO identifier</h2><span id='topic+get_studies_by_efo_id'></span>

<h3>Description</h3>

<p>Gets studies whose phenotypic trait is matched by EFO identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies_by_efo_id(
  efo_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_by_efo_id_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a>
identifiers.</p>
</td></tr>
<tr><td><code id="get_studies_by_efo_id_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_by_efo_id_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_studies_by_efo_id_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>

<hr>
<h2 id='get_studies_by_efo_trait'>Get GWAS Catalog studies by EFO traits</h2><span id='topic+get_studies_by_efo_trait'></span>

<h3>Description</h3>

<p>Gets studies that match <a href="https://www.ebi.ac.uk/efo/">EFO</a> trait
description.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies_by_efo_trait(
  efo_trait = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_by_efo_trait_+3A_efo_trait">efo_trait</code></td>
<td>
<p>A <code>character</code> vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p>
</td></tr>
<tr><td><code id="get_studies_by_efo_trait_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_by_efo_trait_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_studies_by_efo_trait_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>

<hr>
<h2 id='get_studies_by_efo_uri'>Get GWAS Catalog studies by EFO URIs</h2><span id='topic+get_studies_by_efo_uri'></span>

<h3>Description</h3>

<p>Gets studies that match <a href="https://www.ebi.ac.uk/efo/">EFO</a> URI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies_by_efo_uri(
  efo_uri = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_by_efo_uri_+3A_efo_uri">efo_uri</code></td>
<td>
<p>A character vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> URIs.</p>
</td></tr>
<tr><td><code id="get_studies_by_efo_uri_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_by_efo_uri_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_studies_by_efo_uri_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>

<hr>
<h2 id='get_studies_by_full_pvalue_set'>Get GWAS Catalog studies by full summary statistics criterion</h2><span id='topic+get_studies_by_full_pvalue_set'></span>

<h3>Description</h3>

<p>Gets studies that either have full summary statistics or studies that do not
have it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies_by_full_pvalue_set(
  full_pvalue_set = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_by_full_pvalue_set_+3A_full_pvalue_set">full_pvalue_set</code></td>
<td>
<p>A <code>logical</code> (scalar!) indicating to retrieve
studies with full summary statistics (<code>TRUE</code>) or studies without it
(<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="get_studies_by_full_pvalue_set_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_by_full_pvalue_set_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_studies_by_full_pvalue_set_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>

<hr>
<h2 id='get_studies_by_pubmed_id'>Get GWAS Catalog studies by PubMed identifiers</h2><span id='topic+get_studies_by_pubmed_id'></span>

<h3>Description</h3>

<p>Gets studies whose associated publications match
<a href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a> identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies_by_pubmed_id(
  pubmed_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_by_pubmed_id_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>An integer vector of
<a href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a> identifiers.</p>
</td></tr>
<tr><td><code id="get_studies_by_pubmed_id_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_by_pubmed_id_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_studies_by_pubmed_id_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>

<hr>
<h2 id='get_studies_by_reported_trait'>Get GWAS Catalog studies by reported traits</h2><span id='topic+get_studies_by_reported_trait'></span>

<h3>Description</h3>

<p>Gets studies that match the reported traits, as reported by the original
authors' of the study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies_by_reported_trait(
  reported_trait = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_by_reported_trait_+3A_reported_trait">reported_trait</code></td>
<td>
<p>A character vector of phenotypic traits as
reported by the original authors' the study.</p>
</td></tr>
<tr><td><code id="get_studies_by_reported_trait_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_by_reported_trait_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_studies_by_reported_trait_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>

<hr>
<h2 id='get_studies_by_study_id'>Get GWAS Catalog studies by study identifiers</h2><span id='topic+get_studies_by_study_id'></span>

<h3>Description</h3>

<p>Gets studies by GWAS Catalog internal study identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies_by_study_id(
  study_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_by_study_id_+3A_study_id">study_id</code></td>
<td>
<p>A character vector of GWAS Catalog study accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_studies_by_study_id_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_by_study_id_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_studies_by_study_id_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>

<hr>
<h2 id='get_studies_by_user_requested'>Get GWAS Catalog studies that have been requested by users or not</h2><span id='topic+get_studies_by_user_requested'></span>

<h3>Description</h3>

<p>Gets studies that have either been requested by users of the Catalog
or studies that have not been explicitly requested by users.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies_by_user_requested(
  user_requested = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_by_user_requested_+3A_user_requested">user_requested</code></td>
<td>
<p>A <code>logical</code> (scalar!) indicating to retrieve
either studies requested by users of the Catalog (<code>TRUE</code>) or otherwise
(<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="get_studies_by_user_requested_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_by_user_requested_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_studies_by_user_requested_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>

<hr>
<h2 id='get_studies_by_variant_id'>Get GWAS Catalog studies by variant identifiers</h2><span id='topic+get_studies_by_variant_id'></span>

<h3>Description</h3>

<p>Gets studies by variant identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_studies_by_variant_id(
  variant_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_studies_by_variant_id_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of GWAS Catalog variant identifiers.</p>
</td></tr>
<tr><td><code id="get_studies_by_variant_id_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_studies_by_variant_id_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_studies_by_variant_id_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>

<hr>
<h2 id='get_traits'>Get GWAS Catalog EFO traits</h2><span id='topic+get_traits'></span>

<h3>Description</h3>

<p>Retrieves traits via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all traits that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
traits that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_traits(
  study_id = NULL,
  association_id = NULL,
  efo_id = NULL,
  pubmed_id = NULL,
  efo_uri = NULL,
  efo_trait = NULL,
  set_operation = "union",
  verbose = FALSE,
  warnings = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_traits_+3A_study_id">study_id</code></td>
<td>
<p>A <code>character</code> vector of GWAS Catalog study accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_association_id">association_id</code></td>
<td>
<p>A <code>character</code> vector of GWAS Catalog association
identifiers.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a>
identifiers.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>An <code>integer</code> vector of
<a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a> identifiers.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_efo_uri">efo_uri</code></td>
<td>
<p>A <code>character</code> vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a> URIs.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_efo_trait">efo_trait</code></td>
<td>
<p>A <code>character</code> vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_set_operation">set_operation</code></td>
<td>
<p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how traits retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same traits found with different
criteria.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_traits_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note that all search criteria are vectorised, thus allowing for batch
mode search, e.g., one can search by multiple trait identifiers at once by
passing a vector of identifiers to <code>efo_id</code>.
</p>


<h3>Value</h3>

<p>A <a href="#topic+traits-class">traits</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get traits by study identifier
get_traits(study_id = 'GCST001085', warnings = FALSE)

# Get traits by association identifier
get_traits(association_id = '25389945', warnings = FALSE)

# Get a trait by its EFO identifier
get_traits(efo_id = 'EFO_0005537', warnings = FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='get_traits_all'>Get all GWAS Catalog EFO traits</h2><span id='topic+get_traits_all'></span>

<h3>Description</h3>

<p>Gets all EFO traits. Beware this can take several minutes!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_traits_all(verbose = FALSE, warnings = TRUE, page_size = 20L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_traits_all_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_traits_all_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_traits_all_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+traits-class">traits</a> object.
</p>

<hr>
<h2 id='get_traits_by_association_id'>Get GWAS Catalog traits by association identifiers</h2><span id='topic+get_traits_by_association_id'></span>

<h3>Description</h3>

<p>Gets traits by GWAS Catalog internal association identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_traits_by_association_id(
  association_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_traits_by_association_id_+3A_association_id">association_id</code></td>
<td>
<p>A <code>character</code> vector of GWAS Catalog association
identifiers.</p>
</td></tr>
<tr><td><code id="get_traits_by_association_id_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_traits_by_association_id_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_traits_by_association_id_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+traits-class">traits</a> object.
</p>

<hr>
<h2 id='get_traits_by_efo_id'>Get GWAS Catalog traits by EFO identifier</h2><span id='topic+get_traits_by_efo_id'></span>

<h3>Description</h3>

<p>Gets traits whose phenotypic trait is matched by EFO identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_traits_by_efo_id(
  efo_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_traits_by_efo_id_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a>
identifiers.</p>
</td></tr>
<tr><td><code id="get_traits_by_efo_id_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_traits_by_efo_id_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_traits_by_efo_id_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+traits-class">traits</a> object.
</p>

<hr>
<h2 id='get_traits_by_efo_trait'>Get GWAS Catalog traits by EFO traits</h2><span id='topic+get_traits_by_efo_trait'></span>

<h3>Description</h3>

<p>Gets traits that match <a href="https://www.ebi.ac.uk/efo/">EFO</a> trait
description.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_traits_by_efo_trait(
  efo_trait = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_traits_by_efo_trait_+3A_efo_trait">efo_trait</code></td>
<td>
<p>A <code>character</code> vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p>
</td></tr>
<tr><td><code id="get_traits_by_efo_trait_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_traits_by_efo_trait_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_traits_by_efo_trait_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+traits-class">traits</a> object.
</p>

<hr>
<h2 id='get_traits_by_efo_uri'>Get GWAS Catalog traits by EFO URIs</h2><span id='topic+get_traits_by_efo_uri'></span>

<h3>Description</h3>

<p>Gets traits that match <a href="https://www.ebi.ac.uk/efo/">EFO</a> URI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_traits_by_efo_uri(
  efo_uri = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_traits_by_efo_uri_+3A_efo_uri">efo_uri</code></td>
<td>
<p>A <code>character</code> vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> URIs.</p>
</td></tr>
<tr><td><code id="get_traits_by_efo_uri_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_traits_by_efo_uri_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_traits_by_efo_uri_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+traits-class">traits</a> object.
</p>

<hr>
<h2 id='get_traits_by_pubmed_id'>Get GWAS Catalog traits by PubMed identifiers</h2><span id='topic+get_traits_by_pubmed_id'></span>

<h3>Description</h3>

<p>Gets traits whose associated publications match
<a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a> identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_traits_by_pubmed_id(
  pubmed_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_traits_by_pubmed_id_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>An <code>integer</code> vector of
<a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a> identifiers.</p>
</td></tr>
<tr><td><code id="get_traits_by_pubmed_id_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_traits_by_pubmed_id_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_traits_by_pubmed_id_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+traits-class">traits</a> object.
</p>

<hr>
<h2 id='get_traits_by_study_id'>Get GWAS Catalog traits by study identifiers</h2><span id='topic+get_traits_by_study_id'></span>

<h3>Description</h3>

<p>Gets traits by GWAS Catalog internal study identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_traits_by_study_id(
  study_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_traits_by_study_id_+3A_study_id">study_id</code></td>
<td>
<p>A <code>character</code> vector of GWAS Catalog study accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_traits_by_study_id_+3A_verbose">verbose</code></td>
<td>
<p>A <code>logical</code> indicating whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_traits_by_study_id_+3A_warnings">warnings</code></td>
<td>
<p>A <code>logical</code> indicating whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_traits_by_study_id_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+traits-class">traits</a> object.
</p>

<hr>
<h2 id='get_variants'>Get GWAS Catalog variants</h2><span id='topic+get_variants'></span>

<h3>Description</h3>

<p>Retrieves variants via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all variants that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
variants that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants(
  study_id = NULL,
  association_id = NULL,
  variant_id = NULL,
  efo_id = NULL,
  pubmed_id = NULL,
  genomic_range = NULL,
  cytogenetic_band = NULL,
  gene_name = NULL,
  efo_trait = NULL,
  reported_trait = NULL,
  set_operation = "union",
  interactive = TRUE,
  std_chromosomes_only = TRUE,
  verbose = FALSE,
  warnings = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_+3A_study_id">study_id</code></td>
<td>
<p>A character vector of GWAS Catalog study accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of GWAS Catalog association
identifiers.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of GWAS Catalog variant identifiers.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a>
identifiers.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>An integer vector of
<a href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a> identifiers.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_genomic_range">genomic_range</code></td>
<td>
<p>A named list of three vectors:
</p>
 <dl>
<dt>chromosome</dt><dd><p>A character vector of chromosome names of the
form 1&ndash;22, X or Y.</p>
</dd> <dt>start</dt><dd><p>A numeric vector of start positions,
starting at 1.</p>
</dd> <dt>end</dt><dd><p>A numeric vector of end positions.</p>
</dd> </dl>

<p>The three vectors need to be of the same length so that <code>chromosome</code>
names, <code>start</code> and <code>end</code> positions can be matched by position.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_cytogenetic_band">cytogenetic_band</code></td>
<td>
<p>A character vector of cytogenetic bands of the form
<code>'1p36.11'</code>.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_gene_name">gene_name</code></td>
<td>
<p>Gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_efo_trait">efo_trait</code></td>
<td>
<p>A character vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_reported_trait">reported_trait</code></td>
<td>
<p>A character vector of phenotypic traits as
reported by the original authors of the study.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_set_operation">set_operation</code></td>
<td>
<p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how variants retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same variants found with different
criteria.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_interactive">interactive</code></td>
<td>
<p>A logical. If all variants are requested, whether to ask
interactively if we really want to proceed.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_std_chromosomes_only">std_chromosomes_only</code></td>
<td>
<p>Whether to return only variants mapped to
standard chromosomes: 1 thru 22, X, Y, and MT.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note that all search criteria are vectorised, thus allowing for batch
mode search, e.g., one can search by multiple variant identifiers at once by
passing a vector of identifiers to <code>variant_id</code>.
</p>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get variants by study identifier
get_variants(study_id = 'GCST001085', warnings = FALSE)

# Get a variant by its identifier
## Not run: 
get_variants(variant_id = 'rs3798440', warnings = FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='get_variants_all'>Get all GWAS Catalog variants</h2><span id='topic+get_variants_all'></span>

<h3>Description</h3>

<p>Gets all variants. Beware this can take several minutes!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants_all(verbose = FALSE, warnings = TRUE, page_size = 20L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_all_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_all_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_variants_all_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>

<hr>
<h2 id='get_variants_by_association_id'>Get GWAS Catalog variants by their association identifiers</h2><span id='topic+get_variants_by_association_id'></span>

<h3>Description</h3>

<p>Gets variants by GWAS Catalog internal association identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants_by_association_id(
  association_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_by_association_id_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of GWAS Catalog association
identifiers.</p>
</td></tr>
<tr><td><code id="get_variants_by_association_id_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_by_association_id_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_variants_by_association_id_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>

<hr>
<h2 id='get_variants_by_cytogenetic_band'>Get GWAS Catalog variants by cytogenetic band.</h2><span id='topic+get_variants_by_cytogenetic_band'></span>

<h3>Description</h3>

<p>Gets variants that are mapped onto specific regions as specified by
cytogenetic bands. See the dataframe
<code><a href="#topic+cytogenetic_bands">cytogenetic_bands</a></code> for more information on possible
values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants_by_cytogenetic_band(
  cytogenetic_band = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_by_cytogenetic_band_+3A_cytogenetic_band">cytogenetic_band</code></td>
<td>
<p>A <code>character</code> vector of cytogenetic bands of the
form <code>'1p36.11'</code></p>
</td></tr>
<tr><td><code id="get_variants_by_cytogenetic_band_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_by_cytogenetic_band_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_variants_by_cytogenetic_band_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>

<hr>
<h2 id='get_variants_by_efo_id'>Get GWAS Catalog studies by EFO identifier</h2><span id='topic+get_variants_by_efo_id'></span>

<h3>Description</h3>

<p>Gets variants whose phenotypic trait is matched by EFO identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants_by_efo_id(
  efo_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_by_efo_id_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of <a href="https://www.ebi.ac.uk/efo/">EFO</a>
identifiers.</p>
</td></tr>
<tr><td><code id="get_variants_by_efo_id_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_by_efo_id_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_variants_by_efo_id_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>

<hr>
<h2 id='get_variants_by_efo_trait'>Get GWAS Catalog variants by EFO traits</h2><span id='topic+get_variants_by_efo_trait'></span>

<h3>Description</h3>

<p>Gets variants that match <a href="https://www.ebi.ac.uk/efo/">EFO</a> trait
description.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants_by_efo_trait(
  efo_trait = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_by_efo_trait_+3A_efo_trait">efo_trait</code></td>
<td>
<p>A character vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p>
</td></tr>
<tr><td><code id="get_variants_by_efo_trait_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_by_efo_trait_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_variants_by_efo_trait_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>

<hr>
<h2 id='get_variants_by_gene_name'>Get GWAS Catalog variants by gene name.</h2><span id='topic+get_variants_by_gene_name'></span>

<h3>Description</h3>

<p>Gets variants whose genomic context includes a specific gene or genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants_by_gene_name(
  gene_name = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_by_gene_name_+3A_gene_name">gene_name</code></td>
<td>
<p>A <code>character</code> vector of gene names.</p>
</td></tr>
<tr><td><code id="get_variants_by_gene_name_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_by_gene_name_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_variants_by_gene_name_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>

<hr>
<h2 id='get_variants_by_genomic_range'>Get GWAS Catalog variants by genomic range</h2><span id='topic+get_variants_by_genomic_range'></span>

<h3>Description</h3>

<p>Gets variants by genomic range.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants_by_genomic_range(
  chromosome = NULL,
  start = NULL,
  end = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_by_genomic_range_+3A_chromosome">chromosome</code></td>
<td>
<p>A character vector of human chromosome names: autosomal
and sexual chromosomes only, i.e., 1&ndash;22, X and Y.</p>
</td></tr>
<tr><td><code id="get_variants_by_genomic_range_+3A_start">start</code></td>
<td>
<p>Start position of range (starts at 1).</p>
</td></tr>
<tr><td><code id="get_variants_by_genomic_range_+3A_end">end</code></td>
<td>
<p>End position of range (inclusive).</p>
</td></tr>
<tr><td><code id="get_variants_by_genomic_range_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_by_genomic_range_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_variants_by_genomic_range_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>

<hr>
<h2 id='get_variants_by_pubmed_id'>Get GWAS Catalog variants by PubMed identifiers</h2><span id='topic+get_variants_by_pubmed_id'></span>

<h3>Description</h3>

<p>Gets variants whose associated publications match
<a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a> identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants_by_pubmed_id(
  pubmed_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_by_pubmed_id_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>An integer vector of
<a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a> identifiers.</p>
</td></tr>
<tr><td><code id="get_variants_by_pubmed_id_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_by_pubmed_id_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_variants_by_pubmed_id_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>

<hr>
<h2 id='get_variants_by_reported_trait'>Get GWAS Catalog variants by reported traits</h2><span id='topic+get_variants_by_reported_trait'></span>

<h3>Description</h3>

<p>Gets variants that match the reported traits, as reported by the original
authors' of the study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants_by_reported_trait(
  reported_trait = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_by_reported_trait_+3A_reported_trait">reported_trait</code></td>
<td>
<p>A character vector of phenotypic traits as reported by
the original authors' the study. Note: this parameter is case sensitive.</p>
</td></tr>
<tr><td><code id="get_variants_by_reported_trait_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_by_reported_trait_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_variants_by_reported_trait_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>

<hr>
<h2 id='get_variants_by_study_id'>Get GWAS Catalog variants by study identifiers</h2><span id='topic+get_variants_by_study_id'></span>

<h3>Description</h3>

<p>Gets variants by GWAS Catalog internal study identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants_by_study_id(
  study_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_by_study_id_+3A_study_id">study_id</code></td>
<td>
<p>A character vector of GWAS Catalog study accession
identifiers.</p>
</td></tr>
<tr><td><code id="get_variants_by_study_id_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_by_study_id_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
<tr><td><code id="get_variants_by_study_id_+3A_page_size">page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>

<hr>
<h2 id='get_variants_by_variant_id'>Get GWAS Catalog variants by variant identifiers</h2><span id='topic+get_variants_by_variant_id'></span>

<h3>Description</h3>

<p>Gets variants by variant identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_variants_by_variant_id(
  variant_id = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_variants_by_variant_id_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of GWAS Catalog variant identifiers.</p>
</td></tr>
<tr><td><code id="get_variants_by_variant_id_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td></tr>
<tr><td><code id="get_variants_by_variant_id_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings, if any.</p>
</td></tr>
<tr><td><code id="get_variants_by_variant_id_+3A_page_size">page_size</code></td>
<td>
<p>An <code>integer</code> scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>

<hr>
<h2 id='is_association_id'>Is a string a GWAS Catalog association accession ID?</h2><span id='topic+is_association_id'></span>

<h3>Description</h3>

<p>Find which strings are valid GWAS Catalog association IDs (returns
<code>TRUE</code>). Association IDs are tested against the following regular
expression: <code>^\\d+$</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_association_id(str, convert_NA_to_FALSE = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_association_id_+3A_str">str</code></td>
<td>
<p>A character vector of strings.</p>
</td></tr>
<tr><td><code id="is_association_id_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to treat <code>NA</code> as <code>NA</code>
(<code>convert_NA_to_FALSE = FALSE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector.
</p>

<hr>
<h2 id='is_ebi_reachable'>Is the GWAS Catalog REST API server reachable?</h2><span id='topic+is_ebi_reachable'></span>

<h3>Description</h3>

<p>Check if the EBI server where the GWAS Catalog REST API server is running is
reachable. This function attempts to connect to
<a href="https://www.ebi.ac.uk">https://www.ebi.ac.uk</a>, returning <code>TRUE</code> on
success, and <code>FALSE</code> otherwise. Set <code>chatty = TRUE</code> for a step by
step description of the connection attempt.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_ebi_reachable(url = "https://www.ebi.ac.uk", port = 443L, chatty = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_ebi_reachable_+3A_url">url</code></td>
<td>
<p>NHGRI-EBI GWAS Catalog server URL. Default is
<a href="https://www.ebi.ac.uk">https://www.ebi.ac.uk</a>. You should not need to change this parameter.</p>
</td></tr>
<tr><td><code id="is_ebi_reachable_+3A_port">port</code></td>
<td>
<p>Network port on which to ping the server. You should not need to
change this parameter.</p>
</td></tr>
<tr><td><code id="is_ebi_reachable_+3A_chatty">chatty</code></td>
<td>
<p>Whether to be verbose (<code>TRUE</code>) or not (<code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value: <code>TRUE</code> if EBI server is reachable, <code>FALSE</code>
otherwise.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Check if the GWAS Catalog Server is reachable
is_ebi_reachable() # Returns TRUE or FALSE.

# Check if the GWAS Catalog Server is reachable
# and show exactly at what step is it failing (if that is the case)
is_ebi_reachable(chatty = TRUE)

</code></pre>

<hr>
<h2 id='is_efo_id'>Is a string an EFO trait ID?</h2><span id='topic+is_efo_id'></span>

<h3>Description</h3>

<p>Find which strings are valid EFO trait IDs (returns
<code>TRUE</code>). EFO trait IDs are tested against the following regular
expression: <code>^EFO_\\d{7}$</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_efo_id(str, convert_NA_to_FALSE = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_efo_id_+3A_str">str</code></td>
<td>
<p>A character vector of strings.</p>
</td></tr>
<tr><td><code id="is_efo_id_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to treat <code>NA</code> as <code>NA</code>
(<code>convert_NA_to_FALSE = FALSE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector.
</p>

<hr>
<h2 id='is_efo_id2'>Is a string an EFO trait ID in the broad sense?</h2><span id='topic+is_efo_id2'></span>

<h3>Description</h3>

<p>This function is more permissible than <code><a href="#topic+is_efo_id">is_efo_id</a></code>.
This function matches EFO trait IDs against the following regular expression:
<code>^\\w+$</code>. This is very forgiving on the input, any sequence of word
characters are ok. This is useful to match EFO identifiers that do not follow
the regex <code>^EFO_\\d{7}$</code>, such as: <code>'GO_0097334'</code>,
<code>'HP_0001268'</code>, <code>'Orphanet_182098'</code>, and <code>'NCIT_C74532'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_efo_id2(str, convert_NA_to_FALSE = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_efo_id2_+3A_str">str</code></td>
<td>
<p>A character vector of strings.</p>
</td></tr>
<tr><td><code id="is_efo_id2_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to treat <code>NA</code> as <code>NA</code>
(<code>convert_NA_to_FALSE = FALSE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector.
</p>

<hr>
<h2 id='is_embedded'>Is the GWAS response wrapped in an '_embedded' object?</h2><span id='topic+is_embedded'></span>

<h3>Description</h3>

<p>Checks if the response is wrapped in an '_embedded' object by checking if an
element named '_embedded' exists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_embedded(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_embedded_+3A_obj">obj</code></td>
<td>
<p>The response object as return by <code>jsonlite::fromJSON</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value.
</p>

<hr>
<h2 id='is_empty_str'>Is string empty or an all whitespace string?</h2><span id='topic+is_empty_str'></span>

<h3>Description</h3>

<p>Matches the <code>string</code> vector against <code>"^\\s*$"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_empty_str(str, convert_NA_to_FALSE = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_empty_str_+3A_str">str</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> vector.</p>
</td></tr>
<tr><td><code id="is_empty_str_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to treat <code>NA</code> as <code>NA</code>
(<code>convert_NA_to_FALSE = FALSE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="base.html#topic+logical">logical</a></code>.
</p>

<hr>
<h2 id='is_human_chromosome'>Is a string a human chromosome name?</h2><span id='topic+is_human_chromosome'></span>

<h3>Description</h3>

<p>Find which strings are valid human chromosome names. The valid chromosome
names can be specified via the argument <code>chromosomes</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_human_chromosome(
  string,
  chromosomes = c(seq_len(22), "X", "Y", "MT"),
  convert_NA_to_FALSE = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_human_chromosome_+3A_string">string</code></td>
<td>
<p>A character vector of strings.</p>
</td></tr>
<tr><td><code id="is_human_chromosome_+3A_chromosomes">chromosomes</code></td>
<td>
<p>A character vector of valid chromosome names. Default is
autosomal chromosomes 1 thru 22 and, X, Y, and MT.</p>
</td></tr>
<tr><td><code id="is_human_chromosome_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to treat <code>NA</code> as <code>NA</code>
(<code>convert_NA_to_FALSE = FALSE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector.
</p>

<hr>
<h2 id='is_paginated'>Is the GWAS response paginated?</h2><span id='topic+is_paginated'></span>

<h3>Description</h3>

<p>Checks if the response is paginated by checking if an element named 'page'
exists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_paginated(content)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_paginated_+3A_content">content</code></td>
<td>
<p>The response content as return by <code>jsonlite::fromJSON</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value.
</p>

<hr>
<h2 id='is_pubmed_id'>Is a string a PubMed ID?</h2><span id='topic+is_pubmed_id'></span>

<h3>Description</h3>

<p>Find which strings are valid PubMed IDs (returns <code>TRUE</code>). PubMed IDs are
tested against the following regular expression: <code>^\\d+$</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_pubmed_id(str, convert_NA_to_FALSE = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_pubmed_id_+3A_str">str</code></td>
<td>
<p>A character vector of strings.</p>
</td></tr>
<tr><td><code id="is_pubmed_id_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to treat <code>NA</code> as <code>NA</code>
(<code>convert_NA_to_FALSE = FALSE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector.
</p>

<hr>
<h2 id='is_rs_id'>Is a string a valid rsID?</h2><span id='topic+is_rs_id'></span>

<h3>Description</h3>

<p>Find which strings are valid SNP reference IDs, i.e., of the form rs[0-9]+.
Please note that this only does a syntax validation on the strings. It does
not check whether the actual IDs exist in dbSNP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_rs_id(str, convert_NA_to_FALSE = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_rs_id_+3A_str">str</code></td>
<td>
<p>A character vector of putative SNP reference IDs of the form
rs[0-9]+.</p>
</td></tr>
<tr><td><code id="is_rs_id_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to preserve <code>NA</code>
(<code>convert_NA_to_FALSE = TRUE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a logical vector of the same length as <code>str</code>,
<code>TRUE</code> for strings that are valid rs IDs, and <code>FALSE</code> otherwise.
</p>

<hr>
<h2 id='is_study_id'>Is a string a GWAS Catalog study accession ID?</h2><span id='topic+is_study_id'></span>

<h3>Description</h3>

<p>Find which strings are valid GWAS Catalog study accession IDs (returns
<code>TRUE</code>). Study accession IDs are tested against the following regular
expression: <code>^GCST\\d+$</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_study_id(str, convert_NA_to_FALSE = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_study_id_+3A_str">str</code></td>
<td>
<p>A character vector of strings.</p>
</td></tr>
<tr><td><code id="is_study_id_+3A_convert_na_to_false">convert_NA_to_FALSE</code></td>
<td>
<p>Whether to treat <code>NA</code> as <code>NA</code>
(<code>convert_NA_to_FALSE = FALSE</code>) or whether to return <code>FALSE</code> when
an <code>NA</code> is found (<code>convert_NA_to_FALSE = TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector.
</p>

<hr>
<h2 id='list_to_s4'>Convert a named list to an S4 object</h2><span id='topic+list_to_s4'></span>

<h3>Description</h3>

<p>Convert a named list to an S4 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_to_s4(list, class)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="list_to_s4_+3A_list">list</code></td>
<td>
<p>list</p>
</td></tr>
<tr><td><code id="list_to_s4_+3A_class">class</code></td>
<td>
<p>character vector indicating the S4 class</p>
</td></tr>
</table>


<h3>Value</h3>

<p>S4 object of class <code>class</code>.
</p>

<hr>
<h2 id='loci_tbl'>Creates a loci table.</h2><span id='topic+loci_tbl'></span>

<h3>Description</h3>

<p>Creates a loci table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loci_tbl(
  association_id = character(),
  locus_id = integer(),
  haplotype_snp_count = integer(),
  description = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loci_tbl_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of association identifiers.</p>
</td></tr>
<tr><td><code id="loci_tbl_+3A_locus_id">locus_id</code></td>
<td>
<p>An integer vector of locus identifiers.</p>
</td></tr>
<tr><td><code id="loci_tbl_+3A_haplotype_snp_count">haplotype_snp_count</code></td>
<td>
<p>An integer vector indicating the number of
variants in the haplotype.</p>
</td></tr>
<tr><td><code id="loci_tbl_+3A_description">description</code></td>
<td>
<p>A character vector of descriptions, one per locus
identifier.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='lstjoin'>Joins for lists.</h2><span id='topic+lstjoin'></span><span id='topic+lst_left_join'></span><span id='topic+lst_right_join'></span><span id='topic+lst_inner_join'></span><span id='topic+plst_left_join'></span><span id='topic+plst_right_join'></span><span id='topic+plst_inner_join'></span>

<h3>Description</h3>

<p>These functions join lists by matching elements by name. In the case of
<code>lst_left_join</code> <code>lst_x</code> is used as reference for traversal and its
names are looked for in <code>lst_y</code> for joining. In the case of
<code>lst_left_join</code>, <code>lst_y</code> is used as reference.
<code>lst_inner_join</code> only uses names common to both <code>lst_x</code> and
<code>lst_y</code> for combining elements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lst_left_join(lst_x, lst_y)

lst_right_join(lst_x, lst_y)

lst_inner_join(lst_x, lst_y)

plst_left_join(list_of_lsts)

plst_right_join(list_of_lsts)

plst_inner_join(list_of_lsts)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lstjoin_+3A_lst_x">lst_x</code>, <code id="lstjoin_+3A_lst_y">lst_y</code></td>
<td>
<p>lists.</p>
</td></tr>
<tr><td><code id="lstjoin_+3A_list_of_lsts">list_of_lsts</code></td>
<td>
<p>A list of lists to be joined together.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The functions <code>plst_left_join</code>, <code>plst_right_join</code> and
<code>plst_inner_join</code> are parallel versions that allow joining more than two
lists easily, i.e., just pass a list of lists to be joined.
</p>


<h3>Value</h3>

<p>A list.
</p>

<hr>
<h2 id='metadata_lst'>Generate a list to hold GWAS metadata</h2><span id='topic+metadata_lst'></span>

<h3>Description</h3>

<p>Creates a list to hold GWAS Catalog metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metadata_lst(
  ensembl_release_number = NA_integer_,
  genome_build_version = NA_character_,
  dbsnp_version = NA_integer_,
  usage_start_date = lubridate::ymd_hms()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metadata_lst_+3A_ensembl_release_number">ensembl_release_number</code></td>
<td>
<p><a href="https://www.ensembl.org/info/docs/api/versions.html">Ensembl release
number</a>.</p>
</td></tr>
<tr><td><code id="metadata_lst_+3A_genome_build_version">genome_build_version</code></td>
<td>
<p><a href="https://www.ncbi.nlm.nih.gov/grc/human">Genome build version</a>.</p>
</td></tr>
<tr><td><code id="metadata_lst_+3A_dbsnp_version">dbsnp_version</code></td>
<td>
<p><a href="https://www.ncbi.nlm.nih.gov/projects/SNP/snp_summary.cgi">dbSNP
version</a>.</p>
</td></tr>
<tr><td><code id="metadata_lst_+3A_usage_start_date">usage_start_date</code></td>
<td>
<p>Date since this combination of resource versions has
been in use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of 4 named elements as passed as arguments:
<code>ensembl_release_number</code>, <code>genome_build_version</code>,
<code>dbsnp_version</code> and <code>usage_start_date</code>.
</p>

<hr>
<h2 id='missing_to_na'>Recursively map all missing values to <code>NA</code>.</h2><span id='topic+missing_to_na'></span>

<h3>Description</h3>

<p>Traverse a recursive list and map missing values to <code><a href="base.html#topic+NA">NA</a></code>.
Missing values are: </p>
 <ul>
<li> <p><code>NULL</code> </p>
</li>
<li> <p><code>"NR"</code> </p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>missing_to_na(lst, na = NA_character_)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="missing_to_na_+3A_lst">lst</code></td>
<td>
<p>A list.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: This function is case sensitive on <code>"NR"</code>, i.e., it will leave
<code>"nr"</code> untouched.
</p>


<h3>Value</h3>

<p>A list whose missing values have been replaced by NA.
</p>

<hr>
<h2 id='n'>Number of GWAS Catalog entities</h2><span id='topic+n'></span><span id='topic+n+2Cstudies-method'></span><span id='topic+n+2Cassociations-method'></span><span id='topic+n+2Cvariants-method'></span><span id='topic+n+2Ctraits-method'></span>

<h3>Description</h3>

<p>This function returns the number of unique entities in a GWAS Catalog object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n(x, unique = FALSE)

## S4 method for signature 'studies'
n(x, unique = FALSE)

## S4 method for signature 'associations'
n(x, unique = FALSE)

## S4 method for signature 'variants'
n(x, unique = FALSE)

## S4 method for signature 'traits'
n(x, unique = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="n_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+studies-class">studies</a>, an
<a href="#topic+associations-class">associations</a>, a <a href="#topic+variants-class">variants</a>, or a
<a href="#topic+traits-class">traits</a> object.</p>
</td></tr>
<tr><td><code id="n_+3A_unique">unique</code></td>
<td>
<p>Whether to count only unique entries (<code>TRUE</code>) or not
(<code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer scalar.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Determine number of studies
n(studies_ex01)

# Determine number of associations
n(associations_ex01)

# Determine number of variants
n(variants_ex01)

# Determine number of traits
n(traits_ex01)

</code></pre>

<hr>
<h2 id='normalise_obj'>Normalise a JSON-list.</h2><span id='topic+normalise_obj'></span>

<h3>Description</h3>

<p>This function normalises a JSON-list. The concept of JSON-list is here
defined as an ordinary R list object whose elements are either all named or
none is named. These lists map naturally to JSON elements: objects and
arrays. What this function does is wrap certain elements of the nested
list <code>obj</code> in <code>list()</code> to make the tree structure of the <code>obj</code>
list homologous across different responses. This makes all responses of the
same entity type (studies, associations, variants or traits) homologous and
hence joinable with family functions <code>lst_*_join</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalise_obj(obj, resource_url)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalise_obj_+3A_obj">obj</code></td>
<td>
<p>A JSON-list. This is just an ordinary list as returned by
<code><a href="#topic+gc_request">gc_request</a></code> or <code><a href="#topic+gc_request_all">gc_request_all</a></code>.</p>
</td></tr>
<tr><td><code id="normalise_obj_+3A_resource_url">resource_url</code></td>
<td>
<p>The URL endpoint used to obtain the JSON-list <code>obj</code>.
This is used to infer the type of GWAS entity returned. See
<code><a href="#topic+object_type_from_url">object_type_from_url</a></code> for more details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This normalisation is GWAS object type specific. The parameter
<code>resource_url</code> should be mappable to either studies, associations,
variants or traits by <code><a href="#topic+object_type_from_url">object_type_from_url</a></code>.
</p>


<h3>Value</h3>

<p>A normalised JSON-list.
</p>

<hr>
<h2 id='object_type_from_url'>Identify the GWAS object entity from the URL endpoint</h2><span id='topic+object_type_from_url'></span>

<h3>Description</h3>

<p>This function takes URL endpoints and returns one of: studies, associations,
variants or traits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>object_type_from_url(resource_url)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="object_type_from_url_+3A_resource_url">resource_url</code></td>
<td>
<p>A character vector of GWAS URL endpoints.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of either <code>"studies"</code>, <code>"associations"</code>,
<code>"variants"</code> or <code>"traits"</code>.
</p>

<hr>
<h2 id='open_in_dbsnp'>Browse dbSNP from SNP identifiers.</h2><span id='topic+open_in_dbsnp'></span>

<h3>Description</h3>

<p>This function launches the web browser at dbSNP and opens a tab for each SNP
identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>open_in_dbsnp(variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="open_in_dbsnp_+3A_variant_id">variant_id</code></td>
<td>
<p>A variant identifier, a character vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if successful. Note however that this
function is run for its side effect.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>open_in_dbsnp('rs56261590')
</code></pre>

<hr>
<h2 id='open_in_gtex'>Browse GTEx from SNP identifiers.</h2><span id='topic+open_in_gtex'></span>

<h3>Description</h3>

<p>This function launches the web browser at the GTEx Portal and opens a tab for
each SNP identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>open_in_gtex(variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="open_in_gtex_+3A_variant_id">variant_id</code></td>
<td>
<p>A variant identifier, a character vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if successful. Note however that this
function is run for its side effect.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>open_in_gtex('rs56261590')
</code></pre>

<hr>
<h2 id='open_in_gwas_catalog'>Browse GWAS Catalog entities from the GWAS Web Graphical User Interface</h2><span id='topic+open_in_gwas_catalog'></span>

<h3>Description</h3>

<p>This function launches the web browser and opens a tab for each identifier on
the GWAS web graphical user interface: <a href="https://www.ebi.ac.uk/gwas">https://www.ebi.ac.uk/gwas</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>open_in_gwas_catalog(
  identifier,
  gwas_catalog_entity = c("study", "variant", "trait", "gene", "region", "publication")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="open_in_gwas_catalog_+3A_identifier">identifier</code></td>
<td>
<p>A vector of identifiers. The identifiers can be: study
accession identifiers, variant identifiers, EFO trait identifiers, gene
symbol names, cytogenetic regions, or PubMed identifiers.</p>
</td></tr>
<tr><td><code id="open_in_gwas_catalog_+3A_gwas_catalog_entity">gwas_catalog_entity</code></td>
<td>
<p>Either <code>'study'</code> (default), <code>'variant'</code>,
<code>'trait'</code>, <code>'gene'</code>, <code>'region'</code> or <code>'publication'</code>, a
scalar character. This argument indicates the type of the identifiers
passed in <code>identifier</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if successful, or <code>FALSE</code> otherwise. But
note that this function is run for its side effect.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Open studies in GWAS Web Graphical User Interface
open_in_gwas_catalog(c('GCST000016', 'GCST001115'))

# Open variants
open_in_gwas_catalog(c('rs146992477', 'rs56261590'),
  gwas_catalog_entity = 'variant')

# Open EFO traits
open_in_gwas_catalog(c('EFO_0004884', 'EFO_0004343'),
  gwas_catalog_entity = 'trait')

# Open genes
open_in_gwas_catalog(c('DPP6', 'MCCC2'),
  gwas_catalog_entity = 'gene')

# Open cytogenetic regions
open_in_gwas_catalog(c('2q37.1', '1p36.11'),
  gwas_catalog_entity = 'region')

# Open publications
open_in_gwas_catalog(c('25533513', '24376627'),
  gwas_catalog_entity = 'publication')

</code></pre>

<hr>
<h2 id='open_in_pubmed'>Browse PubMed from PubMed identifiers.</h2><span id='topic+open_in_pubmed'></span>

<h3>Description</h3>

<p>This function launches the web browser and opens a tab for each PubMed
citation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>open_in_pubmed(pubmed_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="open_in_pubmed_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p>A PubMed identifier, either a character or an integer
vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if successful. Note however that this
function is run for its side effect.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>open_in_pubmed(c('26301688', '30595370'))

</code></pre>

<hr>
<h2 id='peel_off_embedded'>Peels off the _embedded tier from a JSON-list.</h2><span id='topic+peel_off_embedded'></span>

<h3>Description</h3>

<p>This function removes an element named <code>_embedded</code> from the element
<code>content</code> moving all of its contents one level up. If
<code>obj$content$`_embedded`</code> does not exist then returns <code>obj</code>
untouched.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peel_off_embedded(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peel_off_embedded_+3A_obj">obj</code></td>
<td>
<p>A JSON-list.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A JSON-list.
</p>

<hr>
<h2 id='platforms_tbl'>Creates a platforms table.</h2><span id='topic+platforms_tbl'></span>

<h3>Description</h3>

<p>Creates a platforms table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>platforms_tbl(study_id = character(), manufacturer = character())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="platforms_tbl_+3A_study_id">study_id</code></td>
<td>
<p>GWAS Catalog study accession identifier.</p>
</td></tr>
<tr><td><code id="platforms_tbl_+3A_manufacturer">manufacturer</code></td>
<td>
<p>Platform manufacturer.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='publications_tbl'>Creates a publications table.</h2><span id='topic+publications_tbl'></span>

<h3>Description</h3>

<p>Creates a publications table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>publications_tbl(
  study_id = character(),
  pubmed_id = integer(),
  publication_date = lubridate::ymd(),
  publication = character(),
  title = character(),
  author_fullname = character(),
  author_orcid = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="publications_tbl_+3A_study_id">study_id</code></td>
<td>
<p>GWAS Catalog study accession identifier.</p>
</td></tr>
<tr><td><code id="publications_tbl_+3A_pubmed_id">pubmed_id</code></td>
<td>
<p><a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a>
identifier.</p>
</td></tr>
<tr><td><code id="publications_tbl_+3A_publication_date">publication_date</code></td>
<td>
<p>Publication date (online date if available) formatted
as <code><a href="lubridate.html#topic+ymd">ymd</a></code>.</p>
</td></tr>
<tr><td><code id="publications_tbl_+3A_publication">publication</code></td>
<td>
<p>Abbreviated journal name.</p>
</td></tr>
<tr><td><code id="publications_tbl_+3A_title">title</code></td>
<td>
<p>Publication title.</p>
</td></tr>
<tr><td><code id="publications_tbl_+3A_author_fullname">author_fullname</code></td>
<td>
<p>Last name and initials of first author.</p>
</td></tr>
<tr><td><code id="publications_tbl_+3A_author_orcid">author_orcid</code></td>
<td>
<p>Author's <a href="https://en.wikipedia.org/wiki/ORCID">ORCID
iD</a> (Open Researcher and Contributor ID).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='recursive_apply'>Simple rapply version that deals with NULL values.</h2><span id='topic+recursive_apply'></span>

<h3>Description</h3>

<p>Like <code><a href="base.html#topic+rapply">rapply</a></code>, <code>recursive_apply</code> is a recursive
version <code><a href="base.html#topic+lapply">lapply</a></code> but contrary to
<code><a href="base.html#topic+rapply">rapply</a></code>, <code>recursive_apply</code> does not ignore
<code><a href="base.html#topic+NULL">NULL</a></code> values. Each element of the list which is not itself
a list is replaced by the result of applying <code>fn</code>. If down the line
there are data.frames, then their class is preserved.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recursive_apply(x, fn)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="recursive_apply_+3A_x">x</code></td>
<td>
<p>A list (of potentially many nested lists).</p>
</td></tr>
<tr><td><code id="recursive_apply_+3A_fn">fn</code></td>
<td>
<p>A function of a single argument.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list whose non-list elements have been replaced by the result of
applying <code>fn</code>.
</p>

<hr>
<h2 id='reported_genes_tbl'>Creates an authors' reported genes table.</h2><span id='topic+reported_genes_tbl'></span>

<h3>Description</h3>

<p>Creates an authors' reported genes table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reported_genes_tbl(
  association_id = character(),
  locus_id = integer(),
  gene_name = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reported_genes_tbl_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of association identifiers.</p>
</td></tr>
<tr><td><code id="reported_genes_tbl_+3A_locus_id">locus_id</code></td>
<td>
<p>An integer vector of locus identifiers.</p>
</td></tr>
<tr><td><code id="reported_genes_tbl_+3A_gene_name">gene_name</code></td>
<td>
<p>A character vector of gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='risk_alleles_tbl'>Creates a risk alleles table.</h2><span id='topic+risk_alleles_tbl'></span>

<h3>Description</h3>

<p>Creates a risk alleles table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>risk_alleles_tbl(
  association_id = character(),
  locus_id = integer(),
  variant_id = character(),
  risk_allele = character(),
  risk_frequency = double(),
  genome_wide = logical(),
  limited_list = logical()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="risk_alleles_tbl_+3A_association_id">association_id</code></td>
<td>
<p>A character vector of association identifiers.</p>
</td></tr>
<tr><td><code id="risk_alleles_tbl_+3A_locus_id">locus_id</code></td>
<td>
<p>An integer vector of locus identifiers.</p>
</td></tr>
<tr><td><code id="risk_alleles_tbl_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of variant identifiers.</p>
</td></tr>
<tr><td><code id="risk_alleles_tbl_+3A_risk_allele">risk_allele</code></td>
<td>
<p>A character vector of risk or effect allele names.</p>
</td></tr>
<tr><td><code id="risk_alleles_tbl_+3A_risk_frequency">risk_frequency</code></td>
<td>
<p>A numeric vector of the frequency of risk or effect
alleles.</p>
</td></tr>
<tr><td><code id="risk_alleles_tbl_+3A_genome_wide">genome_wide</code></td>
<td>
<p>A logical vector.</p>
</td></tr>
<tr><td><code id="risk_alleles_tbl_+3A_limited_list">limited_list</code></td>
<td>
<p>A logical vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='s4_to_list'>Convert an S4 object into a list</h2><span id='topic+s4_to_list'></span>

<h3>Description</h3>

<p>Convert an S4 object into a list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>s4_to_list(s4_obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="s4_to_list_+3A_s4_obj">s4_obj</code></td>
<td>
<p>an S4 object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list version of the S4 object.
</p>

<hr>
<h2 id='set_testing_fast'>Setup the environment to skip slow tests</h2><span id='topic+set_testing_fast'></span>

<h3>Description</h3>

<p>This function sets an environment variable <code>TEST_FAST</code> to 'true' which
is used to flag if a test should be skipped or not. If you place
<code><a href="#topic+skip_if_testing_is_fast">skip_if_testing_is_fast</a></code> in a test then it will
check if <code>TEST_FAST</code> is 'true', if it is, then it will skip the next
expectations within a <code>test_that()</code> block.
Please note that this function is to be used in gwasrapidd development.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_testing_fast()
</code></pre>


<h3>Value</h3>

<p>Returns <code>NULL</code>. This function should be used for its side
effect.
</p>

<hr>
<h2 id='set_testing_slow'>Setup the environment to still run slow tests</h2><span id='topic+set_testing_slow'></span>

<h3>Description</h3>

<p>This function sets an environment variable <code>TEST_FAST</code> to 'false' which
is used to flag if a test should be skipped or not. If you place
<code><a href="#topic+skip_if_testing_is_fast">skip_if_testing_is_fast</a></code> in a test then it will
check if <code>TEST_FAST</code> is 'true', if it is, then it will skip the next
expectations with a <code>test_that()</code> block. Please note that this function
is to be used in gwasrapidd development.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_testing_slow()
</code></pre>


<h3>Value</h3>

<p>Returns <code>NULL</code>. This function should be used for its side
effect.
</p>

<hr>
<h2 id='setop'>Set operations on GWAS Catalog objects.</h2><span id='topic+setop'></span><span id='topic+union'></span><span id='topic+intersect'></span><span id='topic+setdiff'></span><span id='topic+setequal'></span>

<h3>Description</h3>

<p>Performs set union, intersection, and (asymmetric!) difference on two objects
of either class <a href="#topic+studies-class">studies</a>, <a href="#topic+associations-class">associations</a>,
<a href="#topic+variants-class">variants</a>, or <a href="#topic+traits-class">traits</a>. Note that <code>union()</code>
removes duplicated entities, whereas <code><a href="#topic+bind">bind</a>()</code> does
not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>union(x, y, ...)

intersect(x, y, ...)

setdiff(x, y, ...)

setequal(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setop_+3A_x">x</code>, <code id="setop_+3A_y">y</code></td>
<td>
<p>Objects of either class <a href="#topic+studies-class">studies</a>,
<a href="#topic+associations-class">associations</a>, <a href="#topic+variants-class">variants</a>, or
<a href="#topic+traits-class">traits</a>.</p>
</td></tr>
<tr><td><code id="setop_+3A_...">...</code></td>
<td>
<p>other arguments passed on to methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same class as <code>x</code> and <code>y</code>, i.e.,
<a href="#topic+studies-class">studies</a>, <a href="#topic+associations-class">associations</a>, <a href="#topic+variants-class">variants</a>,
or <a href="#topic+traits-class">traits</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# union()
#
# Combine studies and remove duplicates
union(studies_ex01, studies_ex02)

# Combine associations and remove duplicates
union(associations_ex01, associations_ex02)

# Combine variants and remove duplicates
union(variants_ex01, variants_ex02)

# Combine traits and remove duplicates
union(traits_ex01, traits_ex02)

#
# intersect()
#
# Intersect common studies
intersect(studies_ex01, studies_ex02)

# Intersect common associations
intersect(associations_ex01, associations_ex02)

# Intersect common variants
intersect(variants_ex01, variants_ex02)

# Intersect common traits
intersect(traits_ex01, traits_ex02)

#
# setdiff()
#
# Remove studies from ex01 that are also present in ex02
setdiff(studies_ex01, studies_ex02)

# Remove associations from ex01 that are also present in ex02
setdiff(associations_ex01, associations_ex02)

# Remove variants from ex01 that are also present in ex02
setdiff(variants_ex01, variants_ex02)

# Remove traits from ex01 that are also present in ex02
setdiff(traits_ex01, traits_ex02)

#
# setequal()
#
# Compare two studies objects
setequal(studies_ex01, studies_ex01)
setequal(studies_ex01, studies_ex02)

# Compare two associations objects
setequal(associations_ex01, associations_ex01)
setequal(associations_ex01, associations_ex02)

# Compare two variants objects
setequal(variants_ex01, variants_ex01)
setequal(variants_ex01, variants_ex02)

# Compare two traits objects
setequal(traits_ex01, traits_ex01)
setequal(traits_ex01, traits_ex02)

</code></pre>

<hr>
<h2 id='skip_if_testing_is_fast'>Skips a test if TEST_FAST is 'true'</h2><span id='topic+skip_if_testing_is_fast'></span>

<h3>Description</h3>

<p>This function checks the value of <code>TEST_FAST</code>, if it is 'true' it skips
the test otherwise it still runs the following expectations. Please note that
this function is to be used in gwasrapidd development.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>skip_if_testing_is_fast()
</code></pre>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if the environment variable <code>'TEST_FAST'</code> is
set to <code>'false'</code>, or does not return anything but triggers the side
effect of skipping the next test (with testthat).
</p>

<hr>
<h2 id='studies'>Constructor for the S4 studies object.</h2><span id='topic+studies'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+studies-class">studies</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>studies(
  studies = studies_tbl(),
  genotyping_techs = genotyping_techs_tbl(),
  platforms = platforms_tbl(),
  ancestries = ancestries_tbl(),
  ancestral_groups = ancestral_groups_tbl(),
  countries_of_origin = countries_tbl(),
  countries_of_recruitment = countries_tbl(),
  publications = publications_tbl()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="studies_+3A_studies">studies</code></td>
<td>
<p>A <code><a href="#topic+studies_tbl">studies_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="studies_+3A_genotyping_techs">genotyping_techs</code></td>
<td>
<p>A <code><a href="#topic+genotyping_techs_tbl">genotyping_techs_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="studies_+3A_platforms">platforms</code></td>
<td>
<p>A <code><a href="#topic+platforms_tbl">platforms_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="studies_+3A_ancestries">ancestries</code></td>
<td>
<p>A <code><a href="#topic+ancestries_tbl">ancestries_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="studies_+3A_ancestral_groups">ancestral_groups</code></td>
<td>
<p>A <code><a href="#topic+ancestral_groups_tbl">ancestral_groups_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="studies_+3A_countries_of_origin">countries_of_origin</code></td>
<td>
<p>A <code><a href="#topic+countries_tbl">countries_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="studies_+3A_countries_of_recruitment">countries_of_recruitment</code></td>
<td>
<p>A <code><a href="#topic+countries_tbl">countries_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="studies_+3A_publications">publications</code></td>
<td>
<p>A <code><a href="#topic+publications_tbl">publications_tbl</a></code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+studies-class">studies</a>.
</p>

<hr>
<h2 id='studies_drop_na'>Drop any NA studies.</h2><span id='topic+studies_drop_na'></span>

<h3>Description</h3>

<p>This function takes a studies S4 object and removes any study identifiers
that might have been NA. This ensures that there is always a non-NA
<code>study_id</code> value in all tables. This is important as the <code>study_id</code>
is the primary key.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>studies_drop_na(s4_studies)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="studies_drop_na_+3A_s4_studies">s4_studies</code></td>
<td>
<p>An object of class <a href="#topic+studies-class">studies</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+studies-class">studies</a>.
</p>

<hr>
<h2 id='studies_tbl'>Creates a studies table.</h2><span id='topic+studies_tbl'></span>

<h3>Description</h3>

<p>Creates a studies table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>studies_tbl(
  study_id = character(),
  reported_trait = character(),
  initial_sample_size = character(),
  replication_sample_size = character(),
  gxe = logical(),
  gxg = logical(),
  snp_count = integer(),
  qualifier = character(),
  imputed = logical(),
  pooled = logical(),
  study_design_comment = character(),
  full_pvalue_set = logical(),
  user_requested = logical()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="studies_tbl_+3A_study_id">study_id</code></td>
<td>
<p>GWAS Catalog study accession identifier.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_reported_trait">reported_trait</code></td>
<td>
<p>Phenotypic trait as reported by the authors of the
study.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_initial_sample_size">initial_sample_size</code></td>
<td>
<p>Free text description of the initial cohort sample
size.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_replication_sample_size">replication_sample_size</code></td>
<td>
<p>Free text description of the replication
cohort sample size.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_gxe">gxe</code></td>
<td>
<p>Whether the study investigates a gene-environment interaction.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_gxg">gxg</code></td>
<td>
<p>Whether the study investigates a gene-gene interaction.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_snp_count">snp_count</code></td>
<td>
<p>Number of variants passing quality control.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_qualifier">qualifier</code></td>
<td>
<p>Qualifier of number of variants passing quality control.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_imputed">imputed</code></td>
<td>
<p>Whether variants were imputed.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_pooled">pooled</code></td>
<td>
<p>Whether samples were pooled.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_study_design_comment">study_design_comment</code></td>
<td>
<p>Any other relevant study design information.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_full_pvalue_set">full_pvalue_set</code></td>
<td>
<p>Whether full summary statistics are available for this
study.</p>
</td></tr>
<tr><td><code id="studies_tbl_+3A_user_requested">user_requested</code></td>
<td>
<p>Whether the addition of this study to the GWAS Catalog
was requested by a user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='studies-class'>An S4 class to represent a set of GWAS Catalog studies</h2><span id='topic+studies-class'></span>

<h3>Description</h3>

<p>The studies object consists of eight slots, each a table
(<code><a href="tibble.html#topic+tibble">tibble</a></code>), that combined form a relational database of a
subset of GWAS Catalog studies. Each study is an observation (row) in the
<code>studies</code> table &mdash; main table. All tables have the column
<code>study_id</code> as primary key.
</p>


<h3>Slots</h3>


<dl>
<dt><code>studies</code></dt><dd>
<dl>
<dt>study_id</dt><dd><p>GWAS Catalog study accession identifier, e.g.,
<code>"GCST002735"</code>.</p>
</dd>
<dt>reported_trait</dt><dd><p>Phenotypic trait as reported by the authors of the
study, e.g. <code>"Breast cancer"</code>.</p>
</dd>
<dt>initial_sample_size</dt><dd><p>Free text description of the initial cohort sample
size.</p>
</dd>
<dt>replication_sample_size</dt><dd><p>Free text description of the replication
cohort sample size.</p>
</dd>
<dt>gxe</dt><dd><p>Whether the study investigates a gene-environment interaction.</p>
</dd>
<dt>gxg</dt><dd><p>Whether the study investigates a gene-gene interaction.</p>
</dd>
<dt>snp_count</dt><dd><p>Number of variants passing quality control.</p>
</dd>
<dt>qualifier</dt><dd><p>Qualifier of number of variants passing quality control.</p>
</dd>
<dt>imputed</dt><dd><p>Whether variants were imputed.</p>
</dd>
<dt>pooled</dt><dd><p>Whether samples were pooled.</p>
</dd>
<dt>study_design_comment</dt><dd><p>Any other relevant study design information.</p>
</dd>
<dt>full_pvalue_set</dt><dd><p>Whether full summary statistics are available for this
study.</p>
</dd>
<dt>user_requested</dt><dd><p>Whether the addition of this study to the GWAS Catalog
was requested by a user.</p>
</dd>
</dl>
</dd>
<dt><code>genotyping_techs</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing genotyping
technologies employed in each study. Columns:
</p>

<dl>
<dt>study_id</dt><dd><p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>genotyping_technology</dt><dd><p>Genotyping technology employed, e.g.
<code>"Exome genotyping array"</code>, <code>"Exome-wide sequencing"</code>,
<code>"Genome-wide genotyping array"</code>, <code>"Genome-wide sequencing"</code>, or
<code>"Targeted genotyping array"</code>.</p>
</dd>
</dl>
</dd>
<dt><code>platforms</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing platforms used per
study.
</p>

<dl>
<dt>study_id</dt><dd><p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>manufacturer</dt><dd><p>Platform manufacturer, e.g., <code>"Affymetrix"</code>,
<code>"Illumina"</code>, or <code>"Perlegen"</code>.</p>
</dd>
</dl>
</dd>
<dt><code>ancestries</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing ancestry of samples
used in each study.
</p>

<dl>
<dt>study_id</dt><dd><p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>ancestry_id</dt><dd><p>Ancestry identifier.</p>
</dd>
<dt>type</dt><dd><p>Stage of the ancestry sample: either <code>'initial'</code> or
<code>'replication'</code>.</p>
</dd>
<dt>number_of_individuals</dt><dd><p>Number of individuals comprising this ancestry
sample.</p>
</dd>
</dl>
</dd>
<dt><code>ancestral_groups</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing ancestral
groups used in each ancestry.
</p>

<dl>
<dt>study_id</dt><dd><p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>ancestry_id</dt><dd><p>Ancestry identifier.</p>
</dd>
<dt>ancestral_group</dt><dd><p>Genetic ancestry groups present in the sample.</p>
</dd>
</dl>
</dd>
<dt><code>countries_of_origin</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing countries of
origin of samples.
</p>

<dl>
<dt>study_id</dt><dd><p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>ancestry_id</dt><dd><p>Ancestry identifier.</p>
</dd>
<dt>country_name</dt><dd><p>Country name, according to
<a href="https://unstats.un.org/unsd/methodology/m49/overview/">The United
Nations M49 Standard of Geographic Regions.</a></p>
</dd>
<dt>major_area</dt><dd><p>Region name, according to
<a href="https://unstats.un.org/unsd/methodology/m49/overview/">The United
Nations M49 Standard of Geographic Regions.</a></p>
</dd> <dt>region</dt><dd><p>Sub-region name,
according to <a href="https://unstats.un.org/unsd/methodology/m49/overview/">The
United Nations M49 Standard of Geographic Regions.</a></p>
</dd>
</dl>
</dd>
<dt><code>countries_of_recruitment</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing
countries of recruitment of samples.
</p>

<dl>
<dt>study_id</dt><dd><p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>ancestry_id</dt><dd><p>Ancestry identifier.</p>
</dd>
<dt>country_name</dt><dd><p>Country name, according to
<a href="https://unstats.un.org/unsd/methodology/m49/overview/">The United
Nations M49 Standard of Geographic Regions.</a></p>
</dd>
<dt>major_area</dt><dd><p>Region name, according to
<a href="https://unstats.un.org/unsd/methodology/m49/overview/">The United
Nations M49 Standard of Geographic Regions.</a></p>
</dd> <dt>region</dt><dd><p>Sub-region name,
according to <a href="https://unstats.un.org/unsd/methodology/m49/overview/">The
United Nations M49 Standard of Geographic Regions.</a></p>
</dd>
</dl>
</dd>
<dt><code>publications</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing publications
associated with each study.
</p>

<dl>
<dt>study_id</dt><dd><p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>pubmed_id</dt><dd><p><a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a>
identifier.</p>
</dd>
<dt>publication_date</dt><dd><p>Publication date (online date if available) formatted
as <code><a href="lubridate.html#topic+ymd">ymd</a></code>.</p>
</dd>
<dt>publication</dt><dd><p>Abbreviated journal name.</p>
</dd>
<dt>title</dt><dd><p>Publication title.</p>
</dd>
<dt>author_fullname</dt><dd><p>Last name and initials of first author.</p>
</dd>
<dt>author_orcid</dt><dd><p>Author's <a href="https://en.wikipedia.org/wiki/ORCID">ORCID
iD</a> (Open Researcher and Contributor ID).</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='study_to_association'>Map a study id to an association id</h2><span id='topic+study_to_association'></span>

<h3>Description</h3>

<p>Map a study accession identifier to an association accession identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>study_to_association(study_id, verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="study_to_association_+3A_study_id">study_id</code></td>
<td>
<p>A character vector of study accession identifiers.</p>
</td></tr>
<tr><td><code id="study_to_association_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="study_to_association_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two identifiers. First column is the study identifier
and the second column is the association identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map GWAS study identifiers to association identifiers
study_to_association(c('GCST001084', 'GCST001085'))

## End(Not run)

</code></pre>

<hr>
<h2 id='study_to_trait'>Map a study id to a EFO trait id</h2><span id='topic+study_to_trait'></span>

<h3>Description</h3>

<p>Map a study accession identifier to a EFO trait identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>study_to_trait(study_id, verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="study_to_trait_+3A_study_id">study_id</code></td>
<td>
<p>A character vector of study accession identifiers.</p>
</td></tr>
<tr><td><code id="study_to_trait_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="study_to_trait_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two identifiers. First column is the study identifier
and the second column is the EFO identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map GWAS study identifiers to EFO trait identifiers
study_to_trait(c('GCST001084', 'GCST001085'))

## End(Not run)

</code></pre>

<hr>
<h2 id='study_to_variant'>Map a study id to a variant id</h2><span id='topic+study_to_variant'></span>

<h3>Description</h3>

<p>Map a study accession identifier to a variant accession identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>study_to_variant(study_id, verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="study_to_variant_+3A_study_id">study_id</code></td>
<td>
<p>A character vector of study accession identifiers.</p>
</td></tr>
<tr><td><code id="study_to_variant_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="study_to_variant_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two identifiers. First column is the study identifier
and the second column is the variant identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map GWAS study identifiers to variant identifiers
study_to_variant(c('GCST001084', 'GCST001085'))

## End(Not run)

</code></pre>

<hr>
<h2 id='subset-associations'>Subset an associations object</h2><span id='topic+subset-associations'></span><span id='topic++5B+2Cassociations+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cassociations+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cassociations+2Ccharacter+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+associations-class">associations</a> by identifier or by position using the
<code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'associations,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'associations,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'associations,character,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset-associations_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+associations-class">associations</a> object.</p>
</td></tr>
<tr><td><code id="subset-associations_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-associations_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-associations_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-associations_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+associations-class">associations</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Subset an associations object by identifier
associations_ex01['22505']

# Or by its position in table associations
associations_ex01[2]

# Keep all associations except the second
associations_ex01[-2]

</code></pre>

<hr>
<h2 id='subset-studies'>Subset a studies object</h2><span id='topic+subset-studies'></span><span id='topic++5B+2Cstudies+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cstudies+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cstudies+2Ccharacter+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+studies-class">studies</a> by identifier or by position using the
<code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'studies,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'studies,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'studies,character,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset-studies_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+studies-class">studies</a> object.</p>
</td></tr>
<tr><td><code id="subset-studies_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-studies_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-studies_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-studies_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+studies-class">studies</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Subset a studies object by identifier
studies_ex01['GCST001585']

# Or by its position in table studies
studies_ex01[1]

# Keep all studies except the first
studies_ex01[-1]

</code></pre>

<hr>
<h2 id='subset-traits'>Subset a traits object</h2><span id='topic+subset-traits'></span><span id='topic++5B+2Ctraits+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Ctraits+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Ctraits+2Ccharacter+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+traits-class">traits</a> by identifier or by position using the
<code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'traits,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'traits,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'traits,character,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset-traits_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+traits-class">traits</a> object.</p>
</td></tr>
<tr><td><code id="subset-traits_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-traits_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-traits_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-traits_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+traits-class">traits</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Subset a traits object by identifier
traits_ex01['EFO_0004884']

# Or by its position in table traits
traits_ex01[1]

# Keep all traits except the second
traits_ex01[-2]

</code></pre>

<hr>
<h2 id='subset-variants'>Subset a variants object</h2><span id='topic+subset-variants'></span><span id='topic++5B+2Cvariants+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cvariants+2Cnumeric+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cvariants+2Ccharacter+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>You can subset <a href="#topic+variants-class">variants</a> by identifier or by position using the
<code>`[`</code> operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'variants,missing,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'variants,numeric,missing,missing'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'variants,character,missing,missing'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset-variants_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+variants-class">variants</a> object.</p>
</td></tr>
<tr><td><code id="subset-variants_+3A_i">i</code></td>
<td>
<p>Position of the identifier or the name of the identifier itself.</p>
</td></tr>
<tr><td><code id="subset-variants_+3A_j">j</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="subset-variants_+3A_...">...</code></td>
<td>
<p>Additional arguments not used here.</p>
</td></tr>
<tr><td><code id="subset-variants_+3A_drop">drop</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+variants-class">variants</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Subset a variants object by identifier
variants_ex01['rs4725504']

# Or by its position in table variants
variants_ex01[3]

# Keep all variants except the third
variants_ex01[-3]

</code></pre>

<hr>
<h2 id='sure'>Are you sure?</h2><span id='topic+sure'></span>

<h3>Description</h3>

<p>This function asks you interactively for permission to continue or not. You
can specify a custom message before the question and also different messages
for a both a positive and negative answer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sure(
  before_question = NULL,
  after_saying_no = NULL,
  after_saying_yes = NULL,
  default_answer = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sure_+3A_before_question">before_question</code></td>
<td>
<p>String with message to be printed before question.</p>
</td></tr>
<tr><td><code id="sure_+3A_after_saying_no">after_saying_no</code></td>
<td>
<p>String with message to be printed after answering
<code>'no'</code>.</p>
</td></tr>
<tr><td><code id="sure_+3A_after_saying_yes">after_saying_yes</code></td>
<td>
<p>String with message to be printed after answering
<code>'yes'</code>.</p>
</td></tr>
<tr><td><code id="sure_+3A_default_answer">default_answer</code></td>
<td>
<p>String with answer to question, if run in
non-interactive mode.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you run this function in non-interactive mode, you should pass an
automatic answer to <code>default_answer</code>: <code>'yes'</code> or <code>'no'</code>.
</p>


<h3>Value</h3>

<p>A logical indicating if answer was <code>'yes'</code>/<code>'y'</code>
(<code>TRUE</code>) or otherwise (<code>FALSE</code>).
</p>

<hr>
<h2 id='trait_to_association'>Map an EFO trait id to an association id</h2><span id='topic+trait_to_association'></span>

<h3>Description</h3>

<p>Map an EFO trait id to an association identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trait_to_association(efo_id, verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trait_to_association_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of EFO trait identifiers.</p>
</td></tr>
<tr><td><code id="trait_to_association_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="trait_to_association_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two identifiers. First column is the EFO trait
identifier and the second column is the association identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map EFO trait identifiers to association identifiers
trait_to_association(c('EFO_0005108', 'EFO_0005109'))

## End(Not run)

</code></pre>

<hr>
<h2 id='trait_to_study'>Map an EFO trait id to a study id</h2><span id='topic+trait_to_study'></span>

<h3>Description</h3>

<p>Map an EFO trait id to a study accession identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trait_to_study(efo_id, verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trait_to_study_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of EFO trait identifiers.</p>
</td></tr>
<tr><td><code id="trait_to_study_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="trait_to_study_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two identifiers. First column is the EFO trait
identifier and the second column is the study identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map EFO trait identifiers to study identifiers
trait_to_study(c('EFO_0005108', 'EFO_0005109'))

## End(Not run)

</code></pre>

<hr>
<h2 id='trait_to_variant'>Map an EFO trait id to a variant id</h2><span id='topic+trait_to_variant'></span>

<h3>Description</h3>

<p>Map an EFO trait id to a variant identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trait_to_variant(efo_id, verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trait_to_variant_+3A_efo_id">efo_id</code></td>
<td>
<p>A character vector of EFO trait identifiers.</p>
</td></tr>
<tr><td><code id="trait_to_variant_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="trait_to_variant_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two identifiers. First column is the EFO trait
identifier and the second column is the variant identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map EFO trait identifiers to variant identifiers
trait_to_variant('EFO_0005229')

## End(Not run)

</code></pre>

<hr>
<h2 id='traits'>Constructor for the S4 traits object.</h2><span id='topic+traits'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+traits-class">traits</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>traits(traits = traits_tbl())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="traits_+3A_traits">traits</code></td>
<td>
<p>A <code><a href="#topic+traits_tbl">traits_tbl</a></code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+traits-class">traits</a>.
</p>

<hr>
<h2 id='traits_drop_na'>Drop any NA traits.</h2><span id='topic+traits_drop_na'></span>

<h3>Description</h3>

<p>This function takes a traits S4 object and removes any EFO trait identifiers
that might have been NA. This ensures that there is always a non-NA
<code>efo_id</code> value in all tables. This is important as the <code>efo_id</code>
is the primary key.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>traits_drop_na(s4_traits)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="traits_drop_na_+3A_s4_traits">s4_traits</code></td>
<td>
<p>An object of class <a href="#topic+traits-class">traits</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+traits-class">traits</a>.
</p>

<hr>
<h2 id='traits_tbl'>Creates a traits tibble</h2><span id='topic+traits_tbl'></span>

<h3>Description</h3>

<p>Creates a traits tibble.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>traits_tbl(efo_id = character(), trait = character(), uri = character())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="traits_tbl_+3A_efo_id">efo_id</code></td>
<td>
<p>A <code>character</code> vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> identifiers.</p>
</td></tr>
<tr><td><code id="traits_tbl_+3A_trait">trait</code></td>
<td>
<p>A <code>character</code> vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> trait descriptions.</p>
</td></tr>
<tr><td><code id="traits_tbl_+3A_uri">uri</code></td>
<td>
<p>A <code>character</code> vector of
<a href="https://www.ebi.ac.uk/efo/">EFO</a> URIs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='traits-class'>An S4 class to represent a set of GWAS Catalog EFO traits.</h2><span id='topic+traits-class'></span>

<h3>Description</h3>

<p>The traits object consists of one slot only, a table
(<code><a href="tibble.html#topic+tibble">tibble</a></code>) of GWAS Catalog EFO traits. Each EFO trait is
an observation (row) in the <code>traits</code> table &mdash; main table.
</p>


<h3>Slots</h3>


<dl>
<dt><code>traits</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing EFO traits. Columns:
</p>

<dl>
<dt>efo_id</dt><dd><p><a href="https://www.ebi.ac.uk/efo/">EFO</a> identifier.</p>
</dd>
<dt>trait</dt><dd><p><a href="https://www.ebi.ac.uk/efo/">EFO</a> trait description.</p>
</dd>
<dt>uri</dt><dd><p>The full URI of the <a href="https://www.ebi.ac.uk/efo/">EFO</a> term.</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='tws'>Trim whitespace.</h2><span id='topic+tws'></span>

<h3>Description</h3>

<p>This function removes leading and trailing white space from strings. Note:
this function does no checking on input for performance reasons. So make sure
the input is really a character vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tws(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tws_+3A_x">x</code></td>
<td>
<p>A <code><a href="base.html#topic+character">character</a></code> vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector.
</p>

<hr>
<h2 id='v_ensembl_ids_tbl'>Creates a gene Ensembl identifiers table.</h2><span id='topic+v_ensembl_ids_tbl'></span>

<h3>Description</h3>

<p>Creates a gene Ensembl identifiers table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>v_ensembl_ids_tbl(
  variant_id = character(),
  gene_name = character(),
  ensembl_id = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="v_ensembl_ids_tbl_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of variant identifiers.</p>
</td></tr>
<tr><td><code id="v_ensembl_ids_tbl_+3A_gene_name">gene_name</code></td>
<td>
<p>A character vector of gene symbols according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</td></tr>
<tr><td><code id="v_ensembl_ids_tbl_+3A_ensembl_id">ensembl_id</code></td>
<td>
<p>A character vector of Ensembl identifiers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='v_entrez_ids_tbl'>Creates a gene Entrez identifiers table.</h2><span id='topic+v_entrez_ids_tbl'></span>

<h3>Description</h3>

<p>Creates a gene Entrez identifiers table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>v_entrez_ids_tbl(
  variant_id = character(),
  gene_name = character(),
  entrez_id = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="v_entrez_ids_tbl_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of variant identifiers.</p>
</td></tr>
<tr><td><code id="v_entrez_ids_tbl_+3A_gene_name">gene_name</code></td>
<td>
<p>A character vector of gene symbols according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</td></tr>
<tr><td><code id="v_entrez_ids_tbl_+3A_entrez_id">entrez_id</code></td>
<td>
<p>A character vector of Entrez identifiers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='variant_name'>Extract variant identifiers from strings of the form rs123-G</h2><span id='topic+variant_name'></span>

<h3>Description</h3>

<p>This function parses strings of the form <code>"rs123-G"</code> and returns
the name of the variant; it uses the regex <code>-([ATCG\?]+)?$</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variant_name(risk_allele_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variant_name_+3A_risk_allele_names">risk_allele_names</code></td>
<td>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of variant names.
</p>

<hr>
<h2 id='variant_to_association'>Map a variant id to an association id</h2><span id='topic+variant_to_association'></span>

<h3>Description</h3>

<p>Map a variant identifier to an association identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variant_to_association(variant_id, verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variant_to_association_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of variant identifiers.</p>
</td></tr>
<tr><td><code id="variant_to_association_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="variant_to_association_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two identifiers. First column is the variant
identifier and the second column is the association identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map GWAS variant identifiers to association identifiers
variant_to_association(c('rs7904579', 'rs138331350'))

## End(Not run)

</code></pre>

<hr>
<h2 id='variant_to_study'>Map a variant id to a study id</h2><span id='topic+variant_to_study'></span>

<h3>Description</h3>

<p>Map a variant identifier to a study accession identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variant_to_study(variant_id, verbose = FALSE, warnings = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variant_to_study_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of variant identifiers.</p>
</td></tr>
<tr><td><code id="variant_to_study_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="variant_to_study_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two identifiers. First column is the variant
identifier and the second column is the study identifier.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map GWAS variant identifiers to study identifiers
variant_to_study(c('rs7904579', 'rs138331350'))

## End(Not run)

</code></pre>

<hr>
<h2 id='variant_to_trait'>Map a variant id to an EFO trait</h2><span id='topic+variant_to_trait'></span>

<h3>Description</h3>

<p>Map a variant identifier to an EFO trait identifier. Variants are first
mapped to association identifiers, and then to EFO traits. Set the option
<code>keep_association_id</code> to <code>TRUE</code> to keep the intermediate mapping,
i.e., the association identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variant_to_trait(
  variant_id,
  keep_association_id = FALSE,
  verbose = FALSE,
  warnings = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variant_to_trait_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of variant identifiers.</p>
</td></tr>
<tr><td><code id="variant_to_trait_+3A_keep_association_id">keep_association_id</code></td>
<td>
<p>Whether to keep the association identifier
in the final output (default is <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="variant_to_trait_+3A_verbose">verbose</code></td>
<td>
<p>Whether the function should be
verbose about the different queries or not.</p>
</td></tr>
<tr><td><code id="variant_to_trait_+3A_warnings">warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of two or three identifiers. If
<code>keep_association_id</code> is set to <code>FALSE</code>, the first column is the
variant identifier and the second column is the EFO trait identifier,
otherwise the variable <code>association_id</code> is also included as the second
column.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Map GWAS variant identifiers to EFO trait identifiers
variant_to_trait(c('rs7904579', 'rs138331350'))

# Map GWAS variant identifiers to EFO trait identifiers
# but keep the intermediate association identifier
variant_to_trait(c('rs7904579', 'rs138331350'), keep_association_id = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='variants'>Constructor for the S4 variants object.</h2><span id='topic+variants'></span>

<h3>Description</h3>

<p>Constructor for the S4 <a href="#topic+variants-class">variants</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variants(
  variants = variants_tbl(),
  genomic_contexts = genomic_contexts_tbl(),
  ensembl_ids = v_ensembl_ids_tbl(),
  entrez_ids = v_entrez_ids_tbl()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variants_+3A_variants">variants</code></td>
<td>
<p>A <code><a href="#topic+variants_tbl">variants_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="variants_+3A_genomic_contexts">genomic_contexts</code></td>
<td>
<p>A <code><a href="#topic+genomic_contexts_tbl">genomic_contexts_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="variants_+3A_ensembl_ids">ensembl_ids</code></td>
<td>
<p>A <code><a href="#topic+v_ensembl_ids_tbl">v_ensembl_ids_tbl</a></code> tibble.</p>
</td></tr>
<tr><td><code id="variants_+3A_entrez_ids">entrez_ids</code></td>
<td>
<p>A <code><a href="#topic+v_entrez_ids_tbl">v_entrez_ids_tbl</a></code> tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+variants-class">variants</a>.
</p>

<hr>
<h2 id='variants_drop_na'>Drop any NA variants.</h2><span id='topic+variants_drop_na'></span>

<h3>Description</h3>

<p>This function takes a variants S4 object and removes any variant identifiers
that might have been NA. This ensures that there is always a non-NA
<code>variant_id</code> value in all tables. This is important as the
<code>variant_id</code> is the primary key.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variants_drop_na(s4_variants)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variants_drop_na_+3A_s4_variants">s4_variants</code></td>
<td>
<p>An object of class <a href="#topic+variants-class">variants</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <a href="#topic+variants-class">variants</a>.
</p>

<hr>
<h2 id='variants_tbl'>Creates a variants table.</h2><span id='topic+variants_tbl'></span>

<h3>Description</h3>

<p>Creates a variants table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variants_tbl(
  variant_id = character(),
  merged = integer(),
  functional_class = character(),
  chromosome_name = character(),
  chromosome_position = integer(),
  chromosome_region = character(),
  last_update_date = lubridate::ymd_hms()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variants_tbl_+3A_variant_id">variant_id</code></td>
<td>
<p>A character vector of variant identifiers.</p>
</td></tr>
<tr><td><code id="variants_tbl_+3A_merged">merged</code></td>
<td>
<p>A logical vector indicating if a SNP has been merged with
another SNP in a newer genome build.</p>
</td></tr>
<tr><td><code id="variants_tbl_+3A_functional_class">functional_class</code></td>
<td>
<p>A character vector of functional classes, see
<code>functional_class</code> in slot <code>variants</code> of <a href="#topic+variants-class">variants</a>.</p>
</td></tr>
<tr><td><code id="variants_tbl_+3A_chromosome_name">chromosome_name</code></td>
<td>
<p>A character vector of chromosome names.</p>
</td></tr>
<tr><td><code id="variants_tbl_+3A_chromosome_position">chromosome_position</code></td>
<td>
<p>An integer vector of chromosome positions.</p>
</td></tr>
<tr><td><code id="variants_tbl_+3A_chromosome_region">chromosome_region</code></td>
<td>
<p>A character vector of cytogenetic regions.</p>
</td></tr>
<tr><td><code id="variants_tbl_+3A_last_update_date">last_update_date</code></td>
<td>
<p>A <code><a href="base.html#topic+DateTimeClasses">POSIXct</a></code> object
indicating the last time the variants have been updated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> whose columns are the named arguments
to the function.
</p>

<hr>
<h2 id='variants-class'>An S4 class to represent a set of GWAS Catalog variants</h2><span id='topic+variants-class'></span>

<h3>Description</h3>

<p>The variants object consists of four slots, each a table
(<code><a href="tibble.html#topic+tibble">tibble</a></code>), that combined form a relational database of a
subset of GWAS Catalog variants. Each variant is an observation (row) in the
<code>variants</code> table &mdash; main table. All tables have the column
<code>variant_id</code> as primary key.
</p>


<h3>Slots</h3>


<dl>
<dt><code>variants</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing variants. Columns:
</p>

<dl>
<dt>variant_id</dt><dd><p>Variant identifier, e.g., <code>'rs1333048'</code>.</p>
</dd>
<dt>merged</dt><dd><p>Whether this SNP has been merged with another SNP in a newer
genome build.</p>
</dd>
<dt>functional_class</dt><dd><p>Class according to Ensembl's predicted consequences
that each variant allele may have on transcripts. See
<a href="https://www.ensembl.org/info/genome/variation/prediction/predicted_data.html">Ensembl
Variation - Calculated variant consequences</a>.</p>
</dd>
<dt>chromosome_name</dt><dd><p>Chromosome name.</p>
</dd>
<dt>chromosome_position</dt><dd><p>Chromosome position.</p>
</dd>
<dt>chromosome_region</dt><dd><p><a href="https://medlineplus.gov/genetics/understanding/howgeneswork/genelocation/">Cytogenetic
location</a>.</p>
</dd>
<dt>last_update_date</dt><dd><p>Last time this variant was updated.</p>
</dd>
</dl>
</dd>
<dt><code>genomic_contexts</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing genomic
contexts associated with each variant. Columns:
</p>

<dl>
<dt>variant_id</dt><dd><p>Variant identifier.</p>
</dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</dd>
<dt>chromosome_name</dt><dd><p>Chromosome name.</p>
</dd>
<dt>chromosome_position</dt><dd><p>Chromosome position.</p>
</dd>
<dt>distance</dt><dd><p>Genomic distance between the variant and the gene (in base
pairs).</p>
</dd>
<dt>is_mapped_gene</dt><dd><p>Whether this is a mapped gene to this variant. A mapped
gene is either an overlapping gene with the variant or the two closest genes
upstream and downstream of the variant. Moreover, only genes whose mapping
source is 'Ensembl' are considered.</p>
</dd>
<dt>is_closest_gene</dt><dd><p>Whether this is the closest gene to this variant.</p>
</dd>
<dt>is_intergenic</dt><dd><p>Whether this variant is intergenic, i.e, if there is no
gene up or downstream within 100kb.</p>
</dd>
<dt>is_upstream</dt><dd><p>Whether this variant is upstream of this gene.</p>
</dd>
<dt>is_downstream</dt><dd><p>Whether this variant is downstream of this gene.</p>
</dd>
<dt>source</dt><dd><p>Gene mapping source, either <code>Ensembl</code> or <code>NCBI</code>.</p>
</dd>
<dt>mapping_method</dt><dd><p>Gene mapping method.</p>
</dd>
</dl>
</dd>
<dt><code>ensembl_ids</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing gene Ensembl
identifiers associated with each genomic context. Columns:
</p>

<dl>
<dt>variant_id</dt><dd><p>Variant identifier.</p>
</dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</dd>
<dt>ensembl_id</dt><dd><p>The Ensembl identifier of an Ensembl gene, see Section
<a href="https://www.ensembl.org/info/genome/genebuild/index.html">Gene
annotation in Ensembl</a> for more information.</p>
</dd>
</dl>
</dd>
<dt><code>entrez_ids</code></dt><dd><p>A <code><a href="tibble.html#topic+tibble">tibble</a></code> listing gene Entrez
identifiers associated with each genomic context. Columns:
</p>

<dl>
<dt>variant_id</dt><dd><p>Variant identifier.</p>
</dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</dd>
<dt>entrez_id</dt><dd><p>The Entrez identifier of a gene, see ref.
<a href="https://doi.org/10.1093/nar/gkq1237">doi:10.1093/nar/gkq1237</a> for
more information.</p>
</dd>
</dl>
</dd>
</dl>

<hr>
<h2 id='write_xlsx'>Export a GWAS Catalog object to xlsx</h2><span id='topic+write_xlsx'></span>

<h3>Description</h3>

<p>This function exports a GWAS Catalog object to Microsoft Excel xlsx file.
Each table (slot) is saved in its own sheet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_xlsx(x, file = stop("`file` must be specified"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write_xlsx_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+studies-class">studies</a>, <a href="#topic+associations-class">associations</a>,
<a href="#topic+variants-class">variants</a> or <a href="#topic+traits-class">traits</a> object.</p>
</td></tr>
<tr><td><code id="write_xlsx_+3A_file">file</code></td>
<td>
<p>A file name to write to.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Although this function is run for its side effect of writing an xlsx
file, the path to the exported file is returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Initial setup
.old_wd &lt;- setwd(tempdir())

# Save an `associations` object, e.g. `associations_ex01`, to xlsx.
write_xlsx(associations_ex01, "associations.xlsx")

# Cleanup
unlink("associations.xlsx")
setwd(.old_wd)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
