<!DOCTYPE html><html><head><title>Help for package GWASinspector</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GWASinspector}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#compare_GWASs'><p>Compare summary statistics of GWAS result files</p></a></li>
<li><a href='#demo_inspector'><p>Runs an example QC</p></a></li>
<li><a href='#get_config'><p>Copies the template configuration file to the local machine</p></a></li>
<li><a href='#get_headerTranslation'><p>Copies the template header translation table to the local machine</p></a></li>
<li><a href='#GWASinspector'><p>Comprehensive and Easy to Use Quality Control of GWAS Results</p></a></li>
<li><a href='#Inspector-class'><p>An S4 class to represent the methods and parameters for inspecting a list of GWAS study result files.</p></a></li>
<li><a href='#manhattan_plot'><p>Creates the Manhattan plot</p></a></li>
<li><a href='#result_inspector'><p>Displays a brief report after running the QC pipeline</p></a></li>
<li><a href='#run_inspector'><p>Runs the QC pipeline on a set of GWAS result files</p></a></li>
<li><a href='#setup_inspector'><p>Importing a configuration file into R for setting the pipeline parameters</p></a></li>
<li><a href='#Study-class'><p>An S4 class to represent an inspected GWAS study result file.</p></a></li>
<li><a href='#StudyList-class'><p>An S4 class to represent a list of inspected GWAS study result files.</p></a></li>
<li><a href='#system_check'><p>Checks which required and optional packages are available</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Comprehensive and Easy to Use Quality Control of GWAS Results</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-27</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alireza Ani &lt;a.ani@umcg.nl&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ini (&ge; 0.3), futile.logger (&ge; 1.4), data.table (&ge; 1.12.8),
hash (&ge; 2.2), tools (&ge; 3.0), ggplot2 (&ge; 3.0), knitr (&ge;
1.1), rmarkdown (&ge; 2.17), gridExtra, grid, RSQLite, kableExtra
(&ge; 0.8), methods, R.utils, openxlsx (&ge; 4.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://GWASinspector.com">http://GWASinspector.com</a></td>
</tr>
<tr>
<td>Description:</td>
<td>When evaluating the results of a genome-wide association study (GWAS), it is important to perform a quality control to ensure that the results are valid, complete, correctly formatted, and, in case of meta-analysis, consistent with other studies that have applied the same analysis. This package was developed to facilitate and streamline this process and provide the user with a comprehensive report. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-27 19:03:52 UTC; Alireza</td>
</tr>
<tr>
<td>Author:</td>
<td>Alireza Ani [aut, cre],
  Peter J. van der Most [aut],
  Ahmad Vaez [aut],
  Ilja M. Nolte [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-27 20:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='compare_GWASs'>Compare summary statistics of GWAS result files</h2><span id='topic+compare_GWASs'></span>

<h3>Description</h3>

<p>This function compares the key metrics of previously inspected files.
This allows the user to check if the results of these studies are comparable (important when running a meta-analysis) and that there are no significant anomalies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare_GWASs(input.file.list, output.path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare_GWASs_+3A_input.file.list">input.file.list</code></td>
<td>
<p>list, full path of the Inspector result files. This file is in RDS format and will be generated for each GWAS result file during the inspection algorithm.</p>
</td></tr>
<tr><td><code id="compare_GWASs_+3A_output.path">output.path</code></td>
<td>
<p>character, full path to the folder where output files should be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Key metrics report and plots of previously inspected files are generated and saved in the specified output folder.
</p>

<hr>
<h2 id='demo_inspector'>Runs an example QC</h2><span id='topic+demo_inspector'></span>

<h3>Description</h3>

<p>This function runs the QC algorithm on a fabricated GWAS result file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_inspector(result.dir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="demo_inspector_+3A_result.dir">result.dir</code></td>
<td>
<p>character. Path to the output folder for saving QC result files</p>
</td></tr>
</table>


<h3>Value</h3>

<p>QC reports from running the algorithm on a sample GWAS file are generated and saved in the specified folder.
</p>

<hr>
<h2 id='get_config'>Copies the template configuration file to the local machine</h2><span id='topic+get_config'></span>

<h3>Description</h3>

<p>This templates should be edited and then used for setting up and running the QC pipeline.
Default filename is <strong>config.ini</strong>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_config(dir.path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_config_+3A_dir.path">dir.path</code></td>
<td>
<p>Path to the folder for saving a sample configuration file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Copies a sample configuration file (config.ini) in the specified folder.
</p>

<hr>
<h2 id='get_headerTranslation'>Copies the template header translation table to the local machine</h2><span id='topic+get_headerTranslation'></span>

<h3>Description</h3>

<p>This template file is used to translate a dataset's column names (the header) into the standard names used by GWASinspector.
The file contains a two-column table, with the left column containing the standard column-names and the right the alternatives.
Both the standard and alternative columns must be fully capitalized. This is a text file which includes most common variable/header names and can be edited according to user specifications.
The default filename is <strong>alt_headers.txt</strong>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_headerTranslation(dir.path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_headerTranslation_+3A_dir.path">dir.path</code></td>
<td>
<p>Path to the folder for saving a header-translation table file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Copies a sample header-translation table in the specified folder.
</p>

<hr>
<h2 id='GWASinspector'>Comprehensive and Easy to Use Quality Control of GWAS Results</h2><span id='topic+GWASinspector'></span><span id='topic+GWASinspector-package'></span>

<h3>Description</h3>

<p>When evaluating the results of a genome-wide association study (GWAS), it is important to perform a quality control to ensure that the results are valid, complete, correctly formatted, and, in case of meta-analysis, consistent with other studies in the same analysis.
This package was developed to facilitate and streamline this process and provide the user with a comprehensive report.
</p>


<h3>Details</h3>

<p>Check out our website for more help and support  <a href="http://GWASinspector.com">http://GWASinspector.com</a>.
</p>


<h3>Functions</h3>

<p><code><a href="#topic+setup_inspector">setup_inspector</a></code> This function Imports a QC-configuration file into R by generating a new instance of <a href="#topic+Inspector-class">Inspector</a> class.
</p>
<p><code><a href="#topic+run_inspector">run_inspector</a></code> This is the main function for running the algorithm on a set of GWAS result files.
</p>
<p><code><a href="#topic+result_inspector">result_inspector</a></code> This function displays a brief report about the results of running the Inspector algorithm on a set of GWAS result files.
</p>
<p><code><a href="#topic+demo_inspector">demo_inspector</a></code> This function runs the algorithm on a fabricated GWAS result file. User should only set the output folder for saving the generated files. The input file and reference dataset are embedded in the package.
</p>
<p><code><a href="#topic+system_check">system_check</a></code> Checks if required and optional packages are installed on the system. Although the optional packages do not contribute to the QC itself, having them available will allow for Excel and HTML formatted report files, which are easier to read and interpret.
</p>
<p><code><a href="#topic+get_config">get_config</a></code> Copies the template configuration file to the local machine.
</p>
<p><code><a href="#topic+get_headerTranslation">get_headerTranslation</a></code> Copies the template configuration file to the local machine.
</p>
<p><code><a href="#topic+compare_GWASs">compare_GWASs</a></code> Generates reports and plots for comparing the summary statistics of GWAS result files that are previously inspected with this package.
</p>
<p><code><a href="#topic+manhattan_plot">manhattan_plot</a></code> Generates the Manhattan plot from a GWAS result file. This function has many features that are described in the package tutorial.
</p>


<h3>Note</h3>

<p>GWASinspector uses the S4 object system of R to conduct the QC.
The QC is configured using an configuration (ini) file (check <code><a href="#topic+get_config">get_config</a></code>), which is imported into R through <code><a href="#topic+setup_inspector">setup_inspector</a></code> and
turns into an object of the <a href="#topic+Inspector-class">Inspector</a> class. To perform the QC, process the object with <code><a href="#topic+run_inspector">run_inspector</a></code>.
A quick scan of the results can be performed via <code><a href="#topic+result_inspector">result_inspector</a></code>, but the primary outcome of the QC are the
report files and graphs generated by <code><a href="#topic+run_inspector">run_inspector</a></code>.
</p>
<p>The main product of the QC is the extensive log file (in Excel/HTML format, depending on your settings)
</p>

<hr>
<h2 id='Inspector-class'>An S4 class to represent the methods and parameters for inspecting a list of GWAS study result files.</h2><span id='topic+Inspector-class'></span><span id='topic+Inspector'></span>

<h3>Description</h3>

<p>An object of this class is created by <code><a href="#topic+setup_inspector">setup_inspector</a></code> function. Each section of the
configuration file is represented as a list of attributes in this object.
</p>


<h3>Slots</h3>


<dl>
<dt><code>paths</code></dt><dd><p>A list of parameters which indicate <strong>Paths</strong> section from configuration file.</p>
</dd>
<dt><code>supplementaryFiles</code></dt><dd><p>A list of parameters which indicate <strong>supplementaryFiles</strong> section from configuration file.</p>
</dd>
<dt><code>input_parameters</code></dt><dd><p>A list of parameters which indicate <strong>input_parameters</strong> section from configuration file.</p>
</dd>
<dt><code>output_parameters</code></dt><dd><p>A list of parameters which indicate <strong>output_parameters</strong> section from configuration file.</p>
</dd>
<dt><code>remove_chromosomes</code></dt><dd><p>A list of parameters which indicate <strong>remove_chromosomes</strong> section from configuration file.</p>
</dd>
<dt><code>plot_specs</code></dt><dd><p>A list of parameters which indicate <strong>plot_specs</strong> section from configuration file.</p>
</dd>
<dt><code>filters</code></dt><dd><p>A list of parameters which indicate <strong>filters</strong> section from configuration file.</p>
</dd>
<dt><code>debug</code></dt><dd><p>A list of parameters which indicate <strong>debug</strong> section from configuration file.</p>
</dd>
<dt><code>input_files</code></dt><dd><p>A list of files that will be inspected during the run.</p>
</dd>
<dt><code>created_at</code></dt><dd><p>The time that object was created.</p>
</dd>
<dt><code>start_time</code></dt><dd><p>The time that object was run.</p>
</dd>
<dt><code>end_time</code></dt><dd><p>The time that run was finished.</p>
</dd>
<dt><code>StudyList</code></dt><dd><p>An object of <a href="#topic+StudyList-class">StudyList</a> class.</p>
</dd>
</dl>

<hr>
<h2 id='manhattan_plot'>Creates the Manhattan plot</h2><span id='topic+manhattan_plot'></span>

<h3>Description</h3>

<p>A function to generate Manhattan plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattan_plot(
  dataset,
  chr,
  pvalue,
  position,
  fileName,
  plot.title = "Manhattan Plot",
  plot.subtitle = "",
  p.threshold = 0.01,
  sig.threshold.log = -log10(5 * 10^-8),
  beta = NULL,
  std.error = NULL,
  check.columns = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="manhattan_plot_+3A_dataset">dataset</code></td>
<td>
<p>Data frame or data table containing the below columns</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_chr">chr</code></td>
<td>
<p>Name of chromosome column</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_pvalue">pvalue</code></td>
<td>
<p>Name of P-value column</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_position">position</code></td>
<td>
<p>Name of position column</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_filename">fileName</code></td>
<td>
<p>Full name and path of file to be saved (file extension should be 'png'). e.g. “c:/users/researcher/study/man_plot.png”</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_plot.title">plot.title</code></td>
<td>
<p>Title of the plot, default value is 'Manhattan plot'</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_plot.subtitle">plot.subtitle</code></td>
<td>
<p>Subtitle of the plot</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_p.threshold">p.threshold</code></td>
<td>
<p>Threshold for plotting variants (i.e. p-values &gt; 0.01 will not be plotted). Setting a higher threshold will significantly increase plotting time</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_sig.threshold.log">sig.threshold.log</code></td>
<td>
<p>The -log10 transformed significance threshold, used for plotting a threshold line (e.g. 8 = 10^-8)</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_beta">beta</code></td>
<td>
<p>(optional) Name of the effect-size column</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_std.error">std.error</code></td>
<td>
<p>(optional) Name of the standard error column</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_check.columns">check.columns</code></td>
<td>
<p>Whether to check input columns for invalid values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Generates and saves a Manhattan plot for the provided data.
</p>

<hr>
<h2 id='result_inspector'>Displays a brief report after running the QC pipeline</h2><span id='topic+result_inspector'></span>

<h3>Description</h3>

<p>This function displays a brief report about the results of running the Inspector algorithm on a set of GWAS result files.
The full report including plots, cleaned files and summary statistics are generated and saved in the output folder during the algorithm run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>result_inspector(inspector)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="result_inspector_+3A_inspector">inspector</code></td>
<td>
<p>An instance of <a href="#topic+Inspector-class">Inspector</a> class. Check <code><a href="#topic+setup_inspector">setup_inspector</a></code> for more details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table containing a brief report about the results.
</p>

<hr>
<h2 id='run_inspector'>Runs the QC pipeline on a set of GWAS result files</h2><span id='topic+run_inspector'></span>

<h3>Description</h3>

<p>This is the main function of the package for running the QC algorithm on a set of GWAS result files.
It requires an object of class <a href="#topic+Inspector-class">Inspector</a> which should be created by <code><a href="#topic+setup_inspector">setup_inspector</a></code>.
Check the package vignette and tutorial for more details on this topic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_inspector(inspector, verbose = TRUE, test.run = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_inspector_+3A_inspector">inspector</code></td>
<td>
<p>An instance of <a href="#topic+Inspector-class">Inspector</a> class. Check <code><a href="#topic+setup_inspector">setup_inspector</a></code> for more details.</p>
</td></tr>
<tr><td><code id="run_inspector_+3A_verbose">verbose</code></td>
<td>
<p>logical. If FALSE, no messages will show up in the terminal and are only saved in the log file.</p>
</td></tr>
<tr><td><code id="run_inspector_+3A_test.run">test.run</code></td>
<td>
<p>logical. If TRUE, only the first 1000 lines of each data file are loaded and analyzed;
plots and saving the cleaned output dataset are skipped. Default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Reports from running the algorithm on a single or a series of GWAS result files are generated and saved.
</p>

<hr>
<h2 id='setup_inspector'>Importing a configuration file into R for setting the pipeline parameters</h2><span id='topic+setup_inspector'></span>

<h3>Description</h3>

<p>To run a QC in GWASinspector, copy a template configuration file to your machine using the <code><a href="#topic+get_config">get_config</a></code> command at first, and edit it to suit your requirements.
Next, use the <code><a href="#topic+setup_inspector">setup_inspector</a></code> function to check the configuration file and import it into R.
This will create an object of the inspector class, which can then be processed using <code><a href="#topic+run_inspector">run_inspector</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setup_inspector(config.file, validate = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setup_inspector_+3A_config.file">config.file</code></td>
<td>
<p>character. Path to a configuration (.ini) file. For a sample configuration file, see <code><a href="#topic+get_config">get_config</a></code>.</p>
</td></tr>
<tr><td><code id="setup_inspector_+3A_validate">validate</code></td>
<td>
<p>logical. Whether to validate the object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a new instance of <a href="#topic+Inspector-class">Inspector</a> class.
</p>

<hr>
<h2 id='Study-class'>An S4 class to represent an inspected GWAS study result file.</h2><span id='topic+Study-class'></span><span id='topic+Study'></span>

<h3>Description</h3>

<p>This class is embedded in the <a href="#topic+StudyList-class">StudyList</a> class and should not be initiated separately.
</p>


<h3>Slots</h3>


<dl>
<dt><code>File</code></dt><dd><p>A list representing GWAS result file specifications</p>
</dd>
<dt><code>Counts</code></dt><dd><p>A list representing different variant counts from the GWAS result file.</p>
</dd>
<dt><code>Correlations</code></dt><dd><p>A list representing different allele frequency and P-value correlations in the GWAS result file.</p>
</dd>
<dt><code>Statistics</code></dt><dd><p>A list representing summary statistics from the GWAS result file.</p>
</dd>
<dt><code>Successful_run</code></dt><dd><p>A logical value indicating whether the run was successful or not.</p>
</dd>
<dt><code>starttime</code></dt><dd><p>The time that file inspection started.</p>
</dd>
<dt><code>endtime</code></dt><dd><p>The time that file inspection ended.</p>
</dd>
</dl>

<hr>
<h2 id='StudyList-class'>An S4 class to represent a list of inspected GWAS study result files.</h2><span id='topic+StudyList-class'></span><span id='topic+StudyList'></span>

<h3>Description</h3>

<p>This class is embedded in the <a href="#topic+Inspector-class">Inspector</a> class and should not be initiated separately.
</p>


<h3>Slots</h3>


<dl>
<dt><code>studyList</code></dt><dd><p>A list of GWAS study result files. Each member of this list is of class <code><a href="#topic+Study-class">Study</a></code>.</p>
</dd>
<dt><code>studyCount</code></dt><dd><p>A numeric value indicating how many items of class <code><a href="#topic+Study-class">Study</a></code> are included.</p>
</dd>
</dl>

<hr>
<h2 id='system_check'>Checks which required and optional packages are available</h2><span id='topic+system_check'></span>

<h3>Description</h3>

<p>Checks if required and optional packages are installed on the system.
Although the optional packages do not contribute to the QC itself, having them available
will allow for Excel and HTML formatted log files, which are easier to read and interpret.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>system_check()
</code></pre>


<h3>Value</h3>

<p>System information and required functionalities for the QC algorithm are checked and reported as a data frame.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
