<!DOCTYPE html><html lang="en"><head><title>Help for package GWASExactHW</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GWASExactHW}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GWASExactHW-package'><p>Exact Hardy-Weinburg testing for Genome Wide Association Studies</p></a></li>
<li><a href='#HWExact'><p>Function to calculate Hardy-Weinburg exact p-values</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Exact Hardy-Weinburg Testing for Genome Wide Association Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Ian Painter, University of Washington</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Stephanie Gogarten &lt;sdmorris@uw.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>
  Exact Hardy-Weinburg testing (using Fisher's test) for SNP genotypes
  as typically obtained in a Genome Wide Association Study (GWAS).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-12 03:48:51 UTC; stephanie</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-12 07:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='GWASExactHW-package'>Exact Hardy-Weinburg testing for Genome Wide Association Studies</h2><span id='topic+GWASExactHW-package'></span><span id='topic+GWASExactHW'></span>

<h3>Description</h3>

<p>This package contains a function to do exact Hardy-Weinburg testing (using Fisher's test) over all or a selection of SNP genotypes as typically obtained in a Genome Wide Association Study (GWAS).</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> GWASExactHW</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.01</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2013-01-04</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GNU </td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The function HWExact runs fast Hardy-Weinburg testing for a set of bi-alleleic genotypes.
</p>


<h3>Author(s)</h3>

<p>Ian Painter, University of Washington
</p>
<p>Maintainer: Ian Painter &lt;ipainter@u.washington.edu&gt;
</p>


<h3>References</h3>

<p> Wigginton, JE, Cutler, DJ, and Abecasis, GR (2005) A Note on Exact Tests of Hardy-Weinberg Equilibrium. American Journal of Human Genetics. 76
</p>

<hr>
<h2 id='HWExact'>Function to calculate Hardy-Weinburg exact p-values
</h2><span id='topic+HWExact'></span>

<h3>Description</h3>

<p>This function calculates Hardy-Weinburg (Fisher's) exact p-values for GWAS SNP data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWExact(GenotypeCounts)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWExact_+3A_genotypecounts">GenotypeCounts</code></td>
<td>
<p>A dataframe of genotype counts, with columns called <code>nAA</code>, <code>nAa</code> and <code>naa</code>, one row for each SNP.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of exact p-values.</p>


<h3>Note</h3>

<p> This function uses a C function <code>SNPHWE.c</code> written by Jan Wigginton as described in the above reference.</p>


<h3>Author(s)</h3>

<p>Ian Painter</p>


<h3>References</h3>

<p> Wigginton, JE, Cutler, DJ, and Abecasis, GR (2005) A Note on Exact Tests of Hardy-Weinberg Equilibrium. American Journal of Human Genetics. 76</p>


<h3>Examples</h3>

<pre><code class='language-R'>
pA&lt;- runif(1)
pAA&lt;- pA^2
pAa&lt;- 2*pA*(1-pA)
paa&lt;- (1-pA)^2

counts&lt;- rmultinom(1000, 3000, c(pAA, pAa, paa) )
genotypes&lt;- data.frame(nAA = counts[1,], nAa = counts[2,], naa = counts[3,])
hwPvalues&lt;- HWExact(genotypes)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
