<!DOCTYPE html><html><head><title>Help for package GWASbyCluster</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GWASbyCluster}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#esSim'>
<p>An ExpressionSet Object Storing Simulated Genotype Data</p></a></li>
<li><a href='#esSimDiffPriors'>
<p>An ExpressionSet Object Storing Simulated Genotype Data</p></a></li>
<li><a href='#estMemSNPs'>
<p>Estimate SNP cluster membership</p></a></li>
<li><a href='#estMemSNPs.oneSetHyperPara'>
<p>Estimate SNP cluster membership</p></a></li>
<li><a href='#simGenoFunc'>
<p>Simulate Genotype Data from a Mixture of 3 Bayesian Hierarchical Models</p></a></li>
<li><a href='#simGenoFuncDiffPriors'>
<p>Simulate Genotype Data from a Mixture of 3 Bayesian Hierarchical Models</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identifying Significant SNPs in Genome Wide Association Studies
(GWAS) via Clustering</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-10-09</td>
</tr>
<tr>
<td>Author:</td>
<td>Yan Xu, Li Xing, Jessica Su, Xuekui Zhang&lt;UBC.X.Zhang@gmail.com&gt;, Weiliang Qiu &lt;Weiliang.Qiu@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Li Xing &lt;sfulxing@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), Biobase</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, snpStats, methods, rootSolve, limma</td>
</tr>
<tr>
<td>Description:</td>
<td>Identifying disease-associated significant SNPs using clustering approach. This package is implementation of method proposed in Xu et al (2019) &lt;<a href="https://doi.org/10.1038%2Fs41598-019-50229-6">doi:10.1038/s41598-019-50229-6</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-10-10 01:10:31 UTC; weiliangqiu</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-10-11 09:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='esSim'>
An ExpressionSet Object Storing Simulated Genotype Data
</h2><span id='topic+esSim'></span>

<h3>Description</h3>

<p>An ExpressionSet object storing simulated genotype data.
The minor allele frequency (MAF) of cases has the same prior
as that of controls. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("esSim")</code></pre>


<h3>Details</h3>

<p>In this simulation, we generate additive-coded genotypes for 3 clusters of SNPs
based on a mixture of 3 Bayesian hierarchical models. 
</p>
<p>In cluster <code class="reqn">+</code>, the minor allele frequency
(MAF) <code class="reqn">\theta_{x+}</code> of cases is greater than the MAF <code class="reqn">\theta_{y+}</code> of
controls.
</p>
<p>In cluster <code class="reqn">0</code>, the MAF <code class="reqn">\theta_{0}</code> of cases is equal to 
the MAF of controls.
</p>
<p>In cluster <code class="reqn">-</code>, the MAF <code class="reqn">\theta_{x-}</code> of cases is smaller than 
the MAF <code class="reqn">\theta_{y-}</code> of controls.
</p>
<p>The proportions of the 3 clusters of SNPs are <code class="reqn">\pi_{+}</code>,
<code class="reqn">\pi_{0}</code>, and <code class="reqn">\pi_{-}</code>, respectively.
</p>
<p>We assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">+</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{+}\left(\theta_{x+}\right)h_{+}\left(\theta_{y+}\right)
I\left(\theta_{x+}&gt;\theta_{y+}\right),
</code>
</p>

<p>where <code class="reqn">I(a)</code> is hte indicator function taking value <code class="reqn">1</code> if the event 
<code class="reqn">a</code> is true, and value <code class="reqn">0</code> otherwise.
The function <code class="reqn">h_{+}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{+}, \beta_{+}\right)</code>.
</p>
<p>We assume <code class="reqn">\theta_{0}</code> has the beta prior <code class="reqn">Beta(\alpha_0, \beta_0)</code>. 
</p>
<p>We also assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">-</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{-}\left(\theta_{x-}\right)h_{-}\left(\theta_{y-}\right)
I\left(\theta_{x-}&gt;\theta_{y-}\right).
</code>
</p>

<p>The function <code class="reqn">h_{-}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{-}, \beta_{-}\right)</code>.
</p>
<p>Given a SNP, we assume Hardy-Weinberg equilibrium holds for its genotypes. 
That is, given MAF <code class="reqn">\theta</code>, the probabilities of genotypes are
</p>
<p style="text-align: center;"><code class="reqn">Pr(geno=2) = \theta^2</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=1) = 2\theta\left(1-\theta\right)</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=0) = \left(1-\theta\right)^2</code>
</p>

<p>We also assume the genotypes <code class="reqn">0</code> (wild-type), <code class="reqn">1</code> (heterozygote), and 
<code class="reqn">2</code> (mutation) follows a multinomial distribution
<code class="reqn">Multinomial\left\{1, \left[
\theta^2, 2\theta\left(1-\theta\right),
\left(1-\theta\right)^2
\right]\right\}</code>
</p>
<p>We set the number of cases as <code class="reqn">100</code>, the number of controls as 
<code class="reqn">100</code>,
and the number of SNPs as <code class="reqn">1000</code>.
</p>
<p>The hyperparameters are
<code class="reqn">\alpha_{+}=2</code>, <code class="reqn">\beta_{+}=5</code>, <code class="reqn">\pi_{+}=0.1</code>,
<code class="reqn">\alpha_{0}=2</code>, <code class="reqn">\beta_{0}=5</code>, <code class="reqn">\pi_{0}=0.8</code>,
<code class="reqn">\alpha_{-}=2</code>, <code class="reqn">\beta_{-}=5</code>, <code class="reqn">\pi_{-}=0.1</code>.
</p>
<p>Note that when we generate MAFs from the half-flat shape bivariate priors, 
we might get very small MAFs or get MAFs <code class="reqn">&gt;0.5</code>. In these cased,
we then delete this SNP.
</p>
<p>So the final number of SNPs generated might be less than the initially-set 
number <code class="reqn">1000</code> of SNPs.
</p>
<p>For the dataset stored in <code>esSim</code>, there are <code class="reqn">872</code> SNPs.
<code class="reqn">83</code> SNPs are in cluster -, <code class="reqn">714</code> SNPs are in cluster <code class="reqn">0</code>,
and <code class="reqn">75</code> SNPs are in cluster <code class="reqn">+</code>.
</p>


<h3>References</h3>

<p>Yan X, Xing L, Su J, Zhang X, Qiu W. 
Model-based clustering for identifying disease-associated SNPs in case-control genome-wide association studies. 
Scientific Reports 9, Article number: 13686 (2019) https://www.nature.com/articles/s41598-019-50229-6.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(esSim)
print(esSim)

pDat=pData(esSim)
print(pDat[1:2,])
print(table(pDat$memSubjs))

fDat=fData(esSim)
print(fDat[1:2,])
print(table(fDat$memGenes))
print(table(fDat$memGenes2))

</code></pre>

<hr>
<h2 id='esSimDiffPriors'>
An ExpressionSet Object Storing Simulated Genotype Data
</h2><span id='topic+esSimDiffPriors'></span>

<h3>Description</h3>

<p>An ExpressionSet object storing simulated genotype data.
The minor allele frequency (MAF) of cases has different prior
than that of controls. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("esSimDiffPriors")</code></pre>


<h3>Details</h3>

<p>In this simulation, we generate additive-coded genotypes for 3 clusters of SNPs
based on a mixture of 3 Bayesian hierarchical models. 
</p>
<p>In cluster <code class="reqn">+</code>, the minor allele frequency
(MAF) <code class="reqn">\theta_{x+}</code> of cases is greater than the MAF <code class="reqn">\theta_{y+}</code> of
controls.
</p>
<p>In cluster <code class="reqn">0</code>, the MAF <code class="reqn">\theta_{0}</code> of cases is equal to 
the MAF of controls.
</p>
<p>In cluster <code class="reqn">-</code>, the MAF <code class="reqn">\theta_{x-}</code> of cases is smaller than 
the MAF <code class="reqn">\theta_{y-}</code> of controls.
</p>
<p>The proportions of the 3 clusters of SNPs are <code class="reqn">\pi_{+}</code>,
<code class="reqn">\pi_{0}</code>, and <code class="reqn">\pi_{-}</code>, respectively.
</p>
<p>We assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">+</code> 
</p>
<p style="text-align: center;"><code class="reqn">2h_{x+}\left(\theta_{x+}\right)h_{y+}\left(\theta_{y+}\right)
I\left(\theta_{x+}&gt;\theta_{y+}\right),
</code>
</p>

<p>where <code class="reqn">I(a)</code> is hte indicator function taking value <code class="reqn">1</code> if the event 
<code class="reqn">a</code> is true, and value <code class="reqn">0</code> otherwise.
The function <code class="reqn">h_{x+}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{x+}, \beta_{x+}\right)</code>.
The function <code class="reqn">h_{y+}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{y+}, \beta_{y+}\right)</code>.
</p>
<p>We assume <code class="reqn">\theta_{0}</code> has the beta prior <code class="reqn">Beta(\alpha_0, \beta_0)</code>. 
</p>
<p>We also assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">-</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{x-}\left(\theta_{x-}\right)h_{y-}\left(\theta_{y-}\right)
I\left(\theta_{x-}&gt;\theta_{y-}\right).
</code>
</p>

<p>The function <code class="reqn">h_{x-}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{x-}, \beta_{x-}\right)</code>.
The function <code class="reqn">h_{y-}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{y-}, \beta_{y-}\right)</code>.
</p>
<p>Given a SNP, we assume Hardy-Weinberg equilibrium holds for its genotypes. 
That is, given MAF <code class="reqn">\theta</code>, the probabilities of genotypes are
</p>
<p style="text-align: center;"><code class="reqn">Pr(geno=2) = \theta^2</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=1) = 2\theta\left(1-\theta\right)</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=0) = \left(1-\theta\right)^2</code>
</p>

<p>We also assume the genotypes <code class="reqn">0</code> (wild-type), <code class="reqn">1</code> (heterozygote), and 
<code class="reqn">2</code> (mutation) follows a multinomial distribution
<code class="reqn">Multinomial\left\{1, \left[
\theta^2, 2\theta\left(1-\theta\right),
\left(1-\theta\right)^2
\right]\right\}</code>
</p>
<p>We set the number of cases as <code class="reqn">100</code>, the number of controls as 
<code class="reqn">100</code>,
and the number of SNPs as <code class="reqn">1000</code>.
</p>
<p>The hyperparameters are
<code class="reqn">\alpha_{x+}=2</code>, <code class="reqn">\beta_{x+}=3</code>, 
<code class="reqn">\alpha_{y+}=2</code>, <code class="reqn">\beta_{y+}=8</code>, 
<code class="reqn">\pi_{+}=0.1</code>,
</p>
<p><code class="reqn">\alpha_{0}=2</code>, <code class="reqn">\beta_{0}=5</code>, <code class="reqn">\pi_{0}=0.8</code>,
</p>
<p><code class="reqn">\alpha_{x-}=2</code>, <code class="reqn">\beta_{x-}=8</code>, 
<code class="reqn">\alpha_{y-}=2</code>, <code class="reqn">\beta_{y-}=3</code>, 
<code class="reqn">\pi_{-}=0.1</code>.
</p>
<p>Note that when we generate MAFs from the half-flat shape bivariate priors, 
we might get very small MAFs or get MAFs <code class="reqn">&gt;0.5</code>. In these cased,
we then delete this SNP.
</p>
<p>So the final number of SNPs generated might be less than the initially-set 
number <code class="reqn">1000</code> of SNPs.
</p>
<p>For the dataset stored in <code>esSim</code>, there are <code class="reqn">838</code> SNPs.
<code class="reqn">64</code> SNPs are in cluster -, <code class="reqn">708</code> SNPs are in cluster <code class="reqn">0</code>,
and <code class="reqn">66</code> SNPs are in cluster <code class="reqn">+</code>.
</p>


<h3>References</h3>

<p>Yan X, Xing L, Su J, Zhang X, Qiu W. 
Model-based clustering for identifying disease-associated SNPs in case-control genome-wide association studies. 
Scientific Reports 9, Article number: 13686 (2019) https://www.nature.com/articles/s41598-019-50229-6.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(esSimDiffPriors)
print(esSimDiffPriors)

pDat=pData(esSimDiffPriors)
print(pDat[1:2,])
print(table(pDat$memSubjs))

fDat=fData(esSimDiffPriors)
print(fDat[1:2,])
print(table(fDat$memGenes))
print(table(fDat$memGenes2))
</code></pre>

<hr>
<h2 id='estMemSNPs'>
Estimate SNP cluster membership
</h2><span id='topic+estMemSNPs'></span>

<h3>Description</h3>

<p>Estimate SNP cluster membership. Only update cluster mixture proportions.
Assume the 3 clusters have different sets of hyperparameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estMemSNPs(es, 
           var.memSubjs = "memSubjs", 
           eps = 0.001, 
           MaxIter = 50, 
           bVec = rep(3, 3), 
           pvalAdjMethod = "fdr", 
           method = "FDR",
           fdr = 0.05,
           verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estMemSNPs_+3A_es">es</code></td>
<td>

<p>An ExpressionSet object storing SNP genotype data.
It contains 3 matrices. The first matrix, which can be extracted by <code>exprs</code> method (e.g., <code>exprs(es)</code>), stores genotype data, with rows are SNPs and columns are subjects.
</p>
<p>The second matrix, which can be extracted by <code>pData</code> method (e.g., <code>pData(es)</code>), stores phenotype data describing subjects. Rows are subjects, and columns are phenotype variables.
</p>
<p>The third matrix, which can be extracted by <code>fData</code> method (e.g., <code>fData(es)</code>), stores feature data describing SNPs. Rows are SNPs and columns are feature variables.
</p>
</td></tr>
<tr><td><code id="estMemSNPs_+3A_var.memsubjs">var.memSubjs</code></td>
<td>

<p>character. The name of the phenotype variable indicating subject's case-control status. It must take only two values: 1 indicating case and 0 indicating control.
</p>
</td></tr>
<tr><td><code id="estMemSNPs_+3A_eps">eps</code></td>
<td>

<p>numeric. A small positive number as threshold for convergence of EM algorithm.
</p>
</td></tr>
<tr><td><code id="estMemSNPs_+3A_maxiter">MaxIter</code></td>
<td>

<p>integer. A positive integer indicating maximum iteration in EM algorithm.
</p>
</td></tr>
<tr><td><code id="estMemSNPs_+3A_bvec">bVec</code></td>
<td>

<p>numeric. A vector of 2 elements. Indicates the parameters of the symmetric Dirichlet prior for proportion mixtures.
</p>
</td></tr>
<tr><td><code id="estMemSNPs_+3A_pvaladjmethod">pvalAdjMethod</code></td>
<td>

<p>character. Indicating p-value adjustment method. c.f. <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>.
</p>
</td></tr>
<tr><td><code id="estMemSNPs_+3A_method">method</code></td>
<td>
<p>method to obtain SNP cluster membership based on the responsibility matrix. The default value is &ldquo;FDR&rdquo;. The other possible 
value is &ldquo;max&rdquo;. see details.</p>
</td></tr>
<tr><td><code id="estMemSNPs_+3A_fdr">fdr</code></td>
<td>
<p>numeric. A small positive FDR threshold used to call SNP cluster membership</p>
</td></tr>
<tr><td><code id="estMemSNPs_+3A_verbose">verbose</code></td>
<td>

<p>logical. Indicating if intermediate and final results should be output.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this simulation, we generate additive-coded genotypes for 3 clusters of SNPs
based on a mixture of 3 Bayesian hierarchical models. 
</p>
<p>In cluster <code class="reqn">+</code>, the minor allele frequency
(MAF) <code class="reqn">\theta_{x+}</code> of cases is greater than the MAF <code class="reqn">\theta_{y+}</code> of
controls.
</p>
<p>In cluster <code class="reqn">0</code>, the MAF <code class="reqn">\theta_{0}</code> of cases is equal to 
the MAF of controls.
</p>
<p>In cluster <code class="reqn">-</code>, the MAF <code class="reqn">\theta_{x-}</code> of cases is smaller than 
the MAF <code class="reqn">\theta_{y-}</code> of controls.
</p>
<p>The proportions of the 3 clusters of SNPs are <code class="reqn">\pi_{+}</code>,
<code class="reqn">\pi_{0}</code>, and <code class="reqn">\pi_{-}</code>, respectively.
</p>
<p>We assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">+</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{+}\left(\theta_{x+}\right)h_{+}\left(\theta_{y+}\right)
I\left(\theta_{x+}&gt;\theta_{y+}\right),
</code>
</p>

<p>where <code class="reqn">I(a)</code> is hte indicator function taking value <code class="reqn">1</code> if the event 
<code class="reqn">a</code> is true, and value <code class="reqn">0</code> otherwise.
The function <code class="reqn">h_{+}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{+}, \beta_{+}\right)</code>.
</p>
<p>We assume <code class="reqn">\theta_{0}</code> has the beta prior <code class="reqn">Beta(\alpha_0, \beta_0)</code>. 
</p>
<p>We also assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">-</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{-}\left(\theta_{x-}\right)h_{-}\left(\theta_{y-}\right)
I\left(\theta_{x-}&gt;\theta_{y-}\right).
</code>
</p>

<p>The function <code class="reqn">h_{-}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{-}, \beta_{-}\right)</code>.
</p>
<p>Given a SNP, we assume Hardy-Weinberg equilibrium holds for its genotypes. 
That is, given MAF <code class="reqn">\theta</code>, the probabilities of genotypes are
</p>
<p style="text-align: center;"><code class="reqn">Pr(geno=2) = \theta^2</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=1) = 2\theta\left(1-\theta\right)</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=0) = \left(1-\theta\right)^2</code>
</p>

<p>We also assume the genotypes <code class="reqn">0</code> (wild-type), <code class="reqn">1</code> (heterozygote), and 
<code class="reqn">2</code> (mutation) follows a multinomial distribution
<code class="reqn">Multinomial\left\{1, \left[
\theta^2, 2\theta\left(1-\theta\right),
\left(1-\theta\right)^2
\right]\right\}</code>
</p>
<p>For each SNP, we calculat its posterior probabilities that it belongs to cluster <code class="reqn">k</code>. This forms a matrix with 3 columns. Rows are SNPs. 
The 1st column is the posterior probability that the SNP belongs to cluster <code class="reqn">-</code>. 
The 2nd column is the posterior probability that the SNP belongs to cluster <code class="reqn">0</code>.
The 3rd column is the posterior probability that the SNP belongs to cluster <code class="reqn">+</code>.
We call this posterior probability matrix as responsibility matrix.
To determine which cluster a SNP eventually belongs to, we can use 2 methods.
The first method (the default method) is &ldquo;FDR&rdquo; method, which will
use FDR criterion to determine SNP cluster membership.
The 2nd method is use the maximum posterior probability to decide which
cluster a SNP belongs to.
</p>


<h3>Value</h3>

<p>A list of 12 elements
</p>
<table>
<tr><td><code>wMat</code></td>
<td>
<p>matrix of posterior probabilities. The rows are SNPs. There are 3 columns. 
The first column is the posterior probability that a SNP belongs to cluster - given genotypes of subjects. 
The second column is the posterior probability that a SNP belongs to cluster 0 given genotypes of subjects. 
The third column is the posterior probability that a SNP belongs to cluster + given genotypes of subjects. 
</p>
</td></tr>
<tr><td><code>memSNPs</code></td>
<td>
<p>a vector of SNP cluster membership for the 3-cluster partitionfrom the mixture of 3 Bayesian hierarchical models.</p>
</td></tr>
<tr><td><code>memSNPs2</code></td>
<td>
<p>a vector of binary SNP cluster membership. 1 indicates the SNP has different MAFs between cases and controls. 0 indicates the SNP has the same MAF in cases as that in controls.</p>
</td></tr>
<tr><td><code>piVec</code></td>
<td>
<p>a vector of cluster mixture proportions.</p>
</td></tr>
<tr><td><code>alpha.p</code></td>
<td>
<p>the first shape parameter of the beta prior for MAF obtaind from initial 3-cluster partitions based on GWAS for cluster +.</p>
</td></tr>
<tr><td><code>beta.p</code></td>
<td>
<p>the second shape parameter of the beta prior for MAF obtaind from initial 3-cluster partitions based on GWAS for cluster +.</p>
</td></tr>
<tr><td><code>alpha0</code></td>
<td>
<p>the first shape parameter of the beta prior for MAF obtaind from initial 3-cluster partitions based on GWAS for cluster 0.</p>
</td></tr>
<tr><td><code>beta0</code></td>
<td>
<p>the second shape parameter of the beta prior for MAF obtaind from initial 3-cluster partitions based on GWAS for cluster 0.</p>
</td></tr>
<tr><td><code>alpha.n</code></td>
<td>
<p>the first shape parameter of the beta prior for MAF obtaind from initial 3-cluster partitions based on GWAS for cluster -.</p>
</td></tr>
<tr><td><code>beta.n</code></td>
<td>
<p>the second shape parameter of the beta prior for MAF obtaind from initial 3-cluster partitions based on GWAS for cluster -.</p>
</td></tr>
<tr><td><code>loop</code></td>
<td>
<p>number of iteration in EM algorithm</p>
</td></tr>
<tr><td><code>diff</code></td>
<td>
<p>sum of the squared difference of cluster mixture proportions between current iteration and previous iteration in EM algorithm. if <code>eps &lt; eps</code>, we claim the EM algorithm converges.</p>
</td></tr>
<tr><td><code>res.limma</code></td>
<td>
<p>object returned by limma</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yan Xu &lt;yanxu@uvic.ca&gt;, 
Li Xing &lt;sfulxing@gmail.com&gt;, 
Jessica Su &lt;rejas@channing.harvard.edu&gt;, 
Xuekui Zhang &lt;xuekui@uvic.ca&gt;, 
Weiliang Qiu &lt;Weiliang.Qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Yan X, Xing L, Su J, Zhang X, Qiu W. 
Model-based clustering for identifying disease-associated SNPs in case-control genome-wide association studies. 
Scientific Reports 9, Article number: 13686 (2019) https://www.nature.com/articles/s41598-019-50229-6.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(esSimDiffPriors)
print(esSimDiffPriors)

es=esSimDiffPriors[1:500,]
fDat = fData(es)
print(fDat[1:2,])
print(table(fDat$memGenes))

res = estMemSNPs(
  es = es, 
  var.memSubjs = "memSubjs")

print(table(fDat$memGenes, res$memSNPs))

</code></pre>

<hr>
<h2 id='estMemSNPs.oneSetHyperPara'>
Estimate SNP cluster membership
</h2><span id='topic+estMemSNPs.oneSetHyperPara'></span>

<h3>Description</h3>

<p>Estimate SNP cluster membership. Only update cluster mixture proportions.
Assume all 3 clusters have the same set of hyperparameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estMemSNPs.oneSetHyperPara(es, 
           var.memSubjs = "memSubjs", 
           eps = 1.0e-3,
           MaxIter = 50, 
           bVec = rep(3, 3), 
           pvalAdjMethod = "none", 
           method = "FDR",
           fdr = 0.05,
           verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estMemSNPs.oneSetHyperPara_+3A_es">es</code></td>
<td>

<p>An ExpressionSet object storing SNP genotype data.
It contains 3 matrices. The first matrix, which can be extracted by <code>exprs</code> method (e.g., <code>exprs(es)</code>), stores genotype data, with rows are SNPs and columns are subjects.
</p>
<p>The second matrix, which can be extracted by <code>pData</code> method (e.g., <code>pData(es)</code>), stores phenotype data describing subjects. Rows are subjects, and columns are phenotype variables.
</p>
<p>The third matrix, which can be extracted by <code>fData</code> method (e.g., <code>fData(es)</code>), stores feature data describing SNPs. Rows are SNPs and columns are feature variables.
</p>
</td></tr>
<tr><td><code id="estMemSNPs.oneSetHyperPara_+3A_var.memsubjs">var.memSubjs</code></td>
<td>

<p>character. The name of the phenotype variable indicating subject's case-control status. It must take only two values: 1 indicating case and 0 indicating control.
</p>
</td></tr>
<tr><td><code id="estMemSNPs.oneSetHyperPara_+3A_eps">eps</code></td>
<td>

<p>numeric. A small positive number as threshold for convergence of EM algorithm.
</p>
</td></tr>
<tr><td><code id="estMemSNPs.oneSetHyperPara_+3A_maxiter">MaxIter</code></td>
<td>

<p>integer. A positive integer indicating maximum iteration in EM algorithm.
</p>
</td></tr>
<tr><td><code id="estMemSNPs.oneSetHyperPara_+3A_bvec">bVec</code></td>
<td>

<p>numeric. A vector of 2 elements. Indicates the parameters of the symmetric Dirichlet prior for proportion mixtures.
</p>
</td></tr>
<tr><td><code id="estMemSNPs.oneSetHyperPara_+3A_pvaladjmethod">pvalAdjMethod</code></td>
<td>

<p>character. Indicating p-value adjustment method. c.f. <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>.
</p>
</td></tr>
<tr><td><code id="estMemSNPs.oneSetHyperPara_+3A_method">method</code></td>
<td>
<p>method to obtain SNP cluster membership based on the responsibility matrix. The default value is &ldquo;FDR&rdquo;. The other possible 
value is &ldquo;max&rdquo;. see details.</p>
</td></tr>
<tr><td><code id="estMemSNPs.oneSetHyperPara_+3A_fdr">fdr</code></td>
<td>
<p>numeric. A small positive FDR threshold used to call SNP cluster membership</p>
</td></tr>
<tr><td><code id="estMemSNPs.oneSetHyperPara_+3A_verbose">verbose</code></td>
<td>

<p>logical. Indicating if intermediate and final results should be output.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We characterize the distribution of genotypes of SNPs by a mixture of 3 Bayesian hierarchical models.  The 3 Bayeisan hierarchical models correspond to 3
clusters of SNPs.
</p>
<p>In cluster <code class="reqn">+</code>, the minor allele frequency
(MAF) <code class="reqn">\theta_{x+}</code> of cases is greater than the MAF <code class="reqn">\theta_{y+}</code> of
controls.
</p>
<p>In cluster <code class="reqn">0</code>, the MAF <code class="reqn">\theta_{0}</code> of cases is equal to 
the MAF of controls.
</p>
<p>In cluster <code class="reqn">-</code>, the MAF <code class="reqn">\theta_{x-}</code> of cases is smaller than 
the MAF <code class="reqn">\theta_{y-}</code> of controls.
</p>
<p>The proportions of the 3 clusters of SNPs are <code class="reqn">\pi_{+}</code>,
<code class="reqn">\pi_{0}</code>, and <code class="reqn">\pi_{-}</code>, respectively.
</p>
<p>We assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">+</code>
</p>
<p style="text-align: center;"><code class="reqn">2h\left(\theta_{x+}\right)h\left(\theta_{y+}\right)
I\left(\theta_{x+}&gt;\theta_{y+}\right),
</code>
</p>

<p>where <code class="reqn">I(a)</code> is hte indicator function taking value <code class="reqn">1</code> if the event 
<code class="reqn">a</code> is true, and value <code class="reqn">0</code> otherwise.
The function <code class="reqn">h</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha, \beta\right)</code>.
</p>
<p>We assume <code class="reqn">\theta_{0}</code> has the beta prior <code class="reqn">Beta(\alpha, \beta)</code>. 
</p>
<p>We also assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">-</code>
</p>
<p style="text-align: center;"><code class="reqn">2h\left(\theta_{x-}\right)h\left(\theta_{y-}\right)
I\left(\theta_{x-}&gt;\theta_{y-}\right).
</code>
</p>

<p>Given a SNP, we assume Hardy-Weinberg equilibrium holds for its genotypes. 
That is, given MAF <code class="reqn">\theta</code>, the probabilities of genotypes are
</p>
<p style="text-align: center;"><code class="reqn">Pr(geno=2) = \theta^2</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=1) = 2\theta\left(1-\theta\right)</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=0) = \left(1-\theta\right)^2</code>
</p>

<p>We also assume the genotypes <code class="reqn">0</code> (wild-type), <code class="reqn">1</code> (heterozygote), and 
<code class="reqn">2</code> (mutation) follows a multinomial distribution
<code class="reqn">Multinomial\left\{1, \left[
\theta^2, 2\theta\left(1-\theta\right),
\left(1-\theta\right)^2
\right]\right\}</code>
</p>
<p>For each SNP, we calculat its posterior probabilities that it belongs to cluster <code class="reqn">k</code>. This forms a matrix with 3 columns. Rows are SNPs. 
The 1st column is the posterior probability that the SNP belongs to cluster <code class="reqn">-</code>. 
The 2nd column is the posterior probability that the SNP belongs to cluster <code class="reqn">0</code>.
The 3rd column is the posterior probability that the SNP belongs to cluster <code class="reqn">+</code>.
We call this posterior probability matrix as responsibility matrix.
To determine which cluster a SNP eventually belongs to, we can use 2 methods.
The first method (the default method) is &ldquo;FDR&rdquo; method, which will
use FDR criterion to determine SNP cluster membership.
The 2nd method is use the maximum posterior probability to decide which
cluster a SNP belongs to.
</p>


<h3>Value</h3>

<p>A list of 10 elements
</p>
<table>
<tr><td><code>wMat</code></td>
<td>
<p>matrix of posterior probabilities. The rows are SNPs. There are 3 columns. 
The first column is the posterior probability that a SNP belongs to cluster - given genotypes of subjects. 
The second column is the posterior probability that a SNP belongs to cluster 0 given genotypes of subjects. 
The third column is the posterior probability that a SNP belongs to cluster + given genotypes of subjects. 
</p>
</td></tr>
<tr><td><code>memSNPs</code></td>
<td>
<p>a vector of SNP cluster membership for the 3-cluster partitionfrom the mixture of 3 Bayesian hierarchical models.</p>
</td></tr>
<tr><td><code>memSNPs2</code></td>
<td>
<p>a vector of binary SNP cluster membership. 1 indicates the SNP has different MAFs between cases and controls. 0 indicates the SNP has the same MAF in cases as that in controls.</p>
</td></tr>
<tr><td><code>piVec</code></td>
<td>
<p>a vector of cluster mixture proportions.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>the first shape parameter of the beta prior for MAF obtaind from initial 3-cluster partitions based on GWAS.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>the second shape parameter of the beta prior for MAF obtaind from initial 3-cluster partitions based on GWAS.</p>
</td></tr>
<tr><td><code>loop</code></td>
<td>
<p>number of iteration in EM algorithm</p>
</td></tr>
<tr><td><code>diff</code></td>
<td>
<p>sum of the squared difference of cluster mixture proportions between current iteration and previous iteration in EM algorithm. if <code>eps &lt; eps</code>, we claim the EM algorithm converges.</p>
</td></tr>
<tr><td><code>res.limma</code></td>
<td>
<p>object returned by limma</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yan Xu &lt;yanxu@uvic.ca&gt;, 
Li Xing &lt;sfulxing@gmail.com&gt;, 
Jessica Su &lt;rejas@channing.harvard.edu&gt;, 
Xuekui Zhang &lt;xuekui@uvic.ca&gt;, 
Weiliang Qiu &lt;Weiliang.Qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Yan X, Xing L, Su J, Zhang X, Qiu W. 
Model-based clustering for identifying disease-associated SNPs in case-control genome-wide association studies. 
Scientific Reports 9, Article number: 13686 (2019) https://www.nature.com/articles/s41598-019-50229-6.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(esSimDiffPriors)
print(esSimDiffPriors)
fDat = fData(esSimDiffPriors)
print(fDat[1:2,])
print(table(fDat$memGenes))

res = estMemSNPs.oneSetHyperPara(
  es = esSimDiffPriors, 
  var.memSubjs = "memSubjs")

print(table(fDat$memGenes, res$memSNPs))

</code></pre>

<hr>
<h2 id='simGenoFunc'>
Simulate Genotype Data from a Mixture of 3 Bayesian Hierarchical Models
</h2><span id='topic+simGenoFunc'></span>

<h3>Description</h3>

<p>Simulate Genotype Data from a Mixture of 3 Bayesian Hierarchical Models.
The minor allele frequency (MAF) of cases has the same prior
as that of controls. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simGenoFunc(nCases = 100, 
            nControls = 100, 
            nSNPs = 1000, 
            alpha.p = 2, 
            beta.p = 5, 
            pi.p = 0.1, 
            alpha0 = 2, 
            beta0 = 5, 
            pi0 = 0.8, 
            alpha.n = 2, 
            beta.n = 5, 
            pi.n = 0.1, 
            low = 0.02, 
            upp = 0.5, 
            verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simGenoFunc_+3A_ncases">nCases</code></td>
<td>

<p>integer. Number of cases.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_ncontrols">nControls</code></td>
<td>

<p>integer. Number of controls.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_nsnps">nSNPs</code></td>
<td>

<p>integer. Number of SNPs.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_alpha.p">alpha.p</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">+</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_beta.p">beta.p</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">+</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_pi.p">pi.p</code></td>
<td>

<p>numeric. Mixture proportion for cluster <code class="reqn">+</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_alpha0">alpha0</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">0</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_beta0">beta0</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">0</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_pi0">pi0</code></td>
<td>

<p>numeric. Mixture proportion for cluster <code class="reqn">0</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_alpha.n">alpha.n</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">-</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_beta.n">beta.n</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">-</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_pi.n">pi.n</code></td>
<td>

<p>numeric. Mixture proportion for cluster <code class="reqn">-</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_low">low</code></td>
<td>

<p>numeric. A small positive value. If a MAF generated from half-flat shape
bivariate prior is smaller than <code>low</code>, we will delete the SNP to be generated.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_upp">upp</code></td>
<td>

<p>numeric. A positive value. If a MAF generated from half-flat shape
bivariate prior is greater than <code>upp</code>, we will delete the SNP to be generated.
</p>
</td></tr>
<tr><td><code id="simGenoFunc_+3A_verbose">verbose</code></td>
<td>

<p>logical. Indicating if intermediate results or final results should be output 
to output screen.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this simulation, we generate additive-coded genotypes for 3 clusters of SNPs
based on a mixture of 3 Bayesian hierarchical models. 
</p>
<p>In cluster <code class="reqn">+</code>, the minor allele frequency
(MAF) <code class="reqn">\theta_{x+}</code> of cases is greater than the MAF <code class="reqn">\theta_{y+}</code> of
controls.
</p>
<p>In cluster <code class="reqn">0</code>, the MAF <code class="reqn">\theta_{0}</code> of cases is equal to 
the MAF of controls.
</p>
<p>In cluster <code class="reqn">-</code>, the MAF <code class="reqn">\theta_{x-}</code> of cases is smaller than 
the MAF <code class="reqn">\theta_{y-}</code> of controls.
</p>
<p>The proportions of the 3 clusters of SNPs are <code class="reqn">\pi_{+}</code>,
<code class="reqn">\pi_{0}</code>, and <code class="reqn">\pi_{-}</code>, respectively.
</p>
<p>We assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">+</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{+}\left(\theta_{x+}\right)h_{+}\left(\theta_{y+}\right)
I\left(\theta_{x+}&gt;\theta_{y+}\right),
</code>
</p>

<p>where <code class="reqn">I(a)</code> is hte indicator function taking value <code class="reqn">1</code> if the event 
<code class="reqn">a</code> is true, and value <code class="reqn">0</code> otherwise.
The function <code class="reqn">h_{+}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{+}, \beta_{+}\right)</code>.
</p>
<p>We assume <code class="reqn">\theta_{0}</code> has the beta prior <code class="reqn">Beta(\alpha_0, \beta_0)</code>. 
</p>
<p>We also assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">-</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{-}\left(\theta_{x-}\right)h_{-}\left(\theta_{y-}\right)
I\left(\theta_{x-}&gt;\theta_{y-}\right).
</code>
</p>

<p>The function <code class="reqn">h_{-}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{-}, \beta_{-}\right)</code>.
</p>
<p>Given a SNP, we assume Hardy-Weinberg equilibrium holds for its genotypes. 
That is, given MAF <code class="reqn">\theta</code>, the probabilities of genotypes are
</p>
<p style="text-align: center;"><code class="reqn">Pr(geno=2) = \theta^2</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=1) = 2\theta\left(1-\theta\right)</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=0) = \left(1-\theta\right)^2</code>
</p>

<p>We also assume the genotypes <code class="reqn">0</code> (wild-type), <code class="reqn">1</code> (heterozygote), and 
<code class="reqn">2</code> (mutation) follows a multinomial distribution
<code class="reqn">Multinomial\left\{1, \left[
\theta^2, 2\theta\left(1-\theta\right),
\left(1-\theta\right)^2
\right]\right\}</code>
</p>
<p>Note that when we generate MAFs from the half-flat shape bivariate priors, 
we might get very small MAFs or get MAFs <code class="reqn">&gt;0.5</code>. In these cased,
we then delete this SNP.
</p>
<p>So the final number of SNPs generated might be less than the initially-set 
number of SNPs.
</p>


<h3>Value</h3>

<p>An ExpressionSet object stores genotype data.
</p>


<h3>Author(s)</h3>

<p>Yan Xu &lt;yanxu@uvic.ca&gt;, 
Li Xing &lt;sfulxing@gmail.com&gt;, 
Jessica Su &lt;rejas@channing.harvard.edu&gt;, 
Xuekui Zhang &lt;xuekui@uvic.ca&gt;, 
Weiliang Qiu &lt;Weiliang.Qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Yan X, Xing L, Su J, Zhang X, Qiu W. 
Model-based clustering for identifying disease-associated SNPs in case-control genome-wide association studies. 
Scientific Reports 9, Article number: 13686 (2019) https://www.nature.com/articles/s41598-019-50229-6.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2) 

esSim = simGenoFunc(
  nCases = 100,
  nControls = 100,
  nSNPs = 500,
  alpha.p = 2, beta.p = 5, pi.p = 0.1,
  alpha0 = 2, beta0 = 5, pi0 = 0.8,
  alpha.n = 2, beta.n = 5, pi.n = 0.1,
  low = 0.02, upp = 0.5, verbose = FALSE
)

print(esSim)
pDat = pData(esSim)
print(pDat[1:2,])
print(table(pDat$memSubjs))

fDat = fData(esSim)
print(fDat[1:2,])
print(table(fDat$memGenes))
print(table(fDat$memGenes2))
</code></pre>

<hr>
<h2 id='simGenoFuncDiffPriors'>
Simulate Genotype Data from a Mixture of 3 Bayesian Hierarchical Models
</h2><span id='topic+simGenoFuncDiffPriors'></span>

<h3>Description</h3>

<p>Simulate Genotype Data from a Mixture of 3 Bayesian Hierarchical Models.
The minor allele frequency (MAF) of cases has different priors
than that of controls. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simGenoFuncDiffPriors(
    nCases = 100, 
    nControls = 100, 
    nSNPs = 1000, 
    alpha.p.ca = 2, 
    beta.p.ca = 3, 
    alpha.p.co = 2, 
    beta.p.co = 8, 
    pi.p = 0.1, 
    alpha0 = 2, 
    beta0 = 5, 
    pi0 = 0.8, 
    alpha.n.ca = 2, 
    beta.n.ca = 8, 
    alpha.n.co = 2, 
    beta.n.co = 3, 
    pi.n = 0.1, 
    low = 0.02, 
    upp = 0.5, 
    verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simGenoFuncDiffPriors_+3A_ncases">nCases</code></td>
<td>

<p>integer. Number of cases.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_ncontrols">nControls</code></td>
<td>

<p>integer. Number of controls.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_nsnps">nSNPs</code></td>
<td>

<p>integer. Number of SNPs.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_alpha.p.ca">alpha.p.ca</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">+</code> for cases.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_beta.p.ca">beta.p.ca</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">+</code> for cases. 
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_alpha.p.co">alpha.p.co</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">+</code> for controls.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_beta.p.co">beta.p.co</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">+</code> for controls. 
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_pi.p">pi.p</code></td>
<td>

<p>numeric. Mixture proportion for cluster <code class="reqn">+</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_alpha0">alpha0</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">0</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_beta0">beta0</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">0</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_pi0">pi0</code></td>
<td>

<p>numeric. Mixture proportion for cluster <code class="reqn">0</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_alpha.n.ca">alpha.n.ca</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">-</code> for cases.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_beta.n.ca">beta.n.ca</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">-</code> for cases.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_alpha.n.co">alpha.n.co</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">-</code> for controls.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_beta.n.co">beta.n.co</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">-</code> for controls.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_pi.n">pi.n</code></td>
<td>

<p>numeric. Mixture proportion for cluster <code class="reqn">-</code>.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_low">low</code></td>
<td>

<p>numeric. A small positive value. If a MAF generated from half-flat shape
bivariate prior is smaller than <code>low</code>, we will delete the SNP to be generated.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_upp">upp</code></td>
<td>

<p>numeric. A positive value. If a MAF generated from half-flat shape
bivariate prior is greater than <code>upp</code>, we will delete the SNP to be generated.
</p>
</td></tr>
<tr><td><code id="simGenoFuncDiffPriors_+3A_verbose">verbose</code></td>
<td>

<p>logical. Indicating if intermediate results or final results should be output 
to output screen.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this simulation, we generate additive-coded genotypes for 3 clusters of SNPs
based on a mixture of 3 Bayesian hierarchical models. 
</p>
<p>In cluster <code class="reqn">+</code>, the minor allele frequency
(MAF) <code class="reqn">\theta_{x+}</code> of cases is greater than the MAF <code class="reqn">\theta_{y+}</code> of
controls.
</p>
<p>In cluster <code class="reqn">0</code>, the MAF <code class="reqn">\theta_{0}</code> of cases is equal to 
the MAF of controls.
</p>
<p>In cluster <code class="reqn">-</code>, the MAF <code class="reqn">\theta_{x-}</code> of cases is smaller than 
the MAF <code class="reqn">\theta_{y-}</code> of controls.
</p>
<p>The proportions of the 3 clusters of SNPs are <code class="reqn">\pi_{+}</code>,
<code class="reqn">\pi_{0}</code>, and <code class="reqn">\pi_{-}</code>, respectively.
</p>
<p>We assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">+</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{+}\left(\theta_{x+}\right)h_{+}\left(\theta_{y+}\right)
I\left(\theta_{x+}&gt;\theta_{y+}\right),
</code>
</p>

<p>where <code class="reqn">I(a)</code> is hte indicator function taking value <code class="reqn">1</code> if the event 
<code class="reqn">a</code> is true, and value <code class="reqn">0</code> otherwise.
The function <code class="reqn">h_{+}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{+}, \beta_{+}\right)</code>.
</p>
<p>We assume <code class="reqn">\theta_{0}</code> has the beta prior <code class="reqn">Beta(\alpha_0, \beta_0)</code>. 
</p>
<p>We also assume a &ldquo;half-flat shape&rdquo; bivariate prior for the MAF in
cluster <code class="reqn">-</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{-}\left(\theta_{x-}\right)h_{-}\left(\theta_{y-}\right)
I\left(\theta_{x-}&gt;\theta_{y-}\right).
</code>
</p>

<p>The function <code class="reqn">h_{-}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{-}, \beta_{-}\right)</code>.
</p>
<p>Given a SNP, we assume Hardy-Weinberg equilibrium holds for its genotypes. 
That is, given MAF <code class="reqn">\theta</code>, the probabilities of genotypes are
</p>
<p style="text-align: center;"><code class="reqn">Pr(geno=2) = \theta^2</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=1) = 2\theta\left(1-\theta\right)</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=0) = \left(1-\theta\right)^2</code>
</p>

<p>We also assume the genotypes <code class="reqn">0</code> (wild-type), <code class="reqn">1</code> (heterozygote), and 
<code class="reqn">2</code> (mutation) follows a multinomial distribution
<code class="reqn">Multinomial\left\{1, \left[
\theta^2, 2\theta\left(1-\theta\right),
\left(1-\theta\right)^2
\right]\right\}</code>
</p>
<p>Note that when we generate MAFs from the half-flat shape bivariate priors, 
we might get very small MAFs or get MAFs <code class="reqn">&gt;0.5</code>. In these cased,
we then delete this SNP.
</p>
<p>So the final number of SNPs generated might be less than the initially-set 
number of SNPs.
</p>


<h3>Value</h3>

<p>An ExpressionSet object stores genotype data.
</p>


<h3>Author(s)</h3>

<p>Yan Xu &lt;yanxu@uvic.ca&gt;, 
Li Xing &lt;sfulxing@gmail.com&gt;, 
Jessica Su &lt;rejas@channing.harvard.edu&gt;, 
Xuekui Zhang &lt;xuekui@uvic.ca&gt;, 
Weiliang Qiu &lt;Weiliang.Qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Yan X, Xing L, Su J, Zhang X, Qiu W. 
Model-based clustering for identifying disease-associated SNPs in case-control genome-wide association studies. 
Scientific Reports 9, Article number: 13686 (2019) https://www.nature.com/articles/s41598-019-50229-6.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2)

esSimDiffPriors = simGenoFuncDiffPriors(
  nCases = 100,
  nControls = 100,
  nSNPs = 500,
  alpha.p.ca = 2, beta.p.ca = 3,
  alpha.p.co = 2, beta.p.co = 8, pi.p = 0.1,
  alpha0 = 2, beta0 = 5, pi0 = 0.8,
  alpha.n.ca = 2, beta.n.ca = 8,
  alpha.n.co = 2, beta.n.co = 3, pi.n = 0.1,
  low = 0.02, upp = 0.5, verbose = FALSE
)

print(esSimDiffPriors)

pDat = pData(esSimDiffPriors)
print(pDat[1:2,])
print(table(pDat$memSubjs))

fDat = fData(esSimDiffPriors)
print(fDat[1:2,])
print(table(fDat$memGenes))
print(table(fDat$memGenes2))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
