<!DOCTYPE html><html lang="en"><head><title>Help for package gwasforest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gwasforest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gwasforest'><p>Make forest plot with GWAS data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Make Forest Plot with GWAS Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Extract and reform data from GWAS (genome-wide association study) results, and then make a single integrated forest plot containing multiple windows of which each shows the result of individual SNPs (or other items of interest).</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/yilixu/gwasforest">https://github.com/yilixu/gwasforest</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/yilixu/gwasforest/issues">https://github.com/yilixu/gwasforest/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>colorspace, data.table, dplyr, ggplot2, ggrepel, glue, utils</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-11-22 11:06:46 UTC; yilixu</td>
</tr>
<tr>
<td>Author:</td>
<td>Yili Xu [aut, cre],
  Anqi Wang [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yili Xu &lt;yilixu@usc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-11-24 08:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='gwasforest'>Make forest plot with GWAS data</h2><span id='topic+gwasforest'></span>

<h3>Description</h3>

<p>Extract and reform data from GWAS results, and then make a single integrated forest plot containing multiple windows of which each shows the result of individual SNPs (or other items of interest).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gwasforest(
  customFilename,
  customFilename_results = NULL,
  customFilename_studyName = NULL,
  keepStudyOrder = TRUE,
  stdColnames = FALSE,
  customColnames = NULL,
  calculateEXP = FALSE,
  calculateCI = TRUE,
  valueFormat = "Effect",
  metaStudy = "Study1",
  colorMode = "mono",
  forestLayout = "auto",
  plotTitle = "auto",
  showMetaValue = TRUE,
  outputFolderPath = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gwasforest_+3A_customfilename">customFilename</code>, <code id="gwasforest_+3A_customfilename_results">customFilename_results</code></td>
<td>
<p>string, relative or full path to the input file. customFilename for raw GWAS data file, and customFilename_results for gwasforest-generated results file. If customFilename_results is provided, certain downstream calculations will be skipped.</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_customfilename_studyname">customFilename_studyName</code></td>
<td>
<p>(optional) string, relative or full path to the study name file, required if users want to use their own study names which is not standardized (see &quot;stdColnames&quot;); all study names should be in one column with a header; also, study names should be in the order of that they first appear in the input data columns.</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_keepstudyorder">keepStudyOrder</code></td>
<td>
<p>logical, whether to keep studies (except for meta study) in the original order provided by user (from customFilename_studyName), or sort them alphabetically on the combined forest plot; meta study will always be put at the bottom of the combined forest plot.</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_stdcolnames">stdColnames</code></td>
<td>
<p>logical, whether the input data has standardized column names as provided in the instruction example, if TRUE, column order doesn't matter (except that study1 needs to be the Meta study); if FALSE, see &quot;customColnames&quot;.</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_customcolnames">customColnames</code></td>
<td>
<p>character, case-sensitive, can be a vector, choose from c(&quot;Value&quot;, &quot;StdErr&quot;) or c(&quot;Value&quot;, &quot;Upper&quot;, &quot;Lower&quot;) based on what columns are contained in the input data; required if stdColnames = FALSE, also the input data need to be grouped by study while in the customColnames order, e.g. Study1__Value, Study1__StdErr, Study2__Value, Study2__StdErr...; in addition, each study should contain the same number of columns.</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_calculateexp">calculateEXP</code></td>
<td>
<p>logical, whether to calculate exp(Value), if TRUE, downstream calculateCI will also take exp into consideration.</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_calculateci">calculateCI</code></td>
<td>
<p>logical, whether to calculate Confidence Interval, if TRUE, input data need to contain &quot;StdErr&quot; column; if FALSE, input data need to contain &quot;Upper&quot; and &quot;Lower&quot; columns.</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_valueformat">valueFormat</code></td>
<td>
<p>character, format of Value column, e.g. &quot;Effect&quot;, &quot;Beta&quot;, &quot;OR&quot;, &quot;HR&quot;, &quot;logRR&quot;...</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_metastudy">metaStudy</code></td>
<td>
<p>character, which study is the meta study, by default &quot;Study1&quot; (the first study appear in the input data columns).</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_colormode">colorMode</code></td>
<td>
<p>character, choose from c(&quot;mono&quot;, &quot;duo&quot;, &quot;diverse&quot;), mono - render all studies including meta study in the same color, duo - highlight meta study, diverse - render all studies in different colors.</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_forestlayout">forestLayout</code></td>
<td>
<p>character, or integer vector, determines the layout of the combined forest plot, by default use &quot;auto&quot; which will automatically arrange the combined forest plot; or user can explicitly set the row/column layout by providing a vector c(rowNum, colNum).</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_plottitle">plotTitle</code></td>
<td>
<p>character/string, the title of the combined forest plot, can be customized or simply set to &quot;auto&quot;.</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_showmetavalue">showMetaValue</code></td>
<td>
<p>logical, whether to show value for meta group on the combined forest plot.</p>
</td></tr>
<tr><td><code id="gwasforest_+3A_outputfolderpath">outputFolderPath</code></td>
<td>
<p>string, relative or full path to the output folder, can be set to NULL (no output file will be written to the file system).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list, users can run the function without assigning the return value to a variable. If assigned to a variable, it will be a list containing GWAS results (dataframe) and GWAS forest plot (ggplot2 object).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># customFilename in dataframe format (with standardized column names)
tempValue = runif(n = 18, min = 0.01, max = 2)
tempStdErr = tempValue / rep(3:5, times = 6)
eg_customFilename = data.frame(paste0("Marker", 1:6), tempValue[1:6],
    tempStdErr[1:6], tempValue[7:12], tempStdErr[7:12], tempValue[13:18],
    tempStdErr[13:18], stringsAsFactors = FALSE)
colnames(eg_customFilename) = c("MarkerName", paste0(rep("Study", times = 6),
    rep(1:3, each = 2), sample(LETTERS, 6)))
rm(tempValue, tempStdErr)
eg_customFilename_studyName = data.frame("studyName" = paste0("Study", 1:3),
    stringsAsFactors = FALSE)
eg_returnList = gwasforest(eg_customFilename, customFilename_studyName =
    eg_customFilename_studyName, stdColnames = FALSE, customColnames = c("Value",
    "StdErr"), valueFormat = "Effect", metaStudy = "Study1", colorMode = "duo")

# customFilename in dataframe format (without standardized column names),
# with customFilename_studyName provided in dataframe format
tempValue = runif(n = 18, min = 0.01, max = 2)
tempStdErr = tempValue / rep(3:5, times = 6)
eg_customFilename = data.frame(paste0("Marker", 1:6), tempValue[1:6],
    tempStdErr[1:6], tempValue[7:12], tempStdErr[7:12], tempValue[13:18],
    tempStdErr[13:18], stringsAsFactors = FALSE)
colnames(eg_customFilename) = c("MarkerName", paste0(rep("Study", times = 6),
    rep(1:3, each = 2), sample(LETTERS, 6)))
rm(tempValue, tempStdErr)
eg_customFilename_studyName = data.frame("studyName" = paste0("Study", 1:3),
    stringsAsFactors = FALSE)
eg_returnList = gwasforest(eg_customFilename, customFilename_studyName =
    eg_customFilename_studyName, stdColnames = FALSE, customColnames = c("Value",
    "StdErr"), valueFormat = "Effect", metaStudy = "Study1", colorMode = "duo")

# customFilename_results in dataframe format (run either of the two examples
# above to see the example results)
eg_customFilename_results = eg_returnList[[1]]

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
