<!DOCTYPE html><html lang="en"><head><title>Help for package gwid</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gwid}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#build_gwas'><p>Open a SNP GDS file and extract information.</p></a></li>
<li><a href='#build_gwid'><p>Open a ibd file and extract information.</p></a></li>
<li><a href='#build_phase'><p>Read .vcf structured text format files and reduce the size of file.</p></a></li>
<li><a href='#case_control'><p>Reload saved case-control list file</p></a></li>
<li><a href='#extract'><p>Extract information from SNP GDS file.</p></a></li>
<li><a href='#extract_window'><p>extract component of an object</p></a></li>
<li><a href='#extract_window.gwid'><p>Extract information from ibd data in a moving window</p></a></li>
<li><a href='#extract.gwas'><p>Extract information from SNP GDS file.</p></a></li>
<li><a href='#extract.gwid'><p>Extract information from ibd data.</p></a></li>
<li><a href='#fisher_test'><p>Fisher test</p></a></li>
<li><a href='#fisher_test.gwas'><p>Fisher's Exact Test for gwas count data</p></a></li>
<li><a href='#fisher_test.gwid'><p>Fisher's Exact Test for gwid count data</p></a></li>
<li><a href='#fisher_test.result_snps'><p>fisher exact test for result_snps count data</p></a></li>
<li><a href='#gtest'><p>perform gtest</p></a></li>
<li><a href='#gtest.haplotype_structure'><p>Perform G-test on haplotype structures extracted from <code>haplotype_structure</code> function</p></a></li>
<li><a href='#haplotype_frequency'><p>haplotype frequency</p></a></li>
<li><a href='#haplotype_frequency.haplotype_structure'><p>haplotype frequency in sliding windows</p></a></li>
<li><a href='#haplotype_structure'><p>haplotype structures in a window</p></a></li>
<li><a href='#haplotype_structure.gwas'><p>extract haplotype structures of individuals in a window</p></a></li>
<li><a href='#haplotype_structure.gwid'><p>extract haplotype structures of pairwise ibd samples in a window</p></a></li>
<li><a href='#launch_app'><p>laucnh a shiny app</p></a></li>
<li><a href='#mcnemar_test'><p>mcnemar test</p></a></li>
<li><a href='#mcnemar_test_permut'><p>mcnemar permutation</p></a></li>
<li><a href='#mcnemar_test_permut.result_snps'><p>mcnemar permutation test</p></a></li>
<li><a href='#mcnemar_test.result_snps'><p>mcnemar test</p></a></li>
<li><a href='#permutation_test'><p>permutation test</p></a></li>
<li><a href='#permutation_test.gwas'><p>Permutation test for gwas object</p></a></li>
<li><a href='#permutation_test.gwid'><p>permutation test for gwid count data</p></a></li>
<li><a href='#permutation_test.haplotype_structure'><p>Permutation test for 'haplotype_structure' object</p></a></li>
<li><a href='#plot.gwas'><p>Line plot of gwas objects</p></a></li>
<li><a href='#plot.gwid'><p>Line plot of gwid objects</p></a></li>
<li><a href='#plot.haplotype_frequency'><p>Line plot of haplotype_frequency object</p></a></li>
<li><a href='#plot.haplotype_structure_frequency'><p>Two type of line plots for haplotype_structure_frequency objects .</p></a></li>
<li><a href='#plot.result_snps'><p>Line plot of result_snps objects</p></a></li>
<li><a href='#plot.test_snps'><p>Line plot of test_snps objects</p></a></li>
<li><a href='#print'><p>print</p></a></li>
<li><a href='#print.gwas'><p>print gwas instants</p></a></li>
<li><a href='#roh'><p>runs of homozygosity</p></a></li>
<li><a href='#roh.phase'><p>runs of homozygosity</p></a></li>
<li><a href='#subset'><p>subset an object</p></a></li>
<li><a href='#subset.gwid'><p>subset gwid object based on snp position</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genome-Wide Identity-by-Descent</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Soroush Mahmoudiandehkordi &lt;soroushmdg@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Methods and tools for the analysis of Genome Wide 
    Identity-by-Descent ('gwid') mapping data, focusing on testing whether there 
    is a higher occurrence of Identity-By-Descent (IBD) segments around potential causal variants 
    in cases compared to controls, which is crucial for identifying rare 
    variants. To enhance its analytical power, 'gwid' incorporates a Sliding 
    Window Approach, allowing for the detection and analysis of signals from 
    multiple Single Nucleotide Polymorphisms (SNPs).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, gdsfmt, SNPRelate, Matrix, ggplot2, plotly, utils,
stats, RcppRoll, methods, piggyback, shiny, lattice, grid</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, magrittr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/soroushmdg/gwid">https://github.com/soroushmdg/gwid</a>,
<a href="https://soroushmdg.github.io/gwid/">https://soroushmdg.github.io/gwid/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/soroushmdg/gwid/issues">https://github.com/soroushmdg/gwid/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-03 22:52:20 UTC; soroush</td>
</tr>
<tr>
<td>Author:</td>
<td>Soroush Mahmoudiandehkordi [aut, cre],
  Steven J Schrodi [aut],
  Mehdi Maadooliat [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-03 23:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='build_gwas'>Open a SNP GDS file and extract information.</h2><span id='topic+build_gwas'></span>

<h3>Description</h3>

<p>Open a SNP GDS file and extract information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_gwas(gds_data = "name.gds", caco = "name.Rda", gwas_generator = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="build_gwas_+3A_gds_data">gds_data</code></td>
<td>
<p>File name</p>
</td></tr>
<tr><td><code id="build_gwas_+3A_caco">caco</code></td>
<td>
<p>An object of class caco. Output of <code>case_control</code> function.</p>
</td></tr>
<tr><td><code id="build_gwas_+3A_gwas_generator">gwas_generator</code></td>
<td>
<p>logical; if <code>TRUE</code> an object of class result_snps
will be saved inside output list.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of seven objects; including smp.id, snp.id, snp.pos, smp.indx,
smp.snp (a matrix with samples in rows and snp in columns), caco,
snps(column sum of smp.snp for each case control)
</p>

<hr>
<h2 id='build_gwid'>Open a ibd file and extract information.</h2><span id='topic+build_gwid'></span>

<h3>Description</h3>

<p>Open a ibd file and extract information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_gwid(
  ibd_data = "name.ibd",
  gwas = "object of class gwas",
  gwid_generator = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="build_gwid_+3A_ibd_data">ibd_data</code></td>
<td>
<p>a file name for output of <a href="http://faculty.washington.edu/browning/refined-ibd.html">Refined IBD</a></p>
</td></tr>
<tr><td><code id="build_gwid_+3A_gwas">gwas</code></td>
<td>
<p>object of class gwas</p>
</td></tr>
<tr><td><code id="build_gwid_+3A_gwid_generator">gwid_generator</code></td>
<td>
<p>logical; if <code>TRUE</code> an object of class result_snps
will be saved inside output list.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a object(list) of class gwid contains
profile object, IBD object and result_snps object.
</p>

<hr>
<h2 id='build_phase'>Read .vcf structured text format files and reduce the size of file.</h2><span id='topic+build_phase'></span>

<h3>Description</h3>

<p>Read .vcf structured text format files and reduce the size of file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_phase(phased_vcf = "name.vcf", caco)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="build_phase_+3A_phased_vcf">phased_vcf</code></td>
<td>
<p>A file name for a variant call format (vcf) file.</p>
</td></tr>
<tr><td><code id="build_phase_+3A_caco">caco</code></td>
<td>
<p>An object of class caco. Output of <code>case_control</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a a list of class phase contains two sparse matrix
for each haplotype.
</p>

<hr>
<h2 id='case_control'>Reload saved case-control list file</h2><span id='topic+case_control'></span>

<h3>Description</h3>

<p>Reload saved case-control list file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>case_control(case_control_rda, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="case_control_+3A_case_control_rda">case_control_rda</code></td>
<td>
<p>A character string giving the name of the case-control
file to load. The file is a list of character vectors including subject names
in each case-control groups or csv file including subject name for a disease.</p>
</td></tr>
<tr><td><code id="case_control_+3A_...">...</code></td>
<td>
<p>name of a column (disease name) of csv file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output will be a list of character vectors include subject names
and groups. The class of returned object is caco.
</p>

<hr>
<h2 id='extract'>Extract information from SNP GDS file.</h2><span id='topic+extract'></span>

<h3>Description</h3>

<p>Extract information from SNP GDS file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract_+3A_obj">obj</code></td>
<td>
<p>an object of class gwas</p>
</td></tr>
<tr><td><code id="extract_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>extract object instants
</p>

<hr>
<h2 id='extract_window'>extract component of an object</h2><span id='topic+extract_window'></span>

<h3>Description</h3>

<p>extract component of an object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_window(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract_window_+3A_obj">obj</code></td>
<td>
<p>obj</p>
</td></tr>
<tr><td><code id="extract_window_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a result_snps (data.table) object including 3 columns
including, &ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo;
</p>

<hr>
<h2 id='extract_window.gwid'>Extract information from ibd data in a moving window</h2><span id='topic+extract_window.gwid'></span>

<h3>Description</h3>

<p>Extract information from ibd data in a moving window
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwid'
extract_window(obj, w = 10, snp_start, snp_end, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract_window.gwid_+3A_obj">obj</code></td>
<td>
<p>object of class gwid(output of function build_gwid)</p>
</td></tr>
<tr><td><code id="extract_window.gwid_+3A_w">w</code></td>
<td>
<p>window size</p>
</td></tr>
<tr><td><code id="extract_window.gwid_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snp, which we want to aggregate.</p>
</td></tr>
<tr><td><code id="extract_window.gwid_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snp, which we want to aggregate.</p>
</td></tr>
<tr><td><code id="extract_window.gwid_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a result_snps (data.table) object including 3 columns
including, &ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)


</code></pre>

<hr>
<h2 id='extract.gwas'>Extract information from SNP GDS file.</h2><span id='topic+extract.gwas'></span>

<h3>Description</h3>

<p>Extract information from SNP GDS file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwas'
extract(obj, type = c("snps", "snp2", "nas"), snp_start, snp_end, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract.gwas_+3A_obj">obj</code></td>
<td>
<p>object of class gwas.</p>
</td></tr>
<tr><td><code id="extract.gwas_+3A_type">type</code></td>
<td>
<p>indicate type of aggregation on sample-snp data and must be
one of snps, snp2, or nas</p>
</td></tr>
<tr><td><code id="extract.gwas_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snp, which we want to aggregate.</p>
</td></tr>
<tr><td><code id="extract.gwas_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snp, which we want to aggregate.</p>
</td></tr>
<tr><td><code id="extract.gwas_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a result_snps (data.table) object including 3 columns
including, snp_pos, case_control, and value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='extract.gwid'>Extract information from ibd data.</h2><span id='topic+extract.gwid'></span>

<h3>Description</h3>

<p>Extract information from ibd data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwid'
extract(obj = "object of class gwid", snp_start, snp_end, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract.gwid_+3A_obj">obj</code></td>
<td>
<p>object of class gwid(output of function build_gwid)</p>
</td></tr>
<tr><td><code id="extract.gwid_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snp, which we want to aggregate.</p>
</td></tr>
<tr><td><code id="extract.gwid_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snp, which we want to aggregate.</p>
</td></tr>
<tr><td><code id="extract.gwid_+3A_...">...</code></td>
<td>
<p>other objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a result_snps (data.table) object including 3 columns
including, &ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='fisher_test'>Fisher test</h2><span id='topic+fisher_test'></span>

<h3>Description</h3>

<p>Fisher test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fisher_test(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fisher_test_+3A_obj">obj</code></td>
<td>
<p>an object</p>
</td></tr>
<tr><td><code id="fisher_test_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>

<hr>
<h2 id='fisher_test.gwas'>Fisher's Exact Test for gwas count data</h2><span id='topic+fisher_test.gwas'></span>

<h3>Description</h3>

<p>Fisher's Exact Test for gwas count data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwas'
fisher_test(
  obj,
  reference,
  snp_start,
  snp_end,
  alternative = c("two.sided", "greater", "less"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fisher_test.gwas_+3A_obj">obj</code></td>
<td>
<p>object of class gwas</p>
</td></tr>
<tr><td><code id="fisher_test.gwas_+3A_reference">reference</code></td>
<td>
<p>reference group of subjects in which we want to perform fisher test
test</p>
</td></tr>
<tr><td><code id="fisher_test.gwas_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snps.</p>
</td></tr>
<tr><td><code id="fisher_test.gwas_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snp.</p>
</td></tr>
<tr><td><code id="fisher_test.gwas_+3A_alternative">alternative</code></td>
<td>
<p>indicates the alternative hypothesis and must be one of &quot;two.sided&quot;, &quot;greater&quot; or &quot;less&quot;.
You can specify just the initial letter. Only used in the 2 by 2 case</p>
</td></tr>
<tr><td><code id="fisher_test.gwas_+3A_...">...</code></td>
<td>
<p>optional arguments to fisher.test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='fisher_test.gwid'>Fisher's Exact Test for gwid count data</h2><span id='topic+fisher_test.gwid'></span>

<h3>Description</h3>

<p>Fisher's Exact Test for gwid count data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwid'
fisher_test(
  obj,
  caco,
  snp_start,
  snp_end,
  reference,
  alternative = c("two.sided", "greater", "less"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fisher_test.gwid_+3A_obj">obj</code></td>
<td>
<p>An object of class gwid. Output of <code>build_gwid</code> function</p>
</td></tr>
<tr><td><code id="fisher_test.gwid_+3A_caco">caco</code></td>
<td>
<p>An object of class caco. Output of <code>case_control</code> function.</p>
</td></tr>
<tr><td><code id="fisher_test.gwid_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snps.</p>
</td></tr>
<tr><td><code id="fisher_test.gwid_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snp.</p>
</td></tr>
<tr><td><code id="fisher_test.gwid_+3A_reference">reference</code></td>
<td>
<p>reference group of subjects in which we want to perform fisher test</p>
</td></tr>
<tr><td><code id="fisher_test.gwid_+3A_alternative">alternative</code></td>
<td>
<p>indicates the alternative hypothesis and must be one of &quot;two.sided&quot;, &quot;greater&quot; or &quot;less&quot;.
You can specify just the initial letter. Only used in the 2 by 2 case</p>
</td></tr>
<tr><td><code id="fisher_test.gwid_+3A_...">...</code></td>
<td>
<p>optional arguments to fisher.test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='fisher_test.result_snps'>fisher exact test for result_snps count data</h2><span id='topic+fisher_test.result_snps'></span>

<h3>Description</h3>

<p>fisher exact test for result_snps count data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'result_snps'
fisher_test(
  obj,
  caco,
  reference,
  alternative = c("two.sided", "greater", "less"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fisher_test.result_snps_+3A_obj">obj</code></td>
<td>
<p>An object of class result_snps</p>
</td></tr>
<tr><td><code id="fisher_test.result_snps_+3A_caco">caco</code></td>
<td>
<p>An object of class caco. Output of <code>case_control</code> function.</p>
</td></tr>
<tr><td><code id="fisher_test.result_snps_+3A_reference">reference</code></td>
<td>
<p>reference group of subjects in which we want to perform fisher test.</p>
</td></tr>
<tr><td><code id="fisher_test.result_snps_+3A_alternative">alternative</code></td>
<td>
<p>indicates the alternative hypothesis and must be one of &quot;two.sided&quot;, &quot;greater&quot; or &quot;less&quot;.
You can specify just the initial letter. Only used in the 2 by 2 case</p>
</td></tr>
<tr><td><code id="fisher_test.result_snps_+3A_...">...</code></td>
<td>
<p>optional arguments to fisher.test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
model_permutation &lt;- permutation_test(ibd_data,snp_data_gds,
snp_start = 119026294,snp_end = 120613594,nperm=20,reference = "cases")
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='gtest'>perform gtest</h2><span id='topic+gtest'></span>

<h3>Description</h3>

<p>perform gtest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtest(haplotype_structure, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gtest_+3A_haplotype_structure">haplotype_structure</code></td>
<td>
<p>object of a class</p>
</td></tr>
<tr><td><code id="gtest_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>

<hr>
<h2 id='gtest.haplotype_structure'>Perform G-test on haplotype structures extracted from <code>haplotype_structure</code> function</h2><span id='topic+gtest.haplotype_structure'></span>

<h3>Description</h3>

<p>Perform G-test on haplotype structures extracted from <code>haplotype_structure</code> function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'haplotype_structure'
gtest(haplotype_structure, reference, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gtest.haplotype_structure_+3A_haplotype_structure">haplotype_structure</code></td>
<td>
<p>An object of class haplotype_structure. Output of
<code>haplotype_structure</code> function.</p>
</td></tr>
<tr><td><code id="gtest.haplotype_structure_+3A_reference">reference</code></td>
<td>
<p>reference group of subjects in which we want to perform G-test</p>
</td></tr>
<tr><td><code id="gtest.haplotype_structure_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>

<hr>
<h2 id='haplotype_frequency'>haplotype frequency</h2><span id='topic+haplotype_frequency'></span>

<h3>Description</h3>

<p>haplotype frequency
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplotype_frequency(haplotype_structure, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haplotype_frequency_+3A_haplotype_structure">haplotype_structure</code></td>
<td>
<p>object of class haplotype structure</p>
</td></tr>
<tr><td><code id="haplotype_frequency_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class haplotype_frequency contains of two objects. first one
is object of haplotype_structure_frequency (data.table) and second one is object of class result_snps(data.table)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='haplotype_frequency.haplotype_structure'>haplotype frequency in sliding windows</h2><span id='topic+haplotype_frequency.haplotype_structure'></span>

<h3>Description</h3>

<p>haplotype frequency in sliding windows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'haplotype_structure'
haplotype_frequency(haplotype_structure, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haplotype_frequency.haplotype_structure_+3A_haplotype_structure">haplotype_structure</code></td>
<td>
<p>An object of class haplotype_structure. Output of
<code>haplotype_structure</code> function.</p>
</td></tr>
<tr><td><code id="haplotype_frequency.haplotype_structure_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class haplotype_frequency contains of two objects. first one
is object of haplotype_structure_frequency (data.table) and second one is object of class result_snps(data.table)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='haplotype_structure'>haplotype structures in a window</h2><span id='topic+haplotype_structure'></span>

<h3>Description</h3>

<p>haplotype structures in a window
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplotype_structure(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haplotype_structure_+3A_obj">obj</code></td>
<td>
<p>object</p>
</td></tr>
<tr><td><code id="haplotype_structure_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output will be an object of class haplotype_structure (data.table) that has information
about subjects haplotype structures in a a window.
</p>

<hr>
<h2 id='haplotype_structure.gwas'>extract haplotype structures of individuals in a window</h2><span id='topic+haplotype_structure.gwas'></span>

<h3>Description</h3>

<p>extract haplotype structures of individuals in a window
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwas'
haplotype_structure(obj, phase, w = 10, snp_start, snp_end, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haplotype_structure.gwas_+3A_obj">obj</code></td>
<td>
<p>object of class gwas</p>
</td></tr>
<tr><td><code id="haplotype_structure.gwas_+3A_phase">phase</code></td>
<td>
<p>An object of class phase. Output of <code>build_phase</code> function</p>
</td></tr>
<tr><td><code id="haplotype_structure.gwas_+3A_w">w</code></td>
<td>
<p>window size</p>
</td></tr>
<tr><td><code id="haplotype_structure.gwas_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snps.</p>
</td></tr>
<tr><td><code id="haplotype_structure.gwas_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snps.</p>
</td></tr>
<tr><td><code id="haplotype_structure.gwas_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output will be an object of class haplotype_structure (data.table) that has information
about subjects haplotype structures in a a window.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='haplotype_structure.gwid'>extract haplotype structures of pairwise ibd samples in a window</h2><span id='topic+haplotype_structure.gwid'></span>

<h3>Description</h3>

<p>extract haplotype structures of pairwise ibd samples in a window
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwid'
haplotype_structure(obj, phase, w = 10, snp_start, snp_end, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haplotype_structure.gwid_+3A_obj">obj</code></td>
<td>
<p>An object of class gwid. Output of <code>build_gwid</code> function.</p>
</td></tr>
<tr><td><code id="haplotype_structure.gwid_+3A_phase">phase</code></td>
<td>
<p>An object of class phase. Output of <code>build_phase</code> function.</p>
</td></tr>
<tr><td><code id="haplotype_structure.gwid_+3A_w">w</code></td>
<td>
<p>window size</p>
</td></tr>
<tr><td><code id="haplotype_structure.gwid_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snps.</p>
</td></tr>
<tr><td><code id="haplotype_structure.gwid_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snps.</p>
</td></tr>
<tr><td><code id="haplotype_structure.gwid_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output will be an object of class haplotype_structure (data.table) that has information
about subjects haplotype structures in a a window.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='launch_app'>laucnh a shiny app</h2><span id='topic+launch_app'></span>

<h3>Description</h3>

<p>laucnh a shiny app
</p>


<h3>Usage</h3>

<pre><code class='language-R'>launch_app(data_folder_address, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="launch_app_+3A_data_folder_address">data_folder_address</code></td>
<td>
<p>address of the folder that your data folders are.
for example if you have two sets of data such as data1 and data2 and they are
in mydata folder then your data_folder_address should be &quot;./mydata&quot;</p>
</td></tr>
<tr><td><code id="launch_app_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>open a shiny app
</p>

<hr>
<h2 id='mcnemar_test'>mcnemar test</h2><span id='topic+mcnemar_test'></span>

<h3>Description</h3>

<p>mcnemar test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcnemar_test(roh, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mcnemar_test_+3A_roh">roh</code></td>
<td>
<p>roh as class result_snp</p>
</td></tr>
<tr><td><code id="mcnemar_test_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>

<hr>
<h2 id='mcnemar_test_permut'>mcnemar permutation</h2><span id='topic+mcnemar_test_permut'></span>

<h3>Description</h3>

<p>mcnemar permutation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcnemar_test_permut(mcnemar, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mcnemar_test_permut_+3A_mcnemar">mcnemar</code></td>
<td>
<p>macnemar test output</p>
</td></tr>
<tr><td><code id="mcnemar_test_permut_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>

<hr>
<h2 id='mcnemar_test_permut.result_snps'>mcnemar permutation test</h2><span id='topic+mcnemar_test_permut.result_snps'></span>

<h3>Description</h3>

<p>mcnemar permutation test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'result_snps'
mcnemar_test_permut(
  mcnemar = "object of class result_snps (output of function mcnemar_test with fun=sum)",
  roh_mat = "output of roh function when roh_mat = TRUE",
  gwas = "object of class gwas",
  nperm = 1000,
  reference = "cases",
  w,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mcnemar_test_permut.result_snps_+3A_mcnemar">mcnemar</code></td>
<td>
<p>macnemar test output</p>
</td></tr>
<tr><td><code id="mcnemar_test_permut.result_snps_+3A_roh_mat">roh_mat</code></td>
<td>
<p>roh matrix</p>
</td></tr>
<tr><td><code id="mcnemar_test_permut.result_snps_+3A_gwas">gwas</code></td>
<td>
<p>gwas</p>
</td></tr>
<tr><td><code id="mcnemar_test_permut.result_snps_+3A_nperm">nperm</code></td>
<td>
<p>number of permutation</p>
</td></tr>
<tr><td><code id="mcnemar_test_permut.result_snps_+3A_reference">reference</code></td>
<td>
<p>reference group</p>
</td></tr>
<tr><td><code id="mcnemar_test_permut.result_snps_+3A_w">w</code></td>
<td>
<p>window</p>
</td></tr>
<tr><td><code id="mcnemar_test_permut.result_snps_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>

<hr>
<h2 id='mcnemar_test.result_snps'>mcnemar test</h2><span id='topic+mcnemar_test.result_snps'></span>

<h3>Description</h3>

<p>mcnemar test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'result_snps'
mcnemar_test(
  roh = "object of class result_snps (output of function roh with fun=sum)",
  reference,
  w = 10,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mcnemar_test.result_snps_+3A_roh">roh</code></td>
<td>
<p>An object of class result_snps (output of function roh with fun=sum)</p>
</td></tr>
<tr><td><code id="mcnemar_test.result_snps_+3A_reference">reference</code></td>
<td>
<p>reference group of subjects in which we want to perform fisher test.</p>
</td></tr>
<tr><td><code id="mcnemar_test.result_snps_+3A_w">w</code></td>
<td>
<p>window size</p>
</td></tr>
<tr><td><code id="mcnemar_test.result_snps_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>

<hr>
<h2 id='permutation_test'>permutation test</h2><span id='topic+permutation_test'></span>

<h3>Description</h3>

<p>permutation test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>permutation_test(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="permutation_test_+3A_obj">obj</code></td>
<td>
<p>object</p>
</td></tr>
<tr><td><code id="permutation_test_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>

<hr>
<h2 id='permutation_test.gwas'>Permutation test for gwas object</h2><span id='topic+permutation_test.gwas'></span>

<h3>Description</h3>

<p>Permutation test for gwas object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwas'
permutation_test(
  obj,
  snp_start,
  snp_end,
  nperm = 1000,
  reference = "cases",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="permutation_test.gwas_+3A_obj">obj</code></td>
<td>
<p>object of class gwas</p>
</td></tr>
<tr><td><code id="permutation_test.gwas_+3A_snp_start">snp_start</code></td>
<td>
<p>elect starting position of snps.</p>
</td></tr>
<tr><td><code id="permutation_test.gwas_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snp.</p>
</td></tr>
<tr><td><code id="permutation_test.gwas_+3A_nperm">nperm</code></td>
<td>
<p>Number of permutations.</p>
</td></tr>
<tr><td><code id="permutation_test.gwas_+3A_reference">reference</code></td>
<td>
<p>reference group of subjects in which we want to perform fisher test</p>
</td></tr>
<tr><td><code id="permutation_test.gwas_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
model_permutation &lt;- permutation_test(ibd_data,snp_data_gds,
snp_start = 119026294,snp_end = 120613594,nperm=20,reference = "cases")
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='permutation_test.gwid'>permutation test for gwid count data</h2><span id='topic+permutation_test.gwid'></span>

<h3>Description</h3>

<p>permutation test for gwid count data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwid'
permutation_test(
  obj,
  gwas,
  snp_start,
  snp_end,
  nperm = 100,
  reference = "cases",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="permutation_test.gwid_+3A_obj">obj</code></td>
<td>
<p>An object of class gwid. Output of <code>build_gwid</code> function</p>
</td></tr>
<tr><td><code id="permutation_test.gwid_+3A_gwas">gwas</code></td>
<td>
<p>object of class gwas</p>
</td></tr>
<tr><td><code id="permutation_test.gwid_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snps.</p>
</td></tr>
<tr><td><code id="permutation_test.gwid_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snp.</p>
</td></tr>
<tr><td><code id="permutation_test.gwid_+3A_nperm">nperm</code></td>
<td>
<p>Number of permutations.</p>
</td></tr>
<tr><td><code id="permutation_test.gwid_+3A_reference">reference</code></td>
<td>
<p>reference group</p>
</td></tr>
<tr><td><code id="permutation_test.gwid_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
model_permutation &lt;- permutation_test(ibd_data,snp_data_gds,
snp_start = 119026294,snp_end = 120613594,nperm=20,reference = "cases")
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='permutation_test.haplotype_structure'>Permutation test for 'haplotype_structure' object</h2><span id='topic+permutation_test.haplotype_structure'></span>

<h3>Description</h3>

<p>Permutation test for 'haplotype_structure' object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'haplotype_structure'
permutation_test(obj, nperm, reference, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="permutation_test.haplotype_structure_+3A_obj">obj</code></td>
<td>
<p>object of class 'haplotype_structure'</p>
</td></tr>
<tr><td><code id="permutation_test.haplotype_structure_+3A_nperm">nperm</code></td>
<td>
<p>Number of permutations.</p>
</td></tr>
<tr><td><code id="permutation_test.haplotype_structure_+3A_reference">reference</code></td>
<td>
<p>reference group of subjects in which we want to perform 'gtest'</p>
</td></tr>
<tr><td><code id="permutation_test.haplotype_structure_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
&ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo; which is
a p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
model_permutation &lt;- permutation_test(ibd_data,snp_data_gds,
snp_start = 119026294,snp_end = 120613594,nperm=20,reference = "cases")
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='plot.gwas'>Line plot of gwas objects</h2><span id='topic+plot.gwas'></span>

<h3>Description</h3>

<p>Line plot of gwas objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwas'
plot(x, y = NA, title = "number of snps", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.gwas_+3A_x">x</code></td>
<td>
<p>object of class gwas.</p>
</td></tr>
<tr><td><code id="plot.gwas_+3A_y">y</code></td>
<td>
<p>default value is NA, if specified it should be a vector of names of
subject groups i.e. y = c(&quot;case&quot;,&quot;control&quot;)</p>
</td></tr>
<tr><td><code id="plot.gwas_+3A_title">title</code></td>
<td>
<p>title of the plot.</p>
</td></tr>
<tr><td><code id="plot.gwas_+3A_...">...</code></td>
<td>
<p>optional argument of <code>plot</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>an interactive line plot of gwas objects for each case control subjects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='plot.gwid'>Line plot of gwid objects</h2><span id='topic+plot.gwid'></span>

<h3>Description</h3>

<p>Line plot of gwid objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwid'
plot(
  x,
  y = NA,
  title = "number of IBD in each snp",
  plot_type = c("result_snps", "profile"),
  reference,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.gwid_+3A_x">x</code></td>
<td>
<p>An object of class gwid. Output of <code>build_gwid</code> function.</p>
</td></tr>
<tr><td><code id="plot.gwid_+3A_y">y</code></td>
<td>
<p>default value is NA, if specified it should be a vector of names of
subject groups i.e. y = c(&quot;case&quot;,&quot;control&quot;)</p>
</td></tr>
<tr><td><code id="plot.gwid_+3A_title">title</code></td>
<td>
<p>title of the plot.</p>
</td></tr>
<tr><td><code id="plot.gwid_+3A_plot_type">plot_type</code></td>
<td>
<p>either &ldquo;result_snps&rdquo; or &ldquo;profile&rdquo;.</p>
</td></tr>
<tr><td><code id="plot.gwid_+3A_reference">reference</code></td>
<td>
<p>reference group of subjects in which we want to have profile plot.</p>
</td></tr>
<tr><td><code id="plot.gwid_+3A_...">...</code></td>
<td>
<p>if plot_type is &ldquo;result_snps&rdquo; it is optional argument of <code>plot</code>.
if plot_type is &ldquo;profile&rdquo; we can subset plot based on snp_start and snp_end locations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if plot_type is &ldquo;result_snps&rdquo; an interactive line plot of result_snps for each case control subjects.
if plot_type is &ldquo;profile&rdquo; an interactive profile plot of identity by descent subjects in subset of locations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='plot.haplotype_frequency'>Line plot of haplotype_frequency object</h2><span id='topic+plot.haplotype_frequency'></span>

<h3>Description</h3>

<p>Line plot of haplotype_frequency object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'haplotype_frequency'
plot(
  x,
  y = NA,
  plot_type = c("haplotype_structure_frequency", "result_snps"),
  type = c("version1", "version2"),
  ly = TRUE,
  nwin,
  title,
  line_size = 0.6,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.haplotype_frequency_+3A_x">x</code></td>
<td>
<p>an object of class haplotype_frequency</p>
</td></tr>
<tr><td><code id="plot.haplotype_frequency_+3A_y">y</code></td>
<td>
<p>default value is 'NA', if specified it should be a vector of names of
subject groups i.e. 'y = c(&quot;case&quot;,&quot;control&quot;)'</p>
</td></tr>
<tr><td><code id="plot.haplotype_frequency_+3A_plot_type">plot_type</code></td>
<td>
<p>either &ldquo;result_snps&rdquo;
or &ldquo;&quot;haplotype_structure_frequency&quot;&rdquo;</p>
</td></tr>
<tr><td><code id="plot.haplotype_frequency_+3A_type">type</code></td>
<td>
<p>either &ldquo;version1&rdquo; or &ldquo;version2&rdquo;
when plot_type is &ldquo;&quot;haplotype_structure_frequency&quot;&rdquo;</p>
</td></tr>
<tr><td><code id="plot.haplotype_frequency_+3A_ly">ly</code></td>
<td>
<p>if TRUE, we have a plotly object and if it is false plot
is going to be a ggplot object.</p>
</td></tr>
<tr><td><code id="plot.haplotype_frequency_+3A_nwin">nwin</code></td>
<td>
<p>window number</p>
</td></tr>
<tr><td><code id="plot.haplotype_frequency_+3A_title">title</code></td>
<td>
<p>title of the plot.</p>
</td></tr>
<tr><td><code id="plot.haplotype_frequency_+3A_line_size">line_size</code></td>
<td>
<p>geom_line size</p>
</td></tr>
<tr><td><code id="plot.haplotype_frequency_+3A_...">...</code></td>
<td>
<p>optional argument of <code>plot</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>an interactive line plot of haplotype_frequency objects for
each case control subjects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",
dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control)
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='plot.haplotype_structure_frequency'>Two type of line plots for haplotype_structure_frequency objects .</h2><span id='topic+plot.haplotype_structure_frequency'></span>

<h3>Description</h3>

<p>Two type of line plots for haplotype_structure_frequency objects .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'haplotype_structure_frequency'
plot(
  x,
  y = NA,
  type = c("version1", "version2"),
  nwin,
  ly = TRUE,
  line_size = 0.6,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.haplotype_structure_frequency_+3A_x">x</code></td>
<td>
<p>an object of class haplotype_structure_frequency</p>
</td></tr>
<tr><td><code id="plot.haplotype_structure_frequency_+3A_y">y</code></td>
<td>
<p>default value is NA, if specified it should be a vector of names of
subject groups i.e. y = c(&quot;case&quot;,&quot;control&quot;)</p>
</td></tr>
<tr><td><code id="plot.haplotype_structure_frequency_+3A_type">type</code></td>
<td>
<p>either &ldquo;version1&rdquo; or &ldquo;version2&rdquo;</p>
</td></tr>
<tr><td><code id="plot.haplotype_structure_frequency_+3A_nwin">nwin</code></td>
<td>
<p>window number</p>
</td></tr>
<tr><td><code id="plot.haplotype_structure_frequency_+3A_ly">ly</code></td>
<td>
<p>if 'TRUE', we have a 'plotly' object and if it is 'FALSE' plot is going to be
a 'ggplot' object.</p>
</td></tr>
<tr><td><code id="plot.haplotype_structure_frequency_+3A_line_size">line_size</code></td>
<td>
<p>geom_line size</p>
</td></tr>
<tr><td><code id="plot.haplotype_structure_frequency_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an interactive line plot of haplotype_structure_frequency objects for each case control subjects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='plot.result_snps'>Line plot of result_snps objects</h2><span id='topic+plot.result_snps'></span>

<h3>Description</h3>

<p>Line plot of result_snps objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'result_snps'
plot(x, y = NA, title, snp_start, snp_end, ly = TRUE, line_size = 0.6, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.result_snps_+3A_x">x</code></td>
<td>
<p>An object of class result_snps.</p>
</td></tr>
<tr><td><code id="plot.result_snps_+3A_y">y</code></td>
<td>
<p>default value is NA, if specified it should be a vector of names of
subject groups i.e. y = c(&quot;case&quot;,&quot;control&quot;)</p>
</td></tr>
<tr><td><code id="plot.result_snps_+3A_title">title</code></td>
<td>
<p>title of the plot.</p>
</td></tr>
<tr><td><code id="plot.result_snps_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snps.</p>
</td></tr>
<tr><td><code id="plot.result_snps_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snps.</p>
</td></tr>
<tr><td><code id="plot.result_snps_+3A_ly">ly</code></td>
<td>
<p>if TRUE, we have a plotly object and if it is false plot is going to be
a ggplot object.</p>
</td></tr>
<tr><td><code id="plot.result_snps_+3A_line_size">line_size</code></td>
<td>
<p>geom_line size</p>
</td></tr>
<tr><td><code id="plot.result_snps_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an interactive line plot of result_snps for each case control subjects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps)
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='plot.test_snps'>Line plot of test_snps objects</h2><span id='topic+plot.test_snps'></span>

<h3>Description</h3>

<p>Line plot of test_snps objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'test_snps'
plot(
  x,
  y = NA,
  title,
  snp_start,
  snp_end,
  ly = TRUE,
  line_size = 0.6,
  log_transformation = TRUE,
  QQplot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.test_snps_+3A_x">x</code></td>
<td>
<p>an object of class test_snps.</p>
</td></tr>
<tr><td><code id="plot.test_snps_+3A_y">y</code></td>
<td>
<p>default value is NA, if specified it should be a vector of names of
subject groups i.e. y = c(&quot;case&quot;,&quot;control&quot;)</p>
</td></tr>
<tr><td><code id="plot.test_snps_+3A_title">title</code></td>
<td>
<p>title of the plot.</p>
</td></tr>
<tr><td><code id="plot.test_snps_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snps.</p>
</td></tr>
<tr><td><code id="plot.test_snps_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snps.</p>
</td></tr>
<tr><td><code id="plot.test_snps_+3A_ly">ly</code></td>
<td>
<p>if 'TRUE', we have a 'plotly' object and if it is 'FALSE' plot
is going to be a 'ggplot' object.</p>
</td></tr>
<tr><td><code id="plot.test_snps_+3A_line_size">line_size</code></td>
<td>
<p>geom_line size</p>
</td></tr>
<tr><td><code id="plot.test_snps_+3A_log_transformation">log_transformation</code></td>
<td>
<p>if 'TRUE' plot -log10 transformation of p_values.</p>
</td></tr>
<tr><td><code id="plot.test_snps_+3A_qqplot">QQplot</code></td>
<td>
<p>if TRUE, plot QQplot of P-values</p>
</td></tr>
<tr><td><code id="plot.test_snps_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an interactive line plot of test_snps objects for each case control subjects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='print'>print</h2><span id='topic+print'></span>

<h3>Description</h3>

<p>print
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print_+3A_x">x</code></td>
<td>
<p>an object</p>
</td></tr>
<tr><td><code id="print_+3A_...">...</code></td>
<td>
<p>other objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>print an object
</p>

<hr>
<h2 id='print.gwas'>print gwas instants</h2><span id='topic+print.gwas'></span>

<h3>Description</h3>

<p>print gwas instants
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwas'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.gwas_+3A_x">x</code></td>
<td>
<p>object gwas</p>
</td></tr>
<tr><td><code id="print.gwas_+3A_...">...</code></td>
<td>
<p>other objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>print number of subjects and number of SNPs of a GWAS object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
print(snp_data_gds)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

<hr>
<h2 id='roh'>runs of homozygosity</h2><span id='topic+roh'></span>

<h3>Description</h3>

<p>runs of homozygosity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>roh(phase, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="roh_+3A_phase">phase</code></td>
<td>
<p>object of phase</p>
</td></tr>
<tr><td><code id="roh_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>runs of homozygosity data table or matrix
</p>

<hr>
<h2 id='roh.phase'>runs of homozygosity</h2><span id='topic+roh.phase'></span>

<h3>Description</h3>

<p>runs of homozygosity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phase'
roh(
  phase,
  gwas,
  w = 10,
  fun = c("sum", "mean"),
  snp_start,
  snp_end,
  roh_mat = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="roh.phase_+3A_phase">phase</code></td>
<td>
<p>An object of class phase. Output of <code>build_phase</code> function</p>
</td></tr>
<tr><td><code id="roh.phase_+3A_gwas">gwas</code></td>
<td>
<p>object of class gwas</p>
</td></tr>
<tr><td><code id="roh.phase_+3A_w">w</code></td>
<td>
<p>window size</p>
</td></tr>
<tr><td><code id="roh.phase_+3A_fun">fun</code></td>
<td>
<p>an aggregate function. either  &ldquo;sum&rdquo; or  &ldquo;mean&rdquo;</p>
</td></tr>
<tr><td><code id="roh.phase_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snps.</p>
</td></tr>
<tr><td><code id="roh.phase_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snps.</p>
</td></tr>
<tr><td><code id="roh.phase_+3A_roh_mat">roh_mat</code></td>
<td>
<p>return roh as matrix</p>
</td></tr>
<tr><td><code id="roh.phase_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a result_snps (data.table) object including 3 columns
including, &ldquo;snp_pos&rdquo;, &ldquo;case_control&rdquo;, and &ldquo;value&rdquo;
</p>

<hr>
<h2 id='subset'>subset an object</h2><span id='topic+subset'></span>

<h3>Description</h3>

<p>subset an object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset_+3A_obj">obj</code></td>
<td>
<p>object</p>
</td></tr>
<tr><td><code id="subset_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a object(list) of class gwid contains
profile object and result_snps object.
</p>

<hr>
<h2 id='subset.gwid'>subset gwid object based on snp position</h2><span id='topic+subset.gwid'></span>

<h3>Description</h3>

<p>subset gwid object based on snp position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gwid'
subset(obj, snp_start, snp_end, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset.gwid_+3A_obj">obj</code></td>
<td>
<p>object of class gwid(output of function build_gwid)</p>
</td></tr>
<tr><td><code id="subset.gwid_+3A_snp_start">snp_start</code></td>
<td>
<p>select starting position of snp, which we want to aggregate.</p>
</td></tr>
<tr><td><code id="subset.gwid_+3A_snp_end">snp_end</code></td>
<td>
<p>select ending position of snp, which we want to aggregate.</p>
</td></tr>
<tr><td><code id="subset.gwid_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the output will be a object(list) of class gwid contains
profile object and result_snps object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
