<!DOCTYPE html><html><head><title>Help for package gwrpvr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {gwrpvr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calc_pvalue'><p>calc_pvalue()</p></a></li>
<li><a href='#close_to_normal'><p>This is a CLT-linked run-time control.</p></a></li>
<li><a href='#gwrpv'><p>Genome-Wide Regression P-Value (gwrpv) in R</p></a></li>
<li><a href='#gwrpv_batch'><p>Batch computation of a list of pvalues of GWA regression beta statistics using a bernoulli-normal mixture distribution</p></a></li>
<li><a href='#gwrpvr'><p>gwrpvr: A package for calculating Genome-Wide Regression P-Values (gwrpv) in R</p></a></li>
<li><a href='#highlow'><p>highlow()</p></a></li>
<li><a href='#loop_calc_pvalue'><p>loop_calc_pvalue()</p></a></li>
<li><a href='#regresults'><p>regresults: sample data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genome-Wide Regression P-Value (Gwrpv)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-10-13</td>
</tr>
<tr>
<td>Author:</td>
<td>Gregory Connor [aut],
  Michael O'Neill [trl, aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael O'Neill &lt;m.oneill@ucd.ie&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Computes the sample probability value (p-value)
    for the estimated coefficient from a standard
    genome-wide univariate regression.  It computes the exact
    finite-sample p-value under the assumption that the measured
    phenotype (the dependent variable in the regression) has
    a known Bernoulli-normal mixture distribution.
    Finite-sample genome-wide regression p-values (Gwrpv) with
    a non-normally distributed phenotype (Gregory Connor and
    Michael O'Neill, bioRxiv 204727 &lt;<a href="https://doi.org/10.1101%2F204727">doi:10.1101/204727</a>&gt;).</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://doi.org/10.1101/204727">https://doi.org/10.1101/204727</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-10-19 17:05:56 UTC; mike</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-10-19 17:47:54 UTC</td>
</tr>
</table>
<hr>
<h2 id='calc_pvalue'>calc_pvalue()</h2><span id='topic+calc_pvalue'></span>

<h3>Description</h3>

<p>calculate the pvalue : called from loop_calc_pvalue()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_pvalue(n0a, n1a, n2a, n0, n1, n2, pa, pb, x, mua, mub, sumsqx, siga, sigb,
  vary, beta, skipiter, pvalue)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_pvalue_+3A_n0a">n0a</code></td>
<td>
<p>outer loop index</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_n1a">n1a</code></td>
<td>
<p>middle loop index</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_n2a">n2a</code></td>
<td>
<p>inner loop index</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_n0">n0</code></td>
<td>
<p>the major allele homozygotes</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_n1">n1</code></td>
<td>
<p>the major allele heterozygotes</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_n2">n2</code></td>
<td>
<p>the minor allele zygotes</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_pa">pa</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_pb">pb</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_x">x</code></td>
<td>
<p>a zero mean explanatory variable from the SNP data set</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_mua">mua</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_mub">mub</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_sumsqx">sumsqx</code></td>
<td>
<p>sum of the squares of x</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_siga">siga</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_sigb">sigb</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_vary">vary</code></td>
<td>
<p>vary &lt;- pa*(mua^2+siga^2)+pb*(mub^2+sigb^2)-(pa*mua+pb*mub)^2</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_beta">beta</code></td>
<td>
<p>the beta from the regression being tested</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_skipiter">skipiter</code></td>
<td>
<p>flag to determine if we can skip some calculations</p>
</td></tr>
<tr><td><code id="calc_pvalue_+3A_pvalue">pvalue</code></td>
<td>
<p>the input pvalue prior to calculating new improved pvalue</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pvalue
</p>

<hr>
<h2 id='close_to_normal'>This is a CLT-linked run-time control.</h2><span id='topic+close_to_normal'></span>

<h3>Description</h3>

<p>If the number of observations is large enough that a normality approximation
holds for the y average across the major homozygote subsample, then the code skips the
time-consuming loop over n0, n1 and n2 and and uses the normal approximation for the average y
for the major homozygote subsample. The remaining loop is only over n1 and n2.
The only new input/output variables are input lognearnorm (the magnitude of maximum allowed
tolerance (in log 10 format) for the sum of squared deviation of skewness and
kurtosis from their normal values and output stopiter
(a zero if the code does not mandate a stop to the iterative estimation and a one if it does).
The input variable lognearnorm has a default value set so that users only have to enter it if they want to over-ride the default value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>close_to_normal(totnobs, n0, n1, n2, pa, pb, mua, mub, siga, sigb, beta,
  nearnorm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="close_to_normal_+3A_totnobs">totnobs</code></td>
<td>
<p>the sum of n0, n1, n2</p>
</td></tr>
<tr><td><code id="close_to_normal_+3A_n0">n0</code></td>
<td>
<p>the major allele homozygotes</p>
</td></tr>
<tr><td><code id="close_to_normal_+3A_n1">n1</code></td>
<td>
<p>the major allele heterozygotes</p>
</td></tr>
<tr><td><code id="close_to_normal_+3A_n2">n2</code></td>
<td>
<p>the minor allele zygotes</p>
</td></tr>
<tr><td><code id="close_to_normal_+3A_pa">pa</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
<tr><td><code id="close_to_normal_+3A_pb">pb</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
<tr><td><code id="close_to_normal_+3A_mua">mua</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="close_to_normal_+3A_mub">mub</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="close_to_normal_+3A_siga">siga</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="close_to_normal_+3A_sigb">sigb</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="close_to_normal_+3A_beta">beta</code></td>
<td>
<p>the beta from the regression being tested</p>
</td></tr>
<tr><td><code id="close_to_normal_+3A_nearnorm">nearnorm</code></td>
<td>
<p>must be in log base 10 format, with default value set to -5</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list(skewbeta = skewbeta, kurtbeta = kurtbeta, sigbeta = sigbeta, skipiter = skipiter)
</p>

<hr>
<h2 id='gwrpv'>Genome-Wide Regression P-Value (gwrpv) in R</h2><span id='topic+gwrpv'></span>

<h3>Description</h3>

<p>Computes the sample probability value (p-value) for the estimated coefficient
from a standard genome-wide univariate regression.
It computes the exact finite-sample p-value under the assumption that
the measured phenotype (the dependent variable in the regression)
has a known Bernoulli-normal mixture distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gwrpv(beta, n0, n1, n2, mua, siga, mub, sigb, pa, pb, logdelta = -16,
  lognearnorm = -5, logtopsum = 8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gwrpv_+3A_beta">beta</code></td>
<td>
<p>the beta being tested</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_n0">n0</code></td>
<td>
<p>number of major allele homozygotes</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_n1">n1</code></td>
<td>
<p>number of major allele heterozygotes</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_n2">n2</code></td>
<td>
<p>number of minor allele zygotes</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_mua">mua</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_siga">siga</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_mub">mub</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_sigb">sigb</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_pa">pa</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_pb">pb</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_logdelta">logdelta</code></td>
<td>
<p>must be in log base 10 format, with default value set to -16</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_lognearnorm">lognearnorm</code></td>
<td>
<p>must be in log base 10 format, with default value set to -5</p>
</td></tr>
<tr><td><code id="gwrpv_+3A_logtopsum">logtopsum</code></td>
<td>
<p>must be in log base 10 format, with default value set to 8</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gwrpv returns a list containing:
</p>

<dl>
<dt>$pvalue</dt><dd><p>p-value of a two-sided hypothesis test for a true coefficient of zero</p>
</dd>
<dt>$skew</dt><dd><p>skewness</p>
</dd>
<dt>$kurt</dt><dd><p>kurtosis of the coefficient estimate under assumed model</p>
</dd>
<dt>$skiptype</dt><dd><p>type of trimming/skip which took place (zero means no trimming)</p>
</dd>
<dt>$totnobs</dt><dd><p>total number of observations</p>
</dd>
<dt>$loopruns</dt><dd><p>number of sums in the main computation for each regression case</p>
</dd>
</dl>

<p>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>beta &lt;- 6.05879
n0 &lt;- 499
n1 &lt;- 1
n2 &lt;- 0
mua &lt;- 13.87226
siga &lt;- 2.58807
mub &lt;- 4.62829
sigb &lt;- 2.51803
pa &lt;- 0.96544
pb &lt;- 0.03456    # alternatively: pb &lt;- 1.0 - pa
gwrpv(beta,n0,n1,n2,mua,siga,mub,sigb,pa,pb)

# note default values have been used for the trim parameters above
# in the following example we explicitly set the trim parameters
#
g &lt;- gwrpv(beta,n0,n1,n2,mua,siga,mub,sigb,pa,pb,logdelta=-16,lognearnorm=-5,logtopsum=8)
g$pvalue
</code></pre>

<hr>
<h2 id='gwrpv_batch'>Batch computation of a list of pvalues of GWA regression beta statistics using a bernoulli-normal mixture distribution</h2><span id='topic+gwrpv_batch'></span>

<h3>Description</h3>

<p>Batch computation of a list of pvalues of GWA regression beta statistics using a bernoulli-normal mixture distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gwrpv_batch(regresults, mua, siga, mub, sigb, pa, pb, logdelta = -16,
  lognearnorm = -5, logtopsum = 8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gwrpv_batch_+3A_regresults">regresults</code></td>
<td>
<p>a list of four lists.
</p>

<dl>
<dt>$beta</dt><dd><p>the list of betas being tested</p>
</dd>
<dt>$n0</dt><dd><p>the list of major allele homozygotes</p>
</dd>
<dt>$n1</dt><dd><p>the list of major allele heterozygotes</p>
</dd>
<dt>$n2</dt><dd><p>the list of minor allele zygotes</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="gwrpv_batch_+3A_mua">mua</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="gwrpv_batch_+3A_siga">siga</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="gwrpv_batch_+3A_mub">mub</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="gwrpv_batch_+3A_sigb">sigb</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="gwrpv_batch_+3A_pa">pa</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
<tr><td><code id="gwrpv_batch_+3A_pb">pb</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
<tr><td><code id="gwrpv_batch_+3A_logdelta">logdelta</code></td>
<td>
<p>must be in log base 10 format, with default value set to -16</p>
</td></tr>
<tr><td><code id="gwrpv_batch_+3A_lognearnorm">lognearnorm</code></td>
<td>
<p>must be in log base 10 format, with default value set to -5</p>
</td></tr>
<tr><td><code id="gwrpv_batch_+3A_logtopsum">logtopsum</code></td>
<td>
<p>must be in log base 10 format, with default value set to 8</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gwrpv_batch returns a list of lists containing the lists:
</p>

<dl>
<dt>$pvalue</dt><dd><p>p-value of a two-sided hypothesis test for a true coefficient of zero</p>
</dd>
<dt>$skew</dt><dd><p>skewness</p>
</dd>
<dt>$kurt</dt><dd><p>kurtosis of the coefficient estimate under assumed model</p>
</dd>
<dt>$skiptype</dt><dd><p>type of trimming/skip which took place (zero means no trimming)</p>
</dd>
<dt>$totnobs</dt><dd><p>total number of observations</p>
</dd>
<dt>$loopruns</dt><dd><p>number of sums in the main computation for each regression case</p>
</dd>
</dl>

<p>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>beta &lt;- c(6.05879, -6.05879, 2.72055, -2.72055, 1.93347,
         -1.93347, 0.88288, -0.88288, 4.28421, -4.28421)
n0 &lt;- c(499, 499, 495, 495, 490, 490, 451, 451, 998, 998)
n1 &lt;- c(1, 1, 5, 5, 10, 10, 48, 48, 2, 2)
n2 &lt;- c(0, 0, 0, 0, 0, 0, 1, 1, 0, 0)
myregresults &lt;- list(beta = beta, n0 = n0, n1 = n1, n2 = n2)
mua &lt;- 13.87226
siga &lt;- 2.58807
mub &lt;- 4.62829
sigb &lt;- 2.51803
pa &lt;- 0.96544
pb &lt;- 1.0 - pa
gwrpv_batch(myregresults,mua,siga,mub,sigb,pa,pb)
# store results in a user-defined variable g
g &lt;- gwrpv_batch(myregresults,mua,siga,mub,sigb,pa,pb,logdelta=-16,lognearnorm=-4,logtopsum=8)
g$pvalue
</code></pre>

<hr>
<h2 id='gwrpvr'>gwrpvr: A package for calculating Genome-Wide Regression P-Values (gwrpv) in R</h2><span id='topic+gwrpvr'></span><span id='topic+gwrpvr-package'></span>

<h3>Description</h3>

<p>Computes the sample probability value (p-value) for the estimated coefficient
from a standard genome-wide univariate regression.
It computes the exact finite-sample p-value under the assumption that
the measured phenotype (the dependent variable in the regression)
has a known Bernoulli-normal mixture distribution.
</p>


<h3>Details</h3>

<p>The gwrpvr package provides two functions:
gwrpv and gwrpv_batch.
</p>

<hr>
<h2 id='highlow'>highlow()</h2><span id='topic+highlow'></span>

<h3>Description</h3>

<p>If possible, trim the upper and lower bounds
</p>


<h3>Usage</h3>

<pre><code class='language-R'>highlow(downtrim, n, pa, pb)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="highlow_+3A_downtrim">downtrim</code></td>
<td>
<p>lower bound</p>
</td></tr>
<tr><td><code id="highlow_+3A_n">n</code></td>
<td>
<p>upper bound</p>
</td></tr>
<tr><td><code id="highlow_+3A_pa">pa</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
<tr><td><code id="highlow_+3A_pb">pb</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>c(lhigh, llow))  # return the new upper and lower bounds
</p>

<hr>
<h2 id='loop_calc_pvalue'>loop_calc_pvalue()</h2><span id='topic+loop_calc_pvalue'></span>

<h3>Description</h3>

<p>calls calc_pvalue()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loop_calc_pvalue(lowone, highone, lowtwo, hightwo, lowthree, highthree, n0a,
  n1a, n2a, n0, n1, n2, pa, pb, x, mua, mub, sumsqx, siga, sigb, vary, beta,
  skipiter, pvalue)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loop_calc_pvalue_+3A_lowone">lowone</code></td>
<td>
<p>lower bound outer loop</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_highone">highone</code></td>
<td>
<p>upper bound outer loop</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_lowtwo">lowtwo</code></td>
<td>
<p>lower bound middle loop</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_hightwo">hightwo</code></td>
<td>
<p>upper bound middle loop</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_lowthree">lowthree</code></td>
<td>
<p>lower bound inner loop</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_highthree">highthree</code></td>
<td>
<p>upper bound inner loop</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_n0a">n0a</code></td>
<td>
<p>outer loop index</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_n1a">n1a</code></td>
<td>
<p>middle loop index</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_n2a">n2a</code></td>
<td>
<p>inner loop index</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_n0">n0</code></td>
<td>
<p>the major allele homozygotes</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_n1">n1</code></td>
<td>
<p>the major allele heterozygotes</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_n2">n2</code></td>
<td>
<p>the minor allele zygotes</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_pa">pa</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_pb">pb</code></td>
<td>
<p>parameter of the mixture distribution, a real number between zero and one with pa+pb=1</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_x">x</code></td>
<td>
<p>a zero mean explanatory variable from the SNP data set</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_mua">mua</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_mub">mub</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_sumsqx">sumsqx</code></td>
<td>
<p>sum of the squares of x</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_siga">siga</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_sigb">sigb</code></td>
<td>
<p>parameter of the mixture distribution, can be any real number</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_vary">vary</code></td>
<td>
<p>vary &lt;- pa*(mua^2+siga^2)+pb*(mub^2+sigb^2)-(pa*mua+pb*mub)^2</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_beta">beta</code></td>
<td>
<p>the beta from the regression being tested</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_skipiter">skipiter</code></td>
<td>
<p>flag to determine if we can skip some calculations</p>
</td></tr>
<tr><td><code id="loop_calc_pvalue_+3A_pvalue">pvalue</code></td>
<td>
<p>the input pvalue prior to calculating new improved pvalue</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pvalue
</p>

<hr>
<h2 id='regresults'>regresults: sample data</h2><span id='topic+regresults'></span>

<h3>Description</h3>

<p>A sample dataset of input regression results based on machine-level accurate cumulative normal values.
Rather than just typing in a few digits of the 2.5
the norminverse function in RATS was used to create sample-case betas which are exact
</p>


<h3>Format</h3>

<p>csv format file with 4 variables (beta, n0, n1, n2) and 120 rows</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
