<!DOCTYPE html><html><head><title>Help for package GxEprs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GxEprs}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Bcov_discovery'><p>Covariate data file of the discovery dataset when the outcome is binary.</p>
This contains covariate information of the individuals in the discovery dataset following confounders.</a></li>
<li><a href='#Bcov_target'><p>Covariate data file of the target dataset when the outcome is binary.</p>
This contains covariate information of the individuals in the target dataset following confounders.</a></li>
<li><a href='#Bphe_discovery'><p>Phenotype data file of the discovery dataset when the outcome is binary.</p>
This contains phenotype information of the individuals in the discovery dataset.</a></li>
<li><a href='#Bphe_target'><p>Phenotype data file of the target dataset when the outcome is binary.</p>
This contains phenotype information of the individuals in the target dataset.</a></li>
<li><a href='#DummyData.bim'><p>PLINK .bim file</p></a></li>
<li><a href='#DummyData.fam'><p>PLINK .fam file</p></a></li>
<li><a href='#DummyData.map'><p>PLINK .map file</p></a></li>
<li><a href='#DummyData.ped'><p>PLINK .ped file</p></a></li>
<li><a href='#GWAS_binary'><p>GWAS_binary function</p>
This function performs GWAS using plink2 and outputs the GWAS summary statistics with additive SNP effects. Users may save the output in a user-specified file (see example).</a></li>
<li><a href='#GWAS_quantitative'><p>GWAS_quantitative function</p>
This function performs GWAS using plink2 and outputs the GWAS summary statistics with additive SNP effects. Users may save the output in a user-specified file (see example).</a></li>
<li><a href='#GWEIS_binary'><p>GWEIS_binary function</p>
This function performs GWEIS using plink2 and outputs the GWEIS summary statistics with additive SNP effects and interaction SNP effects. Users may save the outputs in separate user-specified files (see examples).</a></li>
<li><a href='#GWEIS_quantitative'><p>GWEIS_quantitative function</p>
This function performs GWEIS using plink2 and outputs the GWEIS summary statistics with additive SNP effects and interaction SNP effects separately. It is recommended to save the outputs in separate user-specified files (see examples).</a></li>
<li><a href='#PRS_binary'><p>PRS_binary function</p>
This function uses plink2 and outputs Polygenic Risk Scores (PRSs) of all the individuals, using pre-generated GWAS and/or GWEIS summary statistics. Note that the input used in this function can be generated by using GWAS_binary and/or GWEIS_binary functions. Users may save the output in a user-specified file (see examples).</a></li>
<li><a href='#PRS_quantitative'><p>PRS_quantitative function</p>
This function uses plink2 and outputs Polygenic Risk Scores (PRSs) of all the individuals, using pre-generated GWAS and/or GWEIS summary statistics. Note that the input used in this function can be generated by using GWAS_quantitative and/or GWEIS_quantitative functions. Users may save the output in a user-specified file (see examples).</a></li>
<li><a href='#Qcov_discovery'><p>Covariate data file of the discovery dataset when the outcome is quantitative.</p>
This contains covariate information of the individuals in the discovery dataset following confounders.</a></li>
<li><a href='#Qcov_target'><p>Covariate data file of the target dataset when the outcome is quantitative.</p>
This contains covariate information of the individuals in the target dataset following confounders.</a></li>
<li><a href='#Qphe_discovery'><p>Phenotype data file of the discovery dataset when the outcome is quantitative.</p>
This contains phenotype information of the individuals in the discovery dataset.</a></li>
<li><a href='#Qphe_target'><p>Phenotype data file of the target dataset when the outcome is quantitative.</p>
This contains phenotype information of the individuals in the target dataset.</a></li>
<li><a href='#summary_permuted_binary'><p>summary_permuted_binary function</p>
This function outputs the p value of permuted model in the target dataset, using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_quantitative function. It is recommended to run this function, if you choose to fit 'PRS_gxe x E' interaction component (i.e. novel proposed model, Model 5) when generating risk scores. If the 'PRS_gxe x E' term is significant in Model 5, and insignificant in Model 5* (permuted p value), consider that the 'PRS_gxe x E' interaction component is actually insignificant (always give priority to the p value obtained from the permuted model).</a></li>
<li><a href='#summary_permuted_quantitative'><p>summary_permuted_quantitative function</p>
This function outputs the p value of permuted model in the target dataset, using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_quantitative functions. It is recommended to run this function, if you choose to fit 'PRS_gxe x E' interaction component (i.e. novel proposed model, Model 4) when generating risk scores. If the 'PRS_gxe x E' term is significant in Model 4, and insignificant in Model 4* (permuted p value), consider that the 'PRS_gxe x E' interaction component is actually insignificant (always give priority to the p value obtained from the permuted model).</a></li>
<li><a href='#summary_regular_binary'><p>summary_regular_binary function</p>
This function outputs the summary of regular model and final risk score values of each individual in the target dataset using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_binary function.</a></li>
<li><a href='#summary_regular_quantitative'><p>summary_regular_quantitative function</p>
This function outputs the summary of regular model and final risk score values of each individual in the target dataset using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_quantitative function.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Genotype-by-Environment Interaction in Polygenic Score Models</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A novel PRS model is introduced to enhance the prediction accuracy by utilising GxE effects. This package performs Genome Wide Association Studies (GWAS) and Genome Wide Environment Interaction Studies (GWEIS) using a discovery dataset. The package has the ability to obtain polygenic risk scores (PRSs) for a target sample. Finally it predicts the risk values of each individual in the target sample. Users have the choice of using existing models (Li et al., 2015) &lt;<a href="https://doi.org/10.1093%2Fannonc%2Fmdu565">doi:10.1093/annonc/mdu565</a>&gt;, (Pandis et al., 2013) &lt;<a href="https://doi.org/10.1093%2Fejo%2Fcjt054">doi:10.1093/ejo/cjt054</a>&gt;, (Peyrot et al., 2018) &lt;<a href="https://doi.org/10.1016%2Fj.biopsych.2017.09.009">doi:10.1016/j.biopsych.2017.09.009</a>&gt; and (Song et al., 2022) &lt;<a href="https://doi.org/10.1038%2Fs41467-022-32407-9">doi:10.1038/s41467-022-32407-9</a>&gt;, as well as newly proposed models for genomic risk prediction (refer to the URL for more details).</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/DoviniJ/GxEprs">https://github.com/DoviniJ/GxEprs</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-03 03:12:25 UTC; dj8715</td>
</tr>
<tr>
<td>Author:</td>
<td>Dovini Jayasinghe [aut, cre, cph],
  Hong Lee [aut, cph],
  Moksedul Momin [aut, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dovini Jayasinghe &lt;dovini.jayasinghe@mymail.unisa.edu.au&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-03 06:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='Bcov_discovery'>Covariate data file of the discovery dataset when the outcome is binary.
This contains covariate information of the individuals in the discovery dataset following confounders.</h2><span id='topic+Bcov_discovery'></span>

<h3>Description</h3>

<p>Covariate data file of the discovery dataset when the outcome is binary.
This contains covariate information of the individuals in the discovery dataset following confounders.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Bcov_discovery
</code></pre>


<h3>Format</h3>

<p>A dataframe with 800 rows and 18 columns
</p>

<dl>
<dt>Column 1</dt><dd><p>Family ID</p>
</dd>
<dt>Column 2</dt><dd><p>Individual ID</p>
</dd>
<dt>Column 3</dt><dd><p>Standardized covariate</p>
</dd>
<dt>Column 4</dt><dd><p>Square of the standardized covariate</p>
</dd>
<dt>Column 5</dt><dd><p>Confounder 1</p>
</dd>
<dt>Column 6</dt><dd><p>Confounder 2</p>
</dd>
<dt>Column 7</dt><dd><p>Confounder 3</p>
</dd>
<dt>Column 8</dt><dd><p>Confounder 4</p>
</dd>
<dt>Column 9</dt><dd><p>Confounder 5</p>
</dd>
<dt>Column 10</dt><dd><p>Confounder 6</p>
</dd>
<dt>Column 11</dt><dd><p>Confounder 7</p>
</dd>
<dt>Column 12</dt><dd><p>Confounder 8</p>
</dd>
<dt>Column 13</dt><dd><p>Confounder 9</p>
</dd>
<dt>Column 14</dt><dd><p>Confounder 10</p>
</dd>
<dt>Column 15</dt><dd><p>Confounder 11</p>
</dd>
<dt>Column 16</dt><dd><p>Confounder 12</p>
</dd>
<dt>Column 17</dt><dd><p>Confounder 13</p>
</dd>
<dt>Column 18</dt><dd><p>Confounder 14</p>
</dd>
</dl>


<hr>
<h2 id='Bcov_target'>Covariate data file of the target dataset when the outcome is binary.
This contains covariate information of the individuals in the target dataset following confounders.</h2><span id='topic+Bcov_target'></span>

<h3>Description</h3>

<p>Covariate data file of the target dataset when the outcome is binary.
This contains covariate information of the individuals in the target dataset following confounders.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Bcov_target
</code></pre>


<h3>Format</h3>

<p>A dataframe with 200 rows and 18 columns
</p>

<dl>
<dt>Column 1</dt><dd><p>Family ID</p>
</dd>
<dt>Column 2</dt><dd><p>Individual ID</p>
</dd>
<dt>Column 3</dt><dd><p>Standardized covariate</p>
</dd>
<dt>Column 4</dt><dd><p>Square of the standardized covariate</p>
</dd>
<dt>Column 5</dt><dd><p>Confounder 1</p>
</dd>
<dt>Column 6</dt><dd><p>Confounder 2</p>
</dd>
<dt>Column 7</dt><dd><p>Confounder 3</p>
</dd>
<dt>Column 8</dt><dd><p>Confounder 4</p>
</dd>
<dt>Column 9</dt><dd><p>Confounder 5</p>
</dd>
<dt>Column 10</dt><dd><p>Confounder 6</p>
</dd>
<dt>Column 11</dt><dd><p>Confounder 7</p>
</dd>
<dt>Column 12</dt><dd><p>Confounder 8</p>
</dd>
<dt>Column 13</dt><dd><p>Confounder 9</p>
</dd>
<dt>Column 14</dt><dd><p>Confounder 10</p>
</dd>
<dt>Column 15</dt><dd><p>Confounder 11</p>
</dd>
<dt>Column 16</dt><dd><p>Confounder 12</p>
</dd>
<dt>Column 17</dt><dd><p>Confounder 13</p>
</dd>
<dt>Column 18</dt><dd><p>Confounder 14</p>
</dd>
</dl>


<hr>
<h2 id='Bphe_discovery'>Phenotype data file of the discovery dataset when the outcome is binary.
This contains phenotype information of the individuals in the discovery dataset.</h2><span id='topic+Bphe_discovery'></span>

<h3>Description</h3>

<p>Phenotype data file of the discovery dataset when the outcome is binary.
This contains phenotype information of the individuals in the discovery dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Bphe_discovery
</code></pre>


<h3>Format</h3>

<p>A dataframe with 800 rows and 3 columns
</p>

<dl>
<dt>Column 1</dt><dd><p>Family ID</p>
</dd>
<dt>Column 2</dt><dd><p>Individual ID</p>
</dd>
<dt>Column 3</dt><dd><p>Phenotype (1=controls, 2=cases)</p>
</dd>
</dl>


<hr>
<h2 id='Bphe_target'>Phenotype data file of the target dataset when the outcome is binary.
This contains phenotype information of the individuals in the target dataset.</h2><span id='topic+Bphe_target'></span>

<h3>Description</h3>

<p>Phenotype data file of the target dataset when the outcome is binary.
This contains phenotype information of the individuals in the target dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Bphe_target
</code></pre>


<h3>Format</h3>

<p>A dataframe with 200 rows and 3 columns
</p>

<dl>
<dt>Column 1</dt><dd><p>Family ID</p>
</dd>
<dt>Column 2</dt><dd><p>Individual ID</p>
</dd>
<dt>Column 3</dt><dd><p>Phenotype (0=controls, 1=cases)</p>
</dd>
</dl>


<hr>
<h2 id='DummyData.bim'>PLINK .bim file</h2><span id='topic+DummyData.bim'></span>

<h3>Description</h3>

<p>PLINK .bim file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DummyData.bim
</code></pre>


<h3>Format</h3>

<p>This follows PLINK general format
</p>

<dl>
<dt>Column 1</dt><dd><p>Chromosome ID</p>
</dd>
<dt>Column 2</dt><dd><p>SNP ID</p>
</dd>
<dt>Column 3</dt><dd><p>Position of centimorgans</p>
</dd>
<dt>Column 4</dt><dd><p>Base-pair coordinate</p>
</dd>
<dt>Column 5</dt><dd><p>Minor Allele</p>
</dd>
<dt>Column 6</dt><dd><p>Reference Allele</p>
</dd>
</dl>


<hr>
<h2 id='DummyData.fam'>PLINK .fam file</h2><span id='topic+DummyData.fam'></span>

<h3>Description</h3>

<p>PLINK .fam file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DummyData.fam
</code></pre>


<h3>Format</h3>

<p>This follows PLINK general format
</p>

<dl>
<dt>Column 1</dt><dd><p>Family ID</p>
</dd>
<dt>Column 2</dt><dd><p>Individual ID</p>
</dd>
<dt>Column 3</dt><dd><p>Father's ID</p>
</dd>
<dt>Column 4</dt><dd><p>Mother's ID</p>
</dd>
<dt>Column 5</dt><dd><p>Sex</p>
</dd>
<dt>Column 6</dt><dd><p>Phenotype value</p>
</dd>
</dl>


<hr>
<h2 id='DummyData.map'>PLINK .map file</h2><span id='topic+DummyData.map'></span>

<h3>Description</h3>

<p>PLINK .map file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DummyData.map
</code></pre>


<h3>Format</h3>

<p>This follows PLINK general format
</p>

<hr>
<h2 id='DummyData.ped'>PLINK .ped file</h2><span id='topic+DummyData.ped'></span>

<h3>Description</h3>

<p>PLINK .ped file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DummyData.ped
</code></pre>


<h3>Format</h3>

<p>This follows PLINK general format
</p>

<hr>
<h2 id='GWAS_binary'>GWAS_binary function
This function performs GWAS using plink2 and outputs the GWAS summary statistics with additive SNP effects. Users may save the output in a user-specified file (see example).</h2><span id='topic+GWAS_binary'></span>

<h3>Description</h3>

<p>GWAS_binary function
This function performs GWAS using plink2 and outputs the GWAS summary statistics with additive SNP effects. Users may save the output in a user-specified file (see example).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GWAS_binary(plink_path, b_file, Bphe_discovery, Bcov_discovery, thread = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GWAS_binary_+3A_plink_path">plink_path</code></td>
<td>
<p>Path to the PLINK executable application</p>
</td></tr>
<tr><td><code id="GWAS_binary_+3A_b_file">b_file</code></td>
<td>
<p>Prefix of the binary files, where all .fam, .bed and .bim files have a common prefix</p>
</td></tr>
<tr><td><code id="GWAS_binary_+3A_bphe_discovery">Bphe_discovery</code></td>
<td>
<p>Name (with file extension) of the phenotype file containing family ID, individual ID and phenotype of the discovery dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="GWAS_binary_+3A_bcov_discovery">Bcov_discovery</code></td>
<td>
<p>Name (with file extension) of the covariate file containing family ID, individual ID, standardized covariate, square of standardized covariate, and/or confounders of the discovery dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="GWAS_binary_+3A_thread">thread</code></td>
<td>
<p>Number of threads used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will perform GWAS and output
</p>
<table>
<tr><td><code>B_out.trd.sum</code></td>
<td>
<p>GWAS summary statistics with additive SNP effects</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
x &lt;- GWAS_binary(plink_path, DummyData, Bphe_discovery, Bcov_discovery, 
thread = 20)
sink("B_out.trd.sum") #to create a file in the working directory
write.table(x[c("ID", "A1", "OR")], sep = " ", 
row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output
head(x) #to obtain the head of GWAS summary statistics of additive SNP effects
x$CHROM #to extract the chromosome number 
x$POS #to extract the base pair position 
x$ID #to extract the SNP ID 
x$REF #to extract the reference allele 
x$ALT #to extract the alternate allele 
x$A1 #to extract the minor allele 
x$OBS_CT #to extract the number of allele observations 
x$OR #to extract the odds ratios of the SNP effects
x$LOG_OR_SE #to extract the standard errors of log odds 
x$Z_STAT #to extract the test statistics 
x$P #to extract the p values 

## End(Not run)
</code></pre>

<hr>
<h2 id='GWAS_quantitative'>GWAS_quantitative function
This function performs GWAS using plink2 and outputs the GWAS summary statistics with additive SNP effects. Users may save the output in a user-specified file (see example).</h2><span id='topic+GWAS_quantitative'></span>

<h3>Description</h3>

<p>GWAS_quantitative function
This function performs GWAS using plink2 and outputs the GWAS summary statistics with additive SNP effects. Users may save the output in a user-specified file (see example).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GWAS_quantitative(
  plink_path,
  b_file,
  Qphe_discovery,
  Qcov_discovery,
  thread = 20
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GWAS_quantitative_+3A_plink_path">plink_path</code></td>
<td>
<p>Path to the PLINK executable application</p>
</td></tr>
<tr><td><code id="GWAS_quantitative_+3A_b_file">b_file</code></td>
<td>
<p>Prefix of the binary files, where all .fam, .bed and .bim files have a common prefix</p>
</td></tr>
<tr><td><code id="GWAS_quantitative_+3A_qphe_discovery">Qphe_discovery</code></td>
<td>
<p>Name (with file extension) of the phenotype file containing family ID, individual ID and phenotype of the discovery dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="GWAS_quantitative_+3A_qcov_discovery">Qcov_discovery</code></td>
<td>
<p>Name (with file extension) of the covariate file containing family ID, individual ID, standardized covariate, square of standardized covariate, and/or confounders of the discovery dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="GWAS_quantitative_+3A_thread">thread</code></td>
<td>
<p>Number of threads used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will perform GWAS and output
</p>
<table>
<tr><td><code>Q_out.trd.sum</code></td>
<td>
<p>GWAS summary statistics with additive SNP effects</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
x &lt;- GWAS_quantitative(plink_path, DummyData, Qphe_discovery, Qcov_discovery, 
thread = 20)
sink("Q_out.trd.sum") #to create a file in the working directory
write.table(x[c("ID", "A1", "BETA")], sep = " ", 
row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output
head(x) #to obtain the head of GWAS summary statistics of additive SNP effects
x$CHROM #to extract the chromosome number 
x$POS #to extract the base pair position 
x$ID #to extract the SNP ID 
x$REF #to extract the reference allele 
x$ALT #to extract the alternate allele 
x$A1 #to extract the minor allele 
x$OBS_CT #to extract the number of allele observations 
x$BETA #to extract the SNP effects
x$SE #to extract the standard errors of the SNP effects
x$T_STAT #to extract the test statistics 
x$P #to extract the p values 

## End(Not run)
</code></pre>

<hr>
<h2 id='GWEIS_binary'>GWEIS_binary function
This function performs GWEIS using plink2 and outputs the GWEIS summary statistics with additive SNP effects and interaction SNP effects. Users may save the outputs in separate user-specified files (see examples).</h2><span id='topic+GWEIS_binary'></span>

<h3>Description</h3>

<p>GWEIS_binary function
This function performs GWEIS using plink2 and outputs the GWEIS summary statistics with additive SNP effects and interaction SNP effects. Users may save the outputs in separate user-specified files (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GWEIS_binary(plink_path, b_file, Bphe_discovery, Bcov_discovery, thread = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GWEIS_binary_+3A_plink_path">plink_path</code></td>
<td>
<p>Path to the PLINK executable application</p>
</td></tr>
<tr><td><code id="GWEIS_binary_+3A_b_file">b_file</code></td>
<td>
<p>Prefix of the binary files, where all .fam, .bed and .bim files have a common prefix</p>
</td></tr>
<tr><td><code id="GWEIS_binary_+3A_bphe_discovery">Bphe_discovery</code></td>
<td>
<p>Phenotype file containing family ID, individual ID and phenotype of the discovery dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="GWEIS_binary_+3A_bcov_discovery">Bcov_discovery</code></td>
<td>
<p>Covariate file containing family ID, individual ID, standardized covariate, square of standardized covariate, and/or confounders of the discovery dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="GWEIS_binary_+3A_thread">thread</code></td>
<td>
<p>Number of threads used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will perform GWEIS and output
</p>
<table>
<tr><td><code>B_out.sum</code></td>
<td>
<p>GWEIS summary statistics with additive and interaction SNP effects</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  
x &lt;- GWEIS_binary(plink_path, DummyData, Bphe_discovery, Bcov_discovery, 
thread = 20)
sink("B_out.add.sum") #to create a file in the working directory
write.table(x[c("ID", "A1", "ADD_OR")], sep = " ", 
row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output
sink("B_out.gxe.sum") #to create a file in the working directory
write.table(x[c("ID", "A1", "INTERACTION_OR")], sep = " ", 
row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output
head(x) #to extract the head of all columns in GWEIS summary  
#statistics of additive and interaction SNP effects 
x$CHROM #to extract the chromosome number 
x$POS #to extract the base pair position
x$ID #to extract the SNP ID
x$REF #to extract the reference allele
x$ALT #to extract the alternate allele 
x$A1 #to extract the minor allele
x$OBS_CT #to extract the number of allele observations 
x$ADD_OR #to extract the odds ratios of additive SNP effects
x$ADD_LOG_OR_SE #to extract the standard errors of log  
#odds of additive SNP effects
x$ADD_Z_STAT #to extract the test statistics of additive 
#SNP effects
x$ADD_P #to extract the p values of additive SNP effects
x$INTERACTION_OR #to extract the odds ratios of  
#interaction SNP effects
x$INTERACTION_LOG_OR_SE #to extract the standard errors of  
#log odds of interaction SNP effects
x$INTERACTION_Z_STAT #to extract the test statistics of  
#interaction SNP effects
x$INTERACTION_P #to extract the p values of interaction 
#SNP effects

## End(Not run)
</code></pre>

<hr>
<h2 id='GWEIS_quantitative'>GWEIS_quantitative function
This function performs GWEIS using plink2 and outputs the GWEIS summary statistics with additive SNP effects and interaction SNP effects separately. It is recommended to save the outputs in separate user-specified files (see examples).</h2><span id='topic+GWEIS_quantitative'></span>

<h3>Description</h3>

<p>GWEIS_quantitative function
This function performs GWEIS using plink2 and outputs the GWEIS summary statistics with additive SNP effects and interaction SNP effects separately. It is recommended to save the outputs in separate user-specified files (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GWEIS_quantitative(
  plink_path,
  b_file,
  Qphe_discovery,
  Qcov_discovery,
  thread = 20
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GWEIS_quantitative_+3A_plink_path">plink_path</code></td>
<td>
<p>Path to the PLINK executable application</p>
</td></tr>
<tr><td><code id="GWEIS_quantitative_+3A_b_file">b_file</code></td>
<td>
<p>Prefix of the binary files, where all .fam, .bed and .bim files have a common prefix</p>
</td></tr>
<tr><td><code id="GWEIS_quantitative_+3A_qphe_discovery">Qphe_discovery</code></td>
<td>
<p>Phenotype file containing family ID, individual ID and phenotype of the discovery dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="GWEIS_quantitative_+3A_qcov_discovery">Qcov_discovery</code></td>
<td>
<p>Covariate file containing family ID, individual ID, standardized covariate, square of standardized covariate, and/or confounders of the discovery dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="GWEIS_quantitative_+3A_thread">thread</code></td>
<td>
<p>Number of threads used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will perform GWEIS and output
</p>
<table>
<tr><td><code>Q_out.sum</code></td>
<td>
<p>GWEIS summary statistics with additive and interaction SNP effects</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  
x &lt;- GWEIS_quantitative (plink_path, DummyData, Qphe_discovery, Qcov_discovery, 
thread = 20)
sink("Q_out.add.sum") #to create a file in the working directory
write.table(x[c("ID", "A1", "ADD_BETA")], sep = " ", 
row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output
sink("Q_out.gxe.sum") #to create a file in the working directory
write.table(x[c("ID", "A1", "INTERACTION_BETA")], sep = " ", 
row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output
head(x) #to extract the head of all columns in GWEIS summary  
#statistics of additive and interaction SNP effects 
x$CHROM #to extract the chromosome number 
x$POS #to extract the base pair position
x$ID #to extract the SNP ID
x$REF #to extract the reference allele
x$ALT #to extract the alternate allele 
x$A1 #to extract the minor allele
x$OBS_CT #to extract the number of allele observations 
x$ADD_BETA #to extract the additive SNP effects
x$ADD_SE #to extract the standard errors of the 
#additive SNP effects
x$ADD_T_STAT #to extract the test statistics of additive 
#SNP effects
x$ADD_P #to extract the p values of additive SNP effects
x$INTERACTION_BETA #to extract the interaction SNP effects
x$INTERACTION_SE #to extract the standard errors of the 
#interaction SNP effects
x$INTERACTION_T_STAT #to extract the test statistics of 
#interaction SNP effects
x$INTERACTION_P #to extract the p values of interaction 
#SNP effects

## End(Not run)
</code></pre>

<hr>
<h2 id='PRS_binary'>PRS_binary function
This function uses plink2 and outputs Polygenic Risk Scores (PRSs) of all the individuals, using pre-generated GWAS and/or GWEIS summary statistics. Note that the input used in this function can be generated by using GWAS_binary and/or GWEIS_binary functions. Users may save the output in a user-specified file (see examples).</h2><span id='topic+PRS_binary'></span>

<h3>Description</h3>

<p>PRS_binary function
This function uses plink2 and outputs Polygenic Risk Scores (PRSs) of all the individuals, using pre-generated GWAS and/or GWEIS summary statistics. Note that the input used in this function can be generated by using GWAS_binary and/or GWEIS_binary functions. Users may save the output in a user-specified file (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PRS_binary(plink_path, b_file, summary_input)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PRS_binary_+3A_plink_path">plink_path</code></td>
<td>
<p>Path to the PLINK executable application</p>
</td></tr>
<tr><td><code id="PRS_binary_+3A_b_file">b_file</code></td>
<td>
<p>Prefix of the binary files, where all .fam, .bed and .bim files have a common prefix</p>
</td></tr>
<tr><td><code id="PRS_binary_+3A_summary_input">summary_input</code></td>
<td>
<p>Pre-generated GWAS and/or GWEIS summary statistics</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will output
</p>
<table>
<tr><td><code>prs.sscore</code></td>
<td>
<p>PRSs for each individual</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  
a &lt;- GWAS_binary(plink_path, DummyData, Bphe_discovery, Bcov_discovery)
trd &lt;- a[c("ID", "A1", "OR")]
b &lt;- GWEIS_binary(plink_path, DummyData, Bphe_discovery, Bcov_discovery)
add &lt;- b[c("ID", "A1", "ADD_OR")]
gxe &lt;- b[c("ID", "A1", "INTERACTION_OR")]
x &lt;- PRS_binary(plink_path, DummyData, summary_input = trd)
sink("B_trd.sscore") #to create a file in the working directory
write.table(x, sep = " ", row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output
head(x) #to read the head of all columns in the output
x$FID #to extract the family ID's of full dataset
x$IID #to extract the individual ID's of full dataset 
x$PRS #to extract the polygenic risk scores of full dataset
y &lt;- PRS_binary(plink_path, DummyData, summary_input = add)
sink("B_add.sscore") #to create a file in the working directory
write.table(y, sep = " ", row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output
z &lt;- PRS_binary(plink_path, DummyData, summary_input = gxe)
sink("B_gxe.sscore") #to create a file in the working directory
write.table(z, sep = " ", row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output

## End(Not run)
</code></pre>

<hr>
<h2 id='PRS_quantitative'>PRS_quantitative function
This function uses plink2 and outputs Polygenic Risk Scores (PRSs) of all the individuals, using pre-generated GWAS and/or GWEIS summary statistics. Note that the input used in this function can be generated by using GWAS_quantitative and/or GWEIS_quantitative functions. Users may save the output in a user-specified file (see examples).</h2><span id='topic+PRS_quantitative'></span>

<h3>Description</h3>

<p>PRS_quantitative function
This function uses plink2 and outputs Polygenic Risk Scores (PRSs) of all the individuals, using pre-generated GWAS and/or GWEIS summary statistics. Note that the input used in this function can be generated by using GWAS_quantitative and/or GWEIS_quantitative functions. Users may save the output in a user-specified file (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PRS_quantitative(plink_path, b_file, summary_input)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PRS_quantitative_+3A_plink_path">plink_path</code></td>
<td>
<p>Path to the PLINK executable application</p>
</td></tr>
<tr><td><code id="PRS_quantitative_+3A_b_file">b_file</code></td>
<td>
<p>Prefix of the binary files, where all .fam, .bed and .bim files have a common prefix</p>
</td></tr>
<tr><td><code id="PRS_quantitative_+3A_summary_input">summary_input</code></td>
<td>
<p>Pre-generated GWAS and/or GWEIS summary statistics</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will output
</p>
<table>
<tr><td><code>prs.sscore</code></td>
<td>
<p>PRSs for each individual</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  
a &lt;- GWAS_quantitative(plink_path, DummyData, Qphe_discovery, Qcov_discovery)
trd &lt;- a[c("ID", "A1", "BETA")]
b &lt;- GWEIS_quantitative(plink_path, DummyData, Qphe_discovery, Qcov_discovery)
add &lt;- b[c("ID", "A1", "ADD_BETA")]
gxe &lt;- b[c("ID", "A1", "INTERACTION_BETA")]
x &lt;- PRS_quantitative(plink_path, DummyData, summary_input = trd)
sink("Q_trd.sscore") #to create a file in the working directory
write.table(x, sep = " ", row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output
head(x) #to read the head of all columns in the output
x$FID #to extract the family ID's of full dataset
x$IID #to extract the individual ID's of full dataset 
x$PRS #to extract the polygenic risk scores of full dataset
y &lt;- PRS_quantitative(plink_path, DummyData, summary_input = add)
sink("Q_add.sscore") #to create a file in the working directory
write.table(y, sep = " ", row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output
z &lt;- PRS_quantitative(plink_path, DummyData, summary_input = gxe)
sink("Q_gxe.sscore") #to create a file in the working directory
write.table(z, sep = " ", row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output

## End(Not run)
</code></pre>

<hr>
<h2 id='Qcov_discovery'>Covariate data file of the discovery dataset when the outcome is quantitative.
This contains covariate information of the individuals in the discovery dataset following confounders.</h2><span id='topic+Qcov_discovery'></span>

<h3>Description</h3>

<p>Covariate data file of the discovery dataset when the outcome is quantitative.
This contains covariate information of the individuals in the discovery dataset following confounders.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Qcov_discovery
</code></pre>


<h3>Format</h3>

<p>A dataframe with 800 rows and 18 columns
</p>

<dl>
<dt>Column 1</dt><dd><p>Family ID</p>
</dd>
<dt>Column 2</dt><dd><p>Individual ID</p>
</dd>
<dt>Column 3</dt><dd><p>Standardized covariate</p>
</dd>
<dt>Column 4</dt><dd><p>Square of the standardized covariate</p>
</dd>
<dt>Column 5</dt><dd><p>Confounder 1</p>
</dd>
<dt>Column 6</dt><dd><p>Confounder 2</p>
</dd>
<dt>Column 7</dt><dd><p>Confounder 3</p>
</dd>
<dt>Column 8</dt><dd><p>Confounder 4</p>
</dd>
<dt>Column 9</dt><dd><p>Confounder 5</p>
</dd>
<dt>Column 10</dt><dd><p>Confounder 6</p>
</dd>
<dt>Column 11</dt><dd><p>Confounder 7</p>
</dd>
<dt>Column 12</dt><dd><p>Confounder 8</p>
</dd>
<dt>Column 13</dt><dd><p>Confounder 9</p>
</dd>
<dt>Column 14</dt><dd><p>Confounder 10</p>
</dd>
<dt>Column 15</dt><dd><p>Confounder 11</p>
</dd>
<dt>Column 16</dt><dd><p>Confounder 12</p>
</dd>
<dt>Column 17</dt><dd><p>Confounder 13</p>
</dd>
<dt>Column 18</dt><dd><p>Confounder 14</p>
</dd>
</dl>


<hr>
<h2 id='Qcov_target'>Covariate data file of the target dataset when the outcome is quantitative.
This contains covariate information of the individuals in the target dataset following confounders.</h2><span id='topic+Qcov_target'></span>

<h3>Description</h3>

<p>Covariate data file of the target dataset when the outcome is quantitative.
This contains covariate information of the individuals in the target dataset following confounders.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Qcov_target
</code></pre>


<h3>Format</h3>

<p>A dataframe with 200 rows and 18 columns
</p>

<dl>
<dt>Column 1</dt><dd><p>Family ID</p>
</dd>
<dt>Column 2</dt><dd><p>Individual ID</p>
</dd>
<dt>Column 3</dt><dd><p>Standardized covariate</p>
</dd>
<dt>Column 4</dt><dd><p>Square of the standardized covariate</p>
</dd>
<dt>Column 5</dt><dd><p>Confounder 1</p>
</dd>
<dt>Column 6</dt><dd><p>Confounder 2</p>
</dd>
<dt>Column 7</dt><dd><p>Confounder 3</p>
</dd>
<dt>Column 8</dt><dd><p>Confounder 4</p>
</dd>
<dt>Column 9</dt><dd><p>Confounder 5</p>
</dd>
<dt>Column 10</dt><dd><p>Confounder 6</p>
</dd>
<dt>Column 11</dt><dd><p>Confounder 7</p>
</dd>
<dt>Column 12</dt><dd><p>Confounder 8</p>
</dd>
<dt>Column 13</dt><dd><p>Confounder 9</p>
</dd>
<dt>Column 14</dt><dd><p>Confounder 10</p>
</dd>
<dt>Column 15</dt><dd><p>Confounder 11</p>
</dd>
<dt>Column 16</dt><dd><p>Confounder 12</p>
</dd>
<dt>Column 17</dt><dd><p>Confounder 13</p>
</dd>
<dt>Column 18</dt><dd><p>Confounder 14</p>
</dd>
</dl>


<hr>
<h2 id='Qphe_discovery'>Phenotype data file of the discovery dataset when the outcome is quantitative.
This contains phenotype information of the individuals in the discovery dataset.</h2><span id='topic+Qphe_discovery'></span>

<h3>Description</h3>

<p>Phenotype data file of the discovery dataset when the outcome is quantitative.
This contains phenotype information of the individuals in the discovery dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Qphe_discovery
</code></pre>


<h3>Format</h3>

<p>A dataframe with 800 rows and 3 columns
</p>

<dl>
<dt>Column 1</dt><dd><p>Family ID</p>
</dd>
<dt>Column 2</dt><dd><p>Individual ID</p>
</dd>
<dt>Column 3</dt><dd><p>Phenotype</p>
</dd>
</dl>


<hr>
<h2 id='Qphe_target'>Phenotype data file of the target dataset when the outcome is quantitative.
This contains phenotype information of the individuals in the target dataset.</h2><span id='topic+Qphe_target'></span>

<h3>Description</h3>

<p>Phenotype data file of the target dataset when the outcome is quantitative.
This contains phenotype information of the individuals in the target dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Qphe_target
</code></pre>


<h3>Format</h3>

<p>A dataframe with 200 rows and 3 columns
</p>

<dl>
<dt>Column 1</dt><dd><p>Family ID</p>
</dd>
<dt>Column 2</dt><dd><p>Individual ID</p>
</dd>
<dt>Column 3</dt><dd><p>Phenotype</p>
</dd>
</dl>


<hr>
<h2 id='summary_permuted_binary'>summary_permuted_binary function
This function outputs the p value of permuted model in the target dataset, using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_quantitative function. It is recommended to run this function, if you choose to fit 'PRS_gxe x E' interaction component (i.e. novel proposed model, Model 5) when generating risk scores. If the 'PRS_gxe x E' term is significant in Model 5, and insignificant in Model 5* (permuted p value), consider that the 'PRS_gxe x E' interaction component is actually insignificant (always give priority to the p value obtained from the permuted model).</h2><span id='topic+summary_permuted_binary'></span>

<h3>Description</h3>

<p>summary_permuted_binary function
This function outputs the p value of permuted model in the target dataset, using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_quantitative function. It is recommended to run this function, if you choose to fit 'PRS_gxe x E' interaction component (i.e. novel proposed model, Model 5) when generating risk scores. If the 'PRS_gxe x E' term is significant in Model 5, and insignificant in Model 5* (permuted p value), consider that the 'PRS_gxe x E' interaction component is actually insignificant (always give priority to the p value obtained from the permuted model).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_permuted_binary(
  Bphe_target,
  Bcov_target,
  iterations = 1000,
  add_score,
  gxe_score
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_permuted_binary_+3A_bphe_target">Bphe_target</code></td>
<td>
<p>Phenotype file containing family ID, individual ID and phenotype of the target dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="summary_permuted_binary_+3A_bcov_target">Bcov_target</code></td>
<td>
<p>Covariate file containing family ID, individual ID, standardized covariate, square of standardized covariate, and/or confounders of the target dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="summary_permuted_binary_+3A_iterations">iterations</code></td>
<td>
<p>Number of iterations used in permutation</p>
</td></tr>
<tr><td><code id="summary_permuted_binary_+3A_add_score">add_score</code></td>
<td>
<p>PRSs generated using additive SNP effects of GWEIS summary statistics</p>
</td></tr>
<tr><td><code id="summary_permuted_binary_+3A_gxe_score">gxe_score</code></td>
<td>
<p>PRSs generated using interaction SNP effects of GWEIS summary statistics</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will output
</p>
<table>
<tr><td><code>B_permuted_p</code></td>
<td>
<p>the p value of the permuted model</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  
a &lt;- GWEIS_binary(plink_path, DummyData, Bphe_discovery, Bcov_discovery)
add &lt;- a[c("ID", "A1", "ADD_OR")]
gxe &lt;- a[c("ID", "A1", "INTERACTION_OR")]
p &lt;- PRS_binary(plink_path, DummyData, summary_input = add)
q &lt;- PRS_binary(plink_path, DummyData, summary_input = gxe)
x &lt;- summary_permuted_binary(Bphe_target, Bcov_target, iterations = 1000, 
add_score = p, gxe_score = q)
x

## End(Not run)
</code></pre>

<hr>
<h2 id='summary_permuted_quantitative'>summary_permuted_quantitative function
This function outputs the p value of permuted model in the target dataset, using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_quantitative functions. It is recommended to run this function, if you choose to fit 'PRS_gxe x E' interaction component (i.e. novel proposed model, Model 4) when generating risk scores. If the 'PRS_gxe x E' term is significant in Model 4, and insignificant in Model 4* (permuted p value), consider that the 'PRS_gxe x E' interaction component is actually insignificant (always give priority to the p value obtained from the permuted model).</h2><span id='topic+summary_permuted_quantitative'></span>

<h3>Description</h3>

<p>summary_permuted_quantitative function
This function outputs the p value of permuted model in the target dataset, using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_quantitative functions. It is recommended to run this function, if you choose to fit 'PRS_gxe x E' interaction component (i.e. novel proposed model, Model 4) when generating risk scores. If the 'PRS_gxe x E' term is significant in Model 4, and insignificant in Model 4* (permuted p value), consider that the 'PRS_gxe x E' interaction component is actually insignificant (always give priority to the p value obtained from the permuted model).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_permuted_quantitative(
  Qphe_target,
  Qcov_target,
  iterations = 1000,
  add_score,
  gxe_score
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_permuted_quantitative_+3A_qphe_target">Qphe_target</code></td>
<td>
<p>Phenotype file containing family ID, individual ID and phenotype of the target dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="summary_permuted_quantitative_+3A_qcov_target">Qcov_target</code></td>
<td>
<p>Covariate file containing family ID, individual ID, standardized covariate, square of standardized covariate, and/or confounders of the target dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="summary_permuted_quantitative_+3A_iterations">iterations</code></td>
<td>
<p>Number of iterations used in permutation</p>
</td></tr>
<tr><td><code id="summary_permuted_quantitative_+3A_add_score">add_score</code></td>
<td>
<p>PRSs generated using additive SNP effects of GWEIS summary statistics</p>
</td></tr>
<tr><td><code id="summary_permuted_quantitative_+3A_gxe_score">gxe_score</code></td>
<td>
<p>PRSs generated using interaction SNP effects of GWEIS summary statistics</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will output
</p>
<table>
<tr><td><code>Q_permuted_p.txt</code></td>
<td>
<p>the p value of the permuted model</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  
a &lt;- GWEIS_quantitative(plink_path, DummyData, Qphe_discovery, Qcov_discovery)
add &lt;- a[c("ID", "A1", "ADD_BETA")]
gxe &lt;- a[c("ID", "A1", "INTERACTION_BETA")]
p &lt;- PRS_quantitative(plink_path, DummyData, summary_input = add)
q &lt;- PRS_quantitative(plink_path, DummyData, summary_input = gxe)
x &lt;- summary_permuted_quantitative(Qphe_target, Qcov_target, iterations = 1000, 
add_score = p, gxe_score = q)
x 

## End(Not run)
</code></pre>

<hr>
<h2 id='summary_regular_binary'>summary_regular_binary function
This function outputs the summary of regular model and final risk score values of each individual in the target dataset using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_binary function.</h2><span id='topic+summary_regular_binary'></span>

<h3>Description</h3>

<p>summary_regular_binary function
This function outputs the summary of regular model and final risk score values of each individual in the target dataset using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_binary function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_regular_binary(
  Bphe_target,
  Bcov_target,
  add_score = NULL,
  gxe_score = NULL,
  Model
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_regular_binary_+3A_bphe_target">Bphe_target</code></td>
<td>
<p>Phenotype file containing family ID, individual ID and phenotype of the target dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="summary_regular_binary_+3A_bcov_target">Bcov_target</code></td>
<td>
<p>Covariate file containing family ID, individual ID, standardized covariate, square of standardized covariate, and/or confounders of the target dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="summary_regular_binary_+3A_add_score">add_score</code></td>
<td>
<p>PRSs generated using additive SNP effects of GWAS/GWEIS summary statistics</p>
</td></tr>
<tr><td><code id="summary_regular_binary_+3A_gxe_score">gxe_score</code></td>
<td>
<p>PRSs generated using interaction SNP effects of GWEIS summary statistics</p>
</td></tr>
<tr><td><code id="summary_regular_binary_+3A_model">Model</code></td>
<td>
<p>Specify the model number (0: y = PRS_trd + E + confounders, 1: y = PRS_trd + E + PRS_trd x E + confounders, 2: y = PRS_add + E + PRS_add x E + confounders, 3: y = PRS_add + E + PRS_gxe x E + confounders, 4: y = PRS_add + E + PRS_gxe + PRS_gxe x E + confounders, 5: y = PRS_add + E + E^2 + PRS_gxe + PRS_gxe x E + confounders, where y is the outcome variable, E is the covariate of interest, PRS_trd and PRS_add are the polygenic risk scores computed using additive SNP effects of GWAS and GWEIS summary statistics respectively, and PRS_gxe is the polygenic risk scores computed using GxE interaction SNP effects of GWEIS summary statistics.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will output
</p>
<table>
<tr><td><code>Bsummary</code></td>
<td>
<p>the summary of the fitted model</p>
</td></tr>
<tr><td><code>Individual_risk_values</code></td>
<td>
<p>the estimated risk values of individuals in the target sample</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  
a &lt;- GWAS_binary(plink_path, DummyData, Bphe_discovery, Bcov_discovery)
trd &lt;- a[c("ID", "A1", "OR")]
b &lt;- GWEIS_binary(plink_path, DummyData, Bphe_discovery, Bcov_discovery)
add &lt;- b[c("ID", "A1", "ADD_OR")]
gxe &lt;- b[c("ID", "A1", "INTERACTION_OR")]
p &lt;- PRS_binary(plink_path, DummyData, summary_input = trd)
q &lt;- PRS_binary(plink_path, DummyData, summary_input = add)
r &lt;- PRS_binary(plink_path, DummyData, summary_input = gxe)
summary_regular_binary(Bphe_target, Bcov_target, 
                            add_score = p,
                            Model = 0)
summary_regular_binary(Bphe_target, Bcov_target, 
                            add_score = p,
                            Model = 1)
summary_regular_binary(Bphe_target, Bcov_target, 
                            add_score = q,
                            Model = 2)
summary_regular_binary(Bphe_target, Bcov_target, 
                            add_score = q, 
                            gxe_score = r, 
                            Model = 3) 
summary_regular_binary(Bphe_target, Bcov_target, 
                            add_score = q, 
                            gxe_score = r, 
                            Model = 4) 
x &lt;- summary_regular_binary(Bphe_target, Bcov_target, 
                            add_score = q, 
                            gxe_score = r, 
                            Model = 5) 
sink("Bsummary.txt") #to create a file in the working directory
print(x$summary) #to write the output
sink() #to save the output
sink("Individual_risk_values.txt") #to create a file in the working directory
write.table(x$risk.values, sep = " ", row.names = FALSE, col.names = FALSE, 
quote = FALSE) #to write the output
sink() #to save the output
x$summary #to obtain the model summary output
x$risk.values #to obtain the predicted risk values of target individuals

## End(Not run)
</code></pre>

<hr>
<h2 id='summary_regular_quantitative'>summary_regular_quantitative function
This function outputs the summary of regular model and final risk score values of each individual in the target dataset using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_quantitative function.</h2><span id='topic+summary_regular_quantitative'></span>

<h3>Description</h3>

<p>summary_regular_quantitative function
This function outputs the summary of regular model and final risk score values of each individual in the target dataset using pre-generated Polygenic Risk Scores (PRSs) of all the individuals. Note that the input used in this function can be generated by using PRS_quantitative function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_regular_quantitative(
  Qphe_target,
  Qcov_target,
  add_score = NULL,
  gxe_score = NULL,
  Model
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_regular_quantitative_+3A_qphe_target">Qphe_target</code></td>
<td>
<p>Phenotype file containing family ID, individual ID and phenotype of the target dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="summary_regular_quantitative_+3A_qcov_target">Qcov_target</code></td>
<td>
<p>Covariate file containing family ID, individual ID, standardized covariate, square of standardized covariate, and/or confounders of the target dataset as columns, without heading</p>
</td></tr>
<tr><td><code id="summary_regular_quantitative_+3A_add_score">add_score</code></td>
<td>
<p>PRSs generated using additive SNP effects of GWAS/GWEIS summary statistics</p>
</td></tr>
<tr><td><code id="summary_regular_quantitative_+3A_gxe_score">gxe_score</code></td>
<td>
<p>PRSs generated using interaction SNP effects of GWEIS summary statistics</p>
</td></tr>
<tr><td><code id="summary_regular_quantitative_+3A_model">Model</code></td>
<td>
<p>Specify the model number (0: y = PRS_trd + E + confounders, 1: y = PRS_trd + E + PRS_trd x E + confounders, 2: y = PRS_add + E + PRS_add x E + confounders, 3: y = PRS_add + E + PRS_gxe x E + confounders, 4: y = PRS_add + E + PRS_gxe + PRS_gxe x E + confounders, where y is the outcome variable, E is the covariate of interest, PRS_trd and PRS_add are the polygenic risk scores computed using additive SNP effects of GWAS and GWEIS summary statistics respectively, and PRS_gxe is the polygenic risk scores computed using GxE interaction SNP effects of GWEIS summary statistics.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will output
</p>
<table>
<tr><td><code>Qsummary.txt</code></td>
<td>
<p>the summary of the fitted model</p>
</td></tr>
<tr><td><code>Individual_risk_values.txt</code></td>
<td>
<p>the estimated risk values of individuals in the target sample</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
a &lt;- GWAS_quantitative(plink_path, DummyData, Qphe_discovery, Qcov_discovery)
trd &lt;- a[c("ID", "A1", "BETA")]
b &lt;- GWEIS_quantitative(plink_path, DummyData, Qphe_discovery, Qcov_discovery)
add &lt;- b[c("ID", "A1", "ADD_BETA")]
gxe &lt;- b[c("ID", "A1", "INTERACTION_BETA")]
p &lt;- PRS_quantitative(plink_path, DummyData, summary_input = trd)
q &lt;- PRS_quantitative(plink_path, DummyData, summary_input = add)
r &lt;- PRS_quantitative(plink_path, DummyData, summary_input = gxe)
summary_regular_quantitative(Qphe_target, Qcov_target, 
                            add_score = p,
                            Model = 0)
summary_regular_quantitative(Qphe_target, Qcov_target, 
                            add_score = p,
                            Model = 1)
summary_regular_quantitative(Qphe_target, Qcov_target, 
                            add_score = q,
                            Model = 2)
summary_regular_quantitative(Qphe_target, Qcov_target, 
                            add_score = q, 
                            gxe_score = r, 
                            Model = 3) 
x &lt;- summary_regular_quantitative(Qphe_target, Qcov_target, 
                            add_score = q, 
                            gxe_score = r, 
                            Model = 4) 
sink("Qsummary.txt") #to create a file in the working directory
print(x$summary) #to write the output
sink() #to save the output
sink("Individual_risk_values.txt") #to create a file in the working directory
write.table(x$risk.values, sep = " ", row.names = FALSE, col.names = FALSE, 
quote = FALSE) #to write the output
sink() #to save the output
x$summary #to obtain the model summary output
x$risk.values #to obtain the predicted risk values of target individuals 

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
