<!DOCTYPE html><html><head><title>Help for package GxEScanR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {GxEScanR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gwas'><p>gwas</p></a></li>
<li><a href='#gweis'><p>gweis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Run GWAS/GWEIS Scans Using Binary Dosage Files</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-10-02</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools to run genome-wide association study (GWAS) and
  genome-wide by environment interaction study (GWEIS) scans using the genetic
  data stored in a binary dosage file. The user provides a data frame with
  the subject's covariate data and the information about the binary dosage
  file returned by the BinaryDosage::getbdinfo() routine.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.4), prodlim</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BinaryDosage, testthat (&ge; 2.1.0), covr, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-10-02 18:22:38 UTC; jbike_000</td>
</tr>
<tr>
<td>Author:</td>
<td>John Morrison [aut, cre],
  NCI [fnd] (CA196559),
  NCI [fnd] (CA201407),
  NIEHS [fnd] (ES007048),
  NHLBI [fnd] (HL115606)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>John Morrison &lt;jmorr@usc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-10-02 21:52:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='gwas'>gwas</h2><span id='topic+gwas'></span>

<h3>Description</h3>

<p>Run a gwas using genetic data from a binary dosage file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gwas(data, bdinfo, snps, outfile, skipfile, minmaf, blksize, binary)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gwas_+3A_data">data</code></td>
<td>
<p>Data frame containing the subject ID, phenotype
and covariates</p>
</td></tr>
<tr><td><code id="gwas_+3A_bdinfo">bdinfo</code></td>
<td>
<p>Information about the binary dosage file returned
from the BinaryDosage::getbdinfo routine</p>
</td></tr>
<tr><td><code id="gwas_+3A_snps">snps</code></td>
<td>
<p>The SNPs to be used in the scan. This may be an integer
vector indicate which SNPs to use in the binary dosage file or a 
character vector of the SNP IDs to use. The value may also be &quot;all&quot;,
indicating to use all SNPs. The default value is &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="gwas_+3A_outfile">outfile</code></td>
<td>
<p>The file name for the results Can be blank.
If the value is &quot;&quot;, the results are returned as a data frame. Default
value is &quot;&quot;</p>
</td></tr>
<tr><td><code id="gwas_+3A_skipfile">skipfile</code></td>
<td>
<p>The name of the file to write the SNPs that were not
used and the reason they weren't used. If the value is blank, there is
no output of the unused SNPs. Default value is &quot;&quot;.</p>
</td></tr>
<tr><td><code id="gwas_+3A_minmaf">minmaf</code></td>
<td>
<p>Minimum minor allele frequency of SNPs to include
in analysis. SNPS that have less than 20 minor alleles observed
will be excluded from the analysis regardless of the value of
minmaf. A value of 0 indicates to use all the SNPs that have 20
minor alleles observed. Default value is 0.</p>
</td></tr>
<tr><td><code id="gwas_+3A_blksize">blksize</code></td>
<td>
<p>Size of blocks of SNPs to read in at one time.
Larger blocks can improve overall speed but require larger
amounts of computer memory. A value of 0 indicates to use the
recommended block size. Default value is 0.</p>
</td></tr>
<tr><td><code id="gwas_+3A_binary">binary</code></td>
<td>
<p>Logical value indicating if the phenotype
is a binary value. Default value is false.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>0
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bdinfo &lt;- readRDS(system.file("extdata/pdata_4_1.bdinfo", package = "GxEScanR"))
covdata &lt;- readRDS(system.file("extdata/covdata.rds", package = "GxEScanR"))

results &lt;- gwas(data = covdata, bdinfo = bdinfo, binary = FALSE)
</code></pre>

<hr>
<h2 id='gweis'>gweis</h2><span id='topic+gweis'></span>

<h3>Description</h3>

<p>Run a gweis using genetic data from a binary dosage file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gweis(data, bdinfo, snps, outfile, skipfile, minmaf, blksize, binary)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gweis_+3A_data">data</code></td>
<td>
<p>Data frame containing the subject ID, phenotype
and covariates</p>
</td></tr>
<tr><td><code id="gweis_+3A_bdinfo">bdinfo</code></td>
<td>
<p>Information about the binary dosage file returned
from the BinaryDosage::getbdinfo routine</p>
</td></tr>
<tr><td><code id="gweis_+3A_snps">snps</code></td>
<td>
<p>The SNPs to be used in the scan. This may be an integer
vector indicate which SNPs to use in the binary dosage file or a 
character vector of the SNP IDs to use. The value may also be &quot;all&quot;,
indicating to use all SNPs. The default value is &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="gweis_+3A_outfile">outfile</code></td>
<td>
<p>The file name for the results Can be blank.
If the value is &quot;&quot;, the results are returned as a data frame. Default
value is &quot;&quot;</p>
</td></tr>
<tr><td><code id="gweis_+3A_skipfile">skipfile</code></td>
<td>
<p>The name of the file to write the SNPs that were not
used and the reason they weren't used. If the value is blank, there is
no output of the unused SNPs. Default value is &quot;&quot;.</p>
</td></tr>
<tr><td><code id="gweis_+3A_minmaf">minmaf</code></td>
<td>
<p>Minimum minor allele frequency of SNPs to include
in analysis. SNPS that have less than 20 minor alleles observed
will be excluded from the analysis regardless of the value of
minmaf. A value of 0 indicates to use all the SNPs that have 20
minor alleles observed. Default value is 0.</p>
</td></tr>
<tr><td><code id="gweis_+3A_blksize">blksize</code></td>
<td>
<p>Size of blocks of SNPs to read in at one time.
Larger blocks can improve overall speed but require larger
amounts of computer memory. A value of 0 indicates to use the
recommended block size. Default value is 0.</p>
</td></tr>
<tr><td><code id="gweis_+3A_binary">binary</code></td>
<td>
<p>Logical value indicating if the phenotype
is a binary value. Default value is false.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>0
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bdinfo &lt;- readRDS(system.file("extdata/pdata_4_1.bdinfo", package = "GxEScanR"))
covdata &lt;- readRDS(system.file("extdata/covdata.rds", package = "GxEScanR"))

results &lt;- gweis(data = covdata, bdinfo = bdinfo)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
