<!DOCTYPE html><html><head><title>Help for package HACSim</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HACSim}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HACSim-package'>
<p>Iterative Extrapolation of Species' Haplotype Accumulation</p>
Curves for Genetic Diversity Assessment</a></li>
<li><a href='#accumulate'><p>Internal C++ code</p></a></li>
<li><a href='#envr'><p>Simulation variable storage environment</p></a></li>
<li><a href='#HAC.sim'><p>Internal R code</p></a></li>
<li><a href='#HAC.simrep'><p>Run a simulation of haplotype accumulation curves for hypothetical or</p>
real species</a></li>
<li><a href='#HACClass'><p>Internal R code</p></a></li>
<li><a href='#HACHypothetical'><p>Set up an object to simulate haplotype accumulation curves for a</p>
hypothetical species</a></li>
<li><a href='#HACReal'><p>Set up an object to simulate haplotype accumulation curves for a real</p>
species</a></li>
<li><a href='#launchApp'><p>Launch HACSim R Shiny web app</p></a></li>
<li><a href='#sim.seqs'><p>Simulate DNA sequences according to DNA substitution models</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Iterative Extrapolation of Species' Haplotype Accumulation
Curves for Genetic Diversity Assessment</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.6-1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-23</td>
</tr>
<tr>
<td>Author:</td>
<td>Jarrett D. Phillips [aut, cre], 
        Steven H. French [ctb], Navdeep Singh [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jarrett D. Phillips &lt;phillipsjarrett1@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs iterative extrapolation of species' haplotype accumulation curves using a nonparametric stochastic (Monte Carlo) optimization method for assessment of specimen sampling completeness based on the approach of Phillips et al. (2015) &lt;<a href="https://doi.org/10.1515%2Fdna-2015-0008">doi:10.1515/dna-2015-0008</a>&gt;, Phillips et al. (2019) &lt;<a href="https://doi.org/10.1002%2Fece3.4757">doi:10.1002/ece3.4757</a>&gt; and Phillips et al. (2020) &lt;<a href="https://doi.org/10.7717%2Fpeerj-cs.243">doi:10.7717/peerj-cs.243</a>&gt;. 'HACSim' outputs a number of useful summary statistics of sampling coverage ("Measures of Sampling Closeness"), including an estimate of the likely required sample size (along with desired level confidence intervals) necessary to recover a given number/proportion of observed unique species' haplotypes. Any genomic marker can be targeted to assess likely required specimen sample sizes for genetic diversity assessment. The method is particularly well-suited to assess sampling sufficiency for DNA barcoding initiatives. Users can also simulate their own DNA sequences according to various models of nucleotide substitution. A Shiny app is also available.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td>&lt;<a href="https://github.com/jphill01/HACSim.R&amp;gt;">https://github.com/jphill01/HACSim.R&gt;</a>
&lt;<a href="https://github.com/jphill01/HACSim-RShiny-App&amp;gt;">https://github.com/jphill01/HACSim-RShiny-App&gt;</a>
&lt;<a href="https://jphill01.shinyapps.io/HACSim&amp;gt;">https://jphill01.shinyapps.io/HACSim&gt;</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape (&ge; 5.3), data.table (&ge; 1.12.8), graphics (&ge; 3.6.1),
matrixStats (&ge; 0.56.0), pegas (&ge; 0.13), Rcpp (&ge; 1.0.3),
shiny (&ge; 1.6.0), stats (&ge; 3.6.1), utils (&ge; 3.6.1)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-13 00:18:40 UTC; jarrettphillips</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-13 06:50:16 UTC</td>
</tr>
</table>
<hr>
<h2 id='HACSim-package'>
Iterative Extrapolation of Species' Haplotype Accumulation
Curves for Genetic Diversity Assessment
</h2><span id='topic+HACSim-package'></span>

<h3>Description</h3>

<p>HACSim (<strong>H</strong>aplotype <strong>A</strong>ccumulation <strong>C</strong>urve 
<strong>Sim</strong>ulator) employs a novel nonparametric stochastic (Monte Carlo) optimization method of iteratively generating species' haplotype accumulation 
curves through extrapolation to assess sampling completeness based on the 
approach outlined in Phillips et al. (2015) &lt;doi:10.1515/dna-2015-0008&gt;, 
Phillips et al. (2019) &lt;doi:10.1002/ece3.4757&gt; and Phillips et al. (2020) 
&lt;doi: 10.7717/peerj-cs.243&gt;. HACSim outputs a number of useful summary 
statistics of sampling coverage (&quot;Measures of Sampling Closeness&quot;), including 
an estimate of the likely required sample size (along with desired level 
confidence intervals) necessary to recover a given number/proportion of 
observed unique species' haplotypes. Any genomic marker can be targeted to 
assess likely required specimen sample sizes for genetic diversity assessment. 
The method is particularly well-suited to assess sampling sufficiency for DNA barcoding initiatives. Users can also simulate their own DNA sequences 
according to various models of nucleotide substitution. A Shiny app is also 
available.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> HACSim</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Iterative Extrapolation of Species' Haplotype Accumulation
Curves for Genetic Diversity Assessment</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.6-1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2022-05-23</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Jarrett D. Phillips [aut, cre], 
        Steven H. French [ctb], Navdeep Singh [ctb]</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Jarrett D. Phillips &lt;phillipsjarrett1@gmail.com&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Performs iterative extrapolation of species' haplotype accumulation curves using a nonparametric stochastic (Monte Carlo) optimization method for assessment of specimen sampling completeness based on the approach of Phillips et al. (2015) &lt;doi:10.1515/dna-2015-0008&gt;, Phillips et al. (2019) &lt;doi:10.1002/ece3.4757&gt; and Phillips et al. (2020) &lt;doi: 10.7717/peerj-cs.243&gt;. 'HACSim' outputs a number of useful summary statistics of sampling coverage ("Measures of Sampling Closeness"), including an estimate of the likely required sample size (along with desired level confidence intervals) necessary to recover a given number/proportion of observed unique species' haplotypes. Any genomic marker can be targeted to assess likely required specimen sample sizes for genetic diversity assessment. The method is particularly well-suited to assess sampling sufficiency for DNA barcoding initiatives. Users can also simulate their own DNA sequences according to various models of nucleotide substitution. A Shiny app is also available.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
URL: </td><td style="text-align: left;"> &lt;https://github.com/jphill01/HACSim.R&gt;
&lt;https://github.com/jphill01/HACSim-RShiny-App&gt;
&lt;https://jphill01.shinyapps.io/HACSim&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
NeedsCompilation: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> ape (&gt;= 5.3),
data.table (&gt;= 1.12.8),
graphics (&gt;= 3.6.1),
matrixStats (&gt;= 0.56.0),
pegas (&gt;= 0.13),
Rcpp (&gt;= 1.0.3),
shiny (&gt;= 1.6.0),
stats (&gt;= 3.6.1),
utils (&gt;= 3.6.1)</td>
</tr>
<tr>
 <td style="text-align: left;">
LinkingTo: </td><td style="text-align: left;"> Rcpp, RcppArmadillo</td>
</tr>
<tr>
 <td style="text-align: left;">
RoxygenNote: </td><td style="text-align: left;"> 6.1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Packaged: </td><td style="text-align: left;"> 2019-10-23 16:00:58 UTC; jarrettphillips</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<p>Index of help topics:
</p>
<pre>
HAC.sim                 Internal R code
HAC.simrep              Run a simulation of haplotype accumulation
                        curves for hypothetical or real species
HACClass                Internal R code
HACHypothetical         Set up an object to simulate haplotype
                        accumulation curves for a hypothetical species
HACReal                 Set up an object to simulate haplotype
                        accumulation curves for a real species
HACSim-package          Iterative Extrapolation of Species' Haplotype
                        Accumulation Curves for Genetic Diversity
                        Assessment
accumulate              Internal C++ code
envr                    Simulation variable storage environment
launchApp               Launch HACSim R Shiny web app
sim.seqs                Simulate DNA sequences according to DNA
                        substitution models
</pre>


<h3>Author(s)</h3>

<p>Jarrett D. Phillips [aut, cre], 
        Steven H. French [ctb], Navdeep Singh [ctb]
</p>
<p>Maintainer: Jarrett D. Phillips &lt;phillipsjarrett1@gmail.com&gt;
</p>


<h3>References</h3>

<p>Phillips, J.D., Gwiazdowski, R.A., Ashlock, D. and Hanner, R. (2015). An 
exploration of sufficient sampling effort to describe intraspecific DNA barcode haplotype diversity: examples from the ray-finned fishes (Chordata: 
Actinopterygii). DNA Barcodes, 3: 66-73.
</p>
<p>Phillips, J.D., Gillis, D.J. and Hanner, R.H. (2019). Incomplete estimates of 
genetic diversity within species: Implications for DNA barcoding. Ecology and Evolution, 9(5): 2996-3010.
</p>
<p>Phillips, J.D., Gillis, D.J. and Hanner, R.H. (2020). HACSim: An R package to estimate intraspecific sample sizes for genetic diversity assessment using 
haplotype accumulation curves. PeerJ Computer Science
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Simulate hypothetical species ##

N &lt;- 100 # total number of sampled individuals
Hstar &lt;- 10 # total number of haplotypes
probs &lt;- rep(1/Hstar, Hstar) # equal haplotype frequency distribution

HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar, 
probs = probs, filename = "output") # outputs a CSV 
# file called "output.csv"

## Simulate hypothetical species - subsampling ##
HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar, 
probs = probs, perms = 1000, p = 0.95, 
subsample = TRUE, prop = 0.25, conf.level = 0.95, 
filename = "output")

## Simulate hypothetical species and all paramaters changed - subsampling ##
HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar, probs = probs, 
perms = 10000, p = 0.90, subsample = TRUE, prop = 0.15, 
conf.level = 0.95, filename = "output")

HAC.simrep(HACSObj) # runs a simulation


## Simulate real species ##

## Not run: 
## Simulate real species ##
# outputs file called "output.csv"
HACSObj &lt;- HACReal(filename = "output") 

## Simulate real species - subsampling ##
HACSObj &lt;- HACReal(subsample = TRUE, prop = 0.15, 
conf.level = 0.95, filename = "output")

## Simulate real species and all parameters changed - subsampling ##
HACSObj &lt;- HACReal(perms = 10000, p = 0.90, subsample = TRUE, 
prop = 0.15, conf.level = 0.99, filename = "output")

# user prompted to select appropriate FASTA file
HAC.simrep(HACSObj) 
## End(Not run)

## Not run: 
## Simulate DNA sequences ##

num.seqs &lt;- 100 # number of DNA sequences
num.haps &lt;- 15 # number of haplotypes
length.seqs &lt;- 658 # length of DNA sequences
count.haps &lt;- c(60, rep(10, 2), rep(5, 2), rep(1, 5)) # haplotype frequency distribution
nucl.freqs &lt;- rep(0.25, 4) # nucleotide frequency distribution
subst.model &lt;- "JC69" # desired nucleotide substitution model
mu.rate &lt;- 1e-3 # mutation rate
transi.rate &lt;- NULL # transition rate
transv.rate &lt;- NULL # transversion rate

sim.seqs(num.seqs = num.seqs, num.haps = num.haps, length.seqs = length.seqs,
nucl.freqs = nucl.freqs, count.haps = count.haps, subst.model = subst.model, 
transi.rate = transi.rate, transv.rate = transv.rate)

# outputs file called "output.csv"
HACSObj &lt;- HACReal(filename = "output") 

## Simulate DNA sequences - subsampling ##
HACSObj &lt;- HACReal(subsample = TRUE, prop = 0.15, 
conf.level = 0.95, filename = "output")

## Simulate DNA sequences and all parameters changed - subsampling ##
HACSObj &lt;- HACReal(perms = 10000, p = 0.90, subsample = TRUE, 
prop = 0.15, conf.level = 0.99, filename = "output") 

# user prompted to select appropriate FASTA file
HAC.simrep(HACSObj) 
## End(Not run)

</code></pre>

<hr>
<h2 id='accumulate'>Internal C++ code</h2><span id='topic+accumulate'></span>

<h3>Description</h3>

<p><code>accumulate</code> comprises internal C++ code employed by 
<code>HAC.sim</code>. It is not directly called by the user. 
</p>

<hr>
<h2 id='envr'>Simulation variable storage environment</h2><span id='topic+envr'></span>

<h3>Description</h3>

<p><code>envr</code> is a new (initially empty) environment that is created 
when <code>HACSim</code> is loaded. 
</p>


<h3>Value</h3>

<p>When a simulation is run via <code>HAC.simrep</code>, <code>envr</code> will contain 
26 elements as follows:
</p>
<table>
<tr><td><code>ci.type</code></td>
<td>
<p>Type of confidence interval to compute and plot. Default is 
<code>conf.type = "quantile"</code>.</p>
</td></tr>
<tr><td><code>conf.level</code></td>
<td>
<p>The desired confidence level. Default is 
<code>conf.level = 0.95</code>.</p>
</td></tr>
<tr><td><code>d</code></td>
<td>
<p>A dataframe with <code>Nstar - X</code> rows and five columns: specimens 
(specs), accumulated haplotypes (means), standard deviations (sds) and 
quantiles (both lower and upper)</p>
</td></tr>
<tr><td><code>df.out</code></td>
<td>
<p>A dataframe with <code>iters</code> rows and six columns displaying 
&quot;Measures of Sampling Closeness&quot;.</p>
</td></tr>
<tr><td><code>filename</code></td>
<td>
<p>The name of the file where results are to be saved. Default 
is NULL.</p>
</td></tr>
<tr><td><code>Hstar</code></td>
<td>
<p>Number of unique species' haplotypes</p>
</td></tr>
<tr><td><code>input.seqs</code></td>
<td>
<p>Should DNA sequences be inputted? Default is FALSE.</p>
</td></tr>
<tr><td><code>iters</code></td>
<td>
<p>The number of iterations required to reach convergence</p>
</td></tr>
<tr><td><code>N</code></td>
<td>
<p>The starting sample size used to initialize the algorithm</p>
</td></tr>
<tr><td><code>Nstar</code></td>
<td>
<p>The final (extrapolated) sample size</p>
</td></tr>
<tr><td><code>Nstar.high</code></td>
<td>
<p>The upper endpoint of the desired level confidence interval 
for the 'true' required sample size</p>
</td></tr>
<tr><td><code>Nstar.low</code></td>
<td>
<p>The lower endpoint of the desired level confidence interval 
for the 'true' required sample size</p>
</td></tr>
<tr><td><code>num.iters</code></td>
<td>
<p>Number of iterations to compute. <code>num.iters = NULL</code> by 
default (i.e., all iterations are computed; users can specify 
<code>num.iters</code> = 1 for the first iteration.)</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>The user-specified level of haplotype recovery. Default is 
<code>p</code> = 0.95.</p>
</td></tr>
<tr><td><code>perms</code></td>
<td>
<p>The user-specified number of permutations (replications). Default 
is <code>perms</code> = 10000.</p>
</td></tr>
<tr><td><code>probs</code></td>
<td>
<p>Haplotype frequency distribution vector</p>
</td></tr>
<tr><td><code>progress</code></td>
<td>
<p>Should iteration results be outputted to the console? Default 
is TRUE.</p>
</td></tr>
<tr><td><code>prop.haps</code></td>
<td>
<p>If <code>subset.haps</code> = TRUE, the user-specified proportion of haplotype labels to recover</p>
</td></tr>
<tr><td><code>prop.seqs</code></td>
<td>
<p>If <code>subset.seqs</code> = TRUE, the user-specified proportion of 
DNA sequences to recover</p>
</td></tr>
<tr><td><code>ptm</code></td>
<td>
<p>A timer to track progress of the algorithm in seconds</p>
</td></tr>
<tr><td><code>R</code></td>
<td>
<p>The proportion of haplotypes recovered by the algorithm</p>
</td></tr>
<tr><td><code>R.low</code></td>
<td>
<p>The lower endpoint of the desired level confidence interval for 
the 'true' fraction of haplotypes captured</p>
</td></tr>
<tr><td><code>R.up</code></td>
<td>
<p>The upper endpoint of the desired level confidence interval for the 
'true' fraction of haplotypes captured</p>
</td></tr>
<tr><td><code>subset.haps</code></td>
<td>
<p>Should a subsample of haplotype labels be taken? Default is 
FALSE.</p>
</td></tr>
<tr><td><code>subset.seqs</code></td>
<td>
<p>Should a subsample of DNA sequences be taken? Default is 
FALSE.</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>Mean number of specimens not sampled</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Returns the frequencies of each haplotype in the extrapolated sample 
max(envr$d$specs) * envr$probs

# Returns the extrapolated sample size corresponding to the dotted line 
# in the last iteration plot
envr$d[which(envr$d$means &gt;= envr$p * envr$Hstar), ][1, 1]
</code></pre>

<hr>
<h2 id='HAC.sim'>Internal R code</h2><span id='topic+HAC.sim'></span>

<h3>Description</h3>

<p><code>HAC.sim</code> comprises internal R code used by <code>HAC.simrep</code> 
and is not directly called by the user. 
</p>

<hr>
<h2 id='HAC.simrep'>Run a simulation of haplotype accumulation curves for hypothetical or 
real species</h2><span id='topic+HAC.simrep'></span>

<h3>Description</h3>

<p>Runs the <code>HACSim</code> algorithm by successively calling 
<code>HAC.sim</code> to iteratively extrapolate haplotype accumulation curves to 
determine likely specimen sample sizes for hypothetical or real species
</p>
<p>The algorithm employs the following iterative methods when calculating the 
&quot;Measures of Sampling Closeness&quot;:
</p>

<ul>
<li><p> Mean number of haplotype sampled: <code class="reqn">H_i</code>
</p>
</li>
<li><p> Mean number of haplotypes not sampled <code class="reqn">H^* - H_i</code>
</p>
</li>
<li><p> Proportion of haplotypes sampled: <code class="reqn">\frac{H_i}{H^*}</code>
</p>
</li>
<li><p> Proportion of haplotypes not sampled: <code class="reqn">1 - \frac{H_i}{H^*}</code>
</p>
</li>
<li><p> Mean value of <code class="reqn">N*</code>: <code class="reqn">\frac{N_iH^*}{H_i}</code>
</p>
</li>
<li><p> Mean number of specimens not sampled: <code class="reqn">\frac{N_iH^*}{H_i} - N_i</code>
</p>
</li></ul>

<p>where <code class="reqn">H_i</code> is stochastically-determined through sampling from 
<code>probs</code>, the observed species' haplotype frequency distribution vector. 
</p>
<p>As the algorithm proceeds, <code class="reqn">H_i</code> will approach <code class="reqn">H^*</code> asymptotically 
(and hence, <code class="reqn">N_i</code> will converge to <code class="reqn">N^*</code>), but will likely fluctuate 
randomly from one iteration to the next. However, estimates of <code class="reqn">N^*</code> found 
at each iteration will be monotonically-increasing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HAC.simrep(HACSObject)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HAC.simrep_+3A_hacsobject">HACSObject</code></td>
<td>
<p>object containing the desired simulation 
parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Iteration results are outputted to the console and graphs displayed in 
the plot window. Plots depict haplotype accumulation (along with shaded 
confidence intervals for the mean number of haplotypes found). Dashed lines 
correspond to the endpoint of the curve and reflect haplotype recovery for a 
user-defined cutoff (default <code>p</code> = 0.95, 95% haplotype diversity). Output 
from the first iteration is useful for judging levels of haplotype diversity 
and recovery found in observed intraspecific sequence datasets, reflecting 
current sampling depth. The required sample size is displayed in the second-
last iteration. All other information corresponding to the extrapolated sample 
size can be found in the last iteration. Iteration results can optionally be 
saved to a CSV file. Subsampled DNA sequences are automatically saved to a 
FASTA file.</p>


<h3>Note</h3>

<p>When simulating real species via <code>HACReal(...)</code>, a pop-up window will appear prompting the user to select an intraspecific FASTA file of 
aligned/trimmed DNA sequences. The alignment must not contain missing or 
ambiguous nucleotides (i.e., it should only contain A, C, G or T); otherwise, haplotype diversity may be overestimated. Excluding sequences or alignment 
sites with missing/ambiguous data is an option.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  ## Simulate hypothetical species ##
  
  N &lt;- 100 # total number of sampled individuals
  Hstar &lt;- 10 # total number of haplotypes
  probs &lt;- rep(1/Hstar, Hstar) # equal haplotype frequency distribution
  
  HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar , probs = probs, 
  filename = "output") # outputs a CSV file called "output.csv"
  
  ## Simulate hypothetical species - subsampling ##
  HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar, probs = probs, 
  perms = 1000, p = 0.95, subsample = TRUE, prop = 0.25, 
  conf.level = 0.95, filename = "output")
  
  ## Simulate hypothetical species and all paramaters changed - subsampling ##
  HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar, probs = probs, 
  perms = 10000, p = 0.90, subsample = TRUE, prop = 0.15, conf.level = 0.95, 
  filename = "output")
  
  try(HAC.simrep(HACSObj)) # runs a simulation
  
  ## Simulate real species ##
  
  ## Not run: 
    ## Simulate real species ##
    # outputs file called "output.csv"
    HACSObj &lt;- HACReal(filename = "output") 
    
    ## Simulate real species - subsampling ##
    HACSObj &lt;- HACReal(subsample = TRUE, prop = 0.15, conf.level = 0.95, 
    filename = "output")
    
    ## Simulate real species and all parameters changed - subsampling ##
    HACSObj &lt;- HACReal(perms = 10000, p = 0.90, subsample = TRUE, 
    prop = 0.15, conf.level = 0.99, filename = "output")
    
    # user prompted to select appropriate FASTA file
    try(HAC.simrep(HACSObj))
    
## End(Not run)
</code></pre>

<hr>
<h2 id='HACClass'>Internal R code</h2><span id='topic+HACClass'></span>

<h3>Description</h3>

<p><code>HACClass</code> comprises internal R code used to generate an 
object used by <code>HAC.simrep</code>. It is not directly called by the user.
</p>

<hr>
<h2 id='HACHypothetical'>Set up an object to simulate haplotype accumulation curves for a 
hypothetical species</h2><span id='topic+HACHypothetical'></span>

<h3>Description</h3>

<p>Helper function which creates an object containing necessary 
information to run a simulation of haplotype accumulation for a hypothetical 
species of interest</p>


<h3>Usage</h3>

<pre><code class='language-R'>HACHypothetical(N, Hstar, probs, perms = 10000, p = 0.95, 
conf.level = 0.95, ci.type = "quantile", subsample = FALSE, prop = NULL,
progress = TRUE, num.iters = NULL, filename = NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HACHypothetical_+3A_n">N</code></td>
<td>
<p>Number of individuals</p>
</td></tr>
<tr><td><code id="HACHypothetical_+3A_hstar">Hstar</code></td>
<td>
<p>Number of unique species' haplotypes</p>
</td></tr>
<tr><td><code id="HACHypothetical_+3A_probs">probs</code></td>
<td>
<p>Haplotype frequency distribution vector</p>
</td></tr>
<tr><td><code id="HACHypothetical_+3A_perms">perms</code></td>
<td>
<p>Number of permutations (replications)</p>
</td></tr>
<tr><td><code id="HACHypothetical_+3A_p">p</code></td>
<td>
<p>Proportion of haplotypes to recover</p>
</td></tr>
<tr><td><code id="HACHypothetical_+3A_conf.level">conf.level</code></td>
<td>
<p>Desired confidence level for graphical output and interval estimation</p>
</td></tr>
<tr><td><code id="HACHypothetical_+3A_ci.type">ci.type</code></td>
<td>
<p>Type of confidence interval for graphical output. Choose from &quot;quantile&quot; or &quot;asymptotic&quot;</p>
</td></tr>
<tr><td><code id="HACHypothetical_+3A_subsample">subsample</code></td>
<td>
<p>Is a subsample of haplotype labels desired?</p>
</td></tr>
<tr><td><code id="HACHypothetical_+3A_prop">prop</code></td>
<td>
<p>If subsample = TRUE, the proportion of haplotype labels to 
subsample</p>
</td></tr>
<tr><td><code id="HACHypothetical_+3A_num.iters">num.iters</code></td>
<td>
<p>Number of iterations to compute</p>
</td></tr>
<tr><td><code id="HACHypothetical_+3A_progress">progress</code></td>
<td>
<p>Should iteration output be printed to the R console?</p>
</td></tr>
<tr><td><code id="HACHypothetical_+3A_filename">filename</code></td>
<td>
<p>Name of file where simulation results are to be saved</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object of class &quot;HAC&quot; with 13 elements that can be passed to 
<code>HAC.simrep</code> as follows:
</p>
<table>
<tr><td><code>input.seqs</code></td>
<td>
<p>Should a FASTA file of aligned/trimmed DNA sequences be 
inputted? Default is FALSE</p>
</td></tr>
<tr><td><code>subset.seqs</code></td>
<td>
<p>Should a subsample of DNA sequences be taken? Default is 
FALSE</p>
</td></tr>
<tr><td><code>prop.seqs</code></td>
<td>
<p>Proportion of DNA sequences to subsample. Default is NULL</p>
</td></tr>
<tr><td><code>prop.haps</code></td>
<td>
<p>Proportion of haplotype labels to subsample. Default is NULL 
(can be altered by user)</p>
</td></tr>
<tr><td><code>subset.haps</code></td>
<td>
<p>Should a subsample of haplotype labels be taken? Default 
is NULL (can be altered by user)</p>
</td></tr>
<tr><td><code>N</code></td>
<td>
<p>Number of individuals. NA by default (provided by user)</p>
</td></tr>
<tr><td><code>Hstar</code></td>
<td>
<p>Number of unique species' haplotypes. NA by default (provided 
by user)</p>
</td></tr>
<tr><td><code>probs</code></td>
<td>
<p>Haplotype frequency distribution vector. NA by default (provided 
by user)</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>Proportion of haplotypes to recover. <code>p</code> = 0.95 by default.</p>
</td></tr>
<tr><td><code>perms</code></td>
<td>
<p>Number of permutations (replications). <code>perms</code> = 10000 by 
default.</p>
</td></tr>
<tr><td><code>conf.level</code></td>
<td>
<p>Desired confidence level for graphical output and interval estimation. <code>conf.level</code> = 0.95 by default.</p>
</td></tr>
<tr><td><code>ci.type</code></td>
<td>
<p>Type of confidence interval for graphical output. 
<code>ci.type = "quantile"</code> by default</p>
</td></tr>
<tr><td><code>num.iters</code></td>
<td>
<p>Number of iterations to compute. <code>num.iters = NULL</code> by 
default (i.e., all iterations are computed; users can specify 
<code>num.iters</code> = 1 for the first iteration.)</p>
</td></tr>
<tr><td><code>progress</code></td>
<td>
<p>Should iteration output be printed to the R console? Default 
is TRUE</p>
</td></tr>
<tr><td><code>filename</code></td>
<td>
<p>Name of file where simulation results are to be saved.</p>
</td></tr>
</table>


<h3>Note</h3>

<p><code>N</code> must be greater than 1 and greater than or equal to <code>Hstar</code>.
</p>
<p><code>Hstar</code> must be greater than 1. 
</p>
<p><code>probs</code> must have a length equal to <code>Hstar</code> and its elements must sum 
to 1.</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Simulate hypothetical species ##
  
  N &lt;- 100 # total number of sampled individuals
  Hstar &lt;- 10 # total number of haplotypes
  probs &lt;- rep(1/Hstar, Hstar) # equal haplotype frequency distribution
  
  # outputs a CSV file called "output.csv"
  HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar, probs = probs, 
  filename = "output") 
  
  ## Simulate hypothetical species - subsampling ##
  # subsamples 25% of haplotype labels
  HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar, probs = probs, 
  perms = 1000, p = 0.95, subsample = TRUE, prop = 0.25, 
  conf.level = 0.95, filename = "output") 
  
  ## Simulate hypothetical species and all paramaters changed - subsampling ##
  HACSObj &lt;- HACHypothetical(N = N, Hstar = Hstar, probs = probs, 
  perms = 10000, p = 0.90, subsample = TRUE, prop = 0.15, conf.level = 0.95, 
  num.iters = 1, filename = "output")
  </code></pre>

<hr>
<h2 id='HACReal'>Set up an object to simulate haplotype accumulation curves for a real 
species</h2><span id='topic+HACReal'></span>

<h3>Description</h3>

<p>Helper function which creates an object containing necessary 
information to run a simulation of haplotype accumulation for a real species of interest</p>


<h3>Usage</h3>

<pre><code class='language-R'>HACReal(perms = 10000, p = 0.95, conf.level = 0.95, 
ci.type = "quantile", subsample = FALSE, prop = NULL, 
progress = TRUE, num.iters = NULL, filename = NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HACReal_+3A_perms">perms</code></td>
<td>
<p>Number of permutations (replications)</p>
</td></tr>
<tr><td><code id="HACReal_+3A_p">p</code></td>
<td>
<p>Proportion of haplotypes to recover</p>
</td></tr>
<tr><td><code id="HACReal_+3A_conf.level">conf.level</code></td>
<td>
<p>Desired confidence level for graphical output and interval estimation</p>
</td></tr>
<tr><td><code id="HACReal_+3A_ci.type">ci.type</code></td>
<td>
<p>Type of confidence interval for graphical output. Choose from &quot;quantile&quot; or &quot;asymptotic&quot;</p>
</td></tr>
<tr><td><code id="HACReal_+3A_subsample">subsample</code></td>
<td>
<p>Is a subsample of DNA sequences desired?</p>
</td></tr>
<tr><td><code id="HACReal_+3A_prop">prop</code></td>
<td>
<p>If subsample = TRUE, the proportion of DNA sequences to subsample</p>
</td></tr>
<tr><td><code id="HACReal_+3A_num.iters">num.iters</code></td>
<td>
<p>Number of iterations to compute</p>
</td></tr>
<tr><td><code id="HACReal_+3A_progress">progress</code></td>
<td>
<p>Should iteration output be printed to the R console?</p>
</td></tr>
<tr><td><code id="HACReal_+3A_filename">filename</code></td>
<td>
<p>Name of file where simulation results are to be saved</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object of class &quot;HAC&quot; with 13 elements that can be passed to 
<code>HAC.simrep</code> as follows:
</p>
<table>
<tr><td><code>input.seqs</code></td>
<td>
<p>Should a FASTA file of aligned/trimmed DNA sequences be 
inputted? Default is TRUE</p>
</td></tr>
<tr><td><code>subset.seqs</code></td>
<td>
<p>Should a subsample of DNA sequences be taken? Default is 
FALSE (can be altered by user)</p>
</td></tr>
<tr><td><code>prop.seqs</code></td>
<td>
<p>Proportion of DNA sequences to subsample. Default is NA 
(can be altered by user)</p>
</td></tr>
<tr><td><code>prop.haps</code></td>
<td>
<p>Proportion of haplotype labels to subsample. Default is 
NULL</p>
</td></tr>
<tr><td><code>subset.haps</code></td>
<td>
<p>Should a subsample of haplotype labels be taken? Default 
is NULL</p>
</td></tr>
<tr><td><code>N</code></td>
<td>
<p>Number of individuals. NA by default (computed automatically by 
algorithm)</p>
</td></tr>
<tr><td><code>Hstar</code></td>
<td>
<p>Number of unique species' haplotypes. NA by default (computed automatically by algorithm)</p>
</td></tr>
<tr><td><code>probs</code></td>
<td>
<p>Haplotype frequency distribution vector. NA by default (computed automatically by algorithm)</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>Proportion of haplotypes to recover. <code>p</code> = 0.95 by default.</p>
</td></tr>
<tr><td><code>perms</code></td>
<td>
<p>Number of permutations (replications). <code>perms</code> = 10000 by 
default.</p>
</td></tr>
<tr><td><code>conf.level</code></td>
<td>
<p>Desired confidence level for graphical output and interval estimation. <code>conf.level</code> = 0.95 by default.</p>
</td></tr>
<tr><td><code>ci.type</code></td>
<td>
<p>Type of confidence interval for graphical output. 
<code>ci.type = "quantile"</code> by default</p>
</td></tr>
<tr><td><code>num.iters</code></td>
<td>
<p>Number of iterations to compute. <code>num.iters = NULL</code> by 
default (i.e., all iterations are computed; users can specify 
<code>num.iters</code> = 1 for the first iteration.)</p>
</td></tr>
<tr><td><code>progress</code></td>
<td>
<p>Should iteration output be printed to the R console? Default 
is TRUE</p>
</td></tr>
<tr><td><code>filename</code></td>
<td>
<p>Name of file where simulation results are to be saved.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>    ## Simulate real species ##
    # outputs file called "output.csv"
    HACSObj &lt;- HACReal(filename = "output") 
    
    ## Simulate real species - subsampling ##
    # subsamples 25% of DNA sequences
    HACSObj &lt;- HACReal(subsample = TRUE, prop = 0.25, conf.level = 0.95, 
    filename = "output") 
    
    ## Simulate real species and all parameters changed - subsampling ##
    HACSObj &lt;- HACReal(perms = 10000, p = 0.90, subsample = TRUE, 
    prop = 0.15, conf.level = 0.99, num.iters = 1, filename = "output")
</code></pre>

<hr>
<h2 id='launchApp'>Launch HACSim R Shiny web app</h2><span id='topic+launchApp'></span>

<h3>Description</h3>

<p>Launch HACSim R Shiny web app locally on a user's R session</p>


<h3>Usage</h3>

<pre><code class='language-R'>launchApp()</code></pre>

<hr>
<h2 id='sim.seqs'>Simulate DNA sequences according to DNA substitution models</h2><span id='topic+sim.seqs'></span>

<h3>Description</h3>

<p>Simulates DNA sequences according to various DNA substitution 
models:
</p>

<ul>
<li><p> Jukes-Cantor (1969)
</p>
</li>
<li><p> Kimura (1980)
</p>
</li>
<li><p> Felsenstein (1981)
</p>
</li>
<li><p> Hasegawa-Kishino-Yano (1985)
</p>
</li></ul>

<p>Output can then be passed to <code>HACReal()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.seqs(num.seqs, num.haps, length.seqs, count.haps, nucl.freqs, 
codon.tbl = c("standard", "vertebrate mitochondrial", 
"invertebrate mitochondrial"), subst.model = c("JC69", "K80", "F81", "HKY85"), 
mu.rate, transi.rate, transv.rate)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim.seqs_+3A_num.seqs">num.seqs</code></td>
<td>
<p>Number of simulated DNA sequences</p>
</td></tr>
<tr><td><code id="sim.seqs_+3A_num.haps">num.haps</code></td>
<td>
<p>Number of simulated unique species' haplotypes</p>
</td></tr>
<tr><td><code id="sim.seqs_+3A_length.seqs">length.seqs</code></td>
<td>
<p>Basepair length of DNA sequences</p>
</td></tr>
<tr><td><code id="sim.seqs_+3A_count.haps">count.haps</code></td>
<td>
<p>Haplotype frequency distribution vector</p>
</td></tr>
<tr><td><code id="sim.seqs_+3A_nucl.freqs">nucl.freqs</code></td>
<td>
<p>Nucleotide frequency distribution vector of A, C, G, and 
T respectively</p>
</td></tr>
<tr><td><code id="sim.seqs_+3A_codon.tbl">codon.tbl</code></td>
<td>
<p>Codon table</p>
</td></tr>
<tr><td><code id="sim.seqs_+3A_subst.model">subst.model</code></td>
<td>
<p>Model of DNA substitution</p>
</td></tr>
<tr><td><code id="sim.seqs_+3A_mu.rate">mu.rate</code></td>
<td>
<p>Overall nucleotide mutation rate/site/generation</p>
</td></tr>
<tr><td><code id="sim.seqs_+3A_transi.rate">transi.rate</code></td>
<td>
<p>Nucleotide transition rate/site/generation</p>
</td></tr>
<tr><td><code id="sim.seqs_+3A_transv.rate">transv.rate</code></td>
<td>
<p>Nucleotide transversion rate/site/generation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A FASTA file of DNA sequences</p>


<h3>Note</h3>

<p><code>num.seqs</code> must be greater than or equal to <code>num.haps</code>.
</p>
<p>Both <code>num.seqs</code> and <code>num.haps</code> must be greater than 1.
</p>
<p><code>nucl.freqs</code> must have a length of four and its elements must sum to 1.
</p>
<p><code>count.haps</code> must have a length of <code>num.haps</code> and its elements must 
sum to <code>num.seqs</code>. 
</p>
<p><code>subst.model</code> must be one of &quot;JC69&quot; (Jukes Cantor corrected p-distance), 
&quot;K80&quot; (Kimura-2-Parameter (K2P), &quot;F81&quot; (Felenstein) or &quot;HKY85&quot;
(Hasegawa-Kishino-Yano)
</p>
<p><code>mu.rate</code> must be specified for both &quot;JC69&quot; and &quot;F81&quot; models
</p>
<p><code>transi.rate</code> and <code>transv.rate</code> must be specified for both &quot;K80&quot; 
and &quot;HKY85&quot; models
</p>
<p>All elements <code>nucl.freqs</code> must be equal to 0.25 when <code>subst.model</code> 
is either &quot;JC69&quot; or &quot;K80&quot; 
</p>
<p>All elements <code>nucl.freqs</code> must differ from 0.25 when <code>subst.model</code> 
is either &quot;F81&quot; or &quot;HKY85&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

# Simulate DNA sequences from the 5'-COI DNA barcode region under a Jukes 
# Cantor nucleotide substitution model

num.seqs &lt;- 100 # number of DNA sequences
num.haps &lt;- 10 # number of haplotypes
length.seqs &lt;- 658 # length of DNA sequences
count.haps &lt;- c(65, rep(10, 2), rep(5, 2), rep(1, 5)) # haplotype frequency distribution
nucl.freqs &lt;- rep(0.25, 4) # nucleotide frequency distribution
subst.model &lt;- "JC69" # desired nucleotide substitution model
codon.tbl &lt;- "vertebrate mitochondrial"
mu.rate &lt;- 1e-3 # mutation rate
transi.rate &lt;- NULL # transition rate
transv.rate &lt;- NULL # transversion rate

sim.seqs(num.seqs = num.seqs, num.haps = num.haps, length.seqs = length.seqs, 
count.haps = count.haps, nucl.freqs = nucl.freqs, codon.tbl = codon.tbl, 
subst.model = subst.model, mu.rate = mu.rate, transi.rate = transi.rate, 
transv.rate = transv.rate)


## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
