<!DOCTYPE html><html lang="en-US"><head><title>Help for package HaDeX</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HaDeX}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HaDeX-package'><p>HaDeX</p></a></li>
<li><a href='#add_stat_dependency'><p>Calculates confidence limits</p></a></li>
<li><a href='#calculate_confidence_limit_values'><p>Calculate the value of confidence limit</p></a></li>
<li><a href='#calculate_kinetics'><p>Calculate kinetic data</p></a></li>
<li><a href='#calculate_state_deuteration'><p>Calculate deuteration</p></a></li>
<li><a href='#comparison_plot'><p>Plot comparison plot</p></a></li>
<li><a href='#HaDeX_gui'><p>HaDeX Graphical User Interface</p></a></li>
<li><a href='#plot_coverage'><p>Plot peptide coverage</p></a></li>
<li><a href='#plot_kinetics'><p>Plot kinetics data</p></a></li>
<li><a href='#plot_position_frequency'><p>Plot position frequency</p></a></li>
<li><a href='#prepare_dataset'><p>Calculate data</p></a></li>
<li><a href='#quality_control'><p>Experiment quality control</p></a></li>
<li><a href='#read_hdx'><p>Read HDX-MS data file</p></a></li>
<li><a href='#reconstruct_sequence'><p>Reconstruct protein sequence</p></a></li>
<li><a href='#woods_plot'><p>Plot Woods' plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Analysis and Visualisation of Hydrogen/Deuterium Exchange Mass
Spectrometry Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for processing, analysis and visualization of Hydrogen Deuterium eXchange monitored by 
             Mass Spectrometry experiments (HDX-MS) (10.1093/bioinformatics/btaa587). 'HaDeX' introduces a new 
             standardized and reproducible workflow for the analysis of the HDX-MS data, including novel 
             uncertainty intervals. Additionally, it covers data exploration, quality control and generation of 
             publication-quality figures. All functionalities are also available in the in-built 'Shiny' app.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, dplyr, DT, ggplot2, gsubfn, latex2exp, reshape2,
readr, readxl, shiny, stringr, tidyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>spelling, covr, digest, gridExtra, knitr, pander, renv,
rmarkdown, shinycssloaders, shinyhelper, shinyjs, testthat,
vdiffr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-08-12 13:37:47 UTC; User</td>
</tr>
<tr>
<td>Author:</td>
<td>Weronika Puchala <a href="https://orcid.org/0000-0003-2163-1429"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut],
  Michal Burdukiewicz
    <a href="https://orcid.org/0000-0001-8926-582X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Dominik Rafacz <a href="https://orcid.org/0000-0003-0925-1909"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Weronika Puchala &lt;puchala.weronika@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-08-12 14:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='HaDeX-package'>HaDeX</h2><span id='topic+HaDeX-package'></span><span id='topic+HaDeX'></span>

<h3>Description</h3>

<p>The <code>HaDeX</code> package is a toolbox for the analysis of HDX-MS data.
</p>


<h3>Author(s)</h3>

<p>Weronika Puchala, Michal Burdukiewicz.
</p>

<hr>
<h2 id='add_stat_dependency'>Calculates confidence limits</h2><span id='topic+add_stat_dependency'></span>

<h3>Description</h3>

<p>Returns relation with confidence limits for each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_stat_dependency(
  calc_dat,
  confidence_limit = 0.98,
  theoretical = FALSE,
  relative = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_stat_dependency_+3A_calc_dat">calc_dat</code></td>
<td>
<p>processed data from DynamX file - using <code><a href="#topic+prepare_dataset">prepare_dataset</a></code></p>
</td></tr>
<tr><td><code id="add_stat_dependency_+3A_confidence_limit">confidence_limit</code></td>
<td>
<p>confidence limit chosen by user - from range [0, 1].</p>
</td></tr>
<tr><td><code id="add_stat_dependency_+3A_theoretical">theoretical</code></td>
<td>
<p>logical value to determine if the plot is theoretical or not.</p>
</td></tr>
<tr><td><code id="add_stat_dependency_+3A_relative">relative</code></td>
<td>
<p>logical value to determine if values are relative or absolute.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>...
</p>


<h3>Value</h3>

<p>calc_dat extended by column specifying if given peptide is relevant in given confidence limit. 
The value of the confidence limit is added as an attribute - as well as parameters used to calculate (theoretical/relative)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_hdx">read_hdx</a></code> <code><a href="#topic+prepare_dataset">prepare_dataset</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load example data
dat &lt;- read_hdx(system.file(package = "HaDeX", 
                            "HaDeX/data/KD_180110_CD160_HVEM.csv"))
                            
# prepate dataset for states `CD160` and `CD160_HVEM` in given time parameters 
calc_dat &lt;- prepare_dataset(dat,
                            in_state_first = "CD160_0.001",
                            chosen_state_first = "CD160_1",
                            out_state_first = "CD160_1440",
                            in_state_second = "CD160_HVEM_0.001",
                            chosen_state_second = "CD160_HVEM_1",
                            out_state_second = "CD160_HVEM_1440") 
                            
# add calculated confidence limits for prepared data
add_stat_dependency(calc_dat, 
                    confidence_limit = 0.98, 
                    theoretical = FALSE, 
                    relative = TRUE)
                     

</code></pre>

<hr>
<h2 id='calculate_confidence_limit_values'>Calculate the value of confidence limit</h2><span id='topic+calculate_confidence_limit_values'></span>

<h3>Description</h3>

<p>Calculates confidence limit values for prepared dataset, based on chosen parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_confidence_limit_values(
  calc_dat,
  confidence_limit = 0.98,
  theoretical = FALSE,
  relative = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculate_confidence_limit_values_+3A_calc_dat">calc_dat</code></td>
<td>
<p>processed data from DynamX file - using prepare_dataset</p>
</td></tr>
<tr><td><code id="calculate_confidence_limit_values_+3A_confidence_limit">confidence_limit</code></td>
<td>
<p>confidence limit chosen by user - from range [0, 1].</p>
</td></tr>
<tr><td><code id="calculate_confidence_limit_values_+3A_theoretical">theoretical</code></td>
<td>
<p>logical value to determine if plot is theoretical or not.</p>
</td></tr>
<tr><td><code id="calculate_confidence_limit_values_+3A_relative">relative</code></td>
<td>
<p>logical value to determine if values are relative or absolute.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>...
</p>


<h3>Value</h3>

<p>range of confidence limit interval
</p>


<h3>References</h3>

<p>Houde, D., Berkowitz, S.A., and Engen, J.R. (2011). 
The Utility of Hydrogen/Deuterium Exchange Mass Spectrometry in 
Biopharmaceutical Comparability Studies. J Pharm Sci 100, 2071–2086.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_hdx">read_hdx</a></code> <code><a href="#topic+prepare_dataset">prepare_dataset</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example data
dat &lt;- read_hdx(system.file(package = "HaDeX", "HaDeX/data/KD_180110_CD160_HVEM.csv"))

# prepare dataset for states `CD160` and `CD160_HVEM` in given time parameters 
calc_dat &lt;- prepare_dataset(dat,
                            in_state_first = "CD160_0.001",
                            chosen_state_first = "CD160_1",
                            out_state_first = "CD160_1440",
                            in_state_second = "CD160_HVEM_0.001",
                            chosen_state_second = "CD160_HVEM_1",
                            out_state_second = "CD160_HVEM_1440")   
                            
# calculates confidence limits for prepared data  
calculate_confidence_limit_values(calc_dat = calc_dat,
                                  confidence_limit = 0.99,
                                  theoretical = FALSE, 
                                  relative = TRUE)                          
                              
</code></pre>

<hr>
<h2 id='calculate_kinetics'>Calculate kinetic data</h2><span id='topic+calculate_kinetics'></span>

<h3>Description</h3>

<p>Calculate kinetics of the hydrogen-deuteration exchange 
for given peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_kinetics(
  dat,
  protein = dat[["Protein"]][1],
  sequence,
  state,
  start,
  end,
  time_in,
  time_out,
  deut_part = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculate_kinetics_+3A_dat">dat</code></td>
<td>
<p>dat data read by <code><a href="#topic+read_hdx">read_hdx</a></code></p>
</td></tr>
<tr><td><code id="calculate_kinetics_+3A_protein">protein</code></td>
<td>
<p>protein value for chosen peptide</p>
</td></tr>
<tr><td><code id="calculate_kinetics_+3A_sequence">sequence</code></td>
<td>
<p>sequence of the peptide for which the kinetics is calculated</p>
</td></tr>
<tr><td><code id="calculate_kinetics_+3A_state">state</code></td>
<td>
<p>state of given sequence</p>
</td></tr>
<tr><td><code id="calculate_kinetics_+3A_start">start</code></td>
<td>
<p>end of given sequence</p>
</td></tr>
<tr><td><code id="calculate_kinetics_+3A_end">end</code></td>
<td>
<p>end of given sequence</p>
</td></tr>
<tr><td><code id="calculate_kinetics_+3A_time_in">time_in</code></td>
<td>
<p>time in for experimental calculations</p>
</td></tr>
<tr><td><code id="calculate_kinetics_+3A_time_out">time_out</code></td>
<td>
<p>time out for experimental calculations</p>
</td></tr>
<tr><td><code id="calculate_kinetics_+3A_deut_part">deut_part</code></td>
<td>
<p>percentage of deuterium the protein was exposed to, value in range [0, 1]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates deuteration data for all available data points 
for given peptide. 
All four variants (relative &amp; theoretical combinations) of deuteration computations 
are supported. Manual correction of percentage of deuterium the protein was exposed 
to during the exchange in theoretical calculations is provided.
To visualize obtained data we recommend using <code><a href="#topic+plot_kinetics">plot_kinetics</a></code> function.
The first version doesn't support filled Modification and Fragment columns.
</p>


<h3>Value</h3>

<p>data frame with deuteration calculated for all the data points 
between time_in and time_out. 
The chosen time point for which deuteration in all four variants is calculated 
is available in column 'time_chosen'. The rest of
the returned structure is equivalent to structure returned by
<code><a href="#topic+calculate_state_deuteration">calculate_state_deuteration</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_hdx">read_hdx</a></code> <code><a href="#topic+calculate_state_deuteration">calculate_state_deuteration</a></code> <code><a href="#topic+plot_kinetics">plot_kinetics</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example data
dat &lt;- read_hdx(system.file(package = "HaDeX", 
                            "HaDeX/data/KD_180110_CD160_HVEM.csv"))

# calculate data for sequence INITSSASQEGTRLN in state CD160
(kin1 &lt;- calculate_kinetics(dat, 
                   protein = "db_CD160",
                   sequence = "INITSSASQEGTRLN", 
                   state = "CD160",
                   start = 1, 
                   end = 15,
                   time_in = 0.001, 
                   time_out = 1440))
                   
# calculate data for sequence INITSSASQEGTRLN in state CD160_HVEM
(kin2 &lt;- calculate_kinetics(dat, 
                   protein = "db_CD160",
                   sequence = "INITSSASQEGTRLN", 
                   state = "CD160_HVEM",
                   start = 1, 
                   end = 15,
                   time_in = 0.001, 
                   time_out = 1440))
                   
# load extra libraries
library(dplyr)
library(ggplot2)

# plot example - experimental and relative 
bind_rows(kin1, kin2) %&gt;% 
 plot_kinetics(theoretical = FALSE, 
               relative = TRUE) +
 labs(title = "Kinetic plot for INITSSASQEGTRLN")
 
# plot example - theoretical and absolute
bind_rows(kin1, kin2) %&gt;%
  plot_kinetics(theoretical = TRUE, 
               relative = FALSE) +
  labs(title = "Theoretical kinetics plot for INITSSASQEGTRLN")
       

</code></pre>

<hr>
<h2 id='calculate_state_deuteration'>Calculate deuteration</h2><span id='topic+calculate_state_deuteration'></span>

<h3>Description</h3>

<p>Calculates deuteration uptake based on supplied parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_state_deuteration(
  dat,
  protein,
  state,
  time_in,
  time_chosen,
  time_out,
  deut_part = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculate_state_deuteration_+3A_dat">dat</code></td>
<td>
<p>data as imported by the <code><a href="#topic+read_hdx">read_hdx</a></code> function</p>
</td></tr>
<tr><td><code id="calculate_state_deuteration_+3A_protein">protein</code></td>
<td>
<p>protein included in calculations</p>
</td></tr>
<tr><td><code id="calculate_state_deuteration_+3A_state">state</code></td>
<td>
<p>state included in calculations</p>
</td></tr>
<tr><td><code id="calculate_state_deuteration_+3A_time_in">time_in</code></td>
<td>
<p>experimental 'time_in'</p>
</td></tr>
<tr><td><code id="calculate_state_deuteration_+3A_time_chosen">time_chosen</code></td>
<td>
<p>chosen time point</p>
</td></tr>
<tr><td><code id="calculate_state_deuteration_+3A_time_out">time_out</code></td>
<td>
<p>experimental 'time_out'</p>
</td></tr>
<tr><td><code id="calculate_state_deuteration_+3A_deut_part">deut_part</code></td>
<td>
<p>percentage of deuterium the protein was exposed to, value in range [0, 1]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>calculate_state_deuteration</code> calculates deuteration for peptides in given protein in given state based
on supplied parameters: 'time_in', 'time_out' and 'time_chosen'. All four variants (combinations of theoretical &amp; relative) are 
supplied (mean values and uncertainty). Manual correction of percentage of deuterium the protein was exposed to during the exchange
in theoretical calculations is provided. 
</p>
<p>Methods of calculation and uncertainty are profoundly discussed in the vignette.
</p>


<h3>Value</h3>

<p>a <code><a href="base.html#topic+data.frame">data.frame</a></code> object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_hdx">read_hdx</a></code> <code><a href="#topic+calculate_confidence_limit_values">calculate_confidence_limit_values</a></code> <code><a href="#topic+add_stat_dependency">add_stat_dependency</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example data
dat &lt;- read_hdx(system.file(package = "HaDeX", "HaDeX/data/KD_180110_CD160_HVEM.csv"))

# calculate deuteration for state "CD160"
calculate_state_deuteration(dat, protein = "db_CD160", state = "CD160",
                            time_in = 0, time_chosen = 5.000, time_out = 1440.000)

</code></pre>

<hr>
<h2 id='comparison_plot'>Plot comparison plot</h2><span id='topic+comparison_plot'></span>

<h3>Description</h3>

<p>Produces comparison_plot based on previously processed data - theoretical or experimental. User can change labels if needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comparison_plot(
  calc_dat,
  theoretical = FALSE,
  relative = TRUE,
  state_first = "state_first",
  state_second = "state_second"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="comparison_plot_+3A_calc_dat">calc_dat</code></td>
<td>
<p>processed data from DynamX file - using prepare_dataset</p>
</td></tr>
<tr><td><code id="comparison_plot_+3A_theoretical">theoretical</code></td>
<td>
<p>logical value to determine if plot is theoretical or not. default : false</p>
</td></tr>
<tr><td><code id="comparison_plot_+3A_relative">relative</code></td>
<td>
<p>logical value to determine if values are relative or absolute. default : true</p>
</td></tr>
<tr><td><code id="comparison_plot_+3A_state_first">state_first</code></td>
<td>
<p>first state name</p>
</td></tr>
<tr><td><code id="comparison_plot_+3A_state_second">state_second</code></td>
<td>
<p>second state name</p>
</td></tr>
</table>


<h3>Details</h3>

<p>...
</p>
<p>This is the first version - multi-state calculations are not supported.
</p>


<h3>Value</h3>

<p>a <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_hdx">read_hdx</a></code> <code><a href="#topic+prepare_dataset">prepare_dataset</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example data
dat &lt;- read_hdx(system.file(package = "HaDeX", "HaDeX/data/KD_180110_CD160_HVEM.csv"))

# prepare dataset for states `CD160` and `CD160_HVEM` in given time parameters
calc_dat &lt;- prepare_dataset(dat,
                            in_state_first = "CD160_0.001",
                            chosen_state_first = "CD160_1",
                            out_state_first = "CD160_1440",
                            in_state_second = "CD160_HVEM_0.001",
                            chosen_state_second = "CD160_HVEM_1",
                            out_state_second = "CD160_HVEM_1440")       

# plot comparison plot - theoretical &amp; relative                      
comparison_plot(calc_dat = calc_dat,
                theoretical = TRUE,
                relative = TRUE,
                state_first = "CD160",
                state_second = "CD160_HVEM")
                
# plot comparison plot - experimental &amp; relative                      
comparison_plot(calc_dat = calc_dat,
                theoretical = FALSE,
                relative = TRUE,
                state_first = "CD160",
                state_second = "CD160_HVEM")  
                
# plot comparison plot - theoretical &amp; absolute                      
comparison_plot(calc_dat = calc_dat,
                theoretical = TRUE,
                relative = FALSE,
                state_first = "CD160",
                state_second = "CD160_HVEM")

# plot comparison plot - experimental &amp; absolute                      
comparison_plot(calc_dat = calc_dat,
                theoretical = FALSE,
                relative = FALSE,
                state_first = "CD160",
                state_second = "CD160_HVEM")                

</code></pre>

<hr>
<h2 id='HaDeX_gui'>HaDeX Graphical User Interface</h2><span id='topic+HaDeX_gui'></span>

<h3>Description</h3>

<p>Launches graphical user interface.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HaDeX_gui(port = getOption("shiny.port"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HaDeX_gui_+3A_port">port</code></td>
<td>
<p>The TCP port. See <code><a href="shiny.html#topic+runApp">runApp</a></code>.</p>
</td></tr>
</table>


<h3>Warning </h3>

<p>Any ad-blocking software may cause malfunctions.
</p>

<hr>
<h2 id='plot_coverage'>Plot peptide coverage</h2><span id='topic+plot_coverage'></span>

<h3>Description</h3>

<p>Plots the peptide coverage of the protein sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_coverage(
  dat,
  protein = dat[["Protein"]][1],
  chosen_state = dat[["State"]][1]
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_coverage_+3A_dat">dat</code></td>
<td>
<p>data as imported by the <code><a href="#topic+read_hdx">read_hdx</a></code> function</p>
</td></tr>
<tr><td><code id="plot_coverage_+3A_protein">protein</code></td>
<td>
<p>protein to be included in plot</p>
</td></tr>
<tr><td><code id="plot_coverage_+3A_chosen_state">chosen_state</code></td>
<td>
<p>sequence states to be included in plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>plot_coverage</code> plots sequence coverage based on experimental data for chosen protein in chosen state. 
Only non-duplicated peptides are shown on the plot, next to each other. 
</p>
<p>The aim of this plot is to see the dependence between positions of the peptides on the protein sequence. Their position in y-axis 
does not contain any information.
</p>


<h3>Value</h3>

<p>a <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_hdx">read_hdx</a></code> <code><a href="#topic+plot_position_frequency">plot_position_frequency</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example data
dat &lt;- read_hdx(system.file(package = "HaDeX", 
                            "HaDeX/data/KD_180110_CD160_HVEM.csv"))
                            
# plot coverage with default parameters
plot_coverage(dat)

# plot coverage with explicit parameters
plot_coverage(dat, protein = "db_CD160", chosen_state = "CD160_HVEM")

</code></pre>

<hr>
<h2 id='plot_kinetics'>Plot kinetics data</h2><span id='topic+plot_kinetics'></span>

<h3>Description</h3>

<p>Plots kinetics of the hydrogen-deuterium exchange for specific peptides.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_kinetics(kin_dat, theoretical = FALSE, relative = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_kinetics_+3A_kin_dat">kin_dat</code></td>
<td>
<p>calculated kinetic data by <code><a href="#topic+calculate_kinetics">calculate_kinetics</a></code> 
function</p>
</td></tr>
<tr><td><code id="plot_kinetics_+3A_theoretical">theoretical</code></td>
<td>
<p><code>logical</code>, determines if plot shows theoretical values</p>
</td></tr>
<tr><td><code id="plot_kinetics_+3A_relative">relative</code></td>
<td>
<p><code>logical</code>, determines if values are relative or absolute</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualises the output of the  
<code><a href="#topic+calculate_kinetics">calculate_kinetics</a></code> function. 
Based on supplied parameters appropriate columns are chosen for the plot. 
The uncertainty associated with each peptide is shown as a ribbon. 
Axis are labeled according to the supplied parameters but no title is provided.
</p>
<p>If you want to plot data for more then one peptide in one state, join 
calculated data by using <code>bind_rows</code> from dplyr package and 
pass the result as kin_dat.
</p>


<h3>Value</h3>

<p>a <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculate_kinetics">calculate_kinetics</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example data
dat &lt;- read_hdx(system.file(package = "HaDeX", "HaDeX/data/KD_180110_CD160_HVEM.csv"))

# calculate data for the sequence INITSSASQEGTRLN in the CD160 state 
(kin1 &lt;- calculate_kinetics(dat, 
                           protein = "db_CD160",
                           sequence = "INITSSASQEGTRLN", 
                           state = "CD160",
                           start = 1, 
                           end = 15,
                           time_in = 0.001, 
                           time_out = 1440))

# calculate data for the sequence INITSSASQEGTRLN in the CD160_HVEM state 
(kin2 &lt;- calculate_kinetics(dat, 
                           protein = "db_CD160",
                           sequence = "INITSSASQEGTRLN", 
                           state = "CD160_HVEM",
                           start = 1, 
                           end = 15,
                           time_in = 0.001, 
                           time_out = 1440)) 
                         
# load extra packages 
library(dplyr)
  
# plot a single peptide - theoretical and relative
plot_kinetics(kin_dat = kin1, 
              theoretical = TRUE, 
              relative = TRUE)
                
# plot joined data - experimental and absolute
bind_rows(kin1, kin2) %&gt;%
  plot_kinetics(theoretical = FALSE, 
                relative = FALSE)
                
</code></pre>

<hr>
<h2 id='plot_position_frequency'>Plot position frequency</h2><span id='topic+plot_position_frequency'></span>

<h3>Description</h3>

<p>Plots the frequency of coverage of protein sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_position_frequency(
  dat,
  protein = dat[["Protein"]][1],
  chosen_state = dat[["State"]][1]
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_position_frequency_+3A_dat">dat</code></td>
<td>
<p>data as imported by the <code><a href="#topic+read_hdx">read_hdx</a></code> function</p>
</td></tr>
<tr><td><code id="plot_position_frequency_+3A_protein">protein</code></td>
<td>
<p>protein to be included in plot</p>
</td></tr>
<tr><td><code id="plot_position_frequency_+3A_chosen_state">chosen_state</code></td>
<td>
<p>sequence states to be included in plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>plot_position_frequency</code> plots a histogram of the coverage frequency based on experimental data.
The aim of this plot is to see how many times each position of the sequence was covered (by different peptides).
</p>


<h3>Value</h3>

<p>a <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_hdx">read_hdx</a></code> <code><a href="#topic+plot_coverage">plot_coverage</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example data
dat &lt;- read_hdx(system.file(package = "HaDeX", 
                            "HaDeX/data/KD_180110_CD160_HVEM.csv"))
                            
# function call with default parameters
plot_position_frequency(dat)

# function call with explicit parameters
plot_position_frequency(dat, chosen_state = "CD160_HVEM")

</code></pre>

<hr>
<h2 id='prepare_dataset'>Calculate data</h2><span id='topic+prepare_dataset'></span>

<h3>Description</h3>

<p>Calculates values for visualization  from input data file - both experimental and theoretical. All parameters are needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_dataset(
  dat,
  in_state_first,
  chosen_state_first,
  out_state_first,
  in_state_second,
  chosen_state_second,
  out_state_second
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prepare_dataset_+3A_dat">dat</code></td>
<td>
<p>data frame with data from DynamX file</p>
</td></tr>
<tr><td><code id="prepare_dataset_+3A_in_state_first">in_state_first</code></td>
<td>
<p>string in form &quot;state_time&quot; for first state in in time</p>
</td></tr>
<tr><td><code id="prepare_dataset_+3A_chosen_state_first">chosen_state_first</code></td>
<td>
<p>string in form &quot;state_time&quot; for chosen state in in time</p>
</td></tr>
<tr><td><code id="prepare_dataset_+3A_out_state_first">out_state_first</code></td>
<td>
<p>string in form &quot;state_time&quot; for first state in out time</p>
</td></tr>
<tr><td><code id="prepare_dataset_+3A_in_state_second">in_state_second</code></td>
<td>
<p>string in form &quot;state_time&quot; for second state in in time</p>
</td></tr>
<tr><td><code id="prepare_dataset_+3A_chosen_state_second">chosen_state_second</code></td>
<td>
<p>string in form &quot;state_time&quot; for second state in chosen time</p>
</td></tr>
<tr><td><code id="prepare_dataset_+3A_out_state_second">out_state_second</code></td>
<td>
<p>string in form &quot;state_time&quot; for second state in out time</p>
</td></tr>
</table>


<h3>Details</h3>

<p>...
</p>
<p>This is the first version - multi-state calculations are not supported.
</p>


<h3>Value</h3>

<p>data frame with calculated values
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_hdx">read_hdx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example data
dat &lt;- read_hdx(system.file(package = "HaDeX", "HaDeX/data/KD_180110_CD160_HVEM.csv"))

# prepare dataset for states `CD160` and `CD160_HVEM` in given time parameters
prepare_dataset(dat,
                in_state_first = "CD160_0.001",
                chosen_state_first = "CD160_1",
                out_state_first = "CD160_1440",
                in_state_second = "CD160_HVEM_0.001",
                chosen_state_second = "CD160_HVEM_1",
                out_state_second = "CD160_HVEM_1440") 

</code></pre>

<hr>
<h2 id='quality_control'>Experiment quality control</h2><span id='topic+quality_control'></span>

<h3>Description</h3>

<p>Checks how the uncertainty changes in a function of 'out_time'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quality_control(dat, state_first, state_second, chosen_time, in_time)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="quality_control_+3A_dat">dat</code></td>
<td>
<p>data read by <code><a href="#topic+read_hdx">read_hdx</a></code></p>
</td></tr>
<tr><td><code id="quality_control_+3A_state_first">state_first</code></td>
<td>
<p>state of the first peptide</p>
</td></tr>
<tr><td><code id="quality_control_+3A_state_second">state_second</code></td>
<td>
<p>state of the second peptide</p>
</td></tr>
<tr><td><code id="quality_control_+3A_chosen_time">chosen_time</code></td>
<td>
<p>chosen time point</p>
</td></tr>
<tr><td><code id="quality_control_+3A_in_time">in_time</code></td>
<td>
<p>'in' time</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates mean uncertainty of all peptides and its uncertainty (standard error) based on given 'in_time' and 'chosen_time' 
as a function of 'out_time'. Both theoretical and experimental results for each state and their difference are supplied for comparison but only 
experimental calculations depends on 'out_time' variable. The results are either in form of relative or absolute values depending on the 'relative' 
parameter supplied by the user. 
This data can be useful for general overview of the experiment and analyse of the chosen time parameters.
</p>


<h3>Value</h3>

<p><code>data.frame</code> with mean uncertainty per different 'out_time' value
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_hdx">read_hdx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example data
dat &lt;- read_hdx(system.file(package = "HaDeX", "HaDeX/data/KD_180110_CD160_HVEM.csv"))

# calculate mean uncertainty 
(result &lt;- quality_control(dat = dat,
                           state_first = "CD160",
                           state_second = "CD160_HVEM", 
                           chosen_time = 1, 
                           in_time = 0.001))    
                           
# load extra libraries
library(ggplot2)
library(tidyr)
library(dplyr)

# example of data visualization 
gather(result, 2:7, key = 'type', value = 'value') %&gt;%
filter(startsWith(type, "avg")) %&gt;%
  ggplot(aes(x = factor(out_time), y = value, group = type)) +
  geom_line(aes(color = type)) +
  labs(x = "Out time", 
       y = "Mean uncertainty")

</code></pre>

<hr>
<h2 id='read_hdx'>Read HDX-MS data file</h2><span id='topic+read_hdx'></span>

<h3>Description</h3>

<p>Imports data from a HDX-MS file and validates its content.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_hdx(filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_hdx_+3A_filename">filename</code></td>
<td>
<p>a file supplied by the user. Formats allowed: .csv, .xlsx and .xls.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>First version accepts files produced by DynamX 3.0 and 2.0 in 'cluster data' format. 
The function checks if all necessary columns are provided in correct format. The file must 
include at least two repetitions of the measurement for the uncertainty to be calculated.
</p>


<h3>Value</h3>

<p><code>dat</code> - a <code><a href="base.html#topic+data.frame">data.frame</a></code> with validated content.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculate_kinetics">calculate_kinetics</a></code> <code><a href="#topic+calculate_state_deuteration">calculate_state_deuteration</a></code> <code><a href="#topic+plot_coverage">plot_coverage</a></code> <code><a href="#topic+plot_position_frequency">plot_position_frequency</a></code>
<code><a href="#topic+prepare_dataset">prepare_dataset</a></code> <code><a href="#topic+quality_control">quality_control</a></code> <code><a href="#topic+reconstruct_sequence">reconstruct_sequence</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read example data
head(read_hdx(system.file(package = "HaDeX", 
                     "HaDeX/data/KD_180110_CD160_HVEM.csv")))

</code></pre>

<hr>
<h2 id='reconstruct_sequence'>Reconstruct protein sequence</h2><span id='topic+reconstruct_sequence'></span>

<h3>Description</h3>

<p>Reconstructs protein sequence from supplied file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reconstruct_sequence(dat, protein = dat[["Protein"]][1])
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reconstruct_sequence_+3A_dat">dat</code></td>
<td>
<p>data read by <code><a href="#topic+read_hdx">read_hdx</a></code></p>
</td></tr>
<tr><td><code id="reconstruct_sequence_+3A_protein">protein</code></td>
<td>
<p>the protein of which the structure is to be reconstructed</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function reconstructs protein sequence from supplied experimental data. If a position is not covered, x is shown.
First version doesn't support manual sequence length correction.
</p>


<h3>Value</h3>

<p>reconstructed sequence - <code>character</code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_hdx">read_hdx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- read_hdx(system.file(package = "HaDeX", "HaDeX/data/KD_180110_CD160_HVEM.csv"))
reconstruct_sequence(dat)

</code></pre>

<hr>
<h2 id='woods_plot'>Plot Woods' plot</h2><span id='topic+woods_plot'></span>

<h3>Description</h3>

<p>Produces Woods' plot based on theoretical or experimental HDX-MS data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>woods_plot(
  calc_dat,
  theoretical = FALSE,
  relative = TRUE,
  confidence_limit = 0.98,
  confidence_limit_2 = 0.99
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="woods_plot_+3A_calc_dat">calc_dat</code></td>
<td>
<p>data as imported by the <code><a href="#topic+read_hdx">read_hdx</a></code> function and 
processed by the <code><a href="#topic+prepare_dataset">prepare_dataset</a></code> function.</p>
</td></tr>
<tr><td><code id="woods_plot_+3A_theoretical">theoretical</code></td>
<td>
<p><code>logical</code>, determines if plot shows theoretical values.</p>
</td></tr>
<tr><td><code id="woods_plot_+3A_relative">relative</code></td>
<td>
<p><code>logical</code>, determines if values are relative or absolute.</p>
</td></tr>
<tr><td><code id="woods_plot_+3A_confidence_limit">confidence_limit</code></td>
<td>
<p>confidence limit.</p>
</td></tr>
<tr><td><code id="woods_plot_+3A_confidence_limit_2">confidence_limit_2</code></td>
<td>
<p>confidence limit 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>...
</p>
<p>This is the first version - multi-state calculations are not supported.
</p>


<h3>Value</h3>

<p>a <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code> object.
</p>


<h3>References</h3>

<p>Woods, V.L., and Hamuro, Y. (2001). High resolution, 
high-throughput amide deuterium exchange-mass spectrometry (DXMS) 
determination of protein binding site structure and dynamics: utility 
in pharmaceutical design. J. Cell. Biochem. Suppl. Suppl 37, 89–98.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read_hdx">read_hdx</a></code> <code><a href="#topic+prepare_dataset">prepare_dataset</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example data
dat &lt;- read_hdx(system.file(package = "HaDeX", 
                            "HaDeX/data/KD_180110_CD160_HVEM.csv"))
                            
# prepare dataset for states `CD160` and `CD160_HVEM` 
# in given time parameters                           
calc_dat &lt;- prepare_dataset(dat,
                            in_state_first = "CD160_0.001",
                            chosen_state_first = "CD160_1",
                            out_state_first = "CD160_1440",
                            in_state_second = "CD160_HVEM_0.001",
                            chosen_state_second = "CD160_HVEM_1",
                            out_state_second = "CD160_HVEM_1440") 
                            
# plot Woods plot - theoretical &amp; relative                             
woods_plot(calc_dat = calc_dat,
           theoretical = TRUE,
           relative = TRUE,
           confidence_limit = 0.98,
           confidence_limit_2 = 0.99)
           
# plot Woods plot - experimental  &amp; relative                
woods_plot(calc_dat = calc_dat,
           theoretical = FALSE,
           relative = TRUE,
           confidence_limit = 0.98,
           confidence_limit_2 = 0.99)

# plot Woods plot - theoretical &amp; absolute    
woods_plot(calc_dat = calc_dat,
           theoretical = TRUE,
           relative = FALSE,
           confidence_limit = 0.98,
           confidence_limit_2 = 0.99)

# plot Woods plot - experimental &amp; absolute    
woods_plot(calc_dat = calc_dat,
           theoretical = FALSE,
           relative = FALSE,
           confidence_limit = 0.98,
           confidence_limit_2 = 0.99)
            
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
