<!DOCTYPE html><html><head><title>Help for package hahmmr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hahmmr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#acen_hg19'><p>centromere regions (hg19)</p></a></li>
<li><a href='#acen_hg38'><p>centromere regions (hg38)</p></a></li>
<li><a href='#analyze_allele'><p>Analyze allele profile</p></a></li>
<li><a href='#analyze_joint'><p>Analyze allele and expression profile</p></a></li>
<li><a href='#annot_cm'><p>Annotate genetic distance between markers</p></a></li>
<li><a href='#annot_segs'><p>Annotate copy number segments after HMM decoding</p></a></li>
<li><a href='#approx_phi_post'><p>Laplace approximation of the posterior of expression fold change phi</p></a></li>
<li><a href='#approx_theta_post_s2'><p>Find optimal theta of a CNV segment using forward-backward; uses a 2-state allele HMM</p></a></li>
<li><a href='#approx_theta_post_s3'><p>Find optimal theta of a chromosome using forward-backward; uses a 3-state allele HMM</p></a></li>
<li><a href='#bulk_example'><p>example pseudobulk dataframe</p></a></li>
<li><a href='#calc_allele_lik_s2'><p>Calculate allele likelihoods for 2-state allele HMM</p></a></li>
<li><a href='#calc_allele_lik_s3'><p>Calculate allele likelihoods for 3-state allele HMM</p></a></li>
<li><a href='#calc_allele_LLR'><p>Calculate LLR for an allele HMM</p></a></li>
<li><a href='#calc_exp_LLR'><p>Calculate LLR for an expression HMM</p></a></li>
<li><a href='#calc_trans_mat_s15'><p>Calculate the transition matrix for 15-state joint HMM</p></a></li>
<li><a href='#calc_trans_mat_s7'><p>Calculate the transition matrix for 7-state joint HMM</p></a></li>
<li><a href='#check_cols'><p>Check if columns are present in dataframe</p></a></li>
<li><a href='#check_matrix'><p>Check the format of a count matrix</p></a></li>
<li><a href='#chrom_sizes_hg19'><p>chromosome sizes (hg19)</p></a></li>
<li><a href='#chrom_sizes_hg38'><p>chromosome sizes (hg38)</p></a></li>
<li><a href='#classify_alleles'><p>classify alleles using viterbi and forward-backward</p></a></li>
<li><a href='#combine_bulk'><p>Combine allele and expression pseudobulks</p></a></li>
<li><a href='#dbbinom'><p>Beta-binomial distribution density function</p>
A distribution is beta-binomial if p, the probability of success,
in a binomial distribution has a beta distribution with shape
parameters alpha &gt; 0 and beta &gt; 0
For more details, see extraDistr::dbbinom</a></li>
<li><a href='#df_allele_example'><p>example allele count dataframe</p></a></li>
<li><a href='#dpoilog'><p>Returns the density for the Poisson lognormal distribution with parameters mu and sig</p></a></li>
<li><a href='#filter_genes'><p>filter for mutually expressed genes</p></a></li>
<li><a href='#fit_gamma'><p>fit gamma maximum likelihood</p></a></li>
<li><a href='#fit_lnpois_cpp'><p>Fit MLE of log-normal Poisson model</p></a></li>
<li><a href='#fit_ref_sse'><p>Fit a reference profile from multiple references using constrained least square</p></a></li>
<li><a href='#forward_back_allele'><p>Forward-backward algorithm for allele HMM</p></a></li>
<li><a href='#gaps_hg19'><p>genome gap regions (hg19)</p></a></li>
<li><a href='#gaps_hg38'><p>genome gap regions (hg38)</p></a></li>
<li><a href='#gene_counts_example'><p>example gene expression counts matrix</p></a></li>
<li><a href='#generate_postfix'><p>Generate alphabetical postfixes</p></a></li>
<li><a href='#get_allele_bulk'><p>Aggregate into pseudobulk alelle profile</p></a></li>
<li><a href='#get_allele_hmm_s2'><p>Make a 2-state allele HMM - no transitions to netural state</p></a></li>
<li><a href='#get_allele_hmm_s3'><p>Make a 3-state allele HMM - allow transitions to netural state</p></a></li>
<li><a href='#get_bulk'><p>Produce combined bulk expression and allele profile</p></a></li>
<li><a href='#get_exp_bulk'><p>Aggregate into bulk expression profile</p></a></li>
<li><a href='#get_trans_probs_s15'><p>Helper function to calculate transition porbabilities for 15-state joint HMM</p>
cn/phase are sclars, only p_s is vectorized</a></li>
<li><a href='#get_trans_probs_s7'><p>Helper function to calculate transition porbabilities for 7-state joint HMM</p>
cn/phase are sclars, only p_s is vectorized</a></li>
<li><a href='#gtf_hg19'><p>gene model (hg19)</p></a></li>
<li><a href='#gtf_hg38'><p>gene model (hg38)</p></a></li>
<li><a href='#gtf_mm10'><p>gene model (mm10)</p></a></li>
<li><a href='#l_bbinom'><p>calculate joint likelihood of allele data</p></a></li>
<li><a href='#l_lnpois'><p>calculate joint likelihood of a PLN model</p></a></li>
<li><a href='#likelihood_allele'><p>Only compute total log likelihood from an allele HMM</p></a></li>
<li><a href='#logSumExp'><p>logSumExp function</p></a></li>
<li><a href='#plot_bulks'><p>Plot a group of pseudobulk HMM profiles</p></a></li>
<li><a href='#plot_psbulk'><p>Plot a pseudobulk HMM profile</p></a></li>
<li><a href='#pre_likelihood_hmm'><p>HMM object for unit tests</p></a></li>
<li><a href='#ref_hca'><p>reference expression magnitudes from HCA</p></a></li>
<li><a href='#ref_hca_counts'><p>reference expression counts from HCA</p></a></li>
<li><a href='#run_allele_hmm_s3'><p>Run a 3-state allele HMM</p></a></li>
<li><a href='#run_allele_hmm_s5'><p>Run a 5-state allele-only HMM - two theta levels</p></a></li>
<li><a href='#run_joint_hmm_s15'><p>Run 15-state joint HMM on a pseudobulk profile</p></a></li>
<li><a href='#run_joint_hmm_s7'><p>Run 7-state joint HMM on a pseudobulk profile</p></a></li>
<li><a href='#segs_example'><p>example CNV segments dataframe</p></a></li>
<li><a href='#smooth_segs'><p>Smooth the segments after HMM decoding</p></a></li>
<li><a href='#switch_prob_cm'><p>predict phase switch probablity as a function of genetic distance</p></a></li>
<li><a href='#theta_hat_seg'><p>Estimate of imbalance level theta in a segment</p></a></li>
<li><a href='#vcf_meta'><p>example VCF header</p></a></li>
<li><a href='#viterbi_allele'><p>Viterbi algorithm for allele HMM</p></a></li>
<li><a href='#viterbi_joint'><p>Viterbi algorithm for joint HMM, can only handle one set of observations</p></a></li>
<li><a href='#viterbi_joint_mat'><p>Generalized viterbi algorithm for joint HMM, can handle multiple sets of observations</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Haplotype-Aware Hidden Markov Model for RNA</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Haplotype-aware Hidden Markov Model for RNA (HaHMMR) is a method for detecting copy number variations (CNVs) from bulk RNA-seq data. Additional examples, documentations, and details on the method are available at <a href="https://github.com/kharchenkolab/hahmmr/">https://github.com/kharchenkolab/hahmmr/</a>.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, dplyr, GenomicRanges, ggplot2, glue, IRanges,
methods, patchwork, Rcpp, stringr, tibble, zoo</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggrastr, testthat</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, roptim</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-25 14:22:02 UTC; tenggao</td>
</tr>
<tr>
<td>Author:</td>
<td>Teng Gao <a href="https://orcid.org/0000-0002-0196-689X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Evan Biederstedt [aut],
  Peter Kharchenko [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Teng Gao &lt;tgaoteng@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-25 18:00:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='acen_hg19'>centromere regions (hg19)</h2><span id='topic+acen_hg19'></span>

<h3>Description</h3>

<p>centromere regions (hg19)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>acen_hg19
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 22 rows and 3 columns.
</p>

<hr>
<h2 id='acen_hg38'>centromere regions (hg38)</h2><span id='topic+acen_hg38'></span>

<h3>Description</h3>

<p>centromere regions (hg38)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>acen_hg38
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 22 rows and 3 columns.
</p>

<hr>
<h2 id='analyze_allele'>Analyze allele profile</h2><span id='topic+analyze_allele'></span>

<h3>Description</h3>

<p>Analyze allele profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyze_allele(
  bulk,
  t = 1e-05,
  theta_min = 0.08,
  gamma = 20,
  nu = 0.5,
  r = 0.015,
  hmm = "S5",
  fit_theta = FALSE,
  fit_gamma = FALSE,
  theta_start = 0.05,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="analyze_allele_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Bulk allele profile</p>
</td></tr>
<tr><td><code id="analyze_allele_+3A_t">t</code></td>
<td>
<p>numeric Transition probability</p>
</td></tr>
<tr><td><code id="analyze_allele_+3A_theta_min">theta_min</code></td>
<td>
<p>numeric Minimum allele fraction</p>
</td></tr>
<tr><td><code id="analyze_allele_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion parameter</p>
</td></tr>
<tr><td><code id="analyze_allele_+3A_nu">nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td></tr>
<tr><td><code id="analyze_allele_+3A_r">r</code></td>
<td>
<p>numeric Alternative allele count bias</p>
</td></tr>
<tr><td><code id="analyze_allele_+3A_hmm">hmm</code></td>
<td>
<p>character HMM model to use (S3 or S5)</p>
</td></tr>
<tr><td><code id="analyze_allele_+3A_fit_theta">fit_theta</code></td>
<td>
<p>logical Whether to fit theta_min</p>
</td></tr>
<tr><td><code id="analyze_allele_+3A_fit_gamma">fit_gamma</code></td>
<td>
<p>logical Whether to fit gamma</p>
</td></tr>
<tr><td><code id="analyze_allele_+3A_theta_start">theta_start</code></td>
<td>
<p>numeric Starting value for theta_min</p>
</td></tr>
<tr><td><code id="analyze_allele_+3A_verbose">verbose</code></td>
<td>
<p>logical Whether to print progress</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Bulk allele profile with CNV states
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bulk_example = analyze_allele(bulk_example, hmm = 'S5')
</code></pre>

<hr>
<h2 id='analyze_joint'>Analyze allele and expression profile</h2><span id='topic+analyze_joint'></span>

<h3>Description</h3>

<p>Analyze allele and expression profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyze_joint(
  bulk,
  t = 1e-05,
  gamma = 20,
  theta_min = 0.08,
  logphi_min = 0.25,
  hmm = "S15",
  nu = 1,
  min_genes = 10,
  r = 0.015,
  theta_start = 0.05,
  exclude_neu = TRUE,
  fit_gamma = FALSE,
  fit_theta = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="analyze_joint_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Bulk allele and expression profile</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_t">t</code></td>
<td>
<p>numeric Transition probability</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion parameter</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_theta_min">theta_min</code></td>
<td>
<p>numeric Minimum allele fraction</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_logphi_min">logphi_min</code></td>
<td>
<p>numeric Minimum log2 fold change</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_hmm">hmm</code></td>
<td>
<p>character HMM model to use (S7 or S15)</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_nu">nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_min_genes">min_genes</code></td>
<td>
<p>integer Minimum number of genes per segment</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_r">r</code></td>
<td>
<p>numeric Alternative allele count bias</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_theta_start">theta_start</code></td>
<td>
<p>numeric Starting value for theta_min</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_exclude_neu">exclude_neu</code></td>
<td>
<p>logical Whether to exclude neutral segments in retest</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_fit_gamma">fit_gamma</code></td>
<td>
<p>logical Whether to fit gamma</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_fit_theta">fit_theta</code></td>
<td>
<p>logical Whether to fit theta_min</p>
</td></tr>
<tr><td><code id="analyze_joint_+3A_verbose">verbose</code></td>
<td>
<p>logical Whether to print progress</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Bulk allele and expression profile with CNV states
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bulk_example = analyze_joint(bulk_example, hmm = 'S15')
</code></pre>

<hr>
<h2 id='annot_cm'>Annotate genetic distance between markers</h2><span id='topic+annot_cm'></span>

<h3>Description</h3>

<p>Annotate genetic distance between markers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annot_cm(bulk, genetic_map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annot_cm_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
<tr><td><code id="annot_cm_+3A_genetic_map">genetic_map</code></td>
<td>
<p>dataframe Genetic map</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Annotated pseudobulk profile
</p>

<hr>
<h2 id='annot_segs'>Annotate copy number segments after HMM decoding</h2><span id='topic+annot_segs'></span>

<h3>Description</h3>

<p>Annotate copy number segments after HMM decoding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annot_segs(bulk, var = "cnv_state")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annot_segs_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a pseudobulk dataframe
</p>

<hr>
<h2 id='approx_phi_post'>Laplace approximation of the posterior of expression fold change phi</h2><span id='topic+approx_phi_post'></span>

<h3>Description</h3>

<p>Laplace approximation of the posterior of expression fold change phi
</p>


<h3>Usage</h3>

<pre><code class='language-R'>approx_phi_post(
  Y_obs,
  lambda_ref,
  d,
  alpha = NULL,
  beta = NULL,
  mu = NULL,
  sig = NULL,
  lower = 0.2,
  upper = 10,
  start = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="approx_phi_post_+3A_y_obs">Y_obs</code></td>
<td>
<p>numeric vector Gene expression counts</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_lambda_ref">lambda_ref</code></td>
<td>
<p>numeric vector Reference expression levels</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_d">d</code></td>
<td>
<p>numeric Total library size</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_alpha">alpha</code></td>
<td>
<p>numeric Shape parameter of the gamma distribution</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_beta">beta</code></td>
<td>
<p>numeric Rate parameter of the gamma distribution</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_mu">mu</code></td>
<td>
<p>numeric Mean of the normal distribution</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_sig">sig</code></td>
<td>
<p>numeric Standard deviation of the normal distribution</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_lower">lower</code></td>
<td>
<p>numeric Lower bound of phi</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_upper">upper</code></td>
<td>
<p>numeric Upper bound of phi</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_start">start</code></td>
<td>
<p>numeric Starting value of phi</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric MLE of phi and its standard deviation
</p>

<hr>
<h2 id='approx_theta_post_s2'>Find optimal theta of a CNV segment using forward-backward; uses a 2-state allele HMM</h2><span id='topic+approx_theta_post_s2'></span>

<h3>Description</h3>

<p>Find optimal theta of a CNV segment using forward-backward; uses a 2-state allele HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>approx_theta_post_s2(
  pAD,
  DP,
  R,
  p_s,
  upper = 0.499,
  start = 0.25,
  gamma = 20,
  r = 0.015
)
</code></pre>

<hr>
<h2 id='approx_theta_post_s3'>Find optimal theta of a chromosome using forward-backward; uses a 3-state allele HMM</h2><span id='topic+approx_theta_post_s3'></span>

<h3>Description</h3>

<p>Find optimal theta of a chromosome using forward-backward; uses a 3-state allele HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>approx_theta_post_s3(
  pAD,
  DP,
  R,
  p_s,
  t,
  upper = 0.45,
  start = 0.05,
  gamma = 20,
  r = 0.015
)
</code></pre>

<hr>
<h2 id='bulk_example'>example pseudobulk dataframe</h2><span id='topic+bulk_example'></span>

<h3>Description</h3>

<p>example pseudobulk dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bulk_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 10321 rows and 58 columns.
</p>

<hr>
<h2 id='calc_allele_lik_s2'>Calculate allele likelihoods for 2-state allele HMM</h2><span id='topic+calc_allele_lik_s2'></span>

<h3>Description</h3>

<p>Calculate allele likelihoods for 2-state allele HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_allele_lik_s2(pAD, DP, R, p_s, theta, gamma = 20, r = 0.015)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_allele_lik_s2_+3A_pad">pAD</code></td>
<td>
<p>integer vector Paternal allele counts</p>
</td></tr>
<tr><td><code id="calc_allele_lik_s2_+3A_dp">DP</code></td>
<td>
<p>integer vector Total alelle counts</p>
</td></tr>
<tr><td><code id="calc_allele_lik_s2_+3A_r">R</code></td>
<td>
<p>numeric vector Variant mapping bias direction</p>
</td></tr>
<tr><td><code id="calc_allele_lik_s2_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="calc_allele_lik_s2_+3A_theta">theta</code></td>
<td>
<p>numeric Haplotype imbalance</p>
</td></tr>
<tr><td><code id="calc_allele_lik_s2_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion in the allele-specific expression</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector Allele likelihoods
</p>

<hr>
<h2 id='calc_allele_lik_s3'>Calculate allele likelihoods for 3-state allele HMM</h2><span id='topic+calc_allele_lik_s3'></span>

<h3>Description</h3>

<p>Calculate allele likelihoods for 3-state allele HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_allele_lik_s3(pAD, DP, R, p_s, t, theta, gamma = 20, r = 0.015)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_allele_lik_s3_+3A_pad">pAD</code></td>
<td>
<p>integer vector Paternal allele counts</p>
</td></tr>
<tr><td><code id="calc_allele_lik_s3_+3A_dp">DP</code></td>
<td>
<p>integer vector Total alelle counts</p>
</td></tr>
<tr><td><code id="calc_allele_lik_s3_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="calc_allele_lik_s3_+3A_theta">theta</code></td>
<td>
<p>numeric Haplotype imbalance</p>
</td></tr>
<tr><td><code id="calc_allele_lik_s3_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion in the allele-specific expression</p>
</td></tr>
</table>

<hr>
<h2 id='calc_allele_LLR'>Calculate LLR for an allele HMM</h2><span id='topic+calc_allele_LLR'></span>

<h3>Description</h3>

<p>Calculate LLR for an allele HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_allele_LLR(pAD, DP, R, p_s, theta_mle, theta_0 = 0, gamma = 20, r = 0.015)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_allele_LLR_+3A_pad">pAD</code></td>
<td>
<p>numeric vector Phased allele depth</p>
</td></tr>
<tr><td><code id="calc_allele_LLR_+3A_dp">DP</code></td>
<td>
<p>numeric vector Total allele depth</p>
</td></tr>
<tr><td><code id="calc_allele_LLR_+3A_r">R</code></td>
<td>
<p>numeric vector Allelic bias direction</p>
</td></tr>
<tr><td><code id="calc_allele_LLR_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="calc_allele_LLR_+3A_theta_mle">theta_mle</code></td>
<td>
<p>numeric MLE of imbalance level theta (alternative hypothesis)</p>
</td></tr>
<tr><td><code id="calc_allele_LLR_+3A_theta_0">theta_0</code></td>
<td>
<p>numeric Imbalance level in the null hypothesis</p>
</td></tr>
<tr><td><code id="calc_allele_LLR_+3A_gamma">gamma</code></td>
<td>
<p>numeric Dispersion parameter for the Beta-Binomial allele model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric Log-likelihood ratio
</p>

<hr>
<h2 id='calc_exp_LLR'>Calculate LLR for an expression HMM</h2><span id='topic+calc_exp_LLR'></span>

<h3>Description</h3>

<p>Calculate LLR for an expression HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_exp_LLR(
  Y_obs,
  lambda_ref,
  d,
  phi_mle,
  mu = NULL,
  sig = NULL,
  alpha = NULL,
  beta = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_exp_LLR_+3A_y_obs">Y_obs</code></td>
<td>
<p>numeric vector Gene expression counts</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_lambda_ref">lambda_ref</code></td>
<td>
<p>numeric vector Reference expression levels</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_d">d</code></td>
<td>
<p>numeric vector Total library size</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_phi_mle">phi_mle</code></td>
<td>
<p>numeric MLE of expression fold change phi (alternative hypothesis)</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_mu">mu</code></td>
<td>
<p>numeric Mean parameter for the PLN expression model</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_sig">sig</code></td>
<td>
<p>numeric Dispersion parameter for the PLN expression model</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_alpha">alpha</code></td>
<td>
<p>numeric Hyperparameter for the gamma poisson model (not used)</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_beta">beta</code></td>
<td>
<p>numeric Hyperparameter for the gamma poisson model (not used)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric Log-likelihood ratio
</p>

<hr>
<h2 id='calc_trans_mat_s15'>Calculate the transition matrix for 15-state joint HMM</h2><span id='topic+calc_trans_mat_s15'></span>

<h3>Description</h3>

<p>Calculate the transition matrix for 15-state joint HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_trans_mat_s15(t, p_s, w, states_cn, states_phase)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_trans_mat_s15_+3A_t">t</code></td>
<td>
<p>numeric; CNV state transition probability</p>
</td></tr>
<tr><td><code id="calc_trans_mat_s15_+3A_p_s">p_s</code></td>
<td>
<p>numeric; phase switch probability</p>
</td></tr>
<tr><td><code id="calc_trans_mat_s15_+3A_w">w</code></td>
<td>
<p>numeric; relative abundance of states</p>
</td></tr>
<tr><td><code id="calc_trans_mat_s15_+3A_states_cn">states_cn</code></td>
<td>
<p>character; CNV states</p>
</td></tr>
<tr><td><code id="calc_trans_mat_s15_+3A_states_phase">states_phase</code></td>
<td>
<p>character; haplotype phase states</p>
</td></tr>
</table>


<h3>Value</h3>

<p>array; transition matrix
</p>

<hr>
<h2 id='calc_trans_mat_s7'>Calculate the transition matrix for 7-state joint HMM</h2><span id='topic+calc_trans_mat_s7'></span>

<h3>Description</h3>

<p>Calculate the transition matrix for 7-state joint HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_trans_mat_s7(t, p_s, w, states_cn, states_phase)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_trans_mat_s7_+3A_t">t</code></td>
<td>
<p>numeric; CNV state transition probability</p>
</td></tr>
<tr><td><code id="calc_trans_mat_s7_+3A_p_s">p_s</code></td>
<td>
<p>numeric; phase switch probability</p>
</td></tr>
<tr><td><code id="calc_trans_mat_s7_+3A_w">w</code></td>
<td>
<p>numeric; relative abundance of states</p>
</td></tr>
<tr><td><code id="calc_trans_mat_s7_+3A_states_cn">states_cn</code></td>
<td>
<p>character; CNV states</p>
</td></tr>
<tr><td><code id="calc_trans_mat_s7_+3A_states_phase">states_phase</code></td>
<td>
<p>character; haplotype phase states</p>
</td></tr>
</table>


<h3>Value</h3>

<p>array; transition matrix
</p>

<hr>
<h2 id='check_cols'>Check if columns are present in dataframe</h2><span id='topic+check_cols'></span>

<h3>Description</h3>

<p>Check if columns are present in dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_cols(df, cols)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_cols_+3A_df">df</code></td>
<td>
<p>dataframe Dataframe</p>
</td></tr>
<tr><td><code id="check_cols_+3A_cols">cols</code></td>
<td>
<p>character vector Column names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Dataframe
</p>

<hr>
<h2 id='check_matrix'>Check the format of a count matrix</h2><span id='topic+check_matrix'></span>

<h3>Description</h3>

<p>Check the format of a count matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_matrix(count_mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_matrix_+3A_count_mat">count_mat</code></td>
<td>
<p>matrix Count matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix Count matrix
</p>

<hr>
<h2 id='chrom_sizes_hg19'>chromosome sizes (hg19)</h2><span id='topic+chrom_sizes_hg19'></span>

<h3>Description</h3>

<p>chromosome sizes (hg19)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chrom_sizes_hg19
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 22 rows and 2 columns.
</p>

<hr>
<h2 id='chrom_sizes_hg38'>chromosome sizes (hg38)</h2><span id='topic+chrom_sizes_hg38'></span>

<h3>Description</h3>

<p>chromosome sizes (hg38)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chrom_sizes_hg38
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 22 rows and 2 columns.
</p>

<hr>
<h2 id='classify_alleles'>classify alleles using viterbi and forward-backward</h2><span id='topic+classify_alleles'></span>

<h3>Description</h3>

<p>classify alleles using viterbi and forward-backward
</p>


<h3>Usage</h3>

<pre><code class='language-R'>classify_alleles(bulk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="classify_alleles_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pesudobulk profile</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pesudobulk profile
</p>

<hr>
<h2 id='combine_bulk'>Combine allele and expression pseudobulks</h2><span id='topic+combine_bulk'></span>

<h3>Description</h3>

<p>Combine allele and expression pseudobulks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_bulk(allele_bulk, exp_bulk, gtf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_bulk_+3A_allele_bulk">allele_bulk</code></td>
<td>
<p>dataframe Bulk allele profile</p>
</td></tr>
<tr><td><code id="combine_bulk_+3A_exp_bulk">exp_bulk</code></td>
<td>
<p>dataframe Bulk expression profile</p>
</td></tr>
<tr><td><code id="combine_bulk_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk allele and expression profile
</p>

<hr>
<h2 id='dbbinom'>Beta-binomial distribution density function
A distribution is beta-binomial if p, the probability of success,
in a binomial distribution has a beta distribution with shape
parameters alpha &gt; 0 and beta &gt; 0
For more details, see extraDistr::dbbinom</h2><span id='topic+dbbinom'></span>

<h3>Description</h3>

<p>Beta-binomial distribution density function
A distribution is beta-binomial if p, the probability of success,
in a binomial distribution has a beta distribution with shape
parameters alpha &gt; 0 and beta &gt; 0
For more details, see extraDistr::dbbinom
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbbinom(x, size, alpha = 1, beta = 1, log = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dbbinom_+3A_x">x</code></td>
<td>
<p>vector of quantiles</p>
</td></tr>
<tr><td><code id="dbbinom_+3A_size">size</code></td>
<td>
<p>number of trials (zero or more)</p>
</td></tr>
<tr><td><code id="dbbinom_+3A_alpha">alpha</code></td>
<td>
<p>numeric (default=1)</p>
</td></tr>
<tr><td><code id="dbbinom_+3A_beta">beta</code></td>
<td>
<p>numeric (default=1)</p>
</td></tr>
<tr><td><code id="dbbinom_+3A_log">log</code></td>
<td>
<p>boolean (default=FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric Probability density values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dbbinom(1, 1, 1, 1)
</code></pre>

<hr>
<h2 id='df_allele_example'>example allele count dataframe</h2><span id='topic+df_allele_example'></span>

<h3>Description</h3>

<p>example allele count dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>df_allele_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 9957 rows and 11 columns.
</p>

<hr>
<h2 id='dpoilog'>Returns the density for the Poisson lognormal distribution with parameters mu and sig</h2><span id='topic+dpoilog'></span>

<h3>Description</h3>

<p>Returns the density for the Poisson lognormal distribution with parameters mu and sig
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dpoilog(x, mu, sig, log = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dpoilog_+3A_x">x</code></td>
<td>
<p>vector of integers, the observations</p>
</td></tr>
<tr><td><code id="dpoilog_+3A_mu">mu</code></td>
<td>
<p>mean of lognormal distribution</p>
</td></tr>
<tr><td><code id="dpoilog_+3A_sig">sig</code></td>
<td>
<p>standard deviation of lognormal distribution</p>
</td></tr>
<tr><td><code id="dpoilog_+3A_log">log</code></td>
<td>
<p>boolean Return the log density if TRUE (default=FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric Probability density values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = dpoilog(1, 1, 1)
</code></pre>

<hr>
<h2 id='filter_genes'>filter for mutually expressed genes</h2><span id='topic+filter_genes'></span>

<h3>Description</h3>

<p>filter for mutually expressed genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_genes(count_mat, lambdas_ref, gtf, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_genes_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix Gene expression counts</p>
</td></tr>
<tr><td><code id="filter_genes_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>named numeric vector A reference expression profile</p>
</td></tr>
<tr><td><code id="filter_genes_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector Genes that are kept after filtering
</p>

<hr>
<h2 id='fit_gamma'>fit gamma maximum likelihood</h2><span id='topic+fit_gamma'></span>

<h3>Description</h3>

<p>fit gamma maximum likelihood
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_gamma(AD, DP, r = 0.015, start = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_gamma_+3A_ad">AD</code></td>
<td>
<p>numeric vector Variant allele depth</p>
</td></tr>
<tr><td><code id="fit_gamma_+3A_dp">DP</code></td>
<td>
<p>numeric vector Total allele depth</p>
</td></tr>
<tr><td><code id="fit_gamma_+3A_r">r</code></td>
<td>
<p>numeric Degree of allele bias</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a fit
</p>

<hr>
<h2 id='fit_lnpois_cpp'>Fit MLE of log-normal Poisson model</h2><span id='topic+fit_lnpois_cpp'></span>

<h3>Description</h3>

<p>Fit MLE of log-normal Poisson model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_lnpois_cpp(Y_obs, lambda_ref, d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_lnpois_cpp_+3A_y_obs">Y_obs</code></td>
<td>
<p>Vector of observed counts</p>
</td></tr>
<tr><td><code id="fit_lnpois_cpp_+3A_lambda_ref">lambda_ref</code></td>
<td>
<p>Vector of reference rates</p>
</td></tr>
<tr><td><code id="fit_lnpois_cpp_+3A_d">d</code></td>
<td>
<p>integer Total depth</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NumericVector MLE estimates of mu and sigma
</p>

<hr>
<h2 id='fit_ref_sse'>Fit a reference profile from multiple references using constrained least square</h2><span id='topic+fit_ref_sse'></span>

<h3>Description</h3>

<p>Fit a reference profile from multiple references using constrained least square
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_ref_sse(Y_obs, lambdas_ref, gtf, min_lambda = 2e-06, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_ref_sse_+3A_y_obs">Y_obs</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="fit_ref_sse_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix</p>
</td></tr>
<tr><td><code id="fit_ref_sse_+3A_gtf">gtf</code></td>
<td>
<p>dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>fitted expression profile
</p>

<hr>
<h2 id='forward_back_allele'>Forward-backward algorithm for allele HMM</h2><span id='topic+forward_back_allele'></span>

<h3>Description</h3>

<p>Forward-backward algorithm for allele HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>forward_back_allele(hmm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="forward_back_allele_+3A_hmm">hmm</code></td>
<td>
<p>HMM object; expect variables x (allele depth), d (total depth),
logPi (log transition prob matrix), delta (prior for each state),
alpha (alpha for each state), beta (beta for each state),
states (states), p_s (phase switch probs)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric matrix; posterior probabilities
</p>


<h3>Examples</h3>

<pre><code class='language-R'>forward_back_allele(pre_likelihood_hmm)
</code></pre>

<hr>
<h2 id='gaps_hg19'>genome gap regions (hg19)</h2><span id='topic+gaps_hg19'></span>

<h3>Description</h3>

<p>genome gap regions (hg19)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gaps_hg19
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 28 rows and 3 columns.
</p>

<hr>
<h2 id='gaps_hg38'>genome gap regions (hg38)</h2><span id='topic+gaps_hg38'></span>

<h3>Description</h3>

<p>genome gap regions (hg38)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gaps_hg38
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 30 rows and 3 columns.
</p>

<hr>
<h2 id='gene_counts_example'>example gene expression counts matrix</h2><span id='topic+gene_counts_example'></span>

<h3>Description</h3>

<p>example gene expression counts matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_counts_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1758 rows and 1 columns.
</p>

<hr>
<h2 id='generate_postfix'>Generate alphabetical postfixes</h2><span id='topic+generate_postfix'></span>

<h3>Description</h3>

<p>Generate alphabetical postfixes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_postfix(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_postfix_+3A_n">n</code></td>
<td>
<p>vector of integers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of alphabetical postfixes
</p>

<hr>
<h2 id='get_allele_bulk'>Aggregate into pseudobulk alelle profile</h2><span id='topic+get_allele_bulk'></span>

<h3>Description</h3>

<p>Aggregate into pseudobulk alelle profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_allele_bulk(df_allele, gtf, genetic_map = NULL, nu = 0.5, min_depth = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_allele_bulk_+3A_df_allele">df_allele</code></td>
<td>
<p>dataframe Single-cell allele counts</p>
</td></tr>
<tr><td><code id="get_allele_bulk_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td></tr>
<tr><td><code id="get_allele_bulk_+3A_genetic_map">genetic_map</code></td>
<td>
<p>dataframe Genetic map</p>
</td></tr>
<tr><td><code id="get_allele_bulk_+3A_nu">nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td></tr>
<tr><td><code id="get_allele_bulk_+3A_min_depth">min_depth</code></td>
<td>
<p>integer Minimum coverage to filter SNPs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk allele profile
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bulk_example = get_allele_bulk(
    df_allele = df_allele_example,
    gtf = gtf_hg38)
</code></pre>

<hr>
<h2 id='get_allele_hmm_s2'>Make a 2-state allele HMM - no transitions to netural state</h2><span id='topic+get_allele_hmm_s2'></span>

<h3>Description</h3>

<p>Make a 2-state allele HMM - no transitions to netural state
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_allele_hmm_s2(pAD, DP, R, p_s, theta, gamma = 20, r = 0.015)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_allele_hmm_s2_+3A_pad">pAD</code></td>
<td>
<p>integer vector Paternal allele counts</p>
</td></tr>
<tr><td><code id="get_allele_hmm_s2_+3A_dp">DP</code></td>
<td>
<p>integer vector Total alelle counts</p>
</td></tr>
<tr><td><code id="get_allele_hmm_s2_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="get_allele_hmm_s2_+3A_theta">theta</code></td>
<td>
<p>numeric Haplotype imbalance</p>
</td></tr>
<tr><td><code id="get_allele_hmm_s2_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion in the allele-specific expression</p>
</td></tr>
</table>


<h3>Value</h3>

<p>HMM object
</p>

<hr>
<h2 id='get_allele_hmm_s3'>Make a 3-state allele HMM - allow transitions to netural state</h2><span id='topic+get_allele_hmm_s3'></span>

<h3>Description</h3>

<p>Make a 3-state allele HMM - allow transitions to netural state
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_allele_hmm_s3(pAD, DP, R, p_s, t, theta_min, gamma = 20, r = 0.015)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_allele_hmm_s3_+3A_pad">pAD</code></td>
<td>
<p>integer vector Paternal allele counts</p>
</td></tr>
<tr><td><code id="get_allele_hmm_s3_+3A_dp">DP</code></td>
<td>
<p>integer vector Total alelle counts</p>
</td></tr>
<tr><td><code id="get_allele_hmm_s3_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="get_allele_hmm_s3_+3A_t">t</code></td>
<td>
<p>numeric Transition probability between haplotype states</p>
</td></tr>
<tr><td><code id="get_allele_hmm_s3_+3A_theta_min">theta_min</code></td>
<td>
<p>numeric Haplotype imbalance</p>
</td></tr>
<tr><td><code id="get_allele_hmm_s3_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion in the allele-specific expression</p>
</td></tr>
<tr><td><code id="get_allele_hmm_s3_+3A_r">r</code></td>
<td>
<p>numeric Variant mapping bias</p>
</td></tr>
</table>


<h3>Value</h3>

<p>HMM object
</p>

<hr>
<h2 id='get_bulk'>Produce combined bulk expression and allele profile</h2><span id='topic+get_bulk'></span>

<h3>Description</h3>

<p>Produce combined bulk expression and allele profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_bulk(
  count_mat,
  lambdas_ref,
  df_allele,
  gtf,
  genetic_map = NULL,
  min_depth = 0,
  nu = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_bulk_+3A_count_mat">count_mat</code></td>
<td>
<p>matrix Gene expression counts</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Reference expression profiles</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_df_allele">df_allele</code></td>
<td>
<p>dataframe Allele counts</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_genetic_map">genetic_map</code></td>
<td>
<p>dataframe Genetic map</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_min_depth">min_depth</code></td>
<td>
<p>integer Minimum coverage to filter SNPs</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_nu">nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_verbose">verbose</code></td>
<td>
<p>logical Whether to print progress</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk gene expression and allele profile
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bulk_example = get_bulk(
    count_mat = gene_counts_example,
    lambdas_ref = ref_hca,
    df_allele = df_allele_example,
    gtf = gtf_hg38)
</code></pre>

<hr>
<h2 id='get_exp_bulk'>Aggregate into bulk expression profile</h2><span id='topic+get_exp_bulk'></span>

<h3>Description</h3>

<p>Aggregate into bulk expression profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_exp_bulk(count_mat, lambdas_ref, gtf, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_exp_bulk_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix Gene expression counts</p>
</td></tr>
<tr><td><code id="get_exp_bulk_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Reference expression profiles</p>
</td></tr>
<tr><td><code id="get_exp_bulk_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk gene expression profile
</p>

<hr>
<h2 id='get_trans_probs_s15'>Helper function to calculate transition porbabilities for 15-state joint HMM
cn/phase are sclars, only p_s is vectorized</h2><span id='topic+get_trans_probs_s15'></span>

<h3>Description</h3>

<p>Helper function to calculate transition porbabilities for 15-state joint HMM
cn/phase are sclars, only p_s is vectorized
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_trans_probs_s15(t, p_s, w, cn_from, phase_from, cn_to, phase_to)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_trans_probs_s15_+3A_t">t</code></td>
<td>
<p>numeric; CNV state transition probability</p>
</td></tr>
<tr><td><code id="get_trans_probs_s15_+3A_p_s">p_s</code></td>
<td>
<p>numeric; phase switch probability</p>
</td></tr>
<tr><td><code id="get_trans_probs_s15_+3A_w">w</code></td>
<td>
<p>numeric; relative abundance of states</p>
</td></tr>
<tr><td><code id="get_trans_probs_s15_+3A_cn_from">cn_from</code></td>
<td>
<p>character; originating CNV state</p>
</td></tr>
<tr><td><code id="get_trans_probs_s15_+3A_phase_from">phase_from</code></td>
<td>
<p>character; originating haplotype phase state</p>
</td></tr>
<tr><td><code id="get_trans_probs_s15_+3A_cn_to">cn_to</code></td>
<td>
<p>character; destination CNV state</p>
</td></tr>
<tr><td><code id="get_trans_probs_s15_+3A_phase_to">phase_to</code></td>
<td>
<p>character; destination haplotype phase state</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric; transition probability
</p>

<hr>
<h2 id='get_trans_probs_s7'>Helper function to calculate transition porbabilities for 7-state joint HMM
cn/phase are sclars, only p_s is vectorized</h2><span id='topic+get_trans_probs_s7'></span>

<h3>Description</h3>

<p>Helper function to calculate transition porbabilities for 7-state joint HMM
cn/phase are sclars, only p_s is vectorized
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_trans_probs_s7(t, p_s, w, cn_from, phase_from, cn_to, phase_to)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_trans_probs_s7_+3A_t">t</code></td>
<td>
<p>numeric; CNV state transition probability</p>
</td></tr>
<tr><td><code id="get_trans_probs_s7_+3A_p_s">p_s</code></td>
<td>
<p>numeric; phase switch probability</p>
</td></tr>
<tr><td><code id="get_trans_probs_s7_+3A_w">w</code></td>
<td>
<p>numeric; relative abundance of states</p>
</td></tr>
<tr><td><code id="get_trans_probs_s7_+3A_cn_from">cn_from</code></td>
<td>
<p>character; originating CNV state</p>
</td></tr>
<tr><td><code id="get_trans_probs_s7_+3A_phase_from">phase_from</code></td>
<td>
<p>character; originating haplotype phase state</p>
</td></tr>
<tr><td><code id="get_trans_probs_s7_+3A_cn_to">cn_to</code></td>
<td>
<p>character; destination CNV state</p>
</td></tr>
<tr><td><code id="get_trans_probs_s7_+3A_phase_to">phase_to</code></td>
<td>
<p>character; destination haplotype phase state</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric; transition probability
</p>

<hr>
<h2 id='gtf_hg19'>gene model (hg19)</h2><span id='topic+gtf_hg19'></span>

<h3>Description</h3>

<p>gene model (hg19)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtf_hg19
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 26841 rows and 5 columns.
</p>

<hr>
<h2 id='gtf_hg38'>gene model (hg38)</h2><span id='topic+gtf_hg38'></span>

<h3>Description</h3>

<p>gene model (hg38)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtf_hg38
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 26807 rows and 5 columns.
</p>

<hr>
<h2 id='gtf_mm10'>gene model (mm10)</h2><span id='topic+gtf_mm10'></span>

<h3>Description</h3>

<p>gene model (mm10)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtf_mm10
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 30336 rows and 5 columns.
</p>

<hr>
<h2 id='l_bbinom'>calculate joint likelihood of allele data</h2><span id='topic+l_bbinom'></span>

<h3>Description</h3>

<p>calculate joint likelihood of allele data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>l_bbinom(AD, DP, alpha, beta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="l_bbinom_+3A_ad">AD</code></td>
<td>
<p>numeric vector Variant allele depth</p>
</td></tr>
<tr><td><code id="l_bbinom_+3A_dp">DP</code></td>
<td>
<p>numeric vector Total allele depth</p>
</td></tr>
<tr><td><code id="l_bbinom_+3A_alpha">alpha</code></td>
<td>
<p>numeric Alpha parameter of Beta-Binomial distribution</p>
</td></tr>
<tr><td><code id="l_bbinom_+3A_beta">beta</code></td>
<td>
<p>numeric Beta parameter of Beta-Binomial distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric Joint log likelihood
</p>


<h3>Examples</h3>

<pre><code class='language-R'>l_bbinom(c(1, 2), c(1, 2), 1, 1)
</code></pre>

<hr>
<h2 id='l_lnpois'>calculate joint likelihood of a PLN model</h2><span id='topic+l_lnpois'></span>

<h3>Description</h3>

<p>calculate joint likelihood of a PLN model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>l_lnpois(Y_obs, lambda_ref, d, mu, sig, phi = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="l_lnpois_+3A_y_obs">Y_obs</code></td>
<td>
<p>numeric vector Gene expression counts</p>
</td></tr>
<tr><td><code id="l_lnpois_+3A_lambda_ref">lambda_ref</code></td>
<td>
<p>numeric vector Reference expression levels</p>
</td></tr>
<tr><td><code id="l_lnpois_+3A_d">d</code></td>
<td>
<p>numeric Total library size</p>
</td></tr>
<tr><td><code id="l_lnpois_+3A_mu">mu</code></td>
<td>
<p>numeric Global mean expression</p>
</td></tr>
<tr><td><code id="l_lnpois_+3A_sig">sig</code></td>
<td>
<p>numeric Global standard deviation of expression</p>
</td></tr>
<tr><td><code id="l_lnpois_+3A_phi">phi</code></td>
<td>
<p>numeric Fold change of expression</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric Joint log likelihood
</p>


<h3>Examples</h3>

<pre><code class='language-R'>l_lnpois(c(1, 2), c(1, 2), 1, 1, 1)
</code></pre>

<hr>
<h2 id='likelihood_allele'>Only compute total log likelihood from an allele HMM</h2><span id='topic+likelihood_allele'></span>

<h3>Description</h3>

<p>Only compute total log likelihood from an allele HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood_allele(hmm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="likelihood_allele_+3A_hmm">hmm</code></td>
<td>
<p>HMM object; expect variables x (allele depth), d (total depth),
logPi (log transition prob matrix), delta (prior for each state),
alpha (alpha for each state), beta (beta for each state),
states (states), p_s (phase switch probs)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric; total log likelihood
</p>


<h3>Examples</h3>

<pre><code class='language-R'>likelihood_allele(pre_likelihood_hmm)
</code></pre>

<hr>
<h2 id='logSumExp'>logSumExp function</h2><span id='topic+logSumExp'></span>

<h3>Description</h3>

<p>logSumExp function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logSumExp(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logSumExp_+3A_x">x</code></td>
<td>
<p>NumericVector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>double logSumExp of x
</p>

<hr>
<h2 id='plot_bulks'>Plot a group of pseudobulk HMM profiles</h2><span id='topic+plot_bulks'></span>

<h3>Description</h3>

<p>Plot a group of pseudobulk HMM profiles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_bulks(bulks, ..., ncol = 1, title = TRUE, title_size = 8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_bulks_+3A_bulks">bulks</code></td>
<td>
<p>dataframe Pseudobulk profiles annotated with &quot;sample&quot; column</p>
</td></tr>
<tr><td><code id="plot_bulks_+3A_...">...</code></td>
<td>
<p>additional parameters passed to plot_psbulk()</p>
</td></tr>
<tr><td><code id="plot_bulks_+3A_ncol">ncol</code></td>
<td>
<p>integer Number of columns</p>
</td></tr>
<tr><td><code id="plot_bulks_+3A_title">title</code></td>
<td>
<p>logical Whether to add titles to individual plots</p>
</td></tr>
<tr><td><code id="plot_bulks_+3A_title_size">title_size</code></td>
<td>
<p>numeric Size of titles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = plot_bulks(bulk_example)
</code></pre>

<hr>
<h2 id='plot_psbulk'>Plot a pseudobulk HMM profile</h2><span id='topic+plot_psbulk'></span>

<h3>Description</h3>

<p>Plot a pseudobulk HMM profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_psbulk(
  bulk,
  use_pos = TRUE,
  allele_only = FALSE,
  min_LLR = 5,
  min_depth = 8,
  exp_limit = 2,
  phi_mle = TRUE,
  theta_roll = FALSE,
  dot_size = 0.8,
  dot_alpha = 0.5,
  legend = TRUE,
  exclude_gap = TRUE,
  genome = "hg38",
  text_size = 10,
  raster = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_psbulk_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_use_pos">use_pos</code></td>
<td>
<p>logical Use marker position instead of index as x coordinate</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_allele_only">allele_only</code></td>
<td>
<p>logical Only plot alleles</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_min_llr">min_LLR</code></td>
<td>
<p>numeric LLR threshold for event filtering</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_min_depth">min_depth</code></td>
<td>
<p>numeric Minimum coverage depth for a SNP to be plotted</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_exp_limit">exp_limit</code></td>
<td>
<p>numeric Expression logFC axis limit</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_phi_mle">phi_mle</code></td>
<td>
<p>logical Whether to plot estimates of segmental expression fold change</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_theta_roll">theta_roll</code></td>
<td>
<p>logical Whether to plot rolling estimates of allele imbalance</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_dot_size">dot_size</code></td>
<td>
<p>numeric Size of marker dots</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_dot_alpha">dot_alpha</code></td>
<td>
<p>numeric Transparency of the marker dots</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_legend">legend</code></td>
<td>
<p>logical Whether to show legend</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_exclude_gap">exclude_gap</code></td>
<td>
<p>logical Whether to mark gap regions and centromeres</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_genome">genome</code></td>
<td>
<p>character Genome build, either 'hg38' or 'hg19'</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_text_size">text_size</code></td>
<td>
<p>numeric Size of text in the plot</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_raster">raster</code></td>
<td>
<p>logical Whether to raster images</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot Plot of pseudobulk HMM profile
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = plot_psbulk(bulk_example)
</code></pre>

<hr>
<h2 id='pre_likelihood_hmm'>HMM object for unit tests</h2><span id='topic+pre_likelihood_hmm'></span>

<h3>Description</h3>

<p>HMM object for unit tests
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pre_likelihood_hmm
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 10.
</p>

<hr>
<h2 id='ref_hca'>reference expression magnitudes from HCA</h2><span id='topic+ref_hca'></span>

<h3>Description</h3>

<p>reference expression magnitudes from HCA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ref_hca
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 24756 rows and 12 columns.
</p>

<hr>
<h2 id='ref_hca_counts'>reference expression counts from HCA</h2><span id='topic+ref_hca_counts'></span>

<h3>Description</h3>

<p>reference expression counts from HCA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ref_hca_counts
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 24857 rows and 12 columns.
</p>

<hr>
<h2 id='run_allele_hmm_s3'>Run a 3-state allele HMM</h2><span id='topic+run_allele_hmm_s3'></span>

<h3>Description</h3>

<p>Run a 3-state allele HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_allele_hmm_s3(
  pAD,
  DP,
  R,
  p_s,
  t = 1e-05,
  theta_min = 0.08,
  gamma = 20,
  r = 0.015
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_allele_hmm_s3_+3A_pad">pAD</code></td>
<td>
<p>integer vector Paternal allele counts</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s3_+3A_dp">DP</code></td>
<td>
<p>integer vector Total alelle counts</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s3_+3A_r">R</code></td>
<td>
<p>numeric vector Variant mapping bias direction</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s3_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s3_+3A_t">t</code></td>
<td>
<p>numeric Transition probability between copy number states</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s3_+3A_theta_min">theta_min</code></td>
<td>
<p>numeric Minimum haplotype frequency deviation threshold</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s3_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion in the allele-specific expression</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector Decoded states
</p>

<hr>
<h2 id='run_allele_hmm_s5'>Run a 5-state allele-only HMM - two theta levels</h2><span id='topic+run_allele_hmm_s5'></span>

<h3>Description</h3>

<p>Run a 5-state allele-only HMM - two theta levels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_allele_hmm_s5(
  pAD,
  DP,
  p_s,
  t = 1e-05,
  theta_min = 0.08,
  gamma = 20,
  prior = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_allele_hmm_s5_+3A_pad">pAD</code></td>
<td>
<p>integer vector Paternal allele counts</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s5_+3A_dp">DP</code></td>
<td>
<p>integer vector Total alelle counts</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s5_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s5_+3A_t">t</code></td>
<td>
<p>numeric Transition probability between copy number states</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s5_+3A_theta_min">theta_min</code></td>
<td>
<p>numeric Minimum haplotype frequency deviation threshold</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s5_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion in the allele-specific expression</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s5_+3A_prior">prior</code></td>
<td>
<p>numeric vector Prior probabilities for each state</p>
</td></tr>
<tr><td><code id="run_allele_hmm_s5_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector Decoded states
</p>


<h3>Examples</h3>

<pre><code class='language-R'>with(bulk_example, {
    run_allele_hmm_s5(pAD = pAD, DP = DP, R = R, p_s = p_s, theta_min = 0.08, gamma = 30)
})
</code></pre>

<hr>
<h2 id='run_joint_hmm_s15'>Run 15-state joint HMM on a pseudobulk profile</h2><span id='topic+run_joint_hmm_s15'></span>

<h3>Description</h3>

<p>Run 15-state joint HMM on a pseudobulk profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_joint_hmm_s15(
  pAD,
  DP,
  p_s,
  Y_obs = 0,
  lambda_ref = 0,
  d_total = 0,
  theta_min = 0.08,
  theta_neu = 0,
  bal_cnv = TRUE,
  phi_del = 2^(-0.25),
  phi_amp = 2^(0.25),
  phi_bamp = phi_amp,
  phi_bdel = phi_del,
  mu = 0,
  sig = 1,
  r = 0.015,
  t = 1e-05,
  gamma = 18,
  prior = NULL,
  exp_only = FALSE,
  allele_only = FALSE,
  classify_allele = FALSE,
  debug = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_joint_hmm_s15_+3A_pad">pAD</code></td>
<td>
<p>integer vector Paternal allele counts</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_dp">DP</code></td>
<td>
<p>integer vector Total alelle counts</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_y_obs">Y_obs</code></td>
<td>
<p>numeric vector Observed gene counts</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_lambda_ref">lambda_ref</code></td>
<td>
<p>numeric vector Reference expression rates</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_d_total">d_total</code></td>
<td>
<p>integer Total library size for expression counts</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_theta_min">theta_min</code></td>
<td>
<p>numeric Minimum haplotype imbalance threshold</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_theta_neu">theta_neu</code></td>
<td>
<p>numeric Haplotype imbalance threshold for neutral state</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_bal_cnv">bal_cnv</code></td>
<td>
<p>logical Whether to include balanced CNV states</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_phi_del">phi_del</code></td>
<td>
<p>numeric Expected fold change for deletion</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_phi_amp">phi_amp</code></td>
<td>
<p>numeric Expected fold change for amplification</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_phi_bamp">phi_bamp</code></td>
<td>
<p>numeric Expected fold change for balanced amplification</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_phi_bdel">phi_bdel</code></td>
<td>
<p>numeric Expected fold change for balanced deletion</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_mu">mu</code></td>
<td>
<p>numeric Global expression bias</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_sig">sig</code></td>
<td>
<p>numeric Global expression variance</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_r">r</code></td>
<td>
<p>numeric Variant mapping bias</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_t">t</code></td>
<td>
<p>numeric Transition probability between copy number states</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion in the allele-specific expression</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_prior">prior</code></td>
<td>
<p>numeric vector Prior probabilities for each state</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_exp_only">exp_only</code></td>
<td>
<p>logical Whether to only use expression data</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_allele_only">allele_only</code></td>
<td>
<p>logical Whether to only use allele data</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_classify_allele">classify_allele</code></td>
<td>
<p>logical Whether to classify allele states</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_debug">debug</code></td>
<td>
<p>logical Whether to print debug messages</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s15_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector Decoded states
</p>


<h3>Examples</h3>

<pre><code class='language-R'>with(bulk_example, {
    run_joint_hmm_s15(pAD = pAD, DP = DP, p_s = p_s, Y_obs = Y_obs, lambda_ref = lambda_ref, 
    d_total = na.omit(unique(d_obs)), mu = mu, sig = sig, t = 1e-5, gamma = 30, theta_min = 0.08)
})
</code></pre>

<hr>
<h2 id='run_joint_hmm_s7'>Run 7-state joint HMM on a pseudobulk profile</h2><span id='topic+run_joint_hmm_s7'></span>

<h3>Description</h3>

<p>Run 7-state joint HMM on a pseudobulk profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_joint_hmm_s7(
  pAD,
  DP,
  R,
  p_s,
  Y_obs,
  lambda_ref,
  d_total,
  theta_min = 0.08,
  phi_del = 2^(-0.25),
  phi_amp = 2^(0.25),
  t = 1e-05,
  mu = 0,
  sig = 1,
  gamma = 20,
  r = 0.015,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_joint_hmm_s7_+3A_pad">pAD</code></td>
<td>
<p>integer vector Paternal allele counts</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_dp">DP</code></td>
<td>
<p>integer vector Total alelle counts</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_r">R</code></td>
<td>
<p>numeric vector Variant mapping bias direction</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_y_obs">Y_obs</code></td>
<td>
<p>numeric vector Observed gene counts</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_lambda_ref">lambda_ref</code></td>
<td>
<p>numeric vector Reference expression rates</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_d_total">d_total</code></td>
<td>
<p>integer Total library size for expression counts</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_theta_min">theta_min</code></td>
<td>
<p>numeric Minimum haplotype imbalance threshold</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_phi_del">phi_del</code></td>
<td>
<p>numeric Expected fold change for deletion</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_phi_amp">phi_amp</code></td>
<td>
<p>numeric Expected fold change for amplification</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_t">t</code></td>
<td>
<p>numeric Transition probability between copy number states</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_mu">mu</code></td>
<td>
<p>numeric Global expression bias</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_sig">sig</code></td>
<td>
<p>numeric Global expression variance</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion in the allele-specific expression</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_r">r</code></td>
<td>
<p>numeric Variant mapping bias</p>
</td></tr>
<tr><td><code id="run_joint_hmm_s7_+3A_debug">debug</code></td>
<td>
<p>logical Whether to print debug messages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector Decoded states
</p>

<hr>
<h2 id='segs_example'>example CNV segments dataframe</h2><span id='topic+segs_example'></span>

<h3>Description</h3>

<p>example CNV segments dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>segs_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 27 rows and 30 columns.
</p>

<hr>
<h2 id='smooth_segs'>Smooth the segments after HMM decoding</h2><span id='topic+smooth_segs'></span>

<h3>Description</h3>

<p>Smooth the segments after HMM decoding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smooth_segs(bulk, min_genes = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smooth_segs_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
<tr><td><code id="smooth_segs_+3A_min_genes">min_genes</code></td>
<td>
<p>integer Minimum number of genes to call a segment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk profile
</p>

<hr>
<h2 id='switch_prob_cm'>predict phase switch probablity as a function of genetic distance</h2><span id='topic+switch_prob_cm'></span>

<h3>Description</h3>

<p>predict phase switch probablity as a function of genetic distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>switch_prob_cm(d, nu = 1, min_p = 1e-10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="switch_prob_cm_+3A_d">d</code></td>
<td>
<p>numeric vector Genetic distance in cM</p>
</td></tr>
<tr><td><code id="switch_prob_cm_+3A_nu">nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td></tr>
<tr><td><code id="switch_prob_cm_+3A_min_p">min_p</code></td>
<td>
<p>numeric Minimum phase switch probability</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector Phase switch probability
</p>

<hr>
<h2 id='theta_hat_seg'>Estimate of imbalance level theta in a segment</h2><span id='topic+theta_hat_seg'></span>

<h3>Description</h3>

<p>Estimate of imbalance level theta in a segment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theta_hat_seg(major_count, minor_count)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="theta_hat_seg_+3A_major_count">major_count</code></td>
<td>
<p>vector of major allele count</p>
</td></tr>
<tr><td><code id="theta_hat_seg_+3A_minor_count">minor_count</code></td>
<td>
<p>vector of minor allele count</p>
</td></tr>
</table>


<h3>Value</h3>

<p>estimate of theta
</p>

<hr>
<h2 id='vcf_meta'>example VCF header</h2><span id='topic+vcf_meta'></span>

<h3>Description</h3>

<p>example VCF header
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf_meta
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 65.
</p>

<hr>
<h2 id='viterbi_allele'>Viterbi algorithm for allele HMM</h2><span id='topic+viterbi_allele'></span>

<h3>Description</h3>

<p>Viterbi algorithm for allele HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>viterbi_allele(hmm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="viterbi_allele_+3A_hmm">hmm</code></td>
<td>
<p>HMM object; expect variables x (allele depth), d (total depth),
logPi (log transition prob matrix), delta (prior for each state),
alpha (alpha for each state), beta (beta for each state),
states (states), p_s (phase switch probs)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector Decoded states
</p>

<hr>
<h2 id='viterbi_joint'>Viterbi algorithm for joint HMM, can only handle one set of observations</h2><span id='topic+viterbi_joint'></span>

<h3>Description</h3>

<p>Viterbi algorithm for joint HMM, can only handle one set of observations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>viterbi_joint(hmm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="viterbi_joint_+3A_hmm">hmm</code></td>
<td>
<p>HMM object; expect variables x (allele depth), d (total depth), y (expression count), l (cell total library size),
lambda (reference expression rate), mu (global expression mean), sig (global expression standard deviation),
logPi (log transition prob matrix), phi (expression fold change for each state), delta (prior for each state),
alpha (alpha for each state), beta (beta for each state), states (states), p_s (phase switch probs)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector Decoded states
</p>

<hr>
<h2 id='viterbi_joint_mat'>Generalized viterbi algorithm for joint HMM, can handle multiple sets of observations</h2><span id='topic+viterbi_joint_mat'></span>

<h3>Description</h3>

<p>Generalized viterbi algorithm for joint HMM, can handle multiple sets of observations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>viterbi_joint_mat(hmm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="viterbi_joint_mat_+3A_hmm">hmm</code></td>
<td>
<p>HMM object; expect variables x (allele depth), d (total depth), y (expression count), l (cell total library size),
lambda (reference expression rate), mu (global expression mean), sig (global expression standard deviation),
logPi (log transition prob matrix), phi (expression fold change for each state), delta (prior for each state),
alpha (alpha for each state), beta (beta for each state), states (states), p_s (phase switch probs)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector; decoded states
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
