<!DOCTYPE html><html lang="en"><head><title>Help for package handyFunctions</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {handyFunctions}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#handyFunctions-package'><p>handyFunctions: Useful Functions for Handfully Manipulating and Analyzing Data with Data.frame Format</p></a></li>
<li><a href='#checkCols'><p>check the validation and return index of cols given from input in rawDataFrame</p></a></li>
<li><a href='#checkDtype'><p>Return suggested dtype of vector input</p></a></li>
<li><a href='#grade'><p>Grade records of virtual persons in high school</p></a></li>
<li><a href='#matchIndex'><p>Return the index of source vector matched with query vector</p></a></li>
<li><a href='#mergeCustom'><p>merge two data.frame based on xcol and ycol</p></a></li>
<li><a href='#modifyColNames'><p>Return reformatted data.frame with standard col names</p></a></li>
<li><a href='#modifyColTypes'><p>Return suggested appropriate dtypes for each column in rawDataFrame</p></a></li>
<li><a href='#modifyRowNames'><p>Return reformatted data.frame with standard row names</p></a></li>
<li><a href='#people'><p>Basic information of virtual persons</p></a></li>
<li><a href='#queryingInfo'><p>return index of x data.frame with the given vector/list or ycol in data.frame</p>
(if set the accurate match or not)</a></li>
<li><a href='#ShowSNPDensityPlot'><p>Function of showing SNP density at chromosome level</p></a></li>
<li><a href='#SNV_1MB_density_data'><p>The SNPV number within 1Mb bins at chromosome levels generated from transcriptome dataset of two dog populations</p>
(including wild wolf and domesticated dogs).</a></li>
<li><a href='#splitCol'><p>Return specific-indexed vector according to given delimitator/separator by splitting one col in data.frame</p></a></li>
<li><a href='#unifyDataframe'><p>Reformat dataframe with the all modifiers simultaneously (colNames, rowNames and dtypes)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Useful Functions for Handfully Manipulating and Analyzing Data
with Data.frame Format</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Hongfei Liu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hongfei Liu &lt;lhf563@126.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/LuffyLouis/handyFunctions">https://github.com/LuffyLouis/handyFunctions</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/LuffyLouis/handyFunctions/issues">https://github.com/LuffyLouis/handyFunctions/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Some useful functions for simply manipulating and analyzing data with data.frame format.
  It mainly includes the following sections: ReformatDataframe (reformat dataframe with the modifiers), InteractDataframe, and Post-VCF (for downstream analysis for data generated from 'vcftools' Petr et al (2011) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtr330">doi:10.1093/bioinformatics/btr330</a>&gt;
  or 'plink' Chang et al (2015) &lt;<a href="https://doi.org/10.1186%2Fs13742-015-0047-8">doi:10.1186/s13742-015-0047-8</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2 (&ge; 3.0), methods, rlang, stats (&ge; 3.0), stringr (&ge;
1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-20 01:25:45 UTC; liuhongfei</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-22 14:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='handyFunctions-package'>handyFunctions: Useful Functions for Handfully Manipulating and Analyzing Data with Data.frame Format</h2><span id='topic+handyFunctions'></span><span id='topic+handyFunctions-package'></span>

<h3>Description</h3>

<p>Some useful functions for simply manipulating and analyzing data with data.frame format. It mainly includes the following sections: ReformatDataframe (reformat dataframe with the modifiers), InteractDataframe, and Post-VCF (for downstream analysis for data generated from 'vcftools' Petr et al (2011) doi: <a href="https://doi.org/10.1093/bioinformatics/btr330">10.1093/bioinformatics/btr330</a> or 'plink' Chang et al (2015) doi: <a href="https://doi.org/10.1186/s13742-015-0047-8">10.1186/s13742-015-0047-8</a>.
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/LuffyLouis/handyFunctions">https://github.com/LuffyLouis/handyFunctions</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/LuffyLouis/handyFunctions/issues">https://github.com/LuffyLouis/handyFunctions/issues</a>
</p>
</li></ul>


<hr>
<h2 id='checkCols'>check the validation and return index of cols given from input in rawDataFrame</h2><span id='topic+checkCols'></span>

<h3>Description</h3>

<p>check the validation and return index of cols given from input in rawDataFrame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkCols(rawDataFrame, cols)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkCols_+3A_rawdataframe">rawDataFrame</code></td>
<td>
<p>raw data.frame</p>
</td></tr>
<tr><td><code id="checkCols_+3A_cols">cols</code></td>
<td>
<p>specific cols given from input</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return validation (only FALSE if invaild cols input) or index of cols
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(handyFunctions)
data(people)
checkCols(people, c("..name", "..sex"))
# OR
checkCols(people, c(1, 2))
</code></pre>

<hr>
<h2 id='checkDtype'>Return suggested dtype of vector input</h2><span id='topic+checkDtype'></span>

<h3>Description</h3>

<p>Return suggested dtype of vector input
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkDtype(vector)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkDtype_+3A_vector">vector</code></td>
<td>
<p>vector/list input</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return suggested dtypes of vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(handyFunctions)
vector &lt;- c(1, 2, 3, "", NA, "  ", "four", "NA", 5)
checkDtype(vector)

</code></pre>

<hr>
<h2 id='grade'>Grade records of virtual persons in high school</h2><span id='topic+grade'></span>

<h3>Description</h3>

<p>A dataset containing the personal grade information (chinese, math, english, physics, biology, chemistry)
of virtual persons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grade
</code></pre>


<h3>Format</h3>

<p>A data frame with 6 rows and 7 variables:
</p>

<dl>
<dt>name</dt><dd><p>name, chinese or foreigner, in carats</p>
</dd>
<dt>chinese</dt><dd><p>grade of the chinese, in numbers</p>
</dd>
<dt>math</dt><dd><p>grade of the math, in numbers</p>
</dd>
<dt>english</dt><dd><p>grade of the english, in numbers</p>
</dd>
<dt>physics</dt><dd><p>grade of the physics, in numbers</p>
</dd>
<dt>biology</dt><dd><p>grade of the biology, in numbers</p>
</dd>
<dt>chemistry</dt><dd><p>grade of the chemistry, in numbers</p>
</dd>
</dl>
<p>...
</p>



<h3>Source</h3>

<p>&quot;simulated dataset&quot;
</p>

<hr>
<h2 id='matchIndex'>Return the index of source vector matched with query vector</h2><span id='topic+matchIndex'></span>

<h3>Description</h3>

<p>Return the index of source vector matched with query vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matchIndex(SourceInfo, queryInfo, queryType = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="matchIndex_+3A_sourceinfo">SourceInfo</code></td>
<td>
<p>the source vector</p>
</td></tr>
<tr><td><code id="matchIndex_+3A_queryinfo">queryInfo</code></td>
<td>
<p>the query vector</p>
</td></tr>
<tr><td><code id="matchIndex_+3A_querytype">queryType</code></td>
<td>
<p>logical If set it to accurate match (default: TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the index of source vector matched with query vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(handyFunctions)
data(grade)
matchIndex(grade[, "name"], c("Ming Li", "Bang Wei"))

</code></pre>

<hr>
<h2 id='mergeCustom'>merge two data.frame based on xcol and ycol</h2><span id='topic+mergeCustom'></span>

<h3>Description</h3>

<p>merge two data.frame based on xcol and ycol
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeCustom(x, y, xcol, ycol)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mergeCustom_+3A_x">x</code></td>
<td>
<p>the first data.frame</p>
</td></tr>
<tr><td><code id="mergeCustom_+3A_y">y</code></td>
<td>
<p>the second data.frame</p>
</td></tr>
<tr><td><code id="mergeCustom_+3A_xcol">xcol</code></td>
<td>
<p>colnames which you want to merged in first data.frame</p>
</td></tr>
<tr><td><code id="mergeCustom_+3A_ycol">ycol</code></td>
<td>
<p>colnames which you want to merged in second data.frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return the new data.frame merged
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(handyFunctions)
data(people)
data(grade)
mergeCustom(people, grade, "..name", "name")

</code></pre>

<hr>
<h2 id='modifyColNames'>Return reformatted data.frame with standard col names</h2><span id='topic+modifyColNames'></span>

<h3>Description</h3>

<p>Return reformatted data.frame with standard col names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modifyColNames(rawDataFrame, cols = TRUE, rawSep = "..", sep = "_")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="modifyColNames_+3A_rawdataframe">rawDataFrame</code></td>
<td>
<p>Raw data.frame input</p>
</td></tr>
<tr><td><code id="modifyColNames_+3A_cols">cols</code></td>
<td>
<p>Specific col names or indexes what you want to reformat (default: TRUE, use all cols)</p>
</td></tr>
<tr><td><code id="modifyColNames_+3A_rawsep">rawSep</code></td>
<td>
<p>Raw odd separation symbol in col names of raw data.frame.
Note: it supports regEx (regular expression), so &quot;.&quot; means all possible symbols.
If you want to use the &quot;.&quot; dot notation, please use &quot;[.]&quot;.</p>
</td></tr>
<tr><td><code id="modifyColNames_+3A_sep">sep</code></td>
<td>
<p>Separation symbol in col names of modified data.frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified data.frame with col names separated by your given delimitator
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(handyFunctions)
data(people)
modified_people &lt;- modifyColNames(people,rawSep = "[.][.]")

</code></pre>

<hr>
<h2 id='modifyColTypes'>Return suggested appropriate dtypes for each column in rawDataFrame</h2><span id='topic+modifyColTypes'></span>

<h3>Description</h3>

<p>Return suggested appropriate dtypes for each column in rawDataFrame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modifyColTypes(rawDataFrame, cols = TRUE, dtype = FALSE, custom = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="modifyColTypes_+3A_rawdataframe">rawDataFrame</code></td>
<td>
<p>Raw data.frame</p>
</td></tr>
<tr><td><code id="modifyColTypes_+3A_cols">cols</code></td>
<td>
<p>Specify cols which you want to change its dtypes when custom is FALSE (default: TRUE, for all cols)</p>
</td></tr>
<tr><td><code id="modifyColTypes_+3A_dtype">dtype</code></td>
<td>
<p>Specify indexed matched dtypes whcih you want to update when custom is FALSE (default: FALSE, for automatically update)</p>
</td></tr>
<tr><td><code id="modifyColTypes_+3A_custom">custom</code></td>
<td>
<p>Option whether set to auto/custom , you can specify your custom dtypes for cols given when setting to TRUE (default: FALSE, for auto)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a new data.frame with appropriate dtypes suggested for each cols
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(handyFunctions)
data(people)
modifyColTypes(people)

</code></pre>

<hr>
<h2 id='modifyRowNames'>Return reformatted data.frame with standard row names</h2><span id='topic+modifyRowNames'></span>

<h3>Description</h3>

<p>Return reformatted data.frame with standard row names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modifyRowNames(rawDataFrame, rows = TRUE, rawSep = "..", sep = "_")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="modifyRowNames_+3A_rawdataframe">rawDataFrame</code></td>
<td>
<p>Raw data.frame input</p>
</td></tr>
<tr><td><code id="modifyRowNames_+3A_rows">rows</code></td>
<td>
<p>Specific row names or indexes what you want to reformat (default: TRUE, use all row)</p>
</td></tr>
<tr><td><code id="modifyRowNames_+3A_rawsep">rawSep</code></td>
<td>
<p>Raw odd separation symbol in row names of raw data.frame.
Note: it supports regEx (regular expression), so &quot;.&quot; means all possible symbols.
If you want to use the &quot;.&quot; dot notation, please use &quot;[.]&quot;.</p>
</td></tr>
<tr><td><code id="modifyRowNames_+3A_sep">sep</code></td>
<td>
<p>Separation symbol in row names of modified data.frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified data.frame with row names separated by your given delimitator
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(handyFunctions)
data(people)
modifyRowNames(people)

</code></pre>

<hr>
<h2 id='people'>Basic information of virtual persons</h2><span id='topic+people'></span>

<h3>Description</h3>

<p>A dataset containing the personal basic information (name, sex, age, and death_age) of virtual persons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>people
</code></pre>


<h3>Format</h3>

<p>A data frame with 6 rows and 4 variables:
</p>

<dl>
<dt>..name</dt><dd><p>name, chinese or foreigner, in carats</p>
</dd>
<dt>..sex</dt><dd><p>sex of the person, in carats</p>
</dd>
<dt>..age</dt><dd><p>living age in final record, in numbers</p>
</dd>
<dt>..death..age</dt><dd><p>final age when a person is dead, in numbers</p>
</dd>
</dl>
<p>...
</p>



<h3>Source</h3>

<p>&quot;simulated dataset&quot;
</p>

<hr>
<h2 id='queryingInfo'>return index of x data.frame with the given vector/list or ycol in data.frame
(if set the accurate match or not)</h2><span id='topic+queryingInfo'></span>

<h3>Description</h3>

<p>return index of x data.frame with the given vector/list or ycol in data.frame
(if set the accurate match or not)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>queryingInfo(SourceData, sourceCol, queryCol, queryInfo, queryType = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="queryingInfo_+3A_sourcedata">SourceData</code></td>
<td>
<p>the source data.frame which you want to query</p>
</td></tr>
<tr><td><code id="queryingInfo_+3A_sourcecol">sourceCol</code></td>
<td>
<p>the col names or index of query field in source data.frame</p>
</td></tr>
<tr><td><code id="queryingInfo_+3A_querycol">queryCol</code></td>
<td>
<p>the col names or index of return field in source data.frame</p>
</td></tr>
<tr><td><code id="queryingInfo_+3A_queryinfo">queryInfo</code></td>
<td>
<p>vector/list the query info</p>
</td></tr>
<tr><td><code id="queryingInfo_+3A_querytype">queryType</code></td>
<td>
<p>logical if set it to accurate match (default: TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector in query field matched with query info in source data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(handyFunctions)
data(grade)
queryingInfo(grade, "name", "chinese", c("Ming Li", "Bang Wei"))

</code></pre>

<hr>
<h2 id='ShowSNPDensityPlot'>Function of showing SNP density at chromosome level</h2><span id='topic+ShowSNPDensityPlot'></span>

<h3>Description</h3>

<p>Function of showing SNP density at chromosome level
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ShowSNPDensityPlot(
  densityData,
  binSize,
  densityColorBar = c("grey", "darkgreen", "yellow", "red"),
  chromSet = c(1:22),
  withchr = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ShowSNPDensityPlot_+3A_densitydata">densityData</code></td>
<td>
<p>the raw density data generated from vcftools</p>
</td></tr>
<tr><td><code id="ShowSNPDensityPlot_+3A_binsize">binSize</code></td>
<td>
<p>the bin size set while generating density data</p>
</td></tr>
<tr><td><code id="ShowSNPDensityPlot_+3A_densitycolorbar">densityColorBar</code></td>
<td>
<p>vector Specific the color bar for plotting density plot (generally four colors)</p>
</td></tr>
<tr><td><code id="ShowSNPDensityPlot_+3A_chromset">chromSet</code></td>
<td>
<p>vector Filtered chrom set which you want to plot (it must be matched with the CHROM column in densityData)</p>
</td></tr>
<tr><td><code id="ShowSNPDensityPlot_+3A_withchr">withchr</code></td>
<td>
<p>logical If the chromsome labels of density plot is prefixed with &quot;chr&quot;.
Note: it cannot work when the filtered chrom set contain other uncommon chrom symbols (e.g. NC0*, etc)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object for SNP density plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(handyFunctions)
data(SNV_1MB_density_data)
ShowSNPDensityPlot(SNV_1MB_density_data, binSize = 1e6, chromSet = c(38:1))

</code></pre>

<hr>
<h2 id='SNV_1MB_density_data'>The SNPV number within 1Mb bins at chromosome levels generated from transcriptome dataset of two dog populations
(including wild wolf and domesticated dogs).</h2><span id='topic+SNV_1MB_density_data'></span>

<h3>Description</h3>

<p>A dataset containing the SNV number within 1Mb bins called from transcriptome dataset of wild wolf and domesticated dogs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNV_1MB_density_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 2544 rows and 4 variables:
</p>

<dl>
<dt>CHROM</dt><dd><p>chrom id, reference genome of CanFam3.1, in numbers/carats</p>
</dd>
<dt>BIN_START</dt><dd><p>the start genomic coordinate for one bin at relevant chromosome, in numbers</p>
</dd>
<dt>SNP_COUNT</dt><dd><p>the end genomic coordinate for one bin at relevant chromosome, in numbers</p>
</dd>
<dt>VARIANTS.KB</dt><dd><p>SNV(variants) number within one bin per KB, in numbers</p>
</dd>
</dl>
<p>...
</p>



<h3>Source</h3>

<p>&quot;real dataset&quot;
</p>

<hr>
<h2 id='splitCol'>Return specific-indexed vector according to given delimitator/separator by splitting one col in data.frame</h2><span id='topic+splitCol'></span>

<h3>Description</h3>

<p>Return specific-indexed vector according to given delimitator/separator by splitting one col in data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitCol(data, col = FALSE, sep, index, fixed = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="splitCol_+3A_data">data</code></td>
<td>
<p>vector or data.frame input</p>
</td></tr>
<tr><td><code id="splitCol_+3A_col">col</code></td>
<td>
<p>the col names or indexes if data.frame input</p>
</td></tr>
<tr><td><code id="splitCol_+3A_sep">sep</code></td>
<td>
<p>separation deliminator</p>
</td></tr>
<tr><td><code id="splitCol_+3A_index">index</code></td>
<td>
<p>the index of symbol which you want</p>
</td></tr>
<tr><td><code id="splitCol_+3A_fixed">fixed</code></td>
<td>
<p>logical. If TRUE match split exactly, otherwise use regular expressions, detailed info can be seen in <a href="/library/base/help/strsplit">strsplit</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>specific-indexed vector or factor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(handyFunctions)
data(people)
splitCol(people, col = 1, sep = " ", index = 2)

</code></pre>

<hr>
<h2 id='unifyDataframe'>Reformat dataframe with the all modifiers simultaneously (colNames, rowNames and dtypes)</h2><span id='topic+unifyDataframe'></span>

<h3>Description</h3>

<p>Reformat dataframe with the all modifiers simultaneously (colNames, rowNames and dtypes)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unifyDataframe(
  rawDataFrame,
  rawRowSep = "..",
  rowSep = "_",
  rawColSep = "..",
  colSep = "_",
  changeDtype = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="unifyDataframe_+3A_rawdataframe">rawDataFrame</code></td>
<td>
<p>raw data.frame</p>
</td></tr>
<tr><td><code id="unifyDataframe_+3A_rawrowsep">rawRowSep</code></td>
<td>
<p>raw separation deliminator of row names in raw data.frame</p>
</td></tr>
<tr><td><code id="unifyDataframe_+3A_rowsep">rowSep</code></td>
<td>
<p>the new separation deliminator of row names</p>
</td></tr>
<tr><td><code id="unifyDataframe_+3A_rawcolsep">rawColSep</code></td>
<td>
<p>raw separation deliminator of col names in raw data.frame</p>
</td></tr>
<tr><td><code id="unifyDataframe_+3A_colsep">colSep</code></td>
<td>
<p>the new separation deliminator of col names</p>
</td></tr>
<tr><td><code id="unifyDataframe_+3A_changedtype">changeDtype</code></td>
<td>
<p>if change the dtypes of cols</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified data.frame with applied to above all modifiers
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(handyFunctions)
data(people)
unifyDataframe(people,rawColSep = "[.][.]")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
