<!DOCTYPE html><html><head><title>Help for package hapassoc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hapassoc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#anova.hapassoc'><p>Return likelihood ratio test of haplotype effect</p></a></li>
<li><a href='#hapassoc'><p>EM algorithm to fit maximum likelihood estimates of trait associations with SNP haplotypes</p></a></li>
<li><a href='#hypoDat'><p>Simulated data for a hypothetical binary trait</p></a></li>
<li><a href='#hypoDatGeno'><p>Simulated data for a hypothetical genetic SNPs</p></a></li>
<li><a href='#logLik.hapassoc'><p>Return log-likelihood</p></a></li>
<li><a href='#pre.hapassoc'><p>Pre-process the data before fitting it with hapassoc</p></a></li>
<li><a href='#summary.hapassoc'><p>Summarize results of the hapassoc function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.2-9</td>
</tr>
<tr>
<td>Title:</td>
<td>Inference of Trait Associations with SNP Haplotypes and Other
Attributes using the EM Algorithm</td>
</tr>
<tr>
<td>Author:</td>
<td>K. Burkett &lt;kburkett@uottawa.ca&gt;, B. McNeney &lt;mcneney@sfu.ca&gt;, J.
        Graham &lt;jgraham@stat.sfu.ca&gt;, with code for case-control data
        contributed by Zhijian Chen &lt;z11chen@math.uwaterloo.ca&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>K. Burkett &lt;kburkett@uottawa.ca&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.6.0), stats</td>
</tr>
<tr>
<td>Description:</td>
<td>The following R functions are used for inference of trait
        associations with haplotypes and other covariates in
        generalized linear models.  The functions are developed
        primarily for data collected in cohort or cross-sectional
        studies. They can accommodate uncertain haplotype phase and
        handle missing genotypes at some SNPs.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://sfustatgen.github.io/research/hapassoc.html">https://sfustatgen.github.io/research/hapassoc.html</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-25 15:55:55 UTC; kellyburkett</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-25 20:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='anova.hapassoc'>Return likelihood ratio test of haplotype effect</h2><span id='topic+anova.hapassoc'></span>

<h3>Description</h3>

<p>This function returns the likelihood ratio test statistic comparing
two nested models fit with <code>hapassoc</code> for cohort or cross-sectional
data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'hapassoc'
anova(object, redfit, display=TRUE, ...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anova.hapassoc_+3A_object">object</code></td>
<td>
<p>a list of class <code>hapassoc</code> output by the 
<code><a href="#topic+hapassoc">hapassoc</a></code> function</p>
</td></tr></table>
<p>.
</p>
<table>
<tr><td><code id="anova.hapassoc_+3A_redfit">redfit</code></td>
<td>
<p>A <code>hapassoc</code> object resulting from fitting a reduced 
model</p>
</td></tr>
<tr><td><code id="anova.hapassoc_+3A_display">display</code></td>
<td>
<p>An indicator to suppress output displayed on screen</p>
</td></tr> 
<tr><td><code id="anova.hapassoc_+3A_...">...</code></td>
<td>
<p>additional arguments to the summary function currently unused</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the hapassoc vignette, of the same name as the package, for details.
</p>


<h3>Value</h3>

<table>
<tr><td><code>LRTstat</code></td>
<td>
<p>The likelihood ratio statistic comparing the two models</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>Degrees of freedom of the likelihood ratio statistic</p>
</td></tr>
<tr><td><code>pvalue</code></td>
<td>
<p>The p-value of the test</p>
</td></tr>
</table>


<h3>References</h3>

<p>Burkett K, McNeney B, Graham J (2004).
A note on inference of trait associations with SNP
haplotypes and other attributes in generalized linear models.
Human Heredity, <b>57</b>:200-206
</p>
<p>Burkett K, Graham J and McNeney B (2006). hapassoc: Software for 
Likelihood Inference of Trait Associations with SNP Haplotypes and Other 
Attributes. Journal of Statistical Software, <b>16(2)</b>:1-19
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pre.hapassoc">pre.hapassoc</a></code>,<code><a href="#topic+hapassoc">hapassoc</a></code>, 
<code><a href="#topic+summary.hapassoc">summary.hapassoc</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hypoDatGeno)
example2.pre.hapassoc&lt;-pre.hapassoc(hypoDatGeno, numSNPs=3, allelic=FALSE)
example2.regr &lt;- hapassoc(affected ~ attr + hAAA+ hACA + hACC + hCAA + 
pooled, example2.pre.hapassoc, family=binomial())
example2.regr2 &lt;- hapassoc(affected ~ attr + hAAA, example2.pre.hapassoc, 
family=binomial())
anova(example2.regr,example2.regr2)

# Returns:

#	hapassoc: likelihood ratio test

#Full model: affected ~ attr + hAAA + hACA + hACC + hCAA + pooled 
#Reduced model: affected ~ attr + hAAA 

#LR statistic = 1.5433 , df = 4 , p-value =  0.8189 
</code></pre>

<hr>
<h2 id='hapassoc'>EM algorithm to fit maximum likelihood estimates of trait associations with SNP haplotypes</h2><span id='topic+hapassoc'></span>

<h3>Description</h3>

<p>This function takes a dataset of haplotypes in which rows for
individuals of uncertain phase have been augmented by &ldquo;pseudo-individuals&rdquo;
who carry the possible multilocus genotypes consistent with
the single-locus phenotypes. For cohort or cross-sectional
data, the EM algorithm is used to find
MLE's for trait associations with covariates in generalized linear models.
For case-control data, the algorithm solves a set of unbiased
estimating equations (see <b>Details</b>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapassoc(form,haplos.list,baseline = "missing" ,family = binomial(),
design="cohort",disease.prob=NULL,freq = NULL, maxit = 50, tol = 0.001, 
start = NULL, verbose=FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hapassoc_+3A_form">form</code></td>
<td>
<p>model equation in usual R format</p>
</td></tr>
<tr><td><code id="hapassoc_+3A_haplos.list">haplos.list</code></td>
<td>
<p>list of haplotype data from <code><a href="#topic+pre.hapassoc">pre.hapassoc</a></code></p>
</td></tr>
<tr><td><code id="hapassoc_+3A_baseline">baseline</code></td>
<td>
<p>optional, haplotype to be used for baseline coding if the model formula
either includes all haplotypes or is of the form &quot;y~.&quot; for example. Default
is the most frequent haplotype according to the initial haplotype frequency
estimates returned by <code><a href="#topic+pre.hapassoc">pre.hapassoc</a></code>.</p>
</td></tr>
<tr><td><code id="hapassoc_+3A_family">family</code></td>
<td>
<p>binomial, poisson, gaussian or gamma are supported, 
default=binomial</p>
</td></tr>
<tr><td><code id="hapassoc_+3A_design">design</code></td>
<td>
<p>study design. Default is &ldquo;cohort&rdquo; for cohort or 
cross-sectional sampling. Users may optionally specify &ldquo;cc&rdquo; for 
case-control or retrospective sampling of exposures (i.e. genotypes and 
non-genetic attributes) conditional on disease status. 
When <code>design="cc"</code>, <code>family=binomial()</code> is assumed
and the robust MPSE estimator of the regression parameters 
(Spinka et al., 2005) is returned; see <b>Details</b> for more information.
</p>
</td></tr>
<tr><td><code id="hapassoc_+3A_disease.prob">disease.prob</code></td>
<td>
<p>marginal disease probability [P(D=1)]  to use in 
the MPSE estimator, if <code>design="cc"</code>. If <code>disease.prob=NULL</code> 
(the default value), a rare disease is assumed.
This argument is ignored if <code>design="cohort"</code>.</p>
</td></tr>
<tr><td><code id="hapassoc_+3A_freq">freq</code></td>
<td>
<p>initial estimates of haplotype frequencies, default values are 
calculated in <code><a href="#topic+pre.hapassoc">pre.hapassoc</a></code> using standard haplotype-counting 
(i.e. EM algorithm without adjustment for non-haplotype covariates)</p>
</td></tr>
<tr><td><code id="hapassoc_+3A_maxit">maxit</code></td>
<td>
<p>maximum number of iterations of the EM algorithm; default=50</p>
</td></tr>
<tr><td><code id="hapassoc_+3A_tol">tol</code></td>
<td>
<p>convergence tolerance in terms of either the maximum difference in 
parameter estimates between interations or the maximum relative
difference in parameter estimates between iterations, which ever is larger.</p>
</td></tr>
<tr><td><code id="hapassoc_+3A_start">start</code></td>
<td>
<p>starting values for parameter estimates in the risk model</p>
</td></tr>
<tr><td><code id="hapassoc_+3A_verbose">verbose</code></td>
<td>
<p>should the iteration number and value of the covergence criterion be printed
at each iteration of the EM algorithm? Default=FALSE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the hapassoc vignette, of the same name as the package, for details.
</p>
<p>When the study design is case-control, i.e. genotypes and
non-genetic attributes have been sampled retrospectively given disease
status, naive application of prospective maximum likelihood methods
can yield biased inference (Spinka et al., 2005, Chen, 2006). 
Therefore, when <code>design="cc"</code>, 
the algorithm solves the modified prospective score equations or MPSE
(Spinka et al. 2005) for regression
and haplotype frequency parameters. The implementation in
<span class="pkg">hapassoc</span> is due to Chen (2006). 
In general, the MPSE approach requires that
the marginal probability of disease,  P(D=1), be known. 
An exception is when the disease is rare; hence, when 
<code>disease.prob=NULL</code> (the default) a rare disease is assumed.
The variance-covariance matrix of the regression parameter and
haplotype frequency estimators is approximated as described
in Chen (2006). Limited simulations indicate that the resulting standard errors
for regression parameters perform well, but not the
standard errors for haplotype frequencies, which should be ignored.
For case-control data, we hope to
implement the variance-covariance estimator of Spinka et al. (2005)
in a future version of <span class="pkg">hapassoc</span>.
</p>


<h3>Value</h3>

<table>
<tr><td><code>it</code></td>
<td>
<p>number of iterations of the EM algorithm</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>estimated regression coefficients</p>
</td></tr>
<tr><td><code>freq</code></td>
<td>
<p>estimated haplotype frequencies</p>
</td></tr>
<tr><td><code>fits</code></td>
<td>
<p>fitted values of the trait</p>
</td></tr>
<tr><td><code>wts</code></td>
<td>
<p>final weights calculated in last iteration of the EM algorithm. These are estimates of the
conditional probabilities of each multilocus genotype given the observed 
single-locus genotypes.</p>
</td></tr>
<tr><td><code>var</code></td>
<td>
<p>joint variance-covariance matrix of the estimated regression
coefficients and the estimated haplotype frequencies</p>
</td></tr>
<tr><td><code>dispersion</code></td>
<td>
<p>maximum likelihood estimate of dispersion parameter 
(to get the moment estimate, use <code><a href="#topic+summary.hapassoc">summary.hapassoc</a></code>)
if applicable, otherwise 1</p>
</td></tr>
<tr><td><code>family</code></td>
<td>
<p>family of the generalized linear model (e.g. binomial, 
gaussian, etc.)</p>
</td></tr>
<tr><td><code>response</code></td>
<td>
<p>trait value</p>
</td></tr>
<tr><td><code>converged</code></td>
<td>
<p>TRUE/FALSE indicator of convergence. If the algorithm 
fails to converge, only the <code>converged</code> indicator is returned.</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>model equation</p>
</td></tr>
<tr><td><code>loglik</code></td>
<td>
<p>the log-likelihood evaluated at the maximum likelihood estimates 
of all parameters if <code>design="cohort"</code>, or <code>NA</code> if 
<code>design="cc"</code></p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>the function call</p>
</td></tr>
</table>


<h3>References</h3>

<p>Burkett K, McNeney B, Graham J (2004).
A note on inference of trait associations with SNP
haplotypes and other attributes in generalized linear models.
Human Heredity, <b>57</b>:200-206
</p>
<p>Burkett K, Graham J and McNeney B (2006). hapassoc: Software for
Likelihood Inference of Trait Associations with SNP Haplotypes and Other
Attributes. Journal of Statistical Software, <b>16(2)</b>:1-19
</p>
<p>Chen, Z. (2006): Approximate likelihood inference for haplotype risks
in case-control studies of a rare disease, Masters thesis, Statistics
and Actuarial Science, Simon Fraser University, available at
<a href="https://www.stat.sfu.ca/content/dam/sfu/stat/alumnitheses/MiscellaniousTheses/Chen-2006.pdf">https://www.stat.sfu.ca/content/dam/sfu/stat/alumnitheses/MiscellaniousTheses/Chen-2006.pdf</a>.
</p>
<p>Spinka, C., Carroll, R. J. &amp; Chatterjee, N. (2005). Analysis of 
case-control studies of genetic and environmental factors with missing 
genetic information and haplotype-phase ambiguity. 
Genetic Epidemiology, <b>29</b>, 108-127. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pre.hapassoc">pre.hapassoc</a></code>,<code><a href="#topic+summary.hapassoc">summary.hapassoc</a></code>,<code><a href="stats.html#topic+glm">glm</a></code>,<code><a href="stats.html#topic+family">family</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hypoDat)
example.pre.hapassoc&lt;-pre.hapassoc(hypoDat, 3)

example.pre.hapassoc$initFreq # look at initial haplotype frequencies
#      h000       h001       h010       h011       h100       h101       h110 
#0.25179111 0.26050418 0.23606001 0.09164470 0.10133627 0.02636844 0.01081260 
#      h111 
#0.02148268 


names(example.pre.hapassoc$haploDM)
# "h000"   "h001"   "h010"   "h011"   "h100"   "pooled"

# Columns of the matrix haploDM score the number of copies of each haplotype 
# for each pseudo-individual.

# Logistic regression for a multiplicative odds model having as the baseline 
# group homozygotes '001/001' for the most common haplotype

example.regr &lt;- hapassoc(affected ~ attr + h000+ h010 + h011 + h100 + pooled,
                  example.pre.hapassoc, family=binomial())

# Logistic regression with separate effects for 000 homozygotes, 001 homozygotes 
# and 000/001 heterozygotes

example2.regr &lt;- hapassoc(affected ~ attr + I(h000==2) + I(h001==2) +
                   I(h000==1 &amp; h001==1), example.pre.hapassoc, family=binomial())

</code></pre>

<hr>
<h2 id='hypoDat'>Simulated data for a hypothetical binary trait</h2><span id='topic+hypoDat'></span>

<h3>Description</h3>

<p>Simulated binary trait data used to illustrate the hapassoc package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hypoDat)</code></pre>


<h3>Format</h3>

<p>Matrix with columns:\
</p>

<table>
<tr>
 <td style="text-align: right;">
[,1] </td><td style="text-align: left;"> affected </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> affection status (1=yes, 0=no) </td>
</tr>
<tr>
 <td style="text-align: right;">
[,3] </td><td style="text-align: left;"> attr </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> simulated quantitative attribute </td>
</tr>
<tr>
 <td style="text-align: right;">
[,5] </td><td style="text-align: left;"> M1.1 </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> the first allele of hypothetical SNP M1 </td>
</tr>
<tr>
 <td style="text-align: right;">
[,6] </td><td style="text-align: left;"> M1.2 </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> the second allele of hypothetical SNP M1 </td>
</tr>
<tr>
 <td style="text-align: right;">
[,5] </td><td style="text-align: left;"> M2.1 </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> the first allele of hypothetical SNP M2 </td>
</tr>
<tr>
 <td style="text-align: right;">
[,6] </td><td style="text-align: left;"> M2.2 </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> the second allele of hypothetical SNP M2 </td>
</tr>
<tr>
 <td style="text-align: right;">
[,7] </td><td style="text-align: left;"> M3.1 </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> the first allele of hypothetical SNP M3 </td>
</tr>
<tr>
 <td style="text-align: right;">
[,8] </td><td style="text-align: left;"> M3.2 </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> the second allele of hypotetical SNP  M3 </td>
</tr>

</table>


<hr>
<h2 id='hypoDatGeno'>Simulated data for a hypothetical genetic SNPs</h2><span id='topic+hypoDatGeno'></span>

<h3>Description</h3>

<p>Simulated genetic SNPs data used to illustrate the hapassoc package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hypoDatGeno)</code></pre>


<h3>Format</h3>

<p>Matrix with columns:\
</p>

<table>
<tr>
 <td style="text-align: right;">
[,1] </td><td style="text-align: left;"> affected </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> affection status (1=yes, 0=no) </td>
</tr>
<tr>
 <td style="text-align: right;">
[,2] </td><td style="text-align: left;"> attr </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> simulated quantitative attribute </td>
</tr>
<tr>
 <td style="text-align: right;">
[,3] </td><td style="text-align: left;"> M1 </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> hypothetical SNP M1 </td>
</tr>
<tr>
 <td style="text-align: right;">
[,4] </td><td style="text-align: left;"> M2 </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> hypothetical SNP M2 </td>
</tr>
<tr>
 <td style="text-align: right;">
[,5] </td><td style="text-align: left;"> M3 </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> hypothetical SNP M3 </td>
</tr>
<tr>
 <td style="text-align: right;">  </td>
</tr>

</table>


<hr>
<h2 id='logLik.hapassoc'>Return log-likelihood</h2><span id='topic+logLik.hapassoc'></span>

<h3>Description</h3>

<p>This function is used to return the log-likelihood at the maximum 
likelihood estimates computed by <code>hapassoc</code> and to return
the number of parameters fit by <code>hapassoc</code> 
(i.e. the degrees of freedom in <code>R</code>) for cohort or
cross-sectional data. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'hapassoc'
logLik(object, ...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logLik.hapassoc_+3A_object">object</code></td>
<td>
<p>a list of class <code>hapassoc</code> output by the 
<code><a href="#topic+hapassoc">hapassoc</a></code> function</p>
</td></tr>
<tr><td><code id="logLik.hapassoc_+3A_...">...</code></td>
<td>
<p>additional arguments to the summary 
function (currently unused)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the hapassoc vignette, of the same name as the package, for details.
</p>


<h3>Value</h3>

<table>
<tr><td><code>logLik</code></td>
<td>
<p>log-likelihood computed at the maximum likelihood estimates
if <code>design="cohort"</code>, or <code>NA</code> if <code>design="cc"</code></p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>number of parameters in the model (i.e. regression coefficients,
any dispersion parameters and haplotype 
frequencies). This is not the residual
degrees of freedom, which is the number of subjects minus the number
of parameters estimated.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Burkett K, McNeney B, Graham J (2004).
A note on inference of trait associations with SNP
haplotypes and other attributes in generalized linear models.
Human Heredity, <b>57</b>:200-206
</p>
<p>Burkett K, Graham J and McNeney B (2006). hapassoc: Software for
Likelihood Inference of Trait Associations with SNP Haplotypes and Other
Attributes. Journal of Statistical Software, <b>16(2)</b>:1-19
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pre.hapassoc">pre.hapassoc</a></code>,<code><a href="#topic+hapassoc">hapassoc</a></code>,
<code><a href="#topic+summary.hapassoc">summary.hapassoc</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hypoDatGeno)
example2.pre.hapassoc&lt;-pre.hapassoc(hypoDatGeno, numSNPs=3, allelic=FALSE)
example.regr &lt;- hapassoc(affected ~ attr + hAAA+ hACA + hACC + hCAA + 
pooled, example2.pre.hapassoc, family=binomial())
logLik(example.regr)

# Returns:
# Log Lik: -322.1558 (df=14)
</code></pre>

<hr>
<h2 id='pre.hapassoc'>Pre-process the data before fitting it with hapassoc</h2><span id='topic+pre.hapassoc'></span>

<h3>Description</h3>

<p>This function takes as an argument a dataframe with
non-SNP and SNP data and converts the genotype data at single SNPs
(the single-locus genotypes) into haplotype data.
The rows of the input data frame should correspond to
subjects. Single-locus SNP genotypes  may be specified in 
one of two ways: (i) as pairs of columns, with one column for
each allele of the single-locus genotypes (&ldquo;allelic format&rdquo;),
or (ii) as columns of two-character genotypes (&ldquo;genotypic format&rdquo;).
The SNP data should comprise the
last 2*numSNPs columns (allelic format) or the last numSNPs columns (genotypic
format) of the data frame. 
</p>
<p>If the haplotypes for a subject cannot be
inferred from his or her genotype data, &ldquo;pseudo-individuals&rdquo;
representing all possible haplotype combinations consistent with
the single-locus genotypes are considered.
Missing single-locus genotypes, up to a maximum of maxMissingGenos (see
below), are allowed, but subjects with missing data in more than
maxMissingGenos, or with missing non-SNP data, are removed.
Initial estimates of haplotype frequencies are then obtained using the 
EM algorithm applied to the genotype data pooled over all subjects.
Haplotypes with frequencies below a user-specified tolerance (zero.tol)
are assumed not to exist and are removed from further consideration.
(Pseudo-individuals having haplotypes of negligible frequency are deleted and 
the column in the design matrix corresponding to that haplotype is deleted.)
For the remaining haplotypes, those with non-negligible frequency below a 
user-defined pooling tolerance (pooling.tol) are pooled into a single 
category called &ldquo;pooled&rdquo; in the design matrix for the risk model. 
However, the frequencies of each of these pooled haplotypes are 
still calculated separately. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pre.hapassoc(dat,numSNPs,maxMissingGenos=1,pooling.tol = 0.05, 
       zero.tol = 1/(2 * nrow(dat) * 10), allelic=TRUE, verbose=TRUE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pre.hapassoc_+3A_dat">dat</code></td>
<td>
<p>the non-SNP and SNP data as a data frame. The SNP data should comprise the last 2*numSNPs columns (allelic format) or last numSNPs columns (genotypic format).
Missing allelic data should be coded as <code>NA</code> or <code>""</code> and missing genotypic data should be coded as, e.g., <code>"A"</code> if one allele is missing and <code>""</code> if both alleles are missing.  </p>
</td></tr>
<tr><td><code id="pre.hapassoc_+3A_numsnps">numSNPs</code></td>
<td>
<p>number of SNPs per haplotype</p>
</td></tr>
<tr><td><code id="pre.hapassoc_+3A_maxmissinggenos">maxMissingGenos</code></td>
<td>
<p>maximum number of single-locus genotypes with missing data to allow for each subject. (Subjects with more missing data, or with missing non-SNP data are removed.) The default is 1.</p>
</td></tr>
<tr><td><code id="pre.hapassoc_+3A_pooling.tol">pooling.tol</code></td>
<td>
<p>pooling tolerance &ndash; by default set to 0.05</p>
</td></tr>
<tr><td><code id="pre.hapassoc_+3A_zero.tol">zero.tol</code></td>
<td>
<p>tolerance for haplotype frequencies below which haplotypes
are assumed not to exist &ndash; by default set to 
<code class="reqn">\frac{1}{2*N*10}</code> where N is the number of subjects</p>
</td></tr>
<tr><td><code id="pre.hapassoc_+3A_allelic">allelic</code></td>
<td>
<p>TRUE if single-locus SNP genotypes are in allelic format and FALSE if in genotypic format; default is TRUE.</p>
</td></tr>
<tr><td><code id="pre.hapassoc_+3A_verbose">verbose</code></td>
<td>
<p>indicates whether or not a list of the genotype
variables used to form haplotypes and a list of other non-genetic
variables should be printed; default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the hapassoc vignette, of the same name as the package, for details.
</p>


<h3>Value</h3>

<table>
<tr><td><code>haplotest</code></td>
<td>
<p>logical, TRUE if some haplotypes had frequency less than <code>zero.tol</code> and are assumed not to exist</p>
</td></tr>
<tr><td><code>initFreq</code></td>
<td>
<p>initial estimates of haplotype frequencies</p>
</td></tr>
<tr><td><code>zeroFreqHaplos</code></td>
<td>
<p>list of haplotypes assumed not to exist</p>
</td></tr>
<tr><td><code>pooledHaplos</code></td>
<td>
<p>list of haplotypes pooled into a single category in the design matrix</p>
</td></tr>
<tr><td><code>haploDM</code></td>
<td>
<p>Haplotype portion of the data frame <strong>augmented</strong> with pseudo-individuals. Has <code class="reqn">2^{numSNPs}</code> columns scoring number of copies of each haplotype for each pseudo-individual</p>
</td></tr>
<tr><td><code>nonHaploDM</code></td>
<td>
<p>non-haplotype portion of the data frame <strong>augmented</strong> with pseudo-individuals</p>
</td></tr>
<tr><td><code>haploMat</code></td>
<td>
<p>matrix with 2 columns listing haplotype labels for each pseudo-individual</p>
</td></tr>
<tr><td><code>wt</code></td>
<td>
<p>vector giving initial weights for each pseudo-individual for 
the EM algorithm</p>
</td></tr>
<tr><td><code>ID</code></td>
<td>
<p>index for each individual in the original data frame. Note that all pseudo-individuals have the same ID value</p>
</td></tr>
</table>


<h3>References</h3>

<p>Burkett K, McNeney B, Graham J (2004). 
A note on inference of trait associations with SNP
haplotypes and other attributes in generalized linear models.
Human Heredity, <b>57</b>:200-206
</p>
<p>Burkett K, Graham J and McNeney B (2006). hapassoc: Software for 
Likelihood Inference of Trait Associations with SNP Haplotypes and 
Other Attributes. Journal of Statistical Software, <b>16(2)</b>:1-19
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hapassoc">hapassoc</a></code>,<code><a href="#topic+summary.hapassoc">summary.hapassoc</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#First example data set has single-locus genotypes in "allelic format"
data(hypoDat)
example.pre.hapassoc&lt;-pre.hapassoc(hypoDat, numSNPs=3)

# To get the initial haplotype frequencies:
example.pre.hapassoc$initFreq
#      h000       h001       h010       h011       h100       h101       h110 
#0.25179111 0.26050418 0.23606001 0.09164470 0.10133627 0.02636844 0.01081260 
#      h111 
#0.02148268 
# The '001' haplotype is estimated to be the most frequent

example.pre.hapassoc$pooledHaplos
# "h101" "h110" "h111"
# These haplotypes are to be pooled in the design matrix for the risk model

names(example.pre.hapassoc$haploDM)
# "h000"   "h001"   "h010"   "h011"   "h100"   "pooled"

####
#Second example data set has single-locus genotypes in "genotypic format"
data(hypoDatGeno)
example2.pre.hapassoc&lt;-pre.hapassoc(hypoDatGeno, numSNPs=3, allelic=FALSE)

# To get the initial haplotype frequencies:
example2.pre.hapassoc$initFreq
#      hAAA       hAAC       hACA       hACC       hCAA       hCAC
#0.25179111 0.26050418 0.23606001 0.09164470 0.10133627 0.02636844
#      hCCA           hCCC 
#0.01081260	0.02148268 
# The 'hAAC' haplotype is estimated to be the most frequent

example2.pre.hapassoc$pooledHaplos
#  "hCAC" "hCCA" "hCCC"
# These haplotypes are to be pooled in the design matrix for the risk model

names(example2.pre.hapassoc$haploDM)
#  "hAAA"   "hAAC"   "hACA"   "hACC"   "hCAA"   "pooled"
</code></pre>

<hr>
<h2 id='summary.hapassoc'>Summarize results of the hapassoc function</h2><span id='topic+summary.hapassoc'></span>

<h3>Description</h3>

<p>Summary function for reporting the results of the hapassoc function in a similar  
style to the lm and glm summaries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'hapassoc'
summary(object, ...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.hapassoc_+3A_object">object</code></td>
<td>
<p>a list of class <code>hapassoc</code> output by the <code><a href="#topic+hapassoc">hapassoc</a></code> function</p>
</td></tr>
<tr><td><code id="summary.hapassoc_+3A_...">...</code></td>
<td>
<p>additional arguments to the summary function (currently unused)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the hapassoc vignette, of the same name as the package, for details.
</p>


<h3>Value</h3>

<table>
<tr><td><code>call</code></td>
<td>
<p>The function call to hapassoc</p>
</td></tr>
<tr><td><code>subjects</code></td>
<td>
<p>The number of subjects used in the analysis</p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p>Table of estimated coefficients, standard errors and Wald tests for each variable</p>
</td></tr>
<tr><td><code>frequencies</code></td>
<td>
<p>Table of estimated haplotype frequencies and standard errors</p>
</td></tr>
<tr><td><code>dispersion</code></td>
<td>
<p>Estimate of dispersion parameter (Moment estimator for gamma model)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Burkett K, McNeney B, Graham J (2004).
A note on inference of trait associations with SNP
haplotypes and other attributes in generalized linear models.
Human Heredity, <b>57</b>:200-206
</p>
<p>Burkett K, Graham J and McNeney B (2006). hapassoc: Software for
Likelihood Inference of Trait Associations with SNP Haplotypes and Other
Attributes. Journal of Statistical Software, <b>16(2)</b>:1-19
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pre.hapassoc">pre.hapassoc</a></code>,<code><a href="#topic+hapassoc">hapassoc</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hypoDat)
example.pre.hapassoc&lt;-pre.hapassoc(hypoDat, 3)
example.regr &lt;- hapassoc(affected ~ attr + h000+ h010 + h011 + h100 + pooled,
                     example.pre.hapassoc, family=binomial())

# Summarize the results:
summary(example.regr)

# Results:
#$coefficients
#               Estimate Std. Error      zscore   Pr(&gt;|z|)
#(Intercept) -1.24114270  0.7820977 -1.58694079 0.11252606
#attr         0.74036920  0.2918205  2.53707057 0.01117844
#h000         1.14968352  0.5942542  1.93466627 0.05303126
#h010        -0.59318434  0.6569672 -0.90291311 0.36657201
#h011        -0.03615243  0.9161959 -0.03945928 0.96852422
#h100        -0.85329292  1.0203105 -0.83630709 0.40298217
#pooled       0.38516864  0.8784283  0.43847478 0.66104215
#
#$frequencies
#         Estimate Std. Error
#f.h000 0.26716394 0.03933158
#f.h001 0.25191674 0.03866739
#f.h010 0.21997138 0.03881578
#f.h011 0.10094795 0.02949617
#f.h100 0.09507014 0.02371878
#f.h101 0.02584918 0.01411881
#f.h110 0.01779455 0.01386080
#f.h111 0.02128613 0.01247265
#
#$dispersion
#[1] 1
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
