<!DOCTYPE html><html lang="en"><head><title>Help for package Hapi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Hapi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Hapi-package'><p>Hapi is a novel easy-to-use package that only requires 3 to 5 gametes</p>
to reconstruct accurate and high-resolution haplotypes of an individual.
The gamete genotype data may be generated from various platforms including
genotyping arrays and next generation sequencing even with low-coverage.
Hapi simply takes genotype data of known hetSNPs in single gamete cells
as input and report the high-resolution haplotypes as well as confidence
level of each phased hetSNPs. The package also includes a module allowing
downstream analyses and visualization of crossovers in the gametes.</a></li>
<li><a href='#base2num'><p>Convert genotype coded in A/T/C/G to 0/1</p></a></li>
<li><a href='#crossover'><p>Crossover information across all gamete cells</p></a></li>
<li><a href='#gamete11'><p>Haplotypes of a single gamete cell for visualization</p></a></li>
<li><a href='#gmt'><p>Raw genotyping data</p></a></li>
<li><a href='#hapiAssemble'><p>Consensus haplotype assembly</p></a></li>
<li><a href='#hapiAssembleEnd'><p>Assembly of haplotypes in regions at the end of a chromosome</p></a></li>
<li><a href='#hapiAutoPhase'><p>Automatic inference of haplotypes</p></a></li>
<li><a href='#hapiBlockMPR'><p>Maximum Parsimony of Recombination (MPR) for</p>
proofreading of draft haplotypes</a></li>
<li><a href='#hapiCVCluster'><p>Filter out hetSNPs in potential complex regions</p></a></li>
<li><a href='#hapiCVDistance'><p>Histogram of crossover distance</p></a></li>
<li><a href='#hapiCVMap'><p>Visualization of crossover map</p></a></li>
<li><a href='#hapiCVResolution'><p>Histogram of crossover resolution</p></a></li>
<li><a href='#hapiFilterError'><p>Filter out hetSNPs with potential genotyping errors</p></a></li>
<li><a href='#hapiFrameSelection'><p>Selection of hetSNPs to form a framework</p></a></li>
<li><a href='#hapiGameteView'><p>Visualization of haplotypes in a single gamete cell</p></a></li>
<li><a href='#hapiIdentifyCV'><p>Indentify crossovers in gamete cells</p></a></li>
<li><a href='#hapiImupte'><p>Imputation of missing genotypes in the framework</p></a></li>
<li><a href='#hapiPhase'><p>Phase draft haplotypes by majority voting</p></a></li>
<li><a href='#hg19'><p>Chromosome information of hg19</p></a></li>
<li><a href='#num2base'><p>Convert genotype coded in 0/1 to A/T/C/G</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Inference of Chromosome-Length Haplotypes Using Genomic Data of
Single Gamete Cells</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Ruidong Li,
	Han Qu,
	Jinfeng Chen,
	Shibo Wang,
	Le Zhang,
	Julong Wei,
	Sergio Pietro Ferrante,
	Mikeal L. Roose,
	Zhenyu Jia</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ruidong Li &lt;rli012@ucr.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Inference of chromosome-length haplotypes using a few haploid 
	gametes of an individual. The gamete genotype data may be generated from various platforms 
	including genotyping arrays and sequencing even with low-coverage. Hapi simply takes 
	genotype data of known hetSNPs in single gamete cells as input and report the high-resolution 
	haplotypes as well as confidence of each phased hetSNPs. The package also includes a module 
	allowing downstream analyses and visualization of identified crossovers in the gametes. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>Imports:</td>
<td>HMM, ggplot2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>biocViews:</td>
<td>SNP, GenomicVariation, Genetics, HiddenMarkovModel,
SingleCell, Sequencing, Microarray</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-07-23 04:15:32 UTC; ruidong</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-07-28 15:10:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='Hapi-package'>Hapi is a novel easy-to-use package that only requires 3 to 5 gametes 
to reconstruct accurate and high-resolution haplotypes of an individual. 
The gamete genotype data may be generated from various platforms including 
genotyping arrays and next generation sequencing even with low-coverage. 
Hapi simply takes genotype data of known hetSNPs in single gamete cells 
as input and report the high-resolution haplotypes as well as confidence 
level of each phased hetSNPs. The package also includes a module allowing 
downstream analyses and visualization of crossovers in the gametes.</h2><span id='topic+Hapi-package'></span><span id='topic+Hapi'></span>

<h3>Description</h3>

<p>Hapi is a novel easy-to-use package that only requires 3 to 5 gametes 
to reconstruct accurate and high-resolution haplotypes of an individual. 
The gamete genotype data may be generated from various platforms including 
genotyping arrays and next generation sequencing even with low-coverage. 
Hapi simply takes genotype data of known hetSNPs in single gamete cells 
as input and report the high-resolution haplotypes as well as confidence 
level of each phased hetSNPs. The package also includes a module allowing 
downstream analyses and visualization of crossovers in the gametes.
</p>

<hr>
<h2 id='base2num'>Convert genotype coded in A/T/C/G to 0/1</h2><span id='topic+base2num'></span>

<h3>Description</h3>

<p>Convert base (A/T/C/G) coded genotype to numeric (0/1) coded
</p>


<h3>Usage</h3>

<pre><code class='language-R'>base2num(gmt, ref, alt)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="base2num_+3A_gmt">gmt</code></td>
<td>
<p>a dataframe of genotype data of gamete cells</p>
</td></tr>
<tr><td><code id="base2num_+3A_ref">ref</code></td>
<td>
<p>a character represents reference allele</p>
</td></tr>
<tr><td><code id="base2num_+3A_alt">alt</code></td>
<td>
<p>a character represents alternative allele</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe containing converted genotype
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref &lt;- sample(c('A','T'),500, replace=TRUE)
alt &lt;- sample(c('C','G'),500, replace=TRUE)

gmt &lt;- data.frame(chr=rep(1,500), pos=seq_len(500),
    ref=ref, alt=alt, gmt1=ref, gmt2=alt, gmt3=ref,
    gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
    stringsAsFactors = FALSE)
    
gmtDa &lt;- base2num(gmt=gmt[5:9], ref=ref, alt=alt)
</code></pre>

<hr>
<h2 id='crossover'>Crossover information across all gamete cells</h2><span id='topic+crossover'></span>

<h3>Description</h3>

<p>Crossover information across all gamete cells
</p>

<hr>
<h2 id='gamete11'>Haplotypes of a single gamete cell for visualization</h2><span id='topic+gamete11'></span>

<h3>Description</h3>

<p>Haplotypes of a single gamete cell for visualization
</p>

<hr>
<h2 id='gmt'>Raw genotyping data</h2><span id='topic+gmt'></span>

<h3>Description</h3>

<p>Raw genotyping data
</p>

<hr>
<h2 id='hapiAssemble'>Consensus haplotype assembly</h2><span id='topic+hapiAssemble'></span>

<h3>Description</h3>

<p>Assemble the consensus high-resolution haplotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiAssemble(gmt, draftHap, keepLowConsistency = TRUE,
  consistencyThresh = 0.85)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiAssemble_+3A_gmt">gmt</code></td>
<td>
<p>a dataframe of genotype data of gamete cells</p>
</td></tr>
<tr><td><code id="hapiAssemble_+3A_drafthap">draftHap</code></td>
<td>
<p>a dataframe with draft haplotype information</p>
</td></tr>
<tr><td><code id="hapiAssemble_+3A_keeplowconsistency">keepLowConsistency</code></td>
<td>
<p>logical, if low-consistent gamete cells 
should be kept</p>
</td></tr>
<tr><td><code id="hapiAssemble_+3A_consistencythresh">consistencyThresh</code></td>
<td>
<p>a numeric value of the threshold determining 
low-consistent gamete cells compared with the draft haplotype. 
Default is 0.85</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe containing phased haplotypes
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>finalDraft &lt;- rep(0,500)
names(finalDraft) &lt;- seq_len(500)

ref &lt;- rep(0,500)
alt &lt;- rep(1,500)

gmtDa &lt;- data.frame(gmt1=ref, gmt2=alt, gmt3=ref,
gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
stringsAsFactors = FALSE)

idx1 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))
idx2 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))
idx3 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))

gmtDa[idx1,1] &lt;- NA
gmtDa[idx2,2] &lt;- NA
gmtDa[idx3,3] &lt;- NA

consensusHap &lt;- hapiAssemble(draftHap = finalDraft, gmt = gmtDa)
</code></pre>

<hr>
<h2 id='hapiAssembleEnd'>Assembly of haplotypes in regions at the end of a chromosome</h2><span id='topic+hapiAssembleEnd'></span>

<h3>Description</h3>

<p>Assembly of haplotypes in regions at the end of a chromosome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiAssembleEnd(gmt, draftHap, consensusHap, k = 300)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiAssembleEnd_+3A_gmt">gmt</code></td>
<td>
<p>a dataframe of genotype data of gamete cells</p>
</td></tr>
<tr><td><code id="hapiAssembleEnd_+3A_drafthap">draftHap</code></td>
<td>
<p>a dataframe with draft haplotype information</p>
</td></tr>
<tr><td><code id="hapiAssembleEnd_+3A_consensushap">consensusHap</code></td>
<td>
<p>a dataframe of the consensus haplotype information</p>
</td></tr>
<tr><td><code id="hapiAssembleEnd_+3A_k">k</code></td>
<td>
<p>a numeric value for the number of hetSNPs that will be combined
with markers beyond the framework for assembly. Default is 300</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe containing phased haplotypes
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>finalDraft &lt;- rep(0,500)
names(finalDraft) &lt;- seq_len(500)

ref &lt;- rep(0,500)
alt &lt;- rep(1,500)

gmtDa &lt;- data.frame(gmt1=ref, gmt2=alt, gmt3=ref,
gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
stringsAsFactors = FALSE)

idx1 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))
idx2 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))
idx3 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))

gmtDa[idx1,1] &lt;- NA
gmtDa[idx2,2] &lt;- NA
gmtDa[idx3,3] &lt;- NA

consensusHap &lt;- data.frame(hap1=rep(0,500),hap2=rep(1,500),
total=rep(5,500),rate=rep(1,500),
confidence=rep('F',500),
stringsAsFactors = FALSE)
rownames(consensusHap) &lt;- seq_len(500)

consensusHap &lt;- hapiAssembleEnd(gmt = gmtDa, draftHap = finalDraft, 
consensusHap = consensusHap, k = 300)
</code></pre>

<hr>
<h2 id='hapiAutoPhase'>Automatic inference of haplotypes</h2><span id='topic+hapiAutoPhase'></span>

<h3>Description</h3>

<p>Automatic inference of haplotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiAutoPhase(gmt, code = "atcg")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiAutoPhase_+3A_gmt">gmt</code></td>
<td>
<p>a dataframe of genotype data of gamete cells</p>
</td></tr>
<tr><td><code id="hapiAutoPhase_+3A_code">code</code></td>
<td>
<p>a character indicating the code style of genotype data. 
One of <code>'atcg'</code> and <code>'01'</code>. Default is <code>'atcg'</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of inferred consensus haplotypes
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref &lt;- sample(c('A','T'),500, replace=TRUE)
alt &lt;- sample(c('C','G'),500, replace=TRUE)

gmt &lt;- data.frame(chr=rep(1,500), pos=seq_len(500),
    ref=ref, alt=alt, gmt1=ref, gmt2=alt, gmt3=ref,
    gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
    stringsAsFactors = FALSE)
    
hapOutput &lt;- hapiAutoPhase(gmt=gmt, code='atcg')
</code></pre>

<hr>
<h2 id='hapiBlockMPR'>Maximum Parsimony of Recombination (MPR) for 
proofreading of draft haplotypes</h2><span id='topic+hapiBlockMPR'></span>

<h3>Description</h3>

<p>Maximum Parsimony of Recombination (MPR) for 
proofreading of draft haplotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiBlockMPR(draftHap, gmtFrame, cvlink = 2, smallBlock = 100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiBlockMPR_+3A_drafthap">draftHap</code></td>
<td>
<p>a dataframe with draft haplotype information</p>
</td></tr>
<tr><td><code id="hapiBlockMPR_+3A_gmtframe">gmtFrame</code></td>
<td>
<p>a dataframe of raw genotype data in the framework</p>
</td></tr>
<tr><td><code id="hapiBlockMPR_+3A_cvlink">cvlink</code></td>
<td>
<p>a numeric value of number of cvlinks. Default is <code>2</code></p>
</td></tr>
<tr><td><code id="hapiBlockMPR_+3A_smallblock">smallBlock</code></td>
<td>
<p>a numeric value determining the size of small blocks 
that should be excluded from the draft haplotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of draft haplotypes after proofreading
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref &lt;- rep(0,500)
alt &lt;- rep(1,500)

gmtFrame &lt;- data.frame(gmt1=ref, gmt2=alt, gmt3=ref,
gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
stringsAsFactors = FALSE)

idx1 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))
idx2 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))
idx3 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))

gmtFrame[idx1,1] &lt;- NA
gmtFrame[idx2,2] &lt;- NA
gmtFrame[idx3,3] &lt;- NA

imputedFrame &lt;- data.frame(gmt1=ref, gmt2=alt, gmt3=ref,
gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
stringsAsFactors = FALSE)

draftHap &lt;- hapiPhase(imputedFrame)

finalDraft &lt;- hapiBlockMPR(draftHap, gmtFrame, cvlink=2, smallBlock=100)
</code></pre>

<hr>
<h2 id='hapiCVCluster'>Filter out hetSNPs in potential complex regions</h2><span id='topic+hapiCVCluster'></span>

<h3>Description</h3>

<p>Filter out hetSNPs in potential complex regions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiCVCluster(draftHap, minDistance = 1e+06, cvlink = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiCVCluster_+3A_drafthap">draftHap</code></td>
<td>
<p>a dataframe with draft haplotype information</p>
</td></tr>
<tr><td><code id="hapiCVCluster_+3A_mindistance">minDistance</code></td>
<td>
<p>a numeric value of the distance between two 
genomic positions with cv-links. Default is <code>1000000</code></p>
</td></tr>
<tr><td><code id="hapiCVCluster_+3A_cvlink">cvlink</code></td>
<td>
<p>a numeric value of number of cvlinks. Default is <code>2</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of regions to be filtered out
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref &lt;- rep(0,500)
alt &lt;- rep(1,500)

imputedFrame &lt;- data.frame(gmt1=ref, gmt2=alt, gmt3=ref,
gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
stringsAsFactors = FALSE)

draftHap &lt;- hapiPhase(imputedFrame)
cvCluster &lt;- hapiCVCluster(draftHap = draftHap, cvlink=2)
</code></pre>

<hr>
<h2 id='hapiCVDistance'>Histogram of crossover distance</h2><span id='topic+hapiCVDistance'></span>

<h3>Description</h3>

<p>Histogram of crossover distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiCVDistance(cv)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiCVDistance_+3A_cv">cv</code></td>
<td>
<p>a dataframe of crossover information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a histogram
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(crossover)
hapiCVDistance(cv=crossover)
</code></pre>

<hr>
<h2 id='hapiCVMap'>Visualization of crossover map</h2><span id='topic+hapiCVMap'></span>

<h3>Description</h3>

<p>Visualization of crossover map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiCVMap(cv, chr = hg19, step = 5, gap = gap.hg19, x.limits = 6,
  y.breaks = NULL, y.labels = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiCVMap_+3A_cv">cv</code></td>
<td>
<p>a dataframe of crossover information</p>
</td></tr>
<tr><td><code id="hapiCVMap_+3A_chr">chr</code></td>
<td>
<p>a dataframe of chromosome information, including length, 
and centrometric regions</p>
</td></tr>
<tr><td><code id="hapiCVMap_+3A_step">step</code></td>
<td>
<p>a numeric value of genomic interval in Mb. 
Default is <code>5</code></p>
</td></tr>
<tr><td><code id="hapiCVMap_+3A_gap">gap</code></td>
<td>
<p>a dataframe of unassembled regions with the first column is
chromosme, the second column is start position, and third column is the 
end position of the gap. Default is gap for hg19. 
If no gap region is provided, use <code>gap=NULL</code></p>
</td></tr>
<tr><td><code id="hapiCVMap_+3A_x.limits">x.limits</code></td>
<td>
<p>a numeric value of limits on x axis</p>
</td></tr>
<tr><td><code id="hapiCVMap_+3A_y.breaks">y.breaks</code></td>
<td>
<p>a vector of positions to show labels on y axis. 
Default is <code>NULL</code></p>
</td></tr>
<tr><td><code id="hapiCVMap_+3A_y.labels">y.labels</code></td>
<td>
<p>a vector of labels on the y axis. Default is <code>NULL</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot of crossover map on all the chromosomes
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(crossover)
hapiCVMap(cv=crossover)
</code></pre>

<hr>
<h2 id='hapiCVResolution'>Histogram of crossover resolution</h2><span id='topic+hapiCVResolution'></span>

<h3>Description</h3>

<p>Histogram of crossover resolution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiCVResolution(cv)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiCVResolution_+3A_cv">cv</code></td>
<td>
<p>a dataframe of crossover information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a histogram
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(crossover)
hapiCVResolution(cv=crossover)
</code></pre>

<hr>
<h2 id='hapiFilterError'>Filter out hetSNPs with potential genotyping errors</h2><span id='topic+hapiFilterError'></span>

<h3>Description</h3>

<p>Filter out hetSNPs with potential genotyping errors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiFilterError(gmt, hmm = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiFilterError_+3A_gmt">gmt</code></td>
<td>
<p>a dataframe of genotype data of gamete cells</p>
</td></tr>
<tr><td><code id="hapiFilterError_+3A_hmm">hmm</code></td>
<td>
<p>a list containing probabilities of a HMM. Default is <code>NULL</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of genotype data of gamete cells
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref &lt;- rep(0,500)
alt &lt;- rep(1,500)

gmt &lt;- data.frame(gmt1=ref, gmt2=alt, gmt3=ref,
    gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
    stringsAsFactors = FALSE)
    
idx &lt;- sort(sample(seq_len(500), 10, replace = FALSE))
gmt[idx,1] &lt;- 1

gmtDa &lt;- hapiFilterError(gmt = gmt)
</code></pre>

<hr>
<h2 id='hapiFrameSelection'>Selection of hetSNPs to form a framework</h2><span id='topic+hapiFrameSelection'></span>

<h3>Description</h3>

<p>Selection of hetSNPs to form a framework
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiFrameSelection(gmt, n = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiFrameSelection_+3A_gmt">gmt</code></td>
<td>
<p>a dataframe of genotype data of gamete cells</p>
</td></tr>
<tr><td><code id="hapiFrameSelection_+3A_n">n</code></td>
<td>
<p>a numeric value of the minumum number of gametes with 
observed genotypes at a locus</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of genotype data of gamete cells
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref &lt;- rep(0,500)
alt &lt;- rep(1,500)

gmt &lt;- data.frame(gmt1=ref, gmt2=alt, gmt3=ref,
gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
stringsAsFactors = FALSE)

idx &lt;- sort(sample(seq_len(500), 10, replace = FALSE))

gmt[idx,1] &lt;- NA
gmt[idx,2] &lt;- NA
gmt[idx,3] &lt;- NA

gmtFrame &lt;- hapiFrameSelection(gmt = gmt, n = 3)
</code></pre>

<hr>
<h2 id='hapiGameteView'>Visualization of haplotypes in a single gamete cell</h2><span id='topic+hapiGameteView'></span>

<h3>Description</h3>

<p>Visualization of haplotypes in a single gamete cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiGameteView(hap, chr = hg19, hap.color = c("deepskyblue2",
  "darkorange2"), centromere.fill = "black", x.breaks = NULL,
  x.labels = NULL, y.breaks = NULL, y.labels = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiGameteView_+3A_hap">hap</code></td>
<td>
<p>a dataframe of all the phased hetSNPs in all chromosomes</p>
</td></tr>
<tr><td><code id="hapiGameteView_+3A_chr">chr</code></td>
<td>
<p>a dataframe of chromosome information, including length, 
and centrometric regions</p>
</td></tr>
<tr><td><code id="hapiGameteView_+3A_hap.color">hap.color</code></td>
<td>
<p>a vector of colors for the two haplotypes. 
Default is <code>c('deepskyblue2','darkorange2')</code></p>
</td></tr>
<tr><td><code id="hapiGameteView_+3A_centromere.fill">centromere.fill</code></td>
<td>
<p>a character of the color for the centromeres. 
Default is <code>'black'</code></p>
</td></tr>
<tr><td><code id="hapiGameteView_+3A_x.breaks">x.breaks</code></td>
<td>
<p>a vector of positions to show labels on x axis. 
Default is <code>NULL</code></p>
</td></tr>
<tr><td><code id="hapiGameteView_+3A_x.labels">x.labels</code></td>
<td>
<p>a vector of labels on the x axis. 
Default is <code>NULL</code></p>
</td></tr>
<tr><td><code id="hapiGameteView_+3A_y.breaks">y.breaks</code></td>
<td>
<p>a vector of positions to show labels on y axis. 
Default is <code>NULL</code></p>
</td></tr>
<tr><td><code id="hapiGameteView_+3A_y.labels">y.labels</code></td>
<td>
<p>a vector of labels on the y axis. 
Default is <code>NULL</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot of haplotypes in a single gamete cell
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gamete11)
hapiGameteView(hap=gamete11)
</code></pre>

<hr>
<h2 id='hapiIdentifyCV'>Indentify crossovers in gamete cells</h2><span id='topic+hapiIdentifyCV'></span>

<h3>Description</h3>

<p>Indentify crossovers in gamete cells
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiIdentifyCV(hap, gmt, hmm = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiIdentifyCV_+3A_hap">hap</code></td>
<td>
<p>a dataframe of the two haplotypes</p>
</td></tr>
<tr><td><code id="hapiIdentifyCV_+3A_gmt">gmt</code></td>
<td>
<p>a dataframe of genotype data of gamete cells</p>
</td></tr>
<tr><td><code id="hapiIdentifyCV_+3A_hmm">hmm</code></td>
<td>
<p>a list containing probabilities of a HMM. Default is <code>NULL</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe containing crossover information in each gamete cell
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref &lt;- sample(c('A','T'),500, replace=TRUE)
alt &lt;- sample(c('C','G'),500, replace=TRUE)

hap &lt;- data.frame(hap1=ref, hap2=alt, stringsAsFactors = FALSE)
rownames(hap) &lt;- seq_len(500)

gmt &lt;- data.frame(gmt1=ref, gmt2=alt, gmt3=ref,
    gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
    stringsAsFactors = FALSE)
    
cvOutput &lt;- hapiIdentifyCV(hap=hap, gmt=gmt)
</code></pre>

<hr>
<h2 id='hapiImupte'>Imputation of missing genotypes in the framework</h2><span id='topic+hapiImupte'></span>

<h3>Description</h3>

<p>Imputation of missing genotypes in the framework
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiImupte(gmt, nSPT = 2, allowNA = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiImupte_+3A_gmt">gmt</code></td>
<td>
<p>a dataframe of genotype data of gamete cells in the framework</p>
</td></tr>
<tr><td><code id="hapiImupte_+3A_nspt">nSPT</code></td>
<td>
<p>a numeric value of the minumum number of supports 
for an imputation</p>
</td></tr>
<tr><td><code id="hapiImupte_+3A_allowna">allowNA</code></td>
<td>
<p>a numeric value of the maximum number of gametes with 
NA at a locus</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of imputed genotypes in the framework
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref &lt;- rep(0,500)
alt &lt;- rep(1,500)

gmtFrame &lt;- data.frame(gmt1=ref, gmt2=alt, gmt3=ref,
gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
stringsAsFactors = FALSE)

idx1 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))
idx2 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))
idx3 &lt;- sort(sample(seq_len(500), 30, replace = FALSE))

gmtFrame[idx1,1] &lt;- NA
gmtFrame[idx2,2] &lt;- NA
gmtFrame[idx3,3] &lt;- NA
imputedFrame &lt;- hapiImupte(gmtFrame, nSPT=2, allowNA=0)
</code></pre>

<hr>
<h2 id='hapiPhase'>Phase draft haplotypes by majority voting</h2><span id='topic+hapiPhase'></span>

<h3>Description</h3>

<p>Phase draft haplotypes by majority voting
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapiPhase(gmt)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapiPhase_+3A_gmt">gmt</code></td>
<td>
<p>a dataframe of imputed genotype data of gamete cells</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of inferred draft haplotypes
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref &lt;- rep(0,500)
alt &lt;- rep(1,500)
imputedFrame &lt;- data.frame(gmt1=ref, gmt2=alt, gmt3=ref, 
gmt4=ref, gmt5=c(alt[1:250], ref[251:500]),
stringsAsFactors = FALSE)
draftHap &lt;- hapiPhase(gmt=imputedFrame)
</code></pre>

<hr>
<h2 id='hg19'>Chromosome information of hg19</h2><span id='topic+hg19'></span>

<h3>Description</h3>

<p>Chromosome information of hg19
</p>

<hr>
<h2 id='num2base'>Convert genotype coded in 0/1 to A/T/C/G</h2><span id='topic+num2base'></span>

<h3>Description</h3>

<p>Convert numeric (0/1) coded genotype to base (A/T/C/G) coded
</p>


<h3>Usage</h3>

<pre><code class='language-R'>num2base(hap, ref, alt)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="num2base_+3A_hap">hap</code></td>
<td>
<p>a dataframe of consensus haplotypes</p>
</td></tr>
<tr><td><code id="num2base_+3A_ref">ref</code></td>
<td>
<p>a character represents reference allele</p>
</td></tr>
<tr><td><code id="num2base_+3A_alt">alt</code></td>
<td>
<p>a character represents alternative allele</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe containing converted haplotypes
</p>


<h3>Author(s)</h3>

<p>Ruidong Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref &lt;- sample(c('A','T'),500, replace=TRUE)
alt &lt;- sample(c('C','G'),500, replace=TRUE)

consensusHap &lt;- data.frame(hap1=rep(0,500),hap2=rep(1,500),
    total=rep(5,500),rate=rep(1,500),
    confidence=rep('F',500),
    stringsAsFactors = FALSE)
rownames(consensusHap) &lt;- seq_len(500)

hap &lt;- num2base(hap=consensusHap, ref=ref, alt=alt)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
