<!DOCTYPE html><html lang="en"><head><title>Help for package Haplin</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Haplin}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cbindFiles'><p>Combine a sequence of files by columns</p></a></li>
<li><a href='#convertPed'><p>Convert large ped files by creating unique IDs, converting allele coding and extracting a selection of SNPs</p></a></li>
<li><a href='#f.check.unique.ids'><p>Checking uniqueness of individuals' IDs</p></a></li>
<li><a href='#f.convert.matrix.ff'><p>Converting Haplin-formatted matrix into the new format</p></a></li>
<li><a href='#f.create.missingness.matrix'><p>Create family matrix of data missingness</p></a></li>
<li><a href='#f.create.snp.names'><p>Intern function for creating column names for genotype data</p></a></li>
<li><a href='#f.extract.ff.numeric'><p>Extract a part of an ff array (matrix) and return it as numeric matrix</p></a></li>
<li><a href='#f.get.gen.data.cols'><p>Getter for the list with genetic or environmental data</p></a></li>
<li><a href='#finishParallelRun'><p>Closing the Rmpi cluster</p></a></li>
<li><a href='#genDataGetPart'><p>Extracting part of genetic data.</p></a></li>
<li><a href='#genDataLoad'><p>Loading the data previously read in and saved by &quot;genDataRead&quot;</p></a></li>
<li><a href='#genDataPreprocess'><p>Pre-processing of the genetic data</p></a></li>
<li><a href='#genDataRead'><p>Reading the genetic data from a file</p></a></li>
<li><a href='#getChildren'><p>Getter for all rows with children data</p></a></li>
<li><a href='#getDyads'><p>Getter only for all dyads (child and one parent)</p></a></li>
<li><a href='#getFathers'><p>Getter for all rows with fathers' data</p></a></li>
<li><a href='#getFullTriads'><p>Getter for all full triads</p></a></li>
<li><a href='#getMothers'><p>Getter for all rows with mothers' data</p></a></li>
<li><a href='#gxe'><p>Test for gene-environment interaction</p></a></li>
<li><a href='#haplin'><p>Fitting log-linear models to case-parent triad and/or case-control data</p></a></li>
<li><a href='#haplinSlide'><p>Run haplin analysis in a series of sliding windows over a sequence of markers/SNPs</p></a></li>
<li><a href='#haplinStrat'><p>Fit <code>haplin</code> to each subset/stratum of data, determined by the argument <code>strata</code></p></a></li>
<li><a href='#hapPower'><p>Power simulation for association analyses with Haplin</p></a></li>
<li><a href='#hapPowerAsymp'><p>Asymptotic power calculations for genetic association analyses with Haplin</p></a></li>
<li><a href='#hapRelEff'><p>Relative efficiency comparing different study designs in genetic association analysis with Haplin</p></a></li>
<li><a href='#hapRun'><p>Simulates genetic data and runs Haplin for each simulation</p></a></li>
<li><a href='#hapSim'><p>Simulation of genetic data in Haplin format</p></a></li>
<li><a href='#haptable'><p>Create haplin table</p></a></li>
<li><a href='#initParallelRun'><p>Initialization of the Rmpi cluster</p></a></li>
<li><a href='#lineByLine'><p>Line-by-line modification of files</p></a></li>
<li><a href='#nfam'><p>Count the number of families in the data</p></a></li>
<li><a href='#nindiv'><p>Count the number of individuals in the data</p></a></li>
<li><a href='#nsnps'><p>Count the number of markers in the data</p></a></li>
<li><a href='#output'><p>Save files with summary, table, and plot from a haplin object.</p></a></li>
<li><a href='#pedToHaplin'><p>Convert from ped format data to Haplin format</p></a></li>
<li><a href='#pJohnson'><p>The Johnson distributions</p></a></li>
<li><a href='#plot.haplin'><p>Plot a haplin object</p></a></li>
<li><a href='#plot.haplinSlide'><p>Plotter function for haplinSlide.</p></a></li>
<li><a href='#plot.haplinStrat'><p>Plotter function for haplinStrat results.</p></a></li>
<li><a href='#plot.haptable'><p>Plot a haptable object</p></a></li>
<li><a href='#plotPValues'><p>Plotting p-values for relative risks</p></a></li>
<li><a href='#pQQ'><p>QQ-plot with confidence intervals for a vector of p-values</p></a></li>
<li><a href='#print.haplin'><p>Print a haplin object</p></a></li>
<li><a href='#print.summary.haplin'><p>Print the summary of a haplin object</p></a></li>
<li><a href='#rbindFiles'><p>Combine a sequence of files by rows</p></a></li>
<li><a href='#showGen'><p>Display chosen genotypes</p></a></li>
<li><a href='#showPheno'><p>Display phenotype part of data</p></a></li>
<li><a href='#showSNPnames'><p>Display marker names</p></a></li>
<li><a href='#snpPos'><p>Find the column numbers of SNP identifiers/SNP numbers in a ped file</p></a></li>
<li><a href='#snpPower'><p>Power calculations for a single SNP</p></a></li>
<li><a href='#snpSampleSize'><p>Sample size calculations for a single SNP</p></a></li>
<li><a href='#suest'><p>Compute a joint p-value for a list of haplin fits (usually from a sliding window approach), correcting for multiple testing.</p></a></li>
<li><a href='#summary.haplin'><p>Summary of a haplin object</p></a></li>
<li><a href='#toDataFrame'><p>Stack dataframes from haplinSlide into a single dataframe</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Analyzing Case-Parent Triad and/or Case-Control Data with SNP
Haplotypes</td>
</tr>
<tr>
<td>Version:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-08-16</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hakon K. Gjessing &lt;hakon.gjessing@uib.no&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>tools, mgcv, MASS, ff, rlang, methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, Rmpi, ggplot2, testthat, rmarkdown</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs genetic association analyses of case-parent triad (trio) data with multiple markers. It can also incorporate complete or incomplete control triads, for instance independent control children. Estimation is based on haplotypes, for instance SNP haplotypes, even though phase is not known from the genetic data. 'Haplin' estimates relative risk (RR + conf.int.) and p-value associated with each haplotype. It uses maximum likelihood estimation to make optimal use of data from triads with missing genotypic data, for instance if some SNPs has not been typed for some individuals. 'Haplin' also allows estimation of effects of maternal haplotypes and parent-of-origin effects, particularly appropriate in perinatal epidemiology. 'Haplin' allows special models, like X-inactivation, to be fitted on the X-chromosome. A GxE analysis allows testing interactions between environment and all estimated genetic effects. The models were originally described in "Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396".</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>BuildVignettes:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-08-20 08:50:02 UTC; hkgje</td>
</tr>
<tr>
<td>Author:</td>
<td>Hakon K. Gjessing [aut, cre],
  Miriam Gjerdevik [ctb] (functions 'lineByLine', 'cbindFiles',
    'rbindFiles', 'snpPower', 'snpSampleSize', 'hapSim', 'hapRun',
    'hapPower', 'hapPowerAsymp', and 'hapRelEff'),
  Julia Romanowska <a href="https://orcid.org/0000-0001-6733-1953"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb] (new data format, parallelisation, new documentation),
  Oivind Skare [ctb] (TDT tests)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-08-20 14:30:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='cbindFiles'>Combine a sequence of files by columns</h2><span id='topic+cbindFiles'></span>

<h3>Description</h3>

<p>Takes a sequence of files and combines them column-wise (side-by-side), i.e. reads each file line by line, pastes corresponding lines, then writes to outfile. Combining files line by line is especially useful when working with large datasets, where the reading of entire files may be time consuming and require a large amount of memory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cbindFiles(infiles, outfile, col.sep, ask = TRUE, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cbindFiles_+3A_infiles">infiles</code></td>
<td>
<p>A character vector of names (and paths) of the files to combine.</p>
</td></tr>
<tr><td><code id="cbindFiles_+3A_outfile">outfile</code></td>
<td>
<p>A character string giving the name of the resulting file. The name of the file is relative to the current working directory, unless the file name contains a definite path.</p>
</td></tr>
<tr><td><code id="cbindFiles_+3A_col.sep">col.sep</code></td>
<td>
<p>Specifies the column separator which will be inserted between files.</p>
</td></tr>
<tr><td><code id="cbindFiles_+3A_ask">ask</code></td>
<td>
<p>Logical. Default is &quot;TRUE&quot;. If set to &quot;FALSE&quot;, an already existing outfile will be overwritten without asking.</p>
</td></tr>
<tr><td><code id="cbindFiles_+3A_verbose">verbose</code></td>
<td>
<p>Logical. Default is &quot;TRUE&quot;, which means that the line number is displayed for each iteration, i.e. each combined line.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="base.html#topic+cbind">cbind</a></code> combines R objects by columns. However, reading large datafiles may require a large amount of memory and be extremely time consuming.
<code>cbindFiles</code> combines the files column-wise, one line at a time, and then writes each line to <code>outfile</code>. This avoids reading the full files into memory.<br />
If <code>infiles</code> contains only one file, this file will be copied directly to <code>outfile</code>, i.e. without any line-by-line modifications.
</p>


<h3>Value</h3>

<p>There is no useful output; the objective of <code>cbindFiles</code> is to produce <code>outfile</code>.
</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a>
</p>


<h3>References</h3>

<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+rbindFiles">rbindFiles</a></code>, <code><a href="#topic+lineByLine">lineByLine</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Combines the three infiles side-by-side
cbindFiles(infiles = c("myfile1.txt", "myfile2.txt", 
"myfile3.txt"), outfile = "myfile_combined_by_columns.txt", 
col.sep = " ", ask = TRUE, verbose = TRUE)


## End(Not run)
</code></pre>

<hr>
<h2 id='convertPed'>Convert large ped files by creating unique IDs, converting allele coding and extracting a selection of SNPs</h2><span id='topic+convertPed'></span>

<h3>Description</h3>

<p>NOTE: This function is probably less useful now that GenABEL is no longer used by Haplin. The function is used to prepare a ped file for loading into GenABEL. However, GenABEL requires unique individual IDs in the file, not only unique within family. Furthermore, numeric allele coding 1,2,3,4 is not accepted. To fix this, <code>convertPed</code> can be run prior to running <code>prepPed</code>. This will create unique IDs and do the necessary allele recoding, and possibly also select and reorder SNPs. <code>convertPed</code> will also update the corresponding map file.</p>


<h3>Usage</h3>

<pre><code class='language-R'>convertPed(ped.infile, map.infile, ped.outfile, map.outfile, create.unique.id = FALSE,
convert, snp.select = NULL, choose.lines = NULL, col.sep = " ",
ask = TRUE, blank.lines.skip = TRUE, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convertPed_+3A_ped.infile">ped.infile</code></td>
<td>
<p>A character string giving the name of the standard ped file to be modified. The name of the file is relative to the current working directory, unless the file name contains an absolute path.<br />  See Details for a description of the standard ped format.</p>
</td></tr>
<tr><td><code id="convertPed_+3A_map.infile">map.infile</code></td>
<td>
<p>A character string giving the name and path of the to-be-modified standard map file. Optional if snp.select = NULL. A description of the standard map format is given in the Details section.</p>
</td></tr>
<tr><td><code id="convertPed_+3A_ped.outfile">ped.outfile</code></td>
<td>
<p>A character string of the name and path of the converted ped file.</p>
</td></tr>
<tr><td><code id="convertPed_+3A_map.outfile">map.outfile</code></td>
<td>
<p>A character string giving the name and path of the modified map file.</p>
</td></tr>
<tr><td><code id="convertPed_+3A_create.unique.id">create.unique.id</code></td>
<td>
<p>Logical. If &quot;TRUE&quot;, the function creates a unique individual ID.</p>
</td></tr>
<tr><td><code id="convertPed_+3A_convert">convert</code></td>
<td>
<p>No default. The option &quot;ACGT_to_1234&quot; recodes the SNP alleles from A,C,G,T to 1,2,3,4, whereas &quot;1234_to_ACGT&quot; converts from 1,2,3,4 to A,C,G,T. If &quot;no_recode&quot;, no conversion occurs.</p>
</td></tr>
<tr><td><code id="convertPed_+3A_snp.select">snp.select</code></td>
<td>
<p>A character vector of the SNP identifiers (RS codes) or a numeric vector of the SNP numbers to be extracted. Default is &quot;NULL&quot;, which means that all SNPs are selected without reordering among the SNPs. The RS codes or SNP numbers may be listed in any order. Reordering among the selected SNPs will occur in the modified files corresponding to this listing.</p>
</td></tr>
<tr><td><code id="convertPed_+3A_choose.lines">choose.lines</code></td>
<td>
<p>A numeric vector of lines to be selected from the ped file. If &quot;NULL&quot; (default), all lines are selected.</p>
</td></tr>
<tr><td><code id="convertPed_+3A_col.sep">col.sep</code></td>
<td>
<p>Specifies the separator that splits the columns in <code>ped.infile</code>. By default, col.sep = &quot; &quot; (space). To split at all types of space or blank characters, set col.sep = &quot;[[:space:]]&quot; or col.sep = &quot;[[:blank:]]&quot;.</p>
</td></tr>
<tr><td><code id="convertPed_+3A_ask">ask</code></td>
<td>
<p>Logical. Default is &quot;TRUE&quot;. If set to &quot;FALSE&quot;, an already existing outfile will be overwritten without asking.</p>
</td></tr>
<tr><td><code id="convertPed_+3A_blank.lines.skip">blank.lines.skip</code></td>
<td>
<p>Logical. If &quot;TRUE&quot; (default), <code>convertPed</code> ignores blank lines in <code>ped.infile</code> and <code>map.infile</code>.</p>
</td></tr>
<tr><td><code id="convertPed_+3A_verbose">verbose</code></td>
<td>
<p>Logical. Default is &quot;TRUE&quot;, which means that the line number is displayed for each iteration, i.e. each line read and modified, in addition to the first ten columns of the converted line.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>convertPed</code> assumes a standard ped file as input. 
The format of the ped file should look something like this:
</p>
<pre>
1104  1  2  3  1  2  4  1  3  2  1  1
1104  2  0  0  1  1  4  1  2  2  4  1
1104  3  0  0  2  1  0  0  0  0  0  0
1105  1  2  3  2  2  1  1  2  2  4  1
1105  2  0  0  1  1  1  1  2  2  1  1
1105  3  0  0  2  1  1  1  3  2  4  4
</pre>
<p>The column values are: Family ID, Individual ID, Father's ID, Mother's ID, Sex (1 = male, 2 = female, alternatively: 1 = male, 0 = female), and Case-control status (1 = controls, 2 = cases, alternatively: 0 = controls, 1 = cases).<br />
Column 7 and onwards contain the genotype data, with alleles in separate columns, two columns representing one SNP. A &ldquo;0&rdquo; is used to denote missing data.<br />
<br />
</p>
<p>The corresponding map file should look something like this:
</p>
<pre>
Chromosome SNP-identifier Base-pair-position
1               RS9629043             554636
1              RS12565286             711153 
1              RS12138618             740098
</pre>
<p>Alternatively, the map file could contain four columns. The column values should then be:
Chromosome, SNP-identifier, Genetic-distance, Base-pair-position.<br />
A header must be added to the map file if this does not already have one.<br /><br />
</p>
<p>After creating unique individual IDs and recoding the SNP alleles from 1,2,3,4 to A,C,G,T (using <code>convertPed</code> with options <code>create.unique.id = TRUE</code> and <code>convert = "1234_to_ACGT"</code>), 
the ped file above should look like this:
</p>
<pre>
1104  1104_1  1104_2  1104_3  1  2  T  A  G  C  A  A
1104  1104_2       0       0  1  1  T  A  C  C  T  A
1104  1104_3       0       0  2  1  0  0  0  0  0  0
1105  1105_1  1105_2  1105_3  2  2  A  A  C  C  T  A
1105  1105_2       0       0  1  1  A  A  C  C  A  A
1105  1105_3       0       0  2  1  A  A  G  C  T  T
</pre>


<h3>Value</h3>

<p>There is no useful output; the objective of <code>convertPed</code> is the converted ped file and the modified map file. 
</p>


<h3>Note</h3>

<p>The function does not check if the ped or map file is formatted correctly. For instance, if the alleles follows the generic A/B Illumina coding, <code>convertPed</code> may still be used to create unique individual IDs and extract a selection of SNPs. Using <code>convert = "ACGT_to_1234"</code> would however, result in nonsense.</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lineByLine">lineByLine</a></code>, <code>Haplin:::lineConvert</code>, <code><a href="#topic+snpPos">snpPos</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Create unique individual IDs and recode SNP alleles from 1,2,3,4 to A,C,G,T
convertPed(ped.infile = "mygwas.ped", map.infile = "mygwas.map",
ped.outfile = "mygwas_modified.ped", map.outfile = "mygwas_modified.map",
create.unique.id = TRUE, convert = "1234_to_ACGT", ask = TRUE)


## End(Not run)
</code></pre>

<hr>
<h2 id='f.check.unique.ids'>Checking uniqueness of individuals' IDs</h2><span id='topic+f.check.unique.ids'></span>

<h3>Description</h3>

<p>This checks whether the individuals have unique IDs. If not, it creates new IDs by
adding the family ID to the indiv. ID. Since the function checks for specific
column names, this basically means that the input should be as read from 
a PED file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f.check.unique.ids(data.cov)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="f.check.unique.ids_+3A_data.cov">data.cov</code></td>
<td>
<p>A matrix with &quot;id.fam&quot;, &quot;id.c&quot;, &quot;id.m&quot;, and &quot;id.f&quot; columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with:
</p>

<ul>
<li><p> ids - new IDs in the same order as &quot;id.c&quot; column in the input data
</p>
</li>
<li><p> pedIndex - matrix with &quot;family&quot;, &quot;id.child&quot;, &quot;id.mother&quot;, and
&quot;id.father&quot; columns
</p>
</li></ul>


<hr>
<h2 id='f.convert.matrix.ff'>Converting Haplin-formatted matrix into the new format</h2><span id='topic+f.convert.matrix.ff'></span>

<h3>Description</h3>

<p>Internal function for converting an R matrix object into the new format of data used in Haplin. The output is the same as from <a href="#topic+genDataRead">genDataRead</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f.convert.matrix.ff(
  data = stop("You must give the data to convert!", call. = FALSE),
  n.vars = stop("You must explicitly give the number of columns with covariates!", call.
    = FALSE),
  cov.header,
  gen.levels
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="f.convert.matrix.ff_+3A_data">data</code></td>
<td>
<p>A character matrix (NB: it's assumed that this matrix contains genotypes
and covariate data, if any; i.e., as if it was a table read in from a haplin-formatted
file).</p>
</td></tr>
<tr><td><code id="f.convert.matrix.ff_+3A_n.vars">n.vars</code></td>
<td>
<p>Number of columns with covariate variables - if the data does not contain
covariates, give 0 explicitly.</p>
</td></tr>
<tr><td><code id="f.convert.matrix.ff_+3A_cov.header">cov.header</code></td>
<td>
<p>Optional; if there are covariates in the data, you can give their 
names explicitly here. Otherwise, dummy names will be created.</p>
</td></tr>
<tr><td><code id="f.convert.matrix.ff_+3A_gen.levels">gen.levels</code></td>
<td>
<p>Optional; a vector with all the possible values for alleles in the
genotype part of data. If not given, these will be assessed from the given data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with two elements:
</p>

<ul>
<li> <p><em>cov.data</em> - a <code>data.frame</code> with covariate data (if available in
the input file)
</p>
</li>
<li> <p><em>gen.data</em> - a list with chunks of the genetic data; the data is divided
column-wise, using 10,000 columns per chunk; each element of this list is a
<a href="ff.html#topic+ff">ff</a> matrix
</p>
</li></ul>


<hr>
<h2 id='f.create.missingness.matrix'>Create family matrix of data missingness</h2><span id='topic+f.create.missingness.matrix'></span>

<h3>Description</h3>

<p>Help function that checks the family structure, based on IDs and provided data in .ped
file, and then creates a matrix with families, where each indiv. has TRUE/FALSE value,
depending on whether they have or not genetic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f.create.missingness.matrix(
  data.in = stop("No data given!", call. = FALSE),
  new.ids = stop("No IDs given", call. = FALSE)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="f.create.missingness.matrix_+3A_data.in">data.in</code></td>
<td>
<p>The data object (in format as the output of <a href="#topic+genDataRead">genDataRead</a>); note
that the design of the data is assumed to be triad.</p>
</td></tr>
<tr><td><code id="f.create.missingness.matrix_+3A_new.ids">new.ids</code></td>
<td>
<p>A list with pedIndex and ids, as generated by <a href="#topic+f.check.unique.ids">f.check.unique.ids</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix where each member of the family has TRUE or FALSE
(whether they have any genetic data)
</p>

<hr>
<h2 id='f.create.snp.names'>Intern function for creating column names for genotype data</h2><span id='topic+f.create.snp.names'></span>

<h3>Description</h3>

<p>Creating column names either from a .map file or generating dummy names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f.create.snp.names(map.file, header = FALSE, ncol, format, design)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="f.create.snp.names_+3A_map.file">map.file</code></td>
<td>
<p>Filename (with path if the file is not in current directory) of the
.map file holding the SNP names, if available (see Details).</p>
</td></tr>
<tr><td><code id="f.create.snp.names_+3A_header">header</code></td>
<td>
<p>Logical: does the map.file contain a header in the first row?
Default: FALSE.</p>
</td></tr>
<tr><td><code id="f.create.snp.names_+3A_ncol">ncol</code></td>
<td>
<p>Number of columns IN TOTAL in the dataset containing only the genotype data</p>
</td></tr>
<tr><td><code id="f.create.snp.names_+3A_format">format</code></td>
<td>
<p>Format of data (will influence how data is processed) - choose from:
</p>

<ul>
<li> <p><em>haplin</em> - data already in one row per family,
</p>
</li>
<li> <p><em>ped</em> - data from .ped file, each row represents an individual.
</p>
</li></ul>
<p>.</p>
</td></tr>
<tr><td><code id="f.create.snp.names_+3A_design">design</code></td>
<td>
<p>The design used in the study - choose from:
</p>

<ul>
<li> <p><em>triad</em> - (default), data includes genotypes of mother, father and child;
</p>
</li>
<li> <p><em>cc</em> - classical case-control;
</p>
</li>
<li> <p><em>cc.triad</em> - hybrid design: triads with cases and controls
</p>
</li></ul>
<p>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the .map file is given, the SNP names are read in, if not dummy names are
created by attaching numbers to &quot;m&quot; prefix. Then, a &quot;_a&quot; and &quot;_b&quot; suffix is
attached to each allele, respectively. Finally, if the design is &quot;triad&quot; or
&quot;cc.triad&quot;, the following suffixes are attached: &quot;_m&quot; for the mother's
alleles, &quot;_f&quot; for the father's, and &quot;_c&quot; for the child's.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li> <p><em>gen.data.colnames</em> - a vector with names of columns, length equal to 
the number of columns in the genotype dataset (i.e., depending on the format and
design).
</p>
</li>
<li> <p><em>marker.names</em> - a vector containing the names of markers, as read in
from 'map.file', or dummy names.
</p>
</li></ul>



<h3>Details</h3>

<p>The .map file should contain at least two columns, where the second one contains SNP 
names. Any additional columns should be separated by a whitespace character, but will 
be ignored. The file should contain a header.
</p>

<hr>
<h2 id='f.extract.ff.numeric'>Extract a part of an ff array (matrix) and return it as numeric matrix</h2><span id='topic+f.extract.ff.numeric'></span>

<h3>Description</h3>

<p>This function returns the requested columns and/or rows and converts it to a numeric matrix (by default, the extraction from an ff matrix with defined levels returns a factor vector).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f.extract.ff.numeric(gen.data, cols = NULL, rows = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="f.extract.ff.numeric_+3A_gen.data">gen.data</code></td>
<td>
<p>The ff matrix with genetic data.</p>
</td></tr>
<tr><td><code id="f.extract.ff.numeric_+3A_cols">cols</code></td>
<td>
<p>The sequence numbering the columns which are to be retrieved, if NULL (default) then all the columns will be extracted.</p>
</td></tr>
<tr><td><code id="f.extract.ff.numeric_+3A_rows">rows</code></td>
<td>
<p>The sequence numbering the rows which are to be retrieved, if NULL (default) then all the rows will be extracted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with all the requested columns, data in numeric representation.
</p>

<hr>
<h2 id='f.get.gen.data.cols'>Getter for the list with genetic or environmental data</h2><span id='topic+f.get.gen.data.cols'></span>

<h3>Description</h3>

<p>This function returns the columns requested from the column-wise splitted genetic/environmental data in ff format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f.get.gen.data.cols(gen.data, cols, by.colname = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="f.get.gen.data.cols_+3A_gen.data">gen.data</code></td>
<td>
<p>The list with genetic/environmental data (in ff format).</p>
</td></tr>
<tr><td><code id="f.get.gen.data.cols_+3A_cols">cols</code></td>
<td>
<p>The sequence numbering (or listing) the columns which are to be retrieved; this should enumerate the columns as if all the data was in one large matrix.</p>
</td></tr>
<tr><td><code id="f.get.gen.data.cols_+3A_by.colname">by.colname</code></td>
<td>
<p>Logical: is 'cols' given as a number sequence (FALSE, default) or character vector (TRUE) naming the columns to choose?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix (in ff format) with all the requested columns.
</p>

<hr>
<h2 id='finishParallelRun'>Closing the Rmpi cluster</h2><span id='topic+finishParallelRun'></span>

<h3>Description</h3>

<p>This function closes all the slaves spawned in <a href="#topic+initParallelRun">initParallelRun</a> and
finishes the mpi routines. This function MUST BE called after all the <a href="#topic+haplinSlide">haplinSlide</a>
calls and right before exiting the script/R session!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>finishParallelRun()
</code></pre>

<hr>
<h2 id='genDataGetPart'>Extracting part of genetic data.</h2><span id='topic+genDataGetPart'></span>

<h3>Description</h3>

<p>This function enables to extract (and save for later use) part of genetic data
read in with <a href="#topic+genDataRead">genDataRead</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genDataGetPart(
  data.in = stop("No data given!", call. = FALSE),
  design = stop("Design type must be given!"),
  markers,
  indiv.ids,
  rows,
  cc,
  sex,
  file.out = "my_data_part",
  dir.out = ".",
  overwrite = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genDataGetPart_+3A_data.in">data.in</code></td>
<td>
<p>The data object (in format as the output of <a href="#topic+genDataRead">genDataRead</a>).</p>
</td></tr>
<tr><td><code id="genDataGetPart_+3A_design">design</code></td>
<td>
<p>The design used in the study - choose from:
</p>

<ul>
<li> <p><em>triad</em> - (default), data includes genotypes of mother, father and child;
</p>
</li>
<li> <p><em>cc</em> - classical case-control;
</p>
</li>
<li> <p><em>cc.triad</em> - hybrid design: triads with cases and controls;
</p>
</li></ul>
<p>.
</p>
<p>Any of the following can be given to narrow down the dataset:</p>
</td></tr>
<tr><td><code id="genDataGetPart_+3A_markers">markers</code></td>
<td>
<p>Vector with numbers or names indicating which markers to choose.</p>
</td></tr>
<tr><td><code id="genDataGetPart_+3A_indiv.ids">indiv.ids</code></td>
<td>
<p>Character vector giving IDs of individuals. <strong>CAUTION:</strong>
in a standard PED file, individual IDs are not unique, so this will select
all individuals with given IDs.</p>
</td></tr>
<tr><td><code id="genDataGetPart_+3A_rows">rows</code></td>
<td>
<p>Numeric vector giving the positions - this will select only these rows.</p>
</td></tr>
<tr><td><code id="genDataGetPart_+3A_cc">cc</code></td>
<td>
<p>One or more values to choose based on case-control status ('cc' column).</p>
</td></tr>
<tr><td><code id="genDataGetPart_+3A_sex">sex</code></td>
<td>
<p>One or more values to choose based on the 'sex' column.</p>
</td></tr>
<tr><td><code id="genDataGetPart_+3A_file.out">file.out</code></td>
<td>
<p>The base for the output filename (default: &quot;my_data_part&quot;).</p>
</td></tr>
<tr><td><code id="genDataGetPart_+3A_dir.out">dir.out</code></td>
<td>
<p>The path to the directory where the output files will be saved.</p>
</td></tr>
<tr><td><code id="genDataGetPart_+3A_overwrite">overwrite</code></td>
<td>
<p>Whether to overwrite the output files: if NULL (default), will prompt
the user to give answer; set to TRUE, will automatically overwrite any existing files;
and set to FALSE, will stop if the output files exist.</p>
</td></tr>
<tr><td><code id="genDataGetPart_+3A_...">...</code></td>
<td>
<p>If any additional covariate data are available in <code>data.in</code>, 
the user can choose based on values of these (see the Examples section).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The genetic data from GWAS studies can be quite large, and thus the analysis
is time-consuming. If a user knows where they want to focus the analysis,
they can use this function to extract part of the entire dataset and use
only this part in subsequent Haplin analysis.
</p>


<h3>Value</h3>

<p>A list object with three elements:
</p>

<ul>
<li> <p><em>cov.data</em> - a <code>data.frame</code> with covariate data (if available in
the input file)
</p>
</li>
<li> <p><em>gen.data</em> - a list with chunks of the genetic data; the data is divided
column-wise, using 10,000 columns per chunk; each element of this list is a
<a href="ff.html#topic+ff">ff</a> matrix
</p>
</li>
<li> <p><em>aux</em> - a list with meta-data and important parameters.
</p>
</li></ul>

<p>This now contains only the selected subset of data.
</p>


<h3>Warning</h3>

<p>No checks are performed when choosing a subset of the data - it is the user's
obligation to check whether the data subset contains correct number of individuals
(especially important when using the <code>triad</code> design study) and/or markers!
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # The argument 'overwrite' is set to TRUE!
  # Read the data:
  examples.dir &lt;- system.file( "extdata", package = "Haplin" )
  example.file &lt;- file.path( examples.dir, "HAPLIN.trialdata2.txt" )
  my.gen.data.read &lt;- genDataRead( file.in = example.file, file.out = "trial_data",
   dir.out = tempdir( check = TRUE ), format = "haplin", allele.sep = "", n.vars = 2,
   cov.header = c( "smoking", "sex" ), overwrite = TRUE )
  my.gen.data.read
  # Extract part with only men:
  men.subset &lt;- genDataGetPart( my.gen.data.read, design = "triad", sex = 1,
    dir.out = tempdir( check = TRUE ), file.out = "gen_data_men_only", overwrite = TRUE )
  men.subset
  # Extract the part with only smoking women:
  women.smoke.subset &lt;- genDataGetPart( my.gen.data.read, design = "triad",
    dir.out = tempdir( check = TRUE ), sex = 0, smoking = c( 1,2 ), overwrite = TRUE )
  women.smoke.subset 

</code></pre>

<hr>
<h2 id='genDataLoad'>Loading the data previously read in and saved by &quot;genDataRead&quot;</h2><span id='topic+genDataLoad'></span>

<h3>Description</h3>

<p>This function loads the data from the saved .ffData and .RData files,
and prepares the data to subsequent analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genDataLoad(filename = stop("'filename' must be given!"), dir.in = ".")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genDataLoad_+3A_filename">filename</code></td>
<td>
<p>The base of the filenames; i.e. if the data is saved in
&quot;my_data_gen.ffData&quot;, &quot;my_data_gen.RData&quot; and &quot;my_data_cov.RData&quot;, then the 'filename'
should be &quot;my_data&quot;.</p>
</td></tr>
<tr><td><code id="genDataLoad_+3A_dir.in">dir.in</code></td>
<td>
<p>The path to the directory where files were saved (defaults to the
current directory).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with three elements:
</p>

<ul>
<li> <p><em>cov.data</em> - a <code>data.frame</code> with covariate data (if available in
the input file)
</p>
</li>
<li> <p><em>gen.data</em> - a list with chunks of the genetic data; the data is divided
column-wise, using 10,000 columns per chunk; each element of this list is a
<a href="ff.html#topic+ff">ff</a> matrix
</p>
</li>
<li> <p><em>aux</em> - a list with meta-data and important parameters.
</p>
</li></ul>


<hr>
<h2 id='genDataPreprocess'>Pre-processing of the genetic data</h2><span id='topic+genDataPreprocess'></span>

<h3>Description</h3>

<p>This function prepares the data to be used in Haplin analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genDataPreprocess(
  data.in = stop("You have to give the object to preprocess!"),
  map.file,
  map.header = FALSE,
  design = "triad",
  file.out = "data_preprocessed",
  dir.out = ".",
  ncpu = 1,
  overwrite = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genDataPreprocess_+3A_data.in">data.in</code></td>
<td>
<p>Input data, as loaded by <a href="#topic+genDataRead">genDataRead</a> or <a href="#topic+genDataLoad">genDataLoad</a>.</p>
</td></tr>
<tr><td><code id="genDataPreprocess_+3A_map.file">map.file</code></td>
<td>
<p>Filename (with path if the file is not in current directory) of the
.map file holding the SNP names, if available.</p>
</td></tr>
<tr><td><code id="genDataPreprocess_+3A_map.header">map.header</code></td>
<td>
<p>Logical: does the map.file contain a header in the first row?
Default: FALSE.</p>
</td></tr>
<tr><td><code id="genDataPreprocess_+3A_design">design</code></td>
<td>
<p>The design used in the study - choose from:
</p>

<ul>
<li> <p><em>triad</em> - (default), data includes genotypes of mother, father and child;
</p>
</li>
<li> <p><em>cc</em> - classical case-control;
</p>
</li>
<li> <p><em>cc.triad</em> - hybrid design: triads with cases and controls
</p>
</li></ul>
<p>.</p>
</td></tr>
<tr><td><code id="genDataPreprocess_+3A_file.out">file.out</code></td>
<td>
<p>The core name of the files that will contain the preprocessed data
(character string); ready to load next time with <a href="#topic+genDataLoad">genDataLoad</a> function;
default: &quot;data_preprocessed&quot;.</p>
</td></tr>
<tr><td><code id="genDataPreprocess_+3A_dir.out">dir.out</code></td>
<td>
<p>The directory that will contain the saved data; defaults to current 
working directory.</p>
</td></tr>
<tr><td><code id="genDataPreprocess_+3A_ncpu">ncpu</code></td>
<td>
<p>The number of CPU cores to use - this speeds up the process for large 
datasets significantly. Default is 1 core, maximum is 1 less than the total number
of cores available on a current machine (even if the number given by the user is
more than that).</p>
</td></tr>
<tr><td><code id="genDataPreprocess_+3A_overwrite">overwrite</code></td>
<td>
<p>Whether to overwrite the output files: if NULL (default), will prompt
the user to give answer; set to TRUE, will automatically overwrite any existing files;
and set to FALSE, will stop if the output files exist.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with three elements:
</p>

<ul>
<li> <p><em>cov.data</em> - a <code>data.frame</code> with covariate data (if available in
the input file)
</p>
</li>
<li> <p><em>gen.data</em> - a list with chunks of the genetic data; the data is divided
column-wise, using 10,000 columns per chunk; each element of this list is a
<a href="ff.html#topic+ff">ff</a> matrix
</p>
</li>
<li> <p><em>aux</em> - a list with meta-data and important parameters:
</p>

<ul>
<li> <p><em>variables</em> - tabulated information of the covariate data;
</p>
</li>
<li> <p><em>variables.nas</em> - how many NA values per each column of covariate data;
</p>
</li>
<li> <p><em>alleles</em> - all the possible alleles in each marker;
</p>
</li>
<li> <p><em>alleles.nas</em> - how many NA values in each marker;
</p>
</li>
<li> <p><em>nrows.with.missing</em> - how many rows contain any missing allele information;
</p>
</li>
<li> <p><em>which.rows.with.missing</em> - vector of indices of rows with missing data (if any)
</p>
</li></ul>
<p>.
</p>
</li></ul>



<h3>Details</h3>

<p>The .map file should contain at least two columns, where the second one contains SNP 
names. Any additional columns should be separated by a whitespace character, but will 
be ignored. The file should contain a header.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # The argument 'overwrite' is set to TRUE!
  # First, read the data:
  examples.dir &lt;- system.file( "extdata", package = "Haplin" )
  example.file &lt;- file.path( examples.dir, "exmpl_data.ped" )
  ped.data.read &lt;- genDataRead( example.file, file.out = "exmpl_ped_data", 
   dir.out = tempdir( check = TRUE ), format = "ped", overwrite = TRUE )
  ped.data.read
  # Take only part of the data (if needed)
  ped.data.part &lt;- genDataGetPart( ped.data.read, design = "triad", markers = 10:12,
   dir.out = tempdir( check = TRUE ), file.out = "exmpl_ped_data_part", overwrite = TRUE )
  # Preprocess as "triad" data:
  ped.data.preproc &lt;- genDataPreprocess( ped.data.part, design = "triad",
   dir.out = tempdir( check = TRUE ), file.out = "exmpl_data_preproc", overwrite = TRUE )
  ped.data.preproc

</code></pre>

<hr>
<h2 id='genDataRead'>Reading the genetic data from a file</h2><span id='topic+genDataRead'></span>

<h3>Description</h3>

<p>This function will read in data from PED or haplin formatted file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genDataRead(
  file.in = stop("Filename must be given!", call. = FALSE),
  file.out = NULL,
  dir.out = ".",
  format = stop("Format parameter is required!"),
  header = FALSE,
  n.vars,
  cov.file.in,
  cov.header,
  map.file,
  map.header = FALSE,
  allele.sep = ";",
  na.strings = "NA",
  col.sep = "",
  overwrite = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genDataRead_+3A_file.in">file.in</code></td>
<td>
<p>The name of the main input file with genotype information.</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_file.out">file.out</code></td>
<td>
<p>The base for the output filename (by default, constructed from the
input file name).</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_dir.out">dir.out</code></td>
<td>
<p>The path to the directory where the output files will be saved.</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_format">format</code></td>
<td>
<p>Format of data (will influence how data is processed) - choose from:
</p>

<ul>
<li> <p><em>haplin</em> - data already in one row per family,
</p>
</li>
<li> <p><em>ped</em> - data from .ped file, each row represents an individual.
</p>
</li></ul>
<p>.</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_header">header</code></td>
<td>
<p>Whether the first line of the main input file contains column names;
default: FALSE; NB: this is useful only for 'haplin'-formatted files!</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_n.vars">n.vars</code></td>
<td>
<p>The number of columns with covariate data (if any) in the main file;
NB: if the main file is in PED format, it is assumed that the first 6 columns contain
the standard PED-covariates (i.e., family ID, ID of the child, father and mother,
sex and case-control status), so in this case setting 'n.vars' is useful only
if the PED file contains more than 6 covariate columns.</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_cov.file.in">cov.file.in</code></td>
<td>
<p>Name of the file containing additional covariate data, if any. 
Caution: unless the 'cov.header' argument is used, it is assumed that the first line 
of this file contains the header (i.e., the column names of the additional data).</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_cov.header">cov.header</code></td>
<td>
<p>The character vector containing the names of covariate columns
(in the file with additional covariate data if given by the 'cov.file.in' argument;
or in the main file, if it's a &quot;haplin&quot;-formatted file).</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_map.file">map.file</code></td>
<td>
<p>Filename (with path if the file is not in current directory) of the
.map file holding the SNP names, if available (see Details).</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_map.header">map.header</code></td>
<td>
<p>Logical: does the map.file contain a header in the first row?
Default: FALSE.</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_allele.sep">allele.sep</code></td>
<td>
<p>Character: separator between two alleles (default: &quot;;&quot;).</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_na.strings">na.strings</code></td>
<td>
<p>Character or NA: how the missing data is coded (default: &quot;NA&quot;).</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_col.sep">col.sep</code></td>
<td>
<p>Character: separator between the columns (i.e., markers; default: any
whitespace character).</p>
</td></tr>
<tr><td><code id="genDataRead_+3A_overwrite">overwrite</code></td>
<td>
<p>Whether to overwrite the output files: if NULL (default), will prompt
the user to give answer; set to TRUE, will automatically overwrite any existing files;
and set to FALSE, will stop if the output files exist.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function reads in all the data in the file, creates <a href="ff.html#topic+ff">ff</a> objects to store
the genetic information and <a href="base.html#topic+data.frame">data.frame</a> to store covariate data (if any). These
objects are saved in <code>.RData</code> and <code>.ffData</code> files, which can be later on
easily uploaded to R (with <a href="#topic+genDataLoad">genDataLoad</a>) and re-used.
</p>


<h3>Value</h3>

<p>A list object with three elements:
</p>

<ul>
<li> <p><em>cov.data</em> - a <code>data.frame</code> with covariate data (if available in
the input file)
</p>
</li>
<li> <p><em>gen.data</em> - a list with chunks of the genetic data; the data is divided
column-wise, using 10,000 columns per chunk; each element of this list is a
<a href="ff.html#topic+ff">ff</a> matrix
</p>
</li>
<li> <p><em>aux</em> - a list with meta-data and important parameters.
</p>
</li></ul>



<h3>Details</h3>

<p>The .map file should contain at least two columns, where the second one contains SNP 
names. Any additional columns should be separated by a whitespace character, but will 
be ignored. The file should contain a header.
</p>


<h3>Usage note</h3>

<p>When reading in a covariate file together with the genotype information, it is advised
to include the header in the file, so that there is no doubt to the naming of
the data columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # The argument 'overwrite' is set to TRUE!
  examples.dir &lt;- system.file( "extdata", package = "Haplin" )
  # ped format:
  example.file2 &lt;- file.path( examples.dir, "exmpl_data.ped" )
  ped.data.read &lt;- genDataRead( example.file2, file.out = "exmpl_ped_data", 
   dir.out = tempdir( check = TRUE ), format = "ped", overwrite = TRUE )
  ped.data.read
  # haplin format:
  example.file1 &lt;- file.path( examples.dir, "HAPLIN.trialdata2.txt" )
  haplin.data.read &lt;- genDataRead( file.in = example.file1,
   file.out = "exmpl_haplin_data", format = "haplin", allele.sep = "", n.vars = 2, 
   cov.header = c( "smoking", "sex" ), overwrite = TRUE,
   dir.out = tempdir( check = TRUE ) )
  haplin.data.read

</code></pre>

<hr>
<h2 id='getChildren'>Getter for all rows with children data</h2><span id='topic+getChildren'></span>

<h3>Description</h3>

<p>Wrapper function for <a href="#topic+genDataGetPart">genDataGetPart</a> that returns a subset of the data containing 
only children.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getChildren(
  data.in = stop("No data given!", call. = FALSE),
  file.out = "my_data_onlyChildren",
  dir.out = ".",
  overwrite = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getChildren_+3A_data.in">data.in</code></td>
<td>
<p>The data object (in format as the output of <a href="#topic+genDataRead">genDataRead</a>); note
that the design of the data is assumed to be triad.</p>
</td></tr>
<tr><td><code id="getChildren_+3A_file.out">file.out</code></td>
<td>
<p>The base for the output filename (default: &quot;my_data_onlyChildren&quot;).</p>
</td></tr>
<tr><td><code id="getChildren_+3A_dir.out">dir.out</code></td>
<td>
<p>The path to the directory where the output files will be saved.</p>
</td></tr>
<tr><td><code id="getChildren_+3A_overwrite">overwrite</code></td>
<td>
<p>Whether to overwrite the output files: if NULL (default), will prompt
the user to give answer; set to TRUE, will automatically overwrite any existing files;
and set to FALSE, will stop if the output files exist.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with three elements:
</p>

<ul>
<li> <p><em>cov.data</em> - a <code>data.frame</code> with covariate data (if available in
the input file)
</p>
</li>
<li> <p><em>gen.data</em> - a list with chunks of the genetic data; the data is divided
column-wise, using 10,000 columns per chunk; each element of this list is a
<a href="ff.html#topic+ff">ff</a> matrix
</p>
</li>
<li> <p><em>aux</em> - a list with meta-data and important parameters.
</p>
</li></ul>

<p>This now contains only the selected subset of data.
</p>

<hr>
<h2 id='getDyads'>Getter only for all dyads (child and one parent)</h2><span id='topic+getDyads'></span>

<h3>Description</h3>

<p>Wrapper function for <a href="#topic+genDataGetPart">genDataGetPart</a> that returns a subset of the data containing 
only dyads (where the child and only one parent have genetic data), i.e., not triads.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDyads(
  data.in = stop("No data given!", call. = FALSE),
  file.out = "my_data_onlyDyads",
  dir.out = ".",
  overwrite = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDyads_+3A_data.in">data.in</code></td>
<td>
<p>The data object (in format as the output of <a href="#topic+genDataRead">genDataRead</a>); note
that the design of the data is assumed to be &quot;triad&quot;.</p>
</td></tr>
<tr><td><code id="getDyads_+3A_file.out">file.out</code></td>
<td>
<p>The base for the output filename (default: &quot;my_data_onlyDyads&quot;).</p>
</td></tr>
<tr><td><code id="getDyads_+3A_dir.out">dir.out</code></td>
<td>
<p>The path to the directory where the output files will be saved
(default: &quot;.&quot;, the current directory).</p>
</td></tr>
<tr><td><code id="getDyads_+3A_overwrite">overwrite</code></td>
<td>
<p>Whether to overwrite the output files: if NULL (default), will prompt
the user to give answer; set to TRUE, will automatically overwrite any existing files;
and set to FALSE, will stop if the output files exist.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with three elements:
</p>

<ul>
<li> <p><em>cov.data</em> - a <code>data.frame</code> with covariate data (if available in
the input file)
</p>
</li>
<li> <p><em>gen.data</em> - a list with chunks of the genetic data; the data is divided
column-wise, using 10,000 columns per chunk; each element of this list is a
<a href="ff.html#topic+ff">ff</a> matrix
</p>
</li>
<li> <p><em>aux</em> - a list with meta-data and important parameters.
</p>
</li></ul>

<p>This now contains only the selected subset of data.
</p>

<hr>
<h2 id='getFathers'>Getter for all rows with fathers' data</h2><span id='topic+getFathers'></span>

<h3>Description</h3>

<p>Wrapper function for <a href="#topic+genDataGetPart">genDataGetPart</a> that returns a subset of the data containing 
only fathers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getFathers(
  data.in = stop("No data given!", call. = FALSE),
  file.out = "my_data_onlyFathers",
  dir.out = ".",
  overwrite = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getFathers_+3A_data.in">data.in</code></td>
<td>
<p>The data object (in format as the output of <a href="#topic+genDataRead">genDataRead</a>); note
that the design of the data is assumed to be triad.</p>
</td></tr>
<tr><td><code id="getFathers_+3A_file.out">file.out</code></td>
<td>
<p>The base for the output filename (default: &quot;my_data_onlyFathers&quot;).</p>
</td></tr>
<tr><td><code id="getFathers_+3A_dir.out">dir.out</code></td>
<td>
<p>The path to the directory where the output files will be saved.</p>
</td></tr>
<tr><td><code id="getFathers_+3A_overwrite">overwrite</code></td>
<td>
<p>Whether to overwrite the output files: if NULL (default), will prompt
the user to give answer; set to TRUE, will automatically overwrite any existing files;
and set to FALSE, will stop if the output files exist.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with three elements:
</p>

<ul>
<li> <p><em>cov.data</em> - a <code>data.frame</code> with covariate data (if available in
the input file)
</p>
</li>
<li> <p><em>gen.data</em> - a list with chunks of the genetic data; the data is divided
column-wise, using 10,000 columns per chunk; each element of this list is a
<a href="ff.html#topic+ff">ff</a> matrix
</p>
</li>
<li> <p><em>aux</em> - a list with meta-data and important parameters.
</p>
</li></ul>

<p>This now contains only the selected subset of data.
</p>

<hr>
<h2 id='getFullTriads'>Getter for all full triads</h2><span id='topic+getFullTriads'></span>

<h3>Description</h3>

<p>Wrapper function for <a href="#topic+genDataGetPart">genDataGetPart</a> that returns a subset of the data containing 
only full triads (where all, the child, the mother and the father have genetic data).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getFullTriads(
  data.in = stop("No data given!", call. = FALSE),
  file.out = "my_data_onlyTriads",
  dir.out = ".",
  overwrite = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getFullTriads_+3A_data.in">data.in</code></td>
<td>
<p>The data object (in format as the output of <a href="#topic+genDataRead">genDataRead</a>); note
that the design of the data is assumed to be triad.</p>
</td></tr>
<tr><td><code id="getFullTriads_+3A_file.out">file.out</code></td>
<td>
<p>The base for the output filename (default: &quot;my_data_onlyTriads&quot;).</p>
</td></tr>
<tr><td><code id="getFullTriads_+3A_dir.out">dir.out</code></td>
<td>
<p>The path to the directory where the output files will be saved.</p>
</td></tr>
<tr><td><code id="getFullTriads_+3A_overwrite">overwrite</code></td>
<td>
<p>Whether to overwrite the output files: if NULL (default), will prompt
the user to give answer; set to TRUE, will automatically overwrite any existing files;
and set to FALSE, will stop if the output files exist.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with three elements:
</p>

<ul>
<li> <p><em>cov.data</em> - a <code>data.frame</code> with covariate data (if available in
the input file)
</p>
</li>
<li> <p><em>gen.data</em> - a list with chunks of the genetic data; the data is divided
column-wise, using 10,000 columns per chunk; each element of this list is a
<a href="ff.html#topic+ff">ff</a> matrix
</p>
</li>
<li> <p><em>aux</em> - a list with meta-data and important parameters.
</p>
</li></ul>

<p>This now contains only the selected subset of data.
</p>

<hr>
<h2 id='getMothers'>Getter for all rows with mothers' data</h2><span id='topic+getMothers'></span>

<h3>Description</h3>

<p>Wrapper function for <a href="#topic+genDataGetPart">genDataGetPart</a> that returns a subset of the data containing 
only mothers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMothers(
  data.in = stop("No data given!", call. = FALSE),
  file.out = "my_data_onlyMothers",
  dir.out = ".",
  overwrite = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getMothers_+3A_data.in">data.in</code></td>
<td>
<p>The data object (in format as the output of <a href="#topic+genDataRead">genDataRead</a>); note
that the design of the data is assumed to be triad.</p>
</td></tr>
<tr><td><code id="getMothers_+3A_file.out">file.out</code></td>
<td>
<p>The base for the output filename (default: &quot;my_data_onlyMothers&quot;).</p>
</td></tr>
<tr><td><code id="getMothers_+3A_dir.out">dir.out</code></td>
<td>
<p>The path to the directory where the output files will be saved.</p>
</td></tr>
<tr><td><code id="getMothers_+3A_overwrite">overwrite</code></td>
<td>
<p>Whether to overwrite the output files: if NULL (default), will prompt
the user to give answer; set to TRUE, will automatically overwrite any existing files;
and set to FALSE, will stop if the output files exist.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with three elements:
</p>

<ul>
<li> <p><em>cov.data</em> - a <code>data.frame</code> with covariate data (if available in
the input file)
</p>
</li>
<li> <p><em>gen.data</em> - a list with chunks of the genetic data; the data is divided
column-wise, using 10,000 columns per chunk; each element of this list is a
<a href="ff.html#topic+ff">ff</a> matrix
</p>
</li>
<li> <p><em>aux</em> - a list with meta-data and important parameters.
</p>
</li></ul>

<p>This now contains only the selected subset of data.
</p>

<hr>
<h2 id='gxe'>Test for gene-environment interaction</h2><span id='topic+gxe'></span>

<h3>Description</h3>

<p>Performes a gene-environment test to check if <code>haplin</code> estimates of relative risks change over strata of environment. It is typically applied to the output from <code>haplinStrat</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gxe(object.list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gxe_+3A_object.list">object.list</code></td>
<td>
<p>A list of <code>haplin</code> results, almost always the output from <code>haplinStrat</code>. The first element is the result of running <code>haplin</code> on all data; the remaining elements are the results for each stratum separately.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>haplinStrat</code> runs <code>haplin</code> first on the entire input data file, then on each stratum separately. The results from <code>haplinStrat</code> are similar to just manually splitting the file into strata and running <code>haplin</code> on each, with one important difference, however: Since some strata may be small etc., haplin might conceivably choose different haplotypes in different strata, and also choose different reference haplotypes. When first running haplin on the entire file, <code>haplinStrat</code> saves the selected haplotypes and chosen reference category. Then, in the strata-specific runs haplinStrat forces haplin to choose the same haplotypes/reference category in all runs, so that results from different strata are comparable.
When applying <code>gxe</code> to the output from haplinStrat, it will test whether there is a statistically significant change in parameter estimates from stratum to stratum, i.e. a gene-environment interaction since strata usually are defined by an environmental exposure.
<code>gxe</code> uses Wald tests to test for interactions. It always tests whether there is change in haplotype frequencies from stratum to stratum. More importantly, it separately tests whether any genetic effects, such as fetal genetic effects, maternal effects, or parent-of-origin effects, change significantly over strata.
<code>gxe</code> can also be run from within <code>haplinSlide</code> by using the <code>strata</code> argument in <code>haplinSlide</code>.
</p>


<h3>Value</h3>

<p>A dataframe with one row for each test that is performed (haplo.freq is the first, the remaining depend on the model that has been estimated). The Wald chi-squared test value, degrees-of-freedom, and resulting p-value are reported.
</p>


<h3>NOTE:</h3>

<p>In the future, the structure of the output from gxe will change. In particular, measures of ratios of relative risks will be reported in addition to the p-values</p>


<h3>Note</h3>

<p>Further information is found on the web page.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+haplinStrat">haplinStrat</a></code>, <code><a href="#topic+haplinSlide">haplinSlide</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# All standard haplin runs can be done with haplinStrat. 
# Below is an illustration. See the haplin help page for more 
# examples.
# 
# Analyzing the effect of fetal genes, including triads with missing data,
# using a multiplicative response model. The first column of the data file
# in this example contains the stratification variable.
result &lt;- haplinStrat("C:/work/data.dat", strata = 1, use.missing = T, response = "mult",
reference = "ref.cat", winlength = 1)
# Provide summary of separate results:
lapply(result, summary)
# Plot results separately:
par(ask = T)
lapply(result, plot)
#
# Convert results to table format and stack them over strata:
haptable(result)
# Test for interaction between haplotype risk estimates and the strata variable:
gxe(result)


## End(Not run)

</code></pre>

<hr>
<h2 id='haplin'>Fitting log-linear models to case-parent triad and/or case-control data</h2><span id='topic+haplin'></span>

<h3>Description</h3>

<p><code>haplin</code> fits a log-linear model to case-parent triads, case-control data, or combined (hybrid) case-parent control-parent triads or dyads. It estimates marker or haplotype frequencies, and uses the EM algorithm to reconstruct haplotypes and, if requested, impute missing genotypes. <code>haplin</code> prints and plots estimates of relative risks associated with fetal and maternal haploypes, and in addition allows splitting fetal haplotype effects into maternally and paternally inherited effects. It allows special models, like x-inactivation, to be fitted on the X-chromosome. The result is an object of class <code>haplin</code>, which can be explored with <code>summary</code>, <code>plot</code>, and <code>haptable</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplin( data, markers = "ALL", 
design = "triad", use.missing = FALSE, 
xchrom = FALSE, maternal = FALSE, test.maternal = FALSE, 
poo = FALSE, scoretest = "no", ccvar = NULL, strata = NULL, 
sex = NULL, comb.sex = "double",
reference = "reciprocal", response = "free", 
threshold = 0.01, max.haplos = NULL, haplo.file = NULL, 
resampling = "no", max.EM.iter = 50, data.out = "no", 
verbose = TRUE, printout = TRUE )
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haplin_+3A_data">data</code></td>
<td>
<p>An R-object which is the result of using <a href="#topic+genDataPreprocess">genDataPreprocess</a>. See the web page for a detailed description of how to use this function.</p>
</td></tr>
<tr><td><code id="haplin_+3A_markers">markers</code></td>
<td>
<p>Default is &quot;ALL&quot;, which means <code>haplin</code> uses all available markers in the data set in the analysis. For the current version of <code>haplin</code> the number of markers used at a single run should probably not exceed 4 or 5 due to the computational burden. The markers argument can be used to select appropriate markers from the file without creating a new file for the selected markers. The relevant markers can be specified by giving a vector or numbers (e.g., <code>markers = c(1, 3:10)</code> will use the 10 first markers except marker 2) or characters (e.g., <code>markers = c("m1", "m3", "rs35971")</code>). When running <code>haplin</code>, it may be a good idea to start exploring a few markers at a time, using this argument.  </p>
</td></tr>
<tr><td><code id="haplin_+3A_design">design</code></td>
<td>
<p>The value &quot;triad&quot; is used for the standard case triad design, without independent controls. The value &quot;cc.triad&quot; means a combination of case triads and control triads. This requires the argument <code>ccvar</code> to point to the data column containing the case-control variable. The value &quot;cc&quot; means a simple case-control design, where the parents have not been genotyped (there are no data columns for parental genes). NOTE: <code>design</code> is also set in <code>genDataPreprocess</code>. Almost always, the two arguments should be equal. Occasionally, however, the user might want to override the original argument by switching from 'cc.triad' to 'triad' or vice versa.	</p>
</td></tr>
<tr><td><code id="haplin_+3A_use.missing">use.missing</code></td>
<td>
<p> A logical value used to determine whether triads with missing data should be included in the analysis. When set to TRUE, <code>haplin</code> uses the EM algorithm to obtain risk estimates, also taking into account triads with missing data. The standard errors and p-values are adjusted to correct for this. The default, however, is FALSE. When FALSE, all triads having  any sort of missing data are excluded before the analysis is run. Note that <code>haplin</code> only looks at markers actually used in the analysis,  so that if the markers  argument (see below) is used to select a collection of markers for analysis, <code>haplin</code> only excludes triads with missing data on the included markers. </p>
</td></tr>
<tr><td><code id="haplin_+3A_xchrom">xchrom</code></td>
<td>
<p>Logical, defaults to &quot;FALSE&quot;. If set to &quot;TRUE&quot;, <code>haplin</code> assumes the markers are on the x-chromosome. This option should be combined with specifying the <code>sex</code> argument. In addition, <code>comb.sex</code> can be useful. <code>xchrom = T</code> can be combined with <code>poo = T</code> and/or <code>maternal = T</code>.</p>
</td></tr>
<tr><td><code id="haplin_+3A_maternal">maternal</code></td>
<td>
<p> If TRUE, maternal effects are estimated as well as the standard fetal effects. </p>
</td></tr>
<tr><td><code id="haplin_+3A_test.maternal">test.maternal</code></td>
<td>
<p>Not yet implemented.</p>
</td></tr>
<tr><td><code id="haplin_+3A_poo">poo</code></td>
<td>
<p>Parent-Of-Origin effects. If TRUE, <code>haplin</code> will split single-dose effects into two separate effect estimates, one for the maternally inherited haplotype, and one for the paternally inherited haplotype. Double dose will be estimated as before.</p>
</td></tr>
<tr><td><code id="haplin_+3A_scoretest">scoretest</code></td>
<td>
<p>Special interest only. If &quot;no&quot;, no score test is computed. If &quot;yes&quot;, an overall score p-value is included in the output, and the individual score values are returned in the <code>haplin</code> object. If &quot;only&quot;, <code>haplin</code> is only run under the null hypothesis, and a simple score object is returned instead of the full <code>haplin</code> object. Useful if only score testing is needed.</p>
</td></tr>
<tr><td><code id="haplin_+3A_ccvar">ccvar</code></td>
<td>
<p>Numeric. Should give the column number for the column containing the case-control indicator in the data file. Needed for the &quot;cc&quot; and &quot;cc.triad&quot; designs. The column should contain two numeric values, of which the largest one is always used to denote cases.</p>
</td></tr>
<tr><td><code id="haplin_+3A_strata">strata</code></td>
<td>
<p>Not yet implemented.</p>
</td></tr>
<tr><td><code id="haplin_+3A_sex">sex</code></td>
<td>
<p>To be used with <code>xchrom = TRUE</code>. A numeric value specifying which of the data columns that contains the sex variable. The variable should be coded 1 for males and 2 for females.</p>
</td></tr>
<tr><td><code id="haplin_+3A_comb.sex">comb.sex</code></td>
<td>
<p>To be used with <code>xchrom = TRUE</code>. A character value that specifies how to handle gender differences on the X-chromosome. If set to &quot;males&quot; or &quot;females&quot;, analyses are done either for just males or just females, respectively. If set to &quot;single&quot; or &quot;double&quot;, males and females are used in a combined analysis. Specifically, when &quot;single&quot;, the effect of a (single) allele in males is assumed to equal the effect of a single allele dose in females, and similarly, when &quot;double&quot;, a single allele in males is assumed to have the same effect as a double allele dose in females. Default is &quot;double&quot;, which corresponds to X-inactivation. See separate description for more details.</p>
</td></tr>
<tr><td><code id="haplin_+3A_reference">reference</code></td>
<td>
<p>Decides how <code>haplin</code> chooses its reference category for the effect estimates. Default value is &quot;reciprocal&quot;. With the reciprocal reference the effect of a single or double dose of each haplotype is measured relative to the remaining haplotypes. This means that a new reference category is used for each single haplotype. Other possible values are &quot;population&quot; (which is similar to reciprocal, but where the reference category is always the total population), and &quot;ref.cat&quot;, where a single haplotype is used as reference for all the rest. For ref.cat, the default is to choose the most frequent haplotype as the reference haplotype. The reference haplotype can be set explicitly by giving a numeric value for the reference argument. Note that the numeric value refers to the haplotype's position among the haplotypes selected for analysis by <code>haplin</code>. This means that one should run <code>haplin</code> once first to see what haplotypes are used before giving a numeric value to reference. </p>
</td></tr>
<tr><td><code id="haplin_+3A_response">response</code></td>
<td>
<p>The default value &quot;free&quot; means that both single- and double dose effects are estimated. Choosing &quot;mult&quot; instead specifies a multiplicative dose-response model. </p>
</td></tr>
<tr><td><code id="haplin_+3A_threshold">threshold</code></td>
<td>
<p>Sets the (approximate) lower limit for the haplotype frequencies of those haplotypes that should be retained in the analysis. Hapotypes that are less frequent are removed, and information about this is given in the output. Default is 0.01.</p>
</td></tr>
<tr><td><code id="haplin_+3A_max.haplos">max.haplos</code></td>
<td>
<p>Not yet implemented.</p>
</td></tr>
<tr><td><code id="haplin_+3A_haplo.file">haplo.file</code></td>
<td>
<p>Not yet implemented.</p>
</td></tr>
<tr><td><code id="haplin_+3A_resampling">resampling</code></td>
<td>
<p>Mostly for testing. Default is &quot;no&quot;. When &quot;no&quot;, the individual haplotypes reconstructed by the EM algorithm as assumed known when computing CIs and p-values. If set to &quot;jackknife&quot; a jackknife-based resampling procedure is used when computing confidence intervals and p-values for effect estimates. This takes more time, but corrects the CIs and p-values for the uncertainty contained in unphased data. Note: in all recent versions of <code>haplin</code>, the resampling is no longer needed since the confidence intervals and p-values are already corrected in the standard computation. </p>
</td></tr>
<tr><td><code id="haplin_+3A_max.em.iter">max.EM.iter</code></td>
<td>
<p>The maximum number of iterations used by the EM algorithm. This value can be increased if necessary, which sometimes is the case with e.g. case-control data which a substantial amount of missing. However, for triad data with little missing information there is usually no need for many iterations. </p>
</td></tr>
<tr><td><code id="haplin_+3A_data.out">data.out</code></td>
<td>
<p>Character. Accepts values &quot;no&quot;, &quot;prelim&quot;, &quot;null&quot; or &quot;full&quot;, with &quot;no&quot; as default. For values other than default, <code>haplin</code> returns the data file prepared for analysis rather than the usual <code>haplin</code> estimation results. The data file contains the haplotypes identified for each triad, and a vector of weights giving the probability distribution of different haplotype configurations within a triad. The probabilities are computed from preliminary haplotype frequency estimates, from the null model or from the full likelihood model. The &quot;prelim&quot; option will be much faster but somewhat less precise than the likelihood models.</p>
</td></tr>
<tr><td><code id="haplin_+3A_verbose">verbose</code></td>
<td>
<p> Default is T (=TRUE). During the EM algorithm, <code>haplin</code> prints the estimated parameters and deviance for each step. To avoid the output, set this argument to F (=FALSE). </p>
</td></tr>
<tr><td><code id="haplin_+3A_printout">printout</code></td>
<td>
<p>Logical. If TRUE (default), <code>haplin</code> prints a full summary of the results after finishing the estimation. If FALSE, no such printout is given, but the <code>summary</code> function can later be applied to a saved result to get the same summary.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Input data can be either a <code>haplin</code> format data file, or a PED data. These have to be loaded into R first, using <a href="#topic+genDataRead">genDataRead</a> or <a href="#topic+genDataLoad">genDataLoad</a> functions, and then pre-processed with the <a href="#topic+genDataPreprocess">genDataPreprocess</a> function. If the PED data file is used, the arguments <code>filename</code>, <code>n.vars</code>, <code>sep</code>, <code>allele.sep</code>, <code>na.strings</code>, <code>ccvar</code>, and <code>sex</code> need not be specified.<br />
The output can be examined by <code>print</code>, <code>summary</code>, <code>plot</code> and <code>haptable</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>haplin</code> is returned.
(The only exception is when <code>data.out</code> is set different from &quot;no&quot;, where <code>haplin</code> will produce a data file with haplotypes identified.)
</p>


<h3>Warning</h3>

<p>Typically, some of the included haplotypes will be relatively rare, such as a frequency of 1% - 5%. For those haplotypes there may be too little data to estimate the double doses properly, so the estimates may be unreliable. This is seen from the extremely wide confidence intervals. The rare double dose estimates should be disregarded, but the remaining single and double dose estimates are valid. To avoid the problem one can also reduce the model to a purely multiplicative model by setting <code>response = "mult"</code> combined with <code>reference = "ref.cat"</code>.</p>


<h3>Note</h3>

<p>Further information is found on the web page.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.haplin">summary.haplin</a></code>, <code><a href="#topic+plot.haplin">plot.haplin</a></code>, <code><a href="#topic+pedToHaplin">pedToHaplin</a></code>, <code><a href="#topic+haptable">haptable</a></code>, <code><a href="#topic+haplinSlide">haplinSlide</a></code>, <code><a href="#topic+genDataLoad">genDataLoad</a></code>, <code><a href="#topic+genDataRead">genDataRead</a></code>, <code><a href="#topic+genDataPreprocess">genDataPreprocess</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># setting up the directory with exemplary data
dir.in &lt;- system.file( "extdata", package = "Haplin" )
file.in &lt;- file.path( dir.in, "data.dat" )

# reading data in
data.in &lt;- genDataRead( file.in, file.out = "poo_exmpl_data_read", format = "haplin",
  dir.out = tempdir( check = TRUE ), n.vars = 1, allele.sep = " ", col.sep = " ",
  overwrite = TRUE )
# preprocessing the data
data.preproc &lt;- genDataPreprocess( data.in, design = "triad",
  file.out = "poo_exmpl_data_preproc", dir.out = tempdir( check = TRUE ), overwrite = TRUE )

# running haplin, calculating POO
res.POO &lt;- haplin( data.preproc, markers = 2, poo = TRUE, response = "mult",
  reference = 2, use.missing = TRUE )
res.POO

## Not run: 
# 1. Read the data:
my.haplin.data &lt;- genDataRead( file.in = "HAPLIN.trialdata.txt", file.out =
  "trial_data1", dir.out = ".", format = "haplin", n.vars = 0 )

# 2. Run pre-processing:
haplin.data.prep &lt;- genDataPreprocess( data.in = my.haplin.data, format =
  "haplin", design = "triad", file.out = "trial_data1_prep", dir.out = "." )

# 3. Analyze:
# Standard run:
haplin( haplin.data.prep )

# Specify path, estimate maternal effects:
haplin( haplin.data.prep, maternal = T )

# Specify path, use haplotype no. 2 as reference:
haplin( haplin.data.prep, reference = 2 )

# Remove more haplotypes from estimation by increasing the threshold 
# to 5%:
haplin( haplin.data.prep, threshold = 0.05 )

# Estimate maternal effects, using the most frequent haplotype as reference. 
# Use all data, including triads with missing data. Select 
# markers 3, 4 and 8 from the supplied data.
haplin( haplin.data.prep, use.missing = T, maternal = T, 
reference = "ref.cat", markers = c(3,4,8) )
# Note: in this version of haplin, the jackknife is 
# no longer necessary since the standard errors are already corrected.

# Some examples showing how to save the haplin result and later 
# recall plot and summary results:

# Same analysis as above, saving the result in the object "result.1":
result.1 &lt;- haplin( haplin.data.prep, use.missing = T, maternal = T, 
reference = "ref.cat", markers = c(3,4,8) )

# Replot the saved result (fetal effects):
plot( result.1 )

# Replot the saved result (maternal effects):
plot( result.1, plot.maternal = T )

# Print a very short summary of saved result:
result.1

# A full summary of saved result, with confidence intervals and 
# p-values (the same as haplin prints when running):
summary( result.1 )

# Some examples when the data file contains two covariates, 
# the second is the case-control variable:

# The following standard triad run is INCORRECT since it disregards 
# case status:
haplin("data.dat", use.missing = T, n.vars = 2, design = "triad")

# Combined run on "hybrid" design, correctly using both case-parent 
# triads and control-parent triads:
haplin( my.haplin.data, use.missing = T, n.vars = 2, ccvar = 2, 
design = "cc.triad" )

# If parent columns are not in the file, a plain case-control 
# run can be used:
haplin( my.haplin.data, use.missing = T, n.vars = 2, ccvar = 2, 
design = "cc", response = "mult", reference = "ref.cat" )

# An example of how to produce a data file with all possible haplotypes
# identified for each triad, together with their probaility weights:
result.data &lt;- haplin( my.haplin.data, use.missing = T, 
markers = c(3,4,8), data.out = "prelim" )
# result.data will then contain the data file, with a vector of 
# probabilities (freq) computed from the preliminary haplotype
# frequencies.

## End(Not run)

</code></pre>

<hr>
<h2 id='haplinSlide'>Run haplin analysis in a series of sliding windows over a sequence of markers/SNPs</h2><span id='topic+haplinSlide'></span>

<h3>Description</h3>

<p>Produces a list, each element of which is an object of class <code>haplin</code>, which is the result of fitting the log-linear <code>haplin</code> models to the data one &quot;window&quot; at a time. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplinSlide( data, markers = "ALL", winlength = 1, 
strata = NULL, table.output = TRUE, cpus = 1, para.env = NULL, slaveOutfile = "", 
printout = FALSE, verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haplinSlide_+3A_data">data</code></td>
<td>
<p>R-object of class &quot;haplin.ready&quot;, which is e.g., output from <code><a href="#topic+genDataPreprocess">genDataPreprocess</a></code> or <code><a href="#topic+genDataLoad">genDataLoad</a></code>, and contains covariate and genetic data.</p>
</td></tr>
<tr><td><code id="haplinSlide_+3A_markers">markers</code></td>
<td>
<p>Default is &quot;ALL&quot;, which means haplinSlide uses all available markers in the data set in the analysis. Alternatively, the relevant markers can be specified by giving a vector or numbers (e.g., <code>markers = c(1, 3:10)</code> will use the 10 first markers except marker 2) or characters (e.g., <code>markers = c("m1", "m3", "rs35971")</code>). <code>haplinSlide</code> will then run haplin on a series of windows selected from the supplied <code>markers</code>. The <code>winlength</code> argument decides the length of the windows. See details.</p>
</td></tr>
<tr><td><code id="haplinSlide_+3A_winlength">winlength</code></td>
<td>
<p>Length of the sliding, overlapping windows to be run along the markers. See details.</p>
</td></tr>
<tr><td><code id="haplinSlide_+3A_strata">strata</code></td>
<td>
<p>A single numeric value specifying which data column contains the stratification variable.</p>
</td></tr>
<tr><td><code id="haplinSlide_+3A_table.output">table.output</code></td>
<td>
<p>If <code>TRUE</code>, the <code>haptable</code> function will be applied to each result after estimation, greatly reducing the size of the output. If <code>FALSE</code>, each element of the output list is a standard <code>haplin</code> object. To preserve memory, default is set to TRUE.</p>
</td></tr>
<tr><td><code id="haplinSlide_+3A_cpus">cpus</code></td>
<td>
<p><code>haplinSlide</code> allows parallel processing of its analyses. The <code>cpus</code> argument should preferably be set to the number of available cpu's. If set lower, it will save some capacity for other processes to run. Setting it too high should not cause any serious problems.</p>
</td></tr>
<tr><td><code id="haplinSlide_+3A_para.env">para.env</code></td>
<td>
<p>The user can choose parallel environment to use &mdash; &quot;parallel&quot; (default) or &quot;Rmpi&quot; (for use on clusters); this option is used only when <code>cpus</code> argument is larger than 1.</p>
</td></tr>
<tr><td><code id="haplinSlide_+3A_slaveoutfile">slaveOutfile</code></td>
<td>
<p>Character. To be used when <code>cpus &gt; 1</code>. If <code>slaveOutfile = ""</code> (default), output from all running cores will be printed in the standard R session window. Alternatively, the output can be saved to a file by specifying the file path and name.</p>
</td></tr>
<tr><td><code id="haplinSlide_+3A_printout">printout</code></td>
<td>
<p>Default is FALSE. If TRUE, provides a full summary of each <code>haplin</code> result during the run of <code>haplinSlide</code>.</p>
</td></tr>
<tr><td><code id="haplinSlide_+3A_verbose">verbose</code></td>
<td>
<p>Same as for <code>haplin</code>, but defaults to FALSE to reduce output size.</p>
</td></tr>
<tr><td><code id="haplinSlide_+3A_...">...</code></td>
<td>
<p>Remaining arguments to be used by <code><a href="#topic+haplin">haplin</a></code> in each run.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>haplinSlide</code> runs <code>haplin</code> on a series of overlapping windows of the chosen markers. Except for the <code>markers</code> and <code>winlength</code> arguments, all arguments are used exactly as in <code>haplin</code> itself. For instance, if <code>markers = c(1, 3, 4, 5, 7, 8)</code> and <code>winlength = 4</code>, <code>haplinSlide</code> will run <code>haplin</code> on first the markers <code>c(1, 3, 4, 5)</code>, then on <code>c(3, 4, 5, 7)</code>, and finally on <code>c(4, 5, 7, 8)</code>. The results are returned in a list. The elements are named &quot;1-3-4-5&quot; etc., and can be extracted with, say, <code>summary(res[["1-3-4-5"]])</code> etc., where <code>res</code> is the saved result. Or the output can be examined by, for instance, using <code>lapply(res, summary)</code> and <code>lapply(res, plot)</code>.<br />
When running <code>haplinSlide</code> on a large number of markers, the output can become prohibitively large. In that case <code>table.output</code> should be set to <code>TRUE</code>, and <code>haplinSlide</code> will return a list of summary &quot;haptables&quot;. This list can then be stacked into a single dataframe using <code>toDataFrame</code>. To avoid exessive memory use, the default is <code>table.output = TRUE</code>.<br />
When multiple cores are available, set the <code>cpus</code> to the number of cores that should be used. This will run <code>haplinSlide</code> in parallel on the chosen number of cores. Note that feedback is provided by each of the cores separately, and some cores may start working on markers far out in the sequence.
</p>


<h3>Value</h3>

<p>A list of objects of class haplin is returned.
</p>


<h3>Note</h3>

<p>Further information is found on the web page.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+summary.haplin">summary.haplin</a></code>, <code><a href="#topic+plot.haplin">plot.haplin</a></code>, <code><a href="#topic+haptable">haptable</a></code>, <code><a href="#topic+toDataFrame">toDataFrame</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# (Almost) all standard haplin runs can be done with haplinSlide. 
# Below is an illustration. See the haplin help page for more 
# examples.
# 

# 1. Read the data:
my.haplin.data &lt;- genDataRead( file.in = "HAPLIN.trialdata.txt", file.out =
  "trial_data1", dir.out = tempdir( check = TRUE ), format = "haplin", n.vars = 0 )

# 2. Run pre-processing:
haplin.data.prep &lt;- genDataPreprocess( data.in = my.haplin.data,
  format = "haplin", design = "triad", file.out = "trial_data1_prep",
  dir.out = tempdir( check = TRUE ) )

# 3. Analyze:
# Analyzing the effect of fetal genes, including triads with missing data,
# using a multiplicative response model. When winlength = 1, separate
# markers are used. To make longer windows, winlength can be increased
# correspondingly:
result.1 &lt;- haplinSlide( haplin.data.prep, use.missing = T, response = "mult",
reference = "ref.cat", winlength = 1, table.output = F)
# Provide summary of separate results:
lapply(result.1, summary)
# Plot results:
par(ask = T)
lapply(result.1, plot)


## End(Not run)

</code></pre>

<hr>
<h2 id='haplinStrat'>Fit <code>haplin</code> to each subset/stratum of data, determined by the argument <code>strata</code></h2><span id='topic+haplinStrat'></span>

<h3>Description</h3>

<p>Produces a list, each element of which is an object of class <code>haplin</code>, which is the result of fitting the log-linear <code>haplin</code> models to each <code>strata</code> stratum independently. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplinStrat( data, strata = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haplinStrat_+3A_data">data</code></td>
<td>
<p>R-object of class &quot;haplin.ready&quot;, which is e.g., output from <code><a href="#topic+genDataPreprocess">genDataPreprocess</a></code> or <code><a href="#topic+genDataLoad">genDataLoad</a></code>, and contains covariate and genetic data.</p>
</td></tr>
<tr><td><code id="haplinStrat_+3A_strata">strata</code></td>
<td>
<p>A single integer specifying the number of the column in the covariate data that contains the stratification variable</p>
</td></tr>
<tr><td><code id="haplinStrat_+3A_...">...</code></td>
<td>
<p>Remaining arguments to be used by <code><a href="#topic+haplin">haplin</a></code> in each run.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>haplinStrat</code> runs <code>haplin</code> first on the entire input data file, then on each stratum separately. Strata are defined by the <code>strata</code> variable, which can be coded as numerical or character. However, one should use only a moderate number of levels/strata, since <code>haplin</code> will be run independetly on each, and some strata may otherwise have an insufficient amount of data. 
Running <code>haplinStrat</code> is thus just a simplification of manually splitting the file into strata and running <code>haplin</code> on each; the end result would be the same.
The main reason for running haplinStrat is to test for gene-environment interactions. This is achieved by running <code>postTest</code> on the result from <code>haplinStrat</code>.
<code>haplinStrat</code> can also be run from within <code>haplinSlide</code> by using the <code>strata</code> argument in <code>haplinSlide</code>.
</p>


<h3>Value</h3>

<p>A list of objects of class haplin is returned. The first element contains the result of running <code>haplin</code> on the entire data file; the remaining elements are the results from each of the strata. The names of the list correspond to the values of the <code>strata</code> variable.
</p>


<h3>Note</h3>

<p>Further information is found on the web page.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+summary.haplin">summary.haplin</a></code>, <code><a href="#topic+plot.haplin">plot.haplin</a></code>, <code><a href="#topic+haptable">haptable</a></code>, <code><a href="#topic+toDataFrame">toDataFrame</a></code>, <code><a href="#topic+haplinSlide">haplinSlide</a></code>, <code><a href="#topic+gxe">gxe</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
# setting up the directory with exemplary data
dir.in &lt;- system.file( "extdata", package = "Haplin" )
file.in &lt;- paste0( dir.in, "/data.dat" )

# reading data in
data.in &lt;- genDataRead( file.in, file.out = "poo_exmpl_data_read", format = "haplin",
  dir.out = tempdir( check = TRUE ), n.vars = 1, allele.sep = " ", col.sep = " ",
  overwrite = TRUE )
# preprocessing the data
data.preproc &lt;- genDataPreprocess( data.in, design = "triad",
  file.out = "poo_exmpl_data_preproc", dir.out = tempdir( check = TRUE ), overwrite = TRUE )

# running haplinStrat, checking for gene-environment interactions
res.GxE &lt;- haplinStrat( data.preproc, markers = 2, strata = 1, poo = FALSE,
  response = "mult", reference = 2, use.missing = TRUE )
res.GxE

# running haplinStrat, checking for POO-environment interactions
res.POOxE &lt;- haplinStrat( data.preproc, markers = c(1,2,3), strata = 1, poo = TRUE,
  response = "mult", reference = "ref.cat", use.missing = TRUE )
res.POOxE

## Not run: 
# All standard haplin runs can be done with haplinStrat. 
# Below is an illustration. See the haplin help page for more 
# examples.
# 
# Analyzing the effect of fetal genes, including triads with missing data,
# using a multiplicative response model. The first column of the data file
# in this example contains the stratification variable.
result &lt;- haplinStrat("C:/work/data.dat", strata = 1, use.missing = T, response = "mult",
reference = "ref.cat", winlength = 1)
# Provide summary of separate results:
lapply(result, summary)
# Plot results separately:
par(ask = T)
lapply(result, plot)
#
# Convert results to table format and stack them over strata:
haptable(result)
# Test for interaction between haplotype risk estimates and the strata variable:
postTest(result)


## End(Not run)

</code></pre>

<hr>
<h2 id='hapPower'>Power simulation for association analyses with Haplin</h2><span id='topic+hapPower'></span>

<h3>Description</h3>

<p>Simulates the statistical power of genetic analyses assessing fetal effects, maternal effects and/or parent-of-origin effects. Effects of X-chromosome genes and gene-environment interaction effects are also allowed. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapPower(hapRun.result, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapPower_+3A_haprun.result">hapRun.result</code></td>
<td>
<p>The result of running <code><a href="#topic+hapRun">hapRun</a></code></p>
</td></tr>
<tr><td><code id="hapPower_+3A_alpha">alpha</code></td>
<td>
<p><code>alpha</code> is the Type I Error probablity. Equals 0.05 by default.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Haplin framework includes different modules for assessing genetic effects: <code><a href="#topic+haplin">haplin</a></code>, 
<code><a href="#topic+haplinStrat">haplinStrat</a></code> and <code><a href="#topic+haplinSlide">haplinSlide</a></code>. <code>hapPower</code> simulates the power of these analyses, which enables power calculations of fetal effects, maternal effects and/or parent-of-origin effects. Various family designs, i.e., triads, case-control, the hybrid design, and all intermediate designs, are possible.
It also allows power calculation of gene-environment interaction effects and effects on X-chromosome markers.
</p>
<p><code>hapPower</code> calculates statistical power using the result of <code><a href="#topic+hapRun">hapRun</a></code>, and the target effects must be specified in this function, see Examples below, and details in <a href="https://haplin.bitbucket.io/docu/Haplin_power.pdf">https://haplin.bitbucket.io/docu/Haplin_power.pdf</a>.
</p>


<h3>Value</h3>

<p><code>hapPower</code> returns the simulated power.</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
</p>
<p><a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+haplinSlide">haplinSlide</a></code>, <code><a href="#topic+hapSim">hapSim</a></code>, <code><a href="#topic+hapRun">hapRun</a></code>, <code><a href="#topic+snpPower">snpPower</a></code>, <code><a href="#topic+snpSampleSize">snpSampleSize</a></code> , <code><a href="#topic+hapPowerAsymp">hapPowerAsymp</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Simulate power from 100 files using haplin. 
## The files consist of fetal effects at two diallelic markers,
## corresponding to haplo.freq = rep(0.25, 4), RR = c(2,1,1,1) and RRstar = c(1,1,1,1).
## The power is simulated for the combination of 100 case triads 
## and 100 control triads with no missing data at a 0.05 significance level,
## applying a multiplicative model.
hapRun.res &lt;- hapRun(nall = c(2,2), n.strata = 1, cases = c(mfc=100), controls = c(mfc=100),
haplo.freq = rep(0.25,4), RR = c(2,1,1,1), RRstar = c(1,1,1,1), 
hapfunc = "haplin", response = "mult", n.sim = 100, dire = "simfiles", ask = FALSE)
hapPower(hapRun.res)

## Simulate power from 100 files applying haplinStrat. 
## The files consist of fetal and maternal effects at two diallelic markers.
## The data is simulated for 500 case triads and 200 control families in the first stratum,
## and 500 case triads and 500 control trids in the second.
## The fetal effects vary across strata,
## whereas the maternal effects are the same.
## One percent of the case triads are missing at random in the second stratum.
hapRun.res &lt;- hapRun(nall = c(2,2), n.strata = 2, cases = c(mfc=500),
controls = list(c(mfc=200),c(mfc=500)), haplo.freq = rep(0.25,4), maternal = TRUE, 
RR = list(c(1.5,1,1,1),c(1,1,1,1)), RRstar = c(1,1,1,1),
RR.mat = c(1.5,1,1,1), RRstar.mat = c(1,1,1,1), gen.missing.cases = list(NULL,0.01), 
use.missing = TRUE, hapfunc = "haplinStrat", n.sim = 100, ask = FALSE)
hapPower(hapRun.res)

## Simulate power at the 0.1 significance level from 1000 files using haplin.
## The files consist of fetal effects at one diallelic locus,
## corresponding to haplo.freq = c(0.1,0.9), RR = c(2,1) and RRstar = c(1,1).
## The data consists of a combination of 100 case triads and 100 control triads.
hapRun.res &lt;- hapRun(nall = c(2), cases = c(mfc=100), controls = c(mfc=100),
haplo.freq = c(0.1,0.9), RR = c(2,1), RRstar = c(1,1),
hapfunc = "haplin", response = "mult", n.sim = 1000, ask = FALSE)
hapPower(hapRun.res, alpha= 0.10)

## The latter example, applying response = "mult", should be comparable to 
## the theoretic calculations of snpPower.
snpPower(cases = list(mfc=100), controls = list(mfc=100),
RR = 2, MAF = 0.1, alpha = 0.10)

## End(Not run)
</code></pre>

<hr>
<h2 id='hapPowerAsymp'>Asymptotic power calculations for genetic association analyses with Haplin</h2><span id='topic+hapPowerAsymp'></span>

<h3>Description</h3>

<p>Computes the asymptotic power for genetic analyses assessing fetal effects, maternal effects and/or parent-of-origin effects. Effects of X-chromosome genes and gene-environment interaction effects are also allowed. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapPowerAsymp(nall = 2, n.strata = 1, cases, controls, haplo.freq,
RR, RRcm, RRcf, RRstar, RR.mat, RRstar.mat,
xchrom = F, sim.comb.sex = "double", BR.girls, 
response = "mult", alpha = 0.05, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapPowerAsymp_+3A_nall">nall</code></td>
<td>
<p>A vector of the number of alleles at each locus. By default a diallellic SNP.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_n.strata">n.strata</code></td>
<td>
<p>The number of strata.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_cases">cases</code></td>
<td>
<p>A list of the number of case families. Each element is a vector of the number of families of the specified family design in the corresponding stratum. The possible family designs, i.e., the possible names of the elements, are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child dyad), <code>"fc"</code> (father-child dyad) or <code>"c"</code> (a single case child).</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_controls">controls</code></td>
<td>
<p>A list of the number of control families. Each element is a vector of the number of families of the specified family design in the corresponding stratum. The possible family designs are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child-dyad), <code>"fc"</code> (father-child dyad), <code>"mf"</code> (mother-father dyad), <code>"c"</code> (a single control child), <code>"m"</code> (a single control mother) or <code>"f"</code> (a single control father).</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_haplo.freq">haplo.freq</code></td>
<td>
<p>A list of which each element is a numeric vector of the haplotype frequencies in each stratum. The frequencies are normalized and sum to one. The Details section shows how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_rr">RR</code></td>
<td>
<p>A list of which each element is a numeric vector of the relative risks in each stratum. The Details section shows how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_rrcm">RRcm</code></td>
<td>
<p>A list of numeric vectors. Each vector contains the relative risks associated with the haplotypes transmitted from the mother for this stratum. See Details for description of how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_rrcf">RRcf</code></td>
<td>
<p>A list of numeric vectors. Each vector contains the relative risks associated with the haplotypes transmitted from the father for this stratum. See Details for description of how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_rrstar">RRstar</code></td>
<td>
<p>A list of numeric vectors. Estimates how much double-dose children would deviate from the risk expected in a multiplicative dose-response relationship.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_rr.mat">RR.mat</code></td>
<td>
<p>The interpretation is similar to <code>RR</code> but for maternal genetic effects.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_rrstar.mat">RRstar.mat</code></td>
<td>
<p>The interpretation is similar to <code>RRstar</code> but for maternal genetic effects.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_xchrom">xchrom</code></td>
<td>
<p>Logical. Equals <code>FALSE</code> by default, which indicates analyses of autosomal markers. If <code>TRUE</code>, <code>hapPowerAsymp</code> performs power analyses of X-linked markers.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_sim.comb.sex">sim.comb.sex</code></td>
<td>
<p>To be used with <code>xchrom = TRUE</code>. A character value that specifies how to handle gender differences on the X-chromosome. If <code>"single"</code>, the effect of a (single) allele in males is equal to the effect of a single allele dose in females, and similarly, if <code>"double"</code>, a single allele in males has the same effect as a double allele dose in females. Default is &quot;double&quot;, which corresponds to X-inactivation.</p>
</td></tr> 
<tr><td><code id="hapPowerAsymp_+3A_br.girls">BR.girls</code></td>
<td>
<p>To be used with <code>xchrom = TRUE</code>. Gives the ratio of baseline risk for females relative to the baseline risk for males.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_response">response</code></td>
<td>
<p>The default value &quot;mult&quot; specifies a multiplicative dose-response model. <code>response = "free"</code> is not yet implemented.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_alpha">alpha</code></td>
<td>
<p><code>alpha</code> is the Type I Error probablity. Equals 0.05 by default.</p>
</td></tr>
<tr><td><code id="hapPowerAsymp_+3A_...">...</code></td>
<td>
<p>Could include argument <code>reference</code>. By default, the most frequent allele or haplotype is chosen as reference. The reference haplotype can be set explicitly by giving a numeric value for the reference argument.</p>
</td></tr> 
</table>


<h3>Details</h3>

<p>The Haplin framework includes different modules for assessing genetic effects: <code><a href="#topic+haplin">haplin</a></code>, 
<code><a href="#topic+haplinStrat">haplinStrat</a></code> and <code><a href="#topic+haplinSlide">haplinSlide</a></code>. <code>hapPowerAsymp</code> computes the asymptotic power for these analyses, which enables power calculations of fetal effects, maternal effects and/or parent-of-origin effects. Various family designs, i.e., triads, case-control, the hybrid design, and all intermediate designs, are possible.
It also allows power calculation of gene-environment interaction effects and effects on X-chromosome markers.
</p>
<p><code><a href="#topic+hapPower">hapPower</a></code> computes power through &quot;brute force&quot; simulations using <code><a href="#topic+hapRun">hapRun</a></code>. This is a robust way of checking software implementations, asymptotic approximations
and attained significance level. However, both power and the corresponding sample size calculations can be performed much more efficiently using asymptotic approximations.
The asymptotic power is calculated applying the non-centrality parameter of the Wald tests, which use the asymptotic normal distribution of the log-scale parameter. The function <code>hapCovar</code> (used by <code>hapPowerAsymp</code>) computes the variance-covariance matrix 
by applying the log-linear model combined with the EM algorithm.
</p>
<p>Specifying haplotype risks:<br />
The number of haplotypes used in the simulations is determined by the <code>nall</code> argument, since <code>prod(nall)</code> different haplotypes can be made from the specified number of markers, <code>length(nall)</code>. The arguments <code>haplo.freq</code>, <code>RR</code>, <code>RRcm</code>, <code>RRcf</code>, <code>RRstar</code>, <code>RR.mat</code>, and <code>RRstar.mat</code> are all lists where each element represents a stratum. Within each stratum, the arguments are vectors of length equal to the number of haplotypes, specifying the relative risk etc. associated with each haplotype.
The stratum specific arguments may be simplified if the number of strata is one, or if the arguments are equal across all strata.
The haplotypes are determined by creating all possible haplotypes from the given markers, in a sequence where the first marker varies mostly quickly. For instance, if <code>nall = c(3,2)</code>, the first marker has 3 alleles, the second has 2, and 6 haplotypes are possible. Taken in order, the haplotypes are 1-1, 2-1, 3-1, 1-2, 2-2, and 3-2. When specifying, say, <code>RR = c(1,2,1,1,1,1)</code> the haplotype 2-1 has a double risk compared to the rest. With, for instance, two strata, the specification <code>RR = list(c(1,2,1,1,1,1), c(1,1,1,1,1,1))</code> would mean that the risk associated with 2-1 is elevated only in the first stratum, not the second.
The simplest example would be with <code>nall = c(2)</code> and <code>RR = c(1,2)</code>, which would simulate a single SNP where the second allele has a double risk.
</p>
<p>Specifying genetic effects:<br />
Standard fetal effects are specified by the arguments <code>RR</code> and <code>RRstar</code>, whereas parent-of-origin effects are addressed by the arguments <code>RRcm</code>, <code>RRcf</code> and <code>RRstar</code>.  Maternal effects are included by the additional arguments <code>RRmat</code> and <code>RRstar.mat</code>.
</p>


<h3>Value</h3>

<p><code>hapPowerAsymp</code> returns the asymptotic power for the relevant genetic effects. The first element of the list depicts the power for each haplotype analyzed separately. 
If there are more than two possible haplotypes, the second element displays the overall power for all haplotypes combined.</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
</p>
<p><a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+haplinSlide">haplinSlide</a></code>,  <code><a href="#topic+haplinStrat">haplinStrat</a></code>, <code><a href="#topic+hapSim">hapSim</a></code>, <code><a href="#topic+hapRun">hapRun</a></code>, <code><a href="#topic+snpPower">snpPower</a></code>, <code><a href="#topic+hapPower">hapPower</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Calculate the asymptotic power for a triad design 
## when the minor allele increases the fetal risk by twofold.
## Assumes a multiplicative dose-response relationship.
hapPowerAsymp(nall = c(2), n.strata = 1, cases = list(c(mfc=120)), 
haplo.freq = c(0.1,0.9), RR = c(2,1), RRstar = c(1,1))

## Calculate the asymptotic power for the hybrid design when 
## the minor allele increases the fetal risk by twofold 
## in the first stratum and no effect is seen in the second
## i.e., gene-environment interaction (GxE) effects.
hapPowerAsymp(nall = c(2), n.strata = 2, cases = list(c(mfc=100)), 
controls = list(c(mfc=100)), haplo.freq = c(0.1,0.9), 
RR = list(c(2,1), c(1,1)), RRstar = c(1,1))

## Calculate the asymptotic GxE power assessing maternal- and
## parent-of-origin effects at two diallelic loci.
hapPowerAsymp(nall = c(2,2), n.strata = 2, cases = c(mfc=500), 
haplo.freq = c(0.1,0.2,0.3,0.4), 
RRcm = list(c(3,1,1,1), c(1,1,1,1)), RRcf = c(1,1,1,1), RRstar = c(1,1,1,1),
RR.mat = list(c(1.5,1,1,1),c(1,1,1,1)), RRstar.mat = c(1,1,1,1))

</code></pre>

<hr>
<h2 id='hapRelEff'>Relative efficiency comparing different study designs in genetic association analysis with Haplin</h2><span id='topic+hapRelEff'></span>

<h3>Description</h3>

<p>Computes the relative efficiency for different study designs in genetic association analysis. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapRelEff(nall = 2, cases.comp, controls.comp, 
cases.ref, controls.ref, haplo.freq,
RR, RRcm, RRcf, RRstar, RR.mat, RRstar.mat,
xchrom = F, sim.comb.sex = "double", BR.girls, 
response = "mult", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapRelEff_+3A_nall">nall</code></td>
<td>
<p>A vector of the number of alleles at each locus. By default a diallellic SNP.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_cases.comp">cases.comp</code></td>
<td>
<p>A list of the number of case families in the comparison design. Its element is a vector of the number of families of the specified family design. The possible family designs, i.e., the possible names of the elements, are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child dyad), <code>"fc"</code> (father-child dyad) or <code>"c"</code> (a single case child).</p>
</td></tr>		
<tr><td><code id="hapRelEff_+3A_controls.comp">controls.comp</code></td>
<td>
<p>A list of the number of control families in the comparison design. Its element is a vector of the number of families of the specified family design. The possible family designs are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child-dyad), <code>"fc"</code> (father-child dyad), <code>"mf"</code> (mother-father dyad), <code>"c"</code> (a single control child), <code>"m"</code> (a single control mother) or <code>"f"</code> (a single control father).</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_cases.ref">cases.ref</code></td>
<td>
<p>A list of the number of case families in the reference design. The options are the same as for <code>cases.comp</code>.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_controls.ref">controls.ref</code></td>
<td>
<p>A list of the number of control families in the reference design. The options are the same as for <code>controls.comp</code>.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_haplo.freq">haplo.freq</code></td>
<td>
<p>A list of which its element is a numeric vector of the haplotype frequencies. The frequencies are normalized and sum to one. The Details section shows how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_rr">RR</code></td>
<td>
<p>A list of which its element is a numeric vector of the relative risks. The Details section shows how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_rrcm">RRcm</code></td>
<td>
<p>A numeric vector in list format containing the relative risks associated with the haplotypes transmitted from the mother. See Details for description of how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_rrcf">RRcf</code></td>
<td>
<p>A numeric vector in list format containing the relative risks associated with the haplotypes transmitted from the father. See Details for description of how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_rrstar">RRstar</code></td>
<td>
<p>A numeric vector in list format. Estimates how much double-dose children would deviate from the risk expected in a multiplicative dose-response relationship.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_rr.mat">RR.mat</code></td>
<td>
<p>The interpretation is similar to <code>RR</code> but for maternal genetic effects.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_rrstar.mat">RRstar.mat</code></td>
<td>
<p>The interpretation is similar to <code>RRstar</code> but for maternal genetic effects.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_xchrom">xchrom</code></td>
<td>
<p>Logical. Equals <code>FALSE</code> by default, which indicates analyses of autosomal markers. If <code>TRUE</code>, analyses are perfomed on X-linked markers.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_sim.comb.sex">sim.comb.sex</code></td>
<td>
<p>To be used with <code>xchrom = TRUE</code>. A character value that specifies how to handle gender differences on the X-chromosome. If <code>"single"</code>, the effect of a (single) allele in males is equal to the effect of a single allele dose in females, and similarly, if <code>"double"</code>, a single allele in males has the same effect as a double allele dose in females. Default is &quot;double&quot;, which corresponds to X-inactivation.</p>
</td></tr> 
<tr><td><code id="hapRelEff_+3A_br.girls">BR.girls</code></td>
<td>
<p>To be used with <code>xchrom = TRUE</code>. Gives the ratio of baseline risk for females relative to the baseline risk for males.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_response">response</code></td>
<td>
<p>The default value &quot;mult&quot; specifies a multiplicative dose-response model. <code>response = "free"</code> is not yet implemented.</p>
</td></tr>
<tr><td><code id="hapRelEff_+3A_...">...</code></td>
<td>
<p>Could include argument <code>reference</code>. By default, the most frequent allele or haplotype is chosen as reference. The reference haplotype can be set explicitly by giving a numeric value for the reference argument.</p>
</td></tr> 
</table>


<h3>Details</h3>

<p><code>hapRelEff</code> compares two study designs for genetic association analysis, using a term called relative efficiency. 
The relative efficiency is defined as the ratio of variances of estimators for the same parameter, computed from two
different designs, or equivalently, the ratio of the sample sizes needed for each of the two designs to achieve the same level and power. The number of genotyped individuals within each design is accounted for.
The relative efficiency estimated under the null hypothesis, i.e., when all relative risks are equal to one, is known as the Pitman efficiency.
To compute the variance-covariance estimates for each design, <code>hapRelEff</code> calls the function <code>hapCovar</code>, which calculates the asymptotic variance-covariance matrix 
by applying the log-linear model combined with the EM algorithm.
</p>
<p>The relative efficiency can be computed for fetal effects, maternal effects and parent-of-origin effect, 
as well as effects on X-chromosome markers. 
Various study designs, i.e., case-parent triads, the standard case-control design, the hybrid design, and all intermediate designs, can be compared.
</p>
<p>Note that the exact number of case families and control families in the reference or comparison design is irrelevant, as this will be accounted for in the relative efficiency estimate.
However, the ratio of control families to case families within the reference or comparison design must be specified correctly. See the Examples section.
</p>
<p>Specifying haplotype risks:<br />
The number of haplotypes used in the simulations is determined by the <code>nall</code> argument, since <code>prod(nall)</code> different haplotypes can be made from the specified number of markers, <code>length(nall)</code>. 
The arguments <code>haplo.freq</code>, <code>RR</code>, <code>RRcm</code>, <code>RRcf</code>, <code>RRstar</code>, <code>RR.mat</code>, and <code>RRstar.mat</code> are all lists, containing vectors of length equal to the number of haplotypes, specifying the relative risk etc. associated with each haplotype.
However, the function will work without using the list format.
The haplotypes are determined by creating all possible haplotypes from the given markers, in a sequence where the first marker varies mostly quickly. 
For instance, if <code>nall = c(3,2)</code>, the first marker has 3 alleles, the second has 2, and 6 haplotypes are possible. Taken in order, the haplotypes are 1-1, 2-1, 3-1, 1-2, 2-2, and 3-2. 
When specifying, say, <code>RR = c(1,2,1,1,1,1)</code> the haplotype 2-1 has a double risk compared to the rest. 
The simplest example would be with <code>nall = c(2)</code> and <code>RR = c(1,2)</code>, which would simulate a single SNP where the second allele has a double risk.
</p>
<p>Specifying genetic effects:<br />
Standard fetal effects are specified by the arguments <code>RR</code> and <code>RRstar</code>, 
whereas parent-of-origin effects are addressed by the arguments <code>RRcm</code>, <code>RRcf</code> and <code>RRstar</code>. 
Maternal effects are included by the additional arguments <code>RRmat</code> and <code>RRstar.mat</code>.
</p>


<h3>Value</h3>

<p><code>hapRelEff</code> returns the relative efficiency estimate, comparing two study designs. The first element of the list depicts the relative efficiency for each haplotype analyzed separately. 
If there are more than two possible haplotypes, the second element displays the overall relative efficiency for all haplotypes combined.</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
</p>
<p><a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+hapPowerAsymp">hapPowerAsymp</a></code>, <code><a href="#topic+hapRun">hapRun</a></code>, <code><a href="#topic+snpPower">snpPower</a></code>, <code><a href="#topic+hapPower">hapPower</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Child effects: Calculate the efficiency of the standard case-control design 
## (with an equal number of case and control children) 
## relative to the case-parent triad design
## under the null hypothesis when the minor allele frequency is 0.1.
hapRelEff(nall = c(2), cases.comp = c(c=1), 
controls.comp = c(c=1), cases.ref = c(mfc=1),
haplo.freq = c(0.1,0.9), RR = c(1,1))
	
## Child effects: Calculate the efficiency of the standard case-control design, 
## with twice as many cases as controls, relative to the case-parent triad design
## under the null hypothesis when the minor allele frequency is 0.2.
hapRelEff(nall = c(2), cases.comp = c(c=2), 
controls.comp = c(c=1), cases.ref = c(mfc=1),
haplo.freq = c(0.2,0.8), RR = c(1,1))

## Child and maternal effects: Calculate the efficiency of the case-mother dyad design 
## relative to the case-parent triad design
## under the null hypothesis when the minor allele frequency is 0.1.
hapRelEff(nall = c(2), cases.comp = c(mc=1), cases.ref = c(mfc=1),
haplo.freq = c(0.1,0.9), RR = c(1,1), RR.mat=c(1,1))

## PoO effects: Calculate the efficiency of the full hybrid design, 
## with twice as many control families as case families, 
## relative to the case-parent triad design
## under the null hypothesis when the minor allele frequency is 0.1.
hapRelEff(nall = c(2), cases.comp = c(mfc=1), 
controls.comp = c(mfc=2), cases.ref = c(mfc=1),
haplo.freq = c(0.1,0.9), RRcm = c(1,1), RRcf = c(1,1))

</code></pre>

<hr>
<h2 id='hapRun'>Simulates genetic data and runs Haplin for each simulation</h2><span id='topic+hapRun'></span>

<h3>Description</h3>

<p>Calculates Haplin results by first simulating genetic data, allowing a various number of family designs, and then running Haplin on the simulations.
The simulated data may contain of fetal effects, maternal effects and/or parent-of-origin effects.
The function allows for simulations and calculations on both autosomal and X-chromosome markers, 
assuming Hardy-Weinberg equilibrium.
It enables simulation and calculation of gene-environment interaction effects, i.e, the input (relative risks, number of cases etc.) may vary across strata.
<code>hapRun</code> calls <code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+haplinStrat">haplinStrat</a></code> or <code><a href="#topic+haplinSlide">haplinSlide</a></code> to run on the simulated data files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapRun(nall, n.strata= 1, cases, controls, haplo.freq,
RR, RRcm, RRcf, RRstar, RR.mat, RRstar.mat, hapfunc = "haplin",
gen.missing.cases = NULL, gen.missing.controls = NULL, 
n.sim = 1000, xchrom = FALSE, sim.comb.sex = "double", BR.girls, dire,
ask = TRUE, cpus = 1, slaveOutfile = "", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapRun_+3A_nall">nall</code></td>
<td>
<p>A vector of the number of alleles at each locus.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_n.strata">n.strata</code></td>
<td>
<p>The number of strata.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_cases">cases</code></td>
<td>
<p>A list of the number of case families. Each element is a vector of the number of families of the specified family design(s) in the corresponding stratum. The possible family designs, i.e., the possible names of the elements, are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child dyad), <code>"fc"</code> (father-child dyad) or <code>"c"</code> (a single case child). See Details for a thorough description.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_controls">controls</code></td>
<td>
<p>A list of the number of control families. Each element is a vector of the number of families of the specified family design(s) in the corresponding stratum. The possible family designs are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child-dyad), <code>"fc"</code> (father-child dyad), <code>"mf"</code> (mother-father dyad), <code>"c"</code> (a single control child), <code>"m"</code> (a single control mother) or <code>"f"</code> (a single control father). See Details for a thorough description.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_haplo.freq">haplo.freq</code></td>
<td>
<p>A list of which each element is a numeric vector of the haplotype frequencies in each stratum. The frequencies are normalized and sum to one. The Details section shows how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_rr">RR</code></td>
<td>
<p>A list of which each element is a numeric vector of the relative risks in each stratum. The Details section shows how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_rrcm">RRcm</code></td>
<td>
<p>A list of numeric vectors. Each vector contains the relative risks associated with the haplotypes transmitted from the mother for this stratum. See Details for description of how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_rrcf">RRcf</code></td>
<td>
<p>A list of numeric vectors. Each vector contains the relative risks associated with the haplotypes transmitted from the father for this stratum. See Details for description of how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_rrstar">RRstar</code></td>
<td>
<p>A list of numeric vectors. Estimates how much double-dose children would deviate from the risk expected in a multiplicative dose-response relationship.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_rr.mat">RR.mat</code></td>
<td>
<p>The interpretation is similar to <code>RR</code> when simulating genetic data with maternal effects.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_rrstar.mat">RRstar.mat</code></td>
<td>
<p>The interpretation is similar to <code>RRstar</code> when simulating genetic data with maternal effects.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_hapfunc">hapfunc</code></td>
<td>
<p>Defines which haplin function to run, the options being <code>"haplin"</code>, <code>"haplinSlide"</code> or <code>"haplinStrat"</code>. <code>"haplinSlide"</code> is however only partially implemented.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_gen.missing.cases">gen.missing.cases</code></td>
<td>
<p>Generates missing values at random for the case families. Set to <code>NULL</code> by default, i.e., no missing values generated. See Details for description of how to implement this argument.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_gen.missing.controls">gen.missing.controls</code></td>
<td>
<p>Generates missing values at random for the control families. Set to <code>NULL</code> by default, i.e., no missing values generated. See Details for description of how to implement this argument.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_n.sim">n.sim</code></td>
<td>
<p>The number of simulations, i.e., the number of simulated data files.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_xchrom">xchrom</code></td>
<td>
<p>Logical. Equals <code>FALSE</code> by default, which indicates simulation of autosomal markers. If <code>TRUE</code>, hapSim simulates X-linked genes.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_sim.comb.sex">sim.comb.sex</code></td>
<td>
<p>To be used with <code>xchrom = TRUE</code>. A character value that specifies how to handle gender differences on the X-chromosome. If <code>"single"</code>, the effect of a (single) allele in males is equal to the effect of a single allele dose in females, and similarly, if <code>"double"</code>, a single allele in males has the same effect as a double allele dose in females. Default is &quot;double&quot;, which corresponds to X-inactivation.</p>
</td></tr> 
<tr><td><code id="hapRun_+3A_br.girls">BR.girls</code></td>
<td>
<p>To be used with <code>xchrom = TRUE</code>. Gives the ratio of baseline risk for females to the baseline risk for males.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_dire">dire</code></td>
<td>
<p>Gives the directory of the simulated data files. Missing by default, which means that none of the files are saved to files.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_ask">ask</code></td>
<td>
<p>Logical. If <code>TRUE</code>, hapSim will ask before overwriting the files in an already existing directory.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_cpus">cpus</code></td>
<td>
<p>Allows parallel processing of its analyses. The <code>cpus</code> argument should preferably be set to the number of available CPUs. If set lower, it will save some capacity for other processes to run. Setting it too high should not cause any serious problems.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_slaveoutfile">slaveOutfile</code></td>
<td>
<p>Character. If <code>slaveOutfile = ""</code> (default), output from all running cores will be printed in the standard R session window. Alternatively, the output can be saved to a file by specifying the file path and name.</p>
</td></tr>
<tr><td><code id="hapRun_+3A_...">...</code></td>
<td>
<p>Arguments to be used by <code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+haplinSlide">haplinSlide</a></code> or <code><a href="#topic+haplinStrat">haplinStrat</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>hapRun</code> applies <code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+haplinSlide">haplinSlide</a></code> or <code><a href="#topic+haplinStrat">haplinStrat</a></code>
on each data file simulated by <code><a href="#topic+hapSim">hapSim</a></code>. 
It provides simulations on various family designs, i.e., triads, case-control, the hybrid design, and all intermediate designs.
The simulated files may accomodate fetal effects, maternal effects and/or parent-of-origin effects.
<code>hapRun</code> allows simulation of both autosomal and X-chromosome markers, assuming Hardy-Weinberg equilibrium.
It also enables simulation and calculation of gene-environment interaction effects.
</p>
<p>Details on how to implement the arguments listed above are provided by <code><a href="#topic+hapSim">hapSim</a></code> and the Examples section below.
The stratum specific arguments may be simplified if the number of strata is one, or if the arguments are equal across all strata.
</p>
<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+haplinStrat">haplinStrat</a></code> and <code><a href="#topic+haplinSlide">haplinSlide</a></code> will run with default values unless otherwise specified by <code>hapRun</code>.
For example, if <code>hapfunc = "haplin"</code>, <code><a href="#topic+haplin">haplin</a></code> will use <code>response = "free"</code> unless <code>response = "mult"</code> is explicitly given as an argument in <code>hapRun</code>. 
Moreover, triads with missing data are only included in the haplin analysis if the argument <code>use.missing</code> equals <code>TRUE</code> (default in <code>hapRun</code>). Please confer <a href="https://haplin.bitbucket.io/docu/Haplin_power.pdf">https://haplin.bitbucket.io/docu/Haplin_power.pdf</a> for further details and examples.
</p>
<p>For information on the arguments to be passed on to <code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+haplinStrat">haplinStrat</a></code> and <code><a href="#topic+haplinSlide">haplinSlide</a></code>, please consult their help pages.
</p>
<p>Note that <code>RR.mat</code> and <code>RRstar.mat</code> and <code>RRcm</code> and <code>RRcf</code> are required for <code><a href="#topic+hapSim">hapSim</a></code> to simulate maternal and parent-of-origin effects, respectively.
To calculate these effects, however, arguments <code>maternal = TRUE</code> and/or <code>poo = TRUE</code> must be specified.
</p>
<p><code>gen.missing.cases</code> and <code>gen.missing.controls</code> are flexible arguments. By default, both equal NULL, which means that no missing data are generated at random.
If the arguments are single numbers, missing data are generated at random with this proportion for all cases and/or controls. 
If the arguments are vectors of length equal to the number of loci, missing data are generated with the corresponding proportion for each locus.
The arguments can also be matrices with the number of rows equal to the number of loci and three columns. 
Each row corresponds to a locus, and the columns correspond to mothers, fathers and children, respectively. 
</p>


<h3>Value</h3>

<p>If <code>hapfunc = "haplin"</code>, <code>hapRun</code> returns a dataframe consisting of results from running <code><a href="#topic+haplin">haplin</a></code> on each simulated file.
The first two columns are:
</p>
<table role = "presentation">
<tr><td><code>sim.no</code></td>
<td>
<p>The name of the directory from which the results are calculated, i.e., the simulation number</p>
</td></tr>
<tr><td><code>row.no</code></td>
<td>
<p>The row number within each simulation</p>
</td></tr>
</table>
<p><code><a href="#topic+haptable">haptable</a></code> gives detailed information of the full dataframe. 
</p>
<p>If <code>hapfunc = "haplinSlide"</code>, <code>hapRun</code> returns a list of which each element contains the results from a single run of <code><a href="#topic+haplinSlide">haplinSlide</a></code>.
Consult <code><a href="#topic+suest">suest</a></code> for a thorough description of the output. Note, however, that <code>hapfunc = "haplinSlide"</code> is currently only implemented for diallelic markers, and the reference category is always chosen to be the first haplotype (see <code><a href="#topic+hapSim">hapSim</a></code> for a description of the haplotype grid). 
</p>
<p>If <code>hapfunc = "haplinStrat"</code>, <code><a href="#topic+haplinStrat">haplinStrat</a></code> is used to estimate gene-effects in each stratum of the exposure covariate, and the results from all strata are compared using <code><a href="#topic+gxe">gxe</a></code>.  <code>hapRun</code> returns a list, where each element is the result of a single run of 
<code><a href="#topic+gxe">gxe</a></code>.
</p>
<p>Additionaly, if <code>dire</code> is not missing by default, the simulated files from which the Haplin results are calculated, are stored in the given directory.</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
</p>
<p><a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+haplinSlide">haplinSlide</a></code>, <code><a href="#topic+hapSim">hapSim</a></code>, <code><a href="#topic+haptable">haptable</a></code>, <code><a href="#topic+suest">suest</a></code>, <code><a href="#topic+hapPower">hapPower</a></code>, <code><a href="#topic+hapPowerAsymp">hapPowerAsymp</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Simulate Haplin results from 100 files using the multiplicative model in haplin. 
## The files consist of fetal effects at two diallelic markers,
## corresponding to haplo.freq = rep(0.25, 4), RR = c(2,1,1,1) 
## and RRstar = c(1,1,1,1). That is, the first allele has a doubled risk 
## relative to the rest. The data consists of a combination of 
## 100 case triads and 100 control triads with no missing data.
## No environmental factors are considered, i.e. the number of strata is one.
hapRun(nall = c(2,2), n.strata = 1, cases = c(mfc=100), controls = c(mfc=100),
haplo.freq = rep(0.25,4), RR = c(2,1,1,1), RRstar = c(1,1,1,1), 
hapfunc = "haplin", response = "mult", n.sim = 100, dire = "simfiles", ask = FALSE)

## Simulate power from 100 files applying haplinStrat. 
## The files consist of fetal and maternal effects at two diallelic markers.
## The data is simulated for 500 case triads and 200 control families in the first stratum,
## and 500 case triads and 500 control trids in the second.
## The fetal effects vary across strata,
## whereas the maternal effects are the same.
## One percent of the case triads are missing at random in the second stratum.
hapRun(nall = c(2,2), n.strata = 2, cases = c(mfc=500),
controls = list(c(mfc=200),c(mfc=500)), haplo.freq = rep(0.25,4), maternal = TRUE, 
RR = list(c(1.5,1,1,1),c(1,1,1,1)), RRstar = c(1,1,1,1),
RR.mat = c(1.5,1,1,1), RRstar.mat = c(1,1,1,1), 
gen.missing.cases = list(NULL,0.01), use.missing = TRUE, hapfunc = "haplinStrat", 
n.sim = 100, ask = FALSE)

## Simulate Haplin results from 100 files using haplin. 
## The files consist of fetal effects at one diallelic locus, 
## corresponding to haplo.freq = rep(0.5,2), RR = c(1.5,1) and RRstar = c(1,1).
## We have a combination of 100 case triads and 
## 100 control triads with no missing data. 
## No environmental effects are considered.
hapRun(nall = c(2), n.strata = 1, cases = c(mfc=100), controls = c(mfc=100),
haplo.freq = rep(0.5,2), RR = c(1.5,1), RRstar = c(1,1),
hapfunc = "haplin", n.sim = 100, dire = "simfiles", ask = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='hapSim'>Simulation of genetic data in Haplin format</h2><span id='topic+hapSim'></span>

<h3>Description</h3>

<p>Simulates genetic data in Haplin format, consisting of fetal effects, maternal effects and/or parent-of-origin effects. 
Allows for simulation of both autosomal and X-linked markers, assuming Hardy-Weinberg equilibrium.
Enables stratified simulations for gene-environment interaction analyses, i.e the input (relative risks, number of cases etc) may vary across different strata.</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapSim(nall, n.strata = 1, cases, controls, haplo.freq,
RR, RRcm, RRcf, RRstar, RR.mat, RRstar.mat, 
gen.missing.cases = NULL, gen.missing.controls = NULL,
n.sim = 1000, xchrom = F, sim.comb.sex = "double", BR.girls, dire = "simfiles", 
ask = TRUE, verbose = TRUE, cpus = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hapSim_+3A_nall">nall</code></td>
<td>
<p>A vector of the number of alleles at each locus.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_n.strata">n.strata</code></td>
<td>
<p>The number of strata.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_cases">cases</code></td>
<td>
<p>A list of the number of case families. Each element is a vector of the number of families of the specified family design(s) in the corresponding stratum. The possible family designs, i.e., the possible names of the elements, are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child dyad), <code>"fc"</code> (father-child dyad) or <code>"c"</code> (a single case child). See Details for a thorough description.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_controls">controls</code></td>
<td>
<p>A list of the number of control families. Each element is a vector of the number of families of the specified family design(s) in the corresponding stratum. The possible family designs are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child-dyad), <code>"fc"</code> (father-child dyad), <code>"mf"</code> (mother-father dyad), <code>"c"</code> (a single control child), <code>"m"</code> (a single control mother) or <code>"f"</code> (a single control father). See Details for a thorough description.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_haplo.freq">haplo.freq</code></td>
<td>
<p>A list of which each element is a numeric vector of the haplotype frequencies in each stratum. The frequencies will be normalized so that they sum to one. The Details section shows how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_rr">RR</code></td>
<td>
<p>A list of which each element is a numeric vector of the relative risks in each stratum. The Details section shows how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_rrcm">RRcm</code></td>
<td>
<p>A list of numeric vectors. Each vector contains the relative risks associated with the haplotypes transmitted from the mother for this stratum. See Details for description of how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_rrcf">RRcf</code></td>
<td>
<p>A list of numeric vectors. Each vector contains the relative risks associated with the haplotypes transmitted from the father for this stratum. See Details for description of how to implement this argument in agreement with the possible haplotypes.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_rrstar">RRstar</code></td>
<td>
<p>A list of numeric vectors. Estimates how much double-dose children would deviate from the risk expected in a multiplicative dose-response relationship.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_rr.mat">RR.mat</code></td>
<td>
<p>The interpretation is similar to <code>RR</code> when simulating genetic data with maternal effects.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_rrstar.mat">RRstar.mat</code></td>
<td>
<p>The interpretation is parallel to <code>RRstar</code> when simulating genetic data with maternal effects.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_gen.missing.cases">gen.missing.cases</code></td>
<td>
<p>Generates missing values at random for the case families. Set to <code>NULL</code> by default, i.e., no missing values generated. See Details for description of how to implement this argument.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_gen.missing.controls">gen.missing.controls</code></td>
<td>
<p>Generates missing values at random for the control families. Set to <code>NULL</code> by default, i.e., no missing values generated. See Details for description of how to implement this argument.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_n.sim">n.sim</code></td>
<td>
<p>The number of simulations, i.e., the number of simulated data files.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_xchrom">xchrom</code></td>
<td>
<p>Logical. Equals <code>FALSE</code> by default, which indicates simulation of autosomal markers. If <code>TRUE</code>, <code>hapSim</code> simulates X-linked genes.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_sim.comb.sex">sim.comb.sex</code></td>
<td>
<p>To be used with <code>xchrom = TRUE</code>. A character value which specifies how to handle gender differences on the X-chromosome. If <code>"single"</code>, the effect of a (single) allele in males is equal to the effect of a single allele dose in females, and similarly, if <code>"double"</code>, a single allele in males has the same effect as a double allele dose in females. Default is <code>"double"</code>, which corresponds to X-inactivation.</p>
</td></tr> 
<tr><td><code id="hapSim_+3A_br.girls">BR.girls</code></td>
<td>
<p>To be used with <code>xchrom = TRUE</code>. Gives the ratio of baseline risk for females to the baseline risk for males.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_dire">dire</code></td>
<td>
<p>Gives the directory of the simulated data files.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_ask">ask</code></td>
<td>
<p>Logical. If <code>TRUE</code>, <code>hapSim</code> will ask before overwriting the files in an already existing directory.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_verbose">verbose</code></td>
<td>
<p>Logical. Default is <code>TRUE</code>, which means that the file name is displayed for each iteration. Works only when <code>cpus = 1</code>.</p>
</td></tr>
<tr><td><code id="hapSim_+3A_cpus">cpus</code></td>
<td>
<p>Allows simulations to be performed in parallel. The <code>cpus</code> argument should preferably be set to the number of available cores. If set lower, it will save some capacity for other processes to run. Setting it too high should not cause any serious problems.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>hapSim</code> simulates allele values for case and control families at multiple markers (typically in LD) simultaneously. The number of markers/SNPs involved will typically be in the range 1 to 6. Data are simulated to produce relative risks of disease as specified by the user input. Simulations can be performed separately over a number of strata so as to simulate gene-environment interactions.
</p>
<p>Specifying haplotype risks:<br />
The number of haplotypes used in the simulations is determined by the <code>nall</code> argument, since <code>prod(nall)</code> different haplotypes can be made from the specified number of markers, <code>length(nall)</code>. The arguments <code>haplo.freq</code>, <code>RR</code>, <code>RRcm</code>, <code>RRcf</code>, <code>RRstar</code>, <code>RR.mat</code>, and <code>RRstar.mat</code> are all lists where each element represents a stratum. Within each stratum, the arguments are vectors of length equal to the number of haplotypes, specifying the relative risk etc. associated with each haplotype.
The stratum specific arguments may be simplified if the number of strata is one, or if the arguments are equal across all strata.
</p>
<p>The haplotypes are determined by creating all possible haplotypes from the given markers, in a sequence where the first marker varies mostly quickly. For instance, if <code>nall = c(3,2)</code>, the first marker has 3 alleles, the second has 2, and 6 haplotypes are possible. Taken in order, the haplotypes are 1-1, 2-1, 3-1, 1-2, 2-2, and 3-2. When specifying, say, <code>RR = c(1,2,1,1,1,1)</code> the haplotype 2-1 has a double risk compared to the rest. With, for instance, two strata, the specification <code>RR = list(c(1,2,1,1,1,1), c(1,1,1,1,1,1))</code> would mean that the risk associated with 2-1 is elevated only in the first stratum, not the second. 
</p>
<p>The simplest example would be with <code>nall = c(2)</code> and <code>RR = c(1,2)</code>, which would simulate a single SNP where the second allele has a double risk.
</p>
<p>Output file format:<br />
The format of the simulated files is relatively flexible and allows multi-allelic markers and various designs.
If both case and control families are present, the simulated files contain a leading column of the case/control status (1/0). 
If <code>xchrom=TRUE</code>, the neighboring column to the left of the genetic data contains the sex information (1 = male, 2 = female). 
Each line represents genotypes for a case or control triad. 
</p>
<p>There are six columns for each locus, two for the mother (M), two for the father (F) and two for the child (C). The columns are placed in the following sequence: M11 M12 F11 F12 C11 C12 M21 M22 F21 F22 C21 C22... etc, 
where the first number indicates marker, and the second number indicates the first or second allele at this locus. Columns are separated by a single white space, and missing data are coded as NA.
</p>
<p>Intermediate designs, for instance mother-child dyads, are represented as full triads with columns of absent family members set to missing. In the case of a pure case-control design, however, each line represents a single individual, and there are no columns representing mothers and fathers.<br />
</p>
<p>There are no row or column names in the files.
</p>
<p>Some examples are given below. See <a href="https://haplin.bitbucket.io/docu/haplin_data_format.html">https://haplin.bitbucket.io/docu/haplin_data_format.html</a> for a thorough description of the Haplin format. Note that this description separates the two alleles for an individual within a locus by a semi-colon, such as 1;2. This is, however, not necessary.<br />
Confer the document <a href="https://haplin.bitbucket.io/docu/Haplin_power.pdf">https://haplin.bitbucket.io/docu/Haplin_power.pdf</a> for details and examples on how to perform the simulations.
</p>
<p><code>gen.missing.cases</code> and <code>gen.missing.controls</code> are flexible arguments. By default, both equal NULL, which means that no missing data are generated at random.
If the arguments are single numbers, missing data are generated at random with this proportion for all cases and/or controls. 
If the arguments are vectors of length equal to the number of loci, missing data are generated with the corresponding proportion for each locus.
The arguments can also be matrices with the number of rows equal to the number of loci and three columns. 
Each row corresponds to a locus, and the columns correspond to mothers, fathers and children, respectively.</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
</p>
<p><a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+hapRun">hapRun</a></code>, <code><a href="#topic+hapPower">hapPower</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Simulate genetic data (100 files) at two diallelic markers, consisting of fetal effects 
## corresponding to haplo.freq = rep(0.25, 4), RR = c(2,1,1,1) and RRstar = c(1,1,1,1),
## for the combination of 1000 case triads and 1000 control triads with no missing data.
## Only one stratum.
hapSim(nall = c(2,2), n.strata = 1, cases = c(mfc=1000),
controls = c(mfc=1000), haplo.freq = rep(0.25,4),
RR = c(2,1,1,1), RRstar = c(1,1,1,1), n.sim = 100, dire = "simfiles")

## Simulate genetic data (100 files) at two diallelic markers,
## consisting of fetal and maternal effects corresponding to 
## haplo.freq = rep(0.25, 4), RR = c(2,1,1,1), RRstar = c(1,1,1,1),
## RR.mat = c(2,1,1,1) and RRstar.mat = c(1,1,1,1),
## for 1000 case triads and zero control families. 
## One percent of the case triads are missing at random. One stratum only.
hapSim(nall = c(2,2), n.strata=1, cases = c(mfc=1000),
controls = c(mfc=0), haplo.freq = rep(0.25,4), RR = c(2,1,1,1),
RRstar = c(1,1,1,1), RR.mat = c(2,1,1,1), RRstar.mat = c(1,1,1,1), 
gen.missing.cases = 0.01, n.sim = 100, dire = "simfiles")

## Simulate genetic data (100 files) at two diallelic markers. In the first stratum, 
## we have a combination of 500 case triads and 500 control triads with
## haplo.freq = rep(0.25, 4), RR = c(2,1,1,1) and RRstar = c(1,1,1,1).
## In the second stratum, we have 300 case triads and 500 control triads with
## haplo.freq = rep(0.25, 4), RR = c(1,1,1,1) and RRstar = c(1,1,1,1).
## One percent of the control triads are missing at random in the first stratum.
hapSim(nall = c(2,2), n.strata= 2, cases = list(c(mfc=500),c(mfc=300)), 
controls = c(mfc=500),haplo.freq = rep(0.25,4),
RR = list(c(2,1,1,1),c(1,1,1,1)), RRstar = c(1,1,1,1),
gen.missing.controls = list(0.01,NULL), n.sim = 100, dire = "simfiles")

## End(Not run)
</code></pre>

<hr>
<h2 id='haptable'>Create haplin table </h2><span id='topic+haptable'></span>

<h3>Description</h3>

<p>Create a comprehensive table of haplin output
</p>


<h3>Usage</h3>

<pre><code class='language-R'> haptable(object) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haptable_+3A_object">object</code></td>
<td>
<p>A <code>haplin</code> object, i.e. the result of running <code>haplin</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>haptable</code> extracts the most important information from a <code>haplin</code> object to produce a summary table. The table can then be saved with, for instance, <code>write.table</code>, making the results easily accessible to other applications. You can also use <code>output</code> to produce and save the same results.
</p>


<h3>Value</h3>

<p>&mdash;-
A dataframe is returned, with the following columns:
&mdash;-
</p>
<table role = "presentation">
<tr><td><code>marker</code></td>
<td>
<p>Name(s) of marker(s) investigated</p>
</td></tr>
<tr><td><code>alleles</code></td>
<td>
<p>A listing of the alleles found at each marker</p>
</td></tr>
<tr><td><code>counts</code></td>
<td>
<p>Frequency counts of alleles at each marker</p>
</td></tr>
<tr><td><code>HWE.pv</code></td>
<td>
<p>P-value from Hardy-Weinberg equilibrium test at each marker</p>
</td></tr>
<tr><td><code>Original</code></td>
<td>
<p>Number of triads before removal</p>
</td></tr>
<tr><td><code>After.rem.NA</code></td>
<td>
<p>Number of triads after removal of missing</p>
</td></tr>
<tr><td><code>After.rem.Mend.inc.</code></td>
<td>
<p>Number of triads after removal of Mendelian inconsistencies</p>
</td></tr>
<tr><td><code>After.rem.unused.haplos</code></td>
<td>
<p>Number of triads after removal of unused (rare) haplotypes</p>
</td></tr>
<tr><td><code>pv.overall</code></td>
<td>
<p>Overall likelihood ratio p-value (test of all genetic effects combined)</p>
</td></tr>
<tr><td><code>haplos</code></td>
<td>
<p>Haplotypes (or single-marker alleles) found during estimation</p>
</td></tr>
<tr><td><code>haplofreq</code></td>
<td>
<p>Estimated haplotype frequencies</p>
</td></tr>
<tr><td><code>haplofreq.lower</code></td>
<td>
<p>Lower 95% CI for estimated haplotype frequencies</p>
</td></tr>
<tr><td><code>haplofreq.upper</code></td>
<td>
<p>Upper 95% CI for estimated haplotype frequencies</p>
</td></tr>
<tr><td><code>reference</code></td>
<td>
<p>Reference method. If ref.cat is used, the reference category is labeled &quot;ref&quot;</p>
</td></tr>
<tr><td><code>RR.est.</code></td>
<td>
<p>Estimated single dose relative risk</p>
</td></tr>
<tr><td><code>RR.lower</code></td>
<td>
<p>Lower 95% CI for single dose relative risk</p>
</td></tr>
<tr><td><code>RR.upper</code></td>
<td>
<p>Upper 95% CI for single dose relative risk</p>
</td></tr>
<tr><td><code>RR.p.value</code></td>
<td>
<p>P-values for individual single dose effect</p>
</td></tr>
<tr><td><code>RRdd.est.</code></td>
<td>
<p>Estimated double dose relative risk</p>
</td></tr>
<tr><td><code>RRdd.lower</code></td>
<td>
<p>Lower 95% CI for double dose relative risk</p>
</td></tr>
<tr><td><code>RRdd.upper</code></td>
<td>
<p>Upper 95% CI for double dose relative risk</p>
</td></tr>
<tr><td><code>RRdd.p.value</code></td>
<td>
<p>P-values for individual double dose effect</p>
</td></tr>
<tr><td><code>NOTE1</code></td>
<td>
<p>When maternal = TRUE, there will be additional columns:</p>
</td></tr>
<tr><td><code>RRm.est.</code></td>
<td>
<p>Estimated single dose relative risk for maternal haplotype</p>
</td></tr>
<tr><td><code>RRm.lower</code></td>
<td>
<p>Lower 95% CI for single dose relative risk for maternal haplotype</p>
</td></tr>
<tr><td><code>RRm.upper</code></td>
<td>
<p>Upper 95% CI for single dose relative risk for maternal haplotype</p>
</td></tr>
<tr><td><code>RRm.p.value</code></td>
<td>
<p>P-values for individual single dose effect of maternal haplotype</p>
</td></tr>
<tr><td><code>RRmdd.est.</code></td>
<td>
<p>Estimated double dose relative risk for maternal haplotype</p>
</td></tr>
<tr><td><code>RRmdd.lower</code></td>
<td>
<p>Lower 95% CI for double dose relative risk for maternal haplotype</p>
</td></tr>
<tr><td><code>RRmdd.upper</code></td>
<td>
<p>Upper 95% CI for double dose relative risk for maternal haplotype</p>
</td></tr>
<tr><td><code>RRmdd.p.value</code></td>
<td>
<p>P-values for individual double dose effect of maternal haplotype</p>
</td></tr>
<tr><td><code>NOTE2</code></td>
<td>
<p>When poo = TRUE, the RR.est., RR.lower, and RR.upper columns will be replaced by the following columns:</p>
</td></tr>
<tr><td><code>RRcm.est.</code></td>
<td>
<p>Estimated single dose relative risk, when inherited from the mother</p>
</td></tr>
<tr><td><code>RRcm.lower</code></td>
<td>
<p>Lower 95% CI for single dose relative risk, when inherited from the mother</p>
</td></tr>
<tr><td><code>RRcm.upper</code></td>
<td>
<p>Upper 95% CI for single dose relative risk, when inherited from the mother</p>
</td></tr>
<tr><td><code>RRcm.p.value</code></td>
<td>
<p>P-values for individual single dose effects, when inherited from the mother</p>
</td></tr>
<tr><td><code>RRcf.est.</code></td>
<td>
<p>Estimated single dose relative risk, when inherited from the father</p>
</td></tr>
<tr><td><code>RRcf.lower</code></td>
<td>
<p>Lower 95% CI for single dose relative risk, when inherited from the father</p>
</td></tr>
<tr><td><code>RRcf.upper</code></td>
<td>
<p>Upper 95% CI for single dose relative risk, when inherited from the father</p>
</td></tr>
<tr><td><code>RRcf.p.value</code></td>
<td>
<p>P-values for individual single dose effects, when inherited from the father</p>
</td></tr>
<tr><td><code>RRcm_RRcf.est.</code></td>
<td>
<p>An estimate of parent-of-origin effect, i.e. the ratio RRcm/RRcf</p>
</td></tr>
<tr><td><code>RRcm_RRcf.lower</code></td>
<td>
<p>Lower 95% CI for ratio RRcm/RRcf</p>
</td></tr>
<tr><td><code>RRcm_RRcf.upper</code></td>
<td>
<p>Upper 95% CI for ratio RRcm/RRcf</p>
</td></tr>
<tr><td><code>RRcm_RRcf.p.value</code></td>
<td>
<p>P-value for parent-of-origin effect RRcm/RRcf at that marker</p>
</td></tr>	   
</table>


<h3>Note</h3>

<p>Further information is found on the web page
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+output">output</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Produce a table containing the most important output from haplin:
res &lt;- haplin("data.dat", use.missing = T, maternal = T)
haptable(res)

## End(Not run)</code></pre>

<hr>
<h2 id='initParallelRun'>Initialization of the Rmpi cluster</h2><span id='topic+initParallelRun'></span>

<h3>Description</h3>

<p>This function prepares a cluster using <code>Rmpi</code> package. The initialization
is paired with closing of the cluster using the <code><a href="#topic+finishParallelRun">finishParallelRun</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initParallelRun(cpus)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="initParallelRun_+3A_cpus">cpus</code></td>
<td>
<p>Number of cores to use (optional). If given, only that number of CPUs 
will be used. By default (if not set), the Rmpi will check how many CPUs are
available in the system and take the maximum number.</p>
</td></tr>
</table>

<hr>
<h2 id='lineByLine'>Line-by-line modification of files</h2><span id='topic+lineByLine'></span>

<h3>Description</h3>

<p>Modifies a data file line by line, i.e. reads a file line by line, converts each line, then writes to the modified file. This method is especially useful when modifying large datasets, where the reading of entire files may be time consuming and require a large amount of memory.</p>


<h3>Usage</h3>

<pre><code class='language-R'>lineByLine(infile, outfile, linefunc = identity, choose.lines = NULL,
choose.columns = NULL, col.sep = " ", ask = TRUE, 
blank.lines.skip = TRUE, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lineByLine_+3A_infile">infile</code></td>
<td>
<p>A character string giving the name and path of the file to be modified. </p>
</td></tr>
<tr><td><code id="lineByLine_+3A_outfile">outfile</code></td>
<td>
<p>A character string giving the name of the modified file. The name of the file is relative to the current working directory, unless the file name contains a definite path.</p>
</td></tr>
<tr><td><code id="lineByLine_+3A_linefunc">linefunc</code></td>
<td>
<p><code>lineByLine</code> modifies each line using <code>linefunc</code>. Default is the identity function. The user may define his or her own line-modifying functions, see Details for a thorough description.</p>
</td></tr>
<tr><td><code id="lineByLine_+3A_choose.lines">choose.lines</code></td>
<td>
<p>A numeric vector of lines to be selected or dropped from <code>infile</code>. Positive values refer to lines to be chosen, whereas negative values refer to lines to be skipped. The vector cannot include both positive and negative values at the same time. If &quot;NULL&quot; (default), all lines are selected.</p>
</td></tr>
<tr><td><code id="lineByLine_+3A_choose.columns">choose.columns</code></td>
<td>
<p>A numeric vector of columns to be selected (positive values) or skipped (negative values) from <code>infile</code>. The vector cannot include both positive and negative values at the same time. By default, all columns are selected without reordering among the columns. Duplication and reordering among the selected columns will occur in the modified file corresponding to the order in which the columns are listed.</p>
</td></tr>
<tr><td><code id="lineByLine_+3A_col.sep">col.sep</code></td>
<td>
<p>Specifies the separator that splits the columns in <code>infile</code>. By default, <code>col.sep = " "</code> (space). To split at all types of spaces or blank characters, set <code>col.sep = "[[:space:]]"</code> or <code>col.sep = "[[:blank:]]"</code>.</p>
</td></tr>
<tr><td><code id="lineByLine_+3A_ask">ask</code></td>
<td>
<p>Logical. Default is &quot;TRUE&quot;. If set to &quot;FALSE&quot;, an already existing outfile will be overwritten without asking.</p>
</td></tr>
<tr><td><code id="lineByLine_+3A_blank.lines.skip">blank.lines.skip</code></td>
<td>
<p>Logical. If &quot;TRUE&quot; (default), <code>lineByLine</code> ignores blank lines in the input.</p>
</td></tr>
<tr><td><code id="lineByLine_+3A_verbose">verbose</code></td>
<td>
<p>Logical. Default is &quot;TRUE&quot;, which means that the line number is displayed for each iteration, in addition to output from <code>linefunc</code>. If <code>choose.columns</code> contains invalid column numbers, this will also be displayed.</p>
</td></tr>
<tr><td><code id="lineByLine_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>linefunc</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When reading large datafiles, functions such as <code><a href="utils.html#topic+read.table">read.table</a></code> can use a large amount of memory and be extremely time consuming.
Instead of reading the entire file at once, <code>lineByLine</code> reads one line at a time, modifies the line using <code>linefunc</code>, and then writes the line to <code>outfile</code>. <br />
The user may specify his or her own line-converting function. This function must take the argument <code>x</code>, a character vector representing a single line of the file, split at spaces. However, additional arguments may be included.
If <code>verbose</code> equals &quot;TRUE&quot;, output should be displayed.
The modified vector is returned. <br />	
The framework of the line-modifying function may look something like this:
</p>
<pre>
lineModify &lt;- function(x){
.xnew &lt;- x

## Define any modifications, for instance recoding missing values in a dataset from NA to 0:
.xnew[is.na(.xnew)] &lt;- 0

## Just to monitor progress, display, for instance, 10 first elements, without newline:
cat(paste(.xnew[1:min(10, length(.xnew))], collapse = " "))

## Return converted vector
return(.xnew)
}
</pre>
<p>See <code>Haplin:::lineConvert</code> for an additional example of a line-modifying function.
</p>


<h3>Value</h3>

<p><code>lineByLine</code> returns the number of lines read, although invisible. The main objective is the modified file.</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
</p>
<p><a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+convertPed">convertPed</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

## Extract the first ten columns from "myfile.txt", 
## without reordering
lineByLine(infile = "myfile.txt", outfile = "myfile_modified.txt", 
choose.columns = c(1:10))


## End(Not run)

</code></pre>

<hr>
<h2 id='nfam'>Count the number of families in the data</h2><span id='topic+nfam'></span>

<h3>Description</h3>

<p>This is a help function to count the number of families in an object read in with
<a href="#topic+genDataRead">genDataRead</a> (or loaded with <a href="#topic+genDataLoad">genDataLoad</a>). Note: it is assumed that
the study design is either 'triad' or 'cc.triad'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nfam(data.in)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nfam_+3A_data.in">data.in</code></td>
<td>
<p>The data read in by <a href="#topic+genDataRead">genDataRead</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>How many families (integer).
</p>

<hr>
<h2 id='nindiv'>Count the number of individuals in the data</h2><span id='topic+nindiv'></span>

<h3>Description</h3>

<p>This is a help function to count the number of individuals in an object read in with
<a href="#topic+genDataRead">genDataRead</a> (or loaded with <a href="#topic+genDataLoad">genDataLoad</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nindiv(data.in, design = "triad")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nindiv_+3A_data.in">data.in</code></td>
<td>
<p>The data read in by <a href="#topic+genDataRead">genDataRead</a>.</p>
</td></tr>
<tr><td><code id="nindiv_+3A_design">design</code></td>
<td>
<p>The design used in the study - choose from:
</p>

<ul>
<li> <p><em>triad</em> (default) - data includes genotypes of mother, father and child;
</p>
</li>
<li> <p><em>cc</em> - classical case-control;
</p>
</li>
<li> <p><em>cc.triad</em> - hybrid design: triads with cases and controls
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>How many individuals (integer).
</p>

<hr>
<h2 id='nsnps'>Count the number of markers in the data</h2><span id='topic+nsnps'></span>

<h3>Description</h3>

<p>This is a help function to count the number of markers in an object read in with
<a href="#topic+genDataRead">genDataRead</a> (or loaded with <a href="#topic+genDataLoad">genDataLoad</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nsnps(data.in, design = "triad")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nsnps_+3A_data.in">data.in</code></td>
<td>
<p>The data read in by <a href="#topic+genDataRead">genDataRead</a>.</p>
</td></tr>
<tr><td><code id="nsnps_+3A_design">design</code></td>
<td>
<p>The design used in the study - choose from:
</p>

<ul>
<li> <p><em>triad</em> (default) - data includes genotypes of mother, father and child;
</p>
</li>
<li> <p><em>cc</em> - classical case-control;
</p>
</li>
<li> <p><em>cc.triad</em> - hybrid design: triads with cases and controls
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>How many markers (integer).
</p>

<hr>
<h2 id='output'>Save files with summary, table, and plot from a haplin object.</h2><span id='topic+output'></span>

<h3>Description</h3>

<p>Create summary tables and figure from a <code>haplin</code> object. Save results as separate files in a specified directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	output(object, dirname, ask = T)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="output_+3A_object">object</code></td>
<td>
<p>A <code>haplin</code> object, i.e. the result of running <code>haplin</code>.</p>
</td></tr>
<tr><td><code id="output_+3A_dirname">dirname</code></td>
<td>
<p>Text string, for instance &quot;c:/work/haplinresults&quot;. Name of directory where results should be saved. Default is to save results in the current working directory.</p>
</td></tr>
<tr><td><code id="output_+3A_ask">ask</code></td>
<td>
<p>Logical. If TRUE (default), you will be asked before overwriting any files with the same name. If FALSE, <code>output</code> will overwrite without warning.</p>
</td></tr>
</table>


<h3>Details</h3>

 
<p>After having run <code>haplin</code> and saved the result (in the R workspace), the <code>output</code> function will extract summary results, a summary table, and a plot of the results and save them to the specified directory. The filenames will be haplin_summary.txt, haplin_table.txt and haplin_plot.jpg, respectively. <code>output</code> simply uses the available functions <code>summary</code>, <code>haptable</code>, and <code>plot</code> to produce the files, but is a quick way of saving all the relevant results.
</p>


<h3>Note</h3>

<p>Further information is found on the web page.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Run haplin and save results in separate files 
# in the c:\work\haplinresults directory:
res &lt;- haplin("data.dat", use.missing = T, maternal = T)
output(res, dirname = "c:/work/haplinresults")

## End(Not run)</code></pre>

<hr>
<h2 id='pedToHaplin'>Convert from ped format data to Haplin format</h2><span id='topic+pedToHaplin'></span>

<h3>Description</h3>

<p>Converts an ASCII file from a standard ped format to the Haplin format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedToHaplin(indata, outdata, merge = F, na.strings = "0", sep, 
colnames.out = F)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pedToHaplin_+3A_indata">indata</code></td>
<td>
<p>A character string giving the name and path of the ASCII data file to be converted. </p>
</td></tr>
<tr><td><code id="pedToHaplin_+3A_outdata">outdata</code></td>
<td>
<p>A character string giving the name and path for saving the converted file. </p>
</td></tr>
<tr><td><code id="pedToHaplin_+3A_merge">merge</code></td>
<td>
<p>If the alleles of each genotype are in two separate columns in the <code>indata</code> file, they must be merged (with the &quot;;&quot; separator) in the <code>outdata</code> file. This is done by setting <code>merge = TRUE</code>. Otherwise, it must be set to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="pedToHaplin_+3A_na.strings">na.strings</code></td>
<td>
<p>The symbol used to denote missing data in <code>indata</code>. It is passed directly to R's <code>read.table</code></p>
</td></tr>
<tr><td><code id="pedToHaplin_+3A_sep">sep</code></td>
<td>
<p>Column separator in <code>indata</code>. If unspecified, any white space will be used, as in <code>read.table</code>.</p>
</td></tr> 
<tr><td><code id="pedToHaplin_+3A_colnames.out">colnames.out</code></td>
<td>
<p>Provided just for the purpose of checking data. If TRUE, adds colnames to the <code>outdata</code> file to make it more readable. NOTE: Haplin does currently not use colnames, so this should be set to FALSE when producing the file to run on.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Important: The first 6 columns should always be family id, individual id, father's id, mother's id, sex and casetype, in that order, then followed by the genetic data columns. If the genetic data columns are separated into two individual alleles, one should use the option <code>merge = TRUE</code> to merge them in the output file. If they are already joined in single columns, for instance as CT or C;T, <code>merge</code> should be set to <code>FALSE</code> (default).
</p>
<p>Additional covariates can be included in the input file. If so, they should be placed after the 6 leading columns but before the genetic data. In this case, one should make sure the genetic data columns are already merged, and that <code>merge = FALSE</code>. (The <code>merge = TRUE</code> option when covariates are present will hopefully be implemented at some point...)
</p>
<p>Note that the output file usually has three columns before (to the left of) the columns containing genetic data. These columns are family id, sex, and casetype. When running haplin on the output file one should specify the argument 'n.vars = 3' in haplin. If the data are from the x chromosome the haplin arguments should also include 'sex = 2' and 'xchrom = T'. Similarly, if the casetype variable is a case-control indicator one should use the argument 'ccvar = 3'. If the intention is to only run haplin on the cases the case triads should be saved separately in a new file prior to running haplin on it.
</p>


<h3>Value</h3>

<p>The <code>outdata</code> file is written to disk. <code>pedToHaplin</code> returns (invisibly) the converted data file.
</p>


<h3>Warning</h3>

<p>Data files come in many shapes and formats, so you should always check the output from <code>pedToHaplin</code> before using it.</p>


<h3>Note</h3>

<p>Further information is found on the web page.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

# Standard run on supplied test file:
pedToHaplin("test_pedToHaplin.ped", outdata = "test_pedToHaplin_result.txt", 
colnames.out = F, merge = T)


## End(Not run)

</code></pre>

<hr>
<h2 id='pJohnson'>The Johnson distributions</h2><span id='topic+pJohnson'></span><span id='topic+JohnsonFit'></span>

<h3>Description</h3>

<p>Density of the Johnson distribution; adapted from the orphaned SuppDists package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pJohnson(q, parms, lower.tail = TRUE, log.p = FALSE)

JohnsonFit(t, moment = "quant")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pJohnson_+3A_q">q</code></td>
<td>
<p>vector of quantities.</p>
</td></tr>
<tr><td><code id="pJohnson_+3A_parms">parms</code></td>
<td>
<p>list or list of lists each containing output of <code>JohnsonFit</code>.</p>
</td></tr>
<tr><td><code id="pJohnson_+3A_lower.tail">lower.tail</code></td>
<td>
<p>logical vector; if TRUE (default), probabilities are <code class="reqn">P[X &lt;= x]</code>,
otherwise, <code class="reqn">P[X &gt; x]</code>.</p>
</td></tr>
<tr><td><code id="pJohnson_+3A_log.p">log.p</code></td>
<td>
<p>logical vector; if TRUE, probabilities p are given as log(p).</p>
</td></tr>
<tr><td><code id="pJohnson_+3A_t">t</code></td>
<td>
<p>observation vector, t=x.</p>
</td></tr>
<tr><td><code id="pJohnson_+3A_moment">moment</code></td>
<td>
<p>character scalar specifying t: for now only &quot;quant&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Johnson system (Johnson 1949) is a very flexible system for describing statistical
distributions. It is defined by
</p>
<p style="text-align: center;"><code class="reqn">z=\gamma+\delta \log{f(u)}, u=(x-\xi)/\lambda</code>
</p>

<p>and where <code class="reqn">f( )</code> has four possible forms:
</p>

<table>
<tr>
 <td style="text-align: left;">
  	SL:</td><td style="text-align: left;"> <code class="reqn">f(u)=u</code> the log normal </td>
</tr>
<tr>
 <td style="text-align: left;">
  	SU:</td><td style="text-align: left;"> <code class="reqn">f(u)=u+\sqrt{1+u^2}</code> an unbounded distribution</td>
</tr>
<tr>
 <td style="text-align: left;">
  	SB:</td><td style="text-align: left;"> <code class="reqn">f(u)=u/(1-u)</code> a bounded distribution</td>
</tr>
<tr>
 <td style="text-align: left;">
  	SN:</td><td style="text-align: left;"> <code class="reqn">\exp(u)</code> the normal
  </td>
</tr>

</table>

<p>Estimation of the Johnson parameters may be done from quantiles. The procedure of
Wheeler (1980) is used. They may also be estimated from the moments.
Applied Statistics algorithm 99, due to Hill, Hill, and Holder (1976) has been
translated into C for this implementation.
</p>


<h3>Value</h3>

<p><code>pJohnson()</code> gives the distribution function.
<code>JohnsonFit()</code> outputs a list containing the Johnson parameters
(gamma, delta, xi, lambda, type), where type is one of the Johnson types: &quot;SN&quot;, &quot;SL&quot;,
&quot;SB&quot;, or &quot;SU&quot;. <code>JohnsonFit()</code> does this using 5 order statistics when
moment=&quot;quant&quot;.
</p>


<h3>Author(s)</h3>

<p>Bob Wheeler
</p>


<h3>References</h3>

<p>Hill, I.D., Hill, R., and Holder, R.L. (1976). Fitting Johnson curves
by moments. <em>Applied Statistics.</em> AS99;
Johnson, N.L. (1949). Systems of frequency curves generated by methods of translation.
<em>Biometrika,</em> <b>36.</b> 149-176;
Wheeler, R.E. (1980). Quantile estimators of Johnson curve parameters. <em>Biometrika.</em>
<b>67-3</b> 725-728
</p>

<hr>
<h2 id='plot.haplin'>Plot a haplin object</h2><span id='topic+plot.haplin'></span>

<h3>Description</h3>

<p>Plot a haplin object and (optionally) produce picture files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'haplin'
plot(x, reference, separate.plots = F, filename, 
filetype = "png", use.dd, ...) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.haplin_+3A_x">x</code></td>
<td>
<p>A <code>haplin</code> object, i.e. the result of running <code>haplin</code>. This is the only required argument.</p>
</td></tr>
<tr><td><code id="plot.haplin_+3A_reference">reference</code></td>
<td>
<p>Same as <code>reference</code> argument in <code>haplin</code>. Note that when plotting, you can only choose <code>"reciprocal"</code>, <code>"population"</code> or <code>"ref.cat"</code>. You cannot use a numeric value to change the reference category, to do that <code>haplin</code> must be run over again. (See the <code>reference</code> argument of <code>haplin</code>.)</p>
</td></tr>
<tr><td><code id="plot.haplin_+3A_separate.plots">separate.plots</code></td>
<td>
<p>Logical. If you estimate effects of both fetal and maternal genes you can decide whether or not to plot them in the same plot. The default is the same plot (TRUE), the alternative (FALSE) means in separate plots. If you choose separate plots you may have to set the graphics window to &quot;recording&quot; to make sure you can scroll back to the first plot.</p>
</td></tr>
<tr><td><code id="plot.haplin_+3A_filename">filename</code></td>
<td>
<p>If you want a file containing the plot to be produced, give a character string for the filename.</p>
</td></tr>
<tr><td><code id="plot.haplin_+3A_filetype">filetype</code></td>
<td>
<p>The default filetype is <code>"png"</code>, alternatively you can choose <code>"jpeg"</code>.</p>
</td></tr>
<tr><td><code id="plot.haplin_+3A_use.dd">use.dd</code></td>
<td>
<p>Numeric vector indicating which double dose estimates should be plotted. For instance, if set to c(1,3) only the first and third haplotypes will be drawn with double dose estimates. This is useful if some haplotypes are rare and you want to exclude the uncertain estimates from the plot.</p>
</td></tr>
<tr><td><code id="plot.haplin_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed on to the plot function</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Further information is found on the web page.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Produce separate plots for child and mother, dump plots to files:
res &lt;- haplin("data.dat", use.missing = T, maternal = T)
plot(res, separate.plots = T, filename = "Haplinres.png")


## End(Not run)</code></pre>

<hr>
<h2 id='plot.haplinSlide'>Plotter function for haplinSlide.</h2><span id='topic+plot.haplinSlide'></span>

<h3>Description</h3>

<p>This will plot any haplinSlide object in one figure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'haplinSlide'
plot(
  x,
  filename,
  title,
  windows,
  plot.signif.only = FALSE,
  signif.thresh = 0.05,
  y.lim,
  x.title,
  y.title,
  file.width,
  file.height,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.haplinSlide_+3A_x">x</code></td>
<td>
<p>The <code><a href="#topic+haplinSlide">haplinSlide</a></code> object (NB: only the output produced by
running <code><a href="#topic+haplinSlide">haplinSlide</a></code> with the <code>table.output</code> argument set to TRUE!)</p>
</td></tr>
<tr><td><code id="plot.haplinSlide_+3A_filename">filename</code></td>
<td>
<p>If the plot should be saved to the disk, give the name of the output
file including the file extension.</p>
</td></tr>
<tr><td><code id="plot.haplinSlide_+3A_title">title</code></td>
<td>
<p>If the user wishes to override the default title of the plot, give it here.</p>
</td></tr>
<tr><td><code id="plot.haplinSlide_+3A_windows">windows</code></td>
<td>
<p>Numerical vector. If given, this will only plot the chosen windows.</p>
</td></tr>
<tr><td><code id="plot.haplinSlide_+3A_plot.signif.only">plot.signif.only</code></td>
<td>
<p>Logical: whether to filter out the &quot;non-significant&quot; markers
from the plot. Default: FALSE, i.e., plot everything.</p>
</td></tr>
<tr><td><code id="plot.haplinSlide_+3A_signif.thresh">signif.thresh</code></td>
<td>
<p>The threshold defining the significant p-values: if
<code>plot.signif.only == TRUE</code>, then only the markers with relative risk p-values
lower than the threshold will be kept for plotting. Default: 0.05.</p>
</td></tr>
<tr><td><code id="plot.haplinSlide_+3A_y.lim">y.lim</code></td>
<td>
<p>Vector with two numbers setting the Y limits of the plotted graph.</p>
</td></tr>
<tr><td><code id="plot.haplinSlide_+3A_x.title">x.title</code></td>
<td>
<p>Title for the X axis (default: &quot;marker&quot;).</p>
</td></tr>
<tr><td><code id="plot.haplinSlide_+3A_y.title">y.title</code></td>
<td>
<p>Title for the Y axis (default: &quot;RR (log scale)&quot;).</p>
</td></tr>
<tr><td><code id="plot.haplinSlide_+3A_file.width">file.width</code></td>
<td>
<p>Width (in inches) for the output plot, if a filename was given.</p>
</td></tr>
<tr><td><code id="plot.haplinSlide_+3A_file.height">file.height</code></td>
<td>
<p>Height (in inches) for the output plot, if a filename was given.</p>
</td></tr>
<tr><td><code id="plot.haplinSlide_+3A_...">...</code></td>
<td>
<p>other arguments (ignored).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code><a href="#topic+haplinSlide">haplinSlide</a></code> object is a list of <code><a href="#topic+haplin">haplin</a></code> results - by
default in <code><a href="#topic+haptable">haptable</a></code> form. This is used to plot the relative risk
estimates for all the markers in one plot, to enable easy comparison. NB: those 
estimates that have infinite confidence interval will not be plotted.
</p>


<h3>Value</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code> object.
</p>

<hr>
<h2 id='plot.haplinStrat'>Plotter function for haplinStrat results.</h2><span id='topic+plot.haplinStrat'></span>

<h3>Description</h3>

<p>This will automatically plot all haplinStrat results on one figure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'haplinStrat'
plot(
  x,
  filename,
  title,
  windows,
  plot.signif.only = FALSE,
  signif.thresh = 0.05,
  y.lim,
  x.title,
  y.title,
  file.width,
  file.height,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.haplinStrat_+3A_x">x</code></td>
<td>
<p>The <code><a href="#topic+haplinSlide">haplinSlide</a></code> object (NB: only the output produced by
running <code><a href="#topic+haplinSlide">haplinSlide</a></code> with the <code>table.output</code> argument set to TRUE!)</p>
</td></tr>
<tr><td><code id="plot.haplinStrat_+3A_filename">filename</code></td>
<td>
<p>If the plot should be saved to the disk, give the name of the output
file including the file extension.</p>
</td></tr>
<tr><td><code id="plot.haplinStrat_+3A_title">title</code></td>
<td>
<p>If the user wishes to override the default title of the plot, give it here.</p>
</td></tr>
<tr><td><code id="plot.haplinStrat_+3A_windows">windows</code></td>
<td>
<p>Numerical vector. If given, this will only plot the chosen windows.</p>
</td></tr>
<tr><td><code id="plot.haplinStrat_+3A_plot.signif.only">plot.signif.only</code></td>
<td>
<p>Logical: whether to filter out the &quot;non-significant&quot; markers
from the plot. Default: FALSE, i.e., plot everything.</p>
</td></tr>
<tr><td><code id="plot.haplinStrat_+3A_signif.thresh">signif.thresh</code></td>
<td>
<p>The threshold defining the significant p-values: if
<code>plot.signif.only == TRUE</code>, then only the markers with relative risk p-values
lower than the threshold will be kept for plotting. Default: 0.05.</p>
</td></tr>
<tr><td><code id="plot.haplinStrat_+3A_y.lim">y.lim</code></td>
<td>
<p>Vector with two numbers setting the Y limits of the plotted graph.</p>
</td></tr>
<tr><td><code id="plot.haplinStrat_+3A_x.title">x.title</code></td>
<td>
<p>Title for the X axis (default: &quot;marker&quot;).</p>
</td></tr>
<tr><td><code id="plot.haplinStrat_+3A_y.title">y.title</code></td>
<td>
<p>Title for the Y axis (default: &quot;RR (log scale)&quot;).</p>
</td></tr>
<tr><td><code id="plot.haplinStrat_+3A_file.width">file.width</code></td>
<td>
<p>Width (in inches) for the output plot, if a filename was given.</p>
</td></tr>
<tr><td><code id="plot.haplinStrat_+3A_file.height">file.height</code></td>
<td>
<p>Height (in inches) for the output plot, if a filename was given.</p>
</td></tr>
<tr><td><code id="plot.haplinStrat_+3A_...">...</code></td>
<td>
<p>other arguments (ignored).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses the same style as <code><a href="#topic+plot.haplinSlide">plot.haplinSlide</a></code> and plots all of
the <code><a href="#topic+haplinStrat">haplinStrat</a></code> results on one figure, for easy comparison. NB: those 
estimates that have infinite confidence interval will not be plotted.
</p>


<h3>Value</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code> object.
</p>

<hr>
<h2 id='plot.haptable'>Plot a haptable object</h2><span id='topic+plot.haptable'></span>

<h3>Description</h3>

<p>Plot a haptable object, which is the result of running <code>haptable</code> on a <code>haplin</code> result, and (optionally) produce picture files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'haptable'
plot(x, separate.plots = F, filename, 
filetype = "png", use.dd, verbose = T, ...) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.haptable_+3A_x">x</code></td>
<td>
<p>A <code>haptable</code> object, i.e. the result of running <code>haptable</code> on a result from <code>haplin</code>. This is the only required argument.</p>
</td></tr>
<tr><td><code id="plot.haptable_+3A_separate.plots">separate.plots</code></td>
<td>
<p>Logical. If you estimate effects of both fetal and maternal genes you can decide whether or not to plot them in the same plot. The default is the same plot (TRUE), the alternative (FALSE) means in separate plots. If you choose separate plots you may have to set the graphics window to &quot;recording&quot; to make sure you can scroll back to the first plot.</p>
</td></tr>
<tr><td><code id="plot.haptable_+3A_filename">filename</code></td>
<td>
<p>If you want a file containing the plot to be produced, give a character string for the filename.</p>
</td></tr>
<tr><td><code id="plot.haptable_+3A_filetype">filetype</code></td>
<td>
<p>The default filetype is <code>"png"</code>, alternatively you can choose <code>"jpeg"</code>.</p>
</td></tr>
<tr><td><code id="plot.haptable_+3A_use.dd">use.dd</code></td>
<td>
<p>Numeric vector indicating which double dose estimates should be plotted. For instance, if set to c(1,3) only the first and third haplotypes will be drawn with double dose estimates. This is useful if some haplotypes are rare and you want to exclude the uncertain estimates from the plot.</p>
</td></tr>
<tr><td><code id="plot.haptable_+3A_verbose">verbose</code></td>
<td>
<p>Turns on or off some minor comments when plotting</p>
</td></tr>
<tr><td><code id="plot.haptable_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed on to the plot function</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Further information is found on the web page.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Directly plotting the haplin result. Produce separate plots for child and mother, 
# dump plots to files:
res &lt;- haplin("data.dat", use.missing = T, maternal = T)
plot(res, separate.plots = T, filename = "Haplinres.png")

# Create haptable from the haplin result and plot the results in the table:
res &lt;- haplin("data.dat", use.missing = T, maternal = T)
tab &lt;- haptable(res)
plot(tab)

# Create haptables for the 10 first markers of a data file using haplinSlide. 
# Create plots for each result:
res &lt;- haplinSlide("data.dat", markers = 1:10, use.missing = T, maternal = T, table.output = T)
lapply(res, plot)

## End(Not run)</code></pre>

<hr>
<h2 id='plotPValues'>Plotting p-values for relative risks</h2><span id='topic+plotPValues'></span>

<h3>Description</h3>

<p>This function plots p-values for the relative risks calculated by <code><a href="#topic+haplinSlide">haplinSlide</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPValues(
  object,
  windows,
  which.p.val = "overall",
  plot.signif.only = FALSE,
  signif.thresh = 0.05,
  title,
  filename
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotPValues_+3A_object">object</code></td>
<td>
<p>The <code><a href="#topic+haplinSlide">haplinSlide</a></code> results: list of <code><a href="#topic+haptable">haptable</a></code>
objects.</p>
</td></tr>
<tr><td><code id="plotPValues_+3A_windows">windows</code></td>
<td>
<p>Numerical vector; if given, the plot will be restricted to only those.</p>
</td></tr>
<tr><td><code id="plotPValues_+3A_which.p.val">which.p.val</code></td>
<td>
<p>Character string specifying which p-values to choose for plotting:
&quot;overall&quot; (default), &quot;child&quot;, &quot;child.double&quot;, &quot;maternal&quot;, &quot;maternal.double&quot;,
&quot;paternal&quot;. The last three options can be chosen only if <code><a href="#topic+haplinSlide">haplinSlide</a></code>
was run with <code>maternal = TRUE</code> or <code>poo = TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotPValues_+3A_plot.signif.only">plot.signif.only</code></td>
<td>
<p>Logical: whether to filter out the &quot;non-significant&quot; markers
from the plot. Default: FALSE, i.e., plot everything.</p>
</td></tr>
<tr><td><code id="plotPValues_+3A_signif.thresh">signif.thresh</code></td>
<td>
<p>The threshold defining the significant p-values: if
<code>plot.signif.only == TRUE</code>, then only the markers with relative risk p-values
lower than the threshold will be kept for plotting. Default: 0.05.</p>
</td></tr>
<tr><td><code id="plotPValues_+3A_title">title</code></td>
<td>
<p>Optional character string for the title of the figure.</p>
</td></tr>
<tr><td><code id="plotPValues_+3A_filename">filename</code></td>
<td>
<p>If the plot should be saved to the disk, give the name of the output
file including the file extension.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output of <code><a href="#topic+haplinSlide">haplinSlide</a></code> can be very lengthy and not suitable for
an overall plot of all the relative risks (RR) on one figure. Therefore, it's advised
to first plot only the p-values for each window (user can choose which p-values to
plot - see parameter <code>which.p.val</code>), and only then plot the RRs for specific
windows, for which the p-values are significant.
</p>


<h3>Value</h3>

<p>Invisibly returns the table with only the plotted p-values.
</p>

<hr>
<h2 id='pQQ'>QQ-plot with confidence intervals for a vector of p-values</h2><span id='topic+pQQ'></span>

<h3>Description</h3>

<p>Produces a QQ-plot of p-values. The x-axis is -log10 of the expected p-values (under a null hypothesis of no effects), the y-axis is -log10 values of the actual p-values. A (pointwise) confidence interval can be drawn, and names of snps/genes corresponding to the most significant ones can be added.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pQQ(pvals, nlabs = 6, conf = 0.95, lim, mark = 0.05, ...) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pQQ_+3A_pvals">pvals</code></td>
<td>
<p>A vector of p-values. </p>
</td></tr>
<tr><td><code id="pQQ_+3A_nlabs">nlabs</code></td>
<td>
<p>The number of (most significant) p-values to be labeled using <code>names(pvals)</code>.</p>
</td></tr>
<tr><td><code id="pQQ_+3A_conf">conf</code></td>
<td>
<p>The confidence level of the pointwise confidence band. The default is 0.95. The confidence intervals are computed under the assumption of the p-values being drawn independently from a uniform [0,1] distribution. To leave out the confidence interval, set this to FALSE.</p>
</td></tr>
<tr><td><code id="pQQ_+3A_lim">lim</code></td>
<td>
<p>A vector of length 2 giving the plot limits (on a log10 scal, for instance c(0,4)). Plot limits are computed automatically. However, if other plot limits are desireable, they can be set using this argument. </p>
</td></tr>
<tr><td><code id="pQQ_+3A_mark">mark</code></td>
<td>
<p>By default, the 0.05 significance level is marked by lines. Can be changed to a different value, or set to FALSE. </p>
</td></tr>
<tr><td><code id="pQQ_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to the plotting function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>pvals</code> argument should be a vector of p-values to be plotted. If the vector has names corresponding to marker (snp) names, the plot will label some of the most significant points with the marker names.
</p>


<h3>Value</h3>

<p>No value is returned.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>

<hr>
<h2 id='print.haplin'>Print a haplin object</h2><span id='topic+print.haplin'></span>

<h3>Description</h3>

<p>Print basic information about a haplin object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'haplin'
print(x, ...) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.haplin_+3A_x">x</code></td>
<td>
<p>A <code>haplin</code> object, i.e. the result of running <code>haplin</code>.</p>
</td></tr>
<tr><td><code id="print.haplin_+3A_...">...</code></td>
<td>
<p>Other arguments, passed on to <code>print</code>.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Further information is found on the web page
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code></p>

<hr>
<h2 id='print.summary.haplin'>Print the summary of a haplin object</h2><span id='topic+print.summary.haplin'></span>

<h3>Description</h3>

<p>Print the result of applying <code>summary</code> to a haplin object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'summary.haplin'
print(x, digits, ...) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.summary.haplin_+3A_x">x</code></td>
<td>
<p>A <code>haplin</code> object, i.e. the result of running <code>haplin</code>.</p>
</td></tr>
<tr><td><code id="print.summary.haplin_+3A_digits">digits</code></td>
<td>
<p>The number of digits to be used in the printout. Defaults to 3.</p>
</td></tr>
<tr><td><code id="print.summary.haplin_+3A_...">...</code></td>
<td>
<p>Other arguments (ignored).</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Further information is found on the web page
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Standard summary:
res &lt;- haplin("data.dat", use.missing = T, maternal = T)
summary(res)

# Increase number of digits in printout
print(summary(res), digits = 8)


## End(Not run)</code></pre>

<hr>
<h2 id='rbindFiles'>Combine a sequence of files by rows</h2><span id='topic+rbindFiles'></span>

<h3>Description</h3>

<p>Takes a sequence of files and combines them by rows, without reading the full files into memory. This is especially useful when dealing with large datasets, where the reading of entire files may be time consuming and require a large amount of memory.</p>


<h3>Usage</h3>

<pre><code class='language-R'>rbindFiles(infiles, outfile, col.sep, header = FALSE, ask = TRUE, 
verbose = FALSE, add.file.number = FALSE, blank.lines.skip = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rbindFiles_+3A_infiles">infiles</code></td>
<td>
<p>A character vector of names (and paths) of the files to combine.</p>
</td></tr>
<tr><td><code id="rbindFiles_+3A_outfile">outfile</code></td>
<td>
<p>A character string giving the name of the modified file. The name of the file is relative to the current working directory, unless the file name contains a definite path.</p>
</td></tr>
<tr><td><code id="rbindFiles_+3A_col.sep">col.sep</code></td>
<td>
<p>Specifies the separator used to split the columns in the files. To split at all types of spaces or blank characters, set <code>col.sep = "[[:space:]]"</code> or <code>col.sep = "[[:blank:]]"</code>.</p>
</td></tr>
<tr><td><code id="rbindFiles_+3A_header">header</code></td>
<td>
<p>A logical variable which indicates if the first line in each file contains the names of the variables. If &quot;TRUE&quot;, <code>outfile</code> will display this header in its first row, assuming the headers for each file are identical. Equals FALSE by default, i.e. no headers assumed.</p>
</td></tr>
<tr><td><code id="rbindFiles_+3A_ask">ask</code></td>
<td>
<p>Logical. Default is &quot;TRUE&quot;. If set to &quot;FALSE&quot;, an already existing outfile will be overwritten without asking.</p>
</td></tr>
<tr><td><code id="rbindFiles_+3A_verbose">verbose</code></td>
<td>
<p>Logical. Default is &quot;TRUE&quot;, which means that the line number is displayed for each iteration, i.e. each combined line.</p>
</td></tr>
<tr><td><code id="rbindFiles_+3A_add.file.number">add.file.number</code></td>
<td>
<p>A logical variable which equals &quot;FALSE&quot; by deafult. If &quot;TRUE&quot;, an extra first column will be added to the outfile, consisting of the file numbers for each line.</p>
</td></tr>
<tr><td><code id="rbindFiles_+3A_blank.lines.skip">blank.lines.skip</code></td>
<td>
<p>Logical. If &quot;TRUE&quot; (default), <code><a href="#topic+lineByLine">lineByLine</a></code> ignores blank lines in the input.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="base.html#topic+cbind">rbind</a></code> combines R objects by rows. However, reading large data files may require a large amount of memory and be extremely time consuming.
<code>rbindFiles</code> avoids reading the full files into memory. It reads the files line by line, possibly modifies each line, then writes to outfile. 
If however, <code>header</code>, <code>verbose</code>, <code>add.file.number</code> and <code>blank.lines.skip</code> are all set to &quot;FALSE&quot; 
(their default values), the files are appended directly, thus evading line-by-line modifications. 
In the case where <code>infiles</code> contains only one file and no output or modifications are requested 
(<code>verbose</code>, <code>add.file.number</code> and <code>blank.lines.skip</code> equal &quot;FALSE&quot;), an identical copy of this file is made.
</p>


<h3>Value</h3>

<p>There is no useful output; the objective of <code>rbindFiles</code> is to produce <code>outfile</code>.</p>


<h3>Note</h3>

<p>Combining the files by reading each file line by line is less time efficient than appending the files directly. For this reason, if <code>header = FALSE</code>, changing the values of the logical variables <code>verbose</code>, <code>add.file.number</code>
and <code>blank.lines.skip</code> from &quot;FALSE&quot; to &quot;TRUE&quot; should not be done unless absolutely necessary.
</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cbindFiles">cbindFiles</a></code>, <code><a href="#topic+lineByLine">lineByLine</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Combines the three infiles, by rows
rbindFiles(file.names = c("myfile1.txt", "myfile2.txt", "myfile3.txt"), 
outfile = "myfile_combined_by_rows.txt", col.sep = " ", header = TRUE, verbose = TRUE)


## End(Not run)
</code></pre>

<hr>
<h2 id='showGen'>Display chosen genotypes</h2><span id='topic+showGen'></span>

<h3>Description</h3>

<p>This is a help function to extract genotypes from an object read in with
<a href="#topic+genDataRead">genDataRead</a> (or loaded with <a href="#topic+genDataLoad">genDataLoad</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showGen(data.in, design = "triad", n = 5, from, to, sex, markers = 1:5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="showGen_+3A_data.in">data.in</code></td>
<td>
<p>The data read in by <a href="#topic+genDataRead">genDataRead</a>.</p>
</td></tr>
<tr><td><code id="showGen_+3A_design">design</code></td>
<td>
<p>The design used in the study - choose from:
</p>

<ul>
<li> <p><em>triad</em> - data includes genotypes of mother, father and child;
</p>
</li>
<li> <p><em>cc</em> - classical case-control;
</p>
</li>
<li> <p><em>cc.triad</em> - hybrid design: triads with cases and controls;
</p>
</li></ul>
<p>.</p>
</td></tr>
<tr><td><code id="showGen_+3A_n">n</code></td>
<td>
<p>Number of rows to display or &quot;all&quot; (default: 5).</p>
</td></tr>
<tr><td><code id="showGen_+3A_from">from</code></td>
<td>
<p>From which row to display (optional, default: from the first).</p>
</td></tr>
<tr><td><code id="showGen_+3A_to">to</code></td>
<td>
<p>To which row to display (optional).</p>
</td></tr>
<tr><td><code id="showGen_+3A_sex">sex</code></td>
<td>
<p>If the sex column is part of the phenotypic information, the user can
choose based on one of the categories used in this column (optional);
NB: this does not combine with the 'to' and 'from' arguments.</p>
</td></tr>
<tr><td><code id="showGen_+3A_markers">markers</code></td>
<td>
<p>A vector specifying which markers to display or &quot;all&quot; (default: first
5); NB: the user can specify the markers by numbers or by their names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table with genotypes extracted from 'data.in'.
</p>

<hr>
<h2 id='showPheno'>Display phenotype part of data</h2><span id='topic+showPheno'></span>

<h3>Description</h3>

<p>This is a help function to extract phenotypic (and covariate) data from an object
read in with <a href="#topic+genDataRead">genDataRead</a> (or loaded with <a href="#topic+genDataLoad">genDataLoad</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showPheno(data.in, n = 5, from, to, sex)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="showPheno_+3A_data.in">data.in</code></td>
<td>
<p>The data read in by <a href="#topic+genDataRead">genDataRead</a>).</p>
</td></tr>
<tr><td><code id="showPheno_+3A_n">n</code></td>
<td>
<p>Number of rows to display or &quot;all&quot; (default: 5).</p>
</td></tr>
<tr><td><code id="showPheno_+3A_from">from</code></td>
<td>
<p>From which row to display (optional, default: from the first).</p>
</td></tr>
<tr><td><code id="showPheno_+3A_to">to</code></td>
<td>
<p>To which row to display (optional).</p>
</td></tr>
<tr><td><code id="showPheno_+3A_sex">sex</code></td>
<td>
<p>If the sex column is part of the phenotypic information, the user can
choose based on one of the categories used in this column (optional);
NB: this does not combine with the 'to' and 'from' arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table with phenotypic and covariate data (if any) extracted from 'data.in'.
</p>

<hr>
<h2 id='showSNPnames'>Display marker names</h2><span id='topic+showSNPnames'></span>

<h3>Description</h3>

<p>This is a help function to extract marker names from an object generated by
<a href="#topic+genDataRead">genDataRead</a>, <a href="#topic+genDataLoad">genDataLoad</a>, <a href="#topic+genDataGetPart">genDataGetPart</a> or 
<a href="#topic+genDataPreprocess">genDataPreprocess</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showSNPnames(data.in, n = 5, from, to)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="showSNPnames_+3A_data.in">data.in</code></td>
<td>
<p>The data as outputted by <a href="#topic+genDataRead">genDataRead</a>, <a href="#topic+genDataLoad">genDataLoad</a>, 
<a href="#topic+genDataGetPart">genDataGetPart</a> or <a href="#topic+genDataPreprocess">genDataPreprocess</a>.</p>
</td></tr>
<tr><td><code id="showSNPnames_+3A_n">n</code></td>
<td>
<p>Number of names to display or &quot;all&quot; (default: 5).</p>
</td></tr>
<tr><td><code id="showSNPnames_+3A_from">from</code></td>
<td>
<p>From which marker to display (optional, default: from the first).</p>
</td></tr>
<tr><td><code id="showSNPnames_+3A_to">to</code></td>
<td>
<p>To which marker to display (optional).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with marker names, as read in from map.file or generated dummy names.
</p>

<hr>
<h2 id='snpPos'>Find the column numbers of SNP identifiers/SNP numbers in a ped file</h2><span id='topic+snpPos'></span>

<h3>Description</h3>

<p>Gives the column numbers of SNP identifiers or SNP numbers in a standard ped file, calculated from the SNP's positions in the corresponding map file. The column numbers are sorted in the same order as <code>snp.select</code>.
These positions may be useful when extracting a selection of SNPs from a ped file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpPos(snp.select, map.file, blank.lines.skip = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snpPos_+3A_snp.select">snp.select</code></td>
<td>
<p>A character vector of the SNP identifiers (RS codes) or a numeric vector of the SNP numbers.</p>
</td></tr>
<tr><td><code id="snpPos_+3A_map.file">map.file</code></td>
<td>
<p>A character string giving the name and path of the standard map file to be used.  See Details for a description of the standard map format.</p>
</td></tr>
<tr><td><code id="snpPos_+3A_blank.lines.skip">blank.lines.skip</code></td>
<td>
<p>Logical. If &quot;TRUE&quot; (default), <code>snpPos</code> ignores blank lines in <code>map.file</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To extract certain SNPs from a standard ped file, one has to know their positions in the ped file.
This can be obtained from the corresponding map file.<br />
</p>
<p>The map file should look something like this:
</p>
<pre>
Chromosome SNP-identifier Base-pair-position
1               RS9629043             554636
1              RS12565286             711153 
1              RS12138618             740098
</pre>
<p>Alternatively, the map file could contain four columns. The column values should then be:
Chromosome, SNP-identifier, Genetic-distance, Base-pair-position.<br />
A header must be added to the map file if this does not already exist.<br />
<br />
The format of the corresponding ped file should be something like this:
</p>
<pre>
1104  1104-1  1104-2  1104-3  1  2  4  1  3  2
1104  1104-2       0       0  1  1  4  1  2  2
1104  1104-3       0       0  2  1  0  0  0  0
1105  1105-1  1105-2  1105-3  2  2  1  1  2  2
1105  1105-2       0       0  1  1  1  1  2  2
1105  1105-3       0       0  2  1  1  1  3  2
</pre>
<p>The column values are: Family id, Individual id, Father's id, Mother's id, Sex (1 = male, 2 = female, alternatively: 1 = male, 0 = female), and Case-control status (1 = controls, 2 = cases, alternatively: 0 = controls, 1 = cases).<br />
Column 7 and onwards contain the genotype data, with alleles in separate columns. A &ldquo;0&rdquo; is used to denote missing data.<br />
</p>


<h3>Value</h3>

<p>A vector of the column numbers of the SNP identifiers/SNP numbers in the ped file, sorted in the same order as given in <code>snp.select</code>.
</p>


<h3>Note</h3>

<p>The function does not check if the map file is formatted correctly or if the map and ped file have the same number of SNPs. The corresponding positions of the SNPs in the ped file may not be correct if the ped file has a different format from the given example.</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+convertPed">convertPed</a></code>, <code><a href="#topic+lineByLine">lineByLine</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

# Find the column numbers of the SNP identifiers "RS9629043" and "RS12565286" in 
# a standard ped file
snpPos(snp.select = c("RS9629043", "RS12565286"), map.file = "mygwas.map")

## End(Not run)

</code></pre>

<hr>
<h2 id='snpPower'>Power calculations for a single SNP</h2><span id='topic+snpPower'></span>

<h3>Description</h3>

<p>Calculates power for a single SNP. Allows for power computations of several combinations simultaneously.</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpPower(cases, controls, RR, MAF, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snpPower_+3A_cases">cases</code></td>
<td>
<p>A list of the number of case families. Each element contains the number of families of a specified family design. The possible family designs, i.e. the possible names of the elements, are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child dyad), <code>"fc"</code> (father-child dyad) or <code>"c"</code> (a single case child).</p>
</td></tr>
<tr><td><code id="snpPower_+3A_controls">controls</code></td>
<td>
<p>A list of the number of control families. Each element contains the number of families of a specified family design. The possible family designs are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child-dyad), <code>"fc"</code> (father-child dyad), <code>"mf"</code> (mother-father dyad), <code>"c"</code> (a single control child), <code>"m"</code> (a single control mother) or <code>"f"</code> (a single control father).</p>
</td></tr>
<tr><td><code id="snpPower_+3A_rr">RR</code></td>
<td>
<p>A numeric vector of the relative risks (the effect sizes of interest).</p>
</td></tr>
<tr><td><code id="snpPower_+3A_maf">MAF</code></td>
<td>
<p>A numeric vector of the minor allele frequencies.</p>
</td></tr>
<tr><td><code id="snpPower_+3A_alpha">alpha</code></td>
<td>
<p>A numeric vector of the Type I Errors. Equals 0.05 by default.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>snpPower</code> computes power for a single SNP by counting the number of &quot;real&quot; case alleles, &quot;real&quot; control alleles and pseudo-control alleles. 
The pseudo-control alleles are the non-transmitted alleles, possibly from both case families and control families. It assumes a multiplicative dose-response model.
<code>snpPower</code> uses the asymptotic normal approximation for the natural logarithm of the odds ratio for calculating power 
(the relative risks and odds ratios are used interchangeably due to the &quot;rare disease assumption&quot;).<br />
</p>
<p><code>snpPower</code> allows for power calculations for mixtures of the possible case family designs and control family designs. 
The argument <code>cases</code> could, for example, consist of a combination of 1000 full case triads (family design <code>"mfc"</code>) and 500 single case children (family design <code>"c"</code>). 
It is also feasible to compute power for several combinations of the input variables simultaneously. See Examples for further details.
</p>


<h3>Value</h3>

<p><code>snpPower</code> returns a data frame containing the combinations of input variables and the corresponding power calculations.</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
</p>
<p><a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Skare O, Jugessur A, Lie RT, Wilcox AJ, Murray JC, Lunde A, Nguyen TT, Gjessing HK. Application of a novel hybrid study design to explore gene-environment interactions in orofacial clefts. Annals of Human Genetics (2012) 76, pp. 221-236.<br />
</p>
<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+snpSampleSize">snpSampleSize</a></code>, <code><a href="#topic+hapRun">hapRun</a></code>, <code><a href="#topic+hapPower">hapPower</a></code>, <code><a href="#topic+hapPowerAsymp">hapPowerAsymp</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute power for a single SNP,
## for the combination of 1000 case triads, RR = 1.2, MAF = 0.1 and alpha = 0.05
snpPower(cases = list(mfc = 1000), controls = list(mfc = 0), RR = 1.2, MAF = 0.1)

## Compute power for a single SNP,
## for the combination of 1000 case triads and 500 single case children (altogether),
## 5000 control triads, RR = 1.1, MAF = 0.1 and alpha = 0.05
snpPower(cases = list(mfc = 1000, c = 500), controls = list(mfc=5000),
RR = 1.1, MAF = 0.1, alpha = 0.05)

## Compute power for a single SNP,
## for the combination of 500 case triads, 10000 control triads, 
## relative risk of 1.2 and minor allele frequency of 0.1,
## and also for the combination of 1000 case triads, 10000 control triads,
## relative risk of 1.1 and minor allele frequency of 0.1
snpPower(cases = list(mfc = c(500, 1000)), controls = list(mfc = 10000),
RR = c(1.2, 1.1), MAF = 0.1)

</code></pre>

<hr>
<h2 id='snpSampleSize'>Sample size calculations for a single SNP</h2><span id='topic+snpSampleSize'></span>

<h3>Description</h3>

<p>Sample size calculations for a single SNP. Allows for sample size calculations of several combinations simultaneously.</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpSampleSize(fam.cases, fam.controls, fraction = 0.5,
RR, MAF, alpha = 0.05, power = 0.80)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snpSampleSize_+3A_fam.cases">fam.cases</code></td>
<td>
<p>A character vector of the case family design. The possible family designs are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child dyad), <code>"fc"</code> (father-child dyad) and <code>"c"</code> (a single case child).</p>
</td></tr>
<tr><td><code id="snpSampleSize_+3A_fam.controls">fam.controls</code></td>
<td>
<p>A character vector of the control family design. The possible family designs are <code>"mfc"</code> (full triad), <code>"mc"</code> (mother-child-dyad), <code>"fc"</code> (father-child dyad), <code>"mf"</code> (mother-father dyad), <code>"c"</code> (a single control child), <code>"m"</code> (a single control mother), <code>"f"</code> (a single control father) or <code>"no_controls"</code> (no control families).</p>
</td></tr>
<tr><td><code id="snpSampleSize_+3A_fraction">fraction</code></td>
<td>
<p>A numeric vector of the proportion of case families. Equals 0.5 by default, i.e. there are as many case families as control families. If <code>fam.controls</code> equals <code>"no_controls"</code>, <code>fraction</code> is automatically set to 1.</p>
</td></tr>
<tr><td><code id="snpSampleSize_+3A_rr">RR</code></td>
<td>
<p>A numeric vector of the relative risks (the effect sizes of interest).</p>
</td></tr>
<tr><td><code id="snpSampleSize_+3A_maf">MAF</code></td>
<td>
<p>A numeric vector of the minor allele frequencies.</p>
</td></tr>
<tr><td><code id="snpSampleSize_+3A_alpha">alpha</code></td>
<td>
<p>A numeric vector of the Type I Errors. Equals 0.05 by default.</p>
</td></tr>
<tr><td><code id="snpSampleSize_+3A_power">power</code></td>
<td>
<p>A numeric vector of the desired probability of identifying a difference in the relative risks. Default is 0.80.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>snpSampleSize</code> computes the number of case and control families required for a single SNP to attain the desired power. 
It assumes a multiplicative dose-response model.
<code>snpSampleSize</code> calculates the fraction of case alleles corresponding to the given family designs 
and then uses the asymptotic normal approximation for the natural logarithm of the odds ratio for calculating the sample sizes 
(the relative risks and odds ratios are used interchangeably due to the &quot;rare disease assumption&quot;).<br />
</p>
<p><code>snpSampleSize</code> allows for sample size calculations of several combinations of the input variables at once. The Examples section provides further details. 
</p>


<h3>Value</h3>

<p><code>snpSampleSize</code> returns a data frame containing the combinations of input variables and the corresponding sample size calculations.</p>


<h3>Author(s)</h3>

<p>Miriam Gjerdevik,<br />
with Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
</p>
<p><a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p>Skare O, Jugessur A, Lie RT, Wilcox AJ, Murray JC, Lunde A, Nguyen TT, Gjessing HK. Application of a novel hybrid study design to explore gene-environment interactions in orofacial clefts. Annals of Human Genetics (2012) 76, pp. 221-236.<br />
</p>
<p>Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+snpPower">snpPower</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Compute sample sizes for a single SNP,
## when the specified case family design is a full triad, there are no control families,
## RR = 1.1, MAF = 0.1, alpha = 0.05 and power = 0.9
snpSampleSize(fam.cases = "mfc", fam.controls = "no_controls", 
RR = 1.1, MAF = 0.1, alpha = 0.05, power = 0.9)

## Compute sample sizes for a single SNP,
## for the combination of case triads, control triads, fraction = 0.5, RR = 1.2, 
## MAF = 0.1, alpha = 0.05 and power = 0.8, and also for the combination of case triads,
## control children, fraction = 0.5, RR = 1.2, MAF = 0.1, alpha = 0.05 and power = 0.9
snpSampleSize(fam.cases = "mfc", fam.controls = c("mfc", "c"), 
RR = 1.2, MAF = 0.1, alpha = 0.05, power = c(0.8, 0.9))

</code></pre>

<hr>
<h2 id='suest'>Compute a joint p-value for a list of haplin fits (usually from a sliding window approach), correcting for multiple testing.</h2><span id='topic+suest'></span>

<h3>Description</h3>

<p>The first argument to <code>suest</code> should be a list of haplin estimation results (from the same data file), usually the output from <code><a href="#topic+haplinSlide">haplinSlide</a></code>. <code>suest</code> produces as a result a joint overall p-value based on aggregating the individual p-values and then correcting for multiple testing. The correction is achieved by using the principle of &quot;seemingly unrelated&quot; estimation, taking into account the correlation between the individual estimation results. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>suest(reslist)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="suest_+3A_reslist">reslist</code></td>
<td>
<p>A list whose elements are different haplin runs on the same data file, typically the output of <code><a href="#topic+haplinSlide">haplinSlide</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+haplinSlide">haplinSlide</a></code> runs <code>haplin</code> on a series of overlapping
windows of markers from the same data file, typically within the same gene.
Since each run produces a separate overall p-value, <code>suest</code> computes a
joint overall p-value for the gene (or region) that has been scanned. It
corrects the overall p-value for multiple testing, also taking into account
the fact that the sequence of estimates produced by <code>haplinSlide</code> will
be dependent, both because they are computed on the same data set and also
since the windows are overlapping (if the window length is larger than 1).
If the <code>suest</code> estimation fails (which doesn't happen very often), a
standard Bonferroni correction is used instead. Important:
<code>haplinSlide</code> must be run with the option <code>table.output = FALSE</code>
to provide <code>suest</code> with enough information.
</p>


<h3>Value</h3>

<p>A list is returned, the most important elements of which are:
</p>
<table role = "presentation">
<tr><td><code>pval.obs</code></td>
<td>
<p>The overall score p-values from each haplin run</p>
</td></tr>
<tr><td><code>pval.obs.corr</code></td>
<td>
<p>The joint p-value, corrected for multiple testing</p>
</td></tr>
<tr><td><code>bonferroni</code></td>
<td>
<p>A logical, usually FALSE, which means the <code>suest</code> estimation went well. If TRUE, it means that the <code>suest</code> estimation failed for some reason, and a standard Bonferroni correction was used instead.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Further information is found on the web page.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code>, <code><a href="#topic+haplinSlide">haplinSlide</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# (Almost) all standard haplin runs can be done with haplinSlide. 
# Below is an illustration. See the haplin help page for more 
# examples.
# 
# Analyzing the effect of fetal genes, including triads with missing data,
# using a multiplicative response model. When winlength = 1, separate
# markers are used. To make longer windows, winlength can be increased
# correspondingly:
result.1 &lt;- haplinSlide("C:/work/data.dat", use.missing = T, response = "mult",
reference = "ref.cat", winlength = 1, table.output = F)
# Provide summary of separate results:
lapply(result.1, summary)
# Plot results:
par(ask = T)
lapply(result.1, plot)
# Compute an overall p-value for the scan, corrected for multiple testing
# and dependencies between windows:
suest(result.1)


## End(Not run)

</code></pre>

<hr>
<h2 id='summary.haplin'>Summary of a haplin object</h2><span id='topic+summary.haplin'></span>

<h3>Description</h3>

<p>Provides detailed information about estimation results from a haplin object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'haplin'
summary(object, reference, ...) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.haplin_+3A_object">object</code></td>
<td>
<p>A <code>haplin</code> object, i.e. the result of running <code>haplin</code>.</p>
</td></tr>
<tr><td><code id="summary.haplin_+3A_reference">reference</code></td>
<td>
<p>Same as <code>reference</code> argument in <code>haplin</code>. Note that when producing the summary, you can only choose <code>"reciprocal"</code>, <code>"population"</code> or <code>"ref.cat"</code>. You cannot use a numeric value to change the reference category, to do that <code>haplin</code> must be run over again. (See the <code>reference</code> argument of <code>haplin</code>.)</p>
</td></tr>
<tr><td><code id="summary.haplin_+3A_...">...</code></td>
<td>
<p>Other arguments (ignored).</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Further information is found on the web page
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+haplin">haplin</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Produce separate plots for child and mother, dump plots to files:
res &lt;- haplin("data.dat", use.missing = T, maternal = T)
summary(res)


## End(Not run)</code></pre>

<hr>
<h2 id='toDataFrame'>Stack dataframes from haplinSlide into a single dataframe</h2><span id='topic+toDataFrame'></span>

<h3>Description</h3>

<p>When haplinSlide is run with the option <code>table.output = T</code>, the result is a list of <code>haptable</code>s, i.e. tables with summary haplin results for each window <code>haplinSlide</code> is run on. <code>toDataFrame</code> stacks the separate dataframes into one large dataframe containing all results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toDataFrame(x, reduce = F) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="toDataFrame_+3A_x">x</code></td>
<td>
<p>The output from <code>haplinSlide</code> run with option <code>table.output = TRUE</code>.</p>
</td></tr>
<tr><td><code id="toDataFrame_+3A_reduce">reduce</code></td>
<td>
<p>Reduce output to one line per marker</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When <code>haplinSlide</code> is run with <code>winlength = 1</code> on SNP markers, each table in the output has only two rows, and can be condensed to a single row. By setting the argument <code>reduce</code> to TRUE, <code>toDataFrame</code> reduces each table to one line and returns a dataframe with one line for each SNP. In more general situations, with multi-allelic markers or, more commonly, <code>winlength</code> set to 2 or more, each output table will typically have more than two rows and cannot be reduced, so <code>reduce</code> should be set to FALSE.
</p>


<h3>Value</h3>

<p>The output is a dataframe. First column contains the marker names. Second column are row numbers, counted within each output table. The remaining columns are identical to the individual output columns, which are described in more detail in the help file for <code><a href="#topic+haptable">haptable</a></code>.
</p>


<h3>Author(s)</h3>

<p>Hakon K. Gjessing<br />
Professor of Biostatistics<br />
Division of Epidemiology<br />
Norwegian Institute of Public Health<br />
<a href="mailto:hakon.gjessing@uib.no">hakon.gjessing@uib.no</a></p>


<h3>References</h3>

<p> Gjessing HK and Lie RT. Case-parent triads: Estimating single- and double-dose effects of fetal and maternal disease gene haplotypes. Annals of Human Genetics (2006) 70, pp. 382-396.<br /><br />
Web Site: <a href="https://haplin.bitbucket.io">https://haplin.bitbucket.io</a></p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
