<!DOCTYPE html><html><head><title>Help for package haplotypes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="/home/deepayan/Rinstall/R-devel/lib/R/doc/html/R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {haplotypes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#[-methods'><p>Extract or replace parts of an object of class <code>Dna</code></p></a></li>
<li><a href='#append-methods'><p>Combines two <code>Dna</code> objects</p></a></li>
<li><a href='#as.data.frame-methods'><p>Coerces a <code>Dna</code> object to a data.frame</p></a></li>
<li><a href='#as.dna-methods'><p>Coerces an object to a <code>Dna</code> object</p></a></li>
<li><a href='#as.DNAbin-methods'><p> Coerces an object to a  <code>DNAbin</code> object</p></a></li>
<li><a href='#as.list-methods'><p>Methods for function <code>as.list</code> in the Package <span class="pkg">haplotypes</span></p></a></li>
<li><a href='#as.matrix-methods'><p>Methods for function <code>as.matrix</code> in the Package <span class="pkg">haplotypes</span></p></a></li>
<li><a href='#as.network-methods'><p> Coerces an object to a <code>network</code> object</p></a></li>
<li><a href='#as.networx-methods'><p> Coerces an object to a <code>networx</code> object</p></a></li>
<li><a href='#as.numeric-methods'><p>Coerces a <code>Dna</code> object to a numeric matrix</p></a></li>
<li><a href='#as.phyDat-methods'><p> Coerces an object to a <code>phyDat</code> object</p></a></li>
<li><a href='#basecomp-methods'><p> Calculates base composition</p></a></li>
<li><a href='#boot.dna-methods'><p> Generates single bootstrap replicate</p></a></li>
<li><a href='#distance-methods'><p>Calculates absolute pairwise character difference matrix using a <code>Dna</code> object</p></a></li>
<li><a href='#Dna-class'><p>Class <code>"Dna"</code> in the Package <span class="pkg">haplotypes</span></p></a></li>
<li><a href='#dna.obj'><p>Example DNA sequence data</p></a></li>
<li><a href='#grouping-methods'><p>Groups haplotypes according to the grouping variable (populations, species, etc.)</p></a></li>
<li><a href='#Haplotype-class'><p>Class <code>"Haplotype"</code> in the Package <span class="pkg">haplotypes</span></p></a></li>
<li><a href='#haplotype-methods'><p>Methods for function <code>haplotype</code> in the package <span class="pkg">haplotypes</span></p></a></li>
<li><a href='#haplotypes-internal'><p>Internal function(s)</p></a></li>
<li><a href='#haplotypes-package'>
<p>Manipulating DNA Sequences and Estimating Unambiguous Haplotype Network with Statistical Parsimony</p></a></li>
<li><a href='#hapreord-methods'><p>Reorders haplotypes according to the ordering factor</p></a></li>
<li><a href='#homopoly-methods'><p>Provides the list of homoplastic indels and substitutions</p></a></li>
<li><a href='#image-methods'><p> Display DNA Sequence</p></a></li>
<li><a href='#indelcoder-methods'><p>Codes gaps</p></a></li>
<li><a href='#length-methods'><p>Methods for function <code>length</code> in the package <span class="pkg">haplotypes</span></p></a></li>
<li><a href='#names-methods'><p>Function to get or set names of a <code>Dna</code> object or <code>Parsimnet</code> object</p></a></li>
<li><a href='#ncol-methods'><p>Returns the length of the longest DNA sequence</p></a></li>
<li><a href='#nrow-methods'><p>Returns the number of DNA sequences</p></a></li>
<li><a href='#pairnei-methods'><p>Provides the average number of pairwise Nei's (D) differences between populations</p></a></li>
<li><a href='#pairPhiST-methods'><p> Provides the pairwise PhiST between populations</p></a></li>
<li><a href='#Parsimnet-class'><p>Class <code>"Parsimnet"</code> in the Package <span class="pkg">haplotypes</span></p></a></li>
<li><a href='#parsimnet-methods'><p>Estimates gene genealogies using statistical parsimony</p></a></li>
<li><a href='#pielegend-methods'><p>Add Legends to Plots</p></a></li>
<li><a href='#pieplot-methods'><p>Plots pie charts on statistical parsimony network</p></a></li>
<li><a href='#plot-methods'><p>Methods for function <code>plot</code> in the package <span class="pkg">haplotypes</span></p></a></li>
<li><a href='#polymorp-methods'><p>Displays polymorphic sites (base substitutions and indels) between two sequences</p></a></li>
<li><a href='#range-methods'><p>Returns the minimum and maximum lengths of the DNA sequences</p></a></li>
<li><a href='#read.fas'>
<p>Read sequences from a file in FASTA format</p></a></li>
<li><a href='#remove.gaps-methods'><p> Removing gaps from <code>Dna</code> object</p></a></li>
<li><a href='#rownames-methods'><p>Retrieve or set the row names</p></a></li>
<li><a href='#show-methods'><p>Methods for function <code>show</code> in the package <span class="pkg">haplotypes</span></p></a></li>
<li><a href='#subs-methods'><p>Displays base substitutions</p></a></li>
<li><a href='#tolower-methods, toupper-methods'><p> Convert sequence characters from upper to lower case or vice versa</p></a></li>
<li><a href='#unique-methods'><p>Extract Unique Sequences</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Manipulating DNA Sequences and Estimating Unambiguous Haplotype
Network with Statistical Parsimony</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.3.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-07-12</td>
</tr>
<tr>
<td>Author:</td>
<td>Caner Aktas</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Caner Aktas &lt;caktas.aca@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides S4 classes and methods for reading and manipulating aligned DNA sequences, supporting an indel coding methods (only simple indel coding method is available in the current version), showing base substitutions and indels, calculating absolute pairwise distances between DNA sequences, and collapses identical DNA sequences into haplotypes or inferring haplotypes using user provided absolute pairwise character difference matrix.  This package also includes S4 classes and methods for estimating genealogical relationships among haplotypes using statistical parsimony and plotting parsimony networks.  </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://cran.r-project.org">https://cran.r-project.org</a>,
<a href="https://biolsystematics.wordpress.com/r/">https://biolsystematics.wordpress.com/r/</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, stats, utils, network, sna, ape, phangorn,plotrix</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-07-12 05:58:08 UTC; cnr</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-07-15 05:40:03 UTC</td>
</tr>
<tr>
<td>Built:</td>
<td>R 4.4.0; ; 2024-01-02 08:00:09 UTC; unix</td>
</tr>
</table>
<hr>
<h2 id='+5B-methods'>Extract or replace parts of an object of class <code>Dna</code></h2><span id='topic++5B-methods'></span><span id='topic++5B+2CDna-method'></span><span id='topic++5B+3C-+2CDna-method'></span><span id='topic++5B+2CDna+2CANY+2CANY+2CANY-method'></span><span id='topic++5B+3C-+2CDna+2CANY+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>Operators acting on sequence matrix to extract or
replace parts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
x[i, j,..., drop = FALSE]
## S4 replacement method for signature 'Dna'
x[i, j]&lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B-methods_+3A_x">x</code></td>
<td>
<p>an object of class  <code><a href="haplotypes.html#topic+Dna">Dna</a></code> </p>
</td></tr>
<tr><td><code id="+2B5B-methods_+3A_i">i</code>, <code id="+2B5B-methods_+3A_j">j</code></td>
<td>
<p>elements to extract or replace.</p>
</td></tr>
<tr><td><code id="+2B5B-methods_+3A_...">...</code></td>
<td>

<p>Additional arguments. In this context, ... is used primarily for as.matrix, which is a boolean. If as.matrix is TRUE (default), the function returns a matrix. If drop is also TRUE, and the subset has either a single row or column, the function will return a vector instead. If as.matrix is FALSE, the function returns an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>. The as.matrix argument should be specified as as.matrix=TRUE or as.matrix=FALSE within the call. If it is not specified, the function defaults to TRUE.
</p>
</td></tr>
<tr><td><code id="+2B5B-methods_+3A_drop">drop</code></td>
<td>
<p>boolean; if TRUE and a single row or column is selected, the function returns a vector instead of a matrix. This is only applicable when as.matrix is TRUE.</p>
</td></tr>
<tr><td><code id="+2B5B-methods_+3A_value">value</code></td>
<td>
<p>a character vector or a character matrix. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The S4 method dispatch mechanism matches arguments based on those specified in the signature of the corresponding generic function. However, for some generics that include '...' in their signature, additional arguments can be incorporated in specific methods. Notably, the '[' function does not follow this pattern and restricts the arguments to those defined in its signature.
In this context, the 'as.matrix' argument is not in the signature of the generic '[', so it is included within '...'. Then, within the body of the function, we check whether 'as.matrix' has been provided in the actual arguments when the function is called. If 'as.matrix' is not specified, the function defaults to 'TRUE', preserving the behavior of previous versions of the method.   
</p>


<h3>Value</h3>

<p>returns an object of class matrix, vector or  <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna", i = "ANY", j = "ANY", drop = "ANY")</code></dt><dd>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+Dna">Dna</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("dna.obj")
x&lt;-dna.obj

## Extract parts.
# a matrix object
x[1:2,1:3]

# a Dna object, as.dna(x[1:2,1:3]) gives the same result
x[1:2,1:3,as.matrix=FALSE] 

# a vector object
x[1,1:4,drop=TRUE]

## Replace parts.
#"G" "C"
x[1,1:2] 	
x[1,1:2]&lt;-c("A","T")
x[1,1:2]

</code></pre>

<hr>
<h2 id='append-methods'>Combines two <code>Dna</code> objects</h2><span id='topic+append-methods'></span><span id='topic+append+2CDna-method'></span>

<h3>Description</h3>

<p>Combines two <code><a href="haplotypes.html#topic+Dna">Dna</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
append(x,values)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="append-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>. </p>
</td></tr>
<tr><td><code id="append-methods_+3A_values">values</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna", values= "Dna")</code></dt><dd>
<p>combines two <code>Dna</code> objects.
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>	
data("dna.obj")

x&lt;-dna.obj
y&lt;-dna.obj
nrow(x)

## Combining two 'Dna' objects.
z&lt;- append(x,y)
nrow(z)


</code></pre>

<hr>
<h2 id='as.data.frame-methods'>Coerces a <code>Dna</code> object to a data.frame</h2><span id='topic+as.data.frame-methods'></span><span id='topic+as.data.frame+2CDna-method'></span>

<h3>Description</h3>

<p>Coerces an object to a data.frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
as.data.frame(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.data.frame-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a data frame.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>coerces a Dna object to a data.frame. </p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>
data("dna.obj")

x&lt;-dna.obj
x&lt;-as.dna(x[1:4,1:6])

## Coercing a 'Dna' object to a data.frame.
df&lt;-as.data.frame(x)
df

# TRUE
is.data.frame(df) 

## Not run: 
# gives the same result 
df&lt;-as.data.frame(x@sequence) 
df

## End(Not run)
</code></pre>

<hr>
<h2 id='as.dna-methods'>Coerces an object to a <code>Dna</code> object</h2><span id='topic+as.dna'></span><span id='topic+as.dna-methods'></span><span id='topic+as.dna+2Ccharacter-method'></span><span id='topic+as.dna+2Cdata.frame-method'></span><span id='topic+as.dna+2CDNAbin-method'></span><span id='topic+as.dna+2CHaplotype-method'></span><span id='topic+as.dna+2Clist-method'></span><span id='topic+as.dna+2Cmatrix-method'></span><span id='topic+as.dna+2CphyDat-method'></span>

<h3>Description</h3>

<p>Coerces an object that contains DNA sequences to an object of Class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'matrix'
as.dna(x)
## S4 method for signature 'data.frame'
as.dna(x)
## S4 method for signature 'list'
as.dna(x)
## S4 method for signature 'character'
as.dna(x)
## S4 method for signature 'Haplotype'
as.dna(x)
## S4 method for signature 'DNAbin'
as.dna(x)
## S4 method for signature 'phyDat'
as.dna(x)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.dna-methods_+3A_x">x</code></td>
<td>
<p> a matrix, a data.frame, a list, a character, an object of class <code><a href="haplotypes.html#topic+Haplotype">Haplotype</a></code>, <code><a href="ape.html#topic+DNAbin">DNAbin</a></code> <code>{ape}</code> or <code><a href="phangorn.html#topic+phyDat">phyDat</a></code> <code>{phangorn}</code> object containing the DNA sequences.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Elements of the list must be vectors. Each element of the list contains a single DNA sequence. If the sequence lengths differ, the longest sequence is taken into account and gaps are introduced to the shorter sequences at the end of the matrix in the slot <code>sequence</code>. Sequence length information is stored in the slot <code>seqlengths</code>.
</p>
<p>Valid characters for the slot <code>sequence</code> are &quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;,&quot;a&quot;,&quot;c&quot;,&quot;g&quot;,&quot;t&quot;,&quot;-&quot;,&quot;?&quot;. During the conversion of the object to the class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>, integers 0,1,2,3,4,5 or characters &quot;0&quot;,&quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;,&quot;5&quot; are converted to &quot;?&quot;,&quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;,&quot;-&quot;, respectively. Invalid characters are replaced with &quot;?&quot; with a warning message.  
</p>


<h3>Value</h3>

<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a>.</code>
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "matrix")</code></dt><dd>
<p>coerces matrix to a <code>Dna</code> object.
</p>
</dd>
<dt><code>signature(x = "data.frame")</code></dt><dd>
<p>coerces data.frame to a <code>Dna</code> object.
</p>
</dd>
<dt><code>signature(x = "list")</code></dt><dd>
<p>coerces list to a <code>Dna</code> object.
</p>
</dd>
<dt><code>signature(x = "character")</code></dt><dd>
<p>coerces characters to a <code>Dna</code> object.
</p>
</dd>
<dt><code>signature(x = "Haplotype")</code></dt><dd>
<p>coerces a <code><a href="haplotypes.html#topic+Haplotype">Haplotype</a></code> object to a <code>Dna</code> object.
</p>
</dd>
<dt><code>signature(x = "DNAbin")</code></dt><dd>
<p>coerces a <code><a href="ape.html#topic+DNAbin">DNAbin</a></code> object to a <code>Dna</code> object.
</p>
</dd>
<dt><code>signature(x = "phyDat")</code></dt><dd>
<p>coerces a <code><a href="phangorn.html#topic+phyDat">phyDat</a></code> object to a <code>Dna</code> object.
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>
## Coercing a matrix to a 'Dna' object.
# all valid characters
x&lt;-matrix(c("?","A","C","g","t","-","0","1","2","3","4","5"),4,6)
rownames(x)&lt;-c("seq1","seq2","seq3","seq4") 
dna.obj&lt;-as.dna(x) 
dna.obj
# the sequence matrix
dna.obj@sequence

## Not run: 
# includes invalid characters
x&lt;-matrix(c("X","y","*","?","t","-","0","1","2","3","4","5"),4,6) 
rownames(x)&lt;-c("seq1","seq2","seq3","seq4") 
dna.obj&lt;-as.dna(x) 
dna.obj
dna.obj@sequence

# all valid integers
x&lt;-matrix(c(0,1,2,3,4,5,0,1,2,3,4,5),4,6) 
rownames(x)&lt;-c("seq1","seq2","seq3","seq4") 
dna.obj&lt;-as.dna(x) 
dna.obj
dna.obj@sequence

## Coercing a data.frame to a 'Dna' object.
x&lt;-data.frame(matrix(c("?","A","C","g","t","-","0","1","2","3","4","5"),4,6)) 
rownames(x)&lt;-c("seq1","seq2","seq3","seq4") 
dna.obj&lt;-as.dna(x) 
dna.obj
dna.obj@sequence

## Coercing a list to a 'Dna' object.
seq1&lt;-c("?","A","C","g","t","-","0","1")
seq2&lt;-c("?","A","C","g","t","-","0","1","2")
seq3&lt;-c("?","A","C","g","t","-","0","1","2","3")
x&lt;-list(seq1=seq1,seq2=seq2,seq3=seq3) 
dna.obj&lt;-as.dna(x)

# sequence lengths differ 
dna.obj@seqlengths
dna.obj@sequence

## Coercing a character vector to a Dna object.
seq&lt;-c("?","A","C","g","t","-","0","1")
x&lt;-as.dna(seq)
x


## Coercing a Haplotype object to a Dna object.
data("dna.obj")
x&lt;-dna.obj
h&lt;-haplotype(x)

# DNA Sequences of unique haplotypes  
dna.obj&lt;-as.dna(h)
dna.obj

d&lt;-distance(x)

# if 'Haplotype' object does not contain 'DNA' Sequences
h&lt;-haplotype(d)

# returns an error
as.dna(h)

## Coercing a DNAbin object to a Dna object.
require(ape)
data(woodmouse)
x&lt;-as.dna(woodmouse)
x


## End(Not run)</code></pre>

<hr>
<h2 id='as.DNAbin-methods'> Coerces an object to a  <code>DNAbin</code> object</h2><span id='topic+as.DNAbin'></span><span id='topic+as.DNAbin-methods'></span><span id='topic+as.DNAbin+2CDna-method'></span>

<h3>Description</h3>

<p>This function coerces <code>Dna</code> object to <code><a href="ape.html#topic+DNAbin">DNAbin</a></code> <code>{ape}</code> object .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
as.DNAbin(x, endgaps=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.DNAbin-methods_+3A_x">x</code></td>
<td>
<p> an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
<tr><td><code id="as.DNAbin-methods_+3A_endgaps">endgaps</code></td>
<td>
<p> boolean; gaps at the end of the sequences are included if this is TRUE. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code><a href="ape.html#topic+DNAbin">DNAbin</a>.</code>
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>coerces a <code><a href="haplotypes.html#topic+Dna">Dna</a></code> object to a <code>DNAbin</code> object.
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>
## Coercing a Dna object to a DNAbin object.
data("dna.obj")

x&lt;-dna.obj
dBin&lt;-as.DNAbin(x)
dBin

#gaps at the end removed
dBin&lt;-as.DNAbin(x, endgaps=FALSE)
dBin


</code></pre>

<hr>
<h2 id='as.list-methods'>Methods for function <code>as.list</code> in the Package <span class="pkg">haplotypes</span></h2><span id='topic+as.list-methods'></span><span id='topic+as.list+2CDna-method'></span><span id='topic+as.list+2CHaplotype-method'></span><span id='topic+as.list+2CParsimnet-method'></span>

<h3>Description</h3>

<p>Coerces an object to a list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
as.list(x)
## S4 method for signature 'Haplotype'
as.list(x)
## S4 method for signature 'Parsimnet'
as.list(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.list-methods_+3A_x">x</code></td>
<td>
<p> an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>,<code><a href="haplotypes.html#topic+Haplotype">Haplotype</a></code> or <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>x</code> is a <code><a href="haplotypes.html#topic+Dna">Dna</a></code> object, elements of the list are character vectors that contains the DNA sequences of length equal to corresponding value in the slot <code>seqlengths</code>. If <code>x</code> is <code><a href="haplotypes.html#topic+Haplotype">Haplotype</a></code> or <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code> objects, slots are converted to list elements.
</p>


<h3>Value</h3>

<p>returns a list.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>coerces an object of class Dna to a list. 
</p>
</dd>
<dt><code>signature(x = "Haplotype")</code></dt><dd>
<p>coerces an object of class Haplotype to a list. 
</p>
</dd>
<dt><code>signature(x = "Parsimnet")</code></dt><dd>
<p>coerces an object of class Parsimnet to a list. 
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>
data("dna.obj")

## Coercing a 'Dna' object to a list.
x&lt;-dna.obj[1:3,as.matrix=FALSE]
as.list(x)

## Not run: 
## Coercing a 'Haplotype' object to a list.
x&lt;-dna.obj
h&lt;-haplotype(x)
as.list(h)

## Coercing a 'Parsimnet' object to a list.
x&lt;-dna.obj
p&lt;-parsimnet(x)
as.list(p)

## End(Not run)


</code></pre>

<hr>
<h2 id='as.matrix-methods'>Methods for function <code>as.matrix</code> in the Package <span class="pkg">haplotypes</span></h2><span id='topic+as.matrix-methods'></span><span id='topic+as.matrix+2CDna-method'></span>

<h3>Description</h3>

<p>Coerces an object to a matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
as.matrix(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.matrix-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a character matrix.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>coerces an object of class Dna to a matrix.
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")

## Coercing a 'Dna' object to a matrix.
x&lt;-dna.obj[1:4,1:6,as.matrix=FALSE]
x
as.matrix(x)

## Not run: 
# gives the same result
dna.obj[1:4,1:6,as.matrix=TRUE] 
## End(Not run)

</code></pre>

<hr>
<h2 id='as.network-methods'> Coerces an object to a <code>network</code> object</h2><span id='topic+as.network'></span><span id='topic+as.network-methods'></span><span id='topic+as.network+2CParsimnet-method'></span>

<h3>Description</h3>

<p>This function coerces <code>Parsimnet</code> object to <code><a href="network.html#topic+network">network</a></code> <code>{network}</code> object .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Parsimnet'
as.network(x,net=1,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.network-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code>.</p>
</td></tr>
<tr><td><code id="as.network-methods_+3A_net">net</code></td>
<td>
<p>a numeric vector of length one indicating which network to convert.</p>
</td></tr>
<tr><td><code id="as.network-methods_+3A_...">...</code></td>
<td>
<p>additional arguments to function <code><a href="network.html#topic+network">network</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code><a href="network.html#topic+network">network</a>.</code>
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Parsimnet")</code></dt><dd>
<p>coerces a <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code> object to a <code>network</code> object.
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>
## Coercing a Parsimnet object to a network object.
data("dna.obj")
x&lt;-dna.obj
p&lt;-parsimnet(x)
n&lt;-as.network(p)

#Fourth network (with only two edges)
p&lt;-parsimnet(x,prob=.99)
n&lt;-as.network(p,net=4)



</code></pre>

<hr>
<h2 id='as.networx-methods'> Coerces an object to a <code>networx</code> object</h2><span id='topic+as.networx'></span><span id='topic+as.networx-methods'></span><span id='topic+as.networx+2CParsimnet-method'></span>

<h3>Description</h3>

<p>This function coerces <code>Parsimnet</code> object to <code><a href="phangorn.html#topic+networx">networx</a></code> <code>{phangorn}</code> object .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Parsimnet'
as.networx(x,net=1,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.networx-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code>.</p>
</td></tr>
<tr><td><code id="as.networx-methods_+3A_net">net</code></td>
<td>
<p>a numeric vector of length one indicating which network to convert.</p>
</td></tr>
<tr><td><code id="as.networx-methods_+3A_...">...</code></td>
<td>
<p>additional arguments to <code><a href="phangorn.html#topic+as.splits">as.splits</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code><a href="phangorn.html#topic+networx">networx</a>.</code>
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Parsimnet")</code></dt><dd>
<p>coerces a <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code> object to a <code>networx</code> object.
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>
## Coercing a Parsimnet object to a networx object.
data("dna.obj")
x&lt;-dna.obj
p&lt;-parsimnet(x)
nx&lt;-as.networx(p)
plot(nx, "2D")

</code></pre>

<hr>
<h2 id='as.numeric-methods'>Coerces a <code>Dna</code> object to a numeric matrix</h2><span id='topic+as.numeric-methods'></span><span id='topic+as.numeric+2CDna-method'></span>

<h3>Description</h3>

<p>Converts a character matrix to a numeric matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
as.numeric(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.numeric-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function <code>as.numeric()</code> coerces the character matrix in the slot <code>sequence</code> to a numeric matrix. 
Lower or upper case characters &quot;?&quot;,&quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;,&quot;-&quot; are converted to integers 0,1,2,3,4,5, respectively. 
</p>


<h3>Value</h3>

<p>returns a numeric matrix.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>coerces a Dna object to a numeric matrix.
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>x&lt;-matrix(c("?","A","C","g","t","-","0","1","2","3","4","5"),4,6)
rownames(x)&lt;-c("seq1","seq2","seq3","seq4") 
x&lt;-as.dna(x)
# original character matrix 
as.matrix(x) 

## Coercing a 'Dna' object to a numeric matrix.
# numeric matrix
as.numeric(x)

</code></pre>

<hr>
<h2 id='as.phyDat-methods'> Coerces an object to a <code>phyDat</code> object</h2><span id='topic+as.phyDat'></span><span id='topic+as.phyDat-methods'></span><span id='topic+as.phyDat+2CDna-method'></span>

<h3>Description</h3>

<p>This function coerces <code>Dna</code> object to <code><a href="phangorn.html#topic+phyDat">phyDat</a></code> <code>{phangorn}</code> object .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
as.phyDat(x, indels="sic",...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.phyDat-methods_+3A_x">x</code></td>
<td>
<p>an object of class  <code><a href="haplotypes.html#topic+Dna">Dna</a></code>. </p>
</td></tr>
<tr><td><code id="as.phyDat-methods_+3A_indels">indels</code></td>
<td>
<p>the indel coding method to be used. This must be one of &quot;sic&quot;, &quot;5th&quot; or &quot;missing&quot;. Any unambiguous substring can be given.  See also &lsquo;Details&rsquo; </p>
</td></tr>
<tr><td><code id="as.phyDat-methods_+3A_...">...</code></td>
<td>
<p>additional arguments to <code><a href="haplotypes.html#topic+as.phyDat">as.phyDat</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Available indel coding methods:
</p>

<dl>
<dt><code>sic</code>:</dt><dd><p>Treating gaps as a missing character and coding them separately following the simple indel coding method.</p>
</dd>
<dt><code>5th</code>:</dt><dd><p>Treating gaps as a fifth state character.</p>
</dd>
<dt><code>missing</code>:</dt><dd><p>Treating gaps as a missing character.</p>
</dd>
</dl>



<h3>Value</h3>

<p>an object of class <code><a href="phangorn.html#topic+phyDat">phyDat</a>.</code>
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>coerces a <code><a href="haplotypes.html#topic+Dna">Dna</a></code> object to a <code>phyDat</code> object.
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>

data("dna.obj")
x&lt;-dna.obj


## Coercing a Dna object to a phyDat object.
# Simple indel coding.
phyd&lt;-as.phyDat(x)
phyd

# Gaps as 5th state characters.
phyd&lt;-as.phyDat(x,indels="5")
phyd

# Gaps as 5th state characters.
phyd&lt;-as.phyDat(x,indels="m")
phyd


</code></pre>

<hr>
<h2 id='basecomp-methods'> Calculates base composition </h2><span id='topic+basecomp'></span><span id='topic+basecomp-methods'></span><span id='topic+basecomp+2CDna-method'></span>

<h3>Description</h3>

<p>Calculates base composition of <code><a href="haplotypes.html#topic+Dna">Dna</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
basecomp(x)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="basecomp-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix with sequence as rows, DNA bases as columns and frequencies as entries.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>calculates base composition of <code><a href="haplotypes.html#topic+Dna">Dna</a></code> object.
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x &lt;-dna.obj

## Calculating base compositions.
basecomp(x)



</code></pre>

<hr>
<h2 id='boot.dna-methods'> Generates single bootstrap replicate</h2><span id='topic+boot.dna'></span><span id='topic+boot.dna-methods'></span><span id='topic+boot.dna+2CDna-method'></span>

<h3>Description</h3>

<p>Methods for generating a single bootstrap replicate. </p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
boot.dna(x,replacement=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="boot.dna-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
<tr><td><code id="boot.dna-methods_+3A_replacement">replacement</code></td>
<td>
<p>boolean; whether the sampling is done with replacement or without replacement.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a>.</code>
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>generates single bootstrap replicate from a <code>Dna</code> object.
</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	
	
data("dna.obj")
x&lt;-dna.obj

## Generating a bootstrap replicate.
# with replacement
bxr&lt;-boot.dna(x)
image(bxr)

# without replacement
bx&lt;-boot.dna(x,replacement=FALSE)
image(bx)
</code></pre>

<hr>
<h2 id='distance-methods'>Calculates absolute pairwise character difference matrix using a <code>Dna</code> object</h2><span id='topic+distance'></span><span id='topic+distance-methods'></span><span id='topic+distance+2CDna-method'></span>

<h3>Description</h3>

<p>Computes and returns an absolute pairwise character difference matrix from DNA sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
distance(x,subset=NULL,indels="sic")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distance-methods_+3A_x">x</code></td>
<td>
<p>an object of class  <code><a href="haplotypes.html#topic+Dna">Dna</a></code>. </p>
</td></tr>
<tr><td><code id="distance-methods_+3A_subset">subset</code></td>
<td>
<p>a vector of integers in the range [1,nrow(x)], specifying which sequence(s) are used in the distance calculation. Only distance between selected sequence(s) and the rest of the sequences are calculated. If it is NULL, all comparisons are done. </p>
</td></tr>
<tr><td><code id="distance-methods_+3A_indels">indels</code></td>
<td>
<p>the indel coding method to be used. This must be one of &quot;sic&quot;, &quot;5th&quot; or &quot;missing&quot;. Any unambiguous substring can be given.  See also &lsquo;Details&rsquo; </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Available indel coding methods:
</p>

<dl>
<dt><code>sic</code>:</dt><dd><p>Treating gaps as a missing character and coding them separately following the simple indel coding method.</p>
</dd>
<dt><code>5th</code>:</dt><dd><p>Treating gaps as a fifth state character.</p>
</dd>
<dt><code>missing</code>:</dt><dd><p>Treating gaps as a missing character.</p>
</dd>
</dl>



<h3>Value</h3>

<p>returns an object of class <code>dist</code>.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>Computes and returns an absolute pairwise character difference matrix from <code>Dna</code> objects.
</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>References</h3>

<p>Giribet, G. and Wheeler, W.C. (1999) On gaps. <em>Molecular Phylogenetics and Evolution</em> <b>13</b>, 132-143.
</p>
<p>Simmons, M., Ochoterena, H. (2000) Gaps as characters in sequence-based phylogenetic analyses. <em>Systematic Biology</em> <b>49</b>, 369-381.
</p>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+indelcoder">indelcoder</a></code> and <code><a href="haplotypes.html#topic+subs">subs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x&lt;-dna.obj[4:7,13:22,as.matrix=FALSE]

## Simple indel coding.
distance(x,indels="s")

## Gaps as 5th state characters.
distance(x,indels="5")

## Gaps as missing characters.
distance(x,indels="m")

## Not run: 
## Using 'subset'.
x&lt;-dna.obj[4:10,13:22,as.matrix=FALSE]
distance(x, NULL)
distance(x, subset=c(1))
distance(x, subset=c(2,4))

## End(Not run)
</code></pre>

<hr>
<h2 id='Dna-class'>Class <code>"Dna"</code> in the Package <span class="pkg">haplotypes</span></h2><span id='topic+Dna'></span><span id='topic+Dna-class'></span>

<h3>Description</h3>

<p>S4 class to hold DNA sequence data. </p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("Dna", sequence, seqlengths, seqnames)</code>, however reading fasta file using   <code><a href="haplotypes.html#topic+read.fas">read.fas</a></code> function or coerce matrix, data.frame or list objects to a Dna object using <code><a href="haplotypes.html#topic+as.dna">as.dna</a></code> methods is preferable.
</p>


<h3>Slots</h3>


<dl>
<dt><code>sequence</code>:</dt><dd><p>Object of class <code>"matrix"</code> containing DNA sequence data, rows represent sequences and columns represent sites. See also &lsquo;Note&rsquo;.</p>
</dd>
<dt><code>seqlengths</code>:</dt><dd><p>Object of class <code>"numeric"</code> containing the length of each DNA sequence.</p>
</dd>
<dt><code>seqnames</code>:</dt><dd><p>Object of class <code>"character"</code> containing the name of each DNA sequence.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>[</dt><dd><p><code>signature(x = "Dna", i = "ANY", j = "ANY")</code>: extracts part of a DNA sequence as an object of class matrix.</p>
</dd>
<dt>[&lt;-</dt><dd><p><code>signature(x = "Dna", i = "ANY", j = "ANY", value = "ANY")</code>: replaces part of a Dna sequence with an object of class <code>"matrix"</code>, <code>"numeric"</code> or <code>"character"</code>.</p>
</dd>
<dt>append</dt><dd><p><code>signature(x = "Dna", value = "ANY")</code>: combines two Dna objects.</p>
</dd>
<dt>as.data.frame</dt><dd><p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a data.frame.</p>
</dd>
<dt>as.list</dt><dd><p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a list; elements of the list are character vectors that contains the DNA sequences of length equal to corresponding value in the slot <code>seqlengths</code>.</p>
</dd>  
<dt>as.matrix</dt><dd><p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a matrix.</p>
</dd>
<dt>as.numeric</dt><dd><p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a numeric matrix.</p>
</dd>
<dt>as.DNAbin</dt><dd><p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a  <code>DNAbin</code> object.</p>
</dd>
<dt>as.phyDat</dt><dd><p><code>signature(x = "Dna")</code>: coerces an object of class Dna to a  <code>phyDat</code> object.</p>
</dd>
<dt>basecomp</dt><dd><p><code>signature(x = "Dna")</code>: calculates base composition of Dna object.</p>
</dd>
<dt>boot.dna</dt><dd><p><code>signature(x = "Dna")</code>: generates single bootstrap replicate.</p>
</dd>
<dt>distance</dt><dd><p><code>signature(x = "Dna")</code>: computes and returns an absolute pairwise character difference matrix from DNA sequences. </p>
</dd>
<dt>haplotype</dt><dd><p><code>signature(x = "Dna")</code>: infers haplotypes from DNA sequences. </p>
</dd>
<dt>image</dt><dd><p><code>signature(x = "Dna")</code>: displays DNA sequences </p>
</dd>
<dt>indelcoder</dt><dd><p><code>signature(x = "Dna")</code>: supports simple indel coding method. </p>
</dd>
<dt>length</dt><dd><p><code>signature(x = "Dna")</code>: returns the longest sequence length. </p>
</dd>
<dt>append</dt><dd><p><code>signature(x = "Dna", value = "ANY")</code>: combines two Dna objects.</p>
</dd>
<dt>names</dt><dd><p><code>signature(x = "Dna")</code>: gets the names of an object Dna. </p>
</dd>
<dt>names&lt;-</dt><dd><p><code>signature(x = "Dna")</code>: sets the names of an object Dna. </p>
</dd>
<dt>ncol</dt><dd><p><code>signature(x = "Dna")</code>: returns the longest sequence length. </p>
</dd>
<dt>nrow</dt><dd><p><code>signature(x = "Dna")</code>: returns the total sequence number. </p>
</dd>
<dt>pairnei</dt><dd><p><code>signature(x = "Dna")</code>:  calculates pairwise Nei's average number of differences between populations. </p>
</dd>
<dt>pairPhiST</dt><dd><p><code>signature(x = "Dna")</code>:  calculates pairwise PhiST between populations. </p>
</dd>
<dt>parsimnet</dt><dd><p><code>signature(x = "Dna")</code>: estimates genealogies using statistical parsimony.</p>
</dd>
<dt>polymorp</dt><dd><p><code>signature(x = "Dna")</code>: displays information about DNA polymorphisms of two sequences; indels and base substitutions, respectively.</p>
</dd>
<dt>range</dt><dd><p><code>signature(object = "Dna")</code>: returns a vector containing the minimum and maximum length of DNA sequences. </p>
</dd>
<dt>remove.gaps</dt><dd><p><code>signature(object = "Dna")</code>: removes alignment gaps. </p>
</dd>
<dt>rownames</dt><dd><p><code>signature(object = "Dna")</code>: retrieve the row names of a DNA sequence matrix.</p>
</dd>
<dt>rownames&lt;-</dt><dd><p><code>signature(object = "Dna")</code>: set the row names of a DNA sequence matrix.</p>
</dd>
<dt>show</dt><dd><p><code>signature(object = "Dna")</code>: displays Dna object briefly. </p>
</dd>
<dt>subs</dt><dd><p><code>signature(x = "Dna")</code>: displays information about base substitutions. </p>
</dd>
<dt>tolower</dt><dd><p><code>signature(x = "Dna")</code>: Translate characters in DNA sequence matrix from upper to lower case.</p>
</dd>
<dt>toupper</dt><dd><p><code>signature(x = "Dna")</code>: Translate characters in DNA sequence matrix from lower to upper case.</p>
</dd>
<dt>unique</dt><dd><p><code>signature(x = "Dna")</code>: returns a list with duplicate DNA sequences removed.</p>
</dd>
</dl>



<h3>Note</h3>

<p>Valid characters for the slot <code>sequence</code> are &quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;,&quot;a&quot;,&quot;c&quot;,&quot;g&quot;,&quot;t&quot;,&quot;-&quot;,&quot;?&quot;. Numeric entries (integers) between 0-5 will be converted to &quot;?&quot;,&quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;,&quot;-&quot;, respectively. Invalid characters will be replaced with &quot;?&quot; with a warning message.
</p>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>

<hr>
<h2 id='dna.obj'>Example DNA sequence data</h2><span id='topic+dna.obj'></span>

<h3>Description</h3>

<p>An example object of the class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dna.obj)</code></pre>


<h3>Format</h3>

<p><code>dna.obj</code> contains a <code><a href="haplotypes.html#topic+Dna">Dna</a></code> object. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dna.obj)
dna.obj
</code></pre>

<hr>
<h2 id='grouping-methods'>Groups haplotypes according to the grouping variable (populations, species, etc.)
</h2><span id='topic+grouping'></span><span id='topic+grouping-methods'></span><span id='topic+grouping+2CHaplotype-method'></span>

<h3>Description</h3>

<p>Function for creating a matrix with haplotypes as rows, grouping factor (populations, species, etc.) as columns and abundance as entries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Haplotype'
grouping(x,factors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grouping-methods_+3A_x">x</code></td>
<td>
<p>an object of class  <code><a href="haplotypes.html#topic+Haplotype">Haplotype</a></code>. </p>
</td></tr>
<tr><td><code id="grouping-methods_+3A_factors">factors</code></td>
<td>
<p>a vector or factor giving the grouping variable (populations, species, etc.), with one element per individual. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with two components: 
</p>

<dl>
<dt><code>hapmat</code>:</dt><dd><p>a matrix with haplotypes as rows, levels of the grouping factor (populations, species, etc.) as columns and abundance as entries.</p>
</dd>
<dt><code>hapvec</code>:</dt><dd><p>a vector giving the haplotype identities of individuals. </p>
</dd>
</dl>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Haplotype")</code></dt><dd>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+haplotype">haplotype</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x&lt;-dna.obj[1:6,,as.matrix=FALSE]
# inferring haplotypes from DNA sequences
h&lt;-haplotype(x) 

## Grouping haplotypes.
# character vector 'populations' is a grouping factor.
populations&lt;-c("pop1","pop1","pop2","pop3","pop3","pop3") 

# length of the argument 'factor' is equal to the number of sequences
g&lt;-grouping(h,factors=populations) 
g

</code></pre>

<hr>
<h2 id='Haplotype-class'>Class <code>"Haplotype"</code> in the Package <span class="pkg">haplotypes</span></h2><span id='topic+Haplotype'></span><span id='topic+Haplotype-class'></span>

<h3>Description</h3>

<p>S4 class to store haplotype information.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("Haplotype", haplist, hapind, uniquehapind, sequence, d, freq, nhap)</code>,  however use function <code><a href="haplotypes.html#topic+haplotype">haplotype</a></code>  instead.
</p>


<h3>Slots</h3>


<dl>
<dt><code>haplist</code>:</dt><dd><p>Object of class <code>"list"</code>, containing the names of individuals that share the same haplotype. </p>
</dd>
<dt><code>hapind</code>:</dt><dd><p>Object of class <code>"list"</code>, containing the index of individuals that share the same haplotype. </p>
</dd>
<dt><code>uniquehapind</code>:</dt><dd><p>Object of class <code>"numeric"</code>, containing the index of the first occurrence of unique haplotypes. </p>
</dd>
<dt><code>sequence</code>:</dt><dd><p>Object of class <code>"matrix"</code> if present, giving the DNA sequence matrix of unique haplotypes. </p>
</dd>
<dt><code>d</code>:</dt><dd><p>Object of class <code>"matrix"</code>, giving the absolute pairwise character difference matrix of unique haplotypes. </p>
</dd>
<dt><code>freq</code>:</dt><dd><p>Object of class <code>"numeric"</code>, giving the haplotype frequencies. </p>
</dd>
<dt><code>nhap</code>:</dt><dd><p>Object of class <code>"numeric"</code>, giving the total number of haplotypes.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>as.dna</dt><dd><p><code>signature(x = "Haplotype")</code>: if Haplotype object contains dna sequences, coerces an object of class Haplotype to an object of class Dna, else returns an error message.</p>
</dd>
<dt>as.list</dt><dd><p><code>signature(x = "Haplotype")</code>: assigns slots of an object Haplotype to list elements.</p>
</dd>
<dt>grouping</dt><dd><p><code>signature(x = "Haplotype")</code>: creates a matrix with haplotypes as rows, grouping factor (populations, species, etc.) as columns and abundance as entries. </p>
</dd>
<dt>hapreord</dt><dd><p><code>signature(x = "Haplotype")</code>: reorders haplotypes according to the ordering factor.</p>
</dd>
<dt>length</dt><dd><p><code>signature(x = "Haplotype")</code>: returns the number of haplotypes. </p>
</dd>
<dt>pieplot</dt><dd><p><code>signature(x = "Parsimnet", y = "Haplotype")</code>: plot pie charts on statistical parsimony network.</p>
</dd>
<dt>pielegend</dt><dd><p><code>signature(x = "Parsimnet", y = "Haplotype")</code>: add legends to pie charts produced using <code><a href="haplotypes.html#topic+pieplot">pieplot</a></code>.</p>
</dd>
<dt>show</dt><dd><p><code>signature(object = "Haplotype")</code>: displays the object briefly. </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>

<hr>
<h2 id='haplotype-methods'>Methods for function <code>haplotype</code> in the package <span class="pkg">haplotypes</span></h2><span id='topic+haplotype'></span><span id='topic+haplotype-methods'></span><span id='topic+haplotype+2CDna-method'></span><span id='topic+haplotype+2Cdist-method'></span><span id='topic+haplotype+2Cmatrix-method'></span>

<h3>Description</h3>

<p>Collapses identical DNA sequences into haplotypes or infering haplotypes using user provided absolute pairwise character difference matrix. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
haplotype(x,indels="sic")
## S4 method for signature 'dist'
haplotype(x)
## S4 method for signature 'matrix'
haplotype(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplotype-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>, <code>dist</code>, or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="haplotype-methods_+3A_indels">indels</code></td>
<td>
<p>the indel coding method to be used. This must be one of &quot;sic&quot;, &quot;5th&quot; or &quot;missing&quot;. Any unambiguous substring can be given.  See <code><a href="haplotypes.html#topic+distance">distance</a></code> for details. </p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>haplotype</code> returns an object of class <code><a href="haplotypes.html#topic+Haplotype">Haplotype</a></code>, <code><a href="haplotypes.html#topic+as.list-methods">as.list-methods</a></code> can be used to coerce the object to a list.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>Inferring haplotypes from DNA sequences.
</p>
</dd>
<dt><code>signature(x = "dist")</code></dt><dd>
<p>Inferring haplotypes using an absolute pairwise character difference matrix (dist object).
</p>
</dd>
<dt><code>signature(x = "matrix")</code></dt><dd>
<p>Inferring haplotypes using an absolute pairwise character difference matrix.
</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	
data("dna.obj")
x&lt;-dna.obj[1:6,,as.matrix=FALSE]

##Inferring haplotypes using 'Dna' object.
# coding gaps using simple indel coding method
h&lt;-haplotype(x,indels="sic")
h

# giving DNA sequences of haplotypes
as.dna(h) 

## Not run: 
	
## Slots of an object Haplotype	
h@haplist  #haplotype list (names)
h@hapind  #haplotype list (index)
h@uniquehapind  #getting index of the first occurrence of haplotypes
h@sequence  #DNA sequences of haplotypes
h@d  #distance matrix of haplotypes
h@freq  #haplotype frequencies
h@nhap  #total number of haplotypes

## End(Not run)

## Inferring haplotypes using dist object. 
d&lt;-distance(x)
h&lt;-haplotype(d) 
h
## Not run: 
# returns an error message 
as.dna(h)

## End(Not run) 

## Inferring haplotypes using distance matrix.
d&lt;-as.matrix(distance(x))
h&lt;-haplotype(d) 
h
## Not run: 
# returns an error message
as.dna(h)

## End(Not run)  

</code></pre>

<hr>
<h2 id='haplotypes-internal'>Internal function(s)</h2><span id='topic+add.miss'></span><span id='topic+alltest'></span><span id='topic+as.sequence'></span><span id='topic+calcclust'></span><span id='topic+clsfyindels'></span><span id='topic+clustersG'></span><span id='topic+contr'></span><span id='topic+distance2'></span><span id='topic+ext.nodes'></span><span id='topic+fillendgaps'></span><span id='topic+fillna'></span><span id='topic+geodist.corr'></span><span id='topic+haplo'></span><span id='topic+has.cycle'></span><span id='topic+indtest'></span><span id='topic+multipolypoz'></span><span id='topic+pairneidist'></span><span id='topic+pairPhiSTdist'></span><span id='topic+parsnet'></span><span id='topic+pegas.amova'></span><span id='topic+polys'></span><span id='topic+recdistG'></span><span id='topic+ren.dupli'></span><span id='topic+ri'></span><span id='topic+sic'></span><span id='topic+steplimit'></span><span id='topic+tryconnect'></span><span id='topic+tryconnectG'></span><span id='topic+validity'></span><span id='topic+xhaps'></span><span id='topic+sic'></span><span id='topic+validity'></span><span id='topic+initialize-methods'></span><span id='topic+initialize+2CDna-method'></span><span id='topic+initialize+2CHaplotype-method'></span><span id='topic+initialize+2CParsimnet-method'></span>

<h3>Description</h3>

<p>Internal function(s)</p>

<hr>
<h2 id='haplotypes-package'>
Manipulating DNA Sequences and Estimating Unambiguous Haplotype Network with Statistical Parsimony
</h2><span id='topic+haplotypes-package'></span><span id='topic+haplotypes'></span>

<h3>Description</h3>

<p>This package provides S4 classes and methods for reading and manipulating aligned DNA sequences, supporting an indel coding methods (only simple indel coding method is available in the current version), showing base substitutions and indels, calculating absolute pairwise distances between DNA sequences, and collapses identical DNA sequences into haplotypes or infering haplotypes using user provided absolute pairwise character difference matrix.  This package also includes S4 classes and methods for estimating genealogical relationships among haplotypes using statistical parsimony and plotting parsimony networks. 
</p>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Read example FASTA file.	
f&lt;-system.file("example.fas",package="haplotypes") 
# invalid character 'N' was replaced with '?' with a warning message
x&lt;-read.fas(file=f)
# an object of class 'Dna'
x 

## or load DNA Sequence data set.
data("dna.obj") 
x&lt;-dna.obj
## Not run: 
x

## End(Not run)

## Compute an absolute pairwise character difference matrix from DNA sequences.
# coding gaps using simple indel coding method
d&lt;- distance(x,indels="sic") 
## Not run: 
d

## End(Not run)

## Infer haplotypes using the 'Dna' object.
# coding gaps using simple indel coding method
h&lt;-haplotype(x,indels="s") 
## Not run: 
h

## End(Not run)

## Conduct statistical parsimony analysis with 95% connection limit.
#algortihmic method
## Not run: 
p&lt;-parsimnet(x,prob=.95) 
p
# plot network
plot(p) 

## End(Not run)

## Plotting pie charts on the statistical parsimony network
## Not run: 
data("dna.obj")
x&lt;-dna.obj
h&lt;-haplotypes::haplotype(x)

## Statistical parsimony with 95
p&lt;-parsimnet(x) 

#randomly generated populations
pop&lt;-c("pop1","pop2","pop3","pop4","pop5","pop6","pop7","pop8")
set.seed(5)
pops&lt;-sample(pop,nrow(x),replace=TRUE)

# Plotting with default parameters.
pieplot(p,h,1, pops)


## End(Not run)
</code></pre>

<hr>
<h2 id='hapreord-methods'>Reorders haplotypes according to the ordering factor</h2><span id='topic+hapreord'></span><span id='topic+hapreord-methods'></span><span id='topic+hapreord+2CHaplotype-method'></span>

<h3>Description</h3>

<p>Reorders haplotypes according to the ordering factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Haplotype'
hapreord(x,order=c(1:x@nhap))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hapreord-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Haplotype">Haplotype</a></code>. </p>
</td></tr>
<tr><td><code id="hapreord-methods_+3A_order">order</code></td>
<td>
<p>a vector giving the order of haplotypes, with one element per haplotype. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns an object of class <code><a href="haplotypes.html#topic+Haplotype">Haplotype</a></code>.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Haplotype")</code></dt><dd>
<p>Reorders haplotypes.
</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+haplotype">haplotype</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x&lt;-dna.obj[1:6,,as.matrix=FALSE]
# inferring haplotypes from DNA sequences
h&lt;-haplotype(x)

## Reordering haplotypes.

# length of the argument 'order' is equal to the number of haplotypes
rh&lt;-hapreord(h,order=c(4,3,1,2))
rh

</code></pre>

<hr>
<h2 id='homopoly-methods'>Provides the list of homoplastic indels and substitutions</h2><span id='topic+homopoly'></span><span id='topic+homopoly-methods'></span><span id='topic+homopoly+2CDna-method'></span>

<h3>Description</h3>

<p>This function returs the list of homoplastic indels and substitutions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
homopoly(x,indels="sic",...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="homopoly-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
<tr><td><code id="homopoly-methods_+3A_indels">indels</code></td>
<td>
<p>the indel coding method to be used. This must be one of &quot;sic&quot;, &quot;5th&quot; or &quot;missing&quot;. Any unambiguous substring can be given.  See <code><a href="haplotypes.html#topic+distance">distance</a></code> for details. </p>
</td></tr>
<tr><td><code id="homopoly-methods_+3A_...">...</code></td>
<td>
<p>additional arguments to <code><a href="haplotypes.html#topic+parsimnet">parsimnet</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with following components: 
</p>
<table>
<tr><td><code>indels</code></td>
<td>
<p>a character vector of homoplastic indels sitewise Consistency Index, names of the character vector gives the site of homoplastic indel.</p>
</td></tr>
<tr><td><code>subs</code></td>
<td>
<p>a character vector of homoplastic substitutions sitewise Consistency Index, names of the character vector gives the site of substitution.</p>
</td></tr>
</table>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")

### Method for signature 'Dna'.
x&lt;-dna.obj
homopoly(x)

</code></pre>

<hr>
<h2 id='image-methods'> Display DNA Sequence </h2><span id='topic+image-methods'></span><span id='topic+image+2CDna-method'></span>

<h3>Description</h3>

<p>Display an image of DNA sequences .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
image(x,all=FALSE,fifth=TRUE,
col=c("#BFBFBF","#0B99FD","#FD0B0B","#11A808","#F5FD0B","#F8F8FF"),
chars=TRUE,cex=1,show.names=TRUE,show.sites=TRUE,xlab="",ylab="",...)
 </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="image-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
<tr><td><code id="image-methods_+3A_all">all</code></td>
<td>
<p>boolean; should entire sequence be displayed or only the polymorphic sites? </p>
</td></tr>
<tr><td><code id="image-methods_+3A_fifth">fifth</code></td>
<td>
<p>boolean; if all==FALSE, should gaps be displayed? </p>
</td></tr>
<tr><td><code id="image-methods_+3A_col">col</code></td>
<td>
<p> an integer or character vector for the colors. By default it is blue for &quot;A&quot;, red for &quot;C&quot;, green for &quot;G&quot;, yellow for &quot;T&quot;, white for &quot;-&quot;,  and grey for &quot;?&quot;.</p>
</td></tr>
<tr><td><code id="image-methods_+3A_chars">chars</code></td>
<td>
<p>boolean; should characters be displayed on image?</p>
</td></tr>
<tr><td><code id="image-methods_+3A_cex">cex</code></td>
<td>
<p>a numeric vector of expansion factor for characters. </p>
</td></tr>	
<tr><td><code id="image-methods_+3A_show.names">show.names</code></td>
<td>
<p>boolean; should sequence names be displayed on the left side.</p>
</td></tr>
<tr><td><code id="image-methods_+3A_show.sites">show.sites</code></td>
<td>
<p>boolean; should site labels be displayed on the bottom side.</p>
</td></tr>
<tr><td><code id="image-methods_+3A_xlab">xlab</code></td>
<td>
<p>a title for the x axis.</p>
</td></tr>
<tr><td><code id="image-methods_+3A_ylab">ylab</code></td>
<td>
<p>a title for the y axis.</p>
</td></tr>
<tr><td><code id="image-methods_+3A_...">...</code></td>
<td>
<p>additional arguments to <code><a href="graphics.html#topic+image.default">image.default</a></code>.</p>
</td></tr>
</table>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>Display an image of Dna objects
</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>. 
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+image.default">image.default</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x&lt;-dna.obj

## Display only polymorphic sites without gaps
image(x,all=FALSE,fifth=FALSE,show.names=TRUE,cex=0.6)

## Display only polymorphic sites with gaps
image(x,all=FALSE,fifth=TRUE,show.names=TRUE,cex=0.6)

## Not run: 
## Display entire sequences
image(x,all=FALSE,show.names=TRUE,cex=0.6)

## End(Not run)
</code></pre>

<hr>
<h2 id='indelcoder-methods'>Codes gaps</h2><span id='topic+indelcoder'></span><span id='topic+indelcoder-methods'></span><span id='topic+indelcoder+2CDna-method'></span>

<h3>Description</h3>

<p>Function for coding gaps separately. Only simple indel coding method is available in the current version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
indelcoder(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="indelcoder-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with two components: 
</p>

<dl>
<dt><code>indels</code>:</dt><dd><p>a matrix giving the indel positions (beginnings and ends) and lengths.</p>
</dd>
<dt><code>codematrix</code>:</dt><dd><p>a binary matrix giving the indel codings. Missing values are denoted by -1.</p>
</dd>
</dl>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>Function for coding gaps separately.
</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>References</h3>

<p>Simmons, M., Ochoterena, H. (2000) Gaps as characters in sequence-based phylogenetic analyses. <em>Systematic Biology</em> <b>49</b>, 369-381.		
</p>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+distance">distance</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x&lt;-dna.obj

## Simple indel coding.
indelcoder(x)

</code></pre>

<hr>
<h2 id='length-methods'>Methods for function <code>length</code> in the package <span class="pkg">haplotypes</span></h2><span id='topic+length-methods'></span><span id='topic+length+2CDna-method'></span><span id='topic+length+2CHaplotype-method'></span><span id='topic+length+2CParsimnet-method'></span>

<h3>Description</h3>

<p>Methods for function <code>length</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
length(x)
## S4 method for signature 'Haplotype'
length(x)
## S4 method for signature 'Parsimnet'
length(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="length-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>, <code><a href="haplotypes.html#topic+Haplotype">Haplotype</a></code> or <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a non-negative integer vector.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>returns the longest sequence length.
</p>
</dd>
<dt><code>signature(x = "Haplotype")</code></dt><dd>
<p>returns the number of haplotypes.
</p>
</dd>
<dt><code>signature(x = "Parsimnet")</code></dt><dd>
<p>returns the length of network(s). 
</p>
</dd>
</dl>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+ncol-methods">ncol-methods</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x&lt;-dna.obj

## Longest sequence length
length(x)


## Total number of haplotypes
h&lt;-haplotype(x)
length(h)

## Length of network(s)
p&lt;-parsimnet(x,prob=.95)
# length of the network
length(p)
 
p&lt;-parsimnet(x,prob=.99)
# length of the networks
length(p)

</code></pre>

<hr>
<h2 id='names-methods'>Function to get or set names of a <code>Dna</code> object or <code>Parsimnet</code> object</h2><span id='topic+names-methods'></span><span id='topic+names+3C--methods'></span><span id='topic+names+2CDna-method'></span><span id='topic+names+3C-+2CDna-method'></span><span id='topic+names+2CParsimnet-method'></span><span id='topic+names+3C-+2CParsimnet-method'></span>

<h3>Description</h3>

<p>Function to get or set sequence names of <code>Dna</code> object or names of network in <code>Parsimnet</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
names(x)
## S4 method for signature 'Parsimnet'
names(x)
## S4 replacement method for signature 'Dna'
names(x)&lt;-value
## S4 replacement method for signature 'Parsimnet'
names(x)&lt;-value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="names-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code> or <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code>. </p>
</td></tr>
<tr><td><code id="names-methods_+3A_value">value</code></td>
<td>
<p>a character vector of the same length as number of sequence or networks.</p>
</td></tr>
</table>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>Function to get or set names of an object of Dna.
</p>
</dd>
<dt><code>signature(x = "Parsimnet")</code></dt><dd>
<p>Function to get or set names of networks in Parsimnet object.
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>
data("dna.obj")

x&lt;-dna.obj
x&lt;-as.dna(x[1:4,1:6])

## Getting sequence names.
names(x)

## Setting sequence names.
names(x)&lt;-c("u","v","z","y")
names(x)



x&lt;-dna.obj
p&lt;-parsimnet(x,prob=.99)

##Getting network names in parsimnet object
names(p)
## Setting network names names.
names(p)&lt;-c("a","b","c","d","f","g")
names(p)

</code></pre>

<hr>
<h2 id='ncol-methods'>Returns the length of the longest DNA sequence</h2><span id='topic+ncol-methods'></span><span id='topic+ncol+2CDna-method'></span>

<h3>Description</h3>

<p><code>ncol</code> returns the number of columns present in a matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
ncol(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ncol-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an integer of length one.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p><code>ncol</code> returns the number of columns present in the sequence matrix (length of the longest DNA sequence).
</p>
</dd>
</dl>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+length-methods">length-methods</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x &lt;-dna.obj

## Giving the length of the longest sequence.
ncol(x) 
# gives the same result
length(x) 
</code></pre>

<hr>
<h2 id='nrow-methods'>Returns the number of DNA sequences</h2><span id='topic+nrow-methods'></span><span id='topic+nrow+2CDna-method'></span>

<h3>Description</h3>

<p><code>nrow</code> returns the number of rows present in a matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
nrow(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nrow-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an integer of length one.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p><code>nrow</code> returns the number of rows present in the sequence matrix (number of sequences).
</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x &lt;-dna.obj

## Giving the number of sequences.
nrow(x)

</code></pre>

<hr>
<h2 id='pairnei-methods'>Provides the average number of pairwise Nei's (D) differences between populations</h2><span id='topic+pairnei'></span><span id='topic+pairnei-methods'></span><span id='topic+pairnei+2Cdist-method'></span><span id='topic+pairnei+2CDna-method'></span><span id='topic+pairnei+2Cmatrix-method'></span>

<h3>Description</h3>

<p>Function provides pairwise Nei's raw number of nucleotide differences between populations. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
pairnei(x,populations,indels="sic",nperm=99, subset=NULL,showprogbar=FALSE)
## S4 method for signature 'dist'
pairnei(x,populations,nperm=99, subset=NULL,showprogbar=FALSE)
## S4 method for signature 'matrix'
pairnei(x,populations,nperm=99, subset=NULL,showprogbar=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairnei-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>, <code>"dist"</code> or <code>"matrix"</code>.</p>
</td></tr>
<tr><td><code id="pairnei-methods_+3A_populations">populations</code></td>
<td>
<p>a vector giving the populations, with one element per individual.</p>
</td></tr>
<tr><td><code id="pairnei-methods_+3A_indels">indels</code></td>
<td>
<p>the indel coding method to be used. This must be one of &quot;sic&quot;, &quot;5th&quot; or &quot;missing&quot;. Any unambiguous substring can be given.  See <code><a href="haplotypes.html#topic+distance">distance</a></code> for details. </p>
</td></tr>
<tr><td><code id="pairnei-methods_+3A_nperm">nperm</code></td>
<td>
<p>the number of permutations. Set this to 0 to skip the permutation procedure.</p>
</td></tr>
<tr><td><code id="pairnei-methods_+3A_subset">subset</code></td>
<td>
<p>a vector of integers in the range [1, <code>length(unique(populations))</code>], only distances between selected population(s) and the rest of the populations are calculated. If it is NULL, all comparisons are done.</p>
</td></tr>
<tr><td><code id="pairnei-methods_+3A_showprogbar">showprogbar</code></td>
<td>
<p>boolean; whether the progress bar is displayed or not  displayed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The null distribution of pairwise Nei's differences under the hypothesis of no difference between the populations is obtained by permuting individuals between populations.  
</p>


<h3>Value</h3>

<p>a list with following components: 
</p>
<table>
<tr><td><code>neidist</code></td>
<td>
<p>a matrix giving the average number of pairwise Nei's (D) differences between populations (below diagonal elements) and  average number of pairwise differences within populations (diagonal elements).</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>a matrix giving the p-values, or NULL if permutation test is not performed.</p>
</td></tr>
</table>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
</dd>
<dt><code>signature(x = "dist")</code></dt><dd>
</dd>
<dt><code>signature(x = "matrix")</code></dt><dd>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>References</h3>

<p>Nei, M. and Li, W. H. (1979) Mathematical model for studying genetic variation in terms of restriction endonucleases. <em>Proceedings of the National Academy of Sciences of the United States of America </em> <b>76</b>, 5269-5273.		
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")


### Method for signature 'Dna'.
x&lt;-dna.obj
x&lt;-dna.obj[c(1,20,21,26,27,28,30,3,4,7,13,14,15,16,23,24,25),,as.matrix=FALSE]
populations&lt;-c("pop1","pop1","pop1","pop1","pop1","pop1","pop1","pop2",
"pop2","pop2","pop2","pop2","pop3","pop4","pop3","pop4","pop4")

##skip permutation testing
pn&lt;-pairnei(x, populations, nperm=0)
pn

#Between populations
as.dist(pn$neidist)

#Within populations
diag(pn$neidist)

##Gaps as missing characters.
pn &lt;-pairnei(x, populations, indels="m", nperm=0)
pn

##using subset, third population against others
pn&lt;-pairnei(x, populations, nperm=0,subset=c(3))
pn

## Not run: 
## 999 permutations.
pn&lt;-pairnei(x, populations, nperm=999, showprogbar=TRUE)
pn

## random populations
x&lt;-dna.obj
populations&lt;-sample(1:4,nrow(x),replace=TRUE)
pn&lt;-pairnei(x, populations, nperm=999, showprogbar=TRUE)
pn

## populations based on clusters
x&lt;-dna.obj
d&lt;-distance(x)
hc&lt;-hclust(d,method="ward.D")
populations&lt;-cutree(hc,4)
pn&lt;-pairnei(x, populations, nperm=999, showprogbar=TRUE)
pn

## End(Not run)

### Method for signature 'dist'.
x&lt;-dna.obj
x&lt;-dna.obj[c(1,20,21,26,27,28,30,3,4,7,13,14,15,16,23,24,25),,as.matrix=FALSE]
populations&lt;-c("pop1","pop1","pop1","pop1","pop1","pop1","pop1","pop2",
"pop2","pop2","pop2","pop2","pop3","pop4","pop3","pop4","pop4")
d&lt;-distance(x)
pn&lt;-pairnei(d, populations,nperm=0)
pn

### Method for signature 'matrix'.
x&lt;-dna.obj
x&lt;-dna.obj[c(1,20,21,26,27,28,30,3,4,7,13,14,15,16,23,24,25),,as.matrix=FALSE]
populations&lt;-c("pop1","pop1","pop1","pop1","pop1","pop1","pop1","pop2",
"pop2","pop2","pop2","pop2","pop3","pop4","pop3","pop4","pop4")
d&lt;-as.matrix(distance(x))
pn&lt;-pairnei(d, populations,nperm=0)
pn

</code></pre>

<hr>
<h2 id='pairPhiST-methods'> Provides the pairwise PhiST between populations</h2><span id='topic+pairPhiST'></span><span id='topic+pairPhiST-methods'></span><span id='topic+pairPhiST+2Cdist-method'></span><span id='topic+pairPhiST+2CDna-method'></span><span id='topic+pairPhiST+2Cmatrix-method'></span>

<h3>Description</h3>

<p>This function calculates pairwise PhiST between populations using the AMOVA framework.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
pairPhiST(x,populations,indels="sic",nperm=99, negatives=FALSE, subset =NULL, 
showprogbar=TRUE)
## S4 method for signature 'dist'
pairPhiST(x,populations,nperm=99, negatives=FALSE, subset=NULL,showprogbar=TRUE)
## S4 method for signature 'matrix'
pairPhiST(x,populations,nperm=99,negatives=FALSE, subset=NULL,showprogbar=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairPhiST-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>, <code>"dist"</code> or <code>"matrix"</code>.</p>
</td></tr>
<tr><td><code id="pairPhiST-methods_+3A_populations">populations</code></td>
<td>
<p>a vector giving the populations, with one element per individual.</p>
</td></tr>
<tr><td><code id="pairPhiST-methods_+3A_indels">indels</code></td>
<td>
<p>the indel coding method to be used. This must be one of &quot;sic&quot;, &quot;5th&quot; or &quot;missing&quot;. Any unambiguous substring can be given.  See <code><a href="haplotypes.html#topic+distance">distance</a></code> for details. </p>
</td></tr>
<tr><td><code id="pairPhiST-methods_+3A_nperm">nperm</code></td>
<td>
<p>the number of permutations. Set this to 0 to skip the permutation procedure.</p>
</td></tr>
<tr><td><code id="pairPhiST-methods_+3A_negatives">negatives</code></td>
<td>
<p>boolean; if it is FALSE all negative PhiST values are replaced with zero. </p>
</td></tr>
<tr><td><code id="pairPhiST-methods_+3A_subset">subset</code></td>
<td>
<p>a vector of integers in the range [1, <code>length(unique(populations))</code>], only distances between selected population(s) and the rest of the populations are calculated. If it is NULL, all comparisons are done.</p>
</td></tr>
<tr><td><code id="pairPhiST-methods_+3A_showprogbar">showprogbar</code></td>
<td>
<p>boolean; whether the progress bar is displayed or not  displayed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The null distribution of pairwise PhiST under the hypothesis of no difference between the populations is obtained by permuting individuals between populations.   
</p>


<h3>Value</h3>

<p>a list with following components: 
</p>
<table>
<tr><td><code>PhiST</code></td>
<td>
<p>a matrix giving the PhiST values between populations.</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>a matrix giving the p-values, or NULL if permutation test is not performed.</p>
</td></tr>
</table>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
</dd>
<dt><code>signature(x = "dist")</code></dt><dd>
</dd>
<dt><code>signature(x = "matrix")</code></dt><dd>
</dd>
</dl>


<h3>Note</h3>

<p>An internal code <code>Statphi</code> is taken from package  <span class="pkg">ade4</span> version 1.7-8 without any modification, author Sandrine Pavoine. Function <code>amova</code> from package <span class="pkg">pegas</span> is used internally to estimate variance components, author Emmanuel Paradis.
</p>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>References</h3>

<p>Excoffier, L., Smouse, P.E. and Quattro, J.M. (1992) Analysis of molecular variance inferred
from metric distances among DNA haplotypes: application to human mitochondrial DNA restriction
data. <em>Genetics</em>, <b>131</b>, 479-491.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")

### Method for signature 'Dna'.
x&lt;-dna.obj
x&lt;-dna.obj[c(1,20,21,26,27,28,30,3,4,7,13,14,15,16,23,24,25),,as.matrix=FALSE]
populations&lt;-c("pop1","pop1","pop1","pop1","pop1","pop1","pop1","pop2",
"pop2","pop2","pop2","pop2","pop3","pop4","pop3","pop4","pop4")

##skip permutation testing
pst&lt;-pairPhiST(x, populations, nperm=0)
pst

##allow negative PhiST values
pst&lt;-pairPhiST(x, populations, nperm=0, negatives=TRUE)
pst

##Gaps as missing characters.
pst&lt;-pairPhiST(x, populations, indels="m", nperm=0, negatives=TRUE)
pst

##using subset, second population against others
pst &lt;-pairPhiST(x, populations, nperm=0,subset=c(2))
pst


## Not run: 
## 999 permutations.
pst&lt;-pairPhiST(x, populations, nperm=999,showprogbar=TRUE)
pst

## random populations
x&lt;-dna.obj
populations&lt;-sample(1:4,nrow(x),replace=TRUE)
pst&lt;-pairPhiST(x, populations, nperm=999,showprogbar=TRUE)
pst

## populations based on clusters
x&lt;-dna.obj
d&lt;-distance(x)
hc&lt;-hclust(d,method="ward.D")
populations&lt;-cutree(hc,4)
pst&lt;-pairPhiST(x, populations, nperm=999,showprogbar=TRUE)
pst

## End(Not run)

### Method for signature 'dist'.
x&lt;-dna.obj
x&lt;-dna.obj[c(1,20,21,26,27,28,30,3,4,7,13,14,15,16,23,24,25),,as.matrix=FALSE]
populations&lt;-c("pop1","pop1","pop1","pop1","pop1","pop1","pop1","pop2",
"pop2","pop2","pop2","pop2","pop3","pop4","pop3","pop4","pop4")
d&lt;-distance(x)
pst&lt;-pairPhiST(d, populations, nperm=0)
pst

### Method for signature 'matrix'.
x&lt;-dna.obj
x&lt;-dna.obj[c(1,20,21,26,27,28,30,3,4,7,13,14,15,16,23,24,25),,as.matrix=FALSE]
populations&lt;-c("pop1","pop1","pop1","pop1","pop1","pop1","pop1","pop2",
"pop2","pop2","pop2","pop2","pop3","pop4","pop3","pop4","pop4")
d&lt;-as.matrix(distance(x))
pst&lt;-pairPhiST(d, populations, nperm=0)
pst

</code></pre>

<hr>
<h2 id='Parsimnet-class'>Class <code>"Parsimnet"</code> in the Package <span class="pkg">haplotypes</span> </h2><span id='topic+Parsimnet'></span><span id='topic+Parsimnet-class'></span>

<h3>Description</h3>

<p>S4 class to store statistical parsimony networks and additional information.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("Parsimnet", d, tempProbs, conlimit, prob, nhap, rowindex), however use function <a href="haplotypes.html#topic+parsimnet">parsimnet</a> instead</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>d</code>:</dt><dd><p>Object of class <code>"list"</code> containing the geodesic distance matrix of haplotypes and intermediates for each network. </p>
</dd>
<dt><code>tempProbs</code>:</dt><dd><p>Object of class <code>"numeric"</code> giving the probabilities of parsimony for mutational steps beyond the connection limit.  </p>
</dd>
<dt><code>conlimit</code>:</dt><dd><p>Object of class <code>"numeric"</code> giving the number of maximum connection steps at connection limit.</p>
</dd>
<dt><code>prob</code>:</dt><dd><p>Object of class <code>"numeric"</code> giving the user defined connection limit.  </p>
</dd>
<dt><code>nhap</code>:</dt><dd><p>Object of class <code>"numeric"</code> giving the number of haplotypes in each network.</p>
</dd>
<dt><code>rowindex</code>:</dt><dd><p>Object of class <code>"list"</code> containing vectors giving the index of haplotypes in each network. </p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>as.list</dt><dd><p><code>signature(x = "Parsimnet")</code>: assigns slots of an object Parsimnet to list elements.</p>
</dd>
<dt>as.network</dt><dd><p><code>signature(x = "Parsimnet")</code>: coerces <code>Parsimnet</code> object to <code><a href="network.html#topic+network">network</a></code> <code>{network}</code> object </p>
</dd>
<dt>as.networx</dt><dd><p><code>signature(x = "Parsimnet")</code>: coerces <code>Parsimnet</code> object to <code><a href="phangorn.html#topic+networx">networx</a></code> <code>{phangorn}</code> object  </p>
</dd>
<dt>length</dt><dd><p><code>signature(x = "Parsimnet")</code>: returns the length of network(s). </p>
</dd>
<dt>names</dt><dd><p><code>signature(x = "Parsimnet")</code>: gets names of networks in <code>Parsimnet</code> object </p>
</dd>
<dt>names&lt;-</dt><dd><p><code>signature(x = "Parsimnet")</code>: sets names of networks in <code>Parsimnet</code> object </p>
</dd>
<dt>plot</dt><dd><p><code>signature(x = "Parsimnet")</code>: plots statistical parsimony networks.</p>
</dd>
<dt>pieplot</dt><dd><p><code>signature(x = "Parsimnet", y = "Haplotype")</code>: plots pie charts on statistical parsimony networks </p>
</dd>
<dt>pielegend</dt><dd><p><code>signature(x = "Parsimnet", y = "Haplotype")</code>: add legends to pie charts produced using <code><a href="haplotypes.html#topic+pieplot">pieplot</a></code>.</p>
</dd>
<dt>rownames</dt><dd><p><code>signature(x = "Parsimnet")</code>: gets names of vertices in networks. </p>
</dd>
<dt>rownames&lt;-</dt><dd><p><code>signature(x = "Parsimnet")</code>: gets names of vertices in networks </p>
</dd>
<dt>show</dt><dd><p><code>signature(object = "Parsimnet")</code>: displays the object briefly. </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>

<hr>
<h2 id='parsimnet-methods'>Estimates gene genealogies using statistical parsimony</h2><span id='topic+parsimnet'></span><span id='topic+parsimnet-methods'></span><span id='topic+parsimnet+2CDna-method'></span><span id='topic+parsimnet+2Cdist-method'></span><span id='topic+parsimnet+2Cmatrix-method'></span>

<h3>Description</h3>

<p>Function for estimating gene genealogies from DNA sequences or user provided absolute pairwise character difference matrix using statistical parsimony.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
parsimnet(x,indels="sic",prob=.95)
## S4 method for signature 'dist'
parsimnet(x,seqlength,prob=.95)
## S4 method for signature 'matrix'
parsimnet(x,seqlength,prob=.95)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parsimnet-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>, dist, or matrix.</p>
</td></tr>
<tr><td><code id="parsimnet-methods_+3A_indels">indels</code></td>
<td>
<p>the indel coding method to be used. This must be one of &quot;sic&quot;, &quot;5th&quot; or &quot;missing&quot;. Any unambiguous substring can be given.  See <code><a href="haplotypes.html#topic+distance">distance</a></code> for details.</p>
</td></tr>
<tr><td><code id="parsimnet-methods_+3A_seqlength">seqlength</code></td>
<td>
<p>an integer of length one giving the sequence length information (number of characters). </p>
</td></tr>
<tr><td><code id="parsimnet-methods_+3A_prob">prob</code></td>
<td>
<p>a numeric vector of length one in the range [0.01, 0.99] giving the probability of parsimony as defined in Templeton et al. (1992). In order to set maximum connection steps to Inf (to connect all the haplotypes in a single network), set the probability to NULL. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The network estimation methods implemented in  <code>parsimnet</code> function finds one of the most parsimonious network (or sub-networks if connection between haplotypes exceeds the parsimony limit).  This is an implemetation of the TCS method proposed in Templeton et al. (1992) and Clement et al. (2002).<code>parsimnet</code> function generates an unambiguous haplotype network without loops. If more than one best networks found (results in ambiguous connections), only a network with the lowest average all-pairs distance is returned. Loops may occur only if they are present in initial haplotype distance matrix.
</p>


<h3>Value</h3>

<p>S4 methods for signature 'Dna', 'matrix' or 'dist' returns an object of class <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code>.</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>estimating gene genealogies from DNA sequences.
</p>
</dd>
<dt><code>signature(x = "dist")</code></dt><dd>
<p>estimating gene genealogies from distance matrix (dist object).
</p>
</dd>
<dt><code>signature(x = "matrix")</code></dt><dd>
<p>estimating gene genealogies from distance matrix.
</p>
</dd>
</dl>


<h3>Note</h3>

 
<p>Duplicate names in the final distance matrices in slot d are renamed without warning. An internal function <code>.TempletonProb</code> is taken from package pegas version 0.6 without any modification, authors Emmanuel Paradis, Klaus Schliep. 
</p>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>. 
</p>


<h3>References</h3>

<p>Clement, M., Q. Snell, P. Walker, D. Posada, and K. A. Crandall (2002) TCS: Estimating Gene Genealogies  <em>in First IEEE International Workshop on High Performance Computational Biology (HiCOMB) </em> 
</p>
<p>Templeton, A. R., Crandall, K. A. and Sing, C. F. (1992) A cladistic
analysis of phenotypic associations with haplotypes inferred from
restriction endonuclease mapping and DNA sequence data. III. Cladogram
estimation. <em>Genetics</em>, <b>132</b>, 619-635.
</p>


<h3>See Also</h3>

<p><code><a href="network.html#topic+network">network</a></code>, <code><a href="stats4.html#topic+plot-methods">plot-methods</a></code> and <code><a href="haplotypes.html#topic+pieplot-methods">pieplot-methods</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	
## Not run: 
data("dna.obj")
x&lt;-dna.obj

### Method for signature 'Dna'.
## statistical parsimony with 95
p&lt;-parsimnet(x) 
p
plot(p)

## statistical parsimony with 99
p&lt;-parsimnet(x,prob=.99) 
p
# plot the first network
plot(p,net=1) 

## statistical parsimony with 99
#indels are coded as missing
p&lt;-parsimnet(x,indels="m",prob=.99)
p
plot(p)


# statistical parsimony without connection limit.
p&lt;-parsimnet(x,prob=NULL) 
p
plot(p)


# plot the first network
plot(p,net=1)

### Method for signature 'dist'.
d&lt;-distance(x)
seqlength&lt;-length(x)
## statistical parsimony with 95
p&lt;-parsimnet(d,seqlength) 
p
plot(p)

### Method for signature 'matrix'.
d&lt;-as.matrix(distance(x))
seqlength&lt;-length(x)
## statistical parsimony with 95
p&lt;-parsimnet(d,seqlength)
p
plot(p)


## End(Not run)
</code></pre>

<hr>
<h2 id='pielegend-methods'>Add Legends to Plots</h2><span id='topic+pielegend'></span><span id='topic+pielegend-methods'></span><span id='topic+pielegend+2CParsimnet+2CHaplotype-method'></span>

<h3>Description</h3>

<p>This function can be used to add legends to pie charts produced using <code><a href="haplotypes.html#topic+pieplot">pieplot</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Parsimnet,Haplotype'
pielegend(p,h,net=1,factors,...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pielegend-methods_+3A_p">p</code></td>
<td>
<p> a <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code> object. </p>
</td></tr>
<tr><td><code id="pielegend-methods_+3A_h">h</code></td>
<td>
<p> a <code><a href="haplotypes.html#topic+Haplotype">Haplotype</a></code> object.</p>
</td></tr>
<tr><td><code id="pielegend-methods_+3A_net">net</code></td>
<td>
<p>a numeric vector of length one indicating which network to plot.</p>
</td></tr>
<tr><td><code id="pielegend-methods_+3A_factors">factors</code></td>
<td>
<p>a vector or factor giving the grouping variable (populations, species, etc.), with one element per individual. </p>
</td></tr>
<tr><td><code id="pielegend-methods_+3A_...">...</code></td>
<td>
<p> arguments to be passed to <code><a href="graphics.html#topic+legend">legend</a></code> and others. See &lsquo;Details&rsquo; </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method calls <code><a href="graphics.html#topic+legend">legend</a></code> <code>{graphics}</code>, some default parameters changed:
</p>

<dl>
<dt>col</dt><dd><p> an integer or character vector for the edge colors. By default, it is <code>rainbow</code>. </p>
</dd>
<dt>fill</dt><dd><p> an integer or character vector for the filling colors. By default, it is <code>rainbow</code> </p>
</dd>
<dt>legend</dt><dd><p> a numeric or character vector to appear in the legend. By default, it is the levels of the grouping factor for haplotypes. </p>
</dd>
<dt>x</dt><dd><p> position of the legend. Default is set to <code>"topright"</code>. </p>
</dd>
</dl>



<h3>Value</h3>

<p>See &lsquo; <code><a href="graphics.html#topic+legend">legend</a></code> <code>{graphics}</code>&rsquo;
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(p = "Parsimnet", h = "Haplotype")</code></dt><dd>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>. 
</p>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+plot+2CParsimnet-method">plot,Parsimnet-method</a></code>,<code><a href="plotrix.html#topic+floating.pie">floating.pie</a></code>, <code><a href="graphics.html#topic+plot.default">plot.default</a></code>, <code><a href="network.html#topic+plot.network.default">plot.network.default</a></code> and <code><a href="graphics.html#topic+legend">legend</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x&lt;-dna.obj
h&lt;-haplotypes::haplotype(x)

### Statistical parsimony with 95% connection limit 
p&lt;-parsimnet(x) 

#randomly generated populations
pop&lt;-c("pop1","pop2","pop3","pop4","pop5","pop6","pop7","pop8")
set.seed(5)
pops&lt;-sample(pop,nrow(x),replace=TRUE)

## Plotting with default parameters.
pieplot(p,h,1, pops)

## Add legend with default parameters.
pielegend(p,h,1,pops)

## Change colors for the populations.
#8 colors for 8 populations 
cols&lt;-colors()[c(30,369,552,558,538,642,142,91)]
pieplot(p,h,1, pops,col=cols)
pielegend(p,h,1,pops,col= cols)

</code></pre>

<hr>
<h2 id='pieplot-methods'>Plots pie charts on statistical parsimony network</h2><span id='topic+pieplot'></span><span id='topic+pieplot-methods'></span><span id='topic+pieplot+2CParsimnet+2CHaplotype-method'></span>

<h3>Description</h3>

<p>Plotting pie charts on the statistical parsimony network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Parsimnet,Haplotype'
pieplot(x,y,net=1,factors, coord = NULL,inter.labels=FALSE,interactive=FALSE,rex=1,...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pieplot-methods_+3A_x">x</code></td>
<td>
<p> a <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code> object. </p>
</td></tr>
<tr><td><code id="pieplot-methods_+3A_y">y</code></td>
<td>
<p> a <code><a href="haplotypes.html#topic+Haplotype">Haplotype</a></code> object.</p>
</td></tr>
<tr><td><code id="pieplot-methods_+3A_net">net</code></td>
<td>
<p>a numeric vector of length one indicating which network to plot.</p>
</td></tr>
<tr><td><code id="pieplot-methods_+3A_factors">factors</code></td>
<td>
<p>a vector or factor giving the grouping variable (populations, species, etc.), with one element per individual. </p>
</td></tr>
<tr><td><code id="pieplot-methods_+3A_coord">coord</code></td>
<td>
<p>a matrix that contains user specified coordinates of the vertices, or NULL to  generate vertex layouts with <code><a href="network.html#topic+network.layout">network.layout</a></code> function from package <span class="pkg">network</span>. </p>
</td></tr>
<tr><td><code id="pieplot-methods_+3A_inter.labels">inter.labels</code></td>
<td>
<p>boolean; should vertex labels of intermediate haplotypes be displayed? </p>
</td></tr>
<tr><td><code id="pieplot-methods_+3A_interactive">interactive</code></td>
<td>
<p>boolean; should vertices be interactively adjusted? </p>
</td></tr>
<tr><td><code id="pieplot-methods_+3A_rex">rex</code></td>
<td>
<p> expansion factor for the pie radius.</p>
</td></tr>
<tr><td><code id="pieplot-methods_+3A_...">...</code></td>
<td>
<p> arguments to be passed to <code><a href="plotrix.html#topic+floating.pie">floating.pie</a></code> and others. See &lsquo;Details&rsquo; </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method calls <code><a href="plotrix.html#topic+floating.pie">floating.pie</a></code> <code>{plotrix}</code>, <code><a href="network.html#topic+network.vertex">network.vertex</a></code> <code>{network}</code>, and <code><a href="graphics.html#topic+plot.default">plot.default</a></code>, <code><a href="graphics.html#topic+lines">lines</a></code>, and <code><a href="graphics.html#topic+text">text</a></code>  <code>{graphics}</code>. This method also uses some internal structures of <code><a href="network.html#topic+plot.network.default">plot.network.default</a></code> from package <span class="pkg">network</span>. The following additional arguments can be passed to these functions: 
</p>

<dl>
<dt>mode</dt><dd><p> the vertex placement algorithm. Default is set to <code>"fruchtermanreingold"</code>. </p>
</dd>
<dt>pad</dt><dd><p> amount to pad the plotting range; useful if labels are being clipped. Default is set to 1.</p>
</dd>
<dt>displaylabels</dt><dd><p>boolean; should vertex labels be displayed? </p>
</dd>
<dt>label</dt><dd><p> a vector of vertex labels. By default, the rownames of the distance matrix (<code>rownames(p@d[[net]])</code>) are used. If <code>inter.labels==FALSE</code> only haplotype labels are displayed. </p>
</dd>
<dt>label.cex</dt><dd><p> character expansion factor for labels. Default is set to 0.75.</p>
</dd>
<dt>label.col</dt><dd><p> an integer or character vector for the label colors. By default, it is 1 (black). </p>
</dd>
<dt>label.pos</dt><dd><p> position at which labels should be placed relative to vertices. 0 and 6 results in labels which are placed away from the center of the plotting region; 1, 2, 3, and 4 result in labels being placed below, to the left of, above, and to the right of vertices, respectively; and label.pos 5 or greater than 6 results in labels which are plotted with no offset (i.e., at the vertex positions). Default is set to 0.</p>
</dd>
<dt>label.pad</dt><dd><p> amount to pad the labels. This setting is available only if the labels are plotted with offset relative to vertex positions. Default is set to 1.</p>
</dd>
<dt>vertex.cex</dt><dd><p> a numeric vector of expansion factor for intermediate vertices (only). By default it is <code>(0.5)*min(radius)</code>. Use 'radius' to specify size of pie charts. </p>
</dd>
<dt>col</dt><dd><p> the colors of the pie sectors (i.e., colors for populations), by default <code>rainbow</code>. </p>
</dd>
<dt>vertex.col</dt><dd><p> an integer or character vector for the intermediate vertex colors. By default, it is 1 (black). </p>
</dd>
<dt>edge.col</dt><dd><p> an integer or character vector for the edge colors. By default, it is 1 (black). </p>
</dd>
<dt>edge.lwd</dt><dd><p> a numeric vector, edges line width. By default, it is 1. </p>
</dd>
<dt>edge.lty</dt><dd><p> a numeric vector of length one, specifies the line type for the edges. By default it is 1. </p>
</dd>
<dt>edges</dt><dd><p> the number of lines forming a pie circle, By default, it is 200. </p>
</dd>
<dt>radius</dt><dd><p> a numeric vector of length <code>p@nhap[net]</code> for the radius of drawn pie charts. Useful for specifying the radius independent of the haplotype frequencies. Default is <code>(0.8*(haplotype frequencies)*rex)/max(haplotype frequencies)</code>. </p>
</dd>
<dt>vertex.sides</dt><dd><p> number of polygon sides for vertices. Default is set to 50. </p>
</dd>
<dt>xlab</dt><dd><p>x axis label. </p>
</dd>
<dt>xlab</dt><dd><p>y axis label. </p>
</dd>
</dl>



<h3>Value</h3>

<p>A two-column matrix containing the vertex positions as x,y coordinates.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Parsimnet", y = "Haplotype")</code></dt><dd>
</dd>
</dl>


<h3>Note</h3>

 
<p>Some internal structures of <code><a href="network.html#topic+plot.network.default">plot.network.default</a></code> is taken from package <span class="pkg">network</span> with modifications, author Carter T. Butts.
</p>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>. 
</p>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+plot+2CParsimnet-method">plot,Parsimnet-method</a></code>,<code><a href="plotrix.html#topic+floating.pie">floating.pie</a></code>, <code><a href="graphics.html#topic+plot.default">plot.default</a></code> and <code><a href="network.html#topic+plot.network.default">plot.network.default</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x&lt;-dna.obj
h&lt;-haplotypes::haplotype(x)

### Statistical parsimony with 95% connection limit 
p&lt;-parsimnet(x) 

#randomly generated populations
pop&lt;-c("pop1","pop2","pop3","pop4","pop5","pop6","pop7","pop8")
set.seed(5)
pops&lt;-sample(pop,nrow(x),replace=TRUE)

## Plotting with default parameters.
pieplot(p,h,1, pops)

## Change colors for the populations.
#8 colors for 8 populations 
cols&lt;-colors()[c(30,369,552,558,538,642,142,91)]
pieplot(p,h,1, pops,col=cols)

## Expanding pie charts and intermediate vertices.
pieplot(p,h,1, pops,rex=2)

## Adjusting intermediate vertex sizes.
pieplot(p,h,1, pops, vertex.cex=rep(0.2, nrow(p@d[[1]])-p@nhap))

## Expanding pie charts and intermediate vertices, adjusting intermediate vertex sizes.
pieplot(p,h,1, pops,rex=2, vertex.cex=rep(0.1, nrow(p@d[[1]])-p@nhap))

## Adjusting radius of pie charts.
pieplot(p,h,1, pops,radius=rep(1, p@nhap))

## Not run:  
## Interactively adjusting vertex positions.
pieplot(p,h,1, pops, interactive=TRUE)

## End(Not run)

### Multiple networks with 99% connection limit.
p&lt;-parsimnet(x,prob=.99) 

## Plotting first network with default parameters.
pieplot(p,h,1, pops)

## Change colors for the populations.
#8 colors for 8 populations
cols&lt;-colors()[c(30,369,552,558,538,642,142,91)]
pieplot(p,h,1, pops,col=cols)







</code></pre>

<hr>
<h2 id='plot-methods'>Methods for function <code>plot</code> in the package <span class="pkg">haplotypes</span></h2><span id='topic+plot-methods'></span><span id='topic+plot+2CParsimnet-method'></span><span id='topic+plot+2CParsimnet+2Cmissing-method'></span>

<h3>Description</h3>

<p>Plots statistical parsimony networks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Parsimnet'
plot(x,y,net=1,inter.labels=FALSE,...)
 </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code>.</p>
</td></tr>
<tr><td><code id="plot-methods_+3A_y">y</code></td>
<td>
<p>not used (needed for compatibility with generic plot function).</p>
</td></tr>
<tr><td><code id="plot-methods_+3A_net">net</code></td>
<td>
<p>a numeric vector of length one indicating which network to plot.</p>
</td></tr>
<tr><td><code id="plot-methods_+3A_inter.labels">inter.labels</code></td>
<td>
<p>boolean; should vertex labels of intermediate haplotypes to be displayed? </p>
</td></tr>
<tr><td><code id="plot-methods_+3A_...">...</code></td>
<td>
<p> additional arguments to <code><a href="graphics.html#topic+plot.default">plot.default</a></code>, <code><a href="network.html#topic+plot.network.default">plot.network.default</a></code> and arguments to be passed to plot method for <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code> objects. See &lsquo;Details&rsquo;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These methods call <code><a href="network.html#topic+plot.network.default">plot.network.default</a></code> from package <span class="pkg">network</span>. Some default parameters are changed:
</p>

<dl>
<dt>label</dt><dd><p> a vector of vertex labels. By default the row names of the distance matrices in slot <code>d</code> are used. If <code>inter.labels==FALSE</code> only haplotype labels are displayed. </p>
</dd>
<dt>usearrows</dt><dd><p> boolean; should arrows (rather than line segments) be used to indicate edges? Default is set to FALSE. </p>
</dd>
<dt>mode</dt><dd><p> the vertex placement algorithm. Default is set to <code>"kamadakawai"</code>. </p>
</dd>
<dt>pad</dt><dd><p> amount to pad the plotting range; useful if labels are being clipped. Default is set to 1.</p>
</dd>
<dt>label.cex</dt><dd><p> character expansion factor for label text. Default is set to 0.75.</p>
</dd>
<dt>vertex.cex</dt><dd><p> a numeric vector of expansion factor for vertices. By default it is 0.8 for haplotypes and 0.5 for intermediates. </p>
</dd>
<dt>vertex.col</dt><dd><p> an integer or character vector for the vertex colors. By default it is 2 (red) for haplotypes and 4 (blue) for intermediates. </p>
</dd>
</dl>



<h3>Value</h3>

<p>A two-column matrix containing the vertex positions as x,y coordinates.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Parsimnet",y = "missing")</code></dt><dd>
<p>Plots Parsimnet objects.
</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>. 
</p>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+parsimnet">parsimnet</a></code>, <code><a href="graphics.html#topic+plot.default">plot.default</a></code> and <code><a href="network.html#topic+plot.network.default">plot.network.default</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	
	
## Not run: 
	
data("dna.obj")
x&lt;-dna.obj


### Method for signature 'Parsimnet'.

## Statistical parsimony with 95
p&lt;-parsimnet(x) 
p
## Plotting with default parameters.
plot(p)

## Displaying vertex labels of intermediate haplotypes.
plot(p, inter.labels=TRUE)

## Interactively adjusting vertex positions.
plot(p, interactive=TRUE)

## Interactively adjusting and saving vertex positions.
p&lt;-parsimnet(x)
#saving vertex positions as x,y coordinates. 
coo&lt;-plot(p,interactive=TRUE) 
#reuse saved coordinates.
plot(p,coord=coo)

## Adjusting vertex sizes.
plot(p, vertex.cex=c(rep(3,nrow(p@d[[1]]))))
# different sizes for haplotypes and intermediates
plot(p, vertex.cex=c(rep(3,p@nhap),rep(1,c(nrow(p@d[[1]])-p@nhap)))) 

## Adjusting vertex colors
# different color for haplotypes and intermediates
plot(p, vertex.col=c(rep("magenta",p@nhap),rep("deepskyblue3",c(nrow(p@d[[1]])-p@nhap))))  
 
## Statistical parsimony with 98
p&lt;-parsimnet(x,prob=.99) 
p
#plot the first network
plot(p,net=1)
#plot the second network
plot(p,net=2)
#plot the third network. It is a single vertex.
plot(p,net=3) 






## End(Not run)

</code></pre>

<hr>
<h2 id='polymorp-methods'>Displays polymorphic sites (base substitutions and indels) between two sequences</h2><span id='topic+polymorp'></span><span id='topic+polymorp-methods'></span><span id='topic+polymorp+2CDna-method'></span>

<h3>Description</h3>

<p>This function displays the polymorphic sites (base substitutions and indels) between the two sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
polymorp(x,pair,indels="sic")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="polymorp-methods_+3A_x">x</code></td>
<td>
<p>an object of class  <code><a href="haplotypes.html#topic+Dna">Dna</a></code>. </p>
</td></tr>
<tr><td><code id="polymorp-methods_+3A_pair">pair</code></td>
<td>
<p>a vector of integers in the range [1,nrow(x)] of length two, specifying sequence pair. </p>
</td></tr>
<tr><td><code id="polymorp-methods_+3A_indels">indels</code></td>
<td>
<p>the indel coding method to be used. This must be one of &quot;sic&quot;, &quot;5th&quot; or &quot;missing&quot;. Any unambiguous substring can be given.  See <code><a href="haplotypes.html#topic+distance">distance</a></code> for details. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with two components: 
</p>

<dl>
<dt><code>indels</code>:</dt><dd><p>a list of matrices of the indel regions if <code>indels=="sic"</code>. The component names of the list gives the position of the indels. </p>
</dd>
<dt><code>subst</code>:</dt><dd><p>a list of matrices of the base substitutions. If <code>indels=="5th"</code>, each gap is treated as a base substitution. The component names of the list gives the position of the base substitutions. </p>
</dd>
</dl>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p>Showing base substitutions and indels between the two sequences.
</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+indelcoder">indelcoder</a></code> and <code><a href="haplotypes.html#topic+subs">subs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x&lt;-dna.obj

## Showing base substitutions and indels between seq1 and seq6.

# gaps are coded following the simple indel coding method
polymorp(x,c(1,6),indels="s")

# gaps are coded as a fifth state character
polymorp(x,c(1,6),indels="5")

# gaps are treated as missing character 
polymorp(x,c(1,6),indels="m")


</code></pre>

<hr>
<h2 id='range-methods'>Returns the minimum and maximum lengths of the DNA sequences</h2><span id='topic+range-methods'></span><span id='topic+range+2CDna-method'></span>

<h3>Description</h3>

<p><code>range</code> returns the lengths of shortest and longest DNA sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
range(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="range-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an integer of length two.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
<p><code>range</code> 
</p>
</dd>
</dl>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+length-methods">length-methods</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x &lt;-dna.obj

## shortest and longest DNA sequence lengths
range(x) 

</code></pre>

<hr>
<h2 id='read.fas'>
Read sequences from a file in FASTA format
</h2><span id='topic+read.fas'></span>

<h3>Description</h3>

<p>Read DNA sequences from a file in FASTA Format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.fas(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.fas_+3A_file">file</code></td>
<td>

<p>the name of the file, which the sequence in the FASTA format is to be read from. If it does not contain an <em>absolute</em> path, the file name is <em>relative</em> to the current working directory,
<code><a href="base.html#topic+getwd">getwd</a>()</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>read.fas</code> returns an object of class  <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.
</p>


<h3>Note</h3>

<p>By default, valid characters are &quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;,&quot;a&quot;,&quot;c&quot;,&quot;g&quot;,&quot;t&quot;,&quot;-&quot;,&quot;?&quot; for the class  <code><a href="haplotypes.html#topic+Dna">Dna</a></code>. Numeric entries (integers) between 0-5 will be converted to &quot;?&quot;,&quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;,&quot;-&quot;, respectively. Invalid characters will be replaced with &quot;?&quot; with a warning message. 
</p>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="haplotypes.html#topic+Dna">Dna</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##Reading example file.	
f&lt;-system.file("example.fas",package="haplotypes")
 
# invalid character 'N' was replaced with '?' with a warning message
x&lt;-read.fas(file=f) 

# an object of class 'Dna'
x

</code></pre>

<hr>
<h2 id='remove.gaps-methods'> Removing gaps from <code>Dna</code> object</h2><span id='topic+remove.gaps'></span><span id='topic+remove.gaps-methods'></span><span id='topic+remove.gaps+2CDna-method'></span>

<h3>Description</h3>

<p>Removing gaps(&quot;-&quot;) from <code>Dna</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
remove.gaps(x,entire.col=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove.gaps-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
<tr><td><code id="remove.gaps-methods_+3A_entire.col">entire.col</code></td>
<td>
<p>boolean; entire columns with gaps are removed if this is TRUE. See also &lsquo;Details&rsquo;. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>If entire.col==TRUE, alignment is preserved. If it is FALSE, end gaps are introduced to sequence matrix. 
</p>


<h3>Value</h3>

<p>an object of class  <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("dna.obj")

## original data
x&lt;-dna.obj 
range(x)
x@seqlengths

## Only gaps '-' are removed from sequences.
x&lt;-remove.gaps(dna.obj, entire.col=FALSE)
range(x)
x@seqlengths

## entire columns with gaps are removed.
x&lt;-remove.gaps(dna.obj, entire.col=TRUE)
range(x)
x@seqlengths


</code></pre>

<hr>
<h2 id='rownames-methods'>Retrieve or set the row names</h2><span id='topic+rownames-methods'></span><span id='topic+rownames+3C--methods'></span><span id='topic+rownames+2CDna-method'></span><span id='topic+rownames+3C-+2CDna-method'></span><span id='topic+rownames+2CParsimnet-method'></span><span id='topic+rownames+3C-+2CParsimnet-method'></span>

<h3>Description</h3>

<p>Function to get or set row names of a sequence matrix in a <code>Dna</code> object or distance matrix (or matrices) in a <code>Parsimnet</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
rownames(x)
## S4 method for signature 'Parsimnet'
rownames(x)
## S4 replacement method for signature 'Dna'
rownames(x)&lt;-value
## S4 replacement method for signature 'Parsimnet'
rownames(x)&lt;-value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rownames-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code> or <code><a href="haplotypes.html#topic+Parsimnet">Parsimnet</a></code>. </p>
</td></tr>
<tr><td><code id="rownames-methods_+3A_value">value</code></td>
<td>
<p>a character vector of the same length as number of sequences or a list of the same length as number of networks including vertex names for each network. See &lsquo;Examples&rsquo; </p>
</td></tr>
</table>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
</dd>
<dt><code>signature(x = "Parsimnet")</code></dt><dd>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>
data("dna.obj")
x&lt;-dna.obj

### Method for signature 'Dna'.

## Getting sequence names.
rownames(x)

## Setting sequence names.
rownames(x)&lt;-c(1:nrow(x))
rownames(x)

### Method for signature 'Parsimnet'.
x&lt;-dna.obj

##single network
p&lt;-parsimnet(x)

##Getting vertex names
rownames(p)

## Setting vertex names.
rownames(p)&lt;-list(c(1:nrow(p@d[[1]])))
rownames(p)
plot(p)

## Multiple networks with 99% connection limit.
p&lt;-parsimnet(x,prob=.99) 

## Getting vertex names
rownames(p)

## Setting vertex names.
rownames(p)&lt;-list(1:9, 10, 11,12:13,14,15:16)
rownames(p)



</code></pre>

<hr>
<h2 id='show-methods'>Methods for function <code>show</code> in the package <span class="pkg">haplotypes</span></h2><span id='topic+show-methods'></span><span id='topic+show+2CDna-method'></span><span id='topic+show+2CHaplotype-method'></span><span id='topic+show+2CParsimnet-method'></span>

<h3>Description</h3>

<p>Show objects of classes <code>Dna</code>, <code>Haplotype</code>, and  <code>Parsimnet</code> 
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "Dna")</code></dt><dd>
<p>displays <code>Dna</code> object briefly: The total number of DNA sequences, names of the first six sequences (if nrow(x)&gt;=6), length of the shortest and longest sequences and the names of the slots.
</p>
</dd>
<dt><code>signature(object = "Haplotype")</code></dt><dd>
<p>displays <code>Haplotype</code> object briefly: The list of individuals that share the same haplotypes, the total number of haplotypes and the names of the slots.
</p>
</dd>
<dt><code>signature(object = "Parsimnet")</code></dt><dd>
<p>displays <code>Parsimnet</code> object briefly: The total number of networks, the maximum connection steps at chosen probability, the total number of haplotypes in each network,the total number of intermediates in each network, total network lengths (scores) of each network and the names of the slots.</p>
</dd>
</dl>

<hr>
<h2 id='subs-methods'>Displays base substitutions</h2><span id='topic+subs'></span><span id='topic+subs-methods'></span><span id='topic+subs+2CDna-method'></span>

<h3>Description</h3>

<p>This function displays all base substitutions. If <code>fifth=="TRUE"</code>, each gap is treated as a fifth state character.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
subs(x,fifth=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subs-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>. </p>
</td></tr>
<tr><td><code id="subs-methods_+3A_fifth">fifth</code></td>
<td>
<p> boolean; should gaps be treated as a fifth state character?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with three components: 
</p>

<dl>
<dt><code>subsmat</code>:</dt><dd><p>a sequence matrix showing substitutions.</p>
</dd>
<dt><code>subs</code>:</dt><dd><p>a list of matrices of the substitutions.</p>
</dd>
<dt><code>subsmnum</code>:</dt><dd><p>total number of substitutions.</p>
</dd>
</dl>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Caner Aktas, <a href="mailto:caktas.aca@gmail.com">caktas.aca@gmail.com</a>. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dna.obj")
x&lt;-dna.obj

## Base substitutions.
subs(x)

## Gaps are treated as a fifth state character.
subs(x,fifth=TRUE)

</code></pre>

<hr>
<h2 id='tolower-methods+2C+20toupper-methods'> Convert sequence characters from upper to lower case or vice versa</h2><span id='topic+tolower-methods'></span><span id='topic+tolower+2CDna-method'></span><span id='topic+toupper-methods'></span><span id='topic+toupper+2CDna-method'></span>

<h3>Description</h3>

<p>Convert sequence characters in a <code>Dna</code> object from upper to lower case or vice versa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
tolower(x)
## S4 method for signature 'Dna'
toupper(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tolower-methods+2B2C+2B20toupper-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>	
## Coercing a list to a 'Dna' object.
seq1&lt;-c("?","A","C","g","t","-","0","1")
seq2&lt;-c("?","A","C","g","t","-","0","1","2")
seq3&lt;-c("?","A","C","g","t","-","0","1","2","3")
x&lt;-list(seq1=seq1,seq2=seq2,seq3=seq3) 
dna.obj&lt;-as.dna(x)

#characters in Dna object
table(as.matrix(dna.obj))

##all lower case
lowc&lt;-tolower(dna.obj)
#characters
table(as.matrix(lowc))

##all upper case
upc&lt;-toupper(dna.obj)
#characters
table(as.matrix(upc))

</code></pre>

<hr>
<h2 id='unique-methods'>Extract Unique Sequences</h2><span id='topic+unique-methods'></span><span id='topic+unique+2CDna-method'></span><span id='topic+unique+2CDna+2CANY-method'></span>

<h3>Description</h3>

<p><code>unique</code> returns a list with duplicate sequences removed. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Dna'
unique(x,gaps=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unique-methods_+3A_x">x</code></td>
<td>
<p>an object of class <code><a href="haplotypes.html#topic+Dna">Dna</a></code>.</p>
</td></tr>
<tr><td><code id="unique-methods_+3A_gaps">gaps</code></td>
<td>
<p>boolean; gaps are removed if this is FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function behaves somehow similar to <code><a href="haplotypes.html#topic+haplotype">haplotype</a></code>, however indels and missing characters are not taken into account.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "Dna")</code></dt><dd>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>	
data("dna.obj")
x&lt;-dna.obj[1:6,,as.matrix=FALSE]

##gaps removed.
unique(x)

##gaps not removed.
unique(x,gaps=TRUE)

##unique vs. haplotype
#unique returns 5 unique sequences.
unique(x)
length(unique(x))

#haplotype returns 4 unique haplotypes with simple indel coding.
h&lt;-haplotype(x)
as.list(as.dna(h))
length(h)

#haplotype returns 3 unique haplotypes with gaps as missing.
h&lt;-haplotype(x, indels="m")
as.list(as.dna(h))
length(h)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
