<!DOCTYPE html><html><head><title>Help for package haplo.ccs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {haplo.ccs}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#haplo'><p>Specify an Allele Matrix, Inheritance Mode, and Grouping for Rare Haplotypes</p></a></li>
<li><a href='#haplo.ccs'><p>Estimate Haplotype Relative Risks in Case-Control Data</p></a></li>
<li><a href='#haplo.hpp'><p>Obtain Summaries of Phase Ambiguity</p></a></li>
<li><a href='#renin'><p>Example Dataset for 'haplo.ccs'</p></a></li>
<li><a href='#sandcov'><p>Compute Sandwich Variance-Covariance Estimates</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimate Haplotype Relative Risks in Case-Control Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-04-26</td>
</tr>
<tr>
<td>Author:</td>
<td>Benjamin French[aut, cre],
  Thomas Lumley[aut, cre],
  Shawn Garbett[ctb]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.13.0), haplo.stats, survival</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Benjamin French &lt;b.french@vumc.org&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Haplotype and covariate relative risks in case-control data are estimated by weighted logistic regression. Diplotype probabilities, which are estimated by EM computation with progressive insertion of loci, are utilized as weights. French et al. (2006) &lt;<a href="https://doi.org/10.1002%2Fgepi.20161">doi:10.1002/gepi.20161</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/vubiostat/haplo.ccs">https://github.com/vubiostat/haplo.ccs</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-26 14:56:47 UTC; frencb2</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-28 07:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='haplo'>Specify an Allele Matrix, Inheritance Mode, and Grouping for Rare Haplotypes</h2><span id='topic+haplo'></span>

<h3>Description</h3>

<p>'haplo' specifies an allele matrix and an inheritance mode for the
'haplo.ccs' model formula. 'haplo' also specifies preferences for
grouping rare haplotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
haplo(..., mode, group.rare=TRUE, rare.freq=0.02)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplo_+3A_...">...</code></td>
<td>
<p>a matrix of alleles or list of columns of an allele matrix.
Each locus on the chromosome has a pair of adjacent columns in the
allele matrix, so that the number of columns of the allele matrix is
twice the number of loci on the chromosome. The order of the columns
corresponds to the order of the loci on the chromosome, and each row
corresponds to the alleles for each subject.  The alleles should be
numerically coded, i.e., 1, 2, 3, or 4 for 'A', 'C', 'G', or 'T'.</p>
</td></tr>
<tr><td><code id="haplo_+3A_mode">mode</code></td>
<td>
<p>the inheritance mode, either 'additive', 'dominant', or
'recessive'. Note that the default inheritance mode is 'additive'.</p>
</td></tr>
<tr><td><code id="haplo_+3A_group.rare">group.rare</code></td>
<td>
<p>a logical value indicating whether rare haplotypes
should be grouped in the 'haplo.ccs' model. Note that the default is
to group rare haplotypes.</p>
</td></tr>
<tr><td><code id="haplo_+3A_rare.freq">rare.freq</code></td>
<td>
<p>the population haplotype frequency to define rare
haplotypes. If 'group.rare=TRUE', then haplotypes with an estimated
population frequency less than or equal to 'rare.freq' are grouped
in the 'haplo.ccs' model. Note that the default haplotype frequency
is 0.02. 'rare.freq' is automatically set to 0 if
'group.rare=FALSE'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of alleles with mode, group.rare, and rare.freq assigned as
attributes.
</p>


<h3>Author(s)</h3>

<p>Benjamin French and Thomas Lumley, University of Washington
</p>


<h3>References</h3>

<p>French B, Lumley T, Monks SA, Rice KM, Hindorff LA, Reiner AP, Psaty
BM. Simple estimates of haplotype relative risks in case-control
data. Genetic Epidemiology 2006; 30(6):485-494.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplo.ccs">haplo.ccs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(renin)

## Specify an allele matrix in a model fit by 'haplo.ccs'.

haplo.ccs(case ~ haplo(geno))

## Specify dominant inheritance and define rare haplotypes.

haplo.ccs(case ~ haplo(geno, mode="dominant", rare.freq=0.01))

## Specify the allele matrix without grouping rare haplotypes.

haplo.ccs(case ~ haplo(geno, group.rare=FALSE))

</code></pre>

<hr>
<h2 id='haplo.ccs'>Estimate Haplotype Relative Risks in Case-Control Data</h2><span id='topic+haplo.ccs'></span><span id='topic+haplo.ccs.fit'></span><span id='topic+summary.haplo.ccs'></span><span id='topic+print.haplo.ccs'></span><span id='topic+print.summary.haplo.ccs'></span><span id='topic+coef.haplo.ccs'></span><span id='topic+fitted.haplo.ccs'></span><span id='topic+residuals.haplo.ccs'></span><span id='topic+vcov.haplo.ccs'></span><span id='topic+haplo.freq'></span><span id='topic+anova.haplo.ccs'></span><span id='topic+logLik.haplo.ccs'></span><span id='topic+AIC.haplo.ccs'></span><span id='topic+one'></span><span id='topic+count.haps'></span><span id='topic+return.haps'></span>

<h3>Description</h3>

<p>'haplo.ccs' estimates haplotype and covariate relative risks in
case-control data by weighted logistic regression. Diplotype
probabilities, which are estimated by EM computation with progressive
insertion of loci, are utilized as weights. The model is specified by
a symbolic description of the linear predictor, which includes
specification of an allele matrix, inheritance mode, and preferences
for rare haplotypes using 'haplo'. Note that use of this function
requires installation of the 'haplo.stats' and 'survival'
packages. See 'haplo.em' for a description of EM computation of
diplotype probabilities. Currently missing genotype information is not
allowed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
haplo.ccs(formula, data=NULL, ...)

haplo.ccs.fit(y, x, int, geno, inherit.mode, group.rare, rare.freq, 
              referent, names.x, names.int, ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplo.ccs_+3A_formula">formula</code></td>
<td>
<p>a symbolic description of the model to be fit, which
requires specification of an allele matrix and inheritance mode
using 'haplo'. Note that 'additive' is the default inheritance mode
for 'haplo'. Preferences for grouping rare haplotypes are also
specified using 'haplo'. Note that by default 'haplo' groups
haplotypes with an estimated population frequency less than
0.02. More details on model formulae are given below.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_data">data</code></td>
<td>
<p>an optional data frame, list, or environment (or object
coercible by 'as.data.frame' to a data frame) containing the
variables in the model. If not found in 'data', the variables are
taken from 'environment(formula)', typically the environment from
which 'haplo.ccs' is called.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_referent">referent</code></td>
<td>
<p>a character string representing the haplotype to be used
as the referent. The haplotype with the highest estimated population
frequency is the default referent.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_...">...</code></td>
<td>
<p>optional model-fitting arguments to be passed to 'glm'.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_y">y</code></td>
<td>
<p>a vector of observations.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_x">x</code></td>
<td>
<p>the design matrix for environmental covariates.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_int">int</code></td>
<td>
<p>the design matrix for haplotype-environment interaction.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_geno">geno</code></td>
<td>
<p>the allele matrix.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_inherit.mode">inherit.mode</code></td>
<td>
<p>the inheritance mode specified by 'haplo'.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_group.rare">group.rare</code></td>
<td>
<p>a logical value indicating whether rare haplotypes
should be grouped, specified by 'haplo'.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_rare.freq">rare.freq</code></td>
<td>
<p>the population haplotype frequency used to define the
rare haplotypes, specified by 'haplo'.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_names.x">names.x</code></td>
<td>
<p>the column names of the design matrix for covariates.</p>
</td></tr>
<tr><td><code id="haplo.ccs_+3A_names.int">names.int</code></td>
<td>
<p>the column names of the design matrix for
haplotype-environment interaction.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A formula has the form 'y ~ terms' where 'y' is a numeric vector
indicating case-control status and 'terms' is a series of terms which
specifies a linear predictor for 'y'. A terms specification of the
form 'first + second' indicates all the terms in 'first' together with
all the terms in 'second' with duplicates removed. The terms in the
formula will be re-ordered so that main effects come first, followed
by the interactions, all second-order, all third-order and so on. The
specification 'first*second' indicates the cross of 'first' and
'second'.
</p>
<p>Note that 'haplo.ccs.fit' is the workhorse function. The inputs 'y',
'x', 'geno', and 'int' represent case-control status, the matrix of
covariates, the matrix of alleles, and the matrix of terms that have
interaction with the haplotypes to be estimated from the alleles. The
argument 'inherit.mode' corresponds to the inheritance mode specified
by 'haplo', and the arguments 'group.rare' and 'rare.freq' correspond
to the preferences for grouping rare haplotypes specified by
'haplo'. 'names.x' and 'names.int' correspond to the column names of
'x' and 'int', respectively. The background functions 'one',
'count.haps', and 'return.haps' are used in specifying the model terms
and neatly packaging the results.
</p>


<h3>Value</h3>

<p>'haplo.ccs' returns an object of class inheriting from
'&quot;haplo.ccs&quot;'. More details appear later in this section. The function
'summary' (i.e., 'summary.haplo.ccs') obtains or prints a summary of
the results, which include haplotype and covariate relative risks,
robust standard error estimates, and estimated haplotype
frequencies. The generic accessory functions 'coefficients',
'fitted.values', and 'residuals' extract corresponding features of the
object returned by 'haplo.ccs'. The function 'vcov' (i.e.,
'vcov.haplo.ccs') returns sandwich variance-covariance estimates. The
function 'haplo.freq' extracts information returned by the EM
computation of haplotype frequencies. Note that if rare haplotypes are
grouped, then their individual estimated frequencies are summed. An
object of class '&quot;haplo.ccs&quot;' is a list containing at least the
following components:
</p>
<table>
<tr><td><code>formula</code></td>
<td>
<p>the formula supplied.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>the matched call.</p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p>a named vector of coefficients.</p>
</td></tr>
<tr><td><code>covariance</code></td>
<td>
<p>a named matrix of sandwich variance-covariance
estimates, computed using 'sandcov'.</p>
</td></tr>
<tr><td><code>residuals</code></td>
<td>
<p>the working residuals, i.e., the residuals from the
final iteration of the IWLS fit.</p>
</td></tr>
<tr><td><code>fitted.values</code></td>
<td>
<p>the fitted mean values, obtained by transforming
the linear predictors by the expit function.</p>
</td></tr>
<tr><td><code>linear.predictors</code></td>
<td>
<p>the linear fit on the logit scale.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>the model degrees of freedom.</p>
</td></tr>
<tr><td><code>rank</code></td>
<td>
<p>the numeric rank of the fitted model.</p>
</td></tr>
<tr><td><code>family</code></td>
<td>
<p>the family object, in this case, quasibinomial.</p>
</td></tr>
<tr><td><code>iter</code></td>
<td>
<p>the number of iterations of IWLS used.</p>
</td></tr>
<tr><td><code>weights</code></td>
<td>
<p>the working weights, i.e., the weights from the final
iteration of the IWLS fit.</p>
</td></tr>
<tr><td><code>prior.weights</code></td>
<td>
<p>the weights initially supplied, in this case, the
diplotype probabilities estimated by the EM computation.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>a vector indicating case-control status, expanded for each
subject by the number of plausible diplotypes for that subject.</p>
</td></tr>
<tr><td><code>id</code></td>
<td>
<p>the numeric vector used to identify subjects, expanded for
each subject by the number of plausible diplotypes for that
subject.</p>
</td></tr>
<tr><td><code>converged</code></td>
<td>
<p>a logical indicating whether the IWLS fit converged.</p>
</td></tr>
<tr><td><code>boundary</code></td>
<td>
<p>a logical indicating whether the fitted values are on
the boundary of the attainable values.</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>the model matrix used.</p>
</td></tr>
<tr><td><code>terms</code></td>
<td>
<p>the terms object used.</p>
</td></tr>
<tr><td><code>offset</code></td>
<td>
<p>the offset vector used.</p>
</td></tr>
<tr><td><code>contrasts</code></td>
<td>
<p>the contrasts used.</p>
</td></tr>
<tr><td><code>xlevels</code></td>
<td>
<p>a record of the levels of the factors used in fitting.</p>
</td></tr>
<tr><td><code>inheritance.mode</code></td>
<td>
<p>the method of inheritance.</p>
</td></tr>
<tr><td><code>rare.freq</code></td>
<td>
<p>the value used to define the rare haplotypes.</p>
</td></tr>
<tr><td><code>em.lnlike</code></td>
<td>
<p>the value of the log likelihood at the last EM
iteration.</p>
</td></tr>
<tr><td><code>em.lr</code></td>
<td>
<p>the likelihood ratio statistic used to test the assumed
model against the model that assumes complete linkage equilibrium
among all loci.</p>
</td></tr>
<tr><td><code>em.df.lr</code></td>
<td>
<p>the degrees of freedom for the likelihood ratio
statistic.</p>
</td></tr>
<tr><td><code>em.nreps</code></td>
<td>
<p>the count of haplotype pairs that map to
each subject's marker genotypes.</p>
</td></tr>
<tr><td><code>hap1</code></td>
<td>
<p>character strings representing the possible first
haplotype for each subject.</p>
</td></tr>
<tr><td><code>hap2</code></td>
<td>
<p>character strings representing the possible second
haplotype for each subject.</p>
</td></tr>
<tr><td><code>hap.names</code></td>
<td>
<p>character strings representing the unique
haplotypes.</p>
</td></tr>
<tr><td><code>hap.probs</code></td>
<td>
<p>the estimated frequency of each unique
haplotype. Note that if rare haplotypes are grouped, then their
individual estimated frequencies are summed.</p>
</td></tr>
<tr><td><code>em.converged</code></td>
<td>
<p>a logical indicating whether the EM computation
converged.</p>
</td></tr>
<tr><td><code>em.nreps</code></td>
<td>
<p>the number of haplotype pairs that map to the marker
genotypes for each subject.</p>
</td></tr>
<tr><td><code>em.max.pairs</code></td>
<td>
<p>the maximum number of pairs of haplotypes per
subject that are consistent with their marker data.</p>
</td></tr>
<tr><td><code>em.control</code></td>
<td>
<p>a list of control parameters for the EM computation.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The functions 'anova', 'logLik', and 'AIC' are not appropriate for
models of class '&quot;haplo.ccs&quot;', because 'haplo.ccs' does not fit by maximum
likelihood. Accordingly, model and null deviance are not reported.
</p>


<h3>Author(s)</h3>

<p>Benjamin French and Thomas Lumley, University of Washington
</p>


<h3>References</h3>

<p>French B, Lumley T, Monks SA, Rice KM, Hindorff LA, Reiner AP, Psaty
BM. Simple estimates of haplotype relative risks in case-control
data. Genetic Epidemiology 2006; 30(6):485-494.
</p>
<p>The help files for 'glm', 'haplo.em', and 'haplo.glm' were
instrumental in creating this help file.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+glm">glm</a></code>,
<code><a href="#topic+haplo">haplo</a></code>,
<code><a href="haplo.stats.html#topic+haplo.em">haplo.em</a></code>,
<code><a href="haplo.stats.html#topic+haplo.glm">haplo.glm</a></code>,
<code><a href="#topic+sandcov">sandcov</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(renin)

## Fit a model for haplotype effects.

haplo.ccs(case ~ haplo(geno))

## Fit a model for haplotype and covariate effects.

haplo.ccs(case ~ gender + age + factor(race) + haplo(geno))

## Fit a model for haplotype interaction with gender.

haplo.ccs(case ~ age + factor(race) + gender*haplo(geno))

</code></pre>

<hr>
<h2 id='haplo.hpp'>Obtain Summaries of Phase Ambiguity</h2><span id='topic+haplo.hpp'></span>

<h3>Description</h3>

<p>'haplo.hpp' obtains summary statistics of phase ambiguity. The
proportion of subjects whose highest posterior diplotype probability
is greater than or equal to a specified probability is reported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
haplo.hpp(model, prob=0.95)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplo.hpp_+3A_model">model</code></td>
<td>
<p>a fitted model of class '&quot;haplo.ccs&quot;'.</p>
</td></tr>
<tr><td><code id="haplo.hpp_+3A_prob">prob</code></td>
<td>
<p>the probability to which to compare the highest posterior
diplotype probability for each subject. Note that the default
probability is 0.95. Note also that either a single probability or a
list of probabilities may be specified.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The proportion of subjects whose highest posterior diplotype
probability is greater than or equal to the specified probability or
probabilities.
</p>


<h3>Author(s)</h3>

<p>Benjamin French and Thomas Lumley, University of Washington
</p>


<h3>References</h3>

<p>French B, Lumley T, Monks SA, Rice KM, Hindorff LA, Reiner AP, Psaty
BM. Simple estimates of haplotype relative risks in case-control
data. Genetic Epidemiology 2006; 30(6):485-494.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplo">haplo</a></code>,
<code><a href="#topic+haplo.ccs">haplo.ccs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(renin)

haplo.hpp(model=haplo.ccs(case ~ haplo(geno)), prob=c(0.90, 0.95))

</code></pre>

<hr>
<h2 id='renin'>Example Dataset for 'haplo.ccs'</h2><span id='topic+renin'></span><span id='topic+age'></span><span id='topic+case'></span><span id='topic+gender'></span><span id='topic+geno'></span><span id='topic+race'></span>

<h3>Description</h3>

<p>This dataset serves as the example dataset for 'haplo.ccs'. The
genotypes in this dataset were generated from haplotype frequency data
for renin, one of the genes in the renin-angiotensin system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
data(renin)

</code></pre>


<h3>Format</h3>

<p>All variables are in numeric format.
</p>


<h3>Details</h3>

<p>'case': case-control status (1=Case, 0=Control)
</p>
<p>'geno': a matrix of alleles indicating genotype where each locus
has a pair of adjacent columns of alleles, and the order of columns
corresponds to the order of the loci on the chromosome (1=A, 2=C,
3=T, 4=G)
</p>
<p>'gender': gender (1=Male, 2=Female)
</p>
<p>'age': age in years
</p>
<p>'race': race (1=White, 2=Black, 3=Asian, 4=Other)
</p>


<h3>Note</h3>

<p>Other covariates, such as gender, age, and race were randomly
generated. Therefore no scientific inference should be made from these
data.
</p>


<h3>Author(s)</h3>

<p>Benjamin French and Thomas Lumley, University of Washington
</p>


<h3>References</h3>

<p>French B, Lumley T, Monks SA, Rice KM, Hindorff LA, Reiner AP, Psaty
BM. Simple estimates of haplotype relative risks in case-control
data. Genetic Epidemiology 2006; 30(6):485-494.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplo.ccs">haplo.ccs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(renin)

</code></pre>

<hr>
<h2 id='sandcov'>Compute Sandwich Variance-Covariance Estimates</h2><span id='topic+sandcov'></span>

<h3>Description</h3>

<p>'sandcov' computes sandwich variance-covariance estimates for the
coefficients of a fitted model. These estimates may be used to
calculate robust standard error estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
sandcov(model, id)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sandcov_+3A_model">model</code></td>
<td>
<p>a fitted model of class '&quot;lm&quot;' or '&quot;glm&quot;'.</p>
</td></tr>
<tr><td><code id="sandcov_+3A_id">id</code></td>
<td>
<p>the numeric vector used to identify subjects, expanded for
each subject by the number of observations for that subject.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a model of class '&quot;haplo.ccs&quot;', the sandwich variance-covariance
matrix is automatically provided as the object 'covariance', or may be
extracted by 'vcov' (i.e., 'vcov.haplo.ccs'). See examples below.
</p>


<h3>Value</h3>

<p>A named matrix for the covariance of the regression coefficients
specified in 'model', calculated using the sandwich method.
</p>


<h3>Author(s)</h3>

<p>Benjamin French and Thomas Lumley, University of Washington
</p>


<h3>References</h3>

<p>French B, Lumley T, Monks SA, Rice KM, Hindorff LA, Reiner AP, Psaty
BM. Simple estimates of haplotype relative risks in case-control
data. Genetic Epidemiology 2006; 30(6):485-494.
</p>
<p>Liang KY, Zeger SL. Longitudinal data analysis using generalized
linear models. Biometrika 1986; 73(1):13-22.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+glm">glm</a></code>,
<code><a href="#topic+haplo.ccs">haplo.ccs</a></code>,
<code><a href="stats.html#topic+lm">lm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(renin)

## Fit a model for covariate effects.

m1 &lt;- glm(case ~ age + factor(race) + gender, family=binomial(link=logit))

## Obtain sandwich variance-covariance matrix.

id &lt;- 1:length(case)
v1 &lt;- sandcov(model = m1, id = id)

## Calculate robust standard error estimates.

se1 &lt;- sqrt(diag(v1))

## Fit a model for haplotype and covariate effects.

m2 &lt;- haplo.ccs(case ~ gender + age + factor(race) + haplo(geno))

## Obtain sandwich variance-covariance matrix by one of two methods.

v2 &lt;- m2$covariance
v2 &lt;- vcov(m2)

## Calculate robust standard error estimates.

se2 &lt;- sqrt(diag(v2))

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
