<!DOCTYPE html><html><head><title>Help for package haplotyper</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {haplotyper}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#haplotyper'><p>haplotyper function identifies haplotypes within QTL.</p></a></li>
<li><a href='#rice_qtl'><p>Real experimental data</p></a></li>
<li><a href='#sim_qtl'><p>simple QTL simulated</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tool for Clustering Genotypes in Haplotypes</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-03-28</td>
</tr>
<tr>
<td>Author:</td>
<td>Sebastian Simondi &lt;sebastian.simondi@gmail.com&gt; and Gaston Quero, with
 contributions from Victoria Bonnecarrere and Lucia Gutierrez</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gaston Quero &lt;gastonquero@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Function to  identify  haplotypes
  within QTL (Quantitative Trait Loci). One haplotype is a combination of SNP
  (Single Nucleotide Polymorphisms) within the QTL. This function groups
  together all individuals of a population with the same haplotype.
  Each group contains individual with the same allele in each SNP,
  whether or not missing data. Thus, haplotyper groups individuals,
  that to be imputed, have a non-zero probability of having the same alleles
  in the entire sequence of SNP's. Moreover, haplotyper calculates such
  probability from relative frequencies.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, utils</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-04-06 14:44:27 UTC; GQ</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-04-06 17:23:47</td>
</tr>
</table>
<hr>
<h2 id='haplotyper'>haplotyper function identifies haplotypes within QTL.</h2><span id='topic+haplotyper'></span>

<h3>Description</h3>

<p>This function groups together all individuals of a population
with the same haplotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplotyper(x, Print = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplotyper_+3A_x">x</code></td>
<td>
<p>a data.frame that should be loaded with read.table function.
Each row represents the individuals while each column represents the markers.
The first column contains the names of the genotypes.</p>
</td></tr>
<tr><td><code id="haplotyper_+3A_print">Print</code></td>
<td>
<p>option for print the haplotyper result.
The default is FALSE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each group contains individual with the same allele in each SNP,
whether or not missing data.
</p>


<h3>Value</h3>

<p>a matrix with the haplotypes
</p>


<h3>Author(s)</h3>

<p>Sebastian Simondi, Victoria Bonnecarrere, Lucia Gutierrez,
Gaston Quero
</p>


<h3>See Also</h3>

<p>read.table function
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(rice_qtl)
haplotyper(rice_qtl)

## End(Not run)
</code></pre>

<hr>
<h2 id='rice_qtl'>Real experimental data</h2><span id='topic+rice_qtl'></span>

<h3>Description</h3>

<p>The data is a QTL for rice Grain Quality
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rice_qtl
</code></pre>


<h3>Format</h3>

<p>A data frame 326 rows (individual) and 38 variables (SNPs)</p>


<h3>Source</h3>

<p>Uruguayan Rice Breeding GWAS (URiB)
</p>

<hr>
<h2 id='sim_qtl'>simple QTL simulated</h2><span id='topic+sim_qtl'></span>

<h3>Description</h3>

<p>A dataset containing the marcadores
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_qtl
</code></pre>


<h3>Format</h3>

<p>A data frame 5 rows (individuals) and 7 variables (snps)</p>


<h3>Source</h3>

<p>simulated data
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
