<!DOCTYPE html><html lang="en"><head><title>Help for package HaploCatcher</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HaploCatcher}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#auto_locus'><p>Auto Locus - An Automated Pipeline for Locus Prediction</p></a></li>
<li><a href='#gene_comp'><p>Model Gene Compendium Data Set</p></a></li>
<li><a href='#geno_mat'><p>Model Gene Compendium Data Set</p></a></li>
<li><a href='#locus_cv'><p>Haplotype Prediction: Cross Validation of KNN and RF Models</p></a></li>
<li><a href='#locus_perm_cv'><p>Haplotype Prediction: Permutation Cross Validation of KNN and RF Models</p></a></li>
<li><a href='#locus_pred'><p>Haplotype Prediction: Using Trained Models to Make Predictions</p></a></li>
<li><a href='#locus_train'><p>Haplotype Prediction: Training Models for Use in Forward Prediction</p></a></li>
<li><a href='#marker_info'><p>Model Gene Compendium Data Set</p></a></li>
<li><a href='#plot_locus_perm_cv'><p>Visualize Permutation CV Results</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>A Predictive Haplotyping Package</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-3-23</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Used for predicting a genotypeâ€™s allelic state at a specific locus/QTL/gene. This is accomplished by using both a genotype matrix and a separate file which has categorizations about loci/QTL/genes of interest for the individuals in the genotypic matrix. A training population can be created from a panel of individuals who have been previously screened for specific loci/QTL/genes, and this previous screening could be summarized into a category. Using the categorization of individuals which have been genotyped using a genome wide marker platform, a model can be trained to predict what category (haplotype) an individual belongs in based on their genetic sequence in the region associated with the locus/QTL/gene. These trained models can then be used to predict the haplotype of a locus/QTL/gene for individuals which have been genotyped with a genome wide platform yet not genotyped for the specific locus/QTL/gene. This package is based off work done by Winn et al 2021. For more specific information on this method, refer to &lt;<a href="https://doi.org/10.1007%2Fs00122-022-04178-w">doi:10.1007/s00122-022-04178-w</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>base, parallel, doParallel, foreach, caret, ggplot2, graphics,
knitr, patchwork, lattice, randomForest</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-04-21 15:47:33 UTC; zwinn</td>
</tr>
<tr>
<td>Author:</td>
<td>Zachary Winn <a href="https://orcid.org/0000-0003-1543-1527"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zachary Winn &lt;zwinn@outlook.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-21 23:32:39 UTC</td>
</tr>
</table>
<hr>
<h2 id='auto_locus'>Auto Locus - An Automated Pipeline for Locus Prediction</h2><span id='topic+auto_locus'></span>

<h3>Description</h3>

<p>This function takes all other functions of the HaploCatcher package and weaves them into one pipeline that performs locus prediction with minimal user intervention.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auto_locus(
  geno_mat,
  gene_file,
  gene_name,
  marker_info,
  chromosome,
  training_genotypes,
  testing_genotypes,
  ncor_markers = 50,
  n_neighbors = 50,
  cv_percent_testing = 0.2,
  cv_percent_training = 0.8,
  n_perms = 30,
  model_selection_parameter = "kappa",
  n_votes = 30,
  set_seed = NULL,
  predict_by_vote = FALSE,
  include_hets = FALSE,
  include_models = FALSE,
  verbose = TRUE,
  parallel = FALSE,
  n_cores = NULL,
  plot_cv_results = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="auto_locus_+3A_geno_mat">geno_mat</code></td>
<td>
<p>An imputed, number-coded, genotypic matrix which has n rows of individuals and m columns of markers. Row names of the matrix should be representative of genotypic IDs and column names should be representative of marker IDs. Missing data is not allowed. Numeric coding of genotypes can vary as long as it remains consistant among markers.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_gene_file">gene_file</code></td>
<td>
<p>A dataframe containing at least three columns labeled as follows: 'Gene', 'FullSampleName', and 'Call'. The 'Gene' column contains the name of the gene for which the observation belongs to. The 'FullSampleName' column contains the genotypic ID which corresponds exactly to the column name in the genotypic matrix. The 'Call' column contains the marker call which corresponds to the gene for that genotype. Other information may be present in this dataframe beyond these columns, but the three listed columns above are obligatory.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_gene_name">gene_name</code></td>
<td>
<p>A character string which matches the name of the gene which you are trying to perform cross validation for. This character string must be present in your gene_file 'Gene' column.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_marker_info">marker_info</code></td>
<td>
<p>A dataframe containing the following three columns: 'Marker', 'Chromosome', and 'BP_Position'. The 'Marker' column contains the names of the marker which are present in the genotypic matrix. The 'Chromosome' column contains the corresponding chromosome/linkage group to which the marker belongs. The 'Position' column contains the physical or centimorgan position of the marker. All markers present in the genotypic matrix must be listed in this dataframe. If physical or centimorgan positions are unavailable for the listed markers, a numeric dummy variable ranging from one to n number of markers may be provided instead.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_chromosome">chromosome</code></td>
<td>
<p>A character string which matches the name of the chromosome upon which the gene resides. This chromosome name must be present in the marker_info file.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_training_genotypes">training_genotypes</code></td>
<td>
<p>A character vector of fullsamplenames found in the geno_mat and gene_file. These are lines which will be used in cross validation and to train the model which will predict the lines which have no locus call available.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_testing_genotypes">testing_genotypes</code></td>
<td>
<p>A character vector of fullsamplenames found in the geno_mat. These are lines which will have predictions made for them.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_ncor_markers">ncor_markers</code></td>
<td>
<p>A numeric variable which represents the number of markers the user want to use in model training. Correlation among markers to the gene call is calculated and the top n markers specified are retained for training. The default setting is 50 markers.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_n_neighbors">n_neighbors</code></td>
<td>
<p>A numeric variable which represents the number of neighbors to use in KNN. Default is 50.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_cv_percent_testing">cv_percent_testing</code></td>
<td>
<p>A numeric variable which ranges such that x|0&lt;x&lt;1. This means that this number can be neither zero nor one. This number represents the percent of the total data available the user wants to retain to validate the model. The default setting is 0.20.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_cv_percent_training">cv_percent_training</code></td>
<td>
<p>A numeric variable which ranges such that x|0&lt;x&lt;1. This means that the number can be neither zero nor one. This number represents the percent of the total data available the user wants to retain for training of the model.The default setting is 0.80.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_n_perms">n_perms</code></td>
<td>
<p>A numeric variable defining the number of permutations to perform. This value may range from one to infinity. Default is 30.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_model_selection_parameter">model_selection_parameter</code></td>
<td>
<p>A character string which defines which model performance parameter the user wants to use to select the best performing model. Only two strings are accepted either &quot;kappa&quot; for selection on Cohen's kappa value or &quot;accuracy&quot; for selection on total accuracy. Default is &quot;kappa&quot;.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_n_votes">n_votes</code></td>
<td>
<p>A numeric variable defining the number of models to train and predict with. This method will take n number of predictions over multiple iterations, sum across those multiple iterations and provide a prediction which is based on majority rule. The default is 30.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_set_seed">set_seed</code></td>
<td>
<p>A numeric variable that is used to set a seed for reproducible results if the user is running the &quot;locus_trait&quot; function once for use in the &quot;locus_pred&quot; function. If the user wishes to run the function many times with a random seed and decide the outcome by voting, set 'predict_by_vote' to TRUE instead. The default setting is NULL.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_predict_by_vote">predict_by_vote</code></td>
<td>
<p>A logical variable which defines if the user wants predictions generated by voting over several runs of machine models summarized into a majority rule call. Default is FALSE.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_include_hets">include_hets</code></td>
<td>
<p>A logical variable which determines if the user wishes to include heterozygous calls or not. The default setting is FALSE.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_include_models">include_models</code></td>
<td>
<p>A logical variable which determines if the user wishes to include the trained models in the results object for further testing. Warning: the models are quite large and running this will result in a very large results object. The default setting is FALSE.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_verbose">verbose</code></td>
<td>
<p>A logical variable which determines if the user wants plots displayed and text feedback from each permutation. Regardless of this parameter, the function will display the name of the gene which is being cross validated and the current progress of the permutations. Default setting is TRUE.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_parallel">parallel</code></td>
<td>
<p>A logical variable which determines if the user wants the cross validation and voting predictions to be performed in parallel. Default is FALSE. If the user defines that parallel is TRUE, all visual and textual feedback will not be rendered.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_n_cores">n_cores</code></td>
<td>
<p>A numeric variable that is used to set a seed for reproducible results if the user is running the function once for use in the &quot;locus_pred&quot; function. If the user wishes to run the function many times with a random seed and decide the outcome by voting, set 'predict_by_vote' to true instead. The default setting is NULL.</p>
</td></tr>
<tr><td><code id="auto_locus_+3A_plot_cv_results">plot_cv_results</code></td>
<td>
<p>A logical variable which determines if the user want the cross validation results to be printed as a 'ggplot' image. Default is TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns two types of list-of-list objects. If 'predict_by_vote' is set to FALSE (the default setting), then the list will contain four objects: method, cross_validation_results, prediction_model, and predictions. The method object is a character which denotes what method was used. The cross_validation_results object contains the results of the cross-validation. The prediction model contains the information about the prediction model, and predictions contains the predictions for the testing_genotypes. If 'predict_by_vote' is set to TRUE, then the list will contain four objects: method, cross_validation_results, predictions, and consensus_predictions. Both method and cross_validation_results are the same for this option. Predictions contains the prediction of each model ran to vote. The object consensus_predictions contains a consensus prediction of the alleleic state of a genotype based on majority rule and a tablular summary of how many votes were taken per catigory.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#refer to vignette for an in depth look at the auto_locus function
vignette("An_Intro_to_HaploCatcher", package = "HaploCatcher")

</code></pre>

<hr>
<h2 id='gene_comp'>Model Gene Compendium Data Set</h2><span id='topic+gene_comp'></span>

<h3>Description</h3>

<p>A data frame which contains information from 1345 unique wheat lines on the Sst1 solid stem locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_comp
</code></pre>


<h3>Format</h3>

<p>A data frame with 1345 rows and 7 columns:
</p>

<dl>
<dt>Trait</dt><dd><p>A short discription of the phenotype associated with the gene</p>
</dd>
<dt>Chromosome</dt><dd><p>The chromosome where the gene resides</p>
</dd>
<dt>Gene</dt><dd><p>The name of the gene</p>
</dd>
<dt>Nursery</dt><dd><p>The program which produced the gene call for the genotype</p>
</dd>
<dt>Line</dt><dd><p>A breeder assigned line designation</p>
</dd>
<dt>FullSampleName</dt><dd><p>A designation unique to the line found in the genotypic matrix</p>
</dd>
<dt>Call</dt><dd><p>A 'call' given for the allelic state. For this package, it is best to format the non desiarable allele as &quot;non_gene&quot; and the heterozygous state as &quot;het_gene&quot;.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Generated by Zachary James Winn for the CSU breeding program via USDA-ARS gene reports and in-house gene assays
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("gene_comp") #lazy loads the dataset for use in the package
</code></pre>

<hr>
<h2 id='geno_mat'>Model Gene Compendium Data Set</h2><span id='topic+geno_mat'></span>

<h3>Description</h3>

<p>A numeric matrix which contains molecular marker information on 1345 unique genotypes for 2271 SNP markers located on wheat chromosome 3B. This data set corresponds to the information found in the &quot;gene_comp&quot; and &quot;marker_info&quot; data sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geno_mat
</code></pre>


<h3>Format</h3>

<p>A numeric matrix with 1345 rows and 2271 columns:
</p>


<h3>Source</h3>

<p>Generated by Zachary James Winn for the CSU breeding program via historical in-house GBS data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("geno_mat") #lazy loads the dataset for use in the package
</code></pre>

<hr>
<h2 id='locus_cv'>Haplotype Prediction: Cross Validation of KNN and RF Models</h2><span id='topic+locus_cv'></span>

<h3>Description</h3>

<p>This function performs the analysis featured in Winn et al 2022 where genome wide markers are used to train machine learning models to identify if genotypes have or do not have specific alleles of a QTL/gene. This function is used to perform cross validation where a random partition of the total available data is used to train a model and a reserved testing partition is used to validate. This function is used for a single round of cross validation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locus_cv(
  geno_mat,
  gene_file,
  gene_name,
  marker_info,
  chromosome,
  ncor_markers = 50,
  n_neighbors = 50,
  percent_testing = 0.2,
  percent_training = 0.8,
  include_hets = FALSE,
  include_models = FALSE,
  verbose = TRUE,
  graph = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="locus_cv_+3A_geno_mat">geno_mat</code></td>
<td>
<p>An imputed, number-coded, genotypic matrix which has n rows of individuals and m columns of markers. Row names of the matrix should be representative of genotypic IDs and column names should be representative of marker IDs. Missing data is not allowed. Numeric coding of genotypes can vary as long as it remains consistant among markers.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_gene_file">gene_file</code></td>
<td>
<p>A dataframe containing at least three columns labeled as follows: 'Gene', 'FullSampleName', and 'Call'. The 'Gene' column contains the name of the gene for which the observation belongs to. The 'FullSampleName' column contains the genotypic ID which corresponds exactly to the column name in the genotypic matrix. The 'Call' column contains the marker call which corresponds to the gene for that genotype. Other information may be present in this dataframe beyond these columns, but the three listed columns above are obligatory.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_gene_name">gene_name</code></td>
<td>
<p>A character string which matches the name of the gene which you are trying to perform cross validation for. This character string must be present in your gene_file 'Gene' column.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_marker_info">marker_info</code></td>
<td>
<p>A dataframe containing the following three columns: 'Marker', 'Chromosome', and 'BP_Position'. The 'Marker' column contains the names of the marker which are present in the genotypic matrix. The 'Chromosome' column contains the corresponding chromosome/linkage group to which the marker belongs. The 'Position' column contains the physical or centimorgan position of the marker. All markers present in the genotypic matrix must be listed in this dataframe. If physical or centimorgan positions are unavailable for the listed markers, a numeric dummy variable ranging from one to n number of markers may be provided instead.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_chromosome">chromosome</code></td>
<td>
<p>A character string which matches the name of the chromosome upon which the gene resides. This chromosome name must be present in the marker_info file.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_ncor_markers">ncor_markers</code></td>
<td>
<p>A numeric variable which represents the number of markers the user want to use in model training. Correlation among markers to the gene call is calculated and the top n markers specified are retained for training. The default setting is 50 markers.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_n_neighbors">n_neighbors</code></td>
<td>
<p>A numeric variable which represents the number of neighbors to use in KNN. Default is 50.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_percent_testing">percent_testing</code></td>
<td>
<p>A numeric variable which ranges such that x|0&lt;x&lt;1. This means that this number can be neither zero nor one. This number represents the percent of the total data available the user wants to retain to validate the model. The default setting is 0.20.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_percent_training">percent_training</code></td>
<td>
<p>A numeric variable which ranges such that x|0&lt;x&lt;1. This means that the number can be neither zero nor one. This number represents the percent of the total data available the user wants to retain for training of the model.The default setting is 0.80.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_include_hets">include_hets</code></td>
<td>
<p>A logical variable which determines if the user wishes to include heterozygous calls or not. Default is FALSE.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_include_models">include_models</code></td>
<td>
<p>A logical variable which determines if the user wishes to include the trained models in the results object for further testing. Warning: the models are quite large and running this will result in a very large results object. Default is FALSE.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_verbose">verbose</code></td>
<td>
<p>A logical variable which determines if the user wants text feedback. Default is TRUE.</p>
</td></tr>
<tr><td><code id="locus_cv_+3A_graph">graph</code></td>
<td>
<p>A logical variable which determines if the user wants plots displayed. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a list of list which contains the following list objects: 'confu', 'preds', 'models', and 'data'. The 'confu' list contains the confusion matrix objects for both the random forest and k-nearest neighbors models. The 'preds' list contains the predictions made by the separate models. The 'models' contains the two caret model objects for both the random forest and k-nearest neighbors models. The 'data' list contains the training and test data frames made by the function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#read in the genotypic data matrix
data("geno_mat")

#read in the marker information
data("marker_info")

#read in the gene compendium file
data("gene_comp")

#run the function without hets for a very limited number of markers and neighbors
#due to requirements by cran, this must be commented out
#to run, place this code in the console and remove comments
#fit&lt;-locus_cv(geno_mat=geno_mat, #the genotypic matrix
#             gene_file=gene_comp, #the gene compendium file
#             gene_name="sst1_solid_stem", #the name of the gene
#             marker_info=marker_info, #the marker information file
#             chromosome="3B", #name of the chromosome
#             ncor_markers=2, #number of markers to retain
#             n_neighbors=1, #number of neighbors
#             percent_testing=0.2, #percentage of genotypes in the validation set
#             percent_training=0.8, #percentage of genotypes in the training set
#             include_hets=FALSE, #include hets in the model
#             include_models=TRUE, #include models in the final results
#             verbose=TRUE, #allows text output
#             graph=TRUE) #allows graph output

</code></pre>

<hr>
<h2 id='locus_perm_cv'>Haplotype Prediction: Permutation Cross Validation of KNN and RF Models</h2><span id='topic+locus_perm_cv'></span>

<h3>Description</h3>

<p>This function performs the analysis featured in Winn et al 2022 where genome wide markers are used to train machine learning models to identify if genotypes have or do not have specific alleles of a QTL/gene. This function is used to perform cross validation where a random partition of the total available data is used to train a model and a reserved testing partition is used to validate. This function is used for permutation based cross validation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locus_perm_cv(
  n_perms = 30,
  geno_mat,
  gene_file,
  gene_name,
  marker_info,
  chromosome,
  ncor_markers = 50,
  n_neighbors = 50,
  percent_testing = 0.2,
  percent_training = 0.8,
  include_hets = FALSE,
  include_models = FALSE,
  verbose = FALSE,
  parallel = FALSE,
  n_cores = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="locus_perm_cv_+3A_n_perms">n_perms</code></td>
<td>
<p>A numeric variable defining the number of permutations to perform. This value may range from one to infinity. Default is 30.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_geno_mat">geno_mat</code></td>
<td>
<p>An imputed, number-coded, genotypic matrix which has n rows of individuals and m columns of markers. Row names of the matrix should be representative of genotypic IDs and column names should be representative of marker IDs. Missing data is not allowed. Numeric coding of genotypes can vary as long as it remains consistent among markers.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_gene_file">gene_file</code></td>
<td>
<p>A dataframe containing at least three columns labeled as follows: 'Gene', 'FullSampleName', and 'Call'. The 'Gene' column contains the name of the gene for which the observation belongs to. The 'FullSampleName' column contains the genotypic ID which corresponds exactly to the column name in the genotypic matrix. The 'Call' column contains the marker call which corresponds to the gene for that genotype. Other information may be present in this dataframe beyond these columns, but the three listed columns above are obligatory.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_gene_name">gene_name</code></td>
<td>
<p>A character string which matches the name of the gene which you are trying to perform cross validation for. This character string must be present in your gene_file 'Gene' column.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_marker_info">marker_info</code></td>
<td>
<p>A dataframe containing the following three columns: 'Marker', 'Chromosome', and 'BP_Position'. The 'Marker' column contains the names of the marker which are present in the genotypic matrix. The 'Chromosome' column contains the corresponding chromosome/linkage group to which the marker belongs. The 'Position' column contains the physical or centimorgan position of the marker. All markers present in the genotypic matrix must be listed in this dataframe. If physical or centimorgan positions are unavailable for the listed markers, a numeric dummy variable ranging from one to n number of markers may be provided instead.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_chromosome">chromosome</code></td>
<td>
<p>A character string which matches the name of the chromosome upon which the gene resides. This chromosome name must be present in the marker_info file.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_ncor_markers">ncor_markers</code></td>
<td>
<p>A numeric variable which represents the number of markers the user want to use in model training. Correlation among markers to the gene call is calculated and the top n markers specified are retained for training. The default setting is 50 markers.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_n_neighbors">n_neighbors</code></td>
<td>
<p>A numeric variable which represents the number of neighbors to use in KNN. Default is 50.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_percent_testing">percent_testing</code></td>
<td>
<p>A numeric variable which ranges such that x|0&lt;x&lt;1. This means that this number can be neither zero nor one. This number represents the percent of the total data available the user wants to retain to validate the model. The default setting is 0.20.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_percent_training">percent_training</code></td>
<td>
<p>A numeric variable which ranges such that x|0&lt;x&lt;1. This means that the number can be neither zero nor one. This number represents the percent of the total data available the user wants to retain for training of the model.The default setting is 0.80.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_include_hets">include_hets</code></td>
<td>
<p>A logical variable which determines if the user wishes to include heterozygous calls or not. The default setting is FALSE.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_include_models">include_models</code></td>
<td>
<p>A logical variable which determines if the user wishes to include the trained models in the results object for further testing. Warning: the models are quite large and running this will result in a very large results object. The default setting is FALSE.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_verbose">verbose</code></td>
<td>
<p>A logical variable which determines if the user wants plots displayed and text feedback from each permutation. Regardless of this parameter, the function will display the name of the gene which is being cross validated and the current progress of the permutations. Default setting is FALSE.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_parallel">parallel</code></td>
<td>
<p>A logical variable which determines if the user wants the cross validation performed in parallel. Default is FALSE. If the user defines that parallel is TRUE, all visual and textual feedback will not be rendered.</p>
</td></tr>
<tr><td><code id="locus_perm_cv_+3A_n_cores">n_cores</code></td>
<td>
<p>A numerical vector which denotes the number of cores used for parallel processor. If &quot;parallel&quot; option is TRUE and n_cores is not specified, then the number of available cores minus one will be assigned to processing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a list of list with the following objects: &quot;Overall_Parameters&quot;, &quot;By_Class_Parameters&quot;, &quot;Overall_Summary&quot;, &quot;By_Class_Summary&quot;, and &quot;Raw_Permutation_Info&quot;. The &quot;Overall_Parameters&quot; data frame contains all the relevant parameters for each permutation overall. The &quot;By_Class_Parameters&quot; data frame contains all the relevant parameters for each permutation by class.The &quot;Overall_Summary&quot; data frame contains all the relevant parameters overall summarized across permutations. The &quot;By_Class_Summary&quot; data frame contains all the relevant parameters by class summarized across permutations. The &quot;Raw_Permutation_Info&quot; is a list of list which contains each permutations model info as described in the &quot;locus_cv&quot; function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#read in the genotypic data matrix
data("geno_mat")

#read in the marker information
data("marker_info")

#read in the gene compendium file
data("gene_comp")

#run permutational analysis - commented out for package specifications
#to run, copy and paste without '#' into the console

#fit&lt;-locus_perm_cv(n_perms = 10, #the number of permutations
#                   geno_mat=geno_mat, #the genotypic matrix
#                   gene_file=gene_comp, #the gene compendium file
#                   gene_name="sst1_solid_stem", #the name of the gene
#                   marker_info=marker_info, #the marker information file
#                   chromosome="3B", #name of the chromosome
#                   ncor_markers= 25, #number of markers to retain
#                   n_neighbors = 25, #number of nearest-neighbors
#                   percent_testing=0.2, #percentage of genotypes in the validation set
#                   percent_training=0.8, #percentage of genotypes in the training set
#                   include_hets=FALSE, #excludes hets in the model
#                   include_models=FALSE, #excludes models in results object
#                   verbose = FALSE) #excludes text


</code></pre>

<hr>
<h2 id='locus_pred'>Haplotype Prediction: Using Trained Models to Make Predictions</h2><span id='topic+locus_pred'></span>

<h3>Description</h3>

<p>Haplotype Prediction: Using Trained Models to Make Predictions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locus_pred(locus_train_results, geno_mat, genotypes_to_predict)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="locus_pred_+3A_locus_train_results">locus_train_results</code></td>
<td>
<p>This object is a the results object from the &quot;locus_train&quot; function.</p>
</td></tr>
<tr><td><code id="locus_pred_+3A_geno_mat">geno_mat</code></td>
<td>
<p>This is a genotypic matrix with genotypes of individuals you have genotyped and characterized for the locus/QTL/gene of interest and individuals that have only been genotyped with a genome wide marker panel. It is important to note that the markers in the genome wide panel <em>must</em> be shared between training population and the population you wish to forward predict. In the case of genotyping-by-sequencing markers, the training and test populations should be discovered and produced together into a genotype file. All marker platforms, however, are compatable, as long as the training and forward prediction population share the same markers genome-wide.</p>
</td></tr>
<tr><td><code id="locus_pred_+3A_genotypes_to_predict">genotypes_to_predict</code></td>
<td>
<p>This is a character vector of genotypes in the geno_mat which the user wishes to predict. If this object contains names in the training population, they will be omitted in the results to avoid bias.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with two columns: genotype names and predictions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#set seed for reproducible sampling
set.seed(022294)

#read in the genotypic data matrix
data("geno_mat")

#read in the marker information
data("marker_info")

#read in the gene compendium file
data("gene_comp")

#Note: in practice you would have something like a gene file
#that does not contain any lines you are trying to predict.
#However, this is for illustrative purposes on how to run the function

#sample data in the gene_comp file to make a traning population
train&lt;-gene_comp[gene_comp$FullSampleName %in%
                   sample(gene_comp$FullSampleName,
                          round(length(gene_comp$FullSampleName)*0.8),0),]

#pull vector of names, not in the train, for forward prediction
test&lt;-gene_comp[!gene_comp$FullSampleName
                %in% train$FullSampleName,
                "FullSampleName"]

#run the function with hets
fit&lt;-locus_train(geno_mat=geno_mat, #the genotypic matrix
                 gene_file=train, #the gene compendium file
                 gene_name="sst1_solid_stem", #the name of the gene
                 marker_info=marker_info, #the marker information file
                 chromosome="3B", #name of the chromosome
                 ncor_markers=2, #number of markers to retain
                 n_neighbors=3, #number of neighbors
                 include_hets=FALSE, #include hets in the model
                 verbose = FALSE, #allows for text and graph output
                 set_seed = 022294, #sets a seed for reproduction of results
                 models = "knn") #sets what models are requested

#predict the lines in the test population
pred&lt;-locus_pred(locus_train_results=fit,
                 geno_mat=geno_mat,
                 genotypes_to_predict=test)

#see predictions
head(pred)

</code></pre>

<hr>
<h2 id='locus_train'>Haplotype Prediction: Training Models for Use in Forward Prediction</h2><span id='topic+locus_train'></span>

<h3>Description</h3>

<p>This function is used to train a model for use in forward prediction of lines which have no record
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locus_train(
  geno_mat,
  gene_file,
  gene_name,
  marker_info,
  chromosome,
  ncor_markers = 50,
  n_neighbors = 50,
  include_hets = FALSE,
  verbose = FALSE,
  set_seed = NULL,
  models_request = "all",
  graph = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="locus_train_+3A_geno_mat">geno_mat</code></td>
<td>
<p>An imputed, number-coded, genotypic matrix which has n rows of individuals and m columns of markers. Row names of the matrix should be representative of genotypic IDs and column names should be representative of marker IDs. Missing data is not allowed. Numeric coding of genotypes can vary as long as it remains consistant among markers.</p>
</td></tr>
<tr><td><code id="locus_train_+3A_gene_file">gene_file</code></td>
<td>
<p>A dataframe containing at least three columns labeled as follows: 'Gene', 'FullSampleName', and 'Call'. The 'Gene' column contains the name of the gene for which the observation belongs to. The 'FullSampleName' column contains the genotypic ID which corresponds exactly to the column name in the genotypic matrix. The 'Call' column contains the marker call which corresponds to the gene for that genotype. Other information may be present in this dataframe beyond these columns, but the three listed columns above are obligatory.</p>
</td></tr>
<tr><td><code id="locus_train_+3A_gene_name">gene_name</code></td>
<td>
<p>A character string which matches the name of the gene which you are trying to perform cross validation for. This character string must be present in your gene_file 'Gene' column.</p>
</td></tr>
<tr><td><code id="locus_train_+3A_marker_info">marker_info</code></td>
<td>
<p>A dataframe containing the following three columns: 'Marker', 'Chromosome', and 'BP_Position'. The 'Marker' column contains the names of the marker which are present in the genotypic matrix. The 'Chromosome' column contains the corresponding chromosome/linkage group to which the marker belongs. The 'Position' column contains the physical or centimorgan position of the marker. All markers present in the genotypic matrix must be listed in this dataframe. If physical or centimorgan positions are unavailable for the listed markers, a numeric dummy variable ranging from one to n number of markers may be provided instead.</p>
</td></tr>
<tr><td><code id="locus_train_+3A_chromosome">chromosome</code></td>
<td>
<p>A character string which matches the name of the chromosome upon which the gene resides. This chromosome name must be present in the marker_info file.</p>
</td></tr>
<tr><td><code id="locus_train_+3A_ncor_markers">ncor_markers</code></td>
<td>
<p>A numeric variable which represents the number of markers the user want to use in model training. Correlation among markers to the gene call is calculated and the top n markers specified are retained for training. The default setting is 50 markers.</p>
</td></tr>
<tr><td><code id="locus_train_+3A_n_neighbors">n_neighbors</code></td>
<td>
<p>A numeric variable which represents the number of neighbors to use in KNN. Default is 50.</p>
</td></tr>
<tr><td><code id="locus_train_+3A_include_hets">include_hets</code></td>
<td>
<p>A logical variable which determines if the user wishes to include heterozygous calls or not. The default setting is FALSE.</p>
</td></tr>
<tr><td><code id="locus_train_+3A_verbose">verbose</code></td>
<td>
<p>A logical variable which determines if the user wants text feedback. Default setting is TRUE.</p>
</td></tr>
<tr><td><code id="locus_train_+3A_set_seed">set_seed</code></td>
<td>
<p>A numeric variable that is used to set a seed for reproducible results if the user is running the function once for use in the &quot;locus_pred&quot; function. If the user wishes to run the function many times with a random seed and decide the outcome by voting, use the function &quot;locus_voting&quot; instead. The default setting is NULL.</p>
</td></tr>
<tr><td><code id="locus_train_+3A_models_request">models_request</code></td>
<td>
<p>A character string which defines what models are to be ran. K-nearest neighbors is abbreviated as &quot;knn&quot; and random forest is &quot;rf&quot;. If both models are desired, use the text string &quot;all&quot;. Default setting is &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="locus_train_+3A_graph">graph</code></td>
<td>
<p>A logical variable which determines if the user wants graphs displayed. default setting is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a list of list which contains: &quot;seed&quot;, &quot;models_request&quot; ,&quot;models&quot;, and &quot;data&quot;. The &quot;seed&quot; object is the seed set by the user. If no seed was provided this will appear as a character stating &quot;no_seed_set&quot;. The &quot;models_request&quot; item hold the models requested. The &quot;models&quot; object contains the trained models. The &quot;data&quot; object contains the data used to train the models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#set seed for reproducible sampling
set.seed(022294)

#read in the genotypic data matrix
data("geno_mat")

#read in the marker information
data("marker_info")

#read in the gene compendium file
data("gene_comp")

#Note: in practice you would have something like a gene file
#that does not contain any lines you are trying to predict.
#However, this is for illustrative purposes on how to run the function

#sample data in the gene_comp file to make a traning population
train&lt;-gene_comp[gene_comp$FullSampleName %in%
                   sample(gene_comp$FullSampleName,
                          round(length(gene_comp$FullSampleName)*0.8),0),]

#pull vector of names, not in the train, for forward prediction
test&lt;-gene_comp[!gene_comp$FullSampleName
                %in% train$FullSampleName,
                "FullSampleName"]

#run the function with hets
fit&lt;-locus_train(geno_mat=geno_mat, #the genotypic matrix
                 gene_file=train, #the gene compendium file
                 gene_name="sst1_solid_stem", #the name of the gene
                 marker_info=marker_info, #the marker information file
                 chromosome="3B", #name of the chromosome
                 ncor_markers=2, #number of markers to retain
                 n_neighbors=3, #number of neighbors
                 include_hets=FALSE, #include hets in the model
                 verbose = FALSE, #allows for text and graph output
                 set_seed = 022294, #sets a seed for reproduction of results
                 models = "knn") #sets what models are requested

#predict the lines in the test population
pred&lt;-locus_pred(locus_train_results=fit,
                 geno_mat=geno_mat,
                 genotypes_to_predict=test)

#see predictions
head(pred)

</code></pre>

<hr>
<h2 id='marker_info'>Model Gene Compendium Data Set</h2><span id='topic+marker_info'></span>

<h3>Description</h3>

<p>A data frame which contains marker information of GBS markers found on wheat chromosome 3B. This data pairs with the markers found in &quot;geno_mat&quot; data file associated with the HaploCatcher package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>marker_info
</code></pre>


<h3>Format</h3>

<p>A data frame with 2271 rows and 3 columns:
</p>

<dl>
<dt>Marker</dt><dd><p>The designation of the markers which are found in the genotypic matrix</p>
</dd>
<dt>Chromosome</dt><dd><p>The chromosome where each marker resides</p>
</dd>
<dt>BP_Position</dt><dd><p>The position of each marker in basepairs</p>
</dd>
</dl>



<h3>Source</h3>

<p>Generated by Zachary James Winn for the CSU breeding program via historical in-house GBS data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("marker_info") #lazy loads the dataset for use in the package
</code></pre>

<hr>
<h2 id='plot_locus_perm_cv'>Visualize Permutation CV Results</h2><span id='topic+plot_locus_perm_cv'></span>

<h3>Description</h3>

<p>This function takes a list of list object from the &quot;locus_perm_cv&quot; function and creates a summary graphic of the accuracy, kappa, sensitivity, and specificity of the models ran. If heterozygous individuals were left in the cross validation scheme, by-class sensitivity and specificity will be displayed; otherwise, displayed parameters will be of the overall model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_locus_perm_cv(results, individual_images = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_locus_perm_cv_+3A_results">results</code></td>
<td>
<p>An object of class &quot;list&quot; which is derived from the &quot;locus_perm_cv&quot; function.</p>
</td></tr>
<tr><td><code id="plot_locus_perm_cv_+3A_individual_images">individual_images</code></td>
<td>
<p>A logical argument that defines if the user wants both the composite image and the full image. Default setting is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints a ggplot2 image
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#refer to vignette for an in depth look at the plot_locus_perm_cv function
vignette("An_Intro_to_HaploCatcher", package = "HaploCatcher")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
