<!DOCTYPE html><html><head><title>Help for package haploR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {haploR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#as.num'><p>Converts vector of strings to numeric vector</p></a></li>
<li><a href='#getExtendedView'><p>This function queries HaploReg web-based tool</p>
in order to Extended view for SNP of interest</a></li>
<li><a href='#getStudyList'><p>This function queries HaploReg web-based tool</p>
in order to see a list of GWAS.</a></li>
<li><a href='#queryHaploreg'><p>This function queries HaploReg web-based tool and returns results.</p></a></li>
<li><a href='#queryRegulome'><p>This function queries RegulomeDB web-based tool</p>
and returns results in a data frame.</a></li>
<li><a href='#regulomeSearch'><p>This function queries RegulomeDB web-based tool</p>
and returns results in a data frame.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Query 'HaploReg', 'RegulomeDB'</td>
</tr>
<tr>
<td>Version:</td>
<td>4.0.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-10-08</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ilya Y. Zhbannikov &lt;ilya.zhbannikov@duke.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A set of utilities for querying 
             'HaploReg' <a href="https://pubs.broadinstitute.org/mammals/haploreg/haploreg.php">https://pubs.broadinstitute.org/mammals/haploreg/haploreg.php</a>, 
             'RegulomeDB' <a href="https://www.regulomedb.org/regulome-search/">https://www.regulomedb.org/regulome-search/</a> 
             web-based tools. The package connects to 
             'HaploReg', 'RegulomeDB' searches and downloads results, without 
             opening web pages, directly from R environment. 
             Results are stored in a data frame that can be directly used in various 
             kinds of downstream analyses.</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown,openxlsx</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>httr,XML,tibble,RUnit,plyr,DT,RCurl,RJSONIO,methods</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Author:</td>
<td>Ilya Y. Zhbannikov [aut, cre],
  Konstantin G. Arbeev [aut],
  Anatoliy I. Yashin [aut]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-08 22:36:12 UTC; iz12</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-09 13:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='as.num'>Converts vector of strings to numeric vector</h2><span id='topic+as.num'></span>

<h3>Description</h3>

<p>Converts vector of strings to numeric vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.num(x, na.strings = "NA")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.num_+3A_x">x</code></td>
<td>
<p>Input vector of strings.</p>
</td></tr>
<tr><td><code id="as.num_+3A_na.strings">na.strings</code></td>
<td>
<p>A string which represents <code>NA</code>.
Default: &quot;NA&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(haploR)
as.num(c("1", "2", "X"), na.strings="X")
</code></pre>

<hr>
<h2 id='getExtendedView'>This function queries HaploReg web-based tool 
in order to Extended view for SNP of interest</h2><span id='topic+getExtendedView'></span>

<h3>Description</h3>

<p>This function queries HaploReg web-based tool 
in order to Extended view for SNP of interest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExtendedView(snp, url = Haploreg.settings[["extended.view.url"]])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getExtendedView_+3A_snp">snp</code></td>
<td>
<p>A SNP of interest.</p>
</td></tr>
<tr><td><code id="getExtendedView_+3A_url">url</code></td>
<td>
<p>A url to HaploReg.
Default: &lt;https://pubs.broadinstitute.org/mammals/haploreg/detail_v4.1.php?query=&amp;id=&gt;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of tables t1, t2, ..., etc 
depending on information contained in HaploReg database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tables &lt;- getExtendedView(snp="rs10048158")
tables
</code></pre>

<hr>
<h2 id='getStudyList'>This function queries HaploReg web-based tool 
in order to see a list of GWAS.</h2><span id='topic+getStudyList'></span>

<h3>Description</h3>

<p>This function queries HaploReg web-based tool 
in order to see a list of GWAS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getStudyList(url = Haploreg.settings[["study.url"]])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getStudyList_+3A_url">url</code></td>
<td>
<p>A url to HaploReg.
Default: &lt;https://pubs.broadinstitute.org/mammals/haploreg/haploreg.php&gt;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of studies. Each study is itself a list
of two: <code>name</code>, <code>id</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>studies &lt;- getStudyList()
studies
</code></pre>

<hr>
<h2 id='queryHaploreg'>This function queries HaploReg web-based tool and returns results.</h2><span id='topic+queryHaploreg'></span>

<h3>Description</h3>

<p>This function queries HaploReg web-based tool and returns results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>queryHaploreg(
  query = NULL,
  file = NULL,
  study = NULL,
  ldThresh = 0.8,
  ldPop = "EUR",
  epi = "vanilla",
  cons = "siphy",
  genetypes = "gencode",
  url = Haploreg.settings[["base.url"]],
  timeout = 100,
  encoding = "UTF-8",
  querySNP = FALSE,
  fields = NULL,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="queryHaploreg_+3A_query">query</code></td>
<td>
<p>Query (a vector of rsIDs).</p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_file">file</code></td>
<td>
<p>A text file (one refSNP ID per line).</p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_study">study</code></td>
<td>
<p>A particular study. See function <code>getHaploRegStudyList(...)</code>.
Default: <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_ldthresh">ldThresh</code></td>
<td>
<p>LD threshold, r2 (select NA to only show query variants). 
Default: 0.8.</p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_ldpop">ldPop</code></td>
<td>
<p>1000G Phase 1 population for LD calculation. 
Can be: &quot;AFR&quot;, &quot;AMR&quot;, &quot;ASN&quot;. Default: &quot;EUR&quot;.</p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_epi">epi</code></td>
<td>
<p>Source for epigenomes. 
Possible values: <code>vanilla</code> for ChromHMM (Core 15-state model);
<code>imputed</code> for ChromHMM (25-state model using 12 imputed marks);
<code>methyl</code> for H3K4me1/H3K4me3 peaks;
<code>acetyl</code> for H3K27ac/H3K9ac peaks.
Default: <code>vanilla</code>.</p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_cons">cons</code></td>
<td>
<p>Mammalian conservation algorithm. 
Possible values: <code>gerp</code> for GERP,
<code>siphy</code> for SiPhy-omega,
<code>both</code> for both.
Default: <code>siphy</code>.</p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_genetypes">genetypes</code></td>
<td>
<p>Show position relative to.
Possible values: <code>gencode</code> for Gencode genes;
<code>refseq</code> for RefSeq genes;
<code>both</code> for both.
Default: <code>gencode</code>.</p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_url">url</code></td>
<td>
<p>HaploReg url address. 
Default: &lt;https://pubs.broadinstitute.org/mammals/haploreg/haploreg.php&gt;</p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_timeout">timeout</code></td>
<td>
<p>A <code>timeout</code> parameter for <code>curl</code>.
Default: 100</p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_encoding">encoding</code></td>
<td>
<p>sets the <code>encoding</code> for correct retrieval web-page content.
Default: <code>UTF-8</code></p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_querysnp">querySNP</code></td>
<td>
<p>A flag indicating to return query SNPs only. 
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_fields">fields</code></td>
<td>
<p>A set of fields to extract. Refer to the package vignette 
for available fields. Default: <code>All</code>.</p>
</td></tr>
<tr><td><code id="queryHaploreg_+3A_verbose">verbose</code></td>
<td>
<p>Verbosing output. Default: FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame (table) with results similar to 
HaploReg uses.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(haploR)
data &lt;- queryHaploreg(c("rs10048158","rs4791078"))
head(data)
</code></pre>

<hr>
<h2 id='queryRegulome'>This function queries RegulomeDB web-based tool 
and returns results in a data frame.</h2><span id='topic+queryRegulome'></span>

<h3>Description</h3>

<p>This function queries RegulomeDB web-based tool 
and returns results in a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>queryRegulome(
  query = NULL,
  genomeAssembly = "GRCh37",
  limit = 1000,
  timeout = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="queryRegulome_+3A_query">query</code></td>
<td>
<p>Query (a vector of rsIDs or exact one query region in rsid or like &quot;chr1:39492461-39492462&quot;).</p>
</td></tr>
<tr><td><code id="queryRegulome_+3A_genomeassembly">genomeAssembly</code></td>
<td>
<p>Genome assembly built: can be GRCh37 (default) or GRCh38.</p>
</td></tr>
<tr><td><code id="queryRegulome_+3A_limit">limit</code></td>
<td>
<p>It controls how many variants will be queried and returned for a large region. It can be a number (1000 by default) or &quot;all&quot;. 
Please note that large number or &quot;all&quot; may get yourself hurt because you could get timeout or may even crash the server.</p>
</td></tr>
<tr><td><code id="queryRegulome_+3A_timeout">timeout</code></td>
<td>
<p>A <code>timeout</code> parameter for <code>httr::GET</code>.
Default: 100</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame (table)
OR
a list with the following items:
- guery_coordinates
- features
- regulome_score
- variants
- nearby_snps,
- assembly
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- queryRegulome(c("rs4791078","rs10048158"))
head(data)
</code></pre>

<hr>
<h2 id='regulomeSearch'>This function queries RegulomeDB web-based tool 
and returns results in a data frame.</h2><span id='topic+regulomeSearch'></span><span id='topic+regulomeSummary'></span>

<h3>Description</h3>

<p>This function queries RegulomeDB web-based tool 
and returns results in a data frame.
</p>
<p>This function queries RegulomeDB web-based tool 
and returns results in a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>regulomeSearch(
  query = NULL,
  genomeAssembly = NULL,
  limit = 1000,
  timeout = 100
)

regulomeSummary(
  query = NULL,
  limit = 1000,
  genomeAssembly = NULL,
  timeout = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="regulomeSearch_+3A_query">query</code></td>
<td>
<p>Query (a vector of rsIDs).</p>
</td></tr>
<tr><td><code id="regulomeSearch_+3A_genomeassembly">genomeAssembly</code></td>
<td>
<p>Genome assembly built: can be GRCh37 or GRCh38</p>
</td></tr>
<tr><td><code id="regulomeSearch_+3A_limit">limit</code></td>
<td>
<p>It controls how many variants will be queried and returned for a large region. It can be a number (1000 by default) or &quot;all&quot;. 
Please note that large number or &quot;all&quot; may get yourself hurt because you could get timeout or may even crash the server.</p>
</td></tr>
<tr><td><code id="regulomeSearch_+3A_timeout">timeout</code></td>
<td>
<p>A <code>timeout</code> parameter for <code>httr::GET</code>.
Default: 100</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following items:
- guery_coordinates
- features
- regulome_score
- variants
- nearby_snps,
- assembly
</p>
<p>a data frame (table)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- regulomeSearch("rs4791078")
head(data)
data &lt;- regulomeSummary(c("rs4791078","rs10048158"))
head(data)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
