<!DOCTYPE html><html><head><title>Help for package HaploSim</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HaploSim}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AssignQTL'><p>Assigns QTL to a list of Haplotypes</p></a></li>
<li><a href='#buildhPedigree'><p>Build a haplotype pedrigree from a haplotype list</p></a></li>
<li><a href='#getAll'><p>Get alleles</p></a></li>
<li><a href='#haploList'><p>Function to create objects of class 'haploList'</p></a></li>
<li><a href='#haploList-class'><p>Class &quot;haploList&quot;</p></a></li>
<li><a href='#haplotype-class'><p>Class &quot;haplotype&quot; ~~~</p></a></li>
<li><a href='#hPed2Ped'><p>Transform a haplotype pedigree into a pedigree.</p></a></li>
<li><a href='#RemoveHomozygotes'><p>Remove the homozygotic snp loci from a list of haplotypes</p></a></li>
<li><a href='#SampleBaseHaplotype'><p>Simulate a base population haplotype</p></a></li>
<li><a href='#SampleHaplotype'><p>Simulate a haplotype</p></a></li>
<li><a href='#SampleHaplotypes'><p>Simulate a list of haplotypes</p></a></li>
<li><a href='#SamplePedigree'><p>Simulate a haplotypes in a pedigree</p></a></li>
<li><a href='#validhaploListObject'><p>Function to validate an object of class 'haploList'</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Functions to Simulate Haplotypes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.8.4.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-08-13</td>
</tr>
<tr>
<td>Depends:</td>
<td>methods,utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>pedigree (&ge; 1.4)</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulate haplotypes through meioses. Allows specification
        of population parameters.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Author:</td>
<td>Albart Coster [aut, cre],
  John Bastiaansen [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Albart Coster &lt;albart@dairyconsult.nl&gt;</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-13 20:53:01 UTC; Gebruiker</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-13 21:10:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='AssignQTL'>Assigns QTL to a list of Haplotypes</h2><span id='topic+AssignQTL'></span><span id='topic+ListQTL'></span>

<h3>Description</h3>

<p><code>AssignQTL</code> assigns QTL to the <code>qtl</code> of objects of class
<code>haplotype</code>. <code>ListQTL</code> returns a list of qtl which for the
object of class <code>haploList</code>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AssignQTL(hList,nqtl = NA,frqtl = NA,sigma2qtl = NULL,shQTL = 1,scQTL = 1,
                      nTraits = 1,overlap = 0,MAF = 0.1,rmCausSNP =
TRUE)
ListQTL(hList,nqtl = NA,frqtl = NA,sigma2qtl = NULL,shQTL = 1,scQTL = 1,
                      nTraits = 1,overlap = 0,MAF = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AssignQTL_+3A_hlist">hList</code></td>
<td>
<p>List of haplotype objects.</p>
</td></tr>
<tr><td><code id="AssignQTL_+3A_nqtl">nqtl</code></td>
<td>
<p>If specified, the number of qtl which are placed on the
genome.</p>
</td></tr>
<tr><td><code id="AssignQTL_+3A_frqtl">frqtl</code></td>
<td>
<p>If specified, the fraction of heterozygous SNP loci which
become QTL.</p>
</td></tr>
<tr><td><code id="AssignQTL_+3A_sigma2qtl">sigma2qtl</code></td>
<td>
<p>If specified, the qtl variance. If
<code>length(sigma2qtl)&lt;nqtl</code>, sigma2qtl is replicated until
obtaining a vector of appropriate length. Can be specified as a list
of length equal to nTraits.</p>
</td></tr>
<tr><td><code id="AssignQTL_+3A_shqtl">shQTL</code></td>
<td>
<p>If alpha is not specified, shQTL specifies the shape
parameter of the gamma distribution from which allele substitution
effects are sampled.</p>
</td></tr>
<tr><td><code id="AssignQTL_+3A_scqtl">scQTL</code></td>
<td>
<p>If alpha is not specified, scQTL specifies the scale
parameter of the gamma distribution from which allele substitution
effects are sampled.</p>
</td></tr>
<tr><td><code id="AssignQTL_+3A_ntraits">nTraits</code></td>
<td>
<p>The number of traits.</p>
</td></tr>
<tr><td><code id="AssignQTL_+3A_overlap">overlap</code></td>
<td>
<p>Numeric between 0 and 1. Specifies the percantage of
pleiotropic QTL.</p>
</td></tr>
<tr><td><code id="AssignQTL_+3A_maf">MAF</code></td>
<td>
<p>Minor Allele Frequency. Loci with maf below MAF are not
considered to become QTL. Do not count for frqtl.</p>
</td></tr>
<tr><td><code id="AssignQTL_+3A_rmcaussnp">rmCausSNP</code></td>
<td>
<p>Remove causative SNPs. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length <code>nHaplotpes</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SampleHaplotype">SampleHaplotype</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hList &lt;- SampleHaplotypes(nHaplotypes = 20,nLoc = 100,genDist =
1,nDec = 3) ## create objects
 hListd &lt;- SampleHaplotypes(orig = hList,genDist = 1,nDec = 3)
hListQTL &lt;- AssignQTL(hList,frqtl = 0.1,MAF = 0.0)
hListd &lt;- SampleHaplotypes(orig = hListQTL,genDist = 1,nDec
= 3,QTL = TRUE)
qtlList &lt;- ListQTL(hList,frqtl = 0.1,MAF = 0.0)
</code></pre>

<hr>
<h2 id='buildhPedigree'>Build a haplotype pedrigree from a haplotype list</h2><span id='topic+buildhPedigree'></span>

<h3>Description</h3>

<p>Builds a haplotype pedigree from a list of objects of class
<code>haplotype</code>. Objects have attributes <code>hID</code> and <code>phID0</code>,
<code>phID1</code>, the last two refer to the two haplotypes in the
parent. Function can construct a new pedigree or continue a given pedigree.
Simulate a list of haplotypes, either sampling using population
parameters or from a previous list of haplotypes through a series of
meioses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildhPedigree(hPedigree = NULL,hList)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="buildhPedigree_+3A_hpedigree">hPedigree</code></td>
<td>
<p>If unspecified, the previous haplotype pedigree upon
which buildhPed builds the additional haplotypes.</p>
</td></tr>
<tr><td><code id="buildhPedigree_+3A_hlist">hList</code></td>
<td>
<p>A <code>list</code> with objects of class <code>haplotype</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>buildhPedigree</code> uses the attributes <code>hID</code>,
<code>phID0</code>, <code>phID1</code> of objects of class <code>haplotype to
    build a pedigree of haplotypes. Each haplotype originates
    from a pair of parental haplotypes between which the meiosis event
    occured or has no known parental haplotypes.</code>
</p>


<h3>Value</h3>

<p>A <code>data.frame</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SampleHaplotypes">SampleHaplotypes</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example(SampleHaplotypes)
hPedigree &lt;- buildhPedigree(hList = hList)
for(g in 1:10)
  {
    hList &lt;- SampleHaplotypes(orig = hList,genDist
        = 1,roundDec = 3)
    hPedigree &lt;- buildhPedigree(hPedigree=hPedigree,hList = hList)
  }
</code></pre>

<hr>
<h2 id='getAll'>Get alleles</h2><span id='topic+getAll'></span>

<h3>Description</h3>

<p>Extract the sequence of snp alleles from a list of objects of
class <code>haplotype</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAll(hList,what = c("snp","qtl"),removeHomozygotes =
FALSE,translatePos = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAll_+3A_hlist">hList</code></td>
<td>
<p>List of objects of class <code>haplotype</code>.</p>
</td></tr>
<tr><td><code id="getAll_+3A_what">what</code></td>
<td>
<p>Specifies if snp or qtl alleles are extracted from
haplotype objects.</p>
</td></tr>
<tr><td><code id="getAll_+3A_removehomozygotes">removeHomozygotes</code></td>
<td>
<p>If TRUE, homozygote genotypes are
removed from the marker data.</p>
</td></tr>
<tr><td><code id="getAll_+3A_translatepos">translatePos</code></td>
<td>
<p>Translates marker positions to positions in
Morgan, else keeps the positions as integers.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function removes homozygous snp loci before extracting these.
</p>


<h3>Value</h3>

<p>Matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example(AssignQTL)
hh &lt;- getAll(hList = hList)
qq &lt;- getAll(hList = hList,what = 'qtl')
</code></pre>

<hr>
<h2 id='haploList'>Function to create objects of class 'haploList'</h2><span id='topic+haploList'></span>

<h3>Description</h3>

<p>Function to create objects of class <code>'haploList'</code>. Can eiter copy
the attributes of an earlier object of class <code>'haploList'</code> or
create a new <code>'haploList'</code> object. Function mostly used within
other functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haploList(list = NULL,hList = NULL,nDec,genDist,nChrom = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haploList_+3A_list">list</code></td>
<td>
<p>List of objects of class <code>'haplotype'</code>.</p>
</td></tr>
<tr><td><code id="haploList_+3A_hlist">hList</code></td>
<td>
<p>Object of class <code>'haploList'</code>.</p>
</td></tr>
<tr><td><code id="haploList_+3A_ndec">nDec</code></td>
<td>
<p>Number of decimal positions of new object.</p>
</td></tr>
<tr><td><code id="haploList_+3A_gendist">genDist</code></td>
<td>
<p>Genome size of new object measured in Morgan.</p>
</td></tr>
<tr><td><code id="haploList_+3A_nchrom">nChrom</code></td>
<td>
<p>Number of chromosomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>'haploList'</code>. If <code>list</code> is not NULL,
object of filled with objects of class <code>'haplotype'</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplotype-class">haplotype</a></code>,
<code><a href="#topic+haploList-class">haploList</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hList &lt;- haploList(nDec=  1,genDist = 1)
validhaploListObject(hList)
</code></pre>

<hr>
<h2 id='haploList-class'>Class &quot;haploList&quot;</h2><span id='topic+haploList-class'></span><span id='topic++5B+2ChaploList+2CANY+2Cmissing-method'></span><span id='topic+c+2ChaploList-method'></span><span id='topic+print+2ChaploList-method'></span><span id='topic+show+2ChaploList-method'></span>

<h3>Description</h3>

<p>Class definition of <code>haploList</code>. Extends lists to
contian objects of class <code>haplotype</code>. Attributes are <code>nDec</code>,
the number of decimal positions and <code>genDist</code>, the genome size
measured in Morgan.</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("haploList", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>.Data</code>:</dt><dd><p>Object of class <code>"list"</code> containing
objects of class <code>'haplotype'</code></p>
</dd>
<dt><code>genDist</code>:</dt><dd><p>Object of class <code>"numeric"</code>
expressing the genome size in Morgan.</p>
</dd>
<dt><code>nDec</code>:</dt><dd><p>Object of class <code>"integer"</code>
expressing the number of decimal positions of the
haplotypes.</p>
</dd>
<dt><code>nChrom</code>:</dt><dd><p>Object of class <code>"integer"</code>
expressing the number of chromosomes.</p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"<a href="methods.html#topic+list-class">list</a>"</code>, from data part.
Class <code>"<a href="methods.html#topic+vector-class">vector</a>"</code>, by class &quot;list&quot;, distance 2.
</p>


<h3>Methods</h3>


<dl>
<dt>[</dt><dd><p><code>signature(x = "haploList", i = "ANY", j =
	"missing")</code>: subset and extract from object of class <code>'haploList'</code></p>
</dd>
<dt>c</dt><dd><p><code>signature(x = "haploList")</code>: concatenate object of
class <code>'haploList'</code>.</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+SampleHaplotypes">SampleHaplotypes</a></code>,
<code><a href="#topic+haplotype-class">haplotype</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("haploList")
</code></pre>

<hr>
<h2 id='haplotype-class'>Class &quot;haplotype&quot; ~~~ </h2><span id='topic+haplotype-class'></span><span id='topic+print+2Chaplotype-method'></span><span id='topic+show+2Chaplotype-method'></span>

<h3>Description</h3>

<p>Definition of <code>'haplotype'</code> class.</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("haplotype", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>snp</code>:</dt><dd><p>Object of class <code>"integer"</code> containing the
positions on the genome where the haplotype has a 1 allele. The
number of snp positions is the genome size in Morgan times the
number of decimal positions, both stored in <code>'haploList'</code> objects. </p>
</dd>
<dt><code>qtl</code>:</dt><dd><p>Object of class <code>"list"</code>. Names of the list
are the QTL positions, on the same scale as snp positions. Objects
in the list are the QTL alleles.</p>
</dd>
<dt><code>hID</code>:</dt><dd><p>Object of class <code>"numeric"</code> identifying the
current object of class <code>'haplotype'</code>.</p>
</dd>
<dt><code>phID0</code>:</dt><dd><p>Object of class <code>"numeric"</code> pointing to
the first parental haplotype.</p>
</dd>
<dt><code>phID1</code>:</dt><dd><p>Object of class <code>"numeric"</code> pointing to
the second parental haplotype.</p>
</dd>
</dl>



<h3>Methods</h3>

<p>No methods defined with class &quot;haplotype&quot; in the signature.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SampleHaplotype">SampleHaplotype</a></code>,
<code><a href="#topic+SampleHaplotypes">SampleHaplotypes</a></code>,
<code><a href="#topic+haploList-class">haploList</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("haplotype") </code></pre>

<hr>
<h2 id='hPed2Ped'>Transform a haplotype pedigree into a pedigree.</h2><span id='topic+hPed2Ped'></span>

<h3>Description</h3>

<p>Transforms a haplotype pedigree into a pedigree. Individuals in a
haplotype pedigree are identified through meiosis. The number of rows
of the pedigree equals the number of unique combinations of haplotypes
plus the number of haplotypes which did not 'participate' in a meiosis
event. The latter haplotypes form individuals with only one haplotype
and thus only one parent.
Row number of the pedigree identifies individuals. The pedigree has
four columns. Columns 3 and 4 identify the haplotypes   of the
individual. Columns 1 and 2 identify parental individuals of   the
individuals by their row number in the pedigree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hPed2Ped(hPed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hPed2Ped_+3A_hped">hPed</code></td>
<td>
<p>The haplotype pedigree from which the pedigree is build.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SampleHaplotype">SampleHaplotype</a></code>, <code><a href="#topic+buildhPedigree">buildhPedigree</a></code>  
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example(buildhPedigree)
ped &lt;- hPed2Ped(hPedigree)
</code></pre>

<hr>
<h2 id='RemoveHomozygotes'>Remove the homozygotic snp loci from a list of haplotypes</h2><span id='topic+RemoveHomozygotes'></span>

<h3>Description</h3>

<p>Finds homozygotic marker loci in a list of haplotypes and removes
these from all haplotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RemoveHomozygotes(hList)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RemoveHomozygotes_+3A_hlist">hList</code></td>
<td>
<p>List of objects of class <code>haplotype</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of objects of class <code>haplotype</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SampleHaplotypes">SampleHaplotypes</a></code>, <code><a href="#topic+SampleHaplotype">SampleHaplotype</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hList &lt;- SampleHaplotypes(nHaplotypes = 20,nLoc = 100,genDist =
1,nDec = 3) ## create objects
hList &lt;- RemoveHomozygotes(hList)
</code></pre>

<hr>
<h2 id='SampleBaseHaplotype'>Simulate a base population haplotype</h2><span id='topic+SampleBaseHaplotype'></span>

<h3>Description</h3>

<p>Simulates a base population haplotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SampleBaseHaplotype(genDist,nDec,nLoc,pSnp = seq(0,1,length.out = nLoc))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SampleBaseHaplotype_+3A_gendist">genDist</code></td>
<td>
<p>Map size of the simulated genome in Morgan.</p>
</td></tr>
<tr><td><code id="SampleBaseHaplotype_+3A_ndec">nDec</code></td>
<td>
<p>Number of decimals until which marker positions are
rounded. Partially specifies marker density on the chromosome.</p>
</td></tr>
<tr><td><code id="SampleBaseHaplotype_+3A_nloc">nLoc</code></td>
<td>
<p>Maximal number of snp loci on genome, default is number of
available positions on genome.</p>
</td></tr>
<tr><td><code id="SampleBaseHaplotype_+3A_psnp">pSnp</code></td>
<td>
<p>Optional,if specified the snp positions. Default:
uniform distribution of loci over the whole genome. Consequence will
be that loci have allele frequency equal to one. If NULL, positions
will be randomly sampled for each haplotype and allele frequencies
will be close to 0, depending on <code class="reqn">nLoc / (genDist*10^{nDec})</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Generally called by function <code><a href="#topic+SampleHaplotypes">SampleHaplotypes</a></code>.
</p>


<h3>Value</h3>

<p>An object of class <code>haplotype</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SampleHaplotypes">SampleHaplotypes</a></code>,
<code><a href="#topic+SampleBaseHaplotype">SampleBaseHaplotype</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hList &lt;- sapply(1:10,function(x)SampleBaseHaplotype(genDist = 1,nDec
= 2,nLoc = 50))
</code></pre>

<hr>
<h2 id='SampleHaplotype'>Simulate a haplotype</h2><span id='topic+SampleHaplotype'></span>

<h3>Description</h3>

<p>Simulates a haplotype, either sampling using population
parameters or through a meiosis event with two parental haplotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SampleHaplotype(H0 = NULL,H1 = NULL,genDist,nDec,nChrom = 1,prMut =
1E-5,QTL = F,checkValidity = TRUE) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SampleHaplotype_+3A_h0">H0</code></td>
<td>
<p>If specified, the first parental haplotype.</p>
</td></tr>
<tr><td><code id="SampleHaplotype_+3A_h1">H1</code></td>
<td>
<p>If specified, the second parental haplotype. If neither H0
nor H1 are specified, a new haplotype is sampled from a base
population. Errors message is displayed when only one haplotype is
provided as meiosis occurs between two haplotypes.</p>
</td></tr>
<tr><td><code id="SampleHaplotype_+3A_gendist">genDist</code></td>
<td>
<p>Map size of the simulated genome in Morgan.</p>
</td></tr>
<tr><td><code id="SampleHaplotype_+3A_ndec">nDec</code></td>
<td>
<p>Number of decimals until which marker positions are
rounded. Partially specifies marker density on the chromosome.</p>
</td></tr>
<tr><td><code id="SampleHaplotype_+3A_nchrom">nChrom</code></td>
<td>
<p>Number of chromosomes, default at 1</p>
</td></tr>
<tr><td><code id="SampleHaplotype_+3A_prmut">prMut</code></td>
<td>
<p>Probability of marker bp mutation.</p>
</td></tr>
<tr><td><code id="SampleHaplotype_+3A_qtl">QTL</code></td>
<td>
<p>If TRUE, qtl alleles are inherited to the next
generation. See function <code><a href="#topic+AssignQTL">AssignQTL</a></code> for
assigning qtl to a list of haplotypes.</p>
</td></tr>
<tr><td><code id="SampleHaplotype_+3A_checkvalidity">checkValidity</code></td>
<td>
<p>If TRUE, tests if a pair of haplotypes is
compatible; e.g. if the number of traits in both is equal (or 0) and
if the sizes are equal.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Markers are continually spaced over the whole genome. Marker density
is specified in <code><a href="#topic+SampleBaseHaplotype">SampleBaseHaplotype</a></code>. Position
of 1 alleles is recorded and stored in @snp attribute of the object of
class <code>haplotype</code>.
If QTL is TRUE, haplotypes with QTL's need to be provided. If not,
nothing happens apart from mutations (same prob. as for single
basepairs).
Function <code>SampleHaplotype</code> is generally called by function
<code><a href="#topic+SampleHaplotypes">SampleHaplotypes</a></code>.
</p>


<h3>Value</h3>

<p>An object of class <code>haplotype</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SampleHaplotypes">SampleHaplotypes</a></code>,
<code><a href="#topic+SampleBaseHaplotype">SampleBaseHaplotype</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hList &lt;- SampleHaplotypes(nHaplotypes = 20,genDist =
1,nDec = 3,nLoc = 20) ## create objects
h &lt;- SampleHaplotype(H0 = hList[[1]],H1 = hList[[2]],genDist =
1,nDec = 3)
</code></pre>

<hr>
<h2 id='SampleHaplotypes'>Simulate a list of haplotypes</h2><span id='topic+SampleHaplotypes'></span>

<h3>Description</h3>

<p>Simulate a list of haplotypes, either sampling using population
parameters or from a previous list of haplotypes through a series of
meioses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SampleHaplotypes(orig = NULL,nHaplotypes = 10,nMeioses = 2,gg = NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SampleHaplotypes_+3A_orig">orig</code></td>
<td>
<p>If unspecified, the function samples base population
haplotypes. If specified, the function requests a list
containing objects of class <code>haplotype</code>.</p>
</td></tr>
<tr><td><code id="SampleHaplotypes_+3A_nhaplotypes">nHaplotypes</code></td>
<td>
<p>The requested number of haplotypes.</p>
</td></tr>
<tr><td><code id="SampleHaplotypes_+3A_nmeioses">nMeioses</code></td>
<td>
<p>The number of offspring from each individual. For details
see below.</p>
</td></tr>
<tr><td><code id="SampleHaplotypes_+3A_gg">gg</code></td>
<td>
<p>If specified, the combinations of haplotypes in
individuals. Meiosis only occur within individuals. If not
specified, individuals are sampled as random combinations of haplotypes.</p>
</td></tr>
<tr><td><code id="SampleHaplotypes_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to function
<code>SampleHaplotype</code>, <code>haploList</code> and <code>SampleBaseHaplotype</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>SampleHaplotypes</code> creates individuals by randomly
combining haplotypes from the list. Meiosis events in individuals
create new haplotypes. Argument <code>nOff</code> sets the number of meiosis
events within each individual. Argument <code>nHaplotypes</code> has only
effect when sampling a base population. See
<code><a href="#topic+SampleBaseHaplotype">SampleBaseHaplotype</a></code> for sampling base
haplotypes and for arguments of this function.
</p>


<h3>Value</h3>

<p>A list of length <code>nHaplotpes</code> or <code>nOff * length(orig)</code> of objects of class <code>haplotype</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SampleHaplotype">SampleHaplotype</a></code>,
<code><a href="#topic+SampleBaseHaplotype">SampleBaseHaplotype</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hList &lt;- SampleHaplotypes(nHaplotypes = 20,nLoc = 100,genDist =
1,nDec = 3) ## create objects
for(g in 1:10)hList &lt;- SampleHaplotypes(orig = hList,genDist
= 1,nDec = 3)
</code></pre>

<hr>
<h2 id='SamplePedigree'>Simulate a haplotypes in a pedigree</h2><span id='topic+SamplePedigree'></span>

<h3>Description</h3>

<p>Simulates haplotypes within a given pedigree. Haplotypes for a base
individual are sampled from a list of base haplotypes. Parameters for
sampling haplotypes are passed to function <code>SampleHaplotype</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SamplePedigree(orig,ped,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SamplePedigree_+3A_orig">orig</code></td>
<td>
<p>List of objects of class <code>haplotype</code>. Haplotypes for
base individuals are the result of a meiosis event between two
haplotypes in this list.</p>
</td></tr>
<tr><td><code id="SamplePedigree_+3A_ped">ped</code></td>
<td>
<p><code>data.frame</code> of three columns. Column 1 contains id's,
column two and three id's of parental individuals. Pedigree is
first ordered with function <code>orderPed</code> from the package
<code>pedigree</code>.</p>
</td></tr>
<tr><td><code id="SamplePedigree_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to function <code>SampleHaplotype</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Samples haplotypes for individuals in a pedigree and returns a
pedigree with two additional columns which identify the two haplotypes
of an individual together with a list of haplotypes. Uses function
<code>SampleHaplotype</code> to sample a meiosis event between two parental
haplotypes.
</p>


<h3>Value</h3>

<p>A list with a pedigree and a list of objects of class
<code>haplotype</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SampleHaplotype">SampleHaplotype</a></code>,
<code><a href="pedigree.html#topic+orderPed">orderPed</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example(SampleHaplotypes)
ID &lt;- 1:10
pID0 &lt;- c(rep(0,5),1,1,3,3,5)
pID1 &lt;- c(rep(0,4),2,2,2,4,4,6)
ped &lt;- data.frame(ID,pID0,pID1)
phList &lt;- SamplePedigree(orig = hList,ped = ped)
</code></pre>

<hr>
<h2 id='validhaploListObject'>Function to validate an object of class 'haploList'</h2><span id='topic+validhaploListObject'></span>

<h3>Description</h3>

<p>Checks of object is of class <code>'haploList'</code> and checks if all
entries in <code>object@.Data</code> are of class
<code>'haplotype'</code>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validhaploListObject(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="validhaploListObject_+3A_object">object</code></td>
<td>
<p>Any object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haploList-class">haploList</a></code>,  <code><a href="#topic+haplotype-class">haplotype</a></code> 
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
