<!DOCTYPE html><html><head><title>Help for package hapsim</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hapsim}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ACEdata'><p>ACE data set</p></a></li>
<li><a href='#allelefreqs'><p>Estimates allele frequencies</p></a></li>
<li><a href='#checkpd'><p>Checks that a covariance matrix is positive definite</p></a></li>
<li><a href='#cor2dprime'><p>Correlations to D' coefficients conversion</p></a></li>
<li><a href='#cortocov'><p>Correlation to covariance conversion</p></a></li>
<li><a href='#covtocor'><p>Covariance to correlation conversion</p></a></li>
<li><a href='#divlocus'><p>Diversity score</p></a></li>
<li><a href='#haplodata'><p>Haplotype object creator</p></a></li>
<li><a href='#haplofreqs'><p> Haplotype frequencies</p></a></li>
<li><a href='#haplosim'><p>Haplotype data simulator</p></a></li>
<li><a href='#ldplot'><p>LD plot</p></a></li>
<li><a href='#makepd'><p>Makes a matrix positive definite</p></a></li>
<li><a href='#mergemats'><p>Merges two LD matrices</p></a></li>
<li><a href='#mse'><p>Mean Square Error</p></a></li>
<li><a href='#sumsqscale'><p> Scales a matrix</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Haplotype Data Simulation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.31</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-06-05</td>
</tr>
<tr>
<td>Author:</td>
<td>Giovanni Montana</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Apostolos Dimitromanolakis &lt;apostolis@live.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Package for haplotype-based genotype simulations. Haplotypes are
        generated such that their allele frequencies and linkage
        disequilibrium coefficients match those estimated from an input
        data set.</td>
</tr>
<tr>
<td>Depends:</td>
<td>MASS</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-06-07 21:40:03 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-06-07 16:09:30 UTC; apo</td>
</tr>
</table>
<hr>
<h2 id='ACEdata'>ACE data set</h2><span id='topic+ACEdata'></span>

<h3>Description</h3>

<p>ACE (angiotensin I converting enzyme) data set</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ACEdata)</code></pre>


<h3>Format</h3>

<p>A data set with 22 haplotypes and 52 SNPs.</p>


<h3>References</h3>

<p>Montana, G. HapSim: a simulation tool for generating haplotype data with
pre-specified allele frequencies and LD coefficients. 2005.
</p>

<hr>
<h2 id='allelefreqs'>Estimates allele frequencies</h2><span id='topic+allelefreqs'></span>

<h3>Description</h3>

<p>Estimates allele frequencies from a binary matrix</p>


<h3>Usage</h3>

<pre><code class='language-R'>allelefreqs(dat)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="allelefreqs_+3A_dat">dat</code></td>
<td>
<p> A binary matrix, rows are haplotypes and columns are binary markers </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr><td><code>freqs</code></td>
<td>
<p>Vector of allele &quot;0&quot; frequencies</p>
</td></tr>
<tr><td><code>all.polym</code></td>
<td>
<p>If TRUE, all loci are polymorphic</p>
</td></tr>
<tr><td><code>non.polym</code></td>
<td>
<p>Vector of non-polymorphic loci, if any</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Giovanni Montana</p>


<h3>References</h3>

<p>Montana, G. HapSim: a simulation tool for generating haplotype data with
pre-specified allele frequencies and LD coefficients. 2005.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(ACEdata)
x &lt;- allelefreqs(ACEdata)
hist(x$freqs)

</code></pre>

<hr>
<h2 id='checkpd'>Checks that a covariance matrix is positive definite</h2><span id='topic+checkpd'></span>

<h3>Description</h3>

<p>Checks that a matrix is positive definite
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkpd(sigma, tol = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkpd_+3A_sigma">sigma</code></td>
<td>
<p>Covariance matrix</p>
</td></tr>
<tr><td><code id="checkpd_+3A_tol">tol</code></td>
<td>
<p>Tolerance value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If TRUE, the input matrix is positive definite.
</p>

<hr>
<h2 id='cor2dprime'>Correlations to D' coefficients conversion</h2><span id='topic+cor2dprime'></span>

<h3>Description</h3>

<p>Converts a matrix of correlation coefficients into a D' matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cor2dprime(mat, probs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cor2dprime_+3A_mat">mat</code></td>
<td>
<p>A correlation matrix</p>
</td></tr>
<tr><td><code id="cor2dprime_+3A_probs">probs</code></td>
<td>
<p>A vector of allele frequencies</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The resulting matrix of D' coefficients
</p>


<h3>Author(s)</h3>

<p>Giovanni Montana</p>


<h3>References</h3>

<p>Montana, G. HapSim: a simulation tool for generating haplotype data with
pre-specified allele frequencies and LD coefficients. 2005.
</p>

<hr>
<h2 id='cortocov'>Correlation to covariance conversion</h2><span id='topic+cortocov'></span>

<h3>Description</h3>

<p>Converts a correlation matrix into a covariance matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cortocov(cor.mat, sd)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cortocov_+3A_cor.mat">cor.mat</code></td>
<td>
<p>Correlation matrix</p>
</td></tr>
<tr><td><code id="cortocov_+3A_sd">sd</code></td>
<td>
<p>Vector of standard devations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The covariance matrix
</p>

<hr>
<h2 id='covtocor'>Covariance to correlation conversion</h2><span id='topic+covtocor'></span>

<h3>Description</h3>

<p>Converts a covariance matrix into a correlation matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covtocor(cov.mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covtocor_+3A_cov.mat">cov.mat</code></td>
<td>
<p>Covariance matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The correlation matrix
</p>

<hr>
<h2 id='divlocus'>Diversity score</h2><span id='topic+divlocus'></span>

<h3>Description</h3>

<p>Compute a measure of genetic diversity at each locus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>divlocus(dat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="divlocus_+3A_dat">dat</code></td>
<td>
<p> A binary matrix, rows are haplotypes and columns are binary markers </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements a measure of diversity for a locus <code class="reqn">j</code>
as in Clayton (2002). If <code class="reqn">z_ij</code> represents the allele <code class="reqn">j</code> of haplotype <code class="reqn">i</code>, for
<code class="reqn">i=1,...,N</code> and assuming that alleles are coded as 0 and 1, the
diversity measure can be written as
</p>
<p style="text-align: center;"><code class="reqn">D_j = 2*N( \sum_{i=1}^N z_{ij}^2 - (\sum_{i=1}^N z_{ij}) ^2 )</code>
</p>


<h3>Value</h3>

<p>A vector containing the diversity measure for all markers
</p>


<h3>Author(s)</h3>

<p>Giovanni Montana</p>


<h3>References</h3>

<p>D. Clayton. Choosing a set of haplotype tagging SNPs from a
larger set of diallelic loci. 2002. <a href="www-gene.cimr.cam.ac.uk/clayton/software/stata/htSNP/htsnp.pdf">www-gene.cimr.cam.ac.uk/clayton/software/stata/htSNP/htsnp.pdf</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ACEdata)
divlocus(ACEdata)
</code></pre>

<hr>
<h2 id='haplodata'>Haplotype object creator</h2><span id='topic+haplodata'></span>

<h3>Description</h3>

<p>Creates an haplotype data object needed for simulating haplotypes with <code>haplosim</code>. This object also contains some summary statistics about the real data.</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplodata(dat)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplodata_+3A_dat">dat</code></td>
<td>
<p> A binary matrix, rows are haplotypes and columns are binary markers </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr><td><code>freqs</code></td>
<td>
<p>Allele frequencies</p>
</td></tr>
<tr><td><code>cor</code></td>
<td>
<p>Correlation matrix (LD coefficients) </p>
</td></tr>
<tr><td><code>div</code></td>
<td>
<p>Locus-specific diversity measure </p>
</td></tr>
<tr><td><code>cov</code></td>
<td>
<p>Covariance matrix for the normal distribution</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Giovanni Montana</p>


<h3>References</h3>

<p>Montana, G. HapSim: a simulation tool for generating haplotype data with
pre-specified allele frequencies and LD coefficients. 2005.
</p>


<h3>See Also</h3>

<p>See also <code> <a href="#topic+haplosim">haplosim</a> </code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(ACEdata)

# creates the haplotype object
x &lt;- haplodata(ACEdata) 

# simulates 100 random haplotypes
y &lt;- haplosim(100, x) 

</code></pre>

<hr>
<h2 id='haplofreqs'> Haplotype frequencies </h2><span id='topic+haplofreqs'></span>

<h3>Description</h3>

<p>Compute haplotype frequencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplofreqs(dat, firstl, lastl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplofreqs_+3A_dat">dat</code></td>
<td>
<p> A binary matrix, rows are haplotypes and columns are binary markers </p>
</td></tr>
<tr><td><code id="haplofreqs_+3A_firstl">firstl</code></td>
<td>
<p> Position of the first locus </p>
</td></tr>
<tr><td><code id="haplofreqs_+3A_lastl">lastl</code></td>
<td>
<p> Position of the last locus </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of haplotype frequencies</p>


<h3>Author(s)</h3>

<p>Giovanni Montana</p>


<h3>References</h3>

<p>Montana, G. HapSim: a simulation tool for generating haplotype data with
pre-specified allele frequencies and LD coefficients. 2005.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ACEdata)
freqs &lt;- haplofreqs(ACEdata, 17, 22)
</code></pre>

<hr>
<h2 id='haplosim'>Haplotype data simulator</h2><span id='topic+haplosim'></span>

<h3>Description</h3>

<p>Generates a random sample of haplotypes, given an haplotype object created from a data set</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplosim(n, hap, which.snp = NULL, seed = NULL, force.polym = TRUE, summary = TRUE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplosim_+3A_n">n</code></td>
<td>
<p>Number of haplotypes to generate</p>
</td></tr>
<tr><td><code id="haplosim_+3A_hap">hap</code></td>
<td>
<p>Haplotype object created with <code>haplodata</code></p>
</td></tr>
<tr><td><code id="haplosim_+3A_which.snp">which.snp</code></td>
<td>
<p>A vector specifying which SNPs to include</p>
</td></tr> 
<tr><td><code id="haplosim_+3A_seed">seed</code></td>
<td>
<p>Seed for the random number generator</p>
</td></tr>
<tr><td><code id="haplosim_+3A_force.polym">force.polym</code></td>
<td>
<p>if TRUE, all loci are polymorphic</p>
</td></tr>
<tr><td><code id="haplosim_+3A_summary">summary</code></td>
<td>
<p>if TRUE, additional summary statistics are returned</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr><td><code>data</code></td>
<td>
<p>Simulated sample</p>
</td></tr>
<tr><td><code>freqs</code></td>
<td>
<p>Allele frequency vector</p>
</td></tr>
<tr><td><code>cor</code></td>
<td>
<p>Correlation matrix</p>
</td></tr>
<tr><td><code>div</code></td>
<td>
<p>Locus-specific diversity scores</p>
</td></tr>
<tr><td><code>mse.freqs</code></td>
<td>
<p>MSE of allele frequencies</p>
</td></tr>
<tr><td><code>mse.cor</code></td>
<td>
<p>MSE of correlations</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Giovanni Montana </p>


<h3>References</h3>

<p>Montana, G. HapSim: a simulation tool for generating haplotype data with
pre-specified allele frequencies and LD coefficients. 2005.
</p>


<h3>See Also</h3>

<p>See also <code> <a href="#topic+haplodata">haplodata</a> </code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
#
# Example 1
#

data(ACEdata)

# create the haplotype object 
x &lt;- haplodata(ACEdata) 

# simulates a first sample of 100 haplotypes using all markers
y1 &lt;- haplosim(100, x) 

# compares allele frequencies in real and simulated samples
plot(x$freqs, y1$freqs, title=paste("MSE:",y1$mse.freqs)); abline(a=0, b=1)

# compares LD coefficients in real and simulated samples
ldplot(mergemats(x$cor, y1$cor), ld.type='r') 

# simulates a second sample of 1000 haplotypes using the first 20 markers only
y2 &lt;- haplosim(1000, which.snp=seq(20), x) 

#
# Example 2
#

# simulate a sample of 500 haplotypes based on the ACE data set
set.seed(100)
data(ACEdata)
n &lt;- 500
x &lt;- haplodata(ACEdata)
y &lt;- haplosim(n, x)

# compute the haplotype frequencies
# an haplotype starts at markers 17 and ends at marker 22
freq1 &lt;- haplofreqs(ACEdata, 17, 22)
freq2 &lt;- haplofreqs(y$data, 17, 22)

# extract the set of haplotypic configurations that are shared
# by real and simulated data and their frequencies
commonhapls &lt;- intersect(names(freq1),names(freq2)) 
cfreq1 &lt;- freq1[commonhapls]
cfreq2 &lt;- freq2[commonhapls]

# compare real vs simulated haplotype frequencies
par(mar=c(10.1, 4.1, 4.1, 2.1), xpd=TRUE)
legend.text &lt;- names(cfreq1)
bp &lt;- barplot(cbind(cfreq1,cfreq2), main="Haplotype Frequencies",
       names.arg=c("Real","Simulated"), col=heat.colors(length(legend.text)))
legend(mean(range(bp)), -0.3, legend.text, xjust = 0.5,
       fill=heat.colors(length(legend.text)), horiz = TRUE)
chisq.test(x=n*cfreq2, p=cfreq1, simulate.p.value = TRUE, rescale.p = TRUE)

</code></pre>

<hr>
<h2 id='ldplot'>LD plot</h2><span id='topic+ldplot'></span>

<h3>Description</h3>

<p>Creates a linkage disequilibrium plot from a matrix of pair-wise LD coefficients</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldplot(ld.mat, ld.type, color = heat.colors(50), title = NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ldplot_+3A_ld.mat">ld.mat</code></td>
<td>
<p>A square matrix of LD coefficients</p>
</td></tr>
<tr><td><code id="ldplot_+3A_ld.type">ld.type</code></td>
<td>
<p>A character value specifying what coefficients are used as input: either 'r' for correlation coefficients or 'd' for D/Dprime scores</p>
</td></tr>
<tr><td><code id="ldplot_+3A_color">color</code></td>
<td>
<p>A range of colors to be used for drawing. Default is <code>heat.colors</code></p>
</td></tr>
<tr><td><code id="ldplot_+3A_title">title</code></td>
<td>
<p>Character string for the title of the plot</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Giovanni Montana</p>


<h3>References</h3>

<p>Montana, G. HapSim: a simulation tool for generating haplotype data with
pre-specified allele frequencies and LD coefficients. 2005.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(ACEdata)

# LD plot of ACEdata using r^2 coefficients
ldplot(cor(ACEdata), ld.type='r') 

</code></pre>

<hr>
<h2 id='makepd'>Makes a matrix positive definite</h2><span id='topic+makepd'></span>

<h3>Description</h3>

<p>Modifies a matrix so that it is positive definite
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makepd(mat, eig.tol = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makepd_+3A_mat">mat</code></td>
<td>
<p>Matrix</p>
</td></tr>
<tr><td><code id="makepd_+3A_eig.tol">eig.tol</code></td>
<td>
<p>Tolerance value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A positive-definite matrix
</p>

<hr>
<h2 id='mergemats'>Merges two LD matrices</h2><span id='topic+mergemats'></span>

<h3>Description</h3>

<p>Merges two LD matrices. It can be used to compare the LD coefficients estimated in the real and simulated data sets</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergemats(mat1, mat2)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mergemats_+3A_mat1">mat1</code></td>
<td>
<p>First square matrix</p>
</td></tr>
<tr><td><code id="mergemats_+3A_mat2">mat2</code></td>
<td>
<p>Second square matrix of same dimensions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The resulting matrix has upper triangular matrix from <code>mat1</code> and lower triangular matrix from <code>mat2</code>
</p>


<h3>Author(s)</h3>

<p>Giovanni Montana</p>


<h3>References</h3>

<p>Montana, G. HapSim: a simulation tool for generating haplotype data with
pre-specified allele frequencies and LD coefficients. 2005.
</p>

<hr>
<h2 id='mse'>Mean Square Error</h2><span id='topic+mse'></span>

<h3>Description</h3>

<p>Computes the Mean Square Error between two vectors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mse(x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mse_+3A_x">x</code></td>
<td>
<p>First vector</p>
</td></tr>
<tr><td><code id="mse_+3A_y">y</code></td>
<td>
<p>Second vector, must have same lenght</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The MSE score
</p>

<hr>
<h2 id='sumsqscale'> Scales a matrix </h2><span id='topic+sumsqscale'></span>

<h3>Description</h3>

<p>Scales the columns of a matrix so that the squared elements sum to unity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sumsqscale(mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sumsqscale_+3A_mat">mat</code></td>
<td>
<p>A matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The scaled matrix.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
