<!DOCTYPE html><html lang="en"><head><title>Help for package HardyWeinberg</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HardyWeinberg}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HardyWeinberg-package'><p>Statistical Tests and Graphics for Hardy-Weinberg Equilibrium</p></a></li>
<li><a href='#af'><p>Function to compute allele frequencies</p></a></li>
<li><a href='#AFtest'><p>Exact test of equality of allele frequencies for males and females</p></a></li>
<li><a href='#afx'><p>Function to compute X-chromosomal allele frequencies</p></a></li>
<li><a href='#agcounts'>
<p>Calculate allele and genotype counts for X-chromosomal markers</p></a></li>
<li><a href='#alleles'>
<p>Extract alleles</p></a></li>
<li><a href='#AllelesToTriangular'>
<p>Calculate triangular genotype matrix for vector(s) of alleles.</p></a></li>
<li><a href='#Alzheimer'><p>Genotype frequencies for 70 SNPs related to Alzheimer's disease</p></a></li>
<li><a href='#CEUchr22'>
<p>Biallelic polymorphisms sampled from chromosome 22 of the CEU population of the 1000 Genomes project.</p></a></li>
<li><a href='#dgraffelmanweir.bi'>
<p>Calculate Graffelman-Weir exact density for bi-allelic X-chromosomal variant</p></a></li>
<li><a href='#dlevene'>
<p>Calculate Levene's exact density for k alleles</p></a></li>
<li><a href='#dlevene.bi'>
<p>Calculate Levene's density for a bi-allelic variant</p></a></li>
<li><a href='#EAFExact'>
<p>Exact test for equality of allele frequencies in males and females</p></a></li>
<li><a href='#fisherz'><p>Fisher's z transformation</p></a></li>
<li><a href='#fold'>
<p>Fold a square matrix</p></a></li>
<li><a href='#GenerateSamples'><p>Generate genotypic compositions</p></a></li>
<li><a href='#genlabels'>
<p>Label genotype counts of a vector or matrix</p></a></li>
<li><a href='#Glyoxalase'>
<p>Glyoxalase genotype data</p></a></li>
<li><a href='#HapMapCHBChr1'><p>Genotype frequencies for 225 SNPs on chromosome 1 of the CHB population.</p></a></li>
<li><a href='#he'>
<p>Calculate expected heterozygosity (He)</p></a></li>
<li><a href='#HWABO'>
<p>Estimate allele frequencies and test for Hardy-Weinberg equilibrium with</p>
a tri-allelic ABO system.</a></li>
<li><a href='#HWAIC'><p>Compute Akaike's Information Criterion (AIC) for HWP and EAF models</p></a></li>
<li><a href='#HWAlltests'>
<p>Perform all tests for Hardy-Weinberg equilibrium</p></a></li>
<li><a href='#HWAlr'>
<p>Compute additive log-ratio transformation</p></a></li>
<li><a href='#HWAlrPlot'><p>Plot genetic markers in additive log-ratio coordinates</p></a></li>
<li><a href='#HWChisq'><p>Chi square tests for Hardy Weinberg equilibrium</p></a></li>
<li><a href='#HWChisqMat'><p>Matrix version of HWChisq</p></a></li>
<li><a href='#HWChisqStats'>
<p>Fast computation of chi-square statistics for Hardy-Weinberg equilibrium</p></a></li>
<li><a href='#HWClo'>
<p>Convert genotype counts to compositions</p></a></li>
<li><a href='#HWClr'>
<p>Compute the centred log-ratio transformation</p></a></li>
<li><a href='#HWClrPlot'><p>Plot genetic markers in centred log-ratio coordinates</p></a></li>
<li><a href='#HWCondProbAB'><p> Compute probability of a genotypic sample</p></a></li>
<li><a href='#HWD'>
<p>Compute disequilibrium statistic D</p></a></li>
<li><a href='#HWData'><p>Generate genetic marker data in or out of Hardy-Weinberg Equilibrium</p></a></li>
<li><a href='#HWEM'>
<p>Estimation of contributions of two populations to a sample of genotype frequencies with the EM algorithm.</p></a></li>
<li><a href='#HWExact'><p> Exact test for Hardy-Weinberg equilibrium</p></a></li>
<li><a href='#HWExactMat'><p>Matrix version of HWExact</p></a></li>
<li><a href='#HWExactPrevious'><p> Exact test for Hardy-Weinberg equilibrium</p></a></li>
<li><a href='#HWExactStats'>
<p>Computation of Exact p-values for Hardy-Weinberg equilibrium for sets of SNPs</p></a></li>
<li><a href='#HWf'>
<p>Computation of inbreeding coefficient</p></a></li>
<li><a href='#HWGenotypePlot'>
<p>Scatter plot of the genotype frequencies</p></a></li>
<li><a href='#HWIlr'>
<p>Compute isometric log ratio coordinates.</p></a></li>
<li><a href='#HWIlrPlot'>
<p>Plot bi-allelic genetic markers in isometric log ratio coordinates</p></a></li>
<li><a href='#HWLindley'>
<p>Calculate a posteriori density for Lindley's alpha</p></a></li>
<li><a href='#HWLindley.cri'>
<p>Calculate a credible interval for Lindley's alpha for HWE,</p></a></li>
<li><a href='#HWLRAllTests'>
<p>Perform most relevant likelihood ratio test for Hardy-Weinberg</p>
equilibrium and equality of allele frequencies</a></li>
<li><a href='#HWLratio'><p>Likelihood ratio test for Hardy Weinberg equilibrium</p></a></li>
<li><a href='#HWLRtest'><p>Perform likelihood ratio test comparing two nested scenarios for a</p>
bi-allelic genetic variant, distinguishing the two sexes.</a></li>
<li><a href='#HWMissing'>
<p>Test a bi-allelic marker for Hardy-Weinberg equilibrium in the</p>
presence of missing genotype information.</a></li>
<li><a href='#HWNetwork'>
<p>Autosomal and X-chromosomal exact tests for HWE via a Network algorithm</p></a></li>
<li><a href='#HWPerm'>
<p>Permutation test for Hardy-Weinberg equilibrium</p></a></li>
<li><a href='#HWPerm.mult'>
<p>Permutation tests fo variants with multiple alleles</p></a></li>
<li><a href='#HWPosterior'>
<p>Calculation of posterior probabilities and Bayes factors for</p>
Hardy-Weinberg tests at X-chromosomal variants.</a></li>
<li><a href='#HWPower'>
<p>Compute the power of a test for Hardy-Weinberg equilibrium.</p></a></li>
<li><a href='#HWQqplot'>
<p>A Q-Q plot for Hardy-Weinberg equilibrium</p></a></li>
<li><a href='#HWStr'>
<p>Testing a set of microsatellites (STRs) for Hardy-Weinberg equilibrium</p></a></li>
<li><a href='#HWStrata'>
<p>Asymptotic test for HWE across strata for a single biallelic marker</p></a></li>
<li><a href='#HWTernaryPlot'><p> Ternary plot with the Hardy-Weinberg acceptance region</p></a></li>
<li><a href='#HWTriExact'><p>Exact test for Hardy-Weinberg equilibrium and equality of allele</p>
frequencies for tri-allelic variants.</a></li>
<li><a href='#ifisherz'><p>Inverse Fisher z transformation</p></a></li>
<li><a href='#is.mono'><p>Detects autosomal and X-chromosomal monomorphic variants</p></a></li>
<li><a href='#JPTmultiallelicsChr7'>
<p>Multi-allelic autosomal variants of the Japanese population of the</p>
1000 genomes project</a></li>
<li><a href='#JPTmultiallelicsChrX'>
<p>Multi-allelic X-chromosomal variants of the Japanese population of the</p>
1000 genomes project</a></li>
<li><a href='#JPTsnps'>
<p>Bi-allelic SNPs from a Japanese population stratified by gender</p></a></li>
<li><a href='#JPTtriallelicsChr7'>
<p>Tri-allelic variants on chromosome 7 of the Japanese (JPT) sample of the 1000 genomes project</p></a></li>
<li><a href='#JPTtriallelicsChrX'>
<p>Tri-allelic variants on the X-chromosome of the Japanese (JPT) sample of the 1000 genomes project</p></a></li>
<li><a href='#mac'>
<p>Compute the minor allele count.</p></a></li>
<li><a href='#maf'><p>Function to compute minor allele frequencies</p></a></li>
<li><a href='#MakeCounts'><p>Create genotype counts from bi-allelic marker data</p></a></li>
<li><a href='#MakeFactor'><p>Make factors from genotyping data</p></a></li>
<li><a href='#Markers'><p>SNP data and intensities</p></a></li>
<li><a href='#Mourant'><p>Genotype frequencies for blood group locus MN</p></a></li>
<li><a href='#n.alleles'>
<p>Number of alleles</p></a></li>
<li><a href='#NistSTRs'>
<p>NIST autosomal STR data</p></a></li>
<li><a href='#order.auto'><p>Reordering of autosomal genotype counts</p></a></li>
<li><a href='#order.x'><p>Reordering of X-chromosomal genotype counts</p></a></li>
<li><a href='#qqunif'>
<p>Q-Q plot for a uniform distribution.</p></a></li>
<li><a href='#recode'><p>Recode genotype information</p></a></li>
<li><a href='#shannon'>
<p>Shannon index</p></a></li>
<li><a href='#strsort'><p>Sort tokens of a set of strings</p></a></li>
<li><a href='#ThetatoF'>
<p>Convert theta to an inbreeding coefficient</p></a></li>
<li><a href='#toTriangular'><p>Convert a vector of genotype counts to triangular format</p></a></li>
<li><a href='#TSIXTriAllelics'>
<p>Tri-allelic polymorphisms on the X chromosome of the TSI population</p></a></li>
<li><a href='#UniqueGenotypeCounts'><p>Extract unique genotypic compositions from a matrix</p></a></li>
<li><a href='#vaf'>
<p>Computes the sample variance of the allele frequency for a biallelic marker.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Statistical Tests and Graphics for Hardy-Weinberg Equilibrium</td>
</tr>
<tr>
<td>Version:</td>
<td>1.7.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-04-05</td>
</tr>
<tr>
<td>Author:</td>
<td>Jan Graffelman [aut, cre],
  Christopher Chang [ctb],
  Xavi Puig [ctb],
  Jan Wigginton [ctb],
  Leonardo Ortoleva [ctb],
  William R. Engels [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jan Graffelman &lt;jan.graffelman@upc.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 1.8.0), mice, nnet, Rsolnp, shape</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains tools for exploring Hardy-Weinberg equilibrium (Hardy, 1908;  Weinberg, 1908) for bi and multi-allelic genetic marker data. All classical tests (chi-square, exact, likelihood-ratio and permutation tests) with bi-allelic variants are included in the package, as well as functions for power computation and for the simulation of marker data under equilibrium and disequilibrium. Routines for dealing with markers on the X-chromosome are included (Graffelman &amp; Weir, 2016) &lt;<a href="https://doi.org/10.1038%2Fhdy.2016.20">doi:10.1038/hdy.2016.20</a>&gt;, including Bayesian procedures. Some exact and permutation procedures also work with multi-allelic variants. Special test procedures that jointly address Hardy-Weinberg equilibrium and equality of allele frequencies in both sexes are supplied, for the bi and multi-allelic case. Functions for testing equilibrium in the presence of missing data by using multiple imputation are also provided. Implements several graphics for exploring the equilibrium status of a large set of bi-allelic markers: ternary plots with acceptance regions, log-ratio plots and Q-Q plots. The functionality of the package is explained in detail in a related JSS paper &lt;<a href="https://doi.org/10.18637%2Fjss.v064.i03">doi:10.18637/jss.v064.i03</a>&gt;. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.r-project.org">https://www.r-project.org</a>, <a href="http://www-eio.upc.edu/~jan/">http://www-eio.upc.edu/~jan/</a></td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-05 19:21:35 UTC; TanitSunShine</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-06 09:53:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='HardyWeinberg-package'>Statistical Tests and Graphics for Hardy-Weinberg Equilibrium</h2><span id='topic+HardyWeinberg-package'></span><span id='topic+HardyWeinberg'></span>

<h3>Description</h3>

<p>Contains tools for exploring Hardy-Weinberg equilibrium (Hardy, 1908;  Weinberg, 1908) for bi and multi-allelic genetic marker data. All classical tests (chi-square, exact, likelihood-ratio and permutation tests) with bi-allelic variants are included in the package, as well as functions for power computation and for the simulation of marker data under equilibrium and disequilibrium. Routines for dealing with markers on the X-chromosome are included (Graffelman &amp; Weir, 2016) including Bayesian procedures. Some exact and permutation procedures also work with multi-allelic variants. Special test procedures that jointly address Hardy-Weinberg equilibrium and equality of allele frequencies in both sexes are supplied, for the bi and multi-allelic case. Functions for testing equilibrium in the presence of missing data by using multiple imputation are also provided. Implements several graphics for exploring the equilibrium status of a large set of bi-allelic markers: ternary plots with acceptance regions, log-ratio plots and Q-Q plots. The functionality of the package is explained in detail in a related paper (Graffelman, 2015). 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> HardyWeinberg</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.7.7</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2024-03-31</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL Version 2 or later.</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>With function <code>HWTernaryPlot</code> one can create ternary plots with
acceptance regions for HWE. Several routines implement statistical tests
for HWE such as <code>HWChisq</code>, <code>HWExact</code>, <code>HWLratio</code> and
<code>HWPerm</code>. Bayesian procedures are available using
<code>HWPosterior</code>. Aikaike information criterion for various scenarios
can be calculated with <code>HWAIC</code>. Power computations are possible
with <code>HWPower</code>. 
</p>


<h3>Author(s)</h3>

<p>Jan Graffelman
</p>
<p>Maintainer: Jan Graffelman &lt;jan.graffelman@upc.edu&gt;
</p>


<h3>References</h3>

<p>Weir, B.S. (1996) <em>Genetic Data Analysis II</em>. Sinauer Associates, Massachusetts.
</p>
<p>Graffelman, J. and Morales, J. (2008) Graphical Tests for Hardy-Weinberg Equilibrium 
Based on the Ternary Plot. <em>Human Heredity</em> 65(2):77-84. 
</p>
<p>Graffelman, J. (2015) Exploring Diallelic Genetic Markers: The HardyWeinberg Package.
<em>Journal of Statistical Software</em> 64(3):
1-23. <a href="https://doi.org/10.18637/jss.v064.i03">doi:10.18637/jss.v064.i03</a>.
</p>
<p>Graffelman, J. and Weir, B.S. (2016) Testing for Hardy-Weinberg
equilibrium at bi-allelic genetic markers on
the X chromosome. <em>Heredity</em> 116(6):
558-568. <a href="https://doi.org/10.1038/hdy.2016.20">doi:10.1038/hdy.2016.20</a> Nature Publishing Group.
</p>
<p>Hardy, G.H. (1908) Mendelian Proportions in a Mixed Population.
<em>Science</em> 28(706): 49-50. 
</p>
<p>Weinberg, W. (1908) On the Demonstration of Heredity in Man. In: Papers on Human Genetics.
S. H. Boyer (Ed.) Englewood Cliffs, NJ:Prentice Hall, translated, 1963.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(HardyWeinberg)

# draw random SNPs from a population that is in HWE

set.seed(123)

m &lt;- 100 # number of markers
n &lt;- 100 # sample size

X &lt;- HWData(n,m)
out &lt;- HWTernaryPlot(X,100,region=1,vertex.cex=2,signifcolour=TRUE)

</code></pre>

<hr>
<h2 id='af'>Function to compute allele frequencies</h2><span id='topic+af'></span>

<h3>Description</h3>

<p>Function <code>af</code> computes the allele frequency for a vector with autosomal or X-chromosomal genotype counts or compositions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>af(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="af_+3A_x">x</code></td>
<td>
<p>a vector or matrix with counts or compositions. <code>x</code> must have either three elements (autosomal) or five elements (X-chromosomal)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function <code>af</code> calculates the A allele frequency for the given order of autosomal genotype counts (AA,AB,BB) or the given order of
X chromosomal genotype counts (A,B,AA,AB,BB). The genotypes must be supplied this order. Use <code>maf</code> to calculate the minor allele
frequency and more flexible coding.
</p>


<h3>Value</h3>

<p>a vector with allele frequencies
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>See Also</h3>

<p><code><a href="#topic+maf">maf</a></code>, <code><a href="#topic+afx">afx</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# A simulated random marker
#
X &lt;- as.vector(rmultinom(1,100,c(0.5,0.4,0.1)))
X &lt;- X/sum(X)
print(X)
print(af(X))
#
# MN blood group counts
#
x &lt;- c(MM=298,MN=489,NN=213)
#
# Calculate frequency M allele
#
af(x)
#
# Calculate frequency N allele
#
af(rev(x))
1 - af(x)
#
# Calculated allele frequencies (p) for a matrix of genotype counts.
#
X &lt;- HWData(nm=10,n=100)
X
p &lt;- af(X)
p
</code></pre>

<hr>
<h2 id='AFtest'>Exact test of equality of allele frequencies for males and females
</h2><span id='topic+AFtest'></span>

<h3>Description</h3>

<p>Function <code>AFtest</code> tests equality of allele frequencies for males
and females for bi-allelic marker data by means of a Fisher exact test.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AFtest(x, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AFtest_+3A_x">x</code></td>
<td>
<p>a vector containg the genotypic counts c(A,B,AA,AB,BB) for a
bi-allelic X-chromosomal markers.</p>
</td></tr>
<tr><td><code id="AFtest_+3A_verbose">verbose</code></td>
<td>
<p>verbose = TRUE prints results, verbose = FALSE is silent.</p>
</td></tr>
<tr><td><code id="AFtest_+3A_...">...</code></td>
<td>
<p>additional arguments for function <code>fisher.test</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function <code>AFtest</code> constructs the contingency table of sex by
allele, and call <code>fisher.test</code> to test for equality of allele
frequencies. The test assumes Hardy-Weinberg equilibrium.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>AC</code></td>
<td>
<p>Two-way table of sex by allele</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>p-value of the test</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWChisq">HWChisq</a></code>, <code><a href="#topic+HWExact">HWExact</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>  rs5968922 &lt;-  c(A=392, B=212, AA=275, AB=296, BB=80)
  AFtest(rs5968922)
</code></pre>

<hr>
<h2 id='afx'>Function to compute X-chromosomal allele frequencies</h2><span id='topic+afx'></span>

<h3>Description</h3>

<p>Function <code>afx</code> computes the allele frequency for a vector or matrix of X-chromosomal genotype counts or frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>afx(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="afx_+3A_x">x</code></td>
<td>
<p>a vector or matrix with X-chromosomal counts or compositions. <code>x</code> must have five elements or columns.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function <code>afx</code> calculates the A allele frequency for the given order of X-chromosomal genotype counts (A,B,AA,AB,BB). The genotypes must be supplied this order. Use <code>maf</code> to calculate the minor allele frequency and more flexible coding.
</p>


<h3>Value</h3>

<p>a vector with allele frequencies
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>See Also</h3>

<p><code><a href="#topic+maf">maf</a></code>, <code><a href="#topic+af">af</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# A simulated random X-chromosomal marker
#
X &lt;- as.vector(rmultinom(1,100,c(0.3,0.2,0.1,0.15,0.25)))
X &lt;- X/sum(X)
print(X)
print(afx(X))
#
# 
#
rs5968922 &lt;-  c(A=392, B=212, AA=275, AB=296, BB=80)
afx(rs5968922)
#
#
#
y &lt;- c(C=337, G=129, CC=277, CG=209, GG=48) 
afx(y)
#
# Calculated allele frequencies (p) for a matrix of X-chromosomal genotype counts.
#
X &lt;- HWData(nm=10,n=1000,x.linked=TRUE)
X
p &lt;- afx(X)
p
</code></pre>

<hr>
<h2 id='agcounts'>
Calculate allele and genotype counts for X-chromosomal markers
</h2><span id='topic+agcounts'></span>

<h3>Description</h3>

<p>Function <code>agcounts</code> determines sample size, minor are major allele counts,
allele counts in females, numbers of males and females and allele
frequencies for a vector of genotypes counts of an X-chromosomal markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agcounts(x, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="agcounts_+3A_x">x</code></td>
<td>
<p>a vector of X-chromosomal genotype counts (A,B,AA,AB,BB)</p>
</td></tr>
<tr><td><code id="agcounts_+3A_verbose">verbose</code></td>
<td>
<p>print the counts if (<code>verbose = TRUE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code>nA</code></td>
<td>
<p>number of A alleles</p>
</td></tr>
<tr><td><code>nB</code></td>
<td>
<p>number of B alleles</p>
</td></tr>
<tr><td><code>nf</code></td>
<td>
<p>number of females</p>
</td></tr>
<tr><td><code>nm</code></td>
<td>
<p>number of males</p>
</td></tr>
<tr><td><code>nAf</code></td>
<td>
<p>number of A alleles in females</p>
</td></tr>
<tr><td><code>nBf</code></td>
<td>
<p>number of B alleles in females</p>
</td></tr>
<tr><td><code>nt</code></td>
<td>
<p>total number of alleles</p>
</td></tr>
<tr><td><code>fAA</code></td>
<td>
<p>number of AA females</p>
</td></tr>
<tr><td><code>fAB</code></td>
<td>
<p>number of AB females</p>
</td></tr>
<tr><td><code>fBB</code></td>
<td>
<p>number of BB females</p>
</td></tr>
<tr><td><code>pA</code></td>
<td>
<p>overall A allele frequency</p>
</td></tr>
<tr><td><code>pB</code></td>
<td>
<p>overall B allele frequency</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

<p><code><a href="#topic+mac">mac</a></code>, <code>link{maf}</code></p>


<h3>Examples</h3>

<pre><code class='language-R'>rs5968922 &lt;-  c(A=392, B=212, AA=275, AB=296, BB=80)
counts &lt;- agcounts(rs5968922)
</code></pre>

<hr>
<h2 id='alleles'>
Extract alleles 
</h2><span id='topic+alleles'></span>

<h3>Description</h3>

<p>Function <code>alleles</code> extracts the names of the alleles from a named
genotype vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alleles(x, fromlabels = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="alleles_+3A_x">x</code></td>
<td>
<p>A named or unnamed genotype vector (e.g. c(AA=10,AB=20,BB=5))</p>
</td></tr>
<tr><td><code id="alleles_+3A_fromlabels">fromlabels</code></td>
<td>
<p>extract genotypes from the labels of the vector
elements, or from the vector elements themselves.</p>
</td></tr> 
</table>


<h3>Value</h3>

<p>A character vector with the alleles</p>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>See Also</h3>

<p><code><a href="#topic+n.alleles">n.alleles</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>   x &lt;- c(AA=10,AG=10,GG=10,AT=5)
   als.x &lt;- alleles(x)
   print(als.x)

   y &lt;- rep(names(x),x)
   als.y &lt;- alleles(y,fromlabels = FALSE)
   print(als.y)
</code></pre>

<hr>
<h2 id='AllelesToTriangular'>
Calculate triangular genotype matrix for vector(s) of alleles.
</h2><span id='topic+AllelesToTriangular'></span>

<h3>Description</h3>

<p><code>AllelesToTriangular</code> constructs a lower triangular matrix of genotype counts from one or two vectors of alleles. It is particularly useful to create genotype counts for microsatellite data (STRs). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AllelesToTriangular(A1, A2 = NULL, given=NULL) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AllelesToTriangular_+3A_a1">A1</code></td>
<td>
<p>The first allele of each individual, or a vector with all alleles, two consecutive ones for each individual.</p>
</td></tr>
<tr><td><code id="AllelesToTriangular_+3A_a2">A2</code></td>
<td>
<p>The second allele of each individual (optional).</p>
</td></tr>
<tr><td><code id="AllelesToTriangular_+3A_given">given</code></td>
<td>
<p>A vector of known alleles (optional). This argument can be used to specify alleles that may not exist in the data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the data is a single column vector with two succesive alleles for each individual, then specify <code>A1</code> only. If data consists of two columns, each holding one allele of each individual, then specify <code>A1</code> and <code>A2</code>. Typical STR data that comes in the format of two repeat lengths for a set of individuals can be transformed into a lower triangular matrix with genotype counts. See the examples below.
</p>


<h3>Value</h3>

<p>A lower triangular matrix with genotype counts.
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Graffelman, J. (2015) Exploring Diallelic Genetic Markers: The HardyWeinberg Package.
<em>Journal of Statistical Software</em> 64(3): 1-23. <a href="https://doi.org/10.18637/jss.v064.i03">doi:10.18637/jss.v064.i03</a>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+toTriangular">toTriangular</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(NistSTRs)
A1 &lt;- NistSTRs[,1]
A2 &lt;- NistSTRs[,2]
GM &lt;- AllelesToTriangular(A1,A2)
print(GM)
</code></pre>

<hr>
<h2 id='Alzheimer'>Genotype frequencies for 70 SNPs related to Alzheimer's disease</h2><span id='topic+Alzheimer'></span>

<h3>Description</h3>

<p>The dataframe contains the genotype frequencies MM, Mm and mm for the
70 SNPs for both cases and controls. The data are taken from table
7.11 in Laird &amp; Lange.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Alzheimer)</code></pre>


<h3>Format</h3>

<p>A data frame containing 70 observations.</p>


<h3>Source</h3>

<p>Laird, N. M. and Lange, C. Table 7.11, p. 124</p>


<h3>References</h3>

<p>Laird, N. M. and Lange, C. (2011) The fundamentals of modern statistical genetics.
Springer.
</p>

<hr>
<h2 id='CEUchr22'>
Biallelic polymorphisms sampled from chromosome 22 of the CEU population of the 1000 Genomes project.
</h2><span id='topic+CEUchr22'></span>

<h3>Description</h3>

<p>Matrix <code>CEUchr22</code> contains 10000 single nucleotide polymorphisms sampled from chromosome 22 of sample of
99 Utah residents with Northern and Western European ancestry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("CEUchr22")</code></pre>


<h3>Format</h3>

<p>Matrix</p>


<h3>Details</h3>

<p>The polymorphisms are coded in (0,1,2) format representing the count of the reference allele.
</p>


<h3>Source</h3>

<p>https://www.internationalgenome.org/
</p>


<h3>References</h3>

<p>The 1000 Genomes Project Consortium (2015) A global reference for human genetic variation. Nature 526(7571), pp. 68&ndash;74.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(CEUchr22)
str(CEUchr22)
</code></pre>

<hr>
<h2 id='dgraffelmanweir.bi'>
Calculate Graffelman-Weir exact density for bi-allelic X-chromosomal variant
</h2><span id='topic+dgraffelmanweir.bi'></span>

<h3>Description</h3>

<p>Function <code>dgraffelmanweir</code> calculate the probability P(NAB=nab and
MA=ma|NA=na) for a bi-allelic X-chromosomal variant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dgraffelmanweir.bi(x, y)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dgraffelmanweir.bi_+3A_x">x</code></td>
<td>
<p>vector with male genotype counts (A,B)</p>
</td></tr>
<tr><td><code id="dgraffelmanweir.bi_+3A_y">y</code></td>
<td>
<p>vector with female genotype counts (AA,AB,BB)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a single real number</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a></p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2016) Testing for Hardy-Weinberg
equilibrium at bi-allelic genetic markers on the X chromosome.
Heredity 116(6) pp. 558&ndash;568. <a href="https://doi.org/10.1038/hdy.2016.20">doi:10.1038/hdy.2016.20</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWExact">HWExact</a></code>, <code><a href="#topic+HWExactStats">HWExactStats</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>males &lt;- c(A=392, B=212)
females &lt;- c(AA=275, AB=296, BB=80)
prob &lt;- dgraffelmanweir.bi(males,females)
print(prob)
</code></pre>

<hr>
<h2 id='dlevene'>
Calculate Levene's exact density for k alleles
</h2><span id='topic+dlevene'></span>

<h3>Description</h3>

<p>Function <code>dlevene</code> calculates Levene's exact density for a diploid
system with k alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dlevene(N)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dlevene_+3A_n">N</code></td>
<td>
<p>A lower triangular matrix with genotype counts</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The supplied matrix of genotype counts should be triangular, with the
homozygote counts on the diagonal, and all heterozygote counts below
the diagonal.
</p>


<h3>Value</h3>

<p>a single real number</p>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>References</h3>

<p>Levene, H. (1949) On a matching problem arising in genetics. Annals of
Mathematical Statistics, 20,  pp. 91-94.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWExact">HWExact</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(AA=12,AB=19,AC=13,BB=7,BC=5,CC=0)
x &lt;- toTriangular(x)
prob &lt;- dlevene(x)
print(prob)
</code></pre>

<hr>
<h2 id='dlevene.bi'>
Calculate Levene's density for a bi-allelic variant
</h2><span id='topic+dlevene.bi'></span>

<h3>Description</h3>

<p>Program <code>dlevene.bi</code> calculates Levene's density (P(AB|A)) for a
bi-allelic variant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dlevene.bi(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dlevene.bi_+3A_x">x</code></td>
<td>
<p>a vector of genotype counts (AA,AB,BB)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a single real number</p>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>References</h3>

<p>Levene, H. (1949) On a matching problem arising in genetics. Annals of
Mathematical Statistics, 20,  pp. 91-94.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dlevene">dlevene</a></code>, <code><a href="#topic+HWExact">HWExact</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(AA=298,AB=489,BB=213)
prob &lt;- dlevene.bi(x)
print(prob)
</code></pre>

<hr>
<h2 id='EAFExact'>
Exact test for equality of allele frequencies in males and females
</h2><span id='topic+EAFExact'></span>

<h3>Description</h3>

<p><code>EAFExact</code> uses a Fisher Exact test to compare allele frequencies
in males and females for variants with k alleles (k &gt;= 2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EAFExact(m, f, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="EAFExact_+3A_m">m</code></td>
<td>
<p>vector or triangular matrix with male genotype counts</p>
</td></tr>
<tr><td><code id="EAFExact_+3A_f">f</code></td>
<td>
<p>vector or triangular matrix with female genotype counts</p>
</td></tr>
<tr><td><code id="EAFExact_+3A_verbose">verbose</code></td>
<td>
<p>print output (TRUE) or not (FALSE)</p>
</td></tr>
<tr><td><code id="EAFExact_+3A_...">...</code></td>
<td>
<p>additional arguments for <code>fisher.test</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For bi-allelic autosomal variants the genotype counts can be supplied as vectors
((AA,AB,BB) for males, and (AA,AB,BB) for females). For X-chromosomal
bi-allelic variants the genotype counts can also supplied as vectors
((A,B) for males, and (AA,AB,BB) for females). For multi-allelic autosomal
variants male and genotype counts can be supplied as vectors
(AA,AB,AC,BB,BC,CC,...) or as a triangular
matrix, where matrix rows and colums are labelled with the allele name
(A,B,C,...). For multi-allelic X-chromosomal variants, male genotype
counts must be supplied as a vector (A,B,C,...) and female genotype
counts must be supplied as a triangular matrix. See the examples below.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pval</code></td>
<td>
<p>p-value</p>
</td></tr>
<tr><td><code>tab</code></td>
<td>
<p>table with allele counts</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+fisher.test">fisher.test</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# bi-allelic autosomal
#

m &lt;- c(AA=60,AB=96,BB=44) 
f &lt;- c(AA=44,AB=97,BB=59) 
EAFtest &lt;- EAFExact(m,f)

#
# bi-allelic X-chromosomal
#

males   &lt;- c(A=392, B=212)
females &lt;- c(AA=275, AB=296, BB=80)
EAFtest &lt;- EAFExact(males,females,verbose=TRUE)

#
# tri-allelic autosomal
#

males   &lt;- c(AA=20,AB=52,AC=34,BB=17,BC=51,CC=26)
females &lt;- c(AA=28,AB=55,AC=33,BB=18,BC=50,CC=16)
EAFtest &lt;- EAFExact(males,females,verbose=TRUE)

#
# tri-allelic X-chromosomal
#

males   &lt;- c(A=15,B=17,C=24) 
females &lt;- toTriangular(c(AA=4,AB=2,AC=13,BB=6,BC=19,CC=4)) 
EAFtest &lt;- EAFExact(males,females,verbose=TRUE)

</code></pre>

<hr>
<h2 id='fisherz'>Fisher's z transformation
</h2><span id='topic+fisherz'></span>

<h3>Description</h3>

<p>Calculates Fisher's z transformation for a correlation coefficient
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fisherz(r)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fisherz_+3A_r">r</code></td>
<td>
<p>a correlation coefficient
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a real number
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+cor">cor</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>r &lt;- 0.5
print(fisherz(r))
</code></pre>

<hr>
<h2 id='fold'>
Fold a square matrix
</h2><span id='topic+fold'></span>

<h3>Description</h3>

<p>The function <code>fold</code> sums corresponding below and above diagonal elements of a square matrix to form a triangular matrix. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fold(X, lower = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fold_+3A_x">X</code></td>
<td>
<p>a square matrix</p>
</td></tr>
<tr><td><code id="fold_+3A_lower">lower</code></td>
<td>
<p>logical. If <code>lower=TRUE</code> a lower triangular matrix is formed, if not an upper triangular matrix.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Useful for constructing triangular matrices of genotype counts
</p>


<h3>Value</h3>

<p>A matrix
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+lower.tri">lower.tri</a>, <a href="base.html#topic+upper.tri">upper.tri</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>allelenames &lt;- paste("A",11:13,sep="")
allele1 &lt;- factor(c("A11","A11","A12","A12","A13","A12"),levels=allelenames)
allele2 &lt;- factor(c("A11","A12","A12","A13","A13","A11"),levels=allelenames)
GC  &lt;- table(allele1,allele2)
GC  &lt;- as.matrix(unclass(GC))
GCf &lt;- fold(GC)
</code></pre>

<hr>
<h2 id='GenerateSamples'>Generate genotypic compositions</h2><span id='topic+GenerateSamples'></span>

<h3>Description</h3>

<p><code>GenerateSamples</code> generates all possible genotypic compositions
(AA,AB,BB) for a given sample size <code>n</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenerateSamples(n = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GenerateSamples_+3A_n">n</code></td>
<td>
<p>the desired sample size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a matrix with in each row a possible genotypic compostion
for the given sample size.
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>Examples</h3>

<pre><code class='language-R'>GenerateSamples(5)
</code></pre>

<hr>
<h2 id='genlabels'>
Label genotype counts of a vector or matrix
</h2><span id='topic+genlabels'></span>

<h3>Description</h3>

<p>Function <code>genlabels</code> sets the names of a vector or matrix of
genotype counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genlabels(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genlabels_+3A_x">X</code></td>
<td>
<p>a 3 (or 5) element vector with genotype counts, a matrix of
genotype counts (3 or 5 columns)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector or a matrix</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>See Also</h3>

<p><code><a href="#topic+HWChisq">HWChisq</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(25,50,25)
x &lt;- genlabels(x)
</code></pre>

<hr>
<h2 id='Glyoxalase'>
Glyoxalase genotype data
</h2><span id='topic+Glyoxalase'></span>

<h3>Description</h3>

<p>Biallelic glyoxalase genotype data for 17 populations from India
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Glyoxalase")</code></pre>


<h3>Format</h3>

<p>A data frame with 17 observations on the following 3 variables.
</p>

<dl>
<dt><code>AA</code></dt><dd><p>number of homozygote AA individuals</p>
</dd>
<dt><code>AB</code></dt><dd><p>number of heterozygote AB individuals</p>
</dd>
<dt><code>BB</code></dt><dd><p>number of homozygote BB individuals</p>
</dd>
</dl>



<h3>Source</h3>

<p>Olson, J.M. (1993) Table 3.
</p>


<h3>References</h3>

<p>Olson, J.M. (1993) Testing the Hardy-Weinberg law across strata. Annals of Human Genetics, 57,  pp. 291-295.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Glyoxalase)
</code></pre>

<hr>
<h2 id='HapMapCHBChr1'>Genotype frequencies for 225 SNPs on chromosome 1 of the CHB population.</h2><span id='topic+HapMapCHBChr1'></span>

<h3>Description</h3>

<p>The dataframe contains the genotype frequencies in generic notation,
AA, AB and BB the first 225 polymorphic SNPs without missing data on
chromosome 1 of the Han Chinese in Beijing. The data are compiled
from the HapMap project, phase 3.2, containing genotype information
of 84 individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HapMapCHBChr1)</code></pre>


<h3>Format</h3>

<p>A matrix containing 225 rows and 3 columns (AA, AB, BB).</p>


<h3>Source</h3>

<p>http://hapmap.ncbi.nlm.nih.gov/</p>


<h3>References</h3>

<p>The International HapMap Consortium (2007). A second generation human haplotype map of over 3.1 million SNPs
Nature 449, pp. 851&ndash;861.
</p>

<hr>
<h2 id='he'>
Calculate expected heterozygosity (He)
</h2><span id='topic+he'></span>

<h3>Description</h3>

<p>Function <code>he</code> calculates the expected heterozygosity for a biallelic genetic variant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>he(x, bias.correct = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="he_+3A_x">x</code></td>
<td>
<p>a vector or matrix with genotype counts.</p>
</td></tr>
<tr><td><code id="he_+3A_bias.correct">bias.correct</code></td>
<td>
<p>if <code>bias.correct = TRUE</code> a correction for bias will be applied.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>x</code> can be a vector of genotype counts for a single marker (AA,AB,BB) or a three-column matrix of
genotype counts for multiple markers.
</p>


<h3>Value</h3>

<p>a vector of expected heterozygosities.
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>See Also</h3>

<p><code><a href="#topic+af">af</a></code>, <code><a href="#topic+maf">maf</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# He for a single marker
#
x &lt;- c(MM=298,MN=489,NN=213)
he(x)
#
# He for a matrix of rmarkers
#
set.seed(123)
X &lt;- HWData(10,100)
he(X)
he(X,bias.correct = FALSE)
</code></pre>

<hr>
<h2 id='HWABO'>
Estimate allele frequencies and test for Hardy-Weinberg equilibrium with
a tri-allelic ABO system.
</h2><span id='topic+HWABO'></span>

<h3>Description</h3>

<p>Function <code>HWABO</code> takes four genotype counts (&quot;A&quot;,&quot;B&quot;,&quot;AB&quot;,&quot;OO&quot;) and
estimates the three allele frequencies using the EM algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWABO(x, p = c(1/3, 1/3, 1/3), maxiter = 50, tol = 1e-10, verbose = TRUE) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWABO_+3A_x">x</code></td>
<td>
<p> a vector with genotype counts (&quot;A&quot;,&quot;B&quot;,&quot;AB&quot;,&quot;OO&quot;).</p>
</td></tr>
<tr><td><code id="HWABO_+3A_p">p</code></td>
<td>
<p> a vector with initial allele frequencies (by default (1/3,1/3,1/3)).</p>
</td></tr>
<tr><td><code id="HWABO_+3A_maxiter">maxiter</code></td>
<td>
<p> maximum number of iterations.</p>
</td></tr>
<tr><td><code id="HWABO_+3A_tol">tol</code></td>
<td>
<p> tolerance for convergence, 1e-10 by default</p>
</td></tr>
<tr><td><code id="HWABO_+3A_verbose">verbose</code></td>
<td>
<p> print iteration history or not.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pn</code></td>
<td>
<p> vector with estimated allele frequencies.</p>
</td></tr>
<tr><td><code>It.hist</code></td>
<td>
<p> iteration history with log-likelihood.</p>
</td></tr>
<tr><td><code>expected</code></td>
<td>
<p> expected genotype counts under HWE.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

<p><code><a href="#topic+af">af</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   x &lt;- c(fA=182,fB=60,nAB=17,nOO=176)
   al.fre &lt;- HWABO(x)
#al2 &lt;- HWABO(x,p=c(0.99,0.01,0.01),maxiter=25)
#al3 &lt;- HWABO(x,p=c(0.01,0.99,0.01),maxiter=25)
#al4 &lt;- HWABO(x,p=c(0.01,0.01,0.99),maxiter=25)
</code></pre>

<hr>
<h2 id='HWAIC'>Compute Akaike's Information Criterion (AIC) for HWP and EAF models</h2><span id='topic+HWAIC'></span>

<h3>Description</h3>

<p>Function <code>HWAIC</code> calculates Akaike's Information Criterion for
ten different models that describe a bi-allelic genetic variant: M11:
Hardy-Weinberg proportions and equality of allele frequencies in the
sexes (HWP &amp; EAF); M12: EAF and HWP in males only; M13: EAF and HWP in
females only; M14: EAF and equality of inbreeding coefficients in
the sexes (EIC); M15: EAF only; M21: HWP in both sexes; M22: HWP for
males only; M23: HWP for females only; M24: EIC only; M25: None of the previous.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWAIC(x, y, tracing = 0, tol = 0.000001)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWAIC_+3A_x">x</code></td>
<td>
<p>Male genotype counts (AA,AB,BB)</p>
</td></tr>
<tr><td><code id="HWAIC_+3A_y">y</code></td>
<td>
<p>Female genotype counts (AA,AB,BB)</p>
</td></tr>
<tr><td><code id="HWAIC_+3A_tracing">tracing</code></td>
<td>
<p>Activate tracing in the maximization of some
likelihoods (0=no tracing; 1:tracing)</p>
</td></tr>
<tr><td><code id="HWAIC_+3A_tol">tol</code></td>
<td>
<p>tolerance for iterative maximization of some likelihoods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The log-likelihood for the six models is calculated. For two models (C
and E) this is done numerically using package <code>RSolnp</code>.
</p>


<h3>Value</h3>

<p>A named vector containing 6 values for AIC</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a></p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2018) On the testing of Hardy-Weinberg
proportions and equality of allele frequencies in males and females at
bi-allelic genetic markers. Genetic Epidemiology 42(1) pp. 34-48. <a href="https://doi.org/10.1002/gepi.22079">doi:10.1002/gepi.22079</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWLRtest">HWLRtest</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>males &lt;- c(AA=11,AB=32,BB=13) 
females &lt;- c(AA=14,AB=23,BB=11) 
stats &lt;- HWAIC(males,females)
print(stats)
</code></pre>

<hr>
<h2 id='HWAlltests'>
Perform all tests for Hardy-Weinberg equilibrium
</h2><span id='topic+HWAlltests'></span>

<h3>Description</h3>

<p><code>HWAlltests</code> performs all classical frequentists tests for
Hardy-Weinberg equilibrium and lists their p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWAlltests(x, verbose = TRUE, include.permutation.test = FALSE, x.linked = FALSE) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWAlltests_+3A_x">x</code></td>
<td>
<p>a vector with a set of genotype counts (AA, AB, BB)</p>
</td></tr>
<tr><td><code id="HWAlltests_+3A_verbose">verbose</code></td>
<td>
<p>print output if set to TRUE</p>
</td></tr>
<tr><td><code id="HWAlltests_+3A_include.permutation.test">include.permutation.test</code></td>
<td>
<p>turns on the permutation test if set
to TRUE</p>
</td></tr>
<tr><td><code id="HWAlltests_+3A_x.linked">x.linked</code></td>
<td>
 <p><code>x.linked=FALSE</code> indicates the marker is
autosomal (default), and <code>x.linked=TRUE</code> indicates it resides
on the X-chromosome.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default the permutation test is not performed in order to reduce
computing time. 
</p>


<h3>Value</h3>

<p>A dataframe with test statistics and p-values. 
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWLratio">HWLratio</a></code>, <code><a href="#topic+HWChisq">HWChisq</a></code>, <code><a href="#topic+HWExact">HWExact</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(298,489,213)
names(x) &lt;- c("MM","MN","NN")
HWAlltests(x,verbose=TRUE)
</code></pre>

<hr>
<h2 id='HWAlr'>
Compute additive log-ratio transformation
</h2><span id='topic+HWAlr'></span>

<h3>Description</h3>

<p><code>HWAlr</code> computes the additive log-ratio transformation for
genotype counts of bi-allelic genetic markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWAlr(X, zeroadj = 0.5, denominator = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWAlr_+3A_x">X</code></td>
<td>
<p>A matrix of genotype counts (columns AA, AB and BB)</p>
</td></tr>
<tr><td><code id="HWAlr_+3A_zeroadj">zeroadj</code></td>
<td>
<p>A zero adjustment parameter (0.5 by default)</p>
</td></tr>
<tr><td><code id="HWAlr_+3A_denominator">denominator</code></td>
<td>
<p>The genotype count put in the denominator of the
log-ratio (1=AA, 2=AB, 3=BB)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix or vector of log-ratio coordinates
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>References</h3>

<p>Graffelman, J. and Egozcue, J. J. (2011) Hardy-Weinberg
equilibrium: a non-parametric compositional approach. In: Vera
Pawlowsky-Glahn and Antonella Buccianti (eds.) Compositional Data
Analysis: Theory and Applications, John Wiley &amp; Sons, Ltd, pp. 207-215</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWClr">HWClr</a>,<a href="#topic+HWIlr">HWIlr</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   X &lt;- HWData(100,100)
   Y &lt;- HWAlr(X)
</code></pre>

<hr>
<h2 id='HWAlrPlot'>Plot genetic markers in additive log-ratio coordinates</h2><span id='topic+HWAlrPlot'></span>

<h3>Description</h3>

<p><code>HWAlrPlot</code> creates a scatter plot of the log-ratio coordinates of
bi-allelic genetic markers. Hardy-Weinberg equilibrium is indicated by
a straigh line in the plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWAlrPlot(X, zeroadj = 0.5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWAlrPlot_+3A_x">X</code></td>
<td>
<p>A matrix of genotype counts (columns AA, AB, BB)
</p>
</td></tr>
<tr><td><code id="HWAlrPlot_+3A_zeroadj">zeroadj</code></td>
<td>
<p>Zero-adjustment parameter. Zero counts in the count
matrix are substituted by <code>zeroadj</code> which is 0.5 by default.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>References</h3>

<p>Graffelman, J. and Egozcue, J. J. (2011) Hardy-Weinberg
equilibrium: a non-parametric compositional approach. In: Vera
Pawlowsky-Glahn and Antonella Buccianti (eds.) Compositional Data
Analysis: Theory and Applications, John Wiley &amp; Sons, Ltd, pp. 207-215</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWClrPlot">HWClrPlot</a>,<a href="#topic+HWIlrPlot">HWIlrPlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   X &lt;- HWClo(HWData(100,100))
   HWAlrPlot(X)
</code></pre>

<hr>
<h2 id='HWChisq'>Chi square tests for Hardy Weinberg equilibrium</h2><span id='topic+HWChisq'></span>

<h3>Description</h3>

<p><code>HWChisq</code> performs the chi-square test for Hardy-Weinberg
equilibrium both for autosomal and X-chromosomal markers; it can dealt both with bi-allelic and multi-allelic variants.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWChisq(X, cc = 0.5, verbose = TRUE, x.linked = FALSE, phifixed = NULL) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWChisq_+3A_x">X</code></td>
<td>
<p>For bi-allelic variants, <code>X</code> is a vector containg the genotypic counts (AA,AB,BB for
autosomal markers c(A,B,AA,AB,BB) for X-chromosomal markers). For multi-allelic variants, <code>X</code> is a
lower triangular matrix with genotype counts, homozygotes on the diagonal and heterozygotes below the diagonal.</p>
</td></tr>
<tr><td><code id="HWChisq_+3A_cc">cc</code></td>
<td>
<p><code>cc</code> the continuity correction parameter, the correction is only applied to bi-allelic markers (default
<code>cc = 0.5</code>).</p>
</td></tr>
<tr><td><code id="HWChisq_+3A_verbose">verbose</code></td>
<td>
 <p><code>verbose = TRUE</code> prints results, <code>verbose = FALSE</code>
is silent.</p>
</td></tr>
<tr><td><code id="HWChisq_+3A_x.linked">x.linked</code></td>
<td>
 <p><code>x.linked=FALSE</code> indicates the marker is
autosomal (default), and <code>x.linked=TRUE</code> indicates it resides
on the X-chromosome.</p>
</td></tr>
<tr><td><code id="HWChisq_+3A_phifixed">phifixed</code></td>
<td>
<p>(For X-chromosomal markers only)
<code>phifixed=NULL</code> indicates that the fraction of males
(females) should be estimated from the data (default). If set to any
other value (e.g. <code>phifixed=0.5</code>) then the sample is assumed to
come from a population with the specified fraction of males.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>HWChisq</code> does a chi-square test for Hardy-Weinberg equilibrium,
and by default applies a continuity correction. For extreme allele
frequencies, the continuity correction can lead to excessive type 1
error rates, and is better turned off in that case. The continuity
correction can be turned off by specifying <code>cc=0</code>.
</p>
<p><code>HWChisq</code> can do the chi-square test for both autosomal and
X-chrosomal markers. By setting <code>x.linked = TRUE</code> the marker
will be assumed to be on the X-chromosome, and the count vector
supplied should have 5 elements instead of 3 elements for an
autosomal marker. For X-chromsomal markers argument <code>phifixed</code>
is in general best left to its default value (<code>NULL</code>). Only in
specific situations where the theoretical population sex ratio is known (e.g. in
simulation studies where a universe with known gender ratio is
sampled) <code>phifixed</code> could be set to the theoretical ratio of interest.   
</p>
<p>With bi-allelic variants, when <code>alternative</code> is set to <code>less</code>, a one-sided test for
against a negative inbreeding coefficient (heterozygote excess) is
performed. When <code>alternative</code> is set to <code>greater</code> a one-sided test for
against a positive inbreeding coefficient (lack of heterozygotes) is
performed.
</p>
<p>For multi-allelic variants, which typically do have some rare alleles and rare genotypes, the asymptotic chi-square test is in
general not recommended, and exact test procedures or permutation tests are recommended (see <code>HWExact</code> or <code>HWPerm.mult</code>).
</p>


<h3>Value</h3>

<p><code>HWChisq</code> returns a list with the components:
</p>
<table role = "presentation">
<tr><td><code>chisq</code></td>
<td>
<p>value of the chi-square statistic. NA is returned if the marker is monomorphic.</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>p-value of the chi-square test for Hardy-Weinberg equilibrium.</p>
</td></tr>
<tr><td><code>D</code></td>
<td>
<p>Half the deviation from Hardy-Weinberg equilibrium for the AB genotype.</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>the allele frequency of A.</p>
</td></tr>
<tr><td><code>f</code></td>
<td>
<p>the inbreeding coefficient.</p>
</td></tr>
<tr><td><code>expected</code></td>
<td>
<p>the expected counts under Hardy-Weinberg equilibrium.</p>
</td></tr>
<tr><td><code>chi.contrib</code></td>
<td>
<p>the contributions of the different genotypes to the chi-square statistic.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Weir, B.S. (1996) Genetic data analysis II. Sinauer
Associates, Massachusetts. See Chapter3.
</p>
<p>For the chi-square test for X-linked markers:
</p>
<p>Graffelman, J. &amp; Weir, B.S. (2016) Testing for Hardy-Weinberg
equilibrium at bi-allelic genetic markers on the X chromosome. 
<em>Heredity</em> 116(6) pp. 558&ndash;568. <a href="https://doi.org/10.1038/hdy.2016.20">doi:10.1038/hdy.2016.20</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWLratio">HWLratio</a></code>, <code><a href="#topic+HWChisqStats">HWChisqStats</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Test for an autosomal blood group marker
#
x &lt;- c(MM=298,MN=489,NN=213)
HW.test &lt;- HWChisq(x,verbose=TRUE)
#
# Same test without continuity correction
#
HW.test &lt;- HWChisq(x,cc=0,verbose=TRUE)
#
# Test for X-chromsomal SNPs.
#
rs5968922 &lt;-  c(A=392, B=212, AA=275, AB=296, BB=80)
HW.test &lt;- HWChisq(rs5968922,cc=0,x.linked=TRUE,verbose=TRUE)
#
#
#
y &lt;- c(GG=48, CG=209, CC=277, G=129, C=337) 
HWChisq(y,x.linked=TRUE,cc=0)
#
# Test a multi-allelic microsatellite
#
data(NistSTRs)
A1 &lt;- NistSTRs[,1]
A2 &lt;- NistSTRs[,2]
GC &lt;- AllelesToTriangular(A1,A2)
HW.test &lt;- HWChisq(GC)
#
# retaining only the three  most common alleles
#
ii &lt;- (A1 == 10 | A1 == 11 | A1 == 12) &amp; (A2 == 10 | A2 == 11 | A2 == 12)
A1s &lt;- A1[ii]
A2s &lt;- A2[ii]
GC &lt;- AllelesToTriangular(A1s,A2s)
HW.test &lt;- HWChisq(GC)
</code></pre>

<hr>
<h2 id='HWChisqMat'>Matrix version of HWChisq</h2><span id='topic+HWChisqMat'></span>

<h3>Description</h3>

<p>Function <code>HWChisqMat</code> executes the Chisquare test for HWE for each row in a matrix. This function is
deprecated and it is better to use the faster <code>HWChisqStats</code> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWChisqMat(X, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWChisqMat_+3A_x">X</code></td>
<td>
<p>A n times 3 matrix of genotypic counts (AA,AB,BB)</p>
</td></tr>
<tr><td><code id="HWChisqMat_+3A_...">...</code></td>
<td>
<p>extra arguments that are passed on to <code>HWChisqStats</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with chi-square statistics or p-values</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWChisq">HWChisq</a></code>, <code><a href="#topic+HWChisqStats">HWChisqStats</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- HWData(100,100)
colnames(X) &lt;- c("MM","MN","NN")
Results &lt;- HWChisqMat(X)
</code></pre>

<hr>
<h2 id='HWChisqStats'>
Fast computation of chi-square statistics for Hardy-Weinberg equilibrium
</h2><span id='topic+HWChisqStats'></span>

<h3>Description</h3>

<p><code>HWChisqStats</code> is a function for the fast computation of chi-square
statistics (or the corresponding p-values) for a large set of bi-allelic markers (typically SNPs). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWChisqStats(X, x.linked = FALSE, pvalues = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWChisqStats_+3A_x">X</code></td>
<td>
<p>A matrix with genotype counts, one row per marker. <code>X</code>
should have 5 columns for an X-chromosomal data set and 3 columns
for an autosomal data set.</p>
</td></tr>
<tr><td><code id="HWChisqStats_+3A_x.linked">x.linked</code></td>
<td>
<p>Logical indicating whether the markers are autosomal (<code>x.linked=FALSE</code>)
or X-chromosomal (<code>x.linked=TRUE</code>).</p>
</td></tr>
<tr><td><code id="HWChisqStats_+3A_pvalues">pvalues</code></td>
<td>
<p>Logical indicated whether chi-square statistics should be returned (<code>pvalues=FALSE</code>) or whether p-values should be returned (<code>pvalues=TRUE</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Matrix <code>X</code> should strictly comply with the following format. For
an autosomal dataset is should contain the 3 genotype counts in order
(AA,AB,BB). For an X-chromosomal dataset it should contain the 5
genotype counts in order (A,B,AA,AB,BB) where A and B are the male
counts and AA, AB and BB the female counts.
</p>
<p>This function was written for speed improvement, and should be much
faster than looping over the rows of <code>X</code> with <code>HWChisq</code>. There
is no error checking on the supplied data matrix.
</p>


<h3>Value</h3>

<p>A vector of chi-square statistics</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2016) Testing for Hardy-Weinberg
equilibrium at 2 bi-allelic genetic markers on the X chromosome.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWChisq">HWChisq</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Autosomal example
#
set.seed(123)
X &lt;- HWData(1000,100)
monom &lt;- (X[,2]==0 &amp; X[,1]==0) | (X[,2]==0 &amp; X[,3]==0)
X &lt;- X[!monom,] # exclude monomorphics
Chisq.stats &lt;- HWChisqStats(X,x.linked=FALSE,pvalues=FALSE)
Chisq.pvals &lt;- HWChisqStats(X,x.linked=FALSE,pvalues=TRUE)
#
# X-chromosomal example
#
X &lt;- HWData(1000,100,n.males=50,nA=75,x.linked=TRUE)
Chisq.stats &lt;- HWChisqStats(X,x.linked=TRUE,pvalues=FALSE)
Chisq.pvals &lt;- HWChisqStats(X,x.linked=TRUE,pvalues=TRUE)
</code></pre>

<hr>
<h2 id='HWClo'>
Convert genotype counts to compositions
</h2><span id='topic+HWClo'></span>

<h3>Description</h3>

<p>Function <code>HWClo</code> divides each row of a matrix by its total, and so
produces matrix of compositions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWClo(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWClo_+3A_x">X</code></td>
<td>
<p>A matrix of (genotype) counts</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

<p><code><a href="#topic+HWAlr">HWAlr</a>,<a href="#topic+HWClr">HWClr</a>,<a href="#topic+HWIlr">HWIlr</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  X &lt;- HWData(2,100)
  Y &lt;- HWClo(X)
</code></pre>

<hr>
<h2 id='HWClr'>
Compute the centred log-ratio transformation
</h2><span id='topic+HWClr'></span>

<h3>Description</h3>

<p><code>HWClr</code> computes the centred log-ratio transformation for
genotype counts of bi-allelic genetic markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWClr(X, zeroadj = 0.5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWClr_+3A_x">X</code></td>
<td>
<p>A matrix of genotype counts (columns AA, AB and BB)</p>
</td></tr>
<tr><td><code id="HWClr_+3A_zeroadj">zeroadj</code></td>
<td>
<p>A zero adjustment parameter (0.5 by default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix or vector of log-ratio coordinates
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>References</h3>

<p>Graffelman, J. and Egozcue, J. J. (2011) Hardy-Weinberg
equilibrium: a non-parametric compositional approach. In: Vera
Pawlowsky-Glahn and Antonella Buccianti (eds.) Compositional Data
Analysis: Theory and Applications, John Wiley &amp; Sons, Ltd, pp. 207-215</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWAlr">HWAlr</a>,<a href="#topic+HWIlr">HWIlr</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   X &lt;- HWData(100,100)
   Y &lt;- HWClr(X)
</code></pre>

<hr>
<h2 id='HWClrPlot'>Plot genetic markers in centred log-ratio coordinates</h2><span id='topic+HWClrPlot'></span>

<h3>Description</h3>

<p><code>HWClrPlot</code> creates a scatter plot of the centred log-ratio coordinates of
bi-allelic genetic markers. Hardy-Weinberg equilibrium is indicated by
a straigh line in the plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWClrPlot(X, zeroadj = 0.5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWClrPlot_+3A_x">X</code></td>
<td>
<p>A matrix of genotype counts (columns AA, AB, BB)
</p>
</td></tr>
<tr><td><code id="HWClrPlot_+3A_zeroadj">zeroadj</code></td>
<td>
<p>Zero-adjustment parameter. Zero counts in the count
matrix are substituted by <code>zeroadj</code> which is 0.5 by default.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>References</h3>

<p>Graffelman, J. and Egozcue, J. J. (2011) Hardy-Weinberg
equilibrium: a non-parametric compositional approach. In: Vera
Pawlowsky-Glahn and Antonella Buccianti (eds.) Compositional Data
Analysis: Theory and Applications, John Wiley &amp; Sons, Ltd, pp. 207-215</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWAlrPlot">HWAlrPlot</a>,<a href="#topic+HWIlrPlot">HWIlrPlot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   X &lt;- HWClo(HWData(100,100))
   HWClrPlot(X)
</code></pre>

<hr>
<h2 id='HWCondProbAB'> Compute probability of a genotypic sample  </h2><span id='topic+HWCondProbAB'></span>

<h3>Description</h3>

<p>Computes the probability of a particular genotypic sample given the
allele count, sample size and number of heterozygotes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWCondProbAB(n, nA, nAB)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWCondProbAB_+3A_n">n</code></td>
<td>
 <p><code>n</code> is the total sample size (total number of individuals)</p>
</td></tr>
<tr><td><code id="HWCondProbAB_+3A_na">nA</code></td>
<td>
 <p><code>nA</code> is the number of A alleles in the sample </p>
</td></tr>
<tr><td><code id="HWCondProbAB_+3A_nab">nAB</code></td>
<td>
 <p><code>nAB</code> is the number of heterozygotes in the sample </p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>p</code></td>
<td>
<p>probability of the particular sample</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>See Also</h3>

<p><code><a href="#topic+HWExact">HWExact</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(298,489,213)
names(x) &lt;- c("MM","MN","NN")
n &lt;- sum(x)
nM &lt;- 2*x[1]+x[2]
nMN &lt;- x[2]
p &lt;- HWCondProbAB(n,nM,nMN)
</code></pre>

<hr>
<h2 id='HWD'>
Compute disequilibrium statistic D 
</h2><span id='topic+HWD'></span>

<h3>Description</h3>

<p>Function <code>HWD</code> computes Weir's disequilibrium coefficient D.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWD(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWD_+3A_x">X</code></td>
<td>
<p>a vector of genotype counts (AA, AB, BB)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the disequilibrium coefficient</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Weir, B.S. (1996) Genetic data analysis II. Sinauer
Associates, Massachusetts. See Chapter3.</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWf">HWf</a></code> <code><a href="#topic+HWChisq">HWChisq</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(MM=298,MN=489,NN=213)
D &lt;- HWD(x)
cat("Disequilibrium coefficient: ",D,"\n")
</code></pre>

<hr>
<h2 id='HWData'>Generate genetic marker data in or out of Hardy-Weinberg Equilibrium</h2><span id='topic+HWData'></span>

<h3>Description</h3>

<p>HWData generates samples of genotypic counts under various schemes. It mainly
uses sampling from the multinomial distribution for given or random allele frequencies,
either assuming Hardy-Weinberg proportions or a specified degree of inbreeding. Sampling
can also be performed conditional on the allele frequency. The same procedures are
also available for X linked markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWData(nm = 100, n = rep(100, nm), f = rep(0, nm), p = NULL, conditional = FALSE,
exactequilibrium = FALSE, x.linked = FALSE, nA = NULL, n.males = round(0.5 * n),
shape1 = 1, shape2 = 1, counts = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWData_+3A_nm">nm</code></td>
<td>
<p>The number of bi-allelic markers.</p>
</td></tr>
<tr><td><code id="HWData_+3A_n">n</code></td>
<td>
<p>The sample sizes.</p>
</td></tr>    
<tr><td><code id="HWData_+3A_f">f</code></td>
<td>
<p>The inbreeding coefficients (only for autosomal markers)</p>
</td></tr>
<tr><td><code id="HWData_+3A_p">p</code></td>
<td>
<p>a vector of allele frequencies</p>
</td></tr>
<tr><td><code id="HWData_+3A_conditional">conditional</code></td>
<td>
<p>if <code>TRUE</code> The Levene-Haldane distribution will be used for sampling autosomal markers,
the Graffelman-Weir distribution for sampling X chromosomal markers. If <code>FALSE</code> a multinomial distribution is used</p>
</td></tr>
<tr><td><code id="HWData_+3A_exactequilibrium">exactequilibrium</code></td>
<td>
<p>generates data in exact HWE if set to
<code>TRUE</code></p>
</td></tr>
<tr><td><code id="HWData_+3A_x.linked">x.linked</code></td>
<td>
<p>Simulated autosomal markers (<code>x.linked=FALSE</code>,
the default) or X-chromosomal markers (<code>x.linked=TRUE</code>)</p>
</td></tr>
<tr><td><code id="HWData_+3A_na">nA</code></td>
<td>
<p>A vector of minor allele counts, one for each marker. If not
specified, it will be calculated from <code>p</code></p>
</td></tr>
<tr><td><code id="HWData_+3A_n.males">n.males</code></td>
<td>
<p>The number of males (only relevant if <code>x.linked = TRUE</code>)</p>
</td></tr>
<tr><td><code id="HWData_+3A_shape1">shape1</code></td>
<td>
<p>First shape parameter of the beta distribution used to generate allele frequencies</p>
</td></tr>
<tr><td><code id="HWData_+3A_shape2">shape2</code></td>
<td>
<p>Second shape parameter of the beta distribution used to generate allele frequencies</p>
</td></tr>
<tr><td><code id="HWData_+3A_counts">counts</code></td>
<td>
<p>If <code>counts=TRUE</code> the output are genotype counts, if <code>counts=TRUE</code> relative genotype frequencies</p>
</td></tr>  
</table>


<h3>Details</h3>

<p>Option <code>pfixed</code> is deprecated and replaced by <code>conditional</code>
Option <code>pdist</code> is deprecated and replaced by parameters <code>shape1</code> and <code>shape2</code>
</p>
<p><code>HWData</code> returns a matrix of genotype counts, <code>nm</code> by 3 for
autsomal markers or <code>nm</code> by 5 for X-chromosomal markers.
</p>
<p>If the inbreeding coefficient is specified (<code>f</code>) it will only
take effect for autosomal markers (<code>x.linked=FALSE</code>) and
multinomial sampling (<code>conditional=FALSE</code>).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>X</code></td>
<td>
<p>A matrix containing the genotype counts.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>See Also</h3>

<p><code><a href="#topic+HWTernaryPlot">HWTernaryPlot</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Generate 100 SNPs with uniform allele frequency under the equilibrium assumption.
#
out &lt;- HWData(nm=100,n=100)
#
# Generate genotype frequencies of 100 SNPs with uniform allele frequency assuming exact equilbrium.
#
X &lt;- HWData(nm=100, exactequilibrium = TRUE, counts = FALSE)
#
# Generate 100 SNPs (as counts), all having an expected A allele frequency of 0.50
#
X &lt;- HWData(nm=100,p=0.5)
#
#  Generate 100 SNPs, 50 with A allele frequency 0.25 and 50 with A allele frequency 0.75,
#  assuming fixed allele frequencies. 
#
X &lt;- HWData(nm=100,p=rep(c(0.25,0.75),50), conditional = TRUE)
#
#  Generate 100 SNPs with a skewed (beta) distribution of the allele frequency,
#  rich in variants with a low minor allele frequency.
#
X &lt;- HWData(nm=100,shape1=1,shape2=10)
#
# Generate 100 X chromosomal SNPs with uniformly distributed allele frequency.
#
X &lt;- HWData(nm=100, x.linked = TRUE)
</code></pre>

<hr>
<h2 id='HWEM'>
Estimation of contributions of two populations to a sample of genotype frequencies with the EM algorithm.
</h2><span id='topic+HWEM'></span>

<h3>Description</h3>

<p>Function <code>HWEM</code> estimates the relative contributions of two populations with different allele frequencies that contributed
to the genotype frequencies of sample which is a mixture of these two populations. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWEM(x, p = NULL, G = NULL, delta.init = c(0.5, 0.5), itmax = 50,
     eps = 1e-06, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWEM_+3A_x">x</code></td>
<td>
<p>a 3x1 vector of genotype frequencies of the mixed sample, either counts or proportions.</p>
</td></tr>
<tr><td><code id="HWEM_+3A_p">p</code></td>
<td>
<p>a vector with two allele frequencies.</p>
</td></tr>
<tr><td><code id="HWEM_+3A_g">G</code></td>
<td>
<p>a 3x2 matrix with genotype frequencies as columns.</p>
</td></tr>
<tr><td><code id="HWEM_+3A_delta.init">delta.init</code></td>
<td>
<p>initial values for the contributions of each populations</p>
</td></tr>
<tr><td><code id="HWEM_+3A_itmax">itmax</code></td>
<td>
<p>maximum number of iterations</p>
</td></tr>
<tr><td><code id="HWEM_+3A_eps">eps</code></td>
<td>
<p>tolerance criterion for convergence</p>
</td></tr>
<tr><td><code id="HWEM_+3A_verbose">verbose</code></td>
<td>
<p><code>verbose = TRUE</code> prints results, <code>verbose = FALSE</code> is silent.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>HWEM</code> employs the EM algorithm to estimate the contribution of two populations to the genotype frequencies of a bi-allelic
marker, assumed to be typed for a mixed sample with individuals coming from two different populations with different allele
frequencies. To estimate the contributions, both the mixed sample genotype frequencies (x) and either the genotype frequencies (G)
or the allele frequencies (p) of the original populations must be specified. In case only allele frequencies are specified,
Hardy-Weinberg proportions are assumed for the genotype frequencies. 
</p>


<h3>Value</h3>

<p>a vector with two proportions, ordered according to the specified allele or genotype frequencies.</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Dempster, A.P., Laird, N.M. and Rubin, D.B. (1977) Maximum Likelihood from Incomplete Data via the EM Algorithm. Journal of the Royal
Statistical Society. Series B (Methodological) 39(1) pp. 1&ndash;38.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># genotype frequencies population 1
g1 &lt;- c(0.034, 0.330, 0.636)
# genotype frequencies population 2
g2 &lt;- c(0.349, 0.493, 0.158)
# sample from the mixed population
x  &lt;- c(0.270, 0.453,0.277)
#
# estimation based on genotype frequencies
#
G &lt;- cbind(g1,g2)
contributions &lt;- HWEM(x,G=G)
#
# estimation based on allele frequencies
#
p &lt;- c(af(g1),af(g2))
contributions &lt;- HWEM(x,p=p)
</code></pre>

<hr>
<h2 id='HWExact'> Exact test for Hardy-Weinberg equilibrium </h2><span id='topic+HWExact'></span>

<h3>Description</h3>

<p><code>HWExact</code> performs an exact test for Hardy-Weinberg equilibrium
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWExact(X, alternative = "two.sided", pvaluetype = "selome", eps=1e-10, x.linked =
FALSE, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWExact_+3A_x">X</code></td>
<td>
<p> vector with the genotype counts AA, AB, BB </p>
</td></tr>
<tr><td><code id="HWExact_+3A_alternative">alternative</code></td>
<td>
<p><code>two.sided</code> (default) will perform a two-sided
test where both an excess and a dearth of heterozygotes count as
evidence against HWE. <code>less</code> is a one-sided test where only
dearth of heterozygotes counts a evidence against HWE,
<code>greater</code> is a one-sided test where only excess of
heterozygotes counts as evidence against HWE.</p>
</td></tr>
<tr><td><code id="HWExact_+3A_pvaluetype">pvaluetype</code></td>
<td>
<p>if <code>pvaluetype</code> is set to <code>dost</code>
then the p-value of a two-sided test is computed as twice the tail area
of a one-sided test. When set to <code>selome</code>, the p-value
is computed as the sum of the probabilities of all samples less or
equally likely as the current sample. When set to <code>midp</code>, the
p-value is computed as half the probability of the current sample +
the probabilities of all samples that are more extreme.</p>
</td></tr>
<tr><td><code id="HWExact_+3A_x.linked">x.linked</code></td>
<td>
 <p><code>x.linked=FALSE</code> indicates the marker is
autosomal (default), and <code>x.linked=TRUE</code> indicates it resides
on the X-chromosome.</p>
</td></tr>
<tr><td><code id="HWExact_+3A_eps">eps</code></td>
<td>
<p>a tolerance that can be set for comparing probabilities in order to include
tied outcomes</p>
</td></tr>
<tr><td><code id="HWExact_+3A_verbose">verbose</code></td>
<td>
<p> print results or not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>HWExact</code> uses the recursion equations described by Wigginton
et. al.
</p>
<p>For testing large sets of bi-allelic variants, use the faster code in <code><a href="#topic+HWExactStats">HWExactStats</a></code>.
</p>
<p>For large samples, <code>HWExact</code> may give the error message:
&quot;evaluation nested too deeply: infinite recursion&quot;. This can usually
be resolved by increasing R's limit on nested expressions with
<code>options(expressions=10000)</code> or a higher limit. With higher
limits, the error message &quot;protect(): protection stack overflow&quot; can
occur. This error can usually be resolved by increasing R's
protection stack with the command line option
<code>--max-ppsize 100000</code> or higer values. However, with such large
samples the exact test will give virtually the same result as a
chi-square test, and it may be easier to use <code>HWChisq</code> in these
circumstances.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pval</code></td>
<td>
<p> p-value of the exact test </p>
</td></tr>
<tr><td><code>prob</code></td>
<td>
<p> probabilities of all possible samples with the same
sample size and minor allele count</p>
</td></tr>
<tr><td><code>pofthesample</code></td>
<td>
<p> probability of the observed sample </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>References</h3>

<p>Weir, B.S. (1996) Genetic data analysis II. Sinauer
Associates, Massachusetts. See Chapter3.
</p>
<p>Wigginton, J.E., Cutler, D.J. and Abecasis, G.R. (2005) A note on exact
tests of Hardy-Weinberg equilibrium, American Journal of Human
Genetics (76) pp. 887-893.
</p>
<p>Graffelman, J. and Moreno, V. (2013) The mid p-value in exact tests
for Hardy-Weinberg equilibrium, Statistical Applications in Genetics
and Molecular Biology 12(4) pp. 433-448.</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWLratio">HWLratio</a></code>, <code><a href="#topic+HWChisq">HWChisq</a></code>, <code><a href="#topic+HWExactStats">HWExactStats</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Example for an autosomal marker using the standard exact p-value
#
x &lt;- c(298,489,213)
names(x) &lt;- c("MM","MN","NN")
HW.test &lt;- HWExact(x,verbose=TRUE)
#
# Example for an autosomal marker using the mid p-value
#
HW.test &lt;- HWExact(x,verbose=TRUE,pvaluetype="midp")
#
# Example x-linked markers
#
rs5968922  &lt;- c(A=392, B=212, AA=275, AB=296, BB=80 )
HWExact(rs5968922,x.linked=TRUE,verbose=TRUE)
#
#
#
y &lt;- c(GG=48, CG=209, CC=277, G=129, C=337) 
HWExact(y,x.linked=TRUE)
#
#
</code></pre>

<hr>
<h2 id='HWExactMat'>Matrix version of HWExact</h2><span id='topic+HWExactMat'></span>

<h3>Description</h3>

<p>Function <code>HWExactMat</code> is deprecated; use <code>HWExactStats</code> instead. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWExactMat(X, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWExactMat_+3A_x">X</code></td>
<td>
<p>A n times 3 matrix of genotypic counts (AA,AB,BB)</p>
</td></tr>
<tr><td><code id="HWExactMat_+3A_...">...</code></td>
<td>
<p>extra arguments that are passed on to <code>HWExact</code></p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pvalvec</code></td>
<td>
<p>Vector with the p-values of each test</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWExact">HWExact</a></code> <code><a href="#topic+HWExactStats">HWExactStats</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- HWData(100,100)
colnames(X) &lt;- c("MM","MN","NN")
Results &lt;- HWExactMat(X)
Output &lt;- cbind(X,Results)
print(Output)
</code></pre>

<hr>
<h2 id='HWExactPrevious'> Exact test for Hardy-Weinberg equilibrium </h2><span id='topic+HWExactPrevious'></span>

<h3>Description</h3>

<p><code>HWExactPrevious</code> performs an exact test for Hardy-Weinberg equilibrium
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWExactPrevious(X, alternative = "two.sided", pvaluetype = "selome",
x.linked = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWExactPrevious_+3A_x">X</code></td>
<td>
<p> vector with the genotype counts AA, AB, BB </p>
</td></tr>
<tr><td><code id="HWExactPrevious_+3A_alternative">alternative</code></td>
<td>
<p><code>two.sided</code> (default) will perform a two-sided
test where both an excess and a dearth of heterozygotes count as
evidence against HWE. <code>less</code> is a one-sided test where only
dearth of heterozygotes counts a evidence against HWE,
<code>greater</code> is a one-sided test where only excess of
heterozygotes counts as evidence against HWE.</p>
</td></tr>
<tr><td><code id="HWExactPrevious_+3A_pvaluetype">pvaluetype</code></td>
<td>
<p>if <code>pvaluetype</code> is set to <code>dost</code>
then the p-value of a two-sided test is computed as twice the tail area
of a one-sided test. When set to <code>selome</code>, the p-value
is computed as the sum of the probabilities of all samples less or
equally likely as the current sample. When set to <code>midp</code>, the
p-value is computed as half the probability of the current sample +
the probabilities of all samples that are more extreme.</p>
</td></tr>
<tr><td><code id="HWExactPrevious_+3A_x.linked">x.linked</code></td>
<td>
 <p><code>x.linked=FALSE</code> indicates the marker is
autosomal (default), and <code>x.linked=TRUE</code> indicates it resides
on the X-chromosome.</p>
</td></tr>
<tr><td><code id="HWExactPrevious_+3A_verbose">verbose</code></td>
<td>
<p> print results or not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>HWExactPrevious</code> uses the recursion equations described by Wigginton
et. al.
</p>
<p>For large samples, <code>HWExactPrevious</code> may give the error message:
&quot;evaluation nested too deeply: infinite recursion&quot;. This can usually
be resolved by increasing R's limit on nested expressions with
<code>options(expressions=10000)</code> or a higher limit. With higher
limits, the error message &quot;protect(): protection stack overflow&quot; can
occur. This error can usually be resolved by increasing R's
protection stack with the command line option
<code>--max-ppsize 100000</code> or higer values. However, with such large
samples the exact test will give virtually the same result as a
chi-square test, and it may be easier to use <code>HWChisq</code> in these
circumstances.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pval</code></td>
<td>
<p> p-value of the exact test </p>
</td></tr>
<tr><td><code>prob</code></td>
<td>
<p> probabilities of all possible samples with the same
sample size and minor allele count</p>
</td></tr>
<tr><td><code>pofthesample</code></td>
<td>
<p> probability of the observed sample </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>References</h3>

<p>Weir, B.S. (1996) Genetic data analysis II. Sinauer
Associates, Massachusetts. See Chapter3.
</p>
<p>Wigginton, J.E., Cutler, D.J. and Abecasis, G.R. (2005) A note on exact
tests of Hardy-Weinberg equilibrium, American Journal of Human
Genetics (76) pp. 887-893.</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWLratio">HWLratio</a></code>, <code><a href="#topic+HWChisq">HWChisq</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Example autosomal marker
#
x &lt;- c(298,489,213)
names(x) &lt;- c("MM","MN","NN")
## Not run: HW.test &lt;- HWExactPrevious(x,verbose=TRUE)
#
# Example x-linked marker
#
rs5968922  &lt;- c(A=392, B=212, AA=275, AB=296, BB=80 )
## Not run: HWExactPrevious(rs5968922,x.linked=TRUE,verbose=TRUE) 
</code></pre>

<hr>
<h2 id='HWExactStats'>
Computation of Exact p-values for Hardy-Weinberg equilibrium for sets of SNPs
</h2><span id='topic+HWExactStats'></span>

<h3>Description</h3>

<p><code>HWExactStats</code> is a function for the computation of Exact p-values
for a large set of bi-allelic markers (typically SNPs). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWExactStats(X, x.linked = FALSE, plinkcode = TRUE, midp = FALSE,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWExactStats_+3A_x">X</code></td>
<td>
<p>A matrix with genotype counts, one row per marker. <code>X</code>
should have 5 columns for an X-chromosomal data set and 3 columns
for an autosomal data set.</p>
</td></tr>
<tr><td><code id="HWExactStats_+3A_x.linked">x.linked</code></td>
<td>
<p>Logical indicating whether the markers are autosomal (<code>x.linked=FALSE</code>)
or X-chromosomal (<code>x.linked=TRUE</code>).</p>
</td></tr>
<tr><td><code id="HWExactStats_+3A_plinkcode">plinkcode</code></td>
<td>
<p>Logical indicating whether to use faster C++ code
from the PLINK software.</p>
</td></tr>
<tr><td><code id="HWExactStats_+3A_midp">midp</code></td>
<td>
<p>Logical indicating whether to use the mid p-value for the
C++ code or not</p>
</td></tr>
<tr><td><code id="HWExactStats_+3A_...">...</code></td>
<td>
<p>extra arguments that are passed on to <code>HWExact</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Matrix <code>X</code> should strictly comply with the following format. For
an autosomal dataset is should contain the 3 genotype counts in order
(AA,AB,BB). For an X-chromosomal dataset it should contain the 5
genotype counts in order (A,B,AA,AB,BB) where A and B are the male
counts and AA, AB and BB the female counts.
</p>
<p>Argument <code>plinkcode=TRUE</code> (the default) will use C++ code for faster calculation (functions <code>SNPHWE2</code> and
<code>SNPHWEX</code>) with larger datasets. The C++ code  was generously shared by Christopher
Chang, and the same code is used in the program
<a href="http://www.cog-genomics.org/plink/2.0">PLINK (2.0)</a>.
</p>


<h3>Value</h3>

<p>A vector of p-values</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> (R code) and
Christopher Chang <a href="mailto:chrchang523@gmail.com">chrchang523@gmail.com</a> (C++ code)</p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2016) Testing for Hardy-Weinberg
equilibrium at bi-allelic genetic markers on the X chromosome.
Heredity 116(6) pp. 558&ndash;568. <a href="https://doi.org/10.1038/hdy.2016.20">doi:10.1038/hdy.2016.20</a>
</p>
<p>Purcell et al. (2007) PLINK: A Toolset for Whole-Genome Association and
Population-Based Linkage Analysis. American Journal of Human Genetics
81(3) pp. 559&ndash;575.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWExact">HWExact</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Autosomal example
#
set.seed(123)
X &lt;- HWData(1000,100)
monom &lt;- (X[,2]==0 &amp; X[,1]==0) | (X[,2]==0 &amp; X[,3]==0)
X &lt;- X[!monom,] # exclude monomorphics
Exact.pvalues &lt;- HWExactStats(X,x.linked=FALSE)
#
# X-chromosomal example
#
X &lt;- HWData(1000,100,n.males=50,nA=75,x.linked=TRUE)
Exact.pvalues &lt;- HWExactStats(X,x.linked=TRUE)
</code></pre>

<hr>
<h2 id='HWf'>
Computation of inbreeding coefficient
</h2><span id='topic+HWf'></span>

<h3>Description</h3>

<p><code>HWf</code> computes the inbreeding coefficient for sample
of genotype counts, or a matrix of genotype counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWf(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWf_+3A_x">X</code></td>
<td>
<p>a vector or matrix of genotype counts (AA, AB, BB)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For monomorphic markers a warning is issued, and the estimate for the
inbreeding coefficient is NaN. 
</p>


<h3>Value</h3>

<p>Returns a single inbreeding coefficient (intraclass correlation coefficient), if <code>X</code> is a single sample, or a vector of inbreeding coefficients, if <code>X</code> is a matrix with genotype counts.
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Crow, J. F. and Kimura, M. (1970) An introduction to population genetics theory.
Harper &amp; Row, publishers, New York
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWChisq">HWChisq</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# A single sample
#
x &lt;- c(MM=298,MN=489,NN=213)
fhat &lt;- HWf(x)
fhat
#
# Multiple samples
#
set.seed(123)
X &lt;- HWData(nm=100,n=1000)
fhat &lt;- HWf(X)
</code></pre>

<hr>
<h2 id='HWGenotypePlot'>
Scatter plot of the genotype frequencies 
</h2><span id='topic+HWGenotypePlot'></span>

<h3>Description</h3>

<p><code>HWGenotypePlot</code> makes a scatterplots of the AB or BB frequency
versus the AA frequency and represents a blue curve indicating the
Hardy-Weinberg equilibrium condition.</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWGenotypePlot(X, plottype = 1, xlab = expression(f[AA]), ylab =
ifelse(plottype == 1, expression(f[AB]), expression(f[BB])), asp = 1,
pch = 19, xlim = c(0, 1), ylim = c(0, 1), cex = 1, cex.axis = 2, cex.lab = 2, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWGenotypePlot_+3A_x">X</code></td>
<td>
<p>A matrix of genotype counts or frequencies with three columns (AA, AB, BB)</p>
</td></tr>
<tr><td><code id="HWGenotypePlot_+3A_plottype">plottype</code></td>
<td>
<p><code>plottype=1</code> produces a plot of AB versus AA,
<code>plottype=2</code> produced a plot of BB versus AA.</p>
</td></tr>
<tr><td><code id="HWGenotypePlot_+3A_xlab">xlab</code></td>
<td>
<p>A label for the x axis</p>
</td></tr>
<tr><td><code id="HWGenotypePlot_+3A_ylab">ylab</code></td>
<td>
<p>A label for the y axis</p>
</td></tr>
<tr><td><code id="HWGenotypePlot_+3A_asp">asp</code></td>
<td>
<p>Aspec ratio (1 by default)</p>
</td></tr>
<tr><td><code id="HWGenotypePlot_+3A_pch">pch</code></td>
<td>
<p>Plotting charachter (19 by default)</p>
</td></tr>
<tr><td><code id="HWGenotypePlot_+3A_xlim">xlim</code></td>
<td>
<p>Limits for the x axis (0-1 by default)</p>
</td></tr>
<tr><td><code id="HWGenotypePlot_+3A_ylim">ylim</code></td>
<td>
<p>Limits for the y axis (0-1 by default)</p>
</td></tr>
<tr><td><code id="HWGenotypePlot_+3A_cex">cex</code></td>
<td>
<p>Character expansion factor (1 by default)</p>
</td></tr>
<tr><td><code id="HWGenotypePlot_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Character expansion factor for the axes (2 by default)</p>
</td></tr>
<tr><td><code id="HWGenotypePlot_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Character expansion factor for labels of axis (2 by default)</p>
</td></tr>
<tr><td><code id="HWGenotypePlot_+3A_...">...</code></td>
<td>
<p>Additional arguments for the <code>plot</code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWTernaryPlot">HWTernaryPlot</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>n &lt;- 100 # sample size
m &lt;- 100 # number of markers
Xc &lt;- HWClo(HWData(n,m))
HWGenotypePlot(Xc,plottype=1,main="Heterozygote-homozygote scatterplot")
</code></pre>

<hr>
<h2 id='HWIlr'>
Compute isometric log ratio coordinates.
</h2><span id='topic+HWIlr'></span>

<h3>Description</h3>

<p><code>HWIlr</code> computes isometric log ratio coordinates for genotypic
compositions (AA, AB, BB)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWIlr(X, zeroadj = 0.5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWIlr_+3A_x">X</code></td>
<td>
<p>A matrix of genotype counts, markers in rows, counts for AA,
AB and BB in three columns
</p>
</td></tr>
<tr><td><code id="HWIlr_+3A_zeroadj">zeroadj</code></td>
<td>
<p>Adjustment for zeros (0.5 by defaults)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of log ratio coordinates.
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>References</h3>

<p>Egozcue, J.J., Pawlowsky-Glahn, V., Mateu-Figueras, G. and
Barcelo-Vidal, C. (2003) Isometric Logratio Transformations for Compositional
Data Analysis. Mathematical Geology 35(3), pp. 279-300.
</p>
<p>Graffelman, J. and Egozcue, J. J. (2011) Hardy-Weinberg
equilibrium: a non-parametric compositional approach. In: Vera
Pawlowsky-Glahn and Antonella Buccianti (eds.) Compositional Data
Analysis: Theory and Applications, John Wiley &amp; Sons, Ltd, pp. 207-215
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWAlr">HWAlr</a>,<a href="#topic+HWClr">HWClr</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- HWData(100,100)
Y &lt;- HWIlr(X)
</code></pre>

<hr>
<h2 id='HWIlrPlot'>
Plot bi-allelic genetic markers in isometric log ratio coordinates
</h2><span id='topic+HWIlrPlot'></span>

<h3>Description</h3>

<p><code>HWIlrPlot</code> makes a scatter plot of the isometric log ratio
coordinates for bia-llelic markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWIlrPlot(X, zeroadj = 0.5, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWIlrPlot_+3A_x">X</code></td>
<td>
<p>Matrix of genotype counts, one marker per row, AA, AB and BB
in three columns
</p>
</td></tr>
<tr><td><code id="HWIlrPlot_+3A_zeroadj">zeroadj</code></td>
<td>
<p>Adjustment for zero values (0.5 by default)
</p>
</td></tr>
<tr><td><code id="HWIlrPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments for function <code>plot</code>
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of log ratio coordinates.
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>References</h3>

<p>Graffelman, J. and Egozcue, J. J. (2011) Hardy-Weinberg equilibrium: a
nonparametric compositional approach. 
In Pawlowsky-Glahn, V. and Buccianti A., editors, Compositional Data Analysis: Theory and Applications, pages 208-217, John Wiley &amp; Sons, Ltd. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWAlrPlot">HWAlrPlot</a>,<a href="#topic+HWClrPlot">HWClrPlot</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- HWClo(HWData(100,100))
HWIlrPlot(X)
</code></pre>

<hr>
<h2 id='HWLindley'>
Calculate a posteriori density for Lindley's alpha
</h2><span id='topic+HWLindley'></span>

<h3>Description</h3>

<p>Function <code>HWLindley</code> calculates the posterior density for disequilibrium measure alpha, as defined by Lindley (1988).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWLindley(alphaseq = seq(-3, 3, by = 0.01), x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWLindley_+3A_alphaseq">alphaseq</code></td>
<td>
<p>a single value or a sequence of values for alpha</p>
</td></tr>
<tr><td><code id="HWLindley_+3A_x">x</code></td>
<td>
<p>the genotype count vector in format (AA,AB,BB)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Numerical integration is used to compute the density.
</p>


<h3>Value</h3>

<p>a vector with values of the density for each value in <code>alphaseq</code>
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Lindley, D.V. (1988) Statistical Inference Concerning Hardy-Weinberg Equilibrium. In: 
Bernardo, J.M., DeGroot, M.H., Lindley, D.V. and Smith, A.F.M. Bayesian Statistics, 3,
pp. 307-326. Oxford University Press.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWPosterior">HWPosterior</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(MM=298,MN=489,NN=213)
post.dens &lt;- HWLindley(seq(-1,1,by=0.01),x)
## Not run: 
plot(seq(-1,1,by=0.01),post.dens,type="l",xlab=expression(alpha),
     ylab=expression(pi(alpha)))
segments(0,0,0,HWLindley(0,x),lty="dotted",col="red")

## End(Not run)
</code></pre>

<hr>
<h2 id='HWLindley.cri'>
Calculate a credible interval for Lindley's alpha for HWE,
</h2><span id='topic+HWLindley.cri'></span>

<h3>Description</h3>

<p>Function <code>HWLindley.cri</code> calculates a Bayesian credible interval using Lindley's
posterior density for equilibrium paramater alpha.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWLindley.cri(x, verbose = TRUE, limits = c(0.025, 0.975))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWLindley.cri_+3A_x">x</code></td>
<td>
<p>a vector of three genotype counts in order (AA,AB,BB).</p>
</td></tr>
<tr><td><code id="HWLindley.cri_+3A_verbose">verbose</code></td>
<td>
<p>print output (<code>verbose=TRUE</code>) or be silent</p>
</td></tr>
<tr><td><code id="HWLindley.cri_+3A_limits">limits</code></td>
<td>
<p>upper and lower probability limits of the interval</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The limits are found by numerical integration over Lindley's density.
</p>


<h3>Value</h3>

<p>a vector with the lower and upper limit of the credible interval</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Lindley, D.V. (1988) Statistical Inference Concerning Hardy-Weinberg Equilibrium. In: 
Bernardo, J.M., DeGroot, M.H., Lindley, D.V. and Smith, A.F.M. Bayesian Statistics, 3,
pp. 307-326. Oxford University Press.
</p>
<p>Graffelman, J. (2020) Statistical tests for the Hardy-Weinberg equilibrium.
Wiley StatsRef: Statistics Reference Online <a href="https://doi.org/10.1002/9781118445112.stat08274">doi:10.1002/9781118445112.stat08274</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWLindley">HWLindley</a></code>, <code><a href="#topic+HWPosterior">HWPosterior</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# MN blood group data
#
x &lt;- c(MM=298,MN=489,NN=213)
#
# credible interval of 95%
#
HWLindley.cri(x)
#
# credible interval of 90%
#
HWLindley.cri(x,limits=c(0.05,0.95))
</code></pre>

<hr>
<h2 id='HWLRAllTests'>
Perform most relevant likelihood ratio test for Hardy-Weinberg
equilibrium and equality of allele frequencies
</h2><span id='topic+HWLRAllTests'></span>

<h3>Description</h3>

<p>Function <code>HWLRAllTests</code> performs a set of likelihood ratio tests in
relation with Hardy-Weinberg proportions (HWP) and equality of allele
frequencies (EAF) for autosomal bi-allelic genetic variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWLRAllTests(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWLRAllTests_+3A_x">x</code></td>
<td>
<p>Male genotype counts (AA,AB,BB)</p>
</td></tr>
<tr><td><code id="HWLRAllTests_+3A_y">y</code></td>
<td>
<p>Female genotype counts (AA,AB,BB)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function <code>HWLRAllTests</code> calls <code>HWLRtest</code> and calculates the
p-value of six different tests: 1) joint HWP and EAF (A-F); 2) EAF
irrespective of HWP (C-F); 3) HWP irrespective of EAF (D-F); 4) HWP
versus EIC (given EAF) (A-B); 5) EIC irrespective of EAF (E-F) and 6)
HWP versus EIC. Letters refer to scenarios described by Graffelman &amp;
Weir (2018).
</p>


<h3>Value</h3>

<p>A named vector with six p-values</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a></p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2018) On the testing of Hardy-Weinberg
proportions and equality of allele frequencies in males and females at
bi-allelic genetic
markers. Genetic Epidemiology 42(1): pp. 34&ndash;48. <a href="https://doi.org/10.1002/gepi.22079">doi:10.1002/gepi.22079</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWLRtest">HWLRtest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>males &lt;- c(AA=11,AB=32,BB=13) 
females &lt;- c(AA=14,AB=23,BB=11)
pvalues &lt;- HWLRAllTests(males,females)
print(pvalues)
</code></pre>

<hr>
<h2 id='HWLratio'>Likelihood ratio test for Hardy Weinberg equilibrium</h2><span id='topic+HWLratio'></span>

<h3>Description</h3>

<p><code>HWLratio</code> performs the Likelihood ratio test for Hardy Weinberg
equilibrium, both for autosomal and X-chromosomal markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWLratio(X, verbose = TRUE, x.linked = FALSE) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWLratio_+3A_x">X</code></td>
<td>
<p><code>X</code> a vector containing the genotypic counts (AA,AB,BB).</p>
</td></tr>
<tr><td><code id="HWLratio_+3A_verbose">verbose</code></td>
<td>
<p><code>verbose = TRUE</code> prints results, <code>verbose = FALSE</code>
is silent.</p>
</td></tr>
<tr><td><code id="HWLratio_+3A_x.linked">x.linked</code></td>
<td>
 <p><code>x.linked=FALSE</code> indicates the marker is
autosomal (default), and <code>x.linked=TRUE</code> indicates it resides
on the X-chromosome.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>HWLratio</code> returns a list with the components:
</p>
<table role = "presentation">
<tr><td><code>Lambda</code></td>
<td>
<p>the likelihood ratio</p>
</td></tr>
<tr><td><code>G2</code></td>
<td>
<p>-2*log(Lambda)</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>the p-value</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Weir, B.S. (1996) Genetic data analysis II. Sinauer Associates, Massachusetts. See Chapter 3.</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWChisq">HWChisq</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(298,489,213)
names(x) &lt;- c("MM","MN","NN")
HW.test &lt;- HWLratio(x,verbose=TRUE)
#
# Test for X-chromsomal SNPs.
#
rs5968922 &lt;-  c(A=392, B=212, AA=275, AB=296, BB=80)
HW.test &lt;- HWLratio(rs5968922,x.linked=TRUE,verbose=TRUE)
#
#
#
y &lt;- c(GG=48, CG=209, CC=277, G=129, C=337) 
HWLratio(y,x.linked=TRUE)
#
#
</code></pre>

<hr>
<h2 id='HWLRtest'>Perform likelihood ratio test comparing two nested scenarios for a
bi-allelic genetic variant, distinguishing the two sexes.
</h2><span id='topic+HWLRtest'></span>

<h3>Description</h3>

<p>Program <code>HWLRtest</code> performs a likelihood ratio test comparing two
scenarios for an autosomal bi-allelic genetic variant. The scenarios
concern Hardy-Weinberg proportions (HWP) and equality of allele
frequencies (EAF) in both sexes. The different scenarios are described
by Graffelman &amp; Weir (2017). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWLRtest(x, y, scene.null = "S1", scene.alt = "S6", verbose = TRUE, tracing = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWLRtest_+3A_x">x</code></td>
<td>
<p>Male genotype counts</p>
</td></tr>
<tr><td><code id="HWLRtest_+3A_y">y</code></td>
<td>
<p>Female genotype counts</p>
</td></tr>
<tr><td><code id="HWLRtest_+3A_scene.null">scene.null</code></td>
<td>
<p>Scenario under the null hypothesis (E.g. &quot;S1&quot;)</p>
</td></tr>
<tr><td><code id="HWLRtest_+3A_scene.alt">scene.alt</code></td>
<td>
<p>Scenario under the alternative hypothesis (E.g. &quot;S6&quot;)</p>
</td></tr>
<tr><td><code id="HWLRtest_+3A_verbose">verbose</code></td>
<td>
<p>print output or not</p>
</td></tr>
<tr><td><code id="HWLRtest_+3A_tracing">tracing</code></td>
<td>
<p>Show tracing of the numeric likelihood maximization (1)
or not (0).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The different scenarios are indicated with S1, S2, S3, S4, S6 and
S6. S1 refers to Hardy-Weinber proportions and equality of allele
frequencies. S2 refers to equality of allele frequencies and equality
of inbreeding coefficients for the two sexes. S3 refers to equality of
allele frequencies irrespective of HWP. S4 refers to HWP irrespective
of allele frequencies. S5 refers to equality of inbreeding
coefficients irrespective of allele frequencies. S6 is unrestricted.  
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>G2</code></td>
<td>
<p>Likelihood ratio statistic</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>Degrees of freedom of the likelihood ratio statistic</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>p-value</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a></p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2018) On the testing of Hardy-Weinberg
proportions and equality of allele frequencies in males and females at
bi-allelic genetic
markers. Genetic Epidemiology 42(1) pp. 34&ndash;48 <a href="https://doi.org/10.1002/gepi.22079">doi:10.1002/gepi.22079</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWAIC">HWAIC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>males &lt;- c(AA=11,AB=32,BB=13) 
females &lt;- c(AA=14,AB=23,BB=11)

#
# test EAF
#

lr1.out &lt;- HWLRtest(males,females,scene.null="S3",scene.alt="S6")

#
# test EIC given EAF
#


lr2.out &lt;- HWLRtest(males,females,scene.null="S2",scene.alt="S3")

#
# test HWP versus EIC, given EAF.
#

lr3.out &lt;- HWLRtest(males,females,scene.null="S1",scene.alt="S2")

</code></pre>

<hr>
<h2 id='HWMissing'>
Test a bi-allelic marker for Hardy-Weinberg equilibrium in the
presence of missing genotype information.
</h2><span id='topic+HWMissing'></span>

<h3>Description</h3>

<p>Function <code>HWMissing</code> imputes missing genotype data with a
multinomial logit model that uses information from allele intensities
and/or neighbouring markers. Multiple imputation algorithms
implemented in the Mice package are used to obtain imputed data sets.
Inference for HWE is carried out by estimating the inbreeding
coefficient or exact p-values for each imputed data set, and
by combining all estimates
using Rubin's pooling rules.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWMissing(X, imputecolumn = 1, m = 50, coding = c(0,1,2), verbose = FALSE, alpha = 0.05,
    varest = "oneovern", statistic = "chisquare",  alternative =
"two.sided", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWMissing_+3A_x">X</code></td>
<td>
<p>An input data frame. By default, the first column should
contain the SNP with missing values.</p>
</td></tr>
<tr><td><code id="HWMissing_+3A_imputecolumn">imputecolumn</code></td>
<td>
<p>Indicates which column of the supplied data frame
is to be imputed (by default, the first colum, <code>imputecolumn=1</code></p>
</td></tr>
<tr><td><code id="HWMissing_+3A_m">m</code></td>
<td>
<p>The number of imputations (50 by default)</p>
</td></tr>
<tr><td><code id="HWMissing_+3A_coding">coding</code></td>
<td>
<p>Indicates how the genotype data is coded (e.g. 0 for AA,
1 for AB, and 2 for BB).</p>
</td></tr>
<tr><td><code id="HWMissing_+3A_verbose">verbose</code></td>
<td>
 <p><code>verbose = TRUE</code> prints results, <code>verbose = FALSE</code> is silent.</p>
</td></tr>
<tr><td><code id="HWMissing_+3A_alpha">alpha</code></td>
<td>
<p>significance level (0.05 by default) used when computing
confidence intervals</p>
</td></tr>
<tr><td><code id="HWMissing_+3A_varest">varest</code></td>
<td>
<p>Estimator for the variance of the inbreeding
coefficient. <code>varest="oneovern"</code> is the default and sets the
variance under the null (1/n). <code>varest="bailey"</code> uses an
approximation (see details).</p>
</td></tr>
<tr><td><code id="HWMissing_+3A_statistic">statistic</code></td>
<td>
<p>If <code>statistic = "chisquare"</code> then inbreeding
coefficients (equivalent to chisquare statistics) will be computed
for each imputed data set and then combined. If <code>statistic =
      "exact"</code> then one-sided exact tests will be computed for each
imputed data set and the resulting p-values will be combined.</p>
</td></tr>
<tr><td><code id="HWMissing_+3A_alternative">alternative</code></td>
<td>
<p><code>two.sided</code> (default) will perform a two-sided
test where both an excess and a dearth of heterozygotes count as
evidence against HWE. <code>less</code> is a one-sided test where only
dearth of heterozygotes counts a evidence against HWE,
<code>greater</code> is a one-sided test where only excess of
heterozygotes counts as evidence against HWE.</p>
</td></tr>
<tr><td><code id="HWMissing_+3A_...">...</code></td>
<td>
<p>additional options for function <code>mice</code> of the Mice package</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>HWMissing</code> tests one genetic marker (e.g. a SNP)
with missings for HWE. By default, this marker is supposed to be the
first column of dataframe <code>X</code>. The other columns of <code>X</code>
contain covariates to be used in the imputation model. Covariates
will typically be other, correlated markers or allele intensities of
the SNP to be imputed. Covariate markers should be coded as factor
variables whereas allele intensities should be numerical
variables. By default, a polytomous regression model will be used to
impute the missings. If the covariates also contain missings, an
imputation method for each column of <code>X</code> can be specified by
using the <code>method</code> of mice (see example below).
</p>
<p>If there are no covariates, missings can be imputed under the MCAR
assumption. In that case, missings are imputed by taking a random
sample from the observed data. This is what <code>HWMissing</code> will do
if no covariates are supplied, <code>X</code> being a single factor
variable.
</p>
<p>Several estimators for the variance of the inbreeding coefficient
have been described in the literature. The asymptotic variance of the
inbreeding coefficient under the null hypothesis is 1/n, and is used
if <code>varest = "oneovern"</code> is used. This is the recommended
option. Alternatively, the approximation described in Weir (p. 66) can be used
with <code>varest = "bailey"</code>.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Res</code></td>
<td>
<p>A vector with the inbreeding coefficient, a confidence interval
for the inbreeding coefficient, a p-value for a HWE test and missing
data statistics.</p>
</td></tr>
<tr><td><code>Xmat</code></td>
<td>
<p>A matrix with the genotypic composition of each of the
<code>m</code> imputed data sets.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Little, R. J. A. and Rubin, D. B. (2002) Statistical analysis with
missing data. Second edition, New York, John Wiley &amp; sons.
</p>
<p>Graffelman, J., S\'anchez, M., Cook, S. and Moreno, V. (2013) Statistical
inference for Hardy-Weinberg proportions in the presence of missing
genotype information. PLoS ONE 8(12):
e83316. <a href="https://doi.org/10.1371/journal.pone.0083316">doi:10.1371/journal.pone.0083316</a>
</p>
<p>Graffelman, J. (2015) Exploring Diallelic Genetic Markers: The HardyWeinberg Package.
<em>Journal of Statistical Software</em> 64(3): 1-23. <a href="https://doi.org/10.18637/jss.v064.i03">doi:10.18637/jss.v064.i03</a>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWChisq">HWChisq</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Markers)
## Not run: 
set.seed(123)
Results &lt;- HWMissing(Markers[,1],m=50,verbose=TRUE)$Res # no covariates, imputation assuming MCAR.
set.seed(123)
Results &lt;- HWMissing(Markers[,1:3],m=50,verbose=TRUE)$Res # impute with two allele intensities.
set.seed(123)
Results &lt;- HWMissing(Markers[,c(1,4,5)],m=50,verbose=TRUE)$Res # impute with two covariate SNPs

## End(Not run)
</code></pre>

<hr>
<h2 id='HWNetwork'>
Autosomal and X-chromosomal exact tests for HWE via a Network algorithm 
</h2><span id='topic+HWNetwork'></span>

<h3>Description</h3>

<p>Program <code>HWNetwork</code> implements a network algorithm for efficient calculation of exact test p-values in HWE tests with multiple alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWNetwork(a1, a2, ma = NULL, fe = NULL, gender = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWNetwork_+3A_a1">a1</code></td>
<td>
<p>the first allele (expressed as a number)</p>
</td></tr>
<tr><td><code id="HWNetwork_+3A_a2">a2</code></td>
<td>
<p>the second allele (expressed as a number; NA if the variant is X chromosomal)</p>
</td></tr>
<tr><td><code id="HWNetwork_+3A_ma">ma</code></td>
<td>
<p>alternative format: vector of male X chromosomal allele counts.</p>
</td></tr>
<tr><td><code id="HWNetwork_+3A_fe">fe</code></td>
<td>
<p>triangular matrix of female genotype counts</p>
</td></tr>
<tr><td><code id="HWNetwork_+3A_gender">gender</code></td>
<td>
<p>gender of the individual (1=male; 2=female)</p>
</td></tr>
<tr><td><code id="HWNetwork_+3A_verbose">verbose</code></td>
<td>
<p>be silent (<code>verbose=FALSE</code>) or informative (<code>verbose=TRUE</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function <code>HWNetwork</code> accepts data in two formats. Original genotype data (e.g. repeat numbers of microsatellites) can be supplied, or  the data can be supplied in summarized form as a male genotype count vector and a female genotype count matrix. If one of the two male alleles is missing (NA) the variant will taken to be X-chromosomal. If all males have two alleles, the variant will taken to be autosomal.
</p>


<h3>Value</h3>

<p>the exact p-value of the test.
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a></p>


<h3>References</h3>

<p>Aoki, S. (2003) Network algorithm for the Exact Test of Hardy-Weinberg Proportion for Multiple Alleles. <em>Biometrical Journal</em> 45(4),
pp. 471-490.
</p>
<p>Engels, W. R. (2009) Exact Tests for Hardy-Weinberg Proportions. <em>Genetics</em> 183, pp. 1431-1441.
</p>
<p>Graffelman, J. (2015) Exploring Diallelic Genetic Markers: The HardyWeinberg Package.
<em>Journal of Statistical Software</em> 64(3): 1-23. <a href="https://doi.org/10.18637/jss.v064.i03">doi:10.18637/jss.v064.i03</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWExact">HWExact</a>, <a href="#topic+HWExactStats">HWExactStats</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# From vectors with counts of genotypes
#
data(TSIXTriAllelics)
ma &lt;- as.matrix(TSIXTriAllelics[1,2:4])
names(ma) &lt;- c("A","B","C")

fe &lt;- TSIXTriAllelics[1,5:10]
names(fe) &lt;- c("AA","AB","AC","BB","BC","CC")

fe &lt;- HardyWeinberg:::toTriangularfixed(fe)

HWNetwork(ma=ma,fe=fe)
</code></pre>

<hr>
<h2 id='HWPerm'>
Permutation test for Hardy-Weinberg equilibrium
</h2><span id='topic+HWPerm'></span>

<h3>Description</h3>

<p>Function <code>HWPerm</code> does a permutation test for Hardy-Weinberg
equilibrium using a user-supplied test statistic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWPerm(x, nperm = 17000, verbose = TRUE, x.linked = FALSE,
FUN = ifelse(x.linked,Chisquare.x,Chisquare), eps=1e-10, ...) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWPerm_+3A_x">x</code></td>
<td>
<p>A vector of genotype counts (AA,AB,BB)</p>
</td></tr>
<tr><td><code id="HWPerm_+3A_nperm">nperm</code></td>
<td>
<p>The number of permutations</p>
</td></tr>
<tr><td><code id="HWPerm_+3A_verbose">verbose</code></td>
<td>
<p><code>verbose</code> = TRUE will print results,
<code>verbose</code> = FALSE is silent.</p>
</td></tr>    
<tr><td><code id="HWPerm_+3A_x.linked">x.linked</code></td>
<td>
 <p><code>x.linked=FALSE</code> indicates the marker is
autosomal (default), and <code>x.linked=TRUE</code> indicates it resides
on the X-chromosome.</p>
</td></tr>
<tr><td><code id="HWPerm_+3A_fun">FUN</code></td>
<td>
<p>An function call for calculating the test statistic for HWE
(see examples below)</p>
</td></tr>
<tr><td><code id="HWPerm_+3A_eps">eps</code></td>
<td>
<p>Tolerance for comparison of floating point numbers (1e-10 by default)</p>
</td></tr>
<tr><td><code id="HWPerm_+3A_...">...</code></td>
<td>
<p>Additional parameters for the function call argument <code>FUN</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The set of alleles for the observed sample is permuted. Consequently, the test is
conditional on allele frequency.
</p>


<h3>Value</h3>

<p><code>HWPerm</code> returns a list with the components:
</p>
<table role = "presentation">
<tr><td><code>stat</code></td>
<td>
<p>value of the chosen test statistic for the observed sample.</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>p-value of the permutation test.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Ziegler, A. &amp; Konig, I.R. (2006) A statistical approach to genetic
epidemiology. Wiley.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWChisq">HWChisq</a></code>,<code><a href="#topic+HWExact">HWExact</a></code>,<code><a href="#topic+HWLratio">HWLratio</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(MM=298,MN=489,NN=213)
## Not run: 
#
# use a default chi-square statistic
#
HW.test &lt;- HWPerm(x,nperm=10000,verbose=TRUE)
#
# use a chi-square statistic with continuity correction.
#
HW.test &lt;- HWPerm(x,nperm=10000,verbose=TRUE,
FUN=function(z) HWChisq(z,verbose=FALSE)$chisq,cc=0.5) # 
#
# use a likelihood ratio statistic.
#
HW.test &lt;- HWPerm(x,nperm=10000,verbose=TRUE,
FUN=function(y) HWLratio(y,verbose=FALSE)$G2)
# 
# use an exact test p-value
#
HWPerm(x,nperm=10000,verbose=TRUE,FUN=function(y) 1-HWExact(y,verbose=FALSE)$pval)
# 
#
# Permutation test for a marker on the X chromosome
#
rs5968922 &lt;-  c(A=392, B=212, AA=275, AB=296, BB=80)
HW.test &lt;- HWPerm(rs5968922,nperm=10000,x.linked=TRUE,verbose=TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='HWPerm.mult'>
Permutation tests fo variants with multiple alleles
</h2><span id='topic+HWPerm.mult'></span>

<h3>Description</h3>

<p>Function <code>HWPerm.mult</code> implements permutation tests for
Hardy-Weinberg equilibrium for autosomal and X-chromosomal variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWPerm.mult(x, y = NULL, nperm = 17000, eps = 1e-10, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWPerm.mult_+3A_x">x</code></td>
<td>
<p>vector or triangular matrix with male genotype counts</p>
</td></tr>
<tr><td><code id="HWPerm.mult_+3A_y">y</code></td>
<td>
<p>vector or triangular matrix with female genotype counts</p>
</td></tr>
<tr><td><code id="HWPerm.mult_+3A_nperm">nperm</code></td>
<td>
<p>number of permutations (17.000 by default)</p>
</td></tr>
<tr><td><code id="HWPerm.mult_+3A_eps">eps</code></td>
<td>
<p>a tolerance for the comparison of floating point numbers</p>
</td></tr>
<tr><td><code id="HWPerm.mult_+3A_verbose">verbose</code></td>
<td>
<p>print output or not</p>
</td></tr>
<tr><td><code id="HWPerm.mult_+3A_...">...</code></td>
<td>
<p>addtional arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function approximates exact test probabilities for joint tests
for HWE and equality of allele frequencies for variants with multiple
alleles. For purely bi-allelic variant <code>HWPerm</code> can be used which
allows for more statistics than just probabilities.
</p>
<p>If argument <code>y</code> is not specified, gender is considered
irrelevant, and <code>x</code> contains total genotype counts. If <code>x</code>
and <code>y</code> are specified, <code>x</code> should contain male genotype
counts and <code>y</code> female genotype counts. <code>x</code> and <code>y</code> can
be vectors if the variant is bi-allelic, but are assumed lower
triangular if there are more than two alleles. <code>x</code> is still a
vector if there are multiple alleles but the variant is
X-chromosomal. See the examples given below.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pofthesample</code></td>
<td>
<p>probability of the observed sample</p>
</td></tr>
<tr><td><code>pseudodist</code></td>
<td>
<p>probabilities of simulated samples</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>p-value</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a>
</p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2017) Multi-allelic exact tests for Hardy-Weinberg equilibrium that account for gender.
Molecular Ecology Resources. 18(3) pp. 461&ndash;473. <a href="https://doi.org/10.1111/1755-0998.12748">doi:10.1111/1755-0998.12748</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWPerm">HWPerm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# bi-allelic autosomal
#

x1 &lt;- c(AA=298,AB=489,BB=213)
## Not run: 
out &lt;- HWPerm.mult(x1)

## End(Not run)

#
# bi-allelic X-chromosomal
#

x2.m &lt;- c(A=39, B=21)
x2.f &lt;- toTriangular(c(AA=28, AB=30, BB=8))
## Not run: 
out &lt;- HWPerm.mult(x2.m,x2.f)

## End(Not run)

#
# autosomal k alleles not accounting for gender
#

x3 &lt;- c(AA=12,AB=19,AC=13,BB=7,BC=5,CC=0)
x3 &lt;- toTriangular(x3)
## Not run: 
out &lt;- HWPerm.mult(x3)

## End(Not run)

#
# X-chromosomal k alleles 
#

x4.m &lt;- c(A=15,B=17,C=24) 
x4.f &lt;- toTriangular(c(AA=4,AB=2,AC=13,BB=6,BC=19,CC=4))
## Not run: 
out &lt;- HWPerm.mult(x4.m,x4.f)

## End(Not run)

#
# Autosomal k alleles accounting for gender
#

x5.m &lt;- toTriangular(c(AA=12,AB=19,AC=13,BB=7,BC=5,CC=0))
x5.f &lt;- toTriangular(c(AA=8,AB=12,AC=13,BB=8,BC=7,CC=0))
## Not run: 
out &lt;- HWPerm.mult(x5.m,x5.f)

## End(Not run)

#
# Autosomal STR with multipe alleles
#

data(NistSTRs)
A1 &lt;- NistSTRs[,1]
A2 &lt;- NistSTRs[,2]
GenotypeCounts &lt;- AllelesToTriangular(A1,A2)
print(GenotypeCounts)
## Not run: 
out &lt;- HWPerm.mult(GenotypeCounts)

## End(Not run)
</code></pre>

<hr>
<h2 id='HWPosterior'>
Calculation of posterior probabilities and Bayes factors for
Hardy-Weinberg tests at X-chromosomal variants.
</h2><span id='topic+HWPosterior'></span>

<h3>Description</h3>

<p>Function <code>HWPosterior</code> calculates posterior probabilities and Bayes
factors for tests for Hardy-Weinberg equilibrium of autosomal and X-chromosomal
variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWPosterior(males, females, verbose = TRUE, prior.af = c(0.5,0.5), prior.gf =
            c(0.333,0.333,0.333), x.linked = FALSE, precision = 0.05) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWPosterior_+3A_males">males</code></td>
<td>
<p>A vector of male genotype counts. For autosomal variants, this
should be a three-element named vector like (AA,AB,BB); for X-chromosomal
variants it should be a two-element vector giving the counts of the hemizygous
genotypes like (A,B).</p>
</td></tr>
<tr><td><code id="HWPosterior_+3A_females">females</code></td>
<td>
<p>A vector of female genotype counts. This should be a three-element
named vector like (AA,AB,BB)</p>
</td></tr> 
<tr><td><code id="HWPosterior_+3A_verbose">verbose</code></td>
<td>
<p>prints results if <code>verbose = TRUE</code></p>
</td></tr>
<tr><td><code id="HWPosterior_+3A_prior.af">prior.af</code></td>
<td>
<p>Beta prior parameters for male and female allele frequencies</p>
</td></tr>
<tr><td><code id="HWPosterior_+3A_prior.gf">prior.gf</code></td>
<td>
<p>Dirichlet prior parameters for female genotype
frequencies</p>
</td></tr>
<tr><td><code id="HWPosterior_+3A_x.linked">x.linked</code></td>
<td>
<p>logical indicating whether the variant is autosomal or
X-chromosomal</p>
</td></tr>
<tr><td><code id="HWPosterior_+3A_precision">precision</code></td>
<td>
<p>precision parameter for marginal likelihoods that
require numeric integration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For X-chromosomal variants, four possible models are considered, and the posterior
probabilities and Bayes factors for each model are calculated.
</p>
<p>For autosomal variants, ten possible scenarios are considered, and the
posterior probabilities for all models are calculated.
</p>
<p>In general, default Dirichlet priors are used for genotype
frequencies, and beta prior are used for allele frequencies.
</p>


<h3>Value</h3>

<p>For X-chromosomal variants, a matrix with posterior probabilities and
Bayes factors will be produced.
For autosomal variants, a vector of posterior probabilities is produced.
</p>


<h3>Author(s)</h3>

<p>Xavi Puig <a href="mailto:xavier.puig@upc.edu">xavier.puig@upc.edu</a> and
Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a>
</p>


<h3>References</h3>

<p>Puig, X., Ginebra, J. and Graffelman, J. (2017) A Bayesian test for Hardy-Weinberg
equilibrium of bi-allelic X-chromosomal markers. Heredity 119(4):226&ndash;236.
<a href="https://doi.org/10.1038/hdy.2017.30">doi:10.1038/hdy.2017.30</a>.
</p>
<p>Puig, X., Ginebra, J. and Graffelman, J. (2019) Bayesian model selection for the study
of Hardy-Weinberg proportions and homogeneity of gender allele frequencies.
Heredity  123(5), pp. 549-564. <a href="https://doi.org/10.1038/s41437-019-0232-0">doi:10.1038/s41437-019-0232-0</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWChisq">HWChisq</a></code>, <code><a href="#topic+HWExact">HWExact</a></code>, <code><a href="#topic+HWExactStats">HWExactStats</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# An X-chromosomal example
#
males   &lt;- c(A=43,B=13)
females &lt;- c(AA=26,AB=19,BB=3)
out     &lt;- HWPosterior(males,females,verbose=TRUE,x.linked=TRUE)
#
# An autosomal example
#
data(JPTsnps)
males     &lt;- JPTsnps[1,1:3]
females   &lt;- JPTsnps[1,4:6]
post.prob &lt;- HWPosterior(males,females,x.linked=FALSE)
</code></pre>

<hr>
<h2 id='HWPower'>
Compute the power of a test for Hardy-Weinberg equilibrium.
</h2><span id='topic+HWPower'></span>

<h3>Description</h3>

<p><code>HWPower</code> is a function that computes the power of a test for
Hardy-Weinberg equilibrium.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWPower(n = 100, nA = 100, pA = 0.5, y = c(AA=25,AB=50,BB=25),
alpha = 0.05, theta = 4, f = NULL, test = "exact",
alternative = "two.sided", pvaluetype = "selome", cc = 0.5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWPower_+3A_n">n</code></td>
<td>
<p>The sample size</p>
</td></tr>
<tr><td><code id="HWPower_+3A_na">nA</code></td>
<td>
<p>The minor allele count</p>
</td></tr>
<tr><td><code id="HWPower_+3A_pa">pA</code></td>
<td>
<p>The minor allele frequency</p>
</td></tr>
<tr><td><code id="HWPower_+3A_y">y</code></td>
<td>
<p>A sample of genotype counts (AA,AB,BB)</p>
</td></tr>
<tr><td><code id="HWPower_+3A_alpha">alpha</code></td>
<td>
<p>The significance level (0.05 by default)</p>
</td></tr>
<tr><td><code id="HWPower_+3A_theta">theta</code></td>
<td>
<p>The degree of disequilibrium (<code>theta</code> = 4 is
equilibrium, <code>theta</code> &gt; 4 is heterozygote excess, <code>theta</code> &lt;
4 is heterozygote dearth)</p>
</td></tr>
<tr><td><code id="HWPower_+3A_f">f</code></td>
<td>
<p>The inbreeding coefficient. Overrules <code>theta</code> if specified.</p>
</td></tr>
<tr><td><code id="HWPower_+3A_test">test</code></td>
<td>
<p>The type of test for which power is to be computed. Can be
&quot;exact&quot; (default) or &quot;chisq&quot; (chi-square)</p>
</td></tr>
<tr><td><code id="HWPower_+3A_alternative">alternative</code></td>
<td>
<p>The nature of the alternative hypothesis
(&quot;two.sided&quot; (default), &quot;greater&quot; or &quot;less&quot;)</p>
</td></tr>
<tr><td><code id="HWPower_+3A_pvaluetype">pvaluetype</code></td>
<td>
<p>The type of p-value used in an exact test
(&quot;selome&quot;, &quot;dost&quot; or &quot;midp&quot;)</p>
</td></tr>
<tr><td><code id="HWPower_+3A_cc">cc</code></td>
<td>
<p>Continuity correction parameter for the chi-square test (0.5
by default)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>HWPower</code> uses the Levene-Haldane distribution (distribution of the
number of heterzygotes given the minor allele count) for computing
power.
</p>
<p><code>HWPower</code> can be used in three different way. In principle, the
power is calcuted on the basis of the sample size (<code>n</code>) and the minor
allele count (<code>nA</code>). Alternatively, the user may specifiy sample
size (<code>n</code>) and minor allele frequency (<code>pA</code>). Finally, power
can also be calculated directly from a sample of genotype counts. In
that case the calculated power is the power for a sample of the given
sample size and minor allele count. The three ways to use <code>HWPower</code>
are illustrated in the example section. 
</p>


<h3>Value</h3>

<p>if <code>test</code> = &quot;exact&quot; the power of the exact test is computed for
the given significance level and minor allele count.
</p>
<p>if <code>test</code> = &quot;chisq&quot; the power of the chi-square test is computed for
the given significance level and minor allele count.
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>References</h3>

<p>Graffelman, J. and Moreno, V. (2013) The Mid p-value in exact tests for
Hardy-Weinberg proportions. <em>Statistical Applications in Genetics
and Molecular Biology</em> 12(4): 433-448. 
</p>
<p>Graffelman, J. (2015) Exploring Diallelic Genetic Markers: The HardyWeinberg Package.
<em>Journal of Statistical Software</em> 64(3): 1-23. <a href="https://doi.org/10.18637/jss.v064.i03">doi:10.18637/jss.v064.i03</a>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWExact">HWExact</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>pw.chisq &lt;- HWPower(n=100,nA=100,alpha=0.05,test="chisq",theta=16)
print(pw.chisq)
pw.exact &lt;- HWPower(n=100,nA=100,alpha=0.05,test="exact",theta=16,pvaluetype="selome")
print(pw.exact)
pw.exact &lt;- HWPower(n=100,nA=100)
print(pw.exact)
pw.exact &lt;- HWPower(n=100,pA=0.5)
print(pw.exact)
pw.exact &lt;- HWPower(y=c(AA=25,AB=50,BB=25))
print(pw.exact)
</code></pre>

<hr>
<h2 id='HWQqplot'>
A Q-Q plot for Hardy-Weinberg equilibrium
</h2><span id='topic+HWQqplot'></span><span id='topic+HWQQplot'></span>

<h3>Description</h3>

<p><code>HWQqplot</code> creates a Q-Q plot for the p-values obtained in an Exact
test for Hardy-Weinberg equilibrium. Empirical p-values are plotted
against multiple simulated quantiles of the theoretical p-value distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWQqplot(X, nsim = 100, fit = "curve", logplot = FALSE,
main = "Q-Q plot for HWE", mm = NULL, pvaluetype = "selome", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWQqplot_+3A_x">X</code></td>
<td>
<p>Data matrix with genotype counts, one row for each sample, 3 columns</p>
</td></tr>
<tr><td><code id="HWQqplot_+3A_nsim">nsim</code></td>
<td>
<p>Number of samples drawn from the null distribution (100 by
default)</p>
</td></tr>
<tr><td><code id="HWQqplot_+3A_fit">fit</code></td>
<td>
<p>If <code>fit</code> is set to &quot;line&quot; straight lines will be
fitted to the simulated samples, if set to &quot;curve&quot;, ascending curves
will be shown.</p>
</td></tr>
<tr><td><code id="HWQqplot_+3A_logplot">logplot</code></td>
<td>
<p>If <code>logplot</code> is set to true, then the log10 of the
p-values will be used in the plot. If not, untransformed p-values will
be used.</p>
</td></tr>
<tr><td><code id="HWQqplot_+3A_main">main</code></td>
<td>
<p>Title for the plot</p>
</td></tr>
<tr><td><code id="HWQqplot_+3A_mm">mm</code></td>
<td>
<p>Maximal value for x and y axis in the plot</p>
</td></tr>
<tr><td><code id="HWQqplot_+3A_pvaluetype">pvaluetype</code></td>
<td>
<p>Type of p-value to be used in an exact test. Can be
&quot;selome&quot; (default), &quot;midp&quot; or &quot;dost&quot;.</p>
</td></tr>
<tr><td><code id="HWQqplot_+3A_...">...</code></td>
<td>
<p>Any additional arguments for the <code>plot</code> instruction</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>HWQqplot</code> constructs a Q-Q plot of the p-values of an exact test
for Hardy-Weinberg equilibrium. Under the null, this p-value is not
uniform. HWQqplot samples from the theoretical null distribution, taking
into account that markers may vary in allele frequency and in sample
size (due to missing values). For each simulated sample a grey curve or
line is shown. A green reference line with intercept 0 and slope 1 is
also shown in the plot.
</p>


<h3>Value</h3>

<p>NULL</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Rohlfs, R.V. and Weir, B.S. (2008) Distributions of Hardy-Weinberg
equilibrium test statistics. Genetics 180, pp. 1609-1616.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWTernaryPlot">HWTernaryPlot</a></code>, <code><a href="#topic+HWExact">HWExact</a></code>, <code><a href="stats.html#topic+qqplot">qqplot</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed(1234)
n &lt;- 200 # sample size
m &lt;- 100 # number of markers
X &lt;- HWData(n,m)
HWQqplot(X,logplot=TRUE,pvaluetype="selome",main="Q-Q Plot for HWE")

## End(Not run)
</code></pre>

<hr>
<h2 id='HWStr'>
Testing a set of microsatellites (STRs) for Hardy-Weinberg equilibrium 
</h2><span id='topic+HWStr'></span>

<h3>Description</h3>

<p>Function <code>HWStr</code> is a wrapper function arount <code>HWPerm.mult</code> and <code>HWChisq</code> in order to test a set of STRs for Hardy-Weinberg equilibrium.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWStr(X, test = "permutation", verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWStr_+3A_x">X</code></td>
<td>
<p>A data matrix with STRs in columns, with there two alleles coded in successive columns.</p>
</td></tr>
<tr><td><code id="HWStr_+3A_test">test</code></td>
<td>
<p><code>permutation</code> for a permutation test, or <code>"chisq"</code> for a chisquare test</p>
</td></tr>
<tr><td><code id="HWStr_+3A_verbose">verbose</code></td>
<td>
<p>be silent if set to <code>FALSE</code> or print info for each STR if <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="HWStr_+3A_...">...</code></td>
<td>
<p>possible extra parameters to be passed on to <code>HWPerm.mult</code> such as <code>nperm</code> or <code>eps</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is recommended to test by a permutation test. By default, 17.000 permutations are used. Exact testing is not implemented.
Missing genotypes are excluded on a per STR basis.
</p>


<h3>Value</h3>

<p>A data frame with the 
</p>
<table role = "presentation">
<tr><td><code>strnames</code></td>
<td>
<p>STR name</p>
</td></tr>
<tr><td><code>N</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code>Nt</code></td>
<td>
<p>number of alleles</p>
</td></tr>
<tr><td><code>MinorAF</code></td>
<td>
<p>minor allele frequency</p>
</td></tr>
<tr><td><code>MajorAF</code></td>
<td>
<p>major allele frequency</p>
</td></tr>
<tr><td><code>Ho</code></td>
<td>
<p>observed heterozygosity</p>
</td></tr>
<tr><td><code>He</code></td>
<td>
<p>expected heterozygosity (without bias correction)</p>
</td></tr>
<tr><td><code>Hp</code></td>
<td>
<p>Shannon index of allele frequencies</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>p-value of the HWE test</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

<p><code><a href="#topic+HWPerm.mult">HWPerm.mult</a></code>, <code><a href="#topic+HWChisq">HWChisq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(NistSTRs)
## Not run: 
Results &lt;- HWStr(NistSTRs,test="permutation")

## End(Not run)
</code></pre>

<hr>
<h2 id='HWStrata'>
Asymptotic test for HWE across strata for a single biallelic marker 
</h2><span id='topic+HWStrata'></span>

<h3>Description</h3>

<p>Function <code>HWStrata</code> implements Olson's asymptotic test for HWE for a stratified sample of
single biallelic polymorphism.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWStrata(X, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWStrata_+3A_x">X</code></td>
<td>
<p>A three-column matrix of genotype counts (e.g. with columns AA, AB, BB)</p>
</td></tr>
<tr><td><code id="HWStrata_+3A_verbose">verbose</code></td>
<td>
<p>print output if <code>verbose = TRUE</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the references for the related homogeneity assumption.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>T2</code></td>
<td>
<p>The test statistic</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>The p-value</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a></p>


<h3>References</h3>

<p>Olson J.M. (1993) Testing the Hardy-Weinberg law across strata. Annals of Human Genetics 57(4):291-295. 
</p>
<p>Olson, J.M. and Foley, M. (1996) Testing for homogeneity of Hardy-Weinberg disequilibrium using data sampled from several
populations. Biometrics 52(3) pp. 971-979.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWExactStats">HWExactStats</a></code>, <code><a href="#topic+HWChisqStats">HWChisqStats</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Test across strata
#
data("Glyoxalase")
Glyoxalase &lt;- as.matrix(Glyoxalase)
HWStrata(Glyoxalase)
#
# Stratified exact testing, testing each sample
#
HWExactStats(Glyoxalase)
</code></pre>

<hr>
<h2 id='HWTernaryPlot'> Ternary plot with the Hardy-Weinberg acceptance region </h2><span id='topic+HWTernaryPlot'></span>

<h3>Description</h3>

<p><code>HWTernaryPlot</code> is a routine that draws a ternary plot for three-way genotypic compositions (AA,AB,BB), and represents
the acceptance region for different tests for Hardy-Weinberg equilibrium (HWE) in the plot. This allows for graphical
testing of a large set of markers (e.g. SNPs) for HWE. The (non) significance of the test
for HWE can be inferred from the position of the marker in the ternary plot. Different statistical tests for HWE
can be done graphically with this routine: the ordinary chisquare test, the chisquare test with continuity
correction and the Haldane's exact test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWTernaryPlot(X, n = NA, addmarkers = TRUE, newframe = TRUE, hwcurve = TRUE, 
vbounds = FALSE, mafbounds = FALSE, mafvalue = 0.05, axis = 0, region = 1, 
vertexlab = colnames(X), alpha = 0.05, vertex.cex = 1, pch = 19, cc = 0.5, 
markercol = "black", markerbgcol = "black", cex = 0.75, axislab = "", 
verbose = FALSE, markerlab = NULL, markerpos = NULL, mcex = 1, connect =
FALSE, curvecols = rep("black",5), signifcolour = TRUE, patternsigsymbol = 19,
curtyp = "solid", ssf = "max", pvaluetype = "selome", grid = FALSE,
gridlabels = TRUE, patternramp = FALSE, axisticklabels = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWTernaryPlot_+3A_x">X</code></td>
<td>
<p> a matrix of <code>n</code> genotypic compositions or counts. If it
is a matrix of compositions, X should have (<code>n</code> rows that sum
1, and 3 columns, with the relative frequencies of AA, AB and BB
respectively. Argument <code>n</code> should be supplied as well. If X is
a matrix of raw genotypic counts, it should have 3 columns with the
absolute counts of AA, AB and BB respectively. Argument <code>n</code> may
be supplied and will be used for painting acceptance regions. If not
supplied <code>n</code> is computed from the data in <code>X</code>. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_n">n</code></td>
<td>
<p> the samples size (for a complete composition with no missing data). </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_addmarkers">addmarkers</code></td>
<td>
<p> represent markers by dots in the triangle (<code>addmarkers=TRUE</code>) or not <br /> 
(<code>addmarkers=FALSE</code>). </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_newframe">newframe</code></td>
<td>
<p> allows for plotting additional markers in an already existing ternary plot. Overplotting
is achieved by setting <code>newframe</code> to <code>FALSE</code>. Setting <code>newframe = TRUE</code> (default) will 
create a new ternary plot. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_hwcurve">hwcurve</code></td>
<td>
<p> draw the HW parabola in the plot (<code>hwcurve=TRUE)</code> or not (<code>hwcurve=FALSE</code>). </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_vbounds">vbounds</code></td>
<td>
<p> indicate the area corresponding to expected counts &gt; 5 (<code>vbounds=TRUE</code>) or not 
(<code>vbounds=FALSE</code>).</p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_mafbounds">mafbounds</code></td>
<td>
<p> indicate the area corresponding to MAF &lt; <code>mafvalue</code>. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_mafvalue">mafvalue</code></td>
<td>
<p> a critical value for the minor allele frequency (MAF).</p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_axis">axis</code></td>
<td>
<p> draw a vertex axis <br />
0 = no axis is drawn <br />
1 = draw the AA axis <br />
2 = draw the AB axis <br />
3 = draw the BB axis <br />
4 = draw tick on the A-B axis</p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_region">region</code></td>
<td>
<p> the type of acceptance region to be delimited in the triangle <br /> 
0 = no acceptance region is drawn <br />
1 = draw the acceptance region corresponding to a Chi-square test <br /> 
2 = draw the acceptance region corresponding to a Chi-square test with continuity correction <br />
3 = draw the acceptance region corresponding to a Chi-square test with continuity correction for D &gt; 0 <br />
4 = draw the acceptance region corresponding to a Chi-square test with continuity correction for D &lt; 0 <br /> 
5 = draw the acceptance regions for all preceding tests simultaneously <br /> 
6 = draw the acceptance region corresponding to a Chi-square test with continuity correction with the upper
limit for D &gt; 0 and the lower limit for D &lt; 0 <br />
7 = draw the acceptance region corresponding to a two-sided exact test <br />
</p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_vertexlab">vertexlab</code></td>
<td>
<p> labels for the three vertices of the triangle </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_alpha">alpha</code></td>
<td>
<p> significance level (0.05 by default) </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_vertex.cex">vertex.cex</code></td>
<td>
<p> character expansion factor for the labels of the vertices of the triangle. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_pch">pch</code></td>
<td>
<p> the plotting character used to represent the markers. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_cc">cc</code></td>
<td>
<p> value for the continuity correction parameter (0.5 by default). </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_markercol">markercol</code></td>
<td>
<p> vector with colours for the marker points in the triangle. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_markerbgcol">markerbgcol</code></td>
<td>
<p> vector with background colours for the marker points in the triangle. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_cex">cex</code></td>
<td>
<p> expansion factor for the marker points in the triangle. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_axislab">axislab</code></td>
<td>
<p> a label to be put under the horizontal axis. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_verbose">verbose</code></td>
<td>
<p> print information on the numerically found cut-points between curves of the acceptance region and 
the edges of the triangle. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_markerlab">markerlab</code></td>
<td>
<p> labels for the markers in the triangle. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_markerpos">markerpos</code></td>
<td>
<p> positions for the marker labels in the triangle
(1,2,3 or 4).</p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_mcex">mcex</code></td>
<td>
<p> character expansion factor for the labels of the markers in the ternary plot. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_connect">connect</code></td>
<td>
<p> connect the represented markers by a line in the ternary plot. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_curvecols">curvecols</code></td>
<td>
<p> a vector with four colour specifications for the different curves that can be used
to delimit the HW acceptance region. E.g. <code>curvecols=c("red",
        "green","blue","black","purple")</code> will paint
the Hardy-Weinberg curve red, the limits of the acceptance region for an ordinary chi-square test
for HWE green, the limits of the acceptance region for a chi-square test with continuity correction
when D &gt; 0 blue and the limits of the acceptance region for a chi-square test with continuity 
correction when D &lt; 0 black, and the limits of the exact acceptance region purple. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_signifcolour">signifcolour</code></td>
<td>
<p> colour the marker points automatically according to the result of a signifance test 
(green markers non-siginficant, red markers significant). 
<code>signifcolour</code> only takes effect if <code>region</code> is set to
1, 2 or 7.</p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_patternsigsymbol">patternsigsymbol</code></td>
<td>
<p> plotting character used to represent significant makers when <code>patternramp=TRUE</code></p>
</td></tr> 
<tr><td><code id="HWTernaryPlot_+3A_curtyp">curtyp</code></td>
<td>
<p> style of the drawn curves (<code>"dashed","solid","dotted",...</code>)</p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_ssf">ssf</code></td>
<td>
<p> sample size function (<code>"max","min","mean","median",...</code>). Indicates how the sample size for
drawing acceptance regions is determined from the matrix
of counts.</p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_pvaluetype">pvaluetype</code></td>
<td>
<p> method to compute p-values in an exact test
(<code>"dost"</code> or <code>"selome"</code>)</p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_grid">grid</code></td>
<td>
<p> draw a reference grid for genotype frequencies at (0.2,0.4,0.6,0.8)</p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_gridlabels">gridlabels</code></td>
<td>
<p> if <code>grid=TRUE</code> plot labels at the grid</p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_patternramp">patternramp</code></td>
<td>
<p> paint a colour ramp for patterns of genotype frequencies. This will ignore <code>signifcolour</code>, and
use colour to represent marker frequencies. The ramp is a green-red gradient for the relative frequency of the pattern. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_axisticklabels">axisticklabels</code></td>
<td>
<p> place numeric labels for allele frequencies on the A-B axis. </p>
</td></tr>
<tr><td><code id="HWTernaryPlot_+3A_...">...</code></td>
<td>
<p> other arguments passed on to the plot function (e.g. <code>main</code> for a main title). </p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>HWTernaryPlot</code> automatically colours significant markers in
red, and non-significant markers in green if <code>region</code> is set to
1, 2 or 7.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>minp</code></td>
<td>
<p>minimum allele frequency above which testing for HWE is appropriate (expected counts exceeding 5).</p>
</td></tr>
<tr><td><code>maxp</code></td>
<td>
<p>maximum allele frequency below which testing for HWE is appropriate.</p>
</td></tr>
<tr><td><code>inrange</code></td>
<td>
<p>number of markers in the appropriate range.</p>
</td></tr>
<tr><td><code>percinrange</code></td>
<td>
<p>percentage of markers in the appropriate.</p>
</td></tr>
<tr><td><code>nsignif</code></td>
<td>
<p>number of significant markers (only if <code>region</code> equals 1,2 or 7.)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Graffelman, J. and Morales, J. (2008) Graphical Tests for Hardy-Weinberg Equilibrium 
Based on the Ternary Plot. <em>Human Heredity</em> 65(2):77-84. 
</p>
<p>Graffelman, J. (2015) Exploring Diallelic Genetic Markers: The HardyWeinberg Package.
<em>Journal of Statistical Software</em> 64(3): 1-23. <a href="https://doi.org/10.18637/jss.v064.i03">doi:10.18637/jss.v064.i03</a>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+HWChisq">HWChisq</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Ternary plot with 1000 SNPs and HWE curve
#

X &lt;- HWData(nm=1000,n=100)
HWTernaryPlot(X,100,region=0,vertex.cex=2,pch=1)

#
# Genotype frequency pattern of simulated SNPs with uniform
# allele frequency distribution
#

X &lt;- HWData(nm=1000,n=100)
HWTernaryPlot(X,patternramp = TRUE)

#
# Genotype frequency pattern of simulated SNPs with skewed 
# allele frequency distribution
#

X &lt;- HWData(nm=1000,n=100,shape1=1,shape2=20)
HWTernaryPlot(X,patternramp = TRUE)

#
# Genotype frequency pattern of SNPs simulated under HWE with
# allele frequency of 0.50
#

X &lt;- HWData(nm=1000,n=100,p=0.25)
HWTernaryPlot(X,patternramp = TRUE)

#
# Genotype frequency pattern of SNPs simulated under HWE with
# allele frequency of 0.25
#

X &lt;- HWData(nm=1000,n=100,p=0.25)
HWTernaryPlot(X,patternramp = TRUE)

#
# Genotype frequency pattern of SNPs simulated under HWE with
# allele frequency of 0.25, using a triangle to mark siginficant markers
#

X &lt;- HWData(nm=1000,n=100,p=0.25)
HWTernaryPlot(X,patternramp = TRUE, region=1, patternsigsymbol = 2)

#
# Ternary plot of 1000 SNPs simulated under HWE and uniform allele frequency, with
# acceptance region for a chi-square test and significant markers in red.
#

X &lt;- HWData(nm=1000,n=100)
HWTernaryPlot(X)

#
# Ternary plot of 100 SNPs simulated under HWE and uniform allele frequency, with
# acceptance region for an exact test and significant markers in red.
#

X &lt;- HWData(nm=100,n=100)
## Not run: HWTernaryPlot(X,region=7)
</code></pre>

<hr>
<h2 id='HWTriExact'>Exact test for Hardy-Weinberg equilibrium and equality of allele
frequencies for tri-allelic variants.
</h2><span id='topic+HWTriExact'></span>

<h3>Description</h3>

<p>Function <code>HWTriExact</code> does a standard exact test for Hardy-Weinberg
equilibrium of a tri-allelic variant, and also does joint exact tests
for equilibrium and equality of allele frequencies if the genotype
counts are given separately for both sexes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWTriExact(x, y = NULL, eps = 1e-10, nperm = 17000, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWTriExact_+3A_x">x</code></td>
<td>
<p>vector with 6 genotype counts (AA,AB,AC,BB,BC,CC)</p>
</td></tr>
<tr><td><code id="HWTriExact_+3A_y">y</code></td>
<td>
<p>vector with 6 or 3 genotype counts (AA,AB,AC,BB,BC,CC) or (A,B,C)</p>
</td></tr>
<tr><td><code id="HWTriExact_+3A_eps">eps</code></td>
<td>
<p>a tolerance that can be set for comparing exact probabilities</p>
</td></tr>
<tr><td><code id="HWTriExact_+3A_nperm">nperm</code></td>
<td>
<p>number of permutations (only relevant for autosomal
stratified by gender)</p>
</td></tr>
<tr><td><code id="HWTriExact_+3A_verbose">verbose</code></td>
<td>
<p>print output or not</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If only <code>x</code> is specified, an exact test for an autosomal variant
with three alleles will be performed.
</p>
<p>If both <code>x</code> and <code>y</code> are supplied as vectors with 6 elements,
a permutation test for HWE and equality of allele frequencies (EAF) for an
autosomal variant is performed, using <code>nperm</code> permutations. The
distribution of the probabilities is returned in <code>pseudodist</code>. The
computational cost of a completed enumeration algorithm can be
prohibitive in this case.
</p>
<p>If <code>x</code> is supplied as a length 6 vector, and <code>y</code> as a length 3
vector, the variant is assumed to be X-chromosomal, <code>x</code> containing
female genotype counts and <code>y</code> containing male genotyep counts. In
this case a joint exact test for HWE and EAF for an X-chromosomal
tri-allelic variant is executed.
</p>
<p>See the examples in the example section below.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pval</code></td>
<td>
<p>The p-value of the sample</p>
</td></tr>
<tr><td><code>pseudodist</code></td>
<td>
<p>Distribution of probabilities obtained by simulation</p>
</td></tr>
<tr><td><code>pofthesample</code></td>
<td>
<p>The probability of the observed sample</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2017) Multi-allelic exact tests for Hardy-Weinberg equilibrium that account for gender.
Molecular Ecology Resources. 18(3) pp. 461&ndash;473. <a href="https://doi.org/10.1111/1755-0998.12748">doi:10.1111/1755-0998.12748</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWPerm.mult">HWPerm.mult</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Autosomal tri-allelic (not accounting for gender)
#

x &lt;- c(AA=20,AB=31,AC=26,BB=15,BC=12,CC=0)
## Not run: out &lt;- HWTriExact(x)

#
# Autosomal tri-allelic accounting for gender
#

males   &lt;- c(A=1,B=21,C=34) 
females &lt;- c(AA=0,AB=1,AC=0,BB=8,BC=24,CC=15)
## Not run: out &lt;- HWTriExact(females,males)

#
# X-chromosomal tri-allelic accounting for gender
#

males   &lt;- c(A=1,B=21,C=34) 
females &lt;- c(AA=0,AB=1,AC=0,BB=8,BC=24,CC=15)
## Not run: out &lt;- HWTriExact(females,males)
</code></pre>

<hr>
<h2 id='ifisherz'>Inverse Fisher z transformation
</h2><span id='topic+ifisherz'></span>

<h3>Description</h3>

<p>Calculates the inverse of Fisher's z transformation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ifisherz(y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ifisherz_+3A_y">y</code></td>
<td>
<p>a real number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a correlation coefficient in the range (-1,1)
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+cor">cor</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>r &lt;- 0.5
print(ifisherz(fisherz(r)))
</code></pre>

<hr>
<h2 id='is.mono'>Detects autosomal and X-chromosomal monomorphic variants</h2><span id='topic+is.mono'></span>

<h3>Description</h3>

<p>Function <code>is.mono</code> tests if a bi-allelic variant is monomorphic or not</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.mono(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.mono_+3A_x">x</code></td>
<td>
<p>a vector of three or five genotype counts ((AA,AB,BB) or (A,B,AA,AB,BB)), or a
three-column or five-column matrix with genotype counts (variants in rows, columns)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>is.mono</code> assumes autosomal variants are coded in three-element vectors or
three-column matrices, whereas X-chromosomal variants are coded in five-element
vectors or five-column matrices.
</p>


<h3>Value</h3>

<p>A logical or vector of logicals</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# a polymorphic autosomal marker
#
x &lt;- c(AA=10,AB=20,BB=10)
print(is.mono(x))
#
# a monomorphic autosomal marker
#
x &lt;- c(AT=0,AA=100,TT=0)
print(is.mono(x))
#
# an autosomal marker with only heterozygotes
#
x &lt;- c(AT=100,AA=0,TT=0)
print(is.mono(x))
#
# a matrix with low maf autosomal markers
#
set.seed(123)
X &lt;- HWData(100,50,shape1=1,shape2=20)
number.monomorphics &lt;- sum(is.mono(X))
print(number.monomorphics)
#
# a polymorphic X chromosomal marker
#
x &lt;- c(G=24,C=26,GG=12,CC=13,GC=25)
print(is.mono(x))
#
# another polymorphic X chromosomal marker
#
x &lt;- c(G=24,C=1,GG=25,CC=0,GC=0)
is.mono(x)
#
# a monomorphic X chromosomal marker
#
x &lt;- c(G=24,C=0,GG=12,CC=0,GC=0)
is.mono(x)
#
# a matrix with low maf X-chromosomal markers
#
set.seed(123)
Y &lt;- HWData(100,50,shape=1,shape2=20,x.linked = TRUE)
number.monomorphics &lt;- sum(is.mono(Y))
print(number.monomorphics)
</code></pre>

<hr>
<h2 id='JPTmultiallelicsChr7'>
Multi-allelic autosomal variants of the Japanese population of the
1000 genomes project
</h2><span id='topic+JPTmultiallelicsChr7'></span>

<h3>Description</h3>

<p><code>JPTtriallelicsChrX</code> contains three selected multi-allelic variants
on chromosome 7 from the Japanese sample of the 1000 genomes project.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("JPTmultiallelicsChr7")</code></pre>


<h3>Format</h3>

<p>List object with fields m4,f4; m5,f5; m6,f6;</p>


<h3>Details</h3>

<p>The list object contains male and female genotype counts for 3
multi-allelic variants on chromosome 7 of the JPT sample of the 1000 genomes project.
</p>


<h3>Source</h3>

<p>The <a href="https://www.internationalgenome.org">The 1000 genomes project</a>.
</p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2017) Multi-allelic exact tests for Hardy-Weinberg equilibrium that
account for gender. doi: 10.1101/172874. Table 10.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(JPTmultiallelicsChr7)
str(JPTmultiallelicsChr7)
</code></pre>

<hr>
<h2 id='JPTmultiallelicsChrX'>
Multi-allelic X-chromosomal variants of the Japanese population of the
1000 genomes project
</h2><span id='topic+JPTmultiallelicsChrX'></span>

<h3>Description</h3>

<p><code>JPTtriallelicsChrX</code> contains four selected multi-allelic variants on the
X chromosome from the Japanese sample of the 1000 genomes project.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("JPTmultiallelicsChrX")</code></pre>


<h3>Format</h3>

<p>List object with fields m4,f4; m5,f5; m6,f6; m7,f7</p>


<h3>Details</h3>

<p>The list object contains male and female genotype counts for four
multi-allelic variants of the JPT sample of the 1000 genomes project.
</p>


<h3>Source</h3>

<p>The <a href="https://www.internationalgenome.org">The 1000 genomes project</a>.
</p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2017) Multi-allelic exact tests for Hardy-Weinberg equilibrium that
account for gender. doi: 10.1101/172874. Table 7.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(JPTmultiallelicsChrX)
m4 &lt;- JPTmultiallelicsChrX$m4
f4 &lt;- JPTmultiallelicsChrX$f4
</code></pre>

<hr>
<h2 id='JPTsnps'>
Bi-allelic SNPs from a Japanese population stratified by gender
</h2><span id='topic+JPTsnps'></span>

<h3>Description</h3>

<p><code>JPTsnps</code> contains genotype counts for the two sexes of ten single
nucleotide polymorphisms of the Japanese (JPT) sample of the 1000
Genomes project, consisting of 56 males and 48 females.
</p>
<p>The first three columns contain the male genotype counts, and the last
three columns contain the female genotype counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("JPTsnps")</code></pre>


<h3>Format</h3>

<p>data frame</p>


<h3>Source</h3>

<p>The <a href="https://www.internationalgenome.org">The 1000 genomes project</a>.
</p>


<h3>References</h3>

<p>Puig, X., Ginebra, J. and Graffelman, J. (2019) Bayesian model selection
for the study of Hardy-Weinberg proportions and homogeneity of gender
allele frequencies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(JPTsnps)
</code></pre>

<hr>
<h2 id='JPTtriallelicsChr7'>
Tri-allelic variants on chromosome 7 of the Japanese (JPT) sample of the 1000 genomes project 
</h2><span id='topic+JPTtriallelicsChr7'></span>

<h3>Description</h3>

<p><code>JPTtriallelics</code> contains six selected tri-allelic variants on 
chromosome 7 from the Japanese sample of the 1000 genomes project.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("JPTtriallelicsChr7")</code></pre>


<h3>Format</h3>

<p>A data frame with 6 observations on the following 14 variables.
</p>

<dl>
<dt><code>id</code></dt><dd><p>RS identifier</p>
</dd>
<dt><code>pos</code></dt><dd><p>position in base pairs</p>
</dd>
<dt><code>mAA</code></dt><dd><p>number of AA males</p>
</dd>
<dt><code>mAB</code></dt><dd><p>number of AB males</p>
</dd>
<dt><code>mAC</code></dt><dd><p>number of AC males</p>
</dd>
<dt><code>mBB</code></dt><dd><p>number of BB males</p>
</dd>
<dt><code>mBC</code></dt><dd><p>number of BC males</p>
</dd>
<dt><code>mCC</code></dt><dd><p>number of CC males</p>
</dd>
<dt><code>fAA</code></dt><dd><p>number of AA females</p>
</dd>
<dt><code>fAB</code></dt><dd><p>number of AB females</p>
</dd>
<dt><code>fAC</code></dt><dd><p>number of AC females</p>
</dd>
<dt><code>fBB</code></dt><dd><p>number of BB females</p>
</dd>
<dt><code>fBC</code></dt><dd><p>number of BC females</p>
</dd>
<dt><code>fCC</code></dt><dd><p>number of CC females</p>
</dd>
</dl>



<h3>Source</h3>

<p>The <a href="https://www.internationalgenome.org">The 1000 genomes project</a>.
</p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2017) Multi-allelic exact tests for Hardy-Weinberg equilibrium that
account for gender. doi: 10.1101/172874. Table 9.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(JPTtriallelicsChr7)
str(JPTtriallelicsChr7)
</code></pre>

<hr>
<h2 id='JPTtriallelicsChrX'>
Tri-allelic variants on the X-chromosome of the Japanese (JPT) sample of the 1000 genomes project 
</h2><span id='topic+JPTtriallelicsChrX'></span>

<h3>Description</h3>

<p><code>JPTtriallelicsChrX</code> contains five selected tri-allelic variants on the
X chromosome from the Japanese sample of the 1000 genomes project.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("JPTtriallelicsChrX")</code></pre>


<h3>Format</h3>

<p>A data frame with 5 observations on the following 12 variables.
</p>

<dl>
<dt><code>id</code></dt><dd><p>Identifier of the polymorphism</p>
</dd>
<dt><code>pos</code></dt><dd><p>Position of the polymorphism in base pairs</p>
</dd>
<dt><code>chr</code></dt><dd><p>Chromosome</p>
</dd>
<dt><code>A</code></dt><dd><p>Number of males with A genotype</p>
</dd>
<dt><code>B</code></dt><dd><p>Number of males with B genotype</p>
</dd>
<dt><code>C</code></dt><dd><p>Number of males with C genotype</p>
</dd>
<dt><code>AA</code></dt><dd><p>Number of AA females</p>
</dd>
<dt><code>AB</code></dt><dd><p>Number of AB females</p>
</dd>
<dt><code>AC</code></dt><dd><p>Number of AC females</p>
</dd>
<dt><code>BB</code></dt><dd><p>Number of BB females</p>
</dd>
<dt><code>BC</code></dt><dd><p>Number of BC females</p>
</dd>
<dt><code>CC</code></dt><dd><p>Number of CC females</p>
</dd>
</dl>



<h3>Source</h3>

<p>The <a href="https://www.internationalgenome.org">The 1000 genomes project</a>.
</p>


<h3>References</h3>

<p>Graffelman, J. and Weir, B.S. (2017) Multi-allelic exact tests for Hardy-Weinberg equilibrium that
account for gender. doi: 10.1101/172874. Table 6.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(JPTtriallelicsChrX)
str(JPTtriallelicsChrX)
</code></pre>

<hr>
<h2 id='mac'>
Compute the minor allele count.
</h2><span id='topic+mac'></span>

<h3>Description</h3>

<p><code>mac</code> computes the smallest allele count for a given vector of
genotype counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mac(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mac_+3A_x">X</code></td>
<td>
<p>a vector or matrix with genotype counts (AA, AB, BB)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of the minor allele counts
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>See Also</h3>

<p><code><a href="#topic+maf">maf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   X &lt;- as.vector(rmultinom(1,100,c(0.5,0.4,0.1)))
   names(X) &lt;- c("AA","AB","BB")
   print(X)
   print(mac(X))
</code></pre>

<hr>
<h2 id='maf'>Function to compute minor allele frequencies</h2><span id='topic+maf'></span>

<h3>Description</h3>

<p>Function <code>maf</code> computes the minor allele frequency for
a matrix or vector of genotype counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maf(x, option = 1, verbose = FALSE) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="maf_+3A_x">x</code></td>
<td>
<p>a vector or matrix of with genotype counts</p>
</td></tr>
<tr><td><code id="maf_+3A_option">option</code></td>
<td>
<p>determines output that is returned. <code>option=1</code> calculates the minor allele frequency; <code>option=2</code>
returns all allele frequencies; <code>option=3</code> returns the allele counts.</p>
</td></tr>
<tr><td><code id="maf_+3A_verbose">verbose</code></td>
<td>
<p>be silent (<code>verbose=FALSE</code>) or not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of minor allele frequencies or minor allele counts.
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>See Also</h3>

<p><code><a href="#topic+mac">mac</a></code>, <code><a href="#topic+MakeCounts">MakeCounts</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# MAF of a single random marker
#
set.seed(123)
X &lt;- as.vector(rmultinom(1,100,c(0.5,0.4,0.1)))
names(X) &lt;- c("AA","AB","BB")
print(X)
print(maf(X))
#
# MAF of MN bloodgroup counts
#
x &lt;- c(MM=298,MN=489,NN=213)
maf(x)
#
# Both allele frqeuencies, ordered from minor to major
#
maf(x,2)
#
# allele counts of MN bloodgroup counts, order from minor to major
#
maf(x,3)
#
# MAF of single triallelic marker in triangular format
#
x &lt;- c(AA=20,AB=52,AC=34,BB=17,BC=51,CC=26)
print(x)
GT &lt;- toTriangular(x)
print(GT)
maf(GT)
#
# extract all allele frequencies
#
maf(GT,option=2)
#
# extract all allele counts
#
maf(GT,option=3)
#
# Calculate the MAF for 10 random SNPs under HWE
#
set.seed(123)
Z &lt;- HWData(nm=10)
print(Z)
#
# vector with minor allele frequencies, one per marker
#
maf(Z)
#
# Matrix with minor and major allele frequencies, one row per marker 
#
maf(Z,2)
#
# Matrix with minor and major allele count, one row per marker 
#
maf(Z,3)
</code></pre>

<hr>
<h2 id='MakeCounts'>Create genotype counts from bi-allelic marker data</h2><span id='topic+MakeCounts'></span>

<h3>Description</h3>

<p><code>MakeCounts</code> creates a matrix of genotype counts, with one row for
each bi-allelic marker, containing 4 columns with the counts AA, AB, BB
and NA (missings) respectively
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MakeCounts(X, alleles, pos1 = 1, pos2 = 3, coding = c(AA=0,AB=1,BB=2), sep = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MakeCounts_+3A_x">X</code></td>
<td>
<p>A matrix or dataframe with bi-allelic genotyping information,
markers in columns, individuals in rows</p>
</td></tr>
<tr><td><code id="MakeCounts_+3A_alleles">alleles</code></td>
<td>
<p>a vector of alleles for each marker
(e.g. c(&quot;A/T&quot;,&quot;A/G&quot;,...)). Only relevanit if <code>X</code> is a matrix
with text entries.</p>
</td></tr>
<tr><td><code id="MakeCounts_+3A_pos1">pos1</code></td>
<td>
<p>position of the first allele in the allele string (1 by default)</p>
</td></tr>
<tr><td><code id="MakeCounts_+3A_pos2">pos2</code></td>
<td>
<p>position of the second allele in the allele string (3 by
default)</p>
</td></tr>
<tr><td><code id="MakeCounts_+3A_coding">coding</code></td>
<td>
<p>indicates how homozygotes and heterozygote are coded as
numbers. Only relevant if <code>X</code> is a matrix with numeric
entries.</p>
</td></tr>
<tr><td><code id="MakeCounts_+3A_sep">sep</code></td>
<td>
<p>allele separator character for genotype data in text format (&quot;&quot; for AA; &quot;/&quot; for &quot;A/A&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>MakeCounts</code> is thought for bi-allelic marker data only. Missings
are should be coded by NA. It produces the right input for
<code>HWTernaryPlot</code>.
</p>
<p>Heterozygotes may be coded in the data as &quot;AB&quot; or &quot;BA&quot;. Both entries
will be counted as a heterozygote. 
</p>


<h3>Value</h3>

<p>A matrix of 4 columns</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

<p><code><a href="#topic+HWTernaryPlot">HWTernaryPlot</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>SNP1 &lt;- c("GG","GG","GG","GG","GG","GG","GG","GG","GG")
SNP2 &lt;- c("CG","GG","CC","GG","GG","CG","CG","CG","CG")
SNP3 &lt;- c("AA","AA","AA","AG","AA","AG","AA","AA","AA")
SNP4 &lt;- c("GG","GG","GG","GG","GG","GG","GG","GG","GG")
SNP5 &lt;- c("CC","CC","CC","CC","CC","CC","CT","CT","CT")
X &lt;- cbind(SNP1,SNP2,SNP3,SNP4,SNP5)
Y &lt;- MakeCounts(X,c("A/G","C/G","A/G","A/G","C/T"))
print(Y)
W &lt;- matrix(sample(c(0,1,2,NA),100,replace=TRUE),ncol=5)
Z &lt;- MakeCounts(W,coding=c(0,1,2))
</code></pre>

<hr>
<h2 id='MakeFactor'>Make factors from genotyping data</h2><span id='topic+MakeFactor'></span>

<h3>Description</h3>

<p><code>MakeFactor</code> converts bi-allelic genetic marker data, whether coded numerically
as (0,1,2) or as (GG,GT,TT), etc. into standard factors coded as AA,
AB, BB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MakeFactor(x, coding = c(0, 1, 2))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MakeFactor_+3A_x">x</code></td>
<td>
<p>A vector containing genotyping results</p>
</td></tr>
<tr><td><code id="MakeFactor_+3A_coding">coding</code></td>
<td>
<p>Describes the numerical coding of the genotype data in
order AA, AB and BB. Only relevant if <code>x</code> is numerical</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>x</code> is a factor, it will be coerced to a factor with
levels AA, AB and BB. Important detail: the produced factors will have
only those levels that are observed in the data. E.g., if genotyping
results only consist of (0,1), then the resulting factor will not have
level BB (which would be an empty category)
</p>


<h3>Value</h3>

<p>A factor variable</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

<p><code><a href="#topic+MakeCounts">MakeCounts</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>y &lt;- c(1,1,0,0,2,2)
data.frame(y,MakeFactor(y))

y &lt;- c(2,2,3,3,1,1)
data.frame(y,MakeFactor(y,coding=c(1,2,3)))

data(Markers)
data.frame(Markers[,1],MakeFactor(Markers[,1],coding=c(1,2,3)))
</code></pre>

<hr>
<h2 id='Markers'>SNP data and intensities</h2><span id='topic+Markers'></span>

<h3>Description</h3>

<p>The dataframe contains the genotypes of 3 SNPs and two allele
intensities of 146 individuals. The first column is a GT polymorphism
that has missing values for several individuals. The second and third
column (iG and iG) are the allele intensities of this
polymorphism. Column 4 and 5 are covariate SNPs (an AC and an AG
polymorphism) that have no missing values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Markers)</code></pre>


<h3>Format</h3>

<p>A data frame containing 146 rows and 5 columns</p>


<h3>References</h3>

<p>Graffelman, J. (2015) Exploring Diallelic Genetic Markers: The
HardyWeinberg Package. Journal of Statistical Software, 64(3),
1-23. <a href="https://doi.org/10.18637/jss.v064.i03">doi:10.18637/jss.v064.i03</a>.
</p>

<hr>
<h2 id='Mourant'>Genotype frequencies for blood group locus MN</h2><span id='topic+Mourant'></span>

<h3>Description</h3>

<p>The dataframe contains the genotype frequencies MM, MN and NN for the
MN blood group locus for 216 populations. The data are taken from table 2.5 in Mourant et
al., using only entries with a sample size of at least 500.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Mourant)</code></pre>


<h3>Format</h3>

<p>A data frame containing 216 observations.</p>


<h3>Source</h3>

<p>Mourant et al, Table 2.5</p>


<h3>References</h3>

<p>Mourant, A. E. and Kope\'c, A. C. and Domaniewska-Sobczak, K. (1976) The
Distribution of the Human Blood Groups and other Polymorphisms. Second edition.
Oxford University Press, London.
</p>

<hr>
<h2 id='n.alleles'>
Number of alleles 
</h2><span id='topic+n.alleles'></span>

<h3>Description</h3>

<p>Function <code>n.alleles</code> determines the number of alleles in a named
genotype vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n.alleles(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="n.alleles_+3A_x">x</code></td>
<td>
<p>A named genotype vector (e.g. c(AA=10,AB=20,BB=5))</p>
</td></tr>
<tr><td><code id="n.alleles_+3A_...">...</code></td>
<td>
<p>extra arguments that are passed on to <code>alleles</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>integer</p>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>See Also</h3>

<p><code><a href="#topic+alleles">alleles</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>   x &lt;- c(AA=25,AB=50,BB=25)
   k &lt;- n.alleles(x)
   print(k)
</code></pre>

<hr>
<h2 id='NistSTRs'>
NIST autosomal STR data 
</h2><span id='topic+NistSTRs'></span>

<h3>Description</h3>

<p><code>NistSTRs</code> contains 29 autosomal microsatellites (STRs) of individuals of Caucasian ancestry. The two alleles of an individual are separated into two columns for each STR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("NistSTRs")</code></pre>


<h3>Format</h3>

<p>A data frame with 361 observations on the following 58 variables.
</p>

<dl>
<dt>&lsquo;<span class="samp">&#8288;CSF1PO-1&#8288;</span>&rsquo;</dt><dd><p>First allele of CSF1PO</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;CSF1PO-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of CSF1PO</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D10S1248-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D10S1248</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D10S1248-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D10S1248</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D12S391-1&#8288;</span>&rsquo;</dt><dd><p>First  allele of D12S391</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D12S391-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D12S391</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D13S317-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D13S317</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D13S317-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D13S317</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D16S539-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D16S539</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D16S539-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D16S539</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D18S51-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D18S51</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D18S51-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D18S51</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D19S433-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D19S433</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D19S433-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D19S433</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D1S1656-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D1S1656</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D1S1656-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D1S1656</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D21S11-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D21S11</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D21S11-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D21S11</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D22S1045-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D22S1045</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D22S1045-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D22S1045</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D2S1338-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D2S1338</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D2S1338-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D2S1338</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D2S441-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D2S441</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D2S441-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D2S441</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D3S1358-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D3S1358</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D3S1358-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D3S1358</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D5S818-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D5S818</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D5S818-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D5S818</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D6S1043-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D6S1043</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D6S1043-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D6S1043</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D7S820-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D7S820</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D7S820-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D7S820</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D8S1179-1&#8288;</span>&rsquo;</dt><dd><p>First allele of D8S1179</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;D8S1179-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of D8S1179</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;F13A01-1&#8288;</span>&rsquo;</dt><dd><p>First allele of F13A01</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;F13A01-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of F13A01</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;F13B-1&#8288;</span>&rsquo;</dt><dd><p>First allele of F13B</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;F13B-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of F13B</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;FESFPS-1&#8288;</span>&rsquo;</dt><dd><p>First allele of FESFPS</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;FESFPS-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of FESFPS</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;FGA-1&#8288;</span>&rsquo;</dt><dd><p>First allele of FGA</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;FGA-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of FGA</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;LPL-1&#8288;</span>&rsquo;</dt><dd><p>First allele of LPL</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;LPL-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of LPL</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;Penta_C-1&#8288;</span>&rsquo;</dt><dd><p>First allele of Penta_C</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;Penta_C-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of Penta_C</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;Penta_D-1&#8288;</span>&rsquo;</dt><dd><p>First allele of Penta_D</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;Penta_D-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of Penta_D</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;Penta_E-1&#8288;</span>&rsquo;</dt><dd><p>First allele of Penta_E</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;Penta_E-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of Penta_E</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;SE33-1&#8288;</span>&rsquo;</dt><dd><p>First allele of SE33</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;SE33-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of SE33</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;TH01-1&#8288;</span>&rsquo;</dt><dd><p>First allele of TH01</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;TH01-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of TH01</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;TPOX-1&#8288;</span>&rsquo;</dt><dd><p>First allele of TPOX</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;TPOX-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of TPOX</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;vWA-1&#8288;</span>&rsquo;</dt><dd><p>First allele of vWA</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;vWA-2&#8288;</span>&rsquo;</dt><dd><p>Second allele of vWA</p>
</dd>
</dl>



<h3>Source</h3>

<p>http://strbase.nist.gov
</p>


<h3>References</h3>

<p>Hill, C. et al. (2013) U.S. population data for 29 autosomal STR loci. Forensic science international: Genetics 497 7(3):e82-3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(NistSTRs)
</code></pre>

<hr>
<h2 id='order.auto'>Reordering of autosomal genotype counts</h2><span id='topic+order.auto'></span>

<h3>Description</h3>

<p>Function <code>order.auto</code> reorders a named vector of three genotype
counts, such that the sequence (minor homozygote, heterozygote,
major heterzygote) is establised.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>order.auto(X)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="order.auto_+3A_x">X</code></td>
<td>
<p>a named vector of genotype counts (e.g. c(AA=25,AB=50,BB=25))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Graffelman, J. (2015) Exploring Diallelic Genetic Markers: The HardyWeinberg Package.
<em>Journal of Statistical Software</em> 64(3): 1-23. <a href="https://doi.org/10.18637/jss.v064.i03">doi:10.18637/jss.v064.i03</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+order.x">order.x</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(MN=489,MM=298,NN=213)
print(x)
y &lt;- order.auto(x)
print(y)
</code></pre>

<hr>
<h2 id='order.x'>Reordering of X-chromosomal genotype counts</h2><span id='topic+order.x'></span>

<h3>Description</h3>

<p>Function <code>order.x</code> reorders a named vector of three genotype
counts, such that the sequence (minor hemizygote, major hemizygote,
minor homozygote, heterozygote, major heterzygote) is establised.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>order.x(X)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="order.x_+3A_x">X</code></td>
<td>
<p>a named vector of genotype counts (e.g. c(A=10,B=10,AA=25,AB=50,BB=25))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>References</h3>

<p>Graffelman, J. (2015) Exploring Diallelic Genetic Markers: The HardyWeinberg Package.
<em>Journal of Statistical Software</em> 64(3): 1-23. <a href="https://doi.org/10.18637/jss.v064.i03">doi:10.18637/jss.v064.i03</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+order.auto">order.auto</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;-  c(A=392, B=212, AA=275, AB=296, BB=80)
print(x)
y &lt;- order.x(x)
print(y)
</code></pre>

<hr>
<h2 id='qqunif'>
Q-Q plot for a uniform distribution.
</h2><span id='topic+qqunif'></span>

<h3>Description</h3>

<p><code>qqunif</code> makes a Q-Q plot against a uniform distribution for the supplied data vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qqunif(x, logplot = FALSE, lbs = 1:length(x), texton = FALSE, 
          xylim = NULL, main = "Q-Q plot for a uniform distribution",
          plotline = 0, xlab = "Expected p-value", 
          ylab = "Observed p-value", colvec=rep("black",length(x)),
	  colline = "black", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qqunif_+3A_x">x</code></td>
<td>
<p>The data vector</p>
</td></tr>
<tr><td><code id="qqunif_+3A_logplot">logplot</code></td>
<td>
<p>If <code>logplot</code> is set to true, then the log10 of the
p-values will be used in the plot. If not, untransformed p-values will
be used.</p>
</td></tr>
<tr><td><code id="qqunif_+3A_lbs">lbs</code></td>
<td>
<p>A vector of labels for the points in the Q-Q plot</p>
</td></tr>
<tr><td><code id="qqunif_+3A_texton">texton</code></td>
<td>
<p>Logical indicating whether labels should be plotted or not</p>
</td></tr>
<tr><td><code id="qqunif_+3A_xylim">xylim</code></td>
<td>
<p>Shared upper limit x and y axis in the plot</p>
</td></tr>
<tr><td><code id="qqunif_+3A_main">main</code></td>
<td>
<p>Title for the plot</p>
</td></tr>
<tr><td><code id="qqunif_+3A_plotline">plotline</code></td>
<td>
<p>Setting <code>plotline=0</code> plots a (0,1) line; <code>plotline=1</code> plots a robust fit,</p>
</td></tr>
<tr><td><code id="qqunif_+3A_xlab">xlab</code></td>
<td>
<p>label for the x axis</p>
</td></tr>
<tr><td><code id="qqunif_+3A_ylab">ylab</code></td>
<td>
<p>label for the y axis</p>
</td></tr>
<tr><td><code id="qqunif_+3A_colvec">colvec</code></td>
<td>
<p>vector with colours for the points in the QQ plot</p>
</td></tr>
<tr><td><code id="qqunif_+3A_colline">colline</code></td>
<td>
<p>colour for the line in the plot</p>
</td></tr>
<tr><td><code id="qqunif_+3A_...">...</code></td>
<td>
<p>Any additional arguments for the <code>plot</code> instruction</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pvals</code></td>
<td>
<p>observed probabilities</p>
</td></tr>
<tr><td><code>epvals</code></td>
<td>
<p>expected probabilities</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+qqnorm">qqnorm</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>   x &lt;- runif(1000)
   z &lt;- qqunif(x)
</code></pre>

<hr>
<h2 id='recode'>Recode genotype information</h2><span id='topic+recode'></span>

<h3>Description</h3>

<p>function <code>recode</code> recodes bi-allelic genetic marker information
expressed as strings (e.g. &quot;AA&quot;, &quot;AB&quot;, &quot;BB&quot;) into numerical form.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recode(X, alleles, values = c(0, 1, 2), pos1 = 1, pos2 = 3, minor
= FALSE, verbose = FALSE)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="recode_+3A_x">X</code></td>
<td>
<p>A matrix or dataframe of bi-allelic markers, individuals in
rows, markers in columns</p>
</td></tr>
<tr><td><code id="recode_+3A_alleles">alleles</code></td>
<td>
<p>a vector with the alleles for each marker
(e.g. c(&quot;A/T&quot;, &quot;A/G&quot;, etc))</p>
</td></tr>
<tr><td><code id="recode_+3A_values">values</code></td>
<td>
<p>a vector of numerica values for AA, AB and BB, ((0,1,2)
by default).</p>
</td></tr>
<tr><td><code id="recode_+3A_pos1">pos1</code></td>
<td>
<p>position of the first allele in the allele string (1 by default).</p>
</td></tr>
<tr><td><code id="recode_+3A_pos2">pos2</code></td>
<td>
<p>position of the second allele in the allele string (3 by
default).</p>
</td></tr>
<tr><td><code id="recode_+3A_minor">minor</code></td>
<td>
<p>coding is according to the number of copies of the minor
allele. if <code>minor = TRUE</code>, the value of 2 reflects two copies
of the minor allele, and the value 0 reflects no copies of the minor
allele.</p>
</td></tr>
<tr><td><code id="recode_+3A_verbose">verbose</code></td>
<td>
<p>print progress on the conversion or not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>recode</code> is written for bi-allelic marker data
only. Heterozygotes may be coded both as AB or BA. By default, the
second allele specified (e.g. &quot;T&quot; in &quot;A/T&quot;) is counted in the
recoding, and homozygotes AA are coded as 0 and homozygotes TT as 2. 
</p>


<h3>Value</h3>

<p>A numerical matrix, individuals in rows, markers in columns</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

<p><code><a href="#topic+MakeCounts">MakeCounts</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>SNP1 &lt;- c("GG","GG","GG","GG","GG","GG","GG","GG","GG")
SNP2 &lt;- c("CG","GG","CC","GG","GG","CG","CG","CG","CG")
SNP3 &lt;- c("AA","AA","AA","AG","AA","AG","AA","AA","AA")
SNP4 &lt;- c("GG","GG","GG","GG","GG","GG","GG","GG","GG")
SNP5 &lt;- c("CC","CC","CC","CC","CC","CC","CT","CT","CT")
X &lt;- cbind(SNP1,SNP2,SNP3,SNP4,SNP5)
Y &lt;- recode(X,c("A/G","C/G","A/G","A/G","C/T"))
print(Y)
</code></pre>

<hr>
<h2 id='shannon'>
Shannon index
</h2><span id='topic+shannon'></span>

<h3>Description</h3>

<p>Function <code>shannon</code> calculates the Shannon index and its variance for a vector of counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shannon(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shannon_+3A_x">x</code></td>
<td>
<p>a vector of counts</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Hp</code></td>
<td>
<p>the sample Shannon index</p>
</td></tr>
<tr><td><code>VHp</code></td>
<td>
<p>the sample variance of the Shannon index</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Shannon index for allele frequencies of a biallelic MN blood group polymorphism
#
x &lt;- c(MM=298,MN=489,NN=213)
p &lt;- af(x)
shannon(c(p,1-p))$Hp
#
# Shannon index for the allele frequencies of an STR
#
data("NistSTRs")
AlleleTable &lt;- table(c(NistSTRs[,1],NistSTRs[,2]))
AlleleFreq  &lt;- AlleleTable/sum(AlleleTable)
shannon(AlleleFreq)$Hp
</code></pre>

<hr>
<h2 id='strsort'>Sort tokens of a set of strings</h2><span id='topic+strsort'></span>

<h3>Description</h3>

<p>Function <code>strsort</code> collapses all tokens of a vector of strings in a
single string with sorted tokens</p>


<h3>Usage</h3>

<pre><code class='language-R'>strsort(s)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="strsort_+3A_s">s</code></td>
<td>
<p>a vector of character strings</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a string</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

<p><code><a href="#topic+alleles">alleles</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c("AA","AB","BB","AC","CC")
print(strsort(x))
</code></pre>

<hr>
<h2 id='ThetatoF'>
Convert theta to an inbreeding coefficient
</h2><span id='topic+ThetatoF'></span>

<h3>Description</h3>

<p>Function <code>ThetatoF</code> converts disequilibrium measure theta to an
inbreeding coefficient.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ThetatoF(p, theta = 4)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ThetatoF_+3A_p">p</code></td>
<td>
<p>the allele frequency</p>
</td></tr>
<tr><td><code id="ThetatoF_+3A_theta">theta</code></td>
<td>
<p>the disequilibrium parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a real number
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman (jan.graffelman@upc.edu) </p>


<h3>References</h3>

<p>Rohlfs, R.V. and Weir, B.S. (2008) Distributions of
Hardy-Weinberg Equilibrium Test Statistics, Genetics, 180(3) pp. 1609-1616.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HWf">HWf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>f &lt;- ThetatoF(0.5,4)
</code></pre>

<hr>
<h2 id='toTriangular'>Convert a vector of genotype counts to triangular format</h2><span id='topic+toTriangular'></span>

<h3>Description</h3>

<p>Function <code>toTriangular</code> converts a named vector of genotype counts
into a triangular matrix format, with homozygotes on the diagonal and
heterozygotes below the diagonal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toTriangular(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="toTriangular_+3A_x">x</code></td>
<td>
<p>A vector of genotype counts</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(AA=20,AB=52,AC=34,BB=17,BC=51,CC=26)
print(x)
X &lt;- toTriangular(x)
print(X)
</code></pre>

<hr>
<h2 id='TSIXTriAllelics'>
Tri-allelic polymorphisms on the X chromosome of the TSI population
</h2><span id='topic+TSIXTriAllelics'></span>

<h3>Description</h3>

<p>This dataframe contains genotype counts for six three-allelic polymorphisms (A,B,C) on chromosome X of a sample of individuals from the TSI population (Tuscany, Italy) of the 1,000 genomes project.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(TSIXTriAllelics)</code></pre>


<h3>Format</h3>

<p>A data frame with 6 observations on the following 10 variables.
</p>

<dl>
<dt><code>ID</code></dt><dd><p>Identifier of the polymorphism</p>
</dd>
<dt><code>A</code></dt><dd><p>Male A genotype count</p>
</dd>
<dt><code>B</code></dt><dd><p>Male B genotype count</p>
</dd>
<dt><code>C</code></dt><dd><p>Male C genotype count</p>
</dd>
<dt><code>AA</code></dt><dd><p>Female AA genotype count</p>
</dd>
<dt><code>AB</code></dt><dd><p>Female AB genotype count</p>
</dd>
<dt><code>AC</code></dt><dd><p>Female AC genotype count</p>
</dd>
<dt><code>BB</code></dt><dd><p>Female BB genotype count</p>
</dd>
<dt><code>BC</code></dt><dd><p>Female BC genotype count</p>
</dd>
<dt><code>CC</code></dt><dd><p>Female CC genotype count</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data taken from the 1,000 genomes project at www.internationalgenome.org
</p>


<h3>References</h3>

<p>Graffelman, J. and Ortoleva, L. (2020) A network algorithm for the X chromosomal exact test for Hardy-Weinberg equilibrium with multiple alleles. Under review.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(TSIXTriAllelics)
</code></pre>

<hr>
<h2 id='UniqueGenotypeCounts'>Extract unique genotypic compositions from a matrix</h2><span id='topic+UniqueGenotypeCounts'></span>

<h3>Description</h3>

<p>Function <code>UniqueGenotypeCounts</code> creates a matrix containing
only the unique rows in the given matrix, together with their frequency
of occurrence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UniqueGenotypeCounts(X, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UniqueGenotypeCounts_+3A_x">X</code></td>
<td>
<p>A n by 3 matrix with genotypic counts (AA,AB,BB)</p>
</td></tr>
<tr><td><code id="UniqueGenotypeCounts_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE then print some statistics</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with 4 columns, AA, AB, BB, and frequency of occurrence
</p>


<h3>Author(s)</h3>

<p> Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a> </p>


<h3>See Also</h3>

 <p><code><a href="#topic+GenerateSamples">GenerateSamples</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
X &lt;- HWData(n=100,nm=100)
print(nrow(X))
Y &lt;- UniqueGenotypeCounts(X)
print(nrow(Y))
print(sum(Y$w))
</code></pre>

<hr>
<h2 id='vaf'>
Computes the sample variance of the allele frequency for a biallelic marker.
</h2><span id='topic+vaf'></span>

<h3>Description</h3>

<p>Function <code>vaf</code> computes the sample variance of the allele frequencies
of a single sample or a matrix of samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vaf(X, hw = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vaf_+3A_x">X</code></td>
<td>
<p>vector or matrix with genotype counts (AA,AB,BB)</p>
</td></tr>
<tr><td><code id="vaf_+3A_hw">hw</code></td>
<td>
<p>assume Hardy-Weinberg proportions (<code>hw=TRUE</code>) or not (<code>hw=FALSE</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For biallelic markers the variance of the minor allele frequency
equals the variance of the major allele frequency. 
</p>


<h3>Value</h3>

<p>a numeric vector of variances.
</p>


<h3>Author(s)</h3>

<p>Jan Graffelman <a href="mailto:jan.graffelman@upc.edu">jan.graffelman@upc.edu</a></p>


<h3>References</h3>

<p>Weir, B.S. (1996) Genetic data analysis II. Sinauer Associates, Massachusetts. See Chapter 2.</p>


<h3>See Also</h3>

<p><code><a href="#topic+af">af</a></code>, <code><a href="#topic+maf">maf</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(MM=298,MN=489,NN=213)
pA &lt;- af(x)
vA &lt;- vaf(x)
cat("allele frequency:",pA,"\n")
cat("sample variance allele frequency:",vA,"\n")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
