<!DOCTYPE html><html><head><title>Help for package harrietr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {harrietr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dist_long'><p>Return evolutionary distance in long format</p></a></li>
<li><a href='#get_node_support'><p>Get node support from a tree produced with IQTREE</p></a></li>
<li><a href='#harrietr'><p><code>harrietr</code> package</p></a></li>
<li><a href='#join_metadata'><p>Add metadata to long distance matrix</p></a></li>
<li><a href='#melt_dist'><p>Melt a square distance matrix into long format</p></a></li>
<li><a href='#woodmouse'><p>Woodmouse dataset</p></a></li>
<li><a href='#woodmouse_iqtree'><p>Woodmouse IQTREE newick tree</p></a></li>
<li><a href='#woodmouse_meta'><p>Woodmouse metadata</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Wrangle Phylogenetic Distance Matrices and Other Utilities</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Harriet was Charles Darwin's pet tortoise (possibly). 'harrietr'
    implements some function to manipulate distance matrices and phylogenetic trees
    to make it easier to plot with 'ggplot2' and to manipulate using 'tidyverse'
    tools.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> | file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/andersgs/harrietr">https://github.com/andersgs/harrietr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/andersgs/harrietr/issues">https://github.com/andersgs/harrietr/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape (&ge; 4.1), ggtree (&ge; 1.8.1), magrittr (&ge; 1.5), lazyeval
(&ge; 0.2.0), dplyr (&ge; 0.7.2), tidyr (&ge; 0.7.0), rlang (&ge;
0.1.2)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggplot2 (&ge; 2.2.1)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-12-01 04:23:23 UTC; andersgs</td>
</tr>
<tr>
<td>Author:</td>
<td>Anders Gonçalves da Silva [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Anders Gonçalves da Silva &lt;andersgs@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-12-01 04:30:18 UTC</td>
</tr>
</table>
<hr>
<h2 id='dist_long'>Return evolutionary distance in long format</h2><span id='topic+dist_long'></span>

<h3>Description</h3>

<p>This will take an alignment, will calculate the evolutionary distance between
all pairs of sequence, and will transform the distance matrix to long
format. It will remove upper triangle, and diagonal elements, so you
end with only (n)*(n-1)/2 rows, where n are the total number of rows in
the distance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_long(aln, order = NULL, dist = "N", tree = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist_long_+3A_aln">aln</code></td>
<td>
<p>An object of class matrix, it must be square</p>
</td></tr>
<tr><td><code id="dist_long_+3A_order">order</code></td>
<td>
<p>A character vector of size n with the order of the columns and rows (default: NULL)</p>
</td></tr>
<tr><td><code id="dist_long_+3A_dist">dist</code></td>
<td>
<p>A string naming the model to calculate distances (accepted values are those in ape::dist.dna)</p>
</td></tr>
<tr><td><code id="dist_long_+3A_tree">tree</code></td>
<td>
<p>An object of class phylo</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a tree is optionally given, a fourth column is returned with the cophenetic
distance across all elements of tree. It assumes the tree was generated from
the alignment.
</p>


<h3>Value</h3>

<p>A data.frame with three or four columns: (1) iso1; (2) iso2: (3) dist. If a tree is given then a fourth column (evol_dist) containig the distances from the tree is also supplied.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(woodmouse)
dist_df &lt;- dist_long(woodmouse)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_node_support'>Get node support from a tree produced with IQTREE</h2><span id='topic+get_node_support'></span>

<h3>Description</h3>

<p>In IQTREE it is possible to obtain node support values by SH aproximate
likelihood ratio tests (SH-aLRT), and ultrafast bootstraps (uBS). Often,
we do both, which IQTREE encodes as two numbers separated by a '/' as the
internal node label. This function will return a data.frame with the
number of the internal nodes, and the support values for each.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_node_support(tree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_node_support_+3A_tree">tree</code></td>
<td>
<p>An object of type phylo generated using IQTREE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with internal node information, plus two columns:
(1) SH-aLRT; and (2) uBS
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(woodmouse_iqtree)
get_node_support(woodmouse_iqtree)

## End(Not run)
</code></pre>

<hr>
<h2 id='harrietr'><code>harrietr</code> package</h2><span id='topic+harrietr'></span><span id='topic+harrietr-package'></span>

<h3>Description</h3>

<p>harrietr: Wrangle Phylogenetic Distance Matrices and Other Utilities
</p>


<h3>Details</h3>

<p>See the README on
<a href="https://cran.r-project.org/package=harrietr/README.html">CRAN</a>
or <a href="https://github.com/andersgs/harrietr#readme">GitHub</a>
</p>

<hr>
<h2 id='join_metadata'>Add metadata to long distance matrix</h2><span id='topic+join_metadata'></span>

<h3>Description</h3>

<p>This functions takes the output from <code>dist_long</code>, plus a
data.frame with metadata, and attaches it to the data.frame output from
<code>dist_long</code>. It uses a column in the metadata data.frame as a key to
join the two data.frames. So, it requires a column of data in the metadata
data.frame to have same ID labels as those in the pairwise comparison table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>join_metadata(dist, meta, isolate = "ISOLATES", group = "CLUSTER",
  remove_ind = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="join_metadata_+3A_dist">dist</code></td>
<td>
<p>A data.frame produced by dist_long function</p>
</td></tr>
<tr><td><code id="join_metadata_+3A_meta">meta</code></td>
<td>
<p>A data.frame with one column of IDs that match the IDs in <code>dist_long</code></p>
</td></tr>
<tr><td><code id="join_metadata_+3A_isolate">isolate</code></td>
<td>
<p>A character string with the name of the column in the meta data.frame with the ID data</p>
</td></tr>
<tr><td><code id="join_metadata_+3A_group">group</code></td>
<td>
<p>A character string with the name of column containing the grouping variable</p>
</td></tr>
<tr><td><code id="join_metadata_+3A_remove_ind">remove_ind</code></td>
<td>
<p>A boolean whether to remove all non-essential columns</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output from <code>dist_long</code> with an additional column containing
a factor, with levels composed of joining the categories in the <code>group</code>
colum of the metadata data.frame for each pairwise comparison. For example,
if one row has distance between samples id1 and id2, and in the grouping column
of the metadata id1 is identified as part of group 'A', and id2 is identified
as part of group 'B', then the output from that row will 'AB'. If they were
from the same group, say 'A', the output would be just 'A'. In this way
it is easy to identify pairs of isolates that are from the same group, and
pairs of isolates that are from different groups.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(woodmouse)
data(woodmouse_meta)
dist_df &lt;- dist_long(woodmouse)
join_metadata(dist_df, woodmouse_meta, isolate = 'SAMPLE_ID', group = 'CLUSTER', remove_ind = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='melt_dist'>Melt a square distance matrix into long format</h2><span id='topic+melt_dist'></span>

<h3>Description</h3>

<p>This will take a square distance matrix, and will transform in to long
format. It will remove upper triangle, and diagonal elements, so you
end with only (n)*(n-1)/2 rows, where n are the total number of rows in
the distance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>melt_dist(dist, order = NULL, dist_name = "dist")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="melt_dist_+3A_dist">dist</code></td>
<td>
<p>An object of class matrix, it must be square</p>
</td></tr>
<tr><td><code id="melt_dist_+3A_order">order</code></td>
<td>
<p>A character vector of size n with the order of the columns and rows (default: NULL)</p>
</td></tr>
<tr><td><code id="melt_dist_+3A_dist_name">dist_name</code></td>
<td>
<p>A string to name the distance column in the output (default: dist)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with three columns: (1) iso1; (2) iso2; (3) dist. iso1 and
iso2 indicate the pair being compared, and dist indicates the distance between
that pair.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(woodmouse)
dist &lt;- ape::dist(woodmouse, model = 'N', as.matrix = TRUE)
dist_df &lt;- melt_dist(dist)

## End(Not run)
</code></pre>

<hr>
<h2 id='woodmouse'>Woodmouse dataset</h2><span id='topic+woodmouse'></span>

<h3>Description</h3>

<p>Woodmouse dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>woodmouse
</code></pre>


<h3>Format</h3>

<p>An object of class <code>DNAbin</code> with 15 rows and 965 columns.</p>


<h3>Source</h3>

<p>&quot;ape&quot; package <a href="ape.html#topic+woodmouse">woodmouse</a>
</p>

<hr>
<h2 id='woodmouse_iqtree'>Woodmouse IQTREE newick tree</h2><span id='topic+woodmouse_iqtree'></span>

<h3>Description</h3>

<p>Generated a multiFASTA, and used IQTREE to generate a tree with the following
command:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>woodmouse_iqtree
</code></pre>


<h3>Format</h3>

<p>An object of class <code>phylo</code> of length 5.</p>


<h3>Details</h3>

<p><code>iqtree -s woodmouse.fasta -m TEST -nt 4 -bb 1000 -alrt 1000</code>
</p>
<p>The tree was loaded into 'R' using 'ape::read.tree', and saved to a dataset.
</p>


<h3>Source</h3>

<p>&quot;ape&quot; package <a href="ape.html#topic+woodmouse">woodmouse</a>
</p>

<hr>
<h2 id='woodmouse_meta'>Woodmouse metadata</h2><span id='topic+woodmouse_meta'></span>

<h3>Description</h3>

<p>A dummy metadata table generated to demonstrate the use of <code>add_metadata</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>woodmouse_meta
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 15 rows and 2 columns.</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
