<!DOCTYPE html><html lang="en"><head><title>Help for package HDANOVA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HDANOVA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HDANOVA'><p>HDANOVA</p></a></li>
<li><a href='#apca'><p>ANOVA Principal Component Analysis - APCA</p></a></li>
<li><a href='#asca'><p>Analysis of Variance Simultaneous Component Analysis - ASCA</p></a></li>
<li><a href='#asca_fit'><p>ASCA Fitting Workhorse Function</p></a></li>
<li><a href='#asca_plots'><p>ASCA Plot Methods</p></a></li>
<li><a href='#asca_results'><p>ASCA Result Methods</p></a></li>
<li><a href='#block.data.frame'><p>Block-wise indexable data.frame</p></a></li>
<li><a href='#caldana'><p>Arabidopsis thaliana growth experiment</p></a></li>
<li><a href='#candies'><p>Sensory assessment of candies.</p></a></li>
<li><a href='#dummycode'><p>Dummy-coding of a single vector</p></a></li>
<li><a href='#extended.model.frame'><p>Extracting the Extended Model Frame from a Formula or Fit</p></a></li>
<li><a href='#limmpca'><p>Linear Mixed Model PCA</p></a></li>
<li><a href='#model.frame.asca'><p>Model Frame and Model Matrix for ASCA-like Models</p></a></li>
<li><a href='#msca'><p>Multilevel Simultaneous Component Analysis - MSCA</p></a></li>
<li><a href='#pcanova'><p>Principal Components Analysis of Variance Simultaneous Component Analysis - PC-ANOVA</p></a></li>
<li><a href='#pcanova_plots'><p>PC-ANOVA Result Methods</p></a></li>
<li><a href='#pcanova_results'><p>PC-ANOVA Result Methods</p></a></li>
<li><a href='#permanova'><p>Permutation Based MANOVA - PERMANOVA</p></a></li>
<li><a href='#prc'><p>Principal Response Curves</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#timeplot'><p>Timeplot for Combined Effects</p></a></li>
<li><a href='#update_without_factor'><p>Update a Model without Factor</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>High-Dimensional Analysis of Variance</td>
</tr>
<tr>
<td>Version:</td>
<td>0.8.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-01-20</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions and datasets to support Smilde, Marini, Westerhuis and Liland (2025, ISBN: 978-1-394-21121-0) 
   "Analysis of Variance for High-Dimensional Data - Applications in Life, Food and Chemical Sciences". 
   This implements and imports a  collection of methods for HD-ANOVA data analysis with common interfaces, result- and plotting 
   functions, multiple real data sets and four vignettes covering a range different applications.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>car, lme4, mixlm (&ge; 1.4.2), pls, pracma, progress, RSpectra</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, vegan</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://khliland.github.io/HDANOVA/">https://khliland.github.io/HDANOVA/</a>,
<a href="https://github.com/khliland/HDANOVA/">https://github.com/khliland/HDANOVA/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/khliland/HDANOVA/issues/">https://github.com/khliland/HDANOVA/issues/</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-21 12:07:58 UTC; kristian</td>
</tr>
<tr>
<td>Author:</td>
<td>Kristian Hovde Liland
    <a href="https://orcid.org/0000-0001-6468-9423"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kristian Hovde Liland &lt;kristian.liland@nmbu.no&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-21 13:30:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='HDANOVA'>HDANOVA</h2><span id='topic+HDANOVA-package'></span><span id='topic+HDANOVA'></span>

<h3>Description</h3>

<p>Included methods:
</p>

<ul>
<li><p> ASCA+ - Analysis of Variance Simultaneous Component Analysis
</p>
</li>
<li><p> APCA+ - ANOVA Principal Component Analysis
</p>
</li>
<li><p> LiMM-PCA - Linear Mixed Model PCA
</p>
</li>
<li><p> MSCA - Multilevel Simultaneous Component Analysis
</p>
</li>
<li><p> PC-ANOVA - Principal Component Analysis of Variance
</p>
</li>
<li><p> PRC - Principal Response Curves
</p>
</li>
<li><p> PERMANOVA - Permutation Based MANOVA
</p>
</li></ul>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Kristian Hovde Liland <a href="mailto:kristian.liland@nmbu.no">kristian.liland@nmbu.no</a> (<a href="https://orcid.org/0000-0001-6468-9423">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>

<hr>
<h2 id='apca'>ANOVA Principal Component Analysis - APCA</h2><span id='topic+apca'></span>

<h3>Description</h3>

<p>APCA function for fitting ANOVA Principal Component Analysis models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apca(formula, data, add_error = TRUE, contrasts = "contr.sum", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="apca_+3A_formula">formula</code></td>
<td>
<p>Model formula accepting a single response (block) and predictors.</p>
</td></tr>
<tr><td><code id="apca_+3A_data">data</code></td>
<td>
<p>The data set to analyse.</p>
</td></tr>
<tr><td><code id="apca_+3A_add_error">add_error</code></td>
<td>
<p>Add error to LS means (default = TRUE).</p>
</td></tr>
<tr><td><code id="apca_+3A_contrasts">contrasts</code></td>
<td>
<p>Effect coding: &quot;sum&quot; (default = sum-coding), &quot;weighted&quot;, &quot;reference&quot;, &quot;treatment&quot;.</p>
</td></tr>
<tr><td><code id="apca_+3A_...">...</code></td>
<td>
<p>Additional parameters for the asca_fit function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>apca</code>, inheriting from the general <code>asca</code> class.
Further arguments and plots can be found in the <code><a href="#topic+asca">asca</a></code> documentation.
</p>


<h3>References</h3>

<p>Harrington, P.d.B., Vieira, N.E., Espinoza, J., Nien, J.K., Romero, R., and Yergey, A.L. (2005)
Analysis of variance–principal component analysis: A soft tool for proteomic discovery.
Analytica chimica acta, 544 (1-2), 118–127.
</p>


<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(candies)
ap &lt;- apca(assessment ~ candy, data=candies)
scoreplot(ap)

# Numeric effects
candies$num &lt;- eff &lt;- 1:165
mod &lt;- apca(assessment ~ candy + assessor + num, data=candies)
summary(mod)
scoreplot(mod, factor=3, gr.col=rgb(eff/max(eff), 1-eff/max(eff),0), pch.scores="x")
</code></pre>

<hr>
<h2 id='asca'>Analysis of Variance Simultaneous Component Analysis - ASCA</h2><span id='topic+asca'></span>

<h3>Description</h3>

<p>This is a quite general and flexible implementation of ASCA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>asca(formula, data, contrasts = "contr.sum", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="asca_+3A_formula">formula</code></td>
<td>
<p>Model formula accepting a single response (block) and predictors. See Details for more information.</p>
</td></tr>
<tr><td><code id="asca_+3A_data">data</code></td>
<td>
<p>The data set to analyse.</p>
</td></tr>
<tr><td><code id="asca_+3A_contrasts">contrasts</code></td>
<td>
<p>Effect coding: &quot;sum&quot; (default = sum-coding), &quot;weighted&quot;, &quot;reference&quot;, &quot;treatment&quot;.</p>
</td></tr>
<tr><td><code id="asca_+3A_...">...</code></td>
<td>
<p>Additional arguments to <code><a href="#topic+asca_fit">asca_fit</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ASCA is a method which decomposes a multivariate response according to one or more design
variables. ANOVA is used to split variation into contributions from factors, and PCA is performed
on the corresponding least squares estimates, i.e., <code>Y = X1 B1 + X2 B2 + ... + E = T1 P1' + T2 P2' + ... + E</code>.
This version of ASCA encompasses variants of LiMM-PCA, generalized ASCA and covariates ASCA. It includes
confidence ellipsoids for the balanced crossed-effect ASCA.
</p>
<p>The formula interface is extended with the function r() to indicate random
effects and comb() to indicate effects that should be combined. See Examples
for use cases.
</p>


<h3>Value</h3>

<p>An <code>asca</code> object containing loadings, scores, explained variances, etc. The object has
associated plotting (<code><a href="#topic+asca_plots">asca_plots</a></code>) and result (<code><a href="#topic+asca_results">asca_results</a></code>) functions.
</p>


<h3>References</h3>


<ul>
<li><p> Smilde, A., Jansen, J., Hoefsloot, H., Lamers,R., Van Der Greef, J., and Timmerman, M.(2005). ANOVA-Simultaneous Component Analysis (ASCA): A new tool for analyzing designed metabolomics data. Bioinformatics, 21(13), 3043–3048.
</p>
</li>
<li><p> Liland, K.H., Smilde, A., Marini, F., and Næs,T. (2018). Confidence ellipsoids for ASCA models based on multivariate regression theory. Journal of Chemometrics, 32(e2990), 1–13.
</p>
</li>
<li><p> Martin, M. and Govaerts, B. (2020). LiMM-PCA: Combining ASCA+ and linear mixed models to analyse high-dimensional designed data. Journal of Chemometrics, 34(6), e3232.
</p>
</li></ul>



<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load candies data
data(candies)

# Basic ASCA model with two factors
mod &lt;- asca(assessment ~ candy + assessor, data=candies)
print(mod)

# ASCA model with interaction
mod &lt;- asca(assessment ~ candy * assessor, data=candies)
print(mod)

# Result plotting for first factor
loadingplot(mod, scatter=TRUE, labels="names")
scoreplot(mod)
# No backprojection
scoreplot(mod, projections=FALSE)
# Spider plot
scoreplot(mod, spider=TRUE, projections=FALSE)

# ASCA model with compressed response using 5 principal components
mod.pca &lt;- asca(assessment ~ candy + assessor, data=candies, pca.in=5)

# Mixed Model ASCA, random assessor
mod.mix &lt;- asca(assessment ~ candy + r(assessor), data=candies)
scoreplot(mod.mix)

# Mixed Model ASCA, REML estimation
mod.mix &lt;- asca(assessment ~ candy + r(assessor), data=candies, REML=TRUE)
scoreplot(mod.mix)

# Load Caldana data
data(caldana)

# Combining effects in ASCA
mod.comb &lt;- asca(compounds ~ time + comb(light + time:light), data=caldana)
summary(mod.comb)
timeplot(mod.comb, factor="light", time="time", comb=2)

# Permutation testing
mod.perm &lt;- asca(assessment ~ candy * assessor, data=candies, permute=TRUE)
summary(mod.perm)

</code></pre>

<hr>
<h2 id='asca_fit'>ASCA Fitting Workhorse Function</h2><span id='topic+asca_fit'></span>

<h3>Description</h3>

<p>This function is called by all ASCA related methods in this package. It is documented
so that one can have access to a richer set of parameters from the various methods or call this
function directly. The latter should be done with care as there are many possibilities and not all
have been used in publications or tested thoroughly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>asca_fit(
  formula,
  data,
  subset,
  weights,
  na.action,
  family,
  permute = FALSE,
  perm.type = c("approximate", "exact"),
  unrestricted = FALSE,
  add_error = FALSE,
  aug_error = "denominator",
  use_ED = FALSE,
  pca.in = FALSE,
  contrasts = "contr.sum",
  coding,
  equal_baseline = FALSE,
  SStype = "II",
  REML = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="asca_fit_+3A_formula">formula</code></td>
<td>
<p>Model formula accepting a single response (block) and predictors. See Details for more information.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_data">data</code></td>
<td>
<p>The data set to analyse.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_subset">subset</code></td>
<td>
<p>Expression for subsetting the data before modelling.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_weights">weights</code></td>
<td>
<p>Optional object weights.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_na.action">na.action</code></td>
<td>
<p>How to handle NAs (no action implemented).</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_family">family</code></td>
<td>
<p>Error distributions and link function for Generalized Linear Models.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_permute">permute</code></td>
<td>
<p>Perform approximate permutation testing, default = FALSE (numeric or TRUE = 1000 permutations).</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_perm.type">perm.type</code></td>
<td>
<p>Type of permutation: &quot;approximate&quot; (default) or &quot;exact&quot;.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_unrestricted">unrestricted</code></td>
<td>
<p>Use unrestricted ANOVA decomposition (default = FALSE).</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_add_error">add_error</code></td>
<td>
<p>Add error to LS means, e.g., for APCA.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_aug_error">aug_error</code></td>
<td>
<p>Augment score matrices in backprojection. Default = &quot;denominator&quot;
(of F test), &quot;residual&quot; (force error term), nueric value (alpha-value in LiMM-PCA).</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_use_ed">use_ED</code></td>
<td>
<p>Use &quot;effective dimensions&quot; for score rescaling in LiMM-PCA.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_pca.in">pca.in</code></td>
<td>
<p>Compress response before ASCA (number of components).</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_contrasts">contrasts</code></td>
<td>
<p>Effect coding: &quot;sum&quot; (default = sum-coding), &quot;weighted&quot;, &quot;reference&quot;, &quot;treatment&quot;.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_coding">coding</code></td>
<td>
<p>Defunct. Use 'contrasts' instead.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_equal_baseline">equal_baseline</code></td>
<td>
<p>Experimental: Set to <code>TRUE</code> to let interactions, where a main effect is missing,
e.g., a nested model, be handled with the same baseline as a cross effect model. If <code>TRUE</code> the corresponding
interactions will be put in quotation marks and included in the <code>model.frame</code>.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_sstype">SStype</code></td>
<td>
<p>Type of sum-of-squares: &quot;I&quot; = sequential, &quot;II&quot; (default) = last term, obeying marginality,
&quot;III&quot; = last term, not obeying marginality.</p>
</td></tr>
<tr><td><code id="asca_fit_+3A_reml">REML</code></td>
<td>
<p>Parameter to mixlm: NULL (default) = sum-of-squares, TRUE = REML, FALSE = ML.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>asca</code> object containing loadings, scores, explained variances, etc. The object has
associated plotting (<code><a href="#topic+asca_plots">asca_plots</a></code>) and result (<code><a href="#topic+asca_results">asca_results</a></code>) functions.
</p>

<hr>
<h2 id='asca_plots'>ASCA Plot Methods</h2><span id='topic+asca_plots'></span><span id='topic+loadingplot.asca'></span><span id='topic+scoreplot.asca'></span><span id='topic+permutationplot'></span>

<h3>Description</h3>

<p>Various plotting procedures for <code><a href="#topic+asca">asca</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'asca'
loadingplot(object, factor = 1, comps = 1:2, ...)

## S3 method for class 'asca'
scoreplot(
  object,
  factor = 1,
  comps = 1:2,
  within_level = "all",
  pch.scores = 19,
  pch.projections = 1,
  gr.col = NULL,
  projections = TRUE,
  spider = FALSE,
  ellipsoids,
  confidence,
  xlim,
  ylim,
  xlab,
  ylab,
  legendpos,
  ...
)

permutationplot(object, factor = 1, xlim, xlab = "SSQ", main, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="asca_plots_+3A_object">object</code></td>
<td>
<p><code>asca</code> object.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_factor">factor</code></td>
<td>
<p><code>integer/character</code> for selecting a model factor. If factor &lt;= 0 or &quot;global&quot;,
the PCA of the input is used (negativ factor to include factor level colouring with global PCA).</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_comps">comps</code></td>
<td>
<p><code>integer</code> vector of selected components.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_...">...</code></td>
<td>
<p>additional arguments to underlying methods.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_within_level">within_level</code></td>
<td>
<p>MSCA parameter for chosing plot level (default = &quot;all&quot;).</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_pch.scores">pch.scores</code></td>
<td>
<p><code>integer</code> plotting symbol.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_pch.projections">pch.projections</code></td>
<td>
<p><code>integer</code> plotting symbol.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_gr.col">gr.col</code></td>
<td>
<p><code>integer</code> vector of colours for groups.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_projections">projections</code></td>
<td>
<p>Include backprojections in score plot (default = TRUE).</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_spider">spider</code></td>
<td>
<p>Draw lines between group centers and backprojections (default = FALSE).</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_ellipsoids">ellipsoids</code></td>
<td>
<p><code>character</code> &quot;confidence&quot; or &quot;data&quot; ellipsoids for balanced fixed effect models.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_confidence">confidence</code></td>
<td>
<p><code>numeric</code> vector of ellipsoid confidences, default = c(0.4, 0.68, 0.95).</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_xlim">xlim</code></td>
<td>
<p><code>numeric</code> x limits.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_ylim">ylim</code></td>
<td>
<p><code>numeric</code> y limits.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_xlab">xlab</code></td>
<td>
<p><code>character</code> x label.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_ylab">ylab</code></td>
<td>
<p><code>character</code> y label.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_legendpos">legendpos</code></td>
<td>
<p><code>character</code> position of legend.</p>
</td></tr>
<tr><td><code id="asca_plots_+3A_main">main</code></td>
<td>
<p>Plot title.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Usage of the functions are shown using generics in the examples in <code><a href="#topic+asca">asca</a></code>.
Plot routines are available as
<code>scoreplot.asca</code> and <code>loadingplot.asca</code>.
</p>


<h3>Value</h3>

<p>The plotting routines have no return.
</p>


<h3>References</h3>


<ul>
<li><p> Smilde, A., Jansen, J., Hoefsloot, H., Lamers,R., Van Der Greef, J., and Timmerman, M.(2005). ANOVA-Simultaneous Component Analysis (ASCA): A new tool for analyzing designed metabolomics data. Bioinformatics, 21(13), 3043–3048.
</p>
</li>
<li><p> Liland, K.H., Smilde, A., Marini, F., and Næs,T. (2018). Confidence ellipsoids for ASCA models based on multivariate regression theory. Journal of Chemometrics, 32(e2990), 1–13.
</p>
</li>
<li><p> Martin, M. and Govaerts, B. (2020). LiMM-PCA: Combining ASCA+ and linear mixed models to analyse high-dimensional designed data. Journal of Chemometrics, 34(6), e3232.
</p>
</li></ul>



<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>

<hr>
<h2 id='asca_results'>ASCA Result Methods</h2><span id='topic+asca_results'></span><span id='topic+print.asca'></span><span id='topic+summary.asca'></span><span id='topic+projections'></span><span id='topic+projections.asca'></span><span id='topic+print.summary.asca'></span><span id='topic+loadings.asca'></span><span id='topic+scores.asca'></span>

<h3>Description</h3>

<p>Standard result computation and extraction functions for ASCA (<code><a href="#topic+asca">asca</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'asca'
print(x, ...)

## S3 method for class 'asca'
summary(object, extended = TRUE, df = FALSE, ...)

## S3 method for class 'summary.asca'
print(x, digits = 2, ...)

## S3 method for class 'asca'
loadings(object, factor = 1, ...)

## S3 method for class 'asca'
scores(object, factor = 1, ...)

projections(object, ...)

## S3 method for class 'asca'
projections(object, factor = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="asca_results_+3A_x">x</code></td>
<td>
<p><code>asca</code> object.</p>
</td></tr>
<tr><td><code id="asca_results_+3A_...">...</code></td>
<td>
<p>additional arguments to underlying methods.</p>
</td></tr>
<tr><td><code id="asca_results_+3A_object">object</code></td>
<td>
<p><code>asca</code> object.</p>
</td></tr>
<tr><td><code id="asca_results_+3A_extended">extended</code></td>
<td>
<p>Extended output in summary (default = TRUE).</p>
</td></tr>
<tr><td><code id="asca_results_+3A_df">df</code></td>
<td>
<p>Show degrees of freedom in summary (default = FALSE).</p>
</td></tr>
<tr><td><code id="asca_results_+3A_digits">digits</code></td>
<td>
<p><code>integer</code> number of digits for printing.</p>
</td></tr>
<tr><td><code id="asca_results_+3A_factor">factor</code></td>
<td>
<p><code>integer/character</code> for selecting a model factor.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Usage of the functions are shown using generics in the examples in <code><a href="#topic+asca">asca</a></code>.
Explained variances are available (block-wise and global) through <code>blockexpl</code> and <code>print.rosaexpl</code>.
Object printing and summary are available through:
<code>print.asca</code> and <code>summary.asca</code>.
Scores and loadings have their own extensions of <code>scores()</code> and <code>loadings()</code> through
<code>scores.asca</code> and <code>loadings.asca</code>. Special to ASCA is that scores are on a
factor level basis, while back-projected samples have their own function in <code>projections.asca</code>.
</p>


<h3>Value</h3>

<p>Returns depend on method used, e.g. <code>projections.asca</code> returns projected samples,
<code>scores.asca</code> return scores, while print and summary methods return the object invisibly.
</p>


<h3>References</h3>


<ul>
<li><p> Smilde, A., Jansen, J., Hoefsloot, H., Lamers,R., Van Der Greef, J., and Timmerman, M.(2005). ANOVA-Simultaneous Component Analysis (ASCA): A new tool for analyzing designed metabolomics data. Bioinformatics, 21(13), 3043–3048.
</p>
</li>
<li><p> Liland, K.H., Smilde, A., Marini, F., and Næs,T. (2018). Confidence ellipsoids for ASCA models based on multivariate regression theory. Journal of Chemometrics, 32(e2990), 1–13.
</p>
</li>
<li><p> Martin, M. and Govaerts, B. (2020). LiMM-PCA: Combining ASCA+ and linear mixed models to analyse high-dimensional designed data. Journal of Chemometrics, 34(6), e3232.
</p>
</li></ul>



<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>

<hr>
<h2 id='block.data.frame'>Block-wise indexable data.frame</h2><span id='topic+block.data.frame'></span>

<h3>Description</h3>

<p>This is a convenience function for making <code>data.frame</code>s that are easily
indexed on a block-wise basis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>block.data.frame(X, block_inds = NULL, to.matrix = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="block.data.frame_+3A_x">X</code></td>
<td>
<p>Either a single <code>data.frame</code> to index or a <code>list</code> of matrices/data.frames</p>
</td></tr>
<tr><td><code id="block.data.frame_+3A_block_inds">block_inds</code></td>
<td>
<p>Named <code>list</code> of indexes if <code>X</code> is a single <code>data.frame</code>, otherwise <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="block.data.frame_+3A_to.matrix">to.matrix</code></td>
<td>
<p><code>logical</code> indicating if input list elements should be converted to matrices.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> which can be indexed block-wise.
</p>


<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Random data
M &lt;- matrix(rnorm(200), nrow = 10)
# .. with dimnames
dimnames(M) &lt;- list(LETTERS[1:10], as.character(1:20))

# A named list for indexing
inds &lt;- list(B1 = 1:10, B2 = 11:20)

X &lt;- block.data.frame(M, inds)
str(X)

</code></pre>

<hr>
<h2 id='caldana'>Arabidopsis thaliana growth experiment</h2><span id='topic+caldana'></span>

<h3>Description</h3>

<p>A dataset containing 67 metabolites from plants grown under
different light and temperature conditions. This subset of the data contains
only the light effect and time effect for limited conditions, while the full
data also contains gene expressions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(caldana)
</code></pre>


<h3>Format</h3>

<p>A data.frame having 140 rows and 3 variables:
</p>

<dl>
<dt>light</dt><dd><p>Light levels</p>
</dd>
<dt>time</dt><dd><p>Time of measurement</p>
</dd>
<dt>compound</dt><dd><p>Metabolic compounds</p>
</dd>
</dl>



<h3>References</h3>

<p>Caldana C, Degenkolbe T, Cuadros-Inostroza A, Klie S, Sulpice R,
Leisse A, et al. High-density kinetic analysis of the metabolomic and
transcriptomic response of Arabidopsis to eight environmental conditions.
Plant J. 2011;67(5):869-884.
</p>

<hr>
<h2 id='candies'>Sensory assessment of candies.</h2><span id='topic+candies'></span>

<h3>Description</h3>

<p>A dataset containing 9 sensory attributes for 5 candies assessed
by 11 trained assessors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(candies)
</code></pre>


<h3>Format</h3>

<p>A data.frame having 165 rows and 3 variables:
</p>

<dl>
<dt>assessment</dt><dd><p>Matrix of sensory attributes</p>
</dd>
<dt>assessor</dt><dd><p>Factor of assessors</p>
</dd>
<dt>candy</dt><dd><p>Factor of candies</p>
</dd>
</dl>



<h3>References</h3>

<p>Luciano G, Næs T. Interpreting sensory data by combining principal
component analysis and analysis of variance. Food Qual Prefer. 2009;20(3):167-175.
</p>

<hr>
<h2 id='dummycode'>Dummy-coding of a single vector</h2><span id='topic+dummycode'></span>

<h3>Description</h3>

<p>Flexible dummy-coding allowing for all R's built-in types of contrasts
and optional dropping of a factor level to reduce rank defficiency probability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dummycode(Y, contrast = "contr.sum", drop = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dummycode_+3A_y">Y</code></td>
<td>
<p><code>vector</code> to dummy code.</p>
</td></tr>
<tr><td><code id="dummycode_+3A_contrast">contrast</code></td>
<td>
<p>Contrast type, default = &quot;contr.sum&quot;.</p>
</td></tr>
<tr><td><code id="dummycode_+3A_drop">drop</code></td>
<td>
<p><code>logical</code> indicating if one level should be dropped (default = TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>matrix</code> made by dummy-coding the input vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vec &lt;- c("a","a","b","b","c","c")
dummycode(vec)
</code></pre>

<hr>
<h2 id='extended.model.frame'>Extracting the Extended Model Frame from a Formula or Fit</h2><span id='topic+extended.model.frame'></span>

<h3>Description</h3>

<p>This function attempts to apply <code><a href="stats.html#topic+model.frame">model.frame</a></code> and extend the
result with columns of interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extended.model.frame(formula, data, ..., sep = ".")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extended.model.frame_+3A_formula">formula</code></td>
<td>
<p>a model formula or terms object or an R object.</p>
</td></tr>
<tr><td><code id="extended.model.frame_+3A_data">data</code></td>
<td>
<p>a data.frame, list or environment (see <code><a href="stats.html#topic+model.frame">model.frame</a></code>).</p>
</td></tr>
<tr><td><code id="extended.model.frame_+3A_...">...</code></td>
<td>
<p>further arguments to pass to <code><a href="stats.html#topic+model.frame">model.frame</a></code>.</p>
</td></tr>
<tr><td><code id="extended.model.frame_+3A_sep">sep</code></td>
<td>
<p>separator in contraction of names for interactions (default = &quot;.&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="base.html#topic+data.frame">data.frame</a></code> that includes everything a <code><a href="stats.html#topic+model.frame">model.frame</a></code>
does plus interaction terms.
</p>


<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- data.frame(Y = c(1,2,3,4,5,6),
                  X = factor(LETTERS[c(1,1,2,2,3,3)]),
                  W = factor(letters[c(1,2,1,2,1,2)]))
extended.model.frame(Y ~ X*W, dat)
</code></pre>

<hr>
<h2 id='limmpca'>Linear Mixed Model PCA</h2><span id='topic+limmpca'></span>

<h3>Description</h3>

<p>This function mimics parts of the LiMM-PCA framework, combining ASCA+ and
linear mixed models to analyse high-dimensional designed data. The default
is to use REML estimation and scaling of the backprojected errors. See examples
for alternatives.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>limmpca(
  formula,
  data,
  pca.in = 5,
  aug_error = 0.05,
  use_ED = FALSE,
  REML = TRUE,
  contrasts = "contr.sum",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="limmpca_+3A_formula">formula</code></td>
<td>
<p>Model formula accepting a single response (block) and predictors. See Details for more information.</p>
</td></tr>
<tr><td><code id="limmpca_+3A_data">data</code></td>
<td>
<p>The data set to analyse.</p>
</td></tr>
<tr><td><code id="limmpca_+3A_pca.in">pca.in</code></td>
<td>
<p>Compress response before ASCA (number of components), default = 5.</p>
</td></tr>
<tr><td><code id="limmpca_+3A_aug_error">aug_error</code></td>
<td>
<p>Error term of model (&quot;denominator&quot;, &quot;residual&quot;, numeric alpha-value).
The latter implies the first with a scaling factor.</p>
</td></tr>
<tr><td><code id="limmpca_+3A_use_ed">use_ED</code></td>
<td>
<p>Use Effective Dimensions instead of degrees of freedom when scaling.</p>
</td></tr>
<tr><td><code id="limmpca_+3A_reml">REML</code></td>
<td>
<p>Use restricted maximum likelihood estimation.
Alternatives: TRUE (default), FALSE (ML), NULL (least squares).</p>
</td></tr>
<tr><td><code id="limmpca_+3A_contrasts">contrasts</code></td>
<td>
<p>Effect coding: &quot;sum&quot; (default = sum-coding), &quot;weighted&quot;, &quot;reference&quot;, &quot;treatment&quot;.</p>
</td></tr>
<tr><td><code id="limmpca_+3A_...">...</code></td>
<td>
<p>Additional arguments to <code><a href="#topic+asca_fit">asca_fit</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>limmpca</code>, inheriting from the general <code>asca</code> class.
</p>


<h3>References</h3>


<ul>
<li><p> Martin, M. and Govaerts, B. (2020). LiMM-PCA: Combining ASCA+ and linear mixed models to analyse high-dimensional designed data. Journal of Chemometrics, 34(6), e3232.
</p>
</li></ul>



<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load candies data
data(candies)

# Default LiMM-PCA model with two factors and interaction, 5 PCA components
mod &lt;- limmpca(assessment ~ candy*r(assessor), data=candies)
summary(mod)
scoreplot(mod, factor = "candy")

# LiMM-PCA with least squares estimation and 8 PCA components
modLS &lt;- limmpca(assessment ~ candy*r(assessor), data=candies, REML=NULL, pca.in=8)
summary(modLS)
scoreplot(modLS, factor = "candy")

# Load Caldana data
data(caldana)

# Combining effects in LiMM-PCA (assuming light is a random factor)
mod.comb &lt;- limmpca(compounds ~ time + comb(r(light) + r(time:light)), data=caldana, pca.in=8)
summary(mod.comb)
</code></pre>

<hr>
<h2 id='model.frame.asca'>Model Frame and Model Matrix for ASCA-like Models</h2><span id='topic+model.frame.asca'></span><span id='topic+model.matrix.asca'></span>

<h3>Description</h3>

<p>Extraction functions to retrieve the <code>model.frame</code> and <code>model.matrix</code> of an <code>asca</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'asca'
model.frame(formula, ...)

## S3 method for class 'asca'
model.matrix(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="model.frame.asca_+3A_formula">formula</code></td>
<td>
<p>The <code>asca</code> object.</p>
</td></tr>
<tr><td><code id="model.frame.asca_+3A_...">...</code></td>
<td>
<p>Not implemented</p>
</td></tr>
<tr><td><code id="model.frame.asca_+3A_object">object</code></td>
<td>
<p>The <code>asca</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> or <code>matrix</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load candies data
data(candies)

# Basic ASCA model with two factors
mod &lt;- asca(assessment ~ candy + assessor, data=candies)

# Extract model frame and model matrix
mf &lt;- model.frame(mod)
head(mf)
mm &lt;- model.matrix(mod)
par.old &lt;- par(mar=c(3,3,3,1), mgp=c(1,0.7,0))
image(t(mm[seq(165,1,-1),]), main="Model Matrix", xlab="dummy values", ylab="samples",
     axes=FALSE)
par(par.old)
</code></pre>

<hr>
<h2 id='msca'>Multilevel Simultaneous Component Analysis - MSCA</h2><span id='topic+msca'></span>

<h3>Description</h3>

<p>This MSCA implementation assumes a single factor to be used as between-individuals factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msca(formula, data, contrasts = "contr.sum", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msca_+3A_formula">formula</code></td>
<td>
<p>Model formula accepting a single response (block) and predictors. See Details for more information.</p>
</td></tr>
<tr><td><code id="msca_+3A_data">data</code></td>
<td>
<p>The data set to analyse.</p>
</td></tr>
<tr><td><code id="msca_+3A_contrasts">contrasts</code></td>
<td>
<p>Effect coding: &quot;sum&quot; (default = sum-coding), &quot;weighted&quot;, &quot;reference&quot;, &quot;treatment&quot;.</p>
</td></tr>
<tr><td><code id="msca_+3A_...">...</code></td>
<td>
<p>Additional arguments to <code><a href="#topic+asca_fit">asca_fit</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>asca</code> object containing loadings, scores, explained variances, etc. The object has
associated plotting (<code><a href="#topic+asca_plots">asca_plots</a></code>) and result (<code><a href="#topic+asca_results">asca_results</a></code>) functions.
</p>


<h3>References</h3>


<ul>
<li><p> Smilde, A., Jansen, J., Hoefsloot, H., Lamers,R., Van Der Greef, J., and Timmerman, M.(2005). ANOVA-Simultaneous Component Analysis (ASCA): A new tool for analyzing designed metabolomics data. Bioinformatics, 21(13), 3043–3048.
</p>
</li>
<li><p> Liland, K.H., Smilde, A., Marini, F., and Næs,T. (2018). Confidence ellipsoids for ASCA models based on multivariate regression theory. Journal of Chemometrics, 32(e2990), 1–13.
</p>
</li></ul>



<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load candies data
data(candies)

# Basic MSCA model with a single factor
mod &lt;- msca(assessment ~ candy, data=candies)
print(mod)
summary(mod)

# Result plotting for first factor
loadingplot(mod, scatter=TRUE, labels="names")
scoreplot(mod)

# Within scores
scoreplot(mod, factor="within")

# Within scores per factor level
par.old &lt;- par(mfrow=c(3,2), mar=c(4,4,2,1), mgp=c(2,0.7,0))
for(i in 1:length(mod$scores.within))
  scoreplot(mod, factor="within", within_level=i,
            main=paste0("Level: ", names(mod$scores.within)[i]),
            panel.first=abline(v=0,h=0,col="gray",lty=2))
par(par.old)

# Permutation testing
mod.perm &lt;- asca(assessment ~ candy * assessor, data=candies, permute=TRUE)
summary(mod.perm)

</code></pre>

<hr>
<h2 id='pcanova'>Principal Components Analysis of Variance Simultaneous Component Analysis - PC-ANOVA</h2><span id='topic+pcanova'></span>

<h3>Description</h3>

<p>This is a quite general and flexible implementation of PC-ANOVA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcanova(formula, data, ncomp = 0.9, contrasts = "contr.sum", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pcanova_+3A_formula">formula</code></td>
<td>
<p>Model formula accepting a single response (block) and predictor names separated by + signs.</p>
</td></tr>
<tr><td><code id="pcanova_+3A_data">data</code></td>
<td>
<p>The data set to analyse.</p>
</td></tr>
<tr><td><code id="pcanova_+3A_ncomp">ncomp</code></td>
<td>
<p>The number of components to retain, proportion of variation or default = minimum cross-validation error.</p>
</td></tr>
<tr><td><code id="pcanova_+3A_contrasts">contrasts</code></td>
<td>
<p>Effect coding: &quot;sum&quot; (default = sum-coding), &quot;weighted&quot;, &quot;reference&quot;, &quot;treatment&quot;.</p>
</td></tr>
<tr><td><code id="pcanova_+3A_...">...</code></td>
<td>
<p>Additional parameters for the asca_fit function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PC-ANOVA works in the opposite order of ASCA. First the response matrix
is decomposed using ANOVA. Then the components are analysed using ANOVA
with respect to a design or grouping in the data. The latter can be ordinary
fixed effects modelling or mixed models.
</p>


<h3>Value</h3>

<p>A <code>pcanova</code> object containing loadings, scores, explained variances, etc. The object has
associated plotting (<code><a href="#topic+pcanova_plots">pcanova_plots</a></code>) and result (<code><a href="#topic+pcanova_results">pcanova_results</a></code>) functions.
</p>


<h3>References</h3>

<p>Luciano G, Næs T. Interpreting sensory data by combining principal
component analysis and analysis of variance. Food Qual Prefer. 2009;20(3):167-175.
</p>


<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load candies data
data(candies)

# Basic PC-ANOVA model with two factors, cross-validated opt. of #components
mod &lt;- pcanova(assessment ~ candy + assessor, data = candies)
print(mod)

# PC-ANOVA model with interaction, minimum 90% explained variance
mod &lt;- pcanova(assessment ~ candy * assessor, data = candies, ncomp = 0.9)
print(mod)
summary(mod)

# Tukey group letters for 'candy' per component
lapply(mod$models, function(x)
       mixlm::cld(mixlm::simple.glht(x,
                                     effect = "candy")))

# Result plotting
loadingplot(mod, scatter=TRUE, labels="names")
scoreplot(mod)

# Mixed Model PC-ANOVA, random assessor
mod.mix &lt;- pcanova(assessment ~ candy + r(assessor), data=candies, ncomp = 0.9)
scoreplot(mod.mix)
# Fixed effects
summary(mod.mix)

</code></pre>

<hr>
<h2 id='pcanova_plots'>PC-ANOVA Result Methods</h2><span id='topic+pcanova_plots'></span><span id='topic+scoreplot.pcanova'></span><span id='topic+loadingplot.pcanova'></span>

<h3>Description</h3>

<p>Various plotting procedures for <code><a href="#topic+pcanova">pcanova</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pcanova'
scoreplot(object, factor = 1, comps = 1:2, col = "factor", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pcanova_plots_+3A_object">object</code></td>
<td>
<p><code>pcanova</code> object.</p>
</td></tr>
<tr><td><code id="pcanova_plots_+3A_factor">factor</code></td>
<td>
<p><code>integer/character</code> for selecting a model factor.</p>
</td></tr>
<tr><td><code id="pcanova_plots_+3A_comps">comps</code></td>
<td>
<p><code>integer</code> vector of selected components.</p>
</td></tr>
<tr><td><code id="pcanova_plots_+3A_col">col</code></td>
<td>
<p><code>character</code> for selecting a factor to use for colouring
(default = first factor) or ordinary colour specifications.</p>
</td></tr>
<tr><td><code id="pcanova_plots_+3A_...">...</code></td>
<td>
<p>additional arguments to underlying methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Usage of the functions are shown using generics in the examples in <code><a href="#topic+pcanova">pcanova</a></code>.
Plot routines are available as
<code>scoreplot.pcanova</code> and <code>loadingplot.pcanova</code>.
</p>


<h3>Value</h3>

<p>The plotting routines have no return.
</p>


<h3>References</h3>

<p>Luciano G, Næs T. Interpreting sensory data by combining principal
component analysis and analysis of variance. Food Qual Prefer. 2009;20(3):167-175.
</p>


<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>

<hr>
<h2 id='pcanova_results'>PC-ANOVA Result Methods</h2><span id='topic+pcanova_results'></span><span id='topic+summary.pcanova'></span><span id='topic+projections.pcanova'></span><span id='topic+print.summary.pcanova'></span><span id='topic+print.pcanova'></span>

<h3>Description</h3>

<p>Standard result computation and extraction functions for ASCA (<code><a href="#topic+pcanova">pcanova</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pcanova'
summary(object, ...)

## S3 method for class 'summary.pcanova'
print(x, digits = 2, ...)

## S3 method for class 'pcanova'
print(x, ...)

## S3 method for class 'pcanova'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pcanova_results_+3A_object">object</code></td>
<td>
<p><code>pcanova</code> object.</p>
</td></tr>
<tr><td><code id="pcanova_results_+3A_...">...</code></td>
<td>
<p>additional arguments to underlying methods.</p>
</td></tr>
<tr><td><code id="pcanova_results_+3A_x">x</code></td>
<td>
<p><code>pcanova</code> object.</p>
</td></tr>
<tr><td><code id="pcanova_results_+3A_digits">digits</code></td>
<td>
<p><code>integer</code> number of digits for printing.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Usage of the functions are shown using generics in the examples in <code><a href="#topic+pcanova">pcanova</a></code>.
Explained variances are available (block-wise and global) through <code>blockexpl</code> and <code>print.rosaexpl</code>.
Object printing and summary are available through:
<code>print.pcanova</code> and <code>summary.pcanova</code>.
Scores and loadings have their own extensions of <code>scores()</code> and <code>loadings()</code> through
<code>scores.pcanova</code> and <code>loadings.pcanova</code>. Special to ASCA is that scores are on a
factor level basis, while back-projected samples have their own function in <code>projections.pcanova</code>.
</p>


<h3>Value</h3>

<p>Returns depend on method used, e.g. <code>projections.pcanova</code> returns projected samples,
<code>scores.pcanova</code> return scores, while print and summary methods return the object invisibly.
</p>


<h3>References</h3>

<p>Luciano G, Næs T. Interpreting sensory data by combining principal
component analysis and analysis of variance. Food Qual Prefer. 2009;20(3):167-175.
</p>


<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>

<hr>
<h2 id='permanova'>Permutation Based MANOVA - PERMANOVA</h2><span id='topic+permanova'></span>

<h3>Description</h3>

<p>Wrapper for the <code><a href="vegan.html#topic+adonis2">adonis2</a></code> function to allow ordinary formula input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>permanova(formula, data, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="permanova_+3A_formula">formula</code></td>
<td>
<p>Model formula accepting a single response matrix and predictors.
See details in <code><a href="vegan.html#topic+adonis2">adonis2</a></code>.</p>
</td></tr>
<tr><td><code id="permanova_+3A_data">data</code></td>
<td>
<p>The data set to analyse.</p>
</td></tr>
<tr><td><code id="permanova_+3A_...">...</code></td>
<td>
<p>Additional arguments to <code><a href="vegan.html#topic+adonis2">adonis2</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An ANOVA table with permutation-based p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(caldana)
(pr &lt;- permanova(compounds ~ light * time, caldana))
</code></pre>

<hr>
<h2 id='prc'>Principal Response Curves</h2><span id='topic+prc'></span>

<h3>Description</h3>

<p>Wrapper for the <code><a href="vegan.html#topic+prc">prc</a></code> function to allow for formula input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prc(formula, data, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prc_+3A_formula">formula</code></td>
<td>
<p>Model formula accepting a single response (block) and predictors.
If no predictor is called 'time', time is assumed to be the second predictor.</p>
</td></tr>
<tr><td><code id="prc_+3A_data">data</code></td>
<td>
<p>The data set to analyse.</p>
</td></tr>
<tr><td><code id="prc_+3A_...">...</code></td>
<td>
<p>Additional arguments to <code><a href="vegan.html#topic+prc">prc</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>prc</code>.
</p>


<h3>See Also</h3>

<p>Main methods: <code><a href="#topic+asca">asca</a></code>, <code><a href="#topic+apca">apca</a></code>, <code><a href="#topic+limmpca">limmpca</a></code>, <code><a href="#topic+msca">msca</a></code>, <code><a href="#topic+pcanova">pcanova</a></code>, <code><a href="#topic+prc">prc</a></code> and <code><a href="#topic+permanova">permanova</a></code>.
Workhorse function underpinning most methods: <code><a href="#topic+asca_fit">asca_fit</a></code>.
Extraction of results and plotting: <code><a href="#topic+asca_results">asca_results</a></code>, <code><a href="#topic+asca_plots">asca_plots</a></code>, <code><a href="#topic+pcanova_results">pcanova_results</a></code> and <code><a href="#topic+pcanova_plots">pcanova_plots</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(caldana)
(pr &lt;- prc(compounds ~ light * time, caldana))
summary(pr)
</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+pcr'></span><span id='topic+plsr'></span><span id='topic+MSEP'></span><span id='topic+R2'></span><span id='topic+RMSEP'></span><span id='topic+coefplot'></span><span id='topic+cvsegments'></span><span id='topic+loading.weights'></span><span id='topic+loadingplot'></span><span id='topic+loadings'></span><span id='topic+mvrValstats'></span><span id='topic+predplot'></span><span id='topic+scoreplot'></span><span id='topic+scores'></span><span id='topic+validationplot'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>pls</dt><dd><p><code><a href="pls.html#topic+mvrVal">MSEP</a></code>, <code><a href="pls.html#topic+mvrVal">R2</a></code>, <code><a href="pls.html#topic+mvrVal">RMSEP</a></code>, <code><a href="pls.html#topic+coefplot">coefplot</a></code>, <code><a href="pls.html#topic+cvsegments">cvsegments</a></code>, <code><a href="pls.html#topic+scores">loading.weights</a></code>, <code><a href="pls.html#topic+scoreplot">loadingplot</a></code>, <code><a href="pls.html#topic+scores">loadings</a></code>, <code><a href="pls.html#topic+mvrVal">mvrValstats</a></code>, <code><a href="pls.html#topic+mvr">pcr</a></code>, <code><a href="pls.html#topic+mvr">plsr</a></code>, <code><a href="pls.html#topic+predplot">predplot</a></code>, <code><a href="pls.html#topic+scoreplot">scoreplot</a></code>, <code><a href="pls.html#topic+scores">scores</a></code>, <code><a href="pls.html#topic+validationplot">validationplot</a></code></p>
</dd>
</dl>

<hr>
<h2 id='timeplot'>Timeplot for Combined Effects</h2><span id='topic+timeplot'></span>

<h3>Description</h3>

<p>Timeplot for Combined Effects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>timeplot(
  object,
  factor,
  time,
  comb,
  comp = 1,
  ylim,
  x_time = FALSE,
  xlab = time,
  ylab = paste0("Score ", comp),
  lwd = 2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="timeplot_+3A_object">object</code></td>
<td>
<p><code>asca</code> object.</p>
</td></tr>
<tr><td><code id="timeplot_+3A_factor">factor</code></td>
<td>
<p><code>integer/character</code> main factor.</p>
</td></tr>
<tr><td><code id="timeplot_+3A_time">time</code></td>
<td>
<p><code>integer/character</code> time factor.</p>
</td></tr>
<tr><td><code id="timeplot_+3A_comb">comb</code></td>
<td>
<p><code>integer/character</code> combined effect factor.</p>
</td></tr>
<tr><td><code id="timeplot_+3A_comp">comp</code></td>
<td>
<p><code>integer</code> component number.</p>
</td></tr>
<tr><td><code id="timeplot_+3A_ylim">ylim</code></td>
<td>
<p><code>numeric</code> y limits.</p>
</td></tr>
<tr><td><code id="timeplot_+3A_x_time">x_time</code></td>
<td>
<p><code>logical</code> use time levels as non-equispaced x axis (default = FALSE).</p>
</td></tr>
<tr><td><code id="timeplot_+3A_xlab">xlab</code></td>
<td>
<p><code>character</code> x label.</p>
</td></tr>
<tr><td><code id="timeplot_+3A_ylab">ylab</code></td>
<td>
<p><code>character</code> y label.</p>
</td></tr>
<tr><td><code id="timeplot_+3A_lwd">lwd</code></td>
<td>
<p><code>numeric</code> line width.</p>
</td></tr>
<tr><td><code id="timeplot_+3A_...">...</code></td>
<td>
<p>additional arguments to <code>plot</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("caldana")
mod.comb &lt;- asca(compounds ~ time + comb(light + light:time), data=caldana)

# Default time axis
timeplot(mod.comb, factor="light", time="time", comb=2)

# Non-equispaced time axis (using time levels)
timeplot(mod.comb, factor="light", time="time", comb=2, x_time=TRUE)

# Second component
timeplot(mod.comb, factor="light", time="time", comb=2, comp=2, x_time=TRUE)
</code></pre>

<hr>
<h2 id='update_without_factor'>Update a Model without Factor</h2><span id='topic+update_without_factor'></span>

<h3>Description</h3>

<p>Perform a model update while removing a chosen factor. Hierarchical
corresponds to type &quot;II&quot; sum-of-squares, i.e., obeying marginality,
while non-hierarchical corresponds to type &quot;III&quot; sum-of-squares.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_without_factor(model, fac, hierarchical = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="update_without_factor_+3A_model">model</code></td>
<td>
<p><code>model</code> object to update.</p>
</td></tr>
<tr><td><code id="update_without_factor_+3A_fac">fac</code></td>
<td>
<p><code>character</code> factor to remove.</p>
</td></tr>
<tr><td><code id="update_without_factor_+3A_hierarchical">hierarchical</code></td>
<td>
<p><code>logical</code> obey hierarchy when removing factor (default = TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated model object is returned. If the supplied model is of
type <code>lmerMod</code> and no random effects are left, the model is
automatically converted to a linear model before updating.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
