<!DOCTYPE html><html lang="en"><head><title>Help for package HDBRR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HDBRR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HDBRR'>
<p>High Dimensional Bayesian Ridge Regression without MCMC.</p></a></li>
<li><a href='#matop'>
<p>matop</p></a></li>
<li><a href='#pheno'><p>Durum Wheat</p></a></li>
<li><a href='#phenowheat'><p>Durum wheat dataset</p></a></li>
<li><a href='#X'><p>Durum Wheat X</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>High Dimensional Bayesian Ridge Regression without MCMC</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Ridge regression provide biased estimators of the regression parameters with lower variance. The HDBRR ("High Dimensional Bayesian Ridge Regression") function fits Bayesian Ridge regression without MCMC, this one uses the SVD or QR decomposition for the posterior computation.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>numDeriv, parallel, bigstatsr, MASS, graphics</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-05 22:41:42 UTC; monroy</td>
</tr>
<tr>
<td>Author:</td>
<td>Sergio Perez-Elizalde Developer [aut],
  Blanca Monroy-Castillo Developer [aut, cre],
  Paulino Perez-Rodriguez User [ctb],
  Jose Crossa User [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Blanca Monroy-Castillo Developer &lt;blancamonroy.96@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-05 23:10:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='HDBRR'>
High Dimensional Bayesian Ridge Regression without MCMC.
</h2><span id='topic+HDBRR'></span><span id='topic+summary.HDBRR'></span><span id='topic+print.summary.HDBRR'></span><span id='topic+print.HDBRR'></span><span id='topic+plot.HDBRR'></span><span id='topic+predict.HDBRR'></span><span id='topic+coef.HDBRR'></span>

<h3>Description</h3>

<p>Ridge regression provide biased estimators of the regression parameters with lower variance. The HDBRR (&quot;High Dimensional Bayesian Ridge Regression&quot;) function fits Bayesian Ridge regression without MCMC, this one uses the SVD or QR decomposition for the posterior computation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HDBRR(y, X, n0 = 5, p0 = 5, s20 = NULL, d20 = NULL, h = 0.5,
    intercept = TRUE, vpapp = TRUE,npts = NULL,c = NULL,
    corpred = NULL, method = c("svd","qr"),bigmat = TRUE, ncores = 2, svdx = NULL)

## S3 method for class 'HDBRR'
summary(object, all.coef = FALSE, crit = log(4), ...)

## S3 method for class 'HDBRR'
plot(x, crit = log(4), var_select = FALSE, post = FALSE, ...)

## S3 method for class 'HDBRR'
predict(object,  ...)

## S3 method for class 'summary.HDBRR'
print(x, ...)

## S3 method for class 'HDBRR'
print(x, ...)

## S3 method for class 'HDBRR'
coef(object, all = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HDBRR_+3A_y">y</code></td>
<td>

<p>The data vector (numeric, n) NAs allowed.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_x">X</code></td>
<td>

<p>Design Matrix of dimension <code>n x p</code>.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_n0">n0</code>, <code id="HDBRR_+3A_p0">p0</code></td>
<td>

<p><code>n0/2</code> and <code>p0/2</code> are the shape parameter of the Gamma Inverse prior assigned to the residual variance and the shape parameter of the Gamma Inverse prior assigned to the Beta's variance respectively. The default value for <code>n0/2</code> and <code>p0/2</code> parameter is 5.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_s20">s20</code>, <code id="HDBRR_+3A_d20">d20</code></td>
<td>

<p><code>(n0s20)/2</code> and <code>(p0d20)/2</code> are the scale parameter of the Gamma Inverse prior assigned to the residual variance and the scale parameter of the Gamma Inverse prior assigned to the Beta's variance respectively. The default value for the <code>s20</code> and <code>d20</code> is <code>NULL</code>. If the scale is not specified a value is calculated with <code>h</code> and quantiles.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_h">h</code></td>
<td>

<p>(numeric, 0&lt;<code>h</code>&lt;1) shrinkage factor. Only used if the hyper-parameters are not specified; If h -&gt; 0 then we have greater shrinkage, this is, <code class="reqn">\beta</code> -&gt; 0. If h -&gt; 1 then we have less shrinkage.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_intercept">intercept</code></td>
<td>

<p>Logic value. The default value for the <code>intercept</code> is TRUE.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_vpapp">vpapp</code></td>
<td>

<p>Logic value. Compute an approximation of the predictive variance. The default value for the <code>vpapp</code> is TRUE.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_npts">npts</code></td>
<td>

<p>Number (integer) of points used to evaluate the u's density for the numeric aprroach. The default value for the <code>npts</code> parameter is 200.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_c">c</code></td>
<td>

<p>ratio of Gaussian densities (Spike/Slab) in the prior mixture density of each Beta for variable selection.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_corpred">corpred</code></td>
<td>

<p>The method for the compute of the correlation, there are two methods, Empirical Bayes (<code>"eb"</code>) and Bayesian (<code>"b"</code>) method. The default value for the parameter corpred is NULL. If the values is NULL then the corr and edf values will be NULL.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_method">method</code></td>
<td>

<p>Options for the posterior computation. There are two methods available: <code>"qr"</code> decomposition of <code>X*t(X)</code> and <code>"svd"</code> decomposition of matrix <code>X</code>. The default value for the method is SVD decomposition.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_bigmat">bigmat</code></td>
<td>

<p>Use of the bigstatsr package. The default value for <code>bigmat</code> is <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_ncores">ncores</code></td>
<td>

<p>Number of the cores for computation. The default value for the ncores is 2, you can detect your number of cores with <code>detectCores()</code> and use it (iOS and Linux).
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_object">object</code></td>
<td>

<p>A HDBRR object, typically generated by a call to <code>HDBRR</code>.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_all.coef">all.coef</code></td>
<td>

<p>Logical. Should results be returned for all ridge regression penalty
parameters (<code>all.coef = TRUE</code>), or only  those whose <code>log(bayes factor)&gt;crit</code>.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_crit">crit</code></td>
<td>

<p>Numerical. The lower bound of the log Bayes factor in favour to include a variable in the model. The default value for <code>crit</code> is <code>log(4)</code>.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_...">...</code></td>
<td>

<p>Additional arguments to be passed to or from other methods.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_x">x</code></td>
<td>

<p>A HDBRR object, typically generated by a call to <code>HDBRR</code> (for the
<code>print.HDBRR</code> and <code>plot.HDBRR</code> functions) or an object of class
<code>summary.HDBRR</code> (for the <code>print.summary.HDBRR</code>
function).
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_var_select">var_select</code></td>
<td>

<p>Logical. If is TRUE a plot with variable selection is returned. The default value is FALSE.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_post">post</code></td>
<td>

<p>Logical. If is TRUE a plot with marginal posterior of u is returned. The default value is FALSE.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_all">all</code></td>
<td>

<p>Logical. All coefficients are returned. If is FALSE, then, if <code>p &gt; 250</code> only 250 coefficients are returned. The default value es FALSE.
</p>
</td></tr>
<tr><td><code id="HDBRR_+3A_svdx">svdx</code></td>
<td>

<p>It is possible to add the svd. The default value es NULL.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Ridge regression is a useful tool to deal with colinerity in the homocesastic linear regression model providing biased estimators of the regression parameters with lower variance than the least square estimators.
The model </p>
<p style="text-align: center;"><code class="reqn">y = X\beta + \epsilon</code>
</p>
<p> where <code class="reqn">\epsilon</code> vector is assumed Normal with mean vector 0 and covariance matrix <code class="reqn">\sigma^2 I_n</code>. For further details see the vignettes in the package.
</p>


<h3>Value</h3>

<p>List containing the following components:
</p>
<table role = "presentation">
<tr><td><code>betahat</code></td>
<td>

<p>Vector (numeric, <code>p</code>)  with the betas estimates.
</p>
</td></tr>
<tr><td><code>yhat</code></td>
<td>

<p>Vector (numeric, <code>n</code>) with the y's estimates.
</p>
</td></tr>
<tr><td><code>sdyhat</code></td>
<td>

<p>Vector (numeric, <code>n</code>) with the standard deviation of the predicts values.
</p>
</td></tr>
<tr><td><code>sdpred</code></td>
<td>

<p>Vector (numeric, <code>n</code>) with the standard deviation of predict variances.
</p>
</td></tr>
<tr><td><code>varb</code></td>
<td>

<p>Vector (numeric, <code>p</code>) with the beta's variance.
</p>
</td></tr>
<tr><td><code>sigsqhat</code></td>
<td>

<p>Value (numeric) of the residual variance estimate.
</p>
</td></tr>
<tr><td><code>sigbsqhat</code></td>
<td>

<p>Value (numeric) of the Beta's variance estimate.
</p>
</td></tr>
<tr><td><code>u</code></td>
<td>

<p>Vector (numeric, <code>npts</code>) with the u's values.
</p>
</td></tr>
<tr><td><code>postu</code></td>
<td>

<p>Vector (numeric, <code>npts</code>) with the values of the u posterior.
</p>
</td></tr>
<tr><td><code>uhat</code></td>
<td>

<p>Value (numeric) of u estimated.
</p>
</td></tr>
<tr><td><code>umode</code></td>
<td>

<p>Value (numeric) of the posterior mode of u.
</p>
</td></tr>
<tr><td><code>whichNa</code></td>
<td>

<p>Value (integer) of NAs in the y vector.
</p>
</td></tr>
<tr><td><code>phat</code></td>
<td>

<p>Vector (numeric, <code>p</code>), selection probability of x_i.
</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>

<p>Used in the variable selection.
</p>
</td></tr>
<tr><td><code>edf</code></td>
<td>

<p>Value (numeric) of the effective degrees of freedom for regression.
</p>
</td></tr>
<tr><td><code>corr</code></td>
<td>

<p>Vector (numeric, <code>n</code>) of the correlation between <code>y_i</code> estimates and <code>y_i</code>.
</p>
</td></tr>
<tr><td><code>svdx</code></td>
<td>

<p>The svd decomposition.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sergio Perez-Elizalde, Blanca E. Monroy-Castillo, Paulino Perez-Rodriguez, Jose Crossa.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

data("phenowheat")
mod &lt;- lmer(pheno$HD~pheno$env+(1|pheno$Line))
y &lt;- unlist(ranef(mod))
n &lt;- length(y)
X &lt;- scale(X, scale=F)
fitall &lt;- HDBRR(y,X/sqrt(ncol(X)),intercept = FALSE, corpred = "eb", c = 100)
fitall
sumarry(fitall, crit = 0)
plot(fitall, crit = 0)
predict(fitall)


## End(Not run)
</code></pre>

<hr>
<h2 id='matop'>
matop
</h2><span id='topic+matop'></span>

<h3>Description</h3>

<p>Compute the SVD or QR decomposition of the matrix X.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matop(y = NULL, X, method = c("svd", "qr"), bigmat = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="matop_+3A_y">y</code></td>
<td>

<p>The data vector (numeric, n) NAs allowed. The default value is NULL, It is possible to compute the SVD or QR decomposition without y.
</p>
</td></tr>
<tr><td><code id="matop_+3A_x">X</code></td>
<td>

<p>Design Matrix of dimension n x p.
</p>
</td></tr>
<tr><td><code id="matop_+3A_method">method</code></td>
<td>

<p>Options for the posterior computation. Two methods, <code>"qr"</code> and <code>"svd"</code> decomposition. The default value for the method is SVD descomposition.
</p>
</td></tr>
<tr><td><code id="matop_+3A_bigmat">bigmat</code></td>
<td>

<p>Use of the bigstatsr package. The default value for bigmat is <code>TRUE</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Use the bigstartsr package when <code>p &gt;&gt; n</code>. Auxiliary in the HDBRR function.
</p>


<h3>Value</h3>

<p>If the method used is svd then the list containing the following components:
</p>
<table role = "presentation">
<tr><td><code>y</code></td>
<td>

<p>The data vector (numeric, <code>n</code>) NAs allowed.
</p>
</td></tr>
<tr><td><code>X</code></td>
<td>

<p>Design Matrix of dimension <code>n x p</code>.
</p>
</td></tr>
<tr><td><code>D</code></td>
<td>

<p>A vector containing the singular values of <code>X</code>, of lenght <code>min(n,p)</code>.
</p>
</td></tr>
<tr><td><code>L</code></td>
<td>

<p>A matrix whose columns contain the left singular vectors of <code>X</code>,
</p>
</td></tr>
<tr><td><code>R</code></td>
<td>

<p>A matrix whose columns contain the right singular vectors of <code>X</code>.
</p>
</td></tr>
<tr><td><code>ev</code></td>
<td>

<p>A vector containing the square of <code>D</code>.
</p>
</td></tr>
<tr><td><code>Ly</code></td>
<td>

<p>The cross-product between the matrix <code>L</code> and vector <code>y</code>.
</p>
</td></tr>
<tr><td><code>n</code></td>
<td>

<p>Number of rows of <code>X</code>.
</p>
</td></tr>
<tr><td><code>p</code></td>
<td>

<p>Number of columns of <code>X</code>.
</p>
</td></tr>
</table>
<p>If the method used is qr then the list containing the following components:
</p>
<table role = "presentation">
<tr><td><code>y</code></td>
<td>

<p>The data vector (numeric, <code>n</code>) NAs allowed.
</p>
</td></tr>
<tr><td><code>X</code></td>
<td>

<p>Design Matrix of dimension <code>n x p</code>.
</p>
</td></tr>
<tr><td><code>R</code></td>
<td>

<p>An upper triangular matrix of dimension <code>n x p</code>.
</p>
</td></tr>
<tr><td><code>n</code></td>
<td>

<p>Number of rows of <code>X</code>.
</p>
</td></tr>
<tr><td><code>p</code></td>
<td>

<p>Number of columns of <code>X</code>.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sergio Perez-Elizalde, Blanca E. Monroy-Castillo, Paulino Perez-Rodriguez.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+qr">qr</a></code>, <code><a href="base.html#topic+svd">svd</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n &lt;- 30
p &lt;- 100
X &lt;- matrix(rnorm(n*(p-1),1,1/p),nrow = n,ncol = p-1)
Beta &lt;- sample(1:p,p-1,rep = FALSE)
Beta &lt;- c(1,Beta)
y &lt;- cbind(rep(1,n),X) %*% Beta+rnorm(n,0,1)
matop(y, X, bigmat = TRUE)
</code></pre>

<hr>
<h2 id='pheno'>Durum Wheat</h2><span id='topic+pheno'></span>

<h3>Description</h3>

<p>The final number of SNPs included in the NCCR linkage map was 7594. The markers were centered and standardized. Phenotypic evaluation of the NCCR population was performed during two growing seasons (2010-2011 and 2011-2012) in locations in the Po Valley representative of the target environments where durum wheat is grown: Cadriano in the 2010-2011 growin season (Cad11) and the 2011-2012 growing season (Cad12); Poggio Renatico in the 2010-2011 growing season (Pr11) and Argelato in the 2011-2012  growing season (Arg12).
</p>


<h3>Source</h3>

<p>International Maize and Wheat Improvement Center (CIMMYT), Mexico.
</p>

<hr>
<h2 id='phenowheat'>Durum wheat dataset</h2><span id='topic+phenowheat'></span>

<h3>Description</h3>

<p>This contain data from a multiparental durum wheat (Triticum turgidum L. spp. duram) trial consisting of 334 lines evaluated in a country-year combination. This population is characterized for Grain Yield (GY), grain volume weight (GVW), 1000-kernel weight (GWT) and heading date (HD) in the four environments. For further details see the vignettes in the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(phenowheat)
</code></pre>


<h3>Format</h3>


<ol>
<li><p><code>pheno</code>Matrix phenowheat.pheno contains the phenotypic data.
</p>
</li>
<li><p><code>X</code>The matrix phenowheat.X contains the Genotypic data.
</p>
</li></ol>


<hr>
<h2 id='X'>Durum Wheat X</h2><span id='topic+X'></span>

<h3>Description</h3>

<p>Is a matrix (338 x 7594) with A balanced, four-way multiparental cross population was developed from four elite durum wheat cultivars (Neodur, Claudio,
Colosseo, and Rascon/Tarro) that were chosen as diverse contributors of different alleles of agronomic relevance.
</p>


<h3>Source</h3>

<p>International Maize and Wheat Improvement Center (CIMMYT), Mexico.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
