<!DOCTYPE html><html><head><title>Help for package hdcuremodels</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hdcuremodels}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#amltest'><p>AML test data</p></a></li>
<li><a href='#amltrain'><p>AML training data</p></a></li>
<li><a href='#AUC'><p>AUC for cure prediction using mean score imputation</p></a></li>
<li><a href='#coef.mixturecure'><p>Extract model coefficients from a fitted mixture cure object</p></a></li>
<li><a href='#concordance_mcm'><p>C-statistic for mixture cure models</p></a></li>
<li><a href='#cure_estimate'><p>Estimate cured fraction</p></a></li>
<li><a href='#cureem'><p>Fit penalized mixture cure model using the E-M algorithm</p></a></li>
<li><a href='#curegmifs'><p>Fit penalized parametric mixture cure model using the GMIFS algorithm</p></a></li>
<li><a href='#cv_cureem'><p>Fit penalized mixture cure model using the E-M algorithm with cross-validation for parameter tuning</p></a></li>
<li><a href='#cv_curegmifs'><p>Fit a penalized parametric mixture cure model using the GMIFS algorithm with cross-validation for model selection</p></a></li>
<li><a href='#generate_cure_data'><p>Simulate data under a mixture cure model</p></a></li>
<li><a href='#nonzerocure_test'><p>Non-parametric pest for a non-zero cured fraction</p></a></li>
<li><a href='#plot.mixturecure'><p>Plot fitted mixture cure model</p></a></li>
<li><a href='#predict.mixturecure'><p>Predicted probabilities for susceptibles, linear predictor for latency, and risk class for latency for mixture cure fit</p></a></li>
<li><a href='#print.mixturecure'><p>Print the contents of a mixture cure fitted object</p></a></li>
<li><a href='#sufficient_fu_test'><p>Test for sufficient follow-up</p></a></li>
<li><a href='#summary.mixturecure'><p>Summarize a Fitted Mixture Cure Object.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Penalized Mixture Cure Models for High-Dimensional Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-06-11</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions for fitting various penalized parametric and semi-parametric mixture cure models with different penalty functions, testing for a significant cure fraction, and testing for sufficient follow-up as described in Fu et al (2022)&lt;<a href="https://doi.org/10.1002%2Fsim.9513">doi:10.1002/sim.9513</a>&gt; and Archer et al (2024)&lt;<a href="https://doi.org/10.1186%2Fs13045-024-01553-6">doi:10.1186/s13045-024-01553-6</a>&gt;. False discovery rate controlled variable selection is provided using model-X knock-offs. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>doParallel, flexsurv, flexsurvcure, foreach, ggplot2, ggpubr,
glmnet, knockoff, mvnfast, parallel, plyr, methods, survival</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-11 14:12:41 UTC; archer.43</td>
</tr>
<tr>
<td>Author:</td>
<td>Han Fu [aut],
  Kellie J. Archer <a href="https://orcid.org/0000-0003-1555-5781"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kellie J. Archer &lt;archer.43@osu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-13 10:10:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='amltest'>AML test data</h2><span id='topic+amltest'></span>

<h3>Description</h3>

<p>Duration of complete response for 40 cytogenetically normal AML patients and a subset of 320 transcript expression from RNA-sequencing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>amltest
</code></pre>


<h3>Format</h3>

<p>A data frame with 40 rows (subjects) and 322 columns:
</p>

<dl>
<dt>cryr</dt><dd><p>duration of complete response in years</p>
</dd>
<dt>relapse.death</dt><dd><p>censoring indicator: 1 = relapsed or died; 0 = alive at last follow=up</p>
</dd>
<dt>ENSG00000001561</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000005249</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000006757</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000007062</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000007968</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000008283</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000008405</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000008441</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000010295</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000011028</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000011198</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000012779</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000012817</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000013306</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000013725</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000018189</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000022267</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000023171</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000023909</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000029639</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000047634</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000049192</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000053524</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000058056</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000060138</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000061918</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000065809</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000065923</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000068489</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000069020</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000070404</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000071894</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000072422</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000073605</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000076555</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000080823</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000089723</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000090382</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000090975</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100068</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100077</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100299</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100376</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100418</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100448</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100596</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100916</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000102409</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000102760</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000104689</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000104946</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000105518</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000105808</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000106367</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000106526</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000106546</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000106780</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000107104</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000107742</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000107798</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000107816</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000107957</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000109674</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000110076</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000110237</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000110492</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000110799</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000111275</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000112773</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000113504</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000114268</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000114737</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000115183</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000115414</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000115457</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000115525</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000116574</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000117480</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000119280</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000120594</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000120675</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000120832</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000120913</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000121005</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000121039</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000121274</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000123080</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000123836</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000124019</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000124882</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000126822</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000127152</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000129824</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000130702</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000131188</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000131370</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000132122</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000132530</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000132819</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000132849</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000133401</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000133619</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000134531</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000134897</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135074</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135245</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135272</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135362</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135363</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135916</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000136026</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000136193</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000136231</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000136997</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000137193</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000137198</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000138722</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000139318</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000140287</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000144036</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000144647</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000144677</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000145476</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000145545</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000146243</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000146373</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000147044</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000147180</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000148444</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000148484</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000149131</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000150760</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000150782</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000151135</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000151208</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000151458</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000152409</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000152580</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000152767</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000152778</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000153563</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000154217</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000154743</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000154760</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000154874</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000156381</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000157107</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000157240</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000157873</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000157978</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000158691</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000159339</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000159403</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000159788</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000160685</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000160781</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000161509</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000162433</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000162614</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000162676</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000163412</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000163702</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000163814</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000164086</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000164172</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000164442</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000165272</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000166165</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000166435</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000166987</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000167291</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000167565</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000167851</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000168026</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000168209</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000168502</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000168939</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000169203</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000169247</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000169504</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000169860</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000169991</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000170035</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000170180</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000170456</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000170522</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000170909</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000171121</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000171222</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000171476</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000171813</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000171962</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000172197</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000172236</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000173083</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000173530</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000173926</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000174059</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000174080</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000174130</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000174738</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000175265</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000175352</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000176597</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000179222</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000179630</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000179639</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000179820</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000180096</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000180596</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000180902</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000181104</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000182866</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000182871</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000183087</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000183091</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000184371</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185129</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185201</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185245</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185291</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185304</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185710</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185883</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185986</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000186130</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000186854</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000187010</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000187627</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000187653</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000187837</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188002</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188107</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188211</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188636</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188738</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188856</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000189164</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000189223</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000196155</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000196189</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000196415</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000196565</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197081</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197121</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197253</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197256</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197321</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197561</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197728</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197860</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197937</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197951</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000198743</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000198838</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000199347</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000200243</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000201801</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000203872</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000204172</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000205571</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000205593</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000208772</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213085</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213261</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213626</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213722</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213906</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213967</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000214016</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000214425</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000216316</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000220008</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000223345</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000224080</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000225138</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000226471</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000227097</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000227191</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000227615</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000228049</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000229153</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000230076</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000231160</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000231721</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000233927</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000233974</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000234883</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000236876</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000237298</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000237892</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000238160</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000239437</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000241399</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000241489</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000241529</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000244405</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000247627</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000249592</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000250116</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000250251</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000251079</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000253210</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000253276</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000254415</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000259276</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000260727</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000261377</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000264885</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000264895</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000267136</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000267551</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000267702</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000268001</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000268573</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000270554</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000270562</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000271646</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000273018</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000273033</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11068580/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11068580/</a>
</p>

<hr>
<h2 id='amltrain'>AML training data</h2><span id='topic+amltrain'></span>

<h3>Description</h3>

<p>Duration of complete response for 306 cytogenetically normal AML patients and a subset of 320 transcript expression from RNA-sequencing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>amltrain
</code></pre>


<h3>Format</h3>

<p>A data frame with 306 rows (subjects) and 322 columns:
</p>

<dl>
<dt>cryr</dt><dd><p>duration of complete response in years</p>
</dd>
<dt>relapse.death</dt><dd><p>censoring indicator: 1 = relapsed or died; 0 = alive at last follow=up</p>
</dd>
<dt>ENSG00000001561</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000005249</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000006757</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000007062</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000007968</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000008283</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000008405</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000008441</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000010295</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000011028</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000011198</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000012779</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000012817</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000013306</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000013725</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000018189</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000022267</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000023171</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000023909</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000029639</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000047634</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000049192</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000053524</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000058056</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000060138</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000061918</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000065809</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000065923</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000068489</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000069020</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000070404</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000071894</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000072422</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000073605</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000076555</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000080823</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000089723</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000090382</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000090975</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100068</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100077</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100299</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100376</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100418</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100448</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100596</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000100916</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000102409</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000102760</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000104689</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000104946</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000105518</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000105808</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000106367</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000106526</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000106546</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000106780</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000107104</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000107742</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000107798</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000107816</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000107957</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000109674</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000110076</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000110237</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000110492</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000110799</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000111275</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000112773</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000113504</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000114268</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000114737</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000115183</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000115414</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000115457</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000115525</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000116574</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000117480</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000119280</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000120594</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000120675</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000120832</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000120913</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000121005</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000121039</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000121274</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000123080</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000123836</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000124019</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000124882</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000126822</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000127152</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000129824</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000130702</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000131188</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000131370</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000132122</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000132530</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000132819</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000132849</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000133401</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000133619</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000134531</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000134897</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135074</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135245</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135272</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135362</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135363</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000135916</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000136026</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000136193</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000136231</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000136997</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000137193</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000137198</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000138722</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000139318</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000140287</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000144036</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000144647</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000144677</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000145476</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000145545</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000146243</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000146373</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000147044</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000147180</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000148444</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000148484</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000149131</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000150760</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000150782</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000151135</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000151208</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000151458</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000152409</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000152580</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000152767</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000152778</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000153563</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000154217</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000154743</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000154760</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000154874</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000156381</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000157107</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000157240</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000157873</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000157978</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000158691</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000159339</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000159403</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000159788</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000160685</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000160781</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000161509</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000162433</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000162614</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000162676</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000163412</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000163702</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000163814</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000164086</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000164172</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000164442</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000165272</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000166165</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000166435</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000166987</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000167291</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000167565</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000167851</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000168026</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000168209</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000168502</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000168939</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000169203</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000169247</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000169504</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000169860</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000169991</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000170035</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000170180</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000170456</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000170522</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000170909</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000171121</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000171222</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000171476</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000171813</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000171962</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000172197</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000172236</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000173083</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000173530</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000173926</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000174059</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000174080</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000174130</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000174738</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000175265</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000175352</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000176597</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000179222</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000179630</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000179639</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000179820</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000180096</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000180596</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000180902</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000181104</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000182866</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000182871</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000183087</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000183091</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000184371</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185129</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185201</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185245</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185291</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185304</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185710</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185883</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000185986</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000186130</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000186854</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000187010</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000187627</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000187653</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000187837</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188002</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188107</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188211</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188636</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188738</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000188856</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000189164</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000189223</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000196155</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000196189</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000196415</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000196565</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197081</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197121</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197253</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197256</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197321</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197561</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197728</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197860</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197937</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000197951</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000198743</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000198838</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000199347</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000200243</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000201801</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000203872</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000204172</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000205571</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000205593</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000208772</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213085</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213261</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213626</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213722</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213906</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000213967</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000214016</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000214425</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000216316</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000220008</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000223345</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000224080</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000225138</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000226471</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000227097</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000227191</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000227615</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000228049</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000229153</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000230076</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000231160</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000231721</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000233927</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000233974</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000234883</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000236876</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000237298</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000237892</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000238160</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000239437</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000241399</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000241489</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000241529</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000244405</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000247627</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000249592</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000250116</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000250251</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000251079</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000253210</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000253276</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000254415</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000259276</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000260727</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000261377</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000264885</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000264895</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000267136</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000267551</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000267702</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000268001</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000268573</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000270554</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000270562</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000271646</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000273018</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
<dt>ENSG00000273033</dt><dd><p>normalized expression for indicated transcript</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11068580/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11068580/</a>
</p>

<hr>
<h2 id='AUC'>AUC for cure prediction using mean score imputation</h2><span id='topic+AUC'></span>

<h3>Description</h3>

<p>This function calculates the AUC for cure prediction using the mean score imputation (MSI) method proposed by Asano et al.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AUC(object, newdata, cure_cutoff = 5, model.select = "AIC")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AUC_+3A_object">object</code></td>
<td>
<p>a <code>mixturecure</code> object resulting from <code>curegmifs</code>, <code>cureem</code>, <code>cv_curegmifs</code>, <code>cv_cureem</code>.</p>
</td></tr>
<tr><td><code id="AUC_+3A_newdata">newdata</code></td>
<td>
<p>an optional data.frame that minimally includes the incidence and/or latency variables to use for predicting the response. If omitted, the training data are used.</p>
</td></tr>
<tr><td><code id="AUC_+3A_cure_cutoff">cure_cutoff</code></td>
<td>
<p>cutoff value for cure, used to produce a proxy for the unobserved cure status; default is 5.</p>
</td></tr>
<tr><td><code id="AUC_+3A_model.select">model.select</code></td>
<td>
<p>for models fit using <code>curegmifs</code> or <code>cureem</code> any step along the solution path can be selected. The default is <code>model.select = "AIC"</code> which calculates the predicted values using the coefficients from the model having the lowest AIC. Other options are <code>model.select = "mAIC"</code> for the modified AIC,  <code>model.select = "cAIC"</code> for the corrected AIC, <code>model.select = "BIC"</code>, <code>model.select = "mBIC"</code> for the modified BIC, <code>model.select = "EBIC"</code> for the extended BIC, <code>model.select = "logLik"</code> for the step that maximizes the log-likelihood, or any numeric value from the solution path. This option has no effect for objects fit using <code>cv_curegmifs</code> or <code>cv_cureem</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the AUC value for cure prediction using the mean score imputation (MSI) method.
</p>


<h3>References</h3>

<p>Asano, J., Hirakawa, H., Hamada, C. (2014) Assessing the prediction accuracy of cure in the Cox proportional hazards cure model: an application to breast cancer data. <em>Pharmaceutical Statistics</em>, <b>13</b>:357&ndash;363.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+concordance_mcm">concordance_mcm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 100, J = 10, nTrue = 10, A = 1.8)
training &lt;- temp$Training
testing &lt;- temp$Testing
fit &lt;- curegmifs(Surv(Time, Censor) ~ .,
                  data = training, x.latency = training,
                  model = "weibull", thresh = 1e-4, maxit = 2000,
                  epsilon = 0.01, verbose = FALSE)
AUC(fit)
AUC(fit, newdata = testing)
</code></pre>

<hr>
<h2 id='coef.mixturecure'>Extract model coefficients from a fitted mixture cure object</h2><span id='topic+coef.mixturecure'></span>

<h3>Description</h3>

<p><code>coef.mixturecure</code> is a generic function which extracts the model coefficients from a fitted mixture cure model object fit using <code>curegmifs</code>, <code>cureem</code>, <code>cv_curegmifs</code>, or <code>cv_cureem</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'mixturecure'
coef(object, model.select = "AIC", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.mixturecure_+3A_object">object</code></td>
<td>
<p>a <code>mixturecure</code> object resulting from <code>curegmifs</code>, <code>cureem</code>, <code>cv_curegmifs</code>, or <code>cv_cureem</code>.</p>
</td></tr>
<tr><td><code id="coef.mixturecure_+3A_model.select">model.select</code></td>
<td>
<p>for models fit using <code>curegmifs</code> or <code>cureem</code> any step along the solution path can be selected. The default is <code>model.select = "AIC"</code> which calculates the predicted values using the coefficients from the model having the lowest AIC. Other options are <code>model.select = "mAIC"</code> for the modified AIC,  <code>model.select = "cAIC"</code> for the corrected AIC, <code>model.select = "BIC"</code>, <code>model.select = "mBIC"</code> for the modified BIC, <code>model.select = "EBIC"</code> for the extended BIC, <code>model.select = "logLik"</code> for the step that maximizes the log-likelihood, or any numeric value from the solution path. This option has no effect for objects fit using <code>cv_curegmifs</code> or <code>cv_cureem</code>.</p>
</td></tr>
<tr><td><code id="coef.mixturecure_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of estimated parameters extracted from the model object using the model selection criterion
</p>


<h3>See Also</h3>

<p><code><a href="#topic+curegmifs">curegmifs</a></code>, <code><a href="#topic+cureem">cureem</a></code>, <code><a href="#topic+summary.mixturecure">summary.mixturecure</a></code>, <code><a href="#topic+plot.mixturecure">plot.mixturecure</a></code>, <code><a href="#topic+predict.mixturecure">predict.mixturecure</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 100, J = 10, nTrue = 10, A = 1.8)
training &lt;- temp$Training
fit &lt;- curegmifs(Surv(Time, Censor) ~ .,
                      data = training, x.latency = training,
                      model = "weibull", thresh = 1e-4, maxit = 2000, epsilon = 0.01,
                      verbose = FALSE)
coef(fit)
</code></pre>

<hr>
<h2 id='concordance_mcm'>C-statistic for mixture cure models</h2><span id='topic+concordance_mcm'></span>

<h3>Description</h3>

<p>This function calculates the C-statistic using the cure status weighting (CSW) method proposed by Asano and Hirakawa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>concordance_mcm(object, newdata, cure_cutoff = 5, model.select = "AIC")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concordance_mcm_+3A_object">object</code></td>
<td>
<p>a <code>mixturecure</code> object resulting from <code>curegmifs</code>, <code>cureem</code>, <code>cv_curegmifs</code>, <code>cv_cureem</code>.</p>
</td></tr>
<tr><td><code id="concordance_mcm_+3A_newdata">newdata</code></td>
<td>
<p>an optional data.frame that minimally includes the incidence and/or latency variables to use for predicting the response. If omitted, the training data are used.</p>
</td></tr>
<tr><td><code id="concordance_mcm_+3A_cure_cutoff">cure_cutoff</code></td>
<td>
<p>cutoff value for cure, used to produce a proxy for the unobserved cure status; default is 5.</p>
</td></tr>
<tr><td><code id="concordance_mcm_+3A_model.select">model.select</code></td>
<td>
<p>for models fit using <code>curegmifs</code> or <code>cureem</code> any step along the solution path can be selected. The default is <code>model.select = "AIC"</code> which calculates the predicted values using the coefficients from the model having the lowest AIC. Other options are <code>model.select = "mAIC"</code> for the modified AIC,  <code>model.select = "cAIC"</code> for the corrected AIC, <code>model.select = "BIC"</code>, <code>model.select = "mBIC"</code> for the modified BIC, <code>model.select = "EBIC"</code> for the extended BIC, <code>model.select = "logLik"</code> for the step that maximizes the log-likelihood, or any numeric value from the solution path. This option has no effect for objects fit using <code>cv_curegmifs</code> or <code>cv_cureem</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>value of C-statistic for the cure models.
</p>


<h3>References</h3>

<p>Asano, J. and Hirakawa, H. (2017) Assessing the prediction accuracy of a cure model for censored survival data with long-term survivors: Application to breast cancer data. <em>Journal of Biopharmaceutical Statistics</em>, <b>27</b>:6, 918&ndash;932.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AUC">AUC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 100, J = 10, nTrue = 10, A = 1.8)
training &lt;- temp$Training
testing &lt;- temp$Testing
fit &lt;- curegmifs(Surv(Time, Censor) ~ .,
                  data = training, x.latency = training,
                  model = "weibull", thresh = 1e-4, maxit = 2000,
                  epsilon = 0.01, verbose = FALSE)
concordance_mcm(fit)
concordance_mcm(fit, newdata = testing)
</code></pre>

<hr>
<h2 id='cure_estimate'>Estimate cured fraction</h2><span id='topic+cure_estimate'></span>

<h3>Description</h3>

<p>Estimates the cured fraction using a Kaplan-Meier fitted object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cure_estimate(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cure_estimate_+3A_object">object</code></td>
<td>
<p>a <code>survfit</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>estimated proportion of cured observations
</p>


<h3>See Also</h3>

<p><code><a href="survival.html#topic+survfit">survfit</a></code>, <code><a href="#topic+sufficient_fu_test">sufficient_fu_test</a></code>, <code><a href="#topic+nonzerocure_test">nonzerocure_test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 100, J = 10, nTrue = 10, A = 1.8)
training &lt;- temp$Training
km.fit &lt;- survfit(Surv(Time, Censor) ~ 1, data = training)
cure_estimate(km.fit)
</code></pre>

<hr>
<h2 id='cureem'>Fit penalized mixture cure model using the E-M algorithm</h2><span id='topic+cureem'></span>

<h3>Description</h3>

<p>Fits a penalized parametric and semi-parametric mixture cure model (MCM) using the E-M algorithm with user-specified penalty parameters. The lasso (L1), MCP, and SCAD penalty is supported for the Cox MCM while only lasso is currently supported for parametric MCMs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cureem(
  formula,
  data,
  subset,
  x.latency = NULL,
  model = "cox",
  penalty = "lasso",
  penalty.factor.inc = NULL,
  penalty.factor.lat = NULL,
  thresh = 0.001,
  scale = TRUE,
  maxit = NULL,
  inits = NULL,
  lambda.inc = 0.1,
  lambda.lat = 0.1,
  gamma.inc = 3,
  gamma.lat = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cureem_+3A_formula">formula</code></td>
<td>
<p>an object of class &quot;<code>formula</code>&quot; (or one that can be coerced to that class): a symbolic description of the model to be fitted. The response must be a survival object as returned by the <code>Surv</code> function while the variables on the right side of the formula are the covariates that are included in the incidence portion of the model.</p>
</td></tr>
<tr><td><code id="cureem_+3A_data">data</code></td>
<td>
<p>a data.frame in which to interpret the variables named in the <code>formula</code> or in the <code>subset</code> argument.</p>
</td></tr>
<tr><td><code id="cureem_+3A_subset">subset</code></td>
<td>
<p>an optional expression indicating which subset of observations to be used in the fitting process, either a numeric or factor variable should be used in subset, not a character variable. All observations are included by default.</p>
</td></tr>
<tr><td><code id="cureem_+3A_x.latency">x.latency</code></td>
<td>
<p>specifies the variables to be included in the latency portion of the model and can be either a matrix of predictors, a model formula with the right hand side specifying the latency variables, or the same data.frame passed to the <code>data</code> parameter. Note that when using the model formula syntax for <code>x.latency</code> it cannot handle <code>x.latency = ~ .</code>.</p>
</td></tr>
<tr><td><code id="cureem_+3A_model">model</code></td>
<td>
<p>type of regression model to use for the latency portion of mixture cure model. Can be &quot;cox&quot;, &quot;weibull&quot;, or &quot;exponential&quot; (default is &quot;cox&quot;).</p>
</td></tr>
<tr><td><code id="cureem_+3A_penalty">penalty</code></td>
<td>
<p>type of penalty function. Can be &quot;lasso&quot;, &quot;MCP&quot;, or &quot;SCAD&quot; (default is &quot;lasso&quot;).</p>
</td></tr>
<tr><td><code id="cureem_+3A_penalty.factor.inc">penalty.factor.inc</code></td>
<td>
<p>vector of binary indicators representing the penalty to apply to each incidence coefficient: 0 implies no shrinkage and 1 implies shrinkage. If not supplied, 1 is applied to all incidence variables.</p>
</td></tr>
<tr><td><code id="cureem_+3A_penalty.factor.lat">penalty.factor.lat</code></td>
<td>
<p>vector of binary indicators representing the penalty to apply to each latency coefficient: 0 implies no shrinkage and 1 implies shrinkage. If not supplied, 1 is applied to all latency variables.</p>
</td></tr>
<tr><td><code id="cureem_+3A_thresh">thresh</code></td>
<td>
<p>small numeric value. The iterative process stops when the differences between successive expected penalized complete-data log-likelihoods for both incidence and latency components are less than this specified level of tolerance (default is 10^-3).</p>
</td></tr>
<tr><td><code id="cureem_+3A_scale">scale</code></td>
<td>
<p>logical, if TRUE the predictors are centered and scaled.</p>
</td></tr>
<tr><td><code id="cureem_+3A_maxit">maxit</code></td>
<td>
<p>integer specifying the maximum number of passes over the data for each lambda. If not specified, 100 is applied when <code>penalty = "lasso"</code> and 1000 is applied when <code>penalty = "MCP"</code> or <code>penalty = "SCAD"</code>.</p>
</td></tr>
<tr><td><code id="cureem_+3A_inits">inits</code></td>
<td>
<p>an optional list specifiying the initial value for the incidence intercept (<code>itct</code>), a numeric vector for the unpenalized incidence coefficients (<code>b_u</code>), and a numeric vector for unpenalized latency coefficients (<code>beta_u</code>).  For parametric models, it should also include a numeric value for the rate parameter (<code>lambda</code>) when <code>model = "weibull"</code> or <code>model = "exponential"</code>, and a numeric value for the shape parameter (<code>alpha</code>) when <code>model = "weibull"</code>. When <code>model = "cox"</code>, it should also include a numeric vector for the latency survival probabilities <code class="reqn">S_u(t_i|w_i)</code> for i=1,...,N (<code>survprob</code>). Penalized coefficients are initialized to zero. If <code>inits</code> is not specified or improperly specified, initialization is automatically provided by the function.</p>
</td></tr>
<tr><td><code id="cureem_+3A_lambda.inc">lambda.inc</code></td>
<td>
<p>numeric value for the penalization parameter <code class="reqn">\lambda</code> for variables in the incidence portion of the model.</p>
</td></tr>
<tr><td><code id="cureem_+3A_lambda.lat">lambda.lat</code></td>
<td>
<p>numeric value for the penalization parameter <code class="reqn">\lambda</code> for variables in the latency portion of the model.</p>
</td></tr>
<tr><td><code id="cureem_+3A_gamma.inc">gamma.inc</code></td>
<td>
<p>numeric value for the penalization parameter <code class="reqn">\gamma</code> for variables in the incidence portion of the model when <code>penalty = "MCP"</code> or <code>penalty = "SCAD"</code> (default is 3).</p>
</td></tr>
<tr><td><code id="cureem_+3A_gamma.lat">gamma.lat</code></td>
<td>
<p>numeric value for the penalization parameter <code class="reqn">\gamma</code> for variables in the latency portion of the model when <code>penalty = "MCP"</code> or <code>penalty = "SCAD"</code> (default is 3).</p>
</td></tr>
<tr><td><code id="cureem_+3A_...">...</code></td>
<td>
<p>additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>b_path</code></td>
<td>
<p>Matrix representing the solution path of the coefficients in the incidence portion of the model. Row is step and column is variable.</p>
</td></tr>
<tr><td><code>beta_path</code></td>
<td>
<p>Matrix representing the solution path of lthe coefficients in the latency portion of the model. Row is step and column is variable.</p>
</td></tr>
<tr><td><code>b0_path</code></td>
<td>
<p>Vector representing the solution path of the intercept in the incidence portion of the model.</p>
</td></tr>
<tr><td><code>logLik.inc</code></td>
<td>
<p>Vector representing the expected penalized complete-data log-likelihood for the incidence portion of the model for each step in the solution path.</p>
</td></tr>
<tr><td><code>logLik.lat</code></td>
<td>
<p>Vector representing the expected penalized complete-data log-likelihood for the latency portion of the model for each step in the solution path.</p>
</td></tr>
<tr><td><code>x.incidence</code></td>
<td>
<p>Matrix representing the design matrix of the incidence predictors.</p>
</td></tr>
<tr><td><code>x.latency</code></td>
<td>
<p>Matrix representing the design matrix of the latency predictors.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>Vector representing the survival object response as returned by the <code>Surv</code> function </p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>Character string indicating the type of regression model used for the latency portion of mixture cure model (&quot;weibull&quot; or &quot;exponential&quot;).</p>
</td></tr>
<tr><td><code>scale</code></td>
<td>
<p>Logical value indicating whether the predictors were centered and scaled.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>Character string indicating the EM alogoritm was used in fitting the mixture cure model.</p>
</td></tr>
<tr><td><code>rate_path</code></td>
<td>
<p>Vector representing the solution path of the rate parameter for the Weibull or exponential density in the latency portion of the model.</p>
</td></tr>
<tr><td><code>alpha_path</code></td>
<td>
<p>Vector representing the solution path of the shape parameter for the Weibull density in the latency portion of the model.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>the matched call.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Archer, K. J., Fu, H., Mrozek, K., Nicolet, D., Mims, A. S., Uy, G. L., Stock, W., Byrd, J. C., Hiddemann, W., Braess, J., Spiekermann, K., Metzeler, K. H., Herold, T., Eisfeld, A.-K. (2024) Identifying long-term survivors and those at higher or lower risk of relapse among patients with cytogenetically normal acute myeloid leukemia using a high-dimensional mixture cure model. <em>Journal of Hematology &amp; Oncology</em>, <b>17</b>:28.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cv_cureem">cv_cureem</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 80, J = 100, nTrue = 10, A = 1.8)
training &lt;- temp$Training
fit &lt;- cureem(Surv(Time, Censor) ~ ., data = training, x.latency = training,
                 model = "cox", penalty = "lasso",
                 lambda.inc = 0.1, lambda.lat = 0.1, gamma.inc = 6, gamma.lat = 10)
</code></pre>

<hr>
<h2 id='curegmifs'>Fit penalized parametric mixture cure model using the GMIFS algorithm</h2><span id='topic+curegmifs'></span>

<h3>Description</h3>

<p>Fits a penalized Weibull or exponential mixture cure model using the generalized monotone incremental forward stagewise (GMIFS) algorithm and yields solution paths for parameters in the incidence and latency portions of the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>curegmifs(
  formula,
  data,
  subset,
  x.latency = NULL,
  model = "weibull",
  penalty.factor.inc = NULL,
  penalty.factor.lat = NULL,
  epsilon = 0.001,
  thresh = 1e-05,
  scale = TRUE,
  maxit = 10000,
  inits = NULL,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="curegmifs_+3A_formula">formula</code></td>
<td>
<p>an object of class &quot;<code>formula</code>&quot; (or one that can be coerced to that class): a symbolic description of the model to be fitted. The response must be a survival object as returned by the <code>Surv</code> function while the variables on the right side of the formula are the covariates that are included in the incidence portion of the model.</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_data">data</code></td>
<td>
<p>a data.frame in which to interpret the variables named in the <code>formula</code> or in the <code>subset</code> argument.</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_subset">subset</code></td>
<td>
<p>an optional expression indicating which subset of observations to be used in the fitting process, either a numeric or factor variable should be used in subset, not a character variable. All observations are included by default.</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_x.latency">x.latency</code></td>
<td>
<p>specifies the variables to be included in the latency portion of the model and can be either a matrix of predictors, a model formula with the right hand side specifying the latency variables, or the same data.frame passed to the <code>data</code> parameter. Note that when using the model formula syntax for <code>x.latency</code> it cannot handle <code>x.latency = ~ .</code>.</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_model">model</code></td>
<td>
<p>type of regression model to use for the latency portion of mixture cure model. Can be &quot;weibull&quot; or &quot;exponential&quot;; default is &quot;weibull&quot;.</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_penalty.factor.inc">penalty.factor.inc</code></td>
<td>
<p>vector of binary indicators representing the penalty to apply to each incidence coefficient: 0 implies no shrinkage and 1 implies shrinkage. If not supplied, 1 is applied to all incidence variables.</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_penalty.factor.lat">penalty.factor.lat</code></td>
<td>
<p>vector of binary indicators representing the penalty to apply to each latency coefficient: 0 implies no shrinkage and 1 implies shrinkage. If not supplied, 1 is applied to all latency variables.</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_epsilon">epsilon</code></td>
<td>
<p>small numeric value reflecting the incremental value used to update a coefficient at a given step (default is 0.001).</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_thresh">thresh</code></td>
<td>
<p>small numeric value. The iterative process stops when the differences between successive expected penalized complete-data log-likelihoods for both incidence and latency components are less than this specified level of tolerance (default is 10^-5).</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_scale">scale</code></td>
<td>
<p>logical, if TRUE the predictors are centered and scaled.</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_maxit">maxit</code></td>
<td>
<p>integer specifying the maximum number of steps to run in the iterative algorithm (default is 10^4).</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_inits">inits</code></td>
<td>
<p>an optional list specifiying the initial value for the incidence intercept (<code>itct</code>), a numeric vector for the unpenalized incidence coefficients (<code>b_u</code>), and a numeric vector for unpenalized latency coefficients (<code>beta_u</code>), a numeric value for the rate parameter (<code>lambda</code>), and a numeric value for the shape parameter (<code>alpha</code>) when <code>model = "weibull"</code>. If not supplied or improperly supplied, initialization is automatically provided by the function.</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_verbose">verbose</code></td>
<td>
<p>logical, if TRUE running information is printed to the console (default is FALSE).</p>
</td></tr>
<tr><td><code id="curegmifs_+3A_...">...</code></td>
<td>
<p>additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>b_path</code></td>
<td>
<p>Matrix representing the solution path of the coefficients in the incidence portion of the model. Row is step and column is variable.</p>
</td></tr>
<tr><td><code>beta_path</code></td>
<td>
<p>Matrix representing the solution path of lthe coefficients in the latency portion of the model. Row is step and column is variable.</p>
</td></tr>
<tr><td><code>b0_path</code></td>
<td>
<p>Vector representing the solution path of the intercept in the incidence portion of the model.</p>
</td></tr>
<tr><td><code>rate_path</code></td>
<td>
<p>Vector representing the solution path of the rate parameter for the Weibull or exponential density in the latency portion of the model.</p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>Vector representing the log-likelihood for each step in the solution path.</p>
</td></tr>
<tr><td><code>x.incidence</code></td>
<td>
<p>Matrix representing the design matrix of the incidence predictors.</p>
</td></tr>
<tr><td><code>x.latency</code></td>
<td>
<p>Matrix representing the design matrix of the latency predictors.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>Vector representing the survival object response as returned by the <code>Surv</code> function </p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>Character string indicating the type of regression model used for the latency portion of mixture cure model (&quot;weibull&quot; or &quot;exponential&quot;).</p>
</td></tr>
<tr><td><code>scale</code></td>
<td>
<p>Logical value indicating whether the predictors were centered and scaled.</p>
</td></tr>
<tr><td><code>alpha_path</code></td>
<td>
<p>Vector representing the solution path of the shape parameter for the Weibull density in the latency portion of the model.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>the matched call.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Fu, H., Nicolet, D., Mrozek, K., Stone, R. M., Eisfeld, A. K., Byrd, J. C., Archer, K. J. (2022) Controlled variable selection in Weibull mixture cure models for high-dimensional data. <em>Statistics in Medicine</em>, <b>41</b>(22), 4340&ndash;4366.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cv_curegmifs">cv_curegmifs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 100, J = 10, nTrue = 10, A = 1.8)
training &lt;- temp$Training

fit &lt;- curegmifs(Surv(Time, Censor) ~ .,
         data = training, x.latency = training,
         model = "weibull", thresh = 1e-4, maxit = 2000, epsilon = 0.01,
         verbose = FALSE)
</code></pre>

<hr>
<h2 id='cv_cureem'>Fit penalized mixture cure model using the E-M algorithm with cross-validation for parameter tuning</h2><span id='topic+cv_cureem'></span>

<h3>Description</h3>

<p>Fits a penalized parametric and semi-parametric mixture cure model (MCM) using the E-M algorithm with with k-fold cross-validation for parameter tuning. The lasso (L1), MCP and SCAD penalty are supported for the Cox MCM while only lasso is currently supported for parametric MCMs. When FDR controlled variable selection is used, the model-X knockoffs method is applied and indices of selected variables are returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv_cureem(
  formula,
  data,
  subset,
  x.latency = NULL,
  model = "cox",
  penalty = "lasso",
  penalty.factor.inc = NULL,
  penalty.factor.lat = NULL,
  fdr.control = FALSE,
  fdr = 0.2,
  grid.tuning = FALSE,
  thresh = 0.001,
  scale = TRUE,
  maxit = NULL,
  inits = NULL,
  lambda.inc.list = NULL,
  lambda.lat.list = NULL,
  nlambda.inc = NULL,
  nlambda.lat = NULL,
  gamma.inc = 3,
  gamma.lat = 3,
  lambda.min.ratio.inc = 0.1,
  lambda.min.ratio.lat = 0.1,
  n_folds = 5,
  measure.inc = "c",
  one.se = FALSE,
  cure_cutoff = 5,
  parallel = FALSE,
  seed = NULL,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cv_cureem_+3A_formula">formula</code></td>
<td>
<p>an object of class &quot;<code>formula</code>&quot; (or one that can be coerced to that class): a symbolic description of the model to be fitted. The response must be a survival object as returned by the <code>Surv</code> function while the variables on the right side of the formula are the covariates that are included in the incidence portion of the model.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_data">data</code></td>
<td>
<p>a data.frame in which to interpret the variables named in the <code>formula</code> or in the <code>subset</code> argument.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_subset">subset</code></td>
<td>
<p>an optional expression indicating which subset of observations to be used in the fitting process, either a numeric or factor variable should be used in subset, not a character variable. All observations are included by default.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_x.latency">x.latency</code></td>
<td>
<p>specifies the variables to be included in the latency portion of the model and can be either a matrix of predictors, a model formula with the right hand side specifying the latency variables, or the same data.frame passed to the <code>data</code> parameter. Note that when using the model formula syntax for <code>x.latency</code> it cannot handle <code>x.latency = ~ .</code>.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_model">model</code></td>
<td>
<p>type of regression model to use for the latency portion of mixture cure model. Can be &quot;cox&quot;, &quot;weibull&quot;, or &quot;exponential&quot; (default is &quot;cox&quot;).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_penalty">penalty</code></td>
<td>
<p>type of penalty function. Can be &quot;lasso&quot;, &quot;MCP&quot;, or &quot;SCAD&quot; (default is &quot;lasso&quot;).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_penalty.factor.inc">penalty.factor.inc</code></td>
<td>
<p>vector of binary indicators representing the penalty to apply to each incidence coefficient: 0 implies no shrinkage and 1 implies shrinkage. If not supplied, 1 is applied to all incidence variables.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_penalty.factor.lat">penalty.factor.lat</code></td>
<td>
<p>vector of binary indicators representing the penalty to apply to each latency coefficient: 0 implies no shrinkage and 1 implies shrinkage. If not supplied, 1 is applied to all latency variables.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_fdr.control">fdr.control</code></td>
<td>
<p>logical, if TRUE, model-X knockoffs are used for FDR-controlled variable selection and indices of selected variables are returned (default is FALSE).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_fdr">fdr</code></td>
<td>
<p>numeric value in (0, 1) range specifying the target FDR level to use for variable selection when <code>fdr.control=TRUE</code> (default is 0.2).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_grid.tuning">grid.tuning</code></td>
<td>
<p>logical, if TRUE a 2-D grid tuning approach is used to select the optimal pair of <code class="reqn">\lambda_b</code> and <code class="reqn">\lambda_{\beta}</code> penalty parameters for the incidence and latency portions of the model, respectively. Otherwise the <code class="reqn">\lambda_b</code> and <code class="reqn">\lambda_{\beta}</code> are selected from a 1-D sequence and are equal to one another (default is FALSE).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_thresh">thresh</code></td>
<td>
<p>small numeric value. The iterative process stops when the differences between successive expected penalized complete-data log-likelihoods for both incidence and latency components are less than this specified level of tolerance (default is 10^-3).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_scale">scale</code></td>
<td>
<p>logical, if TRUE the predictors are centered and scaled.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_maxit">maxit</code></td>
<td>
<p>maximum number of passes over the data for each lambda. If not specified, 100 is applied when <code>penalty = "lasso"</code> and 1000 is applied when <code>penalty = "MCP"</code> or <code>penalty = "SCAD"</code>.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_inits">inits</code></td>
<td>
<p>an optional list specifiying the initial value for the incidence intercept (<code>itct</code>), a numeric vector for the unpenalized incidence coefficients (<code>b_u</code>), and a numeric vector for unpenalized latency coefficients (<code>beta_u</code>).  For parametric models, it should also include a numeric value for the rate parameter (<code>lambda</code>) when <code>model = "weibull"</code> or <code>model = "exponential"</code>, and a numeric value for the shape parameter (<code>alpha</code>) when <code>model = "weibull"</code>. When <code>model = "cox"</code>, it should also include a numeric vector for the latency survival probabilities <code class="reqn">S_u(t_i|w_i)</code> for i=1,...,N (<code>survprob</code>). Penalized coefficients are initialized to zero. If <code>inits</code> is not specified or improperly specified, initialization is automatically provided by the function.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_lambda.inc.list">lambda.inc.list</code></td>
<td>
<p>a numeric vector used to search for the optimal <code class="reqn">\lambda_b</code> tuning parameter. If not supplied, the function computes a <code class="reqn">\lambda_b</code> sequence based on <code>nlambda.inc</code> and <code>lambda.min.ratio.inc</code>. If <code>grid.tuning=FALSE</code>, the same sequence should be used for both <code class="reqn">\lambda_b</code> and <code class="reqn">\lambda_{\beta}</code>.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_lambda.lat.list">lambda.lat.list</code></td>
<td>
<p>a numeric vector used to search for the optimal <code class="reqn">\lambda_{\beta}</code> tuning parameter. If not supplied, the function computes a <code class="reqn">\lambda_{\beta}</code> sequence based on <code>nlambda.lat</code> and <code>lambda.min.ratio.lat</code>. If <code>grid.tuning=FALSE</code>, the same sequence should be used for both <code class="reqn">\lambda_b</code> and <code class="reqn">\lambda_{\beta}</code>.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_nlambda.inc">nlambda.inc</code></td>
<td>
<p>an integer specifying the number of values to search for the optimal <code class="reqn">\lambda_b</code> tuning parameter; default is 10 if <code>grid.tuning=TRUE</code> and 50 otherwise.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_nlambda.lat">nlambda.lat</code></td>
<td>
<p>an integer specifying the number of values to search for the optimal <code class="reqn">\lambda_{\beta}</code> tuning parameter; default is 10 if <code>grid.tuning=TRUE</code> and 50 otherwise.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_gamma.inc">gamma.inc</code></td>
<td>
<p>numeric value for the penalization parameter <code class="reqn">\gamma</code> for variables in the incidence portion of the model when <code>penalty = "MCP"</code> or <code>penalty = "SCAD"</code> (default is 3).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_gamma.lat">gamma.lat</code></td>
<td>
<p>numeric value for the penalization parameter <code class="reqn">\gamma</code> for variables in the latency portion of the model when <code>penalty = "MCP"</code> or <code>penalty = "SCAD"</code> (default is 3).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_lambda.min.ratio.inc">lambda.min.ratio.inc</code></td>
<td>
<p>numeric value in (0,1) representing the smallest value for <code class="reqn">\lambda_b</code> as a fraction of <code>lambda.max.inc</code>, the data-derived entry value at which essentially all penalized variables in the incidence portion of the model have a coefficient estimate of 0 (default is 0.1).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_lambda.min.ratio.lat">lambda.min.ratio.lat</code></td>
<td>
<p>numeric value in (0.1) representing the smallest value for <code class="reqn">\lambda_{\beta}</code> as a fraction of <code>lambda.max.lat</code>, the data-derived entry value at essentially all penalized variables in the latency portion of the model have a coefficient estimate of 0 (default is 0.1).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_n_folds">n_folds</code></td>
<td>
<p>an integer specifying the number of folds for the k-fold cross-valiation procedure (default is 5).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_measure.inc">measure.inc</code></td>
<td>
<p>character string specifying the evaluation criterion used in selecting the optimal <code class="reqn">\lambda_b</code>. Can be &quot;c&quot; or &quot;auc&quot;; default is &quot;c&quot;. If <code>measure.inc="c"</code>, the C-statistic using the cure status weighting (CSW) method proposed by Asano and Hirakawa (2017) is used to select both <code class="reqn">\lambda_b</code> and <code class="reqn">\lambda_{\beta}</code>. If <code>measure.inc="auc"</code>, the AUC for cure prediction using the mean score imputation (MSI) method proposed by Asano et al. (2014) is used to select <code class="reqn">\lambda_b</code> while the C-statistic with CSW is used for <code class="reqn">\lambda_{\beta}</code>.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_one.se">one.se</code></td>
<td>
<p>logical, if TRUE then the one standard error rule is applied for selecting the optimal parameters. The one standard error rule selects the most parsimonious model having evaluation criterion no more than one standard error worse than that of the best evaluation criterion (default is FALSE).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_cure_cutoff">cure_cutoff</code></td>
<td>
<p>numeric value representing the cutoff time value that represents subjects not experiencing the event by this time are cured. This value is used to produce a proxy for the unobserved cure status when calculating C-statistic and AUC (default is 5 representing 5 years). Users should be careful to note the time scale of their data and adjust this according to the time scale and clinical application.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_parallel">parallel</code></td>
<td>
<p>logical. If TRUE, parallel processing is performed for K-fold CV using <code>foreach</code> and the <span class="pkg">doMC</span> package is required.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_seed">seed</code></td>
<td>
<p>optional integer representing the random seed. Setting the random seed fosters reproducibility of the results.</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_verbose">verbose</code></td>
<td>
<p>logical, if TRUE running information is printed to the console (default is FALSE).</p>
</td></tr>
<tr><td><code id="cv_cureem_+3A_...">...</code></td>
<td>
<p>additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>b0</code></td>
<td>
<p>Estimated intercept for the incidence portion of the model.</p>
</td></tr>
<tr><td><code>b</code></td>
<td>
<p>Estimated coefficients for the incidence portion of the model.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>Estimated coefficients for the latency portion of the model.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>Estimated shape parameter if the Weibull model is fit.</p>
</td></tr>
<tr><td><code>rate</code></td>
<td>
<p>Estimated rate parameter if the Weibull or exponential model is fit.</p>
</td></tr>
<tr><td><code>logLik.inc</code></td>
<td>
<p>Expected penalized complete-data log-likelihood for the incidence portion of the model.</p>
</td></tr>
<tr><td><code>logLik.lat</code></td>
<td>
<p>Expected penalized complete-data log-likelihood for the latency portion of the model.</p>
</td></tr>
<tr><td><code>selected.lambda.inc</code></td>
<td>
<p>Value of <code class="reqn">\lambda_b</code> selected using cross-validation. NULL when fdr.control is TRUE.</p>
</td></tr>
<tr><td><code>selected.lambda.lat</code></td>
<td>
<p>Value of <code class="reqn">\lambda_{\beta}</code> selected using cross-validation. NULL when fdr.control is TRUE.</p>
</td></tr>
<tr><td><code>max.c</code></td>
<td>
<p>Maximum C-statistic achieved.</p>
</td></tr>
<tr><td><code>max.auc</code></td>
<td>
<p>Maximum AUC for cure prediction achieved; only output when <code>measure.inc="auc"</code>.</p>
</td></tr>
<tr><td><code>selected.index.inc</code></td>
<td>
<p>Indices of selected variables for the incidence portion of the model when <code>fdr.control=TRUE</code>. If no variables are selected, <code>int(0)</code> will be returned.</p>
</td></tr>
<tr><td><code>selected.index.lat</code></td>
<td>
<p>Indices of selected variables for the latency portion of the model when <code>fdr.control=TRUE</code>. If no variables are selected, <code>int(0)</code> will be returned.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>the matched call.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Archer, K. J., Fu, H., Mrozek, K., Nicolet, D., Mims, A. S., Uy, G. L., Stock, W., Byrd, J. C., Hiddemann, W., Braess, J., Spiekermann, K., Metzeler, K. H., Herold, T., Eisfeld, A.-K. (2024) Identifying long-term survivors and those at higher or lower risk of relapse among patients with cytogenetically normal acute myeloid leukemia using a high-dimensional mixture cure model. <em>Journal of Hematology &amp; Oncology</em>, <b>17</b>:28.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cureem">cureem</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 200, J = 25, nTrue = 5, A = 1.8)
training &lt;- temp$Training
fit.cv &lt;- cv_cureem(Surv(Time, Censor) ~ ., data = training,
                 x.latency = training, fdr.control = FALSE,
                 grid.tuning = FALSE, nlambda.inc = 10, nlambda.lat = 10,
                 n_folds = 2, seed = 23, verbose = TRUE)
fit.cv.fdr &lt;- cv_cureem(Surv(Time, Censor) ~ ., data = training,
                 x.latency = training, model = "weibull", penalty = "lasso",
                 fdr.control = TRUE, grid.tuning = FALSE, nlambda.inc = 10,
                 nlambda.lat = 10, n_folds = 2, seed = 23, verbose = TRUE)
</code></pre>

<hr>
<h2 id='cv_curegmifs'>Fit a penalized parametric mixture cure model using the GMIFS algorithm with cross-validation for model selection</h2><span id='topic+cv_curegmifs'></span>

<h3>Description</h3>

<p>Fits a penalized Weibull or exponential mixture cure model using the generalized monotone incremental forward stagewise (GMIFS) algorithm with k-fold cross-validation to select the optimal iteration step along the solution path. When FDR controlled variable selection is used, the model-X knockoffs method is applied and indices of selected variables are returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv_curegmifs(
  formula,
  data,
  subset,
  x.latency = NULL,
  model = "weibull",
  penalty.factor.inc = NULL,
  penalty.factor.lat = NULL,
  fdr.control = FALSE,
  fdr = 0.2,
  epsilon = 0.001,
  thresh = 1e-05,
  scale = TRUE,
  maxit = 10000,
  inits = NULL,
  n_folds = 5,
  measure.inc = "c",
  one.se = FALSE,
  cure_cutoff = 5,
  parallel = FALSE,
  seed = NULL,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cv_curegmifs_+3A_formula">formula</code></td>
<td>
<p>an object of class &quot;<code>formula</code>&quot; (or one that can be coerced to that class): a symbolic description of the model to be fitted. The response must be a survival object as returned by the <code>Surv</code> function while the variables on the right side of the formula are the covariates that are included in the incidence portion of the model.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_data">data</code></td>
<td>
<p>a data.frame in which to interpret the variables named in the <code>formula</code> or in the <code>subset</code> argument.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_subset">subset</code></td>
<td>
<p>an optional expression indicating which subset of observations to be used in the fitting process, either a numeric or factor variable should be used in subset, not a character variable. All observations are included by default.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_x.latency">x.latency</code></td>
<td>
<p>specifies the variables to be included in the latency portion of the model and can be either a matrix of predictors, a model formula with the right hand side specifying the latency variables, or the same data.frame passed to the <code>data</code> parameter. Note that when using the model formula syntax for <code>x.latency</code> it cannot handle <code>x.latency = ~ .</code>.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_model">model</code></td>
<td>
<p>type of regression model to use for the latency portion of mixture cure model. Can be &quot;weibull&quot; or &quot;exponential&quot;; default is &quot;weibull&quot;.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_penalty.factor.inc">penalty.factor.inc</code></td>
<td>
<p>vector of binary indicators representing the penalty to apply to each incidence coefficient: 0 implies no shrinkage and 1 implies shrinkage. If not supplied, 1 is applied to all incidence variables.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_penalty.factor.lat">penalty.factor.lat</code></td>
<td>
<p>vector of binary indicators representing the penalty to apply to each latency coefficient: 0 implies no shrinkage and 1 implies shrinkage. If not supplied, 1 is applied to all latency variables.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_fdr.control">fdr.control</code></td>
<td>
<p>logical, if TRUE, model-X knockoffs are used for FDR-controlled variable selection and indices of selected variables are returned (default is FALSE).</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_fdr">fdr</code></td>
<td>
<p>numeric value in (0, 1) range specifying the target FDR level to use for variable selection when <code>fdr.control=TRUE</code> (default is 0.2).</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_epsilon">epsilon</code></td>
<td>
<p>small numeric value reflecting incremental value used to update a coefficient at a given step (default is 0.001).</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_thresh">thresh</code></td>
<td>
<p>small numeric value. The iterative process stops when the differences between successive expected penalized complete-data log-likelihoods for both incidence and latency components are less than this specified level of tolerance (default is 10^-5).</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_scale">scale</code></td>
<td>
<p>logical, if TRUE the predictors are centered and scaled.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_maxit">maxit</code></td>
<td>
<p>integer specifying the maximum number of steps to run in the iterative algorithm (default is 10^4).</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_inits">inits</code></td>
<td>
<p>an optional list specifiying the initial value for the incidence intercept (<code>itct</code>), a numeric vector for the unpenalized incidence coefficients (<code>b_u</code>), and a numeric vector for unpenalized latency coefficients (<code>beta_u</code>), a numeric value for the rate parameter (<code>lambda</code>), and a numeric value for the shape parameter (<code>alpha</code>) when <code>model = "weibull"</code>. If not supplied or improperly supplied, initialization is automatically provided by the function.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_n_folds">n_folds</code></td>
<td>
<p>an integer specifying the number of folds for the k-fold cross-valiation procedure (default is 5).</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_measure.inc">measure.inc</code></td>
<td>
<p>character string specifying the evaluation criterion used in selecting the optimal <code class="reqn">\lambda_b</code>. Can be &quot;c&quot; or &quot;auc&quot;; default is &quot;c&quot;. If <code>measure.inc="c"</code>, the C-statistic using the cure status weighting (CSW) method proposed by Asano and Hirakawa (2017) is used to select both <code class="reqn">\lambda_b</code> and <code class="reqn">\lambda_{\beta}</code>. If <code>measure.inc="auc"</code>, the AUC for cure prediction using the mean score imputation (MSI) method proposed by Asano et al. (2014) is used to select <code class="reqn">\lambda_b</code> while the C-statistic with CSW is used for <code class="reqn">\lambda_{\beta}</code>.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_one.se">one.se</code></td>
<td>
<p>logical, if TRUE then the one standard error rule is applied for selecting the optimal parameters. The one standard error rule selects the most parsimonious model having evaluation criterion no more than one standard error worse than that of the best evaluation criterion (default is FALSE).</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_cure_cutoff">cure_cutoff</code></td>
<td>
<p>numeric value representing the cutoff time value that represents subjects not experiencing the event by this time are cured. This value is used to produce a proxy for the unobserved cure status when calculating C-statistic and AUC (default is 5 representing 5 years). Users should be careful to note the time scale of their data and adjust this according to the time scale and clinical application.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_parallel">parallel</code></td>
<td>
<p>logical. If TRUE, parallel processing is performed for K-fold CV using <code>foreach</code> and the <span class="pkg">doMC</span> package is required.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_seed">seed</code></td>
<td>
<p>optional integer representing the random seed. Setting the random seed fosters reproducibility of the results.</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_verbose">verbose</code></td>
<td>
<p>logical, if TRUE running information is printed to the console (default is FALSE).</p>
</td></tr>
<tr><td><code id="cv_curegmifs_+3A_...">...</code></td>
<td>
<p>additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>b0</code></td>
<td>
<p>Estimated intercept for the incidence portion of the model.</p>
</td></tr>
<tr><td><code>b</code></td>
<td>
<p>Estimated coefficients for the incidence portion of the model.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>Estimated coefficients for the latency portion of the model.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>Estimated shape parameter if the Weibull model is fit.</p>
</td></tr>
<tr><td><code>rate</code></td>
<td>
<p>Estimated rate parameter if the Weibull or exponential model is fit.</p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>Log-likelihood value.</p>
</td></tr>
<tr><td><code>selected.step.inc</code></td>
<td>
<p>Iteration step selected for the incidence portion of the model using cross-validation. NULL when fdr.control is TRUE.</p>
</td></tr>
<tr><td><code>selected.step.lat</code></td>
<td>
<p>Iteration step selected for the latency portion of the model using cross-validation. NULL when fdr.control is TRUE.</p>
</td></tr>
<tr><td><code>max.c</code></td>
<td>
<p>Maximum C-statistic achieved</p>
</td></tr>
<tr><td><code>max.auc</code></td>
<td>
<p>Maximum AUC for cure prediction achieved; only output when <code>measure.inc="auc"</code>.</p>
</td></tr>
<tr><td><code>selected.index.inc</code></td>
<td>
<p>Indices of selected variables for the incidence portion of the model when <code>fdr.control=TRUE</code>. If none selected, <code>int(0)</code> will be returned.</p>
</td></tr>
<tr><td><code>selected.index.lat</code></td>
<td>
<p>Indices of selected variables for the latency portion of the model when <code>fdr.control=TRUE</code>. If none selected, <code>int(0)</code> will be returned.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>the matched call.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Fu, H., Nicolet, D., Mrozek, K., Stone, R. M., Eisfeld, A. K., Byrd, J. C., Archer, K. J. (2022) Controlled variable selection in Weibull mixture cure models for high-dimensional data. <em>Statistics in Medicine</em>, <b>41</b>(22), 4340&ndash;4366.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+curegmifs">curegmifs</a></code>
</p>
<p><code><a href="#topic+curegmifs">curegmifs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(123)
temp &lt;- generate_cure_data(N = 100, J = 15, nTrue = 3, A = 1.8, rho = 0.2)
training &lt;- temp$Training

fit.cv &lt;- cv_curegmifs(Surv(Time, Censor) ~ ., data = training,
                      x.latency = training, fdr.control = FALSE,
                      maxit = 450, epsilon = 0.01,
                      n_folds = 2, seed = 23, verbose = TRUE)
</code></pre>

<hr>
<h2 id='generate_cure_data'>Simulate data under a mixture cure model</h2><span id='topic+generate_cure_data'></span>

<h3>Description</h3>

<p>Simulate data under a mixture cure model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_cure_data(
  N = 400,
  J = 500,
  nonp = 2,
  train.prop = 3/4,
  nTrue = 10,
  A = 1,
  rho = 0.5,
  itct_mean = 0.5,
  cens_ub = 20,
  alpha = 1,
  lambda = 2,
  same_signs = FALSE,
  model = "weibull"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_cure_data_+3A_n">N</code></td>
<td>
<p>an integer denoting the total sample size.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_j">J</code></td>
<td>
<p>an integer denoting the number of penalized predictors which is the same for both the incidence and latency portions of the model.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_nonp">nonp</code></td>
<td>
<p>an integer less than J denoting the number of unpenalized predictors (which is the same for both the incidence and latency portions of the model.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_train.prop">train.prop</code></td>
<td>
<p>a numeric value in 0, 1 representing the fraction of N to be used in forming the Training dataset.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_ntrue">nTrue</code></td>
<td>
<p>an integer denoting the number of variables truly associated with the outcome (i.e., the number of covariates with nonzero parameter values) among the penalized predictors.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_a">A</code></td>
<td>
<p>a numeric value denoting the effect size which is the same for both the incidence and latency portions of the model.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_rho">rho</code></td>
<td>
<p>a numeric value in 0, 1 representing the correlation between adjacent covariates in the same block. See details below.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_itct_mean">itct_mean</code></td>
<td>
<p>a numeric value representing the expectation of the incidence intercept which controls the cure rate.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_cens_ub">cens_ub</code></td>
<td>
<p>a numeric value representing the upper bound on the censoring time distribition which follows a uniform distribution on 0, <code>cens_ub</code>.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_alpha">alpha</code></td>
<td>
<p>a numeric value representing the shape parameter in the Weibull density.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_lambda">lambda</code></td>
<td>
<p>a numeric value representing the rate parameter in the Weibull density.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_same_signs">same_signs</code></td>
<td>
<p>logical, if TRUE the incidence and latency coefficients have the same signs.</p>
</td></tr>
<tr><td><code id="generate_cure_data_+3A_model">model</code></td>
<td>
<p>type of regression model to use for the latency portion of mixture cure model. Can be &quot;weibull&quot;, &quot;GG&quot;, &quot;Gompertz&quot;, &quot;nonparametric&quot;, or &quot;GG_baseline&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Training</code></td>
<td>
<p>Training data.frame which includes Time, Censor, and covariates.</p>
</td></tr>
<tr><td><code>Testing</code></td>
<td>
<p>Testing data.frame which includes Time, Censor, and covariates.</p>
</td></tr>
<tr><td><code>parameters</code></td>
<td>
<p>A list including: the indices of true incidence signals (<code>nonzero_b</code>), indices of true latency signals (<code>nonzero_beta</code>), unpenalized incidence parameter values (<code>b_u</code>), unpenalized latency parameter values (<code>beta_u</code>), parameter values for the true incidence signals among penalized covariates (<code>b_p_nz</code>), parameter values for the true latency signals among penalized covariates (<code>beta_p_nz</code>), parameter value for the incidence intercept (<code>itct</code>)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
data &lt;- generate_cure_data(N = 200, J = 50, nTrue = 10, A = 1.8, rho = 0.2)
training &lt;- data$Training
testing &lt;- data$Testing
fit &lt;- cureem(Surv(Time, Censor) ~ ., data = training,
              x.latency = training, model = "cox", penalty = "lasso",
              lambda.inc = 0.05, lambda.lat = 0.05,
              gamma.inc = 6, gamma.lat = 10)
</code></pre>

<hr>
<h2 id='nonzerocure_test'>Non-parametric pest for a non-zero cured fraction</h2><span id='topic+nonzerocure_test'></span>

<h3>Description</h3>

<p>Tests the null hypothesis that the proportion of observations susceptible to the event = 1 against the alternative that the proportion of observations susceptible to the event is &lt; 1. If the null hypothesis is rejected, there is a significant cured fraction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nonzerocure_test(object, Reps = 1000, seed = NULL, plot = FALSE, B = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nonzerocure_test_+3A_object">object</code></td>
<td>
<p>a <code>survfit</code> object.</p>
</td></tr>
<tr><td><code id="nonzerocure_test_+3A_reps">Reps</code></td>
<td>
<p>number of simulations on which to base the p-value (default = 1000).</p>
</td></tr>
<tr><td><code id="nonzerocure_test_+3A_seed">seed</code></td>
<td>
<p>optional random seed.</p>
</td></tr>
<tr><td><code id="nonzerocure_test_+3A_plot">plot</code></td>
<td>
<p>logical. If TRUE a histogram of the estimated susceptible proportions over all simulations is produced.</p>
</td></tr>
<tr><td><code id="nonzerocure_test_+3A_b">B</code></td>
<td>
<p>optional. If specified the maximum observed time for the uniform distribution for generating the censoring times. If not specified, an exponential model is used for generating the censoring times (default).</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>proportion_susceptible</code></td>
<td>
<p>estimated proportion of susceptibles</p>
</td></tr>
<tr><td><code>proportion_cured</code></td>
<td>
<p>estimated proportion of those cured</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>p-value testing the null hypothesis that the proportion of susceptibles = 1 (cured fraction = 0) against the alternative that the proportion of susceptibles &lt; 1 (non-zero cured fraction)</p>
</td></tr>
<tr><td><code>time_95_percent_of_events</code></td>
<td>
<p>estimated time at which 95% of events should have occurred</p>
</td></tr>
</table>


<h3>References</h3>

<p>Maller, R. A. and Zhou, X. (1996) <em>Survival Analysis with Long-Term Survivors</em>. John Wiley &amp; Sons.
</p>


<h3>See Also</h3>

<p><code><a href="survival.html#topic+survfit">survfit</a></code>, <code><a href="#topic+cure_estimate">cure_estimate</a></code>, <code><a href="#topic+sufficient_fu_test">sufficient_fu_test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 100, J = 10, nTrue = 10, A = 1.8)
training &lt;- temp$Training
km.fit &lt;- survfit(Surv(Time, Censor) ~ 1, data = training)
nonzerocure_test(km.fit)
</code></pre>

<hr>
<h2 id='plot.mixturecure'>Plot fitted mixture cure model</h2><span id='topic+plot.mixturecure'></span>

<h3>Description</h3>

<p>This function plots either the coefficient path, the AIC, the cAIC, the BIC, or the log-likelihood for a fitted <code>curegmifs</code> or <code>cureem</code> object. This function produces a lollipop plot of the coefficient estimates for a fitted <code>cv_curegmifs</code> or <code>cv_cureem</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'mixturecure'
plot(x, type = "trace", xlab = NULL, ylab = NULL, main = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.mixturecure_+3A_x">x</code></td>
<td>
<p>a <code>mixturecure</code> object resulting from <code>curegmifs</code> or <code>cureem</code>, <code>cv_curegmifs</code> or <code>cv_cureem</code>.</p>
</td></tr>
<tr><td><code id="plot.mixturecure_+3A_type">type</code></td>
<td>
<p>default is <code>"trace"</code> which plots the coefficient path for the fitted object. Also available are <code>"AIC"</code>, <code>"cAIC"</code>, <code>"mAIC"</code>, <code>"BIC"</code>, <code>"mBIC"</code>, <code>"EBIC"</code>, and <code>"logLik"</code>. This option has no effect for objects fit using <code>cv_curegmifs</code> or <code>cv_cureem</code>.</p>
</td></tr>
<tr><td><code id="plot.mixturecure_+3A_xlab">xlab</code></td>
<td>
<p>a default x-axis label will be used which can be changed by specifying a user-defined x-axis label.</p>
</td></tr>
<tr><td><code id="plot.mixturecure_+3A_ylab">ylab</code></td>
<td>
<p>a default y-axis label will be used which can be changed by specifying a user-defined y-axis label.</p>
</td></tr>
<tr><td><code id="plot.mixturecure_+3A_main">main</code></td>
<td>
<p>a default main title will be used which can be changed by specifying a user-defined main title. This option is not used for <code>cv_curegmifs</code> or <code>cv_cureem</code> fitted objects.</p>
</td></tr>
<tr><td><code id="plot.mixturecure_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>this function has no returned value but is called for its side effects
</p>


<h3>See Also</h3>

<p><code><a href="#topic+curegmifs">curegmifs</a></code>, <code><a href="#topic+cureem">cureem</a></code>, <code><a href="#topic+coef.mixturecure">coef.mixturecure</a></code>, <code><a href="#topic+summary.mixturecure">summary.mixturecure</a></code>, <code><a href="#topic+predict.mixturecure">predict.mixturecure</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 100, J = 10, nTrue = 10, A = 1.8)
training &lt;- temp$Training
fit &lt;- curegmifs(Surv(Time, Censor) ~ .,
                   data = training, x.latency = training,
                   model = "weibull", thresh = 1e-4, maxit = 2000,
                   epsilon = 0.01, verbose = FALSE)
plot(fit)
</code></pre>

<hr>
<h2 id='predict.mixturecure'>Predicted probabilities for susceptibles, linear predictor for latency, and risk class for latency for mixture cure fit</h2><span id='topic+predict.mixturecure'></span>

<h3>Description</h3>

<p>This function returns a list the includes the predicted probabilities for susceptibles as well as the linear predictor for the latency distribution and a dichotomous risk for latency for a <code>curegmifs</code>, <code>cureem</code>, <code>cv_curegmifs</code> or <code>cv_cureem</code> fitted object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'mixturecure'
predict(object, newdata, model.select = "AIC", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.mixturecure_+3A_object">object</code></td>
<td>
<p>a <code>mixturecure</code> object resulting from <code>curegmifs</code>, <code>cureem</code>, <code>cv_curegmifs</code>, <code>cv_cureem</code>.</p>
</td></tr>
<tr><td><code id="predict.mixturecure_+3A_newdata">newdata</code></td>
<td>
<p>an optional data.frame that minimally includes the incidence and/or latency variables to use for predicting the response. If omitted, the training data are used.</p>
</td></tr>
<tr><td><code id="predict.mixturecure_+3A_model.select">model.select</code></td>
<td>
<p>for models fit using <code>curegmifs</code> or <code>cureem</code> any step along the solution path can be selected. The default is <code>model.select = "AIC"</code> which calculates the predicted values using the coefficients from the model having the lowest AIC. Other options are <code>model.select = "mAIC"</code> for the modified AIC,  <code>model.select = "cAIC"</code> for the corrected AIC, <code>model.select = "BIC"</code>, <code>model.select = "mBIC"</code> for the modified BIC, <code>model.select = "EBIC"</code> for the extended BIC, <code>model.select = "logLik"</code> for the step that maximizes the log-likelihood, or any numeric value from the solution path. This option has no effect for objects fit using <code>cv_curegmifs</code> or <code>cv_cureem</code>.</p>
</td></tr>
<tr><td><code id="predict.mixturecure_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>p.uncured</code></td>
<td>
<p> a vector of probabilities from the incidence portion of the fitted model representing the P(uncured).</p>
</td></tr>
<tr><td><code>linear.latency</code></td>
<td>
<p> a vector for the linear predictor from the latency portion of the model.</p>
</td></tr>
<tr><td><code>latency.risk</code></td>
<td>
<p> a dichotomous class representing low (below the median) versus high risk for the latency portion of the model.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+curegmifs">curegmifs</a></code>, <code><a href="#topic+cureem">cureem</a></code>, <code><a href="#topic+coef.mixturecure">coef.mixturecure</a></code>, <code><a href="#topic+summary.mixturecure">summary.mixturecure</a></code>, <code><a href="#topic+plot.mixturecure">plot.mixturecure</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 100, J = 10, nTrue = 10, A = 1.8)
training &lt;- temp$Training
fit &lt;- curegmifs(Surv(Time, Censor) ~ .,
                  data = training, x.latency = training,
                  model = "weibull", thresh = 1e-4, maxit = 2000,
                  epsilon = 0.01, verbose = FALSE)
predict.train &lt;- predict(fit)
names(predict.train)
testing &lt;- temp$Testing
predict.test &lt;- predict(fit, newdata = testing)
</code></pre>

<hr>
<h2 id='print.mixturecure'>Print the contents of a mixture cure fitted object</h2><span id='topic+print.mixturecure'></span>

<h3>Description</h3>

<p>This function prints the names of the list objects from a <code>curegmifs</code>, <code>cureem</code>, <code>cv_cureem</code>, or <code>cv_curegmifs</code> fitted model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'mixturecure'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.mixturecure_+3A_x">x</code></td>
<td>
<p>a <code>mixturecure</code> object resulting from <code>curegmifs</code>, <code>cureem</code>, <code>cv_cureem</code>, or <code>cv_curegmifs</code>.</p>
</td></tr>
<tr><td><code id="print.mixturecure_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>names of the objects in a mixturecure object fit using <code>cureem</code>, <code>curegmifs</code>, <code>cv_cureem</code>, or <code>cv_curegmifs</code>.
</p>


<h3>Note</h3>

<p>The contents of an <code>mixturecure</code> fitted object differ depending upon whether the EM (<code>cureem</code>) or GMIFS (<code>curegmifs</code>) algorithm is used for model fitting. Also, the output differs depending upon whether <code>x.latency</code> is specified in the model (i.e., variables are included in the latency portion of the model fit) or only <code>terms</code> on the right hand side of the equation are included (i.e., variables are included in the incidence portion of the model).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+curegmifs">curegmifs</a></code>, <code><a href="#topic+cureem">cureem</a></code>, <code><a href="#topic+coef.mixturecure">coef.mixturecure</a></code>, <code><a href="#topic+summary.mixturecure">summary.mixturecure</a></code>, <code><a href="#topic+plot.mixturecure">plot.mixturecure</a></code>, <code><a href="#topic+predict.mixturecure">predict.mixturecure</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 100, J = 10, nTrue = 10, A = 1.8)
training &lt;- temp$Training
fit &lt;- curegmifs(Surv(Time, Censor) ~ .,
                       data = training, x.latency = training,
                       model = "weibull", thresh = 1e-4, maxit = 2000,
                       epsilon = 0.01, verbose = FALSE)
print(fit)
</code></pre>

<hr>
<h2 id='sufficient_fu_test'>Test for sufficient follow-up</h2><span id='topic+sufficient_fu_test'></span>

<h3>Description</h3>

<p>Tests for sufficient follow-up using a Kaplan-Meier fitted object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sufficient_fu_test(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sufficient_fu_test_+3A_object">object</code></td>
<td>
<p>a <code>survfit</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>p.value</code></td>
<td>
<p>p-value from testing the null hypothesis that there was not sufficient follow-up against the alternative that there was sufficient follow-up</p>
</td></tr>
<tr><td><code>Nn</code></td>
<td>
<p>total number of events that occurred at time &gt; pmax(0, 2*(last observed event time)-(last observed time)) and &lt; the last observed event time</p>
</td></tr>
<tr><td><code>N</code></td>
<td>
<p>number of observations in the dataset</p>
</td></tr>
</table>


<h3>References</h3>

<p>Maller, R. A. and Zhou, X. (1996) <em>Survival Analysis with Long-Term Survivors</em>. John Wiley &amp; Sons.
</p>


<h3>See Also</h3>

<p><code><a href="survival.html#topic+survfit">survfit</a></code>, <code><a href="#topic+cure_estimate">cure_estimate</a></code>, <code><a href="#topic+nonzerocure_test">nonzerocure_test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 100, J = 10, nTrue = 10, A = 1.8)
training &lt;- temp$Training
km.fit &lt;- survfit(Surv(Time, Censor) ~ 1, data = training)
sufficient_fu_test(km.fit)
</code></pre>

<hr>
<h2 id='summary.mixturecure'>Summarize a Fitted Mixture Cure Object.</h2><span id='topic+summary.mixturecure'></span>

<h3>Description</h3>

<p><code>summary</code> method for a mixturecure object fit using <code>curegmifs</code>, <code>cureem</code>, <code>cv_curegmifs</code>, or <code>cv_cureem</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'mixturecure'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.mixturecure_+3A_object">object</code></td>
<td>
<p>a <code>mixturecure</code> object resulting from <code>curegmifs</code>, <code>cureem</code>, <code>cv_curegmifs</code>, or <code>cv_cureem</code>.</p>
</td></tr>
<tr><td><code id="summary.mixturecure_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>prints the following items extracted from the object fit using <code>curegmifs</code> or <code>cureem</code>: the step and value that maximizes the log-likelihood; the step and value that minimizes the AIC, modified AIC (mAIC), corrected AIC (cAIC), BIC, modified BIC (mBIC), and extended BIC (EBIC). Returns log-likelihood, AIC, and BIC if the object was fit using <code>cv_curegmifs</code> or <code>cv_cureem</code> at the optimal cross-validated values if no FDR control; the number of non-zero incidence and latency variables is returned when cross-validation is used together with FDR control.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+curegmifs">curegmifs</a></code>, <code><a href="#topic+cureem">cureem</a></code>, <code><a href="#topic+coef.mixturecure">coef.mixturecure</a></code>, <code><a href="#topic+plot.mixturecure">plot.mixturecure</a></code>, <code><a href="#topic+predict.mixturecure">predict.mixturecure</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
set.seed(1234)
temp &lt;- generate_cure_data(N = 100, J = 10, nTrue = 10, A = 1.8)
training &lt;- temp$Training
fit &lt;- curegmifs(Surv(Time, Censor) ~ .,
                       data = training, x.latency = training,
                       model = "weibull", thresh = 1e-4, maxit = 2000,
                       epsilon = 0.01, verbose = FALSE)
summary(fit)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
