<!DOCTYPE html><html><head><title>Help for package hdmed</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hdmed}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#med_dat'><p>Mediation Example Dataset</p></a></li>
<li><a href='#mediate_bslmm'><p>Bayesian Sparse Linear Mixed Model</p></a></li>
<li><a href='#mediate_hdma'><p>High-Dimensional Mediation Analysis</p></a></li>
<li><a href='#mediate_hdmm'><p>High-dimensional Multivariate Mediation Analysis with</p>
Principal Directions of Mediation</a></li>
<li><a href='#mediate_hilma'><p>High-Dimensional Linear Mediation Analysis</p></a></li>
<li><a href='#mediate_hima'><p>High-Dimensional Mediation Analysis</p></a></li>
<li><a href='#mediate_lvma'><p>Latent Variable Mediation Analysis</p></a></li>
<li><a href='#mediate_medfix'><p>Mediation Analysis via Fixed Effects Model</p></a></li>
<li><a href='#mediate_pcma'><p>Principal Component Mediation Analysis for High-dimensional Mediators</p></a></li>
<li><a href='#mediate_plasso'><p>Pathway LASSO for Mediation Analysis with High-Dimensional Mediators</p></a></li>
<li><a href='#mediate_spcma'><p>Sparse Principal Component Mediation Analysis for High-Dimensional Mediators</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Methods for Mediation Analysis with High-Dimensional Mediators</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dylan Clark-Boucher &lt;dclarkboucher@fas.harvard.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A suite of functions for performing mediation
    analysis with high-dimensional mediators. In addition to centralizing code 
    from several existing packages for high-dimensional mediation analysis, we 
    provide organized, well-documented functions for a handle of methods 
    which, though programmed their original authors, have not previously been 
    formalized into R packages or been made presentable for public use. The 
    methods we include cover a broad array of approaches and objectives, and are
    described in detail by both our companion manuscript&mdash;"Methods for 
    Mediation Analysis with High-Dimensional DNA Methylation Data: Possible 
    Choices and Comparison"&mdash;and the original publications that proposed them. 
    The specific methods offered by our package include the Bayesian sparse 
    linear mixed model (BSLMM) by Song et al. (2019); high-dimensional mediation
    analysis (HDMA) by Gao et al. (2019); high-dimensional multivariate 
    mediation (HDMM) by Chén et al. (2018); high-dimensional linear mediation 
    analysis (HILMA) by Zhou et al. (2020); high-dimensional mediation analysis
    (HIMA) by Zhang et al. (2016); latent variable mediation analysis (LVMA) by 
    Derkach et al. (2019); mediation by fixed-effect model (MedFix) by 
    Zhang (2021); pathway LASSO by Zhao &amp; Luo (2022); principal component 
    mediation analysis (PCMA) by Huang &amp; Pan (2016); and sparse principal 
    component mediation analysis (SPCMA) by Zhao et al. (2020). Citations for 
    the corresponding papers can be found in their respective functions.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>bama (&ge; 1.3.0), foreach (&ge; 1.5.2), freebird (&ge; 1.0), gcdnet
(&ge; 1.0.6), genlasso (&ge; 1.6.1), hdi (&ge; 0.1.9), iterators (&ge;
1.0.14), MASS, mediation (&ge; 4.5.0), ncvreg (&ge; 3.13.0), stats,
utils</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-25 16:46:06 UTC; dc48b</td>
</tr>
<tr>
<td>Author:</td>
<td>Dylan Clark-Boucher [aut, cre],
  Mike Kleinsasser [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-27 08:42:31 UTC</td>
</tr>
</table>
<hr>
<h2 id='med_dat'>Mediation Example Dataset</h2><span id='topic+med_dat'></span>

<h3>Description</h3>

<p>A toy dataset for demonstrating mediation analysis in settings where there
are multiple mediators
</p>


<h3>Usage</h3>

<pre><code class='language-R'>med_dat
</code></pre>


<h3>Format</h3>



<h4><code>med_dat</code></h4>

<p>A list with three items:
</p>

<dl>
<dt>Y</dt><dd><p>numeric vector containing the outcome variable</p>
</dd>
<dt>M</dt><dd><p>numeric matrix of 20 continuous mediators</p>
</dd>
<dt>A</dt><dd><p>numeric vector containing the exposure variable</p>
</dd>
</dl>
<p>...

</p>


<hr>
<h2 id='mediate_bslmm'>Bayesian Sparse Linear Mixed Model</h2><span id='topic+mediate_bslmm'></span>

<h3>Description</h3>

<p><code>mediate_bslmm</code> fits the Bayesian sparse linear mixed model proposed by
Song et al. (2020) for high-dimensional mediation analysis, estimating the
mediation contributions of potential mediators.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediate_bslmm(
  A,
  M,
  Y,
  C1 = NULL,
  C2 = C1,
  burnin = 30000,
  ndraws = 5000,
  ci_level = 0.95,
  weights = NULL,
  k = 2,
  lm0 = 1e-04,
  lm1 = 1,
  lma1 = 1,
  l = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mediate_bslmm_+3A_a">A</code></td>
<td>
<p>length <code>n</code> numeric vector containing exposure variable</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_m">M</code></td>
<td>
<p><code>n x p</code> numeric matrix of high-dimensional mediators.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_y">Y</code></td>
<td>
<p>length <code>n</code> numeric vector containing continuous outcome variable.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_c1">C1</code></td>
<td>
<p>optional numeric matrix of covariates to include in the outcome model.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_c2">C2</code></td>
<td>
<p>optional numeric matrix of covariates to include in the mediator
model. Default is <code>C1</code>.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_burnin">burnin</code></td>
<td>
<p>number of MCMC draws prior to sampling.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_ndraws">ndraws</code></td>
<td>
<p>number of MCMC draws after burn-in.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_ci_level">ci_level</code></td>
<td>
<p>the desired credible interval level. Default is 0.95.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_weights">weights</code></td>
<td>
<p>optional numeric vector of observation weights.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_k">k</code></td>
<td>
<p>shape parameter for the inverse gamma priors. Default is 2.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_lm0">lm0</code></td>
<td>
<p>scale parameter for the inverse gamma prior on the variance of the
smaller-variance normal components. Default is <code>1e-4</code>. If <code>k=2</code>,
this parameter equals the prior mean on the smaller normal variance.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_lm1">lm1</code></td>
<td>
<p>scale parameter for the inverse gamma prior on the variance of the
larger-variance components of <code>beta_m</code>. Default is 1. If <code>k=2</code>,
this parameter equals the prior mean on the larger normal variance of the
mediator-outcome associations.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_lma1">lma1</code></td>
<td>
<p>scale parameter for the inverse gamma prior on the variance of the
larger-variance components of <code>alpha_a</code>. Default is 1. If <code>k=2</code>,
this parameter equals the prior mean on the larger normal variance of the
exposure-mediator associations.</p>
</td></tr>
<tr><td><code id="mediate_bslmm_+3A_l">l</code></td>
<td>
<p>scale parameter for the other inverse gamma priors.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>mediate_bslmm</code> is a wrapper function for the &quot;BSLMM&quot; option from <code><a href="bama.html#topic+bama">bama::bama()</a></code>,
which fits a Bayesian sparse linear mixed model for performing mediation
analysis with high-dimensional mediators. The model assumes that
the mediator-outcome associations (<code class="reqn">\beta_m</code>) and the exposure-mediator
associations (<code class="reqn">\alpha_a</code>) independently follow a mixture of small-variance
and high-variance normal distributions, and that if a mediator <code class="reqn">M_j</code> has both
<code class="reqn">(\beta_m)_j</code> and <code class="reqn">(\alpha_a)_j</code> belonging to the larger-variance distribution,
it has a notably large mediation contribution compared to the others. The
posterior inclusion probability (PIP) of belonging to both larger-variance
distributions is reported for each mediator as <code>ab_pip</code>.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p>contributions: a data frame containing the estimates, Bayesian
credible intervals, and posterior inclusion probabilities of the
mediation contributions.
</p>
</li>
<li><p>effects: a data frame containing the estimated direct, global
mediation, and total effects
</p>
</li></ul>



<h3>References</h3>

<p>Song, Y. et al. Bayesian shrinkage estimation of high dimensional
causal mediation effects in omics studies. Biometrics 76, 700-710 (2020).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- med_dat$A
M &lt;- med_dat$M
Y &lt;- med_dat$Y

# Toy example with small burnin and ndraws
out &lt;- mediate_bslmm(A, M, Y, burnin = 100, ndraws = 10)
out$effects
head(out$contributions)


</code></pre>

<hr>
<h2 id='mediate_hdma'>High-Dimensional Mediation Analysis</h2><span id='topic+mediate_hdma'></span>

<h3>Description</h3>

<p><code>mediate_hdma</code> fits a high-dimensional mediation model with
the de-biased LASSO approach as proposed by Gao et al. (2022),
estimating the mediation contributions of potential mediators.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediate_hdma(
  A,
  M,
  Y,
  C1 = NULL,
  C2 = NULL,
  binary_y = FALSE,
  n_include = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mediate_hdma_+3A_a">A</code></td>
<td>
<p>length <code>n</code> numeric vector containing exposure variable</p>
</td></tr>
<tr><td><code id="mediate_hdma_+3A_m">M</code></td>
<td>
<p><code>n x p</code> numeric matrix of high-dimensional mediators.</p>
</td></tr>
<tr><td><code id="mediate_hdma_+3A_y">Y</code></td>
<td>
<p>length <code>n</code> numeric vector containing continuous or binary outcome variable.</p>
</td></tr>
<tr><td><code id="mediate_hdma_+3A_c1">C1</code></td>
<td>
<p>optional numeric matrix of covariates to include in the outcome model.</p>
</td></tr>
<tr><td><code id="mediate_hdma_+3A_c2">C2</code></td>
<td>
<p>optional numeric matrix of covariates to include in the mediator model.</p>
</td></tr>
<tr><td><code id="mediate_hdma_+3A_binary_y">binary_y</code></td>
<td>
<p>logical flag for whether <code>Y</code> should be interpreted as a
binary variable with 1/0 coding rather than as continuous. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="mediate_hdma_+3A_n_include">n_include</code></td>
<td>
<p>integer specifying the number of top markers from sure
independent screening to be included. Default is <code>NULL</code>, in which case
<code>n_include</code> will be either <code>ceiling(n/log(n))</code> if
<code>binary_Y = F</code>, or <code>ceiling(n/(2*log(n)))</code> if <code>binary_Y = T</code>.
If <code>n_include &gt;= p</code>, all mediators are included with no screening. Note
that if <code>binary_y = F</code>, screening is performed based on the single-mediator
outcome model p-values, and if <code>binary_y = F</code>, screening is based on the
the mediator model p-values.</p>
</td></tr>
<tr><td><code id="mediate_hdma_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="hdi.html#topic+hdi">hdi::hdi()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The first step in HDMA is to perform sure independence
screening (SIS) to choose the <code>n_include</code> mediators that are most
associated with the outcome (when Y is continuous) or the exposure
(when Y is binary), based on p-values from linear regression. The second step
is to fit the outcome model for the remaining mediators using de-sparsified
(A.K.A de-biased) LASSO, which as asymptotic properties allowing for
computation of p-values by the <code>hdi</code> package. HDMA then fits the
mediator models using linear regression among those mediators that have both
survived SIS (in step 1) and been identified by the LASSO (in step 2), obtaining
p-values for the mediation contributions by taking the maximum of the <code class="reqn">\alpha_a</code>
and <code class="reqn">\beta_m</code> p-values. The global indirect effect is estimated by summing the
mediation contributions, and the direct effect is estimated by subtracting
the global indirect effect from an estimate of the total effect. See References for
more detail.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p>contributions: a data frame containing the estimates and p-values
of the mediation contributions
</p>
</li>
<li><p>effects: a data frame containing the estimated direct, global
mediation, and total effects
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://github.com/YuzhaoGao/High-dimensional-mediation-analysis-R">https://github.com/YuzhaoGao/High-dimensional-mediation-analysis-R</a>
</p>


<h3>References</h3>

<p>Gao, Y. et al. Testing Mediation Effects in High-Dimensional
Epigenetic Studies. Front. Genet. 10, 1195 (2019).
</p>
<p>Fan, J. &amp; Lv, J. Sure independence screening for ultrahigh dimensional
feature space. J. R. Stat. Soc. 70, 849-911 (2008)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- med_dat$A
M &lt;- med_dat$M
Y &lt;- med_dat$Y

# Fit hdma with continuous outcomes
out &lt;- mediate_hdma(A, M, Y)
head(out$contributions)
out$effects

</code></pre>

<hr>
<h2 id='mediate_hdmm'>High-dimensional Multivariate Mediation Analysis with
Principal Directions of Mediation</h2><span id='topic+mediate_hdmm'></span>

<h3>Description</h3>

<p><code>mediate_hdmm</code> estimates the first  &quot;direction of mediation&quot; in the
causal mediation mechanism of an exposure <code>A</code>, an outcome <code>Y</code>, and high-dimensional
mediators <code>M</code> as proposed by Chén et al. (2018).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediate_hdmm(
  A,
  M,
  Y,
  sims = 1000,
  boot_ci_type = "bca",
  ci_level = 0.95,
  tol = 10^-5,
  theta = rep(1, 5),
  w1 = rep(1, ncol(M)),
  interval = 10^6,
  step = 10^4,
  imax = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mediate_hdmm_+3A_a">A</code></td>
<td>
<p>numeric vector containing exposure variable.</p>
</td></tr>
<tr><td><code id="mediate_hdmm_+3A_m">M</code></td>
<td>
<p>numeric matrix of high-dimensional mediators. It is not recommended
to supply a matrix with more mediators than observations.</p>
</td></tr>
<tr><td><code id="mediate_hdmm_+3A_y">Y</code></td>
<td>
<p>numeric vector containing continuous outcome variable.</p>
</td></tr>
<tr><td><code id="mediate_hdmm_+3A_sims">sims</code></td>
<td>
<p>number of Monte Carlo draws for nonparametric bootstrap or
quasi-Bayesian approximation. See <code><a href="mediation.html#topic+mediate">mediate</a></code>. Default is 1000.</p>
</td></tr>
<tr><td><code id="mediate_hdmm_+3A_boot_ci_type">boot_ci_type</code></td>
<td>
<p>a character string indicating the type of bootstrap
confidence intervals for when <code>boot = TRUE</code>. If <code>"bca"</code>,
bias-corrected and accelerated (BCa) confidence intervals will be estimated.
If <code>"perc"</code>, percentile confidence intervals will be estimated
(see <code><a href="mediation.html#topic+mediate">mediation::mediate()</a></code>). Default is &quot;bca&quot;.</p>
</td></tr>
<tr><td><code id="mediate_hdmm_+3A_ci_level">ci_level</code></td>
<td>
<p>the designated confidence level. Default 0.95.</p>
</td></tr>
<tr><td><code id="mediate_hdmm_+3A_tol">tol</code></td>
<td>
<p>tolerance. Default 10^-5.</p>
</td></tr>
<tr><td><code id="mediate_hdmm_+3A_theta">theta</code></td>
<td>
<p>numeric vector of length 5 describing starting value of pathway
coefficients. Default is a vector of 1's.</p>
</td></tr>
<tr><td><code id="mediate_hdmm_+3A_w1">w1</code></td>
<td>
<p>numeric vector of the same length of <code>A</code> specifying PDM
starting values. Default is a vector of 1's.</p>
</td></tr>
<tr><td><code id="mediate_hdmm_+3A_interval">interval</code></td>
<td>
<p>numeric vector proportional to the intervals from where the
smoothing parameter is searched. Default is 10^6.</p>
</td></tr>
<tr><td><code id="mediate_hdmm_+3A_step">step</code></td>
<td>
<p>numerical number specifying step width for smoothing parameter
search. Default is 10^4.</p>
</td></tr>
<tr><td><code id="mediate_hdmm_+3A_imax">imax</code></td>
<td>
<p>integer specifying the maximum number of iterations allowed.
Default is 100.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>HDMM provides latent variable approach to high-dimensional mediation analysis.
The function <code>mediate_hdmm</code> uses a likelihood-based approach to compute
principal directions of mediation (PDMs), which are loading weights used to linearly
combine the inputted mediators to form a single, latent variable that replaces
the original mediators in the analysis. Though HDMM cannot be used to estimate
the global mediation effect or the contributions of specific mediators, it can
still can be useful for inferring whether there is mediation occurring through
the set of mediators as a joint system. See the provided reference for more
details.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p>pdm: the first direction of mediation by which mediators are
weighted.
</p>
</li>
<li><p>mediator: the latent mediator corresponding to the first direction
of mediation.
</p>
</li>
<li><p>effects: a data frame containing the estimates, confidence
intervals, and p-values of the mediation effects.
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://github.com/oliverychen/PDM">https://github.com/oliverychen/PDM</a>
</p>


<h3>References</h3>

<p>Chén, O. Y. et al. High-dimensional multivariate mediation with
application to neuroimaging data. Biostatistics 19, 121-136 (2018).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- as.numeric(scale(med_dat$A)) # can help to standardize
M &lt;- scale(med_dat$M[,1:8])
Y &lt;- as.numeric(scale(med_dat$Y))

out &lt;- mediate_hdmm(A, M, Y, sims = 5, tol = 10^-3, imax = 50)
out$effects

</code></pre>

<hr>
<h2 id='mediate_hilma'>High-Dimensional Linear Mediation Analysis</h2><span id='topic+mediate_hilma'></span>

<h3>Description</h3>

<p><code>mediate_hilma</code> applies high-dimensional linear mediation
analysis (HILMA) as proposed by Zhou et al. (2020).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediate_hilma(
  A,
  M,
  Y,
  aic_tuning = FALSE,
  nlambda = 5,
  lambda_minmax_ratio = 0.1,
  center = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mediate_hilma_+3A_a">A</code></td>
<td>
<p>length <code>n</code> numeric vector containing a single exposure variable
or size <code>n x q</code> numeric matrix containing multiple exposures. If a matrix,
should be low-dimensional (<code>q &lt;&lt; n</code>).</p>
</td></tr>
<tr><td><code id="mediate_hilma_+3A_m">M</code></td>
<td>
<p><code>n x p</code> numeric matrix of high-dimensional mediators.</p>
</td></tr>
<tr><td><code id="mediate_hilma_+3A_y">Y</code></td>
<td>
<p>length <code>n</code> numeric vector containing continuous outcome variable.</p>
</td></tr>
<tr><td><code id="mediate_hilma_+3A_aic_tuning">aic_tuning</code></td>
<td>
<p>logical flag for whether to select the tuning parameter using
AIC. Default is <code>FALSE</code> as this was not the preferred approach by the
HILMA authors (see references for more detail).</p>
</td></tr>
<tr><td><code id="mediate_hilma_+3A_nlambda">nlambda</code></td>
<td>
<p>number of candidate lambdas for AIC tuning. Default is 5. If
<code>aic_tuning=F</code> this parameter is ignored.</p>
</td></tr>
<tr><td><code id="mediate_hilma_+3A_lambda_minmax_ratio">lambda_minmax_ratio</code></td>
<td>
<p>ratio of the minimum lambda attempted in
AIC tuning to the maximum. If <code>aic_tuning=F</code>, ignored.</p>
</td></tr>
<tr><td><code id="mediate_hilma_+3A_center">center</code></td>
<td>
<p>logical flag for whether the variables should be centered. Default
is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>mediate_hilma</code> is a wrapper function for the <code><a href="freebird.html#topic+hilma">freebird::hilma()</a></code> function,
which fits the &quot;high-dimensional linear mediation analysis&quot; model proposed by
Zhou et al. (2020) for mediation settings when there are high-dimensional
mediators and one or several exposures. The function returns estimates of
the direct effect, total effect, and global mediation effect, the last of which
is tested for statistical significance with a reported p-value. For additional
detail, see the attached reference as well as the <code><a href="freebird.html#topic+hilma">freebird::hilma()</a></code>
documentation.
</p>


<h3>Value</h3>

<p>A list containing, for each exposure, a data frame of
the estimated direct, total, and global mediation effects. A p-value is
provided for the global mediation effect.
</p>


<h3>References</h3>

<p>Zhou, R. R., Wang, L. &amp; Zhao, S. D. Estimation and inference for
the indirect effect in high-dimensional linear mediation models. Biometrika
107, 573-589 (2020)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- med_dat$A
M &lt;- med_dat$M
Y &lt;- med_dat$Y

# Implement HILMA with one exposure
out &lt;- mediate_hilma(A, M, Y)
out$a1



</code></pre>

<hr>
<h2 id='mediate_hima'>High-Dimensional Mediation Analysis</h2><span id='topic+mediate_hima'></span>

<h3>Description</h3>

<p><code>mediate_hima</code> fits a high-dimensional mediation model with
the minimax concave penalty as proposed by Zhang et al. (2016),
estimating the mediation contributions of potential mediators.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediate_hima(
  A,
  M,
  Y,
  C1 = NULL,
  C2 = NULL,
  binary_y = FALSE,
  n_include = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mediate_hima_+3A_a">A</code></td>
<td>
<p>length <code>n</code> numeric vector containing exposure variable</p>
</td></tr>
<tr><td><code id="mediate_hima_+3A_m">M</code></td>
<td>
<p><code>n x p</code> numeric matrix of high-dimensional mediators.</p>
</td></tr>
<tr><td><code id="mediate_hima_+3A_y">Y</code></td>
<td>
<p>length <code>n</code> numeric vector containing continuous or binary outcome variable.</p>
</td></tr>
<tr><td><code id="mediate_hima_+3A_c1">C1</code></td>
<td>
<p>optional numeric matrix of covariates to include in the outcome model.</p>
</td></tr>
<tr><td><code id="mediate_hima_+3A_c2">C2</code></td>
<td>
<p>optional numeric matrix of covariates to include in the mediator model.</p>
</td></tr>
<tr><td><code id="mediate_hima_+3A_binary_y">binary_y</code></td>
<td>
<p>logical flag for whether <code>Y</code> should be interpreted as a
binary variable with 1/0 coding rather than as continuous. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="mediate_hima_+3A_n_include">n_include</code></td>
<td>
<p>integer specifying the number of top markers from sure
independent screening to be included. Default is <code>NULL</code>, in which case
<code>n_include</code> will be either <code>ceiling(n/log(n))</code> if
<code>binary_Y = F</code>, or <code>ceiling(n/(2*log(n)))</code> if <code>binary_Y = T</code>.
If <code>n_include &gt;= p</code>, all mediators are included with no screening. Note
that if <code>binary_y = F</code>, screening is performed based on the single-mediator
outcome model p-values, and if <code>binary_y = F</code>, screening is based on the
the mediator model p-values.</p>
</td></tr>
<tr><td><code id="mediate_hima_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="hdi.html#topic+hdi">hdi</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The first step in HIMA is to perform sure independence
screening (SIS) to choose the <code>n_include</code> mediators that are most
associated with the outcome (when Y is continuous) or the exposure
(when Y is binary), based on p-values from linear regression. The second step
is to fit the outcome model for the remaining mediators with the minimax
concave penalty. HIMA then fits the mediator models using linear regression
among those mediators that have both survived SIS (in step 1) and been
selected by the MCP (in step 2), which enables estimation of the mediation
contributions. The global indirect effect is estimated by summing these
contributions, and the direct effect is estimated by subtracting the global
indirect effect from an estimate of the total effect. We compute p-values for
the mediation contributions by taking the maximum of the <code class="reqn">\alpha_a</code> and <code class="reqn">\beta_m</code>
p-values, where the beta p-values are obtained via a second, unpenalized generalized
linear model containing only the mediators selected by the MCP. We include this
p-value computation so that our function replicates the behavior of the
<code>HIMA</code> function from <a href="https://CRAN.R-project.org/package=HIMA">HIMA</a>
package, the function on which ours is based, but we caution that the beta
p-values may be over-optimistic due to double-dipping, since the mediators tested in
the unpenalized model are only those chosen by the penalized model. Note also
that the HIMA authors apply Bonferroni correction to the final, maxed p-values
to account for multiple testing, which we choose to leave up to the user. For
more information, see the &quot;HIMA&quot; R package along with the provided reference.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p>contributions: a data frame containing the estimates and p-values
of the mediation contributions
</p>
</li>
<li><p>effects: a data frame containing the estimated direct, global
mediation, and total effects
</p>
</li></ul>



<h3>References</h3>

<p>Zhang, H. et al. Estimating and testing high-dimensional mediation effects
in epigenetic studies. Bioinformatics 32, 3150-3154 (2016).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- med_dat$A
M &lt;- med_dat$M
Y &lt;- med_dat$Y

# Fit hima with continuous outcome
out &lt;- mediate_hima(A, M, Y)
head(out$contributions)
out$effects

# Fit hima with binary outcome
Y1 &lt;- as.numeric(Y &gt; mean(Y))
out1 &lt;- mediate_hima(A, M, Y1, binary_y = TRUE)
head(out1$contributions)
out1$effects

</code></pre>

<hr>
<h2 id='mediate_lvma'>Latent Variable Mediation Analysis</h2><span id='topic+mediate_lvma'></span>

<h3>Description</h3>

<p><code>mediate_lvma</code> fits a high-dimensional mediation model described
by Derkach et al. (2019), in which a small number of latent, unmeasured
mediators replace the original mediators in the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediate_lvma(A, M, Y, q, rhoLM, rhoEL, rhoLY, scale = TRUE, imax = 5000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mediate_lvma_+3A_a">A</code></td>
<td>
<p>length <code>n</code> numeric vector representing the exposure variable</p>
</td></tr>
<tr><td><code id="mediate_lvma_+3A_m">M</code></td>
<td>
<p><code>n x p</code> numeric matrix of high-dimensional mediators.</p>
</td></tr>
<tr><td><code id="mediate_lvma_+3A_y">Y</code></td>
<td>
<p>length <code>n</code> numeric vector representing the continuous outcome variable.</p>
</td></tr>
<tr><td><code id="mediate_lvma_+3A_q">q</code></td>
<td>
<p>number of latent mediators</p>
</td></tr>
<tr><td><code id="mediate_lvma_+3A_rholm">rhoLM</code></td>
<td>
<p>numeric vector of candidate penalty parameters for the latent
mediator-mediator associations in the joint likelihood. Default is a short
toy sequence.</p>
</td></tr>
<tr><td><code id="mediate_lvma_+3A_rhoel">rhoEL</code></td>
<td>
<p>numeric vector of candidate penalty parameters for the exposure-latent mediator
associations in the joint likelihood. Default is a short toy sequence.</p>
</td></tr>
<tr><td><code id="mediate_lvma_+3A_rholy">rhoLY</code></td>
<td>
<p>numeric vector of candidate penalty parameters for the
latent mediator-outcome associations in the joint likelihood. Default is a
short toy sequence.</p>
</td></tr>
<tr><td><code id="mediate_lvma_+3A_scale">scale</code></td>
<td>
<p>logical flag for whether the inputted mediators should be standardized
prior to the analysis. Default is <code>FALSE</code>, but <code>TRUE</code> may be worth
attempting in case of errors.</p>
</td></tr>
<tr><td><code id="mediate_lvma_+3A_imax">imax</code></td>
<td>
<p>integer specifying the maximum number of iterations allowed. Default is 5000.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>LVMA is a latent variable mediation model which assumes, contrary to standard
assumptions, that the inputted set of candidate mediators do not
affect the outcome through the exposure on their own, but rather, occur as result
of latent, unmeasured mediators which themselves transmit effects from the
exposure to outcome. The required parameters for fitting this model are
<code>rhoLE</code>, a regularization parameter for effects of the latent mediators
on the inputted mediators; <code>rhoEL</code>, a regularization parameter for the
effects of the exposure on the latent mediators; and <code>rhoLY</code>, a
regularization parameter for the effects of the latent mediators on the
exposure. These parameters should ideally be supplied by the user as
vectors, so that each combination of the three parameters can be attempted
in the estimation. However, this can be intensely computation costly, and
for simplicity our default values are vectors of length 4, corresponding
to a 64 by 64 parameter grid. In practice, Derkach et al. use a much larger
grid with 5 values of <code>rhoLM</code> (ranging from 6 to 8.5), 40 values of
<code>rhoEY</code> (ranging from 0 to 40), and 40 values of <code>rhoLY</code> (ranging
from 0 to 75). Supplying longer parameter vectors makes the fit more flexible,
but more computationally costly, and to reliably implement LVMA on real data
one should use a larger parameter grid with parallel computation on a remote
computing cluster, as did the authors. For more information on the likelihood,
parameters, and mediation model, see the referenced article and/or its
supplement files.
</p>


<h3>Value</h3>

<p>A list containing the selected models based on AIC, BIC, and EBIC
(recommended) as three sub-lists. The sub-lists include objects indicating
the penalty set that was used (<code>penalty</code>), the values of the chosen parameters
(e.g., <code>EBIC</code>), the exposure-latent mediator effects (<code>AL_effects</code>),
the latent mediator-mediator effects (<code>LM_effects</code>, a data frame), the
direct effect of the exposure on the outcome (<code>AY_direct_effect</code>), the
the latent mediator-outcome effects (<code>LY_effects</code>), and binary vector
indicating whether each mediator was determined to be active. Here, active
mediators are those which are associated with a latent mediator that itself
is associated with both A and Y.
</p>


<h3>Source</h3>

<p><a href="https://pubmed.ncbi.nlm.nih.gov/30859548/">https://pubmed.ncbi.nlm.nih.gov/30859548/</a>
</p>


<h3>References</h3>

<p>Derkach, A., Pfeiffer, R. M., Chen, T.-H. &amp; Sampson, J. N. High dimensional
mediation analysis with latent variables. Biometrics 75, 745-756 (2019).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- med_dat$A
M &lt;- med_dat$M
Y &lt;- med_dat$Y

# Perform latent variable mediation analsis with 4 latent mediators and print
# whether the original 20 mediators are "actively" related to mediation
out &lt;- mediate_lvma(A, M, Y, q = 4, rhoLM = 2, rhoEL = 2, rhoLY = 2, imax = 50)
table(out$EBIC_out$mediator_active)

</code></pre>

<hr>
<h2 id='mediate_medfix'>Mediation Analysis via Fixed Effects Model</h2><span id='topic+mediate_medfix'></span>

<h3>Description</h3>

<p><code>mediate_medfix</code> fits a high-dimensional mediation model with
the adaptive LASSO approach as proposed by Zhang (2021) for the
special case of MedFix that there is only one exposure variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediate_medfix(
  A,
  M,
  Y,
  C1 = NULL,
  C2 = C1,
  nlambda = 100,
  nlambda2 = 50,
  nfolds = 10,
  seed = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mediate_medfix_+3A_a">A</code></td>
<td>
<p>length <code>n</code> numeric vector representing the exposure variable</p>
</td></tr>
<tr><td><code id="mediate_medfix_+3A_m">M</code></td>
<td>
<p><code>n x p</code> numeric matrix of high-dimensional mediators.</p>
</td></tr>
<tr><td><code id="mediate_medfix_+3A_y">Y</code></td>
<td>
<p>length <code>n</code> numeric vector representing the continuous outcome variable.</p>
</td></tr>
<tr><td><code id="mediate_medfix_+3A_c1">C1</code></td>
<td>
<p>optional numeric matrix of covariates to include in the outcome model.
Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="mediate_medfix_+3A_c2">C2</code></td>
<td>
<p>optional numeric matrix of covariates to include in the mediator
model. Default is <code>C1</code>.</p>
</td></tr>
<tr><td><code id="mediate_medfix_+3A_nlambda">nlambda</code></td>
<td>
<p>number of lambdas attempted in the adaptive LASSO.
See <code><a href="gcdnet.html#topic+cv.gcdnet">gcdnet::cv.gcdnet()</a></code>. The specific sequence of <code>lambda</code>s is chosen
by the <code>cv.gcdnet()</code> function.</p>
</td></tr>
<tr><td><code id="mediate_medfix_+3A_nlambda2">nlambda2</code></td>
<td>
<p>number of <code>lambda2</code>s attempted in the initial elastic net
used for computing adaptive weights prior to adaptive LASSO. Default is <code>50</code>.
See <code><a href="gcdnet.html#topic+cv.gcdnet">gcdnet::cv.gcdnet()</a></code> for details on elastic net regression. If <code>0</code>,
<code>lambda2=0</code> is fed to <code>cv.gcdnet()</code> and the initial fit is based on
standard LASSO, not elastic net. If not <code>0)</code>, the specific sequence of
<code>lambda2</code>s is given by <code>exp(seq(1e-4,0.02,length.out=nlambda2))</code>, and
the <code>lambda2</code> with the least cross-validated error is chosen.</p>
</td></tr>
<tr><td><code id="mediate_medfix_+3A_nfolds">nfolds</code></td>
<td>
<p>number of folds for cross-validation. See <code><a href="gcdnet.html#topic+cv.gcdnet">gcdnet::cv.gcdnet()</a></code>.
Default is <code>10</code>.</p>
</td></tr>
<tr><td><code id="mediate_medfix_+3A_seed">seed</code></td>
<td>
<p>numeric random seed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MedFix performs mediation analysis when there are multiple mediators by
applying adaptive LASSO to the outcome model. In order to fit the adaptive LASSO,
we first obtain an initial model fit using either LASSO (which deploys
the L1 penalty) or elastic net (which deploys both the L1 and L2 penalties)
depending on the provided <code>nlambda2</code>. Estimates from this fit are then used
to compute the adaptive weights used in the adaptive LASSO. Once the final
adaptive LASSO estimates (<code class="reqn">\beta_m</code>) are obtained for the outcome model, estimates for
the <code>p</code> mediator models (<code class="reqn">\alpha_a</code>) are obtained by linear regression. The
mediation contributions are computed as <code class="reqn">\alpha_a</code> times <code class="reqn">\beta_m</code>, and the p-value
is taken as the maximum of the <code class="reqn">\alpha_a</code> and <code class="reqn">beta_m</code> p-values. Last, the
global indirect effect is estimated by summing the mediation contributions, and the direct
effect is estimated by subtracting the global indirect effect from an estimate of
the total effect. This function is specific to applying MedFix to the special
case that there is only one exposure; for details on how to apply
MedFix when the exposures are high-dimensional, as proposed by the
authors, see the supplemental files of the referenced manuscript.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p>contributions: a data frame containing the estimates and p-values
of the mediation contributions
</p>
</li>
<li><p>effects: a data frame containing the estimated direct, global
mediation, and total effects
</p>
</li></ul>



<h3>References</h3>

<p>Zhang, Q. High-Dimensional Mediation Analysis with Applications to Causal
Gene Identification. Stat. Biosci. 14, 432-451 (2021).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- med_dat$Y
M &lt;- med_dat$M
Y &lt;- med_dat$Y

out &lt;- mediate_medfix(A, M, Y, nlambda = 10, nlambda2 = 5, seed = 1)
out$effects
head(out$contributions)

</code></pre>

<hr>
<h2 id='mediate_pcma'>Principal Component Mediation Analysis for High-dimensional Mediators</h2><span id='topic+mediate_pcma'></span>

<h3>Description</h3>

<p><code>mediate_pcma</code> applies principal component mediation analysis
(Huang and Pan, 2013) to mediation settings in which the mediators are high-dimensional.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediate_pcma(
  A,
  M,
  Y,
  var_per = 0.8,
  n_pc = NULL,
  sims = 1000,
  boot_ci_type = "bca",
  ci_level = 0.95,
  seed = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mediate_pcma_+3A_a">A</code></td>
<td>
<p>length <code>n</code> numeric vector containing exposure variable</p>
</td></tr>
<tr><td><code id="mediate_pcma_+3A_m">M</code></td>
<td>
<p><code>n x p</code> numeric matrix of high-dimensional mediators.</p>
</td></tr>
<tr><td><code id="mediate_pcma_+3A_y">Y</code></td>
<td>
<p>length <code>n</code> numeric vector containing continuous outcome variable.</p>
</td></tr>
<tr><td><code id="mediate_pcma_+3A_var_per">var_per</code></td>
<td>
<p>a numeric variable with the desired proportion of variance
explained. Default is 0.8.</p>
</td></tr>
<tr><td><code id="mediate_pcma_+3A_n_pc">n_pc</code></td>
<td>
<p>optional numeric variable with the desired number of PCs, in which case
<code>var_per</code> is ignored. Default is <code>NULL</code> and the number of PCs is
determined based on the desired proportion of variance explained.</p>
</td></tr>
<tr><td><code id="mediate_pcma_+3A_sims">sims</code></td>
<td>
<p>number of Monte Carlo draws for nonparametric bootstrap or
quasi-Bayesian approximation (see <code><a href="mediation.html#topic+mediate">mediation::mediate()</a></code>).
Default is 1000.</p>
</td></tr>
<tr><td><code id="mediate_pcma_+3A_boot_ci_type">boot_ci_type</code></td>
<td>
<p>a character string indicating the type of bootstrap
confidence intervals for when <code>boot = TRUE</code>. If <code>"bca"</code>,
bias-corrected and accelerated (BCa) confidence intervals will be estimated.
If <code>"perc"</code>, percentile confidence intervals will be estimated
(see <code><a href="mediation.html#topic+mediate">mediation::mediate()</a></code>). Default is &quot;bca&quot;.</p>
</td></tr>
<tr><td><code id="mediate_pcma_+3A_ci_level">ci_level</code></td>
<td>
<p>the desired confidence level. Default is 0.95.</p>
</td></tr>
<tr><td><code id="mediate_pcma_+3A_seed">seed</code></td>
<td>
<p>seed used for fitting single-mediator models after PCA</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Principal component mediation analysis (PCMA) is a method for estimating
mediation effects when the mediators are high-dimensional. The first step
is to compute the residuals of mediator models (<code class="reqn">M|A</code>), then perform
PCA on those residuals to reduce them to a smaller number of mediators
that efficiently explain the residual variance. Then, since those mediators
are linearly independent conditional on A, one can trivially perform
single-mediator mediation analysis for each PC on its own, in this case
by using the <code><a href="mediation.html#topic+mediate">mediation::mediate()</a></code> function. The global mediation effect is estimated
by summing the mediation effects of the individual PCs.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p>loadings: a matrix of the PC loadings.
</p>
</li>
<li><p>pcs: a matrix of the PCs.
</p>
</li>
<li><p>var_explained: the cumulative proportion of variance explained by the PCs.
</p>
</li>
<li><p>contributions: a data frame containing the estimates, confidence
intervals, and p-values of the mediation contributions.
</p>
</li>
<li><p>effects: a data frame containing the estimated direct, global
mediation, and total effects
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://rdrr.io/github/zhaoyi1026/spcma">https://rdrr.io/github/zhaoyi1026/spcma</a>
</p>


<h3>References</h3>

<p>Huang, Y.-T. &amp; Pan, W.-C. Hypothesis test of mediation effect in
causal mediation model with  high-dimensional continuous mediators.
Biometrics 72, 402-413 (2016).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- med_dat$A
M &lt;- med_dat$M
Y &lt;- med_dat$Y
# Fit PCMA with 3 principal components and print the effects. In practice one
# should choose n_pc (or var_per) and the number sims to be larger
out &lt;- mediate_pcma(A, M, Y, n_pc = 3, sims = 10)
out$effects



</code></pre>

<hr>
<h2 id='mediate_plasso'>Pathway LASSO for Mediation Analysis with High-Dimensional Mediators</h2><span id='topic+mediate_plasso'></span>

<h3>Description</h3>

<p><code>mediate_plasso</code> fits a high-dimensional mediation model
with the penalized likelihood described by Zhao and Luo (2022), estimating the
mediation contributions of each site.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediate_plasso(
  A,
  M,
  Y,
  lambdas = NULL,
  select_lambda = FALSE,
  vss_rep = 5,
  vss_cutoff = 0.1,
  omega_ratio = 1,
  phi = 2,
  maxit = 5000,
  tol = 1e-06
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mediate_plasso_+3A_a">A</code></td>
<td>
<p>numeric vector containing exposure variable.</p>
</td></tr>
<tr><td><code id="mediate_plasso_+3A_m">M</code></td>
<td>
<p>numeric matrix of high-dimensional mediators.</p>
</td></tr>
<tr><td><code id="mediate_plasso_+3A_y">Y</code></td>
<td>
<p>numeric vector containing continuous outcome variable.</p>
</td></tr>
<tr><td><code id="mediate_plasso_+3A_lambdas">lambdas</code></td>
<td>
<p>numeric vector of tuning parameters <code>lambda</code> for
which model is fitted. Default is a vector of length 50 ranging from 10^-5
to 10^4, with more density in the lower range.</p>
</td></tr>
<tr><td><code id="mediate_plasso_+3A_select_lambda">select_lambda</code></td>
<td>
<p>logical flag indicating whether to conduct a tuning
parameter selection using the Variable Selection Stability Criterion
described by Sun et al. (2013). Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="mediate_plasso_+3A_vss_rep">vss_rep</code></td>
<td>
<p>if <code>select_lambda</code> is <code>TRUE</code>, number of VSSC
replications.</p>
</td></tr>
<tr><td><code id="mediate_plasso_+3A_vss_cutoff">vss_cutoff</code></td>
<td>
<p>if <code>select_lambda</code> is <code>TRUE</code>, cutoff used for
VSSC. Default is 0.1.</p>
</td></tr>
<tr><td><code id="mediate_plasso_+3A_omega_ratio">omega_ratio</code></td>
<td>
<p>ratio of the <code>omega</code> parameter to <code>lambda</code>
parameter in the likelihood penalty. Default is 1.</p>
</td></tr>
<tr><td><code id="mediate_plasso_+3A_phi">phi</code></td>
<td>
<p>value of the <code>phi</code> parameter in the likelihood penalty.
Default is 2. Cannot be less than 1/2.</p>
</td></tr>
<tr><td><code id="mediate_plasso_+3A_maxit">maxit</code></td>
<td>
<p>the maximum number of iterations. Default is 5000.</p>
</td></tr>
<tr><td><code id="mediate_plasso_+3A_tol">tol</code></td>
<td>
<p>convergence tolerance. Default is 10^-6.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pathway LASSO fits a high-dimensional mediation model with a likelihood
that directly penalizes the mediator-outcome effects, exposure-mediator
effects, and mediation contributions (i.e., the mediation &quot;pathways&quot;).
The shrinkage of the model is determined by three parameters&ndash;<code>phi</code>,
<code>omega</code>, and <code>lambda</code>&ndash;with higher values tending to result in more
sparsity. Mediation results are returned for every unique value in the
inputted <code>lambdas</code> argument, in increasing order, excluding potentially
any for which the estimation was unsuccessful. For details on the exact
likelihood, see the first reference.
</p>
<p>When implementing multiple lambdas (i.e., using either the default argument
to <code>lambdas</code> or specifying one's own vector), there is an option to
perform a tuning parameter selection using the Variable Selection Stability
Criterion proposed by Sun et al. (2013), which chooses the parameter for
which the variable selection is most stable in repeated data-splitting.
However, implementing this can be computationally costly, especially with a
long list of tuning parameters, since it involves re-fitting pathway LASSO
many times.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p>lambdas: the <code>lambda</code>s attempted in the same order as the objects in <code>all_fits</code>.
</p>
</li>
<li><p>all_fits: a list containing, for each <code>lambda</code>, a <code>data.frame</code> of the estimated mediation effects.
</p>
</li>
<li><p>chosen_lambda: if <code>select_lambda</code> is <code>TRUE</code>, the <code>lambda</code> chosen by VSSC.
</p>
</li>
<li><p>chosen_fit: if <code>select_lambda</code> is <code>TRUE</code>, the fit corresponding to the chosen <code>lambda</code>.
</p>
</li>
<li><p>vss: if <code>select_lambda</code> is<code>TRUE</code>, a <code>data.frame</code> containing the variable selection stabilities.
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://github.com/zhaoyi1026/PathwayLasso">https://github.com/zhaoyi1026/PathwayLasso</a>
</p>


<h3>References</h3>

<p>Zhao, Y. &amp; Luo, X. Pathway LASSO: pathway estimation and
selection with high-dimensional mediators. Stat. Interface 15, 39-50 (2022).
</p>
<p>Sun, W., Wang, J. &amp; Fang, Y. Consistent selection of tuning parameters via
variable selection stability. J. Mach. Learn. Res. 14, 3419-3440 (2013).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- med_dat$A
M &lt;- med_dat$M[,1:8]
Y &lt;- med_dat$Y
# fit pathway LASSO for two tuning parameters and retrieve their fits
out &lt;- mediate_plasso(A, M, Y, lambdas = c(10^-3, 10^-2), tol = 1e-4)
head(out$all_fits$lambda1)
head(out$all_fits$lambda2)


</code></pre>

<hr>
<h2 id='mediate_spcma'>Sparse Principal Component Mediation Analysis for High-Dimensional Mediators</h2><span id='topic+mediate_spcma'></span>

<h3>Description</h3>

<p><code>mediate_spcma</code> applies sparse principal component mediation
analysis to mediation settings in which the mediators are high-dimensional.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediate_spcma(
  A,
  M,
  Y,
  var_per = 0.8,
  n_pc = NULL,
  sims = 1000,
  boot_ci_type = "bca",
  ci_level = 0.95,
  fused = FALSE,
  gamma = 0,
  per_jump = 0.7,
  eps = 1e-04,
  maxsteps = 2000,
  seed = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mediate_spcma_+3A_a">A</code></td>
<td>
<p>length <code>n</code> numeric vector containing exposure variable</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_m">M</code></td>
<td>
<p><code>n x p</code> numeric matrix of high-dimensional mediators.</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_y">Y</code></td>
<td>
<p>length <code>n</code> numeric vector containing continuous outcome variable.</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_var_per">var_per</code></td>
<td>
<p>a numeric variable with the desired proportion of variance
explained. Default is 0.8.</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_n_pc">n_pc</code></td>
<td>
<p>optional numeric variable with the desired number of PCs, in which case
<code>var_per</code> is ignored. Default is <code>NULL</code> and the number of PCs is
determined based on the desired proportion of variance explained.</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_sims">sims</code></td>
<td>
<p>number of Monte Carlo draws for nonparametric bootstrap or
quasi-Bayesian approximation (see <code><a href="mediation.html#topic+mediate">mediation::mediate()</a></code>).
Default is 1000.</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_boot_ci_type">boot_ci_type</code></td>
<td>
<p>character string indicating the type of bootstrap
confidence intervals for when <code>boot = TRUE</code>. If <code>"bca"</code>,
bias-corrected and accelerated (BCa) confidence intervals will be estimated.
If <code>"perc"</code>, percentile confidence intervals will be estimated
(see <code><a href="mediation.html#topic+mediate">mediation::mediate()</a></code>). Default is &quot;bca&quot;.</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_ci_level">ci_level</code></td>
<td>
<p>the designated confidence level. Default 0.95.</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_fused">fused</code></td>
<td>
<p>logical variable for whether the fused LASSO should be used
instead of the ordinary LASSO. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_gamma">gamma</code></td>
<td>
<p>numeric variable <code>&gt;=0</code> indicating the ratio of the
standard LASSO penalty to the fusion penalty (see <code><a href="genlasso.html#topic+genlasso">genlasso::genlasso()</a></code>).
Ignored if <code>fused = FALSE</code>. Default is 0, meaning there is no standard
penalty. Larger values result in more shrinkage and sparser PCs.</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_per_jump">per_jump</code></td>
<td>
<p>numeric value used for tuning parameter selection - the
quantile cut-off for total variance change under different <code>lambda</code> values
in the LASSO. Default is 0.7. Larger values result in more shrinkage and
sparser PCs</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_eps">eps</code></td>
<td>
<p>numeric variable indicating the multiplier for the ridge penalty
in case <code>X</code> is rank deficient (see  <code><a href="genlasso.html#topic+genlasso">genlasso::genlasso()</a></code>).
Default is <code>1e-4</code>.</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_maxsteps">maxsteps</code></td>
<td>
<p>an integer specifying the maximum number of steps for the
algorithm before termination (see <code><a href="genlasso.html#topic+genlasso">genlasso::genlasso()</a></code>). Default
is 2000.</p>
</td></tr>
<tr><td><code id="mediate_spcma_+3A_seed">seed</code></td>
<td>
<p>seed used for fitting single-mediator models after PCA</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>mediate_spcma</code> performs principal component mediation analysis, comparable
to <code>mediate_pcma</code>, with the modification that the PC loadings are sparsified
by a flexible LASSO penalty. This has the potential make the PCs more interpretable,
since, unlike in PCA, they are only linear combinations of a subset of mediators
rather than all of them. The choice of LASSO penalties is determined by
the <code>fused</code> argument - which, when set to <code>TRUE</code>, deploys a fused
LASSO penalty that encourages the model to give consecutive mediators
similar loadings. The default is <code>fused = FALSE</code>, and the standard
LASSO penalty is used instead of the fusion penalty. Once the sparse PCs are
computed, inference proceeds exactly like in PCMA, and the PC-mediators are
evaluated with methods from the <code>mediate</code> package.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p>loadings: a matrix of the sparse PC loadings.
</p>
</li>
<li><p>pcs: a matrix of the PCs.
</p>
</li>
<li><p>var_explained: the cumulative proportion of variance explained by the PCs.
</p>
</li>
<li><p>contributions: a data frame containing the estimates, confidence
intervals, and p-values of the mediation contributions.
</p>
</li>
<li><p>effects: a data frame containing the estimated direct, global
mediation, and total effects
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://rdrr.io/github/zhaoyi1026/spcma">https://rdrr.io/github/zhaoyi1026/spcma</a>
</p>


<h3>References</h3>

<p>Zhao, Y., Lindquist, M. A. &amp; Caffo, B. S. Sparse principal
component based high-dimensional mediation analysis. Comput. Stat.
Data Anal. 142, 106835 (2020).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- med_dat$A
M &lt;- med_dat$M
Y &lt;- med_dat$Y

# Fit SPCMA with the fused LASSO penalty while choosing the number of PCs based
# on the variance they explain. In practice, var_per and sims should be higher.
out &lt;- mediate_spcma(A, M, Y, var_per = 0.25, fused = TRUE, gamma = 2, sims = 10)
out$effects

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
