<!DOCTYPE html><html><head><title>Help for package HDMT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HDMT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adjust_quantile'>
<p>A function to compute the quantiles of the estimated mixture null distribution for pmax using either the approximation</p>
or exact method
</a></li>
<li><a href='#correct_qqplot'>
<p>A function to draw the corrected quantile-quantile plot for p-max using the expected quantiles</p></a></li>
<li><a href='#exercise_input'>
<p>An example dataset to demonstrate the usage of 'HDMT'</p></a></li>
<li><a href='#fdr_est'>
<p>A function to compute the estimated pointwise FDR for every observed p-max</p></a></li>
<li><a href='#fwer_est'>
<p>A function used to compute Family wise error rate (FWER) cutoff for p-max</p>
at a designated level</a></li>
<li><a href='#null_estimation'>
<p>A function to estimate the proportions of the three component nulls</p></a></li>
<li><a href='#snp_input'>
<p>An example dataset to demonstrate the usage of 'HDMT'</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Multiple Testing Procedure for High-Dimensional Mediation
Hypotheses</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-01-24</td>
</tr>
<tr>
<td>Author:</td>
<td>James Dai [aut, cre],
  Xiaoyu Wang [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>James Dai &lt;jdai@fredhutch.org&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A multiple-testing procedure for high-dimensional mediation hypotheses. Mediation analysis is of rising interest in epidemiology and clinical trials. Among existing methods for mediation analyses, the popular joint significance (JS) test yields an overly conservative type I error rate and therefore low power. In the R package 'HDMT' we implement a multiple-testing procedure that accurately controls the family-wise error rate (FWER) and the false discovery rate (FDR) when using JS for testing high-dimensional mediation hypotheses. The core of our procedure is based on estimating the proportions of three component null hypotheses and deriving the corresponding mixture distribution of null p-values. Results of the data examples include better-behaved quantile-quantile plots and improved detection of novel mediation relationships on the role of DNA methylation in genetic regulation of gene expression. With increasing interest in mediation by molecular intermediaries such as gene expression, the proposed method addresses an unmet methodological challenge. Methods used in the package refer to James Y. Dai, Janet L. Stanford &amp; Michael LeBlanc (2020) &lt;<a href="https://doi.org/10.1080%2F01621459.2020.1765785">doi:10.1080/01621459.2020.1765785</a>&gt;.</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>no</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>fdrtool, qvalue</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-01-29 18:30:20 UTC; jdai</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-01-29 19:00:02 UTC</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
</table>
<hr>
<h2 id='adjust_quantile'>
A function to compute the quantiles of the estimated mixture null distribution for pmax using either the approximation
or exact method

</h2><span id='topic+adjust_quantile'></span>

<h3>Description</h3>

<p>A function to generate the quantiles of the estimated three-component mixture null distribution for pmax 
(the maximum of the two p-values  for testing mediation) using either approximation or exact method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust_quantile(alpha00, alpha01, alpha10, alpha1, alpha2, 
input_pvalues, exact = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adjust_quantile_+3A_alpha00">alpha00</code></td>
<td>

<p>A numeric number represents the proportion of null <code class="reqn">H_{00}</code>
</p>
</td></tr>
<tr><td><code id="adjust_quantile_+3A_alpha01">alpha01</code></td>
<td>

<p>A numeric number represents the proportion of null <code class="reqn">H_{01}</code>
</p>
</td></tr>
<tr><td><code id="adjust_quantile_+3A_alpha10">alpha10</code></td>
<td>

<p>A numeric number represents the proportion of null <code class="reqn">H_{10}</code>
</p>
</td></tr>
<tr><td><code id="adjust_quantile_+3A_alpha1">alpha1</code></td>
<td>

<p>A numeric number represents the proportion of null alpha==0 
(association between exposure and mediator)</p>
</td></tr>
<tr><td><code id="adjust_quantile_+3A_alpha2">alpha2</code></td>
<td>

<p>A numeric number represents the proportion of null beta==0 
(association between mediator and outcome adjusted for exposure)</p>
</td></tr>
<tr><td><code id="adjust_quantile_+3A_input_pvalues">input_pvalues</code></td>
<td>

<p>A matrix contains two columns of p-values for candidate mediators.
Column 1 is the p-value of testing if a exposure is associated with 
the mediator (alpha!=0). 
Column 2 is the p-value of testing if a mediator is associated with 
the outcome adjusted for exposure(beta!=0)  
</p>
</td></tr>
<tr><td><code id="adjust_quantile_+3A_exact">exact</code></td>
<td>

<p>Use the option to choose from two methods. 
exact=0: the approximation method without estimating the CDFs when deriving the mixture null distribution;
exact=1: the exact method to estimate the CDFs nonparametrically when deriving the mixture null distribution</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The is a function to computed the expected quantiles for the observed 
p-max values based on the estimated mixture null distribution. The methodology detail can be found in Dai et al (2020).
</p>


<h3>Value</h3>

<p>A vector contains the expected quantiles of p-values based on the estimated mixture null distribution. See Dai et al (2020) for details of how to compute quantiles using the approximation method (exact=0) or the exact method (exact=1).
</p>


<h3>Author(s)</h3>

<p>James Y. Dai and X. Wang
</p>


<h3>References</h3>

<p>James Y. Dai, Janet L. Stanford, Michael LeBlanc. A multiple-testing 
procedure for high-dimensional mediation hypotheses. 
Journal of the American Statistical Association, 2020, DOI: 10.1080/01621459.2020.1765785. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp_input)
input_pvalues &lt;- snp_input

#To save time for illustration, we use 10 percent of rows

input_pvalues &lt;- input_pvalues[sample(1:nrow(input_pvalues),size=ceiling(nrow(input_pvalues)/10)),]

nullprop &lt;- null_estimation(input_pvalues)

pnull &lt;- adjust_quantile(nullprop$alpha00,nullprop$alpha01,nullprop$alpha10,nullprop$alpha1,
                         nullprop$alpha2,input_pvalues,exact=0)
</code></pre>

<hr>
<h2 id='correct_qqplot'>
A function to draw the corrected quantile-quantile plot for p-max using the expected quantiles</h2><span id='topic+correct_qqplot'></span>

<h3>Description</h3>

<p>A function to draw the corrected quantile-quantile (Q-Q) plots. The corrected quantiles were computed from 
the mixture null distribution (green dots) and the naive quantiles were computed from the uniform distribution 
(red dots).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correct_qqplot(pmax, pnull, opt="all")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="correct_qqplot_+3A_pmax">pmax</code></td>
<td>

<p>The vector for maximum  p-values </p>
</td></tr>  
<tr><td><code id="correct_qqplot_+3A_pnull">pnull</code></td>
<td>

<p>The quantiles of pmax based on the estimated mixture null distribution
</p>
</td></tr>
<tr><td><code id="correct_qqplot_+3A_opt">opt</code></td>
<td>

<p>Option to draw the plot. opt=&quot;all&quot;:use all the data points, 
opt=&quot;subset&quot;: use a subset of the data points, in case there 
are too many points in a genome-wide setting, to avoid drawing an overcrowded Q-Q plot with a prohibitive image size.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James Y. Dai and X. Wang
</p>


<h3>References</h3>

<p>James Y. Dai, Janet L. Stanford, Michael LeBlanc. A multiple-testing 
procedure for high-dimensional mediation hypotheses, 
Journal of the American Statistical Association, 2020, DOI: 10.1080/01621459.2020.1765785. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp_input)
input_pvalues &lt;- snp_input
#To save time for illustration, we use 10 percent of rows
input_pvalues &lt;- input_pvalues[sample(1:nrow(input_pvalues),
                               size=ceiling(nrow(input_pvalues)/10)),]

pmax &lt;- apply(input_pvalues,1,max)
nullprop &lt;- null_estimation(input_pvalues)
pnull1 &lt;- adjust_quantile(nullprop$alpha10,nullprop$alpha01,nullprop$alpha00,
                          nullprop$alpha1,nullprop$alpha2,input_pvalues,exact=1)
correct_qqplot(pmax,pnull1)
</code></pre>

<hr>
<h2 id='exercise_input'>
An example dataset to demonstrate the usage of 'HDMT'
</h2><span id='topic+exercise_input'></span>

<h3>Description</h3>

<p>This example dataset was included to assess the mediation role of DNA methylation in the effect of exercise 
on prostate cancer progression in a 
Seattle-based cohort of patients diagnosed with clinically localized 
PCa. The entire data set contains two sets of p-values from genome-wide testing of 450K CpG sites. Due to space limit, 
a subset (10 percent) of the full dataset  is included in the package for illustration.
</p>
<p>The dataset is a matrix containing two columns of p-values for 
candidate mediators. Column 1 contains the p-values for testing if an exposure is associated with 
the mediator (alpha!=0). Column 2 contains the p-value for testing if a mediator is associated with 
the outcome after adjusted for the exposure (beta!=0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("exercise_input")</code></pre>


<h3>Format</h3>

<p>The format of exercise_input is:
num [1:47900, 1:2] 0.4966344 0.1048730 0.1005355 0.4946623 ...
</p>


<h3>References</h3>

<p>James Y. Dai, Janet L. Stanford, Michael LeBlanc. A multiple-testing 
procedure for high-dimensional mediation hypotheses, 
Journal of the American Statistical Association, 2020, DOI: 10.1080/01621459.2020.1765785.  
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exercise_input)
dim(exercise_input)

</code></pre>

<hr>
<h2 id='fdr_est'>
A function to compute the estimated pointwise FDR for every observed p-max</h2><span id='topic+fdr_est'></span>

<h3>Description</h3>

<p>A function to compute the estimated pointwise FDR based on the proposed joint significance mixture null method (JS-mixture).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fdr_est(alpha00, alpha01, alpha10, alpha1, alpha2, input_pvalues, exact = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fdr_est_+3A_alpha00">alpha00</code></td>
<td>

<p>A numeric number represents the proportion of null <code class="reqn">H_{00}</code>
</p>
</td></tr>
<tr><td><code id="fdr_est_+3A_alpha01">alpha01</code></td>
<td>

<p>A numeric number represents the proportion of null <code class="reqn">H_{01}</code>
</p>
</td></tr>
<tr><td><code id="fdr_est_+3A_alpha10">alpha10</code></td>
<td>

<p>A numeric number represents the proportion of null <code class="reqn">H_{10}</code>
</p>
</td></tr>
<tr><td><code id="fdr_est_+3A_alpha1">alpha1</code></td>
<td>

<p>A numeric number represents the proportion of null alpha=0
</p>
</td></tr>
<tr><td><code id="fdr_est_+3A_alpha2">alpha2</code></td>
<td>

<p>A numeric number represents the proportion of null beta=0
</p>
</td></tr>
<tr><td><code id="fdr_est_+3A_input_pvalues">input_pvalues</code></td>
<td>

<p>A matrix contains two columns of p-values for candidate mediators.
Column 1 is the p-value of testing if an exposure  is associated with 
the mediator (alpha!=0). 
Column 2 is the p-value of testing if a mediator is associated with 
the outcome adjusted for the exposure (beta!=0) 
</p>
</td></tr>
<tr><td><code id="fdr_est_+3A_exact">exact</code></td>
<td>

<p>The option to choose from two methods. exact=0: approximation without 
estimating the CDFs; exact=1: estimate the CDFs nonparametrically</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A function to estimate the pointwise FDR based on the proposed  method 
to estimate the mixture null distribution. See Dai et al (2020) for details of how to compute quantiles using the approximation method (exact=0) or the exact method (exact=1).</p>


<h3>Value</h3>

<p>The estimated pointwise FDR for p-max
</p>


<h3>Author(s)</h3>

<p>James Y. Dai and X. Wang
</p>


<h3>References</h3>

<p>James Y. Dai, Janet L. Stanford, Michael LeBlanc. A multiple-testing 
procedure for high-dimensional mediation hypotheses, 
Journal of the American Statistical Association, 2020, DOI: 10.1080/01621459.2020.1765785. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp_input)
input_pvalues &lt;- snp_input
#To save time for illustration, we use 10 percent of rows
input_pvalues &lt;- input_pvalues[sample(1:nrow(input_pvalues),size=ceiling(nrow(input_pvalues)/10)),]

nullprop &lt;- null_estimation(input_pvalues)
fdr &lt;- fdr_est(nullprop$alpha00,nullprop$alpha01,nullprop$alpha10,
               nullprop$alpha1,nullprop$alpha2,input_pvalues,exact=0)

</code></pre>

<hr>
<h2 id='fwer_est'>
A function used to compute Family wise error rate (FWER) cutoff for p-max 
at a designated level</h2><span id='topic+fwer_est'></span>

<h3>Description</h3>

<p>A function to compute the FWER cutoff for p-max using 
the estimated mixture null distribution</p>


<h3>Usage</h3>

<pre><code class='language-R'>fwer_est(alpha10, alpha01, alpha00, alpha1, alpha2, input_pvalues, 
         alpha = 0.05, exact = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fwer_est_+3A_alpha00">alpha00</code></td>
<td>

<p>A numeric number represents the proportion of null <code class="reqn">H_{00}</code>
</p>
</td></tr>
<tr><td><code id="fwer_est_+3A_alpha01">alpha01</code></td>
<td>

<p>A numeric number represents the proportion of null <code class="reqn">H_{01}</code>
</p>
</td></tr>
<tr><td><code id="fwer_est_+3A_alpha10">alpha10</code></td>
<td>

<p>A numeric number represents the proportion of null <code class="reqn">H_{10}</code>
</p>
</td></tr>
<tr><td><code id="fwer_est_+3A_alpha1">alpha1</code></td>
<td>

<p>A numeric number represents the proportion of null alpha==0
</p>
</td></tr>
<tr><td><code id="fwer_est_+3A_alpha2">alpha2</code></td>
<td>

<p>A numeric number represents the proportion of null beta==0
</p>
</td></tr>
<tr><td><code id="fwer_est_+3A_input_pvalues">input_pvalues</code></td>
<td>

<p>A matrix contains two columns of p-values for candidate mediators.
Column 1 is the p-value of testing if an exposure is associated with 
the mediator (alpha!=0). 
Column 2 is the p-value of testing if a mediator is associated with 
the outcome adjusted for the exposure (beta!=0) 
</p>
</td></tr>
<tr><td><code id="fwer_est_+3A_alpha">alpha</code></td>
<td>

<p>The designated significance level for FWER</p>
</td></tr>
<tr><td><code id="fwer_est_+3A_exact">exact</code></td>
<td>

<p>The option to choose from two methods. exact=0: approximation without 
estimating the CDFs; exact=1: estimate the CDFs nonparametrically
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A function to compute FWER cutoff for p-max accounting for 
the mixture null distribution. The methodology detail can be found in Dai et al (2020).
</p>


<h3>Value</h3>

<p>A numeric number represents the output FWER cutoff
</p>


<h3>Author(s)</h3>

<p>James Y. Dai and X. Wang
</p>


<h3>References</h3>

<p>James Y. Dai, Janet L. Stanford, Michael LeBlanc. A multiple-testing 
procedure for high-dimensional mediation hypotheses, 
Journal of the American Statistical Association, 2020, DOI: 10.1080/01621459.2020.1765785. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp_input)
input_pvalues &lt;- snp_input
#To save time for illustration, we use 10 percent of rows
input_pvalues &lt;- input_pvalues[sample(1:nrow(input_pvalues),
                               size = ceiling(nrow(input_pvalues)/10)),]

nullprop &lt;- null_estimation(input_pvalues)
fwercut0 &lt;- fwer_est(nullprop$alpha10,nullprop$alpha01,nullprop$alpha00,nullprop$alpha1,
nullprop$alpha2,input_pvalues,alpha=0.05,exact=0)

fwercut1 &lt;- fwer_est(nullprop$alpha10,nullprop$alpha01,nullprop$alpha00,nullprop$alpha1,
nullprop$alpha2,input_pvalues,alpha=0.05,exact=1)

</code></pre>

<hr>
<h2 id='null_estimation'>
A function to estimate the proportions of the three component nulls
</h2><span id='topic+null_estimation'></span>

<h3>Description</h3>

<p>This is a function to estimate the proportions of the three component
nulls involved in mediation testing. We developed a three component-mixture model method to estimate the proportions of nulls and provide much more accurate control of the family-wise error rate (FWER) and the false discovery rate (FDR), when  compared to the standard approach using the uniform null distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>null_estimation(input_pvalues)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="null_estimation_+3A_input_pvalues">input_pvalues</code></td>
<td>

<p>A matrix contains two columns of p-values for candidate mediators.
Column 1 is the p-value of testing if the exposure is associated with 
the candidate mediator (alpha!=0). 
Column 2 is the p-value of testing if the candidate mediator is associated with 
the outcome adjusted for the exposure (beta!=0).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A function to estimate the proportions of the three types 
of component null hypotheses:
</p>
<p><code class="reqn">H_{00}</code>: alpha==0 and beta==0
</p>
<p><code class="reqn">H_{01}</code>: alpha==0 and beta!=0
</p>
<p><code class="reqn">H_{10}</code>: alpha!=0 and beta==0
</p>
<p>The methodology detail can be found in Dai et al (2020).
</p>


<h3>Value</h3>

<p>A list contains five elements.
</p>
<table>
<tr><td><code>alpha00</code></td>
<td>
<p>A numeric number represents the proportion of null <code class="reqn">H_{00}</code></p>
</td></tr>
<tr><td><code>alpha01</code></td>
<td>
<p>A numeric number represents the proportion of null <code class="reqn">H_{01}</code></p>
</td></tr>
<tr><td><code>alpha10</code></td>
<td>
<p>A numeric number represents the proportion of null <code class="reqn">H_{10}</code></p>
</td></tr>
<tr><td><code>alpha1</code></td>
<td>
<p>A numeric number represents the proportion of null alpha==0</p>
</td></tr>
<tr><td><code>alpha2</code></td>
<td>
<p>A numeric number represents the proportion of null beta==0</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James Y. Dai and X. Wang
</p>


<h3>References</h3>

<p>James Y. Dai, Janet L. Stanford, Michael LeBlanc. A multiple-testing 
procedure for high-dimensional mediation hypotheses, 
Journal of the American Statistical Association, 2020, DOI: 10.1080/01621459.2020.1765785. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp_input)
input_pvalues &lt;- snp_input
#To save computing time for illustration, we use 10 percent of rows (p-values)
input_pvalues &lt;- input_pvalues[sample(1:nrow(input_pvalues),
size &lt;- ceiling(nrow(input_pvalues)/10)),]

nullprop &lt;- null_estimation(input_pvalues)
</code></pre>

<hr>
<h2 id='snp_input'>
An example dataset to demonstrate the usage of 'HDMT'
</h2><span id='topic+snp_input'></span>

<h3>Description</h3>

<p>This example dataset is included in 'HDMT' to assess the mediation role of DNA methylation 
in genetic regulation of gene expression in primary prostate 
cancer (PCa) samples from The Cancer Genome Atlas (TCGA) with risk SNPs 
as the exposure.
</p>
<p>The dataset is a matrix containing two columns of p-values for 
candidate mediators. Column 1 contains the p-values for testing if an exposure is associated with 
the mediator (alpha!=0). Column 2 contains the p-value for testing if a mediator is associated with 
the outcome after adjusted for the exposure (beta!=0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("snp_input")</code></pre>


<h3>Format</h3>

<p>The format of snp_input is:
num [1:69602, 1:2] 0.106 0.999 0.101 0.173 0.89 ...
</p>


<h3>References</h3>

<p>James Y. Dai, Janet L. Stanford, Michael LeBlanc. A multiple-testing 
procedure for high-dimensional mediation hypotheses, 
Journal of the American Statistical Association, 2020, DOI: 10.1080/01621459.2020.1765785. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snp_input)
dim(snp_input)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
