<!DOCTYPE html><html lang="en-US"><head><title>Help for package HDNRA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HDNRA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HDNRA-package'><p>HDNRA: High-Dimensional Location Testing with Normal-Reference Approaches</p></a></li>
<li><a href='#BS1996.TS.NABT'><p>Normal-approximation-based test for two-sample problem proposed by Bai and Saranadasa (1996)</p></a></li>
<li><a href='#corneal'><p>HDNRA_data corneal</p></a></li>
<li><a href='#COVID19'><p>HDNRA_data COVID19</p></a></li>
<li><a href='#CQ2010.TSBF.NABT'><p>Normal-approximation-based test for two-sample BF problem proposed by Chen and Qin (2010)</p></a></li>
<li><a href='#FHW2004.GLHT.NABT'><p>Normal-approximation-based test for GLHT problem proposed by Fujikoshi et al. (2004)</p></a></li>
<li><a href='#NRtest.object'><p>S3 Class &quot;NRtest&quot;</p></a></li>
<li><a href='#print.NRtest'><p>Print Method for S3 Class &quot;NRtest&quot;</p></a></li>
<li><a href='#S2007.ks.NABT'><p>Normal-approximation-based test for one-way MANOVA problem proposed by Schott (2007)</p></a></li>
<li><a href='#SD2008.TS.NABT'><p>Normal-approximation-based test for two-sample problem proposed by Srivastava and Du (2008)</p></a></li>
<li><a href='#SF2006.GLHT.NABT'><p>Normal-approximation-based test for GLHT problem proposed by Srivastava and Fujikoshi (2006)</p></a></li>
<li><a href='#SKK2013.TSBF.NABT'><p>Normal-approximation-based test for two-sample BF problem proposed by Srivastava et al. (2013)</p></a></li>
<li><a href='#YS2012.GLHT.NABT'><p>Normal-approximation-based test for GLHT problem proposed by Yamada and Srivastava (2012)</p></a></li>
<li><a href='#ZGZ2017.GLHT.2cNRT'><p>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for GLHT problem proposed Zhang et al. (2017)</p></a></li>
<li><a href='#ZGZ2017.GLHTBF.NABT'><p>Normal-approximation-based test for GLHT problem under heteroscedasticity proposed by Zhou et al. (2017)</p></a></li>
<li><a href='#ZGZC2020.TS.2cNRT'><p>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for two-sample problem proposed by Zhang et al. (2020)</p></a></li>
<li><a href='#ZWZ2023.TSBF.2cNRT'><p>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for two-sample BF problem proposed by Zhu et al. (2023)</p></a></li>
<li><a href='#ZZ2022.GLHT.3cNRT'><p>Normal-reference-test with three-cumulant (3-c) matched $\chi^2$-approximation for GLHT problem proposed by Zhu and Zhang (2022)</p></a></li>
<li><a href='#ZZ2022.GLHTBF.3cNRT'><p>Normal-reference-test with three-cumulant (3-c) matched $\chi^2$-approximation for GLHT problem under heteroscedasticity proposed by Zhang and Zhu (2022)</p></a></li>
<li><a href='#ZZ2022.TS.3cNRT'><p>Normal-reference-test with three-cumulant (3-c) matched $\chi^2$-approximation for two-sample problem proposed by Zhang and Zhu (2022)</p></a></li>
<li><a href='#ZZ2022.TSBF.3cNRT'><p>Normal-reference-test with three-cumulant (3-c) matched $\chi^2$-approximation for two-sample BF problem proposed by Zhang and Zhu (2022)</p></a></li>
<li><a href='#ZZG2022.GLHTBF.2cNRT'><p>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for GLHT problem under heteroscedasticity proposed by Zhang et al. (2022)</p></a></li>
<li><a href='#ZZGZ2021.TSBF.2cNRT'><p>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for two-sample BF problem proposed by Zhang et al. (2021)</p></a></li>
<li><a href='#ZZZ2020.TS.2cNRT'><p>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for two-sample problem proposed by Zhang et al. (2020)</p></a></li>
<li><a href='#ZZZ2022.GLHT.2cNRT'><p>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for GLHT problem proposed by Zhu et al. (2022)</p></a></li>
<li><a href='#ZZZ2023.TSBF.2cNRT'><p>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for two-sample BF problem proposed by Zhang et al. (2023)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>High-Dimensional Location Testing with Normal-Reference
Approaches</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-10-22</td>
</tr>
<tr>
<td>Author:</td>
<td>Pengfei Wang [aut, cre],
  Shuqi Luo [aut],
  Tianming Zhu [aut],
  Bu Zhou [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Pengfei Wang &lt;nie23.wp8738@e.ntu.edu.sg&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We provide a collection of various classical tests and latest normal-reference tests for comparing high-dimensional mean vectors including two-sample and general linear hypothesis testing (GLHT) problem. Some existing tests for two-sample problem [see Bai, Zhidong, and Hewa Saranadasa.(1996) <a href="https://www.jstor.org/stable/24306018">https://www.jstor.org/stable/24306018</a>; Chen, Song Xi, and Ying-Li Qin.(2010) &lt;<a href="https://doi.org/10.1214%2F09-aos716">doi:10.1214/09-aos716</a>&gt;; Srivastava, Muni S., and Meng Du.(2008) &lt;<a href="https://doi.org/10.1016%2Fj.jmva.2006.11.002">doi:10.1016/j.jmva.2006.11.002</a>&gt;; Srivastava, Muni S., Shota Katayama, and Yutaka Kano.(2013)&lt;<a href="https://doi.org/10.1016%2Fj.jmva.2012.08.014">doi:10.1016/j.jmva.2012.08.014</a>&gt;]. Normal-reference tests for two-sample problem [see Zhang, Jin-Ting, Jia Guo, Bu Zhou, and Ming-Yen Cheng.(2020) &lt;<a href="https://doi.org/10.1080%2F01621459.2019.1604366">doi:10.1080/01621459.2019.1604366</a>&gt;; Zhang, Jin-Ting, Bu Zhou, Jia Guo, and Tianming Zhu.(2021) &lt;<a href="https://doi.org/10.1016%2Fj.jspi.2020.11.008">doi:10.1016/j.jspi.2020.11.008</a>&gt;; Zhang, Liang, Tianming Zhu, and Jin-Ting Zhang.(2020) &lt;<a href="https://doi.org/10.1016%2Fj.ecosta.2019.12.002">doi:10.1016/j.ecosta.2019.12.002</a>&gt;; Zhang, Liang, Tianming Zhu, and Jin-Ting Zhang.(2023) &lt;<a href="https://doi.org/10.1080%2F02664763.2020.1834516">doi:10.1080/02664763.2020.1834516</a>&gt;; Zhang, Jin-Ting, and Tianming Zhu.(2022) &lt;<a href="https://doi.org/10.1080%2F10485252.2021.2015768">doi:10.1080/10485252.2021.2015768</a>&gt;; Zhang, Jin-Ting, and Tianming Zhu.(2022) &lt;<a href="https://doi.org/10.1007%2Fs42519-021-00232-w">doi:10.1007/s42519-021-00232-w</a>&gt;; Zhu, Tianming, Pengfei Wang, and Jin-Ting Zhang.(2023) &lt;<a href="https://doi.org/10.1007%2Fs00180-023-01433-6">doi:10.1007/s00180-023-01433-6</a>&gt;]. Some existing tests for GLHT problem [see Fujikoshi, Yasunori, Tetsuto Himeno, and Hirofumi Wakaki.(2004) &lt;<a href="https://doi.org/10.14490%2Fjjss.34.19">doi:10.14490/jjss.34.19</a>&gt;; Srivastava, Muni S., and Yasunori Fujikoshi.(2006) &lt;<a href="https://doi.org/10.1016%2Fj.jmva.2005.08.010">doi:10.1016/j.jmva.2005.08.010</a>&gt;; Yamada, Takayuki, and Muni S. Srivastava.(2012) &lt;<a href="https://doi.org/10.1080%2F03610926.2011.581786">doi:10.1080/03610926.2011.581786</a>&gt;; Schott, James R.(2007) &lt;<a href="https://doi.org/10.1016%2Fj.jmva.2006.11.007">doi:10.1016/j.jmva.2006.11.007</a>&gt;; Zhou, Bu, Jia Guo, and Jin-Ting Zhang.(2017) &lt;<a href="https://doi.org/10.1016%2Fj.jspi.2017.03.005">doi:10.1016/j.jspi.2017.03.005</a>&gt;]. Normal-reference  tests for GLHT problem [see Zhang, Jin-Ting, Jia Guo, and Bu Zhou.(2017) &lt;<a href="https://doi.org/10.1016%2Fj.jmva.2017.01.002">doi:10.1016/j.jmva.2017.01.002</a>&gt;; Zhang, Jin-Ting, Bu Zhou, and Jia Guo.(2022) &lt;<a href="https://doi.org/10.1016%2Fj.jmva.2021.104816">doi:10.1016/j.jmva.2021.104816</a>&gt;; Zhu, Tianming, Liang Zhang, and Jin-Ting Zhang.(2022) &lt;<a href="https://doi.org/10.5705%2Fss.202020.0362">doi:10.5705/ss.202020.0362</a>&gt;; Zhu, Tianming, and Jin-Ting Zhang.(2022) &lt;<a href="https://doi.org/10.1007%2Fs00180-021-01110-6">doi:10.1007/s00180-021-01110-6</a>&gt;; Zhang, Jin-Ting, and Tianming Zhu.(2022) &lt;<a href="https://doi.org/10.1016%2Fj.csda.2021.107385">doi:10.1016/j.csda.2021.107385</a>&gt;].</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://nie23wp8738.github.io/HDNRA/">https://nie23wp8738.github.io/HDNRA/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/nie23wp8738/HDNRA/issues">https://github.com/nie23wp8738/HDNRA/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Imports:</td>
<td>expm, Rcpp, Rdpack, readr, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>devtools, dplyr, knitr, rmarkdown, spelling, testthat (&ge;
3.0.0), tidyr</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-22 06:45:12 UTC; yehe</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-22 08:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='HDNRA-package'>HDNRA: High-Dimensional Location Testing with Normal-Reference Approaches</h2><span id='topic+HDNRA'></span><span id='topic+HDNRA-package'></span>

<h3>Description</h3>

<p>We provide a collection of various classical tests and latest normal-reference tests for comparing high-dimensional mean vectors including two-sample and general linear hypothesis testing (GLHT) problem. Some existing tests for two-sample problem [see Bai, Zhidong, and Hewa Saranadasa.(1996) <a href="https://www.jstor.org/stable/24306018">https://www.jstor.org/stable/24306018</a>; Chen, Song Xi, and Ying-Li Qin.(2010) <a href="https://doi.org/10.1214/09-aos716">doi:10.1214/09-aos716</a>; Srivastava, Muni S., and Meng Du.(2008) <a href="https://doi.org/10.1016/j.jmva.2006.11.002">doi:10.1016/j.jmva.2006.11.002</a>; Srivastava, Muni S., Shota Katayama, and Yutaka Kano.(2013)<a href="https://doi.org/10.1016/j.jmva.2012.08.014">doi:10.1016/j.jmva.2012.08.014</a>]. Normal-reference tests for two-sample problem [see Zhang, Jin-Ting, Jia Guo, Bu Zhou, and Ming-Yen Cheng.(2020) <a href="https://doi.org/10.1080/01621459.2019.1604366">doi:10.1080/01621459.2019.1604366</a>; Zhang, Jin-Ting, Bu Zhou, Jia Guo, and Tianming Zhu.(2021) <a href="https://doi.org/10.1016/j.jspi.2020.11.008">doi:10.1016/j.jspi.2020.11.008</a>; Zhang, Liang, Tianming Zhu, and Jin-Ting Zhang.(2020) <a href="https://doi.org/10.1016/j.ecosta.2019.12.002">doi:10.1016/j.ecosta.2019.12.002</a>; Zhang, Liang, Tianming Zhu, and Jin-Ting Zhang.(2023) <a href="https://doi.org/10.1080/02664763.2020.1834516">doi:10.1080/02664763.2020.1834516</a>; Zhang, Jin-Ting, and Tianming Zhu.(2022) <a href="https://doi.org/10.1080/10485252.2021.2015768">doi:10.1080/10485252.2021.2015768</a>; Zhang, Jin-Ting, and Tianming Zhu.(2022) <a href="https://doi.org/10.1007/s42519-021-00232-w">doi:10.1007/s42519-021-00232-w</a>; Zhu, Tianming, Pengfei Wang, and Jin-Ting Zhang.(2023) <a href="https://doi.org/10.1007/s00180-023-01433-6">doi:10.1007/s00180-023-01433-6</a>]. Some existing tests for GLHT problem [see Fujikoshi, Yasunori, Tetsuto Himeno, and Hirofumi Wakaki.(2004) <a href="https://doi.org/10.14490/jjss.34.19">doi:10.14490/jjss.34.19</a>; Srivastava, Muni S., and Yasunori Fujikoshi.(2006) <a href="https://doi.org/10.1016/j.jmva.2005.08.010">doi:10.1016/j.jmva.2005.08.010</a>; Yamada, Takayuki, and Muni S. Srivastava.(2012) <a href="https://doi.org/10.1080/03610926.2011.581786">doi:10.1080/03610926.2011.581786</a>; Schott, James R.(2007) <a href="https://doi.org/10.1016/j.jmva.2006.11.007">doi:10.1016/j.jmva.2006.11.007</a>; Zhou, Bu, Jia Guo, and Jin-Ting Zhang.(2017) <a href="https://doi.org/10.1016/j.jspi.2017.03.005">doi:10.1016/j.jspi.2017.03.005</a>]. Normal-reference tests for GLHT problem [see Zhang, Jin-Ting, Jia Guo, and Bu Zhou.(2017) <a href="https://doi.org/10.1016/j.jmva.2017.01.002">doi:10.1016/j.jmva.2017.01.002</a>; Zhang, Jin-Ting, Bu Zhou, and Jia Guo.(2022) <a href="https://doi.org/10.1016/j.jmva.2021.104816">doi:10.1016/j.jmva.2021.104816</a>; Zhu, Tianming, Liang Zhang, and Jin-Ting Zhang.(2022) <a href="https://doi.org/10.5705/ss.202020.0362">doi:10.5705/ss.202020.0362</a>; Zhu, Tianming, and Jin-Ting Zhang.(2022) <a href="https://doi.org/10.1007/s00180-021-01110-6">doi:10.1007/s00180-021-01110-6</a>; Zhang, Jin-Ting, and Tianming Zhu.(2022) <a href="https://doi.org/10.1016/j.csda.2021.107385">doi:10.1016/j.csda.2021.107385</a>].
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Pengfei Wang <a href="mailto:nie23.wp8738@e.ntu.edu.sg">nie23.wp8738@e.ntu.edu.sg</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Shuqi Luo <a href="mailto:nie23.ls4909@e.ntu.edu.sg">nie23.ls4909@e.ntu.edu.sg</a>
</p>
</li>
<li><p> Tianming Zhu <a href="mailto:tianming.zhu@nie.edu.sg">tianming.zhu@nie.edu.sg</a>
</p>
</li>
<li><p> Bu Zhou <a href="mailto:bu.zhou@u.nus.edu">bu.zhou@u.nus.edu</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://nie23wp8738.github.io/HDNRA/">https://nie23wp8738.github.io/HDNRA/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/nie23wp8738/HDNRA/issues">https://github.com/nie23wp8738/HDNRA/issues</a>
</p>
</li></ul>


<hr>
<h2 id='BS1996.TS.NABT'>Normal-approximation-based test for two-sample problem proposed by Bai and Saranadasa (1996)</h2><span id='topic+BS1996.TS.NABT'></span>

<h3>Description</h3>

<p>Bai and Saranadasa (1996)'s test for testing equality of two-sample high-dimensional mean vectors with assuming that two covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BS1996.TS.NABT(y1, y2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BS1996.TS.NABT_+3A_y1">y1</code></td>
<td>
<p>The data matrix (<code class="reqn">n_1 \times p</code>) from the first population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="BS1996.TS.NABT_+3A_y2">y2</code></td>
<td>
<p>The data matrix (<code class="reqn">n_2 \times p</code>) from the second population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have two independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma},i=1,2.
</code>
</p>

<p>The primary object is to test
</p>
<p style="text-align: center;"><code class="reqn">H_{0}: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2\; \operatorname{versus}\; H_{1}: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2.</code>
</p>

<p>Bai and Saranadasa (1996) proposed the following centralised <code class="reqn">L^2</code>-norm-based test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{BS} = \frac{n_1n_2}{n} \|\bar{\boldsymbol{y}}_1 - \bar{\boldsymbol{y}}_2\|^2-\operatorname{tr}(\hat{\boldsymbol{\Sigma}}),</code>
</p>

<p>where  <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,2</code> are the sample mean vectors and <code class="reqn">\hat{\boldsymbol{\Sigma}}</code> is the pooled sample covariance matrix.
They showed that under the null hypothesis, <code class="reqn">T_{BS}</code> is asymptotically normally distributed.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Bai Z, Saranadasa H (1996).
&ldquo;Effect of high dimension: by an example of a two sample problem.&rdquo;
<em>Statistica Sinica</em>, 311&ndash;329.
<a href="https://www.jstor.org/stable/24306018">https://www.jstor.org/stable/24306018</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("COVID19")
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
BS1996.TS.NABT(group1,group2)

</code></pre>

<hr>
<h2 id='corneal'>HDNRA_data corneal</h2><span id='topic+corneal'></span>

<h3>Description</h3>

<p>This dataset was acquired during a keratoconus study, a collaborative project involving Ms.Nancy Tripoli and Dr.Kenneth L.Cohen of Department of Ophthalmology at the University of North Carolina, Chapel Hill. The fitted feature vectors for the complete corneal surface dataset collectively into a feature matrix with dimensions of 150 × 2000.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(corneal)
</code></pre>


<h3>Format</h3>



<h4>'corneal'</h4>

<p>A data frame with 150 observations on the following 4 groups.
</p>

<dl>
<dt>normal group1</dt><dd><p>row 1 to row 43 in total 43 rows of the feature matrix correspond to observations from the normal group</p>
</dd>
<dt>unilateral suspect group2</dt><dd><p>row 44 to row 57 in total 14 rows of the feature matrix correspond to observations from the unilateral suspect group</p>
</dd>
<dt>suspect map group3</dt><dd><p>row 58 to row 78 in total 21 of the feature matrix correspond to observations from the suspect map group</p>
</dd>
<dt> clinical keratoconus group4</dt><dd><p>row 79 to row 150 in total 72 of the feature matrix correspond to observations from the clinical keratoconus group</p>
</dd>
</dl>




<h3>References</h3>

<p>Smaga Ł, Zhang J (2019).
&ldquo;Linear hypothesis testing with functional data.&rdquo;
<em>Technometrics</em>, <b>61</b>(1), 99&ndash;110.
<a href="https://doi.org/10.1080/00401706.2018.1456976">doi:10.1080/00401706.2018.1456976</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(HDNRA)
data(corneal)
dim(corneal)
group1 &lt;- as.matrix(corneal[1:43, ]) ## normal group
dim(group1)
group2 &lt;- as.matrix(corneal[44:57, ]) ## unilateral suspect group
dim(group2)
group3 &lt;- as.matrix(corneal[58:78, ]) ## suspect map group
dim(group3)
group4 &lt;- as.matrix(corneal[79:150, ]) ## clinical keratoconus group
dim(group4)
</code></pre>

<hr>
<h2 id='COVID19'>HDNRA_data COVID19</h2><span id='topic+COVID19'></span>

<h3>Description</h3>

<p>A COVID19 data set from NCBI with ID GSE152641. The data set profiled peripheral blood from 24 healthy controls and 62 prospectively enrolled patients with community-acquired lower respiratory tract infection by SARS-COV-2 within the first 24 hours of hospital admission using RNA sequencing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(COVID19)
</code></pre>


<h3>Format</h3>



<h4>'COVID19'</h4>

<p>A data frame with 86 observations on the following 2 groups.
</p>

<dl>
<dt>healthy group1</dt><dd><p>row 2 to row 19, and row 82 to 87, in total 24 healthy controls</p>
</dd>
<dt>patients group2</dt><dd><p>row 20 to 81, in total 62 prospectively enrolled patients</p>
</dd>
</dl>




<h3>References</h3>

<p>Thair SA, He YD, Hasin-Brumshtein Y, Sakaram S, Pandya R, Toh J, Rawling D, Remmel M, Coyle S, Dalekos GN, others (2021).
&ldquo;Transcriptomic similarities and differences in host response between SARS-CoV-2 and other viral infections.&rdquo;
<em>Iscience</em>, <b>24</b>(1).
<a href="https://doi.org/10.1016/j.isci.2020.101947">doi:10.1016/j.isci.2020.101947</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(HDNRA)
data(COVID19)
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
dim(group1)
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
dim(group2)
</code></pre>

<hr>
<h2 id='CQ2010.TSBF.NABT'>Normal-approximation-based test for two-sample BF problem proposed by Chen and Qin (2010)</h2><span id='topic+CQ2010.TSBF.NABT'></span>

<h3>Description</h3>

<p>Chen and Qin (2010)'s test for testing equality of two-sample high-dimensional mean vectors without assuming that two covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CQ2010.TSBF.NABT(y1, y2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CQ2010.TSBF.NABT_+3A_y1">y1</code></td>
<td>
<p>The data matrix (<code class="reqn">n_1 \times p</code>) from the first population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="CQ2010.TSBF.NABT_+3A_y2">y2</code></td>
<td>
<p>The data matrix (<code class="reqn">n_2 \times p</code>) from the second population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have two independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma}_i,i=1,2.
</code>
</p>

<p>The primary object is to test
</p>
<p style="text-align: center;"><code class="reqn">H_{0}: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2\; \operatorname{versus}\; H_{1}: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2.</code>
</p>

<p>Chen and Qin (2010) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{CQ} = \frac{\sum_{i \neq j}^{n_1} \boldsymbol{y}_{1i}^\top \boldsymbol{y}_{1j}}{n_1 (n_1 - 1)} + \frac{\sum_{i \neq j}^{n_2} \boldsymbol{y}_{2i}^\top \boldsymbol{y}_{2j}}{n_2 (n_2 - 1)} - 2 \frac{\sum_{i = 1}^{n_1} \sum_{j = 1}^{n_2} \boldsymbol{y}_{1i}^\top \boldsymbol{y}_{2j}}{n_1 n_2}.</code>
</p>

<p>They showed that under the null hypothesis, <code class="reqn">T_{CQ}</code> is asymptotically normally distributed.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Chen SX, Qin Y (2010).
&ldquo;A two-sample test for high-dimensional data with applications to gene-set testing.&rdquo;
<em>The Annals of Statistics</em>, <b>38</b>(2).
<a href="https://doi.org/10.1214/09-aos716">doi:10.1214/09-aos716</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("COVID19")
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
CQ2010.TSBF.NABT(group1,group2)


</code></pre>

<hr>
<h2 id='FHW2004.GLHT.NABT'>Normal-approximation-based test for GLHT problem proposed by Fujikoshi et al. (2004)</h2><span id='topic+FHW2004.GLHT.NABT'></span>

<h3>Description</h3>

<p>Fujikoshi et al. (2004)'s test for general linear hypothesis testing (GLHT) problem for high-dimensional data with assuming that underlying covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FHW2004.GLHT.NABT(Y,X,C,n,p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FHW2004.GLHT.NABT_+3A_y">Y</code></td>
<td>
<p>A list of <code class="reqn">k</code> data matrices.  The <code class="reqn">i</code>th element represents the data matrix (<code class="reqn">n_i \times p</code>) from the <code class="reqn">i</code>th population with each row representing a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="FHW2004.GLHT.NABT_+3A_x">X</code></td>
<td>
<p>A known <code class="reqn">n\times k</code> full-rank design matrix with <code class="reqn">\operatorname{rank}(\boldsymbol{X})=k&lt;n</code>.</p>
</td></tr>
<tr><td><code id="FHW2004.GLHT.NABT_+3A_c">C</code></td>
<td>
<p>A known matrix of size <code class="reqn">q\times k</code> with <code class="reqn">\operatorname{rank}(\boldsymbol{C})=q&lt;k</code>.</p>
</td></tr>
<tr><td><code id="FHW2004.GLHT.NABT_+3A_n">n</code></td>
<td>
<p>A vector of <code class="reqn">k</code> sample sizes. The <code class="reqn">i</code>th element represents the sample size of group <code class="reqn">i</code>, <code class="reqn">n_i</code>.</p>
</td></tr>
<tr><td><code id="FHW2004.GLHT.NABT_+3A_p">p</code></td>
<td>
<p>The dimension of data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A high-dimensional linear regression model can be expressed as
</p>
<p style="text-align: center;"><code class="reqn">\boldsymbol{Y}=\boldsymbol{X\Theta}+\boldsymbol{\epsilon},</code>
</p>

<p>where <code class="reqn">\Theta</code> is a <code class="reqn">k\times p</code> unknown parameter matrix and <code class="reqn">\boldsymbol{\epsilon}</code> is an <code class="reqn">n\times p</code> error matrix.
</p>
<p>It is of interest to test the following GLHT problem
</p>
<p style="text-align: center;"><code class="reqn">H_0: \boldsymbol{C\Theta}=\boldsymbol{0}, \quad \text { vs. } \quad H_1: \boldsymbol{C\Theta} \neq \boldsymbol{0}.</code>
</p>

<p>Fujikoshi et al. (2004) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{FHW}=\sqrt{p}\left[(n-k)\frac{\operatorname{tr}(\boldsymbol{S}_h)}{\operatorname{tr}(\boldsymbol{S}_e)}-q\right],</code>
</p>

<p>where <code class="reqn">\boldsymbol{S}_h</code> and <code class="reqn">\boldsymbol{S}_e</code> are the matrices of sums of squares and products due to the hypothesis and the error, respecitively.
</p>
<p>They showed that under the null hypothesis, <code class="reqn">T_{FHW}</code> is asymptotically normally distributed.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Fujikoshi Y, Himeno T, Wakaki H (2004).
&ldquo;Asymptotic results of a high dimensional MANOVA test and power comparison when the dimension is large compared to the sample size.&rdquo;
<em>Journal of the Japan Statistical Society</em>, <b>34</b>(1), 19&ndash;26.
<a href="https://doi.org/10.14490/jjss.34.19">doi:10.14490/jjss.34.19</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("corneal")
dim(corneal)
group1 &lt;- as.matrix(corneal[1:43, ]) ## normal group
group2 &lt;- as.matrix(corneal[44:57, ]) ## unilateral suspect group
group3 &lt;- as.matrix(corneal[58:78, ]) ## suspect map group
group4 &lt;- as.matrix(corneal[79:150, ]) ## clinical keratoconus group
p &lt;- dim(corneal)[2]
Y &lt;- list()
k &lt;- 4
Y[[1]] &lt;- group1
Y[[2]] &lt;- group2
Y[[3]] &lt;- group3
Y[[4]] &lt;- group4
n &lt;- c(nrow(Y[[1]]),nrow(Y[[2]]),nrow(Y[[3]]),nrow(Y[[4]]))
X &lt;- matrix(c(rep(1,n[1]),rep(0,sum(n)),rep(1,n[2]), rep(0,sum(n)),
            rep(1,n[3]),rep(0,sum(n)),rep(1,n[4])),ncol=k,nrow=sum(n))
q &lt;- k-1
C &lt;- cbind(diag(q),-rep(1,q))
FHW2004.GLHT.NABT(Y,X,C,n,p)
</code></pre>

<hr>
<h2 id='NRtest.object'>S3 Class &quot;NRtest&quot;</h2><span id='topic+NRtest.object'></span>

<h3>Description</h3>

<p>The <code>"NRtest"</code> objects provide a comprehensive summary of hypothesis test outcomes,
including test statistics, p-values, parameter estimates, and confidence intervals, if applicable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NRtest.object(
  statistic,
  p.value,
  method,
  null.value,
  alternative,
  parameter = NULL,
  sample.size = NULL,
  sample.dimension = NULL,
  estimation.method = NULL,
  data.name = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="NRtest.object_+3A_statistic">statistic</code></td>
<td>
<p>Numeric scalar containing the value of the test statistic, with a <code>names</code> attribute indicating the name of the test statistic.</p>
</td></tr>
<tr><td><code id="NRtest.object_+3A_p.value">p.value</code></td>
<td>
<p>Numeric scalar containing the p-value for the test.</p>
</td></tr>
<tr><td><code id="NRtest.object_+3A_method">method</code></td>
<td>
<p>Character string giving the name of the test.</p>
</td></tr>
<tr><td><code id="NRtest.object_+3A_null.value">null.value</code></td>
<td>
<p>Character string indicating the null hypothesis.</p>
</td></tr>
<tr><td><code id="NRtest.object_+3A_alternative">alternative</code></td>
<td>
<p>Character string indicating the alternative hypothesis.</p>
</td></tr>
<tr><td><code id="NRtest.object_+3A_parameter">parameter</code></td>
<td>
<p>Numeric vector containing the estimated approximation parameter(s) associated with the approximation method. This vector has a <code>names</code> attribute describing its element(s).</p>
</td></tr>
<tr><td><code id="NRtest.object_+3A_sample.size">sample.size</code></td>
<td>
<p>Numeric vector containing the number of observations in each group used for the hypothesis test.</p>
</td></tr>
<tr><td><code id="NRtest.object_+3A_sample.dimension">sample.dimension</code></td>
<td>
<p>Numeric scalar containing the dimension of the dataset used for the hypothesis test.</p>
</td></tr>
<tr><td><code id="NRtest.object_+3A_estimation.method">estimation.method</code></td>
<td>
<p>Character string giving the name of the approximation approach used to approximate the null distribution of the test statistic.</p>
</td></tr>
<tr><td><code id="NRtest.object_+3A_data.name">data.name</code></td>
<td>
<p>Character string describing the data set used in the hypothesis test.</p>
</td></tr>
<tr><td><code id="NRtest.object_+3A_...">...</code></td>
<td>
<p>Additional optional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A class of objects returned by high-dimensional hypothesis testing functions in the <span class="pkg">HDNRA</span> package,
designed to encapsulate detailed results from statistical hypothesis tests.
These objects are structured similarly to <span class="pkg">htest</span> objects in the package <span class="pkg">EnvStats</span> but are tailored
to the needs of the <span class="pkg">HDNRA</span> package.
</p>


<h3>Value</h3>

<p>An object of class <code>"NRtest"</code> containing both required and optional components depending on the specifics of the hypothesis test,
shown as follows:
</p>


<h3>Required Components</h3>

<p>These components must be present in every <code>"NRtest"</code> object:
</p>

<dl>
<dt><code>statistic</code></dt><dd><p>Must e present.</p>
</dd>
<dt><code>p.value</code></dt><dd><p>Must e present.</p>
</dd>
<dt><code>null.value</code></dt><dd><p>Must e present.</p>
</dd>
<dt><code>alternative</code></dt><dd><p>Must e present.</p>
</dd>
<dt><code>method</code></dt><dd><p>Must e present.</p>
</dd>
</dl>



<h3>Optional Components</h3>

<p>These components are included depending on the specifics of the hypothesis test performed:
</p>

<dl>
<dt><code>parameter</code></dt><dd><p>May be present.</p>
</dd>
<dt><code>sample.size</code></dt><dd><p>May be present.</p>
</dd>
<dt><code>sample.dimension</code></dt><dd><p>May be present.</p>
</dd>
<dt><code>estimation.method</code></dt><dd><p>May be present.</p>
</dd>
<dt><code>data.name</code></dt><dd><p>May be present.</p>
</dd>
</dl>



<h3>Methods</h3>

<p>The class has the following methods:
</p>

<dl>
<dt><code><a href="#topic+print.NRtest">print.NRtest</a></code></dt><dd><p>Printing the contents of the NRtest object in a human-readable form.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'># Example 1: Using Bai and Saranadasa (1996)'s test (two-sample problem)
NRtest.obj1 &lt;- NRtest.object(
  statistic = c("T[BS]" = 2.208),
  p.value = 0.0136,
  method = "Bai and Saranadasa (1996)'s test",
  data.name = "group1 and group2",
  null.value = c("Difference between two mean vectors is o"),
  alternative = "Difference between two mean vectors is not 0",
  parameter = NULL,
  sample.size = c(n1 = 24, n2 = 26),
  sample.dimension = 20460,
  estimation.method = "Normal approximation"
)
print(NRtest.obj1)

# Example 2: Using Fujikoshi et al. (2004)'s test (GLHT problem)
NRtest.obj2 &lt;- NRtest.object(
  statistic = c("T[FHW]" = 6.4015),
  p.value = 0,
  method = "Fujikoshi et al. (2004)'s test",
  data.name = "Y",
  null.value  = "The general linear hypothesis is true",
  alternative = "The general linear hypothesis is not true",
  parameter = NULL,
  sample.size = c(n1 = 43, n2 = 14, n3 = 21, n4 = 72),
  sample.dimension = 2000,
  estimation.method = "Normal approximation"
)
print(NRtest.obj2)

</code></pre>

<hr>
<h2 id='print.NRtest'>Print Method for S3 Class &quot;NRtest&quot;</h2><span id='topic+print.NRtest'></span>

<h3>Description</h3>

<p>Prints the details of the <span class="pkg">NRtest</span> object in a user-friendly manner. This method
provides a clear and concise presentation of the test results contained within
the <span class="pkg">NRtest</span> object, including all relevant statistical metrics and test details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class \pkg{NRtest}
## S3 method for class 'NRtest'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.NRtest_+3A_x">x</code></td>
<td>
<p>an <span class="pkg">NRtest</span> object.</p>
</td></tr>
<tr><td><code id="print.NRtest_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>print.NRtest</code> function formats and presents the contents of the <span class="pkg">NRtest</span>
object, which includes statistical test results and related parameters. This
function is designed to provide a user-friendly display of the object's
contents, making it easier to understand the results of the analysis.
</p>


<h3>Value</h3>

<p>Invisibly returns the input <code>x</code>.
</p>


<h3>Author(s)</h3>

<p>Pengfei Wang <a href="mailto:nie23.wp8738@e.ntu.edu.sg">nie23.wp8738@e.ntu.edu.sg</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+NRtest.object">NRtest.object</a></code>
</p>

<hr>
<h2 id='S2007.ks.NABT'>Normal-approximation-based test for one-way MANOVA problem proposed by Schott (2007)</h2><span id='topic+S2007.ks.NABT'></span>

<h3>Description</h3>

<p>Schott, J. R. (2007)'s test for one-way MANOVA problem for high-dimensional data with assuming that underlying covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>S2007.ks.NABT(Y, n, p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="S2007.ks.NABT_+3A_y">Y</code></td>
<td>
<p>A list of <code class="reqn">k</code> data matrices.  The <code class="reqn">i</code>th element represents the data matrix (<code class="reqn">n_i \times p</code>) from the <code class="reqn">i</code>th population with each row representing a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="S2007.ks.NABT_+3A_n">n</code></td>
<td>
<p>A vector of <code class="reqn">k</code> sample sizes. The <code class="reqn">i</code>th element represents the sample size of group <code class="reqn">i</code>, <code class="reqn">n_i</code>.</p>
</td></tr>
<tr><td><code id="S2007.ks.NABT_+3A_p">p</code></td>
<td>
<p>The dimension of data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have the following <code class="reqn">k</code> independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma},i=1,\ldots,k.
</code>
</p>

<p>It is of interest to test the following one-way MANOVA problem:
</p>
<p style="text-align: center;"><code class="reqn">H_0: \boldsymbol{\mu}_1=\cdots=\boldsymbol{\mu}_k, \quad \text { vs. }\; H_1: H_0 \;\operatorname{is \; not\; ture}.</code>
</p>

<p>Schott (2007) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">
 T_{S}=[\operatorname{tr}(\boldsymbol{H})/h-\operatorname{tr}(\boldsymbol{E})/e]/\sqrt{N-1},
 </code>
</p>

<p>where <code class="reqn">\boldsymbol{H}=\sum_{i=1}^kn_i(\bar{\boldsymbol{y}}_i-\bar{\boldsymbol{y}})(\bar{\boldsymbol{y}}_i-\bar{\boldsymbol{y}})^\top</code>, <code class="reqn">\boldsymbol{E}=\sum_{i=1}^k\sum_{j=1}^{n_i}(\boldsymbol{y}_{ij}-\bar{\boldsymbol{y}}_{i})(\boldsymbol{y}_{ij}-\bar{\boldsymbol{y}}_{i})^\top</code>, <code class="reqn">h=k-1</code>, and <code class="reqn">e=N-k</code>, with <code class="reqn">N=n_1+\cdots+n_k</code>.
They showed that under the null hypothesis, <code class="reqn">T_{S}</code> is asymptotically normally distributed.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Schott JR (2007).
&ldquo;Some high-dimensional tests for a one-way MANOVA.&rdquo;
<em>Journal of Multivariate Analysis</em>, <b>98</b>(9), 1825&ndash;1839.
<a href="https://doi.org/10.1016/j.jmva.2006.11.007">doi:10.1016/j.jmva.2006.11.007</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("corneal")
dim(corneal)
group1 &lt;- as.matrix(corneal[1:43, ]) ## normal group
group2 &lt;- as.matrix(corneal[44:57, ]) ## unilateral suspect group
group3 &lt;- as.matrix(corneal[58:78, ]) ## suspect map group
group4 &lt;- as.matrix(corneal[79:150, ]) ## clinical keratoconus group
p &lt;- dim(corneal)[2]
Y &lt;- list()
Y[[1]] &lt;- group1
Y[[2]] &lt;- group2
Y[[3]] &lt;- group3
Y[[4]] &lt;- group4
n &lt;- c(nrow(Y[[1]]),nrow(Y[[2]]),nrow(Y[[3]]),nrow(Y[[4]]))
S2007.ks.NABT(Y, n, p)

</code></pre>

<hr>
<h2 id='SD2008.TS.NABT'>Normal-approximation-based test for two-sample problem proposed by Srivastava and Du (2008)</h2><span id='topic+SD2008.TS.NABT'></span>

<h3>Description</h3>

<p>Srivastava and Du (2008)'s test for testing equality of two-sample high-dimensional mean vectors with assuming that two covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SD2008.TS.NABT(y1, y2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SD2008.TS.NABT_+3A_y1">y1</code></td>
<td>
<p>The data matrix (<code class="reqn">n_1 \times p</code>) from the first population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="SD2008.TS.NABT_+3A_y2">y2</code></td>
<td>
<p>The data matrix (<code class="reqn">n_2 \times p</code>) from the second population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have two independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma},i=1,2.
</code>
</p>

<p>The primary object is to test
</p>
<p style="text-align: center;"><code class="reqn">H_{0}: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2\; \operatorname{versus}\; H_{1}: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2.</code>
</p>

<p>Srivastava and Du (2008) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{SD} = \frac{n^{-1}n_1n_2(\bar{\boldsymbol{y}}_1 - \bar{\boldsymbol{y}}_2)^\top \boldsymbol{D}_S^{-1}(\bar{\boldsymbol{y}}_1 - \bar{\boldsymbol{y}}_2) - \frac{(n-2)p}{n-4}}{\sqrt{2 \left[\operatorname{tr}(\boldsymbol{R}^2) - \frac{p^2}{n-2}\right] c_{p, n}}},</code>
</p>

<p>where <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,2</code> are the sample mean vectors,  <code class="reqn">\boldsymbol{D}_S</code> is the diagonal matrix of sample variance, <code class="reqn">\boldsymbol{R}</code> is the sample correlation matrix and <code class="reqn">c_{p, n}</code> is the adjustment coefficient proposed by Srivastava and Du (2008).
They showed that under the null hypothesis, <code class="reqn">T_{SD}</code> is asymptotically normally distributed.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Srivastava MS, Du M (2008).
&ldquo;A test for the mean vector with fewer observations than the dimension.&rdquo;
<em>Journal of Multivariate Analysis</em>, <b>99</b>(3), 386&ndash;402.
<a href="https://doi.org/10.1016/j.jmva.2006.11.002">doi:10.1016/j.jmva.2006.11.002</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("COVID19")
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
SD2008.TS.NABT(group1,group2)


</code></pre>

<hr>
<h2 id='SF2006.GLHT.NABT'>Normal-approximation-based test for GLHT problem proposed by Srivastava and Fujikoshi (2006)</h2><span id='topic+SF2006.GLHT.NABT'></span>

<h3>Description</h3>

<p>Srivastava and Fujikoshi (2006)'s test for general linear hypothesis testing (GLHT) problem for high-dimensional data with assuming that underlying covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SF2006.GLHT.NABT(Y,X,C,n,p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SF2006.GLHT.NABT_+3A_y">Y</code></td>
<td>
<p>A list of <code class="reqn">k</code> data matrices.  The <code class="reqn">i</code>th element represents the data matrix (<code class="reqn">n_i \times p</code>) from the <code class="reqn">i</code>th population with each row representing a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="SF2006.GLHT.NABT_+3A_x">X</code></td>
<td>
<p>A known <code class="reqn">n\times k</code> full-rank design matrix with <code class="reqn">\operatorname{rank}(\boldsymbol{X})=k&lt;n</code>.</p>
</td></tr>
<tr><td><code id="SF2006.GLHT.NABT_+3A_c">C</code></td>
<td>
<p>A known matrix of size <code class="reqn">q\times k</code> with <code class="reqn">\operatorname{rank}(\boldsymbol{C})=q&lt;k</code>.</p>
</td></tr>
<tr><td><code id="SF2006.GLHT.NABT_+3A_n">n</code></td>
<td>
<p>A vector of <code class="reqn">k</code> sample sizes. The <code class="reqn">i</code>th element represents the sample size of group <code class="reqn">i</code>, <code class="reqn">n_i</code>.</p>
</td></tr>
<tr><td><code id="SF2006.GLHT.NABT_+3A_p">p</code></td>
<td>
<p>The dimension of data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A high-dimensional linear regression model can be expressed as
</p>
<p style="text-align: center;"><code class="reqn">\boldsymbol{Y}=\boldsymbol{X\Theta}+\boldsymbol{\epsilon},</code>
</p>

<p>where <code class="reqn">\Theta</code> is a <code class="reqn">k\times p</code> unknown parameter matrix and <code class="reqn">\boldsymbol{\epsilon}</code> is an <code class="reqn">n\times p</code> error matrix.
</p>
<p>It is of interest to test the following GLHT problem
</p>
<p style="text-align: center;"><code class="reqn">H_0: \boldsymbol{C\Theta}=\boldsymbol{0}, \quad \text { vs. } \quad H_1: \boldsymbol{C\Theta} \neq \boldsymbol{0}.</code>
</p>

<p>Srivastava and Fujikoshi (2006) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{SF}=\left[2q\hat{a}_2(1+(n-k)^{-1}q)\right]^{-1/2}\left[\frac{\operatorname{tr}(\boldsymbol{B})}{\sqrt{p}}-\frac{q}{\sqrt{n-k}}\frac{\operatorname{tr}(\boldsymbol{W})}{\sqrt{(n-k)p}}\right].</code>
</p>

<p>where <code class="reqn">\boldsymbol{W}</code> and <code class="reqn">\boldsymbol{B}</code> are the matrix of sum of squares and products due to error and the error, respectively, and <code class="reqn">\hat{a}_2=[\operatorname{tr}(\boldsymbol{W}^2)-\operatorname{tr}^2(\boldsymbol{W})/(n-k)]/[(n-k-1)(n-k+2)p]</code>.
They showed that under the null hypothesis, <code class="reqn">T_{SF}</code> is asymptotically normally distributed.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Srivastava MS, Fujikoshi Y (2006).
&ldquo;Multivariate analysis of variance with fewer observations than the dimension.&rdquo;
<em>Journal of Multivariate Analysis</em>, <b>97</b>(9), 1927&ndash;1940.
<a href="https://doi.org/10.1016/j.jmva.2005.08.010">doi:10.1016/j.jmva.2005.08.010</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("corneal")
dim(corneal)
group1 &lt;- as.matrix(corneal[1:43, ]) ## normal group
group2 &lt;- as.matrix(corneal[44:57, ]) ## unilateral suspect group
group3 &lt;- as.matrix(corneal[58:78, ]) ## suspect map group
group4 &lt;- as.matrix(corneal[79:150, ]) ## clinical keratoconus group
p &lt;- dim(corneal)[2]
Y &lt;- list()
k &lt;- 4
Y[[1]] &lt;- group1
Y[[2]] &lt;- group2
Y[[3]] &lt;- group3
Y[[4]] &lt;- group4
n &lt;- c(nrow(Y[[1]]),nrow(Y[[2]]),nrow(Y[[3]]),nrow(Y[[4]]))
X &lt;- matrix(c(rep(1,n[1]),rep(0,sum(n)),rep(1,n[2]), rep(0,sum(n)),
            rep(1,n[3]),rep(0,sum(n)),rep(1,n[4])),ncol=k,nrow=sum(n))
q &lt;- k-1
C &lt;- cbind(diag(q),-rep(1,q))
SF2006.GLHT.NABT(Y,X,C,n,p)


</code></pre>

<hr>
<h2 id='SKK2013.TSBF.NABT'>Normal-approximation-based test for two-sample BF problem proposed by Srivastava et al. (2013)</h2><span id='topic+SKK2013.TSBF.NABT'></span>

<h3>Description</h3>

<p>Srivastava et al. (2013)'s test for testing equality of two-sample high-dimensional mean vectors without assuming that two covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SKK2013.TSBF.NABT(y1, y2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SKK2013.TSBF.NABT_+3A_y1">y1</code></td>
<td>
<p>The data matrix (<code class="reqn">n_1 \times p</code>) from the first population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="SKK2013.TSBF.NABT_+3A_y2">y2</code></td>
<td>
<p>The data matrix (<code class="reqn">n_2 \times p</code> from the second population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have two independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma}_i,i=1,2.
</code>
</p>

<p>The primary object is to test
</p>
<p style="text-align: center;"><code class="reqn">H_{0}: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2\; \operatorname{versus}\; H_{1}: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2.</code>
</p>

<p>Srivastava et al. (2013) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{SKK} = \frac{(\bar{\boldsymbol{y}}_1 - \bar{\boldsymbol{y}}_2)^\top \hat{\boldsymbol{D}}^{-1}(\bar{\boldsymbol{y}}_1 - \bar{\boldsymbol{y}}_2) - p}{\sqrt{2 \widehat{\operatorname{Var}}(\hat{q}_n) c_{p,n}}},</code>
</p>

<p>where <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,2</code> are the sample mean vectors, <code class="reqn">\hat{\boldsymbol{D}}=\hat{\boldsymbol{D}}_1/n_1+\hat{\boldsymbol{D}}_2/n_2</code> with <code class="reqn">\hat{\boldsymbol{D}}_i,i=1,2</code> being the diagonal matrices consisting of only the diagonal elements of the sample covariance matrices. <code class="reqn">\widehat{\operatorname{Var}}(\hat{q}_n)</code> is given by equation (1.18) in Srivastava et al. (2013), and  <code class="reqn">c_{p, n}</code> is the adjustment coefficient proposed by Srivastava et al. (2013).
They showed that under the null hypothesis, <code class="reqn">T_{SKK}</code> is asymptotically normally distributed.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Srivastava MS, Katayama S, Kano Y (2013).
&ldquo;A two sample test in high dimensional data.&rdquo;
<em>Journal of Multivariate Analysis</em>, <b>114</b>, 349&ndash;358.
<a href="https://doi.org/10.1016/j.jmva.2012.08.014">doi:10.1016/j.jmva.2012.08.014</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("COVID19")
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
SKK2013.TSBF.NABT(group1,group2)


</code></pre>

<hr>
<h2 id='YS2012.GLHT.NABT'>Normal-approximation-based test for GLHT problem proposed by Yamada and Srivastava (2012)</h2><span id='topic+YS2012.GLHT.NABT'></span>

<h3>Description</h3>

<p>Yamada and Srivastava (2012)'test for general linear hypothesis testing (GLHT) problem for high-dimensional data with assuming that underlying covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>YS2012.GLHT.NABT(Y,X,C,n,p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="YS2012.GLHT.NABT_+3A_y">Y</code></td>
<td>
<p>A list of <code class="reqn">k</code> data matrices.  The <code class="reqn">i</code>th element represents the data matrix (<code class="reqn">n_i \times p</code>) from the <code class="reqn">i</code>th population with each row representing a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="YS2012.GLHT.NABT_+3A_x">X</code></td>
<td>
<p>A known <code class="reqn">n\times k</code> full-rank design matrix with <code class="reqn">\operatorname{rank}(\boldsymbol{X})=k&lt;n</code>.</p>
</td></tr>
<tr><td><code id="YS2012.GLHT.NABT_+3A_c">C</code></td>
<td>
<p>A known matrix of size <code class="reqn">q\times k</code> with <code class="reqn">\operatorname{rank}(\boldsymbol{C})=q&lt;k</code>.</p>
</td></tr>
<tr><td><code id="YS2012.GLHT.NABT_+3A_n">n</code></td>
<td>
<p>A vector of <code class="reqn">k</code> sample sizes. The <code class="reqn">i</code>th element represents the sample size of group <code class="reqn">i</code>, <code class="reqn">n_i</code>.</p>
</td></tr>
<tr><td><code id="YS2012.GLHT.NABT_+3A_p">p</code></td>
<td>
<p>The dimension of data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A high-dimensional linear regression model can be expressed as
</p>
<p style="text-align: center;"><code class="reqn">\boldsymbol{Y}=\boldsymbol{X\Theta}+\boldsymbol{\epsilon},</code>
</p>

<p>where <code class="reqn">\Theta</code> is a <code class="reqn">k\times p</code> unknown parameter matrix and <code class="reqn">\boldsymbol{\epsilon}</code> is an <code class="reqn">n\times p</code> error matrix.
</p>
<p>It is of interest to test the following GLHT problem
</p>
<p style="text-align: center;"><code class="reqn">H_0: \boldsymbol{C\Theta}=\boldsymbol{0}, \quad \text { vs. } H_1: \boldsymbol{C\Theta} \neq \boldsymbol{0}.</code>
</p>

<p>Yamada and Srivastava (2012) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{YS}=\frac{(n-k)\operatorname{tr}(\boldsymbol{S}_h\boldsymbol{D}_{\boldsymbol{S}_e}^{-1})-(n-k)pq/(n-k-2)}{\sqrt{2q[\operatorname{tr}(\boldsymbol{R}^2)-p^2/(n-k)]c_{p,n}}},</code>
</p>

<p>where <code class="reqn">\boldsymbol{S}_h</code> and <code class="reqn">\boldsymbol{S}_e</code> are the variation matrices due to the hypothesis and error, respectively, and <code class="reqn">\boldsymbol{D}_{\boldsymbol{S}_e}</code> and <code class="reqn">\boldsymbol{R}</code> are diagonal matrix with the diagonal elements of <code class="reqn">\boldsymbol{S}_e</code> and the sample correlation matrix, respectively. <code class="reqn">c_{p, n}</code> is the adjustment coefficient proposed by Yamada and Srivastava (2012).
They showed that under the null hypothesis, <code class="reqn">T_{YS}</code> is asymptotically normally distributed.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Yamada T, Srivastava MS (2012).
&ldquo;A test for multivariate analysis of variance in high dimension.&rdquo;
<em>Communications in Statistics-Theory and Methods</em>, <b>41</b>(13-14), 2602&ndash;2615.
<a href="https://doi.org/10.1080/03610926.2011.581786">doi:10.1080/03610926.2011.581786</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("corneal")
dim(corneal)
group1 &lt;- as.matrix(corneal[1:43, ]) ## normal group
group2 &lt;- as.matrix(corneal[44:57, ]) ## unilateral suspect group
group3 &lt;- as.matrix(corneal[58:78, ]) ## suspect map group
group4 &lt;- as.matrix(corneal[79:150, ]) ## clinical keratoconus group
p &lt;- dim(corneal)[2]
Y &lt;- list()
k &lt;- 4
Y[[1]] &lt;- group1
Y[[2]] &lt;- group2
Y[[3]] &lt;- group3
Y[[4]] &lt;- group4
n &lt;- c(nrow(Y[[1]]),nrow(Y[[2]]),nrow(Y[[3]]),nrow(Y[[4]]))
X &lt;- matrix(c(rep(1,n[1]),rep(0,sum(n)),rep(1,n[2]), rep(0,sum(n)),rep(1,n[3]),
            rep(0,sum(n)),rep(1,n[4])),ncol=k,nrow=sum(n))
q &lt;- k-1
C &lt;- cbind(diag(q),-rep(1,q))
YS2012.GLHT.NABT(Y,X,C,n,p)


</code></pre>

<hr>
<h2 id='ZGZ2017.GLHT.2cNRT'>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for GLHT problem proposed Zhang et al. (2017)</h2><span id='topic+ZGZ2017.GLHT.2cNRT'></span>

<h3>Description</h3>

<p>Zhang et al. (2017)'s test for general linear hypothesis testing (GLHT) problem for high-dimensional data with assuming that underlying covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZGZ2017.GLHT.2cNRT(Y,G,n,p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZGZ2017.GLHT.2cNRT_+3A_y">Y</code></td>
<td>
<p>A list of <code class="reqn">k</code> data matrices.  The <code class="reqn">i</code>th element represents the data matrix (<code class="reqn">n_i\times p</code>) from the <code class="reqn">i</code>th population with each row representing a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZGZ2017.GLHT.2cNRT_+3A_g">G</code></td>
<td>
<p>A known full-rank coefficient matrix (<code class="reqn">q\times k</code>) with <code class="reqn">\operatorname{rank}(\boldsymbol{G})&lt;k</code>.</p>
</td></tr>
<tr><td><code id="ZGZ2017.GLHT.2cNRT_+3A_n">n</code></td>
<td>
<p>A vector of <code class="reqn">k</code> sample sizes. The <code class="reqn">i</code>th element represents the sample size of group <code class="reqn">i</code>, <code class="reqn">n_i</code>.</p>
</td></tr>
<tr><td><code id="ZGZ2017.GLHT.2cNRT_+3A_p">p</code></td>
<td>
<p>The dimension of data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have the following <code class="reqn">k</code> independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma},\;i=1,\ldots,k.
</code>
</p>

<p>It is of interest to test the following GLHT problem:
</p>
<p style="text-align: center;"><code class="reqn">H_0: \boldsymbol{G M}=\boldsymbol{0}, \quad \text { vs. } \quad H_1: \boldsymbol{G M} \neq \boldsymbol{0},</code>
</p>

<p>where
<code class="reqn">\boldsymbol{M}=(\boldsymbol{\mu}_1,\ldots,\boldsymbol{\mu}_k)^\top</code> is a <code class="reqn">k\times p</code> matrix collecting <code class="reqn">k</code> mean vectors and <code class="reqn">\boldsymbol{G}:q\times k</code> is a known full-rank coefficient matrix with <code class="reqn">\operatorname{rank}(\boldsymbol{G})&lt;k</code>.
</p>
<p>Zhang et al. (2017) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">
T_{ZGZ}=\|\boldsymbol{C \hat{\mu}}\|^2,
</code>
</p>

<p>where <code class="reqn">\boldsymbol{C}=[(\boldsymbol{G D G}^\top)^{-1/2}\boldsymbol{G}]\otimes\boldsymbol{I}_p</code>, and <code class="reqn">\hat{\boldsymbol{\mu}}=(\bar{\boldsymbol{y}}_1^\top,\ldots,\bar{\boldsymbol{y}}_k^\top)^\top</code>, with <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,\ldots,k</code> being the sample mean vectors and <code class="reqn">\boldsymbol{D}=\operatorname{diag}(1/n_1,\ldots,1/n_k)</code>.
</p>
<p>They showed that under the null hypothesis, <code class="reqn">T_{ZGZ}</code> and a chi-squared-type mixture have the same normal or non-normal limiting distribution.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhang J, Guo J, Zhou B (2017).
&ldquo;Linear hypothesis testing in high-dimensional one-way MANOVA.&rdquo;
<em>Journal of Multivariate Analysis</em>, <b>155</b>, 200&ndash;216.
<a href="https://doi.org/10.1016/j.jmva.2017.01.002">doi:10.1016/j.jmva.2017.01.002</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("corneal")
dim(corneal)
group1 &lt;- as.matrix(corneal[1:43, ]) ## normal group
group2 &lt;- as.matrix(corneal[44:57, ]) ## unilateral suspect group
group3 &lt;- as.matrix(corneal[58:78, ]) ## suspect map group
group4 &lt;- as.matrix(corneal[79:150, ]) ## clinical keratoconus group
p &lt;- dim(corneal)[2]
Y &lt;- list()
k &lt;- 4
Y[[1]] &lt;- group1
Y[[2]] &lt;- group2
Y[[3]] &lt;- group3
Y[[4]] &lt;- group4
n &lt;- c(nrow(Y[[1]]),nrow(Y[[2]]),nrow(Y[[3]]),nrow(Y[[4]]))
G &lt;- cbind(diag(k-1),rep(-1,k-1))
ZGZ2017.GLHT.2cNRT(Y,G,n,p)
</code></pre>

<hr>
<h2 id='ZGZ2017.GLHTBF.NABT'>Normal-approximation-based test for GLHT problem under heteroscedasticity proposed by Zhou et al. (2017)</h2><span id='topic+ZGZ2017.GLHTBF.NABT'></span>

<h3>Description</h3>

<p>Zhou et al. (2017)'s test for general linear hypothesis testing (GLHT) problem for high-dimensional data under heteroscedasticity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZGZ2017.GLHTBF.NABT(Y,G,n,p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZGZ2017.GLHTBF.NABT_+3A_y">Y</code></td>
<td>
<p>A list of <code class="reqn">k</code> data matrices.  The <code class="reqn">i</code>th element represents the data matrix (<code class="reqn">n_i\times p</code>) from the <code class="reqn">i</code>th population with each row representing a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZGZ2017.GLHTBF.NABT_+3A_g">G</code></td>
<td>
<p>A known full-rank coefficient matrix (<code class="reqn">q\times k</code>) with <code class="reqn">\operatorname{rank}(\boldsymbol{G})&lt; k</code>.</p>
</td></tr>
<tr><td><code id="ZGZ2017.GLHTBF.NABT_+3A_n">n</code></td>
<td>
<p>A vector of <code class="reqn">k</code> sample sizes. The <code class="reqn">i</code>th element represents the sample size of group <code class="reqn">i</code>, <code class="reqn">n_i</code>.</p>
</td></tr>
<tr><td><code id="ZGZ2017.GLHTBF.NABT_+3A_p">p</code></td>
<td>
<p>The dimension of data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have the following <code class="reqn">k</code> independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma}_i,i=1,\ldots,k.
</code>
</p>

<p>It is of interest to test the following GLHT problem:
</p>
<p style="text-align: center;"><code class="reqn">H_0: \boldsymbol{G M}=\boldsymbol{0}, \quad \text { vs. } H_1: \boldsymbol{G M} \neq \boldsymbol{0},</code>
</p>

<p>where
<code class="reqn">\boldsymbol{M}=(\boldsymbol{\mu}_1,\ldots,\boldsymbol{\mu}_k)^\top</code> is a <code class="reqn">k\times p</code> matrix collecting <code class="reqn">k</code> mean vectors and <code class="reqn">\boldsymbol{G}:q\times k</code> is a known full-rank coefficient matrix with <code class="reqn">\operatorname{rank}(\boldsymbol{G})&lt;k</code>.
</p>
<p>Let  <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,\ldots,k</code> be the sample mean vectors and <code class="reqn">\hat{\boldsymbol{\Sigma}}_i,i=1,\ldots,k</code> be the sample covariance matrices.
</p>
<p>Zhou et al. (2017) proposed the following U-statistic based test statistic:
</p>
<p style="text-align: center;"><code class="reqn">
T_{ZGZ}=\|\boldsymbol{C \hat{\mu}}\|^2-\sum_{i=1}^k h_{ii}\operatorname{tr}(\hat{\boldsymbol{\Sigma}}_i)/n_i,
</code>
</p>

<p>where <code class="reqn">\boldsymbol{C}=[(\boldsymbol{G D G}^\top)^{-1/2}\boldsymbol{G}]\otimes\boldsymbol{I}_p</code>, <code class="reqn">\boldsymbol{D}=\operatorname{diag}(1/n_1,\ldots,1/n_k)</code>, and <code class="reqn">h_{ij}</code> is the <code class="reqn">(i,j)</code>th entry of the <code class="reqn">k\times k</code> matrix <code class="reqn">\boldsymbol{H}=\boldsymbol{G}^\top(\boldsymbol{G}\boldsymbol{D}\boldsymbol{G}^\top)^{-1}\boldsymbol{G}</code>.
</p>
<p>They showed that under the null hypothesis, <code class="reqn">T_{ZGZ}</code> is asymptotically normally distributed.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhou B, Guo J, Zhang J (2017).
&ldquo;High-dimensional general linear hypothesis testing under heteroscedasticity.&rdquo;
<em>Journal of Statistical Planning and Inference</em>, <b>188</b>, 36&ndash;54.
<a href="https://doi.org/10.1016/j.jspi.2017.03.005">doi:10.1016/j.jspi.2017.03.005</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("corneal")
dim(corneal)
group1 &lt;- as.matrix(corneal[1:43, ]) ## normal group
group2 &lt;- as.matrix(corneal[44:57, ]) ## unilateral suspect group
group3 &lt;- as.matrix(corneal[58:78, ]) ## suspect map group
group4 &lt;- as.matrix(corneal[79:150, ]) ## clinical keratoconus group
p &lt;- dim(corneal)[2]
Y &lt;- list()
k &lt;- 4
Y[[1]] &lt;- group1
Y[[2]] &lt;- group2
Y[[3]] &lt;- group3
Y[[4]] &lt;- group4
n &lt;- c(nrow(Y[[1]]),nrow(Y[[2]]),nrow(Y[[3]]),nrow(Y[[4]]))
G &lt;- cbind(diag(k-1),rep(-1,k-1))
ZGZ2017.GLHTBF.NABT(Y,G,n,p)

</code></pre>

<hr>
<h2 id='ZGZC2020.TS.2cNRT'>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for two-sample problem proposed by Zhang et al. (2020)</h2><span id='topic+ZGZC2020.TS.2cNRT'></span>

<h3>Description</h3>

<p>Zhang et al. (2020)'s test for testing equality of two-sample high-dimensional mean vectors with assuming that two covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZGZC2020.TS.2cNRT(y1, y2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZGZC2020.TS.2cNRT_+3A_y1">y1</code></td>
<td>
<p>The data matrix (<code class="reqn">n_1 \times p</code>) from the first population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZGZC2020.TS.2cNRT_+3A_y2">y2</code></td>
<td>
<p>The data matrix (<code class="reqn">n_2 \times p</code>) from the second population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have two independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma},i=1,2.
</code>
</p>

<p>The primary object is to test
</p>
<p style="text-align: center;"><code class="reqn">H_{0}: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2\; \operatorname{versus}\; H_{1}: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2.</code>
</p>

<p>Zhang et al.(2020) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{ZGZC} = \frac{n_1n_2}{n} \|\bar{\boldsymbol{y}}_1 - \bar{\boldsymbol{y}}_2\|^2,</code>
</p>

<p>where  <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,2</code> are the sample mean vectors.
They showed that under the null hypothesis, <code class="reqn">T_{ZGZC}</code> and a chi-squared-type mixture have the same normal or non-normal limiting distribution.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhang J, Guo J, Zhou B, Cheng M (2020).
&ldquo;A simple two-sample test in high dimensions based on L 2-norm.&rdquo;
<em>Journal of the American Statistical Association</em>, <b>115</b>(530), 1011&ndash;1027.
<a href="https://doi.org/10.1080/01621459.2019.1604366">doi:10.1080/01621459.2019.1604366</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("COVID19")
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
ZGZC2020.TS.2cNRT(group1, group2)

</code></pre>

<hr>
<h2 id='ZWZ2023.TSBF.2cNRT'>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for two-sample BF problem proposed by Zhu et al. (2023)</h2><span id='topic+ZWZ2023.TSBF.2cNRT'></span>

<h3>Description</h3>

<p>Zhu et al. (2023)'s test for testing equality of two-sample high-dimensional mean vectors without assuming that two covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZWZ2023.TSBF.2cNRT(y1, y2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZWZ2023.TSBF.2cNRT_+3A_y1">y1</code></td>
<td>
<p>The data matrix (<code class="reqn">n_1 \times p</code>) from the first population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZWZ2023.TSBF.2cNRT_+3A_y2">y2</code></td>
<td>
<p>The data matrix (<code class="reqn">n_2 \times p</code>) from the second population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have two independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma}_i,\; i=1,2.
</code>
</p>

<p>The primary object is to test
</p>
<p style="text-align: center;"><code class="reqn">H_{0}: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2\; \operatorname{versus}\; H_{1}: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2.</code>
</p>

<p>Zhu et al. (2023) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{ZWZ}=\frac{n_1n_2n^{-1}\|\bar{\boldsymbol{y}}_1-\bar{\boldsymbol{y}}_2\|^2}{\operatorname{tr}(\hat{\boldsymbol{\Omega}}_n)},</code>
</p>

<p>where  <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,2</code> are the sample mean vectors and <code class="reqn">\hat{\boldsymbol{\Omega}}_n</code> is the estimator of <code class="reqn">\operatorname{Cov}[(n_1n_2/n)^{1/2}(\bar{\boldsymbol{y}}_1-\bar{\boldsymbol{y}}_2)]</code>.
They showed that under the null hypothesis, <code class="reqn">T_{ZWZ}</code> and an F-type mixture have the same normal or non-normal limiting distribution.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhu T, Wang P, Zhang J (2023).
&ldquo;Two-sample Behrens&ndash;Fisher problems for high-dimensional data: a normal reference F-type test.&rdquo;
<em>Computational Statistics</em>, 1&ndash;24.
<a href="https://doi.org/10.1007/s00180-023-01433-6">doi:10.1007/s00180-023-01433-6</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("COVID19")
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
ZWZ2023.TSBF.2cNRT(group1, group2)


</code></pre>

<hr>
<h2 id='ZZ2022.GLHT.3cNRT'>Normal-reference-test with three-cumulant (3-c) matched $\chi^2$-approximation for GLHT problem proposed by Zhu and Zhang (2022)</h2><span id='topic+ZZ2022.GLHT.3cNRT'></span>

<h3>Description</h3>

<p>Zhu and Zhang (2022)'s test for general linear hypothesis testing (GLHT) problem for high-dimensional data with assuming that underlying covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZZ2022.GLHT.3cNRT(Y,G,n,p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZZ2022.GLHT.3cNRT_+3A_y">Y</code></td>
<td>
<p>A list of <code class="reqn">k</code> data matrices.  The <code class="reqn">i</code>th element represents the data matrix (<code class="reqn">n_i\times p</code>) from the <code class="reqn">i</code>th population with each row representing a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZZ2022.GLHT.3cNRT_+3A_g">G</code></td>
<td>
<p>A known full-rank coefficient matrix (<code class="reqn">q\times k</code>) with <code class="reqn">\operatorname{rank}(\boldsymbol{G})&lt;k</code>.</p>
</td></tr>
<tr><td><code id="ZZ2022.GLHT.3cNRT_+3A_n">n</code></td>
<td>
<p>A vector of <code class="reqn">k</code> sample sizes. The <code class="reqn">i</code>th element represents the sample size of group <code class="reqn">i</code>, <code class="reqn">n_i</code>.</p>
</td></tr>
<tr><td><code id="ZZ2022.GLHT.3cNRT_+3A_p">p</code></td>
<td>
<p>The dimension of data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have the following <code class="reqn">k</code> independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma},\; i=1,\ldots,k.
</code>
</p>

<p>It is of interest to test the following GLHT problem:
</p>
<p style="text-align: center;"><code class="reqn">H_0: \boldsymbol{G M}=\boldsymbol{0}, \quad \text { vs. } \quad H_1: \boldsymbol{G M} \neq \boldsymbol{0},</code>
</p>

<p>where
<code class="reqn">\boldsymbol{M}=(\boldsymbol{\mu}_1,\ldots,\boldsymbol{\mu}_k)^\top</code> is a <code class="reqn">k\times p</code> matrix collecting <code class="reqn">k</code> mean vectors and <code class="reqn">\boldsymbol{G}:q\times k</code> is a known full-rank coefficient matrix with <code class="reqn">\operatorname{rank}(\boldsymbol{G})&lt;k</code>.
</p>
<p>Zhu and Zhang (2022) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">
T_{ZZ}=\|\boldsymbol{C} \hat{\boldsymbol{\mu}}\|^2-q \operatorname{tr}(\hat{\boldsymbol{\Sigma}}),
</code>
</p>

<p>where <code class="reqn">\boldsymbol{C}=[(\boldsymbol{G D G}^\top)^{-1/2}\boldsymbol{G}]\otimes\boldsymbol{I}_p</code>, and <code class="reqn">\hat{\boldsymbol{\mu}}=(\bar{\boldsymbol{y}}_1^\top,\ldots,\bar{\boldsymbol{y}}_k^\top)^\top</code>, with <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,\ldots,k</code> being the sample mean vectors and <code class="reqn">\hat{\boldsymbol{\Sigma}}</code> being the usual pooled sample covariance matrix of the <code class="reqn">k</code> samples.
</p>
<p>They showed that under the null hypothesis, <code class="reqn">T_{ZZ}</code> and a chi-squared-type mixture have the same normal or non-normal limiting distribution.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhu T, Zhang J (2022).
&ldquo;Linear hypothesis testing in high-dimensional one-way MANOVA: a new normal reference approach.&rdquo;
<em>Computational Statistics</em>, <b>37</b>(1), 1&ndash;27.
<a href="https://doi.org/10.1007/s00180-021-01110-6">doi:10.1007/s00180-021-01110-6</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("corneal")
dim(corneal)
group1 &lt;- as.matrix(corneal[1:43, ]) ## normal group
group2 &lt;- as.matrix(corneal[44:57, ]) ## unilateral suspect group
group3 &lt;- as.matrix(corneal[58:78, ]) ## suspect map group
group4 &lt;- as.matrix(corneal[79:150, ]) ## clinical keratoconus group
p &lt;- dim(corneal)[2]
Y &lt;- list()
k &lt;- 4
Y[[1]] &lt;- group1
Y[[2]] &lt;- group2
Y[[3]] &lt;- group3
Y[[4]] &lt;- group4
n &lt;- c(nrow(Y[[1]]),nrow(Y[[2]]),nrow(Y[[3]]),nrow(Y[[4]]))
G &lt;- cbind(diag(k-1),rep(-1,k-1))
ZZ2022.GLHT.3cNRT(Y,G,n,p)

</code></pre>

<hr>
<h2 id='ZZ2022.GLHTBF.3cNRT'>Normal-reference-test with three-cumulant (3-c) matched $\chi^2$-approximation for GLHT problem under heteroscedasticity proposed by Zhang and Zhu (2022)</h2><span id='topic+ZZ2022.GLHTBF.3cNRT'></span>

<h3>Description</h3>

<p>Zhang and Zhu (2022)'s test for general linear hypothesis testing (GLHT) problem for high-dimensional data under heteroscedasticity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZZ2022.GLHTBF.3cNRT(Y,G,n,p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZZ2022.GLHTBF.3cNRT_+3A_y">Y</code></td>
<td>
<p>A list of <code class="reqn">k</code> data matrices.  The <code class="reqn">i</code>th element represents the data matrix (<code class="reqn">n_i\times p</code>) from the <code class="reqn">i</code>th population with each row representing a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZZ2022.GLHTBF.3cNRT_+3A_g">G</code></td>
<td>
<p>A known full-rank coefficient matrix (<code class="reqn">q\times k</code>) with <code class="reqn">\operatorname{rank}(\boldsymbol{G})&lt; k</code>.</p>
</td></tr>
<tr><td><code id="ZZ2022.GLHTBF.3cNRT_+3A_n">n</code></td>
<td>
<p>A vector of <code class="reqn">k</code> sample sizes. The <code class="reqn">i</code>th element represents the sample size of group <code class="reqn">i</code>, <code class="reqn">n_i</code>.</p>
</td></tr>
<tr><td><code id="ZZ2022.GLHTBF.3cNRT_+3A_p">p</code></td>
<td>
<p>The dimension of data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have the following <code class="reqn">k</code> independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma}_i,i=1,\ldots,k.
</code>
</p>

<p>It is of interest to test the following GLHT problem:
</p>
<p style="text-align: center;"><code class="reqn">H_0: \boldsymbol{G M}=\boldsymbol{0}, \quad \text { vs. } H_1: \boldsymbol{G M} \neq \boldsymbol{0},</code>
</p>

<p>where
<code class="reqn">\boldsymbol{M}=(\boldsymbol{\mu}_1,\ldots,\boldsymbol{\mu}_k)^\top</code> is a <code class="reqn">k\times p</code> matrix collecting <code class="reqn">k</code> mean vectors and <code class="reqn">\boldsymbol{G}:q\times k</code> is a known full-rank coefficient matrix with <code class="reqn">\operatorname{rank}(\boldsymbol{G})&lt;k</code>.
</p>
<p>Let  <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,\ldots,k</code> be the sample mean vectors and <code class="reqn">\hat{\boldsymbol{\Sigma}}_i,i=1,\ldots,k</code> be the sample covariance matrices.
</p>
<p>Zhang and Zhu (2022) proposed the following U-statistic based test statistic:
</p>
<p style="text-align: center;"><code class="reqn">
T_{ZZ}=\|\boldsymbol{C \hat{\mu}}\|^2-\sum_{i=1}^kh_{ii}\operatorname{tr}(\hat{\boldsymbol{\Sigma}}_i)/n_i,
</code>
</p>

<p>where <code class="reqn">\boldsymbol{C}=[(\boldsymbol{G D G}^\top)^{-1/2}\boldsymbol{G}]\otimes\boldsymbol{I}_p</code>, <code class="reqn">\boldsymbol{D}=\operatorname{diag}(1/n_1,\ldots,1/n_k)</code>, and <code class="reqn">h_{ij}</code> is the <code class="reqn">(i,j)</code>th entry of the <code class="reqn">k\times k</code> matrix <code class="reqn">\boldsymbol{H}=\boldsymbol{G}^\top(\boldsymbol{G}\boldsymbol{D}\boldsymbol{G}^\top)^{-1}\boldsymbol{G}</code>.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhang J, Zhu T (2022).
&ldquo;A new normal reference test for linear hypothesis testing in high-dimensional heteroscedastic one-way MANOVA.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>168</b>, 107385.
<a href="https://doi.org/10.1016/j.csda.2021.107385">doi:10.1016/j.csda.2021.107385</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("corneal")
dim(corneal)
group1 &lt;- as.matrix(corneal[1:43, ]) ## normal group
group2 &lt;- as.matrix(corneal[44:57, ]) ## unilateral suspect group
group3 &lt;- as.matrix(corneal[58:78, ]) ## suspect map group
group4 &lt;- as.matrix(corneal[79:150, ]) ## clinical keratoconus group
p &lt;- dim(corneal)[2]
Y &lt;- list()
k &lt;- 4
Y[[1]] &lt;- group1
Y[[2]] &lt;- group2
Y[[3]] &lt;- group3
Y[[4]] &lt;- group4
n &lt;- c(nrow(Y[[1]]),nrow(Y[[2]]),nrow(Y[[3]]),nrow(Y[[4]]))
G &lt;- cbind(diag(k-1),rep(-1,k-1))
ZZ2022.GLHTBF.3cNRT(Y,G,n,p)

</code></pre>

<hr>
<h2 id='ZZ2022.TS.3cNRT'>Normal-reference-test with three-cumulant (3-c) matched $\chi^2$-approximation for two-sample problem proposed by Zhang and Zhu (2022)</h2><span id='topic+ZZ2022.TS.3cNRT'></span>

<h3>Description</h3>

<p>Zhang and Zhu (2022)'s test for testing equality of two-sample high-dimensional mean vectors with assuming that two covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZZ2022.TS.3cNRT(y1, y2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZZ2022.TS.3cNRT_+3A_y1">y1</code></td>
<td>
<p>The data matrix (<code class="reqn">n_1 \times p</code>) from the first population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZZ2022.TS.3cNRT_+3A_y2">y2</code></td>
<td>
<p>The data matrix (<code class="reqn">n_2 \times p</code>) from the second population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have two independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma},i=1,2.
</code>
</p>

<p>The primary object is to test
</p>
<p style="text-align: center;"><code class="reqn">H_{0}: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2\; \operatorname{versus}\; H_{1}: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2.</code>
</p>

<p>Zhang et al.(2022) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{ZZ} = \frac{n_1n_2}{n} \|\bar{\boldsymbol{y}}_1 - \bar{\boldsymbol{y}}_2\|^2-\operatorname{tr}(\hat{\boldsymbol{\Sigma}}),</code>
</p>

<p>where  <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,2</code> are the sample mean vectors and <code class="reqn">\hat{\boldsymbol{\Sigma}}</code> is the pooled sample covariance matrix.
They showed that under the null hypothesis, <code class="reqn">T_{ZZ}</code> and a chi-squared-type mixture have the same normal or non-normal limiting distribution.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhang J, Zhu T (2022).
&ldquo;A revisit to Bai&ndash;Saranadasa's two-sample test.&rdquo;
<em>Journal of Nonparametric Statistics</em>, <b>34</b>(1), 58&ndash;76.
<a href="https://doi.org/10.1080/10485252.2021.2015768">doi:10.1080/10485252.2021.2015768</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("COVID19")
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
ZZ2022.TS.3cNRT(group1, group2)

</code></pre>

<hr>
<h2 id='ZZ2022.TSBF.3cNRT'>Normal-reference-test with three-cumulant (3-c) matched $\chi^2$-approximation for two-sample BF problem proposed by Zhang and Zhu (2022)</h2><span id='topic+ZZ2022.TSBF.3cNRT'></span>

<h3>Description</h3>

<p>Zhang and Zhu (2022)'s test for testing equality of two-sample high-dimensional mean vectors without assuming that two covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZZ2022.TSBF.3cNRT(y1, y2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZZ2022.TSBF.3cNRT_+3A_y1">y1</code></td>
<td>
<p>The data matrix (<code class="reqn">n_1 \times p</code>) from the first population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZZ2022.TSBF.3cNRT_+3A_y2">y2</code></td>
<td>
<p>The data matrix (<code class="reqn">n_2 \times p</code>) from the second population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have two independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma}_i,i=1,2.
</code>
</p>

<p>The primary object is to test
</p>
<p style="text-align: center;"><code class="reqn">H_{0}: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2\; \operatorname{versus}\; H_{1}: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2.</code>
</p>

<p>Zhang and Zhu (2022) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{ZZ} =  \|\bar{\boldsymbol{y}}_1 - \bar{\boldsymbol{y}}_2\|^2-\operatorname{tr}(\hat{\boldsymbol{\Omega}}_n),</code>
</p>

<p>where  <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,2</code> are the sample mean vectors and <code class="reqn">\hat{\boldsymbol{\Omega}}_n</code> is the estimator of <code class="reqn">\operatorname{Cov}(\bar{\boldsymbol{y}}_1-\bar{\boldsymbol{y}}_2)</code>.
They showed that under the null hypothesis, <code class="reqn">T_{ZZ}</code> and a chi-squared-type mixture have the same normal or non-normal limiting distribution.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhang J, Zhu T (2022).
&ldquo;A further study on Chen-Qin’s test for two-sample Behrens&ndash;Fisher problems for high-dimensional data.&rdquo;
<em>Journal of Statistical Theory and Practice</em>, <b>16</b>(1), 1.
<a href="https://doi.org/10.1007/s42519-021-00232-w">doi:10.1007/s42519-021-00232-w</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("COVID19")
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
ZZ2022.TSBF.3cNRT(group1, group2)

</code></pre>

<hr>
<h2 id='ZZG2022.GLHTBF.2cNRT'>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for GLHT problem under heteroscedasticity proposed by Zhang et al. (2022)</h2><span id='topic+ZZG2022.GLHTBF.2cNRT'></span>

<h3>Description</h3>

<p>Zhang et al. (2022)'s test for general linear hypothesis testing (GLHT) problem for high-dimensional data under heteroscedasticity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZZG2022.GLHTBF.2cNRT(Y,G,n,p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZZG2022.GLHTBF.2cNRT_+3A_y">Y</code></td>
<td>
<p>A list of <code class="reqn">k</code> data matrices.  The <code class="reqn">i</code>th element represents the data matrix (<code class="reqn">n_i\times p</code>) from the <code class="reqn">i</code>th population with each row representing a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZZG2022.GLHTBF.2cNRT_+3A_g">G</code></td>
<td>
<p>A known full-rank coefficient matrix (<code class="reqn">q\times k</code>) with <code class="reqn">\operatorname{rank}(\boldsymbol{G})&lt; k</code>.</p>
</td></tr>
<tr><td><code id="ZZG2022.GLHTBF.2cNRT_+3A_n">n</code></td>
<td>
<p>A vector of <code class="reqn">k</code> sample sizes. The <code class="reqn">i</code>th element represents the sample size of group <code class="reqn">i</code>, <code class="reqn">n_i</code>.</p>
</td></tr>
<tr><td><code id="ZZG2022.GLHTBF.2cNRT_+3A_p">p</code></td>
<td>
<p>The dimension of data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have the following <code class="reqn">k</code> independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma}_i,i=1,\ldots,k.
</code>
</p>

<p>It is of interest to test the following GLHT problem:
</p>
<p style="text-align: center;"><code class="reqn">H_0: \boldsymbol{G M}=\boldsymbol{0}, \quad \text { vs. } \; H_1: \boldsymbol{G M} \neq \boldsymbol{0},</code>
</p>

<p>where
<code class="reqn">\boldsymbol{M}=(\boldsymbol{\mu}_1,\ldots,\boldsymbol{\mu}_k)^\top</code> is a <code class="reqn">k\times p</code> matrix collecting <code class="reqn">k</code> mean vectors and <code class="reqn">\boldsymbol{G}:q\times k</code> is a known full-rank coefficient matrix with <code class="reqn">\operatorname{rank}(\boldsymbol{G})&lt;k</code>.
</p>
<p>Zhang et al. (2022) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">
T_{ZZG}=\|\boldsymbol{C} \hat{\boldsymbol{\mu}}\|^2,
</code>
</p>

<p>where <code class="reqn">\boldsymbol{C}=[(\boldsymbol{G D G}^\top)^{-1/2}\boldsymbol{G}]\otimes\boldsymbol{I}_p</code> with <code class="reqn">\boldsymbol{D}=\operatorname{diag}(1/n_1,\ldots,1/n_k)</code>, and <code class="reqn">\hat{\boldsymbol{\mu}}=(\bar{\boldsymbol{y}}_1^\top,\ldots,\bar{\boldsymbol{y}}_k^\top)^\top</code> with <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,\ldots,k</code> being the sample mean vectors.
</p>
<p>They showed that under the null hypothesis, <code class="reqn">T_{ZZG}</code> and a chi-squared-type mixture have the same normal or non-normal limiting distribution.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhang J, Zhou B, Guo J (2022).
&ldquo;Linear hypothesis testing in high-dimensional heteroscedastic one-way MANOVA: A normal reference <code class="reqn">L^2</code>-norm based test.&rdquo;
<em>Journal of Multivariate Analysis</em>, <b>187</b>, 104816.
<a href="https://doi.org/10.1016/j.jmva.2021.104816">doi:10.1016/j.jmva.2021.104816</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("corneal")
dim(corneal)
group1 &lt;- as.matrix(corneal[1:43, ]) ## normal group
group2 &lt;- as.matrix(corneal[44:57, ]) ## unilateral suspect group
group3 &lt;- as.matrix(corneal[58:78, ]) ## suspect map group
group4 &lt;- as.matrix(corneal[79:150, ]) ## clinical keratoconus group
p &lt;- dim(corneal)[2]
Y &lt;- list()
k &lt;- 4
Y[[1]] &lt;- group1
Y[[2]] &lt;- group2
Y[[3]] &lt;- group3
Y[[4]] &lt;- group4
n &lt;- c(nrow(Y[[1]]),nrow(Y[[2]]),nrow(Y[[3]]),nrow(Y[[4]]))
G &lt;- cbind(diag(k-1),rep(-1,k-1))
ZZG2022.GLHTBF.2cNRT(Y,G,n,p)

</code></pre>

<hr>
<h2 id='ZZGZ2021.TSBF.2cNRT'>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for two-sample BF problem proposed by Zhang et al. (2021)</h2><span id='topic+ZZGZ2021.TSBF.2cNRT'></span>

<h3>Description</h3>

<p>Zhang et al. (2021)'s test for testing equality of two-sample high-dimensional mean vectors without assuming that two covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZZGZ2021.TSBF.2cNRT(y1, y2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZZGZ2021.TSBF.2cNRT_+3A_y1">y1</code></td>
<td>
<p>The data matrix (<code class="reqn">n_1 \times p</code>) from the first population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZZGZ2021.TSBF.2cNRT_+3A_y2">y2</code></td>
<td>
<p>The data matrix (<code class="reqn">n_2 \times p</code>) from the second population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have two independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma}_i,i=1,2.
</code>
</p>

<p>The primary object is to test
</p>
<p style="text-align: center;"><code class="reqn">H_{0}: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2\; \operatorname{versus}\; H_{1}: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2.</code>
</p>

<p>Zhang et al.(2021) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{ZZGZ} = \frac{n_1n_2}{n} \|\bar{\boldsymbol{y}}_1 - \bar{\boldsymbol{y}}_2\|^2,</code>
</p>

<p>where  <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,2</code> are the sample mean vectors.
They showed that under the null hypothesis, <code class="reqn">T_{ZZGZ}</code> and a chi-squared-type mixture have the same normal or non-normal limiting distribution.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhang J, Zhou B, Guo J, Zhu T (2021).
&ldquo;Two-sample Behrens-Fisher problems for high-dimensional data: A normal reference approach.&rdquo;
<em>Journal of Statistical Planning and Inference</em>, <b>213</b>, 142&ndash;161.
<a href="https://doi.org/10.1016/j.jspi.2020.11.008">doi:10.1016/j.jspi.2020.11.008</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("COVID19")
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
ZZGZ2021.TSBF.2cNRT(group1, group2)

</code></pre>

<hr>
<h2 id='ZZZ2020.TS.2cNRT'>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for two-sample problem proposed by Zhang et al. (2020)</h2><span id='topic+ZZZ2020.TS.2cNRT'></span>

<h3>Description</h3>

<p>Zhang et al. (2020)'s test for testing equality of two-sample high-dimensional mean vectors with assuming that two covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZZZ2020.TS.2cNRT(y1, y2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZZZ2020.TS.2cNRT_+3A_y1">y1</code></td>
<td>
<p>The data matrix (<code class="reqn">n_1 \times p</code>) from the first population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZZZ2020.TS.2cNRT_+3A_y2">y2</code></td>
<td>
<p>The data matrix (<code class="reqn">n_2 \times p</code>) from the second population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have two independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma},i=1,2.
</code>
</p>

<p>The primary object is to test
</p>
<p style="text-align: center;"><code class="reqn">H_{0}: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2\; \operatorname{versus}\; H_{1}: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2.</code>
</p>

<p>Zhang et al.(2020) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{ZZZ} = \frac{n_1n_2}{np}(\bar{\boldsymbol{y}}_1 - \bar{\boldsymbol{y}}_2)^\top \hat{\boldsymbol{D}}^{-1}(\bar{\boldsymbol{y}}_1 - \bar{\boldsymbol{y}}_2),</code>
</p>

<p>where <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,2</code> are the sample mean vectors, <code class="reqn">\hat{\boldsymbol{D}}</code> is the diagonal matrix of sample covariance matrix.
They showed that under the null hypothesis, <code class="reqn">T_{ZZZ}</code> and a chi-squared-type mixture have the same limiting distribution.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhang L, Zhu T, Zhang J (2020).
&ldquo;A simple scale-invariant two-sample test for high-dimensional data.&rdquo;
<em>Econometrics and Statistics</em>, <b>14</b>, 131&ndash;144.
<a href="https://doi.org/10.1016/j.ecosta.2019.12.002">doi:10.1016/j.ecosta.2019.12.002</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("COVID19")
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
ZZZ2020.TS.2cNRT(group1,group2)


</code></pre>

<hr>
<h2 id='ZZZ2022.GLHT.2cNRT'>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for GLHT problem proposed by Zhu et al. (2022)</h2><span id='topic+ZZZ2022.GLHT.2cNRT'></span>

<h3>Description</h3>

<p>Zhu et al. (2022)'s test for general linear hypothesis testing (GLHT) problem for high-dimensional data with assuming that underlying covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZZZ2022.GLHT.2cNRT(Y,X,C,n,p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZZZ2022.GLHT.2cNRT_+3A_y">Y</code></td>
<td>
<p>A list of <code class="reqn">k</code> data matrices.  The <code class="reqn">i</code>th element represents the data matrix (<code class="reqn">n_i \times p</code>) from the <code class="reqn">i</code>th population with each row representing a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZZZ2022.GLHT.2cNRT_+3A_x">X</code></td>
<td>
<p>A known <code class="reqn">n\times k</code> full-rank design matrix with <code class="reqn">\operatorname{rank}(\boldsymbol{X})=k&lt;n</code>.</p>
</td></tr>
<tr><td><code id="ZZZ2022.GLHT.2cNRT_+3A_c">C</code></td>
<td>
<p>A known matrix of size <code class="reqn">q\times k</code> with <code class="reqn">\operatorname{rank}(\boldsymbol{C})=q&lt;k</code>.</p>
</td></tr>
<tr><td><code id="ZZZ2022.GLHT.2cNRT_+3A_n">n</code></td>
<td>
<p>A vector of <code class="reqn">k</code> sample sizes. The <code class="reqn">i</code>th element represents the sample size of group <code class="reqn">i</code>, <code class="reqn">n_i</code>.</p>
</td></tr>
<tr><td><code id="ZZZ2022.GLHT.2cNRT_+3A_p">p</code></td>
<td>
<p>The dimension of data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A high-dimensional linear regression model can be expressed as
</p>
<p style="text-align: center;"><code class="reqn">\boldsymbol{Y}=\boldsymbol{X\Theta}+\boldsymbol{\epsilon},</code>
</p>

<p>where <code class="reqn">\Theta</code> is a <code class="reqn">k\times p</code> unknown parameter matrix and <code class="reqn">\boldsymbol{\epsilon}</code> is an <code class="reqn">n\times p</code> error matrix.
</p>
<p>It is of interest to test the following GLHT problem
</p>
<p style="text-align: center;"><code class="reqn">H_0: \boldsymbol{C\Theta}=\boldsymbol{0}, \quad \text { vs. } H_1: \boldsymbol{C\Theta} \neq \boldsymbol{0}.</code>
</p>

<p>Zhu et al. (2022) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{ZZZ}=\frac{(n-k-2)}{(n-k)pq}\operatorname{tr}(\boldsymbol{S}_h\boldsymbol{D}^{-1}),</code>
</p>

<p>where <code class="reqn">\boldsymbol{S}_h</code> and <code class="reqn">\boldsymbol{S}_e</code> are the variation matrices due to the hypothesis and error, respectively, and <code class="reqn">\boldsymbol{D}</code> is the diagonal matrix with the diagonal elements of <code class="reqn">\boldsymbol{S}_e/(n-k)</code>.
They showed that under the null hypothesis, <code class="reqn">T_{ZZZ}</code> and a chi-squared-type mixture have the same limiting distribution.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhu T, Zhang L, Zhang J (2023).
&ldquo;Hypothesis Testing in High-Dimensional Linear Regression: A Normal Reference Scale-Invariant Test.&rdquo;
<em>Statistica Sinica</em>.
<a href="https://doi.org/10.5705/ss.202020.0362">doi:10.5705/ss.202020.0362</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("corneal")
dim(corneal)
group1 &lt;- as.matrix(corneal[1:43, ]) ## normal group
group2 &lt;- as.matrix(corneal[44:57, ]) ## unilateral suspect group
group3 &lt;- as.matrix(corneal[58:78, ]) ## suspect map group
group4 &lt;- as.matrix(corneal[79:150, ]) ## clinical keratoconus group
p &lt;- dim(corneal)[2]
Y &lt;- list()
k &lt;- 4
Y[[1]] &lt;- group1
Y[[2]] &lt;- group2
Y[[3]] &lt;- group3
Y[[4]] &lt;- group4
n &lt;- c(nrow(Y[[1]]),nrow(Y[[2]]),nrow(Y[[3]]),nrow(Y[[4]]))
X &lt;- matrix(c(rep(1,n[1]),rep(0,sum(n)),rep(1,n[2]), rep(0,sum(n)),
            rep(1,n[3]),rep(0,sum(n)),rep(1,n[4])),ncol=k,nrow=sum(n))
q &lt;- k-1
C &lt;- cbind(diag(q),-rep(1,q))
ZZZ2022.GLHT.2cNRT(Y,X,C,n,p)

</code></pre>

<hr>
<h2 id='ZZZ2023.TSBF.2cNRT'>Normal-reference-test with two-cumulant (2-c) matched $\chi^2$-approximation for two-sample BF problem proposed by Zhang et al. (2023)</h2><span id='topic+ZZZ2023.TSBF.2cNRT'></span>

<h3>Description</h3>

<p>Zhang et al. (2023)'s test for testing equality of two-sample high-dimensional mean vectors without assuming that two covariance matrices are the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZZZ2023.TSBF.2cNRT(y1, y2, cutoff)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZZZ2023.TSBF.2cNRT_+3A_y1">y1</code></td>
<td>
<p>The data matrix (<code class="reqn">n_1 \times p</code>) from the first population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZZZ2023.TSBF.2cNRT_+3A_y2">y2</code></td>
<td>
<p>The data matrix (<code class="reqn">n_2 \times p</code>) from the second population. Each row represents a <code class="reqn">p</code>-dimensional observation.</p>
</td></tr>
<tr><td><code id="ZZZ2023.TSBF.2cNRT_+3A_cutoff">cutoff</code></td>
<td>
<p>An empirical criterion for applying the adjustment coefficient</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have two independent high-dimensional samples:
</p>
<p style="text-align: center;"><code class="reqn">
\boldsymbol{y}_{i1},\ldots,\boldsymbol{y}_{in_i}, \;\operatorname{are \; i.i.d. \; with}\; \operatorname{E}(\boldsymbol{y}_{i1})=\boldsymbol{\mu}_i,\; \operatorname{Cov}(\boldsymbol{y}_{i1})=\boldsymbol{\Sigma}_i,i=1,2.
</code>
</p>

<p>The primary object is to test
</p>
<p style="text-align: center;"><code class="reqn">H_{0}: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2\; \operatorname{versus}\; H_{1}: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2.</code>
</p>

<p>Zhang et al.(2023) proposed the following test statistic:
</p>
<p style="text-align: center;"><code class="reqn">T_{ZZZ}=\frac{n_1 n_2}{np}(\bar{\boldsymbol{y}}_1-\bar{\boldsymbol{y}}_2)^{\top} \hat{\boldsymbol{D}}_n^{-1}(\bar{\boldsymbol{y}}_1-\bar{\boldsymbol{y}}_2),</code>
</p>

<p>where <code class="reqn">\bar{\boldsymbol{y}}_{i},i=1,2</code> are the sample mean vectors, and <code class="reqn">\hat{\boldsymbol{D}}_n=\operatorname{diag}(\hat{\boldsymbol{\Sigma}}_1/n+\hat{\boldsymbol{\Sigma}}_2/n)</code> with <code class="reqn">n=n_1+n_2</code>.
They showed that under the null hypothesis, <code class="reqn">T_{ZZZ}</code> and a chi-squared-type mixture have the same limiting distribution.
</p>


<h3>Value</h3>

<p>A list of class <code>"NRtest"</code> containing the results of the hypothesis test. See the help file for <code><a href="#topic+NRtest.object">NRtest.object</a></code> for details.
</p>


<h3>References</h3>

<p>Zhang L, Zhu T, Zhang J (2023).
&ldquo;Two-sample Behrens&ndash;Fisher problems for high-dimensional data: a normal reference scale-invariant test.&rdquo;
<em>Journal of Applied Statistics</em>, <b>50</b>(3), 456&ndash;476.
<a href="https://doi.org/10.1080/02664763.2020.1834516">doi:10.1080/02664763.2020.1834516</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("HDNRA")
data("COVID19")
dim(COVID19)
group1 &lt;- as.matrix(COVID19[c(2:19, 82:87), ]) ## healthy group
group2 &lt;- as.matrix(COVID19[-c(1:19, 82:87), ]) ## COVID-19 patients
ZZZ2023.TSBF.2cNRT(group1,group2,cutoff=1.2)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
