<!DOCTYPE html><html lang="en"><head><title>Help for package HDXBoxeR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HDXBoxeR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#all_summary'><p>Returns full summary table.</p></a></li>
<li><a href='#arg_df'><p>Returns initially processed data.frame from the export from the HDXExaminer</p></a></li>
<li><a href='#arg_UN_FD'><p>Returns initially processed data.frame from the export from the HDXExaminer</p></a></li>
<li><a href='#arguments_call1'><p>Returns default arguments for the output_tp functions. States</p></a></li>
<li><a href='#arguments_call2'><p>Returns default arguments for the output_tp functions. Deut.Time</p></a></li>
<li><a href='#arguments_call3'><p>Returns default arguments for the output_tp functions. # replicates</p></a></li>
<li><a href='#av_tc'><p>Preparatory function for average plot for timecourses</p></a></li>
<li><a href='#av_tp'><p>Preparatory function for average plot</p></a></li>
<li><a href='#ave_timepoint'><p>Returns average value for either uptake of procent data.</p></a></li>
<li><a href='#average_timecourse'><p>Calculates average for time course data.</p></a></li>
<li><a href='#backHX_calculations'><p>Summary of backexchange summary</p></a></li>
<li><a href='#boxplot_tp'><p>Plots boxplots for all the averages in the set</p></a></li>
<li><a href='#CI_2pts'><p>Global confidence interval treshold from experimental standard deviation for 2 samples.</p></a></li>
<li><a href='#CI_single'><p>Global confidence interval treshold from experimental standard deviation for 1 sample</p></a></li>
<li><a href='#CI_tc'><p>Critial interval calculation two sets of timecourses</p></a></li>
<li><a href='#CI_tp'><p>Global confidence interval treshold from experimental standard deviation</p></a></li>
<li><a href='#color_ranges_Blue_Red_heat_map'><p>Returns color pallete from red to blue with number of colors for defined ranges</p></a></li>
<li><a href='#color_ranges_Spectral'><p>Returns Spectral pallette with colors matching defined ranges</p></a></li>
<li><a href='#coverage_residue'><p>Returns coverage per residue</p></a></li>
<li><a href='#deuteration_woods_timecourse'><p>Return woods plots for the timecourse</p></a></li>
<li><a href='#deuteration_woods_timepoints'><p>Return woods plots for the timepoints</p></a></li>
<li><a href='#dif_ave'><p>Returns data frame with difference of averages between State1 and other states provided.</p></a></li>
<li><a href='#dif_tp'><p>Preparatory function for difference plot</p></a></li>
<li><a href='#dif_tp_proc'><p>Preparatory function for difference plot</p></a></li>
<li><a href='#duplicate_sets'><p>Duplicate set function</p></a></li>
<li><a href='#extreme_input_gap'><p>Makes input for Extreme for bimodal analysis.</p></a></li>
<li><a href='#extreme_input_undeut'><p>Makes input for Extreme for bimodal analysis.</p></a></li>
<li><a href='#general_info'><p>Provides summary table for all data.sets.</p></a></li>
<li><a href='#getCoords1'><p>function from plotfunctions package</p></a></li>
<li><a href='#heat_map_tc'><p>Plots heat maps for time courses.</p></a></li>
<li><a href='#heat_map_tp'><p>Preparatory function for heat map</p></a></li>
<li><a href='#heat_map_tp_maxuptake'><p>Preparatory function for heat map of maximum uptake per residue.</p></a></li>
<li><a href='#heat_map_tp_maxuptake_proc'><p>Preparatory function for heat map of maximum procent deuteration per residue.</p></a></li>
<li><a href='#heat_map_tp_proc'><p>Preparatory function for heat map for procent deuteration</p></a></li>
<li><a href='#is.nan.data.frame'><p>Checks for NaN is data.frame</p></a></li>
<li><a href='#lab_dif'><p>Legend for difference in averages plot.</p></a></li>
<li><a href='#lab_dif_proc'><p>Preparatory function for difference plot for procent deuteration</p></a></li>
<li><a href='#lab_vol'><p>Preparatory function for volcano plot legends</p></a></li>
<li><a href='#legend_heat_map'><p>Legend for the heatmaps prep function.</p></a></li>
<li><a href='#legend_heat_map_tc'><p>Legend for the heatmaps for timecourses.</p></a></li>
<li><a href='#legend_heat_map_timecourse'><p>Legend for the heatmaps prep function for timecourses.</p></a></li>
<li><a href='#legend_heat_map_tp'><p>Legend for the heatmaps.Extracts names from data.frame</p></a></li>
<li><a href='#legend_heat_map_tp_proc'><p>Legend for the heatmaps percent.Extracts names from data.frame</p></a></li>
<li><a href='#legend_nm_bottom'><p>Legend, bottom of the plots</p></a></li>
<li><a href='#legend_raw_ave'><p>Legend for average plot.</p></a></li>
<li><a href='#legend_raw_ave_proc'><p>Preparatory function to draw legends for average procent</p></a></li>
<li><a href='#legend_raw_ave_tc'><p>Legend for average deuteration plot for timecourse.</p></a></li>
<li><a href='#legend_sig_peptides'><p>Legend for the significant peptides</p></a></li>
<li><a href='#legend_states_PerD_bottom'><p>Legend, bottom of the plots</p></a></li>
<li><a href='#legend_tc_bottom'><p>Preparatory function returns legends for the timecourses.</p></a></li>
<li><a href='#nb_exch_deut'><p>Number of exchangeable protons</p></a></li>
<li><a href='#nm_states'><p>Lists names of states in data sets</p></a></li>
<li><a href='#output_FD'><p>Prepares output for HDX-MS Full deuteration data</p></a></li>
<li><a href='#output_FD_proc'><p>Prepares output for HDX-MS Full deuteration data  for procent deuteration.</p></a></li>
<li><a href='#output_prep'><p>Prepares output with HDX-MS data for publications</p></a></li>
<li><a href='#output_tc'><p>Prepares output for HDX-MS for the deuteration uptake or percent deuteration for the time courses.</p></a></li>
<li><a href='#output_tp'><p>Prepares output for HDX-MS for the deuteration uptake or percent deuteration for the time points.</p></a></li>
<li><a href='#output_UD'><p>Prepares output for HDX-MS  Undeuterated sample data.</p></a></li>
<li><a href='#output_UD_proc'><p>Prepares output for HDX-MS Undeuterated data for procent deuteration.</p></a></li>
<li><a href='#pallette_legend'><p>Color scheme using heatmap. Legend Extracts names from data.frame</p></a></li>
<li><a href='#pallette_ll'><p>Color scheme using heatmap. Legend extracts names from data frame</p></a></li>
<li><a href='#peptide_pv_tp'><p>Preparatory function for significant peptide plots</p></a></li>
<li><a href='#peptide_pv_tp_proc'><p>Preparatory function for showing peptides with significant differences between sets.</p></a></li>
<li><a href='#pl_gen_ch2'><p>Prepares the plot window for the woods functions</p></a></li>
<li><a href='#pl_gen_uptake'><p>Prepares the plot window for the woods functions</p></a></li>
<li><a href='#plot_heat_map_max_uptake_tp'><p>Plots heat maps for maximum uptake per residue.</p></a></li>
<li><a href='#plot_heat_map_max_uptake_tp_proc'><p>Plots heat maps for maximum procent deuteration per residue.</p></a></li>
<li><a href='#plot_heat_map_tc'><p>Plots heat maps for time courses.</p></a></li>
<li><a href='#plot_heat_map_tp'><p>Plots heat maps for significant peptides.</p></a></li>
<li><a href='#plot_heat_map_tp_proc'><p>Plots heat maps for significant peptides.</p></a></li>
<li><a href='#plot_peptide_sig_tp'><p>Significant peptide plots.</p></a></li>
<li><a href='#plot_peptide_sig_tp_proc'><p>Draws peptides with significant difefrences between sets.</p></a></li>
<li><a href='#plots_av_tcourse'><p>Generates average deuteration plot for the time-course.</p></a></li>
<li><a href='#plots_av_tp'><p>Returns average deuteration plot for timepoints in the data frame</p></a></li>
<li><a href='#plots_av_tp_proc'><p>Returns average procent deuteration plot for time points</p></a></li>
<li><a href='#plots_diff_tp'><p>Returns difference in average plot for timepoints in the data frame</p></a></li>
<li><a href='#plots_diff_tp_proc'><p>Returns difference in average procent deuteration plot for timepoints in the data frame</p></a></li>
<li><a href='#plots_vol_tp'><p>Returns volcano plots for timepoints in the data frame</p></a></li>
<li><a href='#ppar'><p>Preparation of figure window.</p></a></li>
<li><a href='#ppar_bottom_legend'><p>Preparation of figure window with area for figure at the bottom.</p></a></li>
<li><a href='#ppar_wider'><p>Preparation of figure window with more area on west side of plot.</p></a></li>
<li><a href='#pparLM'><p>Preparation of figure window. small margins</p></a></li>
<li><a href='#prep_timecourse_plot_ave'><p>Prepares function for plotting averages in timecourse</p></a></li>
<li><a href='#prep_timecourse_plot_sd'><p>Prepares function for Critical interval for timecourses</p></a></li>
<li><a href='#pv_timecourse'><p>pvalue calculation between two sets of the data at certain timepoint</p></a></li>
<li><a href='#pv_timepoint'><p>Calculation of pvalue between first protein state and any other state from all_states file</p></a></li>
<li><a href='#pymol_script_average_residue'><p>Writes a text files with pymol scripts to list significant residues.</p></a></li>
<li><a href='#pymol_script_significant_peptide'><p>Writes a text files with pymol scripts to list significant peptides</p></a></li>
<li><a href='#pymol_script_significant_peptide_proc'><p>Writes a text files with pymol scripts to list significant peptides</p></a></li>
<li><a href='#pymol_script_significant_residue'><p>Writes a text files with pymol scripts to list significant residues.</p></a></li>
<li><a href='#pymol_script_significant_residue_proc'><p>Writes a text files with pymol scripts to list significant residues.</p></a></li>
<li><a href='#pymol_str'><p>Preparatory function writing pymol scripts</p></a></li>
<li><a href='#qpcr.cbind.na'><p>Hidden function from qpcR package, typical usage as qpcR:::cbind.na</p></a></li>
<li><a href='#ranges_function'><p>Gives ranges for the averages</p></a></li>
<li><a href='#ranges_function_tc'><p>Gives ranges for the averages for time course analysis</p></a></li>
<li><a href='#rbind_na'><p>bind non equal row</p></a></li>
<li><a href='#reset_par'><p>Reset plotting window parameters to default</p></a></li>
<li><a href='#robot_2states_indexes'><p>Returns a robot plot for selected peptides for 2 protein states.</p></a></li>
<li><a href='#robot_indexes'><p>Returns indexes for peptides with significant difference between two sets</p></a></li>
<li><a href='#robot_indexes_df'><p>Returns dataframe with peptides which exhibit significant difference between two sets</p></a></li>
<li><a href='#robot_plot_All'><p>Returns a robot plot for comparisons of the timepoints samples</p></a></li>
<li><a href='#sd_timecourse'><p>Returns standard deviation for uptake data for timecourses.</p></a></li>
<li><a href='#sd_timecourse_proc'><p>Returns standard deviation for percent deuteration data for timecourses.</p></a></li>
<li><a href='#sd_timepoint'><p>Returns standard deviation for dataframe.</p></a></li>
<li><a href='#select_indices'><p>Allows for selecting some peptide from input data</p></a></li>
<li><a href='#significant_peptide_uptake'><p>Function returns which peptides are significantly based of pv_cutoff and Critial interval</p></a></li>
<li><a href='#summary_sd_CI'><p>Provides summary table with Critical interval and standard deviation within the set.</p></a></li>
<li><a href='#uptake_plots'><p>Uptake plots</p></a></li>
<li><a href='#verbose_timecourse_output'><p>Returns csv with averages from analysis for procent deuteration file, standard deviation for time courses.</p></a></li>
<li><a href='#verbose_timepoint_output'><p>Returns csv with averages from analysis for uptake file, standard deviation, p-values.</p></a></li>
<li><a href='#vol_tp'><p>Preparatory function for volcano plot</p></a></li>
<li><a href='#woods_CI_plot'><p>Returns a woods plot for comparisons of the timepoints samples</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysis of Hydrogen-Deuterium Exchange Mass-Spectrometry Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Maria K. Janowska <a href="https://orcid.org/0000-0002-8232-461X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Katherine Reiter [ctb],
  Pearl Magala [ctb],
  Miklos Guttman [ctb],
  Rachel E. Klevit [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Maria K. Janowska &lt;mka.janowska@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A protocol that facilitates the processing and analysis of Hydrogen-Deuterium Exchange Mass Spectrometry data using p-value statistics and Critical Interval analysis. 
    It provides a pipeline for analyzing data from 'HDXExaminer' (Sierra Analytics, Trajan Scientific), 
    automating matching and comparison of protein states through Welch's T-test and the Critical Interval statistical framework. 
    Additionally, it simplifies data export, generates 'PyMol' scripts, and ensures calculations meet publication standards. 
    'HDXBoxeR' assists in various aspects of hydrogen-deuterium exchange data analysis, including reprocessing data, calculating parameters, identifying significant peptides, 
    generating plots, and facilitating comparison between protein states. 
    For details check papers by Hageman and Weis (2019) &lt;<a href="https://doi.org/10.1021%2Facs.analchem.9b01325">doi:10.1021/acs.analchem.9b01325</a>&gt; 
    and Masson et al. (2019) &lt;<a href="https://doi.org/10.1038%2Fs41592-019-0459-y">doi:10.1038/s41592-019-0459-y</a>&gt;.
    'HDXBoxeR' citation: Janowska et al. (2024) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtae479">doi:10.1093/bioinformatics/btae479</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, graphics, grDevices, RColorBrewer, stats, stringr,
tidyr, utils, methods, wrapr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-08-23 19:16:44 UTC; mkaja</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-08-24 07:30:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='all_summary'>Returns full summary table.</h2><span id='topic+all_summary'></span>

<h3>Description</h3>

<p>Returns summary data.
Function returns: Protein states, timepoints, number of replicates,  # peptides, % coveregae, average peptide length and redundancy.
backexchange calculations (average and range), Critical interval and standard deviation.
Function requires undeuterated and Fully deuterated sets marked in Deut.time as 0s and FD respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>all_summary(filepath, replicates = 3, Dfact = 0.85)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="all_summary_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
<tr><td><code id="all_summary_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates. Default set to 3.</p>
</td></tr>
<tr><td><code id="all_summary_+3A_dfact">Dfact</code></td>
<td>
<p>Dfact is the fraction of D/H in the labeling buffer used. Default set up to 0.85</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns summary table.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- all_summary(file_nm, replicates=3, Dfact=0.85)
</code></pre>

<hr>
<h2 id='arg_df'>Returns initially processed data.frame from the export from the HDXExaminer</h2><span id='topic+arg_df'></span>

<h3>Description</h3>

<p>Function used as internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arg_df(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="arg_df_+3A_filepath">filepath</code></td>
<td>
<p>input file location</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame for further processing
</p>

<hr>
<h2 id='arg_UN_FD'>Returns initially processed data.frame from the export from the HDXExaminer</h2><span id='topic+arg_UN_FD'></span>

<h3>Description</h3>

<p>Function used as internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arg_UN_FD(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="arg_UN_FD_+3A_filepath">filepath</code></td>
<td>
<p>input file location</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame for further processing
</p>

<hr>
<h2 id='arguments_call1'>Returns default arguments for the output_tp functions. States</h2><span id='topic+arguments_call1'></span>

<h3>Description</h3>

<p>Function used as internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arguments_call1(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="arguments_call1_+3A_filepath">filepath</code></td>
<td>
<p>input file location</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The default arguments to output_tp functions.
</p>

<hr>
<h2 id='arguments_call2'>Returns default arguments for the output_tp functions. Deut.Time</h2><span id='topic+arguments_call2'></span>

<h3>Description</h3>

<p>Function used as internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arguments_call2(filepath, states)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="arguments_call2_+3A_filepath">filepath</code></td>
<td>
<p>input file location</p>
</td></tr>
<tr><td><code id="arguments_call2_+3A_states">states</code></td>
<td>
<p>states used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The default arguments to output_tp functions.
</p>

<hr>
<h2 id='arguments_call3'>Returns default arguments for the output_tp functions. # replicates</h2><span id='topic+arguments_call3'></span>

<h3>Description</h3>

<p>Function used as internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arguments_call3(filepath, states, times)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="arguments_call3_+3A_filepath">filepath</code></td>
<td>
<p>input file location</p>
</td></tr>
<tr><td><code id="arguments_call3_+3A_states">states</code></td>
<td>
<p>states used</p>
</td></tr>
<tr><td><code id="arguments_call3_+3A_times">times</code></td>
<td>
<p>deuteration times</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The default arguments to output_tp functions.
</p>

<hr>
<h2 id='av_tc'>Preparatory function for average plot for timecourses</h2><span id='topic+av_tc'></span>

<h3>Description</h3>

<p>Returns plots with average deuteration at each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>av_tc(df, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="av_tc_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="av_tc_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots of averages
</p>

<hr>
<h2 id='av_tp'>Preparatory function for average plot</h2><span id='topic+av_tp'></span>

<h3>Description</h3>

<p>Returns plots with average deuteration at each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>av_tp(df, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="av_tp_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="av_tp_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots of averages
</p>

<hr>
<h2 id='ave_timepoint'>Returns average value for either uptake of procent data.</h2><span id='topic+ave_timepoint'></span>

<h3>Description</h3>

<p>Calculates average of uptake or procent data. Returns data frame with average values.
Default for the number of replicates is 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ave_timepoint(df, replicates = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ave_timepoint_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="ave_timepoint_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates used. Default is set to replicates=3</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame with average values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
ave&lt;-ave_timepoint(df=a) ##if number of replicates is equal 3
ave&lt;-ave_timepoint(df=a, replicates=4) ##if number of replicates is equal 4
</code></pre>

<hr>
<h2 id='average_timecourse'>Calculates average for time course data.</h2><span id='topic+average_timecourse'></span>

<h3>Description</h3>

<p>Calculates average for time course data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average_timecourse(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="average_timecourse_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the All_results input file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with average deuteration uptake data.
</p>

<hr>
<h2 id='backHX_calculations'>Summary of backexchange summary</h2><span id='topic+backHX_calculations'></span>

<h3>Description</h3>

<p>Returns average and ranges of backexchange. Function calculates as: 1- (m100%-m0%)/N/Dfact.
m0% is the non-deuterated peptide centroid mass, m100% is the maximally labeled peptide centroid mass, N is
the theoretical number of backbone amides in the peptide and Dfrac is the fraction of D/H in the labeling buffer used.
Function requires undeuterated and Fully deuterated sets marked in Deut.time as 0s and FD respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>backHX_calculations(filepath, Dfact = 0.85)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="backHX_calculations_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
<tr><td><code id="backHX_calculations_+3A_dfact">Dfact</code></td>
<td>
<p>is the fraction of D/H in the labeling buffer used. Default set up to 0.85</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns summary table for backexchange.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- backHX_calculations(filepath=file_nm, Dfact=0.85)
</code></pre>

<hr>
<h2 id='boxplot_tp'>Plots boxplots for all the averages in the set</h2><span id='topic+boxplot_tp'></span>

<h3>Description</h3>

<p>Returns boxplots to compare sets between each other
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boxplot_tp(df, replicates = 3, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="boxplot_tp_+3A_df">df</code></td>
<td>
<p>average data frame. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="boxplot_tp_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
<tr><td><code id="boxplot_tp_+3A_...">...</code></td>
<td>
<p>inherited boxplot parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>boxplots for average deuterium uptake per set.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
boxplot_tp(df=a, replicates=3)
</code></pre>

<hr>
<h2 id='CI_2pts'>Global confidence interval treshold from experimental standard deviation for 2 samples.</h2><span id='topic+CI_2pts'></span>

<h3>Description</h3>

<p>Calculation of global confidence interval using approach by:
Reliable Identification of Significant Differences in Differential Hydrogen Exchange-Mass Spectrometry Measurements Using a Hybrid Significance Testing Approach
Tyler S. Hageman and David D. Weis
Analytical Chemistry 2019 91 (13), 8008-8016 DOI: 10.1021/acs.analchem.9b01325
calculations for alpha 0.99
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CI_2pts(s1, s2, replicates = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CI_2pts_+3A_s1">s1</code></td>
<td>
<p>standard deviation from one sample</p>
</td></tr>
<tr><td><code id="CI_2pts_+3A_s2">s2</code></td>
<td>
<p>standard deviation from seconda sample</p>
</td></tr>
<tr><td><code id="CI_2pts_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>treshold for determining significance.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sd1&lt;-data.frame(c(0.1, 0.12, 0.13, 0.09, 0.11, 0.10))
sd2&lt;-data.frame(c(0.18, 0.11, 0.13, 0.08, 0.11, 0.06))
CI_2pts(s1=sd1, s2=sd2, replicates=3)
</code></pre>

<hr>
<h2 id='CI_single'>Global confidence interval treshold from experimental standard deviation for 1 sample</h2><span id='topic+CI_single'></span>

<h3>Description</h3>

<p>Calculation of global confidence interval using approach by:
Reliable Identification of Significant Differences in Differential Hydrogen Exchange-Mass Spectrometry Measurements Using a Hybrid Significance Testing Approach
Tyler S. Hageman and David D. Weis
Analytical Chemistry 2019 91 (13), 8008-8016 DOI: 10.1021/acs.analchem.9b01325
calculations for alpha 0.99
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CI_single(s1, replicates = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CI_single_+3A_s1">s1</code></td>
<td>
<p>standard deviation from one sample</p>
</td></tr>
<tr><td><code id="CI_single_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>treshold for determining significance.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sd1&lt;-data.frame(c(0.1, 0.12, 0.13, 0.09, 0.11, 0.10))
CI_single(s1=sd1, replicates=3)
</code></pre>

<hr>
<h2 id='CI_tc'>Critial interval calculation two sets of timecourses</h2><span id='topic+CI_tc'></span>

<h3>Description</h3>

<p>Preparatory function for calculation of pvalue between sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CI_tc(sd_c, sd_v, replicates = 3, pv_cutoff = 0.01)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CI_tc_+3A_sd_c">sd_c</code></td>
<td>
<p>dataframe of control</p>
</td></tr>
<tr><td><code id="CI_tc_+3A_sd_v">sd_v</code></td>
<td>
<p>dataframe for variant</p>
</td></tr>
<tr><td><code id="CI_tc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates. Default set to 3.</p>
</td></tr>
<tr><td><code id="CI_tc_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>pvalue cutoff. Default set to 0.01</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Critical interval for 2 sets
</p>

<hr>
<h2 id='CI_tp'>Global confidence interval treshold from experimental standard deviation</h2><span id='topic+CI_tp'></span>

<h3>Description</h3>

<p>Calculation of global confidence interval using approach by for all protein states compared to first state in the data.frame.
Reliable Identification of Significant Differences in Differential Hydrogen Exchange-Mass Spectrometry Measurements Using a Hybrid Significance Testing Approach
Tyler S. Hageman and David D. Weis
Analytical Chemistry 2019 91 (13), 8008-8016 DOI: 10.1021/acs.analchem.9b01325
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CI_tp(df, replicates = 3, alpha = 0.01)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CI_tp_+3A_df">df</code></td>
<td>
<p>standard deviation dataframe.</p>
</td></tr>
<tr><td><code id="CI_tp_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates. Default set to 3.</p>
</td></tr>
<tr><td><code id="CI_tp_+3A_alpha">alpha</code></td>
<td>
<p>significance level. Set as default to 0.01</p>
</td></tr>
</table>


<h3>Value</h3>

<p>treshold for determining significance.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tc(file_nm, seq_match=FALSE)
sd&lt;-sd_timepoint(df=a, replicates=3)
CI_tp(df=sd, replicates=3, alpha=0.01 )
CI_tp(sd)
</code></pre>

<hr>
<h2 id='color_ranges_Blue_Red_heat_map'>Returns color pallete from red to blue with number of colors for defined ranges</h2><span id='topic+color_ranges_Blue_Red_heat_map'></span>

<h3>Description</h3>

<p>Returns color pallete from red to blue with number of colors for defined ranges
</p>


<h3>Usage</h3>

<pre><code class='language-R'>color_ranges_Blue_Red_heat_map(ranges, colors_initial)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="color_ranges_Blue_Red_heat_map_+3A_ranges">ranges</code></td>
<td>
<p>vector of numbers. Should have the same mumber of positive and negative values and contain 0.</p>
</td></tr>
<tr><td><code id="color_ranges_Blue_Red_heat_map_+3A_colors_initial">colors_initial</code></td>
<td>
<p>additional color that should be first in the pallette.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>color scheme for number
</p>


<h3>Examples</h3>

<pre><code class='language-R'>color_ranges_Blue_Red_heat_map(ranges=c(-Inf, -100, -50, 0, 50, 100, Inf), colors_initial="white")
</code></pre>

<hr>
<h2 id='color_ranges_Spectral'>Returns Spectral pallette with colors matching defined ranges</h2><span id='topic+color_ranges_Spectral'></span>

<h3>Description</h3>

<p>Spectral pallette for timecourse data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>color_ranges_Spectral(ranges, colors_initial)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="color_ranges_Spectral_+3A_ranges">ranges</code></td>
<td>
<p>vector of numbers. Should have the same mumber of positive and negative values and contain 0.</p>
</td></tr>
<tr><td><code id="color_ranges_Spectral_+3A_colors_initial">colors_initial</code></td>
<td>
<p>additional color that should be first in the pallette.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>color scheme for number
</p>


<h3>Examples</h3>

<pre><code class='language-R'>color_ranges_Spectral(ranges=c(-Inf, -100, -50, 0, 50, 100, Inf), colors_initial="white")
</code></pre>

<hr>
<h2 id='coverage_residue'>Returns coverage per residue</h2><span id='topic+coverage_residue'></span>

<h3>Description</h3>

<p>returns vector with coverage information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coverage_residue(df1, start_col, end_col)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coverage_residue_+3A_df1">df1</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="coverage_residue_+3A_start_col">start_col</code></td>
<td>
<p>number of &quot;Start&quot; column in data.frame</p>
</td></tr>
<tr><td><code id="coverage_residue_+3A_end_col">end_col</code></td>
<td>
<p>number of &quot;Start&quot; column in data.frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector with coverage per residue
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
coverage_residue(df1=a,start_col=2, end_col=3 )
</code></pre>

<hr>
<h2 id='deuteration_woods_timecourse'>Return woods plots for the timecourse</h2><span id='topic+deuteration_woods_timecourse'></span>

<h3>Description</h3>

<p>All the peptides are plotted based on their uptake.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deuteration_woods_timecourse(
  input_data,
  states,
  replicates = 3,
  ylim = c(0, 120),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="deuteration_woods_timecourse_+3A_input_data">input_data</code></td>
<td>
<p>output from function output_tc(..., percent=TRUE)</p>
</td></tr>
<tr><td><code id="deuteration_woods_timecourse_+3A_states">states</code></td>
<td>
<p>states, if missing all states used</p>
</td></tr>
<tr><td><code id="deuteration_woods_timecourse_+3A_replicates">replicates</code></td>
<td>
<p>replicates</p>
</td></tr>
<tr><td><code id="deuteration_woods_timecourse_+3A_ylim">ylim</code></td>
<td>
<p>y axis limits</p>
</td></tr>
<tr><td><code id="deuteration_woods_timecourse_+3A_...">...</code></td>
<td>
<p>other parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Woods plots for the timecourse
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tc(file_nm, percent=TRUE)
deuteration_woods_timecourse(a)

</code></pre>

<hr>
<h2 id='deuteration_woods_timepoints'>Return woods plots for the timepoints</h2><span id='topic+deuteration_woods_timepoints'></span>

<h3>Description</h3>

<p>All the peptides are plotted based on their uptake.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deuteration_woods_timepoints(
  input_data,
  times,
  replicates = 3,
  cola = NA,
  ylim = c(0, 120),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="deuteration_woods_timepoints_+3A_input_data">input_data</code></td>
<td>
<p>output from function output_tp(..., percent=TRUE)</p>
</td></tr>
<tr><td><code id="deuteration_woods_timepoints_+3A_times">times</code></td>
<td>
<p>Deuteration times, if missing all deuteration times used</p>
</td></tr>
<tr><td><code id="deuteration_woods_timepoints_+3A_replicates">replicates</code></td>
<td>
<p>replicates</p>
</td></tr>
<tr><td><code id="deuteration_woods_timepoints_+3A_cola">cola</code></td>
<td>
<p>colors, default NA</p>
</td></tr>
<tr><td><code id="deuteration_woods_timepoints_+3A_ylim">ylim</code></td>
<td>
<p>y axis limits</p>
</td></tr>
<tr><td><code id="deuteration_woods_timepoints_+3A_...">...</code></td>
<td>
<p>other parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Woods plots for the timepoints
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm, percent=TRUE)
deuteration_woods_timepoints(a[1:12,])

</code></pre>

<hr>
<h2 id='dif_ave'>Returns data frame with difference of averages between State1 and other states provided.</h2><span id='topic+dif_ave'></span>

<h3>Description</h3>

<p>Returns average difference data.frame.
Sets are compared to the first state in the input file.
If other order of the sets is required use
Default for the number of replicates is 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dif_ave(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dif_ave_+3A_df">df</code></td>
<td>
<p>output from functions output_tp, output_tp_proc,  output_tp_states or output_tp_proc_states.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame with difference values btw control and other protein states.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
pv&lt;-pv_timepoint(df=a) ##if number of replicates is equal 3
pv1&lt;-pv_timepoint(df=a, replicates=3) ##if number of replicates is equal 4
#b&lt;-output_tp_states(file_nm, states=c("4EHP", "State2", "State3" ))
#pv_states&lt;-pv_timepoint(df=b) ### here means of State4, will be compared to State2 and State4
</code></pre>

<hr>
<h2 id='dif_tp'>Preparatory function for difference plot</h2><span id='topic+dif_tp'></span>

<h3>Description</h3>

<p>Returns plots with difference deuteration at each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dif_tp(df, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dif_tp_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="dif_tp_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots of difference in average
</p>

<hr>
<h2 id='dif_tp_proc'>Preparatory function for difference plot</h2><span id='topic+dif_tp_proc'></span>

<h3>Description</h3>

<p>Returns plots with difference deuteration at each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dif_tp_proc(df, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dif_tp_proc_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="dif_tp_proc_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots of difference in average
</p>

<hr>
<h2 id='duplicate_sets'>Duplicate set function</h2><span id='topic+duplicate_sets'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>duplicate_sets(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="duplicate_sets_+3A_df">df</code></td>
<td>
<p>dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>duplicate sets
</p>

<hr>
<h2 id='extreme_input_gap'>Makes input for Extreme for bimodal analysis.</h2><span id='topic+extreme_input_gap'></span>

<h3>Description</h3>

<p>Makes input for Extreme for bimodal analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extreme_input_gap(hm_dir, replicates, timepoints, output_path = "NA")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extreme_input_gap_+3A_hm_dir">hm_dir</code></td>
<td>
<p>directory in which all the folders which needs to be processed are</p>
</td></tr>
<tr><td><code id="extreme_input_gap_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample</p>
</td></tr>
<tr><td><code id="extreme_input_gap_+3A_timepoints">timepoints</code></td>
<td>
<p>lists timepoints used in experiments.</p>
</td></tr>
<tr><td><code id="extreme_input_gap_+3A_output_path">output_path</code></td>
<td>
<p>directory where the output files will be saved, hm_dir default</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Inputs for extreme for all data prepared.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
path_to_folders&lt;-system.file("extdata",  package = "HDXBoxeR")

extreme_input_gap(hm_dir =path_to_folders, replicates = 3,
timepoints =c(3, 60, 1800, 72000), output_path=tempdir())

</code></pre>

<hr>
<h2 id='extreme_input_undeut'>Makes input for Extreme for bimodal analysis.</h2><span id='topic+extreme_input_undeut'></span>

<h3>Description</h3>

<p>If data is missing it returns non-deuterated data in these columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extreme_input_undeut(hm_dir, replicates, timepoints, output_path = "NA")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extreme_input_undeut_+3A_hm_dir">hm_dir</code></td>
<td>
<p>directory in which all the folders which needs to be processed are</p>
</td></tr>
<tr><td><code id="extreme_input_undeut_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample</p>
</td></tr>
<tr><td><code id="extreme_input_undeut_+3A_timepoints">timepoints</code></td>
<td>
<p>lists timepoints used in experiments.</p>
</td></tr>
<tr><td><code id="extreme_input_undeut_+3A_output_path">output_path</code></td>
<td>
<p>directory where output should be written</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Inputs for extreme for all data prepared.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
path_to_folders&lt;-system.file("extdata",  package = "HDXBoxeR")
extreme_input_undeut(hm_dir=path_to_folders, replicates = 3,
timepoints =c(3, 60, 1800, 72000), output_path=tempdir())

</code></pre>

<hr>
<h2 id='general_info'>Provides summary table for all data.sets.</h2><span id='topic+general_info'></span>

<h3>Description</h3>

<p>Returns data frame sumamrizing general information about the data sets.
Function returns: Protein states, timepoints, number of replicates,  # peptides, % coveregae, average peptide length and redundancy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>general_info(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="general_info_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns summary table.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- general_info(file_nm)
</code></pre>

<hr>
<h2 id='getCoords1'>function from plotfunctions package</h2><span id='topic+getCoords1'></span>

<h3>Description</h3>

<p>Margin coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCoords1(pos = 1.1, side = 1, input = "p")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCoords1_+3A_pos">pos</code></td>
<td>
<p>position</p>
</td></tr>
<tr><td><code id="getCoords1_+3A_side">side</code></td>
<td>
<p>side of plot</p>
</td></tr>
<tr><td><code id="getCoords1_+3A_input">input</code></td>
<td>
<p>plot or figure position</p>
</td></tr>
</table>


<h3>Value</h3>

<p>coordinates of margins
</p>

<hr>
<h2 id='heat_map_tc'>Plots heat maps for time courses.</h2><span id='topic+heat_map_tc'></span>

<h3>Description</h3>

<p>Returns heat map on timecourses with raw data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heat_map_tc(df, ranges = c(seq(0, 100, by = 10), Inf))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="heat_map_tc_+3A_df">df</code></td>
<td>
<p>timecourse input</p>
</td></tr>
<tr><td><code id="heat_map_tc_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(seq(0, 100, by=10), Inf)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>heat map for timecourses
</p>

<hr>
<h2 id='heat_map_tp'>Preparatory function for heat map</h2><span id='topic+heat_map_tp'></span>

<h3>Description</h3>

<p>Returns heat map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heat_map_tp(
  df,
  pv,
  sd,
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="heat_map_tp_+3A_df">df</code></td>
<td>
<p>average data frame. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="heat_map_tp_+3A_pv">pv</code></td>
<td>
<p>pvalues dataframes calculated using pv_timepoint() function</p>
</td></tr>
<tr><td><code id="heat_map_tp_+3A_sd">sd</code></td>
<td>
<p>standard deviation data.frame generated using sd_timepoint function</p>
</td></tr>
<tr><td><code id="heat_map_tp_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="heat_map_tp_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="heat_map_tp_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>heat map for timepoints
</p>

<hr>
<h2 id='heat_map_tp_maxuptake'>Preparatory function for heat map of maximum uptake per residue.</h2><span id='topic+heat_map_tp_maxuptake'></span>

<h3>Description</h3>

<p>Returns heat map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heat_map_tp_maxuptake(
  df,
  pv,
  sd,
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="heat_map_tp_maxuptake_+3A_df">df</code></td>
<td>
<p>average data frame. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="heat_map_tp_maxuptake_+3A_pv">pv</code></td>
<td>
<p>pvalues dataframes calculated using pv_timepoint() function</p>
</td></tr>
<tr><td><code id="heat_map_tp_maxuptake_+3A_sd">sd</code></td>
<td>
<p>standard deviation data.frame generated using sd_timepoint function</p>
</td></tr>
<tr><td><code id="heat_map_tp_maxuptake_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="heat_map_tp_maxuptake_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="heat_map_tp_maxuptake_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>maxiumum uptake heat map for timepoints
</p>

<hr>
<h2 id='heat_map_tp_maxuptake_proc'>Preparatory function for heat map of maximum procent deuteration per residue.</h2><span id='topic+heat_map_tp_maxuptake_proc'></span>

<h3>Description</h3>

<p>Returns heat map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heat_map_tp_maxuptake_proc(
  df,
  dfup,
  pv,
  sd,
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="heat_map_tp_maxuptake_proc_+3A_df">df</code></td>
<td>
<p>average data frame for procent deuteration. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="heat_map_tp_maxuptake_proc_+3A_dfup">dfup</code></td>
<td>
<p>average data frame for deuteration uptake. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="heat_map_tp_maxuptake_proc_+3A_pv">pv</code></td>
<td>
<p>pvalues dataframes calculated using pv_timepoint() function</p>
</td></tr>
<tr><td><code id="heat_map_tp_maxuptake_proc_+3A_sd">sd</code></td>
<td>
<p>standard deviation data.frame generated using sd_timepoint function</p>
</td></tr>
<tr><td><code id="heat_map_tp_maxuptake_proc_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="heat_map_tp_maxuptake_proc_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="heat_map_tp_maxuptake_proc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Maximum uptake heat map for timepoints
</p>

<hr>
<h2 id='heat_map_tp_proc'>Preparatory function for heat map for procent deuteration</h2><span id='topic+heat_map_tp_proc'></span>

<h3>Description</h3>

<p>Returns heat map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heat_map_tp_proc(
  df,
  dfup,
  pv,
  sd,
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="heat_map_tp_proc_+3A_df">df</code></td>
<td>
<p>average data frame for procent deuteration. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="heat_map_tp_proc_+3A_dfup">dfup</code></td>
<td>
<p>average data frame for deuteration uptake. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="heat_map_tp_proc_+3A_pv">pv</code></td>
<td>
<p>pvalues dataframes calculated using pv_timepoint() function</p>
</td></tr>
<tr><td><code id="heat_map_tp_proc_+3A_sd">sd</code></td>
<td>
<p>standard deviation data.frame generated using sd_timepoint function</p>
</td></tr>
<tr><td><code id="heat_map_tp_proc_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="heat_map_tp_proc_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="heat_map_tp_proc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>heat map for timepoints
</p>

<hr>
<h2 id='is.nan.data.frame'>Checks for NaN is data.frame</h2><span id='topic+is.nan.data.frame'></span>

<h3>Description</h3>

<p>Function by Hong Ooi; https://stackoverflow.com/questions/18142117/how-to-replace-nan-value-with-zero-in-a-huge-data-frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'data.frame'
is.nan(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.nan.data.frame_+3A_x">x</code></td>
<td>
<p>Data frame to be checked for NaN</p>
</td></tr>
</table>


<h3>Value</h3>

<p>logical. Returns info if data.frame contains NaNs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## this function will overwrite the is.nan function that works only on vectors and matrices
df&lt;-data.frame(c(0,NaN), c(1, 2))
is.nan(df)
df[is.nan(df)]&lt;- 0
</code></pre>

<hr>
<h2 id='lab_dif'>Legend for difference in averages plot.</h2><span id='topic+lab_dif'></span>

<h3>Description</h3>

<p>Returns legend for difference in average plots. Preparatory function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lab_dif(df, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lab_dif_+3A_df">df</code></td>
<td>
<p>output from functions average difference</p>
</td></tr>
<tr><td><code id="lab_dif_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for difference in average plot for time points
</p>

<hr>
<h2 id='lab_dif_proc'>Preparatory function for difference plot for procent deuteration</h2><span id='topic+lab_dif_proc'></span>

<h3>Description</h3>

<p>Returns legends for plots procent deuteration at each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lab_dif_proc(df, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lab_dif_proc_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="lab_dif_proc_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from RColorBrewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legends for procent deuteration plots
</p>

<hr>
<h2 id='lab_vol'>Preparatory function for volcano plot legends</h2><span id='topic+lab_vol'></span>

<h3>Description</h3>

<p>Returns volcano plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lab_vol(df, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lab_vol_+3A_df">df</code></td>
<td>
<p>output from functions output_tp</p>
</td></tr>
<tr><td><code id="lab_vol_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legends for volcano plots
</p>

<hr>
<h2 id='legend_heat_map'>Legend for the heatmaps prep function.</h2><span id='topic+legend_heat_map'></span>

<h3>Description</h3>

<p>Returns names for legend for the heatmaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_heat_map(ranges = c(-Inf, seq(-30, 30, by = 10), Inf))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_heat_map_+3A_ranges">ranges</code></td>
<td>
<p>ranges that are to be colored in the legend. Default ranges=c(-Inf,seq(-30, 30, by=10), Inf )</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for the heatmap
</p>

<hr>
<h2 id='legend_heat_map_tc'>Legend for the heatmaps for timecourses.</h2><span id='topic+legend_heat_map_tc'></span>

<h3>Description</h3>

<p>Returns names for legend for the heatmaps. Extracts names from data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_heat_map_tc(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_heat_map_tc_+3A_df">df</code></td>
<td>
<p>generated using output_tcourse()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for the heatmap
</p>

<hr>
<h2 id='legend_heat_map_timecourse'>Legend for the heatmaps prep function for timecourses.</h2><span id='topic+legend_heat_map_timecourse'></span>

<h3>Description</h3>

<p>Returns names for legend for the heatmaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_heat_map_timecourse(ranges = c(-Inf, seq(0, 100, by = 10), Inf))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_heat_map_timecourse_+3A_ranges">ranges</code></td>
<td>
<p>ranges that are to be colored in the legend. Default ranges=c(-Inf,seq(-30, 30, by=10), Inf )</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for the heatmap
</p>

<hr>
<h2 id='legend_heat_map_tp'>Legend for the heatmaps.Extracts names from data.frame</h2><span id='topic+legend_heat_map_tp'></span>

<h3>Description</h3>

<p>Returns names for legend for the heatmaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_heat_map_tp(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_heat_map_tp_+3A_df">df</code></td>
<td>
<p>average data frame. Generated using ave_timepoint() function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for the heatmap
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
legend_heat_map_tp(df=a)
</code></pre>

<hr>
<h2 id='legend_heat_map_tp_proc'>Legend for the heatmaps percent.Extracts names from data.frame</h2><span id='topic+legend_heat_map_tp_proc'></span>

<h3>Description</h3>

<p>Returns names for legend for the heatmaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_heat_map_tp_proc(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_heat_map_tp_proc_+3A_df">df</code></td>
<td>
<p>average data frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for the heatmap prercent
</p>

<hr>
<h2 id='legend_nm_bottom'>Legend, bottom of the plots</h2><span id='topic+legend_nm_bottom'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_nm_bottom(names, cols)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_nm_bottom_+3A_names">names</code></td>
<td>
<p>labels</p>
</td></tr>
<tr><td><code id="legend_nm_bottom_+3A_cols">cols</code></td>
<td>
<p>colors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend at the bottom of the plot
</p>

<hr>
<h2 id='legend_raw_ave'>Legend for average plot.</h2><span id='topic+legend_raw_ave'></span>

<h3>Description</h3>

<p>Returns legend with average plots. Preparatory function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_raw_ave(df, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_raw_ave_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="legend_raw_ave_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for average plot for time points
</p>

<hr>
<h2 id='legend_raw_ave_proc'>Preparatory function to draw legends for average procent</h2><span id='topic+legend_raw_ave_proc'></span>

<h3>Description</h3>

<p>Returns legend with average procent deuteration at each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_raw_ave_proc(df, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_raw_ave_proc_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="legend_raw_ave_proc_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for average deuteration procent for timepoints
</p>

<hr>
<h2 id='legend_raw_ave_tc'>Legend for average deuteration plot for timecourse.</h2><span id='topic+legend_raw_ave_tc'></span>

<h3>Description</h3>

<p>Returns legend with average plots. Preparatory function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_raw_ave_tc(df, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_raw_ave_tc_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="legend_raw_ave_tc_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for average plot for time course
</p>

<hr>
<h2 id='legend_sig_peptides'>Legend for the significant peptides</h2><span id='topic+legend_sig_peptides'></span>

<h3>Description</h3>

<p>Returns names for legend for the significant peptides plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_sig_peptides(ranges = c(-Inf, seq(-30, 30, by = 10), Inf))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_sig_peptides_+3A_ranges">ranges</code></td>
<td>
<p>ranges that are to be colored in the legend. Default ranges=c(-Inf,seq(-30, 30, by=10), Inf )</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for the heatmap
</p>

<hr>
<h2 id='legend_states_PerD_bottom'>Legend, bottom of the plots</h2><span id='topic+legend_states_PerD_bottom'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_states_PerD_bottom(df, cols)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_states_PerD_bottom_+3A_df">df</code></td>
<td>
<p>dataframe</p>
</td></tr>
<tr><td><code id="legend_states_PerD_bottom_+3A_cols">cols</code></td>
<td>
<p>colors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend at the bottom of the plot
</p>

<hr>
<h2 id='legend_tc_bottom'>Preparatory function returns legends for the timecourses.</h2><span id='topic+legend_tc_bottom'></span>

<h3>Description</h3>

<p>Preparatory function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legend_tc_bottom(df, cols)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legend_tc_bottom_+3A_df">df</code></td>
<td>
<p>data frame from which names will be extracted</p>
</td></tr>
<tr><td><code id="legend_tc_bottom_+3A_cols">cols</code></td>
<td>
<p>colors to be used in legend</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend at the bottom of the plot
</p>

<hr>
<h2 id='nb_exch_deut'>Number of exchangeable protons</h2><span id='topic+nb_exch_deut'></span>

<h3>Description</h3>

<p>Provides a vector with number of exchangeable protons, calculated from the input table.
Number of protons calculated as peptide_length - 2 - number of Prolines in the peptide
that are not in the first position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nb_exch_deut(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nb_exch_deut_+3A_df">df</code></td>
<td>
<p>standard deviation from one sample</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector with number of exchangeable protons
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
nb_exch_deut(a)
</code></pre>

<hr>
<h2 id='nm_states'>Lists names of states in data sets</h2><span id='topic+nm_states'></span>

<h3>Description</h3>

<p>Returns vector with name of states used for choosing states for input functions generation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nm_states(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nm_states_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of Protein States.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
names_states&lt;- nm_states(file_nm)
</code></pre>

<hr>
<h2 id='output_FD'>Prepares output for HDX-MS Full deuteration data</h2><span id='topic+output_FD'></span>

<h3>Description</h3>

<p>Returns a data frame for Full deuteration set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>output_FD(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="output_FD_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with reorganized data where in columns is uptake data for Protein States.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;-output_FD(file_nm)

</code></pre>

<hr>
<h2 id='output_FD_proc'>Prepares output for HDX-MS Full deuteration data  for procent deuteration.</h2><span id='topic+output_FD_proc'></span>

<h3>Description</h3>

<p>Returns a data frame for Full deuteration set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>output_FD_proc(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="output_FD_proc_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with reorganized data where in columns is procent deuteration for Protein States.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_FD_proc(file_nm)
</code></pre>

<hr>
<h2 id='output_prep'>Prepares output with HDX-MS data for publications</h2><span id='topic+output_prep'></span>

<h3>Description</h3>

<p>Format prepared based of example from:
Masson, G.R., Burke, J.E., Ahn, N.G. et al. Recommendations for performing, interpreting and reporting hydrogen deuterium exchange mass spectrometry (HDX-MS) experiments. Nat Methods 16, 595602 (2019). https://doi.org/10.1038/s41592-019-0459-y
It generates csv file in format ready for publication of the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>output_prep(filepath, output_name, states, replicates, times, percent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="output_prep_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
<tr><td><code id="output_prep_+3A_output_name">output_name</code></td>
<td>
<p>Name of output file. It has to be csv file</p>
</td></tr>
<tr><td><code id="output_prep_+3A_states">states</code></td>
<td>
<p>function allows to choose what states should be used for analysis. Default all states are used.</p>
</td></tr>
<tr><td><code id="output_prep_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates to be used in analysis. The function takes number of replicates up to specified number. If no argument provided number maximal common number of replicates it used.</p>
</td></tr>
<tr><td><code id="output_prep_+3A_times">times</code></td>
<td>
<p>lists the deuteration times to be used in analysis. Default all states used.</p>
</td></tr>
<tr><td><code id="output_prep_+3A_percent">percent</code></td>
<td>
<p>return either uptake or percent deuteration, default=FALSE, return uptake</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns&amp;saves data.frame in format that is accepted for the publications.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
output_prep(filepath=file_nm, output_name=tempfile())

</code></pre>

<hr>
<h2 id='output_tc'>Prepares output for HDX-MS for the deuteration uptake or percent deuteration for the time courses.</h2><span id='topic+output_tc'></span>

<h3>Description</h3>

<p>Returns a data frame organized for additional analysis.
In columns are deuteration uptake or percent deuteration data for the given protein states.
Function allows for writing csv with data, matching sequences of peptide.
Protein.States, Deut.times, or number of replicates can be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>output_tc(
  filepath,
  replicates,
  states,
  times,
  seq_match = FALSE,
  csv = "NA",
  percent = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="output_tc_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
<tr><td><code id="output_tc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates to be used in analysis. The function takes number of replicates up to specified number. If no argument provided number maximal common number of replicates it used.</p>
</td></tr>
<tr><td><code id="output_tc_+3A_states">states</code></td>
<td>
<p>function allows to choose what states should be used for analysis. Default all states are used.</p>
</td></tr>
<tr><td><code id="output_tc_+3A_times">times</code></td>
<td>
<p>lists the deuteration times to be used in analysis. Default all states used.</p>
</td></tr>
<tr><td><code id="output_tc_+3A_seq_match">seq_match</code></td>
<td>
<p>Flag allows to choose if the peptide sequences should be matched between states. seq_match=FALSE signifies no sequence matching, seq_match=TRUE states that the sequences are matched between the sets.</p>
</td></tr>
<tr><td><code id="output_tc_+3A_csv">csv</code></td>
<td>
<p>Flag allowing saving the output as csv. With default csv=&quot;NA&quot;, data is not saved. If csv output is desided, provide output name.</p>
</td></tr>
<tr><td><code id="output_tc_+3A_percent">percent</code></td>
<td>
<p>Flag allowing to choose output as deteuration uptake (FALSE) or percent deuteration (TRUE). Default deuteration uptake.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with reorganized data where in columns is the deuteration uptake for Protein States.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tc(filepath=file_nm) ###all default parameters used

# all possible flags listed &amp; percent deuteration output,
#with sequences matching for protein states.

a&lt;-output_tc(filepath=file_nm, replicates=3, states=c("bound", "Unbound"),
times=c("3.00s", "72000.00s"), seq_match=TRUE, csv="NA", percent=TRUE)
</code></pre>

<hr>
<h2 id='output_tp'>Prepares output for HDX-MS for the deuteration uptake or percent deuteration for the time points.</h2><span id='topic+output_tp'></span>

<h3>Description</h3>

<p>Returns a data frame organized for additional analysis.
In columns are deuteration uptake or percent deuteration data for the given protein states.
Function allows for writing csv with data, matching sequences of peptide.
Protein.States, Deut.times, or number of replicates can be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>output_tp(
  filepath,
  replicates,
  states,
  times,
  seq_match = FALSE,
  csv = "NA",
  percent = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="output_tp_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
<tr><td><code id="output_tp_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates to be used in analysis. The function takes number of replicates up to specified number. If no argument provided number maximal common number of replicates it used.</p>
</td></tr>
<tr><td><code id="output_tp_+3A_states">states</code></td>
<td>
<p>function allows to choose what states should be used for analysis. Default all states are used.</p>
</td></tr>
<tr><td><code id="output_tp_+3A_times">times</code></td>
<td>
<p>lists the deuteration times to be used in analysis. Default all states used.</p>
</td></tr>
<tr><td><code id="output_tp_+3A_seq_match">seq_match</code></td>
<td>
<p>Flag allows to choose if the peptide sequences should be matched between states. seq_match=FALSE signifies no sequence matching, seq_match=T states that the sequences are matched between the sets.</p>
</td></tr>
<tr><td><code id="output_tp_+3A_csv">csv</code></td>
<td>
<p>Flag allowing saving the output as csv. With default csv=&quot;NA&quot;, data is not saved. If csv output is desided, provide output name.</p>
</td></tr>
<tr><td><code id="output_tp_+3A_percent">percent</code></td>
<td>
<p>Flag allowing to choose output as deteuration uptake (FALSE) or percent deuteration (TRUE). Default deuteration uptake.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with reorganized data where in columns is the deuteration uptake for Protein States.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(filepath=file_nm) ###all default parameters used


# all possible flags listed &amp; percent deuteration output,
# with sequences matching for protein states.

a&lt;-output_tp(filepath=file_nm, replicates=3, states=c("bound", "Unbound"),
times=c("3.00s", "72000.00s"), seq_match=TRUE, csv="NA", percent=TRUE)
</code></pre>

<hr>
<h2 id='output_UD'>Prepares output for HDX-MS  Undeuterated sample data.</h2><span id='topic+output_UD'></span>

<h3>Description</h3>

<p>Returns a data frame for Full deuteration set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>output_UD(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="output_UD_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with reorganized data where in columns is uptake data for Protein States.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_UD(file_nm)
</code></pre>

<hr>
<h2 id='output_UD_proc'>Prepares output for HDX-MS Undeuterated data for procent deuteration.</h2><span id='topic+output_UD_proc'></span>

<h3>Description</h3>

<p>Returns a data frame for Undeuterated control set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>output_UD_proc(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="output_UD_proc_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with reorganized data where in columns is procent deuteration for Protein States.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_UD_proc(file_nm)
</code></pre>

<hr>
<h2 id='pallette_legend'>Color scheme using heatmap. Legend Extracts names from data.frame</h2><span id='topic+pallette_legend'></span>

<h3>Description</h3>

<p>Returns names for legend for the heatmaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pallette_legend(col_pallette)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pallette_legend_+3A_col_pallette">col_pallette</code></td>
<td>
<p>pallette to be used in the heat map</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for the heatmap
</p>

<hr>
<h2 id='pallette_ll'>Color scheme using heatmap. Legend extracts names from data frame</h2><span id='topic+pallette_ll'></span>

<h3>Description</h3>

<p>Returns names for legend for the heatmaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pallette_ll(pallette, lab)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pallette_ll_+3A_pallette">pallette</code></td>
<td>
<p>pallette to be used in the heat map</p>
</td></tr>
<tr><td><code id="pallette_ll_+3A_lab">lab</code></td>
<td>
<p>labels to be used in pallette</p>
</td></tr>
</table>


<h3>Value</h3>

<p>legend for the heatmap
</p>

<hr>
<h2 id='peptide_pv_tp'>Preparatory function for significant peptide plots</h2><span id='topic+peptide_pv_tp'></span>

<h3>Description</h3>

<p>Returns plot where significant peptides are colored in blue-red scheme.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peptide_pv_tp(
  df,
  pv,
  sd,
  nb_row,
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="peptide_pv_tp_+3A_df">df</code></td>
<td>
<p>average data frame. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_+3A_pv">pv</code></td>
<td>
<p>pvalues dataframes calculated using pv_timepoint() function</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_+3A_sd">sd</code></td>
<td>
<p>standard deviation data.frame generated using sd_timepoint function</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_+3A_nb_row">nb_row</code></td>
<td>
<p>number of peptides in each row. Plotting parameter.</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot with peptides which are significantly different between sets.
</p>

<hr>
<h2 id='peptide_pv_tp_proc'>Preparatory function for showing peptides with significant differences between sets.</h2><span id='topic+peptide_pv_tp_proc'></span>

<h3>Description</h3>

<p>Returns plot where significantly different peptides are colored in blue-red scheme.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peptide_pv_tp_proc(
  df,
  dfup,
  pv,
  sd,
  nb_row = 100,
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="peptide_pv_tp_proc_+3A_df">df</code></td>
<td>
<p>average data frame for procent deuteration. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_proc_+3A_dfup">dfup</code></td>
<td>
<p>average data frame for deuteration uptake. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_proc_+3A_pv">pv</code></td>
<td>
<p>pvalues dataframes calculated using pv_timepoint() function</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_proc_+3A_sd">sd</code></td>
<td>
<p>standard deviation data.frame generated using sd_timepoint function</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_proc_+3A_nb_row">nb_row</code></td>
<td>
<p>number of peptides in each row. Plotting parameter.</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_proc_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_proc_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="peptide_pv_tp_proc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot with peptides which are significantly different between sets.
</p>

<hr>
<h2 id='pl_gen_ch2'>Prepares the plot window for the woods functions</h2><span id='topic+pl_gen_ch2'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pl_gen_ch2(df, ddlab = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pl_gen_ch2_+3A_df">df</code></td>
<td>
<p>dataframe</p>
</td></tr>
<tr><td><code id="pl_gen_ch2_+3A_ddlab">ddlab</code></td>
<td>
<p>label</p>
</td></tr>
<tr><td><code id="pl_gen_ch2_+3A_...">...</code></td>
<td>
<p>other</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot window
</p>

<hr>
<h2 id='pl_gen_uptake'>Prepares the plot window for the woods functions</h2><span id='topic+pl_gen_uptake'></span>

<h3>Description</h3>

<p>Internal function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pl_gen_uptake(df, timepoints, ddlab = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pl_gen_uptake_+3A_df">df</code></td>
<td>
<p>dataframe</p>
</td></tr>
<tr><td><code id="pl_gen_uptake_+3A_timepoints">timepoints</code></td>
<td>
<p>deuteration times used</p>
</td></tr>
<tr><td><code id="pl_gen_uptake_+3A_ddlab">ddlab</code></td>
<td>
<p>label</p>
</td></tr>
<tr><td><code id="pl_gen_uptake_+3A_...">...</code></td>
<td>
<p>other</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot window
</p>

<hr>
<h2 id='plot_heat_map_max_uptake_tp'>Plots heat maps for maximum uptake per residue.</h2><span id='topic+plot_heat_map_max_uptake_tp'></span>

<h3>Description</h3>

<p>Returns heat map with maximum uptake per residue.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heat_map_max_uptake_tp(
  df,
  replicates = 3,
  mar_x = 3.5,
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heat_map_max_uptake_tp_+3A_df">df</code></td>
<td>
<p>average data frame. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="plot_heat_map_max_uptake_tp_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
<tr><td><code id="plot_heat_map_max_uptake_tp_+3A_mar_x">mar_x</code></td>
<td>
<p>margin x width. Default=3.5</p>
</td></tr>
<tr><td><code id="plot_heat_map_max_uptake_tp_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="plot_heat_map_max_uptake_tp_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
</table>


<h3>Value</h3>

<p>heat map for maximum uptake per residue
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
plot_heat_map_max_uptake_tp(df=a, replicates=3, pv_cutoff=0.01,
ranges=c(-Inf,-40, -30,-20,-10, 0,10, 20,30,40, Inf) )
plot_heat_map_max_uptake_tp(df=a)
</code></pre>

<hr>
<h2 id='plot_heat_map_max_uptake_tp_proc'>Plots heat maps for maximum procent deuteration per residue.</h2><span id='topic+plot_heat_map_max_uptake_tp_proc'></span>

<h3>Description</h3>

<p>Returns heat map with maximum precent_deuteration per residue.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heat_map_max_uptake_tp_proc(
  input_proc,
  input_up,
  mar_x = 3.5,
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heat_map_max_uptake_tp_proc_+3A_input_proc">input_proc</code></td>
<td>
<p>Dataframe with organized procent deuteration data. Input generated using output_tp_proc() function.</p>
</td></tr>
<tr><td><code id="plot_heat_map_max_uptake_tp_proc_+3A_input_up">input_up</code></td>
<td>
<p>Dataframe with organized deuteration uptake. Input generated using output_tp() function.</p>
</td></tr>
<tr><td><code id="plot_heat_map_max_uptake_tp_proc_+3A_mar_x">mar_x</code></td>
<td>
<p>margin x width. Default=3.5</p>
</td></tr>
<tr><td><code id="plot_heat_map_max_uptake_tp_proc_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="plot_heat_map_max_uptake_tp_proc_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="plot_heat_map_max_uptake_tp_proc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>heat map for average uptake per residue for significant peptides.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a_up&lt;- output_tp(file_nm)
a_proc&lt;- output_tp(file_nm, percent=TRUE)
plot_heat_map_max_uptake_tp_proc(input_proc=a_proc, input_up=a_up, replicates=3, pv_cutoff=0.01,
ranges=c(-Inf,-40, -30,-20,-10, 0,10, 20,30,40, Inf) )
plot_heat_map_max_uptake_tp_proc(input_proc=a_proc, input_up=a_up)
</code></pre>

<hr>
<h2 id='plot_heat_map_tc'>Plots heat maps for time courses.</h2><span id='topic+plot_heat_map_tc'></span>

<h3>Description</h3>

<p>Returns heat map on timecourses with raw data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heat_map_tc(
  df,
  replicates = 3,
  mar_x = 3.5,
  ranges = c(-Inf, seq(0, 100, by = 10), Inf)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heat_map_tc_+3A_df">df</code></td>
<td>
<p>output from function output_tcourse</p>
</td></tr>
<tr><td><code id="plot_heat_map_tc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
<tr><td><code id="plot_heat_map_tc_+3A_mar_x">mar_x</code></td>
<td>
<p>margin x width. Default=3.5</p>
</td></tr>
<tr><td><code id="plot_heat_map_tc_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(seq(0, 100, by=10), Inf)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>heat map for time courses
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tc(file_nm)
plot_heat_map_tc(df=a, replicates=3, ranges=c(seq(0, 100, by=5), Inf))
plot_heat_map_tc(df=a)
</code></pre>

<hr>
<h2 id='plot_heat_map_tp'>Plots heat maps for significant peptides.</h2><span id='topic+plot_heat_map_tp'></span>

<h3>Description</h3>

<p>Returns heat map with average values for significant uptake per residue.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heat_map_tp(
  df,
  mar_x = 3.5,
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heat_map_tp_+3A_df">df</code></td>
<td>
<p>average data frame. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="plot_heat_map_tp_+3A_mar_x">mar_x</code></td>
<td>
<p>margin x width. Default=3.5</p>
</td></tr>
<tr><td><code id="plot_heat_map_tp_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="plot_heat_map_tp_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="plot_heat_map_tp_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>heat map for average uptake per residue for significant peptides.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
plot_heat_map_tp(df=a, replicates=3, pv_cutoff=0.01,
ranges=c(-Inf,-40, -30,-20,-10, 0,10, 20,30,40, Inf) )
plot_heat_map_tp(df=a)
</code></pre>

<hr>
<h2 id='plot_heat_map_tp_proc'>Plots heat maps for significant peptides.</h2><span id='topic+plot_heat_map_tp_proc'></span>

<h3>Description</h3>

<p>Returns heat map with average values for significant uptake per residue.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heat_map_tp_proc(
  input_proc,
  input_up,
  mar_x = 3.5,
  ranges = c(-Inf, -3, -2, -1, 0, 1, 2, 3, Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heat_map_tp_proc_+3A_input_proc">input_proc</code></td>
<td>
<p>Dataframe with organized procent deuteration data. Input generated using output_tp_proc() function.</p>
</td></tr>
<tr><td><code id="plot_heat_map_tp_proc_+3A_input_up">input_up</code></td>
<td>
<p>Dataframe with organized deuteration uptake. Input generated using output_tp() function.</p>
</td></tr>
<tr><td><code id="plot_heat_map_tp_proc_+3A_mar_x">mar_x</code></td>
<td>
<p>margin x width. Default=3.5</p>
</td></tr>
<tr><td><code id="plot_heat_map_tp_proc_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="plot_heat_map_tp_proc_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="plot_heat_map_tp_proc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>heat map for average uptake per residue for significant peptides.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a_up&lt;- output_tp(file_nm)
a_proc&lt;- output_tp(file_nm, percent=TRUE)
plot_heat_map_tp_proc(input_proc=a_proc, input_up=a_up, replicates=3, pv_cutoff=0.01,
ranges=c(-Inf,-40, -30,-20,-10, 0,10, 20,30,40, Inf) )
plot_heat_map_tp_proc(input_proc=a_proc, input_up=a_up)
</code></pre>

<hr>
<h2 id='plot_peptide_sig_tp'>Significant peptide plots.</h2><span id='topic+plot_peptide_sig_tp'></span>

<h3>Description</h3>

<p>Returns plot where significant peptides are colored in blue-red scheme.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_peptide_sig_tp(
  df1,
  replicates = 3,
  nb_pep_row = 100,
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_peptide_sig_tp_+3A_df1">df1</code></td>
<td>
<p>average data frame. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="plot_peptide_sig_tp_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
<tr><td><code id="plot_peptide_sig_tp_+3A_nb_pep_row">nb_pep_row</code></td>
<td>
<p>number of peptides in each row. Plotting parameter. Default set to 100.</p>
</td></tr>
<tr><td><code id="plot_peptide_sig_tp_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="plot_peptide_sig_tp_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot with peptides which are significantly different between sets.
</p>

<hr>
<h2 id='plot_peptide_sig_tp_proc'>Draws peptides with significant difefrences between sets.</h2><span id='topic+plot_peptide_sig_tp_proc'></span>

<h3>Description</h3>

<p>Returns plot where significant peptides are colored in blue-red scheme.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_peptide_sig_tp_proc(
  input_proc,
  input_up,
  nb_pep_row = 100,
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_peptide_sig_tp_proc_+3A_input_proc">input_proc</code></td>
<td>
<p>Dataframe with organized procent deuteration data. Input generated using output_tp_proc() function.</p>
</td></tr>
<tr><td><code id="plot_peptide_sig_tp_proc_+3A_input_up">input_up</code></td>
<td>
<p>Dataframe with organized deuteration uptake. Input generated using output_tp() function.</p>
</td></tr>
<tr><td><code id="plot_peptide_sig_tp_proc_+3A_nb_pep_row">nb_pep_row</code></td>
<td>
<p>number of peptides in each row. Plotting parameter. Default set to 100.</p>
</td></tr>
<tr><td><code id="plot_peptide_sig_tp_proc_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="plot_peptide_sig_tp_proc_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="plot_peptide_sig_tp_proc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot with peptides which are significantly different between sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a_up&lt;- output_tp(file_nm)
a_proc&lt;- output_tp(file_nm, percent=TRUE)
plot_peptide_sig_tp_proc(input_proc=a_proc, input_up=a_up, replicates=3, pv_cutoff=0.01,
ranges=c(-Inf,-40, -30,-20,-10, 0,10, 20,30,40, Inf), nb_pep_row=40 )
</code></pre>

<hr>
<h2 id='plots_av_tcourse'>Generates average deuteration plot for the time-course.</h2><span id='topic+plots_av_tcourse'></span>

<h3>Description</h3>

<p>Returns plots with average deuteration at each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plots_av_tcourse(df, replicates = 3, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plots_av_tcourse_+3A_df">df</code></td>
<td>
<p>output from functions output_tcourse or output_tcourse_proc.</p>
</td></tr>
<tr><td><code id="plots_av_tcourse_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in set as default set to 3.</p>
</td></tr>
<tr><td><code id="plots_av_tcourse_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from RColorBrewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>average deuteration plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tc(file_nm)
plots_av_tcourse(df=a, replicates=3, cola=c(1:4))
plots_av_tcourse(df=a)
</code></pre>

<hr>
<h2 id='plots_av_tp'>Returns average deuteration plot for timepoints in the data frame</h2><span id='topic+plots_av_tp'></span>

<h3>Description</h3>

<p>Returns plots with average deuteration at each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plots_av_tp(df, replicates = 3, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plots_av_tp_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="plots_av_tp_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in set as default set to 3.</p>
</td></tr>
<tr><td><code id="plots_av_tp_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>average deuteration plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
plots_av_tp(df=a, replicates=3, cola=c(1:4))
plots_av_tp(df=a)
</code></pre>

<hr>
<h2 id='plots_av_tp_proc'>Returns average procent deuteration plot for time points</h2><span id='topic+plots_av_tp_proc'></span>

<h3>Description</h3>

<p>Returns plots with average procent deuteration at each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plots_av_tp_proc(df, replicates = 3, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plots_av_tp_proc_+3A_df">df</code></td>
<td>
<p>output from functions output_tp_proc.</p>
</td></tr>
<tr><td><code id="plots_av_tp_proc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in set as default set to 3.</p>
</td></tr>
<tr><td><code id="plots_av_tp_proc_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from RColorBrewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>average deuteration plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm, percent=TRUE)
plots_av_tp_proc(df=a, replicates=3, cola=c(1:4))
plots_av_tp_proc(df=a)
</code></pre>

<hr>
<h2 id='plots_diff_tp'>Returns difference in average plot for timepoints in the data frame</h2><span id='topic+plots_diff_tp'></span>

<h3>Description</h3>

<p>Returns plots with difference in avarage for each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plots_diff_tp(df, replicates = 3, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plots_diff_tp_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="plots_diff_tp_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in set as default set to 3.</p>
</td></tr>
<tr><td><code id="plots_diff_tp_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots of difference of averages
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
plots_diff_tp(df=a, replicates=3, cola=c(1:4))
plots_diff_tp(df=a)
</code></pre>

<hr>
<h2 id='plots_diff_tp_proc'>Returns difference in average procent deuteration plot for timepoints in the data frame</h2><span id='topic+plots_diff_tp_proc'></span>

<h3>Description</h3>

<p>Returns plots with difference in procent deuteration for each peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plots_diff_tp_proc(df, replicates = 3, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plots_diff_tp_proc_+3A_df">df</code></td>
<td>
<p>output from functions output_tp_proc.</p>
</td></tr>
<tr><td><code id="plots_diff_tp_proc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in set as default set to 3.</p>
</td></tr>
<tr><td><code id="plots_diff_tp_proc_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots of difference of average procent deuteration
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm, percent=TRUE)
plots_diff_tp_proc(df=a, replicates=3, cola=c(1:4))
plots_diff_tp_proc(df=a)
</code></pre>

<hr>
<h2 id='plots_vol_tp'>Returns volcano plots for timepoints in the data frame</h2><span id='topic+plots_vol_tp'></span>

<h3>Description</h3>

<p>Returns volcano plots for each peptide. Critical interval is calculated according to #' Reliable Identification of Significant Differences in Differential Hydrogen Exchange-Mass Spectrometry Measurements Using a Hybrid Significance Testing Approach
Tyler S. Hageman and David D. Weis
Analytical Chemistry 2019 91 (13), 8008-8016 DOI: 10.1021/acs.analchem.9b01325
calculations for alpha 0.99
pvalues calculated using Welch t-test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plots_vol_tp(df, replicates = 3, pv_cutoff = 0.01, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plots_vol_tp_+3A_df">df</code></td>
<td>
<p>output from functions output_tp</p>
</td></tr>
<tr><td><code id="plots_vol_tp_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in set as default set to 3.</p>
</td></tr>
<tr><td><code id="plots_vol_tp_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="plots_vol_tp_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>volcano plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
plots_vol_tp(df=a, replicates=3, cola=c(1:4), pv_cutoff=0.01 )
plots_vol_tp(df=a, pv_cutoff=0.05)
</code></pre>

<hr>
<h2 id='ppar'>Preparation of figure window.</h2><span id='topic+ppar'></span>

<h3>Description</h3>

<p>Prepares a plotting window with specified margins with specific number of figure row and columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppar(mfrow2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ppar_+3A_mfrow2">mfrow2</code></td>
<td>
<p>mfrow: number of Multiple Figures (use ROW-wise).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>modified par function with adjusted parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ppar(c(2,1))
</code></pre>

<hr>
<h2 id='ppar_bottom_legend'>Preparation of figure window with area for figure at the bottom.</h2><span id='topic+ppar_bottom_legend'></span>

<h3>Description</h3>

<p>Prepares a plotting window with specified margins with specific number of figure row and columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppar_bottom_legend(mfrow2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ppar_bottom_legend_+3A_mfrow2">mfrow2</code></td>
<td>
<p>mfrow: number of Multiple Figures (use ROW-wise).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>modified par function with adjusted parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ppar_bottom_legend(c(2,3))
</code></pre>

<hr>
<h2 id='ppar_wider'>Preparation of figure window with more area on west side of plot.</h2><span id='topic+ppar_wider'></span>

<h3>Description</h3>

<p>Prepares a plotting window with specified margins with specific number of figure row and columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppar_wider(mfrow2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ppar_wider_+3A_mfrow2">mfrow2</code></td>
<td>
<p>mfrow: number of Multiple Figures (use ROW-wise).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>default plotting window
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ppar_wider(c(2,1))
</code></pre>

<hr>
<h2 id='pparLM'>Preparation of figure window. small margins</h2><span id='topic+pparLM'></span>

<h3>Description</h3>

<p>Prepares a plotting window with specified margins with specific number of figure row and columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pparLM(mfrow2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pparLM_+3A_mfrow2">mfrow2</code></td>
<td>
<p>mfrow: number of Multiple Figures (use ROW-wise).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>modified par function with adjusted parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pparLM(c(2,1))
</code></pre>

<hr>
<h2 id='prep_timecourse_plot_ave'>Prepares function for plotting averages in timecourse</h2><span id='topic+prep_timecourse_plot_ave'></span>

<h3>Description</h3>

<p>Preparatory function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep_timecourse_plot_ave(control_df, variant_df, replicates = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep_timecourse_plot_ave_+3A_control_df">control_df</code></td>
<td>
<p>dataframe of control</p>
</td></tr>
<tr><td><code id="prep_timecourse_plot_ave_+3A_variant_df">variant_df</code></td>
<td>
<p>dataframe for variant</p>
</td></tr>
<tr><td><code id="prep_timecourse_plot_ave_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframes with matched peptides in time course
</p>

<hr>
<h2 id='prep_timecourse_plot_sd'>Prepares function for Critical interval for timecourses</h2><span id='topic+prep_timecourse_plot_sd'></span>

<h3>Description</h3>

<p>Preparatory function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep_timecourse_plot_sd(
  control_df_up,
  variant_df_up,
  replicates = 3,
  pv_cutoff = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep_timecourse_plot_sd_+3A_control_df_up">control_df_up</code></td>
<td>
<p>dataframe of control</p>
</td></tr>
<tr><td><code id="prep_timecourse_plot_sd_+3A_variant_df_up">variant_df_up</code></td>
<td>
<p>dataframe for variant</p>
</td></tr>
<tr><td><code id="prep_timecourse_plot_sd_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates. Default set to 3.</p>
</td></tr>
<tr><td><code id="prep_timecourse_plot_sd_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>cut off of pvalue used in calculation of critical interval. Default set to 0.01</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Critial interval for all sets
</p>

<hr>
<h2 id='pv_timecourse'>pvalue calculation between two sets of the data at certain timepoint</h2><span id='topic+pv_timecourse'></span>

<h3>Description</h3>

<p>Preparatory function for calculation of pvalue between sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pv_timecourse(df_c, df_v, replicates = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pv_timecourse_+3A_df_c">df_c</code></td>
<td>
<p>dataframe of control</p>
</td></tr>
<tr><td><code id="pv_timecourse_+3A_df_v">df_v</code></td>
<td>
<p>dataframe for variant</p>
</td></tr>
<tr><td><code id="pv_timecourse_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pvalue comparisons between two sets.
</p>

<hr>
<h2 id='pv_timepoint'>Calculation of pvalue between first protein state and any other state from all_states file</h2><span id='topic+pv_timepoint'></span>

<h3>Description</h3>

<p>Compares means of sets of uptake data and return dataframe with pvalues.
Welch t.test is used for analysis.
Sets are compared to the first state in the input file.
If other order of the sets is required use
Default for the number of replicates is 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pv_timepoint(df, replicates = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pv_timepoint_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="pv_timepoint_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates used. Default is set to replicates=3</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame with p-values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
pv&lt;-pv_timepoint(df=a) ##if number of replicates is equal 3
# pv1&lt;-pv_timepoint(df=a, replicates=4) ##if number of replicates is equal 4
#b&lt;-output_tp_states(file_nm, states=c("State4", "State2", "State3" ))
#pv_states&lt;-pv_timepoint(df=b) ### here means of State4, will be compared to State2 and State4
</code></pre>

<hr>
<h2 id='pymol_script_average_residue'>Writes a text files with pymol scripts to list significant residues.</h2><span id='topic+pymol_script_average_residue'></span>

<h3>Description</h3>

<p>Function write a script that can be used in pymol to color structure.
Number of colors and corresponding to them ranges can be defined by user.
Residues are being colored by average uptake values from the significant peptides per residues.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pymol_script_average_residue(
  df,
  path = "",
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pymol_script_average_residue_+3A_df">df</code></td>
<td>
<p>output from functions output_tp</p>
</td></tr>
<tr><td><code id="pymol_script_average_residue_+3A_path">path</code></td>
<td>
<p>output folder location</p>
</td></tr>
<tr><td><code id="pymol_script_average_residue_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="pymol_script_average_residue_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="pymol_script_average_residue_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pymol script with residues colored based on average of uptake per residue.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
pymol_script_average_residue(df=a, replicates=3, pv_cutoff=0.01,
ranges=c(-Inf,-40, -30,-20,-10, 0,10, 20,30,40, Inf), path=tempdir() )
pymol_script_average_residue(df=a,  path=tempdir())

</code></pre>

<hr>
<h2 id='pymol_script_significant_peptide'>Writes a text files with pymol scripts to list significant peptides</h2><span id='topic+pymol_script_significant_peptide'></span>

<h3>Description</h3>

<p>Function write a script that can be used in pymol to color structure.
Number of colors and corresponding to them ranges can be defined by user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pymol_script_significant_peptide(
  df,
  path = "",
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3,
  order.pep = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pymol_script_significant_peptide_+3A_df">df</code></td>
<td>
<p>output from functions output_tp</p>
</td></tr>
<tr><td><code id="pymol_script_significant_peptide_+3A_path">path</code></td>
<td>
<p>location where the scripts will be saved</p>
</td></tr>
<tr><td><code id="pymol_script_significant_peptide_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="pymol_script_significant_peptide_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="pymol_script_significant_peptide_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
<tr><td><code id="pymol_script_significant_peptide_+3A_order.pep">order.pep</code></td>
<td>
<p>flag allowing to either order peptide acccording to the peptide length (default), or to position in the protein sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pymol script with colors assigned per peptide
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
pymol_script_significant_peptide(df=a, replicates=3, path=tempdir(), pv_cutoff=0.01,
ranges=c(-Inf,-40, -30,-20,-10, 0,10, 20,30,40, Inf), order.pep=TRUE )
pymol_script_significant_peptide(df=a, path=tempdir())

</code></pre>

<hr>
<h2 id='pymol_script_significant_peptide_proc'>Writes a text files with pymol scripts to list significant peptides</h2><span id='topic+pymol_script_significant_peptide_proc'></span>

<h3>Description</h3>

<p>Function write a script that can be used in pymol to color structure.
Number of colors and corresponding to them ranges can be defined by user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pymol_script_significant_peptide_proc(
  input_proc,
  input_up,
  path = "",
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3,
  order.pep = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pymol_script_significant_peptide_proc_+3A_input_proc">input_proc</code></td>
<td>
<p>Dataframe with organized procent deuteration data. Input generated using output_tp(, percent=T) function.</p>
</td></tr>
<tr><td><code id="pymol_script_significant_peptide_proc_+3A_input_up">input_up</code></td>
<td>
<p>Dataframe with organized deuteration uptake. Input generated using output_tp() function.</p>
</td></tr>
<tr><td><code id="pymol_script_significant_peptide_proc_+3A_path">path</code></td>
<td>
<p>location where the Pymol scripts will be saved</p>
</td></tr>
<tr><td><code id="pymol_script_significant_peptide_proc_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="pymol_script_significant_peptide_proc_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="pymol_script_significant_peptide_proc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
<tr><td><code id="pymol_script_significant_peptide_proc_+3A_order.pep">order.pep</code></td>
<td>
<p>flag allowing to either order peptide acccording to the peptide length (default), or to position in the protein sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pymol script with colors assigned per peptide
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a_up&lt;- output_tp(file_nm)
a_proc&lt;- output_tp(file_nm, percent=TRUE)
pymol_script_significant_peptide_proc(input_proc=a_proc,
input_up=a_up,  path=tempdir(),replicates=3, pv_cutoff=0.01,
ranges=c(-Inf,-40, -30,-20,-10, 0,10, 20,30,40, Inf), order.pep=TRUE)

</code></pre>

<hr>
<h2 id='pymol_script_significant_residue'>Writes a text files with pymol scripts to list significant residues.</h2><span id='topic+pymol_script_significant_residue'></span>

<h3>Description</h3>

<p>Function write a script that can be used in pymol to color structure.
Number of colors and corresponding to them ranges can be defined by user.
Residues are being colored by maximum uptake from significant peptides per residues.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pymol_script_significant_residue(
  df,
  path = "",
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pymol_script_significant_residue_+3A_df">df</code></td>
<td>
<p>average data frame. Generated using ave_timepoint() function.</p>
</td></tr>
<tr><td><code id="pymol_script_significant_residue_+3A_path">path</code></td>
<td>
<p>location where the Pymol scripts will be saved</p>
</td></tr>
<tr><td><code id="pymol_script_significant_residue_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="pymol_script_significant_residue_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="pymol_script_significant_residue_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pymol script with colors assigned per residues by maximum uptake per residue
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
pymol_script_significant_residue(df=a, path=tempdir(), replicates=3, pv_cutoff=0.01,
ranges=c(-Inf,-40, -30,-20,-10, 0,10, 20,30,40, Inf) )
pymol_script_significant_residue(df=a, path=tempdir())

</code></pre>

<hr>
<h2 id='pymol_script_significant_residue_proc'>Writes a text files with pymol scripts to list significant residues.</h2><span id='topic+pymol_script_significant_residue_proc'></span>

<h3>Description</h3>

<p>Function write a script that can be used in pymol to color structure.
Number of colors and corresponding to them ranges can be defined by user.
Residues are colored by average procent_deuteration from the significant peptides per residues.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pymol_script_significant_residue_proc(
  input_up,
  input_proc,
  path = "",
  ranges = c(-Inf, seq(-30, 30, by = 10), Inf),
  pv_cutoff = 0.01,
  replicates = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pymol_script_significant_residue_proc_+3A_input_up">input_up</code></td>
<td>
<p>Dataframe with organized deuteration uptake. Input generated using output_tp() function.</p>
</td></tr>
<tr><td><code id="pymol_script_significant_residue_proc_+3A_input_proc">input_proc</code></td>
<td>
<p>Dataframe with organized procent deuteration data. Input generated using output_tp_proc() function.</p>
</td></tr>
<tr><td><code id="pymol_script_significant_residue_proc_+3A_path">path</code></td>
<td>
<p>location where the Pymol scripts will be saved</p>
</td></tr>
<tr><td><code id="pymol_script_significant_residue_proc_+3A_ranges">ranges</code></td>
<td>
<p>ranges for coloring scheme. Default set to c(-Inf, seq(-30, 30, by=10), Inf)</p>
</td></tr>
<tr><td><code id="pymol_script_significant_residue_proc_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="pymol_script_significant_residue_proc_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pymol script with residues colored based on average of procent deuteration per residue.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a_up&lt;- output_tp(file_nm)
a_proc&lt;- output_tp(file_nm, percent=TRUE)
pymol_script_significant_residue_proc(input_proc=a_proc,
input_up=a_up, path=tempdir(), replicates=3, pv_cutoff=0.01,
ranges=c(-Inf,-40, -30,-20,-10, 0,10, 20,30,40, Inf))

</code></pre>

<hr>
<h2 id='pymol_str'>Preparatory function writing pymol scripts</h2><span id='topic+pymol_str'></span>

<h3>Description</h3>

<p>Function rearrange vector to string by adding + sign between the numbers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pymol_str(ind1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pymol_str_+3A_ind1">ind1</code></td>
<td>
<p>vector of numbers (residues)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>string with + as a separator.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>res&lt;-c(1,5, 19, 100, 109)
pymol_str(res)
</code></pre>

<hr>
<h2 id='qpcr.cbind.na'>Hidden function from qpcR package, typical usage as qpcR:::cbind.na</h2><span id='topic+qpcr.cbind.na'></span>

<h3>Description</h3>

<p>Combine data of unequal row length avoiding repetition or errors by filling with NAs. In contrast to classical cbind, cbind.na can be used to combine data such as
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpcr.cbind.na(..., deparse.level = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qpcr.cbind.na_+3A_...">...</code></td>
<td>
<p>vectors</p>
</td></tr>
<tr><td><code id="qpcr.cbind.na_+3A_deparse.level">deparse.level</code></td>
<td>
<p>set to 1 as default</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with NA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>qpcr.cbind.na(1:10, 1:3)

</code></pre>

<hr>
<h2 id='ranges_function'>Gives ranges for the averages</h2><span id='topic+ranges_function'></span>

<h3>Description</h3>

<p>Function used as internal function to get ranges in the function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ranges_function(df_ave, values_df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ranges_function_+3A_df_ave">df_ave</code></td>
<td>
<p>average per residues</p>
</td></tr>
<tr><td><code id="ranges_function_+3A_values_df">values_df</code></td>
<td>
<p>data frame with values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ranges per set
</p>

<hr>
<h2 id='ranges_function_tc'>Gives ranges for the averages for time course analysis</h2><span id='topic+ranges_function_tc'></span>

<h3>Description</h3>

<p>Function used as internal function to get ranges in the function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ranges_function_tc(df_ave, values_df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ranges_function_tc_+3A_df_ave">df_ave</code></td>
<td>
<p>average per residues</p>
</td></tr>
<tr><td><code id="ranges_function_tc_+3A_values_df">values_df</code></td>
<td>
<p>data frame with values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ranges per set
</p>

<hr>
<h2 id='rbind_na'>bind non equal row</h2><span id='topic+rbind_na'></span>

<h3>Description</h3>

<p>kmezhoud/canceR: A Graphical User Interface for accessing and modeling the Cancer Genomics Data of MSKCC
https://rdrr.io/github/kmezhoud/canceR/src/R/rbind.na.R
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rbind_na(..., deparse.level = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rbind_na_+3A_...">...</code></td>
<td>
<p>(generalized) vectors or matrices.</p>
</td></tr>
<tr><td><code id="rbind_na_+3A_deparse.level">deparse.level</code></td>
<td>
<p>integer controlling the construction of labels in
the case of non-matrix-like arguments (for the default method):
deparse.level = 0 constructs no labels; the default,
deparse.level = 1 or 2 constructs labels from the argument names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with merged rows
</p>


<h3>Examples</h3>

<pre><code class='language-R'>row1 &lt;- c("a","b","c","d")
row2 &lt;- c("A", "B", "C")
row3 &lt;- rbind_na(row1, row2)
</code></pre>

<hr>
<h2 id='reset_par'>Reset plotting window parameters to default</h2><span id='topic+reset_par'></span>

<h3>Description</h3>

<p>function by Farid Cheraghi,
https://stackoverflow.com/questions/9292563/reset-the-graphical-parameters-back-to-default-values-without-use-of-dev-off
function resets plotting window parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reset_par()
</code></pre>


<h3>Value</h3>

<p>default plotting window parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'>reset_par()
</code></pre>

<hr>
<h2 id='robot_2states_indexes'>Returns a robot plot for selected peptides for 2 protein states.</h2><span id='topic+robot_2states_indexes'></span>

<h3>Description</h3>

<p>Modification of butterfly plot. x axis residues.
y axis % deuteration for one variant above the axis and for second peptide below the axis.
Peptides are compared between the sets for the significance change between sets.
If there is significant change beteween sets peptides are plotted for all timepoints.
Significanty different timepoints for the peptides are colored.
Peptides ranges are plotted as a line at corresponding % deuteration values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>robot_2states_indexes(
  thP,
  th,
  indexes,
  states,
  replicates = 3,
  pvalue = 0.01,
  ylim,
  xlim,
  CI_factor = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="robot_2states_indexes_+3A_thp">thP</code></td>
<td>
<p>output of output_tcourse_proc() function. Raw data for procent deuteration for time courses</p>
</td></tr>
<tr><td><code id="robot_2states_indexes_+3A_th">th</code></td>
<td>
<p>output of output_tcourse() function. Raw data for uptake deuteration for time courses</p>
</td></tr>
<tr><td><code id="robot_2states_indexes_+3A_indexes">indexes</code></td>
<td>
<p>indexes of peptides to be drawn.</p>
</td></tr>
<tr><td><code id="robot_2states_indexes_+3A_states">states</code></td>
<td>
<p>Need to choose only two protein states</p>
</td></tr>
<tr><td><code id="robot_2states_indexes_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
<tr><td><code id="robot_2states_indexes_+3A_pvalue">pvalue</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="robot_2states_indexes_+3A_ylim">ylim</code></td>
<td>
<p>y-axis range</p>
</td></tr>
<tr><td><code id="robot_2states_indexes_+3A_xlim">xlim</code></td>
<td>
<p>x-axis range. Set as default from max and minimum residues for the protein</p>
</td></tr>
<tr><td><code id="robot_2states_indexes_+3A_ci_factor">CI_factor</code></td>
<td>
<p>Multiplication factor for Critical Interval. Allows for more restrictive selection of Critial interval.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Robot maps for timecourses for 2 protein states and selected indexes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
tm_df&lt;-output_tc(filepath=file_nm)
tmP_df&lt;-output_tc(filepath=file_nm, percent=TRUE)
names_states&lt;- nm_states(file_nm) ### returns states names
ind1&lt;-robot_indexes(thP = tmP_df, th=tm_df, pvalue=0.001, CI_factor=3, states=names_states[1:2])
robot_2states_indexes(thP = tmP_df, th=tm_df,
 states=names_states[1:2],indexes =ind1, pvalue=0.001, CI_factor=3)
</code></pre>

<hr>
<h2 id='robot_indexes'>Returns indexes for peptides with significant difference between two sets</h2><span id='topic+robot_indexes'></span>

<h3>Description</h3>

<p>Function to help decide which peptides will be drawn on Robot plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>robot_indexes(thP, th, replicates = 3, pvalue = 0.01, states, CI_factor = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="robot_indexes_+3A_thp">thP</code></td>
<td>
<p>output of output_tcourse_proc() function. Raw data for procent deuteration for time courses</p>
</td></tr>
<tr><td><code id="robot_indexes_+3A_th">th</code></td>
<td>
<p>output of output_tcourse() function. Raw data for uptake deuteration for time courses</p>
</td></tr>
<tr><td><code id="robot_indexes_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
<tr><td><code id="robot_indexes_+3A_pvalue">pvalue</code></td>
<td>
<p>p-value cutoff. Default set up to 0.01</p>
</td></tr>
<tr><td><code id="robot_indexes_+3A_states">states</code></td>
<td>
<p>Protein states from the set. As default all states are chosen.</p>
</td></tr>
<tr><td><code id="robot_indexes_+3A_ci_factor">CI_factor</code></td>
<td>
<p>Multiplication factor for Critical Interval. Allows for more restrictive selection of Critial interval.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns indexes of significant peptides
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
tm_df&lt;-output_tc(filepath=file_nm)
tmP_df&lt;-output_tc(filepath=file_nm, percent=TRUE)

 # more restictive peptide selection
robot_indexes(thP = tmP_df, th=tm_df, pvalue=0.01, CI_factor=1.5)
</code></pre>

<hr>
<h2 id='robot_indexes_df'>Returns dataframe with peptides which exhibit significant difference between two sets</h2><span id='topic+robot_indexes_df'></span>

<h3>Description</h3>

<p>Function to help decide which peptides will be drawn on Robot plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>robot_indexes_df(thP, th, replicates = 3, pvalue = 0.01, states, CI_factor = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="robot_indexes_df_+3A_thp">thP</code></td>
<td>
<p>output of output_tcourse_proc() function. Raw data for procent deuteration for time courses</p>
</td></tr>
<tr><td><code id="robot_indexes_df_+3A_th">th</code></td>
<td>
<p>output of output_tcourse() function. Raw data for uptake deuteration for time courses</p>
</td></tr>
<tr><td><code id="robot_indexes_df_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
<tr><td><code id="robot_indexes_df_+3A_pvalue">pvalue</code></td>
<td>
<p>p-value cutoff. Default set up to 0.01</p>
</td></tr>
<tr><td><code id="robot_indexes_df_+3A_states">states</code></td>
<td>
<p>Protein states from the set. As default all states are chosen.</p>
</td></tr>
<tr><td><code id="robot_indexes_df_+3A_ci_factor">CI_factor</code></td>
<td>
<p>Multiplication factor for Critical Interval. Allows for more restrictive selection of Critial interval.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns dataframe listing peptides that are significantly different between sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
tm_df&lt;-output_tc(filepath=file_nm)
tmP_df&lt;-output_tc(filepath=file_nm, percent=TRUE)

# more restictive peptide selection
robot_indexes_df(thP = tmP_df, th=tm_df, pvalue=0.01, CI_factor=1.5)
</code></pre>

<hr>
<h2 id='robot_plot_All'>Returns a robot plot for comparisons of the timepoints samples</h2><span id='topic+robot_plot_All'></span>

<h3>Description</h3>

<p>Modification of butterfly plot. x axis residues.
y axis % deuteration for one variant above the axis and for second peptide below the axis.
Peptides are compared between the sets for the significance change between sets.
If there is significant change beteween sets peptides are plotted for all timepoints.
Significanty different timepoints for the peptides are colored.
Peptides ranges are plotted as a line at corresponding % deuteration values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>robot_plot_All(
  thP,
  th,
  replicates = 3,
  pv_cutoff = 0.01,
  states,
  CI_factor = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="robot_plot_All_+3A_thp">thP</code></td>
<td>
<p>output of output_tcourse_proc() function. Raw data for procent deuteration for time courses</p>
</td></tr>
<tr><td><code id="robot_plot_All_+3A_th">th</code></td>
<td>
<p>output of output_tcourse() function. Raw data for uptake deuteration for time courses</p>
</td></tr>
<tr><td><code id="robot_plot_All_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
<tr><td><code id="robot_plot_All_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="robot_plot_All_+3A_states">states</code></td>
<td>
<p>Protein states from the set. As default all states are chosen.</p>
</td></tr>
<tr><td><code id="robot_plot_All_+3A_ci_factor">CI_factor</code></td>
<td>
<p>Multiplication factor for Critical Interval. Allows for more restrictive selection of Critial interval.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Robot maps for timecourses
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
tm_df&lt;-output_tc(filepath=file_nm)
tmP_df&lt;-output_tc(filepath=file_nm, percent=TRUE)
robot_plot_All(thP = tmP_df, th=tm_df, pv_cutoff=0.001)

# more restrictive peptide selection
robot_plot_All(thP = tmP_df, th=tm_df, pv_cutoff=0.001, CI_factor=3)

</code></pre>

<hr>
<h2 id='sd_timecourse'>Returns standard deviation for uptake data for timecourses.</h2><span id='topic+sd_timecourse'></span>

<h3>Description</h3>

<p>Calculates standard deviation for timecourse data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sd_timecourse(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sd_timecourse_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the All_results input file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame with standard deviation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
sd_timecourse(filepath=file_nm)
</code></pre>

<hr>
<h2 id='sd_timecourse_proc'>Returns standard deviation for percent deuteration data for timecourses.</h2><span id='topic+sd_timecourse_proc'></span>

<h3>Description</h3>

<p>Calculates standard deviation for time course data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sd_timecourse_proc(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sd_timecourse_proc_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the All_results input file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame with standard deviation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
sd_timecourse(filepath=file_nm)
</code></pre>

<hr>
<h2 id='sd_timepoint'>Returns standard deviation for dataframe.</h2><span id='topic+sd_timepoint'></span>

<h3>Description</h3>

<p>Calculates standard deviation for the number of replicates in the function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sd_timepoint(df, replicates = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sd_timepoint_+3A_df">df</code></td>
<td>
<p>output from functions output_tp or output_tp_proc.</p>
</td></tr>
<tr><td><code id="sd_timepoint_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates used. Default is set to replicates=3</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame with standard deviation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
sd&lt;-sd_timepoint(df=a, replicates=3)
</code></pre>

<hr>
<h2 id='select_indices'>Allows for selecting some peptide from input data</h2><span id='topic+select_indices'></span>

<h3>Description</h3>

<p>Function allows for picking indices from the inputs based on:
peptide start or end residue, length, state or timepoint.
If parameters set to NA, condition is skipped.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_indices(df, start = NA, end = NA, length = NA, times = NA, states = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="select_indices_+3A_df">df</code></td>
<td>
<p>input file (output of output_tc or output_tp)</p>
</td></tr>
<tr><td><code id="select_indices_+3A_start">start</code></td>
<td>
<p>provide number for the staring residue, default NA</p>
</td></tr>
<tr><td><code id="select_indices_+3A_end">end</code></td>
<td>
<p>provide number for the end residue, default NA</p>
</td></tr>
<tr><td><code id="select_indices_+3A_length">length</code></td>
<td>
<p>provide max length of the peptide</p>
</td></tr>
<tr><td><code id="select_indices_+3A_times">times</code></td>
<td>
<p>timepoints, only for the output_tp functions</p>
</td></tr>
<tr><td><code id="select_indices_+3A_states">states</code></td>
<td>
<p>states, only for the output_tc functions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Row indices of the peptides that are fulfilling the conditions required.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tp(file_nm)
indb&lt;-select_indices(a,length=12, start=100, end=200)
smaller_df&lt;-a[indb,]
</code></pre>

<hr>
<h2 id='significant_peptide_uptake'>Function returns which peptides are significantly based of pv_cutoff and Critial interval</h2><span id='topic+significant_peptide_uptake'></span>

<h3>Description</h3>

<p>Returns data frame with significant peptides.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>significant_peptide_uptake(df_av, pv, sd, pv_cutoff = 0.01, replicates = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="significant_peptide_uptake_+3A_df_av">df_av</code></td>
<td>
<p>data.frame with averages created using ave_timepoint() function</p>
</td></tr>
<tr><td><code id="significant_peptide_uptake_+3A_pv">pv</code></td>
<td>
<p>data.frame with pvalues created using pv_timepoint() function</p>
</td></tr>
<tr><td><code id="significant_peptide_uptake_+3A_sd">sd</code></td>
<td>
<p>data.frame with standard deviations created using sd_timepoint() function</p>
</td></tr>
<tr><td><code id="significant_peptide_uptake_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>cuttoff for Critical interval. Default=0.01</p>
</td></tr>
<tr><td><code id="significant_peptide_uptake_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates as default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ranges per set
</p>

<hr>
<h2 id='summary_sd_CI'>Provides summary table with Critical interval and standard deviation within the set.</h2><span id='topic+summary_sd_CI'></span>

<h3>Description</h3>

<p>Returns summary data.
Function returns: Protein states, timepoints, number of replicates,  # peptides, % coveregae, average peptide length and redundancy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_sd_CI(filepath, replicates = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary_sd_CI_+3A_filepath">filepath</code></td>
<td>
<p>filepath to the input file. Input file is All_results table from HDX_Examiner, where all the fields are marked for export.</p>
</td></tr>
<tr><td><code id="summary_sd_CI_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates. Default set to 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns summary table.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- summary_sd_CI(file_nm, replicates=3)
</code></pre>

<hr>
<h2 id='uptake_plots'>Uptake plots</h2><span id='topic+uptake_plots'></span>

<h3>Description</h3>

<p>Uptake plots per peptide
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uptake_plots(
  input_data,
  timepoints,
  replicates = 3,
  cola = NA,
  seq_match = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uptake_plots_+3A_input_data">input_data</code></td>
<td>
<p>output from function output_tp(..., percent=T)</p>
</td></tr>
<tr><td><code id="uptake_plots_+3A_timepoints">timepoints</code></td>
<td>
<p>the labeling times</p>
</td></tr>
<tr><td><code id="uptake_plots_+3A_replicates">replicates</code></td>
<td>
<p>replicates</p>
</td></tr>
<tr><td><code id="uptake_plots_+3A_cola">cola</code></td>
<td>
<p>colors, default NA</p>
</td></tr>
<tr><td><code id="uptake_plots_+3A_seq_match">seq_match</code></td>
<td>
<p>Flag TRUE or FALSE, default TRUE, match sequence of the protein states</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Uptake plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tc(file_nm, percent=TRUE)
x=c(3,60, 1800, 72000)
uptake_plots(a, x)

</code></pre>

<hr>
<h2 id='verbose_timecourse_output'>Returns csv with averages from analysis for procent deuteration file, standard deviation for time courses.</h2><span id='topic+verbose_timecourse_output'></span>

<h3>Description</h3>

<p>Returns information from analysis and save it as csv file.
Sets are compared to the first state in the input file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verbose_timecourse_output(filepath, output_name, replicates = 3, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="verbose_timecourse_output_+3A_filepath">filepath</code></td>
<td>
<p>path to All.Data.csv input from HDX-Examiner.</p>
</td></tr>
<tr><td><code id="verbose_timecourse_output_+3A_output_name">output_name</code></td>
<td>
<p>name of the output in csv format.</p>
</td></tr>
<tr><td><code id="verbose_timecourse_output_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates used</p>
</td></tr>
<tr><td><code id="verbose_timecourse_output_+3A_...">...</code></td>
<td>
<p>other variables for output_tc</p>
</td></tr>
</table>


<h3>Value</h3>

<p>csv with analysis for procent deuteration: standard deviation, for all protein states for time courses.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
verbose_timecourse_output(file_nm,tempfile(), replicates=3)
names_states&lt;- nm_states(file_nm)
verbose_timecourse_output(file_nm, tempfile(), seq_match=TRUE, percent=TRUE,
states=names_states, replicates=3, times="3.00s")

</code></pre>

<hr>
<h2 id='verbose_timepoint_output'>Returns csv with averages from analysis for uptake file, standard deviation, p-values.</h2><span id='topic+verbose_timepoint_output'></span>

<h3>Description</h3>

<p>Returns information from analysis and save it as csv file.
Sets are compared to the first state in the input file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verbose_timepoint_output(filepath, output_name, replicates = 3, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="verbose_timepoint_output_+3A_filepath">filepath</code></td>
<td>
<p>path to All.Data.csv input from HDX-Examiner.</p>
</td></tr>
<tr><td><code id="verbose_timepoint_output_+3A_output_name">output_name</code></td>
<td>
<p>name of the output in csv format.</p>
</td></tr>
<tr><td><code id="verbose_timepoint_output_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates used</p>
</td></tr>
<tr><td><code id="verbose_timepoint_output_+3A_...">...</code></td>
<td>
<p>other variables for output_tp</p>
</td></tr>
</table>


<h3>Value</h3>

<p>csv with analysis for uptake file, standard deviation, p-values for all protein states.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
verbose_timepoint_output(file_nm, tempfile())
names_states&lt;- nm_states(file_nm)
verbose_timepoint_output(file_nm, tempfile(), seq_match=TRUE, percent=TRUE,
states=names_states, replicates=3, times="3.00s")

</code></pre>

<hr>
<h2 id='vol_tp'>Preparatory function for volcano plot</h2><span id='topic+vol_tp'></span>

<h3>Description</h3>

<p>Returns volcano plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vol_tp(df1, pv, CI, pv_cutoff = 0.01, cola)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vol_tp_+3A_df1">df1</code></td>
<td>
<p>differences in averages data.frame calculated using diff_ave function</p>
</td></tr>
<tr><td><code id="vol_tp_+3A_pv">pv</code></td>
<td>
<p>pvalues dataframes calculated using pv_timepoint function</p>
</td></tr>
<tr><td><code id="vol_tp_+3A_ci">CI</code></td>
<td>
<p>critical interval, here is multiple sets are using maximun CI is used.</p>
</td></tr>
<tr><td><code id="vol_tp_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="vol_tp_+3A_cola">cola</code></td>
<td>
<p>color pallette for different Protein States. As default Paired pallette from color.Brewer is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>volcano plots
</p>

<hr>
<h2 id='woods_CI_plot'>Returns a woods plot for comparisons of the timepoints samples</h2><span id='topic+woods_CI_plot'></span>

<h3>Description</h3>

<p>Modification of butterfly plot. x axis residues.
y axis % deuteration for
Peptides are compared between the sets for the significance change between sets.
If there is significant change beteween sets peptides are plotted for all timepoints.
Significanty different timepoints for the peptides are colored.
Peptides ranges are plotted as a line at corresponding % deuteration values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>woods_CI_plot(
  thP,
  th,
  replicates = 3,
  pv_cutoff = 0.01,
  states,
  CI_factor = 1,
  ylim = c(0, 120),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="woods_CI_plot_+3A_thp">thP</code></td>
<td>
<p>output of output_tcourse_proc() function. Raw data for procent deuteration for time courses</p>
</td></tr>
<tr><td><code id="woods_CI_plot_+3A_th">th</code></td>
<td>
<p>output of output_tcourse() function. Raw data for uptake deuteration for time courses</p>
</td></tr>
<tr><td><code id="woods_CI_plot_+3A_replicates">replicates</code></td>
<td>
<p>number of replicates in sample. Default set to 3.</p>
</td></tr>
<tr><td><code id="woods_CI_plot_+3A_pv_cutoff">pv_cutoff</code></td>
<td>
<p>p-value cutoff here set up to 0.01</p>
</td></tr>
<tr><td><code id="woods_CI_plot_+3A_states">states</code></td>
<td>
<p>Protein states from the set. As default all states are chosen.</p>
</td></tr>
<tr><td><code id="woods_CI_plot_+3A_ci_factor">CI_factor</code></td>
<td>
<p>Multiplication factor for Critical Interval. Allows for more restrictive selection of Critial interval.</p>
</td></tr>
<tr><td><code id="woods_CI_plot_+3A_ylim">ylim</code></td>
<td>
<p>y axis limit</p>
</td></tr>
<tr><td><code id="woods_CI_plot_+3A_...">...</code></td>
<td>
<p>other variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Woods plots with chosen statistically different peptides
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file_nm&lt;-system.file("extdata", "All_results_table.csv", package = "HDXBoxeR")
a&lt;- output_tc(file_nm)
b&lt;-output_tc(file_nm, percent=TRUE)
woods_CI_plot(thP=b, th=a, pv_cutoff = 0.001, CI_factor = 1, replicates=3)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
