<!DOCTYPE html><html><head><title>Help for package heritability</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {heritability}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BT_LW_H'>
<p>Bolting time and leaf width for the Arabidopsis hapmap population.</p></a></li>
<li><a href='#floweringTime'>
<p>Flowering time data taken from Atwell <em>et al.</em> (2010).</p></a></li>
<li><a href='#heritability-package'>
<p>Marker-Based Estimation of Heritability Using Individual Plant or Plot Data.</p></a></li>
<li><a href='#K_arabidopsis'>
<p>Marker-based relatedness matrices for 3 populations of Arabidopsis thaliana.</p></a></li>
<li><a href='#leafArea'>
<p><em>Arabidopsis</em> leaf area data for the hapmap and Swedish regmap population.</p></a></li>
<li><a href='#marker_h2'>
<p>Compute a marker-based estimate of heritability, given phenotypic observations</p>
at individual plant or plot level.</a></li>
<li><a href='#marker_h2_means'>
<p>Compute a marker-based estimate of heritability, given genotypic means.</p></a></li>
<li><a href='#means_floweringTime'>
<p>Flowering time from Atwell <em>et al.</em> (2010): accession means.</p></a></li>
<li><a href='#R_matrix'>
<p>Covariance matrix of the accession means for flowering time.</p></a></li>
<li><a href='#repeatability'><p>ANOVA-based estimates of repeatability</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Marker-Based Estimation of Heritability Using Individual Plant
or Plot Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-21</td>
</tr>
<tr>
<td>Author:</td>
<td>Willem Kruijer, with a contribution from Ian White (the internal function pin). Contains data collected by Padraic Flood and Rik Kooke.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Willem Kruijer &lt;willem.kruijer@wur.nl&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0), MASS (&ge; 3.1.20)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements marker-based estimation of heritability when observations on genetically identical replicates are available. These can be either observations on individual plants or plot-level data in a field trial. Heritability can then be estimated using a mixed model for the individual plant or plot data. For comparison, also mixed-model based estimation using genotypic means and estimation of repeatability with ANOVA are implemented. For illustration the package contains several datasets for the model species Arabidopsis thaliana.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-24 06:43:18 UTC; kruij025</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-24 07:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='BT_LW_H'>
Bolting time and leaf width for the Arabidopsis hapmap population.
</h2><span id='topic+BT_LW_H'></span>

<h3>Description</h3>

<p>Bolting time and leaf width for the Arabidopsis hapmap population
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(BT_LW_H)</code></pre>


<h3>Format</h3>

<p>A data frame with phenotypic observations on bolting time and leaf width:
</p>

<dl>
<dt><code>genotype</code></dt><dd><p>a factor, the levels being the accession or ecotype identifiers</p>
</dd>
<dt><code>BT</code></dt><dd><p>Bolting time, in number of days</p>
</dd>
<dt><code>LW</code></dt><dd><p>Leaf width</p>
</dd>
<dt><code>replicate</code></dt><dd><p>The replicate (or block) each plant is contained in (factor with levels <code>1</code> to <code>3</code>)</p>
</dd>
<dt><code>rep1</code></dt><dd><p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 1 and 0 otherwise</p>
</dd>
<dt><code>rep2</code></dt><dd><p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 2 and 0 otherwise</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Willem Kruijer &lt;willlem.kruijer@wur.nl&gt;;
experiments conducted by Rik Kooke &lt;rik.kooke@gmail.com&gt;
</p>


<h3>References</h3>


<ul>
<li><p> Kruijer, W. <em>et al.</em> (2015) Marker-based estimation of heritability in immortal
populations. <em>Genetics</em>, Vol. 199(2), p. 1-20.
</p>
</li></ul>



<h3>See Also</h3>

<p>For the corresponding genetic relatedness matrix, see <code><a href="#topic+K_hapmap">K_hapmap</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BT_LW_H)
</code></pre>

<hr>
<h2 id='floweringTime'>
Flowering time data taken from Atwell <em>et al.</em> (2010).
</h2><span id='topic+floweringTime'></span><span id='topic+LD'></span><span id='topic+LDV'></span>

<h3>Description</h3>

<p>Two data-frames containing individual plant data on flowering time under different conditions:
<code>LDV</code> (Flowering time under long days and vernalization) and <code>LD</code> (Flowering time under long days, without vernalization).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LD); data(LDV)</code></pre>


<h3>Format</h3>

<p>Data-frames with flowering time observations, genotype and design information:
</p>

<dl>
<dt><code>genotype</code></dt><dd><p>a factor, the levels being the accession or ecotype identifiers</p>
</dd>
<dt><code>LD</code></dt><dd><p>Flowering time under long days, in number of days</p>
</dd>
<dt><code>LDV</code></dt><dd><p>Flowering time under long days and vernalization, in number of days</p>
</dd>
<dt><code>replicate</code></dt><dd><p>The replicate (or block) each plant is contained in (factor with levels <code>1</code> to <code>6</code>)</p>
</dd>
<dt><code>rep1</code></dt><dd><p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 1 and 0 otherwise</p>
</dd>
<dt><code>rep2</code></dt><dd><p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 2 and 0 otherwise</p>
</dd>
<dt><code>rep3</code></dt><dd><p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 3 and 0 otherwise</p>
</dd>
<dt><code>rep4</code></dt><dd><p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 4 and 0 otherwise</p>
</dd>
<dt><code>rep5</code></dt><dd><p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 5 and 0 otherwise</p>
</dd>
</dl>



<h3>Details</h3>

<p>All plants that had not flowered by the end of the experiment were given a phenotypic value of 200.
Only accessions for which SNP-data are available are included here: 167 accessions in case of LD and 168 accessions in case of LDV.
</p>


<h3>References</h3>


<ul>
<li><p> Atwell, S., Y. S. Huang, B. J. Vilhjalmsson, G. Willems, M. Horton, <em>et al.</em> (2010)
Genome-wide association study of 107 phenotypes in Arabidopsis thaliana inbred lines. Nature
465: 627-631.
</p>
</li>
<li><p> Kruijer, W. <em>et al.</em> (2015) Marker-based estimation of heritability in immortal
populations. <em>Genetics</em>, Vol. 199(2), p. 1-20.
</p>
</li></ul>



<h3>See Also</h3>

<p>For the corresponding genetic relatedness matrix, see <code><a href="#topic+K_atwell">K_atwell</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LD); data(LDV)
</code></pre>

<hr>
<h2 id='heritability-package'>
Marker-Based Estimation of Heritability Using Individual Plant or Plot Data.
</h2><span id='topic+heritability-package'></span><span id='topic+heritability'></span>

<h3>Description</h3>

<p>The package implements marker-based estimation
of heritability when observations on genetically identical replicates are available.
These can be either observations on individual plants (e.g. in a growth chamber) or plot-level data in a field trial.
The function <code>marker_h2</code> estimates heritability using a mixed model for the
individual plant or plot data, as proposed in Kruijer <em>et al.</em>
For comparison, also mixed-model based estimation using genotypic means (<code>marker_h2_means</code>)
and estimation of repeatability with ANOVA (<code>repeatability</code>) are implemented.
For illustration the package contains several datasets for the model species <em>Arabidopsis thaliana</em>.
</p>


<h3>Author(s)</h3>

<p>Willem Kruijer
Maintainer: Willem Kruijer &lt;willlem.kruijer@wur.nl&gt;
</p>


<h3>References</h3>

<p>Kruijer, W. <em>et al.</em> (2015) Marker-based estimation of heritability in immortal
populations. <em>Genetics</em>, Vol. 199(2), p. 1-20.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A) marker-based estimation of heritability, given individual plant-data
# and a marker-based relatedness matrix:
data(LDV)
data(K_atwell)
# This may take up to 30 sec.
#out1 &lt;- marker_h2(data.vector=LDV$LDV,geno.vector=LDV$genotype,
#                  covariates=LDV[,4:8],K=K_atwell)
#
# B) marker-based estimation of heritability, given genotypic means
# and a marker-based relatedness matrix:
data(means_LDV)
data(R_matrix_LDV)
data(K_atwell)
out2 &lt;- marker_h2_means(data.vector=means_LDV$LDV,geno.vector=means_LDV$genotype,
                        K=K_atwell,Dm=R_matrix_LDV)
#
# C) estimation of repeatability using ANOVA:
data(LDV)
out3 &lt;- repeatability(data.vector=LDV$LDV,geno.vector= LDV$genotype,
                      covariates.frame=as.data.frame(LDV[,3]))
</code></pre>

<hr>
<h2 id='K_arabidopsis'>
Marker-based relatedness matrices for 3 populations of Arabidopsis thaliana.
</h2><span id='topic+K_atwell'></span><span id='topic+K_hapmap'></span><span id='topic+K_swedish'></span>

<h3>Description</h3>

<p>Marker-based relatedness matrices based on the SNP-data from Horton <em>et al.</em> (2012).
Three matrices are provided:
(a) <code>K_atwell</code>, for the 199 accessions studied in Atwell <em>et al.</em> (2010).
(b) <code>K_hapmap</code>, for a subset of 350 accessions taken from the Arabidopsis hapmap (Li <em>et al.</em>, 2010).
(c) <code>K_swedish</code>, for 304 Swedish accessions.
All of these are part of the world-wide regmap of 1307 accessions, described in Horton <em>et al.</em> (2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(K_atwell); data(K_hapmap); data(K_swedish)</code></pre>


<h3>Format</h3>

<p>Matrices whose row- and column names are the ecotype or seed-stock IDs of the accessions.
</p>


<h3>Details</h3>

<p>The matrices were computed using equation (2.2) in Astle and Balding (2009); see also
Goddard <em>et al.</em> (2009). The <code>heritability</code>-package does not contain functions
to construct relatedness matrices from genotypic data, but such functions can be found in many other software packages.
For example, GCTA (Yang <em>et al.</em>, 2011), LDAK (Speed <em>et al.</em>, 2012), Fast-LMM (Lippert, 2011) and GEMMA (Zhou and Stephens, 2012).
</p>


<h3>References</h3>


<ul>
<li><p> W. Astle and D.J. Balding (2009) Population Structure and Cryptic Relatedness in Genetic Association Studies.
<em>Statistical Science</em>, Vol. 24, No. 4, 451-471.
</p>
</li>
<li><p> Atwell, S., Y. S. Huang, B. J. Vilhjalmsson, G. Willems, M. Horton, <em>et al.</em> (2010)
Genome-wide association study of 107 phenotypes in Arabidopsis thaliana inbred lines. <em>Nature</em>
465: 627-631.
</p>
</li>
<li><p> Goddard, M.E., Naomi R. Wray, Klara Verbyla and Peter M. Visscher (2009) Estimating Effects and Making Predictions from Genome-Wide Marker Data.
<em>Statistical Science</em>, Vol. 24, No. 4, 517-529.
</p>
</li>
<li><p> Horton, M. W., A. M. Hancock, Y. S. Huang, C. Toomajian, S. Atwell, <em>et al.</em>
(2012) Genome-wide patterns of genetic variation in worldwide Arabidopsis thaliana accessions
from the RegMap panel. <em>Nature Genetics</em> 44: 212-216.
</p>
</li>
<li><p> Li, Y., Y. Huang, J. Bergelson, M. Nordborg, and J. O. Borevitz (2010) Association
mapping of local climate-sensitive quantitative trait loci in arabidopsis thaliana. <em>PNAS</em> vol. 107, number 49.
</p>
</li>
<li><p> Lippert, C., J. Listgarten, Y. Liu, C.M. Kadie, R.I. Davidson, <em>et al.</em> (2011) FaST
linear mixed models for genome-wide association studies. <em>Naure methods</em> 8: 833-835.
</p>
</li>
<li><p> Speed, D., G. Hemani, M. R. Johnson, and D.J. Balding (2012) Improved heritability
estimation from genome-wide snps. <em>the American journal of human genetics</em> 91: 1011-1021.
</p>
</li>
<li><p> Yang, J., S.H. Lee, M.E. Goddard, and P.M. Visscher (2011) GCTA: a tool for genomewide
complex trait analysis. <em>the American journal of human genetics</em> 88: 76-82.
</p>
</li>
<li><p> Zhou, X., and M. Stephens, (2012) Genome-wide efficient mixed-model analysis for association
studies. <em>Nature genetics</em> 44: 821-824.
</p>
</li></ul>



<h3>See Also</h3>

<p>For phenotypic data for the population described in Atwell <em>et al.</em> (2010), see <code><a href="#topic+LD">LD</a></code> and <code><a href="#topic+LDV">LDV</a></code>.
For phenotypic data for the hapmap, see <code><a href="#topic+BT_LW_H">BT_LW_H</a></code> and <code><a href="#topic+LA_H">LA_H</a></code>.
For phenotypic data for the Swedish regmap, see <code><a href="#topic+LA_S">LA_S</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(K_atwell)
data(K_hapmap)
data(K_swedish)
</code></pre>

<hr>
<h2 id='leafArea'>
<em>Arabidopsis</em> leaf area data for the hapmap and Swedish regmap population.
</h2><span id='topic+leafArea'></span><span id='topic+LA_H'></span><span id='topic+LA_S'></span>

<h3>Description</h3>

<p><em>Arabidopsis</em> leaf area data for the hapmap and Swedish regmap population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LA_H); data(LA_S)</code></pre>


<h3>Format</h3>

<p>Data frame with leaf area observations:
</p>

<dl>
<dt><code>genotype</code></dt><dd><p>a factor, the levels being the accession identifiers</p>
</dd>
<dt><code>LA13_H</code></dt><dd><p>Leaf area 13 days after sowing, in numbers of pixels (hapmap)</p>
</dd>
<dt><code>LA13_S</code></dt><dd><p>Leaf area 13 days after sowing, in numbers of pixels (Swedish regmap)</p>
</dd>
<dt><code>replicate</code></dt><dd><p>The replicate (or block) each plant is contained in (factor with levels <code>1</code> to <code>4</code>)</p>
</dd>
<dt><code>rep1</code></dt><dd><p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 1 and 0 otherwise</p>
</dd>
<dt><code>rep2</code></dt><dd><p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 2 and 0 otherwise</p>
</dd>
<dt><code>rep3</code></dt><dd><p>numeric encoding of the factor replicate: equals 1 if the plant is in replicate 3 and 0 otherwise</p>
</dd>
<dt><code>x</code></dt><dd><p>The within image x-coordinate of the plant. A factor with levels <code>1</code> <code>2</code> <code>3</code></p>
</dd>
<dt><code>y</code></dt><dd><p>The within image y-coordinate of the plant. A factor with levels <code>1</code> <code>2</code> <code>3</code> <code>4</code></p>
</dd>
<dt><code>x1</code></dt><dd><p>numeric encoding of the factor <code>x</code>: equals 1 if the plant is in position 1 and 0 otherwise</p>
</dd>
<dt><code>x2</code></dt><dd><p>numeric encoding of the factor <code>x</code>: equals 1 if the plant is in position 2 and 0 otherwise</p>
</dd>
<dt><code>y1</code></dt><dd><p>numeric encoding of the factor <code>y</code>: equals 1 if the plant is in position 1 and 0 otherwise</p>
</dd>
<dt><code>y2</code></dt><dd><p>numeric encoding of the factor <code>y</code>: equals 1 if the plant is in position 2 and 0 otherwise</p>
</dd>
<dt><code>y3</code></dt><dd><p>numeric encoding of the factor <code>y</code>: equals 1 if the plant is in position 3 and 0 otherwise</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Willem Kruijer &lt;willlem.kruijer@wur.nl&gt;;
experiments conducted by Padraic Flood &lt;flood@mpipz.mpg.de&gt;
</p>


<h3>References</h3>


<ul>
<li><p> Kruijer, W. <em>et al.</em> (2015) Marker-based estimation of heritability in immortal
populations. <em>Genetics</em>, Vol. 199(2), p. 1-20.
</p>
</li></ul>



<h3>See Also</h3>

<p>For the corresponding genetic relatedness matrices, see <code><a href="#topic+K_hapmap">K_hapmap</a></code> and <code><a href="#topic+K_swedish">K_swedish</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LA_H); data(LA_S)
</code></pre>

<hr>
<h2 id='marker_h2'>
Compute a marker-based estimate of heritability, given phenotypic observations
at individual plant or plot level. 
</h2><span id='topic+marker_h2'></span>

<h3>Description</h3>

<p>Given a genetic relatedness matrix and phenotypic observations at individual
plant or plot level, this function computes REML-estimates of the genetic and
residual variance and their standard errors, using the AI-algorithm (Gilmour <em>et al.</em> 1995).
Based on this, heritability estimates and confidence intervals are given
(the estimator <code class="reqn">h_r^2</code> in Kruijer <em>et al.</em>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>marker_h2(data.vector, geno.vector, covariates = NULL, K, alpha = 0.05,
          eps = 1e-06, max.iter = 100, fix.h2 = FALSE, h2 = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="marker_h2_+3A_data.vector">data.vector</code></td>
<td>

<p>A vector of phenotypic observations. Needs to be of type numeric. May contain
missing values.
</p>
</td></tr>
<tr><td><code id="marker_h2_+3A_geno.vector">geno.vector</code></td>
<td>

<p>A vector of genotype labels, either a factor or character. This vector should
correspond to <code>data.vector</code>, and hence needs to be of the same length.
</p>
</td></tr>
<tr><td><code id="marker_h2_+3A_covariates">covariates</code></td>
<td>

<p>A data-frame or matrix with optional covariates, the rows corresponding to
the phenotypic observations in <code>data.vector</code> and <code>geno.vector</code>.
May contain missing values. Factors are not allowed, and need to be encoded by
columns of type numeric or integer. The data-frame or matrix should not contain an intercept,
which is included by default.
</p>
</td></tr>
<tr><td><code id="marker_h2_+3A_k">K</code></td>
<td>

<p>A genetic relatedness or kinship matrix, typically marker-based.
Must have row- and column-names corresponding to the levels of <code>geno.vector</code>
</p>
</td></tr>
<tr><td><code id="marker_h2_+3A_alpha">alpha</code></td>
<td>

<p>Confidence level, for the 1-alpha confidence intervals.
</p>
</td></tr>
<tr><td><code id="marker_h2_+3A_eps">eps</code></td>
<td>

<p>Numerical precision, used as convergence criterion in the AI-algorithm.
</p>
</td></tr>
<tr><td><code id="marker_h2_+3A_max.iter">max.iter</code></td>
<td>

<p>Maximal number of iterations in the AI-algorithm.
</p>
</td></tr>
<tr><td><code id="marker_h2_+3A_fix.h2">fix.h2</code></td>
<td>

<p>Compute the log-likelihood and inverse AI-matrix for a fixed heritability value. Default is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="marker_h2_+3A_h2">h2</code></td>
<td>

<p>When <code>fix.h2</code> is <code>TRUE</code>, the value of the heritability. Must be of type numeric, between 0 and 1.
</p>
</td></tr>
</table>


<h3>Details</h3>



<ul>
<li><p> Given phenotypic observations <code class="reqn">Y_{ij}</code> for genotypes <code class="reqn">i=1,...,n</code> and replicates
<code class="reqn">j = 1,...,n_i</code>, the mixed model
<code class="reqn">Y_{ij} = \mu + G_i + E_{ij}</code>
is assumed. The vector of additive genetic effects <code class="reqn">(G_1,...,G_n)'</code> follows a
multivariate normal distribution with mean zero and covariance <code class="reqn">\sigma_A^2 K</code>,
where <code class="reqn">\sigma_A^2</code> is the additive genetic variance, and <code class="reqn">K</code> is a genetic relatedness matrix derived from a dense set of markers.
The errors <code class="reqn">E_{ij}</code> are independent and normally distributed with variance <code class="reqn">\sigma_E^2</code>.
Under certain assumptions (see Speed <em>et al.</em> 2012) the marker- or chip-heritability <code class="reqn">h^2 = \sigma_A^2 / (\sigma_A^2 + \sigma_E^2)</code>
equals the narrow-sense heritability.

</p>
</li>
<li><p> It is assumed that the genetic relatedness matrix <code class="reqn">K</code> is scaled such that <code class="reqn">trace(P K P) = n - 1</code>, where
<code class="reqn">P</code> is the projection matrix <code class="reqn">I_n - 1_n 1_n' / n</code>, for the identity matrix <code class="reqn">I_n</code> and <code class="reqn">1_n</code> being a column vector of ones.
If this is not the case, <code class="reqn">K</code> is automatically scaled prior to fitting the mixed model.

</p>
</li>
<li><p> The model can optionally include a term <code class="reqn">X_{ij} \beta</code>, where <code class="reqn">X_{ij}</code>
is the row vector with observations on <code class="reqn">k</code> extra covariates and the vector <code class="reqn">\beta</code> contains their effects.
In this case the argument <code>covariates</code> should be the (N x k) matrix or
data-frame with rows <code class="reqn">X_{ij}</code> (N being the total number of observations).
Observations where either <code class="reqn">Y_{ij}</code> or any of the covariates is missing are discarded.

</p>
</li>
<li><p> Confidence intervals for heritability are constructed using the delta-method and the inverse AI-matrix.
The delta-method can be applied either directly to the
function <code class="reqn">(\sigma_A^2,\sigma_E^2) -&gt; \sigma_A^2 / (\sigma_A^2 + \sigma_E^2)</code> or to the function
<code class="reqn">(\sigma_A^2,\sigma_E^2) -&gt; log(\sigma_A^2 / \sigma_E^2)</code>. In the latter case,
a confidence interval for <code class="reqn">log(\sigma_A^2 / \sigma_E^2)</code> is obtained, which is back-transformed to a confidence interval for heritability.
This approach (proposed in Kruijer <em>et al.</em>) has the advantage that intervals are always contained in the unit interval.

</p>
</li>
<li><p> The AI-algorithm is run for <code>max.iter</code> iterations. If by then there is no convergence a warning is printed and the current estimates are returned.
</p>
</li></ul>



<h3>Value</h3>

<p>A list with the following components:
</p>

<ul>
<li><p>va:  REML-estimate of the (additive) genetic variance.
</p>
</li>
<li><p>ve:  REML-estimate of the residual variance.
</p>
</li>
<li><p>h2:  Plug-in estimate of heritability: <code class="reqn">va / (va + ve)</code>.
</p>
</li>
<li><p>conf.int1:  1-alpha confidence interval for heritability.
</p>
</li>
<li><p>conf.int2:  1-alpha confidence interval for heritability, obtained by
application of the delta method on a logarithmic scale.
</p>
</li>
<li><p>inv.ai:  The inverse of the average information (AI) matrix.
</p>
</li>
<li><p>loglik:  The log-likelihood.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Willem Kruijer.
</p>


<h3>References</h3>


<ul>
<li><p> Gilmour <em>et al.</em> Gilmour, A.R., R. Thompson and B.R. Cullis (1995) Average Information REML: An Efficient Algorithm for Variance Parameter
Estimation in Linear Mixed Models. <em>Biometrics</em>, volume 51, number 4, 1440-1450.
</p>
</li>
<li><p> Kruijer, W. <em>et al.</em> (2015) Marker-based estimation of heritability in immortal
populations. <em>Genetics</em>, Vol. 199(2), p. 1-20.
</p>
</li>
<li><p> Speed, D., G. Hemani, M. R. Johnson, and D.J. Balding (2012) Improved heritability
estimation from genome-wide snps. <em>the American journal of human genetics</em> 91: 1011-1021.
</p>
</li></ul>



<h3>See Also</h3>

<p>For marker-based estimation of heritability using genotypic means, see
<code><a href="#topic+marker_h2_means">marker_h2_means</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LD)
data(K_atwell)
# Heritability estimation for all observations:
#out &lt;- marker_h2(data.vector=LD$LD,geno.vector=LD$genotype,
#                 covariates=LD[,4:8],K=K_atwell)
# Heritability estimation for a randomly chosen subset of 20 accessions:
set.seed(123)
sub.set &lt;- which(LD$genotype %in% sample(levels(LD$genotype),20))
out &lt;- marker_h2(data.vector=LD$LD[sub.set],geno.vector=LD$genotype[sub.set],
                 covariates=LD[sub.set,4:8],K=K_atwell)
</code></pre>

<hr>
<h2 id='marker_h2_means'>
Compute a marker-based estimate of heritability, given genotypic means.
</h2><span id='topic+marker_h2_means'></span>

<h3>Description</h3>

<p>Given a genetic relatedness matrix and genotypic means,
this function computes REML-estimates of the genetic and
residual variance and their standard errors, using the AI-algorithm (Gilmour <em>et al.</em> 1995).
Based on this, heritability estimates and confidence intervals are given
(the estimator <code class="reqn">h_m^2</code> in Kruijer <em>et al.</em>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>marker_h2_means(data.vector, geno.vector, K, Dm=NULL, alpha = 0.05, eps = 1e-06,
       max.iter = 100, fix.h2 = FALSE, h2 = 0.5, grid.size=99)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="marker_h2_means_+3A_data.vector">data.vector</code></td>
<td>

<p>A vector of phenotypic observations, typically genotypic means. Needs to be of type numeric. May contain
missing values.
</p>
</td></tr>
<tr><td><code id="marker_h2_means_+3A_geno.vector">geno.vector</code></td>
<td>

<p>A vector of genotype labels, either a factor or character. This vector should
correspond to <code>data.vector</code>, and hence needs to be of the same length.
</p>
</td></tr>
<tr><td><code id="marker_h2_means_+3A_k">K</code></td>
<td>

<p>A genetic relatedness or kinship matrix, typically marker-based.
Must have row- and column-names corresponding to the levels of <code>geno.vector</code>
</p>
</td></tr>
<tr><td><code id="marker_h2_means_+3A_dm">Dm</code></td>
<td>

<p>Covariance of the genotypic means contained in data.vector; see details. Should be of class matrix, with row- and column-names corresponding to the levels of <code>geno.vector</code>
</p>
</td></tr>
<tr><td><code id="marker_h2_means_+3A_alpha">alpha</code></td>
<td>

<p>Confidence level, for the 1-alpha confidence intervals.
</p>
</td></tr>
<tr><td><code id="marker_h2_means_+3A_eps">eps</code></td>
<td>

<p>Numerical precision, used as convergence criterion in the AI-algorithm.
</p>
</td></tr>
<tr><td><code id="marker_h2_means_+3A_max.iter">max.iter</code></td>
<td>

<p>Maximal number of iterations in the AI-algorithm.
</p>
</td></tr>
<tr><td><code id="marker_h2_means_+3A_fix.h2">fix.h2</code></td>
<td>

<p>Compute the log-likelihood and inverse AI-matrix for a fixed heritability value. Default is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="marker_h2_means_+3A_h2">h2</code></td>
<td>

<p>When <code>fix.h2</code> is <code>TRUE</code>, the value of the heritability. Must be of type numeric, between 0 and 1.
</p>
</td></tr>
<tr><td><code id="marker_h2_means_+3A_grid.size">grid.size</code></td>
<td>
<p>If the AI-algorithm has not converged after <code>max.iter</code> iterations,
the likelihood is computed on the grid of heritability values 1/(<code>grid.size</code>+1),...,<code>grid.size</code>/(<code>grid.size</code>+1); see details.
</p>
</td></tr>
</table>


<h3>Details</h3>



<ul>
<li><p> Given phenotypic observations <code class="reqn">Y_{i}</code> for genotypes <code class="reqn">i=1,...,n</code>, the mixed model
<code class="reqn">Y_{i} = \mu + G_i + E_{i}</code>
is assumed. Typically, the <code class="reqn">Y_{i}</code> are genotypic means or BLUEs obtained from fitting a linear (mixed) model to
the raw data, containing several plants or plots for each genotype.
The vector of additive genetic effects <code class="reqn">(G_1,...,G_n)'</code> follows a
multivariate normal distribution with mean zero and covariance <code class="reqn">\sigma_A^2 K</code>,
where <code class="reqn">\sigma_A^2</code> is the additive genetic variance, and <code class="reqn">K</code> is a genetic relatedness matrix derived from a dense set of markers.
The vector of errors <code class="reqn">(E_1,...,E_n)'</code> follows a
multivariate normal distribution with mean zero and covariance <code class="reqn">\sigma_E^2 D_m</code>,
where <code class="reqn">D_m</code> is the covariance of the means obtained from the initial analysis.
In case of a completely randomized design with <code class="reqn">r_i</code> replicates for genotypes
<code class="reqn">i=1,...,n</code>, <code class="reqn">D_m</code> is diagonal with elements <code class="reqn">1 / r_i</code>.
Under certain assumptions (see Speed <em>et al.</em> 2012) the marker- or chip-heritability <code class="reqn">h^2 = \sigma_A^2 / (\sigma_A^2 + \sigma_E^2)</code>
equals the narrow-sense heritability.

</p>
</li>
<li><p> As in the <code><a href="#topic+marker_h2">marker_h2</a></code> function, it is assumed that the genetic relatedness matrix <code class="reqn">K</code> is scaled such that <code class="reqn">trace(P K P) = n - 1</code>, where
<code class="reqn">P</code> is the projection matrix <code class="reqn">I_n - 1_n 1_n' / n</code>, for the identity matrix <code class="reqn">I_n</code> and <code class="reqn">1_n</code> being a column vector of ones.
If this is not the case, <code class="reqn">K</code> is automatically scaled prior to fitting the mixed model.

</p>
</li>
<li><p> No covariates can be included, as it is assumed that these are available at plant- or plot level, and accounted for in the genotypic means.

</p>
</li>
<li><p> The resulting heritability estatimes are less accurate than those obtained from individual plant or plot data, and the likelihood can be monotone
in <code class="reqn">h^2 = \sigma_A^2 / (\sigma_A^2 + \sigma_E^2)</code>. If the AI-algorithm has not converged after <code>max.iter</code> iterations,
the likelihood is computed on the grid of heritability values 1/(<code>grid.size</code>+1),...,<code>grid.size</code>/(<code>grid.size</code>+1)

</p>
</li>
<li><p> As in the <code><a href="#topic+marker_h2">marker_h2</a></code> function, confidence intervals for heritability are constructed using the delta-method and the inverse AI-matrix.
The delta-method can be applied either directly to the
function <code class="reqn">(\sigma_A^2,\sigma_E^2) -&gt; \sigma_A^2 / (\sigma_A^2 + \sigma_E^2)</code> or to the function
<code class="reqn">(\sigma_A^2,\sigma_E^2) -&gt; log(\sigma_A^2 / \sigma_E^2)</code>. In the latter case,
a confidence interval for <code class="reqn">log(\sigma_A^2 / \sigma_E^2)</code> is obtained, which is back-transformed to a confidence interval for heritability.
This approach (proposed in Kruijer <em>et al.</em>) has the advantage that intervals are always contained in the unit interval.

</p>
</li></ul>



<h3>Value</h3>

<p>A list with the following components:
</p>

<ul>
<li><p>va:  REML-estimate of the (additive) genetic variance.
</p>
</li>
<li><p>ve:  REML-estimate of the residual variance.
</p>
</li>
<li><p>h2:  Plug-in estimate of heritability: <code class="reqn">va / (va + ve)</code>.
</p>
</li>
<li><p>conf.int1:  1-alpha confidence interval for heritability.
</p>
</li>
<li><p>conf.int2:  1-alpha confidence interval for heritability, obtained by
application of the delta method on a logarithmic scale.
</p>
</li>
<li><p>inv.ai:  The inverse of the average information (AI) matrix.
</p>
</li>
<li><p>loglik:  The log-likelihood.
</p>
</li>
<li><p>loglik.vector:  Empty numeric vector if the AI-algorthm converged within <code class="reqn">max.iter</code> iterations. Otherwise it contains the log-likelihood on a grid.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Willem Kruijer.
</p>


<h3>References</h3>


<ul>
<li><p> Gilmour <em>et al.</em> Gilmour, A.R., R. Thompson and B.R. Cullis (1995) Average Information REML: An Efficient Algorithm for Variance Parameter
Estimation in Linear Mixed Models. <em>Biometrics</em>, volume 51, number 4, 1440-1450.
</p>
</li>
<li><p> Kruijer, W. <em>et al.</em> (2015) Marker-based estimation of heritability in immortal
populations. <em>Genetics</em>, Vol. 199(2), p. 1-20.
</p>
</li>
<li><p> Speed, D., G. Hemani, M. R. Johnson, and D.J. Balding (2012) Improved heritability
estimation from genome-wide snps. <em>the American journal of human genetics</em> 91: 1011-1021.
</p>
</li></ul>



<h3>See Also</h3>

<p>For marker-based estimation of heritability using individual plant or plot data, see
<code><a href="#topic+marker_h2">marker_h2</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(means_LDV)
data(R_matrix_LDV)
data(K_atwell)
out &lt;- marker_h2_means(data.vector=means_LDV$LDV,geno.vector=means_LDV$genotype,
                       K=K_atwell,Dm=R_matrix_LDV)
# Takes about a minute:
#data(means_LD)
#data(R_matrix_LD)
#out &lt;- marker_h2_means(data.vector=means_LD$LD,geno.vector=means_LD$genotype,
#                       K=K_atwell,Dm=R_matrix_LD)
# The likelihood is monotone increasing:
#plot(x=(1:99)/100,y=out$loglik.vector,type="l",ylab="log-likelihood",lwd=2,
#     main='',xlab='h2',cex.lab=2,cex.axis=2.5)
</code></pre>

<hr>
<h2 id='means_floweringTime'>
Flowering time from Atwell <em>et al.</em> (2010): accession means.
</h2><span id='topic+means_floweringTime'></span><span id='topic+means_LD'></span><span id='topic+means_LDV'></span>

<h3>Description</h3>

<p>Accession means for the flowering time data contained in <code><a href="#topic+LD">LD</a></code> and <code><a href="#topic+LDV">LDV</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(means_LD); data(means_LDV)</code></pre>


<h3>Format</h3>

<p>Data-frames with flowering time means:
</p>

<dl>
<dt><code>genotype</code></dt><dd><p>a factor, the levels being the accession or ecotype identifiers</p>
</dd>
<dt><code>LD</code></dt><dd><p>Flowering time under long days, in number of days</p>
</dd>
<dt><code>LDV</code></dt><dd><p>Flowering time under long days and vernalization, in number of days</p>
</dd>
</dl>



<h3>Details</h3>

<p>Following Kruijer <em>et al.</em> (appendix A)
these means were defined as the least-squares estimate for the factor accession, in a linear model
containing both accession and replicate effects.
Consequently there are differences compared to Atwell <em>et al.</em> (2010), where just the
arithmetic averages are considered.
</p>


<h3>References</h3>


<ul>
<li><p> Atwell, S., Y. S. Huang, B. J. Vilhjalmsson, G. Willems, M. Horton, <em>et al.</em> (2010)
Genome-wide association study of 107 phenotypes in Arabidopsis thaliana inbred lines. Nature
465: 627-631.
</p>
</li>
<li><p> Kruijer, W. <em>et al.</em> (2015) Marker-based estimation of heritability in immortal
populations. <em>Genetics</em>, Vol. 199(2), p. 1-20.
</p>
</li></ul>



<h3>See Also</h3>

<p>Together with the covariance matrices contained in <code><a href="#topic+R_matrix_LD">R_matrix_LD</a></code> and
<code><a href="#topic+R_matrix_LDV">R_matrix_LDV</a></code>, the means contained in <code>means_LD</code> and <code>means_LDV</code>
can be used to estimate heritability, using the function <code><a href="#topic+marker_h2_means">marker_h2_means</a></code>.
For the corresponding genetic relatedness matrix, see <code><a href="#topic+K_atwell">K_atwell</a></code>.
For the individual plant data, see <code><a href="#topic+floweringTime">floweringTime</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(means_LD)
data(means_LDV)
</code></pre>

<hr>
<h2 id='R_matrix'>
Covariance matrix of the accession means for flowering time.
</h2><span id='topic+R_matrix'></span><span id='topic+R_matrix_LDV'></span><span id='topic+R_matrix_LD'></span>

<h3>Description</h3>

<p>Covariance matrices of the accession means for flowering time contained in <code><a href="#topic+means_LD">means_LD</a></code>
and <code><a href="#topic+means_LDV">means_LDV</a></code>, derived from the Atwell <em>et al.</em> (2010) data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(R_matrix_LDV);data(R_matrix_LD)</code></pre>


<h3>Format</h3>

<p>Matrix whose row- and column names are the ecotype-IDs of the accessions contained in
<code><a href="#topic+LD">LD</a></code> and <code><a href="#topic+LDV">LDV</a></code>.
</p>


<h3>Details</h3>

<p>The matrix was computed as in Kruijer <em>et al.</em>, Appendix A.
</p>


<h3>References</h3>


<ul>
<li><p> Atwell, S., Y. S. Huang, B. J. Vilhjalmsson, G. Willems, M. Horton, <em>et al.</em> (2010)
Genome-wide association study of 107 phenotypes in Arabidopsis thaliana inbred lines. Nature
465: 627-631.
</p>
</li>
<li><p> Kruijer, W. <em>et al.</em> (2015) Marker-based estimation of heritability in immortal
populations. <em>Genetics</em>, Vol. 199(2), p. 1-20.
</p>
</li></ul>



<h3>See Also</h3>

<p>Together with the corresponding means contained in <code><a href="#topic+means_LD">means_LD</a></code> and <code><a href="#topic+means_LDV">means_LDV</a></code>,
these matrices can be used to estimate heritability, using the function <code><a href="#topic+marker_h2_means">marker_h2_means</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(R_matrix_LD); data(R_matrix_LDV)
</code></pre>

<hr>
<h2 id='repeatability'>ANOVA-based estimates of repeatability</h2><span id='topic+repeatability'></span>

<h3>Description</h3>

<p>Given a population where each genotype is phenotyped for a number of genetically
identical replicates (either individual plants or plots in a field trial), the repeatability
or intra-class correlation can be estimated by
<code class="reqn">V_g / (V_g + V_e)</code>, where <code class="reqn">V_g = (MS(G) - MS(E)) / r</code> and <code class="reqn">V_e = MS(E)</code>. In these expressions,
<code class="reqn">r</code> is the number of replicates per genotype, and <code class="reqn">MS(G)</code> and <code class="reqn">MS(E)</code> are
the mean sums of squares for genotype and residual error obtained from analysis
of variance. In case <code class="reqn">MS(G) &lt; MS(E)</code>, <code class="reqn">V_g</code> is set to zero.
See Singh <em>et al.</em> (1993) or Lynch and Walsh (1998), p.563.
When the genotypes have differing numbers of replicates, <code class="reqn">r</code> is replaced by
<code class="reqn">\bar r = (n-1)^{-1} (R_1 - R_2 / R_1)</code>, where <code class="reqn">R_1 = \sum r_i</code> and <code class="reqn">R_2 = \sum r_i^2</code>.
Under the assumption that all differences between genotypes are genetic,
repeatability equals broad-sense heritability; otherwise it only provides an upper-bound for broad-sense heritability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>repeatability(data.vector, geno.vector, line.repeatability = FALSE,
              covariates.frame = data.frame())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="repeatability_+3A_data.vector">data.vector</code></td>
<td>
<p>A vector of phenotypic observations. Needs to be of type numeric. May contain
missing values.</p>
</td></tr>
<tr><td><code id="repeatability_+3A_geno.vector">geno.vector</code></td>
<td>
<p>A vector of genotype labels, either a factor or character. This vector should
correspond to <code>data.vector</code>, and hence needs to be of the same length.</p>
</td></tr>
<tr><td><code id="repeatability_+3A_line.repeatability">line.repeatability</code></td>
<td>
<p>If <code>TRUE</code>, the line-repeatability or
line-heritability <code class="reqn">\sigma_G^2 / (\sigma_G^2 + \sigma_E^2 / r)</code> is estimated,
otherwise (the default) the repeatability at plot- or plant level, which is <code class="reqn">\sigma_G^2 / (\sigma_G^2 + \sigma_E^2)</code>.
</p>
</td></tr>
<tr><td><code id="repeatability_+3A_covariates.frame">covariates.frame</code></td>
<td>
<p>A data-frame with additional covariates, the rows corresponding to
<code>geno.vector</code> and the phenotypic observations in <code>data.vector</code>.
May contain missing values. Each column can be numeric or a factors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>

<ul>
<li><p>repeatability:  the estimated repeatability.
</p>
</li>
<li><p>gen.variance:  the estimated genetic variance.
</p>
</li>
<li><p>res.variance:  the estimated residual variance.
</p>
</li>
<li><p>line.repeatability:  whether repeatability was estimated at the
individual plant or plot level (the default), or at the level of
genotypic means (in the latter case, <code>line.repeatability=TRUE</code>) 
</p>
</li>
<li><p>average.number.of.replicates:  The average number of replicates. See the description above.
</p>
</li>
<li><p>conf.int:  Confidence interval for repeatability. See Singh <em>et al.</em> (1993) or Lynch and Walsh (1998)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Willem Kruijer <a href="mailto:willem.kruijer@wur.nl">willem.kruijer@wur.nl</a>
</p>


<h3>References</h3>


<ul>
<li><p> Kruijer, W. <em>et al.</em> (2015) Marker-based estimation of heritability in immortal
populations. <em>Genetics</em>, Vol. 199(2), p. 1-20.
</p>
</li>
<li><p> Lynch, M., and B. Walsh (1998) Genetics and Analysis of Quantitative Traits. Sinauer As-
sociates, 1st edition.
</p>
</li>
<li><p> Singh, M., S. Ceccarelli, and J. Hamblin (1993) Estimation of heritability from varietal
trials data. <em>Theoretical and Applied Genetics</em> 86: 437-441.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>repeatability(data.vector=rep(rnorm(26),each=5) + rnorm(5*26),
              geno.vector=rep(letters,each=5))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
