<!DOCTYPE html><html lang="en"><head><title>Help for package HeritSeq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HeritSeq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#computeVPC.CP'><p>Calculate the compound Poisson (CP) variance partition coefficient (VPC)</p>
for one or more features.</a></li>
<li><a href='#computeVPC.NB'><p>Calculate the negative binomial (NB) variance partition coefficient (VPC)</p>
for one or more features.</a></li>
<li><a href='#fit.CP'><p>Fit compound Poisson mixed effect models (CPMM) for one or more features.</p></a></li>
<li><a href='#fit.NB'><p>Fit negative binomial mixed models (NBMM) for one or more features.</p></a></li>
<li><a href='#fitComputeVPC.lmer'><p>Fit linear mixed models (LMM) and compute the VPC values for one or more</p>
features.</a></li>
<li><a href='#getBootCI'><p>Compute variance partition coefficition (VPC) confidence intervals (CI)</p>
for one or more features.</a></li>
<li><a href='#getReadMatrix.CP'><p>Simulate a read matrix from compound Poisson mixed effect models (CPMM).</p></a></li>
<li><a href='#getReadMatrix.NB'><p>Simulate a count matrix from negative binomial mixed effect models (NBMM).</p></a></li>
<li><a href='#para_cp'><p>Parameter matrix obtained from simData by fitting CPMM.</p></a></li>
<li><a href='#para_nb'><p>Parameter matrix obtained from simData by fitting NBMM.</p></a></li>
<li><a href='#simData'><p>A simulated sequencing dataset.</p></a></li>
<li><a href='#simData_voom'><p>Voom transformed version of simData.</p></a></li>
<li><a href='#simData_vst'><p>Variance stabilize transformed version of simData.</p></a></li>
<li><a href='#strains'><p>List of strain names for the samples.</p></a></li>
<li><a href='#weights_voom'><p>Weights used in the voom transformation.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Heritability of Gene Expression for Next-Generation Sequencing</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-07-11</td>
</tr>
<tr>
<td>Author:</td>
<td>W. Jenny Shi [aut, cre],
    Pamela Russell [aut],
    Pratyaydipta Rudra [aut, cre],
    Brian Vestal [aut, cre],
    Katerina Kechris [aut],
    Laura Saba [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>W. Jenny Shi &lt;wjennyshi@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Statistical framework to analyze heritability of gene expression 
    based on next-generation sequencing data and simulating sequencing reads. 
    Variance partition coefficients (VPC) are computed using linear mixed effects 
    and generalized linear mixed effects models. Compound Poisson and negative 
    binomial models are included. Reference: Rudra, Pratyaydipta, et al. "Model based heritability scores for high-throughput sequencing data." BMC bioinformatics 18.1 (2017): 143.</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/KechrisLab/HeritSeq/issues">https://github.com/KechrisLab/HeritSeq/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.3)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>cplm, DESeq2, lme4, pbapply, tweedie, MASS,
SummarizedExperiment</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-07-12 04:27:30 UTC; wenjennyshi</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-07-12 22:30:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='computeVPC.CP'>Calculate the compound Poisson (CP) variance partition coefficient (VPC) 
for one or more features.</h2><span id='topic+computeVPC.CP'></span>

<h3>Description</h3>

<p>Calculate the CP VPC for one or more features following the model fitting 
function fit.CP().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computeVPC.CP(para)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="computeVPC.CP_+3A_para">para</code></td>
<td>
<p>A <code class="reqn">G \times 4</code> matrix of CP fit parameters for 
<code class="reqn">G</code> features, <code class="reqn">G\geq 1</code>. The column order is intercept 
<code class="reqn">\alpha_g</code>, random effect <code class="reqn">\sigma_g^2 (\sigma_g^2\geq0)</code>, 
tweedie parameter <code class="reqn">p_g (1&lt;p_g&lt;2)</code>,
and dispersion <code class="reqn">\phi (\phi&gt;0)</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code class="reqn">G \times 1</code> matrix consisting of VPC for
G features based on compound Poisson mixed models. Column name is 
&quot;CP-fit&quot;; row names are the feature names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute VPC for each feature under compound Poisson mixed models. 
vpc.cp &lt;- computeVPC.CP(para_cp) 

## Visulize the distribution of the VPCs. 
hist(vpc.cp, breaks = 50, col = "cyan")

## Plot sorted VPCs.
plot(sort(vpc.cp), ylab = "Heritability (h2)", ylim = c(0,1), main = "Sorted CP VPC scores")
abline(h = 0.9, lty = 2, col = "red")
text(50, 0.92, "h2 = 0.9", col = "red")
</code></pre>

<hr>
<h2 id='computeVPC.NB'>Calculate the negative binomial (NB) variance partition coefficient (VPC)
for one or more features.</h2><span id='topic+computeVPC.NB'></span>

<h3>Description</h3>

<p>Calculate the NB VPC for one or more features following the model fitting 
function fit.NB().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computeVPC.NB(para)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="computeVPC.NB_+3A_para">para</code></td>
<td>
<p>A <code class="reqn">G \times 3</code> matrix of negative binomial fit 
parameters for <code class="reqn">G</code> features, <code class="reqn">G\geq 1</code>. The column order is intercept 
<code class="reqn">\alpha_g</code>, random effect <code class="reqn">\sigma_g^2 (\sigma_g^2\geq0)</code>, 
and dispersion <code class="reqn">\phi (\phi&gt;0)</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code class="reqn">G \times 1</code> matrix consisting of VPC for
<code class="reqn">G</code> features based on negative binomial mixed model. Column name 
is &quot;NB-fit&quot;; row names are the feature names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute VPC for each feature under negative binomial mixed model.
vpc.nb &lt;- computeVPC.NB(para_nb)

## Visulize the distribution of the VPCs. 
hist(vpc.nb, breaks = 50, col = "cyan")

## Plot sorted VPCs.
plot(sort(vpc.nb), ylab = "Heritability (h2)", ylim = c(0,1), 
main = "Sorted NB VPC scores")
abline(h = 0.9, lty = 2, col = "red")
text(50, 0.92, "h2 = 0.9", col = "red")
</code></pre>

<hr>
<h2 id='fit.CP'>Fit compound Poisson mixed effect models (CPMM) for one or more features.</h2><span id='topic+fit.CP'></span>

<h3>Description</h3>

<p>Fit a CPMM for one or more features and output the fit parameters. 
It is used before the function computeVPC.CP(). This function also allows 
to test the presence of heritability via random effect variance of the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit.CP(CountMatrix, Strains, test = FALSE, optimizer = "nlminb")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit.CP_+3A_countmatrix">CountMatrix</code></td>
<td>
<p>Sequencing count matrix for one or more features. Each 
row is for one feature, and the columns are for samples.</p>
</td></tr>
<tr><td><code id="fit.CP_+3A_strains">Strains</code></td>
<td>
<p>Strain labels for the samples.</p>
</td></tr>
<tr><td><code id="fit.CP_+3A_test">test</code></td>
<td>
<p>TRUE or FALSE (default). Test the presence of heritability 
through examining the random effect variance <code class="reqn">\sigma_g^2 = 0</code>.</p>
</td></tr>
<tr><td><code id="fit.CP_+3A_optimizer">optimizer</code></td>
<td>
<p>A character string that determines which optimization 
routine is to be used. Possible choices are &quot;nlminb&quot; (default), 
&quot;L-BFGS-B&quot;, and &quot;bobyqa&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two objects. The first object is a 
<code class="reqn">G \times 4</code> matrix indicating the fitted parameters for each 
feature. The columns are ordered by intercept <code class="reqn">\alpha_g</code>, tweedie 
parameter <code class="reqn">p_g</code>, random effect variance <code class="reqn">\sigma^2_g</code>, and dispersion
<code class="reqn">\phi_g</code>. Row names are feature names. If the argument test is set to
be true, the second object of the list consists of p-values for testing 
the hypothesis that random effects <code class="reqn">\sigma_a^2 = 0</code>; 
otherwise, the second object is NULL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Fit CPMM for the first two features and test the presence of 
## heritability. 
result.cp &lt;- fit.CP(simData[1:2, ], strains, test = TRUE)
## Extract parameters
para.cp &lt;- result.cp[[1]]
## Extract p-values
pval.cp &lt;- result.cp[[2]]

</code></pre>

<hr>
<h2 id='fit.NB'>Fit negative binomial mixed models (NBMM) for one or more features.</h2><span id='topic+fit.NB'></span>

<h3>Description</h3>

<p>Fit NBMM for one or more features and output the fit parameters. 
It is used before the function computeVPC.NB(). This function also allows 
to test the presence of heritability via random effect variance of the model. To fit a NBMM, the glmmADMB package is needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit.NB(CountMatrix, Strains, test = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit.NB_+3A_countmatrix">CountMatrix</code></td>
<td>
<p>Sequencing count matrix for a list of features. Each row 
is for one feature, and the columns are for samples.</p>
</td></tr>
<tr><td><code id="fit.NB_+3A_strains">Strains</code></td>
<td>
<p>Strain labels for the samples.</p>
</td></tr>
<tr><td><code id="fit.NB_+3A_test">test</code></td>
<td>
<p>TRUE or FALSE (default). Test the presence of heritability 
through examining the random effect variance <code class="reqn">\sigma_g^2 = 0</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two objects. The first object is a <code class="reqn">G \times 3</code> 
matrix indicating the fitted parameters for each feature. The columns are 
ordered by <code class="reqn">\alpha_g, \sigma_g^2, \phi_g</code>. 
Row names are feature names. If the argument test is set to
be true, the second object of the list consists of p-values for testing 
the hypothesis that random effects <code class="reqn">\sigma_a^2 = 0</code>; 
otherwise, the second object is NULL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Compute vpc for each feature under NBMM. This will take a while on the
##  entire dataset. For the purpose of illustration, here we only fit on 
##  the first 2 features.
result.nb &lt;- fit.NB(simData[1:2, ], strains)

</code></pre>

<hr>
<h2 id='fitComputeVPC.lmer'>Fit linear mixed models (LMM) and compute the VPC values for one or more 
features.</h2><span id='topic+fitComputeVPC.lmer'></span>

<h3>Description</h3>

<p>Fit the Gaussian-like data to LMM and compute the VPC values for 
one or more features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitComputeVPC.lmer(
  CountMatrix,
  Strains,
  PriorWeights = NULL,
  test = FALSE,
  VPCname = "LMM"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fitComputeVPC.lmer_+3A_countmatrix">CountMatrix</code></td>
<td>
<p>Sequencing count matrix for one or more features. Each 
row is for one feature, and the columns are for samples.</p>
</td></tr>
<tr><td><code id="fitComputeVPC.lmer_+3A_strains">Strains</code></td>
<td>
<p>Strain labels for the samples.</p>
</td></tr>
<tr><td><code id="fitComputeVPC.lmer_+3A_priorweights">PriorWeights</code></td>
<td>
<p>Weights used in the lmer function in the package lme4. 
It is an optional vector used in the fitting process.</p>
</td></tr>
<tr><td><code id="fitComputeVPC.lmer_+3A_test">test</code></td>
<td>
<p>TRUE or FALSE (default). Test the presence of heritability 
through examining the random effect variance <code class="reqn">\sigma_g^2 = 0</code>.</p>
</td></tr>
<tr><td><code id="fitComputeVPC.lmer_+3A_vpcname">VPCname</code></td>
<td>
<p>Name of the VPC result, default = &quot;LMM&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two objects. The first object is a 
<code class="reqn">1 \times G</code> vector indicating the variance partition coefficients
(VPC). If the argument test is set to be true, the second object of 
the list consists of p-values for testing the hypothesis that random 
effects <code class="reqn">\sigma_a^2 = 0</code>; otherwise, the second 
object is NULL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute VPC for the first two features under linear mixed models for Gaussian-like datasets. 

## Provide normalized data and include hypothesis testing on presence of
## heritability:
result.vst &lt;- fitComputeVPC.lmer(simData_vst[1:2,], strains, test = TRUE)
## Extract parameters
vpc.vst &lt;- result.vst[[1]]
## Extract p-values
pval.vst &lt;- result.vst[[2]]

## Visulize the distribution of p-values.
hist(pval.vst, breaks = 30, col = "cyan")
</code></pre>

<hr>
<h2 id='getBootCI'>Compute variance partition coefficition (VPC) confidence intervals (CI) 
for one or more features.</h2><span id='topic+getBootCI'></span>

<h3>Description</h3>

<p>Compute VPC CI based on parametric bootstrap for one or more features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBootCI(
  CountMatrix,
  Strains,
  which.features,
  num.boot,
  method = "NB-fit",
  alpha = 0.05,
  optimizer = "nlminb"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getBootCI_+3A_countmatrix">CountMatrix</code></td>
<td>
<p>A <code class="reqn">G\times N</code> count matrix. <code class="reqn">G</code> is the number of 
features; <code class="reqn">N</code> is the total number of samples.</p>
</td></tr>
<tr><td><code id="getBootCI_+3A_strains">Strains</code></td>
<td>
<p>A <code class="reqn">1\times N</code> vector of strain labels corresponding to 
each sample.</p>
</td></tr>
<tr><td><code id="getBootCI_+3A_which.features">which.features</code></td>
<td>
<p>A <code class="reqn">1\times k</code> vector of select feature numbers 
for which CI is desired. <code class="reqn">k\leq G</code>.</p>
</td></tr>
<tr><td><code id="getBootCI_+3A_num.boot">num.boot</code></td>
<td>
<p>Number of bootstraps.</p>
</td></tr>
<tr><td><code id="getBootCI_+3A_method">method</code></td>
<td>
<p>Which method should be used, &quot;CP-fit&quot;, &quot;NB-fit&quot; (default), 
or &quot;VST&quot;. &quot;VST&quot; method bootstraps data under negative binomial mixed models.</p>
</td></tr>
<tr><td><code id="getBootCI_+3A_alpha">alpha</code></td>
<td>
<p>A numerical value between 0 and 1, indicating the significance 
level of the CI. The CI will be <code class="reqn">100*(1-\alpha)</code> 
percent CI. Default value is 0.05.</p>
</td></tr>
<tr><td><code id="getBootCI_+3A_optimizer">optimizer</code></td>
<td>
<p>A character string that determines which optimization 
routine is to be used. It is only used for method = &quot;CP-fit&quot;. Possible 
choices are &quot;nlminb&quot; (default), &quot;L-BFGS-B&quot;, and &quot;bobyqa&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two objects. The first object is a <code class="reqn">k \times 2</code> 
matrix containing the CI. The second object consists of a 
<code class="reqn">k \times</code>num.boot matrix of all bootsrapped VPC values.
</p>

<hr>
<h2 id='getReadMatrix.CP'>Simulate a read matrix from compound Poisson mixed effect models (CPMM).</h2><span id='topic+getReadMatrix.CP'></span>

<h3>Description</h3>

<p>Simulate a (possibly unbalanced) read matrix from CPMM.
For a compound Poisson (CP) random variable <code class="reqn">Y_{gsr}</code> with mean 
<code class="reqn">\mu_{gs}</code>, its variance can be expressed as 
<code class="reqn">\phi_g\mu_{gs}^{p_g}</code>, for some <code class="reqn">1&lt;p_g&lt;2</code>. Under the CPMM, with 
a <code class="reqn">\log</code>-link, the regression on the mean has the form:<br />
<code class="reqn">\log(\mu_{gs}) = \alpha_g+ b_{gs}, \;\;b_{gs}\sim N(0, \sigma^2_g).</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getReadMatrix.CP(vec.num.rep, alphas, sigma2s, ps, phis)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getReadMatrix.CP_+3A_vec.num.rep">vec.num.rep</code></td>
<td>
<p>A vector of replicate numbers for each strain.</p>
</td></tr>
<tr><td><code id="getReadMatrix.CP_+3A_alphas">alphas</code></td>
<td>
<p>Intercept vector <code class="reqn">\alpha_g</code>'s, 
<code class="reqn">1 \times \texttt{num.features}</code>.</p>
</td></tr>
<tr><td><code id="getReadMatrix.CP_+3A_sigma2s">sigma2s</code></td>
<td>
<p>Random effect variance vector <code class="reqn">\sigma^2_g</code>'s, 
<code class="reqn">1 \times \texttt{num.features}</code>.</p>
</td></tr>
<tr><td><code id="getReadMatrix.CP_+3A_ps">ps</code></td>
<td>
<p>Tweedie parameter in CP models, <code class="reqn">p_g</code>'s, a 
<code class="reqn">1 \times \texttt{num.features}</code> vector.</p>
</td></tr>
<tr><td><code id="getReadMatrix.CP_+3A_phis">phis</code></td>
<td>
<p>Dispersion parameter in CP models, <code class="reqn">\phi_g</code>'s, a 
<code class="reqn">1 \times \texttt{num.features}</code> vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code class="reqn">G \times N</code> matrix with CP reads. <code class="reqn">N</code> is the 
total number of samples; <code class="reqn">G</code> is the number of features. Column names 
are sample names of the form &quot;Ss_r&quot;, where S stands for sample, s is the 
strain number, r is the replicate number within the strain. Row names 
are the feature names of the form &quot;Gene g&quot;, where g is the feature index.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Generate a sequencing dataset with 5 features and 6 strains. 
## Assign parameter values.
rep.num &lt;- c(3, 5, 2, 3, 4, 2)
a0s &lt;- c(-1, 1, 2, 5, 10)
sig2s &lt;- c(10, 0.2, 0.1, 0.03, 0.01)
ps &lt;- rep(1.5, 5)
phis &lt;- c(1.5, 1, 0.5, 0.1, 0.1)

set.seed(1234)
## Generate reads:
cpData &lt;- getReadMatrix.CP(rep.num, a0s, sig2s, ps, phis)
## Generate strain names:
str &lt;- sapply(1:length(rep.num), function(x){
  str.x &lt;- paste0("S", x)
  return(rep(str.x, rep.num[x]))
})
str &lt;- do.call(c, str)
</code></pre>

<hr>
<h2 id='getReadMatrix.NB'>Simulate a count matrix from negative binomial mixed effect models (NBMM).</h2><span id='topic+getReadMatrix.NB'></span>

<h3>Description</h3>

<p>Simulate a (possibly unbalanced) count matrix from NBMM.
Under NBMM, an observed number of reads aligned to feature/gene <code class="reqn">g</code>, 
<code class="reqn">Y_{gsr}</code>, follows a negative binomial (NB) distribution with mean 
<code class="reqn">\mu_{gs}</code> and variance <code class="reqn">\mu_{gs}+\phi_{g} \mu_{gs}^2</code>, where 
<code class="reqn">\phi_g</code> is the dispersion parameter, shared across strains. The 
generalized linear model uses a <code class="reqn">\log</code>-link:<br />
<code class="reqn">\log(\mu_{gs}) = \alpha_g+ b_{gs}, \;\;b_{gs}\sim N(0, \sigma^2_g).</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getReadMatrix.NB(vec.num.rep, alphas, sigma2s, phis)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getReadMatrix.NB_+3A_vec.num.rep">vec.num.rep</code></td>
<td>
<p>A vector of replicate numbers for each strain.</p>
</td></tr>
<tr><td><code id="getReadMatrix.NB_+3A_alphas">alphas</code></td>
<td>
<p>Intercept vector <code class="reqn">\alpha_g</code>'s, 
<code class="reqn">1 \times \texttt{num.features}</code>.</p>
</td></tr>
<tr><td><code id="getReadMatrix.NB_+3A_sigma2s">sigma2s</code></td>
<td>
<p>Random effect variance vector <code class="reqn">\sigma^2_g</code>'s, 
<code class="reqn">1 \times \texttt{num.features}</code>.</p>
</td></tr>
<tr><td><code id="getReadMatrix.NB_+3A_phis">phis</code></td>
<td>
<p>Dispersion parameter in NB models, <code class="reqn">\phi_g</code>'s, a
<code class="reqn">1 \times \texttt{num.features}</code> vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code class="reqn">G \times N</code> matrix with NB reads. <code class="reqn">N</code> is the 
total number of samples; <code class="reqn">G</code> is the number of features. Column names 
are sample names of the form &quot;Ss_r&quot;, where S stands for sample, s is the 
strain number, r is the replicate number within the strain. Row names 
are the feature names of the form &quot;Gene g&quot;, where g is the feature index.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Generate a sequencing dataset with 5 features and 6 strains. 
## Assign parameter values.
rep.num &lt;- c(3, 5, 2, 3, 4, 2)
a0s &lt;- c(-1, 1, 2, 5, 10)
sig2s &lt;- c(10, 0.2, 0.1, 0.03, 0.01)
phis &lt;- c(0.5, 1, 0.05, 0.01, 0.1)

set.seed(1234)
## Generate reads:
nbData &lt;- getReadMatrix.NB(rep.num, a0s, sig2s, phis)
</code></pre>

<hr>
<h2 id='para_cp'>Parameter matrix obtained from simData by fitting CPMM.</h2><span id='topic+para_cp'></span>

<h3>Description</h3>

<p>Parameter matrix obtained from simData by fitting CPMM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>para_cp
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 100 rows and 4 columns.
</p>

<hr>
<h2 id='para_nb'>Parameter matrix obtained from simData by fitting NBMM.</h2><span id='topic+para_nb'></span>

<h3>Description</h3>

<p>Parameter matrix obtained from simData by fitting NBMM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>para_nb
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 100 rows and 3 columns.
</p>

<hr>
<h2 id='simData'>A simulated sequencing dataset.</h2><span id='topic+simData'></span>

<h3>Description</h3>

<p>A matrix containing simulated counts for 100 features (rows) and 175 samples (columns)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simData
</code></pre>


<h3>Format</h3>

<p>A matrix with 100 rows and 175 columns
</p>

<hr>
<h2 id='simData_voom'>Voom transformed version of simData.</h2><span id='topic+simData_voom'></span>

<h3>Description</h3>

<p>Voom transformed version of simData.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simData_voom
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 881 rows and 175 columns.
</p>

<hr>
<h2 id='simData_vst'>Variance stabilize transformed version of simData.</h2><span id='topic+simData_vst'></span>

<h3>Description</h3>

<p>Variance stabilize transformed version of simData.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simData_vst
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 881 rows and 175 columns.
</p>

<hr>
<h2 id='strains'>List of strain names for the samples.</h2><span id='topic+strains'></span>

<h3>Description</h3>

<p>List of strain names for the samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strains
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 175.
</p>

<hr>
<h2 id='weights_voom'>Weights used in the voom transformation.</h2><span id='topic+weights_voom'></span>

<h3>Description</h3>

<p>Weights used in the voom transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weights_voom
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 881 rows and 175 columns.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
