<!DOCTYPE html><html lang="en"><head><title>Help for package HEssRNA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HEssRNA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#hIndxMeanCalc4Traits'><p>Calculate Mean Heritability Index for Traits</p></a></li>
<li><a href='#powerCalc'><p>Power Calculation from gene expression data information</p></a></li>
<li><a href='#prcesDF4modelInhouse'><p>Process Data Frame in In-House Format for Model Building</p></a></li>
<li><a href='#smplSizPred'><p>Predict Number of Replicates Based on Heritability, Power, and Fold Change</p></a></li>
<li><a href='#smplSizPredModel'><p>Generate a Linear Model for Sample Size Prediction</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Heritability-Based Estimation of Sample Size for RNA-Seq Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides tools for estimating sample sizes primarily based on heritability, while also considering additional parameters such as statistical power and fold change. The package normalizes heritability values according to trait-specific heritability and classification to enhance accuracy in sample size estimation.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>car, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>tidyr, dplyr, magrittr, stats, utils, DESeq2, ssizeRNA, Rdpack</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-12-24 09:09:39 UTC; hp</td>
</tr>
<tr>
<td>Author:</td>
<td>Naina Kumari [aut],
  Jagajjit Sahu [aut],
  Sarika Jaiswal [aut, cre],
  Mir Asif Iquebal [aut],
  Dinesh Kumar [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sarika Jaiswal &lt;sarika@icar.gov.in&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-10 13:50:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='hIndxMeanCalc4Traits'>Calculate Mean Heritability Index for Traits</h2><span id='topic+hIndxMeanCalc4Traits'></span>

<h3>Description</h3>

<p>This function processes heritability index data, filtering out empty trait names, and calculates the mean heritability for each unique trait. The resulting output is a data frame with traits and their corresponding mean heritability values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hIndxMeanCalc4Traits(hIndexValDF)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hIndxMeanCalc4Traits_+3A_hindexvaldf">hIndexValDF</code></td>
<td>
<p>A data frame containing heritability index values with at least two columns: <code>Trait.name</code> and <code>Heritability</code>.
The <code>Trait.name</code> column should contain trait identifiers, and the <code>Heritability</code> column should contain numeric heritability values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with two columns: <code>Trait.name</code> and <code>MeanValue</code>, where <code>MeanValue</code> represents the mean heritability for each trait.
</p>


<h3>References</h3>

<p>Hu et al. (2018) <a href="https://doi.org/10.1093/nar/gky1084">doi:10.1093/nar/gky1084</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Example of usage:
hIndexValDF &lt;- data.frame(Trait.name = c("Trait1", "Trait2", "Trait1", "Trait2"),
                          Heritability = c(0.5, 0.6, 0.7, 0.8))
result &lt;- hIndxMeanCalc4Traits(hIndexValDF)
print(result)


</code></pre>

<hr>
<h2 id='powerCalc'>Power Calculation from gene expression data information</h2><span id='topic+powerCalc'></span>

<h3>Description</h3>

<p>This function takes the required input information such as count data, sample data, etc. to calculate the power.
It filters the input count data, performs DESeq2 analysis to calculate differentially expressed genes (DEGs), and
then calculates the power of detecting these DEGs based on simulations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>powerCalc(
  countDat,
  smplDat,
  alpha = 0.05,
  thrsholdFC = 2,
  inptNoOfReplicates = 3,
  sims = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="powerCalc_+3A_countdat">countDat</code></td>
<td>
<p>A matrix or data frame of raw count data where rows represent genes and columns represent samples.</p>
</td></tr>
<tr><td><code id="powerCalc_+3A_smpldat">smplDat</code></td>
<td>
<p>A data frame of sample information, with at least a <code>condition</code> column that specifies the experimental condition of each sample.</p>
</td></tr>
<tr><td><code id="powerCalc_+3A_alpha">alpha</code></td>
<td>
<p>The significance level (FDR threshold) used to identify differentially expressed genes. Default is 0.05.</p>
</td></tr>
<tr><td><code id="powerCalc_+3A_thrsholdfc">thrsholdFC</code></td>
<td>
<p>The threshold for the absolute value of log2 fold change used to filter DEGs. Default is 2.</p>
</td></tr>
<tr><td><code id="powerCalc_+3A_inptnoofreplicates">inptNoOfReplicates</code></td>
<td>
<p>The input number of replicates based on which the power will be calculated. Default is 3.</p>
</td></tr>
<tr><td><code id="powerCalc_+3A_sims">sims</code></td>
<td>
<p>The number of simulations to run for power calculation. Default is 10.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Example files included with this package:
</p>

<ul>
<li> <p><code>exmplCountDat.csv</code>: A toy dataset with count data.
</p>
</li>
<li> <p><code>exmplSampleDat.csv</code>: A sample dataset with metadata.
</p>
</li></ul>

<p>These files are stored in the <code>inst/extdata</code> directory and can be accessed
using the <code>system.file()</code> function in R.
</p>


<h3>Value</h3>

<p>A data frame containing the calculated power values and related parameters.
</p>


<h3>References</h3>

<p>Bi et al. (2016) <a href="https://doi.org/10.1186/s12859-016-0994-9">doi:10.1186/s12859-016-0994-9</a>
Love et al. (2014) <a href="https://doi.org/10.1186/s13059-014-0550-8">doi:10.1186/s13059-014-0550-8</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load example files
countDatPath &lt;- system.file("extdata", "exmplCountDat.csv", package = "HEssRNA")
smplDatPath &lt;- system.file("extdata", "exmplSampleDat.csv", package = "HEssRNA")

if (file.exists(countDatPath) &amp;&amp; file.exists(smplDatPath)) {
  countDat &lt;- read.csv(countDatPath)
  smplDat &lt;- read.csv(smplDatPath)

  result &lt;- powerCalc(countDat, smplDat)
  print(result$PowerResults)
} else {
  warning("Example data files not found.")
}


</code></pre>

<hr>
<h2 id='prcesDF4modelInhouse'>Process Data Frame in In-House Format for Model Building</h2><span id='topic+prcesDF4modelInhouse'></span>

<h3>Description</h3>

<p>This function takes a data frame in an in-house format and processes it
to make it in longer format and round the value of the power to 3 digits
for building a model. It reshapes the data from a wide format to a long
format, extracting and manipulating columns related to replicate numbers and power values.
This function is needed when user has a data frame similar to the in-house format.
For the purpose of creating model the user should also have Heritability class and log fold change value too.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prcesDF4modelInhouse(df4modelInhouseFmt)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prcesDF4modelInhouse_+3A_df4modelinhousefmt">df4modelInhouseFmt</code></td>
<td>
<p>A data frame containing the input data in in-house format.
The columns should include replicate columns named starting with &quot;R&quot; (e.g., R1, R2, etc.).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame in long format with columns:
</p>
<table role = "presentation">
<tr><td><code>NoOfReplicates</code></td>
<td>
<p>Numeric representation of the replicate number extracted from column names (R1, R2, etc.).</p>
</td></tr>
<tr><td><code>pwr</code></td>
<td>
<p>Power values rounded to 3 decimal places corresponding to the replicate number.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Example of usage:
df &lt;- data.frame(
  Gene = c("Gene1", "Gene2"),
  R1 = c(0.85, 0.90),
  R2 = c(0.88, 0.91),
  R3 = c(0.83, 0.89)
)
result &lt;- prcesDF4modelInhouse(df)
print(result)

</code></pre>

<hr>
<h2 id='smplSizPred'>Predict Number of Replicates Based on Heritability, Power, and Fold Change</h2><span id='topic+smplSizPred'></span>

<h3>Description</h3>

<p>This function predicts the number of replicates required for a given experiment based on heritability, power, fold change, and tissue type. The model is constructed using the provided data, and the prediction is adjusted based on the selected trait's mean heritability value. The function ensures that the predicted replicates are valid, rounding negative or unrealistic values to sensible minimums based on the heritability class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smplSizPred(
  df4model = df4modelInpt,
  hIndexMeanDFinput = hIndexMeanDF,
  heritabilityClass,
  inptPwr,
  fc,
  trait = NULL,
  tissue = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smplSizPred_+3A_df4model">df4model</code></td>
<td>
<p>A data frame containing the input data for the model. It should include the following columns: <code>NoOfReplicates</code>, <code>HeritabilityValue</code>, <code>pwr</code>, <code>FoldChange</code>, and optionally <code>Tissue</code>.</p>
</td></tr>
<tr><td><code id="smplSizPred_+3A_hindexmeandfinput">hIndexMeanDFinput</code></td>
<td>
<p>A data frame containing the mean heritability values for each trait. It should include at least the columns <code>Trait.name</code> and <code>MeanValue</code>.</p>
</td></tr>
<tr><td><code id="smplSizPred_+3A_heritabilityclass">heritabilityClass</code></td>
<td>
<p>A character string specifying the heritability class used for filtering and adjusting the prediction. Possible values are &quot;low&quot;, &quot;mid&quot;, and &quot;high&quot;.</p>
</td></tr>
<tr><td><code id="smplSizPred_+3A_inptpwr">inptPwr</code></td>
<td>
<p>A numeric value representing the power used in the model.</p>
</td></tr>
<tr><td><code id="smplSizPred_+3A_fc">fc</code></td>
<td>
<p>A numeric value representing the fold change used in the model.</p>
</td></tr>
<tr><td><code id="smplSizPred_+3A_trait">trait</code></td>
<td>
<p>An optional parameter specifying the trait. If provided, the heritability value for the trait will be used to adjust the heritability class values.</p>
</td></tr>
<tr><td><code id="smplSizPred_+3A_tissue">tissue</code></td>
<td>
<p>An optional parameter specifying the tissue type. If provided, the model will include tissue as a factor in the regression. If not provided, tissue is excluded.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value representing the predicted number of replicates. The value is rounded to the nearest whole number and adjusted to ensure it is valid for the selected heritability class.
</p>


<h3>References</h3>

<p>Sun et al. (2017) <a href="https://doi.org/10.1093/nar/gkx204">doi:10.1093/nar/gkx204</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Example usage:
df4modelInpt &lt;- data.frame(
    NoOfReplicates = c(3, 5, 7, 9, 11),
    HeritabilityClass = c("high", "mid", "low", "high", "mid"),
    HeritabilityValue = c(0.5, 0.6, 0.7, 0.5, 0.6),
    pwr = c(0.8, 0.9, 0.85, 0.88, 0.86),
    FoldChange = c(2, 3, 2.5, 2.8, 3.2),
    Tissue = c("Liver", "Liver", "Kidney", "Liver", "Kidney")
)
hIndexMeanDF &lt;- data.frame(Trait.name = c("Trait1", "Trait2"),
                           MeanValue = c(0.3, 0.5))
NoOfReplicatesPred &lt;- smplSizPred(df4model = df4modelInpt,
                      hIndexMeanDFinput = hIndexMeanDF,
                      heritabilityClass = "mid",
                      inptPwr = 0.85,
                      fc = 2.5,
                      trait = "Trait1",
                      tissue = "Liver")
print(NoOfReplicatesPred)


</code></pre>

<hr>
<h2 id='smplSizPredModel'>Generate a Linear Model for Sample Size Prediction</h2><span id='topic+smplSizPredModel'></span>

<h3>Description</h3>

<p>This function generates a linear regression model to predict the number of replicates (<code>NoOfReplicates</code>) based on heritability, power, fold change, and tissue type. The model is generated depending on whether the tissue information is provided in the data. The function returns the fitted model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smplSizPredModel(
  df4model = df4modelInpt,
  heritabilityClass,
  inptPwr,
  fc,
  trait = NULL,
  tissue = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smplSizPredModel_+3A_df4model">df4model</code></td>
<td>
<p>A data frame containing the input data for the model. It should include the following columns: <code>NoOfReplicates</code>, <code>HeritabilityValue</code>, <code>pwr</code>, <code>FoldChange</code>, and optionally, <code>Tissue</code>.</p>
</td></tr>
<tr><td><code id="smplSizPredModel_+3A_heritabilityclass">heritabilityClass</code></td>
<td>
<p>A character value indicating the class of heritability used for filtering the data.</p>
</td></tr>
<tr><td><code id="smplSizPredModel_+3A_inptpwr">inptPwr</code></td>
<td>
<p>A numeric value representing the power used in the model.</p>
</td></tr>
<tr><td><code id="smplSizPredModel_+3A_fc">fc</code></td>
<td>
<p>A numeric value representing the fold change used in the model.</p>
</td></tr>
<tr><td><code id="smplSizPredModel_+3A_trait">trait</code></td>
<td>
<p>An optional parameter to specify the trait. If provided, it can be used for further filtering, but it's not currently used in the function.</p>
</td></tr>
<tr><td><code id="smplSizPredModel_+3A_tissue">tissue</code></td>
<td>
<p>An optional parameter specifying the tissue type. If provided, the model will include the tissue information in the regression. If not provided, the model will exclude tissue information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A linear model object (<code>lm</code> class), which contains the fitted linear regression model for the number of replicates prediction.
</p>


<h3>References</h3>

<p>Sun et al. (2017) <a href="https://doi.org/10.1093/nar/gkx204">doi:10.1093/nar/gkx204</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Example usage:
df4modelInpt &lt;- data.frame(
    NoOfReplicates = c(3, 5, 7, 9, 11),
    HeritabilityClass = c("high", "mid", "low", "high", "mid"),
    HeritabilityValue = c(0.5, 0.6, 0.7, 0.5, 0.6),
    pwr = c(0.8, 0.9, 0.85, 0.88, 0.86),
    FoldChange = c(2, 3, 2.5, 2.8, 3.2),
    Tissue = c("Liver", "Liver", "Kidney", "Liver", "Kidney")
)

# Fit the model
model &lt;- smplSizPredModel(
    df4model = df4modelInpt,
    heritabilityClass = "high",
    inptPwr = 0.8,
    fc = 2,
    tissue = "Liver"
)

# Summarize the results
summary(model)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
