<!DOCTYPE html><html lang="en"><head><title>Help for package HetSeq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HetSeq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#distmat'><p>Calculate scSLAM-seq based distance matrices</p></a></li>
<li><a href='#Hetseq'><p>Heterogeneity-seq wrapper</p></a></li>
<li><a href='#HetseqClassify'><p>Heterogeneity-seq: Classifying cellular response by gene expression values</p></a></li>
<li><a href='#HetseqDoubleML'><p>Heterogeneity-seq: Classifying cellular response by gene expression values including causal inference by DoubleML</p></a></li>
<li><a href='#HetseqTest'><p>Heterogeneity-seq: Testing for differential gene expression</p></a></li>
<li><a href='#mincostflow'><p>Min-Cost-Max-Flow for cellular trajectories</p></a></li>
<li><a href='#PlotClassify'><p>Plot Heterogeneity-seq Classifier Results</p></a></li>
<li><a href='#PlotDoubleML'><p>Plot Heterogeneity-seq DoubleML Results</p></a></li>
<li><a href='#prune'><p>Prune trajectories</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Identifying Modulators of Cellular Responses Leveraging
Intercellular Heterogeneity</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Kevin Berg [aut, cre],
  Florian Erhard <a href="https://orcid.org/0000-0002-3574-6983"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Lygeri Sakellaridi [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kevin Berg &lt;Kevin.Berg@informatik.uni-regensburg.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Cellular responses to perturbations are highly heterogeneous
    and depend largely on the initial state of cells. Connecting
    post-perturbation cells via cellular trajectories to untreated cells
    (e.g. by leveraging metabolic labeling information) enables
    exploitation of intercellular heterogeneity as a combined knock-down
    and overexpression screen to identify pathway modulators, termed
    Heterogeneity-seq (see 'Berg et al' &lt;<a href="https://doi.org/10.1101%2F2024.10.28.620481">doi:10.1101/2024.10.28.620481</a>&gt;).
    This package contains functions to generate cellular trajectories based
    on scSLAM-seq (single-cell, thiol-(SH)-linked alkylation of RNA for
    metabolic labelling sequencing) time courses, functions to identify
    pathway modulators and to visualize the results.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (&ge; 2)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/erhard-lab/HetSeq">https://github.com/erhard-lab/HetSeq</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/erhard-lab/HetSeq/issues">https://github.com/erhard-lab/HetSeq/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>cowplot, doParallel, foreach, DoubleML, e1071, igraph,
ggplot2, ggrastr, ggrepel, grandR, lpSolve, mlr3, pROC,
reshape2, scales, Seurat, stats</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-03 11:36:32 UTC; berg</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-03 18:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='distmat'>Calculate scSLAM-seq based distance matrices</h2><span id='topic+distmat'></span>

<h3>Description</h3>

<p>This function calculates distances matrices between cells of different time points based on metabolic labeling RNA profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distmat(prev.t, next.t, prevAssay, nextAssay, gene_subset = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="distmat_+3A_prev.t">prev.t</code></td>
<td>
<p>Cells to be used from the previous time point in distance matrix calculation.</p>
</td></tr>
<tr><td><code id="distmat_+3A_next.t">next.t</code></td>
<td>
<p>Cells to be used from the next time point in distance matrix calculation.</p>
</td></tr>
<tr><td><code id="distmat_+3A_prevassay">prevAssay</code></td>
<td>
<p>Name of the expression assay of cells from the previous time point.</p>
</td></tr>
<tr><td><code id="distmat_+3A_nextassay">nextAssay</code></td>
<td>
<p>Name of the expression assay of cells from the next time point.</p>
</td></tr>
<tr><td><code id="distmat_+3A_gene_subset">gene_subset</code></td>
<td>
<p>Set a subset of genes on which trajectories should be calculated. Other genes will be disregarded.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Distance matrix between cells from two time points.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Full vignette available on https://grandr.erhard-lab.de/articles/web/hetseq.html

  obj.list &lt;- SplitObject(seuratObject, split.by = "time")
  D.list=list(
   distmat(treatment.list[["0h"]],treatment.list[["2h"]], "RNA", "prevRNA"),
   distmat(treatment.list[["2h"]],treatment.list[["4h"]], "RNA", "prevRNA")
  )

</code></pre>

<hr>
<h2 id='Hetseq'>Heterogeneity-seq wrapper</h2><span id='topic+Hetseq'></span>

<h3>Description</h3>

<p>Wrapper function for all Heterogeneity-seq functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Hetseq(method = c("test", "classify", "DoubleML"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Hetseq_+3A_method">method</code></td>
<td>
<p>The method to run Heterogeneity-seq. Calls hetseq.test, hetseq.classify or hetseq.DoubleML.</p>
</td></tr>
<tr><td><code id="Hetseq_+3A_...">...</code></td>
<td>
<p>Parameters given to the chosen Hetseq method. See respective help pages.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Heterogeneity-seq uses intercellular heterogeneity to identify modulators of cellular response to perturbations. Three approaches to identify these factors are currently available:
</p>
<p>- test: Differential Gene Expression testing between cells from two response groups.
</p>
<p>- classify: Predicting cellular outcome by expression of single genes (+informative features) reveals genes with strong predictive capabilities (high AUC) as potential pathway modulators.
</p>
<p>- DoubleML: A strict predictor that utilizes Causal Inference (DoubleML) to distinguish causal factors from simply correlating genes. Genes with high estimated effects on the outcome and high significance are identified as potential pathway modulators.
</p>


<h3>Value</h3>

<p>Table of Heterogeneity-seq results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Full vignette available on https://grandr.erhard-lab.de/articles/web/hetseq.html

  tab &lt;- Hetseq(method="classify", data, trajectories, score.name = "score")

</code></pre>

<hr>
<h2 id='HetseqClassify'>Heterogeneity-seq: Classifying cellular response by gene expression values</h2><span id='topic+HetseqClassify'></span>

<h3>Description</h3>

<p>Classifying the cellular response of control cells using single gene expression (+ informative features) to identify features with the strongest predictive capabilities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HetseqClassify(
  object,
  trajectories,
  score.group = NULL,
  score.name = NULL,
  quantiles = c(0.25, 0.75),
  compareGroups = c("Low", "High"),
  posClass = NULL,
  basefeatures = NULL,
  genes = NULL,
  assay = NULL,
  split = NULL,
  kernel = "radial",
  cross = 10,
  num_cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HetseqClassify_+3A_object">object</code></td>
<td>
<p>Seurat object</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_trajectories">trajectories</code></td>
<td>
<p>Matrix of cell-cell trajectories. Columns represent time points, rows represent trajectories of connected cells over time points.</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_score.group">score.group</code></td>
<td>
<p>A named vector of response groups. Names represent cells, the values represent the score groups. If no score.group is set, use score.name and quantiles parameters must be set to define score groups.</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_score.name">score.name</code></td>
<td>
<p>The name of a numeric Seurat meta data column, which will be used to calculate score groups. Only used if no score.group is given.</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_quantiles">quantiles</code></td>
<td>
<p>Thresholds of the score.name meta data to define 3 response groups. Low, Middle, High.</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_comparegroups">compareGroups</code></td>
<td>
<p>Which score groups to test. Default: Low vs. High</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_posclass">posClass</code></td>
<td>
<p>Define the positive Class for classification.</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_basefeatures">basefeatures</code></td>
<td>
<p>Additional informative features to include in the classification. Must be meta data available in the Seurat object.</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_genes">genes</code></td>
<td>
<p>Vector of genes to test.</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_assay">assay</code></td>
<td>
<p>The name of the Seurat assay to perform Heterogeneity-seq on. If NULL, the default assay will be used.</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_split">split</code></td>
<td>
<p>Set a training-test data split. Must be in [0,1]</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_kernel">kernel</code></td>
<td>
<p>The kernel for the SVM. linear, polynomial, radial or sigmoid. Default: radial.</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_cross">cross</code></td>
<td>
<p>Number of cross-validations.</p>
</td></tr>
<tr><td><code id="HetseqClassify_+3A_num_cores">num_cores</code></td>
<td>
<p>The number of cores used in parallel processing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table of log2FC and AUC values for each gene and an additional AUC value for the baseline features.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Full vignette available on https://grandr.erhard-lab.de/articles/web/hetseq.html

  t &lt;- HetseqClassify(data, trajectories, score.name = "score")
  
  t &lt;- HetseqClassify(data, trajectories, score.group = groups,
          compareGroups = c("Weak", "Strong"))

</code></pre>

<hr>
<h2 id='HetseqDoubleML'>Heterogeneity-seq: Classifying cellular response by gene expression values including causal inference by DoubleML</h2><span id='topic+HetseqDoubleML'></span>

<h3>Description</h3>

<p>Classifying the cellular response of control cells using single gene expression (+ informative features) to identify features with the strongest predictive capabilities and applying causal inference by a DoubleML approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HetseqDoubleML(
  object,
  trajectories,
  score.group = NULL,
  score.name = NULL,
  quantiles = c(0.25, 0.75),
  compareGroups = c("Low", "High"),
  posClass = NULL,
  basefeatures = NULL,
  genes = NULL,
  background = NULL,
  assay = NULL,
  split = NULL,
  cross = 10,
  num_cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HetseqDoubleML_+3A_object">object</code></td>
<td>
<p>Seurat object</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_trajectories">trajectories</code></td>
<td>
<p>Matrix of cell-cell trajectories. Columns represent time points, rows represent trajectories of connected cells over time points.</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_score.group">score.group</code></td>
<td>
<p>A named vector of response groups. Names represent cells, the values represent the score groups. If no score.group is set, use score.name and quantiles parameters must be set to define score groups.</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_score.name">score.name</code></td>
<td>
<p>The name of a numeric Seurat meta data column, which will be used to calculate score groups. Only used if no score.group is given.</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_quantiles">quantiles</code></td>
<td>
<p>Thresholds of the score.name meta data to define 3 response groups. Low, Middle, High.</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_comparegroups">compareGroups</code></td>
<td>
<p>Which score groups to test. Default: Low vs. High</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_posclass">posClass</code></td>
<td>
<p>Define the positive Class for classification.</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_basefeatures">basefeatures</code></td>
<td>
<p>Additional informative features to include in the classification. Must be meta data available in the Seurat object.</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_genes">genes</code></td>
<td>
<p>Vector of genes to test.</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_background">background</code></td>
<td>
<p>A set of genes that will be considered as potential confounding factors in the DoubleML analysis. Must contain all genes set in the genes parameter. By default, all genes are used.</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_assay">assay</code></td>
<td>
<p>The name of the Seurat assay to perform Heterogeneity-seq on. If NULL, the default assay will be used.</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_split">split</code></td>
<td>
<p>Set a training-test data split. Must be in [0,1]</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_cross">cross</code></td>
<td>
<p>Number of cross-validations.</p>
</td></tr>
<tr><td><code id="HetseqDoubleML_+3A_num_cores">num_cores</code></td>
<td>
<p>The number of cores used in parallel processing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table of log2FC and AUC values for each gene and an additional AUC value for the baseline features.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Full vignette available on https://grandr.erhard-lab.de/articles/web/hetseq.html

  t &lt;- HetseqDoubleML(data, trajectories, score.name = "score")
  
  t &lt;- HetseqDoubleML(data, trajectories, score.group = group_vector,
        compareGroups = c("Weak", "Strong"))

</code></pre>

<hr>
<h2 id='HetseqTest'>Heterogeneity-seq: Testing for differential gene expression</h2><span id='topic+HetseqTest'></span>

<h3>Description</h3>

<p>Testing for differential gene expression in predecessors of treated cells
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HetseqTest(mat, A, B)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HetseqTest_+3A_mat">mat</code></td>
<td>
<p>Gene expression matrix of control cells</p>
</td></tr>
<tr><td><code id="HetseqTest_+3A_a">A</code></td>
<td>
<p>Vector of cells (columns) in positive class</p>
</td></tr>
<tr><td><code id="HetseqTest_+3A_b">B</code></td>
<td>
<p>Vector of cells (columns) in negative class</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table of log2FC, p-values and adjusted p-values of differentially expressed genes.
</p>

<hr>
<h2 id='mincostflow'>Min-Cost-Max-Flow for cellular trajectories</h2><span id='topic+mincostflow'></span>

<h3>Description</h3>

<p>Applies Min-Cost-Max-Flow to calculate optimal trajectories from distance matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mincostflow(D.list, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mincostflow_+3A_d.list">D.list</code></td>
<td>
<p>List of (pruned) distance matrices. Can be generated by the distmat function.</p>
</td></tr>
<tr><td><code id="mincostflow_+3A_verbose">verbose</code></td>
<td>
<p>Show verbose output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of cell-cell trajectories spanning all given timepoints.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Full vignette available on https://grandr.erhard-lab.de/articles/web/hetseq.html

  mincostflow(D.list)

</code></pre>

<hr>
<h2 id='PlotClassify'>Plot Heterogeneity-seq Classifier Results</h2><span id='topic+PlotClassify'></span>

<h3>Description</h3>

<p>This plotting functions creates AUC Scatter plots to visualize Classifier Results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotClassify(
  table,
  highlights = NULL,
  highlights.color = NULL,
  auc.cutoff = NULL,
  plot.baseline = TRUE,
  density.color = TRUE,
  density.n = 500,
  point.scale = 0.5,
  xlab = "AUC",
  ylab = bquote(log[2] ~ FC ~ (`0h`)),
  linetype = "dashed"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotClassify_+3A_table">table</code></td>
<td>
<p>Table from HetSeq using the Classifier method.</p>
</td></tr>
<tr><td><code id="PlotClassify_+3A_highlights">highlights</code></td>
<td>
<p>A vector of genes to highlight in the plot.</p>
</td></tr>
<tr><td><code id="PlotClassify_+3A_highlights.color">highlights.color</code></td>
<td>
<p>A vector of colors for gene highlights.</p>
</td></tr>
<tr><td><code id="PlotClassify_+3A_auc.cutoff">auc.cutoff</code></td>
<td>
<p>Inserts a vertical line at the cutoff AUC value.</p>
</td></tr>
<tr><td><code id="PlotClassify_+3A_plot.baseline">plot.baseline</code></td>
<td>
<p>Inserts a vertical line at the baseline AUC value (= AUC of the classifier with basefeatures but no further gene info).</p>
</td></tr>
<tr><td><code id="PlotClassify_+3A_density.color">density.color</code></td>
<td>
<p>Color data points by density. Default is TRUE.</p>
</td></tr>
<tr><td><code id="PlotClassify_+3A_density.n">density.n</code></td>
<td>
<p>Set granularity of 2d density color.</p>
</td></tr>
<tr><td><code id="PlotClassify_+3A_point.scale">point.scale</code></td>
<td>
<p>Set point size.</p>
</td></tr>
<tr><td><code id="PlotClassify_+3A_xlab">xlab</code></td>
<td>
<p>Set label of the x-axis.</p>
</td></tr>
<tr><td><code id="PlotClassify_+3A_ylab">ylab</code></td>
<td>
<p>Set label of the y-axis.</p>
</td></tr>
<tr><td><code id="PlotClassify_+3A_linetype">linetype</code></td>
<td>
<p>Set the linetype of the baseline AUC line.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  tab &lt;- HetseqClassify(data, trajectories, score.name = "score")
  PlotClassify(tab, highlights=c("MYC", "GAPDH", "ISG15"))

</code></pre>

<hr>
<h2 id='PlotDoubleML'>Plot Heterogeneity-seq DoubleML Results</h2><span id='topic+PlotDoubleML'></span>

<h3>Description</h3>

<p>This plotting functions creates a Vulcano Plot to visualize DoubleML Results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotDoubleML(
  table,
  highlights = NULL,
  p.cutoff = 0.05,
  est.cutoff = NULL,
  highlights.color = NULL,
  label.repulsion = 1,
  density.color = TRUE,
  density.n = 500,
  point.scale = 0.5,
  xlab = "Estimate",
  ylab = bquote("-" ~ log[10] ~ FDR),
  linetype = "dashed"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotDoubleML_+3A_table">table</code></td>
<td>
<p>Table from the Hetseq using the doubleML method.</p>
</td></tr>
<tr><td><code id="PlotDoubleML_+3A_highlights">highlights</code></td>
<td>
<p>A vector of genes to highlight in the plot.</p>
</td></tr>
<tr><td><code id="PlotDoubleML_+3A_p.cutoff">p.cutoff</code></td>
<td>
<p>Adds a dashed horizontal line at the given adjusted p-value cutoff.</p>
</td></tr>
<tr><td><code id="PlotDoubleML_+3A_est.cutoff">est.cutoff</code></td>
<td>
<p>Adds two dashed vertical lines (+/-) at the given estimate cutoff.</p>
</td></tr>
<tr><td><code id="PlotDoubleML_+3A_highlights.color">highlights.color</code></td>
<td>
<p>A vector of colors for gene highlights.</p>
</td></tr>
<tr><td><code id="PlotDoubleML_+3A_label.repulsion">label.repulsion</code></td>
<td>
<p>Represents the force parameter of the ggrepel::geom_label_repel() function. Higher values reduce label overlap.</p>
</td></tr>
<tr><td><code id="PlotDoubleML_+3A_density.color">density.color</code></td>
<td>
<p>Color data points by density. Default is TRUE.</p>
</td></tr>
<tr><td><code id="PlotDoubleML_+3A_density.n">density.n</code></td>
<td>
<p>Set granularity of 2d density color.</p>
</td></tr>
<tr><td><code id="PlotDoubleML_+3A_point.scale">point.scale</code></td>
<td>
<p>Set point size.</p>
</td></tr>
<tr><td><code id="PlotDoubleML_+3A_xlab">xlab</code></td>
<td>
<p>Set label of the x-axis.</p>
</td></tr>
<tr><td><code id="PlotDoubleML_+3A_ylab">ylab</code></td>
<td>
<p>Set label of the y-axis.</p>
</td></tr>
<tr><td><code id="PlotDoubleML_+3A_linetype">linetype</code></td>
<td>
<p>Set the linetype of the p-value and estimate cutoff line.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  tab &lt;- HetseqDoubleML(data, trajectories, score.name = "score")
  PlotDoubleML(tab, highlights=c("MYC", "GAPDH", "ISG15"))

</code></pre>

<hr>
<h2 id='prune'>Prune trajectories</h2><span id='topic+prune'></span>

<h3>Description</h3>

<p>Prune trajectories down to top.n candidates to reduce runtime of subsequent mincostflow function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prune(D.list, top.n = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prune_+3A_d.list">D.list</code></td>
<td>
<p>List of distance matrices. Can be generated by the distmat function.</p>
</td></tr>
<tr><td><code id="prune_+3A_top.n">top.n</code></td>
<td>
<p>Prune trajectories to only top.n possible connections to optimize subsequent application of the mincostflow function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Pruned distance matrix between cells from multiple timepoints.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Full vignette available on https://grandr.erhard-lab.de/articles/web/hetseq.html

  prune(D.list, top.n = 10)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
