<!DOCTYPE html><html lang="en"><head><title>Help for package HextractoR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HextractoR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HextractoR'><p>HextractoR: Integrated Tool for Hairping Extraction of RNA Sequences</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Integrated Tool for Hairping Extraction of RNA Sequences</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Author:</td>
<td>Cristian Yones</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Cristian Yones &lt;cyones@sinc.unl.edu.ar&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Simple and integrated tool that automatically extracts and folds all hairpin sequences from raw genome-wide data. It predicts the secondary structure of several overlapped segments, with longer length than the mean length of sequences of interest for the species under processing, ensuring that no one is lost nor inappropriately cut.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License 2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>seqinr, parallel, doParallel, foreach,</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-08-06 16:12:43 UTC; cyones</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-08-06 16:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='HextractoR'>HextractoR: Integrated Tool for Hairping Extraction of RNA Sequences</h2><span id='topic+HextractoR'></span>

<h3>Description</h3>

<p>To preprocess a genome, you need a file containing the raw genome in fasta
format. To run HExtractor, simply call the main function. This function
creates 2 files in the &quot;out&quot; folder and automatically names them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HextractoR(input_file, min_valid_nucleotides = 500, window_size = 160,
  window_step = 30, only_sloop = T, min_length = 60, min_bp = 16,
  trim_sequences = T, margin_bp = 6, blast_evalue = 1,
  identity_threshold = 90, nthreads = 4, nworks = 4,
  filter_files = { })
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HextractoR_+3A_input_file">input_file</code></td>
<td>
<p>filename of the fasta file to proccess</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_min_valid_nucleotides">min_valid_nucleotides</code></td>
<td>
<p>Each input sequence must have this quantity of
valid nucleotides (not 'N') to be processed.</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_window_size">window_size</code></td>
<td>
<p>Number of bases in the windows.</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_window_step">window_step</code></td>
<td>
<p>Window step. This number defines indirectly the overlap:
window_overlap=window_size-window_step</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_only_sloop">only_sloop</code></td>
<td>
<p>Only extract single loop sequence.</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_min_length">min_length</code></td>
<td>
<p>Minimum sequence length. Shorter sequences are discarded.</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_min_bp">min_bp</code></td>
<td>
<p>Minimum number of base-pairs that must form a sequence.</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_trim_sequences">trim_sequences</code></td>
<td>
<p>Use some heuristics to trim the hairpins.</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_margin_bp">margin_bp</code></td>
<td>
<p>When the sequence is trimmed, at least min_bp+margin_bp
base-pairs are left.</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_blast_evalue">blast_evalue</code></td>
<td>
<p>e-value used in blast to match the extracted sequences
with the sequences from the filter files.</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_identity_threshold">identity_threshold</code></td>
<td>
<p>Identity threshold used to match sequences with the
sequences from the filter files.</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_nthreads">nthreads</code></td>
<td>
<p>Allows using more than one thread in the execution.</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_nworks">nworks</code></td>
<td>
<p>Split each sequence in nworks to use less RAM memory.</p>
</td></tr>
<tr><td><code id="HextractoR_+3A_filter_files">filter_files</code></td>
<td>
<p>Fasta files with known sequences to separate the output
stems.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the path of the output files and the result of the
proccessing of each sequence (if it was succesful or failed)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Small example without filter files
library(HextractoR)
# First we get the path of the example FASTA file
fpath &lt;- system.file("Example_tiny.fasta", package="HextractoR")
# To run HextractoR, simply call the main function
HextractoR(input_file = fpath)
# Other example with filter files and bigger input file
fpath1 &lt;- system.file("Example_human.fasta", package="HextractoR")
fpath2 &lt;- system.file("Example_pre-miRNA.fasta", package="HextractoR")
HextractoR(input_file = fpath1, filter_files = {fpath2})
# This function creates 2 files in the working directory and automatically
# names them.

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
