<!DOCTYPE html><html lang="en"><head><title>Help for package HGNChelper</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HGNChelper}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HGNChelper-package'><p>HGNChelper: Identify and Correct Invalid HGNC Human Gene Symbols and MGI Mouse Gene Symbols</p></a></li>
<li><a href='#affyToR'><p>Title function to convert Affymetrix probeset identifiers to valid R names</p></a></li>
<li><a href='#checkGeneSymbols'><p>Identify outdated or Excel-mogrified gene symbols</p></a></li>
<li><a href='#findExcelGeneSymbols'><p>Title function to identify Excel-mogrified gene symbols</p></a></li>
<li><a href='#getCurrentMaps'><p>Get the current maps for correcting gene symbols</p></a></li>
<li><a href='#hgnc.table'><p>All current and withdrawn HGNC gene symbols and Excel-mogrified symbols</p></a></li>
<li><a href='#mouse.table'><p>All current and withdrawn MGI mouse symbols and Excel-mogrified symbols</p></a></li>
<li><a href='#rToAffy'><p>Title function to convert the output of affyToR back to the original Affymetrix probeset identifiers.</p></a></li>
<li><a href='#rToSymbol'><p>Title function to reverse the conversion made by symbolToR</p></a></li>
<li><a href='#symbolToR'><p>Title function to *reversibly* convert HGNC gene symbols to valid R names.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Maintainer:</td>
<td>Levi Waldron &lt;lwaldron.research@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0), methods, utils</td>
</tr>
<tr>
<td>Version:</td>
<td>0.8.15</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-11-16</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2.0)]</td>
</tr>
<tr>
<td>Title:</td>
<td>Identify and Correct Invalid HGNC Human Gene Symbols and MGI
Mouse Gene Symbols</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains functions for
 identifying and correcting HGNC human gene symbols and MGI mouse gene symbols 
 which have been converted to date format by Excel, withdrawn, or aliased.
 Also contains functions for reversibly converting between HGNC
 symbols and valid R names.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/waldronlab/HGNChelper">https://github.com/waldronlab/HGNChelper</a>,
<a href="https://waldronlab.io/HGNChelper/">https://waldronlab.io/HGNChelper/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/waldronlab/HGNChelper/issues">https://github.com/waldronlab/HGNChelper/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>splitstackshape</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, markdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-16 16:08:43 UTC; Levi</td>
</tr>
<tr>
<td>Author:</td>
<td>Sehyun Oh [aut],
  Ayush Aggarwal [aut],
  Markus Riester [aut],
  Levi Waldron [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-16 17:10:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='HGNChelper-package'>HGNChelper: Identify and Correct Invalid HGNC Human Gene Symbols and MGI Mouse Gene Symbols</h2><span id='topic+HGNChelper'></span><span id='topic+HGNChelper-package'></span>

<h3>Description</h3>

<p>Contains functions for identifying and correcting HGNC human gene symbols and MGI mouse gene symbols which have been converted to date format by Excel, withdrawn, or aliased. Also contains functions for reversibly converting between HGNC symbols and valid R names.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Levi Waldron <a href="mailto:lwaldron.research@gmail.com">lwaldron.research@gmail.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Sehyun Oh <a href="mailto:sehyun.oh@sph.cuny.edu">sehyun.oh@sph.cuny.edu</a>
</p>
</li>
<li><p> Ayush Aggarwal <a href="mailto:ayush.aggarwal@igib.in">ayush.aggarwal@igib.in</a>
</p>
</li>
<li><p> Markus Riester <a href="mailto:Markus.riester@novartis.com">Markus.riester@novartis.com</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/waldronlab/HGNChelper">https://github.com/waldronlab/HGNChelper</a>
</p>
</li>
<li> <p><a href="https://waldronlab.io/HGNChelper/">https://waldronlab.io/HGNChelper/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/waldronlab/HGNChelper/issues">https://github.com/waldronlab/HGNChelper/issues</a>
</p>
</li></ul>


<hr>
<h2 id='affyToR'>Title function to convert Affymetrix probeset identifiers to valid R names</h2><span id='topic+affyToR'></span>

<h3>Description</h3>

<p>This function simply prepends &quot;affy.&quot; to the probeset IDs to create
valid R names.  Reverse operation is done by the <a href="#topic+rToAffy">rToAffy</a>
function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>affyToR(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="affyToR_+3A_x">x</code></td>
<td>
<p>vector of Affymetrix probeset identifiers, or any identifier which may with a digit.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector that is simply x with &quot;affy.&quot; prepended to each value.
</p>

<hr>
<h2 id='checkGeneSymbols'>Identify outdated or Excel-mogrified gene symbols</h2><span id='topic+checkGeneSymbols'></span>

<h3>Description</h3>

<p>This function identifies gene symbols which are outdated or may 
have been mogrified by Excel or other spreadsheet programs. If output is assigned 
to a variable, it returns a data.frame of the same number of rows as the input, 
with a second column indicating whether the symbols are valid and a third column 
with a corrected gene list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkGeneSymbols(
  x,
  chromosome = NULL,
  unmapped.as.na = TRUE,
  map = NULL,
  species = "human",
  expand.ambiguous = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkGeneSymbols_+3A_x">x</code></td>
<td>
<p>A character vector of gene symbols to check for modified or outdated values</p>
</td></tr>
<tr><td><code id="checkGeneSymbols_+3A_chromosome">chromosome</code></td>
<td>
<p>An optional integer vector containing the chromosome number of each gene
provided through the argument <code>x</code>. It should be the 
same length as the input for <code>x</code>. Currently, this argument is implemented
only for human gene cases.</p>
</td></tr>
<tr><td><code id="checkGeneSymbols_+3A_unmapped.as.na">unmapped.as.na</code></td>
<td>
<p>If <code>TRUE</code> (default), unmapped symbols will appear as 
NA in the <code>Suggested.Symbol</code> column. If <code>FALSE</code>, the original unmapped 
symbol will be kept.</p>
</td></tr>
<tr><td><code id="checkGeneSymbols_+3A_map">map</code></td>
<td>
<p>Specify if you do not want to use the default maps provided by setting 
species equal to &quot;mouse&quot; or &quot;human&quot;. Map can be any other data.frame with colnames 
identical to <code>c("Symbol", "Approved.Symbol")</code>. The default maps can be updated 
by running the interactive example below.</p>
</td></tr>
<tr><td><code id="checkGeneSymbols_+3A_species">species</code></td>
<td>
<p>A character vector of length 1, either &quot;human&quot; (default) or &quot;mouse&quot;. 
If <code>NULL</code>, or anything other than &quot;human&quot; or &quot;mouse&quot;, then the map argument 
must be provided.</p>
</td></tr>
<tr><td><code id="checkGeneSymbols_+3A_expand.ambiguous">expand.ambiguous</code></td>
<td>
<p>If <code>FALSE</code> (default), genes with multiple mapping will only 
map to its approved symbol as the correct one. If <code>TRUE</code>, genes with multiple/ambiguous mapping
will map to all the symbols linked to it.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function will return a data.frame of the same number of rows as the 
input, with corrections possible from map.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mouse.table">mouse.table</a></code> for the mouse lookup table, <code><a href="#topic+hgnc.table">hgnc.table</a></code> for the human lookup table
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(HGNChelper)

## Human
human &lt;- c("FN1", "TP53", "UNKNOWNGENE","7-Sep", "9/7", "1-Mar", "Oct4", "4-Oct",
      "OCT4-PG4", "C19ORF71", "C19orf71")
checkGeneSymbols(human)

## Mouse
mouse &lt;- c("1-Feb", "Pzp", "A2m")
checkGeneSymbols(mouse, species="mouse")

## expand.ambiguous

## Human
human &lt;- "AAVS1"
checkGeneSymbols(human, expand.ambiguous=FALSE)
checkGeneSymbols(human, expand.ambiguous=TRUE)

## Mouse
mouse &lt;- c("Cpamd8", "Mug2")
checkGeneSymbols(mouse, species = "mouse", expand.ambiguous = FALSE)
checkGeneSymbols(mouse, species = "mouse", expand.ambiguous = TRUE)

## Updating the map
if (interactive()) {
    currentHumanMap &lt;- getCurrentHumanMap()
    checkGeneSymbols(human, map=currentHumanMap)

    # You should save this if you are going to use it multiple times,   
    # then load it from file rather than burdening HGNC's servers.
    save(hgnc.table, file="hgnc.table.rda", compress="bzip2")
    load("hgnc.table.rda")
    checkGeneSymbols(human, map=hgnc.table)
}
  
</code></pre>

<hr>
<h2 id='findExcelGeneSymbols'>Title function to identify Excel-mogrified gene symbols</h2><span id='topic+findExcelGeneSymbols'></span>

<h3>Description</h3>

<p>This function identifies gene symbols which may have been
mogrified by Excel or other spreadsheet programs.  If output is
assigned to a variable, it returns a vector of the same length
where symbols which could be mapped have been mapped.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findExcelGeneSymbols(
  x,
  mog.map = read.csv(system.file("extdata/mog_map.csv", package = "HGNChelper"), as.is =
    TRUE),
  regex = "impossibletomatch^"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findExcelGeneSymbols_+3A_x">x</code></td>
<td>
<p>Vector of gene symbols to check for mogrified values</p>
</td></tr>
<tr><td><code id="findExcelGeneSymbols_+3A_mog.map">mog.map</code></td>
<td>
<p>Map of known mogrifications.  This should be
a dataframe with two columns: original and mogrified, containing
the correct and incorrect symbols, respectively.</p>
</td></tr>
<tr><td><code id="findExcelGeneSymbols_+3A_regex">regex</code></td>
<td>
<p>Regular expression, recognized by the base::grep function which
is called with ignore.case=TRUE, to identify mogrified symbols.
The default regex will not match anything. The regex in the
examples is an attempt to match all Excel-mogrified HGNC human gene symbols.
It is not necessary for all matches to have a corresponding entry
in mog.map$mogrified; values in x which are matched by this regex
but are not found in mog.map$mogrified simply will not be
corrected.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if the return value of the function is assigned to a variable, the
function will return a vector of the same length as the input,
with corrections possible from mog.map made.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Available maps from this package:
human &lt;- read.csv(system.file("extdata/mog_map.csv", 
                              package = "HGNChelper"), as.is=TRUE)
mouse &lt;- read.csv(system.file("extdata/HGNChelper_mog_map_MGI_AMC_2016_03_30.csv", 
                              package = "HGNChelper"), as.is=TRUE)
## This regex is based that provided by Zeeberg et al.,
##  Mistaken Identifiers: Gene name errors can be introduced
## inadvertently when using Excel in bioinformatics.  BMC
##  Bioinformatics 2004, 5:80.
re &lt;- "[0-9]\\-(JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC)|[0-9]\\.[0-9][0-9]E\\+[[0-9][0-9]"
findExcelGeneSymbols(c("2-Apr", "APR2"), mog.map=human, regex=re)
findExcelGeneSymbols(c("1-Feb", "Feb1"), mog.map=mouse)
</code></pre>

<hr>
<h2 id='getCurrentMaps'>Get the current maps for correcting gene symbols</h2><span id='topic+getCurrentMaps'></span><span id='topic+getCurrentHumanMap'></span><span id='topic+getCurrentMouseMap'></span>

<h3>Description</h3>

<p>Valid human and mouse gene symbols can be updated frequently. Use
these functions to get the most current lists of valid symbols, which you can 
then use as an input to the <code>map</code> argument of <code>checkGeneSymbols</code>. 
Make sure to change the default <code>species="human"</code> argument to <code>checkGeneSymbols</code> 
if you are doing this for mouse. Use <code>getCurrentHumanMap</code> for HGNC human gene 
symbols from <a href="https://www.genenames.org/">https://www.genenames.org/</a> and <code>getCurrentMouseMap</code> for MGI mouse gene 
symbols from <a href="https://www.informatics.jax.org/downloads/reports/MGI_EntrezGene.rpt">https://www.informatics.jax.org/downloads/reports/MGI_EntrezGene.rpt</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCurrentHumanMap()
getCurrentMouseMap()
</code></pre>


<h3>Value</h3>

<p>A <code>data.frame</code> that can be used for <code>map</code> argument of <code>checkGeneSymbols</code> function
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## human
new.hgnc.table &lt;- getCurrentHumanMap()
checkGeneSymbols(c("3-Oct", "10-3", "tp53"), map=new.hgnc.table)

## mouse
new.mouse.table &lt;- getCurrentMouseMap()
## Set species to NULL or "mouse" 
checkGeneSymbols(c("Gm46568", "1-Feb"), map=new.mouse.table, species="mouse")

## End(Not run)

</code></pre>

<hr>
<h2 id='hgnc.table'>All current and withdrawn HGNC gene symbols and Excel-mogrified symbols</h2><span id='topic+hgnc.table'></span>

<h3>Description</h3>

<p>A <code>data.frame</code> with the first column providing a gene symbol or 
known alias (including withdrawn symbols), second column providing the approved
HGNC human gene symbol.
</p>

<ul>
<li> <p><code>Symbol</code>: All valid, Excel-mogrified, and withdrawn symbols
</p>
</li>
<li> <p><code>Approved.Symbol</code>: Approved symbols
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>hgnc.table
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 103939 rows and 3 columns.
</p>


<h3>Source</h3>

<p>Extracted from <a href="https://storage.googleapis.com/public-download-files/hgnc/tsv/tsv/hgnc_complete_set.txt">https://storage.googleapis.com/public-download-files/hgnc/tsv/tsv/hgnc_complete_set.txt</a>
and system.file(&quot;extdata/mog_map.csv&quot;, package=&quot;HGNChelper&quot;)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("hgnc.table", package="HGNChelper")
head(hgnc.table)

</code></pre>

<hr>
<h2 id='mouse.table'>All current and withdrawn MGI mouse symbols and Excel-mogrified symbols</h2><span id='topic+mouse.table'></span>

<h3>Description</h3>

<p>A <code>data.frame</code> with the first column providing a gene symbol or 
known alias (including withdrawn symbols), second column providing the approved 
MGI mouse gene symbol.
</p>

<ul>
<li> <p><code>Symbol</code>: All valid, Excel-mogrified, and withdrawn symbols
</p>
</li>
<li> <p><code>Approved.Symbol</code>: Approved symbols
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>mouse.table
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 790110 rows and 2 columns.
</p>


<h3>Source</h3>

<p>Extracted from <a href="http://www.informatics.jax.org/downloads/reports/MGI_EntrezGene.rpt">http://www.informatics.jax.org/downloads/reports/MGI_EntrezGene.rpt</a> and 
system.file(&quot;extdata/HGNChelper_mog_map_MGI_AMC_2016_03_30.csv&quot;, package=&quot;HGNChelper&quot;)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("mouse.table", package="HGNChelper")
head(mouse.table)

</code></pre>

<hr>
<h2 id='rToAffy'>Title function to convert the output of affyToR back to the original Affymetrix probeset identifiers.</h2><span id='topic+rToAffy'></span>

<h3>Description</h3>

<p>This function simply strips the &quot;affy.&quot; added by the <a href="#topic+affyToR">affyToR</a> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rToAffy(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rToAffy_+3A_x">x</code></td>
<td>
<p>the character vector returned by the affyToR function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of Affymetrix probeset identifiers.
</p>

<hr>
<h2 id='rToSymbol'>Title function to reverse the conversion made by symbolToR</h2><span id='topic+rToSymbol'></span>

<h3>Description</h3>

<p>This function reverses the actions of the symbolToR function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rToSymbol(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rToSymbol_+3A_x">x</code></td>
<td>
<p>the character vector returned by the symbolToR function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of HGNC gene symbols, which are not in general valid R names.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+symbolToR">symbolToR</a></code>
</p>

<hr>
<h2 id='symbolToR'>Title function to *reversibly* convert HGNC gene symbols to valid R names.</h2><span id='topic+symbolToR'></span>

<h3>Description</h3>

<p>This function reversibly converts HGNC gene symbols to valid R
names by prepending &quot;symbol.&quot;, and making the following
substitutions: &quot;-&quot; to &quot;hyphen&quot;, &quot;@&quot; to &quot;ampersand&quot;, and &quot;/&quot; to
&quot;forwardslash&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symbolToR(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="symbolToR_+3A_x">x</code></td>
<td>
<p>vector of HGNC symbols</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of valid R names, of the same length as x, which can be
converted to the same HGNC symbols using the rToSymbol function.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rToSymbol">rToSymbol</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
