<!DOCTYPE html><html><head><title>Help for package hibayes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hibayes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ibrm'><p>Bayes model</p></a></li>
<li><a href='#ldmat'><p>LD variance-covariance matrix calculation</p></a></li>
<li><a href='#read_plink'><p>data load</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#sbrm'><p>SBayes model</p></a></li>
<li><a href='#ssbrm'><p>Single-step Bayes model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Individual-Level, Summary-Level and Single-Step Bayesian
Regression Model</td>
</tr>
<tr>
<td>Version:</td>
<td>3.0.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-20</td>
</tr>
<tr>
<td>Description:</td>
<td>A user-friendly tool to fit Bayesian regression models. It can fit 3 types of Bayesian models using individual-level, summary-level, and individual plus pedigree-level (single-step) data for both Genomic prediction/selection (GS) and Genome-Wide Association Study (GWAS), it was designed to estimate joint effects and genetic parameters for a complex trait, including:
    (1) fixed effects and coefficients of covariates,
    (2) environmental random effects, and its corresponding variance, 
    (3) genetic variance, 
    (4) residual variance, 
    (5) heritability, 
    (6) genomic estimated breeding values (GEBV) for both genotyped and non-genotyped individuals,
    (7) SNP effect size, 
    (8) phenotype/genetic variance explained (PVE) for single or multiple SNPs, 
    (9) posterior probability of association of the genomic window (WPPA), 
    (10) posterior inclusive probability (PIP). 
    The functions are not limited, we will keep on going in enriching it with more features. 
    References: Meuwissen et al. (2001) &lt;<a href="https://doi.org/10.1093%2Fgenetics%2F157.4.1819">doi:10.1093/genetics/157.4.1819</a>&gt;; Gustavo et al. (2013) &lt;<a href="https://doi.org/10.1534%2Fgenetics.112.143313">doi:10.1534/genetics.112.143313</a>&gt;; Habier et al. (2011) &lt;<a href="https://doi.org/10.1186%2F1471-2105-12-186">doi:10.1186/1471-2105-12-186</a>&gt;; Yi et al. (2008) &lt;<a href="https://doi.org/10.1534%2Fgenetics.107.085589">doi:10.1534/genetics.107.085589</a>&gt;; Zhou et al. (2013) &lt;<a href="https://doi.org/10.1371%2Fjournal.pgen.1003264">doi:10.1371/journal.pgen.1003264</a>&gt;; Moser et al. (2015) &lt;<a href="https://doi.org/10.1371%2Fjournal.pgen.1004969">doi:10.1371/journal.pgen.1004969</a>&gt;; Lloyd-Jones et al. (2019) &lt;<a href="https://doi.org/10.1038%2Fs41467-019-12653-0">doi:10.1038/s41467-019-12653-0</a>&gt;; Henderson (1976) &lt;<a href="https://doi.org/10.2307%2F2529339">doi:10.2307/2529339</a>&gt;; Fernando et al. (2014) &lt;<a href="https://doi.org/10.1186%2F1297-9686-46-50">doi:10.1186/1297-9686-46-50</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lilin Yin &lt;ylilin@163.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/YinLiLin/hibayes">https://github.com/YinLiLin/hibayes</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/YinLiLin/hibayes/issues">https://github.com/YinLiLin/hibayes/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, stats, methods, stringr, CMplot</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0), bigmemory, Matrix</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo (&ge; 0.9.600.0.0), RcppProgress, BH,
bigmemory, Matrix</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-20 10:17:02 UTC; yinll</td>
</tr>
<tr>
<td>Author:</td>
<td>Lilin Yin [aut, cre, cph],
  Haohao Zhang [aut, cph],
  Xiaolei Liu [aut, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-20 13:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ibrm'>Bayes model</h2><span id='topic+ibrm'></span>

<h3>Description</h3>

<p>Bayes linear regression model using individual level data
</p>
<p style="text-align: center;"><code class="reqn">y = X \beta + R r + M \alpha + e</code>
</p>

<p>where <code class="reqn">\beta</code> is a vector of estimated coefficient for covariates, and <code class="reqn">r</code> is a vector of environmental random effects. <code class="reqn">M</code> is a matrix of genotype covariate, <code class="reqn">\alpha</code> is a vector of estimated marker effect size. <code class="reqn">e</code> is a vector of residuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ibrm(
  formula,
  data = NULL,
  M = NULL,
  M.id = NULL,
  method = c("BayesCpi", "BayesA", "BayesL", "BSLMM", "BayesR", "BayesB", "BayesC",
    "BayesBpi", "BayesRR"),
  map = NULL,
  Pi = NULL,
  fold = NULL,
  niter = NULL,
  nburn = NULL,
  thin = 5,
  windsize = NULL,
  windnum = NULL,
  dfvr = NULL,
  s2vr = NULL,
  vg = NULL,
  dfvg = NULL,
  s2vg = NULL,
  ve = NULL,
  dfve = NULL,
  s2ve = NULL,
  lambda = 0,
  printfreq = 100,
  seed = 666666,
  threads = 4,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ibrm_+3A_formula">formula</code></td>
<td>
<p>a two-sided linear formula object describing both the fixed-effects and random-effects part of the model, with the response on the left of a ‘~’ operator and the terms, separated by ‘+’ operators, on the right. Random-effects terms are distinguished by vertical bars (1|’) separating expressions for design matrices from grouping factors.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_data">data</code></td>
<td>
<p>the data frame containing the variables named in 'formula', NOTE that the first column in 'data' should be the individual id.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_m">M</code></td>
<td>
<p>numeric matrix of genotype with individuals in rows and markers in columns, NAs are not allowed.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_m.id">M.id</code></td>
<td>
<p>vector of id for genotyped individuals, NOTE that no need to adjust the order of id to be the same between 'data' and 'M', the package will do it automatically.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_method">method</code></td>
<td>
<p>bayes methods including: &quot;BayesB&quot;, &quot;BayesA&quot;, &quot;BayesL&quot;, &quot;BayesRR&quot;, &quot;BayesBpi&quot;, &quot;BayesC&quot;, &quot;BayesCpi&quot;, &quot;BayesR&quot;, &quot;BSLMM&quot;.
</p>

<ul>
<li><p> &quot;BayesRR&quot;: Bayes Ridge Regression, all SNPs have non-zero effects and share the same variance, equals to RRBLUP or GBLUP.
</p>
</li>
<li><p> &quot;BayesA&quot;: all SNPs have non-zero effects, and take different variance which follows an inverse chi-square distribution.
</p>
</li>
<li><p> &quot;BayesB&quot;: only a small proportion of SNPs (1-Pi) have non-zero effects, and take different variance which follows an inverse chi-square distribution.
</p>
</li>
<li><p> &quot;BayesBpi&quot;: the same with &quot;BayesB&quot;, but 'Pi' is not fixed.
</p>
</li>
<li><p> &quot;BayesC&quot;: only a small proportion of SNPs (1-Pi) have non-zero effects, and share the same variance.
</p>
</li>
<li><p> &quot;BayesCpi&quot;: the same with &quot;BayesC&quot;, but 'Pi' is not fixed.
</p>
</li>
<li><p> &quot;BayesL&quot;: BayesLASSO, all SNPs have non-zero effects, and take different variance which follows an exponential distribution.
</p>
</li>
<li><p> &quot;BSLMM&quot;: all SNPs have non-zero effects, and take the same variance, but a small proportion of SNPs have additional shared variance.
</p>
</li>
<li><p> &quot;BayesR&quot;: only a small proportion of SNPs have non-zero effects, and the SNPs are allocated into different groups, each group has the same variance.
</p>
</li></ul>
</td></tr>
<tr><td><code id="ibrm_+3A_map">map</code></td>
<td>
<p>(optional, only for GWAS) the map information of genotype, at least 3 columns are: SNPs, chromosome, physical position.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_pi">Pi</code></td>
<td>
<p>vector, the proportion of zero effect and non-zero effect SNPs, the first value must be the proportion of non-effect markers.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_fold">fold</code></td>
<td>
<p>proportion of variance explained for groups of SNPs, the default is c(0, 0.0001, 0.001, 0.01).</p>
</td></tr>
<tr><td><code id="ibrm_+3A_niter">niter</code></td>
<td>
<p>the number of MCMC iteration.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_nburn">nburn</code></td>
<td>
<p>the number of iterations to be discarded.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_thin">thin</code></td>
<td>
<p>the number of thinning after burn-in. Note that smaller thinning frequency may have higher accuracy of estimated parameters, but would result in more memory for collecting process, on contrary, bigger frequency may have negative effect on accuracy of estimations.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_windsize">windsize</code></td>
<td>
<p>window size in bp for GWAS, the default is NULL.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_windnum">windnum</code></td>
<td>
<p>fixed number of SNPs in a window for GWAS, if it is specified, 'windsize' will be invalid, the default is NULL.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_dfvr">dfvr</code></td>
<td>
<p>the number of degrees of freedom for the distribution of environmental variance.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_s2vr">s2vr</code></td>
<td>
<p>scale parameter for the distribution of environmental variance.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_vg">vg</code></td>
<td>
<p>prior value of genetic variance.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_dfvg">dfvg</code></td>
<td>
<p>the number of degrees of freedom for the distribution of genetic variance.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_s2vg">s2vg</code></td>
<td>
<p>scale parameter for the distribution of genetic variance.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_ve">ve</code></td>
<td>
<p>prior value of residual variance.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_dfve">dfve</code></td>
<td>
<p>the number of degrees of freedom for the distribution of residual variance.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_s2ve">s2ve</code></td>
<td>
<p>scale parameter for the distribution of residual variance.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_lambda">lambda</code></td>
<td>
<p>value of ridge regression for inverting a matrix.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_printfreq">printfreq</code></td>
<td>
<p>frequency of printing iterative details on console.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_seed">seed</code></td>
<td>
<p>seed for random sample.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_threads">threads</code></td>
<td>
<p>number of threads used for OpenMP.</p>
</td></tr>
<tr><td><code id="ibrm_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the iteration information on console.</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p>the fixed effects and covariates in 'formula' must be in factors and numeric, respectively. if not, please remember to use 'as.factor' and 'as.numeric' to transform.
</p>
</li>
<li><p>the package has the automatical function of taking the intersection and adjusting the order of id between 'data' and the genotype 'M', thus the first column in 'data' should be the individual id.
</p>
</li>
<li><p>if any one of the options 'windsize' and 'windnum' is specified, the GWAS results will be returned, and the 'map' information must be provided, in which the physical positions should be all in digital values.
</p>
</li>
<li><p>the 'windsize' or 'windnum' option only works for the methods of which the assumption has a proportion of zero effect markers, e.g., BayesB, BayesBpi, BayesC, BayesCpi, BSLMM, and BayesR.
</p>
</li></ul>



<h3>Value</h3>

<p>the function returns a 'blrMod' object containing
</p>

<dl>
<dt>$mu</dt><dd><p>the regression intercept</p>
</dd>
<dt>$pi</dt><dd><p>estimated proportion of zero effect and non-zero effect SNPs</p>
</dd>
<dt>$beta</dt><dd><p>estimated coefficients for all covariates</p>
</dd>
<dt>$r</dt><dd><p>estimated environmental random effects</p>
</dd>
<dt>$Vr</dt><dd><p>estimated variance for all environmental random effect</p>
</dd>
<dt>$Vg</dt><dd><p>estimated genetic variance</p>
</dd>
<dt>$Ve</dt><dd><p>estimated residual variance</p>
</dd>
<dt>$h2</dt><dd><p>estimated heritability (h2 = Vg / (Vr + Vg + Ve))</p>
</dd>
<dt>$alpha</dt><dd><p>estimated effect size of all markers</p>
</dd>
<dt>$g</dt><dd><p>genomic estimated breeding value</p>
</dd>
<dt>$e</dt><dd><p>residuals of the model</p>
</dd>
<dt>$pip</dt><dd><p>the frequency for markers to be included in the model during MCMC iteration, known as posterior inclusive probability (PIP)</p>
</dd>
<dt>$gwas</dt><dd><p>WPPA is defined to be the window posterior probability of association, it is estimated by counting the number of MCMC samples in which </p>
<p style="text-align: center;"><code class="reqn">\alpha</code>
</p>
<p> is nonzero for at least one SNP in the window</p>
</dd>
<dt>$MCMCsamples</dt><dd><p>the collected samples of posterior estimation for all the above parameters across MCMC iterations</p>
</dd>
</dl>



<h3>References</h3>

<p>Meuwissen, Theo HE, Ben J. Hayes, and Michael E. Goddard. &quot;Prediction of total genetic value using genome-wide dense marker maps.&quot; Genetics 157.4 (2001): 1819-1829. <br />
de los Campos, G., Hickey, J. M., Pong-Wong, R., Daetwyler, H. D., and Calus, M. P. (2013). Whole-genome regression and prediction methods applied to plant and animal breeding. Genetics, 193(2), 327-345. <br />
Habier, David, et al. &quot;Extension of the Bayesian alphabet for genomic selection.&quot; BMC bioinformatics 12.1 (2011): 1-12. <br />
Yi, Nengjun, and Shizhong Xu. &quot;Bayesian LASSO for quantitative trait loci mapping.&quot; Genetics 179.2 (2008): 1045-1055. <br />
Zhou, Xiang, Peter Carbonetto, and Matthew Stephens. &quot;Polygenic modeling with Bayesian sparse linear mixed models.&quot; PLoS genetics 9.2 (2013): e1003264. <br />
Moser, Gerhard, et al. &quot;Simultaneous discovery, estimation and prediction analysis of complex traits using a Bayesian mixture model.&quot; PLoS genetics 11.4 (2015): e1004969. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the example data attached in the package
pheno_file_path = system.file("extdata", "demo.phe", package = "hibayes")
pheno = read.table(pheno_file_path, header=TRUE)

bfile_path = system.file("extdata", "demo", package = "hibayes")
bin = read_plink(bfile_path, threads=1)
fam = bin$fam
geno = bin$geno
map = bin$map

# For GS/GP
## no environmental effects:
fit = ibrm(T1~1, data=pheno, M=geno, M.id=fam[,2], method="BayesCpi",
	niter=2000, nburn=1200, thin=5, threads=1)

## overview of the returned results
summary(fit)



## add fixed effects or covariates:
fit = ibrm(T1~sex+season+day+bwt, data=pheno, M=geno, M.id=fam[,2],
	method="BayesCpi")
 
## add environmental random effects:
fit = ibrm(T1~sex+(1|loc)+(1|dam), data=pheno, M=geno, M.id=fam[,2],
	method="BayesCpi")

# For GWAS
fit = ibrm(T1~sex+bwt+(1|dam), data=pheno, M=geno, M.id=fam[,2],
	method="BayesCpi", map=map, windsize=1e6)


# get the SD of estimated SNP effects for markers
summary(fit)$alpha
# get the SD of estimated breeding values
summary(fit)$g

</code></pre>

<hr>
<h2 id='ldmat'>LD variance-covariance matrix calculation</h2><span id='topic+ldmat'></span>

<h3>Description</h3>

<p>To calculate density or sparse LD variance-covariance matrix with genotype in bigmemory format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldmat(
  geno,
  map = NULL,
  gwas.geno = NULL,
  gwas.map = NULL,
  chisq = NULL,
  ldchr = FALSE,
  threads = 4,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ldmat_+3A_geno">geno</code></td>
<td>
<p>the reference genotype panel in bigmemory format.</p>
</td></tr>
<tr><td><code id="ldmat_+3A_map">map</code></td>
<td>
<p>the map information of reference genotype panel, columns are: SNPs, chromosome, physical position.</p>
</td></tr>
<tr><td><code id="ldmat_+3A_gwas.geno">gwas.geno</code></td>
<td>
<p>(optional) the genotype of gwas samples which were used to generate the summary data.</p>
</td></tr>
<tr><td><code id="ldmat_+3A_gwas.map">gwas.map</code></td>
<td>
<p>(optional) the map information of the genotype of gwas samples, columns are: SNPs, chromosome, physical position.</p>
</td></tr>
<tr><td><code id="ldmat_+3A_chisq">chisq</code></td>
<td>
<p>chi-squre value for generating sparse matrix, if n*r2 &lt; chisq, it would be set to zero.</p>
</td></tr>
<tr><td><code id="ldmat_+3A_ldchr">ldchr</code></td>
<td>
<p>lpgical, whether to calulate the LD between chromosomes.</p>
</td></tr>
<tr><td><code id="ldmat_+3A_threads">threads</code></td>
<td>
<p>the number of threads used in computation.</p>
</td></tr>
<tr><td><code id="ldmat_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For full ld matrix, it returns a standard R matrix, for sparse matrix, it returns a 'dgCMatrix'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bfile_path = system.file("extdata", "demo", package = "hibayes")
data = read_plink(bfile_path)
geno = data$geno
map = data$map

xx = ldmat(geno, threads=4, verbose=FALSE)   #chromosome wide full ld matrix
# xx = ldmat(geno, chisq=5, threads=4)   #chromosome wide sparse ld matrix
# xx = ldmat(geno, map, ldchr=FALSE, threads=4)   #chromosome block ld matrix
# xx = ldmat(geno, map, ldchr=FALSE, chisq=5, threads=4)   #chromosome block + sparse ld matrix


</code></pre>

<hr>
<h2 id='read_plink'>data load</h2><span id='topic+read_plink'></span>

<h3>Description</h3>

<p>To load plink binary data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_plink(
  bfile = "",
  maxLine = 10000,
  impute = TRUE,
  mode = c("A", "D"),
  out = NULL,
  threads = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_plink_+3A_bfile">bfile</code></td>
<td>
<p>character, prefix of Plink binary format data.</p>
</td></tr>
<tr><td><code id="read_plink_+3A_maxline">maxLine</code></td>
<td>
<p>number, set the number of lines to read at a time.</p>
</td></tr>
<tr><td><code id="read_plink_+3A_impute">impute</code></td>
<td>
<p>logical, whether to impute missing values in genotype by major alleles.</p>
</td></tr>
<tr><td><code id="read_plink_+3A_mode">mode</code></td>
<td>
<p>&quot;A&quot; or &quot;D&quot;, additive effect or dominant effect.</p>
</td></tr>
<tr><td><code id="read_plink_+3A_out">out</code></td>
<td>
<p>character, path and prefix of output file</p>
</td></tr>
<tr><td><code id="read_plink_+3A_threads">threads</code></td>
<td>
<p>number, the number of used threads for parallel process</p>
</td></tr>
</table>


<h3>Value</h3>

<p>four files will be generated in the directed folder: &quot;xx.desc&quot;, &quot;xx.bin&quot;, &quot;xx.id, &quot;xx.map&quot;, where 'xx' is the prefix of the argument 'out', the memory-mapping files can be fast loaded into memory by 'geno = attach.big.matrix(&quot;xx.desc&quot;)'. Note that hibayes will code the genotype A1A1 as 2, A1A2 as 1, and A2A2 as 0, where A1 is the first allele of each marker in &quot;.bim&quot; file, therefore the estimated effect size is on A1 allele, users should pay attention to it when a process involves marker effect.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bfile_path = system.file("extdata", "demo", package = "hibayes")
data = read_plink(bfile_path, out=tempfile(), mode="A")
fam = data$fam
geno = data$geno
map = data$map
</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+CMplot'></span><span id='topic+new'></span><span id='topic+packageVersion'></span><span id='topic+read.delim'></span><span id='topic+read.table'></span><span id='topic+write.table'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>CMplot</dt><dd><p><code><a href="CMplot.html#topic+CMplot-package">CMplot</a></code></p>
</dd>
<dt>methods</dt><dd><p><code><a href="methods.html#topic+new">new</a></code></p>
</dd>
<dt>utils</dt><dd><p><code><a href="utils.html#topic+packageDescription">packageVersion</a></code>, <code><a href="utils.html#topic+read.table">read.delim</a></code>, <code><a href="utils.html#topic+read.table">read.table</a></code>, <code><a href="utils.html#topic+write.table">write.table</a></code></p>
</dd>
</dl>

<hr>
<h2 id='sbrm'>SBayes model</h2><span id='topic+sbrm'></span>

<h3>Description</h3>

<p>Bayes linear regression model using summary level data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sbrm(
  sumstat,
  ldm,
  method = c("BayesB", "BayesA", "BayesL", "BayesRR", "BayesBpi", "BayesC", "BayesCpi",
    "BayesR", "CG"),
  map = NULL,
  Pi = NULL,
  lambda = NULL,
  fold = NULL,
  niter = NULL,
  nburn = NULL,
  thin = 5,
  windsize = NULL,
  windnum = NULL,
  vg = NULL,
  dfvg = NULL,
  s2vg = NULL,
  ve = NULL,
  dfve = NULL,
  s2ve = NULL,
  printfreq = 100,
  seed = 666666,
  threads = 4,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sbrm_+3A_sumstat">sumstat</code></td>
<td>
<p>matrix of summary data, details refer to https://cnsgenomics.com/software/gcta/#COJO.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_ldm">ldm</code></td>
<td>
<p>dense or sparse matrix, ld for reference panel (m * m, m is the number of SNPs). NOTE that the order of SNPs should be consistent with summary data.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_method">method</code></td>
<td>
<p>bayes methods including: &quot;BayesB&quot;, &quot;BayesA&quot;, &quot;BayesL&quot;, &quot;BayesRR&quot;, &quot;BayesBpi&quot;, &quot;BayesC&quot;, &quot;BayesCpi&quot;, &quot;BayesR&quot;, &quot;CG&quot;.
</p>

<ul>
<li><p> &quot;BayesRR&quot;: Bayes Ridge Regression, all SNPs have non-zero effects and share the same variance, equals to RRBLUP or GBLUP.
</p>
</li>
<li><p> &quot;BayesA&quot;: all SNPs have non-zero effects, and take different variance which follows an inverse chi-square distribution.
</p>
</li>
<li><p> &quot;BayesB&quot;: only a small proportion of SNPs (1-Pi) have non-zero effects, and take different variance which follows an inverse chi-square distribution.
</p>
</li>
<li><p> &quot;BayesBpi&quot;: the same with &quot;BayesB&quot;, but 'Pi' is not fixed.
</p>
</li>
<li><p> &quot;BayesC&quot;: only a small proportion of SNPs (1-Pi) have non-zero effects, and share the same variance.
</p>
</li>
<li><p> &quot;BayesCpi&quot;: the same with &quot;BayesC&quot;, but 'Pi' is not fixed.
</p>
</li>
<li><p> &quot;BayesL&quot;: BayesLASSO, all SNPs have non-zero effects, and take different variance which follows an exponential distribution.
</p>
</li>
<li><p> &quot;BayesR&quot;: only a small proportion of SNPs have non-zero effects, and the SNPs are allocated into different groups, each group has the same variance.
</p>
</li>
<li><p> &quot;CG&quot;: conjugate gradient algorithm with assigned lambda.
</p>
</li></ul>
</td></tr>
<tr><td><code id="sbrm_+3A_map">map</code></td>
<td>
<p>(optional, only for GWAS) the map information of genotype, at least 3 columns are: SNPs, chromosome, physical position.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_pi">Pi</code></td>
<td>
<p>vector, the proportion of zero effect and non-zero effect SNPs, the first value must be the proportion of non-effect markers.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_lambda">lambda</code></td>
<td>
<p>value or vector, the ridge regression value for each SNPs.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_fold">fold</code></td>
<td>
<p>percentage of variance explained for groups of SNPs, the default is c(0, 0.0001, 0.001, 0.01).</p>
</td></tr>
<tr><td><code id="sbrm_+3A_niter">niter</code></td>
<td>
<p>the number of MCMC iteration.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_nburn">nburn</code></td>
<td>
<p>the number of iterations to be discarded.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_thin">thin</code></td>
<td>
<p>the number of thinning after burn-in. Note that smaller thinning frequency may have higher accuracy of estimated parameters, but would result in more memory for collecting process, on contrary, bigger frequency may have negative effect on accuracy of estimations.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_windsize">windsize</code></td>
<td>
<p>window size in bp for GWAS, the default is 1e6.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_windnum">windnum</code></td>
<td>
<p>fixed number of SNPs in a window for GWAS, if it is specified, 'windsize' will be invalid, the default is NULL.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_vg">vg</code></td>
<td>
<p>prior value of genetic variance.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_dfvg">dfvg</code></td>
<td>
<p>the number of degrees of freedom for the distribution of genetic variance.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_s2vg">s2vg</code></td>
<td>
<p>scale parameter for the distribution of genetic variance.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_ve">ve</code></td>
<td>
<p>prior value of residual variance.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_dfve">dfve</code></td>
<td>
<p>the number of degrees of freedom for the distribution of residual variance.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_s2ve">s2ve</code></td>
<td>
<p>scale parameter for the distribution of residual variance.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_printfreq">printfreq</code></td>
<td>
<p>frequency of collecting the estimated parameters and printing on console. Note that smaller frequency may have higher accuracy of estimated parameters, but would result in more time and memory for collecting process, on contrary, bigger frequency may have an negative effect on accuracy of estimations.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_seed">seed</code></td>
<td>
<p>seed for random sample.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_threads">threads</code></td>
<td>
<p>number of threads used for OpenMP.</p>
</td></tr>
<tr><td><code id="sbrm_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the iteration information on console.</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p>if any one of the options 'windsize' and 'windnum' is specified, the GWAS results will be returned, and the 'map' information must be provided, in which the physical positions should be all in digital values.
</p>
</li>
<li><p>the 'windsize' or 'windnum' option only works for the methods of which the assumption has a proportion of zero effect markers, e.g., BayesB, BayesBpi, BayesC, BayesCpi, BSLMM, and BayesR.
</p>
</li></ul>



<h3>Value</h3>

<p>the function returns a 'blrMod' object containing
</p>

<dl>
<dt>$pi</dt><dd><p>estimated proportion of zero effect and non-zero effect SNPs</p>
</dd>
<dt>$Vg</dt><dd><p>estimated genetic variance</p>
</dd>
<dt>$Ve</dt><dd><p>estimated residual variance</p>
</dd>
<dt>$h2</dt><dd><p>estimated heritability (h2 = Vg / (Vg + Ve))</p>
</dd>
<dt>$alpha</dt><dd><p>estimated effect size of all markers</p>
</dd>
<dt>$pip</dt><dd><p>the frequency for markers to be included in the model during MCMC iteration, also known as posterior inclusive probability (PIP)</p>
</dd>
<dt>$gwas</dt><dd><p>WPPA is defined to be the window posterior probability of association, it is estimated by counting the number of MCMC samples in which </p>
<p style="text-align: center;"><code class="reqn">\alpha</code>
</p>
<p> is nonzero for at least one SNP in the window</p>
</dd>
<dt>$MCMCsamples</dt><dd><p>the collected samples of posterior estimation for all the above parameters across MCMC iterations</p>
</dd>
</dl>



<h3>References</h3>

<p>Lloyd-Jones, Luke R., et al. &quot;Improved polygenic prediction by Bayesian multiple regression on summary statistics.&quot; Nature communications 10.1 (2019): 1-11.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bfile_path = system.file("extdata", "demo", package = "hibayes")
bin = read_plink(bfile_path, threads=1)
fam = bin$fam
geno = bin$geno
map = bin$map

sumstat_path = system.file("extdata", "demo.ma", package = "hibayes")
sumstat = read.table(sumstat_path, header=TRUE)
head(sumstat)


# computate ld variance covariance matrix
## construct genome wide full variance-covariance matrix
ldm1 &lt;- ldmat(geno, threads=4)	
## construct genome wide sparse variance-covariance matrix
# ldm2 &lt;- ldmat(geno, chisq=5, threads=4)	
## construct chromosome wide full variance-covariance matrix
# ldm3 &lt;- ldmat(geno, map, ldchr=FALSE, threads=4)	
## construct chromosome wide sparse variance-covariance matrix
# ldm4 &lt;- ldmat(geno, map, ldchr=FALSE, chisq=5, threads=4)

# if the order of SNPs in genotype is not consistent with the order in sumstat file, 
# prior adjusting is necessary.
indx = match(map[, 1], sumstat[, 1])
sumstat = sumstat[indx, ]

# fit model
fit = sbrm(sumstat=sumstat, ldm=ldm1, method="BayesCpi", Pi = c(0.95, 0.05), 
	niter=20000, nburn=12000, seed=666666, map=map, windsize=1e6, threads=1)

# overview of the returned results
summary(fit)

# get the SD of estimated SNP effects for markers
summary(fit)$alpha


</code></pre>

<hr>
<h2 id='ssbrm'>Single-step Bayes model</h2><span id='topic+ssbrm'></span>

<h3>Description</h3>

<p>Single-step Bayes linear regression model using individual level data and pedigree information
</p>
<p style="text-align: center;"><code class="reqn">y = X \beta + R r + M \alpha + U \epsilon + e</code>
</p>

<p>where <code class="reqn">y</code> is the vector of phenotypic values for both genotyped and non-genotyped individuals, <code class="reqn">\beta</code> is a vector of estimated coefficient for covariates, <code class="reqn">M</code> contains the genotype (<code class="reqn">M_2</code>) for genotyped individuals and the imputed genotype (<code class="reqn">M_1 = A_{12}A_{22}^{-1}M_2</code>) for non-genotyped individuals, <code class="reqn">\epsilon</code> is the vector of genotype imputation error, <code class="reqn">e</code> is a vector of residuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssbrm(
  formula,
  data = NULL,
  M = NULL,
  M.id = NULL,
  pedigree = NULL,
  method = c("BayesCpi", "BayesA", "BayesL", "BayesR", "BayesB", "BayesC", "BayesBpi",
    "BayesRR"),
  map = NULL,
  Pi = NULL,
  fold = NULL,
  niter = NULL,
  nburn = NULL,
  thin = 5,
  windsize = NULL,
  windnum = NULL,
  maf = 0.01,
  dfvr = NULL,
  s2vr = NULL,
  vg = NULL,
  dfvg = NULL,
  s2vg = NULL,
  ve = NULL,
  dfve = NULL,
  s2ve = NULL,
  printfreq = 100,
  seed = 666666,
  threads = 4,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ssbrm_+3A_formula">formula</code></td>
<td>
<p>a two-sided linear formula object describing both the fixed-effects and random-effects part of the model, with the response on the left of a ‘~’ operator and the terms, separated by ‘+’ operators, on the right. Random-effects terms are distinguished by vertical bars (1|’) separating expressions for design matrices from grouping factors.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_data">data</code></td>
<td>
<p>the data frame containing the variables named in 'formula', NOTE that the first column in 'data' should be the individual id.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_m">M</code></td>
<td>
<p>numeric matrix of genotype with individuals in rows and markers in columns, NAs are not allowed.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_m.id">M.id</code></td>
<td>
<p>vector of id for genotype.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_pedigree">pedigree</code></td>
<td>
<p>matrix of pedigree, 3 columns limited, the order of columns shoud be &quot;id&quot;, &quot;sir&quot;, &quot;dam&quot;.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_method">method</code></td>
<td>
<p>bayes methods including: &quot;BayesB&quot;, &quot;BayesA&quot;, &quot;BayesL&quot;, &quot;BayesRR&quot;, &quot;BayesBpi&quot;, &quot;BayesC&quot;, &quot;BayesCpi&quot;, &quot;BayesR&quot;.
</p>

<ul>
<li><p> &quot;BayesRR&quot;: Bayes Ridge Regression, all SNPs have non-zero effects and share the same variance, equals to RRBLUP or GBLUP.
</p>
</li>
<li><p> &quot;BayesA&quot;: all SNPs have non-zero effects, and take different variance which follows an inverse chi-square distribution.
</p>
</li>
<li><p> &quot;BayesB&quot;: only a small proportion of SNPs (1-Pi) have non-zero effects, and take different variance which follows an inverse chi-square distribution.
</p>
</li>
<li><p> &quot;BayesBpi&quot;: the same with &quot;BayesB&quot;, but 'Pi' is not fixed.
</p>
</li>
<li><p> &quot;BayesC&quot;: only a small proportion of SNPs (1-Pi) have non-zero effects, and share the same variance.
</p>
</li>
<li><p> &quot;BayesCpi&quot;: the same with &quot;BayesC&quot;, but 'Pi' is not fixed.
</p>
</li>
<li><p> &quot;BayesL&quot;: BayesLASSO, all SNPs have non-zero effects, and take different variance which follows an exponential distribution.
</p>
</li>
<li><p> &quot;BayesR&quot;: only a small proportion of SNPs have non-zero effects, and the SNPs are allocated into different groups, each group has the same variance.
</p>
</li></ul>
</td></tr>
<tr><td><code id="ssbrm_+3A_map">map</code></td>
<td>
<p>(optional, only for GWAS) the map information of genotype, at least 3 columns are: SNPs, chromosome, physical position.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_pi">Pi</code></td>
<td>
<p>vector, the proportion of zero effect and non-zero effect SNPs, the first value must be the proportion of non-effect markers.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_fold">fold</code></td>
<td>
<p>proportion of variance explained for groups of SNPs, the default is c(0, 0.0001, 0.001, 0.01).</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_niter">niter</code></td>
<td>
<p>the number of MCMC iteration.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_nburn">nburn</code></td>
<td>
<p>the number of iterations to be discarded.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_thin">thin</code></td>
<td>
<p>the number of thinning after burn-in. Note that smaller thinning frequency may have higher accuracy of estimated parameters, but would result in more memory for collecting process, on contrary, bigger frequency may have negative effect on accuracy of estimations.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_windsize">windsize</code></td>
<td>
<p>window size in bp for GWAS, the default is NULL.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_windnum">windnum</code></td>
<td>
<p>fixed number of SNPs in a window for GWAS, if it is specified, 'windsize' will be invalid, the default is NULL.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_maf">maf</code></td>
<td>
<p>the effects of markers whose MAF is lower than the threshold will not be estimated.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_dfvr">dfvr</code></td>
<td>
<p>the number of degrees of freedom for the distribution of environmental variance.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_s2vr">s2vr</code></td>
<td>
<p>scale parameter for the distribution of environmental variance.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_vg">vg</code></td>
<td>
<p>prior value of genetic variance.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_dfvg">dfvg</code></td>
<td>
<p>the number of degrees of freedom for the distribution of genetic variance.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_s2vg">s2vg</code></td>
<td>
<p>scale parameter for the distribution of genetic variance.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_ve">ve</code></td>
<td>
<p>prior value of residual variance.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_dfve">dfve</code></td>
<td>
<p>the number of degrees of freedom for the distribution of residual variance.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_s2ve">s2ve</code></td>
<td>
<p>scale parameter for the distribution of residual variance.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_printfreq">printfreq</code></td>
<td>
<p>frequency of printing iterative details on console.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_seed">seed</code></td>
<td>
<p>seed for random sample.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_threads">threads</code></td>
<td>
<p>number of threads used for OpenMP.</p>
</td></tr>
<tr><td><code id="ssbrm_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the iteration information on console.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the function returns a  a 'blrMod' object containing
</p>

<dl>
<dt>$J</dt><dd><p>coefficient for genotype imputation residuals</p>
</dd>
<dt>$Veps</dt><dd><p>estimated variance of genotype imputation residuals</p>
</dd>
<dt>$epsilon</dt><dd><p>genotype imputation residuals</p>
</dd>
<dt>$mu</dt><dd><p>the regression intercept</p>
</dd>
<dt>$pi</dt><dd><p>estimated proportion of zero effect and non-zero effect SNPs</p>
</dd>
<dt>$beta</dt><dd><p>estimated coefficients for all covariates</p>
</dd>
<dt>$r</dt><dd><p>estimated environmental random effects</p>
</dd>
<dt>$Vr</dt><dd><p>estimated variance for all environmental random effect</p>
</dd>
<dt>$Vg</dt><dd><p>estimated genetic variance</p>
</dd>
<dt>$Ve</dt><dd><p>estimated residual variance</p>
</dd>
<dt>$h2</dt><dd><p>estimated heritability (h2 = Vg / (Vr + Vg + Ve))</p>
</dd>
<dt>$g</dt><dd><p>data.frame, the first column is the list of individual id, the second column is the genomic estimated breeding value for all individuals, including genotyped and non-genotyped.</p>
</dd>
<dt>$alpha</dt><dd><p>estimated effect size of all markers</p>
</dd>
<dt>$e</dt><dd><p>residuals of the model</p>
</dd>
<dt>$pip</dt><dd><p>the frequency for markers to be included in the model during MCMC iteration, also known as posterior inclusive probability (PIP)</p>
</dd>
<dt>$gwas</dt><dd><p>WPPA is defined to be the window posterior probability of association, it is estimated by counting the number of MCMC samples in which </p>
<p style="text-align: center;"><code class="reqn">\alpha</code>
</p>
<p> is nonzero for at least one SNP in the window</p>
</dd>
<dt>$MCMCsamples</dt><dd><p>the collected samples of posterior estimation for all the above parameters across MCMC iterations</p>
</dd>
</dl>



<h3>References</h3>

<p>Fernando, Rohan L., Jack CM Dekkers, and Dorian J. Garrick. &quot;A class of Bayesian methods to combine large numbers of genotyped and non-genotyped animals for whole-genome analyses.&quot; Genetics Selection Evolution 46.1 (2014): 1-13. <br />
Henderson, C.R.: A simple method for computing the inverse of a numerator relationship matrix used in prediction of breeding values. Biometrics 32(1), 69-83 (1976).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the example data attached in the package
pheno_file_path = system.file("extdata", "demo.phe", package = "hibayes")
pheno = read.table(pheno_file_path, header=TRUE)

bfile_path = system.file("extdata", "demo", package = "hibayes")
bin = read_plink(bfile_path, threads=1)
fam = bin$fam
geno = bin$geno
map = bin$map

pedigree_file_path = system.file("extdata", "demo.ped", package = "hibayes")
ped = read.table(pedigree_file_path, header=TRUE)

# For GS/GP
## no environmental effects:
fit = ssbrm(T1~1, data=pheno, M=geno, M.id=fam[,2], pedigree=ped,
	method="BayesCpi", niter=1000, nburn=600, thin=5, printfreq=100, threads=1)

## overview of the returned results
summary(fit)



## add fixed effects or covariates:
fit = ssbrm(T1~sex+bwt, data=pheno, M=geno, M.id=fam[,2], pedigree=ped,
	method="BayesCpi")

## add environmental random effects:
fit = ssbrm(T1~(1|loc)+(1|dam), data=pheno, M=geno, M.id=fam[,2],
	pedigree=ped, method="BayesCpi")

# For GWAS
fit = ssbrm(T1~sex+bwt+(1|dam), data=pheno, M=geno, M.id=fam[,2],
	pedigree=ped, method="BayesCpi", map=map, windsize=1e6)


# get the SD of estimated SNP effects for markers
summary(fit)$alpha
# get the SD of estimated breeding values
summary(fit)$g

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
