<!DOCTYPE html><html><head><title>Help for package hidecan</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hidecan}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.check_cols'><p>Checks whether some columns are present in a tibble</p></a></li>
<li><a href='#.compute_chrom_length_genes'><p>Computes chromosomes' length for a tibble of genes</p></a></li>
<li><a href='#apply_threshold'><p>Filters GWAS or DE results based on a threshold</p></a></li>
<li><a href='#CAN_data'><p>Creates a <code>CAN_data</code> object</p></a></li>
<li><a href='#combine_chrom_length'><p>Computes chromosomes' length from list</p></a></li>
<li><a href='#compute_chrom_length'><p>Computes chromosomes' length</p></a></li>
<li><a href='#create_hidecan_plot'><p>Creates a HIDECAN plot</p></a></li>
<li><a href='#DE_data'><p>Creates a <code>DE_data</code> object</p></a></li>
<li><a href='#get_example_data'><p>Example dataset</p></a></li>
<li><a href='#GWAS_data'><p>Creates a <code>GWAS_data</code> object</p></a></li>
<li><a href='#GWAS_data_from_gwaspoly'><p>Extracts information from GWASpoly output</p></a></li>
<li><a href='#hidecan_plot'><p>Wrapper to create a HIDECAN plot</p></a></li>
<li><a href='#hidecan_plot_from_gwaspoly'><p>Creates a HIDECAN plot from GWASpoly output</p></a></li>
<li><a href='#new_CAN_data'><p><code>CAN_data</code> constructor</p></a></li>
<li><a href='#new_DE_data'><p><code>DE_data</code> constructor</p></a></li>
<li><a href='#new_GWAS_data'><p><code>GWAS_data</code> constructor</p></a></li>
<li><a href='#run_hidecan_shiny'><p>Launches the HIDECAN shiny app</p></a></li>
<li><a href='#validate_CAN_data'><p>Checks validity of input for <code>CAN_data</code> constructor</p></a></li>
<li><a href='#validate_DE_data'><p>Checks validity of input for <code>DE_data</code> constructor</p></a></li>
<li><a href='#validate_GWAS_data'><p>Checks validity of input for <code>GWAS_data</code> constructor</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Create HIDECAN Plots for Visualising Genome-Wide Association
Studies and Differential Expression Results</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Generates HIDECAN plots that summarise and combine 
  the results of genome-wide association studies (GWAS) and transcriptomics 
  differential expression analyses (DE), along with manually curated candidate genes of interest. The HIDECAN plot 
  is presented in Angelin-Bonnet et al. (2023) (currently in review).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://plantandfoodresearch.github.io/hidecan/">https://plantandfoodresearch.github.io/hidecan/</a>,
<a href="https://github.com/PlantandFoodResearch/hidecan">https://github.com/PlantandFoodResearch/hidecan</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/PlantandFoodResearch/hidecan/issues">https://github.com/PlantandFoodResearch/hidecan/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, ggnewscale, ggplot2, ggrepel, purrr, shiny, tibble,
tidyr, viridis, vroom</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, stringr, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-09 02:07:11 UTC; HRPOAB</td>
</tr>
<tr>
<td>Author:</td>
<td>Olivia Angelin-Bonnet
    <a href="https://orcid.org/0000-0002-7708-2919"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Olivia Angelin-Bonnet &lt;olivia.angelin-bonnet@plantandfood.co.nz&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-10 09:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='.check_cols'>Checks whether some columns are present in a tibble</h2><span id='topic+.check_cols'></span>

<h3>Description</h3>

<p>Checks whether some columns are present in a tibble
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.check_cols(x, col_names, param_name = "Input data-frame")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".check_cols_+3A_x">x</code></td>
<td>
<p>Tibble</p>
</td></tr>
<tr><td><code id=".check_cols_+3A_col_names">col_names</code></td>
<td>
<p>character vector of column names</p>
</td></tr>
<tr><td><code id=".check_cols_+3A_param_name">param_name</code></td>
<td>
<p>Character, name of the dataframe to use in the error message.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>invisible NULL
</p>

<hr>
<h2 id='.compute_chrom_length_genes'>Computes chromosomes' length for a tibble of genes</h2><span id='topic+.compute_chrom_length_genes'></span>

<h3>Description</h3>

<p>Computes the length (in bp) of each chromosome as the maximum
position of genes on the chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.compute_chrom_length_genes(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".compute_chrom_length_genes_+3A_x">x</code></td>
<td>
<p>Either a <code>DE_data</code> or <code>CAN_data</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with two columns: <code>chromosome</code> (chromosome name) and
<code>length</code> (chromosome length in base pair).
</p>

<hr>
<h2 id='apply_threshold'>Filters GWAS or DE results based on a threshold</h2><span id='topic+apply_threshold'></span><span id='topic+apply_threshold.GWAS_data'></span><span id='topic+apply_threshold.DE_data'></span><span id='topic+apply_threshold.CAN_data'></span><span id='topic+apply_threshold.default'></span>

<h3>Description</h3>

<p>Filters markers or genes/transcripts based on a threshold applied to their
GWAS or DE score, and log2(fold-change) (if applicable). For a set of
candidate genes, simply returns the list. Note that markers or genes with
a missing score or log2(fold-change) will be removed from the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apply_threshold(x, score_thr = 0, log2fc_thr = 0)

## S3 method for class 'GWAS_data'
apply_threshold(x, score_thr = 0, log2fc_thr = 0)

## S3 method for class 'DE_data'
apply_threshold(x, score_thr = 0, log2fc_thr = 0)

## S3 method for class 'CAN_data'
apply_threshold(x, score_thr = 0, log2fc_thr = 0)

## Default S3 method:
apply_threshold(x, score_thr = 0, log2fc_thr = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="apply_threshold_+3A_x">x</code></td>
<td>
<p>Either a <code>GWAS_data</code>, <code>DE_data</code> or <code>CAN_data</code> object.</p>
</td></tr>
<tr><td><code id="apply_threshold_+3A_score_thr">score_thr</code></td>
<td>
<p>Numeric, threshold to use on markers' or genes/transcripts' score.
Only markers or genes with a score equal to or higher than this threshold
will be retained. Default value is 0. Ignored for <code>CAN_data</code>.</p>
</td></tr>
<tr><td><code id="apply_threshold_+3A_log2fc_thr">log2fc_thr</code></td>
<td>
<p>Numeric, threshold to use on the absolute value of genes/
transcripts' log2(fold-change). Only genes/transcripts with an absolute
log2(fold-change) equal to or higher than this threshold will be retained.
Ignored for <code>GWAS_data</code> and <code>CAN_data</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A filtered tibble (of class <code>GWAS_data_thr</code>, <code>DE_data_thr</code> or
<code>CAN_data_thr</code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- get_example_data()

## For GWAS results
apply_threshold(GWAS_data(x[["GWAS"]]), score_thr = 4)

## For DE results - in second line, no threshold is applied
## on the log2(fold-change)
apply_threshold(DE_data(x[["DE"]]), score_thr = -log10(0.05), log2fc_thr = 1)
apply_threshold(DE_data(x[["DE"]]), score_thr = -log10(0.05), log2fc_thr = 0)

## No effect on the Candidate genes
apply_threshold(CAN_data(x[["CAN"]]))
</code></pre>

<hr>
<h2 id='CAN_data'>Creates a <code>CAN_data</code> object</h2><span id='topic+CAN_data'></span>

<h3>Description</h3>

<p>Creates a <code>CAN_data</code> object from a tibble or data-frame of candidate genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CAN_data(dat, keep_rownames_as = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CAN_data_+3A_dat">dat</code></td>
<td>
<p>Tibble, set of candidate genes of interest. See Details.</p>
</td></tr>
<tr><td><code id="CAN_data_+3A_keep_rownames_as">keep_rownames_as</code></td>
<td>
<p>Character, the name of the column in which to save the
rownames of the input data-frame. Default value is <code>NULL</code>, i.e. rownames will
be discarded.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input data should have one row per gene, and at least the
following columns:
</p>

<ul>
<li> <p><code>chromosome</code>: character column, chromosome on which the gene is located.
</p>
</li>
<li> <p><code>start</code> and <code>end</code>: numeric, starting and end position of the gene (in bp).
A column <code>position</code> will be constructed as the middle value (mean) between
<code>start</code> and <code>end</code>.
</p>
</li>
<li> <p><code>name</code>: character, the name of the candidate genes to be displayed.
</p>
</li></ul>



<h3>Value</h3>

<p>A <code>CAN_data</code> object, i.e. a tibble.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- get_example_data()

CAN_data(x[["CAN"]])
</code></pre>

<hr>
<h2 id='combine_chrom_length'>Computes chromosomes' length from list</h2><span id='topic+combine_chrom_length'></span>

<h3>Description</h3>

<p>Computes the length (in bp) of each chromosome from a list of GWAS and
DE results as well as candidate gene lists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_chrom_length(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_chrom_length_+3A_x">x</code></td>
<td>
<p>A list of <code>GWAS_data</code>, <code>DE_data</code> or <code>CAN_data</code> objects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with two columns: <code>chromosome</code> (chromosome name) and
<code>length</code> (chromosome length in base pair).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- get_example_data()
y &lt;- list("GWAS" = GWAS_data(x[["GWAS"]]),
          "DE" = DE_data(x[["DE"]]),
          "CAN" = CAN_data(x[["CAN"]]))

combine_chrom_length(y)
</code></pre>

<hr>
<h2 id='compute_chrom_length'>Computes chromosomes' length</h2><span id='topic+compute_chrom_length'></span><span id='topic+compute_chrom_length.GWAS_data'></span><span id='topic+compute_chrom_length.DE_data'></span><span id='topic+compute_chrom_length.CAN_data'></span>

<h3>Description</h3>

<p>Computes the length (in bp) of each chromosome as the maximum
position of markers or genes on the chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_chrom_length(x)

## S3 method for class 'GWAS_data'
compute_chrom_length(x)

## S3 method for class 'DE_data'
compute_chrom_length(x)

## S3 method for class 'CAN_data'
compute_chrom_length(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_chrom_length_+3A_x">x</code></td>
<td>
<p>Either a <code>GWAS_data</code>, <code>DE_data</code> or <code>CAN_data</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with two columns: <code>chromosome</code> (chromosome name) and
<code>length</code> (chromosome length in base pair).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- get_example_data()

compute_chrom_length(GWAS_data(x[["GWAS"]]))
compute_chrom_length(DE_data(x[["DE"]]))
compute_chrom_length(CAN_data(x[["CAN"]]))
</code></pre>

<hr>
<h2 id='create_hidecan_plot'>Creates a HIDECAN plot</h2><span id='topic+create_hidecan_plot'></span>

<h3>Description</h3>

<p>Creates a HIDECAN plot from a list of filtered GWAS or DE results
and/or candidate genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_hidecan_plot(
  x,
  chrom_length,
  colour_genes_by_score = TRUE,
  remove_empty_chrom = FALSE,
  chroms = NULL,
  chrom_limits = NULL,
  title = NULL,
  subtitle = NULL,
  n_rows = NULL,
  n_cols = 2,
  legend_position = "bottom",
  point_size = 3,
  label_size = 3.5,
  label_padding = 0.15
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_hidecan_plot_+3A_x">x</code></td>
<td>
<p>A list of <code>GWAS_data_thr</code>, <code>DE_data_thr</code> and/or <code>CAN_data_thr</code>
produced by the <code><a href="#topic+apply_threshold">apply_threshold</a>()</code> function. If named, the names
will be appended to the y-axis labels (use <code>' '</code> as empty name in the list).</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_chrom_length">chrom_length</code></td>
<td>
<p>Tibble with columns <code>chromosome</code> and <code>length</code>, giving
for each chromosome its length in bp (see <code><a href="#topic+combine_chrom_length">combine_chrom_length</a>()</code>
function).</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_colour_genes_by_score">colour_genes_by_score</code></td>
<td>
<p>Logical, whether to colour the genes by score
(<code>TRUE</code>) or by log2(fold-change) (<code>FALSE</code>). Default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_remove_empty_chrom">remove_empty_chrom</code></td>
<td>
<p>Logical, should chromosomes with no significant
markers/genes nor candidate genes be removed from the plot? Default value
if <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_chroms">chroms</code></td>
<td>
<p>Character vector, name of chromosomes to include in the plot.</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_chrom_limits">chrom_limits</code></td>
<td>
<p>Integer vector of length 2, or named list where the
elements are integer vectors of length 2. If vector, gives the lower and upper
limit of the chromosomes (in bp) to use in the plot. If a named list, names
should correspond to chromosome names. Gives for each chromosome the lower
and upper limits (in bp) to use in the plot. Doesn't have to be specified
for all chromosomes. Default value is <code>NULL</code>, i.e. no limits are applied
to the chromosomes (they will be plotted in their entirety).</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_title">title</code></td>
<td>
<p>Character, title of the plot. Default value is <code>NULL</code> (i.e.
no title will be added to the plot).</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_subtitle">subtitle</code></td>
<td>
<p>Character, subtitle of the plot. Default value is <code>NULL</code>
(i.e. no subtitle will be added to the plot).</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_n_rows">n_rows</code></td>
<td>
<p>Integer, number of rows of chromosomes to create in the plot.
Default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_n_cols">n_cols</code></td>
<td>
<p>Integer, number of columns of chromosomes to create in the plot.
Default value is 2. Will be set to <code>NULL</code> if <code>n_rows</code> is not <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_legend_position">legend_position</code></td>
<td>
<p>Character, position of the legend in the plot. Can be
<code>bottom</code> (default value), <code>top</code>, <code>right</code>, <code>left</code> or <code>none</code>.</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_point_size">point_size</code></td>
<td>
<p>Numeric, size of the points in the plot. Default value is 3.</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_label_size">label_size</code></td>
<td>
<p>Numeric, size of the gene labels in the plot. Default value is
3.5 (for <code><a href="ggrepel.html#topic+geom_label_repel">geom_label_repel</a></code>).</p>
</td></tr>
<tr><td><code id="create_hidecan_plot_+3A_label_padding">label_padding</code></td>
<td>
<p>Numeric, amount of padding around gene labels in the plot,
as unit or number. Default value is 0.15
(for <a href="ggrepel.html#topic+geom_label_repel">geom_label_repel</a>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
x &lt;- get_example_data()
y &lt;- list("GWAS" = GWAS_data(x[["GWAS"]]),
          "DE" = DE_data(x[["DE"]]),
          "CAN" = CAN_data(x[["CAN"]]))

chrom_length &lt;- combine_chrom_length(y)

z &lt;- list(
  apply_threshold(y[["GWAS"]], score_thr = 4),
  apply_threshold(y[["DE"]], score_thr = 1.3, log2fc_thr = 0.5),
  apply_threshold(y[["CAN"]])
)

create_hidecan_plot(z,
                    chrom_length,
                    label_size = 2)

## Colour genes according to their fold-change
create_hidecan_plot(z,
                    chrom_length,
                    colour_genes_by_score = FALSE,
                    label_size = 2)

## Add names to the datasets
create_hidecan_plot(setNames(z, c("Genomics", "RNAseq", "My list")),
                    chrom_length,
                    colour_genes_by_score = FALSE,
                    label_size = 2)

## Add names to some of the datasets only (e.g. not for GWAS results)
create_hidecan_plot(setNames(z, c(" ", "RNAseq", "My list")),
                    chrom_length,
                    colour_genes_by_score = FALSE,
                    label_size = 2)

## Set limits on all chromosomes (to "zoom in" to the 10-20Mb region)
create_hidecan_plot(z,
                    chrom_length,
                    label_size = 2,
                    chrom_limits = c(10e6, 20e6))

## Set limits on some chromosomes only
create_hidecan_plot(z,
                    chrom_length,
                    label_size = 2,
                    chrom_limits = list("ST4.03ch00" = c(10e6, 20e6),
                                        "ST4.03ch02" = c(15e6, 25e6)))
}
</code></pre>

<hr>
<h2 id='DE_data'>Creates a <code>DE_data</code> object</h2><span id='topic+DE_data'></span>

<h3>Description</h3>

<p>Creates a <code>DE_data</code> object from a tibble or data-frame of differential expression
results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DE_data(dat, keep_rownames_as = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DE_data_+3A_dat">dat</code></td>
<td>
<p>Tibble, results from a differential expression analysis. See Details.</p>
</td></tr>
<tr><td><code id="DE_data_+3A_keep_rownames_as">keep_rownames_as</code></td>
<td>
<p>Character, the name of the column in which to save the
rownames of the input data-frame. Default value is <code>NULL</code>, i.e. rownames will
be discarded.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input data should have one row per gene or transcript, and at least the
following columns:
</p>

<ul>
<li> <p><code>chromosome</code>: character column, chromosome on which the gene/transcript is located.
</p>
</li>
<li> <p><code>start</code> and <code>end</code>: numeric, starting and end position of the gene/transcript
(in bp). A column <code>position</code> will be constructed as the middle value (mean) between
<code>start</code> and <code>end</code>.
</p>
</li>
<li> <p><code>score</code> or <code>padj</code>: numeric, the DE score or adjusted p-value of the
gene/transcript. If column <code>score</code> column is missing, will be constructed
as <code>-log10(padj)</code>.
</p>
</li>
<li> <p><code>foldChange</code> or <code>log2FoldChange</code>: numeric, the fold-change or log2(fold-change)
of the gene/transcript. If column <code>log2FoldChange</code> is missing, will be constructed
as <code>log2(foldChange)</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A <code>DE_data</code> object, i.e. a tibble.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- get_example_data()

DE_data(x[["DE"]])
</code></pre>

<hr>
<h2 id='get_example_data'>Example dataset</h2><span id='topic+get_example_data'></span>

<h3>Description</h3>

<p>Returns a list of example datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_example_data()
</code></pre>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<ul>
<li> <p><code>GWAS</code>: a tibble of GWAS results, with columns <code>id</code>, <code>chromosome</code>,
<code>position</code> and <code>score</code>.
</p>
</li>
<li> <p><code>DE</code>: a tibble of differential expression results, with columns <code>gene</code>,
<code>chromosome</code>, <code>padj</code>, <code>log2FoldChange</code>, <code>start</code>, <code>end</code> and <code>label</code>.
</p>
</li>
<li> <p><code>CAN</code>: a tibble of candidate genes, with columns <code>id</code>, <code>chromosome</code>,
<code>start</code>, <code>end</code>, <code>name</code> and <code>gene_name</code>.
</p>
</li></ul>


<hr>
<h2 id='GWAS_data'>Creates a <code>GWAS_data</code> object</h2><span id='topic+GWAS_data'></span>

<h3>Description</h3>

<p>Creates a <code>GWAS_data</code> object from a tibble or data-frame of GWAS results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GWAS_data(dat, keep_rownames_as = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GWAS_data_+3A_dat">dat</code></td>
<td>
<p>Tibble, results from a GWAS analysis. See Details.</p>
</td></tr>
<tr><td><code id="GWAS_data_+3A_keep_rownames_as">keep_rownames_as</code></td>
<td>
<p>Character, the name of the column in which to save the
rownames of the input data-frame. Default value is <code>NULL</code>, i.e. rownames will
be discarded.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input data should have one row per marker, and at least the
following columns:
</p>

<ul>
<li> <p><code>chromosome</code>: character column, chromosome on which the marker is located.
</p>
</li>
<li> <p><code>position</code>: numeric, the physical position of the marker along the chromosome (in bp).
</p>
</li>
<li> <p><code>score</code> or <code>padj</code>: numeric, the GWAS score or adjusted p-value of the marker.
If column <code>score</code> column is missing, will be constructed as <code>-log10(padj)</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A <code>GWAS_data</code> object, i.e. a tibble.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- get_example_data()

GWAS_data(x[["GWAS"]])
</code></pre>

<hr>
<h2 id='GWAS_data_from_gwaspoly'>Extracts information from GWASpoly output</h2><span id='topic+GWAS_data_from_gwaspoly'></span>

<h3>Description</h3>

<p>Extracts GWAS results and chromosome length from GWASpoly output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GWAS_data_from_gwaspoly(gwaspoly_output, traits = NULL, models = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GWAS_data_from_gwaspoly_+3A_gwaspoly_output">gwaspoly_output</code></td>
<td>
<p>A <code>GWASpoly.fitted</code> or <code>GWASpoly.thresh</code> object
(returned by <code>GWASpoly::GWASpoly()</code> or <code>GWASpoly::set.threshold()</code> functions).</p>
</td></tr>
<tr><td><code id="GWAS_data_from_gwaspoly_+3A_traits">traits</code></td>
<td>
<p>Character vector, traits for which GWAS results should be
extracted. If <code>NULL</code> (default value), all traits present are considered.</p>
</td></tr>
<tr><td><code id="GWAS_data_from_gwaspoly_+3A_models">models</code></td>
<td>
<p>Character vector, genetic models for which GWAS results should be
extracted. If <code>NULL</code> (default value), all genetic models present are considered.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<ul>
<li> <p><code>gwas_data_list</code>: A named list of <code>GWAS_data</code> objects, giving the markers score for
each possible trait/genetic model combination. The names of the list are in
the form <code style="white-space: pre;">&#8288;trait (genetic model)&#8288;</code>.
</p>
</li>
<li> <p><code>gwas_data_thr_list</code>: if the input data is a <code>GWASpoly.thresh</code> object
(from the <code>GWASpoly::set.threshold()</code> function), a named list of <code>GwAS_data_thr</code>,
with the significant markers score for each possible trait/genetic model
combination. The names of the list are in the form <code style="white-space: pre;">&#8288;trait (genetic model)&#8288;</code>.
</p>
</li>
<li> <p><code>chrom_length</code>: A tibble with one row per chromosome, giving the length
(in bp) of each chromosome.
</p>
</li></ul>


<hr>
<h2 id='hidecan_plot'>Wrapper to create a HIDECAN plot</h2><span id='topic+hidecan_plot'></span>

<h3>Description</h3>

<p>Wrapper function to create a HIDECAN plot from GWAS results,
DE results or candidate genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hidecan_plot(
  gwas_list = NULL,
  de_list = NULL,
  can_list = NULL,
  score_thr_gwas = 4,
  score_thr_de = 2,
  log2fc_thr = 1,
  chrom_length = NULL,
  colour_genes_by_score = TRUE,
  remove_empty_chrom = FALSE,
  chroms = NULL,
  chrom_limits = NULL,
  title = NULL,
  subtitle = NULL,
  n_rows = NULL,
  n_cols = 2,
  legend_position = "bottom",
  point_size = 3,
  label_size = 3.5,
  label_padding = 0.15
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hidecan_plot_+3A_gwas_list">gwas_list</code></td>
<td>
<p>Data-frame or list of data-frames containing GWAS results,
each with at least a <code>chromosome</code>, <code>position</code> and either <code>padj</code> or <code>score</code>
columns. If a named list, the names will be used in the plot.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_de_list">de_list</code></td>
<td>
<p>Data-frame or list of data-frames containing DE results,
each with at least a <code>chromosome</code>, <code>start</code>, <code>end</code>, <code>log2FoldChange</code> and
either <code>padj</code> or <code>score</code> columns. If a named list, the names will be used in
the plot.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_can_list">can_list</code></td>
<td>
<p>Data-frame or list of data-frames containing candidate genes,
each with at least a <code>chromosome</code>, <code>start</code>, <code>end</code> and <code>name</code> columns. If a
named list, the names will be used in the plot.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_score_thr_gwas">score_thr_gwas</code></td>
<td>
<p>Numeric, the score threshold for GWAS results that will be used to
select which markers will be plotted. Default value is 4.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_score_thr_de">score_thr_de</code></td>
<td>
<p>Numeric, the score threshold for DE results that will be used to
select which markers will be plotted. Default value is 2.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_log2fc_thr">log2fc_thr</code></td>
<td>
<p>Numeric, the log2(fold-change) threshold that will be used
to select which genes will be plotted. Default value is 1.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_chrom_length">chrom_length</code></td>
<td>
<p>Optional, tibble with columns <code>chromosome</code> and <code>length</code>,
giving for each chromosome its length in bp. If <code>NULL</code> (the default), will
be inferred from the GWAS, DE and candidate gene data.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_colour_genes_by_score">colour_genes_by_score</code></td>
<td>
<p>Logical, whether to colour the genes by score
(<code>TRUE</code>) or by log2(fold-change) (<code>FALSE</code>). Default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_remove_empty_chrom">remove_empty_chrom</code></td>
<td>
<p>Logical, should chromosomes with no significant
markers/genes nor candidate genes be removed from the plot? Default value
if <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_chroms">chroms</code></td>
<td>
<p>Character vector, name of chromosomes to include in the plot.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_chrom_limits">chrom_limits</code></td>
<td>
<p>Integer vector of length 2, or named list where the
elements are integer vectors of length 2. If vector, gives the lower and upper
limit of the chromosomes (in bp) to use in the plot. If a named list, names
should correspond to chromosome names. Gives for each chromosome the lower
and upper limits (in bp) to use in the plot. Doesn't have to be specified
for all chromosomes. Default value is <code>NULL</code>, i.e. no limits are applied
to the chromosomes (they will be plotted in their entirety).</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_title">title</code></td>
<td>
<p>Character, title of the plot. Default value is <code>NULL</code> (i.e.
no title will be added to the plot).</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_subtitle">subtitle</code></td>
<td>
<p>Character, subtitle of the plot. Default value is <code>NULL</code>
(i.e. no subtitle will be added to the plot).</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_n_rows">n_rows</code></td>
<td>
<p>Integer, number of rows of chromosomes to create in the plot.
Default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_n_cols">n_cols</code></td>
<td>
<p>Integer, number of columns of chromosomes to create in the plot.
Default value is 2. Will be set to <code>NULL</code> if <code>n_rows</code> is not <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_legend_position">legend_position</code></td>
<td>
<p>Character, position of the legend in the plot. Can be
<code>bottom</code> (default value), <code>top</code>, <code>right</code>, <code>left</code> or <code>none</code>.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_point_size">point_size</code></td>
<td>
<p>Numeric, size of the points in the plot. Default value is 3.</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_label_size">label_size</code></td>
<td>
<p>Numeric, size of the gene labels in the plot. Default value is
3.5 (for <code><a href="ggrepel.html#topic+geom_label_repel">geom_label_repel</a></code>).</p>
</td></tr>
<tr><td><code id="hidecan_plot_+3A_label_padding">label_padding</code></td>
<td>
<p>Numeric, amount of padding around gene labels in the plot,
as unit or number. Default value is 0.15
(for <a href="ggrepel.html#topic+geom_label_repel">geom_label_repel</a>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
x &lt;- get_example_data()

## Typical example with one GWAs result table, one DE result table and
## one table of candidate genes
hidecan_plot(gwas_list = x[["GWAS"]],
             de_list = x[["DE"]],
             can_list = x[["CAN"]],
             score_thr_gwas = -log10(0.0001),
             score_thr_de = -log10(0.005),
             log2fc_thr = 0,
             label_size = 2)

## Example with two sets of GWAS results
hidecan_plot(gwas_list = list(x[["GWAS"]], x[["GWAS"]]),
             score_thr_gwas = 4)

## Example with two sets of DE results, with names
hidecan_plot(de_list = list("X vs Y" = x[["DE"]],
                            "X vs Z" = x[["DE"]]),
             score_thr_de = -log10(0.05),
             log2fc_thr = 0)

## Set limits on all chromosomes (to "zoom in" to the 10-20Mb region)
hidecan_plot(gwas_list = x[["GWAS"]],
             de_list = x[["DE"]],
             can_list = x[["CAN"]],
             score_thr_gwas = -log10(0.0001),
             score_thr_de = -log10(0.005),
             log2fc_thr = 0,
             label_size = 2,
             chrom_limits = c(10e6, 20e6))

## Set limits on some chromosomes only
hidecan_plot(gwas_list = x[["GWAS"]],
             de_list = x[["DE"]],
             can_list = x[["CAN"]],
             score_thr_gwas = -log10(0.0001),
             score_thr_de = -log10(0.005),
             log2fc_thr = 0,
             label_size = 2,
             chrom_limits = list("ST4.03ch00" = c(10e6, 20e6),
                                  "ST4.03ch02" = c(15e6, 25e6)))
}
</code></pre>

<hr>
<h2 id='hidecan_plot_from_gwaspoly'>Creates a HIDECAN plot from GWASpoly output</h2><span id='topic+hidecan_plot_from_gwaspoly'></span>

<h3>Description</h3>

<p>Creates a HIDECAN plot from the GWAS results from GWASpoly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hidecan_plot_from_gwaspoly(gwaspoly_output, traits = NULL, models = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hidecan_plot_from_gwaspoly_+3A_gwaspoly_output">gwaspoly_output</code></td>
<td>
<p>A <code>GWASpoly.thresh</code> object
(returned by the <code>GWASpoly::set.threshold()</code> function).</p>
</td></tr>
<tr><td><code id="hidecan_plot_from_gwaspoly_+3A_traits">traits</code></td>
<td>
<p>Character vector, traits for which GWAS results should be
extracted. If <code>NULL</code> (default value), all traits present are considered.</p>
</td></tr>
<tr><td><code id="hidecan_plot_from_gwaspoly_+3A_models">models</code></td>
<td>
<p>Character vector, genetic models for which GWAS results should be
extracted. If <code>NULL</code> (default value), all genetic models present are considered.</p>
</td></tr>
<tr><td><code id="hidecan_plot_from_gwaspoly_+3A_...">...</code></td>
<td>
<p>Further arguments passed to the <code><a href="#topic+create_hidecan_plot">create_hidecan_plot</a>()</code>
function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot</code>.
</p>

<hr>
<h2 id='new_CAN_data'><code>CAN_data</code> constructor</h2><span id='topic+new_CAN_data'></span>

<h3>Description</h3>

<p><code>CAN_data</code> constructor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>new_CAN_data(dat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="new_CAN_data_+3A_dat">dat</code></td>
<td>
<p>Tibble, containing information about genes of interest, with at least columns
<code>chromosome</code>, <code>start</code>, <code>end</code>, <code>position</code> and <code>name</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>CAN_data</code> object, i.e. a tibble.
</p>

<hr>
<h2 id='new_DE_data'><code>DE_data</code> constructor</h2><span id='topic+new_DE_data'></span>

<h3>Description</h3>

<p><code>DE_data</code> constructor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>new_DE_data(dat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="new_DE_data_+3A_dat">dat</code></td>
<td>
<p>Tibble, results from a differential expression analysis, with at least columns
<code>chromosome</code>, <code>score</code>, <code>log2FoldChange</code>, <code>start</code>, <code>end</code> and <code>position</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>DE_data</code> object, i.e. a tibble.
</p>

<hr>
<h2 id='new_GWAS_data'><code>GWAS_data</code> constructor</h2><span id='topic+new_GWAS_data'></span>

<h3>Description</h3>

<p><code>GWAS_data</code> constructor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>new_GWAS_data(dat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="new_GWAS_data_+3A_dat">dat</code></td>
<td>
<p>Tibble, results from a GWAS analysis, with at least columns
<code>chromosome</code>, <code>position</code> and <code>score</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>GWAS_data</code> object, i.e. a tibble.
</p>

<hr>
<h2 id='run_hidecan_shiny'>Launches the HIDECAN shiny app</h2><span id='topic+run_hidecan_shiny'></span>

<h3>Description</h3>

<p>Starts the HIDECAN shiny app. The app reads in csv data to produce
a HIDECAN plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_hidecan_shiny()
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects (launching the shiny app).
</p>

<hr>
<h2 id='validate_CAN_data'>Checks validity of input for <code>CAN_data</code> constructor</h2><span id='topic+validate_CAN_data'></span>

<h3>Description</h3>

<p>Checks validity of input for <code>CAN_data</code> constructor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate_CAN_data(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="validate_CAN_data_+3A_x">x</code></td>
<td>
<p>a <code>CAN_data</code> object constructed via <a href="#topic+new_CAN_data">new_CAN_data</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>CAN_data</code> object, i.e. a tibble.
</p>

<hr>
<h2 id='validate_DE_data'>Checks validity of input for <code>DE_data</code> constructor</h2><span id='topic+validate_DE_data'></span>

<h3>Description</h3>

<p>Checks validity of input for <code>DE_data</code> constructor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate_DE_data(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="validate_DE_data_+3A_x">x</code></td>
<td>
<p>a <code>DE_data</code> object constructed via <a href="#topic+new_DE_data">new_DE_data</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>DE_data</code> object, i.e. a tibble.
</p>

<hr>
<h2 id='validate_GWAS_data'>Checks validity of input for <code>GWAS_data</code> constructor</h2><span id='topic+validate_GWAS_data'></span>

<h3>Description</h3>

<p>Checks validity of input for <code>GWAS_data</code> constructor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate_GWAS_data(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="validate_GWAS_data_+3A_x">x</code></td>
<td>
<p>a <code>GWAS_data</code> object constructed via <a href="#topic+new_GWAS_data">new_GWAS_data</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>GWAS_data</code> object, i.e. a tibble.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
