<!DOCTYPE html><html lang="en"><head><title>Help for package hierBipartite</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hierBipartite}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#constructBipartiteGraph'><p>Construct Bipartite Graph Edge Weight Matrix of Gene-drug Association Patterns</p></a></li>
<li><a href='#ctrp2'><p>Processed Cancer Cell Line Encyclopedia (CCLE) and Cancer Therapeutics Response Portal (CTRP2) Dataset</p></a></li>
<li><a href='#getMergeGroupRows'><p>Given index from hclust merge matrix, return X, Y row indices</p>
corresponding to groups specified by index</a></li>
<li><a href='#getSignificantMergedGroups'><p>Select Significant Results from 'HierBipartite'</p></a></li>
<li><a href='#hierBipartite'><p>Bipartite Graph-based Hierarchical Clustering</p></a></li>
<li><a href='#matrixDissimilarity'><p>Matrix dissimilarity</p></a></li>
<li><a href='#newMergedGroup'><p>Given group indices for a merge from hclust merge matrix, return group names of merged cluster</p></a></li>
<li><a href='#null_distri'><p>Null distribution of dissimilarity measures</p></a></li>
<li><a href='#p_value'><p>P-value of Similarity in Gene-drug Associations</p></a></li>
<li><a href='#scale_features'><p>Only scales features (columns) with positive variance</p></a></li>
<li><a href='#scca'><p>Sparse canonical covariance analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bipartite Graph-Based Hierarchical Clustering</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Calvin Chi &lt;calvin.chi@berkeley.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>
    Bipartite graph-based hierarchical clustering, developed for pharmacogenomic 
    datasets and datasets sharing the same data structure. The goal is to 
    construct a hierarchical clustering of groups of samples based on 
    association patterns between two sets of variables. In the context of
    pharmacogenomic datasets, the samples are cell lines, and the two sets of 
    variables are typically expression levels and drug sensitivity values. 
    For this method, sparse canonical correlation analysis from 
    Lee, W., Lee, D., Lee, Y. and Pawitan, Y. (2011) &lt;<a href="https://doi.org/10.2202%2F1544-6115.1638">doi:10.2202/1544-6115.1638</a>&gt; 
    is first applied to extract association patterns for each group of samples. 
    Then, a nuclear norm-based dissimilarity measure is used to construct a 
    dissimilarity matrix between groups based on the extracted associations. 
    Finally, hierarchical clustering is applied.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>parallel, magrittr, irlba</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-02-12 15:02:07 UTC; calvinchi</td>
</tr>
<tr>
<td>Author:</td>
<td>Calvin Chi [aut, cre, cph],
  Woojoo Lee [ctb],
  Donghwan Lee [ctb],
  Youngjo Lee [ctb],
  Yudi Pawitan [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-02-16 09:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='constructBipartiteGraph'>Construct Bipartite Graph Edge Weight Matrix of Gene-drug Association Patterns</h2><span id='topic+constructBipartiteGraph'></span>

<h3>Description</h3>

<p>Constructs edge weight matrix B representing association between set of variables in mat1 and set of variables in mat2 (see paper).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>constructBipartiteGraph(
  mat1,
  mat2,
  n_subsample = 1,
  subsampling_ratio = 1,
  parallel = FALSE,
  maxCores = 7
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="constructBipartiteGraph_+3A_mat1">mat1</code></td>
<td>
<p>an n x p matrix of variable set 1 (e.g. gene expression)</p>
</td></tr>
<tr><td><code id="constructBipartiteGraph_+3A_mat2">mat2</code></td>
<td>
<p>an n x q matrix of variable set 2 (e.g. drug sensitivity)</p>
</td></tr>
<tr><td><code id="constructBipartiteGraph_+3A_n_subsample">n_subsample</code></td>
<td>
<p>number of times to perform subsampling to generate B</p>
</td></tr>
<tr><td><code id="constructBipartiteGraph_+3A_subsampling_ratio">subsampling_ratio</code></td>
<td>
<p>fraction of samples to subsample each time</p>
</td></tr>
<tr><td><code id="constructBipartiteGraph_+3A_parallel">parallel</code></td>
<td>
<p>boolean for whether to parallelize subsampling</p>
</td></tr>
<tr><td><code id="constructBipartiteGraph_+3A_maxcores">maxCores</code></td>
<td>
<p>maximum number of cores to use (only applicable when parallel = TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a p x q matrix of bipartite graph edge weights
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Extract bipartite edge weight matrix B for cell lines from the
# squamous cell carcinoma, esophagus group
data(ctrp2)

groups = ctrp2$groups
X = ctrp2$X
Y = ctrp2$Y

x = X[groups[["squamous_cell_carcinoma_esophagus"]], ]
y = Y[groups[["squamous_cell_carcinoma_esophagus"]], ]

# Extract bipartite edge weight matrix B with subsampling
## Not run: 
B = constructBipartiteGraph(x, y, n_subsample = 100,
                            subsampling_ratio = 0.90,
                            parallel = TRUE, maxCores = 2)

## End(Not run)

</code></pre>

<hr>
<h2 id='ctrp2'>Processed Cancer Cell Line Encyclopedia (CCLE) and Cancer Therapeutics Response Portal (CTRP2) Dataset</h2><span id='topic+ctrp2'></span>

<h3>Description</h3>

<p>Smaller test dataset version of the &quot;CTRP2&quot; carcinoma dataset in the paper. Specifically,
only the top 1,000 transcripts by correlation with drug sensitivity are included instead of 5,000.
Otherwise the dataset has been processed exactly as described in the paper. Note the expression dataset is
provided by CCLE and the drug sensitivity dataset is provided by CTRP2, and the pharmacogenomic datasets in the paper are
are referred to by the resource providing the sensitivity data. The cell lines are grouped by carcinoma subtype and primary site (e.g. lung NSC).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ctrp2)
</code></pre>


<h3>Format</h3>

<p>A list with elements of gene expression, drug sensitivities, and group membership.
</p>

<dl>
<dt>X</dt><dd><p>n x p gene expression matrix</p>
</dd>
<dt>Y</dt><dd><p>n x q drug sensitivities matrix</p>
</dd>
<dt>groups</dt><dd><p>List of starting groups. Each group is represented by a vector of row indices for X, Y.</p>
</dd>
</dl>



<h3>References</h3>

<p>Barretina, J., et al. (2012). The cancer cell line encyclopedia enables predictive modelling of anticancer drug sensitivity.
Nature, 483(7391), 603â€“607. (<a href="https://pubmed.ncbi.nlm.nih.gov/22460905">PubMed</a>)
</p>
<p>Seashore-Ludlow, B., et al. (2015). Harnessing connectivity in a large-scale small-molecule sensitivity dataset.
Cancer discovery, 5(11), 1210-1223. (<a href="https://pubmed.ncbi.nlm.nih.gov/26482930">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ctrp2)
X = ctrp2[["X"]]
Y = ctrp2[["Y"]]
groups = ctrp2[["groups"]]

</code></pre>

<hr>
<h2 id='getMergeGroupRows'>Given index from hclust merge matrix, return X, Y row indices
corresponding to groups specified by index</h2><span id='topic+getMergeGroupRows'></span>

<h3>Description</h3>

<p>Given index from hclust merge matrix, return X, Y row indices
corresponding to groups specified by index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMergeGroupRows(idx, groups, groupMerges)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getMergeGroupRows_+3A_idx">idx</code></td>
<td>
<p>index from hclust merge</p>
</td></tr>
<tr><td><code id="getMergeGroupRows_+3A_groups">groups</code></td>
<td>
<p>list of starting group membership (input to hierBipartite())</p>
</td></tr>
<tr><td><code id="getMergeGroupRows_+3A_groupmerges">groupMerges</code></td>
<td>
<p>list of merged groups thus far (groupMerge[[3]] is a
vector of group names from cluster created at third merge)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of row indices corresponding to X, Y
</p>

<hr>
<h2 id='getSignificantMergedGroups'>Select Significant Results from 'HierBipartite'</h2><span id='topic+getSignificantMergedGroups'></span>

<h3>Description</h3>

<p>Selects clusters from bipartite graph-based hierarchical clustering with p-value less than or equal to a p-value cutoff.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSignificantMergedGroups(results, p = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getSignificantMergedGroups_+3A_results">results</code></td>
<td>
<p>list of results from bipartite graph-based hierarchical clustering</p>
</td></tr>
<tr><td><code id="getSignificantMergedGroups_+3A_p">p</code></td>
<td>
<p>p-value cutoff</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of results from bipartite graph-based hierarchical clustering, but only with clusters with p-value at or below p-value cutoff
</p>


<h3>Examples</h3>

<pre><code class='language-R'># sample bipartite graph-based hierarchical clustering of three groups
data(ctrp2)

groups = ctrp2$groups
X = ctrp2$X
Y = ctrp2$Y

groupNames = names(groups)
groupSmall = groups[groupNames[1:3]]

## Not run: 
result = hierBipartite(X, Y, groupSmall)

# set fictitious p-values, with one cluster with p-value less than the cutoff
# and the other not
result$nodePvals = list(0.03, 0.12)
getSignificantMergedGroups(result, p = 0.05)

## End(Not run)

</code></pre>

<hr>
<h2 id='hierBipartite'>Bipartite Graph-based Hierarchical Clustering</h2><span id='topic+hierBipartite'></span>

<h3>Description</h3>

<p>Main bipartite graph-based hierarchial clustering algorithm. Visit <a href="https://calvintchi.github.io/html/hierBipartite">here</a> for vignette on using the
hierBipartite package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hierBipartite(
  X,
  Y,
  groups,
  link = "ward.D2",
  n_subsample = 1,
  subsampling_ratio = 1,
  p.value = FALSE,
  n_perm = 100,
  parallel = FALSE,
  maxCores = 7,
  p_cutoff = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hierBipartite_+3A_x">X</code></td>
<td>
<p>an n x p matrix of variable set 1 (e.g. gene expression)</p>
</td></tr>
<tr><td><code id="hierBipartite_+3A_y">Y</code></td>
<td>
<p>an n x q matrix of variable set 2 (e.g. drug sensitivity)</p>
</td></tr>
<tr><td><code id="hierBipartite_+3A_groups">groups</code></td>
<td>
<p>a list of starting group membership (e.g. list(&quot;1&quot; = c(1,2,3), &quot;2&quot; = c(4,5,6)) means group 1 has samples 1, 2, 3, and group 2 has samples 4, 5, 6.</p>
</td></tr>
<tr><td><code id="hierBipartite_+3A_link">link</code></td>
<td>
<p>string indicating link function as input to hclust(). One of &quot;ward.D&quot;, &quot;ward.D2&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;average&quot;, &quot;mcquitty&quot;, &quot;median&quot;, &quot;centroid&quot;.</p>
</td></tr>
<tr><td><code id="hierBipartite_+3A_n_subsample">n_subsample</code></td>
<td>
<p>number of subsampling to generate matrix B (see paper)</p>
</td></tr>
<tr><td><code id="hierBipartite_+3A_subsampling_ratio">subsampling_ratio</code></td>
<td>
<p>fraction of samples to sample for subsampling to generate matrix B (see paper)</p>
</td></tr>
<tr><td><code id="hierBipartite_+3A_p.value">p.value</code></td>
<td>
<p>boolean for whether to generate p-values for each merge</p>
</td></tr>
<tr><td><code id="hierBipartite_+3A_n_perm">n_perm</code></td>
<td>
<p>number of permutations for generating p-values. Ignored if p.value = FALSE</p>
</td></tr>
<tr><td><code id="hierBipartite_+3A_parallel">parallel</code></td>
<td>
<p>boolean for whether to parallelize subsampling and p-value generation step</p>
</td></tr>
<tr><td><code id="hierBipartite_+3A_maxcores">maxCores</code></td>
<td>
<p>maximum number of cores to use (only applicable when parallel = TRUE)</p>
</td></tr>
<tr><td><code id="hierBipartite_+3A_p_cutoff">p_cutoff</code></td>
<td>
<p>p-value cutoff that determines whether merge is significant. If p-value &gt; p_cutoff, p-values will not be calculated for future merges involving current group. Ignored if p.value = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of results from bipartite graph-based hierarchical clustering, containing up to
</p>

<ul>
<li><p> hclustObj: hclust object
</p>
</li>
<li><p> groupMerges: list of clusters after each merge, in order of merge.
Each cluster is indicated by a vector of cell line groups
</p>
</li>
<li><p> nodePvals: list of p-value of each new merge,
in order of merge. Only available if p.value = TRUE
</p>
</li>
<li><p> D: dissimilarity matrix
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Get a small subset of the test dataset
data(ctrp2)

groups = ctrp2$groups
X = ctrp2$X
Y = ctrp2$Y

groupNames = names(groups)
groupSmall = groups[groupNames[1:3]]

## Not run: 
# Basic call of hierBipartite() on small test dataset
result0 = hierBipartite(X, Y, groupSmall)

# Calling hierBipartite() with subsampling
result1 = hierBipartite(X, Y, groupSmall, n_subsample = 100, subsampling_ratio = 0.90)

# Calling hierBipartite() with p-value generation
result2 = hierBipartite(X, Y, groupSmall, n_perm = 100, p.value = TRUE, p_cutoff = 0.10)

# Calling hierBipartite() with both subsampling and p-value generation (expensive)
result3 = hierBipartite(X, Y, groupSmall, n_subsample = 100, subsampling_ratio = 0.90,
                        n_perm = 100, p.value = TRUE, p_cutoff = 0.10)

## End(Not run)

</code></pre>

<hr>
<h2 id='matrixDissimilarity'>Matrix dissimilarity</h2><span id='topic+matrixDissimilarity'></span>

<h3>Description</h3>

<p>Computes nuclear norm-based dissimilarity measure between two matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrixDissimilarity(B1, B2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="matrixDissimilarity_+3A_b1">B1</code></td>
<td>
<p>first p x q bipartite graph edge weight matrix</p>
</td></tr>
<tr><td><code id="matrixDissimilarity_+3A_b2">B2</code></td>
<td>
<p>second p x q bipartite graph edge weight matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nuclear norm-based dissimilarity
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Compute matrix dissimilarity in edge weight matrix between squamous cell
# carcinoma, esophagus and squamous cell carcinoma, upper aerodigestive
data(ctrp2)

groups = ctrp2$groups
X = ctrp2$X
Y = ctrp2$Y

x1 = X[groups[["squamous_cell_carcinoma_esophagus"]], ]
y1 = Y[groups[["squamous_cell_carcinoma_esophagus"]], ]

## Not run: 
B1 = constructBipartiteGraph(x1, y1)

## End(Not run)

x2 = X[groups[["squamous_cell_carcinoma_upper_aerodigestive"]], ]
y2 = Y[groups[["squamous_cell_carcinoma_upper_aerodigestive"]], ]

## Not run: 
B2 = constructBipartiteGraph(x2, y2)
matrixDissimilarity(B1, B2)

## End(Not run)

</code></pre>

<hr>
<h2 id='newMergedGroup'>Given group indices for a merge from hclust merge matrix, return group names of merged cluster</h2><span id='topic+newMergedGroup'></span>

<h3>Description</h3>

<p>Given group indices for a merge from hclust merge matrix, return group names of merged cluster
</p>


<h3>Usage</h3>

<pre><code class='language-R'>newMergedGroup(idx1, idx2, groups, groupMerges)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="newMergedGroup_+3A_idx1">idx1</code></td>
<td>
<p>group index of first group from hclust merge</p>
</td></tr>
<tr><td><code id="newMergedGroup_+3A_idx2">idx2</code></td>
<td>
<p>group index of second group from hclust merge</p>
</td></tr>
<tr><td><code id="newMergedGroup_+3A_groups">groups</code></td>
<td>
<p>list of starting group membership (input to hierBipartite())</p>
</td></tr>
<tr><td><code id="newMergedGroup_+3A_groupmerges">groupMerges</code></td>
<td>
<p>list of merged groups thus far (groupMerge[[3]] is a vector of group names from cluster created at third merge)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of group names after merge of group indicated by idx1 and idx2
</p>

<hr>
<h2 id='null_distri'>Null distribution of dissimilarity measures</h2><span id='topic+null_distri'></span>

<h3>Description</h3>

<p>Generates null distribution of dissimilarity measures between group 1 (X1, Y1) and group 2 (X2, Y2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>null_distri(X1, Y1, X2, Y2, n.perm = 100, parallel = FALSE, maxCores = 7)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="null_distri_+3A_x1">X1</code></td>
<td>
<p>an n x p matrix of variable set 1 (e.g. gene expression) from group 1</p>
</td></tr>
<tr><td><code id="null_distri_+3A_y1">Y1</code></td>
<td>
<p>an n x q matrix of variable set 2 (e.g. drug sensitivity) from group 1</p>
</td></tr>
<tr><td><code id="null_distri_+3A_x2">X2</code></td>
<td>
<p>an n x p matrix of variable set 1 (e.g. gene expression) from group 2</p>
</td></tr>
<tr><td><code id="null_distri_+3A_y2">Y2</code></td>
<td>
<p>an n x q matrix of varaible set 2 (e.g. drug sensitivity) from group 2</p>
</td></tr>
<tr><td><code id="null_distri_+3A_n.perm">n.perm</code></td>
<td>
<p>number of null dissimilarity measures to generate</p>
</td></tr>
<tr><td><code id="null_distri_+3A_parallel">parallel</code></td>
<td>
<p>boolean for whether to parallelize permutation</p>
</td></tr>
<tr><td><code id="null_distri_+3A_maxcores">maxCores</code></td>
<td>
<p>maximum number of cores to use (only applicable when parallel = TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of length n.perm of null dissimilarity measures
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get data for group squamous cell carcinoma, esophagus and for group
# squamous cell carcinoma, upper aerodigestive
data(ctrp2)

groups = ctrp2$groups
X = ctrp2$X
Y = ctrp2$Y

x1 = X[groups[["squamous_cell_carcinoma_esophagus"]], ]
y1 = Y[groups[["squamous_cell_carcinoma_esophagus"]], ]

x2 = X[groups[["squamous_cell_carcinoma_upper_aerodigestive"]], ]
y2 = Y[groups[["squamous_cell_carcinoma_upper_aerodigestive"]], ]

## Not run: 
dissimilarities = null_distri(x1, y1, x2, y2, n.perm = 100)

## End(Not run)

</code></pre>

<hr>
<h2 id='p_value'>P-value of Similarity in Gene-drug Associations</h2><span id='topic+p_value'></span>

<h3>Description</h3>

<p>Computes p-value as number of null dissimilarities less than or equal to observed dissimilarity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_value(dissimilarity, dissimilarities)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_value_+3A_dissimilarity">dissimilarity</code></td>
<td>
<p>observed dissimilarity</p>
</td></tr>
<tr><td><code id="p_value_+3A_dissimilarities">dissimilarities</code></td>
<td>
<p>null distribution of dissimilarities</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p-value
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate null distribution of dissimilarities
dissimilarities = runif(100, min = 0, max = 1)

d = 0.10
p_value(d, dissimilarities)

</code></pre>

<hr>
<h2 id='scale_features'>Only scales features (columns) with positive variance</h2><span id='topic+scale_features'></span>

<h3>Description</h3>

<p>Only scales features (columns) with positive variance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_features(mat)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scale_features_+3A_mat">mat</code></td>
<td>
<p>an n x p matrix (gene expression or drug sensitivity)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an n x p matrix with columns with positive variance scaled
</p>

<hr>
<h2 id='scca'>Sparse canonical covariance analysis</h2><span id='topic+scca'></span>

<h3>Description</h3>

<p>'scca' is used to perform sparse canonical covariance analysis (SCCA)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scca(X,Y,penalty="HL",lamx=c(1,2,3),lamy=c(1,2,3),nc=1,
tuning="CV.alt",K=5,seed=NULL,center=TRUE,scale=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scca_+3A_x">X</code></td>
<td>
<p>n-by-p data matrix, where n is the number of subjects and p is the
number of variables</p>
</td></tr>
<tr><td><code id="scca_+3A_y">Y</code></td>
<td>
<p>n-by-q data matrix, where q is the number of variables</p>
</td></tr>
<tr><td><code id="scca_+3A_penalty">penalty</code></td>
<td>
<p>&quot;HL&quot; is the unbounded penalty proposed by Lee and Oh (2009).
&quot;LASSO&quot; (Tibshirani, 1996), &quot;SCAD&quot; (Fan and Li, 2001) and &quot;SOFT&quot;
(soft thresholding) are also available as other penalty options.
Default is &quot;HL&quot;.</p>
</td></tr>
<tr><td><code id="scca_+3A_lamx">lamx</code></td>
<td>
<p>A vector specifying grid points of the tuning parameter for X.
Default is (1,2,3).</p>
</td></tr>
<tr><td><code id="scca_+3A_lamy">lamy</code></td>
<td>
<p>A vector specifying grid points of the tuning parameter for Y.
Default is (1,2,3).</p>
</td></tr>
<tr><td><code id="scca_+3A_nc">nc</code></td>
<td>
<p>Number of components (canonical vectors). Default is 1.</p>
</td></tr>
<tr><td><code id="scca_+3A_tuning">tuning</code></td>
<td>
<p>How to find optimal tuning parameters for the sparsity.
If tuning=&quot;CV.full&quot;, then the tuning parameters are selected
automatically via K-fold cross-validation by using 2-dim'l grid search.
If &quot;CV.alt&quot;, then a sequential 1-dim'l search method is applied instead of
the 2-dim'l grid search. Default is &quot;CV.alt&quot;.</p>
</td></tr>
<tr><td><code id="scca_+3A_k">K</code></td>
<td>
<p>Perform K-fold cross-validation.</p>
</td></tr>
<tr><td><code id="scca_+3A_seed">seed</code></td>
<td>
<p>Seed number for initialization. A random initial point is
generated for tuning=&quot;CV.alt&quot;.</p>
</td></tr>
<tr><td><code id="scca_+3A_center">center</code></td>
<td>
<p>The columns of the data matrix are centered to have mean zero.
Default is TRUE.</p>
</td></tr>
<tr><td><code id="scca_+3A_scale">scale</code></td>
<td>
<p>The columns of the data matrix are scaled to have variance 1.
Default is FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sparse CCA uses a random-effect model approach to obtain sparse
regression. This model gives unbounded gains for zero loadings at the
origin. Various penalty functions can be adapted as well.
</p>


<h3>Value</h3>


<ul>
<li><p> A: p-by-nc matrix, k-th colum of A corresponds to k-th pattern
</p>
</li>
<li><p> B: q-by-nc matrix, k-th colum of B corresponds to k-th pattern
(canonical vector) for Y
</p>
</li>
<li><p> U: n-by-nc matrix. k-th column of U corresponds to k-th score
associated with k-th pattern for X
</p>
</li>
<li><p> V: n-by-nc matrix. k-th column of V corresponds to k-th score
associated with k-th pattern for Y
</p>
</li>
<li><p> lambda: nc-by-2 matrix. k-th row of lambda corresponds to the optimal
tuning parameters for k-th pattern pairs
</p>
</li>
<li><p> CR: average cross-validated sample covariance
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Woojoo Lee, Donghwan Lee, Youngjo Lee and Yudi Pawitan
</p>


<h3>References</h3>

<p>Lee, W., Lee, D., Lee, Y. and Pawitan, Y. (2011) Sparse Canonical
Covariance Analysis for High-throughput Data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example 1
## A very simple simulation example
n&lt;-10; p&lt;-50; q&lt;-20
X = matrix(rnorm(n*p),ncol=p)
Y = matrix(rnorm(n*q),ncol=q)
scca(X,Y)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
