<!DOCTYPE html><html lang="en"><head><title>Help for package highMLR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {highMLR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#hnscc'><p>High dimensional head and neck cancer survival and gene expression data</p></a></li>
<li><a href='#mlclassCox'><p>Applications of machine learning in survival analysis by prognostic classification of genes by CoxPH model.</p></a></li>
<li><a href='#mlclassKap'><p>Applications of machine learning in survival analysis by prognostic classification of genes by Kaplan-Meier estimator.</p></a></li>
<li><a href='#mlhighCox'><p>mlhighCox</p></a></li>
<li><a href='#mlhighFrail'><p>mlhighFrail</p></a></li>
<li><a href='#mlhighHet'><p>mlhighHet</p></a></li>
<li><a href='#mlhighKap'><p>mlhighKap</p></a></li>
<li><a href='#srdata'><p>High dimensional protein gene expression data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Feature Selection for High Dimensional Survival Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-07-17</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>mlr3, mlr3learners, survival, gtools, tibble, dplyr, utils,
coxme, missForest, R6</td>
</tr>
<tr>
<td>LazyData:</td>
<td>Yes</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>Yes</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform high dimensional Feature Selection in the presence of survival outcome.
            Based on Feature Selection method and different survival analysis, it will obtain the best 
            markers with optimal threshold levels according to their effect on disease progression
            and produce the most consistent level according to those threshold values. 
            The functions' methodology is based on by Sonabend et al (2021) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtab039">doi:10.1093/bioinformatics/btab039</a>&gt;  and Bhattacharjee  et al (2021) &lt;<a href="https://doi.org/10.48550/arXiv.2012.02102">doi:10.48550/arXiv.2012.02102</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Atanu Bhattacharjee &lt;atanustat@gmail.com&gt;</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-07-16 22:31:58 UTC; atanu</td>
</tr>
<tr>
<td>Author:</td>
<td>Atanu Bhattacharjee [aut, cre, ctb],
  Gajendra K. Vishwakarma [aut, ctb],
  Souvik Banerjee [aut, ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-07-18 08:10:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='hnscc'>High dimensional head and neck cancer survival and gene expression data</h2><span id='topic+hnscc'></span>

<h3>Description</h3>

<p>High dimensional head and neck cancer gene expression data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hnscc
</code></pre>


<h3>Format</h3>

<p>A dataframe with 565 rows and 104 variables
</p>

<dl>
<dt>ID</dt><dd><p>&quot;Column/Variable name&quot; consisting id of subjects</p>
</dd>
<dt>Death</dt><dd><p>&quot;Column/Variable name&quot; consisting survival event</p>
</dd>
<dt>OS</dt><dd><p>&quot;Column/Variable name&quot; consisting duration of overall survival</p>
</dd>
<dt>PFS</dt><dd><p>&quot;Column/Variable name&quot; consisting duration of progression free survival</p>
</dd>
<dt>Prog</dt><dd><p>&quot;Column/Variable name&quot; consisting progression event</p>
</dd>
<dt>GJB1,...,HMGCS2</dt><dd><p>High dimensional covariates</p>
</dd></dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(hnscc)
</code></pre>

<hr>
<h2 id='mlclassCox'>Applications of machine learning in survival analysis by prognostic classification of genes by CoxPH model.</h2><span id='topic+mlclassCox'></span>

<h3>Description</h3>

<p>Applications of machine learning in survival analysis by prognostic classification of genes by CoxPH model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mlclassCox(m, n, idSurv, idEvent, Time, s_ID, per = 20, fold = 3, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mlclassCox_+3A_m">m</code></td>
<td>
<p>Starting column number from where high dimensional variates to be selected.</p>
</td></tr>
<tr><td><code id="mlclassCox_+3A_n">n</code></td>
<td>
<p>Ending column number till where high dimensional variates to be selected.</p>
</td></tr>
<tr><td><code id="mlclassCox_+3A_idsurv">idSurv</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting duration of survival.</p>
</td></tr>
<tr><td><code id="mlclassCox_+3A_idevent">idEvent</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting survival event.</p>
</td></tr>
<tr><td><code id="mlclassCox_+3A_time">Time</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting Times of repeated observations.</p>
</td></tr>
<tr><td><code id="mlclassCox_+3A_s_id">s_ID</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting unique identification for each subject.</p>
</td></tr>
<tr><td><code id="mlclassCox_+3A_per">per</code></td>
<td>
<p>Percentage value for ordering, default=20.</p>
</td></tr>
<tr><td><code id="mlclassCox_+3A_fold">fold</code></td>
<td>
<p>Number of folds for re-sampling, default=3.</p>
</td></tr>
<tr><td><code id="mlclassCox_+3A_data">data</code></td>
<td>
<p>High dimensional data containing survival observations with multiple covariates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of genes as per their classifications
</p>

<dl>
<dt>GeneClassification</dt><dd><p>List of genes classified using Cox proportional hazard model</p>
</dd>
<dt>GeneClassification$Positive_Gene</dt><dd><p>Sublist of genes classified as positive genes</p>
</dd>
<dt>GeneClassification$Negative_Gene</dt><dd><p>Sublist of genes classified as negative genes</p>
</dd>
<dt>GeneClassification$Volatile_Gene</dt><dd><p>Sublist of genes classified as volatile genes</p>
</dd>
<dt>Result</dt><dd><p>A dataframe consisting threshold values with corresponding coefficients and p-values.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(srdata)
mlclassCox(m=50,n=59,idSurv="OS",idEvent="event",Time="Visit",s_ID="ID",per=20,fold=3,data=srdata)

## End(Not run)
</code></pre>

<hr>
<h2 id='mlclassKap'>Applications of machine learning in survival analysis by prognostic classification of genes by Kaplan-Meier estimator.</h2><span id='topic+mlclassKap'></span>

<h3>Description</h3>

<p>Applications of machine learning in survival analysis by prognostic classification of genes by Kaplan-Meier estimator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mlclassKap(m, n, idSurv, idEvent, Time, s_ID, per = 20, fold = 3, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mlclassKap_+3A_m">m</code></td>
<td>
<p>Starting column number from where high dimensional variates to be selected.</p>
</td></tr>
<tr><td><code id="mlclassKap_+3A_n">n</code></td>
<td>
<p>Ending column number till where high dimensional variates to be selected.</p>
</td></tr>
<tr><td><code id="mlclassKap_+3A_idsurv">idSurv</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting duration of survival.</p>
</td></tr>
<tr><td><code id="mlclassKap_+3A_idevent">idEvent</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting survival event.</p>
</td></tr>
<tr><td><code id="mlclassKap_+3A_time">Time</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting timepoints of repeated observations.</p>
</td></tr>
<tr><td><code id="mlclassKap_+3A_s_id">s_ID</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting unique identification for each subject.</p>
</td></tr>
<tr><td><code id="mlclassKap_+3A_per">per</code></td>
<td>
<p>Percentage value for ordering, default=20.</p>
</td></tr>
<tr><td><code id="mlclassKap_+3A_fold">fold</code></td>
<td>
<p>Number of fold for resampling, default=3.</p>
</td></tr>
<tr><td><code id="mlclassKap_+3A_data">data</code></td>
<td>
<p>High dimensional data containing survival observations and high dimensional covariates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of genes as per their classifications
</p>

<dl>
<dt>GeneClassification</dt><dd><p>List of genes classified using Cox proportional hazard model</p>
</dd>
<dt>GeneClassification$Positive_Gene</dt><dd><p>Sublist of genes classified as positive genes</p>
</dd>
<dt>GeneClassification$Negative_Gene</dt><dd><p>Sublist of genes classified as negative genes</p>
</dd>
<dt>GeneClassification$Volatile_Gene</dt><dd><p>Sublist of genes classified as volatile genes</p>
</dd>
<dt>Result</dt><dd><p>A dataframe consisting threshold values with corresponding coefficients and p-values.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'> ## Not run: 
##
mlclassKap(m=50,n=59,idSurv="OS",idEvent="event",Time="Visit",s_ID="ID",per=20,fold=3,data=srdata)
##

## End(Not run)
</code></pre>

<hr>
<h2 id='mlhighCox'>mlhighCox</h2><span id='topic+mlhighCox'></span>

<h3>Description</h3>

<p>This function extracts desired number of features based on minimum log-Loss function
using Cox proportional hazard model as learner method on a high dimensional survival data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mlhighCox(cols, idSurv, idEvent, per = 20, fold = 3, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mlhighCox_+3A_cols">cols</code></td>
<td>
<p>A numeric vector of column numbers indicating the features for which the log Loss functions are to be computed</p>
</td></tr>
<tr><td><code id="mlhighCox_+3A_idsurv">idSurv</code></td>
<td>
<p>The name of the survival time variable</p>
</td></tr>
<tr><td><code id="mlhighCox_+3A_idevent">idEvent</code></td>
<td>
<p>The name of the survival event variable</p>
</td></tr>
<tr><td><code id="mlhighCox_+3A_per">per</code></td>
<td>
<p>Percentage of total features to be selected, default value 20</p>
</td></tr>
<tr><td><code id="mlhighCox_+3A_fold">fold</code></td>
<td>
<p>An integer denoting number of folds in cross validation, default value 3</p>
</td></tr>
<tr><td><code id="mlhighCox_+3A_data">data</code></td>
<td>
<p>A data frame that contains the survival and covariate information for the subjects</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Performs feature Selection using Cox PH on high-dimensional data
</p>
<p>Using the Cox proportional hazard model on the given survival data, this function selects
the most significant feature based on a performance measure. The performance measure is considered as logarithmic loss function. It is defined as,
</p>
<p style="text-align: center;"><code class="reqn">L(f,t)=-log(f(t))</code>
</p>
<p>. The features with minimum log-loss function are extracted.
</p>


<h3>Value</h3>

<p>A dataframe containing desired number of features and the corresponding log Loss function.
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee, Gajendra K. Vishwakarma &amp; Souvik Banerjee
</p>


<h3>References</h3>

<p>Sonabend, R., Király, F. J., Bender, A., Bernd Bischl B. and Lang M. mlr3proba: An R Package for Machine Learning in Survival Analysis, 2021, Bioinformatics, &lt;https://doi.org/10.1093/bioinformatics/btab039&gt;
</p>


<h3>See Also</h3>

<p>mlhighKap, mlhighFrail
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(hnscc)
mlhighCox(cols=c(6:15), idSurv="OS", idEvent="Death", per=20, fold = 3, data=hnscc)

## End(Not run)
</code></pre>

<hr>
<h2 id='mlhighFrail'>mlhighFrail</h2><span id='topic+mlhighFrail'></span>

<h3>Description</h3>

<p>This function extracts  features based on minimum log-Loss function using Cox proportional hazard model as learner method on a
high dimensional survival data. For those genes, we obtain frailty variances using CoxPH.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mlhighFrail(
  cols,
  idSurv,
  idEvent,
  idFrail,
  dist = "gaussian",
  per = 20,
  fold = 3,
  data
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mlhighFrail_+3A_cols">cols</code></td>
<td>
<p>A numeric vector of column numbers indicating the features for which the log Loss functions are to be computed</p>
</td></tr>
<tr><td><code id="mlhighFrail_+3A_idsurv">idSurv</code></td>
<td>
<p>The name of the survival time variable</p>
</td></tr>
<tr><td><code id="mlhighFrail_+3A_idevent">idEvent</code></td>
<td>
<p>The name of the survival event variable</p>
</td></tr>
<tr><td><code id="mlhighFrail_+3A_idfrail">idFrail</code></td>
<td>
<p>The name of the frailty variable</p>
</td></tr>
<tr><td><code id="mlhighFrail_+3A_dist">dist</code></td>
<td>
<p>The name of the frailty distribution. Options are &quot;gamma&quot;, &quot;gaussian&quot; or &quot;t&quot;, default is &quot;gaussian&quot;</p>
</td></tr>
<tr><td><code id="mlhighFrail_+3A_per">per</code></td>
<td>
<p>Percentage of features to be selected, default value 20</p>
</td></tr>
<tr><td><code id="mlhighFrail_+3A_fold">fold</code></td>
<td>
<p>An integer denoting number of folds in cross validation, default value 3</p>
</td></tr>
<tr><td><code id="mlhighFrail_+3A_data">data</code></td>
<td>
<p>A data frame that contains the survival and covariate information for the subjects</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Performs CoxPH frailty on high doimensional survival data
</p>
<p>Using the Cox proportional hazard model on the given survival data, this function selects the most significant feature based on minimum logarithmic loss function. The logarithmic loss function is defined as,
</p>
<p style="text-align: center;"><code class="reqn">L(f,t)=-log(f(t))</code>
</p>

<p>After selcting the most significant features, a Cox proportional hazard frailty model is fitted on the selected features. The CoxPH frailty model is defined as,
</p>
<p style="text-align: center;"><code class="reqn">\lambda(t)=\lambda 0(t)\nu exp{X'\beta}</code>
</p>
<p> where <code class="reqn">\nu</code> is called the frailty component. The variance of the
frailty term is considered as the heterogeneity among the subjects or patients. The distribution of frailty component is considered as either Gaussian, Gamma or t distribution.
</p>


<h3>Value</h3>

<p>A dataframe containing desired number of features with corresponding frailty variances.
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee, Gajendra K. Vishwakarma &amp; Souvik Banerjee
</p>


<h3>References</h3>

<p>Sonabend, R., Kiraly, F. J., Bender, A., Bernd Bischl B. and Lang M. mlr3proba: An R Package for Machine Learning in Survival Analysis, 2021, Bioinformatics, &lt;https://doi.org/10.1093/bioinformatics/btab039&gt;
</p>


<h3>See Also</h3>

<p>mlhighHet, mlhighCox
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(hnscc)
mlhighFrail(cols=c(10:20), idSurv="OS", idEvent="Death", idFrail="ID", dist="gaussian",
per=20, fold = 3, data=hnscc)

## End(Not run)
</code></pre>

<hr>
<h2 id='mlhighHet'>mlhighHet</h2><span id='topic+mlhighHet'></span>

<h3>Description</h3>

<p>This function extracts features based on ML method, finds optimal cut-off values of features using sequencial
Cox PH model and obtain the most consistent level according to the cut-offs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mlhighHet(cols, idSurv, idEvent, idFrail, num, fold = 3, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mlhighHet_+3A_cols">cols</code></td>
<td>
<p>A numeric vector of column numbers indicating the features for which the log Loss functions are to be computed</p>
</td></tr>
<tr><td><code id="mlhighHet_+3A_idsurv">idSurv</code></td>
<td>
<p>The name of the survival time variable</p>
</td></tr>
<tr><td><code id="mlhighHet_+3A_idevent">idEvent</code></td>
<td>
<p>The name of the survival event variable</p>
</td></tr>
<tr><td><code id="mlhighHet_+3A_idfrail">idFrail</code></td>
<td>
<p>The name of the frailty variable</p>
</td></tr>
<tr><td><code id="mlhighHet_+3A_num">num</code></td>
<td>
<p>Number of features to be selected</p>
</td></tr>
<tr><td><code id="mlhighHet_+3A_fold">fold</code></td>
<td>
<p>An integer denoting number of folds in cross validation, default value 3</p>
</td></tr>
<tr><td><code id="mlhighHet_+3A_data">data</code></td>
<td>
<p>A data frame that contains the survival and covariate information for the subjects</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Performs heterogeneity analysis in gene expression
</p>
<p>This function extracts features based on minimum log-Loss function using Cox proportional hazard model as learner method on a high dimensional survival data.
For those selected genes, we obtain optimal cutoff values using minimum p-value in a Cox PH model. The Cox PH model is used sequencially for each combination of genes
and all possible gene combinations are tested to obtain best possible combination with minimum BIC value. The subjects are classified according to different levels of those genes. Using a Cox PH frailty model, we obtain the most consistent level for which the frailty variance is minimum.
The data is splited using cross validation technique. The performance measure is considered as logarithmic loss function. It is defined as,
</p>
<p style="text-align: center;"><code class="reqn">L(f,t)=-log(f(t))</code>
</p>

<p>The CoxPH frailty model is defined as,
</p>
<p style="text-align: center;"><code class="reqn">\lambda(t)=\lambda 0(t)\nu exp{X'\beta}</code>
</p>
<p> where <code class="reqn">\nu</code> is called the frailty. The variance of the
frailty term is considered as the heterogeneity among the subjects or patients. Gaussian distribution with mean 0 is considered for the distribution of frailty component.
</p>


<h3>Value</h3>

<p>dataframes containing optimal gene cutoff values and most consistent level according to those cut-offs with frailty variance.
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee, Gajendra K. Vishwakarma &amp; Souvik Banerjee
</p>


<h3>References</h3>

<p>Sonabend, R., Király, F. J., Bender, A., Bernd Bischl B. and Lang M. mlr3proba: An R Package for Machine Learning in Survival Analysis, 2021, Bioinformatics, &lt;https://doi.org/10.1093/bioinformatics/btab039&gt;
</p>
<p>Bhattacharjee, A. Vishwakarma, G.K. and Banerjee, S. A modified risk detection approach of biomarkers by frailty effect on multiple time to event data, 2020, &lt;arXiv:2012.02102&gt;.
</p>


<h3>See Also</h3>

<p>mlhighCox, mlhighFrail
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(hnscc)
mlhighHet(cols=c(27:32), idSurv="OS", idEvent="Death", idFrail="ID", num=2, fold = 3, data=hnscc)

## End(Not run)
</code></pre>

<hr>
<h2 id='mlhighKap'>mlhighKap</h2><span id='topic+mlhighKap'></span>

<h3>Description</h3>

<p>This function extracts desired number of features based on minimum log-Loss function using Kaplan Meier model as learner
method on a high dimensional survival data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mlhighKap(cols, idSurv, idEvent, per = 20, fold = 3, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mlhighKap_+3A_cols">cols</code></td>
<td>
<p>A numeric vector of column numbers indicating the features for which the log Loss functions are to be computed</p>
</td></tr>
<tr><td><code id="mlhighKap_+3A_idsurv">idSurv</code></td>
<td>
<p>The name of the survival time variable</p>
</td></tr>
<tr><td><code id="mlhighKap_+3A_idevent">idEvent</code></td>
<td>
<p>The name of the survival event variable</p>
</td></tr>
<tr><td><code id="mlhighKap_+3A_per">per</code></td>
<td>
<p>Percentage of features to be selected, default value 20</p>
</td></tr>
<tr><td><code id="mlhighKap_+3A_fold">fold</code></td>
<td>
<p>An integer denoting number of folds in cross validation, default value 3</p>
</td></tr>
<tr><td><code id="mlhighKap_+3A_data">data</code></td>
<td>
<p>A data frame that contains the survival and covariate information for the subjects</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Performs feature selection using Kaplan Meier method
</p>
<p>Using the Kaplan Meier method on the given survival data, this function selects the most significant feature based on a performance measure. The performance measure is considered as logarithmic loss function. It is defined as,
</p>
<p style="text-align: center;"><code class="reqn">L(f,t)=-log(f(t))</code>
</p>
<p>. The features with minimum log-loss function are extracted.
</p>


<h3>Value</h3>

<p>A dataframe containing desired number of features based on minimum log Loss function
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee, Gajendra K. Vishwakarma &amp; Souvik Banerjee
</p>


<h3>References</h3>

<p>Sonabend, R., Kiraly, F. J., Bender, A., Bernd Bischl B. and Lang M. mlr3proba: An R Package for Machine Learning in Survival Analysis, 2021, Bioinformatics
</p>


<h3>See Also</h3>

<p>mlhighCox
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(hnscc)
mlhighKap(cols=c(6:15), idSurv="OS", idEvent="Death", per=20, fold = 3, data=hnscc)

## End(Not run)
</code></pre>

<hr>
<h2 id='srdata'>High dimensional protein gene expression data</h2><span id='topic+srdata'></span>

<h3>Description</h3>

<p>High dimensional protein gene expression data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srdata
</code></pre>


<h3>Format</h3>

<p>A dataframe with 288 rows and 250 variables
</p>

<dl>
<dt>ID</dt><dd><p>&quot;Column/Variable name&quot; consisting id of subjects</p>
</dd>
<dt>Visit</dt><dd><p>&quot;Column/Variable name&quot; consisting number of times observations recorded</p>
</dd>
<dt>event</dt><dd><p>&quot;Column/Variable name&quot; consisting survival event</p>
</dd>
<dt>OS</dt><dd><p>&quot;Column/Variable name&quot; consisting duration of overall survival</p>
</dd>
<dt>C6kine,.....,GFRalpha4</dt><dd><p>High dimensional covariates</p>
</dd></dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(srdata)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
