<!DOCTYPE html><html><head><title>Help for package hilldiv</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hilldiv}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alpha_div'><p>Alpha diversity</p></a></li>
<li><a href='#bat.diet.hierarchy'><p>Bat diet hierarchy</p></a></li>
<li><a href='#bat.diet.otutable'><p>Bat diet OTU table</p></a></li>
<li><a href='#bat.diet.tree'><p>Bat diet OTU tree</p></a></li>
<li><a href='#beta_dis'><p>Beta dissimilarity</p></a></li>
<li><a href='#copy_filt'><p>OTU/ASV copy number filtering</p></a></li>
<li><a href='#CqN'><p>Sørensen-type overlap</p></a></li>
<li><a href='#depth_cov'><p>Depth coverage assessment</p></a></li>
<li><a href='#depth_filt'><p>Sequencing depth filtering</p></a></li>
<li><a href='#div_part'><p>Multi-level diversity partitioning</p></a></li>
<li><a href='#div_profile'><p>Diversity profile</p></a></li>
<li><a href='#div_profile_plot'><p>Diversity profile plot</p></a></li>
<li><a href='#div_test'><p>Diversity test</p></a></li>
<li><a href='#div_test_plot'><p>Diversity test plotting</p></a></li>
<li><a href='#gamma_div'><p>Gamma diversity</p></a></li>
<li><a href='#hill_div'><p>Hill numbers computation</p></a></li>
<li><a href='#index_div'><p>Diversity index computation</p></a></li>
<li><a href='#is.nested'><p>Check if hierachy is nested</p></a></li>
<li><a href='#match_data'><p>Match data</p></a></li>
<li><a href='#pair_dis'><p>Pairwise dissimilarity</p></a></li>
<li><a href='#pair_dis_plot'><p>Pairwise dissimilarity plot</p></a></li>
<li><a href='#SqN'><p>Jaccard-type turnover-complement</p></a></li>
<li><a href='#to.incidence'><p>Hill numbers computation</p></a></li>
<li><a href='#tree_depth'><p>Tree depth</p></a></li>
<li><a href='#tss'><p>Total Sum Scaling normalisation</p></a></li>
<li><a href='#UqN'><p>Jaccard-type overlap</p></a></li>
<li><a href='#VqN'><p>Sørensen-type turnover-complement</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Integral Analysis of Diversity Based on Hill Numbers</td>
</tr>
<tr>
<td>Version:</td>
<td>1.5.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Antton Alberdi [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Antton Alberdi &lt;antton.alberdi@bio.ku.dk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for analysing, comparing, visualising and partitioning diversity based on Hill numbers.
  'hilldiv' is an R package that provides a set of functions to assist analysis of diversity for
  diet reconstruction, microbial community profiling or more general ecosystem characterisation
  analyses based on Hill numbers, using OTU/ASV tables and associated phylogenetic trees as
  inputs. The package includes functions for (phylo)diversity measurement, (phylo)diversity
  profile plotting, (phylo)diversity comparison between samples and groups, (phylo)diversity
  partitioning and (dis)similarity measurement. All of these grounded in abundance-based and
  incidence-based Hill numbers.
  The statistical framework developed around Hill numbers encompasses many of the most
  broadly employed diversity (e.g. richness, Shannon index, Simpson index),
  phylogenetic diversity (e.g. Faith's PD, Allen's H, Rao's quadratic entropy) and
  dissimilarity (e.g. Sorensen index, Unifrac distances) metrics. This enables the most
  common analyses of diversity to be performed while grounded in a single statistical
  framework. The methods are described in Jost et al. (2007) &lt;<a href="https://doi.org/10.1890%2F06-1736.1">doi:10.1890/06-1736.1</a>&gt;,
  Chao et al. (2010) &lt;<a href="https://doi.org/10.1098%2Frstb.2010.0272">doi:10.1098/rstb.2010.0272</a>&gt; and Chiu et al. (2014)
  &lt;<a href="https://doi.org/10.1890%2F12-0960.1">doi:10.1890/12-0960.1</a>&gt;; and reviewed in the framework of molecularly characterised
  biological systems in Alberdi &amp; Gilbert (2019) &lt;<a href="https://doi.org/10.1111%2F1755-0998.13014">doi:10.1111/1755-0998.13014</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/anttonalberdi/hilldiv">https://github.com/anttonalberdi/hilldiv</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/anttonalberdi/hilldiv/issues">https://github.com/anttonalberdi/hilldiv/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, ggplot2, scales, ggpubr, RColorBrewer, data.table, ape,
vegan, geiger, qgraph, FSA</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Collate:</td>
<td>'index_div.R' 'hill_div.R' 'div_profile.R'
'div_profile_plot.R' 'div_test.R' 'div_test_plot.R'
'depth_cov.R' 'div_part.R' 'alpha_div.R' 'gamma_div.R'
'beta_dis.R' 'pair_dis.R' 'pair_dis_plot.R' 'UqN.R' 'CqN.R'
'VqN.R' 'SqN.R' 'match_data.R' 'depth_filt.R' 'copy_filt.R'
'to.incidence.R' 'tss.R' 'is.nested.R' 'tree_depth.R' 'data.R'</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-10-01 13:32:38 UTC; jpl786</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-10-01 14:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='alpha_div'>Alpha diversity</h2><span id='topic+alpha_div'></span>

<h3>Description</h3>

<p>Compute alpha diversity of a system comprised of multiple samples from a count (OTU/ASV) table. If a tree object is provided, the computed alpha diversity accounts for the phylogenetic relations across OTUs/ASVs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha_div(countable,qvalue,tree,weight)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alpha_div_+3A_countable">countable</code></td>
<td>
<p>A count table (matrix/data.frame) indicating the absolute or relative OTU/ASV abundances of multiple samples. Columns must refer to samples and rows to OTUs/ASVs.</p>
</td></tr>
<tr><td><code id="alpha_div_+3A_qvalue">qvalue</code></td>
<td>
<p>A positive number, usually between 0 and 5, but most commonly 0, 1 or 2. It can be an integer or contain decimals.</p>
</td></tr>
<tr><td><code id="alpha_div_+3A_tree">tree</code></td>
<td>
<p>A phylogenetic tree of class 'phylo'. The tip labels must match the row names in the count table. Use the function match_data() if the count table and tree names do not match.</p>
</td></tr>
<tr><td><code id="alpha_div_+3A_weight">weight</code></td>
<td>
<p>A vector indicating the relative weight of each sample. The order needs to be identical to the order of the samples in the OTU table. The values need to sum up to 1. If empty, all samples are weighed the same.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Alpha diversity computation (based on Hill numbers)
</p>


<h3>Value</h3>

<p>An alpha diversity value.
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Hsieh, T. C. (2012). Proposing a resolution to debates on diversity partitioning. Ecology, 93, 2037-2051.<br /><br />
Jost, L. (2007). Partitioning diversity into independent alpha and beta components. Ecology, 88, 2427-2439.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_part">div_part</a></code>, <code><a href="#topic+gamma_div">gamma_div</a></code>, <code><a href="#topic+match_data">match_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
alpha_div(countable=bat.diet.otutable,qvalue=1)
alpha_div(countable=bat.diet.otutable,qvalue=1,tree=bat.diet.tree)
weight.vector = rep(1/ncol(bat.diet.otutable),ncol(bat.diet.otutable))
alpha_div(bat.diet.otutable,1,bat.diet.tree,weight.vector)
</code></pre>

<hr>
<h2 id='bat.diet.hierarchy'>Bat diet hierarchy</h2><span id='topic+bat.diet.hierarchy'></span>

<h3>Description</h3>

<p>Hierarchy table indicating the relationship between samples and their
respective parent groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bat.diet.hierarchy
</code></pre>


<h3>Format</h3>

<p>A data frame with 40 rows and 2 columns.</p>

<hr>
<h2 id='bat.diet.otutable'>Bat diet OTU table</h2><span id='topic+bat.diet.otutable'></span>

<h3>Description</h3>

<p>An OTU table containing the absolute read abundances of 363 OTUs in 40
faecal samples from 8 different bat species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bat.diet.otutable
</code></pre>


<h3>Format</h3>

<p>A data frame with 363 rows and 40 species.</p>

<hr>
<h2 id='bat.diet.tree'>Bat diet OTU tree</h2><span id='topic+bat.diet.tree'></span>

<h3>Description</h3>

<p>Phylogenetic tree built from the representative sequences of the 363 OTUs
included in the 'bat.diet.otutable' data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bat.diet.tree
</code></pre>


<h3>Format</h3>

<p>A phylo object with 363 tips and 362 internal nodes.</p>

<hr>
<h2 id='beta_dis'>Beta dissimilarity</h2><span id='topic+beta_dis'></span>

<h3>Description</h3>

<p>Compute dissimilarity or similarity values based on beta diversities (neutral or phylogenetic) and sample size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beta_dis(beta, qvalue, N, metric, type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="beta_dis_+3A_beta">beta</code></td>
<td>
<p>A numeric beta diversity value or an object outputted by function div_part() (which contains all the information to compute (dis)similarities).</p>
</td></tr>
<tr><td><code id="beta_dis_+3A_qvalue">qvalue</code></td>
<td>
<p>A positive number, usually between 0 and 5, but most commonly 0, 1 or 2. It can be an integer or contain decimals.</p>
</td></tr>
<tr><td><code id="beta_dis_+3A_n">N</code></td>
<td>
<p>An integer indicating sample size, the number of sampling units to be used to compute the (dis)similarity measures. The argument is ovewritten if a 'div_part' object is used.</p>
</td></tr>
<tr><td><code id="beta_dis_+3A_metric">metric</code></td>
<td>
<p>A vector containing &quot;C&quot;, &quot;U&quot;, &quot;V&quot; or &quot;S&quot;. C: Sørensen-type overlap or complement. U: Jaccard-type overlap or complement. V: Sørensen-type turnover or complement. S: Jaccard-type turnover or complement. See hilldiv wiki for further information.</p>
</td></tr>
<tr><td><code id="beta_dis_+3A_type">type</code></td>
<td>
<p>A character object containing either &quot;similarity&quot; or &quot;dissimilarity&quot;. If 'similarity' is used, similarity metrics (0: completely different composition - 1: identical composition) are returned. If 'dissimilarity' is used, dissimilarity metrics (0: identical composition - 1:completely different composition) are returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>(Dis)similarity computation from beta diversities based on Hill numbers
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Hsieh, T. C. (2012). Proposing a resolution to debates on diversity partitioning. Ecology, 93, 2037-2051.<br /><br />
Jost, L. (2007). Partitioning diversity into independent alpha and beta components. Ecology, 88, 2427-2439.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_part">div_part</a></code>, <code><a href="#topic+gamma_div">gamma_div</a></code>, <code><a href="#topic+pair_dis">pair_dis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
#Manually indicating beta diversity, order of diversity and sample size
beta_dis(beta=4.5,qvalue=1,N=8)
beta_dis(beta=4.5,qvalue=1,N=8,metric="C",type="similarity")
#Using an object created with the function div_part()
divpartobject &lt;- div_part(bat.diet.otutable,qvalue=0,tree=bat.diet.tree)
beta_dis(divpartobject)
beta_dis(divpartobject,metric="S",type="similarity")
</code></pre>

<hr>
<h2 id='copy_filt'>OTU/ASV copy number filtering</h2><span id='topic+copy_filt'></span>

<h3>Description</h3>

<p>As DNA sequencing data include PCR and sequencing errors, copy number thresholds are commonly applied to discard the OTUs with low number of sequence copies. This threshold can be absolute or (ideally) relative to the sequencing depth of each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>copy_filt(abund, threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="copy_filt_+3A_abund">abund</code></td>
<td>
<p>A vector or a matrix/data.frame indicating the relative abundances of one or multiple samples, respectively. If a matrix/data.frame is provided, columns must refer to samples and rows to OTUs.</p>
</td></tr>
<tr><td><code id="copy_filt_+3A_threshold">threshold</code></td>
<td>
<p>An integer or a decimal number indicating the cut-off threshold. If an integer is provided, an absolute threshold is used (same threshold for all samples). If a decimal number is provided a relative copy number threshold is applied (dependent on the sequencing depth of each sample).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>OTU/ASV copy number filtering
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi A, Aizpurua O, Bohmann K, Gopalakrishnan S, Lynggaard C, Nielsen M, Gilbert MTP. 2019. Promises and pitfalls of using high-throughput sequencing for diet analysis. Molecular Ecology Resources, 19(2), 327-348.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+depth_cov">depth_cov</a></code>, <code><a href="#topic+tss">tss</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
#Remove singletons from all samples
copy_filt(bat.diet.otutable,2)
#Remove OTUs represented by less than 0.01% of the total reads per sample.
copy_filt(bat.diet.otutable,0.0001)
</code></pre>

<hr>
<h2 id='CqN'>Sørensen-type overlap</h2><span id='topic+CqN'></span>

<h3>Description</h3>

<p>The Sørensen-type overlap quantifies the effective average proportion of a sub-systems OTUs (or lineages in the case of phylodiversities) that is shared across all subsystems. This is thus a metric that quantifies overlap from the subsystems perspective. Its corresponding dissimilarity measure (1 - CqN) quantifies the effective average proportion of nonshared OTUs or lineages in a system. CqN is integrated in the functions beta_dis() and pair_dis().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CqN(beta, qvalue, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CqN_+3A_beta">beta</code></td>
<td>
<p>A beta diversity value based on Hill numbers.</p>
</td></tr>
<tr><td><code id="CqN_+3A_qvalue">qvalue</code></td>
<td>
<p>The q value used to compute the beta diversity. It needs to be a positive number, usually between 0 and 5, but most commonly 0, 1 or 2. It can be an integer or contain decimals.</p>
</td></tr>
<tr><td><code id="CqN_+3A_n">N</code></td>
<td>
<p>An integer indicating sample size, the number of sampling units to be used to compute the similarity measure.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sørensen-type overlap
</p>


<h3>Value</h3>

<p>A Sørensen-type overlap value
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Hsieh, T. C. (2012). Proposing a resolution to debates on diversity partitioning. Ecology, 93, 2037-2051.<br /><br />
Jost, L. (2007). Partitioning diversity into independent alpha and beta components. Ecology, 88, 2427-2439.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_part">div_part</a></code>, <code><a href="#topic+beta_dis">beta_dis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>CqN(beta=1.24,qvalue=1,N=3)
CqN(1.24,1,3)
</code></pre>

<hr>
<h2 id='depth_cov'>Depth coverage assessment</h2><span id='topic+depth_cov'></span>

<h3>Description</h3>

<p>Coverage of the estimated Hill numbers at different orders of diversity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>depth_cov(abund, qvalue)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="depth_cov_+3A_abund">abund</code></td>
<td>
<p>A vector or a matrix/data.frame indicating the relative abundances of one or multiple samples, respectively. If a matrix/data.frame is provided, columns must refer to samples and rows to OTUs.</p>
</td></tr>
<tr><td><code id="depth_cov_+3A_qvalue">qvalue</code></td>
<td>
<p>A positive integer or decimal number (&gt;=0), usually between 0 and 3.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Depth coverage assessment
</p>


<h3>Value</h3>

<p>A matrix with observed diversity,  estimated diversities and coverage
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Chao, A. &amp; Jost, L. (2015) Estimating diversity and entropy profiles via discovery rates of new species. Methods in Ecology and Evolution, 6, 873-882.<br /><br />
Jost, L. (2006). Entropy and diversity. Oikos, 113, 363-375.<br /><br />
Hill, M. O. (1973). Diversity and evenness: a unifying notation and its consequences. Ecology, 54, 427-432.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hill_div">hill_div</a></code>, <code><a href="#topic+depth_filt">depth_filt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
depth_cov(bat.diet.otutable,0)
depth_cov(bat.diet.otutable,qvalue=1)
</code></pre>

<hr>
<h2 id='depth_filt'>Sequencing depth filtering</h2><span id='topic+depth_filt'></span>

<h3>Description</h3>

<p>Filter samples based on a minimum sequencing depth.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>depth_filt(countable, threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="depth_filt_+3A_countable">countable</code></td>
<td>
<p>An OTU table (matrix/data.frame) indicating the absolute OTU abundances of multiple samples. Columns must refer to samples and rows to OTUs.</p>
</td></tr>
<tr><td><code id="depth_filt_+3A_threshold">threshold</code></td>
<td>
<p>A number indicating the minimum sequencing depth required to keep the sample.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sequencing depth filtering
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi A, Aizpurua O, Bohmann K, Gopalakrishnan S, Lynggaard C, Nielsen M, Gilbert MTP. 2019. Promises and pitfalls of using high-throughput sequencing for diet analysis. Molecular Ecology Resources, 19(2), 327-348.<br /><br />
</p>


<h3>See Also</h3>

<p><code><a href="#topic+depth_cov">depth_cov</a></code>, <code><a href="#topic+copy_filt">copy_filt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
depth_filt(bat.diet.otutable,5000)
depth_filt(bat.diet.otutable,threshold=20000)
</code></pre>

<hr>
<h2 id='div_part'>Multi-level diversity partitioning</h2><span id='topic+div_part'></span>

<h3>Description</h3>

<p>Multi-level diversity partitioning following the multiplicative definition based on Hill numbers. Hierarchical levels are defined from L1 (minimum, sample) to Ln (maximum, whole system), and as many intermediate levels as wanted can be defined in between. The hierarchical structure of the system is defined with the hierarchy table. If no hierarchy table is inputed, the function yields a simple two-level partitioning between alpha (L1), beta and gamma (L2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>div_part(countable, qvalue, tree, hierarchy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="div_part_+3A_countable">countable</code></td>
<td>
<p>A count table (matrix/data.frame) indicating the absolute or relative OTU/ASV abundances of multiple samples. Columns must refer to samples and rows to OTUs/ASVs.</p>
</td></tr>
<tr><td><code id="div_part_+3A_qvalue">qvalue</code></td>
<td>
<p>A positive number, usually between 0 and 5, but most commonly 0, 1 or 2. It can be an integer or contain decimals.</p>
</td></tr>
<tr><td><code id="div_part_+3A_tree">tree</code></td>
<td>
<p>A phylogenetic tree of class 'phylo'. The tip labels must match the row names in the OTU table. Use the function match_data() if the OTU names do not match.</p>
</td></tr>
<tr><td><code id="div_part_+3A_hierarchy">hierarchy</code></td>
<td>
<p>A matrix indicating the relation between samples (first column) and parent group(s).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Multi-level diversity partitioning (based on Hill numbers)
</p>


<h3>Value</h3>

<p>A list object containing details of hierarchical diversity partitioning.
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Hsieh, T. C. (2012). Proposing a resolution to debates on diversity partitioning. Ecology, 93, 2037-2051.<br /><br />
Jost, L. (2007). Partitioning diversity into independent alpha and beta components. Ecology, 88, 2427–2439.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_part">div_part</a></code>, <code><a href="#topic+gamma_div">gamma_div</a></code>, <code><a href="#topic+match_data">match_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
data(bat.diet.hierarchy)
#Two level examples (L1=sample (alpha diversity), L2=whole system (gamma diversity))
div_part(bat.diet.otutable,qvalue=1)
div_part(bat.diet.otutable,qvalue=0,tree=bat.diet.tree)
#Three-level example (L1=sample, L2=species, L3=whole system)
div_part(bat.diet.otutable,qvalue=0,hierarchy=bat.diet.hierarchy)
</code></pre>

<hr>
<h2 id='div_profile'>Diversity profile</h2><span id='topic+div_profile'></span>

<h3>Description</h3>

<p>Create diversity profile vectors (single sample or system) or tables (multiple samples or groups) from count tables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>div_profile(count, qvalues, tree, hierarchy, level)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="div_profile_+3A_count">count</code></td>
<td>
<p>A vector or a matrix indicating the (relative) OTU/ASV counts of one or multiple samples. If a matrix is provided, columns must refer to samples and rows to OTUs.</p>
</td></tr>
<tr><td><code id="div_profile_+3A_qvalues">qvalues</code></td>
<td>
<p>A vector of sequential orders of diversity (default from 0 to 5). order=seq(from = 0, to = 5, by = (0.1))</p>
</td></tr>
<tr><td><code id="div_profile_+3A_tree">tree</code></td>
<td>
<p>A tree of class 'phylo'. The tip labels must match the names of the vector values (if one sample) or matrix rows (if multiple samples).</p>
</td></tr>
<tr><td><code id="div_profile_+3A_hierarchy">hierarchy</code></td>
<td>
<p>A two-column matrix indicating the relation between samples (first column) and groups (second column).</p>
</td></tr>
<tr><td><code id="div_profile_+3A_level">level</code></td>
<td>
<p>Whether to compute alpha or gamma diversities of the system or the groups specified in the hierarchy table.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Diversity profile
</p>


<h3>Value</h3>

<p>A vector or matrix containing diversity values at different orders of diversity (as specified in qvalues).
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Jost, L. (2014). Unifying species diversity, phylogenetic diversity, functional diversity, and related similarity and differentiation measures through hill numbers. Annual Review of Ecology Evolution and Systematics, 45, 297-324.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_profile_plot">div_profile_plot</a></code>, <code><a href="#topic+hill_div">hill_div</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
data(bat.diet.hierarchy)
#One sample example
bat.diet.sample &lt;- bat.diet.otutable[,1]
div_profile(count=bat.diet.sample,qvalues=seq(from = 0, to = 5, by = (0.1)))
#One sample example (phylogenetic Hill numbers)
names(bat.diet.sample) &lt;- rownames(bat.diet.otutable)
div_profile(count=bat.diet.sample,qvalues=seq(from = 0, to = 5, by = (0.1)),tree=bat.diet.tree)
#Multiple samples
div_profile(bat.diet.otutable)
#Multiple groups (gamma diversity)
div_profile(bat.diet.otutable,hierarchy=bat.diet.hierarchy,level="gamma")
#Multiple groups (alpha diversity)
div_profile(bat.diet.otutable,hierarchy=bat.diet.hierarchy,level="alpha")
</code></pre>

<hr>
<h2 id='div_profile_plot'>Diversity profile plot</h2><span id='topic+div_profile_plot'></span>

<h3>Description</h3>

<p>Plot diversity profiles from objects generated with the function div_profile().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>div_profile_plot(profile, colour, log, legend)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="div_profile_plot_+3A_profile">profile</code></td>
<td>
<p>A div_profile() object or a vector/matrix containg diversity profile(s), with columns indicating samples/groups and rows indicating orders of diversity (q-values).</p>
</td></tr>
<tr><td><code id="div_profile_plot_+3A_colour">colour</code></td>
<td>
<p>A vector of RGB colours e.g. c(&quot;#34k235&quot;,&quot;#99cc00&quot;). The number of vector items, must equal the number of samples or groups that are intended to plot.</p>
</td></tr>
<tr><td><code id="div_profile_plot_+3A_log">log</code></td>
<td>
<p>Whether to transform Hill numbers to logarithmic scale (TRUE) or not (FALSE). This is useful when there are large differences between q values (e.g. sharp drop from q=0 to q=1), which might complicate visualization. Default: log=FALSE</p>
</td></tr>
<tr><td><code id="div_profile_plot_+3A_legend">legend</code></td>
<td>
<p>Whether to display the legend (TRUE) or not (FALSE) in diversity profiles containing multiple samples/groups. Default TRUE in multi-sample charts.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Diversity profile plot
</p>


<h3>Value</h3>

<p>A diversity profile plot.
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Jost, L. (2014). Unifying species diversity, phylogenetic diversity, functional diversity, and related similarity and differentiation measures through hill numbers. Annual Review of Ecology Evolution and Systematics, 45, 297-324.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_profile">div_profile</a></code>, <code><a href="#topic+hill_div">hill_div</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.hierarchy)
#One sample example
bat.diet.sample &lt;- bat.diet.otutable[,1]
profile.onesample &lt;- div_profile(count=bat.diet.sample,qvalues=seq(from = 0, to = 5, by = (0.1)))
div_profile_plot(profile.onesample)
#Multiple samples
profile.multiplesamples &lt;- div_profile(bat.diet.otutable)
div_profile_plot(profile.multiplesamples)
#Multiple groups (gamma diversity)
profile.multiplegroups &lt;- div_profile(bat.diet.otutable,hierarchy=bat.diet.hierarchy,level="gamma")
div_profile_plot(profile.multiplegroups)
</code></pre>

<hr>
<h2 id='div_test'>Diversity test</h2><span id='topic+div_test'></span>

<h3>Description</h3>

<p>Diversity comparison test between groups of samples. The function automatically assesses whether the data meets the properties for parametric statistics and performs the appropriate test accordingly: Students' T, ANOVA, Wilcoxon or Kruskal-Wallis. If the posthoc argument is set as TRUE, multiple group comparisons are complemented with post hoc pairwise tests, either Tukey test (parametric) or Dunn test with Benjamini-Hochberg correction (non-parametric).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>div_test(countable, qvalue, hierarchy, tree, posthoc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="div_test_+3A_countable">countable</code></td>
<td>
<p>A matrix indicating the relative abundances of multiple samples. Columns should be samples and rows OTUs.</p>
</td></tr>
<tr><td><code id="div_test_+3A_qvalue">qvalue</code></td>
<td>
<p>A positive integer or decimal number (&gt;=0), usually between 0 and 3.</p>
</td></tr>
<tr><td><code id="div_test_+3A_hierarchy">hierarchy</code></td>
<td>
<p>A two-column matrix indicating the relation between samples (first column) and groups (second column).</p>
</td></tr>
<tr><td><code id="div_test_+3A_tree">tree</code></td>
<td>
<p>A phylogenetic tree of class 'phylo'. The tip labels must match the row names in the OTU table. Use the function match_data() if the OTU names do not match.</p>
</td></tr>
<tr><td><code id="div_test_+3A_posthoc">posthoc</code></td>
<td>
<p>Whether to run post hoc pairwise analyses or not. If TRUE, an ANOVA will be complemented with a Tukey test and a Kruskal-Wallis test will be complemented with a Dunn test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Diversity test
</p>


<h3>Value</h3>

<p>A statistical test output.
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Jost, L. (2014). Unifying species diversity, phylogenetic diversity, functional diversity, and related similarity and differentiation measures through hill numbers. Annual Review of Ecology Evolution and Systematics, 45, 297-324.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hill_div">hill_div</a></code>, <code><a href="#topic+div_part">div_part</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
data(bat.diet.hierarchy)
div_test(bat.diet.otutable,qvalue=0,hierarchy=bat.diet.hierarchy)
div_test(bat.diet.otutable,qvalue=1,hierarchy=bat.diet.hierarchy,tree=bat.diet.tree)
div_test(bat.diet.otutable,2,bat.diet.hierarchy,bat.diet.tree)
div_test(bat.diet.otutable,qvalue=1,hierarchy=bat.diet.hierarchy,posthoc=TRUE)
</code></pre>

<hr>
<h2 id='div_test_plot'>Diversity test plotting</h2><span id='topic+div_test_plot'></span>

<h3>Description</h3>

<p>Plot of diversity comparison between groups of samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>div_test_plot(divtest, chart, colour, posthoc, threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="div_test_plot_+3A_divtest">divtest</code></td>
<td>
<p>Object outputed by the div_test() function</p>
</td></tr>
<tr><td><code id="div_test_plot_+3A_chart">chart</code></td>
<td>
<p>Chart type, either 'box' for boxplot, 'jitter' for jitter plot or 'violin' for violin plot. chart=&quot;box&quot;</p>
</td></tr>
<tr><td><code id="div_test_plot_+3A_colour">colour</code></td>
<td>
<p>The number of vector items (colours, e.g. '#34k235'), must equal the number of groups that are intended to plot.</p>
</td></tr>
<tr><td><code id="div_test_plot_+3A_posthoc">posthoc</code></td>
<td>
<p>If 'TRUE' pairwise p-values of the posthoc analyses will be ploted. It requires the div_test() object to contain posthoc results.</p>
</td></tr>
<tr><td><code id="div_test_plot_+3A_threshold">threshold</code></td>
<td>
<p>Maximum p-value to show in pairwise posthoc results (usually 0.05, but could be any other number between 0 an 1). P-values above the threshold will not be showed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Diversity test plotting
</p>


<h3>Value</h3>

<p>Chart of (mean) diversities of contrasting groups with optional posthoc results.
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_test">div_test</a></code>, <code><a href="#topic+hill_div">hill_div</a></code>, <code><a href="#topic+div_part">div_part</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.hierarchy)
divtestres &lt;- div_test(bat.diet.otutable,qvalue=0,hierarchy=bat.diet.hierarchy)
div_test_plot(divtestres,chart="box")
div_test_plot(divtestres,chart="violin")
divtest.res.ph &lt;- div_test(bat.diet.otutable,qvalue=0,hierarchy=bat.diet.hierarchy,posthoc=TRUE)
div_test_plot(divtest.res.ph,chart="jitter",posthoc=TRUE,threshold=0.5)
</code></pre>

<hr>
<h2 id='gamma_div'>Gamma diversity</h2><span id='topic+gamma_div'></span>

<h3>Description</h3>

<p>Compute gamma diversity of a system from a matrix (OTU table) containing multiple samples. If a tree is provided, the computed gamma diversity accounts for the phylogenetic relations across OTUs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamma_div(countable,qvalue,tree,weight)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gamma_div_+3A_countable">countable</code></td>
<td>
<p>A count table (matrix/data.frame) indicating the absolute or relative OTU/ASV abundances of multiple samples. Columns must refer to samples and rows to OTUs/ASVs.</p>
</td></tr>
<tr><td><code id="gamma_div_+3A_qvalue">qvalue</code></td>
<td>
<p>A positive number, usually between 0 and 5, but most commonly 0, 1 or 2. It can be an integer or contain decimals.</p>
</td></tr>
<tr><td><code id="gamma_div_+3A_tree">tree</code></td>
<td>
<p>A phylogenetic tree of class 'phylo'. The tip labels must match the row names in the count table. Use the function match_data() if the count table and tree names do not match.</p>
</td></tr>
<tr><td><code id="gamma_div_+3A_weight">weight</code></td>
<td>
<p>A vector indicating the relative weight of each sample. The order needs to be identical to the order of the samples in the OTU table. The values need to sum up to 1. If empty, all samples are weighed the same.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Gamma diversity computation (based on Hill numbers)
</p>


<h3>Value</h3>

<p>A gamma diversity value.
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Hsieh, T. C. (2012). Proposing a resolution to debates on diversity partitioning. Ecology, 93, 2037-2051.<br /><br />
Jost, L. (2007). Partitioning diversity into independent alpha and beta components. Ecology, 88, 2427-2439.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_part">div_part</a></code>, <code><a href="#topic+alpha_div">alpha_div</a></code>, <code><a href="#topic+match_data">match_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
data(bat.diet.hierarchy)
gamma_div(countable=bat.diet.otutable,qvalue=1)
gamma_div(countable=bat.diet.otutable,qvalue=1,tree=bat.diet.tree)
weight.vector = rep(1/ncol(bat.diet.otutable),ncol(bat.diet.otutable))
gamma_div(bat.diet.otutable,1,bat.diet.tree,weight.vector)
</code></pre>

<hr>
<h2 id='hill_div'>Hill numbers computation</h2><span id='topic+hill_div'></span>

<h3>Description</h3>

<p>Compute neutral or phylogenetic Hill numbers from a single sample (vector) or count table (matrix). Hill numbers or numbers equivalents of diversity indices are diversity measures that compute diversity in effective number of OTUs, i.e. the number of equally abundant OTUs that would be needed to give the same value of diversity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hill_div(count, qvalue, tree, dist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hill_div_+3A_count">count</code></td>
<td>
<p>A vector or a matrix/data.frame indicating the (relative) counts of one or multiple samples, respectively. If a matrix/data.frame is provided, columns must refer to samples and rows to OTUs.</p>
</td></tr>
<tr><td><code id="hill_div_+3A_qvalue">qvalue</code></td>
<td>
<p>A positive integer or decimal number (&gt;=0), usually between 0 and 3.</p>
</td></tr>
<tr><td><code id="hill_div_+3A_tree">tree</code></td>
<td>
<p>An ultrametic tree of class 'phylo'. The tip labels must match the names of the vector values (if one sample) or matrix rows (if multiple samples). Use the function match_data() if the OTU names do not match.</p>
</td></tr>
<tr><td><code id="hill_div_+3A_dist">dist</code></td>
<td>
<p>A dist object indicating the pairwise distances between samples. NOT implemented yet</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Hill numbers computation
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Jost, L. (2006). Entropy and diversity. Oikos, 113, 363-375.<br /><br />
Hill, M. O. (1973). Diversity and evenness: a unifying notation and its consequences. Ecology, 54, 427-432.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+index_div">index_div</a></code>, <code><a href="#topic+div_part">div_part</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
data(bat.diet.hierarchy)
#One sample
bat.diet.sample &lt;- bat.diet.otutable[,1]
hill_div(bat.diet.sample,0)
hill_div(bat.diet.sample,qvalue=1)
#One sample (phylogenetic)
names(bat.diet.sample) &lt;- rownames(bat.diet.otutable)
hill_div(bat.diet.sample,1,bat.diet.tree)
#Multiple samples
hill_div(bat.diet.otutable,0)
#Incidence-based
bat.diet.otutable.incidence &lt;- to.incidence(bat.diet.otutable,bat.diet.hierarchy)
hill_div(bat.diet.otutable.incidence,qvalue=1)
hill_div(to.incidence(bat.diet.otutable,bat.diet.hierarchy),1)
</code></pre>

<hr>
<h2 id='index_div'>Diversity index computation</h2><span id='topic+index_div'></span>

<h3>Description</h3>

<p>Computes common diversity indices related to Hill numbers. If the input is a vector, the function computes the indices of a single sample, while if the input is a matrix (OTU table), the function computes individual diversity indices for each sample (column). An ultrametic OTU tree is required for computing phylogenetic diversity indices (Faith's PD, Allen's H and Rao's Q). If the relative abundances of each sample (vector or each column of the matrix) do not sum to 1, TSS normalisation is applied.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>index_div(abund, tree, index)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="index_div_+3A_abund">abund</code></td>
<td>
<p>A vector or a matrix/data.frame indicating the relative abundances of one or multiple samples, respectively. If a matrix/data.frame is provided, columns must refer to samples and rows to OTUs.</p>
</td></tr>
<tr><td><code id="index_div_+3A_tree">tree</code></td>
<td>
<p>An ultrametic tree of class 'phylo'. The tip labels must match the names of the vector values (if one sample) or matrix rows (if multiple samples). Use the function match_data() if the OTU names do not match.</p>
</td></tr>
<tr><td><code id="index_div_+3A_index">index</code></td>
<td>
<p>Diversity index to be computed (&quot;richness&quot;, &quot;shannon&quot;, &quot;simpson&quot;, &quot;faith&quot;, &quot;allen&quot;, &quot;rao&quot;). Default without tree argument: index=&quot;richness&quot;. Default with tree argument: index=&quot;faith&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Diversity index computation
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Jost, L. (2006). Entropy and diversity. Oikos, 113, 363-375.<br /><br />
Rao, C. R. (1982). Diversity and dissimilarity coefficients: A unified approach. Theoretical Population Biology, 21, 24-43.<br /><br />
Shannon, C. E. (1948). A mathematical theory of communication. The Bell System Technical Journal, 27, 379-423.<br /><br />
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hill_div">hill_div</a></code>, <code><a href="#topic+div_part">div_part</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
data(bat.diet.hierarchy)
#One sample
bat.diet.sample &lt;- bat.diet.otutable[,1]
index_div(bat.diet.sample)
index_div(bat.diet.sample,index="shannon")
#Multiple samples
index_div(bat.diet.otutable)
index_div(bat.diet.otutable,tree=bat.diet.tree,index="faith")
#Incidence-based
bat.diet.otutable.incidence &lt;- to.incidence(bat.diet.otutable,bat.diet.hierarchy)
index_div(bat.diet.otutable.incidence)
index_div(bat.diet.otutable.incidence,index="simpson")
index_div(to.incidence(bat.diet.otutable,bat.diet.hierarchy),tree=bat.diet.tree)
</code></pre>

<hr>
<h2 id='is.nested'>Check if hierachy is nested</h2><span id='topic+is.nested'></span>

<h3>Description</h3>

<p>Multi-level diversity partitioning requires the groups at different hierarchical levels to be nested. i.e. two samples that belong to a common parent group cannot have different grandparent groups. The best example of nested hierarchy is taxonomy: e.g. two species that belong to the same genus cannot belong to different families. This function checks whether the groups specified in a hierarchy table have a nested structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.nested(hierarchy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.nested_+3A_hierarchy">hierarchy</code></td>
<td>
<p>A matrix indicating the relation between samples (first column) and parent groups.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Check if hierachy is nested
</p>


<h3>Value</h3>

<p>A logical value (TRUE/FALSE).
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.hierarchy)
is.nested(bat.diet.hierarchy)
</code></pre>

<hr>
<h2 id='match_data'>Match data</h2><span id='topic+match_data'></span>

<h3>Description</h3>

<p>Filter count tables and OTU/ASV phylogenetic trees to match OTUs/ASVs present in both data files..
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match_data(countable, tree, output)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="match_data_+3A_countable">countable</code></td>
<td>
<p>A count table (matrix/data.frame) indicating the absolute or relative OTU/ASV abundances of multiple samples. Columns must refer to samples and rows to OTUs/ASVs.</p>
</td></tr>
<tr><td><code id="match_data_+3A_tree">tree</code></td>
<td>
<p>An ultrametic tree of class 'phylo'.</p>
</td></tr>
<tr><td><code id="match_data_+3A_output">output</code></td>
<td>
<p>Whether to output a filtered count table ('countable') or a filtered OTU tree ('tree').</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Match data
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hill_div">hill_div</a></code>, <code><a href="#topic+index_div">index_div</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
match_data(bat.diet.otutable,bat.diet.tree,output="countable")
match_data(bat.diet.otutable,bat.diet.tree,output="tree")
</code></pre>

<hr>
<h2 id='pair_dis'>Pairwise dissimilarity</h2><span id='topic+pair_dis'></span>

<h3>Description</h3>

<p>Computation of pairwise dissimilarities based on Hill numbers diversity partitioning
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pair_dis(countable, qvalue, tree, hierarchy, level, metric)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pair_dis_+3A_countable">countable</code></td>
<td>
<p>A matrix indicating the relative abundances of multiple samples. Columns should be samples and rows OTUs.</p>
</td></tr>
<tr><td><code id="pair_dis_+3A_qvalue">qvalue</code></td>
<td>
<p>A positive integer or decimal number (&gt;=0), usually between 0 and 3.</p>
</td></tr>
<tr><td><code id="pair_dis_+3A_tree">tree</code></td>
<td>
<p>A phylogenetic tree of class 'phylo'. The tip labels must match the row names in the OTU table. Use the function match_data() if the OTU names do not match.</p>
</td></tr>
<tr><td><code id="pair_dis_+3A_hierarchy">hierarchy</code></td>
<td>
<p>A matrix indicating the relation between samples (first column) and groups.</p>
</td></tr>
<tr><td><code id="pair_dis_+3A_level">level</code></td>
<td>
<p>If '1' dissimilarities are computed across samples, while if '2' dissimilarities are computed across groups, as specified by the hierarchy table. level=1.</p>
</td></tr>
<tr><td><code id="pair_dis_+3A_metric">metric</code></td>
<td>
<p>A vector containing any combination of &quot;C&quot;, &quot;U&quot;, &quot;V&quot; or &quot;S&quot;. If not provided, all metrics will be computed. metric=&quot;U&quot;, metric=c(&quot;U&quot;,&quot;S&quot;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pairwise dissimilarity
</p>


<h3>Value</h3>

<p>A list of matrices containing pairwise beta diversities and dissimilarity metrics.
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Hsieh, T. C. (2012). Proposing a resolution to debates on diversity partitioning. Ecology, 93, 2037-2051.<br /><br />
Jost, L. (2007). Partitioning diversity into independent alpha and beta components. Ecology, 88, 2427-2439.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hill_div">hill_div</a></code>, <code><a href="#topic+div_part">div_part</a></code>, <code><a href="#topic+beta_dis">beta_dis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
data(bat.diet.hierarchy)
pair_dis(bat.diet.otutable,qvalue=1)

pair_dis(bat.diet.otutable,qvalue=1,tree=bat.diet.tree,metric="V")

pair_dis(bat.diet.otutable,qvalue=0,hierarchy=bat.diet.hierarchy,level="2")
</code></pre>

<hr>
<h2 id='pair_dis_plot'>Pairwise dissimilarity plot</h2><span id='topic+pair_dis_plot'></span>

<h3>Description</h3>

<p>Visualisation of pairwise dissimilarities
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pair_dis_plot(distance, hierarchy, type, level, colour, magnify)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pair_dis_plot_+3A_distance">distance</code></td>
<td>
<p>Matrix of pairwise dissimilarities, usually one of the matrices listed in the output object of the pair_dis() function.</p>
</td></tr>
<tr><td><code id="pair_dis_plot_+3A_hierarchy">hierarchy</code></td>
<td>
<p>The first column lists the sample names while the second lists the groups. If provided, group profiles are plotted instead of individual profiles.</p>
</td></tr>
<tr><td><code id="pair_dis_plot_+3A_type">type</code></td>
<td>
<p>Whether to plot a NMDS or qgraph chart. type=&quot;NMDS&quot;.</p>
</td></tr>
<tr><td><code id="pair_dis_plot_+3A_level">level</code></td>
<td>
<p>If '1', samples are not grouped (each sample is a different colour), while if '2', samples are coloured by groups. level=1.</p>
</td></tr>
<tr><td><code id="pair_dis_plot_+3A_colour">colour</code></td>
<td>
<p>he number of vector items (colours, e.g. '#34k235'), must equal the number of samples or groups that are intended to plot.</p>
</td></tr>
<tr><td><code id="pair_dis_plot_+3A_magnify">magnify</code></td>
<td>
<p>Only relevant for qgraph. Whether the pairwise dissimilarity values are transformed to 0-1 scale, 0 corresponding to the minimum dissimilarity and 1 to the maximum dissimilarity value. magnify=FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pairwise dissimilarity plot
</p>


<h3>Value</h3>

<p>An NMDS or network plot.
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Hsieh, T. C. (2012). Proposing a resolution to debates on diversity partitioning. Ecology, 93, 2037-2051.<br /><br />
Jost, L. (2007). Partitioning diversity into independent alpha and beta components. Ecology, 88, 2427-2439.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pair_dis">pair_dis</a></code>, <code><a href="#topic+beta_dis">beta_dis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
data(bat.diet.hierarchy)
pairdisres &lt;- pair_dis(bat.diet.otutable,qvalue=0,hierarchy=bat.diet.hierarchy,level="2")
pair_dis_plot(pairdisres$L2_CqN,hierarchy=bat.diet.hierarchy,type="NMDS",level=2)
pair_dis_plot(pairdisres$L2_CqN,hierarchy=bat.diet.hierarchy,type="qgraph",level=2)
</code></pre>

<hr>
<h2 id='SqN'>Jaccard-type turnover-complement</h2><span id='topic+SqN'></span>

<h3>Description</h3>

<p>The Jaccard-type turnover-complement is thecomplement of the Jaccard-type turnover, which quantifies the normalized OTU turnover rate with respect to the whole system (i.e. gamma). SqN is integrated in the functions beta_dis() and pair_dis().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SqN(beta, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SqN_+3A_beta">beta</code></td>
<td>
<p>A beta diversity value based on Hill numbers.</p>
</td></tr>
<tr><td><code id="SqN_+3A_n">N</code></td>
<td>
<p>An integer indicating sample size, the number of sampling units to be used to compute the similarity measure.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Jaccard-type turnover-complement
</p>


<h3>Value</h3>

<p>A Jaccard-type turnover-complement value
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Hsieh, T. C. (2012). Proposing a resolution to debates on diversity partitioning. Ecology, 93, 2037-2051.<br /><br />
Jost, L. (2007). Partitioning diversity into independent alpha and beta components. Ecology, 88, 2427-2439.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_part">div_part</a></code>, <code><a href="#topic+beta_dis">beta_dis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SqN(beta=1.24,N=2)
SqN(1.24,2)
</code></pre>

<hr>
<h2 id='to.incidence'>Hill numbers computation</h2><span id='topic+to.incidence'></span>

<h3>Description</h3>

<p>Transform a count (OTU/ASV) table from abundance to incidence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>to.incidence(otutable, hierarchy, relative)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="to.incidence_+3A_otutable">otutable</code></td>
<td>
<p>A matrix/data.frame indicating the (relative) abundances of multiple samples. Columns must refer to samples and rows to OTUs/ASVs.</p>
</td></tr>
<tr><td><code id="to.incidence_+3A_hierarchy">hierarchy</code></td>
<td>
<p>A two-column matrix indicating the relation between samples (first column) and groups (second column).</p>
</td></tr>
<tr><td><code id="to.incidence_+3A_relative">relative</code></td>
<td>
<p>Whether to transform the incidence vector or matrix to relative (0-1) values. Default: relative=FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To incidence
</p>


<h3>Value</h3>

<p>A vector of incidence data of a single system if no hierarchy table is specified and a matrix of incidence data of multiple systems if a hierarchy table is specified.
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hill_div">hill_div</a></code>, <code><a href="#topic+div_part">div_part</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.hierarchy)
to.incidence(bat.diet.otutable)
to.incidence(bat.diet.otutable,bat.diet.hierarchy)
to.incidence(bat.diet.otutable,bat.diet.hierarchy,relative=TRUE)
to.incidence(otutable=bat.diet.otutable,hierarchy=bat.diet.hierarchy,relative=TRUE)
</code></pre>

<hr>
<h2 id='tree_depth'>Tree depth</h2><span id='topic+tree_depth'></span>

<h3>Description</h3>

<p>Computes phylogenetic tree depth based from a phylogenetic tree and a vector of (relative) abundances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_depth(tree, abund)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree_depth_+3A_tree">tree</code></td>
<td>
<p>A phylogenetic tree of class 'phylo'. The tip labels must match the row names in the OTU table. Use the function match_data() if the OTU names do not match.</p>
</td></tr>
<tr><td><code id="tree_depth_+3A_abund">abund</code></td>
<td>
<p>A vector or a matrix/data.frame indicating the relative abundances of one or multiple samples, respectively. If a matrix/data.frame is provided, columns must refer to samples and rows to OTUs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Tree depth
</p>


<h3>Value</h3>

<p>A tree depth value
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_part">div_part</a></code>, <code><a href="#topic+gamma_div">gamma_div</a></code>, <code><a href="#topic+match_data">match_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
data(bat.diet.tree)
tree_depth(tree=bat.diet.tree,abund=bat.diet.otutable)
tree_depth(bat.diet.tree,bat.diet.otutable)
</code></pre>

<hr>
<h2 id='tss'>Total Sum Scaling normalisation</h2><span id='topic+tss'></span>

<h3>Description</h3>

<p>Normalise a vector or count matrix to the range of 0-1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tss(abund)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tss_+3A_abund">abund</code></td>
<td>
<p>A vector or a matrix/data.frame indicating the relative abundances of one or multiple samples, respectively. If a matrix/data.frame is provided, columns must refer to samples and rows to OTUs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Total Sum Scaling normalisation
</p>


<h3>Value</h3>

<p>Normalised vector or matrix.
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hill_div">hill_div</a></code>, <code><a href="#topic+index_div">index_div</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bat.diet.otutable)
tss(bat.diet.otutable)
bat.diet.sample &lt;- bat.diet.otutable[,1]
tss(bat.diet.sample)
</code></pre>

<hr>
<h2 id='UqN'>Jaccard-type overlap</h2><span id='topic+UqN'></span>

<h3>Description</h3>

<p>The Jaccard-type overlap quantifies the effective proportion of OTUs or lineages in a system that are shared across all subsystems. Hence, this metric quantifies overlap from the perspective of the overall system. Its corresponding dissimilarity (1 - UqN) quantifies the effective proportion of nonshared OTUs or lineages in the overall system. UqN is integrated in the functions beta_dis() and pair_dis().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UqN(beta, qvalue, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="UqN_+3A_beta">beta</code></td>
<td>
<p>A beta diversity value based on Hill numbers.</p>
</td></tr>
<tr><td><code id="UqN_+3A_qvalue">qvalue</code></td>
<td>
<p>The q value used to compute the beta diversity. It needs to be a positive number, usually between 0 and 5, but most commonly 0, 1 or 2. It can be an integer or contain decimals.</p>
</td></tr>
<tr><td><code id="UqN_+3A_n">N</code></td>
<td>
<p>An integer indicating sample size, the number of sampling units to be used to compute the similarity measure.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Jaccard-type overlap
</p>


<h3>Value</h3>

<p>A Jaccard-type overlap value
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Hsieh, T. C. (2012). Proposing a resolution to debates on diversity partitioning. Ecology, 93, 2037-2051.<br /><br />
Jost, L. (2007). Partitioning diversity into independent alpha and beta components. Ecology, 88, 2427-2439.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_part">div_part</a></code>, <code><a href="#topic+beta_dis">beta_dis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>UqN(beta=1.24,qvalue=1,N=2)
UqN(1.24,1,2)
</code></pre>

<hr>
<h2 id='VqN'>Sørensen-type turnover-complement</h2><span id='topic+VqN'></span>

<h3>Description</h3>

<p>The Sørensen-type turnover-complement is the complement of the Sørensen-type turnover, which quantifies the normalized OTU turnover rate with respect to the average subsystem (i.e., alpha), thus provides the proportion of a typical subsystem that changes across subsystems. VqN is integrated in the functions beta_dis() and pair_dis().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VqN(beta, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VqN_+3A_beta">beta</code></td>
<td>
<p>A beta diversity value based on Hill numbers.</p>
</td></tr>
<tr><td><code id="VqN_+3A_n">N</code></td>
<td>
<p>An integer indicating sample size, the number of sampling units to be used to compute the similarity measure.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sørensen-type turnover-complement
</p>


<h3>Value</h3>

<p>A Sørensen-type turnover-complement value
</p>


<h3>Author(s)</h3>

<p>Antton Alberdi, <a href="mailto:anttonalberdi@gmail.com">anttonalberdi@gmail.com</a>
</p>


<h3>References</h3>

<p>Alberdi, A., Gilbert, M.T.P. (2019). A guide to the application of Hill numbers to DNA-based diversity analyses. Molecular Ecology Resources, 19, 804-817.<br /><br />
Chao, A., Chiu, C.H., &amp; Hsieh, T. C. (2012). Proposing a resolution to debates on diversity partitioning. Ecology, 93, 2037-2051.<br /><br />
Jost, L. (2007). Partitioning diversity into independent alpha and beta components. Ecology, 88, 2427-2439.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div_part">div_part</a></code>, <code><a href="#topic+beta_dis">beta_dis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>VqN(beta=1.24,N=2)
VqN(1.24,2)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
