<!DOCTYPE html><html lang="en"><head><title>Help for package HIMA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HIMA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HIMA-package'><p>High-Dimensional Mediation Analysis for 'Omic' Data</p></a></li>
<li><a href='#BinaryOutcome'><p>Binary Outcome Dataset for HIMA Demo</p></a></li>
<li><a href='#ContinuousOutcome'><p>Continuous Outcome Dataset for HIMA Demo</p></a></li>
<li><a href='#hima'><p>High-dimensional Mediation Analysis</p></a></li>
<li><a href='#hima_classic'><p>Classic high-dimensional mediation analysis</p></a></li>
<li><a href='#hima_dblasso'><p>High-dimensional mediation analysis with de-biased lasso penalty</p></a></li>
<li><a href='#hima_efficient'><p>Efficient high-dimensional mediation analysis</p></a></li>
<li><a href='#hima_microbiome'><p>High-dimensional mediation analysis for compositional microbiome data</p></a></li>
<li><a href='#hima_quantile'><p>High-dimensional quantile mediation analysis</p></a></li>
<li><a href='#hima_survival'><p>High-dimensional mediation analysis for survival data</p></a></li>
<li><a href='#MicrobiomeData'><p>Compositional Mediator Dataset for HIMA Demo</p></a></li>
<li><a href='#QuantileData'><p>Quantile Mediation Dataset for HIMA Demo</p></a></li>
<li><a href='#SurvivalData'><p>Survival Outcome Dataset for HIMA Demo</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>High-Dimensional Mediation Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>2.3.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-01-27</td>
</tr>
<tr>
<td>Description:</td>
<td>Allows to estimate and test high-dimensional mediation effects based on advanced mediator screening and penalized regression techniques. Methods used in the package refer to Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W, Schwartz J, Just A, Colicino E, Vokonas P, Zhao L, Lv J, Baccarelli A, Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects in Epigenetic Studies. Bioinformatics. (2016) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw351">doi:10.1093/bioinformatics/btw351</a>&gt;. PMID: 27357171.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0), ncvreg, glmnet</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, stats, MASS, survival, HDMT, hdi, conquer, quantreg,
hommel, iterators, parallel, foreach, doParallel</td>
</tr>
<tr>
<td>Collate:</td>
<td>utils.R hima_classic.R hima_dblasso.R hima_survival.R
hima_microbiome.R hima_quantile.R hima_efficient.R hima.R
hima_data.R onAttach.R HIMA-package.R</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/YinanZheng/HIMA/">https://github.com/YinanZheng/HIMA/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/YinanZheng/HIMA/issues/">https://github.com/YinanZheng/HIMA/issues/</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-27 11:44:52 UTC; y-zhe</td>
</tr>
<tr>
<td>Author:</td>
<td>Yinan Zheng <a href="https://orcid.org/0000-0002-2006-7320"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Haixiang Zhang [aut],
  Lifang Hou [aut],
  Lei Liu [aut, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yinan Zheng &lt;y-zheng@northwestern.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-27 12:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='HIMA-package'>High-Dimensional Mediation Analysis for 'Omic' Data</h2><span id='topic+HIMA-package'></span>

<h3>Description</h3>

<p>HIMA is an R package for estimating and testing high-dimensional mediation effects in omic studies.
HIMA can perform high-dimensional mediation analysis on a wide range of omic data types as
potential mediators, including epigenetics, transcriptomics, proteomics, metabolomics, and microbiomics.
HIMA can also handle survival data mediation analysis and perform quantile mediation analysis.
</p>

<table>
<tr>
 <td style="text-align: left;"> Package: </td><td style="text-align: left;"> HIMA</td>
</tr>
<tr>
 <td style="text-align: left;"> Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.3.0</td>
</tr>
<tr>
 <td style="text-align: left;"> Date: </td><td style="text-align: left;"> 2025-01-27</td>
</tr>
<tr>
 <td style="text-align: left;"> License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Details</h3>

<p># If package &quot;qvalue&quot; is not found during installation, please first install &quot;qvalue&quot; package
# through Bioconductor: https://www.bioconductor.org/packages/release/bioc/html/qvalue.html
</p>


<h3>Author(s)</h3>

<p>Yinan Zheng <a href="mailto:y-zheng@northwestern.edu">y-zheng@northwestern.edu</a>,
Haixiang Zhang <a href="mailto:haixiang.zhang@tju.edu.cn">haixiang.zhang@tju.edu.cn</a>,
Lei liu (Contact) <a href="mailto:lei.liu@wustl.edu">lei.liu@wustl.edu</a>
</p>
<p>Maintainer: Yinan Zheng <a href="mailto:y-zheng@northwestern.edu">y-zheng@northwestern.edu</a>
</p>


<h3>References</h3>

<p>1. Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W, Schwartz J, Just A, Colicino E, Vokonas P, Zhao L,
Lv J, Baccarelli A, Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects in Epigenetic Studies.
Bioinformatics. 2016. DOI: 10.1093/bioinformatics/btw351. PMID: 27357171; PMCID: PMC5048064
</p>
<p>2. Zhang H, Zheng Y, Hou L, Zheng C, Liu L. Mediation Analysis for Survival Data with High-Dimensional Mediators.
Bioinformatics. 2021. DOI: 10.1093/bioinformatics/btab564. PMID: 34343267; PMCID: PMC8570823
</p>
<p>3. Zhang H, Chen J, Feng Y, Wang C, Li H, Liu L. Mediation Effect Selection in High-dimensional and Compositional Microbiome data.
Stat Med. 2021. DOI: 10.1002/sim.8808. PMID: 33205470; PMCID: PMC7855955
</p>
<p>4. Zhang H, Chen J, Li Z, Liu L. Testing for Mediation Effect with Application to Human Microbiome Data.
Stat Biosci. 2021. DOI: 10.1007/s12561-019-09253-3. PMID: 34093887; PMCID: PMC8177450
</p>
<p>5. Perera C, Zhang H, Zheng Y, Hou L, Qu A, Zheng C, Xie K, Liu L. HIMA2: High-dimensional Mediation Analysis and Its Application in
Epigenome-wide DNA Methylation Data. BMC Bioinformatics. 2022. DOI: 10.1186/s12859-022-04748-1. PMID: 35879655; PMCID: PMC9310002
</p>
<p>6. Zhang H, Hong X, Zheng Y, Hou L, Zheng C, Wang X, Liu L. High-Dimensional Quantile Mediation Analysis with Application to a Birth
Cohort Study of Mother–Newborn Pairs. Bioinformatics. 2024. DOI: 10.1093/bioinformatics/btae055. PMID: 38290773; PMCID: PMC10873903
</p>
<p>7. Bai X, Zheng Y, Hou L, Zheng C, Liu L, Zhang H. An Efficient Testing Procedure for High-dimensional Mediators with FDR Control.
Statistics in Biosciences. 2024. DOI: 10.1007/s12561-024-09447-4.
</p>

<hr>
<h2 id='BinaryOutcome'>Binary Outcome Dataset for HIMA Demo</h2><span id='topic+BinaryOutcome'></span>

<h3>Description</h3>

<p>A dataset containing phenotype data and high-dimensional mediators for binary outcome analysis.
The dataset was simulated using parameters generated from real data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryOutcome
</code></pre>


<h3>Format</h3>

<p>A list with the following components:
</p>

<dl>
<dt>PhenoData</dt><dd><p>A data frame containing:
</p>

<dl>
<dt>Treatment</dt><dd><p>treated (value = 1) or not treated (value = 0).</p>
</dd>
<dt>Disease</dt><dd><p>binary outcome: diseased (value = 1) or healthy (value = 0).</p>
</dd>
<dt>Sex</dt><dd><p>female (value = 1) or male (value = 0).</p>
</dd>
<dt>Age</dt><dd><p>age of the participant.</p>
</dd>
</dl>

</dd>
<dt>Mediator</dt><dd><p>A matrix of high-dimensional mediators (rows: samples, columns: variables).</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(BinaryOutcome)
head(BinaryOutcome$PhenoData)

</code></pre>

<hr>
<h2 id='ContinuousOutcome'>Continuous Outcome Dataset for HIMA Demo</h2><span id='topic+ContinuousOutcome'></span>

<h3>Description</h3>

<p>A dataset containing phenotype data and high-dimensional mediators for continuous outcome analysis.
The dataset was simulated using parameters generated from real data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ContinuousOutcome
</code></pre>


<h3>Format</h3>

<p>A list with the following components:
</p>

<dl>
<dt>PhenoData</dt><dd><p>A data frame containing:
</p>

<dl>
<dt>Treatment</dt><dd><p>treated (value = 1) or not treated (value = 0).</p>
</dd>
<dt>Outcome</dt><dd><p>a normally distributed continuous outcome variable.</p>
</dd>
<dt>Sex</dt><dd><p>female (value = 1) or male (value = 0).</p>
</dd>
<dt>Age</dt><dd><p>age of the participant.</p>
</dd>
</dl>

</dd>
<dt>Mediator</dt><dd><p>A matrix of high-dimensional mediators (rows: samples, columns: variables).</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(ContinuousOutcome)
head(ContinuousOutcome$PhenoData)

</code></pre>

<hr>
<h2 id='hima'>High-dimensional Mediation Analysis</h2><span id='topic+hima'></span>

<h3>Description</h3>

<p><code>hima</code> is a wrapper function designed to perform various HIMA methods for estimating and testing high-dimensional mediation effects.
<code>hima</code> can automatically select the appropriate HIMA method based on the outcome and mediator data type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hima(
  formula,
  data.pheno,
  data.M,
  mediator.type = c("gaussian", "negbin", "compositional"),
  penalty = c("DBlasso", "MCP", "SCAD", "lasso"),
  quantile = FALSE,
  efficient = FALSE,
  scale = TRUE,
  sigcut = 0.05,
  contrast = NULL,
  subset = NULL,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hima_+3A_formula">formula</code></td>
<td>
<p>an object of class <code>formula</code> representing the overall effect model to be fitted, specified as <code>outcome ~ exposure + covariates</code>. 
The &quot;exposure&quot; variable (the variable of interest) must be listed first on the right-hand side of the formula.</p>
</td></tr>
<tr><td><code id="hima_+3A_data.pheno">data.pheno</code></td>
<td>
<p>a data frame containing the exposure, outcome, and covariates specified in the formula. Variable names in <code>data.pheno</code> must match those 
in the formula. When <code>scale = TRUE</code>, the exposure and covariates will be scaled (the outcome retains its original scale).</p>
</td></tr>
<tr><td><code id="hima_+3A_data.m">data.M</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators, with rows representing samples and columns representing mediator variables. 
When <code>scale = TRUE</code>, <code>data.M</code> will be scaled.</p>
</td></tr>
<tr><td><code id="hima_+3A_mediator.type">mediator.type</code></td>
<td>
<p>a character string indicating the data type of the high-dimensional mediators (<code>data.M</code>). Options are: <code>'gaussian'</code> (default): 
for continuous mediators. <code>'negbin'</code>: for count data mediators modeled using the negative binomial distribution (e.g., RNA-seq data). <code>'compositional'</code>: 
for compositional data mediators (e.g., microbiome data).</p>
</td></tr>
<tr><td><code id="hima_+3A_penalty">penalty</code></td>
<td>
<p>a character string specifying the penalty method to apply in the model. Options are: <code>'DBlasso'</code>: De-biased LASSO (default). <code>'MCP'</code>: 
Minimax Concave Penalty. <code>'SCAD'</code>: Smoothly Clipped Absolute Deviation. <code>'lasso'</code>: Least Absolute Shrinkage and Selection Operator. Note: Survival HIMA and microbiome 
HIMA can only be performed with <code>'DBlasso'</code>. Quantile HIMA and efficient HIMA cannot use <code>'DBlasso'</code>; they always apply <code>'MCP'</code>.</p>
</td></tr>
<tr><td><code id="hima_+3A_quantile">quantile</code></td>
<td>
<p>logical. Indicates whether to use quantile HIMA (<code>hima_quantile</code>). Default is <code>FALSE</code>. Applicable only for classic HIMA with a continuous outcome and 
<code>mediator.type = 'gaussian'</code>. If <code>TRUE</code>, specify the desired quantile(s) using the <code>tau</code> parameter; otherwise, the default <code>tau = 0.5</code> (i.e., median) is used.</p>
</td></tr>
<tr><td><code id="hima_+3A_efficient">efficient</code></td>
<td>
<p>logical. Indicates whether to use efficient HIMA (<code>hima_efficient</code>). Default is <code>FALSE</code>. Applicable only for classic HIMA with a continuous outcome and 
<code>mediator.type = 'gaussian'</code>.</p>
</td></tr>
<tr><td><code id="hima_+3A_scale">scale</code></td>
<td>
<p>logical. Determines whether the function scales the data (exposure, mediators, and covariates). Default is <code>TRUE</code>. Note: For simulation studies, set 
<code>scale = FALSE</code> to avoid estimate compression (i.e., shrinkage of estimates toward zero due to scaling).</p>
</td></tr>
<tr><td><code id="hima_+3A_sigcut">sigcut</code></td>
<td>
<p>numeric. The significance cutoff for selecting mediators. Default is <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="hima_+3A_contrast">contrast</code></td>
<td>
<p>a named list of contrasts to be applied to factor variables in the covariates (cannot be the variable of interest).</p>
</td></tr>
<tr><td><code id="hima_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of observations to use in the analysis.</p>
</td></tr>
<tr><td><code id="hima_+3A_verbose">verbose</code></td>
<td>
<p>logical. Determines whether the function displays progress messages. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="hima_+3A_...">...</code></td>
<td>
<p>reserved passing parameter (or for future use).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of selected mediators.
</p>

<dl>
<dt>ID: </dt><dd><p>Mediator ID/name.</p>
</dd>
<dt>alpha: </dt><dd><p>Coefficient estimates of exposure (X) &ndash;&gt; mediators (M) (adjusted for covariates).</p>
</dd>
<dt>beta: </dt><dd><p>Coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for covariates and exposure).</p>
</dd>
<dt>alpha*beta: </dt><dd><p>The estimated indirect (mediation) effect of exposure on outcome through each mediator.</p>
</dd>
<dt>Relative Importance: </dt><dd><p>The proportion of each mediator's mediation effect relative to the sum of the absolute mediation effects of all significant mediators.</p>
</dd>
<dt>p-value: </dt><dd><p>The joint p-value assessing the significance of each mediator's indirect effect, calculated based on the corresponding statistical approach.</p>
</dd>
<dt>tau: </dt><dd><p>The quantile level of the outcome (applicable only when using the quantile mediation model).</p>
</dd>
</dl>



<h3>References</h3>

<p>1. Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W, Schwartz J, Just A, Colicino E, Vokonas P, Zhao L,
Lv J, Baccarelli A, Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects in Epigenetic Studies.
Bioinformatics. 2016. DOI: 10.1093/bioinformatics/btw351. PMID: 27357171; PMCID: PMC5048064
</p>
<p>2. Zhang H, Zheng Y, Hou L, Zheng C, Liu L. Mediation Analysis for Survival Data with High-Dimensional Mediators.
Bioinformatics. 2021. DOI: 10.1093/bioinformatics/btab564. PMID: 34343267; PMCID: PMC8570823
</p>
<p>3. Zhang H, Chen J, Feng Y, Wang C, Li H, Liu L. Mediation Effect Selection in High-dimensional and Compositional Microbiome data.
Stat Med. 2021. DOI: 10.1002/sim.8808. PMID: 33205470; PMCID: PMC7855955
</p>
<p>4. Zhang H, Chen J, Li Z, Liu L. Testing for Mediation Effect with Application to Human Microbiome Data.
Stat Biosci. 2021. DOI: 10.1007/s12561-019-09253-3. PMID: 34093887; PMCID: PMC8177450
</p>
<p>5. Perera C, Zhang H, Zheng Y, Hou L, Qu A, Zheng C, Xie K, Liu L. HIMA2: High-dimensional Mediation Analysis and Its Application in
Epigenome-wide DNA Methylation Data. BMC Bioinformatics. 2022. DOI: 10.1186/s12859-022-04748-1. PMID: 35879655; PMCID: PMC9310002
</p>
<p>6. Zhang H, Hong X, Zheng Y, Hou L, Zheng C, Wang X, Liu L. High-Dimensional Quantile Mediation Analysis with Application to a Birth
Cohort Study of Mother–Newborn Pairs. Bioinformatics. 2024. DOI: 10.1093/bioinformatics/btae055. PMID: 38290773; PMCID: PMC10873903
</p>
<p>7. Bai X, Zheng Y, Hou L, Zheng C, Liu L, Zhang H. An Efficient Testing Procedure for High-dimensional Mediators with FDR Control.
Statistics in Biosciences. 2024. DOI: 10.1007/s12561-024-09447-4.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following examples, M1, M2, and M3 are true mediators.

# Example 1 (continuous outcome - linear HIMA):
head(ContinuousOutcome$PhenoData)

e1 &lt;- hima(Outcome ~ Treatment + Sex + Age,
  data.pheno = ContinuousOutcome$PhenoData,
  data.M = ContinuousOutcome$Mediator,
  mediator.type = "gaussian",
  penalty = "MCP", # Can be "DBlasso" for hima_dblasso
  scale = FALSE
) # Disabled only for simulation data
summary(e1)

# Efficient HIMA (only applicable to mediators and outcomes that are
# both continuous and normally distributed.)
e1e &lt;- hima(Outcome ~ Treatment + Sex + Age,
  data.pheno = ContinuousOutcome$PhenoData,
  data.M = ContinuousOutcome$Mediator,
  mediator.type = "gaussian",
  efficient = TRUE,
  penalty = "MCP", # Efficient HIMA does not support DBlasso
  scale = FALSE
) # Disabled only for simulation data
summary(e1e)

# Example 2 (binary outcome - logistic HIMA):
head(BinaryOutcome$PhenoData)

e2 &lt;- hima(Disease ~ Treatment + Sex + Age,
  data.pheno = BinaryOutcome$PhenoData,
  data.M = BinaryOutcome$Mediator,
  mediator.type = "gaussian",
  penalty = "MCP",
  scale = FALSE
) # Disabled only for simulation data
summary(e2)

# Example 3 (time-to-event outcome - survival HIMA):
head(SurvivalData$PhenoData)

e3 &lt;- hima(Surv(Time, Status) ~ Treatment + Sex + Age,
  data.pheno = SurvivalData$PhenoData,
  data.M = SurvivalData$Mediator,
  mediator.type = "gaussian",
  penalty = "DBlasso",
  scale = FALSE
) # Disabled only for simulation data
summary(e3)

# Example 4 (compositional data as mediator, e.g., microbiome):
head(MicrobiomeData$PhenoData)

e4 &lt;- hima(Outcome ~ Treatment + Sex + Age,
  data.pheno = MicrobiomeData$PhenoData,
  data.M = MicrobiomeData$Mediator,
  mediator.type = "compositional",
  penalty = "DBlasso"
) # Scaling is always enabled internally for hima_microbiome
summary(e4)

#' # Example 5 (quantile mediation anlaysis - quantile HIMA):
head(QuantileData$PhenoData)

# Note that the function will prompt input for quantile level.
e5 &lt;- hima(Outcome ~ Treatment + Sex + Age,
  data.pheno = QuantileData$PhenoData,
  data.M = QuantileData$Mediator,
  mediator.type = "gaussian",
  quantile = TRUE,
  penalty = "MCP", # Quantile HIMA does not support DBlasso
  scale = FALSE, # Disabled only for simulation data
  tau = c(0.3, 0.5, 0.7)
) # Specify multiple quantile level
summary(e5)

## End(Not run)

</code></pre>

<hr>
<h2 id='hima_classic'>Classic high-dimensional mediation analysis</h2><span id='topic+hima_classic'></span>

<h3>Description</h3>

<p><code>hima_classic</code> is used to estimate and test classic high-dimensional mediation effects (linear &amp; logistic regression).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hima_classic(
  X,
  M,
  Y,
  COV.XM = NULL,
  COV.MY = COV.XM,
  Y.type = c("continuous", "binary"),
  M.type = c("gaussian", "negbin"),
  penalty = c("MCP", "SCAD", "lasso"),
  topN = NULL,
  parallel = FALSE,
  ncore = 1,
  scale = TRUE,
  Bonfcut = 0.05,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hima_classic_+3A_x">X</code></td>
<td>
<p>a vector of exposure. Do not use <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_m">M</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators. Rows represent samples, columns
represent variables.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_y">Y</code></td>
<td>
<p>a vector of outcome. Can be either continuous or binary (0-1). Do not use <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_cov.xm">COV.XM</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of covariates dataset for testing the association <code>M ~ X</code>.
Covariates specified here will not participate penalization. Default = <code>NULL</code>. If the covariates
contain mixed data types, please make sure all categorical variables are properly formatted as <code>factor</code> type.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_cov.my">COV.MY</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of covariates dataset for testing the association <code>Y ~ M</code>.
Covariates specified here will not participate penalization. If not specified, the same set of covariates for
<code>M ~ X</code> will be applied (i.e., <code>COV.XM</code>. Using different sets of covariates is allowed but this needs to be 
handled carefully.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_y.type">Y.type</code></td>
<td>
<p>data type of outcome (<code>Y</code>). Either <code>'continuous'</code> (default) or <code>'binary'</code>.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_m.type">M.type</code></td>
<td>
<p>data type of mediator (<code>M</code>). Either <code>'gaussian'</code> (default) or <code>'negbin'</code> (i.e., negative binomial).</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_penalty">penalty</code></td>
<td>
<p>the penalty to be applied to the model. Either <code>'MCP'</code> (the default), <code>'SCAD'</code>, or
<code>'lasso'</code>.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_topn">topN</code></td>
<td>
<p>an integer specifying the number of top markers from sure independent screening.
Default = <code>NULL</code>. If <code>NULL</code>, <code>topN</code> will be either <code>ceiling(n/log(n))</code> for continuous outcome,
or <code>ceiling(n/(2*log(n)))</code> for binary outcome, where <code>n</code> is the sample size. If the sample size is greater
than topN (pre-specified or calculated), all mediators will be included in the test (i.e. low-dimensional scenario).</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_parallel">parallel</code></td>
<td>
<p>logical. Enable parallel computing feature? Default = <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_ncore">ncore</code></td>
<td>
<p>number of cores to run parallel computing Valid when <code>parallel = TRUE</code>.
By default max number of cores available in the machine will be utilized.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_scale">scale</code></td>
<td>
<p>logical. Should the function scale the data? Default = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_bonfcut">Bonfcut</code></td>
<td>
<p>Bonferroni-corrected p value cutoff applied to select significant mediators. Default = <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose? Default = <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="hima_classic_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code>ncvreg</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of selected mediators.
</p>

<dl>
<dt>Index: </dt><dd><p>mediation name of selected significant mediator.</p>
</dd>
<dt>alpha_hat: </dt><dd><p>coefficient estimates of exposure (X) &ndash;&gt; mediators (M) (adjusted for covariates).</p>
</dd>
<dt>beta_hat: </dt><dd><p>coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for covariates and exposure).</p>
</dd>
<dt>IDE: </dt><dd><p>mediation (indirect) effect, i.e., alpha*beta.</p>
</dd>
<dt>rimp: </dt><dd><p>relative importance of the mediator.</p>
</dd>
<dt>pmax: </dt><dd><p>joint raw p-value of selected significant mediator (based on Bonferroni method).</p>
</dd>
</dl>



<h3>References</h3>

<p>Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W, Schwartz J, Just A, Colicino E, Vokonas P, Zhao L,
Lv J, Baccarelli A, Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects in Epigenetic Studies.
Bioinformatics. 2016. DOI: 10.1093/bioinformatics/btw351. PMID: 27357171; PMCID: PMC5048064
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following examples, M1, M2, and M3 are true mediators.

# When Y is continuous and normally distributed
# Example 1 (continuous outcome):
head(ContinuousOutcome$PhenoData)

hima.fit &lt;- hima_classic(
  X = ContinuousOutcome$PhenoData$Treatment,
  Y = ContinuousOutcome$PhenoData$Outcome,
  M = ContinuousOutcome$Mediator,
  COV.XM = ContinuousOutcome$PhenoData[, c("Sex", "Age")],
  Y.type = "continuous",
  scale = FALSE, # Disabled only for simulation data
  verbose = TRUE
)
hima.fit

# When Y is binary
# Example 2 (binary outcome):
head(BinaryOutcome$PhenoData)

hima.logistic.fit &lt;- hima_classic(
  X = BinaryOutcome$PhenoData$Treatment,
  Y = BinaryOutcome$PhenoData$Disease,
  M = BinaryOutcome$Mediator,
  COV.XM = BinaryOutcome$PhenoData[, c("Sex", "Age")],
  Y.type = "binary",
  scale = FALSE, # Disabled only for simulation data
  verbose = TRUE
)
hima.logistic.fit

## End(Not run)

</code></pre>

<hr>
<h2 id='hima_dblasso'>High-dimensional mediation analysis with de-biased lasso penalty</h2><span id='topic+hima_dblasso'></span>

<h3>Description</h3>

<p><code>hima_dblasso</code> is used to estimate and test high-dimensional mediation effects using de-biased lasso penalty.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hima_dblasso(
  X,
  M,
  Y,
  COV = NULL,
  topN = NULL,
  scale = TRUE,
  FDRcut = 0.05,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hima_dblasso_+3A_x">X</code></td>
<td>
<p>a vector of exposure. Do not use <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="hima_dblasso_+3A_m">M</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators. Rows represent samples, columns represent variables.</p>
</td></tr>
<tr><td><code id="hima_dblasso_+3A_y">Y</code></td>
<td>
<p>a vector of outcome. Can be either continuous or binary (0-1). Do not use <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="hima_dblasso_+3A_cov">COV</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of covariates dataset for testing the association M ~ X and Y ~ M.</p>
</td></tr>
<tr><td><code id="hima_dblasso_+3A_topn">topN</code></td>
<td>
<p>an integer specifying the number of top markers from sure independent screening.
Default = <code>NULL</code>. If <code>NULL</code>, <code>topN</code> will be <code>ceiling(n/log(n))</code>,
where <code>n</code> is the sample size. If the sample size is greater than topN (pre-specified or calculated), all
mediators will be included in the test (i.e. low-dimensional scenario).</p>
</td></tr>
<tr><td><code id="hima_dblasso_+3A_scale">scale</code></td>
<td>
<p>logical. Should the function scale the data? Default = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="hima_dblasso_+3A_fdrcut">FDRcut</code></td>
<td>
<p>HDMT pointwise FDR cutoff applied to select significant mediators. Default = <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="hima_dblasso_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose? Default = <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of significant mediators (FDR &lt;<code>FDRcut</code>).
</p>

<dl>
<dt>Index: </dt><dd><p>mediation name of selected significant mediator.</p>
</dd>
<dt>alpha_hat: </dt><dd><p>coefficient estimates of exposure (X) &ndash;&gt; mediators (M) (adjusted for covariates).</p>
</dd>
<dt>alpha_se: </dt><dd><p>standard error for alpha.</p>
</dd>
<dt>beta_hat: </dt><dd><p>coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for covariates and exposure).</p>
</dd>
<dt>beta_se: </dt><dd><p>standard error for beta.</p>
</dd>
<dt>IDE: </dt><dd><p>mediation (indirect) effect, i.e., alpha*beta.</p>
</dd>
<dt>rimp: </dt><dd><p>relative importance of the mediator.</p>
</dd>
<dt>pmax: </dt><dd><p>joint raw p-value of selected significant mediator (based on HDMT pointwise FDR method).</p>
</dd>
</dl>



<h3>References</h3>

<p>Perera C, Zhang H, Zheng Y, Hou L, Qu A, Zheng C, Xie K, Liu L.
HIMA2: high-dimensional mediation analysis and its application in epigenome-wide DNA methylation data.
BMC Bioinformatics. 2022. DOI: 10.1186/s12859-022-04748-1. PMID: 35879655; PMCID: PMC9310002
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following examples, M1, M2, and M3 are true mediators.

# Y is continuous and normally distributed
# Example:
head(ContinuousOutcome$PhenoData)

hima_dblasso.fit &lt;- hima_dblasso(
  X = ContinuousOutcome$PhenoData$Treatment,
  Y = ContinuousOutcome$PhenoData$Outcome,
  M = ContinuousOutcome$Mediator,
  COV = ContinuousOutcome$PhenoData[, c("Sex", "Age")],
  scale = FALSE, # Disabled only for simulation data
  FDRcut = 0.05,
  verbose = TRUE
)
hima_dblasso.fit

## End(Not run)

</code></pre>

<hr>
<h2 id='hima_efficient'>Efficient high-dimensional mediation analysis</h2><span id='topic+hima_efficient'></span>

<h3>Description</h3>

<p><code>hima_efficient</code> is used to estimate and test high-dimensional mediation effects using an efficient algorithm. It provides
higher statistical power than the standard <code>hima</code>. Note: efficient HIMA is only applicable to mediators and outcomes that
are both continuous and normally distributed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hima_efficient(
  X,
  M,
  Y,
  COV = NULL,
  topN = NULL,
  scale = TRUE,
  FDRcut = 0.05,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hima_efficient_+3A_x">X</code></td>
<td>
<p>a vector of exposure. Do not use <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="hima_efficient_+3A_m">M</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators. Rows represent samples, columns
represent mediator variables. <code>M</code> has to be continuous and normally distributed.</p>
</td></tr>
<tr><td><code id="hima_efficient_+3A_y">Y</code></td>
<td>
<p>a vector of continuous outcome. Do not use <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="hima_efficient_+3A_cov">COV</code></td>
<td>
<p>a matrix of adjusting covariates. Rows represent samples, columns represent variables. Can be <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="hima_efficient_+3A_topn">topN</code></td>
<td>
<p>an integer specifying the number of top markers from sure independent screening.
Default = <code>NULL</code>. If <code>NULL</code>, <code>topN</code> will be <code>2*ceiling(n/log(n))</code>, where <code>n</code> is the sample size.
If the sample size is greater than topN (pre-specified or calculated), all mediators will be included in the test (i.e. low-dimensional scenario).</p>
</td></tr>
<tr><td><code id="hima_efficient_+3A_scale">scale</code></td>
<td>
<p>logical. Should the function scale the data? Default = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="hima_efficient_+3A_fdrcut">FDRcut</code></td>
<td>
<p>Benjamini-Hochberg FDR cutoff applied to select significant mediators. Default = <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="hima_efficient_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose? Default = <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of significant mediators (FDR &lt;<code>FDRcut</code>).
</p>

<dl>
<dt>Index: </dt><dd><p>mediation name of selected significant mediator.</p>
</dd>
<dt>alpha_hat: </dt><dd><p>coefficient estimates of exposure (X) &ndash;&gt; mediators (M) (adjusted for covariates).</p>
</dd>
<dt>alpha_se: </dt><dd><p>standard error for alpha.</p>
</dd>
<dt>beta_hat: </dt><dd><p>coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for covariates and exposure).</p>
</dd>
<dt>beta_se: </dt><dd><p>standard error for beta.</p>
</dd>
<dt>IDE: </dt><dd><p>mediation (indirect) effect, i.e., alpha*beta.</p>
</dd>
<dt>rimp: </dt><dd><p>relative importance of the mediator.</p>
</dd>
<dt>pmax: </dt><dd><p>joint raw p-value of selected significant mediator (based on divide-aggregate composite-null test [DACT] method).</p>
</dd>
</dl>



<h3>References</h3>

<p>Bai X, Zheng Y, Hou L, Zheng C, Liu L, Zhang H. An Efficient Testing Procedure for High-dimensional Mediators with FDR Control.
Statistics in Biosciences. 2024. DOI: 10.1007/s12561-024-09447-4.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following example, M1, M2, and M3 are true mediators.

# Y is continuous and normally distributed
# Example (continuous outcome):
head(ContinuousOutcome$PhenoData)

hima_efficient.fit &lt;- hima_efficient(
  X = ContinuousOutcome$PhenoData$Treatment,
  Y = ContinuousOutcome$PhenoData$Outcome,
  M = ContinuousOutcome$Mediator,
  COV = ContinuousOutcome$PhenoData[, c("Sex", "Age")],
  scale = FALSE, # Disabled only for simulation data
  FDRcut = 0.05,
  verbose = TRUE
)
hima_efficient.fit

## End(Not run)

</code></pre>

<hr>
<h2 id='hima_microbiome'>High-dimensional mediation analysis for compositional microbiome data</h2><span id='topic+hima_microbiome'></span>

<h3>Description</h3>

<p><code>hima_microbiome</code> is used to estimate and test high-dimensional mediation effects for compositional microbiome data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hima_microbiome(X, OTU, Y, COV = NULL, FDRcut = 0.05, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hima_microbiome_+3A_x">X</code></td>
<td>
<p>a vector of exposure. Do not use <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="hima_microbiome_+3A_otu">OTU</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional Operational Taxonomic Unit (OTU) data (mediators).
Rows represent samples, columns represent variables.</p>
</td></tr>
<tr><td><code id="hima_microbiome_+3A_y">Y</code></td>
<td>
<p>a vector of continuous outcome. Binary outcome is not allowed. Do not use <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="hima_microbiome_+3A_cov">COV</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of adjusting covariates. Rows represent samples, columns represent
microbiome variables. Can be <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="hima_microbiome_+3A_fdrcut">FDRcut</code></td>
<td>
<p>Hommel FDR cutoff applied to select significant mediators. Default = <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="hima_microbiome_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose? Default = <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of significant mediators (FDR &lt;<code>FDRcut</code>).
</p>

<dl>
<dt>Index: </dt><dd><p>mediation name of selected significant mediator.</p>
</dd>
<dt>alpha_hat: </dt><dd><p>coefficient estimates of exposure (X) &ndash;&gt; mediators (M) (adjusted for covariates).</p>
</dd>
<dt>alpha_se: </dt><dd><p>standard error for alpha.</p>
</dd>
<dt>beta_hat: </dt><dd><p>coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for covariates and exposure).</p>
</dd>
<dt>beta_se: </dt><dd><p>standard error for beta.</p>
</dd>
<dt>IDE: </dt><dd><p>mediation (indirect) effect, i.e., alpha*beta.</p>
</dd>
<dt>rimp: </dt><dd><p>relative importance of the mediator.</p>
</dd>
<dt>pmax: </dt><dd><p>joint raw p-value of selected significant mediator (based on Hommel FDR method).</p>
</dd>
</dl>



<h3>References</h3>

<p>1. Zhang H, Chen J, Feng Y, Wang C, Li H, Liu L. Mediation effect selection in high-dimensional and compositional microbiome data.
Stat Med. 2021. DOI: 10.1002/sim.8808. PMID: 33205470; PMCID: PMC7855955
</p>
<p>2. Zhang H, Chen J, Li Z, Liu L. Testing for mediation effect with application to human microbiome data.
Stat Biosci. 2021. DOI: 10.1007/s12561-019-09253-3. PMID: 34093887; PMCID: PMC8177450
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following example, M1, M2, and M3 are true mediators.

head(MicrobiomeData$PhenoData)

hima_microbiome.fit &lt;- hima_microbiome(
  X = MicrobiomeData$PhenoData$Treatment,
  Y = MicrobiomeData$PhenoData$Outcome,
  OTU = MicrobiomeData$Mediator,
  COV = MicrobiomeData$PhenoData[, c("Sex", "Age")],
  FDRcut = 0.05,
  verbose = TRUE
)
hima_microbiome.fit

## End(Not run)

</code></pre>

<hr>
<h2 id='hima_quantile'>High-dimensional quantile mediation analysis</h2><span id='topic+hima_quantile'></span>

<h3>Description</h3>

<p><code>hima_quantile</code> is used to estimate and test high-dimensional quantile mediation effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hima_quantile(
  X,
  M,
  Y,
  COV = NULL,
  penalty = c("MCP", "SCAD", "lasso"),
  topN = NULL,
  tau = 0.5,
  scale = TRUE,
  Bonfcut = 0.05,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hima_quantile_+3A_x">X</code></td>
<td>
<p>a vector of exposure. Do not use <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="hima_quantile_+3A_m">M</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators. Rows represent samples, columns
represent mediator variables.</p>
</td></tr>
<tr><td><code id="hima_quantile_+3A_y">Y</code></td>
<td>
<p>a vector of continuous outcome. Do not use <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="hima_quantile_+3A_cov">COV</code></td>
<td>
<p>a matrix of adjusting covariates. Rows represent samples, columns represent variables. Can be <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="hima_quantile_+3A_penalty">penalty</code></td>
<td>
<p>the penalty to be applied to the model (a parameter passed to function <code>conquer.cv.reg</code> in package <code>conquer</code>.
Either <code>'MCP'</code> (the default), <code>'SCAD'</code>, or <code>'lasso'</code>.</p>
</td></tr>
<tr><td><code id="hima_quantile_+3A_topn">topN</code></td>
<td>
<p>an integer specifying the number of top markers from sure independent screening.
Default = <code>NULL</code>. If <code>NULL</code>, <code>topN</code> will be <code>2*ceiling(n/log(n))</code>, where <code>n</code> is the sample size.
If the sample size is greater than topN (pre-specified or calculated), all mediators will be included in the test (i.e. 
low-dimensional scenario).</p>
</td></tr>
<tr><td><code id="hima_quantile_+3A_tau">tau</code></td>
<td>
<p>quantile level of outcome. Default = <code>0.5</code>. A vector of tau is accepted.</p>
</td></tr>
<tr><td><code id="hima_quantile_+3A_scale">scale</code></td>
<td>
<p>logical. Should the function scale the data? Default = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="hima_quantile_+3A_bonfcut">Bonfcut</code></td>
<td>
<p>Bonferroni-corrected p value cutoff applied to select significant mediators. Default = <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="hima_quantile_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose? Default = <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="hima_quantile_+3A_...">...</code></td>
<td>
<p>reserved passing parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of selected mediators (Bonferroni-adjusted p value &lt;<code>Bonfcut</code>).
</p>

<dl>
<dt>Index: </dt><dd><p>mediation name of selected significant mediator.</p>
</dd>
<dt>alpha_hat: </dt><dd><p>coefficient estimates of exposure (X) &ndash;&gt; mediators (M) (adjusted for covariates).</p>
</dd>
<dt>alpha_se: </dt><dd><p>standard error for alpha.</p>
</dd>
<dt>beta_hat: </dt><dd><p>coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for covariates and exposure).</p>
</dd>
<dt>beta_se: </dt><dd><p>standard error for beta.</p>
</dd>
<dt>IDE: </dt><dd><p>mediation (indirect) effect, i.e., alpha*beta.</p>
</dd>
<dt>rimp: </dt><dd><p>relative importance of the mediator.</p>
</dd>
<dt>pmax: </dt><dd><p>joint raw p-value of selected significant mediator (based on Bonferroni method).</p>
</dd>
</dl>



<h3>References</h3>

<p>Zhang H, Hong X, Zheng Y, Hou L, Zheng C, Wang X, Liu L. High-Dimensional Quantile Mediation Analysis with Application to a Birth
Cohort Study of Mother–Newborn Pairs. Bioinformatics. 2024. DOI: 10.1093/bioinformatics/btae055. PMID: 38290773; PMCID: PMC10873903
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following example, M1, M2, and M3 are true mediators.

head(QuantileData$PhenoData)

hima_quantile.fit &lt;- hima_quantile(
  X = QuantileData$PhenoData$Treatment,
  M = QuantileData$Mediator,
  Y = QuantileData$PhenoData$Outcome,
  COV = QuantileData$PhenoData[, c("Sex", "Age")],
  tau = c(0.3, 0.5, 0.7),
  scale = FALSE, # Disabled only for simulation data
  Bonfcut = 0.05,
  verbose = TRUE
)
hima_quantile.fit

## End(Not run)

</code></pre>

<hr>
<h2 id='hima_survival'>High-dimensional mediation analysis for survival data</h2><span id='topic+hima_survival'></span>

<h3>Description</h3>

<p><code>hima_survival</code> is used to estimate and test high-dimensional mediation effects for survival data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hima_survival(
  X,
  M,
  OT,
  status,
  COV = NULL,
  topN = NULL,
  scale = TRUE,
  FDRcut = 0.05,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hima_survival_+3A_x">X</code></td>
<td>
<p>a vector of exposure. Do not use <code>data.frame</code> or <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="hima_survival_+3A_m">M</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators. Rows represent samples, columns
represent mediator variables.</p>
</td></tr>
<tr><td><code id="hima_survival_+3A_ot">OT</code></td>
<td>
<p>a vector of observed failure times.</p>
</td></tr>
<tr><td><code id="hima_survival_+3A_status">status</code></td>
<td>
<p>a vector of censoring indicator (<code>status = 1</code>: uncensored; <code>status = 0</code>: censored)</p>
</td></tr>
<tr><td><code id="hima_survival_+3A_cov">COV</code></td>
<td>
<p>a matrix of adjusting covariates. Rows represent samples, columns represent variables. Can be <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="hima_survival_+3A_topn">topN</code></td>
<td>
<p>an integer specifying the number of top markers from sure independent screening.
Default = <code>NULL</code>. If <code>NULL</code>, <code>topN</code> will be <code>ceiling(n/log(n))</code>, where <code>n</code> is the sample size.
If the sample size is greater than topN (pre-specified or calculated), all mediators will be included in the test (i.e. low-dimensional scenario).</p>
</td></tr>
<tr><td><code id="hima_survival_+3A_scale">scale</code></td>
<td>
<p>logical. Should the function scale the data? Default = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="hima_survival_+3A_fdrcut">FDRcut</code></td>
<td>
<p>HDMT pointwise FDR cutoff applied to select significant mediators. Default = <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="hima_survival_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose? Default = <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of significant mediators (FDR &lt;<code>FDRcut</code>).
</p>

<dl>
<dt>Index: </dt><dd><p>mediation name of selected significant mediator.</p>
</dd>
<dt>alpha_hat: </dt><dd><p>coefficient estimates of exposure (X) &ndash;&gt; mediators (M) (adjusted for covariates).</p>
</dd>
<dt>alpha_se: </dt><dd><p>standard error for alpha.</p>
</dd>
<dt>beta_hat: </dt><dd><p>coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for covariates and exposure).</p>
</dd>
<dt>beta_se: </dt><dd><p>standard error for beta.</p>
</dd>
<dt>IDE: </dt><dd><p>mediation (indirect) effect, i.e., alpha*beta.</p>
</dd>
<dt>rimp: </dt><dd><p>relative importance of the mediator.</p>
</dd>
<dt>pmax: </dt><dd><p>joint raw p-value of selected significant mediator (based on HDMT pointwise FDR method).</p>
</dd>
</dl>



<h3>References</h3>

<p>Zhang H, Zheng Y, Hou L, Zheng C, Liu L. Mediation Analysis for Survival Data with High-Dimensional Mediators.
Bioinformatics. 2021. DOI: 10.1093/bioinformatics/btab564. PMID: 34343267; PMCID: PMC8570823
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following example, M1, M2, and M3 are true mediators.

head(SurvivalData$PhenoData)

hima_survival.fit &lt;- hima_survival(
  X = SurvivalData$PhenoData$Treatment,
  M = SurvivalData$Mediator,
  OT = SurvivalData$PhenoData$Time,
  status = SurvivalData$PhenoData$Status,
  COV = SurvivalData$PhenoData[, c("Sex", "Age")],
  scale = FALSE, # Disabled only for simulation data
  FDRcut = 0.05,
  verbose = TRUE
)
hima_survival.fit

## End(Not run)

</code></pre>

<hr>
<h2 id='MicrobiomeData'>Compositional Mediator Dataset for HIMA Demo</h2><span id='topic+MicrobiomeData'></span>

<h3>Description</h3>

<p>A dataset containing phenotype data and high-dimensional compositional mediators (e.g., microbiome).
The dataset was simulated using parameters generated from real data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MicrobiomeData
</code></pre>


<h3>Format</h3>

<p>A list with the following components:
</p>

<dl>
<dt>PhenoData</dt><dd><p>A data frame containing:
</p>

<dl>
<dt>Treatment</dt><dd><p>treated (value = 1) or not treated (value = 0).</p>
</dd>
<dt>Outcome</dt><dd><p>a normally distributed continuous outcome variable.</p>
</dd>
<dt>Sex</dt><dd><p>female (value = 1) or male (value = 0).</p>
</dd>
<dt>Age</dt><dd><p>age of the participant.</p>
</dd>
</dl>

</dd>
<dt>Mediator</dt><dd><p>A matrix of high-dimensional compositional mediators (rows: samples, columns: variables).</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(MicrobiomeData)
head(MicrobiomeData$PhenoData)

</code></pre>

<hr>
<h2 id='QuantileData'>Quantile Mediation Dataset for HIMA Demo</h2><span id='topic+QuantileData'></span>

<h3>Description</h3>

<p>A dataset containing phenotype data and high-dimensional mediators for quantile mediation analysis.
The dataset was simulated using parameters generated from real data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QuantileData
</code></pre>


<h3>Format</h3>

<p>A list with the following components:
</p>

<dl>
<dt>PhenoData</dt><dd><p>A data frame containing:
</p>

<dl>
<dt>Treatment</dt><dd><p>treated (value = 1) or not treated (value = 0).</p>
</dd>
<dt>Outcome</dt><dd><p>an abnormally distributed continuous outcome variable.</p>
</dd>
<dt>Sex</dt><dd><p>female (value = 1) or male (value = 0).</p>
</dd>
<dt>Age</dt><dd><p>age of the participant.</p>
</dd>
</dl>

</dd>
<dt>Mediator</dt><dd><p>A matrix of high-dimensional mediators (rows: samples, columns: variables).</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(QuantileData)
head(QuantileData$PhenoData)

</code></pre>

<hr>
<h2 id='SurvivalData'>Survival Outcome Dataset for HIMA Demo</h2><span id='topic+SurvivalData'></span>

<h3>Description</h3>

<p>A dataset containing phenotype data and high-dimensional mediators for survival outcome analysis.
The dataset was simulated using parameters generated from real data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SurvivalData
</code></pre>


<h3>Format</h3>

<p>A list with the following components:
</p>

<dl>
<dt>PhenoData</dt><dd><p>A data frame containing:
</p>

<dl>
<dt>Treatment</dt><dd><p>treated (value = 1) or not treated (value = 0).</p>
</dd>
<dt>Status</dt><dd><p>status indicator: dead (value = 1) or alive (value = 0).</p>
</dd>
<dt>Time</dt><dd><p>time to the event or censoring.</p>
</dd>
<dt>Sex</dt><dd><p>female (value = 1) or male (value = 0).</p>
</dd>
<dt>Age</dt><dd><p>age of the participant.</p>
</dd>
</dl>

</dd>
<dt>Mediator</dt><dd><p>A matrix of high-dimensional mediators (rows: samples, columns: variables).</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(SurvivalData)
head(SurvivalData$PhenoData)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
