<!DOCTYPE html><html><head><title>Help for package HIMA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HIMA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dblassoHIMA'><p>This is the function for high-dimensional mediation analysis using de-biased lasso</p>
HIMA with de-biased lasso</a></li>
<li><a href='#hima'><p>High-dimensional Mediation Analysis</p></a></li>
<li><a href='#HIMA-package'><p>High-Dimensional Mediation Analysis for 'Omic' Data</p></a></li>
<li><a href='#hima2'><p>Advanced High-dimensional Mediation Analysis</p></a></li>
<li><a href='#himaDat'><p>Example datasets for HIMA demo</p></a></li>
<li><a href='#microHIMA'><p>High-dimensional mediation analysis for compositional microbiome data</p></a></li>
<li><a href='#qHIMA'><p>High-dimensional quantile mediation analysis</p></a></li>
<li><a href='#survHIMA'><p>High-dimensional mediation analysis for survival data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>High-Dimensional Mediation Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>2.2.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-10</td>
</tr>
<tr>
<td>Description:</td>
<td>Allows to estimate and test high-dimensional mediation effects based on advanced mediator screening and penalized regression techniques. Methods used in the package refer to Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W, Schwartz J, Just A, Colicino E, Vokonas P, Zhao L, Lv J, Baccarelli A, Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects in Epigenetic Studies. Bioinformatics. (2016) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw351">doi:10.1093/bioinformatics/btw351</a>&gt;. PMID: 27357171.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0), ncvreg, glmnet</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, stats, MASS, survival, HDMT, hdi, conquer, quantreg,
hommel, iterators, parallel, foreach, doParallel</td>
</tr>
<tr>
<td>Collate:</td>
<td>utils.R hima.R survHIMA.R microHIMA.R dblassoHIMA.R qHIMA.R
hima2.R himaDat.R onAttach.R HIMA-package.R</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/YinanZheng/HIMA/">https://github.com/YinanZheng/HIMA/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/YinanZheng/HIMA/issues/">https://github.com/YinanZheng/HIMA/issues/</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-10 15:35:59 UTC; y-zhe</td>
</tr>
<tr>
<td>Author:</td>
<td>Yinan Zheng [aut, cre],
  Haixiang Zhang [aut],
  Lifang Hou [aut],
  Lei Liu [aut, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yinan Zheng &lt;y-zheng@northwestern.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-10 16:02:34 UTC</td>
</tr>
</table>
<hr>
<h2 id='dblassoHIMA'>This is the function for high-dimensional mediation analysis using de-biased lasso
HIMA with de-biased lasso</h2><span id='topic+dblassoHIMA'></span>

<h3>Description</h3>

<p><code>dblassoHIMA</code> is used to estimate and test high-dimensional mediation effects using de-biased lasso penalty.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dblassoHIMA(
  X,
  Y,
  M,
  Z,
  Y.family = c("gaussian", "binomial"),
  topN = NULL,
  scale = TRUE,
  FDRcut = 0.05,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dblassoHIMA_+3A_x">X</code></td>
<td>
<p>a vector of exposure.</p>
</td></tr>
<tr><td><code id="dblassoHIMA_+3A_y">Y</code></td>
<td>
<p>a vector of outcome. Can be either continuous or binary (0-1).</p>
</td></tr>
<tr><td><code id="dblassoHIMA_+3A_m">M</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators. Rows represent samples, columns represent variables.</p>
</td></tr>
<tr><td><code id="dblassoHIMA_+3A_z">Z</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of covariates dataset for testing the association M ~ X and Y ~ M.</p>
</td></tr>
<tr><td><code id="dblassoHIMA_+3A_y.family">Y.family</code></td>
<td>
<p>either 'gaussian' (default) or 'binomial', depending on the data type of outcome (<code>Y</code>). This parameter is passed 
to function <code>lasso.proj</code> in R package <code><a href="bayestestR.html#topic+hdi">hdi</a></code> for de-biased lasso penalization.</p>
</td></tr>
<tr><td><code id="dblassoHIMA_+3A_topn">topN</code></td>
<td>
<p>an integer specifying the number of top markers from sure independent screening. 
Default = <code>NULL</code>. If <code>NULL</code>, <code>topN</code> will be either <code>ceiling(n/log(n))</code> if 
<code>Y.family = 'gaussian'</code>, or <code>ceiling(n/(2*log(n)))</code> if <code>Y.family = 'binomial'</code>, 
where <code>n</code> is the sample size. If the sample size is greater than topN (pre-specified or calculated), all 
mediators will be included in the test (i.e. low-dimensional scenario).</p>
</td></tr>
<tr><td><code id="dblassoHIMA_+3A_scale">scale</code></td>
<td>
<p>logical. Should the function scale the data? Default = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="dblassoHIMA_+3A_fdrcut">FDRcut</code></td>
<td>
<p>FDR cutoff applied to define and select significant mediators. Default = <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="dblassoHIMA_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose? Default = <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of selected mediators (FDR &lt;<code>FDPcut</code>). 
</p>

<ul>
<li><p>alpha: coefficient estimates of exposure (X) &ndash;&gt; mediators (M).
</p>
</li>
<li><p>beta: coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for exposure).
</p>
</li>
<li><p>gamma: coefficient estimates of exposure (X) &ndash;&gt; outcome (Y) (total effect).
</p>
</li>
<li><p>alpha*beta: mediation effect.
</p>
</li>
<li><p>% total effect: alpha*beta / gamma. Percentage of the mediation effect out of the total effect.
</p>
</li>
<li><p>p.joint: joint raw p-value of selected significant mediator (based on FDR).
</p>
</li></ul>

<p>#' @references Perera C, Zhang H, Zheng Y, Hou L, Qu A, Zheng C, Xie K, Liu L. 
HIMA2: high-dimensional mediation analysis and its application in epigenome-wide DNA methylation data. 
BMC Bioinformatics. 2022. DOI: 10.1186/s12859-022-04748-1. PMID: 35879655. PMCID: PMC9310002
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following examples, M1, M2, and M3 are true mediators.
data(himaDat)

# When Y is continuous and normally distributed
# Example 1 (continuous outcome): 
head(himaDat$Example1$PhenoData)

dblassohima.fit &lt;- dblassoHIMA(X = himaDat$Example1$PhenoData$Treatment, 
                               Y = himaDat$Example1$PhenoData$Outcome, 
                               M = himaDat$Example1$Mediator,
                               Z = himaDat$Example1$PhenoData[, c("Sex", "Age")],
                               Y.family = 'gaussian',
                               scale = FALSE,
                               verbose = TRUE) 
dblassohima.fit

# When Y is binary (should specify Y.family)
# Example 2 (binary outcome): 
head(himaDat$Example2$PhenoData)

dblassohima.logistic.fit &lt;- dblassoHIMA(X = himaDat$Example2$PhenoData$Treatment,
                                        Y = himaDat$Example2$PhenoData$Disease,
                                        M = himaDat$Example2$Mediator,
                                        Z = himaDat$Example2$PhenoData[, c("Sex", "Age")],
                                        Y.family = 'binomial',
                                        scale = FALSE,
                                        verbose = TRUE)
dblassohima.logistic.fit

## End(Not run)

</code></pre>

<hr>
<h2 id='hima'>High-dimensional Mediation Analysis</h2><span id='topic+hima'></span>

<h3>Description</h3>

<p><code>hima</code> is used to estimate and test high-dimensional mediation effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hima(
  X,
  Y,
  M,
  COV.XM = NULL,
  COV.MY = COV.XM,
  Y.family = c("gaussian", "binomial"),
  M.family = c("gaussian", "negbin"),
  penalty = c("MCP", "SCAD", "lasso"),
  topN = NULL,
  parallel = FALSE,
  ncore = 1,
  scale = TRUE,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hima_+3A_x">X</code></td>
<td>
<p>a vector of exposure. Do not use data.frame or matrix.</p>
</td></tr>
<tr><td><code id="hima_+3A_y">Y</code></td>
<td>
<p>a vector of outcome. Can be either continuous or binary (0-1). Do not use data.frame or matrix.</p>
</td></tr>
<tr><td><code id="hima_+3A_m">M</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators. Rows represent samples, columns 
represent variables.</p>
</td></tr>
<tr><td><code id="hima_+3A_cov.xm">COV.XM</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of covariates dataset for testing the association <code>M ~ X</code>. 
Covariates specified here will not participate penalization. Default = <code>NULL</code>. If the covariates 
contain mixed data types, please make sure all categorical variables are properly formatted as <code>factor</code> type.</p>
</td></tr>
<tr><td><code id="hima_+3A_cov.my">COV.MY</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of covariates dataset for testing the association <code>Y ~ M</code>. 
Covariates specified here will not participate penalization. If not specified, the same set of covariates for 
<code>M ~ X</code> will be applied. Using different sets of covariates is allowed but this needs to be handled carefully.</p>
</td></tr>
<tr><td><code id="hima_+3A_y.family">Y.family</code></td>
<td>
<p>either 'gaussian' (default) or 'binomial', depending on the data type of outcome (<code>Y</code>). See 
<code>ncvreg</code></p>
</td></tr>
<tr><td><code id="hima_+3A_m.family">M.family</code></td>
<td>
<p>either 'gaussian' (default) or 'negbin' (i.e., negative binomial), depending on the data type of
mediator (<code>M</code>).</p>
</td></tr>
<tr><td><code id="hima_+3A_penalty">penalty</code></td>
<td>
<p>the penalty to be applied to the model. Either 'MCP' (the default), 'SCAD', or 
'lasso'.</p>
</td></tr>
<tr><td><code id="hima_+3A_topn">topN</code></td>
<td>
<p>an integer specifying the number of top markers from sure independent screening. 
Default = <code>NULL</code>. If <code>NULL</code>, <code>topN</code> will be either <code>ceiling(n/log(n))</code> if 
<code>Y.family = 'gaussian'</code>, or <code>ceiling(n/(2*log(n)))</code> if <code>Y.family = 'binomial'</code>, 
where <code>n</code> is the sample size. If the sample size is greater than topN (pre-specified or calculated), all 
mediators will be included in the test (i.e. low-dimensional scenario).</p>
</td></tr>
<tr><td><code id="hima_+3A_parallel">parallel</code></td>
<td>
<p>logical. Enable parallel computing feature? Default = <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="hima_+3A_ncore">ncore</code></td>
<td>
<p>number of cores to run parallel computing Valid when <code>parallel == TRUE</code>. 
By default max number of cores available in the machine will be utilized.</p>
</td></tr>
<tr><td><code id="hima_+3A_scale">scale</code></td>
<td>
<p>logical. Should the function scale the data? Default = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="hima_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose? Default = <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="hima_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code>ncvreg</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of selected mediators. 
</p>

<ul>
<li><p>alpha: coefficient estimates of exposure (X) &ndash;&gt; mediators (M).
</p>
</li>
<li><p>beta: coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for exposure).
</p>
</li>
<li><p>gamma: coefficient estimates of exposure (X) &ndash;&gt; outcome (Y) (total effect).
</p>
</li>
<li><p>alpha*beta: mediation effect.
</p>
</li>
<li><p>% total effect: alpha*beta / gamma. Percentage of the mediation effect out of the total effect.
</p>
</li>
<li><p>Bonferroni.p: statistical significance of the mediator (Bonferroni-corrected p value).
</p>
</li>
<li><p>BH.FDR: statistical significance of the mediator (Benjamini-Hochberg FDR).
</p>
</li></ul>



<h3>References</h3>

<p>Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W, Schwartz J, Just A, Colicino E, Vokonas P, Zhao L, 
Lv J, Baccarelli A, Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects in Epigenetic Studies. 
Bioinformatics. 2016. DOI: 10.1093/bioinformatics/btw351. PMID: 27357171. PMCID: PMC5048064
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following examples, M1, M2, and M3 are true mediators.
data(himaDat)

# When Y is continuous and normally distributed
# Example 1 (continuous outcome): 
head(himaDat$Example1$PhenoData)

hima.fit &lt;- hima(X = himaDat$Example1$PhenoData$Treatment, 
                 Y = himaDat$Example1$PhenoData$Outcome, 
                 M = himaDat$Example1$Mediator,
                 COV.XM = himaDat$Example1$PhenoData[, c("Sex", "Age")],
                 Y.family = 'gaussian',
                 scale = FALSE,
                 verbose = TRUE) 
hima.fit

# When Y is binary (should specify Y.family)
# Example 2 (binary outcome): 
head(himaDat$Example2$PhenoData)

hima.logistic.fit &lt;- hima(X = himaDat$Example2$PhenoData$Treatment,
                          Y = himaDat$Example2$PhenoData$Disease,
                          M = himaDat$Example2$Mediator,
                          COV.XM = himaDat$Example2$PhenoData[, c("Sex", "Age")],
                          Y.family = 'binomial',
                          scale = FALSE,
                          verbose = TRUE)
hima.logistic.fit

## End(Not run)

</code></pre>

<hr>
<h2 id='HIMA-package'>High-Dimensional Mediation Analysis for 'Omic' Data</h2><span id='topic+HIMA-package'></span><span id='topic+HIMA'></span>

<h3>Description</h3>

<p>HIMA is an R package for estimating and testing high-dimensional mediation effects in omic studies.
HIMA can perform high-dimensional mediation analysis on a wide range of omic data types as
potential mediators, including epigenetics, transcriptomics, proteomics, and metabolomics using function <code>hima</code> and 
microbiome data (function <code>microHIMA</code>). HIMA can also handle survival data (function <code>survHIMA</code>).
</p>

<table>
<tr>
 <td style="text-align: left;"> Package: </td><td style="text-align: left;"> HIMA</td>
</tr>
<tr>
 <td style="text-align: left;"> Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.2.1</td>
</tr>
<tr>
 <td style="text-align: left;"> Date: </td><td style="text-align: left;"> 2023-09-10</td>
</tr>
<tr>
 <td style="text-align: left;"> License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Details</h3>

<p># If package &quot;qvalue&quot; is not found during installation, please first install &quot;qvalue&quot; package
# through Bioconductor: https://www.bioconductor.org/packages/release/bioc/html/qvalue.html
</p>


<h3>Author(s)</h3>

<p>Yinan Zheng <a href="mailto:y-zheng@northwestern.edu">y-zheng@northwestern.edu</a>,
Haixiang Zhang <a href="mailto:haixiang.zhang@tju.edu.cn">haixiang.zhang@tju.edu.cn</a>,  
Lifang Hou <a href="mailto:l-hou@northwestern.edu">l-hou@northwestern.edu</a>
Lei liu <a href="mailto:lei.liu@wustl.edu">lei.liu@wustl.edu</a>
</p>
<p>Maintainer: Yinan Zheng <a href="mailto:y-zheng@northwestern.edu">y-zheng@northwestern.edu</a>
</p>


<h3>References</h3>

<p>1. Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W, Schwartz J, Just A, Colicino E, Vokonas P, Zhao L, 
Lv J, Baccarelli A, Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects in Epigenetic Studies. 
Bioinformatics. 2016. DOI: 10.1093/bioinformatics/btw351. PMID: 27357171. PMCID: PMC5048064
</p>
<p>2. Perera C, Zhang H, Zheng Y, Hou L, Qu A, Zheng C, Xie K, Liu L. 
HIMA2: high-dimensional mediation analysis and its application in epigenome-wide DNA methylation data. 
BMC Bioinformatics. 2022. DOI: 10.1186/s12859-022-04748-1. PMID: 35879655. PMCID: PMC9310002
</p>
<p>3. Zhang H, Zheng Y, Hou L, Zheng C, Liu L. Mediation Analysis for Survival Data with High-Dimensional Mediators. 
Bioinformatics. 2021. DOI: 10.1093/bioinformatics/btab564. PMID: 34343267. PMCID: PMC8570823
</p>
<p>4. Zhang H, Chen J, Feng Y, Wang C, Li H, Liu L. Mediation effect selection in high-dimensional and compositional microbiome data. 
Stat Med. 2021. DOI: 10.1002/sim.8808. PMID: 33205470; PMCID: PMC7855955
</p>
<p>5. Zhang H, Chen J, Li Z, Liu L. Testing for mediation effect with application to human microbiome data. 
Stat Biosci. 2021. DOI: 10.1007/s12561-019-09253-3. PMID: 34093887; PMCID: PMC8177450
</p>
<p>6. Zhang H, Hong X, Zheng Y, Hou L, Zheng C, Wang X, Liu L. High-Dimensional Quantile Mediation Analysis with Application to a Birth 
Cohort Study of Mother–Newborn Pairs. Bioinformatics. 2023. (In press)
</p>

<hr>
<h2 id='hima2'>Advanced High-dimensional Mediation Analysis</h2><span id='topic+hima2'></span>

<h3>Description</h3>

<p><code>hima2</code> is an upgraded version of hima for estimating and testing high-dimensional mediation effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hima2(
  formula,
  data.pheno,
  data.M,
  outcome.family = c("gaussian", "binomial", "survival", "quantile"),
  mediator.family = c("gaussian", "negbin", "compositional"),
  penalty = c("DBlasso", "MCP", "SCAD", "lasso"),
  topN = NULL,
  scale = TRUE,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hima2_+3A_formula">formula</code></td>
<td>
<p>an object of class <code>formula</code>: a symbolic description of the overall effect model, i.e., 
<code>outcome ~ exposure + covariates</code>, to be fitted. Make sure the &quot;exposure&quot; is the variable of interest, which
must be listed as the first variable in the right hand side of the formula.
independent variable in the formula. The same covariates will be used in screening and penalized regression.</p>
</td></tr>
<tr><td><code id="hima2_+3A_data.pheno">data.pheno</code></td>
<td>
<p>a data frame containing exposure and covariates that are listed in the right hand side of the <code>formula</code>.
The variable names must match those listed in <code>formula</code>. By default <code>hima2</code> will scale <code>data.pheno</code>.</p>
</td></tr>
<tr><td><code id="hima2_+3A_data.m">data.M</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators. Rows represent samples, columns 
represent variables. By default <code>hima2</code> will scale <code>data.M</code>.</p>
</td></tr>
<tr><td><code id="hima2_+3A_outcome.family">outcome.family</code></td>
<td>
<p>either <code>'gaussian'</code> (default, for normally distributed continuous outcome), <code>'binomial'</code> 
(for binay outcome), <code>'survival'</code> (for time-to-event outcome), or <code>'quantile'</code> (for quantile mediation analysis)</p>
</td></tr>
<tr><td><code id="hima2_+3A_mediator.family">mediator.family</code></td>
<td>
<p>either <code>'gaussian'</code> (default, for continuous mediators), <code>'negbin'</code> (i.e., negative binomial, 
for RNA-seq data as mediators), or <code>'compositional'</code> (for microbiome data as mediators), depending on the data type of 
high-dimensional mediators (<code>data.M</code>).</p>
</td></tr>
<tr><td><code id="hima2_+3A_penalty">penalty</code></td>
<td>
<p>the penalty to be applied to the model. Either <code>'DBlasso'</code> (De-biased LASSO, default), 
<code>'MCP'</code>, <code>'SCAD'</code>, or <code>'lasso'</code>. Please note, survival HIMA and microbiome HIMA can be only performed with <code>'DBlasso'</code>; 
Quantile HIMA cannot be performed with <code>'DBlasso'</code>.</p>
</td></tr>
<tr><td><code id="hima2_+3A_topn">topN</code></td>
<td>
<p>an integer specifying the number of top markers from sure independent screening. 
Default = <code>NULL</code>. If <code>NULL</code>, <code>topN</code> will be <code>ceiling(2 * n/log(n))</code>, where <code>n</code> is the sample size. 
If the sample size is greater than topN (pre-specified or calculated), all mediators will be included in the test 
(i.e. a low-dimensional scenario).</p>
</td></tr>
<tr><td><code id="hima2_+3A_scale">scale</code></td>
<td>
<p>logical. Should the function scale the data (exposure, mediators, and covariates)? Default = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="hima2_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose and shows the progression? Default = <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="hima2_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of selected mediators.
</p>


<h3>References</h3>

<p>Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W, Schwartz J, Just A, Colicino E, Vokonas P, Zhao L, 
Lv J, Baccarelli A, Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects in Epigenetic Studies. 
Bioinformatics. 2016. DOI: 10.1093/bioinformatics/btw351. PMID: 27357171. PMCID: PMC5048064
</p>
<p>Perera C, Zhang H, Zheng Y, Hou L, Qu A, Zheng C, Xie K, Liu L. 
HIMA2: high-dimensional mediation analysis and its application in epigenome-wide DNA methylation data. 
BMC Bioinformatics. 2022. DOI: 10.1186/s12859-022-04748-1. PMID: 35879655. PMCID: PMC9310002
</p>
<p>Zhang H, Zheng Y, Hou L, Zheng C, Liu L. Mediation Analysis for Survival Data with High-Dimensional Mediators. 
Bioinformatics. 2021. DOI: 10.1093/bioinformatics/btab564. PMID: 34343267. PMCID: PMC8570823
</p>
<p>Zhang H, Chen J, Feng Y, Wang C, Li H, Liu L. Mediation effect selection in high-dimensional and compositional microbiome data. 
Stat Med. 2021. DOI: 10.1002/sim.8808. PMID: 33205470; PMCID: PMC7855955
</p>
<p>Zhang H, Chen J, Li Z, Liu L. Testing for mediation effect with application to human microbiome data. 
Stat Biosci. 2021. DOI: 10.1007/s12561-019-09253-3. PMID: 34093887; PMCID: PMC8177450
</p>
<p>Zhang H, Hong X, Zheng Y, Hou L, Zheng C, Wang X, Liu L. High-Dimensional Quantile Mediation Analysis with Application to a Birth 
Cohort Study of Mother–Newborn Pairs. Bioinformatics. 2023. (In press)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following examples, M1, M2, and M3 are true mediators.
data(himaDat)

# Example 1 (continous outcome): 
head(himaDat$Example1$PhenoData)

e1 &lt;- hima2(Outcome ~ Treatment + Sex + Age, 
      data.pheno = himaDat$Example1$PhenoData, 
      data.M = himaDat$Example1$Mediator,
      outcome.family = "gaussian",
      mediator.family = "gaussian",
      penalty = "DBlasso",
      scale = FALSE) # Disabled only for example data
e1
attributes(e1)$variable.labels

# Example 2 (binary outcome): 
head(himaDat$Example2$PhenoData)

e2 &lt;- hima2(Disease ~ Treatment + Sex + Age, 
      data.pheno = himaDat$Example2$PhenoData, 
      data.M = himaDat$Example2$Mediator,
      outcome.family = "binomial",
      mediator.family = "gaussian",
      penalty = "DBlasso",
      scale = FALSE) # Disabled only for example data
e2
attributes(e2)$variable.labels

# Example 3 (time-to-event outcome): 
head(himaDat$Example3$PhenoData)

e3 &lt;- hima2(Surv(Status, Time) ~ Treatment + Sex + Age, 
      data.pheno = himaDat$Example3$PhenoData, 
      data.M = himaDat$Example3$Mediator,
      outcome.family = "survival",
      mediator.family = "gaussian",
      penalty = "DBlasso",
      scale = FALSE) # Disabled only for example data
e3
attributes(e3)$variable.labels

# Example 4 (compositional data as mediator, e.g., microbiome): 
head(himaDat$Example4$PhenoData)

e4 &lt;- hima2(Outcome ~ Treatment + Sex + Age, 
      data.pheno = himaDat$Example4$PhenoData, 
      data.M = himaDat$Example4$Mediator,
      outcome.family = "gaussian",
      mediator.family = "compositional",
      penalty = "DBlasso",
      scale = FALSE) # Disabled only for example data
e4
attributes(e4)$variable.labels

#' # Example 5 (quantile mediation anlaysis): 
head(himaDat$Example5$PhenoData)

# Note that the function will prompt input for quantile level.
e5 &lt;- hima2(Outcome ~ Treatment + Sex + Age, 
      data.pheno = himaDat$Example5$PhenoData, 
      data.M = himaDat$Example5$Mediator,
      outcome.family = "quantile",
      mediator.family = "gaussian",
      penalty = "MCP", # Quantile HIMA does not support DBlasso
      scale = FALSE, # Disabled only for example data
      tau = c(0.3, 0.5, 0.7)) # Specify multiple quantile level
e5
attributes(e5)$variable.labels

## End(Not run)
                  
</code></pre>

<hr>
<h2 id='himaDat'>Example datasets for HIMA demo</h2><span id='topic+himaDat'></span>

<h3>Description</h3>

<p>A <code>list</code> dataset containing datasets for various scenarios of HIMA. Each dataset contains a phenotype 
data frame and a high-dimension mediator data matrix. The datasets are simulated using parameters generated 
from real datasets. The code used to generate the data can be found in /inst/script folder of the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>himaDat
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 5.
</p>


<h3>Details</h3>

<p>Example dataset 1 for HIMA: Continuous outcome 
</p>

<ul>
<li><p> Treatment: treated (value = 1) or not treated (value = 0)
</p>
</li>
<li><p> Outcome: outcome of the treatment- a normally distributed continuous variable
</p>
</li>
<li><p> Sex: female (value = 1) or male (value = 0)
</p>
</li>
<li><p> Age: Age of the participant
</p>
</li></ul>

<p>Example dataset 2 for HIMA: Binary outcome 
</p>

<ul>
<li><p> Treatment: treated (value = 1) or not treated (value = 0)
</p>
</li>
<li><p> Disease: diseased (value = 1) or healthy (value = 0)
</p>
</li>
<li><p> Sex: female (value = 1) or male (value = 0)
</p>
</li>
<li><p> Age: Age of the participant
</p>
</li></ul>

<p>Example dataset 3 for HIMA: Survival data outcome
</p>

<ul>
<li><p> Treatment: treated (value = 1) or not treated (value = 0)
</p>
</li>
<li><p> Status: Status indicator: dead (value = 1) or alive (value = 0)
</p>
</li>
<li><p> Time: time to event
</p>
</li>
<li><p> Sex: female (value = 1) or male (value = 0)
</p>
</li>
<li><p> Age: Age of the participant
</p>
</li></ul>

<p>Example dataset 4 for HIMA: Compositional mediator (e.g., microbiome)
</p>

<ul>
<li><p> Treatment: treated (value = 1) or not treated (value = 0)
</p>
</li>
<li><p> Outcome: outcome of the treatment- a normally distributed continuous variable
</p>
</li>
<li><p> Sex: female (value = 1) or male (value = 0)
</p>
</li>
<li><p> Age: Age of the participant
</p>
</li></ul>

<p>Example dataset 5 for HIMA: High-dimensional quantile mediation analysis
</p>

<ul>
<li><p> Treatment: treated (value = 1) or not treated (value = 0)
</p>
</li>
<li><p> Outcome: outcome of the treatment- abnormally distributed continuous variable
</p>
</li>
<li><p> Sex: female (value = 1) or male (value = 0)
</p>
</li>
<li><p> Age: Age of the participant
</p>
</li></ul>



<h3>Value</h3>

<p>A list of example datasets for HIMA demo and testing.
</p>

<hr>
<h2 id='microHIMA'>High-dimensional mediation analysis for compositional microbiome data</h2><span id='topic+microHIMA'></span>

<h3>Description</h3>

<p><code>microHIMA</code> is used to estimate and test high-dimensional mediation effects for compositional microbiome data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>microHIMA(X, Y, OTU, COV = NULL, FDRcut = 0.05, scale = TRUE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="microHIMA_+3A_x">X</code></td>
<td>
<p>a vector of exposure.</p>
</td></tr>
<tr><td><code id="microHIMA_+3A_y">Y</code></td>
<td>
<p>a vector of outcome.</p>
</td></tr>
<tr><td><code id="microHIMA_+3A_otu">OTU</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional compositional OTUs (mediators). Rows represent samples, 
columns represent variables.</p>
</td></tr>
<tr><td><code id="microHIMA_+3A_cov">COV</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of adjusting covariates. Rows represent samples, columns represent microbiome variables. 
Can be <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="microHIMA_+3A_fdrcut">FDRcut</code></td>
<td>
<p>FDR cutoff applied to define and select significant mediators. Default = <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="microHIMA_+3A_scale">scale</code></td>
<td>
<p>logical. Should the function scale the data? Default = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="microHIMA_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose? Default = <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of selected mediators (FDR &lt; <code>FDRcut</code>). 
</p>

<ul>
<li><p>ID: index of selected significant mediator.
</p>
</li>
<li><p>alpha: coefficient estimates of exposure (X) &ndash;&gt; mediators (M).
</p>
</li>
<li><p>alpha_se: standard error for alpha.
</p>
</li>
<li><p>beta: coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for exposure).
</p>
</li>
<li><p>beta_se: standard error for beta.
</p>
</li>
<li><p>FDR: false discovery rate of selected significant mediator.
</p>
</li></ul>



<h3>References</h3>

<p>Zhang H, Chen J, Feng Y, Wang C, Li H, Liu L. Mediation effect selection in high-dimensional and compositional microbiome data. 
Stat Med. 2021. DOI: 10.1002/sim.8808. PMID: 33205470; PMCID: PMC7855955.
</p>
<p>Zhang H, Chen J, Li Z, Liu L. Testing for mediation effect with application to human microbiome data. 
Stat Biosci. 2021. DOI: 10.1007/s12561-019-09253-3. PMID: 34093887; PMCID: PMC8177450.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following example, M1, M2, and M3 are true mediators.
data(himaDat)

head(himaDat$Example4$PhenoData)

microHIMA.fit &lt;- microHIMA(X = himaDat$Example4$PhenoData$Treatment, 
                           Y = himaDat$Example4$PhenoData$Outcome, 
                           OTU = himaDat$Example4$Mediator, 
                           COV = himaDat$Example4$PhenoData[, c("Sex", "Age")],
                           scale = FALSE)
microHIMA.fit

## End(Not run)

</code></pre>

<hr>
<h2 id='qHIMA'>High-dimensional quantile mediation analysis</h2><span id='topic+qHIMA'></span>

<h3>Description</h3>

<p><code>qHIMA</code> is used to estimate and test high-dimensional quantile mediation effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qHIMA(
  X,
  M,
  Y,
  Z,
  penalty = c("MCP", "SCAD", "lasso"),
  topN = NULL,
  tau = 0.5,
  scale = TRUE,
  Bonfcut = 0.05,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qHIMA_+3A_x">X</code></td>
<td>
<p>a vector of exposure.</p>
</td></tr>
<tr><td><code id="qHIMA_+3A_m">M</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators. Rows represent samples, columns 
represent mediator variables.</p>
</td></tr>
<tr><td><code id="qHIMA_+3A_y">Y</code></td>
<td>
<p>a vector of continuous outcome. Do not use data.frame or matrix.</p>
</td></tr>
<tr><td><code id="qHIMA_+3A_z">Z</code></td>
<td>
<p>a matrix of adjusting covariates. Rows represent samples, columns represent variables. Can be <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="qHIMA_+3A_penalty">penalty</code></td>
<td>
<p>the penalty to be applied to the model (a parameter passed to function <code>conquer.cv.reg</code> in package <code><a href="conquer.html#topic+conquer">conquer</a></code>. 
Either 'MCP' (the default), 'SCAD', or 'lasso'.</p>
</td></tr>
<tr><td><code id="qHIMA_+3A_topn">topN</code></td>
<td>
<p>an integer specifying the number of top markers from sure independent screening. 
Default = <code>NULL</code>. If <code>NULL</code>, <code>topN</code> will be <code>2*ceiling(n/log(n))</code>, where <code>n</code> is the sample size.
If the sample size is greater than topN (pre-specified or calculated), all mediators will be included in the test (i.e. low-dimensional scenario).</p>
</td></tr>
<tr><td><code id="qHIMA_+3A_tau">tau</code></td>
<td>
<p>quantile level of outcome. Default = 0.5. A vector of tau is accepted.</p>
</td></tr>
<tr><td><code id="qHIMA_+3A_scale">scale</code></td>
<td>
<p>logical. Should the function scale the data? Default = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="qHIMA_+3A_bonfcut">Bonfcut</code></td>
<td>
<p>Bonferroni-corrected p value cutoff applied to define and select significant mediators. Default = <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="qHIMA_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose? Default = <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="qHIMA_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of selected mediators (Bonferroni-adjusted p value &lt;<code>Bonfcut</code>). 
</p>

<ul>
<li><p>ID: index of selected significant mediator.
</p>
</li>
<li><p>alpha: coefficient estimates of exposure (X) &ndash;&gt; mediators (M).
</p>
</li>
<li><p>alpha_se: standard error for alpha.
</p>
</li>
<li><p>beta: coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for exposure).
</p>
</li>
<li><p>beta_se: standard error for beta.
</p>
</li>
<li><p>Bonferroni.p: statistical significance of the mediator (Bonferroni-corrected p value).
</p>
</li></ul>



<h3>References</h3>

<p>Zhang H, Hong X, Zheng Y, Hou L, Zheng C, Wang X, Liu L. High-Dimensional Quantile Mediation Analysis with Application to a Birth 
Cohort Study of Mother–Newborn Pairs. Bioinformatics. 2023. (In press)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following example, M1, M2, and M3 are true mediators.
data(himaDat)

head(himaDat$Example5$PhenoData)

qHIMA.fit &lt;- qHIMA(X = himaDat$Example5$PhenoData$Treatment,
                M = himaDat$Example5$Mediator, 
                Y = himaDat$Example5$PhenoData$Outcome, 
                Z = himaDat$Example5$PhenoData[, c("Sex", "Age")], 
                Bonfcut = 0.05,
                tau = c(0.3, 0.5, 0.7),
                scale = FALSE, 
                verbose = TRUE)
qHIMA.fit

## End(Not run)

</code></pre>

<hr>
<h2 id='survHIMA'>High-dimensional mediation analysis for survival data</h2><span id='topic+survHIMA'></span>

<h3>Description</h3>

<p><code>survHIMA</code> is used to estimate and test high-dimensional mediation effects for survival data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survHIMA(X, Z, M, OT, status, FDRcut = 0.05, scale = TRUE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="survHIMA_+3A_x">X</code></td>
<td>
<p>a vector of exposure.</p>
</td></tr>
<tr><td><code id="survHIMA_+3A_z">Z</code></td>
<td>
<p>a matrix of adjusting covariates. Rows represent samples, columns represent variables. Can be <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="survHIMA_+3A_m">M</code></td>
<td>
<p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators. Rows represent samples, columns 
represent mediator variables.</p>
</td></tr>
<tr><td><code id="survHIMA_+3A_ot">OT</code></td>
<td>
<p>a vector of observed failure times.</p>
</td></tr>
<tr><td><code id="survHIMA_+3A_status">status</code></td>
<td>
<p>a vector of censoring indicator (<code>status = 1</code>: uncensored; <code>status = 0</code>: censored)</p>
</td></tr>
<tr><td><code id="survHIMA_+3A_fdrcut">FDRcut</code></td>
<td>
<p>FDR cutoff applied to define and select significant mediators. Default = <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="survHIMA_+3A_scale">scale</code></td>
<td>
<p>logical. Should the function scale the data? Default = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="survHIMA_+3A_verbose">verbose</code></td>
<td>
<p>logical. Should the function be verbose? Default = <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing mediation testing results of selected mediators (FDR &lt;<code>FDPcut</code>). 
</p>

<ul>
<li><p>ID: index of selected significant mediator.
</p>
</li>
<li><p>alpha: coefficient estimates of exposure (X) &ndash;&gt; mediators (M).
</p>
</li>
<li><p>alpha_se: standard error for alpha.
</p>
</li>
<li><p>beta: coefficient estimates of mediators (M) &ndash;&gt; outcome (Y) (adjusted for exposure).
</p>
</li>
<li><p>beta_se: standard error for beta.
</p>
</li>
<li><p>p.joint: joint raw p-value of selected significant mediator (based on FDR).
</p>
</li></ul>



<h3>References</h3>

<p>Zhang H, Zheng Y, Hou L, Zheng C, Liu L. Mediation Analysis for Survival Data with High-Dimensional Mediators. 
Bioinformatics. 2021. DOI: 10.1093/bioinformatics/btab564. PMID: 34343267. PMCID: PMC8570823
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Note: In the following example, M1, M2, and M3 are true mediators.
data(himaDat)

head(himaDat$Example3$PhenoData)

survHIMA.fit &lt;- survHIMA(X = himaDat$Example3$PhenoData$Treatment,
                Z = himaDat$Example3$PhenoData[, c("Sex", "Age")], 
                M = himaDat$Example3$Mediator, 
                OT = himaDat$Example3$PhenoData$Time, 
                status = himaDat$Example3$PhenoData$Status, 
                FDRcut = 0.05,
                scale = FALSE, 
                verbose = TRUE)
survHIMA.fit

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
