<!DOCTYPE html><html lang="en"><head><title>Help for package hiphop</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hiphop}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#combinations'><p>List of triads to be ranked for parentage assignment.</p></a></li>
<li><a href='#genotypes'><p>Genotypes of individuals scored at different loci</p></a></li>
<li><a href='#hothiphop'><p>A function that calculates the genetic mismatches according to the hiphop and hot</p>
test</a></li>
<li><a href='#individuals'><p>List of individuals to be compared for parentage assignment.</p></a></li>
<li><a href='#inspect'><p>A function that inspects the genotypes of individuals</p></a></li>
<li><a href='#topmatch'><p>A function selecting the dams and sires with the fewest genetic mismatches to an offspring</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Parentage Assignment using Bi-Allelic Genetic Markers</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Can be used for paternity and maternity assignment and outperforms 
  conventional methods where closely related individuals occur in the pool of 
  possible parents. The method compares the genotypes of offspring with any 
  combination of potentials parents and scores the number of mismatches of these 
  individuals at bi-allelic genetic markers (e.g. Single Nucleotide Polymorphisms). 
  It elaborates on a prior exclusion method based on the Homozygous Opposite Test 
  (HOT; Huisman 2017 &lt;<a href="https://doi.org/10.1111%2F1755-0998.12665">doi:10.1111/1755-0998.12665</a>&gt;) by introducing the 
  additional exclusion criterion HIPHOP (Homozygous Identical Parents, 
  Heterozygous Offspring are Precluded; Cockburn et al., in revision). Potential 
  parents are excluded if they have more mismatches than can 
  be expected due to genotyping error and mutation, and thereby one can identify the 
  true genetic parents and detect situations where one (or both) of the true 
  parents is not sampled. Package 'hiphop' can deal with (a) the case where there 
  is contextual information about parentage of the mother (i.e. a female has been 
  seen to be involved in reproductive tasks such as nest building), but paternity 
  is unknown (e.g. due to promiscuity), (b) where both parents need to be assigned, 
  because there is no contextual information on which female laid eggs and which 
  male fertilized them (e.g. polygynandrous mating system where multiple females 
  and males deposit young in a common nest, or organisms with external fertilisation 
  that breed in aggregations).
    For details: Cockburn, A., Penalba, J.V.,Jaccoud, D.,Kilian, A., Brouwer, L.,
       Double, M.C., Margraf, N., Osmond, H.L., van de Pol, M. and Kruuk, L.E.B. 
       (in revision). HIPHOP: improved paternity assignment among close relatives 
       using a simple exclusion method for bi-allelic markers. 
       Molecular Ecology Resources, DOI to be added upon acceptance.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://groups.google.com/d/forum/r-hiphop">https://groups.google.com/d/forum/r-hiphop</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-08-12 10:13:11 UTC; marti</td>
</tr>
<tr>
<td>Author:</td>
<td>Martijn van de Pol
    <a href="https://orcid.org/0000-0003-4102-4079"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Lyanne Brouwer <a href="https://orcid.org/0000-0001-6728-4851"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Andrew Cockburn <a href="https://orcid.org/0000-0002-8531-3350"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Martijn van de Pol &lt;martijn@myscience.eu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-08-19 09:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='combinations'>List of triads to be ranked for parentage assignment.</h2><span id='topic+combinations'></span>

<h3>Description</h3>

<p>An example list of how the output from the hothiphop function looks like.
Listed are all possible triad combinations for 8 offspring from the 2018 cohort
This dataframe is also used to generate the vignette.
See the help file of the hothiphop function for an explanation of columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combinations
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 24192 rows and 24 columns.
</p>

<hr>
<h2 id='genotypes'>Genotypes of individuals scored at different loci</h2><span id='topic+genotypes'></span>

<h3>Description</h3>

<p>A dataset containing genotypes of 1407 superb fairy wrens (Malurus cyaneus)
from five cohorts (breeding seasons 2014-2018) from the Australian National Botanic Garden.
Each individual (rows) is scored at 1376 loci (columns) with the scores meaning:
0: homozygotes at common allele; 1: homozygote at rare allele and
2: heterozygotes; NA: locus could not be scored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotypes
</code></pre>


<h3>Format</h3>

<p>A data frame with 1407 rows (individuals) and 1376 variables (loci)
</p>


<h3>Author(s)</h3>

<p>Andrew Cockburn, <a href="mailto:andrew.cockburn@anu.edu.au">andrew.cockburn@anu.edu.au</a>
</p>


<h3>Source</h3>

<p>Cockburn et al. (2020) HIPHOP: improved paternity assignment among
close relatives using a simple exclusion method for biallelic markers.
Molecular Ecology Resources, in revision.
</p>

<hr>
<h2 id='hothiphop'>A function that calculates the genetic mismatches according to the hiphop and hot
test</h2><span id='topic+hothiphop'></span>

<h3>Description</h3>

<p>This function  calculates the number genetic mismatches according to the hiphop and hot
test for any combination of offspring-potential.dam-potential.sire.
The HOT test (Homozygous Opposite Test; Huisman 2017) compares the genotype of an offspring with a potential parent:
a mismatch is scored when both the offspring and parent are homozygous, but for different alleles.
The HIPHOP test (Homozygous Identical Parents, Heterozygous Offspring are Precluded; Cockburn et al. in revision)
compares the genotype of an offspring with both potential parents:
a mismatch is scored when the offspring is heterozygous and both parent are homozygous for the same allele
The resulting output can next be summarized using the 'topmatch()' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hothiphop(ind, gen)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hothiphop_+3A_ind">ind</code></td>
<td>
<p>The input file with individuals, which should contain at least the columns
brood, individual, type, social.parent, year.</p>
</td></tr>
<tr><td><code id="hothiphop_+3A_gen">gen</code></td>
<td>
<p>The input file with genotypes, which should contain the loci-names as column headers
and the individual  names as row-header and should only contain the values 0, 1, 2 or NA.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe with all possible  offspring-potential.dam-potential.sire combinations
and their mismatch scores according to the HOT and HIPHOP test, the number of loci this was based on,
and some additional relevant information about the social parents and potential dam and sires
</p>

<dl>
<dt>year</dt><dd><p> the year or cohort that is being considered, adults can be potential dam or sire in some years, but no in others</p>
</dd>
<dt>brood</dt><dd><p> an identifier of the brood to which the offspring and adults belong/are associated with</p>
</dd>
<dt>offspring</dt><dd><p> an identifier of the offspring </p>
</dd>
<dt>potential.dam</dt><dd><p> an identifier of the potential dam </p>
</dd>
<dt>potential.sire</dt><dd><p> an identifier of the potential sire </p>
</dd>
<dt>hothiphop.parents</dt><dd><p> the sum of the hiphop and hot.parents mismatch score </p>
</dd>
<dt>hiphop</dt><dd><p> the hiphop mismatch score of the offspring with the potential dam and potential sire, expressed as the number of loci giving mismatches </p>
</dd>
<dt>hot.parents</dt><dd><p> the hot score of the offspring with both the potential dam and sire, expressed as the number of loci giving mismatches</p>
</dd>
<dt>hot.dam</dt><dd><p> the hot score of the offspring with the potential dam, expressed as the number of loci giving mismatches</p>
</dd>
<dt>hot.sire</dt><dd><p> the hot mismatch score of the offspring with the potential sire , expressed as the number of loci giving mismatches</p>
</dd>
<dt>hothiphop.dam</dt><dd><p> the sum of the hot.dam and hiphop mismatch score </p>
</dd>
<dt>hothiphop.sire</dt><dd><p> the sum of the hot.sire and hiphop mismatch score </p>
</dd>
<dt>loci.dyad.dam</dt><dd><p> the number of loci at which both the offspring and dam were not NA </p>
</dd>
<dt>loci.dyad.sire</dt><dd><p>the number of loci at which both the offspring and sire were not NA </p>
</dd>
<dt>loci.triad</dt><dd><p> the number of loci at which the offspring, dam and sire were not NA </p>
</dd>
<dt>offspring.heterozygosity</dt><dd><p> proportion of loci at which the offspring was heterozygous </p>
</dd>
<dt>social.mother.sampled</dt><dd><p>if the social.mother genotypic data is in the genotypes file then equal to 1, else 0</p>
</dd>
<dt>social.father.sampled</dt><dd><p>if the social.father genotypic data is in the genotypes file then equal to 1, else 0</p>
</dd>
<dt>is.dam.social</dt><dd><p>if the potential dam is the social mother then equal to 1, else 0</p>
</dd>
<dt>is.sire.social</dt><dd><p>if the potential sire is the social father then equal to 1, else 0</p>
</dd>
<dt>is.dam.within.group</dt><dd><p>if the potential dam is part of the same group (i.e. associated with the same brood) as the offspring, then equal to 1, else 0</p>
</dd>
<dt>is.sire.within.group</dt><dd><p>if the potential sire is part of the same group (i.e. associated with the same brood) as the offspring, then equal to 1, else 0</p>
</dd>
<dt>social.mother</dt><dd><p> identity of the social mother of the offspring</p>
</dd>
<dt>social.father</dt><dd><p> identity of the social father of the offspring</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Martijn van de Pol, <a href="mailto:martijn@myscience.eu">martijn@myscience.eu</a>
</p>


<h3>Source</h3>

<p>Cockburn et al. (2020) HIPHOP: improved paternity assignment among
close relatives using a simple exclusion method for biallelic markers.
Molecular Ecology Resources, in revision.
</p>


<h3>References</h3>

<p>Cockburn et al. (2020) HIPHOP: improved paternity assignment among close relatives using a simple exclusion method for biallelic markers.
Molecular Ecology Resources, in revision.
</p>
<p>Huisman, J. (2017). Pedigree reconstruction from SNP data: parentage assignment, sibship clustering and beyond. Molecular ecology resources, 17(5), 1009-1024.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>results&lt;-hothiphop(ind=individuals[1:22,], gen=genotypes)
head(results)
best&lt;-topmatch(x=results, ranking="hothiphop.parents")
head(best)
</code></pre>

<hr>
<h2 id='individuals'>List of individuals to be compared for parentage assignment.</h2><span id='topic+individuals'></span>

<h3>Description</h3>

<p>A list of 2527 individuals (superb fairy wrens; Malurus cyaneus)) of which their genetics are to be compared to determine parentage.
The dataset consists of 1153 offspring, 469 adult females that are potential dams and 905 adult males that are potential sires.
Data is from five cohorts (breeding seasons 2014-2018) from the Australian National Botanic Garden.
Note that individuals can occur multiple times in the dataset, as adults can have parentage in multiple years.
Also offspring can become adults in future years.
Also if an individual is associated with multiple broods as social parent this means there can be multiple records per year.
Note that the columns social parent and brood are only used to determine whether a potential dam or sire is the social parent,
an extra-group (technically extra-brood) parent or within-group parent that is not the social parent (a subordinate).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>individuals
</code></pre>


<h3>Format</h3>

<p>A data frame with 2527 rows and 6 variables:
</p>

<dl>
<dt>brood</dt><dd><p> an identifier of the brood to which the offspring and adults belong/are associated with</p>
</dd>
<dt>individual</dt><dd><p> an identifier of individual </p>
</dd>
<dt>type</dt><dd><p> denotes whether the individual is an offspring, adult female (potential dam) or adult male (potential sire)</p>
</dd>
<dt>social.parent</dt><dd><p> if the individual is the social parent of the brood then equal to 1, else 0</p>
</dd>
<dt>year</dt><dd><p> the year or cohort that is being considered, adults can be potential dam or sire in some years, but no in others</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Andrew Cockburn, <a href="mailto:andrew.cockburn@anu.edu.au">andrew.cockburn@anu.edu.au</a>
</p>


<h3>Source</h3>

<p>Cockburn et al. (2020) HIPHOP: improved paternity assignment among
close relatives using a simple exclusion method for biallelic markers.
Molecular Ecology Resources, in revision.
</p>

<hr>
<h2 id='inspect'>A function that inspects the genotypes of individuals</h2><span id='topic+inspect'></span>

<h3>Description</h3>

<p>This function allows one to inspect the genotypes of all individuals in your input file to see
whether genetic data is available, and if so what the proportion of
homozygotes, heterozygotes fraction of missing values is.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inspect(ind, gen)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inspect_+3A_ind">ind</code></td>
<td>
<p>The input file with individuals, which should contain at least the columns
brood, individual, type, social.parent, year.</p>
</td></tr>
<tr><td><code id="inspect_+3A_gen">gen</code></td>
<td>
<p>The input file with genotypes, which should contain the loci-names as column headers
and the individual  names as row-header and should only contain the values 0, 1, 2 or NA.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the individuals file with a summary of the genotypes attached.
#' </p>

<dl>
<dt>brood</dt><dd><p> an identifier of the brood to which the offspring and adults belong/are associated with</p>
</dd>
<dt>individual</dt><dd><p> an identifier of the offspring </p>
</dd>
<dt>type</dt><dd><p> denotes whether the individual is an offspring, adult female (potential dam) or adult male (potential sire)</p>
</dd>
<dt>social.parent</dt><dd><p> if the individual is the social parent of the brood then equal to 1, else 0</p>
</dd>
<dt>year</dt><dd><p> the year or cohort that is being considered, adults can be potential dam or sire in some years, but no in others</p>
</dd>
<dt>sampled</dt><dd><p>if the individual's genotypic data is in the genotypes file then equal to 1, else 0</p>
</dd>
<dt>homozygote.0</dt><dd><p> the proportion of loci that was scored as 0 (homozygotes at common allele) </p>
</dd>
<dt>homozygote.1</dt><dd><p> the proportion of loci that was scored as 1 (homozygotes at rare allele) </p>
</dd>
<dt>homozygote.2</dt><dd><p> the proportion of loci that was scored as 2 (heterozygote) </p>
</dd>
<dt>missing</dt><dd><p> the proportion of loci that has missing values (NA) </p>
</dd>
<dt>number.loci</dt><dd><p> the number of loci that has genotypic information  (i.e. not NA) </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Martijn van de Pol, <a href="mailto:martijn@myscience.eu">martijn@myscience.eu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>overview&lt;-inspect(ind=individuals[1:22,], gen=genotypes)
head(overview)
</code></pre>

<hr>
<h2 id='topmatch'>A function selecting the dams and sires with the fewest genetic mismatches to an offspring</h2><span id='topic+topmatch'></span>

<h3>Description</h3>

<p>This function summarizes per offspring the top combinations of dam and sire
that have the least genetic mismatches according to the hot (Huisman 2017) and/or hiphop (Cockburn et al. in revision) test criteria.
In addition to the top matched combinations the summary always also list the social parent, if not among the top X.
The user can choose whether one wants to look for the most likely dam and sire with and without
assuming that the social mother (or father) is the genetic parent. Furthermore, one can choose on which
test score to rank individuals. For more information and worked examples, see the vignette and Cockburn et al. (in revision).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>topmatch(
  x,
  ranking,
  condition = "none",
  thres = 99999,
  top = 3,
  unique = "pair"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="topmatch_+3A_x">x</code></td>
<td>
<p>The file with hot and hiphop scores that is generated by the 'hothiphop()' function.</p>
</td></tr>
<tr><td><code id="topmatch_+3A_ranking">ranking</code></td>
<td>
<p>This sets the mismatch criterion in which dams and sires are ranked,
possibilities include ranking=&quot;hothiphop.parents&quot;, &quot;hiphop&quot;, &quot;hot.parents&quot;, hot.dam&quot;, &quot;hot.sire&quot;, &quot;hothiphop.dam&quot;, &quot;hothiphop.sire&quot;.
In some situations it can be useful to supply two ranking criteria, for example to avoid ties (e.g. ranking=c(&quot;hot.dam&quot;, &quot;hiphop))</p>
</td></tr>
<tr><td><code id="topmatch_+3A_condition">condition</code></td>
<td>
<p>Whether or not one wants to condition on either the social mother (condition=&quot;mother&quot;)
or social father (condition=&quot;father&quot;). The default is condition=&quot;none&quot; in which case all
possible dams and sires are considered as genetic parents.</p>
</td></tr>
<tr><td><code id="topmatch_+3A_thres">thres</code></td>
<td>
<p>Sets a threshold value for the number of mismatches, if parents have less mismatches
then the variable below.threshold in the output will have a value of 1.
If 0&gt;=thres&lt;1, the number of mismatches will be re-expressed as a proportion of all loci sampled (not NA).
If -1&gt;thres&lt;0  the mismatch scores are expressed as the number of mismatched divided by the number of loci at which the offspring is heterozygote (for the HIPHOP test) or divided by  number of loci at which the offspring is homozygote (for the HOT test). This standardization may be sometimes useful for the recalculating the HIPHOP score to account for the fact that some offspring can be scored at more loci than others for this test.
By default thres is set to a large integer (99999).</p>
</td></tr>
<tr><td><code id="topmatch_+3A_top">top</code></td>
<td>
<p>Sets the number of top matches that is shown in the output
#' @return a dataframe with for each offspring the top X (default top 3, giving 3 rows for each offspring) dam and sire combinations
and their mismatch scores according to the HOT and HIPHOP test, the number of loci this was based on,
and some additional relevant information about the social parents and potential dam and sires.
In addition to the top X offspring-dam-sire with the fewest mismatches, the scores of the social parents are also always listed
(if they were not already in the top X ranking).
#' </p>

<dl>
<dt>year</dt><dd><p> the year or cohort that is being considered, adults can be potential dam or sire in some years, but no in others</p>
</dd>
<dt>brood</dt><dd><p> an identifier of the brood to which the offspring and adults belong/are associated with</p>
</dd>
<dt>offspring</dt><dd><p> an identifier of the offspring </p>
</dd>
<dt>rank</dt><dd><p> the ranking among all possible combinations of offspring and potential dams and sires, rank 1 is fewest mismatches according to criterion chosen in the function argument 'ranking' </p>
</dd>
<dt>dam</dt><dd><p> an identifier of the potential dam </p>
</dd>
<dt>sire</dt><dd><p> an identifier of the potential sire </p>
</dd>
<dt>dam.type</dt><dd><p> description of the type of potential dam (social parent, extra-group parent, within-group subordinate) </p>
</dd>
<dt>sire.type</dt><dd><p> description of the type of potential sire (social parent, extra-group parent, within-group subordinate) </p>
</dd>
<dt>hothiphop.parents</dt><dd><p> the sum of the hiphop and hot.parents mismatch score </p>
</dd>
<dt>hiphop</dt><dd><p> the hiphop mismatch score of the offspring with the potential dam and potential sire, expressed as the number of loci giving mismatches </p>
</dd>
<dt>hot.parents</dt><dd><p> the hot score of the offspring with both the potential dam and sire, expressed as the number of loci giving mismatches</p>
</dd>
<dt>hot.dam</dt><dd><p> the hot score of the offspring with the potential dam, expressed as the number of loci giving mismatches</p>
</dd>
<dt>hot.sire</dt><dd><p> the hot mismatch score of the offspring with the potential sire , expressed as the number of loci giving mismatches</p>
</dd>
<dt>hothiphop.dam</dt><dd><p> the sum of the hot.dam and hiphop mismatch score </p>
</dd>
<dt>hothiphop.sire</dt><dd><p> the sum of the hot.sire and hiphop mismatch score </p>
</dd>
<dt>below.threshold</dt><dd><p> if the score chosen in argument 'ranking' is below the chose threshold value, then equals to 1, else 0 </p>
</dd>
<dt>threshold</dt><dd><p> the chosen threshold value in argument 'threshold'</p>
</dd>
<dt>loci.dyad.dam</dt><dd><p> the number of loci at which both the offspring and dam were not NA </p>
</dd>
<dt>loci.dyad.sire</dt><dd><p>the number of loci at which both the offspring and sire were not NA </p>
</dd>
<dt>loci.triad</dt><dd><p> the number of loci at which the offspring, dam and sire were not NA </p>
</dd>
<dt>offspring.heterozygosity</dt><dd><p> the proportion of loci at which the offspring was heterozygous </p>
</dd>
<dt>social.mother.sampled</dt><dd><p>if the social.mother genotypic data is in the genotypes file then equal to 1, else 0</p>
</dd>
<dt>social.father.sampled</dt><dd><p>if the social.father genotypic data is in the genotypes file then equal to 1, else 0</p>
</dd>
<dt>social.mother</dt><dd><p> identity of the social mother of the offspring</p>
</dd>
<dt>social.father</dt><dd><p> identity of the social father of the offspring</p>
</dd>
<dt>condition</dt><dd><p> the value of the argument 'condition'</p>
</dd>
<dt>ranking</dt><dd><p> the value of the (first) value in the argument 'ranking'</p>
</dd>
<dt>ranking2</dt><dd><p> the value of the second value in the argument 'ranking'</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="topmatch_+3A_unique">unique</code></td>
<td>
<p>When using the ranking criteria hot.dam or hot.sire the best ranked dam or sire will occupy the entire top 3 and
this does not allow comparing among dams. By setting unique to &quot;dam&quot; (or &quot;sire&quot;) the output shows the
top 3 with the best record for each unique dam (or sire).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martijn van de Pol, <a href="mailto:martijn@myscience.eu">martijn@myscience.eu</a>
</p>


<h3>Source</h3>

<p>Cockburn et al. (2020) HIPHOP: improved paternity assignment among
close relatives using a simple exclusion method for biallelic markers.
Molecular Ecology Resources, in revision.
</p>


<h3>References</h3>

<p>Cockburn et al. (2020) HIPHOP: improved paternity assignment among close relatives using a simple exclusion method for biallelic markers.
Molecular Ecology Resources, in revision.
</p>
<p>Huisman, J. (2017). Pedigree reconstruction from SNP data: parentage assignment, sibship clustering and beyond. Molecular ecology resources, 17(5), 1009-1024.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>results&lt;-hothiphop(ind=individuals[1:22,], gen=genotypes)
best&lt;-topmatch(x=results, ranking="hothiphop.parents")
head(best)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
