<!DOCTYPE html><html><head><title>Help for package hJAM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hJAM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#betas.Gy'><p>Example beta list of hJAM</p></a></li>
<li><a href='#conditional_A'><p>Example conditional A matrix of hJAM</p></a></li>
<li><a href='#get_cond_A'><p>Compute conditional Z matrix</p></a></li>
<li><a href='#get_cond_alpha'><p>Compute conditional alphas</p></a></li>
<li><a href='#Gl'><p>Example reference data of hJAM</p></a></li>
<li><a href='#hJAM_egger'><p>Fit hJAM with Egger regression</p></a></li>
<li><a href='#hJAM_lnreg'><p>Fit hJAM with linear regression</p></a></li>
<li><a href='#marginal_A'><p>Example marginal A matrix of hJAM</p></a></li>
<li><a href='#output.format'><p>Keep the output as three digits</p></a></li>
<li><a href='#print.hJAM_egger'><p>Print out for hJAM_egger</p></a></li>
<li><a href='#print.hJAM_lnreg'><p>Print out for hJAM_lnreg</p></a></li>
<li><a href='#SNPs_heatmap'><p>Heatmap for all the SNPs used in the analysis</p></a></li>
<li><a href='#SNPs_info'><p>Example SNPs' information of hJAM</p></a></li>
<li><a href='#SNPs_scatter_plot'><p>Scatter plot for all the SNPs used in the analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Hierarchical Joint Analysis of Marginal Summary Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Lai Jiang &lt;jian848@usc.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lai Jiang &lt;jian848@usc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions to implement a hierarchical approach which is designed to perform joint analysis of summary statistics using the framework of Mendelian Randomization or transcriptome analysis. Reference: Lai Jiang, Shujing Xu, Nicholas Mancuso, Paul J. Newcombe, David V. Conti (2020). "A Hierarchical Approach Using Marginal Summary Statistics for Multiple Intermediates in a Mendelian Randomization or Transcriptome Analysis." &lt;bioRxiv&gt;&lt;<a href="https://doi.org/10.1101%2F2020.02.03.924241">doi:10.1101/2020.02.03.924241</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/lailylajiang/hJAM">https://github.com/lailylajiang/hJAM</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/lailylajiang/hJAM/issues">https://github.com/lailylajiang/hJAM/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, ggpubr, dplyr, reshape2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-02-12 01:12:52 UTC; jian848</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-02-20 14:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='betas.Gy'>Example beta list of hJAM</h2><span id='topic+betas.Gy'></span>

<h3>Description</h3>

<p>Example beta list of hJAM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betas.Gy
</code></pre>


<h3>Format</h3>

<p>The <code>betas.Gy</code> is the beta vector in the hJAM model: the association estimates between 210 SNPs and myocardial infarction. The summary data was collected from UK Biobank (n=459,324).</p>


<h3>References</h3>

<p>Sudlow C, Gallacher J, Allen N, et al. UK biobank: an open access resource for identifying the causes of a wide range of complex diseases of middle and old age. PLoS Med 2015; 12: e1001779.
</p>

<hr>
<h2 id='conditional_A'>Example conditional A matrix of hJAM</h2><span id='topic+conditional_A'></span>

<h3>Description</h3>

<p>Example conditional A matrix of hJAM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>conditional_A
</code></pre>


<h3>Format</h3>

<p>The <code>conditional_A</code> is the conditional estimates alpha matrix in the hJAM model: the association estimates between 210 SNPs and body mass index (BMI) and type 2 diabetes (T2D). The summary data was collected from GIANT consortium (n=339,224) and DIAGRAM+GERA+UKB (n=659316) for BMI and T2D, respectively. We converted it from marginal_A, using <code>get_cond_A</code> function in hJAM package.</p>


<h3>References</h3>

<p>1. Locke AE, Kahali B, Berndt SI, et al. Genetic studies of body mass index yield new insights for obesity biology. Nature 2015; 518: 197-206. 2. Xue A, Wu Y, Zhu Z, et al. Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes. Nat Commun 2018; 9: 2941.
</p>

<hr>
<h2 id='get_cond_A'>Compute conditional Z matrix</h2><span id='topic+get_cond_A'></span>

<h3>Description</h3>

<p>The get_cond_A function is to get the conditional A matrix by using marginal A matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cond_A(marginal_A, Gl, N.Gx, ridgeTerm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cond_A_+3A_marginal_a">marginal_A</code></td>
<td>
<p>the marginal effects of SNPs on the exposures (Gx).</p>
</td></tr>
<tr><td><code id="get_cond_A_+3A_gl">Gl</code></td>
<td>
<p>the reference panel (Gl), such as 1000 Genome</p>
</td></tr>
<tr><td><code id="get_cond_A_+3A_n.gx">N.Gx</code></td>
<td>
<p>the sample size of each Gx. It can be a scalar or a vector. If there are multiple X's from different Gx, it should be a vector including the sample size of each Gx. If all alphas are from the same Gx, it could be a scalar.</p>
</td></tr>
<tr><td><code id="get_cond_A_+3A_ridgeterm">ridgeTerm</code></td>
<td>
<p>ridgeTerm = TRUE when the matrix L is singular. Matrix L is obtained from the cholesky decomposition of G0'G0. Default as FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with conditional estimates which are converted from marginal estimates using the JAM model.
</p>


<h3>Author(s)</h3>

<p>Lai Jiang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Gl)
data(betas.Gy)
data(marginal_A)
get_cond_A(marginal_A = marginal_A, Gl = Gl, N.Gx = c(339224, 659316), ridgeTerm = TRUE)
</code></pre>

<hr>
<h2 id='get_cond_alpha'>Compute conditional alphas</h2><span id='topic+get_cond_alpha'></span>

<h3>Description</h3>

<p>The get_cond_alpha function is to compute the conditional alpha vector for each X
If only one X in the model, please use get_cond_alpha instead of get_cond_A
A sub-step in the get_cond_A function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cond_alpha(alphas, Gl, N.Gx, ridgeTerm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cond_alpha_+3A_alphas">alphas</code></td>
<td>
<p>the marginal effects of SNPs on one exposure (Gx).</p>
</td></tr>
<tr><td><code id="get_cond_alpha_+3A_gl">Gl</code></td>
<td>
<p>the reference panel (Gl), such as 1000 Genome</p>
</td></tr>
<tr><td><code id="get_cond_alpha_+3A_n.gx">N.Gx</code></td>
<td>
<p>the sample size of the Gx. It can be a scalar.</p>
</td></tr>
<tr><td><code id="get_cond_alpha_+3A_ridgeterm">ridgeTerm</code></td>
<td>
<p>ridgeTerm = TRUE when the matrix L is singular. Matrix L is obtained from the cholesky decomposition of G0'G0. Default as FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with conditional estimates which are converted from marginal estimates using the JAM model.
</p>


<h3>Author(s)</h3>

<p>Lai Jiang
</p>


<h3>References</h3>

<p>Lai Jiang, Shujing Xu, Nicholas Mancuso, Paul J. Newcombe, David V. Conti (2020).
A Hierarchical Approach Using Marginal Summary Statistics for Multiple Intermediates
in a Mendelian Randomization or Transcriptome Analysis. <em>bioRxiv</em>
<a href="https://doi.org/10.1101/2020.02.03.924241">https://doi.org/10.1101/2020.02.03.924241</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Gl)
data(betas.Gy)
data(marginal_A)
get_cond_alpha(alphas = marginal_A[, 1], Gl = Gl, N.Gx = 339224, ridgeTerm = TRUE)
</code></pre>

<hr>
<h2 id='Gl'>Example reference data of hJAM</h2><span id='topic+Gl'></span>

<h3>Description</h3>

<p>The real data example from hJAM paper
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Gl
</code></pre>


<h3>Format</h3>

<p>The <code>Gl</code> object is a data matrix with 2467 individual of 210 SNPs from 1000 Genome project.</p>


<h3>References</h3>

<p>Consortium GP. A global reference for human genetic variation. Nature 2015; 526: 68.
</p>

<hr>
<h2 id='hJAM_egger'>Fit hJAM with Egger regression</h2><span id='topic+hJAM_egger'></span>

<h3>Description</h3>

<p>The hJAM_egger function is to get the results from the hJAM model with Egger regression. It is for detecting potential pleiotropy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hJAM_egger(betas.Gy, N.Gy, Gl, A, ridgeTerm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hJAM_egger_+3A_betas.gy">betas.Gy</code></td>
<td>
<p>The betas in the paper: the marginal effects of SNPs on the phenotype (Gy)</p>
</td></tr>
<tr><td><code id="hJAM_egger_+3A_n.gy">N.Gy</code></td>
<td>
<p>The sample size of Gy</p>
</td></tr>
<tr><td><code id="hJAM_egger_+3A_gl">Gl</code></td>
<td>
<p>The reference panel (Gl), such as 1000 Genome</p>
</td></tr>
<tr><td><code id="hJAM_egger_+3A_a">A</code></td>
<td>
<p>The A matrix in the paper: the marginal/conditional effects of SNPs on the exposures (Gx)</p>
</td></tr>
<tr><td><code id="hJAM_egger_+3A_ridgeterm">ridgeTerm</code></td>
<td>
<p>ridgeTerm = TRUE when the matrix L is singular. Matrix L is obtained from the cholesky decomposition of G0'G0. Default as FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the hJAM with egger regression results.
</p>

<dl>
<dt>Exposure</dt><dd><p>The intermediates, such as the modifiable risk factors in Mendelian Randomization and
gene expression in transcriptome analysis.</p>
</dd>
<dt>numSNP</dt><dd><p>The number of SNPs that the user use in the instrument set.</p>
</dd>
<dt>Estimate</dt><dd><p>The conditional estimates of the associations between intermediates and the outcome.</p>
</dd>
<dt>StdErr</dt><dd><p>The standard error of the conditional estimates of the associations between intermediates
and the outcome.</p>
</dd>
<dt>Lower.CI</dt><dd><p>The lower bound of the 95% confidence interval of the estimates.</p>
</dd>
<dt>Upper.CI</dt><dd><p>The upper bound of the 95% confidence interval of the estimates.</p>
</dd>
<dt>Pvalue</dt><dd><p>The p value of the estimates with a type-I error equals 0.05.</p>
</dd>
<dt>Est.Int</dt><dd><p>The intercept of the regression of intermediates on the outcome.</p>
</dd>
<dt>StdErr.Int</dt><dd><p>The standard error of the intercept of the regression of intermediates
on the outcome.</p>
</dd>
<dt>Lower.CI.Int</dt><dd><p>The lower bound of the 95% confidence interval of the intercept.</p>
</dd>
<dt>Upper.CI.Int</dt><dd><p>The upper bound of the 95% confidence interval of the intercept.</p>
</dd>
<dt>Pvalue.Int</dt><dd><p>The p value of the intercept with a type-I error equals 0.05.</p>
</dd>
</dl>

<p>An object of hJAM with egger regression results.
</p>


<h3>Author(s)</h3>

<p>Lai Jiang
</p>


<h3>References</h3>

<p>Lai Jiang, Shujing Xu, Nicholas Mancuso, Paul J. Newcombe, David V. Conti (2020).
A Hierarchical Approach Using Marginal Summary Statistics for Multiple Intermediates
in a Mendelian Randomization or Transcriptome Analysis. <em>bioRxiv</em>
<a href="https://doi.org/10.1101/2020.02.03.924241">https://doi.org/10.1101/2020.02.03.924241</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Gl)
data(betas.Gy)
data(conditional_A)
hJAM_egger(betas.Gy = betas.Gy, Gl = Gl, N.Gy = 459324, A = conditional_A, ridgeTerm = TRUE)
</code></pre>

<hr>
<h2 id='hJAM_lnreg'>Fit hJAM with linear regression</h2><span id='topic+hJAM_lnreg'></span>

<h3>Description</h3>

<p>The hJAM function is to get the results from the hJAM model using input data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hJAM_lnreg(betas.Gy, N.Gy, Gl, A, ridgeTerm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hJAM_lnreg_+3A_betas.gy">betas.Gy</code></td>
<td>
<p>The betas in the paper: the marginal effects of SNPs on the phenotype (Gy)</p>
</td></tr>
<tr><td><code id="hJAM_lnreg_+3A_n.gy">N.Gy</code></td>
<td>
<p>The sample size of Gy</p>
</td></tr>
<tr><td><code id="hJAM_lnreg_+3A_gl">Gl</code></td>
<td>
<p>The reference panel (Gl), such as 1000 Genome</p>
</td></tr>
<tr><td><code id="hJAM_lnreg_+3A_a">A</code></td>
<td>
<p>The A matrix in the paper: the marginal/conditional effects of SNPs on the exposures (Gx)</p>
</td></tr>
<tr><td><code id="hJAM_lnreg_+3A_ridgeterm">ridgeTerm</code></td>
<td>
<p>ridgeTerm = TRUE when the matrix L is singular. Matrix L is obtained from the cholesky
decomposition of G0'G0. Default as FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the hJAM with linear regression results.
</p>

<dl>
<dt>Exposure</dt><dd><p>The intermediates, such as the modifiable risk factors in Mendelian Randomization and gene expression in transcriptome analysis.</p>
</dd>
<dt>numSNP</dt><dd><p>The number of SNPs that the user use in the instrument set.</p>
</dd>
<dt>Estimate</dt><dd><p>The conditional estimates of the associations between intermediates and the outcome.</p>
</dd>
<dt>StdErr</dt><dd><p>The standard error of the conditional estimates of the associations between intermediates and the outcome.</p>
</dd>
<dt>Lower.CI</dt><dd><p>The lower bound of the 95% confidence interval of the estimates.</p>
</dd>
<dt>Upper.CI</dt><dd><p>The upper bound of the 95% confidence interval of the estimates.</p>
</dd>
<dt>Pvalue</dt><dd><p>The p value of the estimates with a type-I error equals 0.05.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Lai Jiang
</p>


<h3>References</h3>

<p>Lai Jiang, Shujing Xu, Nicholas Mancuso, Paul J. Newcombe, David V. Conti (2020).
A Hierarchical Approach Using Marginal Summary Statistics for Multiple Intermediates
in a Mendelian Randomization or Transcriptome Analysis. <em>bioRxiv</em>
<a href="https://doi.org/10.1101/2020.02.03.924241">https://doi.org/10.1101/2020.02.03.924241</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Gl)
data(betas.Gy)
data(conditional_A)
hJAM_lnreg(betas.Gy = betas.Gy, Gl = Gl, N.Gy = 459324, A = conditional_A, ridgeTerm = TRUE)

</code></pre>

<hr>
<h2 id='marginal_A'>Example marginal A matrix of hJAM</h2><span id='topic+marginal_A'></span>

<h3>Description</h3>

<p>Example marginal A matrix of hJAM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>marginal_A
</code></pre>


<h3>Format</h3>

<p>The <code>marginal_A</code> is the marginal estimates alpha matrix in the hJAM model: the association estimates between 210 SNPs and body mass index (BMI) and type 2 diabetes (T2D). The summary data was collected from GIANT consortium (n=339,224) and DIAGRAM+GERA+UKB (n=659316) for BMI and T2D, respectively.</p>


<h3>References</h3>

<p>1. Locke AE, Kahali B, Berndt SI, et al. Genetic studies of body mass index yield new insights for obesity biology. Nature 2015; 518: 197-206. 2. Xue A, Wu Y, Zhu Z, et al. Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes. Nat Commun 2018; 9: 2941.
</p>

<hr>
<h2 id='output.format'>Keep the output as three digits</h2><span id='topic+output.format'></span>

<h3>Description</h3>

<p>Keep the output as three digits
</p>


<h3>Usage</h3>

<pre><code class='language-R'>output.format(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="output.format_+3A_x">x</code></td>
<td>
<p>input</p>
</td></tr>
<tr><td><code id="output.format_+3A_...">...</code></td>
<td>
<p>other options you want to put in</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lai Jiang
</p>

<hr>
<h2 id='print.hJAM_egger'>Print out for hJAM_egger</h2><span id='topic+print.hJAM_egger'></span>

<h3>Description</h3>

<p>Print out for hJAM_egger
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'hJAM_egger'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.hJAM_egger_+3A_x">x</code></td>
<td>
<p>input</p>
</td></tr>
<tr><td><code id="print.hJAM_egger_+3A_...">...</code></td>
<td>
<p>other options you want to put in</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lai Jiang
</p>

<hr>
<h2 id='print.hJAM_lnreg'>Print out for hJAM_lnreg</h2><span id='topic+print.hJAM_lnreg'></span>

<h3>Description</h3>

<p>Print out for hJAM_lnreg
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'hJAM_lnreg'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.hJAM_lnreg_+3A_x">x</code></td>
<td>
<p>input</p>
</td></tr>
<tr><td><code id="print.hJAM_lnreg_+3A_...">...</code></td>
<td>
<p>other options you want to put in</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lai Jiang
</p>

<hr>
<h2 id='SNPs_heatmap'>Heatmap for all the SNPs used in the analysis</h2><span id='topic+SNPs_heatmap'></span>

<h3>Description</h3>

<p>To generate the heatmap of all the SNPs that the user use in the analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNPs_heatmap(Gl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNPs_heatmap_+3A_gl">Gl</code></td>
<td>
<p>The reference panel (Gl) of the SNPs that the user use in the analysis, such as 1000 Genome</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lai Jiang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Gl)
t = SNPs_heatmap(Gl = Gl)
t
</code></pre>

<hr>
<h2 id='SNPs_info'>Example SNPs' information of hJAM</h2><span id='topic+SNPs_info'></span>

<h3>Description</h3>

<p>Example SNPs' information of hJAM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNPs_info
</code></pre>


<h3>Format</h3>

<p>The <code>SNPs_info</code> is the information of the 210 SNPs that we used in this data example. It includes three columns: the rsID, major allele, and minor allele frequency of each SNP. The minor allele frequencies were calculated in the 503 European-ancestry subjects in 1000 Genome project.</p>


<h3>References</h3>

<p>Consortium GP. A global reference for human genetic variation. Nature 2015; 526: 68.
</p>

<hr>
<h2 id='SNPs_scatter_plot'>Scatter plot for all the SNPs used in the analysis</h2><span id='topic+SNPs_scatter_plot'></span>

<h3>Description</h3>

<p>To generate the scatter plot of all the SNPs that the user use in the analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNPs_scatter_plot(A, betas.Gy, num_X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNPs_scatter_plot_+3A_a">A</code></td>
<td>
<p>The effects of SNPs on the exposures (Gx).</p>
</td></tr>
<tr><td><code id="SNPs_scatter_plot_+3A_betas.gy">betas.Gy</code></td>
<td>
<p>The betas in the paper: the marginal effects of SNPs on the phenotype (Gy)</p>
</td></tr>
<tr><td><code id="SNPs_scatter_plot_+3A_num_x">num_X</code></td>
<td>
<p>The number of intermediates in the research question.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A set of scatter plots with x-axis being the conditional <code class="reqn">\alpha</code> estimates for each
intermediate and y-axis being the <code class="reqn">\beta</code> estimates.
</p>


<h3>Author(s)</h3>

<p>Lai Jiang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(conditional_A)
data(betas.Gy)
t = SNPs_scatter_plot(A = conditional_A, betas.Gy = betas.Gy, num_X = 2)
t
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
