<!DOCTYPE html><html lang="en"><head><title>Help for package HLAtools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HLAtools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#addCodonLine'><p>Add an 'AA codon' Line to Alignments When Missing.</p></a></li>
<li><a href='#alignmentFull'><p>Build Sets of Protein, Codon, Coding Nucleotide and Genomic Nucleotide Alignments for Specified  Loci</p></a></li>
<li><a href='#alignmentSearch'><p>Search Alignments for Specific Positions in a Specific Allele</p></a></li>
<li><a href='#alleleListHistory'><p>Allele Names Across All Release Versions</p></a></li>
<li><a href='#alleleTrim'><p>Trim All Versions of HLA Allele Names</p></a></li>
<li><a href='#atlasFull'><p>Generate a Complete set of Protein, Coding nucleotide and Genomic Nucleotide Atlases</p></a></li>
<li><a href='#atlasMaker'><p>Identify the Gene-Feature Boundaries in HLA Region Genes</p></a></li>
<li><a href='#BDstrat'><p>Stratify BIGDAWG Datasets by Specific Alleles</p></a></li>
<li><a href='#BDtoPyPop'><p>Convert BIGDAWG datasets to PyPop datasets</p>
</p>
<p>Converts a BIGDAWG-formatted data frame into a pair of PyPop-formatted case and control data frames.</p></a></li>
<li><a href='#buildAlignments'><p>Build Amino Acid, cDNA and gDNA Alignments</p></a></li>
<li><a href='#buildGazeteer'><p>Define Categories of Genes Supported by the IPD-IMGT/HLA Database</p></a></li>
<li><a href='#buildIMGTHLAGeneTypes'><p>Describe IPD-IMGT/HLA Database Genes, Identifying Pseudogenes and Gene Fragments</p></a></li>
<li><a href='#checkAlignType'><p>Ensure that AlignType Values are Valid</p></a></li>
<li><a href='#checkgDNAstart'><p>Identify gDNA Alignments in Which the First Feature Boundary is not Identified as Position +1.</p></a></li>
<li><a href='#checkSource'><p>Ensure that Source Values are Valid</p></a></li>
<li><a href='#checkVersion'><p>Check IPD-IMGT/HLA Release Version Allele Names</p></a></li>
<li><a href='#compareSequences'><p>Identify Sequence Differences Between Two Alleles at a Locus</p></a></li>
<li><a href='#convertAny'><p>Convert Values Across an Entire Data Frame or Vector</p></a></li>
<li><a href='#countSpaces'><p>Count the Spaces in a Character String</p></a></li>
<li><a href='#customAlign'><p>Generate a Customized Peptide, Codon or Nucleotide Sequence Alignment.</p></a></li>
<li><a href='#expandVersion'><p>Add 'Dot' Delimiters to a Numeric Release Version</p></a></li>
<li><a href='#ffN'><p>Identify and Annotate Gene Features in Pseudogenes and Gene Fragments.</p></a></li>
<li><a href='#formatHead'><p>Format PyPop Data Frame Headers</p></a></li>
<li><a href='#fragmentFeatureNames'><p>Gene Features of HLA Pseudogenes and Gene Fragments</p></a></li>
<li><a href='#getAlignmentNames'><p>Retrieve Alignment Filenames for HLA Genes</p></a></li>
<li><a href='#getField'><p>Trim Colon-Delimited HLA Allele Names by Field</p></a></li>
<li><a href='#getLatestVersion'><p>Identify the Latest IPD-IMGT/HLA Database Release</p></a></li>
<li><a href='#GIANT'><p>GL string code-Integrated Allele Name Translation</p></a></li>
<li><a href='#GLSC.ex'><p>Example Data Frame of Genotype List String Code Data</p></a></li>
<li><a href='#GLStoUNI'><p>Translate GL String to UNIFORMAT</p></a></li>
<li><a href='#GLstring.ex'><p>Example Data Frame of Genotype List String Data.</p></a></li>
<li><a href='#GLtoUN'><p>Translate GL Strings to UNIFORMAT Strings</p></a></li>
<li><a href='#GLupdate'><p>Update a GL String Code to a Specified IPD-IMGT/HLA Database Version</p></a></li>
<li><a href='#GLV'><p>Retrieve Version from Input GL String</p></a></li>
<li><a href='#GLV2'><p>Format GL String Code Version Number</p></a></li>
<li><a href='#GLvalidate'><p>Validate a GL String Code</p></a></li>
<li><a href='#GLVhelper'><p>Locate Matches for an Incomplete IPD-IMGT/HLA Database Version</p></a></li>
<li><a href='#HLAatlas'><p>Boundary Positions of Exons, Introns and UTRs in Amino Acid, cDNA and gDNA Alignments</p></a></li>
<li><a href='#HLAgazeteer'><p>Functional and Organizational Categories of Genes Supported by the IPD-IMGT/HLA Database</p></a></li>
<li><a href='#IMGTHLAGeneTypes'><p>Molecular characteristics of the Genes Curated by the IPD-IMGT/HLA Database</p></a></li>
<li><a href='#motifMatch'><p>Identify Alleles that Share a Sequence Motif</p></a></li>
<li><a href='#multiAlign'><p>Generate an Alignment for Specific Alleles at Different Positions</p></a></li>
<li><a href='#multiAlleleTrim'><p>Trim Multiple HLA Allele Names</p></a></li>
<li><a href='#multiGLStoUNI'><p>Translate Multiple GL Strings to UNIFORMAT</p></a></li>
<li><a href='#multiLocusValidation'><p>Apply validateLocus() to Multiple Loci</p></a></li>
<li><a href='#multiQueryRelease'><p>Search Multiple Elements of Allele Names Across Release Versions</p></a></li>
<li><a href='#multiSearch'><p>Search Alignment Sequences at Multiple Positions for a Specified Allele</p></a></li>
<li><a href='#multiTranslateGLstring'><p>Translate a Vector of GL Strings to a Desired IPD-IMGT/HLA Database Version</p></a></li>
<li><a href='#multiUNItoGLS'><p>Translate Multiple UNIFORMAT Strings to GL Strings</p></a></li>
<li><a href='#multiUpdateGL'><p>Update a Vector of GL String Code Data to a Desired IPD-IMGT/HLA Database Version</p></a></li>
<li><a href='#numFields'><p>Identify the Number of Fields in a Colon-Delimited Allele Name</p></a></li>
<li><a href='#parseAlignmentHead'><p>Guides for Parsing the Header Blocks of Alignment Files</p></a></li>
<li><a href='#posSort'><p>Numerical Sort of Alignment Positions that Contain Indels</p></a></li>
<li><a href='#pypopHeaders'><p>Convert BIGDAWG File Headers to PyPop Format</p></a></li>
<li><a href='#queryRelease'><p>Search Allele Names Across Release Versions</p></a></li>
<li><a href='#redec'><p>Reintroduce Version Decimals</p></a></li>
<li><a href='#relRisk'><p>Calculate Relative Risk for Individual Alleles and Genotypes in BIGDAWG-formatted Non-Case-Control Datasets</p></a></li>
<li><a href='#repoVersion'><p>Convert an AlleleListHistory Release Version to the GitHub Repository Version</p></a></li>
<li><a href='#sHLAdata'><p>Synthetic HLA Data for use with Package Examples</p></a></li>
<li><a href='#squashVersion'><p>Reduce a Release Version to Numerals</p></a></li>
<li><a href='#translateAllele'><p>Translate HLA Allele Names across IPD-IMGT/HLA Database Release Versions</p></a></li>
<li><a href='#translateGLstring'><p>Translate a GL String across IPD-IMGT/HLA Database Release Versions</p></a></li>
<li><a href='#typeToSource'><p>Convert AlignType Values to Source Values</p></a></li>
<li><a href='#uniAlign'><p>Generate an Alignment for Specific Alleles at Specific Positions</p></a></li>
<li><a href='#UNIFORMAT.example'><p>Example Data Frame of UNIFORMAT Data.</p></a></li>
<li><a href='#uniSearch'><p>Search Sequences at a Single Position for an Allele</p></a></li>
<li><a href='#UNItoGLS'><p>Translate UNIFORMAT to GL String</p></a></li>
<li><a href='#UNtoGL'><p>Translate UNIFORMAT Strings to GL Strings</p></a></li>
<li><a href='#updateAll'><p>Update All Package Data Objects Derived from IPD-IMGT/HLA Database Resources</p></a></li>
<li><a href='#updateAlleleListHistory'><p>Build the AlleleList History R Object</p></a></li>
<li><a href='#updateGL'><p>Update a GL String Code to a Specified IPD-IMGT/HLA Database Version</p></a></li>
<li><a href='#validateAllele'><p>Validate Allele-Name Format and Presence in HLAalignments</p></a></li>
<li><a href='#validateGLstring'><p>Validate a GL String</p></a></li>
<li><a href='#validateLocus'><p>Determine if a Locus Name is in the HLAgazeteer</p></a></li>
<li><a href='#validateMotif'><p>Determine if a Motif is Properly Formatted</p></a></li>
<li><a href='#validateUniformat'><p>Validate a UNIFORMAT String</p></a></li>
<li><a href='#validateVersion'><p>Validate an IPD-IMGT/HLA Release Version</p></a></li>
<li><a href='#verifyAllele'><p>Determine if an Allele Name Ever Existed, and (if so) in Which IPD-IMGT/HLA Database Releases</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Toolkit for HLA Immunogenomics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Steven Mack &lt;Steven.Mack@ucsf.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A toolkit for the analysis and management of data for genes in the so-called "Human Leukocyte Antigen" (HLA) region. Functions extract reference data from the Anthony Nolan HLA Informatics Group/ImmunoGeneTics HLA 'GitHub' repository (ANHIG/IMGTHLA) <a href="https://github.com/ANHIG/IMGTHLA">https://github.com/ANHIG/IMGTHLA</a>, validate Genotype List (GL) Strings, convert between UNIFORMAT and GL String Code (GLSC) formats, translate HLA alleles and GLSCs across ImmunoPolymorphism Database (IPD) IMGT/HLA Database release versions, identify differences between pairs of alleles at a locus, generate customized, multi-position sequence alignments, trim and convert allele-names across nomenclature epochs, and extend existing data-analysis methods.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>DescTools, dplyr, stringr, tibble, utils, xfun, fmsb</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td>&lt;<a href="https://github.com/sjmack/HLAtools&amp;gt;">https://github.com/sjmack/HLAtools&gt;</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-03 21:39:06 UTC; sjmack</td>
</tr>
<tr>
<td>Author:</td>
<td>Livia Tran [aut],
  Ryan Nickens [aut],
  Leamon Crooms IV [aut],
  Derek Pappas [aut],
  Vinh Luu [ctb],
  Josh Bredeweg [ctb],
  Steven Mack <a href="https://orcid.org/0000-0001-9820-9547"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-03 22:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='addCodonLine'>Add an 'AA codon' Line to Alignments When Missing.</h2><span id='topic+addCodonLine'></span>

<h3>Description</h3>

<p>Modifies cDNA alignment objects that are missing &quot;AA codon&quot; lines to include these lines in the correct location with the correct codon position information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addCodonLine(cDNAalign, firstPos, afterLine = 8, codons = 25)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addCodonLine_+3A_cdnaalign">cDNAalign</code></td>
<td>
<p>A matrix of cDNA alignment lines, generated from an alignment file that is missing &quot;AA codon&quot; lines.</p>
</td></tr>
<tr><td><code id="addCodonLine_+3A_firstpos">firstPos</code></td>
<td>
<p>A numeric value identifying the position number of the transcript's N-terminal codon, based on a complete cDNA alignment in another release.</p>
</td></tr>
<tr><td><code id="addCodonLine_+3A_afterline">afterLine</code></td>
<td>
<p>A numeric value identifying the number of the line below the first &quot;cDNA&quot; line in the alignment. The default value is 8.</p>
</td></tr>
<tr><td><code id="addCodonLine_+3A_codons">codons</code></td>
<td>
<p>A numeric value identifying the number of codons in each line of the nucleotide alignment. The default value is 25.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A complete cDNA alignment data frame that includes &quot;AA codon&quot; rows.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>

<hr>
<h2 id='alignmentFull'>Build Sets of Protein, Codon, Coding Nucleotide and Genomic Nucleotide Alignments for Specified  Loci</h2><span id='topic+alignmentFull'></span>

<h3>Description</h3>

<p>Applies buildAlignments() to build a set of alignments for loci supported in the ANHIG/IMGTHLA GitHub repository.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alignmentFull(loci = "all", alignType = "all", version = HLAgazeteer$version)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="alignmentFull_+3A_loci">loci</code></td>
<td>
<p>A characer vector of the locus names for which alignments should be built. The default value (&quot;all&quot;) generates alignments for all loci.</p>
</td></tr>
<tr><td><code id="alignmentFull_+3A_aligntype">alignType</code></td>
<td>
<p>A character vector of alignment types. The allowed values are &quot;all&quot;, &quot;prot&quot;, &quot;codon&quot;, &quot;nuc&quot;, and &quot;gen&quot;, which specify either all available alignments for a given locus or the respective protein, codon, nucleotide and genomic alignments, as determined by the HLAgazeteer.</p>
</td></tr>
<tr><td><code id="alignmentFull_+3A_version">version</code></td>
<td>
<p>A character string describing the version of the ANHIG/IMGTHLA Github repository from which alignments are obtained. The default value, 'HLAgazeteer$version', generates alignments for the IPD-IMGT/HLA Database release under which the HLAgazeteer was built.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object containing data frames of protein (prot), codon (codon), coding nucleotide (nuc), or genomic nucleotide (gen) alignments, for specified genes in the specified IPD-IMGT/HLA Database release, along with the pertinent reference database release.
</p>


<h3>Note</h3>

<p>AlignmentFull() must be run to build the 'HLAalignments' object (HLAalignments &lt;- alignmentFull()) before other functions that use alignments can be used, and therefore requires internet access.
</p>
<p>Depending on local download speeds, building all available alignments for all loci can take several minutes.
</p>
<p>Prior to IPD-IMGT/HLA Database release version 3.24.0, the HLA-DP and HLA-DQ sequence alignment files in the IPD-IMGT/HLA GitHub Repository did not all include a numerical suffix in the gene name (e.g., the protein sequence alignment file for the DQA1 gene was named 'DQA_prot.txt') because alignment files for the DPA2, DPB2, DQA2 and DQB2 genes had not been made available. Building DPA1, DPB1, DQA1, and DQB1 sequence alignments from releases prior to 3.24.0 require using a gene name that does not include the numerical suffix.
</p>

<hr>
<h2 id='alignmentSearch'>Search Alignments for Specific Positions in a Specific Allele</h2><span id='topic+alignmentSearch'></span>

<h3>Description</h3>

<p>Searches ANHIG/IMGT-HLA alignments and returns protein, codons or nucleotide sequences for a submitted allele name and position(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alignmentSearch(alignType, allelename, positions, prefix = TRUE, sep = "~")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="alignmentSearch_+3A_aligntype">alignType</code></td>
<td>
<p>The type of alignment being searched. Allowed values are &quot;codon&quot;,&quot;gen&quot;, nuc&quot; and &quot;prot&quot;. Only one 'alignType' value is allowed.</p>
</td></tr>
<tr><td><code id="alignmentSearch_+3A_allelename">allelename</code></td>
<td>
<p>A full or 2-field HLA allele name, excluding the &quot;HLA-&quot; prefix.</p>
</td></tr>
<tr><td><code id="alignmentSearch_+3A_positions">positions</code></td>
<td>
<p>A vector of sequence positions (e.g., c(-17,1,22,130)); in nucleotide and genomic alignments, indel positions are named using decimals. So the first indel between positions 26 and 27 is named 26.1, and the second indel between 26 and 27 is 26.2, etc.</p>
</td></tr>
<tr><td><code id="alignmentSearch_+3A_prefix">prefix</code></td>
<td>
<p>A logical that indicates if the position number(s) should be included in the result.</p>
</td></tr>
<tr><td><code id="alignmentSearch_+3A_sep">sep</code></td>
<td>
<p>The value that will be used to separate the sequences for each position. No value can be specified (sep=&quot;&quot;). If prefix=FALSE and sep=&quot;&quot;, a sequence block is returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string containing the corresponding peptide, codon or nucleotide sequence for each position in 'positions' for 'allelename'. A sequence is not returned if 'allelename' is not not found in the pertinent alignment. A position will be empty if 'allelename' does not have a value at the specified position.
</p>


<h3>Note</h3>

<p>This function requires that the HLAalignments object has been populated with alignments via the alignmentFull() function.
</p>
<p>Indel positions must be text-formatted (e.g. &quot;607.12&quot;).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
alignmentSearch("gen","DRB1*15:07:01",11:22)
alignmentSearch("nuc","DRB1*15:07:01",11:22)
alignmentSearch("prot","DRB1*15:07:01",11:22)
alignmentSearch("codon","DRB1*15:07:01",11:22)
alignmentSearch("nuc","DRB1*11:250N",c(605,"607.1","607.12",608))

## End(Not run)

</code></pre>

<hr>
<h2 id='alleleListHistory'>Allele Names Across All Release Versions</h2><span id='topic+alleleListHistory'></span>

<h3>Description</h3>

<p>A large list object of two elements. The first element contains version information. The second element contains all releases of allele name lists in the IMGT/HLA database.
This object is built by the UpdateAlleleListHistory() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(alleleListHistory)
</code></pre>


<h3>Format</h3>

<p>A large list of 2 elements that identify the table of allele names for each IPD-IMGT/HLA reference Database release and the source release
</p>

<ul>
<li><p>(Version: the Date and IPD-IMGT HLA Release version under which the alleleListHistory object was made)
</p>
</li>
<li><p>(AlleleListHistory: a dataframe version of the alleleListHistory.txt file)
</p>
</li></ul>



<h3>Source</h3>

<p>https://raw.githubusercontent.com/ANHIG/IMGTHLA/Latest/Allelelist_history.txt
</p>

<hr>
<h2 id='alleleTrim'>Trim All Versions of HLA Allele Names</h2><span id='topic+alleleTrim'></span>

<h3>Description</h3>

<p>Trims an HLA allele name to a specified number of fields or number of digits, depending on the nomenclature epoch. Expression variant suffixes in full-length allele names can be appended to truncated allele names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alleleTrim(allele, resolution, version = 3, append = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="alleleTrim_+3A_allele">allele</code></td>
<td>
<p>A character string of an HLA allele name formatted as locus*allele_name, optionally including the &quot;HLA-&quot; prefix.</p>
</td></tr>
<tr><td><code id="alleleTrim_+3A_resolution">resolution</code></td>
<td>
<p>A number identifying the number of fields to trim the allele name to.</p>
</td></tr>
<tr><td><code id="alleleTrim_+3A_version">version</code></td>
<td>
<p>A number identifying the HLA nomenclature epoch under which the allele was named. Epoch 1 allele names are found in IPD-IMGT/HLA Database releases 1.0.0 to 1.16.0. Epoch 2 allele names are found in IPD-IMGT/HLA Database releases 2.0.0 to 2.28.0. Epoch 3 allele names are found in IPD_IMGT/HLA Database releases 3.0.0 and onward.</p>
</td></tr>
<tr><td><code id="alleleTrim_+3A_append">append</code></td>
<td>
<p>A logical. When append = TRUE, the expression variant suffix of a full-length allele name is appended to a truncated allele name. The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string of the trimmed allele name, shortened according to the input parameters.
</p>


<h3>Note</h3>

<p>Expression variant suffixes will not be removed from full-length allele names that have fewer than four fields or eight digits.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alleleTrim(allele = "A*03:01:01", resolution = 2)
alleleTrim(allele = "A*030101", resolution = 2,version = 2)
alleleTrim(allele = "A*0303N",resolution = 1,version = 1, append = TRUE)
alleleTrim("HLA-A*24020102L",resolution = 3,version = 2, append = TRUE)

</code></pre>

<hr>
<h2 id='atlasFull'>Generate a Complete set of Protein, Coding nucleotide and Genomic Nucleotide Atlases</h2><span id='topic+atlasFull'></span>

<h3>Description</h3>

<p>Applies atlasMaker() to build a set of 'atlases' identifying the gene-feature (exon, intron, and UTR) boundaries in the protein, coding nucleotide, and genomic nucleotide alignments of all HLA loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>atlasFull(version = "Latest")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="atlasFull_+3A_version">version</code></td>
<td>
<p>A character string identifying the version of the ANHIG/IMGTHLA Github repository used to generate atlases. The default value, &quot;Latest&quot;, generates an atlas for the most recent IPD-IMGT/HLA Database release.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object containing a list of data frames of protein (prot), coding nucleotide (nuc), and genomic nucleotide (gen) atlases for all genes in the IPD-IMGT/HLA Database release, along with a character vector identifying the pertinent IPD-IMGT/HLA Database release.
</p>


<h3>Note</h3>

<p>Data informing the atlases were downloaded from the ANHIG/IMGTHLA Github repository.
</p>
<p>For internal HLAtools use.
</p>

<hr>
<h2 id='atlasMaker'>Identify the Gene-Feature Boundaries in HLA Region Genes</h2><span id='topic+atlasMaker'></span>

<h3>Description</h3>

<p>The 'AA' atlas identifies the amino acid (AA) positions encoded by codons that span each exon (E) boundary. The 'cDNA' atlas describes the 3' nucleotide position in the cDNA sequence that follows each exon boundary, as well as the codon that spans each exon boundary. The 'gDNA' atlas describes first 3' nucleotide position following each UTR (U), exon or intron (I) boundary. Each feature is followed by its rank (e.g U3 is the 3' UTR). Non-standard gene-features (H, J, N, and S) are detailed in the documentation for the ffN() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>atlasMaker(loci, source, version = "Latest")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="atlasMaker_+3A_loci">loci</code></td>
<td>
<p>A character string identifying a specific HLA gene (ex. &quot;DRB1&quot;). More than one gene can be specified (e.g., c(&quot;DRB1&quot;,&quot;DQB1&quot;)).</p>
</td></tr>
<tr><td><code id="atlasMaker_+3A_source">source</code></td>
<td>
<p>A character string identifying the type of alignment used to build the atlas. The allowed options are &quot;AA&quot;, for protein alignments, &quot;cDNA&quot;, for nucleotide alignments, and &quot;gDNA&quot;, for genomic alignments. More than one alignment type can be specified (e.g., c(&quot;AA&quot;,&quot;gDNA&quot;)).</p>
</td></tr>
<tr><td><code id="atlasMaker_+3A_version">version</code></td>
<td>
<p>A character string identifying the release version of the IPD-IMGT/HLA Github repository for which alignments should be generated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Builds an 'atlas' of the gene-feature (exon, intron and UTR) boundaries for IPD-IMGT/HLA loci, using IPD-IMGT/HLA Database alignments.
</p>


<h3>Value</h3>

<p>A list object of 'atlas' dataframes and a 'release version' character string for each locus and alignment type.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>
<p>Nucleotide atlases for pseudogenes will include a 'codon' row populated with NA values.
</p>
<p>Some HLA-DQB1\*05 and \*06 alleles include a 5th exon that is not present in the DQB1\*05:01:01:01 reference allele. In this case, the Exon 4 to Exon 5 boundary is defined as 'Ins' (insertion). For all other alleles there is no E.4-5 boundary.
</p>
<p>Boundaries for non-standard hybrid (H), join (J), novel (N) and segment (S) features may be included in gene fragment and pseudogene atlases.
</p>

<hr>
<h2 id='BDstrat'>Stratify BIGDAWG Datasets by Specific Alleles</h2><span id='topic+BDstrat'></span>

<h3>Description</h3>

<p>Divides a BIGDAWG-formatted dataset into two subsets (strata) that do and do not include specified alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BDstrat(dataset, alleles, warnBelow = 21)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BDstrat_+3A_dataset">dataset</code></td>
<td>
<p>A BIGDAWG-formatted data frame or a path to a BIGDAWG-formatted, tab-delimited text file.</p>
</td></tr>
<tr><td><code id="BDstrat_+3A_alleles">alleles</code></td>
<td>
<p>A vector of allele names in the locus-asterisk-allele_name format (e.g., &quot;A*01:01:01:01&quot;).</p>
</td></tr>
<tr><td><code id="BDstrat_+3A_warnbelow">warnBelow</code></td>
<td>
<p>An integer that defines a low number of subjects in a stratum, generating a warning message. The default value is 21.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list-object of two BIGDAWG-formatted data frames titled dataset$<code style="white-space: pre;">&#8288;&lt;alleles&gt;-positive&#8288;</code> and dataset$<code style="white-space: pre;">&#8288;&lt;alleles&gt;-negative&#8288;</code>. The positive list element includes all subjects with the specified alleles, and the negative list element includes all subjects without those specified alleles.
</p>


<h3>References</h3>

<p><a href="https://CRAN.R-project.org/package=BIGDAWG/vignettes/BIGDAWG.html#input-data">BIGDAWG Data Format</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>HLA_data.multi.strat &lt;- BDstrat(sHLAdata,c("DRB1*16:02:01:01","DRB1*04:07:01:01","A*25:01:01:01"))
HLA_data.single.strat &lt;- BDstrat(sHLAdata,"DRB1*16:02:01:01")

</code></pre>

<hr>
<h2 id='BDtoPyPop'>Convert BIGDAWG datasets to PyPop datasets
Converts a BIGDAWG-formatted data frame into a pair of PyPop-formatted case and control data frames.</h2><span id='topic+BDtoPyPop'></span>

<h3>Description</h3>

<p>Convert BIGDAWG datasets to PyPop datasets
</p>
<p>Converts a BIGDAWG-formatted data frame into a pair of PyPop-formatted case and control data frames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BDtoPyPop(dataset, filename, save.file = TRUE, save.path = tempdir())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BDtoPyPop_+3A_dataset">dataset</code></td>
<td>
<p>A data frame containing a BIGDAWG-formatted case-control dataset</p>
</td></tr>
<tr><td><code id="BDtoPyPop_+3A_filename">filename</code></td>
<td>
<p>A character string identifying the desired path and name for the generated files or the elements of the returned list object.</p>
</td></tr>
<tr><td><code id="BDtoPyPop_+3A_save.file">save.file</code></td>
<td>
<p>A logical that determines if a pair of files should be written (TRUE) or if a list object should be returned. The default value is TRUE.</p>
</td></tr>
<tr><td><code id="BDtoPyPop_+3A_save.path">save.path</code></td>
<td>
<p>A character string identifying the path in which to write the pair of files when save.file is TRUE. The default value is tempdir().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>When save.file = TRUE, a pair of files named &quot;'filename'.positive.pop&quot; and &quot;'filename'.negative.pop&quot; are generated in the working directory. When save.file = FALSE, a list of two-elements, named &quot;'filename'.positive&quot; and &quot;'filename'.negative&quot;, is returned.
</p>


<h3>Note</h3>

<p>Files generated by BDtoPyPop are intended for analysis using Pypop version 1.#.#. PyPop is not an R package, and must be installed separately.
</p>


<h3>References</h3>

<p>Lancaster et al. Front Immunol. 2024 Apr 2;15:1378512. https://pubmed.ncbi.nlm.nih.gov/38629078/
</p>
<p>Pappas et al. Hum Immunol. 2016 Mar 77(3):283-287. https://pubmed.ncbi.nlm.nih.gov/26708359/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>HLAdata.PP &lt;- BDtoPyPop(sHLAdata,"BDHLA",FALSE)

</code></pre>

<hr>
<h2 id='buildAlignments'>Build Amino Acid, cDNA and gDNA Alignments</h2><span id='topic+buildAlignments'></span>

<h3>Description</h3>

<p>Returns a list of data frames of amino acid, codon, nucleotide and genomic alignments, with version information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildAlignments(loci, source, version = "Latest")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildAlignments_+3A_loci">loci</code></td>
<td>
<p>A character vector of HLA gene names (e.g., &quot;DRB1&quot;, c(&quot;A&quot;,&quot;C&quot;)).</p>
</td></tr>
<tr><td><code id="buildAlignments_+3A_source">source</code></td>
<td>
<p>A character vector of alignment types. The allowed values are &quot;AA&quot;, &quot;cDNA&quot;, and &quot;gDNA&quot;. If 'source' is &quot;cDNA&quot;, both codon and cDNA nucleotide alignments are generated. If source is 'AA' or 'gDNA', a single peptide or genomic nucleotide alignment is generated. Up to four alignments will be returned for a locus, as determined by its ability to be transcribed or translated.</p>
</td></tr>
<tr><td><code id="buildAlignments_+3A_version">version</code></td>
<td>
<p>A character string describing the desired release version (branch) of the ANHIG/IMGTHLA Github repository (e.g. '3.53.0'). The default value ('Latest') returns alignments for the most recent release.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with a data frame of all allele names (and trimmed, two-field allele names) and their corresponding sequences (Amino Acid, codon, cDNA, or gDNA) for a specific locus, as well as version details for the returned information. These alignments identify locations of feature boundaries in relation to amino acid, codon, cDNA, and gDNA sequences. When a three- or four-field allele name includes an expression variant suffix, that suffix is appended to the trimmed name.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>

<hr>
<h2 id='buildGazeteer'>Define Categories of Genes Supported by the IPD-IMGT/HLA Database</h2><span id='topic+buildGazeteer'></span>

<h3>Description</h3>

<p>Consumes information in the ANHIG/IMGTHLA GitHub repository and at hla.alleles.org/genes to define specific categories of genes supported by the IPD-IMGT/HLA Database, which are represented as nineteen elements of the HLAgazeteer object.
</p>
<p>Elements:
</p>

<ul>
<li><p> All genes with alignments ($align)
</p>
</li>
<li><p> Genes that that do and do not have amino acid ($prot/$noprot), nucleotide ($nuc/$nonuc), and genomic ($gen/$nogen) alignments
</p>
</li>
<li><p> HLA genes ($hla), pseudogenes ($pseudo), gene fragments ($frag)
</p>
</li>
<li><p> Genes that are expressed ($expressed) or not expressed ($notexpressed)
</p>
</li>
<li><p> Genes in the Class I region ($classireg), Class I HLA genes ($classIhla), Genes in the Class II region ($classiireg), and Class II HLA genes ($classiihla)
</p>
</li>
<li><p> Classical HLA genes ($classical) and non-classical exprssed HLA genes ($nonclassical)
</p>
</li>
<li><p> All genes presented in map order ($map)
</p>
</li></ul>

<p>The twentieth element ($version) identifies the IPD-IMGT/HLA Database version used to build the HLAgazeteer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildGazeteer(version = getLatestVersion())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildGazeteer_+3A_version">version</code></td>
<td>
<p>A string identifying of the desired IPD-IMGT/HLA Database release version to which the gazeteer should be updated. The default value is most recent IPD-IMGT/HLA Database release version.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object of vectors organizing the genes in the IPD-IMGT/HLA Database into specific categories.
</p>


<h3>Note</h3>

<p>The <em>$prot</em> and <em>$nuc</em> vectors include a 'DRB' &quot;gene&quot;. While 'DRB' is not a gene name, the DRB_prot.txt file includes combined alignments for the DRB1, DRB3, DRB4, and DRB5 genes, and the DRB_nuc.txt file includes combined alignments for the DRB1, DRB2, DRB3, DRB4, DRB5, DRB6, DRB7, DRB8, and DRB9 genes. 'DRB' is included in these vectors for the purpose of validation when these combined alignments are desired.
</p>
<p>For interal HLAtools use.
</p>
<p>These elements are constructed using data compiled from hla.alleles.org/genes/, github.com/ANHIG/IMGTHLA/tree/Latest/alignments, DOI:10.2741/a317, DOI:10.1111/tan.15180, and Human Genome Assembly GRCh38.p14 reference assembly NC_000006.12.
</p>
<p>Additional information about these genes can be found in HLAtools::IMGTHLAGeneTypes.
</p>
<p>The $map element does not distinguish the order of the DRB3, DRB4 and DRB5 genes (DRB3/4/5) or the DR6 and DR7 (DR6/7) genes, as the individual elements of these gene sets are found on different DRB haplotypes.
</p>
<p>This function requires internet access to function.
</p>


<h3>Source</h3>

<p>Andersson G. Evolution of the human HLA-DR region. Front. Biosci. 1998 Jul 27:3:d739-45.
</p>
<p>Alexandrov et al. HLA-OLI: A new MHC class I pseudogene and HLA-Y are located on a 60 kb indel in the human MHC between HLA-W and HLA-J. HLA 2023 Nov; 102(5):599-606.
</p>

<hr>
<h2 id='buildIMGTHLAGeneTypes'>Describe IPD-IMGT/HLA Database Genes, Identifying Pseudogenes and Gene Fragments</h2><span id='topic+buildIMGTHLAGeneTypes'></span>

<h3>Description</h3>

<p>buildIMGTHLAGeneTypes() scrapes 'hla.alleles.org/genes/index.html' and generates a data frame identifying each gene. It requires internet access to function. As such, this function always returns data for the current IPD-IMGT/HLA Database release.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildIMGTHLAGeneTypes()
</code></pre>


<h3>Value</h3>

<p>A list object of two elements &ndash; 'version' and 'GeneTypes'. The 'version' element identifies the date that the source table was generated. The 'GeneTypes' element is a data frame of three columns, identifying each gene supported by the IPD-IMGT/HLA Database, along with its molecular characteristics and its status as either a pseudogene or a gene fragment.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>

<hr>
<h2 id='checkAlignType'>Ensure that AlignType Values are Valid</h2><span id='topic+checkAlignType'></span>

<h3>Description</h3>

<p>Evaluates 'alignType' values to ensure that only &quot;prot&quot;, &quot;nuc&quot;, &quot;codon&quot; and &quot;gen&quot; are passed to downstream functions. If any other values are entered, a message describing excluded values is generated. If no valid 'alignType' values are are present, an error is generated, and any calling function is ended.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkAlignType(alignType)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkAlignType_+3A_aligntype">alignType</code></td>
<td>
<p>A vector of character values specifying sequence alignment types to be used for a function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector that includes only allowed 'alignType' values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>checkAlignType(c("nuc","prot","gDNA")) 

</code></pre>

<hr>
<h2 id='checkgDNAstart'>Identify gDNA Alignments in Which the First Feature Boundary is not Identified as Position +1.</h2><span id='topic+checkgDNAstart'></span>

<h3>Description</h3>

<p>Checks the position of the first feature boundary (the 5' UTR - Exon 1 boundary in expressed genes) in each gDNA alignment, and identifies those alignments in which that position is not 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkgDNAstart(verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkgDNAstart_+3A_verbose">verbose</code></td>
<td>
<p>A logical indicating if loci with first feature boundary positions that are not 1 should be reported in the console (verbose = TRUE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function reviews the gDNA atlases in the HLAatlas object and returns a data frame of the first feature boundary position for each locus. For expressed genes and some pseudogenes, this is the position of the start of Exon 1.
</p>


<h3>Value</h3>

<p>A one-row data frame with one column for each locus with a gDNA alignment.
</p>


<h3>Note</h3>

<p>This function requires that the HLAalignments object has been populated with alignments via the alignmentFull() function.
</p>
<p>For internal HLAtools use.
</p>

<hr>
<h2 id='checkSource'>Ensure that Source Values are Valid</h2><span id='topic+checkSource'></span>

<h3>Description</h3>

<p>Evaluates 'source' values to ensure that only &quot;AA&quot;, &quot;cDNA&quot;, and &quot;gDNA&quot; are passed to downstream functions. If any other values are entered, a message describing excluded values is generated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkSource(source)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkSource_+3A_source">source</code></td>
<td>
<p>A vector of character values specifying sequence alignment file sources to be used for a function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector that includes only allowed 'source' values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>checkSource(c("AA","cDNA","codon")) 

</code></pre>

<hr>
<h2 id='checkVersion'>Check IPD-IMGT/HLA Release Version Allele Names</h2><span id='topic+checkVersion'></span>

<h3>Description</h3>

<p>Determines if allele name data for a given IPD-IMGT/HLA Release Version is present in the local HLAtools package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkVersion(version)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkVersion_+3A_version">version</code></td>
<td>
<p>A character string identifying the release version (branch) of the ANHIG/IMGTHLA Github repository (e.g. '3.53.0'). The value 'Latest' refers to the most recent release.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical. TRUE indicates that data for 'version' is available to the local HLAtools package. FALSE means that data for 'version' is not available.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>checkVersion(version = "3.25.0")
checkVersion(version = "Latest")

</code></pre>

<hr>
<h2 id='compareSequences'>Identify Sequence Differences Between Two Alleles at a Locus</h2><span id='topic+compareSequences'></span>

<h3>Description</h3>

<p>Compares the sequences of two alleles at a locus, and identifies the differences between them at specific positions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareSequences(alignType, alleles)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compareSequences_+3A_aligntype">alignType</code></td>
<td>
<p>A character string identifying the type of alignment being searched. Allowed values are &quot;codon&quot;,&quot;gen&quot;, nuc&quot; and &quot;prot&quot;. Only one 'alignType' value is permitted.</p>
</td></tr>
<tr><td><code id="compareSequences_+3A_alleles">alleles</code></td>
<td>
<p>A character vector containing two full-length names for alleles at the same locus.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-row data frame identifying the positions and sequences at which the two alleles differ. E.g., compareSequences(alignType = &quot;gen&quot;, alleles = c(&quot;DPA1\*01:03:38:01&quot;,&quot;DPA1\*01:03:38:02&quot;). Positions for which the sequence of either allele is unknown are ignored.
</p>

<hr>
<h2 id='convertAny'>Convert Values Across an Entire Data Frame or Vector</h2><span id='topic+convertAny'></span>

<h3>Description</h3>

<p>Converts all instances of a value in a data frame or vector to a specified value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convertAny(dataset, change.from = NA, change.to = "****")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convertAny_+3A_dataset">dataset</code></td>
<td>
<p>A data frame or vector of values.</p>
</td></tr>
<tr><td><code id="convertAny_+3A_change.from">change.from</code></td>
<td>
<p>The value present in the data frame or vector that should be changed. The default value is NA.</p>
</td></tr>
<tr><td><code id="convertAny_+3A_change.to">change.to</code></td>
<td>
<p>The value to which 'change.from' should be changed to. The default value is &quot;****&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame or vector in which 'change.from' has been converted to 'change.to'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dataset &lt;- convertAny(sHLAdata)

</code></pre>

<hr>
<h2 id='countSpaces'>Count the Spaces in a Character String</h2><span id='topic+countSpaces'></span>

<h3>Description</h3>

<p>Counts the number of spaces in a character string.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countSpaces(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="countSpaces_+3A_x">x</code></td>
<td>
<p>A character string.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of numeric values identifying the number of instances of spaces (' ') in x, and the number of consecutive spaces in an instance.
</p>


<h3>Author(s)</h3>

<p>Josh Bredeweg, Reddit user jbraids1421
</p>


<h3>Source</h3>

<p>https://www.reddit.com/r/rstats/comments/2th8ic/function_to_count_the_number_of_white_spaces/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>countSpaces("abc def")
countSpaces("abc def  ghi   jkl mno")

</code></pre>

<hr>
<h2 id='customAlign'>Generate a Customized Peptide, Codon or Nucleotide Sequence Alignment.</h2><span id='topic+customAlign'></span>

<h3>Description</h3>

<p>Generates a peptide, codon, coding or genomic nucleotide alignment table for user-specified HLA alleles at user-specified positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>customAlign(alignType, alleles, positions)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="customAlign_+3A_aligntype">alignType</code></td>
<td>
<p>The type of alignment being searched. Allowed values are &quot;prot&quot;, codon&quot;, &quot;nuc&quot; and &quot;gen&quot;. Only one 'alignType' value is allowed.</p>
</td></tr>
<tr><td><code id="customAlign_+3A_alleles">alleles</code></td>
<td>
<p>A vector of un-prefixed HLA allele names.</p>
</td></tr>
<tr><td><code id="customAlign_+3A_positions">positions</code></td>
<td>
<p>Either a vector of variant positions, against which all loci will be aligned, or a list of vectors of nucleotide positions, exactly one vector for each allele, against which each corresponding allele will be aligned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of allele names and the corresponding nucleotide sequences for each desired nucleotide position. an error message is returned if input locus is not available in the ANHIG/IMGTHLA Github Repository.
</p>


<h3>Note</h3>

<p>This function requires that the HLAalignments object has been populated with alignments via the alignmentFull() function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
customAlign("codon",c("DRB1*01:01","DQB1*02:01","DPB1*01:01"),c(1,2,3,7,8,9,13,14,15))
customAlign("codon",c("DPB1*01:01:01:01","DQA1*01:01:01:01","DQB1*05:01:01:01"),list(19:35,1:4,6:9))
customAlign("prot",c("DRB1*08:07","DPB1*14:01"),list(c(12,27,30,39,57,84),c(15,16,39,60,67,81)))

## End(Not run)

</code></pre>

<hr>
<h2 id='expandVersion'>Add 'Dot' Delimiters to a Numeric Release Version</h2><span id='topic+expandVersion'></span>

<h3>Description</h3>

<p>Adds the '.' delimiters to a 'squashed' IPD-IMGT/HLA Database Release Version name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expandVersion(ver)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="expandVersion_+3A_ver">ver</code></td>
<td>
<p>A four character/numeral IPD-IMGT/HLA Database Release Version name (e.g., '3540' or 3450).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of the expanded IPD-IMGT/HLA Database Release Version name.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>
<p>This function assumes single-character-length first and third fields of IPD-IMGT/HLA Database Release Version names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>expandVersion(3450)

</code></pre>

<hr>
<h2 id='ffN'>Identify and Annotate Gene Features in Pseudogenes and Gene Fragments.</h2><span id='topic+ffN'></span>

<h3>Description</h3>

<p>HLA pseudogenes and gene fragments many not share all of the gene features of their expressed homologs. The ffN() function generates a list object identifying the features present in a given gene fragment or pseudogene with a genomic alignment file. This includes annotations of the differences between the pseudogenes and gene fragments, and their expressed homologs.
Pseudogenes and gene fragments are identified in the IMGTHLAGeneTypes data object.
</p>
<p>Standard Features
</p>

<ul>
<li><p> E - Exon, a peptide-encoding sequence
</p>
</li>
<li><p> I - Intron, an intervening sequence found between Exons
</p>
</li>
<li><p> U - UTR, an untranslated region of sequence preceding the first Exon or following the last Exon
</p>
</li></ul>

<p>Additional Features used in these annotations, based on boundaries indicated in the sequence alignment, are:
</p>

<ul>
<li><p> H - Hybrid, a sequence that includes at least one known feature sequence and one nucleotide sequence that does not correspond to a known feature
</p>
</li>
<li><p> J - Join, a sequence that includes two or more features that are separated by a feature boundary in the reference
</p>
</li>
<li><p> N - Novel, a novel sequence that does correspond to a known feature sequence
</p>
</li>
<li><p> S - Segment, a subset of a longer feature sequence
</p>
</li></ul>

<p>Reference Genes
</p>

<ul>
<li><p> HLA-C genomic sequence is used as the reference for class I pseudogenes and gene fragments.
</p>
</li>
<li><p> HLA-DPA1 and -DPB1 genomic sequences are used as the references for class II pseudogenes -DPA2 and -DPB2, respectively.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>ffN(version)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ffN_+3A_version">version</code></td>
<td>
<p>A character string identifying the pertinent IPD-IMGT/HLA Database release version (e.g., &quot;3.55.0&quot;) under which the returned object is generated. This parameter does not impact the generation of the feature names or annotations, and is only included to provide IPD-IMGT/HLA Database release version context.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object where each element is the name of a pseudogene or gene fragment. Each of these elements is a list of 'features' and 'annotation'. &quot;Features&quot; identifies the gene features for that gene, ordered from 5' to 3'. Because these genes are not-expressed, the three standard gene features (U, Untranslated Region (UTR); E, Exon; and I, Intron) may not always be present. In these cases, H, J, N and S features (defined below) are returned. Each feature identifier is followed by an identifying number (e.g. U.5 is the 5' UTR, and E.3 is Exon 3). &quot;Annotation&quot; provides the composition of the non-standard H, J, N and S features for each gene.
#'
</p>


<h3>Note</h3>

<p>Features and their annotations have been identified manually. Feature annotations will not change unless a new pseudogene or gene fragment is added in a future release, in which case new annotations will be generated.
</p>
<p>The H, J, N and S features are described for class I pseudogenes and gene fragment sequences, all of which are described relative to the HLA-C reference sequence. Feature length differences for DPA2 and DPB2, relative to the DPA1 and DPB1 references, are noted in annotations of standard feature abbreviations (E, I and U).
</p>
<p>No annotations are included for the DRB2, DRB6, DRB7, and DRB9 genes, as genomic alignments for these genes are not available.
</p>
<p>For internal HLAtools use.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fragmentFeatureNames &lt;- ffN("3.35.0")

</code></pre>

<hr>
<h2 id='formatHead'>Format PyPop Data Frame Headers</h2><span id='topic+formatHead'></span>

<h3>Description</h3>

<p>Format the header of a PyPop-formatted data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>formatHead(colHead)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="formatHead_+3A_colhead">colHead</code></td>
<td>
<p>A vector of column names. PyPop format requires that paired locus/gene names should end in '_1' and '_2', respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector in which the locus names are suffixed with '_1' and '_2'.
</p>


<h3>Note</h3>

<p>This function assumes that the first two elements the 'colHead' vector are not locus/gene names.
</p>


<h3>References</h3>

<p>Lancaster et al. Front Immunol. 2024 Apr 2;15:1378512. https://pubmed.ncbi.nlm.nih.gov/38629078/
</p>
<p>Pappas et al. Hum Immunol. 2016 Mar 77(3):283-287. https://pubmed.ncbi.nlm.nih.gov/26708359/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>formatHead(colHead = colnames(sHLAdata))

</code></pre>

<hr>
<h2 id='fragmentFeatureNames'>Gene Features of HLA Pseudogenes and Gene Fragments</h2><span id='topic+fragmentFeatureNames'></span>

<h3>Description</h3>

<p>A list object of 15 elements. Each of the 15 elements corresponds to one of the 15 HLA pseudogenes and gene fragments, and contains two items. The first item identifies the gene features for that locus. The second item contains an annotation detailing information regarding non-standard gene structures.
This object is built by the ffN() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fragmentFeatureNames)
</code></pre>


<h3>Format</h3>

<p>A list of 15 elements that identify the gene features for each locus wih a genomic alignment and annotate non-standard features
</p>


<h3>Source</h3>

<p>https://hla.alleles.org/genes/index.html
</p>
<p>https://github.com/ANHIG/IMGTHLA/tree/Latest/alignments
</p>

<hr>
<h2 id='getAlignmentNames'>Retrieve Alignment Filenames for HLA Genes</h2><span id='topic+getAlignmentNames'></span>

<h3>Description</h3>

<p>Retrieves the filenames of the protein, nucleotide and genomic alignments available in a specific branch of the IMGTHLA GitHub Repository
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAlignmentNames(URL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAlignmentNames_+3A_url">URL</code></td>
<td>
<p>A chararacter string containing a Uniform Resource Locator (URL) identifying of the desired IPD-IMGT/HLA Database release version from which the alignment filenames should be retrieved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of all of the filenames.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>

<hr>
<h2 id='getField'>Trim Colon-Delimited HLA Allele Names by Field</h2><span id='topic+getField'></span>

<h3>Description</h3>

<p>Trims a properly formatted colon-delimited HLA allele name to a desired number of fields.
</p>
<p>If an allele name with an expression-variant suffix is truncated, the suffix can be appended to the end of the truncated allele name. If a resolution value greater than the number of fields in the submitted field is specified, the original allele is returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getField(allele, res, append = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getField_+3A_allele">allele</code></td>
<td>
<p>A character string representing an HLA allele.</p>
</td></tr>
<tr><td><code id="getField_+3A_res">res</code></td>
<td>
<p>A numeric describing the resolution desired.</p>
</td></tr>
<tr><td><code id="getField_+3A_append">append</code></td>
<td>
<p>A logical. When append = TRUE, the expression variant suffix of a full-length allele name is appended to a truncated allele name. The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of the 'allele' character string that has been trimmed to 'res' resolution.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>getField("HLA-A*01:01:01:01", 3)
getField("DRB1*11:01:01:12N", 2,TRUE)

</code></pre>

<hr>
<h2 id='getLatestVersion'>Identify the Latest IPD-IMGT/HLA Database Release</h2><span id='topic+getLatestVersion'></span>

<h3>Description</h3>

<p>Identifies the most recent version of the IPD-IMGT/HLA Database available in the ANHIG/IMGTHLA Github repository.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getLatestVersion()
</code></pre>


<h3>Value</h3>

<p>A dot-delimited character string identifying the latest release version (branch) of the ANHIG/IMGTHLA Github repository.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>getLatestVersion()

</code></pre>

<hr>
<h2 id='GIANT'>GL string code-Integrated Allele Name Translation</h2><span id='topic+GIANT'></span>

<h3>Description</h3>

<p>Calls updateGL() to translate the HLA allele names in a vector or data frame between two IPD-IMGT/HLA Database release versions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GIANT(data, transFrom, transTo)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GIANT_+3A_data">data</code></td>
<td>
<p>A data frame or vector of HLA allele name character string data. Allele names in vectors must include locus prefixes. Data frames must include column headers identifying a single locus for the allele names in that column.</p>
</td></tr>
<tr><td><code id="GIANT_+3A_transfrom">transFrom</code></td>
<td>
<p>A dot-formatted character string identifying the IPD-IMGT/HLA Database version (e.g. 3.58.0) under which the HLA data were generated.</p>
</td></tr>
<tr><td><code id="GIANT_+3A_transto">transTo</code></td>
<td>
<p>A dot-formatted character string identifying the IPD-IMGT/HLA Database version (e.g. 2.25.0) to which which the HLA data are to be translated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame or vector of HLA allele name data in the 'transTo' release.
</p>


<h3>Note</h3>

<p>GIANT() will ignore the first two columns of BIGDAWG-formatted data frames, which include non-HLA data. All columns in non-BIGDAWG-formatted data frames are expected to contain HLA allele names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>GIANT(c("A*01:01:01:01","DOA*01:01:01:01"),"3.56.0","2.20.0")
updsHLAdata &lt;- GIANT(sHLAdata,"3.56.0","2.25.0")

</code></pre>

<hr>
<h2 id='GLSC.ex'>Example Data Frame of Genotype List String Code Data</h2><span id='topic+GLSC.ex'></span>

<h3>Description</h3>

<p>A two-column data frame (modified from pould::hla.hap.demo) including GL String data in GL String Code format.
Column one identifies each subject's status, while column two identifies each subject's HLA genotype in GL String Code format.
This data is provided for use in examples and demonstrations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GLSC.ex)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 419 rows and 2 columns.
</p>


<h3>Source</h3>

<p>pould::hla.hap.demo
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023 Oct;102(4):501-507 https://doi.org/10.1111/tan.15145
</p>

<hr>
<h2 id='GLStoUNI'>Translate GL String to UNIFORMAT</h2><span id='topic+GLStoUNI'></span>

<h3>Description</h3>

<p>A wrapper function for GLtoUN, which translates strings from GL String format to UNIFORMAT format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLStoUNI(GLstring, prefix = "HLA-", pre = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GLStoUNI_+3A_glstring">GLstring</code></td>
<td>
<p>A character string of HLA allele names and operators in the GL String format signifying their relation with one another.</p>
</td></tr>
<tr><td><code id="GLStoUNI_+3A_prefix">prefix</code></td>
<td>
<p>A character string of the pertinent gene-system prefix (default is &quot;HLA-&quot;).</p>
</td></tr>
<tr><td><code id="GLStoUNI_+3A_pre">pre</code></td>
<td>
<p>A logical that indicates whether returned allele names should contain 'prefix' (TRUE), or if 'prefix' should be excluded from returned names (FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of 'GLstring' converted to UNIFORMAT format, or FALSE if 'GLstring' is invalid.
</p>


<h3>Note</h3>

<p>This function does not function with the &quot;?&quot; operator, as the &quot;?&quot; operator has no cognate in UNIFORMAT.
</p>


<h3>References</h3>

<p>Nunes Tissue Antigens 2007;69(s1):203-205 https://doi.org/10.1111/j.1399-0039.2006.00808.x
</p>
<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
GLStoUNI("HLA-A*02:01/HLA-A*02:02+HLA-A*03:01/HLA-A*03:02")
GLStoUNI("A+B/C~D^G|E^W+X/Y^Z+J")

</code></pre>

<hr>
<h2 id='GLstring.ex'>Example Data Frame of Genotype List String Data.</h2><span id='topic+GLstring.ex'></span>

<h3>Description</h3>

<p>A two-column example data frame (from pould::hla.hap.demo) including GL String Data.
Column one identifies each subject's status, while column two identifies each subject's HLA genotype in GL String format.
This data is provided for use in examples and demonstrations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GLstring.ex)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 419 rows and 2 columns.
</p>


<h3>Source</h3>

<p>pould::hla.hap.demo
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>

<hr>
<h2 id='GLtoUN'>Translate GL Strings to UNIFORMAT Strings</h2><span id='topic+GLtoUN'></span>

<h3>Description</h3>

<p>A function that translates strings from GL String format to UNIFORMAT format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLtoUN(GLstring, prefix = "HLA-", pre = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GLtoUN_+3A_glstring">GLstring</code></td>
<td>
<p>A character string of HLA allele names and operators in the GL String format signifying their relation with one another.</p>
</td></tr>
<tr><td><code id="GLtoUN_+3A_prefix">prefix</code></td>
<td>
<p>A character string of the desired gene-system prefix (default is &quot;HLA-&quot;).</p>
</td></tr>
<tr><td><code id="GLtoUN_+3A_pre">pre</code></td>
<td>
<p>A logical that indicates whether returned allele names should contain 'prefix' (TRUE), or if 'prefix' should be excluded from returned names (FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of the input string converted to UNIFORMAT format.
</p>


<h3>Note</h3>

<p>For internal use only.
</p>
<p>This function does not function with the &quot;?&quot; operator, as the &quot;?&quot; operator has no cognate in UNIFORMAT.
</p>


<h3>References</h3>

<p>Nunes Tissue Antigens 2007;69(s1):203-205 https://doi.org/10.1111/j.1399-0039.2006.00808.x
</p>
<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>


<h3>Examples</h3>

<pre><code class='language-R'>GLtoUN("HLA-A*02:01/HLA-A*02:02+HLA-A*03:01/HLA-A*03:02")
GLtoUN("A+B/C~D^G|E^W+X/Y^Z+J")

</code></pre>

<hr>
<h2 id='GLupdate'>Update a GL String Code to a Specified IPD-IMGT/HLA Database Version</h2><span id='topic+GLupdate'></span>

<h3>Description</h3>

<p>Updates the elements of a GLString Code across IPD-IMGT/HLA Database release versions. Truncated allele names can be expanded to the list of all allele names that contain the truncated name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLupdate(GLSC, to, expand = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GLupdate_+3A_glsc">GLSC</code></td>
<td>
<p>A GL String Code containing HLA allele names.</p>
</td></tr>
<tr><td><code id="GLupdate_+3A_to">to</code></td>
<td>
<p>A character string identifying the IPD-IMGT/HLA Database release version to translate 'GLSC' to. Values can range from version 1.05.0 to the loaded version of the alleleListHistory object.</p>
</td></tr>
<tr><td><code id="GLupdate_+3A_expand">expand</code></td>
<td>
<p>A logical that indicates if the lowest-numbered truncated allele names that match truncated allele names in 'GLSC' should be returned (expand = FALSE), or if a slash-delimited string of all matching full-length allele names should be returned (expand = TRUE). The default value is FALSE.</p>
</td></tr>
<tr><td><code id="GLupdate_+3A_verbose">verbose</code></td>
<td>
<p>A logical that indicates if messages regarding the update process should be sent to the console (TRUE) or not (FALSE). The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A GL String Code updated to the 'to' release version.
</p>


<h3>Note</h3>

<p>This function is maintained for compatibility with previous releases, and will be removed in a future release. HLAtools users are advised to use updateGL().
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>
<p>Mack et al. HLA 2023;102(4):501-507 https://doi.org/10.1111/tan.15145
</p>


<h3>Examples</h3>

<pre><code class='language-R'>GLupdate(GLSC.ex$GL.String.Code[1],"2.15.0",FALSE,FALSE)
GLupdate(GLSC.ex$GL.String.Code[1],"2.15.0",TRUE,FALSE)

</code></pre>

<hr>
<h2 id='GLV'>Retrieve Version from Input GL String</h2><span id='topic+GLV'></span>

<h3>Description</h3>

<p>Extracts the version data from an input GL String Code, or provides appropriate options if version input is not present in the IPD-IMGT/HLA Database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLV(GLString)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GLV_+3A_glstring">GLString</code></td>
<td>
<p>A character string of HLA allele names and operators in GL String Code format, signifying their relation with one another and the pertinent IPD-IMGT/HLA Database release version.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns A character string of the version of the release as it appears in the pertinent alleleListHistory$AlleleListHistory-column-header.
</p>


<h3>Note</h3>

<p>For internal use only.
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>
<p>Mack et al. HLA 2023;102(4):501-507 https://doi.org/10.1111/tan.15145
</p>


<h3>Examples</h3>

<pre><code class='language-R'>GLV("hla#3.25.0#HLA-B15:35")

</code></pre>

<hr>
<h2 id='GLV2'>Format GL String Code Version Number</h2><span id='topic+GLV2'></span>

<h3>Description</h3>

<p>Returns a compressed IPD-IMGT/HLA Database release version or a vector of dot-delimited release version options if the specified version is not present in the alleleListHistory object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLV2(Version)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GLV2_+3A_version">Version</code></td>
<td>
<p>A character string of the desired IPD-IMGT/HLA Database version, going back to version 1.05.0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns either a single, compressed character string-formatted release version, or a character vector of potential release versions.
</p>


<h3>Note</h3>

<p>For internal use only.
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>
<p>Mack et al. HLA 2023;102(4):501-507 https://doi.org/10.1111/tan.15145
</p>


<h3>Examples</h3>

<pre><code class='language-R'>GLV2("3.34.0")
GLV2("3.0.0")

</code></pre>

<hr>
<h2 id='GLvalidate'>Validate a GL String Code</h2><span id='topic+GLvalidate'></span>

<h3>Description</h3>

<p>A lightweight validator that inspects a GL String Code for correct structure. If the namespace field contains a value other than value of 'namespace', the namespace is changed to 'hla'. The version and GL String fields are not evaluated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLvalidate(GLString, namespace = c("hla", "kir"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GLvalidate_+3A_glstring">GLString</code></td>
<td>
<p>A character string describing a string formatted to GL String Code specifications.</p>
</td></tr>
<tr><td><code id="GLvalidate_+3A_namespace">namespace</code></td>
<td>
<p>A character vector of allowed namespace strings. The default value is 'c(&quot;hla&quot;,&quot;kir&quot;)'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string describing either a well-formatted GL String Code or the value FALSE.
</p>


<h3>Note</h3>

<p>For internal use only.
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>
<p>Mack et al. HLA 2023;102(4):501-507 https://doi.org/10.1111/tan.15145
</p>


<h3>Examples</h3>

<pre><code class='language-R'>GLvalidate("ha#3.25.0#hla-B15:35") 

</code></pre>

<hr>
<h2 id='GLVhelper'>Locate Matches for an Incomplete IPD-IMGT/HLA Database Version</h2><span id='topic+GLVhelper'></span>

<h3>Description</h3>

<p>Uses the provided version information to locate and return possible matches for an incompletely defined IPD-IMGT/HLA Database release version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLVhelper(Version)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GLVhelper_+3A_version">Version</code></td>
<td>
<p>A character string of the desired version to which the alleles should be updated, going back to version 1.05.0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of character strings of possible matches to a given incomplete input.
</p>


<h3>Note</h3>

<p>For internal use only.
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>
<p>Mack et al. HLA 2023;102(4):501-507 https://doi.org/10.1111/tan.15145
</p>


<h3>Examples</h3>

<pre><code class='language-R'>GLVhelper("2.25")
GLVhelper("3.9.0")


</code></pre>

<hr>
<h2 id='HLAatlas'>Boundary Positions of Exons, Introns and UTRs in Amino Acid, cDNA and gDNA Alignments</h2><span id='topic+HLAatlas'></span>

<h3>Description</h3>

<p>A list object of sub-lists of R dataframes (atlases) for each locus with a protein (prot), cDNA (nuc), and gDNA (gen) alignment. Each atlas identifies the position of the exon (E), intron (I) or UTR (U) gene-feature boundaries in an alignment. Boundaries for non-standard hybrid (H), join (J), novel (N) and segment (S) features may be included in gene fragment and pseudogene atlases.
This object is built by the atlasFull() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HLAatlas)
</code></pre>


<h3>Format</h3>

<p>A list of 4 elements that include the gene features boundaries in each ANHIG/IMGTHLA sequence alignment.
</p>

<ul>
<li><p>(prot: peptide-alignment atlases)
</p>
</li>
<li><p>(nuc: cDNA- and codon-alignment atlases)
</p>
</li>
<li><p>(gen: gDNA-alignment atlases)
</p>
</li>
<li><p>(version: The IPD-IMGT/HLA Database release version under which these data were generated)
</p>
</li></ul>



<h3>Source</h3>

<p>https://github.com/ANHIG/IMGTHLA/tree/Latest/alignments
</p>

<hr>
<h2 id='HLAgazeteer'>Functional and Organizational Categories of Genes Supported by the IPD-IMGT/HLA Database</h2><span id='topic+HLAgazeteer'></span>

<h3>Description</h3>

<p>A list object of nineteen vectors that identify genes in the HLA region that share specific commonalities.
This object is built by the buildGazeteer() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HLAgazeteer)
</code></pre>


<h3>Format</h3>

<p>A large list of 19 vectors that define specific categories of genes supported by the IPD-IMGT/HLA Database
</p>

<p>.   </p>
<ul>
<li><p>(align: all genes with alignments in the IPD/IMGT-HLA GitHub Repository)
</p>
</li>
<li><p>(gen: genes with genomic alignments in the IPD/IMGT-HLA GitHub Repository)
</p>
</li>
<li><p>(nuc: genes with nucleotide alignments in the IPD/IMGT-HLA GitHub Repository)
</p>
</li>
<li><p>(prot: genes with protein alignments in the IPD/IMGT-HLA GitHub Repository)
</p>
</li>
<li><p>(nogen: genes with no genomic alignments)
</p>
</li>
<li><p>(nonuc: genes with no nucleotide alignments)
</p>
</li>
<li><p>(noprot: genes with no protein alignments)
</p>
</li>
<li><p>(pseudo: pseudogenes)
</p>
</li>
<li><p>(frag: gene fragments)
</p>
</li>
<li><p>(hla: HLA genes)
</p>
</li>
<li><p>(expressed: genes that are expressed)
</p>
</li>
<li><p>(notexpressed: genes that are not expressed)
</p>
</li>
<li><p>(classireg: genes found in the HLA class I region)
</p>
</li>
<li><p>(classihla: class I HLA genes)
</p>
</li>
<li><p>(classiireg: genes found in the HLA class II region)
</p>
</li>
<li><p>(classiihla: class II HLA genes)
</p>
</li>
<li><p>(classical: classical HLA genes)
</p>
</li>
<li><p>(nonclassical: non-classical HLA genes)
</p>
</li>
<li><p>(map: all genes organized by 5' to 3' map order on the genomic reference + strand)
</p>
</li>
<li><p>(version: IPD-IMGT/HLA Database version used to build the HLAgazeteer)
</p>
</li></ul>



<h3>Source</h3>

<p>https://hla.alleles.org/genes
</p>
<p>https://github.com/ANHIG/IMGTHLA/tree/Latest/alignments
</p>
<p>Andersson Front. Biosci. 1998, 3(4), 739–745. https://doi.org/10.2741/a317
</p>
<p>Alexandrov et al. HLA 2023, Vol.102(5), p.599-606. https://doi.org/10.1111/tan.15180
</p>
<p>https://www.ncbi.nlm.nih.gov/nucleotide/NC_000009.12
</p>

<hr>
<h2 id='IMGTHLAGeneTypes'>Molecular characteristics of the Genes Curated by the IPD-IMGT/HLA Database</h2><span id='topic+IMGTHLAGeneTypes'></span>

<h3>Description</h3>

<p>A data frame of three columns identifying each gene supported by the IPD-IMGT/HLA Database, its molecular characteristics, and its status as a gene fragment or pseudogene.
This object is built by the BuildIMGTHLAGeneTypes() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(IMGTHLAGeneTypes)
</code></pre>


<h3>Format</h3>

<p>A list object of two elements:
</p>

<ul>
<li><p>(GeneTypes: a data frame of three columns)
</p>
</li>
<li><p>(version: a character string identifying the date that the source file was written)
</p>
</li></ul>



<h3>Source</h3>

<p>https://hla.alleles.org/genes/index.html
</p>

<hr>
<h2 id='motifMatch'>Identify Alleles that Share a Sequence Motif</h2><span id='topic+motifMatch'></span>

<h3>Description</h3>

<p>Searches alignments for alleles that share a specific sequence motif.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>motifMatch(motif, alignType, full = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="motifMatch_+3A_motif">motif</code></td>
<td>
<p>A character string identifying a sequence variant motif in the following format: Locus*#$~#$~#$, where ## identifies a variant position, and $ identifies the sequence variant. Both nucleotide and peptide motifs can be provided, and any number of variants can be specified.</p>
</td></tr>
<tr><td><code id="motifMatch_+3A_aligntype">alignType</code></td>
<td>
<p>A character string identifying the type of alignment being searched. Allowed values are &quot;codon&quot;,&quot;gen&quot;, nuc&quot; and &quot;prot&quot;. Only one 'alignType' value is allowed.</p>
</td></tr>
<tr><td><code id="motifMatch_+3A_full">full</code></td>
<td>
<p>A logical value that specifies if full (TRUE) or truncated (FALSE) allele names should be returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of allele names that contain the motif, or NA when no alleles contain the motif, NULL when no alignment is available for specified locus, and FALSE when the locus or motif is invalid.
</p>


<h3>Note</h3>

<p>This function requires an HLAalignments object that has been populated with alignments via alignmentFull().
</p>

<hr>
<h2 id='multiAlign'>Generate an Alignment for Specific Alleles at Different Positions</h2><span id='topic+multiAlign'></span>

<h3>Description</h3>

<p>Generates a peptide, codon, coding nucleotide or genomic alignment for HLA alleles, allowing each allele to be aligned to a different set of positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiAlign(alignType, alleles, positions)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiAlign_+3A_aligntype">alignType</code></td>
<td>
<p>The type of alignment being searched. Allowed values are &quot;prot&quot;, codon&quot;, &quot;nuc&quot; and &quot;gen&quot;.  Only one 'alignType' value is allowed.</p>
</td></tr>
<tr><td><code id="multiAlign_+3A_alleles">alleles</code></td>
<td>
<p>A vector of un-prefixed HLA locus names.</p>
</td></tr>
<tr><td><code id="multiAlign_+3A_positions">positions</code></td>
<td>
<p>A list of vectors of nucleotide positions, exactly one vector for each allele, against which each corresponding allele will be aligned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of 'allele' and the corresponding nucleotide sequence for specified positions designated for an allele. a nucleotide sequence is not returned for a specific allele if input allele is not available in the ANHIG/IMGTHLA Github Repository. position will be left empty if specific allele does not have a nucleotide at the input position.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>

<hr>
<h2 id='multiAlleleTrim'>Trim Multiple HLA Allele Names</h2><span id='topic+multiAlleleTrim'></span>

<h3>Description</h3>

<p>Trim a vector of allele names to a specified number of fields or digits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiAlleleTrim(alleles, resolution, version = 3, append = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiAlleleTrim_+3A_alleles">alleles</code></td>
<td>
<p>A vector of HLA allele names from a single nomenclature epoch.</p>
</td></tr>
<tr><td><code id="multiAlleleTrim_+3A_resolution">resolution</code></td>
<td>
<p>A number identifying the number of fields to which the alleles should be trimmed.</p>
</td></tr>
<tr><td><code id="multiAlleleTrim_+3A_version">version</code></td>
<td>
<p>A number identifying the HLA nomenclature epoch under which the allele was named. Epoch 1 allele names are found in IPD-IMGT/HLA Database releases 1.0.0 to 1.16.0. Epoch 2 allele names are found in IPD-IMGT/HLA Database releases 2.0.0 to 2.28.0. Epoch 3 allele names are found in IPD_IMGT/HLA Database releases 3.0.0 and onward.</p>
</td></tr>
<tr><td><code id="multiAlleleTrim_+3A_append">append</code></td>
<td>
<p>A logical. When append = TRUE, the expression variant suffix of a full-length allele name is appended to a truncated allele name. The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of the trimmed allele name, shortened according to the input parameters.
</p>


<h3>Note</h3>

<p>Expression variant suffixes will not be removed from full-length allele names that have fewer than four fields or eight digits.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alleles &lt;- c("A*02:01:01:02L","DRB1*08:07", "DQB1*04:02:01:16Q")
multiAlleleTrim(alleles,2)
multiAlleleTrim(alleles,2,append = TRUE)

alleles &lt;- c("A*01010102L","DRB1*1613N","HLA-Cw*0322Q")
multiAlleleTrim(alleles,1,2,TRUE)
multiAlleleTrim(alleles,2,2)

</code></pre>

<hr>
<h2 id='multiGLStoUNI'>Translate Multiple GL Strings to UNIFORMAT</h2><span id='topic+multiGLStoUNI'></span>

<h3>Description</h3>

<p>Translate a data frame or a vector of GL Strings to UNIFORMAT strings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiGLStoUNI(GLstringArray, prefix = "HLA-", pre = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiGLStoUNI_+3A_glstringarray">GLstringArray</code></td>
<td>
<p>A data frame or a vector containing GL String formatted data. If 'GLstringArray' is a data frame with more than one column, the first column should contain only identifiers. If 'GLstringArray' is a vector, it should contain only GL Strings.</p>
</td></tr>
<tr><td><code id="multiGLStoUNI_+3A_prefix">prefix</code></td>
<td>
<p>A string of the desired locus prefix (default is &quot;HLA-&quot;).</p>
</td></tr>
<tr><td><code id="multiGLStoUNI_+3A_pre">pre</code></td>
<td>
<p>A logical. If 'pre' is TRUE, all allele names will be prefixed with 'prefix'. If 'pre' is FALSE, no allele names will be prefixed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of 'GLStringArray' in which the GL String data has been converted to UNIFORMAT format. If a 'GLstringArray' was a data frame, a data frame is returned. If 'GLstringArray' was a vector, a vector is returned.
</p>


<h3>Note</h3>

<p>GL Strings that include the &quot;?&quot; operator will not be translated, as the &quot;?&quot; operator has no cognate in UNIFORMAT.
</p>


<h3>References</h3>

<p>Nunes Tissue Antigens 2007;69(s1):203-205 https://doi.org/10.1111/j.1399-0039.2006.00808.x
</p>
<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>


<h3>Examples</h3>

<pre><code class='language-R'>multiGLStoUNI(GLstring.ex[[2]][1:5],version) ## converting a vector of GL Strings to UNIFORMAT
multiGLStoUNI(GLstring.ex[1:5,1:2])  ## converting a data.frame of GL Strings to UNIFORMAT

</code></pre>

<hr>
<h2 id='multiLocusValidation'>Apply validateLocus() to Multiple Loci</h2><span id='topic+multiLocusValidation'></span>

<h3>Description</h3>

<p>Applies validateLocus() to a vector of locus names, validates them against HLAgazeteer$gen, and returns a vector of validated locus names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiLocusValidation(loci, source = "gDNA", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiLocusValidation_+3A_loci">loci</code></td>
<td>
<p>A character vector of locus names found in the current HLAgazeteer.</p>
</td></tr>
<tr><td><code id="multiLocusValidation_+3A_source">source</code></td>
<td>
<p>A character vector describing the type of alignment 'loci' should be validated against. Allowed options are 'AA' (protein), 'cDNA' (nucleotide) and 'gDNA' (genomic). Only a single value should be provided. The default value is 'gen'.</p>
</td></tr>
<tr><td><code id="multiLocusValidation_+3A_verbose">verbose</code></td>
<td>
<p>A logical value. If 'verbose' = TRUE, messages describing invalid 'loci' or 'source' values are generated. If 'verbose' = FALSE, no messages are generated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of locus names that are present in HLAgazeteer$gen.
</p>


<h3>Note</h3>

<p>The results of this check should only be considered valid for the IPD-IMGT/HLA Database release version of the current HLAgazeteer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>multiLocusValidation(loci = c("DRB1","DPB1","DQB8"))
multiLocusValidation(loci = c("A","B","C","D","Q"))

</code></pre>

<hr>
<h2 id='multiQueryRelease'>Search Multiple Elements of Allele Names Across Release Versions</h2><span id='topic+multiQueryRelease'></span>

<h3>Description</h3>

<p>Searches specific release versions in the AlleleListHistory object for a set of user-defined allele-name elements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiQueryRelease(rel, variants = "", all = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiQueryRelease_+3A_rel">rel</code></td>
<td>
<p>An IPD-IMGT/HLA Database release version, represented as either a character (e.g., &quot;3.58.0&quot;) or a numeric (e.g., 3580) value.</p>
</td></tr>
<tr><td><code id="multiQueryRelease_+3A_variants">variants</code></td>
<td>
<p>A vector of character strings. The values in 'variants' can be any part of a locus or allele name (e.g., &quot;DR&quot;, &quot;02:01&quot;, &quot;DRB1*08:07&quot;). The default (&quot;&quot;) specifies all alleles in 'rel'.</p>
</td></tr>
<tr><td><code id="multiQueryRelease_+3A_all">all</code></td>
<td>
<p>A logical. When 'all' = TRUE, a vector of all alleles in 'rel' that share all elements of 'variants' is returned.  When 'all' = FALSE, the number of alleles that share all elements of 'variants' in 'rel' is returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of all alleles that share all 'variants' in 'rel' or the number of alleles that share all 'variants' in 'rel'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Identify DRB1 allele names that include a 'Q' character in release 3.58.0.
multiQueryRelease("3.58.0",c("DRB1","Q"),TRUE) 
# Identify the number of null (N) DRB1 alleles.
multiQueryRelease("3.58.0",c("DRB1","N"),FALSE)

</code></pre>

<hr>
<h2 id='multiSearch'>Search Alignment Sequences at Multiple Positions for a Specified Allele</h2><span id='topic+multiSearch'></span>

<h3>Description</h3>

<p>Generates a character string of multiple protein, codon or nucleotide or genomic sequences at the specified positions for the specified allele name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiSearch(
  alignType,
  locus,
  allele,
  positions,
  prefix = TRUE,
  sep = "~",
  trimmed = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiSearch_+3A_aligntype">alignType</code></td>
<td>
<p>The type of alignment being searched. Allowed values are &quot;prot&quot;, codon&quot;, &quot;nuc&quot; and &quot;gen&quot;. Only one 'alignType' value is allowed.</p>
</td></tr>
<tr><td><code id="multiSearch_+3A_locus">locus</code></td>
<td>
<p>A specific locus.</p>
</td></tr>
<tr><td><code id="multiSearch_+3A_allele">allele</code></td>
<td>
<p>An allele name.</p>
</td></tr>
<tr><td><code id="multiSearch_+3A_positions">positions</code></td>
<td>
<p>Nucleotide positions to search.</p>
</td></tr>
<tr><td><code id="multiSearch_+3A_prefix">prefix</code></td>
<td>
<p>A logical that indicates if the position number(s) should be included in the result.</p>
</td></tr>
<tr><td><code id="multiSearch_+3A_sep">sep</code></td>
<td>
<p>Defines the separator between position sequences.</p>
</td></tr>
<tr><td><code id="multiSearch_+3A_trimmed">trimmed</code></td>
<td>
<p>A logical identifying whether 'allele' is a two-field, 'trimmed' allele name (trimmed=TRUE), or a full-length name (trimmed=FALSE). The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>codons (possibly with position number) at 'positions' for 'allele'. If 'allele' does not have a codon at 'position', &rdquo; will be returned.
</p>


<h3>Note</h3>

<p>For internal HLAtools use only.
</p>

<hr>
<h2 id='multiTranslateGLstring'>Translate a Vector of GL Strings to a Desired IPD-IMGT/HLA Database Version</h2><span id='topic+multiTranslateGLstring'></span>

<h3>Description</h3>

<p>Translates a column from a data frame in GL String Code format to a desired reference database version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiTranslateGLstring(GLstrings, from, to, expand = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiTranslateGLstring_+3A_glstrings">GLstrings</code></td>
<td>
<p>A vector of GL Strings encoded in the same IPD-IMGT/HLA Database release version.</p>
</td></tr>
<tr><td><code id="multiTranslateGLstring_+3A_from">from</code></td>
<td>
<p>A character string identifying the IPD-IMGT/HLA Database release version of 'GLstrings'.</p>
</td></tr>
<tr><td><code id="multiTranslateGLstring_+3A_to">to</code></td>
<td>
<p>A character string identifying the IPD-IMGT/HLA Database release version to translate 'GLstrings' to. Values can range from version 1.05.0 to the loaded version of the alleleListHistory object.</p>
</td></tr>
<tr><td><code id="multiTranslateGLstring_+3A_expand">expand</code></td>
<td>
<p>A logical that indicates if the lowest-numbered truncated allele names that match truncated allele names in 'GLstrings' should be returned (expand = FALSE), or if a slash-delimited string of all matching full-length allele names should be returned (expand = TRUE). The default value is FALSE.</p>
</td></tr>
<tr><td><code id="multiTranslateGLstring_+3A_verbose">verbose</code></td>
<td>
<p>A logical that indicates if messages regarding the trnslation process should be sent to the console (TRUE) or not (FALSE). The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version GLSCs (in a data frame) updated to the desired version. NA values are returned in place of alleles that are not present in 'to'.
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>


<h3>Examples</h3>

<pre><code class='language-R'>multiGLs &lt;- multiTranslateGLstring(GLstring.ex$Gl.String[1:50],"3.01.0","2.15.0")
multiGLs &lt;- multiTranslateGLstring(GLstring.ex$Gl.String[1:50],"3.30.0","3.58.0")

</code></pre>

<hr>
<h2 id='multiUNItoGLS'>Translate Multiple UNIFORMAT Strings to GL Strings</h2><span id='topic+multiUNItoGLS'></span>

<h3>Description</h3>

<p>Translate a data frame or vector of UNIFORMAT strings to GL Strings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiUNItoGLS(UniformatArray, prefix = "HLA-", pre = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiUNItoGLS_+3A_uniformatarray">UniformatArray</code></td>
<td>
<p>A data frame or vector of UNIFORMAT formatted strings. If 'UniformatArray' is a data frame with more than one column, the first column should contain only identifiers. If 'UniformatgArray' is a vector, it should contain only UNIFORMAT strings.</p>
</td></tr>
<tr><td><code id="multiUNItoGLS_+3A_prefix">prefix</code></td>
<td>
<p>A character string of the desired locus prefix (default is &quot;HLA-&quot;).</p>
</td></tr>
<tr><td><code id="multiUNItoGLS_+3A_pre">pre</code></td>
<td>
<p>A logical. If 'pre' is TRUE, all allele names will be prefixed with 'prefix'. If 'pre' is FALSE, no allele names will be prefixed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of 'UniformatArray' in which the UNIFORMAT data have been converted to GL String format. If a 'UniformatArray' was a data frame, a data frame is returned. If 'UniformatArray' was a vector, a vector is returned.
</p>


<h3>Note</h3>

<p>This function does not return the GL String &quot;?&quot; operator, as the &quot;?&quot; operator has no cognate in UNIFORMAT.
</p>


<h3>References</h3>

<p>Nunes Tissue Antigens 2007;69(s1):203-205 https://doi.org/10.1111/j.1399-0039.2006.00808.x
</p>
<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>


<h3>Examples</h3>

<pre><code class='language-R'>multiUNItoGLS(unname(as.vector(UNIFORMAT.example[2]))[[1]])
multiUNItoGLS(UNIFORMAT.example[1:2])

</code></pre>

<hr>
<h2 id='multiUpdateGL'>Update a Vector of GL String Code Data to a Desired IPD-IMGT/HLA Database Version</h2><span id='topic+multiUpdateGL'></span>

<h3>Description</h3>

<p>Updates a column from a data frame in GL String Code format to a desired reference database version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiUpdateGL(GLSCs, to, expand = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiUpdateGL_+3A_glscs">GLSCs</code></td>
<td>
<p>A vector of GL String Codes.</p>
</td></tr>
<tr><td><code id="multiUpdateGL_+3A_to">to</code></td>
<td>
<p>A character string identifying the IPD-IMGT/HLA Database release version to translate 'GLSCs' to. Values can range from version 1.05.0 to the loaded version of the alleleListHistory object.</p>
</td></tr>
<tr><td><code id="multiUpdateGL_+3A_expand">expand</code></td>
<td>
<p>A logical that indicates if the lowest-numbered truncated allele names that match truncated allele names in 'GLSCs' should be returned (expand = FALSE), or if a slash-delimited string of all matching full-length allele names should be returned (expand = TRUE). The default value is FALSE.</p>
</td></tr>
<tr><td><code id="multiUpdateGL_+3A_verbose">verbose</code></td>
<td>
<p>A logical that indicates if messages regarding the update process should be sent to the console (TRUE) or not (FALSE). The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version GLSCs (in a data frame) updated to the desired version. NA values are returned in place of alleles that are not present in 'to'.
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>
<p>Mack et al. HLA 2023;102(4):501-507 https://doi.org/10.1111/tan.15145
</p>


<h3>Examples</h3>

<pre><code class='language-R'>multiUpdateGLs &lt;- multiUpdateGL(GLSC.ex$GL.String.Code[1:50],"2.25.0")
multiUpdateGLs &lt;- multiUpdateGL(GLSC.ex$GL.String.Code[1:50],"3.58.0")

</code></pre>

<hr>
<h2 id='numFields'>Identify the Number of Fields in a Colon-Delimited Allele Name</h2><span id='topic+numFields'></span>

<h3>Description</h3>

<p>Returns the number of fields in a colon-delimited HLA allele name. A value of 1 is returned for digit-delimited HLA allele names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>numFields(allele)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="numFields_+3A_allele">allele</code></td>
<td>
<p>A character string of a colon-delimited HLA allele name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value describing the number of fields in the allele name.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>numFields("HLA-A*01:01")
numFields("DRB1*04:03:01")
numFields("13:02:01:01")
</code></pre>

<hr>
<h2 id='parseAlignmentHead'>Guides for Parsing the Header Blocks of Alignment Files</h2><span id='topic+parseAlignmentHead'></span>

<h3>Description</h3>

<p>Returns a vector describing the location of key information in the header blocks of alignment files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseAlignmentHead(version)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parseAlignmentHead_+3A_version">version</code></td>
<td>
<p>A character string of a validated IPD-IMGT/HLA Database release version (e.g., '3.25.0' or '3250').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either FALSE if the version is not valid, or two-value numerical vector describing (1) the header line in which alignment version data appears and (2) the length of the character string in that line preceding version data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>parseAlignmentHead("3.25.0")

</code></pre>

<hr>
<h2 id='posSort'>Numerical Sort of Alignment Positions that Contain Indels</h2><span id='topic+posSort'></span>

<h3>Description</h3>

<p>Sorts sequence alignment positions that contain indels in numerical order; e.g., three indels following position X are are identified as X.1, X.2 and X.3.
</p>
<p>Positions are validated against a specified locus for a specified alignment type. Positions that do not exist for that locus-alignment combination are not returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posSort(posVec, alignType, locus)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="posSort_+3A_posvec">posVec</code></td>
<td>
<p>A character vector of variant positions.</p>
</td></tr>
<tr><td><code id="posSort_+3A_aligntype">alignType</code></td>
<td>
<p>A character string describing the type of alignment the positions are found in. The values 'prot', 'codon', 'nuc' and 'gen' are valid options. Only one 'alignType' value is allowed.</p>
</td></tr>
<tr><td><code id="posSort_+3A_locus">locus</code></td>
<td>
<p>A character string describing a locus in the HLAalignments object for the specified alignType.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string of the correctly sorted sequence.
</p>


<h3>Note</h3>

<p>Indel positions must be text-formatted (e.g. &quot;607.12&quot;). C.f., posSort(c(2,4,3,1,5), &quot;nuc&quot;,&quot;DRB1&quot;) vs posSort(c(&quot;607.23&quot;,&quot;607.10&quot;,&quot;607.3&quot;,&quot;607.4&quot;),&quot;nuc&quot;,&quot;DRB1&quot;).
</p>

<hr>
<h2 id='pypopHeaders'>Convert BIGDAWG File Headers to PyPop Format</h2><span id='topic+pypopHeaders'></span>

<h3>Description</h3>

<p>Convert the header of a BIGDAWG-formatted data frame into a Python for Population Genomics (PyPop) formatted header.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pypopHeaders(colHead)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pypopHeaders_+3A_colhead">colHead</code></td>
<td>
<p>A vector of column names. BIGDAWG format requires that the first two header values are the sample identifier character string and the subject status (0 or 1). All other header values should be paired locus/gene names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A PyPop-formatted column header vector.
</p>


<h3>Note</h3>

<p>This function returns unique locus names for each column. For example, &quot;A_1&quot; and &quot;A_2&quot; headers will be returned for two &quot;A&quot; locus columns. PyPop configuration (.ini) files allow this format to be customized.
</p>


<h3>References</h3>

<p>Lancaster et al. Front Immunol. 2024 Apr 2;15:1378512. https://pubmed.ncbi.nlm.nih.gov/38629078/
</p>
<p>Pappas et al. Hum Immunol. 2016 Mar 77(3):283-287. https://pubmed.ncbi.nlm.nih.gov/26708359/
</p>


<h3>See Also</h3>

<p><a href="http://pypop.org/docs/guide-chapter-usage.html#a-minimal-configuration-file">PyPop Configuration File</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pyHead &lt;- pypopHeaders(colnames(sHLAdata))

</code></pre>

<hr>
<h2 id='queryRelease'>Search Allele Names Across Release Versions</h2><span id='topic+queryRelease'></span>

<h3>Description</h3>

<p>Searches specific release versions in the AlleleListHistory object for user-defined allele variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>queryRelease(rel, variant = "", all = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="queryRelease_+3A_rel">rel</code></td>
<td>
<p>An IPD-IMGT/HLA Database release version, represented as either a character (e.g., &quot;3.56.0&quot;) or a numeric (e.g., 3560) value.</p>
</td></tr>
<tr><td><code id="queryRelease_+3A_variant">variant</code></td>
<td>
<p>A character string. The value of 'variant' can be any part of a locus or allele name (e.g., &quot;DR&quot;, &quot;02:01&quot;, &quot;DRB1*08:07&quot;). The default (&quot;&quot;) specifies all alleles in 'rel'.</p>
</td></tr>
<tr><td><code id="queryRelease_+3A_all">all</code></td>
<td>
<p>A logical. When 'all' = TRUE, a vector of all instances of 'variant' in 'rel' is returned.  When 'all' = FALSE, the number of instances of 'var' in 'rel' is returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of all matches to 'variant' in 'rel' or the number of all matches to 'variant' in 'rel'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Identify the number of DRB9 alleles in releases 3.30.0 and 3.31.0.
queryRelease("3.30.0","DRB9",FALSE) 
queryRelease("3.31.0","DRB9",FALSE)

# Identify the total number of alleles in release 3.56.0.
queryRelease(3560)

</code></pre>

<hr>
<h2 id='redec'>Reintroduce Version Decimals</h2><span id='topic+redec'></span>

<h3>Description</h3>

<p>Correctly put decimals back into an AlleleListHistory column-formatted version name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>redec(Cname)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="redec_+3A_cname">Cname</code></td>
<td>
<p>A character string describing a version name in AlleleListHistory column format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string describing a version name with decimals in appropriate places according to the IPD-IMGT/HLA Database.
</p>


<h3>Note</h3>

<p>For internal use only.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>redec("X3090")

</code></pre>

<hr>
<h2 id='relRisk'>Calculate Relative Risk for Individual Alleles and Genotypes in BIGDAWG-formatted Non-Case-Control Datasets</h2><span id='topic+relRisk'></span>

<h3>Description</h3>

<p>Returns a list object containing relative risk, confidence interval and p-value data for the individual alleles and individual genotypes at each locus in a BIGDAWG-formatted non-case-control genotype data frame or file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relRisk(dataset, return = TRUE, save.path = tempdir())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="relRisk_+3A_dataset">dataset</code></td>
<td>
<p>A character string describing the name of a non-case-control genotype dataset using the BIGDAWG format. Here, &quot;non-case-control&quot; means that while two subject categories are required, the categories should not be patients and controls; instead, the categories may be, e.g., for a dataset of patients, either of two disease states, where one disease state is coded as 0 and the other is coded as 1 in the second column of the dataset. Either a tab-delimited file or a data frame can be specified.</p>
</td></tr>
<tr><td><code id="relRisk_+3A_return">return</code></td>
<td>
<p>A logical identifying if the list object should be returned (return=TRUE), or if pairs of tab-delimited text files of results (one for alleles and one for genotypes) should be written to the working directory for each locus.</p>
</td></tr>
<tr><td><code id="relRisk_+3A_save.path">save.path</code></td>
<td>
<p>A character string identifying the path in which to write the pair of files when return is FALSE. The default value is tempdir().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object of two lists (&quot;alleles&quot; and &quot;genotypes&quot;), each of which contains a list of nine-column data frames containing results for each unique allele or genotype (in rows) at each locus. Column headers in each dataframe are, <em>Locus</em>, <em>Variant</em>, <em>Status_1</em>, <em>Status_0</em>, <em>RelativeRisk</em>, <em>CI.low</em>, <em>CI.high</em>, <em>p.value</em>, and <em>Significant</em>.
</p>


<h3>References</h3>

<p><a href="https://CRAN.R-project.org/package=BIGDAWG/vignettes/BIGDAWG.html#input-data">BIGDAWG Data Format</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rr &lt;- relRisk(sHLAdata)

</code></pre>

<hr>
<h2 id='repoVersion'>Convert an AlleleListHistory Release Version to the GitHub Repository Version</h2><span id='topic+repoVersion'></span>

<h3>Description</h3>

<p>Removes periods from IMGT/HLA Database release versions and converts release versions '3.00.0'-'3.09.0' to '300'-'390' to facilitate working with these GitHub repo branches.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>repoVersion(version)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="repoVersion_+3A_version">version</code></td>
<td>
<p>A character string of a validated IPD-IMGT/HLA Database release version.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string of the release version appropriate for use in an IMGT/HLA GitHub repository URL.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>repoVersion("3.05.0")

</code></pre>

<hr>
<h2 id='sHLAdata'>Synthetic HLA Data for use with Package Examples</h2><span id='topic+sHLAdata'></span>

<h3>Description</h3>

<p>A BIGDAWG-formatted data frame of 18 columns and 47 rows containing synthetic HLA-A, -C, -B, -DRB1, -DQA1, -DQB1, -DPA1, and -DPB1 genotype data for 24 control subjects and 23 case subjects. Allele name data were recorded under IPD-IMGT/HLA Database release version 3.56.0. These are synthetic data generated for the imaginary UchiTelle population, and do not represent biologically-derived HLA genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sHLAdata)
</code></pre>


<h3>Format</h3>

<p>A data frame of 18 columns and 47 rows.
</p>


<h3>Source</h3>

<p>These synthetic data were generated as part of the 13th International HLA Workshop to demonstrate PyPop functions. See <a href="http://pypop.org/docs/guide-chapter-usage.html#data-minimal-noheader">http://pypop.org/docs/guide-chapter-usage.html#data-minimal-noheader</a> for additional details.
</p>

<hr>
<h2 id='squashVersion'>Reduce a Release Version to Numerals</h2><span id='topic+squashVersion'></span>

<h3>Description</h3>

<p>Removes the '.' delimiters from an IPD-IMGT/HLA Database release version name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>squashVersion(ver, num = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="squashVersion_+3A_ver">ver</code></td>
<td>
<p>A character vector-formatted IPD-IMGT/HLA Database release version name (e.g., '3.54.0').</p>
</td></tr>
<tr><td><code id="squashVersion_+3A_num">num</code></td>
<td>
<p>A Logial. A numeric value is returned when num = TRUE. A character vector is returned when num = FALSE. The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 4-digit value, as either a character string or a number.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>squashVersion("3.45.0",TRUE)

</code></pre>

<hr>
<h2 id='translateAllele'>Translate HLA Allele Names across IPD-IMGT/HLA Database Release Versions</h2><span id='topic+translateAllele'></span>

<h3>Description</h3>

<p>Translates a single HLA allele name across IPD-IMGT/HLA Database release versions.  Truncated allele names in one version can be expanded to a list of all allele names that match the provided allele name in a chosen version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translateAllele(allele, from, to, expand = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="translateAllele_+3A_allele">allele</code></td>
<td>
<p>A full or truncated HLA allele name.</p>
</td></tr>
<tr><td><code id="translateAllele_+3A_from">from</code></td>
<td>
<p>A character string identifying the IPD-IMGT/HLA Database release version of 'allele'.</p>
</td></tr>
<tr><td><code id="translateAllele_+3A_to">to</code></td>
<td>
<p>A character string identifying the IPD-IMGT/HLA Database release version to translate 'allele' to. Values can range from version 1.05.0 to the loaded version of the the alleleListHistory object.</p>
</td></tr>
<tr><td><code id="translateAllele_+3A_expand">expand</code></td>
<td>
<p>A logical that indicates if the lowest-numbered truncated allele name that matches a truncated allele name should be returned (expand = FALSE), or if a slash-delimited string of all matching full-length allele names should be returned (expand = TRUE). The default value is FALSE.</p>
</td></tr>
<tr><td><code id="translateAllele_+3A_verbose">verbose</code></td>
<td>
<p>A logical that indicates if messages regarding allele translations should be sent to the console (TRUE) or not (FALSE). The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string of either a single matching allele or all matching alleles in the 'to' release version. If an allele does not have a cognate match in 'to', NA is returned for that allele.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>translateAllele("A*01:01","3.01.0","2.20.0",FALSE)
translateAllele("A*0101","2.20.0","3.01.0",TRUE)
translateAllele("B*57:01","3.12.0","3.12.0",TRUE)

</code></pre>

<hr>
<h2 id='translateGLstring'>Translate a GL String across IPD-IMGT/HLA Database Release Versions</h2><span id='topic+translateGLstring'></span>

<h3>Description</h3>

<p>Translates a single Genotype List (GL) string across IPD-IMGT/HLA Database release versions. Truncated allele names can be expanded to the list of all allele names that contain the truncated name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translateGLstring(GLstring, from, to, expand = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="translateGLstring_+3A_glstring">GLstring</code></td>
<td>
<p>A GL String of HLA allele names.</p>
</td></tr>
<tr><td><code id="translateGLstring_+3A_from">from</code></td>
<td>
<p>A character string identifying the IPD-IMGT/HLA Database release version of 'GLstring'.</p>
</td></tr>
<tr><td><code id="translateGLstring_+3A_to">to</code></td>
<td>
<p>A character string identifying the IPD-IMGT/HLA Database release version to translate 'GLstring' to. Values can range from version 1.05.0 to the loaded version of the alleleListHistory object.</p>
</td></tr>
<tr><td><code id="translateGLstring_+3A_expand">expand</code></td>
<td>
<p>A logical that indicates if the lowest-numbered truncated allele names that match truncated allele names in the GL String should be returned (expand = FALSE), or if a slash-delimited string of all matching full-length allele names should be returned (expand = TRUE). The default value is FALSE.</p>
</td></tr>
<tr><td><code id="translateGLstring_+3A_verbose">verbose</code></td>
<td>
<p>A logical that indicates if messages regarding allele translations should be sent to the console (TRUE) or not (FALSE). The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A GL String of allele names in the 'to' release version. If the GL String is determined to be invalid, FALSE is returned.
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>


<h3>Examples</h3>

<pre><code class='language-R'>translateGLstring(GLstring.ex$Gl.String[1],"3.01.0","2.15.0")
translateGLstring(GLstring.ex$Gl.String[1],"3.01.0","2.15.0",TRUE)
translateGLstring(GLstring.ex$Gl.String[1],"3.01.0","2.15.0",FALSE,TRUE)
translateGLstring(GLstring.ex$Gl.String[1],"3.01.0","2.15.0",TRUE,TRUE)

</code></pre>

<hr>
<h2 id='typeToSource'>Convert AlignType Values to Source Values</h2><span id='topic+typeToSource'></span>

<h3>Description</h3>

<p>Converts between 'alignType' values, identifying four types of sequence alignments and 'source' values, identifying three kinds of alignment files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>typeToSource(alignVector, toSource = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="typeToSource_+3A_alignvector">alignVector</code></td>
<td>
<p>A vector of character values specifying kinds of four sequence alignment types (&quot;prot&quot;,&quot;nuc&quot;,&quot;codon&quot; and &quot;gen&quot;), or three source alignment files (&quot;AA&quot;,&quot;cDNA&quot;, and &quot;gDNA&quot;).</p>
</td></tr>
<tr><td><code id="typeToSource_+3A_tosource">toSource</code></td>
<td>
<p>A logical. If 'toSource' is true, 'alignType' values are converted to 'source' values. If 'toSource' is false, 'source' values are concerted to 'alignType' values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string of the converted 'align' or 'source' value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>typeToSource(c("nuc","prot","gen"),TRUE) 

</code></pre>

<hr>
<h2 id='uniAlign'>Generate an Alignment for Specific Alleles at Specific Positions</h2><span id='topic+uniAlign'></span>

<h3>Description</h3>

<p>Generates a peptide, codon, coding or genomic nucleotide alignment at a single set of positions for HLA alleles at one or more loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uniAlign(alignType, alleles, positions)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uniAlign_+3A_aligntype">alignType</code></td>
<td>
<p>The type of alignment being searched. Allowed values are &quot;prot&quot;, codon&quot;, &quot;nuc&quot; and &quot;gen&quot;. Only one 'alignType' value is allowed.</p>
</td></tr>
<tr><td><code id="uniAlign_+3A_alleles">alleles</code></td>
<td>
<p>A vector of un-prefixed HLA locus names.</p>
</td></tr>
<tr><td><code id="uniAlign_+3A_positions">positions</code></td>
<td>
<p>A vector of codon positions, against which all loci will be aligned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of allele names and the corresponding codon sequence for specified position. A codon sequence is not returned for alleles that are not present in the loaded HLAalignments object. A sequence position will be populated with whitespace if a specific allele has no sequence at the specified position.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>

<hr>
<h2 id='UNIFORMAT.example'>Example Data Frame of UNIFORMAT Data.</h2><span id='topic+UNIFORMAT.example'></span>

<h3>Description</h3>

<p>A two-column example data frame including UNIFORMAT data.
Column one contains sample identifiers, while column two identifies each subject's HLA genotype in UNIFORMAT format.
To be used for example runs and demonstrations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(UNIFORMAT.example)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 20 rows and 2 columns.
</p>


<h3>Source</h3>

<p>https://hla-net.eu/wp/wp-content/uploads/example-three-loci.unif_.txt
</p>


<h3>References</h3>

<p>Nunes Tissue Antigens 2007;69(s1):203-205 https://doi.org/10.1111/j.1399-0039.2006.00808.x
</p>

<hr>
<h2 id='uniSearch'>Search Sequences at a Single Position for an Allele</h2><span id='topic+uniSearch'></span>

<h3>Description</h3>

<p>Generates a character string of the peptide, codon or nucleotide sequence at the specified position for the specified HLA allele.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uniSearch(alignType, locus, allele, position, prefix = TRUE, trimmed = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uniSearch_+3A_aligntype">alignType</code></td>
<td>
<p>The type of alignment being searched. Allowed values are &quot;prot&quot;, &quot;codon&quot;, &quot;nuc&quot; and &quot;gen&quot;. Only one 'alignType' value is allowed.</p>
</td></tr>
<tr><td><code id="uniSearch_+3A_locus">locus</code></td>
<td>
<p>A specific HLA locus.</p>
</td></tr>
<tr><td><code id="uniSearch_+3A_allele">allele</code></td>
<td>
<p>The name of the allele being searched.</p>
</td></tr>
<tr><td><code id="uniSearch_+3A_position">position</code></td>
<td>
<p>The specified position.</p>
</td></tr>
<tr><td><code id="uniSearch_+3A_prefix">prefix</code></td>
<td>
<p>A logical that indicates if the position number(s) should be included in the result.</p>
</td></tr>
<tr><td><code id="uniSearch_+3A_trimmed">trimmed</code></td>
<td>
<p>A logical identifying whether 'allele' is a two-field, 'trimmed' allele name (trimmed=TRUE), or a full-length name (trimmed=FALSE). The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The peptide residue, codon or nucleotide sequence at specified position for specified allele as available in the ANHIG/IMGTHLA Github Repository.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>

<hr>
<h2 id='UNItoGLS'>Translate UNIFORMAT to GL String</h2><span id='topic+UNItoGLS'></span>

<h3>Description</h3>

<p>A wrapper function for UNtoGL() which translates strings from UNIFORMAT format to GL String format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UNItoGLS(uniformat, prefix = "HLA-", pre = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UNItoGLS_+3A_uniformat">uniformat</code></td>
<td>
<p>A character string of HLA allele names and operators in the UNIFORMAT format signifying their relation with one another.</p>
</td></tr>
<tr><td><code id="UNItoGLS_+3A_prefix">prefix</code></td>
<td>
<p>A character string of the desired gene-system prefix (default is &quot;HLA-&quot;).</p>
</td></tr>
<tr><td><code id="UNItoGLS_+3A_pre">pre</code></td>
<td>
<p>A logical that indicates whether returned allele names should contain 'prefix' (TRUE), or if 'prefix' should be excluded from returned names (FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An version of 'uniformat' converted to GL String format, or FALSE if 'uniformat' is invalid.
</p>


<h3>Note</h3>

<p>This function does not return the &quot;?&quot; operator, as the &quot;?&quot; operator has no cognate in UNIFORMAT.
</p>


<h3>References</h3>

<p>Nunes Tissue Antigens 2007;69(s1):203-205 https://doi.org/10.1111/j.1399-0039.2006.00808.x
</p>
<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>


<h3>Examples</h3>

<pre><code class='language-R'>UNItoGLS("A*02:01,A*03:01|A*02:01,A*03:02|A*02:02,A*03:01|A*02:02,A*03:02")
UNItoGLS("A,B|A,C~D G|E W,X|W,Y Z,J J,Z")

</code></pre>

<hr>
<h2 id='UNtoGL'>Translate UNIFORMAT Strings to GL Strings</h2><span id='topic+UNtoGL'></span>

<h3>Description</h3>

<p>A function that translates strings from UNIFORMAT format to GL String format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UNtoGL(uniformat, prefix = "HLA-", pre = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UNtoGL_+3A_uniformat">uniformat</code></td>
<td>
<p>A character string of HLA allele names and operators in the UNIFORMAT format signifying their relation with one another.</p>
</td></tr>
<tr><td><code id="UNtoGL_+3A_prefix">prefix</code></td>
<td>
<p>A character string of the desired prefix (default is &quot;HLA-&quot;).</p>
</td></tr>
<tr><td><code id="UNtoGL_+3A_pre">pre</code></td>
<td>
<p>A logical that indicates whether user would like all allele names to contain the prefix of their choice (TRUE), or if the prefix should not be appended to allele names (FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An altered version of the input character string, converted to GL String format.
</p>


<h3>Note</h3>

<p>For internal use only.
</p>
<p>This function does not return the &quot;?&quot; operator, as the &quot;?&quot; operator has no cognate in UNIFORMAT.
</p>


<h3>References</h3>

<p>Nunes Tissue Antigens 2007;69(s1):203-205 https://doi.org/10.1111/j.1399-0039.2006.00808.x
</p>
<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>


<h3>Examples</h3>

<pre><code class='language-R'>UNtoGL("A*02:01,A*03:01|A*02:01,A*03:02|A*02:02,A*03:01|A*02:02,A*03:02")
UNtoGL("A,B|A,C|D,B|D,C|E,B|E,C", pre = FALSE)

</code></pre>

<hr>
<h2 id='updateAll'>Update All Package Data Objects Derived from IPD-IMGT/HLA Database Resources</h2><span id='topic+updateAll'></span>

<h3>Description</h3>

<p>Applies updateAlleleListHistory(), atlasFull(), buildGazeteer(), extractGeneTypes(), and ffN() to update the alleleListHistory, HLAatlas, HLAgazeteer, IMGTHLAGeneTypes and fragmentFeatureNames data objects.
</p>
<p>A new alleleListHistory data object should be generated with each IPD-IMGT/HLA Database release. The other data objects will likely only change when new genes are added to the IPD-IMGT/HLA Database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>updateAll(updateType = "all", version = getLatestVersion())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="updateAll_+3A_updatetype">updateType</code></td>
<td>
<p>A character vector of the names of data objects to be updated. By default, updateAll() builds all five data objects (updateType=&quot;all&quot;). Alternatively specific data objects can be updated; e.g., updateType=&quot;alleleListHistory&quot; or updateType=c(&quot;alleleListHistory&quot;,&quot;fragmentFeatureNames&quot;).</p>
</td></tr>
<tr><td><code id="updateAll_+3A_version">version</code></td>
<td>
<p>A numeric value or character string identifying the version of the ANHIG/IMGTHLA Github repository to build these objects from. By default, updateAll() calls the getLatestVersion() function to identify the most recent IPD-IMGT/HLA Database release.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value is returned. The desired versions of the specified data objects are built into the environment that called updateAll().
</p>


<h3>Note</h3>

<p>Generating a new HLAatlas can take 5 minutes or more to complete.
</p>

<hr>
<h2 id='updateAlleleListHistory'>Build the AlleleList History R Object</h2><span id='topic+updateAlleleListHistory'></span>

<h3>Description</h3>

<p>Consumes the ANHIG/IMGTHLA GitHub repository's allelelisthistory.txt file to create the AlleleListHistory object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>updateAlleleListHistory()
</code></pre>


<h3>Value</h3>

<p>A list object containing two data frames. alleleListHistory$Version identifies the date and version of alleleListHistory$AlleleListHistory. alleleListHistory$AlleleListHistory relates each IPD-IMGT/HLA Database Accession ID (HLA_ID) to the corresponding HLA allele name in each release version since 1.05.0.
</p>

<hr>
<h2 id='updateGL'>Update a GL String Code to a Specified IPD-IMGT/HLA Database Version</h2><span id='topic+updateGL'></span>

<h3>Description</h3>

<p>Updates the elements of a GLString Code across IPD-IMGT/HLA Database release versions. Truncated allele names can be expanded to the list of all allele names that contain the truncated name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>updateGL(GLSC, to, expand = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="updateGL_+3A_glsc">GLSC</code></td>
<td>
<p>A GL String Code containing HLA allele names.</p>
</td></tr>
<tr><td><code id="updateGL_+3A_to">to</code></td>
<td>
<p>A character string identifying the IPD-IMGT/HLA Database release version to translate 'GLSC' to. Values can range from version 1.05.0 to the loaded version of the alleleListHistory object.</p>
</td></tr>
<tr><td><code id="updateGL_+3A_expand">expand</code></td>
<td>
<p>A logical that indicates if the lowest-numbered truncated allele names that match truncated allele names in 'GLSC' should be returned (expand = FALSE), or if a slash-delimited string of all matching full-length allele names should be returned (expand = TRUE). The default value is FALSE.</p>
</td></tr>
<tr><td><code id="updateGL_+3A_verbose">verbose</code></td>
<td>
<p>A logical that indicates if messages regarding the update process should be sent to the console (TRUE) or not (FALSE). The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A GL String Code updated the 'to' release version.
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>
<p>Mack et al. HLA 2023;102(4):501-507 https://doi.org/10.1111/tan.15145
</p>


<h3>Examples</h3>

<pre><code class='language-R'>updateGL(GLSC.ex$GL.String.Code[1],"2.15.0",FALSE,FALSE)
updateGL(GLSC.ex$GL.String.Code[1],"2.15.0",TRUE,FALSE)

</code></pre>

<hr>
<h2 id='validateAllele'>Validate Allele-Name Format and Presence in HLAalignments</h2><span id='topic+validateAllele'></span>

<h3>Description</h3>

<p>Returns TRUE if an allele name is found in HLAalignments in either the 'allele_name' column of full-length allele names or the 'trimmed_allele' column of two-field allele names in the pertinent genomic alignment. Returns FALSE if the allele name is not properly formed, or if the allele name is not found in HLAalignments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validateAllele(allele)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validateAllele_+3A_allele">allele</code></td>
<td>
<p>A character string of the colon-delimited HLA allele name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical identifying if the allele name is present in the alignments (TRUE) or, if it is not in the alignments or is not valid (FALSE).
</p>


<h3>Note</h3>

<p>Messages will be returned to the console if the allele name is malformed, or the locus is invalid; e.g., 'validateAllele(&quot;C\*01:01:01:01&quot;)' or 'validateAllele(&quot;A\*01:01:01:01:01&quot;)'.
</p>
<p>The locus being evaluated must be included in HLAalignments.
</p>

<hr>
<h2 id='validateGLstring'>Validate a GL String</h2><span id='topic+validateGLstring'></span>

<h3>Description</h3>

<p>Evaluates a GL String to identify unsupported characters for a specified GL String version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validateGLstring(GLstring, version)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validateGLstring_+3A_glstring">GLstring</code></td>
<td>
<p>A character string of allele names and operators in the GL String format.</p>
</td></tr>
<tr><td><code id="validateGLstring_+3A_version">version</code></td>
<td>
<p>A character string identifying the GL String version for evaluation. Options are &quot;1.0&quot; and &quot;1.1&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical. TRUE is returned when all characters in a 'version' GL String are permitted. FALSE is returned when forbidden characters are present, or when an incorrect 'version' is specified.
</p>


<h3>References</h3>

<p>Mack et al. HLA 2023;102(2):206-212 https://doi.org/10.1111/tan.15126
</p>


<h3>Examples</h3>

<pre><code class='language-R'>validateGLstring("HLA-A*02:01/HLA-A*02:02+HLA-A*03:01/HLA-A*03:02", version = "1.0")

</code></pre>

<hr>
<h2 id='validateLocus'>Determine if a Locus Name is in the HLAgazeteer</h2><span id='topic+validateLocus'></span>

<h3>Description</h3>

<p>Checks a vector of HLA locus names against the HLAgazeteer to determine if the locus name is valid for a specific type of alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validateLocus(loci, source)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validateLocus_+3A_loci">loci</code></td>
<td>
<p>A character vector of HLA gene names (ex. &quot;DRB1&quot;, c(&quot;DRB1&quot;,&quot;DQB1&quot;)).</p>
</td></tr>
<tr><td><code id="validateLocus_+3A_source">source</code></td>
<td>
<p>A character vector of alignment source types. &quot;AA&quot;, &quot;cDNA&quot;, and &quot;gDNA&quot; are allowed types.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value. TRUE indicates that all of the names and source types are valid. FALSE indicates that at least one locus name or alignment source type is invalid.
</p>


<h3>Note</h3>

<p>The results of this check should only be considered valid for the version of the HLAgazeteer included in the HLAtools package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>validateLocus(loci = "DRB1", source = "AA")
validateLocus(loci = c("V"), source = c("cDNA","gDNA"))
validateLocus(loci = c("E","F","G"), source = "gDNA")
</code></pre>

<hr>
<h2 id='validateMotif'>Determine if a Motif is Properly Formatted</h2><span id='topic+validateMotif'></span>

<h3>Description</h3>

<p>Evaluates a motif to determine if the locus is valid and variants are valid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validateMotif(motif, alignType)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validateMotif_+3A_motif">motif</code></td>
<td>
<p>A character string identifying a sequence variant motif in the following format: Locus*#$~#$~#$, where ## identifies a variant position, and $ identifies the sequence variant. Both nucleotide and peptide motifs can be provided, and any number of variants can be specified.</p>
</td></tr>
<tr><td><code id="validateMotif_+3A_aligntype">alignType</code></td>
<td>
<p>A character string identifying the type of alignment being searched. Allowed values are &quot;codon&quot;,&quot;gen&quot;, nuc&quot; and &quot;prot&quot;. Only one 'alignType' value is allowed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If the motif is valid, TRUE is returned. If the locus or body of the motif is invalid, FALSE is returned along with a brief message.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>validateMotif("A*-21M~2P","prot")
validateMotif("A*196G~301A~3046T","gen")

</code></pre>

<hr>
<h2 id='validateUniformat'>Validate a UNIFORMAT String</h2><span id='topic+validateUniformat'></span>

<h3>Description</h3>

<p>Evaluates a UNIFORMAT string to identify unsupported characters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validateUniformat(uniformat)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validateUniformat_+3A_uniformat">uniformat</code></td>
<td>
<p>A character string of allele names and operators in the UNIFORMAT format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical. TRUE is returned when all characters in 'uniformat' are permitted. FALSE is returned when forbidden characters are present.
</p>


<h3>References</h3>

<p>Nunes Tissue Antigens 2007;69(s1):203-205 https://doi.org/10.1111/j.1399-0039.2006.00808.x
</p>


<h3>Examples</h3>

<pre><code class='language-R'>validateUniformat("A*02:01,A*03:01|A*02:01,A*03:02|A*02:02,A*03:01|A*02:02,A*03:02")

</code></pre>

<hr>
<h2 id='validateVersion'>Validate an IPD-IMGT/HLA Release Version</h2><span id='topic+validateVersion'></span>

<h3>Description</h3>

<p>Determines if an IPD-IMGT/HLA Release Version is referenced in the AlleleListHistory file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validateVersion(version)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validateVersion_+3A_version">version</code></td>
<td>
<p>A character string identifying the release version (branch) of the ANHIG/IMGTHLA Github repository (e.g. '3.53.0'). The value 'Latest' refers to the most recent release.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical. TRUE indicates that data for 'version' is available in the ANHIG/IMGTHLA GitHub repository. FALSE means that data for 'version' is not available.
</p>


<h3>Note</h3>

<p>For internal HLAtools use.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>validateVersion(version = "3.31.1")
validateVersion(version = "3.13.0")

</code></pre>

<hr>
<h2 id='verifyAllele'>Determine if an Allele Name Ever Existed, and (if so) in Which IPD-IMGT/HLA Database Releases</h2><span id='topic+verifyAllele'></span>

<h3>Description</h3>

<p>Returns TRUE if an allele name is present in AlleleListHistory or FALSE it is absent, or c(TRUE,version), where 'version' can be the most recent, original, or all IPD-IMGT/HLA Database releases in which that name appears.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verifyAllele(allele, version = FALSE, all = FALSE, first = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="verifyAllele_+3A_allele">allele</code></td>
<td>
<p>A character string of an HLA allele name. Colon-delimited and field-delimited names are both accepted.</p>
</td></tr>
<tr><td><code id="verifyAllele_+3A_version">version</code></td>
<td>
<p>A logical that indicates if the most recent nomenclature release version in which that name was valid should be returned. The default value is FALSE.</p>
</td></tr>
<tr><td><code id="verifyAllele_+3A_all">all</code></td>
<td>
<p>A logical that indicates if all of the nomenclature release versions in which that name was valid should be returned. The default value is FALSE.</p>
</td></tr>
<tr><td><code id="verifyAllele_+3A_first">first</code></td>
<td>
<p>A logical that indicates if only the original nomenclature release version in which that name was valid should be returned. The default value is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical identifying if the allele name is found in AlleleListHistory (TRUE) or not (FALSE), c(TRUE,version) if version = TRUE.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>verifyAllele("A*01:01:01:01")
verifyAllele("A*01:01:01:01",TRUE)
verifyAllele("A*010101",TRUE,TRUE)
verifyAllele("A*0101",TRUE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
