<!DOCTYPE html><html lang="en"><head><title>Help for package HMPTrees</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HMPTrees}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HMPTrees-package'><p>Object Oriented Data Analysis of Taxonomic Trees</p></a></li>
<li><a href='#checkTreeValidity'><p>Check Validity of an RDP-Based Taxonomic Tree</p></a></li>
<li><a href='#compareTwoDataSets'><p>Likelihood-Ratio-Test Statistics to Compare the Distribution of 2 Sets of RDP-Based Taxonomic Trees</p></a></li>
<li><a href='#createAndPlot'><p>Create and Plot a Tree from a Data Set</p></a></li>
<li><a href='#createTrees'><p>Create a Tree Object</p></a></li>
<li><a href='#displayLegend'><p>Displays Tree Plot Legend</p></a></li>
<li><a href='#formatData'><p>Formats a Data Set</p></a></li>
<li><a href='#generateTree'><p>Generate Test Trees</p></a></li>
<li><a href='#getMLEandLoglike'><p>Get MLE and Log Likelihood of a Data Set</p></a></li>
<li><a href='#HMPTrees-internal'><p>Internal Functions</p></a></li>
<li><a href='#mergeDataSets'><p>Merge Multiple Data Sets</p></a></li>
<li><a href='#pairedCompareTwoDataSets'><p>Likelihood-Ratio-Test Statistics to Compare the Distribution of 2 Paired Sets of RDP-Based Taxonomic Trees</p></a></li>
<li><a href='#plotTree'><p>Plots a Tree Object</p></a></li>
<li><a href='#plotTreeDataMDS'><p>Plot an MDS Plot of a Group of Trees</p></a></li>
<li><a href='#saliva'><p>Saliva Data Set</p></a></li>
<li><a href='#stool'><p>Stool Data Set</p></a></li>
<li><a href='#throat'><p>Throat Data Set</p></a></li>
<li><a href='#trimToTaxaLevel'><p>Trim a Tree to a Given Level</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Statistical Object Oriented Data Analysis of RDP-Based Taxonomic
Trees from Human Microbiome Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-07-05</td>
</tr>
<tr>
<td>Author:</td>
<td>Patricio S. La Rosa, Elena Deych, Berkley Shands, William D. Shannon</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Berkley Shands &lt;rpackages@biorankings.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, HMP, dirmult, doParallel, foreach, parallel, stats,
graphics</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools to model, compare, and visualize populations of taxonomic tree objects.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (== 2.0)</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-07-05 18:36:36 UTC; Berkley</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-07-05 22:08:45 UTC</td>
</tr>
</table>
<hr>
<h2 id='HMPTrees-package'>Object Oriented Data Analysis of Taxonomic Trees</h2><span id='topic+HMPTrees-package'></span><span id='topic+HMPTrees'></span>

<h3>Description</h3>

<p>Object Oriented Data Analysis (OODA) methods to analyze Human Microbiome taxonomic trees directly. 
We provide tools to model, compare, and visualize populations of taxonomic trees.
</p>


<h3>Details</h3>

<p>HMP metagenomic sequences in a sample can be represented as a rooted taxonomic tree. Using supervised taxonomic methods a sequence is matched to a 
hierarchical taxa or taxonomy bins defined in a bacterial-taxonomy library such as, for example, the Ribosomal Database Project (RDP) (Cole, 2005). The 
supervised taxonomic analysis allows us to represent each sample (set of sequences) by a rooted taxonomic tree where the root corresponds to taxon at the 
Kingdom level, i.e., bacteria, and the leaves correspond to the taxa at the Genus level, and the width of the edges (paths) between taxonomic levels correspond 
to the 'abundances' of the descending taxon. <br />
</p>
<p>In particular, we combine RDP matches by adding RDP values of common taxon, which allows us to provide a measure 
of taxa abundance weighting on the confidence of each taxa assignment. The resulting taxonomic trees satisfy the following conditions: i) branches closer to 
the root have higher 'abundance' values than branches closer to leaves, and ii) the sum of the 'abundances' of all descending taxa under a common parent taxon 
cannot be larger than the 'abundance' of the corresponding parent taxon. <br />
</p>
<p>It is important to note that due to how the <code>ape</code> package works the following naming conventions apply to taxa names: 
</p>

<ol>
<li><p> Colons cannot be used in the taxa names at all.
</p>
</li>
<li><p> Each taxa name must be unique - you cannot have two seperate branches both have a child named 'unclassified' for example .
(We took the parent name and added a 'U' to the end to signify an unclassified in our data sets)
</p>
</li>
<li><p> There can only be one top level node. (Bacteria and Archaea cannot both exist unless there is an additional single level above them for example)
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Patricio S. La Rosa, Elena Deych, Berkley Shands, William D. Shannon
</p>


<h3>References</h3>


<ol>
<li><p> Cole JR, Chai B, Farris RJ, Wang Q, Kulam SA, McGarrell DM, Garrity GM, Tiedje JM. The Ribosomal Database Project (RDP-II): 
sequences and tools for high-throughput rRNA analysis. Nucleic Acids Research 2005; 33: D294-D296.
</p>
</li>
<li><p> P. S. La Rosa, Berkley Shands, Elena Deych, Yanjiao Zhou, Erica Sodergren, George Weinstock, and William D. Shannon, 
&quot;Object data analysis of taxonomic trees from human microbiome data,&quot;PLoS ONE 7(11): e48996. doi:10.1371/journal.pone.0048996. Nov. 2012.
</p>
</li>
<li><p> Banks D, Constantine GM. Metric Models for Random Graphs. Journal of Classification 1998; 15: 199-223.
</p>
</li>
<li><p> Shannon WD, Banks D. Combining classification trees using MLE. Stat Med 1999; 18: 727-740.
</p>
</li></ol>


<hr>
<h2 id='checkTreeValidity'>Check Validity of an RDP-Based Taxonomic Tree</h2><span id='topic+checkTreeValidity'></span>

<h3>Description</h3>

<p>This function goes through every node in the tree and for each node it checks that the sum of that nodes children 
are less than or equal to the value of that node.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkTreeValidity(data, samples = NULL, epsilon = 0.0001, split = ".")</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkTreeValidity_+3A_data">data</code></td>
<td>
<p>A data frame in which each column contains the rdp read counts for every taxa given in the row names.</p>
</td></tr>
<tr><td><code id="checkTreeValidity_+3A_samples">samples</code></td>
<td>
<p>Deprecated. Only send the columns in data to create.</p>
</td></tr>
<tr><td><code id="checkTreeValidity_+3A_epsilon">epsilon</code></td>
<td>
<p>This value allows for rounding problems or other such small errors in the data such that the (parent + epsilon &gt; sum(children)).</p>
</td></tr>
<tr><td><code id="checkTreeValidity_+3A_split">split</code></td>
<td>
<p>This is the character that separates the taxa levels in the row names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A boolean vector that indicates the validity of every tree tested.</p>


<h3>Author(s)</h3>

<p>Berkley Shands, Patricio S. La Rosa, Elena Deych, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva) 
	
	validTree &lt;- checkTreeValidity(saliva[,1, drop=FALSE])
	validTree
</code></pre>

<hr>
<h2 id='compareTwoDataSets'>Likelihood-Ratio-Test Statistics to Compare the Distribution of 2 Sets of RDP-Based Taxonomic Trees</h2><span id='topic+compareTwoDataSets'></span>

<h3>Description</h3>

<p>This functions compares the distribution of two sets of RDP-based taxonomic trees using Likelihood-Ratio-Test statistics and a 
p-value is computed using permutations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	compareTwoDataSets(data1, data2, numPerms = 1000, parallel = FALSE, cores = 3, 
	maxSteps=50, delta=10^(-6), numBootStraps = NULL, enableMC = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compareTwoDataSets_+3A_data1">data1</code>, <code id="compareTwoDataSets_+3A_data2">data2</code></td>
<td>
<p>Data frames in which each column contains the rdp read counts for every taxa given in the row names.</p>
</td></tr>
<tr><td><code id="compareTwoDataSets_+3A_numperms">numPerms</code></td>
<td>
<p>The number of permutation tests to run.</p>
</td></tr>
<tr><td><code id="compareTwoDataSets_+3A_parallel">parallel</code></td>
<td>
<p>When this is 'TRUE' it allows for parallel calculation of the permutations. Requires the package <code>doParallel</code>.</p>
</td></tr>
<tr><td><code id="compareTwoDataSets_+3A_cores">cores</code></td>
<td>
<p>The number of parallel processes to run if enableMC is 'TRUE'.</p>
</td></tr>
<tr><td><code id="compareTwoDataSets_+3A_maxsteps">maxSteps</code></td>
<td>
<p>The maximum number of times to iterate though for the MLE.</p>
</td></tr>
<tr><td><code id="compareTwoDataSets_+3A_delta">delta</code></td>
<td>
<p>The minimum threshold of change in f to stop the search for the MLE.</p>
</td></tr>
<tr><td><code id="compareTwoDataSets_+3A_numbootstraps">numBootStraps</code></td>
<td>
<p>Deprecated. Replaced with numPerms.</p>
</td></tr>
<tr><td><code id="compareTwoDataSets_+3A_enablemc">enableMC</code></td>
<td>
<p>Deprecated. Replaced with parallel.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: Both data sets should be standardized to the same number of reads.
</p>
<p>We are interested in assessing whether the distributions from two metagenomic populations are the same or different, which is equivalent to evaluating 
whether their respective parameters are the same or different. The corresponding hypothesis is given as follows:
</p>
<p style="text-align: center;"><code class="reqn">H_{\mathrm{o}}: (g_{1}^{*},\tau_{1}) =  (g_{2}^{*},\tau_{2}) = (g_{0}^{*},\tau_{0}) vs H_{\mathrm{A}}: (g_{1}^{*},\tau_{1}) \neq (g_{2}^{*},\tau_{2}) ,</code>
</p>

<p>where <code class="reqn">(g_{0}^{*},\tau_{0})</code> is the unknown common parameter vector. To evaluate this hypothesis we use the likelihood-ratio test (LRT) which is given by,
</p>
<p style="text-align: center;"><code class="reqn">\lambda = -2 \log\left(\frac{L(g_{o}^{*},\tau_{o};{S_{1n},S_{2m}})}{L(g_{1}^{*},\tau_{1};{S_{1n}})+L(g_{2}^{*},\tau_{2};{S_{2m}})} \right),</code>
</p>

<p>where <code class="reqn">S_{1n}</code> and <code class="reqn">S_{2m}</code> are the sets containing <code class="reqn">n</code> and <code class="reqn">m</code> random samples of trees from each metagenomic population, respectively. 
We assume that the model parameters are unknown under both the null and alternative hypothesis, therefore, we estimate these using the MLE procedure proposed 
in La Rosa et al (see reference 2), and compute the corresponding p-value using non-parametric bootstrap.
</p>


<h3>Value</h3>

<p>A p-value for the similarity of the two data sets based on the permutation test.</p>


<h3>Author(s)</h3>

<p>Patricio S. La Rosa, Elena Deych, Berkley Shands, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva)
	data(stool)
	
	### We use 1 for the number of permutations for computation time
	### This value should be at least 1000 for an accurate result
	numPerms &lt;- 1
	
	pval &lt;- compareTwoDataSets(saliva, stool, numPerms)
	pval
</code></pre>

<hr>
<h2 id='createAndPlot'>Create and Plot a Tree from a Data Set</h2><span id='topic+createAndPlot'></span>

<h3>Description</h3>

<p>This function combines the createTrees and plotTree functions to create and plot a set of trees.</p>


<h3>Usage</h3>

<pre><code class='language-R'>	createAndPlot(data, samples = NULL, level = "genus", colors = NULL, 
		divisions = NULL, main = NULL, sub = "", showTipLabel = TRUE, 
		showNodeLabel = FALSE, displayLegend = TRUE, onePerPage = FALSE, 
		split = ".")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createAndPlot_+3A_data">data</code></td>
<td>
<p>A data frame in which each column contains the rdp read counts for every taxa given in the row names.</p>
</td></tr>
<tr><td><code id="createAndPlot_+3A_samples">samples</code></td>
<td>
<p>Deprecated. Only send the columns in data to plot.</p>
</td></tr>
<tr><td><code id="createAndPlot_+3A_level">level</code></td>
<td>
<p>The depth the tree creation will go down to (kingdom, phylum, class, order, family, genus, species, subspecies).</p>
</td></tr>
<tr><td><code id="createAndPlot_+3A_colors">colors</code></td>
<td>
<p>A vector of colors to be applied to the branches in the plot.</p>
</td></tr>
<tr><td><code id="createAndPlot_+3A_divisions">divisions</code></td>
<td>
<p>A vector of numbers to be used as break points to assign different colors.</p>
</td></tr>
<tr><td><code id="createAndPlot_+3A_main">main</code></td>
<td>
<p>A custom title(s) for the plot(s).</p>
</td></tr>
<tr><td><code id="createAndPlot_+3A_sub">sub</code></td>
<td>
<p>A custom subtitle for the plot.</p>
</td></tr>
<tr><td><code id="createAndPlot_+3A_showtiplabel">showTipLabel</code></td>
<td>
<p>Hides the tip labels if 'FALSE' otherwise it shows all non-zero tip labels.</p>
</td></tr>
<tr><td><code id="createAndPlot_+3A_shownodelabel">showNodeLabel</code></td>
<td>
<p>Hides the interior node labels if 'FALSE' otherwise it shows all non-zero node labels.</p>
</td></tr>
<tr><td><code id="createAndPlot_+3A_displaylegend">displayLegend</code></td>
<td>
<p>Enables the display of a legend of the branch colors and divisions when 'TRUE'.</p>
</td></tr>
<tr><td><code id="createAndPlot_+3A_oneperpage">onePerPage</code></td>
<td>
<p>If 'TRUE' one tree will be plotted per page, if 'FALSE' four will be displayed per page.</p>
</td></tr>
<tr><td><code id="createAndPlot_+3A_split">split</code></td>
<td>
<p>This is the character that separates the taxa levels in the row names.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Notes:
</p>

<ol>
<li><p> For 'level' k, p, c, o, f, g, s and ss can be used in place of kingdom, phylum, class, order, family, genus, species 
and subspecies respectively.
</p>
</li>
<li><p> The values for division should directly relate to the values of your data, i.e. if your data ranges from 0 to 50000 reads you 
should adjust the divisions to fit your data.
</p>
</li></ol>



<h3>Value</h3>

<p>A plot of the tree(s).</p>


<h3>Author(s)</h3>

<p>Berkley Shands, Patricio S. La Rosa, Elena Deych, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva)
	
	### Plots the trees in column 2 and 3 in 'Saliva'
	createAndPlot(saliva[,2:3])
</code></pre>

<hr>
<h2 id='createTrees'>Create a Tree Object</h2><span id='topic+createTrees'></span>

<h3>Description</h3>

<p>This function creates a list tree objects of type 'phylo' for use in plotting the trees.</p>


<h3>Usage</h3>

<pre><code class='language-R'>createTrees(data, samples = NULL, level = "genus", split = ".")</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createTrees_+3A_data">data</code></td>
<td>
<p>A data frame in which each column contains the rdp read counts for every taxa given in the row names.</p>
</td></tr>
<tr><td><code id="createTrees_+3A_samples">samples</code></td>
<td>
<p>Deprecated. Only send the columns in data to create.</p>
</td></tr>
<tr><td><code id="createTrees_+3A_level">level</code></td>
<td>
<p>The depth the tree creation will go down to (kingdom, phylum, class, order, family, genus, species, subspecies).</p>
</td></tr>
<tr><td><code id="createTrees_+3A_split">split</code></td>
<td>
<p>This is the character that separates the taxa levels in the row names.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'level' k, p, c, o, f, g, s and ss can be used in place of kingdom, phylum, class, order, family, genus, 
species and subspecies respectively.
</p>


<h3>Value</h3>

<p>A list of 'phylo' objects that can be passed to plotTree to plot them.</p>


<h3>Author(s)</h3>

<p>Berkley Shands, Patricio S. La Rosa, Elena Deych, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva)
	
	### Creates a tree for the 4th sample in 'Saliva'
	salivaTree &lt;- createTrees(saliva[,4, drop=FALSE])
</code></pre>

<hr>
<h2 id='displayLegend'>Displays Tree Plot Legend</h2><span id='topic+displayLegend'></span>

<h3>Description</h3>

<p>This function displays a legend that shows the tree branch sizes/colors divisions.</p>


<h3>Usage</h3>

<pre><code class='language-R'>displayLegend(colors = NULL, divisions = NULL, title = "Confidence Value")</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="displayLegend_+3A_colors">colors</code></td>
<td>
<p>A vector of colors to be used in the plot from lowest ranking to highest ranking.</p>
</td></tr>
<tr><td><code id="displayLegend_+3A_divisions">divisions</code></td>
<td>
<p>A vector of numbers from lowest to highest to separate the tree branches into the color ranking.</p>
</td></tr>
<tr><td><code id="displayLegend_+3A_title">title</code></td>
<td>
<p>The title for the legend.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The values for division should directly relate to the values of your data, i.e. if your data ranges from 0 
to 50000 reads you should adjust the divisions to fit your data.
</p>


<h3>Value</h3>

<p>A blank plot that contains a legend.</p>


<h3>Author(s)</h3>

<p>Berkley Shands, Patricio S. La Rosa, Elena Deych, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	displayLegend(c("red", "orange", "blue"), c(.1, 100, 10000))
</code></pre>

<hr>
<h2 id='formatData'>Formats a Data Set</h2><span id='topic+formatData'></span>

<h3>Description</h3>

<p>This function will take a data set and format it by removing low count trees, and/or normalizing counts.</p>


<h3>Usage</h3>

<pre><code class='language-R'>formatData(data, countThreshold = 1000, normalizeThreshold = 10000)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="formatData_+3A_data">data</code></td>
<td>
<p>A data frame in which each column contains the rdp read counts for every taxa given in the row names.</p>
</td></tr>
<tr><td><code id="formatData_+3A_countthreshold">countThreshold</code></td>
<td>
<p>A cut off threshold for reads - all trees with fewer than this number of reads will be removed.</p>
</td></tr>
<tr><td><code id="formatData_+3A_normalizethreshold">normalizeThreshold</code></td>
<td>
<p>All the trees that are not removed will be normalized to this many reads.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When removing trees with too few reads, the cuts off is based on the value of the top level node, not the sum of all the reads in a sample.
</p>


<h3>Value</h3>

<p>A new data set that is trimmed and standardized based on the specified parameters. The new data is also reordered alphabetically according to row labels.</p>


<h3>Author(s)</h3>

<p>Patricio S. La Rosa, Elena Deych, Berkley Shands, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva)
	
	saliva2 &lt;- formatData(saliva, 1000, 10000)
</code></pre>

<hr>
<h2 id='generateTree'>Generate Test Trees</h2><span id='topic+generateTree'></span>

<h3>Description</h3>

<p>This function will take several initial trees and will randomly populate new trees based on the originals.</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateTree(data, numReadsPerSamp, theta = NULL, level = "genus", split = ".")</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generateTree_+3A_data">data</code></td>
<td>
<p>A data frame in which each column contains the rdp read counts for every taxa given in the row names.</p>
</td></tr>
<tr><td><code id="generateTree_+3A_numreadspersamp">numReadsPerSamp</code></td>
<td>
<p>A vector specifying the number of reads or sequence depth for each sample.</p>
</td></tr>
<tr><td><code id="generateTree_+3A_theta">theta</code></td>
<td>
<p>When theta is not NULL the base tree is generated by using the <code>dirmult</code> function.</p>
</td></tr>
<tr><td><code id="generateTree_+3A_level">level</code></td>
<td>
<p>The depth the tree will go down to (kingdom, phylum, class, order, family, genus, species, subspecies).</p>
</td></tr>
<tr><td><code id="generateTree_+3A_split">split</code></td>
<td>
<p>This is the character that separates the taxa levels in the row names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the generated tree(s).</p>


<h3>Author(s)</h3>

<p>Patricio S. La Rosa, Elena Deych, Berkley Shands, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva)
	
	### Generate a the number of reads per sample
	### The first number is the number of reads and the second is the number of subjects
	nrs &lt;- rep(10000, 2)
	
	gendata &lt;- generateTree(saliva, nrs)
</code></pre>

<hr>
<h2 id='getMLEandLoglike'>Get MLE and Log Likelihood of a Data Set</h2><span id='topic+getMLEandLoglike'></span>

<h3>Description</h3>

<p>This function takes a data set and computes the MLE and its Log-Likelihood value.</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMLEandLoglike(data, maxSteps = 50, weightCols = NULL, delta = 10^(-6), weight = NULL)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getMLEandLoglike_+3A_data">data</code></td>
<td>
<p>A data frame in which each column contains the rdp read counts for every taxa given in the row names.</p>
</td></tr>
<tr><td><code id="getMLEandLoglike_+3A_maxsteps">maxSteps</code></td>
<td>
<p>The maximum number of times to iterate though for the MLE.</p>
</td></tr>
<tr><td><code id="getMLEandLoglike_+3A_weightcols">weightCols</code></td>
<td>
<p>A vector of weights for the subjects.</p>
</td></tr>
<tr><td><code id="getMLEandLoglike_+3A_delta">delta</code></td>
<td>
<p>The minimum threshold of change in f to stop the search for the MLE.</p>
</td></tr>
<tr><td><code id="getMLEandLoglike_+3A_weight">weight</code></td>
<td>
<p>Deprecated, use weightCols instead</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A unimodal probability model for graph-valued random objects has been derived and applied previously to several types of graphs 
(cluster trees, digraphs, and classification and regression trees) (For example, Banks and Constantine, 1998; Shannon and Banks, 1999). 
Here we apply this model to HMP trees constructed from RDP matches.  Let <code class="reqn">G</code> be the finite set of taxonomic trees with elements 
<code class="reqn">g</code>, and <code class="reqn">d: G \times G \to R^{+}</code> an arbitrary metric of distance on <code class="reqn">G</code>. We have the probability measure <code class="reqn">H(g^{*},\tau)</code> defined by 
</p>
<p style="text-align: center;"><code class="reqn">P(g;g^{*},\tau) = c(g^{*},\tau) \exp(-\tau d(g^{*},g) ), for all g \in G,</code>
</p>

<p>where <code class="reqn">g^{*}</code>  is the modal or central tree, <code class="reqn">\tau</code> is a concentration parameter, and <code class="reqn">c(g^{*},\tau)</code> is the normalization constant.  
The distance measure between two trees is the Euclidean norm of the difference between their corresponding adjacency-vectors. To estimate the parameters 
<code class="reqn">(g^{*},\tau)</code>, we use the maximum likelihood estimate (MLE) procedure described in La Rosa et al. (see reference 2)
</p>


<h3>Value</h3>

<p>A list containing the MLE, log-likelihood, tau, the number of iterations it took to run, and some intermediate values</p>


<h3>Author(s)</h3>

<p>Patricio S. La Rosa, Elena Deych, Berkley Shands, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva)
	
	### We use 1 for the maximum number of steps for computation time
	### This value should be much higher to ensure an accurate result
	numSteps &lt;- 1
	
	mle &lt;- getMLEandLoglike(saliva, numSteps)$mleTree
</code></pre>

<hr>
<h2 id='HMPTrees-internal'>Internal Functions</h2><span id='topic+HMPTrees-internal'></span><span id='topic+buildTree'></span><span id='topic+checkTreeValidHelp'></span><span id='topic+getBranchSizes'></span><span id='topic+getTaxaDepth'></span><span id='topic+removeUnclass'></span><span id='topic+subsetData'></span><span id='topic+traverseTree'></span><span id='topic+traverseTreeHelp'></span><span id='topic+transformHMPTreetoHMP'></span><span id='topic+transformHMPtoHMPTree'></span>

<h3>Description</h3>

<p>These files are only called from inside other functions and are therefore not documented here.</p>


<h3>Author(s)</h3>

<p>Patricio S. La Rosa, Elena Deych, Berkley Shands, William D. Shannon</p>

<hr>
<h2 id='mergeDataSets'>Merge Multiple Data Sets</h2><span id='topic+mergeDataSets'></span>

<h3>Description</h3>

<p>This function can take any number of data sets, calculate their individual and combined MLEs and then merge them.</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeDataSets(dataList, calcMLE = FALSE, uniqueNames = FALSE, data = NULL)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mergeDataSets_+3A_datalist">dataList</code></td>
<td>
<p>A list of data frames in which each column contains the rdp read counts for every taxa given in the row names.</p>
</td></tr>
<tr><td><code id="mergeDataSets_+3A_calcmle">calcMLE</code></td>
<td>
<p>If 'FALSE' the MLEs for the data sets will not be calculated, otherwise they are added to the end.</p>
</td></tr>
<tr><td><code id="mergeDataSets_+3A_uniquenames">uniqueNames</code></td>
<td>
<p>If 'TRUE' the column names in the combined data set will be appended to insure uniqueness, otherwise the column names
will follow the naming process from the <code>merge</code> function.</p>
</td></tr>
<tr><td><code id="mergeDataSets_+3A_data">data</code></td>
<td>
<p>Deprecated. Replaced with dataList.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Although not required, all data sets should be standardized to the same number of reads before merging.
</p>


<h3>Value</h3>

<p>A single data set containing all the data from the input data sets, in addition to their individual MLEs and a combined MLE if requested.</p>


<h3>Author(s)</h3>

<p>Berkley Shands, Patricio S. La Rosa, Elena Deych, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva)
	data(stool)
	
	dataComb &lt;- mergeDataSets(list(saliva, stool))
</code></pre>

<hr>
<h2 id='pairedCompareTwoDataSets'>Likelihood-Ratio-Test Statistics to Compare the Distribution of 2 Paired Sets of RDP-Based Taxonomic Trees</h2><span id='topic+pairedCompareTwoDataSets'></span>

<h3>Description</h3>

<p>This functions compares the distribution of two paired sets of RDP-based taxonomic trees using Likelihood-Ratio-Test statistics and a 
p-value is computed using permutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	pairedCompareTwoDataSets(data1, data2, numPerms = 1000, parallel = FALSE, 
		cores = 3, maxSteps=50, delta=10^(-6))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pairedCompareTwoDataSets_+3A_data1">data1</code>, <code id="pairedCompareTwoDataSets_+3A_data2">data2</code></td>
<td>
<p>Data frames in which each column contains the rdp read counts for every taxa given in the row names.</p>
</td></tr>
<tr><td><code id="pairedCompareTwoDataSets_+3A_numperms">numPerms</code></td>
<td>
<p>Number of permutations.  In practice this should be at least 1,000.</p>
</td></tr>
<tr><td><code id="pairedCompareTwoDataSets_+3A_parallel">parallel</code></td>
<td>
<p>When this is 'TRUE' it allows for parallel calculation of the permutations. Requires the package <code>doParallel</code>.</p>
</td></tr>
<tr><td><code id="pairedCompareTwoDataSets_+3A_cores">cores</code></td>
<td>
<p>The number of parallel processes to run if parallel is 'TRUE'.</p>
</td></tr>
<tr><td><code id="pairedCompareTwoDataSets_+3A_maxsteps">maxSteps</code></td>
<td>
<p>The maximum number of times to iterate though for the MLE.</p>
</td></tr>
<tr><td><code id="pairedCompareTwoDataSets_+3A_delta">delta</code></td>
<td>
<p>The minimum threshold of change in f to stop the search for the MLE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: Both data sets should be standardized to the same number of reads.
</p>


<h3>Value</h3>

<p>A p-value for the similarity of the two data sets based on the permutation test.</p>


<h3>Author(s)</h3>

<p>Patricio S. La Rosa, Elena Deych, Berkley Shands, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva)
	data(stool)
	
	### We use 1 for the number of permutations for computation time
	### This value should be at least 1000 for an accurate result
	numPerms &lt;- 1
	
	pval &lt;- pairedCompareTwoDataSets(saliva, stool, numPerms)
	pval
</code></pre>

<hr>
<h2 id='plotTree'>Plots a Tree Object</h2><span id='topic+plotTree'></span>

<h3>Description</h3>

<p>This function takes one or more 'phylo' objects and plots them.</p>


<h3>Usage</h3>

<pre><code class='language-R'>	plotTree(treeList, colors = NULL, divisions = NULL, main = NULL, sub = "", 
		showTipLabel = TRUE, showNodeLabel = FALSE, displayLegend = TRUE, 
		trees = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotTree_+3A_treelist">treeList</code></td>
<td>
<p>A list that contains at least one tree of type 'phylo'.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_colors">colors</code></td>
<td>
<p>A vector of colors to be applied to the branches in the plot.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_divisions">divisions</code></td>
<td>
<p>A vector of numbers to be used as break points to assign different colors.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_main">main</code></td>
<td>
<p>A custom title(s) for the plot(s).</p>
</td></tr>
<tr><td><code id="plotTree_+3A_sub">sub</code></td>
<td>
<p>A custom subtitle for the plot.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_showtiplabel">showTipLabel</code></td>
<td>
<p>Hides the tip labels if 'FALSE' otherwise it shows all non-zero tip labels.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_shownodelabel">showNodeLabel</code></td>
<td>
<p>Hides the interior node labels if 'FALSE' otherwise it shows all non-zero node labels.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_displaylegend">displayLegend</code></td>
<td>
<p>Enables the display of a legend of the branch colors and divisions when 'TRUE'.</p>
</td></tr>
<tr><td><code id="plotTree_+3A_trees">trees</code></td>
<td>
<p>Deprecated. Replaced with treeList.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Notes: 
</p>

<ol>
<li><p> The <code>phylo</code> type is a product of the <code>ape</code> package and the <code>createTrees</code> function in this package 
produces a list of <code>phylo</code> type objects for use with this function.  
</p>
</li>
<li><p> The values for division should directly relate to the values of your data, i.e. if your data ranges from 
0 to 50000 reads you should adjust the divisions to fit your data.
</p>
</li></ol>



<h3>Value</h3>

<p>A plot of the tree(s).</p>


<h3>Author(s)</h3>

<p>Berkley Shands, Patricio S. La Rosa, Elena Deych, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva)
	
	### Creates a tree for the 4th sample in 'Saliva' then plots it
	salivaTree &lt;- createTrees(saliva[,4, drop=FALSE])
	plotTree(salivaTree, displayLegend=FALSE)
</code></pre>

<hr>
<h2 id='plotTreeDataMDS'>Plot an MDS Plot of a Group of Trees</h2><span id='topic+plotTreeDataMDS'></span>

<h3>Description</h3>

<p>This function can take any number of data sets and plots them on an MDS plot to show relative closeness to one another.</p>


<h3>Usage</h3>

<pre><code class='language-R'>	plotTreeDataMDS(dataList, main = "Tree MDS Comparisons", calcMLE = TRUE, 
		mleTitles = NULL, dotColors = NULL, dotSizes = NULL, showNames = FALSE, 
		returnCoords = FALSE, data = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotTreeDataMDS_+3A_datalist">dataList</code></td>
<td>
<p>A list of a data frames in which each column contains the rdp read counts for every taxa given in the row names.</p>
</td></tr>
<tr><td><code id="plotTreeDataMDS_+3A_main">main</code></td>
<td>
<p>A title for the MDS plot.</p>
</td></tr>
<tr><td><code id="plotTreeDataMDS_+3A_calcmle">calcMLE</code></td>
<td>
<p>If 'FALSE' the MLEs for the data sets will not be calculated and plotted.</p>
</td></tr>
<tr><td><code id="plotTreeDataMDS_+3A_mletitles">mleTitles</code></td>
<td>
<p>Deprecated. Replaced with the names in 'dataList'.</p>
</td></tr>
<tr><td><code id="plotTreeDataMDS_+3A_dotcolors">dotColors</code></td>
<td>
<p>The colors to be used when plotting the points and MLE points on the MDS plot.</p>
</td></tr>
<tr><td><code id="plotTreeDataMDS_+3A_dotsizes">dotSizes</code></td>
<td>
<p>A vector in which the first value is the data points CEX and the second value is the MLEs CEX.</p>
</td></tr>
<tr><td><code id="plotTreeDataMDS_+3A_shownames">showNames</code></td>
<td>
<p>When 'TRUE' the column name will be plotted above each corresponding point.</p>
</td></tr>
<tr><td><code id="plotTreeDataMDS_+3A_returncoords">returnCoords</code></td>
<td>
<p>When 'TRUE' this function will return the x and y coordinates for every plotted point.</p>
</td></tr>
<tr><td><code id="plotTreeDataMDS_+3A_data">data</code></td>
<td>
<p>Deprecated. Replaced with dataList.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A MDS plot of the data.</p>


<h3>Author(s)</h3>

<p>Berkley Shands, Patricio S. La Rosa, Elena Deych, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva)
	data(stool)
	
	plotTreeDataMDS(list(Saliva=saliva, Stool=stool))
</code></pre>

<hr>
<h2 id='saliva'>Saliva Data Set</h2><span id='topic+saliva'></span>

<h3>Description</h3>

<p>A data set containing all taxa from 24 subjects.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(saliva)</code></pre>


<h3>Format</h3>

<p>The format is a data frame of 454 rows by 24 columns, with each column being a separate subject and each 
row being a different taxa denoted by the row names.  The taxanomical levels are separated by a '.' in 
their names (Bacteria.Phylum.Class....).  The values in each column are the sum of values that each taxa
had in an RDP file.  It should also be noted that the samples are normalized to 7000 reads and any level 
that ends with a U was unclassified in the RDP file.
</p>

<hr>
<h2 id='stool'>Stool Data Set</h2><span id='topic+stool'></span>

<h3>Description</h3>

<p>A data set containing all taxa from 24 subjects.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(stool)</code></pre>


<h3>Format</h3>

<p>The format is a data frame of 371 rows by 24 columns, with each column being a separate subject and each 
row being a different taxa denoted by the row names.  The taxanomical levels are separated by a '.' in 
their names (Bacteria.Phylum.Class....).  The values in each column are the sum of values that each taxa
had in an RDP file.  It should also be noted that the samples are normalized to 7000 reads and any level 
that ends with a U was unclassified in the RDP file.
</p>

<hr>
<h2 id='throat'>Throat Data Set</h2><span id='topic+throat'></span>

<h3>Description</h3>

<p>A data set containing all taxa from 22 subjects.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(throat)</code></pre>


<h3>Format</h3>

<p>The format is a data frame of 529 rows by 22 columns, with each column being a separate subject and each 
row being a different taxa denoted by the row names.  The taxanomical levels are separated by a '.' in 
their names (Bacteria.Phylum.Class....).  The values in each column are the sum of values that each taxa
had in an RDP file. It should also be noted that the samples have not been normalized and should be 
used with 'formatData'.  Also any level that ends with a U was unclassified in the RDP file.
</p>

<hr>
<h2 id='trimToTaxaLevel'>Trim a Tree to a Given Level</h2><span id='topic+trimToTaxaLevel'></span>

<h3>Description</h3>

<p>This function will take a tree and either remove all nodes lower than the given 
level or will remove all nodes not of the given level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	trimToTaxaLevel(data, level = "genus", eliminateParentNodes = FALSE, 
		trimBelow = NULL, split = ".")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trimToTaxaLevel_+3A_data">data</code></td>
<td>
<p>A data frame in which each column contains the rdp read counts for every taxa given in the row names.</p>
</td></tr>
<tr><td><code id="trimToTaxaLevel_+3A_level">level</code></td>
<td>
<p>The depth the tree will go down to (kingdom, phylum, class, order, family, genus, species, subspecies).</p>
</td></tr>
<tr><td><code id="trimToTaxaLevel_+3A_eliminateparentnodes">eliminateParentNodes</code></td>
<td>

<p>If 'TRUE' the data set returned will only contain rows at the level specified by 'myTaxaLevel'.
If 'FALSE' the data set returned will contain all the nodes up to the level specified by 'myTaxaLevel'.</p>
</td></tr>
<tr><td><code id="trimToTaxaLevel_+3A_trimbelow">trimBelow</code></td>
<td>

<p>If 'NULL' the function will pull out only the data at the level specified by 'myTaxaLevel'.
If 'TRUE' the function will remove all the levels below the specified level.
If 'FALSE' the function will remove all the levels above the specified level.
</p>
</td></tr>
<tr><td><code id="trimToTaxaLevel_+3A_split">split</code></td>
<td>
<p>This is the character that separates the taxa levels in the row names.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Notes: 
</p>

<ol>
<li><p> For 'level' k, p, c, o, f, g, s and ss can be used in place of kingdom, phylum, class, order, family, genus, species 
and subspecies respectively.
</p>
</li>
<li><p> Numbers can also be used for 'level', with no maximum limit.
</p>
</li>
<li><p> The option to 'eliminateParentNodes' only works when 'trimBelow' is NULL.
</p>
</li></ol>



<h3>Value</h3>

<p>A new data set that has been trimmed to the level selected.</p>


<h3>Author(s)</h3>

<p>Berkley Shands, Patricio S. La Rosa, Elena Deych, William D. Shannon</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(saliva)
	
	### Trims saliva to only contain the class level 
	salivaClass &lt;- trimToTaxaLevel(saliva, "class", TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
