<!DOCTYPE html><html lang="en"><head><title>Help for package HodgesTools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HodgesTools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#append_section_to_ini'><p>Append section to ini file</p></a></li>
<li><a href='#cpg_analysis'><p>CpG Analysis</p></a></li>
<li><a href='#createManhattandQQ'><p>Creating a Manhattan Plot and QQ plot</p></a></li>
<li><a href='#helper_collapseTableByLevenSim'><p>helper_collapseTableByLevenSim</p></a></li>
<li><a href='#helper_getMaxLevenSimCol'><p>helper_getMaxLevenSimCol</p></a></li>
<li><a href='#helper_makeBigTableFromListOfStandardTables'><p>makeBigTableFromListofTables</p></a></li>
<li><a href='#plot_HOMERTFs'><p>Plot HOMER TF enrichment results</p></a></li>
<li><a href='#read_bed'><p>Read bed file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Common Use Tools for Genomic Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Built by Hodges lab members for current and future Hodges lab members. Other individuals are welcome to use as well. Provides useful functions that the lab uses everyday to analyze various genomic datasets. Critically, only general use functions are provided; functions specific to a given technique are reserved for a separate package. As the lab grows, we expect to continue adding functions to the package to build on previous lab members code. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, ggplot2, magrittr, RColorBrewer, readr, ini, qqman,
data.table, RecordLinkage</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-05 16:08:26 UTC; tyler</td>
</tr>
<tr>
<td>Author:</td>
<td>Tyler Hansen [aut, cre, cph],
  Tim Scott [aut, ctb, cph],
  Lindsey Guerin [aut, ctb, cph],
  Verda Agan [aut, ctb, cph],
  Emily Hodges [aut, fnd, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tyler Hansen &lt;tyler.j.hansen@vanderbilt.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-08 13:52:36 UTC</td>
</tr>
</table>
<hr>
<h2 id='append_section_to_ini'>Append section to ini file</h2><span id='topic+append_section_to_ini'></span>

<h3>Description</h3>

<p>Takes a new section in ini format and adds to existing ini.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>append_section_to_ini(ini_file, new_section)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="append_section_to_ini_+3A_ini_file">ini_file</code></td>
<td>
<p>file location of config.ini file</p>
</td></tr>
<tr><td><code id="append_section_to_ini_+3A_new_section">new_section</code></td>
<td>
<p>named list of the section list</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The new_section must be a named list of the section list. See examples.
</p>


<h3>Value</h3>

<p>No return value. Edits and overwrites input config.ini file.
</p>


<h3>Author(s)</h3>

<p>Tyler Hansen
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#list of key-value pairs
CHRACC &lt;- list(dir='/chrAcc_peaks/',
               peaks='/chrAcc_peaks/GM12878_genrich.narrowPeak')

#list of section, resulting in list of list.
new_section &lt;- list(CHRACC=CHRACC)

#write ini
ini_file &lt;- system.file("extdata", "config.ini")
append_section_to_ini(ini_file, new_section)

</code></pre>

<hr>
<h2 id='cpg_analysis'>CpG Analysis</h2><span id='topic+cpg_analysis'></span>

<h3>Description</h3>

<p>Compute observed/expected CpG ratio and GC% for regions of interest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpg_analysis(
  list = FALSE,
  count,
  cpg_file,
  nuc_file,
  palette = "Set3",
  plot = "none"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cpg_analysis_+3A_list">list</code></td>
<td>
<p>&quot;boolean of whether input is a list of groups. Default = FALSE.&quot;</p>
</td></tr>
<tr><td><code id="cpg_analysis_+3A_count">count</code></td>
<td>
<p>&quot;numeric value for the number of files included in your list</p>
</td></tr>
<tr><td><code id="cpg_analysis_+3A_cpg_file">cpg_file</code></td>
<td>
<p>&quot;file names or list of files names for your CpGcount.txt files. This is defined in cpg_analysis.sh&quot;</p>
</td></tr>
<tr><td><code id="cpg_analysis_+3A_nuc_file">nuc_file</code></td>
<td>
<p>&quot;file names or list of files names for your nucOutput_gc.txt files. This is defined in cpg_analysis.sh&quot;</p>
</td></tr>
<tr><td><code id="cpg_analysis_+3A_palette">palette</code></td>
<td>
<p>&quot;if choosing to plot, the RColorBrewer palette you would like to be applied to your plot&quot;</p>
</td></tr>
<tr><td><code id="cpg_analysis_+3A_plot">plot</code></td>
<td>
<p>&quot;one of three choices depending on what output you would like: 'none' for no plot, 'ratio' for observed/expected ratios, 'gc_percent' for GC%&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function reads in a nucOutput_gc and CpGcount text file
The function uses the nucOutput_gc and CpGcount file to calculates observed/expected ratio and GC%.
The function allows the option to plot the distribution of these values in ggplot2
</p>


<h3>Value</h3>

<p>ggplot object or tibble if plot=&quot;none&quot;
</p>


<h3>Author(s)</h3>

<p>Lindsey Guerin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load external data

gain_6hr_CpG &lt;- system.file(package = "HodgesTools", "extdata",
"cov5root_6hr_gain.CpGcount.txt")
gain_12hr_CpG &lt;- system.file(package = "HodgesTools", "extdata",
 "cov5root_12hr_gain.CpGcount.txt")
gain_6hr_nuc &lt;- system.file(package = "HodgesTools", "extdata",
 "cov5root_6hr_gain.nucOutput_gc.txt")
gain_12hr_nuc &lt;- system.file(package = "HodgesTools", "extdata",
"cov5root_12hr_gain.nucOutput_gc.txt")

#Make a density plot of GC% values for a list of two region of interest files
cpg_analysis(list = TRUE, count = 2, cpg_file = list(gain_6hr_CpG, gain_12hr_CpG),
nuc_file= list(gain_6hr_nuc, gain_12hr_nuc), palette = "Set3", plot ="gc_percent")

#Make a density plot of observed/expected values for a single set of regions of interest
cpg_analysis(list = FALSE, cpg_file = gain_6hr_CpG,
nuc_file = gain_6hr_nuc, palette = "Set3", plot ="ratio")
</code></pre>

<hr>
<h2 id='createManhattandQQ'>Creating a Manhattan Plot and QQ plot</h2><span id='topic+createManhattandQQ'></span>

<h3>Description</h3>

<p>Creates a Manhattan plot and QQ plot using GWAS results output from PLINK
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createManhattandQQ(
  gwas_results,
  highlights_file = NULL,
  suggestive_line = -log10(0.05),
  set_color_vector = c("gray10", "gray60"),
  genomewide_line = -log10(5e-08),
  annotate_Pval = 0.05,
  y_lim = c(0, 8)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createManhattandQQ_+3A_gwas_results">gwas_results</code></td>
<td>
<p>output file listing SNP-trait association values for GWAS run using PLINK</p>
</td></tr>
<tr><td><code id="createManhattandQQ_+3A_highlights_file">highlights_file</code></td>
<td>
<p>a text file with a 'snp' column listing the SNPs to annotate/color on the Manhattan plot</p>
</td></tr>
<tr><td><code id="createManhattandQQ_+3A_suggestive_line">suggestive_line</code></td>
<td>
<p>where to draw a &quot;suggestive&quot; line; default -log10(1e-5).</p>
</td></tr>
<tr><td><code id="createManhattandQQ_+3A_set_color_vector">set_color_vector</code></td>
<td>
<p>a character vector listing colors in palette of interest (you must create this chr object before calling the createManhattanandQQ function and assign it to set_color_vector)</p>
</td></tr>
<tr><td><code id="createManhattandQQ_+3A_genomewide_line">genomewide_line</code></td>
<td>
<p>where to draw a &quot;genome-wide significant&quot; line; default -log10(5e-8)</p>
</td></tr>
<tr><td><code id="createManhattandQQ_+3A_annotate_pval">annotate_Pval</code></td>
<td>
<p>if set, SNPs below this p-value will be annotated on the plot; default is 0.05</p>
</td></tr>
<tr><td><code id="createManhattandQQ_+3A_y_lim">y_lim</code></td>
<td>
<p>set the y-axis limits; default is c(0,8)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function reads in a GWAS result file output from plink2 listing the coordinates, ids, and associated p-values for SNPs under study
This function also has the option of reading in a &quot;highlights&quot; file listing the IDs of SNPs to annotate/color on the Manhattan plot
</p>


<h3>Value</h3>

<p>a Manhattan plot of SNP-trait associations and QQ plot
</p>


<h3>Author(s)</h3>

<p>Verda Agan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#' #load external data.
gwas_results &lt;- system.file(package = "HodgesTools", "extdata",
"createManhattandQQ_example_sum_stats.txt")
snps_to_annotate &lt;- system.file(package = "HodgesTools", "extdata",
"createManhattandQQ_example_highlights_file.txt")

#Make a Manhattan plot that highlights a select list of SNPs subset from GWAS results
createManhattandQQ(gwas_results, highlights_file=snps_to_annotate,
suggestive_line = -log10(0.001), set_color_vector = c("gray10", "gray60"),
genomewide_line = -log10(5e-8), annotate_Pval = 0.001, y_lim =c(0,8))

#Make a Manhattan plot that doesn't highlight a select list of SNPs subset from GWAS results
createManhattandQQ(gwas_results, suggestive_line = -log10(0.001),
set_color_vector = c("gray10", "gray60"), genomewide_line = -log10(5e-8),
annotate_Pval = 0.001, y_lim =c(0,8))
</code></pre>

<hr>
<h2 id='helper_collapseTableByLevenSim'>helper_collapseTableByLevenSim</h2><span id='topic+helper_collapseTableByLevenSim'></span>

<h3>Description</h3>

<p>Reads in a table and value for Levenshtein threshold and returns a table collapsed by threshold (highest p-value for each group)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>helper_collapseTableByLevenSim(inputTable, levenSimThresholdVal)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="helper_collapseTableByLevenSim_+3A_inputtable">inputTable</code></td>
<td>
<p>dataframe. HOMER output table modified in the parent script&ndash;ready for filtering by Levenshtein similarity.</p>
</td></tr>
<tr><td><code id="helper_collapseTableByLevenSim_+3A_levensimthresholdval">levenSimThresholdVal</code></td>
<td>
<p>float. Value for thresholding TFs. For groups of TFs with similar consensus sequences, the TF with the lowest p-value by HOMER will be retained.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tibble
</p>


<h3>Author(s)</h3>

<p>Tim Scott
</p>

<hr>
<h2 id='helper_getMaxLevenSimCol'>helper_getMaxLevenSimCol</h2><span id='topic+helper_getMaxLevenSimCol'></span>

<h3>Description</h3>

<p>Reads in a vector of motifs and returns a
</p>


<h3>Usage</h3>

<pre><code class='language-R'>helper_getMaxLevenSimCol(vectorOfMotifs)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="helper_getMaxLevenSimCol_+3A_vectorofmotifs">vectorOfMotifs</code></td>
<td>
<p>vector of char. Vector of motifs to filter through.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Tim Scott
</p>

<hr>
<h2 id='helper_makeBigTableFromListOfStandardTables'>makeBigTableFromListofTables</h2><span id='topic+helper_makeBigTableFromListOfStandardTables'></span>

<h3>Description</h3>

<p>Reads in a list of tables and return list of tables with percent Fold Change (enrichment)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>helper_makeBigTableFromListOfStandardTables(inputListOfTables)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="helper_makeBigTableFromListOfStandardTables_+3A_inputlistoftables">inputListOfTables</code></td>
<td>
<p>list of dataframe. List of HOMER TF knownResults Tables.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of tibble
</p>


<h3>Author(s)</h3>

<p>Tim Scott
</p>

<hr>
<h2 id='plot_HOMERTFs'>Plot HOMER TF enrichment results</h2><span id='topic+plot_HOMERTFs'></span>

<h3>Description</h3>

<p>Plot HOMER TF enrichment results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_HOMERTFs(
  dir = "/directory/of/results/",
  show = 3,
  qThreshold = 0.05,
  levenSimThreshold = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_HOMERTFs_+3A_dir">dir</code></td>
<td>
<p>string. Input directory containing HOMER findMotifsGenome.pl output files in format: *knownResults.txt</p>
</td></tr>
<tr><td><code id="plot_HOMERTFs_+3A_show">show</code></td>
<td>
<p>int. Number of rows to show per input file, ranked by p-value.</p>
</td></tr>
<tr><td><code id="plot_HOMERTFs_+3A_qthreshold">qThreshold</code></td>
<td>
<p>int. Value for thresholding HOMER enrichment results by q-value.</p>
</td></tr>
<tr><td><code id="plot_HOMERTFs_+3A_levensimthreshold">levenSimThreshold</code></td>
<td>
<p>float. Value for thresholding TFs. For groups of TFs with similar consensus sequences, the TF with the lowest p-value by HOMER will be retained.</p>
</td></tr>
</table>


<h3>Details</h3>



<h4>Make bigTable of all TFs to pull from so a single TF can have data from different input Files (e.g. across CTs)</h4>

<p>(5) Create bigTable of all q-value TF results concatenated together
(6) Filter by consensus list (4)and make a gg-plot appropriate table and Plot
(7-8) Order factors and plot
</p>



<h3>Value</h3>

<p>ggplot object
</p>


<h3>Strategy:</h3>



<h4>Find motifs to extract</h4>

<p>(1) Filter each element table by q-value. Should basically chop off the bottom portion of list, making the rest of the script less computationally cumbersome
(2) Collapse each element table by by Levenshtein Similarity
(3) Filter each element table (in my case: cell type-specific results file) to top X rows
(4) Extract consensus columns from each element table and store as a variable]
</p>



<h3>Author(s)</h3>

<p>Tim Scott
</p>

<hr>
<h2 id='read_bed'>Read bed file</h2><span id='topic+read_bed'></span>

<h3>Description</h3>

<p>Reads in a tab-delimited BED formatted file into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_bed(file, extra_col_names = c(), length = FALSE, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_bed_+3A_file">file</code></td>
<td>
<p>bed file</p>
</td></tr>
<tr><td><code id="read_bed_+3A_extra_col_names">extra_col_names</code></td>
<td>
<p>list of strings specifying extra column names</p>
</td></tr>
<tr><td><code id="read_bed_+3A_length">length</code></td>
<td>
<p>boolean of whether to add length column</p>
</td></tr>
<tr><td><code id="read_bed_+3A_verbose">verbose</code></td>
<td>
<p>boolean set to see function behavior</p>
</td></tr>
</table>


<h3>Details</h3>

<p>First three columns of file must be the genomic coordinates of the regions (i.e. chr start end).
</p>
<p>read_bed will auto-detect BED3 and BED6 formats. It will also detect BED3+ and BED6+ formats assigning generic or user-defined col_names to the additional column(s).
</p>


<h3>Value</h3>

<p>tibble
</p>


<h3>Author(s)</h3>

<p>Tyler Hansen &amp; Tim Scott
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load external data.
BED3 &lt;- system.file(package = "HodgesTools", "extdata", "test_BED3.bed")
BED6 &lt;- system.file(package = "HodgesTools", "extdata", "test_BED6.bed")
BED4 &lt;- system.file(package = "HodgesTools", "extdata", "test_BED4.bed")
BED8 &lt;- system.file(package = "HodgesTools", "extdata", "test_BED8.bed")

# Read 3-column BED file.
read_bed(BED3)

# Read 6-column BED file.
read_bed(BED6)

# Read 3-column BED file and add length column.
read_bed(BED3, length = TRUE)

# Read 3 column format BED file with additional fourth column. Add generic column names.
read_bed(BED4)

# Read 3 column format BED file with additional fourth column. Specify additional column names.
read_bed(BED4, extra_col_names = c("fourthColumn"))

# Read 6 column format BED file with additional columns. Specify additional column names.
read_bed(BED8, extra_col_names = c("seventhColumn", "eigthColumn"))

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
