<!DOCTYPE html><html><head><title>Help for package homologene</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {homologene}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#autoTranslate'><p>Attempt to automatically translate a gene list</p></a></li>
<li><a href='#getGeneHistory'><p>Download gene history file</p></a></li>
<li><a href='#getGeneInfo'><p>Download gene symbol information</p></a></li>
<li><a href='#getHomologene'><p>Get the latest homologene file</p></a></li>
<li><a href='#homologene'><p>Get homologues of given genes</p></a></li>
<li><a href='#homologeneData'><p>homologeneData</p></a></li>
<li><a href='#homologeneData2'><p>homologeneData2</p></a></li>
<li><a href='#homologeneVersion'><p>Version of homologene used</p></a></li>
<li><a href='#human2mouse'><p>Human/mouse wraper for homologene</p></a></li>
<li><a href='#mouse2human'><p>Mouse/human wraper for homologene</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#taxData'><p>Names and ids of included species</p></a></li>
<li><a href='#updateHomologene'><p>Update homologene database</p></a></li>
<li><a href='#updateIDs'><p>Update gene IDs</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Quick Access to Homologene and Gene Annotation Updates</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.68.19.3.27</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr (&ge; 0.7.4), magrittr (&ge; 1.5), purrr (&ge; 0.2.5), readr
(&ge; 1.3.1), R.utils(&ge; 2.8.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 1.0.2)</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-03-28</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/oganm/homologene/issues">https://github.com/oganm/homologene/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/oganm/homologene">https://github.com/oganm/homologene</a></td>
</tr>
<tr>
<td>Description:</td>
<td>A wrapper for the homologene database by the National Center for
    Biotechnology Information ('NCBI'). It allows searching for gene homologs across 
    species. Data in this package can be found at <a href="ftp://ftp.ncbi.nih.gov/pub/HomoloGene/build68/">ftp://ftp.ncbi.nih.gov/pub/HomoloGene/build68/</a>.
    The package also includes an updated version of the homologene database where 
    gene identifiers and symbols are replaced with their latest (at the time of
    submission) version and functions to fetch latest annotation data to keep updated.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-03-28 20:51:16 UTC; omancarci</td>
</tr>
<tr>
<td>Author:</td>
<td>Ogan Mancarci [aut, cre],
  Leon French [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ogan Mancarci &lt;ogan.mancarci@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-03-28 23:10:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='autoTranslate'>Attempt to automatically translate a gene list</h2><span id='topic+autoTranslate'></span>

<h3>Description</h3>

<p>Given a list of query gene list and a target gene list, the function
tries find the homology pairing that matches the query list to the target list. The query list
is a short list of genes while the target list is supposed to represent a large number of genes from the target
species. The default output will be the largest possible list. If <code>returnAllPossible = TRUE</code> then
all possible pairings with any matches are returned. It is possible to limit the
search by setting <code>possibleOrigins</code> and <code>possibleTargets</code>. Note that gene symbols of some species
are more similar to each other than others. Using this with small gene lists and without providing any
<code>possibleOrigins</code> or <code>possibleTargets</code> might return multiple hits, or if <code>returnAllPossible = TRUE</code>
a wrong match can be returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>autoTranslate(genes, targetGenes, possibleOrigins = NULL,
  possibleTargets = NULL, returnAllPossible = FALSE,
  db = homologene::homologeneData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="autoTranslate_+3A_genes">genes</code></td>
<td>
<p>A list of genes to match the target. Symbols or NCBI ids</p>
</td></tr>
<tr><td><code id="autoTranslate_+3A_targetgenes">targetGenes</code></td>
<td>
<p>The target list. This list is supposed to represent a large number of genes
from the target species.</p>
</td></tr>
<tr><td><code id="autoTranslate_+3A_possibleorigins">possibleOrigins</code></td>
<td>
<p>Taxonomic identifiers of possible origin species</p>
</td></tr>
<tr><td><code id="autoTranslate_+3A_possibletargets">possibleTargets</code></td>
<td>
<p>Taxonomic identifiers of possible target species</p>
</td></tr>
<tr><td><code id="autoTranslate_+3A_returnallpossible">returnAllPossible</code></td>
<td>
<p>if TRUE returns all possible pairings with non zero gene matches. If FALSE (default) returns the best match</p>
</td></tr>
<tr><td><code id="autoTranslate_+3A_db">db</code></td>
<td>
<p>Homologene database to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame if <code>returnAllPossibe = FALSE</code> and a list of data frames if <code>TRUE</code>
</p>

<hr>
<h2 id='getGeneHistory'>Download gene history file</h2><span id='topic+getGeneHistory'></span>

<h3>Description</h3>

<p>Downloads and reads the gene history file from NCBI website. This file is needed for
other functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGeneHistory(destfile = NULL, justRead = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGeneHistory_+3A_destfile">destfile</code></td>
<td>
<p>Path of the output file. If NULL a temp file will be used</p>
</td></tr>
<tr><td><code id="getGeneHistory_+3A_justread">justRead</code></td>
<td>
<p>If TRUE and destfile exists, it reads the file instead of 
downloading the latest one from NCBI</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with latest gene history information
</p>

<hr>
<h2 id='getGeneInfo'>Download gene symbol information</h2><span id='topic+getGeneInfo'></span>

<h3>Description</h3>

<p>This function downloads the gene_info file from NCBI website and returns the
gene symbols for current IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGeneInfo(destfile = NULL, justRead = FALSE, chunk_size = 1e+06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGeneInfo_+3A_destfile">destfile</code></td>
<td>
<p>Path of the output file. If NULL a temp file will be used</p>
</td></tr>
<tr><td><code id="getGeneInfo_+3A_justread">justRead</code></td>
<td>
<p>If TRUE and destfile exists, it reads the file instead of 
downloading the latest one from NCBI</p>
</td></tr>
<tr><td><code id="getGeneInfo_+3A_chunk_size">chunk_size</code></td>
<td>
<p>Chunk size to be used with <code>link[readr]{read_tsv_chunked}</code>.
The gene_info file is big enough to make its intake difficult. If you don't
have large amounts of free memory you may have to reduce this number to read
the file in smaller chunks</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with gene symbols for each current gene id
</p>

<hr>
<h2 id='getHomologene'>Get the latest homologene file</h2><span id='topic+getHomologene'></span>

<h3>Description</h3>

<p>This function downloads the latest homologene file from NCBI. Note that Homologene
has not been updated since 2014 so the output will be identical to <code><a href="#topic+homologeneData">homologeneData</a></code>
included in this package. This function is here for futureproofing purposes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getHomologene(destfile = NULL, justRead = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getHomologene_+3A_destfile">destfile</code></td>
<td>
<p>Path of the output file. If NULL a temp file will be used</p>
</td></tr>
<tr><td><code id="getHomologene_+3A_justread">justRead</code></td>
<td>
<p>If TRUE and destfile exists, it reads the file instead of 
downloading the latest one from NCBI</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with homology groups, gene ids and gene symbols
</p>

<hr>
<h2 id='homologene'>Get homologues of given genes</h2><span id='topic+homologene'></span>

<h3>Description</h3>

<p>Given a list of genes and a taxid, returns a data frame inlcuding the genes and their corresponding homologues
</p>


<h3>Usage</h3>

<pre><code class='language-R'>homologene(genes, inTax, outTax, db = homologene::homologeneData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="homologene_+3A_genes">genes</code></td>
<td>
<p>A vector of gene symbols or NCBI ids</p>
</td></tr>
<tr><td><code id="homologene_+3A_intax">inTax</code></td>
<td>
<p>taxid of the species that the input genes are coming from</p>
</td></tr>
<tr><td><code id="homologene_+3A_outtax">outTax</code></td>
<td>
<p>taxid of the species that you are seeking homology</p>
</td></tr>
<tr><td><code id="homologene_+3A_db">db</code></td>
<td>
<p>Homologene database to use.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>homologene(c('Eno2','17441'), inTax = 10090, outTax = 9606)
</code></pre>

<hr>
<h2 id='homologeneData'>homologeneData</h2><span id='topic+homologeneData'></span>

<h3>Description</h3>

<p>List of gene homologues used by homologene functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>homologeneData
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 275237 rows and 4 columns.</p>

<hr>
<h2 id='homologeneData2'>homologeneData2</h2><span id='topic+homologeneData2'></span>

<h3>Description</h3>

<p>A modified copy of the homologene database. Homologene was updated at 2014 and many of its gene IDs and
symbols are out of date. Here the IDs and symbols are replaced with their most current version
Last update: Wed Mar 27 16:34:11 2019
</p>


<h3>Usage</h3>

<pre><code class='language-R'>homologeneData2
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 269592 rows and 4 columns.</p>

<hr>
<h2 id='homologeneVersion'>Version of homologene used</h2><span id='topic+homologeneVersion'></span>

<h3>Description</h3>

<p>Version of homologene used
</p>


<h3>Usage</h3>

<pre><code class='language-R'>homologeneVersion
</code></pre>


<h3>Format</h3>

<p>An object of class <code>integer</code> of length 1.</p>

<hr>
<h2 id='human2mouse'>Human/mouse wraper for homologene</h2><span id='topic+human2mouse'></span>

<h3>Description</h3>

<p>Human/mouse wraper for homologene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>human2mouse(genes, db = homologene::homologeneData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="human2mouse_+3A_genes">genes</code></td>
<td>
<p>A vector of gene symbols or NCBI ids</p>
</td></tr>
<tr><td><code id="human2mouse_+3A_db">db</code></td>
<td>
<p>Homologene database to use.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>human2mouse(c('ENO2','4340'))
</code></pre>

<hr>
<h2 id='mouse2human'>Mouse/human wraper for homologene</h2><span id='topic+mouse2human'></span>

<h3>Description</h3>

<p>Mouse/human wraper for homologene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mouse2human(genes, db = homologene::homologeneData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mouse2human_+3A_genes">genes</code></td>
<td>
<p>A vector of gene symbols or NCBI ids</p>
</td></tr>
<tr><td><code id="mouse2human_+3A_db">db</code></td>
<td>
<p>Homologene database to use.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>mouse2human(c('Eno2','17441'))
</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic++25+3E+25'></span><span id='topic++25+3C+3E+25'></span><span id='topic++25+24+25'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>magrittr</dt><dd><p><code><a href="magrittr.html#topic++25+3E+25">%&gt;%</a></code>, <code><a href="magrittr.html#topic++25+3C+3E+25">%&lt;&gt;%</a></code>, <code><a href="magrittr.html#topic++25+24+25">%$%</a></code></p>
</dd>
</dl>

<hr>
<h2 id='taxData'>Names and ids of included species</h2><span id='topic+taxData'></span>

<h3>Description</h3>

<p>Names and ids of included species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxData
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 21 rows and 2 columns.</p>

<hr>
<h2 id='updateHomologene'>Update homologene database</h2><span id='topic+updateHomologene'></span>

<h3>Description</h3>

<p>Creates an updated version of the homologene database. This is done by downloading
the latest gene annotation information and tracing changes in gene symbols and 
identifiers over history. <code><a href="#topic+homologeneData2">homologeneData2</a></code> was created using 
this function over the original <code><a href="#topic+homologeneData">homologeneData</a></code>. This function 
requires downloading large amounts of data from the NCBI ftp servers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>updateHomologene(destfile = NULL,
  baseline = homologene::homologeneData2, gene_history = NULL,
  gene_info = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="updateHomologene_+3A_destfile">destfile</code></td>
<td>
<p>Optional. Path of the output file.</p>
</td></tr>
<tr><td><code id="updateHomologene_+3A_baseline">baseline</code></td>
<td>
<p>The baseline homologene file to be used. By default uses the
<code><a href="#topic+homologeneData2">homologeneData2</a></code> that is included in this package. The more ids 
to update, the more time is needed for the update which is why the default option
uses an already updated version of the original database.</p>
</td></tr>
<tr><td><code id="updateHomologene_+3A_gene_history">gene_history</code></td>
<td>
<p>A gene history data frame, possibly returned by <code><a href="#topic+getGeneHistory">getGeneHistory</a></code>
function. Use this if you want to have a static gene_history file to update up to a specific date.
An up to date gene_history object can be set to update to a specific date by trimming
rows that have recent dates. Note that the same is not possible for the gene_info 
If not provided, the latest file will be downloaded.</p>
</td></tr>
<tr><td><code id="updateHomologene_+3A_gene_info">gene_info</code></td>
<td>
<p>A gene info data frame that contatins ID-symbol matches,
possibly returned by <code><a href="#topic+getGeneInfo">getGeneInfo</a></code>. Use this if you
want a static version. Should be in sync with the gene_history file. Note that there is 
no easy way to track changes in gene symbols back in time so if you want to update it up
to a specific date, make sure you don't lose that file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Homologene database in a data frame with updated gene IDs and symbols
</p>

<hr>
<h2 id='updateIDs'>Update gene IDs</h2><span id='topic+updateIDs'></span>

<h3>Description</h3>

<p>Given a list of gene ids and gene history information, traces changes in the 
gene's name to get the latest valid ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>updateIDs(ids, gene_history)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="updateIDs_+3A_ids">ids</code></td>
<td>
<p>Gene ids</p>
</td></tr>
<tr><td><code id="updateIDs_+3A_gene_history">gene_history</code></td>
<td>
<p>Gene history information, probably returned by  <code><a href="#topic+getGeneHistory">getGeneHistory</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector. New ids for genes that changed ids, or &quot;-&quot; for discontinued genes.
the input itself.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gene_history = getGeneHistory()
updateIDs(c("4340964", "4349034", "4332470", "4334151", "4323831"),gene_history)

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
