<!DOCTYPE html><html><head><title>Help for package Hotelling</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Hotelling}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alr'><p>Additive log ratio transformation</p></a></li>
<li><a href='#bottle.df'><p>Bottle data</p></a></li>
<li><a href='#clr'><p>Centered log ratio transformation</p></a></li>
<li><a href='#container.df'><p>Container data</p></a></li>
<li><a href='#Hotelling-package'><p>Hotelling</p></a></li>
<li><a href='#hotelling.stat'><p>Calculate Hotelling's two sample T-squared test statistic</p></a></li>
<li><a href='#hotelling.test'><p>Two-sample Hotelling's T-squared test</p></a></li>
<li><a href='#manova1.df'><p>manova1 data</p></a></li>
<li><a href='#plot.hotelling.test'><p>Plots the results from a permutation based version of Hotelling's T-squared</p>
test for the difference in two multivariate sample means</a></li>
<li><a href='#print.hotelling.test'><p>Prints the results from a Hotelling's T-squared test for the difference in</p>
two multivariate sample means</a></li>
<li><a href='#summarise'><p>Summary statistics for grouped data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.0-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-09-09</td>
</tr>
<tr>
<td>Title:</td>
<td>Hotelling's T^2 Test and Variants</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>James Curran &lt;j.curran@auckland.ac.nz&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A set of R functions which implements Hotelling's T^2 test and some variants of it. Functions are also included for Aitchison's additive log ratio and centred log ratio transformations.</td>
</tr>
<tr>
<td>Depends:</td>
<td>corpcor</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jmcurran/Hotelling">https://github.com/jmcurran/Hotelling</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jmcurran/Hotelling/issues">https://github.com/jmcurran/Hotelling/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-09 01:31:33 UTC; jcur002</td>
</tr>
<tr>
<td>Author:</td>
<td>James Curran [aut, cre],
  Taylor Hersh [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-09 04:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='alr'>Additive log ratio transformation</h2><span id='topic+alr'></span>

<h3>Description</h3>

<p>Aitchison's additive log ratio tranformation for compositional data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alr(form, data, group = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alr_+3A_form">form</code></td>
<td>
<p>a formula which specifies the denominator variable as the
response</p>
</td></tr>
<tr><td><code id="alr_+3A_data">data</code></td>
<td>
<p>a data frame in which the data is stored</p>
</td></tr>
<tr><td><code id="alr_+3A_group">group</code></td>
<td>
<p>if not NULL then a character string specifying the name of the
grouping variable</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will give a warning if zeros are present because the
transformed data will have -Infs.
</p>


<h3>Value</h3>

<p>a data frame with the ALR transformation applied to data. Each row
in the data frame is standardized with respect to a specific variable by
dividing by that variable. The logarithms of the resulting ratios are
returned. If a grouping variable is specified, then this is preserved.
</p>


<h3>Author(s)</h3>

<p>James M. Curran
</p>


<h3>References</h3>

<p>Aitchison, J. (1986). &ldquo;The Statistical Analysis of
Compositional Data&rdquo;, Chapman and Hall, reprinted in 2003 with additional
material by The Blackburn Press
</p>


<h3>See Also</h3>

<p>clr
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bottle.df)

## transform with respect to manganese
alr(Mn~., bottle.df, "Number")

## transform the data with respect to barium, but removing the
## grouping in column 1
alr(Ba~., bottle.df[,-1])
</code></pre>

<hr>
<h2 id='bottle.df'>Bottle data</h2><span id='topic+bottle.df'></span>

<h3>Description</h3>

<p>This data contains the elemental concentration of five different elements
(Manganese, Barium, Strontium, Zirconium, and Titanium) in samples of glass
taken from six different Heineken beer bottles. 20 measurements were taken
from each bottle.
</p>


<h3>References</h3>

<p>R. L. Bennett. <em>Aspects of the analysis and interpretation
of glass trace evidence</em>. Master's thesis, Department of Chemistry,
University of Waikato, 2002.
</p>

<hr>
<h2 id='clr'>Centered log ratio transformation</h2><span id='topic+clr'></span>

<h3>Description</h3>

<p>Aitchison's centered log ratio tranformation for compositional data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clr(data, group = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clr_+3A_data">data</code></td>
<td>
<p>a data frame in which the data is stored</p>
</td></tr>
<tr><td><code id="clr_+3A_group">group</code></td>
<td>
<p>if not NULL then a character string specifying the name of the
grouping variable</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will give a warning if zeros are present because the
transformed data will have -Infs.
</p>


<h3>Value</h3>

<p>a data frame with the CLR transformation applied to data. Each row
in the data frame is standardized by dividing by the geometric mean of that
row. The logarithms of the resulting ratios are returned. If a grouping
variable is specified, then this is preserved.
</p>


<h3>Author(s)</h3>

<p>James M. Curran
</p>


<h3>References</h3>

<p>Aitchison, J. (1986). &ldquo;The Statistical Analysis of
Compositional Data&rdquo;, Chapman and Hall, reprinted in 2003 with additional
material by The Blackburn Press
</p>


<h3>See Also</h3>

<p>alr
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bottle.df)

## transform preserving grouping
clr(bottle.df, "Number")

## transform the data but remove the
## grouping in column 1
clr(bottle.df[,-1])

</code></pre>

<hr>
<h2 id='container.df'>Container data</h2><span id='topic+container.df'></span>

<h3>Description</h3>

<p>This data contains the elemental concentration of nine different elements
(Titanium, Aluminium, Iron, Manganese, Magnesium, Calcium, Barium,
Strontium, and Zirconium) in specimens of glass taken from two different
containers. Ten measurements were taken from each container.
</p>


<h3>References</h3>

<p>Jose R. Almirall. Discrimination of glass samples by solution
based ICP-OES PhD thesis, Department of Chemistry, Florida International
University, 1998.
</p>

<hr>
<h2 id='Hotelling-package'>Hotelling</h2><span id='topic+Hotelling-package'></span><span id='topic+Hotelling'></span>

<h3>Description</h3>

<p>A set of R functions and data sets which implements Hotelling's T^2 test, and some variants of it. 
Functions are also included for Aitchison's additive log ratio and centred log ratio transformations.
</p>


<h3>Author(s)</h3>

<p>James Curran Maintainer: James Curran &lt;j.curran@auckland.ac.nz&gt; ~~
The author and/or maintainer of the package ~~
</p>

<hr>
<h2 id='hotelling.stat'>Calculate Hotelling's two sample T-squared test statistic</h2><span id='topic+hotelling.stat'></span><span id='topic+hotel.stat'></span>

<h3>Description</h3>

<p>Calculate Hotelling's T-squared test statistic for the difference in two
multivariate means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hotelling.stat(x, y, shrinkage = FALSE, var.equal = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hotelling.stat_+3A_x">x</code></td>
<td>
<p>a nx by p matrix containing the data points from sample 1 or a list containing elements <code>mean</code>, <code>cov</code>, and <code>n</code> where
<code>mean</code> is a mean vector of length p, <code>cov</code> is a variance-covariance matrix of dimension p by p, and <code>n</code> is the sample size</p>
</td></tr>
<tr><td><code id="hotelling.stat_+3A_y">y</code></td>
<td>
<p>a ny by p matrix containg the data points from sample 2  or a list containing elements <code>mean</code>, <code>cov</code>, and <code>n</code> where
<code>mean</code> is a mean vector of length p, <code>cov</code> is a variance-covariance matrix of dimension p by p, and <code>n</code> is the sample size</p>
</td></tr>
<tr><td><code id="hotelling.stat_+3A_shrinkage">shrinkage</code></td>
<td>
<p>set to <code>TRUE</code> if the covariance matrices are to be estimated using Schaefer and Strimmer's James-Stein 
shrinkage estimator. Note this only works when raw data is supplied, and will 
not work if summary statistics are supplied.</p>
</td></tr>
<tr><td><code id="hotelling.stat_+3A_var.equal">var.equal</code></td>
<td>
<p>set to <code>TRUE</code> if the covariance matrices are (assumed to be) equal</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note, the sample size requirements are that nx + ny - 1 &gt; p. The procedure
will stop if this is not met and the shrinkage estimator is not being used.
The shrinkage estimator has not been rigorously tested for this application
(small p, smaller n).
</p>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<table>
<tr><td><code>statistic</code></td>
<td>
<p>Hotelling's (unscaled) T-squared statistic</p>
</td></tr> <tr><td><code>m</code></td>
<td>
<p>The
scaling factor - this can be used by by multiplying it with the test
statistic, or dividing the critical F value</p>
</td></tr> <tr><td><code>df</code></td>
<td>
<p>a vector of length
containing the numerator and denominator degrees of freedom</p>
</td></tr> <tr><td><code>nx</code></td>
<td>
<p>The
sample size of sample 1</p>
</td></tr> <tr><td><code>ny</code></td>
<td>
<p>The sample size of sample 2</p>
</td></tr> <tr><td><code>p</code></td>
<td>
<p>The
number of variables to be used in the comparison</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James M. Curran
</p>
<p>Taylor Hersh
</p>


<h3>References</h3>

<p>Hotelling, H. (1931). &ldquo;The generalization of Student's ratio.&rdquo;
Annals of Mathematical Statistics 2 (3): 360&ndash;378.
</p>
<p>Schaefer, J., and K. Strimmer (2005). &ldquo;A shrinkage approach to large-scale
covariance matrix estimation and implications for functional genomics.&rdquo;
Statist. Appl. Genet. Mol. Biol. 4: 32.
</p>
<p>Opgen-Rhein, R., and K. Strimmer (2007). &ldquo;Accurate ranking of
differentially expressed genes by a distribution-free shrinkage approach.&rdquo;
Statist. Appl. Genet. Mol. Biol. 6: 9.
</p>
<p>NEL, D.G. and VAN DER MERWE, C.A. (1986). &ldquo;A solution to the -
multivariate Behrens-Fisher problem.&rdquo;
Comm. Statist. Theor.- Meth., A15, 12, 3719-3736.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(container.df)
split.data = split(container.df[,-1],container.df$gp)
x = split.data[[1]]
y = split.data[[2]]
hotelling.stat(x, y)
hotelling.stat(x, y, TRUE)

</code></pre>

<hr>
<h2 id='hotelling.test'>Two-sample Hotelling's T-squared test</h2><span id='topic+hotelling.test'></span><span id='topic+hotelling.test.default'></span><span id='topic+hotelling.test.formula'></span><span id='topic+hotel.test'></span>

<h3>Description</h3>

<p>Performs a two-sample Hotelling's T-squared test for the difference in two
multivariate means
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hotelling.test(x, ...)

## Default S3 method:
hotelling.test(
  x,
  y,
  shrinkage = FALSE,
  var.equal = TRUE,
  perm = FALSE,
  B = 10000,
  progBar = (perm &amp;&amp; TRUE),
  ...
)

## S3 method for class 'formula'
hotelling.test(x, data = NULL, pair = c(1, 2), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hotelling.test_+3A_x">x</code></td>
<td>
<p>a matrix containing the data points from sample 1, or a formula
specifying the elements to be used as a response and the grouping variable
as a predictor, or a list containing elements <code>mean</code>, <code>cov</code>, and
<code>n</code> where <code>mean</code> is a mean vector of length p, <code>cov</code> is a
variance-covariance matrix of dimension p by p, and <code>n</code> is the sample
size</p>
</td></tr>
<tr><td><code id="hotelling.test_+3A_...">...</code></td>
<td>
<p>any additional arguments. This is useful to pass the optional
arguments for the default call from the formula version</p>
</td></tr>
<tr><td><code id="hotelling.test_+3A_y">y</code></td>
<td>
<p>a matrix containing the data points from sample 2, or a list
containing elements <code>mean</code>, <code>cov</code>, and <code>n</code> where <code>mean</code>
is a mean vector of length p, <code>cov</code> is a variance-covariance matrix of
dimension p by p, and <code>n</code> is the sample size</p>
</td></tr>
<tr><td><code id="hotelling.test_+3A_shrinkage">shrinkage</code></td>
<td>
<p>if <code>TRUE</code> then Shaefer and Strimmer's James-Stein
shrinkage estimator is used to calculate the sample covariance matrices</p>
</td></tr>
<tr><td><code id="hotelling.test_+3A_var.equal">var.equal</code></td>
<td>
<p>set to <code>TRUE</code> if the covariance matrices are (assumed to be) equal</p>
</td></tr>
<tr><td><code id="hotelling.test_+3A_perm">perm</code></td>
<td>
<p>if <code>TRUE</code> then permutation testing is used to estimate the
non-parametric P-value for the hypothesis test</p>
</td></tr>
<tr><td><code id="hotelling.test_+3A_b">B</code></td>
<td>
<p>if perm is TRUE, then B is the number of permutations to perform</p>
</td></tr>
<tr><td><code id="hotelling.test_+3A_progbar">progBar</code></td>
<td>
<p>if <code>TRUE</code> and <code>perm</code> is TRUE then a progress bar
will be displayed whilst the permutation procedure is carried out</p>
</td></tr>
<tr><td><code id="hotelling.test_+3A_data">data</code></td>
<td>
<p>a data frame needs to be specified if a formula is to be used to
perform the test</p>
</td></tr>
<tr><td><code id="hotelling.test_+3A_pair">pair</code></td>
<td>
<p>a vector of length two which can be used when the grouping factor
has more than two levels to select different pairs of groups. For example
for a 3-level factor, pairs could be set to <code>c(1,3)</code> to perform
Hotelling's test between groups 1 an 3</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list (which is also of class 'hotelling.test') with the following
elements:
</p>
<table>
<tr><td><code>stats</code></td>
<td>
<p>a list containing all of the output from
<code>hotelling.stat</code></p>
</td></tr> <tr><td><code>pval</code></td>
<td>
<p>the P-value from the test</p>
</td></tr>
<tr><td><code>results</code></td>
<td>
<p>if <code>perm == TRUE</code>, then all of the permuation test
statisics are stored in results</p>
</td></tr>
</table>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>default</code>: Two-sample Hotelling's T-squared test
</p>
</li>
<li> <p><code>formula</code>: Two-sample Hotelling's T-squared test
</p>
</li></ul>


<h3>Author(s)</h3>

<p>James M. Curran
</p>


<h3>References</h3>

<p>Hotelling, H. (1931). &ldquo;The generalization of Student's ratio.&rdquo;
Annals of Mathematical Statistics 2 (3): 360&ndash;378.
</p>
<p>Schaefer, J., and K. Strimmer (2005). &ldquo;A shrinkage approach to large-scale
covariance matrix estimation and implications for functional genomics.&rdquo;
Statist. Appl. Genet. Mol. Biol. 4: 32.
</p>
<p>Opgen-Rhein, R., and K. Strimmer (2007). &ldquo;Accurate ranking of
differentially expressed genes by a distribution-free shrinkage approach.&rdquo;
Statist. Appl. Genet. Mol. Biol. 6: 9.
</p>
<p>Campbell, G.P. and J. M. Curran (2009). &ldquo;The interpretation of elemental
composition measurements from forensic glass evidence III.&rdquo; Science and
Justice, 49(1),2-7.
</p>


<h3>See Also</h3>

<p>hotelling.stat
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(container.df)
fit = hotelling.test(.~gp, data = container.df)
fit

subs.df = container.df[1:10,]
subs.df$gp = rep(1:2, c(5,5))
fitPerm = hotelling.test(Al+Fe~gp, data  = subs.df, perm =  TRUE)
fitPerm
plot(fitPerm)

data(bottle.df)
fit12 = hotelling.test(.~Number, data = bottle.df)
fit12

fit23 = hotelling.test(.~Number, data = bottle.df, pair = c(2,3))
fit23

data(manova1.df)
fit = hotelling.test(wratr+wrata~treatment, data = manova1.df, var.equal = FALSE)
fit

x = list(mean = c(7.81, 108.77, 44.92),
         cov = matrix(c(0.461, 1.18, 4.49,
                        1.18, 3776.4, -17.35, 
                        4.49, -17.35, 147.24), nc = 3, byrow = TRUE),
         n = 13)
y = list(mean = c(5.89, 41.9, 20.8),
         cov = matrix(c(0.148, -0.679, 0.209, 
                       -0.679, 96.10, 20.20,
                        0.209, 20.20, 24.18), nc = 3, byrow = TRUE),
         n = 10)
fit = hotelling.test(x, y, var.equal = FALSE)
fit
</code></pre>

<hr>
<h2 id='manova1.df'>manova1 data</h2><span id='topic+manova1.df'></span>

<h3>Description</h3>

<p>The data contains example data for testing the unequal variance option in the 
package. The dataset has four varibles, <code>wratr</code>, <code>wrata</code>, <code>treatment</code>, 
and <code>disability</code>. <code>treatment is the grouping variable</code> and <code>wratr</code> and <code>wrata</code>
are the responses. <code>disability</code> is not used.
</p>


<h3>References</h3>

<p><a href="https://ncss-wpengine.netdna-ssl.com/wp-content/themes/ncss/pdf/Procedures/NCSS/Hotellings_Two-Sample_T2.pdf">https://ncss-wpengine.netdna-ssl.com/wp-content/themes/ncss/pdf/Procedures/NCSS/Hotellings_Two-Sample_T2.pdf</a>
</p>

<hr>
<h2 id='plot.hotelling.test'>Plots the results from a permutation based version of Hotelling's T-squared
test for the difference in two multivariate sample means</h2><span id='topic+plot.hotelling.test'></span>

<h3>Description</h3>

<p>Plots a histogram of the distribution of the permuted test statistics for a
permutation version of Hotelling's T-squared
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'hotelling.test'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.hotelling.test_+3A_x">x</code></td>
<td>
<p>an object of type hotelling.test</p>
</td></tr>
<tr><td><code id="plot.hotelling.test_+3A_...">...</code></td>
<td>
<p>any additional arguments to be passed to the hist command</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function only works if you have performed a permutation test. It will
return an error message if not. It could be programmed to draw the relevant
F distribution in the standard case, but this seems rather pointless.
</p>


<h3>Author(s)</h3>

<p>James M. Curran
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bottle.df)
bottle.df = subset(bottle.df, Number == 1)
bottle.df$Number = rep(1:2,c(10,10))
fit = hotelling.test(.~Number, bottle.df, perm = TRUE)
plot(fit)
plot(fit, col = "lightblue")

</code></pre>

<hr>
<h2 id='print.hotelling.test'>Prints the results from a Hotelling's T-squared test for the difference in
two multivariate sample means</h2><span id='topic+print.hotelling.test'></span>

<h3>Description</h3>

<p>Prints the test stastic, degrees of freedom and P-value from Hotelling's
T-squared test for the difference in two multivariate sample means
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'hotelling.test'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.hotelling.test_+3A_x">x</code></td>
<td>
<p>an object of type hotelling.test</p>
</td></tr>
<tr><td><code id="print.hotelling.test_+3A_...">...</code></td>
<td>
<p>any additional arguments to be passed to the hist command</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James M. Curran
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bottle.df)
bottle.df = subset(bottle.df, Number == 1)
bottle.df$Number = rep(1:2,c(10,10))
fit = hotelling.test(.~Number, bottle.df, perm = TRUE)
fit
fit = hotelling.test(.~Number, bottle.df)
fit

## an explict call
print(fit)

</code></pre>

<hr>
<h2 id='summarise'>Summary statistics for grouped data</h2><span id='topic+summarise'></span><span id='topic+summarize'></span><span id='topic+summarise.default'></span><span id='topic+summarise.formula'></span><span id='topic+summarise.data.frame'></span>

<h3>Description</h3>

<p>Easily get summary statistics for each group present in the data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarise(x, ...)

## Default S3 method:
summarise(
  x,
  y,
  stats = list(Mean = mean, Median = median, `Std. Dev.` = sd, N = length),
  ...
)

## S3 method for class 'formula'
summarise(
  x,
  data = NULL,
  stats = list(Mean = mean, Median = median, `Std. Dev.` = sd, N = length),
  ...
)

## S3 method for class 'data.frame'
summarise(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summarise_+3A_x">x</code></td>
<td>
<p>a matrix of multivariate observations, a list of summary statistics from
multivariate observations, a data.frame of multivariate observations, or a formula
with a multivariate response on the right hand side, and a grouping variable/factor
on the left hand side.</p>
</td></tr>
<tr><td><code id="summarise_+3A_y">y</code></td>
<td>
<p>a matrix of multivariate observations, a list of summary statistics from
multivariate observations, OR a data.frame of multivariate observations</p>
</td></tr>
<tr><td><code id="summarise_+3A_stats">stats</code></td>
<td>
<p>a named list of summary statistics to compute on each variable in each 
group. Note 1: Quantiles are not supported yet because I can't think of a good way to
handle the extra arguments. Help welcome. Note 2: The names of the elements in the list 
are used to label the columns of the output. They probably should be unique.</p>
</td></tr>
<tr><td><code id="summarise_+3A_data">data</code></td>
<td>
<p>a data.frame containing the variables used in a formula</p>
</td></tr>
<tr><td><code id="summarise_+3A_...">...</code></td>
<td>
<p>other arguments such as another matrix of multivariate observations:
see <code>summarise.default</code>, or a data to be used with a formula: see
<code>summarise.formula</code></p>
</td></tr>
</table>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>default</code>: Summary statistics for grouped data
</p>
</li>
<li> <p><code>formula</code>: Summary statistics for grouped data
</p>
</li>
<li> <p><code>data.frame</code>: Summary statistics for grouped data
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>
data(container.df)
split.data = split(container.df[,-1],container.df$gp)
x = split.data[[1]]
y = split.data[[2]]
summarise(x, y)

## Using the formula interface
data(container.df)
summarise(gp~., data = container.df)

summarise(gp~Al+Ti, data = container.df)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
