<!DOCTYPE html><html lang="en"><head><title>Help for package hsphase</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hsphase}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#hsphase-package'>
<p>Phasing, Pedigree Reconstruction, Sire Imputation and Recombination Events Identification for Half-sib Families</p></a></li>
<li><a href='#.fastdist'><p>Calculate Genotypic Distances</p></a></li>
<li><a href='#.maf'><p>Calculate Minor Allele Frequency (MAF)</p></a></li>
<li><a href='#.simulateHalfsib'><p>Simulate Half-Sibling Genotypes</p></a></li>
<li><a href='#aio'>
<p>All-in-one Phasing</p></a></li>
<li><a href='#bmh'>
<p>Block Partitioning</p></a></li>
<li><a href='#co'>
<p>Crossover Detection</p></a></li>
<li><a href='#cs'>
<p>Chromosome Splitter</p></a></li>
<li><a href='#genotypes'><p>Example of Genotype Data Set</p></a></li>
<li><a href='#hbp'>
<p>Haplotype Blocks of Phased Data</p></a></li>
<li><a href='#hh'>
<p>Heatmap of Half-sibs</p></a></li>
<li><a href='#hss'>
<p>Half-sib Family Splitter</p></a></li>
<li><a href='#imageplot'>
<p>Image Plot of Blocking Structure</p></a></li>
<li><a href='#impute'>
<p>Impute of Low Density SNP Marker to High Density (Paternal Strand)</p></a></li>
<li><a href='#map'><p>Example Map File for Genetic Data</p></a></li>
<li><a href='#ohd'>
<p>Opposing Homozygote Detection</p></a></li>
<li><a href='#ohg'>
<p>Matrix of Opposing Homozygotes</p></a></li>
<li><a href='#ohplot'>
<p>Opposing Homozygotes Plot</p></a></li>
<li><a href='#para'>
<p>Parallel Analysis of Data</p></a></li>
<li><a href='#pedigree'><p>Example Pedigree Data Set</p></a></li>
<li><a href='#pedigreeNaming'>
<p>Fix Pedigree Errors</p></a></li>
<li><a href='#phf'>
<p>Half-Sib Family Phasing</p></a></li>
<li><a href='#pm'>
<p>Probability Matrix</p></a></li>
<li><a href='#pogc'>
<p>Parent Offspring Group Constructor</p></a></li>
<li><a href='#readGenotype'>
<p>Read and Check the Genotype File</p></a></li>
<li><a href='#recombinations'>
<p>Recombination Number</p></a></li>
<li><a href='#rplot'>
<p>Recombination Plot</p></a></li>
<li><a href='#rpoh'>
<p>Reconstruct Pedigree Based on Matrix of Opposing Homozygotes</p></a></li>
<li><a href='#ssp'>
<p>Sire Imputation and Phasing</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Phasing, Pedigree Reconstruction, Sire Imputation and
Recombination Events Identification of Half-sib Families Using
SNP Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Mohammad Ferdosi &lt;mhferdosi@yahoo.com&gt;, Cedric Gondro &lt;gondroce@msu.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mohammad Ferdosi &lt;mhferdosi@yahoo.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>snowfall, R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>RcppArmadillo (&ge; 0.4.300.8.0), Rcpp (&ge; 0.11.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Description:</td>
<td>Identification of recombination events, haplotype reconstruction, sire imputation and pedigree reconstruction using half-sib family SNP data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-16 07:45:27 UTC; mhf</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-16 11:40:02 UTC</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
</table>
<hr>
<h2 id='hsphase-package'>
Phasing, Pedigree Reconstruction, Sire Imputation and Recombination Events Identification for Half-sib Families
</h2><span id='topic+hsphase-package'></span><span id='topic+hsphase'></span>

<h3>Description</h3>

<p>Identification of recombination events, haplotype reconstruction and sire imputation using half-sib family SNP data.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> hsphase</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2014-6-17</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL 3 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong>Main Functions:</strong> <br />
<code><a href="#topic+bmh">bmh</a></code>: Block partitioning<br />
<code><a href="#topic+ssp">ssp</a></code>: Sire inference<br />
<code><a href="#topic+aio">aio</a></code>: Phasing<br />
<code><a href="#topic+imageplot">imageplot</a></code>: Image plot of the block structure <br />
<code><a href="#topic+rpoh">rpoh</a></code>: Reconstruct pedigree based on opposing homozygote <br />
</p>
<p><strong>Auxiliary Functions</strong> <br />
<code><a href="#topic+hss">hss</a></code>: Half-sib family splitter<br />
<code><a href="#topic+cs">cs</a></code>: Chromosome splitter<br />
<code><a href="#topic+para">para</a></code>: Parallel data analysis<br />
</p>
<p><em>Note</em>: These functions can be used to analyse large datasets.
</p>


<h3>Author(s)</h3>

<p>Mohammad H. Ferdosi &lt;mferdosi@une.udu.au&gt;, Cedric Gondro &lt;cgondro2@une.edu.au&gt;
Maintainer: Mohammad H. Ferdosi &lt;mferdosi@myune.udu.au&gt;
</p>


<h3>References</h3>

<p>Ferdosi, M. H., Kinghorn, B. P., van der Werf, J. H., &amp; Gondro, C (2013). Effect of genotype and pedigree error on detection of recombination events, sire imputation and haplotype inference using the hsphase algorithm. In Proc. Assoc. Advmt. Anim. Breed. Genet (Vol. 20, pp. 546-549). AAABG; Napier, New Zealand.
<br /><br />
Ferdosi, M. H., Kinghorn, B. P., van der Werf, J. H., &amp; Gondro, C. (2014). Detection of recombination events, haplotype reconstruction and imputation of sires using half-sib SNP genotypes. Genetics, selection, evolution: GSE, 46(1), 11.
<br /><br />
Ferdosi, M. H., Kinghorn, B. P., van der Werf, J. H., Lee, S. H., &amp; Gondro, C. (2014). hsphase: an R package for pedigree reconstruction, detection of recombination events, phasing and imputation of half-sib family groups. BMC Bioinformatics, 15(1), 172.
<br /><br />
Ferdosi, M. H., &amp; Boerner, V. (2014). A fast method for evaluating opposing homozygosity in large SNP data sets. Livestock Science.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(
  0,0,0,0,1,2,2,2,0,0,2,0,0,0,
  2,2,2,2,1,0,0,0,2,2,2,2,2,2,
  2,2,2,2,1,2,2,2,0,0,2,2,2,2,
  2,2,2,2,0,0,0,0,2,2,2,2,2,2,
  0,0,0,0,0,2,2,2,2,2,2,0,0,0), ncol = 14, byrow = TRUE) 
ssp(bmh(genotype), genotype)
aio(genotype)
imageplot(bmh(genotype), title = "ImagePlot example")
rplot(genotype, c(1:14))
</code></pre>

<hr>
<h2 id='.fastdist'>Calculate Genotypic Distances</h2><span id='topic+.fastdist'></span>

<h3>Description</h3>

<p>Calculates a symmetric matrix of distances between genotypes, based on a given genotype matrix.
Each row in the 'GenotypeMatrix' represents a genotype, and each column represents a marker.
The genotype is coded as 0 for AA, 1 for AB, and 2 for BB. Use 9 to represent missing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.fastdist(GenotypeMatrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id=".fastdist_+3A_genotypematrix">GenotypeMatrix</code></td>
<td>
<p>A matrix where each row represents a genotype and each column 
represents a marker. Genotypes should be coded as 0 for AA, 1 for AB, and 2 for BB, 
with 9 representing missing data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a symmetric matrix of distances between the genotypes specified in the 
'GenotypeMatrix'. Row and column names of the returned matrix correspond to the row names 
of the 'GenotypeMatrix'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate genotype data for 40 individuals across 1000 SNPs
# genotypes &lt;- simulateHalfsib(numInd = 40, numSNP = 1000, recbound = 0:6, type = "genotype")
# Calculate the distance matrix
# dist_matrix &lt;- fastdist(genotypes)
# Display the distance matrix
# print(dist_matrix)

</code></pre>

<hr>
<h2 id='.maf'>Calculate Minor Allele Frequency (MAF)</h2><span id='topic+.maf'></span>

<h3>Description</h3>

<p>This function calculates the minor allele frequency (MAF) for a given single nucleotide polymorphism (SNP) data. 
The SNP data should be coded numerically: 0 for homozygous for the first allele (AA), 
1 for heterozygous (AB), and 2 for homozygous for the second allele (BB). Missing data should be coded as 9.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.maf(snp)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id=".maf_+3A_snp">snp</code></td>
<td>
<p>A numeric vector representing the genotype of individuals for a single SNP. 
The genotype should be coded as 0 for AA, 1 for AB, and 2 for BB. Use 9 to represent missing data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value representing the minor allele frequency (MAF) for the SNP data provided.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>snp_data &lt;- c(0, 0, 1, 2, 2, 9)
maf_value &lt;- .maf(snp_data)
print(maf_value)

</code></pre>

<hr>
<h2 id='.simulateHalfsib'>Simulate Half-Sibling Genotypes</h2><span id='topic+.simulateHalfsib'></span>

<h3>Description</h3>

<p>This function simulates genotypes for a set of half-siblings based on specified parameters,
including the number of individuals, the number of SNPs, recombination boundaries, and the type of data to return.
It generates a sire genotype, maternal half-sib genotypes, and combines these to simulate offspring genotypes,
optionally returning phased genotypes based on recombination events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.simulateHalfsib(
  numInd = 40,
  numSNP = 10000,
  recbound = 0:6,
  type = "genotype"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id=".simulateHalfsib_+3A_numind">numInd</code></td>
<td>
<p>Integer, the number of half-siblings to simulate.</p>
</td></tr>
<tr><td><code id=".simulateHalfsib_+3A_numsnp">numSNP</code></td>
<td>
<p>Integer, the number of SNPs to simulate for each individual.</p>
</td></tr>
<tr><td><code id=".simulateHalfsib_+3A_recbound">recbound</code></td>
<td>
<p>Numeric vector, specifying the range of possible recombination events to simulate.</p>
</td></tr>
<tr><td><code id=".simulateHalfsib_+3A_type">type</code></td>
<td>
<p>Character string, specifying the type of data to return: &quot;genotype&quot; for genotypic data or any other string for phased genotypic data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depending on the <code>type</code> parameter, this function returns a matrix of simulated genotypic data
for half-siblings. If <code>type</code> is &quot;genotype&quot;, it returns unphased genotypic data; otherwise, it returns phased genotypic data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim_genotypes &lt;- .simulateHalfsib(numInd = 40, numSNP = 10000, recbound = 0:6, type = "genotype")
dim(sim_genotypes) # Should return 40 rows (individuals) and 100 columns (SNPs)

</code></pre>

<hr>
<h2 id='aio'>
All-in-one Phasing
</h2><span id='topic+aio'></span>

<h3>Description</h3>

<p>Phasing of a half-sib family group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aio(genotypeMatrix, bmh_forwardVectorSize = 30, bmh_excludeFP = TRUE,
bmh_nsap = 3, output = "phase")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="aio_+3A_genotypematrix">genotypeMatrix</code></td>
<td>

<p><code>matrix</code> half-sib genotypes  (one half-sib per row, with SNP ordered by mapping position in the columns. Data should be numeric. Use 0, 1 and 2 respectively for AA, AB and BB. Use 9 for missing data)

</p>
</td></tr>
<tr><td><code id="aio_+3A_bmh_forwardvectorsize">bmh_forwardVectorSize</code></td>
<td>

<p><code>integer</code> number of heterozygous sites used to validate recombination events or check for genotyping errors</p>
</td></tr>
<tr><td><code id="aio_+3A_bmh_excludefp">bmh_excludeFP</code></td>
<td>

<p><code>logical</code> excludes SNPs that may cause heterozygous sites in the sire due to genotyping errors or map errors</p>
</td></tr>
<tr><td><code id="aio_+3A_bmh_nsap">bmh_nsap</code></td>
<td>

<p><code>integer</code> number of SNP per block to validate recombinations</p>
</td></tr>
<tr><td><code id="aio_+3A_output">output</code></td>
<td>

<p><code>character</code> if equal to the <code>phase</code> the 'aio' will only return the phasing results 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calls the <code><a href="#topic+bmh">bmh</a></code>,  <code><a href="#topic+ssp">ssp</a></code> and  <code><a href="#topic+phf">phf</a></code> functions. 
</p>


<h3>Value</h3>

<p>Returns a list of matrices. The first element (phasedHalfsibs) is a matrix with two rows (phased haplotypes) per individual (first paternal and second maternal). Data in format 0 (A), 1 (B) and 9 (unphased or missing).
The second (sireHaplotype) and third (blockStructure) elements are the same as the output of  <code><a href="#topic+ssp">ssp</a></code> and  <code><a href="#topic+bmh">bmh</a></code>.
</p>


<h3>Note</h3>

<p>Only this function needs to be called to phase a half-sib family. The genotype's matrix must contain individuals from only one half-sib family and one ordered chromosome.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bmh">bmh</a></code>,  <code><a href="#topic+ssp">ssp</a></code> and  <code><a href="#topic+phf">phf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(       # Define a Half-sib Genotype Matrix
  2,1,0,                    # Individual 1
  2,0,0,                    # Individual 2
  0,0,2                     # Individual 3
  ), byrow = TRUE, ncol = 3)    # There are 3 individulas with three SNPs
  
aio(genotype)               # The genotypes must include only one half-sib family and one chromosome
</code></pre>

<hr>
<h2 id='bmh'>
Block Partitioning
</h2><span id='topic+bmh'></span>

<h3>Description</h3>

<p>Identifies the block structure (chromosome segments) in the half-sib family that each individual inherited from its sire.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bmh(GenotypeMatrix, forwardVectorSize = 30, excludeFP = TRUE, nsap = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bmh_+3A_genotypematrix">GenotypeMatrix</code></td>
<td>

<p><code>matrix</code> half-sib genotypes  (one half-sib per row, with SNP ordered by mapping position in the columns. Data should be numeric. Use 0, 1 and 2 respectively for AA, AB and BB. Use 9 for missing data)</p>
</td></tr>
<tr><td><code id="bmh_+3A_forwardvectorsize">forwardVectorSize</code></td>
<td>

<p><code>integer</code> number of heterozygous sites used to validate recombination events or check for genotyping errors (50k -&gt; 30, 700k -&gt; 120)</p>
</td></tr>
<tr><td><code id="bmh_+3A_excludefp">excludeFP</code></td>
<td>

<p><code>logical</code> excludes SNPs that may cause heterozygous sites in the sire due to genotyping errors or map errors</p>
</td></tr>
<tr><td><code id="bmh_+3A_nsap">nsap</code></td>
<td>

<p><code>integer</code> number of SNP per block to validate recombinations (50k -&gt; 3, 700k -&gt; 10)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix of the blocking structure that contains 1s, 2s and 0s. 1s and 2s are the two sire strands. The choice of strand is arbitrary for each chromosome and not consistent across chromosomes. 0s indicate regions of unknown origin.</p>


<h3>Note</h3>

<p>The genotype's matrix must contain individuals from only one half-sib family and one ordered chromosome.</p>


<h3>See Also</h3>

<p><code><a href="#topic+ssp">ssp</a></code>, <code><a href="#topic+phf">phf</a></code>, <code><a href="#topic+aio">aio</a></code> and <code><a href="#topic+imageplot">imageplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(
0,2,1,1,1,
2,0,1,2,2,
2,2,1,0,2,
2,2,1,1,1,
0,0,2,1,0), ncol = 5, byrow = TRUE) 

(result &lt;- bmh(genotype))

</code></pre>

<hr>
<h2 id='co'>
Crossover Detection</h2><span id='topic+co'></span>

<h3>Description</h3>

<p>Detect all possible crossover events.</p>


<h3>Usage</h3>

<pre><code class='language-R'>co(genotypeMatrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="co_+3A_genotypematrix">genotypeMatrix</code></td>
<td>

<p><code>matrix</code> half-sib genotypes  (one half-sib per row, with SNP ordered by mapping position in the columns. Data should be numeric. Use 0, 1 and 2 for respectively AA, AB and BB. Use 9 for missing data)

</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with the number of crossover events for each site.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(           # Define a Half-sib Genotype Matrix
  2,1,0,                        # Individual 1
  2,0,2,                        # Individual 2
  0,0,2                         # Individual 3
  ), byrow = TRUE, ncol = 3)    # There are 3 individuals with three SNPs
  
co(genotype)               

</code></pre>

<hr>
<h2 id='cs'>
Chromosome Splitter
</h2><span id='topic+cs'></span>

<h3>Description</h3>

<p>This function splits the genotypes list generated by <code><a href="#topic+hss">hss</a></code> into the different chromosomes based on a map file and orders SNP based on chromosomal position.</p>


<h3>Usage</h3>

<pre><code class='language-R'>cs(halfsib, mapPath, separator = " ")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cs_+3A_halfsib">halfsib</code></td>
<td>

<p><code>list</code> list with matrices of half-sib genotypes, one family per list item </p>
</td></tr>
<tr><td><code id="cs_+3A_mappath">mapPath</code></td>
<td>

<p><code>character</code> path to the map file  (column 1 -&gt; SNP names, column 2 -&gt; chromosome name and column 3 -&gt; SNP position in base pairs) or, alternatively, the name of a dataframe with the mapping information (in the same format)  
</p>
</td></tr>
<tr><td><code id="cs_+3A_separator">separator</code></td>
<td>

<p><code>character</code> separator character used in the the map file
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The map file should include only the chromosomes that will be analyzed. For example, the Y and X chromosomes should be excluded (and others optionally). Names of each element in the list can be used for further categorization.
The header must be &quot;Name Chr Position&quot;.
</p>


<h3>Value</h3>

<p>Returns a list of matrices, the number of elements in this list is the number of half-sib families multiplied by the number of chromosomes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Please run demo(hsphase)
</code></pre>

<hr>
<h2 id='genotypes'>Example of Genotype Data Set</h2><span id='topic+genotypes'></span>

<h3>Description</h3>

<p>This data set serves as an example of a genotype matrix intended for use with the <code>hsphase</code> package. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(genotypes)
</code></pre>


<h3>Format</h3>

<p>The data set is a genotype matrix with specific structure, including:
</p>

<ul>
<li> <p><strong>Columns:</strong> Represent Single Nucleotide Polymorphisms (SNPs). Each column corresponds to a specific SNP.
</p>
</li>
<li> <p><strong>Rows:</strong> Represent individual animals. Each row corresponds to the genotypic data for a single animal across various SNPs.
</p>
</li></ul>


<hr>
<h2 id='hbp'>
Haplotype Blocks of Phased Data
</h2><span id='topic+hbp'></span>

<h3>Description</h3>

<p>Creates a blocking structure matrix of the half-sib family based on phased data of the sire and half-sib family.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hbp(PhasedGenotypeMatrix, PhasedSireGenotype, strand = "auto")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hbp_+3A_phasedgenotypematrix">PhasedGenotypeMatrix</code></td>
<td>
<p><code>matrix</code> haplotypes for a half-sib family (two rows per individual) </p>
</td></tr>
<tr><td><code id="hbp_+3A_phasedsiregenotype">PhasedSireGenotype</code></td>
<td>
<p><code>matrix</code> haplotypes of sire</p>
</td></tr>
<tr><td><code id="hbp_+3A_strand">strand</code></td>
<td>
<p><code>character</code> method for identification of paternal strand (1 and 2 for strand one and two of the offsprings)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix where 3 or 4 stands for the SNP originating in, respectively, strands 1 and 2. 0 indicates that the source strand for the SNP is unknown.
</p>


<h3>Note</h3>

<p>The input matrices must only contain individuals from one half-sib family and one ordered chromosome.
The <code>strand</code> option should be set to &quot;auto&quot; (default value).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aio">aio</a></code>, <code><a href="#topic+ssp">ssp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sire &lt;- matrix(c(
  0,0,0,0,0,1,                  # Haplotype one of the sire
  0,1,1,1,1,0                   # Haplotype two of the sire
  ), byrow = TRUE, ncol = 6)
  
haplotypeHalfsib &lt;- matrix(c(
  1,0,1,1,1,1,                  # Individual one, haplotype one
  0,1,0,0,0,0,                  # Individual one, haplotype two
  0,1,1,0,1,1,                  # Individual two, haplotype one
  1,0,0,1,0,0                   # Individual two, haplotype two
  ), byrow = TRUE, ncol = 6)    # 0s and 1s are alelle a and b
  
 hbp(haplotypeHalfsib, sire)
</code></pre>

<hr>
<h2 id='hh'>
Heatmap of Half-sibs
</h2><span id='topic+hh'></span>

<h3>Description</h3>

<p>The <code>hh</code> function creates a heatmap of the half-sib families using the
matrix of opposing homozygotes.</p>


<h3>Usage</h3>

<pre><code class='language-R'>	hh(oh, inferredPedigree, realPedigree, pedOnly = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hh_+3A_oh">oh</code></td>
<td>

<p><code>matrix</code> Opposing homozygotes matrix (output of <code>ohg</code>)
</p>
</td></tr>
<tr><td><code id="hh_+3A_inferredpedigree">inferredPedigree</code></td>
<td>
<p><code>matrix</code>
inferred pedigree (output of <code>rpoh</code>)
</p>
</td></tr>
<tr><td><code id="hh_+3A_realpedigree">realPedigree</code></td>
<td>
<p><code>matrix</code>
original pedigree
</p>
</td></tr>
<tr><td><code id="hh_+3A_pedonly">pedOnly</code></td>
<td>
<p><code>logical</code>
Consider only individuals that are exist in the real pedigree 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the heatmap of the matrix of opposing homozygotes with sidebars colour coded by sires from the inferred and original pedigree.
</p>


<h3>Author(s)</h3>

<p>The fuction uses the colour generated by <em>getcol</em> function in the <em>made4</em> package (Aedin Culhane).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ohg">ohg</a></code> and <code><a href="#topic+rpoh">rpoh</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>c1h1 &lt;- .simulateHalfsib(numInd = 62, numSNP = 5000)
c1h2 &lt;- .simulateHalfsib(numInd = 38, numSNP = 5000)
Genotype &lt;- rbind(c1h1, c1h2)
oh &lt;- ohg(Genotype) # creating the Opposing Homozygote matrix
hh(oh)
</code></pre>

<hr>
<h2 id='hss'>
Half-sib Family Splitter</h2><span id='topic+hss'></span>

<h3>Description</h3>

<p>Splits the dataset into half-sib family groups based on a pedigree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hss(pedigree, genotype, check = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hss_+3A_pedigree">pedigree</code></td>
<td>

<p><code>matrix</code>  the pedigree matrix should contain at least two columns, the first column with the half-sib IDs
and the second column with the sires IDs
</p>
</td></tr>
<tr><td><code id="hss_+3A_genotype">genotype</code></td>
<td>

<p><code>matrix</code> genotype matrix with SNP ordered by mapping position in the columns. Data should be numeric. Use 0, 1 and 2 respectively for AA, AB and BB. Use 9 for missing data
</p>
</td></tr> 
<tr><td><code id="hss_+3A_check">check</code></td>
<td>

<p><code>logical</code> check the genotype file for the possible errors
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Only half-sib groups that have more than 3 individuals will be returned.
</p>


<h3>Value</h3>

<p>Returns a list of numeric matrices, each matrix is a half-sib family.
</p>


<h3>Note</h3>

<p>Pedigree must have at least two columns with sample ids (Column 1) and sire ids (Column 2).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Please run demo(hsphase)
</code></pre>

<hr>
<h2 id='imageplot'>
Image Plot of Blocking Structure</h2><span id='topic+imageplot'></span>

<h3>Description</h3>

<p>Create an imageplot of the blocking structure.</p>


<h3>Usage</h3>

<pre><code class='language-R'>imageplot(x, title, rv = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="imageplot_+3A_x">x</code></td>
<td>
<p><code>matrix</code> blocking structure (output of <a href="#topic+bmh">bmh</a> or <a href="#topic+hbp">hbp</a> functions)
</p>
</td></tr>
<tr><td><code id="imageplot_+3A_title">title</code></td>
<td>
<p><code>character</code> title of imageplot
</p>
</td></tr>
<tr><td><code id="imageplot_+3A_rv">rv</code></td>
<td>
<p><code>logical</code>
reverse the colour
</p>
</td></tr>
<tr><td><code id="imageplot_+3A_...">...</code></td>
<td>
<p>Can be used to set xLabels and yLabels <a href="graphics.html#topic+axis">axis</a>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>White indicates regions of unknown origin, red and blue correspond to the two sire strands.
</p>


<h3>Author(s)</h3>

<p>This is a modified version of a function written by Chris Seidel. <br />
http://www.phaget4.org/R/image_matrix.html
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bmh">bmh</a></code> and <code><a href="#topic+aio">aio</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(
0,2,1,1,1,
2,0,1,2,2,
2,2,1,0,2,
2,2,1,1,1,
0,0,2,1,0), ncol = 5, byrow = TRUE) # each row contains the SNP of individuals
imageplot(bmh(genotype))
</code></pre>

<hr>
<h2 id='impute'>
Impute of Low Density SNP Marker to High Density (Paternal Strand)</h2><span id='topic+impute'></span>

<h3>Description</h3>

<p>Impute the paternal strand from low density to high density utilising high density sire haplotype.</p>


<h3>Usage</h3>

<pre><code class='language-R'>impute(halfsib_genotype_ld, sire_hd, bmh_forwardVectorSize = 30, 
bmh_excludeFP = TRUE, bmh_nsap = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="impute_+3A_halfsib_genotype_ld">halfsib_genotype_ld</code></td>
<td>
<p><code>matrix</code>
half-sib genotypes with low density marker (one half-sib per row, with SNP ordered by mapping position in the columns.
Data should be numeric. Use 0, 1 and 2 respectively for AA, AB and BB. Use 9 for missing data)</p>
</td></tr>
<tr><td><code id="impute_+3A_sire_hd">sire_hd</code></td>
<td>
<p><code>matrix</code>
haplotype of sire (this parameter can be sequence data or any phased sire - the matrix should have rownames which are the sample
IDs and colnames which are the SNP names)</p>
</td></tr>
<tr><td><code id="impute_+3A_bmh_forwardvectorsize">bmh_forwardVectorSize</code></td>
<td>

<p><code>integer</code> number of heterozygous sites used to validate recombination events or check for genotyping errors</p>
</td></tr>
<tr><td><code id="impute_+3A_bmh_excludefp">bmh_excludeFP</code></td>
<td>

<p><code>logical</code> exclude SNPs that may cause heterozygous sites in the sire due to genotyping errors or map errors</p>
</td></tr>
<tr><td><code id="impute_+3A_bmh_nsap">bmh_nsap</code></td>
<td>

<p><code>integer</code> number of SNPs per block</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return an imputed half-sib matrix. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bmh">bmh</a></code>,  <code><a href="#topic+ssp">ssp</a></code> and  <code><a href="#topic+phf">phf</a></code>
</p>

<hr>
<h2 id='map'>Example Map File for Genetic Data</h2><span id='topic+map'></span>

<h3>Description</h3>

<p>This data set is an example of a map file used within the <code>hsphase</code> package to demonstrate the mapping of SNPs to their respective locations on chromosomes. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(map)
</code></pre>


<h3>Format</h3>

<p>The data set is formatted as a data frame with the following columns, providing essential information about each SNP:
</p>

<ul>
<li> <p><strong>Name:</strong> The unique identifier or name of the SNP.
</p>
</li>
<li> <p><strong>Chr:</strong> The chromosome on which the SNP is located.
</p>
</li>
<li> <p><strong>Position:</strong> The position of the SNP on the chromosome, expressed in base pairs.
</p>
</li></ul>


<hr>
<h2 id='ohd'>
Opposing Homozygote Detection</h2><span id='topic+ohd'></span>

<h3>Description</h3>

<p>Counts the number of opposing homozygotes for each animal that caused a heterozygus site in the sire.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ohd(genotypeMatrix, unique_check = FALSE, SNPs = 6000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ohd_+3A_genotypematrix">genotypeMatrix</code></td>
<td>

<p><code>matrix</code> half-sib genotypes (one half-sib per row, with SNP ordered by mapping position in the columns. Data should be numeric. Use 0, 1 and 2 respectively for AA, AB and BB. Use 9 for missing data)

</p>
</td></tr>
<tr><td><code id="ohd_+3A_unique_check">unique_check</code></td>
<td>

<p><code>logical</code> check if samples uniquely originate an opposing homozygote at a locus

</p>
</td></tr>
<tr><td><code id="ohd_+3A_snps">SNPs</code></td>
<td>

<p><code>integer</code> number of SNP to use

</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a vector with the number of heterozygous sites that each sample caused.
</p>


<h3>Note</h3>

<p>This function can be used to identify pedigree errors; i.e., the outliers.
</p>


<h3>Author(s)</h3>

<p>This method is suggested by Bruce Tier &lt;btier@une.edu.au&gt; to identify pedigree errors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(        
  2,1,0,                  
  2,0,0,                  
  0,0,2                   
  ), byrow = TRUE, ncol = 3)    
  
ohd(genotype)             
</code></pre>

<hr>
<h2 id='ohg'>
Matrix of Opposing Homozygotes 
</h2><span id='topic+ohg'></span>

<h3>Description</h3>

<p>Creates a matrix of opposing homozygotes from the genotype matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ohg(genotypeMatrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ohg_+3A_genotypematrix">genotypeMatrix</code></td>
<td>

<p><code>matrix</code> genotype (Data should be numeric. Use 0, 1 and 2 respectively for AA, AB and BB. Use 9 for missing data)

</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a square matrix (sample X sample) with the pairwise counts of opposing homozygotes.
</p>


<h3>Note</h3>

<p>This function can be slow with a large data set. The fast version of this function will be available after publish of the related manuscript.
</p>


<h3>Author(s)</h3>

<p>Ferdosi, M. H., &amp; Boerner, V. (2014). A fast method for evaluating opposing homozygosity in large SNP data sets. Livestock Science.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rpoh">rpoh</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(        
  2,1,0,                  
  2,0,0,                  
  0,0,2                   
  ), byrow = TRUE, ncol = 3)    
  
ohg(genotype)             
</code></pre>

<hr>
<h2 id='ohplot'>
Opposing Homozygotes Plot
</h2><span id='topic+ohplot'></span>

<h3>Description</h3>

<p>Plot the sorted vectorized matrix of Opposing Homozygotes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ohplot(oh, genotype, pedigree, check = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ohplot_+3A_oh">oh</code></td>
<td>

<p><code>integer</code> Opposing homozygotes matrix (Output of <code><a href="#topic+ohg">ohg</a></code>) </p>
</td></tr>
<tr><td><code id="ohplot_+3A_genotype">genotype</code></td>
<td>

<p><code>matrix</code> genotype of one chromosome (data should be numeric. Use 0, 1 and 2 for respectively AA, AB and BB. Use 9 for missing data)
</p>
</td></tr>
<tr><td><code id="ohplot_+3A_pedigree">pedigree</code></td>
<td>

<p><code>matrix</code>  the pedigree matrix should contain at least two columns, the first column with the half-sib IDs
and the second column with the sires IDs. This argument is optional.
</p>
</td></tr>
<tr><td><code id="ohplot_+3A_check">check</code></td>
<td>

<p><code>logical</code> check the genotype file for the possible errors
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>cut off</code> line shows the edge of most different groups. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ohg">ohg</a></code> and <code><a href="#topic+rpoh">rpoh</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100)
chr &lt;- list()
sire &lt;- list()
set.seed(1)
chr &lt;- list()
for(i in 1:5)
{
	chr[[i]] &lt;- .simulateHalfsib(numInd = 20, numSNP = 5000, recbound = 1:10)
	sire[[i]] &lt;- ssp(bmh(chr[[i]]), chr[[i]])
	sire[[i]] &lt;- sire[[i]][1,] + sire[[i]][2,]
	sire[[i]][sire[[i]] == 18] &lt;- 9
}

Genotype &lt;- do.call(rbind, chr)
rownames(Genotype) &lt;- 6:(nrow(Genotype) + 5)
sire &lt;- do.call(rbind, sire)
rownames(sire) &lt;- 1:5
Genotype &lt;- rbind(sire, Genotype)
oh &lt;- ohg(Genotype)  # creating the Opposing Homozygote matrix
pedigree &lt;- as.matrix(data.frame(c(1:5, 6:(nrow(Genotype))), 
rep = c(rep(0,5), rep(1:5, rep(20,5)))))
ohplot(oh, Genotype, pedigree, check = TRUE)
</code></pre>

<hr>
<h2 id='para'>
Parallel Analysis of Data
</h2><span id='topic+para'></span>

<h3>Description</h3>

<p>This function uses the list of matrices (the output of <code><a href="#topic+cs">cs</a></code>) and runs one of the options, on each element of the list, in parallel. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>para(halfsibs, cpus = 1, option = "bmh", type = "SOCK", bmh_forwardVectorSize = 30,
bmh_excludeFP = TRUE, bmh_nsap = 3, pmMethod = "constant")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="para_+3A_halfsibs">halfsibs</code></td>
<td>

<p><code>list</code> list of matrices of half-sibs (can be generated with <code><a href="#topic+hss">hss</a></code> and <code><a href="#topic+cs">cs</a></code> functions)
</p>
</td></tr>
<tr><td><code id="para_+3A_cpus">cpus</code></td>
<td>

<p><code>numeric</code> number of CPUs (thread)
</p>
</td></tr>
<tr><td><code id="para_+3A_option">option</code></td>
<td>

<p><code>character</code>  type of analysis
</p>
</td></tr>
<tr><td><code id="para_+3A_type">type</code></td>
<td>
<p><code>character</code>
type of cluster for parallel analysis
</p>
</td></tr>
<tr><td><code id="para_+3A_bmh_forwardvectorsize">bmh_forwardVectorSize</code></td>
<td>

<p><code>integer</code> number of heterozygous sites used to validate recombination events or check for genotyping errors</p>
</td></tr>
<tr><td><code id="para_+3A_bmh_excludefp">bmh_excludeFP</code></td>
<td>

<p><code>logical</code> exclude SNPs that may cause heterozygous sites in the sire due to genotyping errors or map errors</p>
</td></tr>
<tr><td><code id="para_+3A_bmh_nsap">bmh_nsap</code></td>
<td>

<p><code>integer</code> number of SNPs per block</p>
</td></tr>
<tr><td><code id="para_+3A_pmmethod">pmMethod</code></td>
<td>

<p><code>character</code> method for creating the recombination matrix</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Type of analysis  can be  <code><a href="#topic+bmh">bmh</a></code>, <code><a href="#topic+ssp">ssp</a></code>, <code><a href="#topic+aio">aio</a></code>, <code><a href="#topic+pm">pm</a></code>, or rec (refer to <code><a href="#topic+pm">pm</a></code>, <code><a href="#topic+rplot">rplot</a></code> and vignette for more information about rec).
</p>


<h3>Value</h3>

<p>Returns a list of matrices with the results (formats specific to the option selected).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Please run demo(hsphase)

</code></pre>

<hr>
<h2 id='pedigree'>Example Pedigree Data Set</h2><span id='topic+pedigree'></span>

<h3>Description</h3>

<p>This dataset provides an example of a pedigree, specifically designed for use with the <code>hsphase</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pedigree)
</code></pre>


<h3>Format</h3>

<p>The dataset is structured as a data frame with detailed familial relationships, including:
</p>

<ul>
<li> <p><strong>First Column:</strong> Identifiers for half-sibs. 
</p>
</li>
<li> <p><strong>Second Column:</strong> Identifiers for sires. 
</p>
</li></ul>


<hr>
<h2 id='pedigreeNaming'>
Fix Pedigree Errors
</h2><span id='topic+pedigreeNaming'></span>

<h3>Description</h3>

<p>Tries to link the inferred pedigree from <code><a href="#topic+rpoh">rpoh</a></code> with the sire IDs in the original pedigree and fix pedigree errors.</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigreeNaming(inferredPedigree, realPedigree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pedigreeNaming_+3A_inferredpedigree">inferredPedigree</code></td>
<td>

<p><code>matrix</code> inferred pedigree  (output of <code><a href="#topic+rpoh">rpoh</a></code> )
</p>
</td></tr>
<tr><td><code id="pedigreeNaming_+3A_realpedigree">realPedigree</code></td>
<td>

<p><code>matrix</code>  original pedigree
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calls the <code><a href="#topic+bmh">bmh</a></code> and <code><a href="#topic+recombinations">recombinations</a></code> functions to count the number of recombinations in each half-sib group.  
</p>


<h3>Value</h3>

<p>Returns the inferred pedigree with the best fit to the sire names used in the original pedigree file.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rpoh">rpoh</a></code> and <code><a href="#topic+ohg">ohg</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Please run demo(hsphase)
</code></pre>

<hr>
<h2 id='phf'>
Half-Sib Family Phasing
</h2><span id='topic+phf'></span>

<h3>Description</h3>

<p>Phases the half-sib family by using the blocking structure and imputed sire matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phf(GenotypeMatrix, blockMatrix, sirePhasedMatrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phf_+3A_genotypematrix">GenotypeMatrix</code></td>
<td>
<p><code>matrix</code> half-sib genotypes  (one half-sib per row, with SNP ordered by mapping position in the columns. Data should be numeric. Use 0, 1 and 2 respectively for AA, AB and BB. Use 9 for missing data)
</p>
</td></tr>
<tr><td><code id="phf_+3A_blockmatrix">blockMatrix</code></td>
<td>

<p><code>matrix</code> blocking structure (output of <code><a href="#topic+bmh">bmh</a></code>)
</p>
</td></tr>
<tr><td><code id="phf_+3A_sirephasedmatrix">sirePhasedMatrix</code></td>
<td>
<p><code>matrix</code>  imputed sire (output of <code><a href="#topic+ssp">ssp</a></code>)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix that contains the phased parental haplotypes of the half-sibs. It uses 1, 0 and 9 for A, B and missing.
</p>


<h3>Note</h3>

<p>The genotype matrix must only contain individuals from one half-sib family and one ordered chromosome.
This function is used by the <code><a href="#topic+aio">aio</a></code> function for complete phasing of a half-sib group.</p>


<h3>See Also</h3>

<p><code><a href="#topic+aio">aio</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(
  2,1,0,
  2,0,0,
  0,0,2), byrow = TRUE, ncol = 3)
block &lt;- bmh(genotype)
phf(genotype, block, ssp(block, genotype))
</code></pre>

<hr>
<h2 id='pm'>
Probability Matrix
</h2><span id='topic+pm'></span>

<h3>Description</h3>

<p>Creates a recombination matrix based on the blocking structure.</p>


<h3>Usage</h3>

<pre><code class='language-R'>pm(blockMatrix, method = "constant") 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pm_+3A_blockmatrix">blockMatrix</code></td>
<td>
 
<p><code>matrix</code> blocking structure (Output of <code><a href="#topic+bmh">bmh</a></code>)</p>
</td></tr>
<tr><td><code id="pm_+3A_method">method</code></td>
<td>

<p><code>character</code> method for creating the recombination matrix</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function finds the recombination between two consecutive sites, and marks the recombination site with a 1; if there are unknown sites between two blocks it will also mark these sites with a 1 (<code>constant</code> method) or 1 divided by number of unknown site (<code>relative</code> method).</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(
0,2,0,1,0,
2,0,1,2,2,
2,2,1,0,2,
2,2,1,1,1,
0,0,2,1,0), ncol = 5, byrow = TRUE) # each row contains the SNP of individuals

(result &lt;- bmh(genotype))
pm(result)
</code></pre>

<hr>
<h2 id='pogc'>
Parent Offspring Group Constructor
</h2><span id='topic+pogc'></span>

<h3>Description</h3>

<p>Assign offsprings to the parents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pogc(oh, genotypeError)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pogc_+3A_oh">oh</code></td>
<td>
 <p><code>integer</code> opposing homozygotes matrix (Output of <code><a href="#topic+ohg">ohg</a></code>)
</p>
</td></tr>
<tr><td><code id="pogc_+3A_genotypeerror">genotypeError</code></td>
<td>
<p><code>integer</code>
number of genotypeing error allowed in the <code>oh</code> matrix
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a data frame with two columns. The first column is the animal ID and the second column is the parent ID.  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ohg">ohg</a></code>, <code><a href="#topic+hss">hss</a></code> and <code><a href="#topic+rpoh">rpoh</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100)
chr &lt;- list()
sire &lt;- list()
set.seed(1)
chr &lt;- list()
for(i in 1:5)
{
	chr[[i]] &lt;- .simulateHalfsib(numInd = 20, numSNP = 5000, recbound = 1:10)
	sire[[i]] &lt;- ssp(bmh(chr[[i]]), chr[[i]])
	sire[[i]] &lt;- sire[[i]][1,] + sire[[i]][2,]
	sire[[i]][sire[[i]] == 18] &lt;- 9
}

Genotype &lt;- do.call(rbind, chr)
rownames(Genotype) &lt;- 6:(nrow(Genotype) + 5)
sire &lt;- do.call(rbind, sire)
rownames(sire) &lt;- 1:5
Genotype &lt;- rbind(sire, Genotype)
oh &lt;- ohg(Genotype)  # creating the Opposing Homozygote matrix
pogc(oh, 5)
</code></pre>

<hr>
<h2 id='readGenotype'>
Read and Check the Genotype File</h2><span id='topic+readGenotype'></span>

<h3>Description</h3>

<p>This function reads and checks genotype files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readGenotype(genotypePath, separatorGenotype = " ", check = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readGenotype_+3A_genotypepath">genotypePath</code></td>
<td>

<p><code>character</code> genotype path (animals (rows) and SNP (columns), SNP should be coded as 0, 1 and 2 for respectively AA, AB and BB.
Use 9 for missing data. please refer to vignette for more information)
</p>
</td></tr>
<tr><td><code id="readGenotype_+3A_separatorgenotype">separatorGenotype</code></td>
<td>

<p><code>character</code> separator character for genotype
</p>
</td></tr>
<tr><td><code id="readGenotype_+3A_check">check</code></td>
<td>

<p><code>logical</code> check the genotype file for possible errors
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the genotype matrix.
</p>


<h3>Note</h3>

<p>Please refer to vignette for more information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A comprehensive demo and example dataset is available from
# http://www-personal.une.edu.au/~cgondro2/hsphase.html
</code></pre>

<hr>
<h2 id='recombinations'>
Recombination Number
</h2><span id='topic+recombinations'></span>

<h3>Description</h3>

<p>Counts the number of recombinations for each individual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recombinations(blockMatrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="recombinations_+3A_blockmatrix">blockMatrix</code></td>
<td>

<p><code>matrix</code> block structure (Output of <code><a href="#topic+bmh">bmh</a></code>)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a vector of recombinations. The number of elements in this vector is equal to the number of individuals,  
i.e. each element holds the number of recombinations identified for each sample.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bmh">bmh</a>
</code></p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(          
  2,1,0,0,                    
  2,0,2,2,                    
  0,0,2,2,
  0,2,0,0                     
  ), byrow = TRUE, ncol = 4)      
  
recombinations(bmh(genotype))             
</code></pre>

<hr>
<h2 id='rplot'>
Recombination Plot</h2><span id='topic+rplot'></span>

<h3>Description</h3>

<p>This function creates a plot which shows the sum of all recombination events across a half-sib family.</p>


<h3>Usage</h3>

<pre><code class='language-R'>rplot(x, distance, start = 1, end = ncol(x), maximum = 100,
overwrite = FALSE, method = "constant")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rplot_+3A_x">x</code></td>
<td>
<p><code>matrix</code> of half-sib genotypes  (one half-sib per row, with SNP ordered by mapping position in the columns. Data should be numeric. Use 0, 1 and 2 for respectively AA, AB and BB. Use 9 for missing data).
</p>
</td></tr>
<tr><td><code id="rplot_+3A_distance">distance</code></td>
<td>
<p><code>integer</code> of physical distances between markers
</p>
</td></tr>
<tr><td><code id="rplot_+3A_start">start</code></td>
<td>
<p><code>integer</code>
first marker selected for the plot</p>
</td></tr>
<tr><td><code id="rplot_+3A_end">end</code></td>
<td>
<p><code>integer</code>
last marker selected for the plot</p>
</td></tr>
<tr><td><code id="rplot_+3A_maximum">maximum</code></td>
<td>
<p><code>integer</code>
maximum number of recombinations to show (higher recombination rates will be omitted from the plot)</p>
</td></tr>
<tr><td><code id="rplot_+3A_overwrite">overwrite</code></td>
<td>
<p><code>logical</code>
draw a diagram over the current diagram  (default <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="rplot_+3A_method">method</code></td>
<td>
<p><code>character</code> please refer to the <code><a href="#topic+pm">pm</a></code> document</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(
0,2,0,1,0,
2,0,1,2,2,
2,2,1,0,2,
2,2,1,1,1,
0,0,2,1,0), ncol = 5, byrow = TRUE) # each row contains the SNP of individuals

rplot(genotype, c(1,2,3,4,8)) 
</code></pre>

<hr>
<h2 id='rpoh'>
Reconstruct Pedigree Based on Matrix of Opposing Homozygotes</h2><span id='topic+rpoh'></span>

<h3>Description</h3>

<p>Reconstructs a half-sib pedigree based on a matrix of opposing homozygotes.</p>


<h3>Usage</h3>

<pre><code class='language-R'>rpoh(genotypeMatrix, oh, forwardVectorSize = 30, excludeFP = TRUE, nsap = 3,
maxRec = 15, intercept = 26.3415, coefficient = 77.3171, snpnooh, method, maxsnpnooh)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rpoh_+3A_genotypematrix">genotypeMatrix</code></td>
<td>

<p><code>matrix</code> genotype of one chromosome (data should be numeric. Use 0, 1 and 2 for respectively AA, AB and BB. Use 9 for missing data)
</p>
</td></tr>
<tr><td><code id="rpoh_+3A_oh">oh</code></td>
<td>

<p><code>integer</code> Opposing homozygotes matrix (Output of <code><a href="#topic+ohg">ohg</a></code>) </p>
</td></tr>
<tr><td><code id="rpoh_+3A_forwardvectorsize">forwardVectorSize</code></td>
<td>

<p><code>integer</code> number of heterozygous sites used to validate recombination events or check for genotyping errors</p>
</td></tr>
<tr><td><code id="rpoh_+3A_excludefp">excludeFP</code></td>
<td>

<p><code>logical</code> excludes SNPs that may cause heterozygous sites in the sire due to genotyping errors or map errors</p>
</td></tr>
<tr><td><code id="rpoh_+3A_nsap">nsap</code></td>
<td>

<p><code>integer</code> number of SNP per block to validate recombinations</p>
</td></tr>
<tr><td><code id="rpoh_+3A_maxrec">maxRec</code></td>
<td>

<p><code>integer</code> maximum number of expected recombinations per individual</p>
</td></tr>
<tr><td><code id="rpoh_+3A_intercept">intercept</code></td>
<td>

<p><code>integer</code> intercept of fitted model</p>
</td></tr>
<tr><td><code id="rpoh_+3A_coefficient">coefficient</code></td>
<td>

<p><code>integer</code> coefficient of fitted model</p>
</td></tr>
<tr><td><code id="rpoh_+3A_snpnooh">snpnooh</code></td>
<td>

<p><code>integer</code> number of SNPs used to create <em>oh</em> matrix (this number must be divided by 1000)</p>
</td></tr>
<tr><td><code id="rpoh_+3A_method">method</code></td>
<td>

<p><code>character</code> pedigree reconstruction method</p>
</td></tr>
<tr><td><code id="rpoh_+3A_maxsnpnooh">maxsnpnooh</code></td>
<td>

<p><code>numeric</code> the maximum number of allowing opposing homozygote in a half-sib family
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Four methods <em>simple</em>, <em>recombinations</em>, <em>calus</em> and <em>manual</em> can be
utilized to reconstruct the pedigree. 
<br /><br />
The following examples show the arguments require for each method.
</p>
<p>pedigree1 &lt;- rpoh(oh = oh, snpnooh = 732, method = &quot;simple&quot;) <br />
pedigree2 &lt;- rpoh(genotypeMatrix = genotypeChr1, oh = ohg(genotype), maxRec = 10 , method = &quot;recombinations&quot;)  <br />
pedigree3 &lt;- rpoh(genotypeMatrix = genotype, oh = oh, method = &quot;calus&quot;) <br />
pedigree4 &lt;- rpoh(oh = oh, maxsnpnooh = 31662, method = &quot;manual&quot;)
<br />
</p>


<h3>Value</h3>

<p>Returns a data frame with two columns, the first column is animals' ID and the second column is sire identifiers (randomly generated).
</p>


<h3>Note</h3>

<p>Method can be <em>recombinations</em>, <em>simple</em>, <em>calus</em> or <em>manual</em>. Please refer to vignette for more information.<br />
</p>
<p>The sire genotype should be removed before using this function utilizing  <code><a href="#topic+pogc">pogc</a></code> function.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bmh">bmh</a></code> and <code><a href="#topic+recombinations">recombinations</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Please run demo(hsphase)
</code></pre>

<hr>
<h2 id='ssp'>
Sire Imputation and Phasing
</h2><span id='topic+ssp'></span>

<h3>Description</h3>

<p>Infer (impute) and phase sire's genotype based on the block structure matrix (recombination blocks) and homozygous sites of the half-sib genotype matrix.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssp(blockMatrix, genotypeMatrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ssp_+3A_blockmatrix">blockMatrix</code></td>
<td>

<p><code>matrix</code> block structure (Output of <code><a href="#topic+bmh">bmh</a></code>)
</p>
</td></tr>
<tr><td><code id="ssp_+3A_genotypematrix">genotypeMatrix</code></td>
<td>

<p><code>matrix</code> half-sibs genotype
(each row includes the SNP of individuals, 0, 1 and 2 for respectively AA, AB and BB. Use 9 for missing data) </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix (Imputed Sire) with two rows one for each haplotype of the sire (columns are SNP in the order of the genotype matrix). Alleles are coded as 0 (A) and 1 (B). Alleles that could not be imputed are coded as 9.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phf">phf</a></code>, <code><a href="#topic+aio">aio</a></code> and <code><a href="#topic+imageplot">imageplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- matrix(c(
0,2,1,1,1,
2,0,1,2,2,
2,2,1,0,2,
2,2,1,1,1,
0,0,2,1,0), ncol = 5, byrow = TRUE) # each row contains the SNP of individuals

(result &lt;- ssp(bmh(genotype), genotype)) 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
