<!DOCTYPE html><html><head><title>Help for package hsrecombi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hsrecombi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bestmapfun'><p>Best fitting genetic-map function</p></a></li>
<li><a href='#checkCandidates'><p>Candidates for misplacement</p></a></li>
<li><a href='#countNumbers'><p>Count genotype combinations at 2 SNPs</p></a></li>
<li><a href='#daughterSire'><p>targetregion: allocation of paternal half-sib families</p></a></li>
<li><a href='#editraw'><p>Editing results of hsrecombi</p></a></li>
<li><a href='#felsenstein'><p>Felsenstein's genetic map function</p></a></li>
<li><a href='#geneticPosition'><p>Estimation of genetic position</p></a></li>
<li><a href='#genotype.chr'><p>targetregion: progeny genotypes</p></a></li>
<li><a href='#haldane'><p>Haldane's genetic map function</p></a></li>
<li><a href='#hapSire'><p>targetregion: sire haplotypes</p></a></li>
<li><a href='#hsrecombi'><p>Estimation of recombination rate and maternal LD</p></a></li>
<li><a href='#karlin'><p>Liberman and Karlin's genetic map function</p></a></li>
<li><a href='#kosambi'><p>Kosambi's genetic map function</p></a></li>
<li><a href='#LDHScpp'><p>Expectation Maximisation (EM) algorithm</p></a></li>
<li><a href='#loglikfun'><p>Calculate log-likelihood function</p></a></li>
<li><a href='#makehap'><p>Make list of imputed sire haplotypes</p></a></li>
<li><a href='#makehaplist'><p>Make list of sire haplotypes</p></a></li>
<li><a href='#makehappm'><p>Make list of imputed haplotypes and estimate recombination rate</p></a></li>
<li><a href='#map.chr'><p>targetregion: physical map</p></a></li>
<li><a href='#rao'><p>System of genetic-map functions</p></a></li>
<li><a href='#rao inverse'><p>Approximation to inverse of Rao's system of map functions</p></a></li>
<li><a href='#startvalue'><p>Start value for maternal allele and haplotype frequencies</p></a></li>
<li><a href='#targetregion'><p>Description of the targetregion data set</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimation of Recombination Rate and Maternal LD in Half-Sibs</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-06-07</td>
</tr>
<tr>
<td>Description:</td>
<td>Paternal recombination rate and maternal linkage disequilibrium 
    (LD) are estimated for pairs of biallelic markers such as single nucleotide 
    polymorphisms (SNPs) from progeny genotypes and sire haplotypes. The 
    implementation relies on paternal half-sib families. If maternal half-sib 
    families are used, the roles of sire/dam are swapped. Multiple families can 
    be considered. For parameter estimation, at least one sire has to be double 
    heterozygous at the investigated pairs of SNPs.
    Based on recombination rates, genetic distances between markers can be 
    estimated. Markers with unusually large recombination rate to markers in 
    close proximity (i.e. putatively misplaced markers) shall be discarded in 
    this derivation.
    A workflow description is attached as vignette.
    *A pipeline is available at GitHub*
    <a href="https://github.com/wittenburg/hsrecombi">https://github.com/wittenburg/hsrecombi</a>
    Hampel, Teuscher, Gomez-Raya, Doschoris, Wittenburg (2018) "Estimation of 
    recombination rate and maternal linkage disequilibrium in half-sibs" 
    &lt;<a href="https://doi.org/10.3389%2Ffgene.2018.00186">doi:10.3389/fgene.2018.00186</a>&gt;.
    Gomez-Raya (2012) "Maximum likelihood estimation of linkage disequilibrium 
    in half-sib families" &lt;<a href="https://doi.org/10.1534%2Fgenetics.111.137521">doi:10.1534/genetics.111.137521</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.3), hsphase, dplyr, data.table, rlist, quadprog,
curl, Matrix</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, formatR, AlphaSimR (&ge; 0.13.0), doParallel,
ggplot2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-07 08:00:12 UTC; wittenburg</td>
</tr>
<tr>
<td>Author:</td>
<td>Dörte Wittenburg [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dörte Wittenburg &lt;wittenburg@fbn-dummerstorf.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-07 08:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='bestmapfun'>Best fitting genetic-map function</h2><span id='topic+bestmapfun'></span>

<h3>Description</h3>

<p>Approximation of mixing parameter of system of map functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bestmapfun(theta, dist_M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bestmapfun_+3A_theta">theta</code></td>
<td>
<p>vector of recombination rates</p>
</td></tr>
<tr><td><code id="bestmapfun_+3A_dist_m">dist_M</code></td>
<td>
<p>vector of genetic positions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The genetic mapping function that fits best to the genetic data
(recombination rate and genetic distances) is obtained from Rao's system of
genetic-map functions. The corresponding mixing parameter is estimated via
1-dimensional constrained optimisation.
See vignette for its application to estimated data.
</p>


<h3>Value</h3>

<p>list (LEN 2)
</p>

<dl>
<dt>mixing</dt><dd><p>mixing parameter of system of genetic mapping functions</p>
</dd>
<dt>mse</dt><dd><p>minimum value of target function (theta - dist_M)^2</p>
</dd>
</dl>



<h3>References</h3>

<p>Rao, D.C., Morton, N.E., Lindsten, J., Hulten, M. &amp; Yee, S (1977)
A mapping function for man. Human Heredity 27: 99-104.
doi: <a href="https://doi.org/10.1159/000152856">10.1159/000152856</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  theta &lt;- seq(0, 0.5, 0.01)
  gendist &lt;- -log(1 - 2 * theta) / 2
  bestmapfun(theta, gendist)
</code></pre>

<hr>
<h2 id='checkCandidates'>Candidates for misplacement</h2><span id='topic+checkCandidates'></span>

<h3>Description</h3>

<p>Search for SNPs with unusually large estimates of recombination
rate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkCandidates(final, map1, win = 30, quant = 0.99)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkCandidates_+3A_final">final</code></td>
<td>
<p>table of results produced by <code>editraw</code> with pairwise
estimates of recombination rate between p SNPs within chromosome; minimum
required data frame with columns <code>SNP1</code>, <code>SNP2</code> and <code>theta</code></p>
</td></tr>
<tr><td><code id="checkCandidates_+3A_map1">map1</code></td>
<td>
<p>data.frame containing information on physical map, at least:
</p>

<dl>
<dt><code>SNP</code></dt><dd><p>SNP ID</p>
</dd>
<dt><code>locus_Mb</code></dt><dd><p>physical position in Mbp of SNP on chromosomes</p>
</dd>
<dt><code>Chr</code></dt><dd><p>chromosome of SNP</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="checkCandidates_+3A_win">win</code></td>
<td>
<p>optional value for window size; default value 30</p>
</td></tr>
<tr><td><code id="checkCandidates_+3A_quant">quant</code></td>
<td>
<p>optional value; default value 0.99, see details</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Markers with unusually large estimates of recombination rate to
close SNPs are candidates for misplacement in the underlying assembly. The
mean of recombination rate estimates with <code>win</code> subsequent or
preceding markers is calculated and those SNPs with mean value exceeding
the <code>quant</code> quantile are denoted as candidates which have to be
manually curated!
This can be done, for instance, by visual inspection of a correlation plot
containing estimates of recombination rate in a selected region.
</p>


<h3>Value</h3>

<p>vector of SNP IDs for further verification
</p>


<h3>References</h3>

<p>Hampel, A., Teuscher, F., Gomez-Raya, L., Doschoris, M. &amp; Wittenburg, D.
(2018) Estimation of recombination rate and maternal linkage
disequilibrium in half-sibs. Frontiers in Genetics 9:186.
doi: <a href="https://doi.org/10.3389/fgene.2018.00186">10.3389/fgene.2018.00186</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ### test data
  data(targetregion)
  ### make list for paternal half-sib families
  hap &lt;- makehaplist(daughterSire, hapSire)
  ### parameter estimates on a chromosome
  res &lt;- hsrecombi(hap, genotype.chr)
  ### post-processing to achieve final and valid set of estimates
  final &lt;- editraw(res, map.chr)
  ### check for candidates of misplacement
  snp &lt;- checkCandidates(final, map.chr)
</code></pre>

<hr>
<h2 id='countNumbers'>Count genotype combinations at 2 SNPs</h2><span id='topic+countNumbers'></span>

<h3>Description</h3>

<p>Count genotype combinations at 2 SNPs
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="countNumbers_+3A_x">X</code></td>
<td>
<p>integer matrix of genotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>count</code> vector of counts of 9 possible genotypes at SNP pair
</p>

<hr>
<h2 id='daughterSire'>targetregion: allocation of paternal half-sib families</h2><span id='topic+daughterSire'></span>

<h3>Description</h3>

<p>Vector of sire ID for each progeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>daughterSire
</code></pre>


<h3>Format</h3>

<p>An object of class <code>integer</code> of length 265.
</p>

<hr>
<h2 id='editraw'>Editing results of hsrecombi</h2><span id='topic+editraw'></span>

<h3>Description</h3>

<p>Process raw results from <code>hsrecombi</code>, decide which out of
two sets of estimates is more likely and prepare list of final results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>editraw(Roh, map1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="editraw_+3A_roh">Roh</code></td>
<td>
<p>list of raw results from <code>hsrecombi</code></p>
</td></tr>
<tr><td><code id="editraw_+3A_map1">map1</code></td>
<td>
<p>data.frame containing information on physical map, at least:
</p>

<dl>
<dt><code>SNP</code></dt><dd><p>SNP ID</p>
</dd>
<dt><code>locus_Mb</code></dt><dd><p>physical position in Mbp of SNP on chromosomes</p>
</dd>
<dt><code>Chr</code></dt><dd><p>chromosome of SNP</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Value</h3>

<p>final table of results
</p>

<dl>
<dt><code>SNP1</code></dt><dd><p>index 1. SNP</p>
</dd>
<dt><code>SNP2</code></dt><dd><p>index 2. SNP</p>
</dd>
<dt><code>D</code></dt><dd><p>maternal LD</p>
</dd>
<dt><code>fAA</code></dt><dd><p>frequency of maternal haplotype 1-1</p>
</dd>
<dt><code>fAB</code></dt><dd><p>frequency of maternal haplotype 1-0</p>
</dd>
<dt><code>fBA</code></dt><dd><p>frequency of maternal haplotype 0-1</p>
</dd>
<dt><code>fBB</code></dt><dd><p>frequency of maternal haplotype 0-0</p>
</dd>
<dt><code>p1</code></dt><dd><p>Maternal allele frequency (allele 1) at <code>SNP1</code></p>
</dd>
<dt><code>p2</code></dt><dd><p>Maternal allele frequency (allele 1) at <code>SNP2</code></p>
</dd>
<dt><code>nfam1</code></dt><dd><p>size of genomic family 1</p>
</dd>
<dt><code>nfam2</code></dt><dd><p>size of genomic family 2</p>
</dd>
<dt><code>error</code></dt><dd><p>0 if computations were without error; 1 if EM algorithm
did not converge</p>
</dd>
<dt><code>iteration</code></dt><dd><p>number of EM iterations</p>
</dd>
<dt><code>theta</code></dt><dd><p>paternal recombination rate</p>
</dd>
<dt><code>r2</code></dt><dd><p><code class="reqn">r^2</code> of maternal LD</p>
</dd>
<dt><code>logL</code></dt><dd><p>value of log likelihood function</p>
</dd>
<dt><code>unimodal</code></dt><dd><p>1 if likelihood is unimodal; 0 if likelihood is
bimodal</p>
</dd>
<dt><code>critical</code></dt><dd><p>0 if parameter estimates were unique; 1 if parameter
estimates were obtained via decision process</p>
</dd>
<dt><code>locus_Mb</code></dt><dd><p>physical distance between SNPs in Mbp</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>  ### test data
  data(targetregion)
  ### make list for paternal half-sib families
  hap &lt;- makehaplist(daughterSire, hapSire)
  ### parameter estimates on a chromosome
  res &lt;- hsrecombi(hap, genotype.chr)
  ### post-processing to achieve final and valid set of estimates
  final &lt;- editraw(res, map.chr)
</code></pre>

<hr>
<h2 id='felsenstein'>Felsenstein's genetic map function</h2><span id='topic+felsenstein'></span>

<h3>Description</h3>

<p>Calculation of genetic distances from recombination rates given
an interference parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>felsenstein(K, x, inverse = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="felsenstein_+3A_k">K</code></td>
<td>
<p>parameter (numeric) corresponding to the intensity of crossover
interference</p>
</td></tr>
<tr><td><code id="felsenstein_+3A_x">x</code></td>
<td>
<p>vector of recombination rates</p>
</td></tr>
<tr><td><code id="felsenstein_+3A_inverse">inverse</code></td>
<td>
<p>logical, if FALSE recombination rate is mapped to Morgan unit,
if TRUE Morgan unit is mapped to recombination rate (default is FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of genetic positions in Morgan units
</p>


<h3>References</h3>

<p>Felsenstein, J. (1979) A mathematically tractable family of
genetic mapping functions with different amounts of interference. Genetics
91:769-775.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  felsenstein(0.1, seq(0, 0.5, 0.01))
</code></pre>

<hr>
<h2 id='geneticPosition'>Estimation of genetic position</h2><span id='topic+geneticPosition'></span>

<h3>Description</h3>

<p>Estimation of genetic positions (in centi Morgan)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneticPosition(final, map1, exclude = NULL, threshold = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geneticPosition_+3A_final">final</code></td>
<td>
<p>table of results produced by <code>editraw</code> with pairwise
estimates of recombination rate between p SNPs within chromosome; minimum
required data frame with columns <code>SNP1</code>, <code>SNP2</code> and <code>theta</code></p>
</td></tr>
<tr><td><code id="geneticPosition_+3A_map1">map1</code></td>
<td>
<p>data.frame containing information on physical map, at least:
</p>

<dl>
<dt><code>SNP</code></dt><dd><p>SNP ID</p>
</dd>
<dt><code>locus_Mb</code></dt><dd><p>physical position in Mbp of SNP on chromosomes</p>
</dd>
<dt><code>Chr</code></dt><dd><p>chromosome of SNP</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="geneticPosition_+3A_exclude">exclude</code></td>
<td>
<p>optional vector (LEN &lt; p) of SNP IDs to be excluded (e.g.,
candidates of misplaced SNPs; default NULL)</p>
</td></tr>
<tr><td><code id="geneticPosition_+3A_threshold">threshold</code></td>
<td>
<p>optional value; recombination rates &lt;= threshold are
considered for smoothing approach assuming theta ~ Morgan (default 0.05)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Smoothing of recombination rates (theta) &lt;= 0.05 via quadratic
optimization provides an approximation of genetic distances (in Morgan)
between SNPs. The cumulative sum * 100 yields the genetic positions in cM.
</p>
<p>The minimization problem <code>(theta - D d)^2</code> is solved s.t. d &gt; 0 where
d is the vector of genetic distances between adjacent markers but theta is
not restricted to adjacent markers. The incidence matrix D contains 1's for
those intervals contributing to the total distance relevant for each theta.
</p>
<p>Estimates of theta = 1e-6 are neglected as these values coincide with start
values and indicate that (because of a very flat likelihood surface) no
meaningful estimate of recombination rate has been obtained.
</p>


<h3>Value</h3>

<p>list (LEN 2)
</p>

<dl>
<dt>gen.cM</dt><dd><p>vector (LEN p) of genetic positions of SNPs (in cM)</p>
</dd>
<dt>gen.Mb</dt><dd><p>vector (LEN p) of physical positions of SNPs (in Mbp)</p>
</dd>
</dl>



<h3>References</h3>

<p>Qanbari, S. &amp; Wittenburg, D. (2020) Male recombination map of the autosomal
genome in German Holstein. Genetics Selection Evolution 52:73.
doi: <a href="https://doi.org/10.1186/s12711-020-00593-z">10.1186/s12711-020-00593-z</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ### test data
  data(targetregion)
  ### make list for paternal half-sib families
  hap &lt;- makehaplist(daughterSire, hapSire)
  ### parameter estimates on a chromosome
  res &lt;- hsrecombi(hap, genotype.chr)
  ### post-processing to achieve final and valid set of estimates
  final &lt;- editraw(res, map.chr)
  ### approximation of genetic positions
  pos &lt;- geneticPosition(final, map.chr)
</code></pre>

<hr>
<h2 id='genotype.chr'>targetregion: progeny genotypes</h2><span id='topic+genotype.chr'></span>

<h3>Description</h3>

<p>matrix of progeny genotypes in target region on chromosome BTA1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotype.chr
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 265 rows and 200 columns.
</p>

<hr>
<h2 id='haldane'>Haldane's genetic map function</h2><span id='topic+haldane'></span>

<h3>Description</h3>

<p>Calculation of genetic distances from recombination rates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haldane(x, inverse = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haldane_+3A_x">x</code></td>
<td>
<p>vector of recombination rates</p>
</td></tr>
<tr><td><code id="haldane_+3A_inverse">inverse</code></td>
<td>
<p>logical, if FALSE recombination rate is mapped to Morgan unit,
if TRUE Morgan unit is mapped to recombination rate (default is FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of genetic positions in Morgan units
</p>


<h3>References</h3>

<p>Haldane JBS (1919) The combination of linkage values, and the
calculation of distances between the loci of linked factors. J Genet 8:
299-309.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  haldane(seq(0, 0.5, 0.01))
</code></pre>

<hr>
<h2 id='hapSire'>targetregion: sire haplotypes</h2><span id='topic+hapSire'></span>

<h3>Description</h3>

<p>matrix of sire haplotypes in target region on chromosome BTA1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hapSire
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 10 rows and 201 columns.
</p>

<hr>
<h2 id='hsrecombi'>Estimation of recombination rate and maternal LD</h2><span id='topic+hsrecombi'></span>

<h3>Description</h3>

<p>Wrapper function for estimating recombination rate and maternal
linkage disequilibrium between intra-chromosomal SNP pairs by calling EM
algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hsrecombi(hap, genotype.chr, exclude = NULL, only.adj = FALSE, prec = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hsrecombi_+3A_hap">hap</code></td>
<td>
<p>list (LEN 2) of lists
</p>

<dl>
<dt>famID</dt><dd><p>list (LEN number of sires) of vectors (LEN n.progeny) of
progeny indices relating to lines in genotype matrix</p>
</dd>
<dt>sireHap</dt><dd><p>list (LEN number of sires) of matrices (DIM 2 x p)
of sire haplotypes (0, 1) on investigated chromosome</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="hsrecombi_+3A_genotype.chr">genotype.chr</code></td>
<td>
<p>matrix (DIM n x p) of all progeny genotypes (0, 1, 2) on
a chromosome with p SNPs; 9 indicates missing genotype</p>
</td></tr>
<tr><td><code id="hsrecombi_+3A_exclude">exclude</code></td>
<td>
<p>vector (LEN &lt; p) of SNP IDs (for filtering column names of
<code>genotype.chr)</code> to be excluded from analysis (default NULL)</p>
</td></tr>
<tr><td><code id="hsrecombi_+3A_only.adj">only.adj</code></td>
<td>
<p>logical; if <code>TRUE</code>, recombination rate is calculated
only between neighbouring markers</p>
</td></tr>
<tr><td><code id="hsrecombi_+3A_prec">prec</code></td>
<td>
<p>scalar; precision of estimation</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Paternal recombination rate and maternal linkage disequilibrium (LD)
are estimated for pairs of biallelic markers (such as single nucleotide
polymorphisms; SNPs) from progeny genotypes and sire haplotypes. At least
one sire has to be double heterozygous at the investigated pairs of SNPs.
All progeny are merged in two genomic families: (1) coupling phase family
if sires are double heterozygous 0-0/1-1 and (2) repulsion phase family if
sires are double heterozygous 0-1/1-0.
So far it is recommended processing the chromosomes separately.
If maternal half-sib families are used, the roles of sire/dam are swapped.
Multiple families can be considered.
</p>


<h3>Value</h3>

<p>list (LEN p - 1) of data.frames; for each SNP, parameters are
estimated with all following SNPs; two solutions (prefix sln1 and sln2) are
obtained for two runs of the EM algorithm
</p>

<dl>
<dt><code>SNP1</code></dt><dd><p>ID of 1. SNP</p>
</dd>
<dt><code>SNP2</code></dt><dd><p>ID of 2. SNP</p>
</dd>
<dt><code>D</code></dt><dd><p>maternal LD</p>
</dd>
<dt><code>fAA</code></dt><dd><p>frequency of maternal haplotype 1-1</p>
</dd>
<dt><code>fAB</code></dt><dd><p>frequency of maternal haplotype 1-0</p>
</dd>
<dt><code>fBA</code></dt><dd><p>frequency of maternal haplotype 0-1</p>
</dd>
<dt><code>fBB</code></dt><dd><p>frequency of maternal haplotype 0-0</p>
</dd>
<dt><code>p1</code></dt><dd><p>Maternal allele frequency (allele 1) at <code>SNP1</code></p>
</dd>
<dt><code>p2</code></dt><dd><p>Maternal allele frequency (allele 1) at <code>SNP2</code></p>
</dd>
<dt><code>nfam1</code></dt><dd><p>size of genomic family 1</p>
</dd>
<dt><code>nfam2</code></dt><dd><p>size of genomic family 2</p>
</dd>
<dt><code>error</code></dt><dd><p>0 if computations were without error; 1 if EM algorithm
did not converge</p>
</dd>
<dt><code>iteration</code></dt><dd><p>number of EM iterations</p>
</dd>
<dt><code>theta</code></dt><dd><p>paternal recombination rate</p>
</dd>
<dt><code>r2</code></dt><dd><p><code class="reqn">r^2</code> of maternal LD</p>
</dd>
<dt><code>logL</code></dt><dd><p>value of log likelihood function</p>
</dd>
<dt><code>unimodal</code></dt><dd><p>1 if likelihood is unimodal; 0 if likelihood is
bimodal</p>
</dd>
<dt><code>critical</code></dt><dd><p>0 if parameter estimates are unique; 1 if parameter
estimates at both solutions are valid, then decision process follows in
post-processing function &quot;editraw&quot;</p>
</dd>
</dl>

<p>Afterwards, solutions are compared and processed with function
<code>editraw</code>, yielding the final estimates for each valid pair of SNPs.
</p>


<h3>References</h3>

<p>Hampel, A., Teuscher, F., Gomez-Raya, L., Doschoris, M. &amp; Wittenburg, D.
(2018) Estimation of recombination rate and maternal linkage
disequilibrium in half-sibs. Frontiers in Genetics 9:186.
doi: <a href="https://doi.org/10.3389/fgene.2018.00186">10.3389/fgene.2018.00186</a>
</p>
<p>Gomez-Raya, L. (2012) Maximum likelihood estimation of linkage
disequilibrium in half-sib families. Genetics 191:195-213.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ### test data
  data(targetregion)
  ### make list for paternal half-sib families
  hap &lt;- makehaplist(daughterSire, hapSire)
  ### parameter estimates on a chromosome
  res &lt;- hsrecombi(hap, genotype.chr)
  ### post-processing to achieve final and valid set of estimates
  final &lt;- editraw(res, map.chr)
</code></pre>

<hr>
<h2 id='karlin'>Liberman and Karlin's genetic map function</h2><span id='topic+karlin'></span>

<h3>Description</h3>

<p>Calculation of genetic distances from recombination rates given
a parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>karlin(N, x, inverse = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="karlin_+3A_n">N</code></td>
<td>
<p>parameter (positive integer) required by the binomial model to
assess the count (of crossover) distribution; <code>N = 1</code> corresponds to
Morgan's map function</p>
</td></tr>
<tr><td><code id="karlin_+3A_x">x</code></td>
<td>
<p>vector of recombination rates</p>
</td></tr>
<tr><td><code id="karlin_+3A_inverse">inverse</code></td>
<td>
<p>logical, if FALSE recombination rate is mapped to Morgan unit,
if TRUE Morgan unit is mapped to recombination rate (default is FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of genetic positions in Morgan units
</p>


<h3>References</h3>

<p>Liberman, U. &amp; Karlin, S. (1984) Theoretical models of genetic
map functions. Theor Popul Biol 25:331-346.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  karlin(2, seq(0, 0.5, 0.01))
</code></pre>

<hr>
<h2 id='kosambi'>Kosambi's genetic map function</h2><span id='topic+kosambi'></span>

<h3>Description</h3>

<p>Calculation of genetic distances from recombination rates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kosambi(x, inverse = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kosambi_+3A_x">x</code></td>
<td>
<p>vector of recombination rates</p>
</td></tr>
<tr><td><code id="kosambi_+3A_inverse">inverse</code></td>
<td>
<p>logical, if FALSE recombination rate is mapped to Morgan unit,
if TRUE Morgan unit is mapped to recombination rate (default is FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of genetic positions in Morgan units
</p>


<h3>References</h3>

<p>Kosambi D.D. (1944) The estimation of map distance from
recombination values. Ann. Eugen. 12: 172-175.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  kosambi(seq(0, 0.5, 0.01))
</code></pre>

<hr>
<h2 id='LDHScpp'>Expectation Maximisation (EM) algorithm</h2><span id='topic+LDHScpp'></span>

<h3>Description</h3>

<p>Expectation Maximisation (EM) algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDHScpp(XGF1, XGF2, fAA, fAB, fBA, theta, display, threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDHScpp_+3A_xgf1">XGF1</code></td>
<td>
<p>integer matrix of progeny genotypes in genomic family 1</p>
</td></tr>
<tr><td><code id="LDHScpp_+3A_xgf2">XGF2</code></td>
<td>
<p>integer matrix of progeny genotypes in genomic family 2</p>
</td></tr>
<tr><td><code id="LDHScpp_+3A_faa">fAA</code></td>
<td>
<p>frequency of maternal haplotype 1-1</p>
</td></tr>
<tr><td><code id="LDHScpp_+3A_fab">fAB</code></td>
<td>
<p>frequency of maternal haplotype 1-0</p>
</td></tr>
<tr><td><code id="LDHScpp_+3A_fba">fBA</code></td>
<td>
<p>frequency of maternal haplotype 0-1</p>
</td></tr>
<tr><td><code id="LDHScpp_+3A_theta">theta</code></td>
<td>
<p>paternal recombination rate</p>
</td></tr>
<tr><td><code id="LDHScpp_+3A_display">display</code></td>
<td>
<p>logical for displaying additional information</p>
</td></tr>
<tr><td><code id="LDHScpp_+3A_threshold">threshold</code></td>
<td>
<p>convergence criterion</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of parameter estimates
</p>

<dl>
<dt><code>D</code></dt><dd><p>maternal LD</p>
</dd>
<dt><code>fAA</code></dt><dd><p>frequency of maternal haplotype 1-1</p>
</dd>
<dt><code>fAB</code></dt><dd><p>frequency of maternal haplotype 1-0</p>
</dd>
<dt><code>fBA</code></dt><dd><p>frequency of maternal haplotype 0-1</p>
</dd>
<dt><code>fBB</code></dt><dd><p>frequency of maternal haplotype 0-0</p>
</dd>
<dt><code>p1</code></dt><dd><p>Maternal allele frequency (allele 1) at 1. SNP</p>
</dd>
<dt><code>p2</code></dt><dd><p>Maternal allele frequency (allele 1) at 2. SNP</p>
</dd>
<dt><code>nfam1</code></dt><dd><p>size of genomic family 1</p>
</dd>
<dt><code>nfam2</code></dt><dd><p>size of genomic family 2</p>
</dd>
<dt><code>error</code></dt><dd><p>0 if computations were without error; 1 if EM algorithm
did not converge</p>
</dd>
<dt><code>iteration</code></dt><dd><p>number of EM iterations</p>
</dd>
<dt><code>theta</code></dt><dd><p>paternal recombination rate</p>
</dd>
<dt><code>r2</code></dt><dd><p><code class="reqn">r^2</code> of maternal LD</p>
</dd>
<dt><code>logL</code></dt><dd><p>value of log likelihood function</p>
</dd>
</dl>


<hr>
<h2 id='loglikfun'>Calculate log-likelihood function</h2><span id='topic+loglikfun'></span>

<h3>Description</h3>

<p>Calculate log-likelihood function
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="loglikfun_+3A_counts">counts</code></td>
<td>
<p>integer vector of observed 2-locus genotype</p>
</td></tr>
<tr><td><code id="loglikfun_+3A_faa">fAA</code></td>
<td>
<p>frequency of maternal haplotype 1-1</p>
</td></tr>
<tr><td><code id="loglikfun_+3A_fab">fAB</code></td>
<td>
<p>frequency of maternal haplotype 1-0</p>
</td></tr>
<tr><td><code id="loglikfun_+3A_fba">fBA</code></td>
<td>
<p>frequency of maternal haplotype 0-1</p>
</td></tr>
<tr><td><code id="loglikfun_+3A_fbb">fBB</code></td>
<td>
<p>frequency of maternal haplotype 0-0</p>
</td></tr>
<tr><td><code id="loglikfun_+3A_theta">theta</code></td>
<td>
<p>paternal recombination rate</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>lik</code> value of log likelihood at parameter estimates
</p>

<hr>
<h2 id='makehap'>Make list of imputed sire haplotypes</h2><span id='topic+makehap'></span>

<h3>Description</h3>

<p>List of sire haplotypes is set up in the format required for
hsrecombi. Sire haplotypes are imputed from progeny genotypes using R
package <code>hsphase</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makehap(sireID, daughterSire, genotype.chr, nmin = 30, exclude = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makehap_+3A_sireid">sireID</code></td>
<td>
<p>vector (LEN N) of IDs of all sires</p>
</td></tr>
<tr><td><code id="makehap_+3A_daughtersire">daughterSire</code></td>
<td>
<p>vector (LEN n) of sire ID for each progeny</p>
</td></tr>
<tr><td><code id="makehap_+3A_genotype.chr">genotype.chr</code></td>
<td>
<p>matrix (DIM n x p) of progeny genotypes (0, 1, 2) on a
single chromosome with p SNPs; 9 indicates missing genotype</p>
</td></tr>
<tr><td><code id="makehap_+3A_nmin">nmin</code></td>
<td>
<p>scalar, minimum required number of progeny for proper imputation,
default 30</p>
</td></tr>
<tr><td><code id="makehap_+3A_exclude">exclude</code></td>
<td>
<p>vector (LEN &lt; p) of SNP indices to be excluded from analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list (LEN 2) of lists. For each sire:
</p>

<dl>
<dt><code>famID</code></dt><dd><p>list (LEN N) of vectors (LEN n.progeny) of progeny
indices relating to lines in genotype matrix</p>
</dd>
<dt><code>sireHap</code></dt><dd><p>list (LEN N) of matrices (DIM 2 x p) of sire
haplotypes (0, 1) on investigated chromosome</p>
</dd>
</dl>



<h3>References</h3>

<p>Ferdosi, M., Kinghorn, B., van der Werf, J., Lee, S. &amp; Gondro, C. (2014)
hsphase: an R package for pedigree reconstruction, detection of
recombination events, phasing and imputation of half-sib family groups
BMC Bioinformatics 15:172.
<a href="https://CRAN.R-project.org/package=hsphase">https://CRAN.R-project.org/package=hsphase</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> data(targetregion)
 hap &lt;- makehap(unique(daughterSire), daughterSire, genotype.chr)
</code></pre>

<hr>
<h2 id='makehaplist'>Make list of sire haplotypes</h2><span id='topic+makehaplist'></span>

<h3>Description</h3>

<p>List of sire haplotypes is set up in the format required for
hsrecombi. Haplotypes (obtained by external software) are provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makehaplist(daughterSire, hapSire, nmin = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makehaplist_+3A_daughtersire">daughterSire</code></td>
<td>
<p>vector (LEN n) of sire ID for each progeny</p>
</td></tr>
<tr><td><code id="makehaplist_+3A_hapsire">hapSire</code></td>
<td>
<p>matrix (DIM 2N x p + 1) of sire haplotype at p SNPs; 2 lines
per sire, 1. columns contains sire ID</p>
</td></tr>
<tr><td><code id="makehaplist_+3A_nmin">nmin</code></td>
<td>
<p>scalar, minimum number of progeny required, default 1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list (LEN 2) of lists. For each sire:
</p>

<dl>
<dt><code>famID</code></dt><dd><p>list (LEN N) of vectors (LEN n.progeny) of progeny
indices relating to lines in genotype matrix</p>
</dd>
<dt><code>sireHap</code></dt><dd><p>list (LEN N) of matrices (DIM 2 x p) of sire
haplotypes (0, 1) on investigated chromosome</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>  data(targetregion)
  hap &lt;- makehaplist(daughterSire, hapSire)
</code></pre>

<hr>
<h2 id='makehappm'>Make list of imputed haplotypes and estimate recombination rate</h2><span id='topic+makehappm'></span>

<h3>Description</h3>

<p>List of sire haplotypes is set up in the format required for
hsrecombi. Sire haplotypes are imputed from progeny genotypes using R
package <code>hsphase</code>. Furthermore, recombination rate estimates between
adjacent SNPs from hsphase are reported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makehappm(sireID, daughterSire, genotype.chr, nmin = 30, exclude = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makehappm_+3A_sireid">sireID</code></td>
<td>
<p>vector (LEN N) of IDs of all sires</p>
</td></tr>
<tr><td><code id="makehappm_+3A_daughtersire">daughterSire</code></td>
<td>
<p>vector (LEN n) of sire ID for each progeny</p>
</td></tr>
<tr><td><code id="makehappm_+3A_genotype.chr">genotype.chr</code></td>
<td>
<p>matrix (DIM n x p) of progeny genotypes (0, 1, 2) on a
single chromosome with p SNPs; 9 indicates missing genotype</p>
</td></tr>
<tr><td><code id="makehappm_+3A_nmin">nmin</code></td>
<td>
<p>scalar, minimum required number of progeny for proper imputation,
default 30</p>
</td></tr>
<tr><td><code id="makehappm_+3A_exclude">exclude</code></td>
<td>
<p>vector (LEN &lt; p) of SNP IDs (for filtering column names of
<code>genotype.chr)</code> to be excluded from analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list (LEN 2) of lists. For each sire:
</p>

<dl>
<dt><code>famID</code></dt><dd><p>list (LEN N) of vectors (LEN n.progeny) of progeny
indices relating to lines in genotype matrix</p>
</dd>
<dt><code>sireHap</code></dt><dd><p>list (LEN N) of matrices (DIM 2 x p) of sire
haplotypes (0, 1) on investigated chromosome</p>
</dd>
<dt>probRec</dt><dd><p>vector (LEN p - 1) of proportion of recombinant progeny over
all families between adjacent SNPs</p>
</dd>
<dt>numberRec</dt><dd><p>list (LEN N) of vectors (LEN n.progeny) of number of
recombination events per animal</p>
</dd>
<dt>gen</dt><dd><p>vector (LEN p) of genetic positions of SNPs (in cM)</p>
</dd>
</dl>



<h3>References</h3>

<p>Ferdosi, M., Kinghorn, B., van der Werf, J., Lee, S. &amp; Gondro, C. (2014)
hsphase: an R package for pedigree reconstruction, detection of
recombination events, phasing and imputation of half-sib family groups
BMC Bioinformatics 15:172.
<a href="https://CRAN.R-project.org/package=hsphase">https://CRAN.R-project.org/package=hsphase</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(targetregion)
  hap &lt;- makehappm(unique(daughterSire), daughterSire, genotype.chr, exclude = paste0('V', 301:310))
</code></pre>

<hr>
<h2 id='map.chr'>targetregion: physical map</h2><span id='topic+map.chr'></span>

<h3>Description</h3>

<p>SNP marker map in target region on chromosome BTA1 according to
ARS-UCD1.2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map.chr
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="map.chr_+3A_map.chr">map.chr</code></td>
<td>
<p>data frame
</p>

<dl>
<dt>SNP</dt><dd><p>SNP index</p>
</dd>
<dt>Chr</dt><dd><p>chromosome of SNP</p>
</dd>
<dt>locus_bp</dt><dd><p>physical position of SNP in bp</p>
</dd>
<dt>locus_Mb</dt><dd><p>physical position of SNP in Mbp</p>
</dd>
<dt>markername</dt><dd><p>official SNP name</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 200 rows and 6 columns.
</p>

<hr>
<h2 id='rao'>System of genetic-map functions</h2><span id='topic+rao'></span>

<h3>Description</h3>

<p>Calculation of genetic distances from recombination rates given
a mixing parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rao(p, x, inverse = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rao_+3A_p">p</code></td>
<td>
<p>mixing parameter (see details); <code>0 &lt;= p &lt;= 1</code></p>
</td></tr>
<tr><td><code id="rao_+3A_x">x</code></td>
<td>
<p>vector of recombination rates</p>
</td></tr>
<tr><td><code id="rao_+3A_inverse">inverse</code></td>
<td>
<p>logical, if FALSE recombination rate is mapped to Morgan unit,
if TRUE Morgan unit is mapped to recombination rate (default is FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Mixing parameter <code>p=0</code> would match to Morgan, <code>p=0.25</code> to
Carter, <code>p=0.5</code> to Kosambi and <code>p=1</code> to Haldane map function.
As an inverse of Rao's system of functions does not exist, NA will be
produced if <code>inverse = T</code>. To approximate the inverse call function
<code>rao.inv(p, x)</code>.
</p>


<h3>Value</h3>

<p>vector of genetic positions in Morgan units
</p>


<h3>References</h3>

<p>Rao, D.C., Morton, N.E., Lindsten, J., Hulten, M. &amp; Yee, S (1977)
A mapping function for man. Human Heredity 27: 99-104.
doi: <a href="https://doi.org/10.1159/000152856">10.1159/000152856</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  rao(0.25, seq(0, 0.5, 0.01))
</code></pre>

<hr>
<h2 id='rao+20inverse'>Approximation to inverse of Rao's system of map functions</h2><span id='topic+rao+20inverse'></span><span id='topic+rao.inv'></span>

<h3>Description</h3>

<p>Calculation of recombination rates from genetic distances given
a mixing parameter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rao.inv(p, x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rao+2B20inverse_+3A_p">p</code></td>
<td>
<p>mixing parameter (see details); <code>0 &lt;= p &lt;= 1</code></p>
</td></tr>
<tr><td><code id="rao+2B20inverse_+3A_x">x</code></td>
<td>
<p>vector in Morgan units</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Mixing parameter <code>p=0</code> would match to Morgan, <code>p=0.25</code> to
Carter, <code>p=0.5</code> to Kosambi and <code>p=1</code> to Haldane map function.
</p>


<h3>Value</h3>

<p>vector of recombination rates
</p>


<h3>References</h3>

<p>Rao, D.C., Morton, N.E., Lindsten, J., Hulten, M. &amp; Yee, S (1977)
A mapping function for man. Human Heredity 27: 99-104.
doi: <a href="https://doi.org/10.1159/000152856">10.1159/000152856</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  rao.inv(0.25, seq(0, 01, 0.1))
</code></pre>

<hr>
<h2 id='startvalue'>Start value for maternal allele and haplotype frequencies</h2><span id='topic+startvalue'></span>

<h3>Description</h3>

<p>Determine default start values for Expectation Maximisation (EM)
algorithm that is used to estimate paternal recombination rate and maternal
haplotype frequencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>startvalue(Fam1, Fam2, Dd = 0, prec = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="startvalue_+3A_fam1">Fam1</code></td>
<td>
<p>matrix (DIM n.progeny x 2) of progeny genotypes (0, 1, 2) of
genomic family with coupling phase sires (1) at SNP pair</p>
</td></tr>
<tr><td><code id="startvalue_+3A_fam2">Fam2</code></td>
<td>
<p>matrix (DIM n.progeny x 2) of progeny genotypes (0, 1, 2) of
genomic family with repulsion phase sires (2) at SNP pair</p>
</td></tr>
<tr><td><code id="startvalue_+3A_dd">Dd</code></td>
<td>
<p>maternal LD, default 0</p>
</td></tr>
<tr><td><code id="startvalue_+3A_prec">prec</code></td>
<td>
<p>minimum accepted start value for fAA, fAB, fBA; default
<code>1e-6</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>list (LEN 8)
</p>

<dl>
<dt><code>fAA.start</code></dt><dd><p>frequency of maternal haplotype 1-1</p>
</dd>
<dt><code>fAB.start</code></dt><dd><p>frequency of maternal haplotype 1-0</p>
</dd>
<dt><code>fBA.start</code></dt><dd><p>frequency of maternal haplotype 0-1</p>
</dd>
<dt><code>p1</code></dt><dd><p>estimate of maternal allele frequency (allele 1) when sire
is heterozygous at <code>SNP1</code></p>
</dd>
<dt><code>p2</code></dt><dd><p>estimate of maternal allele frequency (allele 1) when sire
is heterozygous at <code>SNP2</code></p>
</dd>
<dt><code>L1</code></dt><dd><p>lower bound of maternal LD</p>
</dd>
<dt><code>L2</code></dt><dd><p>upper bound for maternal LD</p>
</dd>
<dt><code>critical</code></dt><dd><p>0 if parameter estimates are unique; 1 if parameter
estimates at both solutions are valid</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'> n1 &lt;- 100
 n2 &lt;- 20
 G1 &lt;- matrix(ncol = 2, nrow = n1, sample(c(0:2), replace = TRUE,
  size = 2 * n1))
 G2 &lt;- matrix(ncol = 2, nrow = n2, sample(c(0:2), replace = TRUE,
  size = 2 * n2))
 startvalue(G1, G2)
</code></pre>

<hr>
<h2 id='targetregion'>Description of the targetregion data set</h2><span id='topic+targetregion'></span>

<h3>Description</h3>

<p>The data set contains sire haplotypes, assignment of progeny to
sire, progeny genotypes and physical map information in a target region
</p>

<p>The raw data can be downloaded at the source given below. Then,
executing the following R code leads to the data provided in
<code>targetregion.RData</code>.
</p>
<dl>
<dt><code>hapSire</code></dt><dd><p>matrix of sire haplotypes of each sire; 2 lines per
sire; 1. column contains sireID</p>
</dd>
<dt><code>daughterSire</code></dt><dd><p>vector of sire ID for each progeny</p>
</dd>
<dt><code>genotype.chr</code></dt><dd><p>matrix of progeny genotypes</p>
</dd>
<dt><code>map.chr</code></dt><dd><p>SNP marker map in target region</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are available at RADAR
doi: <a href="https://doi.org/10.22000/280">10.22000/280</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download data from RADAR (requires about 1.4 GB)
url &lt;- "https://www.radar-service.eu/radar-backend/archives/fqSPQoIvjtOGJlav/versions/1/content"
curl_download(url = url, 'tmp.tar')
untar('tmp.tar')
file.remove('tmp.tar')
path &lt;- '10.22000-280/data/dataset'
## list of haplotypes of sires for each chromosome
load(file.path(path, 'sire_haplotypes.RData'))
## assign progeny to sire
daughterSire &lt;- read.table(file.path(path, 'assign_to_family.txt'))[, 1]
## progeny genotypes
X &lt;- as.matrix(read.table(file.path(path, 'XFam-ARS.txt')))
## physical and approximated genetic map
map &lt;- read.table(file.path(path, 'map50K_ARS_reordered.txt'), header = T)
## select target region
chr &lt;- 1
window &lt;- 301:500
## map information of target region
map.chr &lt;- map[map$Chr == chr, ][window, ]
## matrix of sire haplotypes in target region
hapSire &lt;- rlist::list.rbind(haps[[chr]])
sireID &lt;- 1:length(unique(daughterSire))
hapSire &lt;- cbind(rep(sireID, each = 2), hapSire[, window])
## matrix of progeny genotypes
genotype.chr &lt;- X[, map.chr$SNP]
colnames(genotype.chr) &lt;- map.chr$SNP
save(list = c('genotype.chr', 'hapSire', 'map.chr', 'daughterSire'),
     file = 'targetregion.RData', compress = 'xz')

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
