<!DOCTYPE html><html><head><title>Help for package httk</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {httk}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_chemtable'><p>Add a table of chemical information for use in making httk predictions.</p></a></li>
<li><a href='#age_draw_smooth'><p>Draws ages from a smoothed distribution for a given gender/race combination</p></a></li>
<li><a href='#apply_clint_adjustment'><p>Correct the measured intrinsive hepatic clearance for fraction free</p></a></li>
<li><a href='#apply_fup_adjustment'><p>Correct the measured fraction unbound in plasma for lipid binding</p></a></li>
<li><a href='#armitage_estimate_sarea'><p>Estimate well surface area</p></a></li>
<li><a href='#armitage_eval'><p>Evaluate the updated Armitage model</p></a></li>
<li><a href='#armitage_input'><p>Armitage et al. (2014) Model Inputs from Honda et al. (2019)</p></a></li>
<li><a href='#augment.table'><p>Add a parameter value to the chem.physical_and_invitro.data table</p></a></li>
<li><a href='#available_rblood2plasma'><p>Find the best available ratio of the blood to plasma concentration constant.</p></a></li>
<li><a href='#aylward2014'><p>Aylward et al. 2014</p></a></li>
<li><a href='#benchmark_httk'><p>Assess the current performance of httk relative to historical benchmarks</p></a></li>
<li><a href='#blood_mass_correct'><p>Find average blood masses by age.</p></a></li>
<li><a href='#blood_weight'><p>Predict blood mass.</p></a></li>
<li><a href='#bmiage'><p>CDC BMI-for-age charts</p></a></li>
<li><a href='#body_surface_area'><p>Predict body surface area.</p></a></li>
<li><a href='#bone_mass_age'><p>Predict bone mass</p></a></li>
<li><a href='#brain_mass'><p>Predict brain mass.</p></a></li>
<li><a href='#calc_analytic_css'><p>Calculate the analytic steady state plasma concentration.</p></a></li>
<li><a href='#calc_analytic_css_1comp'><p>Calculate the analytic steady state concentration for the one compartment model.</p></a></li>
<li><a href='#calc_analytic_css_3comp'><p>Calculate the analytic steady state concentration for model 3comp</p></a></li>
<li><a href='#calc_analytic_css_3compss'><p>Calculate the analytic steady state concentration for the three compartment</p>
steady-state model</a></li>
<li><a href='#calc_analytic_css_pbtk'><p>Calculate the analytic steady state plasma concentration for model pbtk.</p></a></li>
<li><a href='#calc_css'><p>Find the steady state concentration and the day it is reached.</p></a></li>
<li><a href='#calc_dow'><p>Calculate the distribution coefficient</p></a></li>
<li><a href='#calc_elimination_rate'><p>Calculate the elimination rate for a one compartment model</p></a></li>
<li><a href='#calc_fbio.oral'><p>Functions for calculating the bioavaialble fractions from oral doses</p></a></li>
<li><a href='#calc_fetal_phys'><p>Calculate maternal-fetal physiological parameters</p></a></li>
<li><a href='#calc_fup_correction'><p>Calculate the correction for lipid binding in plasma binding assay</p></a></li>
<li><a href='#calc_half_life'><p>Calculates the half-life for a one compartment model.</p></a></li>
<li><a href='#calc_hep_bioavailability'><p>Calculate first pass heaptic metabolism</p></a></li>
<li><a href='#calc_hep_clearance'><p>Calculate the hepatic clearance.</p></a></li>
<li><a href='#calc_hep_fu'><p>Calculate the free chemical in the hepaitic clearance assay</p></a></li>
<li><a href='#calc_hepatic_clearance'><p>Calculate the hepatic clearance (deprecated).</p></a></li>
<li><a href='#calc_ionization'><p>Calculate the ionization.</p></a></li>
<li><a href='#calc_kair'><p>Calculate air:matrix partition coefficients</p></a></li>
<li><a href='#calc_krbc2pu'><p>Back-calculates the Red Blood Cell to Unbound Plasma Partition Coefficient</p></a></li>
<li><a href='#calc_ma'><p>Calculate the membrane affinity</p></a></li>
<li><a href='#calc_maternal_bw'><p>Calculate maternal body weight</p></a></li>
<li><a href='#calc_mc_css'><p>Distribution of chemical steady state concentration with uncertainty and variability</p></a></li>
<li><a href='#calc_mc_oral_equiv'><p>Calculate Monte Carlo Oral Equivalent Dose</p></a></li>
<li><a href='#calc_mc_tk'><p>Conduct multiple TK simulations using Monte Carlo</p></a></li>
<li><a href='#calc_rblood2plasma'><p>Calculate the constant ratio of the blood concentration to the plasma</p>
concentration.</a></li>
<li><a href='#calc_stats'><p>Calculate toxicokinetic summary statistics (deprecated).</p></a></li>
<li><a href='#calc_tkstats'><p>Calculate toxicokinetic summary statistics.</p></a></li>
<li><a href='#calc_total_clearance'><p>Calculate the total plasma clearance.</p></a></li>
<li><a href='#calc_vdist'><p>Calculate the volume of distribution for a one compartment model.</p></a></li>
<li><a href='#CAS.checksum'><p>Test the check digit of a CAS number to confirm validity</p></a></li>
<li><a href='#chem.invivo.PK.aggregate.data'><p>Parameter Estimates from Wambaugh et al. (2018)</p></a></li>
<li><a href='#chem.invivo.PK.data'><p>Published toxicokinetic time course measurements</p></a></li>
<li><a href='#chem.invivo.PK.summary.data'><p>Summary of published toxicokinetic time course experiments</p></a></li>
<li><a href='#chem.physical_and_invitro.data'><p>Physico-chemical properties and in vitro measurements for toxicokinetics</p></a></li>
<li><a href='#ckd_epi_eq'><p>CKD-EPI equation for GFR.</p></a></li>
<li><a href='#concentration_data_Linakis2020'><p>Concentration data involved in Linakis 2020 vignette analysis.</p></a></li>
<li><a href='#convert_httkpop_1comp'><p>Converts HTTK-Pop physiology into parameters relevant to the one</p>
compartment model</a></li>
<li><a href='#convert_solve_x'><p>convert_solve_x</p></a></li>
<li><a href='#convert_units'><p>convert_units</p></a></li>
<li><a href='#create_mc_samples'><p>Create a table of parameter values for Monte Carlo</p></a></li>
<li><a href='#dawson2021'><p>Dawson et al. 2021 data</p></a></li>
<li><a href='#EPA.ref'><p>Reference for EPA Physico-Chemical Data</p></a></li>
<li><a href='#estimate_gfr'><p>Predict GFR.</p></a></li>
<li><a href='#estimate_gfr_ped'><p>Predict GFR in children.</p></a></li>
<li><a href='#estimate_hematocrit'><p>Generate hematocrit values for a virtual population</p></a></li>
<li><a href='#example.seem'><p>SEEM Example Data</p>
We can grab SEEM daily intake rate predictions already in RData format from
https://github.com/HumanExposure/SEEM3RPackage/tree/main/SEEM3/data
Download the file Ring2018Preds.RData</a></li>
<li><a href='#example.toxcast'><p>ToxCast Example Data</p>
The main page for the ToxCast data is here:
https://www.epa.gov/comptox-tools/exploring-toxcast-data
Most useful to us is a single file containing all the hits across all chemcials
and assays:
https://clowder.edap-cluster.com/datasets/6364026ee4b04f6bb1409eda?space=62bb560ee4b07abf29f88fef</a></li>
<li><a href='#export_pbtk_jarnac'><p>Export model to jarnac.</p></a></li>
<li><a href='#export_pbtk_sbml'><p>Export model to sbml.</p></a></li>
<li><a href='#fetalpcs'><p>Fetal Partition Coefficients</p></a></li>
<li><a href='#Frank2018invivo'><p>Literature In Vivo Data on Doses Causing Neurological Effects</p></a></li>
<li><a href='#gen_age_height_weight'><p>Generate demographic parameters for a virtual population</p></a></li>
<li><a href='#gen_height_weight'><p>Generate heights and weights for a virtual population.</p></a></li>
<li><a href='#gen_serum_creatinine'><p>Generate serum creatinine values for a virtual population.</p></a></li>
<li><a href='#get_caco2'><p>Retrieve in vitro measured Caco-2 membrane permeabilit</p></a></li>
<li><a href='#get_chem_id'><p>Retrieve chemical identity from HTTK package</p></a></li>
<li><a href='#get_cheminfo'><p>Retrieve chemical information available from HTTK package</p></a></li>
<li><a href='#get_clint'><p>Retrieve and parse intrinsic hepatic clearance</p></a></li>
<li><a href='#get_fabsgut'><p>Retrieve or calculate fraction of chemical absorbed from the gut</p></a></li>
<li><a href='#get_fup'><p>Retrieve and parse fraction unbound in plasma</p></a></li>
<li><a href='#get_gfr_category'><p>Categorize kidney function by GFR.</p></a></li>
<li><a href='#get_invitroPK_param'><p>Retrieve species-specific in vitro data from chem.physical_and_invitro.data table</p></a></li>
<li><a href='#get_lit_cheminfo'><p>Get literature Chemical Information.</p></a></li>
<li><a href='#get_lit_css'><p>Get literature Css</p></a></li>
<li><a href='#get_lit_oral_equiv'><p>Get Literature Oral Equivalent Dose</p></a></li>
<li><a href='#get_physchem_param'><p>Get physico-chemical parameters from chem.physical_and_invitro.data table</p></a></li>
<li><a href='#get_rblood2plasma'><p>Get ratio of the blood concentration to the plasma concentration.</p></a></li>
<li><a href='#get_weight_class'><p>Assign weight class (underweight, normal, overweight, obese)</p></a></li>
<li><a href='#get_wetmore_cheminfo'><p>Get literature Chemical Information. (deprecated).</p></a></li>
<li><a href='#get_wetmore_css'><p>Get literature Css (deprecated).</p></a></li>
<li><a href='#get_wetmore_oral_equiv'><p>Get Literature Oral Equivalent Dose (deprecated).</p></a></li>
<li><a href='#hct_h'><p>KDE bandwidths for residual variability in hematocrit</p></a></li>
<li><a href='#hematocrit_infants'><p>Predict hematocrit in infants under 1 year old.</p></a></li>
<li><a href='#honda.ivive'><p>Return the assumptions used in Honda et al. 2019</p></a></li>
<li><a href='#honda2023.data'><p>Measured Caco-2 Apical-Basal Permeability Data</p></a></li>
<li><a href='#honda2023.qspr'><p>Predicted Caco-2 Apical-Basal Permeabilities</p></a></li>
<li><a href='#howgate'><p>Howgate 2006</p></a></li>
<li><a href='#httk-package'><p>httk: High-Throughput Toxicokinetics</p></a></li>
<li><a href='#httk.performance'><p>Historical Performance of R Package httk</p></a></li>
<li><a href='#httkpop'><p>httkpop: Virtual population generator for HTTK.</p></a></li>
<li><a href='#httkpop_biotophys_default'><p>Convert HTTK-Pop-generated parameters to HTTK physiological parameters</p></a></li>
<li><a href='#httkpop_direct_resample'><p>Generate a virtual population by directly resampling the NHANES data.</p></a></li>
<li><a href='#httkpop_direct_resample_inner'><p>Inner loop function called by <code>httkpop_direct_resample</code>.</p></a></li>
<li><a href='#httkpop_generate'><p>Generate a virtual population for PBTK</p></a></li>
<li><a href='#httkpop_mc'><p>httk-pop: Correlated human physiological parameter Monte Carlo</p></a></li>
<li><a href='#httkpop_virtual_indiv'><p>Generate a virtual population by the virtual individuals method.</p></a></li>
<li><a href='#hw_H'><p>KDE bandwidth for residual variability in height/weight</p></a></li>
<li><a href='#in.list'><p>Convenience Boolean (yes/no) functions to identify chemical membership in</p>
several key lists.</a></li>
<li><a href='#invitro_mc'><p>Monte Carlo for in vitro toxicokinetic parameters including uncertainty and variability.</p></a></li>
<li><a href='#is_in_inclusive'><p>Checks whether a value, or all values in a vector, is within inclusive</p>
limits</a></li>
<li><a href='#is.httk'><p>Convenience Boolean (yes/no) function to identify chemical membership and</p>
treatment within the httk project.</a></li>
<li><a href='#johnson'><p>Johnson 2006</p></a></li>
<li><a href='#kapraun2019'><p>Kapraun et al. 2019 data</p></a></li>
<li><a href='#kidney_mass_children'><p>Predict kidney mass for children</p></a></li>
<li><a href='#liver_mass_children'><p>Predict liver mass for children</p></a></li>
<li><a href='#load_dawson2021'><p>Load CLint and Fup QSPR predictions from Dawson et al. 2021.</p></a></li>
<li><a href='#load_honda2023'><p>Load Caco2 QSPR predictions from Honda et al. 2023</p></a></li>
<li><a href='#load_pradeep2020'><p>Load CLint and Fup QSPR predictions predictions from Pradeep et al. 2020.</p></a></li>
<li><a href='#load_sipes2017'><p>Load CLint and Fup QSPR predictions from Sipes et al 2017.</p></a></li>
<li><a href='#lump_tissues'><p>Lump tissue parameters into model compartments</p></a></li>
<li><a href='#lung_mass_children'><p>Predict lung mass for children</p></a></li>
<li><a href='#mcnally_dt'><p>Reference tissue masses and flows from tables in McNally et al. 2014.</p></a></li>
<li><a href='#mecdt'><p>Pre-processed NHANES data.</p></a></li>
<li><a href='#metabolism_data_Linakis2020'><p>Metabolism data involved in Linakis 2020 vignette analysis.</p></a></li>
<li><a href='#monte_carlo'><p>Monte Carlo for toxicokinetic model parameters</p></a></li>
<li><a href='#Obach2008'><p>Published Pharmacokinetic Parameters from Obach et al. 2008</p></a></li>
<li><a href='#onlyp'><p>NHANES Exposure Data</p></a></li>
<li><a href='#pancreas_mass_children'><p>Predict pancreas mass for children</p></a></li>
<li><a href='#parameterize_1comp'><p>Parameters for a one compartment (empirical) toxicokinetic model</p></a></li>
<li><a href='#parameterize_3comp'><p>Parameters for a three-compartment toxicokinetic model (dynamic)</p></a></li>
<li><a href='#parameterize_fetal_pbtk'><p>Parameterize_fetal_PBTK</p></a></li>
<li><a href='#parameterize_gas_pbtk'><p>Parameters for a generic gas inhalation physiologically-based toxicokinetic model</p></a></li>
<li><a href='#parameterize_pbtk'><p>Parameters for a generic physiologically-based toxicokinetic model</p></a></li>
<li><a href='#parameterize_schmitt'><p>Parameters for Schmitt's (2008) Tissue Partition Coefficient Method</p></a></li>
<li><a href='#parameterize_steadystate'><p>Parameters for a three-compartment toxicokinetic model at steady-state</p></a></li>
<li><a href='#pc.data'><p>Partition Coefficient Data</p></a></li>
<li><a href='#pearce2017regression'><p>Pearce et al. 2017 data</p></a></li>
<li><a href='#pharma'><p>DRUGS|NORMAN: Pharmaceutical List with EU, Swiss, US Consumption Data</p></a></li>
<li><a href='#physiology.data'><p>Species-specific physiology parameters</p></a></li>
<li><a href='#pksim.pcs'><p>Partition Coefficients from PK-Sim</p></a></li>
<li><a href='#pradeep2020'><p>Pradeep et al. 2020</p></a></li>
<li><a href='#predict_partitioning_schmitt'><p>Predict partition coefficients using the method from Schmitt (2008).</p></a></li>
<li><a href='#pregnonpregaucs'><p>AUCs for Pregnant and Non-Pregnant Women</p></a></li>
<li><a href='#propagate_invitrouv_1comp'><p>Propagates uncertainty and variability in in vitro HTTK data into one</p>
compartment model parameters</a></li>
<li><a href='#propagate_invitrouv_3comp'><p>Propagates uncertainty and variability in in vitro HTTK data into three</p>
compartment model parameters</a></li>
<li><a href='#propagate_invitrouv_pbtk'><p>Propagates uncertainty and variability in in vitro HTTK data into PBPK</p>
model parameters</a></li>
<li><a href='#r_left_censored_norm'><p>Returns draws from a normal distribution with a lower censoring limit of lod</p>
(limit of detection)</a></li>
<li><a href='#reset_httk'><p>Reset HTTK to Default Data Tables</p></a></li>
<li><a href='#rfun'><p>Randomly draws from a one-dimensional KDE</p></a></li>
<li><a href='#rmed0non0u95'><p>Draw random numbers with LOD median but non-zero upper 95th percentile</p></a></li>
<li><a href='#scale_dosing'><p>Scale mg/kg body weight doses according to body weight and units</p></a></li>
<li><a href='#scr_h'><p>KDE bandwidths for residual variability in serum creatinine</p></a></li>
<li><a href='#set_httk_precision'><p>set_httk_precision</p></a></li>
<li><a href='#sipes2017'><p>Sipes et al. 2017 data</p></a></li>
<li><a href='#skeletal_muscle_mass'><p>Predict skeletal muscle mass</p></a></li>
<li><a href='#skeletal_muscle_mass_children'><p>Predict skeletal muscle mass for children</p></a></li>
<li><a href='#skin_mass_bosgra'><p>Predict skin mass</p></a></li>
<li><a href='#solve_1comp'><p>Solve one compartment TK model</p></a></li>
<li><a href='#solve_3comp'><p>Solve_3comp</p></a></li>
<li><a href='#solve_fetal_pbtk'><p>Solve_fetal_PBTK</p></a></li>
<li><a href='#solve_gas_pbtk'><p>solve_gas_pbtk</p></a></li>
<li><a href='#solve_model'><p>Solve_model</p></a></li>
<li><a href='#solve_pbtk'><p>Solve_PBTK</p></a></li>
<li><a href='#spleen_mass_children'><p>Predict spleen mass for children</p></a></li>
<li><a href='#supptab1_Linakis2020'><p>Supplementary output from Linakis 2020 vignette analysis.</p></a></li>
<li><a href='#supptab2_Linakis2020'><p>More supplementary output from Linakis 2020 vignette analysis.</p></a></li>
<li><a href='#Tables.Rdata.stamp'><p>A timestamp of table creation</p></a></li>
<li><a href='#tissue_masses_flows'><p>Given a data.table describing a virtual population by the NHANES quantities,</p>
generates HTTK physiological parameters for each individual.</a></li>
<li><a href='#tissue_scale'><p>Allometric scaling.</p></a></li>
<li><a href='#tissue.data'><p>Tissue composition and species-specific physiology parameters</p></a></li>
<li><a href='#wambaugh2019'><p>in vitro Toxicokinetic Data from Wambaugh et al. (2019)</p></a></li>
<li><a href='#wambaugh2019.nhanes'><p>NHANES Chemical Intake Rates for chemicals in Wambaugh et al. (2019)</p></a></li>
<li><a href='#wambaugh2019.raw'><p>Raw Bayesian in vitro Toxicokinetic Data Analysis from Wambaugh et al. (2019)</p></a></li>
<li><a href='#wambaugh2019.seem3'><p>ExpoCast SEEM3 Consensus Exposure Model Predictions for Chemical Intake Rates</p></a></li>
<li><a href='#wambaugh2019.tox21'><p>Tox21 2015 Active Hit Calls (EPA)</p></a></li>
<li><a href='#wang2018'><p>Wang et al. 2018</p>
Wang et al. (2018) screened the blood of 75 pregnant women for the presence
of environmental organic acids (EOAs) and identified mass spectral features
corresponding to 453 chemical formulae of which 48 could be mapped to likely
structures. Of the 48 with tentative structures the identity of six were
confirmed with available chemical standards.</a></li>
<li><a href='#well_param'><p>Microtiter Plate Well Descriptions for Armitage et al. (2014) Model</p></a></li>
<li><a href='#Wetmore2012'><p>Published toxicokinetic predictions based on in vitro data from Wetmore et</p>
al. 2012.</a></li>
<li><a href='#wfl'><p>WHO weight-for-length charts</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.3.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-18</td>
</tr>
<tr>
<td>Title:</td>
<td>High-Throughput Toxicokinetics</td>
</tr>
<tr>
<td>Description:</td>
<td>Pre-made models that can be rapidly tailored to various chemicals
             and species using chemical-specific in vitro data and physiological 
             information. These tools allow incorporation of chemical 
             toxicokinetics ("TK") and in vitro-in vivo extrapolation ("IVIVE") 
             into bioinformatics, as described by Pearce et al. (2017) 
             (&lt;<a href="https://doi.org/10.18637%2Fjss.v079.i04">doi:10.18637/jss.v079.i04</a>&gt;). Chemical-specific 
             in vitro data characterizing toxicokinetics have been obtained 
             from relatively high-throughput experiments.   The 
             chemical-independent ("generic") physiologically-based ("PBTK") and empirical 
             (for example, one compartment) "TK" models included here can be 
             parameterized with in vitro data or in silico predictions which are 
             provided for thousands of chemicals, multiple exposure routes, 
             and various species. High throughput toxicokinetics ("HTTK") is the 
             combination of in vitro data and generic models. We establish the
             expected accuracy of HTTK for chemicals without in vivo data 
             through statistical evaluation of HTTK predictions for chemicals
             where in vivo data do exist. The models are systems of ordinary 
             differential equations that are developed in MCSim and solved
             using compiled (C-based) code for speed. A Monte Carlo sampler is
             included for simulating human biological variability
             (Ring et al., 2017 &lt;<a href="https://doi.org/10.1016%2Fj.envint.2017.06.004">doi:10.1016/j.envint.2017.06.004</a>&gt;)
             and propagating parameter uncertainty 
             (Wambaugh et al., 2019 &lt;<a href="https://doi.org/10.1093%2Ftoxsci%2Fkfz205">doi:10.1093/toxsci/kfz205</a>&gt;). 
             Empirically calibrated methods are included for predicting 
             tissue:plasma partition coefficients and volume of distribution  
             (Pearce et al., 2017 &lt;<a href="https://doi.org/10.1007%2Fs10928-017-9548-7">doi:10.1007/s10928-017-9548-7</a>&gt;).
             These functions and data provide a set of tools for using IVIVE to
             convert concentrations from high-throughput screening experiments
             (for example, Tox21, ToxCast) to real-world exposures via reverse 
             dosimetry (also known as "RTK")
             (Wetmore et al., 2015 &lt;<a href="https://doi.org/10.1093%2Ftoxsci%2Fkfv171">doi:10.1093/toxsci/kfv171</a>&gt;).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>deSolve, msm, data.table, survey, mvtnorm, truncnorm, stats,
graphics, utils, magrittr, purrr, methods, Rdpack, ggplot2</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, R.rsp, gplots, scales, EnvStats, MASS,
RColorBrewer, stringr, reshape, viridis, gmodels, colorspace,
cowplot, ggrepel, dplyr, forcats, smatr, gridExtra, readxl, ks</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, R.rsp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.epa.gov/chemical-research/rapid-chemical-exposure-and-dose-research">https://www.epa.gov/chemical-research/rapid-chemical-exposure-and-dose-research</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/USEPA/CompTox-ExpoCast-httk">https://github.com/USEPA/CompTox-ExpoCast-httk</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-21 18:56:25 UTC; jwambaug</td>
</tr>
<tr>
<td>Author:</td>
<td>John Wambaugh <a href="https://orcid.org/0000-0002-4024-534X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Sarah Davidson-Fritz
    <a href="https://orcid.org/0000-0002-2891-9380"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Robert Pearce <a href="https://orcid.org/0000-0003-3168-4049"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Caroline Ring <a href="https://orcid.org/0000-0002-0463-1251"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Greg Honda <a href="https://orcid.org/0000-0001-7713-9850"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Mark Sfeir [aut],
  Matt Linakis <a href="https://orcid.org/0000-0003-0526-2395"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Dustin Kapraun <a href="https://orcid.org/0000-0001-5570-6383"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Nathan Pollesch <a href="https://orcid.org/0000-0001-8992-9294"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Miyuki Breen <a href="https://orcid.org/0000-0001-8511-4653"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Shannon Bell <a href="https://orcid.org/0000-0002-5754-6085"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Xiaoqing Chang <a href="https://orcid.org/0000-0003-0752-1848"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Todor Antonijevic <a href="https://orcid.org/0000-0002-0248-8412"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Jimena Davis [ctb],
  Elaina Kenyon <a href="https://orcid.org/0000-0001-8781-2391"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Katie Paul Friedman
    <a href="https://orcid.org/0000-0002-2710-1691"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  James Sluka <a href="https://orcid.org/0000-0002-5901-1404"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Noelle Sinski [ctb],
  Nisha Sipes <a href="https://orcid.org/0000-0003-4203-6426"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Barbara Wetmore <a href="https://orcid.org/0000-0002-6878-5348"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Lily Whipple [ctb],
  Woodrow Setzer <a href="https://orcid.org/0000-0002-6709-9186"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>John Wambaugh &lt;wambaugh.john@epa.gov&gt;</td>
</tr>
<tr>
<td>Copyright:</td>
<td>This package is primarily developed by employees of the U.S.
Federal government as part of their official duties and is
therefore public domain.</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-27 00:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_chemtable'>Add a table of chemical information for use in making httk predictions.</h2><span id='topic+add_chemtable'></span>

<h3>Description</h3>

<p>This function adds chemical-specific information to the table
chem.physical_and_invitro.data. This table is queried by the model
parameterization functions when attempting to parameterize a model, so
adding sufficient data to this table allows additional chemicals to be
modeled.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_chemtable(
  new.table,
  data.list,
  current.table = NULL,
  reference = NULL,
  species = NULL,
  overwrite = FALSE,
  sig.fig = 4,
  clint.pvalue.overwrite = TRUE,
  allow.na = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_chemtable_+3A_new.table">new.table</code></td>
<td>
<p>Object of class data.frame containing one row per chemical,
with each chemical minimally described by a CAS number.</p>
</td></tr>
<tr><td><code id="add_chemtable_+3A_data.list">data.list</code></td>
<td>
<p>This list identifies which properties are to be read from
the table. Each item in the list should point to a column in the table
new.table. Valid names in the list are: 'Compound', 'CAS', 'DSSTox.GSID'
'SMILES.desalt', 'Reference', 'Species', 'MW', 'logP', 'pKa_Donor',
'pKa_Accept', 'logMA', 'Clint', 'Clint.pValue', 'Funbound.plasma',
'Fabs', 'Fgut', 'Rblood2plasma'.</p>
</td></tr>
<tr><td><code id="add_chemtable_+3A_current.table">current.table</code></td>
<td>
<p>This is the table to which data are being added.</p>
</td></tr>
<tr><td><code id="add_chemtable_+3A_reference">reference</code></td>
<td>
<p>This is the reference for the data in the new table. This
may be omitted if a column in data.list gives the reference value for each
chemical.</p>
</td></tr>
<tr><td><code id="add_chemtable_+3A_species">species</code></td>
<td>
<p>This is the species for the data in the new table. This may
be omitted if a column in data.list gives the species value for each
chemical or if the data are not species-specific (e.g., MW).</p>
</td></tr>
<tr><td><code id="add_chemtable_+3A_overwrite">overwrite</code></td>
<td>
<p>If overwrite=TRUE then data in current.table will be
replaced by any data in new.table that is for the same chemical and
property. If overwrite=FALSE (DEFAULT) then new data for the same chemical
and property are ignored.  Funbound.plasma values of 0 (below limit of
detection) are overwritten either way.</p>
</td></tr>
<tr><td><code id="add_chemtable_+3A_sig.fig">sig.fig</code></td>
<td>
<p>Sets the number of significant figures stored (defaults to 4)</p>
</td></tr>
<tr><td><code id="add_chemtable_+3A_clint.pvalue.overwrite">clint.pvalue.overwrite</code></td>
<td>
<p>If TRUE then the Cl_int p-value is set to NA 
when the Cl_int value is changed unless a new p-value is provided. (defaults
to TRUE)</p>
</td></tr>
<tr><td><code id="add_chemtable_+3A_allow.na">allow.na</code></td>
<td>
<p>If TRUE (default is FALSE) then NA values are written to the
table, otherwise they are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>data.frame</code></td>
<td>
<p>A new data.frame containing the data in
current.table augmented by new.table</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(httk)

my.new.data &lt;- as.data.frame(c("A","B","C"),stringsAsFactors=FALSE)
my.new.data &lt;- cbind(my.new.data,as.data.frame(c(
                     "111-11-2","222-22-0","333-33-5"),
                     stringsAsFactors=FALSE))
my.new.data &lt;- cbind(my.new.data,as.data.frame(c("DTX1","DTX2","DTX3"),
                    stringsAsFactors=FALSE))
my.new.data &lt;- cbind(my.new.data,as.data.frame(c(200,200,200)))
my.new.data &lt;- cbind(my.new.data,as.data.frame(c(2,3,4)))
my.new.data &lt;- cbind(my.new.data,as.data.frame(c(0.01,0.02,0.3)))
my.new.data &lt;- cbind(my.new.data,as.data.frame(c(0,10,100)))
colnames(my.new.data) &lt;- c("Name","CASRN","DTXSID","MW","LogP","Fup","CLint")

chem.physical_and_invitro.data &lt;- add_chemtable(my.new.data,
                                  current.table=
                                    chem.physical_and_invitro.data,
                                  data.list=list(
                                  Compound="Name",
                                  CAS="CASRN",
                                  DTXSID="DTXSID",
                                  MW="MW",
                                  logP="LogP",
                                  Funbound.plasma="Fup",
                                  Clint="CLint"),
                                  species="Human",
                                  reference="MyPaper 2015")
parameterize_steadystate(chem.name="C")  
calc_css(chem.name="B")                                

# Initialize a column describing proton donors ("acids")
my.new.data$pka.a &lt;- NA 
# set chemical C to an acid (pKa_donor = 5):
my.new.data[my.new.data$Name=="C","pka.a"] &lt;- "5"
chem.physical_and_invitro.data &lt;- add_chemtable(my.new.data,
                                  current.table=
                                    chem.physical_and_invitro.data,
                                 data.list=list(
                                 Compound="Name",
                                 CAS="CASRN",
                                 DTXSID="DTXSID",
                                 pKa_Donor="pka.a"),
                                 species="Human",
                                 reference="MyPaper 2015") 

# Note Rblood2plasma and hepatic bioavailability change (relative to above):
parameterize_steadystate(chem.name="C")  

# Initialize a column describing proton acceptors ("bases")
my.new.data$pka.b &lt;- NA 
# set chemical B to a base with multiple pka's (pKa_accept = 7 and 8):
my.new.data[my.new.data$Name=="B","pka.b"] &lt;- "7;8"
chem.physical_and_invitro.data &lt;- add_chemtable(my.new.data,
                                  current.table=
                                    chem.physical_and_invitro.data,
                                 data.list=list(
                                 Compound="Name",
                                 CAS="CASRN",
                                 DTXSID="DTXSID",
                                 pKa_Accept="pka.b"),
                                 species="Human",
                                 reference="MyPaper 2015") 
# Note that average and max change (relative to above):
calc_css(chem.name="B")     


</code></pre>

<hr>
<h2 id='age_draw_smooth'>Draws ages from a smoothed distribution for a given gender/race combination</h2><span id='topic+age_draw_smooth'></span>

<h3>Description</h3>

<p>This function should usually not be called directly by the user. It is used by
<code>httkpop_generate()</code> in &quot;virtual-individuals&quot; mode.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>age_draw_smooth(gender, reth, nsamp, agelim_months, nhanes_mec_svy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="age_draw_smooth_+3A_gender">gender</code></td>
<td>
<p>Gender. Either 'Male' or 'Female'.</p>
</td></tr>
<tr><td><code id="age_draw_smooth_+3A_reth">reth</code></td>
<td>
<p>Race/ethnicity. One of 'Mexican American', 'Other Hispanic',
'Non-Hispanic Black', 'Non-Hispanic White', 'Other'.</p>
</td></tr>
<tr><td><code id="age_draw_smooth_+3A_nsamp">nsamp</code></td>
<td>
<p>Number of ages to draw.</p>
</td></tr>
<tr><td><code id="age_draw_smooth_+3A_agelim_months">agelim_months</code></td>
<td>
<p>Two-element numeric vector giving the minimum and
maximum ages in months to include.</p>
</td></tr>
<tr><td><code id="age_draw_smooth_+3A_nhanes_mec_svy">nhanes_mec_svy</code></td>
<td>
<p><code>surveydesign</code> object created from
<code><a href="#topic+mecdt">mecdt</a></code> using <code><a href="survey.html#topic+svydesign">svydesign</a></code> (this is done in
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list with members 'ages_months' and 'ages_years', each
numeric of length <code>nsamp</code>, giving the sampled ages in months and years.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='apply_clint_adjustment'>Correct the measured intrinsive hepatic clearance for fraction free</h2><span id='topic+apply_clint_adjustment'></span>

<h3>Description</h3>

<p>This function uses the free fraction estimated from Kilford et al. (2008) 
to increase the in vitro measure intrinsic hepatic clearance. The assumption
that chemical that is bound in vitro is not available to be metabolized and
therefore the actual rate of clearance is actually faster. Note that in most
high throughput TK models included in the package this increase is offset by
the assumption of &quot;restrictive clearance&quot; &ndash; that is, the rate of hepatic
metabolism is slowed to account for the free fraction of chemical in plasma.
This adjustment was made starting in Wetmore et al. (2015) in order to better
predict plasma concentrations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apply_clint_adjustment(
  Clint,
  Fu_hep = NULL,
  Pow = NULL,
  pKa_Donor = NULL,
  pKa_Accept = NULL,
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="apply_clint_adjustment_+3A_clint">Clint</code></td>
<td>
<p>In vitro measured intrinsic hepatic clearance in units of
(ul/min/million hepatocytes).</p>
</td></tr>
<tr><td><code id="apply_clint_adjustment_+3A_fu_hep">Fu_hep</code></td>
<td>
<p>Estimated fraction of chemical free for metabolism in the 
in vitro assay, estimated by default from the method of Kilford et al. (2008)
using <code><a href="#topic+calc_hep_fu">calc_hep_fu</a></code></p>
</td></tr>
<tr><td><code id="apply_clint_adjustment_+3A_pow">Pow</code></td>
<td>
<p>The octanal:water equilibrium partition coefficient</p>
</td></tr>
<tr><td><code id="apply_clint_adjustment_+3A_pka_donor">pKa_Donor</code></td>
<td>
<p>A string containing hydrogen donor ionization equilibria, 
concatenated with commas. Can be &quot;NA&quot; if none exist.</p>
</td></tr>
<tr><td><code id="apply_clint_adjustment_+3A_pka_accept">pKa_Accept</code></td>
<td>
<p>A string containing hydrogen acceptance ionization equilibria, 
concatenated with commas. Can be &quot;NA&quot; if none exist.</p>
</td></tr>
<tr><td><code id="apply_clint_adjustment_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Intrinsic hepatic clearance increased to take into account binding
in the in vitro assay
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197. 
Wetmore BA, Wambaugh JF, Allen B, Ferguson SS, Sochaski MA, Setzer RW, Houck KA, Strope CL, Cantwell K, Judson RS, others (2015).
&ldquo;Incorporating high-throughput exposure predictions with dosimetry-adjusted in vitro bioactivity to inform chemical toxicity testing.&rdquo;
<em>Toxicological Sciences</em>, <b>148</b>(1), 121&ndash;136.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_hep_fu">calc_hep_fu</a></code>
</p>

<hr>
<h2 id='apply_fup_adjustment'>Correct the measured fraction unbound in plasma for lipid binding</h2><span id='topic+apply_fup_adjustment'></span>

<h3>Description</h3>

<p>This function uses the lipid binding correction estimated by Pearce
et al. (2017) to decrease the fraction unbound in plasma
(f<sub>up</sub>). This correction
assumes that there is additional in vivo binding to lipid, which
has a greater impact on neutral lipophilic compounds.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apply_fup_adjustment(
  fup,
  fup.correction = NULL,
  Pow = NULL,
  pKa_Donor = NULL,
  pKa_Accept = NULL,
  suppress.messages = FALSE,
  minimum.Funbound.plasma = 1e-04
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="apply_fup_adjustment_+3A_fup">fup</code></td>
<td>
<p>In vitro measured fraction unbound in plasma</p>
</td></tr>
<tr><td><code id="apply_fup_adjustment_+3A_fup.correction">fup.correction</code></td>
<td>
<p>Estimated correction to account for additional lipid
binding in vivo (Pearce et al., 2017) from <code><a href="#topic+calc_fup_correction">calc_fup_correction</a></code></p>
</td></tr>
<tr><td><code id="apply_fup_adjustment_+3A_pow">Pow</code></td>
<td>
<p>The octanal:water equilibrium partition coefficient</p>
</td></tr>
<tr><td><code id="apply_fup_adjustment_+3A_pka_donor">pKa_Donor</code></td>
<td>
<p>A string containing hydrogen donor ionization equilibria, 
concatenated with commas. Can be &quot;NA&quot; if none exist.</p>
</td></tr>
<tr><td><code id="apply_fup_adjustment_+3A_pka_accept">pKa_Accept</code></td>
<td>
<p>A string containing hydrogen acceptance ionization equilibria, 
concatenated with commas. Can be &quot;NA&quot; if none exist.</p>
</td></tr>
<tr><td><code id="apply_fup_adjustment_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="apply_fup_adjustment_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
f<sub>up</sub>
<p>is not allowed to drop below this value (default is 0.0001).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fraction unbound in plasma adjusted to take into account binding
in the in vitro assay
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197. 
Wetmore BA, Wambaugh JF, Allen B, Ferguson SS, Sochaski MA, Setzer RW, Houck KA, Strope CL, Cantwell K, Judson RS, others (2015).
&ldquo;Incorporating high-throughput exposure predictions with dosimetry-adjusted in vitro bioactivity to inform chemical toxicity testing.&rdquo;
<em>Toxicological Sciences</em>, <b>148</b>(1), 121&ndash;136.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_fup_correction">calc_fup_correction</a></code>
</p>

<hr>
<h2 id='armitage_estimate_sarea'>Estimate well surface area</h2><span id='topic+armitage_estimate_sarea'></span>

<h3>Description</h3>

<p>Estimate geometry surface area of plastic in well plate based on well plate
format suggested values from Corning.  option.plastic == TRUE (default) give
nonzero surface area (sarea, m^2) option.bottom == TRUE (default) includes
surface area of the bottom of the well in determining sarea.  Optionally
include user values for working volume (v_working, m^3) and surface area.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>armitage_estimate_sarea(
  tcdata = NA,
  this.well_number = 384,
  this.cell_yield = NA,
  this.v_working = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="armitage_estimate_sarea_+3A_tcdata">tcdata</code></td>
<td>
<p>A data table with well_number corresponding to plate format,
optionally include v_working, sarea, option.bottom, and option.plastic</p>
</td></tr>
<tr><td><code id="armitage_estimate_sarea_+3A_this.well_number">this.well_number</code></td>
<td>
<p>For single value, plate format default is 384, used
if is.na(tcdata)==TRUE</p>
</td></tr>
<tr><td><code id="armitage_estimate_sarea_+3A_this.cell_yield">this.cell_yield</code></td>
<td>
<p>For single value, optionally supply cell_yield,
otherwise estimated based on well number</p>
</td></tr>
<tr><td><code id="armitage_estimate_sarea_+3A_this.v_working">this.v_working</code></td>
<td>
<p>For single value, optionally supply working volume,
otherwise estimated based on well number (m^3)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data table composed of any input data.table <em>tcdata</em>
with only the following columns either created or altered by this function:  
</p>

<table>
<tr>
 <td style="text-align: center;">
<strong>Column Name</strong> </td><td style="text-align: center;"> <strong>Description</strong> </td><td style="text-align: center;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: center;">
well_number </td><td style="text-align: center;"> number of wells on plate </td><td style="text-align: center;"> </td>
</tr>
<tr>
 <td style="text-align: center;">
sarea </td><td style="text-align: center;"> surface area </td><td style="text-align: center;"> m^2 </td>
</tr>
<tr>
 <td style="text-align: center;">
cell_yield </td><td style="text-align: center;"> number of cells </td><td style="text-align: center;"> cells </td>
</tr>
<tr>
 <td style="text-align: center;"> 
v_working </td><td style="text-align: center;"> working (filled) volume of each well </td><td style="text-align: center;"> uL </td>
</tr>
<tr>
 <td style="text-align: center;">
v_total </td><td style="text-align: center;"> total volume of each well </td><td style="text-align: center;"> uL </td>
</tr>
<tr>
 <td style="text-align: center;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Greg Honda
</p>


<h3>References</h3>

<p>Armitage JM, Arnot JA, Wania F, Mackay D (2013).
&ldquo;Development and evaluation of a mechanistic bioconcentration model for ionogenic organic chemicals in fish.&rdquo;
<em>Environmental toxicology and chemistry</em>, <b>32</b>(1), 115&ndash;128.
</p>

<hr>
<h2 id='armitage_eval'>Evaluate the updated Armitage model</h2><span id='topic+armitage_eval'></span>

<h3>Description</h3>

<p>Evaluate the Armitage model for chemical distributon <em>in vitro</em>. Takes input
as data table or vectors of values. Outputs a data table. Updates over
the model published in Armitage et al. (2014) include binding to plastic walls
and lipid and protein compartments in cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>armitage_eval(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  casrn.vector = NA_character_,
  nomconc.vector = 1,
  this.well_number = 384,
  this.FBSf = NA_real_,
  tcdata = NA,
  this.sarea = NA_real_,
  this.v_total = NA_real_,
  this.v_working = NA_real_,
  this.cell_yield = NA_real_,
  this.Tsys = 37,
  this.Tref = 298.15,
  this.option.kbsa2 = FALSE,
  this.option.swat2 = FALSE,
  this.pseudooct = 0.01,
  this.memblip = 0.04,
  this.nlom = 0.2,
  this.P_nlom = 0.035,
  this.P_dom = 0.05,
  this.P_cells = 1,
  this.csalt = 0.15,
  this.celldensity = 1,
  this.cellmass = 3,
  this.f_oc = 1,
  this.conc_ser_alb = 24,
  this.conc_ser_lip = 1.9,
  this.Vdom = 0,
  this.pH = 7,
  restrict.ion.partitioning = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="armitage_eval_+3A_chem.cas">chem.cas</code></td>
<td>
<p>A single or vector of Chemical Abstracts Service Registry 
Number(s) (CAS-RN) of desired chemical(s).</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_chem.name">chem.name</code></td>
<td>
<p>A single or vector of name(s)) of desired chemical(s).</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_dtxsid">dtxsid</code></td>
<td>
<p>A single or vector ofEPA's DSSTox Structure ID(s) 
(<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_casrn.vector">casrn.vector</code></td>
<td>
<p>A deprecated argument specifying a single or vector of 
Chemical Abstracts Service Registry 
Number(s) (CAS-RN) of desired chemical(s).</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_nomconc.vector">nomconc.vector</code></td>
<td>
<p>For vector or single value, micromolar (uM = mol/L) nominal 
concentration (e.g. AC50 value)</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.well_number">this.well_number</code></td>
<td>
<p>For single value, plate format default is 384, used
if is.na(tcdata)==TRUE. This value chooses default surface area settings for
<code><a href="#topic+armitage_estimate_sarea">armitage_estimate_sarea</a></code> based on the number of plates per well.</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.fbsf">this.FBSf</code></td>
<td>
<p>Fraction fetal bovine serum, must be entered by user.</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_tcdata">tcdata</code></td>
<td>
<p>A data.table with casrn, nomconc, MP, gkow, gkaw, gswat, sarea,
v_total, v_working. Otherwise supply single values to this.params (e.g., this.sarea,
this.v_total, etc.). Chemical parameters are taken from 
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>.</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.sarea">this.sarea</code></td>
<td>
<p>Surface area per well (m^2)</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.v_total">this.v_total</code></td>
<td>
<p>Total volume per well (uL)</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.v_working">this.v_working</code></td>
<td>
<p>Working volume per well (uL)</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.cell_yield">this.cell_yield</code></td>
<td>
<p>Number of cells per well</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.tsys">this.Tsys</code></td>
<td>
<p>System temperature (degrees C)</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.tref">this.Tref</code></td>
<td>
<p>Reference temperature (degrees K)</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.option.kbsa2">this.option.kbsa2</code></td>
<td>
<p>Use alternative bovine-serum-albumin partitioning
model</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.option.swat2">this.option.swat2</code></td>
<td>
<p>Use alternative water solubility correction</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.pseudooct">this.pseudooct</code></td>
<td>
<p>Pseudo-octanol cell storage lipid content</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.memblip">this.memblip</code></td>
<td>
<p>Membrane lipid content of cells</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.nlom">this.nlom</code></td>
<td>
<p>Structural protein content of cells</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.p_nlom">this.P_nlom</code></td>
<td>
<p>Proportionality constant to octanol structural protein</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.p_dom">this.P_dom</code></td>
<td>
<p>Proportionality constant to dissolve organic material</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.p_cells">this.P_cells</code></td>
<td>
<p>Proportionality constant to octanol storage lipid</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.csalt">this.csalt</code></td>
<td>
<p>Ionic strength of buffer (M = mol/L)</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.celldensity">this.celldensity</code></td>
<td>
<p>Cell density kg/L, g/mL</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.cellmass">this.cellmass</code></td>
<td>
<p>Mass per cell, ng/cell</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.f_oc">this.f_oc</code></td>
<td>
<p>Everything assumed to be like proteins</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.conc_ser_alb">this.conc_ser_alb</code></td>
<td>
<p>Mass concentration of albumin in serum (g/L)</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.conc_ser_lip">this.conc_ser_lip</code></td>
<td>
<p>Mass concentration of lipids in serum (g/L)</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.vdom">this.Vdom</code></td>
<td>
<p>0 ml, the volume of dissolved organic matter (DOM)</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_this.ph">this.pH</code></td>
<td>
<p>7.0, pH of cell culture</p>
</td></tr>
<tr><td><code id="armitage_eval_+3A_restrict.ion.partitioning">restrict.ion.partitioning</code></td>
<td>
<p>FALSE, Should we restrict the chemical available to partition to only the neutral fraction?</p>
</td></tr>
</table>


<h3>Value</h3>


<table>
<tr>
 <td style="text-align: left;">
<strong>Param</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td><td style="text-align: left;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
casrn </td><td style="text-align: left;"> Chemical Abstracts Service Registry Number </td><td style="text-align: left;"> character </td>
</tr>
<tr>
 <td style="text-align: left;">
nomconc </td><td style="text-align: left;"> Nominal Concentration </td><td style="text-align: left;"> uM=umol/L </td>
</tr>
<tr>
 <td style="text-align: left;">       
well_number </td><td style="text-align: left;"> Number of wells in plate (used to set default surface area) </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">   
sarea </td><td style="text-align: left;"> Surface area of well </td><td style="text-align: left;"> m^2 </td>
</tr>
<tr>
 <td style="text-align: left;">         
v_total </td><td style="text-align: left;"> Total volume of well </td><td style="text-align: left;"> uL </td>
</tr>
<tr>
 <td style="text-align: left;">       
v_working </td><td style="text-align: left;"> Filled volume of well </td><td style="text-align: left;"> uL </td>
</tr>
<tr>
 <td style="text-align: left;">     
cell_yield </td><td style="text-align: left;"> Number of cells </td><td style="text-align: left;"> cells </td>
</tr>
<tr>
 <td style="text-align: left;">    
gkow </td><td style="text-align: left;"> The log10 octanol to water (PC) (logP)</td><td style="text-align: left;"> log10 unitless ratio </td>
</tr>
<tr>
 <td style="text-align: left;">          
logHenry </td><td style="text-align: left;"> The log10 Henry's law constant '</td><td style="text-align: left;"> log10 unitless ratio </td>
</tr>
<tr>
 <td style="text-align: left;">      
gswat </td><td style="text-align: left;"> The log10 water solubility (logWSol) </td><td style="text-align: left;"> log10 mg/L </td>
</tr>
<tr>
 <td style="text-align: left;">         
MP </td><td style="text-align: left;"> The chemical compound melting point </td><td style="text-align: left;"> degrees Kelvin </td>
</tr>
<tr>
 <td style="text-align: left;">           
MW </td><td style="text-align: left;"> The chemical compound molecular weight </td><td style="text-align: left;"> g/mol </td>
</tr>
<tr>
 <td style="text-align: left;">            
gkaw </td><td style="text-align: left;"> The air to water PC </td><td style="text-align: left;"> unitless ratio </td>
</tr>
<tr>
 <td style="text-align: left;">          
dsm </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">           
duow </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">          
duaw </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">          
dumw </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">          
gkmw </td><td style="text-align: left;"> log10 </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">          
gkcw </td><td style="text-align: left;"> The log10 cell/tissue to water PC </td><td style="text-align: left;"> log10 unitless ratio</td>
</tr>
<tr>
 <td style="text-align: left;">          
gkbsa </td><td style="text-align: left;"> The log10 bovine serum albumin to water partitiion coefficient </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">         
gkpl </td><td style="text-align: left;"> log10</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">          
ksalt </td><td style="text-align: left;"> Setschenow constant </td><td style="text-align: left;"> L/mol </td>
</tr>
<tr>
 <td style="text-align: left;">        
Tsys </td><td style="text-align: left;"> System temperature </td><td style="text-align: left;"> degrees C </td>
</tr>
<tr>
 <td style="text-align: left;">          
Tref </td><td style="text-align: left;"> Reference temperature</td><td style="text-align: left;"> degrees K </td>
</tr>
<tr>
 <td style="text-align: left;">          
option.kbsa2 </td><td style="text-align: left;"> Use alternative bovine-serum-albumin partitioning model </td><td style="text-align: left;"> logical </td>
</tr>
<tr>
 <td style="text-align: left;">  
option.swat2 </td><td style="text-align: left;"> Use alternative water solubility correction </td><td style="text-align: left;"> logical </td>
</tr>
<tr>
 <td style="text-align: left;">  
FBSf </td><td style="text-align: left;"> Fraction fetal bovine serum </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">          
pseudooct </td><td style="text-align: left;"> Pseudo-octanol cell storage lipid content </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">     
memblip </td><td style="text-align: left;"> Membrane lipid content of cells </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">       
nlom </td><td style="text-align: left;"> Structural protein content of cells </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
P_nlom </td><td style="text-align: left;"> Proportionality constant to octanol structural protein </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">   
P_dom </td><td style="text-align: left;"> Proportionality constant to dissolved organic material (DOM) </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">         
P_cells </td><td style="text-align: left;"> Proportionality constant to octanol storage lipid </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">      
csalt </td><td style="text-align: left;"> Ionic strength of buffer </td><td style="text-align: left;"> M=mol/L </td>
</tr>
<tr>
 <td style="text-align: left;">
celldensity </td><td style="text-align: left;"> Cell density </td><td style="text-align: left;"> kg/L, g/mL </td>
</tr>
<tr>
 <td style="text-align: left;">   
cellmass </td><td style="text-align: left;"> Mass per cell </td><td style="text-align: left;"> ng/cell </td>
</tr>
<tr>
 <td style="text-align: left;">      
f_oc </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
cellwat </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">       
Tcor </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">          
Vm </td><td style="text-align: left;"> Volume of media </td><td style="text-align: left;"> L </td>
</tr>
<tr>
 <td style="text-align: left;">            
Vwell </td><td style="text-align: left;"> Volume of medium (aqueous phase only) </td><td style="text-align: left;"> L </td>
</tr>
<tr>
 <td style="text-align: left;">         
Vair </td><td style="text-align: left;"> Volume of head space </td><td style="text-align: left;"> L </td>
</tr>
<tr>
 <td style="text-align: left;">          
Vcells </td><td style="text-align: left;"> Volume of cells/tissue</td><td style="text-align: left;"> L </td>
</tr>
<tr>
 <td style="text-align: left;">        
Valb </td><td style="text-align: left;"> Volume of serum albumin </td><td style="text-align: left;"> L </td>
</tr>
<tr>
 <td style="text-align: left;">         
Vslip </td><td style="text-align: left;"> Volume of serum lipids </td><td style="text-align: left;"> L </td>
</tr>
<tr>
 <td style="text-align: left;">         
Vdom </td><td style="text-align: left;"> Volume of dissolved organic matter</td><td style="text-align: left;"> L </td>
</tr>
<tr>
 <td style="text-align: left;">          
F_ratio </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">       
gs1.GSE </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">       
s1.GSE </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">        
gss.GSE </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">       
ss.GSE </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">        
kmw </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">           
kow </td><td style="text-align: left;"> The octanol to water PC (i.e., 10^gkow) </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">           
kaw </td><td style="text-align: left;"> The air to water PC (i.e., 10^gkaw) </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">           
swat </td><td style="text-align: left;"> The water solubility (i.e., 10^gswat) </td><td style="text-align: left;"> mg/L </td>
</tr>
<tr>
 <td style="text-align: left;">         
kpl </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">           
kcw </td><td style="text-align: left;"> The cell/tissue to water PC (i.e., 10^gkcw) </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">           
kbsa </td><td style="text-align: left;"> The bovine serum albumin to water PC </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">          
swat_L </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">        
soct_L </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">        
scell_L </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">       
cinit </td><td style="text-align: left;"> Initial concentration </td><td style="text-align: left;"> uM=umol/L </td>
</tr>
<tr>
 <td style="text-align: left;">         
mtot </td><td style="text-align: left;"> Total micromoles </td><td style="text-align: left;"> umol </td>
</tr>
<tr>
 <td style="text-align: left;">          
cwat </td><td style="text-align: left;"> Total concentration in water </td><td style="text-align: left;"> uM=umol/L </td>
</tr>
<tr>
 <td style="text-align: left;">          
cwat_s </td><td style="text-align: left;"> Dissolved concentration in water </td><td style="text-align: left;"> uM=umol/L </td>
</tr>
<tr>
 <td style="text-align: left;">        
csat </td><td style="text-align: left;"> Is the solution saturated (1/0) </td><td style="text-align: left;"> logical </td>
</tr>
<tr>
 <td style="text-align: left;">         
activity </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">      
cair </td><td style="text-align: left;"> Concentration in head space</td><td style="text-align: left;"> uM=umol/L </td>
</tr>
<tr>
 <td style="text-align: left;">          
calb </td><td style="text-align: left;"> Concentration in serum albumin</td><td style="text-align: left;"> uM=umol/L </td>
</tr>
<tr>
 <td style="text-align: left;">          
cslip </td><td style="text-align: left;"> Concentration in serum lipids</td><td style="text-align: left;"> uM=umol/L </td>
</tr>
<tr>
 <td style="text-align: left;">         
cdom </td><td style="text-align: left;"> Concentration in dissolved organic matter</td><td style="text-align: left;"> uM=umol/L </td>
</tr>
<tr>
 <td style="text-align: left;">          
ccells </td><td style="text-align: left;"> Concentration in cells</td><td style="text-align: left;"> uM=umol/L </td>
</tr>
<tr>
 <td style="text-align: left;">        
cplastic </td><td style="text-align: left;"> Concentration in plastic</td><td style="text-align: left;"> uM=umol/m^2 </td>
</tr>
<tr>
 <td style="text-align: left;">      
mwat_s </td><td style="text-align: left;"> Mass dissolved in water </td><td style="text-align: left;"> umols </td>
</tr>
<tr>
 <td style="text-align: left;">        
mair </td><td style="text-align: left;"> Mass in air/head space </td><td style="text-align: left;"> umols </td>
</tr>
<tr>
 <td style="text-align: left;">          
mbsa </td><td style="text-align: left;"> Mass bound to bovine serum albumin </td><td style="text-align: left;"> umols </td>
</tr>
<tr>
 <td style="text-align: left;">          
mslip </td><td style="text-align: left;"> Mass bound to serum lipids </td><td style="text-align: left;"> umols </td>
</tr>
<tr>
 <td style="text-align: left;">        
mdom </td><td style="text-align: left;"> Mass bound to dissolved organic matter </td><td style="text-align: left;"> umols </td>
</tr>
<tr>
 <td style="text-align: left;">          
mcells </td><td style="text-align: left;"> Mass in cells </td><td style="text-align: left;"> umols </td>
</tr>
<tr>
 <td style="text-align: left;">        
mplastic </td><td style="text-align: left;"> Mass bond to plastic </td><td style="text-align: left;"> umols </td>
</tr>
<tr>
 <td style="text-align: left;">      
mprecip </td><td style="text-align: left;"> Mass precipitated out of solution </td><td style="text-align: left;"> umols</td>
</tr>
<tr>
 <td style="text-align: left;">       
xwat_s </td><td style="text-align: left;"> Fraction dissolved in water </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">        
xair </td><td style="text-align: left;"> Fraction in the air </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">          
xbsa </td><td style="text-align: left;"> Fraction bound to bovine serum albumin </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">          
xslip </td><td style="text-align: left;"> Fraction bound to serum lipids </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">         
xdom </td><td style="text-align: left;"> Fraction bound to dissolved organic matter </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">          
xcells </td><td style="text-align: left;"> Fraction within cells </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">        
xplastic </td><td style="text-align: left;"> Fraction bound to plastic </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">     
xprecip </td><td style="text-align: left;"> Fraction precipitated out of solution </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">       
eta_free </td><td style="text-align: left;"> Effective availability ratio </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">      
<strong>cfree.invitro</strong> </td><td style="text-align: left;"> <strong>Free concentration in the in vitro media</strong> (use for Honda1 and Honda2) </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Greg Honda
</p>


<h3>References</h3>

<p>Armitage, J. M.; Wania, F.; Arnot, J. A. Environ. Sci. Technol. 
2014, 48, 9770-9779. https://doi.org/10.1021/es501955g
</p>
<p>Honda GS, Pearce RG, Pham LL, Setzer RW, Wetmore BA, Sipes NS, Gilbert J, Franz B, Thomas RS, Wambaugh JF (2019).
&ldquo;Using the concordance of in vitro and in vivo data to evaluate extrapolation assumptions.&rdquo;
<em>PloS one</em>, <b>14</b>(5), e0217564.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(httk)

# Check to see if we have info on the chemical:
"80-05-7" %in% get_cheminfo()

#We do:
temp &lt;- armitage_eval(casrn.vector = c("80-05-7", "81-81-2"), this.FBSf = 0.1,
this.well_number = 384, nomconc = 10)
print(temp$cfree.invitro)

# Check to see if we have info on the chemical:
"793-24-8" %in% get_cheminfo()

# Since we don't have any info, let's look up phys-chem from dashboard:
cheminfo &lt;- data.frame(
  Compound="6-PPD",
  CASRN="793-24-8",
  DTXSID="DTXSID9025114",
  logP=4.27, 
  logHenry=log10(7.69e-8),
  logWSol=log10(1.58e-4),
  MP=	99.4,
  MW=268.404
  )
  
# Add the information to HTTK's database:
chem.physical_and_invitro.data &lt;- add_chemtable(
 cheminfo,
 current.table=chem.physical_and_invitro.data,
 data.list=list(
 Compound="Compound",
 CAS="CASRN",
  DTXSID="DTXSID",
  MW="MW",
  logP="logP",
  logHenry="logHenry",
  logWSol="logWSol",
  MP="MP"),
  species="Human",
  reference="CompTox Dashboard 31921")

# Run the Armitage et al. (2014) model:
out &lt;- armitage_eval(
  casrn.vector = "793-24-8", 
  this.FBSf = 0.1,
  this.well_number = 384, 
  nomconc = 10)
  
print(out)

</code></pre>

<hr>
<h2 id='armitage_input'>Armitage et al. (2014) Model Inputs from Honda et al. (2019)</h2><span id='topic+armitage_input'></span>

<h3>Description</h3>

<p>Armitage et al. (2014) Model Inputs from Honda et al. (2019)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>armitage_input
</code></pre>


<h3>Format</h3>

<p>A data frame with 53940 rows and 10 variables:
</p>

<dl>
<dt>MP</dt><dd></dd>
<dt>MW</dt><dd></dd>
<dt>casrn</dt><dd></dd>
<dt>compound_name</dt><dd></dd>
<dt>gkaw</dt><dd></dd>
<dt>gkow</dt><dd></dd>
<dt>gswat</dt><dd></dd>
</dl>



<h3>Author(s)</h3>

<p>Greg Honda
</p>


<h3>Source</h3>

<p><a href="https://www.diamondse.info/">https://www.diamondse.info/</a>
</p>


<h3>References</h3>

<p>Armitage, J. M.; Wania, F.; Arnot, J. A. Environ. Sci. Technol.
2014, 48, 9770-9779. dx.doi.org/10.1021/es501955g
Honda GS, Pearce RG, Pham LL, Setzer RW, Wetmore BA, Sipes NS, Gilbert J, Franz B, Thomas RS, Wambaugh JF (2019).
&ldquo;Using the concordance of in vitro and in vivo data to evaluate extrapolation assumptions.&rdquo;
<em>PloS one</em>, <b>14</b>(5), e0217564.
</p>

<hr>
<h2 id='augment.table'>Add a parameter value to the chem.physical_and_invitro.data table</h2><span id='topic+augment.table'></span>

<h3>Description</h3>

<p>This internal function is used by <code><a href="#topic+add_chemtable">add_chemtable</a></code> to add a single 
new parameter to the table of chemical parameters. It should not be typically
used from the command line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>augment.table(
  this.table,
  this.CAS,
  compound.name = NULL,
  this.property,
  value,
  species = NULL,
  reference,
  overwrite = FALSE,
  sig.fig = 4,
  clint.pvalue.overwrite = TRUE,
  allow.na = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="augment.table_+3A_this.table">this.table</code></td>
<td>
<p>Object of class data.frame containing one row per chemical.</p>
</td></tr>
<tr><td><code id="augment.table_+3A_this.cas">this.CAS</code></td>
<td>
<p>The Chemical Abstracts Service registry number (CAS-RN)
correponding to the parameter value</p>
</td></tr>
<tr><td><code id="augment.table_+3A_compound.name">compound.name</code></td>
<td>
<p>A name associated with the chemical (defaults to NULL)</p>
</td></tr>
<tr><td><code id="augment.table_+3A_this.property">this.property</code></td>
<td>
<p>The property being added/modified.</p>
</td></tr>
<tr><td><code id="augment.table_+3A_value">value</code></td>
<td>
<p>The value being assigned to this.property.</p>
</td></tr>
<tr><td><code id="augment.table_+3A_species">species</code></td>
<td>
<p>This is the species for the data in the new table. This may
be omitted if a column in data.list gives the species value for each
chemical or if the data are not species-specific (e.g., MW).</p>
</td></tr>
<tr><td><code id="augment.table_+3A_reference">reference</code></td>
<td>
<p>This is the reference for the data in the new table. This
may be omitted if a column in data.list gives the reference value for each
chemical.</p>
</td></tr>
<tr><td><code id="augment.table_+3A_overwrite">overwrite</code></td>
<td>
<p>If overwrite=TRUE then data in current.table will be
replaced by any data in new.table that is for the same chemical and
property. If overwrite=FALSE (DEFAULT) then new data for the same chemical
and property are ignored.  Funbound.plasma values of 0 (below limit of
detection) are overwritten either way.</p>
</td></tr>
<tr><td><code id="augment.table_+3A_sig.fig">sig.fig</code></td>
<td>
<p>Sets the number of significant figures stored (defaults to 4)</p>
</td></tr>
<tr><td><code id="augment.table_+3A_clint.pvalue.overwrite">clint.pvalue.overwrite</code></td>
<td>
<p>If TRUE then the Cl_int p-value is set to NA 
when the Cl_int value is changed unless a new p-value is provided. (defaults
to TRUE)</p>
</td></tr>
<tr><td><code id="augment.table_+3A_allow.na">allow.na</code></td>
<td>
<p>If TRUE (default is FALSE) then NA values are written to the
table, otherwise they are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>data.frame</code></td>
<td>
<p>A new data.frame containing the data in
current.table augmented by new.table</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>

<hr>
<h2 id='available_rblood2plasma'>Find the best available ratio of the blood to plasma concentration constant.</h2><span id='topic+available_rblood2plasma'></span>

<h3>Description</h3>

<p>This function finds the best available constant ratio of the blood
concentration to the plasma concentration, using 
<code><a href="#topic+get_rblood2plasma">get_rblood2plasma</a></code> and <code><a href="#topic+calc_rblood2plasma">calc_rblood2plasma</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_rblood2plasma(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  adjusted.Funbound.plasma = TRUE,
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="available_rblood2plasma_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the CAS number or the chemical name must be
specified.</p>
</td></tr>
<tr><td><code id="available_rblood2plasma_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="available_rblood2plasma_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="available_rblood2plasma_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="available_rblood2plasma_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Whether or not to use Funbound.plasma
adjustment if calculating Rblood2plasma.</p>
</td></tr>
<tr><td><code id="available_rblood2plasma_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not to display relevant warning messages
to user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Either retrieves a measured blood:plasma concentration ratio from the
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> table or calculates it using the red blood cell
partition coefficient predicted with Schmitt's method
</p>
<p>If available, in vivo data (from <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>) 
for the
given species is returned, substituting the human in vivo value when missing
for other species.  In the absence of in vivo data, the value is calculated
with <code><a href="#topic+calc_rblood2plasma">calc_rblood2plasma</a></code> for the given species. If Funbound.plasma is
unvailable for the given species, the human Funbound.plasma is substituted.
If none of these are available, the mean human Rblood2plasma from
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> is returned.  
details than the description above ~~
</p>


<h3>Value</h3>

<p>The blood to plasma chemical concentration ratio &ndash; measured if available,
calculated if not.
</p>


<h3>Author(s)</h3>

<p>Robert Pearce
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_rblood2plasma">calc_rblood2plasma</a></code>
</p>
<p><code><a href="#topic+get_rblood2plasma">get_rblood2plasma</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
available_rblood2plasma(chem.name="Bisphenol A",adjusted.Funbound.plasma=FALSE)
available_rblood2plasma(chem.name="Bisphenol A",species="Rat")

</code></pre>

<hr>
<h2 id='aylward2014'>Aylward et al. 2014</h2><span id='topic+aylward2014'></span><span id='topic+Aylward2014'></span>

<h3>Description</h3>

<p>Aylward et al. (2014) compiled measurements of the ratio of maternal to fetal 
cord blood chemical concentrations at birth for a range of chemicals with 
environmental routes of exposure, including bromodiphenyl ethers, fluorinated 
compounds, organochlorine pesticides, polyaromatic hydrocarbons, tobacco smoke 
components, and vitamins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aylward2014
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Source</h3>

<p>Kapraun DF, Sfeir M, Pearce RG, Davidson-Fritz SE, Lumen A, Dallmann A, Judson RS, Wambaugh JF (2022).
&ldquo;Evaluation of a rapid, generic human gestational dose model.&rdquo;
<em>Reproductive Toxicology</em>, <b>113</b>, 172&ndash;188.
</p>


<h3>References</h3>

<p>Aylward LL, Hays SM, Kirman CR, Marchitti SA, Kenneke JF, English C, Mattison DR, Becker RA (2014).
&ldquo;Relationships of chemical concentrations in maternal and cord blood: a review of available data.&rdquo;
<em>Journal of Toxicology and Environmental Health, Part B</em>, <b>17</b>(3), 175&ndash;203.
<a href="https://doi.org/10.1080/10937404.2014.884956">doi:10.1080/10937404.2014.884956</a>.
</p>

<hr>
<h2 id='benchmark_httk'>Assess the current performance of httk relative to historical benchmarks</h2><span id='topic+benchmark_httk'></span>

<h3>Description</h3>

<p>The function performs a series of &quot;sanity checks&quot; and predictive performance
benchmarks so that the impact of changes to the data, models, and 
implementation of the R package can be tested. Plots can be generated showing 
how the performance of the current version compares with past releases of
httk.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>benchmark_httk(
  basic.check = TRUE,
  calc_mc_css.check = TRUE,
  in_vivo_stats.check = TRUE,
  tissuepc.check = TRUE,
  suppress.messages = TRUE,
  make.plots = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="benchmark_httk_+3A_basic.check">basic.check</code></td>
<td>
<p>Whether to run the basic checks, including uM and 
mg/L units for <code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code>, <code><a href="#topic+calc_mc_css">calc_mc_css</a></code>, 
and <code><a href="#topic+solve_pbtk">solve_pbtk</a></code> as well as 
the number of chemicals with sufficient data to run the steady_state model 
(defaults to TRUE)</p>
</td></tr>
<tr><td><code id="benchmark_httk_+3A_calc_mc_css.check">calc_mc_css.check</code></td>
<td>
<p>Whether to check the Monte Carlo sample. A 
comparison of the output of <code><a href="#topic+calc_mc_css">calc_mc_css</a></code> to the SimCyp outputs 
reported in 
the Wetmore et al. (2012,2015) papers is performed. A comparison between the
output of <code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code> (no Monte Carlo) to the median of the output of
<code><a href="#topic+calc_mc_css">calc_mc_css</a></code> is also performed. (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="benchmark_httk_+3A_in_vivo_stats.check">in_vivo_stats.check</code></td>
<td>
<p>Whether to compare the outputs of 
<code><a href="#topic+calc_mc_css">calc_mc_css</a></code>
and <code><a href="#topic+calc_tkstats">calc_tkstats</a></code> to in vivo measurements of Css, AUC, and 
Cmax collected
by Wambaugh et al. (2018). (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="benchmark_httk_+3A_tissuepc.check">tissuepc.check</code></td>
<td>
<p>Whether to compare the tissue-specific partition
coefficient predictions from the calibrated Schmitt (2008) model to the 
in vivo data-derived estimates compiled by Pearce et al. (2017). (defaults 
to TRUE)</p>
</td></tr>
<tr><td><code id="benchmark_httk_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not output messages are suppressed
(defaults to TRUE)</p>
</td></tr>
<tr><td><code id="benchmark_httk_+3A_make.plots">make.plots</code></td>
<td>
<p>Whether current benchmarks should be plotted with 
historical performance (defaults to TRUE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Historically some refinements made to one aspect of httk have unintentionally
impacted other aspects. Most notably errors have occasionally been introduced 
with respect to units (v1.9, v2.1.0). This benchmarking tool is intended to
reduce the chance of these errors occurring in the future.
</p>
<p>Past performance was retroactively evaluated by manually installing previous 
versions of 
the package from <a href="https://cran.r-project.org/src/contrib/Archive/httk/">https://cran.r-project.org/src/contrib/Archive/httk/</a> and
then adding the code for <code>benchmark_httk</code> at the command line
interface. 
</p>
<p>The basic tests are important &ndash; if the output units for key functions are wrong, not
much can be right. Past unit errors were linked to an incorrect unit 
conversions made within an individual function. Since the usage of 
<code><a href="#topic+convert_units">convert_units</a></code> became standard throughout httk,
unit problems are hopefully less likely.
</p>
<p>There are two Monte Carlo tests. One compares <code><a href="#topic+calc_mc_css">calc_mc_css</a></code> 95th percentile
steady-state plasma concentrations for a 1 mg/kg/day exposure
against the Css values calculated by SimCyp and reported in Wetmore et al.
(2012,2015). These have gradually diverged as the assumptions for httk have
shifted to better describe non-pharmaceutical, commercial chemicals.
</p>
<p>The in vivo tests are in some ways the most important, as they establish the
overall predictability for httk for Cmax, AUC, and Css. The in vivo 
statistics are currently based on comparisons to the in vivo
data compiled by Wambaugh et al. (2018). We see that when the tissue
partition coefficient calibrations were introduced in v1.6 that the
overall predictability for in vivo endpoints was reduced (increased RMSLE).
If this phenomena continues as new in vivo evaluation data become available,
we may need to revisit whether evaluation against experimentally-derived 
partition coefficients can actually be used for calibration, or just merely
for establishing confidence intervals.
</p>
<p>The partition coefficient tests provide an important check of the httk
implementation of the Schmitt (2008) model for tissue:plasma equilibrium 
distribution. These predictions heavily rely on accurate description of 
tissue composition and the ability to predict the ionization state of the
compounds being modeled.
</p>


<h3>Value</h3>

<p>named list, whose elements depend on the selected checks
</p>

<table>
<tr>
 <td style="text-align: left;">
  basic </td><td style="text-align: left;"> A list with four metrics:
 N.steadystate -- Number of chemicals with sufficient data for steady-state IVIVE
    calc_analytic.units -- Ratio of mg/L to uM * 1000 / molecular weight -- should be 1
    calc_mc.units -- Ratio should be 1
    solve_pbtk.units -- Ratio should be 1 </td>
</tr>
<tr>
 <td style="text-align: left;">

  calc_mc_css </td><td style="text-align: left;"> A list with four metrics:
RMSLE.Wetmore -- Root mean squared log10 error (RMSLE) in predicted Css between literature values (SimCyp, Wetmore et al. 2012,2015) and <code><a href="#topic+calc_mc_css">calc_mc_css</a></code>
N.Wetmore -- Number of chemicals in Wetmore comparison
RMSLE.noMC -- RMSLE between <code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code> and <code><a href="#topic+calc_mc_css">calc_mc_css</a></code>
    N.noMC -- Number of chemicals in noMC comparison </td>
</tr>
<tr>
 <td style="text-align: left;">

  in_vivo_stats </td><td style="text-align: left;"> A list with two metrics:
RMSLE.InVivoCss -- RMSLE between the predictions of <code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code> and in vivo estimates of Css
N.InVivoCss -- Number of chemicals in comparison </td>
</tr>
<tr>
 <td style="text-align: left;">                    
  
  units.plot </td><td style="text-align: left;"> A ggplot2 figure showing units tests of various functions. 
Output is generated for mg/L and uM, and then the ratio mg/L/uM*1000/MW is
calculated. If the units are correct the ratio should be 1 (within the 
precision of the functions -- usually four significant figures). </td>
</tr>
<tr>
 <td style="text-align: left;">

  invivo.rmsle.plot </td><td style="text-align: left;"> A ggplot2 figure comparing
model predictions to in vivo measured values. 
Output generated is the root mean square log10 error for parameters estimated
by the package. </td>
</tr>
<tr>
 <td style="text-align: left;">

  model.rmsle.plot </td><td style="text-align: left;"> A ggplot2 figure comparing various functions values
against values predicted by other models (chiefly SimCyp predictions from
Wetmore et al. 2012 and 2015. 
Output generated is the root mean square log10 error for parameters estimated
by the package. </td>
</tr>
<tr>
 <td style="text-align: left;">

  count.plot </td><td style="text-align: left;"> A ggplot2 figure showing count of chemicals of various functions. 
Output generated is a count of the chemicals available for the each of the
parameters estimated by and used for benchmarking the package. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Davidson-Fritz SE, Evans MV, Chang X, Breen M, Honda GS, Kenyon E, Linakis MW, Meade A, Pearce RG, Purucker T, Ring CL, Sfeir MA, Setzer RW, Sluka JP, Vitense K, Devito MJ, Wambaugh JF (2023).
&ldquo;Transparent and Evaluated Toxicokinetic Models for Bioinformatics and Public Health Risk Assessment.&rdquo;
Unpublished.
</p>

<hr>
<h2 id='blood_mass_correct'>Find average blood masses by age.</h2><span id='topic+blood_mass_correct'></span>

<h3>Description</h3>

<p>If blood mass from <code><a href="#topic+blood_weight">blood_weight</a></code> is negative or very small,
then just default to the mean blood mass by age. (Geigy Scientific Tables,
7th ed.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blood_mass_correct(blood_mass, age_months, age_years, gender, weight)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blood_mass_correct_+3A_blood_mass">blood_mass</code></td>
<td>
<p>A vector of blood masses in kg to be replaced with
averages.</p>
</td></tr>
<tr><td><code id="blood_mass_correct_+3A_age_months">age_months</code></td>
<td>
<p>A vector of ages in months.</p>
</td></tr>
<tr><td><code id="blood_mass_correct_+3A_age_years">age_years</code></td>
<td>
<p>A vector of ages in years.</p>
</td></tr>
<tr><td><code id="blood_mass_correct_+3A_gender">gender</code></td>
<td>
<p>A vector of genders (either 'Male' or 'Female').</p>
</td></tr>
<tr><td><code id="blood_mass_correct_+3A_weight">weight</code></td>
<td>
<p>A vector of body weights in kg.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of blood masses in kg.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Geigy Pharmaceuticals, &quot;Scientific Tables&quot;, 7th Edition, 
John Wiley and Sons (1970)
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='blood_weight'>Predict blood mass.</h2><span id='topic+blood_weight'></span>

<h3>Description</h3>

<p>Predict blood mass based on body surface area and gender, using equations
from Bosgra et al. 2012
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blood_weight(BSA, gender)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blood_weight_+3A_bsa">BSA</code></td>
<td>
<p>Body surface area in m^2. May be a vector.</p>
</td></tr>
<tr><td><code id="blood_weight_+3A_gender">gender</code></td>
<td>
<p>Either 'Male' or 'Female'. May be a vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of blood masses in kg the same length as <code>BSA</code> and
<code>gender</code>.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Bosgra, Sieto, et al. &quot;An improved model to predict 
physiologically based model parameters and their inter-individual variability 
from anthropometry.&quot; Critical reviews in toxicology 42.9 (2012): 751-767.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='bmiage'>CDC BMI-for-age charts</h2><span id='topic+bmiage'></span>

<h3>Description</h3>

<p>Charts giving the BMI-for-age percentiles for boys and girls ages 2-18
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bmiage
</code></pre>


<h3>Format</h3>

<p>A data.table with 434 rows and 5 variables: </p>

<dl>
<dt>Sex</dt><dd><p>Female or Male</p>
</dd> <dt>Agemos</dt><dd><p>Age in months</p>
</dd> <dt>P5</dt><dd><p>The 5th
percentile BMI for the corresponding sex and age</p>
</dd> <dt>P85</dt><dd><p>The 85th
percentile BMI for the corresponding sex and age</p>
</dd> <dt>P95</dt><dd><p>The 95th
percentile BMI for the corresponding sex and age</p>
</dd> </dl>



<h3>Details</h3>

<p>For children ages 2 to 18, weight class depends on the BMI-for-age percentile.
</p>

<dl>
<dt>Underweight</dt><dd><p>&lt;5th percentile</p>
</dd>
<dt>Normal weight</dt><dd><p>5th-85th percentile</p>
</dd>
<dt>Overweight</dt><dd><p>85th-95th percentile</p>
</dd>
<dt>Obese</dt><dd><p>&gt;=95th percentile</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>Source</h3>

<p><a href="https://www.cdc.gov/growthcharts/data/zscore/bmiagerev.csv">https://www.cdc.gov/growthcharts/data/zscore/bmiagerev.csv</a>
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='body_surface_area'>Predict body surface area.</h2><span id='topic+body_surface_area'></span>

<h3>Description</h3>

<p>Predict body surface area from weight, height, and age, using Mosteller's
formula for age&gt;18 and Haycock's formula for age&lt;18
</p>


<h3>Usage</h3>

<pre><code class='language-R'>body_surface_area(BW, H, age_years)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="body_surface_area_+3A_bw">BW</code></td>
<td>
<p>A vector of body weights in kg.</p>
</td></tr>
<tr><td><code id="body_surface_area_+3A_h">H</code></td>
<td>
<p>A vector of heights in cm.</p>
</td></tr>
<tr><td><code id="body_surface_area_+3A_age_years">age_years</code></td>
<td>
<p>A vector of ages in years.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of body surface areas in cm^2.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Mosteller, R. D. &quot;Simplified calculation of body surface area.&quot; 
N Engl J Med 317 (1987): 1098..
</p>
<p>Haycock, George B., George J. Schwartz, and David H. Wisotsky. &quot;Geometric 
method for measuring body surface area: a height-weight formula validated in 
infants, children, and adults.&quot; The Journal of pediatrics 93.1 (1978): 62-66.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='bone_mass_age'>Predict bone mass</h2><span id='topic+bone_mass_age'></span>

<h3>Description</h3>

<p>Predict bone mass from age_years, height, weight, gender, using logistic
equations fit to data from Baxter-Jones et al. 2011, or for infants &lt; 1
year, using equation from Koo et al. 2000 (See Price et al. 2003)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bone_mass_age(age_years, age_months, height, weight, gender)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bone_mass_age_+3A_age_years">age_years</code></td>
<td>
<p>Vector of ages in years.</p>
</td></tr>
<tr><td><code id="bone_mass_age_+3A_age_months">age_months</code></td>
<td>
<p>Vector of ages in months.</p>
</td></tr>
<tr><td><code id="bone_mass_age_+3A_height">height</code></td>
<td>
<p>Vector of heights in cm.</p>
</td></tr>
<tr><td><code id="bone_mass_age_+3A_weight">weight</code></td>
<td>
<p>Vector of body weights in kg.</p>
</td></tr>
<tr><td><code id="bone_mass_age_+3A_gender">gender</code></td>
<td>
<p>Vector of genders, either 'Male' or 'Female'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of bone masses.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Baxter-Jones, Adam DG, et al. &quot;Bone mineral accrual from 8 to 30 years of age: 
an estimation of peak bone mass.&quot; Journal of Bone and Mineral Research 26.8 
(2011): 1729-1739.
</p>
<p>Koo, Winston WK, and Elaine M. Hockman. &quot;Physiologic predictors of lumbar 
spine bone mass in neonates.&quot; Pediatric research 48.4 (2000): 485-489.
</p>
<p>Price, Paul S., et al. &quot;Modeling interindividual variation in physiological 
factors used in PBPK models of humans.&quot; Critical reviews in toxicology 33.5 
(2003): 469-503.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='brain_mass'>Predict brain mass.</h2><span id='topic+brain_mass'></span>

<h3>Description</h3>

<p>Predict brain mass from gender and age.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brain_mass(gender, age_years)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="brain_mass_+3A_gender">gender</code></td>
<td>
<p>Vector of genders, either 'Male' or 'Female'</p>
</td></tr>
<tr><td><code id="brain_mass_+3A_age_years">age_years</code></td>
<td>
<p>Vector of ages in years.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of brain masses in kg.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='calc_analytic_css'>Calculate the analytic steady state plasma concentration.</h2><span id='topic+calc_analytic_css'></span>

<h3>Description</h3>

<p>This function calculates the analytic steady state plasma or venous blood 
concentrations as a result of infusion dosing for the three compartment and 
multiple compartment PBTK models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_analytic_css(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "human",
  daily.dose = 1,
  route = "oral",
  exp.conc = 1,
  period = 24,
  exp.duration = 24,
  output.units = "uM",
  model = "pbtk",
  concentration = "plasma",
  suppress.messages = FALSE,
  tissue = NULL,
  restrictive.clearance = TRUE,
  bioactive.free.invivo = FALSE,
  IVIVE = NULL,
  Caco2.options = list(),
  parameterize.args = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_analytic_css_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must 
be specified.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must 
be specified.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_pbtk (for model = 
'pbtk'), parameterize_3comp (for model = '3compartment), 
parameterize_1comp(for model = '1compartment') or parameterize_steadystate 
(for model = '3compartmentss'), overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose, mg/kg BW.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_route">route</code></td>
<td>
<p>Route of exposure (either &quot;oral&quot;, &quot;iv&quot;, or &quot;inhalation&quot;
default &quot;oral&quot;).</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_exp.conc">exp.conc</code></td>
<td>
<p>Specified inhalation exposure concentration for use in assembling
'forcings' data series argument for integrator. Defaults to uM/L</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_period">period</code></td>
<td>
<p>For use in assembling forcing function data series 'forcings'
argument, specified in hours</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_exp.duration">exp.duration</code></td>
<td>
<p>For use in assembling forcing function data 
series 'forcings' argument, specified in hours</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_output.units">output.units</code></td>
<td>
<p>Units for returned concentrations, defaults to uM 
(specify units = &quot;uM&quot;) but can also be mg/L.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_model">model</code></td>
<td>
<p>Model used in calculation,'gas_pbtk' for the gas pbtk model, 
'pbtk' for the multiple compartment model,
'3compartment' for the three compartment model, '3compartmentss' for 
the three compartment steady state model, and '1compartment' for one 
compartment model.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_concentration">concentration</code></td>
<td>
<p>Desired concentration type: 'blood','tissue', or default 
'plasma'. In the case that the concentration is for plasma, selecting &quot;blood&quot;
will use the blood:plasma ratio to estimate blood concentration. In the case
that the argument 'tissue' specifies a particular tissue of the body, 
concentration defaults to 'tissue' &ndash; that is, the concentration in the 
If cocentration is set to 'blood' or 'plasma' and 'tissue' specifies a
specific tissue then the value returned is for the plasma or blood in that
specific tissue.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_tissue">tissue</code></td>
<td>
<p>Desired steady state tissue concentration. Default is of NULL
typically gives whole body plasma concentration.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>If TRUE (default), then only the fraction of
chemical not bound to protein is available for metabolism in the liver. If 
FALSE, then all chemical in the liver is metabolized (faster metabolism due
to rapid off-binding).</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_bioactive.free.invivo">bioactive.free.invivo</code></td>
<td>
<p>If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with tissue = NULL in current implementation.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_ivive">IVIVE</code></td>
<td>
<p>Honda et al. (2019) identified four plausible sets of 
assumptions for <em>in vitro-in vivo</em> extrapolation (IVIVE) assumptions. 
Argument may be set to &quot;Honda1&quot; through &quot;Honda4&quot;. If used, this function 
overwrites the tissue, restrictive.clearance, and bioactive.free.invivo arguments. 
See Details below for more information.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_parameterize.args">parameterize.args</code></td>
<td>
<p>List of arguments passed to model's associated
parameterization function, including default.to.human, 
adjusted.Funbound.plasma, regression, and minimum.Funbound.plasma. The 
default.to.human argument substitutes missing animal values with human values
if true, adjusted.Funbound.plasma returns adjusted Funbound.plasma when set 
to TRUE along with parition coefficients calculated with this value, 
regression indicates whether or not to use the regressions in calculating
partition coefficients, and minimum.Funbound.plasma is the value to which
Monte Carlo draws less than this value are set (default is 0.0001 &ndash; half
the lowest measured Fup in our dataset).</p>
</td></tr>
<tr><td><code id="calc_analytic_css_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to parameterize function if 
parameters is NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Concentrations are calculated for the specifed model with constant 
oral infusion dosing.  All tissues other than gut, liver, and lung are the 
product of the steady state plasma concentration and the tissue to plasma 
partition coefficient. 
</p>
<p>Only four sets of IVIVE assumptions that performed well in Honda et al. 
(2019) are currently included in <code><a href="#topic+honda.ivive">honda.ivive</a></code>:
&quot;Honda1&quot; through &quot;Honda4&quot;. The use of max (peak) 
concentration can not be currently be calculated with <code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code>. 
The httk default settings correspond to &quot;Honda3&quot;:
</p>

<table>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: right;"> <em>In Vivo</em> Conc. </td><td style="text-align: right;"> Metabolic Clearance </td><td style="text-align: right;"> Bioactive Chemical Conc. <em>In Vivo</em> </td><td style="text-align: right;"> TK Statistic Used* </td><td style="text-align: right;"> Bioactive Chemical Conc. <em>In Vitro</em> </td>
</tr>
<tr>
 <td style="text-align: left;">
Honda1 </td><td style="text-align: right;"> Veinous (Plasma) </td><td style="text-align: right;"> Restrictive </td><td style="text-align: right;"> Free </td><td style="text-align: right;"> Mean Conc. In Vivo </td><td style="text-align: right;"> Free Conc. In Vitro </td>
</tr>
<tr>
 <td style="text-align: left;">
Honda2 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;"> Restrictive </td><td style="text-align: right;"> Free </td><td style="text-align: right;"> Mean Conc. In Vivo </td><td style="text-align: right;"> Nominal Conc. In Vitro </td>
</tr>
<tr>
 <td style="text-align: left;">
Honda3 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;"> Restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Mean Conc. In Vivo </td><td style="text-align: right;"> Nominal Conc. In Vitro </td>
</tr>
<tr>
 <td style="text-align: left;">
Honda4 </td><td style="text-align: right;"> Target Tissue </td><td style="text-align: right;"> Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Mean Conc. In Vivo </td><td style="text-align: right;"> Nominal Conc. In Vitro </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>&quot;Honda1&quot; uses plasma concentration, restrictive clearance, and treats the 
unbound invivo concentration as bioactive. For IVIVE, any input nominal 
concentration in vitro should be converted to cfree.invitro using 
<code><a href="#topic+armitage_eval">armitage_eval</a></code>, otherwise performance will be the same as 
&quot;Honda2&quot;.
</p>


<h3>Value</h3>

<p>Steady state plasma concentration in specified units
</p>


<h3>Author(s)</h3>

<p>Robert Pearce, John Wambaugh, Greg Honda, Miyuki Breen
</p>


<h3>References</h3>

<p>Honda GS, Pearce RG, Pham LL, Setzer RW, Wetmore BA, Sipes NS, Gilbert J, Franz B, Thomas RS, Wambaugh JF (2019).
&ldquo;Using the concordance of in vitro and in vivo data to evaluate extrapolation assumptions.&rdquo;
<em>PloS one</em>, <b>14</b>(5), e0217564.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_css">calc_css</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>calc_analytic_css(chem.name='Bisphenol-A',output.units='mg/L',
                 model='3compartment',concentration='blood')

calc_analytic_css(chem.name='Bisphenol-A',tissue='liver',species='rabbit',
                 parameterize.args = list(
                                default.to.human=TRUE,
                                adjusted.Funbound.plasma=TRUE,
                                regression=TRUE,
                                minimum.Funbound.plasma=1e-4),daily.dose=2)

calc_analytic_css(chem.name="bisphenol a",model="1compartment")

calc_analytic_css(chem.cas="80-05-7",model="3compartmentss")

params &lt;- parameterize_pbtk(chem.cas="80-05-7") 

calc_analytic_css(parameters=params,model="pbtk")

# Try various chemicals with differing parameter sources/issues:
calc_analytic_css(chem.name="Betaxolol")
calc_analytic_css(chem.name="Tacrine",model="pbtk")
calc_analytic_css(chem.name="Dicofol",model="1compartment")
calc_analytic_css(chem.name="Diflubenzuron",model="3compartment")
calc_analytic_css(chem.name="Theobromine",model="3compartmentss")

</code></pre>

<hr>
<h2 id='calc_analytic_css_1comp'>Calculate the analytic steady state concentration for the one compartment model.</h2><span id='topic+calc_analytic_css_1comp'></span>

<h3>Description</h3>

<p>This function calculates the analytic steady state plasma or venous blood 
concentrations as a result of infusion dosing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_analytic_css_1comp(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  hourly.dose = 1/24,
  concentration = "plasma",
  suppress.messages = FALSE,
  recalc.blood2plasma = FALSE,
  tissue = NULL,
  restrictive.clearance = TRUE,
  bioactive.free.invivo = FALSE,
  Caco2.options = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_analytic_css_1comp_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must 
be specified.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must 
be specified.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_pbtk (for model = 
'pbtk'), parameterize_3comp (for model = '3compartment), 
parameterize_1comp(for model = '1compartment') or parameterize_steadystate 
(for model = '3compartmentss'), overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_hourly.dose">hourly.dose</code></td>
<td>
<p>Hourly dose rate mg/kg BW/h.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_concentration">concentration</code></td>
<td>
<p>Desired concentration type, 'blood' or default 'plasma'.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_recalc.blood2plasma">recalc.blood2plasma</code></td>
<td>
<p>Recalculates the ratio of the amount of chemical 
in the blood to plasma using the input parameters. Use this if you have 
altered hematocrit, Funbound.plasma, or Krbc2pu.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_tissue">tissue</code></td>
<td>
<p>Desired tissue conentration (defaults to whole body 
concentration.)</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>If TRUE (default), then only the fraction of
chemical not bound to protein is available for metabolism in the liver. If 
FALSE, then all chemical in the liver is metabolized (faster metabolism due
to rapid off-binding).</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_bioactive.free.invivo">bioactive.free.invivo</code></td>
<td>
<p>If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with tissue = NULL in current implementation.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_1comp_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to parameterize function if 
parameters is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Steady state plasma concentration in mg/L units
</p>


<h3>Author(s)</h3>

<p>Robert Pearce and John Wambaugh
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code>
</p>
<p><code><a href="#topic+parameterize_1comp">parameterize_1comp</a></code>
</p>

<hr>
<h2 id='calc_analytic_css_3comp'>Calculate the analytic steady state concentration for model 3comp</h2><span id='topic+calc_analytic_css_3comp'></span>

<h3>Description</h3>

<p>This function calculates the analytic steady state plasma or venous blood 
concentrations as a result of infusion dosing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_analytic_css_3comp(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  hourly.dose = 1/24,
  concentration = "plasma",
  suppress.messages = FALSE,
  recalc.blood2plasma = FALSE,
  tissue = NULL,
  restrictive.clearance = TRUE,
  bioactive.free.invivo = FALSE,
  Caco2.options = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_analytic_css_3comp_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must 
be specified.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must 
be specified.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_pbtk (for model = 
'pbtk'), parameterize_3comp (for model = '3compartment), 
parameterize_1comp(for model = '1compartment') or parameterize_steadystate 
(for model = '3compartmentss'), overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_hourly.dose">hourly.dose</code></td>
<td>
<p>Hourly dose rate mg/kg BW/h.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_concentration">concentration</code></td>
<td>
<p>Desired concentration type, 'blood' or default 'plasma'.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_recalc.blood2plasma">recalc.blood2plasma</code></td>
<td>
<p>Recalculates the ratio of the amount of chemical 
in the blood to plasma using the input parameters. Use this if you have 
'altered hematocrit, Funbound.plasma, or Krbc2pu.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_tissue">tissue</code></td>
<td>
<p>Desired tissue conentration (defaults to whole body 
concentration.)</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>If TRUE (default), then only the fraction of
chemical not bound to protein is available for metabolism in the liver. If 
FALSE, then all chemical in the liver is metabolized (faster metabolism due
to rapid off-binding).</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_bioactive.free.invivo">bioactive.free.invivo</code></td>
<td>
<p>If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with <code>tissue = NULL</code> in current implementation.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3comp_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to parameterize function if 
parameters is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Steady state plasma concentration in mg/L units
</p>


<h3>Author(s)</h3>

<p>Robert Pearce and John Wambaugh
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code>
</p>
<p><code><a href="#topic+parameterize_3comp">parameterize_3comp</a></code>
</p>

<hr>
<h2 id='calc_analytic_css_3compss'>Calculate the analytic steady state concentration for the three compartment
steady-state model</h2><span id='topic+calc_analytic_css_3compss'></span>

<h3>Description</h3>

<p>This function calculates the analytic steady state plasma or venous blood 
concentrations as a result of infusion dosing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_analytic_css_3compss(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  hourly.dose = 1/24,
  concentration = "plasma",
  suppress.messages = FALSE,
  recalc.blood2plasma = FALSE,
  tissue = NULL,
  restrictive.clearance = TRUE,
  bioactive.free.invivo = FALSE,
  Caco2.options = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_analytic_css_3compss_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must 
be specified.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must 
be specified.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_pbtk (for model = 
'pbtk'), parameterize_3comp (for model = '3compartment), 
parameterize_1comp(for model = '1compartment') or parameterize_steadystate 
(for model = '3compartmentss'), overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_hourly.dose">hourly.dose</code></td>
<td>
<p>Hourly dose rate mg/kg BW/h.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_concentration">concentration</code></td>
<td>
<p>Desired concentration type, 'blood' or default 'plasma'.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_recalc.blood2plasma">recalc.blood2plasma</code></td>
<td>
<p>Recalculates the ratio of the amount of chemical 
in the blood to plasma using the input parameters. Use this if you have 
'altered hematocrit, Funbound.plasma, or Krbc2pu.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_tissue">tissue</code></td>
<td>
<p>Desired tissue concentration (defaults to whole body 
concentration.)</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>If TRUE (default), then only the fraction of
chemical not bound to protein is available for metabolism in the liver. If 
FALSE, then all chemical in the liver is metabolized (faster metabolism due
to rapid off-binding).</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_bioactive.free.invivo">bioactive.free.invivo</code></td>
<td>
<p>If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with tissue = NULL in current implementation.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_3compss_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to parameterize function if 
parameters is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Steady state plasma concentration in mg/L units
</p>


<h3>Author(s)</h3>

<p>Robert Pearce and John Wambaugh
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code>
</p>
<p><code><a href="#topic+parameterize_steadystate">parameterize_steadystate</a></code>
</p>

<hr>
<h2 id='calc_analytic_css_pbtk'>Calculate the analytic steady state plasma concentration for model pbtk.</h2><span id='topic+calc_analytic_css_pbtk'></span>

<h3>Description</h3>

<p>This function calculates the analytic steady state concentration (mg/L) as a result
of oral infusion dosing. Concentrations are returned for plasma by default, but various
tissues or blood concentrations can also be given as specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_analytic_css_pbtk(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  hourly.dose = 1/24,
  concentration = "plasma",
  suppress.messages = FALSE,
  recalc.blood2plasma = FALSE,
  tissue = NULL,
  restrictive.clearance = TRUE,
  bioactive.free.invivo = FALSE,
  Caco2.options = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_analytic_css_pbtk_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must 
be specified.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must 
be specified.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_pbtk (for model = 
'pbtk'), parameterize_3comp (for model = '3compartment), 
parameterize_1comp(for model = '1compartment') or parameterize_steadystate 
(for model = '3compartmentss'), overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_hourly.dose">hourly.dose</code></td>
<td>
<p>Hourly dose rate mg/kg BW/h.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_concentration">concentration</code></td>
<td>
<p>Desired concentration type, 'blood', 'tissue', or default 'plasma'.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_recalc.blood2plasma">recalc.blood2plasma</code></td>
<td>
<p>Recalculates the ratio of the amount of chemical 
in the blood to plasma using the input parameters. Use this if you have 
'altered hematocrit, Funbound.plasma, or Krbc2pu.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_tissue">tissue</code></td>
<td>
<p>Desired tissue conentration (defaults to whole body 
concentration.)</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>If TRUE (default), then only the fraction of
chemical not bound to protein is available for metabolism in the liver. If 
FALSE, then all chemical in the liver is metabolized (faster metabolism due
to rapid off-binding).</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_bioactive.free.invivo">bioactive.free.invivo</code></td>
<td>
<p>If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with tissue = NULL in current implementation.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
<tr><td><code id="calc_analytic_css_pbtk_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to parameterize function if 
parameters is NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The PBTK model (Pearce et al., 2017) predicts the amount of chemical in
various tissues of the body. A system of oridinary
differential equations describes how the amounts in each tissue change as 
a function of time. The analytic steady-state equation was found by 
algebraically solving for the tissue concentrations that result in each
equation being zero &ndash; thus determining the concentration at which there is no change
over time as the result of a fixed infusion dose rate. 
</p>
<p>The analytical solution is:
</p>
<p style="text-align: center;"><code class="reqn">C^{ss}_{ven} = \frac{dose rate * \frac{Q_{liver} + Q_{gut}}{\frac{f_{up}}{R_{b:p}}*Cl_{metabolism} + (Q_{liver}+Q_{gut})}}{Q_{cardiac} - \frac{(Q_{liver} + Q_{gut})^2}{\frac{f_{up}}{R_{b:p}}*Cl_{metabolism} + (Q_{liver}+Q_{gut})} - \frac{(Q_{kidney})^2}{\frac{f_{up}}{R_{b:p}}*Q_{GFR}+Q_{kideny}}-Q_{rest}}</code>
</p>

<p style="text-align: center;"><code class="reqn">C^{ss}_{plasma} = \frac{C^{ss}_{ven}}{R_{b:p}}</code>
</p>

<p style="text-align: center;"><code class="reqn">C^{ss}_{tissue} = \frac{K_{tissue:fuplasma}*f_{up}}{R_{b:p}}*C^{ss}_{ven}</code>
</p>

<p>where Q_cardiac is the cardiace output, Q_gfr is the glomerular filtration
rate in the kidney, other Q's indicate blood flows to various tissues, 
Cl_metabolism is the chemical-specific whole liver metabolism clearance,
f_up is the chemical-specific fraction unbound n plasma, R_b2p is the 
chemical specific ratio of concentrations in blood:plasma, K_tissue2fuplasma
is the chemical- and tissue-specufic equilibrium partition coefficient
and dose rate has  units of mg/kg/day.
</p>


<h3>Value</h3>

<p>Steady state plasma concentration in mg/L units
</p>


<h3>Author(s)</h3>

<p>Robert Pearce and John Wambaugh
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code>
</p>
<p><code><a href="#topic+parameterize_pbtk">parameterize_pbtk</a></code>
</p>

<hr>
<h2 id='calc_css'>Find the steady state concentration and the day it is reached.</h2><span id='topic+calc_css'></span>

<h3>Description</h3>

<p>This function finds the day a chemical comes within the specified range of
the analytical steady state venous blood or plasma concentration(from
calc_analytic_css) for the multiple compartment, three compartment, and one
compartment models, the fraction of the true steady state value reached on
that day, the maximum concentration, and the average concentration at the
end of the simulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_css(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  f = 0.01,
  daily.dose = 1,
  doses.per.day = 3,
  days = 21,
  output.units = "uM",
  suppress.messages = FALSE,
  tissue = NULL,
  model = "pbtk",
  default.to.human = FALSE,
  f.change = 1e-05,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  well.stirred.correction = TRUE,
  restrictive.clearance = TRUE,
  dosing = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_css_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or parameters must be
specified.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or parameters must be
specified.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_css_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_pbtk function,
overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_css_+3A_f">f</code></td>
<td>
<p>Fractional distance from the final steady state concentration that
the average concentration must come within to be considered at steady state.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose, mg/kg BW.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_doses.per.day">doses.per.day</code></td>
<td>
<p>Number of doses per day.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_days">days</code></td>
<td>
<p>Initial number of days to run simulation that is multiplied on
each iteration.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_output.units">output.units</code></td>
<td>
<p>Units for returned concentrations, defaults to uM
(specify units = &quot;uM&quot;) but can also be mg/L.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not to suppress messages.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_tissue">tissue</code></td>
<td>
<p>Desired tissue concentration (default value is NULL, will
depend on model &ndash; see <code>steady.state.compartment</code> in model.info file for
further details.)</p>
</td></tr>
<tr><td><code id="calc_css_+3A_model">model</code></td>
<td>
<p>Model used in calculation, 'pbtk' for the multiple compartment
model,'3compartment' for the three compartment model, and '1compartment' for
the one compartment model.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).</p>
</td></tr>
<tr><td><code id="calc_css_+3A_f.change">f.change</code></td>
<td>
<p>Fractional change of daily steady state concentration
reached to stop calculating.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_well.stirred.correction">well.stirred.correction</code></td>
<td>
<p>Uses correction in calculation of hepatic
clearance for well-stirred model if TRUE for model 1compartment elimination
rate.  This assumes clearance relative to amount unbound in whole blood
instead of plasma, but converted to use with plasma concentration.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to
1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="calc_css_+3A_dosing">dosing</code></td>
<td>
<p>The dosing object for more complicated scenarios. Defaults to
repeated <code>daily.dose</code> spread out over <code>doses.per.day</code></p>
</td></tr>
<tr><td><code id="calc_css_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to model solver (default of
<code><a href="#topic+solve_pbtk">solve_pbtk</a></code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>frac</code></td>
<td>
<p>Ratio of the mean concentration on the day steady state
is reached (baed on doses.per.day) to the analytical Css (based on infusion
dosing).</p>
</td></tr> <tr><td><code>max</code></td>
<td>
<p>The maximum concentration of the simulation.</p>
</td></tr>
<tr><td><code>avg</code></td>
<td>
<p>The average concentration on the final day of the simulation.</p>
</td></tr>
<tr><td><code>the.day</code></td>
<td>
<p>The day the average concentration comes within 100 * p
percent of the true steady state concentration.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Pearce, John Wambaugh
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
calc_css(chem.name='Bisphenol-A',doses.per.day=5,f=.001,output.units='mg/L')

parms &lt;- parameterize_3comp(chem.name='Bisphenol-A')
parms$Funbound.plasma &lt;- .07
calc_css(chem.name='Bisphenol-A',parameters=parms,model='3compartment')

out &lt;- solve_pbtk(chem.name = "Bisphenol A",
  days = 50,                                   
  daily.dose=1,
  doses.per.day = 3)
plot.data &lt;- as.data.frame(out)

css &lt;- calc_analytic_css(chem.name = "Bisphenol A")
library("ggplot2")
c.vs.t &lt;- ggplot(plot.data,aes(time, Cplasma)) + geom_line() +
geom_hline(yintercept = css) + ylab("Plasma Concentration (uM)") +
xlab("Day") + theme(axis.text = element_text(size = 16), axis.title =
element_text(size = 16), plot.title = element_text(size = 17)) +
ggtitle("Bisphenol A")

print(c.vs.t)

</code></pre>

<hr>
<h2 id='calc_dow'>Calculate the distribution coefficient</h2><span id='topic+calc_dow'></span>

<h3>Description</h3>

<p>This function estimates the ratio of the equilibrium concentrations of
a compound in octanol and water, taking into account the charge of the
compound. Given the pH, we assume the neutral (uncharged) fraction of
compound partitions according to the hydrophobicity 
(P<sub>ow</sub>). We assume that
only a fraction alpha (defaults to 0.001 &ndash; Schmitt (2008)) of the charged
compound partitions into lipid (octanol):
D<sub>ow</sub> = P<sub>ow</sub>*(F<sub>neutral</sub> + alpha*F<sub>charged</sub>)
Fractions charged are calculated
according to hydrogen ionization equilibria (pKa_Donor, pKa_Accept) using
<code><a href="#topic+calc_ionization">calc_ionization</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_dow(
  Pow = NULL,
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  pH = NULL,
  pKa_Donor = NULL,
  pKa_Accept = NULL,
  fraction_charged = NULL,
  alpha = 0.001
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_dow_+3A_pow">Pow</code></td>
<td>
<p>Octanol:water partition coefficient (ratio of concentrations)</p>
</td></tr>
<tr><td><code id="calc_dow_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="calc_dow_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="calc_dow_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_dow_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from a parameterize_MODEL function,
overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_dow_+3A_ph">pH</code></td>
<td>
<p>pH where ionization is evaluated.</p>
</td></tr>
<tr><td><code id="calc_dow_+3A_pka_donor">pKa_Donor</code></td>
<td>
<p>Compound H dissociation equilibirum constant(s).
Overwrites chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_dow_+3A_pka_accept">pKa_Accept</code></td>
<td>
<p>Compound H association equilibirum constant(s).
Overwrites chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_dow_+3A_fraction_charged">fraction_charged</code></td>
<td>
<p>Fraction of chemical charged at the given pH</p>
</td></tr>
<tr><td><code id="calc_dow_+3A_alpha">alpha</code></td>
<td>
<p>Ratio of Distribution coefficient D of totally charged species and that of the neutral form</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Distribution coefficient (numeric)
</p>


<h3>Author(s)</h3>

<p>Robert Pearce and John Wambaugh
</p>


<h3>References</h3>

<p>Schmitt, Walter. &quot;General approach for the calculation of tissue to plasma 
partition coefficients.&quot; Toxicology in vitro 22.2 (2008): 457-467.
</p>
<p>Pearce, Robert G., et al. &quot;Evaluation and calibration of
high-throughput predictions of chemical distribution to tissues.&quot; Journal of
Pharmacokinetics and Pharmacodynamics 44.6 (2017): 549-565.
</p>
<p>Strope, Cory L., et al. &quot;High-throughput in-silico prediction of ionization 
equilibria for pharmacokinetic modeling.&quot; Science of The Total Environment 
615 (2018): 150-160.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_ionization">calc_ionization</a></code>
</p>

<hr>
<h2 id='calc_elimination_rate'>Calculate the elimination rate for a one compartment model</h2><span id='topic+calc_elimination_rate'></span>

<h3>Description</h3>

<p>This function calculates an elimination rate from the three compartment
steady state model where elimination is entirely due to metablism by the
liver and glomerular filtration in the kidneys.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_elimination_rate(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  suppress.messages = TRUE,
  default.to.human = FALSE,
  restrictive.clearance = TRUE,
  adjusted.Funbound.plasma = TRUE,
  adjusted.Clint = TRUE,
  regression = TRUE,
  well.stirred.correction = TRUE,
  clint.pvalue.threshold = 0.05,
  minimum.Funbound.plasma = 1e-04
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_elimination_rate_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the cas number or the chemical name must be
specified.</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the cas number must be
specified.</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_steadystate or
1compartment function, overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true.</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>In calculating elimination rate, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_adjusted.clint">adjusted.Clint</code></td>
<td>
<p>Uses Kilford et al. (2008) hepatocyte incubation
binding adjustment for Clint when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_well.stirred.correction">well.stirred.correction</code></td>
<td>
<p>Uses correction in calculation of hepatic
clearance for -stirred model if TRUE.  This assumes clearance relative
to amount unbound in whole blood instead of plasma, but converted to use
with plasma concentration.</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_clint.pvalue.threshold">clint.pvalue.threshold</code></td>
<td>
<p>Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.</p>
</td></tr>
<tr><td><code id="calc_elimination_rate_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Elimination rate calculated by dividing the total clearance (using the
default -stirred hepatic model) by the volume of distribution. When
species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>


<h3>Value</h3>

<table>
<tr><td><code>Elimination rate</code></td>
<td>
<p>Units of 1/h.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Schmitt, Walter. &quot;General approach for the calculation of tissue 
to plasma partition coefficients.&quot; Toxicology in vitro 22.2 (2008): 457-467.
</p>
<p>Dawson DE, Ingle BL, Phillips KA, Nichols JW, Wambaugh JF, Tornero-Velez R (2021).
&ldquo;Designing QSARs for Parameters of High-Throughput Toxicokinetic Models Using Open-Source Descriptors.&rdquo;
<em>Environmental Science &amp; Technology</em>, <b>55</b>(9), 6505-6517.
<a href="https://doi.org/10.1021/acs.est.0c06117">doi:10.1021/acs.est.0c06117</a>, PMID: 33856768, https://doi.org/10.1021/acs.est.0c06117. 
</p>
<p>Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
calc_elimination_rate(chem.name="Bisphenol A")
calc_elimination_rate(chem.name="Bisphenol A",species="Rat")
calc_elimination_rate(chem.cas="80-05-7")

</code></pre>

<hr>
<h2 id='calc_fbio.oral'>Functions for calculating the bioavaialble fractions from oral doses</h2><span id='topic+calc_fbio.oral'></span><span id='topic+calc_fabs.oral'></span><span id='topic+calc_fgut.oral'></span>

<h3>Description</h3>

<p>These functions calculate the fraction of chemical absorbed from the gut
based upon in vitro measured Caco-2 membrane permeability data.
Caco-2 permeabilities (<code class="reqn">10^{-6}</code> cm/s) are related to 
effective permeability based on Yang et al. (2007).
These functions calculate the fraction absorbed (calc_fabs.oral &ndash; 
S Darwich et al. (2010)), the fraction
surviving first pass gut metabolism (calc_fgut.oral), and the overall systemic
oral bioavailability
(calc_fbio.oral). Note that the first pass hepatic clearance is calculated within the
parameterization and other functions. using <code><a href="#topic+calc_hep_bioavailability">calc_hep_bioavailability</a></code> 
We assume that systemic oral bioavailability (<code class="reqn">F_{bio}</code>)
consists of three components: 
(1) the fraction of chemical absorbed from intestinal lumen into enterocytes 
(<code class="reqn">F_{abs}</code>), 
(2) the fraction surviving intestinal metabolism 
(<code class="reqn">F_{gut}</code>), and 
(3) the fraction surviving first-pass hepatic metabolism 
(<code class="reqn">F_{hep}</code>). This function returns (<code class="reqn">F_{abs}*F_{gut}</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_fbio.oral(
  parameters = NULL,
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  suppress.messages = FALSE
)

calc_fabs.oral(
  parameters = NULL,
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  suppress.messages = FALSE,
  Caco2.Pab.default = 1.6
)

calc_fgut.oral(
  parameters = NULL,
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  suppress.messages = FALSE,
  Caco2.Pab.default = 1.6
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_fbio.oral_+3A_parameters">parameters</code></td>
<td>
<p>(List) A list of the parameters (Caco2.Pab, Funbound.Plasma, Rblood2plasma,
Clint, BW, Qsmallintestine, Fabs, Fgut) used in the calculation, either supplied by user
or calculated in <code><a href="#topic+parameterize_steadystate">parameterize_steadystate</a></code>.</p>
</td></tr>
<tr><td><code id="calc_fbio.oral_+3A_chem.cas">chem.cas</code></td>
<td>
<p>(Character) Chemical CAS number. (Defaults to 'NULL'.)
(Note: Either the chemical name, CAS number, or EPA's DSSTox Structure ID
must be specified).</p>
</td></tr>
<tr><td><code id="calc_fbio.oral_+3A_chem.name">chem.name</code></td>
<td>
<p>(Character) Chemical name. (Defaults to 'NULL'.)
(Note: Either the chemical name, CAS number, or EPA's DSSTox Structure ID
must be specified).</p>
</td></tr>
<tr><td><code id="calc_fbio.oral_+3A_dtxsid">dtxsid</code></td>
<td>
<p>(Character) EPA's DSSTox Structure ID
(<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>). (Defaults to 'NULL'.)
(Note: Either the chemical name, CAS number, or EPA's DSSTox Structure ID
must be specified).</p>
</td></tr>
<tr><td><code id="calc_fbio.oral_+3A_species">species</code></td>
<td>
<p>(Character) Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;,
&quot;Mouse&quot;, or default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_fbio.oral_+3A_default.to.human">default.to.human</code></td>
<td>
<p>(Logical) Substitutes missing rat values with
human values if TRUE. (Not applicable for 'calc_fabs.oral'.)
(Defaults to 'FALSE'.)</p>
</td></tr>
<tr><td><code id="calc_fbio.oral_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>(Logical) Whether or not the output message is
suppressed. (Defaults to 'FALSE'.)</p>
</td></tr>
<tr><td><code id="calc_fbio.oral_+3A_caco2.pab.default">Caco2.Pab.default</code></td>
<td>
<p>(Numeric) Caco2 apical to basolateral data.
(Defaults to  1.6.) (Not applicable for 'calc_fbio.oral'.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We model systemic oral bioavailability as 
<code class="reqn">F_{bio}=F_{abs}*F_{gut}*F_{hep}</code>.
<code class="reqn">F_{hep}</code> is estimated from in vitro TK data using 
<code><a href="#topic+calc_hep_bioavailability">calc_hep_bioavailability</a></code>.
If <code class="reqn">F_{bio}</code> has been measured in vivo and is found in
table <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> then we set 
<code class="reqn">F_{abs}*F_{gut}</code> to the measured value divided by <code class="reqn">F_{hep}</code>.
Otherwise, if Caco2 membrane permeability data or predictions
are available <code class="reqn">F_{abs}</code> is estimated using <code><a href="#topic+calc_fgut.oral">calc_fgut.oral</a></code>.
Intrinsic hepatic metabolism is used to very roughly estimate (<code class="reqn">F_{gut}</code>)
using <code><a href="#topic+calc_fgut.oral">calc_fgut.oral</a></code>.
If argument keepit100 is used then there is complete absorption from the gut
(that is, <code class="reqn">F_{abs}=F_{gut}=1</code>).
</p>


<h3>Value</h3>

<table>
<tr><td><code>fbio.oral</code></td>
<td>
<p>Oral bioavailability, the fraction of oral dose 
reaching systemic distribution in the body.</p>
</td></tr>
<tr><td><code>fabs.oral</code></td>
<td>
<p>Fraction of dose absorbed, i.e. the fraction of the dose
that enters the gutlumen.</p>
</td></tr> <tr><td><code>fgut.oral</code></td>
<td>
<p>Fraction of chemical surviving
first pass metabolism in the gut.</p>
</td></tr> <tr><td><code>fhep.oral</code></td>
<td>
<p>Fraction of chemical
surviving first pass hepatic clearance.</p>
</td></tr>
</table>


<h3>Functions</h3>


<ul>
<li> <p><code>calc_fabs.oral()</code>: Calculate the fraction absorbed in the gut (Darwich et al., 2010)
</p>
</li>
<li> <p><code>calc_fgut.oral()</code>: Calculate the fraction of chemical surviving first pass metabolism in the gut
</p>
</li></ul>


<h3>Author(s)</h3>

<p>Gregory Honda
</p>


<h3>References</h3>

<p>S Darwich A, Neuhoff S, Jamei M, Rostami-Hodjegan A (2010).
&ldquo;Interplay of metabolism and transport in determining oral drug absorption and gut wall metabolism: a simulation assessment using the 'Advanced Dissolution, Absorption, Metabolism (ADAM)' model.&rdquo;
<em>Current drug metabolism</em>, <b>11</b>(9), 716&ndash;729.
Yang J, Jamei M, Yeo KR, Tucker GT, Rostami-Hodjegan A (2007).
&ldquo;Prediction of intestinal first-pass drug metabolism.&rdquo;
<em>Current drug metabolism</em>, <b>8</b>(7), 676&ndash;684.
Honda G, Kenyon EM, Davidson-Fritz SE, Dinallo R, El-Masri H, Korel-Bexell E, Li L, Paul-Friedman K, Pearce R, Sayre R, Strock C, Thomas R, Wetmore BA, Wambaugh JF (2023).
&ldquo;Impact of Gut Permeability on Estimation of Oral Bioavailability for Chemicals in Commerce and the Environment.&rdquo;
Unpublished.
</p>

<hr>
<h2 id='calc_fetal_phys'>Calculate maternal-fetal physiological parameters</h2><span id='topic+calc_fetal_phys'></span>

<h3>Description</h3>

<p>This function uses the equations from Kapraun (2019) to calculate chemical-
independent physiological paramreters as a function of gestational age in 
weeks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_fetal_phys(week = 12, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_fetal_phys_+3A_week">week</code></td>
<td>
<p>Gestational week</p>
</td></tr>
<tr><td><code id="calc_fetal_phys_+3A_...">...</code></td>
<td>
<p>Additional arguments to parameterize_fetal_pbtk</p>
</td></tr>
</table>


<h3>Details</h3>

<p style="text-align: center;"><code class="reqn">BW = pre_pregnant_BW + 
   BW_cubic_theta1 * tw + 
   BW_cubic_theta2 * tw^2 + 
   BW_cubic_theta3 * tw^3
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Wadipose =  Wadipose_linear_theta0 + Wadipose_linear_theta1 * tw ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Wfkidney = 0.001 * Wfkidney_gompertz_theta0 * exp ( Wfkidney_gompertz_theta1 / Wfkidney_gompertz_theta2 * ( 1 - exp ( - Wfkidney_gompertz_theta2 * tw ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Wfthyroid = 0.001 * Wfthyroid_gompertz_theta0 * exp ( Wfthyroid_gompertz_theta1 / Wfthyroid_gompertz_theta2 * ( 1 - exp ( - Wfthyroid_gompertz_theta2 * tw ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Wfliver = 0.001 * Wfliver_gompertz_theta0 * exp ( Wfliver_gompertz_theta1 / Wfliver_gompertz_theta2 * ( 1 - exp ( - Wfliver_gompertz_theta2 * tw ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Wfbrain = 0.001 * Wfbrain_gompertz_theta0 * exp ( Wfbrain_gompertz_theta1 / Wfbrain_gompertz_theta2 * ( 1 - exp ( - Wfbrain_gompertz_theta2 * tw ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Wfgut = 0.001 * Wfgut_gompertz_theta0 * exp ( Wfgut_gompertz_theta1 / Wfgut_gompertz_theta2 * ( 1 - exp ( - Wfgut_gompertz_theta2 * tw ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Wflung = 0.001 * Wflung_gompertz_theta0 * exp ( Wflung_gompertz_theta1 / Wflung_gompertz_theta2 * ( 1 - exp ( - Wflung_gompertz_theta2 * tw ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 hematocrit = ( hematocrit_quadratic_theta0 + hematocrit_quadratic_theta1 * tw + hematocrit_quadratic_theta2 * pow ( tw , 2 ) ) / 100 ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Rblood2plasma = 1 - hematocrit + hematocrit * Krbc2pu * Fraction_unbound_plasma ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 fhematocrit = ( fhematocrit_cubic_theta1 * tw + fhematocrit_cubic_theta2 * pow ( tw , 2 ) + fhematocrit_cubic_theta3 * pow ( tw , 3 ) ) / 100 ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Rfblood2plasma = 1 - fhematocrit + fhematocrit * Kfrbc2pu * Fraction_unbound_plasma_fetus ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 fBW = 0.001 * fBW_gompertz_theta0 * exp ( fBW_gompertz_theta1 / fBW_gompertz_theta2 * ( 1 - exp ( - fBW_gompertz_theta2 * tw ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vplacenta = 0.001 * ( Vplacenta_cubic_theta1 * tw + Vplacenta_cubic_theta2 * pow ( tw , 2 ) + Vplacenta_cubic_theta3 * pow ( tw , 3 ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vamnf = 0.001 * Vamnf_logistic_theta0 / ( 1 + exp ( - Vamnf_logistic_theta1 * ( tw - Vamnf_logistic_theta2 ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vplasma = Vplasma_mod_logistic_theta0 / ( 1 + exp ( - Vplasma_mod_logistic_theta1 * ( tw - Vplasma_mod_logistic_theta2 ) ) ) + Vplasma_mod_logistic_theta3 ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vrbcs = hematocrit / ( 1 - hematocrit ) * Vplasma ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vven = venous_blood_fraction * ( Vrbcs + Vplasma ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vart = arterial_blood_fraction * ( Vrbcs + Vplasma ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vadipose = 1 / adipose_density * Wadipose ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vffmx = 1 / ffmx_density * ( BW - Wadipose - ( fBW + placenta_density * Vplacenta + amnf_density * Vamnf ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vallx = Vart + Vven + Vthyroid + Vkidney + Vgut + Vliver + Vlung ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vrest = Vffmx - Vallx ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vfart = 0.001 * arterial_blood_fraction * fblood_weight_ratio * fBW ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vfven = 0.001 * venous_blood_fraction * fblood_weight_ratio * fBW ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vfkidney = 1 / kidney_density * Wfkidney ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vfthyroid = 1 / thyroid_density * Wfthyroid ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vfliver = 1 / liver_density * Wfliver ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vfbrain = 1 / brain_density * Wfbrain ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vfgut = 1 / gut_density * Wfgut ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vflung = 1 / lung_density * Wflung ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Vfrest = fBW - ( Vfart + Vfven + Vfbrain + Vfkidney + Vfthyroid + Vfliver + Vfgut + Vflung ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qcardiac = 24 * ( Qcardiac_cubic_theta0 + Qcardiac_cubic_theta1 * tw + Qcardiac_cubic_theta2 * pow ( tw , 2 ) + Qcardiac_cubic_theta3 * pow ( tw , 3 ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qgut = 0.01 * ( Qgut_percent_initial + ( Qgut_percent_terminal - Qgut_percent_initial ) / term * tw ) * Qcardiac ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qkidney = 24 * ( Qkidney_cubic_theta0 + Qkidney_cubic_theta1 * tw + Qkidney_cubic_theta2 * pow ( tw , 2 ) + Qkidney_cubic_theta3 * pow ( tw , 3 ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qliver = 0.01 * ( Qliver_percent_initial + ( Qliver_percent_terminal - Qliver_percent_initial ) / term * tw ) * Qcardiac ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qthyroid = 0.01 * ( Qthyroid_percent_initial + ( Qthyroid_percent_terminal - Qthyroid_percent_terminal ) / term * tw ) * Qcardiac ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qplacenta = 24 * Qplacenta_linear_theta1 * 1000 * Vplacenta ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qadipose = 0.01 * ( Qadipose_percent_initial + ( Qadipose_percent_terminal - Qadipose_percent_initial ) / term * tw ) * Qcardiac ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qrest = Qcardiac - ( Qgut + Qkidney + Qliver + Qthyroid + Qplacenta + Qadipose ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qgfr = 60 * 24 * 0.001 * ( Qgfr_quadratic_theta0 + Qgfr_quadratic_theta1 * tw + Qgfr_quadratic_theta2 * pow ( tw , 2 ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfrvtl = 60 * 24 * 0.001 * Qfrvtl_logistic_theta0 / ( 1 + exp ( - Qfrvtl_logistic_theta1 * ( tw - Qfrvtl_logistic_theta2 ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qflvtl = 60 * 24 * 0.001 * Qflvtl_logistic_theta0 / ( 1 + exp ( - Qflvtl_logistic_theta1 * ( tw - Qflvtl_logistic_theta2 ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfda = 60 * 24 * 0.001 * Qfda_logistic_theta0 / ( 1 + exp ( - Qfda_logistic_theta1 * ( tw - Qfda_logistic_theta2 ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfartb = Qflvtl + Qfda ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfcardiac = Qfartb ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qflung = Qfrvtl - Qfda ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfplacenta = 60 * 24 * 0.001 * Qfplacenta_logistic_theta0 / ( 1 + exp ( - Qfplacenta_logistic_theta1 * ( tw - Qfplacenta_logistic_theta2 ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfdv = 60 * 24 * 0.001 * Qfdv_gompertz_theta0 * exp ( Qfdv_gompertz_theta1 / Qfdv_gompertz_theta2 * ( 1 - exp ( - Qfdv_gompertz_theta2 * tw ) ) ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfgut = Qfgut_percent / Qfnonplacental_percent * ( 1 - Qfplacenta / Qfartb ) * Qfartb ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfkidney = Qfkidney_percent / Qfnonplacental_percent * ( 1 - Qfplacenta / Qfartb ) * Qfartb ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfbrain = Qfbrain_percent / Qfnonplacental_percent * ( 1 - Qfplacenta / Qfartb ) * Qfartb ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfliver = Qfliver_percent / ( 100 - ( Qbrain_percent + Qkidney_percent + Qgut_percent ) ) * ( 1 - ( Qfbrain_percent +
Qfkidney_percent + Qfgut_percent ) / Qfnonplacental_percent ) * ( 1 - Qfplacenta / Qfartb ) * Qfartb ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfthyroid = Qfthyroid_percent / ( 100 - ( Qbrain_percent + Qkidney_percent + Qgut_percent ) ) * ( 1 - ( Qfbrain_percent +
Qfkidney_percent + Qfgut_percent ) / Qfnonplacental_percent ) * ( 1 - Qfplacenta / Qfartb ) * Qfartb ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfrest = Qfcardiac - ( Qfplacenta + Qfgut + Qfliver + Qfthyroid + Qfkidney + Qfbrain ) ;
 </code>
</p>

<p style="text-align: center;"><code class="reqn">
 Qfbypass = Qfcardiac - Qflung ;
 </code>
</p>



<h3>Value</h3>

<p>list containing:
</p>
<table>
<tr><td><code>BW</code></td>
<td>
<p>Maternal body weight, kg</p>
</td></tr>
<tr><td><code>Wadipose</code></td>
<td>
<p>Maternal adipose fraction of total weight</p>
</td></tr>
<tr><td><code>Wfkidney</code></td>
<td>
<p>Fetal kidney fraction of total weight</p>
</td></tr>
<tr><td><code>Wfthyroid</code></td>
<td>
<p>Fetal thyroid fraction of total weight</p>
</td></tr>
<tr><td><code>Wfliver</code></td>
<td>
<p>Fetal liver fraction of total weight</p>
</td></tr>
<tr><td><code>Wfbrain</code></td>
<td>
<p>Fetal brain fraction of total weight</p>
</td></tr>
<tr><td><code>Wfgut</code></td>
<td>
<p>Fetal gut fraction of total weight</p>
</td></tr>
<tr><td><code>Wflung</code></td>
<td>
<p>Fetal lung fraction of total weight</p>
</td></tr>
<tr><td><code>hematocrit</code></td>
<td>
<p>Maternal hematocrit fraction of blood</p>
</td></tr>
<tr><td><code>Rblood2plasma</code></td>
<td>
<p>Maternal Rblood2plasma</p>
</td></tr>
<tr><td><code>fhematocrit</code></td>
<td>
<p>Fetal hematocrit fraction of blood</p>
</td></tr>
<tr><td><code>Rfblood2plasma</code></td>
<td>
<p>Fetal Rfblood2plasma</p>
</td></tr>
<tr><td><code>fBW</code></td>
<td>
<p>Fetal body weight, kg</p>
</td></tr>
<tr><td><code>Vplacenta</code></td>
<td>
<p>Volume of Vplacenta, L</p>
</td></tr>
<tr><td><code>Vamnf</code></td>
<td>
<p>Volume of amniotic fluid, L</p>
</td></tr>
<tr><td><code>Vplasma</code></td>
<td>
<p>Maternal volume of plasma, L</p>
</td></tr>
<tr><td><code>Vrbcs</code></td>
<td>
<p>Maternal volume of red blood cells, L</p>
</td></tr>
<tr><td><code>Vven</code></td>
<td>
<p>Maternal volume of venous blood, L</p>
</td></tr>
<tr><td><code>Vart</code></td>
<td>
<p>Maternal volume of arterial blood, L</p>
</td></tr>
<tr><td><code>Vadipose</code></td>
<td>
<p>Maternal volume of adipose, L</p>
</td></tr>
<tr><td><code>Vffmx</code></td>
<td>
<p>Fetal volume ofVffmx, L</p>
</td></tr>
<tr><td><code>Vallx</code></td>
<td>
<p>Vallx, L</p>
</td></tr>
<tr><td><code>Vrest</code></td>
<td>
<p>Maternal volume of rest of body, L</p>
</td></tr>
<tr><td><code>Vfart</code></td>
<td>
<p>Fetal volume of arterial blood, L</p>
</td></tr>
<tr><td><code>Vfven</code></td>
<td>
<p>Fetal volume of venous blood, L</p>
</td></tr>
<tr><td><code>Vfkidney</code></td>
<td>
<p>Fetal volume of kidney, L</p>
</td></tr>
<tr><td><code>Vfthyroid</code></td>
<td>
<p>Fetal volume of thyroid, L</p>
</td></tr>
<tr><td><code>Vfliver</code></td>
<td>
<p>Fetal volume of liver, L</p>
</td></tr>
<tr><td><code>Vfbrain</code></td>
<td>
<p>Fetal volume of brain, L</p>
</td></tr>
<tr><td><code>Vfgut</code></td>
<td>
<p>Fetal volume of gut, L</p>
</td></tr>
<tr><td><code>Vflung</code></td>
<td>
<p>Fetal volume of lung, L</p>
</td></tr>
<tr><td><code>Vfrest</code></td>
<td>
<p>Fetal volume of rest of body, L</p>
</td></tr>
<tr><td><code>Qcardiac</code></td>
<td>
<p>Maternal cardiac output blood flow, L/day</p>
</td></tr>
<tr><td><code>Qgut</code></td>
<td>
<p>Maternal blood flow to gut, L/day</p>
</td></tr>
<tr><td><code>Qkidney</code></td>
<td>
<p>Maternal blood flow to kidney, L/day</p>
</td></tr>
<tr><td><code>Qliver</code></td>
<td>
<p>Maternal blood flow to liver, L/day</p>
</td></tr>
<tr><td><code>Qthyroid</code></td>
<td>
<p>Maternal blood flow to thyroid, L/day</p>
</td></tr>
<tr><td><code>Qplacenta</code></td>
<td>
<p>Maternal blood flow to placenta, L/day</p>
</td></tr>
<tr><td><code>Qadipose</code></td>
<td>
<p>Maternal blood flow to adipose, L/day</p>
</td></tr>
<tr><td><code>Qrest</code></td>
<td>
<p>Maternal blood flow to rest, L/day</p>
</td></tr>
<tr><td><code>Qgfr</code></td>
<td>
<p>Maternal glomerular filtration rate in kidney, L/day</p>
</td></tr>
<tr><td><code>Qfrvtl</code></td>
<td>
<p>Fetal blood flow to right ventricle, L/day</p>
</td></tr>
<tr><td><code>Qflvtl</code></td>
<td>
<p>Fetal blood flow to left ventircle, L/day</p>
</td></tr>
<tr><td><code>Qfda</code></td>
<td>
<p>Fetal blood flow to Qfda, L/day</p>
</td></tr>
<tr><td><code>Qfartb</code></td>
<td>
<p>Fetal blood flow to Qfartb, L/day</p>
</td></tr>
<tr><td><code>Qfcardiac</code></td>
<td>
<p>Fetal cardiac output blood flow, L/day</p>
</td></tr>
<tr><td><code>Qflung</code></td>
<td>
<p>Fetal blood flow to lung, L/day</p>
</td></tr>
<tr><td><code>Qfplacenta</code></td>
<td>
<p>Fetal blood flow to placenta, L/day</p>
</td></tr>
<tr><td><code>Qfdv</code></td>
<td>
<p>Fetal blood flow to Qfdv, L/day</p>
</td></tr>
<tr><td><code>Qfgut</code></td>
<td>
<p>Fetal blood flow to gut, L/day</p>
</td></tr>
<tr><td><code>Qfkidney</code></td>
<td>
<p>Fetal blood flow to kidney, L/day</p>
</td></tr>
<tr><td><code>Qfbrain</code></td>
<td>
<p>Fetal blood flow to brain, L/day</p>
</td></tr>
<tr><td><code>Qfliver</code></td>
<td>
<p>Fetal blood flow to liver, L/day</p>
</td></tr>
<tr><td><code>Qfthyroid</code></td>
<td>
<p>Fetal blood flow to thyroid, L/day</p>
</td></tr>
<tr><td><code>Qfrest</code></td>
<td>
<p>Fetal blood flow to rest, L/day</p>
</td></tr>
<tr><td><code>Qfbypass</code></td>
<td>
<p>Fetal blood flow to Qfbypass, L/day</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Kapraun DF, Wambaugh JF, Setzer RW, Judson RS (2019).
&ldquo;Empirical models for anatomical and physiological changes in a human mother and fetus during pregnancy and gestation.&rdquo;
<em>PLOS ONE</em>, <b>14</b>(5), 1-56.
<a href="https://doi.org/10.1371/journal.pone.0215906">doi:10.1371/journal.pone.0215906</a>.
</p>

<hr>
<h2 id='calc_fup_correction'>Calculate the correction for lipid binding in plasma binding assay</h2><span id='topic+calc_fup_correction'></span>

<h3>Description</h3>

<p>Poulin and Haddad (2012) observed &quot;...that for
a highly lipophilic compound, the calculated 
f<sub>up</sub> is by
far [less than] the experimental values observed under
in vitro conditions.&quot; Pearce et al. (2017) hypothesized that there was additional lipid
binding in vivo that acted as a sink for lipophilic compounds, reducing the
effective f<sub>up</sub> in vivo. It is 
possible that this is due to the binding of lipophilic compounds on the non
plasma-side of the rapid equilibrium dialysis plates (Waters et al., 2008).
Pearce et al. (2017) compared predicted and observed 
tissue partition coefficients
for a range of compounds. They showed that predictions were improved by 
adding additional binding proportional to the distribution coefficient 
D<sub>ow</sub> 
(<code><a href="#topic+calc_dow">calc_dow</a></code>)
and the fractional volume of lipid in
plasma (F<sub>lipid</sub>). 
We calculate
F<sub>lipid</sub> as the     
sum of the physiological plasma neutral lipid fractional volume and 30 percent of 
the plasma neutral phospholipid fractional volume. We use values
from Peyret et al. (2010) for rats and Poulin and Haddad (2012)
for humans. The estimate of 30 percent of the
neutral phospholipid volume as neutral lipid was used for simplictity's sake in
place of our membrane affinity predictor. To account for additional binding to lipid, 
plasma to water partitioning
(K<sub>plasma:water</sub> = 1/f<sub>up</sub>)
is increased as such:
K<sup>corrected</sup><sub>plasma:water</sub> = 1/f<sup>corrected</sup><sub>up</sub> = 1/f<sup>in vitro</sup><sub>up</sub> + D<sub>ow</sub>*F<sub>lipid</sub>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_fup_correction(
  fup = NULL,
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  Flipid = NULL,
  plasma.pH = 7.4,
  dow74 = NULL,
  species = "Human",
  default.to.human = FALSE,
  force.human.fup = FALSE,
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_fup_correction_+3A_fup">fup</code></td>
<td>
<p>Fraction unbound in plasma, if provided this argument overides
values from argument parameters and <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code></p>
</td></tr>
<tr><td><code id="calc_fup_correction_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_fup_correction_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_fup_correction_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_fup_correction_+3A_parameters">parameters</code></td>
<td>
<p>Parameters from the appropriate parameterization function
for the model indicated by argument model</p>
</td></tr>
<tr><td><code id="calc_fup_correction_+3A_flipid">Flipid</code></td>
<td>
<p>The fractional volume of lipid in plasma (from <code><a href="#topic+physiology.data">physiology.data</a></code>)</p>
</td></tr>
<tr><td><code id="calc_fup_correction_+3A_plasma.ph">plasma.pH</code></td>
<td>
<p>pH of plasma (default 7.4)</p>
</td></tr>
<tr><td><code id="calc_fup_correction_+3A_dow74">dow74</code></td>
<td>
<p>The octanol-water distribution ratio (DOW).</p>
</td></tr>
<tr><td><code id="calc_fup_correction_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_fup_correction_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing fraction of unbound plasma with
human values if true.</p>
</td></tr>
<tr><td><code id="calc_fup_correction_+3A_force.human.fup">force.human.fup</code></td>
<td>
<p>Returns human fraction of unbound plasma in
calculation for rats if true.
When species is specified as rabbit, dog, or mouse, the human unbound
fraction is substituted.</p>
</td></tr>
<tr><td><code id="calc_fup_correction_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that octanal:water partitioning above 1:1,000,000 
(LogD<sub>ow</sub> > 6)
are truncated at 1:1,000,000 because greater partitioning would
likely take longer than protein binding assay itself.
</p>


<h3>Value</h3>

<p>A numeric fraction unpbound in plasma between zero and one
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Davis JL, Wambaugh JF (2017).
&ldquo;Evaluation and calibration of high-throughput predictions of chemical distribution to tissues.&rdquo;
<em>Journal of pharmacokinetics and pharmacodynamics</em>, <b>44</b>, 549&ndash;565.
</p>
<p>Peyret T, Poulin P, Krishnan K (2010).
&ldquo;A unified algorithm for predicting partition coefficients for PBPK modeling of drugs and environmental chemicals.&rdquo;
<em>Toxicology and applied pharmacology</em>, <b>249</b>(3), 197&ndash;207.
</p>
<p>Poulin P, Haddad S (2012).
&ldquo;Advancing prediction of tissue distribution and volume of distribution of highly lipophilic compounds from a simplified tissue-composition-based model as a mechanistic animal alternative method.&rdquo;
<em>Journal of pharmaceutical sciences</em>, <b>101</b>(6), 2250&ndash;2261.
</p>
<p>Schmitt W (2008).
&ldquo;General approach for the calculation of tissue to plasma partition coefficients.&rdquo;
<em>Toxicology in vitro</em>, <b>22</b>(2), 457&ndash;467.
</p>
<p>Waters, Nigel J., et al. &quot;Validation of a rapid equilibrium dialysis 
approach for the measurement of plasma protein binding.&quot; Journal of 
pharmaceutical sciences 97.10 (2008): 4586-4595.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+apply_fup_adjustment">apply_fup_adjustment</a></code>
</p>
<p><code><a href="#topic+calc_dow">calc_dow</a></code>
</p>

<hr>
<h2 id='calc_half_life'>Calculates the half-life for a one compartment model.</h2><span id='topic+calc_half_life'></span>

<h3>Description</h3>

<p>This function calculates the half life from the three compartment
steady state model where elimination is entirely due to metabolism by the
liver and glomerular filtration in the kidneys.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_half_life(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  suppress.messages = TRUE,
  default.to.human = FALSE,
  restrictive.clearance = TRUE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  well.stirred.correction = TRUE,
  clint.pvalue.threshold = 0.05,
  minimum.Funbound.plasma = 1e-04
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_half_life_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the cas number or the chemical name must be specified.</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the cas number must be specified.</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_steadystate or
1compartment function, overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true.</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>In calculating elimination rate, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_well.stirred.correction">well.stirred.correction</code></td>
<td>
<p>Uses correction in calculation of hepatic
clearance for -stirred model if TRUE.  This assumes clearance relative
to amount unbound in whole blood instead of plasma, but converted to use
with plasma concentration.</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_clint.pvalue.threshold">clint.pvalue.threshold</code></td>
<td>
<p>Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.</p>
</td></tr>
<tr><td><code id="calc_half_life_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Half life is calculated by dividing the natural-log of 2 by the elimination
rate from the one compartment model.
</p>


<h3>Value</h3>

<table>
<tr><td><code>Half life</code></td>
<td>
<p>Units of h.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sarah E. Davidson
</p>


<h3>See Also</h3>

<p>[calc_elimination_rate()] for the elimination rate calculation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
calc_half_life(chem.name="Bisphenol A")
calc_half_life(chem.name="Bisphenol A",species="Rat")
calc_half_life(chem.cas="80-05-7")

</code></pre>

<hr>
<h2 id='calc_hep_bioavailability'>Calculate first pass heaptic metabolism</h2><span id='topic+calc_hep_bioavailability'></span>

<h3>Description</h3>

<p>For models that don't described first pass blood flow from the gut, need to
cacluate a hepatic bioavailability, that is, the fraction of chemical 
systemically available after metabolism during the first pass through the 
liver (Rowland, 1973 Equation 29, where k21 is blood flow through the liver
and k23 is clearance from the liver in Figure 1 in that paper).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_hep_bioavailability(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  restrictive.clearance = TRUE,
  flow.34 = TRUE,
  suppress.messages = FALSE,
  species = "Human"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_hep_bioavailability_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_hep_bioavailability_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_hep_bioavailability_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_hep_bioavailability_+3A_parameters">parameters</code></td>
<td>
<p>Parameters from the appropriate parameterization function
for the model indicated by argument model</p>
</td></tr>
<tr><td><code id="calc_hep_bioavailability_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to 1) in 
liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="calc_hep_bioavailability_+3A_flow.34">flow.34</code></td>
<td>
<p>A logical constraint</p>
</td></tr>
<tr><td><code id="calc_hep_bioavailability_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not to suppress the output message.</p>
</td></tr>
<tr><td><code id="calc_hep_bioavailability_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table whose columns are the parameters of the HTTK model
specified in <code>model</code>.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Rowland, Malcolm, Leslie Z. Benet, and Garry G. Graham. 
Rowland M, Benet LZ, Graham GG (1973).
&ldquo;Clearance concepts in pharmacokinetics.&rdquo;
<em>Journal of pharmacokinetics and biopharmaceutics</em>, <b>1</b>(2), 123&ndash;136.
</p>

<hr>
<h2 id='calc_hep_clearance'>Calculate the hepatic clearance.</h2><span id='topic+calc_hep_clearance'></span>

<h3>Description</h3>

<p>This function calculates the hepatic clearance in plasma for using the
Houston (2004) are also available. In vitro measured hepatic clearace is 
corrected for the free fraction in the assay using the model of Kilford et 
al. (2008).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_hep_clearance(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  hepatic.model = "well-stirred",
  suppress.messages = FALSE,
  well.stirred.correction = TRUE,
  restrictive.clearance = TRUE,
  species = "Human",
  adjusted.Funbound.plasma = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_hep_clearance_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters
must be specified.</p>
</td></tr>
<tr><td><code id="calc_hep_clearance_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must
be specified.</p>
</td></tr>
<tr><td><code id="calc_hep_clearance_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_hep_clearance_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_steadystate
function, overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_hep_clearance_+3A_hepatic.model">hepatic.model</code></td>
<td>
<p>Model used in calculating hepatic clearance, unscaled,
parallel tube, dispersion, or default well-stirred.</p>
</td></tr>
<tr><td><code id="calc_hep_clearance_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not to suppress the output message.</p>
</td></tr>
<tr><td><code id="calc_hep_clearance_+3A_well.stirred.correction">well.stirred.correction</code></td>
<td>
<p>Uses correction in calculation of hepatic
clearance for well-stirred model if TRUE for hepatic.model well-stirred.
This assumes clearance relative to amount unbound in whole blood instead of
plasma, but converted to use with plasma concentration.</p>
</td></tr>
<tr><td><code id="calc_hep_clearance_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to
1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="calc_hep_clearance_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_hep_clearance_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses Pearce et al. (2017) lipid binding adjustment
for Funbound.plasma (which impacts partition coefficients) when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="calc_hep_clearance_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to parameterize_steadystate if
parameters is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Hepatic Clearance</code></td>
<td>
<p>Units of L/h/kg BW.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>References</h3>

<p>Ito, K., &amp; Houston, J. B. (2004). &quot;Comparison of the use of liver models for 
predicting drug clearance using in vitro kinetic data from hepatic microsomes 
and isolated hepatocytes.&quot; Pharmaceutical Research, 21(5), 785-792.
</p>
<p>Carlile, David J., Katayoun Zomorodi, and J. Brian Houston. &quot;Scaling factors 
to relate drug metabolic clearance in hepatic microsomes, isolated 
hepatocytes, and the intact liver: studies with induced livers involving 
diazepam.&quot; Drug metabolism and disposition 25.8 (1997): 903-911.
</p>
<p>Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
calc_hep_clearance(chem.name="Ibuprofen",hepatic.model='unscaled')
calc_hep_clearance(chem.name="Ibuprofen",well.stirred.correction=FALSE)

</code></pre>

<hr>
<h2 id='calc_hep_fu'>Calculate the free chemical in the hepaitic clearance assay</h2><span id='topic+calc_hep_fu'></span>

<h3>Description</h3>

<p>This function uses the method from Kilford et al. (2008) to calculate the
fraction of unbound chemical in the 
hepatocyte intrinsic clearance assay. The bound chemical is presumed to be
unavailable during the performance of the assay, so this fraction can be
used to increase the apparent clearance rate to better estimate in vivo 
clearance. 
For bases, the fraction of chemical unbound in hepatocyte clearance assays 
(fu<sub>hep</sub>) is calculated in terms of 
logP<sub>ow</sub>
but for neutrual and acidic compounds we use 
logD<sub>ow</sub> (from <code><a href="#topic+calc_dow">calc_dow</a></code>). 
Here we denote the appropriate partition coefficient as &quot;logP/D&quot;.
Kilford et al. (2008) calculates
fu<sub>hep</sub> = 1/(1 + 125*V<sub>R</sub>*10^(0.072*logP/D<sup>2</sup> + 0.067*logP/D-1.126))
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_hep_fu(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  Vr = 0.005,
  pH = 7.4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_hep_fu_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_hep_fu_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_hep_fu_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_hep_fu_+3A_parameters">parameters</code></td>
<td>
<p>Parameters from the appropriate parameterization function
for the model indicated by argument model</p>
</td></tr>
<tr><td><code id="calc_hep_fu_+3A_vr">Vr</code></td>
<td>
<p>Ratio of cell volume to incubation volume. Default (0.005) is taken from</p>
</td></tr>
<tr><td><code id="calc_hep_fu_+3A_ph">pH</code></td>
<td>
<p>pH of the incupation medium.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that octanal:water partitioning above 1:1,000,000 
(LogP<sub>ow</sub> > 6)
are truncated at 1:1,000,000 because greater partitioning would
likely take longer than hepatocyte assay itself.
</p>


<h3>Value</h3>

<p>A numeric fraction between zero and one
</p>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>References</h3>

<p>Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197. 
</p>
<p>Wetmore BA, Wambaugh JF, Allen B, Ferguson SS, Sochaski MA, Setzer RW, Houck KA, Strope CL, Cantwell K, Judson RS, others (2015).
&ldquo;Incorporating high-throughput exposure predictions with dosimetry-adjusted in vitro bioactivity to inform chemical toxicity testing.&rdquo;
<em>Toxicological Sciences</em>, <b>148</b>(1), 121&ndash;136.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+apply_clint_adjustment">apply_clint_adjustment</a></code>
</p>

<hr>
<h2 id='calc_hepatic_clearance'>Calculate the hepatic clearance (deprecated).</h2><span id='topic+calc_hepatic_clearance'></span>

<h3>Description</h3>

<p>This function is included for backward compatibility. It calls
<code><a href="#topic+calc_hep_clearance">calc_hep_clearance</a></code> which
calculates the hepatic clearance in plasma for a well-stirred model
or other type if specified. Based on  Ito and Houston (2004)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_hepatic_clearance(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  default.to.human = FALSE,
  hepatic.model = "well-stirred",
  suppress.messages = FALSE,
  well.stirred.correction = TRUE,
  restrictive.clearance = TRUE,
  adjusted.Funbound.plasma = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_hepatic_clearance_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters
must be specified.</p>
</td></tr>
<tr><td><code id="calc_hepatic_clearance_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must
be specified.</p>
</td></tr>
<tr><td><code id="calc_hepatic_clearance_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_hepatic_clearance_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_steadystate
function, overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_hepatic_clearance_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_hepatic_clearance_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true.</p>
</td></tr>
<tr><td><code id="calc_hepatic_clearance_+3A_hepatic.model">hepatic.model</code></td>
<td>
<p>Model used in calculating hepatic clearance, unscaled,
parallel tube, dispersion, or default well-stirred.</p>
</td></tr>
<tr><td><code id="calc_hepatic_clearance_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not to suppress the output message.</p>
</td></tr>
<tr><td><code id="calc_hepatic_clearance_+3A_well.stirred.correction">well.stirred.correction</code></td>
<td>
<p>Uses correction in calculation of hepatic
clearance for well-stirred model if TRUE for hepatic.model well-stirred.
This assumes clearance relative to amount unbound in whole blood instead of
plasma, but converted to use with plasma concentration.</p>
</td></tr>
<tr><td><code id="calc_hepatic_clearance_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to
1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="calc_hepatic_clearance_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Whether or not to use Funbound.plasma
adjustment if calculating Rblood2plasma.</p>
</td></tr>
<tr><td><code id="calc_hepatic_clearance_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to parameterize_steadystate if
parameters is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Hepatic Clearance</code></td>
<td>
<p>Units of L/h/kg BW.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>References</h3>

<p>Ito, K., &amp; Houston, J. B. (2004). &quot;Comparison of the use of liver models for 
predicting drug clearance using in vitro kinetic data from hepatic microsomes 
and isolated hepatocytes.&quot; Pharmaceutical Tesearch, 21(5), 785-792.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
calc_hep_clearance(chem.name="Ibuprofen",hepatic.model='unscaled')
calc_hep_clearance(chem.name="Ibuprofen",well.stirred.correction=FALSE)


</code></pre>

<hr>
<h2 id='calc_ionization'>Calculate the ionization.</h2><span id='topic+calc_ionization'></span>

<h3>Description</h3>

<p>This function calculates the ionization of a compound at a given pH. The 
pKa's are either entered as parameters or taken from a specific compound in
the package. The arguments pKa_Donor and pKa_Accept may be single numbers, characters, or 
vectors. We support characters because there are many instances with multiple 
predicted values and all those values can be included by concatenating with 
commas (for example, pKa_Donor = &quot;8.1,8.6&quot;. Finally, pka_Donor and pKa_Accept 
may be vectors of characters representing different chemicals or instances of
chemical parameters to allow for uncertainty analysis. A null value for
pKa_Donor or pKa_Accept is interpretted as no argument provided, while NA 
is taken as no equlibria
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_ionization(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  pH = NULL,
  pKa_Donor = NULL,
  pKa_Accept = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_ionization_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="calc_ionization_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="calc_ionization_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_ionization_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from a parameterize_MODEL function,
overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_ionization_+3A_ph">pH</code></td>
<td>
<p>pH where ionization is evaluated.</p>
</td></tr>
<tr><td><code id="calc_ionization_+3A_pka_donor">pKa_Donor</code></td>
<td>
<p>Compound H dissociation equilibirum constant(s).
Overwrites chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_ionization_+3A_pka_accept">pKa_Accept</code></td>
<td>
<p>Compound H association equilibirum constant(s).
Overwrites chem.name and chem.cas.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The fractions are calculated by determining the coefficients for each
species and dividing the particular species by the sum of all three.  The
positive, negative and zwitterionic/neutral coefficients are given by:
</p>
<p style="text-align: center;"><code class="reqn">zwitter/netural = 1</code>
</p>
 <p style="text-align: center;"><code class="reqn">for(i in 1:pkabove) negative = negative +
10^(i * pH - pKa1 - ... - pKai)</code>
</p>
 <p style="text-align: center;"><code class="reqn">for(i in 1:pkbelow) positive =
positive + 10^(pKa1 + ... + pKai - i * pH)</code>
</p>
<p> where i begins at 1 and ends at
the number of points above(for negative) or below(for positive) the
neutral/zwitterionic range.  The neutral/zwitterionic range is either the pH
range between 2 pKa's where the number of acceptors above is equal to the
number of donors below, everything above the pKa acceptors if there are no
donors, or everything below the pKa donors if there are no acceptors.  Each
of the terms in the sums represent a different ionization.
</p>


<h3>Value</h3>

<table>
<tr><td><code>fraction_neutral</code></td>
<td>
<p>fraction of compound neutral</p>
</td></tr>
<tr><td><code>fraction_charged</code></td>
<td>
<p>fraction of compound charged</p>
</td></tr>
<tr><td><code>fraction_negative</code></td>
<td>
<p>fraction of compound negative</p>
</td></tr>
<tr><td><code>fraction_positive</code></td>
<td>
<p>fraction of compound positive</p>
</td></tr>
<tr><td><code>fraction_zwitter</code></td>
<td>
<p>fraction of compound zwitterionic</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Pearce and John Wambaugh
</p>


<h3>References</h3>

<p>Pearce, Robert G., et al. &quot;Evaluation and calibration of
high-throughput predictions of chemical distribution to tissues.&quot; Journal of
Pharmacokinetics and Pharmacodynamics 44.6 (2017): 549-565.
</p>
<p>Strope, Cory L., et al. &quot;High-throughput in-silico prediction of ionization 
equilibria for pharmacokinetic modeling.&quot; Science of The Total Environment 
615 (2018): 150-160.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Donor pKa's 9.78,10.39 -- Should be almost all neutral at plasma pH:
out &lt;- calc_ionization(chem.name='bisphenola',pH=7.4)
print(out)
out[["fraction_neutral"]]==max(unlist(out))

# Donor pKa's 9.78,10.39 -- Should be almost all negative (anion) at higher pH:
out &lt;- calc_ionization(chem.name='bisphenola',pH=11)
print(out)
out[["fraction_negative"]]==max(unlist(out))

# Fictitious compound, should be almost all all negative (anion):
out &lt;- calc_ionization(pKa_Donor=8,pKa_Accept="1,4",pH=9)
print(out)
out[["fraction_negative"]]&gt;0.9

# Donor pKa 6.54 -- Should be mostly negative (anion):
out &lt;- calc_ionization(chem.name='Acephate',pH=7)
print(out)
out[["fraction_negative"]]==max(unlist(out))

#Acceptor pKa's "9.04,6.04"  -- Should be almost all positive (cation) at plasma pH:
out &lt;- calc_ionization(chem.cas="145742-28-5",pH=7.4)
print(out)
out[["fraction_positive"]]==max(unlist(out))

#Fictious Zwitteron:
out &lt;- calc_ionization(pKa_Donor=6,pKa_Accept="8",pH=7.4)
print(out)
out[["fraction_zwitter"]]==max(unlist(out))

</code></pre>

<hr>
<h2 id='calc_kair'>Calculate air:matrix partition coefficients</h2><span id='topic+calc_kair'></span>

<h3>Description</h3>

<p>This function uses the methods colleced by Linakis et al. (2020) to calculate
air partition coefficients for blood, water, and mucus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_kair(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  adjusted.Funbound.plasma = TRUE,
  default.to.human = FALSE,
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_kair_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_kair_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_kair_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_kair_+3A_parameters">parameters</code></td>
<td>
<p>Parameters from the appropriate parameterization function
for the model indicated by argument model. Can include parameters &quot;logHenry&quot;
and &quot;body_temp&quot;, but if not included standard values are looked up from httk tables.</p>
</td></tr>
<tr><td><code id="calc_kair_+3A_species">species</code></td>
<td>
<p>Species used for body temperature, defaults to &quot;Human&quot;</p>
</td></tr>
<tr><td><code id="calc_kair_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses Pearce et al. (2017) lipid binding adjustment
for Funbound.plasma (which impacts partition coefficients) when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="calc_kair_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing species-specific values with human values if
TRUE (default is FALSE).</p>
</td></tr>
<tr><td><code id="calc_kair_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output messages are suppressed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The blood:air partition coefficient (PB:A) was calculated as 
P<sub>B:A</sub> = P<sub>B:A</sub> * R<sub>B:P</sub> / f<sub>up</sub>
where P_B:A is the blood:air partition, RB:P is the blood:plasma partition
ratio, fup is the fraction unbound in the plasma, and
P_W:A is the water:air partition coefficient:
R * T<sub>body</sub> / HLC / P
where R is the gas constant (8.314 J/mol/K), T_body is the 
species-specific body temperature (K) from <code><a href="#topic+physiology.data">physiology.data</a></code>,
HLC is the Henry's Law Constant (atm*m^3 / mol), and P is conversion factor 
from atmospheres to Pascals (1 atm = 101325 Pa).
</p>
<p>In the isopropanol PBTK
model published by Clewell et al. (2001) it was noted that certain 
chemicals are likely to be absorbed into the mucus or otherwise
trapped in the upper respiratory tract (URT). Following Scott (2014), 
the air:mucus partition coefficient (PA:M) calculated as
log<sub>10</sub>(1/K<sub>water2air</sub>) - (log<sub>10</sub>(P<sub>ow</sub>) - 1) * 0.524
where Pow is the octanol/water partition coefficient
</p>


<h3>Value</h3>

<p>A named list containing the blood:air, water:air, and mucus:air 
partition coefficients
</p>


<h3>Author(s)</h3>

<p>John Wambaugh and Matt Linakis
</p>


<h3>References</h3>

<p>Linakis, Matthew W., et al. &quot;Development and evaluation of a 
high throughput inhalation model for organic chemicals.&quot; Journal of exposure 
Science &amp; Environmental Epidemiology 30.5 (2020): 866-877.
</p>
<p>Clewell III, Harvey J., et al. &quot;Development of a physiologically based 
pharmacokinetic model of isopropanol and its metabolite acetone.&quot; 
Toxicological Sciences 63.2 (2001): 160-172.
</p>
<p>Scott, John W., et al. &quot;Tuning to odor solubility and sorption pattern in 
olfactory epithelial responses.&quot; Journal of Neuroscience 34.6 (2014): 
2025-2036.
</p>

<hr>
<h2 id='calc_krbc2pu'>Back-calculates the Red Blood Cell to Unbound Plasma Partition Coefficient</h2><span id='topic+calc_krbc2pu'></span>

<h3>Description</h3>

<p>Given an observed ratio of chemical concentration in blood to plasma, this
function calculates a Red Blood Cell to unbound plasma (Krbc2pu) partition
coefficient that would be consistent with that observation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_krbc2pu(
  Rb2p,
  Funbound.plasma,
  hematocrit = NULL,
  default.to.human = FALSE,
  species = "Human",
  suppress.messages = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_krbc2pu_+3A_rb2p">Rb2p</code></td>
<td>
<p>The chemical blood:plasma concentration ratop</p>
</td></tr>
<tr><td><code id="calc_krbc2pu_+3A_funbound.plasma">Funbound.plasma</code></td>
<td>
<p>The free fraction of chemical in the presence of 
plasma protein
Rblood2plasma.</p>
</td></tr>
<tr><td><code id="calc_krbc2pu_+3A_hematocrit">hematocrit</code></td>
<td>
<p>Overwrites default hematocrit value in calculating
Rblood2plasma.</p>
</td></tr>
<tr><td><code id="calc_krbc2pu_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true.</p>
</td></tr>
<tr><td><code id="calc_krbc2pu_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_krbc2pu_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Determine whether to display certain usage
feedback.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The red blood cell to unbound chemical in plasma partition coefficient.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>References</h3>

<p>Pearce, Robert G., et al. &quot;Evaluation and calibration of high-throughput 
predictions of chemical distribution to tissues.&quot; Journal of 
pharmacokinetics and pharmacodynamics 44.6 (2017): 549-565.
</p>
<p>Ruark, Christopher D., et al. &quot;Predicting passive and active tissue: plasma 
partition coefficients: interindividual and interspecies variability.&quot; 
Journal of pharmaceutical sciences 103.7 (2014): 2189-2198.
</p>

<hr>
<h2 id='calc_ma'>Calculate the membrane affinity</h2><span id='topic+calc_ma'></span>

<h3>Description</h3>

<p>Membrane affinity (MA) is the membrane:water partition coefficient. MA
chacterizes chemical partitioning into membranes formed
from neutral phospholipids (K<sub>nPL</sub>). 
Pearce et al. (2017) compared five different methods for predicting 
membrane affinity using measured data for 59 compounds. The method of
Yun and Edgington (2013) was identified as the best: 
MA = 10^(1.294 + 0.304 * log<sub>10</sub>(P<sub>ow</sub>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_ma(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_ma_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_ma_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_ma_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_ma_+3A_parameters">parameters</code></td>
<td>
<p>Parameters from the appropriate parameterization function
for the model indicated by argument model</p>
</td></tr>
<tr><td><code id="calc_ma_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric fraction unpbound in plasma between zero and one
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Pearce, Robert G., et al. &quot;Evaluation and calibration of high-throughput 
predictions of chemical distribution to tissues.&quot; Journal of pharmacokinetics 
and pharmacodynamics 44.6 (2017): 549-565.
</p>
<p>Yun, Y. E., and A. N. Edginton. &quot;Correlation-based prediction of 
tissue-to-plasma partition coefficients using readily available input 
parameters.&quot; Xenobiotica 43.10 (2013): 839-852.
</p>

<hr>
<h2 id='calc_maternal_bw'>Calculate maternal body weight</h2><span id='topic+calc_maternal_bw'></span>

<h3>Description</h3>

<p>This function initializes the parameters needed in the functions
solve_fetal_pbtk by calling solve_pbtk and adding additional parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_maternal_bw(week = 12)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_maternal_bw_+3A_week">week</code></td>
<td>
<p>Gestational week</p>
</td></tr>
</table>


<h3>Details</h3>

<p>BW &lt;- params$pre_pregnant_BW + 
params$BW_cubic_theta1 * tw + 
params$BW_cubic_theta2 * tw^2 + 
params$BW_cubic_theta3 * tw^3
</p>


<h3>Value</h3>

<table>
<tr><td><code>BW</code></td>
<td>
<p>Maternal Body Weight, kg.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Kapraun DF, Wambaugh JF, Setzer RW, Judson RS (2019).
&ldquo;Empirical models for anatomical and physiological changes in a human mother and fetus during pregnancy and gestation.&rdquo;
<em>PLOS ONE</em>, <b>14</b>(5), 1-56.
<a href="https://doi.org/10.1371/journal.pone.0215906">doi:10.1371/journal.pone.0215906</a>.
</p>

<hr>
<h2 id='calc_mc_css'>Distribution of chemical steady state concentration with uncertainty and variability</h2><span id='topic+calc_mc_css'></span>

<h3>Description</h3>

<p>For a given chemical and fixed dose rate this function determines a 
distribution of steady-state concentrations reflecting measurement uncertainty
an population variability. Uncertainty and variability are simulated via the
Monte Carlo method &ndash; many sets of model parameters are drawn according to
probability distributions described in 
Ring et al. (2017) (<a href="https://doi.org/10.1016/j.envint.2017.06.004">doi:10.1016/j.envint.2017.06.004</a>)
for human variability and Wambaugh et al. (2019) 
(<a href="https://doi.org/10.1093/toxsci/kfz205">doi:10.1093/toxsci/kfz205</a>) 
for measurement uncertainty. Monte Carlo samples are generated by the 
function <code><a href="#topic+create_mc_samples">create_mc_samples</a></code>. To allow rapid application of the 
Monte Carlo 
method we make use of analytical solutions for the steady-state concentration
for a particular model via a given route (when available) as opposed to
solving the model numerically (that is, using differential equations).
For each sample of the Monte Carlo method (as specified by argument
<code>samples</code>) the parameters for the analytical solution are varied. An
ensemble of steady-state predictions are produced, though by default only
the quantiles specified by argument <code>which.quantile</code> are provided. If 
the full set of predicted values are desired use set the argument 
<code>return.samples</code> to <code>TRUE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_mc_css(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  samples = 1000,
  which.quantile = 0.95,
  species = "Human",
  daily.dose = 1,
  suppress.messages = FALSE,
  model = "3compartmentss",
  httkpop = TRUE,
  invitrouv = TRUE,
  calcrb2p = TRUE,
  censored.params = list(),
  vary.params = list(),
  return.samples = FALSE,
  tissue = NULL,
  concentration = "plasma",
  output.units = "mg/L",
  invitro.mc.arg.list = NULL,
  httkpop.generate.arg.list = list(method = "direct resampling"),
  convert.httkpop.arg.list = NULL,
  parameterize.arg.list = NULL,
  calc.analytic.css.arg.list = NULL,
  Caco2.options = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_mc_css_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_parameters">parameters</code></td>
<td>
<p>Parameters from the appropriate parameterization function
for the model indicated by argument model</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_samples">samples</code></td>
<td>
<p>Number of samples generated in calculating quantiles.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_which.quantile">which.quantile</code></td>
<td>
<p>Which quantile from Monte Carlo simulation is
requested. Can be a vector.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).  Species must be set to &quot;Human&quot; to run httkpop model.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose, mg/kg BW.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not to suppress output message.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_model">model</code></td>
<td>
<p>Model used in calculation,'gas_pbtk' for the gas pbtk model, 
'pbtk' for the multiple compartment model,
'3compartment' for the three compartment model, '3compartmentss' for 
the three compartment steady state model, and '1compartment' for one 
compartment model. This only applies when httkpop=TRUE and species=&quot;Human&quot;,
otherwise '3compartmentss' is used.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_httkpop">httkpop</code></td>
<td>
<p>Whether or not to use population generator and sampler from
httkpop.  This is overwrites censored.params and vary.params and is only for
human physiology.  Species must also be set to 'Human'.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_invitrouv">invitrouv</code></td>
<td>
<p>Logical to indicate whether to include in vitro parameters
in uncertainty and variability analysis</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_calcrb2p">calcrb2p</code></td>
<td>
<p>Logical determining whether or not to recalculate the 
chemical ratio of blood to plasma</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_censored.params">censored.params</code></td>
<td>
<p>The parameters listed in censored.params are sampled
from a normal distribution that is censored for values less than the limit
of detection (specified separately for each parameter). This argument should
be a list of sublists. Each sublist is named for a parameter in
&quot;parameters&quot; and contains two elements: &quot;CV&quot; (coefficient of variation) and
&quot;LOD&quot; (limit of detection, below which parameter values are censored. New
values are sampled with mean equal to the value in &quot;parameters&quot; and standard
deviation equal to the mean times the CV.  Censored values are sampled on a
uniform distribution between 0 and the limit of detection. Not used with
httkpop model.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_vary.params">vary.params</code></td>
<td>
<p>The parameters listed in vary.params are sampled from a
normal distribution that is truncated at zero. This argument should be a
list of coefficients of variation (CV) for the normal distribution. Each
entry in the list is named for a parameter in &quot;parameters&quot;. New values are
sampled with mean equal to the value in &quot;parameters&quot; and standard deviation
equal to the mean times the CV. Not used with httkpop model.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_return.samples">return.samples</code></td>
<td>
<p>Whether or not to return the vector containing the
samples from the simulation instead of the selected quantile.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_tissue">tissue</code></td>
<td>
<p>Desired steady state tissue concentration. Default is of NULL
typically gives whole body plasma concentration.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_concentration">concentration</code></td>
<td>
<p>Desired concentration type: 'blood','tissue', or default 
'plasma'. In the case that the concentration is for plasma, selecting &quot;blood&quot;
will use the blood:plasma ratio to estimate blood concentration. In the case
that the argument 'tissue' specifies a particular tissue of the body, 
concentration defaults to 'tissue' &ndash; that is, the concentration in the 
If cocentration is set to 'blood' or 'plasma' and 'tissue' specifies a
specific tissue then the value returned is for the plasma or blood in that
specific tissue.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_output.units">output.units</code></td>
<td>
<p>Plasma concentration units, either uM or default mg/L.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_invitro.mc.arg.list">invitro.mc.arg.list</code></td>
<td>
<p>List of additional parameters passed to 
<code><a href="#topic+invitro_mc">invitro_mc</a></code></p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_httkpop.generate.arg.list">httkpop.generate.arg.list</code></td>
<td>
<p>Additional parameters passed to 
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_convert.httkpop.arg.list">convert.httkpop.arg.list</code></td>
<td>
<p>Additional parameters passed to the 
convert_httkpop_* function for the model.</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_parameterize.arg.list">parameterize.arg.list</code></td>
<td>
<p>A list of arguments to be passed to the model
parameterization function (that is, parameterize_MODEL) corresponding to
argument &quot;model&quot;. (Defaults to NULL.)</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_calc.analytic.css.arg.list">calc.analytic.css.arg.list</code></td>
<td>
<p>Additional parameters passed to</p>
</td></tr>
<tr><td><code id="calc_mc_css_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>Arguments describing how to handle Caco2 absorption data
that are passed to <code><a href="#topic+invitro_mc">invitro_mc</a></code> and the parameterize_[MODEL] 
functions.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.
</p>
<p><code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The chemical-specific steady-state concentration for a dose rate of 
1 mg/kg body weight/day can be used for in <em>in vitro-in vivo</em> 
extrapolation (IVIVE) of
a bioactive <em>in vitro</em> concentration by dividing the <em>in vitro</em>
concentration
by the steady-state concentration to predict a dose rate (mg/kg/day) that
would produce that concentration in plasma. Using quantiles of the 
distribution (such as the upper 95th percentile) allow incorporation of 
uncertainty and variability into IVIVE.
</p>
<p>Reverse Dosimetry Toxicodynamic IVIVE
</p>
<p><img src="../help/figures/ivive.png" width="60%" alt="Reverse Dosimetry Toxicodynamic IVIVE" />

</p>
<p>Figure from Breen et al. (2021) (<a href="https://doi.org/10.1080/17425255.2021.1935867">doi:10.1080/17425255.2021.1935867</a>) shows 
reverse dosimetry toxicodynamic IVIVE. Equivalent external dose is
determined by solving the TK model in reverse by deriving the external dose
(that is, TK model input) that produces a specified internal concentration 
(that is, TK model output). Reverse dosimetry and IVIVE using HTTK relies on 
the linearity of the models. We calculate a scaling factor to relate 
<em>in vitro</em> 
concentrations (uM) to administered equivalent doses (AED). The scaling 
factor is the inverse of the steady state plasma concentration (Css) 
predicted for a 1 mg/kg/day exposure dose rate. We use Monte Carlo to 
simulate variability and propagate uncertainty; for example, to calculate an 
upper 95th percentile Css,95 for individuals who get higher plasma 
concentrations from the same exposure.
</p>
<p>The Monte Carlo methods used here were recently updated and described by
Breen et al. (submitted).
</p>
<p>httk-pop is used only for humans. For non-human species biological 
variability is simulated by drawing parameters from uncorellated log-normal
distributions. 
</p>
<p>Chemical-specific httk data are available primarily for human and for a few
hundred chemicals in rat. All in silico predictions are for human. Thus, 
when species is specified as rabbit, dog, or mouse, the user can choose
to set the argument <code>default.to.human</code> to <code>TRUE</code> so that this 
function uses the
appropriate physiological data (volumes and flows) but substitutes human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>
<p>If the argument <code>tissue</code> is used, the steady-state concentration in that
tissue, if available, is provided. If that tissue is included in the model
used (specified by arguement <code>model</code>) then the actual tissue 
concentration is
provided. Otherwise, the tissue-specific partition coefficient is used to
estimate the concentration from the plasma. 
</p>
<p>The six sets of plausible IVIVE
assumptions identified by Honda et al. (2019) 
(<a href="https://doi.org/10.1371/journal.pone.0217564">doi:10.1371/journal.pone.0217564</a>) 
are: </p>

<table>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: right;"> <em>in vivo</em> Conc. </td><td style="text-align: right;"> Metabolic Clearance </td><td style="text-align: right;"> Bioactive Chemical
Conc. </td><td style="text-align: right;"> TK Statistic Used* </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda1 </td><td style="text-align: right;"> Veinous (Plasma) </td><td style="text-align: right;">
Restrictive </td><td style="text-align: right;"> Free </td><td style="text-align: right;"> Mean Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda2 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;">
Restrictive </td><td style="text-align: right;"> Free </td><td style="text-align: right;"> Max Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda3 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;">
Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Mean Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda4 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;">
Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Max Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda5 </td><td style="text-align: right;"> Target Tissue </td><td style="text-align: right;">
Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Mean Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda6 </td><td style="text-align: right;"> Target Tissue
</td><td style="text-align: right;"> Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Max Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>
<p> *Assumption is
currently ignored because analytical steady-state solutions are currently
used by this function.
</p>


<h3>Value</h3>

<p>Quantiles (specified by which.quantile) of the distribution of plasma
steady-stae concentration (Css) from the Monte Carlo simulation
</p>


<h3>Author(s)</h3>

<p>Caroline Ring, Robert Pearce, John Wambaugh, Miyuki Breen, and Greg Honda
</p>


<h3>References</h3>

<p>Wambaugh, John F., et al. &quot;Toxicokinetic triage for 
environmental chemicals.&quot; Toxicological Sciences 147.1 (2015): 55-67.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>
<p>Honda GS, Pearce RG, Pham LL, Setzer RW, Wetmore BA, Sipes NS, Gilbert J, Franz B, Thomas RS, Wambaugh JF (2019).
&ldquo;Using the concordance of in vitro and in vivo data to evaluate extrapolation assumptions.&rdquo;
<em>PloS one</em>, <b>14</b>(5), e0217564. 
</p>
<p>Rowland M, Benet LZ, Graham GG (1973).
&ldquo;Clearance concepts in pharmacokinetics.&rdquo;
<em>Journal of pharmacokinetics and biopharmaceutics</em>, <b>1</b>(2), 123&ndash;136.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code>
</p>
<p><code><a href="#topic+create_mc_samples">create_mc_samples</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Set the number of samples (NSAMP) low for rapid testing, increase NSAMP 
# for more stable results. Default value is 1000:
NSAMP = 10

# Basic in vitro - in vivo extrapolation with httk, convert 3 uM in vitro
# concentration of chemical with CAS 2451-62-9 to mg/kg/day:
set.seed(1234)
3/calc_mc_css(chem.cas="2451-62-9", samples=NSAMP, output.units="uM")
# The significant digits should give the same answer as:
set.seed(1234)
calc_mc_oral_equiv(chem.cas="2451-62-9", conc=3, samples=NSAMP)  

 set.seed(1234)
 calc_mc_css(chem.name='Bisphenol A', output.units='uM',
             samples=NSAMP, return.samples=TRUE)

 set.seed(1234)
 calc_mc_css(chem.name='Bisphenol A', output.units='uM',
             samples=NSAMP,
             httkpop.generate.arg.list=list(method='vi'))
                          
 # The following example should result in an error since we do not 
 # estimate tissue partitioning with '3compartmentss'.                         
 set.seed(1234)
 try(calc_mc_css(chem.name='2,4-d', which.quantile=.9,
             samples=NSAMP,
             httkpop=FALSE, tissue='heart'))
 
# But heart will work with PBTK, even though it's lumped since we estimate
# a partition coefficient before lumping:
 set.seed(1234)
 calc_mc_css(chem.name='2,4-d', model='pbtk',
             samples=NSAMP,
             which.quantile=.9, httkpop=FALSE, tissue='heart')

 set.seed(1234)
 calc_mc_css(chem.cas = "80-05-7", which.quantile = 0.5,
             output.units = "uM", samples = NSAMP,
             httkpop.generate.arg.list=list(method='vi', gendernum=NULL, 
             agelim_years=NULL, agelim_months=NULL,
             weight_category = c("Underweight","Normal","Overweight","Obese")))

 params &lt;- parameterize_pbtk(chem.cas="80-05-7")
 set.seed(1234)
 calc_mc_css(parameters=params,model="pbtk", samples=NSAMP)

 set.seed(1234)
 # Standard HTTK Monte Carlo 
 calc_mc_css(chem.cas="90-43-7", model="pbtk", samples=NSAMP)
 set.seed(1234)
 # HTTK Monte Carlo with no measurement uncertainty (pre v1.10.0):
 calc_mc_css(chem.cas="90-43-7",
 model="pbtk",
 samples=NSAMP,
 invitro.mc.arg.list = list(
   adjusted.Funbound.plasma = TRUE,
   poormetab = TRUE, 
   fup.censored.dist = FALSE, 
   fup.lod = 0.01, 
   fup.meas.cv = 0.0, 
   clint.meas.cv = 0.0, 
   fup.pop.cv = 0.3, 
   clint.pop.cv = 0.3))

 # HTTK Monte Carlo with no HTTK-Pop physiological variability):
 set.seed(1234)
 calc_mc_css(chem.cas="90-43-7",model="pbtk",samples=NSAMP,httkpop=FALSE)

 # HTTK Monte Carlo with no in vitro uncertainty and variability):
 set.seed(1234)
 calc_mc_css(chem.cas="90-43-7",model="pbtk",samples=NSAMP,invitrouv=FALSE)

 # HTTK Monte Carlo with no HTTK-Pop and no in vitro uncertainty and variability):
 set.seed(1234)
 calc_mc_css(chem.cas="90-43-7" ,model="pbtk",
             samples=NSAMP, httkpop=FALSE, invitrouv=FALSE)

 # Should be the same as the mean result:
 calc_analytic_css(chem.cas="90-43-7",model="pbtk",output.units="mg/L")

 # HTTK Monte Carlo using basic Monte Carlo sampler:
 set.seed(1234)
 calc_mc_css(chem.cas="90-43-7",
             model="pbtk",
             samples=NSAMP,
             httkpop=FALSE,
             invitrouv=FALSE,
             vary.params=list(Pow=0.3))
 
# We can also use the Monte Carlo functions by passing a table
# where each row represents a different Monte Carlo draw of parameters:
p &lt;- create_mc_samples(chem.cas="80-05-7")
# Use data.table for steady-state plasma concentration (Css) Monte Carlo:
calc_mc_css(parameters=p)
# Using the same table gives the same answer:
calc_mc_css(parameters=p)
# Use Css for 1 mg/kg/day for simple reverse toxicokinetics 
# in Vitro-In Vivo Extrapolation to convert 15 uM to mg/kg/day:
15/calc_mc_css(parameters=p, output.units="uM")
# Can do the same with calc_mc_oral_equiv:
calc_mc_oral_equiv(15, parameters=p)


</code></pre>

<hr>
<h2 id='calc_mc_oral_equiv'>Calculate Monte Carlo Oral Equivalent Dose</h2><span id='topic+calc_mc_oral_equiv'></span>

<h3>Description</h3>

<p>This function converts a chemical plasma concetration to an oral adminstered
equivalent
dose (AED) using a concentration obtained from <code><a href="#topic+calc_mc_css">calc_mc_css</a></code>.  
This function uses reverse dosimetry-based 
'<em>in vitro-in vivo</em> extrapolation (IVIVE) 
for high throughput risk screening.  The user can input the
chemical and <em>in vitro</em> bioactive concentration, select the TK
model, and then automatically predict the <em>in vivo</em> AED
which would produce a body concentration equal to
the <em>in vitro</em> bioactive concentration. This function relies on the
Monte Carlo method (via funcion <code><a href="#topic+create_mc_samples">create_mc_samples</a></code> to simulate
both uncertainty and variability so that the result is a distribution of
equivalent doses, from which we provide specific quantiles
(specified by <code>which.quantile</code>), though the full set of predictions can
be obtained by setting <code>return.samples</code> to <code>TRUE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_mc_oral_equiv(
  conc,
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  which.quantile = 0.95,
  species = "Human",
  input.units = "uM",
  output.units = "mgpkgpday",
  suppress.messages = FALSE,
  return.samples = FALSE,
  restrictive.clearance = TRUE,
  bioactive.free.invivo = FALSE,
  tissue = NULL,
  concentration = "plasma",
  IVIVE = NULL,
  model = "3compartmentss",
  Caco2.options = list(),
  calc.analytic.css.arg.list = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_mc_oral_equiv_+3A_conc">conc</code></td>
<td>
<p>Bioactive in vitro concentration in units of uM.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the CAS number or the chemical name must be
specified.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_parameters">parameters</code></td>
<td>
<p>Parameters from the appropriate parameterization function
for the model indicated by argument model</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_which.quantile">which.quantile</code></td>
<td>
<p>Which quantile from Monte Carlo steady-state
simulation (<code><a href="#topic+calc_mc_css">calc_mc_css</a></code>) is requested. Can be a vector. Note that 95th
concentration quantile is the same population as the 5th dose quantile.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_input.units">input.units</code></td>
<td>
<p>Units of given concentration, default of uM but can also
be mg/L.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_output.units">output.units</code></td>
<td>
<p>Units of dose, default of 'mgpkgpday' for mg/kg BW/ day or
'umolpkgpday' for umol/ kg BW/ day.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Suppress text messages.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_return.samples">return.samples</code></td>
<td>
<p>Whether or not to return the vector containing the
samples from the simulation instead of the selected quantile.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to
1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_bioactive.free.invivo">bioactive.free.invivo</code></td>
<td>
<p>If FALSE (default), then the total concentration is treated
as bioactive in vivo. If TRUE, the the unbound (free) plasma concentration is treated as 
bioactive in vivo. Only works with tissue = NULL in current implementation.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_tissue">tissue</code></td>
<td>
<p>Desired steady state tissue concentration. Default is of NULL
typically gives whole body plasma concentration.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_concentration">concentration</code></td>
<td>
<p>Desired concentration type: 'blood','tissue', or default 
'plasma'. In the case that the concentration is for plasma, selecting &quot;blood&quot;
will use the blood:plasma ratio to estimate blood concentration. In the case
that the argument 'tissue' specifies a particular tissue of the body, 
concentration defaults to 'tissue' &ndash; that is, the concentration in the 
If cocentration is set to 'blood' or 'plasma' and 'tissue' specifies a
specific tissue then the value returned is for the plasma or blood in that
specific tissue.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_ivive">IVIVE</code></td>
<td>
<p>Honda et al. (2019) identified six plausible sets of
assumptions for <em>in vitro-in vivo</em> extrapolation (IVIVE) assumptions.
Argument may be set to &quot;Honda1&quot; through &quot;Honda6&quot;. If used, this function
overwrites the tissue, restrictive.clearance, and bioactive.free.invivo arguments.
See Details below for more information.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_model">model</code></td>
<td>
<p>Model used in calculation,'gas_pbtk' for the gas pbtk model, 
'pbtk' for the multiple compartment model,
'3compartment' for the three compartment model, '3compartmentss' for 
the three compartment steady state model, and '1compartment' for one 
compartment model. This only applies when httkpop=TRUE and species=&quot;Human&quot;,
otherwise '3compartmentss' is used.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_calc.analytic.css.arg.list">calc.analytic.css.arg.list</code></td>
<td>
<p>A list of options to pass to the analytic steady-state calculation function.
This includes 'restrictive.clearance', 'bioactive.free.invivo', 'IVIVE',
'wellstirred.correction', and 'adjusted.Funbound.plasma'.</p>
</td></tr>
<tr><td><code id="calc_mc_oral_equiv_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to <code><a href="#topic+calc_mc_css">calc_mc_css</a></code> for httkpop and
variance of parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The chemical-specific steady-state concentration for a dose rate of 
1 mg/kg body weight/day can be used for in IVIVE of
a bioactive <em>in vitro</em> concentration by dividing the <em>in vitro</em>
concentration
by the steady-state concentration to predict a dose rate (mg/kg/day) that
would produce that concentration in plasma. Using quantiles of the 
distribution (such as the upper 95th percentile) allow incorporation of 
uncertainty and variability into IVIVE.
</p>
<p>This approach relies on thelinearity
of the models to calculate a scaling factor to relate in vitro
concentrations (uM) with AED. The scaling factor is the
inverse of the steady-state plasma concentration (Css) predicted
for a 1 mg/kg/day exposure dose rate
where <em>in vitro</em> concentration [X] and Css must be in the same
units. Note that it is typical for <em>in vitro</em> concentrations to be
reported in units of uM and Css in units of mg/L, in which case
one must be converted to the other. 
</p>
<p>Reverse Dosimetry Toxicodynamic IVIVE
</p>
<p><img src="../help/figures/ivive.png" width="60%" alt="Reverse Dosimetry Toxicodynamic IVIVE" />

</p>
<p>Figure from Breen et al. (2021) (<a href="https://doi.org/10.1080/17425255.2021.1935867">doi:10.1080/17425255.2021.1935867</a>) shows 
reverse dosimetry toxicodynamic IVIVE. Equivalent external dose is
determined by solving the TK model in reverse by deriving the external dose
(that is, TK model input) that produces a specified internal concentration 
(that is, TK model output). Reverse dosimetry and IVIVE using HTTK relies on 
the linearity of the models. We calculate a scaling factor to relate 
<em>in vitro</em> 
concentrations (uM) to AEDs. The scaling 
factor is the inverse of the Css 
predicted for a 1 mg/kg/day exposure dose rate. We use Monte Carlo to 
simulate variability and propagate uncertainty; for example, to calculate an 
upper 95th percentile Css,95 for individuals who get higher plasma 
concentrations from the same exposure.
</p>
<p>The Monte Carlo methods used here were recently updated and described by
Breen et al. (submitted).
</p>
<p>All arguments after httkpop only apply if httkpop is set to TRUE and species
to &quot;Human&quot;.
</p>
<p>When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitutes human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>
<p>Tissue concentrations are calculated for the pbtk model with oral infusion
dosing.  All tissues other than gut, liver, and lung are the product of the
steady state plasma concentration and the tissue to plasma partition
coefficient.
</p>
<p>The six sets of plausible IVIVE
assumptions identified by Honda et al. (2019) 
(<a href="https://doi.org/10.1371/journal.pone.0217564">doi:10.1371/journal.pone.0217564</a>) 
are: </p>

<table>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: right;"> <em>in vivo</em> Conc. </td><td style="text-align: right;"> Metabolic Clearance </td><td style="text-align: right;"> Bioactive Chemical
Conc. </td><td style="text-align: right;"> TK Statistic Used* </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda1 </td><td style="text-align: right;"> Veinous (Plasma) </td><td style="text-align: right;">
Restrictive </td><td style="text-align: right;"> Free </td><td style="text-align: right;"> Mean Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda2 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;">
Restrictive </td><td style="text-align: right;"> Free </td><td style="text-align: right;"> Max Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda3 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;">
Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Mean Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda4 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;">
Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Max Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda5 </td><td style="text-align: right;"> Target Tissue </td><td style="text-align: right;">
Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Mean Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda6 </td><td style="text-align: right;"> Target Tissue
</td><td style="text-align: right;"> Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Max Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>
<p> *Assumption is
currently ignored because analytical steady-state solutions are currently
used by this function.
</p>


<h3>Value</h3>

<p>Equivalent dose in specified units, default of mg/kg BW/day.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118. 
</p>
<p>Wetmore BA, Wambaugh JF, Allen B, Ferguson SS, Sochaski MA, Setzer RW, Houck KA, Strope CL, Cantwell K, Judson RS, others (2015).
&ldquo;Incorporating high-throughput exposure predictions with dosimetry-adjusted in vitro bioactivity to inform chemical toxicity testing.&rdquo;
<em>Toxicological Sciences</em>, <b>148</b>(1), 121&ndash;136.
</p>
<p>Honda GS, Pearce RG, Pham LL, Setzer RW, Wetmore BA, Sipes NS, Gilbert J, Franz B, Thomas RS, Wambaugh JF (2019).
&ldquo;Using the concordance of in vitro and in vivo data to evaluate extrapolation assumptions.&rdquo;
<em>PloS one</em>, <b>14</b>(5), e0217564.
</p>
<p>Rowland M, Benet LZ, Graham GG (1973).
&ldquo;Clearance concepts in pharmacokinetics.&rdquo;
<em>Journal of pharmacokinetics and biopharmaceutics</em>, <b>1</b>(2), 123&ndash;136.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_mc_css">calc_mc_css</a></code>
</p>
<p><code><a href="#topic+create_mc_samples">create_mc_samples</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Set the number of samples (NSAMP) low for rapid testing, increase NSAMP 
# for more stable results. Default value is 1000:
NSAMP = 10

# Basic in vitro - in vivo extrapolation with httk, convert 0.5 uM in vitro
# concentration of chemical Surinabant to mg/kg/day:
set.seed(1234)
0.5/calc_mc_css(chem.name="Surinabant", samples=NSAMP, output.units="uM")

# The significant digits should give the same answer as:
set.seed(1234)
calc_mc_oral_equiv(chem.name="Surinabant",conc=0.5,samples=NSAMP)  

# Note that we use set.seed to get the same sequence of random numbers for
# the two different function calls (calc_mc_css and calc_mc_oral_equiv)

# The following example should result in an error since we do not 
# estimate tissue partitioning with '3compartmentss'. 
set.seed(1234)                        
try(calc_mc_oral_equiv(0.1, chem.cas="34256-82-1",
                       which.quantile=c(0.05,0.5,0.95),
                       samples=NSAMP,
                       tissue='brain'))
       
set.seed(1234)
calc_mc_oral_equiv(0.1,chem.cas="34256-82-1", model='pbtk',
                   samples=NSAMP,
                   which.quantile=c(0.05,0.5,0.95), tissue='brain')
 
# We can also use the Monte Carlo functions by passing a table
# where each row represents a different Monte Carlo draw of parameters:
p &lt;- create_mc_samples(chem.cas="80-05-7")
# Use data.table for steady-state plasma concentration (Css) Monte Carlo:
calc_mc_css(parameters=p)
# Using the same table gives the same answer:
calc_mc_css(parameters=p)
# Use Css for 1 mg/kg/day for simple reverse toxicokinetics 
# in Vitro-In Vivo Extrapolation to convert 15 uM to mg/kg/day:
15/calc_mc_css(parameters=p, output.units="uM")
# Can do the same with calc_mc_oral_equiv:
calc_mc_oral_equiv(15, parameters=p)


</code></pre>

<hr>
<h2 id='calc_mc_tk'>Conduct multiple TK simulations using Monte Carlo</h2><span id='topic+calc_mc_tk'></span>

<h3>Description</h3>

<p>This function finds the analytical steady state plasma concentration(from
calc_analytic_css) using a monte carlo simulation (monte_carlo).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_mc_tk(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  samples = 1000,
  species = "Human",
  suppress.messages = FALSE,
  model = "pbtk",
  httkpop = TRUE,
  invitrouv = TRUE,
  calcrb2p = TRUE,
  censored.params = list(),
  vary.params = list(),
  return.samples = FALSE,
  tissue = NULL,
  output.units = "mg/L",
  solvemodel.arg.list = list(times = c(0, 0.25, 0.5, 0.75, 1, 1.5, 2, 2.5, 3, 4, 5)),
  Caco2.options = list(),
  invitro.mc.arg.list = NULL,
  httkpop.generate.arg.list = list(method = "direct resampling"),
  convert.httkpop.arg.list = NULL,
  parameterize.arg.list = NULL,
  return.all.sims = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_mc_tk_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the CAS number, parameters, or the chemical name must
be specified.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical parameters, name, or the CAS number
must be specified.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_parameters">parameters</code></td>
<td>
<p>Parameters from parameterize_steadystate. Not used with
httkpop model.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_samples">samples</code></td>
<td>
<p>Number of samples generated in calculating quantiles.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).  Species must be set to &quot;Human&quot; to run httkpop model.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not to suppress output message.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_model">model</code></td>
<td>
<p>Model used in calculation: 'pbtk' for the multiple compartment
model,'3compartment' for the three compartment model, '3compartmentss' for
the three compartment steady state model, and '1compartment' for one
compartment model.  This only applies when httkpop=TRUE and species=&quot;Human&quot;,
otherwise '3compartmentss' is used.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_httkpop">httkpop</code></td>
<td>
<p>Whether or not to use population generator and sampler from
httkpop.  This is overwrites censored.params and vary.params and is only for
human physiology.  Species must also be set to 'Human'.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_invitrouv">invitrouv</code></td>
<td>
<p>Logical to indicate whether to include in vitro parameters
in uncertainty and variability analysis</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_calcrb2p">calcrb2p</code></td>
<td>
<p>Logical determining whether or not to recalculate the 
chemical ratio of blood to plasma</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_censored.params">censored.params</code></td>
<td>
<p>The parameters listed in censored.params are sampled
from a normal distribution that is censored for values less than the limit
of detection (specified separately for each parameter). This argument should
be a list of sub-lists. Each sublist is named for a parameter in
&quot;parameters&quot; and contains two elements: &quot;CV&quot; (coefficient of variation) and
&quot;LOD&quot; (limit of detection, below which parameter values are censored. New
values are sampled with mean equal to the value in &quot;parameters&quot; and standard
deviation equal to the mean times the CV.  Censored values are sampled on a
uniform distribution between 0 and the limit of detection. Not used with
httkpop model.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_vary.params">vary.params</code></td>
<td>
<p>The parameters listed in vary.params are sampled from a
normal distribution that is truncated at zero. This argument should be a
list of coefficients of variation (CV) for the normal distribution. Each
entry in the list is named for a parameter in &quot;parameters&quot;. New values are
sampled with mean equal to the value in &quot;parameters&quot; and standard deviation
equal to the mean times the CV. Not used with httkpop model.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_return.samples">return.samples</code></td>
<td>
<p>Whether or not to return the vector containing the
samples from the simulation instead of the selected quantile.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_tissue">tissue</code></td>
<td>
<p>Desired steady state tissue conentration.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_output.units">output.units</code></td>
<td>
<p>Plasma concentration units, either uM or default mg/L.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_solvemodel.arg.list">solvemodel.arg.list</code></td>
<td>
<p>Additional arguments ultimately passed to 
<code><a href="#topic+solve_model">solve_model</a></code></p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.default = 2,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_invitro.mc.arg.list">invitro.mc.arg.list</code></td>
<td>
<p>List of additional parameters passed to 
<code><a href="#topic+invitro_mc">invitro_mc</a></code></p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_httkpop.generate.arg.list">httkpop.generate.arg.list</code></td>
<td>
<p>Additional parameters passed to 
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_convert.httkpop.arg.list">convert.httkpop.arg.list</code></td>
<td>
<p>Additional parameters passed to the 
convert_httkpop_* function for the model.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_parameterize.arg.list">parameterize.arg.list</code></td>
<td>
<p>Additional parameters passed to the 
parameterize_* function for the model.</p>
</td></tr>
<tr><td><code id="calc_mc_tk_+3A_return.all.sims">return.all.sims</code></td>
<td>
<p>Logical indicating whether to return the results
of all simulations, in addition to the default toxicokinetic statistics</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Monte Carlo methods used here were recently updated and described by
Breen et al. (submitted).
</p>
<p>All arguments after httkpop only apply if httkpop is set to TRUE and species
to &quot;Human&quot;.
</p>
<p>When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>
<p>Tissue concentrations are calculated for the pbtk model with oral infusion
dosing.  All tissues other than gut, liver, and lung are the product of the
steady state plasma concentration and the tissue to plasma partition
coefficient.
</p>
<p>The six sets of plausible <em>in vitro-in vivo</em> extrpolation (IVIVE)
assumptions identified by Honda et al. (2019) 
(<a href="https://doi.org/10.1371/journal.pone.0217564">doi:10.1371/journal.pone.0217564</a>) 
are: </p>

<table>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: right;"> <em>in vivo</em> Conc. </td><td style="text-align: right;"> Metabolic Clearance </td><td style="text-align: right;"> Bioactive Chemical
Conc. </td><td style="text-align: right;"> TK Statistic Used* </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda1 </td><td style="text-align: right;"> Veinous (Plasma) </td><td style="text-align: right;">
Restrictive </td><td style="text-align: right;"> Free </td><td style="text-align: right;"> Mean Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda2 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;">
Restrictive </td><td style="text-align: right;"> Free </td><td style="text-align: right;"> Max Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda3 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;">
Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Mean Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda4 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;">
Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Max Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda5 </td><td style="text-align: right;"> Target Tissue </td><td style="text-align: right;">
Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Mean Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> Honda6 </td><td style="text-align: right;"> Target Tissue
</td><td style="text-align: right;"> Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Max Conc. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>
<p> *Assumption is
currently ignored because analytical steady-state solutions are currently
used by this function.
</p>


<h3>Value</h3>

<p>If return.all.sims == FALSE (default) a list with:
</p>
<table>
<tr><td><code>means</code></td>
<td>
<p>The mean concentration for each model compartment as a function
of time across the Monte Carlo simulation</p>
</td></tr>
<tr><td><code>sds</code></td>
<td>
<p>The standard deviation for each model compartment as a function
of time across the Monte Carlo simulation</p>
</td></tr>
</table>
<p>If return.all.sums == TRUE then a list is returned with:
</p>
<table>
<tr><td><code>stats</code></td>
<td>
<p>The list of means and sds from return.all.sums=FALSE</p>
</td></tr>
<tr><td><code>sims</code></td>
<td>
<p>The concentration vs. time results for each compartment for 
every (samples) set of parameters in the Monte Carlo simulation</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_mc_samples">create_mc_samples</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

NSAMP &lt;- 50
chemname="Abamectin"
times&lt;- c(0,0.25,0.5,0.75,1,1.5,2,2.5,3,4,5)
age.ranges &lt;- seq(6,80,by=10)
forward &lt;- NULL
for (age.lower in age.ranges)
{
  label &lt;- paste("Ages ",age.lower,"-",age.lower+4,sep="")
  set.seed(1234)
  forward[[label]] &lt;- calc_mc_tk(
                        chem.name=chemname,
                        samples=NSAMP,
                        httkpop.generate.arg.list=list(
                          method="d",
                          agelim_years = c(age.lower, age.lower+9)),
                        solvemodel.arg.list = list(
                          times=times))
}


</code></pre>

<hr>
<h2 id='calc_rblood2plasma'>Calculate the constant ratio of the blood concentration to the plasma
concentration.</h2><span id='topic+calc_rblood2plasma'></span>

<h3>Description</h3>

<p>This function calculates the constant ratio of the blood concentration to
the plasma concentration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_rblood2plasma(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  hematocrit = NULL,
  Krbc2pu = NULL,
  Funbound.plasma = NULL,
  default.to.human = FALSE,
  species = "Human",
  adjusted.Funbound.plasma = TRUE,
  suppress.messages = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_rblood2plasma_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the CAS number or the chemical name must be
specified.</p>
</td></tr>
<tr><td><code id="calc_rblood2plasma_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="calc_rblood2plasma_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_rblood2plasma_+3A_parameters">parameters</code></td>
<td>
<p>Parameters from <code><a href="#topic+parameterize_schmitt">parameterize_schmitt</a></code></p>
</td></tr>
<tr><td><code id="calc_rblood2plasma_+3A_hematocrit">hematocrit</code></td>
<td>
<p>Overwrites default hematocrit value in calculating
Rblood2plasma.</p>
</td></tr>
<tr><td><code id="calc_rblood2plasma_+3A_krbc2pu">Krbc2pu</code></td>
<td>
<p>The red blood cell to unbound plasma chemical partition
coefficient, typically from <code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code></p>
</td></tr>
<tr><td><code id="calc_rblood2plasma_+3A_funbound.plasma">Funbound.plasma</code></td>
<td>
<p>The fraction of chemical unbound (free) in the
presence of plasma protein</p>
</td></tr>
<tr><td><code id="calc_rblood2plasma_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true.</p>
</td></tr>
<tr><td><code id="calc_rblood2plasma_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_rblood2plasma_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Whether or not to use Funbound.plasma
adjustment.</p>
</td></tr>
<tr><td><code id="calc_rblood2plasma_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Determine whether to display certain usage
feedback.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The red blood cell (RBC) parition coefficient as predicted by the Schmitt
(2008) method is used in the calculation. The value is calculated with the
equation: 1 - hematocrit + hematocrit * Krbc2pu * Funbound.plasma, summing
the red blood cell to plasma and plasma:plasma (equal to 1) partition
coefficients multiplied by their respective fractional volumes. When
species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data (hematocrit and temperature), but substitutes
human fraction unbound and tissue volumes.
</p>


<h3>Value</h3>

<p>The blood to plasma chemical concentration ratio
</p>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>References</h3>

<p>Schmitt W. &quot;General approach for the calculation of tissue to
plasma partition coefficients.&quot; Toxicology In Vitro, 22, 457-467 (2008).
</p>
<p>Pearce, Robert G., et al. &quot;Evaluation and calibration of high-throughput 
predictions of chemical distribution to tissues.&quot; Journal of 
pharmacokinetics and pharmacodynamics 44.6 (2017): 549-565.
</p>
<p>Ruark, Christopher D., et al. &quot;Predicting passive and active tissue: plasma 
partition coefficients: interindividual and interspecies variability.&quot; 
Journal of pharmaceutical sciences 103.7 (2014): 2189-2198.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
calc_rblood2plasma(chem.name="Bisphenol A")
calc_rblood2plasma(chem.name="Bisphenol A",species="Rat")

</code></pre>

<hr>
<h2 id='calc_stats'>Calculate toxicokinetic summary statistics (deprecated).</h2><span id='topic+calc_stats'></span>

<h3>Description</h3>

<p>#' This function is included for backward compatibility. It calls
<code><a href="#topic+calc_tkstats">calc_tkstats</a></code> which 
calculates the area under the curve, the mean, and the peak values
for the venous blood or plasma concentration of a specified chemical or all
chemicals if none is specified for the multiple compartment model with a
given number of days, dose, and number of doses per day.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_stats(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  route = "oral",
  stats = c("AUC", "peak", "mean"),
  species = "Human",
  days = 28,
  daily.dose = 1,
  dose = NULL,
  doses.per.day = 1,
  output.units = "uM",
  concentration = "plasma",
  tissue = "plasma",
  model = "pbtk",
  default.to.human = FALSE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = TRUE,
  suppress.messages = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_stats_+3A_chem.name">chem.name</code></td>
<td>
<p>Name of desired chemical.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_chem.cas">chem.cas</code></td>
<td>
<p>CAS number of desired chemical.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_pbtk function,
overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_route">route</code></td>
<td>
<p>String specification of route of exposure for simulation:
&quot;oral&quot;, &quot;iv&quot;, &quot;inhalation&quot;, ...</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_stats">stats</code></td>
<td>
<p>Desired values (either 'AUC', 'mean', 'peak', or a vector
containing any combination).</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_days">days</code></td>
<td>
<p>Length of the simulation.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose, mg/kg BW.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_dose">dose</code></td>
<td>
<p>Amount of a single dose at time zero, mg/kg BW.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_doses.per.day">doses.per.day</code></td>
<td>
<p>Number of doses per day.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_output.units">output.units</code></td>
<td>
<p>Desired units (either &quot;mg/L&quot;, &quot;mg&quot;, &quot;umol&quot;, or default
&quot;uM&quot;).</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_concentration">concentration</code></td>
<td>
<p>Desired concentration type, 'blood' or default
'plasma'.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_tissue">tissue</code></td>
<td>
<p>Desired steady state tissue conentration.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_model">model</code></td>
<td>
<p>Model used in calculation, 'pbtk' for the multiple compartment
model,'3compartment' for the three compartment model, '3compartmentss' for
the three compartment steady state model, and '1compartment' for one
compartment model.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to
1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether to suppress output message.</p>
</td></tr>
<tr><td><code id="calc_stats_+3A_...">...</code></td>
<td>
<p>Arguments passed to solve function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Default value of 0 for doses.per.day solves for a single dose.
</p>
<p>When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>


<h3>Value</h3>

<table>
<tr><td><code>AUC</code></td>
<td>
<p>Area under the plasma concentration curve.</p>
</td></tr>
<tr><td><code>mean.conc</code></td>
<td>
<p>The area under the curve divided by the number of days.</p>
</td></tr>
<tr><td><code>peak.conc</code></td>
<td>
<p>The highest concentration.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Pearce and John Wambaugh
</p>

<hr>
<h2 id='calc_tkstats'>Calculate toxicokinetic summary statistics.</h2><span id='topic+calc_tkstats'></span>

<h3>Description</h3>

<p>This function calculates the area under the curve, the mean, and the peak values
for the venous blood or plasma concentration of a specified chemical or all
chemicals if none is specified for the multiple compartment model with a
given number of days, dose, and number of doses per day.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_tkstats(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  route = "oral",
  stats = c("AUC", "peak", "mean"),
  species = "Human",
  days = 28,
  daily.dose = 1,
  dose = NULL,
  forcings = NULL,
  doses.per.day = 1,
  output.units = "uM",
  concentration = "plasma",
  tissue = "plasma",
  model = "pbtk",
  default.to.human = FALSE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = TRUE,
  suppress.messages = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_tkstats_+3A_chem.name">chem.name</code></td>
<td>
<p>Name of desired chemical.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_chem.cas">chem.cas</code></td>
<td>
<p>CAS number of desired chemical.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_pbtk function,
overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_route">route</code></td>
<td>
<p>String specification of route of exposure for simulation:
&quot;oral&quot;, &quot;iv&quot;, &quot;inhalation&quot;, ...</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_stats">stats</code></td>
<td>
<p>Desired values (either 'AUC', 'mean', 'peak', or a vector
containing any combination).</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_days">days</code></td>
<td>
<p>Length of the simulation.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose, mg/kg BW.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_dose">dose</code></td>
<td>
<p>Amount of a single dose at time zero, mg/kg BW.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_forcings">forcings</code></td>
<td>
<p>Manual input of 'forcings' data series argument for ode
integrator, defaults is NULL. Then other input parameters
(see exp.start.time, exp.conc, exp.duration, and period) provide the
necessary information to assemble a forcings data series.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_doses.per.day">doses.per.day</code></td>
<td>
<p>Number of doses per day.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_output.units">output.units</code></td>
<td>
<p>Desired units (either &quot;mg/L&quot;, &quot;mg&quot;, &quot;umol&quot;, or default
&quot;uM&quot;).</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_concentration">concentration</code></td>
<td>
<p>Desired concentration type, 'blood' or default
'plasma'.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_tissue">tissue</code></td>
<td>
<p>Desired steady state tissue conentration.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_model">model</code></td>
<td>
<p>Model used in calculation, 'pbtk' for the multiple compartment
model,'3compartment' for the three compartment model, '3compartmentss' for
the three compartment steady state model, and '1compartment' for one
compartment model.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to
1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether to suppress output message.</p>
</td></tr>
<tr><td><code id="calc_tkstats_+3A_...">...</code></td>
<td>
<p>Arguments passed to solve function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Default value of 0 for doses.per.day solves for a single dose.
</p>
<p>When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>


<h3>Value</h3>

<table>
<tr><td><code>AUC</code></td>
<td>
<p>Area under the plasma concentration curve.</p>
</td></tr>
<tr><td><code>mean.conc</code></td>
<td>
<p>The area under the curve divided by the number of days.</p>
</td></tr>
<tr><td><code>peak.conc</code></td>
<td>
<p>The highest concentration.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Pearce and John Wambaugh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
calc_tkstats(chem.name='Bisphenol-A',days=100,stats='mean',model='3compartment')


calc_tkstats(chem.name='Bisphenol-A',days=100,stats=c('peak','mean'),species='Rat')

triclosan.stats &lt;- calc_tkstats(days=10, chem.name = "triclosan")


</code></pre>

<hr>
<h2 id='calc_total_clearance'>Calculate the total plasma clearance.</h2><span id='topic+calc_total_clearance'></span>

<h3>Description</h3>

<p>This function calculates the total clearance rate for a one compartment model
for plasma
where clearance is entirely due to metablism by the liver and glomerular
filtration in the kidneys, identical to clearance of three compartment
steady state model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_total_clearance(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  suppress.messages = FALSE,
  default.to.human = FALSE,
  well.stirred.correction = TRUE,
  restrictive.clearance = TRUE,
  adjusted.Funbound.plasma = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_total_clearance_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must
be specified.</p>
</td></tr>
<tr><td><code id="calc_total_clearance_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters
must be specified.</p>
</td></tr>
<tr><td><code id="calc_total_clearance_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_total_clearance_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_steadystate
function, overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="calc_total_clearance_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_total_clearance_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="calc_total_clearance_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true.</p>
</td></tr>
<tr><td><code id="calc_total_clearance_+3A_well.stirred.correction">well.stirred.correction</code></td>
<td>
<p>Uses correction in calculation of hepatic
clearance for well-stirred model if TRUE.  This assumes clearance relative
to amount unbound in whole blood instead of plasma, but converted to use
with plasma concentration.</p>
</td></tr>
<tr><td><code id="calc_total_clearance_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding is not taken into account (set
to 1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="calc_total_clearance_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE.</p>
</td></tr>
<tr><td><code id="calc_total_clearance_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to parameterize_steadystate if
parameters is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Total Clearance</code></td>
<td>
<p>Units of L/h/kg BW.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
calc_total_clearance(chem.name="Ibuprofen") 


</code></pre>

<hr>
<h2 id='calc_vdist'>Calculate the volume of distribution for a one compartment model.</h2><span id='topic+calc_vdist'></span>

<h3>Description</h3>

<p>This function predicts partition coefficients for all tissues using
<code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>, then lumps them
into a single compartment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_vdist(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  default.to.human = FALSE,
  species = "Human",
  suppress.messages = FALSE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  minimum.Funbound.plasma = 1e-04
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_vdist_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the CAS number or the chemical name must be specified
when Funbound.plasma is not given in parameter list.</p>
</td></tr>
<tr><td><code id="calc_vdist_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified when Funbound.plasma is not given in parameter list.</p>
</td></tr>
<tr><td><code id="calc_vdist_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="calc_vdist_+3A_parameters">parameters</code></td>
<td>
<p>Parameters from parameterize_3comp, parameterize_pbtk or
predict_partitioning_schmitt.</p>
</td></tr>
<tr><td><code id="calc_vdist_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true.</p>
</td></tr>
<tr><td><code id="calc_vdist_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="calc_vdist_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="calc_vdist_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with parition coefficients calculated with this value.</p>
</td></tr>
<tr><td><code id="calc_vdist_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="calc_vdist_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The effective volume of distribution is calculated by summing each tissues
volume times it's partition coefficient relative to plasma. Plasma, and the
paritioning into RBCs are also added to get the total volume of distribution
in L/KG BW. Partition coefficients are calculated using Schmitt's (2008)
method.  When species is specified as rabbit, dog, or mouse, the function
uses the appropriate physiological data(volumes and flows) but substitues
human fraction unbound, partition coefficients, and intrinsic hepatic
clearance.
</p>


<h3>Value</h3>

<table>
<tr><td><code>Volume of distribution</code></td>
<td>
<p>Units of L/ kg BW.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>References</h3>

<p>Schmitt W (2008).
&ldquo;General approach for the calculation of tissue to plasma partition coefficients.&rdquo;
<em>Toxicology in vitro</em>, <b>22</b>(2), 457&ndash;467.
</p>
<p>Peyret T, Poulin P, Krishnan K (2010).
&ldquo;A unified algorithm for predicting partition coefficients for PBPK modeling of drugs and environmental chemicals.&rdquo;
<em>Toxicology and applied pharmacology</em>, <b>249</b>(3), 197&ndash;207.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>
</p>
<p><code><a href="#topic+tissue.data">tissue.data</a></code>
</p>
<p><code><a href="#topic+physiology.data">physiology.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
calc_vdist(chem.cas="80-05-7")
calc_vdist(chem.name="Bisphenol A")
calc_vdist(chem.name="Bisphenol A",species="Rat")

</code></pre>

<hr>
<h2 id='CAS.checksum'>Test the check digit of a CAS number to confirm validity</h2><span id='topic+CAS.checksum'></span>

<h3>Description</h3>

<p>Chemical abstracts services registry numbers (CAS-RN) include a final digit
as a &quot;checksum&quot; to test for validity (that is, that the number has not been
corrupted).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CAS.checksum(CAS.string)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CAS.checksum_+3A_cas.string">CAS.string</code></td>
<td>
<p>A character string of three numbers separated by two dashes</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The check digit (final number) is calculated by working from right to left,
starting with the second to last digit of the CAS-RN. We multiply each digit
by an increasing digit (1, 2, 3...) and sum as we work from right to left.
The check digit should equal the final digit of the sum.
</p>


<h3>Value</h3>

<p>logical (TRUE if final digit of CAS is consistent with other digits)
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>

<hr>
<h2 id='chem.invivo.PK.aggregate.data'>Parameter Estimates from Wambaugh et al. (2018)</h2><span id='topic+chem.invivo.PK.aggregate.data'></span>

<h3>Description</h3>

<p>This table includes 1 and 2 compartment fits of plasma concentration vs time
data aggregated from chem.invivo.PK.data, performed in Wambaugh et al. 2018.
Data includes volume of distribution (Vdist, L/kg), elimination rate (kelim,
1/h), gut absorption rate (kgutabs, 1/h), fraction absorbed (Fabsgut), and
steady state concentration (Css, mg/L).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chem.invivo.PK.aggregate.data
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Source</h3>

<p>Wambaugh et al. 2018 Toxicological Sciences, in press
</p>

<hr>
<h2 id='chem.invivo.PK.data'>Published toxicokinetic time course measurements</h2><span id='topic+chem.invivo.PK.data'></span>

<h3>Description</h3>

<p>This data set includes time and dose specific measurements of chemical
concentration in tissues taken from animals administered control doses of
the chemicals either orally or intravenously. This plasma concentration-time
data is from rat experiments reported in public sources. Toxicokinetic data
were retrieved from those studies by the Netherlands Organisation for
Applied Scientific Research (TNO) using curve stripping (TechDig v2).  This
data is provided for statistical analysis as in Wambaugh et al. 2018.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chem.invivo.PK.data
</code></pre>


<h3>Format</h3>

<p>A data.frame containing 597 rows and 13 columns.
</p>


<h3>Author(s)</h3>

<p>Sieto Bosgra
</p>


<h3>Source</h3>

<p>Wambaugh et al. 2018 Toxicological Sciences, in press
</p>


<h3>References</h3>

<p>Aanderud L, Bakke OM (1983). Pharmacokinetics of antipyrine,
paracetamol, and morphine in rat at 71 ATA. Undersea Biomed Res.
10(3):193-201. PMID: 6636344
</p>
<p>Aasmoe L, Mathiesen M, Sager G (1999). Elimination of methoxyacetic acid and
ethoxyacetic acid in rat. Xenobiotica. 29(4):417-24. PMID: 10375010
</p>
<p>Ako RA. Pharmacokinetics/pharmacodynamics (PK/PD) of oral diethylstilbestrol
(DES) in recurrent prostate cancer patients and of oral dissolving film
(ODF)-DES in rats. PhD dissertation, College of Pharmacy, University of
Houston, USA, 2011.
</p>
<p>Anadon A, Martinez-Larranaga MR, Fernandez-Cruz ML, Diaz MJ, Fernandez MC,
Martinez MA (1996). Toxicokinetics of deltamethrin and its 4'-HO-metabolite
in the rat. Toxicol Appl Pharmacol. 141(1):8-16. PMID: 8917670
</p>
<p>Binkerd PE, Rowland JM, Nau H, Hendrickx AG (1988). Evaluation of valproic
acid (VPA) developmental toxicity and pharmacokinetics in Sprague-Dawley
rats. Fundam Appl Toxicol. 11(3):485-93. PMID: 3146521
</p>
<p>Boralli VB, Coelho EB, Cerqueira PM, Lanchote VL (2005). Stereoselective
analysis of metoprolol and its metabolites in rat plasma with application to
oxidative metabolism. J Chromatogr B Analyt Technol Biomed Life Sci.
823(2):195-202. PMID: 16029965
</p>
<p>Chan MP, Morisawa S, Nakayama A, Kawamoto Y, Sugimoto M, Yoneda M (2005).
Toxicokinetics of 14C-endosulfan in male Sprague-Dawley rats following oral
administration of single or repeated doses. Environ Toxicol. 20(5):533-41.
PMID: 16161119
</p>
<p>Cruz L, Castaneda-Hernandez G, Flores-Murrieta FJ, Garcia-Lopez P,
Guizar-Sahagun G (2002). Alteration of phenacetin pharmacokinetics after
experimental spinal cord injury. Proc West Pharmacol Soc. 45:4-5. PMID:
12434508
</p>
<p>Della Paschoa OE, Mandema JW, Voskuyl RA, Danhof M (1998).
Pharmacokinetic-pharmacodynamic modeling of the anticonvulsant and
electroencephalogram effects of phenytoin in rats. J Pharmacol Exp Ther.
284(2):460-6. PMID: 9454785
</p>
<p>Du B, Li X, Yu Q, A Y, Chen C (2010). Pharmacokinetic comparison of orally
disintegrating, beta-cyclodextrin inclusion complex and conventional tablets
of nicardipine in rats. Life Sci J. 7(2):80-4.
</p>
<p>Farris FF, Dedrick RL, Allen PV, Smith JC (1993). Physiological model for
the pharmacokinetics of methyl mercury in the growing rat. Toxicol Appl
Pharmacol. 119(1):74-90. PMID: 8470126
</p>
<p>Hays SM, Elswick BA, Blumenthal GM, Welsch F, Conolly RB, Gargas ML (2000).
Development of a physiologically based pharmacokinetic model of
2-methoxyethanol and 2-methoxyacetic acid disposition in pregnant rats.
Toxicol Appl Pharmacol. 163(1):67-74. PMID: 10662606
</p>
<p>Igari Y, Sugiyama Y, Awazu S, Hanano M (1982). Comparative physiologically
based pharmacokinetics of hexobarbital, phenobarbital and thiopental in the
rat. J Pharmacokinet Biopharm. 10(1):53-75. PMID: 7069578
</p>
<p>Ito K, Houston JB (2004). Comparison of the use of liver models for
predicting drug clearance using in vitro kinetic data from hepatic
microsomes and isolated hepatocytes. Pharm Res. 21(5):785-92. PMID: 15180335
</p>
<p>Jia L, Wong H, Wang Y, Garza M, Weitman SD (2003). Carbendazim: disposition,
cellular permeability, metabolite identification, and pharmacokinetic
comparison with its nanoparticle. J Pharm Sci. 92(1):161-72. PMID: 12486692
</p>
<p>Kawai R, Mathew D, Tanaka C, Rowland M (1998). Physiologically based
pharmacokinetics of cyclosporine A: extension to tissue distribution
kinetics in rats and scale-up to human. J Pharmacol Exp Ther. 287(2):457-68.
PMID: 9808668
</p>
<p>Kim YC, Kang HE, Lee MG (2008). Pharmacokinetics of phenytoin and its
metabolite, 4'-HPPH, after intravenous and oral administration of phenytoin
to diabetic rats induced by alloxan or streptozotocin. Biopharm Drug Dispos.
29(1):51-61. PMID: 18022993
</p>
<p>Kobayashi S, Takai K, Iga T, Hanano M (1991). Pharmacokinetic analysis of
the disposition of valproate in pregnant rats. Drug Metab Dispos.
19(5):972-6. PMID: 1686245
</p>
<p>Kotegawa T, Laurijssens BE, Von Moltke LL, Cotreau MM, Perloff MD,
Venkatakrishnan K, Warrington JS, Granda BW, Harmatz JS, Greenblatt DJ
(2002). In vitro, pharmacokinetic, and pharmacodynamic interactions of
ketoconazole and midazolam in the rat. J Pharmacol Exp Ther. 302(3):1228-37.
PMID: 12183684
</p>
<p>Krug AK, Kolde R, Gaspar JA, Rempel E, Balmer NV, Meganathan K, Vojnits K,
Baquie M, Waldmann T, Ensenat-Waser R, Jagtap S, Evans RM, Julien S,
Peterson H, Zagoura D, Kadereit S, Gerhard D, Sotiriadou I, Heke M,
Natarajan K, Henry M, Winkler J, Marchan R, Stoppini L, Bosgra S, Westerhout
J, Verwei M, Vilo J, Kortenkamp A, Hescheler J, Hothorn L, Bremer S, van
Thriel C, Krause KH, Hengstler JG, Rahnenfuhrer J, Leist M, Sachinidis A
(2013). Human embryonic stem cell-derived test systems for developmental
neurotoxicity: a transcriptomics approach. Arch Toxicol. 87(1):123-43. PMID:
23179753
</p>
<p>Leon-Reyes MR, Castaneda-Hernandez G, Ortiz MI (2009). Pharmacokinetic of
diclofenac in the presence and absence of glibenclamide in the rat. J Pharm
Pharm Sci. 12(3):280-7. PMID: 20067705
</p>
<p>Nagata M, Hidaka M, Sekiya H, Kawano Y, Yamasaki K, Okumura M, Arimori K
(2007). Effects of pomegranate juice on human cytochrome P450 2C9 and
tolbutamide pharmacokinetics in rats. Drug Metab Dispos. 35(2):302-5. PMID:
17132763
</p>
<p>Okiyama M, Ueno K, Ohmori S, Igarashi T, Kitagawa H (1988). Drug
interactions between imipramine and benzodiazepines in rats. J Pharm Sci.
77(1):56-63. PMID: 2894451
</p>
<p>Pelissier-Alicot AL, Schreiber-Deturmeny E, Simon N, Gantenbein M,
Bruguerolle B (2002). Time-of-day dependent pharmacodynamic and
pharmacokinetic profiles of caffeine in rats. Naunyn Schmiedebergs Arch
Pharmacol. 365(4):318-25. PMID: 11919657
</p>
<p>Piersma AH, Bosgra S, van Duursen MB, Hermsen SA, Jonker LR, Kroese ED, van
der Linden SC, Man H, Roelofs MJ, Schulpen SH, Schwarz M, Uibel F, van
Vugt-Lussenburg BM, Westerhout J, Wolterbeek AP, van der Burg B (2013).
Evaluation of an alternative in vitro test battery for detecting
reproductive toxicants. Reprod Toxicol. 38:53-64. PMID: 23511061
</p>
<p>Pollack GM, Li RC, Ermer JC, Shen DD (1985). Effects of route of
administration and repetitive dosing on the disposition kinetics of
di(2-ethylhexyl) phthalate and its mono-de-esterified metabolite in rats.
Toxicol Appl Pharmacol. Jun 30;79(2):246-56. PMID: 4002226
</p>
<p>Saadeddin A, Torres-Molina F, Carcel-Trullols J, Araico A, Peris JE (2004).
Pharmacokinetics of the time-dependent elimination of all-trans-retinoic
acid in rats. AAPS J. 6(1):1-9. PMID: 18465253
</p>
<p>Satterwhite JH, Boudinot FD (1991). Effects of age and dose on the
pharmacokinetics of ibuprofen in the rat. Drug Metab Dispos. 19(1):61-7.
PMID: 1673423
</p>
<p>Szymura-Oleksiak J, Panas M, Chrusciel W (1983). Pharmacokinetics of
imipramine after single and multiple intravenous administration in rats. Pol
J Pharmacol Pharm. 35(2):151-7. PMID: 6622297
</p>
<p>Tanaka C, Kawai R, Rowland M (2000). Dose-dependent pharmacokinetics of
cyclosporin A in rats: events in tissues. Drug Metab Dispos. 28(5):582-9.
PMID: 10772639
</p>
<p>Timchalk C, Nolan RJ, Mendrala AL, Dittenber DA, Brzak KA, Mattsson JL
(2002). A Physiologically based pharmacokinetic and pharmacodynamic
(PBPK/PD) model for the organophosphate insecticide chlorpyrifos in rats and
humans. Toxicol Sci. Mar;66(1):34-53. PMID: 11861971
</p>
<p>Tokuma Y, Sekiguchi M, Niwa T, Noguchi H (1988). Pharmacokinetics of
nilvadipine, a new dihydropyridine calcium antagonist, in mice, rats,
rabbits and dogs. Xenobiotica 18(1):21-8. PMID: 3354229
</p>
<p>Treiber A, Schneiter R, Delahaye S, Clozel M (2004). Inhibition of organic
anion transporting polypeptide-mediated hepatic uptake is the major
determinant in the pharmacokinetic interaction between bosentan and
cyclosporin A in the rat. J Pharmacol Exp Ther. 308(3):1121-9. PMID:
14617681
</p>
<p>Tsui BC, Feng JD, Buckley SJ, Yeung PK (1994). Pharmacokinetics and
metabolism of diltiazem in rats following a single intra-arterial or single
oral dose. Eur J Drug Metab Pharmacokinet. 19(4):369-73. PMID: 7737239
</p>
<p>Wambaugh, John F., et al. &quot;Toxicokinetic triage for environmental
chemicals.&quot; Toxicological Sciences (2015): 228-237.
</p>
<p>Wang Y, Roy A, Sun L, Lau CE (1999). A double-peak phenomenon in the
pharmacokinetics of alprazolam after oral administration. Drug Metab Dispos.
27(8):855-9. PMID: 10421610
</p>
<p>Wang X, Lee WY, Or PM, Yeung JH (2010). Pharmacokinetic interaction studies
of tanshinones with tolbutamide, a model CYP2C11 probe substrate, using
liver microsomes, primary hepatocytes and in vivo in the rat. Phytomedicine.
17(3-4):203-11. PMID: 19679455
</p>
<p>Yang SH, Lee MG (2008). Dose-independent pharmacokinetics of ondansetron in
rats: contribution of hepatic and intestinal first-pass effects to low
bioavailability. Biopharm Drug Dispos. 29(7):414-26. PMID: 18697186
</p>
<p>Yeung PK, Alcos A, Tang J (2009). Pharmacokinetics and Hemodynamic Effects
of Diltiazem in Rats Following Single vs Multiple Doses In Vivo. Open Drug
Metab J. 3:56-62.
</p>

<hr>
<h2 id='chem.invivo.PK.summary.data'>Summary of published toxicokinetic time course experiments</h2><span id='topic+chem.invivo.PK.summary.data'></span>

<h3>Description</h3>

<p>This data set summarizes the time course data in the chem.invivo.PK.data
table. Maximum concentration (Cmax), time integrated plasma concentration
for the duration of treatment (AUC.treatment) and extrapolated to zero
concentration (AUC.infinity) as well as half-life are calculated. Summary
values are given for each study and dosage. These data can be used to
evaluate toxicokinetic model predictions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chem.invivo.PK.summary.data
</code></pre>


<h3>Format</h3>

<p>A data.frame containing 100 rows and 25 columns.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Source</h3>

<p>Wambaugh et al. 2018 Toxicological Sciences, in press
</p>


<h3>References</h3>

<p>Aanderud L, Bakke OM (1983). Pharmacokinetics of antipyrine,
paracetamol, and morphine in rat at 71 ATA. Undersea Biomed Res.
10(3):193-201. PMID: 6636344
</p>
<p>Aasmoe L, Mathiesen M, Sager G (1999). Elimination of methoxyacetic acid and
ethoxyacetic acid in rat. Xenobiotica. 29(4):417-24. PMID: 10375010
</p>
<p>Ako RA. Pharmacokinetics/pharmacodynamics (PK/PD) of oral diethylstilbestrol
(DES) in recurrent prostate cancer patients and of oral dissolving film
(ODF)-DES in rats. PhD dissertation, College of Pharmacy, University of
Houston, USA, 2011.
</p>
<p>Anadon A, Martinez-Larranaga MR, Fernandez-Cruz ML, Diaz MJ, Fernandez MC,
Martinez MA (1996). Toxicokinetics of deltamethrin and its 4'-HO-metabolite
in the rat. Toxicol Appl Pharmacol. 141(1):8-16. PMID: 8917670
</p>
<p>Binkerd PE, Rowland JM, Nau H, Hendrickx AG (1988). Evaluation of valproic
acid (VPA) developmental toxicity and pharmacokinetics in Sprague-Dawley
rats. Fundam Appl Toxicol. 11(3):485-93. PMID: 3146521
</p>
<p>Boralli VB, Coelho EB, Cerqueira PM, Lanchote VL (2005). Stereoselective
analysis of metoprolol and its metabolites in rat plasma with application to
oxidative metabolism. J Chromatogr B Analyt Technol Biomed Life Sci.
823(2):195-202. PMID: 16029965
</p>
<p>Chan MP, Morisawa S, Nakayama A, Kawamoto Y, Sugimoto M, Yoneda M (2005).
Toxicokinetics of 14C-endosulfan in male Sprague-Dawley rats following oral
administration of single or repeated doses. Environ Toxicol. 20(5):533-41.
PMID: 16161119
</p>
<p>Cruz L, Castaneda-Hernandez G, Flores-Murrieta FJ, Garcia-Lopez P,
Guizar-Sahagun G (2002). Alteration of phenacetin pharmacokinetics after
experimental spinal cord injury. Proc West Pharmacol Soc. 45:4-5. PMID:
12434508
</p>
<p>Della Paschoa OE, Mandema JW, Voskuyl RA, Danhof M (1998).
Pharmacokinetic-pharmacodynamic modeling of the anticonvulsant and
electroencephalogram effects of phenytoin in rats. J Pharmacol Exp Ther.
284(2):460-6. PMID: 9454785
</p>
<p>Du B, Li X, Yu Q, A Y, Chen C (2010). Pharmacokinetic comparison of orally
disintegrating, beta-cyclodextrin inclusion complex and conventional tablets
of nicardipine in rats. Life Sci J. 7(2):80-4.
</p>
<p>Farris FF, Dedrick RL, Allen PV, Smith JC (1993). Physiological model for
the pharmacokinetics of methyl mercury in the growing rat. Toxicol Appl
Pharmacol. 119(1):74-90. PMID: 8470126
</p>
<p>Hays SM, Elswick BA, Blumenthal GM, Welsch F, Conolly RB, Gargas ML (2000).
Development of a physiologically based pharmacokinetic model of
2-methoxyethanol and 2-methoxyacetic acid disposition in pregnant rats.
Toxicol Appl Pharmacol. 163(1):67-74. PMID: 10662606
</p>
<p>Igari Y, Sugiyama Y, Awazu S, Hanano M (1982). Comparative physiologically
based pharmacokinetics of hexobarbital, phenobarbital and thiopental in the
rat. J Pharmacokinet Biopharm. 10(1):53-75. PMID: 7069578
</p>
<p>Ito K, Houston JB (2004). Comparison of the use of liver models for
predicting drug clearance using in vitro kinetic data from hepatic
microsomes and isolated hepatocytes. Pharm Res. 21(5):785-92. PMID: 15180335
</p>
<p>Jia L, Wong H, Wang Y, Garza M, Weitman SD (2003). Carbendazim: disposition,
cellular permeability, metabolite identification, and pharmacokinetic
comparison with its nanoparticle. J Pharm Sci. 92(1):161-72. PMID: 12486692
</p>
<p>Kawai R, Mathew D, Tanaka C, Rowland M (1998). Physiologically based
pharmacokinetics of cyclosporine A: extension to tissue distribution
kinetics in rats and scale-up to human. J Pharmacol Exp Ther. 287(2):457-68.
PMID: 9808668
</p>
<p>Kim YC, Kang HE, Lee MG (2008). Pharmacokinetics of phenytoin and its
metabolite, 4'-HPPH, after intravenous and oral administration of phenytoin
to diabetic rats induced by alloxan or streptozotocin. Biopharm Drug Dispos.
29(1):51-61. PMID: 18022993
</p>
<p>Kobayashi S, Takai K, Iga T, Hanano M (1991). Pharmacokinetic analysis of
the disposition of valproate in pregnant rats. Drug Metab Dispos.
19(5):972-6. PMID: 1686245
</p>
<p>Kotegawa T, Laurijssens BE, Von Moltke LL, Cotreau MM, Perloff MD,
Venkatakrishnan K, Warrington JS, Granda BW, Harmatz JS, Greenblatt DJ
(2002). In vitro, pharmacokinetic, and pharmacodynamic interactions of
ketoconazole and midazolam in the rat. J Pharmacol Exp Ther. 302(3):1228-37.
PMID: 12183684
</p>
<p>Krug AK, Kolde R, Gaspar JA, Rempel E, Balmer NV, Meganathan K, Vojnits K,
Baquie M, Waldmann T, Ensenat-Waser R, Jagtap S, Evans RM, Julien S,
Peterson H, Zagoura D, Kadereit S, Gerhard D, Sotiriadou I, Heke M,
Natarajan K, Henry M, Winkler J, Marchan R, Stoppini L, Bosgra S, Westerhout
J, Verwei M, Vilo J, Kortenkamp A, Hescheler J, Hothorn L, Bremer S, van
Thriel C, Krause KH, Hengstler JG, Rahnenfuhrer J, Leist M, Sachinidis A
(2013). Human embryonic stem cell-derived test systems for developmental
neurotoxicity: a transcriptomics approach. Arch Toxicol. 87(1):123-43. PMID:
23179753
</p>
<p>Leon-Reyes MR, Castaneda-Hernandez G, Ortiz MI (2009). Pharmacokinetic of
diclofenac in the presence and absence of glibenclamide in the rat. J Pharm
Pharm Sci. 12(3):280-7. PMID: 20067705
</p>
<p>Nagata M, Hidaka M, Sekiya H, Kawano Y, Yamasaki K, Okumura M, Arimori K
(2007). Effects of pomegranate juice on human cytochrome P450 2C9 and
tolbutamide pharmacokinetics in rats. Drug Metab Dispos. 35(2):302-5. PMID:
17132763
</p>
<p>Okiyama M, Ueno K, Ohmori S, Igarashi T, Kitagawa H (1988). Drug
interactions between imipramine and benzodiazepines in rats. J Pharm Sci.
77(1):56-63. PMID: 2894451
</p>
<p>Pelissier-Alicot AL, Schreiber-Deturmeny E, Simon N, Gantenbein M,
Bruguerolle B (2002). Time-of-day dependent pharmacodynamic and
pharmacokinetic profiles of caffeine in rats. Naunyn Schmiedebergs Arch
Pharmacol. 365(4):318-25. PMID: 11919657
</p>
<p>Piersma AH, Bosgra S, van Duursen MB, Hermsen SA, Jonker LR, Kroese ED, van
der Linden SC, Man H, Roelofs MJ, Schulpen SH, Schwarz M, Uibel F, van
Vugt-Lussenburg BM, Westerhout J, Wolterbeek AP, van der Burg B (2013).
Evaluation of an alternative in vitro test battery for detecting
reproductive toxicants. Reprod Toxicol. 38:53-64. PMID: 23511061
</p>
<p>Pollack GM, Li RC, Ermer JC, Shen DD (1985). Effects of route of
administration and repetitive dosing on the disposition kinetics of
di(2-ethylhexyl) phthalate and its mono-de-esterified metabolite in rats.
Toxicol Appl Pharmacol. Jun 30;79(2):246-56. PMID: 4002226
</p>
<p>Saadeddin A, Torres-Molina F, Carcel-Trullols J, Araico A, Peris JE (2004).
Pharmacokinetics of the time-dependent elimination of all-trans-retinoic
acid in rats. AAPS J. 6(1):1-9. PMID: 18465253
</p>
<p>Satterwhite JH, Boudinot FD (1991). Effects of age and dose on the
pharmacokinetics of ibuprofen in the rat. Drug Metab Dispos. 19(1):61-7.
PMID: 1673423
</p>
<p>Szymura-Oleksiak J, Panas M, Chrusciel W (1983). Pharmacokinetics of
imipramine after single and multiple intravenous administration in rats. Pol
J Pharmacol Pharm. 35(2):151-7. PMID: 6622297
</p>
<p>Tanaka C, Kawai R, Rowland M (2000). Dose-dependent pharmacokinetics of
cyclosporin A in rats: events in tissues. Drug Metab Dispos. 28(5):582-9.
PMID: 10772639
</p>
<p>Timchalk C, Nolan RJ, Mendrala AL, Dittenber DA, Brzak KA, Mattsson JL
(2002). A Physiologically based pharmacokinetic and pharmacodynamic
(PBPK/PD) model for the organophosphate insecticide chlorpyrifos in rats and
humans. Toxicol Sci. Mar;66(1):34-53. PMID: 11861971
</p>
<p>Tokuma Y, Sekiguchi M, Niwa T, Noguchi H (1988). Pharmacokinetics of
nilvadipine, a new dihydropyridine calcium antagonist, in mice, rats,
rabbits and dogs. Xenobiotica 18(1):21-8. PMID: 3354229
</p>
<p>Treiber A, Schneiter R, Delahaye S, Clozel M (2004). Inhibition of organic
anion transporting polypeptide-mediated hepatic uptake is the major
determinant in the pharmacokinetic interaction between bosentan and
cyclosporin A in the rat. J Pharmacol Exp Ther. 308(3):1121-9. PMID:
14617681
</p>
<p>Tsui BC, Feng JD, Buckley SJ, Yeung PK (1994). Pharmacokinetics and
metabolism of diltiazem in rats following a single intra-arterial or single
oral dose. Eur J Drug Metab Pharmacokinet. 19(4):369-73. PMID: 7737239
</p>
<p>Wambaugh, John F., et al. &quot;Toxicokinetic triage for environmental
chemicals.&quot; Toxicological Sciences (2015): 228-237.
</p>
<p>Wang Y, Roy A, Sun L, Lau CE (1999). A double-peak phenomenon in the
pharmacokinetics of alprazolam after oral administration. Drug Metab Dispos.
27(8):855-9. PMID: 10421610
</p>
<p>Wang X, Lee WY, Or PM, Yeung JH (2010). Pharmacokinetic interaction studies
of tanshinones with tolbutamide, a model CYP2C11 probe substrate, using
liver microsomes, primary hepatocytes and in vivo in the rat. Phytomedicine.
17(3-4):203-11. PMID: 19679455
</p>
<p>Yang SH, Lee MG (2008). Dose-independent pharmacokinetics of ondansetron in
rats: contribution of hepatic and intestinal first-pass effects to low
bioavailability. Biopharm Drug Dispos. 29(7):414-26. PMID: 18697186
</p>
<p>Yeung PK, Alcos A, Tang J (2009). Pharmacokinetics and Hemodynamic Effects
of Diltiazem in Rats Following Single vs Multiple Doses In Vivo. Open Drug
Metab J. 3:56-62.
</p>

<hr>
<h2 id='chem.physical_and_invitro.data'>Physico-chemical properties and in vitro measurements for toxicokinetics</h2><span id='topic+chem.physical_and_invitro.data'></span>

<h3>Description</h3>

<p>This data set contains the necessary information to make basic,
high-throughput toxicokinetic (HTTK) predictions for compounds, including
Funbound.plasma, molecular weight (g/mol), logP, logMA (membrane affinity),
intrinsic clearance(uL/min/10^6 cells), and pKa. These data have been
compiled from multiple sources, and can be used to parameterize a variety of
toxicokinetic models. See variable EPA.ref for information on the reference EPA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chem.physical_and_invitro.data
</code></pre>


<h3>Format</h3>

<p>A data.frame containing 9411 rows and 54 columns.
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Column Name</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td><td style="text-align: left;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
 Compound </td><td style="text-align: left;"> The preferred name of the chemical compound </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;">                      
 CAS</td><td style="text-align: left;"> The preferred Chemical Abstracts Service Registry Number </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;">                     
 CAS.Checksum </td><td style="text-align: left;"> A logical indicating whether the CAS number is valid </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;">                   
 DTXSID </td><td style="text-align: left;"> DSSTox Structure ID 
(<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>) </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;">                  
 Formula </td><td style="text-align: left;"> The proportions of atoms within the chemical compound  </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;">                   
 All.Compound.Names </td><td style="text-align: left;"> All names of the chemical as they occured in the
data </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;">              
 logHenry </td><td style="text-align: left;"> The log10 Henry's law constant </td><td style="text-align: left;"> 
log10(atmosphers*m^3/mole) </td>
</tr>
<tr>
 <td style="text-align: left;">                 
 logHenry.Reference </td><td style="text-align: left;"> Reference for Henry's law constant </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">           
 logP </td><td style="text-align: left;"> The log10 octanol:water partition coefficient (PC)</td><td style="text-align: left;"> log10 unitless ratio </td>
</tr>
<tr>
 <td style="text-align: left;">                 
 logP.Reference </td><td style="text-align: left;"> Reference for logPow </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">               
 logPwa </td><td style="text-align: left;"> The log10 water:air PC </td><td style="text-align: left;"> log10 unitless ratio </td>
</tr>
<tr>
 <td style="text-align: left;">                 
 logPwa.Reference </td><td style="text-align: left;"> Reference for logPwa </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">             
 logMA </td><td style="text-align: left;"> The log10 phospholipid:water PC or
"Membrane affinity" </td><td style="text-align: left;"> unitless ratio </td>
</tr>
<tr>
 <td style="text-align: left;">                
 logMA.Reference </td><td style="text-align: left;"> Reference for membrane affinity </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">   
 logWSol </td><td style="text-align: left;"> The log10 water solubility </td><td style="text-align: left;"> log10(mole/L) </td>
</tr>
<tr>
 <td style="text-align: left;">                  
 logWSol.Reference </td><td style="text-align: left;"> Reference for logWsol </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">              
 MP </td><td style="text-align: left;"> The chemical compound melting point </td><td style="text-align: left;"> degrees Celsius </td>
</tr>
<tr>
 <td style="text-align: left;">                  
 MP.Reference </td><td style="text-align: left;"> Reference for melting point </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">                   
 MW </td><td style="text-align: left;"> The chemical compound molecular weight </td><td style="text-align: left;"> g/mol </td>
</tr>
<tr>
 <td style="text-align: left;">                
 MW.Reference </td><td style="text-align: left;"> Reference for molecular weight </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">                 
 pKa_Accept </td><td style="text-align: left;"> The hydrogen acceptor equilibria concentrations 
 </td><td style="text-align: left;"> logarithm </td>
</tr>
<tr>
 <td style="text-align: left;">              
 pKa_Accept.Reference </td><td style="text-align: left;"> Reference for pKa_Accept </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">           
 pKa_Donor </td><td style="text-align: left;"> The hydrogen acceptor equilibria concentrations 
</td><td style="text-align: left;"> logarithm </td>
</tr>
<tr>
 <td style="text-align: left;">               
 pKa_Donor.Reference </td><td style="text-align: left;"> Reference for pKa_Donor </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">             
 All.Species </td><td style="text-align: left;"> All species for which data were available </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;">                
 DTXSID.Reference </td><td style="text-align: left;"> Reference for DTXSID </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">               
 Formula.Reference </td><td style="text-align: left;"> Reference for chemical formulat </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">             
 [SPECIES].Clint </td><td style="text-align: left;"> (Primary hepatocyte suspension) 
intrinsic hepatic clearance. <em>Entries with comma separated values are Bayesian estimates of
the Clint distribution - displayed as the median, 95th credible interval
(that is quantile 2.5 and 97.5, respectively), and p-value.</em> </td><td style="text-align: left;"> uL/min/10^6 hepatocytes </td>
</tr>
<tr>
 <td style="text-align: left;">                   
 [SPECIES].Clint.pValue </td><td style="text-align: left;"> Probability that there is no clearance observed.
 Values close to 1 indicate clearance is not statistically significant. </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;">           
 [SPECIES].Clint.pValue.Ref </td><td style="text-align: left;"> Reference for Clint pValue </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">   
 [SPECIES].Clint.Reference </td><td style="text-align: left;"> Reference for Clint </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">         
 [SPECIES].Caco2.Pab </td><td style="text-align: left;"> Caco-2 Apical-to-Basal Membrane Permeability </td><td style="text-align: left;"> 10^-6 cm/s </td>
</tr>
<tr>
 <td style="text-align: left;">           
 [SPECIES].Caco2.Pab.Reference </td><td style="text-align: left;"> Reference for Caco-2 Membrane Permeability </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
 [SPECIES].Fabs </td><td style="text-align: left;"> In vivo measured fraction of an oral dose of chemical 
absorbed from the gut lumen into the gut </td><td style="text-align: left;"> unitless fraction </td>
</tr>
<tr>
 <td style="text-align: left;">           
 [SPECIES].Fabs.Reference </td><td style="text-align: left;"> Reference for Fabs </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">        
 [SPECIES].Fgut </td><td style="text-align: left;"> In vivo measured fraction of an oral dose of chemical 
that passes gut metabolism and clearance </td><td style="text-align: left;"> unitless fraction </td>
</tr>
<tr>
 <td style="text-align: left;">           
 [SPECIES].Fgut.Reference </td><td style="text-align: left;"> Reference for Fgut </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">        
 [SPECIES].Foral </td><td style="text-align: left;"> In vivo measued fractional systemic bioavailability of 
an oral dose, modeled as he product of Fabs * Fgut * Fhep (where Fhep is 
first pass hepatic metabolism). </td><td style="text-align: left;"> unitless fraction </td>
</tr>
<tr>
 <td style="text-align: left;">           
 [SPECIES].Foral.Reference </td><td style="text-align: left;"> Reference for Foral </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">        
 [SPECIES].Funbound.plasma </td><td style="text-align: left;"> Chemical fraction unbound in presence of 
plasma proteins (fup). <em>Entries with comma separated values are Bayesian estimates of
the fup distribution - displayed as the median and 95th credible interval
(that is quantile 2.5 and 97.5, respectively).</em> </td><td style="text-align: left;"> unitless fraction </td>
</tr>
<tr>
 <td style="text-align: left;">         
 [SPECIES].Funbound.plasma.Ref</td><td style="text-align: left;"> Reference for Funbound.plasma </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;"> 
 [SPECIES].Rblood2plasma </td><td style="text-align: left;"> Chemical concentration blood to plasma ratio </td><td style="text-align: left;"> unitless ratio </td>
</tr>
<tr>
 <td style="text-align: left;">         
 [SPECIES].Rblood2plasma.Ref </td><td style="text-align: left;"> Reference for Rblood2plasma </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">        
 Chemical.Class </td><td style="text-align: left;"> All classes to which the chemical has been assigned </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Details</h3>

<p>In some cases the rapid equilbrium dailysis method (Waters et al., 2008)
fails to yield detectable concentrations for the free fraction of chemical. 
In those cases we assume the compound is highly bound (that is, Fup approaches
zero). For some calculations (for example, steady-state plasma concentration)
there is precendent (Rotroff et al., 2010) for using half the average limit 
of detection, that is 0.005. We do not recomend using other models where 
quantities like partition coefficients must be predicted using Fup. We also
do not recomend including the value 0.005 in training sets for Fup predictive
models. 
</p>
<p><strong>Note</strong> that in some cases the <strong>Funbound.plasma</strong> and the 
<strong>intrinsic clearance</strong> are
<em>provided as a series of numbers separated by commas</em>. These values are the 
result of Bayesian analysis and characterize a distribution: the first value
is the median of the distribution, while the second and third values are the 
lower and upper 95th percentile (that is qunatile 2.5 and 97.5) respectively.
For intrinsic clearance a fourth value indicating a p-value for a decrease is
provided. Typically 4000 samples were used for the Bayesian analusis, such
that a p-value of &quot;0&quot; is equivale to &quot;&lt;0.00025&quot;. See Wambaugh et al. (2019)
for more details.
</p>
<p>Any one chemical compound <em>may have multiple ionization equilibria</em> 
(see Strope et al., 2018) may both for donating or accepting a proton (and
therefore changing charge state). If there are multiple equlibria of the same
type (donor/accept])the are concatonated by commas.
</p>
<p>All species-specific information is initially from experimental measurements.
The functions <code><a href="#topic+load_sipes2017">load_sipes2017</a></code>, <code><a href="#topic+load_pradeep2020">load_pradeep2020</a></code>, 
and <code><a href="#topic+load_dawson2021">load_dawson2021</a></code> may be used to add in silico, structure-based
predictions for many thousands of additional compounds to this table.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Source</h3>

<p>Wambaugh, John F., et al. &quot;Toxicokinetic triage for environmental
chemicals.&quot; Toxicological Sciences (2015): 228-237.
</p>


<h3>References</h3>

<p>CompTox Chemicals Dashboard (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
</p>
<p>EPI Suite, https://www.epa.gov/opptintr/exposure/pubs/episuite.htm
</p>
<p>Brown, Hayley S., Michael Griffin, and J. Brian Houston. &quot;Evaluation of 
cryopreserved human hepatocytes as an alternative in vitro system to 
microsomes for the prediction of metabolic clearance.&quot; Drug metabolism and 
disposition 35.2 (2007): 293-301.
</p>
<p>Gulden, Michael, et al. &quot;Impact of protein binding on the availability and 
cytotoxic potency of organochlorine pesticides and chlorophenols in vitro.&quot; 
Toxicology 175.1-3 (2002): 201-213.
</p>
<p>Hilal, S., Karickhoff, S. and Carreira, L. (1995). A rigorous test for
SPARC's chemical reactivity models: Estimation of more than 4300 ionization
pKas. Quantitative Structure-Activity Relationships 14(4), 348-355.
</p>
<p>Honda, G. S., Pearce, R. G., Pham, L. L., Setzer, R. W., Wetmore, B. A., 
Sipes, N. S., ... &amp; Wambaugh, J. F. (2019). Using the concordance of in 
vitro and in vivo data to evaluate extrapolation assumptions. PloS one, 
14(5), e0217564.
</p>
<p>Ito, K. and Houston, J. B. (2004). Comparison of the use of liver models for
predicting drug clearance using in vitro kinetic data from hepatic
microsomes and isolated hepatocytes. Pharm Res 21(5), 785-92.
</p>
<p>Jones, O. A., Voulvoulis, N. and Lester, J. N. (2002). Aquatic environmental
assessment of the top 25 English prescription pharmaceuticals. Water
research 36(20), 5013-22.
</p>
<p>Jones, Barry C., et al. &quot;An investigation into the prediction of in vivo 
clearance for a range of flavin-containing monooxygenase substrates.&quot; 
Drug metabolism and disposition 45.10 (2017): 1060-1067.
</p>
<p>Lau, Y. Y., Sapidou, E., Cui, X., White, R. E. and Cheng, K. C. (2002).
Development of a novel in vitro model to predict hepatic clearance using
fresh, cryopreserved, and sandwich-cultured hepatocytes. Drug Metabolism and
Disposition 30(12), 1446-54.
</p>
<p>Linakis, M. W., Sayre, R. R., Pearce, R. G., Sfeir, M. A., Sipes, N. S., 
Pangburn, H. A., ... &amp; Wambaugh, J. F. (2020). Development and evaluation of 
a high-throughput inhalation model for organic chemicals. Journal of 
Exposure Science &amp; Environmental Epidemiology, 1-12.
</p>
<p>Lombardo, F., Berellini, G., &amp; Obach, R. S. (2018). Trend analysis of a 
database of intravenous pharmacokinetic parameters in humans for 1352 drug 
compounds. Drug Metabolism and Disposition, 46(11), 1466-1477.
</p>
<p>McGinnity, D. F., Soars, M. G., Urbanowicz, R. A. and Riley, R. J. (2004).
Evaluation of fresh and cryopreserved hepatocytes as in vitro drug
metabolism tools for the prediction of metabolic clearance. Drug Metabolism
and Disposition 32(11), 1247-53, 10.1124/dmd.104.000026.
</p>
<p>Naritomi, Y., Terashita, S., Kagayama, A. and Sugiyama, Y. (2003). Utility
of Hepatocytes in Predicting Drug Metabolism: Comparison of Hepatic
Intrinsic Clearance in Rats and Humans in Vivo and in Vitro. Drug Metabolism
and Disposition 31(5), 580-588, 10.1124/dmd.31.5.580.
</p>
<p>Obach, R. S. (1999). Prediction of human clearance of twenty-nine drugs from
hepatic microsomal intrinsic clearance data: An examination of in vitro
half-life approach and nonspecific binding to microsomes. Drug Metabolism
and Disposition 27(11), 1350-9.
</p>
<p>Paini, Alicia; Cole, Thomas; Meinero, Maria; Carpi, Donatella; Deceuninck, 
Pierre; Macko, Peter; Palosaari, Taina; Sund, Jukka; Worth, Andrew; Whelan, 
Maurice (2020):  EURL ECVAM in vitro hepatocyte clearance and blood plasma 
protein binding dataset for 77 chemicals. European Commission, Joint Research 
Centre (JRC) [Dataset] PID: https://data.europa.eu/89h/a2ff867f-db80-4acf-8e5c-e45502713bee
</p>
<p>Paixao, P., Gouveia, L. F., &amp; Morais, J. A. (2012). Prediction of the human
oral bioavailability by using in vitro and in silico drug related parameters
in a physiologically based absorption model. International journal of
pharmaceutics, 429(1), 84-98.
</p>
<p>Pirovano, Alessandra, et al. &quot;QSARs for estimating intrinsic hepatic
clearance of organic chemicals in humans.&quot; Environmental toxicology and
pharmacology 42 (2016): 190-197.
</p>
<p>Riley, Robert J., Dermot F. McGinnity, and Rupert P. Austin. &quot;A unified 
model for predicting human hepatic, metabolic clearance from in vitro 
intrinsic clearance data in hepatocytes and microsomes.&quot; Drug Metabolism and 
Disposition 33.9 (2005): 1304-1311.
</p>
<p>Schmitt, W. (2008). General approach for the calculation of tissue to plasma
partition coefficients. Toxicology in vitro : an international journal
published in association with BIBRA 22(2), 457-67,
10.1016/j.tiv.2007.09.010.
</p>
<p>Shibata, Y., Takahashi, H., Chiba, M. and Ishii, Y. (2002). Prediction of
Hepatic Clearance and Availability by Cryopreserved Human Hepatocytes: An
Application of Serum Incubation Method. Drug Metabolism and Disposition
30(8), 892-896, 10.1124/dmd.30.8.892.
</p>
<p>Sohlenius-Sternbeck, Anna-Karin, et al. &quot;Practical use of the regression 
offset approach for the prediction of in vivo intrinsic clearance from 
hepatocytes.&quot; Xenobiotica 42.9 (2012): 841-853.
</p>
<p>Tonnelier, A., Coecke, S. and Zaldivar, J.-M. (2012). Screening of chemicals
for human bioaccumulative potential with a physiologically based
toxicokinetic model. Archives of Toxicology 86(3), 393-403,
10.1007/s00204-011-0768-0.
</p>
<p>Uchimura, Takahide, et al. &quot;Prediction of human blood-to-plasma drug
concentration ratio.&quot; Biopharmaceutics &amp; drug disposition 31.5-6 (2010):
286-297.
</p>
<p>Wambaugh, J. F., Wetmore, B. A., Ring, C. L., Nicolas, C. I., Pearce, R. G., 
Honda, G. S., ... &amp; Badrinarayanan, A. (2019). Assessing Toxicokinetic 
Uncertainty and Variability in Risk Prioritization. Toxicological Sciences, 
172(2), 235-251.
</p>
<p>Wetmore, B. A., Wambaugh, J. F., Ferguson, S. S., Sochaski, M. A., Rotroff,
D. M., Freeman, K., Clewell, H. J., 3rd, Dix, D. J., Andersen, M. E., Houck,
K. A., Allen, B., Judson, R. S., Singh, R., Kavlock, R. J., Richard, A. M.
and Thomas, R. S. (2012). Integration of dosimetry, exposure, and
high-throughput screening data in chemical toxicity assessment.
Toxicological sciences : an official journal of the Society of Toxicology
125(1), 157-74, 10.1093/toxsci/kfr254.
</p>
<p>Wetmore, B. A., Wambaugh, J. F., Ferguson, S. S., Li, L., Clewell, H. J.,
Judson, R. S., Freeman, K., Bao, W., Sochaski, M. A., Chu, T.-M., Black, M.
B., Healy, E., Allen, B., Andersen, M. E., Wolfinger, R. D. and Thomas, R.
S. (2013). Relative Impact of Incorporating Pharmacokinetics on Predicting
In Vivo Hazard and Mode of Action from High-Throughput In Vitro Toxicity
Assays. Toxicological Sciences 132(2), 327-346, 10.1093/toxsci/kft012.
</p>
<p>Wetmore BA, Wambaugh JF, Allen B, Ferguson SS, Sochaski MA, Setzer RW, Houck KA, Strope CL, Cantwell K, Judson RS, others (2015).
&ldquo;Incorporating high-throughput exposure predictions with dosimetry-adjusted in vitro bioactivity to inform chemical toxicity testing.&rdquo;
<em>Toxicological Sciences</em>, <b>148</b>(1), 121&ndash;136.
</p>
<p>F. L. Wood, J. B. Houston and D. Hallifax
'Drug Metabolism and Disposition November 1, 2017, 45 (11) 1178-1188; 
DOI: https://doi.org/10.1124/dmd.117.077040
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_physchem_param">get_physchem_param</a></code>
</p>
<p><code><a href="#topic+get_invitroPK_param">get_invitroPK_param</a></code>
</p>
<p><code><a href="#topic+add_chemtable">add_chemtable</a></code>
</p>

<hr>
<h2 id='ckd_epi_eq'>CKD-EPI equation for GFR.</h2><span id='topic+ckd_epi_eq'></span>

<h3>Description</h3>

<p>Predict GFR from serum creatinine, gender, and age.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ckd_epi_eq(scr, gender, reth, age_years, ckd_epi_race_coeff = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ckd_epi_eq_+3A_scr">scr</code></td>
<td>
<p>Vector of serum creatinine values in mg/dL.</p>
</td></tr>
<tr><td><code id="ckd_epi_eq_+3A_gender">gender</code></td>
<td>
<p>Vector of genders (either 'Male' or 'Female').</p>
</td></tr>
<tr><td><code id="ckd_epi_eq_+3A_reth">reth</code></td>
<td>
<p>Vector of races/ethnicities. Not used unless ckd_epi_race_coeff is TRUE.</p>
</td></tr>
<tr><td><code id="ckd_epi_eq_+3A_age_years">age_years</code></td>
<td>
<p>Vector of ages in years.</p>
</td></tr>
<tr><td><code id="ckd_epi_eq_+3A_ckd_epi_race_coeff">ckd_epi_race_coeff</code></td>
<td>
<p>Whether to use the &quot;race coefficient&quot; in the CKD-EPI equation. Default is FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>From Levey AS, Stevens LA, Schmid CH, Zhang YL, Castro AF, Feldman HI, et al. A new
equation to estimate glomerular filtration rate. Ann Intern Med 2009;
150(9):604-612. doi:10.7326/0003-4819-150-9-200905050-00006
</p>


<h3>Value</h3>

<p>Vector of GFR values in mL/min/1.73m^2.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='concentration_data_Linakis2020'>Concentration data involved in Linakis 2020 vignette analysis.</h2><span id='topic+concentration_data_Linakis2020'></span>

<h3>Description</h3>

<p>Concentration data involved in Linakis 2020 vignette analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>concentration_data_Linakis2020
</code></pre>


<h3>Format</h3>

<p>A data.frame containing x rows and y columns.
</p>


<h3>Author(s)</h3>

<p>Matt Linakis
</p>


<h3>Source</h3>

<p>Matt Linakis
</p>


<h3>References</h3>

<p>DSStox database (https:// www.epa.gov/ncct/dsstox
</p>

<hr>
<h2 id='convert_httkpop_1comp'>Converts HTTK-Pop physiology into parameters relevant to the one
compartment model</h2><span id='topic+convert_httkpop_1comp'></span>

<h3>Description</h3>

<p>Converts HTTK-Pop physiology into parameters relevant to the one
compartment model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_httkpop_1comp(parameters.dt, httkpop.dt, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_httkpop_1comp_+3A_parameters.dt">parameters.dt</code></td>
<td>
<p>Data table returned by <code><a href="#topic+create_mc_samples">create_mc_samples</a></code></p>
</td></tr>
<tr><td><code id="convert_httkpop_1comp_+3A_httkpop.dt">httkpop.dt</code></td>
<td>
<p>Data table returned by <code><a href="#topic+httkpop_generate">httkpop_generate</a></code></p>
</td></tr>
<tr><td><code id="convert_httkpop_1comp_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+propagate_invitrouv_1comp">propagate_invitrouv_1comp</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table whose columns are the parameters of the HTTK model
specified in <code>model</code>.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring, John Wambaugh, and Greg Honda
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='convert_solve_x'>convert_solve_x</h2><span id='topic+convert_solve_x'></span>

<h3>Description</h3>

<p>This function is designed to convert compartment values estimated from one
of the HTTK models (e.g. &quot;1compartment) using the solve_model function.
It takes the HTTK model output matrix, model name, desired output units,
and compound information to perform the conversion default model units to
user specified units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_solve_x(
  model.output.mat,
  model = NULL,
  output.units = NULL,
  MW = NULL,
  vol = NULL,
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  monitor.vars = NULL,
  suppress.messages = FALSE,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_solve_x_+3A_model.output.mat">model.output.mat</code></td>
<td>
<p>Matrix of results from HTTK solve_model function.</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_model">model</code></td>
<td>
<p>Specified model to use in simulation: &quot;pbtk&quot;, &quot;3compartment&quot;,
&quot;3compartmentss&quot;, &quot;1compartment&quot;, &quot;schmitt&quot;, ...</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_output.units">output.units</code></td>
<td>
<p>Output units of interest for the compiled components.
Defaults to NULL, and will provide values in model units if unspecified.</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_mw">MW</code></td>
<td>
<p>Molecular weight of substance of interest in g/mole</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_vol">vol</code></td>
<td>
<p>Volume for the target tissue of interest in liters (L).
NOTE: Volume should not be in units of per BW, i.e. &quot;kg&quot;.</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must
be specified.</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters
must be specified.</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID .
(<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>) the chemical must be identified by
either CAS, name, or DTXSIDs.</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_parameters">parameters</code></td>
<td>
<p>A set of model parameters, especially a set that
includes MW (molecular weight) for our conversions.</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_monitor.vars">monitor.vars</code></td>
<td>
<p>A vector of character strings indicating the model
component variables to retain in the conversion factor table
(assuming suppress.messages == FALSE).  It should also be noted this option
does NOT exclude columns from the input matrix provided in the
'model.output.mat' parameter. (Default is NULL, i.e. conversion factors for
all model components are included in the reporting matrix.)</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output messages are suppressed.
(Default is FALSE, i.e. show messages.)</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_verbose">verbose</code></td>
<td>
<p>Whether or not to display the full conversion factor table.
(Default is FALSE, i.e. only include rows where the conversion factor is 1.)</p>
</td></tr>
<tr><td><code id="convert_solve_x_+3A_...">...</code></td>
<td>
<p>Other parameters that can be passed to <code>convert_units</code>, e.g.
temperature and compound state.  See details in <code><a href="#topic+convert_units">convert_units</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function can be used to convert all compartments to a single unit,
only units for a single model compartment, or units for a set of model
compartments.
</p>
<p>More details on the unit conversion can be found in the documentation for
<code><a href="#topic+convert_units">convert_units</a></code>.
</p>


<h3>Value</h3>

<p>'new.ouput.matrix' A matrix with a column for time (in days), each
compartment, and the area under the curve (AUC) and a row
for each time point. The compartment and AUC columns are
converted from model specified units to user specified units.
</p>
<p>'output.units.vector' A vector of character strings providing the
model compartments and their corresponding units after <code>convert_solve_x</code>.
</p>


<h3>Author(s)</h3>

<p>Sarah E. Davidson
</p>


<h3>See Also</h3>

<p>convert_units
</p>


<h3>Examples</h3>

<pre><code class='language-R'>output.mat &lt;- solve_1comp(dtxsid = "DTXSID0020573")
new.output.mat &lt;- convert_solve_x(output.units = "mg",
                                  model.output.mat = output.mat,
                                  model = "1compartment",
                                  dtxsid = "DTXSID0020573")

</code></pre>

<hr>
<h2 id='convert_units'>convert_units</h2><span id='topic+convert_units'></span>

<h3>Description</h3>

<p>This function is designed to accept input units, output units, and the 
molecular weight (MW) of a substance of interest to then use a table lookup
to return a scaling factor that can be readily applied for the intended
conversion. It can also take chemical identifiers in the place of a 
specified molecular weight value to retrieve that value for its own use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_units(
  input.units = NULL,
  output.units = NULL,
  MW = NULL,
  vol = NULL,
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  temp = 25,
  liquid.density = 1,
  state = "liquid"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_units_+3A_input.units">input.units</code></td>
<td>
<p>Assigned input units of interest</p>
</td></tr>
<tr><td><code id="convert_units_+3A_output.units">output.units</code></td>
<td>
<p>Desired output units</p>
</td></tr>
<tr><td><code id="convert_units_+3A_mw">MW</code></td>
<td>
<p>Molecular weight of substance of interest in g/mole</p>
</td></tr>
<tr><td><code id="convert_units_+3A_vol">vol</code></td>
<td>
<p>Volume for the target tissue of interest in liters (L).
NOTE: Volume should not be in units of per BW, i.e. &quot;kg&quot;.</p>
</td></tr>
<tr><td><code id="convert_units_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must
be specified.</p>
</td></tr>
<tr><td><code id="convert_units_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters
must be specified.</p>
</td></tr>
<tr><td><code id="convert_units_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID 
(<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>) the chemical must be identified by
either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="convert_units_+3A_parameters">parameters</code></td>
<td>
<p>A set of model parameters, especially a set that
includes MW (molecular weight) for our conversions</p>
</td></tr>
<tr><td><code id="convert_units_+3A_temp">temp</code></td>
<td>
<p>Temperature for conversions (default = 25 degreees C)</p>
</td></tr>
<tr><td><code id="convert_units_+3A_liquid.density">liquid.density</code></td>
<td>
<p>Density of the specified chemical in liquid state,
numeric value, (default 1.0 g/mL).</p>
</td></tr>
<tr><td><code id="convert_units_+3A_state">state</code></td>
<td>
<p>Chemical state (gas or default liquid).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If input or output units not contained in the table are queried,
it gives a corresponding error message. It gives a warning message about the
handling of 'ppmv,' as the function is only set up to convert between ppmv 
and mass-based units (like 
mg/m<sup>3</sup> or umol/L) 
in the context of ideal gases.
</p>
<p>convert_units is not directly configured to accept and convert units based
on BW, like mg/kg. For this purpose, see <code><a href="#topic+scale_dosing">scale_dosing</a></code>.
</p>
<p>The function supports a limited set of most relevant units across
toxicological models, currently including umol, uM, mg, mg/L, 
mg/m<sup>3</sup> or umol/L), and
in the context of gases assumed to be ideal, ppmv. 
</p>
<p><em>Andersen and Clewell's Rules of PBPK Modeling:</em>
</p>

<ol>
<li><p> Check Your Units
</p>
</li>
<li> <p><strong>Check Your Units</strong>
</p>
</li>
<li><p> Check Mass Balance
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Mark Sfeir, John Wambaugh, and Sarah E. Davidson
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# MW BPA is 228.29 g/mol
# 1 mg/L -&gt; 1/228.29*1000 = 4.38 uM
convert_units("mg/L","uM",chem.cas="80-05-7")

# MW Diclofenac is 296.148 g/mol
# 1 uM -&gt; 296.148/1000 =  0.296
convert_units("uM","mg/L",chem.name="diclofenac")

# ppmv only works for gasses:
try(convert_units("uM","ppmv",chem.name="styrene"))
convert_units("uM","ppmv",chem.name="styrene",state="gas")

# Compare with https://www3.epa.gov/ceampubl/learn2model/part-two/onsite/ia_unit_conversion.html
# 1 ug/L Toluene -&gt; 0.263 ppmv
convert_units("ug/L","ppmv",chem.name="toluene",state="gas")
# 1 pppmv Toluene, 0.0038 mg/L
convert_units("ppmv","mg/L",chem.name="toluene",state="gas")

MW_pyrene &lt;- get_physchem_param(param='MW', chem.name='pyrene')
conversion_factor &lt;- convert_units(input.units='mg/L', output.units ='uM',
  MW=MW_pyrene)

calc_mc_oral_equiv(15, parameters=p)
</code></pre>

<hr>
<h2 id='create_mc_samples'>Create a table of parameter values for Monte Carlo</h2><span id='topic+create_mc_samples'></span>

<h3>Description</h3>

<p>This is the HTTK master function for creating a data table for use with 
Monte Carlo methods to simulate parameter uncertainty and variabilit.
Each column of the output table corresponds to an HTTK model parameter and 
each row corresponds
to a different random draw (for example, different individuals when 
considering biological variability). This function call three different key
functions to simulate parameter parameter uncertainty and/or variability in
one of three ways. First parameters can be varied in an uncorrelated manner
using truncated normal distributions by the function 
<code><a href="#topic+monte_carlo">monte_carlo</a></code>. Then,
physiological parameters can be varied in a correlated manner according to
the Ring et al. (2017) 
(<a href="https://doi.org/10.1016/j.envint.2017.06.004">doi:10.1016/j.envint.2017.06.004</a>) <em>httk-pop</em> 
approach by the function <code><a href="#topic+httkpop_mc">httkpop_mc</a></code>. Next, both uncertainty
and variability of in vitro HTTK parameters can be simulated by the function
<code><a href="#topic+invitro_mc">invitro_mc</a></code> as described by 
Wambaugh et al. (2019) 
(<a href="https://doi.org/10.1093/toxsci/kfz205">doi:10.1093/toxsci/kfz205</a>). Finally, tissue-specific partition 
coefficients are predicted for each draw using the 
Schmitt (2008)
(<a href="https://doi.org/10.1016/j.tiv.2007.09.010">doi:10.1016/j.tiv.2007.09.010</a>) method as calibrated to <em>in vivo</em>
data by Pearce et al. (2017) 
(<a href="https://doi.org/10.1007/s10928-017-9548-7">doi:10.1007/s10928-017-9548-7</a>) and 
implemented in <code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_mc_samples(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  samples = 1000,
  species = "Human",
  suppress.messages = FALSE,
  model = "3compartmentss",
  httkpop = TRUE,
  invitrouv = TRUE,
  calcrb2p = TRUE,
  censored.params = list(),
  vary.params = list(),
  return.samples = FALSE,
  tissue = NULL,
  httkpop.dt = NULL,
  invitro.mc.arg.list = NULL,
  adjusted.Funbound.plasma = TRUE,
  adjusted.Clint = TRUE,
  httkpop.generate.arg.list = list(method = "direct resampling"),
  convert.httkpop.arg.list = NULL,
  propagate.invitrouv.arg.list = NULL,
  parameterize.arg.list = NULL,
  Caco2.options = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_mc_samples_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_parameters">parameters</code></td>
<td>
<p>Parameters from the appropriate parameterization function
for the model indicated by argument model</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_samples">samples</code></td>
<td>
<p>Number of samples generated in calculating quantiles.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;). Species must be set to &quot;Human&quot; to run httkpop model.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not to suppress output message.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_model">model</code></td>
<td>
<p>Model used in calculation: 'pbtk' for the multiple compartment
model,'3compartment' for the three compartment model, '3compartmentss' for
the three compartment steady state model, and '1compartment' for one
compartment model.  This only applies when httkpop=TRUE and species=&quot;Human&quot;,
otherwise '3compartmentss' is used.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_httkpop">httkpop</code></td>
<td>
<p>Whether or not to use the 
Ring et al. (2017) &quot;httkpop&quot;
population generator. Species must be 'Human'.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_invitrouv">invitrouv</code></td>
<td>
<p>Logical to indicate whether to include in vitro parameters
such as intrinsic hepatic clearance rate and fraction unbound in plasma
in uncertainty and variability analysis</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_calcrb2p">calcrb2p</code></td>
<td>
<p>Logical determining whether or not to recalculate the 
chemical ratio of blood to plasma</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_censored.params">censored.params</code></td>
<td>
<p>The parameters listed in censored.params are sampled
from a normal distribution that is censored for values less than the limit
of detection (specified separately for each parameter). This argument should
be a list of sub-lists. Each sublist is named for a parameter in
&quot;parameters&quot; and contains two elements: &quot;CV&quot; (coefficient of variation) and
&quot;LOD&quot; (limit of detection, below which parameter values are censored. New
values are sampled with mean equal to the value in &quot;parameters&quot; and standard
deviation equal to the mean times the CV.  Censored values are sampled on a
uniform distribution between 0 and the limit of detection. Not used with
httkpop model.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_vary.params">vary.params</code></td>
<td>
<p>The parameters listed in vary.params are sampled from a
normal distribution that is truncated at zero. This argument should be a
list of coefficients of variation (CV) for the normal distribution. Each
entry in the list is named for a parameter in &quot;parameters&quot;. New values are
sampled with mean equal to the value in &quot;parameters&quot; and standard deviation
equal to the mean times the CV. Not used with httkpop model.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_return.samples">return.samples</code></td>
<td>
<p>Whether or not to return the vector containing the
samples from the simulation instead of the selected quantile.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_tissue">tissue</code></td>
<td>
<p>Desired steady state tissue conentration.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_httkpop.dt">httkpop.dt</code></td>
<td>
<p>A data table generated by <code><a href="#topic+httkpop_generate">httkpop_generate</a></code>.
This defaults to NULL, in which case <code><a href="#topic+httkpop_generate">httkpop_generate</a></code> is 
called to generate this table.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_invitro.mc.arg.list">invitro.mc.arg.list</code></td>
<td>
<p>Additional parameters passed to 
<code><a href="#topic+invitro_mc">invitro_mc</a></code>.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses 
Pearce et al. (2017) lipid binding adjustment
for Funbound.plasma when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_adjusted.clint">adjusted.Clint</code></td>
<td>
<p>Uses 
Kilford et al. (2008) hepatocyte incubation
binding adjustment for Clint when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_httkpop.generate.arg.list">httkpop.generate.arg.list</code></td>
<td>
<p>Additional parameters passed to 
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_convert.httkpop.arg.list">convert.httkpop.arg.list</code></td>
<td>
<p>Additional parameters passed to the 
convert_httkpop_* function for the model.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_propagate.invitrouv.arg.list">propagate.invitrouv.arg.list</code></td>
<td>
<p>Additional parameters passed to model's
associated in vitro uncertainty and variability propagation function</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_parameterize.arg.list">parameterize.arg.list</code></td>
<td>
<p>Additional parameters passed to the 
parameterize_* function for the model.</p>
</td></tr>
<tr><td><code id="create_mc_samples_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>Arguments describing how to handle Caco2 absorption data
that are passed to <code><a href="#topic+invitro_mc">invitro_mc</a></code> and the parameterize_[MODEL] 
functions.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Monte Carlo methods used here were recently updated and described by
Breen et al. (2022).
</p>
<p>We aim to make any function that uses chemical identifiers 
(name, CAS, DTXSID) 
also work if passed a complete vector of parameters (that is, a row from the
table generated by this function). This allows the
use of Monte Carlo to vary the parameters and therefore vary the function
output. Depending on the type of parameters (for example, physiological vs. 
in vitro measurements) we vary the parameters in different ways with 
different functions.
</p>


<h3>Value</h3>

<p>A data table where each column corresponds to parameters needed for the 
specified model and each row represents a different Monte Carlo sample of
parameter values.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring, Robert Pearce, and John Wambaugh
</p>


<h3>References</h3>

<p>Breen M, Wambaugh JF, Bernstein A, Sfeir M, Ring CL (2022).
&ldquo;Simulating toxicokinetic variability to identify susceptible and highly exposed populations.&rdquo;
<em>Journal of Exposure Science &amp; Environmental Epidemiology</em>, <b>32</b>(6), 855&ndash;863.<br /><br /> Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197.<br /><br /> Pearce RG, Setzer RW, Davis JL, Wambaugh JF (2017).
&ldquo;Evaluation and calibration of high-throughput predictions of chemical distribution to tissues.&rdquo;
<em>Journal of pharmacokinetics and pharmacodynamics</em>, <b>44</b>, 549&ndash;565.<br /><br /> Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.<br /><br /> Schmitt W (2008).
&ldquo;General approach for the calculation of tissue to plasma partition coefficients.&rdquo;
<em>Toxicology in vitro</em>, <b>22</b>(2), 457&ndash;467.<br /><br /> Wambaugh JF, Wetmore BA, Ring CL, Nicolas CI, Pearce RG, Honda GS, Dinallo R, Angus D, Gilbert J, Sierra T, others (2019).
&ldquo;Assessing toxicokinetic uncertainty and variability in risk prioritization.&rdquo;
<em>Toxicological Sciences</em>, <b>172</b>(2), 235&ndash;251.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# We can use the Monte Carlo functions by passing a table
# where each row represents a different Monte Carlo draw of parameters:
p &lt;- create_mc_samples(chem.cas="80-05-7")
# Use data.table for steady-state plasma concentration (Css) Monte Carlo:
calc_mc_css(parameters=p)
# Using the same table gives the same answer:
calc_mc_css(parameters=p)
# Use Css for 1 mg/kg/day for simple reverse toxicokinetics 
# in Vitro-In Vivo Extrapolation to convert 15 uM to mg/kg/day:
15/calc_mc_css(parameters=p, output.units="uM")
# Can do the same with calc_mc_oral_equiv:
calc_mc_oral_equiv(15, parameters=p)


</code></pre>

<hr>
<h2 id='dawson2021'>Dawson et al. 2021 data</h2><span id='topic+dawson2021'></span><span id='topic+Dawson2021'></span>

<h3>Description</h3>

<p>This table includes QSAR (Random Forest) model predicted values for unbound
fraction plasma protein (fup) and intrinsic hepatic clearance (clint) for a
subset of chemicals in the Tox21 library
(see <a href="https://www.epa.gov/chemical-research/toxicology-testing-21st-century-tox21">https://www.epa.gov/chemical-research/toxicology-testing-21st-century-tox21</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dawson2021
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Details</h3>

<p>Predictions were made with a set of Random Forest QSAR models,
as reported in Dawson et al. (2021).
</p>


<h3>Author(s)</h3>

<p>Daniel E. Dawson
</p>


<h3>References</h3>

<p>Dawson DE, Ingle BL, Phillips KA, Nichols JW, Wambaugh JF, Tornero-Velez R (2021).
&ldquo;Designing QSARs for Parameters of High-Throughput Toxicokinetic Models Using Open-Source Descriptors.&rdquo;
<em>Environmental Science &amp; Technology</em>, <b>55</b>(9), 6505-6517.
<a href="https://doi.org/10.1021/acs.est.0c06117">doi:10.1021/acs.est.0c06117</a>, PMID: 33856768, https://doi.org/10.1021/acs.est.0c06117.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_dawson2021">load_dawson2021</a></code>
</p>

<hr>
<h2 id='EPA.ref'>Reference for EPA Physico-Chemical Data</h2><span id='topic+EPA.ref'></span>

<h3>Description</h3>

<p>The physico-chemical data in the chem.phys_and_invitro.data table are
obtained from EPA's Comptox Chemicals dashboard. This variable indicates
the date the Dashboard was accessed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EPA.ref
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 1.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Source</h3>

<p><a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>
</p>

<hr>
<h2 id='estimate_gfr'>Predict GFR.</h2><span id='topic+estimate_gfr'></span>

<h3>Description</h3>

<p>Predict GFR using CKD-EPI equation (for adults) or BSA-based equation (for children).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_gfr(gfrtmp.dt, gfr_resid_var = TRUE, ckd_epi_race_coeff = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_gfr_+3A_gfrtmp.dt">gfrtmp.dt</code></td>
<td>
<p>A data.table with columns <code>gender</code>, <code>reth</code>, 
<code>age_years</code>, <code>age_months</code>, <code>BSA_adj</code>, <code>serum_creat</code>.</p>
</td></tr>
<tr><td><code id="estimate_gfr_+3A_gfr_resid_var">gfr_resid_var</code></td>
<td>
<p>Logical value indicating whether or not to include
residual variability when generating GFR values. (Default is TRUE.)</p>
</td></tr>
<tr><td><code id="estimate_gfr_+3A_ckd_epi_race_coeff">ckd_epi_race_coeff</code></td>
<td>
<p>Logical value indicating whether or not to use the
&quot;race coefficient&quot; from the CKD-EPI equation when estimating GFR values.
(Default is FALSE.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Add residual variability based on reported residuals for each equation.
</p>


<h3>Value</h3>

<p>The same data.table with a <code>gfr_est</code> column added, containing 
estimated GFR values.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='estimate_gfr_ped'>Predict GFR in children.</h2><span id='topic+estimate_gfr_ped'></span>

<h3>Description</h3>

<p>BSA-based equation from Johnson et al. 2006, Clin Pharmacokinet 45(9) 931-56. 
Used in Wetmore et al. 2014.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_gfr_ped(BSA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_gfr_ped_+3A_bsa">BSA</code></td>
<td>
<p>Vector of body surface areas in m^2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of GFRs in mL/min/1.73m^2.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='estimate_hematocrit'>Generate hematocrit values for a virtual population</h2><span id='topic+estimate_hematocrit'></span>

<h3>Description</h3>

<p>Predict hematocrit from age using smoothing splines and kernel density
estimates of residual variability fitted to NHANES data, for a given
combination of gender and NHANES race/ethnicity category.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_hematocrit(gender, reth, age_years, age_months, nhanes_mec_svy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_hematocrit_+3A_gender">gender</code></td>
<td>
<p>Gender for which to generate hematocrit values (&quot;Male&quot; or
&quot;Female&quot;)</p>
</td></tr>
<tr><td><code id="estimate_hematocrit_+3A_reth">reth</code></td>
<td>
<p>NHANES race/ethnicity category for which to generate serum
creatinine values (&quot;Mexican American&quot;, &quot;Non-Hispanic Black&quot;, &quot;Non-Hispanic
White&quot;, &quot;Other&quot;, or &quot;Other Hispanic&quot;)</p>
</td></tr>
<tr><td><code id="estimate_hematocrit_+3A_age_years">age_years</code></td>
<td>
<p>Vector of ages in years for individuals for whom to generate
hematocrit values (corresponding to age_months)</p>
</td></tr>
<tr><td><code id="estimate_hematocrit_+3A_age_months">age_months</code></td>
<td>
<p>vector of ages in months for individuals for whom to
generate hematocrit values (between 0-959 months)</p>
</td></tr>
<tr><td><code id="estimate_hematocrit_+3A_nhanes_mec_svy">nhanes_mec_svy</code></td>
<td>
<p><code>surveydesign</code> object created from
<code><a href="#topic+mecdt">mecdt</a></code> using <code><a href="survey.html#topic+svydesign">svydesign</a></code> (this is done in
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function should usually not be called directly by the user. It is used by
<code>httkpop_generate()</code> in &quot;virtual-individuals&quot; mode, after drawing gender,
NHANES race/ethnicity category, and age from their NHANES
proportions/distributions.
</p>


<h3>Value</h3>

<p>A vector of numeric generated hematocrit values (blood percentage red
blood cells by volume).
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='example.seem'>SEEM Example Data
We can grab SEEM daily intake rate predictions already in RData format from
https://github.com/HumanExposure/SEEM3RPackage/tree/main/SEEM3/data
Download the file Ring2018Preds.RData</h2><span id='topic+example.seem'></span>

<h3>Description</h3>

<p>We do not have the space to distribute all the SEEM predictions within
this R package, but we can give you our &quot;Intro to IVIVE&quot; example chemicals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example.seem
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>References</h3>

<p>Ring CL, Arnot JA, Bennett DH, Egeghy PP, Fantke P, Huang L, Isaacs KK, Jolliet O, Phillips KA, Price PS, others (2018).
&ldquo;Consensus modeling of median chemical intake for the US population based on predictions of exposure pathways.&rdquo;
<em>Environmental science &amp; technology</em>, <b>53</b>(2), 719&ndash;732.
</p>

<hr>
<h2 id='example.toxcast'>ToxCast Example Data
The main page for the ToxCast data is here:
https://www.epa.gov/comptox-tools/exploring-toxcast-data
Most useful to us is a single file containing all the hits across all chemcials
and assays:
https://clowder.edap-cluster.com/datasets/6364026ee4b04f6bb1409eda?space=62bb560ee4b07abf29f88fef</h2><span id='topic+example.toxcast'></span>

<h3>Description</h3>

<p>As of November, 2022 the most recent version was 3.5 and was available as an
.Rdata file (invitrodb_3_5_mc5.Rdata)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example.toxcast
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Details</h3>

<p>Unfortunately for this vignette there are too many ToxCast data to fit into a
5mb R package. So we will subset to just the shemicals for the 
&quot;Intro to IVIVE&quot; vignette and distribute
only those data. In addition, out of 78 columns in the data, we will keep only 
eight.
</p>

<hr>
<h2 id='export_pbtk_jarnac'>Export model to jarnac.</h2><span id='topic+export_pbtk_jarnac'></span>

<h3>Description</h3>

<p>This function exports the multiple compartment PBTK model to a jarnac file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_pbtk_jarnac(
  chem.cas = NULL,
  chem.name = NULL,
  species = "Human",
  initial.amounts = list(Agutlumen = 0),
  filename = "default.jan",
  digits = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export_pbtk_jarnac_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name or CAS number must be specified.</p>
</td></tr>
<tr><td><code id="export_pbtk_jarnac_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or CAS number must be specified.</p>
</td></tr>
<tr><td><code id="export_pbtk_jarnac_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, or default
&quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="export_pbtk_jarnac_+3A_initial.amounts">initial.amounts</code></td>
<td>
<p>Must specify initial amounts in units of choice.</p>
</td></tr>
<tr><td><code id="export_pbtk_jarnac_+3A_filename">filename</code></td>
<td>
<p>The name of the jarnac file containing the model.</p>
</td></tr>
<tr><td><code id="export_pbtk_jarnac_+3A_digits">digits</code></td>
<td>
<p>Desired number of decimal places to round the parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Compartments to enter into the initial.amounts list includes Agutlumen,
Aart, Aven, Alung, Agut, Aliver, Akidney, and Arest.
</p>
<p>When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>


<h3>Value</h3>

<p>Text containing a Jarnac language version of the PBTK model.
</p>


<h3>Author(s)</h3>

<p>Robert Pearce
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

export_pbtk_jarnac(chem.name='Nicotine',initial.amounts=list(Agutlumen=1),filename='PBTKmodel.jan')


</code></pre>

<hr>
<h2 id='export_pbtk_sbml'>Export model to sbml.</h2><span id='topic+export_pbtk_sbml'></span>

<h3>Description</h3>

<p>This function exports the multiple compartment PBTK model to an sbml file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_pbtk_sbml(
  chem.cas = NULL,
  chem.name = NULL,
  species = "Human",
  initial.amounts = list(Agutlumen = 0),
  filename = "default.xml",
  digits = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export_pbtk_sbml_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name or CAS number must be specified.</p>
</td></tr>
<tr><td><code id="export_pbtk_sbml_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or CAS number must be specified.</p>
</td></tr>
<tr><td><code id="export_pbtk_sbml_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, or default
&quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="export_pbtk_sbml_+3A_initial.amounts">initial.amounts</code></td>
<td>
<p>Must specify initial amounts in units of choice.</p>
</td></tr>
<tr><td><code id="export_pbtk_sbml_+3A_filename">filename</code></td>
<td>
<p>The name of the jarnac file containing the model.</p>
</td></tr>
<tr><td><code id="export_pbtk_sbml_+3A_digits">digits</code></td>
<td>
<p>Desired number of decimal places to round the parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Compartments to enter into the initial.amounts list includes Agutlumen,
Aart, Aven, Alung, Agut, Aliver, Akidney, and Arest.
</p>
<p>When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>


<h3>Value</h3>

<p>Text describing the PBTK model in SBML.
</p>


<h3>Author(s)</h3>

<p>Robert Pearce
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

export_pbtk_sbml(chem.name='Nicotine',initial.amounts=list(Agutlumen=1),filename='PBTKmodel.xml')


</code></pre>

<hr>
<h2 id='fetalpcs'>Fetal Partition Coefficients</h2><span id='topic+fetalpcs'></span><span id='topic+fetalPCs'></span>

<h3>Description</h3>

<p>Partition coefficients were measured for tissues, including placenta, in 
vitro by Csanady et al. (2002) for Bisphenol A and Diadzen. Curley et al. 
(1969) measured the concentration of a variety of pesticides in the cord 
blood of newborns and in the tissues of infants that were stillborn.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fetalpcs
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Details</h3>

<p>Three of the chemicals studied by Curley et al. (1969) were modeled by 
Weijs et al. (2013) using the same partition coefficients for mother and 
fetus. The values used represented &quot;prior knowledge&quot; summarizing the 
available literature.
</p>


<h3>Source</h3>

<p>Kapraun DF, Sfeir M, Pearce RG, Davidson-Fritz SE, Lumen A, Dallmann A, Judson RS, Wambaugh JF (2022).
&ldquo;Evaluation of a rapid, generic human gestational dose model.&rdquo;
<em>Reproductive Toxicology</em>, <b>113</b>, 172&ndash;188.
</p>


<h3>References</h3>

<p>Csanady G, Oberste-Frielinghaus H, Semder B, Baur C, Schneider K, Filser J (2002).
&ldquo;Distribution and unspecific protein binding of the xenoestrogens bisphenol A and daidzein.&rdquo;
<em>Archives of toxicology</em>, <b>76</b>(5-6), 299&ndash;305.
Curley A, Copeland MF, Kimbrough RD (1969).
&ldquo;Chlorinated hydrocarbon insecticides in organs of stillborn and blood of newborn babies.&rdquo;
<em>Archives of Environmental Health: An International Journal</em>, <b>19</b>(5), 628&ndash;632.
Weijs L, Yang RS, Das K, Covaci A, Blust R (2013).
&ldquo;Application of Bayesian population physiologically based pharmacokinetic (PBPK) modeling and Markov chain Monte Carlo simulations to pesticide kinetics studies in protected marine mammals: DDT, DDE, and DDD in harbor porpoises.&rdquo;
<em>Environmental science &amp; technology</em>, <b>47</b>(9), 4365&ndash;4374.
</p>

<hr>
<h2 id='Frank2018invivo'>Literature In Vivo Data on Doses Causing Neurological Effects</h2><span id='topic+Frank2018invivo'></span>

<h3>Description</h3>

<p>Studies were selected from Table 1 in Mundy et al., 2015, as
the studies in that publication were cited as examples of
compounds with evidence for developmental neurotoxicity. There
were sufficient in vitro toxicokinetic data available for this
package for only 6 of the 42 chemicals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Frank2018invivo
</code></pre>


<h3>Format</h3>

<p>A data.frame containing 14 rows and 16 columns.
</p>


<h3>Author(s)</h3>

<p>Timothy J. Shafer
</p>


<h3>References</h3>

<p>Frank, Christopher L., et al. &quot;Defining toxicological tipping points
in neuronal network development.&quot; Toxicology and Applied
Pharmacology 354 (2018): 81-93.
</p>
<p>Mundy, William R., et al. &quot;Expanding the test set: Chemicals with
potential to disrupt mammalian brain development.&quot; Neurotoxicology
and Teratology 52 (2015): 25-35.
</p>

<hr>
<h2 id='gen_age_height_weight'>Generate demographic parameters for a virtual population</h2><span id='topic+gen_age_height_weight'></span>

<h3>Description</h3>

<p>Generate gender, NHANES race/ethnicity category, ages, heights, and weights
for a virtual population, based on NHANES data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gen_age_height_weight(
  nsamp = NULL,
  gendernum = NULL,
  reths,
  weight_category,
  agelim_years,
  agelim_months,
  nhanes_mec_svy
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gen_age_height_weight_+3A_nsamp">nsamp</code></td>
<td>
<p>The desired number of individuals in the virtual population.
<code>nsamp</code> need not be provided if <code>gendernum</code> is provided.</p>
</td></tr>
<tr><td><code id="gen_age_height_weight_+3A_gendernum">gendernum</code></td>
<td>
<p>Optional: A named list giving the numbers of male and female
individuals to include in the population, e.g. <code>list(Male=100,
Female=100)</code>. Default is NULL, meaning both males and females are included,
in their proportions in the NHANES data. If both <code>nsamp</code> and
<code>gendernum</code> are provided, they must agree (i.e., <code>nsamp</code> must be
the sum of <code>gendernum</code>).</p>
</td></tr>
<tr><td><code id="gen_age_height_weight_+3A_reths">reths</code></td>
<td>
<p>Optional: a character vector giving the races/ethnicities to
include in the population. Default is <code>c('Mexican American','Other
Hispanic','Non-Hispanic White','Non-Hispanic Black','Other')</code>, to include
all races and ethnicities in their proportions in the NHANES data.
User-supplied vector must contain one or more of these strings.</p>
</td></tr>
<tr><td><code id="gen_age_height_weight_+3A_weight_category">weight_category</code></td>
<td>
<p>Optional: The weight categories to include in the
population. Default is <code>c('Underweight', 'Normal', 'Overweight',
'Obese')</code>. User-supplied vector must contain one or more of these strings.</p>
</td></tr>
<tr><td><code id="gen_age_height_weight_+3A_agelim_years">agelim_years</code></td>
<td>
<p>Optional: A two-element numeric vector giving the minimum
and maximum ages (in years) to include in the population. Default is
c(0,79). If <code>agelim_years</code> is provided and <code>agelim_months</code> is not,
<code>agelim_years</code> will override the default value of <code>agelim_months</code>.</p>
</td></tr>
<tr><td><code id="gen_age_height_weight_+3A_agelim_months">agelim_months</code></td>
<td>
<p>Optional: A two-element numeric vector giving the minimum
and maximum ages (in months) to include in the population. Default is c(0,
959), equivalent to the default <code>agelim_years</code>. If <code>agelim_months</code>
is provided and <code>agelim_years</code> is not, agelim_months will override the
default values of <code>agelim_years</code>.</p>
</td></tr>
<tr><td><code id="gen_age_height_weight_+3A_nhanes_mec_svy">nhanes_mec_svy</code></td>
<td>
<p><code>surveydesign</code> object created from
<code><a href="#topic+mecdt">mecdt</a></code> using <code><a href="survey.html#topic+svydesign">svydesign</a></code> (this is done in
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function should usually not be called directly by the user. It is used by
<code>httkpop_generate()</code> in &quot;virtual-individuals&quot; mode.
</p>


<h3>Value</h3>

<p>A data.table containing variables </p>

<dl>
<dt><code>gender</code></dt><dd><p>Gender of each virtual individual</p>
</dd>
<dt><code>reth</code></dt><dd><p>Race/ethnicity of each virtual individual</p>
</dd>
<dt><code>age_months</code></dt><dd><p>Age in months of each virtual individual</p>
</dd>
<dt><code>age_years</code></dt><dd><p>Age in years of each virtual individual</p>
</dd>
<dt><code>weight</code></dt><dd><p>Body weight in kg of each virtual individual</p>
</dd>
<dt><code>height</code></dt><dd><p>Height in cm of each virtual individual</p>
</dd> </dl>



<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118. 
</p>
<p>importFrom survey svymean
</p>

<hr>
<h2 id='gen_height_weight'>Generate heights and weights for a virtual population.</h2><span id='topic+gen_height_weight'></span>

<h3>Description</h3>

<p>Predict height and weight from age using smoothing splines, and then add
residual variability from a 2-D KDE, both fitted to NHANES data, for a given
combination of gender and NHANES race/ethnicity category.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gen_height_weight(gender, reth, age_months, nhanes_mec_svy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gen_height_weight_+3A_gender">gender</code></td>
<td>
<p>Gender for which to calculate height/weight (&quot;Male&quot; or &quot;Female&quot;)</p>
</td></tr>
<tr><td><code id="gen_height_weight_+3A_reth">reth</code></td>
<td>
<p>NHANES race/ethnicity category for which to calculate
height/weight (&quot;Mexican American&quot;, &quot;Non-Hispanic Black&quot;, &quot;Non-Hispanic
White&quot;, &quot;Other&quot;, or &quot;Other Hispanic&quot;)</p>
</td></tr>
<tr><td><code id="gen_height_weight_+3A_age_months">age_months</code></td>
<td>
<p>vector of ages in months for individuals for whom to
calculate height/weight (between 0-959 months)</p>
</td></tr>
<tr><td><code id="gen_height_weight_+3A_nhanes_mec_svy">nhanes_mec_svy</code></td>
<td>
<p><code>surveydesign</code> object created from
<code><a href="#topic+mecdt">mecdt</a></code> using <code><a href="survey.html#topic+svydesign">svydesign</a></code> (this is done in
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function should usually not be called directly by the user. It is used by
<code>httkpop_generate()</code> in &quot;virtual-individuals&quot; mode, after drawing gender,
NHANES race/ethnicity category, and age from their NHANES
proportions/distributions.
</p>


<h3>Value</h3>

<p>A list containing two named elements, <code>weight</code> and <code>height</code>,
each of which is a numeric vector. <code>weight</code> gives individual body
weights in kg, and <code>height</code> gives individual heights in cm,
corresponding to each item in the input <code>age_months</code>.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='gen_serum_creatinine'>Generate serum creatinine values for a virtual population.</h2><span id='topic+gen_serum_creatinine'></span>

<h3>Description</h3>

<p>Predict serum creatinine from age using smoothing splines and kernel density
estimates of residual variability fitted to NHANES data,, for a given
combination of gender and NHANES race/ethnicity category.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gen_serum_creatinine(gender, reth, age_years, age_months, nhanes_mec_svy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gen_serum_creatinine_+3A_gender">gender</code></td>
<td>
<p>Gender for which to generate serum creatinine values (&quot;Male&quot; or
&quot;Female&quot;)</p>
</td></tr>
<tr><td><code id="gen_serum_creatinine_+3A_reth">reth</code></td>
<td>
<p>NHANES race/ethnicity category for which to generate serum
creatinine values (&quot;Mexican American&quot;, &quot;Non-Hispanic Black&quot;, &quot;Non-Hispanic
White&quot;, &quot;Other&quot;, or &quot;Other Hispanic&quot;)</p>
</td></tr>
<tr><td><code id="gen_serum_creatinine_+3A_age_years">age_years</code></td>
<td>
<p>Vector of ages in years for individuals for whom to generate
serum creatinine values (corresponding to age_months)</p>
</td></tr>
<tr><td><code id="gen_serum_creatinine_+3A_age_months">age_months</code></td>
<td>
<p>vector of ages in months for individuals for whom to
generate serum creatinine values (between 0-959 months)</p>
</td></tr>
<tr><td><code id="gen_serum_creatinine_+3A_nhanes_mec_svy">nhanes_mec_svy</code></td>
<td>
<p><code>surveydesign</code> object created from
<code><a href="#topic+mecdt">mecdt</a></code> using <code><a href="survey.html#topic+svydesign">svydesign</a></code> (this is done in
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function should usually not be called directly by the user. It is used
by <code>httkpop_generate()</code> in &quot;virtual-individuals&quot; mode, after drawing
gender, NHANES race/ethnicity category, and age from their NHANES
proportions/distributions.
</p>


<h3>Value</h3>

<p>A vector of numeric generated serum creatinine values (mg/dL).
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='get_caco2'>Retrieve in vitro measured Caco-2 membrane permeabilit</h2><span id='topic+get_caco2'></span>

<h3>Description</h3>

<p>This function checks for chemical-specific in vitro measurements of the 
Caco-2 membrane permeability 
in the <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> table. If no value is 
available argument <code>Caco2.Pab.default</code> is returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_caco2(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  Caco2.Pab.default = 1.6,
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_caco2_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="get_caco2_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="get_caco2_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="get_caco2_+3A_caco2.pab.default">Caco2.Pab.default</code></td>
<td>
<p>sets the default value for  Caco2.Pab if Caco2.Pab is 
unavailable.</p>
</td></tr>
<tr><td><code id="get_caco2_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>

<hr>
<h2 id='get_chem_id'>Retrieve chemical identity from HTTK package</h2><span id='topic+get_chem_id'></span>

<h3>Description</h3>

<p>Given one of chem.name, chem.cas (Chemical Abstract Service Registry Number),
or DTXSID (DSStox Substance Identifier <a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>) this
function checks if the chemical is available and, if so, returns all three
pieces of information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_chem_id(chem.cas = NULL, chem.name = NULL, dtxsid = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_chem_id_+3A_chem.cas">chem.cas</code></td>
<td>
<p>CAS regstry number</p>
</td></tr>
<tr><td><code id="get_chem_id_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name</p>
</td></tr>
<tr><td><code id="get_chem_id_+3A_dtxsid">dtxsid</code></td>
<td>
<p>DSSTox Substance identifier</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following chemical identifiers:
</p>
<table>
<tr><td><code>chem.cas</code></td>
<td>
<p>CAS registry number</p>
</td></tr>
<tr><td><code>chem.name</code></td>
<td>
<p>Name</p>
</td></tr>
<tr><td><code>dtxsid</code></td>
<td>
<p>DTXSID</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>

<hr>
<h2 id='get_cheminfo'>Retrieve chemical information available from HTTK package</h2><span id='topic+get_cheminfo'></span>

<h3>Description</h3>

<p>This function lists information on all the chemicals within HTTK for which 
there are sufficient data for the specified model and species. 
By default the function returns only CAS (that is, info=&quot;CAS&quot;). 
The type of information available includes chemical identifiers 
(&quot;Compound&quot;, &quot;CASRN&quot;, &quot;DTXSID&quot;), in vitro
measurements (&quot;Clint&quot;, &quot;Clint.pvalue&quot;, &quot;Funbound plasma&quot;, &quot;Rblood2plasma&quot;), 
and physico-chemical information (&quot;Formula&quot;, &quot;logMA&quot;, &quot;logP&quot;, &quot;MW&quot;,
&quot;pKa_Accept&quot;, &quot;pKa_Donor&quot;). The argument &quot;info&quot; can be a single type of 
information, &quot;all&quot; information, or a vector of specific types of information.
The argument &quot;model&quot; defaults to 
&quot;3compartmentss&quot; and the argument &quot;species&quot; defaults to &quot;human&quot;.  
Since different models have different 
requirements and not all chemicals have complete data, this function will 
return different numbers of chemicals depending on the model specified. If
a chemical is not listed by get_cheminfo then either the in vitro or
physico-chemical data needed are currently missing (but could potentially
be added using <code><a href="#topic+add_chemtable">add_chemtable</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cheminfo(
  info = "CAS",
  species = "Human",
  fup.lod.default = 0.005,
  model = "3compartmentss",
  default.to.human = FALSE,
  median.only = FALSE,
  fup.ci.cutoff = TRUE,
  clint.pvalue.threshold = 0.05,
  class.exclude = TRUE,
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cheminfo_+3A_info">info</code></td>
<td>
<p>A single character vector (or collection of character vectors)
from &quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID, &quot;logP&quot;, &quot;pKa_Donor&quot;,&quot; pKa_Accept&quot;, &quot;MW&quot;, &quot;Clint&quot;,
&quot;Clint.pValue&quot;, &quot;Funbound.plasma&quot;,&quot;Structure_Formula&quot;, or &quot;Substance_Type&quot;. info=&quot;all&quot;
gives all information for the model and species.</p>
</td></tr>
<tr><td><code id="get_cheminfo_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="get_cheminfo_+3A_fup.lod.default">fup.lod.default</code></td>
<td>
<p>Default value used for fraction of unbound plasma for
chemicals where measured value was below the limit of detection. Default
value is 0.0005.</p>
</td></tr>
<tr><td><code id="get_cheminfo_+3A_model">model</code></td>
<td>
<p>Model used in calculation, 'pbtk' for the multiple compartment
model, '1compartment' for the one compartment model, '3compartment' for
three compartment model, '3compartmentss' for the three compartment model
without partition coefficients, or 'schmitt' for chemicals with logP and
fraction unbound (used in predict_partitioning_schmitt).</p>
</td></tr>
<tr><td><code id="get_cheminfo_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing values with human values if
true.</p>
</td></tr>
<tr><td><code id="get_cheminfo_+3A_median.only">median.only</code></td>
<td>
<p>Use median values only for fup and clint.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="get_cheminfo_+3A_fup.ci.cutoff">fup.ci.cutoff</code></td>
<td>
<p>Cutoff for the level of uncertainty in fup estimates.
This value should be between (0,1). Default is 'NULL' specifying no filtering.</p>
</td></tr>
<tr><td><code id="get_cheminfo_+3A_clint.pvalue.threshold">clint.pvalue.threshold</code></td>
<td>
<p>Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.</p>
</td></tr>
<tr><td><code id="get_cheminfo_+3A_class.exclude">class.exclude</code></td>
<td>
<p>Exclude chemical classes identified as outside of 
domain of applicability by relevant modelinfo_[MODEL] file (default TRUE).</p>
</td></tr>
<tr><td><code id="get_cheminfo_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output messages are suppressed 
(default FALSE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When default.to.human is set to TRUE, and the species-specific data,
Funbound.plasma and Clint, are missing from 
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>, human values are given instead.
</p>
<p>In some cases the rapid equilibrium dialysis method (Waters et al., 2008)
fails to yield detectable concentrations for the free fraction of chemical. 
In those cases we assume the compound is highly bound (that is, Fup approaches
zero). For some calculations (for example, steady-state plasma concentration)
there is precedent (Rotroff et al., 2010) for using half the average limit 
of detection, that is, 0.005 (this value is configurable via the argument
fup.lod.default). We do not recommend using other models where 
quantities like partition coefficients must be predicted using Fup. We also
do not recommend including the value 0.005 in training sets for Fup predictive
models.
</p>
<p><strong>Note</strong> that in some cases the <strong>Funbound.plasma</strong> (fup) and the 
<strong>intrinsic clearance</strong> (clint) are
<em>provided as a series of numbers separated by commas</em>. These values are the 
result of Bayesian analysis and characterize a distribution: the first value
is the median of the distribution, while the second and third values are the 
lower and upper 95th percentile (that is quantile 2.5 and 97.5) respectively.
For intrinsic clearance a fourth value indicating a p-value for a decrease is
provided. Typically 4000 samples were used for the Bayesian analysis, such
that a p-value of &quot;0&quot; is equivalent to &quot;&lt;0.00025&quot;. See Wambaugh et al. (2019)
for more details. If argument median.only == TRUE then only the median is
reported for parameters with Bayesian analysis distributions. If the 95
credible interval is larger than fup.ci.cutoff (defaults
to NULL) then the Fup is treated as too uncertain and the value NA is given.
</p>


<h3>Value</h3>

<table>
<tr><td><code>vector/data.table</code></td>
<td>
<p>Table (if info has multiple entries) or 
vector containing a column for each valid entry 
specified in the argument &quot;info&quot; and a row for each chemical with sufficient
data for the model specified by argument &quot;model&quot;:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Column</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td><td style="text-align: left;"> <strong>units</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
Compound </td><td style="text-align: left;"> The preferred name of the chemical compound </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;"> 
CAS </td><td style="text-align: left;"> The preferred Chemical Abstracts Service Registry Number </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;">  
DTXSID </td><td style="text-align: left;"> DSSTox Structure ID 
(<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>) </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;"> 
logP </td><td style="text-align: left;"> The log10 octanol:water partition coefficient</td><td style="text-align: left;"> log10 unitless ratio </td>
</tr>
<tr>
 <td style="text-align: left;"> 
MW </td><td style="text-align: left;"> The chemical compound molecular weight </td><td style="text-align: left;"> g/mol </td>
</tr>
<tr>
 <td style="text-align: left;"> 
pKa_Accept </td><td style="text-align: left;"> The hydrogen acceptor equilibria concentrations 
</td><td style="text-align: left;"> logarithm </td>
</tr>
<tr>
 <td style="text-align: left;">   
pKa_Donor </td><td style="text-align: left;"> The hydrogen donor equilibria concentrations 
 </td><td style="text-align: left;"> logarithm </td>
</tr>
<tr>
 <td style="text-align: left;">   
[SPECIES].Clint </td><td style="text-align: left;"> (Primary hepatocyte suspension) 
intrinsic hepatic clearance. <em>Entries with comma separated values are Bayesian estimates of
the Clint distribution - displayed as the median, 95th credible interval
(that is quantile 2.5 and 97.5, respectively), and p-value.</em> </td><td style="text-align: left;"> uL/min/10^6 hepatocytes </td>
</tr>
<tr>
 <td style="text-align: left;">    
[SPECIES].Clint.pValue </td><td style="text-align: left;"> Probability that there is no clearance observed.
Values close to 1 indicate clearance is not statistically significant. </td><td style="text-align: left;"> none </td>
</tr>
<tr>
 <td style="text-align: left;">  
[SPECIES].Funbound.plasma </td><td style="text-align: left;"> Chemical fraction unbound in presence of 
plasma proteins (fup). <em>Entries with comma separated values are Bayesian estimates of
the fup distribution - displayed as the median and 95th credible interval
(that is quantile 2.5 and 97.5, respectively).</em> </td><td style="text-align: left;"> unitless fraction </td>
</tr>
<tr>
 <td style="text-align: left;"> 
[SPECIES].Rblood2plasma </td><td style="text-align: left;"> Chemical concentration blood to plasma ratio </td><td style="text-align: left;"> unitless ratio </td>
</tr>
<tr>
 <td style="text-align: left;">  
</td>
</tr>

</table>

</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh, Robert Pearce, and Sarah E. Davidson
</p>


<h3>References</h3>

<p>Rotroff, Daniel M., et al. &quot;Incorporating human dosimetry and exposure into 
high-throughput in vitro toxicity screening.&quot; Toxicological Sciences 117.2 
(2010): 348-358.
</p>
<p>Waters, Nigel J., et al. &quot;Validation of a rapid equilibrium dialysis approach 
for the measurement of plasma protein binding.&quot; Journal of pharmaceutical 
sciences 97.10 (2008): 4586-4595.
</p>
<p>Wambaugh, John F., et al. &quot;Assessing toxicokinetic uncertainty and 
variability in risk prioritization.&quot; Toxicological Sciences 172.2 (2019): 
235-251.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# List all CAS numbers for which the 3compartmentss model can be run in humans: 
get_cheminfo()

get_cheminfo(info=c('compound','funbound.plasma','logP'),model='pbtk') 
# See all the data for humans:
get_cheminfo(info="all")

TPO.cas &lt;- c("741-58-2", "333-41-5", "51707-55-2", "30560-19-1", "5598-13-0", 
"35575-96-3", "142459-58-3", "1634-78-2", "161326-34-7", "133-07-3", "533-74-4", 
"101-05-3", "330-54-1", "6153-64-6", "15299-99-7", "87-90-1", "42509-80-8", 
"10265-92-6", "122-14-5", "12427-38-2", "83-79-4", "55-38-9", "2310-17-0", 
"5234-68-4", "330-55-2", "3337-71-1", "6923-22-4", "23564-05-8", "101-02-0", 
"140-56-7", "120-71-8", "120-12-7", "123-31-9", "91-53-2", "131807-57-3", 
"68157-60-8", "5598-15-2", "115-32-2", "298-00-0", "60-51-5", "23031-36-9", 
"137-26-8", "96-45-7", "16672-87-0", "709-98-8", "149877-41-8", "145701-21-9", 
"7786-34-7", "54593-83-8", "23422-53-9", "56-38-2", "41198-08-7", "50-65-7", 
"28434-00-6", "56-72-4", "62-73-7", "6317-18-6", "96182-53-5", "87-86-5", 
"101-54-2", "121-69-7", "532-27-4", "91-59-8", "105-67-9", "90-04-0", 
"134-20-3", "599-64-4", "148-24-3", "2416-94-6", "121-79-9", "527-60-6", 
"99-97-8", "131-55-5", "105-87-3", "136-77-6", "1401-55-4", "1948-33-0", 
"121-00-6", "92-84-2", "140-66-9", "99-71-8", "150-13-0", "80-46-6", "120-95-6",
"128-39-2", "2687-25-4", "732-11-6", "5392-40-5", "80-05-7", "135158-54-2", 
"29232-93-7", "6734-80-1", "98-54-4", "97-53-0", "96-76-4", "118-71-8", 
"2451-62-9", "150-68-5", "732-26-3", "99-59-2", "59-30-3", "3811-73-2", 
"101-61-1", "4180-23-8", "101-80-4", "86-50-0", "2687-96-9", "108-46-3", 
"95-54-5", "101-77-9", "95-80-7", "420-04-2", "60-54-8", "375-95-1", "120-80-9",
"149-30-4", "135-19-3", "88-58-4", "84-16-2", "6381-77-7", "1478-61-1", 
"96-70-8", "128-04-1", "25956-17-6", "92-52-4", "1987-50-4", "563-12-2", 
"298-02-2", "79902-63-9", "27955-94-8")
httk.TPO.rat.table &lt;- subset(get_cheminfo(info="all",species="rat"),
 CAS %in% TPO.cas)
 
httk.TPO.human.table &lt;- subset(get_cheminfo(info="all",species="human"),
 CAS %in% TPO.cas)
 
# create a data.frame with all the Fup values, we ask for model="schmitt" since
# that model only needs fup, we ask for "median.only" because we don't care
# about uncertainty intervals here:
fup.tab &lt;- get_cheminfo(info="all",median.only=TRUE,model="schmitt")
# calculate the median, making sure to convert to numeric values:
median(as.numeric(fup.tab$Human.Funbound.plasma),na.rm=TRUE)
# calculate the mean:
mean(as.numeric(fup.tab$Human.Funbound.plasma),na.rm=TRUE)
# count how many non-NA values we have (should be the same as the number of 
# rows in the table but just in case we ask for non NA values:
sum(!is.na(fup.tab$Human.Funbound.plasma))


</code></pre>

<hr>
<h2 id='get_clint'>Retrieve and parse intrinsic hepatic clearance</h2><span id='topic+get_clint'></span>

<h3>Description</h3>

<p>This function retrieves the chemical- and species-specific intinsic
hepatic clearance (Cl<sub>int</sub>,
inits of uL/min/million hepatocytes)  
from <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>. 
If that parameter is described by a distribution (that is, a median, 
lower-, upper-95th percentile and p-value separated by commas) this 
function splits those quantiles into separate values. Most 
Cl<sub>int</sub> values have an
accompanying p-value indicating the probability that no decrease was 
observed. If the p-values exceeds a threhsold (default 0.05) the clearance is
set to zero (no clearance). Some values extracted from the literature do not
have a p-value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clint(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  force.human.clint = FALSE,
  suppress.messages = FALSE,
  clint.pvalue.threshold = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_clint_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="get_clint_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="get_clint_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="get_clint_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="get_clint_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing hepatic clearance with
human values if true.</p>
</td></tr>
<tr><td><code id="get_clint_+3A_force.human.clint">force.human.clint</code></td>
<td>
<p>If a non-human species value (matching argument
species) is available, it is ignored and the human intrinsic clearance is 
used</p>
</td></tr>
<tr><td><code id="get_clint_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="get_clint_+3A_clint.pvalue.threshold">clint.pvalue.threshold</code></td>
<td>
<p>Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing:
</p>
<table>
<tr><td><code>CLint.point</code></td>
<td>
<p>Point estimate (central tendency) of the intrinsic hepatic clearance</p>
</td></tr>
<tr><td><code>Clint.dist</code></td>
<td>
<p>Quantiles of a distribution (median, lower, upper 95th percentiles) and pvalue</p>
</td></tr>
<tr><td><code>Clint.pvalue</code></td>
<td>
<p>pvalue for whether disapperance of parent compound was observed</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>See Also</h3>

<p><code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>
</p>

<hr>
<h2 id='get_fabsgut'>Retrieve or calculate fraction of chemical absorbed from the gut</h2><span id='topic+get_fabsgut'></span>

<h3>Description</h3>

<p>This function checks for chemical-specific in vivo measurements of the 
fraction absorbed from the
gut in the <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> table. If in vivo
data are unavailable (or <code>keepit100 == TRUE</code>) we attempt to use 
in vitro Caco-2 membrane permeability to predict the fractions according to
<code><a href="#topic+calc_fbio.oral">calc_fbio.oral</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_fabsgut(
  parameters = NULL,
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  Caco2.Pab.default = 1.6,
  Caco2.Fgut = TRUE,
  Caco2.Fabs = TRUE,
  overwrite.invivo = FALSE,
  keepit100 = FALSE,
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_fabsgut_+3A_parameters">parameters</code></td>
<td>
<p>A list of the parameters (Caco2.Pab, Funbound.Plasma, Rblood2plasma,
Clint, BW, Qsmallintestine, Fabs, Fgut) used in the calculation, either supplied by user
or calculated in parameterize_steady_state.</p>
</td></tr>
<tr><td><code id="get_fabsgut_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="get_fabsgut_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="get_fabsgut_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="get_fabsgut_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="get_fabsgut_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing rat values with human values if
true.</p>
</td></tr>
<tr><td><code id="get_fabsgut_+3A_caco2.pab.default">Caco2.Pab.default</code></td>
<td>
<p>sets the default value for  Caco2.Pab if Caco2.Pab is 
unavailable.</p>
</td></tr>
<tr><td><code id="get_fabsgut_+3A_caco2.fgut">Caco2.Fgut</code></td>
<td>
<p>= TRUE uses Caco2.Pab to calculate  fgut.oral, otherwise 
fgut.oral = <code>Fgut</code>.</p>
</td></tr>
<tr><td><code id="get_fabsgut_+3A_caco2.fabs">Caco2.Fabs</code></td>
<td>
<p>= TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>.</p>
</td></tr>
<tr><td><code id="get_fabsgut_+3A_overwrite.invivo">overwrite.invivo</code></td>
<td>
<p>= TRUE overwrites Fabs and Fgut in vivo values from 
literature with Caco2 derived values if available.</p>
</td></tr>
<tr><td><code id="get_fabsgut_+3A_keepit100">keepit100</code></td>
<td>
<p>TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) 
regardless of other settings.</p>
</td></tr>
<tr><td><code id="get_fabsgut_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Greg Honda and John Wambaugh
</p>

<hr>
<h2 id='get_fup'>Retrieve and parse fraction unbound in plasma</h2><span id='topic+get_fup'></span>

<h3>Description</h3>

<p>This function retrieves the chemical- and species-specific fraction
unbound in plasma (f<sub>up</sub>) 
from <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>. 
If that parameter is described by a distribution (that is, a median, 
lower-, and upper-95th percentile separated by commas) this 
function splits those quantiles into separate values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_fup(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  force.human.fup = FALSE,
  suppress.messages = FALSE,
  minimum.Funbound.plasma = 1e-04
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_fup_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="get_fup_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="get_fup_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="get_fup_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="get_fup_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing fraction of unbound plasma with
human values if true.</p>
</td></tr>
<tr><td><code id="get_fup_+3A_force.human.fup">force.human.fup</code></td>
<td>
<p>If a non-human species value (matching argument
species) is available, it is ignored and the human fraction unbound is
returned</p>
</td></tr>
<tr><td><code id="get_fup_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="get_fup_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
f<sub>up</sub>
<p>is not allowed to drop below this value (default is 0.0001).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing:
</p>
<table>
<tr><td><code>Funbound.plasma.point</code></td>
<td>
<p>Point estimate (central tendency) of the Unbound fraction in plasma</p>
</td></tr>
<tr><td><code>Funbound.plasma.dist</code></td>
<td>
<p>Quantiles of a distribution (median, lower and upper 95th percentiles) for the unbound fraction</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>See Also</h3>

<p><code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>
</p>

<hr>
<h2 id='get_gfr_category'>Categorize kidney function by GFR.</h2><span id='topic+get_gfr_category'></span>

<h3>Description</h3>

<p>For adults: 
In general GFR &gt; 60 is considered normal 
15 &lt; GFR &lt; 60 is considered kidney disease 
GFR &lt; 15 is considered kidney failure
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gfr_category(age_years, age_months, gfr_est)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gfr_category_+3A_age_years">age_years</code></td>
<td>
<p>Vector of ages in years.</p>
</td></tr>
<tr><td><code id="get_gfr_category_+3A_age_months">age_months</code></td>
<td>
<p>Vector of ages in months.</p>
</td></tr>
<tr><td><code id="get_gfr_category_+3A_gfr_est">gfr_est</code></td>
<td>
<p>Vector of estimated GFR values in mL/min/1.73m^2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These values can also be used for children 2 years old and greater (see PEDIATRICS IN
REVIEW Vol. 29 No. 10 October 1, 2008 pp. 335-341 (doi:
10.1542/pir.29-10-335))
</p>


<h3>Value</h3>

<p>Vector of GFR categories: 'Normal', 'Kidney Disease', 'Kidney
Failure'.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='get_invitroPK_param'>Retrieve species-specific in vitro data from chem.physical_and_invitro.data table</h2><span id='topic+get_invitroPK_param'></span>

<h3>Description</h3>

<p>This function retrieves in vitro PK data (for example, intrinsic metabolic clearance 
or fraction unbound in plasma) for the the chemical specified by argument &quot;chem.name&quot;, &quot;dtxsid&quot;, 
or chem.cas from the table <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>.
This function looks for species-specific values based on the argument &quot;species&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_invitroPK_param(
  param,
  species,
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_invitroPK_param_+3A_param">param</code></td>
<td>
<p>The desired parameters, a vector or single value.</p>
</td></tr>
<tr><td><code id="get_invitroPK_param_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="get_invitroPK_param_+3A_chem.name">chem.name</code></td>
<td>
<p>The chemical names that you want parameters for, a vector or single value</p>
</td></tr>
<tr><td><code id="get_invitroPK_param_+3A_chem.cas">chem.cas</code></td>
<td>
<p>The chemical CAS numbers that you want parameters for, a vector or single value</p>
</td></tr>
<tr><td><code id="get_invitroPK_param_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this function works with a local version of the 
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> table to allow users to 
add/modify chemical
data (for example, adding new data via <code><a href="#topic+add_chemtable">add_chemtable</a></code> or 
loading in silico predictions distributed with httk via
<code><a href="#topic+load_sipes2017">load_sipes2017</a></code>, <code><a href="#topic+load_pradeep2020">load_pradeep2020</a></code>,
<code><a href="#topic+load_dawson2021">load_dawson2021</a></code>, or <code><a href="#topic+load_honda2023">load_honda2023</a></code>).
</p>
<p>User can request via argument param (case-insensitive):
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Parameter</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td><td style="text-align: left;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
 [SPECIES].Clint </td><td style="text-align: left;">  (Primary hepatocyte suspension) 
intrinsic hepatic clearance. <em>Entries with comma separated values are Bayesian estimates of
the Clint distribution - displayed as the median, 95th credible interval
(that is quantile 2.5 and 97.5, respectively), and p-value.</em> </td><td style="text-align: left;">  uL/min/10^6 hepatocytes </td>
</tr>
<tr>
 <td style="text-align: left;">                   
 [SPECIES].Clint.pValue </td><td style="text-align: left;">  Probability that there is no clearance observed.
 Values close to 1 indicate clearance is not statistically significant. </td><td style="text-align: left;">  none </td>
</tr>
<tr>
 <td style="text-align: left;">       
 [SPECIES].Caco2.Pab </td><td style="text-align: left;">  Caco-2 Apical-to-Basal Membrane Permeability </td><td style="text-align: left;">  10^-6 cm/s </td>
</tr>
<tr>
 <td style="text-align: left;">            
 [SPECIES].Fabs </td><td style="text-align: left;">  In vivo measured fraction of an oral dose of chemical 
absorbed from the gut lumen into the gut </td><td style="text-align: left;">  unitless fraction </td>
</tr>
<tr>
 <td style="text-align: left;">            
 [SPECIES].Fgut </td><td style="text-align: left;">  In vivo measured fraction of an oral dose of chemical 
that passes gut metabolism and clearance </td><td style="text-align: left;">  unitless fraction </td>
</tr>
<tr>
 <td style="text-align: left;">          
 [SPECIES].Foral </td><td style="text-align: left;">  In vivo measued fractional systemic bioavailability of 
an oral dose, modeled as he product of Fabs * Fgut * Fhep (where Fhep is 
first pass hepatic metabolism). </td><td style="text-align: left;">  unitless fraction </td>
</tr>
<tr>
 <td style="text-align: left;">
 [SPECIES].Funbound.plasma </td><td style="text-align: left;">  Chemical fraction unbound in presence of 
plasma proteins (fup). <em>Entries with comma separated values are Bayesian estimates of
the fup distribution - displayed as the median and 95th credible interval
(that is quantile 2.5 and 97.5, respectively).</em> </td><td style="text-align: left;">  unitless fraction </td>
</tr>
<tr>
 <td style="text-align: left;">
 [SPECIES].Rblood2plasma </td><td style="text-align: left;">  Chemical concentration blood to plasma ratio </td><td style="text-align: left;">  unitless ratio </td>
</tr>
<tr>
 <td style="text-align: left;">       
</td>
</tr>

</table>



<h3>Value</h3>

<p>The parameters, either a single value, a named list for a single chemical, or a list of lists
</p>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>See Also</h3>

<p><code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>
</p>
<p><code><a href="#topic+get_invitroPK_param">get_invitroPK_param</a></code>
</p>
<p><code><a href="#topic+add_chemtable">add_chemtable</a></code>
</p>

<hr>
<h2 id='get_lit_cheminfo'>Get literature Chemical Information.</h2><span id='topic+get_lit_cheminfo'></span>

<h3>Description</h3>

<p>This function provides the information specified in &quot;info=&quot; for all
chemicals with data from the Wetmore et al. (2012) and (2013) publications
and other literature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_lit_cheminfo(info = "CAS", species = "Human")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_lit_cheminfo_+3A_info">info</code></td>
<td>
<p>A single character vector (or collection of character vectors)
from
&quot;Compound&quot;, &quot;CAS&quot;, &quot;MW&quot;, &quot;Raw.Experimental.Percentage.Unbound&quot;,
&quot;Entered.Experimental.Percentage.Unbound&quot;, &quot;Fub&quot;, &quot;source_PPB&quot;,
&quot;Renal_Clearance&quot;, &quot;Met_Stab&quot;, &quot;Met_Stab_entered&quot;,
&quot;r2&quot;, &quot;p.val&quot;, &quot;Concentration..uM.&quot;, &quot;Css_lower_5th_perc.mg.L.&quot;, 
&quot;Css_median_perc.mg.L.&quot;, &quot;Css_upper_95th_perc.mg.L.&quot;,
&quot;Css_lower_5th_perc.uM.&quot;,&quot;Css_median_perc.uM.&quot;,&quot;Css_upper_95th_perc.uM.&quot;,
and &quot;Species&quot;.</p>
</td></tr>
<tr><td><code id="get_lit_cheminfo_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot; or default &quot;Human&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>info</code></td>
<td>
<p>Table/vector containing values specified in &quot;info&quot; for
valid chemicals.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., &quot;Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment,&quot; Toxicological Sciences 125
157-174 (2012)
</p>
<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Li, L., Clewell, H.J. III,
Judson, R.S., Freeman, K., Bao, W, Sochaski, M.A., Chu T.-M., Black, M.B.,
Healy, E, Allen, B., Andersen M.E., Wolfinger, R.D., and Thomas R.S., &quot;The
Relative Impact of Incorporating Pharmacokinetics on Predicting in vivo
Hazard and Mode-of-Action from High-Throughput in vitro Toxicity Assays&quot;
Toxicological Sciences, 132:327-346 (2013).
</p>
<p>Wetmore, B. A., Wambaugh, J. F., Allen, B., Ferguson, S. S., Sochaski, M.
A., Setzer, R. W., Houck, K. A., Strope, C. L., Cantwell, K., Judson, R. S.,
LeCluyse, E., Clewell, H.J. III, Thomas, R.S., and Andersen, M. E. (2015).
&quot;Incorporating High-Throughput Exposure Predictions with Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing&quot; Toxicological
Sciences, kfv171.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
get_lit_cheminfo()
get_lit_cheminfo(info=c('CAS','MW'))

</code></pre>

<hr>
<h2 id='get_lit_css'>Get literature Css</h2><span id='topic+get_lit_css'></span>

<h3>Description</h3>

<p>This function retrieves a steady-state plasma concentration as a result of
infusion dosing from the Wetmore et al. (2012) and (2013) publications and
other literature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_lit_css(
  chem.cas = NULL,
  chem.name = NULL,
  daily.dose = 1,
  which.quantile = 0.95,
  species = "Human",
  clearance.assay.conc = NULL,
  output.units = "mg/L",
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_lit_css_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the cas number or the chemical name must be
specified.</p>
</td></tr>
<tr><td><code id="get_lit_css_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="get_lit_css_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose infused in units of mg/kg BW/day.
Defaults to 1 mg/kg/day.</p>
</td></tr>
<tr><td><code id="get_lit_css_+3A_which.quantile">which.quantile</code></td>
<td>
<p>Which quantile from the SimCYP Monte Carlo simulation
is requested. Can be a vector.</p>
</td></tr>
<tr><td><code id="get_lit_css_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot; or default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="get_lit_css_+3A_clearance.assay.conc">clearance.assay.conc</code></td>
<td>
<p>Concentration of chemical used in measureing
intrinsic clearance data, 1 or 10 uM.</p>
</td></tr>
<tr><td><code id="get_lit_css_+3A_output.units">output.units</code></td>
<td>
<p>Returned units for function, defaults to mg/L but can
also be uM (specify units = &quot;uM&quot;).</p>
</td></tr>
<tr><td><code id="get_lit_css_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector with the literature steady-state plasma 
concentration (1 mg/kg/day) for the requested quantiles
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., &quot;Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment,&quot; Toxicological Sciences 125
157-174 (2012)
</p>
<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Li, L., Clewell, H.J. III,
Judson, R.S., Freeman, K., Bao, W, Sochaski, M.A., Chu T.-M., Black, M.B.,
Healy, E, Allen, B., Andersen M.E., Wolfinger, R.D., and Thomas R.S., &quot;The
Relative Impact of Incorporating Pharmacokinetics on Predicting in vivo
Hazard and Mode-of-Action from High-Throughput in vitro Toxicity Assays&quot;
Toxicological Sciences, 132:327-346 (2013).
</p>
<p>Wetmore, B. A., Wambaugh, J. F., Allen, B., Ferguson, S. S., Sochaski, M.
A., Setzer, R. W., Houck, K. A., Strope, C. L., Cantwell, K., Judson, R. S.,
LeCluyse, E., Clewell, H.J. III, Thomas, R.S., and Andersen, M. E. (2015).
&quot;Incorporating High-Throughput Exposure Predictions with Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing&quot; Toxicological
Sciences, kfv171.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_lit_css(chem.cas="34256-82-1")

get_lit_css(chem.cas="34256-82-1",species="Rat",which.quantile=0.5)

get_lit_css(chem.cas="80-05-7", daily.dose = 1,which.quantile = 0.5, output.units = "uM")

</code></pre>

<hr>
<h2 id='get_lit_oral_equiv'>Get Literature Oral Equivalent Dose</h2><span id='topic+get_lit_oral_equiv'></span>

<h3>Description</h3>

<p>This function converts a chemical plasma concetration to an oral equivalent
dose using the values from the Wetmore et al. (2012) and (2013) publications
and other literature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_lit_oral_equiv(
  conc,
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  suppress.messages = FALSE,
  which.quantile = 0.95,
  species = "Human",
  input.units = "uM",
  output.units = "mg",
  clearance.assay.conc = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_lit_oral_equiv_+3A_conc">conc</code></td>
<td>
<p>Bioactive in vitro concentration in units of specified
input.units, default of uM.</p>
</td></tr>
<tr><td><code id="get_lit_oral_equiv_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="get_lit_oral_equiv_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the CAS number or the chemical name must be
specified.</p>
</td></tr>
<tr><td><code id="get_lit_oral_equiv_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="get_lit_oral_equiv_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Suppress output messages.</p>
</td></tr>
<tr><td><code id="get_lit_oral_equiv_+3A_which.quantile">which.quantile</code></td>
<td>
<p>Which quantile from the SimCYP Monte Carlo simulation
is requested. Can be a vector.  Papers include 0.05, 0.5, and 0.95 for
humans and 0.5 for rats.</p>
</td></tr>
<tr><td><code id="get_lit_oral_equiv_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot; or default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="get_lit_oral_equiv_+3A_input.units">input.units</code></td>
<td>
<p>Units of given concentration, default of uM but can also
be mg/L.</p>
</td></tr>
<tr><td><code id="get_lit_oral_equiv_+3A_output.units">output.units</code></td>
<td>
<p>Units of dose, default of 'mg' for mg/kg BW/ day or
'mol' for mol/ kg BW/ day.</p>
</td></tr>
<tr><td><code id="get_lit_oral_equiv_+3A_clearance.assay.conc">clearance.assay.conc</code></td>
<td>
<p>Concentration of chemical used in measureing
intrinsic clearance data, 1 or 10 uM.</p>
</td></tr>
<tr><td><code id="get_lit_oral_equiv_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to get_lit_css.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Equivalent dose in specified units, default of mg/kg BW/day.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., &quot;Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment,&quot; Toxicological Sciences 125
157-174 (2012)
</p>
<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Li, L., Clewell, H.J. III,
Judson, R.S., Freeman, K., Bao, W, Sochaski, M.A., Chu T.-M., Black, M.B.,
Healy, E, Allen, B., Andersen M.E., Wolfinger, R.D., and Thomas R.S., &quot;The
Relative Impact of Incorporating Pharmacokinetics on Predicting in vivo
Hazard and Mode-of-Action from High-Throughput in vitro Toxicity Assays&quot;
Toxicological Sciences, 132:327-346 (2013).
</p>
<p>Wetmore, B. A., Wambaugh, J. F., Allen, B., Ferguson, S. S., Sochaski, M.
A., Setzer, R. W., Houck, K. A., Strope, C. L., Cantwell, K., Judson, R. S.,
LeCluyse, E., Clewell, H.J. III, Thomas, R.S., and Andersen, M. E. (2015).
&quot;Incorporating High-Throughput Exposure Predictions with Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing&quot; Toxicological
Sciences, kfv171.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
table &lt;- NULL
for(this.cas in sample(get_lit_cheminfo(),50)) table &lt;- rbind(table,cbind(
as.data.frame(this.cas),as.data.frame(get_lit_oral_equiv(conc=1,chem.cas=this.cas))))




get_lit_oral_equiv(0.1,chem.cas="34256-82-1")

get_lit_oral_equiv(0.1,chem.cas="34256-82-1",which.quantile=c(0.05,0.5,0.95))

</code></pre>

<hr>
<h2 id='get_physchem_param'>Get physico-chemical parameters from chem.physical_and_invitro.data table</h2><span id='topic+get_physchem_param'></span>

<h3>Description</h3>

<p>This function retrieves physico-chemical properties (&quot;param&quot;) for the 
chemical specified by chem.name or chem.cas from the table
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>. This 
function is distinguished from <code><a href="#topic+get_invitroPK_param">get_invitroPK_param</a></code> in that
there are no species-specific values. Physically meaningful values for 
ionization equilibria are NA/none (that is, no ionization), a single value, 
or a series of values separated by commas. If logMA (log10 membrane affinity) 
is NA, we use calc_ma() to predict it later on in the model parameterization 
functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_physchem_param(param, chem.name = NULL, chem.cas = NULL, dtxsid = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_physchem_param_+3A_param">param</code></td>
<td>
<p>The desired parameters, a vector or single value.</p>
</td></tr>
<tr><td><code id="get_physchem_param_+3A_chem.name">chem.name</code></td>
<td>
<p>The chemical names that you want parameters for, a vector or single value</p>
</td></tr>
<tr><td><code id="get_physchem_param_+3A_chem.cas">chem.cas</code></td>
<td>
<p>The chemical CAS numbers that you want parameters for, a vector or single value</p>
</td></tr>
<tr><td><code id="get_physchem_param_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this function works with a local version of the 
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> table to allow users to 
add/modify chemical
data (for example, adding new data via <code><a href="#topic+add_chemtable">add_chemtable</a></code> or 
loading in silico predictions distributed with httk via
<code><a href="#topic+load_sipes2017">load_sipes2017</a></code>, <code><a href="#topic+load_pradeep2020">load_pradeep2020</a></code>,
<code><a href="#topic+load_dawson2021">load_dawson2021</a></code>, or <code><a href="#topic+load_honda2023">load_honda2023</a></code>).
</p>
<p>User can request the following via argument param (case-insensitive):
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Parameter</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td><td style="text-align: left;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
MW  </td><td style="text-align: left;"> Molecular weight </td><td style="text-align: left;"> g/mole </td>
</tr>
<tr>
 <td style="text-align: left;"> 
pKa_Donor </td><td style="text-align: left;"> Hydrogen donor ionization equilibria (acidic pKa) </td><td style="text-align: left;"> pH </td>
</tr>
<tr>
 <td style="text-align: left;">
pKa_Accept </td><td style="text-align: left;"> Hyrdogen acceptor ionization equilibria (basic pKa </td><td style="text-align: left;"> pH </td>
</tr>
<tr>
 <td style="text-align: left;"> 
logMA </td><td style="text-align: left;"> log10 Membrane Affinity </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">
logP </td><td style="text-align: left;"> log10 Octanol:Water Partition Coefficient (hydrophobicity) </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">
logPwa </td><td style="text-align: left;"> log10 Water:Air Partition Coefficient </td><td style="text-align: left;"> unitless </td>
</tr>
<tr>
 <td style="text-align: left;">
logHenry </td><td style="text-align: left;"> log10 Henry's Law Constant </td><td style="text-align: left;"> atm-m3/mole </td>
</tr>
<tr>
 <td style="text-align: left;">
logWSol </td><td style="text-align: left;"> log10 Water Solubility </td><td style="text-align: left;"> moles/L: Water solubility at 25C </td>
</tr>
<tr>
 <td style="text-align: left;">
MP </td><td style="text-align: left;"> Melting point </td><td style="text-align: left;">  deg C </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>The parameters, either a single value, a named list for a single chemical, or a list of lists
</p>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>See Also</h3>

<p><code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>
</p>
<p><code><a href="#topic+get_invitroPK_param">get_invitroPK_param</a></code>
</p>
<p><code><a href="#topic+add_chemtable">add_chemtable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
get_physchem_param(param = 'logP', chem.cas = '80-05-7')
get_physchem_param(param = c('logP','MW'), chem.cas = c('80-05-7','81-81-2'))
# This function should be case-insensitive:
try(get_physchem_param(chem.cas="80-05-7","LogP"))
# Asking for a parameter we "don't" have produces an error:
try(get_physchem_param(chem.cas="80-05-7","MA"))
get_physchem_param(chem.cas="80-05-7","logMA")
# Ionization equilibria can be NA/none, a single value, or a series of values
# separated by commas:
get_physchem_param(chem.cas="80-05-7","pKa_Donor")
get_physchem_param(chem.cas="80-05-7","pKa_Accept")
get_physchem_param(chem.cas="71751-41-2","pKa_Donor")
get_physchem_param(chem.cas="71751-41-2","pKa_Accept")
# If logMA (log10 membrane affinity) is NA, we use calc_ma() to predict it
# in the parameterization functions:
get_physchem_param(chem.cas="71751-41-2","logMA")
parameterize_steadystate(chem.cas="71751-41-2")

</code></pre>

<hr>
<h2 id='get_rblood2plasma'>Get ratio of the blood concentration to the plasma concentration.</h2><span id='topic+get_rblood2plasma'></span>

<h3>Description</h3>

<p>This function attempts to retrieve a measured species- and chemical-specific 
blood:plasma concentration ratio.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_rblood2plasma(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_rblood2plasma_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="get_rblood2plasma_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the CAS number or the chemical name must be
specified.</p>
</td></tr>
<tr><td><code id="get_rblood2plasma_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="get_rblood2plasma_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="get_rblood2plasma_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A value of NA is returned when the requested value is unavailable.  Values
are retrieved from chem.physical_and_invitro.data. 
details than the description above ~~
</p>


<h3>Value</h3>

<p>A numeric value for the steady-state ratio of chemical concentration in blood
to plasma
</p>


<h3>Author(s)</h3>

<p>Robert Pearce
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
get_rblood2plasma(chem.name="Bisphenol A")
get_rblood2plasma(chem.name="Bisphenol A",species="Rat")

</code></pre>

<hr>
<h2 id='get_weight_class'>Assign weight class (underweight, normal, overweight, obese)</h2><span id='topic+get_weight_class'></span>

<h3>Description</h3>

<p>Given vectors of age, BMI, recumbent length, weight, and gender, categorizes
weight classes using CDC and WHO categories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_weight_class(age_years, age_months, bmi, recumlen, weight, gender)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_weight_class_+3A_age_years">age_years</code></td>
<td>
<p>A vector of ages in years.</p>
</td></tr>
<tr><td><code id="get_weight_class_+3A_age_months">age_months</code></td>
<td>
<p>A vector of ages in months.</p>
</td></tr>
<tr><td><code id="get_weight_class_+3A_bmi">bmi</code></td>
<td>
<p>A vector of BMIs.</p>
</td></tr>
<tr><td><code id="get_weight_class_+3A_recumlen">recumlen</code></td>
<td>
<p>A vector of heights or recumbent lengths in cm.</p>
</td></tr>
<tr><td><code id="get_weight_class_+3A_weight">weight</code></td>
<td>
<p>A vector of body weights in kg.</p>
</td></tr>
<tr><td><code id="get_weight_class_+3A_gender">gender</code></td>
<td>
<p>A vector of genders (as 'Male' or 'Female').</p>
</td></tr>
</table>


<h3>Details</h3>

<p>According to the CDC
(<a href="https://www.cdc.gov/obesity/basics/adult-defining.html">https://www.cdc.gov/obesity/basics/adult-defining.html</a>), adult weight
classes are defined using BMI as follows: </p>
 <dl>
<dt>Underweight</dt><dd><p>BMI
less than 18.5</p>
</dd> <dt>Normal</dt><dd><p>BMI between 18.5 and 25</p>
</dd> <dt>Overweight</dt><dd><p>BMI
between 25 and 30</p>
</dd> <dt>Obese</dt><dd><p>BMI greater than 30</p>
</dd> </dl>

<p>For children ages 2 years and older, weight classes are defined using
percentiles of sex-specific BMI for age, as follows (Barlow et al., 2007):
</p>
 <dl>
<dt>Underweight</dt><dd><p>Below 5th percentile BMI for age</p>
</dd>
<dt>Normal</dt><dd><p>5th-85th percentile BMI for age</p>
</dd> <dt>Overweight</dt><dd><p>85th-95th
percentile BMI for age</p>
</dd> <dt>Obese</dt><dd><p>Above 95th percentile BMI for age</p>
</dd> </dl>

<p>For children birth to age 2, weight classes are defined using percentiles of
sex-specific weight-for-length (Grummer-Strawn et al., 2009). Weight above the
97.7th percentile, or below the 2.3rd percentile, of weight-for-length is
considered potentially indicative of adverse health conditions. Here, weight
below the 2.3rd percentile is categorized as &quot;Underweight&quot; and weight above
the 97.7th percentile is categorized as &quot;Obese.&quot;
</p>


<h3>Value</h3>

<p>A character vector of weight classes. Each element will be one of
'Underweight', 'Normal', 'Overweight', or 'Obese'.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>
<p>Barlow SE. Expert committee recommendations regarding the prevention,
assessment, and treatment of child and adolescent overweight and obesity:
summary report. Pediatrics. 2007;120 Suppl 4. doi:10.1542/peds.2007-2329C
</p>
<p>Grummer-Strawn LM, Reinold C, Krebs NF. Use of World Health Organization and
CDC growth charts for children Aged 0-59 months in the United States. Morb
Mortal Wkly Rep. 2009;59(RR-9).
https://www.cdc.gov/mmwr/preview/mmwrhtml/rr5909a1.htm
</p>

<hr>
<h2 id='get_wetmore_cheminfo'>Get literature Chemical Information. (deprecated).</h2><span id='topic+get_wetmore_cheminfo'></span>

<h3>Description</h3>

<p>This function is included for backward compatibility. It calls
<code><a href="#topic+get_lit_cheminfo">get_lit_cheminfo</a></code> which
provides the information specified in &quot;info=&quot; for all
chemicals with data from the Wetmore et al. (2012) and (2013) publications
and other literature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_wetmore_cheminfo(
  info = "CAS",
  species = "Human",
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_wetmore_cheminfo_+3A_info">info</code></td>
<td>
<p>A single character vector (or collection of character vectors)
from
&quot;Compound&quot;, &quot;CAS&quot;, &quot;MW&quot;, &quot;Raw.Experimental.Percentage.Unbound&quot;,
&quot;Entered.Experimental.Percentage.Unbound&quot;, &quot;Fub&quot;, &quot;source_PPB&quot;,
&quot;Renal_Clearance&quot;, &quot;Met_Stab&quot;, &quot;Met_Stab_entered&quot;,
&quot;r2&quot;, &quot;p.val&quot;, &quot;Concentration..uM.&quot;, &quot;Css_lower_5th_perc.mg.L.&quot;, 
&quot;Css_median_perc.mg.L.&quot;, &quot;Css_upper_95th_perc.mg.L.&quot;,
&quot;Css_lower_5th_perc.uM.&quot;,&quot;Css_median_perc.uM.&quot;,&quot;Css_upper_95th_perc.uM.&quot;,
and &quot;Species&quot;.</p>
</td></tr>
<tr><td><code id="get_wetmore_cheminfo_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot; or default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="get_wetmore_cheminfo_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>info</code></td>
<td>
<p>Table/vector containing values specified in &quot;info&quot; for
valid chemicals.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., &quot;Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment,&quot; Toxicological Sciences 125
157-174 (2012)
</p>
<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Li, L., Clewell, H.J. III,
Judson, R.S., Freeman, K., Bao, W, Sochaski, M.A., Chu T.-M., Black, M.B.,
Healy, E, Allen, B., Andersen M.E., Wolfinger, R.D., and Thomas R.S., &quot;The
Relative Impact of Incorporating Pharmacokinetics on Predicting in vivo
Hazard and Mode-of-Action from High-Throughput in vitro Toxicity Assays&quot;
Toxicological Sciences, 132:327-346 (2013).
</p>
<p>Wetmore, B. A., Wambaugh, J. F., Allen, B., Ferguson, S. S., Sochaski, M.
A., Setzer, R. W., Houck, K. A., Strope, C. L., Cantwell, K., Judson, R. S.,
LeCluyse, E., Clewell, H.J. III, Thomas, R.S., and Andersen, M. E. (2015).
&quot;Incorporating High-Throughput Exposure Predictions with Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing&quot; Toxicological
Sciences, kfv171.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
get_lit_cheminfo()
get_lit_cheminfo(info=c('CAS','MW'))

</code></pre>

<hr>
<h2 id='get_wetmore_css'>Get literature Css (deprecated).</h2><span id='topic+get_wetmore_css'></span>

<h3>Description</h3>

<p>This function is included for backward compatibility. It calls
<code><a href="#topic+get_lit_css">get_lit_css</a></code> which
retrieves a steady-state plasma concentration as a result of
infusion dosing from the Wetmore et al. (2012) and (2013) publications and
other literature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_wetmore_css(
  chem.cas = NULL,
  chem.name = NULL,
  daily.dose = 1,
  which.quantile = 0.95,
  species = "Human",
  clearance.assay.conc = NULL,
  output.units = "mg/L",
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_wetmore_css_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the cas number or the chemical name must be
specified.</p>
</td></tr>
<tr><td><code id="get_wetmore_css_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="get_wetmore_css_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose infused in units of mg/kg BW/day.
Defaults to 1 mg/kg/day.</p>
</td></tr>
<tr><td><code id="get_wetmore_css_+3A_which.quantile">which.quantile</code></td>
<td>
<p>Which quantile from the SimCYP Monte Carlo simulation
is requested. Can be a vector.</p>
</td></tr>
<tr><td><code id="get_wetmore_css_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot; or default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="get_wetmore_css_+3A_clearance.assay.conc">clearance.assay.conc</code></td>
<td>
<p>Concentration of chemical used in measureing
intrinsic clearance data, 1 or 10 uM.</p>
</td></tr>
<tr><td><code id="get_wetmore_css_+3A_output.units">output.units</code></td>
<td>
<p>Returned units for function, defaults to mg/L but can
also be uM (specify units = &quot;uM&quot;).</p>
</td></tr>
<tr><td><code id="get_wetmore_css_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector with the literature steady-state plasma 
concentration (1 mg/kg/day) for the requested quantiles
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., &quot;Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment,&quot; Toxicological Sciences 125
157-174 (2012)
</p>
<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Li, L., Clewell, H.J. III,
Judson, R.S., Freeman, K., Bao, W, Sochaski, M.A., Chu T.-M., Black, M.B.,
Healy, E, Allen, B., Andersen M.E., Wolfinger, R.D., and Thomas R.S., &quot;The
Relative Impact of Incorporating Pharmacokinetics on Predicting in vivo
Hazard and Mode-of-Action from High-Throughput in vitro Toxicity Assays&quot;
Toxicological Sciences, 132:327-346 (2013).
</p>
<p>Wetmore, B. A., Wambaugh, J. F., Allen, B., Ferguson, S. S., Sochaski, M.
A., Setzer, R. W., Houck, K. A., Strope, C. L., Cantwell, K., Judson, R. S.,
LeCluyse, E., Clewell, H.J. III, Thomas, R.S., and Andersen, M. E. (2015).
&quot;Incorporating High-Throughput Exposure Predictions with Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing&quot; Toxicological
Sciences, kfv171.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_lit_css(chem.cas="34256-82-1")

get_lit_css(chem.cas="34256-82-1",species="Rat",which.quantile=0.5)

get_lit_css(chem.cas="80-05-7", daily.dose = 1,which.quantile = 0.5, output.units = "uM")
</code></pre>

<hr>
<h2 id='get_wetmore_oral_equiv'>Get Literature Oral Equivalent Dose (deprecated).</h2><span id='topic+get_wetmore_oral_equiv'></span>

<h3>Description</h3>

<p>This function is included for backward compatibility. It calls
<code><a href="#topic+get_lit_oral_equiv">get_lit_oral_equiv</a></code> which
converts a chemical plasma concetration to an oral equivalent
dose using the values from the Wetmore et al. (2012) and (2013) publications
and other literature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_wetmore_oral_equiv(
  conc,
  chem.name = NULL,
  chem.cas = NULL,
  suppress.messages = FALSE,
  which.quantile = 0.95,
  species = "Human",
  input.units = "uM",
  output.units = "mg",
  clearance.assay.conc = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_wetmore_oral_equiv_+3A_conc">conc</code></td>
<td>
<p>Bioactive in vitro concentration in units of specified
input.units, default of uM.</p>
</td></tr>
<tr><td><code id="get_wetmore_oral_equiv_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="get_wetmore_oral_equiv_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the CAS number or the chemical name must be
specified.</p>
</td></tr>
<tr><td><code id="get_wetmore_oral_equiv_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Suppress output messages.</p>
</td></tr>
<tr><td><code id="get_wetmore_oral_equiv_+3A_which.quantile">which.quantile</code></td>
<td>
<p>Which quantile from the SimCYP Monte Carlo simulation
is requested. Can be a vector.  Papers include 0.05, 0.5, and 0.95 for
humans and 0.5 for rats.</p>
</td></tr>
<tr><td><code id="get_wetmore_oral_equiv_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot; or default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="get_wetmore_oral_equiv_+3A_input.units">input.units</code></td>
<td>
<p>Units of given concentration, default of uM but can also
be mg/L.</p>
</td></tr>
<tr><td><code id="get_wetmore_oral_equiv_+3A_output.units">output.units</code></td>
<td>
<p>Units of dose, default of 'mg' for mg/kg BW/ day or
'mol' for mol/ kg BW/ day.</p>
</td></tr>
<tr><td><code id="get_wetmore_oral_equiv_+3A_clearance.assay.conc">clearance.assay.conc</code></td>
<td>
<p>Concentration of chemical used in measureing
intrinsic clearance data, 1 or 10 uM.</p>
</td></tr>
<tr><td><code id="get_wetmore_oral_equiv_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to get_lit_css.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Equivalent dose in specified units, default of mg/kg BW/day.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., &quot;Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment,&quot; Toxicological Sciences 125
157-174 (2012)
</p>
<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Li, L., Clewell, H.J. III,
Judson, R.S., Freeman, K., Bao, W, Sochaski, M.A., Chu T.-M., Black, M.B.,
Healy, E, Allen, B., Andersen M.E., Wolfinger, R.D., and Thomas R.S., &quot;The
Relative Impact of Incorporating Pharmacokinetics on Predicting in vivo
Hazard and Mode-of-Action from High-Throughput in vitro Toxicity Assays&quot;
Toxicological Sciences, 132:327-346 (2013).
</p>
<p>Wetmore, B. A., Wambaugh, J. F., Allen, B., Ferguson, S. S., Sochaski, M.
A., Setzer, R. W., Houck, K. A., Strope, C. L., Cantwell, K., Judson, R. S.,
LeCluyse, E., Clewell, H.J. III, Thomas, R.S., and Andersen, M. E. (2015).
&quot;Incorporating High-Throughput Exposure Predictions with Dosimetry-Adjusted
In Vitro Bioactivity to Inform Chemical Toxicity Testing&quot; Toxicological
Sciences, kfv171.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
table &lt;- NULL
for(this.cas in sample(get_lit_cheminfo(),50)) table &lt;- rbind(table,cbind(
as.data.frame(this.cas),as.data.frame(get_lit_oral_equiv(conc=1,chem.cas=this.cas))))




get_lit_oral_equiv(0.1,chem.cas="34256-82-1")

get_lit_oral_equiv(0.1,chem.cas="34256-82-1",which.quantile=c(0.05,0.5,0.95))

</code></pre>

<hr>
<h2 id='hct_h'>KDE bandwidths for residual variability in hematocrit</h2><span id='topic+hct_h'></span>

<h3>Description</h3>

<p>Bandwidths used for a one-dimensional kernel density estimation of the
distribution of residual errors around smoothing spline fits of hematocrit vs.
age  for NHANES respondents in each of ten combinations of sex and
race/ethnicity categories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hct_h
</code></pre>


<h3>Format</h3>

<p>A named list with 10 elements, each a numeric value. Each list element
corresponds to, and is named for, one combination of NHANES sex categories
(Male and Female) and NHANES race/ethnicity categories (Mexican American,
Other Hispanic, Non-Hispanic White, Non-Hispanic Black, and Other).
</p>


<h3>Details</h3>

<p>Each matrix is the standard deviation for a normal distribution: this is the
bandwidth to be used for a kernel density estimation (KDE) (using a normal
kernel) of the distribution of residual errors around smoothing spline fits of
hematocrit vs. age for NHANES respondents in the specified sex and
race/ethnicity category. Optimal bandwidths were pre-calculated by doing the
smoothing spline fits, getting the residuals, then calling
<code><a href="ks.html#topic+kde">kde</a></code> on the residuals (which calls <code><a href="ks.html#topic+hpi">hpi</a></code> to
compute the plug-in bandwidth).
</p>
<p>Used by HTTK-Pop only in &quot;virtual individuals&quot; mode (i.e.
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code> with <code>method = "v"</code>), in
<code><a href="#topic+estimate_hematocrit">estimate_hematocrit</a></code>.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='hematocrit_infants'>Predict hematocrit in infants under 1 year old.</h2><span id='topic+hematocrit_infants'></span>

<h3>Description</h3>

<p>For infants under 1 year, hematocrit was not measured in NHANES. Assume a
log-normal distribution where plus/minus 1 standard deviation of the
underlying normal distribution is given by the reference range. Draw
hematocrit values from these distributions by age.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hematocrit_infants(age_months)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hematocrit_infants_+3A_age_months">age_months</code></td>
<td>
<p>Vector of ages in months; all must be &lt;= 12.</p>
</td></tr>
</table>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: center;"> 
Age </td><td style="text-align: center;"> Reference range</td>
</tr>
<tr>
 <td style="text-align: center;"> 
&lt;1 month </td><td style="text-align: center;"> 31-49</td>
</tr>
<tr>
 <td style="text-align: center;"> 
1-6 months </td><td style="text-align: center;"> 29-42</td>
</tr>
<tr>
 <td style="text-align: center;"> 
7-12 months </td><td style="text-align: center;"> 33-38 
</td>
</tr>

</table>



<h3>Value</h3>

<p>Vector of hematocrit percentages corresponding to the input vector
of ages.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='honda.ivive'>Return the assumptions used in Honda et al. 2019</h2><span id='topic+honda.ivive'></span>

<h3>Description</h3>

<p>This function returns four of the better performing sets of assumptions evaluated in Honda et al. 2019 
(https://doi.org/10.1371/journal.pone.0217564).These include four different combinations of hepatic clearance assumption, in vivo bioactivity assumption, 
and relevant tissue assumption. Generally, this function is not called directly by the user, but instead
called by setting the IVIVE option in calc_mc_oral_equiv, calc_mc_css, and calc_analytic functions. Currently, these IVIVE option 
is not implemented the solve_1comp etc. functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>honda.ivive(method = "Honda1", tissue = "liver")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="honda.ivive_+3A_method">method</code></td>
<td>
<p>This is set to one of &quot;Honda1&quot;, &quot;Honda2&quot;, &quot;Honda3&quot;, or &quot;Honda4&quot;.</p>
</td></tr>
<tr><td><code id="honda.ivive_+3A_tissue">tissue</code></td>
<td>
<p>This is only relevant to &quot;Honda4&quot; and indicates the relevant tissue compartment.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Only four sets of IVIVE assumptions that performed well in Honda et al. 
(2019) are currently included:
&quot;Honda1&quot; through &quot;Honda4&quot;. The use of max (peak)  
concentration can not be currently be calculated with <code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code>. 
The httk default settings correspond to &quot;Honda3&quot;:
</p>

<table>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: right;"> <em>In Vivo</em> Conc. </td><td style="text-align: right;"> Metabolic Clearance </td><td style="text-align: right;"> Bioactive Chemical Conc. <em>In Vivo</em> </td><td style="text-align: right;"> TK Statistic Used* </td><td style="text-align: right;"> Bioactive Chemical Conc. <em>In Vitro</em> </td>
</tr>
<tr>
 <td style="text-align: left;">
Honda1 </td><td style="text-align: right;"> Veinous (Plasma) </td><td style="text-align: right;"> Restrictive </td><td style="text-align: right;"> Free </td><td style="text-align: right;"> Mean Conc. In Vivo </td><td style="text-align: right;"> Free Conc. In Vitro </td>
</tr>
<tr>
 <td style="text-align: left;">
Honda2 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;"> Restrictive </td><td style="text-align: right;"> Free </td><td style="text-align: right;"> Mean Conc. In Vivo </td><td style="text-align: right;"> Nominal Conc. In Vitro </td>
</tr>
<tr>
 <td style="text-align: left;">
Honda3 </td><td style="text-align: right;"> Veinous </td><td style="text-align: right;"> Restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Mean Conc. In Vivo </td><td style="text-align: right;"> Nominal Conc. In Vitro </td>
</tr>
<tr>
 <td style="text-align: left;">
Honda4 </td><td style="text-align: right;"> Target Tissue </td><td style="text-align: right;"> Non-restrictive </td><td style="text-align: right;"> Total </td><td style="text-align: right;"> Mean Conc. In Vivo </td><td style="text-align: right;"> Nominal Conc. In Vitro </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>&quot;Honda1&quot; uses plasma concentration, restrictive clearance, and treats the 
unbound invivo concentration as bioactive. For IVIVE, any input nominal 
concentration in vitro should be converted to cfree.invitro using 
<code><a href="#topic+armitage_eval">armitage_eval</a></code>, otherwise performance will be the same as 
&quot;Honda2&quot;.
</p>


<h3>Value</h3>

<p>A list of tissue, bioactive.free.invivo, and restrictive.clearance assumptions.
</p>


<h3>Author(s)</h3>

<p>Greg Honda and John Wambaugh
</p>


<h3>References</h3>

<p>Honda GS, Pearce RG, Pham LL, Setzer RW, Wetmore BA, Sipes NS, Gilbert J, Franz B, Thomas RS, Wambaugh JF (2019).
&ldquo;Using the concordance of in vitro and in vivo data to evaluate extrapolation assumptions.&rdquo;
<em>PloS one</em>, <b>14</b>(5), e0217564.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>honda.ivive(method = "Honda1", tissue = NULL)

</code></pre>

<hr>
<h2 id='honda2023.data'>Measured Caco-2 Apical-Basal Permeability Data</h2><span id='topic+honda2023.data'></span>

<h3>Description</h3>

<p>In vitro Caco-2 membrane permeabilities characterize how readily absobed/transported
a chemical is. These measurements are all for the apical-to-basal Caco-2
orientation.
These data were either measured by EPA or collected by other others, as
indicated by the column 'Data Origin'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>honda2023.data
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 634 rows and 5 columns.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
<strong>Column Name</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td><td style="text-align: left;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
DTXSID </td><td style="text-align: left;"> EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>) </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
Pab </td><td style="text-align: left;"> Apical-to-basal Caco-2 permeability </td><td style="text-align: left;"> 10^-6 cm/s </td>
</tr>
<tr>
 <td style="text-align: left;">
Data Origin </td><td style="text-align: left;"> The reference which collected/generated the measurement </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
Test </td><td style="text-align: left;"> Whether (1) or not (0) the data was withheld from model building to be used in the QSPR test set </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
CAS </td><td style="text-align: left;"> Chemical Abstracts Service Registry Number </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>References</h3>

<p>Obringer C, Manwaring J, Goebel C, Hewitt NJ, Rothe H (2016).
&ldquo;Suitability of the in vitro Caco-2 assay to predict the oral absorption of aromatic amine hair dyes.&rdquo;
<em>Toxicology in Vitro</em>, <b>32</b>, 1&ndash;7.
</p>
<p>Lanevskij K, Didziapetris R (2019).
&ldquo;Physicochemical QSAR analysis of passive permeability across Caco-2 monolayers.&rdquo;
<em>Journal of Pharmaceutical Sciences</em>, <b>108</b>(1), 78&ndash;86.
</p>
<p>Gaulton A, Bellis LJ, Bento AP, Chambers J, Davies M, Hersey A, Light Y, McGlinchey S, Michalovich D, Al-Lazikani B, others (2012).
&ldquo;ChEMBL: a large-scale bioactivity database for drug discovery.&rdquo;
<em>Nucleic Acids Research</em>, <b>40</b>(D1), D1100&ndash;D1107.
</p>
<p>Honda G, Kenyon EM, Davidson-Fritz SE, Dinallo R, El-Masri H, Korel-Bexell E, Li L, Paul-Friedman K, Pearce R, Sayre R, Strock C, Thomas R, Wetmore BA, Wambaugh JF (2023).
&ldquo;Impact of Gut Permeability on Estimation of Oral Bioavailability for Chemicals in Commerce and the Environment.&rdquo;
Unpublished.
</p>

<hr>
<h2 id='honda2023.qspr'>Predicted Caco-2 Apical-Basal Permeabilities</h2><span id='topic+honda2023.qspr'></span>

<h3>Description</h3>

<p>Honda et al. (2023) describes the construction of a machine-learning 
quantitative structure-property relationship (QSPR )model for in vitro Caco-2 
membrane permeabilites. That model was used to make chemical-specific 
predictions provided in this table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>honda2023.qspr
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 14033 rows and 5 columns.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
<strong>Column Name</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td><td style="text-align: left;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
DTXSID </td><td style="text-align: left;"> EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>) </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
Pab.Class.Pred </td><td style="text-align: left;"> Predicted Pab rate of slow (1), moderate (2), or fast (3) </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
Pab.Pred.AD </td><td style="text-align: left;"> Whether (1) or not (0) the chemical is anticipated to be withing the QSPR domain of applicability </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
CAS </td><td style="text-align: left;"> Chemical Abstracts Service Registry Number </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
Pab.Quant.Pred </td><td style="text-align: left;"> Median and 95-percent interval for values within the predicted class's training data  moderate (2), or fast (3) </td><td style="text-align: left;"> 10^-6 cm/s </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>References</h3>

<p>Honda G, Kenyon EM, Davidson-Fritz SE, Dinallo R, El-Masri H, Korel-Bexell E, Li L, Paul-Friedman K, Pearce R, Sayre R, Strock C, Thomas R, Wetmore BA, Wambaugh JF (2023).
&ldquo;Impact of Gut Permeability on Estimation of Oral Bioavailability for Chemicals in Commerce and the Environment.&rdquo;
Unpublished.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_honda2023">load_honda2023</a></code>
</p>

<hr>
<h2 id='howgate'>Howgate 2006</h2><span id='topic+howgate'></span>

<h3>Description</h3>

<p>This data set is only used in Vignette 5.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>howgate
</code></pre>


<h3>Format</h3>

<p>A data.table containing 24 rows and 11 columns.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Howgate, E. M., et al. &quot;Prediction of in vivo drug clearance from in vitro
data. I: impact of inter-individual variability.&quot; Xenobiotica 36.6 (2006):
473-497.
</p>

<hr>
<h2 id='httk-package'>httk: High-Throughput Toxicokinetics</h2><span id='topic+httk'></span><span id='topic+httk-package'></span>

<h3>Description</h3>

<p>Pre-made models that can be rapidly tailored to various chemicals and species using chemical-specific in vitro data and physiological information. These tools allow incorporation of chemical toxicokinetics (&quot;TK&quot;) and in vitro-in vivo extrapolation (&quot;IVIVE&quot;) into bioinformatics, as described by Pearce et al. (2017) (<a href="https://doi.org/10.18637/jss.v079.i04">doi:10.18637/jss.v079.i04</a>). Chemical-specific in vitro data characterizing toxicokinetics have been obtained from relatively high-throughput experiments. The chemical-independent (&quot;generic&quot;) physiologically-based (&quot;PBTK&quot;) and empirical (for example, one compartment) &quot;TK&quot; models included here can be parameterized with in vitro data or in silico predictions which are provided for thousands of chemicals, multiple exposure routes, and various species. High throughput toxicokinetics (&quot;HTTK&quot;) is the combination of in vitro data and generic models. We establish the expected accuracy of HTTK for chemicals without in vivo data through statistical evaluation of HTTK predictions for chemicals where in vivo data do exist. The models are systems of ordinary differential equations that are developed in MCSim and solved using compiled (C-based) code for speed. A Monte Carlo sampler is included for simulating human biological variability (Ring et al., 2017 <a href="https://doi.org/10.1016/j.envint.2017.06.004">doi:10.1016/j.envint.2017.06.004</a>) and propagating parameter uncertainty (Wambaugh et al., 2019 <a href="https://doi.org/10.1093/toxsci/kfz205">doi:10.1093/toxsci/kfz205</a>). Empirically calibrated methods are included for predicting tissue:plasma partition coefficients and volume of distribution (Pearce et al., 2017 <a href="https://doi.org/10.1007/s10928-017-9548-7">doi:10.1007/s10928-017-9548-7</a>). These functions and data provide a set of tools for using IVIVE to convert concentrations from high-throughput screening experiments (for example, Tox21, ToxCast) to real-world exposures via reverse dosimetry (also known as &quot;RTK&quot;) (Wetmore et al., 2015 <a href="https://doi.org/10.1093/toxsci/kfv171">doi:10.1093/toxsci/kfv171</a>).
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: John Wambaugh <a href="mailto:wambaugh.john@epa.gov">wambaugh.john@epa.gov</a> (<a href="https://orcid.org/0000-0002-4024-534X">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Sarah Davidson-Fritz <a href="mailto:davidsonfritz.sarah@epa.gov">davidsonfritz.sarah@epa.gov</a> (<a href="https://orcid.org/0000-0002-2891-9380">ORCID</a>)
</p>
</li>
<li><p> Robert Pearce (<a href="https://orcid.org/0000-0003-3168-4049">ORCID</a>)
</p>
</li>
<li><p> Caroline Ring <a href="mailto:Ring.Caroline@epa.gov">Ring.Caroline@epa.gov</a> (<a href="https://orcid.org/0000-0002-0463-1251">ORCID</a>)
</p>
</li>
<li><p> Greg Honda <a href="mailto:honda.gregory@epa.gov">honda.gregory@epa.gov</a> (<a href="https://orcid.org/0000-0001-7713-9850">ORCID</a>)
</p>
</li>
<li><p> Mark Sfeir
</p>
</li>
<li><p> Matt Linakis <a href="mailto:MLINAKIS@ramboll.com">MLINAKIS@ramboll.com</a> (<a href="https://orcid.org/0000-0003-0526-2395">ORCID</a>)
</p>
</li>
<li><p> Dustin Kapraun <a href="mailto:kapraun.dustin@epa.gov">kapraun.dustin@epa.gov</a> (<a href="https://orcid.org/0000-0001-5570-6383">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Nathan Pollesch <a href="mailto:pollesch.nathan@epa.gov">pollesch.nathan@epa.gov</a> (<a href="https://orcid.org/0000-0001-8992-9294">ORCID</a>) [contributor]
</p>
</li>
<li><p> Miyuki Breen <a href="mailto:breen.miyuki@epa.gov">breen.miyuki@epa.gov</a> (<a href="https://orcid.org/0000-0001-8511-4653">ORCID</a>) [contributor]
</p>
</li>
<li><p> Shannon Bell <a href="mailto:Shannon.bell@inotivco.com">Shannon.bell@inotivco.com</a> (<a href="https://orcid.org/0000-0002-5754-6085">ORCID</a>) [contributor]
</p>
</li>
<li><p> Xiaoqing Chang <a href="mailto:Xiaoqing.chang@inotivco.com">Xiaoqing.chang@inotivco.com</a> (<a href="https://orcid.org/0000-0003-0752-1848">ORCID</a>) [contributor]
</p>
</li>
<li><p> Todor Antonijevic <a href="mailto:tantonijevic@toxstrategies.com">tantonijevic@toxstrategies.com</a> (<a href="https://orcid.org/0000-0002-0248-8412">ORCID</a>) [contributor]
</p>
</li>
<li><p> Jimena Davis [contributor]
</p>
</li>
<li><p> Elaina Kenyon <a href="mailto:kenyon.elaina@epa.gov">kenyon.elaina@epa.gov</a> (<a href="https://orcid.org/0000-0001-8781-2391">ORCID</a>) [contributor]
</p>
</li>
<li><p> Katie Paul Friedman <a href="mailto:Paul-Friedman.Katie@epa.gov">Paul-Friedman.Katie@epa.gov</a> (<a href="https://orcid.org/0000-0002-2710-1691">ORCID</a>) [contributor]
</p>
</li>
<li><p> James Sluka <a href="mailto:jsluka@indiana.edu">jsluka@indiana.edu</a> (<a href="https://orcid.org/0000-0002-5901-1404">ORCID</a>) [contributor]
</p>
</li>
<li><p> Noelle Sinski <a href="mailto:Noelle.Sinski@icf.com">Noelle.Sinski@icf.com</a> [contributor]
</p>
</li>
<li><p> Nisha Sipes <a href="mailto:sipes.nisha@epa.gov">sipes.nisha@epa.gov</a> (<a href="https://orcid.org/0000-0003-4203-6426">ORCID</a>) [contributor]
</p>
</li>
<li><p> Barbara Wetmore <a href="mailto:wetmore.barbara@epa.gov">wetmore.barbara@epa.gov</a> (<a href="https://orcid.org/0000-0002-6878-5348">ORCID</a>) [contributor]
</p>
</li>
<li><p> Lily Whipple [contributor]
</p>
</li>
<li><p> Woodrow Setzer (<a href="https://orcid.org/0000-0002-6709-9186">ORCID</a>) [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://www.epa.gov/chemical-research/rapid-chemical-exposure-and-dose-research">https://www.epa.gov/chemical-research/rapid-chemical-exposure-and-dose-research</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/USEPA/CompTox-ExpoCast-httk">https://github.com/USEPA/CompTox-ExpoCast-httk</a>
</p>
</li></ul>


<hr>
<h2 id='httk.performance'>Historical Performance of R Package httk</h2><span id='topic+httk.performance'></span>

<h3>Description</h3>

<p>This table records the historical performance and other metrics 
of the R package &quot;httk&quot; as
profiled with the function <code><a href="#topic+benchmark_httk">benchmark_httk</a></code>. There is a row for
each version and a column for each benchmark or metric. This table is used
to generate graphs comparing the current version to past performance in order
to help identify unintended degradtion of package capabilities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>httk.performance
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 23 rows and 18 columns.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
<strong>Column Name</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
 Version </td><td style="text-align: left;"> The release of httk (major.minor.patch) </td>
</tr>
<tr>
 <td style="text-align: left;">                      
 N.steadystate </td><td style="text-align: left;"> The number of chemicals for which Css can be predicted for the steady-state model </td>
</tr>
<tr>
 <td style="text-align: left;">
 calc_analytic.units </td><td style="text-align: left;"> The ratio of the output of <code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code> in mg/L to uM multiplied by 1000/MW (should be 1)</td>
</tr>
<tr>
 <td style="text-align: left;">
 calc_mc.units  </td><td style="text-align: left;"> The ratio of the output of <code><a href="#topic+calc_mc_css">calc_mc_css</a></code> in mg/L to uM multiplied by 1000/MW (should be 1) </td>
</tr>
<tr>
 <td style="text-align: left;">
 solve_pbtk.units  </td><td style="text-align: left;"> The ratio of a Cplasma value from <code><a href="#topic+solve_pbtk">solve_pbtk</a></code> in mg/L to uM multiplied by 1000/MW (should be 1) </td>
</tr>
<tr>
 <td style="text-align: left;">  
 RMSLE.Wetmore </td><td style="text-align: left;"> Root mean squared log10 error between Css predictions from httk and published values from Wetmore papers (both 95th percentile) </td>
</tr>
<tr>
 <td style="text-align: left;">
 N.Wetmore  </td><td style="text-align: left;">  Number of chemicals used in RMSLE evaluation </td>
</tr>
<tr>
 <td style="text-align: left;">          
 RMSLE.noMC  </td><td style="text-align: left;">  RMSLE between 95th percentile Css prediction and median prediction </td>
</tr>
<tr>
 <td style="text-align: left;"> 
 N.noMC  </td><td style="text-align: left;"> Number of chemicals used in RMSLE evaluation </td>
</tr>
<tr>
 <td style="text-align: left;"> 
 RMSLE.InVivoCss  </td><td style="text-align: left;"> RMSLE for predictions of in vivo measured Css </td>
</tr>
<tr>
 <td style="text-align: left;"> 
 N.InVivoCss  </td><td style="text-align: left;"> Number of chemicals used in RMSLE evaluation  </td>
</tr>
<tr>
 <td style="text-align: left;"> 
 RMSLE.InVivoAUC   </td><td style="text-align: left;"> RMSLE for predictions of in vivo measured AUCs </td>
</tr>
<tr>
 <td style="text-align: left;"> 
 N.InVivoAUC  </td><td style="text-align: left;"> Number of chemicals used in RMSLE evaluation </td>
</tr>
<tr>
 <td style="text-align: left;"> 
 RMSLE.InVivoCmax </td><td style="text-align: left;"> RMSLE for predictions of in vivo measured Cmax </td>
</tr>
<tr>
 <td style="text-align: left;"> 
 N.InVivoCmax  </td><td style="text-align: left;"> Number of chemicals used in RMSLE evaluation </td>
</tr>
<tr>
 <td style="text-align: left;"> 
 RMSLE.TissuePC </td><td style="text-align: left;"> RMSLE for predicted tissue:plasma partition coefficients </td>
</tr>
<tr>
 <td style="text-align: left;">  
 N.TissuePC  </td><td style="text-align: left;"> Number of chemicals used in RMSLE evaluation </td>
</tr>
<tr>
 <td style="text-align: left;">                
 Notes  </td><td style="text-align: left;"> Why benchmarks/metrics may have changed </td>
</tr>
<tr>
 <td style="text-align: left;">       
</td>
</tr>

</table>



<h3>References</h3>

<p>Davidson-Fritz SE, Evans MV, Chang X, Breen M, Honda GS, Kenyon E, Linakis MW, Meade A, Pearce RG, Purucker T, Ring CL, Sfeir MA, Setzer RW, Sluka JP, Vitense K, Devito MJ, Wambaugh JF (2023).
&ldquo;Transparent and Evaluated Toxicokinetic Models for Bioinformatics and Public Health Risk Assessment.&rdquo;
Unpublished.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+benchmark_httk">benchmark_httk</a></code>
</p>

<hr>
<h2 id='httkpop'>httkpop: Virtual population generator for HTTK.</h2><span id='topic+httkpop'></span>

<h3>Description</h3>

<p>The httkpop package generates virtual population physiologies for use in 
population TK.
</p>


<h3>Details</h3>

<p>To simulate inter-individual variability in the TK model, a MC approach
is used: the model parameters are sampled from known or assumed
distributions, and the model is evaluated for each sampled set of
parameters. To simulate variability across subpopulations, the MC approach
needs to capture the parameter correlation structure. For example,
kidney function changes with age (Levey et al., 2009), thus the
distribution of GFR is likely different in 6-year-olds than in 65-yearolds.
To directly measure the parameter correlation structure, all parameters
need to be measured in each individual in a representative
sample population. Such direct measurements are extremely limited.
However, the correlation structure of the physiological parameters can
be inferred from their known individual correlations with demographic
and anthropometric quantities for which direct population measurements
do exist. These quantities are sex, race/ethnicity, age, height, and
weight (Howgate et al., 2006; Jamei et al., 2009a; Johnson et al., 2006;
McNally et al., 2014; Price et al., 2003). Direct measurements of these
quantities in a large, representative sample of the U.S. population are
publicly available from NHANES. NHANES also includes laboratory
measurements, including both serum creatinine, which can be used to
estimate GFR (Levey et al., 2009), and hematocrit. For conciseness, sex,
race/ethnicity, age, height, weight, serum creatinine, and hematocrit
will be called the NHANES quantities.
</p>
<p>HTTK-Pop's correlated MC approach begins by sampling from the
joint distribution of the NHANES quantities to simulate a population.
Then, for each individual in the simulated population, HTTKePop
predicts the physiological parameters from the NHANES
quantities using regression equations from the literature (Barter et al.,
2007; Baxter-Jones et al., 2011; Bosgra et al., 2012; Koo et al., 2000;
Levey et al., 2009; Looker et al., 2013; McNally et al., 2014; Ogiu et al.,
1997; Price et al., 2003; Schwartz and Work, 2009; Webber and Barr 2012). 
Correlations among the physiological parameters are induced by
their mutual dependence on the correlated NHANES quantities. Finally,
residual variability is added to the predicted physiological parameters
using estimates of residual marginal variance (i.e., variance not explained
by the regressions on the NHANES quantities) (McNally et al.,
2014).
</p>
<p>Data were combined from the three most recent publicly-available
NHANES cycles: 2007-2008, 2009-2010, and 2011-2012. For each
cycle, some NHANES quantities - height, weight, serum creatinine,
and hematocrit - were measured only in a subset of respondents. Only
these subsets were included in HTTKePop. The pooled subsets from the
three cycles contained 29,353 unique respondents. Some respondents
were excluded from analysis: those with age recorded as 80 years (because
all NHANES respondents 80 years and older were marked as
&quot;80&quot;); those with missing height, weight or hematocrit data; and those
aged 12 years or older with missing serum creatinine data. These criteria
excluded 4807 respondents, leaving 24,546 unique respondents. Each
NHANES respondent was assigned a cycle-specific sample weight,
which can be interpreted as the number of individuals in the total U.S.
population represented by each NHANES respondent in each cycle
(Johnson et al., 2013). Because data from three cycles were combined,
the sample weights were rescaled (divided by the number of cycles
being combined, as recommended in NHANES data analysis documentation)
(Johnson et al., 2013). To handle the complex NHANES
sampling structure, the R survey package was used to analyze the
NHANES data (Lumley, 2004). 
</p>
<p>To allow generation of virtual populations specified by weight class,
we coded a categorical variable for each NHANES respondent. The
categories Underweight, Normal, Overweight, or Obese were assigned
based on weight, age, and height/length (Grummer-Strawn et al., 2010;
Kuczmarski et al., 2002; Ogden et al., 2014; WHO, 2006, 2010). 
We implemented two population simulation methods within HTTK-Pop:
the direct-resampling method and the virtual-individuals method.
The direct-resampling method simulated a population by sampling
NHANES respondents with replacement, with probabilities proportional
to the sample weights. Each individual in the resulting simulated population
was an NHANES respondent, identified by a unique NHANES
sequence number. By contrast, the second method generates &quot;virtual
individuals&quot; - sets of NHANES quantities that obey the approximate
joint distribution of the NHANES quantities (calculated using weighted
smoothing functions and kernel density estimators), but do not necessarily correspond to
any particular NHANES respondent. The direct-resampling method removed
the possibility of generating unrealistic combinations of the
NHANES quantities; the virtual-individuals method allowed the use of
interpolation to simulate subpopulations represented by only a small
number of NHANES respondents. 
</p>
<p>For either method, HTTK-Pop
takes optional specifications about the population to be simulated
and then samples from the appropriate conditional joint
distribution of the NHANES quantities.
</p>
<p>Once HTTK-Pop has simulated a population characterized by the
NHANES quantities, the physiological parameters of the TK model
are predicted from the NHANES quantities using regression
equations from the literature. Liver mass was predicted for individuals
over age 18 using allometric scaling with height from Reference Man
(Valentin, 2002), and for individuals under 18 using regression relationships
with height and weight published by Ogiu et al. (1997).
Residual marginal variability was added for each individual as in
PopGen (McNally et al., 2014). Similarly, hepatic portal vein blood
flows (in L/h) are predicted as fixed fractions of a cardiac output allometrically
scaled with height from Reference Man (Valentin, 2002),
and residual marginal variability is added for each individual (McNally
et al., 2014). Glomerular filtration rate (GFR) (in L/h/1.73 m2 body
surface area) is predicted from age, race, sex, and serum creatinine
using the CKD-EPI equation, for individuals over age 18 (Levey et al.,
2009). For individuals under age 18, GFR is estimated from body surface
area (BSA) (Johnson et al., 2006); BSA is predicted using Mosteller's
formula (Verbraecken et al., 2006) for adults and Haycock's
formula (Haycock et al., 1978) for children. Hepatocellularity (in millions
of cells per gram of liver tissue) is predicted from age using an
equation developed by Barter et al. (2007). Hematocrit is estimated
from NHANES data for individuals 1 year and older. For individuals
younger than 1 year, for whom NHANES did not measure hematocrit
directly, hematocrit was predicted from age in months, using published
reference ranges (Lubin, 1987).
</p>
<p>In addition to the HTTK physiological parameters, the HTTK models
include chemical-specific parameters representing the fraction of chemical
unbound in plasma (Fup) and intrinsic clearance (CLint). Because
these parameters represent interactions of the chemical with the body,
their values will vary between individuals. To simulate this variability,
Fub and CLint were included in MC simulations, by sampling from estimated
or assumed distributions for the parameters defining them.
</p>
<p>Variability in hematocrit was simulated either using NHANES data
(for individuals ages 1 and older) or using age-based reference ranges
(for individuals under age 1). Fup was treated as a random variable
obeying a distribution censored below the average limit of quantification
(LOQ) of the in vitro assay. Specifically, Fup was assumed to obey a
normal distribution truncated below at 0 and above at 1, centered at the
Fup value measured in vitro, with a 30
the average LOQ (0.01), Fup was instead drawn from a uniform distribution
between 0 and 0.01. Fup was assumed to be independent of all
other parameters. This censored normal distribution was chosen to
match that used in Wambaugh et al. (2015).
</p>
<p>Variability in hepatocellularity (106 cells/g liver) and Mliver (kg)
were simulated. The remaining source of variability in CLint,h is variability
in CLint, which was simulated using a Gaussian mixture distribution
to represent the population proportions of poor metabolizers
(PMs) and non-PMs of each substance. The true prevalence of PMs is
isozyme-specific (Ma et al., 2002; Yasuda et al., 2008); however, isozyme-
specific metabolism data were not available for the majority of
chemicals considered. We therefore made a simplifying assumption that
5
slower than average.
With 95
a normal distribution truncated below at zero, centered at the value
measured in vitro, with a 30
CLint was drawn from a PM distribution: a truncated normal
distribution centered on one-tenth of the in vitro value with 30
Both CLint itself and the probability of being a PM were assumed to be
independent of all other parameters. The truncated normal nonePM
distribution was chosen because it has been used (with 100
in previous work (Rotroff et al., 2010; Wambaugh et al., 2015;
Wetmore et al., 2014; Wetmore et al., 2015; Wetmore et al., 2012); the
PM distribution was chosen to comport with the nonePM distribution.
</p>


<h3>Main function to generate a population</h3>

<p>If you just want to generate
a table of (chemical-independent) population physiology parameters, use 
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>.
</p>


<h3>Using HTTK-Pop with HTTK</h3>

<p>To generate a population and then run an 
HTTK model for that population, the workflow is as follows: </p>
 
<ol>
<li><p> Generate a population using <code><a href="#topic+httkpop_generate">httkpop_generate</a></code>. </p>
</li>
<li><p> For
a given HTTK chemical and general model, convert the population data to 
corresponding sets of HTTK model parameters using 
<code><a href="#topic+httkpop_mc">httkpop_mc</a></code>.</p>
</li></ol>



<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.     
</p>
<p>Levey, A.S., Stevens, L.A., Schmid, C.H., Zhang, Y.L., Castro, A.F., Feldman, H.I., et al.,
2009. A new equation to estimate glomerular filtration rate. Ann. Intern. Med. 150,
604-612.
</p>
<p>Howgate, E., Rowland-Yeo, K., Proctor, N., Tucker, G., Rostami-Hodjegan, A., 2006.
Prediction of in vivo drug clearance from in vitro data. I: impact of inter-individual
variability. Xenobiotica 36, 473-497.
</p>
<p>Jamei, M., Dickinson, G.L., Rostami-Hodjegan, A., 2009a. A framework for assessing
inter-individual variability in pharmacokinetics using virtual human populations and
integrating general knowledge of physical chemistry, biology, anatomy, physiology
and genetics: a tale of 'bottom-up' vs 'top-down' recognition of covariates. Drug Metab.
Pharmacokinet. 24, 53-75.
</p>
<p>Johnson, T.N., Rostami-Hodjegan, A., Tucker, G.T., 2006. Prediction of the clearance of
eleven drugs and associated variability in neonates, infants and children. Clin.
Pharmacokinet. 45, 931-956.
</p>
<p>McNally, K., Cotton, R., Hogg, A., Loizou, G., 2014. PopGen: a virtual human population
generator. Toxicology 315, 70-85.
</p>
<p>Price, P.S., Conolly, R.B., Chaisson, C.F., Gross, E.A., Young, J.S., Mathis, E.T., et al.,
2003. Modeling interindividual variation in physiological factors used in PBPK
models of humans. Crit. Rev. Toxicol. 33, 469-503.
</p>
<p>Barter, Z.E., Bayliss, M.K., Beaune, P.H., Boobis, A.R., Carlile, D.J., Edwards, R.J., et al.,
2007. Scaling factors for the extrapolation of in vivo metabolic drug clearance from
in vitro data: reaching a consensus on values of human micro-somal protein and
hepatocellularity per gram of liver. Curr. Drug Metab. 8, 33-45.
</p>
<p>Baxter-Jones, A.D., Faulkner, R.A., Forwood, M.R., Mirwald, R.L., Bailey, D.A., 2011.
Bone mineral accrual from 8 to 30 years of age: an estimation of peak bone mass. J.
Bone Miner. Res. 26, 1729-1739.
</p>
<p>Bosgra, S., van Eijkeren, J., Bos, P., Zeilmaker, M., Slob, W., 2012. An improved model to
predict physiologically based model parameters and their inter-individual variability
from anthropometry. Crit. Rev. Toxicol. 42, 751-767.
</p>
<p>Koo, W.W., Walters, J.C., Hockman, E.M., 2000. Body composition in human infants at
birth and postnatally. J. Nutr. 130, 2188-2194.
</p>
<p>Looker, A., Borrud, L., Hughes, J., Fan, B., Shepherd, J., Sherman, M., 2013. Total body
bone area, bone mineral content, and bone mineral density for individuals aged 8
years and over: United States, 1999-2006. In: Vital and health statistics Series 11,
Data from the National Health Survey, pp. 1-78.
</p>
<p>Ogiu, N., Nakamura, Y., Ijiri, I., Hiraiwa, K., Ogiu, T., 1997. A statistical analysis of the
internal organ weights of normal Japanese people. Health Phys. 72, 368-383.
</p>
<p>Schwartz, G.J., Work, D.F., 2009. Measurement and estimation of GFR in children and
adolescents. Clin. J. Am. Soc. Nephrol. 4, 1832-1843.
</p>
<p>Webber, C.E., Barr, R.D., 2012. Age-and gender-dependent values of skeletal muscle mass
in healthy children and adolescents. J. Cachex. Sarcopenia Muscle 3, 25-29.
</p>
<p>Johnson, C.L., Paulose-Ram, R., Ogden, C.L., Carroll, M.D., Kruszon-Moran, D.,
Dohrmann, S.M., et al., 2013. National health and nutrition examination survey:
analytic guidelines, 1999-2010. Vital and health statistics Series 2. Data Eval.
Methods Res. 1-24.
</p>
<p>Lumley, T., 2004. Analysis of complex survey samples. J. Stat. Softw. 9, 1-19.
</p>
<p>Grummer-Strawn, L.M., Reinold, C.M., Krebs, N.F., Control, C.f.D.; Prevention, 2010. Use
of World Health Organization and CDC Growth Charts for Children Aged 0-59
Months in the United States. Department of Health and Human Services, Centers for
Disease Control and Prevention.
</p>
<p>Kuczmarski, R.J., Ogden, C.L., Guo, S.S., Grummer-Strawn, L.M., Flegal, K.M., Mei, Z.,
et al., 2002. 2000 CDC growth charts for the United States: methods and development.
Vital Health Stat. Series 11, Data from the national health survey 246, 1-190.
</p>
<p>Ogden, C.L., Carroll, M.D., Kit, B.K., Flegal, K.M., 2014. Prevalence of childhood and
adult obesity in the United States, 2011-2012. JAMA 311, 806-814.
</p>
<p>WHO, 2006. In: WHO D.o.N.f.H.a.D. (Ed.), WHO Child Growth Standards: Length/Heightfor-
Age, Weight-for-Age, Weight-for-Length, Weight-for-Height and Body Mass Indexfor-
Age: Methods and Development.
</p>
<p>WHO, 2010. In: (WHO) W.H.O. (Ed.), WHO Anthro for Personal Computers Manual:
Software for Assessing Growth and Development of the World's Children, Version
3.2.2, 2011. WHO, Geneva.
</p>
<p>Valentin, J., 2002. Basic anatomical and physiological data for use in radiological protection:
reference values: ICRP publication 89. Ann. ICRP 32, 1-277.
</p>
<p>Johnson, T.N., Rostami-Hodjegan, A., Tucker, G.T., 2006. Prediction of the clearance of
eleven drugs and associated variability in neonates, infants and children. Clin.
Pharmacokinet. 45, 931-956.
</p>
<p>Verbraecken, J., Van de Heyning, P., De Backer, W., Van Gaal, L., 2006. Body surface area
in normal-weight, overweight, and obese adults. A comparison study. Metabolism 55,
515-524
</p>
<p>Haycock, G.B., Schwartz, G.J., Wisotsky, D.H., 1978. Geometric method for measuring
body surface area: a height-weight formula validated in infants, children, and adults.
J. Pediatr. 93, 62-66.
</p>
<p>Lubin, B., 1987. Reference values in infancy and childhood. In: Nathan, D., Oski, F. (Eds.),
Hematology of Infancy and Childhood.
</p>
<p>Wambaugh, J.F., Wetmore, B.A., Pearce, R., Strope, C., Goldsmith, R., Sluka, J.P., et al.,
2015. Toxicokinetic triage for environmental chemicals. Toxicol. Sci. 147, 55-67
</p>
<p>Ma, M.K., Woo, M.H., Mcleod, H.L., 2002. Genetic basis of drug metabolism. Am. J.
Health Syst. Pharm. 59, 2061-2069.
</p>
<p>Yasuda, S.U., Zhang, L., Huang, S.M., 2008. The role of ethnicity in variability in response
to drugs: focus on clinical pharmacology studies. Clin. Pharmacol. Ther. 84, 417-423.
</p>
<p>Rotroff, D.M., Wetmore, B.A., Dix, D.J., Ferguson, S.S., Clewell, H.J., Houck, K.A., et al.,
2010. Incorporating human dosimetry and exposure into high-throughput in vitro
toxicity screening. Toxicol. Sci. 117, 348-358.
</p>
<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A., Rotroff, D.M., Freeman,
K., et al., 2012. Integration of dosimetry, exposure, and high-throughput screening
data in chemical toxicity assessment. Toxicol. Sci. 125, 157-174.
</p>
<p>Wetmore, B.A., Allen, B., Clewell 3rd, H.J., Parker, T., Wambaugh, J.F., Almond, L.M.,
et al., 2014. Incorporating population variability and susceptible subpopulations into
dosimetry for high-throughput toxicity testing. Toxicol. Sci. 142, 210-224.
</p>
<p>Wetmore, B.A., Wambaugh, J.F., Allen, B., Ferguson, S.S., Sochaski, M.A., Setzer, R.W.,
et al., 2015. Incorporating high-throughput exposure predictions with Dosimetryadjusted
in vitro bioactivity to inform chemical toxicity testing. Toxicol. Sci. 148,
121-136.
</p>

<hr>
<h2 id='httkpop_biotophys_default'>Convert HTTK-Pop-generated parameters to HTTK physiological parameters</h2><span id='topic+httkpop_biotophys_default'></span>

<h3>Description</h3>

<p>Convert HTTK-Pop-generated parameters to HTTK physiological parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>httkpop_biotophys_default(indiv_dt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="httkpop_biotophys_default_+3A_indiv_dt">indiv_dt</code></td>
<td>
<p>The data.table object returned by <code>httkpop_generate()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table with the physiological parameters expected by any HTTK 
model, including body weight (BW), hematocrit, tissue volumes per kg body
weight, tissue flows as fraction of CO, CO per (kg BW)^3/4, GFR per (kg
BW)^3/4, portal vein flow per (kg BW)^3/4, and liver density.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='httkpop_direct_resample'>Generate a virtual population by directly resampling the NHANES data.</h2><span id='topic+httkpop_direct_resample'></span>

<h3>Description</h3>

<p>Generate a virtual population by directly resampling the NHANES data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>httkpop_direct_resample(
  nsamp = NULL,
  gendernum = NULL,
  agelim_years = NULL,
  agelim_months = NULL,
  weight_category = c("Underweight", "Normal", "Overweight", "Obese"),
  gfr_category = c("Normal", "Kidney Disease", "Kidney Failure"),
  reths = c("Mexican American", "Other Hispanic", "Non-Hispanic White",
    "Non-Hispanic Black", "Other"),
  gfr_resid_var = TRUE,
  ckd_epi_race_coeff = FALSE,
  nhanes_mec_svy
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="httkpop_direct_resample_+3A_nsamp">nsamp</code></td>
<td>
<p>The desired number of individuals in the virtual population.
<code>nsamp</code> need not be provided if <code>gendernum</code> is provided.</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_+3A_gendernum">gendernum</code></td>
<td>
<p>Optional: A named list giving the numbers of male and
female individuals to include in the population, e.g. <code>list(Male=100,
Female=100)</code>. Default is NULL, meaning both males and females are included,
in their proportions in the NHANES data. If both <code>nsamp</code> and
<code>gendernum</code> are provided, they must agree (i.e., <code>nsamp</code> must be
the sum of <code>gendernum</code>).</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_+3A_agelim_years">agelim_years</code></td>
<td>
<p>Optional: A two-element numeric vector giving the
minimum and maximum ages (in years) to include in the population. Default is
c(0,79). If <code>agelim_years</code> is provided and <code>agelim_months</code> is not,
<code>agelim_years</code> will override the default value of <code>agelim_months</code>.</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_+3A_agelim_months">agelim_months</code></td>
<td>
<p>Optional: A two-element numeric vector giving the
minimum and maximum ages (in months) to include in the population. Default
is c(0, 959), equivalent to the default <code>agelim_years</code>. If
<code>agelim_months</code> is provided and <code>agelim_years</code> is not,
agelim_months will override the default values of <code>agelim_years</code>.</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_+3A_weight_category">weight_category</code></td>
<td>
<p>Optional: The weight categories to include in the
population. Default is <code>c('Underweight', 'Normal', 'Overweight',
'Obese')</code>. User-supplied vector must contain one or more of these strings.</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_+3A_gfr_category">gfr_category</code></td>
<td>
<p>The kidney function categories to include in the
population. Default is <code>c('Normal','Kidney Disease', 'Kidney Failure')</code>
to include all kidney function levels.</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_+3A_reths">reths</code></td>
<td>
<p>Optional: a character vector giving the races/ethnicities to
include in the population. Default is <code>c('Mexican American','Other
Hispanic','Non-Hispanic White','Non-Hispanic Black','Other')</code>, to include
all races and ethnicities in their proportions in the NHANES data.
User-supplied vector must contain one or more of these strings.</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_+3A_gfr_resid_var">gfr_resid_var</code></td>
<td>
<p>Logical value indicating whether or not to include
residual variability when generating GFR values. (Default is TRUE.)</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_+3A_ckd_epi_race_coeff">ckd_epi_race_coeff</code></td>
<td>
<p>Logical value indicating whether or not to use the
&quot;race coefficient&quot; from the CKD-EPI equation when estimating GFR values.
(Default is FALSE.)</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_+3A_nhanes_mec_svy">nhanes_mec_svy</code></td>
<td>
<p><code>surveydesign</code> object created from
<code><a href="#topic+mecdt">mecdt</a></code> using <code><a href="survey.html#topic+svydesign">svydesign</a></code> (this is done in
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table where each row represents an individual, and each
column represents a demographic, anthropometric, or physiological parameter.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='httkpop_direct_resample_inner'>Inner loop function called by <code>httkpop_direct_resample</code>.</h2><span id='topic+httkpop_direct_resample_inner'></span>

<h3>Description</h3>

<p>Inner loop function called by <code>httkpop_direct_resample</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>httkpop_direct_resample_inner(
  nsamp,
  gendernum,
  agelim_months,
  agelim_years,
  reths,
  weight_category,
  gfr_resid_var,
  ckd_epi_race_coeff,
  nhanes_mec_svy
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="httkpop_direct_resample_inner_+3A_nsamp">nsamp</code></td>
<td>
<p>The desired number of individuals in the virtual population. 
<code>nsamp</code> need not be provided if <code>gendernum</code> is provided.</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_inner_+3A_gendernum">gendernum</code></td>
<td>
<p>Optional: A named list giving the numbers of male and female 
individuals to include in the population, e.g. <code>list(Male=100, 
Female=100)</code>. Default is NULL, meaning both males and females are included, 
in their proportions in the NHANES data. If both <code>nsamp</code> and 
<code>gendernum</code> are provided, they must agree (i.e., <code>nsamp</code> must be
the sum of <code>gendernum</code>).</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_inner_+3A_agelim_months">agelim_months</code></td>
<td>
<p>Optional: A two-element numeric vector giving the minimum
and maximum ages (in months) to include in the population. Default is c(0, 
959), equivalent to the default <code>agelim_years</code>. If <code>agelim_months</code>
is provided and <code>agelim_years</code> is not, agelim_months will override the 
default values of <code>agelim_years</code>.</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_inner_+3A_agelim_years">agelim_years</code></td>
<td>
<p>Optional: A two-element numeric vector giving the minimum 
and maximum ages (in years) to include in the population. Default is 
c(0,79). If <code>agelim_years</code> is provided and <code>agelim_months</code> is not,
<code>agelim_years</code> will override the default value of <code>agelim_months</code>.</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_inner_+3A_reths">reths</code></td>
<td>
<p>Optional: a character vector giving the races/ethnicities to 
include in the population. Default is <code>c('Mexican American','Other 
Hispanic','Non-Hispanic White','Non-Hispanic Black','Other')</code>, to include 
all races and ethnicities in their proportions in the NHANES data. 
User-supplied vector must contain one or more of these strings.</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_inner_+3A_weight_category">weight_category</code></td>
<td>
<p>Optional: The weight categories to include in the 
population. Default is <code>c('Underweight', 'Normal', 'Overweight', 
'Obese')</code>. User-supplied vector must contain one or more of these strings.</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_inner_+3A_gfr_resid_var">gfr_resid_var</code></td>
<td>
<p>Logical value indicating whether or not to include
residual variability when generating GFR values. (Default is TRUE, passed from
'httkpop_direct_resample'.)</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_inner_+3A_ckd_epi_race_coeff">ckd_epi_race_coeff</code></td>
<td>
<p>Logical value indicating whether or not to use the
&quot;race coefficient&quot; from the CKD-EPI equation when estimating GFR values.
(Default is FALSE, passed from 'httkpop_direct_resample'.)</p>
</td></tr>
<tr><td><code id="httkpop_direct_resample_inner_+3A_nhanes_mec_svy">nhanes_mec_svy</code></td>
<td>
<p><code>surveydesign</code> object created from
<code><a href="#topic+mecdt">mecdt</a></code> using <code><a href="survey.html#topic+svydesign">svydesign</a></code> (this is done in
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table where each row represents an individual, and
each column represents a demographic, anthropometric, or physiological
parameter.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='httkpop_generate'>Generate a virtual population for PBTK</h2><span id='topic+httkpop_generate'></span>

<h3>Description</h3>

<p>Generate a virtual population characterized by demographic, anthropometric, and physiological parameters relevant to PBTK.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>httkpop_generate(
  method,
  nsamp = NULL,
  gendernum = NULL,
  agelim_years = NULL,
  agelim_months = NULL,
  weight_category = c("Underweight", "Normal", "Overweight", "Obese"),
  gfr_category = c("Normal", "Kidney Disease", "Kidney Failure"),
  reths = c("Mexican American", "Other Hispanic", "Non-Hispanic White",
    "Non-Hispanic Black", "Other"),
  gfr_resid_var = TRUE,
  ckd_epi_race_coeff = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="httkpop_generate_+3A_method">method</code></td>
<td>
<p>The population-generation method to use. Either &quot;virtual
individuals&quot; or &quot;direct resampling.&quot; Short names may be used: &quot;d&quot; or &quot;dr&quot;
for &quot;direct resampling&quot;, and &quot;v&quot; or &quot;vi&quot; for &quot;virtual individuals&quot;.</p>
</td></tr>
<tr><td><code id="httkpop_generate_+3A_nsamp">nsamp</code></td>
<td>
<p>The desired number of individuals in the virtual population.
<code>nsamp</code> need not be provided if <code>gendernum</code> is provided.</p>
</td></tr>
<tr><td><code id="httkpop_generate_+3A_gendernum">gendernum</code></td>
<td>
<p>Optional: A named list giving the numbers of male and female
individuals to include in the population, e.g. <code>list(Male=100,
Female=100)</code>. Default is NULL, meaning both males and females are included,
in their proportions in the NHANES data. If both <code>nsamp</code> and
<code>gendernum</code> are provided, they must agree (i.e., <code>nsamp</code> must be
the sum of <code>gendernum</code>).</p>
</td></tr>
<tr><td><code id="httkpop_generate_+3A_agelim_years">agelim_years</code></td>
<td>
<p>Optional: A two-element numeric vector giving the minimum
and maximum ages (in years) to include in the population. Default is
c(0,79). If only a single value is provided, both minimum and maximum ages
will be set to that value; e.g. <code>agelim_years=3</code> is equivalent to
<code>agelim_years=c(3,3)</code>. If <code>agelim_years</code> is provided and
<code>agelim_months</code> is not, <code>agelim_years</code> will override the default
value of <code>agelim_months</code>.</p>
</td></tr>
<tr><td><code id="httkpop_generate_+3A_agelim_months">agelim_months</code></td>
<td>
<p>Optional: A two-element numeric vector giving the
minimum and maximum ages (in months) to include in the population. Default
is c(0, 959), equivalent to the default <code>agelim_years</code>. If only a
single value is provided, both minimum and maximum ages will be set to that
value; e.g. <code>agelim_months=36</code> is equivalent to
<code>agelim_months=c(36,36)</code>. If <code>agelim_months</code> is provided and
<code>agelim_years</code> is not, <code>agelim_months</code> will override the default
values of <code>agelim_years</code>.</p>
</td></tr>
<tr><td><code id="httkpop_generate_+3A_weight_category">weight_category</code></td>
<td>
<p>Optional: The weight categories to include in the
population. Default is <code>c('Underweight', 'Normal', 'Overweight',
'Obese')</code>. User-supplied vector must contain one or more of these strings.</p>
</td></tr>
<tr><td><code id="httkpop_generate_+3A_gfr_category">gfr_category</code></td>
<td>
<p>The kidney function categories to include in the
population. Default is <code>c('Normal','Kidney Disease', 'Kidney
Failure')</code> to include all kidney function levels.</p>
</td></tr>
<tr><td><code id="httkpop_generate_+3A_reths">reths</code></td>
<td>
<p>Optional: a character vector giving the races/ethnicities to
include in the population. Default is <code>c('Mexican American','Other
Hispanic','Non-Hispanic White','Non-Hispanic Black','Other')</code>, to include
all races and ethnicities in their proportions in the NHANES data.
User-supplied vector must contain one or more of these strings.</p>
</td></tr>
<tr><td><code id="httkpop_generate_+3A_gfr_resid_var">gfr_resid_var</code></td>
<td>
<p>TRUE to add residual variability to GFR predicted from
serum creatinine; FALSE to not add residual variability</p>
</td></tr>
<tr><td><code id="httkpop_generate_+3A_ckd_epi_race_coeff">ckd_epi_race_coeff</code></td>
<td>
<p>TRUE to use the CKD-EPI equation as originally
published (with a coefficient changing predicted GFR for individuals
identified as &quot;Non-Hispanic Black&quot;); FALSE to set this coefficient to 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Demographic and anthropometric (body measures) variables, along with serum creatinine and hematocrit,
are generated from survey data from the Centers for Disease Control's National Health and Nutrition Examination Survey (NHANES).
Those data are stored in the object <code>nhanes_mec_svy</code> (a <code>survey.design</code> object, see package <code>survey</code>).
With <code>method = "d"</code>, these variables will be sampled with replacement directly from NHANES data. Each NHANES respondent's likelihood of being sampled is given by their sample weight.
With <code>method = "v"</code>, these variables will be sampled from distributions fitted to NHANES data.
Tissue masses and flows are generated based on demographic, body measures, and serum creatinine values, using regression equations from the literature and/or allometric scaling based on height. 
Extensive details about how each of these parameters are generated are available in the supplemental material of Ring et al. (2017) (see References for full citation).
</p>


<h3>Value</h3>

<p>A data.table where each row represents an individual, and each column
represents a demographic, anthropometric, or physiological parameter.
Details of the parameters returned and their units are in the following tables.
</p>


<h3>Demographic variables</h3>


<table>
<tr>
 <td style="text-align: right;">
  <strong>Name</strong> </td><td style="text-align: right;"> <strong>Definition</strong> </td><td style="text-align: right;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>seqn</code> </td><td style="text-align: right;"> NHANES unique identifier (only included if <code>method = "direct resampling"</code>) </td><td style="text-align: right;"> NA </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>gender</code> </td><td style="text-align: right;"> Sex: "Male" or "Female" </td><td style="text-align: right;"> NA </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>reth</code> </td><td style="text-align: right;"> Race/ethnicity: "Non-Hispanic Black", "Non-Hispanic white", "Mexican American", "Other Hispanic", or "Other". </td><td style="text-align: right;"> NA </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>age_years</code> </td><td style="text-align: right;"> Age (0-79 years) </td><td style="text-align: right;"> years </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>age_months</code> </td><td style="text-align: right;"> Age (0-959 months) </td><td style="text-align: right;"> months
  </td>
</tr>

</table>



<h3>Body measures and laboratory measurements</h3>


<table>
<tr>
 <td style="text-align: right;">
   <strong>Name</strong> </td><td style="text-align: right;"> <strong>Definition</strong> </td><td style="text-align: right;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>height</code> </td><td style="text-align: right;"> Height </td><td style="text-align: right;"> cm </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>weight</code> </td><td style="text-align: right;"> Body weight </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>serum_creat</code> </td><td style="text-align: right;"> Serum creatinine </td><td style="text-align: right;"> mg/dL </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>hematocrit</code> </td><td style="text-align: right;"> Hematocrit (percentage by volume of red blood cells in blood) </td><td style="text-align: right;"> % 
  </td>
</tr>

</table>



<h3>Tissue masses</h3>


<table>
<tr>
 <td style="text-align: right;">
   <strong>Name</strong> </td><td style="text-align: right;"> <strong>Definition</strong> </td><td style="text-align: right;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Blood_mass</code> </td><td style="text-align: right;"> Mass of blood </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Brain_mass</code> </td><td style="text-align: right;"> Mass of brain </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Gonads_mass</code> </td><td style="text-align: right;"> Mass of gonads </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Heart_mass</code> </td><td style="text-align: right;"> Mass of heart </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Kidneys_mass</code> </td><td style="text-align: right;"> Mass of kidneys </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Large_intestine_mass</code> </td><td style="text-align: right;"> Mass of large intestine </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Liver_mass</code> </td><td style="text-align: right;"> Mass of liver </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Lung_mass</code> </td><td style="text-align: right;"> Mass of lungs </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Muscle_mass</code> </td><td style="text-align: right;"> Mass of skeletal muscle </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Pancreas_mass</code> </td><td style="text-align: right;"> Mass of pancreas </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Skeleton_mass</code> </td><td style="text-align: right;"> Mass of skeleton (including bone, red and yellow marrow, cartilage, periarticular tissue) </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Skin_mass</code> </td><td style="text-align: right;"> Mass of skin </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Small_intestine_mass</code> </td><td style="text-align: right;"> Mass of small intestine </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Spleen_mass</code> </td><td style="text-align: right;"> Mass of spleen </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Stomach_mass</code> </td><td style="text-align: right;"> Mass of stomach tissue </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Other_mass</code> </td><td style="text-align: right;"> Mass of GI tract contents (1.4% of body weight) and tissues not otherwise enumerated (3.3% of body weight). </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>org_mass_sum</code> </td><td style="text-align: right;"> Sum of the above tissue masses.  A check to ensure this is less than body weight. </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Adipose_mass</code> </td><td style="text-align: right;"> Mass of adipose tissue. Assigned as <code>weight - org_mass_sum</code>. </td><td style="text-align: right;"> kg
  </td>
</tr>

</table>



<h3>Tissue flows</h3>


<table>
<tr>
 <td style="text-align: right;">
 <strong>Name</strong> </td><td style="text-align: right;"> <strong>Definition</strong> </td><td style="text-align: right;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Adipose_flow</code> </td><td style="text-align: right;"> Blood flow to adipose tissue </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Brain_flow</code> </td><td style="text-align: right;"> Blood flow to brain tissue </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>CO</code> </td><td style="text-align: right;"> Cardiac output </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Gonads_flow</code> </td><td style="text-align: right;"> Blood flow to gonads tissue </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Heart_flow</code> </td><td style="text-align: right;"> Blood flow to heart tissue </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Kidneys_flow</code> </td><td style="text-align: right;"> Blood flow to kidneys tissue (not for glomerular filtration!) </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Large_intestine_flow</code> </td><td style="text-align: right;"> Blood flow to large intestine tissue </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Liver_flow</code> </td><td style="text-align: right;"> Blood flow to liver tissue </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Lung_flow</code> </td><td style="text-align: right;"> Blood flow to lung tissue </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Muscle_flow</code> </td><td style="text-align: right;"> Blood flow to skeletal muscle tissue </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Pancreas_flow</code> </td><td style="text-align: right;"> Blood flow to pancreas tissue </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Skeleton_flow</code> </td><td style="text-align: right;"> Blood flow to skeleton </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Skin_flow</code> </td><td style="text-align: right;"> Blood flow to skin </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Small_intestine_flow</code> </td><td style="text-align: right;"> Blood flow to small intestine </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Spleen_flow</code> </td><td style="text-align: right;"> Blood flow to spleen </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>Stomach_flow</code> </td><td style="text-align: right;"> Blood flow to stomach </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>org_flow_check</code> </td><td style="text-align: right;"> Sum of blood flows as a fraction of cardiac output (CO). A check to make sure this is less than 1. </td><td style="text-align: right;"> Unitless fraction
  </td>
</tr>

</table>



<h3>Adjusted variables</h3>


<table>
<tr>
 <td style="text-align: right;">
 <strong>Name</strong> </td><td style="text-align: right;"> <strong>Definition</strong> </td><td style="text-align: right;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>weight_adj</code> </td><td style="text-align: right;"> Adjusted body weight: Sum of all tissue masses. </td><td style="text-align: right;"> kg </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>BSA_adj</code> </td><td style="text-align: right;"> Adjusted body surface area, based on <code>height</code> and <code>weight_adj</code>. </td><td style="text-align: right;"> cm^2 </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>million.cells.per.gliver</code> </td><td style="text-align: right;"> Hepatocellularity </td><td style="text-align: right;"> 1e6 cells/g liver </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>gfr_est</code> </td><td style="text-align: right;"> Glomerular filtration rate (GFR) estimated using either the CKD-EPI equation (for adults) or a body-surface-area-based equation (for children). </td><td style="text-align: right;"> mL/min/1.73 m^2 body surface area </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>bmi_adj</code> </td><td style="text-align: right;"> Body mass index (BMI), adjusted to match <code>weight_adj</code> and <code>height</code>. </td><td style="text-align: right;"> kg/m^2 </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>weight_class</code> </td><td style="text-align: right;"> Weight category based on <code>bmi_adj</code>: "Underweight" (BMI &lt; 18.5), "Normal" (18.5 &lt; BMI &lt; 24.9), "Overweight" (25.0 &lt; BMI &lt; 29.9), or "Obese" (BMI &gt;= 30) </td><td style="text-align: right;"> Unitless category </td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>gfr_class</code> </td><td style="text-align: right;"> Kidney function category based on GFR: "Normal" (GFR &gt;=60 mL/min/1.73 m^2), "Kidney Disease" (15 &lt;= GFR &lt;= 60), or "Kidney Failure" (GFR &lt; 15). </td><td style="text-align: right;"> Unitless category
  </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

#Simply generate a virtual population of 100 individuals,
 #using the direct-resampling method
 set.seed(42)
httkpop_generate(method='direct resampling', nsamp=100)
#Generate a population using the virtual-individuals method,
#including 80 females and 20 males,
#including only ages 20-65,
#including only Mexican American and
 #Non-Hispanic Black individuals,
 #including only non-obese individuals
httkpop_generate(method = 'virtual individuals',
gendernum=list(Female=80,
Male=20),
agelim_years=c(20,65),
reths=c('Mexican American',
'Non-Hispanic Black'),
weight_category=c('Underweight',
'Normal',
'Overweight'))




</code></pre>

<hr>
<h2 id='httkpop_mc'>httk-pop: Correlated human physiological parameter Monte Carlo</h2><span id='topic+httkpop_mc'></span>

<h3>Description</h3>

<p>This is the core function for httk-pop correlated human physiological 
variability simulation as described by Ring et al. (2017) 
(<a href="https://doi.org/10.1016/j.envint.2017.06.004">doi:10.1016/j.envint.2017.06.004</a>). This functions 
takes the data table of population biometrics (one individual per row)
generated by <code><a href="#topic+httkpop_generate">httkpop_generate</a></code>, and converts it
to the corresponding table of HTTK model parameters for a specified HTTK 
model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>httkpop_mc(model, samples = 1000, httkpop.dt = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="httkpop_mc_+3A_model">model</code></td>
<td>
<p>One of the HTTK models: &quot;1compartment&quot;, &quot;3compartmentss&quot;,
&quot;3compartment&quot;, or &quot;pbtk&quot;.</p>
</td></tr>
<tr><td><code id="httkpop_mc_+3A_samples">samples</code></td>
<td>
<p>The number of Monte Carlo samples to use (can often think of these
as separate individuals)</p>
</td></tr>
<tr><td><code id="httkpop_mc_+3A_httkpop.dt">httkpop.dt</code></td>
<td>
<p>A data table generated by <code><a href="#topic+httkpop_generate">httkpop_generate</a></code>.
This defaults to NULL, in which case <code><a href="#topic+httkpop_generate">httkpop_generate</a></code> is 
called to generate this table.</p>
</td></tr>
<tr><td><code id="httkpop_mc_+3A_...">...</code></td>
<td>
<p>Additional arugments passed on to <code><a href="#topic+httkpop_generate">httkpop_generate</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Monte Carlo methods used here were recently updated and described by
Breen et al. (submitted).
</p>


<h3>Value</h3>

<p>A data.table with a row for each individual in the sample and a column for
each parater in the model.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring and John Wambaugh
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118. 
</p>
<p>Breen M, Wambaugh JF, Bernstein A, Sfeir M, Ring CL (2022).
&ldquo;Simulating toxicokinetic variability to identify susceptible and highly exposed populations.&rdquo;
<em>Journal of Exposure Science &amp; Environmental Epidemiology</em>, <b>32</b>(6), 855&ndash;863. 
</p>
<p>Rowland M, Benet LZ, Graham GG (1973).
&ldquo;Clearance concepts in pharmacokinetics.&rdquo;
<em>Journal of pharmacokinetics and biopharmaceutics</em>, <b>1</b>(2), 123&ndash;136.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(42)
indiv_examp &lt;- httkpop_generate(method="d", nsamp=10)

httk_param &lt;- httkpop_mc(httkpop.dt=indiv_examp, 
                        samples=10,
                        model="1compartment")

</code></pre>

<hr>
<h2 id='httkpop_virtual_indiv'>Generate a virtual population by the virtual individuals method.</h2><span id='topic+httkpop_virtual_indiv'></span>

<h3>Description</h3>

<p>Generate a virtual population by the virtual individuals method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>httkpop_virtual_indiv(
  nsamp = NULL,
  gendernum = NULL,
  agelim_years = NULL,
  agelim_months = NULL,
  weight_category = c("Underweight", "Normal", "Overweight", "Obese"),
  gfr_category = c("Normal", "Kidney Disease", "Kidney Failure"),
  reths = c("Mexican American", "Other Hispanic", "Non-Hispanic White",
    "Non-Hispanic Black", "Other"),
  gfr_resid_var = TRUE,
  ckd_epi_race_coeff = FALSE,
  nhanes_mec_svy
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="httkpop_virtual_indiv_+3A_nsamp">nsamp</code></td>
<td>
<p>The desired number of individuals in the virtual population.
<code>nsamp</code> need not be provided if <code>gendernum</code> is provided.</p>
</td></tr>
<tr><td><code id="httkpop_virtual_indiv_+3A_gendernum">gendernum</code></td>
<td>
<p>Optional: A named list giving the numbers of male and female
individuals to include in the population, e.g. <code>list(Male=100,
Female=100)</code>. Default is NULL, meaning both males and females are included,
in their proportions in the NHANES data. If both <code>nsamp</code> and
<code>gendernum</code> are provided, they must agree (i.e., <code>nsamp</code> must be
the sum of <code>gendernum</code>).</p>
</td></tr>
<tr><td><code id="httkpop_virtual_indiv_+3A_agelim_years">agelim_years</code></td>
<td>
<p>Optional: A two-element numeric vector giving the minimum
and maximum ages (in years) to include in the population. Default is
c(0,79). If <code>agelim_years</code> is provided and <code>agelim_months</code> is not,
<code>agelim_years</code> will override the default value of <code>agelim_months</code>.</p>
</td></tr>
<tr><td><code id="httkpop_virtual_indiv_+3A_agelim_months">agelim_months</code></td>
<td>
<p>Optional: A two-element numeric vector giving the minimum
and maximum ages (in months) to include in the population. Default is c(0,
959), equivalent to the default <code>agelim_years</code>. If <code>agelim_months</code>
is provided and <code>agelim_years</code> is not, agelim_months will override the
default values of <code>agelim_years</code>.</p>
</td></tr>
<tr><td><code id="httkpop_virtual_indiv_+3A_weight_category">weight_category</code></td>
<td>
<p>Optional: The weight categories to include in the
population. Default is <code>c('Underweight', 'Normal', 'Overweight',
'Obese')</code>. User-supplied vector must contain one or more of these strings.</p>
</td></tr>
<tr><td><code id="httkpop_virtual_indiv_+3A_gfr_category">gfr_category</code></td>
<td>
<p>The kidney function categories to include in the
population. Default is <code>c('Normal','Kidney Disease', 'Kidney Failure')</code>
to include all kidney function levels.</p>
</td></tr>
<tr><td><code id="httkpop_virtual_indiv_+3A_reths">reths</code></td>
<td>
<p>Optional: a character vector giving the races/ethnicities to
include in the population. Default is <code>c('Mexican American','Other
Hispanic','Non-Hispanic White','Non-Hispanic Black','Other')</code>, to include
all races and ethnicities in their proportions in the NHANES data.
User-supplied vector must contain one or more of these strings.</p>
</td></tr>
<tr><td><code id="httkpop_virtual_indiv_+3A_gfr_resid_var">gfr_resid_var</code></td>
<td>
<p>Logical value indicating whether or not to include
residual variability when generating GFR values. (Default is TRUE.)</p>
</td></tr>
<tr><td><code id="httkpop_virtual_indiv_+3A_ckd_epi_race_coeff">ckd_epi_race_coeff</code></td>
<td>
<p>Logical value indicating whether or not to use the
&quot;race coefficient&quot; from the CKD-EPI equation when estimating GFR values.
(Default is FALSE.)</p>
</td></tr>
<tr><td><code id="httkpop_virtual_indiv_+3A_nhanes_mec_svy">nhanes_mec_svy</code></td>
<td>
<p><code>surveydesign</code> object created from
<code><a href="#topic+mecdt">mecdt</a></code> using <code><a href="survey.html#topic+svydesign">svydesign</a></code> (this is done in
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code>, which calls this function)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table where each row represents an individual, and each column
represents a demographic, anthropometric, or physiological parameter.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='hw_H'>KDE bandwidth for residual variability in height/weight</h2><span id='topic+hw_H'></span>

<h3>Description</h3>

<p>Bandwidths used for a two-dimensional kernel density estimation of the joint
distribution of residual errors around smoothing spline fits of height vs. age
and weight vs. age for NHANES respondents in each of ten combinations of sex
and race/ethnicity categories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hw_H
</code></pre>


<h3>Format</h3>

<p>A named list with 10 elements, each a matrix with 2 rows and 2
columns. Each list element corresponds to, and is named for, one combination
of NHANES sex categories (Male and Female) and NHANES race/ethnicity
categories (Mexican American, Other Hispanic, Non-Hispanic White,
Non-Hispanic Black, and Other).
</p>


<h3>Details</h3>

<p>Each matrix is a variance-covariance matrix for a two-dimensional normal
distribution: this is the bandwidth to be used for a two-dimensional kernel
density estimation (KDE) (using a two-dimensional normal kernel) of the joint
distribution of residual errors around smoothing spline fits of height vs. age
and weight vs. age for NHANES respondents in the specified sex and
race/ethnicity category. Optimal bandwidths were pre-calculated by doing the
smoothing spline fits, getting the residuals, then calling
<code><a href="ks.html#topic+kde">kde</a></code> on the residuals (which calls <code><a href="ks.html#topic+Hpi">Hpi</a></code> to
compute the plug-in bandwidth).
</p>
<p>Used by HTTK-Pop only in &quot;virtual individuals&quot; mode (i.e.
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code> with <code>method = "v"</code>), in
<code><a href="#topic+gen_height_weight">gen_height_weight</a></code>.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='in.list'>Convenience Boolean (yes/no) functions to identify chemical membership in
several key lists.</h2><span id='topic+in.list'></span><span id='topic+is.nhanes'></span><span id='topic+is.nhanes.serum.parent'></span><span id='topic+is.nhanes.serum.analyte'></span><span id='topic+is.nhanes.blood.parent'></span><span id='topic+is.nhanes.blood.analyte'></span><span id='topic+is.nhanes.urine.parent'></span><span id='topic+is.nhanes.urine.analyte'></span><span id='topic+is.tox21'></span><span id='topic+is.toxcast'></span><span id='topic+is.expocast'></span><span id='topic+is.pharma'></span>

<h3>Description</h3>

<p>These functions allow easy identification of whether or not a chemical CAS
is included in various research projects. While it is our intent to keep
these lists up-to-date, the information here is only for convenience and
should not be considered to be definitive.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>in.list(chem.cas = NULL, which.list = "ToxCast")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="in.list_+3A_chem.cas">chem.cas</code></td>
<td>
<p>The Chemical Abstracts Service Resgistry Number (CAS-RN)
corresponding to the chemical of interest.</p>
</td></tr>
<tr><td><code id="in.list_+3A_which.list">which.list</code></td>
<td>
<p>A character string that can take the following values:
&quot;ToxCast&quot;, &quot;Tox21&quot;, &quot;ExpoCast&quot;, &quot;NHANES&quot;, &quot;&quot;NHANES.serum.parent&quot;,
&quot;NHANES.serum.analyte&quot;,&quot;NHANES.blood.parent&quot;,&quot;NHANES.blood.analyte&quot;,
&quot;NHANES.urine.parent&quot;,&quot;NHANES.urine.analyte&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Tox21: Toxicology in the 21st Century (Tox21) is a U.S. federal High
Throughput Screening (HTS) collaboration among EPA, NIH, including National
Center for Advancing Translational Sciences and the National Toxicology
Program at the National Institute of Environmental Health Sciences, and the
Food and Drug Administration.  (Bucher et al., 2008)
</p>
<p>ToxCast: The Toxicity Forecaster (ToxCast) is a HTS screening project led by
the U.S. EPA to perform additional testing of a subset of Tox21 chemicals.
(Judson et al. 2010)
</p>
<p>ExpoCast: ExpoCast (Exposure Forecaster) is an U.S. EPA research project to
generate tenetative exposure estimates (e.g., mg/kg BW/day) for thousands of
chemicals that have little other information using models and informatics.
(Wambaugh et al. 2014)
</p>
<p>NHANES: The U.S. Centers for Disease Control (CDC) National Health and
Nutrition Examination Survery (NHANES) is an on-going survey to characterize
the health and biometrics (e.g., weight, height) of the U.S. population. One
set of measurments includes the quantification of xenobiotic chemicals in
various samples (blood, serum, urine) of the thousands of surveyed
individuals. (CDC, 2014)
</p>


<h3>Value</h3>

<table>
<tr><td><code>logical</code></td>
<td>
<p>A Boolean (1/0) value that is TRUE if the chemical is
in the list.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Bucher, J. R. (2008). Guest Editorial: NTP: New Initiatives, New
Alignment. Environ Health Perspect 116(1).
</p>
<p>Judson, R. S., Houck, K. A., Kavlock, R. J., Knudsen, T. B., Martin, M. T.,
Mortensen, H. M., Reif, D. M., Rotroff, D. M., Shah, I., Richard, A. M. and
Dix, D. J. (2010). In Vitro Screening of Environmental Chemicals for
Targeted Testing Prioritization: The ToxCast Project. Environmental Health
Perspectives 118(4), 485-492.
</p>
<p>Wambaugh, J. F., Wang, A., Dionisio, K. L., Frame, A., Egeghy, P., Judson,
R. and Setzer, R. W. (2014). High Throughput Heuristics for Prioritizing
Human Exposure to Environmental Chemicals. Environmental Science &amp;
Technology, 10.1021/es503583j.
</p>
<p>CDC (2014). National Health and Nutrition Examination Survey. Available at:
https://www.cdc.gov/nchs/nhanes.htm.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+is.httk">is.httk</a></code> for determining inclusion in httk project
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

httk.table &lt;- get_cheminfo(info=c("CAS","Compound"))
httk.table[,"Rat"] &lt;- ""
httk.table[,"NHANES"] &lt;- ""
httk.table[,"Tox21"] &lt;- ""
httk.table[,"ToxCast"] &lt;- ""
httk.table[,"ExpoCast"] &lt;- ""
httk.table[,"PBTK"] &lt;- ""
# To make this example run quickly, this loop is only over the first five 
# chemicals. To build a table with all available chemicals use:
# for (this.cas in httk.table$CAS)
for (this.cas in httk.table$CAS[1:5])
{
  this.index &lt;- httk.table$CAS==this.cas
  if (is.nhanes(this.cas)) httk.table[this.index,"NHANES"] &lt;- "Y"
  if (is.tox21(this.cas)) httk.table[this.index,"Tox21"] &lt;- "Y"
  if (is.toxcast(this.cas)) httk.table[this.index,"ToxCast"] &lt;- "Y"
  if (is.expocast(this.cas)) httk.table[this.index,"ExpoCast"] &lt;- "Y"
  if (is.httk(this.cas,model="PBTK")) httk.table[this.index,"PBTK"] &lt;- "Y"
  if (is.httk(this.cas,species="rat")) httk.table[this.index,"Rat"] &lt;- "Y"
}


</code></pre>

<hr>
<h2 id='invitro_mc'>Monte Carlo for in vitro toxicokinetic parameters including uncertainty and variability.</h2><span id='topic+invitro_mc'></span>

<h3>Description</h3>

<p>Given a CAS in the HTTK data set, a virtual population from HTTK-Pop, some
user specifications on the assumed distributions of Funbound.plasma and
Clint, draw &quot;individual&quot; values of Funbound.plasma and Clint from those
distributions. The methodology for this function was developed and described
by Wambaugh et al. (2019)
(<a href="https://doi.org/10.1093/toxsci/kfz205">doi:10.1093/toxsci/kfz205</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>invitro_mc(
  parameters.dt = NULL,
  samples,
  fup.meas.mc = TRUE,
  fup.pop.mc = TRUE,
  clint.meas.mc = TRUE,
  clint.pop.mc = TRUE,
  fup.meas.cv = 0.4,
  clint.meas.cv = 0.3,
  fup.pop.cv = 0.3,
  clint.pop.cv = 0.3,
  caco2.meas.sd = 0.3,
  caco2.pop.sd = 0.3,
  Caco2.Fgut = TRUE,
  Caco2.Fabs = TRUE,
  keepit100 = FALSE,
  poormetab = TRUE,
  fup.lod = 0.01,
  fup.censored.dist = FALSE,
  adjusted.Funbound.plasma = TRUE,
  adjusted.Clint = TRUE,
  clint.pvalue.threshold = 0.05,
  minimum.Funbound.plasma = 1e-04
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="invitro_mc_+3A_parameters.dt">parameters.dt</code></td>
<td>
<p>A data table of physiological and chemical-specific parameters</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_samples">samples</code></td>
<td>
<p>The number of samples to draw.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_fup.meas.mc">fup.meas.mc</code></td>
<td>
<p>Logical &ndash; should we perform measurment (uncertainty)
Monte Carlo for <code>Funbound.plasma</code> values (Default TRUE). If FALSE, 
the user may choose to provide columns for &quot;unadjusted.Funbound.plasma&quot; or
&quot;fup.mean&quot; from their own methods.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_fup.pop.mc">fup.pop.mc</code></td>
<td>
<p>Logical &ndash; should we perform population (variability)
Monte Carlo for <code>Funbound.plasma</code> values (Default TRUE)</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_clint.meas.mc">clint.meas.mc</code></td>
<td>
<p>Logical &ndash; should we perform measurment (uncertainty)
Monte Carlo for <code>Clint</code> values (Default TRUE)</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_clint.pop.mc">clint.pop.mc</code></td>
<td>
<p>Logical &ndash; should we perform population (variability)
Monte Carlo for <code>Clint</code> values (Default TRUE)</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_fup.meas.cv">fup.meas.cv</code></td>
<td>
<p>Coefficient of variation of distribution of measured
<code>Funbound.plasma</code> values.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_clint.meas.cv">clint.meas.cv</code></td>
<td>
<p>Coefficient of variation of distribution of measured 
<code>Clint</code> values.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_fup.pop.cv">fup.pop.cv</code></td>
<td>
<p>Coefficient of variation of distribution of population
<code>Funbound.plasma</code> values.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_clint.pop.cv">clint.pop.cv</code></td>
<td>
<p>Coefficient of variation of distribution of population
<code>Clint</code> values.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_caco2.meas.sd">caco2.meas.sd</code></td>
<td>
<p>Standard deviation of the measured oral absorption - numeric value (Default 0.3).</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_caco2.pop.sd">caco2.pop.sd</code></td>
<td>
<p>Standard deviation of the population level oral absorption - numeric value (Default 0.3).</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_caco2.fgut">Caco2.Fgut</code></td>
<td>
<p>= TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_caco2.fabs">Caco2.Fabs</code></td>
<td>
<p>= TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_keepit100">keepit100</code></td>
<td>
<p>= TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_poormetab">poormetab</code></td>
<td>
<p>Logical. Whether to include poor metabolizers in the Clint
distribution or not.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_fup.lod">fup.lod</code></td>
<td>
<p>The average limit of detection for <code>Funbound.plasma</code>, below
which distribution will be censored if fup.censored.dist is TRUE. Default 0.01.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_fup.censored.dist">fup.censored.dist</code></td>
<td>
<p>Logical. Whether to draw <code>Funbound.plasma</code> from a
censored distribution or not.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses the
Pearce et al. (2017) lipid binding adjustment
for Funbound.plasma when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_adjusted.clint">adjusted.Clint</code></td>
<td>
<p>Uses Kilford et al. (2008)
hepatocyte incubation
binding adjustment for Clint when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_clint.pvalue.threshold">clint.pvalue.threshold</code></td>
<td>
<p>Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
<tr><td><code id="invitro_mc_+3A_parameters">parameters</code></td>
<td>
<p>A list of chemical-specific model parameters containing at
least Funbound.plasma, Clint, and Fhep.assay.correction.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Monte Carlo methods used here were recently updated and described by
Breen et al. (2022).
</p>


<h3>Value</h3>

<p>A data.table with three columns: <code>Funbound.plasma</code> and
<code>Clint</code>, containing the sampled values, and
<code>Fhep.assay.correction</code>, containing the value for fraction unbound in
hepatocyte assay.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring and John Wambaugh
</p>


<h3>References</h3>

<p>Breen M, Wambaugh JF, Bernstein A, Sfeir M, Ring CL (2022).
&ldquo;Simulating toxicokinetic variability to identify susceptible and highly exposed populations.&rdquo;
<em>Journal of Exposure Science &amp; Environmental Epidemiology</em>, <b>32</b>(6), 855&ndash;863.<br /><br /> Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197.<br /><br /> Pearce RG, Setzer RW, Davis JL, Wambaugh JF (2017).
&ldquo;Evaluation and calibration of high-throughput predictions of chemical distribution to tissues.&rdquo;
<em>Journal of pharmacokinetics and pharmacodynamics</em>, <b>44</b>, 549&ndash;565.<br /><br /> Wambaugh JF, Wetmore BA, Ring CL, Nicolas CI, Pearce RG, Honda GS, Dinallo R, Angus D, Gilbert J, Sierra T, others (2019).
&ldquo;Assessing toxicokinetic uncertainty and variability in risk prioritization.&rdquo;
<em>Toxicological Sciences</em>, <b>172</b>(2), 235&ndash;251.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Simply generate a virtual population of 100 individuals,
#using the direct-resampling method
set.seed(42)

# Pull mean vchemical=specific values:
chem.props &lt;- parameterize_pbtk(chem.name="bisphenolaf")

# Convert to data.table with one row per sample:
parameters.dt &lt;- monte_carlo(chem.props,samples=100)

# Use httk-pop to generate a population:
pop &lt;- httkpop_generate(method='direct resampling', nsamp=100)

# Overwrite parameters specified by httk-pop:
parameters.dt[,names(pop):=pop]
# Vary in vitro parameters:
parameters.dt &lt;- invitro_mc(parameters.dt,samples=100)


</code></pre>

<hr>
<h2 id='is_in_inclusive'>Checks whether a value, or all values in a vector, is within inclusive
limits</h2><span id='topic+is_in_inclusive'></span>

<h3>Description</h3>

<p>Checks whether a value, or all values in a vector, is within inclusive
limits
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_in_inclusive(x, lims)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_in_inclusive_+3A_x">x</code></td>
<td>
<p>A numeric value, or vector of values.</p>
</td></tr>
<tr><td><code id="is_in_inclusive_+3A_lims">lims</code></td>
<td>
<p>A two-element vector of (min, max) values for the inclusive
limits. If <code>x</code> is a vector, <code>lims</code> may also be a two-column matrix
with <code>nrow=length(x)</code> where the first column is lower limits and the
second column is upper limits. If <code>x</code> is a vector and <code>lims</code> is a
two-element vector, then each element of <code>x</code> will be checked against
the same limits. If <code>x</code> is a vector and <code>lims</code> is a matrix, then
each element of <code>x</code> will be checked against the limits given by the
corresponding row of <code>lims</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector the same length as <code>x</code>, indicating whether
each element of <code>x</code> is within the inclusive limits given by
<code>lims</code>.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='is.httk'>Convenience Boolean (yes/no) function to identify chemical membership and
treatment within the httk project.</h2><span id='topic+is.httk'></span>

<h3>Description</h3>

<p>Allows easy identification of whether or not a chemical CAS is included in
various aspects of the httk research project (by model type and species of
interest). While it is our intent to keep these lists up-to-date, the 
information here is only for convenience and should not be considered
definitive.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.httk(chem.cas, species = "Human", model = "3compartmentss")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.httk_+3A_chem.cas">chem.cas</code></td>
<td>
<p>The Chemical Abstracts Service Resgistry Number (CAS-RN)
corresponding to the chemical of interest.</p>
</td></tr>
<tr><td><code id="is.httk_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="is.httk_+3A_model">model</code></td>
<td>
<p>Model used in calculation, 'pbtk' for the multiple compartment
model, '1compartment' for the one compartment model, '3compartment' for
three compartment model, '3compartmentss' for the three compartment model
without partition coefficients, or 'schmitt' for chemicals with logP and
fraction unbound (used in predict_partitioning_schmitt).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Tox21: Toxicology in the 21st Century (Tox21) is a U.S. federal High
Throughput Screening (HTS) collaboration among EPA, NIH, including National
Center for Advancing Translational Sciences and the National Toxicology
Program at the National Institute of Environmental Health Sciences, and the
Food and Drug Administration.  (Bucher et al., 2008)
</p>
<p>ToxCast: The Toxicity Forecaster (ToxCast) is a HTS screening project led by
the U.S. EPA to perform additional testing of a subset of Tox21 chemicals.
(Judson et al. 2010)
</p>
<p>ExpoCast: ExpoCast (Exposure Forecaster) is an U.S. EPA research project to
generate tenetative exposure estimates (e.g., mg/kg BW/day) for thousands of
chemicals that have little other information using models and informatics.
(Wambaugh et al. 2014)
</p>
<p>NHANES: The U.S. Centers for Disease Control (CDC) National Health and
Nutrition Examination Survery (NHANES) is an on-going survey to characterize
the health and biometrics (e.g., weight, height) of the U.S. population. One
set of measurments includes the quantification of xenobiotic chemicals in
various samples (blood, serum, urine) of the thousands of surveyed
individuals. (CDC, 2014)
</p>


<h3>Value</h3>

<table>
<tr><td><code>logical</code></td>
<td>
<p>A Boolean (1/0) value that is TRUE if the chemical
is included in the httk project with a given modeling scheme (PBTK) and 
a given species</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Bucher, J. R. (2008). Guest Editorial: NTP: New Initiatives, New
Alignment. Environ Health Perspect 116(1).
</p>
<p>Judson, R. S., Houck, K. A., Kavlock, R. J., Knudsen, T. B., Martin, M. T.,
Mortensen, H. M., Reif, D. M., Rotroff, D. M., Shah, I., Richard, A. M. and
Dix, D. J. (2010). In Vitro Screening of Environmental Chemicals for
Targeted Testing Prioritization: The ToxCast Project. Environmental Health
Perspectives 118(4), 485-492.
</p>
<p>Wambaugh, J. F., Wang, A., Dionisio, K. L., Frame, A., Egeghy, P., Judson,
R. and Setzer, R. W. (2014). High Throughput Heuristics for Prioritizing
Human Exposure to Environmental Chemicals. Environmental Science &amp;
Technology, 10.1021/es503583j.
</p>
<p>CDC (2014). National Health and Nutrition Examination Survey. Available at:
https://www.cdc.gov/nchs/nhanes.htm.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+in.list">in.list</a></code> for determining chemical membership in 
several other key lists
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

httk.table &lt;- get_cheminfo(info=c("CAS","Compound"))
httk.table[,"Rat"] &lt;- ""
httk.table[,"NHANES"] &lt;- ""
httk.table[,"Tox21"] &lt;- ""
httk.table[,"ToxCast"] &lt;- ""
httk.table[,"ExpoCast"] &lt;- ""
httk.table[,"PBTK"] &lt;- ""
# To make this example run quickly, this loop is only over the first five 
# chemicals. To build a table with all available chemicals use:
# for (this.cas in httk.table$CAS)
for (this.cas in httk.table$CAS[1:5])
{
  this.index &lt;- httk.table$CAS==this.cas
  if (is.nhanes(this.cas)) httk.table[this.index,"NHANES"] &lt;- "Y"
  if (is.tox21(this.cas)) httk.table[this.index,"Tox21"] &lt;- "Y"
  if (is.toxcast(this.cas)) httk.table[this.index,"ToxCast"] &lt;- "Y"
  if (is.expocast(this.cas)) httk.table[this.index,"ExpoCast"] &lt;- "Y"
  if (is.httk(this.cas,model="PBTK")) httk.table[this.index,"PBTK"] &lt;- "Y"
  if (is.httk(this.cas,species="rat")) httk.table[this.index,"Rat"] &lt;- "Y"
}


</code></pre>

<hr>
<h2 id='johnson'>Johnson 2006</h2><span id='topic+johnson'></span>

<h3>Description</h3>

<p>This data set is only used in Vignette 5.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>johnson
</code></pre>


<h3>Format</h3>

<p>A data.table containing 60 rows and 11 columns.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Johnson, Trevor N., Amin Rostami-Hodjegan, and Geoffrey T. Tucker.
&quot;Prediction of the clearance of eleven drugs and associated variability in
neonates, infants and children.&quot; Clinical pharmacokinetics 45.9 (2006):
931-956.
</p>

<hr>
<h2 id='kapraun2019'>Kapraun et al. 2019 data</h2><span id='topic+kapraun2019'></span><span id='topic+Kapraun2019'></span>

<h3>Description</h3>

<p>A list object containing time-varying parameters for the human maternal-fetal
HTTK model. List elements contain scalar coefficients for the polynomial,
logistic, Gompertz, and other functions of time describing blood flow rates,
tissue volumes, hematocrits, and other anatomical/physiological quantities
that change in the human mother and her fetus during pregnancy and gestation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kapraun2019
</code></pre>


<h3>Format</h3>

<p>list
</p>


<h3>Author(s)</h3>

<p>Dustin F. Kapraun
</p>


<h3>Source</h3>

<p>Kapraun DF, Sfeir M, Pearce RG, Davidson-Fritz SE, Lumen A, Dallmann A, Judson RS, Wambaugh JF (2022).
&ldquo;Evaluation of a rapid, generic human gestational dose model.&rdquo;
<em>Reproductive Toxicology</em>, <b>113</b>, 172&ndash;188.
</p>


<h3>References</h3>

<p>Kapraun DF, Wambaugh JF, Setzer RW, Judson RS (2019).
&ldquo;Empirical models for anatomical and physiological changes in a human mother and fetus during pregnancy and gestation.&rdquo;
<em>PLOS ONE</em>, <b>14</b>(5), 1-56.
<a href="https://doi.org/10.1371/journal.pone.0215906">doi:10.1371/journal.pone.0215906</a>.
</p>

<hr>
<h2 id='kidney_mass_children'>Predict kidney mass for children</h2><span id='topic+kidney_mass_children'></span>

<h3>Description</h3>

<p>For individuals under age 18, predict kidney mass from weight, height, and
gender. using equations from Ogiu et al. 1997
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kidney_mass_children(weight, height, gender)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kidney_mass_children_+3A_weight">weight</code></td>
<td>
<p>Vector of weights in kg.</p>
</td></tr>
<tr><td><code id="kidney_mass_children_+3A_height">height</code></td>
<td>
<p>Vector of heights in cm.</p>
</td></tr>
<tr><td><code id="kidney_mass_children_+3A_gender">gender</code></td>
<td>
<p>Vector of genders (either 'Male' or 'Female').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of kidney masses in kg.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ogiu, Nobuko, et al. &quot;A statistical analysis of the internal 
organ weights of normal Japanese people.&quot; Health physics 72.3 (1997): 368-383.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='liver_mass_children'>Predict liver mass for children</h2><span id='topic+liver_mass_children'></span>

<h3>Description</h3>

<p>For individuals under 18, predict the liver mass from height, weight, and
gender, using equations from Ogiu et al. 1997
</p>


<h3>Usage</h3>

<pre><code class='language-R'>liver_mass_children(height, weight, gender)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="liver_mass_children_+3A_height">height</code></td>
<td>
<p>Vector of heights in cm.</p>
</td></tr>
<tr><td><code id="liver_mass_children_+3A_weight">weight</code></td>
<td>
<p>Vector of weights in kg.</p>
</td></tr>
<tr><td><code id="liver_mass_children_+3A_gender">gender</code></td>
<td>
<p>Vector of genders (either 'Male' or 'Female').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of liver masses in kg.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ogiu, Nobuko, et al. &quot;A statistical analysis of the internal 
organ weights of normal Japanese people.&quot; Health physics 72.3 (1997): 368-383.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='load_dawson2021'>Load CLint and Fup QSPR predictions from Dawson et al. 2021.</h2><span id='topic+load_dawson2021'></span>

<h3>Description</h3>

<p>This function returns an updated version of 
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>
that includes Clint and Fup predictions from the Random Forest quantitative 
structure-property relationship (QSPR) models developed and
presented in Dawson et al. 2021, included in table <code><a href="#topic+dawson2021">dawson2021</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_dawson2021(overwrite = FALSE, exclude_oad = TRUE, target.env = .GlobalEnv)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_dawson2021_+3A_overwrite">overwrite</code></td>
<td>
<p>Only matters if load.image=FALSE. If overwrite=TRUE then
existing data in chem.physical_and_invitro.data will be replaced by any
predictions in Dawson et al. (2021) that is for the same chemical and
property. If overwrite=FALSE (DEFAULT) then new data for the same chemical
and property are ignored.  Funbound.plasma values of 0 (below limit of
detection) are overwritten either way.</p>
</td></tr>
<tr><td><code id="load_dawson2021_+3A_exclude_oad">exclude_oad</code></td>
<td>
<p>Include the chemicals only within the applicability domain.
If exclude_oad=TRUE (DEFAULT) chemicals outside the applicability domain do not
have their predicted values loaded.</p>
</td></tr>
<tr><td><code id="load_dawson2021_+3A_target.env">target.env</code></td>
<td>
<p>The environment where the new
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> is loaded. Defaults to global environment.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Because Clint and Fup are the only measurements required for many HTTK models,
changing the number of chemicals for which a value is available will change 
the number of chemicals which are listed with the <code><a href="#topic+get_cheminfo">get_cheminfo</a></code>
command. Use the command <code><a href="#topic+reset_httk">reset_httk</a></code> to return to the initial 
(measured only) <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> (for all
parameters).
</p>


<h3>Value</h3>

<table>
<tr><td><code>data.frame</code></td>
<td>
<p>An updated version of
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sarah E. Davidson
</p>


<h3>References</h3>

<p>Dawson DE, Ingle BL, Phillips KA, Nichols JW, Wambaugh JF, Tornero-Velez R (2021).
&ldquo;Designing QSARs for Parameters of High-Throughput Toxicokinetic Models Using Open-Source Descriptors.&rdquo;
<em>Environmental Science &amp; Technology</em>, <b>55</b>(9), 6505-6517.
<a href="https://doi.org/10.1021/acs.est.0c06117">doi:10.1021/acs.est.0c06117</a>, PMID: 33856768, https://doi.org/10.1021/acs.est.0c06117.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reset_httk">reset_httk</a></code>
</p>
<p><code><a href="#topic+get_cheminfo">get_cheminfo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# Count how many chemicals for which HTTK is available without the QSPR:
num.chems &lt;- length(get_cheminfo())
print(num.chems)

# For chemicals with Dawson et al. (2021) Clint and Fup QSPR predictions, 
# add them to our chemical information wherever measured values are 
# unavailable:
load_dawson2021()
# For chemicals with Dawson et al. (2021) QSPR predictions, add them to
# our chemical information -- overwriting measured values where we had them:
load_dawson2021(overwrite=TRUE)

# Let's see how many chemicals we have now with the Dawson et al. (2021) 
# predictions loaded:
length(get_cheminfo()) 

# Now let us reset the chemical data to the initial version:
reset_httk()

# We should be back to our original number:
num.chems == length(get_cheminfo())

## End(Not run)                        

</code></pre>

<hr>
<h2 id='load_honda2023'>Load Caco2 QSPR predictions from Honda et al. 2023</h2><span id='topic+load_honda2023'></span>

<h3>Description</h3>

<p>This function returns an updated version of 
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>
that includes Caco2 Pab predictions from the Random Forest quantitative 
structure-property relationship (QSPR) models developed and
presented in Honda et al. 2023, included in table <code><a href="#topic+honda2023.qspr">honda2023.qspr</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_honda2023(overwrite = FALSE, exclude_oad = TRUE, target.env = .GlobalEnv)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_honda2023_+3A_overwrite">overwrite</code></td>
<td>
<p>Only matters if load.image=FALSE. If overwrite=TRUE then
existing data in chem.physical_and_invitro.data will be replaced by any
prediction in Honda et al. (2023) that is for the same chemical and
property. If overwrite=FALSE (DEFAULT) then new data for the same chemical
and property are ignored.</p>
</td></tr>
<tr><td><code id="load_honda2023_+3A_exclude_oad">exclude_oad</code></td>
<td>
<p>Include the chemicals only within the applicability domain.
If exclude_oad=TRUE (DEFAULT) chemicals outside the applicability domain do not
have their predicted values loaded.</p>
</td></tr>
<tr><td><code id="load_honda2023_+3A_target.env">target.env</code></td>
<td>
<p>The environment where the new
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> is loaded. Defaults to global environment.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that because Pab is not required for most HTTK models, changing
the number of chemicals for which a value is available will not change the
number of chemicals which are listed with the <code><a href="#topic+get_cheminfo">get_cheminfo</a></code>
command. Use the command <code><a href="#topic+reset_httk">reset_httk</a></code> to return to the initial 
(measured only) <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> (for all
parameters).
</p>


<h3>Value</h3>

<table>
<tr><td><code>data.frame</code></td>
<td>
<p>An updated version of
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reset_httk">reset_httk</a></code>
</p>
<p><code><a href="#topic+get_cheminfo">get_cheminfo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# For chemicals with Honda et al. (2023) Caco2 Pab QSPR predictions, 
# add them to our chemical information wherever measured values are 
# unavailable:
load_honda2023()

# Or, for chemicals with Honda et al. (2023) QSPR predictions, add them to
# our chemical information but overwrite measured values where we had them:
load_honda2023(overwrite=TRUE) 

# Now let us reset the chemical data to the initial version:
reset_httk()

## End(Not run)                        

</code></pre>

<hr>
<h2 id='load_pradeep2020'>Load CLint and Fup QSPR predictions predictions from Pradeep et al. 2020.</h2><span id='topic+load_pradeep2020'></span>

<h3>Description</h3>

<p>This function returns an updated version of 
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>
that includes quantitative 
structure-property relationship (QSPR) predictions from
Support Vector Machine and Random Forest 
models developed and presented in Pradeep et al. 2020, included in
<code><a href="#topic+pradeep2020">pradeep2020</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_pradeep2020(overwrite = FALSE, target.env = .GlobalEnv)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_pradeep2020_+3A_overwrite">overwrite</code></td>
<td>
<p>Only matters if load.image=FALSE. If overwrite=TRUE then
existing data in chem.physical_and_invitro.data will be replaced by any
predictions in Pradeep et al. (2020) that is for the same chemical and
property. If overwrite=FALSE (DEFAULT) then new data for the same chemical
and property are ignored.  Funbound.plasma values of 0 (below limit of
detection) are overwritten either way.</p>
</td></tr>
<tr><td><code id="load_pradeep2020_+3A_target.env">target.env</code></td>
<td>
<p>The environment where the new
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> is loaded. Defaults to global environment.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Because Clint and Fup are the only measurements required for many HTTK models,
changing the number of chemicals for which a value is available will change 
the number of chemicals which are listed with the <code><a href="#topic+get_cheminfo">get_cheminfo</a></code>
command. Use the command <code><a href="#topic+reset_httk">reset_httk</a></code> to return to the initial 
(measured only) <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> (for all
parameters).
</p>


<h3>Value</h3>

<table>
<tr><td><code>data.frame</code></td>
<td>
<p>An updated version of
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sarah E. Davidson
</p>


<h3>References</h3>

<p>Pradeep P, Patlewicz G, Pearce R, Wambaugh J, Wetmore B, Judson R (2020).
&ldquo;Using chemical structure information to develop predictive models for in vitro toxicokinetic parameters to inform high-throughput risk-assessment.&rdquo;
<em>Computational Toxicology</em>, <b>16</b>, 100136.
ISSN 2468-1113, <a href="https://doi.org/10.1016/j.comtox.2020.100136">doi:10.1016/j.comtox.2020.100136</a>, <a href="https://www.sciencedirect.com/science/article/pii/S2468111320300463">https://www.sciencedirect.com/science/article/pii/S2468111320300463</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reset_httk">reset_httk</a></code>
</p>
<p><code><a href="#topic+get_cheminfo">get_cheminfo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# Count how many chemicals for which HTTK is available without the QSPR:
num.chems &lt;- length(get_cheminfo())
print(num.chems)

# For chemicals with Pradeep et al. (2020) Clint and Fup QSPR predictions, 
# add them to our chemical information wherever measured values are 
# unavailable:
load_pradeep2020()

# Or, for chemicals with Pradeep et al. (2020) QSPR predictions, add them to
# our chemical information but overwrite measured values where we had them:
load_pradeep2020(overwrite=TRUE) 

# Let's see how many chemicals we have now with the Pradeep et al. (2020)
# predictions data loaded:
length(get_cheminfo())

# Now let us reset the chemical data to the initial version:
reset_httk()

# We should be back to our original number:
num.chems == length(get_cheminfo())

## End(Not run)                        

</code></pre>

<hr>
<h2 id='load_sipes2017'>Load CLint and Fup QSPR predictions from Sipes et al 2017.</h2><span id='topic+load_sipes2017'></span>

<h3>Description</h3>

<p>This function returns an updated version of 
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>
that includes quantitative 
structure-property relationship (QSPR) predictions from Simulations Plus' 
ADMET predictor as 
used in Sipes et al. 2017, included in <code><a href="#topic+sipes2017">sipes2017</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_sipes2017(overwrite = FALSE, target.env = .GlobalEnv)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_sipes2017_+3A_overwrite">overwrite</code></td>
<td>
<p>Only matters if load.image=FALSE. If overwrite=TRUE then
existing data in chem.physical_and_invitro.data will be replaced by any
predictions in Sipes et al. (2017) that is for the same chemical and
property. If overwrite=FALSE (DEFAULT) then new data for the same chemical
and property are ignored. Funbound.plasma values of 0 (below limit of
detection) are overwritten either way.</p>
</td></tr>
<tr><td><code id="load_sipes2017_+3A_target.env">target.env</code></td>
<td>
<p>The environment where the new
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> is loaded. Defaults to global environment.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Because Clint and Fup are the only measurements required for many HTTK models,
changing the number of chemicals for which a value is available will change 
the number of chemicals which are listed with the <code><a href="#topic+get_cheminfo">get_cheminfo</a></code>
command. Use the command <code><a href="#topic+reset_httk">reset_httk</a></code> to return to the initial 
(measured only) <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> (for all
parameters).
</p>


<h3>Value</h3>

<table>
<tr><td><code>data.frame</code></td>
<td>
<p>An updated version of
<code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Pearce and John Wambaugh
</p>


<h3>References</h3>

<p>Sipes, Nisha S., et al. &quot;An intuitive approach for predicting
potential human health risk with the Tox21 10k library.&quot; Environmental
Science &amp; Technology 51.18 (2017): 10786-10796.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reset_httk">reset_httk</a></code>
</p>
<p><code><a href="#topic+get_cheminfo">get_cheminfo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Count how many chemicals for which HTTK is available without the QSPR:
num.chems &lt;- length(get_cheminfo())
print(num.chems)

# For chemicals with Sipes et al. (2017) Clint and Fup QSPR predictions, 
# add them to our chemical information wherever measured values are 
# unavailable:
load_sipes2017()

# Here's a chemical we didn't have before (this one is a good test since the 
# logP is nearly 10 and it probably wouldn't work in vitro):
calc_css(chem.cas="26040-51-7")

# Let's see how many chemicals we have now with the Sipes et al. (2017) 
# predictions data loaded:
length(get_cheminfo())

# Now let us reset the chemical data to the initial version:
reset_httk()

# We should be back to our original number:
num.chems == length(get_cheminfo())
                        

</code></pre>

<hr>
<h2 id='lump_tissues'>Lump tissue parameters into model compartments</h2><span id='topic+lump_tissues'></span>

<h3>Description</h3>

<p>This function takes the tissue:plasma partition coefficients from 
<code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code> along with the tissue-specific
volumes and flows and aggregates (or &quot;lumps&quot;) them
according to the needed scheme of toxicokinetic model tissue comparments.
</p>
<p><code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code> makes tissue-specific predictions
drawing from those tissues described in <code><a href="#topic+tissue.data">tissue.data</a></code>. Since
different physiologically-based toxicokinetic (PBTK) models use diffeent 
schemes for rganizing the
tissues of the body into differing compartments (for example, &quot;rapidly
perfused tissues&quot;), this function lumps tissues into
compartments as specified by the argument 'tissuelist'. Aggregate flows, 
volumes, and partition coefficients are provided for the
lumped tissue compartments. Flows and volumes are summed while
partition coefficients is calculated using averaging weighted by 
species-specific tissue volumes.
</p>
<p>The name of each entry in 'tissuelist' is its own compartment. The
modelinfo_MODEL.R file corresponding to the model specified by argument
'model' includes both a 'tissuelist' describing to the model's 
compartmentallumping schme as well as a vector of 'tissuenames' specifying 
all tissues to be lumped into those compartments.
</p>
<p>Alternatively the 'tissuelist' and 'tissuenames' can also be manually
specified for alternate lumping schemes not necessarily related to a
pre-made httk model. For example,
tissuelist&lt;-list(Rapid=c(&quot;Brain&quot;,&quot;Kidney&quot;)).
</p>
<p>The tissues contained in 'tissuenames' that are unused in 'tissuelist' 
are aggregated into a single compartment termed
&quot;rest&quot;.
</p>
<p>NOTE: The partition coefficients of lumped compartments vary according to
individual and species differences since the volumes of the consitutent 
tissues may vary.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lump_tissues(
  Ktissue2pu.in,
  parameters = NULL,
  tissuelist = NULL,
  species = "Human",
  tissue.vols = NULL,
  tissue.flows = NULL,
  tissuenames = NULL,
  model = "pbtk",
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lump_tissues_+3A_ktissue2pu.in">Ktissue2pu.in</code></td>
<td>
<p>List of partition coefficients from
<code>predict_partitioning_schmitt</code>. The tissues named in this list are 
lumped into the compartments specified by <code>tissuelist</code> unless they are
not present the specified <code>model</code>'s associated <code>alltissues</code>.</p>
</td></tr>
<tr><td><code id="lump_tissues_+3A_parameters">parameters</code></td>
<td>
<p>A list of physiological parameters including flows and
volumes for tissues named in <code>Ktissue2pu.in</code></p>
</td></tr>
<tr><td><code id="lump_tissues_+3A_tissuelist">tissuelist</code></td>
<td>
<p>Manually specifies compartment names and tissues, which
override the standard compartment names and tissues that are usually
specified in a model's associated modelinfo file. Remaining tissues in the
model's associated <code>alltissues</code> listing are lumped in the rest of the body.</p>
</td></tr>
<tr><td><code id="lump_tissues_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="lump_tissues_+3A_tissue.vols">tissue.vols</code></td>
<td>
<p>A list of volumes for tissues in <code>tissuelist</code>.</p>
</td></tr>
<tr><td><code id="lump_tissues_+3A_tissue.flows">tissue.flows</code></td>
<td>
<p>A list of flows for tissues in <code>tissuelist</code>.</p>
</td></tr>
<tr><td><code id="lump_tissues_+3A_tissuenames">tissuenames</code></td>
<td>
<p>A list of tissue names in <code>tissuenames</code>.</p>
</td></tr>
<tr><td><code id="lump_tissues_+3A_model">model</code></td>
<td>
<p>Specify which model (and therefore which tissues) are being 
considered.</p>
</td></tr>
<tr><td><code id="lump_tissues_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Krbc2pu</code></td>
<td>
<p>Ratio of concentration of chemical in red blood cells
to unbound concentration in plasma.</p>
</td></tr> <tr><td><code>Krest2pu</code></td>
<td>
<p>Ratio of concentration
of chemical in rest of body tissue to unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>Vrestc</code></td>
<td>
<p> Volume of the rest of the body per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vliverc</code></td>
<td>
<p> Volume of the liver per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Qtotal.liverf</code></td>
<td>
<p>Fraction of cardiac output flowing to the gut and
liver, i.e. out of the liver.</p>
</td></tr> <tr><td><code>Qgutf</code></td>
<td>
<p>Fraction of cardiac output
flowing to the gut.</p>
</td></tr> <tr><td><code>Qkidneyf</code></td>
<td>
<p>Fraction of cardiac output flowing to
the kidneys.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>References</h3>

<p>Pearce, Robert G., et al. &quot;Evaluation and calibration of 
high-throughput predictions of chemical distribution to tissues.&quot; Journal of
pharmacokinetics and pharmacodynamics 44.6 (2017): 549-565.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>
</p>
<p><code><a href="#topic+tissue.data">tissue.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 pcs &lt;- predict_partitioning_schmitt(chem.name='bisphenola')
 tissuelist &lt;- list(
   liver=c("liver"),
   rapid=c("lung","kidney","muscle","brain"),
   fat=c("adipose"),
   slow=c('bone'))
 lump_tissues(pcs,tissuelist=tissuelist)

</code></pre>

<hr>
<h2 id='lung_mass_children'>Predict lung mass for children</h2><span id='topic+lung_mass_children'></span>

<h3>Description</h3>

<p>For individuals under 18, predict the liver mass from height, weight, and
gender, using equations from Ogiu et al. 1997
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lung_mass_children(height, weight, gender)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lung_mass_children_+3A_height">height</code></td>
<td>
<p>Vector of heights in cm.</p>
</td></tr>
<tr><td><code id="lung_mass_children_+3A_weight">weight</code></td>
<td>
<p>Vector of weights in kg.</p>
</td></tr>
<tr><td><code id="lung_mass_children_+3A_gender">gender</code></td>
<td>
<p>Vector of genders (either 'Male' or 'Female').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of lung masses in kg.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ogiu, Nobuko, et al. &quot;A statistical analysis of the internal 
organ weights of normal Japanese people.&quot; Health physics 72.3 (1997): 368-383.
</p>
<p>Price, Paul S., et al. &quot;Modeling interindividual variation in physiological 
factors used in PBPK models of humans.&quot; Critical reviews in toxicology 33.5 
(2003): 469-503.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='mcnally_dt'>Reference tissue masses and flows from tables in McNally et al. 2014.</h2><span id='topic+mcnally_dt'></span>

<h3>Description</h3>

<p>Reference tissue masses, flows, and residual variance distributions from
Tables 1, 4, and 5 of McNally et al. 2014.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcnally_dt
</code></pre>


<h3>Format</h3>

<p>A data.table with variables: </p>
<dl>
<dt><code>tissue</code></dt><dd><p>Body
tissue</p>
</dd> <dt><code>gender</code></dt><dd><p>Gender: Male or Female</p>
</dd>
<dt><code>mass_ref</code></dt><dd><p>Reference mass in kg, from Reference Man</p>
</dd>
<dt><code>mass_cv</code></dt><dd><p>Coefficient of variation for mass</p>
</dd>
<dt><code>mass_dist</code></dt><dd><p>Distribution for mass: Normal or Log-normal</p>
</dd>
<dt><code>flow_ref</code></dt><dd><p>Reference flow in L/h, from Reference Man</p>
</dd>
<dt><code>flow_cv</code></dt><dd><p>Coefficient of variation for flow (all normally
distributed)</p>
</dd> <dt><code>height_ref</code></dt><dd><p>Reference heights (by gender)</p>
</dd>
<dt><code>CO_ref</code></dt><dd><p>Reference cardiac output by gender</p>
</dd>
<dt><code>flow_frac</code></dt><dd><p>Fraction of CO flowing to each tissue:
<code>flow_ref</code>/<code>CO_ref</code></p>
</dd></dl>



<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>Source</h3>

<p>McNally K, Cotton R, Hogg A, Loizou G. &quot;PopGen: A virtual human
population generator.&quot; Toxicology 315, 70-85, 2004.
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='mecdt'>Pre-processed NHANES data.</h2><span id='topic+mecdt'></span>

<h3>Description</h3>

<p>NHANES data on demographics, anthropometrics, and some laboratory measures,
cleaned and combined into a single data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mecdt
</code></pre>


<h3>Format</h3>

<p>A data.table with 23620 rows and 12
variables.  </p>
 <dl>
<dt>seqn</dt><dd><p>NHANES unique identifier for individual
respondents.</p>
</dd> <dt>sddsrvyr</dt><dd><p>NHANES two-year cycle: one of &quot;NHANES
2013-2014&quot;, &quot;NHANES 2015-2016&quot;, &quot;NHANES 2017-2018&quot;.</p>
</dd> <dt>riagendr</dt><dd><p>Gender:
&quot;Male&quot; or &quot;Female&quot;</p>
</dd> <dt>ridreth1</dt><dd><p>Race/ethnicity category: one of &quot;Mexican
American&quot;, &quot;Non-Hispanic White&quot;, &quot;Non-Hispanic Black&quot;, &quot;Other&quot;, &quot;Other
Hispanic&quot;.</p>
</dd> <dt>ridexagm</dt><dd><p>Age in months at the time of examination (if not
recorded by NHANES, it was imputed based on age at the time of screening)</p>
</dd>
<dt>ridexagy</dt><dd><p>Age in years at the time of examination (if not recorded by
NHANES, it was imputed based on age at the time of screening)</p>
</dd>
<dt>bmxwt</dt><dd><p>Weight in kg</p>
</dd> <dt>lbxscr</dt><dd><p>Serum creatinine, mg/dL</p>
</dd>
<dt>lbxhct</dt><dd><p>Hematocrit, percent by volume of blood composed of red blood
cells</p>
</dd> <dt><code>wtmec6yr</code></dt><dd><p>6-year sample weights for combining 3 cycles,
computed by dividing 2-year sample weights by 3.</p>
</dd>
<dt><code>bmxhtlenavg</code></dt><dd><p>Average of height and recumbent length if both
were measured; if only one was measured, takes value of the one that was
measured.</p>
</dd> <dt><code>weight_class</code></dt><dd><p>One of Underweight, Normal,
Overweight, or Obese. Assigned using methods in
<code><a href="#topic+get_weight_class">get_weight_class</a></code>.</p>
</dd> </dl>



<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>Source</h3>

<p><a href="https://wwwn.cdc.gov/nchs/nhanes/Default.aspx">https://wwwn.cdc.gov/nchs/nhanes/Default.aspx</a>
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='metabolism_data_Linakis2020'>Metabolism data involved in Linakis 2020 vignette analysis.</h2><span id='topic+metabolism_data_Linakis2020'></span>

<h3>Description</h3>

<p>Metabolism data involved in Linakis 2020 vignette analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metabolism_data_Linakis2020
</code></pre>


<h3>Format</h3>

<p>A data.frame containing x rows and y columns.
</p>


<h3>Author(s)</h3>

<p>Matt Linakis
</p>


<h3>Source</h3>

<p>Matt Linakis
</p>


<h3>References</h3>

<p>DSStox database (https:// www.epa.gov/ncct/dsstox
</p>

<hr>
<h2 id='monte_carlo'>Monte Carlo for toxicokinetic model parameters</h2><span id='topic+monte_carlo'></span>

<h3>Description</h3>

<p>This function performs basic, uncorrelated Monte Carlo to simulate
uncertainty and/or variability for parameters of 
toxicokinetic models. Parameters can be varied according to either a
normal distribution that is truncated at zero 
(using argument <code>cv.params</code>) or from a normal distribution that is 
censored for values less than the limit of detection 
(<code>censored.params</code>). Coefficient of variation (cv) and limit of
of detectin can be specified separately for each parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>monte_carlo(
  parameters,
  cv.params = NULL,
  censored.params = NULL,
  samples = 1000,
  suppress.messages = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="monte_carlo_+3A_parameters">parameters</code></td>
<td>
<p>These parameters that are also listed in either
cv.params or censored.params are sampled using Monte Carlo.</p>
</td></tr>
<tr><td><code id="monte_carlo_+3A_cv.params">cv.params</code></td>
<td>
<p>The parameters listed in cv.params are sampled from a
normal distribution that is truncated at zero. This argument should be a
list of coefficients of variation (cv) for the normal distribution. Each
entry in the list is named for a parameter in &quot;parameters&quot;. New values are
sampled with mean equal to the value in &quot;parameters&quot; and standard deviation
equal to the mean times the cv.</p>
</td></tr>
<tr><td><code id="monte_carlo_+3A_censored.params">censored.params</code></td>
<td>
<p>The parameters listed in censored.params are sampled
from a normal distribution that is censored for values less than the limit
of detection (specified separately for each parameter). This argument should
be a list of sub-lists. Each sublist is named for a parameter in &quot;params&quot;
and contains two elements: &quot;cv&quot; (coefficient of variation) and &quot;LOD&quot; (limit
of detection), below which parameter values are censored. New values are
sampled with mean equal to the value in &quot;params&quot; and standard deviation
equal to the mean times the cv. Censored values are sampled on a uniform
distribution between 0 and the limit of detection.</p>
</td></tr>
<tr><td><code id="monte_carlo_+3A_samples">samples</code></td>
<td>
<p>This argument is the number of samples to be generated for
calculating quantiles.</p>
</td></tr>
<tr><td><code id="monte_carlo_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table with a row for each individual in the sample and a column for
each parater in the model.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Pearce, Robert G., et al. &quot;Httk: R package for high-throughput 
toxicokinetics.&quot; Journal of statistical software 79.4 (2017): 1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Example based on Pearce et al. (2017):

# Set up means:
params &lt;- parameterize_pbtk(chem.name="zoxamide")
# Nothing changes:
monte_carlo(params)

vary.params &lt;- NULL
for (this.param in names(params)[!(names(params) %in%
  c("Funbound.plasma", "pKa_Donor", "pKa_Accept" )) &amp;
  !is.na(as.numeric(params))]) vary.params[this.param] &lt;- 0.2
# Most everything varies with CV of 0.2:
monte_carlo(
  parameters=params, 
  cv.params = vary.params)

censored.params &lt;- list(Funbound.plasma = list(cv = 0.2, lod = 0.01))
# Fup is censored below 0.01:
monte_carlo(
  parameters=params, 
  cv.params = vary.params,
  censored.params = censored.params)

</code></pre>

<hr>
<h2 id='Obach2008'>Published Pharmacokinetic Parameters from Obach et al. 2008</h2><span id='topic+Obach2008'></span>

<h3>Description</h3>

<p>This data set is used in Vignette 4 for steady state concentration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Obach2008
</code></pre>


<h3>Format</h3>

<p>A data.frame containing 670 rows and 8 columns.
</p>


<h3>References</h3>

<p>Obach, R. Scott, Franco Lombardo, and Nigel J. Waters. &quot;Trend
analysis of a database of intravenous pharmacokinetic parameters in humans
for 670 drug compounds.&quot; Drug Metabolism and Disposition 36.7 (2008):
1385-1405.
</p>

<hr>
<h2 id='onlyp'>NHANES Exposure Data</h2><span id='topic+onlyp'></span>

<h3>Description</h3>

<p>This data set is only used in Vignette 6.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>onlyp
</code></pre>


<h3>Format</h3>

<p>A data.table containing 1060 rows and 5 columns.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Wambaugh, John F., et al. &quot;High throughput heuristics for prioritizing human
exposure to environmental chemicals.&quot; Environmental science &amp; technology
48.21 (2014): 12760-12767.
</p>

<hr>
<h2 id='pancreas_mass_children'>Predict pancreas mass for children</h2><span id='topic+pancreas_mass_children'></span>

<h3>Description</h3>

<p>For individuals under 18, predict the pancreas mass from height, weight, and
gender, using equations from Ogiu et al.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pancreas_mass_children(height, weight, gender)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pancreas_mass_children_+3A_height">height</code></td>
<td>
<p>Vector of heights in cm.</p>
</td></tr>
<tr><td><code id="pancreas_mass_children_+3A_weight">weight</code></td>
<td>
<p>Vector of weights in kg.</p>
</td></tr>
<tr><td><code id="pancreas_mass_children_+3A_gender">gender</code></td>
<td>
<p>Vector of genders (either 'Male' or 'Female').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of pancreas masses in kg.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ogiu, Nobuko, et al. &quot;A statistical analysis of the internal 
organ weights of normal Japanese people.&quot; Health physics 72.3 (1997): 368-383.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='parameterize_1comp'>Parameters for a one compartment (empirical) toxicokinetic model</h2><span id='topic+parameterize_1comp'></span>

<h3>Description</h3>

<p>This function initializes the parameters needed in the function solve_1comp.
Volume of distribution is estimated by using a modified Schmitt (2008) method
to predict tissue particition coefficients (Pearce et al., 2017) and then
lumping the compartments weighted by tissue volume:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameterize_1comp(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  adjusted.Funbound.plasma = TRUE,
  adjusted.Clint = TRUE,
  regression = TRUE,
  restrictive.clearance = TRUE,
  well.stirred.correction = TRUE,
  suppress.messages = FALSE,
  clint.pvalue.threshold = 0.05,
  minimum.Funbound.plasma = 1e-04,
  Caco2.options = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameterize_1comp_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing rat values with human values if
true.</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses Pearce et al. (2017) lipid binding adjustment
for Funbound.plasma (which impacts volume of distribution) when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_adjusted.clint">adjusted.Clint</code></td>
<td>
<p>Uses Kilford et al. (2008) hepatocyte incubation
binding adjustment for Clint when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients in volume of distribution calculation.</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>In calculating elimination rate and hepatic
bioavailability, protein binding is not taken into account (set to 1) in
liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_well.stirred.correction">well.stirred.correction</code></td>
<td>
<p>Uses correction in calculation of hepatic
clearance for well-stirred model if TRUE.  This assumes clearance relative
to amount unbound in whole blood instead of plasma, but converted to use
with plasma concentration.</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not to suppress messages.</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_clint.pvalue.threshold">clint.pvalue.threshold</code></td>
<td>
<p>Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-value greater than the threshold are
set to zero.</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
<tr><td><code id="parameterize_1comp_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
</table>


<h3>Details</h3>



<p>V_d,steady-state = Sum over all tissues (K_i * V_i) + V_plasma

</p>
<p>where K_i is the tissue:unbound plasma concentration partition coefficient
for tissue i.
</p>


<h3>Value</h3>

<table>
<tr><td><code>Vdist</code></td>
<td>
<p>Volume of distribution, units of L/kg BW.</p>
</td></tr>
<tr><td><code>Fabsgut</code></td>
<td>
<p>Fraction of the oral dose absorbed and surviving gut metabolism, i.e. the 
fraction of the dose that enters the gutlumen.</p>
</td></tr> <tr><td><code>kelim</code></td>
<td>
<p>Elimination rate, units of
1/h.</p>
</td></tr> <tr><td><code>hematocrit</code></td>
<td>
<p>Percent volume of red blood cells in the blood.</p>
</td></tr>
<tr><td><code>Fabsgut</code></td>
<td>
<p>Fraction of the oral dose absorbed, i.e. the fraction of the
dose that enters the gutlumen.</p>
</td></tr> 
<tr><td><code>Fhep.assay.correction</code></td>
<td>
<p>The fraction of chemical unbound in hepatocyte 
assay using the method of Kilford et al. (2008)</p>
</td></tr> 
<tr><td><code>kelim</code></td>
<td>
<p>Elimination rate, units of 1/h.</p>
</td></tr> 
<tr><td><code>hematocrit</code></td>
<td>
<p>Percent volume of red blood cells in the blood.</p>
</td></tr>
<tr><td><code>kgutabs</code></td>
<td>
<p>Rate chemical is absorbed, 1/h.</p>
</td></tr>
<tr><td><code>million.cells.per.gliver</code></td>
<td>
<p>Millions cells per gram of liver tissue.</p>
</td></tr>
<tr><td><code>MW</code></td>
<td>
<p>Molecular Weight, g/mol.</p>
</td></tr> 
<tr><td><code>Rblood2plasma</code></td>
<td>
<p>The ratio of the concentration of the chemical in the 
blood to the concentration in the plasma. Not used in calculations but 
included for the conversion of plasma outputs.</p>
</td></tr> 
<tr><td><code>hepatic.bioavailability</code></td>
<td>
<p>Fraction of dose remaining after
first pass clearance, calculated from the corrected well-stirred model.</p>
</td></tr>
<tr><td><code>BW</code></td>
<td>
<p>Body Weight, kg.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>
<p>Schmitt W (2008).
&ldquo;General approach for the calculation of tissue to plasma partition coefficients.&rdquo;
<em>Toxicology in vitro</em>, <b>22</b>(2), 457&ndash;467.
</p>
<p>Pearce RG, Setzer RW, Davis JL, Wambaugh JF (2017).
&ldquo;Evaluation and calibration of high-throughput predictions of chemical distribution to tissues.&rdquo;
<em>Journal of pharmacokinetics and pharmacodynamics</em>, <b>44</b>, 549&ndash;565.
</p>
<p>Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+solve_1comp">solve_1comp</a></code>
</p>
<p><code><a href="#topic+calc_analytic_css_1comp">calc_analytic_css_1comp</a></code>
</p>
<p><code><a href="#topic+calc_vdist">calc_vdist</a></code>
</p>
<p><code><a href="#topic+parameterize_steadystate">parameterize_steadystate</a></code>
</p>
<p><code><a href="#topic+apply_clint_adjustment">apply_clint_adjustment</a></code>
</p>
<p><code><a href="#topic+tissue.data">tissue.data</a></code>
</p>
<p><code><a href="#topic+physiology.data">physiology.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 parameters &lt;- parameterize_1comp(chem.name='Bisphenol-A',species='Rat')
 parameters &lt;- parameterize_1comp(chem.cas='80-05-7',
                                  restrictive.clearance=FALSE,
                                  species='rabbit',
                                  default.to.human=TRUE)
 out &lt;- solve_1comp(parameters=parameters)

</code></pre>

<hr>
<h2 id='parameterize_3comp'>Parameters for a three-compartment toxicokinetic model (dynamic)</h2><span id='topic+parameterize_3comp'></span>

<h3>Description</h3>

<p>This function generates the chemical- and species-specific parameters needed 
for model '3compartment', for example <code><a href="#topic+solve_3comp">solve_3comp</a></code>. A call is 
masde to <code><a href="#topic+parameterize_pbtk">parameterize_pbtk</a></code> to use Schmitt (2008)'s method
as modified by Pearce et al. (2017) to predict partition coefficients based
on descriptions in <code><a href="#topic+tissue.data">tissue.data</a></code>. Organ volumes and flows are
retrieved from table <code><a href="#topic+physiology.data">physiology.data</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameterize_3comp(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  force.human.clint.fup = FALSE,
  clint.pvalue.threshold = 0.05,
  adjusted.Funbound.plasma = TRUE,
  adjusted.Clint = TRUE,
  regression = TRUE,
  suppress.messages = FALSE,
  restrictive.clearance = TRUE,
  minimum.Funbound.plasma = 1e-04,
  Caco2.options = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameterize_3comp_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (https://comptox.epa.gov/dashboard)
&ndash; the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true.</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_force.human.clint.fup">force.human.clint.fup</code></td>
<td>
<p>Forces use of human values for hepatic
intrinsic clearance and fraction of unbound plasma if true.</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_clint.pvalue.threshold">clint.pvalue.threshold</code></td>
<td>
<p>Hepatic clearances with clearance assays
having p-values greater than the threshold are set to zero.</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses Pearce et al. (2017) lipid binding adjustment
for Funbound.plasma (which impacts partition coefficients) when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_adjusted.clint">adjusted.Clint</code></td>
<td>
<p>Uses Kilford et al. (2008) hepatocyte incubation
binding adjustment for Clint when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>In calculating hepatic.bioavailability, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
<tr><td><code id="parameterize_3comp_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>BW</code></td>
<td>
<p>Body Weight, kg.</p>
</td></tr> 
<tr><td><code>Clmetabolismc</code></td>
<td>
<p>Hepatic Clearance, L/h/kg BW.</p>
</td></tr> 
<tr><td><code>Fabsgut</code></td>
<td>
<p>Fraction of the oral dose absorbed, i.e. the 
fraction of the dose that enters the gutlumen.</p>
</td></tr> 
<tr><td><code>Funbound.plasma</code></td>
<td>
<p>Fraction of plasma that is not bound.</p>
</td></tr> 
<tr><td><code>Fhep.assay.correction</code></td>
<td>
<p>The fraction of chemical unbound in hepatocyte 
assay using the method of Kilford et al. (2008)</p>
</td></tr> 
<tr><td><code>hematocrit</code></td>
<td>
<p>Percent volume of red blood cells in the blood.</p>
</td></tr>
<tr><td><code>Kgut2pu</code></td>
<td>
<p>Ratio of concentration of chemical in gut tissue to unbound
concentration in plasma.</p>
</td></tr> 
<tr><td><code>Kliver2pu</code></td>
<td>
<p>Ratio of concentration of
chemical in liver tissue to unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>Krbc2pu</code></td>
<td>
<p>Ratio of concentration of chemical in red blood cells to
unbound concentration in plasma.</p>
</td></tr> 
<tr><td><code>Krest2pu</code></td>
<td>
<p>Ratio of concentration of
chemical in rest of body tissue to unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>million.cells.per.gliver</code></td>
<td>
<p>Millions cells per gram of liver tissue.</p>
</td></tr>
<tr><td><code>MW</code></td>
<td>
<p>Molecular Weight, g/mol.</p>
</td></tr> 
<tr><td><code>Qcardiacc</code></td>
<td>
<p>Cardiac Output, L/h/kg
BW^3/4.</p>
</td></tr> <tr><td><code>Qgfrc</code></td>
<td>
<p>Glomerular Filtration Rate, L/h/kg BW^3/4, volume of
fluid filtered from kidney and excreted.</p>
</td></tr> 
<tr><td><code>Qgutf</code></td>
<td>
<p>Fraction of cardiac output flowing to the gut.</p>
</td></tr> 
<tr><td><code>Qliverf</code></td>
<td>
<p>Fraction of cardiac output flowing to the liver.</p>
</td></tr> 
<tr><td><code>Rblood2plasma</code></td>
<td>
<p>The ratio of the concentration
of the chemical in the blood to the concentration in the plasma.</p>
</td></tr>
<tr><td><code>Vgutc</code></td>
<td>
<p>Volume of the gut per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vliverc</code></td>
<td>
<p>Volume of the liver per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vrestc</code></td>
<td>
<p> Volume of the rest of the body per kg body weight, L/kg BW.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Pearce and John Wambaugh
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>
<p>Schmitt W (2008).
&ldquo;General approach for the calculation of tissue to plasma partition coefficients.&rdquo;
<em>Toxicology in vitro</em>, <b>22</b>(2), 457&ndash;467.
</p>
<p>Pearce RG, Setzer RW, Davis JL, Wambaugh JF (2017).
&ldquo;Evaluation and calibration of high-throughput predictions of chemical distribution to tissues.&rdquo;
<em>Journal of pharmacokinetics and pharmacodynamics</em>, <b>44</b>, 549&ndash;565.
</p>
<p>Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+solve_3comp">solve_3comp</a></code>
</p>
<p><code><a href="#topic+calc_analytic_css_3comp">calc_analytic_css_3comp</a></code>
</p>
<p><code><a href="#topic+parameterize_pbtk">parameterize_pbtk</a></code>
</p>
<p><code><a href="#topic+apply_clint_adjustment">apply_clint_adjustment</a></code>
</p>
<p><code><a href="#topic+tissue.data">tissue.data</a></code>
</p>
<p><code><a href="#topic+physiology.data">physiology.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 parameters &lt;- parameterize_3comp(chem.name='Bisphenol-A',species='Rat')
 parameters &lt;- parameterize_3comp(chem.cas='80-05-7',
                                  species='rabbit',default.to.human=TRUE)
 out &lt;- solve_3comp(parameters=parameters,plots=TRUE)

</code></pre>

<hr>
<h2 id='parameterize_fetal_pbtk'>Parameterize_fetal_PBTK</h2><span id='topic+parameterize_fetal_pbtk'></span>

<h3>Description</h3>

<p>This function initializes the parameters needed in the functions
solve_fetal_pbtk by calling solve_pbtk and adding additional parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameterize_fetal_pbtk(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  fetal_fup_adjustment = TRUE,
  return.kapraun2019 = TRUE,
  suppress.messages = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameterize_fetal_pbtk_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="parameterize_fetal_pbtk_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="parameterize_fetal_pbtk_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="parameterize_fetal_pbtk_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;). Currently only a narrow human model is supported.</p>
</td></tr>
<tr><td><code id="parameterize_fetal_pbtk_+3A_fetal_fup_adjustment">fetal_fup_adjustment</code></td>
<td>
<p>Logical indicator of whether to use an adjusted
estimate for fetal fup based on the fetal:maternal plasma protein binding
ratios presented in McNamara and Alcorn's 2002 study &quot;Protein Binding
Predictions in Infants.&quot; Defaults to TRUE.</p>
</td></tr>
<tr><td><code id="parameterize_fetal_pbtk_+3A_return.kapraun2019">return.kapraun2019</code></td>
<td>
<p>If TRUE (default) the empirical parameters for the
Kapraun et al. (2019) maternal-fetal growth parameters are provided.</p>
</td></tr>
<tr><td><code id="parameterize_fetal_pbtk_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="parameterize_fetal_pbtk_+3A_...">...</code></td>
<td>
<p>Arguments passed to parameterize_pbtk.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>pre_pregnant_BW</code></td>
<td>
<p>Body Weight before pregnancy, kg.</p>
</td></tr>
<tr><td><code>Clmetabolismc</code></td>
<td>
<p>Hepatic Clearance, L/h/kg BW.</p>
</td></tr> <tr><td><code>Fabsgut</code></td>
<td>
<p>Fraction
of the oral dose absorbed, i.e. the fraction of the dose that enters the
gutlumen.</p>
</td></tr> <tr><td><code>Funbound.plasma</code></td>
<td>
<p>Fraction of plasma that is not bound.</p>
</td></tr>
<tr><td><code>Fhep.assay.correction</code></td>
<td>
<p>The fraction of chemical unbound in hepatocyte
assay using the method of Kilford et al. (2008)</p>
</td></tr> <tr><td><code>hematocrit</code></td>
<td>
<p>Percent
volume of red blood cells in the blood.</p>
</td></tr> <tr><td><code>Kgut2pu</code></td>
<td>
<p>Ratio of
concentration of chemical in gut tissue to unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>kgutabs</code></td>
<td>
<p>Rate that chemical enters the gut from gutlumen, 1/h.</p>
</td></tr>
<tr><td><code>Kkidney2pu</code></td>
<td>
<p>Ratio of concentration of chemical in kidney tissue to
unbound concentration in plasma.</p>
</td></tr> <tr><td><code>Kliver2pu</code></td>
<td>
<p>Ratio of concentration of
chemical in liver tissue to unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>Klung2pu</code></td>
<td>
<p>Ratio of concentration of chemical in lung tissue to unbound
concentration in plasma.</p>
</td></tr> <tr><td><code>Krbc2pu</code></td>
<td>
<p>Ratio of concentration of chemical
in red blood cells to unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>Krest2pu</code></td>
<td>
<p>Ratio of concentration of chemical in rest of body tissue to
unbound concentration in plasma.</p>
</td></tr> <tr><td><code>million.cells.per.gliver</code></td>
<td>
<p>Millions
cells per gram of liver tissue.</p>
</td></tr> <tr><td><code>MW</code></td>
<td>
<p>Molecular Weight, g/mol.</p>
</td></tr>
<tr><td><code>Qgfrc</code></td>
<td>
<p>Glomerular Filtration Rate, L/h/kg BW^3/4, volume of fluid
filtered from kidney and excreted.</p>
</td></tr> <tr><td><code>Rblood2plasma</code></td>
<td>
<p>The ratio of the
concentration of the chemical in the blood to the concentration in the
plasma from available_rblood2plasma.</p>
</td></tr> <tr><td><code>Vgutc</code></td>
<td>
<p>Volume of the gut per kg body
weight, L/kg BW.</p>
</td></tr> <tr><td><code>Vkidneyc</code></td>
<td>
<p>Volume of the kidneys per kg body weight, L/kg
BW.</p>
</td></tr> <tr><td><code>Vliverc</code></td>
<td>
<p>Volume of the liver per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vlungc</code></td>
<td>
<p>Volume of the lungs per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vthyroidc</code></td>
<td>
<p>Volume of the thyroid per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Kfgut2pu</code></td>
<td>
<p>Ratio of concentration of chemical in fetal gut tissue to
unbound concentration in plasma.</p>
</td></tr> <tr><td><code>Kfkidney2pu</code></td>
<td>
<p>Ratio of concentration
of chemical in fetal kidney tissue to unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>Kfliver2pu</code></td>
<td>
<p>Ratio of concentration of chemical in fetal liver tissue
to unbound concentration in plasma.</p>
</td></tr> <tr><td><code>Kflung2pu</code></td>
<td>
<p>Ratio of concentration
of chemical in fetal lung tissue to unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>Kfrest2pu</code></td>
<td>
<p>Ratio of concentration of chemical in fetal rest of body
tissue to unbound concentration in plasma.</p>
</td></tr> <tr><td><code>Kfbrain2pu</code></td>
<td>
<p>Ratio of
concentration of chemical in fetal brain tissue to unbound concentration in
plasma.</p>
</td></tr> <tr><td><code>Kthyroid2pu</code></td>
<td>
<p>Ratio of concentration of chemical in fetal
thyroid tissue to unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>Kfthyroid2pu</code></td>
<td>
<p>Ratio of concentration of chemical in fetal thyroid
tissue to unbound concentration in plasma.</p>
</td></tr> <tr><td><code>Kplacenta2pu</code></td>
<td>
<p>Ratio of
concentration of chemical in placental tissue to unbound concentration in
maternal plasma.</p>
</td></tr> <tr><td><code>Kfplacenta2pu</code></td>
<td>
<p>Ratio of concentration of chemical in
placental tissue to unbound concentration in fetal plasma.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Pearce, Mark Sfeir, John Wambaugh, and Dustin Kapraun
</p>
<p>Mark Sfeir, Dustin Kapraun, John Wambaugh
</p>


<h3>References</h3>

<p>Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197.
</p>
<p>McNamara PJ, Alcorn J. Protein binding predictions in infants. 
AAPS PharmSci. 2002;4(1):E4. doi: 10.1208/ps040104. PMID: 12049488.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+solve_fetal_pbtk">solve_fetal_pbtk</a></code>
</p>
<p><code><a href="#topic+parameterize_pbtk">parameterize_pbtk</a></code>
</p>
<p><code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>
</p>
<p><code><a href="#topic+apply_clint_adjustment">apply_clint_adjustment</a></code>
</p>
<p><code><a href="#topic+tissue.data">tissue.data</a></code>
</p>
<p><code><a href="#topic+physiology.data">physiology.data</a></code>
</p>
<p><code><a href="#topic+kapraun2019">kapraun2019</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

 parameters &lt;- parameterize_fetal_pbtk(chem.cas='80-05-7')

 parameters &lt;- parameterize_fetal_pbtk(chem.name='Bisphenol-A',species='Rat')

 
</code></pre>

<hr>
<h2 id='parameterize_gas_pbtk'>Parameters for a generic gas inhalation physiologically-based toxicokinetic model</h2><span id='topic+parameterize_gas_pbtk'></span>

<h3>Description</h3>

<p>This function initializes the parameters needed for the model 'gas_pbtk', for
example <code><a href="#topic+solve_gas_pbtk">solve_gas_pbtk</a></code>. Chemical- and species-specific model 
parameters are generated. These include tissue:plasma partition coefficients 
via Schmitt (2008)'s method as modified by Pearce et al. (2017). Organ volumes 
and flows are retrieved from table <code><a href="#topic+physiology.data">physiology.data</a></code>). This model
was first described by Linakis et al. (2020).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameterize_gas_pbtk(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  tissuelist = list(liver = c("liver"), kidney = c("kidney"), lung = c("lung"), gut =
    c("gut")),
  force.human.clint.fup = FALSE,
  clint.pvalue.threshold = 0.05,
  adjusted.Funbound.plasma = TRUE,
  adjusted.Clint = TRUE,
  regression = TRUE,
  vmax = 0,
  km = 1,
  exercise = FALSE,
  fR = 12,
  VT = 0.75,
  VD = 0.15,
  suppress.messages = FALSE,
  minimum.Funbound.plasma = 1e-04,
  Caco2.options = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameterize_gas_pbtk_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_tissuelist">tissuelist</code></td>
<td>
<p>Specifies compartment names and tissues groupings.
Remaining tissues in tissue.data are lumped in the rest of the body.
However, solve_pbtk only works with the default parameters.</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_force.human.clint.fup">force.human.clint.fup</code></td>
<td>
<p>Forces use of human values for hepatic
intrinsic clearance and fraction of unbound plasma if true.</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_clint.pvalue.threshold">clint.pvalue.threshold</code></td>
<td>
<p>Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses Pearce et al. (2017) lipid binding adjustment
for Funbound.plasma (which impacts partition coefficients) when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_adjusted.clint">adjusted.Clint</code></td>
<td>
<p>Uses Kilford et al. (2008) hepatocyte incubation
binding adjustment for Clint when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_vmax">vmax</code></td>
<td>
<p>Michaelis-Menten vmax value in reactions/min</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_km">km</code></td>
<td>
<p>Michaelis-Menten concentration of half-maximal reaction velocity
in desired output concentration units.</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_exercise">exercise</code></td>
<td>
<p>Logical indicator of whether to simulate an exercise-induced
heightened respiration rate</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_fr">fR</code></td>
<td>
<p>Respiratory frequency (breaths/minute), used especially to adjust
breathing rate in the case of exercise. This parameter, along with VT and VD
(below) gives another option for calculating Qalv (Alveolar ventilation) 
in case pulmonary ventilation rate is not known</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_vt">VT</code></td>
<td>
<p>Tidal volume (L), to be modulated especially as part of simulating
the state of exercise</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_vd">VD</code></td>
<td>
<p>Anatomical dead space (L), to be modulated especially as part of
simulating the state of exercise</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output messages are suppressed.</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
<tr><td><code id="parameterize_gas_pbtk_+3A_...">...</code></td>
<td>
<p>Other parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>BW</code></td>
<td>
<p>Body Weight, kg.</p>
</td></tr> 
<tr><td><code>Clint</code></td>
<td>
<p>Hepatic intrinsic clearance, uL/min/10^6 cells</p>
</td></tr>
<tr><td><code>Clint.dist</code></td>
<td>
<p>Distribution of hepatic intrinsic clearance values
(median, lower 95th, upper 95th, p value)</p>
</td></tr> 
<tr><td><code>Clmetabolismc</code></td>
<td>
<p>Hepatic Clearance, L/h/kg BW.</p>
</td></tr> 
<tr><td><code>Fabsgut</code></td>
<td>
<p>Fraction of the oral dose absorbed, i.e. the fraction of the
dose that enters the gut lumen.</p>
</td></tr>
<tr><td><code>Fhep.assay.correction</code></td>
<td>
<p>The fraction of chemical unbound in hepatocyte
assay using the method of Kilford et al. (2008)</p>
</td></tr> 
<tr><td><code>Funbound.plasma</code></td>
<td>
<p>Fraction of chemical unbound to plasma.</p>
</td></tr> 
<tr><td><code>Funbound.plasma.adjustment</code></td>
<td>
<p>Fraction unbound to plasma adjusted as
described in Pearce et al. 2017</p>
</td></tr>
<tr><td><code>Funbound.plasma.dist</code></td>
<td>
<p>Distribution of fraction unbound to plasma
(median, lower 95th, upper 95th)</p>
</td></tr>
<tr><td><code>hematocrit</code></td>
<td>
<p>Percent volume of red blood cells in the blood.</p>
</td></tr>
<tr><td><code>Kblood2air</code></td>
<td>
<p>Ratio of concentration of chemical in blood to air</p>
</td></tr>
<tr><td><code>Kgut2pu</code></td>
<td>
<p>Ratio of concentration of chemical in gut tissue to unbound
concentration in plasma.</p>
</td></tr> 
<tr><td><code>kgutabs</code></td>
<td>
<p>Rate that chemical enters the gut from gutlumen, 1/h.</p>
</td></tr> 
<tr><td><code>Kkidney2pu</code></td>
<td>
<p>Ratio of concentration of chemical in kidney tissue to
unbound concentration in plasma.</p>
</td></tr> 
<tr><td><code>Kliver2pu</code></td>
<td>
<p>Ratio of concentration of chemical in liver tissue to
unbound concentration in plasma.</p>
</td></tr> 
<tr><td><code>Klung2pu</code></td>
<td>
<p>Ratio of concentration of chemical in lung tissue
to unbound concentration in plasma.</p>
</td></tr> 
<tr><td><code>km</code></td>
<td>
<p>Michaelis-Menten concentration of half-maximal activity</p>
</td></tr>
<tr><td><code>Kmuc2air</code></td>
<td>
<p>Mucus to air partition coefficient</p>
</td></tr>
<tr><td><code>Krbc2pu</code></td>
<td>
<p>Ratio of concentration of chemical in red blood cells to
unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>Krest2pu</code></td>
<td>
<p>Ratio of concentration of chemical in rest of body tissue to
unbound concentration in plasma.</p>
</td></tr> 
<tr><td><code>kUrtc</code></td>
<td>
<p>Unscaled upper respiratory tract uptake parameter (L/h/kg^0.75)</p>
</td></tr>
<tr><td><code>liver.density</code></td>
<td>
<p>Density of liver in g/mL</p>
</td></tr>
<tr><td><code>MA</code></td>
<td>
<p>phospholipid:water distribution coefficient, membrane affinity</p>
</td></tr>
<tr><td><code>million.cells.per.gliver</code></td>
<td>
<p>Millions cells per gram of liver tissue.</p>
</td></tr> 
<tr><td><code>MW</code></td>
<td>
<p>Molecular Weight, g/mol.</p>
</td></tr>
<tr><td><code>pKa_Accept</code></td>
<td>
<p>compound H association equilibrium constant(s)</p>
</td></tr>
<tr><td><code>pKa_Donor</code></td>
<td>
<p>compound H dissociation equilibirum constant(s)</p>
</td></tr>
<tr><td><code>Pow</code></td>
<td>
<p>octanol:water partition coefficient (not log transformed)</p>
</td></tr>
<tr><td><code>Qalvc</code></td>
<td>
<p>Unscaled alveolar ventilation rate (L/h/kg^0.75)</p>
</td></tr>
<tr><td><code>Qcardiacc</code></td>
<td>
<p>Cardiac Output, L/h/kg BW^3/4.</p>
</td></tr> 
<tr><td><code>Qgfrc</code></td>
<td>
<p>Glomerular Filtration Rate, L/h/kg BW^0.75, volume of fluid
filtered from kidney and excreted.</p>
</td></tr> 
<tr><td><code>Qgutf</code></td>
<td>
<p>Fraction of cardiac output flowing to the gut.</p>
</td></tr>
<tr><td><code>Qkidneyf</code></td>
<td>
<p>Fraction of cardiac output flowing to the kidneys.</p>
</td></tr>
<tr><td><code>Qliverf</code></td>
<td>
<p>Fraction of cardiac output flowing to the liver.</p>
</td></tr>
<tr><td><code>Qlungf</code></td>
<td>
<p>Fraction of cardiac output flowing to lung tissue.</p>
</td></tr>
<tr><td><code>Qrestf</code></td>
<td>
<p>Fraction of blood flow to rest of body</p>
</td></tr>
<tr><td><code>Rblood2plasma</code></td>
<td>
<p>The ratio of the concentration of the chemical in the
blood to the concentration in the plasma from available_rblood2plasma.</p>
</td></tr>
<tr><td><code>Vartc</code></td>
<td>
<p>Volume of the arteries per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vgutc</code></td>
<td>
<p>Volume of the gut per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vkidneyc</code></td>
<td>
<p>Volume of the kidneys per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vliverc</code></td>
<td>
<p>Volume of the liver per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vlungc</code></td>
<td>
<p>Volume of the lungs per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>vmax</code></td>
<td>
<p>Michaelis-Menten maximum reaction velocity (1/min)</p>
</td></tr>
<tr><td><code>Vmucc</code></td>
<td>
<p>Unscaled mucosal volume (L/kg BW^0.75</p>
</td></tr>
<tr><td><code>Vrestc</code></td>
<td>
<p> Volume of the rest of the body per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vvenc</code></td>
<td>
<p>Volume of the veins per kg body weight, L/kg BW.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Matt Linakis, Robert Pearce, John Wambaugh
</p>


<h3>References</h3>

<p>Linakis MW, Sayre RR, Pearce RG, Sfeir MA, Sipes NS, Pangburn HA, Gearhart JM, Wambaugh JF (2020).
&ldquo;Development and evaluation of a high-throughput inhalation model for organic chemicals.&rdquo;
<em>Journal of exposure science &amp; environmental epidemiology</em>, <b>30</b>(5), 866&ndash;877.
</p>
<p>Schmitt W (2008).
&ldquo;General approach for the calculation of tissue to plasma partition coefficients.&rdquo;
<em>Toxicology in vitro</em>, <b>22</b>(2), 457&ndash;467.
</p>
<p>Pearce RG, Setzer RW, Davis JL, Wambaugh JF (2017).
&ldquo;Evaluation and calibration of high-throughput predictions of chemical distribution to tissues.&rdquo;
<em>Journal of pharmacokinetics and pharmacodynamics</em>, <b>44</b>, 549&ndash;565.
</p>
<p>Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+solve_gas_pbtk">solve_gas_pbtk</a></code>
</p>
<p><code><a href="#topic+apply_clint_adjustment">apply_clint_adjustment</a></code>
</p>
<p><code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>
</p>
<p><code><a href="#topic+available_rblood2plasma">available_rblood2plasma</a></code>
</p>
<p><code><a href="#topic+calc_kair">calc_kair</a></code>
</p>
<p><code><a href="#topic+tissue.data">tissue.data</a></code>
</p>
<p><code><a href="#topic+physiology.data">physiology.data</a></code>
</p>
<p><code><a href="#topic+get_clint">get_clint</a></code>
</p>
<p><code><a href="#topic+get_fup">get_fup</a></code>
</p>
<p><code><a href="#topic+get_physchem_param">get_physchem_param</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>parameters &lt;- parameterize_gas_pbtk(chem.cas='129-00-0')

parameters &lt;- parameterize_gas_pbtk(chem.name='pyrene',species='Rat')

parameterize_gas_pbtk(chem.cas = '56-23-5')

parameters &lt;- parameterize_gas_pbtk(chem.name='Carbon tetrachloride',species='Rat')

# Change the tissue lumping:
compartments &lt;- list(liver=c("liver"),fast=c("heart","brain","muscle","kidney"),
                      lung=c("lung"),gut=c("gut"),slow=c("bone"))
parameterize_gas_pbtk(chem.name="Bisphenol a",species="Rat",default.to.human=TRUE,
                   tissuelist=compartments) 

</code></pre>

<hr>
<h2 id='parameterize_pbtk'>Parameters for a generic physiologically-based toxicokinetic model</h2><span id='topic+parameterize_pbtk'></span>

<h3>Description</h3>

<p>Generate a chemical- and species-specific set of PBPK model parameters.
Parameters include 
tissue:plasma partition coefficients, organ volumes, and flows 
for the tissue lumping scheme specified by argument tissuelist.
Tissure:(fraction unbound in) plasma partitition coefficients are predicted
via Schmitt (2008)'s method as modified by Pearce et al. (2017) using
<code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>. Organ volumes and flows are
retrieved from table <code><a href="#topic+physiology.data">physiology.data</a></code>.
Tissues must be described in table <code><a href="#topic+tissue.data">tissue.data</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameterize_pbtk(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  default.to.human = FALSE,
  tissuelist = list(liver = c("liver"), kidney = c("kidney"), lung = c("lung"), gut =
    c("gut")),
  force.human.clint.fup = FALSE,
  clint.pvalue.threshold = 0.05,
  adjusted.Funbound.plasma = TRUE,
  adjusted.Clint = TRUE,
  regression = TRUE,
  suppress.messages = FALSE,
  restrictive.clearance = TRUE,
  minimum.Funbound.plasma = 1e-04,
  million.cells.per.gliver = 110,
  liver.density = 1.05,
  kgutabs = 2.18,
  Caco2.options = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameterize_pbtk_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_tissuelist">tissuelist</code></td>
<td>
<p>Specifies compartment names and tissues groupings.
Remaining tissues in tissue.data are lumped in the rest of the body.
However, <code><a href="#topic+solve_pbtk">solve_pbtk</a></code> only works with the default parameters.</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_force.human.clint.fup">force.human.clint.fup</code></td>
<td>
<p>Forces use of human values for hepatic
intrinsic clearance and fraction of unbound plasma if true.</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_clint.pvalue.threshold">clint.pvalue.threshold</code></td>
<td>
<p>Hepatic clearance for chemicals where the in
vitro clearance assay result has a p-values greater than the threshold are
set to zero.</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses Pearce et al. (2017) lipid binding adjustment
for Funbound.plasma (which impacts partition coefficients) when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_adjusted.clint">adjusted.Clint</code></td>
<td>
<p>Uses Kilford et al. (2008) hepatocyte incubation
binding adjustment for Clint when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>In calculating hepatic.bioavailability, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p><code class="reqn">f_{up}</code> is not allowed to drop below
this value (default is 0.0001).</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_million.cells.per.gliver">million.cells.per.gliver</code></td>
<td>
<p>Hepatocellularity (defaults to 110 10^6 cells/g-liver, from Carlile et al. (1997))</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_liver.density">liver.density</code></td>
<td>
<p>Liver density (defaults to 1.05 g/mL from International Commission on Radiological Protection (1975))</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_kgutabs">kgutabs</code></td>
<td>
<p>Oral absorption rate from gut (defaults to 2.18 1/h from Wambaugh et al. (2018))</p>
</td></tr>
<tr><td><code id="parameterize_pbtk_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, this function initializes the parameters needed in the functions 
<code><a href="#topic+solve_pbtk">solve_pbtk</a></code>, <code><a href="#topic+calc_css">calc_css</a></code>, and others using the httk 
default generic PBTK model (for oral and intravenous dosing only).
</p>
<p>The default PBTK model includes an explicit first pass of the chemical through
the liver before it becomes available to systemic blood. We model systemic oral bioavailability as 
F<sub>bio</sub>=F<sub>abs</sub>*F<sub>gut</sub>*F<sub>hep</sub>.
Only if F<sub>bio</sub>
has been measured in vivo and is found in
table <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> then we set 
F<sub>abs</sub>*F<sub>gut</sub> 
to the measured value divided by 
F<sub>hep</sub> 
where F<sub>hep</sub>
is estimated from in vitro TK data using 
<code><a href="#topic+calc_hep_bioavailability">calc_hep_bioavailability</a></code>.
If Caco2 membrane permeability data or predictions
are available F<sub>abs</sub> is estimated
using <code><a href="#topic+calc_fabs.oral">calc_fabs.oral</a></code>.
Intrinsic hepatic metabolism is used to very roughly estimate
F<sub>gut</sub>
using <code><a href="#topic+calc_fgut.oral">calc_fgut.oral</a></code>.
</p>


<h3>Value</h3>

<table>
<tr><td><code>BW</code></td>
<td>
<p>Body Weight, kg.</p>
</td></tr> 
<tr><td><code>Clmetabolismc</code></td>
<td>
<p>Hepatic Clearance, L/h/kg BW.</p>
</td></tr> 
<tr><td><code>Fabsgut</code></td>
<td>
<p>Fraction of the oral dose absorbed, i.e. the fraction of
the dose that enters the gutlumen.</p>
</td></tr> 
<tr><td><code>Funbound.plasma</code></td>
<td>
<p>Fraction of plasma that is not bound.</p>
</td></tr>
<tr><td><code>Fhep.assay.correction</code></td>
<td>
<p>The fraction of chemical unbound in hepatocyte
assay using the method of Kilford et al. (2008)</p>
</td></tr>
<tr><td><code>hematocrit</code></td>
<td>
<p>Percent volume of red blood cells in the blood.</p>
</td></tr>
<tr><td><code>Kgut2pu</code></td>
<td>
<p>Ratio of concentration of chemical in gut tissue to unbound
concentration in plasma.</p>
</td></tr>
<tr><td><code>kgutabs</code></td>
<td>
<p>Rate that chemical enters the gut from gutlumen, 1/h.</p>
</td></tr>
<tr><td><code>Kkidney2pu</code></td>
<td>
<p>Ratio of concentration of chemical in kidney tissue to
unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>Kliver2pu</code></td>
<td>
<p>Ratio of concentration of chemical in liver tissue to
unbound concentration in plasma.</p>
</td></tr> 
<tr><td><code>Klung2pu</code></td>
<td>
<p>Ratio of concentration of chemical in lung tissue
to unbound concentration in plasma.</p>
</td></tr> 
<tr><td><code>Krbc2pu</code></td>
<td>
<p>Ratio of concentration of chemical in red blood cells to
unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>Krest2pu</code></td>
<td>
<p>Ratio of concentration of chemical in rest of body tissue to
unbound concentration in plasma.</p>
</td></tr>
<tr><td><code>million.cells.per.gliver</code></td>
<td>
<p>Millions cells per gram of liver tissue.</p>
</td></tr>
<tr><td><code>MW</code></td>
<td>
<p>Molecular Weight, g/mol.</p>
</td></tr>
<tr><td><code>Qcardiacc</code></td>
<td>
<p>Cardiac Output, L/h/kg BW^3/4.</p>
</td></tr>
<tr><td><code>Qgfrc</code></td>
<td>
<p>Glomerular Filtration Rate, L/h/kg BW^3/4, volume of fluid
filtered from kidney and excreted.</p>
</td></tr>
<tr><td><code>Qgutf</code></td>
<td>
<p>Fraction of cardiac output flowing to the gut.</p>
</td></tr>
<tr><td><code>Qkidneyf</code></td>
<td>
<p>Fraction of cardiac output flowing to the kidneys.</p>
</td></tr>
<tr><td><code>Qliverf</code></td>
<td>
<p>Fraction of cardiac output flowing to the liver.</p>
</td></tr>
<tr><td><code>Rblood2plasma</code></td>
<td>
<p>The ratio of the concentration of the chemical in the
blood to the concentration in the plasma from available_rblood2plasma.</p>
</td></tr>
<tr><td><code>Vartc</code></td>
<td>
<p>Volume of the arteries per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vgutc</code></td>
<td>
<p>Volume of the gut per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vkidneyc</code></td>
<td>
<p>Volume of the kidneys per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vliverc</code></td>
<td>
<p>Volume of the liver per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vlungc</code></td>
<td>
<p>Volume of the lungs per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vrestc</code></td>
<td>
<p> Volume of the rest of the body per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>Vvenc</code></td>
<td>
<p>Volume of the veins per kg body weight, L/kg BW.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>
<p>Schmitt W (2008).
&ldquo;General approach for the calculation of tissue to plasma partition coefficients.&rdquo;
<em>Toxicology in vitro</em>, <b>22</b>(2), 457&ndash;467.
</p>
<p>Pearce RG, Setzer RW, Davis JL, Wambaugh JF (2017).
&ldquo;Evaluation and calibration of high-throughput predictions of chemical distribution to tissues.&rdquo;
<em>Journal of pharmacokinetics and pharmacodynamics</em>, <b>44</b>, 549&ndash;565.
</p>
<p>Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197.
</p>
<p>International Commission on Radiological Protection. Report of the task 
group on reference man. Vol. 23. Pergamon, Oxford. 1975.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+solve_pbtk">solve_pbtk</a></code>
</p>
<p><code><a href="#topic+calc_analytic_css_pbtk">calc_analytic_css_pbtk</a></code>
</p>
<p><code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>
</p>
<p><code><a href="#topic+apply_clint_adjustment">apply_clint_adjustment</a></code>
</p>
<p><code><a href="#topic+tissue.data">tissue.data</a></code>
</p>
<p><code><a href="#topic+physiology.data">physiology.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 parameters &lt;- parameterize_pbtk(chem.cas='80-05-7')

 parameters &lt;- parameterize_pbtk(chem.name='Bisphenol-A',species='Rat')

 # Change the tissue lumping (note, these model parameters will not work with our current solver):
 compartments &lt;- list(liver=c("liver"),fast=c("heart","brain","muscle","kidney"),
                      lung=c("lung"),gut=c("gut"),slow=c("bone"))
 parameterize_pbtk(chem.name="Bisphenol a",species="Rat",default.to.human=TRUE,
                   tissuelist=compartments) 
</code></pre>

<hr>
<h2 id='parameterize_schmitt'>Parameters for Schmitt's (2008) Tissue Partition Coefficient Method</h2><span id='topic+parameterize_schmitt'></span>

<h3>Description</h3>

<p>This function provides the necessary parameters to run
<code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>, excluding the data in table
<code><a href="#topic+tissue.data">tissue.data</a></code>. The model is based on the Schmitt (2008) method
for predicting tissue:plasma partition coefficients as modified by Pearce 
et al. (2017). The modifications include approaches adapted from Peyret 
et al. (2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameterize_schmitt(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  parameters = NULL,
  species = "Human",
  default.to.human = FALSE,
  force.human.fup = FALSE,
  adjusted.Funbound.plasma = TRUE,
  suppress.messages = FALSE,
  minimum.Funbound.plasma = 1e-04
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameterize_schmitt_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="parameterize_schmitt_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  if
parameters is not specified then the chemical must be identified by either
CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="parameterize_schmitt_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; if parameters is not specified then the chemical must be identified by 
either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="parameterize_schmitt_+3A_parameters">parameters</code></td>
<td>
<p>Chemcial and physiological description parameters needed
to run the Schmitt et al. (2008) model</p>
</td></tr>
<tr><td><code id="parameterize_schmitt_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="parameterize_schmitt_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing fraction of unbound plasma with
human values if true.</p>
</td></tr>
<tr><td><code id="parameterize_schmitt_+3A_force.human.fup">force.human.fup</code></td>
<td>
<p>Returns human fraction of unbound plasma in
calculation for rats if true.
When species is specified as rabbit, dog, or mouse, the human unbound
fraction is substituted.</p>
</td></tr>
<tr><td><code id="parameterize_schmitt_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses Pearce et al. (2017) lipid binding adjustment
for Funbound.plasma (which impacts partition coefficients) when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="parameterize_schmitt_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="parameterize_schmitt_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Funbound.plasma</code></td>
<td>
<p>Unbound fraction in plasma, adjusted for lipid binding according to Pearce et al. (2017)</p>
</td></tr>
<tr><td><code>unadjusted.Funbound.plasma</code></td>
<td>
<p>measured unbound fraction in plasma (0.005
if below limit of detection)</p>
</td></tr> <tr><td><code>Pow</code></td>
<td>
<p>octanol:water partition coefficient
(not log transformed)</p>
</td></tr> <tr><td><code>pKa_Donor</code></td>
<td>
<p>compound H dissociation equilibrium
constant(s)</p>
</td></tr> <tr><td><code>pKa_Accept</code></td>
<td>
<p>compound H association equilibrium
constant(s)</p>
</td></tr> <tr><td><code>MA</code></td>
<td>
<p>phospholipid:water distribution coefficient, membrane
affinity</p>
</td></tr> <tr><td><code>Fprotein.plasma</code></td>
<td>
<p>protein fraction in plasma</p>
</td></tr>
<tr><td><code>plasma.pH</code></td>
<td>
<p>pH of the plasma</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Robert Pearce and John Wambaugh
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>
<p>Schmitt W (2008).
&ldquo;General approach for the calculation of tissue to plasma partition coefficients.&rdquo;
<em>Toxicology in vitro</em>, <b>22</b>(2), 457&ndash;467.
</p>
<p>Schmitt W (2008).
&ldquo;Corrigendum to:'General approach for the calculation of tissue to plasma partition coefficients'[Toxicology in Vitro 22 (2008) 457&ndash;467].&rdquo;
<em>Toxicology in Vitro</em>, <b>22</b>(6), 1666.
</p>
<p>Pearce RG, Setzer RW, Davis JL, Wambaugh JF (2017).
&ldquo;Evaluation and calibration of high-throughput predictions of chemical distribution to tissues.&rdquo;
<em>Journal of pharmacokinetics and pharmacodynamics</em>, <b>44</b>, 549&ndash;565.
</p>
<p>Peyret T, Poulin P, Krishnan K (2010).
&ldquo;A unified algorithm for predicting partition coefficients for PBPK modeling of drugs and environmental chemicals.&rdquo;
<em>Toxicology and applied pharmacology</em>, <b>249</b>(3), 197&ndash;207.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>
</p>
<p><code><a href="#topic+tissue.data">tissue.data</a></code>
</p>
<p><code><a href="#topic+calc_ma">calc_ma</a></code>
</p>
<p><code><a href="#topic+apply_fup_adjustment">apply_fup_adjustment</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
parameterize_schmitt(chem.name='bisphenola')

</code></pre>

<hr>
<h2 id='parameterize_steadystate'>Parameters for a three-compartment toxicokinetic model at steady-state</h2><span id='topic+parameterize_steadystate'></span>

<h3>Description</h3>

<p>This function initializes the parameters needed in the functions
<code><a href="#topic+calc_mc_css">calc_mc_css</a></code>, <code><a href="#topic+calc_mc_oral_equiv">calc_mc_oral_equiv</a></code>, and 
<code><a href="#topic+calc_analytic_css">calc_analytic_css</a></code> for the three
compartment steady state model ('3compartmentss') as used in 
Rotroff et al. (2010), Wetmore et al. (2012), Wetmore et al. (2015), and 
elsewhere. By assuming that enough time has passed to reach steady-state, we
eliminate the need for tissue-specific parititon coefficients because we 
assume all tissues have come to equilibrium with the unbound concentration
in plasma. However, we still use chemical properties to predict the 
blood:plasma ratio for estimating first-pass hepatic metabolism for oral
exposures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameterize_steadystate(
  chem.cas = NULL,
  chem.name = NULL,
  dtxsid = NULL,
  species = "Human",
  clint.pvalue.threshold = 0.05,
  default.to.human = FALSE,
  force.human.clint.fup = FALSE,
  adjusted.Funbound.plasma = TRUE,
  adjusted.Clint = TRUE,
  restrictive.clearance = TRUE,
  fup.lod.default = 0.005,
  suppress.messages = FALSE,
  minimum.Funbound.plasma = 1e-04,
  Caco2.options = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameterize_steadystate_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Chemical Abstract Services Registry Number (CAS-RN) &ndash; the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_chem.name">chem.name</code></td>
<td>
<p>Chemical name (spaces and capitalization ignored) &ndash;  the 
chemical must be identified by either CAS, name, or DTXISD</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
&ndash; the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_clint.pvalue.threshold">clint.pvalue.threshold</code></td>
<td>
<p>Hepatic clearances with clearance assays
having p-values greater than the threshold are set to zero.</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing species-specific values with human values if
TRUE (default is FALSE).</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_force.human.clint.fup">force.human.clint.fup</code></td>
<td>
<p>Uses human hepatic intrinsic clearance and fraction
of unbound plasma in calculation of partition coefficients for rats if true.</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses Pearce et al. (2017) lipid binding adjustment
for Funbound.plasma (which impacts partition coefficients) when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_adjusted.clint">adjusted.Clint</code></td>
<td>
<p>Uses Kilford et al. (2008) hepatocyte incubation
binding adjustment for Clint when set to TRUE (Default).</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>In calculating hepatic.bioavailability, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_fup.lod.default">fup.lod.default</code></td>
<td>
<p>Default value used for fraction of unbound plasma for
chemicals where measured value was below the limit of detection. Default
value is 0.0005.</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
<tr><td><code id="parameterize_steadystate_+3A_...">...</code></td>
<td>
<p>Other parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We model systemic oral bioavailability as 
F<sub>bio</sub>=F<sub>abs</sub>*F<sub>gut</sub>*F<sub>hep</sub>.
F<sub>hep</sub>
is estimated from in vitro TK data using 
<code><a href="#topic+calc_hep_bioavailability">calc_hep_bioavailability</a></code>.
If F<sub>bio</sub>
has been measured in vivo and is found in
table <code><a href="#topic+chem.physical_and_invitro.data">chem.physical_and_invitro.data</a></code> then we set 
F<sub>abs</sub>*F<sub>gut</sub> 
to the measured value divided by 
F<sub>hep</sub> 
Otherwise, if Caco2 membrane permeability data or predictions
are available F<sub>abs</sub> is estimated
using <code><a href="#topic+calc_fabs.oral">calc_fabs.oral</a></code>.
Intrinsic hepatic metabolism is used to very roughly estimate
F<sub>gut</sub>
using <code><a href="#topic+calc_fgut.oral">calc_fgut.oral</a></code>.
</p>


<h3>Value</h3>

<table>
<tr><td><code>Clint</code></td>
<td>
<p>Hepatic Intrinsic Clearance, uL/min/10^6 cells.</p>
</td></tr>
<tr><td><code>Fabsgut</code></td>
<td>
<p>Fraction of the oral dose absorbed and surviving gut metabolism, 
that is, the fraction of the dose that enters the gutlumen.</p>
</td></tr>  
<tr><td><code>Funbound.plasma</code></td>
<td>
<p>Fraction of plasma that is not bound.</p>
</td></tr> 
<tr><td><code>Qtotal.liverc</code></td>
<td>
<p>Flow rate of blood exiting the liver, L/h/kg BW^3/4.</p>
</td></tr> 
<tr><td><code>Qgfrc</code></td>
<td>
<p>Glomerular Filtration Rate, L/h/kg
BW^3/4, volume of fluid filtered from kidney and excreted.</p>
</td></tr> 
<tr><td><code>BW</code></td>
<td>
<p>Body Weight, kg</p>
</td></tr> 
<tr><td><code>MW</code></td>
<td>
<p>Molecular Weight, g/mol</p>
</td></tr>
<tr><td><code>million.cells.per.gliver</code></td>
<td>
<p>Millions cells per gram of liver tissue.</p>
</td></tr>
<tr><td><code>Vliverc</code></td>
<td>
<p>Volume of the liver per kg body weight, L/kg BW.</p>
</td></tr>
<tr><td><code>liver.density</code></td>
<td>
<p>Liver tissue density, kg/L.</p>
</td></tr>
<tr><td><code>Fhep.assay.correction</code></td>
<td>
<p>The fraction of chemical unbound in hepatocyte
assay using the method of Kilford et al. (2008)</p>
</td></tr>
<tr><td><code>hepatic.bioavailability</code></td>
<td>
<p>Fraction of dose remaining after first pass
clearance, calculated from the corrected well-stirred model.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh and Greg Honda
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>
<p>Kilford PJ, Gertz M, Houston JB, Galetin A (2008).
&ldquo;Hepatocellular binding of drugs: correction for unbound fraction in hepatocyte incubations using microsomal binding or drug lipophilicity data.&rdquo;
<em>Drug Metabolism and Disposition</em>, <b>36</b>(7), 1194&ndash;1197.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc_analytic_css_3compss">calc_analytic_css_3compss</a></code>
</p>
<p><code><a href="#topic+apply_clint_adjustment">apply_clint_adjustment</a></code>
</p>
<p><code><a href="#topic+tissue.data">tissue.data</a></code>
</p>
<p><code><a href="#topic+physiology.data">physiology.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 parameters &lt;- parameterize_steadystate(chem.name='Bisphenol-A',species='Rat')
 parameters &lt;- parameterize_steadystate(chem.cas='80-05-7')

</code></pre>

<hr>
<h2 id='pc.data'>Partition Coefficient Data</h2><span id='topic+pc.data'></span>

<h3>Description</h3>

<p>Measured rat in vivo partition coefficients and data for predicting them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pc.data
</code></pre>


<h3>Format</h3>

<p>A data.frame.
</p>


<h3>Author(s)</h3>

<p>Jimena Davis and Robert Pearce
</p>


<h3>References</h3>

<p>Schmitt, W., General approach for the calculation of tissue to
plasma partition coefficients. Toxicology in Vitro, 2008. 22(2): p. 457-467.
</p>
<p>Schmitt, W., Corrigendum to:&quot;General approach for the calculation of tissue
to plasma partition coefficients&quot;[Toxicology in Vitro 22 (2008) 457-467].
Toxicology in Vitro, 2008. 22(6): p. 1666.
</p>
<p>Poulin, P. and F.P. Theil, A priori prediction of tissue: plasma partition
coefficients of drugs to facilitate the use of physiologically based
pharmacokinetic models in drug discovery. Journal of pharmaceutical
sciences, 2000. 89(1): p. 16-35.
</p>
<p>Rodgers, T. and M. Rowland, Physiologically based pharmacokinetic modelling
2: predicting the tissue distribution of acids, very weak bases, neutrals
and zwitterions. Journal of pharmaceutical sciences, 2006. 95(6): p.
1238-1257.
</p>
<p>Rodgers, T., D. Leahy, and M. Rowland, Physiologically based pharmacokinetic
modeling 1: predicting the tissue distribution of moderate-to-strong bases.
Journal of pharmaceutical sciences, 2005. 94(6): p. 1259-1276.
</p>
<p>Rodgers, T., D. Leahy, and M. Rowland, Tissue distribution of basic drugs:
Accounting for enantiomeric, compound and regional differences amongst
beta-blocking drugs in rat. Journal of pharmaceutical sciences, 2005. 94(6):
p. 1237-1248.
</p>
<p>Gueorguieva, I., et al., Development of a whole body physiologically based
model to characterise the pharmacokinetics of benzodiazepines. 1: Estimation
of rat tissue-plasma partition ratios. Journal of pharmacokinetics and
pharmacodynamics, 2004. 31(4): p. 269-298.
</p>
<p>Poulin, P., K. Schoenlein, and F.P. Theil, Prediction of adipose tissue:
plasma partition coefficients for structurally unrelated drugs. Journal of
pharmaceutical sciences, 2001. 90(4): p. 436-447.
</p>
<p>Bjorkman, S., Prediction of the volume of distribution of a drug: which
tissue-plasma partition coefficients are needed? Journal of pharmacy and
pharmacology, 2002. 54(9): p. 1237-1245.
</p>
<p>Yun, Y. and A. Edginton, Correlation-based prediction of tissue-to-plasma
partition coefficients using readily available input parameters.
Xenobiotica, 2013. 43(10): p. 839-852.
</p>
<p>Uchimura, T., et al., Prediction of human blood-to-plasma drug concentration
ratio. Biopharmaceutics &amp; drug disposition, 2010. 31(5-6): p. 286-297.
</p>

<hr>
<h2 id='pearce2017regression'>Pearce et al. 2017 data</h2><span id='topic+pearce2017regression'></span><span id='topic+Pearce2017Regression'></span>

<h3>Description</h3>

<p>This table includes the adjusted and unadjusted regression parameter
estimates for the chemical-specifc plasma
protein unbound fraction (fup) in 12 different tissue types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pearce2017regression
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Details</h3>

<p>Predictions were made with regression models, 
as reported in Pearce et al. (2017).
</p>


<h3>Author(s)</h3>

<p>Robert G. Pearce
</p>


<h3>Source</h3>

<p>Pearce et al. 2017 Regression Models
</p>


<h3>References</h3>

<p>Pearce, Robert G., et al. &quot;Evaluation and calibration of 
high-throughput predictions of chemical distribution to tissues.&quot;
Journal of pharmacokinetics and pharmacodynamics 44.6 (2017): 549-565.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>
</p>

<hr>
<h2 id='pharma'>DRUGS|NORMAN: Pharmaceutical List with EU, Swiss, US Consumption Data</h2><span id='topic+pharma'></span>

<h3>Description</h3>

<p>SWISSPHARMA is a list of pharmaceuticals with consumption data from
Switzerland, France, Germany and the USA, used for a suspect
screening/exposure modelling approach described in
Singer et al 2016, DOI: 10.1021/acs.est.5b03332. The original data is
available on the NORMAN Suspect List Exchange.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pharma
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 14 rows and 954 columns.
</p>


<h3>Source</h3>

<p><a href="https://comptox.epa.gov/dashboard/chemical_lists/swisspharma">https://comptox.epa.gov/dashboard/chemical_lists/swisspharma</a>
</p>


<h3>References</h3>

<p>Wambaugh et al. (2019) &quot;Assessing Toxicokinetic Uncertainty and
Variability in Risk Prioritization&quot;, Toxicological Sciences, 172(2), 235-251.
</p>

<hr>
<h2 id='physiology.data'>Species-specific physiology parameters</h2><span id='topic+physiology.data'></span>

<h3>Description</h3>

<p>This data set contains values from Davies and Morris (1993) necessary to
paramaterize a toxicokinetic model for human, mouse, rat, dog, or rabbit.
The temperature for each species are taken from Robertshaw et al. (2004),
Gordon (1993), and Stammers(1926).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>physiology.data
</code></pre>


<h3>Format</h3>

<p>A data.frame containing 11 rows and 7 columns.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh and Nisha Sipes
</p>


<h3>Source</h3>

<p>Wambaugh, John F., et al. &quot;Toxicokinetic triage for environmental
chemicals.&quot; Toxicological Sciences (2015): 228-237.
</p>


<h3>References</h3>

<p>Davies, B. and Morris, T. (1993). Physiological Parameters in
Laboratory Animals and Humans. Pharmaceutical Research 10(7), 1093-1095,
10.1023/a:1018943613122.  

Environment, in Dukes' Physiology of Domestic Animals, 12th ed., Reece W.O.,
Ed. Copyright 2004 by Cornell University.  Stammers (1926) The blood count
and body temperature in normal rats Gordon (1993) Temperature Regulation in
Laboratory Rodents
</p>

<hr>
<h2 id='pksim.pcs'>Partition Coefficients from PK-Sim</h2><span id='topic+pksim.pcs'></span>

<h3>Description</h3>

<p>Dallmann et al. (2018) made use of PK-Sim to predict chemical- and tissue-
specific partition coefficients. The methods include both the default
PK-Sim approach and PK-Sim Standard and Rodgers &amp; Rowland (2006).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pksim.pcs
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Source</h3>

<p>Kapraun DF, Sfeir M, Pearce RG, Davidson-Fritz SE, Lumen A, Dallmann A, Judson RS, Wambaugh JF (2022).
&ldquo;Evaluation of a rapid, generic human gestational dose model.&rdquo;
<em>Reproductive Toxicology</em>, <b>113</b>, 172&ndash;188.
</p>


<h3>References</h3>

<p>Dallmann A, Ince I, Coboeken K, Eissing T, Hempel G (2018).
&ldquo;A physiologically based pharmacokinetic model for pregnant women to predict the pharmacokinetics of drugs metabolized via several enzymatic pathways.&rdquo;
<em>Clinical pharmacokinetics</em>, <b>57</b>(6), 749&ndash;768.
</p>

<hr>
<h2 id='pradeep2020'>Pradeep et al. 2020</h2><span id='topic+pradeep2020'></span><span id='topic+Pradeep2020'></span>

<h3>Description</h3>

<p>This table includes Support Vector Machine and Random Forest model predicted
values for unbound fraction plasma protein (fup) and intrinsic hepatic
clearance (clint) values for a subset of chemicals in the Tox21 library
(see <a href="https://www.epa.gov/chemical-research/toxicology-testing-21st-century-tox21">https://www.epa.gov/chemical-research/toxicology-testing-21st-century-tox21</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pradeep2020
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Details</h3>

<p>Prediction were made with Support Vector Machine and Random Forest models,
as reported in Pradeep et al. (2020).
</p>


<h3>References</h3>

<p>Pradeep P, Patlewicz G, Pearce R, Wambaugh J, Wetmore B, Judson R (2020).
&ldquo;Using chemical structure information to develop predictive models for in vitro toxicokinetic parameters to inform high-throughput risk-assessment.&rdquo;
<em>Computational Toxicology</em>, <b>16</b>, 100136.
ISSN 2468-1113, <a href="https://doi.org/10.1016/j.comtox.2020.100136">doi:10.1016/j.comtox.2020.100136</a>, <a href="https://www.sciencedirect.com/science/article/pii/S2468111320300463">https://www.sciencedirect.com/science/article/pii/S2468111320300463</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_pradeep2020">load_pradeep2020</a></code>
</p>

<hr>
<h2 id='predict_partitioning_schmitt'>Predict partition coefficients using the method from Schmitt (2008).</h2><span id='topic+predict_partitioning_schmitt'></span>

<h3>Description</h3>

<p>This function implements the method from Schmitt (2008) for predicting the 
tissue to unbound plasma partition coefficients for the tissues contained 
in the <code><a href="#topic+tissue.data">tissue.data</a></code> table. The method has been modified
by Pearce et al. (2017) based on an evaluation using in vivo measured 
partition coefficients.
</p>
<p>To understand this method, it is 
important to recognize that in  a given media the fraction unbound in that 
media is inverse of the media:water partition coefficient. 
In Schmitt's model, each tissue is composed of cells and
interstitium, with each cell consisting of neutral lipid,
neutral phospholipid, water, protein, and acidic phospholipid.
Each tissue cell is defined as the sum of separate
compartments for each constituent, all of which partition
with a shared water compartment. The partitioning between
the cell components and cell water is compound specific
and determined by log Pow (in neutral lipid partitioning),
membrane affinity (phospholipid and protein partitioning),
and pKa (neutral lipid and acidic phospholipid partitioning).
For a given compound the partitioning into each
component is identical across tissues. Thus the differences
among tissues are driven by their composition, that is, the
varying volumes of components such as neutral lipid.
However, pH differences across tissues also determine
small differences in partitioning between cell and plasma
water. The fup is used as the plasma water to total plasma
partition coefficient and to approximate the partitioning
between interstitial protein and water.
</p>
<p>A regression is used to predict membrane affinity when measured values are 
not available (<code><a href="#topic+calc_ma">calc_ma</a></code>).  The
regressions for correcting each tissue are performed on tissue plasma
partition coefficients (Ktissue2pu * Funbound.plasma) calculated with the
corrected Funbound.plasma value and divided by this value to get Ktissue2pu.
Thus the regressions should be used with the corrected Funbound.plasma.
</p>
<p>A separate regression is used when adjusted.Funbound.plasma is FALSE.
</p>
<p>The red blood cell regression can be used but is not by default because of
the span of the data used for evaluation, reducing confidence in the 
regression for higher and lower predicted values.
</p>
<p>Human tissue volumes are used for species other than Rat.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_partitioning_schmitt(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  species = "Human",
  model = "pbtk",
  default.to.human = FALSE,
  parameters = NULL,
  alpha = 0.001,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  regression.list = c("brain", "adipose", "gut", "heart", "kidney", "liver", "lung",
    "muscle", "skin", "spleen", "bone"),
  tissues = NULL,
  minimum.Funbound.plasma = 1e-04,
  suppress.messages = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_partitioning_schmitt_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name or the CAS number must be
specified.</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_model">model</code></td>
<td>
<p>Model for which partition coefficients are neeeded (for example,
&quot;pbtk&quot;, &quot;3compartment&quot;)</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from <code><a href="#topic+parameterize_schmitt">parameterize_schmitt</a></code>
overrides chem.name, dtxsid, and chem.cas.</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_alpha">alpha</code></td>
<td>
<p>Ratio of Distribution coefficient D of totally charged species
and that of the neutral form</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Whether or not to use Funbound.plasma
adjustment.</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions.  Regressions are
used by default.</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_regression.list">regression.list</code></td>
<td>
<p>Tissues to use regressions on.</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_tissues">tissues</code></td>
<td>
<p>Vector of desired partition coefficients.  Returns all by
default.</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
<tr><td><code id="predict_partitioning_schmitt_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns tissue to unbound plasma partition coefficients for each
tissue.
</p>


<h3>Author(s)</h3>

<p>Robert Pearce
</p>


<h3>References</h3>

<p>Schmitt, Walter. &quot;General approach for the calculation of tissue to plasma 
partition coefficients.&quot; Toxicology in Vitro 22.2 (2008): 457-467.
</p>
<p>Birnbaum, L., et al. &quot;Physiological parameter values for PBPK models.&quot; 
International Life Sciences Institute, Risk Science Institute, Washington, 
DC (1994).
</p>
<p>Pearce, Robert G., et al. &quot;Evaluation and calibration of high-throughput 
predictions of chemical distribution to tissues.&quot; Journal of pharmacokinetics 
and pharmacodynamics 44.6 (2017): 549-565.
</p>
<p>Yun, Y. E., and A. N. Edginton. &quot;Correlation-based prediction of 
tissue-to-plasma partition coefficients using readily available input 
parameters.&quot; Xenobiotica 43.10 (2013): 839-852.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+parameterize_schmitt">parameterize_schmitt</a></code>
</p>
<p><code><a href="#topic+tissue.data">tissue.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
predict_partitioning_schmitt(chem.name='ibuprofen',regression=FALSE)

</code></pre>

<hr>
<h2 id='pregnonpregaucs'>AUCs for Pregnant and Non-Pregnant Women</h2><span id='topic+pregnonpregaucs'></span>

<h3>Description</h3>

<p>Dallmann et al. (2018) includes compiled literature descriptions of 
toxicokinetic summary statistics, including time-integrated plasma 
concentrations (area under the curve or AUC) for drugs administered to a 
sample of subjects including both pregnant and non-pregnant women. The 
circumstances of the dosing varied slightly between drugs and are summarized 
in the table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pregnonpregaucs
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Source</h3>

<p>Kapraun DF, Sfeir M, Pearce RG, Davidson-Fritz SE, Lumen A, Dallmann A, Judson RS, Wambaugh JF (2022).
&ldquo;Evaluation of a rapid, generic human gestational dose model.&rdquo;
<em>Reproductive Toxicology</em>, <b>113</b>, 172&ndash;188.
</p>


<h3>References</h3>

<p>Dallmann A, Ince I, Coboeken K, Eissing T, Hempel G (2018).
&ldquo;A physiologically based pharmacokinetic model for pregnant women to predict the pharmacokinetics of drugs metabolized via several enzymatic pathways.&rdquo;
<em>Clinical pharmacokinetics</em>, <b>57</b>(6), 749&ndash;768.
</p>

<hr>
<h2 id='propagate_invitrouv_1comp'>Propagates uncertainty and variability in in vitro HTTK data into one
compartment model parameters</h2><span id='topic+propagate_invitrouv_1comp'></span>

<h3>Description</h3>

<p>Propagates uncertainty and variability in in vitro HTTK data into one
compartment model parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>propagate_invitrouv_1comp(parameters.dt, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="propagate_invitrouv_1comp_+3A_parameters.dt">parameters.dt</code></td>
<td>
<p>The data table of parameters being used by the Monte
Carlo sampler</p>
</td></tr>
<tr><td><code id="propagate_invitrouv_1comp_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+calc_elimination_rate">calc_elimination_rate</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table whose columns are the parameters of the HTTK model
specified in <code>model</code>.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>

<hr>
<h2 id='propagate_invitrouv_3comp'>Propagates uncertainty and variability in in vitro HTTK data into three
compartment model parameters</h2><span id='topic+propagate_invitrouv_3comp'></span>

<h3>Description</h3>

<p>Propagates uncertainty and variability in in vitro HTTK data into three
compartment model parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>propagate_invitrouv_3comp(parameters.dt, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="propagate_invitrouv_3comp_+3A_parameters.dt">parameters.dt</code></td>
<td>
<p>The data table of parameters being used by the Monte
Carlo sampler</p>
</td></tr>
<tr><td><code id="propagate_invitrouv_3comp_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+calc_hep_clearance">calc_hep_clearance</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table whose columns are the parameters of the HTTK model
specified in <code>model</code>.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>

<hr>
<h2 id='propagate_invitrouv_pbtk'>Propagates uncertainty and variability in in vitro HTTK data into PBPK
model parameters</h2><span id='topic+propagate_invitrouv_pbtk'></span>

<h3>Description</h3>

<p>Propagates uncertainty and variability in in vitro HTTK data into PBPK
model parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>propagate_invitrouv_pbtk(parameters.dt, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="propagate_invitrouv_pbtk_+3A_parameters.dt">parameters.dt</code></td>
<td>
<p>The data table of parameters being used by the Monte
Carlo sampler</p>
</td></tr>
<tr><td><code id="propagate_invitrouv_pbtk_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+calc_hep_clearance">calc_hep_clearance</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table whose columns are the parameters of the HTTK model
specified in <code>model</code>.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>

<hr>
<h2 id='r_left_censored_norm'>Returns draws from a normal distribution with a lower censoring limit of lod
(limit of detection)</h2><span id='topic+r_left_censored_norm'></span>

<h3>Description</h3>

<p>Returns draws from a normal distribution with a lower censoring limit of lod
(limit of detection)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>r_left_censored_norm(n, mean = 0, sd = 1, lod = 0.005, lower = 0, upper = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="r_left_censored_norm_+3A_n">n</code></td>
<td>
<p>Number of samples to take</p>
</td></tr>
<tr><td><code id="r_left_censored_norm_+3A_mean">mean</code></td>
<td>
<p>Mean of censored distribution. Default 0.</p>
</td></tr>
<tr><td><code id="r_left_censored_norm_+3A_sd">sd</code></td>
<td>
<p>Standard deviation of censored distribution. Default 1.</p>
</td></tr>
<tr><td><code id="r_left_censored_norm_+3A_lod">lod</code></td>
<td>
<p>Bound below which to censor. Default 0.005.</p>
</td></tr>
<tr><td><code id="r_left_censored_norm_+3A_lower">lower</code></td>
<td>
<p>Lower bound on censored distribution. Default 0.</p>
</td></tr>
<tr><td><code id="r_left_censored_norm_+3A_upper">upper</code></td>
<td>
<p>Upper bound on censored distribution. Default 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of samples from the specified censored distribution.
</p>

<hr>
<h2 id='reset_httk'>Reset HTTK to Default Data Tables</h2><span id='topic+reset_httk'></span>

<h3>Description</h3>

<p>This function returns an updated version of chem.physical_and_invitro.data
that includes data predicted with Simulations Plus' ADMET predictor that was
used in Sipes et al. 2017, included in admet.data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reset_httk(target.env = .GlobalEnv)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reset_httk_+3A_target.env">target.env</code></td>
<td>
<p>The environment where the new
chem.physical_and_invitro.data is loaded. Defaults to global environment.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>data.frame</code></td>
<td>
<p>The package default version of 
chem.physical_and_invitro.data.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

chem.physical_and_invitro.data &lt;- load_sipes2017()
reset_httk()
                        

</code></pre>

<hr>
<h2 id='rfun'>Randomly draws from a one-dimensional KDE</h2><span id='topic+rfun'></span>

<h3>Description</h3>

<p>Randomly draws from a one-dimensional KDE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rfun(n, fhat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rfun_+3A_n">n</code></td>
<td>
<p>Number of samples to draw</p>
</td></tr>
<tr><td><code id="rfun_+3A_fhat">fhat</code></td>
<td>
<p>A list with elements x, w, and h (h is the KDE bandwidth).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of n samples from the KDE fhat
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='rmed0non0u95'>Draw random numbers with LOD median but non-zero upper 95th percentile</h2><span id='topic+rmed0non0u95'></span>

<h3>Description</h3>

<p>This function draws N random numbers from a distribution that approximates
a median that is equal to the limit of detection (LOD, value x.LOD) but has 
an upper 95th percentile (x.u95) that is above x.LOD. We make the assumption 
that values above x.u95 are uniformly distributed between x.u95 and x.u95 +
(x.u95 - x.LOD)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmed0non0u95(n, x.u95, x.min = 0, x.LOD = 0.005)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rmed0non0u95_+3A_n">n</code></td>
<td>
<p>Number of samples to draw</p>
</td></tr>
<tr><td><code id="rmed0non0u95_+3A_x.u95">x.u95</code></td>
<td>
<p>The upper limit on the 95th confidence/credible intervale (this
is the 97.5 percentile)</p>
</td></tr>
<tr><td><code id="rmed0non0u95_+3A_x.min">x.min</code></td>
<td>
<p>The minimum allowed value (defaults to 0)</p>
</td></tr>
<tr><td><code id="rmed0non0u95_+3A_x.lod">x.LOD</code></td>
<td>
<p>The limit of detection (defaults to 0.005)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of N samples where the 50th and 97.5th quantiles approximate
x.LOD and x.u95 respectively
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>References</h3>

<p>Breen M, Wambaugh JF, Bernstein A, Sfeir M, Ring CL (2022).
&ldquo;Simulating toxicokinetic variability to identify susceptible and highly exposed populations.&rdquo;
<em>Journal of Exposure Science &amp; Environmental Epidemiology</em>, <b>32</b>(6), 855&ndash;863.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Fup.95 &lt;- 0.02
N &lt;- 1000

set.seed(1235)
Fup.vec &lt;- rmed0non0u95(n=N, x.u95=Fup.95)
quantile(Fup.vec,c(0.5,0.975))

quantile(rmed0non0u95(200,x.u95=0.05,x.min=10^-4,x.LOD=0.01),c(0.5,0.975))
hist(rmed0non0u95(1000,x.u95=0.05,x.min=10^-4,x.LOD=0.01))

quantile(rmed0non0u95(200,x.u95=0.005,x.min=10^-4,x.LOD=0.01),c(0.5,0.975))
hist(rmed0non0u95(1000,x.u95=0.005,x.min=10^-4,x.LOD=0.01))

</code></pre>

<hr>
<h2 id='scale_dosing'>Scale mg/kg body weight doses according to body weight and units</h2><span id='topic+scale_dosing'></span>

<h3>Description</h3>

<p>This function transforms the dose (in mg/kg) into the appropriate units. It
handles single doses, matrices of doses, or daily repeated doses at varying
intervals. Gut absorption is also factored in through the parameter Fabsgut,
and scaling is currently avoided in the inhalation exposure case with a 
scale factor of 1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_dosing(
  dosing,
  parameters,
  route,
  input.units = NULL,
  output.units = "uM",
  vol = NULL,
  state = "liquid"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scale_dosing_+3A_dosing">dosing</code></td>
<td>
<p>List of dosing metrics used in simulation, which must include
the general entries with names &quot;initial.dose&quot;, &quot;doses.per.day&quot;, 
&quot;daily.dose&quot;, and &quot;dosing.matrix&quot;. The &quot;dosing.matrix&quot; is used for more
precise dose regimen specification, and is a matrix consisting of two
columns or rows named &quot;time&quot; and &quot;dose&quot; containing the time and amount,
in mg/kg BW, of each dose. The minimal usage case involves all entries but
&quot;initial.dose&quot; set to NULL in value.</p>
</td></tr>
<tr><td><code id="scale_dosing_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_pbtk function,
overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="scale_dosing_+3A_route">route</code></td>
<td>
<p>String specification of route of exposure for simulation:
&quot;oral&quot;, &quot;iv&quot;, &quot;inhalation&quot;, ...</p>
</td></tr>
<tr><td><code id="scale_dosing_+3A_input.units">input.units</code></td>
<td>
<p>Units of the dose values being scaled. (Default is NULL.) 
Currently supported units &quot;mg/L&quot;, &quot;ug/L&quot;,&quot;ug/mL&quot;, &quot;uM&quot;, &quot;umol/L&quot;, &quot;ug/dL&quot;,
&quot;ug/g&quot;, &quot;nmol/L&quot;, &quot;nM&quot;, and &quot;ppmw&quot; (supported input.units subject to change).</p>
</td></tr>
<tr><td><code id="scale_dosing_+3A_output.units">output.units</code></td>
<td>
<p>Desired units (either &quot;mg/L&quot;, &quot;mg&quot;, &quot;umol&quot;, or default
&quot;uM&quot;).</p>
</td></tr>
<tr><td><code id="scale_dosing_+3A_vol">vol</code></td>
<td>
<p>Volume for the target tissue of interest.
NOTE: Volume should not be in units of per BW, i.e. &quot;kg&quot;.</p>
</td></tr>
<tr><td><code id="scale_dosing_+3A_state">state</code></td>
<td>
<p>Chemical state of matter (gas or default liquid).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of numeric values for doses converted to output.units, potentially
(depending on argument dosing) including:
</p>
<table>
<tr><td><code>initial.dose</code></td>
<td>
<p>The first dose given</p>
</td></tr>
<tr><td><code>dosing.matrix</code></td>
<td>
<p>A 2xN matrix where the first column is dose time and
the second is dose amount for N doses</p>
</td></tr>
<tr><td><code>daily.dose</code></td>
<td>
<p>The total cumulative daily dose</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Wambaugh and Sarah E. Davidson
</p>

<hr>
<h2 id='scr_h'>KDE bandwidths for residual variability in serum creatinine</h2><span id='topic+scr_h'></span>

<h3>Description</h3>

<p>Bandwidths used for a one-dimensional kernel density estimation of the
distribution of residual errors around smoothing spline fits of serum
creatinine vs. age  for NHANES respondents in each of ten combinations of sex
and race/ethnicity categories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scr_h
</code></pre>


<h3>Format</h3>

<p>A named list with 10 elements, each a numeric value. Each list element
corresponds to, and is named for, one combination of NHANES sex categories
(Male and Female) and NHANES race/ethnicity categories (Mexican American,
Other Hispanic, Non-Hispanic White, Non-Hispanic Black, and Other).
</p>


<h3>Details</h3>

<p>Each matrix is the standard deviation for a normal distribution: this is the
bandwidth to be used for a kernel density estimation (KDE) (using a normal
kernel) of the distribution of residual errors around smoothing spline fits of
serum creatinine vs. age for NHANES respondents in the specified sex and
race/ethnicity category. Optimal bandwidths were pre-calculated by doing the
smoothing spline fits, getting the residuals, then calling
<code><a href="ks.html#topic+kde">kde</a></code> on the residuals (which calls <code><a href="ks.html#topic+hpi">hpi</a></code> to
compute the plug-in bandwidth).
</p>
<p>Used by HTTK-Pop only in &quot;virtual individuals&quot; mode (i.e.
<code><a href="#topic+httkpop_generate">httkpop_generate</a></code> with <code>method = "v"</code>), in
<code><a href="#topic+gen_serum_creatinine">gen_serum_creatinine</a></code>.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='set_httk_precision'>set_httk_precision</h2><span id='topic+set_httk_precision'></span>

<h3>Description</h3>

<p>Although the ODE solver and other functions return very precise numbers,
we cannot (or at least do not spend enough computing time to) be sure of the 
precioion to an arbitrary level. This function both limits the number of
signficant figures reported and truncates the numerical precision.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_httk_precision(in.num, sig.fig = 4, num.prec = 9)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_httk_precision_+3A_in.num">in.num</code></td>
<td>
<p>The numeric variable (or assembly of numerics) to be
processed.</p>
</td></tr>
<tr><td><code id="set_httk_precision_+3A_sig.fig">sig.fig</code></td>
<td>
<p>The number of significant figures reported. Defaults to 4.</p>
</td></tr>
<tr><td><code id="set_httk_precision_+3A_num.prec">num.prec</code></td>
<td>
<p>The precision maintained, digits below 10^num.prec are 
dropped. Defaults to 9.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric values
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>

<hr>
<h2 id='sipes2017'>Sipes et al. 2017 data</h2><span id='topic+sipes2017'></span><span id='topic+Sipes2017'></span>

<h3>Description</h3>

<p>This table includes in silico predicted chemical-specifc plasma protein 
unbound fraction (fup) and intrinsic hepatic clearance values for the entire
Tox21 library 
(see <a href="https://www.epa.gov/chemical-research/toxicology-testing-21st-century-tox21">https://www.epa.gov/chemical-research/toxicology-testing-21st-century-tox21</a>). 
Predictions were made with Simulations Plus ADMET predictor,
as reported in Sipes et al. (2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sipes2017
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Nisha Sipes
</p>


<h3>Source</h3>

<p>ADMET, Simulations Plus
</p>


<h3>References</h3>

<p>Sipes NS, Wambaugh JF, Pearce R, Auerbach SS, Wetmore BA, Hsieh J, Shapiro AJ, Svoboda D, DeVito MJ, Ferguson SS (2017).
&ldquo;An intuitive approach for predicting potential human health risk with the Tox21 10k library.&rdquo;
<em>Environmental science &amp; technology</em>, <b>51</b>(18), 10786&ndash;10796.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_sipes2017">load_sipes2017</a></code>
</p>

<hr>
<h2 id='skeletal_muscle_mass'>Predict skeletal muscle mass</h2><span id='topic+skeletal_muscle_mass'></span>

<h3>Description</h3>

<p>Predict skeletal muscle mass from age, height, and gender.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>skeletal_muscle_mass(smm, age_years, height, gender)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="skeletal_muscle_mass_+3A_smm">smm</code></td>
<td>
<p>Vector of allometrically-scaled skeletal muscle masses.</p>
</td></tr>
<tr><td><code id="skeletal_muscle_mass_+3A_age_years">age_years</code></td>
<td>
<p>Vector of ages in years.</p>
</td></tr>
<tr><td><code id="skeletal_muscle_mass_+3A_height">height</code></td>
<td>
<p>Vector of heights in cm.</p>
</td></tr>
<tr><td><code id="skeletal_muscle_mass_+3A_gender">gender</code></td>
<td>
<p>Vector of genders, either 'Male' or 'Female.'</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For individuals over age 18, use allometrically-scaled muscle mass with an
age-based scaling factor, to account for loss of muscle mass with age
(Janssen et al. 2000). For individuals under age 18, use
<code><a href="#topic+skeletal_muscle_mass_children">skeletal_muscle_mass_children</a></code>.
</p>


<h3>Value</h3>

<p>Vector of skeletal muscle masses in kg.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Janssen, Ian, et al. &quot;Skeletal muscle mass and distribution in 468 men and
women aged 18-88 yer.&quot; Journal of Applied Physiology 89.1 (2000): 81-88
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+skeletal_muscle_mass_children">skeletal_muscle_mass_children</a></code>
</p>

<hr>
<h2 id='skeletal_muscle_mass_children'>Predict skeletal muscle mass for children</h2><span id='topic+skeletal_muscle_mass_children'></span>

<h3>Description</h3>

<p>For individuals under age 18, predict skeletal muscle mass from gender and
age, using a nonlinear equation from Webber and Barr (2012)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>skeletal_muscle_mass_children(gender, age_years)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="skeletal_muscle_mass_children_+3A_gender">gender</code></td>
<td>
<p>Vector of genders (either 'Male' or 'Female').</p>
</td></tr>
<tr><td><code id="skeletal_muscle_mass_children_+3A_age_years">age_years</code></td>
<td>
<p>Vector of ages in years.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of skeletal muscle masses in kg.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Webber, Colin E., and Ronald D. Barr. &quot;Age-and gender-dependent values of 
skeletal muscle mass in healthy children and adolescents.&quot; Journal of 
cachexia, sarcopenia and muscle 3.1 (2012): 25-29.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='skin_mass_bosgra'>Predict skin mass</h2><span id='topic+skin_mass_bosgra'></span>

<h3>Description</h3>

<p>Using equation from Bosgra et al. 2012, predict skin mass from body surface
area.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>skin_mass_bosgra(BSA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="skin_mass_bosgra_+3A_bsa">BSA</code></td>
<td>
<p>Vector of body surface areas in cm^2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of skin masses in kg.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Bosgra, Sieto, et al. &quot;An improved model to predict 
physiologically based model parameters and their inter-individual variability 
from anthropometry.&quot; Critical reviews in toxicology 42.9 (2012): 751-767.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='solve_1comp'>Solve one compartment TK model</h2><span id='topic+solve_1comp'></span>

<h3>Description</h3>

<p>This function solves for the amount or concentration of a chemical in plasma
for a one compartment model as a function of time based on the dose and
dosing frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solve_1comp(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  times = NULL,
  parameters = NULL,
  days = 10,
  tsteps = 4,
  daily.dose = NULL,
  dose = NULL,
  doses.per.day = NULL,
  initial.values = NULL,
  plots = FALSE,
  suppress.messages = FALSE,
  species = "Human",
  iv.dose = FALSE,
  input.units = "mg/kg",
  output.units = NULL,
  method = "lsoda",
  rtol = 1e-08,
  atol = 1e-12,
  default.to.human = FALSE,
  recalc.blood2plasma = FALSE,
  recalc.clearance = FALSE,
  dosing.matrix = NULL,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = TRUE,
  minimum.Funbound.plasma = 1e-04,
  monitor.vars = NULL,
  Caco2.options = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="solve_1comp_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters
must be specified.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must
be specified.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_times">times</code></td>
<td>
<p>Optional time sequence for specified number of days.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_1comp function,
overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_days">days</code></td>
<td>
<p>Length of the simulation.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_tsteps">tsteps</code></td>
<td>
<p>The number time steps per hour.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose, default is mg/kg BW.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_dose">dose</code></td>
<td>
<p>Amount of a single dose, default is mg/kg BW.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_doses.per.day">doses.per.day</code></td>
<td>
<p>Number of doses per day.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_initial.values">initial.values</code></td>
<td>
<p>Vector containing the initial concentrations or
amounts of the chemical in specified tissues with units corresponding to
output.units.  Defaults are zero.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_plots">plots</code></td>
<td>
<p>Plots all outputs if true.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, or default
&quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_iv.dose">iv.dose</code></td>
<td>
<p>Simulates a single i.v. dose if true.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_input.units">input.units</code></td>
<td>
<p>Input units of interest assigned to dosing, defaults to
&quot;mg/kg&quot; BW.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_output.units">output.units</code></td>
<td>
<p>A named vector of output units expected for the model
results. Default, NULL, returns model results in units specified in the
'modelinfo' file. See table below for details.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_method">method</code></td>
<td>
<p>Method used by integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_rtol">rtol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_atol">atol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing rat values with human values if
true.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_recalc.blood2plasma">recalc.blood2plasma</code></td>
<td>
<p>Whether or not to recalculate the blood:plasma
chemical concentrationr ratio</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_recalc.clearance">recalc.clearance</code></td>
<td>
<p>Whether or not to recalculate the elimination
rate.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_dosing.matrix">dosing.matrix</code></td>
<td>
<p>Vector of dosing times or a matrix consisting of two
columns or rows named &quot;dose&quot; and &quot;time&quot; containing the time and amount, in
mg/kg BW by default, of each dose.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with volume of distribution calculated with this value.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients in volume of distribution calculation.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>In calculating elimination rate, protein
binding is not taken into account (set to 1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_monitor.vars">monitor.vars</code></td>
<td>
<p>Which variables are returned as a function of time. 
Defaults value of NULL provides &quot;Agutlumen&quot;, &quot;Ccompartment&quot;, &quot;Ametabolized&quot;,
&quot;AUC&quot;</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
<tr><td><code id="solve_1comp_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the integrator.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the timescales for the model parameters have units of hours while 
the model output is in days.
</p>
<p>Default value of NULL for doses.per.day solves for a single dose.
</p>
<p>When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitutes human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>
<p>AUC is area under plasma concentration curve.
</p>
<p>Model Figure 
<img src="../help/figures/1comp.png" width="60%" alt="Figure: One Compartment Model Schematic" />

</p>


<h3>Value</h3>

<p>A matrix with a column for time(in days) and a column for the
compartment and the area under the curve (concentration only).
</p>


<h3>Author(s)</h3>

<p>Robert Pearce
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+solve_model">solve_model</a></code>
</p>
<p><code><a href="#topic+parameterize_1comp">parameterize_1comp</a></code>
</p>
<p><code><a href="#topic+calc_analytic_css_1comp">calc_analytic_css_1comp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
solve_1comp(chem.name='Bisphenol-A', days=1)

# By storing the model parameters in a vector first, you can potentially
# edit them before using the model:
params &lt;- parameterize_1comp(chem.cas="80-05-7")
solve_1comp(parameters=params, days=1)

head(solve_1comp(chem.name="Terbufos", daily.dose=NULL, dose=1, days=1))
head(solve_1comp(chem.name="Terbufos", daily.dose=NULL,
                 dose=1,days=1, iv.dose=TRUE))

# A dose matrix specifies times and magnitudes of doses:
dm &lt;- matrix(c(0,1,2,5,5,5),nrow=3)
colnames(dm) &lt;- c("time","dose")
solve_1comp(chem.name="Methenamine", dosing.matrix=dm,
            days=2.5, dose=NULL,daily.dose=NULL)

solve_1comp(chem.name="Besonprodil", daily.dose=1, dose=NULL,
            days=2.5, doses.per.day=4)

</code></pre>

<hr>
<h2 id='solve_3comp'>Solve_3comp</h2><span id='topic+solve_3comp'></span>

<h3>Description</h3>

<p>This function solves for the amounts or concentrations of a chemical in
different tissues as functions of time based on the dose and dosing
frequency.  It uses a three compartment model with partition coefficients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solve_3comp(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  times = NULL,
  parameters = NULL,
  days = 10,
  tsteps = 4,
  daily.dose = NULL,
  dose = NULL,
  doses.per.day = NULL,
  initial.values = NULL,
  plots = FALSE,
  suppress.messages = FALSE,
  species = "Human",
  iv.dose = FALSE,
  input.units = "mg/kg",
  output.units = NULL,
  method = "lsoda",
  rtol = 1e-08,
  atol = 1e-12,
  default.to.human = FALSE,
  recalc.blood2plasma = FALSE,
  recalc.clearance = FALSE,
  clint.pvalue.threshold = 0.05,
  dosing.matrix = NULL,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = TRUE,
  minimum.Funbound.plasma = 1e-04,
  Caco2.options = list(),
  monitor.vars = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="solve_3comp_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters
must be specified.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must
be specified.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's 'DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_times">times</code></td>
<td>
<p>Optional time sequence for specified number of days.  The
dosing sequence begins at the beginning of times.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_3comp function,
overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_days">days</code></td>
<td>
<p>Length of the simulation.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_tsteps">tsteps</code></td>
<td>
<p>The number time steps per hour.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose, mg/kg BW.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_dose">dose</code></td>
<td>
<p>Amount of a single dose, mg/kg BW.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_doses.per.day">doses.per.day</code></td>
<td>
<p>Number of doses per day.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_initial.values">initial.values</code></td>
<td>
<p>Vector containing the initial concentrations or
amounts of the chemical in specified tissues with units corresponding to
output.units.  Defaults are zero.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_plots">plots</code></td>
<td>
<p>Plots all outputs if true.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_iv.dose">iv.dose</code></td>
<td>
<p>Simulates a single i.v. dose if true.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_input.units">input.units</code></td>
<td>
<p>Input units of interest assigned to dosing,
defaults to mg/kg BW</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_output.units">output.units</code></td>
<td>
<p>A named vector of output units expected for the model
results. Default, NULL, returns model results in units specified in the
'modelinfo' file. See table below for details.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_method">method</code></td>
<td>
<p>Method used by integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_rtol">rtol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_atol">atol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_recalc.blood2plasma">recalc.blood2plasma</code></td>
<td>
<p>Recalculates the ratio of the amount of chemical
in the blood to plasma using the input parameters, calculated with
hematocrit, Funbound.plasma, and Krbc2pu.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_recalc.clearance">recalc.clearance</code></td>
<td>
<p>Recalculates the the hepatic clearance
(Clmetabolism) with new million.cells.per.gliver parameter.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_clint.pvalue.threshold">clint.pvalue.threshold</code></td>
<td>
<p>Hepatic clearances with clearance assays
having p-values greater than the threshold are set to zero.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_dosing.matrix">dosing.matrix</code></td>
<td>
<p>Vector of dosing times or a matrix consisting of two
columns or rows named &quot;dose&quot; and &quot;time&quot; containing the time and amount, in
mg/kg BW, of each dose.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to
1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_monitor.vars">monitor.vars</code></td>
<td>
<p>Which variables are returned as a function of time. 
Defaults value of NULL provides &quot;Cliver&quot;, &quot;Csyscomp&quot;, &quot;Atubules&quot;, 
&quot;Ametabolized&quot;, &quot;AUC&quot;</p>
</td></tr>
<tr><td><code id="solve_3comp_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the integrator.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the timescales for the model parameters have units of hours while 
the model output is in days.
</p>
<p>Default of NULL for doses.per.day solves for a single dose.
</p>
<p>The compartments used in this model are the gutlumen, gut, liver, and
rest-of-body, with the plasma equivalent to the liver plasma.
</p>
<p>Model Figure 
<img src="../help/figures/3comp.jpg" width="60%" alt="Figure: Three Compartment Model Schematic" /> 

</p>
<p>When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitues human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>


<h3>Value</h3>

<p>A matrix of class deSolve with a column for time(in days) and each
compartment, the plasma concentration, area under the curve, and a row for
each time point.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+solve_model">solve_model</a></code>
</p>
<p><code><a href="#topic+parameterize_3comp">parameterize_3comp</a></code>
</p>
<p><code><a href="#topic+calc_analytic_css_3comp">calc_analytic_css_3comp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
solve_3comp(chem.name='Bisphenol-A', 
            doses.per.day=2, 
            daily.dose=.5,
            days=1,
            tsteps=2)

# By storing the model parameters in a vector first, you can potentially
# edit them before using the model:
params &lt;-parameterize_3comp(chem.cas="80-05-7")
solve_3comp(parameters=params, days=1)

head(solve_3comp(chem.name="Terbufos", daily.dose=NULL, dose=1, days=1))
head(solve_3comp(chem.name="Terbufos", daily.dose=NULL, dose=1, 
                 days=1, iv.dose=TRUE))

# A dose matrix specifies times and magnitudes of doses:
dm &lt;- matrix(c(0,1,2,5,5,5),nrow=3)
colnames(dm) &lt;- c("time","dose")
solve_3comp(chem.name="Methenamine", dosing.matrix=dm,
            dose=NULL, daily.dose=NULL,
            days=2.5)

solve_3comp(chem.name="Besonprodil",
            daily.dose=1, dose=NULL,
            days=2.5, doses.per.day=4)

</code></pre>

<hr>
<h2 id='solve_fetal_pbtk'>Solve_fetal_PBTK</h2><span id='topic+solve_fetal_pbtk'></span>

<h3>Description</h3>

<p>This function solves for the amounts or concentrations in uM of a chemical
in different tissues of a maternofetal system as functions of time based on
the dose and dosing frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solve_fetal_pbtk(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  times = seq(13 * 7, 40 * 7, 1),
  parameters = NULL,
  days = NULL,
  species = "human",
  tsteps = 4,
  dose = NULL,
  dosing.matrix = NULL,
  daily.dose = NULL,
  doses.per.day = NULL,
  initial.values = NULL,
  plots = FALSE,
  suppress.messages = FALSE,
  iv.dose = FALSE,
  input.units = "mg/kg",
  output.units = NULL,
  method = "lsoda",
  rtol = 1e-08,
  atol = 1e-12,
  default.to.human = FALSE,
  recalc.blood2plasma = FALSE,
  recalc.clearance = FALSE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = TRUE,
  minimum.Funbound.plasma = 1e-04,
  monitor.vars = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="solve_fetal_pbtk_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters
must be specified.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must
be specified.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_times">times</code></td>
<td>
<p>Optional time sequence in days. Dosing sequence begins at the
beginning of times. Default is from 13th week of pregnancy to 40th due to 
data constraints.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_fetal_pbtk function,
overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_days">days</code></td>
<td>
<p>Length of the simulation.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_species">species</code></td>
<td>
<p>Included for compatibility with other functions, but the model
will not run for non-human species (default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_tsteps">tsteps</code></td>
<td>
<p>The number time steps per hour. Default of 4.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_dose">dose</code></td>
<td>
<p>Amount of a single, initial oral dose in mg/kg BW.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_dosing.matrix">dosing.matrix</code></td>
<td>
<p>A matrix of either one column (or row) with a set of
dosing times or with two columns (or rows) correspondingly named &quot;dose&quot; and
&quot;time&quot; containing the time and amount, in mg/kg BW, of each dose.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose, mg/kg BW.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_doses.per.day">doses.per.day</code></td>
<td>
<p>Number of doses per day.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_initial.values">initial.values</code></td>
<td>
<p>Vector containing the initial concentrations or
amounts of the chemical in specified tissues with units corresponding to
compartment.units.  Defaults are zero.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_plots">plots</code></td>
<td>
<p>Plots all outputs if true.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_iv.dose">iv.dose</code></td>
<td>
<p>Simulates a single i.v. dose if true.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_input.units">input.units</code></td>
<td>
<p>Input units of interest assigned to dosing, defaults to
mg/kg BW</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_output.units">output.units</code></td>
<td>
<p>A named vector of output units expected for the model
results. Default, NULL, returns model results in units specified in the
'modelinfo' file. See table below for details.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_method">method</code></td>
<td>
<p>Method used by integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_rtol">rtol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_atol">atol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_recalc.blood2plasma">recalc.blood2plasma</code></td>
<td>
<p>Recalculates the ratio of the amount of chemical
in the blood to plasma using the input parameters, calculated with
hematocrit, Funbound.plasma, and Krbc2pu.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_recalc.clearance">recalc.clearance</code></td>
<td>
<p>Recalculates the the hepatic clearance
(Clmetabolism) with new million.cells.per.gliver parameter.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to
1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_monitor.vars">monitor.vars</code></td>
<td>
<p>Which variables to track by default</p>
</td></tr>
<tr><td><code id="solve_fetal_pbtk_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the integrator.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The stage of pregnancy simulated here begins by default at the 13th week due
to a relative lack of data to support parameterization prior, in line with 
the recommendations of Kapraun et al. 2019 (&quot;Empirical models for anatomical
and physiological...&quot;), and ends at the 40th week of pregnancy.
</p>
<p>Note that the model parameters have units of hours while the model output is
in days. Dose is in mg, not scaled for body weight.
</p>
<p>Default NULL value for doses.per.day solves for a single dose.
</p>
<p>The maternal compartments used in this model are the gut lumen, gut, liver, 
venous blood, arterial blood, lung, adipose tissue, kidney, thyroid, 
and rest of body. A placenta is modeled as a joint organ shared by mother
and fetus, through which chemical exchange can occur with the fetus. Fetal
compartments include arterial blood, venous blood, kidney, thyroid, liver,
lung, gut, brain, and rest of body. 
</p>
<p>The extra compartments include the amounts or concentrations metabolized by
the liver and excreted by the kidneys through the tubules.
</p>
<p>AUC is the area under the curve of the plasma concentration.
</p>
<p>This gestational model is only parameterized for humans.
</p>


<h3>Value</h3>

<p>A matrix of class deSolve with a column for time(in days), each
compartment, the area under the curve, and plasma concentration and a row
for each time point.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh, Mark Sfeir, and Dustin Kapraun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+solve_model">solve_model</a></code>
</p>
<p><code><a href="#topic+parameterize_fetal_pbtk">parameterize_fetal_pbtk</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
out = solve_fetal_pbtk(chem.name = 'bisphenol a', daily.dose = 1,
doses.per.day = 3)

# With adjustement to fraction unbound plasma for fetus:
fetal_parms_fup_adjusted &lt;- 
  parameterize_fetal_pbtk(chem.name = 'perfluorooctane sulfonic acid')
head(solve_fetal_pbtk(parameters = fetal_parms_fup_adjusted))

# Without adjustement to fraction unbound plasma for fetus:
fetal_parms_fup_unadjusted &lt;-  
  parameterize_fetal_pbtk(chem.name = 'perfluorooctane sulfonic acid',
                                fetal_fup_adjustment = FALSE)
head(solve_fetal_pbtk(parameters = fetal_parms_fup_unadjusted))


</code></pre>

<hr>
<h2 id='solve_gas_pbtk'>solve_gas_pbtk</h2><span id='topic+solve_gas_pbtk'></span>

<h3>Description</h3>

<p>This function solves for the amounts or concentrations of a chemical
in different tissues as functions of time as a result of inhalation 
exposure to an ideal gas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solve_gas_pbtk(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  parameters = NULL,
  times = NULL,
  days = 10,
  tsteps = 4,
  daily.dose = NULL,
  doses.per.day = NULL,
  dose = NULL,
  dosing.matrix = NULL,
  forcings = NULL,
  exp.start.time = 0,
  exp.conc = 1,
  period = 24,
  exp.duration = 12,
  initial.values = NULL,
  plots = FALSE,
  suppress.messages = FALSE,
  species = "Human",
  iv.dose = FALSE,
  input.units = "ppmv",
  output.units = NULL,
  method = "lsoda",
  rtol = 1e-08,
  atol = 1e-12,
  default.to.human = FALSE,
  recalc.blood2plasma = FALSE,
  recalc.clearance = FALSE,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = TRUE,
  minimum.Funbound.plasma = 1e-04,
  monitor.vars = NULL,
  vmax = 0,
  km = 1,
  exercise = FALSE,
  fR = 12,
  VT = 0.75,
  VD = 0.15,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="solve_gas_pbtk_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters
must be specified.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must
be specified.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_gas_pbtk (or other
bespoke) function, overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_times">times</code></td>
<td>
<p>Optional time sequence for specified number of days.  Dosing
sequence begins at the beginning of times.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_days">days</code></td>
<td>
<p>Length of the simulation.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_tsteps">tsteps</code></td>
<td>
<p>The number of time steps per hour.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_doses.per.day">doses.per.day</code></td>
<td>
<p>Number of doses per day.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_dose">dose</code></td>
<td>
<p>Amount of a single dose</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_dosing.matrix">dosing.matrix</code></td>
<td>
<p>Vector of dosing times or a matrix consisting of two
columns or rows named &quot;dose&quot; and &quot;time&quot; containing the time and amount of 
each dose.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_forcings">forcings</code></td>
<td>
<p>Manual input of 'forcings' data series argument for ode
integrator. If left unspecified, 'forcings' defaults to NULL, and then other 
input parameters (see exp.start.time, exp.conc, exp.duration, and period)
provide the necessary information to assemble a forcings data series.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_exp.start.time">exp.start.time</code></td>
<td>
<p>Start time in specifying forcing exposure series,
default 0.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_exp.conc">exp.conc</code></td>
<td>
<p>Specified inhalation exposure concentration for use in 
assembling &quot;forcings&quot; data series argument for integrator. Defaults to
units of ppmv.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_period">period</code></td>
<td>
<p>For use in assembling forcing function data series 'forcings'
argument, specified in hours</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_exp.duration">exp.duration</code></td>
<td>
<p>For use in assembling forcing function data 
series 'forcings' argument, specified in hours</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_initial.values">initial.values</code></td>
<td>
<p>Vector containing the initial concentrations or
amounts of the chemical in specified tissues with units corresponding to
those specified for the model outputs. Default values are zero.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_plots">plots</code></td>
<td>
<p>Plots all outputs if true.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_iv.dose">iv.dose</code></td>
<td>
<p>Simulates a single i.v. dose if true.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_input.units">input.units</code></td>
<td>
<p>Input units of interest assigned to dosing, including 
forcings. Defaults to &quot;ppmv&quot; as applied to the default forcings scheme.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_output.units">output.units</code></td>
<td>
<p>A named vector of output units expected for the model
results. Default, NULL, returns model results in units specified in the
'modelinfo' file. See table below for details.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_method">method</code></td>
<td>
<p>Method used by integrator (deSolve).
(Note: There are precision differences between M1 Mac and other OS systems
for this function due to how long doubles are handled. To replicate results
between various OS systems we suggest changing the default method of &quot;lsoda&quot;
to &quot;lsode&quot; and also adding the argument mf = 10.
See [deSolve::ode()] for further details.)</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_rtol">rtol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_atol">atol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_recalc.blood2plasma">recalc.blood2plasma</code></td>
<td>
<p>Recalculates the ratio of the amount of chemical
in the blood to plasma using the input parameters, calculated with
hematocrit, Funbound.plasma, and Krbc2pu.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_recalc.clearance">recalc.clearance</code></td>
<td>
<p>Recalculates the hepatic clearance
(Clmetabolism) with new million.cells.per.gliver parameter.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to
1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_monitor.vars">monitor.vars</code></td>
<td>
<p>Which variables are returned as a function of time. 
Defaults value of NULL provides &quot;Cgut&quot;, &quot;Cliver&quot;, &quot;Cven&quot;, &quot;Clung&quot;, &quot;Cart&quot;,
&quot;Crest&quot;, &quot;Ckidney&quot;, &quot;Cplasma&quot;, &quot;Calv&quot;, &quot;Cendexh&quot;, &quot;Cmixexh&quot;, &quot;Cmuc&quot;, 
&quot;Atubules&quot;, &quot;Ametabolized&quot;, &quot;AUC&quot;</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_vmax">vmax</code></td>
<td>
<p>Michaelis-Menten vmax value in reactions/min</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_km">km</code></td>
<td>
<p>Michaelis-Menten concentration of half-maximal reaction velocity
in desired output concentration units.</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_exercise">exercise</code></td>
<td>
<p>Logical indicator of whether to simulate an exercise-induced
heightened respiration rate</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_fr">fR</code></td>
<td>
<p>Respiratory frequency (breaths/minute), used especially to adjust
breathing rate in the case of exercise. This parameter, along with VT and VD
(below) gives another option for calculating Qalv (Alveolar ventilation) 
in case pulmonary ventilation rate is not known</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_vt">VT</code></td>
<td>
<p>Tidal volume (L), to be modulated especially as part of simulating
the state of exercise</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_vd">VD</code></td>
<td>
<p>Anatomical dead space (L), to be modulated especially as part of
simulating the state of exercise</p>
</td></tr>
<tr><td><code id="solve_gas_pbtk_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the integrator.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default dosing scheme involves a specification of the start time
of exposure (exp.start.time), the concentration of gas inhaled (exp.conc),
the period of a cycle of exposure and non-exposure (period), the
duration of the exposure during that period (exp.duration), and the total
days simulated. Together,these arguments determine the &quot;forcings&quot; passed to
the ODE integrator. Forcings can also be specified manually, or effectively
turned off by setting exposure concentration to zero, if the user prefers to 
simulate dosing by other means. 
</p>
<p>The &quot;forcings&quot; object is configured to be passed to the integrator with,
at the most, a basic unit conversion among ppmv, mg/L, and uM. No scaling by
BW is set to be performed on the forcings series.
</p>
<p>Note that the model parameters have units of hours while the model output is
in days.
</p>
<p>Default NULL value for doses.per.day solves for a single dose.
</p>
<p>The compartments used in this model are the gut lumen, gut, liver, kidneys,
veins, arteries, lungs, and the rest of the body.
</p>
<p>The extra compartments include the amounts or concentrations metabolized by
the liver and excreted by the kidneys through the tubules.
</p>
<p>AUC is the area under the curve of the plasma concentration.
</p>
<p>Model Figure from (Linakis et al. 2020):
<img src="../help/figures/gaspbtk.jpg" width="100%" alt="Figure: Gas PBTK  Model Schematic" />

</p>
<p>Model parameters are named according to the following convention:</p>

<table>
<tr>
 <td style="text-align: left;">
prefix </td><td style="text-align: right;"> suffic </td><td style="text-align: right;"> Meaning </td><td style="text-align: right;"> units </td>
</tr>
<tr>
 <td style="text-align: left;">
K </td><td style="text-align: right;"> </td><td style="text-align: right;"> Partition coefficient for tissue to free plasma \ tab unitless </td>
</tr>
<tr>
 <td style="text-align: left;">
V </td><td style="text-align: right;"> </td><td style="text-align: right;"> Volume </td><td style="text-align: right;"> L </td>
</tr>
<tr>
 <td style="text-align: left;">
Q </td><td style="text-align: right;"> </td><td style="text-align: right;"> Flow </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: left;">
k </td><td style="text-align: right;"> </td><td style="text-align: right;"> Rate </td><td style="text-align: right;"> 1/h </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: right;"> c </td><td style="text-align: right;"> Parameter is proportional to body weight </td><td style="text-align: right;"> 1 / kg for volumes
and 1/kg^(3/4) for flows </td>
</tr>

</table>

<p>When species is specified but chemical-specific in vitro data are not
available, the function uses the appropriate physiological data (volumes and 
flows) but default.to.human = TRUE must be used to substitute human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>


<h3>Value</h3>

<p>A matrix of class deSolve with a column for time(in days), each
compartment, the area under the curve, and plasma concentration and a row
for each time point.
</p>


<h3>Author(s)</h3>

<p>Matt Linakis, John Wambaugh, Mark Sfeir, Miyuki Breen
</p>


<h3>References</h3>

<p>Linakis MW, Sayre RR, Pearce RG, Sfeir MA, Sipes NS, Pangburn HA, Gearhart JM, Wambaugh JF (2020).
&ldquo;Development and evaluation of a high-throughput inhalation model for organic chemicals.&rdquo;
<em>Journal of exposure science &amp; environmental epidemiology</em>, <b>30</b>(5), 866&ndash;877.
</p>
<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+solve_model">solve_model</a></code>
</p>
<p><code><a href="#topic+parameterize_gas_pbtk">parameterize_gas_pbtk</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

solve_gas_pbtk(chem.name = 'pyrene', exp.conc = 1, period = 24, expduration = 24)

out &lt;- solve_gas_pbtk(chem.name='pyrene',
                      exp.conc = 0, doses.per.day = 2,
                      daily.dose = 3, input.units = "umol",
                      days=2.5, 
                      plots=TRUE, initial.values=c(Aven=20))

out &lt;- solve_gas_pbtk(chem.name = 'pyrene', exp.conc = 3, 
                      period = 24, days=2.5,
                      exp.duration = 6, exercise = TRUE)
                  
params &lt;- parameterize_gas_pbtk(chem.cas="80-05-7")
solve_gas_pbtk(parameters=params, days=2.5)

# Oral dose with exhalation as a route of elimination:
out &lt;- solve_gas_pbtk(chem.name = 'bisphenol a', exp.conc = 0, dose=100,
                      days=2.5, input.units="mg/kg")

# Note that different model compartments for this model have different units 
# and that the final units can be controlled with the output.units argument:
head(solve_gas_pbtk(chem.name="lindane", days=2.5))
# Convert all compartment units to mg/L:
head(solve_gas_pbtk(chem.name="lindane", days=2.5, output.units="mg/L"))
# Convert just the plasma to mg/L:
head(solve_gas_pbtk(chem.name="lindane", days=2.5, 
                    output.units=list(Cplasma="mg/L")))


</code></pre>

<hr>
<h2 id='solve_model'>Solve_model</h2><span id='topic+solve_model'></span>

<h3>Description</h3>

<p>solve_model is designed to accept systematized metadata (provided by the
model.list defined in the modelinfo files) for a given 
toxicokinetic model, including names of variables, parameterization
functions, and key units, and use it along with chemical information
to prepare an ode system for numerical solution over time of the amounts
or concentrations of chemical in different bodily compartments of a given
species (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or default &quot;Human&quot;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solve_model(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  times = NULL,
  parameters = NULL,
  model = NULL,
  route = "oral",
  dosing = NULL,
  days = 10,
  tsteps = 4,
  initial.values = NULL,
  initial.value.units = NULL,
  plots = FALSE,
  monitor.vars = NULL,
  suppress.messages = FALSE,
  species = "Human",
  input.units = "mg/kg",
  output.units = NULL,
  method = "lsoda",
  rtol = 1e-08,
  atol = 1e-12,
  recalc.blood2plasma = FALSE,
  recalc.clearance = FALSE,
  restrictive.clearance = TRUE,
  adjusted.Funbound.plasma = TRUE,
  minimum.Funbound.plasma = 1e-04,
  parameterize.arg.list = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="solve_model_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters
must be specified.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must
be specified.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="solve_model_+3A_times">times</code></td>
<td>
<p>Optional time sequence for specified number of output times (in days) to be returned by the function.
The model is solved explicitly at the time sequence specified. Dosing sequence begins at the first time provided.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_parameters">parameters</code></td>
<td>
<p>List of chemical parameters, as output by 
parameterize_pbtk function. Overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_model">model</code></td>
<td>
<p>Specified model to use in simulation: &quot;pbtk&quot;, &quot;3compartment&quot;,
&quot;3compartmentss&quot;, &quot;1compartment&quot;, &quot;schmitt&quot;, ...</p>
</td></tr>
<tr><td><code id="solve_model_+3A_route">route</code></td>
<td>
<p>String specification of route of exposure for simulation:
&quot;oral&quot;, &quot;iv&quot;, &quot;inhalation&quot;, ...</p>
</td></tr>
<tr><td><code id="solve_model_+3A_dosing">dosing</code></td>
<td>
<p>List of dosing metrics used in simulation, which includes
the namesake entries of a model's associated dosing.params. In the case
of most httk models, these should include &quot;initial.dose&quot;, &quot;doses.per.day&quot;, 
&quot;daily.dose&quot;, and &quot;dosing.matrix&quot;. The &quot;dosing.matrix&quot; is used for more
precise dose regimen specification, and is a matrix consisting of two
columns or rows named &quot;time&quot; and &quot;dose&quot; containing the time and amount of 
each dose. If none of the namesake entries of the dosing list is set to a
non-NULL value, solve_model uses a default dose of 1 mg/kg BW along with the 
dose type (add/multiply) specified for a given route (e.g. add the dose to gut
lumen for oral route)</p>
</td></tr>
<tr><td><code id="solve_model_+3A_days">days</code></td>
<td>
<p>Simulated period. Default 10 days.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_tsteps">tsteps</code></td>
<td>
<p>The number of time steps per hour. Default of 4.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_initial.values">initial.values</code></td>
<td>
<p>Vector of numeric values containing the initial
concentrations or amounts of the chemical in specified tissues with units
corresponding to those specified for the model outputs. Default values are zero.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_initial.value.units">initial.value.units</code></td>
<td>
<p>Vector of character strings containing the units
corresponding to 'initial.values' specified for the model outputs.
Default is assuming the units match expected compartment units for the model.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_plots">plots</code></td>
<td>
<p>Plots all outputs if true.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_monitor.vars">monitor.vars</code></td>
<td>
<p>Which variables are returned as a function of time. 
Default values of NULL looks up variables specified in modelinfo_MODEL.R</p>
</td></tr>
<tr><td><code id="solve_model_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output messages are suppressed.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_species">species</code></td>
<td>
<p>Species desired (models have been designed to be
parameterized for some subset of the following species: &quot;Rat&quot;, &quot;Rabbit&quot;, 
&quot;Dog&quot;, &quot;Mouse&quot;, or default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="solve_model_+3A_input.units">input.units</code></td>
<td>
<p>Input units of interest assigned to dosing. Defaults
to mg/kg BW, in line with the default dosing scheme of a one-time dose of
1 mg/kg in which no other dosing parameters are specified.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_output.units">output.units</code></td>
<td>
<p>Output units of interest for the compiled components.
Defaults to NULL, and will provide values in model units if unspecified.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_method">method</code></td>
<td>
<p>Method used by integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_model_+3A_rtol">rtol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_model_+3A_atol">atol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_model_+3A_recalc.blood2plasma">recalc.blood2plasma</code></td>
<td>
<p>Recalculates the ratio of the amount of chemical
in the blood to plasma using the input parameters, calculated with
hematocrit, Funbound.plasma, and Krbc2pu.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_recalc.clearance">recalc.clearance</code></td>
<td>
<p>Recalculates the the hepatic clearance
(Clmetabolism) with new million.cells.per.gliver parameter.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to
1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset)</p>
</td></tr>
<tr><td><code id="solve_model_+3A_parameterize.arg.list">parameterize.arg.list</code></td>
<td>
<p>Additional parameterized passed to the model
parameterization function.</p>
</td></tr>
<tr><td><code id="solve_model_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the integrator.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Dosing values with certain acceptable associated input.units (like mg/kg BW)
are configured to undergo a unit conversion. All model simulations are 
intended to run with units as specifed by &quot;compartment.units&quot; in the 
model.list (as defined by the modelinfo files).
</p>
<p>The 'dosing' argument includes all parameters needed to describe exposure
in terms of route of administration, frequency, and quantity short of 
scenarios that require use of a more precise forcing function. If the dosing
argument's namesake entries are left NULL, solve_model defaults to a
single-time dose of 1 mg/kg BW according to the given dosing route and 
associated type (either add/multiply, for example we typically add a dose to 
gut lumen 
when oral route is specified).
</p>
<p>AUC is the area under the curve of the plasma concentration.
</p>
<p>Model parameters are named according to the following convention:
</p>

<table>
<tr>
 <td style="text-align: left;">
prefix </td><td style="text-align: right;"> suffix </td><td style="text-align: right;"> Meaning </td><td style="text-align: right;"> units </td>
</tr>
<tr>
 <td style="text-align: left;">
K </td><td style="text-align: right;"> </td><td style="text-align: right;"> Partition coefficient for tissue to free plasma \ tab unitless </td>
</tr>
<tr>
 <td style="text-align: left;">
V </td><td style="text-align: right;"> </td><td style="text-align: right;"> Volume </td><td style="text-align: right;"> L </td>
</tr>
<tr>
 <td style="text-align: left;">
Q </td><td style="text-align: right;"> </td><td style="text-align: right;"> Flow </td><td style="text-align: right;"> L/h </td>
</tr>
<tr>
 <td style="text-align: left;">
k </td><td style="text-align: right;"> </td><td style="text-align: right;"> Rate </td><td style="text-align: right;"> 1/h </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: right;"> c </td><td style="text-align: right;"> Parameter is proportional to body weight </td><td style="text-align: right;"> 1 / kg for volumes
and 1/kg^(3/4) for flows </td>
</tr>

</table>

<p>When species is specified but chemical-specific in vitro data are not
available, the function uses the appropriate physiological data (volumes and
flows) but default.to.human = TRUE must be used to substitute human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
(NOTE: The 'default.to.human' specification should be included as part of the
arguments listed in 'parameterize.arg.list'.)
</p>
<p>For both plotting purposes and helping the numerical equation solver, it is
helpful to specify that time points shortly before and after dosing are 
included. This function automatically add these points, and they are returned
to the user unless the times argument is used, in which case only the time
points specified by that argument are provided.
</p>


<h3>Value</h3>

<p>A matrix of class deSolve with a column for time(in days), each
compartment, the area under the curve, and plasma concentration and a row
for each time point.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh, Robert Pearce, Miyuki Breen, Mark Sfeir, and
Sarah E. Davidson
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The various "solve_x" functions are wrappers for solve_model:
head(solve_pbtk(chem.name="Terbufos", days=1))

head(solve_model(chem.name="Terbufos",model="pbtk",
                 days=1,
                 dosing=list(
                   initial.dose = 1, # Assume dose is in mg/kg BW/day  
                   doses.per.day=NULL,
                   dosing.matrix = NULL,
                   daily.dose = NULL)))

# A dose matrix specifies times and magnitudes of doses:
dm &lt;- matrix(c(0,1,2,5,5,5),nrow=3)
colnames(dm) &lt;- c("time","dose")

solve_pbtk(chem.name="Methenamine",
           dosing.matrix=dm,
           dose=NULL,
           days=2.5,
           daily.dose=NULL)

solve_model(chem.name="Methenamine",
            model="pbtk",
            days=2.5,
            dosing=list(
              initial.dose =NULL,
              doses.per.day=NULL,
              daily.dose=NULL,
              dosing.matrix=dm))

solve_model(chem.name="Besonprodil",
            model="pbtk",
            days=2.5,
            dosing=list(
              initial.dose=NULL,
              doses.per.day=4,
              daily.dose=1,
              dosing.matrix=NULL))
  
solve_pbtk(chem.name="Besonprodil",
           daily.dose=1,
           dose=NULL,
           doses.per.day=4,
           days=2.5)

</code></pre>

<hr>
<h2 id='solve_pbtk'>Solve_PBTK</h2><span id='topic+solve_pbtk'></span>

<h3>Description</h3>

<p>This function solves for the amounts or concentrations in uM of a chemical
in different tissues as functions of time based on the dose and dosing
frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solve_pbtk(
  chem.name = NULL,
  chem.cas = NULL,
  dtxsid = NULL,
  times = NULL,
  parameters = NULL,
  days = 10,
  tsteps = 4,
  daily.dose = NULL,
  dose = NULL,
  doses.per.day = NULL,
  initial.values = NULL,
  plots = FALSE,
  suppress.messages = FALSE,
  species = "Human",
  iv.dose = FALSE,
  input.units = "mg/kg",
  output.units = NULL,
  method = "lsoda",
  rtol = 1e-08,
  atol = 1e-12,
  default.to.human = FALSE,
  recalc.blood2plasma = FALSE,
  recalc.clearance = FALSE,
  dosing.matrix = NULL,
  adjusted.Funbound.plasma = TRUE,
  regression = TRUE,
  restrictive.clearance = TRUE,
  minimum.Funbound.plasma = 1e-04,
  Caco2.options = list(),
  monitor.vars = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="solve_pbtk_+3A_chem.name">chem.name</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters
must be specified.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_chem.cas">chem.cas</code></td>
<td>
<p>Either the chemical name, CAS number, or the parameters must
be specified.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_dtxsid">dtxsid</code></td>
<td>
<p>EPA's DSSTox Structure ID (<a href="https://comptox.epa.gov/dashboard">https://comptox.epa.gov/dashboard</a>)
the chemical must be identified by either CAS, name, or DTXSIDs</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_times">times</code></td>
<td>
<p>Optional time sequence for specified number of days.  Dosing
sequence begins at the beginning of times.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_parameters">parameters</code></td>
<td>
<p>Chemical parameters from parameterize_pbtk function,
overrides chem.name and chem.cas.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_days">days</code></td>
<td>
<p>Length of the simulation.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_tsteps">tsteps</code></td>
<td>
<p>The number of time steps per hour.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_daily.dose">daily.dose</code></td>
<td>
<p>Total daily dose, defaults to mg/kg BW.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_dose">dose</code></td>
<td>
<p>Amount of a single, initial oral dose in mg/kg BW.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_doses.per.day">doses.per.day</code></td>
<td>
<p>Number of doses per day.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_initial.values">initial.values</code></td>
<td>
<p>Vector containing the initial concentrations or
amounts of the chemical in specified tissues with units corresponding to
output.units.  Defaults are zero.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_plots">plots</code></td>
<td>
<p>Plots all outputs if true.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_suppress.messages">suppress.messages</code></td>
<td>
<p>Whether or not the output message is suppressed.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_species">species</code></td>
<td>
<p>Species desired (either &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Mouse&quot;, or
default &quot;Human&quot;).</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_iv.dose">iv.dose</code></td>
<td>
<p>Simulates a single i.v. dose if true.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_input.units">input.units</code></td>
<td>
<p>Input units of interest assigned to dosing, defaults to
mg/kg BW</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_output.units">output.units</code></td>
<td>
<p>A named vector of output units expected for the model
results. Default, NULL, returns model results in units specified in the
'modelinfo' file. See table below for details.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_method">method</code></td>
<td>
<p>Method used by integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_rtol">rtol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_atol">atol</code></td>
<td>
<p>Argument passed to integrator (deSolve).</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_default.to.human">default.to.human</code></td>
<td>
<p>Substitutes missing animal values with human values
if true (hepatic intrinsic clearance or fraction of unbound plasma).</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_recalc.blood2plasma">recalc.blood2plasma</code></td>
<td>
<p>Recalculates the ratio of the amount of chemical
in the blood to plasma using the input parameters, calculated with
hematocrit, Funbound.plasma, and Krbc2pu.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_recalc.clearance">recalc.clearance</code></td>
<td>
<p>Recalculates the the hepatic clearance
(Clmetabolism) with new million.cells.per.gliver parameter.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_dosing.matrix">dosing.matrix</code></td>
<td>
<p>Vector of dosing times or a matrix consisting of two
columns or rows named &quot;dose&quot; and &quot;time&quot; containing the time and amount, in
mg/kg BW, of each dose.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_adjusted.funbound.plasma">adjusted.Funbound.plasma</code></td>
<td>
<p>Uses adjusted Funbound.plasma when set to
TRUE along with partition coefficients calculated with this value.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_regression">regression</code></td>
<td>
<p>Whether or not to use the regressions in calculating
partition coefficients.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_restrictive.clearance">restrictive.clearance</code></td>
<td>
<p>Protein binding not taken into account (set to
1) in liver clearance if FALSE.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_minimum.funbound.plasma">minimum.Funbound.plasma</code></td>
<td>
<p>Monte Carlo draws less than this value are set 
equal to this value (default is 0.0001 &ndash; half the lowest measured Fup in our
dataset).</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_caco2.options">Caco2.options</code></td>
<td>
<p>A list of options to use when working with Caco2 apical to
basolateral data <code>Caco2.Pab</code>, default is Caco2.options = list(Caco2.Pab.default = 1.6,
Caco2.Fabs = TRUE, Caco2.Fgut = TRUE, overwrite.invivo = FALSE, keepit100 = FALSE). Caco2.Pab.default sets the default value for 
Caco2.Pab if Caco2.Pab is unavailable. Caco2.Fabs = TRUE uses Caco2.Pab to calculate
fabs.oral, otherwise fabs.oral = <code>Fabs</code>. Caco2.Fgut = TRUE uses Caco2.Pab to calculate 
fgut.oral, otherwise fgut.oral = <code>Fgut</code>. overwrite.invivo = TRUE overwrites Fabs and Fgut in vivo values from literature with 
Caco2 derived values if available. keepit100 = TRUE overwrites Fabs and Fgut with 1 (i.e. 100 percent) regardless of other settings.
See <code><a href="#topic+get_fabsgut">get_fabsgut</a></code> for further details.</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_monitor.vars">monitor.vars</code></td>
<td>
<p>Which variables are returned as a function of time. 
The default value of NULL provides &quot;Cgut&quot;, &quot;Cliver&quot;, &quot;Cven&quot;, &quot;Clung&quot;, &quot;Cart&quot;, 
&quot;Crest&quot;, &quot;Ckidney&quot;, &quot;Cplasma&quot;, &quot;Atubules&quot;, &quot;Ametabolized&quot;, and &quot;AUC&quot;</p>
</td></tr>
<tr><td><code id="solve_pbtk_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the integrator.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the model parameters have units of hours while the model output is
in days.
</p>
<p>Default NULL value for doses.per.day solves for a single dose.
</p>
<p>The compartments used in this model are the gutlumen, gut, liver, kidneys,
veins, arteries, lungs, and the rest of the body.
</p>
<p>The extra compartments include the amounts or concentrations metabolized by
the liver and excreted by the kidneys through the tubules.
</p>
<p>AUC is the area under the curve of the plasma concentration.
</p>
<p>Model Figure 
<img src="../help/figures/pbtk.jpg" width="60%" alt="Figure: PBTK Model Schematic" />

</p>
<p>When species is specified as rabbit, dog, or mouse, the function uses the
appropriate physiological data(volumes and flows) but substitutes human
fraction unbound, partition coefficients, and intrinsic hepatic clearance.
</p>


<h3>Value</h3>

<p>A matrix of class deSolve with a column for time(in days), each
compartment, the area under the curve, and plasma concentration and a row
for each time point.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh and Robert Pearce
</p>


<h3>References</h3>

<p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS (2017).
&ldquo;Httk: R package for high-throughput toxicokinetics.&rdquo;
<em>Journal of Statistical Software</em>, <b>79</b>(4), 1.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+solve_model">solve_model</a></code>
</p>
<p><code><a href="#topic+parameterize_gas_pbtk">parameterize_gas_pbtk</a></code>
</p>
<p><code><a href="#topic+calc_analytic_css_pbtk">calc_analytic_css_pbtk</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Multiple doses per day:
head(solve_pbtk(
  chem.name='Bisphenol-A',
  daily.dose=.5,
  days=2.5,
  doses.per.day=2,
  tsteps=2))

# Starting with an initial concentration:
out &lt;- solve_pbtk(
  chem.name='bisphenola',
  dose=0,
  days=2.5,
  output.units="mg/L", 
  initial.values=c(Agut=200))

# Working with parameters (rather than having solve_pbtk retrieve them):
params &lt;- parameterize_pbtk(chem.cas="80-05-7")
head(solve_pbtk(parameters=params, days=2.5))
                  
# We can change the parameters given to us by parameterize_pbtk:
params &lt;- parameterize_pbtk(dtxsid="DTXSID4020406", species = "rat")
params["Funbound.plasma"] &lt;- 0.1
out &lt;- solve_pbtk(parameters=params, days=2.5)

# A fifty day simulation:
out &lt;- solve_pbtk(
  chem.name = "Bisphenol A", 
  days = 50, 
  daily.dose=1,
  doses.per.day = 3)
plot.data &lt;- as.data.frame(out)
css &lt;- calc_analytic_css(chem.name = "Bisphenol A")

library("ggplot2")
c.vs.t &lt;- ggplot(plot.data, aes(time, Cplasma)) + 
  geom_line() +
  geom_hline(yintercept = css) + 
  ylab("Plasma Concentration (uM)") +
  xlab("Day") + 
  theme(
    axis.text = element_text(size = 16), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 17)) +
  ggtitle("Bisphenol A")
print(c.vs.t)


</code></pre>

<hr>
<h2 id='spleen_mass_children'>Predict spleen mass for children</h2><span id='topic+spleen_mass_children'></span>

<h3>Description</h3>

<p>For individuals under 18, predict the spleen mass from height, weight, and
gender, using equations from Ogiu et al. (1997)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spleen_mass_children(height, weight, gender)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spleen_mass_children_+3A_height">height</code></td>
<td>
<p>Vector of heights in cm.</p>
</td></tr>
<tr><td><code id="spleen_mass_children_+3A_weight">weight</code></td>
<td>
<p>Vector of weights in kg.</p>
</td></tr>
<tr><td><code id="spleen_mass_children_+3A_gender">gender</code></td>
<td>
<p>Vector of genders (either 'Male' or 'Female').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of spleen masses in kg.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ogiu, Nobuko, et al. &quot;A statistical analysis of the internal 
organ weights of normal Japanese people.&quot; Health physics 72.3 (1997): 368-383.
</p>
<p>Price, Paul S., et al. &quot;Modeling interindividual variation in physiological 
factors used in PBPK models of humans.&quot; Critical reviews in toxicology 33.5 
(2003): 469-503.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='supptab1_Linakis2020'>Supplementary output from Linakis 2020 vignette analysis.</h2><span id='topic+supptab1_Linakis2020'></span>

<h3>Description</h3>

<p>Supplementary output from Linakis 2020 vignette analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>supptab1_Linakis2020
</code></pre>


<h3>Format</h3>

<p>A data.frame containing x rows and y columns.
</p>


<h3>Author(s)</h3>

<p>Matt Linakis
</p>


<h3>Source</h3>

<p>Matt Linakis
</p>


<h3>References</h3>

<p>DSStox database (https:// www.epa.gov/ncct/dsstox
</p>

<hr>
<h2 id='supptab2_Linakis2020'>More supplementary output from Linakis 2020 vignette analysis.</h2><span id='topic+supptab2_Linakis2020'></span>

<h3>Description</h3>

<p>More supplementary output from Linakis 2020 vignette analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>supptab2_Linakis2020
</code></pre>


<h3>Format</h3>

<p>A data.frame containing x rows and y columns.
</p>


<h3>Author(s)</h3>

<p>Matt Linakis
</p>


<h3>Source</h3>

<p>Matt Linakis
</p>


<h3>References</h3>

<p>DSStox database (https:// www.epa.gov/ncct/dsstox
</p>

<hr>
<h2 id='Tables.Rdata.stamp'>A timestamp of table creation</h2><span id='topic+Tables.Rdata.stamp'></span>

<h3>Description</h3>

<p>The Tables.RData file is separately created as part of building a new
release of HTTK. This time stamp indicates the script used to build the file
and when it was run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tables.Rdata.stamp
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 1.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>

<hr>
<h2 id='tissue_masses_flows'>Given a data.table describing a virtual population by the NHANES quantities, 
generates HTTK physiological parameters for each individual.</h2><span id='topic+tissue_masses_flows'></span>

<h3>Description</h3>

<p>Given a data.table describing a virtual population by the NHANES quantities, 
generates HTTK physiological parameters for each individual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tissue_masses_flows(tmf_dt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tissue_masses_flows_+3A_tmf_dt">tmf_dt</code></td>
<td>
<p>A data.table generated by
<code>gen_age_height_weight()</code>, containing variables <code>gender</code>,
<code>reth</code>, <code>age_months</code>, <code>age_years</code>, <code>weight</code>, and
<code>height</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The same data.table, with aditional variables describing tissue masses
and flows.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Barter, Zoe E., et al. &quot;Scaling factors for the extrapolation of in vivo 
metabolic drug clearance from in vitro data: reaching a consensus on values 
of human micro-somal protein and hepatocellularity per gram of liver.&quot; Current 
Drug Metabolism 8.1 (2007): 33-45.
</p>
<p>Birnbaum, L., et al. &quot;Physiological parameter values for PBPK models.&quot; 
International Life Sciences Institute, Risk Science Institute, Washington, 
DC (1994).
</p>
<p>Geigy Pharmaceuticals, &quot;Scientific Tables&quot;, 7th Edition, 
John Wiley and Sons (1970)
</p>
<p>McNally, Kevin, et al. &quot;PopGen: a virtual human population generator.&quot; 
Toxicology 315 (2014): 70-85.
</p>
<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='tissue_scale'>Allometric scaling.</h2><span id='topic+tissue_scale'></span>

<h3>Description</h3>

<p>Allometrically scale a tissue mass or flow based on height^(3/4).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tissue_scale(height_ref, height_indiv, tissue_mean_ref)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tissue_scale_+3A_height_ref">height_ref</code></td>
<td>
<p>Reference height in cm.</p>
</td></tr>
<tr><td><code id="tissue_scale_+3A_height_indiv">height_indiv</code></td>
<td>
<p>Individual height in cm.</p>
</td></tr>
<tr><td><code id="tissue_scale_+3A_tissue_mean_ref">tissue_mean_ref</code></td>
<td>
<p>Reference tissue mass or flow.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Allometrically scaled tissue mass or flow, in the same units as
<code>tissue_mean_ref</code>.
</p>


<h3>Author(s)</h3>

<p>Caroline Ring
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>

<hr>
<h2 id='tissue.data'>Tissue composition and species-specific physiology parameters</h2><span id='topic+tissue.data'></span>

<h3>Description</h3>

<p>This data set contains values from Schmitt (2008) and Ruark et al. (2014)
describing the composition of specific tissues and from Birnbaum et al.
(1994) describing volumes of and blood flows to those tissues, allowing
parameterization of toxicokinetic models for human, mouse, rat, dog, or
rabbit. Tissue volumes were calculated by converting the fractional mass of
each tissue with its density (both from ICRP), lumping the remaining tissues
into the rest-of-body, excluding the mass of the gastrointestinal contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tissue.data
</code></pre>


<h3>Format</h3>

<p>A data.frame containing 406 rows and 5 columns.
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Column</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
Tissue </td><td style="text-align: left;"> The tissue being described </td>
</tr>
<tr>
 <td style="text-align: left;">                      
Species </td><td style="text-align: left;"> The species being described </td>
</tr>
<tr>
 <td style="text-align: left;">                      
Reference </td><td style="text-align: left;"> The reference for the value reported </td>
</tr>
<tr>
 <td style="text-align: left;">                      
variable </td><td style="text-align: left;"> The aspect of the tissue being characterized </td>
</tr>
<tr>
 <td style="text-align: left;">                      
value </td><td style="text-align: left;"> The value for the variable for the given tissue and species </td>
</tr>
<tr>
 <td style="text-align: left;">                      
</td>
</tr>

</table>



<h3>Details</h3>

<p>Many of the parameters were compiled initially in Table 2 of Schmitt (2009).
The full list of tissue variables described is:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Variable</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td><td style="text-align: left;"> <strong>Units</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
Fcell </td><td style="text-align: left;"> Cellular fraction of total tissue volume </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">
Fint </td><td style="text-align: left;"> Interstitial fraction of total tissue volume </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">
FWc </td><td style="text-align: left;"> Fraction of cell volume that is water </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">
FLc </td><td style="text-align: left;"> Fraction of cell volume that is lipid </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">
FPc </td><td style="text-align: left;"> Fraction of cell volume that is protein </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">
Fn_Lc </td><td style="text-align: left;"> Fraction of cellular lipid tht is neutral lipid </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">
Fn_PLc </td><td style="text-align: left;"> Fraction of cellular lipid tht is neutral phospholipid </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">
Fa_PLc </td><td style="text-align: left;"> Fraction of cellular lipid tht is acidic phospholipid </td><td style="text-align: left;"> fraction </td>
</tr>
<tr>
 <td style="text-align: left;">
pH </td><td style="text-align: left;"> Negative logarithm of H+ ion concentration </td><td style="text-align: left;"> unitless</td>
</tr>
<tr>
 <td style="text-align: left;">
Density </td><td style="text-align: left;"> Tissue density </td><td style="text-align: left;"> g/cm^3 </td>
</tr>
<tr>
 <td style="text-align: left;">
Vol </td><td style="text-align: left;"> Tissue volume </td><td style="text-align: left;"> L/kg </td>
</tr>
<tr>
 <td style="text-align: left;">
Flow </td><td style="text-align: left;"> Blood flow to tissue </td><td style="text-align: left;"> mL/min/kg^(3/4) </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>New tissues can be added to this table to generate
their partition coefficients.
</p>


<h3>Author(s)</h3>

<p>John Wambaugh, Robert Pearce, and Nisha Sipes
</p>


<h3>References</h3>

<p>Birnbaum L, Brown R, Bischoff K, Foran J, Blancato J, Clewell H, Dedrick R (1994).
&ldquo;Physiological parameter values for PBPK models.&rdquo;
<em>International Life Sciences Institute, Risk Science Institute, Washington, DC</em>.
</p>
<p>Ruark CD, Hack CE, Robinson PJ, Mahle DA, Gearhart JM (2014).
&ldquo;Predicting passive and active tissue: plasma partition coefficients: interindividual and interspecies variability.&rdquo;
<em>Journal of pharmaceutical sciences</em>, <b>103</b>(7), 2189&ndash;2198.
</p>
<p>Schmitt W (2008).
&ldquo;General approach for the calculation of tissue to plasma partition coefficients.&rdquo;
<em>Toxicology in vitro</em>, <b>22</b>(2), 457&ndash;467.
</p>
<p>Snyder WS (1974).
&ldquo;Report of the task group on reference man.&rdquo;
<em>ICRP publication</em>.
</p>
<p>Wambaugh JF, Wetmore BA, Pearce R, Strope C, Goldsmith R, Sluka JP, Sedykh A, Tropsha A, Bosgra S, Shah I, others (2015).
&ldquo;Toxicokinetic triage for environmental chemicals.&rdquo;
<em>Toxicological Sciences</em>, <b>147</b>(1), 55&ndash;67.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+predict_partitioning_schmitt">predict_partitioning_schmitt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># We can add thyroid to the tissue data by making a row containing
# its data, subtracting the volumes and flows from the rest-of-body, 
# and binding the row to tissue.data. Here we assume it contains the same 
# partition coefficient data as the spleen and a tenth of the volume and  
# blood flow:
new.tissue &lt;- subset(tissue.data,Tissue == "spleen")
new.tissue[, "Tissue"] &lt;- "thyroid"
new.tissue[new.tissue$variable %in% c("Vol (L/kg)",
"Flow (mL/min/kg^(3/4))"),"value"] &lt;- new.tissue[new.tissue$variable
%in% c("Vol (L/kg)","Flow (mL/min/kg^(3/4))"),"value"] / 10
tissue.data[tissue.data$Tissue == "rest", "value"] &lt;-
tissue.data[tissue.data$Tissue == "rest", "value"] -
new.tissue[new.tissue$variable %in% c("Vol (L/kg)",
"Flow (mL/min/kg^(3/4))"),"value"]
tissue.data &lt;- rbind(tissue.data, new.tissue)

</code></pre>

<hr>
<h2 id='wambaugh2019'>in vitro Toxicokinetic Data from Wambaugh et al. (2019)</h2><span id='topic+wambaugh2019'></span>

<h3>Description</h3>

<p>These data are the new HTTK in vitro data for chemicals reported in Wambaugh
et al. (2019) They
are the processed values used to make the figures in that manuscript.
These data summarize the results of Bayesian analysis of the in vitro
toxicokinetic experiments conducted by Cyprotex to characterize fraction
unbound in the presence of pooled human plasma protein and the intrnsic
hepatic clearance of the chemical by pooled human hepatocytes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wambaugh2019
</code></pre>


<h3>Format</h3>

<p>A data frame with 496 rows and 17 variables:
</p>

<dl>
<dt>Compound</dt><dd><p>The name of the chemical</p>
</dd>
<dt>CAS</dt><dd><p>The Chemical Abstracts Service Registry Number</p>
</dd>
<dt>Human.Clint</dt><dd><p>Median of Bayesian credible interval for intrinsic
hepatic clearance (uL/min/million hepatocytes)]</p>
</dd>
<dt>Human.Clint.pValue</dt><dd><p>Probability that there is no clearance</p>
</dd>
<dt>Human.Funbound.plasma</dt><dd><p>Median of Bayesian credibl interval for
fraction of chemical free in the presence of plasma</p>
</dd>
<dt>pKa_Accept</dt><dd><p>pH(s) at which hydrogen acceptor sites (if any) are at
equilibrium</p>
</dd>
<dt>pKa_Donor</dt><dd><p>pH(s) at which hydrogne donor sites (if any) are at
equilibrium</p>
</dd>
<dt>DSSTox_Substance_Id</dt><dd><p>Identifier for CompTox Chemical Dashboard</p>
</dd>
<dt>SMILES</dt><dd><p>Simplified Molecular-Input Line-Entry System structure
description</p>
</dd>
<dt>Human.Clint.Low95</dt><dd><p>Lower 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance (uL/min/million hepatocytes)</p>
</dd>
<dt>Human.Clint.High95</dt><dd><p>Uppper 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance (uL/min/million hepatocytes)</p>
</dd>
<dt>Human.Clint.Point</dt><dd><p>Point estimate of intrinsic hepatic clearance
(uL/min/million hepatocytes)</p>
</dd>
<dt>Human.Funbound.plasma.Low95</dt><dd><p>Lower 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma</p>
</dd>
<dt>Human.Funbound.plasma.High95</dt><dd><p>Upper 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma</p>
</dd>
<dt>Human.Funbound.plasma.Point</dt><dd><p>Point estimate of the fraction of
chemical free in the presence of plasma</p>
</dd>
<dt>MW</dt><dd><p>Molecular weight (Daltons)</p>
</dd>
<dt>logP</dt><dd><p>log base ten of octanol:water partiion coefficient</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Source</h3>

<p>Wambaugh et al. (2019)
</p>


<h3>References</h3>

<p>Wambaugh et al. (2019) &quot;Assessing Toxicokinetic Uncertainty and
Variability in Risk Prioritization&quot;, Toxicological Sciences, 172(2), 235-251.
</p>

<hr>
<h2 id='wambaugh2019.nhanes'>NHANES Chemical Intake Rates for chemicals in Wambaugh et al. (2019)</h2><span id='topic+wambaugh2019.nhanes'></span>

<h3>Description</h3>

<p>These data are a subset of the Bayesian inferrences reported by Ring et al.
(2017) from the U.S. Centers for Disease Control and Prevention (CDC)
National Health and Nutrition Examination Survey (NHANES). They reflect the
populaton median intake rate (mg/kg body weight/day), with uncertainty.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wambaugh2019.nhanes
</code></pre>


<h3>Format</h3>

<p>A data frame with 20 rows and 4 variables:
</p>

<dl>
<dt>lP</dt><dd><p>The median of the Bayesian credible interval for median population
intake rate (mg/kg bodyweight/day)</p>
</dd>
<dt>lP.min</dt><dd><p>The lower 95th percentile of the Bayesian credible interval for median population
intake rate (mg/kg bodyweight/day)</p>
</dd>
<dt>lP.max</dt><dd><p>The upper 95th percentile of the Bayesian credible interval for median population
intake rate (mg/kg bodyweight/day)</p>
</dd>
<dt>CASRN</dt><dd><p>The Chemical Abstracts Service Registry Number</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Source</h3>

<p>Wambaugh et al. (2019)
</p>


<h3>References</h3>

<p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF (2017).
&ldquo;Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability.&rdquo;
<em>Environment International</em>, <b>106</b>, 105&ndash;118.
</p>
<p>Wambaugh et al. (2019) &quot;Assessing Toxicokinetic Uncertainty and
Variability in Risk Prioritization&quot;, Toxicological Sciences, 172(2), 235-251.
</p>

<hr>
<h2 id='wambaugh2019.raw'>Raw Bayesian in vitro Toxicokinetic Data Analysis from Wambaugh et al. (2019)</h2><span id='topic+wambaugh2019.raw'></span>

<h3>Description</h3>

<p>These data are the new HTTK in vitro data for chemicals reported in Wambaugh
et al. (2019) They
are the output of different Bayesian models evaluated to compare using a
single protein concentration vs. the new three concentration titration
protocol. These data summarize the results of Bayesian analysis of the in vitro
toxicokinetic experiments conducted by Cyprotex to characterize fraction
unbound in the presence of pooled human plasma protein and the intrnsic
hepatic clearance of the chemical by pooled human hepatocytes.
This file includes replicates (diferent CompoundName id's but same chemical')
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wambaugh2019.raw
</code></pre>


<h3>Format</h3>

<p>A data frame with 530 rows and 28 variables:
</p>

<dl>
<dt>DTXSID</dt><dd><p>Identifier for CompTox Chemical Dashboard</p>
</dd>
<dt>Name</dt><dd><p>The name of the chemical</p>
</dd>
<dt>CAS</dt><dd><p>The Chemical Abstracts Service Registry Number</p>
</dd>
<dt>CompoundName</dt><dd><p>Sample name provided by EPA to Cyprotex</p>
</dd>
<dt>Fup.point</dt><dd><p>Point estimate of the fraction of
chemical free in the presence of plasma</p>
</dd>
<dt>Base.Fup.Med</dt><dd><p>Median of Bayesian credible interval for
fraction of chemical free in the presence of plasma for analysis of 100
physiological plasma protein data only (base model)</p>
</dd>
<dt>Base.Fup.Low</dt><dd><p>Lower 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma for analysis of 100
physiological plasma protein data only (base model)</p>
</dd>
<dt>Base.Fup.High</dt><dd><p>Upper 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma for analysis of 100
physiological plasma protein data only (base model)</p>
</dd>
<dt>Affinity.Fup.Med</dt><dd><p>Median of Bayesian credible interval for
fraction of chemical free in the presence of plasma for analysis of protein
titration protocol data (affinity model)</p>
</dd>
<dt>Affinity.Fup.Low</dt><dd><p>Lower 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma for analysis of protein
titration protocol data (affinity model)</p>
</dd>
<dt>Affinity.Fup.High</dt><dd><p>Upper 95th percentile of Bayesian credible
interval for fraction of chemical free in the presence of plasma for analysis of protein
titration protocol data (affinity model)</p>
</dd>
<dt>Affinity.Kd.Med</dt><dd><p>Median of Bayesian credible interval for
protein binding affinity from analysis of protein
titration protocol data (affinity model)</p>
</dd>
<dt>Affinity.Kd.Low</dt><dd><p>Lower 95th percentile of Bayesian credible
interval for protein binding affinity from analysis of protein
titration protocol data (affinity model)</p>
</dd>
<dt>Affinity.Kd.High</dt><dd><p>Upper 95th percentile of Bayesian credible
interval for protein binding affinity from analysis of protein
titration protocol data (affinity model)</p>
</dd>
<dt>Decreases.Prob</dt><dd><p>Probability that the chemical concentration decreased
systematiclally during hepatic clearance assay.</p>
</dd>
<dt>Saturates.Prob</dt><dd><p>Probability that the rate of chemical concentration
decrease varied between the 1 and 10 uM hepatic clearance experiments.</p>
</dd>
<dt>Slope.1uM.Median</dt><dd><p>Estimated slope for chemcial concentration decrease
in the 1 uM hepatic clearance assay.</p>
</dd>
<dt>Slope.10uM.Median</dt><dd><p>Estimated slope for chemcial concentration decrease
in the 10 uM hepatic clearance assay.</p>
</dd>
<dt>CLint.1uM.Median</dt><dd><p>Median of Bayesian credible interval for intrinsic
hepatic clearance at 1 uM initital chemical concentration (uL/min/million hepatocytes)]</p>
</dd>
<dt>CLint.1uM.Low95th</dt><dd><p>Lower 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance at 1 uM initital chemical
concentration (uL/min/million hepatocytes)</p>
</dd>
<dt>CLint.1uM.High95th</dt><dd><p>Uppper 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance at 1 uM initital chemical
concentration(uL/min/million hepatocytes)</p>
</dd>
<dt>CLint.10uM.Median</dt><dd><p>Median of Bayesian credible interval for intrinsic
hepatic clearance at 10 uM initital chemical concentration (uL/min/million hepatocytes)]</p>
</dd>
<dt>CLint.10uM.Low95th</dt><dd><p>Lower 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance at 10 uM initital chemical
concentration (uL/min/million hepatocytes)</p>
</dd>
<dt>CLint.10uM.High95th</dt><dd><p>Uppper 95th percentile of Bayesian credible
interval for intrinsic hepatic clearance at 10 uM initital chemical
concentration(uL/min/million hepatocytes)</p>
</dd>
<dt>CLint.1uM.Point</dt><dd><p>Point estimate of intrinsic hepatic clearance
(uL/min/million hepatocytes) for 1 uM initial chemical concentration</p>
</dd>
<dt>CLint.10uM.Point</dt><dd><p>Point estimate of intrinsic hepatic clearance
(uL/min/million hepatocytes) for 10 uM initial chemical concentration</p>
</dd>
<dt>Fit</dt><dd><p>Classification of clearance observed</p>
</dd>
<dt>SMILES</dt><dd><p>Simplified Molecular-Input Line-Entry System structure
description</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Source</h3>

<p>Wambaugh et al. (2019)
</p>


<h3>References</h3>

<p>Wambaugh et al. (2019) &quot;Assessing Toxicokinetic Uncertainty and
Variability in Risk Prioritization&quot;, Toxicological Sciences, 172(2), 235-251.
</p>

<hr>
<h2 id='wambaugh2019.seem3'>ExpoCast SEEM3 Consensus Exposure Model Predictions for Chemical Intake Rates</h2><span id='topic+wambaugh2019.seem3'></span>

<h3>Description</h3>

<p>These data are a subset of the Bayesian inferrences reported by Ring et al.
(2019) for a consensus model of twelve exposue predictors. The predictors
were calibrated based upon their ability to predict intake rates inferred
National Health and Nutrition Examination Survey (NHANES). They reflect the
populaton median intake rate (mg/kg body weight/day), with uncertainty.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wambaugh2019.seem3
</code></pre>


<h3>Format</h3>

<p>A data frame with 385 rows and 38 variables:
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Source</h3>

<p>Wambaugh et al. (2019)
</p>


<h3>References</h3>

<p>Ring, Caroline L., et al. &quot;Consensus modeling of median chemical
intake for the US population based on predictions of exposure pathways.&quot;
Environmental science &amp; technology 53.2 (2018): 719-732.
</p>
<p>Wambaugh et al. (2019) &quot;Assessing Toxicokinetic Uncertainty and
Variability in Risk Prioritization&quot;, Toxicological Sciences, 172(2), 235-251.
</p>

<hr>
<h2 id='wambaugh2019.tox21'>Tox21 2015 Active Hit Calls (EPA)</h2><span id='topic+wambaugh2019.tox21'></span>

<h3>Description</h3>

<p>The ToxCast and Tox21 research programs employ batteries of high-throughput
assays to assess chemical bioactivity in vitro. Not every chemical is tested
through every assay. Most assays are conducted in concentration response,
and each corresponding assay endpoint is analyzed statistically to determine
if there is a concentration-dependent response or &quot;hit&quot; using the ToxCast
Pipeline.  Most assay endpoint-chemical combinations are non-responsive.
Here, only the hits are treated as potential indicators of bioactivity. This
bioactivity does not have a direct toxicological interpretation. The October
2015 release (invitrodb_v2) of the ToxCast and Tox21 data were used for this
analysis. This object contains just the chemicals in Wambaugh et al. (2019)
and only the quantiles across all assays for the ACC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wambaugh2019.tox21
</code></pre>


<h3>Format</h3>

<p>A data.table with 401 rows and 6 columns
</p>


<h3>Author(s)</h3>

<p>John Wambaugh
</p>


<h3>Source</h3>

<p><a href="https://gaftp.epa.gov/comptox/High_Throughput_Screening_Data/Previous_Data/ToxCast_Data_Release_Oct_2015/MySQL_Data/">https://gaftp.epa.gov/comptox/High_Throughput_Screening_Data/Previous_Data/ToxCast_Data_Release_Oct_2015/MySQL_Data/</a>
</p>


<h3>References</h3>

<p>Kavlock, Robert, et al. &quot;Update on EPA's ToxCast program:
providing high-throughput decision support tools for chemical risk
management.&quot; Chemical research in toxicology 25.7 (2012): 1287-1302.
</p>
<p>Tice, Raymond R., et al. &quot;Improving the human hazard characterization of
chemicals: a Tox21 update.&quot; Environmental health perspectives 121.7 (2013):
756-765.
</p>
<p>Richard, Ann M., et al. &quot;ToxCast chemical landscape: paving the road to 21st
century toxicology.&quot; Chemical research in toxicology 29.8 (2016): 1225-1251.
</p>
<p>Filer, Dayne L., et al. &quot;tcpl: the ToxCast pipeline for high-throughput
screening data.&quot; Bioinformatics 33.4 (2016): 618-620.
</p>
<p>Wambaugh, John F., et al. &quot;Assessing Toxicokinetic Uncertainty and 
Variability in Risk Prioritization.&quot; Toxicological Sciences 172.2 (2019): 
235-251.
</p>

<hr>
<h2 id='wang2018'>Wang et al. 2018
Wang et al. (2018) screened the blood of 75 pregnant women for the presence 
of environmental organic acids (EOAs) and identified mass spectral features 
corresponding to 453 chemical formulae of which 48 could be mapped to likely 
structures. Of the 48 with tentative structures the identity of six were 
confirmed with available chemical standards.</h2><span id='topic+wang2018'></span><span id='topic+Wang2018'></span>

<h3>Description</h3>

<p>Wang et al. 2018
Wang et al. (2018) screened the blood of 75 pregnant women for the presence 
of environmental organic acids (EOAs) and identified mass spectral features 
corresponding to 453 chemical formulae of which 48 could be mapped to likely 
structures. Of the 48 with tentative structures the identity of six were 
confirmed with available chemical standards.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wang2018
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Source</h3>

<p>Kapraun DF, Sfeir M, Pearce RG, Davidson-Fritz SE, Lumen A, Dallmann A, Judson RS, Wambaugh JF (2022).
&ldquo;Evaluation of a rapid, generic human gestational dose model.&rdquo;
<em>Reproductive Toxicology</em>, <b>113</b>, 172&ndash;188.
</p>


<h3>References</h3>

<p>Wang A, Gerona RR, Schwartz JM, Lin T, Sirota M, Morello-Frosch R, Woodruff TJ (2018).
&ldquo;A Suspect Screening Method for Characterizing Multiple Chemical Exposures among a Demographically Diverse Population of Pregnant Women in San Francisco.&rdquo;
<em>Environmental Health Perspectives</em>, <b>126</b>(7), 077009.
<a href="https://doi.org/10.1289/EHP2920">doi:10.1289/EHP2920</a>, https://ehp.niehs.nih.gov/doi/pdf/10.1289/EHP2920, <a href="https://ehp.niehs.nih.gov/doi/full/10.1289/EHP2920">https://ehp.niehs.nih.gov/doi/full/10.1289/EHP2920</a>.
</p>

<hr>
<h2 id='well_param'>Microtiter Plate Well Descriptions for Armitage et al. (2014) Model</h2><span id='topic+well_param'></span>

<h3>Description</h3>

<p>Microtiter Plate Well Descriptions for Armitage et al. (2014) model from
Honda et al. (2019)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>well_param
</code></pre>


<h3>Format</h3>

<p>A data frame / data table with 11 rows and 8 variables:
</p>

<dl>
<dt>sysID</dt><dd><p>Identifier for each multi-well plate system</p>
</dd>
<dt>well_desc</dt><dd><p>Well description</p>
</dd>
<dt>well_number</dt><dd><p>Number of wells on plate</p>
</dd>
<dt>area_bottom</dt><dd><p>Area of well bottom in mm^2</p>
</dd>
<dt>cell_yield</dt><dd><p>Number of cells</p>
</dd>
<dt>diam</dt><dd><p>Diameter of well in mm</p>
</dd>
<dt>v_total</dt><dd><p>Total volume of well in uL)</p>
</dd>
<dt>v_working</dt><dd><p>Working volume of well in uL</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Greg Honda
</p>


<h3>Source</h3>

<p><a href="https://www.corning.com/catalog/cls/documents/application-notes/CLS-AN-209.pdf">https://www.corning.com/catalog/cls/documents/application-notes/CLS-AN-209.pdf</a>
</p>


<h3>References</h3>

<p>Armitage, J. M.; Wania, F.; Arnot, J. A. Environ. Sci. Technol.
2014, 48, 9770-9779. dx.doi.org/10.1021/es501955g
Honda GS, Pearce RG, Pham LL, Setzer RW, Wetmore BA, Sipes NS, Gilbert J, Franz B, Thomas RS, Wambaugh JF (2019).
&ldquo;Using the concordance of in vitro and in vivo data to evaluate extrapolation assumptions.&rdquo;
<em>PloS one</em>, <b>14</b>(5), e0217564.
</p>

<hr>
<h2 id='Wetmore2012'>Published toxicokinetic predictions based on in vitro data from Wetmore et
al. 2012.</h2><span id='topic+Wetmore2012'></span>

<h3>Description</h3>

<p>This data set overlaps with Wetmore.data and is used only in Vignette 4 for
steady state concentration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Wetmore2012
</code></pre>


<h3>Format</h3>

<p>A data.frame containing 13 rows and 15 columns.
</p>


<h3>References</h3>

<p>Wetmore, B.A., Wambaugh, J.F., Ferguson, S.S., Sochaski, M.A.,
Rotroff, D.M., Freeman, K., Clewell, H.J., Dix, D.H., Andersen, M.E., Houck,
K.A., Allen, B., Judson, R.S., Sing, R., Kavlock, R.J., Richard, A.M., and
Thomas, R.S., &quot;Integration of Dosimetry, Exposure and High-Throughput
Screening Data in Chemical Toxicity Assessment,&quot; Toxicological Sciences 125
157-174 (2012)
</p>

<hr>
<h2 id='wfl'>WHO weight-for-length charts</h2><span id='topic+wfl'></span>

<h3>Description</h3>

<p>Charts giving weight-for-length percentiles for boys and girls under age 2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wfl
</code></pre>


<h3>Format</h3>

<p>a data.table with 262 rows and 4 variables:
</p>
 <dl>
<dt>Sex</dt><dd><p>&quot;Male&quot; or &quot;Female&quot;</p>
</dd> <dt>Length</dt><dd><p>Recumbent length in
cm</p>
</dd> <dt>P2.3</dt><dd><p>The 2.3rd percentile weight in kg for the corresponding sex
and recumbent length</p>
</dd> <dt>P97.7</dt><dd><p>The 97.7th percentile weight in kg for
the corresponding sex and recumbent length</p>
</dd></dl>



<h3>Details</h3>

<p>For infants under age 2, weight class depends on weight for length percentile.
#'</p>
 <dl>
<dt>Underweight</dt><dd><p>&lt;2.3rd percentile</p>
</dd> <dt>Normal
weight</dt><dd><p>2.3rd-97.7th percentile</p>
</dd> <dt>Obese</dt><dd><p>&gt;=97.7th percentile</p>
</dd> </dl>



<h3>Source</h3>

<p><a href="https://www.cdc.gov/growthcharts/who/boys_weight_head_circumference.htm">https://www.cdc.gov/growthcharts/who/boys_weight_head_circumference.htm</a>
and
<a href="https://www.cdc.gov/growthcharts/who/girls_weight_head_circumference.htm">https://www.cdc.gov/growthcharts/who/girls_weight_head_circumference.htm</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
