<!DOCTYPE html><html><head><title>Help for package hwde</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {hwde}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#decode.genotypes'><p>Creates contrasts that relate to a single locus</p></a></li>
<li><a href='#hwde'><p>Fit relevant models, and test for various types of departure from</p>
Hardy-Weinberg equilibrium.  Allows only 2 alleles per locus.  The
number of loci is arbitrary.</a></li>
<li><a href='#hwexact'><p>Exact test of HWE</p></a></li>
<li><a href='#IndianIrish'><p>Observed genotype frequencies at MN and S loci, for 2 populations</p></a></li>
<li><a href='#make.contrasts'><p>Calculate contrasts required by the function</p></a></li>
<li><a href='#mendelABC'><p>Mendel's F2 trifactorial data for seed shape (A: round or wrinkled),</p>
cotyledon color (B: albumen yellow or green), and seed coat color
(C: grey-brown or white)</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.67-3</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-21</td>
</tr>
<tr>
<td>Title:</td>
<td>Models and Tests for Departure from Hardy-Weinberg Equilibrium
and Independence Between Loci</td>
</tr>
<tr>
<td>Author:</td>
<td>John Maindonald, with \code{hwexact()} from Randall Johnson.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>John Maindonald &lt;jhmaindonald@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.0)</td>
</tr>
<tr>
<td>Description:</td>
<td>Fits models for genotypic disequilibria, as described in
  Huttley and Wilson (2000) &lt;<a href="https://doi.org/10.1093%2Fgenetics%2F156.4.2127">doi:10.1093/genetics/156.4.2127</a>&gt;, Weir (1996) and Weir and Wilson (1986).
  Contrast terms are available that account for first order interactions
  between loci.  Also implements, for a single locus in a single
  population, a conditional exact test for Hardy-Weinberg equilibrium.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Copyright:</td>
<td>(c) John Curtin Research School of Medical Sciences and
Centre for Mathematics and Its Applications, The Australian
National University</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jhmaindonald/hwde">https://github.com/jhmaindonald/hwde</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-21 07:48:02 UTC; johnm1</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-21 08:12:40 UTC</td>
</tr>
</table>
<hr>
<h2 id='decode.genotypes'>Creates contrasts that relate to a single locus</h2><span id='topic+decode.genotypes'></span>

<h3>Description</h3>

<p>Vectors ma, maa and oset (offset) are returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decode.genotypes(genotype)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="decode.genotypes_+3A_genotype">genotype</code></td>
<td>
<p>The vector <code>genotype</code> holds two-letter codes
for the three genotypes.  For example, the values may be AA, Aa and aa.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>oset</code></td>
<td>
<p>The offset values that would be appropriate, in the
multiplicative version of the model, if there was just this one
locus.</p>
</td></tr>
<tr><td><code>ma</code></td>
<td>
<p>A contrast for the Hardy-Weinberg model, at this locus.</p>
</td></tr>
<tr><td><code>maa</code></td>
<td>
<p>A contrast that measures departure from the Hardy-Weinberg
model, at this locus.</p>
</td></tr>
<tr><td><code>types</code></td>
<td>
<p>A vector of length three whose elements are the
two-letter codes used for the three genotypes.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Called by <code>make.contrasts</code></p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>See Also</h3>

<p><code>hwde</code></p>


<h3>Examples</h3>

<pre><code class='language-R'>decode.genotypes(rep(c("AA","Aa","aa"),2))
</code></pre>

<hr>
<h2 id='hwde'>Fit relevant models, and test for various types of departure from
Hardy-Weinberg equilibrium.  Allows only 2 alleles per locus.  The
number of loci is arbitrary.
</h2><span id='topic+hwde'></span>

<h3>Description</h3>

<p>Fits models for genotypic disequilibria, as described in Huttley and
Wilson (2000), Weir (1996) and Weir and Wilson (1986).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hwde(data = hwde::IndianIrish, gp = "Population", termlist = NULL, refmodel = NULL,
loci = paste("locus", 1:(dim(data)[2] - 1), sep = ""), observed = "Observed",
keep.models = FALSE, aovtable.print = TRUE, group.terms = TRUE, allele.chars = letters)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hwde_+3A_data">data</code></td>
<td>
<p>Must have a column of frequencies, by default called
<code>Observed</code>, and one or more columns giving genotype
information,
in the form AA, Aa, aa. or e.g., MM, MN, NN.  (The choice of letters
is arbitrary.)  Additionally, there may be a column that gives
information on groupings, by default called <code>Population</code>.
</p>
</td></tr>
<tr><td><code id="hwde_+3A_gp">gp</code></td>
<td>
<p>Gives the name of the column, if any, that has information
on groups within the data.</p>
</td></tr>
<tr><td><code id="hwde_+3A_termlist">termlist</code></td>
<td>
<p>Use to specify a user-defined sequence of models.
See the vignette <b>hwde.pdf</b> or <b>hwde.html</b></p>
</td></tr>
<tr><td><code id="hwde_+3A_refmodel">refmodel</code></td>
<td>
<p>For each model in <code>termlist</code>, specifies a
reference model, which will be updated to include the additional terms.</p>
</td></tr>
<tr><td><code id="hwde_+3A_loci">loci</code></td>
<td>
<p>Gives name(s) of columns that hold information on
genotypes. By default, these are taken to be <code>locus1</code>,
<code>locus2</code>, etc.</p>
</td></tr>
<tr><td><code id="hwde_+3A_observed">observed</code></td>
<td>
<p>Name (by default <code>Observed</code>) of the column that
holds the frequenceies.</p>
</td></tr>
<tr><td><code id="hwde_+3A_keep.models">keep.models</code></td>
<td>
<p>Should a list be returned that holds the full
sequence of models that were fitted?</p>
</td></tr>
<tr><td><code id="hwde_+3A_aovtable.print">aovtable.print</code></td>
<td>
<p>Should the anova table be printed?</p>
</td></tr>
<tr><td><code id="hwde_+3A_group.terms">group.terms</code></td>
<td>
<p>Should model terms be grouped according to
hierarchy,
for the anova table?</p>
</td></tr>
<tr><td><code id="hwde_+3A_allele.chars">allele.chars</code></td>
<td>
<p>A sequence of letters used to code for the loci.
By default <em>a</em>, <em>b</em>, <em>c</em>, ... are used</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the document <b>hwde.pdf</b> or <b>hwde.html</b> for details.
See the references (below) for information on the interpretation of
model parameters.
</p>


<h3>Value</h3>

<table>
<tr><td><code>anovatab</code></td>
<td>
<p>anova (analysis of deviance) table</p>
</td></tr>
<tr><td><code>data.df</code></td>
<td>
<p>Data, and contrasts used in fitting the various
models.</p>
</td></tr>
<tr><td><code>aovtab.terms</code></td>
<td>
<p>This string holds, for each model that is
fitted. the terms that have appeared in the model formula.
The text strings for the distinct models are concatenated.</p>
</td></tr>
<tr><td><code>models</code></td>
<td>
<p>Optionally, this holds the complete sequence of
qmodel objects that were fitted</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>References</h3>

<p>1. Huttley, G.A. and Wilson, S.R. 2000. Testing for concordant
equilibria between population samples. <em>Genetics</em> <b>156</b>:2127-2135. <br />
2. Weir, B.S. 1996.  <em>Genetic Data Analysis II.</em> Sinauer.<br />
3. Weir, B.S. and Wilson, S.R. 1986.  Log-linear models for
linked loci. <em>Biometrics</em> <b>42</b>:665-670.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make.contrasts">make.contrasts</a></code>, <code><a href="#topic+decode.genotypes">decode.genotypes</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(IndianIrish)
hwde(data=IndianIrish)
data(mendelABC)
hwde(data=mendelABC, loci=c("seedshape", "cotylcolor", "coatcolor"))
</code></pre>

<hr>
<h2 id='hwexact'>Exact test of HWE</h2><span id='topic+hwexact'></span><span id='topic+snphwe'></span>

<h3>Description</h3>

<p>This function implements an exact SNP test of
Hardy-Weinberg equilibrium. It is a wrapper for <code>snphwe</code></p>


<h3>Usage</h3>

<pre><code class='language-R'>hwexact(obs.hom1, obs.hets, obs.hom2, data, loci, observed)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hwexact_+3A_obs.hom1">obs.hom1</code></td>
<td>
<p>Observed number of individuals homozygous for the common allele.</p>
</td></tr>
<tr><td><code id="hwexact_+3A_obs.hets">obs.hets</code></td>
<td>
<p>Observed number of heterozygous individuals.</p>
</td></tr>
<tr><td><code id="hwexact_+3A_obs.hom2">obs.hom2</code></td>
<td>
<p>Observed number of individuals homozygous for the rare allele.</p>
</td></tr>
<tr><td><code id="hwexact_+3A_data">data</code></td>
<td>
<p>If not <code>NULL</code>, specifies a data frame holding the data.</p>
</td></tr>
<tr><td><code id="hwexact_+3A_loci">loci</code></td>
<td>
<p>If <code>data</code> is not <code>NULL</code>, column that holds some
combination of symbols of the form <code>AA</code>, <code>Aa</code> and <code>aa</code>.</p>
</td></tr>
<tr><td><code id="hwexact_+3A_observed">observed</code></td>
<td>
<p>Column that holds the frequencies, if <code>data</code> is not
<code>NULL</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The exact p-value testing departure from Hardy-Weinberg
Equilibrium, conditional on the observed relative numbers of alleles
of the two types</p>


<h3>Author(s)</h3>

<p>Randall Johnson &ndash; adapted from code by Wigginton et al</p>


<h3>References</h3>

<p>Wigginton, J.E.; Cutler, D.J. &amp; Abecasis, G.R.; A Note on
exact tests of Hardy-Weinberg equilibrium; American Journal of Human
Genetics, 2005, 76, 887-93</p>


<h3>Examples</h3>

<pre><code class='language-R'>hwexact(68, 28, 4)</code></pre>

<hr>
<h2 id='IndianIrish'>Observed genotype frequencies at MN and S loci, for 2 populations</h2><span id='topic+IndianIrish'></span>

<h3>Description</h3>

<p>The <code>IndianIrish</code> data frame has 18 rows and 4 columns.
The data are genotype frequencies for two locations, for
Xavante Indian and Irish populations respectively
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(IndianIrish)</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Population</dt><dd><p>Factor with levels:
<code>Indian</code> and <code>Irish</code> </p>
</dd>
<dt>locus1</dt><dd><p>Factor with levels:
<code>MM</code>, <code>MN</code> and <code>NN</code></p>
</dd>
<dt>locus2</dt><dd><p>Factor with levels:
<code>SS</code>, <code>Ss</code> and <code>ss</code></p>
</dd>
<dt>Observed</dt><dd><p>a numeric vector giving the frequency for each
category of the tale</p>
</dd>
</dl>



<h3>Source</h3>

<p>Mourant et al (1977) and Huttley and Wilson (2000).
</p>


<h3>References</h3>

<p>1. Huttley, G.A. and Wilson, S.R. 2000. Testing for concordant
equilibria between population samples. <em>Genetics</em> <b>156</b>, 2127-2135. <br />
2. Mourant, A.E., Kopec, A.C. and Domaniewska-Sobczak, K. 1976.
<em>The Distribution of the Human Blood Groups and Other Polymorphisms.</em>
Oxford University Press.<br />
3. Weir, B.S. 1996.  <em>Genetic Data Analysis II.</em> Sinauer.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hwde">hwde</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(IndianIrish)
hwde(data=IndianIrish)
</code></pre>

<hr>
<h2 id='make.contrasts'>Calculate contrasts required by the function</h2><span id='topic+make.contrasts'></span>

<h3>Description</h3>

<p>Calculates the contrasts that <code>hwde</code> uses in fitting models
for genotypic disequilibrium.  At present these are limited to
first order interaction terms between the distinct loci,
possibly with distinct values foe each different level of a
grouping factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make.contrasts(data = data[, loci], allele.chars = letters)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make.contrasts_+3A_data">data</code></td>
<td>
<p>Data frame, with one column per locus, and as
many rows as there are observations. Genotypes are represented
using a two-letter code, e.g., AA, Aa, aa</p>
</td></tr>
<tr><td><code id="make.contrasts_+3A_allele.chars">allele.chars</code></td>
<td>
<p>By default the letters &quot;a&quot;, &quot;b&quot;, ...,
one for each locus, are used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Any pair of letters can be used for the two alleles.  The setting
of <code>allele.chars</code> determines the coding used in the model
formulae.  This function is called by <code>hwde</code></p>


<h3>Value</h3>

<table>
<tr><td><code>contrasts.df</code></td>
<td>
<p>Data frame whose columns hold the contrasts that
will be explanatory variables for the Poisson regression model.</p>
</td></tr>
<tr><td><code>oset</code></td>
<td>
<p>Offset vector for the GLM model.</p>
</td></tr>
<tr><td><code>list.columns</code></td>
<td>
<p>List in which each element is a text string
that holds the codes for a sequence of model terms.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>See Also</h3>

<p><code>identify.genotypes</code>, called by <code>make.contrasts</code></p>


<h3>Examples</h3>

<pre><code class='language-R'>make.contrasts(data=c("AA","Aa","aa"))
</code></pre>

<hr>
<h2 id='mendelABC'>Mendel's F2 trifactorial data for seed shape (A: round or wrinkled), 
cotyledon color (B: albumen yellow or green), and seed coat color 
(C: grey-brown or white)</h2><span id='topic+mendelABC'></span>

<h3>Description</h3>

<p>The <code>mendel3</code> data frame has 27 rows and 4 columns. Data are
from Mendel (1886), and are reproduced in Fisher (1936) and Weir (1996).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mendelABC)</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>seedshape</dt><dd><p>Factor with levels:
<code>AA</code>, <code>Aa</code> and <code>aa</code></p>
</dd>
<dt>cotylcolor</dt><dd><p>Factor with levels: 
<code>BB</code>, <code>Bb</code> and <code>bb</code></p>
</dd>
<dt>coatcolor</dt><dd><p>Factor with levels:
<code>CC</code>, <code>Cc</code> and <code>cc</code></p>
</dd>
<dt>Observed</dt><dd><p>a numeric vector that holds the frequencies.</p>
</dd>
</dl>



<h3>Details</h3>

<p>The data are reviewed in detail in Fisher (1936).  For a brief
discussion, and references to work that revisits Fisher's
conclusions, see Weir (1996).
</p>


<h3>Source</h3>

<p>Data are from Mendel (1886), and are reproduced in Fisher (1936) and
Weir (1996).
</p>


<h3>References</h3>

<p>1. Fisher, R.A. 1936. Has Mendel's work been rediscovered?
<em>Annals of Science</em> <b>1</b>:115-137.
</p>
<p>2. Mendel, G. 1886. Versuche uber Pflanzen-Hybriden.  Verhandlugen
des Naturforschenden Vereines in Brunn <b>4</b>:3-47.
(An English translation. with annotations, is available from 
<a href="http://www.esp.org/foundations/genetics/classical/gm-65.pdf">http://www.esp.org/foundations/genetics/classical/gm-65.pdf</a>
NB also the English translation by Royal Horticultural Society of 
London, reprinted in Peters, J.A. 1959. <em>Classic Papers in Genetics.</em>
Prentice-Hall.)
</p>
<p>3. Weir, B.S. 1996.  <em>Genetic Data Analysis II.</em> Sinauer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Lay table out as a 3D array, as in Fisher (1936)
abc &lt;- aperm(array(mendelABC[,4], dim=c(3,3,3)), c(1,3,2))
dimnames(abc) &lt;- list(B=c('BB','Bb','bb'), 
                      A=c('AA','Aa','aa'),
                      C=c('CC','Cc','cc'))
abc
## Fit Hardy-Weinberg disequilibium model
hwde(mendelABC, loci=c("seedshape","cotylcolor","coatcolor"))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
