<!DOCTYPE html><html lang="en"><head><title>Help for package HWEintrinsic</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HWEintrinsic}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HWEintrinsic-package'>
<p>Objective Bayesian Testing for the Hardy-Weinberg Equilibrium Problem</p></a></li>
<li><a href='#cip.2'><p>Graph of the Conditional Intrinsic Prior for a Two Alleles Dataset</p></a></li>
<li><a href='#cip.tmp'><p>Utility Function</p></a></li>
<li><a href='#GuoThompson8'><p>Guo and Thompson (1992) Eight Alleles Simulated Data</p></a></li>
<li><a href='#GuoThompson9'><p>Guo and Thompson (1992) Nine Alleles Data</p></a></li>
<li><a href='#hwe.bf'><p>Standard Bayes Factor for the Hardy-Weinebrg Testing Problem</p></a></li>
<li><a href='#hwe.ibf'><p>Testing Hardy-Weinberg Equilibrium Using an Intrinsic Prior Approach</p></a></li>
<li><a href='#hwe.ibf.mc'><p>Testing Hardy-Weinberg Equilibrium Using an Intrinsic Prior Approach</p></a></li>
<li><a href='#hwe.ibf.plot'><p>Plot of an Hardy-Weinberg Testing Analysis</p></a></li>
<li><a href='#HWEdata-class'><p>Class &quot;HWEdata&quot;. Data specification for the Hardy-Weinberg Testing Problem Using the Intrinsic Prior Approach.</p></a></li>
<li><a href='#HWEintr-class'><p>Class &quot;HWEintr&quot;. Result for the Hardy-Weinberg Testing Problem Using the Intrinsic Prior Approach.</p></a></li>
<li><a href='#ip.2'><p>Graph of the (Unconditional) Intrinsic Prior for a Two Alleles Dataset</p></a></li>
<li><a href='#ip.tmp'><p>Utility Function</p></a></li>
<li><a href='#lik.multin'><p>Utility Function</p></a></li>
<li><a href='#Lindley'><p>Lindley (1988) Two Alleles Data</p></a></li>
<li><a href='#LouisDempster'><p>Louis and Dempster (1987) Four Alleles Data</p></a></li>
<li><a href='#plot-methods'><p>Plot of Hardy-Weinberg Data and Analysis</p></a></li>
<li><a href='#simdata'><p>Two Alleles Simulated Data</p></a></li>
<li><a href='#summary-methods'><p>Summary of Hardy-Weinberg Data and Analysis</p></a></li>
<li><a href='#Wordsworth'><p>Wordsworth et al. (1992) Four Alleles Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Objective Bayesian Testing for the Hardy-Weinberg Equilibrium
Problem</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-05</td>
</tr>
<tr>
<td>Description:</td>
<td>General (multi-allelic) Hardy-Weinberg equilibrium problem from an objective Bayesian testing standpoint. This aim is achieved through the identification of a class of priors specifically designed for this testing problem. A class of intrinsic priors under the full model is considered. This class is indexed by a tuning quantity, the training sample size, as discussed in Consonni, Moreno and Venturini (2010). These priors are objective, satisfy Savage's continuity condition and have proved to behave extremely well for many statistical testing problems.</td>
</tr>
<tr>
<td>Author:</td>
<td>Sergio Venturini</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sergio Venturini &lt;sergio.venturini@unicatt.it&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>methods, utils</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, grDevices, stats</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> | file LICENSE [expanded from: GPL (&ge; 2) | file LICENSE]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-05 14:59:00 UTC; Sergio</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-06 06:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='HWEintrinsic-package'>
Objective Bayesian Testing for the Hardy-Weinberg Equilibrium Problem
</h2><span id='topic+HWEintrinsic-package'></span><span id='topic+HWEintrinsic'></span>

<h3>Description</h3>

<p>General (multi-allelic) Hardy-Weinberg equilibrium problem from an objective Bayesian testing standpoint. This aim is achieved through the identification of a class of priors specifically designed for this testing problem. A class of intrinsic priors under the full model is considered. This class is indexed by a tuning quantity, the training sample size, as discussed in Consonni, Moreno and Venturini (2010). These priors are objective, satisfy Savage's continuity condition and have proved to behave extremely well for many statistical testing problems.
</p>


<h3>Details</h3>

<p>The package is loaded with the usual <code>library(HWEintrinsic)</code> command. The most important functions are <code><a href="#topic+hwe.ibf">hwe.ibf</a></code>, <code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code>, and <code><a href="#topic+hwe.ibf.plot">hwe.ibf.plot</a></code>.
It contains also some data sets which have been extensively used in the literature.   
</p>


<h3>Author(s)</h3>

<p>Sergio Venturini
</p>
<p>Maintainer: Sergio Venturini &lt;sergio.venturini@unicatt.it&gt;
</p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hwe.ibf">hwe.ibf</a></code>, 
<code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code>,
<code><a href="#topic+hwe.ibf.plot">hwe.ibf.plot</a></code>,
<code><a href="#topic+hwe.bf">hwe.bf</a></code>.
</p>

<hr>
<h2 id='cip.2'>Graph of the Conditional Intrinsic Prior for a Two Alleles Dataset</h2><span id='topic+cip.2'></span>

<h3>Description</h3>

<p>This function produces the 3D plot for the conditional intrinsic prior based on a sample of two alleles data as described in Consonni et al. (2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  cip.2(t, p, k = 30)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cip.2_+3A_t">t</code></td>
<td>
<p>training sample size.</p>
</td></tr>
<tr><td><code id="cip.2_+3A_p">p</code></td>
<td>
<p>allele frequency used to condition the prior upon.</p>
</td></tr>
<tr><td><code id="cip.2_+3A_k">k</code></td>
<td>
<p>number of grid points for the alleles frequencies at which the prior is evaluated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No object is returned by this function.
</p>


<h3>Note</h3>

<p>This function provides the plot of the conditional intrinsic prior only for two alleles data.
</p>


<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ip.2">ip.2</a></code>,
<code><a href="#topic+plot-methods">plot-methods</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ATTENTION: the following code may take a long time to run! #

### The following code reproduces Figure 1 in Consonni et al. (2011) ###
p &lt;- 0.5
t &lt;- 5
cip.2(t, p, 30)
t &lt;- 30
cip.2(t, p, 50)

## End(Not run)
</code></pre>

<hr>
<h2 id='cip.tmp'>Utility Function</h2><span id='topic+cip.tmp'></span>

<h3>Description</h3>

<p>This function provides the calculations needed to plot the conditional intrinsic prior based for a two alleles dataset as described in Consonni et al. (2011). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  cip.tmp(p11, p21, t, p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cip.tmp_+3A_p11">p11</code></td>
<td>
<p>gentotype proportion for the alleles pair <code class="reqn">A_{1}A_{1}</code>.</p>
</td></tr>
<tr><td><code id="cip.tmp_+3A_p21">p21</code></td>
<td>
<p>gentotype proportion for the alleles pair <code class="reqn">A_{2}A_{1}</code>.</p>
</td></tr>
<tr><td><code id="cip.tmp_+3A_t">t</code></td>
<td>
<p>training sample size.</p>
</td></tr>
<tr><td><code id="cip.tmp_+3A_p">p</code></td>
<td>
<p>allele frequency used to condition the prior upon.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>cip.tmp</code> returns the value of the conditional intrinsic prior evaluated at the arguments values.
</p>


<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cip.2">cip.2</a></code>,
<code><a href="#topic+plot-methods">plot-methods</a></code>.
</p>

<hr>
<h2 id='GuoThompson8'>Guo and Thompson (1992) Eight Alleles Simulated Data</h2><span id='topic+GuoThompson8'></span>

<h3>Description</h3>

<p>A sample of size <code class="reqn">n = 30</code> of genotype frequencies from a population of <code class="reqn">r = 8</code> alleles simulated under the Hardy-Weinberg equilibrium when the underlying gene frequencies are (.2, .2, .2, .2, .05, .05, .05, .05); see Guo and Thompson (1992), Example 2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GuoThompson8)</code></pre>


<h3>Format</h3>

<p>An object of class <code><a href="#topic+HWEdata">HWEdata</a></code>.
</p>


<h3>Source</h3>

<p>Guo, S.W. and Thompson, E.A. (1992), &quot;Performing the Exact Test of Hardy-Weinberg Proportion for Multiple Alleles&quot;. Biometrics, <b>Vol. 48</b>, No. 2, 361&ndash;372.
</p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1 #
## Not run: 
# ATTENTION: the following code may take a long time to run! #


data(GuoThompson8)
plot(GuoThompson8)
n &lt;- sum(GuoThompson8@data.vec, na.rm = TRUE)
out &lt;- hwe.ibf.mc(GuoThompson8, t = n/2, M = 100000, verbose = TRUE)
summary(out, plot = TRUE)

## End(Not run)

# Example 2 #
## Not run: 
# ATTENTION: the following code may take a long time to run! #

M &lt;- 300000
f &lt;- seq(.1, 1, .05)
n &lt;- sum(GuoThompson8@data.vec, na.rm = TRUE)
out &lt;- hwe.ibf.plot(y = GuoThompson8, t.vec = round(f*n), M = M)

## End(Not run)
</code></pre>

<hr>
<h2 id='GuoThompson9'>Guo and Thompson (1992) Nine Alleles Data</h2><span id='topic+GuoThompson9'></span>

<h3>Description</h3>

<p>These data are extracted (Guo and Thompson, 1992) from the Rhesus data in Cavalli-Sforza and Bodmer (1971). They consists of information on 8297 individuals with <code class="reqn">r = 9</code> groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GuoThompson9)</code></pre>


<h3>Format</h3>

<p>An object of class <code><a href="#topic+HWEdata">HWEdata</a></code>.
</p>


<h3>Source</h3>

<p>Guo, S.W. and Thompson, E.A. (1992), &quot;Performing the Exact Test of Hardy-Weinberg Proportion for Multiple Alleles&quot;. Biometrics, <b>Vol. 49</b>, No. 2, 361&ndash;372.
</p>


<h3>References</h3>

<p>Cavalli-Sforza, L. and Bodmer, W. (1971), &quot;The Genetics of Human Populations&quot;. W.H. Freeman and Company, San Francisco.
Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1 #
## Not run: 
# ATTENTION: the following code may take a long time to run! #

data(GuoThompson9)
plot(GuoThompson9)
n &lt;- sum(GuoThompson9@data.vec, na.rm = TRUE)
out &lt;- hwe.ibf.mc(GuoThompson9, t = n/2, M = 100000, verbose = TRUE)
summary(out, plot = TRUE)

## End(Not run)

# Example 2 #
## Not run: 
# ATTENTION: the following code may take a long time to run! #

M &lt;- 300000
f &lt;- seq(.1, 1, .05)
n &lt;- sum(GuoThompson9@data.vec, na.rm = TRUE)
out &lt;- hwe.ibf.plot(y = GuoThompson9, t.vec = round(f*n), M = M)

## End(Not run)
</code></pre>

<hr>
<h2 id='hwe.bf'>Standard Bayes Factor for the Hardy-Weinebrg Testing Problem</h2><span id='topic+hwe.bf'></span>

<h3>Description</h3>

<p>This function provides the calculations for obtaining the standard Bayes factor for the Hardy-Weinberg testing problem. It implements a common default prior (constant) for both the null and the alternative models.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  hwe.bf(y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hwe.bf_+3A_y">y</code></td>
<td>
<p>an object of <a href="base.html#topic+class">class</a> &quot;<a href="#topic+HWEdata">HWEdata</a>&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>hwe.bf</code> returns the standard Bayes Factor value for the Hardy-Weinberg testing problem (see the references for the details).
</p>


<h3>Note</h3>

<p>The Bayes factor computed here is for the unrestricted model (<code class="reqn">M_1</code>) against the Hardy-Weinberg case (<code class="reqn">M_0</code>).
</p>


<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hwe.ibf">hwe.ibf</a></code>, 
<code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code>,
<code><a href="#topic+hwe.ibf.plot">hwe.ibf.plot</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1 #
data(GuoThompson8)
plot(GuoThompson8)
hwe.bf(GuoThompson8)

# Example 2 #
data(LouisDempster)
plot(LouisDempster)
hwe.bf(LouisDempster)
</code></pre>

<hr>
<h2 id='hwe.ibf'>Testing Hardy-Weinberg Equilibrium Using an Intrinsic Prior Approach</h2><span id='topic+hwe.ibf'></span>

<h3>Description</h3>

<p>This function implements the exact calculation of the Bayes factor based on intrinsic priors for the Hardy-Weinberg testing problem as described in Consonni et al. (2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hwe.ibf(y, t)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hwe.ibf_+3A_y">y</code></td>
<td>
<p>an object of <a href="base.html#topic+class">class</a> &quot;<a href="#topic+HWEdata">HWEdata</a>&quot;.</p>
</td></tr>
<tr><td><code id="hwe.ibf_+3A_t">t</code></td>
<td>
<p>training sample size.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements the exact formula for the Bayes factor based on intrinsic priors.
</p>


<h3>Value</h3>

<p><code>hwe.ibf</code> returns the value of the Bayes factor based on intrinsic priors.
</p>


<h3>Note</h3>

<p>The Bayes factor computed here is for the unrestricted model (<code class="reqn">M_1</code>) against the Hardy-Weinberg case (<code class="reqn">M_0</code>). This function provides the output only for the two alleles case.
</p>


<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hwe.ibf.plot">hwe.ibf.plot</a></code>,
<code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ATTENTION: the following code may take a long time to run! #

data(Lindley)
hwe.ibf.exact &lt;- Vectorize(hwe.ibf, "t")
f &lt;- seq(.05, 1, .05)
n &lt;- sum(dataL1@data.vec, na.rm = TRUE)

# Dataset 1 #
plot(dataL1)
npp.exact &lt;- 1/(1 + hwe.ibf.exact(round(f*n), y = dataL1))
npp.std &lt;- 1/(1 + hwe.bf(dataL1))
plot(f, npp.exact, type="l", lwd = 2, xlab = "f = t/n",
	ylab = "Null posterior probability")
abline(h = npp.std, col = gray(.5), lty = "longdash")

# Dataset 2 #
plot(dataL2)
npp.exact &lt;- 1/(1 + hwe.ibf.exact(round(f*n), y = dataL2))
npp.std &lt;- 1/(1 + hwe.bf(dataL2))
plot(f, npp.exact, type="l", lwd = 2, xlab = "f = t/n",
	ylab = "Null posterior probability")
abline(h = npp.std, col = gray(.5), lty = "longdash")

# Dataset 3 #
plot(dataL3)
npp.exact &lt;- 1/(1 + hwe.ibf.exact(round(f*n), y = dataL3))
npp.std &lt;- 1/(1 + hwe.bf(dataL3))
plot(f, npp.exact, type="l", lwd = 2, xlab = "f = t/n",
	ylab = "Null posterior probability")
abline(h = npp.std, col = gray(.5), lty = "longdash")

# Dataset 4 #
plot(dataL4)
npp.exact &lt;- 1/(1 + hwe.ibf.exact(round(f*n), y = dataL4))
npp.std &lt;- 1/(1 + hwe.bf(dataL4))
plot(f, npp.exact, type="l", lwd = 2, xlab = "f = t/n",
	ylab = "Null posterior probability")
abline(h = npp.std, col = gray(.5), lty = "longdash")

## End(Not run)
</code></pre>

<hr>
<h2 id='hwe.ibf.mc'>Testing Hardy-Weinberg Equilibrium Using an Intrinsic Prior Approach</h2><span id='topic+hwe.ibf.mc'></span>

<h3>Description</h3>

<p>This function implements the Monte Carlo estimation of the Bayes factor based on intrinsic priors for the Hardy-Weinberg testing problem as described in Consonni et al. (2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hwe.ibf.mc(y, t, M = 10000, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hwe.ibf.mc_+3A_y">y</code></td>
<td>
<p>an object of <a href="base.html#topic+class">class</a> &quot;<a href="#topic+HWEdata">HWEdata</a>&quot;.</p>
</td></tr>
<tr><td><code id="hwe.ibf.mc_+3A_t">t</code></td>
<td>
<p>training sample size.</p>
</td></tr>
<tr><td><code id="hwe.ibf.mc_+3A_m">M</code></td>
<td>
<p>number of Monte Carlo iterations.</p>
</td></tr>
<tr><td><code id="hwe.ibf.mc_+3A_verbose">verbose</code></td>
<td>
<p>logical; if TRUE the function prints the detailed calculation progress.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements a Monte Carlo approximation using importance sampling of the Bayes factor based on intrinsic priors.
</p>


<h3>Value</h3>

<p><code>hwe.ibf.mc</code> returns an object of the class &quot;<a href="#topic+HWEintr">HWEintr</a>&quot;.
</p>


<h3>Note</h3>

<p>The Bayes factor computed here is for the unrestricted model (<code class="reqn">M_1</code>) against the Hardy-Weinberg case (<code class="reqn">M_0</code>).
</p>


<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hwe.ibf">hwe.ibf</a></code>, 
<code><a href="#topic+hwe.ibf.plot">hwe.ibf.plot</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1 #
## Not run: 
# ATTENTION: the following code may take a long time to run! #


data(GuoThompson9)
plot(GuoThompson9)
n &lt;- sum(GuoThompson9@data.vec, na.rm = TRUE)
out &lt;- hwe.ibf.mc(GuoThompson9, t = n/2, M = 100000, verbose = TRUE)
summary(out, plot = TRUE)

## End(Not run)

# Example 2 #
## Not run: 
# ATTENTION: the following code may take a long time to run! #

M &lt;- 300000
f &lt;- seq(.1, 1, .05)
n &lt;- sum(GuoThompson9@data.vec, na.rm = TRUE)
out &lt;- hwe.ibf.plot(y = GuoThompson9, t.vec = round(f*n), M = M)

## End(Not run)
</code></pre>

<hr>
<h2 id='hwe.ibf.plot'>Plot of an Hardy-Weinberg Testing Analysis</h2><span id='topic+hwe.ibf.plot'></span>

<h3>Description</h3>

<p>Plot of the null posterior probability of a Hardy-Weinberg testing problem based on intrinsic priors as described in Consonni et al. (2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hwe.ibf.plot(y, t.vec, M = 1e+05, bf = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hwe.ibf.plot_+3A_y">y</code></td>
<td>
<p>an object of <a href="base.html#topic+class">class</a> &quot;<a href="#topic+HWEdata">HWEdata</a>&quot;.</p>
</td></tr>
<tr><td><code id="hwe.ibf.plot_+3A_t.vec">t.vec</code></td>
<td>
<p>vector of training sample sizes.</p>
</td></tr>
<tr><td><code id="hwe.ibf.plot_+3A_m">M</code></td>
<td>
<p>number of Monte Carlo iterations.</p>
</td></tr>
<tr><td><code id="hwe.ibf.plot_+3A_bf">bf</code></td>
<td>
<p>logical: if TRUE the plot reports the Bayes factor based on intrinsic priors.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function allows to create a plot of the null posterior probability versus a given set of training sample sizes. It simply performs a repeated analysis using <code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code> on each of the training sample sizes contained in <code>t.vec</code>.
</p>


<h3>Value</h3>

<p><code><a href="#topic+hwe.ibf.plot">hwe.ibf.plot</a></code> returns as the output an invisible list with the following components:
</p>
<table role = "presentation">
<tr><td><code>mc</code></td>
<td>
<p>matrix containing the Monte Carlo estimates of the Bayes factor and the null posterior probability for each training sample size in <code>t.vec</code>.</p>
</td></tr>
<tr><td><code>std</code></td>
<td>
<p>vector containing the standard Bayes factor and the null posterior probability for the data in hand.</p>
</td></tr>
<tr><td><code>exact</code></td>
<td>
<p>matrix containing the exact values of the Bayes factor and the null posterior probability for each training sample size in <code>t.vec</code>; available only for the two alleles case.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The Bayes factor computed here is for the unrestricted model (<code class="reqn">M_1</code>) against the Hardy-Weinberg case (<code class="reqn">M_0</code>).
</p>


<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hwe.ibf">hwe.ibf</a></code>, 
<code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The following code reproduces Figure 4 in Consonni et al. (2011) #
## Not run: 
# ATTENTION: it may take a long time to run! #

data(simdata)
n &lt;- sum(dataset1@data.vec, na.rm = TRUE)
f &lt;- c(.1,.5,1)
t &lt;- round(f*n)
p11 &lt;- p21 &lt;- seq(0,1,length.out=100)
ip &lt;- array(NA,c(length(f),length(p11),length(p21)))
for (k in 1:length(f)) {
	ip[k,,] &lt;- outer(X = p11, Y = p21, FUN = Vectorize(ip.tmp), t[k])
	print(paste(k," / ",length(f),sep=""), quote = FALSE)
}

r &lt;- 2
R &lt;- r*(r + 1)/2
l &lt;- 4
tables &lt;- matrix(NA, nrow = R, ncol = l)
tables[, 1] &lt;- dataset1@data.vec
tables[, 2] &lt;- dataset2@data.vec
tables[, 3] &lt;- dataset3@data.vec
tables[, 4] &lt;- dataset4@data.vec
lik &lt;- array(NA, c(l, length(p11), length(p21)))
M &lt;- 300000
par(mfrow = c(4, 4))
for (k in 1:l) {
	y &lt;- new("HWEdata", data = tables[, k])
	lik[k,,] &lt;- lik.multin(y, p11, p21)
	
	nlev &lt;- 10
	for (q in 1:length(f)) {
		contour(p11, p21, ip[q,,], xlab = expression(p[11]),
				ylab = expression(p[21]), nlevels = nlev, col = gray(0),
				main = "", cex.axis = 1.75, cex.lab = 1.75, labcex = 1.4)
		lines(p11^2, 2*p11*(1 - p11), lty = "longdash", col = gray(0), lwd = 2)
		contour(p11, p21, lik[k,,], nlevels = nlev, add = TRUE,
				col = gray(.7), labcex = 1.2)
		abline(a = 1, b = -1, lty = 3, col = gray(.8))
	}
	hwe.ibf.plot(y = y, t.vec = seq(1,n,1), M = M)
}

## End(Not run)
</code></pre>

<hr>
<h2 id='HWEdata-class'>Class &quot;HWEdata&quot;. Data specification for the Hardy-Weinberg Testing Problem Using the Intrinsic Prior Approach.</h2><span id='topic+HWEdata-class'></span><span id='topic+HWEdata'></span><span id='topic+initialize+2CHWEdata-method'></span>

<h3>Description</h3>

<p>This class encapsulates the data specification for a Bayesian objective analysis via intrinsic priors of the Hardy-Weinberg Testing Problem as described in Consonni et al. (2011).
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("HWEdata", data)</code>, where <code>data</code> are the data in vector form.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data.mat</code>:</dt><dd><p>Object of class <code>"matrix"</code>; data in matrix form.</p>
</dd>
<dt><code>size</code>:</dt><dd><p>Object of class <code>"numeric"</code>; number of alleles included in the data.</p>
</dd>
<dt><code>data.vec</code>:</dt><dd><p>Object of class <code>"numeric"</code>; data in vector form.</p>
</dd>
</dl>



<h3>Methods</h3>



<dl>
<dt>plot</dt><dd><p><code>signature(x = "HWEdata", y = "missing")</code>: Provides a pictorial representation for a sample of genotype counts.</p>
</dd>
<dt>summary</dt><dd><p><code>signature(object = "HWEdata")</code>: Extracts the contents of an <code><a href="#topic+HWEdata">HWEdata</a></code> object.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hwe.ibf">hwe.ibf</a></code>, 
<code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code>,
<code><a href="#topic+hwe.ibf.plot">hwe.ibf.plot</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data.tmp &lt;- c(3, 9, 8)
dataset &lt;- new("HWEdata", data = data.tmp)
</code></pre>

<hr>
<h2 id='HWEintr-class'>Class &quot;HWEintr&quot;. Result for the Hardy-Weinberg Testing Problem Using the Intrinsic Prior Approach.</h2><span id='topic+HWEintr-class'></span><span id='topic+HWEintr'></span><span id='topic+initialize+2CHWEintr-method'></span>

<h3>Description</h3>

<p>This class encapsulates the results of a Bayesian objective analysis via intrinsic priors for the Hardy-Weinberg Testing Problem as described in Consonni et al. (2011).
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("HWEintr", bf, npp, draws, data)</code>, but most often as the result of a call to <code><a href="#topic+hwe.ibf">hwe.ibf</a></code> or to <code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>bf</code>:</dt><dd><p>Object of class <code>"numeric"</code>; Bayes factor based on intrinsic priors.</p>
</dd>
<dt><code>npp</code>:</dt><dd><p>Object of class <code>"numeric"</code>; posterior probability of the null Hardy-Weinberg model.</p>
</dd>
<dt><code>draws</code>:</dt><dd><p>Object of class <code>"numeric"</code>; individual terms of the Monte Carlo sum using importance sampling.</p>
</dd>
<dt><code>data.mat</code>:</dt><dd><p>Object of class <code>"matrix"</code>; original data in matrix form.</p>
</dd>
</dl>



<h3>Methods</h3>



<dl>
<dt>plot</dt><dd><p><code>signature(x = "HWEintr", y = "missing")</code>: Provides a graphical representation of the estimates.</p>
</dd>
<dt>summary</dt><dd><p><code>signature(object = "HWEintr")</code>: Summarizes the information about the estimates.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hwe.ibf">hwe.ibf</a></code>, 
<code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code>,
<code><a href="#topic+hwe.ibf.plot">hwe.ibf.plot</a></code>.
</p>

<hr>
<h2 id='ip.2'>Graph of the (Unconditional) Intrinsic Prior for a Two Alleles Dataset</h2><span id='topic+ip.2'></span>

<h3>Description</h3>

<p>This function produces the 3D-plot for the (unconditional) intrinsic prior based on a sample of two alleles data as described in Consonni et al. (2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ip.2(t, k = 30)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ip.2_+3A_t">t</code></td>
<td>
<p>training sample size.</p>
</td></tr>
<tr><td><code id="ip.2_+3A_k">k</code></td>
<td>
<p>number of grid points for the alleles frequencies at which the prior is evaluated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No object is returned by this function.
</p>


<h3>Note</h3>

<p>This function provides the plot of the (unconditional) intrinsic prior only for two alleles data.
</p>


<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cip.2">cip.2</a></code>,
<code><a href="#topic+plot-methods">plot-methods</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ATTENTION: the following code may take a long time to run! #

### The following code reproduces Figure 3 in Consonni et al. (2011) ###
t &lt;- 30
ip.2(t, 40)

## End(Not run)
</code></pre>

<hr>
<h2 id='ip.tmp'>Utility Function</h2><span id='topic+ip.tmp'></span>

<h3>Description</h3>

<p>This function provides the calculations needed to plot the (unconditional) intrinsic prior based for a two alleles dataset as described in Consonni et al. (2011). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ip.tmp(p11, p21, t)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ip.tmp_+3A_p11">p11</code></td>
<td>
<p>gentotype proportion for the the pair of alleles <code class="reqn">A_{1}A_{1}</code>.</p>
</td></tr>
<tr><td><code id="ip.tmp_+3A_p21">p21</code></td>
<td>
<p>gentotype proportion for the the pair of alleles <code class="reqn">A_{2}A_{1}</code>.</p>
</td></tr>
<tr><td><code id="ip.tmp_+3A_t">t</code></td>
<td>
<p>training sample size.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>ip.tmp</code> returns the value of the (unconditional) intrinsic prior evaluated at the arguments values.
</p>


<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ip.2">ip.2</a></code>,
<code><a href="#topic+plot-methods">plot-methods</a></code>.
</p>

<hr>
<h2 id='lik.multin'>Utility Function</h2><span id='topic+lik.multin'></span>

<h3>Description</h3>

<p>This function provides the value of the likelihood function of the full (unrestricted) model, as described in Consonni et al. (2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lik.multin(y, p11, p21)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lik.multin_+3A_y">y</code></td>
<td>
<p>an object of <a href="base.html#topic+class">class</a> &quot;<a href="#topic+HWEdata">HWEdata</a>&quot;.</p>
</td></tr>
<tr><td><code id="lik.multin_+3A_p11">p11</code></td>
<td>
<p>gentotype proportion for the alleles pair <code class="reqn">A_{1}A_{1}</code>.</p>
</td></tr>
<tr><td><code id="lik.multin_+3A_p21">p21</code></td>
<td>
<p>gentotype proportion for the alleles pair <code class="reqn">A_{2}A_{1}</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function has been used to generate the likelihood contours that appear in Figure 4 of Consonni et al. (2011) (see the example below). 
</p>


<h3>Value</h3>

<p>This function returns the numerical value of the likelihood in correspondence of the argument values.
</p>


<h3>Note</h3>

<p>This function provides the likelihood function value only for the two alleles case.
</p>


<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hwe.ibf">hwe.ibf</a></code>, 
<code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code>,
<code><a href="#topic+hwe.ibf.plot">hwe.ibf.plot</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The following code reproduces Figure 4 in Consonni et al. (2011) #
## Not run: 
# ATTENTION: it may take a long time to run! #

data(simdata)
n &lt;- sum(dataset1@data.vec, na.rm = TRUE)
f &lt;- c(.1,.5,1)
t &lt;- round(f*n)
p11 &lt;- p21 &lt;- seq(0,1,length.out=100)
ip &lt;- array(NA,c(length(f),length(p11),length(p21)))
for (k in 1:length(f)) {
	ip[k,,] &lt;- outer(X = p11, Y = p21, FUN = Vectorize(ip.tmp), t[k])
	print(paste(k," / ",length(f),sep=""), quote = FALSE)
}

r &lt;- 2
R &lt;- r*(r + 1)/2
l &lt;- 4
tables &lt;- matrix(NA, nrow = R, ncol = l)
tables[, 1] &lt;- dataset1@data.vec
tables[, 2] &lt;- dataset2@data.vec
tables[, 3] &lt;- dataset3@data.vec
tables[, 4] &lt;- dataset4@data.vec
lik &lt;- array(NA, c(l, length(p11), length(p21)))
M &lt;- 300000
par(mfrow = c(4, 4))
for (k in 1:l) {
	y &lt;- new("HWEdata", data = tables[, k])
	lik[k,,] &lt;- lik.multin(y, p11, p21)
	
	nlev &lt;- 10
	for (q in 1:length(f)) {
		contour(p11, p21, ip[q,,], xlab = expression(p[11]),
				ylab = expression(p[21]), nlevels = nlev, col = gray(0),
				main = "", cex.axis = 1.75, cex.lab = 1.75, labcex = 1.4)
		lines(p11^2, 2*p11*(1 - p11), lty = "longdash", col = gray(0), lwd = 2)
		contour(p11, p21, lik[k,,], nlevels = nlev, add = TRUE,
				col = gray(.7), labcex = 1.2)
		abline(a = 1, b = -1, lty = 3, col = gray(.8))
	}
	hwe.ibf.plot(y = y, t.vec = seq(1,n,1), M = M)
}

## End(Not run)
</code></pre>

<hr>
<h2 id='Lindley'>Lindley (1988) Two Alleles Data</h2><span id='topic+Lindley'></span><span id='topic+dataL1'></span><span id='topic+dataL2'></span><span id='topic+dataL3'></span><span id='topic+dataL4'></span>

<h3>Description</h3>

<p>Four samples of <code class="reqn">n = 100</code> genotype counts previously discussed in previously analyzed by Lindley (1988). For the first three sets, the classical &quot;exact&quot; test rejects the null hypothesis of Hardy-Weinberg equilibrium with significance level below 3.4%, whereas for the last data set the Hardy-Weinberg model is not rejected, its p-value being around 20%.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Lindley)</code></pre>


<h3>Format</h3>

<p>Four objects of class <code><a href="#topic+HWEdata">HWEdata</a></code>.
</p>


<h3>Source</h3>

<p>Consonni, G., Gutierrez-Pena, E. and Veronese, P. (2008), &quot;Compatible priors for Bayesian model comparison with an application to the Hardy-Weinberg equilibrium model&quot;. Test, <b>Vol. 17</b>, No. 3, 585&ndash;605.  
</p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
Guo, S.W. and Thompson, E.A. (1992), &quot;Performing the Exact Test of Hardy-Weinberg Proportion for Multiple Alleles&quot;. Biometrics, <b>49</b>, 361&ndash;372.
Lindley D.V. (1988), &quot;Statistical inference concerning Hardy-Weinberg equilibrium&quot;. In: Bernardo, J.M., DeGroot, M.H., Lindley, D.V. and Smith, A.F.M. (eds.), &quot;Bayesian statistics 3&quot;. Oxford University Press, 307&ndash;326.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ATTENTION: the following code may take a long time to run! #

data(Lindley)
hwe.ibf.exact &lt;- Vectorize(hwe.ibf, "t")
f &lt;- seq(.05, 1, .05)
n &lt;- sum(dataL1@data.vec, na.rm = TRUE)

# Dataset 1 #
plot(dataL1)
npp.exact &lt;- 1/(1 + hwe.ibf.exact(round(f*n), y = dataL1))
npp.std &lt;- 1/(1 + hwe.bf(dataL1))
plot(f, npp.exact, type="l", lwd = 2, xlab = "f = t/n",
	ylab = "Null posterior probability")
abline(h = npp.std, col = gray(.5), lty = "longdash")

# Dataset 2 #
plot(dataL2)
npp.exact &lt;- 1/(1 + hwe.ibf.exact(round(f*n), y = dataL2))
npp.std &lt;- 1/(1 + hwe.bf(dataL2))
plot(f, npp.exact, type="l", lwd = 2, xlab = "f = t/n",
	ylab = "Null posterior probability")
abline(h = npp.std, col = gray(.5), lty = "longdash")

# Dataset 3 #
plot(dataL3)
npp.exact &lt;- 1/(1 + hwe.ibf.exact(round(f*n), y = dataL3))
npp.std &lt;- 1/(1 + hwe.bf(dataL3))
plot(f, npp.exact, type="l", lwd = 2, xlab = "f = t/n",
	ylab = "Null posterior probability")
abline(h = npp.std, col = gray(.5), lty = "longdash")

# Dataset 4 #
plot(dataL4)
npp.exact &lt;- 1/(1 + hwe.ibf.exact(round(f*n), y = dataL4))
npp.std &lt;- 1/(1 + hwe.bf(dataL4))
plot(f, npp.exact, type="l", lwd = 2, xlab = "f = t/n",
	ylab = "Null posterior probability")
abline(h = npp.std, col = gray(.5), lty = "longdash")

## End(Not run)
</code></pre>

<hr>
<h2 id='LouisDempster'>Louis and Dempster (1987) Four Alleles Data</h2><span id='topic+LouisDempster'></span>

<h3>Description</h3>

<p>Sample of <code class="reqn">n = 45</code> genotype counts previously discussed in Louis and Dempster (1987) and Guo and Thompson (1992, Example 1). These data are described in Thomson et al. (1986) and concern the antigen class of 45 French type 1 diabetes patients, with the classes being DR1, DR3, DR4, and Y, a fourth class corresponding to all other antigens.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LouisDempster)</code></pre>


<h3>Format</h3>

<p>An object of class <code><a href="#topic+HWEdata">HWEdata</a></code>.
</p>


<h3>Source</h3>

<p>Louis, E. and Dempster, E. (1987), &quot;An Exact Test for Hardy-Weinberg and Multiple Alleles&quot;. Biometrics <b>Vol. 43</b>, No. 4, 805&ndash;811.
</p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
Guo, S.W. and Thompson, E.A. (1992), &quot;Performing the Exact Test of Hardy-Weinberg Proportion for Multiple Alleles&quot;. Biometrics, <b>Vol. 49</b>, No. 2, 361&ndash;372.
Thomson, G., Klitz, W., Louis, E., Lo, S., Bertrams, L., Baur, M., and Neugebauer, M. (1986), &quot;HLA and IDDM predisposition: New aspects&quot;. Genetic Epidemiology, <b>Vol. 1</b>, No. 2, 363&ndash;368.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1 #
## Not run: 
# ATTENTION: the following code may take a long time to run! #


data(LouisDempster)
plot(LouisDempster)
n &lt;- sum(LouisDempster@data.vec, na.rm = TRUE)
out &lt;- hwe.ibf.mc(LouisDempster, t = n/2, M = 100000, verbose = TRUE)
summary(out, plot = TRUE)

## End(Not run)

# Example 2 #
## Not run: 
# ATTENTION: the following code may take a long time to run! #

M &lt;- 300000
f &lt;- seq(.1, 1, .05)
n &lt;- sum(LouisDempster@data.vec, na.rm = TRUE)
out &lt;- hwe.ibf.plot(y = LouisDempster, t.vec = round(f*n), M = M)

## End(Not run)
</code></pre>

<hr>
<h2 id='plot-methods'>Plot of Hardy-Weinberg Data and Analysis</h2><span id='topic+plot-methods'></span><span id='topic+plot+2CANY+2CANY-method'></span><span id='topic+plot+2CHWEdata+2Cmissing-method'></span><span id='topic+plot+2CHWEintr+2Cmissing-method'></span>

<h3>Description</h3>

<p>Methods for function <code>plot</code> in Package &lsquo;graphics&rsquo; to be used with &quot;<a href="#topic+HWEdata">HWEdata</a>&quot; and &quot;<a href="#topic+HWEintr">HWEintr</a>&quot; objects.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "HWEdata", y = "missing")</code></dt><dd>
<p>Pictorial representation for a &quot;<a href="#topic+HWEdata">HWEdata</a>&quot; object.
</p>
</dd>
<dt><code>signature(x = "HWEintr", y = "missing")</code></dt><dd>
<p>Graphical representation of Monte Carlo sums for a &quot;<a href="#topic+HWEintr">HWEintr</a>&quot; object.
</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hwe.ibf">hwe.ibf</a></code>, 
<code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code>,
<code><a href="#topic+hwe.ibf.plot">hwe.ibf.plot</a></code>.
</p>

<hr>
<h2 id='simdata'>Two Alleles Simulated Data</h2><span id='topic+simdata'></span><span id='topic+dataset1'></span><span id='topic+dataset2'></span><span id='topic+dataset3'></span><span id='topic+dataset4'></span>

<h3>Description</h3>

<p>Four different samples of <code class="reqn">n = 20</code> genotype counts simulated under the Hardy-Weinberg equilibrium model. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(simdata)</code></pre>


<h3>Format</h3>

<p>Four objects of class <code><a href="#topic+HWEdata">HWEdata</a></code>.
</p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(simdata)
summary(dataset1)
plot(dataset1)
summary(dataset2)
plot(dataset2)
summary(dataset3)
plot(dataset3)
summary(dataset4)
plot(dataset4)

# The following code reproduces Figure 4 in Consonni et al. (2011) #
## Not run: 
# ATTENTION: it may take a long time to run! #

n &lt;- sum(dataset1@data.vec, na.rm = TRUE)
f &lt;- c(.1,.5,1)
t &lt;- round(f*n)
p11 &lt;- p21 &lt;- seq(0,1,length.out=100)
ip &lt;- array(NA,c(length(f),length(p11),length(p21)))
for (k in 1:length(f)) {
	ip[k,,] &lt;- outer(X = p11, Y = p21, FUN = Vectorize(ip.tmp), t[k])
	print(paste(k," / ",length(f),sep=""), quote = FALSE)
}

r &lt;- 2
R &lt;- r*(r + 1)/2
l &lt;- 4
tables &lt;- matrix(NA, nrow = R, ncol = l)
tables[, 1] &lt;- dataset1@data.vec
tables[, 2] &lt;- dataset2@data.vec
tables[, 3] &lt;- dataset3@data.vec
tables[, 4] &lt;- dataset4@data.vec
lik &lt;- array(NA, c(l, length(p11), length(p21)))
M &lt;- 300000
par(mfrow = c(4, 4))
for (k in 1:l) {
	y &lt;- new("HWEdata", data = tables[, k])
	lik[k,,] &lt;- lik.multin(y, p11, p21)
	
	nlev &lt;- 10
	for (q in 1:length(f)) {
		contour(p11, p21, ip[q,,], xlab = expression(p[11]),
				ylab = expression(p[21]), nlevels = nlev, col = gray(0),
				main = "", cex.axis = 1.75, cex.lab = 1.75, labcex = 1.4)
		lines(p11^2, 2*p11*(1 - p11), lty = "longdash", col = gray(0), lwd = 2)
		contour(p11, p21, lik[k,,], nlevels = nlev, add = TRUE,
				col = gray(.7), labcex = 1.2)
		abline(a = 1, b = -1, lty = 3, col = gray(.8))
	}
	hwe.ibf.plot(y = y, t.vec = seq(1,n,1), M = M)
}

## End(Not run)
</code></pre>

<hr>
<h2 id='summary-methods'>Summary of Hardy-Weinberg Data and Analysis</h2><span id='topic+summary-methods'></span><span id='topic+summary+2CANY-method'></span><span id='topic+summary+2CHWEdata-method'></span><span id='topic+summary+2CHWEintr-method'></span>

<h3>Description</h3>

<p>Methods for function <code>summary</code> in Package &lsquo;base&rsquo; to be used with &quot;<a href="#topic+HWEdata">HWEdata</a>&quot; and &quot;<a href="#topic+HWEintr">HWEintr</a>&quot; objects.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "HWEdata")</code></dt><dd>
<p>Extracts the slots of a &quot;<a href="#topic+HWEdata">HWEdata</a>&quot; object.
</p>
</dd>
<dt><code>signature(object = "HWEintr")</code></dt><dd>
<p>Extracts the slots of a &quot;<a href="#topic+HWEintr">HWEintr</a>&quot; object.
</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Sergio Venturini <a href="mailto:sergio.venturini@unicatt.it">sergio.venturini@unicatt.it</a> </p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hwe.ibf">hwe.ibf</a></code>, 
<code><a href="#topic+hwe.ibf.mc">hwe.ibf.mc</a></code>,
<code><a href="#topic+hwe.ibf.plot">hwe.ibf.plot</a></code>.
</p>

<hr>
<h2 id='Wordsworth'>Wordsworth et al. (1992) Four Alleles Data</h2><span id='topic+Wordsworth'></span>

<h3>Description</h3>

<p>Sample of <code class="reqn">n = 230</code> genotype counts discussed in Lauretto et al. (2009, Example 3). These data come from a rheumatoid arthritis (RA) study performed by Wordsworth et al. (1992), where two hundred and thirty RA patients were genotyped for the HLA-DR locus. The DR4 allele was subdivided into Dw4, Dw14 and other subtypes. DRX represents all non-DR1, non-Dw4, non-Dw14 alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Wordsworth)</code></pre>


<h3>Format</h3>

<p>An object of class <code><a href="#topic+HWEdata">HWEdata</a></code>.
</p>


<h3>Source</h3>

<p>Lauretto, M.S., Nakano, F., Faria, S.R., Pereira, C.A.B. and Stern, J.M. (2009), &quot;A straightforward multiallelic significance test for the Hardy-Weinberg equilibrium law&quot;. Genetics and Molecular Biology, <b>Vol. 32</b>, No. 3, 619&ndash;625.
</p>


<h3>References</h3>

<p>Consonni, G., Moreno, E., and Venturini, S. (2011). &quot;Testing Hardy-Weinberg equilibrium: an objective Bayesian analysis&quot;. Statistics in Medicine, <b>30</b>, 62&ndash;74. <a href="https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract">https://onlinelibrary.wiley.com/doi/10.1002/sim.4084/abstract</a>
Wordsworth, P., Pile, K.D., Buckley, J.D., Lanchbury, J.S.S., Ollier, B., Lathrop, M. and Bell, J.I. (1992), &quot;HLA heterozygosity contributes to susceptibility to rheumatoid arthritis&quot;. American Journal of Human Genetics, <b>51</b>, 3, 585&ndash;591.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1 #
## Not run: 
# ATTENTION: the following code may take a long time to run! #


data(Wordsworth)
plot(Wordsworth)
n &lt;- sum(Wordsworth@data.vec, na.rm = TRUE)
out &lt;- hwe.ibf.mc(Wordsworth, t = n/2, M = 100000, verbose = TRUE)
summary(out, plot = TRUE)

## End(Not run)

# Example 2 #
## Not run: 
# ATTENTION: the following code may take a long time to run! #

data(Wordsworth)
n &lt;- sum(Wordsworth@data.vec, na.rm = TRUE)
M &lt;- 300000
f &lt;- seq(.1, 1, .05)
out &lt;- hwe.ibf.plot(y = Wordsworth, t.vec = round(f*n), M = M)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
