<!DOCTYPE html><html><head><title>Help for package HybridMicrobiomes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {HybridMicrobiomes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FourHbootstrap'>
<p>Bootstraps of the incidence-based 4H Index</p></a></li>
<li><a href='#FourHbootstrapA'>
<p>Bootstraps of the abundance-based 4H Index</p></a></li>
<li><a href='#FourHcentroid'>
<p>Centroid of bootstrapped 4H-indices</p></a></li>
<li><a href='#FourHcompare'>
<p>Tests whether multiple systems have significantly different 4H-indices</p></a></li>
<li><a href='#FourHnull'>
<p>Bootstrapped null models of the 4H-index</p></a></li>
<li><a href='#FourHnullplane'>
<p>Null Plane</p></a></li>
<li><a href='#FourHnullplaneD'>
<p>Bootstrap Distribution Around the Null Plane</p></a></li>
<li><a href='#FourHpreanalysis'>
<p>Bootstraps of the percentage of gamma diversity that is core</p></a></li>
<li><a href='#FourHquaternary'>
<p>Plot bootstrapped 4H-indices</p></a></li>
<li><a href='#FourHquaternaryC'>
<p>Plot centroid of bootstrapped 4H-indices</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysis of Host-Associated Microbiomes from Hybrid Organisms</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A set of tools to analyze and visualize the relationships between host-associated microbiomes of hybrid organisms and those of their progenitor species. Though not necessary, installing the microViz package is recommended as a check for phyloseq objects. To install microViz from R Universe use the following command: install.packages("microViz", repos = c(davidbarnett = "https://david-barnett.r-universe.dev", getOption("repos"))). To install microViz from GitHub use the following commands: install.packages("devtools") followed by devtools::install_github("david-barnett/microViz").</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Software, phyloseq</td>
</tr>
<tr>
<td>Imports:</td>
<td>rgl, compositions, rlang, PERMANOVA, graphics, stats,vegan,
ks, ape, geometry,StereoMorph, phyloseq</td>
</tr>
<tr>
<td>Suggests:</td>
<td>microViz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-05 00:33:49 UTC; sbewick</td>
</tr>
<tr>
<td>Author:</td>
<td>Benjamin Camper [aut],
  Zachary Lauglin [ctb],
  Daniel Malagon [ctb],
  Robert Denton [ctb],
  Sharon Bewick [aut, cre],
  National Science Foundation Division of Integrative Organismal Systems
    (award #2104605) [fnd],
  Clemson University Support for Early Exploration and Development (CU
    SEED) Grant [fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sharon Bewick &lt;sbewick@clemson.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-05 23:20:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='FourHbootstrap'>
Bootstraps of the incidence-based 4H Index
</h2><span id='topic+FourHbootstrap'></span>

<h3>Description</h3>

<p>Calculates incidence-based 4H Indices for bootstrapped samples drawn from an object of class phyloseq containing host-associated microbial community data from both hybrid hosts and hosts of each parent species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FourHbootstrap(x, class_grouping, core_fraction, boot_no, sample_no,
core_average_abundance = 0, core_max_abundance = 0,
core_all_average_abundance = 0, core_all_max_abundance = 0,
replace_hosts=FALSE, reads=NULL, rarefy_each_step=TRUE,seed=NULL,dist='Jaccard')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FourHbootstrap_+3A_x">x</code></td>
<td>

<p>(Required) Host-associated microbial community data . This must be in the form of a phyloseq object and must contain an OTU abundance table with information on the host-associated microbial communities from both hybrid hosts and hosts of each progenitor taxon. The OTU abundance table should contain read counts, not fractional abundances.
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_class_grouping">class_grouping</code></td>
<td>

<p>(Required) A numerical vector identifying the host class of each host from the OTU table. This vector must be ordered according to the OTU abundance table, and must use the following notation: Progenitor One = 1, Hybrids = 2, Progenitor Two = 3.
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_core_fraction">core_fraction</code></td>
<td>

<p>(Required) The fraction of hosts that a microbial taxon must be found on to be considered part of a host class's 'core' microbiome
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_boot_no">boot_no</code></td>
<td>

<p>(Required) The desired number of bootstrap samples
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_sample_no">sample_no</code></td>
<td>

<p>(Required) The number of hosts of each host class that should be selected to generate each bootstrap sample. This number should be less than the minimum number of hosts in any given class.
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_core_average_abundance">core_average_abundance</code></td>
<td>

<p>The minimum average abundance at which a microbial taxon must be found across all hosts within a class to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum average abundance)
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_core_max_abundance">core_max_abundance</code></td>
<td>

<p>The minimum abundance at which a microbial taxon must be found on at least one host within a class to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum abundance on any single host)
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_core_all_average_abundance">core_all_average_abundance</code></td>
<td>

<p>The minimum average abundance at which a microbial taxon must be found across all hosts (regardless of class) to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum average abundance)
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_core_all_max_abundance">core_all_max_abundance</code></td>
<td>

<p>The minimum abundance at which a microbial taxon must be found on at least one host (regardless of class) to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum abundance on any single host)
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_replace_hosts">replace_hosts</code></td>
<td>

<p>Whether or not to replace hosts during bootstrapping. The default is FALSE (i.e., all hosts in a given bootstrap are unique).
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_reads">reads</code></td>
<td>

<p>The number of reads to rarefy to for each microbiome sample. The default is to use the number of reads in the sample with the lowest number of reads. Any user-specified value must be lower than the number of reads in the sample with the lowest number of reads.
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_rarefy_each_step">rarefy_each_step</code></td>
<td>

<p>Whether or not to rarefy each bootstrap sample separately. The default is TRUE (i.e., perform a new rarefaction on the OTU table for each bootstrap sample). However, for cases with large numbers of samples, large numbers of reads per sample or high microbial diversity, this can be slow; thus, there is the option to rarefy the microbiome samples a single time prior to bootstrapping.
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_seed">seed</code></td>
<td>

<p>Optional seed for random number generator. The default is NULL (i.e., seed picked at random at time of simulation).
</p>
</td></tr>
<tr><td><code id="FourHbootstrap_+3A_dist">dist</code></td>
<td>

<p>How to count taxa present on one, two or three host classes when calculating the 4H Index. Use 'Jaccard' to count each taxon once, regardless of the number of host classes it occurs on (i.e., inspired by the Jaccard Index for beta diversity). Use 'Sorensen' to count each taxon proportional to the number of host classes it occurs on (i.e., inspired by the Sorensen Index for beta diversity). The default is 'Jaccard'.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>FourHbootstrap</code> generates bootstrapped samples of the incidence-based 4H Index. Each bootstrap begins by rarefying the OTU table such that all samples have the same library size. Once the OTU table has been rarefied, a subset, <code>sample_no</code>, of hosts is selected from each host class. Hosts can be sampled with or without replacement. The default is to sample hosts without replacement, such that all hosts in any given bootstrap sample are unique. Subsampled individuals from each host class are then used to determine core microbiomes. This is done by selecting all microbial taxa that occur on a threshold number of individuals of that host class (alternately, different abundance thresholds can also be used, see above). Richness of core microbiomes shared among various host classes are then used to calculate a 4H Index for each bootstrap sample.  Specifically, we define:
</p>
<p style="text-align: center;"><code class="reqn">a=|(P_1 \cap P_2) \cap H|</code>
</p>

<p style="text-align: center;"><code class="reqn">b=|(P_1 \cup P_2) \cap H|-a</code>
</p>

<p style="text-align: center;"><code class="reqn">b_1=|P_1 \cap H|-a</code>
</p>

<p style="text-align: center;"><code class="reqn">b_2=|P_2 \cap H|-a</code>
</p>

<p style="text-align: center;"><code class="reqn">c=|H|-a-b</code>
</p>

<p style="text-align: center;"><code class="reqn">d=|P_1 \cup P_2 \cup H|-a-b-c</code>
</p>

<p style="text-align: center;"><code class="reqn">d_1=|P_1|-|P_1 \cap P_2| - |P_1 \cap H|+a</code>
</p>

<p style="text-align: center;"><code class="reqn">d_2=|P_2|-|P_1 \cap P_2| - |P_2 \cap H|+a</code>
</p>

<p style="text-align: center;"><code class="reqn">d_12=d-d_1-d_2</code>
</p>

<p>where <code class="reqn">P_1</code> is the set of core microbial taxa from the first parent species, <code class="reqn">P_2</code> is the set of core microbial taxa from the second parent species, <code class="reqn">H</code> is the set of core microbial taxa from hybrid organisms, and <code class="reqn">|S|</code> denotes the cardinality of set <code class="reqn">S</code>. Thus, <code class="reqn">a</code> is the number of microbial taxa shared by both progenitors and the hybrid, <code class="reqn">b</code> is the number of microbial taxa shared by one progenitor (but not both) and the hybrid, <code class="reqn">b_1</code> is the number of microbial taxa shared by the first progenitor and the hybrid, <code class="reqn">b_2</code> is the number of microbial taxa shared by the second progenitor and the hybrid, <code class="reqn">c</code> is the number of microbial taxa found only on the hybrid, <code class="reqn">d</code> is the number of microbial taxa found only on one or both progenitors, <code class="reqn">d_1</code> is the number of microbial taxa found only on the first progenitor, <code class="reqn">d_2</code> is the number of microbial taxa found only on the second progenitor, and <code class="reqn">d_12</code> is the number of microbial taxa found on both progenitors.
</p>
<p>When dist='Jaccard', the 4H Index is calculated as:
</p>
<p style="text-align: center;"><code class="reqn">\mathcal{U}=\frac{b}{a+b+c+d}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{U}_1=\frac{b_1}{a+b+c+d}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{U}_2=\frac{b_2}{a+b+c+d}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{I}=\frac{a}{a+b+c+d}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{G}=\frac{c}{a+b+c+d}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{L}=\frac{d}{a+b+c+d}=1 - \mathcal{U} -\mathcal{I} -\mathcal{G}</code>
</p>

<p><code class="reqn">\mathcal{U}</code>, <code class="reqn">\mathcal{I}</code>, <code class="reqn">\mathcal{G}</code> and <code class="reqn">\mathcal{L}</code> represent the extent of the Union, Intersection, Gain and Loss Models respectively (see Camper et al., 2023) and <code class="reqn">\mathcal{U}_1</code> and <code class="reqn">\mathcal{U}_2</code> partition the Union model into components associated with the first and second progenitor.
</p>
<p>When dist = 'Sorensen', the 4H Index is calculated as:
</p>
<p style="text-align: center;"><code class="reqn">\mathcal{U}=\frac{2b}{3a+2b+c+d_1+d_2+2d_12}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{U}_1=\frac{2b_1}{3a+2b+c+d_1+d_2+2d_12}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{U}_2=\frac{2b_2}{3a+2b+c+d_1+d_2+2d_12}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{I}=\frac{3a}{3a+2b+c+d_1+d_2+2d_12}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{G}=\frac{c}{3a+2b+c+d_1+d_2+2d_12}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{L}=\frac{d_1+d_2+2d_12}{3a+2b+c+d_1+d_2+2d_12}=1 - \mathcal{U} -\mathcal{I} -\mathcal{G}</code>
</p>

<p>In addition to the four dimensions of the 4H Index, <code>FourHbootstrap</code> also calculates <code class="reqn">\sigma</code> for each bootstrap sample:
</p>
<p>Jaccard:
</p>
<p style="text-align: center;"><code class="reqn">\sigma=\frac{a+d_12}{a+d_12+d_1+d_2}</code>
</p>

<p>Sorensen:
</p>
<p style="text-align: center;"><code class="reqn">\sigma=\frac{2(a+d_12)}{2(a+d_12)+d_1+d_2}</code>
</p>

<p>where <code class="reqn">\sigma</code> is the fraction of parental microbial taxa that are found on both parent species. This will be used later for calculating the null plane of a given hybrid complex (see <code>FourHnullplane</code>).
</p>


<h3>Value</h3>

<p>This function returns a matrix, where rows are individual bootstrap samples and columns are the four dimensions of the 4H Index, as well as <code class="reqn">\sigma</code>, <code class="reqn">\mathcal{U}_1</code>, and <code class="reqn">\mathcal{U}_2</code>.
</p>


<h3>References</h3>

<p>Camper, B., Laughlin, Z. et al. &quot;A Conceptual Framework for Host-Associated Microbiomes of Hybrid Organisms&quot; &lt;doi:10.1101/2023.05.01.538925&gt;
</p>
<p>Henry, L., Wickham H., et al. &quot;rlang: Functions for Base Types and Core R and 'Tidyverse' Features&quot; https://CRAN.R-project.org/package=rlang
</p>
<p>McMurdie, Paul J., and Susan Holmes. &quot;phyloseq: an R package for reproducible interactive analysis and graphics of microbiome census data.&quot; PloS one 8.4 (2013): e61217.
</p>
<p>McMurdie, Paul J., and Susan Holmes. &quot;Phyloseq: a bioconductor package for handling and analysis of high-throughput phylogenetic sequence data.&quot; Biocomputing 2012. 2012. 235-246.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Test with enterotype dataset
library(phyloseq)
data(enterotype)
#Covert the OTU table to reads, rather than fractional abundances
otu_table(enterotype)&lt;-round(10000*otu_table(enterotype))

#Randomly assign host classes (these should be known in a real hybrid microbiome dataset)
#The two parent species are assigned '1' and '3' respectively, the hybrid is assigned '2'
hybrid_status&lt;-sample(1:3,280, replace=TRUE)

FourHbootstrap(enterotype,hybrid_status,0.5,5,10)

</code></pre>

<hr>
<h2 id='FourHbootstrapA'>
Bootstraps of the abundance-based 4H Index
</h2><span id='topic+FourHbootstrapA'></span>

<h3>Description</h3>

<p>Calculates abundance-based 4H Indices for bootstrapped samples drawn from an object of class phyloseq containing host-associated microbial community data from both hybrid hosts and hosts of each parent species. The phyloseq object must, at a minimum, contain an OTU table AND sample data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FourHbootstrapA(x, class_grouping, core_fraction, boot_no, sample_no,
core_average_abundance = 0, core_max_abundance = 0,
core_all_average_abundance = 0, core_all_max_abundance = 0,
replace_hosts=FALSE, reads=NULL, rarefy_each_step=TRUE,seed=NULL,
dist='Bray-Curtis',representative='mean',rescale_core=FALSE, use_microViz = 'yes')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FourHbootstrapA_+3A_x">x</code></td>
<td>

<p>(Required) Host-associated microbial community data . This must be in the form of a phyloseq object and must contain an OTU abundance table with information on the host-associated microbial communities from both hybrid hosts and hosts of each progenitor taxon. The OTU abundance table should contain read counts, not fractional abundances.
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_class_grouping">class_grouping</code></td>
<td>

<p>(Required) A numerical vector identifying the host class of each host from the OTU table. This vector must be ordered according to the OTU abundance table, and must use the following notation: Progenitor One = 1, Hybrids = 2, Progenitor Two = 3.
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_core_fraction">core_fraction</code></td>
<td>

<p>(Required) The fraction of hosts that a microbial taxon must be found on to be considered part of a host class's 'core' microbiome
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_boot_no">boot_no</code></td>
<td>

<p>(Required) The desired number of bootstrap samples
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_sample_no">sample_no</code></td>
<td>

<p>(Required) The number of hosts of each host class that should be selected to generate each bootstrap sample. This number should be less than the minimum number of hosts in any given class.
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_core_average_abundance">core_average_abundance</code></td>
<td>

<p>The minimum average abundance at which a microbial taxon must be found across all hosts within a class to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum average abundance)
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_core_max_abundance">core_max_abundance</code></td>
<td>

<p>The minimum abundance at which a microbial taxon must be found on at least one host within a class to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum abundance on any single host)
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_core_all_average_abundance">core_all_average_abundance</code></td>
<td>

<p>The minimum average abundance at which a microbial taxon must be found across all hosts (regardless of class) to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum average abundance)
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_core_all_max_abundance">core_all_max_abundance</code></td>
<td>

<p>The minimum abundance at which a microbial taxon must be found on at least one host (regardless of class) to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum abundance on any single host)
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_replace_hosts">replace_hosts</code></td>
<td>

<p>Whether or not to replace hosts during bootstrapping. The default is FALSE (i.e., all hosts in a given bootstrap are unique).
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_reads">reads</code></td>
<td>

<p>The number of reads to rarefy to for each microbiome sample. The default is to use the number of reads in the sample with the lowest number of reads. Any user-specified value must be lower than the number of reads in the sample with the lowest number of reads.
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_rarefy_each_step">rarefy_each_step</code></td>
<td>

<p>Whether or not to rarefy each bootstrap sample separately. The default is TRUE (i.e., perform a new rarefaction on the OTU table for each bootstrap sample). However, for cases with large numbers of samples, large numbers of reads per sample or high microbial diversity, this can be slow; thus, there is the option to rarefy the microbiome samples a single time prior to bootstrapping.
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_seed">seed</code></td>
<td>

<p>Optional seed for random number generator. The default is NULL (i.e., seed picked at random at time of simulation).
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_dist">dist</code></td>
<td>

<p>How to count taxa present on one, two or three host classes when calculating the 4H Index. Use 'Ruzicka' to count each taxon once, regardless of the number of host classes it occurs on (i.e., inspired by the Ruzicka Index for beta diversity). Use 'Bray-Curtis' to count each taxon proportional to the number of host classes it occurs on (i.e., inspired by the Bray-Curtis Index for beta diversity).
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_representative">representative</code></td>
<td>

<p>The method used to find representative microbial relative abundances for each host class. Options are 'mean', which is the average abundance of all microbial taxa found on individuals within the host class, and 'median' which is the median abundance of all microbial taxa found on individuals within the host class. If 'median' is chosen, then representative relative abundances are renormalized for each host class such that total abundances of all microbial taxa (core and non-core) are the same across host classes. The default is 'mean'.
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_rescale_core">rescale_core</code></td>
<td>

<p>Whether or not to renormalize relative abundances across the subset of microbial taxa that comprise the core of each host class. Renormalizing the core microbial abundances of each host class results in the 4H Index being density invariant. However, this also constrains the 4H Index to a 2D plane, since there is no variation in total abundance across host classes. The default is to use raw relative abundances without renormalizing. Although this means that the 4H Index is not density invariant, in this case, differences in total abundance of the core microbiome between host classes are meaningful since they represent variation in the extent to which the overall microbiome of a host class is shared among individual animalss.
</p>
</td></tr>
<tr><td><code id="FourHbootstrapA_+3A_use_microviz">use_microViz</code></td>
<td>

<p>Whether or not to use the microViz package to check your phyloseq object. If installation of the microViz package is problematic, this test can be skipped, but the user should verify themselves that a phyloseq object exists AND includes sample data (the sample data placeholder is necessary for the FourHbootstrapA code to work).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>FourHbootstrapA</code> generates bootstrapped samples of the abundance-based 4H Index. Each bootstrap begins by rarefying the OTU table such that all samples have the same library size. Once the OTU table has been rarefied, a subset, <code>sample_no</code>, of hosts is selected from each host class. Hosts can be sampled with or without replacement. The default is to sample hosts without replacement, such that all hosts in any given bootstrap sample are unique. Subsampled individuals from each host class are then used to (i) find representative relative abundances of each microbial taxon found on that host class and (ii) identify the core microbiome of the host class. Representative relative abundances are found by averaging or finding the median relative abundance of each microbial taxon across a particular host class. Core microbial taxa are identified by selecting all microbial taxa that occur on a threshold number of individuals of that host class (alternately, different abundance thresholds can also be used, see above). Once representative relative abundances of core microbial taxa have been determined, the core microbiome may or may not be renormalized (the default is to use the core microbial abundances without renormalization). Representative relative abundances are then used to determine whether certain host classes have an excess/deficit of abundance of any particular microbial taxa. Specifically, we define:
</p>
<p style="text-align: center;"><code class="reqn">A=\sum_{i=1}^{S}min(min(x_{iP_1},x_{iP_2}),x_{iH})=\sum_{i=1}^{S}\alpha_i</code>
</p>

<p style="text-align: center;"><code class="reqn">B=\sum_{i=1}^{S}[min(x_{iP_1}-\alpha_i,x_{iH}-\alpha_i)+min(x_{iP_2}-\alpha_i,x_{iH}-\alpha_i)]=\sum_{i=1}^{S}(\beta_{1i}+\beta_{2i})</code>
</p>

<p style="text-align: center;"><code class="reqn">B_1=\sum_{i=1}^{S}min(x_{iP_1}-\alpha_i,x_{iH}-\alpha_i)=\sum_{i=1}^{S}(\beta_{1i})</code>
</p>

<p style="text-align: center;"><code class="reqn">B_2=\sum_{i=1}^{S}min(x_{iP_2}-\alpha_i,x_{iH}-\alpha_i)=\sum_{i=1}^{S}(\beta_{2i})</code>
</p>

<p style="text-align: center;"><code class="reqn">C=\sum_{i=1}^{S}(x_{iH}-\alpha_i-\beta_{1i}-\beta_{2i})</code>
</p>

<p style="text-align: center;"><code class="reqn">D=\sum_{i=1}^{S}[x_{iP_1}+x_{iP_2}-\alpha_i-\beta_{1i}-\beta_{2i}-min(x_{iP_1},x_{iP_2})]</code>
</p>

<p style="text-align: center;"><code class="reqn">D_1=\sum_{i=1}^{S}[x_{iP_1}-\beta_{1i}-min(x_{iP_1},x_{iP_2})]</code>
</p>

<p style="text-align: center;"><code class="reqn">D_2=\sum_{i=1}^{S}[x_{iP_2}-\beta_{2i}-min(x_{iP_1},x_{iP_2})]</code>
</p>

<p style="text-align: center;"><code class="reqn">D_{12}=\sum_{i=1}^{S}[min(x_{iP_1},x_{iP_2})-\alpha_i]</code>
</p>

<p>where <code class="reqn">x_{iP_1}</code> is the relative abundance of microbial taxon <code class="reqn">i</code> in a representative microbiome of the first parent species, <code class="reqn">x_{iP_2}</code> is the relative abundance of microbial taxon <code class="reqn">i</code> in a representative microbiome of the second parent species, <code class="reqn">x_{iH}</code> is the relative abundance of microbial taxon <code class="reqn">i</code> in a representative microbiome of the hybrid organism. Thus, <code class="reqn">A</code> is the relative abundance of microbial taxa shared by both progenitors and the hybrid, <code class="reqn">B</code> is the relative abundance of microbial taxa shared by one progenitor (but not both) and the hybrid, <code class="reqn">B_1</code> is the relative abundance of microbial taxa shared by the first progenitor and the hybrid, <code class="reqn">B_2</code> is the relative abundance of microbial taxa shared by the second progenitor and the hybrid, <code class="reqn">C</code> is the relative abundance of microbial taxa found only on the hybrid, <code class="reqn">D</code> is the relative abundance of microbial taxa found only on one or both progenitors, <code class="reqn">D_1</code> is the relative abundance of microbial taxa found only on the first progenitor, <code class="reqn">D_2</code> is the relative abundance of microbial taxa found only on the second progenitor, and <code class="reqn">D_12</code> is the relative abundance of microbial taxa found on both progenitors.
</p>
<p>When dist='Ruzicka', the 4H Index is calculated as:
</p>
<p style="text-align: center;"><code class="reqn">\mathcal{U}=\frac{B}{A+B+C+D}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{U}_1=\frac{B_1}{A+B+C+D}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{U}_2=\frac{B_2}{A+B+C+D}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{I}=\frac{A}{A+B+C+D}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{G}=\frac{C}{A+B+C+D}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{L}=\frac{D}{A+B+C+D}=1 - \mathcal{U} -\mathcal{I} -\mathcal{G}</code>
</p>

<p><code class="reqn">\mathcal{U}</code>, <code class="reqn">\mathcal{I}</code>, <code class="reqn">\mathcal{G}</code> and <code class="reqn">\mathcal{L}</code> represent the extent of the Union, Intersection, Gain and Loss Models respectively (see Camper et al., 2023) and <code class="reqn">\mathcal{U}_1</code> and <code class="reqn">\mathcal{U}_2</code> partition the Union model into components associated with the first and second progenitor.
</p>
<p>When dist = 'Bray-Curtis', the 4H Index is calculated as:
</p>
<p>This allows calculation of the four dimensions of the 4H Index:
</p>
<p style="text-align: center;"><code class="reqn">\mathcal{U}=\frac{2B}{3A+2B+C+D_1+D_2+2D_{12}}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{U}_1=\frac{2B_1}{3A+2B+C+D_1+D_2+2D_{12}}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{U}_2=\frac{2B_2}{3A+2B+C+D_1+D_2+2D_{12}}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{I}=\frac{3A}{3A+2B+C+D_1+D_2+2D_{12}}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{G}=\frac{C}{3A+2B+C+D_1+D_2+2D_{12}}</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{L}=\frac{D_1+D_2+2D_{12}}{3A+2B+C+D_1+D_2+2D_{12}}=1 - \mathcal{U} -\mathcal{I} -\mathcal{G}</code>
</p>

<p>In addition to the four dimensions of the 4H Index, <code>FourHbootstrapA</code> also calculates <code class="reqn">\sigma</code> for each bootstrap sample:
</p>
<p>Ruzicka:
</p>
<p style="text-align: center;"><code class="reqn">\sigma=\frac{A+D_{12}}{A+D_{12}+D_1+D_2}</code>
</p>

<p>Bray-Curtis:
</p>
<p style="text-align: center;"><code class="reqn">\sigma=\frac{2(A+D_{12})}{2(A+D_{12})+D_1+D_2}</code>
</p>

<p>where <code class="reqn">\sigma</code> is the fraction of parental microbial taxa that are found on both parent species. This will be used later for calculating the null plane of a given hybrid complex (see <code>FourHnullplane</code>).
</p>


<h3>Value</h3>

<p>This function returns a matrix, where rows are individual bootstrap samples and columns are the four dimensions of the 4H Index, as well as <code class="reqn">\sigma</code>, <code class="reqn">\mathcal{U}_1</code>, and <code class="reqn">\mathcal{U}_2</code>.
</p>


<h3>References</h3>

<p>Camper, B., Laughlin, Z. et al. &quot;A Conceptual Framework for Host-Associated Microbiomes of Hybrid Organisms&quot; &lt;doi:10.1101/2023.05.01.538925&gt;
</p>
<p>Henry, L., Wickham H., et al. &quot;rlang: Functions for Base Types and Core R and 'Tidyverse' Features&quot; https://CRAN.R-project.org/package=rlang
</p>
<p>McMurdie, Paul J., and Susan Holmes. &quot;phyloseq: an R package for reproducible interactive analysis and graphics of microbiome census data.&quot; PloS one 8.4 (2013): e61217.
</p>
<p>McMurdie, Paul J., and Susan Holmes. &quot;Phyloseq: a bioconductor package for handling and analysis of high-throughput phylogenetic sequence data.&quot; Biocomputing 2012. 2012. 235-246.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Test with enterotype dataset
library(phyloseq)
data(enterotype)
#Covert the OTU table to reads, rather than fractional abundances
otu_table(enterotype)&lt;-round(10000*otu_table(enterotype))

#Randomly assign host classes (these should be known in a real hybrid microbiome dataset)
#The two parent species are assigned '1' and '3' respectively, the hybrid is assigned '2'
hybrid_status&lt;-sample(1:3,280, replace=TRUE)

FourHbootstrapA(enterotype,hybrid_status,0.5,5,10, use_microViz='no')

</code></pre>

<hr>
<h2 id='FourHcentroid'>
Centroid of bootstrapped 4H-indices
</h2><span id='topic+FourHcentroid'></span>

<h3>Description</h3>

<p>Calculates the centroid of a set of 4H-indices from bootstrapped samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FourHcentroid(boots)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FourHcentroid_+3A_boots">boots</code></td>
<td>
<p>(Required) Output matrix from the <code>FourHbootstrap</code> function
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>FourHcentroid</code> averages over each column of the output matrix from <code>FourHbootstrap</code> and prints out the average value of each dimension (<code class="reqn">\mathcal{U}</code>, <code class="reqn">\mathcal{I}</code>, <code class="reqn">\mathcal{G}</code>, <code class="reqn">\mathcal{L}</code>) of the 4H-index. It also calculates and prints out the average value for <code class="reqn">\sigma.
</code>
</p>


<h3>Value</h3>

<p>This function outputs a dataframe reporting the average values of each dimension of the 4H-index, and the average value of <code class="reqn">\sigma</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Test with enterotype dataset
library(phyloseq)
data(enterotype)
#Covert the OTU table to reads, rather than fractional abundances
otu_table(enterotype)&lt;-round(10000*otu_table(enterotype))

#Randomly assign host classes (these should be known in a real hybrid microbiome dataset)
#The two parent species are assigned '1' and '3' respectively, the hybrid is assigned '2'
hybrid_status&lt;-sample(1:3,280, replace=TRUE)

#Bootstrap the dataset
boot_samples&lt;-FourHbootstrap(enterotype,hybrid_status,0.5,5,10)

#Find the centroid of the bootstrap samples
FourHcentroid(boot_samples)

</code></pre>

<hr>
<h2 id='FourHcompare'>
Tests whether multiple systems have significantly different 4H-indices
</h2><span id='topic+FourHcompare'></span>

<h3>Description</h3>

<p><code>FourHcompare</code> uses a PERMANOVA test on the log-ratio transformed bootstrap samples of the 4H-index from multiple different hybrid systems to determine whether the systems are significantly different in terms of the relative importance of the Union, Intersection, Gain and Loss Models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FourHcompare(boots_sets, boots_types, method='ilr', permutations = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FourHcompare_+3A_boots_sets">boots_sets</code></td>
<td>

<p>(Required) A matrix of bootstrapped 4H-indices from all of the systems that are to be compared. Rows are individual bootstraps, and columns are the four dimensions of the 4H-index and sigma from a particular bootstrap sample. The simplest way to generate this matrix is to use rbind() to merge the outputs of <code>FourHbootstrap</code> from multiple different systems.
</p>
</td></tr>
<tr><td><code id="FourHcompare_+3A_boots_types">boots_types</code></td>
<td>

<p>(Required) A vector specifying which hybrid system each bootstrap sample belongs to. The order of this vector must be the same as the order of the bootstrapped samples in the matrix of bootstrapped 4H indices.
</p>
</td></tr>
<tr><td><code id="FourHcompare_+3A_method">method</code></td>
<td>

<p>The method used to transform the compositional 4H-index in the 4-part Aitchison Simplex to a 3-dimensional euclidean vector. Statistical tests are performed on the 3-dimensional euclidean vectors. Currently, options for transformation are 'ilr' for an isometric log-ratio transformation, 'clr' for a centered log-ratio transformation, 'alr' for an additive log-ratio transformation and 'none' for non-transformed data. The default is to use an isometric log-ratio transformation.
</p>
</td></tr>
<tr><td><code id="FourHcompare_+3A_permutations">permutations</code></td>
<td>

<p>The desired number of permutations to use for the PERMANOVA. The default is 1000.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>FourHcompare</code> takes bootstrapped samples of the 4H-index from multiple different hybrid systems. First, each 4H-index from each bootstrapped sample is trasnformed to a 3-dimensional euclidean vector using a log-ratio transformation. This is done using the <code>ilr</code> or <code>clr</code> functions from the <code>compositions</code> R package. Next, the <code>PERMANOVA</code> function from the <code>PERMANOVA</code> R package is used to test whether there are significant differences between the 4H indices of the different hybrid systems.
</p>


<h3>Value</h3>

<p>This function creates a list with contents inherited from the <code>PERMANOVA</code> function.
</p>


<h3>References</h3>

<p>Van den Boogaart, K. Gerald, and Raimon Tolosana-Delgado. &quot;“Compositions”: a unified R package to analyze compositional data.&quot; Computers &amp; Geosciences 34.4 (2008): 320-338.
</p>
<p>van den Boogaart, K. Gerald, et al. &quot;Package ‘compositions’.&quot; Compositional data analysis Ver (2013): 1-40.
</p>
<p>van den Boogaart, K. Gerald, Tolosana-Delgado, R., Bren, M. &quot;compositions: Compositional Data Analysis&quot; https://CRAN.R-project.org/package=compositions
</p>
<p>Vicente-Gonzalez, L., J.L. Vicente-Villardon. &quot;PERMANOVA: Multivariate Analysis of Variance Based on Distances and Permutations&quot; https://CRAN.R-project.org/package=PERMANOVA
</p>
<p>Henry, L., Wickham H., et al. &quot;rlang: Functions for Base Types and Core R and 'Tidyverse' Features&quot; https://CRAN.R-project.org/package=rlang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Test with enterotype dataset
library(phyloseq)
data(enterotype)
#Covert the OTU table to reads, rather than fractional abundances
otu_table(enterotype)&lt;-round(10000*otu_table(enterotype))

#Use a subset of the enterotype dataset for one hybrid system...
#...and a subset for a second hybrid system
TS&lt;-prune_samples(grepl('TS',sample_names(enterotype)),enterotype)
MH&lt;-prune_samples(grepl('MH',sample_names(enterotype)),enterotype)

#Randomly assign host classes (these should be known in real hybrid microbiome datasets)
#The two parent species are assigned '1' and '3' respectively, the hybrid is assigned '2'
hybrid_statusTS&lt;-sample(1:3,154, replace=TRUE)
hybrid_statusMH&lt;-sample(1:3,85, replace=TRUE)

#Calculate bootstrapped samples of the 4H-index for each of the two systems
bootstrapTS&lt;-FourHbootstrap(TS,hybrid_statusTS,0.5,5,10)
bootstrapMH&lt;-FourHbootstrap(MH,hybrid_statusMH,0.5,5,10)

#Bind the two system 4H-index matrices together
boots_to_compare&lt;-rbind(bootstrapTS,bootstrapMH)
system_info&lt;-c(rep(1,5),rep(2,5))

#Perform the PERMANOVA
FourHcompare(boots_to_compare, system_info, method='ilr', permutations = 1000)

</code></pre>

<hr>
<h2 id='FourHnull'>
Bootstrapped null models of the 4H-index
</h2><span id='topic+FourHnull'></span>

<h3>Description</h3>

<p>Calculates 4H-indices for bootstrapped samples of hybrid null models drawn from an object of class phyloseq containing host-associated microbial community data from both hybrid hosts and hosts of each parent species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FourHnull(x, class_grouping, core_fraction, boot_no,
              sample_no, null_model=1,replace_hosts=FALSE, reads=NULL,
              rarefy_each_step=TRUE,seed=NULL, use_microViz='yes')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FourHnull_+3A_x">x</code></td>
<td>

<p>(Required) Host-associated microbial community data . This must be in the form of a phyloseq object and must contain an OTU abundance table with information on the host-associated microbial communities from both hybrid hosts and hosts of each progenitor taxon. The OTU abundance table should contain read counts, not fractional abundances.
</p>
</td></tr>
<tr><td><code id="FourHnull_+3A_class_grouping">class_grouping</code></td>
<td>

<p>(Required) A numerical vector identifying the host class of each host from the OTU table. This vector must be ordered according to the OTU abundance table, and must use the following notation: Progenitor One = 1, Hybrids = 2, Progenitor Two = 3.
</p>
</td></tr>
<tr><td><code id="FourHnull_+3A_core_fraction">core_fraction</code></td>
<td>

<p>(Required) The fraction of hosts that a microbial taxon must be found on to be considered part of a host class's 'core' microbiome
</p>
</td></tr>
<tr><td><code id="FourHnull_+3A_boot_no">boot_no</code></td>
<td>

<p>(Required) The desired number of bootstrap samples
</p>
</td></tr>
<tr><td><code id="FourHnull_+3A_sample_no">sample_no</code></td>
<td>

<p>(Required) The number of hosts of each host class that should be selected to generate each bootstrap sample. This number should be less than the minimum number of hosts in any given class.
</p>
</td></tr>
<tr><td><code id="FourHnull_+3A_null_model">null_model</code></td>
<td>

<p>The particular null model to use (see below for a description of each null model). The default is 1 (averaging microbial taxon abundances from a randomly drawn individual of each progenitor species)
</p>
</td></tr>
<tr><td><code id="FourHnull_+3A_replace_hosts">replace_hosts</code></td>
<td>

<p>Whether or not to replace hosts during bootstrapping. The default is FALSE (i.e., all hosts in a given bootstrap are unique).
</p>
</td></tr>
<tr><td><code id="FourHnull_+3A_reads">reads</code></td>
<td>

<p>The number of reads to rarefy to for each microbiome sample. The default is to use the number of reads in the sample with the lowest number of reads. Any user-specified value must be lower than the number of reads in the sample with the lowest number of reads.
</p>
</td></tr>
<tr><td><code id="FourHnull_+3A_rarefy_each_step">rarefy_each_step</code></td>
<td>

<p>Whether or not to rarefy each bootstrap sample separately. The default is TRUE (i.e., perform a new rarefaction on the OTU table for each bootstrap sample). However, for cases with large numbers of samples, large numbers of reads per sample or high microbial diversity, this can be slow; thus, there is the option to rarefy the microbiome samples a single time prior to bootstrapping.
</p>
</td></tr>
<tr><td><code id="FourHnull_+3A_seed">seed</code></td>
<td>

<p>Optional seed for random number generator. The default is NULL (i.e., seed picked at random at time of simulation).
</p>
</td></tr>
<tr><td><code id="FourHnull_+3A_use_microviz">use_microViz</code></td>
<td>

<p>Whether or not to use the microViz package to check your phyloseq object. If installation of the microViz package is problematic, this test can be skipped, but the user should verify themselves that a phyloseq object exists AND includes sample data (the sample data placeholder is necessary for the FourHnull code to work).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>FourHnull</code> generates bootstrapped samples of hybrid null models of the 4H index. As in <code>FourHbootstrap</code>, each iteration begins by rarefying the OTU table such that all samples have the same library size. Once the OTU table has been rarefied, a subset, <code>sample_no</code>, of hosts is selected from each progenitor class. Hosts can be sampled with or without replacement. The default is to sample hosts without replacement, such that all progenitors in any given bootstrap sample are unique. Next,<code>sample_no</code> hybrid individuals are generated according to the chosen null model. Null models are as follows:
</p>
<p><code>null_model= 1</code>: each hybrid individual is created by randomly selecting one individual from the progenitor 1 class and one individual from the progenitor 2 class and then averaging the microbial abundances from the two progenitor individuals.
</p>
<p><code>null_model= 2</code>: each hybrid individual is created by randomly selecting two individuals from the progenitor 1 class and one individual from the progenitor 2 class and then averaging the microbial abundances from the three progenitor individuals.
</p>
<p><code>null_model= 3</code>: each hybrid individual is created by randomly selecting one individual from the progenitor 1 class and two individuals from the progenitor 2 class and then averaging the microbial abundances from the three progenitor individuals.
</p>
<p><code>null_model= 4</code>: each hybrid individual is created by randomly selecting two individual from the progenitor 1 class and then averaging the microbial abundances from these two progenitor individuals.
</p>
<p><code>null_model= 5</code>: each hybrid individual is created by randomly selecting two individual from the progenitor 2 class and then averaging the microbial abundances from these two progenitor individuals.
</p>
<p><code>null_model= 6</code>: each hybrid individual is created by randomly selecting one individual from the progenitor 1 class and using its microbial abundances.
</p>
<p><code>null_model= 7</code>: each hybrid individual is created by randomly selecting one individual from the progenitor 2 class and using its microbial abundances.
</p>
<p><code>null_model= 10</code>: each hybrid individual is created by randomly selecting one individual from the progenitor 1 class and one individual from the progenitor 2 class, randomly setting half of the microbial taxon abundances only found on the progenitor 1 individual to zero, randomly setting half of the microbial taxon abundances only found on the progenitor 2 individual to zero, and then setting all other microbial taxon abundances to one.
</p>
<p>Once null model hybrid iterates have been generated, <code>FourHnull</code> proceeds as in <code>FourHbootstrap</code> and calculates the 4H-index and <code class="reqn">\sigma</code> for each bootstrap sample.
</p>


<h3>Value</h3>

<p>This function returns a matrix, where rows are individual bootstrap samples and columns are the four dimensions of the 4H-index, as well as <code class="reqn">\sigma</code>.
</p>


<h3>References</h3>

<p>Camper, B., Laughlin, Z. et al. &quot;A Conceptual Framework for Host-Associated Microbiomes of Hybrid Organisms&quot; &lt;doi:10.1101/2023.05.01.538925&gt;
</p>
<p>Henry, L., Wickham H., et al. &quot;rlang: Functions for Base Types and Core R and 'Tidyverse' Features&quot; https://CRAN.R-project.org/package=rlang
</p>
<p>McMurdie, Paul J., and Susan Holmes. &quot;phyloseq: an R package for reproducible interactive analysis and graphics of microbiome census data.&quot; PloS one 8.4 (2013): e61217.
</p>
<p>McMurdie, Paul J., and Susan Holmes. &quot;Phyloseq: a bioconductor package for handling and analysis of high-throughput phylogenetic sequence data.&quot; Biocomputing 2012. 2012. 235-246.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Test with enterotype dataset
library(phyloseq)
data(enterotype)
#Covert the OTU table to reads, rather than fractional abundances
otu_table(enterotype)&lt;-round(10000*otu_table(enterotype))

#Randomly assign host classes (these should be known in a real hybrid microbiome dataset)
#The two parent species are assigned '1' and '3' respectively, the hybrid is assigned '2'
hybrid_status&lt;-sample(1:3,280, replace=TRUE)

FourHnull(enterotype,hybrid_status,0.5,5,10, use_microViz='no')

</code></pre>

<hr>
<h2 id='FourHnullplane'>
Null Plane
</h2><span id='topic+FourHnullplane'></span>

<h3>Description</h3>

<p>Calculates and draws the null plane on a quaternary (four dimensional barycentric, Aitchison Simplex) plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FourHnullplane(boots,col='red')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FourHnullplane_+3A_boots">boots</code></td>
<td>
<p>(Required) Output matrix from the <code>FourHbootstrap</code> function
</p>
</td></tr>
<tr><td><code id="FourHnullplane_+3A_col">col</code></td>
<td>
<p>The color of the plane in the quaternary plot. The default color is red.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>FourHnullplane</code> uses the average value of <code class="reqn">\sigma= (P_1 \cap\ P_2)/(P_1 \cup\ P_2)</code> across all bootstrap samples to draw a null plane on an existing quaternary plot. For any given value of <code class="reqn">\theta = \mathcal{G}+\mathcal{L}</code>, the null plane is defined as follows:
</p>
<p style="text-align: center;"><code class="reqn">\mathcal{U}_{null} = (1-\sigma)(1-\theta)</code>
</p>

<p style="text-align: center;"><code class="reqn">\mathcal{I}_{null} = \sigma(1-\theta)</code>
</p>

<p>The primary purpose of the null plane is visualization. In particular, the null plane reflects the ratio of <code class="reqn">\mathcal(I)/(\mathcal{U}+\mathcal{I})</code> that would be expected if hybrids were equally likely to acquire any microbial taxa found on one or both parent species. 4H-indices that lie closer to the <code class="reqn">\mathcal{I}</code> vertex than the null plane indicate that hybrid organisms are more likely to retain microbial taxa shared by both parent species than microbial taxa found on only one parent species (and by corollary, hybrid organisms are more likely to lose microbial taxa found on only one parent species than they are to lose microbial taxa found on both parent species). 4H-indices that lie closer to the <code class="reqn">\mathcal{U}</code> vertex than the null plane indicate that hybrid organisms are more likely to retain microbial taxa found on only one parent species than microbial taxa found on both parent species (and by corollary, hybrid organisms are more likely to lose microbial taxa found on both parent species than they are to lose microbial taxa found on only one parent species).
</p>


<h3>Value</h3>

<p>This function outputs a plane on an existing quaternary plot.
</p>


<h3>References</h3>

<p>Henry, L., Wickham H., et al. &quot;rlang: Functions for Base Types and Core R and 'Tidyverse' Features&quot; https://CRAN.R-project.org/package=rlang
</p>
<p>Van den Boogaart, K. Gerald, and Raimon Tolosana-Delgado. &quot;“Compositions”: a unified R package to analyze compositional data.&quot; Computers &amp; Geosciences 34.4 (2008): 320-338.
</p>
<p>van den Boogaart, K. Gerald, et al. &quot;Package ‘compositions’.&quot; Compositional data analysis Ver (2013): 1-40.
</p>
<p>van den Boogaart, K. Gerald, Tolosana-Delgado, R., Bren, M. &quot;compositions: Compositional Data Analysis&quot; https://CRAN.R-project.org/package=compositions
</p>
<p>Adler, Daniel, Oleg Nenadic, and Walter Zucchini. &quot;Rgl: A r-library for 3d visualization with opengl.&quot; Proceedings of the 35th symposium of the interface: computing science and statistics, Salt Lake City. Vol. 35. 2003.
</p>
<p>Adler, D., Murdoch, M. D., Suggests, M. A. S. S., WebGL, P. L. Y., OBJ, S., &amp; OpenGL, S. (2019). Package ‘rgl’.
</p>
<p>Murdoch, Duncan, Daniel Adler, and Oleg Nenadic. &quot;Package ‘rgl’.&quot; R Package (2023).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Test with enterotype dataset
library(phyloseq)
data(enterotype)
#Covert the OTU table to reads, rather than fractional abundances
otu_table(enterotype)&lt;-round(10000*otu_table(enterotype))


#Randomly assign host classes (these should be known in a real hybrid microbiome dataset)
#The two parent species are assigned '1' and '3' respectively, the hybrid is assigned '2'
hybrid_status&lt;-sample(1:3,280, replace=TRUE)

#Bootstrap the dataset
boot_samples&lt;-FourHbootstrap(enterotype,hybrid_status, 0.5,5,10)

#Plot the null plane on an existing quaternary plot
#An existing quaternary plot should already be open
FourHnullplane(boot_samples,col='red')

</code></pre>

<hr>
<h2 id='FourHnullplaneD'>
Bootstrap Distribution Around the Null Plane
</h2><span id='topic+FourHnullplaneD'></span>

<h3>Description</h3>

<p>Calculates the distribution of bootstrap samples around the null expectation for Union vs. Intersection models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FourHnullplaneD(boots)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FourHnullplaneD_+3A_boots">boots</code></td>
<td>
<p>(Required) Output matrix from the <code>FourHbootstrap</code> function
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each bootstrap sample, <code>FourHnullplaneD</code> uses the fraction of microbes shared by both progenitors to calculate null model expectations for the value of <code class="reqn">\mathcal{U}</code> and <code class="reqn">\mathcal{I}</code>. It then calculates the difference between the observed and expected values along the intersection and union dimensions <code class="reqn">\mathcal{I}-\mathcal{I}_{null}</code> and <code class="reqn">\mathcal{U}-\mathcal{U}_{null}</code>. Finally, <code>FourHnullplaneD</code> plots a histogram of the difference between observed and expected intersection dimensions, <code class="reqn">\mathcal{I}-\mathcal{I}_{null}</code> and calculates the average distance between the observed and expected intersection dimensions, as well as the standard deviation of <code class="reqn">\mathcal{I}-\mathcal{I}_{null}</code> and the fraction of bootstrapped samples where <code class="reqn">\mathcal{I}-\mathcal{I}_{null}&lt;0</code> (i.e., hybrids preferentially retain microbial taxa only found on one progenitor). Preferential retention of microbial taxa found on both progenitors can then be assessed as &gt;95% of bootstrapped samples having <code class="reqn">\mathcal{I}-\mathcal{I}_{null}&gt;0</code>.
</p>


<h3>Value</h3>

<p>This function outputs a list including a dataframe with the following lists:
</p>
<table>
<tr><td><code>sigma</code></td>
<td>
<p>The inputted values of <code class="reqn">\sigma</code> for each bootstrapped sample.</p>
</td></tr>
<tr><td><code>theta</code></td>
<td>
<p>The calculated values of <code class="reqn">\theta = \mathcal{G}+\mathcal{L}</code> for each bootstrapped sample.</p>
</td></tr>
<tr><td><code>union</code></td>
<td>
<p>The observed values of the union dimension, <code class="reqn">\mathcal{U}</code>, for each bootstrapped sample.</p>
</td></tr>
<tr><td><code>union_expectation</code></td>
<td>
<p>The expected values of the union dimension, <code class="reqn">\mathcal{U}_{null}</code>, for each bootstrapped sample.</p>
</td></tr>
<tr><td><code>intersection</code></td>
<td>
<p>The observed values of the intersection dimension, <code class="reqn">\mathcal{I}</code>, for each bootstrapped sample.</p>
</td></tr>
<tr><td><code>intersection_expectation</code></td>
<td>
<p>The expected values of the intersection dimension, <code class="reqn">\mathcal{I}_{null}</code>, for each bootstrapped sample.</p>
</td></tr>
<tr><td><code>diffI</code></td>
<td>
<p>The difference between the observed and expected values of the intersection dimension for each bootstrapped sample.</p>
</td></tr>
<tr><td><code>diffU</code></td>
<td>
<p>The difference between the observed and expected values of the union dimension for each bootstrapped sample.</p>
</td></tr>
</table>
<p>as well as the average distance and standard deviation between the observed and expected intersection dimensions, and the fraction of bootstrapped samples where hybrids preferentially retain microbial taxa only found on one progenitor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Test with enterotype dataset
library(phyloseq)
data(enterotype)
#Covert the OTU table to reads, rather than fractional abundances
otu_table(enterotype)&lt;-round(10000*otu_table(enterotype))


#Randomly assign host classes (these should be known in a real hybrid microbiome dataset)
#The two parent species are assigned '1' and '3' respectively, the hybrid is assigned '2'
hybrid_status&lt;-sample(1:3,280, replace=TRUE)

#Bootstrap the dataset
boot_samples&lt;-FourHbootstrap(enterotype,hybrid_status,0.5,5,10)

#Calculate the expected and observed distributions of bootstrapped samples around the null plane
FourHnullplaneD(boot_samples)

</code></pre>

<hr>
<h2 id='FourHpreanalysis'>
Bootstraps of the percentage of gamma diversity that is core
</h2><span id='topic+FourHpreanalysis'></span>

<h3>Description</h3>

<p>Calculates the percentage of the total gamma diversity that is part of the core microbiota for bootstrapped samples drawn from an object of class phyloseq containing host-associated microbial community data from both hybrid hosts and hosts of each parent species. Prints out a warning when the percentage of the total gamma diversity that is core for hybrid hosts is less than half of the average percentage that is core for the two parent populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FourHpreanalysis(x, class_grouping, core_fraction, boot_no, sample_no,
core_average_abundance = 0, core_max_abundance = 0,
core_all_average_abundance = 0, core_all_max_abundance = 0,
replace_hosts=FALSE, reads=NULL, rarefy_each_step=TRUE,seed=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FourHpreanalysis_+3A_x">x</code></td>
<td>

<p>(Required) Host-associated microbial community data . This must be in the form of a phyloseq object and must contain an OTU abundance table with information on the host-associated microbial communities from both hybrid hosts and hosts of each progenitor taxon. The OTU abundance table should contain read counts, not fractional abundances.
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_class_grouping">class_grouping</code></td>
<td>

<p>(Required) A numerical vector identifying the host class of each host from the OTU table. This vector must be ordered according to the OTU abundance table, and must use the following notation: Progenitor One = 1, Hybrids = 2, Progenitor Two = 3.
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_core_fraction">core_fraction</code></td>
<td>

<p>(Required) The fraction of hosts that a microbial taxon must be found on to be considered part of a host class's 'core' microbiome
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_boot_no">boot_no</code></td>
<td>

<p>(Required) The desired number of bootstrap samples
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_sample_no">sample_no</code></td>
<td>

<p>(Required) The number of hosts of each host class that should be selected to generate each bootstrap sample. This number should be less than the minimum number of hosts in any given class.
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_core_average_abundance">core_average_abundance</code></td>
<td>

<p>The minimum average abundance at which a microbial taxon must be found across all hosts within a class to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum average abundance)
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_core_max_abundance">core_max_abundance</code></td>
<td>

<p>The minimum abundance at which a microbial taxon must be found on at least one host within a class to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum abundance on any single host)
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_core_all_average_abundance">core_all_average_abundance</code></td>
<td>

<p>The minimum average abundance at which a microbial taxon must be found across all hosts (regardless of class) to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum average abundance)
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_core_all_max_abundance">core_all_max_abundance</code></td>
<td>

<p>The minimum abundance at which a microbial taxon must be found on at least one host (regardless of class) to be considered part of a host class's 'core' microbiome. The default is zero (i.e., it does not need to occur at any minimum abundance on any single host)
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_replace_hosts">replace_hosts</code></td>
<td>

<p>Whether or not to replace hosts during bootstrapping. The default is FALSE (i.e., all hosts in a given bootstrap are unique).
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_reads">reads</code></td>
<td>

<p>The number of reads to rarefy to for each microbiome sample. The default is to use the number of reads in the sample with the lowest number of reads. Any user-specified value must be lower than the number of reads in the sample with the lowest number of reads.
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_rarefy_each_step">rarefy_each_step</code></td>
<td>

<p>Whether or not to rarefy each bootstrap sample separately. The default is TRUE (i.e., perform a new rarefaction on the OTU table for each bootstrap sample). However, for cases with large numbers of samples, large numbers of reads per sample or high microbial diversity, this can be slow; thus, there is the option to rarefy the microbiota samples a single time prior to bootstrapping.
</p>
</td></tr>
<tr><td><code id="FourHpreanalysis_+3A_seed">seed</code></td>
<td>

<p>Optional seed for random number generator. The default is NULL (i.e., seed picked at random at time of simulation).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>FourHpreanalysis</code> generates bootstrapped samples of the percentage of gamma diversity for hybrids and each parent species that is part of the core. Each bootstrap begins by rarefying the OTU table such that all samples have the same library size. Once the OTU table has been rarefied, a subset, <code>sample_no</code>, of hosts is selected from each host class. Hosts can be sampled with or without replacement. The default is to sample hosts without replacement, such that all hosts in any given bootstrap sample are unique. Subsampled individuals from each host class are then used to determine overall gamma diversity and diversity of core microbiotas for hybrid hosts and hosts of each parent species. This allows calculation of the percentage of gamma diversity that is part of the core. When this percentage among hybrids is less than half of what it is, on average, among parent species, a warning is printed.
</p>


<h3>References</h3>

<p>Camper, B., Laughlin, Z. et al. &quot;A Conceptual Framework for Host-Associated Microbiomes of Hybrid Organisms&quot; &lt;doi:10.1101/2023.05.01.538925&gt;
</p>
<p>Henry, L., Wickham H., et al. &quot;rlang: Functions for Base Types and Core R and 'Tidyverse' Features&quot; https://CRAN.R-project.org/package=rlang
</p>
<p>McMurdie, Paul J., and Susan Holmes. &quot;phyloseq: an R package for reproducible interactive analysis and graphics of microbiome census data.&quot; PloS one 8.4 (2013): e61217.
</p>
<p>McMurdie, Paul J., and Susan Holmes. &quot;Phyloseq: a bioconductor package for handling and analysis of high-throughput phylogenetic sequence data.&quot; Biocomputing 2012. 2012. 235-246.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Test with enterotype dataset
library(phyloseq)
data(enterotype)
#Covert the OTU table to reads, rather than fractional abundances
otu_table(enterotype)&lt;-round(10000*otu_table(enterotype))

#Randomly assign host classes (these should be known in a real hybrid microbiome dataset)
#The two parent species are assigned '1' and '3' respectively, the hybrid is assigned '2'
hybrid_status&lt;-sample(1:3,280, replace=TRUE)

FourHpreanalysis(enterotype,hybrid_status,0.5,5,10)

</code></pre>

<hr>
<h2 id='FourHquaternary'>
Plot bootstrapped 4H-indices
</h2><span id='topic+FourHquaternary'></span>

<h3>Description</h3>

<p>Draws bootstrapped values of the 4H-index on a quaternary (four dimensional barycentric, Aitchison Simplex) plot with an additional option to draw the centroid of the bootstrapped samples as well.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FourHquaternary(boots,col='red',addplot=FALSE,plotcentroid=TRUE,
               plotgrid=TRUE, size_bootstrap=1,size_centroid=15,size_font=2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FourHquaternary_+3A_boots">boots</code></td>
<td>
<p>(Required) Output matrix from the <code>FourHbootstrap</code> function
</p>
</td></tr>
<tr><td><code id="FourHquaternary_+3A_col">col</code></td>
<td>
<p>The color of the points in the quaternary plot. The default is RED</p>
</td></tr>
<tr><td><code id="FourHquaternary_+3A_addplot">addplot</code></td>
<td>
<p>Whether or not to add the data to an existing plot. The default is FALSE</p>
</td></tr>
<tr><td><code id="FourHquaternary_+3A_plotcentroid">plotcentroid</code></td>
<td>
<p>Whether or not to plot the centroid of the bootstrap samples. The default is TRUE.</p>
</td></tr>
<tr><td><code id="FourHquaternary_+3A_plotgrid">plotgrid</code></td>
<td>
<p>Whether or not to plot gridlines on the quaternary plot. The default is TRUE</p>
</td></tr>
<tr><td><code id="FourHquaternary_+3A_size_bootstrap">size_bootstrap</code></td>
<td>
<p>The size of each point representing a single bootstrap sample. The default is 1.</p>
</td></tr>
<tr><td><code id="FourHquaternary_+3A_size_centroid">size_centroid</code></td>
<td>
<p>The size of the point representing the centroid of the bootstrap samples. The default is 15.</p>
</td></tr>
<tr><td><code id="FourHquaternary_+3A_size_font">size_font</code></td>
<td>
<p>The size of the font used to label the four vertices. The default is 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>FourHquaternary</code> allows for visualization of bootstrapped 4H-indices within the framework of a 4-dimensional Aitchison Simplex, also known as a 4-dimensional barycentric plot or quaternary plot. Commonly used for compositional data, the Aitchison Simplex is convenient for the 4H-index, because <code class="reqn">\mathcal{U} + \mathcal{I} + \mathcal{G} + \mathcal{L} = 1</code>, reflecting the fact that each of the four different conceptual models - Union, Intersection, Gain and Loss - comprise one part of the whole, where the whole is the total microbial diversity across the entire hybrid host complex. <code>FourHquaternary</code> uses functions from the <code>compositions</code> R package to transform the four coordinates of the 4H-index, such that they can be drawn on a quaternary plot.
</p>


<h3>Value</h3>

<p>This function plots bootstrapped 4H-indices on quaternary plot (Aitchison Simplex, 4-dimensional barycentric plot).
</p>


<h3>References</h3>

<p>Henry, L., Wickham H., et al. &quot;rlang: Functions for Base Types and Core R and 'Tidyverse' Features&quot; https://CRAN.R-project.org/package=rlang
</p>
<p>Van den Boogaart, K. Gerald, and Raimon Tolosana-Delgado. &quot;“Compositions”: a unified R package to analyze compositional data.&quot; Computers &amp; Geosciences 34.4 (2008): 320-338.
</p>
<p>van den Boogaart, K. Gerald, et al. &quot;Package ‘compositions’.&quot; Compositional data analysis Ver (2013): 1-40.
</p>
<p>van den Boogaart, K. Gerald, Tolosana-Delgado, R., Bren, M. &quot;compositions: Compositional Data Analysis&quot; https://CRAN.R-project.org/package=compositions
</p>
<p>Adler, Daniel, Oleg Nenadic, and Walter Zucchini. &quot;Rgl: A r-library for 3d visualization with opengl.&quot; Proceedings of the 35th symposium of the interface: computing science and statistics, Salt Lake City. Vol. 35. 2003.
</p>
<p>Adler, D., Murdoch, M. D., Suggests, M. A. S. S., WebGL, P. L. Y., OBJ, S., &amp; OpenGL, S. (2019). Package ‘rgl’.
</p>
<p>Murdoch, Duncan, Daniel Adler, and Oleg Nenadic. &quot;Package ‘rgl’.&quot; R Package (2023).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Test with enterotype dataset
library(phyloseq)
data(enterotype)
#Covert the OTU table to reads, rather than fractional abundances
otu_table(enterotype)&lt;-round(10000*otu_table(enterotype))

#Randomly assign host classes (these should be known in a real hybrid microbiome dataset)
#The two parent species are assigned '1' and '3' respectively, the hybrid is assigned '2'
hybrid_status&lt;-sample(1:3,280, replace=TRUE)

#Bootstrap the dataset
boot_samples&lt;-FourHbootstrap(enterotype,hybrid_status,0.5,5,10)

#Plot the bootstrapped samples as a new quaternary plot including the centroid
FourHquaternary(boot_samples,col='red')

#Plot the bootstrapped samples on an existing quaternary plot without including the centroid
#An existing quaternary plot should already be open
FourHquaternary(boot_samples,col='red',addplot=TRUE, plotcentroid=FALSE)

</code></pre>

<hr>
<h2 id='FourHquaternaryC'>
Plot centroid of bootstrapped 4H-indices
</h2><span id='topic+FourHquaternaryC'></span>

<h3>Description</h3>

<p>Draws the centroid of bootstrapped values of the 4H-index on a quaternary (four dimensional barycentric, Aitchison Simplex) plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FourHquaternaryC(boots,col='red',addplot=FALSE,plotgrid=TRUE,
                                            size_centroid=15,size_font=2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FourHquaternaryC_+3A_boots">boots</code></td>
<td>
<p>Output from the <code>FourHbootstrap</code> function
</p>
</td></tr>
<tr><td><code id="FourHquaternaryC_+3A_col">col</code></td>
<td>
<p>The color of the points in the quaternary plot. The default is 'red'</p>
</td></tr>
<tr><td><code id="FourHquaternaryC_+3A_addplot">addplot</code></td>
<td>
<p>Whether or not to add the data to an existing plot. The default is FALSE</p>
</td></tr>
<tr><td><code id="FourHquaternaryC_+3A_plotgrid">plotgrid</code></td>
<td>
<p>Whether or not to plot gridlines on the quaternary plot. The default is TRUE</p>
</td></tr>
<tr><td><code id="FourHquaternaryC_+3A_size_centroid">size_centroid</code></td>
<td>
<p>The size of the point representing the centroid of the bootstrap samples. The default is 15.</p>
</td></tr>
<tr><td><code id="FourHquaternaryC_+3A_size_font">size_font</code></td>
<td>
<p>The size of the font used to label the four verticies. The default is 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>FourHquaternaryC</code> allows for visualization of the centroid of bootstrapped 4H-indices within the framework of a 4-dimensional Aitchison Simplex, also known as a 4-dimensional barycentric plot or quaternary plot. Commonly used for compositional data, the Aitchison Simplex is convenient for the 4H-index, because <code class="reqn">\mathcal{U} + \mathcal{I} + \mathcal{G} + \mathcal{L} = 1</code>, reflecting the fact that each of the four different conceptual models - Union, Intersection, Gain and Loss - comprise one part of the whole, where the whole is the total microbial diversity across the entire hybrid host complex. <code>FourHquaternaryC</code> uses functions from the <code>compositions</code> R package to transform the four coordinates of the 4H-index, such that they can be drawn on a quaternary plot.
</p>


<h3>Value</h3>

<p>This function plots the centroid of bootstrapped 4H-indices on quaternary plot (Aitchison Simplex, 4-dimensional barycentric plot).
</p>


<h3>References</h3>

<p>Henry, L., Wickham H., et al. &quot;rlang: Functions for Base Types and Core R and 'Tidyverse' Features&quot; https://CRAN.R-project.org/package=rlang
</p>
<p>Van den Boogaart, K. Gerald, and Raimon Tolosana-Delgado. &quot;“Compositions”: a unified R package to analyze compositional data.&quot; Computers &amp; Geosciences 34.4 (2008): 320-338.
</p>
<p>van den Boogaart, K. Gerald, et al. &quot;Package ‘compositions’.&quot; Compositional data analysis Ver (2013): 1-40.
</p>
<p>van den Boogaart, K. Gerald, Tolosana-Delgado, R., Bren, M. &quot;compositions: Compositional Data Analysis&quot; https://CRAN.R-project.org/package=compositions
</p>
<p>Adler, Daniel, Oleg Nenadic, and Walter Zucchini. &quot;Rgl: A r-library for 3d visualization with opengl.&quot; Proceedings of the 35th symposium of the interface: computing science and statistics, Salt Lake City. Vol. 35. 2003.
</p>
<p>Adler, D., Murdoch, M. D., Suggests, M. A. S. S., WebGL, P. L. Y., OBJ, S., &amp; OpenGL, S. (2019). Package ‘rgl’.
</p>
<p>Murdoch, Duncan, Daniel Adler, and Oleg Nenadic. &quot;Package ‘rgl’.&quot; R Package (2023).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Test with enterotype dataset
library(phyloseq)
data(enterotype)
#Covert the OTU table to reads, rather than fractional abundances
otu_table(enterotype)&lt;-round(10000*otu_table(enterotype))

#Randomly assign host classes (these should be known in a real hybrid microbiome dataset)
#The two parent species are assigned '1' and '3' respectively, the hybrid is assigned '2'
hybrid_status&lt;-sample(1:3,280, replace=TRUE)

#Bootstrap the dataset
boot_samples&lt;-FourHbootstrap(enterotype,hybrid_status,0.5,5,10)

#Plot the centroid of the bootstrapped samples as a new quaternary plot
FourHquaternaryC(boot_samples,col='red')

#Plot the centroid of the bootstrapped samples on an existing quaternary plot
#An existing quaternary plot should already be open
FourHquaternaryC(boot_samples,col='red',addplot=TRUE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
