<!DOCTYPE html><html lang="en-US"><head><title>Help for package IBCF.MTME</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {IBCF.MTME}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#IBCF.MTME'><p>IBCF.MTME: Item Based Collaborative Filtering for Multi-Trait and Multi-Environment Data.</p></a></li>
<li><a href='#barplot.IBCFY'><p>barplot.IBCFY</p></a></li>
<li><a href='#CV.RandomPart'><p>Cross-Validation with Random Partitions</p></a></li>
<li><a href='#getMatrixForm'><p>Tidy data format to Matrix format</p></a></li>
<li><a href='#getTidyForm'><p>Matrix format to Tidy data format</p></a></li>
<li><a href='#IBCF'><p>IBCF</p></a></li>
<li><a href='#IBCF.Years'><p>IBCF.Years</p></a></li>
<li><a href='#plot.IBCF'><p>Plot IBCF graph</p></a></li>
<li><a href='#print.IBCF'><p>Print IBCF information object</p></a></li>
<li><a href='#print.IBCFY'><p>Print IBCFY information object</p></a></li>
<li><a href='#summary.IBCF'><p>Summary</p></a></li>
<li><a href='#summary.IBCFY'><p>Summary</p></a></li>
<li><a href='#Wheat_IBCF'><p>Wheat Data</p></a></li>
<li><a href='#Year_IBCF'><p>Year_IBCF Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Item Based Collaborative Filtering for Multi-Trait and
Multi-Environment Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6-0</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-03-17</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Francisco Javier Luna-Vazquez &lt;frahik@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the item based collaborative filtering (IBCF) method for continues phenotypes in the context of plant breeding where data are collected for various traits that were studied in various environments proposed by Montesinos-LÃ³pez et al. (2017) &lt;<a href="https://doi.org/10.1534%2Fg3.117.300309">doi:10.1534/g3.117.300309</a>&gt;. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/frahik/IBCF.MTME">https://github.com/frahik/IBCF.MTME</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/frahik/IBCF.MTME/issues/new">https://github.com/frahik/IBCF.MTME/issues/new</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Collate:</td>
<td>CrossValidation.R IBCF.R IBCF.MTME.R IBCFYear.R methods.R
Td2M.R Title.R Wheat_IBCF.R Year_IBCF.R</td>
</tr>
<tr>
<td>Imports:</td>
<td>lsa, tidyr, dplyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, covr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-03-23 04:37:48 UTC; frahik</td>
</tr>
<tr>
<td>Author:</td>
<td>Francisco Javier Luna-Vazquez
    <a href="https://orcid.org/0000-0002-5370-7152"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Osval Antonio Montesinos-Lopez
    <a href="https://orcid.org/0000-0002-3973-6547"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Abelardo Montesinos-Lopez [aut],
  Jose Crossa <a href="https://orcid.org/0000-0001-9429-5855"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-03-23 05:18:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='IBCF.MTME'>IBCF.MTME: Item Based Collaborative Filtering for Multi-Trait and Multi-Environment Data.</h2><span id='topic+IBCF.MTME'></span><span id='topic+IBCF.MTME-package'></span>

<h3>Description</h3>

<p>The Item Based Collaborative Filtering for Multi-Trait and Multi-Environment Data (IBCF.MTME) package was developed to implement the item based collaborative filtering (IBCF) method for continues phenotypes in the context of plant breeding where data are collected for various traits that were studied in various environments. It is important to point out that the main difference of this package with the available packages that can implement IBCF is that this package was developed for continuous phenotypes which cannot be implemented in the current packages that can implement IBCF that only work for binary and ordinary phenotypes.
</p>

<hr>
<h2 id='barplot.IBCFY'>barplot.IBCFY</h2><span id='topic+barplot.IBCFY'></span>

<h3>Description</h3>

<p>Barplot of the results from IBCFY object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'IBCFY'
barplot(height, select = "Pearson", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="barplot.IBCFY_+3A_height">height</code></td>
<td>
<p><code>IBCFY object</code> IBCFY object, result of use the IBCF.Years() function</p>
</td></tr>
<tr><td><code id="barplot.IBCFY_+3A_select">select</code></td>
<td>
<p><code>character</code> By default ('Pearson'), plot the Pearson Correlations of the IBCF Object, else ('MAAPE'), plot the MAAPE of the IBCF Object.</p>
</td></tr>
<tr><td><code id="barplot.IBCFY_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>

<hr>
<h2 id='CV.RandomPart'>Cross-Validation with Random Partitions</h2><span id='topic+CV.RandomPart'></span>

<h3>Description</h3>

<p>This method consists of randomly dividing the training data set and the test data set.
For each division, the approximation function is adjusted from the training data and calculates the output values for the test data set.
The result corresponds to the arithmetic mean of the values obtained for the different divisions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CV.RandomPart(DataSet, NPartitions = 10, PTesting = 0.35,
  Traits.testing = NULL, Set_seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CV.RandomPart_+3A_dataset">DataSet</code></td>
<td>
<p><code>data.frame</code> The data set object is a data.frame object that contains 4 columns in the Tidy data format:
<code>$Line</code> is the Line or genotype identifier, and the name of this column could change.
<code>$Env</code> is the name of the evaluated environment (s).
<code>$Trait</code> is the name of the evaluated trait (s).
<code>$Response</code> Variable response obtained for the row corresponding to line and environment.</p>
</td></tr>
<tr><td><code id="CV.RandomPart_+3A_npartitions">NPartitions</code></td>
<td>
<p><code>integer</code> Number of Partitions for the Cross-Validation. Is 10 by default.</p>
</td></tr>
<tr><td><code id="CV.RandomPart_+3A_ptesting">PTesting</code></td>
<td>
<p><code>Double</code> Percentage of Testing for the Cross-Validation. Is 0.35 by default.</p>
</td></tr>
<tr><td><code id="CV.RandomPart_+3A_traits.testing">Traits.testing</code></td>
<td>
<p><code>character</code> By default is null and use all the traits to fit the model, else only part of the traits specified be used to fit the model.</p>
</td></tr>
<tr><td><code id="CV.RandomPart_+3A_set_seed">Set_seed</code></td>
<td>
<p><code>integer</code> Number of seed for reproducible research. Is NULL by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>List</code> A list object with length of <code>NPartitions</code>, every index has a <code>matrix</code> <code class="reqn">n \times x</code>, where <code class="reqn">n</code> is the number of <code>NLines</code> and <code class="reqn">x</code> is the number of  <code>NEnv</code> <code class="reqn">\times</code> <code>NTraits</code>. The values inside is 1 for training and 2 for testing.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  library(IBCF.MTME)
  data('Wheat_IBCF')

  CV.RandomPart(Wheat_IBCF)
  CV.RandomPart(Wheat_IBCF, NPartitions = 10)
  CV.RandomPart(Wheat_IBCF, Traits.testing = 'DH')
  CV.RandomPart(Wheat_IBCF, NPartitions = 10, PTesting = .35)
  CV.RandomPart(Wheat_IBCF, NPartitions = 10, Traits.testing = 'DH')
  CV.RandomPart(Wheat_IBCF, NPartitions = 10, PTesting = .35, Set_seed = 5)
  CV.RandomPart(Wheat_IBCF, NPartitions = 10, PTesting = .35, Traits.testing = 'DH')
  CV.RandomPart(Wheat_IBCF, NPartitions = 10, PTesting = .35, Traits.testing = 'DH', Set_seed = 5 )

## End(Not run)
</code></pre>

<hr>
<h2 id='getMatrixForm'>Tidy data format to Matrix format</h2><span id='topic+getMatrixForm'></span>

<h3>Description</h3>

<p>Tidy data format to Matrix format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMatrixForm(Tidy_DataSet, onlyTrait = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getMatrixForm_+3A_tidy_dataset">Tidy_DataSet</code></td>
<td>
<p><code>data.frame</code> object that contains 4 columns:
<code>$Line</code>: Line or genotype identifier, and the name of this column could change.
<code>$Env</code>: Name of the evaluated environment (s).
<code>$Trait</code>: Name of the evaluated trait (s).
<code>$Response</code>: Variable response obtained for the row corresponding to line and environment.</p>
</td></tr>
<tr><td><code id="getMatrixForm_+3A_onlytrait">onlyTrait</code></td>
<td>
<p><code>logical</code> by default is <code>FALSE</code>, if is <code>TRUE</code> only the column <code>$Trait</code> is transformed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> object with the <code>$Response</code> divided by <code>$Traits</code> columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  data('Wheat_IBCF')
  M &lt;- getMatrixForm(Wheat_IBCF)

## End(Not run)

## Not run: 
  data('Year_IBCF')
  M.Y &lt;- getMatrixForm(Year_IBCF, onlyTrait = T)

## End(Not run)
</code></pre>

<hr>
<h2 id='getTidyForm'>Matrix format to Tidy data format</h2><span id='topic+getTidyForm'></span>

<h3>Description</h3>

<p>Matrix format to Tidy data format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTidyForm(Matrix_DataSet, onlyTrait = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getTidyForm_+3A_matrix_dataset">Matrix_DataSet</code></td>
<td>
<p>A data.frame object with the response values divided in <code class="reqn">n</code> environments or traits columns</p>
</td></tr>
<tr><td><code id="getTidyForm_+3A_onlytrait">onlyTrait</code></td>
<td>
<p><code>logical</code> by default is <code>FALSE</code>, if is <code>TRUE</code> only is considered the <code>$Trait</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> object with the <code>$Response</code> divided by <code>$Traits</code> columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  data('Wheat_IBCF')
  M &lt;- getMatrixForm(Wheat_IBCF)
  Tidy &lt;- getTidyForm(M)

## End(Not run)

## Not run: 
  data('Year_IBCF')
  M.Y &lt;- getMatrixForm(Year_IBCF, onlyTrait = T)
  Tidy.Y &lt;- getTidyForm(M.Y, onlyTrait = T)

## End(Not run)

</code></pre>

<hr>
<h2 id='IBCF'>IBCF</h2><span id='topic+IBCF'></span>

<h3>Description</h3>

<p>Item Based Collaborative Filtering for multi-trait and multi-environment data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IBCF(object, dec = 4)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="IBCF_+3A_object">object</code></td>
<td>
<p><code>list</code> CrossValidation object, is obtained from CV.RandomPartition function.</p>
</td></tr>
<tr><td><code id="IBCF_+3A_dec">dec</code></td>
<td>
<p><code>integer</code> Number of decimals to print in the results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the next components
</p>
<table role = "presentation">
<tr><td><code>NPartitions</code></td>
<td>
<p><code>integer</code> Number of partitions used for testing data</p>
</td></tr>
<tr><td><code>predictions_Summary</code></td>
<td>
<p><code>data.frame</code> A data.frame with the results of the test</p>
</td></tr>
<tr><td><code>Predictions</code></td>
<td>
<p><code>list</code> A list with the predicted results for each partition</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'> ## Not run: 
  library(IBCF.MTME)
  data('Wheat_IBCF')

  CV &lt;- CV.RandomPart(Wheat_IBCF)
  IBCF(CV)

## End(Not run)

</code></pre>

<hr>
<h2 id='IBCF.Years'>IBCF.Years</h2><span id='topic+IBCF.Years'></span>

<h3>Description</h3>

<p>Item Based Collaborative Filtering for Years data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IBCF.Years(DataSet, colYears = 1, colID = 2, Years.testing = "",
  Traits.testing = "", dec = 4)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="IBCF.Years_+3A_dataset">DataSet</code></td>
<td>
<p><code>data.frame</code> A data set in Matrix Form.</p>
</td></tr>
<tr><td><code id="IBCF.Years_+3A_colyears">colYears</code></td>
<td>
<p><code>string or integer</code> A name or the position of the 'Years' column just in case that is not the first column.</p>
</td></tr>
<tr><td><code id="IBCF.Years_+3A_colid">colID</code></td>
<td>
<p><code>string or integer</code> A name or the position of the 'ID' column, just in case that is not the second column.</p>
</td></tr>
<tr><td><code id="IBCF.Years_+3A_years.testing">Years.testing</code></td>
<td>
<p><code>vector</code> A vector with the names of the years to use in test.</p>
</td></tr>
<tr><td><code id="IBCF.Years_+3A_traits.testing">Traits.testing</code></td>
<td>
<p><code>vector</code> A vector with the names of the traits to use in test.</p>
</td></tr>
<tr><td><code id="IBCF.Years_+3A_dec">dec</code></td>
<td>
<p><code>integer</code> Number of decimals to print in the results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the next components
</p>
<table role = "presentation">
<tr><td><code>Year.testing</code></td>
<td>
<p><code>vector</code> a vector with the Years used for the testing data</p>
</td></tr>
<tr><td><code>Traits.testing</code></td>
<td>
<p><code>vector</code> a vector with the Traits used for the testing data</p>
</td></tr>
<tr><td><code>Data_Obs_Pred</code></td>
<td>
<p><code>data.frame</code> Contains the values observed and predicted (the predicted values has '.1' after the name)</p>
</td></tr>
<tr><td><code>predictions_Summary</code></td>
<td>
<p><code>data.frame</code> Contains the summary of the correlation of the predictions and the MAAPE</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  library(IBCF.MTME)
  data('Year_IBCF')
  DataSet &lt;- getMatrixForm(Year_IBCF, onlyTrait = TRUE)
  IBCF.Years(DataSet , Years.testing = c('2015', '2016'), Traits.testing = c('T5', 'T6'))


## End(Not run)

</code></pre>

<hr>
<h2 id='plot.IBCF'>Plot IBCF graph</h2><span id='topic+plot.IBCF'></span>

<h3>Description</h3>

<p>Plot from IBCF object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'IBCF'
plot(x, select = "Pearson", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.IBCF_+3A_x">x</code></td>
<td>
<p><code>IBCF object</code> IBCF object, result of use the IBCF() function</p>
</td></tr>
<tr><td><code id="plot.IBCF_+3A_select">select</code></td>
<td>
<p><code>character</code> By default ('Pearson'), plot the Pearson Correlations of the IBCF Object, else ('MAAPE'), plot the MAAPE of the IBCF Object.</p>
</td></tr>
<tr><td><code id="plot.IBCF_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>

<hr>
<h2 id='print.IBCF'>Print IBCF information object</h2><span id='topic+print.IBCF'></span>

<h3>Description</h3>

<p>Print IBCF information object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'IBCF'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.IBCF_+3A_x">x</code></td>
<td>
<p>IBCF object</p>
</td></tr>
<tr><td><code id="print.IBCF_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>printeable object
</p>

<hr>
<h2 id='print.IBCFY'>Print IBCFY information object</h2><span id='topic+print.IBCFY'></span>

<h3>Description</h3>

<p>Print IBCFY information object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'IBCFY'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.IBCFY_+3A_x">x</code></td>
<td>
<p>IBCFY object</p>
</td></tr>
<tr><td><code id="print.IBCFY_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>printeable object
</p>

<hr>
<h2 id='summary.IBCF'>Summary</h2><span id='topic+summary.IBCF'></span>

<h3>Description</h3>

<p>Summary of IBCF object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'IBCF'
summary(object, information = "compact", digits = 4,
  ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.IBCF_+3A_object">object</code></td>
<td>
<p><code>IBCF object</code> IBCF object, result of use the IBCF() function</p>
</td></tr>
<tr><td><code id="summary.IBCF_+3A_information">information</code></td>
<td>
<p><code>string</code> ...</p>
</td></tr>
<tr><td><code id="summary.IBCF_+3A_digits">digits</code></td>
<td>
<p><code>numeric</code> ...</p>
</td></tr>
<tr><td><code id="summary.IBCF_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>

<hr>
<h2 id='summary.IBCFY'>Summary</h2><span id='topic+summary.IBCFY'></span>

<h3>Description</h3>

<p>Summary of IBCFY object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'IBCFY'
summary(object, digits = 4, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.IBCFY_+3A_object">object</code></td>
<td>
<p><code>IBCFY object</code> IBCFY object, result of use the IBCF.Years() function</p>
</td></tr>
<tr><td><code id="summary.IBCFY_+3A_digits">digits</code></td>
<td>
<p><code>numeric</code> Number of digits of the output.</p>
</td></tr>
<tr><td><code id="summary.IBCFY_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>

<hr>
<h2 id='Wheat_IBCF'>Wheat Data</h2><span id='topic+Wheat_IBCF'></span>

<h3>Description</h3>

<p>The package includes a data set based on a portion of the data used in the study of
Montesinos-Lopez, O. A.; Montesinos-Lopez, A.; Crossa, J.; Toledo, F. H.; Montesinos-Lopez, J. C.; Singh, P. &amp; Salinas-Ruiz, J. (2017). A Bayesian Poisson-lognormal Model for Count Data for Multiple-Trait Multiple-Environment Genomic-Enabled Prediction. G3: Genes|Genomes|Genetics 7(5):1595â1606. http://doi.org/10.1534/g3.117.039974.
The data set consists of 250 wheat lines evaluated in 3 environments and 4 distinct traits, i.e. 3000 observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Wheat_IBCF)
</code></pre>


<h3>Format</h3>

<p>a <code>TidyData</code> format, 3000 row per 4 columns.</p>


<h3>Author(s)</h3>

<p>Montesinos-Lopez, O. A.
</p>

<hr>
<h2 id='Year_IBCF'>Year_IBCF Data</h2><span id='topic+Year_IBCF'></span>

<h3>Description</h3>

<p>Dataset based on simulated data with the next code:
</p>
<pre>
set.seed(2)
A &lt;- matrix(0.65,ncol=12,nrow=12)
diag(A) &lt;- 1
Sdv &lt;- diag(c(0.9^0.5,0.8^0.5,0.9^0.5,0.8^0.5,0.86^0.5,0.7^0.5,0.9^0.5,0.8^0.5,0.9^0.5,0.7^0.5,0.7^0.5,0.85^0.5))

Sigma &lt;- Sdv
No.Lines &lt;- 60
Z &lt;- rmvnorm(No.Lines,mean=c(5,5.5,6,5.5,7,6.5,6.0,7,6.6,8,6.3,8),sigma=Sigma)
Years &lt;- c(rep(2014,20),rep(2015,20),rep(2016,20))
Gids &lt;- c(1:No.Lines)

Data.Final &lt;- data.frame(cbind(Years,Gids,Z))

colnames(Data.Final) &lt;- c("Years","Gids","T1","T2","T3","T4","T5","T6","T7","T8","T9","T10","T11","T12")
head(Data.Final)
Year_IBCF &lt;- getTidyForm(Data.Final, onlyTrait = T)
</pre>


<h3>Usage</h3>

<pre><code class='language-R'>data(Year_IBCF)
</code></pre>


<h3>Format</h3>

<p>a <code>TidyData</code> format, 750 row per 4 columns.</p>


<h3>Author(s)</h3>

<p>Montesinos-Lopez, O. A.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
