<!DOCTYPE html><html><head><title>Help for package IBDsim</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {IBDsim}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alleleSummary'>
<p>Allele sharing summary</p></a></li>
<li><a href='#DecodeMap'>
<p>Decode recombination map</p></a></li>
<li><a href='#dominant1'>
<p>Autosomal dominant pedigree</p></a></li>
<li><a href='#IBDsim'>
<p>IBD simulation</p></a></li>
<li><a href='#oneLocusIBD'>
<p>Estimating pairwise IBD coefficients</p></a></li>
<li><a href='#oneLocusJacquard'>
<p>Estimating Jacquard's condensed identity coefficients</p></a></li>
<li><a href='#twoLocusIBD'>
<p>Estimating two-locus IBD coefficients</p></a></li>
<li><a href='#twoLocusJacquard'>
<p>Estimating two-locus Jacquard coefficients</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulation of Chromosomal Regions Shared by Family Members</td>
</tr>
<tr>
<td>Version:</td>
<td>0.9-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-04-6</td>
</tr>
<tr>
<td>Author:</td>
<td>Magnus Dehli Vigeland</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Magnus Dehli Vigeland &lt;m.d.vigeland@medisin.uio.no&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulation of segments shared identical-by-descent (IBD) by 
    pedigree members. Using sex specific recombination rates along the human
    genome (Kong et. al (2010) &lt;<a href="https://doi.org/10.1038%2Fnature09525">doi:10.1038/nature09525</a>&gt;), phased chromosomes
    are simulated for all pedigree members, either by unconditional gene 
    dropping or conditional on a specified IBD pattern. Additional functions 
    provide summaries and further analysis of the simulated genomes.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3), paramlink (&ge; 1.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>Yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-04-06 11:14:47 UTC; magnusdv</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-04-06 11:42:45 UTC</td>
</tr>
</table>
<hr>
<h2 id='alleleSummary'>
Allele sharing summary
</h2><span id='topic+alleleSummary'></span>

<h3>Description</h3>

<p>This function summarises the allele flow for selected individuals, for a single genome simulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alleleSummary(x, ids, ibd.status = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alleleSummary_+3A_x">x</code></td>
<td>

<p>A single genome simulation, i.e. a list of simulated chromosomes. Each chromosome is a list, with one entry for each individual. Each of these entries is a list of two matrices (one for each strand). The matrices have 2 columns (start position; allele) and one row for each segment unbroken by recombination.
</p>
</td></tr>
<tr><td><code id="alleleSummary_+3A_ids">ids</code></td>
<td>

<p>A vector of numerical IDs. If missing, all individuals are included.
</p>
</td></tr>
<tr><td><code id="alleleSummary_+3A_ibd.status">ibd.status</code></td>
<td>

<p>A logical, meaningful only if <code>length(ids)==2</code>. If TRUE the IBD status (number of alleles shared IBD, either 0 1 or 2) of each segment is computed, as well as the breakdown of their parental origin.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is useful for downstream analysis of simulations produced by <code><a href="#topic+IBDsim">IBDsim</a></code>.
</p>


<h3>Value</h3>

<p>A numerical matrix. Each row corresponds to a chromosomal segment. The first 4 columns describe the segment (chromosome, start, end, length), and are followed by two columns (paternal allele, maternal allele) for each of the selected individuals. If <code>ibd.status=TRUE</code> five more columns are added: <code>ibd</code>, <code>ibd_pp</code>, <code>ibd_pm</code>, <code>ibd_mp</code> and <code>ibd_mm</code>. The first of these indicate the IBD status (0, 1 or 2) in the segment, while the latter 4 give the parental breakdown of this number. For instance, <code>ibd_pm</code> is 1 if the _p_aternal allele of the first individual is IBD with the _m_aternal allele of the second individual, and 0 otherwise.
</p>


<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>See Also</h3>

<p><code><a href="#topic+IBDsim">IBDsim</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Sibling simulation (3 sims of chromosomes 1 and 2)
x = nuclearPed(2)
sim = IBDsim(x, sims=3, chromosomes=1:2)

alleleSummary(sim[[1]]) # First sim, summary of all individuals
alleleSummary(sim[[1]], ids=3:4) # Summary of the siblings
alleleSummary(sim[[1]], ids=3:4, ibd.status=TRUE) # IBD breakdown of the siblings

# Trivial example: Summary of the father.
# Being the first founder, his alleles are denoted 1 and 2 everywhere.
fa = alleleSummary(sim[[1]], ids=1) 
stopifnot(all(fa[,c('1p', '1m')]==c(1,1,2,2)))

</code></pre>

<hr>
<h2 id='DecodeMap'>
Decode recombination map
</h2><span id='topic+DecodeMap'></span>

<h3>Description</h3>

<p>A recombination map of the human genome, adapted from the dataset published in (Kong et al., 2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DecodeMap</code></pre>


<h3>Format</h3>

<p>List of 23 components (one for each chromosome pair). Each is a list of 2 matrices, containing the male and female recombination maps of the given chromosome respectively. The 23rd component has NULL as its first (male) element, as we assume no recombination between sex chromosomes in males.
The recombination map matrices have 2 columns, named &quot;Mb&quot; and &quot;cM&quot;. The &quot;Mb&quot; column has the base positions of the markers used by (Kong et. al, 2010), and &quot;cM&quot; the genetic distance from the left telomere.
</p>


<h3>Source</h3>

<p>Kong, A. et al. (October 2010) Fine scale recombination rate differences between sexes, populations and individuals. <em>Nature</em>, 467, 1099&ndash;1103. doi:10.1038/nature09525.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#the first entries of the male map of chromosome 1:
head(DecodeMap[[1]]$male)
</code></pre>

<hr>
<h2 id='dominant1'>
Autosomal dominant pedigree
</h2><span id='topic+dominant1'></span>

<h3>Description</h3>

<p>A three generation pedigree affected with an autosomal dominant disorder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dominant1</code></pre>


<h3>Format</h3>

<p>A data frame describing the pedigree in standard LINKAGE format (i.e. the columns are: Family ID, individual ID, father, mother, sex, affection status).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = linkdat(dominant1)
plot(x)
</code></pre>

<hr>
<h2 id='IBDsim'>
IBD simulation
</h2><span id='topic+IBDsim'></span>

<h3>Description</h3>

<p>This is the main function of the package. Gene dropping of chromosomes is simulated down the pedigree, either unconditionally or conditional on the 'condition' pattern if this is given. Regions showing the 'query' pattern are collected and summarized. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IBDsim(x, sims, query=NULL, condition=NULL, map="decode", 
       chromosomes=NULL, model="chi", merged=TRUE, simdata=NULL, 
       skip.recomb = "noninf_founders", seed=NULL, verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IBDsim_+3A_x">x</code></td>
<td>

<p>A pedigree in the form of a <code><a href="paramlink.html#topic+linkdat">linkdat</a></code> object.
</p>
</td></tr>
<tr><td><code id="IBDsim_+3A_sims">sims</code></td>
<td>

<p>The number of simulations.
</p>
</td></tr>
<tr><td><code id="IBDsim_+3A_query">query</code>, <code id="IBDsim_+3A_condition">condition</code></td>
<td>

<p>Single allele patterns (SAPs), described as lists with numerical entries named &quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;atleast1&quot;, &quot;atmost1&quot;.
</p>
</td></tr>
<tr><td><code id="IBDsim_+3A_map">map</code></td>
<td>

<p>The genetic map(s) to be used in the simulations: One of the character strings &quot;decode&quot;, &quot;uniform.sex.spec&quot;, &quot;uniform.sex.aver&quot;. (See Details.)
</p>
</td></tr>
<tr><td><code id="IBDsim_+3A_chromosomes">chromosomes</code></td>
<td>

<p>A numeric vector indicating chromosome numbers, or either of the words &quot;AUTOSOMAL&quot; or &quot;X&quot;. The default is 1:22, i.e. the human autosomes.
</p>
</td></tr>
<tr><td><code id="IBDsim_+3A_model">model</code></td>
<td>

<p>A character indicating the statistical model for recombination: Either &quot;chi&quot; (the default) or &quot;haldane&quot;. (See details.)
</p>
</td></tr>
<tr><td><code id="IBDsim_+3A_merged">merged</code></td>
<td>

<p>A logical, indicating if overlapping/adjacent regions should be merged or not.
</p>
</td></tr>
<tr><td><code id="IBDsim_+3A_simdata">simdata</code></td>
<td>

<p>Either NULL, in which case simulation is performed before collecting IBD regions, or an object containing simulation data (typically generated by a previous run of <code>IBDsim</code>). 
</p>
</td></tr>
<tr><td><code id="IBDsim_+3A_skip.recomb">skip.recomb</code></td>
<td>

<p>A numeric containing individuals whose meioses should be simulated without recombination (i.e. a random strand is passed on to each offspring). If NULL, nobody is skipped. The default value (the character &quot;noninf_founders&quot;) computes the set of pedigree founders that cannot be carriers of the alleles described in the <code>condition</code> and/or <code>query</code> SAPs. 
</p>
</td></tr>
<tr><td><code id="IBDsim_+3A_seed">seed</code></td>
<td>

<p>NULL, or an integer to be passed on to <code><a href="base.html#topic+set.seed">set.seed</a></code>).
</p>
</td></tr>
<tr><td><code id="IBDsim_+3A_verbose">verbose</code></td>
<td>

<p>A logical.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each simulation starts by unique alleles being distributed to the pedigree founders.  In each subsequent meiosis, homologue chromosomes are made to recombine according to a renewal process along the four-strand bundle, with chi square distributed waiting times.  (For comparison purposes, Haldane's Poisson model for recombination is also implemented.)
</p>
<p>Recombination rates are sex-dependent, and vary along each chromosome according to the recombination map specified by the <code>map</code> parameter. By default, the complete Decode map of the human autosome is used (see References).  If <code>map="uniform.sex.spec"</code>, the genetic chromosome <em>lengths</em> are as in the Decode map, but the recombination rate is kept constant along each chromosome.  If <code>map="uniform.sex.aver"</code>, sex averaged genetic chromosome lengths are used (and constant recombination rates along each chromosome). 
</p>
<p>IBD patterns are described as combinations of Single Allele Patterns (SAPs).  A SAP is a specification for a given allele of the number of copies carried by various individuals, and must be given as a list of numerical vectors (containing ID labels) named '0', '1', '2', 'atleast1' and 'atmost1' (some of these can be absent or NULL; see Examples).
</p>
<p>If a condition SAP is given (i.e. if <code>condition</code> is non-null), simulation of each complete chromosome set (all autosomes by default) is performed as follows:  A 'disease chromosome' is sampled at random (using the physical chromosome lengths as weights), followed by a random 'disease locus' on this chromosome.  For this chromosome, gene dropping down the pedigree is carried out in such a way that the 'disease locus' has the condition SAP.  (In a bit more detail: First, the program computes all possible sets of obligate carriers, with suitable weights, and samples one of these.  Included in the obligate carriers will be exactly one founder, one of whose alleles is taken as the 'disease allele'.  In each meiosis involving obligate carriers, recombination is performed as usual, but the strand carrying the 'disease allele' is always the one passed on.) For the other chromosomes, simulation is done unconditionally. 
</p>


<h3>Value</h3>

<p>If the <code>query</code> is NULL, no identification of IBD segments is done, and only the simulated genomes are (invisibly) returned.  
</p>
<p>If <code>query</code> is non-null, the segments with the query SAP are identified, and a list of three elements is returned. These are the 'simdata' (the simulated genomes), the 'segments' (a list of matrices describing all identified regions) and finally 'stats' (a matrix with one column per simulation, summarizing the regions from that genome).
A summary is printed on the screen, with some or all of the following slots: 
</p>
<table>
<tr><td><code>count.all</code></td>
<td>

<p>The average count of all IBD segments (i.e. counting both random regions and the disease region in case of conditional simulation).
</p>
</td></tr>
<tr><td><code>fraction.all</code></td>
<td>

<p>The average fraction (in %) of the genome covered by IBD segments.
</p>
</td></tr>
<tr><td><code>average.all</code></td>
<td>

<p>The average length (in Mb) of IBD segments.
</p>
</td></tr>
<tr><td><code>longest.all</code></td>
<td>

<p>The average length (in Mb) of the longest IBD segment from each simulated genome.
</p>
</td></tr>
<tr><td><code>count.rand</code></td>
<td>

<p>The average count of random IBD segments.
</p>
</td></tr>
<tr><td><code>fraction.rand</code></td>
<td>

<p>The average fraction (in %) of the chromosomes covered by random IBD segments.
</p>
</td></tr>
<tr><td><code>average.rand</code></td>
<td>

<p>The average length (in Mb) of random IBD segments.
</p>
</td></tr>
<tr><td><code>longest.rand</code></td>
<td>

<p>The average length (in Mb) of the longest random IBD segment from each simulated genome.
</p>
</td></tr>
<tr><td><code>length.dis</code></td>
<td>

<p>The average length of the disease segment (only with conditional simulation).
</p>
</td></tr>
<tr><td><code>rank.dis</code></td>
<td>

<p>The average rank of the disease segment length among all the segments.
</p>
</td></tr>
<tr><td><code>zeroprob</code></td>
<td>

<p>The percentage of simulations resulting in zero IBD segments.
</p>
</td></tr>
</table>
<p>The term 'IBD segment' in the above always refers to 'IBD segment matching the query SAP'. 
The suffixes <code>.dis</code>, <code>.rand</code> and <code>.all</code> points to respectively the 
<em>disease</em> IBD segments (only relevant in conditional simulations), the <em>random</em>, i.e. non-disease, IBD segments (only relevant in conditional simulations), 
and <em>all</em> IBD segments (in any simulation). 
</p>


<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>References</h3>

<p>The Decode map:
</p>
<p>Kong, A. et al. (2010) Fine scale recombination rate differences between sexes, populations and individuals. <em>Nature</em>, 467, 1099&ndash;1103. doi:10.1038/nature09525.
</p>
<p>The chi-square model:
</p>
<p>Zhao, H., Speed, T. P., McPeek, M. S. (1995) Statistical analysis of crossover interference using the chi-square model. <em>Genetics</em>, 139(2), 1045&ndash;1056.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># In all examples below, the 'sims' parameter is set to 1 to 
# decrease runtime. For realistic results increase to e.g. 1000.

#### An example with a recessive disease in a consanguineous pedigree:

x = linkdat(twoloops)
plot(x)

# If individuals 15, 16 and 17 are available for sequencing, we can 
# predict the number and size of disease-compatible IBD segments as 
# follows:

sap1 = list('2'=15:16, 'atmost1'=17)
IBDsim(x, sims=1, condition=sap1, query=sap1)

# If 16 is unavailable, his parents and healthy sibling are still 
# informative. The regions we are looking for now are those with 
# an allele present in 2 copies in 15, 1 copy in 12 and 14, and 
# at most one in 17. Note that the condition SAP remains as above.

IBDsim(x, sims=1, condition=sap1, 
       query=list('2'=15, '1'=c(12,14), 'atmost1'=17))

	   
#### Example with an autosomal dominant disorder:
y = linkdat(dominant1) # lazy load data
plot(y)

# Suppose a 20 Mb linkage peak is found. 
# How often would this occur by chance?

aff = which(dominant1[,'AFF']==2)   # the affected
nonaff = which(dominant1[,'AFF']==1)   # the non-affected
dom_pattern = list('1'=aff, '0'=nonaff)
res = IBDsim(y, sims=1, query=dom_pattern) 
mean(res$stats['longest.all',] &gt; 20)   # the estimated p-value


#### Another example: Unconditional simulation of regions shared 
# IBD by third cousins. The "zeroprob" entry in the output shows 
# the percentage of simulations having no IBD-sharing among the 
# two cousins. (Again: Increase 'sims' for more accurate results.)

x_male = cousinPed(3)
plot(x_male)
IBDsim(x_male, sims=1, query=list('1'=15:16))

# Changing the genders of the individuals connecting the cousins 
# can have a big impact on the distribution of IBD segments:

x_female = swapSex(x_male, c(3,4,7,8,11,12))
plot(x_female)
IBDsim(x_female, sims=1, query=list('1'=15:16))

## Given that the two third cousins have at least one segment in 
# common, what is the expected length of this segment? We simulate 
# conditional on one allele in common between the cousins. The 
# "length.dis" entry of the summary shows the average length of 
# the disease region (which should be quite a lot larger than 
# an average random segment).

sap = list('1'=15:16)
IBDsim(x_male, sims=1, condition=sap, query=sap)


#### Let us look at a different relationship: Half first cousins. 
# Two such cousins will on average share 1/8 = 12.5% of the autosome.  

z = halfCousinPed(1)
plot(z)
res = IBDsim(z, sims=1, query=list('1'=8:9)) 
res$stats

# visualizing the spread in total IBD sharing and the number of segments:

IBD_percent = res$stats['fraction.all', ]
IBD_count = res$stats['count.all', ]
hist(IBD_percent) 
hist(IBD_count)
</code></pre>

<hr>
<h2 id='oneLocusIBD'>
Estimating pairwise IBD coefficients
</h2><span id='topic+oneLocusIBD'></span>

<h3>Description</h3>

<p>Estimates by simulation the IBD coefficients of a non-inbred pairwise relationship. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oneLocusIBD(x, ind1, ind2, Nsim, Xchrom=FALSE, verbose=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oneLocusIBD_+3A_x">x</code></td>
<td>

<p>A pedigree in the form of a <code><a href="paramlink.html#topic+linkdat">linkdat</a></code> object.
</p>
</td></tr>
<tr><td><code id="oneLocusIBD_+3A_ind1">ind1</code>, <code id="oneLocusIBD_+3A_ind2">ind2</code></td>
<td>

<p>Numeric ID labels of the two individuals.
</p>
</td></tr>
<tr><td><code id="oneLocusIBD_+3A_nsim">Nsim</code></td>
<td>

<p>The number of simulations to be performed.
</p>
</td></tr>
<tr><td><code id="oneLocusIBD_+3A_xchrom">Xchrom</code></td>
<td>

<p>A logical indicating if the locus is X-linked (if TRUE) or autosomal (FALSE).
</p>
</td></tr>
<tr><td><code id="oneLocusIBD_+3A_verbose">verbose</code></td>
<td>

<p>A logical.
</p>
</td></tr>
<tr><td><code id="oneLocusIBD_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed on to <code><a href="#topic+IBDsim">IBDsim</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For any pair of non-inbred individuals, the IBD coefficients <code class="reqn">\kappa=(\kappa[0], \kappa[1], \kappa[2])</code> associated with the relationship, are defined as the probabilities
</p>
<p style="text-align: center;"><code class="reqn">\kappa[i] = Pr(i alleles shared identically by descent).</code>
</p>

<p>For an X-chromosomal locus, and if at least one of the individuals is male, <code class="reqn">\kappa[i]</code> is defined only for <code class="reqn">i=0,1</code>.
</p>


<h3>Value</h3>

<p>A numeric of length 3 (autosomal) or 2 (X-linked), estimating <code class="reqn">\kappa</code>.
</p>


<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>See Also</h3>

<p><code><a href="#topic+twoLocusIBD">twoLocusIBD</a></code>, <code><a href="#topic+oneLocusJacquard">oneLocusJacquard</a></code>, <code><a href="#topic+twoLocusJacquard">twoLocusJacquard</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example 1: Full siblings
x &lt;- nuclearPed(2) 
Nsim &lt;- 100 # Should be increased substantially

# Autosomal kappa estimate (exact = c(0.25, 0.5, 0.25))
oneLocusIBD(x, ind1=3, ind2=4, Nsim=Nsim)

# X-chromosomal kappa estimate (exact = c(0.5, 0.5))
oneLocusIBD(x, ind1=3, ind2=4, Nsim=Nsim, Xchrom=TRUE)

</code></pre>

<hr>
<h2 id='oneLocusJacquard'>
Estimating Jacquard's condensed identity coefficients
</h2><span id='topic+oneLocusJacquard'></span>

<h3>Description</h3>

<p>Estimates by simulation Jacquard's 9 condensed identity coefficients for a pairwise relationship. This function is rarely needed, as exact values can be obtained by using <code><a href="paramlink.html#topic+jacquard">jacquard</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oneLocusJacquard(x, ind1, ind2, Nsim, verbose=TRUE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oneLocusJacquard_+3A_x">x</code></td>
<td>

<p>A pedigree in the form of a <code><a href="paramlink.html#topic+linkdat">linkdat</a></code> object.
</p>
</td></tr>
<tr><td><code id="oneLocusJacquard_+3A_ind1">ind1</code>, <code id="oneLocusJacquard_+3A_ind2">ind2</code></td>
<td>

<p>Numeric ID labels of the two individuals.
</p>
</td></tr>
<tr><td><code id="oneLocusJacquard_+3A_nsim">Nsim</code></td>
<td>

<p>The number of simulations to be performed.
</p>
</td></tr>
<tr><td><code id="oneLocusJacquard_+3A_verbose">verbose</code></td>
<td>

<p>A logical.
</p>
</td></tr>
<tr><td><code id="oneLocusJacquard_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed on to <code><a href="#topic+IBDsim">IBDsim</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the definition and further details about these coefficients, see Jacquard (1970).
</p>


<h3>Value</h3>

<p>A numeric of length 9, estimating the condensed Jacquard identity coefficients <code class="reqn">\Delta</code>.
</p>


<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>See Also</h3>

<p><code><a href="paramlink.html#topic+jacquard">jacquard</a></code>, <code><a href="#topic+oneLocusIBD">oneLocusIBD</a></code>, <code><a href="#topic+twoLocusIBD">twoLocusIBD</a></code>, <code><a href="#topic+twoLocusJacquard">twoLocusJacquard</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Siblings whose parents are full siblings.
x = fullSibMating(generations=2)
Nsim = 100 # (increase to improve accuracy) 

# Estimating the 9 identity coefficients
j_est = oneLocusJacquard(x, ind1=5, ind2=6, Nsim=Nsim)

# Exact: c(2,1,4,1,4,1,7,10,2)/32

# With the "identity" package:
## Not run: 
  j_exact = jacquard(x, 5:6)

## End(Not run)
</code></pre>

<hr>
<h2 id='twoLocusIBD'>
Estimating two-locus IBD coefficients
</h2><span id='topic+twoLocusIBD'></span>

<h3>Description</h3>

<p>Estimates by simulation the two-locus IBD coefficients of a non-inbred pairwise relationship.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twoLocusIBD(x, ind1, ind2, rho=NULL, cM=NULL, Nsim, Xchrom=FALSE, verbose=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="twoLocusIBD_+3A_x">x</code></td>
<td>

<p>A pedigree in the form of a <code><a href="paramlink.html#topic+linkdat">linkdat</a></code> object.
</p>
</td></tr>
<tr><td><code id="twoLocusIBD_+3A_ind1">ind1</code>, <code id="twoLocusIBD_+3A_ind2">ind2</code></td>
<td>

<p>Numeric ID labels of the two individuals.
</p>
</td></tr>
<tr><td><code id="twoLocusIBD_+3A_rho">rho</code></td>
<td>

<p>NULL, or a number in the interval [0, 0.5]: the recombination fraction between the two loci. If non-NULL, it is converted to centiMorgan using Haldanes map function: <code>cM = -50*log(1-2*rho)</code>.
</p>
</td></tr>
<tr><td><code id="twoLocusIBD_+3A_cm">cM</code></td>
<td>

<p>NULL, or a non-negative number: the distance in centiMorgan between the two loci.
</p>
</td></tr>
<tr><td><code id="twoLocusIBD_+3A_nsim">Nsim</code></td>
<td>

<p>The number of simulations to be performed.
</p>
</td></tr>
<tr><td><code id="twoLocusIBD_+3A_xchrom">Xchrom</code></td>
<td>

<p>A logical indicating if the loci are X-linked (if TRUE) or autosomal (FALSE).
</p>
</td></tr>
<tr><td><code id="twoLocusIBD_+3A_verbose">verbose</code></td>
<td>

<p>A logical.
</p>
</td></tr>
<tr><td><code id="twoLocusIBD_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed on to <code><a href="#topic+IBDsim">IBDsim</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a given pair of autosomal loci, the two-locus IBD coefficients of a non-inbred pairwise relationship are defined by
</p>
<p style="text-align: center;"><code class="reqn">\kappa[i,j] = Pr(sharing i alleles IBD at locus 1, and j alleles IBD at locus 2),</code>
</p>

<p>where <code class="reqn">0 \le i,j \le 2</code>. For X-chromosomal loci, and if at least one of the individuals is male, <code class="reqn">\kappa[i,j]</code> is defined only for <code class="reqn">0 \le i,j \le 1</code>.
</p>
<p>While the single-locus IBD coefficients <code class="reqn">\kappa=(\kappa[0], \kappa[1], \kappa[2])</code> (see <code><a href="#topic+oneLocusIBD">oneLocusIBD</a></code>) depend only on the genealogy relating the two individuals, the two-locus coefficients also depend on the genetic distance between the loci. In particular, if the loci are completely linked (rho=0; cM=0) we have <code class="reqn">\kappa[i,j] = \kappa[i]</code> if i=j and 0 otherwise. If the loci are unlinked (rho=0.5; cM=Inf) we have <code class="reqn">\kappa[i,j] = \kappa[i]*\kappa[j]</code>. (See examples.)
</p>


<h3>Value</h3>

<p>A numerical matrix, where the entry in row <code class="reqn">a</code> and column <code class="reqn">b</code> is the estimate of <code class="reqn">\kappa[a-1, b-1]</code> defined above. 
</p>


<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>See Also</h3>

<p><code><a href="#topic+oneLocusIBD">oneLocusIBD</a></code>, <code><a href="#topic+oneLocusJacquard">oneLocusJacquard</a></code>, <code><a href="#topic+twoLocusJacquard">twoLocusJacquard</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example 1: Full siblings
x &lt;- nuclearPed(2) 
Nsim &lt;- 100 # Should be increased substantially

# One-locus kappa estimates (autosomal and X):
k_est = oneLocusIBD(x, ind1=3, ind2=4, Nsim=Nsim, seed=123)
k_est_X = oneLocusIBD(x, ind1=3, ind2=4, Nsim=Nsim, Xchrom=TRUE, seed=123)

### Two-locus IBD estimation
# Completely linked, autosomal
rho = 0 
k2_linked = twoLocusIBD(x, ind1=3, ind2=4, rho=rho, Nsim=Nsim, seed=123)
stopifnot(identical(diag(k2_linked), k_est))

# Completely unlinked, autosomal
rho = 0.5
k2_unlinked = twoLocusIBD(x, ind1=3, ind2=4, rho=rho, Nsim=Nsim, seed=123)
stopifnot(identical(k2_unlinked, outer(k_est, k_est)))

# Recombination rate 10%, autosomal
rho &lt;- 0.1 
cM &lt;- -50*log(1-2*rho)
r1 &lt;- twoLocusIBD(x, ind1=3, ind2=4, rho=rho, Nsim=Nsim, seed=17)
r2 &lt;- twoLocusIBD(x, ind1=3, ind2=4, cM=cM, Nsim=Nsim, seed=17)
stopifnot(identical(r1, r2))

### Two-locus IBD on X
# Completely linked, X chromosome
rho = 0 
k2_linked_X = twoLocusIBD(x, ind1=3, ind2=4, rho=rho, Nsim=Nsim, Xchrom=TRUE, seed=123)
stopifnot(identical(diag(k2_linked_X), k_est_X))

# Completely unlinked, X chromosome
rho = 0.5
k2_unlinked_X = twoLocusIBD(x, ind1=3, ind2=4, rho=rho, Nsim=Nsim, Xchrom=TRUE, seed=123)
stopifnot(identical(k2_unlinked_X, outer(k_est_X, k_est_X)))

# Recombination rate 10%, X chromosome
rho &lt;- 0.1 
cM &lt;- -50*log(1-2*rho)
r1_X &lt;- twoLocusIBD(x, ind1=3, ind2=4, rho=rho, Nsim=Nsim, Xchrom=TRUE, seed=123)
r2_X &lt;- twoLocusIBD(x, ind1=3, ind2=4, cM=cM, Nsim=Nsim, Xchrom=TRUE, seed=123)
stopifnot(identical(r1_X, r2_X))


### Example 2: Comparing half sib, grandparent and uncle
# These are indistinguishable with unlinked loci, see e.g. 
# pages 182-183 in Egeland, Kling and Mostad (2016).
# Each simulations followed by exact counterpart.
x &lt;- addSon(addSon(nuclearPed(2, sex=1:2), 4), 5)
plot(x)
rho &lt;- 0.25
Nsim &lt;- 10 # Should be increased to at least 10000
twoLocusIBD(x, 1, 6, rho=rho, Nsim=Nsim, verbose=FALSE)[2,2]; .5*(1-rho)
twoLocusIBD(x, 8, 6, rho=rho, Nsim=Nsim, verbose=FALSE)[2,2]; .5*(rho^2+(1-rho)^2)
twoLocusIBD(x, 3, 6, rho=rho, Nsim=Nsim, verbose=FALSE)[2,2]; .5*((1-rho)*(rho^2+(1-rho)^2) + rho/2)

### Example 3: X chromosome, granddaughter vs maternal grandfather.
y &lt;- addDaughter(nuclearPed(1, sex=2), 3)
plot(y)
rho &lt;- 0.25
Nsim &lt;-10
twoLocusIBD(y, 1, 5, rho=rho, Nsim=Nsim, Xchrom=TRUE)

# Exact
matrix(c(1-rho, rho, rho, 1-rho)/2, ncol=2) 

</code></pre>

<hr>
<h2 id='twoLocusJacquard'>
Estimating two-locus Jacquard coefficients
</h2><span id='topic+twoLocusJacquard'></span>

<h3>Description</h3>

<p>Estimates by simulation the two-locus version of Jacquard's condensed identity coefficients for a pairwise relationship.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twoLocusJacquard(x, ind1, ind2, rho=NULL, cM=NULL, Nsim, verbose=TRUE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="twoLocusJacquard_+3A_x">x</code></td>
<td>

<p>A pedigree in the form of a <code><a href="paramlink.html#topic+linkdat">linkdat</a></code> object.
</p>
</td></tr>
<tr><td><code id="twoLocusJacquard_+3A_ind1">ind1</code>, <code id="twoLocusJacquard_+3A_ind2">ind2</code></td>
<td>

<p>Numeric ID labels of the two individuals.
</p>
</td></tr>
<tr><td><code id="twoLocusJacquard_+3A_rho">rho</code></td>
<td>

<p>NULL, or a number in the interval [0, 0.5]: the recombination fraction between the two loci. If non-NULL, it is converted to centiMorgan using Haldanes map function: <code>cM = -50*log(1-2*rho)</code>.
</p>
</td></tr>
<tr><td><code id="twoLocusJacquard_+3A_cm">cM</code></td>
<td>

<p>NULL, or a non-negative number: the distance in centiMorgan between the two loci. The numeric <code>Inf</code> is allowed, and corresponds to unlinked loci.
</p>
</td></tr>
<tr><td><code id="twoLocusJacquard_+3A_nsim">Nsim</code></td>
<td>

<p>The number of simulations to be performed.
</p>
</td></tr>
<tr><td><code id="twoLocusJacquard_+3A_verbose">verbose</code></td>
<td>

<p>A logical.
</p>
</td></tr>
<tr><td><code id="twoLocusJacquard_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed on to <code><a href="#topic+IBDsim">IBDsim</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As in the case of IBD coefficients (see <code><a href="#topic+twoLocusIBD">twoLocusIBD</a></code>), we can generalise Jacquard's identity coefficients to two loci: Given any pair of individuals, and any pair of autosomal loci, we define the two-locus Jacquard coefficient <code class="reqn">\Delta[i,j]</code> (where <code class="reqn">1 \le i,j \le 9</code>) by
</p>
<p style="text-align: center;"><code class="reqn">\Delta[i,j] = Pr(Jacquard state \Sigma[i] at locus 1, and state \Sigma[j] at locus 2).</code>
</p>



<h3>Value</h3>

<p>A numerical 9*9 matrix. The entry in row <code class="reqn">a</code> and column <code class="reqn">b</code> is the estimate of <code class="reqn">\Delta[a-1, b-1]</code> defined above. 
</p>


<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>See Also</h3>

<p><code><a href="paramlink.html#topic+jacquard">jacquard</a></code>, <code><a href="#topic+oneLocusIBD">oneLocusIBD</a></code>, <code><a href="#topic+twoLocusIBD">twoLocusIBD</a></code>, <code><a href="#topic+oneLocusJacquard">oneLocusJacquard</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Siblings whose parents are full siblings.
x = fullSibMating(generations=2)
Nsim = 100 # (increase to improve accuracy) 

# Estimate of the 9 identity coefficients
j_est = oneLocusJacquard(x, ind1=5, ind2=6, Nsim=Nsim, seed=123)

### Two-locus Jacquard coefficients
# Completely linked loci
rho = 0 
j2_linked = twoLocusJacquard(x, ind1=5, ind2=6, rho=rho, Nsim=Nsim, seed=123)
stopifnot(identical(diag(j2_linked), j_est))

# Completely unlinked
rho = 0.5
j2_unlinked = twoLocusJacquard(x, ind1=5, ind2=6, rho=rho, Nsim=Nsim, seed=123)
stopifnot(identical(j2_unlinked, outer(j_est, j_est)))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
