<!DOCTYPE html><html lang="en"><head><title>Help for package iC10</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {iC10}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#iC10-package'>
<p>A Copy Number and Expression-Based Classifier for Breast Tumours</p></a></li>
<li><a href='#compare'>
<p>Compare results of the iC10 classifier</p></a></li>
<li><a href='#getCNfeatures'>
<p>Internal function for matching copy number features.</p></a></li>
<li><a href='#getExpfeatures'>
<p>Internal function for matching expression features.</p></a></li>
<li><a href='#goodnessOfFit'>
<p>Goodness of fit results of the iC10 classifier</p></a></li>
<li><a href='#iC10'><p>A copy number and expression-based classfier for breast cancers</p></a></li>
<li><a href='#matchFeatures'>
<p>Matching features from the classifier to the test data.</p></a></li>
<li><a href='#normalizeFeatures'>
<p>Normalization of expression features</p></a></li>
<li><a href='#plot.iC10'>
<p>Plot results of the iC10 classifier</p></a></li>
<li><a href='#print.iC10'>
<p>Print results of the iC10 classifier</p></a></li>
<li><a href='#summary.iC10'>
<p>Summary results of the iC10 classifier</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Copy Number and Expression-Based Classifier for Breast Tumours</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-07-16</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Oscar M Rueda &lt;Oscar.Rueda@mrc-bsu.cam.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of the classifier described in the paper Ali HR et al (2014) &lt;<a href="https://doi.org/10.1186%2Fs13059-014-0431-1">doi:10.1186/s13059-014-0431-1</a>&gt;. It uses copy number and/or expression form breast cancer data, trains a Tibshirani's 'pamr' classifier with the features available and predicts the iC10 group.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>pamr, impute, iC10TrainingData</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-19 06:32:13 UTC; oscar</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-19 09:00:26 UTC</td>
</tr>
<tr>
<td>Author:</td>
<td>Oscar M Rueda <a href="https://orcid.org/0000-0003-0008-4884"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
</table>
<hr>
<h2 id='iC10-package'>
A Copy Number and Expression-Based Classifier for Breast Tumours
</h2><span id='topic+iC10-package'></span>

<h3>Description</h3>

<p>Implementation of the classifier described in the paper Ali HR et al (2014) &lt;doi:10.1186/s13059-014-0431-1&gt;. It uses copy number and/or expression form breast cancer data, trains a Tibshirani's 'pamr' classifier with the features available and predicts the iC10 group.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> iC10</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> A Copy Number and Expression-Based Classifier for Breast Tumours</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2024-07-16</td>
</tr>
<tr>
 <td style="text-align: left;">
Authors@R: </td><td style="text-align: left;"> 
	person("Oscar M", "Rueda", , "Oscar.Rueda@mrc-bsu.cam.ac.uk", role = c("aut", "cre"),  comment = c(ORCID = "0000-0003-0008-4884"))</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Oscar M Rueda &lt;Oscar.Rueda@mrc-bsu.cam.ac.uk&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Implementation of the classifier described in the paper Ali HR et al (2014) &lt;doi:10.1186/s13059-014-0431-1&gt;. It uses copy number and/or expression form breast cancer data, trains a Tibshirani's 'pamr' classifier with the features available and predicts the iC10 group.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> pamr,
impute,
iC10TrainingData</td>
</tr>
<tr>
 <td style="text-align: left;">
Packaged: </td><td style="text-align: left;"> 2015-09-23 02:37:41 UTC; rueda01</td>
</tr>
<tr>
 <td style="text-align: left;">
NeedsCompilation: </td><td style="text-align: left;"> no</td>
</tr>
<tr>
 <td style="text-align: left;">
Repository: </td><td style="text-align: left;"> CRAN</td>
</tr>
<tr>
 <td style="text-align: left;">
Date/Publication: </td><td style="text-align: left;"> 2015-09-23 08:05:08</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Oscar M Rueda [aut, cre] (&lt;https://orcid.org/0000-0003-0008-4884&gt;)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<p>Index of help topics:
</p>
<pre>
compare                 Compare results of the iC10 classifier
getCNfeatures           Internal function for matching copy number
                        features.
getExpfeatures          Internal function for matching expression
                        features.
goodnessOfFit           Goodness of fit results of the iC10 classifier
iC10                    A copy number and expression-based classfier
                        for breast cancers
iC10-package            A Copy Number and Expression-Based Classifier
                        for Breast Tumours
matchFeatures           Matching features from the classifier to the
                        test data.
normalizeFeatures       Normalization of expression features
plot.iC10               Plot results of the iC10 classifier
print.iC10              Print results of the iC10 classifier
summary.iC10            Summary results of the iC10 classifier
</pre>
<p>iC10 implements the classifier described in the paper 'Genome-driven integrated classification of breast cancer validated in over 7,500 samples' (Ali HR et al., Genome Biology 2014). It uses copy number and/or expression form breast cancer data, trains a pamr classifier (Tibshirani et al.) with the features available and predicts the iC10 group.
</p>


<h3>Author(s)</h3>

<p>NA
</p>
<p>Maintainer: Oscar M Rueda &lt;Oscar.Rueda@mrc-bsu.cam.ac.uk&gt;
</p>


<h3>References</h3>

<p>Ali HR et al. Genome-driven integrated classification of breast cancer
validated in over 7,500 samples. Genome Biology 2014; 15:431.
Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
Tibshirani et al. Diagnosis of multiple cancer types by shrunken
centroids of gene expression. PNAS 2002; 99(10):6567-6572.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(iC10TrainingData)
data(train.CN)
data(train.Exp)
features &lt;- matchFeatures(Exp=train.Exp, Exp.by.feat="probe")
features &lt;- normalizeFeatures(features, "scale")
res &lt;- iC10(features)
summary(res)
goodnessOfFit(res, newdata=features)
compare(res, iC10=1:2, newdata=features)
compare(res, iC10=2:4, newdata=features)
</code></pre>

<hr>
<h2 id='compare'>
Compare results of the iC10 classifier
</h2><span id='topic+compare'></span><span id='topic+compare.iC10'></span>

<h3>Description</h3>

<p>This function plots the centroids of the training set versus the average profiles
of the new data classified in each group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare(obj, iC10=1:10, newdata, name.test="Test",...)
## S3 method for class 'iC10'
compare(obj, iC10=1:10, newdata, name.test="Test",...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compare_+3A_obj">obj</code></td>
<td>

<p>An object of class <code>iC10</code>, a result of a call to <code>iC10()</code>
</p>
</td></tr>
<tr><td><code id="compare_+3A_ic10">iC10</code></td>
<td>

<p>Groups to plot
</p>
</td></tr>
<tr><td><code id="compare_+3A_newdata">newdata</code></td>
<td>

<p>Set of features of the new data to compare. They must be the same samples classified and contained in
<code>x</code>. A result of a call to <code>matchFeatures()</code> or <code>normalizeFeatures()</code>
</p>
</td></tr>
<tr><td><code id="compare_+3A_name.test">name.test</code></td>
<td>

<p>Name of the new data set to appear in the text of the plot
</p>
</td></tr>
<tr><td><code id="compare_+3A_...">...</code></td>
<td>

<p>Additional arguments passed to <code>plot()</code>
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot is returned with two plots per groups requested.
</p>


<h3>Author(s)</h3>

<p>Oscar M. Rueda
</p>


<h3>References</h3>

<p>Ali HR et al. Genome-driven integrated classification of breast cancer
validated in over 7,500 samples. Genome Biology 2014; 15:431.
Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+iC10">iC10</a></code>, <code><a href="#topic+plot.iC10">plot.iC10</a></code>, <code><a href="#topic+matchFeatures">matchFeatures</a></code>, <code><a href="#topic+normalizeFeatures">normalizeFeatures</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(iC10TrainingData)
data(train.CN)
data(train.Exp)
features &lt;- matchFeatures(Exp=train.Exp, Exp.by.feat="probe")
features &lt;- normalizeFeatures(features, "scale")
res &lt;- iC10(features)
compare(res, 1:3, newdata=features)
</code></pre>

<hr>
<h2 id='getCNfeatures'>
Internal function for matching copy number features.
</h2><span id='topic+getCNfeatures'></span>

<h3>Description</h3>

<p>This function should not be called directly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCNfeatures(CN, Probes, Map, by.feat, ref, Synonyms)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCNfeatures_+3A_cn">CN</code></td>
<td>

<p>CN features matrix
</p>
</td></tr>
<tr><td><code id="getCNfeatures_+3A_probes">Probes</code></td>
<td>

<p>Vector with the probes to match
</p>
</td></tr>
<tr><td><code id="getCNfeatures_+3A_map">Map</code></td>
<td>

<p>data.frame with the genomic description of the features to match
</p>
</td></tr>
<tr><td><code id="getCNfeatures_+3A_by.feat">by.feat</code></td>
<td>

<p>&quot;probe&quot; or &quot;gene&quot;, indicating if match should be done by probe position or gene name.
</p>
</td></tr>
<tr><td><code id="getCNfeatures_+3A_ref">ref</code></td>
<td>

<p>hg18 or hg19 (only relevant if matching is done by probe position).
</p>
</td></tr>
<tr><td><code id="getCNfeatures_+3A_synonyms">Synonyms</code></td>
<td>

<p>data.frame with available synonym gene names to match (only relevant if matching is done by gene name).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with the copy number features
</p>


<h3>Author(s)</h3>

<p>Oscar M Rueda
</p>

<hr>
<h2 id='getExpfeatures'>
Internal function for matching expression features.
</h2><span id='topic+getExpfeatures'></span>

<h3>Description</h3>

<p>Internal function for matching expression features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExpfeatures(Exp, Probes, Synonyms, by.feat)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getExpfeatures_+3A_exp">Exp</code></td>
<td>

<p>Matrix of expression features
</p>
</td></tr>
<tr><td><code id="getExpfeatures_+3A_probes">Probes</code></td>
<td>

<p>Vector of probes to match
</p>
</td></tr>
<tr><td><code id="getExpfeatures_+3A_synonyms">Synonyms</code></td>
<td>

<p>vector of synonyms fo gene names
</p>
</td></tr>
<tr><td><code id="getExpfeatures_+3A_by.feat">by.feat</code></td>
<td>

<p>either &quot;probe&quot; or &quot;gene&quot;
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with the <code>Probes</code> in Exp.
</p>


<h3>Note</h3>

<p>This function is not supposed to be called directly. use <code>matchFeatures</code> instead.
</p>


<h3>Author(s)</h3>

<p>Oscar M Rueda
</p>


<h3>References</h3>

<p>Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+matchFeatures">matchFeatures</a></code>
</p>

<hr>
<h2 id='goodnessOfFit'>
Goodness of fit results of the iC10 classifier
</h2><span id='topic+goodnessOfFit'></span><span id='topic+goodnessOfFit.iC10'></span>

<h3>Description</h3>

<p>Goodness of fit results of the iC10 classifier: this function computes correlations between the signatures
of the training dataset and the classified features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>goodnessOfFit(obj, iC10=1:10, newdata=NULL,...)
## S3 method for class 'iC10'
goodnessOfFit(obj, iC10=1:10, newdata=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="goodnessOfFit_+3A_obj">obj</code></td>
<td>

<p>An object of <code>iC10</code> class.
</p>
</td></tr>
<tr><td><code id="goodnessOfFit_+3A_ic10">iC10</code></td>
<td>

<p>Groups to compute goodness of fit.
</p>
</td></tr>
<tr><td><code id="goodnessOfFit_+3A_newdata">newdata</code></td>
<td>

<p>The feature data to compute the goodness of fit. Must be the samples classified in <code>obj</code>.
It can be a call to <code>matchFeatures</code> or <code>normalizeFeatures</code>. If NULL, <code>obj$fitted</code> is
used.
</p>
</td></tr>
<tr><td><code id="goodnessOfFit_+3A_...">...</code></td>
<td>

<p>Additional arguments passed to <code>cor</code> (like <code>method</code>;  Default is pearson)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It prints the correlation for each iC10.
</p>


<h3>Author(s)</h3>

<p>Oscar M Rueda
</p>


<h3>References</h3>

<p>Ali HR et al. Genome-driven integrated classification of breast cancer
validated in over 7,500 samples. Genome Biology 2014; 15:431.
Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>See Also</h3>

<p><code>iC10</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(iC10TrainingData)
data(train.CN)
data(train.Exp)
features &lt;- matchFeatures(Exp=train.Exp, Exp.by.feat="probe")
features &lt;- normalizeFeatures(features, "scale")
res &lt;- iC10(features)
goodnessOfFit(res, newdata=features)
</code></pre>

<hr>
<h2 id='iC10'>A copy number and expression-based classfier for breast cancers</h2><span id='topic+iC10'></span>

<h3>Description</h3>

<p>iC10 implements the classifier described in the paper 'Genome-driven integrated classification of breast cancer validated in over 7,500 samples' (Ali HR et al., Genome Biology 2014). It uses copy number and/or expression form breast cancer data, trains a pamr classifier (Tibshirani et al.) with the features available and predicts the iC10 group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iC10(x, seed=25435)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iC10_+3A_x">x</code></td>
<td>
<p>An object with class <code>iC10features</code>: A list with elements 'train.CN', 'train.Exp', 'train.iC10',
'CN', 'Exp', 'map.cn', 'map.exp'</p>
</td></tr>
<tr><td><code id="iC10_+3A_seed">seed</code></td>
<td>
<p>seed to initialize random number generator. It is passed to <code>set.seed()</code>.
See details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function trains a <code>pamr</code> classifier and predicts the set of
samples. The shrinkage parameter is obtained with crossvalidation,
therefore different runs can give different results (unless a seed is specified).
</p>


<h3>Value</h3>

<p>An object of class <code>iC10</code>. A list with the following elements:
</p>
<table role = "presentation">
<tr><td><code>class</code></td>
<td>
<p>Prediction classes for the samples</p>
</td></tr>
<tr><td><code>posterior</code></td>
<td>
<p>Probablitites for each sample to belong to each of
the 10 groups</p>
</td></tr>
<tr><td><code>centroids</code></td>
<td>
<p>Shrunken Centroids for each of the 10 groups.</p>
</td></tr>
<tr><td><code>fitted</code></td>
<td>
<p>Normalized features for the samples classified.</p>
</td></tr>
<tr><td><code>map.cn</code></td>
<td>
<p>Annotation data for the copy number features</p>
</td></tr>
<tr><td><code>map.exp</code></td>
<td>
<p>Annotation data for the expression features</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Oscar M. Rueda
</p>


<h3>References</h3>

<p>Ali HR et al. Genome-driven integrated classification of breast cancer
validated in over 7,500 samples. Genome Biology 2014; 15:431.
Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
Tibshirani et al. Diagnosis of multiple cancer types by shrunken
centroids of gene expression. PNAS 2002; 99(10):6567-6572.
</p>


<h3>See Also</h3>

<p>See <code>pamr.train</code>, <code>pamr.cv</code> and <code>pamr.predict</code> in package <code>pamr</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(iC10TrainingData)
data(train.CN)
data(train.Exp)
features &lt;- matchFeatures(Exp=train.Exp, Exp.by.feat="probe")
features &lt;- normalizeFeatures(features, "scale")
res &lt;- iC10(features)

</code></pre>

<hr>
<h2 id='matchFeatures'>
Matching features from the classifier to the test data.
</h2><span id='topic+matchFeatures'></span>

<h3>Description</h3>

<p>This function matches available copy number and/or expression data features to the training signatures;
using either genomic position or HUGO gene name for copy number features and either Illumina
probe names or HUGO gene name for expression features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matchFeatures(CN = NULL, Exp = NULL,
CN.by.feat = c("gene", "probe"),
Exp.by.feat = c("gene", "probe"),
ref="hg19")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="matchFeatures_+3A_cn">CN</code></td>
<td>

<p>Data must be log2 copy number ratios. Two formats are allowed:
- a matrix where each row represents a gene and each column a sample.
In this case <code>CN.by.feat</code> must be &quot;gene&quot; and the rownames must be the hgnc gene names.
- a data.frame with segmented data. The following columns must exist: 'ID' for the sample name,
'chromosome_name' for the chromosome (must be numeric), 'loc.start' for the start position of the region,
'loc.end' for the end position of the region, 'seg.mean' for the log2ratio of the segment.
If NULL, copy number is not used in the classifier.
</p>
</td></tr>
<tr><td><code id="matchFeatures_+3A_exp">Exp</code></td>
<td>

<p>Matrix with the expression data to classify. Each row must be a gene or an Illumina probe, and each
column must correspond to a sample.
Rownames must be either Illumina probes, in which case <code>Exp.by.feat</code> must be &quot;probe&quot;;  or hgnc gene names,
in which case <code>Exp.by.feat</code> must be &quot;gene&quot;.
If NULL, expression is not used in the classifier.
</p>
</td></tr>
<tr><td><code id="matchFeatures_+3A_cn.by.feat">CN.by.feat</code></td>
<td>

<p>Either &quot;probe&quot; or &quot;gene&quot;, Default is &quot;probe&quot;.
</p>
</td></tr>
<tr><td><code id="matchFeatures_+3A_exp.by.feat">Exp.by.feat</code></td>
<td>

<p>Either &quot;probe&quot; or &quot;gene&quot;, Default is &quot;gene&quot;.
</p>
</td></tr>
<tr><td><code id="matchFeatures_+3A_ref">ref</code></td>
<td>

<p>Either &quot;hg18&quot;, &quot;hg19&quot; or &quot;hg38&quot;. It is used to match the copy number probes if <code>CN.by.feat</code> is &quot;probe&quot;
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>One of <code>CN</code> or <code>Exp</code> must be not NULL.
If matching is done by gene, hgnc gene name is used to match the rownames of the features. A list of
synonym gene names is used (see <code>Map.All</code>).
For copy number features matched by probe, the maximum log ratio in absolute value inside the limits
of the feature is used. If there is no copy number in that region, the value of the probe before it is used.
</p>


<h3>Value</h3>

<p>A list with the following elements is returned:
</p>
<table role = "presentation">
<tr><td><code>CN</code></td>
<td>
<p>copy number data to classify</p>
</td></tr>
<tr><td><code>train.CN</code></td>
<td>
<p>copy number training data</p>
</td></tr>
<tr><td><code>Exp</code></td>
<td>
<p>expression data to classify</p>
</td></tr>
<tr><td><code>train.Exp</code></td>
<td>
<p>expression training data</p>
</td></tr>
<tr><td><code>train.iC10</code></td>
<td>
<p>iC10 assignments for the training data</p>
</td></tr>
<tr><td><code>map.cn</code></td>
<td>
<p>annotation data for the copy number features</p>
</td></tr>
<tr><td><code>map.exp</code></td>
<td>
<p>annotation data for the expression features</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Note that the training set will be different, depending on the features
matched.
Genomic annotation for the training dataset has been obtained from Mark
Dunning's lluminaHumanv3.db package.
</p>


<h3>Author(s)</h3>

<p>Oscar M Rueda
</p>


<h3>References</h3>

<p>Ali HR et al. Genome-driven integrated classification of breast cancer
validated in over 7,500 samples. Genome Biology 2014; 15:431.
Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+normalizeFeatures">normalizeFeatures</a></code>, <code><a href="#topic+iC10">iC10</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(iC10TrainingData)
data(train.CN)
data(train.Exp)
features &lt;- matchFeatures(Exp=train.Exp,Exp.by.feat="probe", ref="hg18")
str(features)
</code></pre>

<hr>
<h2 id='normalizeFeatures'>
Normalization of expression features
</h2><span id='topic+normalizeFeatures'></span>

<h3>Description</h3>

<p>Normalization of expression features. Several methods available in the package <code>CONOR</code> can be used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalizeFeatures(x, method=c("none", "scale"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="normalizeFeatures_+3A_x">x</code></td>
<td>

<p>An object result of a call to <code>matchFeatures</code>
</p>
</td></tr>
<tr><td><code id="normalizeFeatures_+3A_method">method</code></td>
<td>

<p>Several methods are available:
&quot;none&quot;: No normalization is done
&quot;scale&quot;: Each expression feature is scaled to have zero mean and standard deviation 1
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>No further normalization is needed on the copy number, as log2 ratios
are comparable between platforms.
</p>


<h3>Value</h3>

<p>A list of the same format as <code>matchFeatures</code>, but with <code>train.Exp</code> anfd <code>Exp</code> normalized.
</p>


<h3>Note</h3>

<p>As <code>CONOR</code> package is no longer maintained, the methods are not available temporarily. We will include more normalization methods in the next version of this package.
</p>


<h3>Author(s)</h3>

<p>Oscar M Rueda
</p>


<h3>References</h3>

<p>Ali HR et al. Genome-driven integrated classification of breast cancer
validated in over 7,500 samples. Genome Biology 2014; 15:431.
Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(iC10TrainingData)
data(train.CN)
data(train.Exp)
features &lt;- matchFeatures(Exp=train.Exp,
Exp.by.feat="probe", ref="hg18")
features &lt;- normalizeFeatures(features, "scale")
</code></pre>

<hr>
<h2 id='plot.iC10'>
Plot results of the iC10 classifier
</h2><span id='topic+plot.iC10'></span>

<h3>Description</h3>

<p>Plot results of the iC10 classifier, in two different formats: either the signatures of the training
set or the signatures of the new data classified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'iC10'
plot(x, sample.name=1, newdata = NULL,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.iC10_+3A_x">x</code></td>
<td>

<p>An object of <code>iC10</code> class:
</p>
</td></tr>
<tr><td><code id="plot.iC10_+3A_sample.name">sample.name</code></td>
<td>

<p>Number of sample to plot (if <code>newdata</code> is NULL). It can be either a number or a character with the
sample name.
</p>
</td></tr>
<tr><td><code id="plot.iC10_+3A_newdata">newdata</code></td>
<td>

<p>An object result to call to <code>matchFeatures</code> or <code>normalizeFeatures</code> containing the features of the samples
to plot.
</p>
</td></tr>
<tr><td><code id="plot.iC10_+3A_...">...</code></td>
<td>

<p>Additional arguments passed to <code>plot</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two types of plots can be produced. If <code>newdata</code> is NULL, a panel 6x2 is drawn with the 10 profiles
of the signatures of the training set and the profile of the features of <code>sample.name</code> and the
distribution of the probabilities of classification to each iC10 for that sample.
If <code>newdata</code> is not nutll, a panel 6x2 (with the 11th panel empty) is drawn with the 10 profiles of
<code>newdata</code> samples and their distribution into the clusters.
The features are sorted by type: copy number (if available) are drawn in grey, and then expression, each
of them are sorted by genomic position.
</p>


<h3>Value</h3>

<p>A 6x2 plot is produced.
</p>


<h3>Author(s)</h3>

<p>Oscar M Rueda
</p>


<h3>References</h3>

<p>Ali HR et al. Genome-driven integrated classification of breast cancer
validated in over 7,500 samples. Genome Biology 2014; 15:431.
Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>See Also</h3>

<p><code>iC10</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(iC10TrainingData)
data(train.CN)
data(train.Exp)
features &lt;- matchFeatures(Exp=train.Exp, Exp.by.feat="probe")
features &lt;- normalizeFeatures(features, "scale")
res &lt;- iC10(features)
plot(res, sample.name=10)
plot(res, newdata=features)
</code></pre>

<hr>
<h2 id='print.iC10'>
Print results of the iC10 classifier
</h2><span id='topic+print.iC10'></span>

<h3>Description</h3>

<p>Print results of the iC10 classifier
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'iC10'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.iC10_+3A_x">x</code></td>
<td>

<p>An object of <code>iC10</code> class:
</p>
</td></tr>
<tr><td><code id="print.iC10_+3A_...">...</code></td>
<td>

<p>Additional arguments passed to <code>print</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a call to <code>str</code>.
</p>


<h3>Author(s)</h3>

<p>Oscar M Rueda
</p>


<h3>References</h3>

<p>Ali HR et al. Genome-driven integrated classification of breast cancer
validated in over 7,500 samples. Genome Biology 2014; 15:431.
Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>See Also</h3>

<p><code>iC10</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(iC10TrainingData)
data(train.CN)
data(train.Exp)
features &lt;- matchFeatures(Exp=train.Exp, Exp.by.feat="probe")
features &lt;- normalizeFeatures(features, "scale")
res &lt;- iC10(features)
res
</code></pre>

<hr>
<h2 id='summary.iC10'>
Summary results of the iC10 classifier
</h2><span id='topic+summary.iC10'></span>

<h3>Description</h3>

<p>Summary results of the iC10 classifier: shows the distribution of samples classified into each iC10 group and
a summary of the maximum posterior probablity for each sample. Small values pinpoint samples with no clear
group assigned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'iC10'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.iC10_+3A_object">object</code></td>
<td>

<p>An object of <code>iC10</code> class.
</p>
</td></tr>
<tr><td><code id="summary.iC10_+3A_...">...</code></td>
<td>

<p>Additional arguments passed to <code>summary</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function prints a table of the classification ad a summary of the maximum posterior probability for
each sample.
</p>


<h3>Author(s)</h3>

<p>Oscar M Rueda
</p>


<h3>References</h3>

<p>Ali HR et al. Genome-driven integrated classification of breast cancer
validated in over 7,500 samples. Genome Biology 2014; 15:431.
Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
Tibshirani et al. Diagnosis of multiple cancer types by shrunken
centroids of gene expression. PNAS 2002; 99(10):6567-6572.
</p>


<h3>See Also</h3>

<p>See <code>iC10</code> and <code>pamr.train</code>, <code>pamr.cv</code> and <code>pamr.predict</code> in package <code>pamr</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(iC10TrainingData)
data(train.CN)
data(train.Exp)
features &lt;- matchFeatures(Exp=train.Exp,
Exp.by.feat="probe", ref="hg18")
features &lt;- normalizeFeatures(features, "scale")
res &lt;- iC10(features)
summary(res)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
