<!DOCTYPE html><html><head><title>Help for package iC10TrainingData</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {iC10TrainingData}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#iC10TrainingData-package'>
<p>Training Datasets for iC10 Package</p></a></li>
<li><a href='#IntClustMemb'>
<p>Class Membership for the training set</p></a></li>
<li><a href='#Map.All'>
<p>Probe mapping of the complete set of features of the training set</p></a></li>
<li><a href='#Map.CN'>
<p>Probe mapping of the copy number features of the training set.</p></a></li>
<li><a href='#Map.Exp'>
<p>Probe mapping of the Expression features of the training set</p></a></li>
<li><a href='#train.CN'>
<p>Copy number data for the training set</p></a></li>
<li><a href='#train.Exp'>
<p>Expression data for the training set.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Training Datasets for iC10 Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-08-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Oscar M Rueda and Jose Antonio Seoane Fernandez</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Oscar M. Rueda &lt;Oscar.Rueda@cruk.cam.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Training datasets for iC10; which implements the classifier described in the paper 'Genome-driven integrated classification of breast cancer validated in over 7,500 samples' (Ali HR et al., Genome Biology 2014). It uses copy number and/or expression form breast cancer data, trains a pamr classifier (Tibshirani et al.) with the features available and predicts the iC10 group. Genomic annotation for the training dataset has been obtained from Mark Dunning's lluminaHumanv3.db package.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-08-20 15:18:40 UTC; rueda01</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-08-24 14:02:15 UTC</td>
</tr>
</table>
<hr>
<h2 id='iC10TrainingData-package'>
Training Datasets for iC10 Package
</h2><span id='topic+iC10TrainingData-package'></span><span id='topic+iC10TrainingData'></span>

<h3>Description</h3>

<p>Training datasets for iC10; which implements the classifier described in the paper 'Genome-driven integrated classification of breast cancer validated in over 7,500 samples' (Ali HR et al., Genome Biology 2014). It uses copy number and/or expression form breast cancer data, trains a pamr classifier (Tibshirani et al.) with the features available and predicts the iC10 group. Genomic annotation for the training dataset has been obtained from Mark Dunning's lluminaHumanv3.db package.
Training datasets for iC10; which implements the classifier described in the METABRIC 
paper 'The genomic and transcriptomic architecture of 2,000 breast 
tumours reveals novel subgroups' (Curtis et al., Nature 2012). 
It uses copy number and/or expression form breast cancer data, 
trains a pamr classifier (Tibshirani et al.) with the features available and predicts the iC10 group.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> iC10TrainingData</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Training Datasets for iC10 Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.3.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2018-08-20</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Oscar M Rueda and Jose Antonio Seoane Fernandez</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Oscar M. Rueda &lt;Oscar.Rueda@cruk.cam.ac.uk&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Training datasets for iC10; which implements the classifier described in the paper 'Genome-driven integrated classification of breast cancer validated in over 7,500 samples' (Ali HR et al., Genome Biology 2014). It uses copy number and/or expression form breast cancer data, trains a pamr classifier (Tibshirani et al.) with the features available and predicts the iC10 group. Genomic annotation for the training dataset has been obtained from Mark Dunning's lluminaHumanv3.db package.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
Packaged: </td><td style="text-align: left;"> 2014-09-15 12:22:07 UTC; rueda01</td>
</tr>
<tr>
 <td style="text-align: left;">
NeedsCompilation: </td><td style="text-align: left;"> no</td>
</tr>
<tr>
 <td style="text-align: left;">
Repository: </td><td style="text-align: left;"> CRAN</td>
</tr>
<tr>
 <td style="text-align: left;">
Date/Publication: </td><td style="text-align: left;"> 2014-09-26 12:13:01</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Index of help topics:
</p>
<pre>
IntClustMemb            Class Membership for the training set
Map.All                 Probe mapping of the complete set of features
                        of the training set
Map.CN                  Probe mapping of the copy number features of
                        the training set.
Map.Exp                 Probe mapping of the Expression features of the
                        training set
iC10TrainingData-package
                        Training Datasets for iC10 Package
train.CN                Copy number data for the training set
train.Exp               Expression data for the training set.
</pre>
<p>Training datasets for iC10; which implements the classifier described in the METABRIC 
paper 'The genomic and transcriptomic architecture of 2,000 breast 
tumours reveals novel subgroups' (Curtis et al., Nature 2012). 
It uses copy number and/or expression form breast cancer data, 
trains a pamr classifier (Tibshirani et al.) with the features available and predicts the iC10 group.
</p>


<h3>Author(s)</h3>

<p>Oscar M Rueda and Jose Antonio Seoane Fernandez
</p>
<p>Maintainer: Oscar M. Rueda &lt;Oscar.Rueda@cruk.cam.ac.uk&gt;
</p>


<h3>References</h3>

<p>Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
Tibshirani et al. Diagnosis of multiple cancer types by shrunken
centroids of gene expression. PNAS 2002; 99(10):6567-6572.
</p>


<h3>See Also</h3>

<p><code>iC10</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(train.CN)
data(train.Exp)
</code></pre>

<hr>
<h2 id='IntClustMemb'>
Class Membership for the training set
</h2><span id='topic+IntClustMemb'></span>

<h3>Description</h3>

<p>iC10 assignment for the Metabric training dataset (997 samples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(IntClustMemb)</code></pre>


<h3>Format</h3>

<p>The format is:
Factor w/ 10 levels &quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;,..: 2 9 3 3 8 6 7 7 7 3 ...
- attr(*, &quot;names&quot;)= chr [1:997] &quot;MB.0135&quot; &quot;MB.0167&quot; &quot;MB.0136&quot; &quot;MB.3403&quot; ...
</p>


<h3>Source</h3>

<p>Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(IntClustMemb)
barplot(table(IntClustMemb))
</code></pre>

<hr>
<h2 id='Map.All'>
Probe mapping of the complete set of features of the training set
</h2><span id='topic+Map.All'></span>

<h3>Description</h3>

<p>Probe mapping of the complete set of features of the training set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Map.All)</code></pre>


<h3>Format</h3>

<p>A data frame with 714 observations on the following 10 variables:
</p>

<dl>
<dt><code>Probe_ID</code></dt><dd><p>a character vector with the Illumina probe ids that flank the features</p>
</dd>
<dt><code>Gene_symbol</code></dt><dd><p>a factor with the hugo gene names</p>
</dd>
<dt><code>Ensembl_ID</code></dt><dd><p>a factor with the ensemble ids</p>
</dd>
<dt><code>Cytoband</code></dt><dd><p>a factor with the cytobands (on hg18)</p>
</dd>
<dt><code>Genomic_location_hg18</code></dt><dd><p>a factor with the genomic locations on hg18</p>
</dd>
<dt><code>chromosome_name_hg18</code></dt><dd><p>a numeric vector with the chromosome on hg18</p>
</dd>
<dt><code>start_position_hg18</code></dt><dd><p>a numeric vector with the start position on hg18</p>
</dd>
<dt><code>end_position_hg18</code></dt><dd><p>a numeric vector with the end position on hg18</p>
</dd>
<dt><code>Synonyms_0</code></dt><dd><p>a character vector with the gene name synonyms of the feature</p>
</dd>
<dt><code>Gene.Chosen</code></dt><dd><p>a character vector (YES or NO) specifiying the probe chosen for gene-based selection</p>
</dd>
<dt><code>Genomic_location_hg19</code></dt><dd><p>a factor with the genomic locations on hg19</p>
</dd>
<dt><code>chromosome_name_hg19</code></dt><dd><p>a numeric vector with the chromosome on hg19</p>
</dd>
<dt><code>start_position_hg19</code></dt><dd><p>a numeric vector with the start position on hg19</p>
</dd>
<dt><code>end_position_hg19</code></dt><dd><p>a numeric vector with the end position on hg19</p>
</dd>
</dl>



<h3>Source</h3>

<p>Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Map.All)
head(Map.All)
</code></pre>

<hr>
<h2 id='Map.CN'>
Probe mapping of the copy number features of the training set.
</h2><span id='topic+Map.CN'></span>

<h3>Description</h3>

<p>Probe mapping of the copy number features of the training set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Map.CN)</code></pre>


<h3>Format</h3>

<p>A data frame with 38 observations on the following 8 variables.
</p>

<dl>
<dt><code>Probe_ID</code></dt><dd><p>a character vector with the Illumina probe ids that flank the features</p>
</dd>
<dt><code>Gene_symbol</code></dt><dd><p>a factor with the hugo gene names</p>
</dd>
<dt><code>Ensembl_ID</code></dt><dd><p>a factor with the ensemble ids</p>
</dd>
<dt><code>Cytoband</code></dt><dd><p>a factor with the cytobands (on hg18)</p>
</dd>
<dt><code>Genomic_location_hg18</code></dt><dd><p>a factor with the genomic locations on hg18</p>
</dd>
<dt><code>chromosome_name_hg18</code></dt><dd><p>a numeric vector with the chromosome on hg18</p>
</dd>
<dt><code>start_position_hg18</code></dt><dd><p>a numeric vector with the start position on hg18</p>
</dd>
<dt><code>end_position_hg18</code></dt><dd><p>a numeric vector with the end position on hg18</p>
</dd>
<dt><code>Genomic_location_hg19</code></dt><dd><p>a factor with the genomic locations on hg19</p>
</dd>
<dt><code>chromosome_name_hg19</code></dt><dd><p>a numeric vector with the chromosome on hg19</p>
</dd>
<dt><code>start_position_hg19</code></dt><dd><p>a numeric vector with the start position on hg19</p>
</dd>
<dt><code>end_position_hg19</code></dt><dd><p>a numeric vector with the end position on hg19</p>
</dd>
</dl>



<h3>Source</h3>

<p>Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Map.CN)
head(Map.CN)
</code></pre>

<hr>
<h2 id='Map.Exp'>
Probe mapping of the Expression features of the training set
</h2><span id='topic+Map.Exp'></span>

<h3>Description</h3>

<p>Probe mapping of the Expression features of the training set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Map.Exp)</code></pre>


<h3>Format</h3>

<p>A data frame with 711 observations on the following 10 variables.
</p>

<dl>
<dt><code>Probe_ID</code></dt><dd><p>a character vector with the Illumina probe ids that flank the features</p>
</dd>
<dt><code>Gene_symbol</code></dt><dd><p>a factor with the hugo gene names</p>
</dd>
<dt><code>Ensembl_ID</code></dt><dd><p>a factor with the ensemble ids</p>
</dd>
<dt><code>Cytoband</code></dt><dd><p>a factor with the cytobands (on hg18)</p>
</dd>
<dt><code>Genomic_location_hg18</code></dt><dd><p>a factor with the genomic locations on hg18</p>
</dd>
<dt><code>chromosome_name_hg18</code></dt><dd><p>a numeric vector with the chromosome on hg18</p>
</dd>
<dt><code>start_position_hg18</code></dt><dd><p>a numeric vector with the start position on hg18</p>
</dd>
<dt><code>end_position_hg18</code></dt><dd><p>a numeric vector with the end position on hg18</p>
</dd>
<dt><code>Synonyms_0</code></dt><dd><p>a character vector with the gene name synonyms of the feature</p>
</dd>
<dt><code>Gene.Chosen</code></dt><dd><p>a character vector (YES or NO) specifiying the probe chosen for gene-based selection</p>
</dd>
<dt><code>Genomic_location_hg19</code></dt><dd><p>a factor with the genomic locations on hg19</p>
</dd>
<dt><code>chromosome_name_hg19</code></dt><dd><p>a numeric vector with the chromosome on hg19</p>
</dd>
<dt><code>start_position_hg19</code></dt><dd><p>a numeric vector with the start position on hg19</p>
</dd>
<dt><code>end_position_hg19</code></dt><dd><p>a numeric vector with the end position on hg19</p>
</dd>
</dl>



<h3>Source</h3>

<p>Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Map.Exp)
head(Map.Exp)
</code></pre>

<hr>
<h2 id='train.CN'>
Copy number data for the training set
</h2><span id='topic+train.CN'></span>

<h3>Description</h3>

<p>Copy number data for the training set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(train.CN)</code></pre>


<h3>Format</h3>

<p>A matrix with 714 rows and 997 columns. Rows are features and columns are training samples.
</p>


<h3>Details</h3>

<p>Each row corresponds to one copy number feature for all samples in the training set. Note that it includes 
all features in the classifier. Note also that, depending on the data available and the type of matching 
(gene or probe) only some of the features will be used.
</p>


<h3>Source</h3>

<p>Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(train.CN)
summary(train.CN)
</code></pre>

<hr>
<h2 id='train.Exp'>
Expression data for the training set.
</h2><span id='topic+train.Exp'></span>

<h3>Description</h3>

<p>Expression data for the training set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(train.Exp)</code></pre>


<h3>Format</h3>

<p>A matrix with 714 rows and 997 columns. Rows are features and columns are training samples.
</p>


<h3>Details</h3>

<p>Each row corresponds to one expression feature for all samples in the training set. Note that it includes 
all features in the classifier. Note that, depending on the data available and the type of matching 
(gene or probe) only some of the features will be used.
</p>


<h3>Source</h3>

<p>Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(train.Exp)
summary(train.Exp)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
